diff --git a/_Headers/SFall/define_lite.h b/_Headers/ReDefine/define_lite.h
index a0226d3..d4e4164 100644
--- a/_Headers/SFall/define_lite.h
+++ b/_Headers/ReDefine/define_lite.h
@@ -164,9 +164,7 @@
 // ----------------------------------------------------------------------- //
 
 // Trait defines //
-#define TRAIT_PERK   (0)
-#define TRAIT_OBJECT (1)
-#define TRAIT_TRAIT  (2)
+// ReDefine : TRAIT_PERK, TRAIT_OBJECT, TRAIT_TRAIT : moved to define_trait.h
 
 // Perk Trait Types
 #define PERK_bonus_awareness      (0)
@@ -607,7 +605,7 @@
 #define drug_influence(WHO)      metarule(METARULE_WHO_ON_DRUGS, WHO)
 #define METARULE_MAP_KNOWN     19
 #define map_is_known(X)          metarule(METARULE_MAP_KNOWN, X)
-//#define DEFINE_METARULE_IS_LOADGAME  22
+//#define METARULE_IS_LOADGAME  22
 #define METARULE_CAR_CURRENT_TOWN  30
 #define METARULE_GIVE_CAR_TO_PARTY 31
 #define METARULE_GIVE_CAR_GAS      32
@@ -658,10 +656,10 @@
 #define KILL_TYPE_gecko_kills         15
 #define KILL_TYPE_alien_kills         16
 #define KILL_TYPE_giant_ant_kills     17
-#define DEFINE_METARULE_SET_CAR_CARRY_AMOUNT 52
-#define set_car_carry_amount(VALUE)   metarule(DEFINE_METARULE_SET_CAR_CARRY_AMOUNT, VALUE)
-#define DEFINE_METARULE_GET_CAR_CARRY_AMOUNT 53
-#define car_carry_amount   metarule(DEFINE_METARULE_GET_CAR_CARRY_AMOUNT, 0)
+#define METARULE_SET_CAR_CARRY_AMOUNT 52
+#define set_car_carry_amount(VALUE)   metarule(METARULE_SET_CAR_CARRY_AMOUNT, VALUE)
+#define METARULE_GET_CAR_CARRY_AMOUNT 53
+#define car_carry_amount   metarule(METARULE_GET_CAR_CARRY_AMOUNT, 0)
 // Script MetaRule3's
 #define METARULE3_CLR_FIXED_TIMED_EVENTS  100
 #define rm_fixed_timer_event(WHO,FIXED_VAL) metarule3(METARULE3_CLR_FIXED_TIMED_EVENTS,WHO,FIXED_VAL,0)
