diff --git a/Mods/DemoInFO2/Original/DEMOGEN.ssl b/Mods/DemoInFO2/ReDefine/DEMOGEN.ssl
index d16d9e3..1d3c66f 100644
--- a/Mods/DemoInFO2/Original/DEMOGEN.ssl
+++ b/Mods/DemoInFO2/ReDefine/DEMOGEN.ssl
@@ -21,10 +21,10 @@ procedure description_p_proc
 begin
     script_overrides;
     if (map_var(2)) then begin
-        display_msg(message_str(1314, 100));
+        display_mstr(100);
     end
     else begin
-        display_msg(message_str(1314, 101));
+        display_mstr(101);
     end
 end
 
@@ -32,10 +32,10 @@ procedure look_at_p_proc
 begin
     script_overrides;
     if (map_var(2)) then begin
-        display_msg(message_str(1314, 100));
+        display_mstr(100);
     end
     else begin
-        display_msg(message_str(1314, 101));
+        display_mstr(101);
     end
 end
 
@@ -50,10 +50,10 @@ begin
         set_map_var(3, map_var(3) + 1);
     end
     if (not(map_var(2))) then begin
-        display_msg(message_str(1314, 118));
+        display_mstr(118);
     end
     else begin
-        display_msg(message_str(1314, 119));
+        display_mstr(119);
     end
 end
 
@@ -63,14 +63,14 @@ begin
     if ((obj_pid(obj_being_used_with) == 75) and not(map_var(2))) then begin
         set_map_var(2, 1);
         give_exp_points(250);
-        display_msg(message_str(1314, 102));
+        display_mstr(102);
     end
     else begin
         if (obj_being_used_with == 75) then begin
-            display_msg(message_str(1314, 103));
+            display_mstr(103);
         end
         else begin
-            display_msg(message_str(1314, 104));
+            display_mstr(104);
         end
     end
 end
@@ -84,15 +84,15 @@ begin
     if ((action_being_used == 12) or (action_being_used == 13)) then begin
         if ((action_being_used == 12) and not(map_var(2))) then begin
             if (local_var(4)) then begin
-                display_msg(message_str(1314, 105));
+                display_mstr(105);
             end
             else begin
                 if (is_success(roll_vs_skill(dude_obj, 12, 30))) then begin
-                    set_global_var(702, 1);
-                    display_msg(message_str(1314, 106));
+                    set_global_var(GVAR_GENERATOR_3, 1);
+                    display_mstr(106);
                 end
                 else begin
-                    display_msg(message_str(1314, 107));
+                    display_mstr(107);
                 end
             end
             set_local_var(4, local_var(4) + 1);
@@ -100,14 +100,14 @@ begin
         else begin
             if (action_being_used == 12) then begin
                 if (local_var(6)) then begin
-                    display_msg(message_str(1314, 108));
+                    display_mstr(108);
                 end
                 else begin
                     if (is_success(roll_vs_skill(dude_obj, 12, 0))) then begin
-                        display_msg(message_str(1314, 109));
+                        display_mstr(109);
                     end
                     else begin
-                        display_msg(message_str(1314, 110));
+                        display_mstr(110);
                     end
                 end
                 set_local_var(6, local_var(6) + 1);
@@ -115,27 +115,27 @@ begin
         end
         if ((action_being_used == 13) and not(map_var(2))) then begin
             if (local_var(5)) then begin
-                display_msg(message_str(1314, 111));
+                display_mstr(111);
             end
             if (is_success(roll_vs_skill(dude_obj, 13, 0))) then begin
-                display_msg(message_str(1314, 112));
+                display_mstr(112);
             end
             else begin
-                display_msg(message_str(1314, 113));
+                display_mstr(113);
             end
             set_local_var(4, local_var(4) + 1);
         end
         else begin
             if (action_being_used == 13) then begin
                 if (local_var(5)) then begin
-                    display_msg(message_str(1314, 116));
+                    display_mstr(116);
                 end
                 else begin
                     if (is_success(roll_vs_skill(dude_obj, 13, 0))) then begin
-                        display_msg(message_str(1314, 114));
+                        display_mstr(114);
                     end
                     else begin
-                        display_msg(message_str(1314, 115));
+                        display_mstr(115);
                     end
                 end
                 add_timer_event(self_obj, game_ticks(5), 1);
@@ -144,7 +144,7 @@ begin
         end
     end
     else begin
-        display_msg(message_str(1314, 117));
+        display_mstr(117);
     end
 end
 
@@ -164,7 +164,6 @@ procedure timed_event_p_proc
 begin
     if (fixed_param == 1) then begin
         play_gmovie(7);
-        metarule(13, 0);
+        signal_end_game;
     end
 end
-
