diff --git a/Mods/DemoInFO2/Original/DEMODOOR.ssl b/Mods/DemoInFO2/ReDefine/DEMODOOR.ssl
index 46df40d..4a5474c 100644
--- a/Mods/DemoInFO2/Original/DEMODOOR.ssl
+++ b/Mods/DemoInFO2/ReDefine/DEMODOOR.ssl
@@ -23,14 +23,14 @@ procedure description_p_proc
 begin
     script_overrides;
     if (local_var(4) == 0) then begin
-        display_msg(message_str(1313, 102));
+        display_mstr(102);
     end
     else begin
         if (local_var(5)) then begin
-            display_msg(message_str(1313, 103));
+            display_mstr(103);
         end
         else begin
-            display_msg(message_str(1313, 104));
+            display_mstr(104);
         end
     end
 end
@@ -38,7 +38,7 @@ end
 procedure look_at_p_proc
 begin
     script_overrides;
-    display_msg(message_str(1313, 106));
+    display_mstr(106);
 end
 
 procedure damage_p_proc
@@ -61,7 +61,7 @@ begin
         else begin
             script_overrides;
             set_map_var(13, map_var(13) + local_var(6));
-            display_msg(message_str(1313, 105));
+            display_mstr(105);
         end
     end
     else begin
@@ -90,7 +90,7 @@ begin
         if (obj_is_open(self_obj) == 1) then begin
         end
         else begin
-            display_msg(message_str(1313, 108));
+            display_mstr(108);
         end
         add_timer_event(self_obj, 5, 1);
     end
@@ -100,21 +100,21 @@ begin
             if (local_var(4) == 0) then begin
                 test := roll_vs_skill(dude_obj, 9, 0);
                 if (is_success(test)) then begin
-                    display_msg(message_str(1313, 100));
+                    display_mstr(100);
                     set_local_var(4, 1);
                 end
                 else begin
-                    display_msg(message_str(1313, 101));
+                    display_mstr(101);
                 end
             end
             else begin
-                display_msg(message_str(1313, 107));
+                display_mstr(107);
             end
         end
         else begin
             script_overrides;
             set_map_var(13, map_var(13) + local_var(7));
-            display_msg(message_str(1313, 108));
+            display_mstr(108);
         end
     end
 end
@@ -137,4 +137,3 @@ begin
         set_map_var(13, 0);
     end
 end
-
