diff --git a/Mods/DemoInFO2/Original/DEMOPHIL.ssl b/Mods/DemoInFO2/ReDefine/DEMOPHIL.ssl
index a3b0fcc..f4d8291 100644
--- a/Mods/DemoInFO2/Original/DEMOPHIL.ssl
+++ b/Mods/DemoInFO2/ReDefine/DEMOPHIL.ssl
@@ -49,10 +49,10 @@ begin
             if (anim_busy(self_obj) == 0) then begin
                 if ((tile_distance_objs(self_obj, dude_obj) < 8) or obj_can_see_obj(self_obj, dude_obj)) then begin
                     if (anim_busy(self_obj) == 0) then begin
-                        global_temp := rotation_to_tile(tile_num(dude_obj), tile_num(self_obj));
+                        global_temp := rotation_to_tile(dude_tile, tile_num(self_obj));
                         animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(self_obj), global_temp, random(3, 10)), 1);
                         global_temp := (global_temp + 1) % 6;
-                        while ((anim_busy(self_obj) == 0) and (global_temp != rotation_to_tile(tile_num(dude_obj), tile_num(self_obj)))) do begin
+                        while ((anim_busy(self_obj) == 0) and (global_temp != rotation_to_tile(dude_tile, tile_num(self_obj)))) do begin
                             debug_msg("flee loop: rot == " + global_temp);
                             animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(self_obj), global_temp, random(3, 10)), 1);
                             global_temp := (global_temp + 1) % 6;
@@ -62,7 +62,7 @@ begin
             end
         end
         else begin
-            attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+            attack(dude_obj);
         end
     end
     if (growling) then begin
@@ -87,7 +87,7 @@ end
 procedure talk_p_proc
 begin
     call Goodbyes();
-    if (global_var(707)) then begin
+    if (global_var(GVAR_DOG_EMPATHY)) then begin
         call Node001();
     end
     else begin
@@ -95,7 +95,7 @@ begin
             call Node009();
         end
         else begin
-            if ((obj_item_subtype(critter_inven_obj(dude_obj, 1)) == 3) or (obj_item_subtype(critter_inven_obj(dude_obj, 2)) == 3)) then begin
+            if ((obj_item_subtype(critter_inven_obj(dude_obj, INVEN_TYPE_RIGHT_HAND)) == 3) or (obj_item_subtype(critter_inven_obj(dude_obj, INVEN_TYPE_LEFT_HAND)) == 3)) then begin
                 call Node000();
             end
             else begin
@@ -124,59 +124,59 @@ end
 procedure destroy_p_proc
 begin
     if (source_obj == dude_obj) then begin
-        set_global_var(5, global_var(5) + 1);
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(4) > (3 * global_var(5))) or (global_var(2) == 1)) and (global_var(1) == 0)) then begin
-            set_global_var(2, 1);
-            set_global_var(3, 0);
+        set_global_var(GVAR_GOOD_MONSTER, global_var(GVAR_GOOD_MONSTER) + 1);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(4) > (3 * global_var(GVAR_GOOD_MONSTER))) or (global_var(GVAR_CHAMPION_REPUTATION) == 1)) and (global_var(GVAR_CHILDKILLER_REPUTATION) == 0)) then begin
+            set_global_var(GVAR_CHAMPION_REPUTATION, 1);
+            set_global_var(GVAR_BERSERKER_REPUTATION, 0);
         end
-        if (metarule(51, self_obj) == 2) then begin
+        if (critter_kill_type(self_obj) == KILL_TYPE_children_kills) then begin
             if (source_obj == dude_obj) then begin
-                set_global_var(2, 0);
-                set_global_var(1, global_var(1) + 1);
-                set_global_var(0, global_var(0) + -15);
+                set_global_var(GVAR_CHAMPION_REPUTATION, 0);
+                set_global_var(GVAR_CHILDKILLER_REPUTATION, global_var(GVAR_CHILDKILLER_REPUTATION) + 1);
+                set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + -15);
                 debug_msg("Player gains " + -15 + " Karma Points.");
-                set_global_var(37, 0);
-                set_global_var(38, 0);
-                set_global_var(39, 0);
-                set_global_var(40, 0);
-                set_global_var(41, 0);
-                set_global_var(42, 0);
-                set_global_var(43, 0);
-                set_global_var(44, 0);
-                set_global_var(45, 0);
-                if (global_var(0) >= 1000) then begin
-                    set_global_var(37, 1);
+                set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+                set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+                set_global_var(GVAR_KARMA_DEFENDER, 0);
+                set_global_var(GVAR_KARMA_WANDERER, 0);
+                set_global_var(GVAR_KARMA_BETRAYER, 0);
+                set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+                set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+                set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+                    set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
                 end
                 else begin
-                    if (global_var(0) >= 750) then begin
-                        set_global_var(38, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
                     end
                     else begin
-                        if (global_var(0) >= 500) then begin
-                            set_global_var(39, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                            set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                         end
                         else begin
-                            if (global_var(0) >= 250) then begin
-                                set_global_var(40, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                                set_global_var(GVAR_KARMA_DEFENDER, 1);
                             end
                             else begin
-                                if (global_var(0) > -250) then begin
-                                    set_global_var(41, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                                    set_global_var(GVAR_KARMA_WANDERER, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -500) then begin
-                                        set_global_var(42, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                        set_global_var(GVAR_KARMA_BETRAYER, 1);
                                     end
                                     else begin
-                                        if (global_var(0) > -750) then begin
-                                            set_global_var(43, 1);
+                                        if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                            set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                         end
                                         else begin
-                                            if (global_var(0) > -1000) then begin
-                                                set_global_var(44, 1);
+                                            if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                                set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                             end
                                             else begin
-                                                set_global_var(45, 1);
+                                                set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                             end
                                         end
                                     end
@@ -185,58 +185,58 @@ begin
                         end
                     end
                 end
-                set_global_var(696, global_var(696) + -8);
+                set_global_var(GVAR_TOWN_REP_SCRAPHEAP, global_var(GVAR_TOWN_REP_SCRAPHEAP) + -8);
                 debug_msg("Added " + -8 + " to Town Rep");
             end
         end
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(5) > (2 * global_var(4))) or (global_var(3) == 1))) then begin
-            set_global_var(3, 1);
-            set_global_var(2, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
+            set_global_var(GVAR_BERSERKER_REPUTATION, 1);
+            set_global_var(GVAR_CHAMPION_REPUTATION, 0);
         end
-        set_global_var(0, global_var(0) + -10);
+        set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + -10);
         debug_msg("Player gains " + -10 + " Karma Points.");
-        set_global_var(37, 0);
-        set_global_var(38, 0);
-        set_global_var(39, 0);
-        set_global_var(40, 0);
-        set_global_var(41, 0);
-        set_global_var(42, 0);
-        set_global_var(43, 0);
-        set_global_var(44, 0);
-        set_global_var(45, 0);
-        if (global_var(0) >= 1000) then begin
-            set_global_var(37, 1);
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+        set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+        set_global_var(GVAR_KARMA_DEFENDER, 0);
+        set_global_var(GVAR_KARMA_WANDERER, 0);
+        set_global_var(GVAR_KARMA_BETRAYER, 0);
+        set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+        set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+            set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
         end
         else begin
-            if (global_var(0) >= 750) then begin
-                set_global_var(38, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
             end
             else begin
-                if (global_var(0) >= 500) then begin
-                    set_global_var(39, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                 end
                 else begin
-                    if (global_var(0) >= 250) then begin
-                        set_global_var(40, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                        set_global_var(GVAR_KARMA_DEFENDER, 1);
                     end
                     else begin
-                        if (global_var(0) > -250) then begin
-                            set_global_var(41, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                            set_global_var(GVAR_KARMA_WANDERER, 1);
                         end
                         else begin
-                            if (global_var(0) > -500) then begin
-                                set_global_var(42, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                set_global_var(GVAR_KARMA_BETRAYER, 1);
                             end
                             else begin
-                                if (global_var(0) > -750) then begin
-                                    set_global_var(43, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -1000) then begin
-                                        set_global_var(44, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                     end
                                     else begin
-                                        set_global_var(45, 1);
+                                        set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                     end
                                 end
                             end
@@ -245,7 +245,7 @@ begin
                 end
             end
         end
-        set_global_var(696, global_var(696) + -5);
+        set_global_var(GVAR_TOWN_REP_SCRAPHEAP, global_var(GVAR_TOWN_REP_SCRAPHEAP) + -5);
         debug_msg("Added " + -5 + " to Town Rep");
     end
 end
@@ -254,10 +254,10 @@ procedure look_at_p_proc
 begin
     script_overrides;
     if (local_var(4) == 0) then begin
-        display_msg(message_str(1318, 100));
+        display_mstr(100);
     end
     else begin
-        display_msg(message_str(1318, 101));
+        display_mstr(101);
     end
 end
 
@@ -265,10 +265,10 @@ procedure description_p_proc
 begin
     script_overrides;
     if (local_var(4) == 0) then begin
-        display_msg(message_str(1318, 100));
+        display_mstr(100);
     end
     else begin
-        display_msg(message_str(1318, 101));
+        display_mstr(101);
     end
 end
 
@@ -278,17 +278,17 @@ end
 
 procedure damage_p_proc
 begin
-    if (party_member_obj(obj_pid(source_obj)) != 0) then begin
+    if (obj_in_party(source_obj)) then begin
         set_local_var(6, 1);
     end
 end
 
 procedure map_enter_p_proc
 begin
-    if (metarule(22, 0) == 0) then begin
-        if (metarule(14, 0)) then begin
-            critter_add_trait(self_obj, 1, 6, 214);
-            critter_add_trait(self_obj, 1, 5, 187);
+    if (is_loading_game == 0) then begin
+        if (map_first_run) then begin
+            critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 214);
+            critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_DEMO_BERSERKER);
             add_timer_event(self_obj, game_ticks(10), 1);
         end
     end
@@ -297,7 +297,7 @@ end
 procedure timed_event_p_proc
 begin
     if (fixed_param == 2) then begin
-        float_msg(self_obj, message_str(1318, 102), 0);
+        float_msg(self_obj, mstr(102), FLOAT_MSG_NORMAL);
     end
     else begin
         if (map_var(4)) then begin
@@ -317,56 +317,56 @@ end
 
 procedure Node000
 begin
-    float_msg(self_obj, message_str(1318, 103), 0);
+    float_msg(self_obj, mstr(103), FLOAT_MSG_NORMAL);
 end
 
 procedure Node001
 begin
-    float_msg(self_obj, message_str(1318, 104), 0);
+    float_msg(self_obj, mstr(104), FLOAT_MSG_NORMAL);
 end
 
 procedure Node002
 begin
     set_local_var(4, 1);
-    gsay_reply(1318, message_str(1318, 105 + (get_critter_stat(dude_obj, 34) == 1)) + message_str(1318, 107));
-    giq_option(4, 1318, 108, Node004, 50);
-    giq_option(4, 1318, 109, Node999, 50);
-    giq_option(6, 1318, 110, Node005, 50);
-    giq_option(-3, 1318, 111, Node003, 50);
+    Reply(mstr(105 + (dude_gender == 1)) + mstr(107));
+    giq_option(4, NAME, 108, Node004, NEUTRAL_REACTION);
+    giq_option(4, NAME, 109, Node999, NEUTRAL_REACTION);
+    giq_option(6, NAME, 110, Node005, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 111, Node003, NEUTRAL_REACTION);
 end
 
 procedure Node003
 begin
-    gsay_message(1318, 112, 50);
+    gsay_message(NAME, 112, NEUTRAL_REACTION);
 end
 
 procedure Node004
 begin
-    gsay_reply(1318, 113);
-    giq_option(4, 1318, 114, Node999, 50);
-    giq_option(4, 1318, 115, Node005, 50);
+    Reply(113);
+    giq_option(4, NAME, 114, Node999, NEUTRAL_REACTION);
+    giq_option(4, NAME, 115, Node005, NEUTRAL_REACTION);
 end
 
 procedure Node005
 begin
-    gsay_reply(1318, 116);
-    giq_option(4, 1318, 117, Node999, 50);
-    giq_option(5, 1318, 118, Node006, 50);
-    giq_option(6, 1318, 119, Node007, 50);
+    Reply(116);
+    giq_option(4, NAME, 117, Node999, NEUTRAL_REACTION);
+    giq_option(5, NAME, 118, Node006, NEUTRAL_REACTION);
+    giq_option(6, NAME, 119, Node007, NEUTRAL_REACTION);
 end
 
 procedure Node006
 begin
     if (cont) then begin
         cont := 0;
-        gsay_reply(1318, 120);
-        giq_option(0, 1318, message_str(1304, 409), Node006, 50);
+        Reply(120);
+        giq_option(0, NAME, message_str(SCRIPT_JUNKDEMO, 409), Node006, NEUTRAL_REACTION);
     end
     else begin
         cont := 1;
-        gsay_reply(1318, 121);
-        giq_option(6, 1318, 122, Node007, 50);
-        giq_option(4, 1318, Gbye, Node999, 50);
+        Reply(121);
+        giq_option(6, NAME, 122, Node007, NEUTRAL_REACTION);
+        giq_option(4, NAME, Gbye, Node999, NEUTRAL_REACTION);
     end
 end
 
@@ -374,24 +374,24 @@ procedure Node007
 begin
     if (cont) then begin
         cont := 0;
-        gsay_reply(1318, 123);
-        giq_option(0, 1318, message_str(1304, 409), Node007, 50);
+        Reply(123);
+        giq_option(0, NAME, message_str(SCRIPT_JUNKDEMO, 409), Node007, NEUTRAL_REACTION);
     end
     else begin
         cont := 1;
-        gsay_reply(1318, 124);
-        giq_option(4, 1318, 125, Node999, 50);
+        Reply(124);
+        giq_option(4, NAME, 125, Node999, NEUTRAL_REACTION);
     end
 end
 
 procedure Node008
 begin
-    float_msg(self_obj, message_str(1318, 126), 0);
+    float_msg(self_obj, mstr(126), FLOAT_MSG_NORMAL);
 end
 
 procedure Node009
 begin
-    float_msg(self_obj, message_str(1318, 127), 0);
+    float_msg(self_obj, mstr(127), FLOAT_MSG_NORMAL);
 end
 
 procedure run_away
@@ -409,6 +409,5 @@ end
 
 procedure Goodbyes
 begin
-    Gbye := message_str(1304, random(403, 408));
+    Gbye := message_str(SCRIPT_JUNKDEMO, random(403, 408));
 end
-
