diff --git a/Mods/DemoInFO2/Original/GL_NIGHTP2.ssl b/Mods/DemoInFO2/ReDefine/GL_NIGHTP2.ssl
index 0a6cfcc..2113b39 100644
--- a/Mods/DemoInFO2/Original/GL_NIGHTP2.ssl
+++ b/Mods/DemoInFO2/ReDefine/GL_NIGHTP2.ssl
@@ -19,70 +19,69 @@ end
 
 procedure RepeatNightPerson
 begin
-    if (has_trait(2, dude_obj, 13)) then begin
+    if (has_trait(TRAIT_TRAIT, dude_obj, TRAIT_sex_appeal)) then begin
         if ((game_time_hour > 601) andAlso (game_time_hour < 1800)) then begin
             set_sfall_global("time0001", 1);
-            if (global_var(708) == 0) then begin
-                set_global_var(708, 1);
+            if (global_var(GVAR_NIGHTPERSON) == 0) then begin
+                set_global_var(GVAR_NIGHTPERSON, 1);
             end
         end
         else begin
             set_sfall_global("time0001", 2);
-            if (global_var(708) == 0) then begin
-                set_global_var(708, 2);
+            if (global_var(GVAR_NIGHTPERSON) == 0) then begin
+                set_global_var(GVAR_NIGHTPERSON, 2);
             end
         end
-        if ((global_var(708) == 1) andAlso (get_sfall_global_int("time0001") == 1)) then begin
-            if (global_var(709) == 0) then begin
+        if ((global_var(GVAR_NIGHTPERSON) == 1) andAlso (get_sfall_global_int("time0001") == 1)) then begin
+            if (global_var(GVAR_NP_DAY) == 0) then begin
                 set_pc_extra_stat(1, get_pc_extra_stat(1) - 1);
                 set_pc_extra_stat(4, get_pc_extra_stat(4) - 1);
                 mod_skill_points_per_level(-2);
-                set_global_var(708, 2);
-                set_global_var(710, 1);
+                set_global_var(GVAR_NIGHTPERSON, 2);
+                set_global_var(GVAR_NP_NIGHT, 1);
             end
             else begin
                 set_pc_extra_stat(1, get_pc_extra_stat(1) - 2);
                 set_pc_extra_stat(4, get_pc_extra_stat(4) - 2);
                 mod_skill_points_per_level(-2);
-                set_global_var(708, 2);
-                set_global_var(710, 1);
+                set_global_var(GVAR_NIGHTPERSON, 2);
+                set_global_var(GVAR_NP_NIGHT, 1);
             end
         end
         else begin
-            if ((global_var(708) == 2) andAlso (get_sfall_global_int("time0001") == 2)) then begin
-                if (global_var(710) == 0) then begin
+            if ((global_var(GVAR_NIGHTPERSON) == 2) andAlso (get_sfall_global_int("time0001") == 2)) then begin
+                if (global_var(GVAR_NP_NIGHT) == 0) then begin
                     set_pc_extra_stat(1, get_pc_extra_stat(1) + 1);
                     set_pc_extra_stat(4, get_pc_extra_stat(4) + 1);
                     mod_skill_points_per_level(2);
-                    set_global_var(708, 1);
-                    set_global_var(709, 1);
+                    set_global_var(GVAR_NIGHTPERSON, 1);
+                    set_global_var(GVAR_NP_DAY, 1);
                 end
                 else begin
                     set_pc_extra_stat(1, get_pc_extra_stat(1) + 2);
                     set_pc_extra_stat(4, get_pc_extra_stat(4) + 2);
                     mod_skill_points_per_level(2);
-                    set_global_var(708, 1);
-                    set_global_var(709, 1);
+                    set_global_var(GVAR_NIGHTPERSON, 1);
+                    set_global_var(GVAR_NP_DAY, 1);
                 end
             end
         end
     end
     else begin
-        if (global_var(708) > 0) then begin
+        if (global_var(GVAR_NIGHTPERSON) > 0) then begin
             if ((game_time_hour > 601) andAlso (game_time_hour < 1800)) then begin
                 set_pc_extra_stat(1, get_pc_extra_stat(1) + 1);
                 set_pc_extra_stat(4, get_pc_extra_stat(4) + 1);
                 mod_skill_points_per_level(0);
-                set_global_var(708, -1);
+                set_global_var(GVAR_NIGHTPERSON, -1);
             end
             else begin
                 set_pc_extra_stat(1, get_pc_extra_stat(1) - 1);
                 set_pc_extra_stat(4, get_pc_extra_stat(4) - 1);
                 mod_skill_points_per_level(0);
-                set_global_var(708, -1);
+                set_global_var(GVAR_NIGHTPERSON, -1);
             end
             set_sfall_global("repeat00", -1);
         end
     end
 end
-
