diff --git a/Mods/DemoInFO2/Original/LEX.ssl b/Mods/DemoInFO2/ReDefine/LEX.ssl
index 9e1343a..bcbf6f4 100644
--- a/Mods/DemoInFO2/Original/LEX.ssl
+++ b/Mods/DemoInFO2/ReDefine/LEX.ssl
@@ -49,10 +49,10 @@ begin
             if (anim_busy(self_obj) == 0) then begin
                 if ((tile_distance_objs(self_obj, dude_obj) < 8) or obj_can_see_obj(self_obj, dude_obj)) then begin
                     if (anim_busy(self_obj) == 0) then begin
-                        global_temp := rotation_to_tile(tile_num(dude_obj), tile_num(self_obj));
+                        global_temp := rotation_to_tile(dude_tile, tile_num(self_obj));
                         animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(self_obj), global_temp, random(3, 10)), 1);
                         global_temp := (global_temp + 1) % 6;
-                        while ((anim_busy(self_obj) == 0) and (global_temp != rotation_to_tile(tile_num(dude_obj), tile_num(self_obj)))) do begin
+                        while ((anim_busy(self_obj) == 0) and (global_temp != rotation_to_tile(dude_tile, tile_num(self_obj)))) do begin
                             debug_msg("flee loop: rot == " + global_temp);
                             animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(self_obj), global_temp, random(3, 10)), 1);
                             global_temp := (global_temp + 1) % 6;
@@ -62,7 +62,7 @@ begin
             end
         end
         else begin
-            attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+            attack(dude_obj);
         end
     end
     if (obj_can_see_obj(self_obj, dude_obj) and (local_var(4) == 0) and (tile_distance_objs(self_obj, dude_obj) < 12)) then begin
@@ -70,7 +70,7 @@ begin
     end
     if (local_var(8) == 1) then begin
         if (tile_distance_objs(dude_obj, self_obj) > 3) then begin
-            animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(dude_obj), 1, 1), 0);
+            animate_move_obj_to_tile(self_obj, tile_num_in_direction(dude_tile, 1, 1), 0);
         end
     end
 end
@@ -84,7 +84,7 @@ end
 
 procedure talk_p_proc
 begin
-    start_gdialog(-1, self_obj, 4, -1, -1);
+    start_dialogue(self_obj, 4);
     gsay_start;
     if (local_var(4) == 0) then begin
         call Node000();
@@ -94,7 +94,7 @@ begin
             call Node009();
         end
         else begin
-            if (global_var(697) == 1) then begin
+            if (global_var(GVAR_GANG_WAR) == 1) then begin
                 call Node012();
             end
             else begin
@@ -114,59 +114,59 @@ end
 procedure destroy_p_proc
 begin
     if (source_obj == dude_obj) then begin
-        set_global_var(5, global_var(5) + 1);
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(4) > (3 * global_var(5))) or (global_var(2) == 1)) and (global_var(1) == 0)) then begin
-            set_global_var(2, 1);
-            set_global_var(3, 0);
+        set_global_var(GVAR_GOOD_MONSTER, global_var(GVAR_GOOD_MONSTER) + 1);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(4) > (3 * global_var(GVAR_GOOD_MONSTER))) or (global_var(GVAR_CHAMPION_REPUTATION) == 1)) and (global_var(GVAR_CHILDKILLER_REPUTATION) == 0)) then begin
+            set_global_var(GVAR_CHAMPION_REPUTATION, 1);
+            set_global_var(GVAR_BERSERKER_REPUTATION, 0);
         end
-        if (metarule(51, self_obj) == 2) then begin
+        if (critter_kill_type(self_obj) == KILL_TYPE_children_kills) then begin
             if (source_obj == dude_obj) then begin
-                set_global_var(2, 0);
-                set_global_var(1, global_var(1) + 1);
-                set_global_var(0, global_var(0) + -15);
+                set_global_var(GVAR_CHAMPION_REPUTATION, 0);
+                set_global_var(GVAR_CHILDKILLER_REPUTATION, global_var(GVAR_CHILDKILLER_REPUTATION) + 1);
+                set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + -15);
                 debug_msg("Player gains " + -15 + " Karma Points.");
-                set_global_var(37, 0);
-                set_global_var(38, 0);
-                set_global_var(39, 0);
-                set_global_var(40, 0);
-                set_global_var(41, 0);
-                set_global_var(42, 0);
-                set_global_var(43, 0);
-                set_global_var(44, 0);
-                set_global_var(45, 0);
-                if (global_var(0) >= 1000) then begin
-                    set_global_var(37, 1);
+                set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+                set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+                set_global_var(GVAR_KARMA_DEFENDER, 0);
+                set_global_var(GVAR_KARMA_WANDERER, 0);
+                set_global_var(GVAR_KARMA_BETRAYER, 0);
+                set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+                set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+                set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+                    set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
                 end
                 else begin
-                    if (global_var(0) >= 750) then begin
-                        set_global_var(38, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
                     end
                     else begin
-                        if (global_var(0) >= 500) then begin
-                            set_global_var(39, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                            set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                         end
                         else begin
-                            if (global_var(0) >= 250) then begin
-                                set_global_var(40, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                                set_global_var(GVAR_KARMA_DEFENDER, 1);
                             end
                             else begin
-                                if (global_var(0) > -250) then begin
-                                    set_global_var(41, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                                    set_global_var(GVAR_KARMA_WANDERER, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -500) then begin
-                                        set_global_var(42, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                        set_global_var(GVAR_KARMA_BETRAYER, 1);
                                     end
                                     else begin
-                                        if (global_var(0) > -750) then begin
-                                            set_global_var(43, 1);
+                                        if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                            set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                         end
                                         else begin
-                                            if (global_var(0) > -1000) then begin
-                                                set_global_var(44, 1);
+                                            if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                                set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                             end
                                             else begin
-                                                set_global_var(45, 1);
+                                                set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                             end
                                         end
                                     end
@@ -175,58 +175,58 @@ begin
                         end
                     end
                 end
-                set_global_var(696, global_var(696) + -8);
+                set_global_var(GVAR_TOWN_REP_SCRAPHEAP, global_var(GVAR_TOWN_REP_SCRAPHEAP) + -8);
                 debug_msg("Added " + -8 + " to Town Rep");
             end
         end
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(5) > (2 * global_var(4))) or (global_var(3) == 1))) then begin
-            set_global_var(3, 1);
-            set_global_var(2, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
+            set_global_var(GVAR_BERSERKER_REPUTATION, 1);
+            set_global_var(GVAR_CHAMPION_REPUTATION, 0);
         end
-        set_global_var(0, global_var(0) + -10);
+        set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + -10);
         debug_msg("Player gains " + -10 + " Karma Points.");
-        set_global_var(37, 0);
-        set_global_var(38, 0);
-        set_global_var(39, 0);
-        set_global_var(40, 0);
-        set_global_var(41, 0);
-        set_global_var(42, 0);
-        set_global_var(43, 0);
-        set_global_var(44, 0);
-        set_global_var(45, 0);
-        if (global_var(0) >= 1000) then begin
-            set_global_var(37, 1);
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+        set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+        set_global_var(GVAR_KARMA_DEFENDER, 0);
+        set_global_var(GVAR_KARMA_WANDERER, 0);
+        set_global_var(GVAR_KARMA_BETRAYER, 0);
+        set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+        set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+            set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
         end
         else begin
-            if (global_var(0) >= 750) then begin
-                set_global_var(38, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
             end
             else begin
-                if (global_var(0) >= 500) then begin
-                    set_global_var(39, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                 end
                 else begin
-                    if (global_var(0) >= 250) then begin
-                        set_global_var(40, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                        set_global_var(GVAR_KARMA_DEFENDER, 1);
                     end
                     else begin
-                        if (global_var(0) > -250) then begin
-                            set_global_var(41, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                            set_global_var(GVAR_KARMA_WANDERER, 1);
                         end
                         else begin
-                            if (global_var(0) > -500) then begin
-                                set_global_var(42, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                set_global_var(GVAR_KARMA_BETRAYER, 1);
                             end
                             else begin
-                                if (global_var(0) > -750) then begin
-                                    set_global_var(43, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -1000) then begin
-                                        set_global_var(44, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                     end
                                     else begin
-                                        set_global_var(45, 1);
+                                        set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                     end
                                 end
                             end
@@ -235,7 +235,7 @@ begin
                 end
             end
         end
-        set_global_var(696, global_var(696) + -5);
+        set_global_var(GVAR_TOWN_REP_SCRAPHEAP, global_var(GVAR_TOWN_REP_SCRAPHEAP) + -5);
         debug_msg("Added " + -5 + " to Town Rep");
     end
 end
@@ -244,10 +244,10 @@ procedure look_at_p_proc
 begin
     script_overrides;
     if (local_var(4) == 1) then begin
-        display_msg(message_str(1307, 100));
+        display_mstr(100);
     end
     else begin
-        display_msg(message_str(1307, 101));
+        display_mstr(101);
     end
 end
 
@@ -255,10 +255,10 @@ procedure description_p_proc
 begin
     script_overrides;
     if (local_var(4) == 1) then begin
-        display_msg(message_str(1307, 100));
+        display_mstr(100);
     end
     else begin
-        display_msg(message_str(1307, 101));
+        display_mstr(101);
     end
 end
 
@@ -275,10 +275,10 @@ end
 
 procedure map_enter_p_proc
 begin
-    if (metarule(22, 0) == 0) then begin
-        if (metarule(14, 0)) then begin
-            critter_add_trait(self_obj, 1, 6, 213);
-            critter_add_trait(self_obj, 1, 5, 188);
+    if (is_loading_game == 0) then begin
+        if (map_first_run) then begin
+            critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 213);
+            critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_DEMO_CAREFUL);
         end
     end
 end
@@ -295,196 +295,195 @@ end
 procedure Node000
 begin
     set_local_var(4, 1);
-    gsay_reply(1307, 104);
-    giq_option(4, 1307, 105, Node001, 50);
-    giq_option(4, 1307, 106, Node007, 50);
+    Reply(104);
+    giq_option(4, NAME, 105, Node001, NEUTRAL_REACTION);
+    giq_option(4, NAME, 106, Node007, NEUTRAL_REACTION);
 end
 
 procedure Node001
 begin
     set_map_var(6, 1);
-    gsay_reply(1307, 107);
-    giq_option(4, 1307, 108, Node002, 50);
-    giq_option(4, 1307, 109, Node999, 50);
+    Reply(107);
+    giq_option(4, NAME, 108, Node002, NEUTRAL_REACTION);
+    giq_option(4, NAME, 109, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node002
 begin
-    gsay_reply(1307, 110);
-    giq_option(4, 1307, 111, Node003, 50);
+    Reply(110);
+    giq_option(4, NAME, 111, Node003, NEUTRAL_REACTION);
 end
 
 procedure Node003
 begin
-    set_global_var(699, 1);
+    set_global_var(GVAR_GANG_2, 1);
     set_map_var(5, 1);
-    gsay_reply(1307, 112);
-    giq_option(4, 1307, 113, Node004, 50);
+    Reply(112);
+    giq_option(4, NAME, 113, Node004, NEUTRAL_REACTION);
 end
 
 procedure Node004
 begin
-    gsay_reply(1307, 114);
-    giq_option(4, 1307, 115, Node005, 50);
+    Reply(114);
+    giq_option(4, NAME, 115, Node005, NEUTRAL_REACTION);
 end
 
 procedure Node005
 begin
-    gsay_reply(1307, 116);
-    giq_option(4, 1307, 117, Node006, 50);
+    Reply(116);
+    giq_option(4, NAME, 117, Node006, NEUTRAL_REACTION);
 end
 
 procedure Node006
 begin
-    gsay_reply(1307, 118);
-    giq_option(4, 1307, 119, Node999, 50);
-    giq_option(4, 1307, 120, Node017, 50);
+    Reply(118);
+    giq_option(4, NAME, 119, Node999, NEUTRAL_REACTION);
+    giq_option(4, NAME, 120, Node017, NEUTRAL_REACTION);
 end
 
 procedure Node007
 begin
-    gsay_reply(1307, 121);
-    giq_option(4, 1307, 122, Node008, 50);
+    Reply(121);
+    giq_option(4, NAME, 122, Node008, NEUTRAL_REACTION);
 end
 
 procedure Node008
 begin
-    gsay_reply(1307, 123);
-    giq_option(4, 1307, 124, Node999, 50);
-    giq_option(4, 1307, 120, Node017, 50);
+    Reply(123);
+    giq_option(4, NAME, 124, Node999, NEUTRAL_REACTION);
+    giq_option(4, NAME, 120, Node017, NEUTRAL_REACTION);
 end
 
 procedure Node009
 begin
-    set_global_var(700, 1);
-    gsay_reply(1307, 125);
-    giq_option(4, 1307, 126, Node010, 50);
-    giq_option(4, 1307, 127, Node011, 50);
+    set_global_var(GVAR_GENERATOR_1, 1);
+    Reply(125);
+    giq_option(4, NAME, 126, Node010, NEUTRAL_REACTION);
+    giq_option(4, NAME, 127, Node011, NEUTRAL_REACTION);
 end
 
 procedure Node010
 begin
-    gsay_reply(1307, 128);
-    giq_option(4, 1307, 103, Node999, 50);
+    Reply(128);
+    giq_option(4, NAME, 103, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node011
 begin
-    gsay_reply(1307, 129);
-    giq_option(4, 1307, 103, Node999, 50);
+    Reply(129);
+    giq_option(4, NAME, 103, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node012
 begin
-    gsay_reply(1307, 130);
-    giq_option(4, 1307, 131, Node013, 50);
-    giq_option(4, 1307, 132, Node999, 50);
+    Reply(130);
+    giq_option(4, NAME, 131, Node013, NEUTRAL_REACTION);
+    giq_option(4, NAME, 132, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node013
 begin
-    gsay_reply(1307, 133);
-    giq_option(4, 1307, 134, Node022, 50);
-    giq_option(4, 1307, 135, Node999, 50);
+    Reply(133);
+    giq_option(4, NAME, 134, Node022, NEUTRAL_REACTION);
+    giq_option(4, NAME, 135, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node014
 begin
-    gsay_reply(1307, 136);
-    giq_option(4, 1307, 137, Node015, 50);
-    giq_option(4, 1307, 138, Node016, 50);
-    giq_option(4, 1307, 139, Node999, 50);
+    Reply(136);
+    giq_option(4, NAME, 137, Node015, NEUTRAL_REACTION);
+    giq_option(4, NAME, 138, Node016, NEUTRAL_REACTION);
+    giq_option(4, NAME, 139, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node015
 begin
-    gsay_reply(1307, 140);
-    giq_option(4, 1307, 103, Node998, 50);
+    Reply(140);
+    giq_option(4, NAME, 103, Node998, NEUTRAL_REACTION);
 end
 
 procedure Node016
 begin
-    gsay_reply(1307, 141);
-    giq_option(4, 1307, 142, Node999, 50);
-    giq_option(4, 1307, 120, Node017, 50);
+    Reply(141);
+    giq_option(4, NAME, 142, Node999, NEUTRAL_REACTION);
+    giq_option(4, NAME, 120, Node017, NEUTRAL_REACTION);
 end
 
 procedure Node017
 begin
-    gsay_reply(1307, 143);
+    Reply(143);
     if (map_var(7)) then begin
-        giq_option(4, 1307, 144, Node019, 50);
+        giq_option(4, NAME, 144, Node019, NEUTRAL_REACTION);
     end
     if (map_var(6)) then begin
-        giq_option(4, 1307, 145, Node020, 50);
+        giq_option(4, NAME, 145, Node020, NEUTRAL_REACTION);
     end
     if (map_var(5)) then begin
-        giq_option(4, 1307, 146, Node021, 50);
+        giq_option(4, NAME, 146, Node021, NEUTRAL_REACTION);
     end
-    giq_option(4, 1307, 147, Node018, 50);
-    giq_option(4, 1307, 148, Node999, 50);
+    giq_option(4, NAME, 147, Node018, NEUTRAL_REACTION);
+    giq_option(4, NAME, 148, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node018
 begin
-    gsay_reply(1307, 149);
+    Reply(149);
     if (map_var(7)) then begin
-        giq_option(4, 1307, 144, Node019, 50);
+        giq_option(4, NAME, 144, Node019, NEUTRAL_REACTION);
     end
     if (map_var(6)) then begin
-        giq_option(4, 1307, 145, Node020, 50);
+        giq_option(4, NAME, 145, Node020, NEUTRAL_REACTION);
     end
     if (map_var(5)) then begin
-        giq_option(4, 1307, 146, Node021, 50);
+        giq_option(4, NAME, 146, Node021, NEUTRAL_REACTION);
     end
-    giq_option(4, 1307, 148, Node999, 50);
+    giq_option(4, NAME, 148, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node019
 begin
-    set_global_var(701, 1);
-    set_global_var(699, 1);
-    gsay_reply(1307, 150);
+    set_global_var(GVAR_GENERATOR_2, 1);
+    set_global_var(GVAR_GANG_2, 1);
+    Reply(150);
     if (map_var(6)) then begin
-        giq_option(4, 1307, 145, Node020, 50);
+        giq_option(4, NAME, 145, Node020, NEUTRAL_REACTION);
     end
     if (map_var(5)) then begin
-        giq_option(4, 1307, 146, Node021, 50);
+        giq_option(4, NAME, 146, Node021, NEUTRAL_REACTION);
     end
-    giq_option(4, 1307, 147, Node018, 50);
-    giq_option(4, 1307, 148, Node999, 50);
+    giq_option(4, NAME, 147, Node018, NEUTRAL_REACTION);
+    giq_option(4, NAME, 148, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node020
 begin
-    gsay_reply(1307, 151);
+    Reply(151);
     if (map_var(7)) then begin
-        giq_option(4, 1307, 144, Node019, 50);
+        giq_option(4, NAME, 144, Node019, NEUTRAL_REACTION);
     end
     if (map_var(5)) then begin
-        giq_option(4, 1307, 146, Node021, 50);
+        giq_option(4, NAME, 146, Node021, NEUTRAL_REACTION);
     end
-    giq_option(4, 1307, 147, Node018, 50);
-    giq_option(4, 1307, 148, Node999, 50);
+    giq_option(4, NAME, 147, Node018, NEUTRAL_REACTION);
+    giq_option(4, NAME, 148, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node021
 begin
     set_map_var(7, 1);
-    gsay_reply(1307, 152);
+    Reply(152);
     if (map_var(7)) then begin
-        giq_option(4, 1307, 144, Node019, 50);
+        giq_option(4, NAME, 144, Node019, NEUTRAL_REACTION);
     end
     if (map_var(6)) then begin
-        giq_option(4, 1307, 145, Node020, 50);
+        giq_option(4, NAME, 145, Node020, NEUTRAL_REACTION);
     end
-    giq_option(4, 1307, 147, Node018, 50);
-    giq_option(4, 1307, 148, Node999, 50);
+    giq_option(4, NAME, 147, Node018, NEUTRAL_REACTION);
+    giq_option(4, NAME, 148, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node022
 begin
-    critter_add_trait(self_obj, 1, 6, 0);
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 0);
     set_local_var(8, 1);
 end
-
