diff --git a/Mods/DemoInFO2/Original/PENSMELL.ssl b/Mods/DemoInFO2/ReDefine/PENSMELL.ssl
index 323316a..b4d9f45 100644
--- a/Mods/DemoInFO2/Original/PENSMELL.ssl
+++ b/Mods/DemoInFO2/ReDefine/PENSMELL.ssl
@@ -6,7 +6,7 @@ procedure spatial_p_proc
 begin
     if (source_obj == dude_obj) then begin
         if (not(local_var(4))) then begin
-            display_msg(message_str(1321, 100));
+            display_msg(mstr(100));
             set_local_var(4, 1);
         end
     end
@@ -15,4 +15,3 @@ end
 procedure start
 begin
 end
-
