diff --git a/Mods/DemoInFO2/Original/SKIPPY.ssl b/Mods/DemoInFO2/ReDefine/SKIPPY.ssl
index e802d0a..72fbd07 100644
--- a/Mods/DemoInFO2/Original/SKIPPY.ssl
+++ b/Mods/DemoInFO2/ReDefine/SKIPPY.ssl
@@ -50,10 +50,10 @@ begin
     variable LVar0 := 0;
     variable LVar1 := 0;
     if (map_var(0)) then begin
-        critter_add_trait(self_obj, 1, 6, 0);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 0);
     end
     else begin
-        critter_add_trait(self_obj, 1, 6, 212);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 212);
     end
     if ((local_var(5) > 1) and obj_can_see_obj(self_obj, dude_obj)) then begin
         set_local_var(5, 1);
@@ -61,10 +61,10 @@ begin
             if (anim_busy(self_obj) == 0) then begin
                 if ((tile_distance_objs(self_obj, dude_obj) < 8) or obj_can_see_obj(self_obj, dude_obj)) then begin
                     if (anim_busy(self_obj) == 0) then begin
-                        global_temp := rotation_to_tile(tile_num(dude_obj), tile_num(self_obj));
+                        global_temp := rotation_to_tile(dude_tile, tile_num(self_obj));
                         animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(self_obj), global_temp, random(3, 10)), 1);
                         global_temp := (global_temp + 1) % 6;
-                        while ((anim_busy(self_obj) == 0) and (global_temp != rotation_to_tile(tile_num(dude_obj), tile_num(self_obj)))) do begin
+                        while ((anim_busy(self_obj) == 0) and (global_temp != rotation_to_tile(dude_tile, tile_num(self_obj)))) do begin
                             debug_msg("flee loop: rot == " + global_temp);
                             animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(self_obj), global_temp, random(3, 10)), 1);
                             global_temp := (global_temp + 1) % 6;
@@ -74,7 +74,7 @@ begin
             end
         end
         else begin
-            attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+            attack(dude_obj);
         end
     end
     if (Skippy_Talk == 1) then begin
@@ -84,36 +84,36 @@ begin
     if (I_Am_A_Fool and Crypt_Taunt) then begin
         Crypt_Taunt := 0;
         if (random(1, 5) == 1) then begin
-            float_msg(self_obj, message_str(1304, random(100, 113)), -1);
+            float_msg(self_obj, message_str(SCRIPT_JUNKDEMO, random(100, 113)), FLOAT_MSG_SEQUENTIAL);
         end
     end
     else begin
         if (I_Am_A_Crypt and Fool_Taunt) then begin
             Fool_Taunt := 0;
             if (random(1, 5) == 1) then begin
-                float_msg(self_obj, message_str(1304, random(200, 218)), -1);
+                float_msg(self_obj, message_str(SCRIPT_JUNKDEMO, random(200, 218)), FLOAT_MSG_SEQUENTIAL);
             end
         end
         else begin
-            if ((global_var(697) == 1) and (tile_num(self_obj) != 20725)) then begin
+            if ((global_var(GVAR_GANG_WAR) == 1) and (tile_num(self_obj) != 20725)) then begin
                 animate_move_obj_to_tile(self_obj, 20725, 0);
             end
             else begin
-                if ((global_var(697) == 1) and (tile_num(self_obj) == 20725)) then begin
-                    set_global_var(697, 0);
+                if ((global_var(GVAR_GANG_WAR) == 1) and (tile_num(self_obj) == 20725)) then begin
+                    set_global_var(GVAR_GANG_WAR, 0);
                     add_timer_event(self_obj, game_ticks(2), 1);
                 end
                 else begin
                     if (map_var(2) and (tile_num(self_obj) != 27504)) then begin
                         if ((self_obj == Phrax_Pointer) and (This_Once == 0)) then begin
                             This_Once := 1;
-                            float_msg(self_obj, message_str(1304, 300), 2);
+                            float_msg(self_obj, message_str(SCRIPT_JUNKDEMO, 300), FLOAT_MSG_RED);
                         end
-                        set_global_var(697, 2);
+                        set_global_var(GVAR_GANG_WAR, 2);
                         animate_move_obj_to_tile(self_obj, 27504, 0);
                     end
                     else begin
-                        if ((tile_num(self_obj) == 27504) and (global_var(697) == 2)) then begin
+                        if ((tile_num(self_obj) == 27504) and (global_var(GVAR_GANG_WAR) == 2)) then begin
                             set_obj_visibility(self_obj, 1);
                             if (I_Am_A_Fool) then begin
                                 set_map_var(10, map_var(10) - 1);
@@ -142,7 +142,7 @@ begin
     if (source_obj == dude_obj) then begin
         set_local_var(5, 2);
     end
-    float_msg(self_obj, message_str(1323, 122), 0);
+    float_msg(self_obj, mstr(122), FLOAT_MSG_NORMAL);
 end
 
 procedure talk_p_proc
@@ -212,55 +212,55 @@ end
 procedure destroy_p_proc
 begin
     if (source_obj == dude_obj) then begin
-        set_global_var(4, global_var(4) + 1);
-        if (metarule(51, self_obj) == 2) then begin
+        set_global_var(GVAR_BAD_MONSTER, global_var(GVAR_BAD_MONSTER) + 1);
+        if (critter_kill_type(self_obj) == 2) then begin
             if (source_obj == dude_obj) then begin
-                set_global_var(2, 0);
-                set_global_var(1, global_var(1) + 1);
-                set_global_var(0, global_var(0) + -15);
+                set_global_var(GVAR_CHAMPION_REPUTATION, 0);
+                set_global_var(GVAR_CHILDKILLER_REPUTATION, global_var(GVAR_CHILDKILLER_REPUTATION) + 1);
+                set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + -15);
                 debug_msg("Player gains " + -15 + " Karma Points.");
-                set_global_var(37, 0);
-                set_global_var(38, 0);
-                set_global_var(39, 0);
-                set_global_var(40, 0);
-                set_global_var(41, 0);
-                set_global_var(42, 0);
-                set_global_var(43, 0);
-                set_global_var(44, 0);
-                set_global_var(45, 0);
-                if (global_var(0) >= 1000) then begin
-                    set_global_var(37, 1);
+                set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+                set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+                set_global_var(GVAR_KARMA_DEFENDER, 0);
+                set_global_var(GVAR_KARMA_WANDERER, 0);
+                set_global_var(GVAR_KARMA_BETRAYER, 0);
+                set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+                set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+                set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+                    set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
                 end
                 else begin
-                    if (global_var(0) >= 750) then begin
-                        set_global_var(38, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
                     end
                     else begin
-                        if (global_var(0) >= 500) then begin
-                            set_global_var(39, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                            set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                         end
                         else begin
-                            if (global_var(0) >= 250) then begin
-                                set_global_var(40, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                                set_global_var(GVAR_KARMA_DEFENDER, 1);
                             end
                             else begin
-                                if (global_var(0) > -250) then begin
-                                    set_global_var(41, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                                    set_global_var(GVAR_KARMA_WANDERER, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -500) then begin
-                                        set_global_var(42, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                        set_global_var(GVAR_KARMA_BETRAYER, 1);
                                     end
                                     else begin
-                                        if (global_var(0) > -750) then begin
-                                            set_global_var(43, 1);
+                                        if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                            set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                         end
                                         else begin
-                                            if (global_var(0) > -1000) then begin
-                                                set_global_var(44, 1);
+                                            if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                                set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                             end
                                             else begin
-                                                set_global_var(45, 1);
+                                                set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                             end
                                         end
                                     end
@@ -269,62 +269,62 @@ begin
                         end
                     end
                 end
-                set_global_var(696, global_var(696) + -8);
+                set_global_var(GVAR_TOWN_REP_SCRAPHEAP, global_var(GVAR_TOWN_REP_SCRAPHEAP) + -8);
                 debug_msg("Added " + -8 + " to Town Rep");
             end
         end
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(5) > (2 * global_var(4))) or (global_var(3) == 1))) then begin
-            set_global_var(3, 1);
-            set_global_var(2, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
+            set_global_var(GVAR_BERSERKER_REPUTATION, 1);
+            set_global_var(GVAR_CHAMPION_REPUTATION, 0);
         end
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(4) > (3 * global_var(5))) or (global_var(2) == 1)) and (global_var(1) == 0)) then begin
-            set_global_var(2, 1);
-            set_global_var(3, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(4) > (3 * global_var(GVAR_GOOD_MONSTER))) or (global_var(GVAR_CHAMPION_REPUTATION) == 1)) and (global_var(GVAR_CHILDKILLER_REPUTATION) == 0)) then begin
+            set_global_var(GVAR_CHAMPION_REPUTATION, 1);
+            set_global_var(GVAR_BERSERKER_REPUTATION, 0);
         end
-        set_global_var(0, global_var(0) + 5);
+        set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + 5);
         debug_msg("Player gains " + 5 + " Karma Points.");
-        set_global_var(37, 0);
-        set_global_var(38, 0);
-        set_global_var(39, 0);
-        set_global_var(40, 0);
-        set_global_var(41, 0);
-        set_global_var(42, 0);
-        set_global_var(43, 0);
-        set_global_var(44, 0);
-        set_global_var(45, 0);
-        if (global_var(0) >= 1000) then begin
-            set_global_var(37, 1);
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+        set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+        set_global_var(GVAR_KARMA_DEFENDER, 0);
+        set_global_var(GVAR_KARMA_WANDERER, 0);
+        set_global_var(GVAR_KARMA_BETRAYER, 0);
+        set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+        set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+            set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
         end
         else begin
-            if (global_var(0) >= 750) then begin
-                set_global_var(38, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
             end
             else begin
-                if (global_var(0) >= 500) then begin
-                    set_global_var(39, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                 end
                 else begin
-                    if (global_var(0) >= 250) then begin
-                        set_global_var(40, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                        set_global_var(GVAR_KARMA_DEFENDER, 1);
                     end
                     else begin
-                        if (global_var(0) > -250) then begin
-                            set_global_var(41, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                            set_global_var(GVAR_KARMA_WANDERER, 1);
                         end
                         else begin
-                            if (global_var(0) > -500) then begin
-                                set_global_var(42, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                set_global_var(GVAR_KARMA_BETRAYER, 1);
                             end
                             else begin
-                                if (global_var(0) > -750) then begin
-                                    set_global_var(43, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -1000) then begin
-                                        set_global_var(44, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                     end
                                     else begin
-                                        set_global_var(45, 1);
+                                        set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                     end
                                 end
                             end
@@ -333,7 +333,7 @@ begin
                 end
             end
         end
-        set_global_var(696, global_var(696) + 2);
+        set_global_var(GVAR_TOWN_REP_SCRAPHEAP, global_var(GVAR_TOWN_REP_SCRAPHEAP) + 2);
         debug_msg("Added " + 2 + " to Town Rep");
     end
     set_map_var(10, map_var(10) - 1);
@@ -343,13 +343,13 @@ procedure look_at_p_proc
 begin
     script_overrides;
     if (local_var(4) == 1) then begin
-        display_msg(message_str(1323, 100));
+        display_msg(mstr(100));
     end
     else begin
-        display_msg(message_str(1323, 101));
+        display_msg(mstr(101));
     end
     if (obj_is_carrying_obj_pid(self_obj, 75)) then begin
-        display_msg(message_str(1323, 102));
+        display_msg(mstr(102));
     end
 end
 
@@ -357,13 +357,13 @@ procedure description_p_proc
 begin
     script_overrides;
     if (local_var(4) == 1) then begin
-        display_msg(message_str(1323, 100));
+        display_msg(mstr(100));
     end
     else begin
-        display_msg(message_str(1323, 101));
+        display_msg(mstr(101));
     end
     if (obj_is_carrying_obj_pid(self_obj, 75)) then begin
-        display_msg(message_str(1323, 102));
+        display_msg(mstr(102));
     end
 end
 
@@ -381,11 +381,11 @@ end
 procedure map_enter_p_proc
 begin
     variable LVar0 := 0;
-    if (metarule(22, 0) == 0) then begin
-        if (metarule(14, 0)) then begin
-            critter_add_trait(self_obj, 1, 6, 212);
-            critter_add_trait(self_obj, 1, 5, 187);
-            LVar0 := create_object_sid(75, 0, 0, -1);
+    if (is_loading_game == 0) then begin
+        if (map_first_run) then begin
+            critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 212);
+            critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_DEMO_BERSERKER);
+            LVar0 := create_object(75, 0, 0);
             add_mult_objs_to_inven(self_obj, LVar0, 1);
         end
     end
@@ -394,25 +394,25 @@ end
 procedure timed_event_p_proc
 begin
     if (fixed_param == 1) then begin
-        set_global_var(697, 0);
+        set_global_var(GVAR_GANG_WAR, 0);
         if (I_Am_A_Fool) then begin
             Fool_Taunt := 1;
-            float_msg(self_obj, message_str(1304, random(100, 115)), -1);
+            float_msg(self_obj, message_str(SCRIPT_JUNKDEMO, random(100, 115)), FLOAT_MSG_SEQUENTIAL);
         end
         else begin
             if (I_Am_A_Crypt) then begin
                 Crypt_Taunt := 1;
-                float_msg(self_obj, message_str(1304, random(200, 218)), -1);
+                float_msg(self_obj, message_str(SCRIPT_JUNKDEMO, random(200, 218)), FLOAT_MSG_SEQUENTIAL);
             end
         end
         add_timer_event(self_obj, 50, 2);
     end
     if (fixed_param == 2) then begin
         if (I_Am_A_Fool) then begin
-            attack_complex(Crypt_Pointer, 0, 1, 0, 0, 30000, 0, 0);
+            attack(Crypt_Pointer);
         end
         else begin
-            attack_complex(Fool_Pointer, 0, 1, 0, 0, 30000, 0, 0);
+            attack(Fool_Pointer);
         end
     end
 end
@@ -428,35 +428,35 @@ end
 
 procedure Node000
 begin
-    gsay_reply(1323, 105);
-    giq_option(4, 1323, 106, Node001, 50);
-    giq_option(4, 1323, 107, Node006, 50);
-    giq_option(4, 1323, 108, Node004, 50);
+    Reply(105);
+    giq_option(4, NAME, 106, Node001, NEUTRAL_REACTION);
+    giq_option(4, NAME, 107, Node006, NEUTRAL_REACTION);
+    giq_option(4, NAME, 108, Node004, NEUTRAL_REACTION);
 end
 
 procedure Node001
 begin
-    gsay_reply(1323, 109);
-    giq_option(4, 1323, 110, Node002, 50);
-    giq_option(4, 1323, 111, Node003, 50);
+    Reply(109);
+    giq_option(4, NAME, 110, Node002, NEUTRAL_REACTION);
+    giq_option(4, NAME, 111, Node003, NEUTRAL_REACTION);
 end
 
 procedure Node002
 begin
-    gsay_reply(1323, 112);
-    giq_option(4, 1323, 104, Node998, 50);
+    Reply(112);
+    giq_option(4, NAME, 104, Node998, NEUTRAL_REACTION);
 end
 
 procedure Node003
 begin
-    gsay_reply(1323, 113);
-    giq_option(4, 1323, 114, Node004, 50);
+    Reply(113);
+    giq_option(4, NAME, 114, Node004, NEUTRAL_REACTION);
 end
 
 procedure Node004
 begin
-    gsay_reply(1323, 115);
-    giq_option(4, 1323, 116, Node04a, 50);
+    Reply(115);
+    giq_option(4, NAME, 116, Node04a, NEUTRAL_REACTION);
 end
 
 procedure Node04a
@@ -476,40 +476,39 @@ begin
         LVar0 := obj_carrying_pid_obj(self_obj, 75);
         rm_obj_from_inven(self_obj, LVar0);
         add_obj_to_inven(dude_obj, LVar0);
-        gsay_reply(1323, 117);
+        Reply(117);
     end
-    giq_option(4, 1323, 104, Node999, 50);
+    giq_option(4, NAME, 104, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node006
 begin
-    gsay_reply(1323, 118);
-    giq_option(4, 1323, 104, Node998, 50);
+    Reply(118);
+    giq_option(4, NAME, 104, Node998, NEUTRAL_REACTION);
 end
 
 procedure Node007
 begin
-    gsay_reply(1323, 119);
-    giq_option(4, 1323, 104, Node999, 50);
+    Reply(119);
+    giq_option(4, NAME, 104, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node008
 begin
-    gsay_reply(1323, 120);
-    giq_option(4, 1323, 104, Node999, 50);
+    Reply(120);
+    giq_option(4, NAME, 104, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node009
 begin
-    gsay_reply(1323, 121);
-    giq_option(4, 1323, 104, Node999, 50);
+    Reply(121);
+    giq_option(4, NAME, 104, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node011
 begin
     Skippy_Talk := 0;
-    float_msg(self_obj, message_str(1323, 125), 0);
-    animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(dude_obj), 0, 1), 0);
+    float_msg(self_obj, mstr(125), FLOAT_MSG_NORMAL);
+    animate_move_obj_to_tile(self_obj, tile_num_in_direction(dude_tile, 0, 1), 0);
     set_local_var(5, 2);
 end
-
