diff --git a/Mods/MegaMod/Original/A1TRIBAL.ssl b/Mods/MegaMod/ReDefine/A1TRIBAL.ssl
index efeecef..3657583 100644
--- a/Mods/MegaMod/Original/A1TRIBAL.ssl
+++ b/Mods/MegaMod/ReDefine/A1TRIBAL.ssl
@@ -31,8 +31,8 @@ end
 procedure timed_event_p_proc
 begin
     if (fixed_param == 20) then begin
-        metarule3(100, self_obj, 20, 0);
-        if ((combat_is_initialized == 0) and obj_can_see_obj(self_obj, dude_obj) and (tile_distance(tile_num(self_obj), tile_num(dude_obj)) <= 5)) then begin
+        rm_fixed_timer_event(self_obj, 20);
+        if ((combat_is_initialized == 0) and obj_can_see_obj(self_obj, dude_obj) and (tile_distance(tile_num(self_obj), dude_tile) <= 5)) then begin
             call Node001();
         end
         add_timer_event(self_obj, game_ticks(random(5, 10) * 10), 1);
@@ -41,8 +41,8 @@ end
 
 procedure talk_p_proc
 begin
-    if (global_var(11) != 0) then begin
-        float_msg(self_obj, message_str(2278, 600), 2);
+    if (global_var(GVAR_REPUTATION_SLAVER) != 0) then begin
+        float_msg(self_obj, mstr(600), FLOAT_MSG_RED);
     end
     else begin
         if (local_var(4) == 1) then begin
@@ -67,63 +67,63 @@ end
 procedure destroy_p_proc
 begin
     if (source_obj == dude_obj) then begin
-        set_global_var(5, global_var(5) + 1);
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(4) > (3 * global_var(5))) or (global_var(2) == 1)) and (global_var(1) == 0)) then begin
-            set_global_var(2, 1);
-            set_global_var(3, 0);
+        set_global_var(GVAR_GOOD_MONSTER, global_var(GVAR_GOOD_MONSTER) + 1);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(4) > (3 * global_var(GVAR_GOOD_MONSTER))) or (global_var(GVAR_CHAMPION_REPUTATION) == 1)) and (global_var(GVAR_CHILDKILLER_REPUTATION) == 0)) then begin
+            set_global_var(GVAR_CHAMPION_REPUTATION, 1);
+            set_global_var(GVAR_BERSERKER_REPUTATION, 0);
         end
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(5) > (2 * global_var(4))) or (global_var(3) == 1))) then begin
-            set_global_var(3, 1);
-            set_global_var(2, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
+            set_global_var(GVAR_BERSERKER_REPUTATION, 1);
+            set_global_var(GVAR_CHAMPION_REPUTATION, 0);
         end
-        set_global_var(0, global_var(0) + -10);
+        set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + -10);
         debug_msg("Player gains " + -10 + " Karma Points.");
-        set_global_var(37, 0);
-        set_global_var(38, 0);
-        set_global_var(39, 0);
-        set_global_var(40, 0);
-        set_global_var(41, 0);
-        set_global_var(42, 0);
-        set_global_var(43, 0);
-        set_global_var(44, 0);
-        set_global_var(45, 0);
-        tmp_gen_rep := global_var(0);
-        if (has_trait(0, dude_obj, 95)) then begin
-            tmp_gen_rep := global_var(0) * 2;
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+        set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+        set_global_var(GVAR_KARMA_DEFENDER, 0);
+        set_global_var(GVAR_KARMA_WANDERER, 0);
+        set_global_var(GVAR_KARMA_BETRAYER, 0);
+        set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+        set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+        tmp_gen_rep := global_var(GVAR_PLAYER_REPUTATION);
+        if (has_trait(TRAIT_PERK, dude_obj, PERK_karma_beacon_perk)) then begin
+            tmp_gen_rep := global_var(GVAR_PLAYER_REPUTATION) * 2;
         end
         if (tmp_gen_rep >= 1000) then begin
-            set_global_var(37, 1);
+            set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
         end
         else begin
             if (tmp_gen_rep >= 750) then begin
-                set_global_var(38, 1);
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
             end
             else begin
                 if (tmp_gen_rep >= 500) then begin
-                    set_global_var(39, 1);
+                    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                 end
                 else begin
                     if (tmp_gen_rep >= 250) then begin
-                        set_global_var(40, 1);
+                        set_global_var(GVAR_KARMA_DEFENDER, 1);
                     end
                     else begin
                         if (tmp_gen_rep > -250) then begin
-                            set_global_var(41, 1);
+                            set_global_var(GVAR_KARMA_WANDERER, 1);
                         end
                         else begin
                             if (tmp_gen_rep > -500) then begin
-                                set_global_var(42, 1);
+                                set_global_var(GVAR_KARMA_BETRAYER, 1);
                             end
                             else begin
                                 if (tmp_gen_rep > -750) then begin
-                                    set_global_var(43, 1);
+                                    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                 end
                                 else begin
                                     if (tmp_gen_rep > -1000) then begin
-                                        set_global_var(44, 1);
+                                        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                     end
                                     else begin
-                                        set_global_var(45, 1);
+                                        set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                     end
                                 end
                             end
@@ -132,7 +132,7 @@ begin
                 end
             end
         end
-        set_global_var(63, global_var(63) + -5);
+        set_global_var(GVAR_TOWN_REP_ABBEY, global_var(GVAR_TOWN_REP_ABBEY) + -5);
         debug_msg("Added " + -5 + " to Town Rep");
     end
 end
@@ -142,11 +142,11 @@ begin
     if (source_obj == dude_obj) then begin
         if (local_var(7) == 1) then begin
             set_local_var(5, 2);
-            set_global_var(63, global_var(63) - 30);
-            attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+            set_global_var(GVAR_TOWN_REP_ABBEY, global_var(GVAR_TOWN_REP_ABBEY) - 30);
+            attack(dude_obj);
         end
         else begin
-            float_msg(self_obj, message_str(2278, 500), 2);
+            float_msg(self_obj, mstr(500), FLOAT_MSG_RED);
             set_local_var(7, 1);
         end
     end
@@ -154,34 +154,34 @@ end
 
 procedure Node001
 begin
-    float_msg(self_obj, message_str(2278, random(200, 207)), 8);
+    float_msg(self_obj, mstr(random(200, 207)), FLOAT_MSG_YELLOW);
 end
 
 procedure critter_p_proc
 begin
-    if ((global_var(63) <= -30) or (local_var(5) == 2) and obj_can_see_obj(self_obj, dude_obj)) then begin
-        attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+    if ((global_var(GVAR_TOWN_REP_ABBEY) <= -30) or (local_var(5) == 2) and obj_can_see_obj(self_obj, dude_obj)) then begin
+        attack(dude_obj);
     end
 end
 
 procedure look_at_p_proc
 begin
     script_overrides;
-    display_msg(message_str(2278, 100));
+    display_mstr(100);
 end
 
 procedure description_p_proc
 begin
     script_overrides;
-    display_msg(message_str(2278, 101));
+    display_mstr(101);
 end
 
 procedure damage_p_proc
 begin
     if (source_obj == dude_obj) then begin
         set_local_var(5, 2);
-        if (global_var(63) > -30) then begin
-            set_global_var(63, -30);
+        if (global_var(GVAR_TOWN_REP_ABBEY) > -30) then begin
+            set_global_var(GVAR_TOWN_REP_ABBEY, -30);
         end
     end
 end
@@ -192,74 +192,73 @@ end
 
 procedure Node010
 begin
-    gsay_reply(2278, 700);
-    giq_option(4, 2278, message_str(2278, 701) + obj_name(dude_obj) + message_str(2278, 702), Node012, 49);
-    giq_option(4, 2278, 703, Node012, 50);
-    giq_option(6, 2278, 704, Node012, 50);
-    giq_option(-3, 2278, message_str(2278, 705) + obj_name(dude_obj) + message_str(2278, 706), Node011, 50);
+    Reply(700);
+    giq_option(4, NAME, mstr(701) + dude_name + mstr(702), Node012, GOOD_REACTION);
+    giq_option(4, NAME, 703, Node012, NEUTRAL_REACTION);
+    giq_option(6, NAME, 704, Node012, NEUTRAL_REACTION);
+    giq_option(-3, NAME, mstr(705) + dude_name + mstr(706), Node011, NEUTRAL_REACTION);
 end
 
 procedure Node011
 begin
-    gsay_reply(2278, 710);
-    giq_option(-3, 2278, 711, Node999, 50);
+    Reply(710);
+    giq_option(-3, NAME, 711, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node012
 begin
-    gsay_reply(2278, 720);
-    giq_option(4, 2278, 721, Node013, 50);
-    giq_option(4, 2278, 722, Node013, 50);
-    giq_option(4, 2278, 723, Node016, 50);
-    giq_option(4, 2278, 724, Node016, 50);
-    giq_option(4, 2278, 725, Node999, 50);
+    Reply(720);
+    giq_option(4, NAME, 721, Node013, NEUTRAL_REACTION);
+    giq_option(4, NAME, 722, Node013, NEUTRAL_REACTION);
+    giq_option(4, NAME, 723, Node016, NEUTRAL_REACTION);
+    giq_option(4, NAME, 724, Node016, NEUTRAL_REACTION);
+    giq_option(4, NAME, 725, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node013
 begin
-    gsay_reply(2278, 730);
-    giq_option(4, 2278, 731, Node014, 50);
-    giq_option(4, 2278, 732, Node015, 50);
-    giq_option(4, 2278, 733, Node016, 50);
-    giq_option(4, 2278, 734, Node999, 50);
+    Reply(730);
+    giq_option(4, NAME, 731, Node014, NEUTRAL_REACTION);
+    giq_option(4, NAME, 732, Node015, NEUTRAL_REACTION);
+    giq_option(4, NAME, 733, Node016, NEUTRAL_REACTION);
+    giq_option(4, NAME, 734, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node014
 begin
-    gsay_reply(2278, 760);
-    giq_option(4, 2278, 732, Node015, 50);
-    giq_option(4, 2278, 733, Node016, 50);
-    giq_option(4, 2278, 761, Node999, 50);
+    Reply(760);
+    giq_option(4, NAME, 732, Node015, NEUTRAL_REACTION);
+    giq_option(4, NAME, 733, Node016, NEUTRAL_REACTION);
+    giq_option(4, NAME, 761, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node015
 begin
-    gsay_reply(2278, 770);
-    giq_option(4, 2278, 771, Node016, 50);
-    giq_option(4, 2278, 731, Node014, 50);
-    giq_option(4, 2278, 772, Node999, 50);
+    Reply(770);
+    giq_option(4, NAME, 771, Node016, NEUTRAL_REACTION);
+    giq_option(4, NAME, 731, Node014, NEUTRAL_REACTION);
+    giq_option(4, NAME, 772, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node016
 begin
-    gsay_reply(2278, 740);
-    giq_option(4, 2278, 741, Node017, 50);
-    giq_option(4, 2278, 742, Node017, 50);
-    giq_option(4, 2278, 743, Node999, 50);
+    Reply(740);
+    giq_option(4, NAME, 741, Node017, NEUTRAL_REACTION);
+    giq_option(4, NAME, 742, Node017, NEUTRAL_REACTION);
+    giq_option(4, NAME, 743, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node017
 begin
-    gsay_reply(2278, 750);
-    giq_option(4, 2278, 751, Node018, 50);
-    giq_option(6, 2278, 752, Node018, 50);
-    giq_option(4, 2278, 743, Node999, 50);
+    Reply(750);
+    giq_option(4, NAME, 751, Node018, NEUTRAL_REACTION);
+    giq_option(6, NAME, 752, Node018, NEUTRAL_REACTION);
+    giq_option(4, NAME, 743, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node018
 begin
-    gsay_reply(2278, 755);
-    giq_option(4, 2278, 756, Node999, 50);
-    giq_option(4, 2278, 757, Node999, 50);
+    Reply(755);
+    giq_option(4, NAME, 756, Node999, NEUTRAL_REACTION);
+    giq_option(4, NAME, 757, Node999, NEUTRAL_REACTION);
 end
-
