diff --git a/Mods/MegaMod/Original/ABBEY.ssl b/Mods/MegaMod/ReDefine/ABBEY.ssl
index 757131c..14f4858 100644
--- a/Mods/MegaMod/Original/ABBEY.ssl
+++ b/Mods/MegaMod/ReDefine/ABBEY.ssl
@@ -9,13 +9,13 @@ end
 
 procedure map_enter_p_proc
 begin
-    if (metarule(14, 0)) then begin
-        display_msg(message_str(1435, 100));
+    if (map_first_run) then begin
+        display_mstr(100);
     end
     else begin
-        display_msg(message_str(1435, 102));
+        display_mstr(102);
     end
-    if (elevation(dude_obj) == 0) then begin
+    if (dude_elevation == 0) then begin
         if ((get_month >= 3) and (get_month < 5)) then begin
             if ((game_time_hour >= 600) and (game_time_hour < (600 + 100))) then begin
                 set_light_level(game_time_hour - 600 + 40);
@@ -97,7 +97,7 @@ end
 
 procedure map_update_p_proc
 begin
-    if (elevation(dude_obj) == 0) then begin
+    if (dude_elevation == 0) then begin
         if ((get_month >= 3) and (get_month < 5)) then begin
             if ((game_time_hour >= 600) and (game_time_hour < (600 + 100))) then begin
                 set_light_level(game_time_hour - 600 + 40);
@@ -179,4 +179,3 @@ begin
         set_light_level(50);
     end
 end
-
