diff --git a/Mods/MegaMod/Original/ABBEY1.ssl b/Mods/MegaMod/ReDefine/ABBEY1.ssl
index 01804f0..3e5aad1 100644
--- a/Mods/MegaMod/Original/ABBEY1.ssl
+++ b/Mods/MegaMod/ReDefine/ABBEY1.ssl
@@ -11,8 +11,8 @@ end
 
 procedure map_enter_p_proc
 begin
-    if (metarule(14, 0)) then begin
-        display_msg(message_str(2247, 100));
+    if (map_first_run) then begin
+        display_mstr(100);
     end
     if (elevation(dude_obj) == 0) then begin
         if ((get_month >= 3) and (get_month < 5)) then begin
@@ -178,4 +178,3 @@ begin
         set_light_level(50);
     end
 end
-
