diff --git a/Mods/MegaMod/Original/ABGRAVE1.ssl b/Mods/MegaMod/ReDefine/ABGRAVE1.ssl
index f76d14f..473cbac 100644
--- a/Mods/MegaMod/Original/ABGRAVE1.ssl
+++ b/Mods/MegaMod/ReDefine/ABGRAVE1.ssl
@@ -19,13 +19,13 @@ end
 procedure look_at_p_proc
 begin
     script_overrides;
-    display_msg(message_str(2254, 100));
+    display_mstr(100);
 end
 
 procedure description_p_proc
 begin
     script_overrides;
-    display_msg(message_str(2254, 101) + message_str(2254, local_var(0)));
+    display_msg(mstr(101) + mstr(local_var(0)));
 end
 
 procedure num_to_bit(variable arg0)
@@ -65,8 +65,7 @@ end
 
 procedure map_enter_p_proc
 begin
-    if (metarule(14, 0)) then begin
+    if (map_first_run) then begin
         call give_headstone_name();
     end
 end
-
