diff --git a/Mods/MegaMod/Original/ABNDCITY.ssl b/Mods/MegaMod/ReDefine/ABNDCITY.ssl
index c47baa3..6f8aa18 100644
--- a/Mods/MegaMod/Original/ABNDCITY.ssl
+++ b/Mods/MegaMod/ReDefine/ABNDCITY.ssl
@@ -138,110 +138,110 @@ end
 procedure map_enter_p_proc
 begin
     debug_msg("Check_Create_Car(" + 11508 + "," + 0 + ")");
-    if (not(metarule(22, 0))) then begin
-        if (global_var(18) == 1) then begin
-            debug_msg("car_current_town == " + metarule(30, 0) + " / map_get_load_area == " + metarule3(111, 0, 0, 0));
-            if ((metarule(30, 0) == metarule3(111, 0, 0, 0)) or ((26 <= metarule(30, 0)) and (metarule(30, 0) <= 29) and (metarule3(111, 0, 0, 0) == -1))) then begin
+    if (not(is_loading_game)) then begin
+        if (global_var(GVAR_PLAYER_GOT_CAR) == 1) then begin
+            debug_msg("car_current_town == " + car_current_town + " / map_get_load_area == " + map_get_load_area);
+            if ((car_current_town == map_get_load_area) or ((26 <= car_current_town) and (car_current_town <= 29) and (map_get_load_area == -1))) then begin
                 debug_msg("Place_Car(" + 11508 + "," + 0 + ")");
-                debug_msg("   GVAR_CAR_PLACED_TILE == " + global_var(633));
+                debug_msg("   GVAR_CAR_PLACED_TILE == " + global_var(GVAR_CAR_PLACED_TILE));
                 debug_msg("   Car_At_Loc == " + (tile_contains_pid_obj(11508, 0, 33555441) != 0));
-                set_global_var(726, cur_map_index);
+                set_global_var(GVAR_NOWA_KARMA2, cur_map_index);
                 debug_msg("Car now at Ghost town");
-                debug_msg("Car Current map = " + global_var(726));
-                debug_msg("Vertibird current map = " + global_var(724));
-                if ((global_var(633) <= 0) or (11508 == global_var(633))) then begin
-                    set_global_var(633, 11508);
+                debug_msg("Car Current map = " + global_var(GVAR_NOWA_KARMA2));
+                debug_msg("Vertibird current map = " + global_var(GVAR_CAR_CURRENT_MAP));
+                if ((global_var(GVAR_CAR_PLACED_TILE) <= 0) or (11508 == global_var(GVAR_CAR_PLACED_TILE))) then begin
+                    set_global_var(GVAR_CAR_PLACED_TILE, 11508);
                     if (not(tile_contains_pid_obj(11508, 0, 33555441) != 0)) then begin
-                        Scenery_Creation := create_object_sid(33555441, 11508, 0, 304);
+                        Scenery_Creation := create_object_sid(33555441, 11508, 0, SCRIPT_ZSDRVCAR);
                         Scenery_Creation_Hex := 11508;
                         Scenery_Creation_Count := 0;
                         while (Scenery_Creation_Count < 2) do begin
                             Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 1, 1);
-                            Scenery_Creation := create_object_sid(33554499, Scenery_Creation_Hex, 0, -1);
+                            Scenery_Creation := create_object(33554499, Scenery_Creation_Hex, 0);
                             Scenery_Creation_Count := Scenery_Creation_Count + 1;
                         end
                         Scenery_Creation_Count := 0;
                         while (Scenery_Creation_Count < 2) do begin
                             Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 2, 1);
-                            Scenery_Creation := create_object_sid(33554499, Scenery_Creation_Hex, 0, -1);
+                            Scenery_Creation := create_object(33554499, Scenery_Creation_Hex, 0);
                             Scenery_Creation_Count := Scenery_Creation_Count + 1;
                         end
                         Scenery_Creation_Count := 0;
                         while (Scenery_Creation_Count < 2) do begin
                             Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 3, 1);
-                            Scenery_Creation := create_object_sid(33554499, Scenery_Creation_Hex, 0, -1);
+                            Scenery_Creation := create_object(33554499, Scenery_Creation_Hex, 0);
                             Scenery_Creation_Count := Scenery_Creation_Count + 1;
                         end
                         Scenery_Creation_Count := 0;
                         while (Scenery_Creation_Count < 2) do begin
                             Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 4, 1);
-                            Scenery_Creation := create_object_sid(33554499, Scenery_Creation_Hex, 0, -1);
+                            Scenery_Creation := create_object(33554499, Scenery_Creation_Hex, 0);
                             Scenery_Creation_Count := Scenery_Creation_Count + 1;
                         end
                         Scenery_Creation_Count := 0;
                         while (Scenery_Creation_Count < 1) do begin
                             Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 5, 1);
-                            Scenery_Creation := create_object_sid(33554499, Scenery_Creation_Hex, 0, -1);
+                            Scenery_Creation := create_object(33554499, Scenery_Creation_Hex, 0);
                             Scenery_Creation_Count := Scenery_Creation_Count + 1;
                         end
                         Scenery_Creation_Count := 0;
                         while (Scenery_Creation_Count < 1) do begin
                             Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 4, 1);
-                            Scenery_Creation := create_object_sid(33554499, Scenery_Creation_Hex, 0, -1);
+                            Scenery_Creation := create_object(33554499, Scenery_Creation_Hex, 0);
                             Scenery_Creation_Count := Scenery_Creation_Count + 1;
                         end
                         Scenery_Creation_Count := 0;
                         while (Scenery_Creation_Count < 1) do begin
                             Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 5, 1);
-                            Scenery_Creation := create_object_sid(33554499, Scenery_Creation_Hex, 0, -1);
+                            Scenery_Creation := create_object(33554499, Scenery_Creation_Hex, 0);
                             Scenery_Creation_Count := Scenery_Creation_Count + 1;
                         end
                         Scenery_Creation_Count := 0;
                         while (Scenery_Creation_Count < 1) do begin
                             Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 4, 1);
-                            Scenery_Creation := create_object_sid(33554499, Scenery_Creation_Hex, 0, -1);
+                            Scenery_Creation := create_object(33554499, Scenery_Creation_Hex, 0);
                             Scenery_Creation_Count := Scenery_Creation_Count + 1;
                         end
                         Scenery_Creation_Count := 0;
                         while (Scenery_Creation_Count < 1) do begin
                             Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 5, 1);
-                            Scenery_Creation := create_object_sid(33554499, Scenery_Creation_Hex, 0, -1);
+                            Scenery_Creation := create_object(33554499, Scenery_Creation_Hex, 0);
                             Scenery_Creation_Count := Scenery_Creation_Count + 1;
                         end
                         Scenery_Creation_Count := 0;
                         while (Scenery_Creation_Count < 1) do begin
                             Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 4, 1);
-                            Scenery_Creation := create_object_sid(33554499, Scenery_Creation_Hex, 0, -1);
+                            Scenery_Creation := create_object(33554499, Scenery_Creation_Hex, 0);
                             Scenery_Creation_Count := Scenery_Creation_Count + 1;
                         end
                         Scenery_Creation_Count := 0;
                         while (Scenery_Creation_Count < 1) do begin
                             Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 0, 1);
-                            Scenery_Creation := create_object_sid(33554499, Scenery_Creation_Hex, 0, -1);
+                            Scenery_Creation := create_object(33554499, Scenery_Creation_Hex, 0);
                             Scenery_Creation_Count := Scenery_Creation_Count + 1;
                         end
                         Scenery_Creation_Count := 0;
                         while (Scenery_Creation_Count < 1) do begin
                             Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 5, 1);
-                            Scenery_Creation := create_object_sid(33554499, Scenery_Creation_Hex, 0, -1);
+                            Scenery_Creation := create_object(33554499, Scenery_Creation_Hex, 0);
                             Scenery_Creation_Count := Scenery_Creation_Count + 1;
                         end
                         Scenery_Creation_Count := 0;
                         while (Scenery_Creation_Count < 2) do begin
                             Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 0, 1);
-                            Scenery_Creation := create_object_sid(33554499, Scenery_Creation_Hex, 0, -1);
+                            Scenery_Creation := create_object(33554499, Scenery_Creation_Hex, 0);
                             Scenery_Creation_Count := Scenery_Creation_Count + 1;
                         end
                         Scenery_Creation_Count := 0;
                         while (Scenery_Creation_Count < 1) do begin
                             Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 1, 1);
-                            Scenery_Creation := create_object_sid(33554499, Scenery_Creation_Hex, 0, -1);
+                            Scenery_Creation := create_object(33554499, Scenery_Creation_Hex, 0);
                             Scenery_Creation_Count := Scenery_Creation_Count + 1;
                         end
                         Scenery_Creation_Count := 0;
                         while (Scenery_Creation_Count < 1) do begin
                             Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 2, 1);
-                            Scenery_Creation := create_object_sid(33554499, Scenery_Creation_Hex, 0, -1);
+                            Scenery_Creation := create_object(33554499, Scenery_Creation_Hex, 0);
                             Scenery_Creation_Count := Scenery_Creation_Count + 1;
                         end
                     end
@@ -254,8 +254,8 @@ begin
                             Scenery_Creation_Ptr := party_member_obj(455);
                         end
                         else begin
-                            if (global_var(18) == 0) then begin
-                                Scenery_Creation_Ptr := create_object_sid(455, 0, 0, 920);
+                            if (global_var(GVAR_PLAYER_GOT_CAR) == 0) then begin
+                                Scenery_Creation_Ptr := create_object_sid(455, 0, 0, SCRIPT_ZICRTRNK);
                             end
                         end
                         if (Scenery_Creation_Ptr != 0) then begin
@@ -264,13 +264,13 @@ begin
                             Scenery_Creation_Count := 0;
                             while (Scenery_Creation_Count < 1) do begin
                                 Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 1, 1);
-                                Scenery_Creation := create_object_sid(33554499, Scenery_Creation_Hex, 0, -1);
+                                Scenery_Creation := create_object(33554499, Scenery_Creation_Hex, 0);
                                 Scenery_Creation_Count := Scenery_Creation_Count + 1;
                             end
                             Scenery_Creation_Count := 0;
                             while (Scenery_Creation_Count < 1) do begin
                                 Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 2, 1);
-                                Scenery_Creation := create_object_sid(33554499, Scenery_Creation_Hex, 0, -1);
+                                Scenery_Creation := create_object(33554499, Scenery_Creation_Hex, 0);
                                 Scenery_Creation_Count := Scenery_Creation_Count + 1;
                             end
                         end
@@ -426,7 +426,7 @@ begin
     else begin
         debug_msg("sorry, can't place the car, the game's loading");
     end
-    if (elevation(dude_obj) == 1) then begin
+    if (dude_elevation == 1) then begin
         set_light_level(40);
     end
     else begin
@@ -507,12 +507,12 @@ begin
             end
         end
     end
-    set_global_var(27, 0);
+    set_global_var(GVAR_LOAD_MAP_INDEX, 0);
 end
 
 procedure map_update_p_proc
 begin
-    if (elevation(dude_obj) == 1) then begin
+    if (dude_elevation == 1) then begin
         set_light_level(40);
     end
     else begin
@@ -594,4 +594,3 @@ begin
         end
     end
 end
-
