diff --git a/Mods/MegaMod/Original/ACKLINT.ssl b/Mods/MegaMod/ReDefine/ACKLINT.ssl
index 9fa5d58..c841d1a 100644
--- a/Mods/MegaMod/Original/ACKLINT.ssl
+++ b/Mods/MegaMod/ReDefine/ACKLINT.ssl
@@ -96,48 +96,48 @@ procedure start;
 
 procedure EvaluateKarma
 begin
-    set_global_var(37, 0);
-    set_global_var(38, 0);
-    set_global_var(39, 0);
-    set_global_var(40, 0);
-    set_global_var(41, 0);
-    set_global_var(42, 0);
-    set_global_var(43, 0);
-    set_global_var(44, 0);
-    set_global_var(45, 0);
-    if (global_var(0) >= 1000) then begin
-        set_global_var(37, 1);
+    set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+    set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+    set_global_var(GVAR_KARMA_DEFENDER, 0);
+    set_global_var(GVAR_KARMA_WANDERER, 0);
+    set_global_var(GVAR_KARMA_BETRAYER, 0);
+    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+    set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+    set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+    if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
     end
     else begin
-        if (global_var(0) >= 750) then begin
-            set_global_var(38, 1);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+            set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
         end
         else begin
-            if (global_var(0) >= 500) then begin
-                set_global_var(39, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
             end
             else begin
-                if (global_var(0) >= 250) then begin
-                    set_global_var(40, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                    set_global_var(GVAR_KARMA_DEFENDER, 1);
                 end
                 else begin
-                    if (global_var(0) > -250) then begin
-                        set_global_var(41, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                        set_global_var(GVAR_KARMA_WANDERER, 1);
                     end
                     else begin
-                        if (global_var(0) > -500) then begin
-                            set_global_var(42, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                            set_global_var(GVAR_KARMA_BETRAYER, 1);
                         end
                         else begin
-                            if (global_var(0) > -750) then begin
-                                set_global_var(43, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                             end
                             else begin
-                                if (global_var(0) > -1000) then begin
-                                    set_global_var(44, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                    set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                 end
                                 else begin
-                                    set_global_var(45, 1);
+                                    set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                 end
                             end
                         end
@@ -154,27 +154,27 @@ begin
     variable LVar1 := 0;
     if ((local_var(7) == 2) and obj_can_see_obj(self_obj, dude_obj)) then begin
         set_local_var(7, 1);
-        attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+        attack(dude_obj);
     end
-    if ((global_var(7) == 1) and obj_can_see_obj(self_obj, dude_obj)) then begin
-        attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+    if ((global_var(GVAR_ENEMY_ARROYO) == 1) and obj_can_see_obj(self_obj, dude_obj)) then begin
+        attack(dude_obj);
     end
-    if (global_var(643) == 1) then begin
-        LVar0 := tile_num_in_direction(tile_num(dude_obj), random(1, 4), random(1, 3));
-        if (tile_distance(tile_num(self_obj), tile_num(dude_obj)) > 5) then begin
+    if (global_var(GVAR_ACKLINT_JOIN) == 1) then begin
+        LVar0 := tile_num_in_direction(dude_tile, random(1, 4), random(1, 3));
+        if (tile_distance(tile_num(self_obj), dude_tile) > 5) then begin
             animate_move_obj_to_tile(self_obj, LVar0, 1);
         end
-        if (cur_map_index == 3) then begin
+        if (CUR_MAP_ARCAVES) then begin
             LVar1 := tile_contains_pid_obj(19928, 2, 33555366);
             if (tile_distance_objs(LVar1, self_obj) < 6) then begin
                 party_remove(self_obj);
-                set_global_var(643, 0);
-                float_msg(self_obj, message_str(751, 792), 0);
+                set_global_var(GVAR_ACKLINT_JOIN, 0);
+                float_msg(self_obj, mstr(792), FLOAT_MSG_NORMAL);
             end
         end
     end
-    if ((get_critter_stat(dude_obj, 34) == 1) and (local_var(5) == 0) and (global_var(10) != 2)) then begin
-        LVar0 := tile_num_in_direction(tile_num(dude_obj), random(1, 5), random(1, 3));
+    if ((dude_gender == 1) and (local_var(5) == 0) and (global_var(GVAR_START_ARROYO_TRIAL) != 2)) then begin
+        LVar0 := tile_num_in_direction(dude_tile, random(1, 5), random(1, 3));
         animate_move_obj_to_tile(self_obj, LVar0, 1);
         dialogue_system_enter;
     end
@@ -189,42 +189,42 @@ end
 
 procedure destroy_p_proc
 begin
-    set_global_var(676, 1);
+    set_global_var(GVAR_RESERVED, 1);
     if (source_obj == dude_obj) then begin
-        set_global_var(5, global_var(5) + 1);
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(4) > (3 * global_var(5))) or (global_var(2) == 1)) and (global_var(1) == 0)) then begin
-            set_global_var(2, 1);
-            set_global_var(3, 0);
+        set_global_var(GVAR_GOOD_MONSTER, global_var(GVAR_GOOD_MONSTER) + 1);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(4) > (3 * global_var(GVAR_GOOD_MONSTER))) or (global_var(GVAR_CHAMPION_REPUTATION) == 1)) and (global_var(GVAR_CHILDKILLER_REPUTATION) == 0)) then begin
+            set_global_var(GVAR_CHAMPION_REPUTATION, 1);
+            set_global_var(GVAR_BERSERKER_REPUTATION, 0);
         end
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(5) > (2 * global_var(4))) or (global_var(3) == 1))) then begin
-            set_global_var(3, 1);
-            set_global_var(2, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
+            set_global_var(GVAR_BERSERKER_REPUTATION, 1);
+            set_global_var(GVAR_CHAMPION_REPUTATION, 0);
         end
-        set_global_var(0, global_var(0) - 15);
+        set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) - 15);
         call EvaluateKarma();
-        set_global_var(47, global_var(47) - 50);
-        set_global_var(7, 1);
+        set_global_var(GVAR_TOWN_REP_ARROYO, global_var(GVAR_TOWN_REP_ARROYO) - 50);
+        set_global_var(GVAR_ENEMY_ARROYO, 1);
     end
 end
 
 procedure map_enter_p_proc
 begin
     Only_Once := 0;
-    if (global_var(676) == 1) then begin
+    if (global_var(GVAR_RESERVED) == 1) then begin
         set_obj_visibility(self_obj, 1);
     end
     else begin
-        if ((cur_map_index == 3) and (global_var(10) == 2)) then begin
+        if ((CUR_MAP_ARCAVES) and (global_var(GVAR_START_ARROYO_TRIAL) == 2)) then begin
             set_obj_visibility(self_obj, 1);
         end
         else begin
-            if ((cur_map_index == 126) and (global_var(10) == 2)) then begin
+            if ((CUR_MAP_ARTEMPLE) and (global_var(GVAR_START_ARROYO_TRIAL) == 2)) then begin
                 set_obj_visibility(self_obj, 1);
             end
             else begin
                 if (party_member_obj(obj_pid(self_obj)) != 1) then begin
-                    critter_add_trait(self_obj, 1, 6, 1);
-                    critter_add_trait(self_obj, 1, 5, 1);
+                    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 1);
+                    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_ARROYO_WARRIOR);
                 end
             end
         end
@@ -234,20 +234,20 @@ end
 procedure look_at_p_proc
 begin
     script_overrides;
-    display_msg(message_str(751, 100));
+    display_mstr(100);
 end
 
 procedure description_p_proc
 begin
     script_overrides;
-    display_msg(message_str(751, 101));
+    display_mstr(101);
 end
 
 procedure talk_p_proc
 begin
-    if (global_var(10) != 2) then begin
-        if (get_critter_stat(dude_obj, 34) == 0) then begin
-            if (global_var(651) == 0) then begin
+    if (global_var(GVAR_START_ARROYO_TRIAL) != 2) then begin
+        if (dude_gender == 0) then begin
+            if (global_var(GVAR_ARTEMPLE_WITHHELP) == 0) then begin
                 if (local_var(5) == 0) then begin
                     start_gdialog(751, self_obj, 4, -1, -1);
                     gsay_start;
@@ -272,12 +272,12 @@ begin
             end
         end
         else begin
-            if (global_var(651) == 0) then begin
-                if (global_var(643) == 1) then begin
+            if (global_var(GVAR_ARTEMPLE_WITHHELP) == 0) then begin
+                if (global_var(GVAR_ACKLINT_JOIN) == 1) then begin
                     call Node087();
                 end
                 else begin
-                    if (global_var(642) == 0) then begin
+                    if (global_var(GVAR_ACKLINT_PERMIT) == 0) then begin
                         start_gdialog(751, self_obj, 4, -1, -1);
                         gsay_start;
                         call Node070();
@@ -303,7 +303,7 @@ begin
         end
     end
     else begin
-        if ((global_var(639) == 3) or (global_var(640) == 3)) then begin
+        if ((global_var(GVAR_ARROYO_MEMBER) == 3) or (global_var(GVAR_ARTEMPLE_PROVE) == 3)) then begin
             start_gdialog(751, self_obj, 4, -1, -1);
             gsay_start;
             call Node050();
@@ -311,7 +311,7 @@ begin
             end_dialogue;
         end
         else begin
-            if (global_var(640) == 1) then begin
+            if (global_var(GVAR_ARTEMPLE_PROVE) == 1) then begin
                 start_gdialog(751, self_obj, 4, -1, -1);
                 gsay_start;
                 call Node044();
@@ -332,21 +332,21 @@ end
 procedure damage_p_proc
 begin
     if (source_obj == dude_obj) then begin
-        if (party_member_obj(obj_pid(self_obj)) != 0) then begin
+        if (obj_in_party(self_obj)) then begin
             party_remove(self_obj);
         end
-        if (global_var(643) == 1) then begin
+        if (global_var(GVAR_ACKLINT_JOIN) == 1) then begin
             set_local_var(8, local_var(8) + 1);
             if (local_var(8) >= 2) then begin
                 party_remove(self_obj);
-                set_global_var(643, 0);
+                set_global_var(GVAR_ACKLINT_JOIN, 0);
                 set_local_var(7, 2);
-                set_global_var(7, 1);
+                set_global_var(GVAR_ENEMY_ARROYO, 1);
             end
         end
         else begin
             set_local_var(7, 2);
-            set_global_var(7, 1);
+            set_global_var(GVAR_ENEMY_ARROYO, 1);
         end
     end
 end
@@ -358,390 +358,390 @@ end
 procedure Node001
 begin
     set_local_var(5, local_var(5) + 1);
-    gsay_reply(751, 202);
-    giq_option(-1, 751, 204, Node002, 50);
-    if (get_critter_stat(dude_obj, 4) <= 4) then begin
-        giq_option(2, 751, 206, Node002, 50);
+    Reply(202);
+    giq_option(-1, NAME, 204, Node002, NEUTRAL_REACTION);
+    if (dude_iq <= 4) then begin
+        giq_option(2, NAME, 206, Node002, NEUTRAL_REACTION);
     end
-    if (get_critter_stat(dude_obj, 4) <= 4) then begin
-        giq_option(2, 751, 208, Node002, 50);
+    if (dude_iq <= 4) then begin
+        giq_option(2, NAME, 208, Node002, NEUTRAL_REACTION);
     end
-    if (has_skill(dude_obj, 14) >= 50) then begin
-        if (get_critter_stat(dude_obj, 3) > (local_var(5) * 1)) then begin
-            giq_option(6, 751, 210, Node002, 50);
+    if (dude_skill(SKILL_SPEECH) >= 50) then begin
+        if (dude_charisma > (local_var(5) * 1)) then begin
+            giq_option(6, NAME, 210, Node002, NEUTRAL_REACTION);
         end
         else begin
-            giq_option(6, 751, 210, Node018, 50);
+            giq_option(6, NAME, 210, Node018, NEUTRAL_REACTION);
         end
     end
-    if (get_critter_stat(dude_obj, 3) > (local_var(6) * 1)) then begin
-        giq_option(5, 751, 212, Node004, 50);
+    if (dude_charisma > (local_var(6) * 1)) then begin
+        giq_option(5, NAME, 212, Node004, NEUTRAL_REACTION);
     end
     else begin
-        giq_option(5, 751, 212, Node009fool, 50);
+        giq_option(5, NAME, 212, Node009fool, NEUTRAL_REACTION);
     end
-    if (get_critter_stat(dude_obj, 35) > 10) then begin
-        giq_option(2, 751, 214, Node005fool, 50);
+    if (get_critter_stat(dude_obj, STAT_current_hp) > 10) then begin
+        giq_option(2, NAME, 214, Node005fool, NEUTRAL_REACTION);
     end
     else begin
-        giq_option(2, 751, 214, Node019, 50);
+        giq_option(2, NAME, 214, Node019, NEUTRAL_REACTION);
     end
-    giq_option(1, 751, 216, Node999, 50);
+    giq_option(1, NAME, 216, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node002
 begin
-    gsay_reply(751, 218);
-    if (get_critter_stat(dude_obj, 35) > 10) then begin
-        giq_option(-1, 751, 220, Node005fool, 50);
+    Reply(218);
+    if (get_critter_stat(dude_obj, STAT_current_hp) > 10) then begin
+        giq_option(-1, NAME, 220, Node005fool, NEUTRAL_REACTION);
     end
     else begin
-        giq_option(-1, 751, 220, Node019, 50);
+        giq_option(-1, NAME, 220, Node019, NEUTRAL_REACTION);
     end
-    if (get_critter_stat(dude_obj, 4) <= 4) then begin
-        giq_option(2, 751, 222, Node018fool, 50);
+    if (dude_iq <= 4) then begin
+        giq_option(2, NAME, 222, Node018fool, NEUTRAL_REACTION);
     end
-    if (get_critter_stat(dude_obj, 4) <= 4) then begin
-        giq_option(2, 751, 224, Node013fool, 50);
+    if (dude_iq <= 4) then begin
+        giq_option(2, NAME, 224, Node013fool, NEUTRAL_REACTION);
     end
-    giq_option(5, 751, 226, Node010fool, 50);
-    if (get_critter_stat(dude_obj, 3) > (local_var(6) * 1)) then begin
-        giq_option(5, 751, 228, Node005fool, 50);
+    giq_option(5, NAME, 226, Node010fool, NEUTRAL_REACTION);
+    if (dude_charisma > (local_var(6) * 1)) then begin
+        giq_option(5, NAME, 228, Node005fool, NEUTRAL_REACTION);
     end
     else begin
-        giq_option(5, 751, 228, Node013fool, 50);
+        giq_option(5, NAME, 228, Node013fool, NEUTRAL_REACTION);
     end
-    if (has_skill(dude_obj, 14) >= 50) then begin
-        if (get_critter_stat(dude_obj, 3) > (local_var(6) * 1)) then begin
-            giq_option(6, 751, 230, Node005fool, 50);
+    if (dude_skill(SKILL_SPEECH) >= 50) then begin
+        if (dude_charisma > (local_var(6) * 1)) then begin
+            giq_option(6, NAME, 230, Node005fool, NEUTRAL_REACTION);
         end
         else begin
-            giq_option(6, 751, 230, Node013fool, 50);
+            giq_option(6, NAME, 230, Node013fool, NEUTRAL_REACTION);
         end
     end
-    giq_option(1, 751, 232, Node999, 50);
+    giq_option(1, NAME, 232, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node003
 begin
     set_local_var(5, local_var(5) + 1);
-    gsay_reply(751, 234);
-    giq_option(-1, 751, 236, Node013fool, 50);
-    if (get_critter_stat(dude_obj, 4) <= 4) then begin
-        if (get_critter_stat(dude_obj, 3) > (local_var(6) * 1)) then begin
-            giq_option(2, 751, 238, Node002fool, 50);
+    Reply(234);
+    giq_option(-1, NAME, 236, Node013fool, NEUTRAL_REACTION);
+    if (dude_iq <= 4) then begin
+        if (dude_charisma > (local_var(6) * 1)) then begin
+            giq_option(2, NAME, 238, Node002fool, NEUTRAL_REACTION);
         end
         else begin
-            giq_option(2, 751, 238, Node013fool, 50);
+            giq_option(2, NAME, 238, Node013fool, NEUTRAL_REACTION);
         end
     end
-    if (get_critter_stat(dude_obj, 3) > (local_var(6) * 1)) then begin
-        giq_option(5, 751, 240, Node004, 50);
+    if (dude_charisma > (local_var(6) * 1)) then begin
+        giq_option(5, NAME, 240, Node004, NEUTRAL_REACTION);
     end
     else begin
-        giq_option(5, 751, 240, Node009fool, 50);
+        giq_option(5, NAME, 240, Node009fool, NEUTRAL_REACTION);
     end
-    if (get_critter_stat(dude_obj, 35) < get_critter_stat(dude_obj, 7)) then begin
-        if (get_critter_stat(dude_obj, 35) <= 10) then begin
-            giq_option(2, 751, 242, Node019, 50);
+    if (get_critter_stat(dude_obj, STAT_current_hp) < get_critter_stat(dude_obj, STAT_max_hp)) then begin
+        if (get_critter_stat(dude_obj, STAT_current_hp) <= 10) then begin
+            giq_option(2, NAME, 242, Node019, NEUTRAL_REACTION);
         end
         else begin
             if (local_var(5) >= 20) then begin
-                giq_option(2, 751, 242, Node019, 50);
+                giq_option(2, NAME, 242, Node019, NEUTRAL_REACTION);
             end
             else begin
-                giq_option(2, 751, 242, Node006fool, 50);
+                giq_option(2, NAME, 242, Node006fool, NEUTRAL_REACTION);
             end
         end
     end
     else begin
-        giq_option(2, 751, 242, Node007fool, 50);
+        giq_option(2, NAME, 242, Node007fool, NEUTRAL_REACTION);
     end
-    if (get_critter_stat(dude_obj, 35) < get_critter_stat(dude_obj, 7)) then begin
-        if (get_critter_stat(dude_obj, 35) <= 10) then begin
-            giq_option(5, 751, 244, Node019, 50);
+    if (get_critter_stat(dude_obj, STAT_current_hp) < get_critter_stat(dude_obj, STAT_max_hp)) then begin
+        if (get_critter_stat(dude_obj, STAT_current_hp) <= 10) then begin
+            giq_option(5, NAME, 244, Node019, NEUTRAL_REACTION);
         end
         else begin
             if (local_var(5) >= 20) then begin
-                giq_option(5, 751, 244, Node019, 50);
+                giq_option(5, NAME, 244, Node019, NEUTRAL_REACTION);
             end
             else begin
-                giq_option(5, 751, 244, Node006fool, 50);
+                giq_option(5, NAME, 244, Node006fool, NEUTRAL_REACTION);
             end
         end
     end
     else begin
-        giq_option(5, 751, 244, Node007fool, 50);
+        giq_option(5, NAME, 244, Node007fool, NEUTRAL_REACTION);
     end
-    giq_option(6, 751, 246, Node006fool, 50);
-    giq_option(2, 751, 248, Node999, 50);
+    giq_option(6, NAME, 246, Node006fool, NEUTRAL_REACTION);
+    giq_option(2, NAME, 248, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node004
 begin
-    gsay_reply(751, 250);
-    giq_option(5, 751, 252, Node010fool, 50);
-    giq_option(5, 751, 254, Node012fool, 50);
-    giq_option(6, 751, 256, Node014fool, 50);
-    giq_option(6, 751, 258, Node002fool, 50);
-    if (get_critter_stat(dude_obj, 3) > (local_var(6) * 1)) then begin
-        giq_option(6, 751, 260, Node005fool, 50);
+    Reply(250);
+    giq_option(5, NAME, 252, Node010fool, NEUTRAL_REACTION);
+    giq_option(5, NAME, 254, Node012fool, NEUTRAL_REACTION);
+    giq_option(6, NAME, 256, Node014fool, NEUTRAL_REACTION);
+    giq_option(6, NAME, 258, Node002fool, NEUTRAL_REACTION);
+    if (dude_charisma > (local_var(6) * 1)) then begin
+        giq_option(6, NAME, 260, Node005fool, NEUTRAL_REACTION);
     end
     else begin
-        giq_option(6, 751, 260, Node013fool, 50);
+        giq_option(6, NAME, 260, Node013fool, NEUTRAL_REACTION);
     end
-    if (get_critter_stat(dude_obj, 3) > (local_var(6) * 1)) then begin
-        giq_option(8, 751, 262, Node005fool, 50);
+    if (dude_charisma > (local_var(6) * 1)) then begin
+        giq_option(8, NAME, 262, Node005fool, NEUTRAL_REACTION);
     end
     else begin
-        giq_option(8, 751, 262, Node013fool, 50);
+        giq_option(8, NAME, 262, Node013fool, NEUTRAL_REACTION);
     end
-    giq_option(1, 751, 264, Node999, 50);
+    giq_option(1, NAME, 264, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node005
 begin
-    gsay_reply(751, 266);
-    giq_option(-1, 751, 268, Node006fool, 50);
-    if (get_critter_stat(dude_obj, 4) <= 6) then begin
-        giq_option(2, 751, 270, Node006fool, 50);
+    Reply(266);
+    giq_option(-1, NAME, 268, Node006fool, NEUTRAL_REACTION);
+    if (dude_iq <= 6) then begin
+        giq_option(2, NAME, 270, Node006fool, NEUTRAL_REACTION);
     end
-    giq_option(7, 751, 272, Node006fool, 50);
-    giq_option(7, 751, 274, Node006fool, 50);
-    giq_option(5, 751, 276, Node999, 50);
-    giq_option(1, 751, 278, Node999, 50);
+    giq_option(7, NAME, 272, Node006fool, NEUTRAL_REACTION);
+    giq_option(7, NAME, 274, Node006fool, NEUTRAL_REACTION);
+    giq_option(5, NAME, 276, Node999, NEUTRAL_REACTION);
+    giq_option(1, NAME, 278, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node006
 begin
-    gsay_reply(751, 280);
-    if ((local_var(6) <= 20) or (global_var(10) != 0)) then begin
-        giq_option(2, 751, 282, Node006, 50);
+    Reply(280);
+    if ((local_var(6) <= 20) or (global_var(GVAR_START_ARROYO_TRIAL) != 0)) then begin
+        giq_option(2, NAME, 282, Node006, NEUTRAL_REACTION);
     end
     else begin
-        giq_option(2, 751, 282, Node008, 50);
+        giq_option(2, NAME, 282, Node008, NEUTRAL_REACTION);
     end
-    giq_option(2, 751, 284, Node999, 50);
-    giq_option(-1, 751, 286, Node999, 50);
+    giq_option(2, NAME, 284, Node999, NEUTRAL_REACTION);
+    giq_option(-1, NAME, 286, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node007
 begin
-    gsay_reply(751, 288);
-    if ((local_var(6) <= 20) or (global_var(10) != 0)) then begin
-        giq_option(7, 751, 290, Node017fool, 50);
+    Reply(288);
+    if ((local_var(6) <= 20) or (global_var(GVAR_START_ARROYO_TRIAL) != 0)) then begin
+        giq_option(7, NAME, 290, Node017fool, NEUTRAL_REACTION);
     end
     else begin
-        giq_option(7, 751, 290, Node008, 50);
+        giq_option(7, NAME, 290, Node008, NEUTRAL_REACTION);
     end
-    giq_option(1, 751, 292, Node999, 50);
+    giq_option(1, NAME, 292, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node008
 begin
-    gsay_reply(751, 294);
-    giq_option(1, 751, 296, Node999, 50);
+    Reply(294);
+    giq_option(1, NAME, 296, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node009
 begin
-    gsay_reply(751, 298);
-    giq_option(1, 751, 300, Node006fool, 50);
-    giq_option(1, 751, 302, Node006fool, 50);
-    if (get_critter_stat(dude_obj, 3) > (local_var(5) * 1)) then begin
-        giq_option(7, 751, 304, Node005fool, 50);
+    Reply(298);
+    giq_option(1, NAME, 300, Node006fool, NEUTRAL_REACTION);
+    giq_option(1, NAME, 302, Node006fool, NEUTRAL_REACTION);
+    if (dude_charisma > (local_var(5) * 1)) then begin
+        giq_option(7, NAME, 304, Node005fool, NEUTRAL_REACTION);
     end
     else begin
-        giq_option(7, 751, 304, Node017fool, 50);
+        giq_option(7, NAME, 304, Node017fool, NEUTRAL_REACTION);
     end
-    if (get_critter_stat(dude_obj, 3) > (local_var(5) * 1)) then begin
-        giq_option(5, 751, 306, Node005fool, 50);
+    if (dude_charisma > (local_var(5) * 1)) then begin
+        giq_option(5, NAME, 306, Node005fool, NEUTRAL_REACTION);
     end
     else begin
-        giq_option(5, 751, 306, Node017fool, 50);
+        giq_option(5, NAME, 306, Node017fool, NEUTRAL_REACTION);
     end
 end
 
 procedure Node010
 begin
-    gsay_reply(751, 308);
+    Reply(308);
     if (local_var(6) <= 20) then begin
-        giq_option(1, 751, 310, Node016fool, 50);
+        giq_option(1, NAME, 310, Node016fool, NEUTRAL_REACTION);
     end
     else begin
-        giq_option(1, 751, 310, Node017fool, 50);
+        giq_option(1, NAME, 310, Node017fool, NEUTRAL_REACTION);
     end
-    if (get_critter_stat(dude_obj, 35) <= 10) then begin
-        giq_option(5, 751, 312, Node019, 50);
+    if (get_critter_stat(dude_obj, STAT_current_hp) <= 10) then begin
+        giq_option(5, NAME, 312, Node019, NEUTRAL_REACTION);
     end
     else begin
-        giq_option(5, 751, 312, Node017fool, 50);
+        giq_option(5, NAME, 312, Node017fool, NEUTRAL_REACTION);
     end
-    if (has_skill(dude_obj, 14) >= 50) then begin
+    if (dude_skill(SKILL_SPEECH) >= 50) then begin
         if (local_var(6) <= 20) then begin
-            giq_option(6, 751, 314, Node009fool, 50);
+            giq_option(6, NAME, 314, Node009fool, NEUTRAL_REACTION);
         end
         else begin
-            giq_option(6, 751, 314, Node017fool, 50);
+            giq_option(6, NAME, 314, Node017fool, NEUTRAL_REACTION);
         end
     end
     if (obj_is_carrying_obj_pid(dude_obj, 7) == 0) then begin
-        if (get_critter_stat(dude_obj, 35) <= 10) then begin
-            giq_option(5, 751, 316, Node019, 50);
+        if (get_critter_stat(dude_obj, STAT_current_hp) <= 10) then begin
+            giq_option(5, NAME, 316, Node019, NEUTRAL_REACTION);
         end
         else begin
-            giq_option(5, 751, 316, Node005fool, 50);
+            giq_option(5, NAME, 316, Node005fool, NEUTRAL_REACTION);
         end
     end
     if (obj_is_carrying_obj_pid(dude_obj, 273) == 0) then begin
-        if (get_critter_stat(dude_obj, 35) <= 10) then begin
-            giq_option(5, 751, 318, Node019, 50);
+        if (get_critter_stat(dude_obj, STAT_current_hp) <= 10) then begin
+            giq_option(5, NAME, 318, Node019, NEUTRAL_REACTION);
         end
         else begin
-            giq_option(5, 751, 318, Node005fool, 50);
+            giq_option(5, NAME, 318, Node005fool, NEUTRAL_REACTION);
         end
     end
-    if (get_critter_stat(dude_obj, 34) == 0) then begin
-        giq_option(1, 751, 320, Node999, 50);
+    if (dude_gender == 0) then begin
+        giq_option(1, NAME, 320, Node999, NEUTRAL_REACTION);
     end
 end
 
 procedure Node012
 begin
-    gsay_reply(751, 322);
-    giq_option(1, 751, 324, Node999, 50);
+    Reply(322);
+    giq_option(1, NAME, 324, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node013
 begin
-    gsay_reply(751, 326);
-    if (has_skill(dude_obj, 14) >= 50) then begin
-        if (get_critter_stat(dude_obj, 3) > (local_var(5) * 1)) then begin
-            giq_option(6, 751, 328, Node009fool, 50);
+    Reply(326);
+    if (dude_skill(SKILL_SPEECH) >= 50) then begin
+        if (dude_charisma > (local_var(5) * 1)) then begin
+            giq_option(6, NAME, 328, Node009fool, NEUTRAL_REACTION);
         end
         else begin
-            giq_option(6, 751, 328, Node017fool, 50);
+            giq_option(6, NAME, 328, Node017fool, NEUTRAL_REACTION);
         end
     end
-    giq_option(5, 751, 330, Node999, 50);
-    giq_option(-4, 751, 332, Node999, 50);
+    giq_option(5, NAME, 330, Node999, NEUTRAL_REACTION);
+    giq_option(-4, NAME, 332, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node014
 begin
-    gsay_reply(751, 334);
-    giq_option(1, 751, 336, Node015fool, 50);
-    if (get_critter_stat(dude_obj, 34) == 0) then begin
-        if (get_critter_stat(dude_obj, 35) <= 10) then begin
-            giq_option(1, 751, 338, Node019, 50);
+    Reply(334);
+    giq_option(1, NAME, 336, Node015fool, NEUTRAL_REACTION);
+    if (dude_gender == 0) then begin
+        if (get_critter_stat(dude_obj, STAT_current_hp) <= 10) then begin
+            giq_option(1, NAME, 338, Node019, NEUTRAL_REACTION);
         end
         else begin
-            giq_option(1, 751, 338, Node017fool, 50);
+            giq_option(1, NAME, 338, Node017fool, NEUTRAL_REACTION);
         end
     end
-    giq_option(5, 751, 340, Node999, 50);
-    giq_option(-4, 751, 342, Node999, 50);
+    giq_option(5, NAME, 340, Node999, NEUTRAL_REACTION);
+    giq_option(-4, NAME, 342, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node015
 begin
-    gsay_reply(751, 344);
-    if (get_critter_stat(dude_obj, 3) > (local_var(6) * 1)) then begin
-        giq_option(6, 751, 346, Node005fool, 50);
+    Reply(344);
+    if (dude_charisma > (local_var(6) * 1)) then begin
+        giq_option(6, NAME, 346, Node005fool, NEUTRAL_REACTION);
     end
     else begin
-        giq_option(6, 751, 346, Node017fool, 50);
+        giq_option(6, NAME, 346, Node017fool, NEUTRAL_REACTION);
     end
-    if (get_critter_stat(dude_obj, 3) > (local_var(6) * 1)) then begin
-        giq_option(8, 751, 348, Node005fool, 50);
+    if (dude_charisma > (local_var(6) * 1)) then begin
+        giq_option(8, NAME, 348, Node005fool, NEUTRAL_REACTION);
     end
     else begin
-        giq_option(8, 751, 348, Node013fool, 50);
+        giq_option(8, NAME, 348, Node013fool, NEUTRAL_REACTION);
     end
-    giq_option(1, 751, 350, Node999, 50);
+    giq_option(1, NAME, 350, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node016
 begin
-    gsay_reply(751, 352);
-    giq_option(1, 751, 354, Node015fool, 50);
-    if (get_critter_stat(dude_obj, 3) > (local_var(6) * 1)) then begin
-        giq_option(6, 751, 356, Node005fool, 50);
+    Reply(352);
+    giq_option(1, NAME, 354, Node015fool, NEUTRAL_REACTION);
+    if (dude_charisma > (local_var(6) * 1)) then begin
+        giq_option(6, NAME, 356, Node005fool, NEUTRAL_REACTION);
     end
     else begin
-        giq_option(6, 751, 356, Node017fool, 50);
+        giq_option(6, NAME, 356, Node017fool, NEUTRAL_REACTION);
     end
-    if (get_critter_stat(dude_obj, 3) > (local_var(6) * 1)) then begin
-        giq_option(8, 751, 358, Node012fool, 50);
+    if (dude_charisma > (local_var(6) * 1)) then begin
+        giq_option(8, NAME, 358, Node012fool, NEUTRAL_REACTION);
     end
     else begin
-        giq_option(8, 751, 358, Node018fool, 50);
+        giq_option(8, NAME, 358, Node018fool, NEUTRAL_REACTION);
     end
-    giq_option(5, 751, 360, Node999, 50);
-    giq_option(1, 751, 362, Node999, 50);
+    giq_option(5, NAME, 360, Node999, NEUTRAL_REACTION);
+    giq_option(1, NAME, 362, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node017
 begin
-    gsay_reply(751, 364);
-    giq_option(5, 751, 366, Node999, 50);
-    giq_option(-4, 751, 368, Node999, 50);
+    Reply(364);
+    giq_option(5, NAME, 366, Node999, NEUTRAL_REACTION);
+    giq_option(-4, NAME, 368, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node018
 begin
-    gsay_reply(751, 370);
-    giq_option(-1, 751, 372, Node999, 50);
-    if (get_critter_stat(dude_obj, 4) <= 4) then begin
-        giq_option(2, 751, 374, Node013fool, 50);
+    Reply(370);
+    giq_option(-1, NAME, 372, Node999, NEUTRAL_REACTION);
+    if (dude_iq <= 4) then begin
+        giq_option(2, NAME, 374, Node013fool, NEUTRAL_REACTION);
     end
-    giq_option(5, 751, 376, Node006fool, 50);
-    giq_option(2, 751, 378, NodeCombat, 50);
-    giq_option(1, 751, 380, Node999, 50);
+    giq_option(5, NAME, 376, Node006fool, NEUTRAL_REACTION);
+    giq_option(2, NAME, 378, NodeCombat, NEUTRAL_REACTION);
+    giq_option(1, NAME, 380, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node019
 begin
-    gsay_reply(751, 382);
+    Reply(382);
     if ((obj_is_carrying_obj_pid(dude_obj, 7) == 0) and (obj_is_carrying_obj_pid(self_obj, 7) > 1)) then begin
-        giq_option(1, 751, 384, Node020spear, 50);
+        giq_option(1, NAME, 384, Node020spear, NEUTRAL_REACTION);
     end
     if ((obj_is_carrying_obj_pid(dude_obj, 273) == 0) and (obj_is_carrying_obj_pid(self_obj, 273) != 0)) then begin
-        giq_option(1, 751, 386, Node021powder, 50);
+        giq_option(1, NAME, 386, Node021powder, NEUTRAL_REACTION);
     end
-    if (has_skill(dude_obj, 14) >= 50) then begin
-        giq_option(8, 751, 388, Node009fool, 50);
+    if (dude_skill(SKILL_SPEECH) >= 50) then begin
+        giq_option(8, NAME, 388, Node009fool, NEUTRAL_REACTION);
     end
-    giq_option(5, 751, 390, Node999, 50);
-    giq_option(5, 751, 392, Node018fool, 50);
-    giq_option(1, 751, 394, Node022, 50);
+    giq_option(5, NAME, 390, Node999, NEUTRAL_REACTION);
+    giq_option(5, NAME, 392, Node018fool, NEUTRAL_REACTION);
+    giq_option(1, NAME, 394, Node022, NEUTRAL_REACTION);
 end
 
 procedure Node020spear
 begin
     variable LVar0 := 0;
     LVar0 := obj_carrying_pid_obj(self_obj, 7);
-    metarule(43, self_obj);
+    inven_unwield(self_obj);
     Junk := rm_mult_objs_from_inven(self_obj, LVar0, 1);
     add_mult_objs_to_inven(dude_obj, LVar0, 1);
     LVar0 := obj_carrying_pid_obj(self_obj, 7);
     wield_obj_critter(self_obj, LVar0);
-    gsay_reply(751, 396);
-    giq_option(1, 751, 398, Node999, 50);
+    Reply(396);
+    giq_option(1, NAME, 398, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node021powder
 begin
     Junk := rm_mult_objs_from_inven(self_obj, obj_carrying_pid_obj(self_obj, 273), 1);
-    add_mult_objs_to_inven(dude_obj, create_object_sid(273, 0, 0, -1), 1);
-    gsay_reply(751, 400);
-    giq_option(1, 751, 402, Node999, 50);
+    add_mult_objs_to_inven(dude_obj, create_object(273, 0, 0), 1);
+    Reply(400);
+    giq_option(1, NAME, 402, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node022
 begin
-    gsay_reply(751, 404);
-    giq_option(1, 751, 406, Node023, 50);
+    Reply(404);
+    giq_option(1, NAME, 406, Node023, NEUTRAL_REACTION);
 end
 
 procedure Node023
@@ -749,157 +749,157 @@ begin
     gfade_out(600);
     gfade_in(600);
     game_time_advance(18000);
-    gsay_reply(751, 408);
-    giq_option(-1, 751, 409, NodeMoveToArroyo, 50);
-    giq_option(2, 751, 410, NodeMoveToArroyo, 50);
+    Reply(408);
+    giq_option(-1, NAME, 409, NodeMoveToArroyo, NEUTRAL_REACTION);
+    giq_option(2, NAME, 410, NodeMoveToArroyo, NEUTRAL_REACTION);
 end
 
 procedure Node037
 begin
-    gsay_reply(751, 412);
-    giq_option(-2, 751, 414, Node038, 50);
-    if (get_critter_stat(dude_obj, 4) <= 5) then begin
-        giq_option(2, 751, 416, Node039, 50);
+    Reply(412);
+    giq_option(-2, NAME, 414, Node038, NEUTRAL_REACTION);
+    if (dude_iq <= 5) then begin
+        giq_option(2, NAME, 416, Node039, NEUTRAL_REACTION);
     end
-    if (get_critter_stat(dude_obj, 4) <= 5) then begin
-        giq_option(2, 751, 418, Node005, 50);
+    if (dude_iq <= 5) then begin
+        giq_option(2, NAME, 418, Node005, NEUTRAL_REACTION);
     end
-    giq_option(6, 751, 420, Node038, 50);
-    giq_option(8, 751, 422, Node010fool, 50);
-    giq_option(6, 751, 424, Node005fool, 50);
-    giq_option(6, 751, 426, Node002fool, 50);
+    giq_option(6, NAME, 420, Node038, NEUTRAL_REACTION);
+    giq_option(8, NAME, 422, Node010fool, NEUTRAL_REACTION);
+    giq_option(6, NAME, 424, Node005fool, NEUTRAL_REACTION);
+    giq_option(6, NAME, 426, Node002fool, NEUTRAL_REACTION);
 end
 
 procedure Node038
 begin
-    gsay_reply(751, 428);
-    giq_option(-1, 751, 430, Node039, 50);
-    giq_option(2, 751, 432, Node039, 50);
+    Reply(428);
+    giq_option(-1, NAME, 430, Node039, NEUTRAL_REACTION);
+    giq_option(2, NAME, 432, Node039, NEUTRAL_REACTION);
 end
 
 procedure Node039
 begin
-    gsay_reply(751, 434);
-    giq_option(1, 751, 436, Node023, 50);
+    Reply(434);
+    giq_option(1, NAME, 436, Node023, NEUTRAL_REACTION);
 end
 
 procedure Node040
 begin
-    gsay_reply(751, 438);
-    giq_option(-2, 751, 440, Node999, 50);
-    if ((get_critter_stat(dude_obj, 4) <= 4) and (global_var(639) != 2) and (global_var(639) != 3)) then begin
-        giq_option(3, 751, 442, Node041, 50);
+    Reply(438);
+    giq_option(-2, NAME, 440, Node999, NEUTRAL_REACTION);
+    if ((dude_iq <= 4) and (global_var(GVAR_ARROYO_MEMBER) != 2) and (global_var(GVAR_ARROYO_MEMBER) != 3)) then begin
+        giq_option(3, NAME, 442, Node041, NEUTRAL_REACTION);
     end
-    if ((global_var(639) != 2) and (global_var(639) != 3)) then begin
-        giq_option(5, 751, 444, Node999, 50);
+    if ((global_var(GVAR_ARROYO_MEMBER) != 2) and (global_var(GVAR_ARROYO_MEMBER) != 3)) then begin
+        giq_option(5, NAME, 444, Node999, NEUTRAL_REACTION);
     end
-    if (global_var(647) == 1) then begin
-        giq_option(5, 751, 446, Node042, 50);
+    if (global_var(GVAR_ACFIRST_Q1_SKINS) == 1) then begin
+        giq_option(5, NAME, 446, Node042, NEUTRAL_REACTION);
     end
-    if (global_var(646) == 1) then begin
-        giq_option(5, 751, 448, Node041, 50);
+    if (global_var(GVAR_ACJORDAN_DUDE_ESCAPED) == 1) then begin
+        giq_option(5, NAME, 448, Node041, NEUTRAL_REACTION);
     end
-    if (global_var(650) == 1) then begin
-        giq_option(5, 751, 450, Node041, 50);
+    if (global_var(GVAR_ACTEMVIL_TALK) == 1) then begin
+        giq_option(5, NAME, 450, Node041, NEUTRAL_REACTION);
     end
-    if (global_var(644) == 10) then begin
-        giq_option(5, 751, 452, Node041, 50);
+    if (global_var(GVAR_AHELDER_Q1_AGRO) == 10) then begin
+        giq_option(5, NAME, 452, Node041, NEUTRAL_REACTION);
     end
-    if ((global_var(639) != 2) and (global_var(639) != 3)) then begin
-        giq_option(3, 751, 454, Node041, 50);
+    if ((global_var(GVAR_ARROYO_MEMBER) != 2) and (global_var(GVAR_ARROYO_MEMBER) != 3)) then begin
+        giq_option(3, NAME, 454, Node041, NEUTRAL_REACTION);
     end
-    giq_option(3, 751, 456, Node999, 50);
+    giq_option(3, NAME, 456, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node041
 begin
-    gsay_reply(751, 458);
-    giq_option(5, 751, 460, Node043, 50);
-    giq_option(-4, 751, 462, Node999, 50);
-    giq_option(5, 751, 464, Node999, 50);
+    Reply(458);
+    giq_option(5, NAME, 460, Node043, NEUTRAL_REACTION);
+    giq_option(-4, NAME, 462, Node999, NEUTRAL_REACTION);
+    giq_option(5, NAME, 464, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node042
 begin
-    gsay_reply(751, 466);
-    giq_option(5, 751, 468, Node043, 50);
-    giq_option(-4, 751, 470, Node999, 50);
-    giq_option(5, 751, 472, Node999, 50);
+    Reply(466);
+    giq_option(5, NAME, 468, Node043, NEUTRAL_REACTION);
+    giq_option(-4, NAME, 470, Node999, NEUTRAL_REACTION);
+    giq_option(5, NAME, 472, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node043
 begin
-    gsay_reply(751, 474);
-    giq_option(1, 751, 476, Node999, 50);
+    Reply(474);
+    giq_option(1, NAME, 476, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node044
 begin
-    if (get_critter_stat(dude_obj, 34) == 0) then begin
-        gsay_reply(751, 478);
+    if (dude_gender == 0) then begin
+        Reply(478);
     end
     else begin
-        gsay_reply(751, 479);
+        Reply(479);
     end
-    giq_option(-1, 751, 480, Node046, 50);
-    giq_option(-1, 751, 482, Node045, 50);
-    giq_option(2, 751, 484, Node046, 50);
-    giq_option(2, 751, 486, Node045, 50);
-    giq_option(5, 751, 488, Node045, 50);
-    giq_option(8, 751, 490, Node045fool, 50);
-    giq_option(5, 751, 492, Node999, 50);
+    giq_option(-1, NAME, 480, Node046, NEUTRAL_REACTION);
+    giq_option(-1, NAME, 482, Node045, NEUTRAL_REACTION);
+    giq_option(2, NAME, 484, Node046, NEUTRAL_REACTION);
+    giq_option(2, NAME, 486, Node045, NEUTRAL_REACTION);
+    giq_option(5, NAME, 488, Node045, NEUTRAL_REACTION);
+    giq_option(8, NAME, 490, Node045fool, NEUTRAL_REACTION);
+    giq_option(5, NAME, 492, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node045
 begin
-    gsay_reply(751, 494);
-    giq_option(-1, 751, 496, Node047, 50);
-    giq_option(2, 751, 498, Node046, 50);
-    giq_option(2, 751, 500, Node046, 50);
-    giq_option(5, 751, 502, Node048, 50);
-    giq_option(5, 751, 504, Node049, 50);
+    Reply(494);
+    giq_option(-1, NAME, 496, Node047, NEUTRAL_REACTION);
+    giq_option(2, NAME, 498, Node046, NEUTRAL_REACTION);
+    giq_option(2, NAME, 500, Node046, NEUTRAL_REACTION);
+    giq_option(5, NAME, 502, Node048, NEUTRAL_REACTION);
+    giq_option(5, NAME, 504, Node049, NEUTRAL_REACTION);
 end
 
 procedure Node046
 begin
-    gsay_reply(751, 506);
-    giq_option(-1, 751, 508, Node999, 50);
-    giq_option(-1, 751, 510, Node047, 50);
-    if (get_critter_stat(dude_obj, 4) <= 4) then begin
-        giq_option(2, 751, 512, Node048, 50);
+    Reply(506);
+    giq_option(-1, NAME, 508, Node999, NEUTRAL_REACTION);
+    giq_option(-1, NAME, 510, Node047, NEUTRAL_REACTION);
+    if (dude_iq <= 4) then begin
+        giq_option(2, NAME, 512, Node048, NEUTRAL_REACTION);
     end
-    if (get_critter_stat(dude_obj, 4) <= 7) then begin
-        giq_option(5, 751, 516, Node047, 50);
+    if (dude_iq <= 7) then begin
+        giq_option(5, NAME, 516, Node047, NEUTRAL_REACTION);
     end
-    giq_option(5, 751, 518, Node048, 50);
-    giq_option(5, 751, 520, Node049, 50);
-    giq_option(1, 751, 522, Node999, 50);
+    giq_option(5, NAME, 518, Node048, NEUTRAL_REACTION);
+    giq_option(5, NAME, 520, Node049, NEUTRAL_REACTION);
+    giq_option(1, NAME, 522, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node047
 begin
-    gsay_reply(751, 524);
-    giq_option(1, 751, 526, Node999, 50);
+    Reply(524);
+    giq_option(1, NAME, 526, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node048
 begin
-    gsay_reply(751, 528);
-    giq_option(1, 751, 530, Node999, 50);
+    Reply(528);
+    giq_option(1, NAME, 530, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node049
 begin
-    gsay_reply(751, 532);
-    giq_option(5, 751, 534, Node999, 50);
-    giq_option(6, 751, 536, Node006, 50);
-    giq_option(1, 751, 538, Node999, 50);
+    Reply(532);
+    giq_option(5, NAME, 534, Node999, NEUTRAL_REACTION);
+    giq_option(6, NAME, 536, Node006, NEUTRAL_REACTION);
+    giq_option(1, NAME, 538, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node050
 begin
-    gsay_reply(751, 540);
-    giq_option(1, 751, 542, Node999, 50);
+    Reply(540);
+    giq_option(1, NAME, 542, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node002fool
@@ -994,72 +994,72 @@ end
 procedure NodeMoveToArroyo
 begin
     give_exp_points(300);
-    display_msg(message_str(14, 100) + 300 + message_str(14, 101));
-    set_global_var(10, 2);
-    if (cur_map_index == 3) then begin
+    display_msg(g_mstr(100) + 300 + g_mstr(101));
+    set_global_var(GVAR_START_ARROYO_TRIAL, 2);
+    if (CUR_MAP_ARCAVES) then begin
         set_map_var(20, 2);
     end
-    set_global_var(653, 1);
+    set_global_var(GVAR_WOOZ_TEACH, 1);
     load_map(4, 0);
 end
 
 procedure Node058
 begin
-    gsay_reply(751, 600);
-    giq_option(5, 751, 602, Node061, 50);
-    giq_option(5, 751, 604, Node062, 50);
-    giq_option(6, 751, 606, Node063, 50);
-    giq_option(6, 751, 608, Node071, 50);
-    giq_option(1, 751, 610, Node999, 50);
+    Reply(600);
+    giq_option(5, NAME, 602, Node061, NEUTRAL_REACTION);
+    giq_option(5, NAME, 604, Node062, NEUTRAL_REACTION);
+    giq_option(6, NAME, 606, Node063, NEUTRAL_REACTION);
+    giq_option(6, NAME, 608, Node071, NEUTRAL_REACTION);
+    giq_option(1, NAME, 610, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node060
 begin
-    gsay_reply(751, 612);
-    giq_option(2, 751, 614, Node060, 50);
-    giq_option(2, 751, 616, Node999, 50);
-    giq_option(-1, 751, 618, Node999, 50);
+    Reply(612);
+    giq_option(2, NAME, 614, Node060, NEUTRAL_REACTION);
+    giq_option(2, NAME, 616, Node999, NEUTRAL_REACTION);
+    giq_option(-1, NAME, 618, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node061
 begin
-    gsay_reply(751, 620);
-    if (((get_critter_stat(dude_obj, 2) + get_critter_stat(dude_obj, 0) + get_critter_stat(dude_obj, 5)) < 19) or (get_critter_stat(dude_obj, 3) > 4)) then begin
-        giq_option(1, 751, 622, Node080, 50);
+    Reply(620);
+    if (((dude_endurance + dude_strength + get_critter_stat(dude_obj, STAT_ag)) < 19) or (dude_charisma > 4)) then begin
+        giq_option(1, NAME, 622, Node080, NEUTRAL_REACTION);
     end
     else begin
-        giq_option(1, 751, 622, Node091, 50);
+        giq_option(1, NAME, 622, Node091, NEUTRAL_REACTION);
     end
-    if (((get_critter_stat(dude_obj, 2) + get_critter_stat(dude_obj, 0) + get_critter_stat(dude_obj, 5)) < 19) or (get_critter_stat(dude_obj, 3) > 4)) then begin
-        giq_option(1, 751, 624, Node080, 50);
+    if (((dude_endurance + dude_strength + get_critter_stat(dude_obj, STAT_ag)) < 19) or (dude_charisma > 4)) then begin
+        giq_option(1, NAME, 624, Node080, NEUTRAL_REACTION);
     end
     else begin
-        giq_option(1, 751, 624, Node091, 50);
+        giq_option(1, NAME, 624, Node091, NEUTRAL_REACTION);
     end
-    giq_option(1, 751, 626, Node089, 50);
+    giq_option(1, NAME, 626, Node089, NEUTRAL_REACTION);
 end
 
 procedure Node062
 begin
-    gsay_reply(751, 628);
-    giq_option(1, 751, 630, Node089, 50);
+    Reply(628);
+    giq_option(1, NAME, 630, Node089, NEUTRAL_REACTION);
 end
 
 procedure Node063
 begin
-    gsay_reply(751, 632);
-    giq_option(1, 751, 634, Node064, 50);
-    giq_option(1, 751, 636, Node064, 50);
-    giq_option(5, 751, 638, Node079, 50);
-    giq_option(-4, 751, 640, Node089, 50);
+    Reply(632);
+    giq_option(1, NAME, 634, Node064, NEUTRAL_REACTION);
+    giq_option(1, NAME, 636, Node064, NEUTRAL_REACTION);
+    giq_option(5, NAME, 638, Node079, NEUTRAL_REACTION);
+    giq_option(-4, NAME, 640, Node089, NEUTRAL_REACTION);
 end
 
 procedure Node064
 begin
-    gsay_reply(751, 642);
-    giq_option(6, 751, 644, Node079, 50);
-    giq_option(8, 751, 646, Node077, 50);
-    giq_option(1, 751, 648, Node089, 50);
+    Reply(642);
+    giq_option(6, NAME, 644, Node079, NEUTRAL_REACTION);
+    giq_option(8, NAME, 646, Node077, NEUTRAL_REACTION);
+    giq_option(1, NAME, 648, Node089, NEUTRAL_REACTION);
 end
 
 procedure Node069
@@ -1067,268 +1067,267 @@ begin
     gfade_out(600);
     gfade_in(600);
     game_time_advance(18000);
-    gsay_reply(751, 650);
-    giq_option(1, 751, 652, NodeMoveToArroyo, 50);
+    Reply(650);
+    giq_option(1, NAME, 652, NodeMoveToArroyo, NEUTRAL_REACTION);
 end
 
 procedure Node070
 begin
     set_local_var(5, local_var(5) + 1);
-    gsay_reply(751, 654);
-    giq_option(-1, 751, 656, Node074, 50);
-    if (get_critter_stat(dude_obj, 4) <= 4) then begin
-        giq_option(2, 751, 658, Node074, 50);
+    Reply(654);
+    giq_option(-1, NAME, 656, Node074, NEUTRAL_REACTION);
+    if (dude_iq <= 4) then begin
+        giq_option(2, NAME, 658, Node074, NEUTRAL_REACTION);
     end
-    giq_option(2, 751, 660, Node074, 50);
-    if (get_critter_stat(dude_obj, 4) <= 4) then begin
-        giq_option(2, 751, 662, Node071, 50);
+    giq_option(2, NAME, 660, Node074, NEUTRAL_REACTION);
+    if (dude_iq <= 4) then begin
+        giq_option(2, NAME, 662, Node071, NEUTRAL_REACTION);
     end
-    if (get_critter_stat(dude_obj, 4) <= 4) then begin
-        giq_option(2, 751, 664, Node062, 50);
+    if (dude_iq <= 4) then begin
+        giq_option(2, NAME, 664, Node062, NEUTRAL_REACTION);
     end
-    giq_option(2, 751, 666, Node076, 50);
-    giq_option(5, 751, 668, Node076, 50);
-    giq_option(5, 751, 670, Node071, 50);
-    giq_option(5, 751, 672, Node072, 50);
-    if (get_critter_stat(dude_obj, 35) <= 10) then begin
-        giq_option(3, 751, 674, Node080, 50);
+    giq_option(2, NAME, 666, Node076, NEUTRAL_REACTION);
+    giq_option(5, NAME, 668, Node076, NEUTRAL_REACTION);
+    giq_option(5, NAME, 670, Node071, NEUTRAL_REACTION);
+    giq_option(5, NAME, 672, Node072, NEUTRAL_REACTION);
+    if (get_critter_stat(dude_obj, STAT_current_hp) <= 10) then begin
+        giq_option(3, NAME, 674, Node080, NEUTRAL_REACTION);
     end
-    if (get_critter_stat(dude_obj, 35) <= 10) then begin
-        giq_option(-2, 751, 676, Node080, 50);
+    if (get_critter_stat(dude_obj, STAT_current_hp) <= 10) then begin
+        giq_option(-2, NAME, 676, Node080, NEUTRAL_REACTION);
     end
 end
 
 procedure Node071
 begin
-    gsay_reply(751, 678);
-    giq_option(2, 751, 680, Node077, 50);
-    giq_option(2, 751, 682, Node089, 50);
-    giq_option(-1, 751, 684, Node089, 50);
+    Reply(678);
+    giq_option(2, NAME, 680, Node077, NEUTRAL_REACTION);
+    giq_option(2, NAME, 682, Node089, NEUTRAL_REACTION);
+    giq_option(-1, NAME, 684, Node089, NEUTRAL_REACTION);
 end
 
 procedure Node072
 begin
-    gsay_reply(751, 686);
-    giq_option(6, 751, 688, Node073, 50);
-    if (has_skill(dude_obj, 14) >= 50) then begin
-        giq_option(8, 751, 690, Node073, 50);
+    Reply(686);
+    giq_option(6, NAME, 688, Node073, NEUTRAL_REACTION);
+    if (dude_skill(SKILL_SPEECH) >= 50) then begin
+        giq_option(8, NAME, 690, Node073, NEUTRAL_REACTION);
     end
-    giq_option(5, 751, 692, Node076, 50);
-    giq_option(5, 751, 694, Node058, 50);
-    giq_option(5, 751, 696, Node071, 50);
-    giq_option(5, 751, 698, Node089, 50);
+    giq_option(5, NAME, 692, Node076, NEUTRAL_REACTION);
+    giq_option(5, NAME, 694, Node058, NEUTRAL_REACTION);
+    giq_option(5, NAME, 696, Node071, NEUTRAL_REACTION);
+    giq_option(5, NAME, 698, Node089, NEUTRAL_REACTION);
 end
 
 procedure Node073
 begin
-    gsay_reply(751, 700);
-    giq_option(5, 751, 702, Node079, 50);
-    giq_option(5, 751, 704, Node075, 50);
-    if (metarule(40, 3)) then begin
-        giq_option(3, 751, 705, Node074, 50);
+    Reply(700);
+    giq_option(5, NAME, 702, Node079, NEUTRAL_REACTION);
+    giq_option(5, NAME, 704, Node075, NEUTRAL_REACTION);
+    if (is_skill_tagged(SKILL_UNARMED_COMBAT)) then begin
+        giq_option(3, NAME, 705, Node074, NEUTRAL_REACTION);
     end
-    giq_option(1, 751, 706, Node089, 50);
+    giq_option(1, NAME, 706, Node089, NEUTRAL_REACTION);
 end
 
 procedure Node074
 begin
-    gsay_reply(751, 708);
-    giq_option(-1, 751, 710, Node089, 50);
-    giq_option(2, 751, 712, Node089, 50);
-    giq_option(5, 751, 714, Node089, 50);
-    if (metarule(40, 3)) then begin
-        giq_option(3, 751, 715, Node073, 50);
+    Reply(708);
+    giq_option(-1, NAME, 710, Node089, NEUTRAL_REACTION);
+    giq_option(2, NAME, 712, Node089, NEUTRAL_REACTION);
+    giq_option(5, NAME, 714, Node089, NEUTRAL_REACTION);
+    if (is_skill_tagged(SKILL_UNARMED_COMBAT)) then begin
+        giq_option(3, NAME, 715, Node073, NEUTRAL_REACTION);
     end
-    if (has_skill(dude_obj, 14) >= 50) then begin
-        giq_option(8, 751, 716, Node060, 50);
+    if (dude_skill(SKILL_SPEECH) >= 50) then begin
+        giq_option(8, NAME, 716, Node060, NEUTRAL_REACTION);
     end
 end
 
 procedure Node075
 begin
-    gsay_reply(751, 718);
-    giq_option(5, 751, 720, Node079, 50);
-    giq_option(5, 751, 722, Node075, 50);
-    giq_option(1, 751, 724, Node089, 50);
+    Reply(718);
+    giq_option(5, NAME, 720, Node079, NEUTRAL_REACTION);
+    giq_option(5, NAME, 722, Node075, NEUTRAL_REACTION);
+    giq_option(1, NAME, 724, Node089, NEUTRAL_REACTION);
 end
 
 procedure Node076
 begin
-    gsay_reply(751, 726);
-    giq_option(5, 751, 728, Node077, 50);
-    if (metarule(40, 3)) then begin
-        giq_option(3, 751, 729, Node079, 50);
+    Reply(726);
+    giq_option(5, NAME, 728, Node077, NEUTRAL_REACTION);
+    if (is_skill_tagged(SKILL_UNARMED_COMBAT)) then begin
+        giq_option(3, NAME, 729, Node079, NEUTRAL_REACTION);
     end
-    giq_option(5, 751, 730, Node073, 50);
-    giq_option(-4, 751, 732, Node073, 50);
+    giq_option(5, NAME, 730, Node073, NEUTRAL_REACTION);
+    giq_option(-4, NAME, 732, Node073, NEUTRAL_REACTION);
 end
 
 procedure Node077
 begin
-    gsay_reply(751, 734);
-    giq_option(5, 751, 736, Node077, 50);
-    giq_option(5, 751, 738, Node089, 50);
-    giq_option(-4, 751, 740, Node089, 50);
+    Reply(734);
+    giq_option(5, NAME, 736, Node077, NEUTRAL_REACTION);
+    giq_option(5, NAME, 738, Node089, NEUTRAL_REACTION);
+    giq_option(-4, NAME, 740, Node089, NEUTRAL_REACTION);
 end
 
 procedure Node079
 begin
-    gsay_reply(751, 742);
-    giq_option(3, 751, 744, Node093, 50);
-    if (get_critter_stat(dude_obj, 4) <= 2) then begin
-        if (((get_critter_stat(dude_obj, 2) + get_critter_stat(dude_obj, 0) + get_critter_stat(dude_obj, 5)) < 19) or (get_critter_stat(dude_obj, 3) > 4)) then begin
-            giq_option(1, 751, 747, Node080, 50);
+    Reply(742);
+    giq_option(3, NAME, 744, Node093, NEUTRAL_REACTION);
+    if (dude_iq <= 2) then begin
+        if (((dude_endurance + dude_strength + get_critter_stat(dude_obj, STAT_ag)) < 19) or (dude_charisma > 4)) then begin
+            giq_option(1, NAME, 747, Node080, NEUTRAL_REACTION);
         end
         else begin
-            giq_option(1, 751, 747, Node091, 50);
+            giq_option(1, NAME, 747, Node091, NEUTRAL_REACTION);
         end
     end
-    if (((get_critter_stat(dude_obj, 2) + get_critter_stat(dude_obj, 0) + get_critter_stat(dude_obj, 5)) < 19) or (get_critter_stat(dude_obj, 3) > 4)) then begin
-        giq_option(3, 751, 746, Node080, 50);
+    if (((dude_endurance + dude_strength + get_critter_stat(dude_obj, STAT_ag)) < 19) or (dude_charisma > 4)) then begin
+        giq_option(3, NAME, 746, Node080, NEUTRAL_REACTION);
     end
     else begin
-        giq_option(3, 751, 746, Node091, 50);
+        giq_option(3, NAME, 746, Node091, NEUTRAL_REACTION);
     end
     if ((obj_is_carrying_obj_pid(dude_obj, 7) == 0) and (obj_is_carrying_obj_pid(self_obj, 7) > 1)) then begin
-        giq_option(3, 751, 748, Node081spear, 50);
+        giq_option(3, NAME, 748, Node081spear, NEUTRAL_REACTION);
     end
     if ((obj_is_carrying_obj_pid(dude_obj, 273) == 0) and (obj_is_carrying_obj_pid(self_obj, 273) != 0)) then begin
-        giq_option(3, 751, 750, Node082powder, 50);
+        giq_option(3, NAME, 750, Node082powder, NEUTRAL_REACTION);
     end
-    giq_option(3, 751, 752, Node089, 50);
+    giq_option(3, NAME, 752, Node089, NEUTRAL_REACTION);
 end
 
 procedure Node080
 begin
     party_add(self_obj);
-    critter_add_trait(self_obj, 1, 6, 0);
-    set_global_var(643, 1);
-    set_global_var(642, 1);
-    gsay_reply(751, 754);
-    giq_option(1, 751, 755, Node999, 50);
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 0);
+    set_global_var(GVAR_ACKLINT_JOIN, 1);
+    set_global_var(GVAR_ACKLINT_PERMIT, 1);
+    Reply(754);
+    giq_option(1, NAME, 755, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node081spear
 begin
-    set_global_var(642, 1);
+    set_global_var(GVAR_ACKLINT_PERMIT, 1);
     Junk := rm_mult_objs_from_inven(self_obj, obj_carrying_pid_obj(self_obj, 7), 1);
-    add_mult_objs_to_inven(dude_obj, create_object_sid(7, 0, 0, -1), 1);
-    gsay_reply(751, 756);
-    giq_option(1, 751, 758, Node999, 50);
+    add_mult_objs_to_inven(dude_obj, create_object(7, 0, 0), 1);
+    Reply(756);
+    giq_option(1, NAME, 758, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node082powder
 begin
-    set_global_var(642, 1);
+    set_global_var(GVAR_ACKLINT_PERMIT, 1);
     Junk := rm_mult_objs_from_inven(self_obj, obj_carrying_pid_obj(self_obj, 273), 1);
-    add_mult_objs_to_inven(dude_obj, create_object_sid(273, 0, 0, -1), 1);
-    gsay_reply(751, 760);
-    giq_option(1, 751, 762, Node999, 50);
+    add_mult_objs_to_inven(dude_obj, create_object(273, 0, 0), 1);
+    Reply(760);
+    giq_option(1, NAME, 762, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node083
 begin
-    gsay_reply(751, 764);
-    giq_option(-2, 751, 766, Node084, 50);
-    if (get_critter_stat(dude_obj, 4) <= 5) then begin
-        giq_option(2, 751, 768, Node085, 50);
+    Reply(764);
+    giq_option(-2, NAME, 766, Node084, NEUTRAL_REACTION);
+    if (dude_iq <= 5) then begin
+        giq_option(2, NAME, 768, Node085, NEUTRAL_REACTION);
     end
-    giq_option(6, 751, 770, Node085, 50);
+    giq_option(6, NAME, 770, Node085, NEUTRAL_REACTION);
 end
 
 procedure Node084
 begin
-    gsay_reply(751, 772);
-    giq_option(-2, 751, 430, Node085, 50);
-    giq_option(3, 751, 776, Node085, 50);
+    Reply(772);
+    giq_option(-2, NAME, 430, Node085, NEUTRAL_REACTION);
+    giq_option(3, NAME, 776, Node085, NEUTRAL_REACTION);
 end
 
 procedure Node085
 begin
-    gsay_reply(751, 778);
-    giq_option(1, 751, 780, Node069, 50);
+    Reply(778);
+    giq_option(1, NAME, 780, Node069, NEUTRAL_REACTION);
 end
 
 procedure Node086
 begin
-    gsay_reply(751, 782);
-    if (get_critter_stat(dude_obj, 35) <= 10) then begin
-        giq_option(3, 751, 784, Node080, 50);
+    Reply(782);
+    if (get_critter_stat(dude_obj, STAT_current_hp) <= 10) then begin
+        giq_option(3, NAME, 784, Node080, NEUTRAL_REACTION);
     end
-    if (get_critter_stat(dude_obj, 35) <= 10) then begin
-        giq_option(-2, 751, 786, Node080, 50);
+    if (get_critter_stat(dude_obj, STAT_current_hp) <= 10) then begin
+        giq_option(-2, NAME, 786, Node080, NEUTRAL_REACTION);
     end
-    giq_option(1, 751, 788, Node999, 50);
+    giq_option(1, NAME, 788, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node087
 begin
-    float_msg(self_obj, message_str(751, 790), 0);
+    float_msg(self_obj, mstr(790), FLOAT_MSG_NORMAL);
 end
 
 procedure Node089
 begin
-    gsay_reply(751, 794);
-    giq_option(3, 751, 796, Node077, 50);
-    giq_option(-2, 751, 798, Node077, 50);
-    if (((get_critter_stat(dude_obj, 2) + get_critter_stat(dude_obj, 0) + get_critter_stat(dude_obj, 5)) < 19) or (get_critter_stat(dude_obj, 3) > 4)) then begin
-        giq_option(3, 751, 800, Node080, 50);
+    Reply(794);
+    giq_option(3, NAME, 796, Node077, NEUTRAL_REACTION);
+    giq_option(-2, NAME, 798, Node077, NEUTRAL_REACTION);
+    if (((dude_endurance + dude_strength + get_critter_stat(dude_obj, STAT_ag)) < 19) or (dude_charisma > 4)) then begin
+        giq_option(3, NAME, 800, Node080, NEUTRAL_REACTION);
     end
     else begin
-        giq_option(3, 751, 800, Node091, 50);
+        giq_option(3, NAME, 800, Node091, NEUTRAL_REACTION);
     end
-    if (((get_critter_stat(dude_obj, 2) + get_critter_stat(dude_obj, 0) + get_critter_stat(dude_obj, 5)) < 19) or (get_critter_stat(dude_obj, 3) > 4)) then begin
-        giq_option(-2, 751, 802, Node080, 50);
+    if (((dude_endurance + dude_strength + get_critter_stat(dude_obj, STAT_ag)) < 19) or (dude_charisma > 4)) then begin
+        giq_option(-2, NAME, 802, Node080, NEUTRAL_REACTION);
     end
     else begin
-        giq_option(-2, 751, 802, Node091, 50);
+        giq_option(-2, NAME, 802, Node091, NEUTRAL_REACTION);
     end
-    giq_option(3, 751, 804, Node092, 50);
-    giq_option(-2, 751, 806, Node092, 50);
+    giq_option(3, NAME, 804, Node092, NEUTRAL_REACTION);
+    giq_option(-2, NAME, 806, Node092, NEUTRAL_REACTION);
 end
 
 procedure Node091
 begin
-    set_global_var(642, 1);
-    gsay_reply(751, 808);
-    giq_option(1, 751, 810, Node999, 50);
+    set_global_var(GVAR_ACKLINT_PERMIT, 1);
+    Reply(808);
+    giq_option(1, NAME, 810, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node092
 begin
-    gsay_reply(751, 812);
-    giq_option(3, 751, 814, Node093, 50);
-    if (get_critter_stat(dude_obj, 4) <= 2) then begin
-        if (((get_critter_stat(dude_obj, 2) + get_critter_stat(dude_obj, 0) + get_critter_stat(dude_obj, 5)) < 19) or (get_critter_stat(dude_obj, 3) > 4)) then begin
-            giq_option(1, 751, 813, Node080, 50);
+    Reply(812);
+    giq_option(3, NAME, 814, Node093, NEUTRAL_REACTION);
+    if (dude_iq <= 2) then begin
+        if (((dude_endurance + dude_strength + get_critter_stat(dude_obj, STAT_ag)) < 19) or (dude_charisma > 4)) then begin
+            giq_option(1, NAME, 813, Node080, NEUTRAL_REACTION);
         end
         else begin
-            giq_option(1, 751, 813, Node091, 50);
+            giq_option(1, NAME, 813, Node091, NEUTRAL_REACTION);
         end
     end
-    if (((get_critter_stat(dude_obj, 2) + get_critter_stat(dude_obj, 0) + get_critter_stat(dude_obj, 5)) < 19) or (get_critter_stat(dude_obj, 3) > 4)) then begin
-        giq_option(3, 751, 816, Node080, 50);
+    if (((dude_endurance + dude_strength + get_critter_stat(dude_obj, STAT_ag)) < 19) or (dude_charisma > 4)) then begin
+        giq_option(3, NAME, 816, Node080, NEUTRAL_REACTION);
     end
     else begin
-        giq_option(3, 751, 816, Node091, 50);
+        giq_option(3, NAME, 816, Node091, NEUTRAL_REACTION);
     end
     if ((obj_is_carrying_obj_pid(dude_obj, 7) == 0) and (obj_is_carrying_obj_pid(self_obj, 7) > 1)) then begin
-        giq_option(3, 751, 818, Node081spear, 50);
+        giq_option(3, NAME, 818, Node081spear, NEUTRAL_REACTION);
     end
     if ((obj_is_carrying_obj_pid(dude_obj, 273) == 0) and (obj_is_carrying_obj_pid(self_obj, 273) != 0)) then begin
-        giq_option(3, 751, 820, Node082powder, 50);
+        giq_option(3, NAME, 820, Node082powder, NEUTRAL_REACTION);
     end
-    giq_option(3, 751, 822, Node089, 50);
+    giq_option(3, NAME, 822, Node089, NEUTRAL_REACTION);
 end
 
 procedure Node093
 begin
-    set_global_var(642, 1);
-    gsay_reply(751, 824);
-    giq_option(1, 751, 826, Node999, 50);
+    set_global_var(GVAR_ACKLINT_PERMIT, 1);
+    Reply(824);
+    giq_option(1, NAME, 826, Node999, NEUTRAL_REACTION);
 end
 
 procedure start
 begin
 end
-
