diff --git a/Mods/MegaMod/Original/ACOMPU.ssl b/Mods/MegaMod/ReDefine/ACOMPU.ssl
index a4731b6..f99d681 100644
--- a/Mods/MegaMod/Original/ACOMPU.ssl
+++ b/Mods/MegaMod/ReDefine/ACOMPU.ssl
@@ -105,13 +105,13 @@ end
 procedure look_at_p_proc
 begin
     script_overrides;
-    display_msg(message_str(2022, 101));
+    display_mstr(101);
 end
 
 procedure description_p_proc
 begin
     script_overrides;
-    display_msg(message_str(2022, 102));
+    display_mstr(102);
 end
 
 procedure use_skill_on_p_proc
@@ -143,7 +143,7 @@ procedure talk_p_proc
 begin
     set_local_var(3, 0);
     set_local_var(4, 0);
-    if (get_critter_stat(dude_obj, 4) < 4) then begin
+    if (dude_iq < 4) then begin
         start_gdialog(2022, self_obj, 4, -1, -1);
         gsay_start;
         call Node012();
@@ -170,156 +170,156 @@ begin
         call Node002a();
     end
     else begin
-        gsay_reply(2022, 106);
-        giq_option(4, 2022, 113, Node999, 50);
+        Reply(106);
+        giq_option(4, NAME, 113, Node999, NEUTRAL_REACTION);
     end
 end
 
 procedure Node002a
 begin
-    gsay_reply(2022, 107);
-    giq_option(4, 2022, 114, Node002b, 50);
+    Reply(107);
+    giq_option(4, NAME, 114, Node002b, NEUTRAL_REACTION);
 end
 
 procedure Node002b
 begin
-    gsay_reply(2022, 108);
-    giq_option(4, 2022, 109, Node003, 50);
-    giq_option(4, 2022, 111, Node005, 50);
-    giq_option(6, 2022, 130, Node015, 50);
-    giq_option(6, 2022, 110, Node100, 50);
-    giq_option(4, 2022, 113, Node999, 50);
+    Reply(108);
+    giq_option(4, NAME, 109, Node003, NEUTRAL_REACTION);
+    giq_option(4, NAME, 111, Node005, NEUTRAL_REACTION);
+    giq_option(6, NAME, 130, Node015, NEUTRAL_REACTION);
+    giq_option(6, NAME, 110, Node100, NEUTRAL_REACTION);
+    giq_option(4, NAME, 113, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node002c
 begin
-    gsay_reply(2022, 403);
-    giq_option(4, 2022, 404, Node002d, 50);
-    giq_option(4, 2022, 405, Node002e, 50);
-    giq_option(4, 2022, 406, Node002f, 50);
-    giq_option(4, 2022, 407, Node002g, 50);
-    giq_option(4, 2022, 408, Node002h, 50);
-    giq_option(4, 2022, 430, Node002b, 50);
+    Reply(403);
+    giq_option(4, NAME, 404, Node002d, NEUTRAL_REACTION);
+    giq_option(4, NAME, 405, Node002e, NEUTRAL_REACTION);
+    giq_option(4, NAME, 406, Node002f, NEUTRAL_REACTION);
+    giq_option(4, NAME, 407, Node002g, NEUTRAL_REACTION);
+    giq_option(4, NAME, 408, Node002h, NEUTRAL_REACTION);
+    giq_option(4, NAME, 430, Node002b, NEUTRAL_REACTION);
 end
 
 procedure Node002d
 begin
     if (local_var(8) == 0) then begin
         set_local_var(8, 1);
-        gsay_reply(2022, 410);
-        critter_mod_skill(dude_obj, 12, 5);
+        Reply(410);
+        critter_mod_skill(dude_obj, SKILL_SCIENCE, 5);
         game_time_advance(60 * (60 * 10) * 2);
     end
     else begin
-        gsay_reply(2022, 420);
+        Reply(420);
     end
-    giq_option(4, 2022, 430, Node002c, 50);
+    giq_option(4, NAME, 430, Node002c, NEUTRAL_REACTION);
 end
 
 procedure Node002e
 begin
     if (local_var(9) == 0) then begin
         set_local_var(9, 1);
-        gsay_reply(2022, 411);
-        critter_mod_skill(dude_obj, 13, 5);
+        Reply(411);
+        critter_mod_skill(dude_obj, SKILL_REPAIR, 5);
         game_time_advance(60 * (60 * 10) * 2);
     end
     else begin
-        gsay_reply(2022, 421);
+        Reply(421);
     end
-    giq_option(4, 2022, 430, Node002c, 50);
+    giq_option(4, NAME, 430, Node002c, NEUTRAL_REACTION);
 end
 
 procedure Node002f
 begin
     if (local_var(10) == 0) then begin
         set_local_var(10, 1);
-        gsay_reply(2022, 412);
-        critter_mod_skill(dude_obj, 7, 5);
-        critter_mod_skill(dude_obj, 6, 5);
+        Reply(412);
+        critter_mod_skill(dude_obj, SKILL_DOCTOR, 5);
+        critter_mod_skill(dude_obj, SKILL_FIRST_AID, 5);
         game_time_advance(60 * (60 * 10) * 2);
     end
     else begin
-        gsay_reply(2022, 422);
+        Reply(422);
     end
-    giq_option(4, 2022, 430, Node002c, 50);
+    giq_option(4, NAME, 430, Node002c, NEUTRAL_REACTION);
 end
 
 procedure Node002g
 begin
     if (local_var(11) == 0) then begin
         set_local_var(11, 1);
-        gsay_reply(2022, 413);
-        critter_mod_skill(dude_obj, 4, 5);
-        critter_mod_skill(dude_obj, 5, 5);
-        critter_mod_skill(dude_obj, 1, 5);
-        critter_mod_skill(dude_obj, 0, 5);
-        critter_mod_skill(dude_obj, 2, 5);
+        Reply(413);
+        critter_mod_skill(dude_obj, SKILL_MELEE, 5);
+        critter_mod_skill(dude_obj, SKILL_THROWING, 5);
+        critter_mod_skill(dude_obj, SKILL_BIG_GUNS, 5);
+        critter_mod_skill(dude_obj, SKILL_SMALL_GUNS, 5);
+        critter_mod_skill(dude_obj, SKILL_ENERGY_WEAPONS, 5);
         game_time_advance(60 * (60 * 10) * 4);
     end
     else begin
-        gsay_reply(2022, 423);
+        Reply(423);
     end
-    giq_option(4, 2022, 430, Node002c, 50);
+    giq_option(4, NAME, 430, Node002c, NEUTRAL_REACTION);
 end
 
 procedure Node002h
 begin
     if (local_var(12) == 0) then begin
         set_local_var(12, 1);
-        gsay_reply(2022, 414);
-        critter_mod_skill(dude_obj, 14, 5);
+        Reply(414);
+        critter_mod_skill(dude_obj, SKILL_SPEECH, 5);
         game_time_advance(60 * (60 * 10) * 2);
     end
     else begin
-        gsay_reply(2022, 424);
+        Reply(424);
     end
-    giq_option(4, 2022, 430, Node002c, 50);
+    giq_option(4, NAME, 430, Node002c, NEUTRAL_REACTION);
 end
 
 procedure Node003
 begin
-    if ((get_critter_stat(dude_obj, 4) > 5) and (has_skill(dude_obj, 12) > 69)) then begin
-        gsay_reply(2022, 115);
-        giq_option(4, 2022, 116, Node006, 50);
-        giq_option(4, 2022, 117, Node007, 50);
-        giq_option(4, 2022, 112, Node002b, 50);
+    if ((dude_iq > 5) and (dude_skill(SKILL_SCIENCE) > 69)) then begin
+        Reply(115);
+        giq_option(4, NAME, 116, Node006, NEUTRAL_REACTION);
+        giq_option(4, NAME, 117, Node007, NEUTRAL_REACTION);
+        giq_option(4, NAME, 112, Node002b, NEUTRAL_REACTION);
     end
     else begin
-        gsay_reply(2022, 118);
-        giq_option(4, 2022, 112, Node002b, 50);
+        Reply(118);
+        giq_option(4, NAME, 112, Node002b, NEUTRAL_REACTION);
     end
 end
 
 procedure Node004
 begin
-    gsay_reply(2022, 122);
-    giq_option(4, 2022, 402, Node002c, 50);
-    giq_option(4, 2022, 112, Node002b, 50);
+    Reply(122);
+    giq_option(4, NAME, 402, Node002c, NEUTRAL_REACTION);
+    giq_option(4, NAME, 112, Node002b, NEUTRAL_REACTION);
 end
 
 procedure Node004a
 begin
-    gsay_reply(2022, 122);
-    giq_option(4, 2022, 402, Node002c, 50);
-    giq_option(4, 2022, 112, Node002b, 50);
+    Reply(122);
+    giq_option(4, NAME, 402, Node002c, NEUTRAL_REACTION);
+    giq_option(4, NAME, 112, Node002b, NEUTRAL_REACTION);
 end
 
 procedure Node004b
 begin
-    gsay_reply(2022, 121);
-    giq_option(4, 2022, 112, Node002b, 50);
+    Reply(121);
+    giq_option(4, NAME, 112, Node002b, NEUTRAL_REACTION);
 end
 
 procedure Node005
 begin
     set_local_var(5, 0);
     set_local_var(6, 0);
-    gsay_reply(2022, message_str(2022, 220) + local_var(3) + message_str(2022, 221) + local_var(4) + message_str(2022, 222));
-    giq_option(4, 2022, 201, Node005a, 50);
-    giq_option(4, 2022, 202, Node005b, 50);
-    giq_option(4, 2022, 203, Node005c, 50);
-    giq_option(4, 2022, 204, Node002b, 50);
+    Reply(mstr(220) + local_var(3) + mstr(221) + local_var(4) + mstr(222));
+    giq_option(4, NAME, 201, Node005a, NEUTRAL_REACTION);
+    giq_option(4, NAME, 202, Node005b, NEUTRAL_REACTION);
+    giq_option(4, NAME, 203, Node005c, NEUTRAL_REACTION);
+    giq_option(4, NAME, 204, Node002b, NEUTRAL_REACTION);
 end
 
 procedure Node005a
@@ -342,49 +342,49 @@ end
 
 procedure Node006
 begin
-    gsay_reply(2022, 119);
-    giq_option(4, 2022, 123, Node006a, 50);
-    giq_option(4, 2022, 112, Node003, 50);
+    Reply(119);
+    giq_option(4, NAME, 123, Node006a, NEUTRAL_REACTION);
+    giq_option(4, NAME, 112, Node003, NEUTRAL_REACTION);
 end
 
 procedure Node006a
 begin
-    gsay_reply(2022, 124);
-    giq_option(4, 2022, 400, Node013, 50);
-    giq_option(4, 2022, 112, Node003, 50);
+    Reply(124);
+    giq_option(4, NAME, 400, Node013, NEUTRAL_REACTION);
+    giq_option(4, NAME, 112, Node003, NEUTRAL_REACTION);
 end
 
 procedure Node007
 begin
-    gsay_reply(2022, 120);
-    giq_option(4, 2022, 125, Node007a, 50);
-    giq_option(4, 2022, 112, Node003, 50);
+    Reply(120);
+    giq_option(4, NAME, 125, Node007a, NEUTRAL_REACTION);
+    giq_option(4, NAME, 112, Node003, NEUTRAL_REACTION);
 end
 
 procedure Node007a
 begin
-    gsay_reply(2022, 126);
-    giq_option(4, 2022, 112, Node003, 50);
+    Reply(126);
+    giq_option(4, NAME, 112, Node003, NEUTRAL_REACTION);
 end
 
 procedure Node008
 begin
     set_local_var(3, local_var(3) + 1);
-    gsay_reply(2022, message_str(2022, 205) + message_str(2022, local_var(6)) + message_str(2022, 206));
-    giq_option(4, 2022, 112, Node005, 50);
+    Reply(mstr(205) + mstr(local_var(6)) + mstr(206));
+    giq_option(4, NAME, 112, Node005, NEUTRAL_REACTION);
 end
 
 procedure Node009
 begin
     set_local_var(4, local_var(4) + 1);
-    gsay_reply(2022, message_str(2022, 205) + message_str(2022, local_var(6)) + message_str(2022, 207));
-    giq_option(4, 2022, 112, Node005, 50);
+    Reply(mstr(205) + mstr(local_var(6)) + mstr(207));
+    giq_option(4, NAME, 112, Node005, NEUTRAL_REACTION);
 end
 
 procedure Node010
 begin
-    gsay_reply(2022, message_str(2022, 205) + message_str(2022, local_var(6)) + message_str(2022, 208));
-    giq_option(4, 2022, 112, Node005, 50);
+    Reply(mstr(205) + mstr(local_var(6)) + mstr(208));
+    giq_option(4, NAME, 112, Node005, NEUTRAL_REACTION);
 end
 
 procedure Node011
@@ -435,68 +435,68 @@ end
 
 procedure Node012
 begin
-    gsay_reply(2022, 300);
-    giq_option(1, 2022, 301, Node999, 50);
-    giq_option(1, 2022, 302, Node999, 50);
+    Reply(300);
+    giq_option(1, NAME, 301, Node999, NEUTRAL_REACTION);
+    giq_option(1, NAME, 302, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node013
 begin
-    gsay_reply(2022, 500);
-    giq_option(4, 2022, 400, Node014, 50);
-    giq_option(4, 2022, 401, Node003, 50);
+    Reply(500);
+    giq_option(4, NAME, 400, Node014, NEUTRAL_REACTION);
+    giq_option(4, NAME, 401, Node003, NEUTRAL_REACTION);
 end
 
 procedure Node014
 begin
-    gsay_reply(2022, 501);
-    giq_option(4, 2022, 502, Node014a, 50);
-    giq_option(4, 2022, 401, Node003, 50);
-    giq_option(4, 2022, 113, Node999, 50);
+    Reply(501);
+    giq_option(4, NAME, 502, Node014a, NEUTRAL_REACTION);
+    giq_option(4, NAME, 401, Node003, NEUTRAL_REACTION);
+    giq_option(4, NAME, 113, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node014a
 begin
-    gsay_reply(2022, 503);
-    giq_option(4, 2022, 401, Node003, 50);
-    giq_option(4, 2022, 113, Node999, 50);
+    Reply(503);
+    giq_option(4, NAME, 401, Node003, NEUTRAL_REACTION);
+    giq_option(4, NAME, 113, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node015
 begin
-    gsay_reply(2022, 130);
-    if (global_var(1133) == 3) then begin
-        giq_option(4, 2022, 132, Node019, 50);
+    Reply(130);
+    if (global_var(GVAR_DEN_ORPHANAGE) == 3) then begin
+        giq_option(4, NAME, 132, Node019, NEUTRAL_REACTION);
     end
     else begin
-        giq_option(4, 2022, 131, Node002b, 50);
+        giq_option(4, NAME, 131, Node002b, NEUTRAL_REACTION);
     end
 end
 
 procedure Node016
 begin
-    gsay_reply(2022, 141);
-    giq_option(6, 2022, 142, Node017, 50);
-    giq_option(4, 2022, 143, Node100, 50);
+    Reply(141);
+    giq_option(6, NAME, 142, Node017, NEUTRAL_REACTION);
+    giq_option(4, NAME, 143, Node100, NEUTRAL_REACTION);
 end
 
 procedure Node017
 begin
-    gsay_reply(2022, 144);
-    giq_option(4, 2022, 145, Node018, 50);
+    Reply(144);
+    giq_option(4, NAME, 145, Node018, NEUTRAL_REACTION);
 end
 
 procedure Node018
 begin
-    gsay_reply(2022, 146);
-    giq_option(4, 2022, 147, Node900, 50);
+    Reply(146);
+    giq_option(4, NAME, 147, Node900, NEUTRAL_REACTION);
 end
 
 procedure Node019
 begin
-    gsay_reply(2022, 150);
-    giq_option(4, 2022, 112, Node002b, 50);
-    giq_option(4, 2022, 113, Node999, 50);
+    Reply(150);
+    giq_option(4, NAME, 112, Node002b, NEUTRAL_REACTION);
+    giq_option(4, NAME, 113, Node999, NEUTRAL_REACTION);
     if (obj_is_carrying_obj_pid(dude_obj, 664)) then begin
         removed_qty := obj_is_carrying_obj_pid(dude_obj, 664);
     end
@@ -507,17 +507,17 @@ begin
         restock_obj := obj_carrying_pid_obj(dude_obj, 664);
         if (obj_type(self_obj) == 1) then begin
             restock_obj := obj_carrying_pid_obj(dude_obj, 664);
-            if (critter_inven_obj(self_obj, 0) == restock_obj) then begin
-                if (obj_item_subtype(critter_inven_obj(dude_obj, 0)) == 0) then begin
-                    restock_obj := critter_inven_obj(dude_obj, 0);
+            if (critter_inven_obj(self_obj, INVEN_TYPE_WORN) == restock_obj) then begin
+                if (obj_item_subtype(critter_inven_obj(dude_obj, INVEN_TYPE_WORN)) == 0) then begin
+                    restock_obj := critter_inven_obj(dude_obj, INVEN_TYPE_WORN);
                     debug_msg("armour pid == " + obj_pid(restock_obj));
                     rm_obj_from_inven(dude_obj, restock_obj);
                     add_obj_to_inven(dude_obj, restock_obj);
                 end
             end
             else begin
-                if ((critter_inven_obj(dude_obj, 2) == restock_obj) or (critter_inven_obj(dude_obj, 1) == restock_obj)) then begin
-                    metarule(43, dude_obj);
+                if ((critter_inven_obj(dude_obj, INVEN_TYPE_LEFT_HAND) == restock_obj) or (critter_inven_obj(dude_obj, INVEN_TYPE_RIGHT_HAND) == restock_obj)) then begin
+                    inven_unwield(dude_obj);
                 end
             end
         end
@@ -526,64 +526,64 @@ begin
     end
     restock_obj := 0;
     restock_amt := 0;
-    set_global_var(1133, 4);
-    display_msg(message_str(2022, 1000));
+    set_global_var(GVAR_DEN_ORPHANAGE, 4);
+    display_mstr(1000);
     give_exp_points(750);
-    display_msg(message_str(14, 100) + 750 + message_str(14, 101));
+    display_msg(g_mstr(100) + 750 + g_mstr(101));
 end
 
 procedure Node900
 begin
-    set_global_var(963, 2);
+    set_global_var(GVAR_RESERVED, 2);
     set_local_var(7, 1);
-    set_global_var(63, -30);
-    display_msg(message_str(2022, 2000));
+    set_global_var(GVAR_TOWN_REP_ABBEY, -30);
+    display_mstr(2000);
     give_exp_points(500);
-    display_msg(message_str(14, 100) + 500 + message_str(14, 101));
-    set_global_var(0, global_var(0) + -5);
+    display_msg(g_mstr(100) + 500 + g_mstr(101));
+    set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + -5);
     debug_msg("Player gains " + -5 + " Karma Points.");
-    set_global_var(37, 0);
-    set_global_var(38, 0);
-    set_global_var(39, 0);
-    set_global_var(40, 0);
-    set_global_var(41, 0);
-    set_global_var(42, 0);
-    set_global_var(43, 0);
-    set_global_var(44, 0);
-    set_global_var(45, 0);
-    if (global_var(0) >= 1000) then begin
-        set_global_var(37, 1);
+    set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+    set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+    set_global_var(GVAR_KARMA_DEFENDER, 0);
+    set_global_var(GVAR_KARMA_WANDERER, 0);
+    set_global_var(GVAR_KARMA_BETRAYER, 0);
+    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+    set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+    set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+    if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
     end
     else begin
-        if (global_var(0) >= 750) then begin
-            set_global_var(38, 1);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+            set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
         end
         else begin
-            if (global_var(0) >= 500) then begin
-                set_global_var(39, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
             end
             else begin
-                if (global_var(0) >= 250) then begin
-                    set_global_var(40, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                    set_global_var(GVAR_KARMA_DEFENDER, 1);
                 end
                 else begin
-                    if (global_var(0) > -250) then begin
-                        set_global_var(41, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                        set_global_var(GVAR_KARMA_WANDERER, 1);
                     end
                     else begin
-                        if (global_var(0) > -500) then begin
-                            set_global_var(42, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                            set_global_var(GVAR_KARMA_BETRAYER, 1);
                         end
                         else begin
-                            if (global_var(0) > -750) then begin
-                                set_global_var(43, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                             end
                             else begin
-                                if (global_var(0) > -1000) then begin
-                                    set_global_var(44, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                    set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                 end
                                 else begin
-                                    set_global_var(45, 1);
+                                    set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                 end
                             end
                         end
@@ -596,8 +596,8 @@ end
 
 procedure Node901
 begin
-    gsay_reply(2022, 148);
-    giq_option(4, 2022, 147, Node999, 50);
+    Reply(148);
+    giq_option(4, NAME, 147, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node999
@@ -610,45 +610,45 @@ end
 
 procedure Node100
 begin
-    if (has_skill(dude_obj, 12) > 110) then begin
-        gsay_reply(2022, 122);
-        giq_option(4, 2022, 600, Node150, 50);
-        giq_option(4, 2022, 601, Node200, 50);
-        giq_option(4, 2022, 602, Node250, 50);
-        giq_option(4, 2022, 603, Node300, 50);
-        giq_option(6, 2022, 604, Node350, 50);
-        giq_option(4, 2022, 112, Node002b, 50);
+    if (dude_skill(SKILL_SCIENCE) > 110) then begin
+        Reply(122);
+        giq_option(4, NAME, 600, Node150, NEUTRAL_REACTION);
+        giq_option(4, NAME, 601, Node200, NEUTRAL_REACTION);
+        giq_option(4, NAME, 602, Node250, NEUTRAL_REACTION);
+        giq_option(4, NAME, 603, Node300, NEUTRAL_REACTION);
+        giq_option(6, NAME, 604, Node350, NEUTRAL_REACTION);
+        giq_option(4, NAME, 112, Node002b, NEUTRAL_REACTION);
     end
     else begin
-        gsay_reply(2022, 121);
-        giq_option(4, 2022, 112, Node002b, 50);
+        Reply(121);
+        giq_option(4, NAME, 112, Node002b, NEUTRAL_REACTION);
     end
-    giq_option(4, 2022, 140, Node016, 50);
+    giq_option(4, NAME, 140, Node016, NEUTRAL_REACTION);
 end
 
 procedure Node150
 begin
     if ((local_var(13) bwand 1) and (local_var(13) bwand 2)) then begin
-        gsay_reply(2022, message_str(2022, 625) + message_str(2022, 626));
+        Reply(mstr(625) + mstr(626));
     end
     else begin
-        gsay_reply(2022, message_str(2022, 625) + message_str(2022, 627));
+        Reply(mstr(625) + mstr(627));
     end
     if ((local_var(13) bwand 1) == 0) then begin
-        giq_option(4, 2022, 630, Node151, 50);
+        giq_option(4, NAME, 630, Node151, NEUTRAL_REACTION);
     end
     if ((local_var(13) bwand 2) == 0) then begin
-        giq_option(4, 2022, 631, Node152, 50);
+        giq_option(4, NAME, 631, Node152, NEUTRAL_REACTION);
     end
-    giq_option(4, 2022, 635, Node100, 50);
+    giq_option(4, NAME, 635, Node100, NEUTRAL_REACTION);
 end
 
 procedure Node151
 begin
     gfade_out(60);
     set_local_var(13, local_var(13) bwor 1);
-    display_msg(message_str(2022, 650));
-    critter_mod_skill(dude_obj, 16, 10);
+    display_mstr(650);
+    critter_mod_skill(dude_obj, SKILL_GAMBLING, 10);
     game_time_advance(72000);
     call Node151a();
 end
@@ -656,16 +656,16 @@ end
 procedure Node151a
 begin
     gfade_in(60);
-    gsay_reply(2022, 640);
-    giq_option(4, 2022, 635, Node150, 50);
+    Reply(640);
+    giq_option(4, NAME, 635, Node150, NEUTRAL_REACTION);
 end
 
 procedure Node152
 begin
     gfade_out(60);
     set_local_var(13, local_var(13) bwor 2);
-    display_msg(message_str(2022, 651));
-    critter_mod_skill(dude_obj, 9, 10);
+    display_mstr(651);
+    critter_mod_skill(dude_obj, SKILL_LOCKPICK, 10);
     game_time_advance(72000);
     call Node152a();
 end
@@ -673,30 +673,30 @@ end
 procedure Node152a
 begin
     gfade_in(60);
-    gsay_reply(2022, 641);
-    giq_option(4, 2022, 635, Node150, 50);
+    Reply(641);
+    giq_option(4, NAME, 635, Node150, NEUTRAL_REACTION);
 end
 
 procedure Node200
 begin
     if (local_var(13) bwand 4) then begin
-        gsay_reply(2022, message_str(2022, 625) + message_str(2022, 626));
+        Reply(mstr(625) + mstr(626));
     end
     else begin
-        gsay_reply(2022, message_str(2022, 625) + message_str(2022, 627));
+        Reply(mstr(625) + mstr(627));
     end
     if ((local_var(13) bwand 4) == 0) then begin
-        giq_option(4, 2022, 632, Node201, 50);
+        giq_option(4, NAME, 632, Node201, NEUTRAL_REACTION);
     end
-    giq_option(4, 2022, 635, Node100, 50);
+    giq_option(4, NAME, 635, Node100, NEUTRAL_REACTION);
 end
 
 procedure Node201
 begin
     gfade_out(60);
     set_local_var(13, local_var(13) bwor 4);
-    display_msg(message_str(2022, 652));
-    critter_mod_skill(dude_obj, 8, 10);
+    display_mstr(652);
+    critter_mod_skill(dude_obj, SKILL_SNEAK, 10);
     game_time_advance(72000);
     call Node201a();
 end
@@ -704,36 +704,36 @@ end
 procedure Node201a
 begin
     gfade_in(60);
-    gsay_reply(2022, 642);
-    giq_option(4, 2022, 635, Node200, 50);
+    Reply(642);
+    giq_option(4, NAME, 635, Node200, NEUTRAL_REACTION);
 end
 
 procedure Node250
 begin
     if ((local_var(13) bwand 8) and (local_var(13) bwand 16) and (local_var(13) bwand 1024)) then begin
-        gsay_reply(2022, message_str(2022, 625) + message_str(2022, 626));
+        Reply(mstr(625) + mstr(626));
     end
     else begin
-        gsay_reply(2022, message_str(2022, 625) + message_str(2022, 627));
+        Reply(mstr(625) + mstr(627));
     end
     if ((local_var(13) bwand 8) == 0) then begin
-        giq_option(4, 2022, 633, Node251, 50);
+        giq_option(4, NAME, 633, Node251, NEUTRAL_REACTION);
     end
     if ((local_var(13) bwand 16) == 0) then begin
-        giq_option(4, 2022, 634, Node252, 50);
+        giq_option(4, NAME, 634, Node252, NEUTRAL_REACTION);
     end
     if ((local_var(13) bwand 1024) == 0) then begin
-        giq_option(4, 2022, 750, Node253, 50);
+        giq_option(4, NAME, 750, Node253, NEUTRAL_REACTION);
     end
-    giq_option(4, 2022, 635, Node100, 50);
+    giq_option(4, NAME, 635, Node100, NEUTRAL_REACTION);
 end
 
 procedure Node251
 begin
     gfade_out(60);
     set_local_var(13, local_var(13) bwor 8);
-    display_msg(message_str(2022, 653));
-    critter_mod_skill(dude_obj, 15, 10);
+    display_mstr(653);
+    critter_mod_skill(dude_obj, SKILL_BARTER, 10);
     game_time_advance(72000);
     call Node251a();
 end
@@ -741,16 +741,16 @@ end
 procedure Node251a
 begin
     gfade_in(60);
-    gsay_reply(2022, 643);
-    giq_option(4, 2022, 635, Node250, 50);
+    Reply(643);
+    giq_option(4, NAME, 635, Node250, NEUTRAL_REACTION);
 end
 
 procedure Node252
 begin
     gfade_out(60);
     set_local_var(13, local_var(13) bwor 16);
-    display_msg(message_str(2022, 654));
-    critter_mod_skill(dude_obj, 14, 10);
+    display_mstr(654);
+    critter_mod_skill(dude_obj, SKILL_SPEECH, 10);
     game_time_advance(72000);
     call Node252a();
 end
@@ -758,8 +758,8 @@ end
 procedure Node252a
 begin
     gfade_in(60);
-    gsay_reply(2022, 644);
-    giq_option(4, 2022, 635, Node250, 50);
+    Reply(644);
+    giq_option(4, NAME, 635, Node250, NEUTRAL_REACTION);
 end
 
 procedure Node253
@@ -772,47 +772,47 @@ end
 procedure Node253a
 begin
     gfade_in(60);
-    gsay_reply(2022, 755);
-    giq_option(4, 2022, 756, Node253b, 50);
-    if (get_critter_stat(dude_obj, 34)) then begin
-        if (global_var(6) != 0) then begin
-            giq_option(4, 2022, message_str(2022, 757) + message_str(2022, 754), Node253b, 50);
+    Reply(755);
+    giq_option(4, NAME, 756, Node253b, NEUTRAL_REACTION);
+    if (dude_gender) then begin
+        if (global_var(GVAR_PLAYER_MARRIED) != 0) then begin
+            giq_option(4, NAME, mstr(757) + mstr(754), Node253b, NEUTRAL_REACTION);
         end
         else begin
-            giq_option(4, 2022, message_str(2022, 757) + message_str(2022, 759), Node253b, 50);
+            giq_option(4, NAME, mstr(757) + mstr(759), Node253b, NEUTRAL_REACTION);
         end
     end
     else begin
-        if (global_var(6) != 0) then begin
-            giq_option(4, 2022, 758, Node253b, 50);
+        if (global_var(GVAR_PLAYER_MARRIED) != 0) then begin
+            giq_option(4, NAME, 758, Node253b, NEUTRAL_REACTION);
         end
     end
-    giq_option(4, 2022, 635, Node250, 50);
+    giq_option(4, NAME, 635, Node250, NEUTRAL_REACTION);
 end
 
 procedure Node253b
 begin
     set_local_var(13, local_var(13) bwor 1024);
-    gsay_reply(2022, 760);
-    if ((global_var(6) != 0) and get_critter_stat(dude_obj, 34)) then begin
-        giq_option(4, 2022, 766, Node253c, 50);
+    Reply(760);
+    if ((global_var(GVAR_PLAYER_MARRIED) != 0) and dude_gender) then begin
+        giq_option(4, NAME, 766, Node253c, NEUTRAL_REACTION);
     end
     else begin
-        giq_option(4, 2022, 767, Node253c, 50);
+        giq_option(4, NAME, 767, Node253c, NEUTRAL_REACTION);
     end
-    giq_option(4, 2022, 768, Node253c, 50);
+    giq_option(4, NAME, 768, Node253c, NEUTRAL_REACTION);
 end
 
 procedure Node253c
 begin
-    gsay_reply(2022, 765);
-    giq_option(4, 2022, 635, Node250, 50);
+    Reply(765);
+    giq_option(4, NAME, 635, Node250, NEUTRAL_REACTION);
 end
 
 procedure Node300
 begin
-    gsay_reply(2022, message_str(2022, 625) + message_str(2022, 626));
-    giq_option(4, 2022, 635, Node100, 50);
+    Reply(mstr(625) + mstr(626));
+    giq_option(4, NAME, 635, Node100, NEUTRAL_REACTION);
 end
 
 procedure Node350
@@ -847,27 +847,27 @@ begin
         LVar1 := LVar1 + 1;
     end
     if (LVar0 == 5) then begin
-        gsay_reply(2022, message_str(2022, 625) + message_str(2022, 626));
+        Reply(mstr(625) + mstr(626));
     end
     else begin
-        gsay_reply(2022, message_str(2022, 625) + message_str(2022, 627));
+        Reply(mstr(625) + mstr(627));
     end
     if (((local_var(13) bwand 32) == 0) and (LVar1 == 4)) then begin
-        giq_option(4, 2022, 700, Node351, 50);
+        giq_option(4, NAME, 700, Node351, NEUTRAL_REACTION);
     end
     if ((local_var(13) bwand 64) == 0) then begin
-        giq_option(4, 2022, 701, Node352, 50);
+        giq_option(4, NAME, 701, Node352, NEUTRAL_REACTION);
     end
     if ((local_var(13) bwand 128) == 0) then begin
-        giq_option(4, 2022, 702, Node353, 50);
+        giq_option(4, NAME, 702, Node353, NEUTRAL_REACTION);
     end
     if ((local_var(13) bwand 256) == 0) then begin
-        giq_option(4, 2022, 703, Node354, 50);
+        giq_option(4, NAME, 703, Node354, NEUTRAL_REACTION);
     end
     if ((local_var(13) bwand 512) == 0) then begin
-        giq_option(4, 2022, 704, Node355, 50);
+        giq_option(4, NAME, 704, Node355, NEUTRAL_REACTION);
     end
-    giq_option(4, 2022, 635, Node100, 50);
+    giq_option(4, NAME, 635, Node100, NEUTRAL_REACTION);
 end
 
 procedure Node351
@@ -880,23 +880,23 @@ end
 procedure Node351a
 begin
     gfade_in(60);
-    gsay_reply(2022, 710);
-    giq_option(4, 2022, 502, Node351b, 50);
-    giq_option(4, 2022, 635, Node350, 50);
+    Reply(710);
+    giq_option(4, NAME, 502, Node351b, NEUTRAL_REACTION);
+    giq_option(4, NAME, 635, Node350, NEUTRAL_REACTION);
 end
 
 procedure Node351b
 begin
-    gsay_reply(2022, 711);
-    giq_option(4, 2022, 502, Node351c, 50);
-    giq_option(4, 2022, 635, Node350, 50);
+    Reply(711);
+    giq_option(4, NAME, 502, Node351c, NEUTRAL_REACTION);
+    giq_option(4, NAME, 635, Node350, NEUTRAL_REACTION);
 end
 
 procedure Node351c
 begin
     set_local_var(13, local_var(13) bwor 32);
-    gsay_reply(2022, 712);
-    giq_option(4, 2022, 635, Node350, 50);
+    Reply(712);
+    giq_option(4, NAME, 635, Node350, NEUTRAL_REACTION);
 end
 
 procedure Node352
@@ -909,23 +909,23 @@ end
 procedure Node352a
 begin
     gfade_in(60);
-    gsay_reply(2022, 715);
-    giq_option(4, 2022, 502, Node352b, 50);
-    giq_option(4, 2022, 635, Node350, 50);
+    Reply(715);
+    giq_option(4, NAME, 502, Node352b, NEUTRAL_REACTION);
+    giq_option(4, NAME, 635, Node350, NEUTRAL_REACTION);
 end
 
 procedure Node352b
 begin
-    gsay_reply(2022, 716);
-    giq_option(4, 2022, 502, Node352c, 50);
-    giq_option(4, 2022, 635, Node350, 50);
+    Reply(716);
+    giq_option(4, NAME, 502, Node352c, NEUTRAL_REACTION);
+    giq_option(4, NAME, 635, Node350, NEUTRAL_REACTION);
 end
 
 procedure Node352c
 begin
     set_local_var(13, local_var(13) bwor 64);
-    gsay_reply(2022, 717);
-    giq_option(4, 2022, 635, Node350, 50);
+    Reply(717);
+    giq_option(4, NAME, 635, Node350, NEUTRAL_REACTION);
 end
 
 procedure Node353
@@ -938,16 +938,16 @@ end
 procedure Node353a
 begin
     gfade_in(60);
-    gsay_reply(2022, 720);
-    giq_option(4, 2022, 502, Node353b, 50);
-    giq_option(4, 2022, 635, Node350, 50);
+    Reply(720);
+    giq_option(4, NAME, 502, Node353b, NEUTRAL_REACTION);
+    giq_option(4, NAME, 635, Node350, NEUTRAL_REACTION);
 end
 
 procedure Node353b
 begin
     set_local_var(13, local_var(13) bwor 128);
-    gsay_reply(2022, 721);
-    giq_option(4, 2022, 635, Node350, 50);
+    Reply(721);
+    giq_option(4, NAME, 635, Node350, NEUTRAL_REACTION);
 end
 
 procedure Node354
@@ -960,16 +960,16 @@ end
 procedure Node354a
 begin
     gfade_in(60);
-    gsay_reply(2022, 725);
-    giq_option(4, 2022, 502, Node354b, 50);
-    giq_option(4, 2022, 635, Node350, 50);
+    Reply(725);
+    giq_option(4, NAME, 502, Node354b, NEUTRAL_REACTION);
+    giq_option(4, NAME, 635, Node350, NEUTRAL_REACTION);
 end
 
 procedure Node354b
 begin
     set_local_var(13, local_var(13) bwor 256);
-    gsay_reply(2022, 726);
-    giq_option(4, 2022, 635, Node350, 50);
+    Reply(726);
+    giq_option(4, NAME, 635, Node350, NEUTRAL_REACTION);
 end
 
 procedure Node355
@@ -982,29 +982,28 @@ end
 procedure Node355a
 begin
     gfade_in(60);
-    gsay_reply(2022, 730);
-    giq_option(4, 2022, 502, Node355b, 50);
-    giq_option(4, 2022, 635, Node350, 50);
+    Reply(730);
+    giq_option(4, NAME, 502, Node355b, NEUTRAL_REACTION);
+    giq_option(4, NAME, 635, Node350, NEUTRAL_REACTION);
 end
 
 procedure Node355b
 begin
-    gsay_reply(2022, 731);
-    giq_option(4, 2022, 502, Node355c, 50);
-    giq_option(4, 2022, 635, Node350, 50);
+    Reply(731);
+    giq_option(4, NAME, 502, Node355c, NEUTRAL_REACTION);
+    giq_option(4, NAME, 635, Node350, NEUTRAL_REACTION);
 end
 
 procedure Node355c
 begin
-    gsay_reply(2022, 732);
-    giq_option(4, 2022, 502, Node355d, 50);
-    giq_option(4, 2022, 635, Node350, 50);
+    Reply(732);
+    giq_option(4, NAME, 502, Node355d, NEUTRAL_REACTION);
+    giq_option(4, NAME, 635, Node350, NEUTRAL_REACTION);
 end
 
 procedure Node355d
 begin
     set_local_var(13, local_var(13) bwor 512);
-    gsay_reply(2022, 733);
-    giq_option(4, 2022, 635, Node350, 50);
+    Reply(733);
+    giq_option(4, NAME, 635, Node350, NEUTRAL_REACTION);
 end
-
