diff --git a/Mods/MegaMod/Original/ACSHANIA.ssl b/Mods/MegaMod/ReDefine/ACSHANIA.ssl
index 17aff3c..6a6637c 100644
--- a/Mods/MegaMod/Original/ACSHANIA.ssl
+++ b/Mods/MegaMod/ReDefine/ACSHANIA.ssl
@@ -223,10 +223,10 @@ begin
         set_local_var(5, 1);
         if ((tile_distance_objs(self_obj, dude_obj) < 8) or obj_can_see_obj(self_obj, dude_obj)) then begin
             if (anim_busy(self_obj) == 0) then begin
-                global_temp := rotation_to_tile(tile_num(dude_obj), tile_num(self_obj));
+                global_temp := rotation_to_tile(dude_tile, tile_num(self_obj));
                 animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(self_obj), global_temp, random(3, 10)), 1);
                 global_temp := (global_temp + 1) % 6;
-                while ((anim_busy(self_obj) == 0) and (global_temp != rotation_to_tile(tile_num(dude_obj), tile_num(self_obj)))) do begin
+                while ((anim_busy(self_obj) == 0) and (global_temp != rotation_to_tile(dude_tile, tile_num(self_obj)))) do begin
                     debug_msg("flee loop: rot == " + global_temp);
                     animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(self_obj), global_temp, random(3, 10)), 1);
                     global_temp := (global_temp + 1) % 6;
@@ -234,8 +234,8 @@ begin
             end
         end
     end
-    if ((global_var(7) == 1) and obj_can_see_obj(self_obj, dude_obj)) then begin
-        attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+    if ((global_var(GVAR_ENEMY_ARROYO) == 1) and obj_can_see_obj(self_obj, dude_obj)) then begin
+        attack(dude_obj);
     end
 end
 
@@ -250,7 +250,7 @@ procedure talk_p_proc
 begin
     if (local_var(2) == 0) then begin
         set_local_var(2, 1);
-        set_local_var(3, (get_critter_stat(dude_obj, 3) - get_critter_stat(self_obj, 3)) * 5);
+        set_local_var(3, (dude_charisma - self_charisma) * 5);
         debug_msg("Base Reaction == " + local_var(3));
     end
     Static_Reaction := 0;
@@ -262,14 +262,14 @@ begin
     else begin
         reaction_bonus_karma := -1 * (-1 * reaction_bonus_karma / 1000);
     end
-    if (global_var(47) >= 0) then begin
-        reaction_bonus_town_rep := global_var(47) / 2;
+    if (global_var(GVAR_TOWN_REP_ARROYO) >= 0) then begin
+        reaction_bonus_town_rep := global_var(GVAR_TOWN_REP_ARROYO) / 2;
     end
     else begin
-        reaction_bonus_town_rep := -1 * (-1 * global_var(47) / 2);
+        reaction_bonus_town_rep := -1 * (-1 * global_var(GVAR_TOWN_REP_ARROYO) / 2);
     end
-    if (has_trait(2, dude_obj, 13)) then begin
-        if (get_critter_stat(dude_obj, 34) == get_critter_stat(self_obj, 34)) then begin
+    if (has_trait(TRAIT_TRAIT, dude_obj, TRAIT_sex_appeal)) then begin
+        if (dude_gender == self_gender) then begin
             Static_Reaction := Static_Reaction + -20;
         end
         else begin
@@ -295,52 +295,52 @@ begin
             Static_Reaction := Static_Reaction + reaction_bonus_town_rep;
         end
         debug_msg("Town Rep Bonus == " + Static_Reaction);
-        set_global_var(37, 0);
-        set_global_var(38, 0);
-        set_global_var(39, 0);
-        set_global_var(40, 0);
-        set_global_var(41, 0);
-        set_global_var(42, 0);
-        set_global_var(43, 0);
-        set_global_var(44, 0);
-        set_global_var(45, 0);
-        tmp_gen_rep := global_var(0);
-        if (has_trait(0, dude_obj, 95)) then begin
-            tmp_gen_rep := global_var(0) * 2;
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+        set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+        set_global_var(GVAR_KARMA_DEFENDER, 0);
+        set_global_var(GVAR_KARMA_WANDERER, 0);
+        set_global_var(GVAR_KARMA_BETRAYER, 0);
+        set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+        set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+        tmp_gen_rep := global_var(GVAR_PLAYER_REPUTATION);
+        if (has_trait(TRAIT_PERK, dude_obj, PERK_karma_beacon_perk)) then begin
+            tmp_gen_rep := global_var(GVAR_PLAYER_REPUTATION) * 2;
         end
         if (tmp_gen_rep >= 1000) then begin
-            set_global_var(37, 1);
+            set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
         end
         else begin
             if (tmp_gen_rep >= 750) then begin
-                set_global_var(38, 1);
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
             end
             else begin
                 if (tmp_gen_rep >= 500) then begin
-                    set_global_var(39, 1);
+                    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                 end
                 else begin
                     if (tmp_gen_rep >= 250) then begin
-                        set_global_var(40, 1);
+                        set_global_var(GVAR_KARMA_DEFENDER, 1);
                     end
                     else begin
                         if (tmp_gen_rep > -250) then begin
-                            set_global_var(41, 1);
+                            set_global_var(GVAR_KARMA_WANDERER, 1);
                         end
                         else begin
                             if (tmp_gen_rep > -500) then begin
-                                set_global_var(42, 1);
+                                set_global_var(GVAR_KARMA_BETRAYER, 1);
                             end
                             else begin
                                 if (tmp_gen_rep > -750) then begin
-                                    set_global_var(43, 1);
+                                    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                 end
                                 else begin
                                     if (tmp_gen_rep > -1000) then begin
-                                        set_global_var(44, 1);
+                                        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                     end
                                     else begin
-                                        set_global_var(45, 1);
+                                        set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                     end
                                 end
                             end
@@ -349,35 +349,35 @@ begin
                 end
             end
         end
-        if (global_var(37) == 1) then begin
+        if (global_var(GVAR_KARMA_HOLY_WARRIOR) == 1) then begin
             Static_Reaction := Static_Reaction - 20;
         end
         else begin
-            if (global_var(38) == 1) then begin
+            if (global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES) == 1) then begin
                 Static_Reaction := Static_Reaction - 15;
             end
             else begin
-                if (global_var(39) == 1) then begin
+                if (global_var(GVAR_KARMA_SHIELD_OF_HOPE) == 1) then begin
                     Static_Reaction := Static_Reaction - 10;
                 end
                 else begin
-                    if (global_var(40) == 1) then begin
+                    if (global_var(GVAR_KARMA_DEFENDER) == 1) then begin
                         Static_Reaction := Static_Reaction - 5;
                     end
                     else begin
-                        if (global_var(41) == 1) then begin
+                        if (global_var(GVAR_KARMA_WANDERER) == 1) then begin
                             Static_Reaction := Static_Reaction - 0;
                         end
                         else begin
-                            if (global_var(42) == 1) then begin
+                            if (global_var(GVAR_KARMA_BETRAYER) == 1) then begin
                                 Static_Reaction := Static_Reaction + -5;
                             end
                             else begin
-                                if (global_var(43) == 1) then begin
+                                if (global_var(GVAR_KARMA_SWORD_OF_DESPAIR) == 1) then begin
                                     Static_Reaction := Static_Reaction + -10;
                                 end
                                 else begin
-                                    if (global_var(44) == 1) then begin
+                                    if (global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE) == 1) then begin
                                         Static_Reaction := Static_Reaction + -15;
                                     end
                                     else begin
@@ -397,52 +397,52 @@ begin
         debug_msg("Karma Reaction Bonus == " + Static_Reaction);
         Static_Reaction := Static_Reaction + reaction_bonus_town_rep;
         debug_msg("Town Rep Bonus == " + Static_Reaction);
-        set_global_var(37, 0);
-        set_global_var(38, 0);
-        set_global_var(39, 0);
-        set_global_var(40, 0);
-        set_global_var(41, 0);
-        set_global_var(42, 0);
-        set_global_var(43, 0);
-        set_global_var(44, 0);
-        set_global_var(45, 0);
-        tmp_gen_rep := global_var(0);
-        if (has_trait(0, dude_obj, 95)) then begin
-            tmp_gen_rep := global_var(0) * 2;
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+        set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+        set_global_var(GVAR_KARMA_DEFENDER, 0);
+        set_global_var(GVAR_KARMA_WANDERER, 0);
+        set_global_var(GVAR_KARMA_BETRAYER, 0);
+        set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+        set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+        tmp_gen_rep := global_var(GVAR_PLAYER_REPUTATION);
+        if (has_trait(TRAIT_PERK, dude_obj, PERK_karma_beacon_perk)) then begin
+            tmp_gen_rep := global_var(GVAR_PLAYER_REPUTATION) * 2;
         end
         if (tmp_gen_rep >= 1000) then begin
-            set_global_var(37, 1);
+            set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
         end
         else begin
             if (tmp_gen_rep >= 750) then begin
-                set_global_var(38, 1);
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
             end
             else begin
                 if (tmp_gen_rep >= 500) then begin
-                    set_global_var(39, 1);
+                    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                 end
                 else begin
                     if (tmp_gen_rep >= 250) then begin
-                        set_global_var(40, 1);
+                        set_global_var(GVAR_KARMA_DEFENDER, 1);
                     end
                     else begin
                         if (tmp_gen_rep > -250) then begin
-                            set_global_var(41, 1);
+                            set_global_var(GVAR_KARMA_WANDERER, 1);
                         end
                         else begin
                             if (tmp_gen_rep > -500) then begin
-                                set_global_var(42, 1);
+                                set_global_var(GVAR_KARMA_BETRAYER, 1);
                             end
                             else begin
                                 if (tmp_gen_rep > -750) then begin
-                                    set_global_var(43, 1);
+                                    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                 end
                                 else begin
                                     if (tmp_gen_rep > -1000) then begin
-                                        set_global_var(44, 1);
+                                        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                     end
                                     else begin
-                                        set_global_var(45, 1);
+                                        set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                     end
                                 end
                             end
@@ -451,35 +451,35 @@ begin
                 end
             end
         end
-        if (global_var(37) == 1) then begin
+        if (global_var(GVAR_KARMA_HOLY_WARRIOR) == 1) then begin
             Static_Reaction := Static_Reaction + 20;
         end
         else begin
-            if (global_var(38) == 1) then begin
+            if (global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES) == 1) then begin
                 Static_Reaction := Static_Reaction + 15;
             end
             else begin
-                if (global_var(39) == 1) then begin
+                if (global_var(GVAR_KARMA_SHIELD_OF_HOPE) == 1) then begin
                     Static_Reaction := Static_Reaction + 10;
                 end
                 else begin
-                    if (global_var(40) == 1) then begin
+                    if (global_var(GVAR_KARMA_DEFENDER) == 1) then begin
                         Static_Reaction := Static_Reaction + 5;
                     end
                     else begin
-                        if (global_var(41) == 1) then begin
+                        if (global_var(GVAR_KARMA_WANDERER) == 1) then begin
                             Static_Reaction := Static_Reaction + 0;
                         end
                         else begin
-                            if (global_var(42) == 1) then begin
+                            if (global_var(GVAR_KARMA_BETRAYER) == 1) then begin
                                 Static_Reaction := Static_Reaction + -5;
                             end
                             else begin
-                                if (global_var(43) == 1) then begin
+                                if (global_var(GVAR_KARMA_SWORD_OF_DESPAIR) == 1) then begin
                                     Static_Reaction := Static_Reaction + -10;
                                 end
                                 else begin
-                                    if (global_var(44) == 1) then begin
+                                    if (global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE) == 1) then begin
                                         Static_Reaction := Static_Reaction + -15;
                                     end
                                     else begin
@@ -495,32 +495,32 @@ begin
         debug_msg("Good Person Reaction Bonus == " + Static_Reaction);
     end
     if (Slavery_Tolerant == 0) then begin
-        if (global_var(11) == 1) then begin
+        if (global_var(GVAR_REPUTATION_SLAVER) == 1) then begin
             Static_Reaction := Static_Reaction + (2 * -25);
         end
     end
     else begin
         if (Slavery_Tolerant == 1) then begin
-            if (global_var(11) == 1) then begin
+            if (global_var(GVAR_REPUTATION_SLAVER) == 1) then begin
                 Static_Reaction := Static_Reaction + -25;
             end
         end
         else begin
             if (Slavery_Tolerant == 3) then begin
-                if (global_var(11) == 1) then begin
+                if (global_var(GVAR_REPUTATION_SLAVER) == 1) then begin
                     Static_Reaction := Static_Reaction + 25;
                 end
             end
         end
     end
     debug_msg("Slaver + Aligned Reaction == " + Static_Reaction);
-    Static_Reaction := Static_Reaction + (10 * has_trait(0, dude_obj, 10));
+    Static_Reaction := Static_Reaction + (10 * has_trait(TRAIT_PERK, dude_obj, PERK_presence));
     debug_msg("Presence Reaction == " + Static_Reaction);
-    if (global_var(1) >= 2) then begin
+    if (global_var(GVAR_CHILDKILLER_REPUTATION) >= 2) then begin
         Static_Reaction := Static_Reaction + -30;
     end
     debug_msg("Childkiller Reaction == " + Static_Reaction);
-    if (has_trait(0, dude_obj, 39)) then begin
+    if (has_trait(TRAIT_PERK, dude_obj, PERK_cult_of_personality)) then begin
         if (Evil_Critter == 1) then begin
             if (Static_Reaction > 0) then begin
                 Static_Reaction := Static_Reaction * -1;
@@ -534,13 +534,13 @@ begin
     end
     set_local_var(0, local_var(3) + Static_Reaction);
     debug_msg("Initial Reaction == " + local_var(0));
-    if ((global_var(7) == 1) or (local_var(6) == 1)) then begin
+    if ((global_var(GVAR_ENEMY_ARROYO) == 1) or (local_var(6) == 1)) then begin
         call Node009();
     end
     else begin
-        if (cur_map_index == 4) then begin
+        if (CUR_MAP_ARVILLAG) then begin
             if (local_var(9) == 0) then begin
-                if ((get_critter_stat(dude_obj, 4) <= 3) and (local_var(4) == 0)) then begin
+                if ((dude_is_stupid) and (local_var(4) == 0)) then begin
                     start_gdialog(1364, self_obj, 4, -1, -1);
                     gsay_start;
                     call Node007();
@@ -548,7 +548,7 @@ begin
                     end_dialogue;
                 end
                 else begin
-                    if (get_critter_stat(dude_obj, 4) <= 3) then begin
+                    if (dude_is_stupid) then begin
                         start_gdialog(1364, self_obj, 4, -1, -1);
                         gsay_start;
                         call Node037();
@@ -574,7 +574,7 @@ begin
                 end
             end
             else begin
-                float_msg(self_obj, message_str(1364, 263), 1);
+                float_msg(self_obj, mstr(263), FLOAT_MSG_BLACK);
             end
         end
     end
@@ -583,63 +583,63 @@ end
 procedure destroy_p_proc
 begin
     if (source_obj == dude_obj) then begin
-        set_global_var(5, global_var(5) + 1);
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(4) > (3 * global_var(5))) or (global_var(2) == 1)) and (global_var(1) == 0)) then begin
-            set_global_var(2, 1);
-            set_global_var(3, 0);
+        set_global_var(GVAR_GOOD_MONSTER, global_var(GVAR_GOOD_MONSTER) + 1);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(4) > (3 * global_var(GVAR_GOOD_MONSTER))) or (global_var(GVAR_CHAMPION_REPUTATION) == 1)) and (global_var(GVAR_CHILDKILLER_REPUTATION) == 0)) then begin
+            set_global_var(GVAR_CHAMPION_REPUTATION, 1);
+            set_global_var(GVAR_BERSERKER_REPUTATION, 0);
         end
-        if (metarule(51, self_obj) == 2) then begin
+        if (critter_kill_type(self_obj) == KILL_TYPE_children_kills) then begin
             if (source_obj == dude_obj) then begin
-                set_global_var(2, 0);
-                set_global_var(1, global_var(1) + 1);
-                set_global_var(0, global_var(0) + -15);
+                set_global_var(GVAR_CHAMPION_REPUTATION, 0);
+                set_global_var(GVAR_CHILDKILLER_REPUTATION, global_var(GVAR_CHILDKILLER_REPUTATION) + 1);
+                set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + -15);
                 debug_msg("Player gains " + -15 + " Karma Points.");
-                set_global_var(37, 0);
-                set_global_var(38, 0);
-                set_global_var(39, 0);
-                set_global_var(40, 0);
-                set_global_var(41, 0);
-                set_global_var(42, 0);
-                set_global_var(43, 0);
-                set_global_var(44, 0);
-                set_global_var(45, 0);
-                tmp_gen_rep := global_var(0);
-                if (has_trait(0, dude_obj, 95)) then begin
-                    tmp_gen_rep := global_var(0) * 2;
+                set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+                set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+                set_global_var(GVAR_KARMA_DEFENDER, 0);
+                set_global_var(GVAR_KARMA_WANDERER, 0);
+                set_global_var(GVAR_KARMA_BETRAYER, 0);
+                set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+                set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+                set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+                tmp_gen_rep := global_var(GVAR_PLAYER_REPUTATION);
+                if (has_trait(TRAIT_PERK, dude_obj, PERK_karma_beacon_perk)) then begin
+                    tmp_gen_rep := global_var(GVAR_PLAYER_REPUTATION) * 2;
                 end
                 if (tmp_gen_rep >= 1000) then begin
-                    set_global_var(37, 1);
+                    set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
                 end
                 else begin
                     if (tmp_gen_rep >= 750) then begin
-                        set_global_var(38, 1);
+                        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
                     end
                     else begin
                         if (tmp_gen_rep >= 500) then begin
-                            set_global_var(39, 1);
+                            set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                         end
                         else begin
                             if (tmp_gen_rep >= 250) then begin
-                                set_global_var(40, 1);
+                                set_global_var(GVAR_KARMA_DEFENDER, 1);
                             end
                             else begin
                                 if (tmp_gen_rep > -250) then begin
-                                    set_global_var(41, 1);
+                                    set_global_var(GVAR_KARMA_WANDERER, 1);
                                 end
                                 else begin
                                     if (tmp_gen_rep > -500) then begin
-                                        set_global_var(42, 1);
+                                        set_global_var(GVAR_KARMA_BETRAYER, 1);
                                     end
                                     else begin
                                         if (tmp_gen_rep > -750) then begin
-                                            set_global_var(43, 1);
+                                            set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                         end
                                         else begin
                                             if (tmp_gen_rep > -1000) then begin
-                                                set_global_var(44, 1);
+                                                set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                             end
                                             else begin
-                                                set_global_var(45, 1);
+                                                set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                             end
                                         end
                                     end
@@ -648,62 +648,62 @@ begin
                         end
                     end
                 end
-                set_global_var(47, global_var(47) + -8);
+                set_global_var(GVAR_TOWN_REP_ARROYO, global_var(GVAR_TOWN_REP_ARROYO) + -8);
                 debug_msg("Added " + -8 + " to Town Rep");
             end
         end
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(5) > (2 * global_var(4))) or (global_var(3) == 1))) then begin
-            set_global_var(3, 1);
-            set_global_var(2, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
+            set_global_var(GVAR_BERSERKER_REPUTATION, 1);
+            set_global_var(GVAR_CHAMPION_REPUTATION, 0);
         end
-        set_global_var(0, global_var(0) + -10);
+        set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + -10);
         debug_msg("Player gains " + -10 + " Karma Points.");
-        set_global_var(37, 0);
-        set_global_var(38, 0);
-        set_global_var(39, 0);
-        set_global_var(40, 0);
-        set_global_var(41, 0);
-        set_global_var(42, 0);
-        set_global_var(43, 0);
-        set_global_var(44, 0);
-        set_global_var(45, 0);
-        tmp_gen_rep := global_var(0);
-        if (has_trait(0, dude_obj, 95)) then begin
-            tmp_gen_rep := global_var(0) * 2;
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+        set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+        set_global_var(GVAR_KARMA_DEFENDER, 0);
+        set_global_var(GVAR_KARMA_WANDERER, 0);
+        set_global_var(GVAR_KARMA_BETRAYER, 0);
+        set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+        set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+        tmp_gen_rep := global_var(GVAR_PLAYER_REPUTATION);
+        if (has_trait(TRAIT_PERK, dude_obj, PERK_karma_beacon_perk)) then begin
+            tmp_gen_rep := global_var(GVAR_PLAYER_REPUTATION) * 2;
         end
         if (tmp_gen_rep >= 1000) then begin
-            set_global_var(37, 1);
+            set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
         end
         else begin
             if (tmp_gen_rep >= 750) then begin
-                set_global_var(38, 1);
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
             end
             else begin
                 if (tmp_gen_rep >= 500) then begin
-                    set_global_var(39, 1);
+                    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                 end
                 else begin
                     if (tmp_gen_rep >= 250) then begin
-                        set_global_var(40, 1);
+                        set_global_var(GVAR_KARMA_DEFENDER, 1);
                     end
                     else begin
                         if (tmp_gen_rep > -250) then begin
-                            set_global_var(41, 1);
+                            set_global_var(GVAR_KARMA_WANDERER, 1);
                         end
                         else begin
                             if (tmp_gen_rep > -500) then begin
-                                set_global_var(42, 1);
+                                set_global_var(GVAR_KARMA_BETRAYER, 1);
                             end
                             else begin
                                 if (tmp_gen_rep > -750) then begin
-                                    set_global_var(43, 1);
+                                    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                 end
                                 else begin
                                     if (tmp_gen_rep > -1000) then begin
-                                        set_global_var(44, 1);
+                                        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                     end
                                     else begin
-                                        set_global_var(45, 1);
+                                        set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                     end
                                 end
                             end
@@ -712,7 +712,7 @@ begin
                 end
             end
         end
-        set_global_var(47, global_var(47) + -5);
+        set_global_var(GVAR_TOWN_REP_ARROYO, global_var(GVAR_TOWN_REP_ARROYO) + -5);
         debug_msg("Added " + -5 + " to Town Rep");
     end
 end
@@ -721,10 +721,10 @@ procedure look_at_p_proc
 begin
     script_overrides;
     if (local_var(4) == 0) then begin
-        display_msg(message_str(1364, 260));
+        display_mstr(260);
     end
     else begin
-        display_msg(message_str(1364, 100));
+        display_mstr(100);
     end
 end
 
@@ -732,10 +732,10 @@ procedure description_p_proc
 begin
     script_overrides;
     if (local_var(4) == 0) then begin
-        display_msg(message_str(1364, 261));
+        display_mstr(261);
     end
     else begin
-        display_msg(message_str(1364, 101));
+        display_mstr(101);
     end
 end
 
@@ -743,54 +743,54 @@ procedure use_skill_on_p_proc
 begin
     if (action_being_used == 10) then begin
         if (is_success(roll_vs_skill(dude_obj, action_being_used, 0))) then begin
-            set_global_var(0, global_var(0) + -3);
+            set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + -3);
             debug_msg("Player gains " + -3 + " Karma Points.");
-            set_global_var(37, 0);
-            set_global_var(38, 0);
-            set_global_var(39, 0);
-            set_global_var(40, 0);
-            set_global_var(41, 0);
-            set_global_var(42, 0);
-            set_global_var(43, 0);
-            set_global_var(44, 0);
-            set_global_var(45, 0);
-            tmp_gen_rep := global_var(0);
-            if (has_trait(0, dude_obj, 95)) then begin
-                tmp_gen_rep := global_var(0) * 2;
+            set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+            set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+            set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+            set_global_var(GVAR_KARMA_DEFENDER, 0);
+            set_global_var(GVAR_KARMA_WANDERER, 0);
+            set_global_var(GVAR_KARMA_BETRAYER, 0);
+            set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+            set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+            set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+            tmp_gen_rep := global_var(GVAR_PLAYER_REPUTATION);
+            if (has_trait(TRAIT_PERK, dude_obj, PERK_karma_beacon_perk)) then begin
+                tmp_gen_rep := global_var(GVAR_PLAYER_REPUTATION) * 2;
             end
             if (tmp_gen_rep >= 1000) then begin
-                set_global_var(37, 1);
+                set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
             end
             else begin
                 if (tmp_gen_rep >= 750) then begin
-                    set_global_var(38, 1);
+                    set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
                 end
                 else begin
                     if (tmp_gen_rep >= 500) then begin
-                        set_global_var(39, 1);
+                        set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                     end
                     else begin
                         if (tmp_gen_rep >= 250) then begin
-                            set_global_var(40, 1);
+                            set_global_var(GVAR_KARMA_DEFENDER, 1);
                         end
                         else begin
                             if (tmp_gen_rep > -250) then begin
-                                set_global_var(41, 1);
+                                set_global_var(GVAR_KARMA_WANDERER, 1);
                             end
                             else begin
                                 if (tmp_gen_rep > -500) then begin
-                                    set_global_var(42, 1);
+                                    set_global_var(GVAR_KARMA_BETRAYER, 1);
                                 end
                                 else begin
                                     if (tmp_gen_rep > -750) then begin
-                                        set_global_var(43, 1);
+                                        set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                     end
                                     else begin
                                         if (tmp_gen_rep > -1000) then begin
-                                            set_global_var(44, 1);
+                                            set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                         end
                                         else begin
-                                            set_global_var(45, 1);
+                                            set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                         end
                                     end
                                 end
@@ -807,14 +807,14 @@ procedure damage_p_proc
 begin
     if (source_obj == dude_obj) then begin
         set_local_var(6, 1);
-        set_global_var(7, 1);
+        set_global_var(GVAR_ENEMY_ARROYO, 1);
     end
 end
 
 procedure map_enter_p_proc
 begin
-    critter_add_trait(self_obj, 1, 6, 1);
-    critter_add_trait(self_obj, 1, 5, 2);
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 1);
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_ARROYO_VILLAGER);
     if (local_var(10) == 1) then begin
         add_timer_event(self_obj, game_ticks(random(20, 40)), 20);
     end
@@ -824,12 +824,12 @@ procedure timed_event_p_proc
 begin
     if (fixed_param == 20) then begin
         if ((local_var(5) == 0) and (local_var(6) == 0)) then begin
-            float_msg(self_obj, message_str(1364, random(190, 192)), 3);
-            metarule3(100, self_obj, 20, 0);
+            float_msg(self_obj, mstr(random(190, 192)), FLOAT_MSG_GREEN);
+            rm_fixed_timer_event(self_obj, 20);
             add_timer_event(self_obj, game_ticks(random(20, 40)), 20);
         end
         else begin
-            metarule3(100, self_obj, 20, 0);
+            rm_fixed_timer_event(self_obj, 20);
         end
     end
 end
@@ -855,69 +855,69 @@ end
 procedure Node001
 begin
     set_local_var(4, 1);
-    gsay_reply(1364, 110);
-    giq_option(4, 1364, 111, Node002, 51);
-    if (get_critter_stat(dude_obj, 34) == 0) then begin
-        giq_option(4, 1364, 112, Node004, 49);
+    Reply(110);
+    giq_option(4, NAME, 111, Node002, BAD_REACTION);
+    if (dude_gender == 0) then begin
+        giq_option(4, NAME, 112, Node004, GOOD_REACTION);
     end
-    if (get_critter_stat(dude_obj, 34) == 1) then begin
-        giq_option(4, 1364, 250, Node014, 49);
+    if (dude_gender == 1) then begin
+        giq_option(4, NAME, 250, Node014, GOOD_REACTION);
     end
 end
 
 procedure Node002
 begin
-    gsay_reply(1364, 120);
-    giq_option(4, 1364, 121, Node996, 51);
-    if (global_var(644) < 20) then begin
-        giq_option(4, 1364, 900, Node003, 49);
+    Reply(120);
+    giq_option(4, NAME, 121, Node996, BAD_REACTION);
+    if (global_var(GVAR_AHELDER_Q1_AGRO) < 20) then begin
+        giq_option(4, NAME, 900, Node003, GOOD_REACTION);
     end
     else begin
-        giq_option(4, 1364, 122, Node003, 49);
+        giq_option(4, NAME, 122, Node003, GOOD_REACTION);
     end
 end
 
 procedure Node003
 begin
-    gsay_reply(1364, 130);
-    giq_option(4, 1364, 131, Node999, 50);
-    giq_option(4, 1364, 262, Node996, 50);
+    Reply(130);
+    giq_option(4, NAME, 131, Node999, NEUTRAL_REACTION);
+    giq_option(4, NAME, 262, Node996, NEUTRAL_REACTION);
 end
 
 procedure Node004
 begin
-    gsay_reply(1364, 140);
-    if (global_var(644) < 20) then begin
-        giq_option(4, 1364, 901, Node999, 50);
-        giq_option(4, 1364, 902, Node005, 49);
+    Reply(140);
+    if (global_var(GVAR_AHELDER_Q1_AGRO) < 20) then begin
+        giq_option(4, NAME, 901, Node999, NEUTRAL_REACTION);
+        giq_option(4, NAME, 902, Node005, GOOD_REACTION);
     end
     else begin
-        giq_option(4, 1364, 141, Node999, 50);
-        giq_option(4, 1364, 142, Node005, 49);
+        giq_option(4, NAME, 141, Node999, NEUTRAL_REACTION);
+        giq_option(4, NAME, 142, Node005, GOOD_REACTION);
     end
 end
 
 procedure Node005
 begin
-    gsay_reply(1364, 150);
-    giq_option(4, 1364, 151, Node999, 50);
-    giq_option(4, 1364, 152, Node006, 50);
+    Reply(150);
+    giq_option(4, NAME, 151, Node999, NEUTRAL_REACTION);
+    giq_option(4, NAME, 152, Node006, NEUTRAL_REACTION);
 end
 
 procedure Node006
 begin
     gfade_out(20);
-    if (global_var(590) == 1) then begin
+    if (global_var(GVAR_DUDE_VIRGIN) == 1) then begin
         set_local_var(11, 1);
     end
-    if (has_trait(0, dude_obj, 94) or (global_var(232) > 0) or (global_var(588) > 0)) then begin
+    if (has_trait(TRAIT_PERK, dude_obj, PERK_kama_sutra_perk) or (global_var(GVAR_NEW_RENO_PORN_STAR) > 0) or (global_var(GVAR_SEXPERT) > 0)) then begin
         set_local_var(11, 2);
     end
-    if (message_str(1364, 150) != 0) then begin
+    if (mstr(150) != 0) then begin
         if (obj_is_carrying_obj_pid(dude_obj, 314) > 0) then begin
             restock_obj := obj_carrying_pid_obj(dude_obj, 314);
-            if ((critter_inven_obj(dude_obj, 2) == restock_obj) or (critter_inven_obj(dude_obj, 1) == restock_obj)) then begin
-                metarule(43, dude_obj);
+            if ((critter_inven_obj(dude_obj, INVEN_TYPE_LEFT_HAND) == restock_obj) or (critter_inven_obj(dude_obj, INVEN_TYPE_RIGHT_HAND) == restock_obj)) then begin
+                inven_unwield(dude_obj);
             end
             rm_obj_from_inven(dude_obj, restock_obj);
             destroy_object(restock_obj);
@@ -926,8 +926,8 @@ begin
         else begin
             if (obj_is_carrying_obj_pid(dude_obj, 315) > 0) then begin
                 restock_obj := obj_carrying_pid_obj(dude_obj, 315);
-                if ((critter_inven_obj(dude_obj, 2) == restock_obj) or (critter_inven_obj(dude_obj, 1) == restock_obj)) then begin
-                    metarule(43, dude_obj);
+                if ((critter_inven_obj(dude_obj, INVEN_TYPE_LEFT_HAND) == restock_obj) or (critter_inven_obj(dude_obj, INVEN_TYPE_RIGHT_HAND) == restock_obj)) then begin
+                    inven_unwield(dude_obj);
                 end
                 rm_obj_from_inven(dude_obj, restock_obj);
                 destroy_object(restock_obj);
@@ -936,8 +936,8 @@ begin
             else begin
                 if (obj_is_carrying_obj_pid(dude_obj, 316) > 0) then begin
                     restock_obj := obj_carrying_pid_obj(dude_obj, 316);
-                    if ((critter_inven_obj(dude_obj, 2) == restock_obj) or (critter_inven_obj(dude_obj, 1) == restock_obj)) then begin
-                        metarule(43, dude_obj);
+                    if ((critter_inven_obj(dude_obj, INVEN_TYPE_LEFT_HAND) == restock_obj) or (critter_inven_obj(dude_obj, INVEN_TYPE_RIGHT_HAND) == restock_obj)) then begin
+                        inven_unwield(dude_obj);
                     end
                     rm_obj_from_inven(dude_obj, restock_obj);
                     destroy_object(restock_obj);
@@ -946,18 +946,18 @@ begin
             end
         end
     end
-    set_global_var(358, global_var(358) + 1);
-    set_global_var(194, global_var(194) + (((((get_critter_stat(dude_obj, 3) * 50) + (get_critter_stat(dude_obj, 2) * 25) + (get_critter_stat(dude_obj, 5) * 13) + (get_critter_stat(dude_obj, 0) * 12)) / 100) + (has_trait(0, dude_obj, 94) * 2) + has_trait(2, dude_obj, 13) + (((global_var(194) >= 100) or (global_var(232) > 0)) * 2 * (dude_obj == dude_obj))) * 1));
-    if (global_var(358) >= 10) then begin
-        set_global_var(589, 1);
+    set_global_var(GVAR_SEX_COUNTER, global_var(GVAR_SEX_COUNTER) + 1);
+    set_global_var(GVAR_PLAYER_SEX_LEVEL, global_var(194) + (((((dude_charisma * 50) + (dude_endurance * 25) + (get_critter_stat(dude_obj, STAT_ag) * 13) + (dude_strength * 12)) / 100) + (has_trait(TRAIT_PERK, dude_obj, PERK_kama_sutra_perk) * 2) + has_trait(2, dude_obj, 13) + (((global_var(GVAR_PLAYER_SEX_LEVEL) >= 100) or (global_var(GVAR_NEW_RENO_PORN_STAR) > 0)) * 2 * (dude_obj == dude_obj))) * 1));
+    if (global_var(GVAR_SEX_COUNTER) >= 10) then begin
+        set_global_var(GVAR_GIGALO, 1);
     end
-    if (((((get_critter_stat(dude_obj, 3) * 50) + (get_critter_stat(dude_obj, 2) * 25) + (get_critter_stat(dude_obj, 5) * 13) + (get_critter_stat(dude_obj, 0) * 12)) / 100) + (has_trait(0, dude_obj, 94) * 2) + has_trait(2, dude_obj, 13) + (((global_var(194) >= 100) or (global_var(232) > 0)) * 2 * (dude_obj == dude_obj))) >= 9) then begin
-        set_global_var(588, 1);
+    if (((((dude_charisma * 50) + (dude_endurance * 25) + (get_critter_stat(dude_obj, STAT_ag) * 13) + (dude_strength * 12)) / 100) + (has_trait(TRAIT_PERK, dude_obj, PERK_kama_sutra_perk) * 2) + has_trait(2, dude_obj, 13) + (((global_var(GVAR_PLAYER_SEX_LEVEL) >= 100) or (global_var(GVAR_NEW_RENO_PORN_STAR) > 0)) * 2 * (dude_obj == dude_obj))) >= 9) then begin
+        set_global_var(GVAR_SEXPERT, 1);
     end
-    if (global_var(590) == 1) then begin
-        set_global_var(590, 0);
+    if (global_var(GVAR_DUDE_VIRGIN) == 1) then begin
+        set_global_var(GVAR_DUDE_VIRGIN, 0);
     end
-    game_time_advance(get_critter_stat(dude_obj, 2) * random(3, 5) * 600);
+    game_time_advance(dude_endurance * random(3, 5) * 600);
     set_local_var(10, 1);
     add_timer_event(self_obj, game_ticks(random(30, 50)), 20);
     call Node010();
@@ -966,398 +966,398 @@ end
 procedure Node007
 begin
     set_local_var(4, 1);
-    gsay_reply(1364, 500);
-    giq_option(-3, 1364, 501, Node008, 50);
-    giq_option(-3, 1364, 502, Node999, 50);
+    Reply(500);
+    giq_option(-3, NAME, 501, Node008, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 502, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node008
 begin
-    add_obj_to_inven(dude_obj, create_object_sid(378, 0, 0, -1));
-    if (global_var(644) < 20) then begin
-        gsay_reply(1364, 911);
+    add_obj_to_inven(dude_obj, create_object(378, 0, 0));
+    if (global_var(GVAR_AHELDER_Q1_AGRO) < 20) then begin
+        Reply(911);
     end
     else begin
-        gsay_reply(1364, 503);
+        Reply(503);
     end
-    if (global_var(644) < 20) then begin
-        giq_option(-3, 1364, 912, Node999, 50);
+    if (global_var(GVAR_AHELDER_Q1_AGRO) < 20) then begin
+        giq_option(-3, NAME, 912, Node999, NEUTRAL_REACTION);
     end
     else begin
-        giq_option(-3, 1364, 504, Node999, 50);
+        giq_option(-3, NAME, 504, Node999, NEUTRAL_REACTION);
     end
-    giq_option(-3, 1364, 505, Node999, 50);
+    giq_option(-3, NAME, 505, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node009
 begin
-    float_msg(self_obj, message_str(9, random(176, 182)), 8);
+    float_msg(self_obj, message_str(SCRIPT_ACVILLGR, random(176, 182)), FLOAT_MSG_YELLOW);
 end
 
 procedure Node010
 begin
     gfade_in(20);
     if (local_var(11) == 0) then begin
-        gsay_reply(1364, 170);
-        if (global_var(644) < 20) then begin
-            giq_option(4, 1364, 905, Node999, 50);
+        Reply(170);
+        if (global_var(GVAR_AHELDER_Q1_AGRO) < 20) then begin
+            giq_option(4, NAME, 905, Node999, NEUTRAL_REACTION);
         end
         else begin
-            giq_option(4, 1364, 171, Node999, 50);
+            giq_option(4, NAME, 171, Node999, NEUTRAL_REACTION);
         end
-        giq_option(4, 1364, 172, Node999, 50);
+        giq_option(4, NAME, 172, Node999, NEUTRAL_REACTION);
     end
     if (local_var(11) == 1) then begin
-        if (global_var(644) < 20) then begin
-            gsay_reply(1364, 903);
+        if (global_var(GVAR_AHELDER_Q1_AGRO) < 20) then begin
+            Reply(903);
         end
         else begin
-            gsay_reply(1364, 160);
+            Reply(160);
         end
-        if (global_var(644) < 20) then begin
-            giq_option(4, 1364, 904, Node999, 50);
+        if (global_var(GVAR_AHELDER_Q1_AGRO) < 20) then begin
+            giq_option(4, NAME, 904, Node999, NEUTRAL_REACTION);
         end
         else begin
-            giq_option(4, 1364, 161, Node999, 50);
+            giq_option(4, NAME, 161, Node999, NEUTRAL_REACTION);
         end
-        giq_option(4, 1364, 162, Node999, 50);
+        giq_option(4, NAME, 162, Node999, NEUTRAL_REACTION);
     end
     if (local_var(11) == 2) then begin
-        gsay_reply(1364, 180);
-        giq_option(4, 1364, 181, Node999, 50);
-        if (global_var(644) < 20) then begin
-            giq_option(4, 1364, 906, Node999, 50);
+        Reply(180);
+        giq_option(4, NAME, 181, Node999, NEUTRAL_REACTION);
+        if (global_var(GVAR_AHELDER_Q1_AGRO) < 20) then begin
+            giq_option(4, NAME, 906, Node999, NEUTRAL_REACTION);
         end
         else begin
-            giq_option(4, 1364, 182, Node999, 50);
+            giq_option(4, NAME, 182, Node999, NEUTRAL_REACTION);
         end
     end
 end
 
 procedure Node011
 begin
-    if (global_var(644) < 20) then begin
-        gsay_reply(1364, 907);
+    if (global_var(GVAR_AHELDER_Q1_AGRO) < 20) then begin
+        Reply(907);
     end
     else begin
-        gsay_reply(1364, 200);
+        Reply(200);
     end
-    if (get_critter_stat(dude_obj, 34) == 0) then begin
+    if (dude_gender == 0) then begin
         if (local_var(10) == 1) then begin
-            giq_option(4, 1364, 201, Node012, 50);
+            giq_option(4, NAME, 201, Node012, NEUTRAL_REACTION);
         end
-        giq_option(4, 1364, 202, Node999, 50);
+        giq_option(4, NAME, 202, Node999, NEUTRAL_REACTION);
     end
-    if (get_critter_stat(dude_obj, 34) == 1) then begin
-        giq_option(4, 1364, 202, Node999, 50);
-        giq_option(4, 1364, 300, Node017, 50);
+    if (dude_gender == 1) then begin
+        giq_option(4, NAME, 202, Node999, NEUTRAL_REACTION);
+        giq_option(4, NAME, 300, Node017, NEUTRAL_REACTION);
     end
 end
 
 procedure Node012
 begin
-    if (global_var(644) < 20) then begin
-        gsay_reply(1364, 908);
+    if (global_var(GVAR_AHELDER_Q1_AGRO) < 20) then begin
+        Reply(908);
     end
     else begin
-        gsay_reply(1364, 210);
+        Reply(210);
     end
-    giq_option(4, 1364, 211, Node999, 50);
-    if (global_var(644) < 20) then begin
-        giq_option(4, 1364, 909, Node999, 51);
+    giq_option(4, NAME, 211, Node999, NEUTRAL_REACTION);
+    if (global_var(GVAR_AHELDER_Q1_AGRO) < 20) then begin
+        giq_option(4, NAME, 909, Node999, BAD_REACTION);
     end
     else begin
-        giq_option(4, 1364, 212, Node999, 51);
+        giq_option(4, NAME, 212, Node999, BAD_REACTION);
     end
 end
 
 procedure Node014
 begin
-    gsay_reply(1364, 220);
-    giq_option(4, 1364, 142, Node015, 49);
-    giq_option(4, 1364, 131, Node999, 50);
+    Reply(220);
+    giq_option(4, NAME, 142, Node015, GOOD_REACTION);
+    giq_option(4, NAME, 131, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node015
 begin
-    gsay_reply(1364, 230);
-    giq_option(4, 1364, 231, Node016, 50);
-    giq_option(4, 1364, 232, Node999, 50);
+    Reply(230);
+    giq_option(4, NAME, 231, Node016, NEUTRAL_REACTION);
+    giq_option(4, NAME, 232, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node016
 begin
-    gsay_reply(1364, 240);
-    giq_option(4, 1364, 241, Node999, 49);
-    giq_option(4, 1364, 242, Node996, 51);
+    Reply(240);
+    giq_option(4, NAME, 241, Node999, GOOD_REACTION);
+    giq_option(4, NAME, 242, Node996, BAD_REACTION);
 end
 
 procedure Node017
 begin
-    gsay_reply(1364, message_str(1364, 304) + obj_name(dude_obj) + message_str(1364, 305));
-    giq_option(4, 1364, 306, Node018, 50);
-    giq_option(4, 1364, 307, Node024, 50);
-    giq_option(4, 1364, 308, Node040, 50);
+    Reply(mstr(304) + dude_name + mstr(305));
+    giq_option(4, NAME, 306, Node018, NEUTRAL_REACTION);
+    giq_option(4, NAME, 307, Node024, NEUTRAL_REACTION);
+    giq_option(4, NAME, 308, Node040, NEUTRAL_REACTION);
 end
 
 procedure Node018
 begin
-    if (message_str(1364, 100) != 0) then begin
+    if (mstr(100) != 0) then begin
         if ((game_time / 864000) > 180) then begin
-            gsay_reply(1364, 360);
+            Reply(360);
         end
         else begin
             if ((game_time / 864000) > 90) then begin
-                gsay_reply(1364, 340);
-                giq_option(4, 1364, 351, Node022, 50);
-                giq_option(4, 1364, 353, Node023, 50);
+                Reply(340);
+                giq_option(4, NAME, 351, Node022, NEUTRAL_REACTION);
+                giq_option(4, NAME, 353, Node023, NEUTRAL_REACTION);
             end
             else begin
-                gsay_reply(1364, 310);
-                giq_option(4, 1364, 312, Node019, 50);
-                giq_option(4, 1364, 313, Node021, 50);
-                giq_option(4, 1364, 311, Node020, 50);
+                Reply(310);
+                giq_option(4, NAME, 312, Node019, NEUTRAL_REACTION);
+                giq_option(4, NAME, 313, Node021, NEUTRAL_REACTION);
+                giq_option(4, NAME, 311, Node020, NEUTRAL_REACTION);
             end
         end
     end
-    giq_option(4, 1364, 314, Node999, 50);
-    giq_option(4, 1364, 335, Node017, 50);
+    giq_option(4, NAME, 314, Node999, NEUTRAL_REACTION);
+    giq_option(4, NAME, 335, Node017, NEUTRAL_REACTION);
 end
 
 procedure Node019
 begin
-    gsay_reply(1364, 325);
-    giq_option(4, 1364, 705, Node019a, 50);
+    Reply(325);
+    giq_option(4, NAME, 705, Node019a, NEUTRAL_REACTION);
 end
 
 procedure Node019a
 begin
-    gsay_reply(1364, 706);
-    giq_option(4, 1364, 326, Node020, 50);
-    giq_option(4, 1364, 313, Node021, 50);
-    giq_option(4, 1364, 335, Node017, 50);
-    if (global_var(644) < 20) then begin
-        giq_option(4, 1364, 910, Node999, 50);
+    Reply(706);
+    giq_option(4, NAME, 326, Node020, NEUTRAL_REACTION);
+    giq_option(4, NAME, 313, Node021, NEUTRAL_REACTION);
+    giq_option(4, NAME, 335, Node017, NEUTRAL_REACTION);
+    if (global_var(GVAR_AHELDER_Q1_AGRO) < 20) then begin
+        giq_option(4, NAME, 910, Node999, NEUTRAL_REACTION);
     end
     else begin
-        giq_option(4, 1364, 327, Node999, 50);
+        giq_option(4, NAME, 327, Node999, NEUTRAL_REACTION);
     end
 end
 
 procedure Node020
 begin
-    gsay_reply(1364, 320);
-    giq_option(4, 1364, 700, Node020a, 50);
+    Reply(320);
+    giq_option(4, NAME, 700, Node020a, NEUTRAL_REACTION);
 end
 
 procedure Node020a
 begin
-    gsay_reply(1364, 701);
-    giq_option(4, 1364, 321, Node019, 50);
-    giq_option(4, 1364, 313, Node021, 50);
-    giq_option(4, 1364, 322, Node999, 50);
-    giq_option(4, 1364, 335, Node017, 50);
+    Reply(701);
+    giq_option(4, NAME, 321, Node019, NEUTRAL_REACTION);
+    giq_option(4, NAME, 313, Node021, NEUTRAL_REACTION);
+    giq_option(4, NAME, 322, Node999, NEUTRAL_REACTION);
+    giq_option(4, NAME, 335, Node017, NEUTRAL_REACTION);
 end
 
 procedure Node021
 begin
-    gsay_reply(1364, 330);
-    giq_option(4, 1364, 331, Node020, 50);
-    giq_option(4, 1364, 332, Node019, 50);
-    giq_option(4, 1364, 335, Node017, 50);
-    giq_option(4, 1364, 333, Node999, 50);
+    Reply(330);
+    giq_option(4, NAME, 331, Node020, NEUTRAL_REACTION);
+    giq_option(4, NAME, 332, Node019, NEUTRAL_REACTION);
+    giq_option(4, NAME, 335, Node017, NEUTRAL_REACTION);
+    giq_option(4, NAME, 333, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node022
 begin
-    gsay_reply(1364, 352);
-    giq_option(4, 1364, 353, Node023, 50);
-    giq_option(4, 1364, 350, Node017, 50);
-    giq_option(4, 1364, 333, Node999, 50);
+    Reply(352);
+    giq_option(4, NAME, 353, Node023, NEUTRAL_REACTION);
+    giq_option(4, NAME, 350, Node017, NEUTRAL_REACTION);
+    giq_option(4, NAME, 333, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node023
 begin
-    gsay_reply(1364, 320);
-    giq_option(4, 1364, 701, Node023a, 50);
+    Reply(320);
+    giq_option(4, NAME, 701, Node023a, NEUTRAL_REACTION);
 end
 
 procedure Node023a
 begin
-    gsay_reply(1364, 701);
-    giq_option(4, 1364, 321, Node022, 50);
-    giq_option(4, 1364, 335, Node017, 50);
-    giq_option(4, 1364, 314, Node999, 50);
+    Reply(701);
+    giq_option(4, NAME, 321, Node022, NEUTRAL_REACTION);
+    giq_option(4, NAME, 335, Node017, NEUTRAL_REACTION);
+    giq_option(4, NAME, 314, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node024
 begin
-    gsay_reply(1364, 369);
-    giq_option(4, 1364, 366, Node017, 50);
+    Reply(369);
+    giq_option(4, NAME, 366, Node017, NEUTRAL_REACTION);
     if (local_var(12) == 0) then begin
-        giq_option(4, 1364, 361, Node025, 50);
+        giq_option(4, NAME, 361, Node025, NEUTRAL_REACTION);
     end
     if (local_var(13) == 0) then begin
-        giq_option(4, 1364, 362, Node026, 50);
+        giq_option(4, NAME, 362, Node026, NEUTRAL_REACTION);
     end
     if (local_var(14) == 0) then begin
-        giq_option(4, 1364, 363, Node027, 50);
+        giq_option(4, NAME, 363, Node027, NEUTRAL_REACTION);
     end
     if (local_var(12) == 1) then begin
-        giq_option(4, 1364, 364, Node028, 50);
+        giq_option(4, NAME, 364, Node028, NEUTRAL_REACTION);
     end
     if (local_var(13) == 1) then begin
-        giq_option(4, 1364, 365, Node029, 50);
+        giq_option(4, NAME, 365, Node029, NEUTRAL_REACTION);
     end
     if (local_var(14) == 1) then begin
-        giq_option(4, 1364, 368, Node030, 50);
+        giq_option(4, NAME, 368, Node030, NEUTRAL_REACTION);
     end
     if ((local_var(12) + local_var(13) + local_var(14)) > 4) then begin
-        giq_option(4, 1364, 420, Node031, 50);
+        giq_option(4, NAME, 420, Node031, NEUTRAL_REACTION);
     end
     if ((local_var(12) + local_var(13) + local_var(14)) > 5) then begin
-        giq_option(4, 1364, 430, Node032, 50);
+        giq_option(4, NAME, 430, Node032, NEUTRAL_REACTION);
     end
-    if ((global_var(399) bwand 1) and (local_var(8) == 0)) then begin
-        giq_option(4, 1364, 367, Node036, 50);
+    if ((global_var(GVAR_RND_KAGA_STATE) bwand 1) and (local_var(8) == 0)) then begin
+        giq_option(4, NAME, 367, Node036, NEUTRAL_REACTION);
     end
 end
 
 procedure Node025
 begin
     set_local_var(12, 1);
-    gsay_reply(1364, 370);
-    giq_option(4, 1364, 371, Node024, 50);
-    giq_option(4, 1364, 372, Node017, 50);
-    giq_option(4, 1364, 373, Node999, 50);
+    Reply(370);
+    giq_option(4, NAME, 371, Node024, NEUTRAL_REACTION);
+    giq_option(4, NAME, 372, Node017, NEUTRAL_REACTION);
+    giq_option(4, NAME, 373, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node026
 begin
     set_local_var(13, 1);
-    gsay_reply(1364, 375);
-    giq_option(4, 1364, 376, Node024, 50);
-    giq_option(4, 1364, 372, Node017, 50);
-    giq_option(4, 1364, 373, Node999, 50);
+    Reply(375);
+    giq_option(4, NAME, 376, Node024, NEUTRAL_REACTION);
+    giq_option(4, NAME, 372, Node017, NEUTRAL_REACTION);
+    giq_option(4, NAME, 373, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node027
 begin
     set_local_var(14, 1);
-    gsay_reply(1364, 380);
-    giq_option(4, 1364, 710, Node027a, 50);
+    Reply(380);
+    giq_option(4, NAME, 710, Node027a, NEUTRAL_REACTION);
 end
 
 procedure Node027a
 begin
-    gsay_reply(1364, 711);
-    giq_option(4, 1364, 381, Node024, 50);
-    giq_option(4, 1364, 372, Node017, 50);
-    giq_option(4, 1364, 373, Node999, 50);
+    Reply(711);
+    giq_option(4, NAME, 381, Node024, NEUTRAL_REACTION);
+    giq_option(4, NAME, 372, Node017, NEUTRAL_REACTION);
+    giq_option(4, NAME, 373, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node028
 begin
     set_local_var(12, 2);
-    gsay_reply(1364, 390);
-    giq_option(4, 1364, 391, Node024, 50);
-    giq_option(4, 1364, 372, Node017, 50);
-    giq_option(4, 1364, 373, Node999, 50);
+    Reply(390);
+    giq_option(4, NAME, 391, Node024, NEUTRAL_REACTION);
+    giq_option(4, NAME, 372, Node017, NEUTRAL_REACTION);
+    giq_option(4, NAME, 373, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node029
 begin
     set_local_var(13, 2);
-    gsay_reply(1364, 400);
-    giq_option(4, 1364, 715, Node029a, 50);
+    Reply(400);
+    giq_option(4, NAME, 715, Node029a, NEUTRAL_REACTION);
 end
 
 procedure Node029a
 begin
-    gsay_reply(1364, 716);
-    giq_option(4, 1364, 401, Node024, 50);
-    giq_option(4, 1364, 372, Node017, 50);
-    giq_option(4, 1364, 373, Node999, 50);
+    Reply(716);
+    giq_option(4, NAME, 401, Node024, NEUTRAL_REACTION);
+    giq_option(4, NAME, 372, Node017, NEUTRAL_REACTION);
+    giq_option(4, NAME, 373, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node030
 begin
     set_local_var(14, 2);
-    gsay_reply(1364, 410);
-    giq_option(4, 1364, 720, Node030a, 50);
+    Reply(410);
+    giq_option(4, NAME, 720, Node030a, NEUTRAL_REACTION);
 end
 
 procedure Node030a
 begin
-    gsay_reply(1364, 721);
-    giq_option(4, 1364, 411, Node024, 50);
-    giq_option(4, 1364, 372, Node017, 50);
-    giq_option(4, 1364, 373, Node999, 50);
+    Reply(721);
+    giq_option(4, NAME, 411, Node024, NEUTRAL_REACTION);
+    giq_option(4, NAME, 372, Node017, NEUTRAL_REACTION);
+    giq_option(4, NAME, 373, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node031
 begin
-    gsay_reply(1364, 421);
-    giq_option(4, 1364, 422, Node024, 50);
-    giq_option(4, 1364, 423, Node024, 50);
-    giq_option(4, 1364, 372, Node017, 50);
-    giq_option(4, 1364, 373, Node999, 50);
+    Reply(421);
+    giq_option(4, NAME, 422, Node024, NEUTRAL_REACTION);
+    giq_option(4, NAME, 423, Node024, NEUTRAL_REACTION);
+    giq_option(4, NAME, 372, Node017, NEUTRAL_REACTION);
+    giq_option(4, NAME, 373, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node032
 begin
-    if (message_str(1364, 100) != 0) then begin
-        if (has_trait(2, dude_obj, 13)) then begin
+    if (mstr(100) != 0) then begin
+        if (has_trait(TRAIT_TRAIT, dude_obj, TRAIT_sex_appeal)) then begin
             if ((game_time / 864000) > 180) then begin
-                gsay_reply(1364, message_str(1364, 431) + get_critter_stat(dude_obj, 33) + message_str(1364, 434) + message_str(1364, 437) + message_str(1364, 440));
+                Reply(mstr(431) + dude_age + mstr(434) + mstr(437) + mstr(440));
             end
             else begin
                 if ((game_time / 864000) > 90) then begin
-                    gsay_reply(1364, message_str(1364, 431) + get_critter_stat(dude_obj, 33) + message_str(1364, 434) + message_str(1364, 437) + message_str(1364, 439));
+                    Reply(mstr(431) + dude_age + mstr(434) + mstr(437) + mstr(439));
                 end
                 else begin
-                    gsay_reply(1364, message_str(1364, 431) + get_critter_stat(dude_obj, 33) + message_str(1364, 434) + message_str(1364, 437) + message_str(1364, 438));
+                    Reply(mstr(431) + dude_age + mstr(434) + mstr(437) + mstr(438));
                 end
             end
         end
         else begin
-            if (has_trait(2, dude_obj, 10) or (get_critter_stat(dude_obj, 3) > 7)) then begin
+            if (has_trait(TRAIT_TRAIT, dude_obj, TRAIT_good_natured) or (dude_charisma > 7)) then begin
                 if ((game_time / 864000) > 180) then begin
-                    gsay_reply(1364, message_str(1364, 431) + get_critter_stat(dude_obj, 33) + message_str(1364, 433) + message_str(1364, 437) + message_str(1364, 440));
+                    Reply(mstr(431) + dude_age + mstr(433) + mstr(437) + mstr(440));
                 end
                 else begin
                     if ((game_time / 864000) > 90) then begin
-                        gsay_reply(1364, message_str(1364, 431) + get_critter_stat(dude_obj, 33) + message_str(1364, 433) + message_str(1364, 437) + message_str(1364, 439));
+                        Reply(mstr(431) + dude_age + mstr(433) + mstr(437) + mstr(439));
                     end
                     else begin
-                        gsay_reply(1364, message_str(1364, 431) + get_critter_stat(dude_obj, 33) + message_str(1364, 433) + message_str(1364, 437) + message_str(1364, 438));
+                        Reply(mstr(431) + dude_age + mstr(433) + mstr(437) + mstr(438));
                     end
                 end
             end
             else begin
-                if (get_critter_stat(dude_obj, 2) > 7) then begin
+                if (dude_endurance > 7) then begin
                     if ((game_time / 864000) > 180) then begin
-                        gsay_reply(1364, message_str(1364, 431) + get_critter_stat(dude_obj, 33) + message_str(1364, 436) + message_str(1364, 437) + message_str(1364, 440));
+                        Reply(mstr(431) + dude_age + mstr(436) + mstr(437) + mstr(440));
                     end
                     else begin
                         if ((game_time / 864000) > 90) then begin
-                            gsay_reply(1364, message_str(1364, 431) + get_critter_stat(dude_obj, 33) + message_str(1364, 436) + message_str(1364, 437) + message_str(1364, 439));
+                            Reply(mstr(431) + dude_age + mstr(436) + mstr(437) + mstr(439));
                         end
                         else begin
-                            gsay_reply(1364, message_str(1364, 431) + get_critter_stat(dude_obj, 33) + message_str(1364, 436) + message_str(1364, 437) + message_str(1364, 438));
+                            Reply(mstr(431) + dude_age + mstr(436) + mstr(437) + mstr(438));
                         end
                     end
                 end
                 else begin
                     if ((game_time / 864000) > 180) then begin
-                        gsay_reply(1364, message_str(1364, 431) + get_critter_stat(dude_obj, 33) + message_str(1364, 435) + message_str(1364, 437) + message_str(1364, 440));
+                        Reply(mstr(431) + dude_age + mstr(435) + mstr(437) + mstr(440));
                     end
                     else begin
                         if ((game_time / 864000) > 90) then begin
-                            gsay_reply(1364, message_str(1364, 431) + get_critter_stat(dude_obj, 33) + message_str(1364, 435) + message_str(1364, 437) + message_str(1364, 439));
+                            Reply(mstr(431) + dude_age + mstr(435) + mstr(437) + mstr(439));
                         end
                         else begin
-                            gsay_reply(1364, message_str(1364, 431) + get_critter_stat(dude_obj, 33) + message_str(1364, 435) + message_str(1364, 437) + message_str(1364, 438));
+                            Reply(mstr(431) + dude_age + mstr(435) + mstr(437) + mstr(438));
                         end
                     end
                 end
@@ -1366,141 +1366,141 @@ begin
     end
     if ((party_member_obj(16777380) != 0) or (party_member_obj(16777379) != 0)) then begin
         if (party_member_obj(16777380) != 0) then begin
-            giq_option(4, 1364, 441, Node033, 51);
+            giq_option(4, NAME, 441, Node033, BAD_REACTION);
         end
         if ((party_member_obj(16777379) != 0) and (local_var(17) == 0)) then begin
-            giq_option(4, 1364, 442, Node034, 51);
+            giq_option(4, NAME, 442, Node034, BAD_REACTION);
         end
         if ((party_member_obj(16777379) != 0) and (local_var(17) == 1)) then begin
-            giq_option(4, 1364, 456, Node034b, 50);
+            giq_option(4, NAME, 456, Node034b, NEUTRAL_REACTION);
         end
     end
-    giq_option(4, 1364, 443, Node017, 50);
-    if (message_str(1364, 101) != 0) then begin
-        if (get_critter_stat(dude_obj, 2) > 7) then begin
-            giq_option(4, 1364, 444, Node999, 50);
+    giq_option(4, NAME, 443, Node017, NEUTRAL_REACTION);
+    if (mstr(101) != 0) then begin
+        if (dude_endurance > 7) then begin
+            giq_option(4, NAME, 444, Node999, NEUTRAL_REACTION);
         end
         else begin
-            giq_option(4, 1364, 314, Node999, 50);
+            giq_option(4, NAME, 314, Node999, NEUTRAL_REACTION);
         end
     end
 end
 
 procedure Node033
 begin
-    gsay_reply(1364, 445);
-    giq_option(4, 1364, 446, Node999, 50);
-    giq_option(4, 1364, 447, Node999, 50);
+    Reply(445);
+    giq_option(4, NAME, 446, Node999, NEUTRAL_REACTION);
+    giq_option(4, NAME, 447, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node034
 begin
     set_local_var(17, 1);
-    gsay_reply(1364, 450);
-    giq_option(4, 1364, 451, Node017, 50);
-    giq_option(4, 1364, 447, Node035, 50);
+    Reply(450);
+    giq_option(4, NAME, 451, Node017, NEUTRAL_REACTION);
+    giq_option(4, NAME, 447, Node035, NEUTRAL_REACTION);
 end
 
 procedure Node034b
 begin
-    gsay_reply(1364, 457);
-    giq_option(4, 1364, 454, Node999, 50);
+    Reply(457);
+    giq_option(4, NAME, 454, Node999, NEUTRAL_REACTION);
     if (party_member_obj(16777305) != 0) then begin
-        giq_option(4, 1364, 455, Node017, 50);
+        giq_option(4, NAME, 455, Node017, NEUTRAL_REACTION);
     end
 end
 
 procedure Node035
 begin
-    gsay_reply(1364, 453);
-    giq_option(4, 1364, 454, Node999, 50);
+    Reply(453);
+    giq_option(4, NAME, 454, Node999, NEUTRAL_REACTION);
     if (party_member_obj(16777305) != 0) then begin
-        giq_option(4, 1364, 455, Node017, 50);
+        giq_option(4, NAME, 455, Node017, NEUTRAL_REACTION);
     end
 end
 
 procedure Node036
 begin
     set_local_var(8, 1);
-    gsay_reply(1364, 460);
-    giq_option(4, 1364, 725, Node036a, 50);
+    Reply(460);
+    giq_option(4, NAME, 725, Node036a, NEUTRAL_REACTION);
 end
 
 procedure Node036a
 begin
-    gsay_reply(1364, 726);
-    giq_option(4, 1364, 461, Node017, 50);
-    giq_option(4, 1364, 462, Node999, 50);
+    Reply(726);
+    giq_option(4, NAME, 461, Node017, NEUTRAL_REACTION);
+    giq_option(4, NAME, 462, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node037
 begin
-    gsay_reply(1364, 506);
-    giq_option(-3, 1364, 507, Node038, 50);
-    giq_option(-3, 1364, 508, Node039, 50);
+    Reply(506);
+    giq_option(-3, NAME, 507, Node038, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 508, Node039, NEUTRAL_REACTION);
 end
 
 procedure Node038
 begin
-    if (global_var(644) < 20) then begin
-        gsay_reply(1364, 913);
+    if (global_var(GVAR_AHELDER_Q1_AGRO) < 20) then begin
+        Reply(913);
     end
     else begin
-        gsay_reply(1364, 509);
+        Reply(509);
     end
-    giq_option(-3, 1364, 510, Node999, 50);
+    giq_option(-3, NAME, 510, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node039
 begin
-    if (global_var(644) < 20) then begin
-        gsay_reply(1364, 914);
+    if (global_var(GVAR_AHELDER_Q1_AGRO) < 20) then begin
+        Reply(914);
     end
     else begin
-        gsay_reply(1364, 511);
+        Reply(511);
     end
-    if (global_var(644) < 20) then begin
-        giq_option(-3, 1364, 915, Node999, 50);
+    if (global_var(GVAR_AHELDER_Q1_AGRO) < 20) then begin
+        giq_option(-3, NAME, 915, Node999, NEUTRAL_REACTION);
     end
     else begin
-        giq_option(-3, 1364, 512, Node999, 50);
+        giq_option(-3, NAME, 512, Node999, NEUTRAL_REACTION);
     end
-    giq_option(-3, 1364, 510, Node999, 50);
+    giq_option(-3, NAME, 510, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node040
 begin
     variable LVar0 := 0;
-    gsay_reply(1364, message_str(1364, 550) + obj_name(dude_obj) + message_str(1364, 551));
-    if ((local_var(15) == 0) and (((get_critter_stat(dude_obj, 1) + get_critter_stat(dude_obj, 4)) / 2) > 6)) then begin
+    Reply(mstr(550) + dude_name + mstr(551));
+    if ((local_var(15) == 0) and (((dude_perception + dude_iq) / 2) > 6)) then begin
         LVar0 := 1;
-        giq_option(4, 1364, 555, Node041, 50);
+        giq_option(4, NAME, 555, Node041, NEUTRAL_REACTION);
     end
-    if ((global_var(590) == 1) and (local_var(16) == 0)) then begin
+    if ((global_var(GVAR_DUDE_VIRGIN) == 1) and (local_var(16) == 0)) then begin
         if (LVar0 == 0) then begin
             LVar0 := 1;
-            giq_option(4, 1364, 575, Node042, 50);
+            giq_option(4, NAME, 575, Node042, NEUTRAL_REACTION);
         end
     end
     if ((local_var(17) == 0) and (party_member_obj(16777379) != 0) and (local_var(18) == 0)) then begin
         if (LVar0 == 0) then begin
             LVar0 := 1;
-            giq_option(4, 1364, 600, Node046, 51);
+            giq_option(4, NAME, 600, Node046, BAD_REACTION);
         end
     end
     if ((local_var(17) == 1) and (party_member_obj(16777379) != 0) and (local_var(18) == 0)) then begin
         if (LVar0 == 0) then begin
             LVar0 := 1;
-            giq_option(4, 1364, 608, Node048, 50);
+            giq_option(4, NAME, 608, Node048, NEUTRAL_REACTION);
         end
     end
-    if ((global_var(449) == 2) and (global_var(803) != 0)) then begin
+    if ((global_var(GVAR_PLAYER_WAS_MARRIED) == 2) and (global_var(GVAR_PLAYER_IS_FOOL) != 0)) then begin
         if (LVar0 == 0) then begin
-            giq_option(4, 1364, 630, Node052, 49);
+            giq_option(4, NAME, 630, Node052, GOOD_REACTION);
         end
     end
-    giq_option(4, 1364, 335, Node017, 50);
-    giq_option(4, 1364, 589, Node999, 50);
+    giq_option(4, NAME, 335, Node017, NEUTRAL_REACTION);
+    giq_option(4, NAME, 589, Node999, NEUTRAL_REACTION);
     return 0;
     LVar0 := 0;
 end
@@ -1508,42 +1508,42 @@ end
 procedure Node041
 begin
     set_local_var(15, 1);
-    gsay_reply(1364, 570);
-    giq_option(4, 1364, 730, Node041a, 50);
+    Reply(570);
+    giq_option(4, NAME, 730, Node041a, NEUTRAL_REACTION);
 end
 
 procedure Node041a
 begin
-    if (global_var(644) < 20) then begin
-        gsay_reply(1364, 916);
+    if (global_var(GVAR_AHELDER_Q1_AGRO) < 20) then begin
+        Reply(916);
     end
     else begin
-        gsay_reply(1364, 731);
+        Reply(731);
     end
-    giq_option(4, 1364, 732, Node041b, 50);
+    giq_option(4, NAME, 732, Node041b, NEUTRAL_REACTION);
 end
 
 procedure Node041b
 begin
-    gsay_reply(1364, 733);
-    giq_option(4, 1364, 571, Node040, 50);
-    giq_option(4, 1364, 572, Node017, 50);
-    giq_option(4, 1364, 573, Node999, 50);
+    Reply(733);
+    giq_option(4, NAME, 571, Node040, NEUTRAL_REACTION);
+    giq_option(4, NAME, 572, Node017, NEUTRAL_REACTION);
+    giq_option(4, NAME, 573, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node042
 begin
     set_local_var(16, 1);
-    gsay_reply(1364, 576);
-    giq_option(4, 1364, 577, Node043, 50);
-    giq_option(4, 1364, 578, Node017, 50);
+    Reply(576);
+    giq_option(4, NAME, 577, Node043, NEUTRAL_REACTION);
+    giq_option(4, NAME, 578, Node017, NEUTRAL_REACTION);
 end
 
 procedure Node043
 begin
-    gsay_reply(1364, 580);
-    giq_option(4, 1364, 581, Node044, 50);
-    giq_option(4, 1364, 582, Node999, 50);
+    Reply(580);
+    giq_option(4, NAME, 581, Node044, NEUTRAL_REACTION);
+    giq_option(4, NAME, 582, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node044
@@ -1551,80 +1551,79 @@ begin
     gfade_out(5);
     give_exp_points(250);
     game_time_advance(72000);
-    display_msg(message_str(1364, 588));
+    display_mstr(588);
     gfade_in(5);
     call Node045();
 end
 
 procedure Node045
 begin
-    gsay_reply(1364, 585);
-    giq_option(4, 1364, 586, Node999, 50);
-    giq_option(4, 1364, 587, Node017, 50);
+    Reply(585);
+    giq_option(4, NAME, 586, Node999, NEUTRAL_REACTION);
+    giq_option(4, NAME, 587, Node017, NEUTRAL_REACTION);
 end
 
 procedure Node046
 begin
     set_local_var(17, 1);
-    gsay_reply(1364, 605);
-    giq_option(4, 1364, 606, Node047, 50);
+    Reply(605);
+    giq_option(4, NAME, 606, Node047, NEUTRAL_REACTION);
 end
 
 procedure Node047
 begin
-    gsay_reply(1364, 607);
-    giq_option(4, 1364, 610, Node049, 50);
+    Reply(607);
+    giq_option(4, NAME, 610, Node049, NEUTRAL_REACTION);
 end
 
 procedure Node048
 begin
-    gsay_reply(1364, 609);
-    giq_option(4, 1364, 610, Node049, 50);
+    Reply(609);
+    giq_option(4, NAME, 610, Node049, NEUTRAL_REACTION);
 end
 
 procedure Node049
 begin
     set_local_var(18, 1);
-    gsay_reply(1364, 610);
-    giq_option(4, 1364, 620, Node017, 50);
-    giq_option(4, 1364, 616, Node050, 50);
-    giq_option(4, 1364, 617, Node050, 50);
-    giq_option(4, 1364, 618, Node050, 50);
-    giq_option(4, 1364, 619, Node050, 50);
+    Reply(610);
+    giq_option(4, NAME, 620, Node017, NEUTRAL_REACTION);
+    giq_option(4, NAME, 616, Node050, NEUTRAL_REACTION);
+    giq_option(4, NAME, 617, Node050, NEUTRAL_REACTION);
+    giq_option(4, NAME, 618, Node050, NEUTRAL_REACTION);
+    giq_option(4, NAME, 619, Node050, NEUTRAL_REACTION);
 end
 
 procedure Node050
 begin
     give_exp_points(250);
-    display_msg(message_str(1364, 626));
+    display_mstr(626);
     call Node051();
 end
 
 procedure Node051
 begin
-    gsay_reply(1364, 625);
-    giq_option(4, 1364, 627, Node017, 50);
-    giq_option(4, 1364, 628, Node999, 50);
+    Reply(625);
+    giq_option(4, NAME, 627, Node017, NEUTRAL_REACTION);
+    giq_option(4, NAME, 628, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node052
 begin
-    if (message_str(1364, 100) != 0) then begin
+    if (mstr(100) != 0) then begin
         if (local_var(17) == 0) then begin
-            gsay_reply(1364, 632);
+            Reply(632);
         end
         else begin
-            gsay_reply(1364, 633);
+            Reply(633);
         end
     end
-    giq_option(4, 1364, 634, Node053, 49);
-    giq_option(4, 1364, 635, Node053, 49);
-    giq_option(4, 1364, 636, Node053, 49);
-    giq_option(4, 1364, 631, Node999, 50);
+    giq_option(4, NAME, 634, Node053, GOOD_REACTION);
+    giq_option(4, NAME, 635, Node053, GOOD_REACTION);
+    giq_option(4, NAME, 636, Node053, GOOD_REACTION);
+    giq_option(4, NAME, 631, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node053
 begin
-    display_msg(message_str(1364, 637));
+    display_mstr(637);
 end
-
