diff --git a/Mods/MegaMod/Original/AHHAKUN.ssl b/Mods/MegaMod/ReDefine/AHHAKUN.ssl
index fdcd85d..43e5d85 100644
--- a/Mods/MegaMod/Original/AHHAKUN.ssl
+++ b/Mods/MegaMod/ReDefine/AHHAKUN.ssl
@@ -269,48 +269,48 @@ end
 
 procedure EvaluateKarma
 begin
-    set_global_var(37, 0);
-    set_global_var(38, 0);
-    set_global_var(39, 0);
-    set_global_var(40, 0);
-    set_global_var(41, 0);
-    set_global_var(42, 0);
-    set_global_var(43, 0);
-    set_global_var(44, 0);
-    set_global_var(45, 0);
-    if (global_var(0) >= 1000) then begin
-        set_global_var(37, 1);
+    set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+    set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+    set_global_var(GVAR_KARMA_DEFENDER, 0);
+    set_global_var(GVAR_KARMA_WANDERER, 0);
+    set_global_var(GVAR_KARMA_BETRAYER, 0);
+    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+    set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+    set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+    if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
     end
     else begin
-        if (global_var(0) >= 750) then begin
-            set_global_var(38, 1);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+            set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
         end
         else begin
-            if (global_var(0) >= 500) then begin
-                set_global_var(39, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
             end
             else begin
-                if (global_var(0) >= 250) then begin
-                    set_global_var(40, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                    set_global_var(GVAR_KARMA_DEFENDER, 1);
                 end
                 else begin
-                    if (global_var(0) > -250) then begin
-                        set_global_var(41, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                        set_global_var(GVAR_KARMA_WANDERER, 1);
                     end
                     else begin
-                        if (global_var(0) > -500) then begin
-                            set_global_var(42, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                            set_global_var(GVAR_KARMA_BETRAYER, 1);
                         end
                         else begin
-                            if (global_var(0) > -750) then begin
-                                set_global_var(43, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                             end
                             else begin
-                                if (global_var(0) > -1000) then begin
-                                    set_global_var(44, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                    set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                 end
                                 else begin
-                                    set_global_var(45, 1);
+                                    set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                 end
                             end
                         end
@@ -323,33 +323,33 @@ end
 
 procedure CheckRep
 begin
-    if (((global_var(4) + global_var(5)) >= 25) and ((global_var(4) > (3 * global_var(5))) or (global_var(2) == 1)) and (global_var(1) == 0)) then begin
-        set_global_var(2, 1);
-        set_global_var(3, 0);
+    if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(4) > (3 * global_var(GVAR_GOOD_MONSTER))) or (global_var(GVAR_CHAMPION_REPUTATION) == 1)) and (global_var(GVAR_CHILDKILLER_REPUTATION) == 0)) then begin
+        set_global_var(GVAR_CHAMPION_REPUTATION, 1);
+        set_global_var(GVAR_BERSERKER_REPUTATION, 0);
     end
-    if (((global_var(4) + global_var(5)) >= 25) and ((global_var(5) > (2 * global_var(4))) or (global_var(3) == 1))) then begin
-        set_global_var(3, 1);
-        set_global_var(2, 0);
+    if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
+        set_global_var(GVAR_BERSERKER_REPUTATION, 1);
+        set_global_var(GVAR_CHAMPION_REPUTATION, 0);
     end
 end
 
 procedure critter_p_proc
 begin
-    if (cur_map_index == 127) then begin
+    if (CUR_MAP_ARDEAD) then begin
         if (local_var(4) == 0) then begin
-            if (metarule(17, 15) == 0) then begin
+            if (town_known(15) == MARK_STATE_UNKNOWN) then begin
                 debug_msg("  mark_on_map(" + 15 + ")");
-                mark_area_known(0, 15, 1);
+                mark_area_known(MARK_TYPE_TOWN, 15, MARK_STATE_KNOWN);
             end
-            if (global_var(494) < 4) then begin
-                set_global_var(494, 4);
+            if (global_var(GVAR_FALLOUT_2) < 4) then begin
+                set_global_var(GVAR_FALLOUT_2, 4);
             end
             dialogue_system_enter;
         end
     end
     else begin
-        if ((global_var(7) == 1) and obj_can_see_obj(self_obj, dude_obj)) then begin
-            attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+        if ((global_var(GVAR_ENEMY_ARROYO) == 1) and obj_can_see_obj(self_obj, dude_obj)) then begin
+            attack(dude_obj);
         end
         else begin
             if (local_var(15) == 1) then begin
@@ -361,7 +361,7 @@ begin
     end
     if ((local_var(5) == 2) and obj_can_see_obj(self_obj, dude_obj)) then begin
         set_local_var(5, 1);
-        attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+        attack(dude_obj);
     end
 end
 
@@ -380,12 +380,12 @@ begin
     end
     if (fixed_param == 3) then begin
         game_time_advance(60 * 10 * 60 * 12);
-        override_map_start(27322 % 200, 27322 / 200, elevation(dude_obj), 2);
+        override_map_start(27322 % 200, 27322 / 200, dude_elevation, 2);
         gfade_in(3);
         add_timer_event(self_obj, 80, 4);
     end
     if (fixed_param == 4) then begin
-        display_msg(message_str(12, 734));
+        display_mstr(734);
         reg_anim_func(2, dude_obj);
         reg_anim_func(1, 1);
         reg_anim_animate(dude_obj, 37, -1);
@@ -394,8 +394,8 @@ begin
     end
     if (fixed_param == 5) then begin
         give_exp_points(300);
-        display_msg(message_str(14, 100) + 300 + message_str(14, 101));
-        float_msg(dude_obj, message_str(12, 736), 0);
+        display_msg(g_mstr(100) + 300 + g_mstr(101));
+        float_msg(dude_obj, mstr(736), FLOAT_MSG_NORMAL);
         game_ui_enable;
     end
 end
@@ -411,35 +411,35 @@ procedure destroy_p_proc
 begin
     set_map_var(6, 0);
     if (source_obj == dude_obj) then begin
-        set_global_var(5, global_var(5) + 1);
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(4) > (3 * global_var(5))) or (global_var(2) == 1)) and (global_var(1) == 0)) then begin
-            set_global_var(2, 1);
-            set_global_var(3, 0);
+        set_global_var(GVAR_GOOD_MONSTER, global_var(GVAR_GOOD_MONSTER) + 1);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(4) > (3 * global_var(GVAR_GOOD_MONSTER))) or (global_var(GVAR_CHAMPION_REPUTATION) == 1)) and (global_var(GVAR_CHILDKILLER_REPUTATION) == 0)) then begin
+            set_global_var(GVAR_CHAMPION_REPUTATION, 1);
+            set_global_var(GVAR_BERSERKER_REPUTATION, 0);
         end
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(5) > (2 * global_var(4))) or (global_var(3) == 1))) then begin
-            set_global_var(3, 1);
-            set_global_var(2, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
+            set_global_var(GVAR_BERSERKER_REPUTATION, 1);
+            set_global_var(GVAR_CHAMPION_REPUTATION, 0);
         end
-        set_global_var(0, global_var(0) - 15);
+        set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) - 15);
         call EvaluateKarma();
     end
-    set_global_var(47, global_var(47) - 50);
-    set_global_var(7, 1);
+    set_global_var(GVAR_TOWN_REP_ARROYO, global_var(GVAR_TOWN_REP_ARROYO) - 50);
+    set_global_var(GVAR_ENEMY_ARROYO, 1);
 end
 
 procedure look_at_p_proc
 begin
     script_overrides;
-    if (not(metarule(17, 11) == 2)) then begin
-        display_msg(message_str(12, random(200, 201)));
+    if (not(town_known(11) == MARK_STATE_VISITED)) then begin
+        display_mstr(random(200, 201));
     end
     else begin
         if (local_var(9) == 0) then begin
             set_local_var(9, local_var(9) + 1);
-            display_msg(message_str(12, 101));
+            display_mstr(101);
         end
         else begin
-            display_msg(message_str(12, 102));
+            display_mstr(102);
         end
     end
 end
@@ -447,11 +447,11 @@ end
 procedure description_p_proc
 begin
     script_overrides;
-    if (not(metarule(17, 11) == 2)) then begin
-        display_msg(message_str(12, 204));
+    if (not(town_known(11) == MARK_STATE_VISITED)) then begin
+        display_mstr(204);
     end
     else begin
-        display_msg(message_str(12, 103));
+        display_mstr(103);
     end
 end
 
@@ -459,19 +459,19 @@ procedure damage_p_proc
 begin
     if (source_obj == dude_obj) then begin
         set_local_var(6, 1);
-        set_global_var(7, 1);
+        set_global_var(GVAR_ENEMY_ARROYO, 1);
     end
 end
 
 procedure map_enter_p_proc
 begin
     Only_Once := 0;
-    critter_add_trait(self_obj, 1, 6, 1);
-    critter_add_trait(self_obj, 1, 5, 4);
-    if (cur_map_index == 127) then begin
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 1);
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_ARROYO_SHAMAN);
+    if (CUR_MAP_ARDEAD) then begin
         anim(self_obj, 49, 0);
     end
-    if (metarule(14, 0)) then begin
+    if (map_first_run) then begin
         set_local_var(13, 1);
         set_local_var(12, 1);
     end
@@ -479,186 +479,186 @@ end
 
 procedure Node003
 begin
-    gsay_reply(12, 206);
-    giq_option(1, 14, 350, "Node999", 50);
+    Reply(206);
+    giq_option(1, SCRIPT_GENERIC, 350, "Node999", NEUTRAL_REACTION);
 end
 
 procedure Node004
 begin
     set_local_var(11, local_var(11) + 1);
-    gsay_reply(12, 208);
-    giq_option(-1, 12, 210, "Node005", 50);
-    if (get_critter_stat(dude_obj, 4) <= 4) then begin
-        giq_option(2, 12, 212, "Node006", 50);
-    end
-    giq_option(5, 12, 216, "Node066", 50);
-    if ((global_var(0) >= 15) and (get_critter_stat(dude_obj, 4) <= 7)) then begin
-        if ((get_critter_stat(dude_obj, 1) >= 6) and (global_var(9) != -1)) then begin
-            giq_option(5, 12, 218, "Node009", 50);
+    Reply(208);
+    giq_option(-1, NAME, 210, "Node005", NEUTRAL_REACTION);
+    if (dude_iq <= 4) then begin
+        giq_option(2, NAME, 212, "Node006", NEUTRAL_REACTION);
+    end
+    giq_option(5, NAME, 216, "Node066", NEUTRAL_REACTION);
+    if ((global_var(GVAR_PLAYER_REPUTATION) >= 15) and (dude_iq <= 7)) then begin
+        if ((dude_perception >= 6) and (global_var(GVAR_KILL_EVIL_PLANTS) != -1)) then begin
+            giq_option(5, NAME, 218, "Node009", NEUTRAL_REACTION);
         end
         else begin
-            giq_option(5, 12, 218, "Node008", 50);
+            giq_option(5, NAME, 218, "Node008", NEUTRAL_REACTION);
         end
     end
-    if (global_var(0) >= 15) then begin
-        if ((get_critter_stat(dude_obj, 1) >= 6) and (global_var(9) != -1)) then begin
-            giq_option(8, 12, 220, "Node009", 50);
+    if (global_var(GVAR_PLAYER_REPUTATION) >= 15) then begin
+        if ((dude_perception >= 6) and (global_var(GVAR_KILL_EVIL_PLANTS) != -1)) then begin
+            giq_option(8, NAME, 220, "Node009", NEUTRAL_REACTION);
         end
         else begin
-            giq_option(8, 12, 220, "Node008", 50);
+            giq_option(8, NAME, 220, "Node008", NEUTRAL_REACTION);
         end
     end
-    if ((global_var(0) < 15) and (get_critter_stat(dude_obj, 34) == 0)) then begin
-        giq_option(5, 12, 222, "Node007", 50);
+    if ((global_var(GVAR_PLAYER_REPUTATION) < 15) and (dude_gender == 0)) then begin
+        giq_option(5, NAME, 222, "Node007", NEUTRAL_REACTION);
     end
-    if ((global_var(0) < 15) and (get_critter_stat(dude_obj, 34) == 1)) then begin
-        giq_option(5, 12, 224, "Node007", 50);
+    if ((global_var(GVAR_PLAYER_REPUTATION) < 15) and (dude_gender == 1)) then begin
+        giq_option(5, NAME, 224, "Node007", NEUTRAL_REACTION);
     end
-    giq_option(5, 12, 226, "Node006", 50);
-    giq_option(1, 12, 228, "Node999", 50);
+    giq_option(5, NAME, 226, "Node006", NEUTRAL_REACTION);
+    giq_option(1, NAME, 228, "Node999", NEUTRAL_REACTION);
 end
 
 procedure Node005
 begin
-    gsay_reply(12, 230);
-    giq_option(1, 14, 350, "Node999", 50);
+    Reply(230);
+    giq_option(1, SCRIPT_GENERIC, 350, "Node999", NEUTRAL_REACTION);
 end
 
 procedure Node006
 begin
     set_local_var(16, 1);
-    gsay_reply(12, 232);
-    giq_option(1, 14, 350, "Node999", 50);
+    Reply(232);
+    giq_option(1, SCRIPT_GENERIC, 350, "Node999", NEUTRAL_REACTION);
 end
 
 procedure Node007
 begin
     set_local_var(16, 1);
-    gsay_reply(12, 234);
-    giq_option(1, 14, 350, "Node999", 50);
+    Reply(234);
+    giq_option(1, SCRIPT_GENERIC, 350, "Node999", NEUTRAL_REACTION);
 end
 
 procedure Node008
 begin
     set_local_var(16, 1);
-    gsay_reply(12, 236);
-    giq_option(1, 14, 350, "Node999", 50);
+    Reply(236);
+    giq_option(1, SCRIPT_GENERIC, 350, "Node999", NEUTRAL_REACTION);
 end
 
 procedure Node009
 begin
-    gsay_reply(12, 238);
-    giq_option(1, 12, 240, "Node999", 50);
-    if ((get_critter_stat(dude_obj, 34) == 0) and (get_critter_stat(dude_obj, 0) >= 5)) then begin
-        giq_option(8, 12, 242, "Node010", 50);
+    Reply(238);
+    giq_option(1, NAME, 240, "Node999", NEUTRAL_REACTION);
+    if ((dude_gender == 0) and (dude_strength >= 5)) then begin
+        giq_option(8, NAME, 242, "Node010", NEUTRAL_REACTION);
     end
-    if (get_critter_stat(dude_obj, 3) >= 6) then begin
-        giq_option(8, 12, 244, "Node010", 50);
+    if (dude_charisma >= 6) then begin
+        giq_option(8, NAME, 244, "Node010", NEUTRAL_REACTION);
     end
-    giq_option(1, 12, 246, "Node011", 50);
+    giq_option(1, NAME, 246, "Node011", NEUTRAL_REACTION);
 end
 
 procedure Node010
 begin
     set_local_var(16, 1);
-    gsay_reply(12, message_str(12, 248) + obj_name(dude_obj) + message_str(12, 249));
-    giq_option(1, 12, 250, "Node999", 50);
+    Reply(mstr(248) + dude_name + mstr(249));
+    giq_option(1, NAME, 250, "Node999", NEUTRAL_REACTION);
 end
 
 procedure Node011
 begin
-    gsay_reply(12, 252);
-    giq_option(1, 12, 254, "Node014", 50);
-    giq_option(8, 12, 256, "Node012", 50);
+    Reply(252);
+    giq_option(1, NAME, 254, "Node014", NEUTRAL_REACTION);
+    giq_option(8, NAME, 256, "Node012", NEUTRAL_REACTION);
 end
 
 procedure Node012
 begin
-    gsay_reply(12, 258);
-    giq_option(1, 12, 260, "Node999", 50);
-    giq_option(1, 12, 262, Node008, 50);
-    giq_option(1, 12, 264, "Node014", 50);
+    Reply(258);
+    giq_option(1, NAME, 260, "Node999", NEUTRAL_REACTION);
+    giq_option(1, NAME, 262, Node008, NEUTRAL_REACTION);
+    giq_option(1, NAME, 264, "Node014", NEUTRAL_REACTION);
 end
 
 procedure Node014
 begin
     if (local_var(19) == 0) then begin
-        add_mult_objs_to_inven(dude_obj, create_object_sid(534, 0, 0, -1), 2);
+        add_mult_objs_to_inven(dude_obj, create_object(534, 0, 0), 2);
         set_local_var(19, 1);
     end
-    set_global_var(9, 1);
-    gsay_reply(12, 268);
-    if (get_critter_stat(dude_obj, 2) > 4) then begin
-        giq_option(8, 12, 270, Node012, 50);
+    set_global_var(GVAR_KILL_EVIL_PLANTS, 1);
+    Reply(268);
+    if (dude_endurance > 4) then begin
+        giq_option(8, NAME, 270, Node012, NEUTRAL_REACTION);
     end
     else begin
-        giq_option(8, 12, 270, Node010, 50);
+        giq_option(8, NAME, 270, Node010, NEUTRAL_REACTION);
     end
-    if (get_critter_stat(dude_obj, 34) == 0) then begin
-        giq_option(1, 12, 272, "Node015", 50);
+    if (dude_gender == 0) then begin
+        giq_option(1, NAME, 272, "Node015", NEUTRAL_REACTION);
     end
     else begin
-        giq_option(1, 12, 272, "Node016", 50);
+        giq_option(1, NAME, 272, "Node016", NEUTRAL_REACTION);
     end
-    if (get_critter_stat(dude_obj, 1) >= 8) then begin
-        giq_option(8, 12, 274, "Node017", 50);
+    if (dude_perception >= 8) then begin
+        giq_option(8, NAME, 274, "Node017", NEUTRAL_REACTION);
     end
-    giq_option(1, 12, 276, "Node999", 50);
+    giq_option(1, NAME, 276, "Node999", NEUTRAL_REACTION);
 end
 
 procedure Node015
 begin
-    gsay_reply(12, 278);
-    giq_option(1, 14, 350, "Node999", 50);
+    Reply(278);
+    giq_option(1, SCRIPT_GENERIC, 350, "Node999", NEUTRAL_REACTION);
 end
 
 procedure Node016
 begin
-    gsay_reply(12, 280);
-    giq_option(1, 14, 350, "Node999", 50);
+    Reply(280);
+    giq_option(1, SCRIPT_GENERIC, 350, "Node999", NEUTRAL_REACTION);
 end
 
 procedure Node017
 begin
-    gsay_reply(12, 282);
-    giq_option(1, 14, 350, "Node999", 50);
+    Reply(282);
+    giq_option(1, SCRIPT_GENERIC, 350, "Node999", NEUTRAL_REACTION);
 end
 
 procedure Node018
 begin
-    gsay_reply(12, message_str(12, 284) + obj_name(dude_obj) + message_str(12, 285));
-    giq_option(1, 14, 350, "Node999", 50);
+    Reply(mstr(284) + dude_name + mstr(285));
+    giq_option(1, SCRIPT_GENERIC, 350, "Node999", NEUTRAL_REACTION);
 end
 
 procedure Node019
 begin
     set_local_var(11, local_var(11) + 1);
-    gsay_reply(12, message_str(12, 286) + obj_name(dude_obj) + message_str(12, 287));
-    giq_option(1, 12, 288, "Node028", 50);
-    giq_option(1, 12, 290, "Node999", 50);
+    Reply(mstr(286) + dude_name + mstr(287));
+    giq_option(1, NAME, 288, "Node028", NEUTRAL_REACTION);
+    giq_option(1, NAME, 290, "Node999", NEUTRAL_REACTION);
 end
 
 procedure Node020
 begin
     set_local_var(11, local_var(11) + 1);
-    gsay_reply(12, 292);
-    if ((global_var(9) == 0) or (global_var(9) == 1)) then begin
-        if (local_var(11) <= get_critter_stat(dude_obj, 6)) then begin
-            giq_option(1, 12, 294, "Node022", 50);
+    Reply(292);
+    if ((global_var(GVAR_KILL_EVIL_PLANTS) == 0) or (global_var(GVAR_KILL_EVIL_PLANTS) == 1)) then begin
+        if (local_var(11) <= dude_luck) then begin
+            giq_option(1, NAME, 294, "Node022", NEUTRAL_REACTION);
         end
         else begin
-            giq_option(1, 12, 294, "Node021", 50);
+            giq_option(1, NAME, 294, "Node021", NEUTRAL_REACTION);
         end
-        giq_option(8, 12, 296, "Node021", 50);
+        giq_option(8, NAME, 296, "Node021", NEUTRAL_REACTION);
     end
     else begin
-        if (global_var(9) == 2) then begin
-            giq_option(1, 12, 298, "Node027", 50);
-            giq_option(8, 12, 300, "Node027", 50);
+        if (global_var(GVAR_KILL_EVIL_PLANTS) == 2) then begin
+            giq_option(1, NAME, 298, "Node027", NEUTRAL_REACTION);
+            giq_option(8, NAME, 300, "Node027", NEUTRAL_REACTION);
         end
         else begin
-            giq_option(1, 12, 298, "Node026", 50);
-            giq_option(8, 12, 302, "Node026", 50);
+            giq_option(1, NAME, 298, "Node026", NEUTRAL_REACTION);
+            giq_option(8, NAME, 302, "Node026", NEUTRAL_REACTION);
         end
     end
 end
@@ -666,123 +666,123 @@ end
 procedure Node021
 begin
     set_local_var(16, 1);
-    gsay_reply(12, message_str(12, 304) + obj_name(dude_obj) + message_str(12, 305));
-    giq_option(1, 14, 350, "Node999", 50);
+    Reply(mstr(304) + dude_name + mstr(305));
+    giq_option(1, SCRIPT_GENERIC, 350, "Node999", NEUTRAL_REACTION);
 end
 
 procedure Node022
 begin
-    gsay_reply(12, 306);
-    giq_option(1, 12, 308, Node010, 50);
-    giq_option(8, 12, 310, "Node025", 50);
-    giq_option(1, 12, 312, "Node024", 50);
-    giq_option(8, 12, 314, "Node023", 50);
+    Reply(306);
+    giq_option(1, NAME, 308, Node010, NEUTRAL_REACTION);
+    giq_option(8, NAME, 310, "Node025", NEUTRAL_REACTION);
+    giq_option(1, NAME, 312, "Node024", NEUTRAL_REACTION);
+    giq_option(8, NAME, 314, "Node023", NEUTRAL_REACTION);
 end
 
 procedure Node023
 begin
-    gsay_reply(12, 316);
-    giq_option(1, 14, 350, "Node999", 50);
+    Reply(316);
+    giq_option(1, SCRIPT_GENERIC, 350, "Node999", NEUTRAL_REACTION);
 end
 
 procedure Node024
 begin
-    gsay_reply(12, 318);
-    giq_option(1, 14, 350, "Node999", 50);
+    Reply(318);
+    giq_option(1, SCRIPT_GENERIC, 350, "Node999", NEUTRAL_REACTION);
 end
 
 procedure Node025
 begin
-    gsay_reply(12, 320);
-    giq_option(1, 14, 350, "Node999", 50);
+    Reply(320);
+    giq_option(1, SCRIPT_GENERIC, 350, "Node999", NEUTRAL_REACTION);
 end
 
 procedure Node026
 begin
     set_local_var(16, 1);
-    gsay_reply(12, 322);
-    giq_option(1, 14, 350, "Node999", 50);
+    Reply(322);
+    giq_option(1, SCRIPT_GENERIC, 350, "Node999", NEUTRAL_REACTION);
 end
 
 procedure Node027
 begin
-    set_global_var(9, 4);
-    gsay_reply(12, 324);
-    giq_option(1, 12, 326, "Node028", 50);
-    giq_option(1, 12, 328, "Node028", 50);
+    set_global_var(GVAR_KILL_EVIL_PLANTS, 4);
+    Reply(324);
+    giq_option(1, NAME, 326, "Node028", NEUTRAL_REACTION);
+    giq_option(1, NAME, 328, "Node028", NEUTRAL_REACTION);
 end
 
 procedure Node028
 begin
-    gsay_reply(12, 330);
-    giq_option(1, 12, 332, "Node029", 50);
+    Reply(330);
+    giq_option(1, NAME, 332, "Node029", NEUTRAL_REACTION);
 end
 
 procedure Node029
 begin
-    gsay_reply(12, 334);
-    giq_option(1, 12, 336, "Node030", 50);
+    Reply(334);
+    giq_option(1, NAME, 336, "Node030", NEUTRAL_REACTION);
 end
 
 procedure Node030
 begin
-    gsay_reply(12, 338);
-    giq_option(1, 12, 340, "Node999", 50);
-    giq_option(1, 12, 342, "Node031", 50);
-    giq_option(1, 12, 344, "Node033", 50);
+    Reply(338);
+    giq_option(1, NAME, 340, "Node999", NEUTRAL_REACTION);
+    giq_option(1, NAME, 342, "Node031", NEUTRAL_REACTION);
+    giq_option(1, NAME, 344, "Node033", NEUTRAL_REACTION);
 end
 
 procedure Node031
 begin
-    gsay_reply(12, 346);
-    giq_option(1, 12, 348, "Node032", 50);
+    Reply(346);
+    giq_option(1, NAME, 348, "Node032", NEUTRAL_REACTION);
 end
 
 procedure Node032
 begin
-    gsay_reply(12, 350);
-    giq_option(1, 12, 352, "Node033", 50);
+    Reply(350);
+    giq_option(1, NAME, 352, "Node033", NEUTRAL_REACTION);
 end
 
 procedure Node033
 begin
     game_time_advance(60 * 60 * 10);
-    critter_mod_skill(dude_obj, 7, 4);
-    display_msg(message_str(12, 718));
-    critter_mod_skill(dude_obj, 12, 4);
-    display_msg(message_str(12, 720));
-    set_global_var(650, 1);
-    gsay_reply(12, 354);
-    giq_option(1, 12, 356, "Node036", 50);
-    giq_option(1, 12, 358, "Node034", 50);
+    critter_mod_skill(dude_obj, SKILL_DOCTOR, 4);
+    display_mstr(718);
+    critter_mod_skill(dude_obj, SKILL_SCIENCE, 4);
+    display_mstr(720);
+    set_global_var(GVAR_ACTEMVIL_TALK, 1);
+    Reply(354);
+    giq_option(1, NAME, 356, "Node036", NEUTRAL_REACTION);
+    giq_option(1, NAME, 358, "Node034", NEUTRAL_REACTION);
 end
 
 procedure Node034
 begin
-    gsay_reply(12, 360);
-    giq_option(1, 12, 362, "Node036", 50);
+    Reply(360);
+    giq_option(1, NAME, 362, "Node036", NEUTRAL_REACTION);
 end
 
 procedure Node035
 begin
-    gsay_reply(12, 366);
-    giq_option(1, 14, 350, "Node999", 50);
+    Reply(366);
+    giq_option(1, SCRIPT_GENERIC, 350, "Node999", NEUTRAL_REACTION);
 end
 
 procedure Node036
 begin
-    set_global_var(639, 2);
-    gsay_reply(12, 368);
-    giq_option(1, 12, 370, "Node042", 50);
-    giq_option(1, 12, 372, "Node037", 50);
-    giq_option(1, 12, 374, "Node999", 50);
+    set_global_var(GVAR_ARROYO_MEMBER, 2);
+    Reply(368);
+    giq_option(1, NAME, 370, "Node042", NEUTRAL_REACTION);
+    giq_option(1, NAME, 372, "Node037", NEUTRAL_REACTION);
+    giq_option(1, NAME, 374, "Node999", NEUTRAL_REACTION);
 end
 
 procedure Node037
 begin
-    gsay_reply(12, 376);
-    giq_option(1, 12, 378, "Node038", 50);
-    giq_option(1, 12, 380, "Node042", 50);
+    Reply(376);
+    giq_option(1, NAME, 378, "Node038", NEUTRAL_REACTION);
+    giq_option(1, NAME, 380, "Node042", NEUTRAL_REACTION);
 end
 
 procedure Node038
@@ -790,205 +790,205 @@ begin
     gfade_out(3);
     game_time_advance(60 * 60 * 10);
     gfade_in(3);
-    gsay_reply(12, 382);
-    giq_option(1, 12, 384, "Node040", 50);
-    giq_option(1, 12, 386, "Node039", 50);
-    giq_option(1, 12, 388, "Node040", 50);
-    giq_option(1, 12, 390, "Node039", 50);
+    Reply(382);
+    giq_option(1, NAME, 384, "Node040", NEUTRAL_REACTION);
+    giq_option(1, NAME, 386, "Node039", NEUTRAL_REACTION);
+    giq_option(1, NAME, 388, "Node040", NEUTRAL_REACTION);
+    giq_option(1, NAME, 390, "Node039", NEUTRAL_REACTION);
 end
 
 procedure Node039
 begin
-    gsay_reply(12, 392);
-    giq_option(1, 14, 350, "Node999", 50);
+    Reply(392);
+    giq_option(1, SCRIPT_GENERIC, 350, "Node999", NEUTRAL_REACTION);
 end
 
 procedure Node040
 begin
-    gfade_out(1);
+    fade_out;
     game_time_advance(10 * 60 * 10);
-    gfade_in(1);
-    gsay_reply(12, 394);
-    giq_option(1, 12, 396, Node040, 50);
-    giq_option(1, 12, 398, "Node041", 50);
-    giq_option(1, 12, 400, Node039, 50);
+    fade_in;
+    Reply(394);
+    giq_option(1, NAME, 396, Node040, NEUTRAL_REACTION);
+    giq_option(1, NAME, 398, "Node041", NEUTRAL_REACTION);
+    giq_option(1, NAME, 400, Node039, NEUTRAL_REACTION);
 end
 
 procedure Node041
 begin
-    gsay_reply(12, message_str(12, 402) + obj_name(dude_obj) + message_str(12, 403));
-    giq_option(1, 12, 404, "Node042", 50);
-    giq_option(1, 12, 406, "Node999", 50);
+    Reply(mstr(402) + dude_name + mstr(403));
+    giq_option(1, NAME, 404, "Node042", NEUTRAL_REACTION);
+    giq_option(1, NAME, 406, "Node999", NEUTRAL_REACTION);
 end
 
 procedure Node042
 begin
-    gsay_reply(12, 408);
-    giq_option(1, 12, 410, Node037, 50);
-    giq_option(1, 12, 412, "Node999", 50);
+    Reply(408);
+    giq_option(1, NAME, 410, Node037, NEUTRAL_REACTION);
+    giq_option(1, NAME, 412, "Node999", NEUTRAL_REACTION);
 end
 
 procedure Node043
 begin
     set_local_var(11, local_var(11) + 1);
-    gsay_reply(12, 414);
-    giq_option(1, 12, 416, Node037, 50);
-    giq_option(1, 12, 418, Node037, 50);
-    giq_option(1, 12, 420, Node042, 50);
-    giq_option(1, 12, 422, "Node999", 50);
+    Reply(414);
+    giq_option(1, NAME, 416, Node037, NEUTRAL_REACTION);
+    giq_option(1, NAME, 418, Node037, NEUTRAL_REACTION);
+    giq_option(1, NAME, 420, Node042, NEUTRAL_REACTION);
+    giq_option(1, NAME, 422, "Node999", NEUTRAL_REACTION);
 end
 
 procedure Node044
 begin
-    gsay_reply(12, message_str(12, 424) + obj_name(dude_obj) + message_str(12, 425));
-    if (global_var(0) >= 30) then begin
-        giq_option(1, 12, 426, "Node045", 50);
+    Reply(mstr(424) + dude_name + mstr(425));
+    if (global_var(GVAR_PLAYER_REPUTATION) >= 30) then begin
+        giq_option(1, NAME, 426, "Node045", NEUTRAL_REACTION);
     end
     else begin
-        giq_option(1, 12, 426, "Node047", 50);
+        giq_option(1, NAME, 426, "Node047", NEUTRAL_REACTION);
     end
-    giq_option(1, 12, 428, "Node048", 50);
+    giq_option(1, NAME, 428, "Node048", NEUTRAL_REACTION);
 end
 
 procedure Node045
 begin
-    gsay_reply(12, 430);
-    giq_option(1, 12, 432, "NodeStupka1", 50);
+    Reply(430);
+    giq_option(1, NAME, 432, "NodeStupka1", NEUTRAL_REACTION);
 end
 
 procedure NodeStupka1
 begin
-    gsay_reply(12, 760);
-    giq_option(1, 12, 762, "NodeStupka2", 50);
+    Reply(760);
+    giq_option(1, NAME, 762, "NodeStupka2", NEUTRAL_REACTION);
 end
 
 procedure NodeStupka2
 begin
-    gsay_reply(12, 764);
-    giq_option(1, 12, 766, "NodeSmoke", 50);
+    Reply(764);
+    giq_option(1, NAME, 766, "NodeSmoke", NEUTRAL_REACTION);
 end
 
 procedure NodeSmoke
 begin
-    gsay_reply(12, 768);
-    giq_option(1, 12, 770, "Node046", 50);
-    giq_option(1, 12, 772, "Node048", 50);
+    Reply(768);
+    giq_option(1, NAME, 770, "Node046", NEUTRAL_REACTION);
+    giq_option(1, NAME, 772, "Node048", NEUTRAL_REACTION);
 end
 
 procedure NodeStars
 begin
-    gsay_reply(12, 780);
-    giq_option(1, 12, 782, "Node999", 50);
+    Reply(780);
+    giq_option(1, NAME, 782, "Node999", NEUTRAL_REACTION);
 end
 
 procedure Node046
 begin
-    add_mult_objs_to_inven(dude_obj, create_object_sid(606, 0, 0, -1), 1);
-    add_mult_objs_to_inven(dude_obj, create_object_sid(533, 0, 0, 1326), 1);
-    set_global_var(650, 2);
-    set_global_var(685, 1);
+    add_mult_objs_to_inven(dude_obj, create_object(606, 0, 0), 1);
+    add_mult_objs_to_inven(dude_obj, create_object_sid(533, 0, 0, SCRIPT_ZISTUPKA), 1);
+    set_global_var(GVAR_ACTEMVIL_TALK, 2);
+    set_global_var(GVAR_JENFARM_QUEST, 1);
     set_local_var(15, 1);
-    set_global_var(634, 1);
-    gsay_reply(12, 436);
-    giq_option(1, 12, 438, "Node999", 50);
+    set_global_var(GVAR_HAKUNIN_TEACH, 1);
+    Reply(436);
+    giq_option(1, NAME, 438, "Node999", NEUTRAL_REACTION);
 end
 
 procedure Node047
 begin
-    gsay_reply(12, 444);
-    giq_option(1, 14, 350, "Node999", 50);
+    Reply(444);
+    giq_option(1, SCRIPT_GENERIC, 350, "Node999", NEUTRAL_REACTION);
 end
 
 procedure Node048
 begin
-    gsay_reply(12, 446);
-    giq_option(1, 14, 350, "Node999", 50);
+    Reply(446);
+    giq_option(1, SCRIPT_GENERIC, 350, "Node999", NEUTRAL_REACTION);
 end
 
 procedure Node049
 begin
-    gsay_reply(12, 448);
-    giq_option(1, 14, 350, "Node999", 50);
+    Reply(448);
+    giq_option(1, SCRIPT_GENERIC, 350, "Node999", NEUTRAL_REACTION);
 end
 
 procedure Node051
 begin
-    gsay_reply(12, message_str(12, 452) + obj_name(dude_obj) + message_str(12, 453));
-    if ((get_critter_stat(dude_obj, 2) < 4) and (local_var(12) > 0)) then begin
-        if (global_var(0) >= 30) then begin
-            giq_option(1, 12, 454, "Node059", 50);
+    Reply(mstr(452) + dude_name + mstr(453));
+    if ((dude_endurance < 4) and (local_var(12) > 0)) then begin
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 30) then begin
+            giq_option(1, NAME, 454, "Node059", NEUTRAL_REACTION);
         end
         else begin
-            giq_option(1, 12, 454, "Node063", 50);
+            giq_option(1, NAME, 454, "Node063", NEUTRAL_REACTION);
         end
     end
-    if (((get_critter_stat(dude_obj, 0) + get_critter_stat(dude_obj, 5)) < 12) and (local_var(12) > 0)) then begin
-        if (global_var(0) >= 30) then begin
-            giq_option(1, 12, 456, "Node059", 50);
+    if (((dude_strength + get_critter_stat(dude_obj, STAT_ag)) < 12) and (local_var(12) > 0)) then begin
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 30) then begin
+            giq_option(1, NAME, 456, "Node059", NEUTRAL_REACTION);
         end
         else begin
-            giq_option(1, 12, 456, "Node063", 50);
+            giq_option(1, NAME, 456, "Node063", NEUTRAL_REACTION);
         end
     end
     if (local_var(12) > 0) then begin
-        if (global_var(0) >= 30) then begin
-            giq_option(1, 12, 458, "Node055", 50);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 30) then begin
+            giq_option(1, NAME, 458, "Node055", NEUTRAL_REACTION);
         end
         else begin
-            giq_option(1, 12, 458, "Node063", 50);
+            giq_option(1, NAME, 458, "Node063", NEUTRAL_REACTION);
         end
     end
     if (local_var(13) > 0) then begin
-        giq_option(5, 12, 528, "Node089", 50);
+        giq_option(5, NAME, 528, "Node089", NEUTRAL_REACTION);
     end
     else begin
-        giq_option(5, 12, 528, "Node054", 50);
+        giq_option(5, NAME, 528, "Node054", NEUTRAL_REACTION);
     end
-    if (get_critter_stat(dude_obj, 4) <= 4) then begin
+    if (dude_iq <= 4) then begin
         if (local_var(13) > 0) then begin
-            giq_option(2, 12, 530, "Node089", 50);
+            giq_option(2, NAME, 530, "Node089", NEUTRAL_REACTION);
         end
         else begin
-            giq_option(2, 12, 530, "Node054", 50);
+            giq_option(2, NAME, 530, "Node054", NEUTRAL_REACTION);
         end
     end
     if ((obj_is_carrying_obj_pid(dude_obj, 757) > 0) and (local_var(10) > 0)) then begin
-        giq_option(1, 12, 774, "Node094one", 50);
+        giq_option(1, NAME, 774, "Node094one", NEUTRAL_REACTION);
     end
     if ((obj_is_carrying_obj_pid(dude_obj, 757) > 1) and (local_var(10) > 0)) then begin
-        giq_option(1, 12, 776, "Node094all", 50);
+        giq_option(1, NAME, 776, "Node094all", NEUTRAL_REACTION);
     end
     if (local_var(10) == 0) then begin
-        giq_option(1, 12, 460, "Node092", 50);
+        giq_option(1, NAME, 460, "Node092", NEUTRAL_REACTION);
     end
     else begin
-        giq_option(1, 12, 778, "Node999", 50);
+        giq_option(1, NAME, 778, "Node999", NEUTRAL_REACTION);
     end
 end
 
 procedure Node052
 begin
-    gsay_reply(12, message_str(12, 462) + obj_name(dude_obj) + message_str(12, 463));
-    giq_option(1, 12, 464, "Node999", 50);
+    Reply(mstr(462) + dude_name + mstr(463));
+    giq_option(1, NAME, 464, "Node999", NEUTRAL_REACTION);
 end
 
 procedure Node053
 begin
-    gsay_reply(12, 466);
-    giq_option(1, 12, 468, "Node999", 50);
+    Reply(466);
+    giq_option(1, NAME, 468, "Node999", NEUTRAL_REACTION);
 end
 
 procedure Node054
 begin
-    gsay_reply(12, 470);
-    giq_option(1, 12, 472, "Node999", 50);
+    Reply(470);
+    giq_option(1, NAME, 472, "Node999", NEUTRAL_REACTION);
 end
 
 procedure Node055
 begin
-    gsay_reply(12, 474);
-    giq_option(1, 12, 476, "Node056", 50);
-    giq_option(1, 12, 478, "Node057", 50);
+    Reply(474);
+    giq_option(1, NAME, 476, "Node056", NEUTRAL_REACTION);
+    giq_option(1, NAME, 478, "Node057", NEUTRAL_REACTION);
 end
 
 procedure Node056
@@ -996,39 +996,39 @@ begin
     gfade_out(3);
     game_time_advance(120 * 60 * 10);
     gfade_in(3);
-    critter_mod_skill(dude_obj, 12, 6);
-    display_msg(message_str(12, 722));
+    critter_mod_skill(dude_obj, SKILL_SCIENCE, 6);
+    display_mstr(722);
     set_local_var(12, local_var(12) - 1);
-    gsay_reply(12, 480);
-    giq_option(1, 12, 482, "Node999", 50);
+    Reply(480);
+    giq_option(1, NAME, 482, "Node999", NEUTRAL_REACTION);
 end
 
 procedure Node057
 begin
-    gsay_reply(12, 484);
-    giq_option(1, 12, 488, "Node999", 50);
+    Reply(484);
+    giq_option(1, NAME, 488, "Node999", NEUTRAL_REACTION);
 end
 
 procedure Node059
 begin
-    gsay_reply(12, 494);
+    Reply(494);
     if (local_var(12) > 0) then begin
-        giq_option(1, 12, 496, "Node060", 50);
+        giq_option(1, NAME, 496, "Node060", NEUTRAL_REACTION);
     end
     else begin
-        giq_option(1, 12, 496, Node052, 50);
+        giq_option(1, NAME, 496, Node052, NEUTRAL_REACTION);
     end
     if (local_var(13) > 0) then begin
-        giq_option(1, 12, 498, "Node062", 50);
+        giq_option(1, NAME, 498, "Node062", NEUTRAL_REACTION);
     end
     else begin
-        giq_option(1, 12, 498, Node054, 50);
+        giq_option(1, NAME, 498, Node054, NEUTRAL_REACTION);
     end
     if (local_var(12) > 0) then begin
-        giq_option(1, 12, 500, "Node061", 50);
+        giq_option(1, NAME, 500, "Node061", NEUTRAL_REACTION);
     end
     else begin
-        giq_option(1, 12, 500, Node052, 50);
+        giq_option(1, NAME, 500, Node052, NEUTRAL_REACTION);
     end
 end
 
@@ -1038,10 +1038,10 @@ begin
     game_time_advance(120 * 60 * 10);
     gfade_in(3);
     set_local_var(12, local_var(12) - 1);
-    critter_mod_skill(dude_obj, 17, 6);
-    display_msg(message_str(12, 724));
-    gsay_reply(12, 502);
-    giq_option(1, 12, 504, "Node999", 50);
+    critter_mod_skill(dude_obj, SKILL_OUTDOORSMAN, 6);
+    display_mstr(724);
+    Reply(502);
+    giq_option(1, NAME, 504, "Node999", NEUTRAL_REACTION);
 end
 
 procedure Node061
@@ -1050,21 +1050,21 @@ begin
     game_time_advance(120 * 60 * 10);
     gfade_in(3);
     set_local_var(12, local_var(12) - 1);
-    critter_mod_skill(dude_obj, 7, 6);
-    display_msg(message_str(12, 726));
-    gsay_reply(12, 506);
-    giq_option(1, 12, 508, "Node999", 50);
+    critter_mod_skill(dude_obj, SKILL_DOCTOR, 6);
+    display_mstr(726);
+    Reply(506);
+    giq_option(1, NAME, 508, "Node999", NEUTRAL_REACTION);
 end
 
 procedure Node062
 begin
-    add_mult_objs_to_inven(dude_obj, create_object_sid(755, 0, 0, -1), 3);
-    add_mult_objs_to_inven(dude_obj, create_object_sid(754, 0, 0, -1), 2);
-    add_mult_objs_to_inven(dude_obj, create_object_sid(273, 0, 0, -1), 1);
+    add_mult_objs_to_inven(dude_obj, create_object(755, 0, 0), 3);
+    add_mult_objs_to_inven(dude_obj, create_object(754, 0, 0), 2);
+    add_mult_objs_to_inven(dude_obj, create_object(273, 0, 0), 1);
     set_local_var(13, local_var(13) - 1);
-    gsay_reply(12, 510);
-    giq_option(1, 12, 512, "Node999", 50);
-    giq_option(1, 12, 514, "Node999", 50);
+    Reply(510);
+    giq_option(1, NAME, 512, "Node999", NEUTRAL_REACTION);
+    giq_option(1, NAME, 514, "Node999", NEUTRAL_REACTION);
 end
 
 procedure Node063
@@ -1072,164 +1072,164 @@ begin
     gfade_out(3);
     game_time_advance(60 * 60 * 10);
     gfade_in(3);
-    gsay_reply(12, 516);
-    giq_option(1, 12, 518, "Node999", 50);
+    Reply(516);
+    giq_option(1, NAME, 518, "Node999", NEUTRAL_REACTION);
 end
 
 procedure Node064
 begin
-    gsay_reply(12, message_str(12, 520) + obj_name(dude_obj) + message_str(12, 521));
-    giq_option(-1, 12, 522, Node005, 50);
-    if ((get_critter_stat(dude_obj, 4) <= 4) and (local_var(14) == 0)) then begin
-        giq_option(2, 12, 524, "Node065", 50);
+    Reply(mstr(520) + dude_name + mstr(521));
+    giq_option(-1, NAME, 522, Node005, NEUTRAL_REACTION);
+    if ((dude_iq <= 4) and (local_var(14) == 0)) then begin
+        giq_option(2, NAME, 524, "Node065", NEUTRAL_REACTION);
     end
     if (local_var(14) == 0) then begin
-        giq_option(5, 12, 526, "Node065", 50);
+        giq_option(5, NAME, 526, "Node065", NEUTRAL_REACTION);
     end
     if (local_var(13) > 0) then begin
-        giq_option(5, 12, 528, "Node089", 50);
+        giq_option(5, NAME, 528, "Node089", NEUTRAL_REACTION);
     end
     else begin
-        giq_option(5, 12, 528, Node054, 50);
+        giq_option(5, NAME, 528, Node054, NEUTRAL_REACTION);
     end
-    if (get_critter_stat(dude_obj, 4) <= 4) then begin
+    if (dude_iq <= 4) then begin
         if (local_var(13) > 0) then begin
-            giq_option(2, 12, 530, "Node089", 50);
+            giq_option(2, NAME, 530, "Node089", NEUTRAL_REACTION);
         end
         else begin
-            giq_option(2, 12, 530, Node054, 50);
+            giq_option(2, NAME, 530, Node054, NEUTRAL_REACTION);
         end
     end
     if (local_var(14) == 3) then begin
-        giq_option(5, 12, 532, "Node075", 50);
+        giq_option(5, NAME, 532, "Node075", NEUTRAL_REACTION);
     end
     if (local_var(14) == 3) then begin
-        giq_option(-4, 12, 534, "Node075", 50);
+        giq_option(-4, NAME, 534, "Node075", NEUTRAL_REACTION);
     end
-    giq_option(2, 12, 535, "Node999", 50);
+    giq_option(2, NAME, 535, "Node999", NEUTRAL_REACTION);
 end
 
 procedure Node065
 begin
-    gsay_reply(12, 538);
-    giq_option(-7, 12, 540, "Node084", 50);
-    giq_option(-4, 12, 542, "Node074", 50);
-    giq_option(5, 12, 546, "Node084", 50);
-    giq_option(5, 12, 552, "Node084", 50);
+    Reply(538);
+    giq_option(-7, NAME, 540, "Node084", NEUTRAL_REACTION);
+    giq_option(-4, NAME, 542, "Node074", NEUTRAL_REACTION);
+    giq_option(5, NAME, 546, "Node084", NEUTRAL_REACTION);
+    giq_option(5, NAME, 552, "Node084", NEUTRAL_REACTION);
 end
 
 procedure Node066
 begin
-    gsay_reply(12, 558);
-    if (global_var(644) != 20) then begin
-        giq_option(1, 12, 560, "Node069", 50);
+    Reply(558);
+    if (global_var(GVAR_AHELDER_Q1_AGRO) != 20) then begin
+        giq_option(1, NAME, 560, "Node069", NEUTRAL_REACTION);
     end
-    giq_option(1, 12, 562, "Node070", 50);
-    giq_option(1, 12, 564, "Node071", 50);
-    giq_option(1, 12, 568, "Node999", 50);
+    giq_option(1, NAME, 562, "Node070", NEUTRAL_REACTION);
+    giq_option(1, NAME, 564, "Node071", NEUTRAL_REACTION);
+    giq_option(1, NAME, 568, "Node999", NEUTRAL_REACTION);
 end
 
 procedure Node067
 begin
-    gsay_reply(12, message_str(12, 570) + obj_name(dude_obj) + message_str(12, 571));
-    giq_option(1, 12, 572, "Node071", 50);
-    giq_option(1, 12, 574, "Node999", 50);
+    Reply(mstr(570) + dude_name + mstr(571));
+    giq_option(1, NAME, 572, "Node071", NEUTRAL_REACTION);
+    giq_option(1, NAME, 574, "Node999", NEUTRAL_REACTION);
 end
 
 procedure Node068
 begin
-    gsay_reply(12, 576);
-    giq_option(1, 12, 580, "Node071", 50);
+    Reply(576);
+    giq_option(1, NAME, 580, "Node071", NEUTRAL_REACTION);
 end
 
 procedure Node069
 begin
-    gsay_reply(12, 582);
-    giq_option(1, 12, 584, "Node070", 50);
-    giq_option(1, 12, 586, "Node071", 50);
-    giq_option(1, 12, 588, "Node999", 50);
+    Reply(582);
+    giq_option(1, NAME, 584, "Node070", NEUTRAL_REACTION);
+    giq_option(1, NAME, 586, "Node071", NEUTRAL_REACTION);
+    giq_option(1, NAME, 588, "Node999", NEUTRAL_REACTION);
 end
 
 procedure Node070
 begin
-    gsay_reply(12, 590);
-    if (global_var(644) != 20) then begin
-        giq_option(1, 12, 592, Node069, 50);
+    Reply(590);
+    if (global_var(GVAR_AHELDER_Q1_AGRO) != 20) then begin
+        giq_option(1, NAME, 592, Node069, NEUTRAL_REACTION);
     end
-    giq_option(1, 12, 594, "Node071", 50);
-    giq_option(1, 12, 596, "Node999", 50);
+    giq_option(1, NAME, 594, "Node071", NEUTRAL_REACTION);
+    giq_option(1, NAME, 596, "Node999", NEUTRAL_REACTION);
 end
 
 procedure Node071
 begin
-    gsay_reply(12, 598);
-    giq_option(1, 12, 600, "Node999", 50);
-    giq_option(1, 12, 602, "Node999", 50);
+    Reply(598);
+    giq_option(1, NAME, 600, "Node999", NEUTRAL_REACTION);
+    giq_option(1, NAME, 602, "Node999", NEUTRAL_REACTION);
 end
 
 procedure Node072
 begin
-    gsay_reply(12, 604);
-    giq_option(1, 14, 350, "Node999", 50);
+    Reply(604);
+    giq_option(1, SCRIPT_GENERIC, 350, "Node999", NEUTRAL_REACTION);
 end
 
 procedure Node073
 begin
     set_local_var(0, 0);
     set_local_var(14, 1);
-    gsay_reply(12, 606);
-    giq_option(1, 14, 350, "Node999", 50);
+    Reply(606);
+    giq_option(1, SCRIPT_GENERIC, 350, "Node999", NEUTRAL_REACTION);
 end
 
 procedure Node074
 begin
     set_local_var(14, 1);
-    gsay_reply(12, 608);
-    giq_option(1, 14, 350, "Node999", 50);
+    Reply(608);
+    giq_option(1, SCRIPT_GENERIC, 350, "Node999", NEUTRAL_REACTION);
 end
 
 procedure Node075
 begin
-    gsay_reply(12, message_str(12, 610) + obj_name(dude_obj) + message_str(12, 611));
+    Reply(mstr(610) + dude_name + mstr(611));
     if ((obj_is_carrying_obj_pid(dude_obj, 365) < 2) or (obj_is_carrying_obj_pid(dude_obj, 92) < 1)) then begin
-        giq_option(1, 12, 612, "Node076", 50);
-        giq_option(-4, 12, 618, "Node999", 50);
-        giq_option(5, 12, 620, "Node999", 50);
+        giq_option(1, NAME, 612, "Node076", NEUTRAL_REACTION);
+        giq_option(-4, NAME, 618, "Node999", NEUTRAL_REACTION);
+        giq_option(5, NAME, 620, "Node999", NEUTRAL_REACTION);
     end
     else begin
-        giq_option(-4, 12, 614, "Node077", 50);
-        giq_option(5, 12, 616, "Node077", 50);
+        giq_option(-4, NAME, 614, "Node077", NEUTRAL_REACTION);
+        giq_option(5, NAME, 616, "Node077", NEUTRAL_REACTION);
     end
 end
 
 procedure Node076
 begin
-    gsay_reply(12, obj_name(dude_obj) + message_str(12, 622));
-    giq_option(-4, 12, 624, Node073, 50);
-    giq_option(5, 12, 626, "Node999", 50);
-    giq_option(8, 12, 628, Node073, 50);
+    Reply(dude_name + mstr(622));
+    giq_option(-4, NAME, 624, Node073, NEUTRAL_REACTION);
+    giq_option(5, NAME, 626, "Node999", NEUTRAL_REACTION);
+    giq_option(8, NAME, 628, Node073, NEUTRAL_REACTION);
 end
 
 procedure Node077
 begin
-    gsay_reply(12, 630);
-    giq_option(1, 12, 632, "Node078", 50);
-    giq_option(8, 12, 636, "Node078", 50);
+    Reply(630);
+    giq_option(1, NAME, 632, "Node078", NEUTRAL_REACTION);
+    giq_option(8, NAME, 636, "Node078", NEUTRAL_REACTION);
 end
 
 procedure Node078
 begin
-    gsay_reply(12, 640);
+    Reply(640);
     if (obj_is_carrying_obj_pid(dude_obj, 7) > 0) then begin
-        giq_option(1, 12, 642, "Node079", 50);
+        giq_option(1, NAME, 642, "Node079", NEUTRAL_REACTION);
     end
     if (obj_is_carrying_obj_pid(dude_obj, 4) > 0) then begin
-        giq_option(1, 12, 646, "Node081", 50);
+        giq_option(1, NAME, 646, "Node081", NEUTRAL_REACTION);
     end
     if (obj_is_carrying_obj_pid(dude_obj, 45) > 0) then begin
-        giq_option(1, 12, 648, "Node082", 50);
+        giq_option(1, NAME, 648, "Node082", NEUTRAL_REACTION);
     end
-    giq_option(1, 12, 650, "Node999", 50);
+    giq_option(1, NAME, 650, "Node999", NEUTRAL_REACTION);
 end
 
 procedure Node079
@@ -1240,11 +1240,11 @@ begin
     rm_mult_objs_from_inven(dude_obj, obj_carrying_pid_obj(dude_obj, 365), 2);
     rm_mult_objs_from_inven(dude_obj, obj_carrying_pid_obj(dude_obj, 92), 1);
     rm_mult_objs_from_inven(dude_obj, obj_carrying_pid_obj(dude_obj, 7), 1);
-    add_mult_objs_to_inven(dude_obj, create_object_sid(701, 0, 0, -1), 1);
-    gsay_reply(12, 652);
-    giq_option(-4, 12, 654, "Node999", 50);
-    giq_option(5, 12, 656, "Node083", 50);
-    giq_option(5, 12, 658, "Node999", 50);
+    add_mult_objs_to_inven(dude_obj, create_object(701, 0, 0), 1);
+    Reply(652);
+    giq_option(-4, NAME, 654, "Node999", NEUTRAL_REACTION);
+    giq_option(5, NAME, 656, "Node083", NEUTRAL_REACTION);
+    giq_option(5, NAME, 658, "Node999", NEUTRAL_REACTION);
 end
 
 procedure Node081
@@ -1255,11 +1255,11 @@ begin
     rm_mult_objs_from_inven(dude_obj, obj_carrying_pid_obj(dude_obj, 365), 2);
     rm_mult_objs_from_inven(dude_obj, obj_carrying_pid_obj(dude_obj, 92), 1);
     rm_mult_objs_from_inven(dude_obj, obj_carrying_pid_obj(dude_obj, 4), 1);
-    add_mult_objs_to_inven(dude_obj, create_object_sid(700, 0, 0, -1), 1);
-    gsay_reply(12, 668);
-    giq_option(-4, 12, 670, "Node999", 50);
-    giq_option(8, 12, 672, "Node083", 50);
-    giq_option(5, 12, 674, "Node999", 50);
+    add_mult_objs_to_inven(dude_obj, create_object(700, 0, 0), 1);
+    Reply(668);
+    giq_option(-4, NAME, 670, "Node999", NEUTRAL_REACTION);
+    giq_option(8, NAME, 672, "Node083", NEUTRAL_REACTION);
+    giq_option(5, NAME, 674, "Node999", NEUTRAL_REACTION);
 end
 
 procedure Node082
@@ -1270,18 +1270,18 @@ begin
     rm_mult_objs_from_inven(dude_obj, obj_carrying_pid_obj(dude_obj, 365), 2);
     rm_mult_objs_from_inven(dude_obj, obj_carrying_pid_obj(dude_obj, 92), 1);
     rm_mult_objs_from_inven(dude_obj, obj_carrying_pid_obj(dude_obj, 45), 1);
-    add_mult_objs_to_inven(dude_obj, create_object_sid(703, 0, 0, -1), 1);
-    gsay_reply(12, 676);
-    giq_option(-4, 12, 678, "Node999", 50);
-    giq_option(8, 12, 680, "Node083", 50);
-    giq_option(5, 12, 682, "Node999", 50);
+    add_mult_objs_to_inven(dude_obj, create_object(703, 0, 0), 1);
+    Reply(676);
+    giq_option(-4, NAME, 678, "Node999", NEUTRAL_REACTION);
+    giq_option(8, NAME, 680, "Node083", NEUTRAL_REACTION);
+    giq_option(5, NAME, 682, "Node999", NEUTRAL_REACTION);
 end
 
 procedure Node083
 begin
-    gsay_reply(12, 684);
-    giq_option(-1, 12, 686, "Node999", 50);
-    giq_option(2, 12, 688, "Node999", 50);
+    Reply(684);
+    giq_option(-1, NAME, 686, "Node999", NEUTRAL_REACTION);
+    giq_option(2, NAME, 688, "Node999", NEUTRAL_REACTION);
 end
 
 procedure Node084
@@ -1289,239 +1289,239 @@ begin
     if (local_var(14) == 0) then begin
         set_local_var(14, 3);
     end
-    gsay_reply(12, 690);
+    Reply(690);
     if ((obj_is_carrying_obj_pid(dude_obj, 365) >= 2) or (obj_is_carrying_obj_pid(dude_obj, 92) >= 1)) then begin
-        giq_option(1, 12, 692, Node078, 50);
+        giq_option(1, NAME, 692, Node078, NEUTRAL_REACTION);
     end
     else begin
-        giq_option(1, 12, 694, "Node085", 50);
+        giq_option(1, NAME, 694, "Node085", NEUTRAL_REACTION);
     end
 end
 
 procedure Node085
 begin
-    gsay_reply(12, 696);
-    giq_option(1, 14, 350, "Node999", 50);
+    Reply(696);
+    giq_option(1, SCRIPT_GENERIC, 350, "Node999", NEUTRAL_REACTION);
 end
 
 procedure Node086
 begin
-    gsay_reply(12, 698);
-    giq_option(1, 14, 350, "Node999", 50);
+    Reply(698);
+    giq_option(1, SCRIPT_GENERIC, 350, "Node999", NEUTRAL_REACTION);
 end
 
 procedure Node088
 begin
-    gsay_reply(12, 700);
-    giq_option(1, 14, 350, "Node999", 50);
+    Reply(700);
+    giq_option(1, SCRIPT_GENERIC, 350, "Node999", NEUTRAL_REACTION);
 end
 
 procedure Node089
 begin
-    gsay_reply(12, 702);
-    if (get_critter_stat(dude_obj, 4) <= 6) then begin
-        giq_option(3, 12, 704, "Node090", 50);
+    Reply(702);
+    if (dude_iq <= 6) then begin
+        giq_option(3, NAME, 704, "Node090", NEUTRAL_REACTION);
     end
-    giq_option(-2, 12, 706, "Node999", 50);
-    giq_option(7, 12, 708, Node062, 50);
+    giq_option(-2, NAME, 706, "Node999", NEUTRAL_REACTION);
+    giq_option(7, NAME, 708, Node062, NEUTRAL_REACTION);
 end
 
 procedure Node090
 begin
-    gsay_reply(12, 710);
-    giq_option(1, 12, 712, "Node999", 50);
-    giq_option(1, 12, 714, Node062, 50);
+    Reply(710);
+    giq_option(1, NAME, 712, "Node999", NEUTRAL_REACTION);
+    giq_option(1, NAME, 714, Node062, NEUTRAL_REACTION);
 end
 
 procedure Node091
 begin
-    if (global_var(650) == 1) then begin
-        set_global_var(650, 3);
+    if (global_var(GVAR_ACTEMVIL_TALK) == 1) then begin
+        set_global_var(GVAR_ACTEMVIL_TALK, 3);
     end
-    gsay_reply(12, 715);
-    giq_option(1, 14, 350, "Node999", 50);
+    Reply(715);
+    giq_option(1, SCRIPT_GENERIC, 350, "Node999", NEUTRAL_REACTION);
 end
 
 procedure Node092
 begin
-    gsay_reply(12, 740);
-    giq_option(1, 12, 742, "Node093", 50);
+    Reply(740);
+    giq_option(1, NAME, 742, "Node093", NEUTRAL_REACTION);
     if (obj_is_carrying_obj_pid(dude_obj, 757) > 0) then begin
-        giq_option(1, 12, 744, "Node094one", 50);
+        giq_option(1, NAME, 744, "Node094one", NEUTRAL_REACTION);
     end
     if (obj_is_carrying_obj_pid(dude_obj, 757) > 1) then begin
-        giq_option(1, 12, 746, "Node094all", 50);
+        giq_option(1, NAME, 746, "Node094all", NEUTRAL_REACTION);
     end
-    giq_option(1, 12, 748, "Node999", 50);
+    giq_option(1, NAME, 748, "Node999", NEUTRAL_REACTION);
 end
 
 procedure Node093
 begin
-    gsay_reply(12, 750);
+    Reply(750);
     if (obj_is_carrying_obj_pid(dude_obj, 757) > 0) then begin
-        giq_option(1, 12, 744, "Node094one", 50);
+        giq_option(1, NAME, 744, "Node094one", NEUTRAL_REACTION);
     end
     if (obj_is_carrying_obj_pid(dude_obj, 757) > 1) then begin
-        giq_option(1, 12, 746, "Node094all", 50);
+        giq_option(1, NAME, 746, "Node094all", NEUTRAL_REACTION);
     end
-    giq_option(1, 12, 748, "Node999", 50);
+    giq_option(1, NAME, 748, "Node999", NEUTRAL_REACTION);
 end
 
 procedure Node094one
 begin
-    gsay_reply(12, 752);
-    giq_option(1, 12, 754, "Node095one", 50);
+    Reply(752);
+    giq_option(1, NAME, 754, "Node095one", NEUTRAL_REACTION);
 end
 
 procedure Node094all
 begin
-    gsay_reply(12, 752);
-    giq_option(1, 12, 754, "Node095all", 50);
+    Reply(752);
+    giq_option(1, NAME, 754, "Node095all", NEUTRAL_REACTION);
 end
 
 procedure Node095one
 begin
-    add_mult_objs_to_inven(dude_obj, create_object_sid(754, 0, 0, -1), 3);
-    add_mult_objs_to_inven(dude_obj, create_object_sid(273, 0, 0, -1), 1);
+    add_mult_objs_to_inven(dude_obj, create_object(754, 0, 0), 3);
+    add_mult_objs_to_inven(dude_obj, create_object(273, 0, 0), 1);
     rm_mult_objs_from_inven(dude_obj, obj_carrying_pid_obj(dude_obj, 757), 1);
     set_local_var(10, 1);
-    gsay_reply(12, 756);
-    giq_option(1, 12, 758, "Node999", 50);
+    Reply(756);
+    giq_option(1, NAME, 758, "Node999", NEUTRAL_REACTION);
 end
 
 procedure Node095all
 begin
-    add_mult_objs_to_inven(dude_obj, create_object_sid(754, 0, 0, -1), 3 * obj_is_carrying_obj_pid(dude_obj, 757));
-    add_mult_objs_to_inven(dude_obj, create_object_sid(273, 0, 0, -1), obj_is_carrying_obj_pid(dude_obj, 757));
+    add_mult_objs_to_inven(dude_obj, create_object(754, 0, 0), 3 * obj_is_carrying_obj_pid(dude_obj, 757));
+    add_mult_objs_to_inven(dude_obj, create_object(273, 0, 0), obj_is_carrying_obj_pid(dude_obj, 757));
     rm_mult_objs_from_inven(dude_obj, obj_carrying_pid_obj(dude_obj, 757), obj_is_carrying_obj_pid(dude_obj, 757));
     set_local_var(10, 1);
-    gsay_reply(12, 756);
-    giq_option(1, 12, 758, "Node999", 50);
+    Reply(756);
+    giq_option(1, NAME, 758, "Node999", NEUTRAL_REACTION);
 end
 
 procedure Node101
 begin
-    gsay_reply(12, 947);
-    giq_option(-3, 12, 948, "Node102", 50);
-    giq_option(4, 12, 949, "Node109", 50);
-    giq_option(4, 12, 950, "Node122", 50);
+    Reply(947);
+    giq_option(-3, NAME, 948, "Node102", NEUTRAL_REACTION);
+    giq_option(4, NAME, 949, "Node109", NEUTRAL_REACTION);
+    giq_option(4, NAME, 950, "Node122", NEUTRAL_REACTION);
 end
 
 procedure Node102
 begin
-    gsay_reply(12, 951);
-    giq_option(-3, 12, 952, "Node103", 50);
+    Reply(951);
+    giq_option(-3, NAME, 952, "Node103", NEUTRAL_REACTION);
 end
 
 procedure Node103
 begin
-    gsay_reply(12, 953);
-    giq_option(-3, 12, 954, "Node104", 50);
+    Reply(953);
+    giq_option(-3, NAME, 954, "Node104", NEUTRAL_REACTION);
 end
 
 procedure Node104
 begin
-    gsay_reply(12, 955);
-    giq_option(-3, 12, 956, "Node105", 50);
+    Reply(955);
+    giq_option(-3, NAME, 956, "Node105", NEUTRAL_REACTION);
 end
 
 procedure Node105
 begin
-    gsay_reply(12, 957);
-    giq_option(-3, 12, 958, "Node106", 50);
+    Reply(957);
+    giq_option(-3, NAME, 958, "Node106", NEUTRAL_REACTION);
 end
 
 procedure Node106
 begin
-    gsay_reply(12, 959);
-    giq_option(-3, 12, 960, "Node107", 50);
+    Reply(959);
+    giq_option(-3, NAME, 960, "Node107", NEUTRAL_REACTION);
 end
 
 procedure Node107
 begin
-    gsay_reply(12, 961);
-    giq_option(-3, 12, 962, "Node108", 50);
+    Reply(961);
+    giq_option(-3, NAME, 962, "Node108", NEUTRAL_REACTION);
 end
 
 procedure Node108
 begin
-    gsay_reply(12, 963);
-    giq_option(-3, 12, 964, "Node999", 50);
+    Reply(963);
+    giq_option(-3, NAME, 964, "Node999", NEUTRAL_REACTION);
 end
 
 procedure Node109
 begin
-    gsay_reply(12, 965);
-    giq_option(4, 12, 966, "Node110", 50);
+    Reply(965);
+    giq_option(4, NAME, 966, "Node110", NEUTRAL_REACTION);
 end
 
 procedure Node110
 begin
-    gsay_reply(12, 967);
-    giq_option(4, 12, 968, "Node112", 50);
+    Reply(967);
+    giq_option(4, NAME, 968, "Node112", NEUTRAL_REACTION);
 end
 
 procedure Node112
 begin
-    gsay_reply(12, 969);
-    giq_option(4, 12, 970, "Node113", 50);
+    Reply(969);
+    giq_option(4, NAME, 970, "Node113", NEUTRAL_REACTION);
 end
 
 procedure Node113
 begin
-    gsay_reply(12, 971);
-    giq_option(4, 12, 972, "Node121", 50);
+    Reply(971);
+    giq_option(4, NAME, 972, "Node121", NEUTRAL_REACTION);
 end
 
 procedure Node114
 begin
-    gsay_reply(12, 973);
-    giq_option(4, 12, 974, "Node115", 50);
+    Reply(973);
+    giq_option(4, NAME, 974, "Node115", NEUTRAL_REACTION);
 end
 
 procedure Node115
 begin
-    gsay_reply(12, 975);
-    giq_option(4, 12, 976, "Node116", 50);
-    giq_option(4, 12, 977, "Node116", 50);
+    Reply(975);
+    giq_option(4, NAME, 976, "Node116", NEUTRAL_REACTION);
+    giq_option(4, NAME, 977, "Node116", NEUTRAL_REACTION);
 end
 
 procedure Node116
 begin
-    gsay_reply(12, 978);
-    giq_option(4, 12, 979, "Node117", 50);
-    giq_option(4, 12, 980, "Node120", 50);
+    Reply(978);
+    giq_option(4, NAME, 979, "Node117", NEUTRAL_REACTION);
+    giq_option(4, NAME, 980, "Node120", NEUTRAL_REACTION);
 end
 
 procedure Node117
 begin
-    gsay_reply(12, 981);
-    giq_option(4, 12, 982, "Node118", 50);
+    Reply(981);
+    giq_option(4, NAME, 982, "Node118", NEUTRAL_REACTION);
 end
 
 procedure Node118
 begin
-    gsay_reply(12, 983);
-    giq_option(4, 12, 984, "Node999", 50);
-    giq_option(4, 12, 985, "Node999", 50);
+    Reply(983);
+    giq_option(4, NAME, 984, "Node999", NEUTRAL_REACTION);
+    giq_option(4, NAME, 985, "Node999", NEUTRAL_REACTION);
 end
 
 procedure Node120
 begin
-    gsay_reply(12, 986);
-    giq_option(4, 12, 987, Node117, 50);
+    Reply(986);
+    giq_option(4, NAME, 987, Node117, NEUTRAL_REACTION);
 end
 
 procedure Node121
 begin
-    gsay_reply(12, 988);
-    giq_option(4, 12, 989, Node114, 50);
+    Reply(988);
+    giq_option(4, NAME, 989, Node114, NEUTRAL_REACTION);
 end
 
 procedure Node122
 begin
-    gsay_reply(12, 990);
-    giq_option(4, 12, 991, Node109, 50);
+    Reply(990);
+    giq_option(4, NAME, 991, Node109, NEUTRAL_REACTION);
 end
 
 procedure Node999
@@ -1535,7 +1535,7 @@ end
 
 procedure talk_p_proc
 begin
-    if (cur_map_index == 127) then begin
+    if (CUR_MAP_ARDEAD) then begin
         set_local_var(4, 1);
         start_gdialog(12, self_obj, 4, 12, 16);
         gsay_start;
@@ -1545,7 +1545,7 @@ begin
         kill_critter(self_obj, 49);
     end
     else begin
-        if ((global_var(0) >= 0) and (local_var(1) != -2) and (metarule(17, 11) == 2)) then begin
+        if ((global_var(GVAR_PLAYER_REPUTATION) >= 0) and (local_var(1) != -2) and (town_known(11) == MARK_STATE_VISITED)) then begin
             start_gdialog(12, self_obj, 4, 10, 19);
             gsay_start;
             call Node101();
@@ -1553,7 +1553,7 @@ begin
             end_dialogue;
         end
         else begin
-            if (global_var(0) <= (-1 * 50)) then begin
+            if (global_var(GVAR_PLAYER_REPUTATION) <= (-1 * 50)) then begin
                 set_local_var(0, 0);
                 start_gdialog(12, self_obj, 4, 10, 19);
                 gsay_start;
@@ -1562,7 +1562,7 @@ begin
                 end_dialogue;
             end
             else begin
-                if (global_var(640) == 1) then begin
+                if (global_var(GVAR_ARTEMPLE_PROVE) == 1) then begin
                     set_local_var(0, 50);
                     start_gdialog(12, self_obj, 4, 10, 19);
                     gsay_start;
@@ -1571,9 +1571,9 @@ begin
                     end_dialogue;
                 end
                 else begin
-                    if (global_var(639) == 1) then begin
+                    if (global_var(GVAR_ARROYO_MEMBER) == 1) then begin
                         if (local_var(16) != 1) then begin
-                            if ((get_critter_stat(dude_obj, 4) >= 5) and (get_critter_stat(dude_obj, 1) >= 6) and ((game_time / (10 * 60 * 60 * 24)) >= 2)) then begin
+                            if ((dude_iq >= 5) and (dude_perception >= 6) and ((game_time / (10 * 60 * 60 * 24)) >= 2)) then begin
                                 set_local_var(0, 50);
                                 start_gdialog(12, self_obj, 4, 10, 19);
                                 gsay_start;
@@ -1582,7 +1582,7 @@ begin
                                 end_dialogue;
                             end
                             else begin
-                                if (global_var(9) <= 0) then begin
+                                if (global_var(GVAR_KILL_EVIL_PLANTS) <= 0) then begin
                                     set_local_var(0, 50);
                                     start_gdialog(12, self_obj, 4, 10, 19);
                                     gsay_start;
@@ -1591,7 +1591,7 @@ begin
                                     end_dialogue;
                                 end
                                 else begin
-                                    if ((global_var(9) > 0) and (global_var(9) < 4)) then begin
+                                    if ((global_var(GVAR_KILL_EVIL_PLANTS) > 0) and (global_var(GVAR_KILL_EVIL_PLANTS) < 4)) then begin
                                         set_local_var(0, 50);
                                         start_gdialog(12, self_obj, 4, 10, 19);
                                         gsay_start;
@@ -1620,7 +1620,7 @@ begin
                         end
                     end
                     else begin
-                        if (global_var(650) == 1) then begin
+                        if (global_var(GVAR_ACTEMVIL_TALK) == 1) then begin
                             if (get_month == 8) then begin
                                 if (get_day < 5) then begin
                                     set_local_var(0, 50);
@@ -1669,7 +1669,7 @@ begin
                             end
                         end
                         else begin
-                            if (global_var(650) == 2) then begin
+                            if (global_var(GVAR_ACTEMVIL_TALK) == 2) then begin
                                 set_local_var(0, 50);
                                 start_gdialog(12, self_obj, 4, 10, 19);
                                 gsay_start;
@@ -1678,7 +1678,7 @@ begin
                                 end_dialogue;
                             end
                             else begin
-                                if (global_var(650) == 3) then begin
+                                if (global_var(GVAR_ACTEMVIL_TALK) == 3) then begin
                                     set_local_var(0, 50);
                                     start_gdialog(12, self_obj, 4, 10, 19);
                                     gsay_start;
@@ -1706,4 +1706,3 @@ end
 procedure start
 begin
 end
-
