diff --git a/Mods/MegaMod/Original/AICONSOL.ssl b/Mods/MegaMod/ReDefine/AICONSOL.ssl
index 03c03a9..40b7356 100644
--- a/Mods/MegaMod/Original/AICONSOL.ssl
+++ b/Mods/MegaMod/ReDefine/AICONSOL.ssl
@@ -159,13 +159,13 @@ end
 procedure look_at_p_proc
 begin
     script_overrides;
-    display_msg(message_str(1619, 100));
+    display_mstr(100);
 end
 
 procedure description_p_proc
 begin
     script_overrides;
-    display_msg(message_str(1619, 100));
+    display_mstr(100);
 end
 
 procedure use_p_proc
@@ -209,7 +209,7 @@ end
 
 procedure set_lg5
 begin
-    if ((get_critter_stat(dude_obj, 6) > 7) and (random(1, 4) == 2)) then begin
+    if ((dude_luck > 7) and (random(1, 4) == 2)) then begin
         login := 4;
     end
     call Node002();
@@ -217,7 +217,7 @@ end
 
 procedure xak_lg
 begin
-    if (has_skill(dude_obj, 12) > 135) then begin
+    if (has_skill(dude_obj, SKILL_SCIENCE) > 135) then begin
         call access_allowed();
     end
     else begin
@@ -252,7 +252,7 @@ end
 
 procedure set_ps5
 begin
-    if ((get_critter_stat(dude_obj, 6) > 7) and (random(1, 6) == 2)) then begin
+    if ((dude_luck > 7) and (random(1, 6) == 2)) then begin
         call access_allowed();
     end
     else begin
@@ -262,256 +262,255 @@ end
 
 procedure access_allowed
 begin
-    if (global_var(1099) == 3) then begin
+    if (global_var(GVAR_FORMER_P_TRIBE_LOCATION_KNOWN) == 3) then begin
         give_exp_points(1500);
-        display_msg(message_str(14, 100) + 1500 + message_str(14, 101));
-        set_global_var(1099, 4);
+        display_msg(g_mstr(100) + 1500 + g_mstr(101));
+        set_global_var(GVAR_FORMER_P_TRIBE_LOCATION_KNOWN, 4);
     end
     set_local_var(1, 1);
-    gsay_reply(1619, 400);
-    gsay_option(1619, 410, Node004, -1);
-    gsay_option(1619, 411, Node005, -1);
-    gsay_option(1619, 412, Node006, -1);
-    gsay_option(1619, 413, Node007, -1);
-    gsay_option(1619, 414, Node999, -1);
+    Reply(400);
+    giq_option(1, NAME, 410, Node004, -1);
+    giq_option(1, NAME, 411, Node005, -1);
+    giq_option(1, NAME, 412, Node006, -1);
+    giq_option(1, NAME, 413, Node007, -1);
+    giq_option(1, NAME, 414, Node999, -1);
 end
 
 procedure Node001
 begin
-    gsay_reply(1619, 200);
-    gsay_option(1619, 210, set_lg1, -1);
-    gsay_option(1619, 211, set_lg2, -1);
-    gsay_option(1619, 212, set_lg3, -1);
-    if (global_var(1099) > 2) then begin
-        gsay_option(1619, 213, set_lg4, -1);
+    Reply(200);
+    giq_option(1, NAME, 210, set_lg1, -1);
+    giq_option(1, NAME, 211, set_lg2, -1);
+    giq_option(1, NAME, 212, set_lg3, -1);
+    if (global_var(GVAR_FORMER_P_TRIBE_LOCATION_KNOWN) > 2) then begin
+        giq_option(1, NAME, 213, set_lg4, -1);
     end
-    gsay_option(1619, 214, set_lg5, -1);
-    gsay_option(1619, 215, xak_lg, -1);
-    gsay_option(1619, 216, Node999, -1);
+    giq_option(1, NAME, 214, set_lg5, -1);
+    giq_option(1, NAME, 215, xak_lg, -1);
+    giq_option(1, NAME, 216, Node999, -1);
 end
 
 procedure Node002
 begin
-    gsay_reply(1619, 300);
-    gsay_option(1619, 310, set_ps1, -1);
-    gsay_option(1619, 311, set_ps2, -1);
-    gsay_option(1619, 312, set_ps3, -1);
-    if (global_var(1099) > 2) then begin
-        gsay_option(1619, 313, access_allowed, -1);
+    Reply(300);
+    giq_option(1, NAME, 310, set_ps1, -1);
+    giq_option(1, NAME, 311, set_ps2, -1);
+    giq_option(1, NAME, 312, set_ps3, -1);
+    if (global_var(GVAR_FORMER_P_TRIBE_LOCATION_KNOWN) > 2) then begin
+        giq_option(1, NAME, 313, access_allowed, -1);
     end
-    gsay_option(1619, 314, set_ps5, -1);
-    gsay_option(1619, 316, Node999, -1);
+    giq_option(1, NAME, 314, set_ps5, -1);
+    giq_option(1, NAME, 316, Node999, -1);
 end
 
 procedure Node003
 begin
-    gsay_reply(1619, 1000);
-    gsay_option(1619, 1203, Node001, -1);
-    gsay_option(1619, 414, Node999, -1);
+    Reply(1000);
+    giq_option(1, NAME, 1203, Node001, -1);
+    giq_option(1, NAME, 414, Node999, -1);
 end
 
 procedure Node004
 begin
-    gsay_reply(1619, 1200);
-    gsay_option(1619, 1208, vault_8, -1);
-    gsay_option(1619, 1212, vault_12, -1);
-    gsay_option(1619, 1213, vault_13, -1);
-    gsay_option(1619, 1215, vault_15, -1);
-    gsay_option(1619, 1227, vault_27, -1);
-    gsay_option(1619, 1229, vault_29, -1);
-    gsay_option(1619, 1201, Node008, -1);
-    gsay_option(1619, 1203, access_allowed, -1);
+    Reply(1200);
+    giq_option(1, NAME, 1208, vault_8, -1);
+    giq_option(1, NAME, 1212, vault_12, -1);
+    giq_option(1, NAME, 1213, vault_13, -1);
+    giq_option(1, NAME, 1215, vault_15, -1);
+    giq_option(1, NAME, 1227, vault_27, -1);
+    giq_option(1, NAME, 1229, vault_29, -1);
+    giq_option(1, NAME, 1201, Node008, -1);
+    giq_option(1, NAME, 1203, access_allowed, -1);
 end
 
 procedure Node005
 begin
-    gsay_reply(1619, 1100);
-    gsay_option(1619, 1203, access_allowed, -1);
-    gsay_option(1619, 1204, Node999, -1);
+    Reply(1100);
+    giq_option(1, NAME, 1203, access_allowed, -1);
+    giq_option(1, NAME, 1204, Node999, -1);
 end
 
 procedure Node006
 begin
-    gsay_reply(1619, 1105);
-    gsay_option(1619, 1203, access_allowed, -1);
-    gsay_option(1619, 1204, Node999, -1);
+    Reply(1105);
+    giq_option(1, NAME, 1203, access_allowed, -1);
+    giq_option(1, NAME, 1204, Node999, -1);
 end
 
 procedure Node007
 begin
-    gsay_reply(1619, 2000);
-    gsay_option(1619, 1203, access_allowed, -1);
-    gsay_option(1619, 1204, Node999, -1);
+    Reply(2000);
+    giq_option(1, NAME, 1203, access_allowed, -1);
+    giq_option(1, NAME, 1204, Node999, -1);
 end
 
 procedure Node008
 begin
-    gsay_reply(1619, 1200);
-    gsay_option(1619, 1234, vault_34, -1);
-    gsay_option(1619, 1236, vault_36, -1);
-    gsay_option(1619, 1242, vault_42, -1);
-    gsay_option(1619, 1253, vault_53, -1);
-    gsay_option(1619, 1255, vault_55, -1);
-    gsay_option(1619, 1202, Node004, -1);
-    gsay_option(1619, 1201, Node009, -1);
-    gsay_option(1619, 1203, access_allowed, -1);
+    Reply(1200);
+    giq_option(1, NAME, 1234, vault_34, -1);
+    giq_option(1, NAME, 1236, vault_36, -1);
+    giq_option(1, NAME, 1242, vault_42, -1);
+    giq_option(1, NAME, 1253, vault_53, -1);
+    giq_option(1, NAME, 1255, vault_55, -1);
+    giq_option(1, NAME, 1202, Node004, -1);
+    giq_option(1, NAME, 1201, Node009, -1);
+    giq_option(1, NAME, 1203, access_allowed, -1);
 end
 
 procedure Node009
 begin
-    gsay_reply(1619, 1200);
-    gsay_option(1619, 1256, vault_56, -1);
-    gsay_option(1619, 1268, vault_68, -1);
-    gsay_option(1619, 1269, vault_69, -1);
-    gsay_option(1619, 1270, vault_70, -1);
-    gsay_option(1619, 1306, vault_106, -1);
-    gsay_option(1619, 1202, Node008, -1);
-    gsay_option(1619, 1203, access_allowed, -1);
+    Reply(1200);
+    giq_option(1, NAME, 1256, vault_56, -1);
+    giq_option(1, NAME, 1268, vault_68, -1);
+    giq_option(1, NAME, 1269, vault_69, -1);
+    giq_option(1, NAME, 1270, vault_70, -1);
+    giq_option(1, NAME, 1306, vault_106, -1);
+    giq_option(1, NAME, 1202, Node008, -1);
+    giq_option(1, NAME, 1203, access_allowed, -1);
 end
 
 procedure vault_8
 begin
-    gsay_reply(1619, 1308);
-    gsay_option(1619, 1201, vault_12, -1);
-    gsay_option(1619, 1203, access_allowed, -1);
-    gsay_option(1619, 1204, Node999, -1);
+    Reply(1308);
+    giq_option(1, NAME, 1201, vault_12, -1);
+    giq_option(1, NAME, 1203, access_allowed, -1);
+    giq_option(1, NAME, 1204, Node999, -1);
 end
 
 procedure vault_12
 begin
-    gsay_reply(1619, 1312);
-    gsay_option(1619, 1202, vault_8, -1);
-    gsay_option(1619, 1201, vault_13, -1);
-    gsay_option(1619, 1203, access_allowed, -1);
-    gsay_option(1619, 1204, Node999, -1);
+    Reply(1312);
+    giq_option(1, NAME, 1202, vault_8, -1);
+    giq_option(1, NAME, 1201, vault_13, -1);
+    giq_option(1, NAME, 1203, access_allowed, -1);
+    giq_option(1, NAME, 1204, Node999, -1);
 end
 
 procedure vault_13
 begin
-    gsay_reply(1619, 1313);
-    gsay_option(1619, 1202, vault_12, -1);
-    gsay_option(1619, 1201, vault_15, -1);
-    gsay_option(1619, 1203, access_allowed, -1);
-    gsay_option(1619, 1204, Node999, -1);
+    Reply(1313);
+    giq_option(1, NAME, 1202, vault_12, -1);
+    giq_option(1, NAME, 1201, vault_15, -1);
+    giq_option(1, NAME, 1203, access_allowed, -1);
+    giq_option(1, NAME, 1204, Node999, -1);
 end
 
 procedure vault_15
 begin
-    gsay_reply(1619, 1315);
-    gsay_option(1619, 1202, vault_13, -1);
-    gsay_option(1619, 1201, vault_27, -1);
-    gsay_option(1619, 1203, access_allowed, -1);
-    gsay_option(1619, 1204, Node999, -1);
+    Reply(1315);
+    giq_option(1, NAME, 1202, vault_13, -1);
+    giq_option(1, NAME, 1201, vault_27, -1);
+    giq_option(1, NAME, 1203, access_allowed, -1);
+    giq_option(1, NAME, 1204, Node999, -1);
 end
 
 procedure vault_27
 begin
-    gsay_reply(1619, 1327);
-    gsay_option(1619, 1202, vault_15, -1);
-    gsay_option(1619, 1201, vault_29, -1);
-    gsay_option(1619, 1203, access_allowed, -1);
-    gsay_option(1619, 1204, Node999, -1);
+    Reply(1327);
+    giq_option(1, NAME, 1202, vault_15, -1);
+    giq_option(1, NAME, 1201, vault_29, -1);
+    giq_option(1, NAME, 1203, access_allowed, -1);
+    giq_option(1, NAME, 1204, Node999, -1);
 end
 
 procedure vault_29
 begin
-    gsay_reply(1619, 1329);
-    gsay_option(1619, 1202, vault_27, -1);
-    gsay_option(1619, 1201, vault_34, -1);
-    gsay_option(1619, 1203, access_allowed, -1);
-    gsay_option(1619, 1204, Node999, -1);
+    Reply(1329);
+    giq_option(1, NAME, 1202, vault_27, -1);
+    giq_option(1, NAME, 1201, vault_34, -1);
+    giq_option(1, NAME, 1203, access_allowed, -1);
+    giq_option(1, NAME, 1204, Node999, -1);
 end
 
 procedure vault_34
 begin
-    gsay_reply(1619, 1334);
-    gsay_option(1619, 1202, vault_29, -1);
-    gsay_option(1619, 1201, vault_36, -1);
-    gsay_option(1619, 1203, access_allowed, -1);
-    gsay_option(1619, 1204, Node999, -1);
+    Reply(1334);
+    giq_option(1, NAME, 1202, vault_29, -1);
+    giq_option(1, NAME, 1201, vault_36, -1);
+    giq_option(1, NAME, 1203, access_allowed, -1);
+    giq_option(1, NAME, 1204, Node999, -1);
 end
 
 procedure vault_36
 begin
-    gsay_reply(1619, 1336);
-    gsay_option(1619, 1202, vault_34, -1);
-    gsay_option(1619, 1201, vault_42, -1);
-    gsay_option(1619, 1203, access_allowed, -1);
-    gsay_option(1619, 1204, Node999, -1);
+    Reply(1336);
+    giq_option(1, NAME, 1202, vault_34, -1);
+    giq_option(1, NAME, 1201, vault_42, -1);
+    giq_option(1, NAME, 1203, access_allowed, -1);
+    giq_option(1, NAME, 1204, Node999, -1);
 end
 
 procedure vault_42
 begin
-    gsay_reply(1619, 1342);
-    gsay_option(1619, 1202, vault_36, -1);
-    gsay_option(1619, 1201, vault_53, -1);
-    gsay_option(1619, 1203, access_allowed, -1);
-    gsay_option(1619, 1204, Node999, -1);
+    Reply(1342);
+    giq_option(1, NAME, 1202, vault_36, -1);
+    giq_option(1, NAME, 1201, vault_53, -1);
+    giq_option(1, NAME, 1203, access_allowed, -1);
+    giq_option(1, NAME, 1204, Node999, -1);
 end
 
 procedure vault_53
 begin
-    gsay_reply(1619, 1353);
-    gsay_option(1619, 1202, vault_42, -1);
-    gsay_option(1619, 1201, vault_55, -1);
-    gsay_option(1619, 1203, access_allowed, -1);
-    gsay_option(1619, 1204, Node999, -1);
+    Reply(1353);
+    giq_option(1, NAME, 1202, vault_42, -1);
+    giq_option(1, NAME, 1201, vault_55, -1);
+    giq_option(1, NAME, 1203, access_allowed, -1);
+    giq_option(1, NAME, 1204, Node999, -1);
 end
 
 procedure vault_55
 begin
-    gsay_reply(1619, 1355);
-    gsay_option(1619, 1202, vault_53, -1);
-    gsay_option(1619, 1201, vault_56, -1);
-    gsay_option(1619, 1203, access_allowed, -1);
-    gsay_option(1619, 1204, Node999, -1);
+    Reply(1355);
+    giq_option(1, NAME, 1202, vault_53, -1);
+    giq_option(1, NAME, 1201, vault_56, -1);
+    giq_option(1, NAME, 1203, access_allowed, -1);
+    giq_option(1, NAME, 1204, Node999, -1);
 end
 
 procedure vault_56
 begin
-    gsay_reply(1619, 1356);
-    gsay_option(1619, 1202, vault_55, -1);
-    gsay_option(1619, 1201, vault_68, -1);
-    gsay_option(1619, 1203, access_allowed, -1);
-    gsay_option(1619, 1204, Node999, -1);
+    Reply(1356);
+    giq_option(1, NAME, 1202, vault_55, -1);
+    giq_option(1, NAME, 1201, vault_68, -1);
+    giq_option(1, NAME, 1203, access_allowed, -1);
+    giq_option(1, NAME, 1204, Node999, -1);
 end
 
 procedure vault_68
 begin
-    gsay_reply(1619, 1368);
-    gsay_option(1619, 1202, vault_56, -1);
-    gsay_option(1619, 1201, vault_69, -1);
-    gsay_option(1619, 1203, access_allowed, -1);
-    gsay_option(1619, 1204, Node999, -1);
+    Reply(1368);
+    giq_option(1, NAME, 1202, vault_56, -1);
+    giq_option(1, NAME, 1201, vault_69, -1);
+    giq_option(1, NAME, 1203, access_allowed, -1);
+    giq_option(1, NAME, 1204, Node999, -1);
 end
 
 procedure vault_69
 begin
-    gsay_reply(1619, 1369);
-    gsay_option(1619, 1202, vault_68, -1);
-    gsay_option(1619, 1201, vault_70, -1);
-    gsay_option(1619, 1203, access_allowed, -1);
-    gsay_option(1619, 1204, Node999, -1);
+    Reply(1369);
+    giq_option(1, NAME, 1202, vault_68, -1);
+    giq_option(1, NAME, 1201, vault_70, -1);
+    giq_option(1, NAME, 1203, access_allowed, -1);
+    giq_option(1, NAME, 1204, Node999, -1);
 end
 
 procedure vault_70
 begin
-    gsay_reply(1619, 1370);
-    gsay_option(1619, 1202, vault_69, -1);
-    gsay_option(1619, 1201, vault_106, -1);
-    gsay_option(1619, 1203, access_allowed, -1);
-    gsay_option(1619, 1204, Node999, -1);
+    Reply(1370);
+    giq_option(1, NAME, 1202, vault_69, -1);
+    giq_option(1, NAME, 1201, vault_106, -1);
+    giq_option(1, NAME, 1203, access_allowed, -1);
+    giq_option(1, NAME, 1204, Node999, -1);
 end
 
 procedure vault_106
 begin
-    gsay_reply(1619, 1406);
-    gsay_option(1619, 1202, vault_70, -1);
-    gsay_option(1619, 1203, access_allowed, -1);
-    gsay_option(1619, 1204, Node999, -1);
+    Reply(1406);
+    giq_option(1, NAME, 1202, vault_70, -1);
+    giq_option(1, NAME, 1203, access_allowed, -1);
+    giq_option(1, NAME, 1204, Node999, -1);
 end
 
 procedure Node999
 begin
 end
-
