diff --git a/Mods/MegaMod/Original/AMONK4.ssl b/Mods/MegaMod/ReDefine/AMONK4.ssl
index f3b251c..08ac4f9 100644
--- a/Mods/MegaMod/Original/AMONK4.ssl
+++ b/Mods/MegaMod/ReDefine/AMONK4.ssl
@@ -37,69 +37,69 @@ end
 procedure description_p_proc
 begin
     script_overrides;
-    display_msg(message_str(2276, 101));
+    display_mstr(101);
 end
 
 procedure destroy_p_proc
 begin
     if (source_obj == dude_obj) then begin
-        set_global_var(5, global_var(5) + 1);
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(4) > (3 * global_var(5))) or (global_var(2) == 1)) and (global_var(1) == 0)) then begin
-            set_global_var(2, 1);
-            set_global_var(3, 0);
+        set_global_var(GVAR_GOOD_MONSTER, global_var(GVAR_GOOD_MONSTER) + 1);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(4) > (3 * global_var(GVAR_GOOD_MONSTER))) or (global_var(GVAR_CHAMPION_REPUTATION) == 1)) and (global_var(GVAR_CHILDKILLER_REPUTATION) == 0)) then begin
+            set_global_var(GVAR_CHAMPION_REPUTATION, 1);
+            set_global_var(GVAR_BERSERKER_REPUTATION, 0);
         end
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(5) > (2 * global_var(4))) or (global_var(3) == 1))) then begin
-            set_global_var(3, 1);
-            set_global_var(2, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
+            set_global_var(GVAR_BERSERKER_REPUTATION, 1);
+            set_global_var(GVAR_CHAMPION_REPUTATION, 0);
         end
-        set_global_var(0, global_var(0) + -10);
+        set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + -10);
         debug_msg("Player gains " + -10 + " Karma Points.");
-        set_global_var(37, 0);
-        set_global_var(38, 0);
-        set_global_var(39, 0);
-        set_global_var(40, 0);
-        set_global_var(41, 0);
-        set_global_var(42, 0);
-        set_global_var(43, 0);
-        set_global_var(44, 0);
-        set_global_var(45, 0);
-        tmp_gen_rep := global_var(0);
-        if (has_trait(0, dude_obj, 95)) then begin
-            tmp_gen_rep := global_var(0) * 2;
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+        set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+        set_global_var(GVAR_KARMA_DEFENDER, 0);
+        set_global_var(GVAR_KARMA_WANDERER, 0);
+        set_global_var(GVAR_KARMA_BETRAYER, 0);
+        set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+        set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+        tmp_gen_rep := global_var(GVAR_PLAYER_REPUTATION);
+        if (has_trait(TRAIT_PERK, dude_obj, PERK_karma_beacon_perk)) then begin
+            tmp_gen_rep := global_var(GVAR_PLAYER_REPUTATION) * 2;
         end
         if (tmp_gen_rep >= 1000) then begin
-            set_global_var(37, 1);
+            set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
         end
         else begin
             if (tmp_gen_rep >= 750) then begin
-                set_global_var(38, 1);
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
             end
             else begin
                 if (tmp_gen_rep >= 500) then begin
-                    set_global_var(39, 1);
+                    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                 end
                 else begin
                     if (tmp_gen_rep >= 250) then begin
-                        set_global_var(40, 1);
+                        set_global_var(GVAR_KARMA_DEFENDER, 1);
                     end
                     else begin
                         if (tmp_gen_rep > -250) then begin
-                            set_global_var(41, 1);
+                            set_global_var(GVAR_KARMA_WANDERER, 1);
                         end
                         else begin
                             if (tmp_gen_rep > -500) then begin
-                                set_global_var(42, 1);
+                                set_global_var(GVAR_KARMA_BETRAYER, 1);
                             end
                             else begin
                                 if (tmp_gen_rep > -750) then begin
-                                    set_global_var(43, 1);
+                                    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                 end
                                 else begin
                                     if (tmp_gen_rep > -1000) then begin
-                                        set_global_var(44, 1);
+                                        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                     end
                                     else begin
-                                        set_global_var(45, 1);
+                                        set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                     end
                                 end
                             end
@@ -108,7 +108,7 @@ begin
                 end
             end
         end
-        set_global_var(63, global_var(63) + -5);
+        set_global_var(GVAR_TOWN_REP_ABBEY, global_var(GVAR_TOWN_REP_ABBEY) + -5);
         debug_msg("Added " + -5 + " to Town Rep");
     end
 end
@@ -116,7 +116,7 @@ end
 procedure map_enter_p_proc
 begin
     Only_Once := 0;
-    critter_add_trait(self_obj, 1, 5, 17);
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_STORE_OWNER);
     if (local_var(7) == 0) then begin
         set_local_var(7, 1);
         call Make_Ainven();
@@ -127,7 +127,7 @@ end
 procedure look_at_p_proc
 begin
     script_overrides;
-    display_msg(message_str(2276, 100));
+    display_mstr(100);
 end
 
 procedure pickup_p_proc
@@ -135,22 +135,22 @@ begin
     variable LVar0 := 0;
     if (source_obj == dude_obj) then begin
         if (local_var(8) == 1) then begin
-            set_global_var(63, -30);
-            attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+            set_global_var(GVAR_TOWN_REP_ABBEY, -30);
+            attack(dude_obj);
         end
         else begin
-            float_msg(self_obj, message_str(2276, 500), 2);
+            float_msg(self_obj, mstr(500), FLOAT_MSG_RED);
             set_local_var(8, 1);
-            LVar0 := create_object_sid(236, 0, 0, -1);
+            LVar0 := create_object(236, 0, 0);
             add_obj_to_inven(self_obj, LVar0);
-            wield_obj_critter(self_obj, LVar0);
+            wield_obj(LVar0);
         end
     end
 end
 
 procedure timed_event_p_proc
 begin
-    if ((fixed_param == 1) and not(combat_is_initialized) and obj_can_see_obj(self_obj, dude_obj) and (tile_distance(tile_num(self_obj), tile_num(dude_obj)) <= 4)) then begin
+    if ((fixed_param == 1) and not(combat_is_initialized) and obj_can_see_obj(self_obj, dude_obj) and (tile_distance(tile_num(self_obj), dude_tile) <= 4)) then begin
         call Node001();
     end
     add_timer_event(self_obj, random(5, 10) * 10, 1);
@@ -158,7 +158,7 @@ end
 
 procedure talk_p_proc
 begin
-    if (not(get_critter_stat(dude_obj, 4) <= 3)) then begin
+    if (not(dude_is_stupid)) then begin
         start_gdialog(2276, self_obj, 4, -1, -1);
         gsay_start;
         call Node001();
@@ -166,7 +166,7 @@ begin
         end_dialogue;
     end
     else begin
-        float_msg(self_obj, message_str(2276, 203), 7);
+        float_msg(self_obj, mstr(203), FLOAT_MSG_LIGHT_RED);
     end
 end
 
@@ -174,81 +174,81 @@ procedure Make_Ainven
 begin
     variable LVar0 := 0;
     if (random(1, 20) <= 15) then begin
-        LVar0 := create_object_sid(71, 0, 0, -1);
+        LVar0 := create_object(71, 0, 0);
         add_mult_objs_to_inven(self_obj, LVar0, random(2, 3));
     end
     if (random(1, 20) <= 10) then begin
-        LVar0 := create_object_sid(379, 0, 0, -1);
+        LVar0 := create_object(379, 0, 0);
         add_mult_objs_to_inven(self_obj, LVar0, 1);
     end
     if (random(1, 20) <= 3) then begin
-        LVar0 := create_object_sid(40, 0, 0, -1);
+        LVar0 := create_object(40, 0, 0);
         add_mult_objs_to_inven(self_obj, LVar0, random(1, 2));
     end
     if (random(1, 20) <= 5) then begin
-        LVar0 := create_object_sid(300, 0, 0, -1);
+        LVar0 := create_object(300, 0, 0);
         add_mult_objs_to_inven(self_obj, LVar0, random(1, 2));
     end
     if (random(1, 20) <= 7) then begin
-        LVar0 := create_object_sid(121, 0, 0, -1);
+        LVar0 := create_object(121, 0, 0);
         add_mult_objs_to_inven(self_obj, LVar0, random(1, 2));
     end
     if (random(1, 20) <= 1) then begin
-        LVar0 := create_object_sid(1, 0, 0, -1);
+        LVar0 := create_object(1, 0, 0);
         add_mult_objs_to_inven(self_obj, LVar0, 1);
     end
     if (random(1, 20) <= 2) then begin
-        LVar0 := create_object_sid(122, 0, 0, -1);
+        LVar0 := create_object(122, 0, 0);
         add_mult_objs_to_inven(self_obj, LVar0, 1);
     end
     if (random(1, 20) <= 3) then begin
-        LVar0 := create_object_sid(360, 0, 0, -1);
+        LVar0 := create_object(360, 0, 0);
         add_mult_objs_to_inven(self_obj, LVar0, random(1, 2));
     end
     if (random(1, 20) <= 5) then begin
-        LVar0 := create_object_sid(159, 0, 0, -1);
+        LVar0 := create_object(159, 0, 0);
         add_mult_objs_to_inven(self_obj, LVar0, random(1, 2));
     end
 end
 
 procedure fNode
 begin
-    float_msg(self_obj, message_str(2276, random(200, 202)), 8);
+    float_msg(self_obj, mstr(random(200, 202)), FLOAT_MSG_YELLOW);
 end
 
 procedure Node001
 begin
-    gsay_reply(2276, 300);
-    giq_option(4, 2276, 301, Node002, 50);
-    giq_option(4, 2276, 302, NodeBarter, 50);
-    giq_option(4, 2276, 303, Node999, 50);
+    Reply(300);
+    giq_option(4, NAME, 301, Node002, NEUTRAL_REACTION);
+    giq_option(4, NAME, 302, NodeBarter, NEUTRAL_REACTION);
+    giq_option(4, NAME, 303, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node002
 begin
-    gsay_reply(2276, 400);
-    giq_option(4, 2276, 401, Node005, 50);
-    giq_option(4, 2276, 402, Node006, 50);
+    Reply(400);
+    giq_option(4, NAME, 401, Node005, NEUTRAL_REACTION);
+    giq_option(4, NAME, 402, Node006, NEUTRAL_REACTION);
 end
 
 procedure NodeBarter
 begin
-    gdialog_mod_barter(0);
+    gdialog_barter;
     call Node001();
 end
 
 procedure Node005
 begin
-    gsay_reply(2276, 403);
-    giq_option(4, 2276, 402, Node006, 50);
-    giq_option(4, 2276, 405, Node001, 50);
+    Reply(403);
+    giq_option(4, NAME, 402, Node006, NEUTRAL_REACTION);
+    giq_option(4, NAME, 405, Node001, NEUTRAL_REACTION);
 end
 
 procedure Node006
 begin
-    gsay_reply(2276, 404);
-    giq_option(4, 2276, 401, Node005, 50);
-    giq_option(4, 2276, 406, Node001, 50);
+    Reply(404);
+    giq_option(4, NAME, 401, Node005, NEUTRAL_REACTION);
+    giq_option(4, NAME, 406, Node001, NEUTRAL_REACTION);
 end
 
 procedure Node999
@@ -257,8 +257,8 @@ end
 
 procedure critter_p_proc
 begin
-    if ((global_var(63) <= -30) and obj_can_see_obj(self_obj, dude_obj)) then begin
-        attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+    if ((global_var(GVAR_TOWN_REP_ABBEY) <= -30) and obj_can_see_obj(self_obj, dude_obj)) then begin
+        attack(dude_obj);
     end
 end
 
@@ -267,9 +267,8 @@ begin
     if (source_obj == dude_obj) then begin
         set_local_var(0, local_var(0) bwor 8);
         set_local_var(0, local_var(0) bwor 2);
-        if (global_var(63) > -30) then begin
-            set_global_var(63, -30);
+        if (global_var(GVAR_TOWN_REP_ABBEY) > -30) then begin
+            set_global_var(GVAR_TOWN_REP_ABBEY, -30);
         end
     end
 end
-
