diff --git a/Mods/MegaMod/Original/ARBRIDGE.ssl b/Mods/MegaMod/ReDefine/ARBRIDGE.ssl
index 0bb54c4..6b6c5f3 100644
--- a/Mods/MegaMod/Original/ARBRIDGE.ssl
+++ b/Mods/MegaMod/ReDefine/ARBRIDGE.ssl
@@ -141,21 +141,21 @@ end
 
 procedure map_enter_p_proc
 begin
-    if (tile_num(dude_obj) == 26317) then begin
-        if ((global_var(640) == 0) and (days_since_visited >= 5) and (global_var(683) != 0)) then begin
-            set_global_var(640, 1);
+    if (dude_tile == 26317) then begin
+        if ((global_var(GVAR_ARTEMPLE_PROVE) == 0) and (days_since_visited >= 5) and (global_var(GVAR_DUDE_SHAMAN) != 0)) then begin
+            set_global_var(GVAR_ARTEMPLE_PROVE, 1);
         end
     end
-    if (global_var(7) == 1) then begin
-        set_global_var(47, -50);
+    if (global_var(GVAR_ENEMY_ARROYO) == 1) then begin
+        set_global_var(GVAR_TOWN_REP_ARROYO, -50);
     end
     else begin
-        if (global_var(644) == 20) then begin
-            set_global_var(47, 50);
+        if (global_var(GVAR_AHELDER_Q1_AGRO) == 20) then begin
+            set_global_var(GVAR_TOWN_REP_ARROYO, 50);
         end
     end
-    if (metarule(14, 0)) then begin
-        display_msg(message_str(28, 3100));
+    if (map_first_run) then begin
+        display_mstr(3100);
     end
     if ((get_month >= 3) and (get_month < 5)) then begin
         if ((game_time_hour >= 600) and (game_time_hour < (600 + 100))) then begin
@@ -233,17 +233,17 @@ begin
             end
         end
     end
-    set_global_var(27, 0);
+    set_global_var(GVAR_LOAD_MAP_INDEX, 0);
 end
 
 procedure map_update_p_proc
 begin
-    if (global_var(7) == 1) then begin
-        set_global_var(47, -50);
+    if (global_var(GVAR_ENEMY_ARROYO) == 1) then begin
+        set_global_var(GVAR_TOWN_REP_ARROYO, -50);
     end
     else begin
-        if (global_var(644) == 20) then begin
-            set_global_var(47, 50);
+        if (global_var(GVAR_AHELDER_Q1_AGRO) == 20) then begin
+            set_global_var(GVAR_TOWN_REP_ARROYO, 50);
         end
     end
     if ((get_month >= 3) and (get_month < 5)) then begin
@@ -323,4 +323,3 @@ begin
         end
     end
 end
-
