diff --git a/Mods/MegaMod/Original/B21.ssl b/Mods/MegaMod/ReDefine/B21.ssl
index 80248d8..bca42ec 100644
--- a/Mods/MegaMod/Original/B21.ssl
+++ b/Mods/MegaMod/ReDefine/B21.ssl
@@ -9,11 +9,11 @@ end
 
 procedure map_enter_p_proc
 begin
-    if (metarule(14, 0)) then begin
-        display_msg(message_str(1421, 100));
+    if (map_first_run) then begin
+        display_mstr(100);
     end
     else begin
-        display_msg(message_str(1421, 102));
+        display_mstr(102);
     end
     if ((get_month >= 3) and (get_month < 5)) then begin
         if ((game_time_hour >= 600) and (game_time_hour < (600 + 100))) then begin
@@ -91,7 +91,7 @@ begin
             end
         end
     end
-    set_global_var(27, 0);
+    set_global_var(GVAR_LOAD_MAP_INDEX, 0);
 end
 
 procedure map_update_p_proc
@@ -173,4 +173,3 @@ begin
         end
     end
 end
-
