diff --git a/Mods/MegaMod/Original/BAZEUN.ssl b/Mods/MegaMod/ReDefine/BAZEUN.ssl
index f5f7d0f..292bddd 100644
--- a/Mods/MegaMod/Original/BAZEUN.ssl
+++ b/Mods/MegaMod/ReDefine/BAZEUN.ssl
@@ -146,8 +146,8 @@ end
 procedure map_enter_p_proc
 begin
     set_light_level(100);
-    if (global_var(763) != 2) then begin
-        create_object_sid(33554499, 24292, 1, -1);
+    if (global_var(GVAR_SKYNET_BROUGHT_ONLINE) != 2) then begin
+        create_object(33554499, 24292, 1);
     end
 end
 
@@ -170,4 +170,3 @@ end
 *           Not implemented                            *
 *                                                      *
 *******************************************************/
-
