diff --git a/Mods/MegaMod/Original/BHRNDMTN.ssl b/Mods/MegaMod/ReDefine/BHRNDMTN.ssl
index cce9147..faabb89 100644
--- a/Mods/MegaMod/Original/BHRNDMTN.ssl
+++ b/Mods/MegaMod/ReDefine/BHRNDMTN.ssl
@@ -146,65 +146,65 @@ end
 procedure map_enter_p_proc
 begin
     k := elevation(dude_obj);
-    master := create_object_sid(16777256, 0, 0, 591);
-    critter_add_trait(master, 1, 5, 22);
-    master_hex := tile_num_in_direction(tile_num(dude_obj), 0, 3);
-    cguard := create_object_sid(16777256, 0, 0, 592);
-    critter_add_trait(cguard, 1, 5, 22);
+    master := create_object_sid(16777256, 0, 0, SCRIPT_HCENCHA);
+    critter_add_trait(master, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_TOUGH_GUARD);
+    master_hex := tile_num_in_direction(dude_tile, 0, 3);
+    cguard := create_object_sid(16777256, 0, 0, SCRIPT_HCENCHB);
+    critter_add_trait(cguard, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_TOUGH_GUARD);
     cguard_hex := tile_num_in_direction(master_hex, 1, 2);
     critter_attempt_placement(cguard, cguard_hex, k);
-    cguard := create_object_sid(16777256, 0, 0, 592);
-    critter_add_trait(cguard, 1, 5, 22);
+    cguard := create_object_sid(16777256, 0, 0, SCRIPT_HCENCHB);
+    critter_add_trait(cguard, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_TOUGH_GUARD);
     cguard_hex := tile_num_in_direction(master_hex, 2, 2);
     critter_attempt_placement(cguard, cguard_hex, k);
-    cguard := create_object_sid(16777256, 0, 0, 592);
-    critter_add_trait(cguard, 1, 5, 22);
+    cguard := create_object_sid(16777256, 0, 0, SCRIPT_HCENCHB);
+    critter_add_trait(cguard, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_TOUGH_GUARD);
     cguard_hex := tile_num_in_direction(master_hex, 4, 8);
     critter_attempt_placement(cguard, master_hex, k);
     critter_attempt_placement(master, cguard_hex, k);
     stuff := tile_num_in_direction(master_hex, 5, 4);
-    Scenery_Creation := create_object_sid(33554959, stuff, k, -1);
+    Scenery_Creation := create_object(33554959, stuff, k);
     Scenery_Creation_Hex := stuff;
     Scenery_Creation_Count := 0;
     while (Scenery_Creation_Count < 1) do begin
         Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 1, 1);
-        Scenery_Creation := create_object_sid(33554499, Scenery_Creation_Hex, k, -1);
+        Scenery_Creation := create_object(33554499, Scenery_Creation_Hex, k);
         Scenery_Creation_Count := Scenery_Creation_Count + 1;
     end
     Scenery_Creation_Count := 0;
     while (Scenery_Creation_Count < 2) do begin
         Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 3, 1);
-        Scenery_Creation := create_object_sid(33554499, Scenery_Creation_Hex, k, -1);
+        Scenery_Creation := create_object(33554499, Scenery_Creation_Hex, k);
         Scenery_Creation_Count := Scenery_Creation_Count + 1;
     end
     Scenery_Creation_Count := 0;
     while (Scenery_Creation_Count < 2) do begin
         Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 4, 1);
-        Scenery_Creation := create_object_sid(33554499, Scenery_Creation_Hex, k, -1);
+        Scenery_Creation := create_object(33554499, Scenery_Creation_Hex, k);
         Scenery_Creation_Count := Scenery_Creation_Count + 1;
     end
     Scenery_Creation_Count := 0;
     while (Scenery_Creation_Count < 1) do begin
         Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 5, 1);
-        Scenery_Creation := create_object_sid(33554499, Scenery_Creation_Hex, k, -1);
+        Scenery_Creation := create_object(33554499, Scenery_Creation_Hex, k);
         Scenery_Creation_Count := Scenery_Creation_Count + 1;
     end
     Scenery_Creation_Count := 0;
     while (Scenery_Creation_Count < 1) do begin
         Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 0, 1);
-        Scenery_Creation := create_object_sid(33554499, Scenery_Creation_Hex, k, -1);
+        Scenery_Creation := create_object(33554499, Scenery_Creation_Hex, k);
         Scenery_Creation_Count := Scenery_Creation_Count + 1;
     end
     Scenery_Creation_Count := 0;
     while (Scenery_Creation_Count < 1) do begin
         Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 5, 1);
-        Scenery_Creation := create_object_sid(33554499, Scenery_Creation_Hex, k, -1);
+        Scenery_Creation := create_object(33554499, Scenery_Creation_Hex, k);
         Scenery_Creation_Count := Scenery_Creation_Count + 1;
     end
     Scenery_Creation_Count := 0;
     while (Scenery_Creation_Count < 2) do begin
         Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 1, 1);
-        Scenery_Creation := create_object_sid(33554499, Scenery_Creation_Hex, k, -1);
+        Scenery_Creation := create_object(33554499, Scenery_Creation_Hex, k);
         Scenery_Creation_Count := Scenery_Creation_Count + 1;
     end
     if ((get_month >= 3) and (get_month < 5)) then begin
@@ -364,4 +364,3 @@ begin
         end
     end
 end
-
