diff --git a/Mods/MegaMod/Original/BOSBURKE.ssl b/Mods/MegaMod/ReDefine/BOSBURKE.ssl
index e8f2944..478320a 100644
--- a/Mods/MegaMod/Original/BOSBURKE.ssl
+++ b/Mods/MegaMod/ReDefine/BOSBURKE.ssl
@@ -153,28 +153,28 @@ begin
     end
     if (critter_inven_obj(self_obj, 0) != arg0) then begin
         if (critter_inven_obj(self_obj, 0) == 0) then begin
-            metarule3(107, self_obj, 16777216 + arg1, 0);
+            art_change_fid_num(self_obj, 16777216 + arg1);
         end
         if ((obj_pid(critter_inven_obj(self_obj, 0)) == 1) or (obj_pid(critter_inven_obj(self_obj, 0)) == 379)) then begin
-            metarule3(107, self_obj, 16777228, 0);
+            art_change_fid_num(self_obj, 16777228);
         end
         if ((obj_pid(critter_inven_obj(self_obj, 0)) == 232) or (obj_pid(critter_inven_obj(self_obj, 0)) == 3)) then begin
-            metarule3(107, self_obj, 16777217, 0);
+            art_change_fid_num(self_obj, 16777217);
         end
         if ((obj_pid(critter_inven_obj(self_obj, 0)) == 349) or (obj_pid(critter_inven_obj(self_obj, 0)) == 348)) then begin
-            metarule3(107, self_obj, 16777287, 0);
+            art_change_fid_num(self_obj, 16777287);
         end
         if ((obj_pid(critter_inven_obj(self_obj, 0)) == 240) or (obj_pid(critter_inven_obj(self_obj, 0)) == 380) or (obj_pid(critter_inven_obj(self_obj, 0)) == 2)) then begin
-            metarule3(107, self_obj, 16777230, 0);
+            art_change_fid_num(self_obj, 16777230);
         end
         if ((obj_pid(critter_inven_obj(self_obj, 0)) == 74) or (obj_pid(critter_inven_obj(self_obj, 0)) == 265)) then begin
-            metarule3(107, self_obj, 16777229, 0);
+            art_change_fid_num(self_obj, 16777229);
         end
         if ((obj_pid(critter_inven_obj(self_obj, 0)) == 239) or (obj_pid(critter_inven_obj(self_obj, 0)) == 17) or (obj_pid(critter_inven_obj(self_obj, 0)) == 381)) then begin
-            metarule3(107, self_obj, 16777226, 0);
+            art_change_fid_num(self_obj, 16777226);
         end
         if ((obj_pid(critter_inven_obj(self_obj, 0)) == 524) or (obj_pid(critter_inven_obj(self_obj, 0)) == 113)) then begin
-            metarule3(107, self_obj, 16777218, 0);
+            art_change_fid_num(self_obj, 16777218);
         end
         LVar2 := critter_inven_obj(self_obj, 1);
         debug_msg("armour pid == " + obj_pid(LVar2));
@@ -209,14 +209,14 @@ end
 procedure pickup_p_proc
 begin
     if (source_obj == dude_obj) then begin
-        set_global_var(1098, 1);
+        set_global_var(GVAR_PRIMIT_MONK_STATE, 1);
     end
 end
 
 procedure critter_p_proc
 begin
-    if ((global_var(1098) == 1) and obj_can_see_obj(self_obj, dude_obj)) then begin
-        attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+    if ((global_var(GVAR_PRIMIT_MONK_STATE) == 1) and obj_can_see_obj(self_obj, dude_obj)) then begin
+        attack(dude_obj);
     end
 end
 
@@ -228,17 +228,17 @@ procedure look_at_p_proc
 begin
     script_overrides;
     if (local_var(4) == 0) then begin
-        display_msg(message_str(1671, 200));
+        display_msg(message_str(SCRIPT_BOSBARTN, 200));
     end
     else begin
-        display_msg(message_str(1671, 201));
+        display_msg(message_str(SCRIPT_BOSBARTN, 201));
     end
 end
 
 procedure description_p_proc
 begin
     script_overrides;
-    display_msg(message_str(1671, 202));
+    display_msg(message_str(SCRIPT_BOSBARTN, 202));
 end
 
 procedure use_skill_on_p_proc
@@ -251,19 +251,19 @@ end
 
 procedure damage_p_proc
 begin
-    if ((source_obj == dude_obj) and (has_trait(1, self_obj, 6) != 0)) then begin
-        set_global_var(1098, 1);
+    if ((source_obj == dude_obj) and (has_trait(TRAIT_OBJECT, self_obj, OBJECT_TEAM_NUM) != 0)) then begin
+        set_global_var(GVAR_PRIMIT_MONK_STATE, 1);
     end
 end
 
 procedure map_enter_p_proc
 begin
-    if (cur_map_index == 238) then begin
-        if ((global_var(1079) > 0) and (global_var(1079) < 6)) then begin
-            use_obj_on_obj(create_object_sid(110, 0, 0, -1), self_obj);
-            use_obj_on_obj(create_object_sid(110, 0, 0, -1), self_obj);
-            use_obj_on_obj(create_object_sid(53, 0, 0, -1), self_obj);
-            use_obj_on_obj(create_object_sid(259, 0, 0, -1), self_obj);
+    if (CUR_MAP_BROHDENT) then begin
+        if ((global_var(GVAR_BROTHERHOOD_HELP_QUEST) > 0) and (global_var(GVAR_BROTHERHOOD_HELP_QUEST) < 6)) then begin
+            use_obj_on_obj(create_object(110, 0, 0), self_obj);
+            use_obj_on_obj(create_object(110, 0, 0), self_obj);
+            use_obj_on_obj(create_object(53, 0, 0), self_obj);
+            use_obj_on_obj(create_object(259, 0, 0), self_obj);
         end
     end
     if (local_var(3) == 20703) then begin
@@ -273,48 +273,48 @@ end
 
 procedure map_update_p_proc
 begin
-    if ((cur_map_index == 238) and (only_once_end == 0) and (global_var(1079) == 6)) then begin
+    if ((CUR_MAP_BROHDENT) and (only_once_end == 0) and (global_var(GVAR_BROTHERHOOD_HELP_QUEST) == 6)) then begin
         only_once_end := 1;
         gfade_out(1000);
         critter_attempt_placement(dude_obj, 21103, 0);
-        move_to(dude_obj, tile_num(dude_obj), 0);
+        move_to(dude_obj, dude_tile, 0);
         kill_critter_type(16777457, -1);
         kill_critter_type(16777459, -1);
         kill_critter_type(16777460, -1);
         kill_critter_type(16777461, -1);
         kill_critter_type(16777463, -1);
         kill_critter_type(16777465, -1);
-        display_msg(message_str(1671, 1300));
+        display_msg(message_str(SCRIPT_BOSBARTN, 1300));
         call combat_complete();
     end
 end
 
 procedure talk_p_proc
 begin
-    if ((global_var(1079) > 6) and (global_var(1076) > 3)) then begin
-        float_msg(self_obj, message_str(1671, 1200), 8);
+    if ((global_var(GVAR_BROTHERHOOD_HELP_QUEST) > 6) and (global_var(GVAR_BRTHD_DISK_FROM_DEN) > 3)) then begin
+        float_msg(self_obj, message_str(SCRIPT_BOSBARTN, 1200), FLOAT_MSG_YELLOW);
     end
     else begin
         start_gdialog(1671, self_obj, -1, -1, -1);
         gsay_start;
         set_local_var(4, 1);
-        if (global_var(1080) == 2) then begin
+        if (global_var(GVAR_ABBEY_TO_BH_QUEST) == 2) then begin
             call Node006();
         end
         else begin
-            if (global_var(1079) == 6) then begin
+            if (global_var(GVAR_BROTHERHOOD_HELP_QUEST) == 6) then begin
                 call Node007();
             end
             else begin
-                if (not(global_var(1079) > 6)) then begin
+                if (not(global_var(GVAR_BROTHERHOOD_HELP_QUEST) > 6)) then begin
                     call Node001();
                 end
                 else begin
-                    if (global_var(1076) == 0) then begin
+                    if (global_var(GVAR_BRTHD_DISK_FROM_DEN) == 0) then begin
                         call Node013();
                     end
                     else begin
-                        if (global_var(1076) >= 2) then begin
+                        if (global_var(GVAR_BRTHD_DISK_FROM_DEN) >= 2) then begin
                             call Node012();
                         end
                     end
@@ -328,99 +328,99 @@ end
 
 procedure Node001
 begin
-    gsay_reply(1671, 203);
-    giq_option(4, 1671, 210, Node002, 50);
-    giq_option(4, 1671, 211, Node003, 50);
-    giq_option(4, 1671, 212, Node005, 50);
-    giq_option(4, 1671, 213, Node004, 50);
-    giq_option(4, 1671, 214, Node999, 50);
+    gsay_reply(SCRIPT_BOSBARTN, 203);
+    giq_option(4, SCRIPT_BOSBARTN, 210, Node002, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_BOSBARTN, 211, Node003, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_BOSBARTN, 212, Node005, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_BOSBARTN, 213, Node004, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_BOSBARTN, 214, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node002
 begin
-    gsay_reply(1671, 300);
-    giq_option(4, 1671, 211, Node003, 50);
-    giq_option(4, 1671, 212, Node005, 50);
-    giq_option(4, 1671, 213, Node004, 50);
-    giq_option(4, 1671, 214, Node999, 50);
+    gsay_reply(SCRIPT_BOSBARTN, 300);
+    giq_option(4, SCRIPT_BOSBARTN, 211, Node003, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_BOSBARTN, 212, Node005, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_BOSBARTN, 213, Node004, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_BOSBARTN, 214, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node003
 begin
-    gsay_reply(1671, 400);
-    giq_option(4, 1671, 210, Node002, 50);
-    giq_option(4, 1671, 212, Node005, 50);
-    giq_option(4, 1671, 213, Node004, 50);
-    giq_option(4, 1671, 214, Node999, 50);
+    gsay_reply(SCRIPT_BOSBARTN, 400);
+    giq_option(4, SCRIPT_BOSBARTN, 210, Node002, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_BOSBARTN, 212, Node005, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_BOSBARTN, 213, Node004, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_BOSBARTN, 214, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node004
 begin
-    gsay_reply(1671, 500);
-    giq_option(4, 1671, 210, Node002, 50);
-    giq_option(4, 1671, 211, Node003, 50);
-    giq_option(4, 1671, 212, Node005, 50);
-    giq_option(4, 1671, 214, Node999, 50);
+    gsay_reply(SCRIPT_BOSBARTN, 500);
+    giq_option(4, SCRIPT_BOSBARTN, 210, Node002, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_BOSBARTN, 211, Node003, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_BOSBARTN, 212, Node005, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_BOSBARTN, 214, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node005
 begin
-    if (global_var(1080) == 0) then begin
+    if (global_var(GVAR_ABBEY_TO_BH_QUEST) == 0) then begin
         mark_area_known(0, 83, 1);
-        add_obj_to_inven(dude_obj, create_object_sid(673, 0, 0, -1));
-        gsay_message(1671, 600, 49);
-        set_global_var(1080, 1);
+        add_obj_to_inven(dude_obj, create_object(673, 0, 0));
+        gsay_message(SCRIPT_BOSBARTN, 600, GOOD_REACTION);
+        set_global_var(GVAR_ABBEY_TO_BH_QUEST, 1);
     end
     else begin
-        gsay_reply(1671, 1200);
-        giq_option(4, 1671, 214, Node999, 50);
+        gsay_reply(SCRIPT_BOSBARTN, 1200);
+        giq_option(4, SCRIPT_BOSBARTN, 214, Node999, NEUTRAL_REACTION);
     end
 end
 
 procedure Node006
 begin
-    gsay_reply(1671, 700);
-    if (message_str(1671, 100) != 0) then begin
+    gsay_reply(SCRIPT_BOSBARTN, 700);
+    if (message_str(SCRIPT_BOSBARTN, 100) != 0) then begin
         if (obj_is_carrying_obj_pid(dude_obj, 229)) then begin
-            giq_option(4, 1671, 711, Node015, 50);
+            giq_option(4, SCRIPT_BOSBARTN, 711, Node015, NEUTRAL_REACTION);
         end
         else begin
-            giq_option(4, 1671, 710, Node015, 50);
+            giq_option(4, SCRIPT_BOSBARTN, 710, Node015, NEUTRAL_REACTION);
         end
     end
-    gsay_option(1671, 214, Node999, -1);
+    giq_option(1, SCRIPT_BOSBARTN, 214, Node999, -1);
 end
 
 procedure Node007
 begin
-    set_global_var(1079, 7);
-    gsay_message(1671, 1100, 49);
+    set_global_var(GVAR_BROTHERHOOD_HELP_QUEST, 7);
+    gsay_message(SCRIPT_BOSBARTN, 1100, GOOD_REACTION);
 end
 
 procedure Node008
 begin
-    set_global_var(1076, 2);
-    gsay_reply(1671, 188);
-    gsay_option(1671, 190, Node999, 50);
-    gsay_option(1671, 191, Node999, 50);
-    if ((global_var(220) == 1) and (global_var(221) == 1) and (global_var(222) == 1) and (global_var(223) == 1) and (global_var(224) == 1) and ((obj_is_carrying_obj_pid(dude_obj, 462) == 1) and (obj_is_carrying_obj_pid(dude_obj, 461) == 1) and (obj_is_carrying_obj_pid(dude_obj, 460) == 1) and (obj_is_carrying_obj_pid(dude_obj, 459) == 1) and (obj_is_carrying_obj_pid(dude_obj, 458) == 1))) then begin
-        gsay_option(1671, 192, Node010, 50);
+    set_global_var(GVAR_BRTHD_DISK_FROM_DEN, 2);
+    gsay_reply(SCRIPT_BOSBARTN, 188);
+    giq_option(1, SCRIPT_BOSBARTN, 190, Node999, NEUTRAL_REACTION);
+    giq_option(1, SCRIPT_BOSBARTN, 191, Node999, NEUTRAL_REACTION);
+    if ((global_var(GVAR_HOLODISK_MB_OUTSIDE) == 1) and (global_var(GVAR_HOLODISK_MB_LEVEL_1) == 1) and (global_var(GVAR_HOLODISK_MB_LEVEL_2) == 1) and (global_var(GVAR_HOLODISK_MB_LEVEL_3) == 1) and (global_var(GVAR_HOLODISK_MB_LEVEL_4) == 1) and ((obj_is_carrying_obj_pid(dude_obj, 462) == 1) and (obj_is_carrying_obj_pid(dude_obj, 461) == 1) and (obj_is_carrying_obj_pid(dude_obj, 460) == 1) and (obj_is_carrying_obj_pid(dude_obj, 459) == 1) and (obj_is_carrying_obj_pid(dude_obj, 458) == 1))) then begin
+        giq_option(1, SCRIPT_BOSBARTN, 192, Node010, NEUTRAL_REACTION);
     end
     else begin
-        if ((global_var(220) == 1) and (global_var(221) == 1) and (global_var(222) == 1) and (global_var(223) == 1) and (global_var(224) == 1) and ((obj_is_carrying_obj_pid(dude_obj, 462) == 0) or (obj_is_carrying_obj_pid(dude_obj, 461) == 0) or (obj_is_carrying_obj_pid(dude_obj, 460) == 0) or (obj_is_carrying_obj_pid(dude_obj, 459) == 0) or (obj_is_carrying_obj_pid(dude_obj, 458) == 0))) then begin
-            gsay_option(1671, 193, Node009, 50);
+        if ((global_var(GVAR_HOLODISK_MB_OUTSIDE) == 1) and (global_var(GVAR_HOLODISK_MB_LEVEL_1) == 1) and (global_var(GVAR_HOLODISK_MB_LEVEL_2) == 1) and (global_var(GVAR_HOLODISK_MB_LEVEL_3) == 1) and (global_var(GVAR_HOLODISK_MB_LEVEL_4) == 1) and ((obj_is_carrying_obj_pid(dude_obj, 462) == 0) or (obj_is_carrying_obj_pid(dude_obj, 461) == 0) or (obj_is_carrying_obj_pid(dude_obj, 460) == 0) or (obj_is_carrying_obj_pid(dude_obj, 459) == 0) or (obj_is_carrying_obj_pid(dude_obj, 458) == 0))) then begin
+            giq_option(1, SCRIPT_BOSBARTN, 193, Node009, NEUTRAL_REACTION);
         end
         else begin
-            gsay_option(1671, 189, Node999, 50);
+            giq_option(1, SCRIPT_BOSBARTN, 189, Node999, NEUTRAL_REACTION);
         end
     end
 end
 
 procedure Node009
 begin
-    gsay_reply(1671, 195);
-    gsay_option(1671, 197, Node999, 50);
-    gsay_option(1671, 198, Node999, 50);
+    gsay_reply(SCRIPT_BOSBARTN, 195);
+    giq_option(1, SCRIPT_BOSBARTN, 197, Node999, NEUTRAL_REACTION);
+    giq_option(1, SCRIPT_BOSBARTN, 198, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node010
@@ -457,10 +457,10 @@ begin
         destroy_object(LVar0);
     end
     give_exp_points(2000);
-    display_msg(message_str(14, 100) + 2000 + message_str(14, 101));
-    set_global_var(1076, 4);
-    gsay_reply(1671, 194);
-    gsay_option(1671, 196, Node999, 50);
+    display_msg(g_mstr(100) + 2000 + g_mstr(101));
+    set_global_var(GVAR_BRTHD_DISK_FROM_DEN, 4);
+    gsay_reply(SCRIPT_BOSBARTN, 194);
+    giq_option(1, SCRIPT_BOSBARTN, 196, Node999, NEUTRAL_REACTION);
     return 0;
     LVar0 := 0;
 end
@@ -472,42 +472,42 @@ end
 
 procedure Node012
 begin
-    gsay_reply(1671, 150);
-    if ((global_var(220) == 1) and (global_var(221) == 1) and (global_var(222) == 1) and (global_var(223) == 1) and (global_var(224) == 1) and ((obj_is_carrying_obj_pid(dude_obj, 462) == 1) and (obj_is_carrying_obj_pid(dude_obj, 461) == 1) and (obj_is_carrying_obj_pid(dude_obj, 460) == 1) and (obj_is_carrying_obj_pid(dude_obj, 459) == 1) and (obj_is_carrying_obj_pid(dude_obj, 458) == 1))) then begin
-        gsay_option(1671, 192, Node010, 50);
+    gsay_reply(SCRIPT_BOSBARTN, 150);
+    if ((global_var(GVAR_HOLODISK_MB_OUTSIDE) == 1) and (global_var(GVAR_HOLODISK_MB_LEVEL_1) == 1) and (global_var(GVAR_HOLODISK_MB_LEVEL_2) == 1) and (global_var(GVAR_HOLODISK_MB_LEVEL_3) == 1) and (global_var(GVAR_HOLODISK_MB_LEVEL_4) == 1) and ((obj_is_carrying_obj_pid(dude_obj, 462) == 1) and (obj_is_carrying_obj_pid(dude_obj, 461) == 1) and (obj_is_carrying_obj_pid(dude_obj, 460) == 1) and (obj_is_carrying_obj_pid(dude_obj, 459) == 1) and (obj_is_carrying_obj_pid(dude_obj, 458) == 1))) then begin
+        giq_option(1, SCRIPT_BOSBARTN, 192, Node010, NEUTRAL_REACTION);
     end
     else begin
-        if ((global_var(220) == 1) and (global_var(221) == 1) and (global_var(222) == 1) and (global_var(223) == 1) and (global_var(224) == 1) and ((obj_is_carrying_obj_pid(dude_obj, 462) == 0) or (obj_is_carrying_obj_pid(dude_obj, 461) == 0) or (obj_is_carrying_obj_pid(dude_obj, 460) == 0) or (obj_is_carrying_obj_pid(dude_obj, 459) == 0) or (obj_is_carrying_obj_pid(dude_obj, 458) == 0))) then begin
-            gsay_option(1671, 193, Node009, 50);
+        if ((global_var(GVAR_HOLODISK_MB_OUTSIDE) == 1) and (global_var(GVAR_HOLODISK_MB_LEVEL_1) == 1) and (global_var(GVAR_HOLODISK_MB_LEVEL_2) == 1) and (global_var(GVAR_HOLODISK_MB_LEVEL_3) == 1) and (global_var(GVAR_HOLODISK_MB_LEVEL_4) == 1) and ((obj_is_carrying_obj_pid(dude_obj, 462) == 0) or (obj_is_carrying_obj_pid(dude_obj, 461) == 0) or (obj_is_carrying_obj_pid(dude_obj, 460) == 0) or (obj_is_carrying_obj_pid(dude_obj, 459) == 0) or (obj_is_carrying_obj_pid(dude_obj, 458) == 0))) then begin
+            giq_option(1, SCRIPT_BOSBARTN, 193, Node009, NEUTRAL_REACTION);
         end
         else begin
-            gsay_option(1671, 199, Node999, 50);
+            giq_option(1, SCRIPT_BOSBARTN, 199, Node999, NEUTRAL_REACTION);
         end
     end
 end
 
 procedure Node013
 begin
-    gsay_reply(1671, 150);
-    gsay_option(1671, 181, Node014, 50);
-    gsay_option(1671, 182, Node999, 50);
-    gsay_option(1671, 183, Node999, 50);
+    gsay_reply(SCRIPT_BOSBARTN, 150);
+    giq_option(1, SCRIPT_BOSBARTN, 181, Node014, NEUTRAL_REACTION);
+    giq_option(1, SCRIPT_BOSBARTN, 182, Node999, NEUTRAL_REACTION);
+    giq_option(1, SCRIPT_BOSBARTN, 183, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node014
 begin
-    gsay_reply(1671, 184);
-    gsay_option(1671, 185, Node008, 50);
-    gsay_option(1671, 186, Node999, 50);
-    gsay_option(1671, 187, Node999, 50);
+    gsay_reply(SCRIPT_BOSBARTN, 184);
+    giq_option(1, SCRIPT_BOSBARTN, 185, Node008, NEUTRAL_REACTION);
+    giq_option(1, SCRIPT_BOSBARTN, 186, Node999, NEUTRAL_REACTION);
+    giq_option(1, SCRIPT_BOSBARTN, 187, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node015
 begin
-    set_global_var(1080, 3);
-    gsay_reply(1671, 801);
-    giq_option(4, 1671, 802, Node999, 50);
-    giq_option(4, 1671, 803, Node999, 50);
+    set_global_var(GVAR_ABBEY_TO_BH_QUEST, 3);
+    gsay_reply(SCRIPT_BOSBARTN, 801);
+    giq_option(4, SCRIPT_BOSBARTN, 802, Node999, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_BOSBARTN, 803, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node999
@@ -517,9 +517,9 @@ end
 procedure combat_complete
 begin
     gfade_in(1000);
-    set_global_var(1079, 6);
+    set_global_var(GVAR_BROTHERHOOD_HELP_QUEST, 6);
     game_ui_disable;
-    float_msg(self_obj, "" + message_str(1671, 1000) + obj_name(dude_obj) + message_str(1671, 1001) + "", 0);
+    float_msg(self_obj, "" + message_str(SCRIPT_BOSBARTN, 1000) + dude_name + message_str(SCRIPT_BOSBARTN, 1001) + "", FLOAT_MSG_NORMAL);
     add_timer_event(self_obj, game_ticks(6), 3);
     terminate_combat;
 end
@@ -527,60 +527,60 @@ end
 procedure destroy_p_proc
 begin
     if (source_obj == dude_obj) then begin
-        set_global_var(1098, 1);
-        set_global_var(5, global_var(5) + 1);
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(4) > (3 * global_var(5))) or (global_var(2) == 1)) and (global_var(1) == 0)) then begin
-            set_global_var(2, 1);
-            set_global_var(3, 0);
+        set_global_var(GVAR_PRIMIT_MONK_STATE, 1);
+        set_global_var(GVAR_GOOD_MONSTER, global_var(GVAR_GOOD_MONSTER) + 1);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(4) > (3 * global_var(GVAR_GOOD_MONSTER))) or (global_var(GVAR_CHAMPION_REPUTATION) == 1)) and (global_var(GVAR_CHILDKILLER_REPUTATION) == 0)) then begin
+            set_global_var(GVAR_CHAMPION_REPUTATION, 1);
+            set_global_var(GVAR_BERSERKER_REPUTATION, 0);
         end
-        if (metarule(51, self_obj) == 2) then begin
+        if (critter_kill_type(self_obj) == KILL_TYPE_children_kills) then begin
             if (source_obj == dude_obj) then begin
-                set_global_var(2, 0);
-                set_global_var(1, global_var(1) + 1);
-                set_global_var(0, global_var(0) + -15);
+                set_global_var(GVAR_CHAMPION_REPUTATION, 0);
+                set_global_var(GVAR_CHILDKILLER_REPUTATION, global_var(GVAR_CHILDKILLER_REPUTATION) + 1);
+                set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + -15);
                 debug_msg("Player gains " + -15 + " Karma Points.");
-                set_global_var(37, 0);
-                set_global_var(38, 0);
-                set_global_var(39, 0);
-                set_global_var(40, 0);
-                set_global_var(41, 0);
-                set_global_var(42, 0);
-                set_global_var(43, 0);
-                set_global_var(44, 0);
-                set_global_var(45, 0);
-                if (global_var(0) >= 1000) then begin
-                    set_global_var(37, 1);
+                set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+                set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+                set_global_var(GVAR_KARMA_DEFENDER, 0);
+                set_global_var(GVAR_KARMA_WANDERER, 0);
+                set_global_var(GVAR_KARMA_BETRAYER, 0);
+                set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+                set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+                set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+                    set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
                 end
                 else begin
-                    if (global_var(0) >= 750) then begin
-                        set_global_var(38, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
                     end
                     else begin
-                        if (global_var(0) >= 500) then begin
-                            set_global_var(39, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                            set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                         end
                         else begin
-                            if (global_var(0) >= 250) then begin
-                                set_global_var(40, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                                set_global_var(GVAR_KARMA_DEFENDER, 1);
                             end
                             else begin
-                                if (global_var(0) > -250) then begin
-                                    set_global_var(41, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                                    set_global_var(GVAR_KARMA_WANDERER, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -500) then begin
-                                        set_global_var(42, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                        set_global_var(GVAR_KARMA_BETRAYER, 1);
                                     end
                                     else begin
-                                        if (global_var(0) > -750) then begin
-                                            set_global_var(43, 1);
+                                        if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                            set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                         end
                                         else begin
-                                            if (global_var(0) > -1000) then begin
-                                                set_global_var(44, 1);
+                                            if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                                set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                             end
                                             else begin
-                                                set_global_var(45, 1);
+                                                set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                             end
                                         end
                                     end
@@ -589,63 +589,63 @@ begin
                         end
                     end
                 end
-                if ((cur_map_index == 236) or (cur_map_index == 242)) then begin
-                    set_global_var(1061, global_var(1061) + -8);
+                if ((CUR_MAP_EPA2) or (CUR_MAP_EPAENT)) then begin
+                    set_global_var(GVAR_LARS_MOTEL_QUEST, global_var(GVAR_LARS_MOTEL_QUEST) + -8);
                 end
                 else begin
-                    set_global_var(1062, global_var(1062) + -8);
+                    set_global_var(GVAR_JOIN_ABBEY, global_var(GVAR_JOIN_ABBEY) + -8);
                 end
                 debug_msg("Added " + -8 + " to Town Rep");
             end
         end
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(5) > (2 * global_var(4))) or (global_var(3) == 1))) then begin
-            set_global_var(3, 1);
-            set_global_var(2, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
+            set_global_var(GVAR_BERSERKER_REPUTATION, 1);
+            set_global_var(GVAR_CHAMPION_REPUTATION, 0);
         end
-        set_global_var(0, global_var(0) + -10);
+        set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + -10);
         debug_msg("Player gains " + -10 + " Karma Points.");
-        set_global_var(37, 0);
-        set_global_var(38, 0);
-        set_global_var(39, 0);
-        set_global_var(40, 0);
-        set_global_var(41, 0);
-        set_global_var(42, 0);
-        set_global_var(43, 0);
-        set_global_var(44, 0);
-        set_global_var(45, 0);
-        if (global_var(0) >= 1000) then begin
-            set_global_var(37, 1);
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+        set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+        set_global_var(GVAR_KARMA_DEFENDER, 0);
+        set_global_var(GVAR_KARMA_WANDERER, 0);
+        set_global_var(GVAR_KARMA_BETRAYER, 0);
+        set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+        set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+            set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
         end
         else begin
-            if (global_var(0) >= 750) then begin
-                set_global_var(38, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
             end
             else begin
-                if (global_var(0) >= 500) then begin
-                    set_global_var(39, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                 end
                 else begin
-                    if (global_var(0) >= 250) then begin
-                        set_global_var(40, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                        set_global_var(GVAR_KARMA_DEFENDER, 1);
                     end
                     else begin
-                        if (global_var(0) > -250) then begin
-                            set_global_var(41, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                            set_global_var(GVAR_KARMA_WANDERER, 1);
                         end
                         else begin
-                            if (global_var(0) > -500) then begin
-                                set_global_var(42, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                set_global_var(GVAR_KARMA_BETRAYER, 1);
                             end
                             else begin
-                                if (global_var(0) > -750) then begin
-                                    set_global_var(43, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -1000) then begin
-                                        set_global_var(44, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                     end
                                     else begin
-                                        set_global_var(45, 1);
+                                        set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                     end
                                 end
                             end
@@ -654,13 +654,12 @@ begin
                 end
             end
         end
-        if ((cur_map_index == 236) or (cur_map_index == 242)) then begin
-            set_global_var(1061, global_var(1061) + -5);
+        if ((CUR_MAP_EPA2) or (CUR_MAP_EPAENT)) then begin
+            set_global_var(GVAR_LARS_MOTEL_QUEST, global_var(GVAR_LARS_MOTEL_QUEST) + -5);
         end
         else begin
-            set_global_var(1062, global_var(1062) + -5);
+            set_global_var(GVAR_JOIN_ABBEY, global_var(GVAR_JOIN_ABBEY) + -5);
         end
         debug_msg("Added " + -5 + " to Town Rep");
     end
 end
-
