diff --git a/Mods/MegaMod/Original/BOSBSDOR.ssl b/Mods/MegaMod/ReDefine/BOSBSDOR.ssl
index 8079070..3eb5e5d 100644
--- a/Mods/MegaMod/Original/BOSBSDOR.ssl
+++ b/Mods/MegaMod/ReDefine/BOSBSDOR.ssl
@@ -146,7 +146,7 @@ end
 
 procedure use_p_proc
 begin
-    if (global_var(1080) == 3) then begin
+    if (global_var(GVAR_ABBEY_TO_BH_QUEST) == 3) then begin
         if (obj_is_locked(self_obj)) then begin
             obj_unlock(self_obj);
         end
@@ -190,7 +190,7 @@ end
 
 procedure map_enter_p_proc
 begin
-    if (not(metarule(22, 0))) then begin
+    if (not(is_loading_game)) then begin
         add_timer_event(self_obj, 10, 1);
     end
 end
@@ -224,4 +224,3 @@ begin
         add_timer_event(self_obj, 10, 1);
     end
 end
-
