diff --git a/Mods/MegaMod/Original/BOSFIND.ssl b/Mods/MegaMod/ReDefine/BOSFIND.ssl
index a7c5f80..a5197ed 100644
--- a/Mods/MegaMod/Original/BOSFIND.ssl
+++ b/Mods/MegaMod/ReDefine/BOSFIND.ssl
@@ -139,10 +139,9 @@ end
 
 procedure spatial_p_proc
 begin
-    if ((source_obj == dude_obj) and (global_var(650) == 1)) then begin
-        float_msg(dude_obj, message_str(1319, 225), 3);
-        display_msg(message_str(1319, 221));
-        set_global_var(650, 3);
+    if ((source_obj == dude_obj) and (global_var(GVAR_ACTEMVIL_TALK) == 1)) then begin
+        float_msg(dude_obj, message_str(SCRIPT_JCKMKN, 225), FLOAT_MSG_GREEN);
+        display_msg(message_str(SCRIPT_JCKMKN, 221));
+        set_global_var(GVAR_ACTEMVIL_TALK, 3);
     end
 end
-
