diff --git a/Mods/MegaMod/Original/BOSHLPCM.ssl b/Mods/MegaMod/ReDefine/BOSHLPCM.ssl
index 2157320..cd34134 100644
--- a/Mods/MegaMod/Original/BOSHLPCM.ssl
+++ b/Mods/MegaMod/ReDefine/BOSHLPCM.ssl
@@ -156,40 +156,40 @@ procedure party_armor_fid(variable arg0, variable arg1)
 begin
     variable LVar2 := 0;
     variable LVar3 := 0;
-    LVar3 := critter_inven_obj(self_obj, 1);
+    LVar3 := critter_inven_obj(self_obj, INVEN_TYPE_RIGHT_HAND);
     if (obj_item_subtype(LVar3) != 3) then begin
-        LVar3 := critter_inven_obj(self_obj, 2);
+        LVar3 := critter_inven_obj(self_obj, INVEN_TYPE_LEFT_HAND);
     end
-    if (critter_inven_obj(self_obj, 0) != arg0) then begin
-        if (critter_inven_obj(self_obj, 0) == 0) then begin
-            metarule3(107, self_obj, 16777216 + arg1, 0);
+    if (critter_inven_obj(self_obj, INVEN_TYPE_WORN) != arg0) then begin
+        if (critter_inven_obj(self_obj, INVEN_TYPE_WORN) == 0) then begin
+            art_change_fid_num(self_obj, 16777216 + arg1);
         end
-        if ((obj_pid(critter_inven_obj(self_obj, 0)) == 1) or (obj_pid(critter_inven_obj(self_obj, 0)) == 379)) then begin
-            metarule3(107, self_obj, 16777228, 0);
+        if ((obj_pid(critter_inven_obj(self_obj, INVEN_TYPE_WORN)) == 1) or (obj_pid(critter_inven_obj(self_obj, INVEN_TYPE_WORN)) == 379)) then begin
+            art_change_fid_num(self_obj, 16777228);
         end
-        if ((obj_pid(critter_inven_obj(self_obj, 0)) == 232) or (obj_pid(critter_inven_obj(self_obj, 0)) == 3)) then begin
-            metarule3(107, self_obj, 16777217, 0);
+        if ((obj_pid(critter_inven_obj(self_obj, INVEN_TYPE_WORN)) == 232) or (obj_pid(critter_inven_obj(self_obj, INVEN_TYPE_WORN)) == 3)) then begin
+            art_change_fid_num(self_obj, 16777217);
         end
-        if ((obj_pid(critter_inven_obj(self_obj, 0)) == 349) or (obj_pid(critter_inven_obj(self_obj, 0)) == 348)) then begin
-            metarule3(107, self_obj, 16777287, 0);
+        if ((obj_pid(critter_inven_obj(self_obj, INVEN_TYPE_WORN)) == 349) or (obj_pid(critter_inven_obj(self_obj, INVEN_TYPE_WORN)) == 348)) then begin
+            art_change_fid_num(self_obj, 16777287);
         end
-        if ((obj_pid(critter_inven_obj(self_obj, 0)) == 240) or (obj_pid(critter_inven_obj(self_obj, 0)) == 380) or (obj_pid(critter_inven_obj(self_obj, 0)) == 2)) then begin
-            metarule3(107, self_obj, 16777230, 0);
+        if ((obj_pid(critter_inven_obj(self_obj, INVEN_TYPE_WORN)) == 240) or (obj_pid(critter_inven_obj(self_obj, INVEN_TYPE_WORN)) == 380) or (obj_pid(critter_inven_obj(self_obj, INVEN_TYPE_WORN)) == 2)) then begin
+            art_change_fid_num(self_obj, 16777230);
         end
-        if ((obj_pid(critter_inven_obj(self_obj, 0)) == 74) or (obj_pid(critter_inven_obj(self_obj, 0)) == 265)) then begin
-            metarule3(107, self_obj, 16777229, 0);
+        if ((obj_pid(critter_inven_obj(self_obj, INVEN_TYPE_WORN)) == 74) or (obj_pid(critter_inven_obj(self_obj, INVEN_TYPE_WORN)) == 265)) then begin
+            art_change_fid_num(self_obj, 16777229);
         end
-        if ((obj_pid(critter_inven_obj(self_obj, 0)) == 239) or (obj_pid(critter_inven_obj(self_obj, 0)) == 17) or (obj_pid(critter_inven_obj(self_obj, 0)) == 381)) then begin
-            metarule3(107, self_obj, 16777226, 0);
+        if ((obj_pid(critter_inven_obj(self_obj, INVEN_TYPE_WORN)) == 239) or (obj_pid(critter_inven_obj(self_obj, INVEN_TYPE_WORN)) == 17) or (obj_pid(critter_inven_obj(self_obj, INVEN_TYPE_WORN)) == 381)) then begin
+            art_change_fid_num(self_obj, 16777226);
         end
-        if ((obj_pid(critter_inven_obj(self_obj, 0)) == 524) or (obj_pid(critter_inven_obj(self_obj, 0)) == 113)) then begin
-            metarule3(107, self_obj, 16777218, 0);
+        if ((obj_pid(critter_inven_obj(self_obj, INVEN_TYPE_WORN)) == 524) or (obj_pid(critter_inven_obj(self_obj, INVEN_TYPE_WORN)) == 113)) then begin
+            art_change_fid_num(self_obj, 16777218);
         end
-        LVar2 := critter_inven_obj(self_obj, 1);
+        LVar2 := critter_inven_obj(self_obj, INVEN_TYPE_RIGHT_HAND);
         debug_msg("armour pid == " + obj_pid(LVar2));
         rm_obj_from_inven(self_obj, LVar2);
         add_obj_to_inven(self_obj, LVar2);
-        wield_obj_critter(self_obj, LVar3);
+        wield_obj(LVar3);
     end
     return 0;
     arg0 := 0;
@@ -205,54 +205,54 @@ end
 procedure destroy_p_proc
 begin
     if (source_obj == dude_obj) then begin
-        set_global_var(5, global_var(5) + 1);
-        if (metarule(51, self_obj) == 2) then begin
+        set_global_var(GVAR_GOOD_MONSTER, global_var(GVAR_GOOD_MONSTER) + 1);
+        if (critter_kill_type(self_obj) == KILL_TYPE_children_kills) then begin
             if (source_obj == dude_obj) then begin
-                set_global_var(2, 0);
-                set_global_var(1, global_var(1) + 1);
-                set_global_var(0, global_var(0) - -15);
-                set_global_var(37, 0);
-                set_global_var(38, 0);
-                set_global_var(39, 0);
-                set_global_var(40, 0);
-                set_global_var(41, 0);
-                set_global_var(42, 0);
-                set_global_var(43, 0);
-                set_global_var(44, 0);
-                set_global_var(45, 0);
-                if (global_var(0) >= 1000) then begin
-                    set_global_var(37, 1);
+                set_global_var(GVAR_CHAMPION_REPUTATION, 0);
+                set_global_var(GVAR_CHILDKILLER_REPUTATION, global_var(GVAR_CHILDKILLER_REPUTATION) + 1);
+                set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) - -15);
+                set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+                set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+                set_global_var(GVAR_KARMA_DEFENDER, 0);
+                set_global_var(GVAR_KARMA_WANDERER, 0);
+                set_global_var(GVAR_KARMA_BETRAYER, 0);
+                set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+                set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+                set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+                    set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
                 end
                 else begin
-                    if (global_var(0) >= 750) then begin
-                        set_global_var(38, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
                     end
                     else begin
-                        if (global_var(0) >= 500) then begin
-                            set_global_var(39, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                            set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                         end
                         else begin
-                            if (global_var(0) >= 250) then begin
-                                set_global_var(40, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                                set_global_var(GVAR_KARMA_DEFENDER, 1);
                             end
                             else begin
-                                if (global_var(0) > -250) then begin
-                                    set_global_var(41, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                                    set_global_var(GVAR_KARMA_WANDERER, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -500) then begin
-                                        set_global_var(42, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                        set_global_var(GVAR_KARMA_BETRAYER, 1);
                                     end
                                     else begin
-                                        if (global_var(0) > -750) then begin
-                                            set_global_var(43, 1);
+                                        if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                            set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                         end
                                         else begin
-                                            if (global_var(0) > -1000) then begin
-                                                set_global_var(44, 1);
+                                            if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                                set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                             end
                                             else begin
-                                                set_global_var(45, 1);
+                                                set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                             end
                                         end
                                     end
@@ -263,58 +263,58 @@ begin
                 end
             end
         end
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(4) > (3 * global_var(5))) or (global_var(2) == 1)) and (global_var(1) == 0)) then begin
-            set_global_var(2, 1);
-            set_global_var(3, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(4) > (3 * global_var(GVAR_GOOD_MONSTER))) or (global_var(GVAR_CHAMPION_REPUTATION) == 1)) and (global_var(GVAR_CHILDKILLER_REPUTATION) == 0)) then begin
+            set_global_var(GVAR_CHAMPION_REPUTATION, 1);
+            set_global_var(GVAR_BERSERKER_REPUTATION, 0);
         end
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(5) > (2 * global_var(4))) or (global_var(3) == 1))) then begin
-            set_global_var(3, 1);
-            set_global_var(2, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
+            set_global_var(GVAR_BERSERKER_REPUTATION, 1);
+            set_global_var(GVAR_CHAMPION_REPUTATION, 0);
         end
-        set_global_var(0, global_var(0) + -10);
+        set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + -10);
         debug_msg("Player gains " + -10 + " Karma Points.");
-        set_global_var(37, 0);
-        set_global_var(38, 0);
-        set_global_var(39, 0);
-        set_global_var(40, 0);
-        set_global_var(41, 0);
-        set_global_var(42, 0);
-        set_global_var(43, 0);
-        set_global_var(44, 0);
-        set_global_var(45, 0);
-        if (global_var(0) >= 1000) then begin
-            set_global_var(37, 1);
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+        set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+        set_global_var(GVAR_KARMA_DEFENDER, 0);
+        set_global_var(GVAR_KARMA_WANDERER, 0);
+        set_global_var(GVAR_KARMA_BETRAYER, 0);
+        set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+        set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+            set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
         end
         else begin
-            if (global_var(0) >= 750) then begin
-                set_global_var(38, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
             end
             else begin
-                if (global_var(0) >= 500) then begin
-                    set_global_var(39, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                 end
                 else begin
-                    if (global_var(0) >= 250) then begin
-                        set_global_var(40, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                        set_global_var(GVAR_KARMA_DEFENDER, 1);
                     end
                     else begin
-                        if (global_var(0) > -250) then begin
-                            set_global_var(41, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                            set_global_var(GVAR_KARMA_WANDERER, 1);
                         end
                         else begin
-                            if (global_var(0) > -500) then begin
-                                set_global_var(42, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                set_global_var(GVAR_KARMA_BETRAYER, 1);
                             end
                             else begin
-                                if (global_var(0) > -750) then begin
-                                    set_global_var(43, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -1000) then begin
-                                        set_global_var(44, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                     end
                                     else begin
-                                        set_global_var(45, 1);
+                                        set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                     end
                                 end
                             end
@@ -330,8 +330,8 @@ end
 procedure critter_p_proc
 begin
     if ((map_var(2) == 9) and (map_var(5) == 0)) then begin
-        anim(dude_obj, 1000, rotation_to_tile(tile_num(dude_obj), tile_num(self_obj)));
-        anim(self_obj, 1000, rotation_to_tile(tile_num(self_obj), tile_num(dude_obj)));
+        anim(dude_obj, 1000, rotation_to_tile(dude_tile, tile_num(self_obj)));
+        anim(self_obj, 1000, rotation_to_tile(tile_num(self_obj), dude_tile));
         call talk_p_proc();
     end
 end
@@ -362,8 +362,8 @@ end
 
 procedure map_enter_p_proc
 begin
-    critter_add_trait(self_obj, 1, 6, 0);
-    if ((global_var(1081) > 1) and (((game_time / (10 * 60 * 60)) - map_var(1)) > 24)) then begin
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 0);
+    if ((global_var(GVAR_BOS_TO_BH_QUEST) > 1) and (((game_time / (10 * 60 * 60)) - map_var(1)) > 24)) then begin
         destroy_object(self_obj);
     end
     BoSCommander_obj := self_obj;
@@ -384,9 +384,9 @@ begin
         gsay_end;
         end_dialogue;
         if (map_var(5) == 0) then begin
-            float_msg(self_obj, message_str(1660, 200), 3);
+            float_msg(self_obj, message_str(SCRIPT_BRHDHLP, 200), FLOAT_MSG_GREEN);
             give_exp_points(5000);
-            display_msg(message_str(14, 100) + 5000 + message_str(14, 101));
+            display_msg(g_mstr(100) + 5000 + g_mstr(101));
             set_map_var(5, 1);
         end
     end
@@ -395,33 +395,33 @@ end
 procedure Node001
 begin
     if (from_nod == "") then begin
-        gsay_reply(1661, 200);
+        Reply(200);
     end
     else begin
         if (from_nod == "Node002") then begin
-            gsay_reply(1661, 300);
+            Reply(300);
         end
         else begin
             if (from_nod == "Node003") then begin
-                gsay_reply(1661, 400);
+                Reply(400);
             end
             else begin
                 if (from_nod == "Node004") then begin
-                    gsay_reply(1661, 500);
+                    Reply(500);
                 end
             end
         end
     end
     if (from_nod != "Node002") then begin
-        gsay_option(1661, 210, Node002, -1);
+        giq_option(1, NAME, 210, Node002, -1);
     end
     if (from_nod != "Node003") then begin
-        gsay_option(1661, 211, Node003, -1);
+        giq_option(1, NAME, 211, Node003, -1);
     end
     if (from_nod != "Node004") then begin
-        gsay_option(1661, 212, Node004, -1);
+        giq_option(1, NAME, 212, Node004, -1);
     end
-    gsay_option(1661, 213, Node999, -1);
+    giq_option(1, NAME, 213, Node999, -1);
 end
 
 procedure Node002
@@ -449,4 +449,3 @@ end
 procedure timed_event_p_proc
 begin
 end
-
