diff --git a/Mods/MegaMod/Original/BOSHLPR2.ssl b/Mods/MegaMod/ReDefine/BOSHLPR2.ssl
index 14bc87f..cbbc7f7 100644
--- a/Mods/MegaMod/Original/BOSHLPR2.ssl
+++ b/Mods/MegaMod/ReDefine/BOSHLPR2.ssl
@@ -159,8 +159,7 @@ end
 
 procedure map_enter_p_proc
 begin
-    if (global_var(1081) > 2) then begin
+    if (global_var(GVAR_BOS_TO_BH_QUEST) > 2) then begin
         destroy_object(self_obj);
     end
 end
-
