diff --git a/Mods/MegaMod/Original/BOSIMCMP.ssl b/Mods/MegaMod/ReDefine/BOSIMCMP.ssl
index f2094b2..74298df 100644
--- a/Mods/MegaMod/Original/BOSIMCMP.ssl
+++ b/Mods/MegaMod/ReDefine/BOSIMCMP.ssl
@@ -38,58 +38,58 @@ begin
     gsay_end;
     end_dialogue;
     if (is_done == 1) then begin
-        set_global_var(1074, 3);
-        display_msg(message_str(1678, 700));
+        set_global_var(GVAR_BRTHD_MARIPOZA_QUEST, 3);
+        display_mstr(700);
         give_exp_points(4000);
     end
 end
 
 procedure Node001
 begin
-    gsay_reply(1678, 200);
-    gsay_option(1678, 210, Node999, -1);
-    gsay_option(1678, 211, Node002, -1);
-    if (global_var(1074) == 2) then begin
-        gsay_option(1678, 212, Node004, -1);
+    Reply(200);
+    giq_option(1, NAME, 210, Node999, -1);
+    giq_option(1, NAME, 211, Node002, -1);
+    if (global_var(GVAR_BRTHD_MARIPOZA_QUEST) == 2) then begin
+        giq_option(1, NAME, 212, Node004, -1);
     end
-    gsay_option(1678, 213, Node003, -1);
-    gsay_option(1678, 214, Node999, -1);
+    giq_option(1, NAME, 213, Node003, -1);
+    giq_option(1, NAME, 214, Node999, -1);
 end
 
 procedure Node002
 begin
-    gsay_reply(1678, 300);
-    gsay_option(1678, 211, Node002, -1);
-    if (global_var(1074) == 2) then begin
-        gsay_option(1678, 212, Node004, -1);
+    Reply(300);
+    giq_option(1, NAME, 211, Node002, -1);
+    if (global_var(GVAR_BRTHD_MARIPOZA_QUEST) == 2) then begin
+        giq_option(1, NAME, 212, Node004, -1);
     end
-    gsay_option(1678, 213, Node003, -1);
-    gsay_option(1678, 214, Node999, -1);
+    giq_option(1, NAME, 213, Node003, -1);
+    giq_option(1, NAME, 214, Node999, -1);
 end
 
 procedure Node003
 begin
-    gsay_reply(1678, 400);
-    gsay_option(1678, 214, Node999, -1);
+    Reply(400);
+    giq_option(1, NAME, 214, Node999, -1);
 end
 
 procedure Node004
 begin
-    gsay_reply(1678, 100);
+    Reply(100);
     if (is_a) then begin
         if (map_var(3) == 0) then begin
-            gsay_option(1678, 450, turn_on, -1);
+            giq_option(1, NAME, 450, turn_on, -1);
         end
         else begin
-            gsay_option(1678, 460, turn_off, -1);
+            giq_option(1, NAME, 460, turn_off, -1);
         end
     end
     if (is_b) then begin
         if (map_var(4) == 0) then begin
-            gsay_option(1678, 450, turn_on, -1);
+            giq_option(1, NAME, 450, turn_on, -1);
         end
         else begin
-            gsay_option(1678, 460, turn_off, -1);
+            giq_option(1, NAME, 460, turn_off, -1);
         end
     end
 end
@@ -125,4 +125,3 @@ begin
         set_map_var(4, 0);
     end
 end
-
