diff --git a/Mods/MegaMod/Original/BOSISCMP.ssl b/Mods/MegaMod/ReDefine/BOSISCMP.ssl
index 3a82318..bac11f8 100644
--- a/Mods/MegaMod/Original/BOSISCMP.ssl
+++ b/Mods/MegaMod/ReDefine/BOSISCMP.ssl
@@ -49,82 +49,81 @@ begin
     if (is_1) then begin
         if ((map_var(5) bwand 1) != 0) then begin
             set_map_var(5, map_var(5) bwand -2);
-            float_msg(self_obj, message_str(1678, 600), 8);
+            float_msg(self_obj, message_str(SCRIPT_BOSIMCMP, 600), FLOAT_MSG_YELLOW);
         end
         else begin
             set_map_var(5, map_var(5) bwor 1);
-            float_msg(self_obj, message_str(1678, 500), 8);
+            float_msg(self_obj, message_str(SCRIPT_BOSIMCMP, 500), FLOAT_MSG_YELLOW);
         end
     end
     if (is_2) then begin
         if ((map_var(5) bwand 2) != 0) then begin
             set_map_var(5, map_var(5) bwand -3);
-            float_msg(self_obj, message_str(1678, 600), 8);
+            float_msg(self_obj, message_str(SCRIPT_BOSIMCMP, 600), FLOAT_MSG_YELLOW);
         end
         else begin
             set_map_var(5, map_var(5) bwor 2);
-            float_msg(self_obj, message_str(1678, 500), 8);
+            float_msg(self_obj, message_str(SCRIPT_BOSIMCMP, 500), FLOAT_MSG_YELLOW);
         end
     end
     if (is_3) then begin
         if ((map_var(5) bwand 4) != 0) then begin
             set_map_var(5, map_var(5) bwand -5);
-            float_msg(self_obj, message_str(1678, 600), 8);
+            float_msg(self_obj, message_str(SCRIPT_BOSIMCMP, 600), FLOAT_MSG_YELLOW);
         end
         else begin
             set_map_var(5, map_var(5) bwor 4);
-            float_msg(self_obj, message_str(1678, 500), 8);
+            float_msg(self_obj, message_str(SCRIPT_BOSIMCMP, 500), FLOAT_MSG_YELLOW);
         end
     end
     if (is_4) then begin
         if ((map_var(5) bwand 8) != 0) then begin
             set_map_var(5, map_var(5) bwand -9);
-            float_msg(self_obj, message_str(1678, 600), 8);
+            float_msg(self_obj, message_str(SCRIPT_BOSIMCMP, 600), FLOAT_MSG_YELLOW);
         end
         else begin
             set_map_var(5, map_var(5) bwor 8);
-            float_msg(self_obj, message_str(1678, 500), 8);
+            float_msg(self_obj, message_str(SCRIPT_BOSIMCMP, 500), FLOAT_MSG_YELLOW);
         end
     end
     if (is_5) then begin
         if ((map_var(5) bwand 16) != 0) then begin
             set_map_var(5, map_var(5) bwand -17);
-            float_msg(self_obj, message_str(1678, 600), 8);
+            float_msg(self_obj, message_str(SCRIPT_BOSIMCMP, 600), FLOAT_MSG_YELLOW);
         end
         else begin
             set_map_var(5, map_var(5) bwor 16);
-            float_msg(self_obj, message_str(1678, 500), 8);
+            float_msg(self_obj, message_str(SCRIPT_BOSIMCMP, 500), FLOAT_MSG_YELLOW);
         end
     end
     if (is_6) then begin
         if ((map_var(5) bwand 32) != 0) then begin
             set_map_var(5, map_var(5) bwand -33);
-            float_msg(self_obj, message_str(1678, 600), 8);
+            float_msg(self_obj, message_str(SCRIPT_BOSIMCMP, 600), FLOAT_MSG_YELLOW);
         end
         else begin
             set_map_var(5, map_var(5) bwor 32);
-            float_msg(self_obj, message_str(1678, 500), 8);
+            float_msg(self_obj, message_str(SCRIPT_BOSIMCMP, 500), FLOAT_MSG_YELLOW);
         end
     end
     if (is_7) then begin
         if ((map_var(5) bwand 64) != 0) then begin
             set_map_var(5, map_var(5) bwand -65);
-            float_msg(self_obj, message_str(1678, 600), 8);
+            float_msg(self_obj, message_str(SCRIPT_BOSIMCMP, 600), FLOAT_MSG_YELLOW);
         end
         else begin
             set_map_var(5, map_var(5) bwor 64);
-            float_msg(self_obj, message_str(1678, 500), 8);
+            float_msg(self_obj, message_str(SCRIPT_BOSIMCMP, 500), FLOAT_MSG_YELLOW);
         end
     end
     if (is_8) then begin
         if ((map_var(5) bwand 128) != 0) then begin
             set_map_var(5, map_var(5) bwand -129);
-            float_msg(self_obj, message_str(1678, 600), 8);
+            float_msg(self_obj, message_str(SCRIPT_BOSIMCMP, 600), FLOAT_MSG_YELLOW);
         end
         else begin
             set_map_var(5, map_var(5) bwor 128);
-            float_msg(self_obj, message_str(1678, 500), 8);
+            float_msg(self_obj, message_str(SCRIPT_BOSIMCMP, 500), FLOAT_MSG_YELLOW);
         end
     end
 end
-
