diff --git a/Mods/MegaMod/Original/BOSKLIFF.ssl b/Mods/MegaMod/ReDefine/BOSKLIFF.ssl
index 6694807..82d6a84 100644
--- a/Mods/MegaMod/Original/BOSKLIFF.ssl
+++ b/Mods/MegaMod/ReDefine/BOSKLIFF.ssl
@@ -153,9 +153,9 @@ end
 
 procedure critter_p_proc
 begin
-    if ((local_var(5) == 2) or (global_var(1098) == 1) and obj_can_see_obj(self_obj, dude_obj)) then begin
+    if ((local_var(5) == 2) or (global_var(GVAR_PRIMIT_MONK_STATE) == 1) and obj_can_see_obj(self_obj, dude_obj)) then begin
         set_local_var(5, 1);
-        attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+        attack(dude_obj);
     end
 end
 
@@ -167,17 +167,17 @@ procedure look_at_p_proc
 begin
     script_overrides;
     if (local_var(4) == 0) then begin
-        display_msg(message_str(1672, 100));
+        display_mstr(100);
     end
     else begin
-        display_msg(message_str(1672, 101));
+        display_mstr(101);
     end
 end
 
 procedure description_p_proc
 begin
     script_overrides;
-    display_msg(message_str(1672, 102));
+    display_mstr(102);
 end
 
 procedure use_skill_on_p_proc
@@ -191,14 +191,14 @@ end
 procedure damage_p_proc
 begin
     if (source_obj == dude_obj) then begin
-        set_global_var(1098, 1);
+        set_global_var(GVAR_PRIMIT_MONK_STATE, 1);
     end
 end
 
 procedure map_enter_p_proc
 begin
     set_local_var(0, tile_num(self_obj));
-    set_local_var(1, has_trait(1, self_obj, 10));
+    set_local_var(1, obj_get_rot(self_obj));
 end
 
 procedure map_update_p_proc
@@ -207,11 +207,11 @@ end
 
 procedure talk_p_proc
 begin
-    if (global_var(1078) < 3) then begin
+    if (global_var(GVAR_BRTHD_HDISK_TO_EPA) < 3) then begin
         start_gdialog(1672, self_obj, -1, -1, -1);
         gsay_start;
         set_local_var(4, 1);
-        if (global_var(1078) > 0) then begin
+        if (global_var(GVAR_BRTHD_HDISK_TO_EPA) > 0) then begin
             call Node006();
         end
         else begin
@@ -221,61 +221,61 @@ begin
         end_dialogue;
     end
     else begin
-        if (global_var(1078) == 3) then begin
-            float_msg(self_obj, message_str(1672, 1200), 8);
+        if (global_var(GVAR_BRTHD_HDISK_TO_EPA) == 3) then begin
+            float_msg(self_obj, mstr(1200), FLOAT_MSG_YELLOW);
         end
     end
 end
 
 procedure Node001
 begin
-    gsay_reply(1672, 200);
-    giq_option(4, 1672, 210, Node002, 50);
-    giq_option(4, 1672, 211, Node003, 50);
-    giq_option(4, 1672, 212, Node004, 50);
-    giq_option(4, 1672, 213, Node999, 50);
+    Reply(200);
+    giq_option(4, NAME, 210, Node002, NEUTRAL_REACTION);
+    giq_option(4, NAME, 211, Node003, NEUTRAL_REACTION);
+    giq_option(4, NAME, 212, Node004, NEUTRAL_REACTION);
+    giq_option(4, NAME, 213, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node002
 begin
-    gsay_reply(1672, 300);
-    giq_option(4, 1672, 211, Node003, 50);
-    giq_option(4, 1672, 212, Node004, 50);
-    giq_option(4, 1672, 213, Node999, 50);
+    Reply(300);
+    giq_option(4, NAME, 211, Node003, NEUTRAL_REACTION);
+    giq_option(4, NAME, 212, Node004, NEUTRAL_REACTION);
+    giq_option(4, NAME, 213, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node003
 begin
-    gsay_reply(1672, 400);
-    giq_option(4, 1672, 210, Node002, 50);
-    giq_option(4, 1672, 212, Node004, 50);
-    giq_option(4, 1672, 213, Node999, 50);
+    Reply(400);
+    giq_option(4, NAME, 210, Node002, NEUTRAL_REACTION);
+    giq_option(4, NAME, 212, Node004, NEUTRAL_REACTION);
+    giq_option(4, NAME, 213, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node004
 begin
-    set_global_var(1078, 1);
-    gsay_reply(1672, 600);
-    giq_option(4, 1672, 650, Node999, 50);
+    set_global_var(GVAR_BRTHD_HDISK_TO_EPA, 1);
+    Reply(600);
+    giq_option(4, NAME, 650, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node006
 begin
-    gsay_reply(1672, 700);
-    if (global_var(1078) == 2) then begin
-        giq_option(4, 1672, 710, Node007, 50);
+    Reply(700);
+    if (global_var(GVAR_BRTHD_HDISK_TO_EPA) == 2) then begin
+        giq_option(4, NAME, 710, Node007, NEUTRAL_REACTION);
     end
-    giq_option(4, 1672, 213, Node999, 50);
+    giq_option(4, NAME, 213, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node007
 begin
-    set_global_var(1078, 3);
+    set_global_var(GVAR_BRTHD_HDISK_TO_EPA, 3);
     give_exp_points(200);
     item_caps_adjust(dude_obj, 5);
-    gsay_reply(1672, 800);
-    giq_option(4, 1672, 850, Node999, 50);
-    giq_option(4, 1672, 851, Node999, 50);
+    Reply(800);
+    giq_option(4, NAME, 850, Node999, NEUTRAL_REACTION);
+    giq_option(4, NAME, 851, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node999
@@ -285,59 +285,59 @@ end
 procedure destroy_p_proc
 begin
     if (source_obj == dude_obj) then begin
-        set_global_var(5, global_var(5) + 1);
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(4) > (3 * global_var(5))) or (global_var(2) == 1)) and (global_var(1) == 0)) then begin
-            set_global_var(2, 1);
-            set_global_var(3, 0);
+        set_global_var(GVAR_GOOD_MONSTER, global_var(GVAR_GOOD_MONSTER) + 1);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(4) > (3 * global_var(GVAR_GOOD_MONSTER))) or (global_var(GVAR_CHAMPION_REPUTATION) == 1)) and (global_var(GVAR_CHILDKILLER_REPUTATION) == 0)) then begin
+            set_global_var(GVAR_CHAMPION_REPUTATION, 1);
+            set_global_var(GVAR_BERSERKER_REPUTATION, 0);
         end
-        if (metarule(51, self_obj) == 2) then begin
+        if (critter_kill_type(self_obj) == KILL_TYPE_children_kills) then begin
             if (source_obj == dude_obj) then begin
-                set_global_var(2, 0);
-                set_global_var(1, global_var(1) + 1);
-                set_global_var(0, global_var(0) + -15);
+                set_global_var(GVAR_CHAMPION_REPUTATION, 0);
+                set_global_var(GVAR_CHILDKILLER_REPUTATION, global_var(GVAR_CHILDKILLER_REPUTATION) + 1);
+                set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + -15);
                 debug_msg("Player gains " + -15 + " Karma Points.");
-                set_global_var(37, 0);
-                set_global_var(38, 0);
-                set_global_var(39, 0);
-                set_global_var(40, 0);
-                set_global_var(41, 0);
-                set_global_var(42, 0);
-                set_global_var(43, 0);
-                set_global_var(44, 0);
-                set_global_var(45, 0);
-                if (global_var(0) >= 1000) then begin
-                    set_global_var(37, 1);
+                set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+                set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+                set_global_var(GVAR_KARMA_DEFENDER, 0);
+                set_global_var(GVAR_KARMA_WANDERER, 0);
+                set_global_var(GVAR_KARMA_BETRAYER, 0);
+                set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+                set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+                set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+                    set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
                 end
                 else begin
-                    if (global_var(0) >= 750) then begin
-                        set_global_var(38, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
                     end
                     else begin
-                        if (global_var(0) >= 500) then begin
-                            set_global_var(39, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                            set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                         end
                         else begin
-                            if (global_var(0) >= 250) then begin
-                                set_global_var(40, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                                set_global_var(GVAR_KARMA_DEFENDER, 1);
                             end
                             else begin
-                                if (global_var(0) > -250) then begin
-                                    set_global_var(41, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                                    set_global_var(GVAR_KARMA_WANDERER, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -500) then begin
-                                        set_global_var(42, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                        set_global_var(GVAR_KARMA_BETRAYER, 1);
                                     end
                                     else begin
-                                        if (global_var(0) > -750) then begin
-                                            set_global_var(43, 1);
+                                        if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                            set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                         end
                                         else begin
-                                            if (global_var(0) > -1000) then begin
-                                                set_global_var(44, 1);
+                                            if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                                set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                             end
                                             else begin
-                                                set_global_var(45, 1);
+                                                set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                             end
                                         end
                                     end
@@ -346,63 +346,63 @@ begin
                         end
                     end
                 end
-                if ((cur_map_index == 236) or (cur_map_index == 242)) then begin
-                    set_global_var(1061, global_var(1061) + -8);
+                if ((CUR_MAP_EPA2) or (CUR_MAP_EPAENT)) then begin
+                    set_global_var(GVAR_LARS_MOTEL_QUEST, global_var(GVAR_LARS_MOTEL_QUEST) + -8);
                 end
                 else begin
-                    set_global_var(1062, global_var(1062) + -8);
+                    set_global_var(GVAR_JOIN_ABBEY, global_var(GVAR_JOIN_ABBEY) + -8);
                 end
                 debug_msg("Added " + -8 + " to Town Rep");
             end
         end
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(5) > (2 * global_var(4))) or (global_var(3) == 1))) then begin
-            set_global_var(3, 1);
-            set_global_var(2, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
+            set_global_var(GVAR_BERSERKER_REPUTATION, 1);
+            set_global_var(GVAR_CHAMPION_REPUTATION, 0);
         end
-        set_global_var(0, global_var(0) + -10);
+        set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + -10);
         debug_msg("Player gains " + -10 + " Karma Points.");
-        set_global_var(37, 0);
-        set_global_var(38, 0);
-        set_global_var(39, 0);
-        set_global_var(40, 0);
-        set_global_var(41, 0);
-        set_global_var(42, 0);
-        set_global_var(43, 0);
-        set_global_var(44, 0);
-        set_global_var(45, 0);
-        if (global_var(0) >= 1000) then begin
-            set_global_var(37, 1);
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+        set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+        set_global_var(GVAR_KARMA_DEFENDER, 0);
+        set_global_var(GVAR_KARMA_WANDERER, 0);
+        set_global_var(GVAR_KARMA_BETRAYER, 0);
+        set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+        set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+            set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
         end
         else begin
-            if (global_var(0) >= 750) then begin
-                set_global_var(38, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
             end
             else begin
-                if (global_var(0) >= 500) then begin
-                    set_global_var(39, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                 end
                 else begin
-                    if (global_var(0) >= 250) then begin
-                        set_global_var(40, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                        set_global_var(GVAR_KARMA_DEFENDER, 1);
                     end
                     else begin
-                        if (global_var(0) > -250) then begin
-                            set_global_var(41, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                            set_global_var(GVAR_KARMA_WANDERER, 1);
                         end
                         else begin
-                            if (global_var(0) > -500) then begin
-                                set_global_var(42, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                set_global_var(GVAR_KARMA_BETRAYER, 1);
                             end
                             else begin
-                                if (global_var(0) > -750) then begin
-                                    set_global_var(43, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -1000) then begin
-                                        set_global_var(44, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                     end
                                     else begin
-                                        set_global_var(45, 1);
+                                        set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                     end
                                 end
                             end
@@ -411,13 +411,12 @@ begin
                 end
             end
         end
-        if ((cur_map_index == 236) or (cur_map_index == 242)) then begin
-            set_global_var(1061, global_var(1061) + -5);
+        if ((CUR_MAP_EPA2) or (CUR_MAP_EPAENT)) then begin
+            set_global_var(GVAR_LARS_MOTEL_QUEST, global_var(GVAR_LARS_MOTEL_QUEST) + -5);
         end
         else begin
-            set_global_var(1062, global_var(1062) + -5);
+            set_global_var(GVAR_JOIN_ABBEY, global_var(GVAR_JOIN_ABBEY) + -5);
         end
         debug_msg("Added " + -5 + " to Town Rep");
     end
 end
-
