diff --git a/Mods/MegaMod/Original/BOSMARTN.ssl b/Mods/MegaMod/ReDefine/BOSMARTN.ssl
index 5872714..b878efb 100644
--- a/Mods/MegaMod/Original/BOSMARTN.ssl
+++ b/Mods/MegaMod/ReDefine/BOSMARTN.ssl
@@ -43,72 +43,72 @@ end
 procedure timed_event_p_proc
 begin
     if (fixed_param == 3) then begin
-        critter_add_trait(self_obj, 1, 6, 0);
-        critter_add_trait(self_obj, 1, 5, 47);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 0);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_SUPER_MUTANT);
     end
     if (fixed_param == 6) then begin
-        critter_add_trait(self_obj, 1, 6, 67);
-        critter_add_trait(self_obj, 1, 5, 47);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 67);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_SUPER_MUTANT);
     end
 end
 
 procedure destroy_p_proc
 begin
     if (source_obj == dude_obj) then begin
-        set_global_var(1098, 1);
-        set_global_var(5, global_var(5) + 1);
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(4) > (3 * global_var(5))) or (global_var(2) == 1)) and (global_var(1) == 0)) then begin
-            set_global_var(2, 1);
-            set_global_var(3, 0);
+        set_global_var(GVAR_PRIMIT_MONK_STATE, 1);
+        set_global_var(GVAR_GOOD_MONSTER, global_var(GVAR_GOOD_MONSTER) + 1);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(4) > (3 * global_var(GVAR_GOOD_MONSTER))) or (global_var(GVAR_CHAMPION_REPUTATION) == 1)) and (global_var(GVAR_CHILDKILLER_REPUTATION) == 0)) then begin
+            set_global_var(GVAR_CHAMPION_REPUTATION, 1);
+            set_global_var(GVAR_BERSERKER_REPUTATION, 0);
         end
-        if (metarule(51, self_obj) == 2) then begin
+        if (critter_kill_type(self_obj) == KILL_TYPE_children_kills) then begin
             if (source_obj == dude_obj) then begin
-                set_global_var(2, 0);
-                set_global_var(1, global_var(1) + 1);
-                set_global_var(0, global_var(0) + -15);
+                set_global_var(GVAR_CHAMPION_REPUTATION, 0);
+                set_global_var(GVAR_CHILDKILLER_REPUTATION, global_var(GVAR_CHILDKILLER_REPUTATION) + 1);
+                set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + -15);
                 debug_msg("Player gains " + -15 + " Karma Points.");
-                set_global_var(37, 0);
-                set_global_var(38, 0);
-                set_global_var(39, 0);
-                set_global_var(40, 0);
-                set_global_var(41, 0);
-                set_global_var(42, 0);
-                set_global_var(43, 0);
-                set_global_var(44, 0);
-                set_global_var(45, 0);
-                if (global_var(0) >= 1000) then begin
-                    set_global_var(37, 1);
+                set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+                set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+                set_global_var(GVAR_KARMA_DEFENDER, 0);
+                set_global_var(GVAR_KARMA_WANDERER, 0);
+                set_global_var(GVAR_KARMA_BETRAYER, 0);
+                set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+                set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+                set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+                    set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
                 end
                 else begin
-                    if (global_var(0) >= 750) then begin
-                        set_global_var(38, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
                     end
                     else begin
-                        if (global_var(0) >= 500) then begin
-                            set_global_var(39, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                            set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                         end
                         else begin
-                            if (global_var(0) >= 250) then begin
-                                set_global_var(40, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                                set_global_var(GVAR_KARMA_DEFENDER, 1);
                             end
                             else begin
-                                if (global_var(0) > -250) then begin
-                                    set_global_var(41, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                                    set_global_var(GVAR_KARMA_WANDERER, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -500) then begin
-                                        set_global_var(42, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                        set_global_var(GVAR_KARMA_BETRAYER, 1);
                                     end
                                     else begin
-                                        if (global_var(0) > -750) then begin
-                                            set_global_var(43, 1);
+                                        if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                            set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                         end
                                         else begin
-                                            if (global_var(0) > -1000) then begin
-                                                set_global_var(44, 1);
+                                            if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                                set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                             end
                                             else begin
-                                                set_global_var(45, 1);
+                                                set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                             end
                                         end
                                     end
@@ -117,63 +117,63 @@ begin
                         end
                     end
                 end
-                if ((cur_map_index == 236) or (cur_map_index == 242)) then begin
-                    set_global_var(1061, global_var(1061) + -8);
+                if ((CUR_MAP_EPA2) or (CUR_MAP_EPAENT)) then begin
+                    set_global_var(GVAR_LARS_MOTEL_QUEST, global_var(GVAR_LARS_MOTEL_QUEST) + -8);
                 end
                 else begin
-                    set_global_var(1062, global_var(1062) + -8);
+                    set_global_var(GVAR_JOIN_ABBEY, global_var(GVAR_JOIN_ABBEY) + -8);
                 end
                 debug_msg("Added " + -8 + " to Town Rep");
             end
         end
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(5) > (2 * global_var(4))) or (global_var(3) == 1))) then begin
-            set_global_var(3, 1);
-            set_global_var(2, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
+            set_global_var(GVAR_BERSERKER_REPUTATION, 1);
+            set_global_var(GVAR_CHAMPION_REPUTATION, 0);
         end
-        set_global_var(0, global_var(0) + -10);
+        set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + -10);
         debug_msg("Player gains " + -10 + " Karma Points.");
-        set_global_var(37, 0);
-        set_global_var(38, 0);
-        set_global_var(39, 0);
-        set_global_var(40, 0);
-        set_global_var(41, 0);
-        set_global_var(42, 0);
-        set_global_var(43, 0);
-        set_global_var(44, 0);
-        set_global_var(45, 0);
-        if (global_var(0) >= 1000) then begin
-            set_global_var(37, 1);
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+        set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+        set_global_var(GVAR_KARMA_DEFENDER, 0);
+        set_global_var(GVAR_KARMA_WANDERER, 0);
+        set_global_var(GVAR_KARMA_BETRAYER, 0);
+        set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+        set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+            set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
         end
         else begin
-            if (global_var(0) >= 750) then begin
-                set_global_var(38, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
             end
             else begin
-                if (global_var(0) >= 500) then begin
-                    set_global_var(39, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                 end
                 else begin
-                    if (global_var(0) >= 250) then begin
-                        set_global_var(40, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                        set_global_var(GVAR_KARMA_DEFENDER, 1);
                     end
                     else begin
-                        if (global_var(0) > -250) then begin
-                            set_global_var(41, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                            set_global_var(GVAR_KARMA_WANDERER, 1);
                         end
                         else begin
-                            if (global_var(0) > -500) then begin
-                                set_global_var(42, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                set_global_var(GVAR_KARMA_BETRAYER, 1);
                             end
                             else begin
-                                if (global_var(0) > -750) then begin
-                                    set_global_var(43, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -1000) then begin
-                                        set_global_var(44, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                     end
                                     else begin
-                                        set_global_var(45, 1);
+                                        set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                     end
                                 end
                             end
@@ -182,11 +182,11 @@ begin
                 end
             end
         end
-        if ((cur_map_index == 236) or (cur_map_index == 242)) then begin
-            set_global_var(1061, global_var(1061) + -5);
+        if ((CUR_MAP_EPA2) or (CUR_MAP_EPAENT)) then begin
+            set_global_var(GVAR_LARS_MOTEL_QUEST, global_var(GVAR_LARS_MOTEL_QUEST) + -5);
         end
         else begin
-            set_global_var(1062, global_var(1062) + -5);
+            set_global_var(GVAR_JOIN_ABBEY, global_var(GVAR_JOIN_ABBEY) + -5);
         end
         debug_msg("Added " + -5 + " to Town Rep");
     end
@@ -194,40 +194,40 @@ end
 
 procedure map_enter_p_proc
 begin
-    critter_add_trait(self_obj, 1, 6, 67);
-    critter_add_trait(self_obj, 1, 5, 47);
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 67);
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_SUPER_MUTANT);
     set_local_var(3, tile_num(self_obj));
-    set_local_var(2, has_trait(1, self_obj, 10));
+    set_local_var(2, obj_get_rot(self_obj));
     martin_obj := self_obj;
 end
 
 procedure damage_p_proc
 begin
     if ((has_trait(source_obj, 1, 6) != 0) and (source_obj == dude_obj)) then begin
-        set_global_var(1098, 1);
+        set_global_var(GVAR_PRIMIT_MONK_STATE, 1);
     end
 end
 
 procedure pickup_p_proc
 begin
     if (source_obj == dude_obj) then begin
-        set_global_var(1098, 1);
+        set_global_var(GVAR_PRIMIT_MONK_STATE, 1);
     end
 end
 
 procedure critter_p_proc
 begin
-    if ((global_var(1098) == 1) and obj_can_see_obj(self_obj, dude_obj)) then begin
-        attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+    if ((global_var(GVAR_PRIMIT_MONK_STATE) == 1) and obj_can_see_obj(self_obj, dude_obj)) then begin
+        attack(dude_obj);
     end
     if (map_var(7) == 1) then begin
         set_map_var(7, 2);
     end
-    if ((global_var(1081) == 13) and (do_once == 0)) then begin
-        mark_area_known(0, 82, -66);
+    if ((global_var(GVAR_BOS_TO_BH_QUEST) == 13) and (do_once == 0)) then begin
+        mark_area_known(MARK_TYPE_TOWN, 82, MARK_STATE_INVISIBLE);
         do_once := 1;
     end
-    if ((tile_distance_objs(self_obj, dude_obj) < 4) and (global_var(1081) < 13) and (local_var(0) == 0)) then begin
+    if ((tile_distance_objs(self_obj, dude_obj) < 4) and (global_var(GVAR_BOS_TO_BH_QUEST) < 13) and (local_var(0) == 0)) then begin
         set_local_var(0, 1);
         dialogue_system_enter;
     end
@@ -240,7 +240,7 @@ begin
     if (not(combat_is_initialized) and (tile_num(self_obj) != local_var(3))) then begin
         animate_move_obj_to_tile(self_obj, local_var(3), 0);
     end
-    if ((tile_num(self_obj) == local_var(3)) and not(combat_is_initialized) and (has_trait(1, self_obj, 10) != local_var(2))) then begin
+    if ((tile_num(self_obj) == local_var(3)) and not(combat_is_initialized) and (obj_get_rot(self_obj) != local_var(2))) then begin
         anim(self_obj, 1000, local_var(2));
     end
 end
@@ -249,27 +249,27 @@ procedure look_at_p_proc
 begin
     script_overrides;
     if (local_var(4) == 0) then begin
-        display_msg(message_str(1658, 100));
+        display_mstr(100);
     end
     else begin
-        display_msg(message_str(1658, 101));
+        display_mstr(101);
     end
 end
 
 procedure description_p_proc
 begin
     script_overrides;
-    display_msg(message_str(1658, 102));
+    display_mstr(102);
 end
 
 procedure talk_p_proc
 begin
     if (local_var(7)) then begin
-        float_msg(self_obj, message_str(1658, 700), 8);
+        float_msg(self_obj, mstr(700), FLOAT_MSG_YELLOW);
     end
     else begin
-        if (global_var(1081) < 13) then begin
-            if (global_var(1081) > 0) then begin
+        if (global_var(GVAR_BOS_TO_BH_QUEST) < 13) then begin
+            if (global_var(GVAR_BOS_TO_BH_QUEST) > 0) then begin
                 start_gdialog(1658, self_obj, 4, -1, -1);
                 gsay_start;
                 call Node016();
@@ -285,27 +285,27 @@ begin
             end
         end
         else begin
-            float_msg(self_obj, message_str(1658, 2300), 8);
+            float_msg(self_obj, mstr(2300), FLOAT_MSG_YELLOW);
         end
     end
 end
 
 procedure Node001
 begin
-    gsay_reply(1658, 200);
-    giq_option(1, 1658, 210, Node002, 50);
-    giq_option(1, 1658, 211, Node004a, 50);
-    giq_option(1, 1658, 212, Node005, 50);
-    giq_option(1, 1658, 213, Node999, 50);
+    Reply(200);
+    giq_option(1, NAME, 210, Node002, NEUTRAL_REACTION);
+    giq_option(1, NAME, 211, Node004a, NEUTRAL_REACTION);
+    giq_option(1, NAME, 212, Node005, NEUTRAL_REACTION);
+    giq_option(1, NAME, 213, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node002
 begin
-    gsay_reply(1658, 300);
-    giq_option(1, 1658, 310, Node004a, 50);
-    giq_option(1, 1658, 311, Node004, 50);
-    giq_option(1, 1658, 312, Node005, 50);
-    giq_option(1, 1658, 314, Node999, 50);
+    Reply(300);
+    giq_option(1, NAME, 310, Node004a, NEUTRAL_REACTION);
+    giq_option(1, NAME, 311, Node004, NEUTRAL_REACTION);
+    giq_option(1, NAME, 312, Node005, NEUTRAL_REACTION);
+    giq_option(1, NAME, 314, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node999
@@ -315,22 +315,22 @@ end
 procedure Node004
 begin
     set_local_var(7, 1);
-    gsay_reply(1658, 700);
+    Reply(700);
 end
 
 procedure Node004a
 begin
-    gsay_reply(1658, 400);
-    giq_option(1, 1658, 410, Node005, 50);
-    giq_option(1, 1658, 411, Node004, 50);
-    giq_option(1, 1658, 412, Node999, 50);
+    Reply(400);
+    giq_option(1, NAME, 410, Node005, NEUTRAL_REACTION);
+    giq_option(1, NAME, 411, Node004, NEUTRAL_REACTION);
+    giq_option(1, NAME, 412, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node005
 begin
     set_map_var(0, 1);
-    gsay_reply(1658, 500);
-    gsay_option(14, 353, Node006, 50);
+    Reply(500);
+    giq_option(1, SCRIPT_GENERIC, 353, Node006, NEUTRAL_REACTION);
     fadeOut(500);
     game_time_advance(game_ticks(600));
     fadeIn(500);
@@ -338,147 +338,146 @@ end
 
 procedure Node006
 begin
-    gsay_reply(1658, 600);
-    giq_option(1, 1658, 610, Node007, 50);
-    giq_option(1, 1658, 611, Node009, 50);
-    giq_option(1, 1658, 612, Node011, 50);
-    giq_option(1, 1658, 613, Node999, 50);
+    Reply(600);
+    giq_option(1, NAME, 610, Node007, NEUTRAL_REACTION);
+    giq_option(1, NAME, 611, Node009, NEUTRAL_REACTION);
+    giq_option(1, NAME, 612, Node011, NEUTRAL_REACTION);
+    giq_option(1, NAME, 613, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node007
 begin
-    if (metarule(17, 82) == 0) then begin
+    if (town_known(82) == MARK_STATE_UNKNOWN) then begin
         debug_msg("  mark_on_map(" + 82 + ")");
-        mark_area_known(0, 82, 1);
+        mark_area_known(MARK_TYPE_TOWN, 82, MARK_STATE_KNOWN);
     end
-    gsay_reply(1658, 1600);
-    giq_option(1, 1658, 1610, Node008, 50);
-    giq_option(1, 1658, 1611, Node008a, 50);
-    giq_option(1, 1658, 1612, Node999, 50);
+    Reply(1600);
+    giq_option(1, NAME, 1610, Node008, NEUTRAL_REACTION);
+    giq_option(1, NAME, 1611, Node008a, NEUTRAL_REACTION);
+    giq_option(1, NAME, 1612, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node008
 begin
-    gsay_reply(1658, 1700);
-    giq_option(1, 1658, 1710, Node008a, 50);
-    giq_option(1, 1658, 1711, Node999, 50);
+    Reply(1700);
+    giq_option(1, NAME, 1710, Node008a, NEUTRAL_REACTION);
+    giq_option(1, NAME, 1711, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node008a
 begin
-    set_global_var(1081, 1);
-    gsay_reply(1658, 1800);
-    giq_option(1, 1658, 1810, Node999, 50);
+    set_global_var(GVAR_BOS_TO_BH_QUEST, 1);
+    Reply(1800);
+    giq_option(1, NAME, 1810, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node009
 begin
-    gsay_reply(1658, 1400);
-    giq_option(1, 1658, 1410, Node007, 50);
-    giq_option(1, 1658, 1411, Node010, 50);
-    giq_option(1, 1658, 1412, Node999, 50);
+    Reply(1400);
+    giq_option(1, NAME, 1410, Node007, NEUTRAL_REACTION);
+    giq_option(1, NAME, 1411, Node010, NEUTRAL_REACTION);
+    giq_option(1, NAME, 1412, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node010
 begin
-    gsay_reply(1658, 1500);
-    giq_option(1, 1658, 1510, Node007, 50);
-    giq_option(1, 1658, 1511, Node999, 50);
+    Reply(1500);
+    giq_option(1, NAME, 1510, Node007, NEUTRAL_REACTION);
+    giq_option(1, NAME, 1511, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node011
 begin
-    gsay_reply(1658, 900);
-    giq_option(1, 1658, 910, Node012, 50);
-    giq_option(1, 1658, 911, Node013, 50);
-    giq_option(1, 1658, 912, Node014, 50);
-    giq_option(1, 1658, 913, Node015, 50);
-    giq_option(1, 1658, 914, Node007, 50);
+    Reply(900);
+    giq_option(1, NAME, 910, Node012, NEUTRAL_REACTION);
+    giq_option(1, NAME, 911, Node013, NEUTRAL_REACTION);
+    giq_option(1, NAME, 912, Node014, NEUTRAL_REACTION);
+    giq_option(1, NAME, 913, Node015, NEUTRAL_REACTION);
+    giq_option(1, NAME, 914, Node007, NEUTRAL_REACTION);
 end
 
 procedure Node012
 begin
-    gsay_reply(1658, 1000);
-    giq_option(1, 1658, 1010, Node011, 50);
-    giq_option(1, 1658, 1011, Node007, 50);
+    Reply(1000);
+    giq_option(1, NAME, 1010, Node011, NEUTRAL_REACTION);
+    giq_option(1, NAME, 1011, Node007, NEUTRAL_REACTION);
 end
 
 procedure Node013
 begin
-    gsay_reply(1658, 1100);
-    giq_option(1, 1658, 1110, Node011, 50);
-    giq_option(1, 1658, 1111, Node007, 50);
+    Reply(1100);
+    giq_option(1, NAME, 1110, Node011, NEUTRAL_REACTION);
+    giq_option(1, NAME, 1111, Node007, NEUTRAL_REACTION);
 end
 
 procedure Node014
 begin
-    gsay_reply(1658, 1200);
-    giq_option(1, 1658, 1210, Node011, 50);
-    giq_option(1, 1658, 1211, Node007, 50);
+    Reply(1200);
+    giq_option(1, NAME, 1210, Node011, NEUTRAL_REACTION);
+    giq_option(1, NAME, 1211, Node007, NEUTRAL_REACTION);
 end
 
 procedure Node015
 begin
-    gsay_reply(1658, 1300);
-    giq_option(1, 1658, 1310, Node011, 50);
-    giq_option(1, 1658, 1311, Node007, 50);
+    Reply(1300);
+    giq_option(1, NAME, 1310, Node011, NEUTRAL_REACTION);
+    giq_option(1, NAME, 1311, Node007, NEUTRAL_REACTION);
 end
 
 procedure Node016
 begin
-    gsay_reply(1658, 1900);
-    if ((global_var(1081) == 2) or (global_var(1081) == 6)) then begin
-        giq_option(1, 1658, 1913, Node017, 50);
+    Reply(1900);
+    if ((global_var(GVAR_BOS_TO_BH_QUEST) == 2) or (global_var(GVAR_BOS_TO_BH_QUEST) == 6)) then begin
+        giq_option(1, NAME, 1913, Node017, NEUTRAL_REACTION);
     end
-    if (global_var(1081) == 8) then begin
-        giq_option(1, 1658, 1912, Node019, 50);
+    if (global_var(GVAR_BOS_TO_BH_QUEST) == 8) then begin
+        giq_option(1, NAME, 1912, Node019, NEUTRAL_REACTION);
     end
-    if ((global_var(1081) > 8) or (global_var(1081) == 3)) then begin
-        if ((global_var(1081) == 12) or (global_var(1081) == 3)) then begin
-            giq_option(1, 1658, 1916, Node017a, 50);
+    if ((global_var(GVAR_BOS_TO_BH_QUEST) > 8) or (global_var(GVAR_BOS_TO_BH_QUEST) == 3)) then begin
+        if ((global_var(GVAR_BOS_TO_BH_QUEST) == 12) or (global_var(GVAR_BOS_TO_BH_QUEST) == 3)) then begin
+            giq_option(1, NAME, 1916, Node017a, NEUTRAL_REACTION);
         end
-        if ((global_var(1081) < 12) and (global_var(1081) != 3)) then begin
-            giq_option(1, 1658, 1917, Node017a, 50);
+        if ((global_var(GVAR_BOS_TO_BH_QUEST) < 12) and (global_var(GVAR_BOS_TO_BH_QUEST) != 3)) then begin
+            giq_option(1, NAME, 1917, Node017a, NEUTRAL_REACTION);
         end
     end
-    if (global_var(1081) == 5) then begin
-        giq_option(1, 1658, 1915, Node017a, 50);
+    if (global_var(GVAR_BOS_TO_BH_QUEST) == 5) then begin
+        giq_option(1, NAME, 1915, Node017a, NEUTRAL_REACTION);
     end
     if (map_var(7) == 0) then begin
-        giq_option(1, 1658, 1914, Node016a, 50);
+        giq_option(1, NAME, 1914, Node016a, NEUTRAL_REACTION);
     end
-    giq_option(1, 1658, 1911, Node999, 50);
+    giq_option(1, NAME, 1911, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node016a
 begin
     set_map_var(7, 1);
-    add_obj_to_inven(dude_obj, create_object_sid(32, 0, 0, -1));
-    add_obj_to_inven(dude_obj, create_object_sid(32, 0, 0, -1));
-    gsay_reply(1658, 2600);
+    add_obj_to_inven(dude_obj, create_object(32, 0, 0));
+    add_obj_to_inven(dude_obj, create_object(32, 0, 0));
+    Reply(2600);
     call Node016();
 end
 
 procedure Node017
 begin
     set_map_var(0, 4);
-    set_global_var(1081, 4);
-    gsay_reply(1658, 2200);
-    giq_option(1, 1658, 1710, Node999, 50);
+    set_global_var(GVAR_BOS_TO_BH_QUEST, 4);
+    Reply(2200);
+    giq_option(1, NAME, 1710, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node017a
 begin
-    set_global_var(1081, 13);
-    set_global_var(58, global_var(58) + 10);
-    gsay_reply(1658, 1950);
+    set_global_var(GVAR_BOS_TO_BH_QUEST, 13);
+    set_global_var(GVAR_TOWN_REP_BURIED_VAULT, global_var(GVAR_TOWN_REP_BURIED_VAULT) + 10);
+    Reply(1950);
 end
 
 procedure Node019
 begin
-    set_global_var(1081, 13);
-    set_global_var(58, global_var(58) + 10);
-    gsay_reply(1658, 2000);
-    giq_option(1, 1658, 1710, Node999, 50);
+    set_global_var(GVAR_BOS_TO_BH_QUEST, 13);
+    set_global_var(GVAR_TOWN_REP_BURIED_VAULT, global_var(GVAR_TOWN_REP_BURIED_VAULT) + 10);
+    Reply(2000);
+    giq_option(1, NAME, 1710, Node999, NEUTRAL_REACTION);
 end
-
