diff --git a/Mods/MegaMod/Original/BOSPUGAL.ssl b/Mods/MegaMod/ReDefine/BOSPUGAL.ssl
index d72c3ff..edab661 100644
--- a/Mods/MegaMod/Original/BOSPUGAL.ssl
+++ b/Mods/MegaMod/ReDefine/BOSPUGAL.ssl
@@ -146,9 +146,9 @@ end
 
 procedure destroy_p_proc
 begin
-    set_global_var(1075, 1);
-    set_global_var(58, global_var(58) + 20);
-    set_global_var(1106, 1);
+    set_global_var(GVAR_DUDE_IS_SCRIBE, 1);
+    set_global_var(GVAR_TOWN_REP_BURIED_VAULT, global_var(GVAR_TOWN_REP_BURIED_VAULT) + 20);
+    set_global_var(GVAR_PRIMITIVE_TRIBE_HELP, 1);
 end
 
 procedure use_p_proc
@@ -190,4 +190,3 @@ end
 procedure timed_event_p_proc
 begin
 end
-
