diff --git a/Mods/MegaMod/Original/BROHD12.ssl b/Mods/MegaMod/ReDefine/BROHD12.ssl
index 380e073..ab81e24 100644
--- a/Mods/MegaMod/Original/BROHD12.ssl
+++ b/Mods/MegaMod/ReDefine/BROHD12.ssl
@@ -144,24 +144,23 @@ end
 
 procedure map_enter_p_proc
 begin
-    if (global_var(734) == 2) then begin
+    if (global_var(GVAR_EPADOOR_LOCK) == 2) then begin
         move_to(dude_obj, 18740, 0);
-        set_global_var(734, 6);
+        set_global_var(GVAR_EPADOOR_LOCK, 6);
     end
-    if (global_var(734) == 3) then begin
+    if (global_var(GVAR_EPADOOR_LOCK) == 3) then begin
         move_to(dude_obj, 18736, 1);
-        set_global_var(734, 6);
+        set_global_var(GVAR_EPADOOR_LOCK, 6);
     end
-    if (global_var(734) == 7) then begin
+    if (global_var(GVAR_EPADOOR_LOCK) == 7) then begin
         move_to(dude_obj, 14912, 0);
-        set_global_var(734, 6);
+        set_global_var(GVAR_EPADOOR_LOCK, 6);
     end
-    if (global_var(1081) < 13) then begin
-        set_global_var(1098, 1);
+    if (global_var(GVAR_BOS_TO_BH_QUEST) < 13) then begin
+        set_global_var(GVAR_PRIMIT_MONK_STATE, 1);
     end
 end
 
 procedure map_update_p_proc
 begin
 end
-
