diff --git a/Mods/MegaMod/Original/BROHD34.ssl b/Mods/MegaMod/ReDefine/BROHD34.ssl
index c7d557b..ed49645 100644
--- a/Mods/MegaMod/Original/BROHD34.ssl
+++ b/Mods/MegaMod/ReDefine/BROHD34.ssl
@@ -144,17 +144,16 @@ end
 
 procedure map_enter_p_proc
 begin
-    if (global_var(734) == 4) then begin
+    if (global_var(GVAR_EPADOOR_LOCK) == 4) then begin
         move_to(dude_obj, 21140, 0);
-        set_global_var(734, 6);
+        set_global_var(GVAR_EPADOOR_LOCK, 6);
     end
-    if (global_var(734) == 5) then begin
+    if (global_var(GVAR_EPADOOR_LOCK) == 5) then begin
         move_to(dude_obj, 21140, 1);
-        set_global_var(734, 6);
+        set_global_var(GVAR_EPADOOR_LOCK, 6);
     end
 end
 
 procedure map_update_p_proc
 begin
 end
-
