diff --git a/Mods/MegaMod/Original/BSITEM15.ssl b/Mods/MegaMod/ReDefine/BSITEM15.ssl
index c73f7e2..72dec5f 100644
--- a/Mods/MegaMod/Original/BSITEM15.ssl
+++ b/Mods/MegaMod/ReDefine/BSITEM15.ssl
@@ -144,7 +144,7 @@ begin
         call talk_p_proc();
     end
     else begin
-        display_msg(message_str(2027, 600));
+        display_mstr(600);
     end
 end
 
@@ -163,15 +163,14 @@ end
 
 procedure Node001
 begin
-    gsay_reply(2027, 100);
-    giq_option(2, 2027, 300, Node999, 50);
+    Reply(100);
+    giq_option(2, NAME, 300, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node002
 begin
     game_time_advance(100);
-    gsay_reply(2027, 100);
+    Reply(100);
     end_dialogue;
     load_map(198, 0);
 end
-
