diff --git a/Mods/MegaMod/Original/BSITEM14.ssl b/Mods/MegaMod/ReDefine/BSITEM14.ssl
index bb8480a..d2b77a1 100644
--- a/Mods/MegaMod/Original/BSITEM14.ssl
+++ b/Mods/MegaMod/ReDefine/BSITEM14.ssl
@@ -139,12 +139,12 @@ begin
     variable LVar0 := 0;
     variable LVar1 := 0;
     variable LVar2 := 0;
-    LVar0 := create_object_sid(915, 0, 0, -1);
-    LVar1 := create_object_sid(916, 0, 0, -1);
+    LVar0 := create_object(915, 0, 0);
+    LVar1 := create_object(916, 0, 0);
     LVar2 := random(1, 2);
     if ((obj_is_carrying_obj_pid(dude_obj, 915) >= 1) and (LVar2 == 1)) then begin
         add_obj_to_inven(dude_obj, LVar1);
-        float_msg(dude_obj, message_str(219, 2398), 8);
+        float_msg(dude_obj, message_str(SCRIPT_KLADWTWN, 2398), FLOAT_MSG_YELLOW);
         removed_qty := obj_is_carrying_obj_pid(dude_obj, 915);
         if (1 < removed_qty) then begin
             removed_qty := 1;
@@ -153,17 +153,17 @@ begin
             restock_obj := obj_carrying_pid_obj(dude_obj, 915);
             if (obj_type(self_obj) == 1) then begin
                 restock_obj := obj_carrying_pid_obj(dude_obj, 915);
-                if (critter_inven_obj(self_obj, 0) == restock_obj) then begin
-                    if (obj_item_subtype(critter_inven_obj(dude_obj, 0)) == 0) then begin
-                        restock_obj := critter_inven_obj(dude_obj, 0);
+                if (critter_inven_obj(self_obj, INVEN_TYPE_WORN) == restock_obj) then begin
+                    if (obj_item_subtype(critter_inven_obj(dude_obj, INVEN_TYPE_WORN)) == 0) then begin
+                        restock_obj := critter_inven_obj(dude_obj, INVEN_TYPE_WORN);
                         debug_msg("armour pid == " + obj_pid(restock_obj));
                         rm_obj_from_inven(dude_obj, restock_obj);
                         add_obj_to_inven(dude_obj, restock_obj);
                     end
                 end
                 else begin
-                    if ((critter_inven_obj(dude_obj, 2) == restock_obj) or (critter_inven_obj(dude_obj, 1) == restock_obj)) then begin
-                        metarule(43, dude_obj);
+                    if ((critter_inven_obj(dude_obj, INVEN_TYPE_LEFT_HAND) == restock_obj) or (critter_inven_obj(dude_obj, INVEN_TYPE_RIGHT_HAND) == restock_obj)) then begin
+                        inven_unwield(dude_obj);
                     end
                 end
             end
@@ -176,7 +176,7 @@ begin
     else begin
         if ((obj_is_carrying_obj_pid(dude_obj, 915) >= 1) and (LVar2 == 2)) then begin
             add_obj_to_inven(dude_obj, LVar0);
-            float_msg(dude_obj, message_str(219, 2397), 8);
+            float_msg(dude_obj, message_str(SCRIPT_KLADWTWN, 2397), FLOAT_MSG_YELLOW);
             removed_qty := obj_is_carrying_obj_pid(dude_obj, 915);
             if (1 < removed_qty) then begin
                 removed_qty := 1;
@@ -185,17 +185,17 @@ begin
                 restock_obj := obj_carrying_pid_obj(dude_obj, 915);
                 if (obj_type(self_obj) == 1) then begin
                     restock_obj := obj_carrying_pid_obj(dude_obj, 915);
-                    if (critter_inven_obj(self_obj, 0) == restock_obj) then begin
-                        if (obj_item_subtype(critter_inven_obj(dude_obj, 0)) == 0) then begin
-                            restock_obj := critter_inven_obj(dude_obj, 0);
+                    if (critter_inven_obj(self_obj, INVEN_TYPE_WORN) == restock_obj) then begin
+                        if (obj_item_subtype(critter_inven_obj(dude_obj, INVEN_TYPE_WORN)) == 0) then begin
+                            restock_obj := critter_inven_obj(dude_obj, INVEN_TYPE_WORN);
                             debug_msg("armour pid == " + obj_pid(restock_obj));
                             rm_obj_from_inven(dude_obj, restock_obj);
                             add_obj_to_inven(dude_obj, restock_obj);
                         end
                     end
                     else begin
-                        if ((critter_inven_obj(dude_obj, 2) == restock_obj) or (critter_inven_obj(dude_obj, 1) == restock_obj)) then begin
-                            metarule(43, dude_obj);
+                        if ((critter_inven_obj(dude_obj, INVEN_TYPE_LEFT_HAND) == restock_obj) or (critter_inven_obj(dude_obj, INVEN_TYPE_RIGHT_HAND) == restock_obj)) then begin
+                            inven_unwield(dude_obj);
                         end
                     end
                 end
@@ -208,7 +208,7 @@ begin
         else begin
             if ((obj_is_carrying_obj_pid(dude_obj, 916) >= 1) and (LVar2 == 1)) then begin
                 add_obj_to_inven(dude_obj, LVar0);
-                float_msg(dude_obj, message_str(219, 2397), 8);
+                float_msg(dude_obj, message_str(SCRIPT_KLADWTWN, 2397), FLOAT_MSG_YELLOW);
                 removed_qty := obj_is_carrying_obj_pid(dude_obj, 916);
                 if (1 < removed_qty) then begin
                     removed_qty := 1;
@@ -217,17 +217,17 @@ begin
                     restock_obj := obj_carrying_pid_obj(dude_obj, 916);
                     if (obj_type(self_obj) == 1) then begin
                         restock_obj := obj_carrying_pid_obj(dude_obj, 916);
-                        if (critter_inven_obj(self_obj, 0) == restock_obj) then begin
-                            if (obj_item_subtype(critter_inven_obj(dude_obj, 0)) == 0) then begin
-                                restock_obj := critter_inven_obj(dude_obj, 0);
+                        if (critter_inven_obj(self_obj, INVEN_TYPE_WORN) == restock_obj) then begin
+                            if (obj_item_subtype(critter_inven_obj(dude_obj, INVEN_TYPE_WORN)) == 0) then begin
+                                restock_obj := critter_inven_obj(dude_obj, INVEN_TYPE_WORN);
                                 debug_msg("armour pid == " + obj_pid(restock_obj));
                                 rm_obj_from_inven(dude_obj, restock_obj);
                                 add_obj_to_inven(dude_obj, restock_obj);
                             end
                         end
                         else begin
-                            if ((critter_inven_obj(dude_obj, 2) == restock_obj) or (critter_inven_obj(dude_obj, 1) == restock_obj)) then begin
-                                metarule(43, dude_obj);
+                            if ((critter_inven_obj(dude_obj, INVEN_TYPE_LEFT_HAND) == restock_obj) or (critter_inven_obj(dude_obj, INVEN_TYPE_RIGHT_HAND) == restock_obj)) then begin
+                                inven_unwield(dude_obj);
                             end
                         end
                     end
@@ -240,7 +240,7 @@ begin
             else begin
                 if ((obj_is_carrying_obj_pid(dude_obj, 916) >= 1) and (LVar2 == 2)) then begin
                     add_obj_to_inven(dude_obj, LVar1);
-                    float_msg(dude_obj, message_str(219, 2398), 8);
+                    float_msg(dude_obj, message_str(SCRIPT_KLADWTWN, 2398), FLOAT_MSG_YELLOW);
                     removed_qty := obj_is_carrying_obj_pid(dude_obj, 916);
                     if (1 < removed_qty) then begin
                         removed_qty := 1;
@@ -249,17 +249,17 @@ begin
                         restock_obj := obj_carrying_pid_obj(dude_obj, 916);
                         if (obj_type(self_obj) == 1) then begin
                             restock_obj := obj_carrying_pid_obj(dude_obj, 916);
-                            if (critter_inven_obj(self_obj, 0) == restock_obj) then begin
-                                if (obj_item_subtype(critter_inven_obj(dude_obj, 0)) == 0) then begin
-                                    restock_obj := critter_inven_obj(dude_obj, 0);
+                            if (critter_inven_obj(self_obj, INVEN_TYPE_WORN) == restock_obj) then begin
+                                if (obj_item_subtype(critter_inven_obj(dude_obj, INVEN_TYPE_WORN)) == 0) then begin
+                                    restock_obj := critter_inven_obj(dude_obj, INVEN_TYPE_WORN);
                                     debug_msg("armour pid == " + obj_pid(restock_obj));
                                     rm_obj_from_inven(dude_obj, restock_obj);
                                     add_obj_to_inven(dude_obj, restock_obj);
                                 end
                             end
                             else begin
-                                if ((critter_inven_obj(dude_obj, 2) == restock_obj) or (critter_inven_obj(dude_obj, 1) == restock_obj)) then begin
-                                    metarule(43, dude_obj);
+                                if ((critter_inven_obj(dude_obj, INVEN_TYPE_LEFT_HAND) == restock_obj) or (critter_inven_obj(dude_obj, INVEN_TYPE_RIGHT_HAND) == restock_obj)) then begin
+                                    inven_unwield(dude_obj);
                                 end
                             end
                         end
@@ -277,4 +277,3 @@ begin
     LVar1 := 0;
     LVar2 := 0;
 end
-
