diff --git a/Mods/MegaMod/Original/BSITEM28.ssl b/Mods/MegaMod/ReDefine/BSITEM28.ssl
index e92b11f..adf70b2 100644
--- a/Mods/MegaMod/Original/BSITEM28.ssl
+++ b/Mods/MegaMod/ReDefine/BSITEM28.ssl
@@ -144,13 +144,13 @@ end
 procedure look_at_p_proc
 begin
     script_overrides;
-    display_msg(message_str(219, 5870));
+    display_msg(message_str(SCRIPT_KLADWTWN, 5870));
 end
 
 procedure description_p_proc
 begin
     script_overrides;
-    display_msg(message_str(219, 5871));
+    display_msg(message_str(SCRIPT_KLADWTWN, 5871));
 end
 
 procedure talk_p_proc
@@ -178,37 +178,36 @@ end
 
 procedure Node001
 begin
-    if (global_var(846) > 1) then begin
-        gsay_reply(219, 5872);
-        giq_option(2, 219, 5874, Node003, 50);
+    if (global_var(GVAR_SODA_AND_POP_ROCKS) > 1) then begin
+        gsay_reply(SCRIPT_KLADWTWN, 5872);
+        giq_option(2, SCRIPT_KLADWTWN, 5874, Node003, NEUTRAL_REACTION);
     end
     else begin
-        gsay_reply(219, 5872);
-        giq_option(2, 219, 5873, Node002, 50);
+        gsay_reply(SCRIPT_KLADWTWN, 5872);
+        giq_option(2, SCRIPT_KLADWTWN, 5873, Node002, NEUTRAL_REACTION);
     end
 end
 
 procedure Node002
 begin
-    gsay_reply(219, 5876);
-    giq_option(2, 219, 5878, Node999, 50);
+    gsay_reply(SCRIPT_KLADWTWN, 5876);
+    giq_option(2, SCRIPT_KLADWTWN, 5878, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node003
 begin
     variable LVar0 := 0;
-    gsay_reply(219, 5875);
-    LVar0 := create_object_sid(901, 0, 0, -1);
+    gsay_reply(SCRIPT_KLADWTWN, 5875);
+    LVar0 := create_object(901, 0, 0);
     add_obj_to_inven(dude_obj, LVar0);
     set_local_var(0, 1);
-    giq_option(2, 219, 5879, Node999, 50);
+    giq_option(2, SCRIPT_KLADWTWN, 5879, Node999, NEUTRAL_REACTION);
     return 0;
     LVar0 := 0;
 end
 
 procedure Node004
 begin
-    gsay_reply(219, 5877);
-    giq_option(2, 219, 5880, Node999, 50);
+    gsay_reply(SCRIPT_KLADWTWN, 5877);
+    giq_option(2, SCRIPT_KLADWTWN, 5880, Node999, NEUTRAL_REACTION);
 end
-
