diff --git a/Mods/MegaMod/Original/BSITEM26.ssl b/Mods/MegaMod/ReDefine/BSITEM26.ssl
index 58334f6..0cd5e9c 100644
--- a/Mods/MegaMod/Original/BSITEM26.ssl
+++ b/Mods/MegaMod/ReDefine/BSITEM26.ssl
@@ -137,6 +137,5 @@ end
 procedure description_p_proc
 begin
     script_overrides;
-    display_msg(message_str(1329, 100) + obj_name(dude_obj) + message_str(1329, 110));
+    display_msg(message_str(SCRIPT_KCCVMRAT, 100) + dude_name + message_str(SCRIPT_KCCVMRAT, 110));
 end
-
