diff --git a/Mods/MegaMod/Original/BSITEM49.ssl b/Mods/MegaMod/ReDefine/BSITEM49.ssl
index b39dd91..a4bf420 100644
--- a/Mods/MegaMod/Original/BSITEM49.ssl
+++ b/Mods/MegaMod/ReDefine/BSITEM49.ssl
@@ -138,11 +138,10 @@ procedure use_p_proc
 begin
     if (local_var(0) == 0) then begin
         set_local_var(0, 1);
-        display_msg(message_str(2037, 100));
-        critter_mod_skill(dude_obj, 16, 10);
+        display_mstr(100);
+        critter_mod_skill(dude_obj, SKILL_GAMBLING, 10);
     end
     else begin
-        display_msg(message_str(2037, 110));
+        display_mstr(110);
     end
 end
-
