diff --git a/Mods/MegaMod/Original/BSITEM18.ssl b/Mods/MegaMod/ReDefine/BSITEM18.ssl
index 2933d4c..3fa5147 100644
--- a/Mods/MegaMod/Original/BSITEM18.ssl
+++ b/Mods/MegaMod/ReDefine/BSITEM18.ssl
@@ -157,13 +157,13 @@ end
 procedure look_at_p_proc
 begin
     script_overrides;
-    display_msg(message_str(219, 2200));
+    display_msg(message_str(SCRIPT_KLADWTWN, 2200));
 end
 
 procedure description_p_proc
 begin
     script_overrides;
-    display_msg(message_str(219, 2201));
+    display_msg(message_str(SCRIPT_KLADWTWN, 2201));
 end
 
 procedure talk_p_proc
@@ -191,75 +191,75 @@ end
 
 procedure Node001
 begin
-    gsay_reply(219, 2202);
-    giq_option(2, 219, 2203, Node007, 50);
-    giq_option(2, 219, 2204, Node005, 50);
-    giq_option(2, 219, 2205, Node006, 50);
-    giq_option(2, 219, 2206, Node002, 50);
-    giq_option(2, 219, 2207, Node003, 50);
+    gsay_reply(SCRIPT_KLADWTWN, 2202);
+    giq_option(2, SCRIPT_KLADWTWN, 2203, Node007, NEUTRAL_REACTION);
+    giq_option(2, SCRIPT_KLADWTWN, 2204, Node005, NEUTRAL_REACTION);
+    giq_option(2, SCRIPT_KLADWTWN, 2205, Node006, NEUTRAL_REACTION);
+    giq_option(2, SCRIPT_KLADWTWN, 2206, Node002, NEUTRAL_REACTION);
+    giq_option(2, SCRIPT_KLADWTWN, 2207, Node003, NEUTRAL_REACTION);
 end
 
 procedure Node002
 begin
-    gsay_reply(219, 2208);
-    giq_option(2, 219, 2207, Node003, 50);
+    gsay_reply(SCRIPT_KLADWTWN, 2208);
+    giq_option(2, SCRIPT_KLADWTWN, 2207, Node003, NEUTRAL_REACTION);
 end
 
 procedure Node003
 begin
-    gsay_reply(219, 2211);
-    giq_option(2, 219, 2209, Node004, 50);
+    gsay_reply(SCRIPT_KLADWTWN, 2211);
+    giq_option(2, SCRIPT_KLADWTWN, 2209, Node004, NEUTRAL_REACTION);
 end
 
 procedure Node004
 begin
-    gsay_reply(219, 2210);
-    giq_option(2, 219, 2203, Node007, 50);
-    giq_option(2, 219, 2204, Node005, 50);
-    giq_option(2, 219, 2205, Node006, 50);
-    giq_option(2, 219, 2206, Node002, 50);
-    giq_option(2, 219, 2207, Node003, 50);
+    gsay_reply(SCRIPT_KLADWTWN, 2210);
+    giq_option(2, SCRIPT_KLADWTWN, 2203, Node007, NEUTRAL_REACTION);
+    giq_option(2, SCRIPT_KLADWTWN, 2204, Node005, NEUTRAL_REACTION);
+    giq_option(2, SCRIPT_KLADWTWN, 2205, Node006, NEUTRAL_REACTION);
+    giq_option(2, SCRIPT_KLADWTWN, 2206, Node002, NEUTRAL_REACTION);
+    giq_option(2, SCRIPT_KLADWTWN, 2207, Node003, NEUTRAL_REACTION);
 end
 
 procedure Node005
 begin
-    gsay_reply(219, 2212);
-    giq_option(2, 219, 2209, Node004, 50);
+    gsay_reply(SCRIPT_KLADWTWN, 2212);
+    giq_option(2, SCRIPT_KLADWTWN, 2209, Node004, NEUTRAL_REACTION);
 end
 
 procedure Node006
 begin
-    gsay_reply(219, 2213);
-    giq_option(2, 219, 2214, Node999, 50);
+    gsay_reply(SCRIPT_KLADWTWN, 2213);
+    giq_option(2, SCRIPT_KLADWTWN, 2214, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node007
 begin
-    gsay_reply(219, 2215);
-    giq_option(2, 219, 2216, Node008, 50);
-    giq_option(2, 219, 2217, Node999, 50);
-    giq_option(2, 219, 2218, Node009, 50);
+    gsay_reply(SCRIPT_KLADWTWN, 2215);
+    giq_option(2, SCRIPT_KLADWTWN, 2216, Node008, NEUTRAL_REACTION);
+    giq_option(2, SCRIPT_KLADWTWN, 2217, Node999, NEUTRAL_REACTION);
+    giq_option(2, SCRIPT_KLADWTWN, 2218, Node009, NEUTRAL_REACTION);
 end
 
 procedure Node008
 begin
-    gsay_reply(219, 2219);
-    giq_option(2, 219, 2220, Node999, 50);
+    gsay_reply(SCRIPT_KLADWTWN, 2219);
+    giq_option(2, SCRIPT_KLADWTWN, 2220, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node009
 begin
-    gsay_reply(219, 2221);
+    gsay_reply(SCRIPT_KLADWTWN, 2221);
     set_local_var(0, 1);
-    giq_option(2, 219, 2222, Node999, 50);
-    giq_option(2, 219, 2237, Node017, 50);
-    giq_option(2, 219, 2238, Node999, 51);
+    giq_option(2, SCRIPT_KLADWTWN, 2222, Node999, NEUTRAL_REACTION);
+    giq_option(2, SCRIPT_KLADWTWN, 2237, Node017, NEUTRAL_REACTION);
+    giq_option(2, SCRIPT_KLADWTWN, 2238, Node999, BAD_REACTION);
 end
 
 procedure Node010
 begin
     if ((obj_is_carrying_obj_pid(dude_obj, 918) >= 1) and (obj_is_carrying_obj_pid(dude_obj, 936) >= 1) and (obj_is_carrying_obj_pid(dude_obj, 937) >= 1) and (obj_is_carrying_obj_pid(dude_obj, 938) >= 1)) then begin
-        gsay_reply(219, 2223);
+        gsay_reply(SCRIPT_KLADWTWN, 2223);
         set_local_var(0, 2);
         removed_qty := obj_is_carrying_obj_pid(dude_obj, 918);
         if (1 < removed_qty) then begin
@@ -269,17 +269,17 @@ begin
             restock_obj := obj_carrying_pid_obj(dude_obj, 918);
             if (obj_type(self_obj) == 1) then begin
                 restock_obj := obj_carrying_pid_obj(dude_obj, 918);
-                if (critter_inven_obj(self_obj, 0) == restock_obj) then begin
-                    if (obj_item_subtype(critter_inven_obj(dude_obj, 0)) == 0) then begin
-                        restock_obj := critter_inven_obj(dude_obj, 0);
+                if (critter_inven_obj(self_obj, INVEN_TYPE_WORN) == restock_obj) then begin
+                    if (obj_item_subtype(critter_inven_obj(dude_obj, INVEN_TYPE_WORN)) == 0) then begin
+                        restock_obj := critter_inven_obj(dude_obj, INVEN_TYPE_WORN);
                         debug_msg("armour pid == " + obj_pid(restock_obj));
                         rm_obj_from_inven(dude_obj, restock_obj);
                         add_obj_to_inven(dude_obj, restock_obj);
                     end
                 end
                 else begin
-                    if ((critter_inven_obj(dude_obj, 2) == restock_obj) or (critter_inven_obj(dude_obj, 1) == restock_obj)) then begin
-                        metarule(43, dude_obj);
+                    if ((critter_inven_obj(dude_obj, INVEN_TYPE_LEFT_HAND) == restock_obj) or (critter_inven_obj(dude_obj, INVEN_TYPE_RIGHT_HAND) == restock_obj)) then begin
+                        inven_unwield(dude_obj);
                     end
                 end
             end
@@ -296,17 +296,17 @@ begin
             restock_obj := obj_carrying_pid_obj(dude_obj, 936);
             if (obj_type(self_obj) == 1) then begin
                 restock_obj := obj_carrying_pid_obj(dude_obj, 936);
-                if (critter_inven_obj(self_obj, 0) == restock_obj) then begin
-                    if (obj_item_subtype(critter_inven_obj(dude_obj, 0)) == 0) then begin
-                        restock_obj := critter_inven_obj(dude_obj, 0);
+                if (critter_inven_obj(self_obj, INVEN_TYPE_WORN) == restock_obj) then begin
+                    if (obj_item_subtype(critter_inven_obj(dude_obj, INVEN_TYPE_WORN)) == 0) then begin
+                        restock_obj := critter_inven_obj(dude_obj, INVEN_TYPE_WORN);
                         debug_msg("armour pid == " + obj_pid(restock_obj));
                         rm_obj_from_inven(dude_obj, restock_obj);
                         add_obj_to_inven(dude_obj, restock_obj);
                     end
                 end
                 else begin
-                    if ((critter_inven_obj(dude_obj, 2) == restock_obj) or (critter_inven_obj(dude_obj, 1) == restock_obj)) then begin
-                        metarule(43, dude_obj);
+                    if ((critter_inven_obj(dude_obj, INVEN_TYPE_LEFT_HAND) == restock_obj) or (critter_inven_obj(dude_obj, INVEN_TYPE_RIGHT_HAND) == restock_obj)) then begin
+                        inven_unwield(dude_obj);
                     end
                 end
             end
@@ -323,17 +323,17 @@ begin
             restock_obj := obj_carrying_pid_obj(dude_obj, 937);
             if (obj_type(self_obj) == 1) then begin
                 restock_obj := obj_carrying_pid_obj(dude_obj, 937);
-                if (critter_inven_obj(self_obj, 0) == restock_obj) then begin
-                    if (obj_item_subtype(critter_inven_obj(dude_obj, 0)) == 0) then begin
-                        restock_obj := critter_inven_obj(dude_obj, 0);
+                if (critter_inven_obj(self_obj, INVEN_TYPE_WORN) == restock_obj) then begin
+                    if (obj_item_subtype(critter_inven_obj(dude_obj, INVEN_TYPE_WORN)) == 0) then begin
+                        restock_obj := critter_inven_obj(dude_obj, INVEN_TYPE_WORN);
                         debug_msg("armour pid == " + obj_pid(restock_obj));
                         rm_obj_from_inven(dude_obj, restock_obj);
                         add_obj_to_inven(dude_obj, restock_obj);
                     end
                 end
                 else begin
-                    if ((critter_inven_obj(dude_obj, 2) == restock_obj) or (critter_inven_obj(dude_obj, 1) == restock_obj)) then begin
-                        metarule(43, dude_obj);
+                    if ((critter_inven_obj(dude_obj, INVEN_TYPE_LEFT_HAND) == restock_obj) or (critter_inven_obj(dude_obj, INVEN_TYPE_RIGHT_HAND) == restock_obj)) then begin
+                        inven_unwield(dude_obj);
                     end
                 end
             end
@@ -350,17 +350,17 @@ begin
             restock_obj := obj_carrying_pid_obj(dude_obj, 938);
             if (obj_type(self_obj) == 1) then begin
                 restock_obj := obj_carrying_pid_obj(dude_obj, 938);
-                if (critter_inven_obj(self_obj, 0) == restock_obj) then begin
-                    if (obj_item_subtype(critter_inven_obj(dude_obj, 0)) == 0) then begin
-                        restock_obj := critter_inven_obj(dude_obj, 0);
+                if (critter_inven_obj(self_obj, INVEN_TYPE_WORN) == restock_obj) then begin
+                    if (obj_item_subtype(critter_inven_obj(dude_obj, INVEN_TYPE_WORN)) == 0) then begin
+                        restock_obj := critter_inven_obj(dude_obj, INVEN_TYPE_WORN);
                         debug_msg("armour pid == " + obj_pid(restock_obj));
                         rm_obj_from_inven(dude_obj, restock_obj);
                         add_obj_to_inven(dude_obj, restock_obj);
                     end
                 end
                 else begin
-                    if ((critter_inven_obj(dude_obj, 2) == restock_obj) or (critter_inven_obj(dude_obj, 1) == restock_obj)) then begin
-                        metarule(43, dude_obj);
+                    if ((critter_inven_obj(dude_obj, INVEN_TYPE_LEFT_HAND) == restock_obj) or (critter_inven_obj(dude_obj, INVEN_TYPE_RIGHT_HAND) == restock_obj)) then begin
+                        inven_unwield(dude_obj);
                     end
                 end
             end
@@ -369,64 +369,63 @@ begin
         end
         restock_obj := 0;
         restock_amt := 0;
-        set_global_var(846, global_var(846) + 1);
+        set_global_var(GVAR_SODA_AND_POP_ROCKS, global_var(GVAR_SODA_AND_POP_ROCKS) + 1);
         give_exp_points(700);
-        display_msg(message_str(219, 2229));
-        giq_option(2, 219, 2225, Node011, 50);
+        display_msg(message_str(SCRIPT_KLADWTWN, 2229));
+        giq_option(2, SCRIPT_KLADWTWN, 2225, Node011, NEUTRAL_REACTION);
     end
     else begin
-        gsay_reply(219, 2223);
-        giq_option(2, 219, 2224, Node999, 50);
+        gsay_reply(SCRIPT_KLADWTWN, 2223);
+        giq_option(2, SCRIPT_KLADWTWN, 2224, Node999, NEUTRAL_REACTION);
     end
 end
 
 procedure Node011
 begin
-    gsay_reply(219, 2226);
-    giq_option(2, 219, 2227, Node012, 50);
-    giq_option(2, 219, 2228, Node016, 50);
-    giq_option(2, 219, 2241, Node999, 50);
+    gsay_reply(SCRIPT_KLADWTWN, 2226);
+    giq_option(2, SCRIPT_KLADWTWN, 2227, Node012, NEUTRAL_REACTION);
+    giq_option(2, SCRIPT_KLADWTWN, 2228, Node016, NEUTRAL_REACTION);
+    giq_option(2, SCRIPT_KLADWTWN, 2241, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node012
 begin
-    gsay_reply(219, 2231);
-    giq_option(2, 219, 2230, Node013, 50);
-    giq_option(2, 219, 2242, Node999, 50);
+    gsay_reply(SCRIPT_KLADWTWN, 2231);
+    giq_option(2, SCRIPT_KLADWTWN, 2230, Node013, NEUTRAL_REACTION);
+    giq_option(2, SCRIPT_KLADWTWN, 2242, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node013
 begin
-    gsay_reply(219, 2232);
-    giq_option(2, 219, 2230, Node014, 50);
-    giq_option(2, 219, 2242, Node999, 50);
+    gsay_reply(SCRIPT_KLADWTWN, 2232);
+    giq_option(2, SCRIPT_KLADWTWN, 2230, Node014, NEUTRAL_REACTION);
+    giq_option(2, SCRIPT_KLADWTWN, 2242, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node014
 begin
-    gsay_reply(219, 2233);
-    giq_option(2, 219, 2230, Node015, 50);
-    giq_option(2, 219, 2242, Node999, 50);
+    gsay_reply(SCRIPT_KLADWTWN, 2233);
+    giq_option(2, SCRIPT_KLADWTWN, 2230, Node015, NEUTRAL_REACTION);
+    giq_option(2, SCRIPT_KLADWTWN, 2242, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node015
 begin
-    gsay_reply(219, 2234);
-    giq_option(2, 219, 2243, Node016, 49);
-    giq_option(2, 219, 2242, Node999, 50);
+    gsay_reply(SCRIPT_KLADWTWN, 2234);
+    giq_option(2, SCRIPT_KLADWTWN, 2243, Node016, GOOD_REACTION);
+    giq_option(2, SCRIPT_KLADWTWN, 2242, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node016
 begin
-    gsay_reply(219, 2235);
-    set_global_var(728, global_var(728) + 1);
-    giq_option(2, 219, 2236, Node999, 50);
+    gsay_reply(SCRIPT_KLADWTWN, 2235);
+    set_global_var(GVAR_ARENA, global_var(GVAR_ARENA) + 1);
+    giq_option(2, SCRIPT_KLADWTWN, 2236, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node017
 begin
-    gsay_reply(219, 2239);
-    giq_option(2, 219, 2240, Node999, 49);
-    giq_option(2, 219, 2238, Node999, 51);
+    gsay_reply(SCRIPT_KLADWTWN, 2239);
+    giq_option(2, SCRIPT_KLADWTWN, 2240, Node999, GOOD_REACTION);
+    giq_option(2, SCRIPT_KLADWTWN, 2238, Node999, BAD_REACTION);
 end
-
