diff --git a/Mods/MegaMod/Original/BSITEM1.ssl b/Mods/MegaMod/ReDefine/BSITEM1.ssl
index 84fe88b..6a81e53 100644
--- a/Mods/MegaMod/Original/BSITEM1.ssl
+++ b/Mods/MegaMod/ReDefine/BSITEM1.ssl
@@ -137,34 +137,33 @@ end
 procedure use_p_proc
 begin
     script_overrides;
-    if (metarule(17, 6) == 0) then begin
+    if (town_known(6) == 0) then begin
         debug_msg("  mark_on_map(" + 6 + ")");
         mark_area_known(0, 6, 1);
     end
-    if (metarule(17, 13) == 0) then begin
+    if (town_known(13) == 0) then begin
         debug_msg("  mark_on_map(" + 13 + ")");
         mark_area_known(0, 13, 1);
     end
-    if (metarule(17, 7) == 0) then begin
+    if (town_known(7) == 0) then begin
         debug_msg("  mark_on_map(" + 7 + ")");
         mark_area_known(0, 7, 1);
     end
-    if (metarule(17, 14) == 0) then begin
+    if (town_known(14) == 0) then begin
         debug_msg("  mark_on_map(" + 14 + ")");
         mark_area_known(0, 14, 1);
     end
-    if (metarule(17, 4) == 0) then begin
+    if (town_known(4) == 0) then begin
         debug_msg("  mark_on_map(" + 4 + ")");
         mark_area_known(0, 4, 1);
     end
-    if (metarule(17, 1) == 0) then begin
+    if (town_known(1) == 0) then begin
         debug_msg("  mark_on_map(" + 1 + ")");
         mark_area_known(0, 1, 1);
     end
-    if (metarule(17, 3) == 0) then begin
+    if (town_known(3) == 0) then begin
         debug_msg("  mark_on_map(" + 3 + ")");
         mark_area_known(0, 3, 1);
     end
-    display_msg(message_str(2020, 100));
+    display_msg(message_str(SCRIPT_ECKAGPAL, 100));
 end
-
