diff --git a/Mods/MegaMod/Original/BSITEM21.ssl b/Mods/MegaMod/ReDefine/BSITEM21.ssl
index bb0d93e..33eee65 100644
--- a/Mods/MegaMod/Original/BSITEM21.ssl
+++ b/Mods/MegaMod/ReDefine/BSITEM21.ssl
@@ -155,7 +155,7 @@ begin
         call talk_p_proc();
     end
     else begin
-        display_msg(message_str(1306, 4150));
+        display_msg(message_str(SCRIPT_AICRSHDR, 4150));
     end
 end
 
@@ -175,98 +175,97 @@ end
 procedure Node001
 begin
     if (obj_is_carrying_obj_pid(dude_obj, 948) >= 1) then begin
-        gsay_reply(1306, 4132);
-        giq_option(2, 1306, 4133, Node002, 50);
-        giq_option(2, 1306, 4134, Node008, 50);
-        giq_option(2, 1306, 4131, Node999, 50);
+        gsay_reply(SCRIPT_AICRSHDR, 4132);
+        giq_option(2, SCRIPT_AICRSHDR, 4133, Node002, NEUTRAL_REACTION);
+        giq_option(2, SCRIPT_AICRSHDR, 4134, Node008, NEUTRAL_REACTION);
+        giq_option(2, SCRIPT_AICRSHDR, 4131, Node999, NEUTRAL_REACTION);
     end
     else begin
-        gsay_reply(1306, 4130);
-        giq_option(2, 1306, 4131, Node999, 50);
+        gsay_reply(SCRIPT_AICRSHDR, 4130);
+        giq_option(2, SCRIPT_AICRSHDR, 4131, Node999, NEUTRAL_REACTION);
     end
 end
 
 procedure Node002
 begin
-    gsay_reply(1306, 4138);
-    giq_option(2, 1306, 4137, Node003, 50);
-    giq_option(2, 1306, 4131, Node999, 50);
+    gsay_reply(SCRIPT_AICRSHDR, 4138);
+    giq_option(2, SCRIPT_AICRSHDR, 4137, Node003, NEUTRAL_REACTION);
+    giq_option(2, SCRIPT_AICRSHDR, 4131, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node003
 begin
-    gsay_reply(1306, 4139);
-    giq_option(2, 1306, 4135, Node004, 50);
-    giq_option(2, 1306, 4131, Node999, 50);
+    gsay_reply(SCRIPT_AICRSHDR, 4139);
+    giq_option(2, SCRIPT_AICRSHDR, 4135, Node004, NEUTRAL_REACTION);
+    giq_option(2, SCRIPT_AICRSHDR, 4131, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node004
 begin
-    gsay_reply(1306, 4140);
-    giq_option(2, 1306, 4137, Node005, 50);
-    giq_option(2, 1306, 4131, Node999, 50);
+    gsay_reply(SCRIPT_AICRSHDR, 4140);
+    giq_option(2, SCRIPT_AICRSHDR, 4137, Node005, NEUTRAL_REACTION);
+    giq_option(2, SCRIPT_AICRSHDR, 4131, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node005
 begin
-    gsay_reply(1306, 4141);
-    giq_option(2, 1306, 4136, Node001, 50);
-    giq_option(2, 1306, 4131, Node999, 50);
+    gsay_reply(SCRIPT_AICRSHDR, 4141);
+    giq_option(2, SCRIPT_AICRSHDR, 4136, Node001, NEUTRAL_REACTION);
+    giq_option(2, SCRIPT_AICRSHDR, 4131, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node006
 begin
-    gsay_reply(1306, 4142);
-    giq_option(2, 1306, 4137, Node007, 50);
-    giq_option(2, 1306, 4131, Node999, 50);
+    gsay_reply(SCRIPT_AICRSHDR, 4142);
+    giq_option(2, SCRIPT_AICRSHDR, 4137, Node007, NEUTRAL_REACTION);
+    giq_option(2, SCRIPT_AICRSHDR, 4131, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node007
 begin
-    gsay_reply(1306, 4143);
-    giq_option(2, 1306, 4136, Node001, 50);
-    giq_option(2, 1306, 4131, Node999, 50);
+    gsay_reply(SCRIPT_AICRSHDR, 4143);
+    giq_option(2, SCRIPT_AICRSHDR, 4136, Node001, NEUTRAL_REACTION);
+    giq_option(2, SCRIPT_AICRSHDR, 4131, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node008
 begin
-    gsay_reply(1306, 4142);
-    giq_option(2, 1306, 4137, Node009, 50);
-    giq_option(2, 1306, 4131, Node999, 50);
+    gsay_reply(SCRIPT_AICRSHDR, 4142);
+    giq_option(2, SCRIPT_AICRSHDR, 4137, Node009, NEUTRAL_REACTION);
+    giq_option(2, SCRIPT_AICRSHDR, 4131, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node009
 begin
-    gsay_reply(1306, 4143);
-    giq_option(2, 1306, 4135, Node010, 50);
-    giq_option(2, 1306, 4131, Node999, 50);
+    gsay_reply(SCRIPT_AICRSHDR, 4143);
+    giq_option(2, SCRIPT_AICRSHDR, 4135, Node010, NEUTRAL_REACTION);
+    giq_option(2, SCRIPT_AICRSHDR, 4131, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node010
 begin
-    gsay_reply(1306, 4144);
-    giq_option(2, 1306, 4137, Node011, 50);
-    giq_option(2, 1306, 4131, Node999, 50);
+    gsay_reply(SCRIPT_AICRSHDR, 4144);
+    giq_option(2, SCRIPT_AICRSHDR, 4137, Node011, NEUTRAL_REACTION);
+    giq_option(2, SCRIPT_AICRSHDR, 4131, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node011
 begin
-    gsay_reply(1306, 4145);
-    giq_option(2, 1306, 4136, Node001, 50);
-    giq_option(2, 1306, 4131, Node999, 50);
+    gsay_reply(SCRIPT_AICRSHDR, 4145);
+    giq_option(2, SCRIPT_AICRSHDR, 4136, Node001, NEUTRAL_REACTION);
+    giq_option(2, SCRIPT_AICRSHDR, 4131, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node012
 begin
-    gsay_reply(1306, 4146);
-    giq_option(2, 1306, 4137, Node013, 50);
-    giq_option(2, 1306, 4131, Node999, 50);
+    gsay_reply(SCRIPT_AICRSHDR, 4146);
+    giq_option(2, SCRIPT_AICRSHDR, 4137, Node013, NEUTRAL_REACTION);
+    giq_option(2, SCRIPT_AICRSHDR, 4131, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node013
 begin
-    gsay_reply(1306, 4147);
-    giq_option(2, 1306, 4136, Node001, 50);
-    giq_option(2, 1306, 4131, Node999, 50);
+    gsay_reply(SCRIPT_AICRSHDR, 4147);
+    giq_option(2, SCRIPT_AICRSHDR, 4136, Node001, NEUTRAL_REACTION);
+    giq_option(2, SCRIPT_AICRSHDR, 4131, Node999, NEUTRAL_REACTION);
 end
-
