diff --git a/Mods/MegaMod/Original/BSITEM12.ssl b/Mods/MegaMod/ReDefine/BSITEM12.ssl
index d71ebc9..5e82b55 100644
--- a/Mods/MegaMod/Original/BSITEM12.ssl
+++ b/Mods/MegaMod/ReDefine/BSITEM12.ssl
@@ -141,11 +141,11 @@ end
 
 procedure use_p_proc
 begin
-    if ((obj_pid(critter_inven_obj(dude_obj, 1)) == 935) or (obj_pid(critter_inven_obj(dude_obj, 2)) == 935)) then begin
+    if ((obj_pid(critter_inven_obj(dude_obj, INVEN_TYPE_RIGHT_HAND)) == 935) or (obj_pid(critter_inven_obj(dude_obj, INVEN_TYPE_LEFT_HAND)) == 935)) then begin
         call talk_p_proc();
     end
     else begin
-        display_msg(message_str(2025, 600));
+        display_msg(message_str(SCRIPT_A042, 600));
     end
 end
 
@@ -164,20 +164,19 @@ end
 
 procedure Node001
 begin
-    gsay_reply(2025, 100);
-    giq_option(2, 2025, 200, Node002, 50);
-    giq_option(2, 2025, 300, Node999, 50);
+    gsay_reply(SCRIPT_A042, 100);
+    giq_option(2, SCRIPT_A042, 200, Node002, NEUTRAL_REACTION);
+    giq_option(2, SCRIPT_A042, 300, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node002
 begin
-    gsay_reply(2025, 210);
-    giq_option(2, 2025, 300, Node999, 50);
+    gsay_reply(SCRIPT_A042, 210);
+    giq_option(2, SCRIPT_A042, 300, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node003
 begin
-    gsay_reply(2025, 100);
-    giq_option(2, 2025, 300, Node999, 50);
+    gsay_reply(SCRIPT_A042, 100);
+    giq_option(2, SCRIPT_A042, 300, Node999, NEUTRAL_REACTION);
 end
-
