diff --git a/Mods/MegaMod/Original/CAVE06.ssl b/Mods/MegaMod/ReDefine/CAVE06.ssl
index a91dcd1..e1ba032 100644
--- a/Mods/MegaMod/Original/CAVE06.ssl
+++ b/Mods/MegaMod/ReDefine/CAVE06.ssl
@@ -26,14 +26,14 @@ procedure map_update_p_proc;
 
 procedure place_car(variable arg0)
 begin
-    set_global_var(633, arg0);
-    set_global_var(1646, elevation(dude_obj));
-    while (global_var(1645) != cur_map_index) do begin
-        set_global_var(1645, cur_map_index);
+    set_global_var(GVAR_CAR_PLACED_TILE, arg0);
+    set_global_var(GVAR_CAR_HAS_BEEN_DRIVEN, elevation(dude_obj));
+    while (global_var(GVAR_CAR_CURRENT_TOWN) != cur_map_index) do begin
+        set_global_var(GVAR_CAR_CURRENT_TOWN, cur_map_index);
     end
     if (not(tile_contains_pid_obj(arg0, elevation(dude_obj), 33556384) != 0) or not(tile_contains_pid_obj(arg0, elevation(dude_obj), 33555441) != 0) or not(tile_contains_pid_obj(arg0, elevation(dude_obj), 33556385) != 0) or not(tile_contains_pid_obj(arg0, elevation(dude_obj), 33556386) != 0) or not(tile_contains_pid_obj(arg0, elevation(dude_obj), 33556387) != 0) or not(tile_contains_pid_obj(arg0, elevation(dude_obj), 33556388) != 0) or not(tile_contains_pid_obj(arg0, elevation(dude_obj), 33556389) != 0)) then begin
-        if (global_var(1644) == 0) then begin
-            Scenery_Creation := create_object_sid(33556384, arg0, elevation(dude_obj), 304);
+        if (global_var(GVAR_CAR_CURRENT_ELEVATION) == 0) then begin
+            Scenery_Creation := create_object_sid(33556384, arg0, elevation(dude_obj), SCRIPT_ZSDRVCAR);
         end
         else begin
             Scenery_Creation_Ptr := tile_contains_pid_obj(arg0, elevation(dude_obj), 33556384);
@@ -41,8 +41,8 @@ begin
                 destroy_object(Scenery_Creation_Ptr);
             end
         end
-        if (global_var(1644) == 1) then begin
-            Scenery_Creation := create_object_sid(33555441, arg0, elevation(dude_obj), 304);
+        if (global_var(GVAR_CAR_CURRENT_ELEVATION) == 1) then begin
+            Scenery_Creation := create_object_sid(33555441, arg0, elevation(dude_obj), SCRIPT_ZSDRVCAR);
         end
         else begin
             Scenery_Creation_Ptr := tile_contains_pid_obj(arg0, elevation(dude_obj), 33555441);
@@ -50,8 +50,8 @@ begin
                 destroy_object(Scenery_Creation_Ptr);
             end
         end
-        if (global_var(1644) == 2) then begin
-            Scenery_Creation := create_object_sid(33556385, arg0, elevation(dude_obj), 304);
+        if (global_var(GVAR_CAR_CURRENT_ELEVATION) == 2) then begin
+            Scenery_Creation := create_object_sid(33556385, arg0, elevation(dude_obj), SCRIPT_ZSDRVCAR);
         end
         else begin
             Scenery_Creation_Ptr := tile_contains_pid_obj(arg0, elevation(dude_obj), 33556385);
@@ -59,8 +59,8 @@ begin
                 destroy_object(Scenery_Creation_Ptr);
             end
         end
-        if (global_var(1644) == 3) then begin
-            Scenery_Creation := create_object_sid(33556386, arg0, elevation(dude_obj), 304);
+        if (global_var(GVAR_CAR_CURRENT_ELEVATION) == 3) then begin
+            Scenery_Creation := create_object_sid(33556386, arg0, elevation(dude_obj), SCRIPT_ZSDRVCAR);
         end
         else begin
             Scenery_Creation_Ptr := tile_contains_pid_obj(arg0, elevation(dude_obj), 33556386);
@@ -68,8 +68,8 @@ begin
                 destroy_object(Scenery_Creation_Ptr);
             end
         end
-        if (global_var(1644) == 4) then begin
-            Scenery_Creation := create_object_sid(33556387, arg0, elevation(dude_obj), 304);
+        if (global_var(GVAR_CAR_CURRENT_ELEVATION) == 4) then begin
+            Scenery_Creation := create_object_sid(33556387, arg0, elevation(dude_obj), SCRIPT_ZSDRVCAR);
         end
         else begin
             Scenery_Creation_Ptr := tile_contains_pid_obj(arg0, elevation(dude_obj), 33556387);
@@ -77,8 +77,8 @@ begin
                 destroy_object(Scenery_Creation_Ptr);
             end
         end
-        if (global_var(1644) == 5) then begin
-            Scenery_Creation := create_object_sid(33556388, arg0, elevation(dude_obj), 304);
+        if (global_var(GVAR_CAR_CURRENT_ELEVATION) == 5) then begin
+            Scenery_Creation := create_object_sid(33556388, arg0, elevation(dude_obj), SCRIPT_ZSDRVCAR);
         end
         else begin
             Scenery_Creation_Ptr := tile_contains_pid_obj(arg0, elevation(dude_obj), 33556388);
@@ -86,8 +86,8 @@ begin
                 destroy_object(Scenery_Creation_Ptr);
             end
         end
-        if (global_var(1644) == 6) then begin
-            Scenery_Creation := create_object_sid(33556389, arg0, elevation(dude_obj), 304);
+        if (global_var(GVAR_CAR_CURRENT_ELEVATION) == 6) then begin
+            Scenery_Creation := create_object_sid(33556389, arg0, elevation(dude_obj), SCRIPT_ZSDRVCAR);
         end
         else begin
             Scenery_Creation_Ptr := tile_contains_pid_obj(arg0, elevation(dude_obj), 33556389);
@@ -99,7 +99,7 @@ begin
         Scenery_Creation_Count := 0;
         while (Scenery_Creation_Count < 2) do begin
             Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 1, 1);
-            Scenery_Creation := create_object_sid(33554499, Scenery_Creation_Hex, elevation(dude_obj), -1);
+            Scenery_Creation := create_object(33554499, Scenery_Creation_Hex, elevation(dude_obj));
             if (tile_num(Scenery_Creation) != Scenery_Creation_Hex) then begin
                 destroy_object(Scenery_Creation);
             end
@@ -108,7 +108,7 @@ begin
         Scenery_Creation_Count := 0;
         while (Scenery_Creation_Count < 2) do begin
             Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 2, 1);
-            Scenery_Creation := create_object_sid(33554499, Scenery_Creation_Hex, elevation(dude_obj), -1);
+            Scenery_Creation := create_object(33554499, Scenery_Creation_Hex, elevation(dude_obj));
             if (tile_num(Scenery_Creation) != Scenery_Creation_Hex) then begin
                 destroy_object(Scenery_Creation);
             end
@@ -117,7 +117,7 @@ begin
         Scenery_Creation_Count := 0;
         while (Scenery_Creation_Count < 2) do begin
             Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 3, 1);
-            Scenery_Creation := create_object_sid(33554499, Scenery_Creation_Hex, elevation(dude_obj), -1);
+            Scenery_Creation := create_object(33554499, Scenery_Creation_Hex, elevation(dude_obj));
             if (tile_num(Scenery_Creation) != Scenery_Creation_Hex) then begin
                 destroy_object(Scenery_Creation);
             end
@@ -126,7 +126,7 @@ begin
         Scenery_Creation_Count := 0;
         while (Scenery_Creation_Count < 2) do begin
             Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 4, 1);
-            Scenery_Creation := create_object_sid(33554499, Scenery_Creation_Hex, elevation(dude_obj), -1);
+            Scenery_Creation := create_object(33554499, Scenery_Creation_Hex, elevation(dude_obj));
             if (tile_num(Scenery_Creation) != Scenery_Creation_Hex) then begin
                 destroy_object(Scenery_Creation);
             end
@@ -135,7 +135,7 @@ begin
         Scenery_Creation_Count := 0;
         while (Scenery_Creation_Count < 1) do begin
             Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 5, 1);
-            Scenery_Creation := create_object_sid(33554499, Scenery_Creation_Hex, elevation(dude_obj), -1);
+            Scenery_Creation := create_object(33554499, Scenery_Creation_Hex, elevation(dude_obj));
             if (tile_num(Scenery_Creation) != Scenery_Creation_Hex) then begin
                 destroy_object(Scenery_Creation);
             end
@@ -144,7 +144,7 @@ begin
         Scenery_Creation_Count := 0;
         while (Scenery_Creation_Count < 1) do begin
             Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 4, 1);
-            Scenery_Creation := create_object_sid(33554499, Scenery_Creation_Hex, elevation(dude_obj), -1);
+            Scenery_Creation := create_object(33554499, Scenery_Creation_Hex, elevation(dude_obj));
             if (tile_num(Scenery_Creation) != Scenery_Creation_Hex) then begin
                 destroy_object(Scenery_Creation);
             end
@@ -153,7 +153,7 @@ begin
         Scenery_Creation_Count := 0;
         while (Scenery_Creation_Count < 1) do begin
             Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 5, 1);
-            Scenery_Creation := create_object_sid(33554499, Scenery_Creation_Hex, elevation(dude_obj), -1);
+            Scenery_Creation := create_object(33554499, Scenery_Creation_Hex, elevation(dude_obj));
             if (tile_num(Scenery_Creation) != Scenery_Creation_Hex) then begin
                 destroy_object(Scenery_Creation);
             end
@@ -162,7 +162,7 @@ begin
         Scenery_Creation_Count := 0;
         while (Scenery_Creation_Count < 1) do begin
             Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 4, 1);
-            Scenery_Creation := create_object_sid(33554499, Scenery_Creation_Hex, elevation(dude_obj), -1);
+            Scenery_Creation := create_object(33554499, Scenery_Creation_Hex, elevation(dude_obj));
             if (tile_num(Scenery_Creation) != Scenery_Creation_Hex) then begin
                 destroy_object(Scenery_Creation);
             end
@@ -171,7 +171,7 @@ begin
         Scenery_Creation_Count := 0;
         while (Scenery_Creation_Count < 1) do begin
             Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 5, 1);
-            Scenery_Creation := create_object_sid(33554499, Scenery_Creation_Hex, elevation(dude_obj), -1);
+            Scenery_Creation := create_object(33554499, Scenery_Creation_Hex, elevation(dude_obj));
             if (tile_num(Scenery_Creation) != Scenery_Creation_Hex) then begin
                 destroy_object(Scenery_Creation);
             end
@@ -180,7 +180,7 @@ begin
         Scenery_Creation_Count := 0;
         while (Scenery_Creation_Count < 1) do begin
             Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 4, 1);
-            Scenery_Creation := create_object_sid(33554499, Scenery_Creation_Hex, elevation(dude_obj), -1);
+            Scenery_Creation := create_object(33554499, Scenery_Creation_Hex, elevation(dude_obj));
             if (tile_num(Scenery_Creation) != Scenery_Creation_Hex) then begin
                 destroy_object(Scenery_Creation);
             end
@@ -189,7 +189,7 @@ begin
         Scenery_Creation_Count := 0;
         while (Scenery_Creation_Count < 1) do begin
             Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 0, 1);
-            Scenery_Creation := create_object_sid(33554499, Scenery_Creation_Hex, elevation(dude_obj), -1);
+            Scenery_Creation := create_object(33554499, Scenery_Creation_Hex, elevation(dude_obj));
             if (tile_num(Scenery_Creation) != Scenery_Creation_Hex) then begin
                 destroy_object(Scenery_Creation);
             end
@@ -198,7 +198,7 @@ begin
         Scenery_Creation_Count := 0;
         while (Scenery_Creation_Count < 1) do begin
             Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 5, 1);
-            Scenery_Creation := create_object_sid(33554499, Scenery_Creation_Hex, elevation(dude_obj), -1);
+            Scenery_Creation := create_object(33554499, Scenery_Creation_Hex, elevation(dude_obj));
             if (tile_num(Scenery_Creation) != Scenery_Creation_Hex) then begin
                 destroy_object(Scenery_Creation);
             end
@@ -207,7 +207,7 @@ begin
         Scenery_Creation_Count := 0;
         while (Scenery_Creation_Count < 2) do begin
             Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 0, 1);
-            Scenery_Creation := create_object_sid(33554499, Scenery_Creation_Hex, elevation(dude_obj), -1);
+            Scenery_Creation := create_object(33554499, Scenery_Creation_Hex, elevation(dude_obj));
             if (tile_num(Scenery_Creation) != Scenery_Creation_Hex) then begin
                 destroy_object(Scenery_Creation);
             end
@@ -216,7 +216,7 @@ begin
         Scenery_Creation_Count := 0;
         while (Scenery_Creation_Count < 1) do begin
             Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 1, 1);
-            Scenery_Creation := create_object_sid(33554499, Scenery_Creation_Hex, elevation(dude_obj), -1);
+            Scenery_Creation := create_object(33554499, Scenery_Creation_Hex, elevation(dude_obj));
             if (tile_num(Scenery_Creation) != Scenery_Creation_Hex) then begin
                 destroy_object(Scenery_Creation);
             end
@@ -225,7 +225,7 @@ begin
         Scenery_Creation_Count := 0;
         while (Scenery_Creation_Count < 1) do begin
             Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 2, 1);
-            Scenery_Creation := create_object_sid(33554499, Scenery_Creation_Hex, elevation(dude_obj), -1);
+            Scenery_Creation := create_object(33554499, Scenery_Creation_Hex, elevation(dude_obj));
             if (tile_num(Scenery_Creation) != Scenery_Creation_Hex) then begin
                 destroy_object(Scenery_Creation);
             end
@@ -235,12 +235,12 @@ begin
     trunk_tile := tile_num_in_direction(tile_num_in_direction(arg0, 5, 2), 4, 1);
     if (tile_contains_pid_obj(arg0, elevation(dude_obj), 33556384) != 0) then begin
         Scenery_Creation_Ptr := 0;
-        if (global_var(1644) == 0) then begin
+        if (global_var(GVAR_CAR_CURRENT_ELEVATION) == 0) then begin
             if (party_member_obj(831) != 0) then begin
                 Scenery_Creation_Ptr := party_member_obj(831);
             end
             else begin
-                Scenery_Creation_Ptr := create_object_sid(831, 0, 0, 920);
+                Scenery_Creation_Ptr := create_object_sid(831, 0, 0, SCRIPT_ZICRTRNK);
                 party_add(Scenery_Creation_Ptr);
             end
         end
@@ -249,12 +249,12 @@ begin
         end
     end
     if (tile_contains_pid_obj(arg0, elevation(dude_obj), 33555441) != 0) then begin
-        if (global_var(1644) == 1) then begin
+        if (global_var(GVAR_CAR_CURRENT_ELEVATION) == 1) then begin
             if (party_member_obj(455) != 0) then begin
                 Scenery_Creation_Ptr := party_member_obj(455);
             end
             else begin
-                Scenery_Creation_Ptr := create_object_sid(455, 0, 0, 920);
+                Scenery_Creation_Ptr := create_object_sid(455, 0, 0, SCRIPT_ZICRTRNK);
                 party_add(Scenery_Creation_Ptr);
             end
         end
@@ -263,12 +263,12 @@ begin
         end
     end
     if (tile_contains_pid_obj(arg0, elevation(dude_obj), 33556385) != 0) then begin
-        if (global_var(1644) == 2) then begin
+        if (global_var(GVAR_CAR_CURRENT_ELEVATION) == 2) then begin
             if (party_member_obj(832) != 0) then begin
                 Scenery_Creation_Ptr := party_member_obj(832);
             end
             else begin
-                Scenery_Creation_Ptr := create_object_sid(832, 0, 0, 920);
+                Scenery_Creation_Ptr := create_object_sid(832, 0, 0, SCRIPT_ZICRTRNK);
                 party_add(Scenery_Creation_Ptr);
             end
         end
@@ -277,12 +277,12 @@ begin
         end
     end
     if (tile_contains_pid_obj(arg0, elevation(dude_obj), 33556386) != 0) then begin
-        if (global_var(1644) == 3) then begin
+        if (global_var(GVAR_CAR_CURRENT_ELEVATION) == 3) then begin
             if (party_member_obj(833) != 0) then begin
                 Scenery_Creation_Ptr := party_member_obj(833);
             end
             else begin
-                Scenery_Creation_Ptr := create_object_sid(833, 0, 0, 920);
+                Scenery_Creation_Ptr := create_object_sid(833, 0, 0, SCRIPT_ZICRTRNK);
                 party_add(Scenery_Creation_Ptr);
             end
         end
@@ -291,12 +291,12 @@ begin
         end
     end
     if (tile_contains_pid_obj(arg0, elevation(dude_obj), 33556387) != 0) then begin
-        if (global_var(1644) == 4) then begin
+        if (global_var(GVAR_CAR_CURRENT_ELEVATION) == 4) then begin
             if (party_member_obj(834) != 0) then begin
                 Scenery_Creation_Ptr := party_member_obj(834);
             end
             else begin
-                Scenery_Creation_Ptr := create_object_sid(834, 0, 0, 920);
+                Scenery_Creation_Ptr := create_object_sid(834, 0, 0, SCRIPT_ZICRTRNK);
                 party_add(Scenery_Creation_Ptr);
             end
         end
@@ -305,12 +305,12 @@ begin
         end
     end
     if (tile_contains_pid_obj(arg0, elevation(dude_obj), 33556388) != 0) then begin
-        if (global_var(1644) == 5) then begin
+        if (global_var(GVAR_CAR_CURRENT_ELEVATION) == 5) then begin
             if (party_member_obj(835) != 0) then begin
                 Scenery_Creation_Ptr := party_member_obj(835);
             end
             else begin
-                Scenery_Creation_Ptr := create_object_sid(835, 0, 0, 920);
+                Scenery_Creation_Ptr := create_object_sid(835, 0, 0, SCRIPT_ZICRTRNK);
                 party_add(Scenery_Creation_Ptr);
             end
         end
@@ -319,12 +319,12 @@ begin
         end
     end
     if (tile_contains_pid_obj(arg0, elevation(dude_obj), 33556389) != 0) then begin
-        if (global_var(1644) == 6) then begin
+        if (global_var(GVAR_CAR_CURRENT_ELEVATION) == 6) then begin
             if (party_member_obj(836) != 0) then begin
                 Scenery_Creation_Ptr := party_member_obj(836);
             end
             else begin
-                Scenery_Creation_Ptr := create_object_sid(836, 0, 0, 920);
+                Scenery_Creation_Ptr := create_object_sid(836, 0, 0, SCRIPT_ZICRTRNK);
                 party_add(Scenery_Creation_Ptr);
             end
         end
@@ -338,7 +338,7 @@ begin
         Scenery_Creation_Count := 0;
         while (Scenery_Creation_Count < 1) do begin
             Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 1, 1);
-            Scenery_Creation := create_object_sid(33554499, Scenery_Creation_Hex, elevation(dude_obj), -1);
+            Scenery_Creation := create_object(33554499, Scenery_Creation_Hex, elevation(dude_obj));
             if (tile_num(Scenery_Creation) != Scenery_Creation_Hex) then begin
                 destroy_object(Scenery_Creation);
             end
@@ -347,17 +347,17 @@ begin
         Scenery_Creation_Count := 0;
         while (Scenery_Creation_Count < 1) do begin
             Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 2, 1);
-            Scenery_Creation := create_object_sid(33554499, Scenery_Creation_Hex, elevation(dude_obj), -1);
+            Scenery_Creation := create_object(33554499, Scenery_Creation_Hex, elevation(dude_obj));
             if (tile_num(Scenery_Creation) != Scenery_Creation_Hex) then begin
                 destroy_object(Scenery_Creation);
             end
             Scenery_Creation_Count := Scenery_Creation_Count + 1;
         end
     end
-    set_global_var(633, arg0);
-    set_global_var(1646, elevation(dude_obj));
-    while (global_var(1645) != cur_map_index) do begin
-        set_global_var(1645, cur_map_index);
+    set_global_var(GVAR_CAR_PLACED_TILE, arg0);
+    set_global_var(GVAR_CAR_HAS_BEEN_DRIVEN, elevation(dude_obj));
+    while (global_var(GVAR_CAR_CURRENT_TOWN) != cur_map_index) do begin
+        set_global_var(GVAR_CAR_CURRENT_TOWN, cur_map_index);
     end
 end
 
@@ -610,8 +610,8 @@ begin
             destroy_object(Scenery_Creation_Ptr);
         end
     end
-    set_global_var(633, -1);
-    set_global_var(1645, -1);
+    set_global_var(GVAR_CAR_PLACED_TILE, -1);
+    set_global_var(GVAR_CAR_CURRENT_TOWN, -1);
 end
 
 procedure checkPartyMembersNearDoor
@@ -726,31 +726,31 @@ end
 procedure map_enter_p_proc
 begin
     set_light_level(50);
-    if ((map_var(0) == 0) and (global_var(8) != 0) and (metarule(30, 0) == 11)) then begin
+    if ((map_var(0) == 0) and (global_var(GVAR_KNOWLEDGE_HEALING_POWDER) != 0) and (car_current_town == 11)) then begin
         set_map_var(0, 1);
-        if (not(metarule(22, 0))) then begin
+        if (not(is_loading_game)) then begin
             call place_car(19500);
         end
     end
     else begin
-        if ((map_var(0) == 0) and (metarule(30, 0) != 11)) then begin
+        if ((map_var(0) == 0) and (car_current_town != 11)) then begin
             set_map_var(0, 1);
-            if (not(metarule(22, 0))) then begin
+            if (not(is_loading_game)) then begin
                 call place_car(19500);
             end
         end
     end
-    if (global_var(633) != 19500) then begin
+    if (global_var(GVAR_CAR_PLACED_TILE) != 19500) then begin
         call delete_car(19500);
     end
-    if (not(metarule(22, 0))) then begin
-        if (global_var(213) == 1) then begin
-            set_global_var(213, 2);
-            if (metarule(17, 45) != 0) then begin
+    if (not(is_loading_game)) then begin
+        if (global_var(GVAR_NCR_FAKE_VAULT13_MAP) == 1) then begin
+            set_global_var(GVAR_NCR_FAKE_VAULT13_MAP, 2);
+            if (town_known(45) != 0) then begin
                 debug_msg(" unmark_on_map(" + 45 + ")");
                 mark_area_known(0, 45, 0);
             end
-            if (metarule(17, 46) == 0) then begin
+            if (town_known(46) == 0) then begin
                 debug_msg("  mark_on_map(" + 46 + ")");
                 mark_area_known(0, 46, 1);
             end
@@ -762,4 +762,3 @@ procedure map_update_p_proc
 begin
     set_light_level(50);
 end
-
