diff --git a/Mods/MegaMod/Original/CAVE7.ssl b/Mods/MegaMod/ReDefine/CAVE7.ssl
index eb15c4b..de99c2c 100644
--- a/Mods/MegaMod/Original/CAVE7.ssl
+++ b/Mods/MegaMod/ReDefine/CAVE7.ssl
@@ -138,16 +138,15 @@ end
 procedure map_enter_p_proc
 begin
     set_light_level(50);
-    if (not(metarule(22, 0))) then begin
+    if (not(is_loading_game)) then begin
         override_map_start(26148 % 200, 26148 / 200, 0, 1);
     end
     debug_msg("CAVE7: here I am");
-    display_msg(message_str(1146, 100));
-    set_global_var(27, 0);
+    display_mstr(100);
+    set_global_var(GVAR_LOAD_MAP_INDEX, 0);
 end
 
 procedure map_update_p_proc
 begin
     set_light_level(50);
 end
-
