diff --git a/Mods/MegaMod/Original/CAVE6.ssl b/Mods/MegaMod/ReDefine/CAVE6.ssl
index def7e7f..708df20 100644
--- a/Mods/MegaMod/Original/CAVE6.ssl
+++ b/Mods/MegaMod/ReDefine/CAVE6.ssl
@@ -138,10 +138,10 @@ end
 procedure map_enter_p_proc
 begin
     set_light_level(50);
-    if (not(metarule(22, 0))) then begin
-        if (global_var(213) == 1) then begin
-            set_global_var(213, 2);
-            if (metarule(17, 46) == 0) then begin
+    if (not(is_loading_game)) then begin
+        if (global_var(GVAR_NCR_FAKE_VAULT13_MAP) == 1) then begin
+            set_global_var(GVAR_NCR_FAKE_VAULT13_MAP, 2);
+            if (town_known(46) == 0) then begin
                 debug_msg("  mark_on_map(" + 46 + ")");
                 mark_area_known(0, 46, 1);
             end
@@ -149,12 +149,11 @@ begin
         override_map_start(26148 % 200, 26148 / 200, 0, 1);
     end
     debug_msg("CAVE6: here I am");
-    display_msg(message_str(1030, 100));
-    set_global_var(27, 0);
+    display_mstr(100);
+    set_global_var(GVAR_LOAD_MAP_INDEX, 0);
 end
 
 procedure map_update_p_proc
 begin
     set_light_level(50);
 end
-
