diff --git a/Mods/MegaMod/Original/CCCOMP2.ssl b/Mods/MegaMod/ReDefine/CCCOMP2.ssl
index d703247..3d39076 100644
--- a/Mods/MegaMod/Original/CCCOMP2.ssl
+++ b/Mods/MegaMod/ReDefine/CCCOMP2.ssl
@@ -42,7 +42,7 @@ end
 procedure critter_p_proc
 begin
     if (obj_can_see_obj(self_obj, dude_obj)) then begin
-        if ((metarule(16, 0) - 1 - (party_member_obj(16777687) != 0)) >= 1) then begin
+        if ((party_member_count(DONT_LIST_HIDDEN_MEMBERS) - 1 - (party_member_obj(16777687) != 0)) >= 1) then begin
             if (said_intruder == 0) then begin
                 said_intruder := 1;
                 call Node008();
@@ -62,7 +62,7 @@ procedure talk_p_proc
 begin
     if (local_var(2) == 0) then begin
         set_local_var(2, 1);
-        set_local_var(3, (get_critter_stat(dude_obj, 3) - get_critter_stat(self_obj, 3)) * 5);
+        set_local_var(3, (dude_charisma - self_charisma) * 5);
         debug_msg("Base Reaction == " + local_var(3));
     end
     Static_Reaction := 0;
@@ -74,14 +74,14 @@ begin
     else begin
         reaction_bonus_karma := -1 * (-1 * reaction_bonus_karma / 1000);
     end
-    if (global_var(628) >= 0) then begin
-        reaction_bonus_town_rep := global_var(628) / 2;
+    if (global_var(GVAR_TOWN_REP_NAVARRO) >= 0) then begin
+        reaction_bonus_town_rep := global_var(GVAR_TOWN_REP_NAVARRO) / 2;
     end
     else begin
-        reaction_bonus_town_rep := -1 * (-1 * global_var(628) / 2);
+        reaction_bonus_town_rep := -1 * (-1 * global_var(GVAR_TOWN_REP_NAVARRO) / 2);
     end
-    if (has_trait(2, dude_obj, 13)) then begin
-        if (get_critter_stat(dude_obj, 34) == get_critter_stat(self_obj, 34)) then begin
+    if (has_trait(TRAIT_TRAIT, dude_obj, TRAIT_sex_appeal)) then begin
+        if (dude_gender == self_gender) then begin
             Static_Reaction := Static_Reaction + -20;
         end
         else begin
@@ -107,52 +107,52 @@ begin
             Static_Reaction := Static_Reaction + reaction_bonus_town_rep;
         end
         debug_msg("Town Rep Bonus == " + Static_Reaction);
-        set_global_var(37, 0);
-        set_global_var(38, 0);
-        set_global_var(39, 0);
-        set_global_var(40, 0);
-        set_global_var(41, 0);
-        set_global_var(42, 0);
-        set_global_var(43, 0);
-        set_global_var(44, 0);
-        set_global_var(45, 0);
-        tmp_gen_rep := global_var(0);
-        if (has_trait(0, dude_obj, 95)) then begin
-            tmp_gen_rep := global_var(0) * 2;
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+        set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+        set_global_var(GVAR_KARMA_DEFENDER, 0);
+        set_global_var(GVAR_KARMA_WANDERER, 0);
+        set_global_var(GVAR_KARMA_BETRAYER, 0);
+        set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+        set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+        tmp_gen_rep := global_var(GVAR_PLAYER_REPUTATION);
+        if (has_trait(TRAIT_PERK, dude_obj, PERK_karma_beacon_perk)) then begin
+            tmp_gen_rep := global_var(GVAR_PLAYER_REPUTATION) * 2;
         end
         if (tmp_gen_rep >= 1000) then begin
-            set_global_var(37, 1);
+            set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
         end
         else begin
             if (tmp_gen_rep >= 750) then begin
-                set_global_var(38, 1);
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
             end
             else begin
                 if (tmp_gen_rep >= 500) then begin
-                    set_global_var(39, 1);
+                    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                 end
                 else begin
                     if (tmp_gen_rep >= 250) then begin
-                        set_global_var(40, 1);
+                        set_global_var(GVAR_KARMA_DEFENDER, 1);
                     end
                     else begin
                         if (tmp_gen_rep > -250) then begin
-                            set_global_var(41, 1);
+                            set_global_var(GVAR_KARMA_WANDERER, 1);
                         end
                         else begin
                             if (tmp_gen_rep > -500) then begin
-                                set_global_var(42, 1);
+                                set_global_var(GVAR_KARMA_BETRAYER, 1);
                             end
                             else begin
                                 if (tmp_gen_rep > -750) then begin
-                                    set_global_var(43, 1);
+                                    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                 end
                                 else begin
                                     if (tmp_gen_rep > -1000) then begin
-                                        set_global_var(44, 1);
+                                        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                     end
                                     else begin
-                                        set_global_var(45, 1);
+                                        set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                     end
                                 end
                             end
@@ -161,35 +161,35 @@ begin
                 end
             end
         end
-        if (global_var(37) == 1) then begin
+        if (global_var(GVAR_KARMA_HOLY_WARRIOR) == 1) then begin
             Static_Reaction := Static_Reaction - 20;
         end
         else begin
-            if (global_var(38) == 1) then begin
+            if (global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES) == 1) then begin
                 Static_Reaction := Static_Reaction - 15;
             end
             else begin
-                if (global_var(39) == 1) then begin
+                if (global_var(GVAR_KARMA_SHIELD_OF_HOPE) == 1) then begin
                     Static_Reaction := Static_Reaction - 10;
                 end
                 else begin
-                    if (global_var(40) == 1) then begin
+                    if (global_var(GVAR_KARMA_DEFENDER) == 1) then begin
                         Static_Reaction := Static_Reaction - 5;
                     end
                     else begin
-                        if (global_var(41) == 1) then begin
+                        if (global_var(GVAR_KARMA_WANDERER) == 1) then begin
                             Static_Reaction := Static_Reaction - 0;
                         end
                         else begin
-                            if (global_var(42) == 1) then begin
+                            if (global_var(GVAR_KARMA_BETRAYER) == 1) then begin
                                 Static_Reaction := Static_Reaction + -5;
                             end
                             else begin
-                                if (global_var(43) == 1) then begin
+                                if (global_var(GVAR_KARMA_SWORD_OF_DESPAIR) == 1) then begin
                                     Static_Reaction := Static_Reaction + -10;
                                 end
                                 else begin
-                                    if (global_var(44) == 1) then begin
+                                    if (global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE) == 1) then begin
                                         Static_Reaction := Static_Reaction + -15;
                                     end
                                     else begin
@@ -209,52 +209,52 @@ begin
         debug_msg("Karma Reaction Bonus == " + Static_Reaction);
         Static_Reaction := Static_Reaction + reaction_bonus_town_rep;
         debug_msg("Town Rep Bonus == " + Static_Reaction);
-        set_global_var(37, 0);
-        set_global_var(38, 0);
-        set_global_var(39, 0);
-        set_global_var(40, 0);
-        set_global_var(41, 0);
-        set_global_var(42, 0);
-        set_global_var(43, 0);
-        set_global_var(44, 0);
-        set_global_var(45, 0);
-        tmp_gen_rep := global_var(0);
-        if (has_trait(0, dude_obj, 95)) then begin
-            tmp_gen_rep := global_var(0) * 2;
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+        set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+        set_global_var(GVAR_KARMA_DEFENDER, 0);
+        set_global_var(GVAR_KARMA_WANDERER, 0);
+        set_global_var(GVAR_KARMA_BETRAYER, 0);
+        set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+        set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+        tmp_gen_rep := global_var(GVAR_PLAYER_REPUTATION);
+        if (has_trait(TRAIT_PERK, dude_obj, PERK_karma_beacon_perk)) then begin
+            tmp_gen_rep := global_var(GVAR_PLAYER_REPUTATION) * 2;
         end
         if (tmp_gen_rep >= 1000) then begin
-            set_global_var(37, 1);
+            set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
         end
         else begin
             if (tmp_gen_rep >= 750) then begin
-                set_global_var(38, 1);
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
             end
             else begin
                 if (tmp_gen_rep >= 500) then begin
-                    set_global_var(39, 1);
+                    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                 end
                 else begin
                     if (tmp_gen_rep >= 250) then begin
-                        set_global_var(40, 1);
+                        set_global_var(GVAR_KARMA_DEFENDER, 1);
                     end
                     else begin
                         if (tmp_gen_rep > -250) then begin
-                            set_global_var(41, 1);
+                            set_global_var(GVAR_KARMA_WANDERER, 1);
                         end
                         else begin
                             if (tmp_gen_rep > -500) then begin
-                                set_global_var(42, 1);
+                                set_global_var(GVAR_KARMA_BETRAYER, 1);
                             end
                             else begin
                                 if (tmp_gen_rep > -750) then begin
-                                    set_global_var(43, 1);
+                                    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                 end
                                 else begin
                                     if (tmp_gen_rep > -1000) then begin
-                                        set_global_var(44, 1);
+                                        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                     end
                                     else begin
-                                        set_global_var(45, 1);
+                                        set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                     end
                                 end
                             end
@@ -263,35 +263,35 @@ begin
                 end
             end
         end
-        if (global_var(37) == 1) then begin
+        if (global_var(GVAR_KARMA_HOLY_WARRIOR) == 1) then begin
             Static_Reaction := Static_Reaction + 20;
         end
         else begin
-            if (global_var(38) == 1) then begin
+            if (global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES) == 1) then begin
                 Static_Reaction := Static_Reaction + 15;
             end
             else begin
-                if (global_var(39) == 1) then begin
+                if (global_var(GVAR_KARMA_SHIELD_OF_HOPE) == 1) then begin
                     Static_Reaction := Static_Reaction + 10;
                 end
                 else begin
-                    if (global_var(40) == 1) then begin
+                    if (global_var(GVAR_KARMA_DEFENDER) == 1) then begin
                         Static_Reaction := Static_Reaction + 5;
                     end
                     else begin
-                        if (global_var(41) == 1) then begin
+                        if (global_var(GVAR_KARMA_WANDERER) == 1) then begin
                             Static_Reaction := Static_Reaction + 0;
                         end
                         else begin
-                            if (global_var(42) == 1) then begin
+                            if (global_var(GVAR_KARMA_BETRAYER) == 1) then begin
                                 Static_Reaction := Static_Reaction + -5;
                             end
                             else begin
-                                if (global_var(43) == 1) then begin
+                                if (global_var(GVAR_KARMA_SWORD_OF_DESPAIR) == 1) then begin
                                     Static_Reaction := Static_Reaction + -10;
                                 end
                                 else begin
-                                    if (global_var(44) == 1) then begin
+                                    if (global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE) == 1) then begin
                                         Static_Reaction := Static_Reaction + -15;
                                     end
                                     else begin
@@ -307,32 +307,32 @@ begin
         debug_msg("Good Person Reaction Bonus == " + Static_Reaction);
     end
     if (Slavery_Tolerant == 0) then begin
-        if (global_var(11) == 1) then begin
+        if (global_var(GVAR_REPUTATION_SLAVER) == 1) then begin
             Static_Reaction := Static_Reaction + (2 * -25);
         end
     end
     else begin
         if (Slavery_Tolerant == 1) then begin
-            if (global_var(11) == 1) then begin
+            if (global_var(GVAR_REPUTATION_SLAVER) == 1) then begin
                 Static_Reaction := Static_Reaction + -25;
             end
         end
         else begin
             if (Slavery_Tolerant == 3) then begin
-                if (global_var(11) == 1) then begin
+                if (global_var(GVAR_REPUTATION_SLAVER) == 1) then begin
                     Static_Reaction := Static_Reaction - -25;
                 end
             end
         end
     end
     debug_msg("Slaver + Aligned Reaction == " + Static_Reaction);
-    Static_Reaction := Static_Reaction + (10 * has_trait(0, dude_obj, 10));
+    Static_Reaction := Static_Reaction + (10 * has_trait(TRAIT_PERK, dude_obj, PERK_presence));
     debug_msg("Presence Reaction == " + Static_Reaction);
-    if (global_var(1) >= 2) then begin
+    if (global_var(GVAR_CHILDKILLER_REPUTATION) >= 2) then begin
         Static_Reaction := Static_Reaction + -30;
     end
     debug_msg("Childkiller Reaction == " + Static_Reaction);
-    if (has_trait(0, dude_obj, 39)) then begin
+    if (has_trait(TRAIT_PERK, dude_obj, PERK_cult_of_personality)) then begin
         if (Evil_Critter == 1) then begin
             if (Static_Reaction > 0) then begin
                 Static_Reaction := Static_Reaction * -1;
@@ -346,15 +346,15 @@ begin
     end
     set_local_var(0, local_var(3) + Static_Reaction);
     debug_msg("Initial Reaction == " + local_var(0));
-    if (global_var(511) > 0) then begin
+    if (global_var(GVAR_NAVARRO_BASE_ALERT) > 0) then begin
         call Node009();
     end
     else begin
-        if ((metarule(16, 0) - 1 - (party_member_obj(16777687) != 0)) >= 1) then begin
+        if ((party_member_count(DONT_LIST_HIDDEN_MEMBERS) - 1 - (party_member_obj(16777687) != 0)) >= 1) then begin
             call Node008();
         end
         else begin
-            if ((obj_pid(critter_inven_obj(dude_obj, 0)) != 3) and (obj_pid(critter_inven_obj(dude_obj, 0)) != 232) and (obj_pid(critter_inven_obj(dude_obj, 0)) != 348) and (obj_pid(critter_inven_obj(dude_obj, 0)) != 349)) then begin
+            if ((obj_pid(critter_inven_obj(dude_obj, INVEN_TYPE_WORN)) != 3) and (obj_pid(critter_inven_obj(dude_obj, INVEN_TYPE_WORN)) != 232) and (obj_pid(critter_inven_obj(dude_obj, INVEN_TYPE_WORN)) != 348) and (obj_pid(critter_inven_obj(dude_obj, INVEN_TYPE_WORN)) != 349)) then begin
                 call Node007();
             end
             else begin
@@ -371,63 +371,63 @@ end
 procedure destroy_p_proc
 begin
     if (source_obj == dude_obj) then begin
-        set_global_var(4, global_var(4) + 1);
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(5) > (2 * global_var(4))) or (global_var(3) == 1))) then begin
-            set_global_var(3, 1);
-            set_global_var(2, 0);
+        set_global_var(GVAR_BAD_MONSTER, global_var(GVAR_BAD_MONSTER) + 1);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
+            set_global_var(GVAR_BERSERKER_REPUTATION, 1);
+            set_global_var(GVAR_CHAMPION_REPUTATION, 0);
         end
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(4) > (3 * global_var(5))) or (global_var(2) == 1)) and (global_var(1) == 0)) then begin
-            set_global_var(2, 1);
-            set_global_var(3, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(4) > (3 * global_var(GVAR_GOOD_MONSTER))) or (global_var(GVAR_CHAMPION_REPUTATION) == 1)) and (global_var(GVAR_CHILDKILLER_REPUTATION) == 0)) then begin
+            set_global_var(GVAR_CHAMPION_REPUTATION, 1);
+            set_global_var(GVAR_BERSERKER_REPUTATION, 0);
         end
-        set_global_var(0, global_var(0) + 5);
+        set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + 5);
         debug_msg("Player gains " + 5 + " Karma Points.");
-        set_global_var(37, 0);
-        set_global_var(38, 0);
-        set_global_var(39, 0);
-        set_global_var(40, 0);
-        set_global_var(41, 0);
-        set_global_var(42, 0);
-        set_global_var(43, 0);
-        set_global_var(44, 0);
-        set_global_var(45, 0);
-        tmp_gen_rep := global_var(0);
-        if (has_trait(0, dude_obj, 95)) then begin
-            tmp_gen_rep := global_var(0) * 2;
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+        set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+        set_global_var(GVAR_KARMA_DEFENDER, 0);
+        set_global_var(GVAR_KARMA_WANDERER, 0);
+        set_global_var(GVAR_KARMA_BETRAYER, 0);
+        set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+        set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+        tmp_gen_rep := global_var(GVAR_PLAYER_REPUTATION);
+        if (has_trait(TRAIT_PERK, dude_obj, PERK_karma_beacon_perk)) then begin
+            tmp_gen_rep := global_var(GVAR_PLAYER_REPUTATION) * 2;
         end
         if (tmp_gen_rep >= 1000) then begin
-            set_global_var(37, 1);
+            set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
         end
         else begin
             if (tmp_gen_rep >= 750) then begin
-                set_global_var(38, 1);
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
             end
             else begin
                 if (tmp_gen_rep >= 500) then begin
-                    set_global_var(39, 1);
+                    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                 end
                 else begin
                     if (tmp_gen_rep >= 250) then begin
-                        set_global_var(40, 1);
+                        set_global_var(GVAR_KARMA_DEFENDER, 1);
                     end
                     else begin
                         if (tmp_gen_rep > -250) then begin
-                            set_global_var(41, 1);
+                            set_global_var(GVAR_KARMA_WANDERER, 1);
                         end
                         else begin
                             if (tmp_gen_rep > -500) then begin
-                                set_global_var(42, 1);
+                                set_global_var(GVAR_KARMA_BETRAYER, 1);
                             end
                             else begin
                                 if (tmp_gen_rep > -750) then begin
-                                    set_global_var(43, 1);
+                                    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                 end
                                 else begin
                                     if (tmp_gen_rep > -1000) then begin
-                                        set_global_var(44, 1);
+                                        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                     end
                                     else begin
-                                        set_global_var(45, 1);
+                                        set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                     end
                                 end
                             end
@@ -436,7 +436,7 @@ begin
                 end
             end
         end
-        set_global_var(628, global_var(628) + 2);
+        set_global_var(GVAR_TOWN_REP_NAVARRO, global_var(GVAR_TOWN_REP_NAVARRO) + 2);
         debug_msg("Added " + 2 + " to Town Rep");
     end
 end
@@ -446,17 +446,17 @@ begin
     script_overrides;
     if (local_var(7) == 0) then begin
         set_local_var(7, 1);
-        display_msg(message_str(1068, 100));
+        display_mstr(100);
     end
     else begin
-        display_msg(message_str(1068, 101));
+        display_mstr(101);
     end
 end
 
 procedure description_p_proc
 begin
     script_overrides;
-    display_msg(message_str(1068, 102));
+    display_mstr(102);
 end
 
 procedure use_skill_on_p_proc
@@ -473,93 +473,92 @@ end
 procedure map_enter_p_proc
 begin
     Only_Once := 0;
-    critter_add_trait(self_obj, 1, 6, 153);
-    critter_add_trait(self_obj, 1, 5, 175);
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 153);
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_NAVARRO_GUARD);
 end
 
 procedure Node001
 begin
-    gsay_reply(1068, 103);
-    giq_option(-3, 1068, 104, Node002, 50);
-    giq_option(-3, 1068, 105, Node999, 50);
-    giq_option(4, 1068, 106, Node002, 50);
-    giq_option(4, 1068, 107, Node999, 50);
+    Reply(103);
+    giq_option(-3, NAME, 104, Node002, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 105, Node999, NEUTRAL_REACTION);
+    giq_option(4, NAME, 106, Node002, NEUTRAL_REACTION);
+    giq_option(4, NAME, 107, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node002
 begin
-    gsay_reply(1068, 108);
-    giq_option(-3, 1068, 109, Node003, 50);
-    giq_option(4, 1068, 110, Node003, 50);
+    Reply(108);
+    giq_option(-3, NAME, 109, Node003, NEUTRAL_REACTION);
+    giq_option(4, NAME, 110, Node003, NEUTRAL_REACTION);
 end
 
 procedure Node003
 begin
-    gsay_reply(1068, 111);
-    giq_option(-3, 1068, 112, Node004, 50);
-    giq_option(-3, 1068, 113, Node999, 50);
-    giq_option(4, 1068, 114, Node004, 50);
-    giq_option(4, 1068, 115, Node999, 50);
+    Reply(111);
+    giq_option(-3, NAME, 112, Node004, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 113, Node999, NEUTRAL_REACTION);
+    giq_option(4, NAME, 114, Node004, NEUTRAL_REACTION);
+    giq_option(4, NAME, 115, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node004
 begin
-    gsay_reply(1068, 116);
-    giq_option(-3, 1068, 117, Node005, 50);
-    giq_option(4, 1068, 118, Node005, 50);
+    Reply(116);
+    giq_option(-3, NAME, 117, Node005, NEUTRAL_REACTION);
+    giq_option(4, NAME, 118, Node005, NEUTRAL_REACTION);
 end
 
 procedure Node005
 begin
     if (is_success(roll_vs_skill(dude_obj, 14, -10))) then begin
         set_map_var(2, 1);
-        gsay_reply(1068, 119);
-        giq_option(-3, 1068, 121, Node006, 50);
-        giq_option(4, 1068, 123, Node006, 50);
+        Reply(119);
+        giq_option(-3, NAME, 121, Node006, NEUTRAL_REACTION);
+        giq_option(4, NAME, 123, Node006, NEUTRAL_REACTION);
     end
     else begin
-        gsay_reply(1068, 120);
-        giq_option(-3, 1068, 122, Node999, 50);
-        giq_option(4, 1068, 124, Node999, 50);
+        Reply(120);
+        giq_option(-3, NAME, 122, Node999, NEUTRAL_REACTION);
+        giq_option(4, NAME, 124, Node999, NEUTRAL_REACTION);
     end
 end
 
 procedure Node006
 begin
-    gsay_reply(1068, 125);
-    giq_option(-3, 1068, 126, Node999, 50);
-    giq_option(4, 1068, 127, Node999, 50);
+    Reply(125);
+    giq_option(-3, NAME, 126, Node999, NEUTRAL_REACTION);
+    giq_option(4, NAME, 127, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node007
 begin
-    float_msg(self_obj, message_str(1068, random(128, 130)), 8);
+    float_msg(self_obj, mstr(random(128, 130)), FLOAT_MSG_YELLOW);
 end
 
 procedure Node008
 begin
-    set_global_var(511, 1);
-    float_msg(self_obj, message_str(1068, random(131, 133)), 8);
+    set_global_var(GVAR_NAVARRO_BASE_ALERT, 1);
+    float_msg(self_obj, mstr(random(131, 133)), FLOAT_MSG_YELLOW);
 end
 
 procedure Node009
 begin
     said_intruder := 1;
-    float_msg(self_obj, message_str(1068, random(134, 136)), 8);
+    float_msg(self_obj, mstr(random(134, 136)), FLOAT_MSG_YELLOW);
 end
 
 procedure Node010
 begin
-    float_msg(self_obj, message_str(1068, random(137, 139)), 8);
+    float_msg(self_obj, mstr(random(137, 139)), FLOAT_MSG_YELLOW);
 end
 
 procedure Node998
 begin
     set_local_var(5, 2);
-    set_global_var(511, 1);
+    set_global_var(GVAR_NAVARRO_BASE_ALERT, 1);
 end
 
 procedure Node999
 begin
 end
-
