diff --git a/Mods/MegaMod/Original/CCFAK9.ssl b/Mods/MegaMod/ReDefine/CCFAK9.ssl
index fa6a300..cba77e0 100644
--- a/Mods/MegaMod/Original/CCFAK9.ssl
+++ b/Mods/MegaMod/ReDefine/CCFAK9.ssl
@@ -65,14 +65,14 @@ begin
     if (obj_can_see_obj(self_obj, dude_obj)) then begin
         if (local_var(0) == 2) then begin
             set_local_var(0, 1);
-            attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+            attack(dude_obj);
         end
     end
-    if ((global_var(1180) >= 1) and (get_critter_extra_stat(self_obj, 22) < 10)) then begin
+    if ((global_var(GVAR_MAKE_MAJOR_STIMS) >= 1) and (get_critter_extra_stat(self_obj, 22) < 10)) then begin
         set_critter_extra_stat(self_obj, 22, 10);
         set_critter_extra_stat(self_obj, 8, get_critter_extra_stat(self_obj, 8) + 1);
     end
-    if ((global_var(1180) == 2) and (get_critter_extra_stat(self_obj, 29) < 600)) then begin
+    if ((global_var(GVAR_MAKE_MAJOR_STIMS) == 2) and (get_critter_extra_stat(self_obj, 29) < 600)) then begin
         set_critter_extra_stat(self_obj, 29, 600);
         set_critter_extra_stat(self_obj, 8, get_critter_extra_stat(self_obj, 8) + 1);
     end
@@ -96,17 +96,17 @@ begin
     script_overrides;
     if (local_var(1) == 0) then begin
         set_local_var(1, 1);
-        display_msg(message_str(1057, 100));
+        display_msg(message_str(SCRIPT_CCK9, 100));
     end
     else begin
-        display_msg(message_str(1057, 101));
+        display_msg(message_str(SCRIPT_CCK9, 101));
     end
 end
 
 procedure description_p_proc
 begin
     script_overrides;
-    display_msg(message_str(1057, 102));
+    display_msg(message_str(SCRIPT_CCK9, 102));
 end
 
 procedure use_p_proc
@@ -176,15 +176,15 @@ begin
                     call Node800();
                 end
                 else begin
-                    float_msg(self_obj, message_str(1057, 155), 8);
+                    float_msg(self_obj, message_str(SCRIPT_CCK9, 155), FLOAT_MSG_YELLOW);
                 end
             end
             else begin
-                display_msg(message_str(1057, 400));
+                display_msg(message_str(SCRIPT_CCK9, 400));
             end
         end
         else begin
-            display_msg(message_str(1057, 410));
+            display_msg(message_str(SCRIPT_CCK9, 410));
         end
     end
 end
@@ -206,11 +206,11 @@ begin
                 call Node800();
             end
             else begin
-                float_msg(self_obj, message_str(1057, 155), 8);
+                float_msg(self_obj, message_str(SCRIPT_CCK9, 155), FLOAT_MSG_YELLOW);
             end
         end
         else begin
-            display_msg(message_str(1057, 420));
+            display_msg(message_str(SCRIPT_CCK9, 420));
         end
     end
 end
@@ -224,14 +224,14 @@ end
 
 procedure map_enter_p_proc
 begin
-    if ((cur_map_index == 19) or (cur_map_index == 37) or (cur_map_index == 38)) then begin
-        if ((elevation(self_obj) != elevation(dude_obj)) and (local_var(5) != 0)) then begin
-            move_to(self_obj, tile_num(self_obj), elevation(dude_obj));
+    if ((CUR_MAP_MODINN) or (CUR_MAP_V15ENT) or (CUR_MAP_V15ENTB)) then begin
+        if ((elevation(self_obj) != dude_elevation) and (local_var(5) != 0)) then begin
+            move_to(self_obj, tile_num(self_obj), dude_elevation);
         end
     end
     Only_Once := 0;
-    critter_add_trait(self_obj, 1, 6, 35);
-    critter_add_trait(self_obj, 1, 5, 50);
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 35);
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_CYBERDOG);
 end
 
 procedure map_update_p_proc
@@ -239,8 +239,8 @@ begin
     if (combat_is_initialized == 0) then begin
         if (anim_busy(self_obj) == 0) then begin
             if (party_member_obj(obj_pid(self_obj)) != 0) then begin
-                if (metarule3(109, self_obj, 0, 0) == 1) then begin
-                    if ((100 * get_critter_stat(self_obj, 35) / get_critter_stat(self_obj, 7)) <= 60) then begin
+                if (chem_use_level(self_obj) == 1) then begin
+                    if ((100 * get_critter_stat(self_obj, STAT_current_hp) / get_critter_stat(self_obj, STAT_max_hp)) <= 60) then begin
                         global_temp := 1;
                         if (obj_is_carrying_obj_pid(self_obj, 144)) then begin
                             if ((global_temp == 0) and (combat_is_initialized == 0)) then begin
@@ -252,7 +252,7 @@ begin
                             end
                             PartyHealingItem := obj_carrying_pid_obj(self_obj, 144);
                             use_obj_on_obj(PartyHealingItem, self_obj);
-                            debug_msg(obj_name(self_obj) + " used super stimpak.");
+                            debug_msg(self_name + " used super stimpak.");
                         end
                         else begin
                             if (obj_is_carrying_obj_pid(self_obj, 40)) then begin
@@ -264,7 +264,7 @@ begin
                                     gfade_in(600);
                                 end
                                 use_obj_on_obj(obj_carrying_pid_obj(self_obj, 40), self_obj);
-                                debug_msg(obj_name(self_obj) + " used stimpak.");
+                                debug_msg(self_name + " used stimpak.");
                             end
                             else begin
                                 if (obj_is_carrying_obj_pid(self_obj, 273)) then begin
@@ -276,7 +276,7 @@ begin
                                         gfade_in(600);
                                     end
                                     use_obj_on_obj(obj_carrying_pid_obj(self_obj, 273), self_obj);
-                                    debug_msg(obj_name(self_obj) + " used healing powder.");
+                                    debug_msg(self_name + " used healing powder.");
                                 end
                                 else begin
                                     if (obj_is_carrying_obj_pid(self_obj, 91)) then begin
@@ -288,7 +288,7 @@ begin
                                             gfade_in(600);
                                         end
                                         use_obj_on_obj(obj_carrying_pid_obj(self_obj, 91), self_obj);
-                                        debug_msg(obj_name(self_obj) + " used doctor's bag kit.");
+                                        debug_msg(self_name + " used doctor's bag kit.");
                                     end
                                     else begin
                                         if (obj_is_carrying_obj_pid(self_obj, 47)) then begin
@@ -300,7 +300,7 @@ begin
                                                 gfade_in(600);
                                             end
                                             use_obj_on_obj(obj_carrying_pid_obj(self_obj, 47), self_obj);
-                                            debug_msg(obj_name(self_obj) + " used First Aid kit.");
+                                            debug_msg(self_name + " used First Aid kit.");
                                         end
                                     end
                                 end
@@ -309,8 +309,8 @@ begin
                     end
                 end
                 else begin
-                    if (metarule3(109, self_obj, 0, 0) == 2) then begin
-                        if ((100 * get_critter_stat(self_obj, 35) / get_critter_stat(self_obj, 7)) <= 30) then begin
+                    if (chem_use_level(self_obj) == 2) then begin
+                        if ((100 * get_critter_stat(self_obj, STAT_current_hp) / get_critter_stat(self_obj, STAT_max_hp)) <= 30) then begin
                             global_temp := 1;
                             if (obj_is_carrying_obj_pid(self_obj, 144)) then begin
                                 if ((global_temp == 0) and (combat_is_initialized == 0)) then begin
@@ -322,7 +322,7 @@ begin
                                 end
                                 PartyHealingItem := obj_carrying_pid_obj(self_obj, 144);
                                 use_obj_on_obj(PartyHealingItem, self_obj);
-                                debug_msg(obj_name(self_obj) + " used super stimpak.");
+                                debug_msg(self_name + " used super stimpak.");
                             end
                             else begin
                                 if (obj_is_carrying_obj_pid(self_obj, 40)) then begin
@@ -334,7 +334,7 @@ begin
                                         gfade_in(600);
                                     end
                                     use_obj_on_obj(obj_carrying_pid_obj(self_obj, 40), self_obj);
-                                    debug_msg(obj_name(self_obj) + " used stimpak.");
+                                    debug_msg(self_name + " used stimpak.");
                                 end
                                 else begin
                                     if (obj_is_carrying_obj_pid(self_obj, 273)) then begin
@@ -346,7 +346,7 @@ begin
                                             gfade_in(600);
                                         end
                                         use_obj_on_obj(obj_carrying_pid_obj(self_obj, 273), self_obj);
-                                        debug_msg(obj_name(self_obj) + " used healing powder.");
+                                        debug_msg(self_name + " used healing powder.");
                                     end
                                     else begin
                                         if (obj_is_carrying_obj_pid(self_obj, 91)) then begin
@@ -358,7 +358,7 @@ begin
                                                 gfade_in(600);
                                             end
                                             use_obj_on_obj(obj_carrying_pid_obj(self_obj, 91), self_obj);
-                                            debug_msg(obj_name(self_obj) + " used doctor's bag kit.");
+                                            debug_msg(self_name + " used doctor's bag kit.");
                                         end
                                         else begin
                                             if (obj_is_carrying_obj_pid(self_obj, 47)) then begin
@@ -370,7 +370,7 @@ begin
                                                     gfade_in(600);
                                                 end
                                                 use_obj_on_obj(obj_carrying_pid_obj(self_obj, 47), self_obj);
-                                                debug_msg(obj_name(self_obj) + " used First Aid kit.");
+                                                debug_msg(self_name + " used First Aid kit.");
                                             end
                                         end
                                     end
@@ -379,8 +379,8 @@ begin
                         end
                     end
                     else begin
-                        if (metarule3(109, self_obj, 0, 0) == 3) then begin
-                            if ((100 * get_critter_stat(self_obj, 35) / get_critter_stat(self_obj, 7)) <= 50) then begin
+                        if (chem_use_level(self_obj) == 3) then begin
+                            if ((100 * get_critter_stat(self_obj, STAT_current_hp) / get_critter_stat(self_obj, STAT_max_hp)) <= 50) then begin
                                 global_temp := 1;
                                 if (obj_is_carrying_obj_pid(self_obj, 144)) then begin
                                     if ((global_temp == 0) and (combat_is_initialized == 0)) then begin
@@ -392,7 +392,7 @@ begin
                                     end
                                     PartyHealingItem := obj_carrying_pid_obj(self_obj, 144);
                                     use_obj_on_obj(PartyHealingItem, self_obj);
-                                    debug_msg(obj_name(self_obj) + " used super stimpak.");
+                                    debug_msg(self_name + " used super stimpak.");
                                 end
                                 else begin
                                     if (obj_is_carrying_obj_pid(self_obj, 40)) then begin
@@ -404,7 +404,7 @@ begin
                                             gfade_in(600);
                                         end
                                         use_obj_on_obj(obj_carrying_pid_obj(self_obj, 40), self_obj);
-                                        debug_msg(obj_name(self_obj) + " used stimpak.");
+                                        debug_msg(self_name + " used stimpak.");
                                     end
                                     else begin
                                         if (obj_is_carrying_obj_pid(self_obj, 273)) then begin
@@ -416,7 +416,7 @@ begin
                                                 gfade_in(600);
                                             end
                                             use_obj_on_obj(obj_carrying_pid_obj(self_obj, 273), self_obj);
-                                            debug_msg(obj_name(self_obj) + " used healing powder.");
+                                            debug_msg(self_name + " used healing powder.");
                                         end
                                         else begin
                                             if (obj_is_carrying_obj_pid(self_obj, 91)) then begin
@@ -428,7 +428,7 @@ begin
                                                     gfade_in(600);
                                                 end
                                                 use_obj_on_obj(obj_carrying_pid_obj(self_obj, 91), self_obj);
-                                                debug_msg(obj_name(self_obj) + " used doctor's bag kit.");
+                                                debug_msg(self_name + " used doctor's bag kit.");
                                             end
                                             else begin
                                                 if (obj_is_carrying_obj_pid(self_obj, 47)) then begin
@@ -440,7 +440,7 @@ begin
                                                         gfade_in(600);
                                                     end
                                                     use_obj_on_obj(obj_carrying_pid_obj(self_obj, 47), self_obj);
-                                                    debug_msg(obj_name(self_obj) + " used First Aid kit.");
+                                                    debug_msg(self_name + " used First Aid kit.");
                                                 end
                                             end
                                         end
@@ -449,8 +449,8 @@ begin
                             end
                         end
                         else begin
-                            if (metarule3(109, self_obj, 0, 0) == 4) then begin
-                                if ((100 * get_critter_stat(self_obj, 35) / get_critter_stat(self_obj, 7)) <= 50) then begin
+                            if (chem_use_level(self_obj) == 4) then begin
+                                if ((100 * get_critter_stat(self_obj, STAT_current_hp) / get_critter_stat(self_obj, STAT_max_hp)) <= 50) then begin
                                     global_temp := 1;
                                     if (obj_is_carrying_obj_pid(self_obj, 144)) then begin
                                         if ((global_temp == 0) and (combat_is_initialized == 0)) then begin
@@ -462,7 +462,7 @@ begin
                                         end
                                         PartyHealingItem := obj_carrying_pid_obj(self_obj, 144);
                                         use_obj_on_obj(PartyHealingItem, self_obj);
-                                        debug_msg(obj_name(self_obj) + " used super stimpak.");
+                                        debug_msg(self_name + " used super stimpak.");
                                     end
                                     else begin
                                         if (obj_is_carrying_obj_pid(self_obj, 40)) then begin
@@ -474,7 +474,7 @@ begin
                                                 gfade_in(600);
                                             end
                                             use_obj_on_obj(obj_carrying_pid_obj(self_obj, 40), self_obj);
-                                            debug_msg(obj_name(self_obj) + " used stimpak.");
+                                            debug_msg(self_name + " used stimpak.");
                                         end
                                         else begin
                                             if (obj_is_carrying_obj_pid(self_obj, 273)) then begin
@@ -486,7 +486,7 @@ begin
                                                     gfade_in(600);
                                                 end
                                                 use_obj_on_obj(obj_carrying_pid_obj(self_obj, 273), self_obj);
-                                                debug_msg(obj_name(self_obj) + " used healing powder.");
+                                                debug_msg(self_name + " used healing powder.");
                                             end
                                             else begin
                                                 if (obj_is_carrying_obj_pid(self_obj, 91)) then begin
@@ -498,7 +498,7 @@ begin
                                                         gfade_in(600);
                                                     end
                                                     use_obj_on_obj(obj_carrying_pid_obj(self_obj, 91), self_obj);
-                                                    debug_msg(obj_name(self_obj) + " used doctor's bag kit.");
+                                                    debug_msg(self_name + " used doctor's bag kit.");
                                                 end
                                                 else begin
                                                     if (obj_is_carrying_obj_pid(self_obj, 47)) then begin
@@ -510,7 +510,7 @@ begin
                                                             gfade_in(600);
                                                         end
                                                         use_obj_on_obj(obj_carrying_pid_obj(self_obj, 47), self_obj);
-                                                        debug_msg(obj_name(self_obj) + " used First Aid kit.");
+                                                        debug_msg(self_name + " used First Aid kit.");
                                                     end
                                                 end
                                             end
@@ -528,149 +528,149 @@ end
 
 procedure Node001
 begin
-    gsay_reply(1057, 103);
-    giq_option(-3, 1057, 104, Node002, 50);
-    giq_option(-3, 1057, 105, Node008, 50);
-    giq_option(-3, 1057, 106, Node010, 50);
-    giq_option(4, 1057, 107, Node002, 50);
-    giq_option(4, 1057, 292, Node008, 50);
-    giq_option(4, 1057, 293, Node010, 50);
+    gsay_reply(SCRIPT_CCK9, 103);
+    giq_option(-3, SCRIPT_CCK9, 104, Node002, NEUTRAL_REACTION);
+    giq_option(-3, SCRIPT_CCK9, 105, Node008, NEUTRAL_REACTION);
+    giq_option(-3, SCRIPT_CCK9, 106, Node010, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_CCK9, 107, Node002, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_CCK9, 292, Node008, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_CCK9, 293, Node010, NEUTRAL_REACTION);
 end
 
 procedure Node002
 begin
-    if (get_critter_stat(dude_obj, 4) < 4) then begin
-        gsay_reply(1057, 108);
+    if (dude_iq < 4) then begin
+        gsay_reply(SCRIPT_CCK9, 108);
     end
     else begin
-        gsay_reply(1057, 109);
+        gsay_reply(SCRIPT_CCK9, 109);
     end
-    giq_option(-3, 1057, 110, Node003, 50);
-    giq_option(1, 1057, 111, Node005, 50);
-    giq_option(1, 1057, 112, Node010, 50);
+    giq_option(-3, SCRIPT_CCK9, 110, Node003, NEUTRAL_REACTION);
+    giq_option(1, SCRIPT_CCK9, 111, Node005, NEUTRAL_REACTION);
+    giq_option(1, SCRIPT_CCK9, 112, Node010, NEUTRAL_REACTION);
 end
 
 procedure Node003
 begin
-    gsay_reply(1057, 113);
-    giq_option(-3, 1057, 114, Node004, 50);
-    giq_option(-3, 1057, 115, Node005, 50);
+    gsay_reply(SCRIPT_CCK9, 113);
+    giq_option(-3, SCRIPT_CCK9, 114, Node004, NEUTRAL_REACTION);
+    giq_option(-3, SCRIPT_CCK9, 115, Node005, NEUTRAL_REACTION);
 end
 
 procedure Node004
 begin
-    gsay_reply(1057, 116);
-    giq_option(-3, 1057, 117, Node005, 50);
+    gsay_reply(SCRIPT_CCK9, 116);
+    giq_option(-3, SCRIPT_CCK9, 117, Node005, NEUTRAL_REACTION);
 end
 
 procedure Node005
 begin
-    gsay_reply(1057, 118);
-    giq_option(-3, 1057, 119, Node006, 50);
-    giq_option(-3, 1057, 120, Node008, 50);
-    giq_option(-3, 1057, 121, Node010, 50);
-    giq_option(1, 1057, 122, Node008, 50);
-    giq_option(1, 1057, 123, Node010, 50);
+    gsay_reply(SCRIPT_CCK9, 118);
+    giq_option(-3, SCRIPT_CCK9, 119, Node006, NEUTRAL_REACTION);
+    giq_option(-3, SCRIPT_CCK9, 120, Node008, NEUTRAL_REACTION);
+    giq_option(-3, SCRIPT_CCK9, 121, Node010, NEUTRAL_REACTION);
+    giq_option(1, SCRIPT_CCK9, 122, Node008, NEUTRAL_REACTION);
+    giq_option(1, SCRIPT_CCK9, 123, Node010, NEUTRAL_REACTION);
 end
 
 procedure Node006
 begin
-    gsay_reply(1057, 124);
-    giq_option(-3, 1057, 125, Node007, 50);
+    gsay_reply(SCRIPT_CCK9, 124);
+    giq_option(-3, SCRIPT_CCK9, 125, Node007, NEUTRAL_REACTION);
 end
 
 procedure Node007
 begin
-    gsay_reply(1057, 126);
-    giq_option(-3, 1057, 127, Node008, 50);
-    giq_option(-3, 1057, 128, Node010, 50);
+    gsay_reply(SCRIPT_CCK9, 126);
+    giq_option(-3, SCRIPT_CCK9, 127, Node008, NEUTRAL_REACTION);
+    giq_option(-3, SCRIPT_CCK9, 128, Node010, NEUTRAL_REACTION);
 end
 
 procedure Node008
 begin
     set_local_var(9, 1);
-    gsay_reply(1057, 129);
-    giq_option(-3, 1057, 130, Node009, 50);
-    giq_option(4, 1057, 131, Node009, 50);
+    gsay_reply(SCRIPT_CCK9, 129);
+    giq_option(-3, SCRIPT_CCK9, 130, Node009, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_CCK9, 131, Node009, NEUTRAL_REACTION);
 end
 
 procedure Node009
 begin
-    if (global_var(513) == 0) then begin
-        set_global_var(513, 1);
+    if (global_var(GVAR_NAVARRO_K9) == 0) then begin
+        set_global_var(GVAR_NAVARRO_K9, 1);
     end
-    gsay_reply(1057, 132);
-    giq_option(-3, 1057, 133, Node999, 50);
-    giq_option(4, 1057, 134, Node999, 50);
+    gsay_reply(SCRIPT_CCK9, 132);
+    giq_option(-3, SCRIPT_CCK9, 133, Node999, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_CCK9, 134, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node010
 begin
-    gsay_reply(1057, 135);
-    giq_option(-3, 1057, 136, Node999, 50);
-    giq_option(4, 1057, 137, Node999, 50);
+    gsay_reply(SCRIPT_CCK9, 135);
+    giq_option(-3, SCRIPT_CCK9, 136, Node999, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_CCK9, 137, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node011
 begin
-    gsay_reply(1057, 138);
-    giq_option(-3, 1057, 139, Node008, 50);
-    giq_option(-3, 1057, 140, Node010, 50);
-    giq_option(4, 1057, 141, Node008, 50);
-    giq_option(4, 1057, 142, Node010, 50);
+    gsay_reply(SCRIPT_CCK9, 138);
+    giq_option(-3, SCRIPT_CCK9, 139, Node008, NEUTRAL_REACTION);
+    giq_option(-3, SCRIPT_CCK9, 140, Node010, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_CCK9, 141, Node008, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_CCK9, 142, Node010, NEUTRAL_REACTION);
 end
 
 procedure Node012
 begin
-    gsay_reply(1057, 143);
-    giq_option(1, 1057, 144, Node999, 50);
+    gsay_reply(SCRIPT_CCK9, 143);
+    giq_option(1, SCRIPT_CCK9, 144, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node013
 begin
-    gsay_reply(1057, 145);
+    gsay_reply(SCRIPT_CCK9, 145);
     if (obj_is_carrying_obj_pid(dude_obj, 488)) then begin
-        giq_option(-3, 1057, 147, Node015, 50);
-        giq_option(4, 1057, 149, Node015, 50);
+        giq_option(-3, SCRIPT_CCK9, 147, Node015, NEUTRAL_REACTION);
+        giq_option(4, SCRIPT_CCK9, 149, Node015, NEUTRAL_REACTION);
     end
     else begin
-        giq_option(-3, 1057, 146, Node014, 50);
-        giq_option(4, 1057, 148, Node014, 50);
+        giq_option(-3, SCRIPT_CCK9, 146, Node014, NEUTRAL_REACTION);
+        giq_option(4, SCRIPT_CCK9, 148, Node014, NEUTRAL_REACTION);
     end
 end
 
 procedure Node014
 begin
-    gsay_reply(1057, 150);
-    giq_option(4, 1057, 151, Node999, 50);
+    gsay_reply(SCRIPT_CCK9, 150);
+    giq_option(4, SCRIPT_CCK9, 151, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node015
 begin
-    gsay_reply(1057, 152);
-    giq_option(4, 1057, 153, Node999, 50);
+    gsay_reply(SCRIPT_CCK9, 152);
+    giq_option(4, SCRIPT_CCK9, 153, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node017
 begin
-    float_msg(self_obj, message_str(1057, random(160, 162)), 8);
+    float_msg(self_obj, message_str(SCRIPT_CCK9, random(160, 162)), FLOAT_MSG_YELLOW);
 end
 
 procedure Node018
 begin
-    float_msg(self_obj, message_str(1057, random(163, 165)), 8);
+    float_msg(self_obj, message_str(SCRIPT_CCK9, random(163, 165)), FLOAT_MSG_YELLOW);
 end
 
 procedure Node800
 begin
     if (local_var(8) == 0) then begin
         set_local_var(8, 1);
-        display_msg(message_str(1057, 800));
+        display_msg(message_str(SCRIPT_CCK9, 800));
         give_exp_points(3500);
-        display_msg(message_str(14, 100) + 3500 + message_str(14, 101));
-        set_global_var(513, 3);
-        dog := create_object_sid(16777687, 0, 0, 1057);
-        move_to(dog, 16914, elevation(dude_obj));
+        display_msg(g_mstr(100) + 3500 + g_mstr(101));
+        set_global_var(GVAR_NAVARRO_K9, 3);
+        dog := create_object_sid(16777687, 0, 0, SCRIPT_CCK9);
+        move_to(dog, 16914, dude_elevation);
         destroy_object(self_obj);
     end
 end
@@ -686,36 +686,36 @@ end
 
 procedure Node1000
 begin
-    gsay_reply(1057, 1000);
+    gsay_reply(SCRIPT_CCK9, 1000);
     if ((local_var(5) != 0) == 0) then begin
         if (local_var(6) == 0) then begin
             set_local_var(6, 6);
         end
-        if (message_str(14, 10001) != 0) then begin
-            if (get_critter_stat(self_obj, 35) < get_critter_stat(self_obj, 7)) then begin
-                giq_option(4, 1057, message_str(14, 10001), Node1001, 50);
+        if (g_mstr(10001) != 0) then begin
+            if (get_critter_stat(self_obj, STAT_current_hp) < get_critter_stat(self_obj, STAT_max_hp)) then begin
+                giq_option(4, SCRIPT_CCK9, g_mstr(10001), Node1001, NEUTRAL_REACTION);
             end
         end
-        if (message_str(14, 10008) != 0) then begin
-            giq_option(4, 1057, message_str(14, 10008), Node1007, 50);
+        if (g_mstr(10008) != 0) then begin
+            giq_option(4, SCRIPT_CCK9, g_mstr(10008), Node1007, NEUTRAL_REACTION);
         end
-        if (message_str(14, 10009) != 0) then begin
-            giq_option(4, 1057, message_str(14, 10009), Node1008, 50);
+        if (g_mstr(10009) != 0) then begin
+            giq_option(4, SCRIPT_CCK9, g_mstr(10009), Node1008, NEUTRAL_REACTION);
         end
-        if (message_str(14, 10002) != 0) then begin
-            giq_option(4, 1057, message_str(14, 10002), Node1002, 50);
+        if (g_mstr(10002) != 0) then begin
+            giq_option(4, SCRIPT_CCK9, g_mstr(10002), Node1002, NEUTRAL_REACTION);
         end
     end
     else begin
-        if (message_str(14, 10100) != 0) then begin
-            giq_option(4, 1057, message_str(14, 10100), Node1100, 50);
+        if (g_mstr(10100) != 0) then begin
+            giq_option(4, SCRIPT_CCK9, g_mstr(10100), Node1100, NEUTRAL_REACTION);
         end
     end
-    if (message_str(14, 10007) != 0) then begin
-        giq_option(4, 1057, message_str(14, 10007), Node999, 50);
+    if (g_mstr(10007) != 0) then begin
+        giq_option(4, SCRIPT_CCK9, g_mstr(10007), Node999, NEUTRAL_REACTION);
     end
-    if (message_str(14, random(300, 303)) != 0) then begin
-        giq_option(-3, 1057, message_str(14, random(300, 303)), Node999, 50);
+    if (g_mstr(random(300, 303)) != 0) then begin
+        giq_option(-3, SCRIPT_CCK9, g_mstr(random(300, 303)), Node999, NEUTRAL_REACTION);
     end
     debug_msg("USED PARTY MEMBER OPTIONS");
 end
@@ -732,7 +732,7 @@ begin
         end
         PartyHealingItem := obj_carrying_pid_obj(self_obj, 144);
         use_obj_on_obj(PartyHealingItem, self_obj);
-        debug_msg(obj_name(self_obj) + " used super stimpak.");
+        debug_msg(self_name + " used super stimpak.");
     end
     else begin
         if (obj_is_carrying_obj_pid(self_obj, 40)) then begin
@@ -744,7 +744,7 @@ begin
                 gfade_in(600);
             end
             use_obj_on_obj(obj_carrying_pid_obj(self_obj, 40), self_obj);
-            debug_msg(obj_name(self_obj) + " used stimpak.");
+            debug_msg(self_name + " used stimpak.");
         end
         else begin
             if (obj_is_carrying_obj_pid(self_obj, 273)) then begin
@@ -756,7 +756,7 @@ begin
                     gfade_in(600);
                 end
                 use_obj_on_obj(obj_carrying_pid_obj(self_obj, 273), self_obj);
-                debug_msg(obj_name(self_obj) + " used healing powder.");
+                debug_msg(self_name + " used healing powder.");
             end
             else begin
                 if (obj_is_carrying_obj_pid(self_obj, 91)) then begin
@@ -768,7 +768,7 @@ begin
                         gfade_in(600);
                     end
                     use_obj_on_obj(obj_carrying_pid_obj(self_obj, 91), self_obj);
-                    debug_msg(obj_name(self_obj) + " used doctor's bag kit.");
+                    debug_msg(self_name + " used doctor's bag kit.");
                 end
                 else begin
                     if (obj_is_carrying_obj_pid(self_obj, 47)) then begin
@@ -780,10 +780,10 @@ begin
                             gfade_in(600);
                         end
                         use_obj_on_obj(obj_carrying_pid_obj(self_obj, 47), self_obj);
-                        debug_msg(obj_name(self_obj) + " used First Aid kit.");
+                        debug_msg(self_name + " used First Aid kit.");
                     end
                     else begin
-                        if (has_skill(self_obj, 7) > has_skill(self_obj, 6)) then begin
+                        if (has_skill(self_obj, SKILL_DOCTOR) > has_skill(self_obj, SKILL_FIRST_AID)) then begin
                             if ((global_temp == 0) and (combat_is_initialized == 0)) then begin
                                 gfade_out(600);
                                 if ((60 * (60 * 10)) > 0) then begin
@@ -793,7 +793,7 @@ begin
                             end
                             if (is_success(roll_vs_skill(self_obj, 7, 0))) then begin
                                 critter_heal(self_obj, random(10, 25));
-                                debug_msg(obj_name(self_obj) + " healed some using doctor skill.");
+                                debug_msg(self_name + " healed some using doctor skill.");
                             end
                         end
                         else begin
@@ -806,7 +806,7 @@ begin
                             end
                             if (is_success(roll_vs_skill(self_obj, 6, 0))) then begin
                                 critter_heal(self_obj, random(5, 15));
-                                debug_msg(obj_name(self_obj) + " healed some using first aid skill.");
+                                debug_msg(self_name + " healed some using first aid skill.");
                             end
                         end
                     end
@@ -815,19 +815,19 @@ begin
         end
     end
     global_temp := 0;
-    if (get_critter_stat(self_obj, 35) == get_critter_stat(self_obj, 7)) then begin
-        gsay_reply(1057, 1100);
+    if (get_critter_stat(self_obj, STAT_current_hp) == get_critter_stat(self_obj, STAT_max_hp)) then begin
+        gsay_reply(SCRIPT_CCK9, 1100);
     end
     else begin
-        if ((get_critter_stat(self_obj, 35) * 100 / get_critter_stat(self_obj, 7)) >= 90) then begin
-            gsay_reply(1057, 1200);
+        if ((get_critter_stat(self_obj, STAT_current_hp) * 100 / get_critter_stat(self_obj, STAT_max_hp)) >= 90) then begin
+            gsay_reply(SCRIPT_CCK9, 1200);
         end
         else begin
-            if ((get_critter_stat(self_obj, 35) * 100 / get_critter_stat(self_obj, 7)) >= 70) then begin
-                gsay_reply(1057, 1300);
+            if ((get_critter_stat(self_obj, STAT_current_hp) * 100 / get_critter_stat(self_obj, STAT_max_hp)) >= 70) then begin
+                gsay_reply(SCRIPT_CCK9, 1300);
             end
             else begin
-                gsay_reply(1057, 1400);
+                gsay_reply(SCRIPT_CCK9, 1400);
             end
         end
     end
@@ -835,31 +835,31 @@ begin
         if (local_var(6) == 0) then begin
             set_local_var(6, 6);
         end
-        if (message_str(14, 10001) != 0) then begin
-            if (get_critter_stat(self_obj, 35) < get_critter_stat(self_obj, 7)) then begin
-                giq_option(4, 1057, message_str(14, 10001), Node1001, 50);
+        if (g_mstr(10001) != 0) then begin
+            if (get_critter_stat(self_obj, STAT_current_hp) < get_critter_stat(self_obj, STAT_max_hp)) then begin
+                giq_option(4, SCRIPT_CCK9, g_mstr(10001), Node1001, NEUTRAL_REACTION);
             end
         end
-        if (message_str(14, 10008) != 0) then begin
-            giq_option(4, 1057, message_str(14, 10008), Node1007, 50);
+        if (g_mstr(10008) != 0) then begin
+            giq_option(4, SCRIPT_CCK9, g_mstr(10008), Node1007, NEUTRAL_REACTION);
         end
-        if (message_str(14, 10009) != 0) then begin
-            giq_option(4, 1057, message_str(14, 10009), Node1008, 50);
+        if (g_mstr(10009) != 0) then begin
+            giq_option(4, SCRIPT_CCK9, g_mstr(10009), Node1008, NEUTRAL_REACTION);
         end
-        if (message_str(14, 10002) != 0) then begin
-            giq_option(4, 1057, message_str(14, 10002), Node1002, 50);
+        if (g_mstr(10002) != 0) then begin
+            giq_option(4, SCRIPT_CCK9, g_mstr(10002), Node1002, NEUTRAL_REACTION);
         end
     end
     else begin
-        if (message_str(14, 10100) != 0) then begin
-            giq_option(4, 1057, message_str(14, 10100), Node1100, 50);
+        if (g_mstr(10100) != 0) then begin
+            giq_option(4, SCRIPT_CCK9, g_mstr(10100), Node1100, NEUTRAL_REACTION);
         end
     end
-    if (message_str(14, 10007) != 0) then begin
-        giq_option(4, 1057, message_str(14, 10007), Node999, 50);
+    if (g_mstr(10007) != 0) then begin
+        giq_option(4, SCRIPT_CCK9, g_mstr(10007), Node999, NEUTRAL_REACTION);
     end
-    if (message_str(14, random(300, 303)) != 0) then begin
-        giq_option(-3, 1057, message_str(14, random(300, 303)), Node999, 50);
+    if (g_mstr(random(300, 303)) != 0) then begin
+        giq_option(-3, SCRIPT_CCK9, g_mstr(random(300, 303)), Node999, NEUTRAL_REACTION);
     end
     debug_msg("USED PARTY MEMBER OPTIONS");
 end
@@ -868,72 +868,72 @@ procedure Node1002
 begin
     set_local_var(5, game_time);
     party_remove(self_obj);
-    gsay_reply(1057, 1500);
+    gsay_reply(SCRIPT_CCK9, 1500);
     if ((local_var(5) != 0) == 0) then begin
         if (local_var(6) == 0) then begin
             set_local_var(6, 6);
         end
-        if (message_str(14, 10001) != 0) then begin
-            if (get_critter_stat(self_obj, 35) < get_critter_stat(self_obj, 7)) then begin
-                giq_option(4, 1057, message_str(14, 10001), Node1001, 50);
+        if (g_mstr(10001) != 0) then begin
+            if (get_critter_stat(self_obj, STAT_current_hp) < get_critter_stat(self_obj, STAT_max_hp)) then begin
+                giq_option(4, SCRIPT_CCK9, g_mstr(10001), Node1001, NEUTRAL_REACTION);
             end
         end
-        if (message_str(14, 10008) != 0) then begin
-            giq_option(4, 1057, message_str(14, 10008), Node1007, 50);
+        if (g_mstr(10008) != 0) then begin
+            giq_option(4, SCRIPT_CCK9, g_mstr(10008), Node1007, NEUTRAL_REACTION);
         end
-        if (message_str(14, 10009) != 0) then begin
-            giq_option(4, 1057, message_str(14, 10009), Node1008, 50);
+        if (g_mstr(10009) != 0) then begin
+            giq_option(4, SCRIPT_CCK9, g_mstr(10009), Node1008, NEUTRAL_REACTION);
         end
-        if (message_str(14, 10002) != 0) then begin
-            giq_option(4, 1057, message_str(14, 10002), Node1002, 50);
+        if (g_mstr(10002) != 0) then begin
+            giq_option(4, SCRIPT_CCK9, g_mstr(10002), Node1002, NEUTRAL_REACTION);
         end
     end
     else begin
-        if (message_str(14, 10100) != 0) then begin
-            giq_option(4, 1057, message_str(14, 10100), Node1100, 50);
+        if (g_mstr(10100) != 0) then begin
+            giq_option(4, SCRIPT_CCK9, g_mstr(10100), Node1100, NEUTRAL_REACTION);
         end
     end
-    if (message_str(14, 10007) != 0) then begin
-        giq_option(4, 1057, message_str(14, 10007), Node999, 50);
+    if (g_mstr(10007) != 0) then begin
+        giq_option(4, SCRIPT_CCK9, g_mstr(10007), Node999, NEUTRAL_REACTION);
     end
-    if (message_str(14, random(300, 303)) != 0) then begin
-        giq_option(-3, 1057, message_str(14, random(300, 303)), Node999, 50);
+    if (g_mstr(random(300, 303)) != 0) then begin
+        giq_option(-3, SCRIPT_CCK9, g_mstr(random(300, 303)), Node999, NEUTRAL_REACTION);
     end
     debug_msg("USED PARTY MEMBER OPTIONS");
 end
 
 procedure Node1003
 begin
-    gsay_reply(1057, 1600);
+    gsay_reply(SCRIPT_CCK9, 1600);
     if ((local_var(5) != 0) == 0) then begin
         if (local_var(6) == 0) then begin
             set_local_var(6, 6);
         end
-        if (message_str(14, 10001) != 0) then begin
-            if (get_critter_stat(self_obj, 35) < get_critter_stat(self_obj, 7)) then begin
-                giq_option(4, 1057, message_str(14, 10001), Node1001, 50);
+        if (g_mstr(10001) != 0) then begin
+            if (get_critter_stat(self_obj, STAT_current_hp) < get_critter_stat(self_obj, STAT_max_hp)) then begin
+                giq_option(4, SCRIPT_CCK9, g_mstr(10001), Node1001, NEUTRAL_REACTION);
             end
         end
-        if (message_str(14, 10008) != 0) then begin
-            giq_option(4, 1057, message_str(14, 10008), Node1007, 50);
+        if (g_mstr(10008) != 0) then begin
+            giq_option(4, SCRIPT_CCK9, g_mstr(10008), Node1007, NEUTRAL_REACTION);
         end
-        if (message_str(14, 10009) != 0) then begin
-            giq_option(4, 1057, message_str(14, 10009), Node1008, 50);
+        if (g_mstr(10009) != 0) then begin
+            giq_option(4, SCRIPT_CCK9, g_mstr(10009), Node1008, NEUTRAL_REACTION);
         end
-        if (message_str(14, 10002) != 0) then begin
-            giq_option(4, 1057, message_str(14, 10002), Node1002, 50);
+        if (g_mstr(10002) != 0) then begin
+            giq_option(4, SCRIPT_CCK9, g_mstr(10002), Node1002, NEUTRAL_REACTION);
         end
     end
     else begin
-        if (message_str(14, 10100) != 0) then begin
-            giq_option(4, 1057, message_str(14, 10100), Node1100, 50);
+        if (g_mstr(10100) != 0) then begin
+            giq_option(4, SCRIPT_CCK9, g_mstr(10100), Node1100, NEUTRAL_REACTION);
         end
     end
-    if (message_str(14, 10007) != 0) then begin
-        giq_option(4, 1057, message_str(14, 10007), Node999, 50);
+    if (g_mstr(10007) != 0) then begin
+        giq_option(4, SCRIPT_CCK9, g_mstr(10007), Node999, NEUTRAL_REACTION);
     end
-    if (message_str(14, random(300, 303)) != 0) then begin
-        giq_option(-3, 1057, message_str(14, random(300, 303)), Node999, 50);
+    if (g_mstr(random(300, 303)) != 0) then begin
+        giq_option(-3, SCRIPT_CCK9, g_mstr(random(300, 303)), Node999, NEUTRAL_REACTION);
     end
     debug_msg("USED PARTY MEMBER OPTIONS");
 end
@@ -941,36 +941,36 @@ end
 procedure Node1004
 begin
     set_local_var(6, 3);
-    gsay_reply(1057, 1700);
+    gsay_reply(SCRIPT_CCK9, 1700);
     if ((local_var(5) != 0) == 0) then begin
         if (local_var(6) == 0) then begin
             set_local_var(6, 6);
         end
-        if (message_str(14, 10001) != 0) then begin
-            if (get_critter_stat(self_obj, 35) < get_critter_stat(self_obj, 7)) then begin
-                giq_option(4, 1057, message_str(14, 10001), Node1001, 50);
+        if (g_mstr(10001) != 0) then begin
+            if (get_critter_stat(self_obj, STAT_current_hp) < get_critter_stat(self_obj, STAT_max_hp)) then begin
+                giq_option(4, SCRIPT_CCK9, g_mstr(10001), Node1001, NEUTRAL_REACTION);
             end
         end
-        if (message_str(14, 10008) != 0) then begin
-            giq_option(4, 1057, message_str(14, 10008), Node1007, 50);
+        if (g_mstr(10008) != 0) then begin
+            giq_option(4, SCRIPT_CCK9, g_mstr(10008), Node1007, NEUTRAL_REACTION);
         end
-        if (message_str(14, 10009) != 0) then begin
-            giq_option(4, 1057, message_str(14, 10009), Node1008, 50);
+        if (g_mstr(10009) != 0) then begin
+            giq_option(4, SCRIPT_CCK9, g_mstr(10009), Node1008, NEUTRAL_REACTION);
         end
-        if (message_str(14, 10002) != 0) then begin
-            giq_option(4, 1057, message_str(14, 10002), Node1002, 50);
+        if (g_mstr(10002) != 0) then begin
+            giq_option(4, SCRIPT_CCK9, g_mstr(10002), Node1002, NEUTRAL_REACTION);
         end
     end
     else begin
-        if (message_str(14, 10100) != 0) then begin
-            giq_option(4, 1057, message_str(14, 10100), Node1100, 50);
+        if (g_mstr(10100) != 0) then begin
+            giq_option(4, SCRIPT_CCK9, g_mstr(10100), Node1100, NEUTRAL_REACTION);
         end
     end
-    if (message_str(14, 10007) != 0) then begin
-        giq_option(4, 1057, message_str(14, 10007), Node999, 50);
+    if (g_mstr(10007) != 0) then begin
+        giq_option(4, SCRIPT_CCK9, g_mstr(10007), Node999, NEUTRAL_REACTION);
     end
-    if (message_str(14, random(300, 303)) != 0) then begin
-        giq_option(-3, 1057, message_str(14, random(300, 303)), Node999, 50);
+    if (g_mstr(random(300, 303)) != 0) then begin
+        giq_option(-3, SCRIPT_CCK9, g_mstr(random(300, 303)), Node999, NEUTRAL_REACTION);
     end
     debug_msg("USED PARTY MEMBER OPTIONS");
 end
@@ -978,36 +978,36 @@ end
 procedure Node1005
 begin
     set_local_var(6, 6);
-    gsay_reply(1057, 1800);
+    gsay_reply(SCRIPT_CCK9, 1800);
     if ((local_var(5) != 0) == 0) then begin
         if (local_var(6) == 0) then begin
             set_local_var(6, 6);
         end
-        if (message_str(14, 10001) != 0) then begin
-            if (get_critter_stat(self_obj, 35) < get_critter_stat(self_obj, 7)) then begin
-                giq_option(4, 1057, message_str(14, 10001), Node1001, 50);
+        if (g_mstr(10001) != 0) then begin
+            if (get_critter_stat(self_obj, STAT_current_hp) < get_critter_stat(self_obj, STAT_max_hp)) then begin
+                giq_option(4, SCRIPT_CCK9, g_mstr(10001), Node1001, NEUTRAL_REACTION);
             end
         end
-        if (message_str(14, 10008) != 0) then begin
-            giq_option(4, 1057, message_str(14, 10008), Node1007, 50);
+        if (g_mstr(10008) != 0) then begin
+            giq_option(4, SCRIPT_CCK9, g_mstr(10008), Node1007, NEUTRAL_REACTION);
         end
-        if (message_str(14, 10009) != 0) then begin
-            giq_option(4, 1057, message_str(14, 10009), Node1008, 50);
+        if (g_mstr(10009) != 0) then begin
+            giq_option(4, SCRIPT_CCK9, g_mstr(10009), Node1008, NEUTRAL_REACTION);
         end
-        if (message_str(14, 10002) != 0) then begin
-            giq_option(4, 1057, message_str(14, 10002), Node1002, 50);
+        if (g_mstr(10002) != 0) then begin
+            giq_option(4, SCRIPT_CCK9, g_mstr(10002), Node1002, NEUTRAL_REACTION);
         end
     end
     else begin
-        if (message_str(14, 10100) != 0) then begin
-            giq_option(4, 1057, message_str(14, 10100), Node1100, 50);
+        if (g_mstr(10100) != 0) then begin
+            giq_option(4, SCRIPT_CCK9, g_mstr(10100), Node1100, NEUTRAL_REACTION);
         end
     end
-    if (message_str(14, 10007) != 0) then begin
-        giq_option(4, 1057, message_str(14, 10007), Node999, 50);
+    if (g_mstr(10007) != 0) then begin
+        giq_option(4, SCRIPT_CCK9, g_mstr(10007), Node999, NEUTRAL_REACTION);
     end
-    if (message_str(14, random(300, 303)) != 0) then begin
-        giq_option(-3, 1057, message_str(14, random(300, 303)), Node999, 50);
+    if (g_mstr(random(300, 303)) != 0) then begin
+        giq_option(-3, SCRIPT_CCK9, g_mstr(random(300, 303)), Node999, NEUTRAL_REACTION);
     end
     debug_msg("USED PARTY MEMBER OPTIONS");
 end
@@ -1015,36 +1015,36 @@ end
 procedure Node1006
 begin
     set_local_var(6, 9);
-    gsay_reply(1057, 1900);
+    gsay_reply(SCRIPT_CCK9, 1900);
     if ((local_var(5) != 0) == 0) then begin
         if (local_var(6) == 0) then begin
             set_local_var(6, 6);
         end
-        if (message_str(14, 10001) != 0) then begin
-            if (get_critter_stat(self_obj, 35) < get_critter_stat(self_obj, 7)) then begin
-                giq_option(4, 1057, message_str(14, 10001), Node1001, 50);
+        if (g_mstr(10001) != 0) then begin
+            if (get_critter_stat(self_obj, STAT_current_hp) < get_critter_stat(self_obj, STAT_max_hp)) then begin
+                giq_option(4, SCRIPT_CCK9, g_mstr(10001), Node1001, NEUTRAL_REACTION);
             end
         end
-        if (message_str(14, 10008) != 0) then begin
-            giq_option(4, 1057, message_str(14, 10008), Node1007, 50);
+        if (g_mstr(10008) != 0) then begin
+            giq_option(4, SCRIPT_CCK9, g_mstr(10008), Node1007, NEUTRAL_REACTION);
         end
-        if (message_str(14, 10009) != 0) then begin
-            giq_option(4, 1057, message_str(14, 10009), Node1008, 50);
+        if (g_mstr(10009) != 0) then begin
+            giq_option(4, SCRIPT_CCK9, g_mstr(10009), Node1008, NEUTRAL_REACTION);
         end
-        if (message_str(14, 10002) != 0) then begin
-            giq_option(4, 1057, message_str(14, 10002), Node1002, 50);
+        if (g_mstr(10002) != 0) then begin
+            giq_option(4, SCRIPT_CCK9, g_mstr(10002), Node1002, NEUTRAL_REACTION);
         end
     end
     else begin
-        if (message_str(14, 10100) != 0) then begin
-            giq_option(4, 1057, message_str(14, 10100), Node1100, 50);
+        if (g_mstr(10100) != 0) then begin
+            giq_option(4, SCRIPT_CCK9, g_mstr(10100), Node1100, NEUTRAL_REACTION);
         end
     end
-    if (message_str(14, 10007) != 0) then begin
-        giq_option(4, 1057, message_str(14, 10007), Node999, 50);
+    if (g_mstr(10007) != 0) then begin
+        giq_option(4, SCRIPT_CCK9, g_mstr(10007), Node999, NEUTRAL_REACTION);
     end
-    if (message_str(14, random(300, 303)) != 0) then begin
-        giq_option(-3, 1057, message_str(14, random(300, 303)), Node999, 50);
+    if (g_mstr(random(300, 303)) != 0) then begin
+        giq_option(-3, SCRIPT_CCK9, g_mstr(random(300, 303)), Node999, NEUTRAL_REACTION);
     end
     debug_msg("USED PARTY MEMBER OPTIONS");
 end
@@ -1067,9 +1067,9 @@ end
 
 procedure Node1100
 begin
-    if (((metarule(16, 0) - 1) >= (floor(get_critter_stat(dude_obj, 3) / 2) + has_trait(0, dude_obj, 98))) or ((metarule(16, 0) - 1) >= 5)) then begin
-        gsay_reply(1057, 2000);
-        giq_option(1, 1057, message_str(14, 10007), Node999, 50);
+    if (((party_member_count(0) - 1) >= (floor(dude_charisma / 2) + has_trait(TRAIT_PERK, dude_obj, PERK_magnetic_personality_perk))) or ((party_member_count(0) - 1) >= 5)) then begin
+        gsay_reply(SCRIPT_CCK9, 2000);
+        giq_option(1, SCRIPT_CCK9, g_mstr(10007), Node999, NEUTRAL_REACTION);
     end
     else begin
         set_local_var(5, 0);
@@ -1078,45 +1078,44 @@ begin
                 set_local_var(6, 6);
             end
             set_local_var(5, 0);
-            if (has_trait(1, self_obj, 6) != 0) then begin
-                set_local_var(7, has_trait(1, self_obj, 6));
+            if (has_trait(TRAIT_OBJECT, self_obj, OBJECT_TEAM_NUM) != 0) then begin
+                set_local_var(7, has_trait(TRAIT_OBJECT, self_obj, OBJECT_TEAM_NUM));
             end
-            critter_add_trait(self_obj, 1, 6, 0);
+            critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 0);
             party_add(self_obj);
         end
-        debug_msg("join party: " + obj_name(self_obj));
-        gsay_reply(1057, 2200);
+        debug_msg("join party: " + self_name);
+        gsay_reply(SCRIPT_CCK9, 2200);
         if ((local_var(5) != 0) == 0) then begin
             if (local_var(6) == 0) then begin
                 set_local_var(6, 6);
             end
-            if (message_str(14, 10001) != 0) then begin
-                if (get_critter_stat(self_obj, 35) < get_critter_stat(self_obj, 7)) then begin
-                    giq_option(4, 1057, message_str(14, 10001), Node1001, 50);
+            if (g_mstr(10001) != 0) then begin
+                if (get_critter_stat(self_obj, STAT_current_hp) < get_critter_stat(self_obj, STAT_max_hp)) then begin
+                    giq_option(4, SCRIPT_CCK9, g_mstr(10001), Node1001, NEUTRAL_REACTION);
                 end
             end
-            if (message_str(14, 10008) != 0) then begin
-                giq_option(4, 1057, message_str(14, 10008), Node1007, 50);
+            if (g_mstr(10008) != 0) then begin
+                giq_option(4, SCRIPT_CCK9, g_mstr(10008), Node1007, NEUTRAL_REACTION);
             end
-            if (message_str(14, 10009) != 0) then begin
-                giq_option(4, 1057, message_str(14, 10009), Node1008, 50);
+            if (g_mstr(10009) != 0) then begin
+                giq_option(4, SCRIPT_CCK9, g_mstr(10009), Node1008, NEUTRAL_REACTION);
             end
-            if (message_str(14, 10002) != 0) then begin
-                giq_option(4, 1057, message_str(14, 10002), Node1002, 50);
+            if (g_mstr(10002) != 0) then begin
+                giq_option(4, SCRIPT_CCK9, g_mstr(10002), Node1002, NEUTRAL_REACTION);
             end
         end
         else begin
-            if (message_str(14, 10100) != 0) then begin
-                giq_option(4, 1057, message_str(14, 10100), Node1100, 50);
+            if (g_mstr(10100) != 0) then begin
+                giq_option(4, SCRIPT_CCK9, g_mstr(10100), Node1100, NEUTRAL_REACTION);
             end
         end
-        if (message_str(14, 10007) != 0) then begin
-            giq_option(4, 1057, message_str(14, 10007), Node999, 50);
+        if (g_mstr(10007) != 0) then begin
+            giq_option(4, SCRIPT_CCK9, g_mstr(10007), Node999, NEUTRAL_REACTION);
         end
-        if (message_str(14, random(300, 303)) != 0) then begin
-            giq_option(-3, 1057, message_str(14, random(300, 303)), Node999, 50);
+        if (g_mstr(random(300, 303)) != 0) then begin
+            giq_option(-3, SCRIPT_CCK9, g_mstr(random(300, 303)), Node999, NEUTRAL_REACTION);
         end
         debug_msg("USED PARTY MEMBER OPTIONS");
     end
 end
-
