diff --git a/Mods/MegaMod/Original/CHLADR1.ssl b/Mods/MegaMod/ReDefine/CHLADR1.ssl
index 4bbb984..20d1d8c 100644
--- a/Mods/MegaMod/Original/CHLADR1.ssl
+++ b/Mods/MegaMod/ReDefine/CHLADR1.ssl
@@ -139,12 +139,12 @@ end
 
 procedure use_p_proc
 begin
-    if (global_var(991) == 0) then begin
+    if (global_var(GVAR_PA_MISSING) == 0) then begin
         script_overrides;
-        display_msg(message_str(1538, 100));
+        display_mstr(100);
     end
     else begin
-        if (global_var(991) == 1) then begin
+        if (global_var(GVAR_PA_MISSING) == 1) then begin
         end
     end
 end
@@ -160,4 +160,3 @@ end
 procedure drop_p_proc
 begin
 end
-
