diff --git a/Mods/MegaMod/Original/CRSHTRCK.ssl b/Mods/MegaMod/ReDefine/CRSHTRCK.ssl
index e5b91a0..164a8d0 100644
--- a/Mods/MegaMod/Original/CRSHTRCK.ssl
+++ b/Mods/MegaMod/ReDefine/CRSHTRCK.ssl
@@ -158,12 +158,11 @@ end
 procedure look_at_p_proc
 begin
     script_overrides;
-    display_msg(message_str(1740, 100));
+    display_mstr(100);
 end
 
 procedure description_p_proc
 begin
     script_overrides;
-    display_msg(message_str(1740, 101));
+    display_mstr(101);
 end
-
