diff --git a/Mods/MegaMod/Original/DCTUBBY.ssl b/Mods/MegaMod/ReDefine/DCTUBBY.ssl
index 909e022..28bb241 100644
--- a/Mods/MegaMod/Original/DCTUBBY.ssl
+++ b/Mods/MegaMod/ReDefine/DCTUBBY.ssl
@@ -68,10 +68,10 @@ begin
                 if (anim_busy(self_obj) == 0) then begin
                     if ((tile_distance_objs(self_obj, dude_obj) < 8) or obj_can_see_obj(self_obj, dude_obj)) then begin
                         if (anim_busy(self_obj) == 0) then begin
-                            global_temp := rotation_to_tile(tile_num(dude_obj), tile_num(self_obj));
+                            global_temp := rotation_to_tile(dude_tile, tile_num(self_obj));
                             animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(self_obj), global_temp, random(3, 10)), 1);
                             global_temp := (global_temp + 1) % 6;
-                            while ((anim_busy(self_obj) == 0) and (global_temp != rotation_to_tile(tile_num(dude_obj), tile_num(self_obj)))) do begin
+                            while ((anim_busy(self_obj) == 0) and (global_temp != rotation_to_tile(dude_tile, tile_num(self_obj)))) do begin
                                 debug_msg("flee loop: rot == " + global_temp);
                                 animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(self_obj), global_temp, random(3, 10)), 1);
                                 global_temp := (global_temp + 1) % 6;
@@ -81,7 +81,7 @@ begin
                 end
             end
             else begin
-                attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+                attack(dude_obj);
             end
         end
     end
@@ -90,7 +90,7 @@ begin
             animate_move_obj_to_tile(self_obj, local_var(7), 0);
         end
         else begin
-            if (has_trait(1, self_obj, 10) != local_var(8)) then begin
+            if (obj_get_rot(self_obj) != local_var(8)) then begin
                 anim(self_obj, 1000, local_var(8));
             end
         end
@@ -114,7 +114,7 @@ begin
     move_obj_inven_to_obj(den_tubby_box_obj, self_obj);
     if (local_var(2) == 0) then begin
         set_local_var(2, 1);
-        set_local_var(3, (get_critter_stat(dude_obj, 3) - get_critter_stat(self_obj, 3)) * 5);
+        set_local_var(3, (dude_charisma - self_charisma) * 5);
         debug_msg("Base Reaction == " + local_var(3));
     end
     Static_Reaction := 0;
@@ -126,14 +126,14 @@ begin
     else begin
         reaction_bonus_karma := -1 * (-1 * reaction_bonus_karma / 1000);
     end
-    if (global_var(49) >= 0) then begin
-        reaction_bonus_town_rep := global_var(49) / 2;
+    if (global_var(GVAR_TOWN_REP_THE_DEN) >= 0) then begin
+        reaction_bonus_town_rep := global_var(GVAR_TOWN_REP_THE_DEN) / 2;
     end
     else begin
-        reaction_bonus_town_rep := -1 * (-1 * global_var(49) / 2);
+        reaction_bonus_town_rep := -1 * (-1 * global_var(GVAR_TOWN_REP_THE_DEN) / 2);
     end
-    if (has_trait(2, dude_obj, 13)) then begin
-        if (get_critter_stat(dude_obj, 34) == get_critter_stat(self_obj, 34)) then begin
+    if (has_trait(TRAIT_TRAIT, dude_obj, TRAIT_sex_appeal)) then begin
+        if (dude_gender == self_gender) then begin
             Static_Reaction := Static_Reaction + -20;
         end
         else begin
@@ -159,52 +159,52 @@ begin
             Static_Reaction := Static_Reaction + reaction_bonus_town_rep;
         end
         debug_msg("Town Rep Bonus == " + Static_Reaction);
-        set_global_var(37, 0);
-        set_global_var(38, 0);
-        set_global_var(39, 0);
-        set_global_var(40, 0);
-        set_global_var(41, 0);
-        set_global_var(42, 0);
-        set_global_var(43, 0);
-        set_global_var(44, 0);
-        set_global_var(45, 0);
-        tmp_gen_rep := global_var(0);
-        if (has_trait(0, dude_obj, 95)) then begin
-            tmp_gen_rep := global_var(0) * 2;
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+        set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+        set_global_var(GVAR_KARMA_DEFENDER, 0);
+        set_global_var(GVAR_KARMA_WANDERER, 0);
+        set_global_var(GVAR_KARMA_BETRAYER, 0);
+        set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+        set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+        tmp_gen_rep := global_var(GVAR_PLAYER_REPUTATION);
+        if (has_trait(TRAIT_PERK, dude_obj, PERK_karma_beacon_perk)) then begin
+            tmp_gen_rep := global_var(GVAR_PLAYER_REPUTATION) * 2;
         end
         if (tmp_gen_rep >= 1000) then begin
-            set_global_var(37, 1);
+            set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
         end
         else begin
             if (tmp_gen_rep >= 750) then begin
-                set_global_var(38, 1);
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
             end
             else begin
                 if (tmp_gen_rep >= 500) then begin
-                    set_global_var(39, 1);
+                    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                 end
                 else begin
                     if (tmp_gen_rep >= 250) then begin
-                        set_global_var(40, 1);
+                        set_global_var(GVAR_KARMA_DEFENDER, 1);
                     end
                     else begin
                         if (tmp_gen_rep > -250) then begin
-                            set_global_var(41, 1);
+                            set_global_var(GVAR_KARMA_WANDERER, 1);
                         end
                         else begin
                             if (tmp_gen_rep > -500) then begin
-                                set_global_var(42, 1);
+                                set_global_var(GVAR_KARMA_BETRAYER, 1);
                             end
                             else begin
                                 if (tmp_gen_rep > -750) then begin
-                                    set_global_var(43, 1);
+                                    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                 end
                                 else begin
                                     if (tmp_gen_rep > -1000) then begin
-                                        set_global_var(44, 1);
+                                        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                     end
                                     else begin
-                                        set_global_var(45, 1);
+                                        set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                     end
                                 end
                             end
@@ -213,35 +213,35 @@ begin
                 end
             end
         end
-        if (global_var(37) == 1) then begin
+        if (global_var(GVAR_KARMA_HOLY_WARRIOR) == 1) then begin
             Static_Reaction := Static_Reaction - 20;
         end
         else begin
-            if (global_var(38) == 1) then begin
+            if (global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES) == 1) then begin
                 Static_Reaction := Static_Reaction - 15;
             end
             else begin
-                if (global_var(39) == 1) then begin
+                if (global_var(GVAR_KARMA_SHIELD_OF_HOPE) == 1) then begin
                     Static_Reaction := Static_Reaction - 10;
                 end
                 else begin
-                    if (global_var(40) == 1) then begin
+                    if (global_var(GVAR_KARMA_DEFENDER) == 1) then begin
                         Static_Reaction := Static_Reaction - 5;
                     end
                     else begin
-                        if (global_var(41) == 1) then begin
+                        if (global_var(GVAR_KARMA_WANDERER) == 1) then begin
                             Static_Reaction := Static_Reaction - 0;
                         end
                         else begin
-                            if (global_var(42) == 1) then begin
+                            if (global_var(GVAR_KARMA_BETRAYER) == 1) then begin
                                 Static_Reaction := Static_Reaction + 5;
                             end
                             else begin
-                                if (global_var(43) == 1) then begin
+                                if (global_var(GVAR_KARMA_SWORD_OF_DESPAIR) == 1) then begin
                                     Static_Reaction := Static_Reaction + 10;
                                 end
                                 else begin
-                                    if (global_var(44) == 1) then begin
+                                    if (global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE) == 1) then begin
                                         Static_Reaction := Static_Reaction + 15;
                                     end
                                     else begin
@@ -261,52 +261,52 @@ begin
         debug_msg("Karma Reaction Bonus == " + Static_Reaction);
         Static_Reaction := Static_Reaction + reaction_bonus_town_rep;
         debug_msg("Town Rep Bonus == " + Static_Reaction);
-        set_global_var(37, 0);
-        set_global_var(38, 0);
-        set_global_var(39, 0);
-        set_global_var(40, 0);
-        set_global_var(41, 0);
-        set_global_var(42, 0);
-        set_global_var(43, 0);
-        set_global_var(44, 0);
-        set_global_var(45, 0);
-        tmp_gen_rep := global_var(0);
-        if (has_trait(0, dude_obj, 95)) then begin
-            tmp_gen_rep := global_var(0) * 2;
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+        set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+        set_global_var(GVAR_KARMA_DEFENDER, 0);
+        set_global_var(GVAR_KARMA_WANDERER, 0);
+        set_global_var(GVAR_KARMA_BETRAYER, 0);
+        set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+        set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+        tmp_gen_rep := global_var(GVAR_PLAYER_REPUTATION);
+        if (has_trait(TRAIT_PERK, dude_obj, PERK_karma_beacon_perk)) then begin
+            tmp_gen_rep := global_var(GVAR_PLAYER_REPUTATION) * 2;
         end
         if (tmp_gen_rep >= 1000) then begin
-            set_global_var(37, 1);
+            set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
         end
         else begin
             if (tmp_gen_rep >= 750) then begin
-                set_global_var(38, 1);
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
             end
             else begin
                 if (tmp_gen_rep >= 500) then begin
-                    set_global_var(39, 1);
+                    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                 end
                 else begin
                     if (tmp_gen_rep >= 250) then begin
-                        set_global_var(40, 1);
+                        set_global_var(GVAR_KARMA_DEFENDER, 1);
                     end
                     else begin
                         if (tmp_gen_rep > -250) then begin
-                            set_global_var(41, 1);
+                            set_global_var(GVAR_KARMA_WANDERER, 1);
                         end
                         else begin
                             if (tmp_gen_rep > -500) then begin
-                                set_global_var(42, 1);
+                                set_global_var(GVAR_KARMA_BETRAYER, 1);
                             end
                             else begin
                                 if (tmp_gen_rep > -750) then begin
-                                    set_global_var(43, 1);
+                                    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                 end
                                 else begin
                                     if (tmp_gen_rep > -1000) then begin
-                                        set_global_var(44, 1);
+                                        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                     end
                                     else begin
-                                        set_global_var(45, 1);
+                                        set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                     end
                                 end
                             end
@@ -315,35 +315,35 @@ begin
                 end
             end
         end
-        if (global_var(37) == 1) then begin
+        if (global_var(GVAR_KARMA_HOLY_WARRIOR) == 1) then begin
             Static_Reaction := Static_Reaction + 20;
         end
         else begin
-            if (global_var(38) == 1) then begin
+            if (global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES) == 1) then begin
                 Static_Reaction := Static_Reaction + 15;
             end
             else begin
-                if (global_var(39) == 1) then begin
+                if (global_var(GVAR_KARMA_SHIELD_OF_HOPE) == 1) then begin
                     Static_Reaction := Static_Reaction + 10;
                 end
                 else begin
-                    if (global_var(40) == 1) then begin
+                    if (global_var(GVAR_KARMA_DEFENDER) == 1) then begin
                         Static_Reaction := Static_Reaction + 5;
                     end
                     else begin
-                        if (global_var(41) == 1) then begin
+                        if (global_var(GVAR_KARMA_WANDERER) == 1) then begin
                             Static_Reaction := Static_Reaction + 0;
                         end
                         else begin
-                            if (global_var(42) == 1) then begin
+                            if (global_var(GVAR_KARMA_BETRAYER) == 1) then begin
                                 Static_Reaction := Static_Reaction + -5;
                             end
                             else begin
-                                if (global_var(43) == 1) then begin
+                                if (global_var(GVAR_KARMA_SWORD_OF_DESPAIR) == 1) then begin
                                     Static_Reaction := Static_Reaction + -10;
                                 end
                                 else begin
-                                    if (global_var(44) == 1) then begin
+                                    if (global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE) == 1) then begin
                                         Static_Reaction := Static_Reaction + -15;
                                     end
                                     else begin
@@ -359,32 +359,32 @@ begin
         debug_msg("Good Person Reaction Bonus == " + Static_Reaction);
     end
     if (Slavery_Tolerant == 0) then begin
-        if (global_var(11) == 1) then begin
+        if (global_var(GVAR_REPUTATION_SLAVER) == 1) then begin
             Static_Reaction := Static_Reaction + (2 * -25);
         end
     end
     else begin
         if (Slavery_Tolerant == 1) then begin
-            if (global_var(11) == 1) then begin
+            if (global_var(GVAR_REPUTATION_SLAVER) == 1) then begin
                 Static_Reaction := Static_Reaction + -25;
             end
         end
         else begin
             if (Slavery_Tolerant == 3) then begin
-                if (global_var(11) == 1) then begin
+                if (global_var(GVAR_REPUTATION_SLAVER) == 1) then begin
                     Static_Reaction := Static_Reaction + 25;
                 end
             end
         end
     end
     debug_msg("Slaver + Aligned Reaction == " + Static_Reaction);
-    Static_Reaction := Static_Reaction + (10 * has_trait(0, dude_obj, 10));
+    Static_Reaction := Static_Reaction + (10 * has_trait(TRAIT_PERK, dude_obj, PERK_presence));
     debug_msg("Presence Reaction == " + Static_Reaction);
-    if (global_var(1) >= 2) then begin
+    if (global_var(GVAR_CHILDKILLER_REPUTATION) >= 2) then begin
         Static_Reaction := Static_Reaction + -30;
     end
     debug_msg("Childkiller Reaction == " + Static_Reaction);
-    if (has_trait(0, dude_obj, 39)) then begin
+    if (has_trait(TRAIT_PERK, dude_obj, PERK_cult_of_personality)) then begin
         if (Evil_Critter == 1) then begin
             if (Static_Reaction > 0) then begin
                 Static_Reaction := Static_Reaction * -1;
@@ -441,16 +441,16 @@ begin
     end
     start_gdialog(48, self_obj, 4, -1, -1);
     gsay_start;
-    if (get_critter_stat(dude_obj, 4) <= 3) then begin
+    if (dude_is_stupid) then begin
         call Node007();
     end
     else begin
-        if (((local_var(4) bwand 2) != 0) or (global_var(14) == 1) or ((global_var(446) bwand 524288) != 0) or ((global_var(445) bwand 64) != 0)) then begin
+        if (((local_var(4) bwand 2) != 0) or (global_var(GVAR_ENEMY_DEN) == 1) or ((global_var(GVAR_DEN_FLAG_2) bwand 524288) != 0) or ((global_var(GVAR_DEN_FLAG_1) bwand 64) != 0)) then begin
             call Node001();
         end
         else begin
             if ((local_var(4) bwand 1) != 0) then begin
-                if ((local_var(1) <= -2) and (has_trait(0, dude_obj, 39) == 0)) then begin
+                if ((local_var(1) <= -2) and (has_trait(TRAIT_PERK, dude_obj, PERK_cult_of_personality) == 0)) then begin
                     call Node001();
                 end
                 else begin
@@ -472,15 +472,15 @@ end
 procedure destroy_p_proc
 begin
     tubby_obj := 0;
-    if (((global_var(445) bwand 4) != 0) == 0) then begin
+    if (((global_var(GVAR_DEN_FLAG_1) bwand 4) != 0) == 0) then begin
         if ((tile_contains_pid_obj(19504, 0, 33555587) == 0) and (tile_contains_pid_obj(19504, 0, 33555588) == 0)) then begin
-            global_temp := create_object_sid(33555587, 0, 0, 529);
+            global_temp := create_object_sid(33555587, 0, 0, SCRIPT_ZICLOSED);
             set_obj_visibility(global_temp, 1);
             move_to(global_temp, 19504, 0);
         end
         debug_msg("created out of business @ tile == " + 19504 + " / elevation == " + 0);
     end
-    set_global_var(445, global_var(445) bwor 4);
+    set_global_var(GVAR_DEN_FLAG_1, global_var(GVAR_DEN_FLAG_1) bwor 4);
     move_obj_inven_to_obj(den_tubby_box_obj, self_obj);
     set_map_var(5, -1);
 end
@@ -488,13 +488,13 @@ end
 procedure look_at_p_proc
 begin
     script_overrides;
-    display_msg(message_str(48, 100 + ((local_var(4) bwand 1) != 0)));
+    display_mstr(100 + ((local_var(4) bwand 1) != 0));
 end
 
 procedure description_p_proc
 begin
     script_overrides;
-    display_msg(message_str(48, 150 + ((local_var(4) bwand 1) != 0)));
+    display_mstr(150 + ((local_var(4) bwand 1) != 0));
 end
 
 procedure use_skill_on_p_proc
@@ -511,12 +511,12 @@ end
 procedure map_enter_p_proc
 begin
     tubby_obj := self_obj;
-    if (metarule(22, 0) == 0) then begin
-        if (metarule(14, 0)) then begin
-            critter_add_trait(self_obj, 1, 6, 19);
-            critter_add_trait(self_obj, 1, 5, 17);
+    if (is_loading_game == 0) then begin
+        if (map_first_run) then begin
+            critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 19);
+            critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_STORE_OWNER);
             set_local_var(7, tile_num(self_obj));
-            set_local_var(8, has_trait(1, self_obj, 10));
+            set_local_var(8, obj_get_rot(self_obj));
         end
     end
 end
@@ -533,186 +533,186 @@ end
 procedure Node001
 begin
     set_local_var(9, 1);
-    gsay_reply(48, 153);
-    giq_option(5, 48, 154, Node996, 50);
-    giq_option(4, 48, 155, Node005, 50);
-    giq_option(4, 48, 156, Node999, 50);
-    if (global_var(1002) == 1) then begin
-        giq_option(4, 48, 513, Node011, 50);
+    Reply(153);
+    giq_option(5, NAME, 154, Node996, NEUTRAL_REACTION);
+    giq_option(4, NAME, 155, Node005, NEUTRAL_REACTION);
+    giq_option(4, NAME, 156, Node999, NEUTRAL_REACTION);
+    if (global_var(GVAR_ENTER_ELDRIDGE_BASEMENT) == 1) then begin
+        giq_option(4, NAME, 513, Node011, NEUTRAL_REACTION);
     end
-    if ((global_var(996) == 2) and (global_var(1002) == 5)) then begin
-        giq_option(4, 48, 526, Node016, 50);
+    if ((global_var(GVAR_KILL_TREASURE_HUNTER) == 2) and (global_var(GVAR_ENTER_ELDRIDGE_BASEMENT) == 5)) then begin
+        giq_option(4, NAME, 526, Node016, NEUTRAL_REACTION);
     end
 end
 
 procedure Node002
 begin
-    gsay_reply(48, 157);
-    giq_option(4, 48, 158, Node995, 50);
-    giq_option(4, 48, 159, Node006, 50);
-    if (global_var(887) == 0) then begin
-        giq_option(4, 48, 600, Node002a, 50);
+    Reply(157);
+    giq_option(4, NAME, 158, Node995, NEUTRAL_REACTION);
+    giq_option(4, NAME, 159, Node006, NEUTRAL_REACTION);
+    if (global_var(GVAR_WHORING_EXPERT) == 0) then begin
+        giq_option(4, NAME, 600, Node002a, NEUTRAL_REACTION);
     end
     if (map_var(12) != 0) then begin
-        giq_option(4, 48, 605, Node002b, 50);
+        giq_option(4, NAME, 605, Node002b, NEUTRAL_REACTION);
     end
-    giq_option(4, 48, 160, Node999, 50);
-    if (global_var(1002) == 1) then begin
-        giq_option(4, 48, 513, Node011, 50);
+    giq_option(4, NAME, 160, Node999, NEUTRAL_REACTION);
+    if (global_var(GVAR_ENTER_ELDRIDGE_BASEMENT) == 1) then begin
+        giq_option(4, NAME, 513, Node011, NEUTRAL_REACTION);
     end
-    if ((global_var(996) == 2) and (global_var(1002) == 5)) then begin
-        giq_option(4, 48, 526, Node016, 50);
+    if ((global_var(GVAR_KILL_TREASURE_HUNTER) == 2) and (global_var(GVAR_ENTER_ELDRIDGE_BASEMENT) == 5)) then begin
+        giq_option(4, NAME, 526, Node016, NEUTRAL_REACTION);
     end
 end
 
 procedure Node002a
 begin
-    set_global_var(887, 1);
-    gsay_reply(48, 601);
-    giq_option(4, 48, 602, Node002, 50);
+    set_global_var(GVAR_WHORING_EXPERT, 1);
+    Reply(601);
+    giq_option(4, NAME, 602, Node002, NEUTRAL_REACTION);
 end
 
 procedure Node002b
 begin
     if (map_var(12) == 2) then begin
-        gsay_reply(48, 606);
+        Reply(606);
     end
     else begin
-        gsay_reply(48, 607);
+        Reply(607);
     end
-    giq_option(4, 48, 602, Node002, 50);
+    giq_option(4, NAME, 602, Node002, NEUTRAL_REACTION);
 end
 
 procedure Node003
 begin
-    gsay_reply(48, 161);
-    giq_option(4, 48, 162, Node995, 50);
-    giq_option(4, 48, 163, Node006, 50);
-    giq_option(4, 48, 164, Node999, 50);
-    if (global_var(1002) == 1) then begin
-        giq_option(4, 48, 513, Node011, 50);
+    Reply(161);
+    giq_option(4, NAME, 162, Node995, NEUTRAL_REACTION);
+    giq_option(4, NAME, 163, Node006, NEUTRAL_REACTION);
+    giq_option(4, NAME, 164, Node999, NEUTRAL_REACTION);
+    if (global_var(GVAR_ENTER_ELDRIDGE_BASEMENT) == 1) then begin
+        giq_option(4, NAME, 513, Node011, NEUTRAL_REACTION);
     end
-    if ((global_var(996) == 2) and (global_var(1002) == 5)) then begin
-        giq_option(4, 48, 526, Node016, 50);
+    if ((global_var(GVAR_KILL_TREASURE_HUNTER) == 2) and (global_var(GVAR_ENTER_ELDRIDGE_BASEMENT) == 5)) then begin
+        giq_option(4, NAME, 526, Node016, NEUTRAL_REACTION);
     end
 end
 
 procedure Node004
 begin
-    gsay_reply(48, 165);
-    giq_option(4, 48, 166, Node995, 50);
-    giq_option(4, 48, 167, Node006, 50);
-    giq_option(4, 48, 168, Node999, 50);
-    if (global_var(1002) == 1) then begin
-        giq_option(4, 48, 513, Node011, 50);
+    Reply(165);
+    giq_option(4, NAME, 166, Node995, NEUTRAL_REACTION);
+    giq_option(4, NAME, 167, Node006, NEUTRAL_REACTION);
+    giq_option(4, NAME, 168, Node999, NEUTRAL_REACTION);
+    if (global_var(GVAR_ENTER_ELDRIDGE_BASEMENT) == 1) then begin
+        giq_option(4, NAME, 513, Node011, NEUTRAL_REACTION);
     end
-    if ((global_var(996) == 2) and (global_var(1002) == 5)) then begin
-        giq_option(4, 48, 526, Node016, 50);
+    if ((global_var(GVAR_KILL_TREASURE_HUNTER) == 2) and (global_var(GVAR_ENTER_ELDRIDGE_BASEMENT) == 5)) then begin
+        giq_option(4, NAME, 526, Node016, NEUTRAL_REACTION);
     end
 end
 
 procedure Node005
 begin
-    gsay_reply(48, 169);
-    giq_option(4, 48, 170, Node999, 50);
+    Reply(169);
+    giq_option(4, NAME, 170, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node006
 begin
-    gsay_reply(48, 171);
-    giq_option(4, 48, 172, Node008, 50);
-    giq_option(4, 48, 173, Node006b, 50);
-    if ((metarule(17, 11) != 0) == 0) then begin
-        giq_option(4, 48, 174, Node010, 50);
+    Reply(171);
+    giq_option(4, NAME, 172, Node008, NEUTRAL_REACTION);
+    giq_option(4, NAME, 173, Node006b, NEUTRAL_REACTION);
+    if ((town_known(11) != 0) == 0) then begin
+        giq_option(4, NAME, 174, Node010, NEUTRAL_REACTION);
     end
-    giq_option(4, 48, 175, Node995, 50);
-    giq_option(4, 48, 176, Node999, 50);
+    giq_option(4, NAME, 175, Node995, NEUTRAL_REACTION);
+    giq_option(4, NAME, 176, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node007
 begin
-    gsay_reply(48, 177);
-    giq_option(-3, 48, 178, Node995, 50);
-    giq_option(-3, 48, 179, Node999, 50);
+    Reply(177);
+    giq_option(-3, NAME, 178, Node995, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 179, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node008
 begin
-    gsay_reply(48, 180);
-    giq_option(4, 48, 181, Node006, 50);
+    Reply(180);
+    giq_option(4, NAME, 181, Node006, NEUTRAL_REACTION);
 end
 
 procedure Node009
 begin
     set_local_var(6, 1);
-    if (metarule(17, 13) == 0) then begin
+    if (town_known(13) == 0) then begin
         debug_msg("  mark_on_map(" + 13 + ")");
         mark_area_known(0, 13, 1);
     end
-    gsay_reply(48, 182);
-    giq_option(4, 48, 183, Node006, 50);
+    Reply(182);
+    giq_option(4, NAME, 183, Node006, NEUTRAL_REACTION);
 end
 
 procedure Node010
 begin
-    gsay_reply(48, 184);
-    giq_option(4, 48, 185, Node006, 50);
+    Reply(184);
+    giq_option(4, NAME, 185, Node006, NEUTRAL_REACTION);
 end
 
 procedure Node011
 begin
-    gsay_reply(48, 514);
-    giq_option(4, 48, 515, Node012, 50);
+    Reply(514);
+    giq_option(4, NAME, 515, Node012, NEUTRAL_REACTION);
 end
 
 procedure Node012
 begin
-    gsay_reply(48, 516);
-    giq_option(4, 48, 517, Node013, 50);
+    Reply(516);
+    giq_option(4, NAME, 517, Node013, NEUTRAL_REACTION);
 end
 
 procedure Node013
 begin
-    gsay_reply(48, 518);
-    giq_option(4, 48, 519, Node014b, 50);
-    if (has_skill(dude_obj, 14) > 100) then begin
-        giq_option(4, 48, 520, Node014, 50);
+    Reply(518);
+    giq_option(4, NAME, 519, Node014b, NEUTRAL_REACTION);
+    if (has_skill(dude_obj, SKILL_CONVERSANT) > 100) then begin
+        giq_option(4, NAME, 520, Node014, NEUTRAL_REACTION);
     end
-    if (get_pc_stat(4) < 0) then begin
-        giq_option(4, 48, 521, Node015, 50);
+    if (get_pc_stat(PCSTAT_karma) < 0) then begin
+        giq_option(4, NAME, 521, Node015, NEUTRAL_REACTION);
     end
 end
 
 procedure Node014
 begin
-    gsay_reply(48, 522);
-    giq_option(4, 48, 523, Node014a, 50);
+    Reply(522);
+    giq_option(4, NAME, 523, Node014a, NEUTRAL_REACTION);
 end
 
 procedure Node014a
 begin
-    set_global_var(1002, 2);
-    restock_obj := create_object_sid(572, 0, 0, -1);
+    set_global_var(GVAR_ENTER_ELDRIDGE_BASEMENT, 2);
+    restock_obj := create_object(572, 0, 0);
     add_mult_objs_to_inven(dude_obj, restock_obj, 1);
 end
 
 procedure Node014b
 begin
-    set_global_var(1002, 3);
+    set_global_var(GVAR_ENTER_ELDRIDGE_BASEMENT, 3);
 end
 
 procedure Node015
 begin
-    gsay_reply(48, 524);
-    giq_option(4, 48, 525, Node014a, 50);
+    Reply(524);
+    giq_option(4, NAME, 525, Node014a, NEUTRAL_REACTION);
 end
 
 procedure Node016
 begin
-    gsay_reply(48, 527);
-    giq_option(4, 48, 528, Node999, 50);
-    set_global_var(1002, 4);
-    restock_obj := create_object_sid(572, 0, 0, -1);
+    Reply(527);
+    giq_option(4, NAME, 528, Node999, NEUTRAL_REACTION);
+    set_global_var(GVAR_ENTER_ELDRIDGE_BASEMENT, 4);
+    restock_obj := create_object(572, 0, 0);
     add_mult_objs_to_inven(dude_obj, restock_obj, 1);
 end
 
@@ -731,15 +731,15 @@ begin
         gdialog_mod_barter(30);
     end
     else begin
-        gdialog_mod_barter(0);
+        gdialog_barter;
     end
-    gsay_reply(48, 550);
-    giq_option(-3, 48, 179, Node999, 50);
+    Reply(550);
+    giq_option(-3, NAME, 179, Node999, NEUTRAL_REACTION);
     if (local_var(9) > 0) then begin
-        giq_option(4, 48, 170, Node999, 50);
+        giq_option(4, NAME, 170, Node999, NEUTRAL_REACTION);
     end
     else begin
-        giq_option(4, 48, 168, Node999, 50);
+        giq_option(4, NAME, 168, Node999, NEUTRAL_REACTION);
     end
 end
 
@@ -749,21 +749,21 @@ begin
         gdialog_mod_barter(30);
     end
     else begin
-        gdialog_mod_barter(0);
+        gdialog_barter;
     end
-    gsay_reply(48, 550);
-    giq_option(-3, 48, 179, Node999, 50);
+    Reply(550);
+    giq_option(-3, NAME, 179, Node999, NEUTRAL_REACTION);
     if (local_var(9) > 0) then begin
-        giq_option(4, 48, 170, Node999, 50);
+        giq_option(4, NAME, 170, Node999, NEUTRAL_REACTION);
     end
     else begin
-        giq_option(4, 48, 168, Node999, 50);
+        giq_option(4, NAME, 168, Node999, NEUTRAL_REACTION);
     end
 end
 
 procedure Node994
 begin
-    if (get_critter_stat(dude_obj, 34) == 0) then begin
+    if (dude_gender == 0) then begin
         call Node003();
     end
     else begin
@@ -779,18 +779,18 @@ begin
         call Node009();
     end
     else begin
-        gsay_reply(48, 192);
-        giq_option(4, 48, 191, Node006, 50);
-        giq_option(4, 48, 193, Node999, 50);
+        Reply(192);
+        giq_option(4, NAME, 191, Node006, NEUTRAL_REACTION);
+        giq_option(4, NAME, 193, Node999, NEUTRAL_REACTION);
     end
 end
 
 procedure Node992
 begin
     if (item_caps_total(dude_obj) < 50) then begin
-        gsay_reply(48, 192);
-        giq_option(4, 48, 191, Node006, 50);
-        giq_option(4, 48, 193, Node999, 50);
+        Reply(192);
+        giq_option(4, NAME, 191, Node006, NEUTRAL_REACTION);
+        giq_option(4, NAME, 193, Node999, NEUTRAL_REACTION);
     end
     else begin
         if (is_success(roll_vs_skill(dude_obj, 14, 0))) then begin
@@ -799,16 +799,16 @@ begin
             call Node009();
         end
         else begin
-            gsay_reply(48, 194);
-            giq_option(4, 48, 191, Node006, 50);
-            giq_option(4, 48, 190, Node999, 50);
+            Reply(194);
+            giq_option(4, NAME, 191, Node006, NEUTRAL_REACTION);
+            giq_option(4, NAME, 190, Node999, NEUTRAL_REACTION);
         end
     end
 end
 
 procedure Node991
 begin
-    if (has_trait(0, dude_obj, 10) or (get_critter_stat(dude_obj, 0) >= 8)) then begin
+    if (has_trait(TRAIT_PERK, dude_obj, PERK_presence) or (dude_strength >= 8)) then begin
         set_local_var(3, local_var(3) + -10);
         set_local_var(0, local_var(0) + -10);
         if (local_var(0) <= -75) then begin
@@ -867,19 +867,18 @@ begin
         call Node009();
     end
     else begin
-        gsay_reply(48, 186);
-        giq_option(4, 48, 191, Node006, 50);
-        if (((10 * has_trait(0, dude_obj, 17)) + has_skill(dude_obj, 15)) > 65) then begin
-            giq_option(4, 48, 188, Node992, 50);
+        Reply(186);
+        giq_option(4, NAME, 191, Node006, NEUTRAL_REACTION);
+        if (((10 * has_trait(TRAIT_PERK, dude_obj, PERK_master_trader)) + has_skill(dude_obj, SKILL_BARTER)) > 65) then begin
+            giq_option(4, NAME, 188, Node992, NEUTRAL_REACTION);
         end
         else begin
-            giq_option(4, 48, 187, Node993, 50);
+            giq_option(4, NAME, 187, Node993, NEUTRAL_REACTION);
         end
-        if (has_skill(dude_obj, 14) > 70) then begin
-            giq_option(6, 48, 195, Node009, 50);
+        if (has_skill(dude_obj, SKILL_CONVERSANT) > 70) then begin
+            giq_option(6, NAME, 195, Node009, NEUTRAL_REACTION);
         end
-        giq_option(4, 48, 189, Node991, 50);
-        giq_option(4, 48, 190, Node999, 50);
+        giq_option(4, NAME, 189, Node991, NEUTRAL_REACTION);
+        giq_option(4, NAME, 190, Node999, NEUTRAL_REACTION);
     end
 end
-
