diff --git a/Mods/MegaMod/Original/DEMOGEN.ssl b/Mods/MegaMod/ReDefine/DEMOGEN.ssl
index 46dbaf5..f7002b0 100644
--- a/Mods/MegaMod/Original/DEMOGEN.ssl
+++ b/Mods/MegaMod/ReDefine/DEMOGEN.ssl
@@ -22,10 +22,10 @@ procedure look_p_proc
 begin
     script_overrides;
     if (Broken_Gen) then begin
-        display_msg(message_str(1495, 100));
+        display_mstr(100);
     end
     else begin
-        display_msg(message_str(1495, 101));
+        display_mstr(101);
     end
 end
 
@@ -33,10 +33,10 @@ procedure description_p_proc
 begin
     script_overrides;
     if (Broken_Gen) then begin
-        display_msg(message_str(1495, 100));
+        display_mstr(100);
     end
     else begin
-        display_msg(message_str(1495, 101));
+        display_mstr(101);
     end
 end
 
@@ -45,44 +45,44 @@ begin
     script_overrides;
     if (map_var(7) > 0) then begin
         if (is_success(roll_vs_skill(dude_obj, 8, -10))) then begin
-            display_msg(message_str(1495, 123));
+            display_mstr(123);
             if (map_var(2) == 0) then begin
-                display_msg(message_str(1495, 118));
+                display_mstr(118);
             end
             if (map_var(2) == 1) then begin
-                display_msg(message_str(1495, 119));
+                display_mstr(119);
             end
             if (map_var(2) == 2) then begin
-                display_msg(message_str(1495, 118));
+                display_mstr(118);
             end
         end
         else begin
-            display_msg(message_str(1495, 122));
+            display_mstr(122);
             set_map_var(3, map_var(3) + 1);
             if (map_var(3) == 1) then begin
-                float_msg(Phrax_Pointer, message_str(1494, 131), 0);
+                float_msg(Phrax_Pointer, message_str(SCRIPT_PHRAX, 131), FLOAT_MSG_NORMAL);
             end
             if (map_var(3) == 2) then begin
-                float_msg(Phrax_Pointer, message_str(1494, 132), 0);
+                float_msg(Phrax_Pointer, message_str(SCRIPT_PHRAX, 132), FLOAT_MSG_NORMAL);
             end
             if (map_var(3) == 3) then begin
-                float_msg(Phrax_Pointer, message_str(1494, 133), 0);
+                float_msg(Phrax_Pointer, message_str(SCRIPT_PHRAX, 133), FLOAT_MSG_NORMAL);
             end
             if (map_var(3) == 4) then begin
-                float_msg(Phrax_Pointer, message_str(1494, 134), 0);
+                float_msg(Phrax_Pointer, message_str(SCRIPT_PHRAX, 134), FLOAT_MSG_NORMAL);
                 set_map_var(10, 1);
             end
         end
     end
     else begin
         if (map_var(2) == 0) then begin
-            display_msg(message_str(1495, 118));
+            display_mstr(118);
         end
         if (map_var(2) == 1) then begin
-            display_msg(message_str(1495, 119));
+            display_mstr(119);
         end
         if (map_var(2) == 2) then begin
-            display_msg(message_str(1495, 118));
+            display_mstr(118);
         end
     end
 end
@@ -97,27 +97,27 @@ begin
                 if (not(Broken_Gen)) then begin
                     set_map_var(2, 1);
                     give_exp_points(100);
-                    display_msg(message_str(1495, 120));
-                    set_global_var(0, global_var(0) - 50);
+                    display_mstr(120);
+                    set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) - 50);
                     Broken_Gen := 1;
                 end
                 else begin
-                    display_msg(message_str(1495, 103));
+                    display_mstr(103);
                 end
             end
             if (source_obj == dude_obj) then begin
                 LVar0 := roll_vs_skill(source_obj, 8, -10);
                 if (is_success(LVar0)) then begin
-                    display_msg(message_str(1495, 123));
+                    display_mstr(123);
                     if ((map_var(7) > 0) and (map_var(6) > 0)) then begin
                         if (not(Broken_Gen)) then begin
                             set_map_var(2, 1);
                             give_exp_points(250);
-                            display_msg(message_str(964, 10400));
+                            display_msg(message_str(SCRIPT_KLATRAP, 10400));
                             Broken_Gen := 1;
                         end
                         else begin
-                            display_msg(message_str(1495, 103));
+                            display_mstr(103);
                         end
                     end
                     else begin
@@ -125,12 +125,12 @@ begin
                             if (not(Broken_Gen)) then begin
                                 set_map_var(2, 1);
                                 give_exp_points(100);
-                                display_msg(message_str(1495, 120));
-                                set_global_var(0, global_var(0) - 50);
+                                display_mstr(120);
+                                set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) - 50);
                                 Broken_Gen := 1;
                             end
                             else begin
-                                display_msg(message_str(1495, 103));
+                                display_mstr(103);
                             end
                         end
                     end
@@ -139,16 +139,16 @@ begin
                     if (map_var(7) > 0) then begin
                         set_map_var(3, map_var(3) + 1);
                         if (map_var(3) == 1) then begin
-                            float_msg(Phrax_Pointer, message_str(1494, 131), 0);
+                            float_msg(Phrax_Pointer, message_str(SCRIPT_PHRAX, 131), FLOAT_MSG_NORMAL);
                         end
                         if (map_var(3) == 2) then begin
-                            float_msg(Phrax_Pointer, message_str(1494, 132), 0);
+                            float_msg(Phrax_Pointer, message_str(SCRIPT_PHRAX, 132), FLOAT_MSG_NORMAL);
                         end
                         if (map_var(3) == 3) then begin
-                            float_msg(Phrax_Pointer, message_str(1494, 133), 0);
+                            float_msg(Phrax_Pointer, message_str(SCRIPT_PHRAX, 133), FLOAT_MSG_NORMAL);
                         end
                         if (map_var(3) == 4) then begin
-                            float_msg(Phrax_Pointer, message_str(1494, 134), 0);
+                            float_msg(Phrax_Pointer, message_str(SCRIPT_PHRAX, 134), FLOAT_MSG_NORMAL);
                             set_map_var(10, 1);
                         end
                     end
@@ -156,28 +156,28 @@ begin
             end
         end
         else begin
-            display_msg(message_str(1495, 104));
+            display_mstr(104);
         end
     end
     if ((map_var(15) == 0) and (map_var(7) > 0)) then begin
-        display_msg(message_str(1495, 124));
+        display_mstr(124);
         set_map_var(3, map_var(3) + 1);
         if (map_var(3) == 1) then begin
-            float_msg(Phrax_Pointer, message_str(1494, 131), 0);
+            float_msg(Phrax_Pointer, message_str(SCRIPT_PHRAX, 131), FLOAT_MSG_NORMAL);
         end
         if (map_var(3) == 2) then begin
-            float_msg(Phrax_Pointer, message_str(1494, 132), 0);
+            float_msg(Phrax_Pointer, message_str(SCRIPT_PHRAX, 132), FLOAT_MSG_NORMAL);
         end
         if (map_var(3) == 3) then begin
-            float_msg(Phrax_Pointer, message_str(1494, 133), 0);
+            float_msg(Phrax_Pointer, message_str(SCRIPT_PHRAX, 133), FLOAT_MSG_NORMAL);
         end
         if (map_var(3) == 4) then begin
-            float_msg(Phrax_Pointer, message_str(1494, 134), 0);
+            float_msg(Phrax_Pointer, message_str(SCRIPT_PHRAX, 134), FLOAT_MSG_NORMAL);
             set_map_var(10, 1);
         end
     end
     if ((map_var(15) == 0) and (map_var(7) == 0)) then begin
-        display_msg(message_str(1495, 126));
+        display_mstr(126);
     end
     return 0;
     LVar0 := 0;
@@ -191,25 +191,25 @@ begin
         if (map_var(15) == 0) then begin
             LVar0 := roll_vs_skill(source_obj, 12, 0);
             if (is_success(LVar0)) then begin
-                display_msg(message_str(1495, 106));
+                display_mstr(106);
                 set_map_var(15, 1);
             end
             else begin
-                display_msg(message_str(1495, 107));
+                display_mstr(107);
             end
         end
         else begin
             if ((map_var(15) == 1) and (map_var(2) == 0)) then begin
-                display_msg(message_str(1495, 108));
+                display_mstr(108);
             end
             else begin
                 if (map_var(2) == 1) then begin
                     LVar0 := roll_vs_skill(source_obj, 12, 0);
                     if (is_success(LVar0)) then begin
-                        display_msg(message_str(1495, 109));
+                        display_mstr(109);
                     end
                     else begin
-                        display_msg(message_str(1495, 110));
+                        display_mstr(110);
                     end
                 end
             end
@@ -217,42 +217,42 @@ begin
     end
     if (action_being_used == 13) then begin
         if (map_var(7) > 0) then begin
-            if (has_skill(dude_obj, 8) >= 60) then begin
-                display_msg(message_str(1495, 123));
+            if (has_skill(dude_obj, SKILL_SNEAK) >= 60) then begin
+                display_mstr(123);
                 if (map_var(2) == 0) then begin
-                    display_msg(message_str(1495, 118));
+                    display_mstr(118);
                 end
                 if (map_var(2) == 1) then begin
-                    display_msg(message_str(1495, 119));
+                    display_mstr(119);
                 end
                 if (map_var(2) == 2) then begin
-                    display_msg(message_str(1495, 118));
+                    display_mstr(118);
                 end
             end
             else begin
-                display_msg(message_str(1495, 122));
+                display_mstr(122);
                 set_map_var(3, map_var(3) + 1);
                 if (map_var(3) == 1) then begin
-                    float_msg(Phrax_Pointer, message_str(1494, 131), 0);
+                    float_msg(Phrax_Pointer, message_str(SCRIPT_PHRAX, 131), FLOAT_MSG_NORMAL);
                 end
                 if (map_var(3) == 2) then begin
-                    float_msg(Phrax_Pointer, message_str(1494, 132), 0);
+                    float_msg(Phrax_Pointer, message_str(SCRIPT_PHRAX, 132), FLOAT_MSG_NORMAL);
                 end
                 if (map_var(3) == 3) then begin
-                    float_msg(Phrax_Pointer, message_str(1494, 133), 0);
+                    float_msg(Phrax_Pointer, message_str(SCRIPT_PHRAX, 133), FLOAT_MSG_NORMAL);
                 end
                 if (map_var(3) == 4) then begin
-                    float_msg(Phrax_Pointer, message_str(1494, 134), 0);
+                    float_msg(Phrax_Pointer, message_str(SCRIPT_PHRAX, 134), FLOAT_MSG_NORMAL);
                     set_map_var(10, 1);
                 end
             end
         end
         if (map_var(2) == 2) then begin
-            display_msg(message_str(1495, 112));
+            display_mstr(112);
         end
         if (map_var(2) == 1) then begin
             if (repair_attempt > 3) then begin
-                display_msg(message_str(1495, 127));
+                display_mstr(127);
                 call cantfix();
             end
             else begin
@@ -261,13 +261,13 @@ begin
                     game_time_advance(200000);
                     Broken_Gen := 0;
                     set_map_var(2, 2);
-                    display_msg(message_str(1495, 114));
-                    set_global_var(0, global_var(0) + 25);
+                    display_mstr(114);
+                    set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + 25);
                     give_exp_points(1000);
                     gfade_in(600);
                 end
                 else begin
-                    display_msg(message_str(1495, 116));
+                    display_mstr(116);
                     repair_attempt := repair_attempt + 1;
                 end
             end
@@ -275,14 +275,14 @@ begin
         if ((map_var(2) == 0) and (map_var(7) == 0)) then begin
             set_map_var(2, 1);
             give_exp_points(100);
-            display_msg(message_str(1495, 120));
-            set_global_var(0, global_var(0) - 25);
+            display_mstr(120);
+            set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) - 25);
             Broken_Gen := 1;
         end
     end
     else begin
         if ((action_being_used != 12) or (action_being_used != 13)) then begin
-            display_msg(message_str(1495, 117));
+            display_mstr(117);
         end
     end
     return 0;
@@ -292,4 +292,3 @@ end
 procedure cantfix
 begin
 end
-
