diff --git a/Mods/MegaMod/Original/DEMODOG.ssl b/Mods/MegaMod/ReDefine/DEMODOG.ssl
index 1996c0a..b52b0a1 100644
--- a/Mods/MegaMod/Original/DEMODOG.ssl
+++ b/Mods/MegaMod/ReDefine/DEMODOG.ssl
@@ -39,10 +39,10 @@ procedure look_at_p_proc
 begin
     script_overrides;
     if (not(dog_is_angry)) then begin
-        display_msg(message_str(1491, 100));
+        display_mstr(100);
     end
     else begin
-        display_msg(message_str(1491, 101));
+        display_mstr(101);
     end
 end
 
@@ -50,25 +50,25 @@ procedure description_p_proc
 begin
     script_overrides;
     if (not(dog_is_angry)) then begin
-        display_msg(message_str(1491, 100));
+        display_mstr(100);
     end
     else begin
-        display_msg(message_str(1491, 101));
+        display_mstr(101);
     end
 end
 
 procedure map_enter_p_proc
 begin
-    critter_add_trait(self_obj, 1, 5, 186);
-    if (global_var(801) == 0) then begin
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_DOG_TOUGH);
+    if (global_var(GVAR_RESERVED) == 0) then begin
         dog_is_angry := 1;
         set_map_var(14, 1);
-        critter_add_trait(self_obj, 1, 6, 51);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 51);
         add_timer_event(self_obj, game_ticks(25), 1);
     end
     else begin
-        if (global_var(801) == 1) then begin
-            critter_add_trait(self_obj, 1, 6, 0);
+        if (global_var(GVAR_RESERVED) == 1) then begin
+            critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 0);
             call follow_player();
         end
     end
@@ -94,10 +94,10 @@ begin
     dog_is_angry := map_var(14);
     if (not(dog_is_angry)) then begin
         if (tile_distance_objs(self_obj, dude_obj) > 4) then begin
-            animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(dude_obj), random(0, 5), 1), 1);
+            animate_move_obj_to_tile(self_obj, tile_num_in_direction(dude_tile, random(0, 5), 1), 1);
         end
         else begin
-            animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(dude_obj), random(0, 5), 3), 0);
+            animate_move_obj_to_tile(self_obj, tile_num_in_direction(dude_tile, random(0, 5), 3), 0);
         end
     end
 end
@@ -111,74 +111,74 @@ procedure use_obj_on_p_proc
 begin
     script_overrides;
     if ((obj_pid(obj_being_used_with) == 103) or (obj_pid(obj_being_used_with) == 81) or (obj_pid(obj_being_used_with) == 284)) then begin
-        set_global_var(801, 1);
-        display_msg(message_str(1491, 109));
+        set_global_var(GVAR_RESERVED, 1);
+        display_mstr(109);
         give_exp_points(550);
-        critter_add_trait(self_obj, 1, 6, 0);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 0);
         dog_is_angry := 0;
         set_map_var(14, 0);
         add_timer_event(self_obj, game_ticks(1), 2);
     end
     else begin
-        display_msg(message_str(1491, 111));
+        display_mstr(111);
     end
 end
 
 procedure destroy_p_proc
 begin
     if (source_obj == dude_obj) then begin
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(4) > (3 * global_var(5))) or (global_var(2) == 1)) and (global_var(1) == 0)) then begin
-            set_global_var(2, 1);
-            set_global_var(3, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(4) > (3 * global_var(GVAR_GOOD_MONSTER))) or (global_var(GVAR_CHAMPION_REPUTATION) == 1)) and (global_var(GVAR_CHILDKILLER_REPUTATION) == 0)) then begin
+            set_global_var(GVAR_CHAMPION_REPUTATION, 1);
+            set_global_var(GVAR_BERSERKER_REPUTATION, 0);
         end
-        if (metarule(51, self_obj) == 2) then begin
+        if (critter_kill_type(self_obj) == 2) then begin
             if (source_obj == dude_obj) then begin
-                set_global_var(2, 0);
-                set_global_var(1, global_var(1) + 1);
-                set_global_var(0, global_var(0) + -10);
+                set_global_var(GVAR_CHAMPION_REPUTATION, 0);
+                set_global_var(GVAR_CHILDKILLER_REPUTATION, global_var(GVAR_CHILDKILLER_REPUTATION) + 1);
+                set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + -10);
                 debug_msg("Player gains " + -10 + " Karma Points.");
-                set_global_var(37, 0);
-                set_global_var(38, 0);
-                set_global_var(39, 0);
-                set_global_var(40, 0);
-                set_global_var(41, 0);
-                set_global_var(42, 0);
-                set_global_var(43, 0);
-                set_global_var(44, 0);
-                set_global_var(45, 0);
-                if (global_var(0) >= 1000) then begin
-                    set_global_var(37, 1);
+                set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+                set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+                set_global_var(GVAR_KARMA_DEFENDER, 0);
+                set_global_var(GVAR_KARMA_WANDERER, 0);
+                set_global_var(GVAR_KARMA_BETRAYER, 0);
+                set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+                set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+                set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+                    set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
                 end
                 else begin
-                    if (global_var(0) >= 750) then begin
-                        set_global_var(38, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
                     end
                     else begin
-                        if (global_var(0) >= 500) then begin
-                            set_global_var(39, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                            set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                         end
                         else begin
-                            if (global_var(0) >= 250) then begin
-                                set_global_var(40, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                                set_global_var(GVAR_KARMA_DEFENDER, 1);
                             end
                             else begin
-                                if (global_var(0) > -250) then begin
-                                    set_global_var(41, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                                    set_global_var(GVAR_KARMA_WANDERER, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -500) then begin
-                                        set_global_var(42, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                        set_global_var(GVAR_KARMA_BETRAYER, 1);
                                     end
                                     else begin
-                                        if (global_var(0) > -750) then begin
-                                            set_global_var(43, 1);
+                                        if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                            set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                         end
                                         else begin
-                                            if (global_var(0) > -1000) then begin
-                                                set_global_var(44, 1);
+                                            if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                                set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                             end
                                             else begin
-                                                set_global_var(45, 1);
+                                                set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                             end
                                         end
                                     end
@@ -187,58 +187,58 @@ begin
                         end
                     end
                 end
-                set_global_var(55, global_var(808) + 5);
+                set_global_var(GVAR_TOWN_REP_NEW_RENO, global_var(GVAR_GANG_2) + 5);
                 debug_msg("Added " + 5 + " to Town Rep");
             end
         end
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(5) > (2 * global_var(4))) or (global_var(3) == 1))) then begin
-            set_global_var(3, 1);
-            set_global_var(2, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
+            set_global_var(GVAR_BERSERKER_REPUTATION, 1);
+            set_global_var(GVAR_CHAMPION_REPUTATION, 0);
         end
-        set_global_var(0, global_var(0) + -1);
+        set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + -1);
         debug_msg("Player gains " + -1 + " Karma Points.");
-        set_global_var(37, 0);
-        set_global_var(38, 0);
-        set_global_var(39, 0);
-        set_global_var(40, 0);
-        set_global_var(41, 0);
-        set_global_var(42, 0);
-        set_global_var(43, 0);
-        set_global_var(44, 0);
-        set_global_var(45, 0);
-        if (global_var(0) >= 1000) then begin
-            set_global_var(37, 1);
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+        set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+        set_global_var(GVAR_KARMA_DEFENDER, 0);
+        set_global_var(GVAR_KARMA_WANDERER, 0);
+        set_global_var(GVAR_KARMA_BETRAYER, 0);
+        set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+        set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+            set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
         end
         else begin
-            if (global_var(0) >= 750) then begin
-                set_global_var(38, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
             end
             else begin
-                if (global_var(0) >= 500) then begin
-                    set_global_var(39, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                 end
                 else begin
-                    if (global_var(0) >= 250) then begin
-                        set_global_var(40, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                        set_global_var(GVAR_KARMA_DEFENDER, 1);
                     end
                     else begin
-                        if (global_var(0) > -250) then begin
-                            set_global_var(41, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                            set_global_var(GVAR_KARMA_WANDERER, 1);
                         end
                         else begin
-                            if (global_var(0) > -500) then begin
-                                set_global_var(42, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                set_global_var(GVAR_KARMA_BETRAYER, 1);
                             end
                             else begin
-                                if (global_var(0) > -750) then begin
-                                    set_global_var(43, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -1000) then begin
-                                        set_global_var(44, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                     end
                                     else begin
-                                        set_global_var(45, 1);
+                                        set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                     end
                                 end
                             end
@@ -248,8 +248,8 @@ begin
             end
         end
     end
-    if (global_var(801) == 0) then begin
-        display_msg(message_str(1491, 110));
+    if (global_var(GVAR_RESERVED) == 0) then begin
+        display_mstr(110);
         give_exp_points(200);
         dog_is_angry := 0;
         set_map_var(14, 0);
@@ -259,20 +259,20 @@ end
 
 procedure growl
 begin
-    float_msg(self_obj, message_str(1491, 103), 0);
+    float_msg(self_obj, mstr(103), FLOAT_MSG_NORMAL);
     growling := 1;
     smartass2 := 1;
 end
 
 procedure do_dialogue
 begin
-    float_msg(self_obj, message_str(1491, 102), 0);
+    float_msg(self_obj, mstr(102), FLOAT_MSG_NORMAL);
 end
 
 procedure follow_player
 begin
     variable LVar0 := 0;
-    LVar0 := tile_num_in_direction(tile_num(dude_obj), random(0, 5), 1);
+    LVar0 := tile_num_in_direction(dude_tile, random(0, 5), 1);
     if (tile_distance_objs(self_obj, dude_obj) < 8) then begin
         animate_move_obj_to_tile(self_obj, LVar0, 0 bwor 16);
     end
@@ -280,14 +280,14 @@ begin
         animate_move_obj_to_tile(self_obj, LVar0, 1 bwor 16);
     end
     add_timer_event(self_obj, game_ticks(2), 2);
-    critter_add_trait(self_obj, 1, 6, 0);
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 0);
     return 0;
     LVar0 := 0;
 end
 
 procedure push_p_proc
 begin
-    if (global_var(801) == 1) then begin
+    if (global_var(GVAR_RESERVED) == 1) then begin
         call Node001();
     end
     else begin
@@ -297,6 +297,5 @@ end
 
 procedure Node001
 begin
-    float_msg(self_obj, message_str(1491, random(102, 107)), 8);
+    float_msg(self_obj, mstr(random(102, 107)), FLOAT_MSG_YELLOW);
 end
-
