diff --git a/Mods/MegaMod/Original/DEMOPHIL.ssl b/Mods/MegaMod/ReDefine/DEMOPHIL.ssl
index b7e53c9..4ecf521 100644
--- a/Mods/MegaMod/Original/DEMOPHIL.ssl
+++ b/Mods/MegaMod/ReDefine/DEMOPHIL.ssl
@@ -50,14 +50,14 @@ procedure start
 begin
     if (local_var(0) == 0) then begin
         set_local_var(0, 1);
-        critter_add_trait(self_obj, 1, 6, 184);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 184);
     end
 end
 
 procedure damage_p_proc
 begin
     if (source_obj == dude_obj) then begin
-        attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+        attack(dude_obj);
     end
 end
 
@@ -69,7 +69,7 @@ end
 procedure critter_p_proc
 begin
     if (fixed_param == 2) then begin
-        float_msg(self_obj, message_str(1493, 102), 0);
+        float_msg(self_obj, mstr(102), FLOAT_MSG_NORMAL);
     end
     if (dog_is_angry == 0) then begin
         animate_move_obj_to_tile(self_obj, 15686, 0);
@@ -79,55 +79,55 @@ end
 procedure destroy_p_proc
 begin
     if (source_obj == dude_obj) then begin
-        set_global_var(5, global_var(5) + 1);
-        if (metarule(51, self_obj) == 2) then begin
+        set_global_var(GVAR_GOOD_MONSTER, global_var(GVAR_GOOD_MONSTER) + 1);
+        if (critter_kill_type(self_obj) == KILL_TYPE_children_kills) then begin
             if (source_obj == dude_obj) then begin
-                set_global_var(2, 0);
-                set_global_var(1, global_var(1) + 1);
-                set_global_var(0, global_var(0) + -15);
+                set_global_var(GVAR_CHAMPION_REPUTATION, 0);
+                set_global_var(GVAR_CHILDKILLER_REPUTATION, global_var(GVAR_CHILDKILLER_REPUTATION) + 1);
+                set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + -15);
                 debug_msg("Player gains " + -15 + " Karma Points.");
-                set_global_var(37, 0);
-                set_global_var(38, 0);
-                set_global_var(39, 0);
-                set_global_var(40, 0);
-                set_global_var(41, 0);
-                set_global_var(42, 0);
-                set_global_var(43, 0);
-                set_global_var(44, 0);
-                set_global_var(45, 0);
-                if (global_var(0) >= 1000) then begin
-                    set_global_var(37, 1);
+                set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+                set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+                set_global_var(GVAR_KARMA_DEFENDER, 0);
+                set_global_var(GVAR_KARMA_WANDERER, 0);
+                set_global_var(GVAR_KARMA_BETRAYER, 0);
+                set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+                set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+                set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+                    set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
                 end
                 else begin
-                    if (global_var(0) >= 750) then begin
-                        set_global_var(38, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
                     end
                     else begin
-                        if (global_var(0) >= 500) then begin
-                            set_global_var(39, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                            set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                         end
                         else begin
-                            if (global_var(0) >= 250) then begin
-                                set_global_var(40, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                                set_global_var(GVAR_KARMA_DEFENDER, 1);
                             end
                             else begin
-                                if (global_var(0) > -250) then begin
-                                    set_global_var(41, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                                    set_global_var(GVAR_KARMA_WANDERER, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -500) then begin
-                                        set_global_var(42, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                        set_global_var(GVAR_KARMA_BETRAYER, 1);
                                     end
                                     else begin
-                                        if (global_var(0) > -750) then begin
-                                            set_global_var(43, 1);
+                                        if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                            set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                         end
                                         else begin
-                                            if (global_var(0) > -1000) then begin
-                                                set_global_var(44, 1);
+                                            if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                                set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                             end
                                             else begin
-                                                set_global_var(45, 1);
+                                                set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                             end
                                         end
                                     end
@@ -136,62 +136,62 @@ begin
                         end
                     end
                 end
-                set_global_var(808, global_var(808) + -5);
+                set_global_var(GVAR_GANG_2, global_var(GVAR_GANG_2) + -5);
                 debug_msg("Added " + -5 + " to Town Rep");
             end
         end
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(4) > (3 * global_var(5))) or (global_var(2) == 1)) and (global_var(1) == 0)) then begin
-            set_global_var(2, 1);
-            set_global_var(3, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(4) > (3 * global_var(GVAR_GOOD_MONSTER))) or (global_var(GVAR_CHAMPION_REPUTATION) == 1)) and (global_var(GVAR_CHILDKILLER_REPUTATION) == 0)) then begin
+            set_global_var(GVAR_CHAMPION_REPUTATION, 1);
+            set_global_var(GVAR_BERSERKER_REPUTATION, 0);
         end
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(5) > (2 * global_var(4))) or (global_var(3) == 1))) then begin
-            set_global_var(3, 1);
-            set_global_var(2, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
+            set_global_var(GVAR_BERSERKER_REPUTATION, 1);
+            set_global_var(GVAR_CHAMPION_REPUTATION, 0);
         end
-        set_global_var(0, global_var(0) + -5);
+        set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + -5);
         debug_msg("Player gains " + -5 + " Karma Points.");
-        set_global_var(37, 0);
-        set_global_var(38, 0);
-        set_global_var(39, 0);
-        set_global_var(40, 0);
-        set_global_var(41, 0);
-        set_global_var(42, 0);
-        set_global_var(43, 0);
-        set_global_var(44, 0);
-        set_global_var(45, 0);
-        if (global_var(0) >= 1000) then begin
-            set_global_var(37, 1);
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+        set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+        set_global_var(GVAR_KARMA_DEFENDER, 0);
+        set_global_var(GVAR_KARMA_WANDERER, 0);
+        set_global_var(GVAR_KARMA_BETRAYER, 0);
+        set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+        set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+            set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
         end
         else begin
-            if (global_var(0) >= 750) then begin
-                set_global_var(38, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
             end
             else begin
-                if (global_var(0) >= 500) then begin
-                    set_global_var(39, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                 end
                 else begin
-                    if (global_var(0) >= 250) then begin
-                        set_global_var(40, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                        set_global_var(GVAR_KARMA_DEFENDER, 1);
                     end
                     else begin
-                        if (global_var(0) > -250) then begin
-                            set_global_var(41, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                            set_global_var(GVAR_KARMA_WANDERER, 1);
                         end
                         else begin
-                            if (global_var(0) > -500) then begin
-                                set_global_var(42, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                set_global_var(GVAR_KARMA_BETRAYER, 1);
                             end
                             else begin
-                                if (global_var(0) > -750) then begin
-                                    set_global_var(43, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -1000) then begin
-                                        set_global_var(44, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                     end
                                     else begin
-                                        set_global_var(45, 1);
+                                        set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                     end
                                 end
                             end
@@ -200,7 +200,7 @@ begin
                 end
             end
         end
-        set_global_var(808, global_var(808) + -3);
+        set_global_var(GVAR_GANG_2, global_var(GVAR_GANG_2) + -3);
         debug_msg("Added " + -3 + " to Town Rep");
     end
 end
@@ -209,17 +209,17 @@ procedure look_at_p_proc
 begin
     script_overrides;
     if (known == 1) then begin
-        display_msg(message_str(1493, 101));
+        display_mstr(101);
     end
     else begin
-        display_msg(message_str(1493, 100));
+        display_mstr(100);
     end
 end
 
 procedure description_p_proc
 begin
     script_overrides;
-    display_msg(message_str(1493, 100));
+    display_mstr(100);
 end
 
 procedure do_dialogue
@@ -249,7 +249,7 @@ end
 
 procedure run_away
 begin
-    float_msg(self_obj, message_str(1493, 124), 0);
+    float_msg(self_obj, mstr(124), FLOAT_MSG_NORMAL);
     animate_move_obj_to_tile(self_obj, 16892, 1);
     growling := 0;
     add_timer_event(self_obj, game_ticks(random(10, 30)), 1);
@@ -277,84 +277,84 @@ end
 
 procedure Node000
 begin
-    float_msg(self_obj, message_str(1493, 103), 0);
+    float_msg(self_obj, mstr(103), FLOAT_MSG_NORMAL);
 end
 
 procedure Node001
 begin
-    float_msg(self_obj, message_str(1493, 104), 0);
+    float_msg(self_obj, mstr(104), FLOAT_MSG_NORMAL);
 end
 
 procedure Node002
 begin
     set_map_var(0, 1);
-    if (get_critter_stat(dude_obj, 34) == 0) then begin
-        gsay_reply(1493, message_str(1493, 105) + message_str(1493, 107));
+    if (dude_gender == 0) then begin
+        Reply(mstr(105) + mstr(107));
     end
     else begin
-        gsay_reply(1493, message_str(1493, 106) + message_str(1493, 107));
+        Reply(mstr(106) + mstr(107));
     end
-    giq_option(4, 1493, 108, Node004, 50);
-    giq_option(4, 1493, 109, Node999, 50);
-    giq_option(4, 1493, 110, Node005, 50);
-    giq_option(4, 1493, 111, Node004, 50);
+    giq_option(4, NAME, 108, Node004, NEUTRAL_REACTION);
+    giq_option(4, NAME, 109, Node999, NEUTRAL_REACTION);
+    giq_option(4, NAME, 110, Node005, NEUTRAL_REACTION);
+    giq_option(4, NAME, 111, Node004, NEUTRAL_REACTION);
 end
 
 procedure Node004
 begin
-    gsay_reply(1493, 113);
-    giq_option(4, 1493, 114, Node999, 50);
-    giq_option(4, 1493, 115, Node005, 50);
+    Reply(113);
+    giq_option(4, NAME, 114, Node999, NEUTRAL_REACTION);
+    giq_option(4, NAME, 115, Node005, NEUTRAL_REACTION);
 end
 
 procedure Node005
 begin
-    gsay_reply(1493, 116);
-    giq_option(4, 1493, 117, Node999, 50);
-    giq_option(4, 1493, 118, Node006, 50);
-    giq_option(4, 1493, 119, Node007, 50);
+    Reply(116);
+    giq_option(4, NAME, 117, Node999, NEUTRAL_REACTION);
+    giq_option(4, NAME, 118, Node006, NEUTRAL_REACTION);
+    giq_option(4, NAME, 119, Node007, NEUTRAL_REACTION);
 end
 
 procedure Node006
 begin
-    gsay_reply(1493, 120);
-    giq_option(4, 1493, 122, Node007, 50);
+    Reply(120);
+    giq_option(4, NAME, 122, Node007, NEUTRAL_REACTION);
 end
 
 procedure Node007
 begin
-    gsay_reply(1493, 123);
-    giq_option(4, 1493, 125, Node999, 50);
+    Reply(123);
+    giq_option(4, NAME, 125, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node008
 begin
-    float_msg(self_obj, message_str(1493, 126), 0);
+    float_msg(self_obj, mstr(126), FLOAT_MSG_NORMAL);
 end
 
 procedure Node009
 begin
-    float_msg(self_obj, message_str(1493, 127), 0);
+    float_msg(self_obj, mstr(127), FLOAT_MSG_NORMAL);
 end
 
 procedure get_reaction
 begin
     if (not(SVar7)) then begin
         SVar7 := 1;
-        SVar5 := SVar5 + ((get_critter_stat(dude_obj, 3) - 5) * 5);
+        SVar5 := SVar5 + ((dude_charisma - 5) * 5);
         if (not(SVar8 == 1)) then begin
-            SVar5 := SVar5 - global_var(808);
+            SVar5 := SVar5 - global_var(GVAR_GANG_2);
         end
         else begin
-            SVar5 := SVar5 + global_var(808);
+            SVar5 := SVar5 + global_var(GVAR_GANG_2);
         end
-        if (global_var(1) > 2) then begin
+        if (global_var(GVAR_CHILDKILLER_REPUTATION) > 2) then begin
             SVar5 := SVar5 - 30;
         end
-        if ((global_var(4) > (3 * global_var(5))) or (global_var(2) == 1)) then begin
+        if ((global_var(GVAR_BAD_MONSTER) > (3 * global_var(GVAR_GOOD_MONSTER))) or (global_var(GVAR_CHAMPION_REPUTATION) == 1)) then begin
             SVar5 := SVar5 + 20;
         end
-        if ((global_var(5) > (2 * global_var(4))) or (global_var(3) == 1)) then begin
+        if ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1)) then begin
             SVar5 := SVar5 - 20;
         end
         call ReactToLevel();
@@ -444,4 +444,3 @@ begin
     end
     call LevelToReact();
 end
-
