diff --git a/Mods/MegaMod/Original/DEMOCOMP.ssl b/Mods/MegaMod/ReDefine/DEMOCOMP.ssl
index c87e9a6..07938b8 100644
--- a/Mods/MegaMod/Original/DEMOCOMP.ssl
+++ b/Mods/MegaMod/ReDefine/DEMOCOMP.ssl
@@ -16,7 +16,7 @@ variable used_before;
 
 procedure start
 begin
-    if ((script_action == 11) and (broken_gen != 1)) then begin
+    if ((script_action == talk_proc) and (broken_gen != 1)) then begin
         start_gdialog(1490, self_obj, 4, -1, -1);
         gsay_start;
         call Node001();
@@ -24,7 +24,7 @@ begin
         end_dialogue;
     end
     else begin
-        if ((script_action == 6) and (broken_gen != 1)) then begin
+        if ((script_action == use_proc) and (broken_gen != 1)) then begin
             script_overrides;
             if (WORKING == 1) then begin
                 dialogue_system_enter;
@@ -34,9 +34,9 @@ begin
             end
         end
         else begin
-            if ((script_action == 21) or (script_action == 3)) then begin
+            if ((script_action == look_at_proc) or (script_action == description_proc)) then begin
                 script_overrides;
-                display_msg(message_str(1490, 100));
+                display_mstr(100);
             end
         end
     end
@@ -50,21 +50,21 @@ procedure Node000
 begin
     variable LVar0 := 0;
     LVar0 := random(20, 40);
-    if (has_skill(dude_obj, 12) > has_skill(dude_obj, 13)) then begin
+    if (has_skill(dude_obj, SKILL_SCIENCE) > has_skill(dude_obj, SKILL_REPAIR)) then begin
         test := roll_vs_skill(dude_obj, 12, 0);
     end
     else begin
         test := roll_vs_skill(dude_obj, 13, 0);
     end
     if (1) then begin
-        display_msg(message_str(1490, 101));
+        display_mstr(101);
         WORKING := 1;
     end
     else begin
-        display_msg(message_str(1490, 103));
+        display_mstr(103);
         if (is_critical(test)) then begin
             critter_dmg(dude_obj, LVar0, 6 bwor 256);
-            display_msg(message_str(1490, 104));
+            display_mstr(104);
         end
     end
     LVar0 := 0;
@@ -72,41 +72,40 @@ end
 
 procedure Node001
 begin
-    gsay_reply(1490, 105);
-    giq_option(4, 1490, 106, Node002, 50);
-    giq_option(4, 1490, 107, Node999, 50);
+    Reply(105);
+    giq_option(4, NAME, 106, Node002, NEUTRAL_REACTION);
+    giq_option(4, NAME, 107, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node002
 begin
-    gsay_reply(1490, 108);
-    giq_option(4, 1490, 109, Node003, 50);
-    giq_option(4, 1490, 110, Node999, 50);
+    Reply(108);
+    giq_option(4, NAME, 109, Node003, NEUTRAL_REACTION);
+    giq_option(4, NAME, 110, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node003
 begin
-    gsay_reply(1490, 111);
-    giq_option(4, 1490, 112, Node004, 50);
-    giq_option(4, 1490, 113, Node005, 50);
-    giq_option(4, 1490, 114, Node999, 50);
+    Reply(111);
+    giq_option(4, NAME, 112, Node004, NEUTRAL_REACTION);
+    giq_option(4, NAME, 113, Node005, NEUTRAL_REACTION);
+    giq_option(4, NAME, 114, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node004
 begin
-    gsay_reply(1490, 115);
-    giq_option(4, 1490, 116, Node003, 50);
-    giq_option(4, 1490, 117, Node003, 50);
+    Reply(115);
+    giq_option(4, NAME, 116, Node003, NEUTRAL_REACTION);
+    giq_option(4, NAME, 117, Node003, NEUTRAL_REACTION);
 end
 
 procedure Node005
 begin
     if (not(used_before)) then begin
         give_exp_points(100);
-        display_msg(message_str(1490, 119));
+        display_mstr(119);
         used_before := 1;
         game_time_advance(4 * (60 * (10 * 10)));
     end
-    float_msg(dude_obj, message_str(1490, 118), 0);
+    float_msg(dude_obj, mstr(118), FLOAT_MSG_NORMAL);
 end
-
