diff --git a/Mods/MegaMod/Original/DIMETBOX.ssl b/Mods/MegaMod/ReDefine/DIMETBOX.ssl
index 00a4181..3c3d205 100644
--- a/Mods/MegaMod/Original/DIMETBOX.ssl
+++ b/Mods/MegaMod/ReDefine/DIMETBOX.ssl
@@ -151,15 +151,15 @@ begin
     variable LVar1 := 0;
     variable LVar2 := 0;
     den_metzger_box_obj := self_obj;
-    if (metarule(22, 0) == 0) then begin
+    if (is_loading_game == 0) then begin
         if (local_var(0) < game_time) then begin
-            if ((global_var(445) bwand 4194304) != 0) then begin
-                set_global_var(445, global_var(445) bwand (-1 - 4194304));
-                LVar0 := create_object_sid(130, 0, 0, -1);
+            if ((global_var(GVAR_DEN_FLAG_1) bwand 4194304) != 0) then begin
+                set_global_var(GVAR_DEN_FLAG_1, global_var(GVAR_DEN_FLAG_1) bwand (-1 - 4194304));
+                LVar0 := create_object(130, 0, 0);
                 move_obj_inven_to_obj(self_obj, LVar0);
                 destroy_object(LVar0);
             end
-            LVar1 := create_object_sid(128, 5, 0, -1);
+            LVar1 := create_object(128, 5, 0);
             if (obj_is_carrying_obj_pid(self_obj, 479) > 0) then begin
                 LVar2 := obj_carrying_pid_obj(self_obj, 479);
                 rm_obj_from_inven(self_obj, LVar2);
@@ -405,7 +405,7 @@ begin
                 rm_obj_from_inven(self_obj, LVar2);
                 add_obj_to_inven(LVar1, LVar2);
             end
-            global_temp := create_object_sid(128, 0, 0, -1);
+            global_temp := create_object(128, 0, 0);
             move_obj_inven_to_obj(self_obj, global_temp);
             destroy_object(global_temp);
             item_caps_adjust(self_obj, random(5, 250) - item_caps_total(self_obj));
@@ -413,7 +413,7 @@ begin
             if (obj_is_carrying_obj_pid(self_obj, 7) < restock_amt) then begin
                 if (100 >= random(1, 100)) then begin
                     if (obj_is_carrying_obj_pid(self_obj, 7) < restock_amt) then begin
-                        restock_obj := create_object_sid(7, 0, 0, -1);
+                        restock_obj := create_object(7, 0, 0);
                         add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 7));
                     end
                     else begin
@@ -426,17 +426,17 @@ begin
                                 restock_obj := obj_carrying_pid_obj(self_obj, 7);
                                 if (obj_type(self_obj) == 1) then begin
                                     restock_obj := obj_carrying_pid_obj(self_obj, 7);
-                                    if (critter_inven_obj(self_obj, 0) == restock_obj) then begin
-                                        if (obj_item_subtype(critter_inven_obj(self_obj, 0)) == 0) then begin
-                                            restock_obj := critter_inven_obj(self_obj, 0);
+                                    if (critter_inven_obj(self_obj, INVEN_TYPE_WORN) == restock_obj) then begin
+                                        if (obj_item_subtype(critter_inven_obj(self_obj, INVEN_TYPE_WORN)) == 0) then begin
+                                            restock_obj := critter_inven_obj(self_obj, INVEN_TYPE_WORN);
                                             debug_msg("armour pid == " + obj_pid(restock_obj));
                                             rm_obj_from_inven(self_obj, restock_obj);
                                             add_obj_to_inven(self_obj, restock_obj);
                                         end
                                     end
                                     else begin
-                                        if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                            metarule(43, self_obj);
+                                        if ((critter_inven_obj(self_obj, INVEN_TYPE_LEFT_HAND) == restock_obj) or (critter_inven_obj(self_obj, INVEN_TYPE_RIGHT_HAND) == restock_obj)) then begin
+                                            inven_unwield(self_obj);
                                         end
                                     end
                                 end
@@ -454,7 +454,7 @@ begin
             end
             else begin
                 if (obj_is_carrying_obj_pid(self_obj, 7) < restock_amt) then begin
-                    restock_obj := create_object_sid(7, 0, 0, -1);
+                    restock_obj := create_object(7, 0, 0);
                     add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 7));
                 end
                 else begin
@@ -467,17 +467,17 @@ begin
                             restock_obj := obj_carrying_pid_obj(self_obj, 7);
                             if (obj_type(self_obj) == 1) then begin
                                 restock_obj := obj_carrying_pid_obj(self_obj, 7);
-                                if (critter_inven_obj(self_obj, 0) == restock_obj) then begin
-                                    if (obj_item_subtype(critter_inven_obj(self_obj, 0)) == 0) then begin
-                                        restock_obj := critter_inven_obj(self_obj, 0);
+                                if (critter_inven_obj(self_obj, INVEN_TYPE_WORN) == restock_obj) then begin
+                                    if (obj_item_subtype(critter_inven_obj(self_obj, INVEN_TYPE_WORN)) == 0) then begin
+                                        restock_obj := critter_inven_obj(self_obj, INVEN_TYPE_WORN);
                                         debug_msg("armour pid == " + obj_pid(restock_obj));
                                         rm_obj_from_inven(self_obj, restock_obj);
                                         add_obj_to_inven(self_obj, restock_obj);
                                     end
                                 end
                                 else begin
-                                    if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                        metarule(43, self_obj);
+                                    if ((critter_inven_obj(self_obj, INVEN_TYPE_LEFT_HAND) == restock_obj) or (critter_inven_obj(self_obj, INVEN_TYPE_RIGHT_HAND) == restock_obj)) then begin
+                                        inven_unwield(self_obj);
                                     end
                                 end
                             end
@@ -496,7 +496,7 @@ begin
             if (obj_is_carrying_obj_pid(self_obj, 4) < restock_amt) then begin
                 if (100 >= random(1, 100)) then begin
                     if (obj_is_carrying_obj_pid(self_obj, 4) < restock_amt) then begin
-                        restock_obj := create_object_sid(4, 0, 0, -1);
+                        restock_obj := create_object(4, 0, 0);
                         add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 4));
                     end
                     else begin
@@ -509,17 +509,17 @@ begin
                                 restock_obj := obj_carrying_pid_obj(self_obj, 4);
                                 if (obj_type(self_obj) == 1) then begin
                                     restock_obj := obj_carrying_pid_obj(self_obj, 4);
-                                    if (critter_inven_obj(self_obj, 0) == restock_obj) then begin
-                                        if (obj_item_subtype(critter_inven_obj(self_obj, 0)) == 0) then begin
-                                            restock_obj := critter_inven_obj(self_obj, 0);
+                                    if (critter_inven_obj(self_obj, INVEN_TYPE_WORN) == restock_obj) then begin
+                                        if (obj_item_subtype(critter_inven_obj(self_obj, INVEN_TYPE_WORN)) == 0) then begin
+                                            restock_obj := critter_inven_obj(self_obj, INVEN_TYPE_WORN);
                                             debug_msg("armour pid == " + obj_pid(restock_obj));
                                             rm_obj_from_inven(self_obj, restock_obj);
                                             add_obj_to_inven(self_obj, restock_obj);
                                         end
                                     end
                                     else begin
-                                        if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                            metarule(43, self_obj);
+                                        if ((critter_inven_obj(self_obj, INVEN_TYPE_LEFT_HAND) == restock_obj) or (critter_inven_obj(self_obj, INVEN_TYPE_RIGHT_HAND) == restock_obj)) then begin
+                                            inven_unwield(self_obj);
                                         end
                                     end
                                 end
@@ -537,7 +537,7 @@ begin
             end
             else begin
                 if (obj_is_carrying_obj_pid(self_obj, 4) < restock_amt) then begin
-                    restock_obj := create_object_sid(4, 0, 0, -1);
+                    restock_obj := create_object(4, 0, 0);
                     add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 4));
                 end
                 else begin
@@ -550,17 +550,17 @@ begin
                             restock_obj := obj_carrying_pid_obj(self_obj, 4);
                             if (obj_type(self_obj) == 1) then begin
                                 restock_obj := obj_carrying_pid_obj(self_obj, 4);
-                                if (critter_inven_obj(self_obj, 0) == restock_obj) then begin
-                                    if (obj_item_subtype(critter_inven_obj(self_obj, 0)) == 0) then begin
-                                        restock_obj := critter_inven_obj(self_obj, 0);
+                                if (critter_inven_obj(self_obj, INVEN_TYPE_WORN) == restock_obj) then begin
+                                    if (obj_item_subtype(critter_inven_obj(self_obj, INVEN_TYPE_WORN)) == 0) then begin
+                                        restock_obj := critter_inven_obj(self_obj, INVEN_TYPE_WORN);
                                         debug_msg("armour pid == " + obj_pid(restock_obj));
                                         rm_obj_from_inven(self_obj, restock_obj);
                                         add_obj_to_inven(self_obj, restock_obj);
                                     end
                                 end
                                 else begin
-                                    if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                        metarule(43, self_obj);
+                                    if ((critter_inven_obj(self_obj, INVEN_TYPE_LEFT_HAND) == restock_obj) or (critter_inven_obj(self_obj, INVEN_TYPE_RIGHT_HAND) == restock_obj)) then begin
+                                        inven_unwield(self_obj);
                                     end
                                 end
                             end
@@ -579,7 +579,7 @@ begin
             if (obj_is_carrying_obj_pid(self_obj, 8) < restock_amt) then begin
                 if (100 >= random(1, 100)) then begin
                     if (obj_is_carrying_obj_pid(self_obj, 8) < restock_amt) then begin
-                        restock_obj := create_object_sid(8, 0, 0, -1);
+                        restock_obj := create_object(8, 0, 0);
                         add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 8));
                     end
                     else begin
@@ -592,17 +592,17 @@ begin
                                 restock_obj := obj_carrying_pid_obj(self_obj, 8);
                                 if (obj_type(self_obj) == 1) then begin
                                     restock_obj := obj_carrying_pid_obj(self_obj, 8);
-                                    if (critter_inven_obj(self_obj, 0) == restock_obj) then begin
-                                        if (obj_item_subtype(critter_inven_obj(self_obj, 0)) == 0) then begin
-                                            restock_obj := critter_inven_obj(self_obj, 0);
+                                    if (critter_inven_obj(self_obj, INVEN_TYPE_WORN) == restock_obj) then begin
+                                        if (obj_item_subtype(critter_inven_obj(self_obj, INVEN_TYPE_WORN)) == 0) then begin
+                                            restock_obj := critter_inven_obj(self_obj, INVEN_TYPE_WORN);
                                             debug_msg("armour pid == " + obj_pid(restock_obj));
                                             rm_obj_from_inven(self_obj, restock_obj);
                                             add_obj_to_inven(self_obj, restock_obj);
                                         end
                                     end
                                     else begin
-                                        if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                            metarule(43, self_obj);
+                                        if ((critter_inven_obj(self_obj, INVEN_TYPE_LEFT_HAND) == restock_obj) or (critter_inven_obj(self_obj, INVEN_TYPE_RIGHT_HAND) == restock_obj)) then begin
+                                            inven_unwield(self_obj);
                                         end
                                     end
                                 end
@@ -620,7 +620,7 @@ begin
             end
             else begin
                 if (obj_is_carrying_obj_pid(self_obj, 8) < restock_amt) then begin
-                    restock_obj := create_object_sid(8, 0, 0, -1);
+                    restock_obj := create_object(8, 0, 0);
                     add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 8));
                 end
                 else begin
@@ -633,17 +633,17 @@ begin
                             restock_obj := obj_carrying_pid_obj(self_obj, 8);
                             if (obj_type(self_obj) == 1) then begin
                                 restock_obj := obj_carrying_pid_obj(self_obj, 8);
-                                if (critter_inven_obj(self_obj, 0) == restock_obj) then begin
-                                    if (obj_item_subtype(critter_inven_obj(self_obj, 0)) == 0) then begin
-                                        restock_obj := critter_inven_obj(self_obj, 0);
+                                if (critter_inven_obj(self_obj, INVEN_TYPE_WORN) == restock_obj) then begin
+                                    if (obj_item_subtype(critter_inven_obj(self_obj, INVEN_TYPE_WORN)) == 0) then begin
+                                        restock_obj := critter_inven_obj(self_obj, INVEN_TYPE_WORN);
                                         debug_msg("armour pid == " + obj_pid(restock_obj));
                                         rm_obj_from_inven(self_obj, restock_obj);
                                         add_obj_to_inven(self_obj, restock_obj);
                                     end
                                 end
                                 else begin
-                                    if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                        metarule(43, self_obj);
+                                    if ((critter_inven_obj(self_obj, INVEN_TYPE_LEFT_HAND) == restock_obj) or (critter_inven_obj(self_obj, INVEN_TYPE_RIGHT_HAND) == restock_obj)) then begin
+                                        inven_unwield(self_obj);
                                     end
                                 end
                             end
@@ -662,7 +662,7 @@ begin
             if (obj_is_carrying_obj_pid(self_obj, 125) < restock_amt) then begin
                 if (100 >= random(1, 100)) then begin
                     if (obj_is_carrying_obj_pid(self_obj, 125) < restock_amt) then begin
-                        restock_obj := create_object_sid(125, 0, 0, -1);
+                        restock_obj := create_object(125, 0, 0);
                         add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 125));
                     end
                     else begin
@@ -675,17 +675,17 @@ begin
                                 restock_obj := obj_carrying_pid_obj(self_obj, 125);
                                 if (obj_type(self_obj) == 1) then begin
                                     restock_obj := obj_carrying_pid_obj(self_obj, 125);
-                                    if (critter_inven_obj(self_obj, 0) == restock_obj) then begin
-                                        if (obj_item_subtype(critter_inven_obj(self_obj, 0)) == 0) then begin
-                                            restock_obj := critter_inven_obj(self_obj, 0);
+                                    if (critter_inven_obj(self_obj, INVEN_TYPE_WORN) == restock_obj) then begin
+                                        if (obj_item_subtype(critter_inven_obj(self_obj, INVEN_TYPE_WORN)) == 0) then begin
+                                            restock_obj := critter_inven_obj(self_obj, INVEN_TYPE_WORN);
                                             debug_msg("armour pid == " + obj_pid(restock_obj));
                                             rm_obj_from_inven(self_obj, restock_obj);
                                             add_obj_to_inven(self_obj, restock_obj);
                                         end
                                     end
                                     else begin
-                                        if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                            metarule(43, self_obj);
+                                        if ((critter_inven_obj(self_obj, INVEN_TYPE_LEFT_HAND) == restock_obj) or (critter_inven_obj(self_obj, INVEN_TYPE_RIGHT_HAND) == restock_obj)) then begin
+                                            inven_unwield(self_obj);
                                         end
                                     end
                                 end
@@ -703,7 +703,7 @@ begin
             end
             else begin
                 if (obj_is_carrying_obj_pid(self_obj, 125) < restock_amt) then begin
-                    restock_obj := create_object_sid(125, 0, 0, -1);
+                    restock_obj := create_object(125, 0, 0);
                     add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 125));
                 end
                 else begin
@@ -716,17 +716,17 @@ begin
                             restock_obj := obj_carrying_pid_obj(self_obj, 125);
                             if (obj_type(self_obj) == 1) then begin
                                 restock_obj := obj_carrying_pid_obj(self_obj, 125);
-                                if (critter_inven_obj(self_obj, 0) == restock_obj) then begin
-                                    if (obj_item_subtype(critter_inven_obj(self_obj, 0)) == 0) then begin
-                                        restock_obj := critter_inven_obj(self_obj, 0);
+                                if (critter_inven_obj(self_obj, INVEN_TYPE_WORN) == restock_obj) then begin
+                                    if (obj_item_subtype(critter_inven_obj(self_obj, INVEN_TYPE_WORN)) == 0) then begin
+                                        restock_obj := critter_inven_obj(self_obj, INVEN_TYPE_WORN);
                                         debug_msg("armour pid == " + obj_pid(restock_obj));
                                         rm_obj_from_inven(self_obj, restock_obj);
                                         add_obj_to_inven(self_obj, restock_obj);
                                     end
                                 end
                                 else begin
-                                    if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                        metarule(43, self_obj);
+                                    if ((critter_inven_obj(self_obj, INVEN_TYPE_LEFT_HAND) == restock_obj) or (critter_inven_obj(self_obj, INVEN_TYPE_RIGHT_HAND) == restock_obj)) then begin
+                                        inven_unwield(self_obj);
                                     end
                                 end
                             end
@@ -745,7 +745,7 @@ begin
             if (obj_is_carrying_obj_pid(self_obj, 29) < restock_amt) then begin
                 if (100 >= random(1, 100)) then begin
                     if (obj_is_carrying_obj_pid(self_obj, 29) < restock_amt) then begin
-                        restock_obj := create_object_sid(29, 0, 0, -1);
+                        restock_obj := create_object(29, 0, 0);
                         add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 29));
                     end
                     else begin
@@ -758,17 +758,17 @@ begin
                                 restock_obj := obj_carrying_pid_obj(self_obj, 29);
                                 if (obj_type(self_obj) == 1) then begin
                                     restock_obj := obj_carrying_pid_obj(self_obj, 29);
-                                    if (critter_inven_obj(self_obj, 0) == restock_obj) then begin
-                                        if (obj_item_subtype(critter_inven_obj(self_obj, 0)) == 0) then begin
-                                            restock_obj := critter_inven_obj(self_obj, 0);
+                                    if (critter_inven_obj(self_obj, INVEN_TYPE_WORN) == restock_obj) then begin
+                                        if (obj_item_subtype(critter_inven_obj(self_obj, INVEN_TYPE_WORN)) == 0) then begin
+                                            restock_obj := critter_inven_obj(self_obj, INVEN_TYPE_WORN);
                                             debug_msg("armour pid == " + obj_pid(restock_obj));
                                             rm_obj_from_inven(self_obj, restock_obj);
                                             add_obj_to_inven(self_obj, restock_obj);
                                         end
                                     end
                                     else begin
-                                        if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                            metarule(43, self_obj);
+                                        if ((critter_inven_obj(self_obj, INVEN_TYPE_LEFT_HAND) == restock_obj) or (critter_inven_obj(self_obj, INVEN_TYPE_RIGHT_HAND) == restock_obj)) then begin
+                                            inven_unwield(self_obj);
                                         end
                                     end
                                 end
@@ -786,7 +786,7 @@ begin
             end
             else begin
                 if (obj_is_carrying_obj_pid(self_obj, 29) < restock_amt) then begin
-                    restock_obj := create_object_sid(29, 0, 0, -1);
+                    restock_obj := create_object(29, 0, 0);
                     add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 29));
                 end
                 else begin
@@ -799,17 +799,17 @@ begin
                             restock_obj := obj_carrying_pid_obj(self_obj, 29);
                             if (obj_type(self_obj) == 1) then begin
                                 restock_obj := obj_carrying_pid_obj(self_obj, 29);
-                                if (critter_inven_obj(self_obj, 0) == restock_obj) then begin
-                                    if (obj_item_subtype(critter_inven_obj(self_obj, 0)) == 0) then begin
-                                        restock_obj := critter_inven_obj(self_obj, 0);
+                                if (critter_inven_obj(self_obj, INVEN_TYPE_WORN) == restock_obj) then begin
+                                    if (obj_item_subtype(critter_inven_obj(self_obj, INVEN_TYPE_WORN)) == 0) then begin
+                                        restock_obj := critter_inven_obj(self_obj, INVEN_TYPE_WORN);
                                         debug_msg("armour pid == " + obj_pid(restock_obj));
                                         rm_obj_from_inven(self_obj, restock_obj);
                                         add_obj_to_inven(self_obj, restock_obj);
                                     end
                                 end
                                 else begin
-                                    if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                        metarule(43, self_obj);
+                                    if ((critter_inven_obj(self_obj, INVEN_TYPE_LEFT_HAND) == restock_obj) or (critter_inven_obj(self_obj, INVEN_TYPE_RIGHT_HAND) == restock_obj)) then begin
+                                        inven_unwield(self_obj);
                                     end
                                 end
                             end
@@ -828,7 +828,7 @@ begin
             if (obj_is_carrying_obj_pid(self_obj, 276) < restock_amt) then begin
                 if (100 >= random(1, 100)) then begin
                     if (obj_is_carrying_obj_pid(self_obj, 276) < restock_amt) then begin
-                        restock_obj := create_object_sid(276, 0, 0, -1);
+                        restock_obj := create_object(276, 0, 0);
                         add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 276));
                     end
                     else begin
@@ -841,17 +841,17 @@ begin
                                 restock_obj := obj_carrying_pid_obj(self_obj, 276);
                                 if (obj_type(self_obj) == 1) then begin
                                     restock_obj := obj_carrying_pid_obj(self_obj, 276);
-                                    if (critter_inven_obj(self_obj, 0) == restock_obj) then begin
-                                        if (obj_item_subtype(critter_inven_obj(self_obj, 0)) == 0) then begin
-                                            restock_obj := critter_inven_obj(self_obj, 0);
+                                    if (critter_inven_obj(self_obj, INVEN_TYPE_WORN) == restock_obj) then begin
+                                        if (obj_item_subtype(critter_inven_obj(self_obj, INVEN_TYPE_WORN)) == 0) then begin
+                                            restock_obj := critter_inven_obj(self_obj, INVEN_TYPE_WORN);
                                             debug_msg("armour pid == " + obj_pid(restock_obj));
                                             rm_obj_from_inven(self_obj, restock_obj);
                                             add_obj_to_inven(self_obj, restock_obj);
                                         end
                                     end
                                     else begin
-                                        if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                            metarule(43, self_obj);
+                                        if ((critter_inven_obj(self_obj, INVEN_TYPE_LEFT_HAND) == restock_obj) or (critter_inven_obj(self_obj, INVEN_TYPE_RIGHT_HAND) == restock_obj)) then begin
+                                            inven_unwield(self_obj);
                                         end
                                     end
                                 end
@@ -869,7 +869,7 @@ begin
             end
             else begin
                 if (obj_is_carrying_obj_pid(self_obj, 276) < restock_amt) then begin
-                    restock_obj := create_object_sid(276, 0, 0, -1);
+                    restock_obj := create_object(276, 0, 0);
                     add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 276));
                 end
                 else begin
@@ -882,17 +882,17 @@ begin
                             restock_obj := obj_carrying_pid_obj(self_obj, 276);
                             if (obj_type(self_obj) == 1) then begin
                                 restock_obj := obj_carrying_pid_obj(self_obj, 276);
-                                if (critter_inven_obj(self_obj, 0) == restock_obj) then begin
-                                    if (obj_item_subtype(critter_inven_obj(self_obj, 0)) == 0) then begin
-                                        restock_obj := critter_inven_obj(self_obj, 0);
+                                if (critter_inven_obj(self_obj, INVEN_TYPE_WORN) == restock_obj) then begin
+                                    if (obj_item_subtype(critter_inven_obj(self_obj, INVEN_TYPE_WORN)) == 0) then begin
+                                        restock_obj := critter_inven_obj(self_obj, INVEN_TYPE_WORN);
                                         debug_msg("armour pid == " + obj_pid(restock_obj));
                                         rm_obj_from_inven(self_obj, restock_obj);
                                         add_obj_to_inven(self_obj, restock_obj);
                                     end
                                 end
                                 else begin
-                                    if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                        metarule(43, self_obj);
+                                    if ((critter_inven_obj(self_obj, INVEN_TYPE_LEFT_HAND) == restock_obj) or (critter_inven_obj(self_obj, INVEN_TYPE_RIGHT_HAND) == restock_obj)) then begin
+                                        inven_unwield(self_obj);
                                     end
                                 end
                             end
@@ -911,7 +911,7 @@ begin
             if (obj_is_carrying_obj_pid(self_obj, 273) < restock_amt) then begin
                 if (100 >= random(1, 100)) then begin
                     if (obj_is_carrying_obj_pid(self_obj, 273) < restock_amt) then begin
-                        restock_obj := create_object_sid(273, 0, 0, -1);
+                        restock_obj := create_object(273, 0, 0);
                         add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 273));
                     end
                     else begin
@@ -924,17 +924,17 @@ begin
                                 restock_obj := obj_carrying_pid_obj(self_obj, 273);
                                 if (obj_type(self_obj) == 1) then begin
                                     restock_obj := obj_carrying_pid_obj(self_obj, 273);
-                                    if (critter_inven_obj(self_obj, 0) == restock_obj) then begin
-                                        if (obj_item_subtype(critter_inven_obj(self_obj, 0)) == 0) then begin
-                                            restock_obj := critter_inven_obj(self_obj, 0);
+                                    if (critter_inven_obj(self_obj, INVEN_TYPE_WORN) == restock_obj) then begin
+                                        if (obj_item_subtype(critter_inven_obj(self_obj, INVEN_TYPE_WORN)) == 0) then begin
+                                            restock_obj := critter_inven_obj(self_obj, INVEN_TYPE_WORN);
                                             debug_msg("armour pid == " + obj_pid(restock_obj));
                                             rm_obj_from_inven(self_obj, restock_obj);
                                             add_obj_to_inven(self_obj, restock_obj);
                                         end
                                     end
                                     else begin
-                                        if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                            metarule(43, self_obj);
+                                        if ((critter_inven_obj(self_obj, INVEN_TYPE_LEFT_HAND) == restock_obj) or (critter_inven_obj(self_obj, INVEN_TYPE_RIGHT_HAND) == restock_obj)) then begin
+                                            inven_unwield(self_obj);
                                         end
                                     end
                                 end
@@ -952,7 +952,7 @@ begin
             end
             else begin
                 if (obj_is_carrying_obj_pid(self_obj, 273) < restock_amt) then begin
-                    restock_obj := create_object_sid(273, 0, 0, -1);
+                    restock_obj := create_object(273, 0, 0);
                     add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 273));
                 end
                 else begin
@@ -965,17 +965,17 @@ begin
                             restock_obj := obj_carrying_pid_obj(self_obj, 273);
                             if (obj_type(self_obj) == 1) then begin
                                 restock_obj := obj_carrying_pid_obj(self_obj, 273);
-                                if (critter_inven_obj(self_obj, 0) == restock_obj) then begin
-                                    if (obj_item_subtype(critter_inven_obj(self_obj, 0)) == 0) then begin
-                                        restock_obj := critter_inven_obj(self_obj, 0);
+                                if (critter_inven_obj(self_obj, INVEN_TYPE_WORN) == restock_obj) then begin
+                                    if (obj_item_subtype(critter_inven_obj(self_obj, INVEN_TYPE_WORN)) == 0) then begin
+                                        restock_obj := critter_inven_obj(self_obj, INVEN_TYPE_WORN);
                                         debug_msg("armour pid == " + obj_pid(restock_obj));
                                         rm_obj_from_inven(self_obj, restock_obj);
                                         add_obj_to_inven(self_obj, restock_obj);
                                     end
                                 end
                                 else begin
-                                    if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                        metarule(43, self_obj);
+                                    if ((critter_inven_obj(self_obj, INVEN_TYPE_LEFT_HAND) == restock_obj) or (critter_inven_obj(self_obj, INVEN_TYPE_RIGHT_HAND) == restock_obj)) then begin
+                                        inven_unwield(self_obj);
                                     end
                                 end
                             end
@@ -994,7 +994,7 @@ begin
             if (obj_is_carrying_obj_pid(self_obj, 272) < restock_amt) then begin
                 if (100 >= random(1, 100)) then begin
                     if (obj_is_carrying_obj_pid(self_obj, 272) < restock_amt) then begin
-                        restock_obj := create_object_sid(272, 0, 0, -1);
+                        restock_obj := create_object(272, 0, 0);
                         add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 272));
                     end
                     else begin
@@ -1007,17 +1007,17 @@ begin
                                 restock_obj := obj_carrying_pid_obj(self_obj, 272);
                                 if (obj_type(self_obj) == 1) then begin
                                     restock_obj := obj_carrying_pid_obj(self_obj, 272);
-                                    if (critter_inven_obj(self_obj, 0) == restock_obj) then begin
-                                        if (obj_item_subtype(critter_inven_obj(self_obj, 0)) == 0) then begin
-                                            restock_obj := critter_inven_obj(self_obj, 0);
+                                    if (critter_inven_obj(self_obj, INVEN_TYPE_WORN) == restock_obj) then begin
+                                        if (obj_item_subtype(critter_inven_obj(self_obj, INVEN_TYPE_WORN)) == 0) then begin
+                                            restock_obj := critter_inven_obj(self_obj, INVEN_TYPE_WORN);
                                             debug_msg("armour pid == " + obj_pid(restock_obj));
                                             rm_obj_from_inven(self_obj, restock_obj);
                                             add_obj_to_inven(self_obj, restock_obj);
                                         end
                                     end
                                     else begin
-                                        if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                            metarule(43, self_obj);
+                                        if ((critter_inven_obj(self_obj, INVEN_TYPE_LEFT_HAND) == restock_obj) or (critter_inven_obj(self_obj, INVEN_TYPE_RIGHT_HAND) == restock_obj)) then begin
+                                            inven_unwield(self_obj);
                                         end
                                     end
                                 end
@@ -1035,7 +1035,7 @@ begin
             end
             else begin
                 if (obj_is_carrying_obj_pid(self_obj, 272) < restock_amt) then begin
-                    restock_obj := create_object_sid(272, 0, 0, -1);
+                    restock_obj := create_object(272, 0, 0);
                     add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 272));
                 end
                 else begin
@@ -1048,17 +1048,17 @@ begin
                             restock_obj := obj_carrying_pid_obj(self_obj, 272);
                             if (obj_type(self_obj) == 1) then begin
                                 restock_obj := obj_carrying_pid_obj(self_obj, 272);
-                                if (critter_inven_obj(self_obj, 0) == restock_obj) then begin
-                                    if (obj_item_subtype(critter_inven_obj(self_obj, 0)) == 0) then begin
-                                        restock_obj := critter_inven_obj(self_obj, 0);
+                                if (critter_inven_obj(self_obj, INVEN_TYPE_WORN) == restock_obj) then begin
+                                    if (obj_item_subtype(critter_inven_obj(self_obj, INVEN_TYPE_WORN)) == 0) then begin
+                                        restock_obj := critter_inven_obj(self_obj, INVEN_TYPE_WORN);
                                         debug_msg("armour pid == " + obj_pid(restock_obj));
                                         rm_obj_from_inven(self_obj, restock_obj);
                                         add_obj_to_inven(self_obj, restock_obj);
                                     end
                                 end
                                 else begin
-                                    if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                        metarule(43, self_obj);
+                                    if ((critter_inven_obj(self_obj, INVEN_TYPE_LEFT_HAND) == restock_obj) or (critter_inven_obj(self_obj, INVEN_TYPE_RIGHT_HAND) == restock_obj)) then begin
+                                        inven_unwield(self_obj);
                                     end
                                 end
                             end
@@ -1077,7 +1077,7 @@ begin
             if (obj_is_carrying_obj_pid(self_obj, 117) < restock_amt) then begin
                 if (100 >= random(1, 100)) then begin
                     if (obj_is_carrying_obj_pid(self_obj, 117) < restock_amt) then begin
-                        restock_obj := create_object_sid(117, 0, 0, -1);
+                        restock_obj := create_object(117, 0, 0);
                         add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 117));
                     end
                     else begin
@@ -1090,17 +1090,17 @@ begin
                                 restock_obj := obj_carrying_pid_obj(self_obj, 117);
                                 if (obj_type(self_obj) == 1) then begin
                                     restock_obj := obj_carrying_pid_obj(self_obj, 117);
-                                    if (critter_inven_obj(self_obj, 0) == restock_obj) then begin
-                                        if (obj_item_subtype(critter_inven_obj(self_obj, 0)) == 0) then begin
-                                            restock_obj := critter_inven_obj(self_obj, 0);
+                                    if (critter_inven_obj(self_obj, INVEN_TYPE_WORN) == restock_obj) then begin
+                                        if (obj_item_subtype(critter_inven_obj(self_obj, INVEN_TYPE_WORN)) == 0) then begin
+                                            restock_obj := critter_inven_obj(self_obj, INVEN_TYPE_WORN);
                                             debug_msg("armour pid == " + obj_pid(restock_obj));
                                             rm_obj_from_inven(self_obj, restock_obj);
                                             add_obj_to_inven(self_obj, restock_obj);
                                         end
                                     end
                                     else begin
-                                        if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                            metarule(43, self_obj);
+                                        if ((critter_inven_obj(self_obj, INVEN_TYPE_LEFT_HAND) == restock_obj) or (critter_inven_obj(self_obj, INVEN_TYPE_RIGHT_HAND) == restock_obj)) then begin
+                                            inven_unwield(self_obj);
                                         end
                                     end
                                 end
@@ -1118,7 +1118,7 @@ begin
             end
             else begin
                 if (obj_is_carrying_obj_pid(self_obj, 117) < restock_amt) then begin
-                    restock_obj := create_object_sid(117, 0, 0, -1);
+                    restock_obj := create_object(117, 0, 0);
                     add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 117));
                 end
                 else begin
@@ -1131,17 +1131,17 @@ begin
                             restock_obj := obj_carrying_pid_obj(self_obj, 117);
                             if (obj_type(self_obj) == 1) then begin
                                 restock_obj := obj_carrying_pid_obj(self_obj, 117);
-                                if (critter_inven_obj(self_obj, 0) == restock_obj) then begin
-                                    if (obj_item_subtype(critter_inven_obj(self_obj, 0)) == 0) then begin
-                                        restock_obj := critter_inven_obj(self_obj, 0);
+                                if (critter_inven_obj(self_obj, INVEN_TYPE_WORN) == restock_obj) then begin
+                                    if (obj_item_subtype(critter_inven_obj(self_obj, INVEN_TYPE_WORN)) == 0) then begin
+                                        restock_obj := critter_inven_obj(self_obj, INVEN_TYPE_WORN);
                                         debug_msg("armour pid == " + obj_pid(restock_obj));
                                         rm_obj_from_inven(self_obj, restock_obj);
                                         add_obj_to_inven(self_obj, restock_obj);
                                     end
                                 end
                                 else begin
-                                    if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                        metarule(43, self_obj);
+                                    if ((critter_inven_obj(self_obj, INVEN_TYPE_LEFT_HAND) == restock_obj) or (critter_inven_obj(self_obj, INVEN_TYPE_RIGHT_HAND) == restock_obj)) then begin
+                                        inven_unwield(self_obj);
                                     end
                                 end
                             end
@@ -1162,4 +1162,3 @@ begin
         end
     end
 end
-
