diff --git a/Mods/MegaMod/Original/DITUBBOX.ssl b/Mods/MegaMod/ReDefine/DITUBBOX.ssl
index f2ec522..cb177c8 100644
--- a/Mods/MegaMod/Original/DITUBBOX.ssl
+++ b/Mods/MegaMod/ReDefine/DITUBBOX.ssl
@@ -142,9 +142,9 @@ begin
     variable LVar0 := 0;
     variable LVar1 := 0;
     den_tubby_box_obj := self_obj;
-    if (metarule(22, 0) == 0) then begin
+    if (is_loading_game == 0) then begin
         if (local_var(0) < game_time) then begin
-            LVar0 := create_object_sid(128, 5, 0, -1);
+            LVar0 := create_object(128, 5, 0);
             if (obj_is_carrying_obj_pid(self_obj, 479) > 0) then begin
                 LVar1 := obj_carrying_pid_obj(self_obj, 479);
                 rm_obj_from_inven(self_obj, LVar1);
@@ -390,7 +390,7 @@ begin
                 rm_obj_from_inven(self_obj, LVar1);
                 add_obj_to_inven(LVar0, LVar1);
             end
-            global_temp := create_object_sid(128, 0, 0, -1);
+            global_temp := create_object(128, 0, 0);
             move_obj_inven_to_obj(self_obj, global_temp);
             destroy_object(global_temp);
             item_caps_adjust(self_obj, random(151, 161) - item_caps_total(self_obj));
@@ -398,7 +398,7 @@ begin
             if (obj_is_carrying_obj_pid(self_obj, 259) < restock_amt) then begin
                 if (100 >= random(1, 100)) then begin
                     if (obj_is_carrying_obj_pid(self_obj, 259) < restock_amt) then begin
-                        restock_obj := create_object_sid(259, 0, 0, -1);
+                        restock_obj := create_object(259, 0, 0);
                         add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 259));
                     end
                     else begin
@@ -421,7 +421,7 @@ begin
                                     end
                                     else begin
                                         if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                            metarule(43, self_obj);
+                                            inven_unwield(self_obj);
                                         end
                                     end
                                 end
@@ -439,7 +439,7 @@ begin
             end
             else begin
                 if (obj_is_carrying_obj_pid(self_obj, 259) < restock_amt) then begin
-                    restock_obj := create_object_sid(259, 0, 0, -1);
+                    restock_obj := create_object(259, 0, 0);
                     add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 259));
                 end
                 else begin
@@ -462,7 +462,7 @@ begin
                                 end
                                 else begin
                                     if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                        metarule(43, self_obj);
+                                        inven_unwield(self_obj);
                                     end
                                 end
                             end
@@ -481,7 +481,7 @@ begin
             if (obj_is_carrying_obj_pid(self_obj, 110) < restock_amt) then begin
                 if (100 >= random(1, 100)) then begin
                     if (obj_is_carrying_obj_pid(self_obj, 110) < restock_amt) then begin
-                        restock_obj := create_object_sid(110, 0, 0, -1);
+                        restock_obj := create_object(110, 0, 0);
                         add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 110));
                     end
                     else begin
@@ -504,7 +504,7 @@ begin
                                     end
                                     else begin
                                         if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                            metarule(43, self_obj);
+                                            inven_unwield(self_obj);
                                         end
                                     end
                                 end
@@ -522,7 +522,7 @@ begin
             end
             else begin
                 if (obj_is_carrying_obj_pid(self_obj, 110) < restock_amt) then begin
-                    restock_obj := create_object_sid(110, 0, 0, -1);
+                    restock_obj := create_object(110, 0, 0);
                     add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 110));
                 end
                 else begin
@@ -545,7 +545,7 @@ begin
                                 end
                                 else begin
                                     if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                        metarule(43, self_obj);
+                                        inven_unwield(self_obj);
                                     end
                                 end
                             end
@@ -564,7 +564,7 @@ begin
             if (obj_is_carrying_obj_pid(self_obj, 87) < restock_amt) then begin
                 if (100 >= random(1, 100)) then begin
                     if (obj_is_carrying_obj_pid(self_obj, 87) < restock_amt) then begin
-                        restock_obj := create_object_sid(87, 0, 0, -1);
+                        restock_obj := create_object(87, 0, 0);
                         add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 87));
                     end
                     else begin
@@ -587,7 +587,7 @@ begin
                                     end
                                     else begin
                                         if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                            metarule(43, self_obj);
+                                            inven_unwield(self_obj);
                                         end
                                     end
                                 end
@@ -605,7 +605,7 @@ begin
             end
             else begin
                 if (obj_is_carrying_obj_pid(self_obj, 87) < restock_amt) then begin
-                    restock_obj := create_object_sid(87, 0, 0, -1);
+                    restock_obj := create_object(87, 0, 0);
                     add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 87));
                 end
                 else begin
@@ -628,7 +628,7 @@ begin
                                 end
                                 else begin
                                     if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                        metarule(43, self_obj);
+                                        inven_unwield(self_obj);
                                     end
                                 end
                             end
@@ -647,7 +647,7 @@ begin
             if (obj_is_carrying_obj_pid(self_obj, 8) < restock_amt) then begin
                 if (100 >= random(1, 100)) then begin
                     if (obj_is_carrying_obj_pid(self_obj, 8) < restock_amt) then begin
-                        restock_obj := create_object_sid(8, 0, 0, -1);
+                        restock_obj := create_object(8, 0, 0);
                         add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 8));
                     end
                     else begin
@@ -670,7 +670,7 @@ begin
                                     end
                                     else begin
                                         if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                            metarule(43, self_obj);
+                                            inven_unwield(self_obj);
                                         end
                                     end
                                 end
@@ -688,7 +688,7 @@ begin
             end
             else begin
                 if (obj_is_carrying_obj_pid(self_obj, 8) < restock_amt) then begin
-                    restock_obj := create_object_sid(8, 0, 0, -1);
+                    restock_obj := create_object(8, 0, 0);
                     add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 8));
                 end
                 else begin
@@ -711,7 +711,7 @@ begin
                                 end
                                 else begin
                                     if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                        metarule(43, self_obj);
+                                        inven_unwield(self_obj);
                                     end
                                 end
                             end
@@ -730,7 +730,7 @@ begin
             if (obj_is_carrying_obj_pid(self_obj, 29) < restock_amt) then begin
                 if (100 >= random(1, 100)) then begin
                     if (obj_is_carrying_obj_pid(self_obj, 29) < restock_amt) then begin
-                        restock_obj := create_object_sid(29, 0, 0, -1);
+                        restock_obj := create_object(29, 0, 0);
                         add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 29));
                     end
                     else begin
@@ -753,7 +753,7 @@ begin
                                     end
                                     else begin
                                         if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                            metarule(43, self_obj);
+                                            inven_unwield(self_obj);
                                         end
                                     end
                                 end
@@ -771,7 +771,7 @@ begin
             end
             else begin
                 if (obj_is_carrying_obj_pid(self_obj, 29) < restock_amt) then begin
-                    restock_obj := create_object_sid(29, 0, 0, -1);
+                    restock_obj := create_object(29, 0, 0);
                     add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 29));
                 end
                 else begin
@@ -794,7 +794,7 @@ begin
                                 end
                                 else begin
                                     if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                        metarule(43, self_obj);
+                                        inven_unwield(self_obj);
                                     end
                                 end
                             end
@@ -813,7 +813,7 @@ begin
             if (obj_is_carrying_obj_pid(self_obj, 30) < restock_amt) then begin
                 if (100 >= random(1, 100)) then begin
                     if (obj_is_carrying_obj_pid(self_obj, 30) < restock_amt) then begin
-                        restock_obj := create_object_sid(30, 0, 0, -1);
+                        restock_obj := create_object(30, 0, 0);
                         add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 30));
                     end
                     else begin
@@ -836,7 +836,7 @@ begin
                                     end
                                     else begin
                                         if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                            metarule(43, self_obj);
+                                            inven_unwield(self_obj);
                                         end
                                     end
                                 end
@@ -854,7 +854,7 @@ begin
             end
             else begin
                 if (obj_is_carrying_obj_pid(self_obj, 30) < restock_amt) then begin
-                    restock_obj := create_object_sid(30, 0, 0, -1);
+                    restock_obj := create_object(30, 0, 0);
                     add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 30));
                 end
                 else begin
@@ -877,7 +877,7 @@ begin
                                 end
                                 else begin
                                     if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                        metarule(43, self_obj);
+                                        inven_unwield(self_obj);
                                     end
                                 end
                             end
@@ -896,7 +896,7 @@ begin
             if (obj_is_carrying_obj_pid(self_obj, 21) < restock_amt) then begin
                 if (100 >= random(1, 100)) then begin
                     if (obj_is_carrying_obj_pid(self_obj, 21) < restock_amt) then begin
-                        restock_obj := create_object_sid(21, 0, 0, -1);
+                        restock_obj := create_object(21, 0, 0);
                         add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 21));
                     end
                     else begin
@@ -919,7 +919,7 @@ begin
                                     end
                                     else begin
                                         if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                            metarule(43, self_obj);
+                                            inven_unwield(self_obj);
                                         end
                                     end
                                 end
@@ -937,7 +937,7 @@ begin
             end
             else begin
                 if (obj_is_carrying_obj_pid(self_obj, 21) < restock_amt) then begin
-                    restock_obj := create_object_sid(21, 0, 0, -1);
+                    restock_obj := create_object(21, 0, 0);
                     add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 21));
                 end
                 else begin
@@ -960,7 +960,7 @@ begin
                                 end
                                 else begin
                                     if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                        metarule(43, self_obj);
+                                        inven_unwield(self_obj);
                                     end
                                 end
                             end
@@ -979,7 +979,7 @@ begin
             if (obj_is_carrying_obj_pid(self_obj, 74) < restock_amt) then begin
                 if (100 >= random(1, 100)) then begin
                     if (obj_is_carrying_obj_pid(self_obj, 74) < restock_amt) then begin
-                        restock_obj := create_object_sid(74, 0, 0, -1);
+                        restock_obj := create_object(74, 0, 0);
                         add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 74));
                     end
                     else begin
@@ -1002,7 +1002,7 @@ begin
                                     end
                                     else begin
                                         if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                            metarule(43, self_obj);
+                                            inven_unwield(self_obj);
                                         end
                                     end
                                 end
@@ -1020,7 +1020,7 @@ begin
             end
             else begin
                 if (obj_is_carrying_obj_pid(self_obj, 74) < restock_amt) then begin
-                    restock_obj := create_object_sid(74, 0, 0, -1);
+                    restock_obj := create_object(74, 0, 0);
                     add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 74));
                 end
                 else begin
@@ -1043,7 +1043,7 @@ begin
                                 end
                                 else begin
                                     if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                        metarule(43, self_obj);
+                                        inven_unwield(self_obj);
                                     end
                                 end
                             end
@@ -1062,7 +1062,7 @@ begin
             if (obj_is_carrying_obj_pid(self_obj, 40) < restock_amt) then begin
                 if (100 >= random(1, 100)) then begin
                     if (obj_is_carrying_obj_pid(self_obj, 40) < restock_amt) then begin
-                        restock_obj := create_object_sid(40, 0, 0, -1);
+                        restock_obj := create_object(40, 0, 0);
                         add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 40));
                     end
                     else begin
@@ -1085,7 +1085,7 @@ begin
                                     end
                                     else begin
                                         if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                            metarule(43, self_obj);
+                                            inven_unwield(self_obj);
                                         end
                                     end
                                 end
@@ -1103,7 +1103,7 @@ begin
             end
             else begin
                 if (obj_is_carrying_obj_pid(self_obj, 40) < restock_amt) then begin
-                    restock_obj := create_object_sid(40, 0, 0, -1);
+                    restock_obj := create_object(40, 0, 0);
                     add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 40));
                 end
                 else begin
@@ -1126,7 +1126,7 @@ begin
                                 end
                                 else begin
                                     if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                        metarule(43, self_obj);
+                                        inven_unwield(self_obj);
                                     end
                                 end
                             end
@@ -1145,7 +1145,7 @@ begin
             if (obj_is_carrying_obj_pid(self_obj, 9) < restock_amt) then begin
                 if (100 >= random(1, 100)) then begin
                     if (obj_is_carrying_obj_pid(self_obj, 9) < restock_amt) then begin
-                        restock_obj := create_object_sid(9, 0, 0, -1);
+                        restock_obj := create_object(9, 0, 0);
                         add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 9));
                     end
                     else begin
@@ -1168,7 +1168,7 @@ begin
                                     end
                                     else begin
                                         if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                            metarule(43, self_obj);
+                                            inven_unwield(self_obj);
                                         end
                                     end
                                 end
@@ -1186,7 +1186,7 @@ begin
             end
             else begin
                 if (obj_is_carrying_obj_pid(self_obj, 9) < restock_amt) then begin
-                    restock_obj := create_object_sid(9, 0, 0, -1);
+                    restock_obj := create_object(9, 0, 0);
                     add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 9));
                 end
                 else begin
@@ -1209,7 +1209,7 @@ begin
                                 end
                                 else begin
                                     if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                        metarule(43, self_obj);
+                                        inven_unwield(self_obj);
                                     end
                                 end
                             end
@@ -1228,7 +1228,7 @@ begin
             if (obj_is_carrying_obj_pid(self_obj, 18) < restock_amt) then begin
                 if (100 >= random(1, 100)) then begin
                     if (obj_is_carrying_obj_pid(self_obj, 18) < restock_amt) then begin
-                        restock_obj := create_object_sid(18, 0, 0, -1);
+                        restock_obj := create_object(18, 0, 0);
                         add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 18));
                     end
                     else begin
@@ -1251,7 +1251,7 @@ begin
                                     end
                                     else begin
                                         if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                            metarule(43, self_obj);
+                                            inven_unwield(self_obj);
                                         end
                                     end
                                 end
@@ -1269,7 +1269,7 @@ begin
             end
             else begin
                 if (obj_is_carrying_obj_pid(self_obj, 18) < restock_amt) then begin
-                    restock_obj := create_object_sid(18, 0, 0, -1);
+                    restock_obj := create_object(18, 0, 0);
                     add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 18));
                 end
                 else begin
@@ -1292,7 +1292,7 @@ begin
                                 end
                                 else begin
                                     if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                        metarule(43, self_obj);
+                                        inven_unwield(self_obj);
                                     end
                                 end
                             end
@@ -1311,7 +1311,7 @@ begin
             if (obj_is_carrying_obj_pid(self_obj, 6) < restock_amt) then begin
                 if (100 >= random(1, 100)) then begin
                     if (obj_is_carrying_obj_pid(self_obj, 6) < restock_amt) then begin
-                        restock_obj := create_object_sid(6, 0, 0, -1);
+                        restock_obj := create_object(6, 0, 0);
                         add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 6));
                     end
                     else begin
@@ -1334,7 +1334,7 @@ begin
                                     end
                                     else begin
                                         if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                            metarule(43, self_obj);
+                                            inven_unwield(self_obj);
                                         end
                                     end
                                 end
@@ -1352,7 +1352,7 @@ begin
             end
             else begin
                 if (obj_is_carrying_obj_pid(self_obj, 6) < restock_amt) then begin
-                    restock_obj := create_object_sid(6, 0, 0, -1);
+                    restock_obj := create_object(6, 0, 0);
                     add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 6));
                 end
                 else begin
@@ -1375,7 +1375,7 @@ begin
                                 end
                                 else begin
                                     if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                        metarule(43, self_obj);
+                                        inven_unwield(self_obj);
                                     end
                                 end
                             end
@@ -1394,7 +1394,7 @@ begin
             if (obj_is_carrying_obj_pid(self_obj, 48) < restock_amt) then begin
                 if (100 >= random(1, 100)) then begin
                     if (obj_is_carrying_obj_pid(self_obj, 48) < restock_amt) then begin
-                        restock_obj := create_object_sid(48, 0, 0, -1);
+                        restock_obj := create_object(48, 0, 0);
                         add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 48));
                     end
                     else begin
@@ -1417,7 +1417,7 @@ begin
                                     end
                                     else begin
                                         if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                            metarule(43, self_obj);
+                                            inven_unwield(self_obj);
                                         end
                                     end
                                 end
@@ -1435,7 +1435,7 @@ begin
             end
             else begin
                 if (obj_is_carrying_obj_pid(self_obj, 48) < restock_amt) then begin
-                    restock_obj := create_object_sid(48, 0, 0, -1);
+                    restock_obj := create_object(48, 0, 0);
                     add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 48));
                 end
                 else begin
@@ -1458,7 +1458,7 @@ begin
                                 end
                                 else begin
                                     if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                        metarule(43, self_obj);
+                                        inven_unwield(self_obj);
                                     end
                                 end
                             end
@@ -1479,4 +1479,3 @@ begin
         end
     end
 end
-
