diff --git a/Mods/MegaMod/Original/ECARTHUR.ssl b/Mods/MegaMod/ReDefine/ECARTHUR.ssl
index 4eeeaa6..8b0e330 100644
--- a/Mods/MegaMod/Original/ECARTHUR.ssl
+++ b/Mods/MegaMod/ReDefine/ECARTHUR.ssl
@@ -176,15 +176,15 @@ end
 procedure start
 begin
     i_arthur_obj := self_obj;
-    if ((Only_Once == 0) and not(metarule(22, 0))) then begin
+    if ((Only_Once == 0) and not(is_loading_game)) then begin
         Only_Once := 1;
-        debug_msg("ECARTHUR: 1 and_spec_flag(ENC_ARTHUR_ONE): " + ((global_var(323) bwand 1) != 0));
-        if (not((global_var(323) bwand 1) != 0)) then begin
-            set_global_var(323, global_var(323) bwor 1);
+        debug_msg("ECARTHUR: 1 and_spec_flag(ENC_ARTHUR_ONE): " + ((global_var(GVAR_SPECIAL_ENCOUNTER_FLAGS) bwand 1) != 0));
+        if (not((global_var(GVAR_SPECIAL_ENCOUNTER_FLAGS) bwand 1) != 0)) then begin
+            set_global_var(GVAR_SPECIAL_ENCOUNTER_FLAGS, global_var(GVAR_SPECIAL_ENCOUNTER_FLAGS) bwor 1);
         end
         else begin
-            if (not((global_var(323) bwand 2) != 0)) then begin
-                set_global_var(323, global_var(323) bwor 2);
+            if (not((global_var(GVAR_SPECIAL_ENCOUNTER_FLAGS) bwand 2) != 0)) then begin
+                set_global_var(GVAR_SPECIAL_ENCOUNTER_FLAGS, global_var(GVAR_SPECIAL_ENCOUNTER_FLAGS) bwor 2);
             end
         end
     end
@@ -194,12 +194,12 @@ procedure critter_p_proc
 begin
     if ((local_var(5) == 2) and obj_can_see_obj(self_obj, dude_obj)) then begin
         set_local_var(5, 1);
-        attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+        attack(dude_obj);
     end
-    if (not((global_var(323) bwand 2) != 0)) then begin
+    if (not((global_var(GVAR_SPECIAL_ENCOUNTER_FLAGS) bwand 2) != 0)) then begin
         if (((map_var(0) bwand 1) != 0) and not((map_var(0) bwand 2) != 0)) then begin
             set_map_var(0, map_var(0) bwor 2);
-            float_msg(self_obj, message_str(683, 207), 0);
+            float_msg(self_obj, mstr(207), FLOAT_MSG_NORMAL);
             add_timer_event(self_obj, 0, 2);
             debug_msg("ECARTHUR: moving to pos a");
         end
@@ -210,8 +210,8 @@ begin
         end
     end
     else begin
-        if ((i_vorpal_obj != -1) and (elevation(dude_obj) == elevation(self_obj))) then begin
-            attack_complex(i_vorpal_obj, 0, 1, 0, 0, 30000, 0, 0);
+        if ((i_vorpal_obj != -1) and (dude_elevation == elevation(self_obj))) then begin
+            attack(i_vorpal_obj);
         end
     end
 end
@@ -231,8 +231,8 @@ begin
     Evil_Critter := 0;
     Slavery_Tolerant := 2;
     Karma_Perception := 10;
-    debug_msg("ECARTHUR: Arthur_encounter_one: " + not((global_var(323) bwand 2) != 0));
-    if (not((global_var(323) bwand 2) != 0)) then begin
+    debug_msg("ECARTHUR: Arthur_encounter_one: " + not((global_var(GVAR_SPECIAL_ENCOUNTER_FLAGS) bwand 2) != 0));
+    if (not((global_var(GVAR_SPECIAL_ENCOUNTER_FLAGS) bwand 2) != 0)) then begin
         if (local_var(4) == 0) then begin
             start_gdialog(683, self_obj, 4, -1, -1);
             gsay_start;
@@ -254,59 +254,59 @@ begin
     if (LVar0) then begin
         destroy_object(LVar0);
     end
-    if (global_var(606) == 0) then begin
-        set_global_var(606, 1);
+    if (global_var(GVAR_SPECIAL_ENCOUNTER_HOLY2) == 0) then begin
+        set_global_var(GVAR_SPECIAL_ENCOUNTER_HOLY2, 1);
     end
     if (source_obj == dude_obj) then begin
-        set_global_var(5, global_var(5) + 1);
-        if (metarule(51, self_obj) == 2) then begin
+        set_global_var(GVAR_GOOD_MONSTER, global_var(GVAR_GOOD_MONSTER) + 1);
+        if (critter_kill_type(self_obj) == KILL_TYPE_children_kills) then begin
             if (source_obj == dude_obj) then begin
-                set_global_var(2, 0);
-                set_global_var(1, global_var(1) + 1);
-                set_global_var(0, global_var(0) + -15);
+                set_global_var(GVAR_CHAMPION_REPUTATION, 0);
+                set_global_var(GVAR_CHILDKILLER_REPUTATION, global_var(GVAR_CHILDKILLER_REPUTATION) + 1);
+                set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + -15);
                 debug_msg("Player gains " + -15 + " Karma Points.");
-                set_global_var(37, 0);
-                set_global_var(38, 0);
-                set_global_var(39, 0);
-                set_global_var(40, 0);
-                set_global_var(41, 0);
-                set_global_var(42, 0);
-                set_global_var(43, 0);
-                set_global_var(44, 0);
-                set_global_var(45, 0);
-                if (global_var(0) >= 1000) then begin
-                    set_global_var(37, 1);
+                set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+                set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+                set_global_var(GVAR_KARMA_DEFENDER, 0);
+                set_global_var(GVAR_KARMA_WANDERER, 0);
+                set_global_var(GVAR_KARMA_BETRAYER, 0);
+                set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+                set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+                set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+                    set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
                 end
                 else begin
-                    if (global_var(0) >= 750) then begin
-                        set_global_var(38, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
                     end
                     else begin
-                        if (global_var(0) >= 500) then begin
-                            set_global_var(39, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                            set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                         end
                         else begin
-                            if (global_var(0) >= 250) then begin
-                                set_global_var(40, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                                set_global_var(GVAR_KARMA_DEFENDER, 1);
                             end
                             else begin
-                                if (global_var(0) > -250) then begin
-                                    set_global_var(41, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                                    set_global_var(GVAR_KARMA_WANDERER, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -500) then begin
-                                        set_global_var(42, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                        set_global_var(GVAR_KARMA_BETRAYER, 1);
                                     end
                                     else begin
-                                        if (global_var(0) > -750) then begin
-                                            set_global_var(43, 1);
+                                        if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                            set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                         end
                                         else begin
-                                            if (global_var(0) > -1000) then begin
-                                                set_global_var(44, 1);
+                                            if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                                set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                             end
                                             else begin
-                                                set_global_var(45, 1);
+                                                set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                             end
                                         end
                                     end
@@ -317,58 +317,58 @@ begin
                 end
             end
         end
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(4) > (3 * global_var(5))) or (global_var(2) == 1)) and (global_var(1) == 0)) then begin
-            set_global_var(2, 1);
-            set_global_var(3, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(4) > (3 * global_var(GVAR_GOOD_MONSTER))) or (global_var(GVAR_CHAMPION_REPUTATION) == 1)) and (global_var(GVAR_CHILDKILLER_REPUTATION) == 0)) then begin
+            set_global_var(GVAR_CHAMPION_REPUTATION, 1);
+            set_global_var(GVAR_BERSERKER_REPUTATION, 0);
         end
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(5) > (2 * global_var(4))) or (global_var(3) == 1))) then begin
-            set_global_var(3, 1);
-            set_global_var(2, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
+            set_global_var(GVAR_BERSERKER_REPUTATION, 1);
+            set_global_var(GVAR_CHAMPION_REPUTATION, 0);
         end
-        set_global_var(0, global_var(0) + -10);
+        set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + -10);
         debug_msg("Player gains " + -10 + " Karma Points.");
-        set_global_var(37, 0);
-        set_global_var(38, 0);
-        set_global_var(39, 0);
-        set_global_var(40, 0);
-        set_global_var(41, 0);
-        set_global_var(42, 0);
-        set_global_var(43, 0);
-        set_global_var(44, 0);
-        set_global_var(45, 0);
-        if (global_var(0) >= 1000) then begin
-            set_global_var(37, 1);
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+        set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+        set_global_var(GVAR_KARMA_DEFENDER, 0);
+        set_global_var(GVAR_KARMA_WANDERER, 0);
+        set_global_var(GVAR_KARMA_BETRAYER, 0);
+        set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+        set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+            set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
         end
         else begin
-            if (global_var(0) >= 750) then begin
-                set_global_var(38, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
             end
             else begin
-                if (global_var(0) >= 500) then begin
-                    set_global_var(39, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                 end
                 else begin
-                    if (global_var(0) >= 250) then begin
-                        set_global_var(40, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                        set_global_var(GVAR_KARMA_DEFENDER, 1);
                     end
                     else begin
-                        if (global_var(0) > -250) then begin
-                            set_global_var(41, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                            set_global_var(GVAR_KARMA_WANDERER, 1);
                         end
                         else begin
-                            if (global_var(0) > -500) then begin
-                                set_global_var(42, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                set_global_var(GVAR_KARMA_BETRAYER, 1);
                             end
                             else begin
-                                if (global_var(0) > -750) then begin
-                                    set_global_var(43, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -1000) then begin
-                                        set_global_var(44, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                     end
                                     else begin
-                                        set_global_var(45, 1);
+                                        set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                     end
                                 end
                             end
@@ -386,17 +386,17 @@ end
 procedure look_at_p_proc
 begin
     script_overrides;
-    display_msg(message_str(683, 100));
+    display_mstr(100);
 end
 
 procedure description_p_proc
 begin
     script_overrides;
     if (local_var(4) == 0) then begin
-        display_msg(message_str(683, 101));
+        display_mstr(101);
     end
     else begin
-        display_msg(message_str(683, 102));
+        display_mstr(102);
     end
 end
 
@@ -414,9 +414,9 @@ end
 procedure map_enter_p_proc
 begin
     Only_Once := 0;
-    critter_add_trait(self_obj, 1, 6, 151);
-    critter_add_trait(self_obj, 1, 5, 22);
-    if (not((global_var(323) bwand 2) != 0)) then begin
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 151);
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_TOUGH_GUARD);
+    if (not((global_var(GVAR_SPECIAL_ENCOUNTER_FLAGS) bwand 2) != 0)) then begin
         move_to(self_obj, 19713, 0);
         reg_anim_func(2, self_obj);
         reg_anim_func(1, 1);
@@ -437,14 +437,14 @@ end
 
 procedure Node001
 begin
-    gsay_reply(683, 103);
-    giq_option(4, 683, 104, Node010, 50);
-    giq_option(-3, 683, 106, Node018, 50);
+    Reply(103);
+    giq_option(4, NAME, 104, Node010, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 106, Node018, NEUTRAL_REACTION);
 end
 
 procedure Node002
 begin
-    float_msg(self_obj, message_str(683, 107 + Node2count), 0);
+    float_msg(self_obj, mstr(107 + Node2count), FLOAT_MSG_NORMAL);
     Node2count := Node2count + 1;
     if (Node2count == 4) then begin
         Node2count := 0;
@@ -456,84 +456,84 @@ begin
     if (Node3count == 3) then begin
         Node3count := 0;
     end
-    float_msg(self_obj, message_str(683, 111 + Node3count), 2);
+    float_msg(self_obj, mstr(111 + Node3count), FLOAT_MSG_RED);
     Node3count := Node3count + 1;
 end
 
 procedure Node010
 begin
-    gsay_reply(683, 114);
-    giq_option(4, 683, 115, Node011, 50);
+    Reply(114);
+    giq_option(4, NAME, 115, Node011, NEUTRAL_REACTION);
 end
 
 procedure Node011
 begin
-    gsay_reply(683, 116);
-    giq_option(4, 683, 117, Node012, 50);
+    Reply(116);
+    giq_option(4, NAME, 117, Node012, NEUTRAL_REACTION);
 end
 
 procedure Node012
 begin
-    gsay_reply(683, 118);
-    giq_option(4, 683, 119, Node013, 50);
+    Reply(118);
+    giq_option(4, NAME, 119, Node013, NEUTRAL_REACTION);
 end
 
 procedure Node013
 begin
-    gsay_reply(683, 120);
-    giq_option(4, 683, 121, Node014, 50);
+    Reply(120);
+    giq_option(4, NAME, 121, Node014, NEUTRAL_REACTION);
 end
 
 procedure Node014
 begin
-    gsay_reply(683, 122);
-    giq_option(4, 683, 123, Node015, 50);
-    giq_option(4, 683, 124, Node017, 50);
+    Reply(122);
+    giq_option(4, NAME, 123, Node015, NEUTRAL_REACTION);
+    giq_option(4, NAME, 124, Node017, NEUTRAL_REACTION);
 end
 
 procedure Node015
 begin
-    gsay_reply(683, 125);
-    giq_option(4, 683, 126, Node016, 50);
+    Reply(125);
+    giq_option(4, NAME, 126, Node016, NEUTRAL_REACTION);
 end
 
 procedure Node016
 begin
     item_caps_adjust(dude_obj, 500);
-    gsay_reply(683, 127);
-    giq_option(4, 683, 128, Node019, 50);
+    Reply(127);
+    giq_option(4, NAME, 128, Node019, NEUTRAL_REACTION);
 end
 
 procedure Node017
 begin
     set_map_var(0, 1);
-    gsay_reply(683, 129);
-    giq_option(0, 683, message_str(14, 350), Node999, 50);
+    Reply(129);
+    giq_option(0, NAME, g_mstr(350), Node999, NEUTRAL_REACTION);
 end
 
 procedure Node018
 begin
     set_map_var(0, 1);
-    gsay_reply(683, 131);
-    giq_option(0, 683, message_str(14, 350), Node999, 50);
+    Reply(131);
+    giq_option(0, NAME, g_mstr(350), Node999, NEUTRAL_REACTION);
 end
 
 procedure Node019
 begin
-    gsay_reply(683, 133);
-    giq_option(4, 683, 134, Node020, 50);
+    Reply(133);
+    giq_option(4, NAME, 134, Node020, NEUTRAL_REACTION);
 end
 
 procedure Node020
 begin
     set_map_var(0, 1);
-    gsay_reply(683, 135);
-    giq_option(0, 683, message_str(14, 350), Node999, 50);
+    Reply(135);
+    giq_option(0, NAME, g_mstr(350), Node999, NEUTRAL_REACTION);
 end
 
 procedure cycleTalk
 begin
-    float_msg(self_obj, message_str(683, 200 + cycleNum), 0);
+    float_msg(self_obj, mstr(200 + cycleNum), FLOAT_MSG_NORMAL);
     cycleNum := cycleNum + 1;
     if (cycleNum > 6) then begin
         cycleNum := 0;
@@ -608,4 +608,3 @@ begin
         end
     end
 end
-
