diff --git a/Mods/MegaMod/Original/ECCLAW.ssl b/Mods/MegaMod/ReDefine/ECCLAW.ssl
index 86e5f64..36a03b0 100644
--- a/Mods/MegaMod/Original/ECCLAW.ssl
+++ b/Mods/MegaMod/ReDefine/ECCLAW.ssl
@@ -144,28 +144,28 @@ end
 procedure look_at_p_proc
 begin
     script_overrides;
-    display_msg(message_str(1313, 100));
+    display_mstr(100);
 end
 
 procedure description_p_proc
 begin
     script_overrides;
-    display_msg(message_str(1313, 101));
+    display_mstr(101);
 end
 
 procedure critter_p_proc
 begin
     if (tile_distance_objs(self_obj, dude_obj) <= 6) then begin
         set_obj_visibility(self_obj, 0);
-        attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+        attack(dude_obj);
     end
 end
 
 procedure map_enter_p_proc
 begin
     set_obj_visibility(self_obj, 1);
-    critter_add_trait(self_obj, 1, 6, 187);
-    critter_add_trait(self_obj, 1, 5, 10);
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 187);
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_DEATHCLAW);
 end
 
 procedure map_exit_p_proc
@@ -187,4 +187,3 @@ end
 procedure start
 begin
 end
-
