diff --git a/Mods/MegaMod/Original/ECELRON.ssl b/Mods/MegaMod/ReDefine/ECELRON.ssl
index 3546e3b..c6235e8 100644
--- a/Mods/MegaMod/Original/ECELRON.ssl
+++ b/Mods/MegaMod/ReDefine/ECELRON.ssl
@@ -12,31 +12,30 @@ end
 
 procedure talk_p_proc
 begin
-    float_msg(self_obj, message_str(766, random(101, 105)), 8);
+    float_msg(self_obj, mstr(random(101, 105)), FLOAT_MSG_YELLOW);
 end
 
 procedure destroy_p_proc
 begin
-    if (cur_map_index == 94) then begin
+    if (CUR_MAP_BHRNDDST) then begin
         set_map_var(1, map_var(1) - 1);
     end
 end
 
 procedure map_enter_p_proc
 begin
-    critter_add_trait(self_obj, 1, 6, 177);
-    critter_add_trait(self_obj, 1, 5, 19);
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 177);
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_DOCTOR);
 end
 
 procedure description_p_proc
 begin
     script_overrides;
-    display_msg(message_str(766, 100));
+    display_mstr(100);
 end
 
 procedure look_at_p_proc
 begin
     script_overrides;
-    display_msg(message_str(766, 100));
+    display_mstr(100);
 end
-
