diff --git a/Mods/MegaMod/Original/ECENCLAW.ssl b/Mods/MegaMod/ReDefine/ECENCLAW.ssl
index 0972493..6561606 100644
--- a/Mods/MegaMod/Original/ECENCLAW.ssl
+++ b/Mods/MegaMod/ReDefine/ECENCLAW.ssl
@@ -143,9 +143,9 @@ end
 
 procedure critter_p_proc
 begin
-    if ((global_var(1109) < 3) or (global_var(1109) > 4)) then begin
+    if ((global_var(GVAR_MARY_ANN_SON_QUEST) < 3) or (global_var(GVAR_MARY_ANN_SON_QUEST) > 4)) then begin
         if (obj_can_see_obj(self_obj, dude_obj)) then begin
-            attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+            attack(dude_obj);
         end
     end
     if (not(anim_busy(self_obj))) then begin
@@ -158,8 +158,8 @@ end
 
 procedure map_enter_p_proc
 begin
-    critter_add_trait(self_obj, 1, 6, 190);
-    critter_add_trait(self_obj, 1, 5, 10);
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 190);
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_DEATHCLAW);
     starttile := tile_num(self_obj);
     reg_anim_func(2, self_obj);
     reg_anim_func(1, 1);
@@ -173,8 +173,7 @@ end
 
 procedure destroy_p_proc
 begin
-    if (cur_map_index == 94) then begin
+    if (CUR_MAP_BHRNDDST) then begin
         set_map_var(1, map_var(1) - 1);
     end
 end
-
