diff --git a/Mods/MegaMod/Original/ECJECK.ssl b/Mods/MegaMod/ReDefine/ECJECK.ssl
index ed5154b..cb544f5 100644
--- a/Mods/MegaMod/Original/ECJECK.ssl
+++ b/Mods/MegaMod/ReDefine/ECJECK.ssl
@@ -76,48 +76,48 @@ procedure start;
 
 procedure EvaluateKarma
 begin
-    set_global_var(37, 0);
-    set_global_var(38, 0);
-    set_global_var(39, 0);
-    set_global_var(40, 0);
-    set_global_var(41, 0);
-    set_global_var(42, 0);
-    set_global_var(43, 0);
-    set_global_var(44, 0);
-    set_global_var(45, 0);
-    if (global_var(0) >= 1000) then begin
-        set_global_var(37, 1);
+    set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+    set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+    set_global_var(GVAR_KARMA_DEFENDER, 0);
+    set_global_var(GVAR_KARMA_WANDERER, 0);
+    set_global_var(GVAR_KARMA_BETRAYER, 0);
+    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+    set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+    set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+    if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
     end
     else begin
-        if (global_var(0) >= 750) then begin
-            set_global_var(38, 1);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+            set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
         end
         else begin
-            if (global_var(0) >= 500) then begin
-                set_global_var(39, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
             end
             else begin
-                if (global_var(0) >= 250) then begin
-                    set_global_var(40, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                    set_global_var(GVAR_KARMA_DEFENDER, 1);
                 end
                 else begin
-                    if (global_var(0) > -250) then begin
-                        set_global_var(41, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                        set_global_var(GVAR_KARMA_WANDERER, 1);
                     end
                     else begin
-                        if (global_var(0) > -500) then begin
-                            set_global_var(42, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                            set_global_var(GVAR_KARMA_BETRAYER, 1);
                         end
                         else begin
-                            if (global_var(0) > -750) then begin
-                                set_global_var(43, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                             end
                             else begin
-                                if (global_var(0) > -1000) then begin
-                                    set_global_var(44, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                    set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                 end
                                 else begin
-                                    set_global_var(45, 1);
+                                    set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                 end
                             end
                         end
@@ -130,60 +130,60 @@ end
 
 procedure CheckRep
 begin
-    if (((global_var(4) + global_var(5)) >= 25) and ((global_var(4) > (3 * global_var(5))) or (global_var(2) == 1)) and (global_var(1) == 0)) then begin
-        set_global_var(2, 1);
-        set_global_var(3, 0);
+    if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(4) > (3 * global_var(GVAR_GOOD_MONSTER))) or (global_var(GVAR_CHAMPION_REPUTATION) == 1)) and (global_var(GVAR_CHILDKILLER_REPUTATION) == 0)) then begin
+        set_global_var(GVAR_CHAMPION_REPUTATION, 1);
+        set_global_var(GVAR_BERSERKER_REPUTATION, 0);
     end
-    if (((global_var(4) + global_var(5)) >= 25) and ((global_var(5) > (2 * global_var(4))) or (global_var(3) == 1))) then begin
-        set_global_var(3, 1);
-        set_global_var(2, 0);
+    if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
+        set_global_var(GVAR_BERSERKER_REPUTATION, 1);
+        set_global_var(GVAR_CHAMPION_REPUTATION, 0);
     end
 end
 
 procedure Node002
 begin
     set_local_var(0, 1);
-    gsay_reply(1307, 202);
-    giq_option(1, 1307, 203, Node003, 50);
+    Reply(202);
+    giq_option(1, NAME, 203, Node003, NEUTRAL_REACTION);
 end
 
 procedure Node003
 begin
-    gsay_reply(1307, 204);
-    if (get_critter_stat(self_obj, 4) >= 2) then begin
-        giq_option(-4, 1307, 205, Node004, 50);
+    Reply(204);
+    if (self_iq >= 2) then begin
+        giq_option(-4, NAME, 205, Node004, NEUTRAL_REACTION);
     end
-    giq_option(5, 1307, 206, Node004, 50);
-    if (get_critter_stat(self_obj, 4) >= 2) then begin
-        giq_option(-4, 1307, 207, Node008, 50);
+    giq_option(5, NAME, 206, Node004, NEUTRAL_REACTION);
+    if (self_iq >= 2) then begin
+        giq_option(-4, NAME, 207, Node008, NEUTRAL_REACTION);
     end
-    giq_option(5, 1307, 208, Node008, 50);
-    giq_option(-1, 1307, 209, Node004, 50);
+    giq_option(5, NAME, 208, Node008, NEUTRAL_REACTION);
+    giq_option(-1, NAME, 209, Node004, NEUTRAL_REACTION);
 end
 
 procedure Node004
 begin
-    gsay_reply(1307, 210);
-    if (get_critter_stat(self_obj, 4) >= 2) then begin
-        giq_option(-4, 1307, 211, Node005, 50);
-    end
-    giq_option(5, 1307, 212, Node005, 50);
-    if (get_critter_stat(self_obj, 4) >= 2) then begin
-        giq_option(-4, 1307, 213, Node007, 50);
-    end
-    giq_option(5, 1307, 214, Node007, 50);
-    giq_option(-1, 1307, 215, Node005, 50);
-    giq_option(-4, 1307, 216, Node008, 50);
-    giq_option(5, 1307, 217, Node008, 50);
+    Reply(210);
+    if (self_iq >= 2) then begin
+        giq_option(-4, NAME, 211, Node005, NEUTRAL_REACTION);
+    end
+    giq_option(5, NAME, 212, Node005, NEUTRAL_REACTION);
+    if (self_iq >= 2) then begin
+        giq_option(-4, NAME, 213, Node007, NEUTRAL_REACTION);
+    end
+    giq_option(5, NAME, 214, Node007, NEUTRAL_REACTION);
+    giq_option(-1, NAME, 215, Node005, NEUTRAL_REACTION);
+    giq_option(-4, NAME, 216, Node008, NEUTRAL_REACTION);
+    giq_option(5, NAME, 217, Node008, NEUTRAL_REACTION);
 end
 
 procedure Node005
 begin
-    add_mult_objs_to_inven(dude_obj, create_object_sid(532, 0, 0, -1), 1);
+    add_mult_objs_to_inven(dude_obj, create_object(532, 0, 0), 1);
     set_local_var(2, 1);
-    gsay_reply(1307, 218);
-    giq_option(5, 1307, 219, Node999, 50);
-    giq_option(-4, 1307, 220, Node999, 50);
+    Reply(218);
+    giq_option(5, NAME, 219, Node999, NEUTRAL_REACTION);
+    giq_option(-4, NAME, 220, Node999, NEUTRAL_REACTION);
 end
 
 /*******************************************************
@@ -198,34 +198,34 @@ end
 
 procedure Node007
 begin
-    gsay_reply(1307, 221);
-    giq_option(5, 1307, 222, Node005, 50);
-    if (get_critter_stat(self_obj, 4) >= 2) then begin
-        giq_option(-4, 1307, 223, Node005, 50);
+    Reply(221);
+    giq_option(5, NAME, 222, Node005, NEUTRAL_REACTION);
+    if (self_iq >= 2) then begin
+        giq_option(-4, NAME, 223, Node005, NEUTRAL_REACTION);
     end
-    giq_option(5, 1307, 224, Node008, 50);
-    giq_option(-4, 1307, 225, Node008, 50);
+    giq_option(5, NAME, 224, Node008, NEUTRAL_REACTION);
+    giq_option(-4, NAME, 225, Node008, NEUTRAL_REACTION);
 end
 
 procedure Node008
 begin
-    gsay_reply(1307, 226);
-    giq_option(1, 1307, 227, Node999, 50);
+    Reply(226);
+    giq_option(1, NAME, 227, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node009
 begin
-    gsay_reply(1307, 228);
-    if (get_critter_stat(self_obj, 4) >= 2) then begin
-        giq_option(-4, 1307, 229, Node005, 50);
+    Reply(228);
+    if (self_iq >= 2) then begin
+        giq_option(-4, NAME, 229, Node005, NEUTRAL_REACTION);
     end
-    giq_option(5, 1307, 230, Node005, 50);
-    if (get_critter_stat(self_obj, 4) >= 2) then begin
-        giq_option(-4, 1307, 231, Node007, 50);
+    giq_option(5, NAME, 230, Node005, NEUTRAL_REACTION);
+    if (self_iq >= 2) then begin
+        giq_option(-4, NAME, 231, Node007, NEUTRAL_REACTION);
     end
-    giq_option(5, 1307, 232, Node007, 50);
-    giq_option(-4, 1307, 233, Node008, 50);
-    giq_option(5, 1307, 234, Node008, 50);
+    giq_option(5, NAME, 232, Node007, NEUTRAL_REACTION);
+    giq_option(-4, NAME, 233, Node008, NEUTRAL_REACTION);
+    giq_option(5, NAME, 234, Node008, NEUTRAL_REACTION);
 end
 
 procedure Node010
@@ -233,70 +233,70 @@ begin
     variable LVar0 := 0;
     set_local_var(1, 1);
     give_exp_points(200);
-    display_msg(message_str(14, 100) + 200 + message_str(14, 101));
-    add_mult_objs_to_inven(dude_obj, create_object_sid(71, 0, 0, -1), 5);
-    gsay_reply(1307, 235);
+    display_msg(g_mstr(100) + 200 + g_mstr(101));
+    add_mult_objs_to_inven(dude_obj, create_object(71, 0, 0), 5);
+    Reply(235);
     LVar0 := random(1, 3);
     if (LVar0 == 1) then begin
-        giq_option(1, 1307, 236, Node011, 50);
+        giq_option(1, NAME, 236, Node011, NEUTRAL_REACTION);
     end
     if (LVar0 == 2) then begin
-        giq_option(1, 1307, 236, Node012, 50);
+        giq_option(1, NAME, 236, Node012, NEUTRAL_REACTION);
     end
     if (LVar0 == 3) then begin
-        giq_option(1, 1307, 236, Node013, 50);
+        giq_option(1, NAME, 236, Node013, NEUTRAL_REACTION);
     end
-    giq_option(5, 1307, 237, Node016, 50);
-    giq_option(-4, 1307, 238, Node016, 50);
+    giq_option(5, NAME, 237, Node016, NEUTRAL_REACTION);
+    giq_option(-4, NAME, 238, Node016, NEUTRAL_REACTION);
 end
 
 procedure Node011
 begin
-    gsay_reply(1307, 239);
-    giq_option(5, 1307, 240, Node014, 50);
-    giq_option(5, 1307, 241, Node016, 50);
-    giq_option(-4, 1307, 242, Node016, 50);
+    Reply(239);
+    giq_option(5, NAME, 240, Node014, NEUTRAL_REACTION);
+    giq_option(5, NAME, 241, Node016, NEUTRAL_REACTION);
+    giq_option(-4, NAME, 242, Node016, NEUTRAL_REACTION);
 end
 
 procedure Node012
 begin
-    gsay_reply(1307, 243);
-    giq_option(5, 1307, 244, Node014, 50);
-    giq_option(5, 1307, 245, Node016, 50);
-    giq_option(-4, 1307, 246, Node016, 50);
+    Reply(243);
+    giq_option(5, NAME, 244, Node014, NEUTRAL_REACTION);
+    giq_option(5, NAME, 245, Node016, NEUTRAL_REACTION);
+    giq_option(-4, NAME, 246, Node016, NEUTRAL_REACTION);
 end
 
 procedure Node013
 begin
-    gsay_reply(1307, 247);
-    giq_option(5, 1307, 248, Node014, 50);
-    giq_option(5, 1307, 249, Node016, 50);
-    giq_option(-4, 1307, 250, Node016, 50);
+    Reply(247);
+    giq_option(5, NAME, 248, Node014, NEUTRAL_REACTION);
+    giq_option(5, NAME, 249, Node016, NEUTRAL_REACTION);
+    giq_option(-4, NAME, 250, Node016, NEUTRAL_REACTION);
 end
 
 procedure Node014
 begin
-    gsay_reply(1307, 251);
-    giq_option(5, 1307, 252, Node015, 50);
-    giq_option(5, 1307, 253, Node016, 50);
+    Reply(251);
+    giq_option(5, NAME, 252, Node015, NEUTRAL_REACTION);
+    giq_option(5, NAME, 253, Node016, NEUTRAL_REACTION);
 end
 
 procedure Node015
 begin
-    gsay_reply(1307, 254);
-    giq_option(1, 1307, 255, Node016, 50);
+    Reply(254);
+    giq_option(1, NAME, 255, Node016, NEUTRAL_REACTION);
 end
 
 procedure Node016
 begin
-    if (get_critter_stat(dude_obj, 34) == 0) then begin
-        gsay_reply(1307, 256);
+    if (dude_gender == 0) then begin
+        Reply(256);
     end
-    if (get_critter_stat(dude_obj, 34) == 1) then begin
-        gsay_reply(1307, 257);
+    if (dude_gender == 1) then begin
+        Reply(257);
     end
-    giq_option(1, 1307, obj_name(dude_obj) + message_str(1307, 258) + obj_name(dude_obj) + message_str(1307, 300), Node999, 50);
-    giq_option(5, 1307, 259, Node999, 50);
+    giq_option(1, NAME, dude_name + mstr(258) + dude_name + mstr(300), Node999, NEUTRAL_REACTION);
+    giq_option(5, NAME, 259, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node017
@@ -410,7 +410,7 @@ end
 
 procedure talk_p_proc
 begin
-    if (global_var(682) == 0) then begin
+    if (global_var(GVAR_RNDBAND_FLAG) == 0) then begin
         if (local_var(0) == 0) then begin
             start_gdialog(1307, self_obj, 4, -1, -1);
             gsay_start;
@@ -427,7 +427,7 @@ begin
                 end_dialogue;
             end
             else begin
-                float_msg(self_obj, message_str(1307, 105), 3);
+                float_msg(self_obj, mstr(105), FLOAT_MSG_GREEN);
             end
         end
     end
@@ -440,33 +440,33 @@ begin
             end_dialogue;
         end
         else begin
-            float_msg(self_obj, message_str(1307, 260), 3);
+            float_msg(self_obj, mstr(260), FLOAT_MSG_GREEN);
         end
     end
 end
 
 procedure map_enter_p_proc
 begin
-    if ((global_var(682) == 0) and (map_var(2) == 1)) then begin
+    if ((global_var(GVAR_RNDBAND_FLAG) == 0) and (map_var(2) == 1)) then begin
         set_obj_visibility(self_obj, 1);
         set_local_var(7, 1);
     end
     else begin
-        critter_add_trait(self_obj, 1, 6, 1);
-        critter_add_trait(self_obj, 1, 5, 1);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 1);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_ARROYO_WARRIOR);
     end
 end
 
 procedure look_at_p_proc
 begin
     script_overrides;
-    display_msg(message_str(1307, random(200, 201)));
+    display_mstr(random(200, 201));
 end
 
 procedure description_p_proc
 begin
     script_overrides;
-    display_msg(message_str(1307, random(200, 201)));
+    display_mstr(random(200, 201));
 end
 
 procedure critter_p_proc
@@ -478,7 +478,7 @@ begin
         end
         if ((local_var(3) == 2) and obj_can_see_obj(self_obj, dude_obj)) then begin
             set_local_var(3, 1);
-            attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+            attack(dude_obj);
         end
         if (tile_distance_objs(self_obj, dude_obj) <= 4) then begin
             if ((talk_flag == 0) and (combat_is_initialized == 0)) then begin
@@ -510,15 +510,15 @@ end
 procedure timed_event_p_proc
 begin
     if ((fixed_param == 1) and (combat_is_initialized == 0)) then begin
-        float_msg(self_obj, message_str(1307, random(100, 104)), 0);
+        float_msg(self_obj, mstr(random(100, 104)), FLOAT_MSG_NORMAL);
         In_Timed_Event := 0;
     end
 end
 
 procedure map_exit_p_proc
 begin
-    if ((local_var(0) == 1) and (global_var(682) == 0)) then begin
-        set_global_var(0, global_var(0) - 5);
+    if ((local_var(0) == 1) and (global_var(GVAR_RNDBAND_FLAG) == 0)) then begin
+        set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) - 5);
     end
 end
 
@@ -532,23 +532,23 @@ end
 procedure destroy_p_proc
 begin
     if (source_obj == dude_obj) then begin
-        set_global_var(5, global_var(5) + 1);
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(4) > (3 * global_var(5))) or (global_var(2) == 1)) and (global_var(1) == 0)) then begin
-            set_global_var(2, 1);
-            set_global_var(3, 0);
+        set_global_var(GVAR_GOOD_MONSTER, global_var(GVAR_GOOD_MONSTER) + 1);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(4) > (3 * global_var(GVAR_GOOD_MONSTER))) or (global_var(GVAR_CHAMPION_REPUTATION) == 1)) and (global_var(GVAR_CHILDKILLER_REPUTATION) == 0)) then begin
+            set_global_var(GVAR_CHAMPION_REPUTATION, 1);
+            set_global_var(GVAR_BERSERKER_REPUTATION, 0);
         end
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(5) > (2 * global_var(4))) or (global_var(3) == 1))) then begin
-            set_global_var(3, 1);
-            set_global_var(2, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
+            set_global_var(GVAR_BERSERKER_REPUTATION, 1);
+            set_global_var(GVAR_CHAMPION_REPUTATION, 0);
         end
-        set_global_var(0, global_var(0) - 10);
+        set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) - 10);
         call EvaluateKarma();
     end
 end
 
 procedure damage_p_proc
 begin
-    set_global_var(65, global_var(65) - 50);
+    set_global_var(GVAR_TOWN_REP_PRIMITIVE_TRIBE, global_var(GVAR_TOWN_REP_PRIMITIVE_TRIBE) - 50);
     if (source_obj == dude_obj) then begin
         set_local_var(3, 2);
     end
@@ -557,4 +557,3 @@ end
 procedure start
 begin
 end
-
