diff --git a/Mods/MegaMod/Original/ECKAGA2.ssl b/Mods/MegaMod/ReDefine/ECKAGA2.ssl
index bf9fad4..a56208d 100644
--- a/Mods/MegaMod/Original/ECKAGA2.ssl
+++ b/Mods/MegaMod/ReDefine/ECKAGA2.ssl
@@ -161,17 +161,17 @@ begin
     if (local_var(6)) then begin
         if (obj_on_screen(self_obj)) then begin
             if (not(anim_busy(self_obj))) then begin
-                animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(self_obj), rotation_to_tile(tile_num(dude_obj), tile_num(self_obj)), 5), 1);
+                animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(self_obj), rotation_to_tile(dude_tile, tile_num(self_obj)), 5), 1);
             end
         end
         else begin
-            set_global_var(399, global_var(399) bwor 64);
+            set_global_var(GVAR_RND_KAGA_STATE, global_var(GVAR_RND_KAGA_STATE) bwor 64);
             destroy_object(self_obj);
         end
     end
     else begin
         if (obj_can_see_obj(self_obj, dude_obj)) then begin
-            attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+            attack(dude_obj);
         end
     end
 end
@@ -190,54 +190,54 @@ end
 procedure destroy_p_proc
 begin
     if (source_obj == dude_obj) then begin
-        set_global_var(4, global_var(4) + 1);
-        if (metarule(51, self_obj) == 2) then begin
+        set_global_var(GVAR_BAD_MONSTER, global_var(GVAR_BAD_MONSTER) + 1);
+        if (critter_kill_type(self_obj) == KILL_TYPE_children_kills) then begin
             if (source_obj == dude_obj) then begin
-                set_global_var(2, 0);
-                set_global_var(1, global_var(1) + 1);
-                set_global_var(0, global_var(0) - -15);
-                set_global_var(37, 0);
-                set_global_var(38, 0);
-                set_global_var(39, 0);
-                set_global_var(40, 0);
-                set_global_var(41, 0);
-                set_global_var(42, 0);
-                set_global_var(43, 0);
-                set_global_var(44, 0);
-                set_global_var(45, 0);
-                if (global_var(0) >= 1000) then begin
-                    set_global_var(37, 1);
+                set_global_var(GVAR_CHAMPION_REPUTATION, 0);
+                set_global_var(GVAR_CHILDKILLER_REPUTATION, global_var(GVAR_CHILDKILLER_REPUTATION) + 1);
+                set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) - -15);
+                set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+                set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+                set_global_var(GVAR_KARMA_DEFENDER, 0);
+                set_global_var(GVAR_KARMA_WANDERER, 0);
+                set_global_var(GVAR_KARMA_BETRAYER, 0);
+                set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+                set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+                set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+                    set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
                 end
                 else begin
-                    if (global_var(0) >= 750) then begin
-                        set_global_var(38, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
                     end
                     else begin
-                        if (global_var(0) >= 500) then begin
-                            set_global_var(39, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                            set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                         end
                         else begin
-                            if (global_var(0) >= 250) then begin
-                                set_global_var(40, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                                set_global_var(GVAR_KARMA_DEFENDER, 1);
                             end
                             else begin
-                                if (global_var(0) > -250) then begin
-                                    set_global_var(41, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                                    set_global_var(GVAR_KARMA_WANDERER, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -500) then begin
-                                        set_global_var(42, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                        set_global_var(GVAR_KARMA_BETRAYER, 1);
                                     end
                                     else begin
-                                        if (global_var(0) > -750) then begin
-                                            set_global_var(43, 1);
+                                        if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                            set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                         end
                                         else begin
-                                            if (global_var(0) > -1000) then begin
-                                                set_global_var(44, 1);
+                                            if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                                set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                             end
                                             else begin
-                                                set_global_var(45, 1);
+                                                set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                             end
                                         end
                                     end
@@ -248,58 +248,58 @@ begin
                 end
             end
         end
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(5) > (2 * global_var(4))) or (global_var(3) == 1))) then begin
-            set_global_var(3, 1);
-            set_global_var(2, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
+            set_global_var(GVAR_BERSERKER_REPUTATION, 1);
+            set_global_var(GVAR_CHAMPION_REPUTATION, 0);
         end
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(4) > (3 * global_var(5))) or (global_var(2) == 1)) and (global_var(1) == 0)) then begin
-            set_global_var(2, 1);
-            set_global_var(3, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(4) > (3 * global_var(GVAR_GOOD_MONSTER))) or (global_var(GVAR_CHAMPION_REPUTATION) == 1)) and (global_var(GVAR_CHILDKILLER_REPUTATION) == 0)) then begin
+            set_global_var(GVAR_CHAMPION_REPUTATION, 1);
+            set_global_var(GVAR_BERSERKER_REPUTATION, 0);
         end
-        set_global_var(0, global_var(0) + 5);
+        set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + 5);
         debug_msg("Player gains " + 5 + " Karma Points.");
-        set_global_var(37, 0);
-        set_global_var(38, 0);
-        set_global_var(39, 0);
-        set_global_var(40, 0);
-        set_global_var(41, 0);
-        set_global_var(42, 0);
-        set_global_var(43, 0);
-        set_global_var(44, 0);
-        set_global_var(45, 0);
-        if (global_var(0) >= 1000) then begin
-            set_global_var(37, 1);
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+        set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+        set_global_var(GVAR_KARMA_DEFENDER, 0);
+        set_global_var(GVAR_KARMA_WANDERER, 0);
+        set_global_var(GVAR_KARMA_BETRAYER, 0);
+        set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+        set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+            set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
         end
         else begin
-            if (global_var(0) >= 750) then begin
-                set_global_var(38, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
             end
             else begin
-                if (global_var(0) >= 500) then begin
-                    set_global_var(39, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                 end
                 else begin
-                    if (global_var(0) >= 250) then begin
-                        set_global_var(40, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                        set_global_var(GVAR_KARMA_DEFENDER, 1);
                     end
                     else begin
-                        if (global_var(0) > -250) then begin
-                            set_global_var(41, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                            set_global_var(GVAR_KARMA_WANDERER, 1);
                         end
                         else begin
-                            if (global_var(0) > -500) then begin
-                                set_global_var(42, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                set_global_var(GVAR_KARMA_BETRAYER, 1);
                             end
                             else begin
-                                if (global_var(0) > -750) then begin
-                                    set_global_var(43, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -1000) then begin
-                                        set_global_var(44, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                     end
                                     else begin
-                                        set_global_var(45, 1);
+                                        set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                     end
                                 end
                             end
@@ -309,26 +309,26 @@ begin
             end
         end
     end
-    set_global_var(399, global_var(399) bwor 32);
-    set_global_var(399, global_var(399) bwor 64);
-    set_global_var(764, 1);
+    set_global_var(GVAR_RND_KAGA_STATE, global_var(GVAR_RND_KAGA_STATE) bwor 32);
+    set_global_var(GVAR_RND_KAGA_STATE, global_var(GVAR_RND_KAGA_STATE) bwor 64);
+    set_global_var(GVAR_RESEARCH_BASE_LEVEL, 1);
 end
 
 procedure look_at_p_proc
 begin
     script_overrides;
     if (local_var(4) == 0) then begin
-        display_msg(message_str(832, 100));
+        display_mstr(100);
     end
     else begin
-        display_msg(message_str(832, 101));
+        display_mstr(101);
     end
 end
 
 procedure description_p_proc
 begin
     script_overrides;
-    display_msg(message_str(832, 102));
+    display_mstr(102);
 end
 
 procedure use_skill_on_p_proc
@@ -346,14 +346,14 @@ procedure map_enter_p_proc
 begin
     variable LVar0 := 0;
     Only_Once := 0;
-    critter_add_trait(self_obj, 1, 6, 194);
-    critter_add_trait(self_obj, 1, 5, 141);
-    set_global_var(399, global_var(399) bwor 2);
-    restock_amt := random(1, 1);
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 194);
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_KAGA);
+    set_global_var(GVAR_RND_KAGA_STATE, global_var(GVAR_RND_KAGA_STATE) bwor 2);
+    restock_amt := 1;
     if (obj_is_carrying_obj_pid(self_obj, 22) < restock_amt) then begin
         if (100 >= random(1, 100)) then begin
             if (obj_is_carrying_obj_pid(self_obj, 22) < restock_amt) then begin
-                restock_obj := create_object_sid(22, 0, 0, -1);
+                restock_obj := create_object(22, 0, 0);
                 add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 22));
             end
             else begin
@@ -366,17 +366,17 @@ begin
                         restock_obj := obj_carrying_pid_obj(self_obj, 22);
                         if (obj_type(self_obj) == 1) then begin
                             restock_obj := obj_carrying_pid_obj(self_obj, 22);
-                            if (critter_inven_obj(self_obj, 0) == restock_obj) then begin
-                                if (obj_item_subtype(critter_inven_obj(self_obj, 0)) == 0) then begin
-                                    restock_obj := critter_inven_obj(self_obj, 0);
+                            if (critter_inven_obj(self_obj, INVEN_TYPE_WORN) == restock_obj) then begin
+                                if (obj_item_subtype(critter_inven_obj(self_obj, INVEN_TYPE_WORN)) == 0) then begin
+                                    restock_obj := critter_inven_obj(self_obj, INVEN_TYPE_WORN);
                                     debug_msg("armour pid == " + obj_pid(restock_obj));
                                     rm_obj_from_inven(self_obj, restock_obj);
                                     add_obj_to_inven(self_obj, restock_obj);
                                 end
                             end
                             else begin
-                                if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                    metarule(43, self_obj);
+                                if ((critter_inven_obj(self_obj, INVEN_TYPE_LEFT_HAND) == restock_obj) or (critter_inven_obj(self_obj, INVEN_TYPE_RIGHT_HAND) == restock_obj)) then begin
+                                    inven_unwield(self_obj);
                                 end
                             end
                         end
@@ -394,7 +394,7 @@ begin
     end
     else begin
         if (obj_is_carrying_obj_pid(self_obj, 22) < restock_amt) then begin
-            restock_obj := create_object_sid(22, 0, 0, -1);
+            restock_obj := create_object(22, 0, 0);
             add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 22));
         end
         else begin
@@ -407,17 +407,17 @@ begin
                     restock_obj := obj_carrying_pid_obj(self_obj, 22);
                     if (obj_type(self_obj) == 1) then begin
                         restock_obj := obj_carrying_pid_obj(self_obj, 22);
-                        if (critter_inven_obj(self_obj, 0) == restock_obj) then begin
-                            if (obj_item_subtype(critter_inven_obj(self_obj, 0)) == 0) then begin
-                                restock_obj := critter_inven_obj(self_obj, 0);
+                        if (critter_inven_obj(self_obj, INVEN_TYPE_WORN) == restock_obj) then begin
+                            if (obj_item_subtype(critter_inven_obj(self_obj, INVEN_TYPE_WORN)) == 0) then begin
+                                restock_obj := critter_inven_obj(self_obj, INVEN_TYPE_WORN);
                                 debug_msg("armour pid == " + obj_pid(restock_obj));
                                 rm_obj_from_inven(self_obj, restock_obj);
                                 add_obj_to_inven(self_obj, restock_obj);
                             end
                         end
                         else begin
-                            if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                metarule(43, self_obj);
+                            if ((critter_inven_obj(self_obj, INVEN_TYPE_LEFT_HAND) == restock_obj) or (critter_inven_obj(self_obj, INVEN_TYPE_RIGHT_HAND) == restock_obj)) then begin
+                                inven_unwield(self_obj);
                             end
                         end
                     end
@@ -432,11 +432,11 @@ begin
             end
         end
     end
-    restock_amt := random(1, 1);
+    restock_amt := 1;
     if (obj_is_carrying_obj_pid(self_obj, 33) < restock_amt) then begin
         if (100 >= random(1, 100)) then begin
             if (obj_is_carrying_obj_pid(self_obj, 33) < restock_amt) then begin
-                restock_obj := create_object_sid(33, 0, 0, -1);
+                restock_obj := create_object(33, 0, 0);
                 add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 33));
             end
             else begin
@@ -449,17 +449,17 @@ begin
                         restock_obj := obj_carrying_pid_obj(self_obj, 33);
                         if (obj_type(self_obj) == 1) then begin
                             restock_obj := obj_carrying_pid_obj(self_obj, 33);
-                            if (critter_inven_obj(self_obj, 0) == restock_obj) then begin
-                                if (obj_item_subtype(critter_inven_obj(self_obj, 0)) == 0) then begin
-                                    restock_obj := critter_inven_obj(self_obj, 0);
+                            if (critter_inven_obj(self_obj, INVEN_TYPE_WORN) == restock_obj) then begin
+                                if (obj_item_subtype(critter_inven_obj(self_obj, INVEN_TYPE_WORN)) == 0) then begin
+                                    restock_obj := critter_inven_obj(self_obj, INVEN_TYPE_WORN);
                                     debug_msg("armour pid == " + obj_pid(restock_obj));
                                     rm_obj_from_inven(self_obj, restock_obj);
                                     add_obj_to_inven(self_obj, restock_obj);
                                 end
                             end
                             else begin
-                                if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                    metarule(43, self_obj);
+                                if ((critter_inven_obj(self_obj, INVEN_TYPE_LEFT_HAND) == restock_obj) or (critter_inven_obj(self_obj, INVEN_TYPE_RIGHT_HAND) == restock_obj)) then begin
+                                    inven_unwield(self_obj);
                                 end
                             end
                         end
@@ -477,7 +477,7 @@ begin
     end
     else begin
         if (obj_is_carrying_obj_pid(self_obj, 33) < restock_amt) then begin
-            restock_obj := create_object_sid(33, 0, 0, -1);
+            restock_obj := create_object(33, 0, 0);
             add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 33));
         end
         else begin
@@ -490,17 +490,17 @@ begin
                     restock_obj := obj_carrying_pid_obj(self_obj, 33);
                     if (obj_type(self_obj) == 1) then begin
                         restock_obj := obj_carrying_pid_obj(self_obj, 33);
-                        if (critter_inven_obj(self_obj, 0) == restock_obj) then begin
-                            if (obj_item_subtype(critter_inven_obj(self_obj, 0)) == 0) then begin
-                                restock_obj := critter_inven_obj(self_obj, 0);
+                        if (critter_inven_obj(self_obj, INVEN_TYPE_WORN) == restock_obj) then begin
+                            if (obj_item_subtype(critter_inven_obj(self_obj, INVEN_TYPE_WORN)) == 0) then begin
+                                restock_obj := critter_inven_obj(self_obj, INVEN_TYPE_WORN);
                                 debug_msg("armour pid == " + obj_pid(restock_obj));
                                 rm_obj_from_inven(self_obj, restock_obj);
                                 add_obj_to_inven(self_obj, restock_obj);
                             end
                         end
                         else begin
-                            if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                metarule(43, self_obj);
+                            if ((critter_inven_obj(self_obj, INVEN_TYPE_LEFT_HAND) == restock_obj) or (critter_inven_obj(self_obj, INVEN_TYPE_RIGHT_HAND) == restock_obj)) then begin
+                                inven_unwield(self_obj);
                             end
                         end
                     end
@@ -515,11 +515,11 @@ begin
             end
         end
     end
-    restock_amt := random(1, 1);
+    restock_amt := 1;
     if (obj_is_carrying_obj_pid(self_obj, 23) < restock_amt) then begin
         if (100 >= random(1, 100)) then begin
             if (obj_is_carrying_obj_pid(self_obj, 23) < restock_amt) then begin
-                restock_obj := create_object_sid(23, 0, 0, -1);
+                restock_obj := create_object(23, 0, 0);
                 add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 23));
             end
             else begin
@@ -532,17 +532,17 @@ begin
                         restock_obj := obj_carrying_pid_obj(self_obj, 23);
                         if (obj_type(self_obj) == 1) then begin
                             restock_obj := obj_carrying_pid_obj(self_obj, 23);
-                            if (critter_inven_obj(self_obj, 0) == restock_obj) then begin
-                                if (obj_item_subtype(critter_inven_obj(self_obj, 0)) == 0) then begin
-                                    restock_obj := critter_inven_obj(self_obj, 0);
+                            if (critter_inven_obj(self_obj, INVEN_TYPE_WORN) == restock_obj) then begin
+                                if (obj_item_subtype(critter_inven_obj(self_obj, INVEN_TYPE_WORN)) == 0) then begin
+                                    restock_obj := critter_inven_obj(self_obj, INVEN_TYPE_WORN);
                                     debug_msg("armour pid == " + obj_pid(restock_obj));
                                     rm_obj_from_inven(self_obj, restock_obj);
                                     add_obj_to_inven(self_obj, restock_obj);
                                 end
                             end
                             else begin
-                                if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                    metarule(43, self_obj);
+                                if ((critter_inven_obj(self_obj, INVEN_TYPE_LEFT_HAND) == restock_obj) or (critter_inven_obj(self_obj, INVEN_TYPE_RIGHT_HAND) == restock_obj)) then begin
+                                    inven_unwield(self_obj);
                                 end
                             end
                         end
@@ -560,7 +560,7 @@ begin
     end
     else begin
         if (obj_is_carrying_obj_pid(self_obj, 23) < restock_amt) then begin
-            restock_obj := create_object_sid(23, 0, 0, -1);
+            restock_obj := create_object(23, 0, 0);
             add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 23));
         end
         else begin
@@ -573,17 +573,17 @@ begin
                     restock_obj := obj_carrying_pid_obj(self_obj, 23);
                     if (obj_type(self_obj) == 1) then begin
                         restock_obj := obj_carrying_pid_obj(self_obj, 23);
-                        if (critter_inven_obj(self_obj, 0) == restock_obj) then begin
-                            if (obj_item_subtype(critter_inven_obj(self_obj, 0)) == 0) then begin
-                                restock_obj := critter_inven_obj(self_obj, 0);
+                        if (critter_inven_obj(self_obj, INVEN_TYPE_WORN) == restock_obj) then begin
+                            if (obj_item_subtype(critter_inven_obj(self_obj, INVEN_TYPE_WORN)) == 0) then begin
+                                restock_obj := critter_inven_obj(self_obj, INVEN_TYPE_WORN);
                                 debug_msg("armour pid == " + obj_pid(restock_obj));
                                 rm_obj_from_inven(self_obj, restock_obj);
                                 add_obj_to_inven(self_obj, restock_obj);
                             end
                         end
                         else begin
-                            if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                metarule(43, self_obj);
+                            if ((critter_inven_obj(self_obj, INVEN_TYPE_LEFT_HAND) == restock_obj) or (critter_inven_obj(self_obj, INVEN_TYPE_RIGHT_HAND) == restock_obj)) then begin
+                                inven_unwield(self_obj);
                             end
                         end
                     end
@@ -602,7 +602,7 @@ begin
     if (obj_is_carrying_obj_pid(self_obj, 41) < restock_amt) then begin
         if (100 >= random(1, 100)) then begin
             if (obj_is_carrying_obj_pid(self_obj, 41) < restock_amt) then begin
-                restock_obj := create_object_sid(41, 0, 0, -1);
+                restock_obj := create_object(41, 0, 0);
                 add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 41));
             end
             else begin
@@ -615,17 +615,17 @@ begin
                         restock_obj := obj_carrying_pid_obj(self_obj, 41);
                         if (obj_type(self_obj) == 1) then begin
                             restock_obj := obj_carrying_pid_obj(self_obj, 41);
-                            if (critter_inven_obj(self_obj, 0) == restock_obj) then begin
-                                if (obj_item_subtype(critter_inven_obj(self_obj, 0)) == 0) then begin
-                                    restock_obj := critter_inven_obj(self_obj, 0);
+                            if (critter_inven_obj(self_obj, INVEN_TYPE_WORN) == restock_obj) then begin
+                                if (obj_item_subtype(critter_inven_obj(self_obj, INVEN_TYPE_WORN)) == 0) then begin
+                                    restock_obj := critter_inven_obj(self_obj, INVEN_TYPE_WORN);
                                     debug_msg("armour pid == " + obj_pid(restock_obj));
                                     rm_obj_from_inven(self_obj, restock_obj);
                                     add_obj_to_inven(self_obj, restock_obj);
                                 end
                             end
                             else begin
-                                if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                    metarule(43, self_obj);
+                                if ((critter_inven_obj(self_obj, INVEN_TYPE_LEFT_HAND) == restock_obj) or (critter_inven_obj(self_obj, INVEN_TYPE_RIGHT_HAND) == restock_obj)) then begin
+                                    inven_unwield(self_obj);
                                 end
                             end
                         end
@@ -643,7 +643,7 @@ begin
     end
     else begin
         if (obj_is_carrying_obj_pid(self_obj, 41) < restock_amt) then begin
-            restock_obj := create_object_sid(41, 0, 0, -1);
+            restock_obj := create_object(41, 0, 0);
             add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 41));
         end
         else begin
@@ -656,17 +656,17 @@ begin
                     restock_obj := obj_carrying_pid_obj(self_obj, 41);
                     if (obj_type(self_obj) == 1) then begin
                         restock_obj := obj_carrying_pid_obj(self_obj, 41);
-                        if (critter_inven_obj(self_obj, 0) == restock_obj) then begin
-                            if (obj_item_subtype(critter_inven_obj(self_obj, 0)) == 0) then begin
-                                restock_obj := critter_inven_obj(self_obj, 0);
+                        if (critter_inven_obj(self_obj, INVEN_TYPE_WORN) == restock_obj) then begin
+                            if (obj_item_subtype(critter_inven_obj(self_obj, INVEN_TYPE_WORN)) == 0) then begin
+                                restock_obj := critter_inven_obj(self_obj, INVEN_TYPE_WORN);
                                 debug_msg("armour pid == " + obj_pid(restock_obj));
                                 rm_obj_from_inven(self_obj, restock_obj);
                                 add_obj_to_inven(self_obj, restock_obj);
                             end
                         end
                         else begin
-                            if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                metarule(43, self_obj);
+                            if ((critter_inven_obj(self_obj, INVEN_TYPE_LEFT_HAND) == restock_obj) or (critter_inven_obj(self_obj, INVEN_TYPE_RIGHT_HAND) == restock_obj)) then begin
+                                inven_unwield(self_obj);
                             end
                         end
                     end
@@ -683,9 +683,9 @@ begin
     end
     LVar0 := obj_carrying_pid_obj(self_obj, 22);
     if (LVar0) then begin
-        wield_obj_critter(self_obj, LVar0);
+        wield_obj(LVar0);
     end
-    if (global_var(399) bwand 64) then begin
+    if (global_var(GVAR_RND_KAGA_STATE) bwand 64) then begin
         set_local_var(4, 1);
     end
     return 0;
@@ -700,14 +700,14 @@ begin
         end
     end
     if (fixed_param == 4) then begin
-        if (get_critter_stat(self_obj, 35) <= (get_critter_stat(self_obj, 7) / 2)) then begin
+        if (get_critter_stat(self_obj, STAT_current_hp) <= (get_critter_stat(self_obj, STAT_max_hp) / 2)) then begin
             critter_set_flee_state(self_obj, 1);
         end
     end
     if (critter_is_fleeing(self_obj)) then begin
         call NodeFleeing();
         if (not(obj_on_screen(self_obj))) then begin
-            set_global_var(399, global_var(399) bwor 64);
+            set_global_var(GVAR_RND_KAGA_STATE, global_var(GVAR_RND_KAGA_STATE) bwor 64);
             destroy_object(self_obj);
         end
     end
@@ -731,22 +731,21 @@ end
 
 procedure NodeFleeing
 begin
-    float_msg(self_obj, message_str(832, random(110, 113)), 8);
+    float_msg(self_obj, mstr(random(110, 113)), FLOAT_MSG_YELLOW);
 end
 
 procedure NodeFighting
 begin
     variable LVar0 := 0;
     LVar0 := random(103, 109);
-    if ((LVar0 == 103) and not(global_var(399) bwand 64)) then begin
+    if ((LVar0 == 103) and not(global_var(GVAR_RND_KAGA_STATE) bwand 64)) then begin
         LVar0 := 104;
     end
-    float_msg(self_obj, message_str(832, LVar0), 2);
+    float_msg(self_obj, mstr(LVar0), FLOAT_MSG_RED);
     if ((LVar0 == 105) or (LVar0 == 106) or (LVar0 == 107)) then begin
         set_local_var(4, 1);
-        set_global_var(399, global_var(399) bwor 64);
+        set_global_var(GVAR_RND_KAGA_STATE, global_var(GVAR_RND_KAGA_STATE) bwor 64);
     end
     return 0;
     LVar0 := 0;
 end
-
