diff --git a/Mods/MegaMod/Original/ECMELCHR.ssl b/Mods/MegaMod/ReDefine/ECMELCHR.ssl
index e41e6e8..f8535ef 100644
--- a/Mods/MegaMod/Original/ECMELCHR.ssl
+++ b/Mods/MegaMod/ReDefine/ECMELCHR.ssl
@@ -155,14 +155,14 @@ procedure critter_p_proc
 begin
     if (obj_can_see_obj(self_obj, dude_obj)) then begin
         if ((local_var(5) bwand 2) != 0) then begin
-            attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+            attack(dude_obj);
         end
     end
     if ((tile_distance_objs(self_obj, dude_obj) < 20) and (tile_distance_objs(self_obj, dude_obj) > 10)) then begin
         if (local_var(4) == 0) then begin
             reg_anim_func(2, dude_obj);
             set_local_var(4, local_var(4) + 1);
-            float_msg(self_obj, message_str(368, 102), 8);
+            float_msg(self_obj, mstr(102), FLOAT_MSG_YELLOW);
             add_timer_event(self_obj, game_ticks(3), 1);
             reg_anim_func(2, dude_obj);
         end
@@ -175,10 +175,10 @@ begin
     end
     else begin
         if (tile_distance_objs(self_obj, dude_obj) < 10) then begin
-            float_msg(self_obj, message_str(368, 107), 8);
+            float_msg(self_obj, mstr(107), FLOAT_MSG_YELLOW);
             rm_timer_event(self_obj);
             set_local_var(5, local_var(5) bwor 2);
-            attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+            attack(dude_obj);
         end
     end
 end
@@ -190,57 +190,57 @@ end
 procedure destroy_p_proc
 begin
     variable LVar0 := 0;
-    LVar0 := create_object_sid(265, 0, 0, -1);
+    LVar0 := create_object(265, 0, 0);
     add_obj_to_inven(self_obj, LVar0);
     if (source_obj == dude_obj) then begin
-        set_global_var(4, global_var(4) + 1);
-        if (metarule(51, self_obj) == 2) then begin
+        set_global_var(GVAR_BAD_MONSTER, global_var(GVAR_BAD_MONSTER) + 1);
+        if (critter_kill_type(self_obj) == KILL_TYPE_children_kills) then begin
             if (source_obj == dude_obj) then begin
-                set_global_var(2, 0);
-                set_global_var(1, global_var(1) + 1);
-                set_global_var(0, global_var(0) - -15);
-                set_global_var(37, 0);
-                set_global_var(38, 0);
-                set_global_var(39, 0);
-                set_global_var(40, 0);
-                set_global_var(41, 0);
-                set_global_var(42, 0);
-                set_global_var(43, 0);
-                set_global_var(44, 0);
-                set_global_var(45, 0);
-                if (global_var(0) >= 1000) then begin
-                    set_global_var(37, 1);
+                set_global_var(GVAR_CHAMPION_REPUTATION, 0);
+                set_global_var(GVAR_CHILDKILLER_REPUTATION, global_var(GVAR_CHILDKILLER_REPUTATION) + 1);
+                set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) - -15);
+                set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+                set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+                set_global_var(GVAR_KARMA_DEFENDER, 0);
+                set_global_var(GVAR_KARMA_WANDERER, 0);
+                set_global_var(GVAR_KARMA_BETRAYER, 0);
+                set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+                set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+                set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+                    set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
                 end
                 else begin
-                    if (global_var(0) >= 750) then begin
-                        set_global_var(38, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
                     end
                     else begin
-                        if (global_var(0) >= 500) then begin
-                            set_global_var(39, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                            set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                         end
                         else begin
-                            if (global_var(0) >= 250) then begin
-                                set_global_var(40, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                                set_global_var(GVAR_KARMA_DEFENDER, 1);
                             end
                             else begin
-                                if (global_var(0) > -250) then begin
-                                    set_global_var(41, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                                    set_global_var(GVAR_KARMA_WANDERER, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -500) then begin
-                                        set_global_var(42, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                        set_global_var(GVAR_KARMA_BETRAYER, 1);
                                     end
                                     else begin
-                                        if (global_var(0) > -750) then begin
-                                            set_global_var(43, 1);
+                                        if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                            set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                         end
                                         else begin
-                                            if (global_var(0) > -1000) then begin
-                                                set_global_var(44, 1);
+                                            if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                                set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                             end
                                             else begin
-                                                set_global_var(45, 1);
+                                                set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                             end
                                         end
                                     end
@@ -251,58 +251,58 @@ begin
                 end
             end
         end
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(5) > (2 * global_var(4))) or (global_var(3) == 1))) then begin
-            set_global_var(3, 1);
-            set_global_var(2, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
+            set_global_var(GVAR_BERSERKER_REPUTATION, 1);
+            set_global_var(GVAR_CHAMPION_REPUTATION, 0);
         end
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(4) > (3 * global_var(5))) or (global_var(2) == 1)) and (global_var(1) == 0)) then begin
-            set_global_var(2, 1);
-            set_global_var(3, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(4) > (3 * global_var(GVAR_GOOD_MONSTER))) or (global_var(GVAR_CHAMPION_REPUTATION) == 1)) and (global_var(GVAR_CHILDKILLER_REPUTATION) == 0)) then begin
+            set_global_var(GVAR_CHAMPION_REPUTATION, 1);
+            set_global_var(GVAR_BERSERKER_REPUTATION, 0);
         end
-        set_global_var(0, global_var(0) + 5);
+        set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + 5);
         debug_msg("Player gains " + 5 + " Karma Points.");
-        set_global_var(37, 0);
-        set_global_var(38, 0);
-        set_global_var(39, 0);
-        set_global_var(40, 0);
-        set_global_var(41, 0);
-        set_global_var(42, 0);
-        set_global_var(43, 0);
-        set_global_var(44, 0);
-        set_global_var(45, 0);
-        if (global_var(0) >= 1000) then begin
-            set_global_var(37, 1);
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+        set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+        set_global_var(GVAR_KARMA_DEFENDER, 0);
+        set_global_var(GVAR_KARMA_WANDERER, 0);
+        set_global_var(GVAR_KARMA_BETRAYER, 0);
+        set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+        set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+            set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
         end
         else begin
-            if (global_var(0) >= 750) then begin
-                set_global_var(38, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
             end
             else begin
-                if (global_var(0) >= 500) then begin
-                    set_global_var(39, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                 end
                 else begin
-                    if (global_var(0) >= 250) then begin
-                        set_global_var(40, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                        set_global_var(GVAR_KARMA_DEFENDER, 1);
                     end
                     else begin
-                        if (global_var(0) > -250) then begin
-                            set_global_var(41, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                            set_global_var(GVAR_KARMA_WANDERER, 1);
                         end
                         else begin
-                            if (global_var(0) > -500) then begin
-                                set_global_var(42, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                set_global_var(GVAR_KARMA_BETRAYER, 1);
                             end
                             else begin
-                                if (global_var(0) > -750) then begin
-                                    set_global_var(43, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -1000) then begin
-                                        set_global_var(44, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                     end
                                     else begin
-                                        set_global_var(45, 1);
+                                        set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                     end
                                 end
                             end
@@ -319,21 +319,21 @@ end
 procedure look_at_p_proc
 begin
     script_overrides;
-    display_msg(message_str(368, 100));
+    display_mstr(100);
 end
 
 procedure description_p_proc
 begin
     script_overrides;
-    display_msg(message_str(368, 101));
+    display_mstr(101);
 end
 
 procedure map_enter_p_proc
 begin
     melchior_obj := self_obj;
     Only_Once := 0;
-    critter_add_trait(self_obj, 1, 6, 67);
-    critter_add_trait(self_obj, 1, 5, 47);
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 67);
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_SUPER_MUTANT);
 end
 
 procedure timed_event_p_proc
@@ -345,15 +345,15 @@ begin
         set_local_var(5, local_var(5) bwand (-1 - 32768));
         if (obj_can_see_obj(self_obj, dude_obj)) then begin
             if (local_var(4) < 5) then begin
-                float_msg(self_obj, message_str(368, 102 + local_var(4)), 8);
+                float_msg(self_obj, mstr(102 + local_var(4)), FLOAT_MSG_YELLOW);
                 set_local_var(4, local_var(4) + 1);
                 add_timer_event(self_obj, game_ticks(5), 1);
             end
             else begin
                 if (combat_is_initialized == 0) then begin
-                    float_msg(self_obj, message_str(368, 107), 8);
+                    float_msg(self_obj, mstr(107), FLOAT_MSG_YELLOW);
                     rm_timer_event(self_obj);
-                    attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+                    attack(dude_obj);
                 end
             end
         end
@@ -365,7 +365,7 @@ begin
     variable LVar0 := 0;
     variable LVar1 := 0;
     variable LVar2 := 0;
-    if (elevation(dude_obj) == elevation(self_obj)) then begin
+    if (dude_elevation == elevation(self_obj)) then begin
         if (fixed_param == 4) then begin
             if (local_var(6) <= 3) then begin
                 if (map_var(local_var(6)) <= 0) then begin
@@ -383,8 +383,8 @@ begin
                         end
                         if (((tile_contains_pid_obj(LVar2, elevation(self_obj), 33554499) != 0) == 0) and ((tile_contains_pid_obj(tile_num_in_direction(LVar2, 0, 1), elevation(self_obj), 33554499) != 0) == 0) and ((tile_contains_pid_obj(tile_num_in_direction(LVar2, 1, 1), elevation(self_obj), 33554499) != 0) == 0) and ((tile_contains_pid_obj(tile_num_in_direction(LVar2, 2, 1), elevation(self_obj), 33554499) != 0) == 0) and ((tile_contains_pid_obj(tile_num_in_direction(LVar2, 3, 1), elevation(self_obj), 33554499) != 0) == 0) and ((tile_contains_pid_obj(tile_num_in_direction(LVar2, 4, 1), elevation(self_obj), 33554499) != 0) == 0) and ((tile_contains_pid_obj(tile_num_in_direction(LVar2, 5, 1), elevation(self_obj), 33554499) != 0) == 0)) then begin
                             LVar1 := 1000 + (10 * local_var(6)) + random(0, 2);
-                            debug_msg("message: #" + LVar1 + " == " + message_str(368, LVar1));
-                            float_msg(self_obj, message_str(368, LVar1), 8);
+                            debug_msg("message: #" + LVar1 + " == " + mstr(LVar1));
+                            float_msg(self_obj, mstr(LVar1), FLOAT_MSG_YELLOW);
                             anim(self_obj, 1000, rotation_to_tile(tile_num(self_obj), (13928 * ((LVar2 % 4) <= 1)) + (14120 * ((LVar2 % 4) >= 2))));
                             anim(self_obj, 18, -1);
                             if ((4 - local_var(7)) > 1) then begin
@@ -399,7 +399,7 @@ begin
                         else begin
                             debug_msg("PORTAL BLOCKED!!!");
                         end
-                        attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+                        attack(dude_obj);
                     end
                     else begin
                         debug_msg("TOO MANY PETS OUT, NEED TO WAIT FOR ONE TO DIE!!!");
@@ -408,14 +408,14 @@ begin
                 else begin
                     if (((local_var(5) bwand 16384) != 0) == 0) then begin
                         set_local_var(5, local_var(5) bwor 16384);
-                        float_msg(self_obj, message_str(368, 1050), 8 + ((7 - 8) * has_trait(0, dude_obj, 22)));
+                        float_msg(self_obj, mstr(1050), 8 + ((7 - 8) * has_trait(TRAIT_PERK, dude_obj, PERK_empathy)));
                     end
                 end
             end
             else begin
                 if (((local_var(5) bwand 16384) != 0) == 0) then begin
                     set_local_var(5, local_var(5) bwor 16384);
-                    float_msg(self_obj, message_str(368, 1050), 8 + ((7 - 8) * has_trait(0, dude_obj, 22)));
+                    float_msg(self_obj, mstr(1050), 8 + ((7 - 8) * has_trait(TRAIT_PERK, dude_obj, PERK_empathy)));
                 end
             end
         end
@@ -444,4 +444,3 @@ end
 procedure use_skill_on_p_proc
 begin
 end
-
