diff --git a/Mods/MegaMod/Original/ECNOMSHA.ssl b/Mods/MegaMod/ReDefine/ECNOMSHA.ssl
index 83c7733..e921834 100644
--- a/Mods/MegaMod/Original/ECNOMSHA.ssl
+++ b/Mods/MegaMod/ReDefine/ECNOMSHA.ssl
@@ -177,7 +177,7 @@ procedure critter_p_proc
 begin
     if ((local_var(1) == 2) and obj_can_see_obj(self_obj, dude_obj)) then begin
         set_local_var(1, 1);
-        attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+        attack(dude_obj);
     end
 end
 
@@ -201,54 +201,54 @@ end
 procedure destroy_p_proc
 begin
     if (source_obj == dude_obj) then begin
-        set_global_var(5, global_var(5) + 1);
-        if (metarule(51, self_obj) == 2) then begin
+        set_global_var(GVAR_GOOD_MONSTER, global_var(GVAR_GOOD_MONSTER) + 1);
+        if (critter_kill_type(self_obj) == KILL_TYPE_children_kills) then begin
             if (source_obj == dude_obj) then begin
-                set_global_var(2, 0);
-                set_global_var(1, global_var(1) + 1);
-                set_global_var(0, global_var(0) - -15);
-                set_global_var(37, 0);
-                set_global_var(38, 0);
-                set_global_var(39, 0);
-                set_global_var(40, 0);
-                set_global_var(41, 0);
-                set_global_var(42, 0);
-                set_global_var(43, 0);
-                set_global_var(44, 0);
-                set_global_var(45, 0);
-                if (global_var(0) >= 1000) then begin
-                    set_global_var(37, 1);
+                set_global_var(GVAR_CHAMPION_REPUTATION, 0);
+                set_global_var(GVAR_CHILDKILLER_REPUTATION, global_var(GVAR_CHILDKILLER_REPUTATION) + 1);
+                set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) - -15);
+                set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+                set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+                set_global_var(GVAR_KARMA_DEFENDER, 0);
+                set_global_var(GVAR_KARMA_WANDERER, 0);
+                set_global_var(GVAR_KARMA_BETRAYER, 0);
+                set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+                set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+                set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+                    set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
                 end
                 else begin
-                    if (global_var(0) >= 750) then begin
-                        set_global_var(38, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
                     end
                     else begin
-                        if (global_var(0) >= 500) then begin
-                            set_global_var(39, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                            set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                         end
                         else begin
-                            if (global_var(0) >= 250) then begin
-                                set_global_var(40, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                                set_global_var(GVAR_KARMA_DEFENDER, 1);
                             end
                             else begin
-                                if (global_var(0) > -250) then begin
-                                    set_global_var(41, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                                    set_global_var(GVAR_KARMA_WANDERER, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -500) then begin
-                                        set_global_var(42, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                        set_global_var(GVAR_KARMA_BETRAYER, 1);
                                     end
                                     else begin
-                                        if (global_var(0) > -750) then begin
-                                            set_global_var(43, 1);
+                                        if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                            set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                         end
                                         else begin
-                                            if (global_var(0) > -1000) then begin
-                                                set_global_var(44, 1);
+                                            if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                                set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                             end
                                             else begin
-                                                set_global_var(45, 1);
+                                                set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                             end
                                         end
                                     end
@@ -259,58 +259,58 @@ begin
                 end
             end
         end
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(4) > (3 * global_var(5))) or (global_var(2) == 1)) and (global_var(1) == 0)) then begin
-            set_global_var(2, 1);
-            set_global_var(3, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(4) > (3 * global_var(GVAR_GOOD_MONSTER))) or (global_var(GVAR_CHAMPION_REPUTATION) == 1)) and (global_var(GVAR_CHILDKILLER_REPUTATION) == 0)) then begin
+            set_global_var(GVAR_CHAMPION_REPUTATION, 1);
+            set_global_var(GVAR_BERSERKER_REPUTATION, 0);
         end
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(5) > (2 * global_var(4))) or (global_var(3) == 1))) then begin
-            set_global_var(3, 1);
-            set_global_var(2, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
+            set_global_var(GVAR_BERSERKER_REPUTATION, 1);
+            set_global_var(GVAR_CHAMPION_REPUTATION, 0);
         end
-        set_global_var(0, global_var(0) + -10);
+        set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + -10);
         debug_msg("Player gains " + -10 + " Karma Points.");
-        set_global_var(37, 0);
-        set_global_var(38, 0);
-        set_global_var(39, 0);
-        set_global_var(40, 0);
-        set_global_var(41, 0);
-        set_global_var(42, 0);
-        set_global_var(43, 0);
-        set_global_var(44, 0);
-        set_global_var(45, 0);
-        if (global_var(0) >= 1000) then begin
-            set_global_var(37, 1);
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+        set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+        set_global_var(GVAR_KARMA_DEFENDER, 0);
+        set_global_var(GVAR_KARMA_WANDERER, 0);
+        set_global_var(GVAR_KARMA_BETRAYER, 0);
+        set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+        set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+            set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
         end
         else begin
-            if (global_var(0) >= 750) then begin
-                set_global_var(38, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
             end
             else begin
-                if (global_var(0) >= 500) then begin
-                    set_global_var(39, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                 end
                 else begin
-                    if (global_var(0) >= 250) then begin
-                        set_global_var(40, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                        set_global_var(GVAR_KARMA_DEFENDER, 1);
                     end
                     else begin
-                        if (global_var(0) > -250) then begin
-                            set_global_var(41, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                            set_global_var(GVAR_KARMA_WANDERER, 1);
                         end
                         else begin
-                            if (global_var(0) > -500) then begin
-                                set_global_var(42, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                set_global_var(GVAR_KARMA_BETRAYER, 1);
                             end
                             else begin
-                                if (global_var(0) > -750) then begin
-                                    set_global_var(43, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -1000) then begin
-                                        set_global_var(44, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                     end
                                     else begin
-                                        set_global_var(45, 1);
+                                        set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                     end
                                 end
                             end
@@ -326,17 +326,17 @@ procedure look_at_p_proc
 begin
     script_overrides;
     if (local_var(0) == 0) then begin
-        display_msg(message_str(770, 100));
+        display_mstr(100);
     end
     else begin
-        display_msg(message_str(770, 101));
+        display_mstr(101);
     end
 end
 
 procedure description_p_proc
 begin
     script_overrides;
-    display_msg(message_str(770, 102));
+    display_mstr(102);
 end
 
 procedure use_skill_on_p_proc
@@ -352,13 +352,13 @@ end
 
 procedure map_enter_p_proc
 begin
-    critter_add_trait(self_obj, 1, 6, 129);
-    critter_add_trait(self_obj, 1, 5, 24);
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 129);
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_TOUGH_CITIZEN);
     restock_amt := random(3, 5);
     if (obj_is_carrying_obj_pid(self_obj, 4) < restock_amt) then begin
         if (30 >= random(1, 100)) then begin
             if (obj_is_carrying_obj_pid(self_obj, 4) < restock_amt) then begin
-                restock_obj := create_object_sid(4, 0, 0, -1);
+                restock_obj := create_object(4, 0, 0);
                 add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 4));
             end
             else begin
@@ -371,17 +371,17 @@ begin
                         restock_obj := obj_carrying_pid_obj(self_obj, 4);
                         if (obj_type(self_obj) == 1) then begin
                             restock_obj := obj_carrying_pid_obj(self_obj, 4);
-                            if (critter_inven_obj(self_obj, 0) == restock_obj) then begin
-                                if (obj_item_subtype(critter_inven_obj(self_obj, 0)) == 0) then begin
-                                    restock_obj := critter_inven_obj(self_obj, 0);
+                            if (critter_inven_obj(self_obj, INVEN_TYPE_WORN) == restock_obj) then begin
+                                if (obj_item_subtype(critter_inven_obj(self_obj, INVEN_TYPE_WORN)) == 0) then begin
+                                    restock_obj := critter_inven_obj(self_obj, INVEN_TYPE_WORN);
                                     debug_msg("armour pid == " + obj_pid(restock_obj));
                                     rm_obj_from_inven(self_obj, restock_obj);
                                     add_obj_to_inven(self_obj, restock_obj);
                                 end
                             end
                             else begin
-                                if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                    metarule(43, self_obj);
+                                if ((critter_inven_obj(self_obj, INVEN_TYPE_LEFT_HAND) == restock_obj) or (critter_inven_obj(self_obj, INVEN_TYPE_RIGHT_HAND) == restock_obj)) then begin
+                                    inven_unwield(self_obj);
                                 end
                             end
                         end
@@ -399,7 +399,7 @@ begin
     end
     else begin
         if (obj_is_carrying_obj_pid(self_obj, 4) < restock_amt) then begin
-            restock_obj := create_object_sid(4, 0, 0, -1);
+            restock_obj := create_object(4, 0, 0);
             add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 4));
         end
         else begin
@@ -412,17 +412,17 @@ begin
                     restock_obj := obj_carrying_pid_obj(self_obj, 4);
                     if (obj_type(self_obj) == 1) then begin
                         restock_obj := obj_carrying_pid_obj(self_obj, 4);
-                        if (critter_inven_obj(self_obj, 0) == restock_obj) then begin
-                            if (obj_item_subtype(critter_inven_obj(self_obj, 0)) == 0) then begin
-                                restock_obj := critter_inven_obj(self_obj, 0);
+                        if (critter_inven_obj(self_obj, INVEN_TYPE_WORN) == restock_obj) then begin
+                            if (obj_item_subtype(critter_inven_obj(self_obj, INVEN_TYPE_WORN)) == 0) then begin
+                                restock_obj := critter_inven_obj(self_obj, INVEN_TYPE_WORN);
                                 debug_msg("armour pid == " + obj_pid(restock_obj));
                                 rm_obj_from_inven(self_obj, restock_obj);
                                 add_obj_to_inven(self_obj, restock_obj);
                             end
                         end
                         else begin
-                            if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                metarule(43, self_obj);
+                            if ((critter_inven_obj(self_obj, INVEN_TYPE_LEFT_HAND) == restock_obj) or (critter_inven_obj(self_obj, INVEN_TYPE_RIGHT_HAND) == restock_obj)) then begin
+                                inven_unwield(self_obj);
                             end
                         end
                     end
@@ -441,7 +441,7 @@ begin
     if (obj_is_carrying_obj_pid(self_obj, 589) < restock_amt) then begin
         if (40 >= random(1, 100)) then begin
             if (obj_is_carrying_obj_pid(self_obj, 589) < restock_amt) then begin
-                restock_obj := create_object_sid(589, 0, 0, -1);
+                restock_obj := create_object(589, 0, 0);
                 add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 589));
             end
             else begin
@@ -454,17 +454,17 @@ begin
                         restock_obj := obj_carrying_pid_obj(self_obj, 589);
                         if (obj_type(self_obj) == 1) then begin
                             restock_obj := obj_carrying_pid_obj(self_obj, 589);
-                            if (critter_inven_obj(self_obj, 0) == restock_obj) then begin
-                                if (obj_item_subtype(critter_inven_obj(self_obj, 0)) == 0) then begin
-                                    restock_obj := critter_inven_obj(self_obj, 0);
+                            if (critter_inven_obj(self_obj, INVEN_TYPE_WORN) == restock_obj) then begin
+                                if (obj_item_subtype(critter_inven_obj(self_obj, INVEN_TYPE_WORN)) == 0) then begin
+                                    restock_obj := critter_inven_obj(self_obj, INVEN_TYPE_WORN);
                                     debug_msg("armour pid == " + obj_pid(restock_obj));
                                     rm_obj_from_inven(self_obj, restock_obj);
                                     add_obj_to_inven(self_obj, restock_obj);
                                 end
                             end
                             else begin
-                                if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                    metarule(43, self_obj);
+                                if ((critter_inven_obj(self_obj, INVEN_TYPE_LEFT_HAND) == restock_obj) or (critter_inven_obj(self_obj, INVEN_TYPE_RIGHT_HAND) == restock_obj)) then begin
+                                    inven_unwield(self_obj);
                                 end
                             end
                         end
@@ -482,7 +482,7 @@ begin
     end
     else begin
         if (obj_is_carrying_obj_pid(self_obj, 589) < restock_amt) then begin
-            restock_obj := create_object_sid(589, 0, 0, -1);
+            restock_obj := create_object(589, 0, 0);
             add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 589));
         end
         else begin
@@ -495,17 +495,17 @@ begin
                     restock_obj := obj_carrying_pid_obj(self_obj, 589);
                     if (obj_type(self_obj) == 1) then begin
                         restock_obj := obj_carrying_pid_obj(self_obj, 589);
-                        if (critter_inven_obj(self_obj, 0) == restock_obj) then begin
-                            if (obj_item_subtype(critter_inven_obj(self_obj, 0)) == 0) then begin
-                                restock_obj := critter_inven_obj(self_obj, 0);
+                        if (critter_inven_obj(self_obj, INVEN_TYPE_WORN) == restock_obj) then begin
+                            if (obj_item_subtype(critter_inven_obj(self_obj, INVEN_TYPE_WORN)) == 0) then begin
+                                restock_obj := critter_inven_obj(self_obj, INVEN_TYPE_WORN);
                                 debug_msg("armour pid == " + obj_pid(restock_obj));
                                 rm_obj_from_inven(self_obj, restock_obj);
                                 add_obj_to_inven(self_obj, restock_obj);
                             end
                         end
                         else begin
-                            if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                metarule(43, self_obj);
+                            if ((critter_inven_obj(self_obj, INVEN_TYPE_LEFT_HAND) == restock_obj) or (critter_inven_obj(self_obj, INVEN_TYPE_RIGHT_HAND) == restock_obj)) then begin
+                                inven_unwield(self_obj);
                             end
                         end
                     end
@@ -524,7 +524,7 @@ begin
     if (obj_is_carrying_obj_pid(self_obj, 71) < restock_amt) then begin
         if (50 >= random(1, 100)) then begin
             if (obj_is_carrying_obj_pid(self_obj, 71) < restock_amt) then begin
-                restock_obj := create_object_sid(71, 0, 0, -1);
+                restock_obj := create_object(71, 0, 0);
                 add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 71));
             end
             else begin
@@ -537,17 +537,17 @@ begin
                         restock_obj := obj_carrying_pid_obj(self_obj, 71);
                         if (obj_type(self_obj) == 1) then begin
                             restock_obj := obj_carrying_pid_obj(self_obj, 71);
-                            if (critter_inven_obj(self_obj, 0) == restock_obj) then begin
-                                if (obj_item_subtype(critter_inven_obj(self_obj, 0)) == 0) then begin
-                                    restock_obj := critter_inven_obj(self_obj, 0);
+                            if (critter_inven_obj(self_obj, INVEN_TYPE_WORN) == restock_obj) then begin
+                                if (obj_item_subtype(critter_inven_obj(self_obj, INVEN_TYPE_WORN)) == 0) then begin
+                                    restock_obj := critter_inven_obj(self_obj, INVEN_TYPE_WORN);
                                     debug_msg("armour pid == " + obj_pid(restock_obj));
                                     rm_obj_from_inven(self_obj, restock_obj);
                                     add_obj_to_inven(self_obj, restock_obj);
                                 end
                             end
                             else begin
-                                if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                    metarule(43, self_obj);
+                                if ((critter_inven_obj(self_obj, INVEN_TYPE_LEFT_HAND) == restock_obj) or (critter_inven_obj(self_obj, INVEN_TYPE_RIGHT_HAND) == restock_obj)) then begin
+                                    inven_unwield(self_obj);
                                 end
                             end
                         end
@@ -565,7 +565,7 @@ begin
     end
     else begin
         if (obj_is_carrying_obj_pid(self_obj, 71) < restock_amt) then begin
-            restock_obj := create_object_sid(71, 0, 0, -1);
+            restock_obj := create_object(71, 0, 0);
             add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 71));
         end
         else begin
@@ -578,17 +578,17 @@ begin
                     restock_obj := obj_carrying_pid_obj(self_obj, 71);
                     if (obj_type(self_obj) == 1) then begin
                         restock_obj := obj_carrying_pid_obj(self_obj, 71);
-                        if (critter_inven_obj(self_obj, 0) == restock_obj) then begin
-                            if (obj_item_subtype(critter_inven_obj(self_obj, 0)) == 0) then begin
-                                restock_obj := critter_inven_obj(self_obj, 0);
+                        if (critter_inven_obj(self_obj, INVEN_TYPE_WORN) == restock_obj) then begin
+                            if (obj_item_subtype(critter_inven_obj(self_obj, INVEN_TYPE_WORN)) == 0) then begin
+                                restock_obj := critter_inven_obj(self_obj, INVEN_TYPE_WORN);
                                 debug_msg("armour pid == " + obj_pid(restock_obj));
                                 rm_obj_from_inven(self_obj, restock_obj);
                                 add_obj_to_inven(self_obj, restock_obj);
                             end
                         end
                         else begin
-                            if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                metarule(43, self_obj);
+                            if ((critter_inven_obj(self_obj, INVEN_TYPE_LEFT_HAND) == restock_obj) or (critter_inven_obj(self_obj, INVEN_TYPE_RIGHT_HAND) == restock_obj)) then begin
+                                inven_unwield(self_obj);
                             end
                         end
                     end
@@ -607,7 +607,7 @@ begin
     if (obj_is_carrying_obj_pid(self_obj, 276) < restock_amt) then begin
         if (20 >= random(1, 100)) then begin
             if (obj_is_carrying_obj_pid(self_obj, 276) < restock_amt) then begin
-                restock_obj := create_object_sid(276, 0, 0, -1);
+                restock_obj := create_object(276, 0, 0);
                 add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 276));
             end
             else begin
@@ -620,17 +620,17 @@ begin
                         restock_obj := obj_carrying_pid_obj(self_obj, 276);
                         if (obj_type(self_obj) == 1) then begin
                             restock_obj := obj_carrying_pid_obj(self_obj, 276);
-                            if (critter_inven_obj(self_obj, 0) == restock_obj) then begin
-                                if (obj_item_subtype(critter_inven_obj(self_obj, 0)) == 0) then begin
-                                    restock_obj := critter_inven_obj(self_obj, 0);
+                            if (critter_inven_obj(self_obj, INVEN_TYPE_WORN) == restock_obj) then begin
+                                if (obj_item_subtype(critter_inven_obj(self_obj, INVEN_TYPE_WORN)) == 0) then begin
+                                    restock_obj := critter_inven_obj(self_obj, INVEN_TYPE_WORN);
                                     debug_msg("armour pid == " + obj_pid(restock_obj));
                                     rm_obj_from_inven(self_obj, restock_obj);
                                     add_obj_to_inven(self_obj, restock_obj);
                                 end
                             end
                             else begin
-                                if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                    metarule(43, self_obj);
+                                if ((critter_inven_obj(self_obj, INVEN_TYPE_LEFT_HAND) == restock_obj) or (critter_inven_obj(self_obj, INVEN_TYPE_RIGHT_HAND) == restock_obj)) then begin
+                                    inven_unwield(self_obj);
                                 end
                             end
                         end
@@ -648,7 +648,7 @@ begin
     end
     else begin
         if (obj_is_carrying_obj_pid(self_obj, 276) < restock_amt) then begin
-            restock_obj := create_object_sid(276, 0, 0, -1);
+            restock_obj := create_object(276, 0, 0);
             add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 276));
         end
         else begin
@@ -661,17 +661,17 @@ begin
                     restock_obj := obj_carrying_pid_obj(self_obj, 276);
                     if (obj_type(self_obj) == 1) then begin
                         restock_obj := obj_carrying_pid_obj(self_obj, 276);
-                        if (critter_inven_obj(self_obj, 0) == restock_obj) then begin
-                            if (obj_item_subtype(critter_inven_obj(self_obj, 0)) == 0) then begin
-                                restock_obj := critter_inven_obj(self_obj, 0);
+                        if (critter_inven_obj(self_obj, INVEN_TYPE_WORN) == restock_obj) then begin
+                            if (obj_item_subtype(critter_inven_obj(self_obj, INVEN_TYPE_WORN)) == 0) then begin
+                                restock_obj := critter_inven_obj(self_obj, INVEN_TYPE_WORN);
                                 debug_msg("armour pid == " + obj_pid(restock_obj));
                                 rm_obj_from_inven(self_obj, restock_obj);
                                 add_obj_to_inven(self_obj, restock_obj);
                             end
                         end
                         else begin
-                            if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                metarule(43, self_obj);
+                            if ((critter_inven_obj(self_obj, INVEN_TYPE_LEFT_HAND) == restock_obj) or (critter_inven_obj(self_obj, INVEN_TYPE_RIGHT_HAND) == restock_obj)) then begin
+                                inven_unwield(self_obj);
                             end
                         end
                     end
@@ -722,40 +722,40 @@ procedure Node001
 begin
     n1count := n1count + 1;
     if (n1count == 1) then begin
-        gsay_reply(770, 103);
+        Reply(103);
     end
     else begin
-        gsay_reply(770, 104);
+        Reply(104);
     end
-    if ((get_critter_stat(dude_obj, 35) != get_critter_stat(dude_obj, 7)) or ((critter_state(dude_obj) bwand 4) or (critter_state(dude_obj) bwand 8) or (critter_state(dude_obj) bwand 16) or (critter_state(dude_obj) bwand 32)) or get_poison(dude_obj)) then begin
-        giq_option(4, 770, 105, Node002, 50);
+    if ((get_critter_stat(dude_obj, STAT_current_hp) != get_critter_stat(dude_obj, STAT_max_hp)) or ((critter_state(dude_obj) bwand 4) or (critter_state(dude_obj) bwand 8) or (critter_state(dude_obj) bwand 16) or (critter_state(dude_obj) bwand 32)) or get_poison(dude_obj)) then begin
+        giq_option(4, NAME, 105, Node002, NEUTRAL_REACTION);
     end
-    giq_option(4, 770, 106, Node008, 50);
-    giq_option(4, 770, 107, Node999, 50);
-    giq_option(-3, 770, 108, Node011, 50);
+    giq_option(4, NAME, 106, Node008, NEUTRAL_REACTION);
+    giq_option(4, NAME, 107, Node999, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 108, Node011, NEUTRAL_REACTION);
 end
 
 procedure Node002
 begin
-    gsay_reply(770, 109);
-    giq_option(4, 770, 110, Node003, 50);
+    Reply(109);
+    giq_option(4, NAME, 110, Node003, NEUTRAL_REACTION);
 end
 
 procedure Node003
 begin
     variable LVar0 := 0;
-    LVar0 := message_str(770, 111);
+    LVar0 := mstr(111);
     if ((critter_state(dude_obj) bwand 4) or (critter_state(dude_obj) bwand 8) or (critter_state(dude_obj) bwand 16) or (critter_state(dude_obj) bwand 32)) then begin
-        LVar0 := LVar0 + (" " + message_str(770, 139));
+        LVar0 := LVar0 + (" " + mstr(139));
     end
-    if (get_critter_stat(dude_obj, 35) < get_critter_stat(dude_obj, 7)) then begin
-        LVar0 := LVar0 + (" " + message_str(770, 137));
+    if (get_critter_stat(dude_obj, STAT_current_hp) < get_critter_stat(dude_obj, STAT_max_hp)) then begin
+        LVar0 := LVar0 + (" " + mstr(137));
     end
     if (get_poison(dude_obj)) then begin
-        LVar0 := LVar0 + (" " + message_str(770, 136));
+        LVar0 := LVar0 + (" " + mstr(136));
     end
-    gsay_reply(770, LVar0);
-    giq_option(4, 770, 112, Node004, 50);
+    Reply(LVar0);
+    giq_option(4, NAME, 112, Node004, NEUTRAL_REACTION);
     return 0;
     LVar0 := 0;
 end
@@ -763,87 +763,87 @@ end
 procedure Node004
 begin
     variable LVar0 := 0;
-    LVar0 := message_str(770, 113);
+    LVar0 := mstr(113);
     if ((critter_state(dude_obj) bwand 4) or (critter_state(dude_obj) bwand 8) or (critter_state(dude_obj) bwand 16) or (critter_state(dude_obj) bwand 32)) then begin
-        LVar0 := LVar0 + (" " + message_str(770, 141) + cripple_price);
+        LVar0 := LVar0 + (" " + mstr(141) + cripple_price);
     end
-    if (get_critter_stat(dude_obj, 35) < get_critter_stat(dude_obj, 7)) then begin
-        LVar0 := LVar0 + (" " + message_str(770, 142) + healing_price);
+    if (get_critter_stat(dude_obj, STAT_current_hp) < get_critter_stat(dude_obj, STAT_max_hp)) then begin
+        LVar0 := LVar0 + (" " + mstr(142) + healing_price);
     end
     if (get_poison(dude_obj)) then begin
-        LVar0 := LVar0 + (" " + message_str(770, 140) + poison_price);
+        LVar0 := LVar0 + (" " + mstr(140) + poison_price);
     end
-    gsay_reply(770, LVar0);
+    Reply(LVar0);
     if (get_poison(dude_obj)) then begin
-        giq_option(4, 770, 114, NodeCurePoison, 50);
+        giq_option(4, NAME, 114, NodeCurePoison, NEUTRAL_REACTION);
     end
-    if (get_critter_stat(dude_obj, 35) < get_critter_stat(dude_obj, 7)) then begin
-        giq_option(4, 770, 115, NodeCureHits, 50);
+    if (get_critter_stat(dude_obj, STAT_current_hp) < get_critter_stat(dude_obj, STAT_max_hp)) then begin
+        giq_option(4, NAME, 115, NodeCureHits, NEUTRAL_REACTION);
     end
     if ((critter_state(dude_obj) bwand 4) or (critter_state(dude_obj) bwand 8) or (critter_state(dude_obj) bwand 16) or (critter_state(dude_obj) bwand 32)) then begin
-        giq_option(4, 770, 116, NodeCureCripple, 50);
+        giq_option(4, NAME, 116, NodeCureCripple, NEUTRAL_REACTION);
     end
-    giq_option(4, 770, 117, Node999, 50);
+    giq_option(4, NAME, 117, Node999, NEUTRAL_REACTION);
     return 0;
     LVar0 := 0;
 end
 
 procedure Node005
 begin
-    gsay_reply(770, 118);
-    if (get_poison(dude_obj) or (get_critter_stat(dude_obj, 35) < get_critter_stat(dude_obj, 7)) or ((critter_state(dude_obj) bwand 4) or (critter_state(dude_obj) bwand 8) or (critter_state(dude_obj) bwand 16) or (critter_state(dude_obj) bwand 32))) then begin
-        giq_option(4, 770, 121, Node004, 50);
+    Reply(118);
+    if (get_poison(dude_obj) or (get_critter_stat(dude_obj, STAT_current_hp) < get_critter_stat(dude_obj, STAT_max_hp)) or ((critter_state(dude_obj) bwand 4) or (critter_state(dude_obj) bwand 8) or (critter_state(dude_obj) bwand 16) or (critter_state(dude_obj) bwand 32))) then begin
+        giq_option(4, NAME, 121, Node004, NEUTRAL_REACTION);
     end
-    giq_option(4, 770, 122, Node999, 50);
+    giq_option(4, NAME, 122, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node006
 begin
-    gsay_reply(770, 123);
-    giq_option(4, 770, 124, Node004, 50);
+    Reply(123);
+    giq_option(4, NAME, 124, Node004, NEUTRAL_REACTION);
 end
 
 procedure Node007
 begin
-    gsay_reply(770, 125);
-    giq_option(4, 770, 126, Node004, 50);
+    Reply(125);
+    giq_option(4, NAME, 126, Node004, NEUTRAL_REACTION);
 end
 
 procedure Node008
 begin
-    gsay_reply(770, message_str(770, 127) + " " + powder_price + " " + message_str(770, 150));
+    Reply(mstr(127) + " " + powder_price + " " + mstr(150));
     if (obj_carrying_pid_obj(dude_obj, 272) and obj_carrying_pid_obj(dude_obj, 271)) then begin
-        giq_option(4, 770, 128, NodeCombine, 50);
+        giq_option(4, NAME, 128, NodeCombine, NEUTRAL_REACTION);
     end
-    giq_option(4, 770, 129, Node999, 50);
+    giq_option(4, NAME, 129, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node009
 begin
-    gsay_reply(770, 130);
-    giq_option(4, 770, 131, Node999, 50);
+    Reply(130);
+    giq_option(4, NAME, 131, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node010
 begin
-    gsay_reply(770, 132);
-    giq_option(4, 770, 133, Node999, 50);
+    Reply(132);
+    giq_option(4, NAME, 133, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node011
 begin
-    gsay_reply(770, 134);
-    giq_option(4, 770, 135, Node999, 50);
+    Reply(134);
+    giq_option(4, NAME, 135, Node999, NEUTRAL_REACTION);
 end
 
 procedure NodeFleeing
 begin
-    float_msg(self_obj, message_str(770, random(120, 122)), 8);
+    float_msg(self_obj, mstr(random(120, 122)), FLOAT_MSG_YELLOW);
 end
 
 procedure NodeFighting
 begin
-    float_msg(self_obj, message_str(770, random(116, 119)), 2);
+    float_msg(self_obj, mstr(random(116, 119)), FLOAT_MSG_RED);
 end
 
 procedure NodeCurePoison
@@ -872,7 +872,7 @@ begin
     if (item_caps_total(dude_obj) >= healing_price) then begin
         item_caps_adjust(dude_obj, -healing_price);
         item_caps_adjust(self_obj, healing_price);
-        critter_heal(dude_obj, get_critter_stat(dude_obj, 7) - get_critter_stat(dude_obj, 35));
+        critter_heal(dude_obj, get_critter_stat(dude_obj, STAT_max_hp) - get_critter_stat(dude_obj, STAT_current_hp));
         gfade_out(90);
         game_time_advance(random(60 * 10 * 30, 60 * (60 * 10) * 3));
         gfade_in(90);
@@ -909,7 +909,7 @@ begin
     item_caps_adjust(self_obj, powder_price);
     rm_obj_from_inven(dude_obj, LVar0);
     rm_obj_from_inven(dude_obj, LVar1);
-    add_mult_objs_to_inven(dude_obj, create_object_sid(273, tile_num(self_obj), 0, -1), 1);
+    add_mult_objs_to_inven(dude_obj, create_object(273, tile_num(self_obj), 0), 1);
     gfade_out(90);
     game_time_advance(random(60 * 10 * 30, 60 * (60 * 10) * 2));
     gfade_in(90);
@@ -918,4 +918,3 @@ begin
     LVar0 := 0;
     LVar1 := 0;
 end
-
