diff --git a/Mods/MegaMod/Original/ECNOMCHF.ssl b/Mods/MegaMod/ReDefine/ECNOMCHF.ssl
index a9c0f9e..959e332 100644
--- a/Mods/MegaMod/Original/ECNOMCHF.ssl
+++ b/Mods/MegaMod/ReDefine/ECNOMCHF.ssl
@@ -50,7 +50,7 @@ procedure critter_p_proc
 begin
     if ((local_var(5) == 2) and obj_can_see_obj(self_obj, dude_obj)) then begin
         set_local_var(5, 1);
-        attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+        attack(dude_obj);
     end
 end
 
@@ -72,66 +72,66 @@ begin
         set_local_var(4, 1);
     end
     else begin
-        float_msg(self_obj, message_str(765, random(116, 120)), 2);
+        float_msg(self_obj, mstr(random(116, 120)), FLOAT_MSG_RED);
     end
 end
 
 procedure destroy_p_proc
 begin
     if (source_obj == dude_obj) then begin
-        set_global_var(5, global_var(5) + 1);
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(4) > (3 * global_var(5))) or (global_var(2) == 1)) and (global_var(1) == 0)) then begin
-            set_global_var(2, 1);
-            set_global_var(3, 0);
+        set_global_var(GVAR_GOOD_MONSTER, global_var(GVAR_GOOD_MONSTER) + 1);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(4) > (3 * global_var(GVAR_GOOD_MONSTER))) or (global_var(GVAR_CHAMPION_REPUTATION) == 1)) and (global_var(GVAR_CHILDKILLER_REPUTATION) == 0)) then begin
+            set_global_var(GVAR_CHAMPION_REPUTATION, 1);
+            set_global_var(GVAR_BERSERKER_REPUTATION, 0);
         end
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(5) > (2 * global_var(4))) or (global_var(3) == 1))) then begin
-            set_global_var(3, 1);
-            set_global_var(2, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
+            set_global_var(GVAR_BERSERKER_REPUTATION, 1);
+            set_global_var(GVAR_CHAMPION_REPUTATION, 0);
         end
-        set_global_var(0, global_var(0) + -10);
+        set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + -10);
         debug_msg("Player gains " + -10 + " Karma Points.");
-        set_global_var(37, 0);
-        set_global_var(38, 0);
-        set_global_var(39, 0);
-        set_global_var(40, 0);
-        set_global_var(41, 0);
-        set_global_var(42, 0);
-        set_global_var(43, 0);
-        set_global_var(44, 0);
-        set_global_var(45, 0);
-        if (global_var(0) >= 1000) then begin
-            set_global_var(37, 1);
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+        set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+        set_global_var(GVAR_KARMA_DEFENDER, 0);
+        set_global_var(GVAR_KARMA_WANDERER, 0);
+        set_global_var(GVAR_KARMA_BETRAYER, 0);
+        set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+        set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+            set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
         end
         else begin
-            if (global_var(0) >= 750) then begin
-                set_global_var(38, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
             end
             else begin
-                if (global_var(0) >= 500) then begin
-                    set_global_var(39, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                 end
                 else begin
-                    if (global_var(0) >= 250) then begin
-                        set_global_var(40, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                        set_global_var(GVAR_KARMA_DEFENDER, 1);
                     end
                     else begin
-                        if (global_var(0) > -250) then begin
-                            set_global_var(41, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                            set_global_var(GVAR_KARMA_WANDERER, 1);
                         end
                         else begin
-                            if (global_var(0) > -500) then begin
-                                set_global_var(42, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                set_global_var(GVAR_KARMA_BETRAYER, 1);
                             end
                             else begin
-                                if (global_var(0) > -750) then begin
-                                    set_global_var(43, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -1000) then begin
-                                        set_global_var(44, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                     end
                                     else begin
-                                        set_global_var(45, 1);
+                                        set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                     end
                                 end
                             end
@@ -147,17 +147,17 @@ procedure look_at_p_proc
 begin
     script_overrides;
     if (local_var(4) == 0) then begin
-        display_msg(message_str(765, 100));
+        display_mstr(100);
     end
     else begin
-        display_msg(message_str(765, 101));
+        display_mstr(101);
     end
 end
 
 procedure description_p_proc
 begin
     script_overrides;
-    display_msg(message_str(765, 102));
+    display_mstr(102);
 end
 
 procedure use_skill_on_p_proc
@@ -173,48 +173,48 @@ end
 
 procedure map_enter_p_proc
 begin
-    if (message_str(765, 100) != 0) then begin
-        if (cur_map_index == 122) then begin
+    if (mstr(100) != 0) then begin
+        if (CUR_MAP_MOUNTN4) then begin
             move_to(self_obj, 17278, 0);
         end
         else begin
-            if (cur_map_index == 123) then begin
+            if (CUR_MAP_MOUNTN5) then begin
                 move_to(self_obj, 17891, 0);
             end
             else begin
-                if (cur_map_index == 81) then begin
+                if (CUR_MAP_DESERT4) then begin
                     move_to(self_obj, 18897, 0);
                 end
                 else begin
-                    if (cur_map_index == 82) then begin
+                    if (CUR_MAP_DESERT5) then begin
                         move_to(self_obj, 19101, 0);
                     end
                     else begin
-                        if (cur_map_index == 114) then begin
+                        if (CUR_MAP_DESERT9) then begin
                             move_to(self_obj, 18293, 0);
                         end
                         else begin
-                            if (cur_map_index == 89) then begin
+                            if (CUR_MAP_COAST7) then begin
                                 move_to(self_obj, 20702, 0);
                             end
                             else begin
-                                if (cur_map_index == 90) then begin
+                                if (CUR_MAP_COAST8) then begin
                                     move_to(self_obj, 18306, 0);
                                 end
                                 else begin
-                                    if (cur_map_index == 91) then begin
+                                    if (CUR_MAP_COAST9) then begin
                                         move_to(self_obj, 17490, 0);
                                     end
                                     else begin
-                                        if (cur_map_index == 110) then begin
+                                        if (CUR_MAP_COAST10) then begin
                                             move_to(self_obj, 16706, 0);
                                         end
                                         else begin
-                                            if (cur_map_index == 111) then begin
+                                            if (CUR_MAP_COAST11) then begin
                                                 move_to(self_obj, 18293, 0);
                                             end
                                             else begin
-                                                if (cur_map_index == 112) then begin
+                                                if (CUR_MAP_07DESERT) then begin
                                                     move_to(self_obj, 19708, 0);
                                                 end
                                             end
@@ -228,14 +228,14 @@ begin
             end
         end
     end
-    critter_add_trait(self_obj, 1, 6, 129);
-    critter_add_trait(self_obj, 1, 5, 24);
-    if (message_str(765, 101) != 0) then begin
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 129);
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_TOUGH_CITIZEN);
+    if (mstr(101) != 0) then begin
         restock_amt := random(2, 5);
         if (obj_is_carrying_obj_pid(self_obj, 7) < restock_amt) then begin
             if (80 >= random(1, 100)) then begin
                 if (obj_is_carrying_obj_pid(self_obj, 7) < restock_amt) then begin
-                    restock_obj := create_object_sid(7, 0, 0, -1);
+                    restock_obj := create_object(7, 0, 0);
                     add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 7));
                 end
                 else begin
@@ -248,17 +248,17 @@ begin
                             restock_obj := obj_carrying_pid_obj(self_obj, 7);
                             if (obj_type(self_obj) == 1) then begin
                                 restock_obj := obj_carrying_pid_obj(self_obj, 7);
-                                if (critter_inven_obj(self_obj, 0) == restock_obj) then begin
-                                    if (obj_item_subtype(critter_inven_obj(self_obj, 0)) == 0) then begin
-                                        restock_obj := critter_inven_obj(self_obj, 0);
+                                if (critter_inven_obj(self_obj, INVEN_TYPE_WORN) == restock_obj) then begin
+                                    if (obj_item_subtype(critter_inven_obj(self_obj, INVEN_TYPE_WORN)) == 0) then begin
+                                        restock_obj := critter_inven_obj(self_obj, INVEN_TYPE_WORN);
                                         debug_msg("armour pid == " + obj_pid(restock_obj));
                                         rm_obj_from_inven(self_obj, restock_obj);
                                         add_obj_to_inven(self_obj, restock_obj);
                                     end
                                 end
                                 else begin
-                                    if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                        metarule(43, self_obj);
+                                    if ((critter_inven_obj(self_obj, INVEN_TYPE_LEFT_HAND) == restock_obj) or (critter_inven_obj(self_obj, INVEN_TYPE_RIGHT_HAND) == restock_obj)) then begin
+                                        inven_unwield(self_obj);
                                     end
                                 end
                             end
@@ -284,17 +284,17 @@ begin
                     restock_obj := obj_carrying_pid_obj(self_obj, 7);
                     if (obj_type(self_obj) == 1) then begin
                         restock_obj := obj_carrying_pid_obj(self_obj, 7);
-                        if (critter_inven_obj(self_obj, 0) == restock_obj) then begin
-                            if (obj_item_subtype(critter_inven_obj(self_obj, 0)) == 0) then begin
-                                restock_obj := critter_inven_obj(self_obj, 0);
+                        if (critter_inven_obj(self_obj, INVEN_TYPE_WORN) == restock_obj) then begin
+                            if (obj_item_subtype(critter_inven_obj(self_obj, INVEN_TYPE_WORN)) == 0) then begin
+                                restock_obj := critter_inven_obj(self_obj, INVEN_TYPE_WORN);
                                 debug_msg("armour pid == " + obj_pid(restock_obj));
                                 rm_obj_from_inven(self_obj, restock_obj);
                                 add_obj_to_inven(self_obj, restock_obj);
                             end
                         end
                         else begin
-                            if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                metarule(43, self_obj);
+                            if ((critter_inven_obj(self_obj, INVEN_TYPE_LEFT_HAND) == restock_obj) or (critter_inven_obj(self_obj, INVEN_TYPE_RIGHT_HAND) == restock_obj)) then begin
+                                inven_unwield(self_obj);
                             end
                         end
                     end
@@ -309,12 +309,12 @@ begin
             end
         end
     end
-    if (message_str(765, 102) != 0) then begin
+    if (mstr(102) != 0) then begin
         restock_amt := random(5, 7);
         if (obj_is_carrying_obj_pid(self_obj, 5) < restock_amt) then begin
             if (80 >= random(1, 100)) then begin
                 if (obj_is_carrying_obj_pid(self_obj, 5) < restock_amt) then begin
-                    restock_obj := create_object_sid(5, 0, 0, -1);
+                    restock_obj := create_object(5, 0, 0);
                     add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 5));
                 end
                 else begin
@@ -327,17 +327,17 @@ begin
                             restock_obj := obj_carrying_pid_obj(self_obj, 5);
                             if (obj_type(self_obj) == 1) then begin
                                 restock_obj := obj_carrying_pid_obj(self_obj, 5);
-                                if (critter_inven_obj(self_obj, 0) == restock_obj) then begin
-                                    if (obj_item_subtype(critter_inven_obj(self_obj, 0)) == 0) then begin
-                                        restock_obj := critter_inven_obj(self_obj, 0);
+                                if (critter_inven_obj(self_obj, INVEN_TYPE_WORN) == restock_obj) then begin
+                                    if (obj_item_subtype(critter_inven_obj(self_obj, INVEN_TYPE_WORN)) == 0) then begin
+                                        restock_obj := critter_inven_obj(self_obj, INVEN_TYPE_WORN);
                                         debug_msg("armour pid == " + obj_pid(restock_obj));
                                         rm_obj_from_inven(self_obj, restock_obj);
                                         add_obj_to_inven(self_obj, restock_obj);
                                     end
                                 end
                                 else begin
-                                    if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                        metarule(43, self_obj);
+                                    if ((critter_inven_obj(self_obj, INVEN_TYPE_LEFT_HAND) == restock_obj) or (critter_inven_obj(self_obj, INVEN_TYPE_RIGHT_HAND) == restock_obj)) then begin
+                                        inven_unwield(self_obj);
                                     end
                                 end
                             end
@@ -363,17 +363,17 @@ begin
                     restock_obj := obj_carrying_pid_obj(self_obj, 5);
                     if (obj_type(self_obj) == 1) then begin
                         restock_obj := obj_carrying_pid_obj(self_obj, 5);
-                        if (critter_inven_obj(self_obj, 0) == restock_obj) then begin
-                            if (obj_item_subtype(critter_inven_obj(self_obj, 0)) == 0) then begin
-                                restock_obj := critter_inven_obj(self_obj, 0);
+                        if (critter_inven_obj(self_obj, INVEN_TYPE_WORN) == restock_obj) then begin
+                            if (obj_item_subtype(critter_inven_obj(self_obj, INVEN_TYPE_WORN)) == 0) then begin
+                                restock_obj := critter_inven_obj(self_obj, INVEN_TYPE_WORN);
                                 debug_msg("armour pid == " + obj_pid(restock_obj));
                                 rm_obj_from_inven(self_obj, restock_obj);
                                 add_obj_to_inven(self_obj, restock_obj);
                             end
                         end
                         else begin
-                            if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                metarule(43, self_obj);
+                            if ((critter_inven_obj(self_obj, INVEN_TYPE_LEFT_HAND) == restock_obj) or (critter_inven_obj(self_obj, INVEN_TYPE_RIGHT_HAND) == restock_obj)) then begin
+                                inven_unwield(self_obj);
                             end
                         end
                     end
@@ -388,12 +388,12 @@ begin
             end
         end
     end
-    if (message_str(765, 103) != 0) then begin
+    if (mstr(103) != 0) then begin
         restock_amt := random(3, 5);
         if (obj_is_carrying_obj_pid(self_obj, 4) < restock_amt) then begin
             if (80 >= random(1, 100)) then begin
                 if (obj_is_carrying_obj_pid(self_obj, 4) < restock_amt) then begin
-                    restock_obj := create_object_sid(4, 0, 0, -1);
+                    restock_obj := create_object(4, 0, 0);
                     add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 4));
                 end
                 else begin
@@ -406,17 +406,17 @@ begin
                             restock_obj := obj_carrying_pid_obj(self_obj, 4);
                             if (obj_type(self_obj) == 1) then begin
                                 restock_obj := obj_carrying_pid_obj(self_obj, 4);
-                                if (critter_inven_obj(self_obj, 0) == restock_obj) then begin
-                                    if (obj_item_subtype(critter_inven_obj(self_obj, 0)) == 0) then begin
-                                        restock_obj := critter_inven_obj(self_obj, 0);
+                                if (critter_inven_obj(self_obj, INVEN_TYPE_WORN) == restock_obj) then begin
+                                    if (obj_item_subtype(critter_inven_obj(self_obj, INVEN_TYPE_WORN)) == 0) then begin
+                                        restock_obj := critter_inven_obj(self_obj, INVEN_TYPE_WORN);
                                         debug_msg("armour pid == " + obj_pid(restock_obj));
                                         rm_obj_from_inven(self_obj, restock_obj);
                                         add_obj_to_inven(self_obj, restock_obj);
                                     end
                                 end
                                 else begin
-                                    if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                        metarule(43, self_obj);
+                                    if ((critter_inven_obj(self_obj, INVEN_TYPE_LEFT_HAND) == restock_obj) or (critter_inven_obj(self_obj, INVEN_TYPE_RIGHT_HAND) == restock_obj)) then begin
+                                        inven_unwield(self_obj);
                                     end
                                 end
                             end
@@ -442,17 +442,17 @@ begin
                     restock_obj := obj_carrying_pid_obj(self_obj, 4);
                     if (obj_type(self_obj) == 1) then begin
                         restock_obj := obj_carrying_pid_obj(self_obj, 4);
-                        if (critter_inven_obj(self_obj, 0) == restock_obj) then begin
-                            if (obj_item_subtype(critter_inven_obj(self_obj, 0)) == 0) then begin
-                                restock_obj := critter_inven_obj(self_obj, 0);
+                        if (critter_inven_obj(self_obj, INVEN_TYPE_WORN) == restock_obj) then begin
+                            if (obj_item_subtype(critter_inven_obj(self_obj, INVEN_TYPE_WORN)) == 0) then begin
+                                restock_obj := critter_inven_obj(self_obj, INVEN_TYPE_WORN);
                                 debug_msg("armour pid == " + obj_pid(restock_obj));
                                 rm_obj_from_inven(self_obj, restock_obj);
                                 add_obj_to_inven(self_obj, restock_obj);
                             end
                         end
                         else begin
-                            if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                metarule(43, self_obj);
+                            if ((critter_inven_obj(self_obj, INVEN_TYPE_LEFT_HAND) == restock_obj) or (critter_inven_obj(self_obj, INVEN_TYPE_RIGHT_HAND) == restock_obj)) then begin
+                                inven_unwield(self_obj);
                             end
                         end
                     end
@@ -467,12 +467,12 @@ begin
             end
         end
     end
-    if (message_str(765, 104) != 0) then begin
+    if (mstr(104) != 0) then begin
         restock_amt := random(4, 6);
         if (obj_is_carrying_obj_pid(self_obj, 320) < restock_amt) then begin
             if (80 >= random(1, 100)) then begin
                 if (obj_is_carrying_obj_pid(self_obj, 320) < restock_amt) then begin
-                    restock_obj := create_object_sid(320, 0, 0, -1);
+                    restock_obj := create_object(320, 0, 0);
                     add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 320));
                 end
                 else begin
@@ -485,17 +485,17 @@ begin
                             restock_obj := obj_carrying_pid_obj(self_obj, 320);
                             if (obj_type(self_obj) == 1) then begin
                                 restock_obj := obj_carrying_pid_obj(self_obj, 320);
-                                if (critter_inven_obj(self_obj, 0) == restock_obj) then begin
-                                    if (obj_item_subtype(critter_inven_obj(self_obj, 0)) == 0) then begin
-                                        restock_obj := critter_inven_obj(self_obj, 0);
+                                if (critter_inven_obj(self_obj, INVEN_TYPE_WORN) == restock_obj) then begin
+                                    if (obj_item_subtype(critter_inven_obj(self_obj, INVEN_TYPE_WORN)) == 0) then begin
+                                        restock_obj := critter_inven_obj(self_obj, INVEN_TYPE_WORN);
                                         debug_msg("armour pid == " + obj_pid(restock_obj));
                                         rm_obj_from_inven(self_obj, restock_obj);
                                         add_obj_to_inven(self_obj, restock_obj);
                                     end
                                 end
                                 else begin
-                                    if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                        metarule(43, self_obj);
+                                    if ((critter_inven_obj(self_obj, INVEN_TYPE_LEFT_HAND) == restock_obj) or (critter_inven_obj(self_obj, INVEN_TYPE_RIGHT_HAND) == restock_obj)) then begin
+                                        inven_unwield(self_obj);
                                     end
                                 end
                             end
@@ -521,17 +521,17 @@ begin
                     restock_obj := obj_carrying_pid_obj(self_obj, 320);
                     if (obj_type(self_obj) == 1) then begin
                         restock_obj := obj_carrying_pid_obj(self_obj, 320);
-                        if (critter_inven_obj(self_obj, 0) == restock_obj) then begin
-                            if (obj_item_subtype(critter_inven_obj(self_obj, 0)) == 0) then begin
-                                restock_obj := critter_inven_obj(self_obj, 0);
+                        if (critter_inven_obj(self_obj, INVEN_TYPE_WORN) == restock_obj) then begin
+                            if (obj_item_subtype(critter_inven_obj(self_obj, INVEN_TYPE_WORN)) == 0) then begin
+                                restock_obj := critter_inven_obj(self_obj, INVEN_TYPE_WORN);
                                 debug_msg("armour pid == " + obj_pid(restock_obj));
                                 rm_obj_from_inven(self_obj, restock_obj);
                                 add_obj_to_inven(self_obj, restock_obj);
                             end
                         end
                         else begin
-                            if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                metarule(43, self_obj);
+                            if ((critter_inven_obj(self_obj, INVEN_TYPE_LEFT_HAND) == restock_obj) or (critter_inven_obj(self_obj, INVEN_TYPE_RIGHT_HAND) == restock_obj)) then begin
+                                inven_unwield(self_obj);
                             end
                         end
                     end
@@ -546,12 +546,12 @@ begin
             end
         end
     end
-    if (message_str(765, 105) != 0) then begin
+    if (mstr(105) != 0) then begin
         restock_amt := random(5, 6);
         if (obj_is_carrying_obj_pid(self_obj, 276) < restock_amt) then begin
             if (80 >= random(1, 100)) then begin
                 if (obj_is_carrying_obj_pid(self_obj, 276) < restock_amt) then begin
-                    restock_obj := create_object_sid(276, 0, 0, -1);
+                    restock_obj := create_object(276, 0, 0);
                     add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 276));
                 end
                 else begin
@@ -564,17 +564,17 @@ begin
                             restock_obj := obj_carrying_pid_obj(self_obj, 276);
                             if (obj_type(self_obj) == 1) then begin
                                 restock_obj := obj_carrying_pid_obj(self_obj, 276);
-                                if (critter_inven_obj(self_obj, 0) == restock_obj) then begin
-                                    if (obj_item_subtype(critter_inven_obj(self_obj, 0)) == 0) then begin
-                                        restock_obj := critter_inven_obj(self_obj, 0);
+                                if (critter_inven_obj(self_obj, INVEN_TYPE_WORN) == restock_obj) then begin
+                                    if (obj_item_subtype(critter_inven_obj(self_obj, INVEN_TYPE_WORN)) == 0) then begin
+                                        restock_obj := critter_inven_obj(self_obj, INVEN_TYPE_WORN);
                                         debug_msg("armour pid == " + obj_pid(restock_obj));
                                         rm_obj_from_inven(self_obj, restock_obj);
                                         add_obj_to_inven(self_obj, restock_obj);
                                     end
                                 end
                                 else begin
-                                    if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                        metarule(43, self_obj);
+                                    if ((critter_inven_obj(self_obj, INVEN_TYPE_LEFT_HAND) == restock_obj) or (critter_inven_obj(self_obj, INVEN_TYPE_RIGHT_HAND) == restock_obj)) then begin
+                                        inven_unwield(self_obj);
                                     end
                                 end
                             end
@@ -600,17 +600,17 @@ begin
                     restock_obj := obj_carrying_pid_obj(self_obj, 276);
                     if (obj_type(self_obj) == 1) then begin
                         restock_obj := obj_carrying_pid_obj(self_obj, 276);
-                        if (critter_inven_obj(self_obj, 0) == restock_obj) then begin
-                            if (obj_item_subtype(critter_inven_obj(self_obj, 0)) == 0) then begin
-                                restock_obj := critter_inven_obj(self_obj, 0);
+                        if (critter_inven_obj(self_obj, INVEN_TYPE_WORN) == restock_obj) then begin
+                            if (obj_item_subtype(critter_inven_obj(self_obj, INVEN_TYPE_WORN)) == 0) then begin
+                                restock_obj := critter_inven_obj(self_obj, INVEN_TYPE_WORN);
                                 debug_msg("armour pid == " + obj_pid(restock_obj));
                                 rm_obj_from_inven(self_obj, restock_obj);
                                 add_obj_to_inven(self_obj, restock_obj);
                             end
                         end
                         else begin
-                            if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                metarule(43, self_obj);
+                            if ((critter_inven_obj(self_obj, INVEN_TYPE_LEFT_HAND) == restock_obj) or (critter_inven_obj(self_obj, INVEN_TYPE_RIGHT_HAND) == restock_obj)) then begin
+                                inven_unwield(self_obj);
                             end
                         end
                     end
@@ -625,12 +625,12 @@ begin
             end
         end
     end
-    if (message_str(765, 106) != 0) then begin
+    if (mstr(106) != 0) then begin
         restock_amt := random(1, 2);
         if (obj_is_carrying_obj_pid(self_obj, 101) < restock_amt) then begin
             if (80 >= random(1, 100)) then begin
                 if (obj_is_carrying_obj_pid(self_obj, 101) < restock_amt) then begin
-                    restock_obj := create_object_sid(101, 0, 0, -1);
+                    restock_obj := create_object(101, 0, 0);
                     add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 101));
                 end
                 else begin
@@ -643,17 +643,17 @@ begin
                             restock_obj := obj_carrying_pid_obj(self_obj, 101);
                             if (obj_type(self_obj) == 1) then begin
                                 restock_obj := obj_carrying_pid_obj(self_obj, 101);
-                                if (critter_inven_obj(self_obj, 0) == restock_obj) then begin
-                                    if (obj_item_subtype(critter_inven_obj(self_obj, 0)) == 0) then begin
-                                        restock_obj := critter_inven_obj(self_obj, 0);
+                                if (critter_inven_obj(self_obj, INVEN_TYPE_WORN) == restock_obj) then begin
+                                    if (obj_item_subtype(critter_inven_obj(self_obj, INVEN_TYPE_WORN)) == 0) then begin
+                                        restock_obj := critter_inven_obj(self_obj, INVEN_TYPE_WORN);
                                         debug_msg("armour pid == " + obj_pid(restock_obj));
                                         rm_obj_from_inven(self_obj, restock_obj);
                                         add_obj_to_inven(self_obj, restock_obj);
                                     end
                                 end
                                 else begin
-                                    if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                        metarule(43, self_obj);
+                                    if ((critter_inven_obj(self_obj, INVEN_TYPE_LEFT_HAND) == restock_obj) or (critter_inven_obj(self_obj, INVEN_TYPE_RIGHT_HAND) == restock_obj)) then begin
+                                        inven_unwield(self_obj);
                                     end
                                 end
                             end
@@ -679,17 +679,17 @@ begin
                     restock_obj := obj_carrying_pid_obj(self_obj, 101);
                     if (obj_type(self_obj) == 1) then begin
                         restock_obj := obj_carrying_pid_obj(self_obj, 101);
-                        if (critter_inven_obj(self_obj, 0) == restock_obj) then begin
-                            if (obj_item_subtype(critter_inven_obj(self_obj, 0)) == 0) then begin
-                                restock_obj := critter_inven_obj(self_obj, 0);
+                        if (critter_inven_obj(self_obj, INVEN_TYPE_WORN) == restock_obj) then begin
+                            if (obj_item_subtype(critter_inven_obj(self_obj, INVEN_TYPE_WORN)) == 0) then begin
+                                restock_obj := critter_inven_obj(self_obj, INVEN_TYPE_WORN);
                                 debug_msg("armour pid == " + obj_pid(restock_obj));
                                 rm_obj_from_inven(self_obj, restock_obj);
                                 add_obj_to_inven(self_obj, restock_obj);
                             end
                         end
                         else begin
-                            if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                metarule(43, self_obj);
+                            if ((critter_inven_obj(self_obj, INVEN_TYPE_LEFT_HAND) == restock_obj) or (critter_inven_obj(self_obj, INVEN_TYPE_RIGHT_HAND) == restock_obj)) then begin
+                                inven_unwield(self_obj);
                             end
                         end
                     end
@@ -704,12 +704,12 @@ begin
             end
         end
     end
-    if (message_str(765, 108) != 0) then begin
+    if (mstr(108) != 0) then begin
         restock_amt := random(5, 6);
         if (obj_is_carrying_obj_pid(self_obj, 106) < restock_amt) then begin
             if (80 >= random(1, 100)) then begin
                 if (obj_is_carrying_obj_pid(self_obj, 106) < restock_amt) then begin
-                    restock_obj := create_object_sid(106, 0, 0, -1);
+                    restock_obj := create_object(106, 0, 0);
                     add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 106));
                 end
                 else begin
@@ -722,17 +722,17 @@ begin
                             restock_obj := obj_carrying_pid_obj(self_obj, 106);
                             if (obj_type(self_obj) == 1) then begin
                                 restock_obj := obj_carrying_pid_obj(self_obj, 106);
-                                if (critter_inven_obj(self_obj, 0) == restock_obj) then begin
-                                    if (obj_item_subtype(critter_inven_obj(self_obj, 0)) == 0) then begin
-                                        restock_obj := critter_inven_obj(self_obj, 0);
+                                if (critter_inven_obj(self_obj, INVEN_TYPE_WORN) == restock_obj) then begin
+                                    if (obj_item_subtype(critter_inven_obj(self_obj, INVEN_TYPE_WORN)) == 0) then begin
+                                        restock_obj := critter_inven_obj(self_obj, INVEN_TYPE_WORN);
                                         debug_msg("armour pid == " + obj_pid(restock_obj));
                                         rm_obj_from_inven(self_obj, restock_obj);
                                         add_obj_to_inven(self_obj, restock_obj);
                                     end
                                 end
                                 else begin
-                                    if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                        metarule(43, self_obj);
+                                    if ((critter_inven_obj(self_obj, INVEN_TYPE_LEFT_HAND) == restock_obj) or (critter_inven_obj(self_obj, INVEN_TYPE_RIGHT_HAND) == restock_obj)) then begin
+                                        inven_unwield(self_obj);
                                     end
                                 end
                             end
@@ -758,17 +758,17 @@ begin
                     restock_obj := obj_carrying_pid_obj(self_obj, 106);
                     if (obj_type(self_obj) == 1) then begin
                         restock_obj := obj_carrying_pid_obj(self_obj, 106);
-                        if (critter_inven_obj(self_obj, 0) == restock_obj) then begin
-                            if (obj_item_subtype(critter_inven_obj(self_obj, 0)) == 0) then begin
-                                restock_obj := critter_inven_obj(self_obj, 0);
+                        if (critter_inven_obj(self_obj, INVEN_TYPE_WORN) == restock_obj) then begin
+                            if (obj_item_subtype(critter_inven_obj(self_obj, INVEN_TYPE_WORN)) == 0) then begin
+                                restock_obj := critter_inven_obj(self_obj, INVEN_TYPE_WORN);
                                 debug_msg("armour pid == " + obj_pid(restock_obj));
                                 rm_obj_from_inven(self_obj, restock_obj);
                                 add_obj_to_inven(self_obj, restock_obj);
                             end
                         end
                         else begin
-                            if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                metarule(43, self_obj);
+                            if ((critter_inven_obj(self_obj, INVEN_TYPE_LEFT_HAND) == restock_obj) or (critter_inven_obj(self_obj, INVEN_TYPE_RIGHT_HAND) == restock_obj)) then begin
+                                inven_unwield(self_obj);
                             end
                         end
                     end
@@ -783,12 +783,12 @@ begin
             end
         end
     end
-    if (message_str(765, 109) != 0) then begin
+    if (mstr(109) != 0) then begin
         restock_amt := random(1, 3);
         if (obj_is_carrying_obj_pid(self_obj, 548) < restock_amt) then begin
             if (80 >= random(1, 100)) then begin
                 if (obj_is_carrying_obj_pid(self_obj, 548) < restock_amt) then begin
-                    restock_obj := create_object_sid(548, 0, 0, -1);
+                    restock_obj := create_object(548, 0, 0);
                     add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 548));
                 end
                 else begin
@@ -801,17 +801,17 @@ begin
                             restock_obj := obj_carrying_pid_obj(self_obj, 548);
                             if (obj_type(self_obj) == 1) then begin
                                 restock_obj := obj_carrying_pid_obj(self_obj, 548);
-                                if (critter_inven_obj(self_obj, 0) == restock_obj) then begin
-                                    if (obj_item_subtype(critter_inven_obj(self_obj, 0)) == 0) then begin
-                                        restock_obj := critter_inven_obj(self_obj, 0);
+                                if (critter_inven_obj(self_obj, INVEN_TYPE_WORN) == restock_obj) then begin
+                                    if (obj_item_subtype(critter_inven_obj(self_obj, INVEN_TYPE_WORN)) == 0) then begin
+                                        restock_obj := critter_inven_obj(self_obj, INVEN_TYPE_WORN);
                                         debug_msg("armour pid == " + obj_pid(restock_obj));
                                         rm_obj_from_inven(self_obj, restock_obj);
                                         add_obj_to_inven(self_obj, restock_obj);
                                     end
                                 end
                                 else begin
-                                    if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                        metarule(43, self_obj);
+                                    if ((critter_inven_obj(self_obj, INVEN_TYPE_LEFT_HAND) == restock_obj) or (critter_inven_obj(self_obj, INVEN_TYPE_RIGHT_HAND) == restock_obj)) then begin
+                                        inven_unwield(self_obj);
                                     end
                                 end
                             end
@@ -837,17 +837,17 @@ begin
                     restock_obj := obj_carrying_pid_obj(self_obj, 548);
                     if (obj_type(self_obj) == 1) then begin
                         restock_obj := obj_carrying_pid_obj(self_obj, 548);
-                        if (critter_inven_obj(self_obj, 0) == restock_obj) then begin
-                            if (obj_item_subtype(critter_inven_obj(self_obj, 0)) == 0) then begin
-                                restock_obj := critter_inven_obj(self_obj, 0);
+                        if (critter_inven_obj(self_obj, INVEN_TYPE_WORN) == restock_obj) then begin
+                            if (obj_item_subtype(critter_inven_obj(self_obj, INVEN_TYPE_WORN)) == 0) then begin
+                                restock_obj := critter_inven_obj(self_obj, INVEN_TYPE_WORN);
                                 debug_msg("armour pid == " + obj_pid(restock_obj));
                                 rm_obj_from_inven(self_obj, restock_obj);
                                 add_obj_to_inven(self_obj, restock_obj);
                             end
                         end
                         else begin
-                            if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                metarule(43, self_obj);
+                            if ((critter_inven_obj(self_obj, INVEN_TYPE_LEFT_HAND) == restock_obj) or (critter_inven_obj(self_obj, INVEN_TYPE_RIGHT_HAND) == restock_obj)) then begin
+                                inven_unwield(self_obj);
                             end
                         end
                     end
@@ -862,12 +862,12 @@ begin
             end
         end
     end
-    if (message_str(765, 110) != 0) then begin
+    if (mstr(110) != 0) then begin
         restock_amt := random(3, 6);
         if (obj_is_carrying_obj_pid(self_obj, 71) < restock_amt) then begin
             if (80 >= random(1, 100)) then begin
                 if (obj_is_carrying_obj_pid(self_obj, 71) < restock_amt) then begin
-                    restock_obj := create_object_sid(71, 0, 0, -1);
+                    restock_obj := create_object(71, 0, 0);
                     add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 71));
                 end
                 else begin
@@ -880,17 +880,17 @@ begin
                             restock_obj := obj_carrying_pid_obj(self_obj, 71);
                             if (obj_type(self_obj) == 1) then begin
                                 restock_obj := obj_carrying_pid_obj(self_obj, 71);
-                                if (critter_inven_obj(self_obj, 0) == restock_obj) then begin
-                                    if (obj_item_subtype(critter_inven_obj(self_obj, 0)) == 0) then begin
-                                        restock_obj := critter_inven_obj(self_obj, 0);
+                                if (critter_inven_obj(self_obj, INVEN_TYPE_WORN) == restock_obj) then begin
+                                    if (obj_item_subtype(critter_inven_obj(self_obj, INVEN_TYPE_WORN)) == 0) then begin
+                                        restock_obj := critter_inven_obj(self_obj, INVEN_TYPE_WORN);
                                         debug_msg("armour pid == " + obj_pid(restock_obj));
                                         rm_obj_from_inven(self_obj, restock_obj);
                                         add_obj_to_inven(self_obj, restock_obj);
                                     end
                                 end
                                 else begin
-                                    if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                        metarule(43, self_obj);
+                                    if ((critter_inven_obj(self_obj, INVEN_TYPE_LEFT_HAND) == restock_obj) or (critter_inven_obj(self_obj, INVEN_TYPE_RIGHT_HAND) == restock_obj)) then begin
+                                        inven_unwield(self_obj);
                                     end
                                 end
                             end
@@ -916,17 +916,17 @@ begin
                     restock_obj := obj_carrying_pid_obj(self_obj, 71);
                     if (obj_type(self_obj) == 1) then begin
                         restock_obj := obj_carrying_pid_obj(self_obj, 71);
-                        if (critter_inven_obj(self_obj, 0) == restock_obj) then begin
-                            if (obj_item_subtype(critter_inven_obj(self_obj, 0)) == 0) then begin
-                                restock_obj := critter_inven_obj(self_obj, 0);
+                        if (critter_inven_obj(self_obj, INVEN_TYPE_WORN) == restock_obj) then begin
+                            if (obj_item_subtype(critter_inven_obj(self_obj, INVEN_TYPE_WORN)) == 0) then begin
+                                restock_obj := critter_inven_obj(self_obj, INVEN_TYPE_WORN);
                                 debug_msg("armour pid == " + obj_pid(restock_obj));
                                 rm_obj_from_inven(self_obj, restock_obj);
                                 add_obj_to_inven(self_obj, restock_obj);
                             end
                         end
                         else begin
-                            if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                metarule(43, self_obj);
+                            if ((critter_inven_obj(self_obj, INVEN_TYPE_LEFT_HAND) == restock_obj) or (critter_inven_obj(self_obj, INVEN_TYPE_RIGHT_HAND) == restock_obj)) then begin
+                                inven_unwield(self_obj);
                             end
                         end
                     end
@@ -941,12 +941,12 @@ begin
             end
         end
     end
-    if (message_str(765, 111) != 0) then begin
+    if (mstr(111) != 0) then begin
         restock_amt := random(1, 2);
         if (obj_is_carrying_obj_pid(self_obj, 127) < restock_amt) then begin
             if (80 >= random(1, 100)) then begin
                 if (obj_is_carrying_obj_pid(self_obj, 127) < restock_amt) then begin
-                    restock_obj := create_object_sid(127, 0, 0, -1);
+                    restock_obj := create_object(127, 0, 0);
                     add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 127));
                 end
                 else begin
@@ -959,17 +959,17 @@ begin
                             restock_obj := obj_carrying_pid_obj(self_obj, 127);
                             if (obj_type(self_obj) == 1) then begin
                                 restock_obj := obj_carrying_pid_obj(self_obj, 127);
-                                if (critter_inven_obj(self_obj, 0) == restock_obj) then begin
-                                    if (obj_item_subtype(critter_inven_obj(self_obj, 0)) == 0) then begin
-                                        restock_obj := critter_inven_obj(self_obj, 0);
+                                if (critter_inven_obj(self_obj, INVEN_TYPE_WORN) == restock_obj) then begin
+                                    if (obj_item_subtype(critter_inven_obj(self_obj, INVEN_TYPE_WORN)) == 0) then begin
+                                        restock_obj := critter_inven_obj(self_obj, INVEN_TYPE_WORN);
                                         debug_msg("armour pid == " + obj_pid(restock_obj));
                                         rm_obj_from_inven(self_obj, restock_obj);
                                         add_obj_to_inven(self_obj, restock_obj);
                                     end
                                 end
                                 else begin
-                                    if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                        metarule(43, self_obj);
+                                    if ((critter_inven_obj(self_obj, INVEN_TYPE_LEFT_HAND) == restock_obj) or (critter_inven_obj(self_obj, INVEN_TYPE_RIGHT_HAND) == restock_obj)) then begin
+                                        inven_unwield(self_obj);
                                     end
                                 end
                             end
@@ -995,17 +995,17 @@ begin
                     restock_obj := obj_carrying_pid_obj(self_obj, 127);
                     if (obj_type(self_obj) == 1) then begin
                         restock_obj := obj_carrying_pid_obj(self_obj, 127);
-                        if (critter_inven_obj(self_obj, 0) == restock_obj) then begin
-                            if (obj_item_subtype(critter_inven_obj(self_obj, 0)) == 0) then begin
-                                restock_obj := critter_inven_obj(self_obj, 0);
+                        if (critter_inven_obj(self_obj, INVEN_TYPE_WORN) == restock_obj) then begin
+                            if (obj_item_subtype(critter_inven_obj(self_obj, INVEN_TYPE_WORN)) == 0) then begin
+                                restock_obj := critter_inven_obj(self_obj, INVEN_TYPE_WORN);
                                 debug_msg("armour pid == " + obj_pid(restock_obj));
                                 rm_obj_from_inven(self_obj, restock_obj);
                                 add_obj_to_inven(self_obj, restock_obj);
                             end
                         end
                         else begin
-                            if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                metarule(43, self_obj);
+                            if ((critter_inven_obj(self_obj, INVEN_TYPE_LEFT_HAND) == restock_obj) or (critter_inven_obj(self_obj, INVEN_TYPE_RIGHT_HAND) == restock_obj)) then begin
+                                inven_unwield(self_obj);
                             end
                         end
                     end
@@ -1020,12 +1020,12 @@ begin
             end
         end
     end
-    if (message_str(765, 112) != 0) then begin
+    if (mstr(112) != 0) then begin
         restock_amt := random(5, 6);
         if (obj_is_carrying_obj_pid(self_obj, 284) < restock_amt) then begin
             if (80 >= random(1, 100)) then begin
                 if (obj_is_carrying_obj_pid(self_obj, 284) < restock_amt) then begin
-                    restock_obj := create_object_sid(284, 0, 0, -1);
+                    restock_obj := create_object(284, 0, 0);
                     add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 284));
                 end
                 else begin
@@ -1038,17 +1038,17 @@ begin
                             restock_obj := obj_carrying_pid_obj(self_obj, 284);
                             if (obj_type(self_obj) == 1) then begin
                                 restock_obj := obj_carrying_pid_obj(self_obj, 284);
-                                if (critter_inven_obj(self_obj, 0) == restock_obj) then begin
-                                    if (obj_item_subtype(critter_inven_obj(self_obj, 0)) == 0) then begin
-                                        restock_obj := critter_inven_obj(self_obj, 0);
+                                if (critter_inven_obj(self_obj, INVEN_TYPE_WORN) == restock_obj) then begin
+                                    if (obj_item_subtype(critter_inven_obj(self_obj, INVEN_TYPE_WORN)) == 0) then begin
+                                        restock_obj := critter_inven_obj(self_obj, INVEN_TYPE_WORN);
                                         debug_msg("armour pid == " + obj_pid(restock_obj));
                                         rm_obj_from_inven(self_obj, restock_obj);
                                         add_obj_to_inven(self_obj, restock_obj);
                                     end
                                 end
                                 else begin
-                                    if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                        metarule(43, self_obj);
+                                    if ((critter_inven_obj(self_obj, INVEN_TYPE_LEFT_HAND) == restock_obj) or (critter_inven_obj(self_obj, INVEN_TYPE_RIGHT_HAND) == restock_obj)) then begin
+                                        inven_unwield(self_obj);
                                     end
                                 end
                             end
@@ -1074,17 +1074,17 @@ begin
                     restock_obj := obj_carrying_pid_obj(self_obj, 284);
                     if (obj_type(self_obj) == 1) then begin
                         restock_obj := obj_carrying_pid_obj(self_obj, 284);
-                        if (critter_inven_obj(self_obj, 0) == restock_obj) then begin
-                            if (obj_item_subtype(critter_inven_obj(self_obj, 0)) == 0) then begin
-                                restock_obj := critter_inven_obj(self_obj, 0);
+                        if (critter_inven_obj(self_obj, INVEN_TYPE_WORN) == restock_obj) then begin
+                            if (obj_item_subtype(critter_inven_obj(self_obj, INVEN_TYPE_WORN)) == 0) then begin
+                                restock_obj := critter_inven_obj(self_obj, INVEN_TYPE_WORN);
                                 debug_msg("armour pid == " + obj_pid(restock_obj));
                                 rm_obj_from_inven(self_obj, restock_obj);
                                 add_obj_to_inven(self_obj, restock_obj);
                             end
                         end
                         else begin
-                            if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                metarule(43, self_obj);
+                            if ((critter_inven_obj(self_obj, INVEN_TYPE_LEFT_HAND) == restock_obj) or (critter_inven_obj(self_obj, INVEN_TYPE_RIGHT_HAND) == restock_obj)) then begin
+                                inven_unwield(self_obj);
                             end
                         end
                     end
@@ -1126,109 +1126,109 @@ end
 
 procedure Node001
 begin
-    if ((cur_map_index == 122) or (cur_map_index == 123) or (cur_map_index == 81) or (cur_map_index == 82) or (cur_map_index == 114) or (cur_map_index == 89) or (cur_map_index == 90) or (cur_map_index == 91) or (cur_map_index == 110) or (cur_map_index == 111) or (cur_map_index == 112)) then begin
-        gsay_reply(765, 103);
-        giq_option(4, 765, 104, Node002, 50);
-        giq_option(4, 765, 105, Node003, 50);
-        giq_option(-3, 765, 106, Node005, 50);
+    if ((CUR_MAP_MOUNTN4) or (CUR_MAP_MOUNTN5) or (CUR_MAP_DESERT4) or (CUR_MAP_DESERT5) or (CUR_MAP_DESERT9) or (CUR_MAP_COAST7) or (CUR_MAP_COAST8) or (CUR_MAP_COAST9) or (CUR_MAP_COAST10) or (CUR_MAP_COAST11) or (CUR_MAP_07DESERT)) then begin
+        Reply(103);
+        giq_option(4, NAME, 104, Node002, NEUTRAL_REACTION);
+        giq_option(4, NAME, 105, Node003, NEUTRAL_REACTION);
+        giq_option(-3, NAME, 106, Node005, NEUTRAL_REACTION);
     end
     else begin
-        gsay_reply(765, 200);
-        giq_option(4, 765, 201, Node002, 50);
-        giq_option(4, 765, 105, Node003, 50);
-        giq_option(-3, 765, 106, Node005, 50);
+        Reply(200);
+        giq_option(4, NAME, 201, Node002, NEUTRAL_REACTION);
+        giq_option(4, NAME, 105, Node003, NEUTRAL_REACTION);
+        giq_option(-3, NAME, 106, Node005, NEUTRAL_REACTION);
     end
 end
 
 procedure Node002
 begin
-    gsay_reply(765, 107);
-    giq_option(6, 765, message_str(765, random(250, 253)), Node010, 50);
-    giq_option(4, 765, 108, NodeBarter, 50);
-    if ((cur_map_index == 122) or (cur_map_index == 123) or (cur_map_index == 81) or (cur_map_index == 82) or (cur_map_index == 114) or (cur_map_index == 89) or (cur_map_index == 90) or (cur_map_index == 91) or (cur_map_index == 110) or (cur_map_index == 111) or (cur_map_index == 112)) then begin
-        giq_option(4, 765, 109, Node004, 50);
+    Reply(107);
+    giq_option(6, NAME, mstr(random(250, 253)), Node010, NEUTRAL_REACTION);
+    giq_option(4, NAME, 108, NodeBarter, NEUTRAL_REACTION);
+    if ((CUR_MAP_MOUNTN4) or (CUR_MAP_MOUNTN5) or (CUR_MAP_DESERT4) or (CUR_MAP_DESERT5) or (CUR_MAP_DESERT9) or (CUR_MAP_COAST7) or (CUR_MAP_COAST8) or (CUR_MAP_COAST9) or (CUR_MAP_COAST10) or (CUR_MAP_COAST11) or (CUR_MAP_07DESERT)) then begin
+        giq_option(4, NAME, 109, Node004, NEUTRAL_REACTION);
     end
     else begin
-        giq_option(4, 765, 109, Node004a, 50);
+        giq_option(4, NAME, 109, Node004a, NEUTRAL_REACTION);
     end
 end
 
 procedure Node003
 begin
-    gsay_reply(765, 110);
-    giq_option(4, 765, 111, Node999, 50);
+    Reply(110);
+    giq_option(4, NAME, 111, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node004
 begin
-    gsay_reply(765, 112);
-    giq_option(6, 765, message_str(765, random(253, 255)), Node010, 50);
-    giq_option(4, 765, 113, Node999, 50);
+    Reply(112);
+    giq_option(6, NAME, mstr(random(253, 255)), Node010, NEUTRAL_REACTION);
+    giq_option(4, NAME, 113, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node004a
 begin
-    gsay_reply(765, 202);
-    giq_option(6, 765, message_str(765, random(253, 255)), Node010, 50);
-    giq_option(4, 765, 113, Node999, 50);
+    Reply(202);
+    giq_option(6, NAME, mstr(random(253, 255)), Node010, NEUTRAL_REACTION);
+    giq_option(4, NAME, 113, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node005
 begin
-    gsay_reply(765, 114);
-    giq_option(-3, 765, 115, Node999, 50);
+    Reply(114);
+    giq_option(-3, NAME, 115, Node999, NEUTRAL_REACTION);
 end
 
 procedure NodeFleeing
 begin
-    float_msg(self_obj, message_str(765, random(120, 122)), 8);
+    float_msg(self_obj, mstr(random(120, 122)), FLOAT_MSG_YELLOW);
 end
 
 procedure NodeFighting
 begin
-    float_msg(self_obj, message_str(765, random(116, 119)), 2);
+    float_msg(self_obj, mstr(random(116, 119)), FLOAT_MSG_RED);
 end
 
 procedure NodeBarter
 begin
-    gdialog_mod_barter(0);
-    gsay_reply(765, 203);
-    giq_option(4, 765, 390, Node024, 50);
-    giq_option(4, 765, 204, Node999, 50);
+    gdialog_barter;
+    Reply(203);
+    giq_option(4, NAME, 390, Node024, NEUTRAL_REACTION);
+    giq_option(4, NAME, 204, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node010
 begin
     set_local_var(6, 0);
-    if (message_str(765, 100) != 0) then begin
+    if (mstr(100) != 0) then begin
         if (random(1, 12) == 7) then begin
-            if (get_critter_stat(dude_obj, 34) == 0) then begin
-                gsay_reply(765, 281);
+            if (dude_gender == 0) then begin
+                Reply(281);
             end
             else begin
-                gsay_reply(765, 282);
+                Reply(282);
             end
         end
         else begin
             set_local_var(6, random(270, 280));
-            gsay_reply(765, message_str(765, local_var(6)));
+            Reply(mstr(local_var(6)));
         end
     end
-    if (message_str(765, 101) != 0) then begin
+    if (mstr(101) != 0) then begin
         if (random(1, 3) == 3) then begin
-            giq_option(4, 765, message_str(765, random(290, 292)), Node011, 50);
+            giq_option(4, NAME, mstr(random(290, 292)), Node011, NEUTRAL_REACTION);
         end
         else begin
-            giq_option(4, 765, message_str(765, random(290, 292)), Node012a, 50);
+            giq_option(4, NAME, mstr(random(290, 292)), Node012a, NEUTRAL_REACTION);
         end
-        giq_option(4, 765, message_str(765, random(300, 302)), NodeBarter, 50);
-        giq_option(4, 765, message_str(765, random(305, 306)), Node999, 50);
+        giq_option(4, NAME, mstr(random(300, 302)), NodeBarter, NEUTRAL_REACTION);
+        giq_option(4, NAME, mstr(random(305, 306)), Node999, NEUTRAL_REACTION);
     end
 end
 
 procedure Node011
 begin
-    if (message_str(765, 100) != 0) then begin
+    if (mstr(100) != 0) then begin
         if (local_var(6) == 280) then begin
             set_local_var(6, 270);
         end
@@ -1236,16 +1236,16 @@ begin
             set_local_var(6, local_var(6) + 1);
         end
     end
-    gsay_reply(765, local_var(6));
-    if (message_str(765, 101) != 0) then begin
+    Reply(local_var(6));
+    if (mstr(101) != 0) then begin
         if (random(1, 3) == 3) then begin
-            giq_option(4, 765, message_str(765, random(290, 292)), Node011, 50);
+            giq_option(4, NAME, mstr(random(290, 292)), Node011, NEUTRAL_REACTION);
         end
         else begin
-            giq_option(4, 765, message_str(765, random(290, 292)), Node012a, 50);
+            giq_option(4, NAME, mstr(random(290, 292)), Node012a, NEUTRAL_REACTION);
         end
-        giq_option(4, 765, message_str(765, random(300, 302)), NodeBarter, 50);
-        giq_option(4, 765, message_str(765, random(305, 306)), Node999, 50);
+        giq_option(4, NAME, mstr(random(300, 302)), NodeBarter, NEUTRAL_REACTION);
+        giq_option(4, NAME, mstr(random(305, 306)), Node999, NEUTRAL_REACTION);
     end
 end
 
@@ -1253,84 +1253,84 @@ procedure Node012
 begin
     variable LVar0 := 0;
     if (local_var(6) != 0) then begin
-        gsay_reply(765, message_str(765, random(331, 333)));
+        Reply(mstr(random(331, 333)));
     end
     else begin
-        gsay_reply(765, message_str(765, random(310, 312)) + message_str(765, random(315, 316)));
+        Reply(mstr(random(310, 312)) + mstr(random(315, 316)));
     end
-    if (global_var(452) < 2) then begin
+    if (global_var(GVAR_DEN_VIC_KNOWN) < 2) then begin
         if (local_var(6) != 1) then begin
             LVar0 := LVar0 + 1;
-            giq_option(4, 765, 320, Node013, 50);
+            giq_option(4, NAME, 320, Node013, NEUTRAL_REACTION);
         end
     end
-    if (global_var(629) == 0) then begin
+    if (global_var(GVAR_GAVE_GECK_EXP) == 0) then begin
         if (local_var(6) != 2) then begin
             LVar0 := LVar0 + 1;
-            giq_option(4, 765, 335, Node014, 50);
+            giq_option(4, NAME, 335, Node014, NEUTRAL_REACTION);
         end
     end
-    if (global_var(494) == 0) then begin
+    if (global_var(GVAR_FALLOUT_2) == 0) then begin
         if (local_var(6) != 3) then begin
             LVar0 := LVar0 + 1;
-            giq_option(4, 765, 340, Node015, 50);
+            giq_option(4, NAME, 340, Node015, NEUTRAL_REACTION);
         end
     end
-    if ((global_var(494) >= 4) and (global_var(494) < 8)) then begin
+    if ((global_var(GVAR_FALLOUT_2) >= 4) and (global_var(GVAR_FALLOUT_2) < 8)) then begin
         if (local_var(6) != 4) then begin
             LVar0 := LVar0 + 1;
-            giq_option(4, 765, 345, Node016, 50);
+            giq_option(4, NAME, 345, Node016, NEUTRAL_REACTION);
         end
     end
-    if (global_var(494) > 7) then begin
+    if (global_var(GVAR_FALLOUT_2) > 7) then begin
         if (local_var(6) != 5) then begin
             LVar0 := LVar0 + 1;
-            giq_option(4, 765, 350, Node017, 50);
+            giq_option(4, NAME, 350, Node017, NEUTRAL_REACTION);
         end
     end
-    if (message_str(765, 100) != 0) then begin
-        if ((metarule(17, 50) == 0) and (global_var(716) == 1) and (((get_critter_stat(dude_obj, 1) + (get_critter_stat(dude_obj, 4) + has_trait(0, dude_obj, 34))) / 2) >= 7)) then begin
+    if (mstr(100) != 0) then begin
+        if ((town_known(50) == MARK_STATE_UNKNOWN) and (global_var(GVAR_DEN_FRED_NO) == 1) and (((dude_perception + (dude_iq + has_trait(TRAIT_PERK, dude_obj, PERK_speaker))) / 2) >= 7)) then begin
             if ((LVar0 < 4) and (local_var(6) != 6)) then begin
                 LVar0 := LVar0 + 1;
-                giq_option(4, 765, 355, Node018, 50);
+                giq_option(4, NAME, 355, Node018, NEUTRAL_REACTION);
             end
         end
         else begin
-            if ((global_var(1128) == 1) and (global_var(445) bwand 1) and (global_var(11) == 0)) then begin
+            if ((global_var(GVAR_VAULT_EXPERIMENTS) == 1) and (global_var(GVAR_DEN_FLAG_1) bwand 1) and (global_var(GVAR_REPUTATION_SLAVER) == 0)) then begin
                 if ((LVar0 < 4) and (local_var(6) != 7)) then begin
                     LVar0 := LVar0 + 1;
-                    giq_option(4, 765, 360, Node019, 50);
+                    giq_option(4, NAME, 360, Node019, NEUTRAL_REACTION);
                 end
             end
         end
     end
-    if (global_var(82) == 8) then begin
+    if (global_var(GVAR_VAULT_GECKO_PLANT) == 8) then begin
         if ((local_var(6) != 8) and (LVar0 < 4)) then begin
             LVar0 := LVar0 + 1;
-            giq_option(4, 765, 365, Node020, 50);
+            giq_option(4, NAME, 365, Node020, NEUTRAL_REACTION);
         end
     end
-    if ((global_var(390) != 0) and (local_var(6) != 9)) then begin
+    if ((global_var(GVAR_QUEST_RAT_GOD) != 0) and (local_var(6) != 9)) then begin
         if (LVar0 < 4) then begin
-            if (global_var(390) < 2) then begin
+            if (global_var(GVAR_QUEST_RAT_GOD) < 2) then begin
                 LVar0 := LVar0 + 1;
-                giq_option(4, 765, 370, Node021, 50);
+                giq_option(4, NAME, 370, Node021, NEUTRAL_REACTION);
             end
             else begin
-                if (global_var(390) == 2) then begin
+                if (global_var(GVAR_QUEST_RAT_GOD) == 2) then begin
                     LVar0 := LVar0 + 1;
-                    giq_option(4, 765, 375, Node022, 50);
+                    giq_option(4, NAME, 375, Node022, NEUTRAL_REACTION);
                 end
             end
         end
     end
-    if ((global_var(399) bwand 1) and ((global_var(399) bwand 128) == 0)) then begin
+    if ((global_var(GVAR_RND_KAGA_STATE) bwand 1) and ((global_var(GVAR_RND_KAGA_STATE) bwand 128) == 0)) then begin
         if ((local_var(6) != 10) and (LVar0 < 4)) then begin
-            giq_option(4, 765, 380, Node023, 50);
+            giq_option(4, NAME, 380, Node023, NEUTRAL_REACTION);
         end
     end
-    giq_option(4, 765, message_str(765, random(325, 327)), NodeBarter, 50);
-    giq_option(4, 765, message_str(765, random(328, 330)), Node999, 50);
+    giq_option(4, NAME, mstr(random(325, 327)), NodeBarter, NEUTRAL_REACTION);
+    giq_option(4, NAME, mstr(random(328, 330)), Node999, NEUTRAL_REACTION);
 end
 
 procedure Node012a
@@ -1342,122 +1342,121 @@ end
 procedure Node013
 begin
     set_local_var(6, 1);
-    gsay_reply(765, 321);
-    giq_option(4, 765, message_str(765, random(322, 324)), Node012, 50);
-    giq_option(4, 765, message_str(765, random(325, 327)), NodeBarter, 50);
-    giq_option(4, 765, message_str(765, random(328, 330)), Node999, 50);
+    Reply(321);
+    giq_option(4, NAME, mstr(random(322, 324)), Node012, NEUTRAL_REACTION);
+    giq_option(4, NAME, mstr(random(325, 327)), NodeBarter, NEUTRAL_REACTION);
+    giq_option(4, NAME, mstr(random(328, 330)), Node999, NEUTRAL_REACTION);
 end
 
 procedure Node014
 begin
     set_local_var(6, 2);
-    gsay_reply(765, 336);
-    giq_option(4, 765, 324, Node012, 50);
-    giq_option(4, 765, message_str(765, random(325, 327)), NodeBarter, 50);
-    giq_option(4, 765, message_str(765, random(328, 330)), Node999, 50);
+    Reply(336);
+    giq_option(4, NAME, 324, Node012, NEUTRAL_REACTION);
+    giq_option(4, NAME, mstr(random(325, 327)), NodeBarter, NEUTRAL_REACTION);
+    giq_option(4, NAME, mstr(random(328, 330)), Node999, NEUTRAL_REACTION);
 end
 
 procedure Node015
 begin
     set_local_var(6, 3);
-    gsay_reply(765, 341);
-    giq_option(4, 765, message_str(765, random(322, 324)), Node012, 50);
-    giq_option(4, 765, message_str(765, random(325, 327)), NodeBarter, 50);
-    giq_option(4, 765, message_str(765, random(328, 330)), Node999, 50);
+    Reply(341);
+    giq_option(4, NAME, mstr(random(322, 324)), Node012, NEUTRAL_REACTION);
+    giq_option(4, NAME, mstr(random(325, 327)), NodeBarter, NEUTRAL_REACTION);
+    giq_option(4, NAME, mstr(random(328, 330)), Node999, NEUTRAL_REACTION);
 end
 
 procedure Node016
 begin
     set_local_var(6, 4);
-    gsay_reply(765, 346);
-    giq_option(4, 765, 324, Node012, 50);
-    giq_option(4, 765, message_str(765, random(325, 327)), NodeBarter, 50);
-    giq_option(4, 765, message_str(765, random(328, 330)), Node999, 50);
+    Reply(346);
+    giq_option(4, NAME, 324, Node012, NEUTRAL_REACTION);
+    giq_option(4, NAME, mstr(random(325, 327)), NodeBarter, NEUTRAL_REACTION);
+    giq_option(4, NAME, mstr(random(328, 330)), Node999, NEUTRAL_REACTION);
 end
 
 procedure Node017
 begin
     set_local_var(6, 5);
-    gsay_reply(765, 351);
-    giq_option(4, 765, 324, Node012, 50);
-    giq_option(4, 765, message_str(765, random(325, 327)), NodeBarter, 50);
-    giq_option(4, 765, message_str(765, random(328, 330)), Node999, 50);
+    Reply(351);
+    giq_option(4, NAME, 324, Node012, NEUTRAL_REACTION);
+    giq_option(4, NAME, mstr(random(325, 327)), NodeBarter, NEUTRAL_REACTION);
+    giq_option(4, NAME, mstr(random(328, 330)), Node999, NEUTRAL_REACTION);
 end
 
 procedure Node018
 begin
-    mark_area_known(0, 50, 1);
+    mark_area_known(MARK_TYPE_TOWN, 50, MARK_STATE_KNOWN);
     set_local_var(6, 6);
-    gsay_reply(765, 356);
-    giq_option(4, 765, 357, Node012, 50);
-    giq_option(4, 765, message_str(765, random(325, 327)), NodeBarter, 50);
-    giq_option(4, 765, message_str(765, random(328, 330)), Node999, 50);
+    Reply(356);
+    giq_option(4, NAME, 357, Node012, NEUTRAL_REACTION);
+    giq_option(4, NAME, mstr(random(325, 327)), NodeBarter, NEUTRAL_REACTION);
+    giq_option(4, NAME, mstr(random(328, 330)), Node999, NEUTRAL_REACTION);
 end
 
 procedure Node019
 begin
     set_local_var(6, 7);
-    gsay_reply(765, 361);
-    giq_option(4, 765, 324, Node012, 50);
-    giq_option(4, 765, message_str(765, random(325, 327)), NodeBarter, 50);
-    giq_option(4, 765, message_str(765, random(328, 330)), Node999, 50);
+    Reply(361);
+    giq_option(4, NAME, 324, Node012, NEUTRAL_REACTION);
+    giq_option(4, NAME, mstr(random(325, 327)), NodeBarter, NEUTRAL_REACTION);
+    giq_option(4, NAME, mstr(random(328, 330)), Node999, NEUTRAL_REACTION);
 end
 
 procedure Node020
 begin
     set_local_var(6, 8);
-    gsay_reply(765, 366);
-    giq_option(4, 765, 324, Node012, 50);
-    giq_option(4, 765, message_str(765, random(325, 327)), NodeBarter, 50);
-    giq_option(4, 765, message_str(765, random(328, 330)), Node999, 50);
+    Reply(366);
+    giq_option(4, NAME, 324, Node012, NEUTRAL_REACTION);
+    giq_option(4, NAME, mstr(random(325, 327)), NodeBarter, NEUTRAL_REACTION);
+    giq_option(4, NAME, mstr(random(328, 330)), Node999, NEUTRAL_REACTION);
 end
 
 procedure Node021
 begin
     set_local_var(6, 9);
-    gsay_reply(765, 371);
-    giq_option(4, 765, 324, Node012, 50);
-    giq_option(4, 765, message_str(765, random(325, 327)), NodeBarter, 50);
-    giq_option(4, 765, message_str(765, random(328, 330)), Node999, 50);
+    Reply(371);
+    giq_option(4, NAME, 324, Node012, NEUTRAL_REACTION);
+    giq_option(4, NAME, mstr(random(325, 327)), NodeBarter, NEUTRAL_REACTION);
+    giq_option(4, NAME, mstr(random(328, 330)), Node999, NEUTRAL_REACTION);
 end
 
 procedure Node022
 begin
     set_local_var(6, 9);
-    gsay_reply(765, 376);
-    giq_option(4, 765, 324, Node012, 50);
-    giq_option(4, 765, message_str(765, random(325, 327)), NodeBarter, 50);
-    giq_option(4, 765, message_str(765, random(328, 330)), Node999, 50);
+    Reply(376);
+    giq_option(4, NAME, 324, Node012, NEUTRAL_REACTION);
+    giq_option(4, NAME, mstr(random(325, 327)), NodeBarter, NEUTRAL_REACTION);
+    giq_option(4, NAME, mstr(random(328, 330)), Node999, NEUTRAL_REACTION);
 end
 
 procedure Node023
 begin
     set_local_var(6, 10);
-    gsay_reply(765, 381);
-    giq_option(4, 765, 324, Node012, 50);
-    giq_option(4, 765, message_str(765, random(325, 327)), NodeBarter, 50);
-    giq_option(4, 765, message_str(765, random(328, 330)), Node999, 50);
+    Reply(381);
+    giq_option(4, NAME, 324, Node012, NEUTRAL_REACTION);
+    giq_option(4, NAME, mstr(random(325, 327)), NodeBarter, NEUTRAL_REACTION);
+    giq_option(4, NAME, mstr(random(328, 330)), Node999, NEUTRAL_REACTION);
 end
 
 procedure Node024
 begin
     set_local_var(6, 11);
-    gsay_reply(765, 333);
-    if (global_var(452) < 2) then begin
-        giq_option(4, 765, 400, Node013, 50);
+    Reply(333);
+    if (global_var(GVAR_DEN_VIC_KNOWN) < 2) then begin
+        giq_option(4, NAME, 400, Node013, NEUTRAL_REACTION);
     end
-    if (global_var(629) == 0) then begin
-        giq_option(4, 765, 401, Node014, 50);
+    if (global_var(GVAR_GAVE_GECK_EXP) == 0) then begin
+        giq_option(4, NAME, 401, Node014, NEUTRAL_REACTION);
     end
-    if (global_var(494) == 0) then begin
-        giq_option(4, 765, 402, Node015, 50);
+    if (global_var(GVAR_FALLOUT_2) == 0) then begin
+        giq_option(4, NAME, 402, Node015, NEUTRAL_REACTION);
     end
-    if ((global_var(494) >= 4) and (global_var(494) < 8)) then begin
-        giq_option(4, 765, 403, Node016, 50);
+    if ((global_var(GVAR_FALLOUT_2) >= 4) and (global_var(GVAR_FALLOUT_2) < 8)) then begin
+        giq_option(4, NAME, 403, Node016, NEUTRAL_REACTION);
     end
-    if ((metarule(17, 50) == 0) and (global_var(1123) == 1) and (((get_critter_stat(dude_obj, 1) + (get_critter_stat(dude_obj, 4) + has_trait(0, dude_obj, 34))) / 2) >= 7)) then begin
-        giq_option(4, 765, 404, Node018, 50);
+    if ((town_known(50) == MARK_STATE_UNKNOWN) and (global_var(GVAR_SULIK_DEAD) == 1) and (((dude_perception + (dude_iq + has_trait(TRAIT_PERK, dude_obj, PERK_speaker))) / 2) >= 7)) then begin
+        giq_option(4, NAME, 404, Node018, NEUTRAL_REACTION);
     end
-    giq_option(4, 765, 395, Node999, 50);
+    giq_option(4, NAME, 395, Node999, NEUTRAL_REACTION);
 end
-
