diff --git a/Mods/MegaMod/Original/ECPLANT.ssl b/Mods/MegaMod/ReDefine/ECPLANT.ssl
index fec8848..65fd03d 100644
--- a/Mods/MegaMod/Original/ECPLANT.ssl
+++ b/Mods/MegaMod/ReDefine/ECPLANT.ssl
@@ -144,15 +144,15 @@ end
 
 procedure map_enter_p_proc
 begin
-    if (metarule(14, 0)) then begin
-        add_mult_objs_to_inven(self_obj, create_object_sid(365, 0, 0, -1), 10);
+    if (map_first_run) then begin
+        add_mult_objs_to_inven(self_obj, create_object(365, 0, 0), 10);
     end
 end
 
 procedure critter_p_proc
 begin
     if (obj_can_see_obj(self_obj, dude_obj) and (PowderUsed == 0)) then begin
-        attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+        attack(dude_obj);
     end
 end
 
@@ -180,7 +180,7 @@ end
 procedure timed_event_p_proc
 begin
     if (fixed_param == 1) then begin
-        float_msg(self_obj, message_str(614, 100), 0);
+        float_msg(self_obj, mstr(100), FLOAT_MSG_NORMAL);
         reg_anim_func(1, 1);
         reg_anim_animate(self_obj, 20, -1);
         reg_anim_func(3, 0);
@@ -196,16 +196,15 @@ procedure DestroyPlant
 begin
     if (PowderUsed) then begin
         give_exp_points(85);
-        display_msg(message_str(14, 100) + 85 + message_str(14, 101));
+        display_msg(g_mstr(100) + 85 + g_mstr(101));
     end
-    if (cur_map_index == 94) then begin
+    if (CUR_MAP_BHRNDDST) then begin
         set_map_var(map_var(1) - 1, 1);
     end
 end
 
 procedure start
 begin
-    critter_add_trait(self_obj, 1, 6, 121);
-    critter_add_trait(self_obj, 1, 5, 5);
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 121);
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_SPORE_PLANT);
 end
-
