diff --git a/Mods/MegaMod/Original/ECRAVPTY.ssl b/Mods/MegaMod/ReDefine/ECRAVPTY.ssl
index 117acda..c663592 100644
--- a/Mods/MegaMod/Original/ECRAVPTY.ssl
+++ b/Mods/MegaMod/ReDefine/ECRAVPTY.ssl
@@ -185,7 +185,7 @@ procedure critter_p_proc
 begin
     if ((local_var(5) == 2) and obj_can_see_obj(self_obj, dude_obj)) then begin
         set_local_var(5, 1);
-        attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+        attack(dude_obj);
     end
 end
 
@@ -221,76 +221,76 @@ begin
     end
     if (doSex == 1) then begin
         doSex := 0;
-        gfade_out(1);
-        game_time_advance(get_critter_stat(dude_obj, 2) * random(4, 8) * (60 * 10));
-        set_global_var(194, global_var(194) + (((((get_critter_stat(dude_obj, 3) * 50) + (get_critter_stat(dude_obj, 2) * 25) + (get_critter_stat(dude_obj, 5) * 13) + (get_critter_stat(dude_obj, 0) * 12)) / 100) + (has_trait(0, dude_obj, 94) * 2) + has_trait(2, dude_obj, 13) + (((global_var(194) >= 100) or (global_var(232) > 0)) * 2 * (dude_obj == dude_obj))) * 1));
-        if (global_var(358) >= 9) then begin
-            set_global_var(588, 1);
+        fade_out;
+        game_time_advance(dude_endurance * random(4, 8) * (60 * 10));
+        set_global_var(GVAR_PLAYER_SEX_LEVEL, global_var(194) + (((((dude_charisma * 50) + (dude_endurance * 25) + (get_critter_stat(dude_obj, STAT_ag) * 13) + (dude_strength * 12)) / 100) + (has_trait(TRAIT_PERK, dude_obj, PERK_kama_sutra_perk) * 2) + has_trait(2, dude_obj, 13) + (((global_var(GVAR_PLAYER_SEX_LEVEL) >= 100) or (global_var(GVAR_NEW_RENO_PORN_STAR) > 0)) * 2 * (dude_obj == dude_obj))) * 1));
+        if (global_var(GVAR_SEX_COUNTER) >= 9) then begin
+            set_global_var(GVAR_SEXPERT, 1);
         end
-        if (((((get_critter_stat(dude_obj, 3) * 50) + (get_critter_stat(dude_obj, 2) * 25) + (get_critter_stat(dude_obj, 5) * 13) + (get_critter_stat(dude_obj, 0) * 12)) / 100) + (has_trait(0, dude_obj, 94) * 2) + has_trait(2, dude_obj, 13) + (((global_var(194) >= 100) or (global_var(232) > 0)) * 2 * (dude_obj == dude_obj))) >= 9) then begin
-            set_global_var(589, 1);
+        if (((((dude_charisma * 50) + (dude_endurance * 25) + (get_critter_stat(dude_obj, STAT_ag) * 13) + (dude_strength * 12)) / 100) + (has_trait(TRAIT_PERK, dude_obj, PERK_kama_sutra_perk) * 2) + has_trait(2, dude_obj, 13) + (((global_var(GVAR_PLAYER_SEX_LEVEL) >= 100) or (global_var(GVAR_NEW_RENO_PORN_STAR) > 0)) * 2 * (dude_obj == dude_obj))) >= 9) then begin
+            set_global_var(GVAR_GIGALO, 1);
         end
-        if (global_var(590) == 1) then begin
-            set_global_var(590, 0);
+        if (global_var(GVAR_DUDE_VIRGIN) == 1) then begin
+            set_global_var(GVAR_DUDE_VIRGIN, 0);
         end
-        set_global_var(358, global_var(358) + 1);
+        set_global_var(GVAR_SEX_COUNTER, global_var(GVAR_SEX_COUNTER) + 1);
         set_local_var(8, 2);
-        gfade_in(1);
+        fade_in;
     end
 end
 
 procedure destroy_p_proc
 begin
     if (source_obj == dude_obj) then begin
-        set_global_var(5, global_var(5) + 1);
-        if (metarule(51, self_obj) == 2) then begin
+        set_global_var(GVAR_GOOD_MONSTER, global_var(GVAR_GOOD_MONSTER) + 1);
+        if (critter_kill_type(self_obj) == KILL_TYPE_children_kills) then begin
             if (source_obj == dude_obj) then begin
-                set_global_var(2, 0);
-                set_global_var(1, global_var(1) + 1);
-                set_global_var(0, global_var(0) + -15);
+                set_global_var(GVAR_CHAMPION_REPUTATION, 0);
+                set_global_var(GVAR_CHILDKILLER_REPUTATION, global_var(GVAR_CHILDKILLER_REPUTATION) + 1);
+                set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + -15);
                 debug_msg("Player gains " + -15 + " Karma Points.");
-                set_global_var(37, 0);
-                set_global_var(38, 0);
-                set_global_var(39, 0);
-                set_global_var(40, 0);
-                set_global_var(41, 0);
-                set_global_var(42, 0);
-                set_global_var(43, 0);
-                set_global_var(44, 0);
-                set_global_var(45, 0);
-                if (global_var(0) >= 1000) then begin
-                    set_global_var(37, 1);
+                set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+                set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+                set_global_var(GVAR_KARMA_DEFENDER, 0);
+                set_global_var(GVAR_KARMA_WANDERER, 0);
+                set_global_var(GVAR_KARMA_BETRAYER, 0);
+                set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+                set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+                set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+                    set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
                 end
                 else begin
-                    if (global_var(0) >= 750) then begin
-                        set_global_var(38, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
                     end
                     else begin
-                        if (global_var(0) >= 500) then begin
-                            set_global_var(39, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                            set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                         end
                         else begin
-                            if (global_var(0) >= 250) then begin
-                                set_global_var(40, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                                set_global_var(GVAR_KARMA_DEFENDER, 1);
                             end
                             else begin
-                                if (global_var(0) > -250) then begin
-                                    set_global_var(41, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                                    set_global_var(GVAR_KARMA_WANDERER, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -500) then begin
-                                        set_global_var(42, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                        set_global_var(GVAR_KARMA_BETRAYER, 1);
                                     end
                                     else begin
-                                        if (global_var(0) > -750) then begin
-                                            set_global_var(43, 1);
+                                        if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                            set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                         end
                                         else begin
-                                            if (global_var(0) > -1000) then begin
-                                                set_global_var(44, 1);
+                                            if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                                set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                             end
                                             else begin
-                                                set_global_var(45, 1);
+                                                set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                             end
                                         end
                                     end
@@ -301,58 +301,58 @@ begin
                 end
             end
         end
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(4) > (3 * global_var(5))) or (global_var(2) == 1)) and (global_var(1) == 0)) then begin
-            set_global_var(2, 1);
-            set_global_var(3, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(4) > (3 * global_var(GVAR_GOOD_MONSTER))) or (global_var(GVAR_CHAMPION_REPUTATION) == 1)) and (global_var(GVAR_CHILDKILLER_REPUTATION) == 0)) then begin
+            set_global_var(GVAR_CHAMPION_REPUTATION, 1);
+            set_global_var(GVAR_BERSERKER_REPUTATION, 0);
         end
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(5) > (2 * global_var(4))) or (global_var(3) == 1))) then begin
-            set_global_var(3, 1);
-            set_global_var(2, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
+            set_global_var(GVAR_BERSERKER_REPUTATION, 1);
+            set_global_var(GVAR_CHAMPION_REPUTATION, 0);
         end
-        set_global_var(0, global_var(0) + -10);
+        set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + -10);
         debug_msg("Player gains " + -10 + " Karma Points.");
-        set_global_var(37, 0);
-        set_global_var(38, 0);
-        set_global_var(39, 0);
-        set_global_var(40, 0);
-        set_global_var(41, 0);
-        set_global_var(42, 0);
-        set_global_var(43, 0);
-        set_global_var(44, 0);
-        set_global_var(45, 0);
-        if (global_var(0) >= 1000) then begin
-            set_global_var(37, 1);
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+        set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+        set_global_var(GVAR_KARMA_DEFENDER, 0);
+        set_global_var(GVAR_KARMA_WANDERER, 0);
+        set_global_var(GVAR_KARMA_BETRAYER, 0);
+        set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+        set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+            set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
         end
         else begin
-            if (global_var(0) >= 750) then begin
-                set_global_var(38, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
             end
             else begin
-                if (global_var(0) >= 500) then begin
-                    set_global_var(39, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                 end
                 else begin
-                    if (global_var(0) >= 250) then begin
-                        set_global_var(40, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                        set_global_var(GVAR_KARMA_DEFENDER, 1);
                     end
                     else begin
-                        if (global_var(0) > -250) then begin
-                            set_global_var(41, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                            set_global_var(GVAR_KARMA_WANDERER, 1);
                         end
                         else begin
-                            if (global_var(0) > -500) then begin
-                                set_global_var(42, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                set_global_var(GVAR_KARMA_BETRAYER, 1);
                             end
                             else begin
-                                if (global_var(0) > -750) then begin
-                                    set_global_var(43, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -1000) then begin
-                                        set_global_var(44, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                     end
                                     else begin
-                                        set_global_var(45, 1);
+                                        set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                     end
                                 end
                             end
@@ -369,18 +369,18 @@ begin
     script_overrides;
     if (local_var(4) == 0) then begin
         if (local_var(7)) then begin
-            display_msg(message_str(629, 119));
+            display_mstr(119);
         end
         else begin
-            display_msg(message_str(629, 100));
+            display_mstr(100);
         end
     end
     else begin
         if (local_var(7)) then begin
-            display_msg(message_str(629, 120));
+            display_mstr(120);
         end
         else begin
-            display_msg(message_str(629, 101));
+            display_mstr(101);
         end
     end
 end
@@ -389,14 +389,14 @@ procedure description_p_proc
 begin
     script_overrides;
     if (local_var(7)) then begin
-        display_msg(message_str(629, 121));
+        display_mstr(121);
     end
     else begin
-        if (get_critter_stat(self_obj, 34) == 0) then begin
-            display_msg(message_str(629, 102));
+        if (self_gender == 0) then begin
+            display_mstr(102);
         end
         else begin
-            display_msg(message_str(629, 210));
+            display_mstr(210);
         end
     end
 end
@@ -417,15 +417,15 @@ procedure map_enter_p_proc
 begin
     Only_Once := 0;
     starttile := tile_num(self_obj);
-    critter_add_trait(self_obj, 1, 6, 169);
-    if (not(metarule(22, 0))) then begin
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 169);
+    if (not(is_loading_game)) then begin
         call chooseType();
     end
     if (local_var(7) == 1) then begin
-        critter_add_trait(self_obj, 1, 5, 11);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_DRUG_ADDICT);
     end
     else begin
-        critter_add_trait(self_obj, 1, 5, 11);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_DRUG_ADDICT);
     end
     add_timer_event(self_obj, 0, 1);
 end
@@ -455,127 +455,127 @@ end
 
 procedure Node001A
 begin
-    gsay_reply(629, 122);
-    giq_option(4, 629, 123, Node32Aa, 50);
-    giq_option(4, 629, 124, Node32Ab, 50);
-    giq_option(4, 629, 125, Node32Ac, 50);
-    giq_option(4, 629, 126, Node999, 50);
-    giq_option(-3, 629, 127, Node004D, 50);
-    if (not(metarule(47, 0)) and (local_var(8) == 1)) then begin
-        giq_option(4, 629, 178, NodeDoSexA, 50);
-        giq_option(-3, 629, 179, NodeDoSexA, 50);
+    Reply(122);
+    giq_option(4, NAME, 123, Node32Aa, NEUTRAL_REACTION);
+    giq_option(4, NAME, 124, Node32Ab, NEUTRAL_REACTION);
+    giq_option(4, NAME, 125, Node32Ac, NEUTRAL_REACTION);
+    giq_option(4, NAME, 126, Node999, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 127, Node004D, NEUTRAL_REACTION);
+    if (not(language_filter_is_on) and (local_var(8) == 1)) then begin
+        giq_option(4, NAME, 178, NodeDoSexA, NEUTRAL_REACTION);
+        giq_option(-3, NAME, 179, NodeDoSexA, NEUTRAL_REACTION);
     end
 end
 
 procedure Node002A
 begin
-    gsay_reply(629, 128);
-    giq_option(0, 629, message_str(14, 350), Node999, 50);
+    Reply(128);
+    giq_option(0, NAME, g_mstr(350), Node999, NEUTRAL_REACTION);
 end
 
 procedure Node003A
 begin
-    gsay_reply(629, 130);
-    giq_option(0, 629, message_str(14, 350), Node999, 50);
+    Reply(130);
+    giq_option(0, NAME, g_mstr(350), Node999, NEUTRAL_REACTION);
 end
 
 procedure Node004A
 begin
-    gsay_reply(629, 131);
+    Reply(131);
     if (item_caps_total(dude_obj) >= 1000) then begin
-        giq_option(-3, 629, 132, Node005Aa, 50);
+        giq_option(-3, NAME, 132, Node005Aa, NEUTRAL_REACTION);
     end
-    giq_option(-3, 629, 133, Node999, 50);
+    giq_option(-3, NAME, 133, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node005A
 begin
-    gsay_reply(629, 134);
-    giq_option(0, 629, message_str(14, 350), Node999, 50);
+    Reply(134);
+    giq_option(0, NAME, g_mstr(350), Node999, NEUTRAL_REACTION);
 end
 
 procedure Node001D
 begin
-    gsay_reply(629, 103);
-    giq_option(4, 629, 104, Node62Da, 50);
-    giq_option(4, 629, 105, Node63Da, 50);
-    giq_option(4, 629, 106, Node65Da, 50);
-    giq_option(4, 629, 107, Node999, 50);
-    giq_option(-3, 629, 108, Node005D, 50);
-    if (not(metarule(47, 0)) and (local_var(8) == 1)) then begin
-        giq_option(4, 629, 158, NodeDoSexD, 50);
-        giq_option(-3, 629, 159, NodeDoSexD, 50);
+    Reply(103);
+    giq_option(4, NAME, 104, Node62Da, NEUTRAL_REACTION);
+    giq_option(4, NAME, 105, Node63Da, NEUTRAL_REACTION);
+    giq_option(4, NAME, 106, Node65Da, NEUTRAL_REACTION);
+    giq_option(4, NAME, 107, Node999, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 108, Node005D, NEUTRAL_REACTION);
+    if (not(language_filter_is_on) and (local_var(8) == 1)) then begin
+        giq_option(4, NAME, 158, NodeDoSexD, NEUTRAL_REACTION);
+        giq_option(-3, NAME, 159, NodeDoSexD, NEUTRAL_REACTION);
     end
 end
 
 procedure Node002D
 begin
-    gsay_reply(629, 109);
-    giq_option(4, 629, 110, Node999, 50);
+    Reply(109);
+    giq_option(4, NAME, 110, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node003D
 begin
-    gsay_reply(629, 111);
-    giq_option(4, 629, 112, Node999, 50);
+    Reply(111);
+    giq_option(4, NAME, 112, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node004D
 begin
-    gsay_reply(629, 113);
-    giq_option(4, 629, 114, Node999, 50);
+    Reply(113);
+    giq_option(4, NAME, 114, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node005D
 begin
-    gsay_reply(629, 115);
-    giq_option(0, 629, message_str(14, 350), Node999, 50);
+    Reply(115);
+    giq_option(0, NAME, g_mstr(350), Node999, NEUTRAL_REACTION);
 end
 
 procedure Node006D
 begin
-    gsay_reply(629, 117);
-    giq_option(0, 629, message_str(14, 350), Node999, 50);
+    Reply(117);
+    giq_option(0, NAME, g_mstr(350), Node999, NEUTRAL_REACTION);
 end
 
 procedure NodeFleeing
 begin
-    float_msg(self_obj, message_str(629, 203 + random(0, 2)), 8);
+    float_msg(self_obj, mstr(203 + random(0, 2)), FLOAT_MSG_YELLOW);
 end
 
 procedure NodeFighting
 begin
-    float_msg(self_obj, message_str(629, 200 + random(0, 2)), 2);
+    float_msg(self_obj, mstr(200 + random(0, 2)), FLOAT_MSG_RED);
 end
 
 procedure checkProp
 begin
     if (local_var(7)) then begin
-        if (not(metarule(47, 0))) then begin
+        if (not(language_filter_is_on)) then begin
             if (local_var(8) != 2) then begin
-                float_msg(self_obj, message_str(629, 170 + random(0, 4)), 3);
+                float_msg(self_obj, mstr(170 + random(0, 4)), FLOAT_MSG_GREEN);
                 set_local_var(8, 1);
             end
             else begin
-                float_msg(self_obj, message_str(629, 175 + random(0, 2)), random(0, 12));
+                float_msg(self_obj, mstr(175 + random(0, 2)), random(0, 12));
             end
         end
         else begin
-            float_msg(self_obj, message_str(629, 175), random(0, 12));
+            float_msg(self_obj, mstr(175), random(0, 12));
         end
     end
     else begin
-        if (not(metarule(47, 0))) then begin
+        if (not(language_filter_is_on)) then begin
             if (local_var(8) != 2) then begin
-                float_msg(self_obj, message_str(629, 150 + random(0, 4)), 3);
+                float_msg(self_obj, mstr(150 + random(0, 4)), FLOAT_MSG_GREEN);
                 set_local_var(8, 1);
             end
             else begin
-                float_msg(self_obj, message_str(629, 155 + random(0, 2)), 5);
+                float_msg(self_obj, mstr(155 + random(0, 2)), FLOAT_MSG_PURPLE);
             end
         end
         else begin
-            float_msg(self_obj, message_str(629, 156), 5);
+            float_msg(self_obj, mstr(156), FLOAT_MSG_PURPLE);
         end
     end
 end
@@ -623,14 +623,14 @@ end
 
 procedure NodeDoSexD
 begin
-    if ((((((get_critter_stat(dude_obj, 3) * 50) + (get_critter_stat(dude_obj, 2) * 25) + (get_critter_stat(dude_obj, 5) * 13) + (get_critter_stat(dude_obj, 0) * 12)) / 100) + (has_trait(0, dude_obj, 94) * 2) + has_trait(2, dude_obj, 13) + (((global_var(194) >= 100) or (global_var(232) > 0)) * 2 * (dude_obj == dude_obj))) > ((((get_critter_stat(self_obj, 3) * 50) + (get_critter_stat(self_obj, 2) * 25) + (get_critter_stat(self_obj, 5) * 13) + (get_critter_stat(self_obj, 0) * 12)) / 100) + (has_trait(0, self_obj, 94) * 2) + has_trait(2, self_obj, 13) + (((global_var(194) >= 100) or (global_var(232) > 0)) * 2 * (self_obj == dude_obj)))) or (random(1, 100) <= (10 * get_critter_stat(dude_obj, 3)))) then begin
+    if ((((((dude_charisma * 50) + (dude_endurance * 25) + (get_critter_stat(dude_obj, STAT_ag) * 13) + (dude_strength * 12)) / 100) + (has_trait(TRAIT_PERK, dude_obj, PERK_kama_sutra_perk) * 2) + has_trait(2, dude_obj, 13) + (((global_var(GVAR_PLAYER_SEX_LEVEL) >= 100) or (global_var(GVAR_NEW_RENO_PORN_STAR) > 0)) * 2 * (dude_obj == dude_obj))) > ((((self_charisma * 50) + (self_endurance * 25) + (get_critter_stat(self_obj, STAT_ag) * 13) + (self_strength * 12)) / 100) + (has_trait(TRAIT_PERK, self_obj, PERK_kama_sutra_perk) * 2) + has_trait(2, self_obj, 13) + (((global_var(GVAR_PLAYER_SEX_LEVEL) >= 100) or (global_var(GVAR_NEW_RENO_PORN_STAR) > 0)) * 2 * (self_obj == dude_obj)))) or (random(1, 100) <= (10 * dude_charisma))) then begin
         doSex := 1;
-        gsay_reply(629, 161);
+        Reply(161);
     end
     else begin
-        gsay_reply(629, 160);
+        Reply(160);
     end
-    giq_option(0, 629, message_str(14, 350), Node999, 50);
+    giq_option(0, NAME, g_mstr(350), Node999, NEUTRAL_REACTION);
 end
 
 procedure Node62Da
@@ -638,7 +638,7 @@ begin
     if (item_caps_total(dude_obj) >= 6) then begin
         item_caps_adjust(dude_obj, -6);
         item_caps_adjust(self_obj, 6);
-        add_obj_to_inven(dude_obj, create_object_sid(106, tile_num(self_obj), 0, -1));
+        add_obj_to_inven(dude_obj, create_object(106, tile_num(self_obj), 0));
         call Node002D();
     end
     else begin
@@ -651,7 +651,7 @@ begin
     if (item_caps_total(dude_obj) >= 15) then begin
         item_caps_adjust(dude_obj, -15);
         item_caps_adjust(self_obj, 15);
-        add_obj_to_inven(dude_obj, create_object_sid(124, tile_num(self_obj), 0, -1));
+        add_obj_to_inven(dude_obj, create_object(124, tile_num(self_obj), 0));
         call Node003D();
     end
     else begin
@@ -664,7 +664,7 @@ begin
     if (item_caps_total(dude_obj) >= 30) then begin
         item_caps_adjust(dude_obj, -30);
         item_caps_adjust(self_obj, 30);
-        add_obj_to_inven(dude_obj, create_object_sid(125, tile_num(self_obj), 0, -1));
+        add_obj_to_inven(dude_obj, create_object(125, tile_num(self_obj), 0));
         call Node005D();
     end
     else begin
@@ -674,14 +674,14 @@ end
 
 procedure NodeDoSexA
 begin
-    if ((((((get_critter_stat(dude_obj, 3) * 50) + (get_critter_stat(dude_obj, 2) * 25) + (get_critter_stat(dude_obj, 5) * 13) + (get_critter_stat(dude_obj, 0) * 12)) / 100) + (has_trait(0, dude_obj, 94) * 2) + has_trait(2, dude_obj, 13) + (((global_var(194) >= 100) or (global_var(232) > 0)) * 2 * (dude_obj == dude_obj))) > ((((get_critter_stat(self_obj, 3) * 50) + (get_critter_stat(self_obj, 2) * 25) + (get_critter_stat(self_obj, 5) * 13) + (get_critter_stat(self_obj, 0) * 12)) / 100) + (has_trait(0, self_obj, 94) * 2) + has_trait(2, self_obj, 13) + (((global_var(194) >= 100) or (global_var(232) > 0)) * 2 * (self_obj == dude_obj)))) or (random(1, 100) <= (10 * get_critter_stat(dude_obj, 3)))) then begin
+    if ((((((dude_charisma * 50) + (dude_endurance * 25) + (get_critter_stat(dude_obj, STAT_ag) * 13) + (dude_strength * 12)) / 100) + (has_trait(TRAIT_PERK, dude_obj, PERK_kama_sutra_perk) * 2) + has_trait(2, dude_obj, 13) + (((global_var(GVAR_PLAYER_SEX_LEVEL) >= 100) or (global_var(GVAR_NEW_RENO_PORN_STAR) > 0)) * 2 * (dude_obj == dude_obj))) > ((((self_charisma * 50) + (self_endurance * 25) + (get_critter_stat(self_obj, STAT_ag) * 13) + (self_strength * 12)) / 100) + (has_trait(TRAIT_PERK, self_obj, PERK_kama_sutra_perk) * 2) + has_trait(2, self_obj, 13) + (((global_var(GVAR_PLAYER_SEX_LEVEL) >= 100) or (global_var(GVAR_NEW_RENO_PORN_STAR) > 0)) * 2 * (self_obj == dude_obj)))) or (random(1, 100) <= (10 * dude_charisma))) then begin
         doSex := 1;
-        gsay_reply(629, 181);
+        Reply(181);
     end
     else begin
-        gsay_reply(629, 180);
+        Reply(180);
     end
-    giq_option(0, 629, message_str(14, 350), Node999, 50);
+    giq_option(0, NAME, g_mstr(350), Node999, NEUTRAL_REACTION);
 end
 
 procedure Node32Aa
@@ -690,7 +690,7 @@ begin
     if (item_caps_total(dude_obj) >= 560) then begin
         item_caps_adjust(dude_obj, -560);
         item_caps_adjust(self_obj, 560);
-        LVar0 := create_object_sid(53, tile_num(self_obj), 0, -1);
+        LVar0 := create_object(53, tile_num(self_obj), 0);
         if (LVar0) then begin
             use_obj_on_obj(LVar0, dude_obj);
         end
@@ -707,7 +707,7 @@ begin
     if (item_caps_total(dude_obj) >= 400) then begin
         item_caps_adjust(dude_obj, -400);
         item_caps_adjust(self_obj, 400);
-        LVar0 := create_object_sid(87, tile_num(self_obj), 0, -1);
+        LVar0 := create_object(87, tile_num(self_obj), 0);
         if (LVar0) then begin
             use_obj_on_obj(LVar0, dude_obj);
         end
@@ -724,7 +724,7 @@ begin
     if (item_caps_total(dude_obj) >= 800) then begin
         item_caps_adjust(dude_obj, -800);
         item_caps_adjust(self_obj, 800);
-        LVar0 := create_object_sid(110, tile_num(self_obj), 0, -1);
+        LVar0 := create_object(110, tile_num(self_obj), 0);
         if (LVar0) then begin
             use_obj_on_obj(LVar0, dude_obj);
         end
@@ -740,10 +740,9 @@ begin
     variable LVar0 := 0;
     item_caps_adjust(dude_obj, -1000);
     item_caps_adjust(self_obj, 1000);
-    LVar0 := create_object_sid(53, tile_num(self_obj), 0, -1);
+    LVar0 := create_object(53, tile_num(self_obj), 0);
     if (LVar0) then begin
         use_obj_on_obj(LVar0, dude_obj);
     end
     call Node005A();
 end
-
