diff --git a/Mods/MegaMod/Original/ECTRAPR2.ssl b/Mods/MegaMod/ReDefine/ECTRAPR2.ssl
index 99b8e1d..415bd6a 100644
--- a/Mods/MegaMod/Original/ECTRAPR2.ssl
+++ b/Mods/MegaMod/ReDefine/ECTRAPR2.ssl
@@ -172,14 +172,14 @@ procedure critter_p_proc
 begin
     if ((local_var(5) == 2) and obj_can_see_obj(self_obj, dude_obj)) then begin
         set_local_var(5, 1);
-        attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+        attack(dude_obj);
     end
     if (local_var(8) == 1) then begin
         if (obj_on_screen(self_obj)) then begin
             if (not(anim_busy(self_obj))) then begin
                 reg_anim_func(2, self_obj);
                 reg_anim_func(1, 1);
-                animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(self_obj), rotation_to_tile(tile_num(dude_obj), tile_num(self_obj)), 6), 0);
+                animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(self_obj), rotation_to_tile(dude_tile, tile_num(self_obj)), 6), 0);
                 reg_anim_func(3, 0);
             end
         end
@@ -229,62 +229,62 @@ end
 procedure destroy_p_proc
 begin
     if (source_obj == dude_obj) then begin
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(4) > (3 * global_var(5))) or (global_var(2) == 1)) and (global_var(1) == 0)) then begin
-            set_global_var(2, 1);
-            set_global_var(3, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(4) > (3 * global_var(GVAR_GOOD_MONSTER))) or (global_var(GVAR_CHAMPION_REPUTATION) == 1)) and (global_var(GVAR_CHILDKILLER_REPUTATION) == 0)) then begin
+            set_global_var(GVAR_CHAMPION_REPUTATION, 1);
+            set_global_var(GVAR_BERSERKER_REPUTATION, 0);
         end
-        if (metarule(51, self_obj) == 2) then begin
+        if (critter_kill_type(self_obj) == 2) then begin
             if (source_obj == dude_obj) then begin
-                set_global_var(2, 0);
-                set_global_var(1, global_var(1) + 1);
-                set_global_var(0, global_var(0) + -15);
+                set_global_var(GVAR_CHAMPION_REPUTATION, 0);
+                set_global_var(GVAR_CHILDKILLER_REPUTATION, global_var(GVAR_CHILDKILLER_REPUTATION) + 1);
+                set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + -15);
                 debug_msg("Player gains " + -15 + " Karma Points.");
-                set_global_var(37, 0);
-                set_global_var(38, 0);
-                set_global_var(39, 0);
-                set_global_var(40, 0);
-                set_global_var(41, 0);
-                set_global_var(42, 0);
-                set_global_var(43, 0);
-                set_global_var(44, 0);
-                set_global_var(45, 0);
-                tmp_gen_rep := global_var(0);
-                if (has_trait(0, dude_obj, 95)) then begin
-                    tmp_gen_rep := global_var(0) * 2;
+                set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+                set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+                set_global_var(GVAR_KARMA_DEFENDER, 0);
+                set_global_var(GVAR_KARMA_WANDERER, 0);
+                set_global_var(GVAR_KARMA_BETRAYER, 0);
+                set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+                set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+                set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+                tmp_gen_rep := global_var(GVAR_PLAYER_REPUTATION);
+                if (has_trait(TRAIT_PERK, dude_obj, PERK_karma_beacon_perk)) then begin
+                    tmp_gen_rep := global_var(GVAR_PLAYER_REPUTATION) * 2;
                 end
                 if (tmp_gen_rep >= 1000) then begin
-                    set_global_var(37, 1);
+                    set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
                 end
                 else begin
                     if (tmp_gen_rep >= 750) then begin
-                        set_global_var(38, 1);
+                        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
                     end
                     else begin
                         if (tmp_gen_rep >= 500) then begin
-                            set_global_var(39, 1);
+                            set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                         end
                         else begin
                             if (tmp_gen_rep >= 250) then begin
-                                set_global_var(40, 1);
+                                set_global_var(GVAR_KARMA_DEFENDER, 1);
                             end
                             else begin
                                 if (tmp_gen_rep > -250) then begin
-                                    set_global_var(41, 1);
+                                    set_global_var(GVAR_KARMA_WANDERER, 1);
                                 end
                                 else begin
                                     if (tmp_gen_rep > -500) then begin
-                                        set_global_var(42, 1);
+                                        set_global_var(GVAR_KARMA_BETRAYER, 1);
                                     end
                                     else begin
                                         if (tmp_gen_rep > -750) then begin
-                                            set_global_var(43, 1);
+                                            set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                         end
                                         else begin
                                             if (tmp_gen_rep > -1000) then begin
-                                                set_global_var(44, 1);
+                                                set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                             end
                                             else begin
-                                                set_global_var(45, 1);
+                                                set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                             end
                                         end
                                     end
@@ -295,58 +295,58 @@ begin
                 end
             end
         end
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(5) > (2 * global_var(4))) or (global_var(3) == 1))) then begin
-            set_global_var(3, 1);
-            set_global_var(2, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
+            set_global_var(GVAR_BERSERKER_REPUTATION, 1);
+            set_global_var(GVAR_CHAMPION_REPUTATION, 0);
         end
-        set_global_var(0, global_var(0) + 0);
+        set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + 0);
         debug_msg("Player gains " + 0 + " Karma Points.");
-        set_global_var(37, 0);
-        set_global_var(38, 0);
-        set_global_var(39, 0);
-        set_global_var(40, 0);
-        set_global_var(41, 0);
-        set_global_var(42, 0);
-        set_global_var(43, 0);
-        set_global_var(44, 0);
-        set_global_var(45, 0);
-        tmp_gen_rep := global_var(0);
-        if (has_trait(0, dude_obj, 95)) then begin
-            tmp_gen_rep := global_var(0) * 2;
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+        set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+        set_global_var(GVAR_KARMA_DEFENDER, 0);
+        set_global_var(GVAR_KARMA_WANDERER, 0);
+        set_global_var(GVAR_KARMA_BETRAYER, 0);
+        set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+        set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+        tmp_gen_rep := global_var(GVAR_PLAYER_REPUTATION);
+        if (has_trait(TRAIT_PERK, dude_obj, PERK_karma_beacon_perk)) then begin
+            tmp_gen_rep := global_var(GVAR_PLAYER_REPUTATION) * 2;
         end
         if (tmp_gen_rep >= 1000) then begin
-            set_global_var(37, 1);
+            set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
         end
         else begin
             if (tmp_gen_rep >= 750) then begin
-                set_global_var(38, 1);
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
             end
             else begin
                 if (tmp_gen_rep >= 500) then begin
-                    set_global_var(39, 1);
+                    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                 end
                 else begin
                     if (tmp_gen_rep >= 250) then begin
-                        set_global_var(40, 1);
+                        set_global_var(GVAR_KARMA_DEFENDER, 1);
                     end
                     else begin
                         if (tmp_gen_rep > -250) then begin
-                            set_global_var(41, 1);
+                            set_global_var(GVAR_KARMA_WANDERER, 1);
                         end
                         else begin
                             if (tmp_gen_rep > -500) then begin
-                                set_global_var(42, 1);
+                                set_global_var(GVAR_KARMA_BETRAYER, 1);
                             end
                             else begin
                                 if (tmp_gen_rep > -750) then begin
-                                    set_global_var(43, 1);
+                                    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                 end
                                 else begin
                                     if (tmp_gen_rep > -1000) then begin
-                                        set_global_var(44, 1);
+                                        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                     end
                                     else begin
-                                        set_global_var(45, 1);
+                                        set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                     end
                                 end
                             end
@@ -361,11 +361,11 @@ end
 procedure look_at_p_proc
 begin
     script_overrides;
-    if (get_critter_stat(self_obj, 34) == 0) then begin
-        display_msg(message_str(1355, 100));
+    if (self_gender == 0) then begin
+        display_mstr(100);
     end
     else begin
-        display_msg(message_str(1355, 200));
+        display_mstr(200);
     end
 end
 
@@ -373,10 +373,10 @@ procedure description_p_proc
 begin
     script_overrides;
     if (local_var(4) == 0) then begin
-        display_msg(message_str(1355, 101));
+        display_mstr(101);
     end
     else begin
-        display_msg(message_str(1355, 102));
+        display_mstr(102);
     end
 end
 
@@ -393,8 +393,8 @@ end
 
 procedure map_enter_p_proc
 begin
-    critter_add_trait(self_obj, 1, 6, 133);
-    critter_add_trait(self_obj, 1, 5, 12);
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 133);
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_GENERIC_GUARDS);
     if ((game_time < 0) and (local_var(9) == 0)) then begin
         set_local_var(7, game_time - 1);
         set_local_var(9, 1);
@@ -404,7 +404,7 @@ begin
         if (obj_is_carrying_obj_pid(self_obj, 276) < restock_amt) then begin
             if (80 >= random(1, 100)) then begin
                 if (obj_is_carrying_obj_pid(self_obj, 276) < restock_amt) then begin
-                    restock_obj := create_object_sid(276, 0, 0, -1);
+                    restock_obj := create_object(276, 0, 0);
                     add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 276));
                 end
                 else begin
@@ -427,7 +427,7 @@ begin
                                 end
                                 else begin
                                     if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                        metarule(43, self_obj);
+                                        inven_unwield(self_obj);
                                     end
                                 end
                             end
@@ -445,7 +445,7 @@ begin
         end
         else begin
             if (obj_is_carrying_obj_pid(self_obj, 276) < restock_amt) then begin
-                restock_obj := create_object_sid(276, 0, 0, -1);
+                restock_obj := create_object(276, 0, 0);
                 add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 276));
             end
             else begin
@@ -468,7 +468,7 @@ begin
                             end
                             else begin
                                 if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                    metarule(43, self_obj);
+                                    inven_unwield(self_obj);
                                 end
                             end
                         end
@@ -487,7 +487,7 @@ begin
         if (obj_is_carrying_obj_pid(self_obj, 277) < restock_amt) then begin
             if (60 >= random(1, 100)) then begin
                 if (obj_is_carrying_obj_pid(self_obj, 277) < restock_amt) then begin
-                    restock_obj := create_object_sid(277, 0, 0, -1);
+                    restock_obj := create_object(277, 0, 0);
                     add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 277));
                 end
                 else begin
@@ -510,7 +510,7 @@ begin
                                 end
                                 else begin
                                     if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                        metarule(43, self_obj);
+                                        inven_unwield(self_obj);
                                     end
                                 end
                             end
@@ -528,7 +528,7 @@ begin
         end
         else begin
             if (obj_is_carrying_obj_pid(self_obj, 277) < restock_amt) then begin
-                restock_obj := create_object_sid(277, 0, 0, -1);
+                restock_obj := create_object(277, 0, 0);
                 add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 277));
             end
             else begin
@@ -551,7 +551,7 @@ begin
                             end
                             else begin
                                 if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                    metarule(43, self_obj);
+                                    inven_unwield(self_obj);
                                 end
                             end
                         end
@@ -593,28 +593,28 @@ end
 
 procedure NodeFleeing
 begin
-    float_msg(self_obj, message_str(1355, 136 + random(0, 2)), 8);
+    float_msg(self_obj, mstr(136 + random(0, 2)), FLOAT_MSG_YELLOW);
 end
 
 procedure NodeFighting
 begin
-    float_msg(self_obj, message_str(1355, 130 + random(0, 3)), 2);
+    float_msg(self_obj, mstr(130 + random(0, 3)), FLOAT_MSG_RED);
 end
 
 procedure Node001
 begin
-    gsay_reply(1355, 103);
-    giq_option(4, 1355, 104, Node002, 50);
-    giq_option(4, 1355, 105, Node999, 50);
-    giq_option(-3, 1355, 106, Node007, 50);
+    Reply(103);
+    giq_option(4, NAME, 104, Node002, NEUTRAL_REACTION);
+    giq_option(4, NAME, 105, Node999, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 106, Node007, NEUTRAL_REACTION);
 end
 
 procedure Node002
 begin
-    gsay_reply(1355, 107);
-    giq_option(4, 1355, 108, Node003, 50);
-    giq_option(4, 1355, 109, Node999, 50);
-    giq_option(-3, 1355, 110, Node999, 50);
+    Reply(107);
+    giq_option(4, NAME, 108, Node003, NEUTRAL_REACTION);
+    giq_option(4, NAME, 109, Node999, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 110, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node003
@@ -626,7 +626,7 @@ begin
     gold_gecko_pelt := obj_carrying_pid_obj(self_obj, 277);
     gecko_pelt_num := obj_is_carrying_obj_pid(self_obj, 276);
     gecko_gold_pelt_num := obj_is_carrying_obj_pid(self_obj, 277);
-    LVar0 := has_skill(self_obj, 15) - has_skill(dude_obj, 15);
+    LVar0 := has_skill(self_obj, SKILL_BARTER) - has_skill(dude_obj, SKILL_BARTER);
     if (LVar0 < 0.00000) then begin
         LVar0 := LVar0 * -1;
         LVar1 := 1;
@@ -648,29 +648,29 @@ begin
         golden_price := floor(125 - (125 * LVar0));
     end
     if ((gecko_pelt_num == 0) and (gecko_gold_pelt_num > 0)) then begin
-        LVar2 := message_str(1355, 140) + gecko_gold_pelt_num + message_str(1355, 113);
+        LVar2 := mstr(140) + gecko_gold_pelt_num + mstr(113);
     end
     else begin
         if ((gecko_pelt_num > 0) and (gecko_gold_pelt_num == 0)) then begin
-            LVar2 := message_str(1355, 141) + gecko_pelt_num + message_str(1355, 112);
+            LVar2 := mstr(141) + gecko_pelt_num + mstr(112);
         end
         else begin
             if ((gecko_pelt_num == 0) and (gecko_gold_pelt_num == 0)) then begin
-                LVar2 := message_str(1355, 142);
+                LVar2 := mstr(142);
             end
             else begin
-                LVar2 := message_str(1355, 111) + gecko_pelt_num + message_str(1355, 112) + " " + gecko_gold_pelt_num + message_str(1355, 113);
+                LVar2 := mstr(111) + gecko_pelt_num + mstr(112) + " " + gecko_gold_pelt_num + mstr(113);
             end
         end
     end
-    gsay_reply(1355, LVar2);
+    Reply(LVar2);
     if (gecko_pelt_num > 0) then begin
-        giq_option(4, 1355, message_str(1355, 134) + normal_price, NodeBuyNormal, 50);
+        giq_option(4, NAME, mstr(134) + normal_price, NodeBuyNormal, NEUTRAL_REACTION);
     end
     if (gecko_gold_pelt_num > 0) then begin
-        giq_option(4, 1355, message_str(1355, 135) + golden_price, NodeBuyGolden, 50);
+        giq_option(4, NAME, mstr(135) + golden_price, NodeBuyGolden, NEUTRAL_REACTION);
     end
-    giq_option(4, 1355, 115, Node999, 50);
+    giq_option(4, NAME, 115, Node999, NEUTRAL_REACTION);
     return 0;
     LVar0 := 0;
     LVar1 := 0;
@@ -679,80 +679,80 @@ end
 
 procedure Node004
 begin
-    gsay_reply(1355, 116);
-    giq_option(4, 1355, 117, Node999, 50);
+    Reply(116);
+    giq_option(4, NAME, 117, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node005
 begin
-    if (get_critter_stat(self_obj, 34) == 0) then begin
+    if (self_gender == 0) then begin
         if (party_member_obj(16777380) != 0) then begin
-            gsay_reply(1355, 150);
-            giq_option(4, 1355, 160, Node006a, 50);
-            giq_option(4, 1355, 161, Node999, 50);
+            Reply(150);
+            giq_option(4, NAME, 160, Node006a, NEUTRAL_REACTION);
+            giq_option(4, NAME, 161, Node999, NEUTRAL_REACTION);
         end
         else begin
-            if (get_critter_stat(dude_obj, 34) == 0) then begin
+            if (dude_gender == 0) then begin
                 if (random(1, 10) == 1) then begin
                     if (party_member_obj(16777379) != 0) then begin
-                        gsay_reply(1355, 153);
-                        giq_option(4, 1355, 160, Node006a, 50);
-                        giq_option(4, 1355, 162, Node999, 50);
+                        Reply(153);
+                        giq_option(4, NAME, 160, Node006a, NEUTRAL_REACTION);
+                        giq_option(4, NAME, 162, Node999, NEUTRAL_REACTION);
                     end
                     else begin
-                        gsay_reply(1355, 156);
-                        giq_option(4, 1355, 163, Node006a, 50);
-                        giq_option(4, 1355, 164, Node006a, 50);
-                        giq_option(4, 1355, random(165, 168), Node999, 50);
+                        Reply(156);
+                        giq_option(4, NAME, 163, Node006a, NEUTRAL_REACTION);
+                        giq_option(4, NAME, 164, Node006a, NEUTRAL_REACTION);
+                        giq_option(4, NAME, random(165, 168), Node999, NEUTRAL_REACTION);
                     end
                 end
                 else begin
-                    gsay_reply(1355, 185);
-                    giq_option(4, 1355, 186, Node999, 50);
-                    giq_option(4, 1355, 187, Node999, 50);
+                    Reply(185);
+                    giq_option(4, NAME, 186, Node999, NEUTRAL_REACTION);
+                    giq_option(4, NAME, 187, Node999, NEUTRAL_REACTION);
                 end
             end
             else begin
-                gsay_reply(1355, 154);
-                giq_option(4, 1355, 163, Node006a, 50);
-                giq_option(4, 1355, 164, Node006a, 50);
-                giq_option(4, 1355, random(165, 168), Node999, 50);
+                Reply(154);
+                giq_option(4, NAME, 163, Node006a, NEUTRAL_REACTION);
+                giq_option(4, NAME, 164, Node006a, NEUTRAL_REACTION);
+                giq_option(4, NAME, random(165, 168), Node999, NEUTRAL_REACTION);
             end
         end
     end
     else begin
-        if (get_critter_stat(self_obj, 34) == 1) then begin
+        if (self_gender == 1) then begin
             if (party_member_obj(16777379) != 0) then begin
-                gsay_reply(1355, 151);
-                giq_option(4, 1355, 160, Node006a, 50);
-                giq_option(4, 1355, 162, Node999, 50);
+                Reply(151);
+                giq_option(4, NAME, 160, Node006a, NEUTRAL_REACTION);
+                giq_option(4, NAME, 162, Node999, NEUTRAL_REACTION);
             end
             else begin
-                if (get_critter_stat(dude_obj, 34) == 1) then begin
+                if (dude_gender == 1) then begin
                     if (random(1, 10) == 1) then begin
                         if (party_member_obj(16777380) != 0) then begin
-                            gsay_reply(1355, 152);
-                            giq_option(4, 1355, 161, Node006a, 50);
-                            giq_option(4, 1355, 162, Node999, 50);
+                            Reply(152);
+                            giq_option(4, NAME, 161, Node006a, NEUTRAL_REACTION);
+                            giq_option(4, NAME, 162, Node999, NEUTRAL_REACTION);
                         end
                         else begin
-                            gsay_reply(1355, 155);
-                            giq_option(4, 1355, 163, Node006a, 50);
-                            giq_option(4, 1355, 164, Node006a, 50);
-                            giq_option(4, 1355, random(165, 168), Node999, 50);
+                            Reply(155);
+                            giq_option(4, NAME, 163, Node006a, NEUTRAL_REACTION);
+                            giq_option(4, NAME, 164, Node006a, NEUTRAL_REACTION);
+                            giq_option(4, NAME, random(165, 168), Node999, NEUTRAL_REACTION);
                         end
                     end
                     else begin
-                        gsay_reply(1355, 185);
-                        giq_option(4, 1355, 186, Node999, 50);
-                        giq_option(4, 1355, 187, Node999, 50);
+                        Reply(185);
+                        giq_option(4, NAME, 186, Node999, NEUTRAL_REACTION);
+                        giq_option(4, NAME, 187, Node999, NEUTRAL_REACTION);
                     end
                 end
                 else begin
-                    gsay_reply(1355, 157);
-                    giq_option(4, 1355, 163, Node006a, 50);
-                    giq_option(4, 1355, 169, Node006a, 50);
-                    giq_option(4, 1355, 122, Node999, 50);
+                    Reply(157);
+                    giq_option(4, NAME, 163, Node006a, NEUTRAL_REACTION);
+                    giq_option(4, NAME, 169, Node006a, NEUTRAL_REACTION);
+                    giq_option(4, NAME, 122, Node999, NEUTRAL_REACTION);
                 end
             end
         end
@@ -761,38 +761,38 @@ end
 
 procedure Node006
 begin
-    gfade_in(1);
-    add_obj_to_inven(dude_obj, create_object_sid(276, tile_num(self_obj), 0, -1));
-    add_obj_to_inven(dude_obj, create_object_sid(277, tile_num(self_obj), 0, -1));
+    fade_in;
+    add_obj_to_inven(dude_obj, create_object(276, tile_num(self_obj), 0));
+    add_obj_to_inven(dude_obj, create_object(277, tile_num(self_obj), 0));
     set_local_var(8, 1);
     if (party_member_obj(16777380) != 0) then begin
-        if (get_critter_stat(self_obj, 34) == 0) then begin
-            display_msg(message_str(1355, 171));
+        if (self_gender == 0) then begin
+            display_mstr(171);
         end
         else begin
-            display_msg(message_str(1355, 170));
+            display_mstr(170);
         end
     end
     else begin
         if (party_member_obj(16777379) != 0) then begin
-            if (get_critter_stat(self_obj, 34) == 0) then begin
-                display_msg(message_str(1355, random(173, 174)));
+            if (self_gender == 0) then begin
+                display_mstr(random(173, 174));
             end
             else begin
-                display_msg(message_str(1355, 174));
+                display_mstr(174);
             end
         end
         else begin
-            if (get_critter_stat(self_obj, 34) == 0) then begin
-                if (get_critter_stat(dude_obj, 34) == 0) then begin
-                    display_msg(message_str(1355, random(176, 181)));
+            if (self_gender == 0) then begin
+                if (dude_gender == 0) then begin
+                    display_mstr(random(176, 181));
                 end
                 else begin
-                    display_msg(message_str(1355, random(175, 180)));
+                    display_mstr(random(175, 180));
                 end
             end
             else begin
-                display_msg(message_str(1355, random(176, 180)));
+                display_mstr(random(176, 180));
             end
         end
     end
@@ -800,14 +800,14 @@ end
 
 procedure Node007
 begin
-    add_obj_to_inven(dude_obj, create_object_sid(276, tile_num(self_obj), 0, -1));
-    gsay_reply(1355, 127);
-    giq_option(-3, 1355, 128, Node999, 50);
+    add_obj_to_inven(dude_obj, create_object(276, tile_num(self_obj), 0));
+    Reply(127);
+    giq_option(-3, NAME, 128, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node008
 begin
-    float_msg(self_obj, message_str(1355, 129), 0);
+    float_msg(self_obj, mstr(129), FLOAT_MSG_NORMAL);
 end
 
 procedure Node999
@@ -844,23 +844,23 @@ end
 
 procedure Node006a
 begin
-    gfade_out(1);
+    fade_out;
     playerSoldSex := 1;
     if ((party_member_obj(16777380) != 0) or (party_member_obj(16777379) != 0)) then begin
         debug_msg("you are married.");
     end
     else begin
-        set_global_var(194, global_var(194) + (((((get_critter_stat(dude_obj, 3) * 50) + (get_critter_stat(dude_obj, 2) * 25) + (get_critter_stat(dude_obj, 5) * 13) + (get_critter_stat(dude_obj, 0) * 12)) / 100) + (has_trait(0, dude_obj, 94) * 2) + has_trait(2, dude_obj, 13) + (((global_var(194) >= 100) or (global_var(232) > 0)) * 2 * (dude_obj == dude_obj))) * 1));
-        if (global_var(358) >= 9) then begin
-            set_global_var(589, 1);
+        set_global_var(GVAR_PLAYER_SEX_LEVEL, global_var(194) + (((((dude_charisma * 50) + (dude_endurance * 25) + (get_critter_stat(dude_obj, STAT_ag) * 13) + (dude_strength * 12)) / 100) + (has_trait(TRAIT_PERK, dude_obj, PERK_kama_sutra_perk) * 2) + has_trait(2, dude_obj, 13) + (((global_var(GVAR_PLAYER_SEX_LEVEL) >= 100) or (global_var(GVAR_NEW_RENO_PORN_STAR) > 0)) * 2 * (dude_obj == dude_obj))) * 1));
+        if (global_var(GVAR_SEX_COUNTER) >= 9) then begin
+            set_global_var(GVAR_GIGALO, 1);
         end
-        if (((((get_critter_stat(dude_obj, 3) * 50) + (get_critter_stat(dude_obj, 2) * 25) + (get_critter_stat(dude_obj, 5) * 13) + (get_critter_stat(dude_obj, 0) * 12)) / 100) + (has_trait(0, dude_obj, 94) * 2) + has_trait(2, dude_obj, 13) + (((global_var(194) >= 100) or (global_var(232) > 0)) * 2 * (dude_obj == dude_obj))) >= 9) then begin
-            set_global_var(588, 1);
+        if (((((dude_charisma * 50) + (dude_endurance * 25) + (get_critter_stat(dude_obj, STAT_ag) * 13) + (dude_strength * 12)) / 100) + (has_trait(TRAIT_PERK, dude_obj, PERK_kama_sutra_perk) * 2) + has_trait(2, dude_obj, 13) + (((global_var(GVAR_PLAYER_SEX_LEVEL) >= 100) or (global_var(GVAR_NEW_RENO_PORN_STAR) > 0)) * 2 * (dude_obj == dude_obj))) >= 9) then begin
+            set_global_var(GVAR_SEXPERT, 1);
         end
-        if (global_var(590) == 1) then begin
-            set_global_var(590, 0);
+        if (global_var(GVAR_DUDE_VIRGIN) == 1) then begin
+            set_global_var(GVAR_DUDE_VIRGIN, 0);
         end
-        set_global_var(358, global_var(358) + 1);
+        set_global_var(GVAR_SEX_COUNTER, global_var(GVAR_SEX_COUNTER) + 1);
     end
     call Node999();
 end
@@ -868,4 +868,3 @@ end
 procedure use_obj_on_p_proc
 begin
 end
-
