diff --git a/Mods/MegaMod/Original/ECTRADER.ssl b/Mods/MegaMod/ReDefine/ECTRADER.ssl
index 95bab4c..6812b32 100644
--- a/Mods/MegaMod/Original/ECTRADER.ssl
+++ b/Mods/MegaMod/ReDefine/ECTRADER.ssl
@@ -44,14 +44,14 @@ procedure critter_p_proc
 begin
     if ((local_var(5) == 2) and obj_can_see_obj(self_obj, dude_obj)) then begin
         set_local_var(5, 1);
-        attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+        attack(dude_obj);
     end
     if (local_var(8) == 1) then begin
         if (obj_on_screen(self_obj)) then begin
             if (not(anim_busy(self_obj))) then begin
                 reg_anim_func(2, self_obj);
                 reg_anim_func(1, 1);
-                animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(self_obj), rotation_to_tile(tile_num(dude_obj), tile_num(self_obj)), 6), 0);
+                animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(self_obj), rotation_to_tile(dude_tile, tile_num(self_obj)), 6), 0);
                 reg_anim_func(3, 0);
             end
         end
@@ -71,7 +71,7 @@ end
 procedure talk_p_proc
 begin
     if (local_var(9) == 0) then begin
-        if (global_var(360) == 0) then begin
+        if (global_var(GVAR_RND_SALES_ENCOUNTER) == 0) then begin
             start_gdialog(741, self_obj, 4, -1, -1);
             gsay_start;
             call Node001();
@@ -85,7 +85,7 @@ begin
             gsay_end;
             end_dialogue;
         end
-        set_global_var(360, global_var(360) + 1);
+        set_global_var(GVAR_RND_SALES_ENCOUNTER, global_var(GVAR_RND_SALES_ENCOUNTER) + 1);
     end
     else begin
         if (local_var(9) == 2) then begin
@@ -103,63 +103,63 @@ procedure destroy_p_proc
 begin
     variable LVar0 := 0;
     if ((obj_pid(self_obj) >= 16777358) and (obj_pid(self_obj) <= 16777359)) then begin
-        LVar0 := create_object_sid(1, 0, 0, -1);
+        LVar0 := create_object(1, 0, 0);
         add_obj_to_inven(self_obj, LVar0);
     end
     if (source_obj == dude_obj) then begin
-        set_global_var(5, global_var(5) + 1);
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(4) > (3 * global_var(5))) or (global_var(2) == 1)) and (global_var(1) == 0)) then begin
-            set_global_var(2, 1);
-            set_global_var(3, 0);
+        set_global_var(GVAR_GOOD_MONSTER, global_var(GVAR_GOOD_MONSTER) + 1);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(4) > (3 * global_var(GVAR_GOOD_MONSTER))) or (global_var(GVAR_CHAMPION_REPUTATION) == 1)) and (global_var(GVAR_CHILDKILLER_REPUTATION) == 0)) then begin
+            set_global_var(GVAR_CHAMPION_REPUTATION, 1);
+            set_global_var(GVAR_BERSERKER_REPUTATION, 0);
         end
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(5) > (2 * global_var(4))) or (global_var(3) == 1))) then begin
-            set_global_var(3, 1);
-            set_global_var(2, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
+            set_global_var(GVAR_BERSERKER_REPUTATION, 1);
+            set_global_var(GVAR_CHAMPION_REPUTATION, 0);
         end
-        set_global_var(0, global_var(0) + -10);
+        set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + -10);
         debug_msg("Player gains " + -10 + " Karma Points.");
-        set_global_var(37, 0);
-        set_global_var(38, 0);
-        set_global_var(39, 0);
-        set_global_var(40, 0);
-        set_global_var(41, 0);
-        set_global_var(42, 0);
-        set_global_var(43, 0);
-        set_global_var(44, 0);
-        set_global_var(45, 0);
-        if (global_var(0) >= 1000) then begin
-            set_global_var(37, 1);
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+        set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+        set_global_var(GVAR_KARMA_DEFENDER, 0);
+        set_global_var(GVAR_KARMA_WANDERER, 0);
+        set_global_var(GVAR_KARMA_BETRAYER, 0);
+        set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+        set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+            set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
         end
         else begin
-            if (global_var(0) >= 750) then begin
-                set_global_var(38, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
             end
             else begin
-                if (global_var(0) >= 500) then begin
-                    set_global_var(39, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                 end
                 else begin
-                    if (global_var(0) >= 250) then begin
-                        set_global_var(40, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                        set_global_var(GVAR_KARMA_DEFENDER, 1);
                     end
                     else begin
-                        if (global_var(0) > -250) then begin
-                            set_global_var(41, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                            set_global_var(GVAR_KARMA_WANDERER, 1);
                         end
                         else begin
-                            if (global_var(0) > -500) then begin
-                                set_global_var(42, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                set_global_var(GVAR_KARMA_BETRAYER, 1);
                             end
                             else begin
-                                if (global_var(0) > -750) then begin
-                                    set_global_var(43, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -1000) then begin
-                                        set_global_var(44, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                     end
                                     else begin
-                                        set_global_var(45, 1);
+                                        set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                     end
                                 end
                             end
@@ -169,8 +169,8 @@ begin
             end
         end
     end
-    set_global_var(359, random(143, 151));
-    set_global_var(360, 0);
+    set_global_var(GVAR_RND_SALES_NAME, random(143, 151));
+    set_global_var(GVAR_RND_SALES_ENCOUNTER, 0);
     set_map_var(2, map_var(2) + 1);
     return 0;
     LVar0 := 0;
@@ -179,17 +179,17 @@ end
 procedure look_at_p_proc
 begin
     script_overrides;
-    display_msg(message_str(741, 100));
+    display_mstr(100);
 end
 
 procedure description_p_proc
 begin
     script_overrides;
     if (local_var(4) == 0) then begin
-        display_msg(message_str(741, 101));
+        display_mstr(101);
     end
     else begin
-        display_msg(message_str(741, global_var(359)));
+        display_mstr(global_var(GVAR_RND_SALES_NAME));
     end
 end
 
@@ -206,16 +206,16 @@ end
 
 procedure map_enter_p_proc
 begin
-    critter_add_trait(self_obj, 1, 6, 164);
-    critter_add_trait(self_obj, 1, 5, 12);
-    if (not(metarule(22, 0))) then begin
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 164);
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_GENERIC_GUARDS);
+    if (not(is_loading_game)) then begin
         call ChooseItem();
         if (local_var(7) < game_time) then begin
             restock_amt := random(1, 5);
             if (obj_is_carrying_obj_pid(self_obj, local_var(10)) < restock_amt) then begin
                 if (100 >= random(1, 100)) then begin
                     if (obj_is_carrying_obj_pid(self_obj, local_var(10)) < restock_amt) then begin
-                        restock_obj := create_object_sid(local_var(10), 0, 0, -1);
+                        restock_obj := create_object(local_var(10), 0, 0);
                         add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, local_var(10)));
                     end
                     else begin
@@ -228,17 +228,17 @@ begin
                                 restock_obj := obj_carrying_pid_obj(self_obj, local_var(10));
                                 if (obj_type(self_obj) == 1) then begin
                                     restock_obj := obj_carrying_pid_obj(self_obj, local_var(10));
-                                    if (critter_inven_obj(self_obj, 0) == restock_obj) then begin
-                                        if (obj_item_subtype(critter_inven_obj(self_obj, 0)) == 0) then begin
-                                            restock_obj := critter_inven_obj(self_obj, 0);
+                                    if (critter_inven_obj(self_obj, INVEN_TYPE_WORN) == restock_obj) then begin
+                                        if (obj_item_subtype(critter_inven_obj(self_obj, INVEN_TYPE_WORN)) == 0) then begin
+                                            restock_obj := critter_inven_obj(self_obj, INVEN_TYPE_WORN);
                                             debug_msg("armour pid == " + obj_pid(restock_obj));
                                             rm_obj_from_inven(self_obj, restock_obj);
                                             add_obj_to_inven(self_obj, restock_obj);
                                         end
                                     end
                                     else begin
-                                        if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                            metarule(43, self_obj);
+                                        if ((critter_inven_obj(self_obj, INVEN_TYPE_LEFT_HAND) == restock_obj) or (critter_inven_obj(self_obj, INVEN_TYPE_RIGHT_HAND) == restock_obj)) then begin
+                                            inven_unwield(self_obj);
                                         end
                                     end
                                 end
@@ -256,7 +256,7 @@ begin
             end
             else begin
                 if (obj_is_carrying_obj_pid(self_obj, local_var(10)) < restock_amt) then begin
-                    restock_obj := create_object_sid(local_var(10), 0, 0, -1);
+                    restock_obj := create_object(local_var(10), 0, 0);
                     add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, local_var(10)));
                 end
                 else begin
@@ -269,17 +269,17 @@ begin
                             restock_obj := obj_carrying_pid_obj(self_obj, local_var(10));
                             if (obj_type(self_obj) == 1) then begin
                                 restock_obj := obj_carrying_pid_obj(self_obj, local_var(10));
-                                if (critter_inven_obj(self_obj, 0) == restock_obj) then begin
-                                    if (obj_item_subtype(critter_inven_obj(self_obj, 0)) == 0) then begin
-                                        restock_obj := critter_inven_obj(self_obj, 0);
+                                if (critter_inven_obj(self_obj, INVEN_TYPE_WORN) == restock_obj) then begin
+                                    if (obj_item_subtype(critter_inven_obj(self_obj, INVEN_TYPE_WORN)) == 0) then begin
+                                        restock_obj := critter_inven_obj(self_obj, INVEN_TYPE_WORN);
                                         debug_msg("armour pid == " + obj_pid(restock_obj));
                                         rm_obj_from_inven(self_obj, restock_obj);
                                         add_obj_to_inven(self_obj, restock_obj);
                                     end
                                 end
                                 else begin
-                                    if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                        metarule(43, self_obj);
+                                    if ((critter_inven_obj(self_obj, INVEN_TYPE_LEFT_HAND) == restock_obj) or (critter_inven_obj(self_obj, INVEN_TYPE_RIGHT_HAND) == restock_obj)) then begin
+                                        inven_unwield(self_obj);
                                     end
                                 end
                             end
@@ -297,8 +297,8 @@ begin
             set_local_var(7, (random(6, 12) * (24 * (60 * (60 * 10)))) + game_time);
         end
     end
-    if (global_var(359) == 0) then begin
-        set_global_var(359, random(143, 151));
+    if (global_var(GVAR_RND_SALES_NAME) == 0) then begin
+        set_global_var(GVAR_RND_SALES_NAME, random(143, 151));
     end
 end
 
@@ -325,7 +325,7 @@ end
 
 procedure NodeFleeing
 begin
-    float_msg(self_obj, message_str(741, 138 + nodeFCount), 8);
+    float_msg(self_obj, mstr(138 + nodeFCount), FLOAT_MSG_YELLOW);
     nodeFCount := nodeFCount + 1;
     if (nodeFCount > 4) then begin
         nodeFCount := 4;
@@ -334,20 +334,20 @@ end
 
 procedure NodeFighting
 begin
-    float_msg(self_obj, message_str(741, 135 + random(0, 2)), 2);
+    float_msg(self_obj, mstr(135 + random(0, 2)), FLOAT_MSG_RED);
 end
 
 procedure Node001
 begin
-    gsay_reply(741, 103);
-    giq_option(1, 741, 104, Node002a, 50);
+    Reply(103);
+    giq_option(1, NAME, 104, Node002a, NEUTRAL_REACTION);
 end
 
 procedure Node002
 begin
-    gsay_reply(741, 105);
-    giq_option(4, 741, 106, Node003, 50);
-    giq_option(-3, 741, 107, Node006, 50);
+    Reply(105);
+    giq_option(4, NAME, 106, Node003, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 107, Node006, NEUTRAL_REACTION);
 end
 
 procedure Node003
@@ -381,9 +381,9 @@ begin
             end
         end
     end
-    gsay_reply(741, message_str(741, LVar0) + local_var(12) + message_str(741, 200));
-    giq_option(4, 741, 114, Node0405a, 50);
-    giq_option(4, 741, 115, Node005, 50);
+    Reply(mstr(LVar0) + local_var(12) + mstr(200));
+    giq_option(4, NAME, 114, Node0405a, NEUTRAL_REACTION);
+    giq_option(4, NAME, 115, Node005, NEUTRAL_REACTION);
     return 0;
     LVar0 := 0;
 end
@@ -399,8 +399,8 @@ begin
         rm_obj_from_inven(self_obj, LVar0);
         add_obj_to_inven(dude_obj, LVar0);
     end
-    gsay_reply(741, 116);
-    giq_option(1, 741, 117, Node999, 50);
+    Reply(116);
+    giq_option(1, NAME, 117, Node999, NEUTRAL_REACTION);
     return 0;
     LVar0 := 0;
 end
@@ -409,45 +409,45 @@ procedure Node005
 begin
     set_local_var(9, 2);
     set_local_var(8, 1);
-    gsay_reply(741, 118);
-    giq_option(1, 741, 119, Node999, 50);
+    Reply(118);
+    giq_option(1, NAME, 119, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node006
 begin
-    gsay_reply(741, message_str(741, 120) + local_var(12) + message_str(741, 152));
-    giq_option(-3, 741, 121, Node004, 50);
-    giq_option(-3, 741, 122, Node005, 50);
+    Reply(mstr(120) + local_var(12) + mstr(152));
+    giq_option(-3, NAME, 121, Node004, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 122, Node005, NEUTRAL_REACTION);
 end
 
 procedure Node007
 begin
-    gsay_reply(741, 123);
-    giq_option(4, 741, 124, Node003, 50);
-    giq_option(4, 741, 125, Node005, 50);
-    giq_option(-3, 741, 126, Node004, 50);
-    giq_option(-3, 741, 127, Node005, 50);
+    Reply(123);
+    giq_option(4, NAME, 124, Node003, NEUTRAL_REACTION);
+    giq_option(4, NAME, 125, Node005, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 126, Node004, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 127, Node005, NEUTRAL_REACTION);
 end
 
 procedure Node008
 begin
-    float_msg(self_obj, message_str(741, 128 + random(0, 2)), 0);
+    float_msg(self_obj, mstr(128 + random(0, 2)), FLOAT_MSG_NORMAL);
 end
 
 procedure Node009
 begin
-    float_msg(self_obj, message_str(741, 131 + random(0, 3)), 0);
+    float_msg(self_obj, mstr(131 + random(0, 3)), FLOAT_MSG_NORMAL);
 end
 
 procedure ChooseItem
 begin
     variable LVar0 := 0;
     variable LVar1 := 0;
-    LVar1 := 11 - get_critter_stat(dude_obj, 6);
+    LVar1 := 11 - dude_luck;
     while (local_var(10) == 0) do begin
         LVar0 := random(1, 25);
         if (LVar0 == 1) then begin
-            if ((get_pc_stat(1) >= 1) and (get_pc_stat(1) <= 3)) then begin
+            if ((get_pc_stat(PCSTAT_level) >= 1) and (get_pc_stat(PCSTAT_level) <= 3)) then begin
                 set_local_var(10, 4);
                 set_local_var(11, 1);
                 set_local_var(12, 40 * LVar1);
@@ -455,7 +455,7 @@ begin
         end
         else begin
             if (LVar0 == 2) then begin
-                if ((get_pc_stat(1) >= 3) and (get_pc_stat(1) <= 5)) then begin
+                if ((get_pc_stat(PCSTAT_level) >= 3) and (get_pc_stat(PCSTAT_level) <= 5)) then begin
                     set_local_var(10, 236);
                     set_local_var(11, 1);
                     set_local_var(12, 165 * LVar1);
@@ -463,7 +463,7 @@ begin
             end
             else begin
                 if (LVar0 == 3) then begin
-                    if ((get_pc_stat(1) >= 1) and (get_pc_stat(1) <= 5)) then begin
+                    if ((get_pc_stat(PCSTAT_level) >= 1) and (get_pc_stat(PCSTAT_level) <= 5)) then begin
                         set_local_var(10, 45);
                         set_local_var(11, 1);
                         set_local_var(12, 100 * LVar1);
@@ -471,7 +471,7 @@ begin
                 end
                 else begin
                     if (LVar0 == 4) then begin
-                        if ((get_pc_stat(1) >= 8) and (get_pc_stat(1) <= 1000)) then begin
+                        if ((get_pc_stat(PCSTAT_level) >= 8) and (get_pc_stat(PCSTAT_level) <= 1000)) then begin
                             set_local_var(10, 116);
                             set_local_var(11, 1);
                             set_local_var(12, 900 * LVar1);
@@ -479,7 +479,7 @@ begin
                     end
                     else begin
                         if (LVar0 == 5) then begin
-                            if ((get_pc_stat(1) >= 1) and (get_pc_stat(1) <= 5)) then begin
+                            if ((get_pc_stat(PCSTAT_level) >= 1) and (get_pc_stat(PCSTAT_level) <= 5)) then begin
                                 set_local_var(10, 8);
                                 set_local_var(11, 2);
                                 set_local_var(12, 250 * LVar1);
@@ -487,7 +487,7 @@ begin
                         end
                         else begin
                             if (LVar0 == 6) then begin
-                                if ((get_pc_stat(1) >= 3) and (get_pc_stat(1) <= 5)) then begin
+                                if ((get_pc_stat(PCSTAT_level) >= 3) and (get_pc_stat(PCSTAT_level) <= 5)) then begin
                                     set_local_var(10, 18);
                                     set_local_var(11, 2);
                                     set_local_var(12, 800 * LVar1);
@@ -495,7 +495,7 @@ begin
                             end
                             else begin
                                 if (LVar0 == 7) then begin
-                                    if ((get_pc_stat(1) >= 6) and (get_pc_stat(1) <= 10)) then begin
+                                    if ((get_pc_stat(PCSTAT_level) >= 6) and (get_pc_stat(PCSTAT_level) <= 10)) then begin
                                         set_local_var(10, 9);
                                         set_local_var(11, 2);
                                         set_local_var(12, 1000 * LVar1);
@@ -503,7 +503,7 @@ begin
                                 end
                                 else begin
                                     if (LVar0 == 8) then begin
-                                        if ((get_pc_stat(1) >= 6) and (get_pc_stat(1) <= 1000)) then begin
+                                        if ((get_pc_stat(PCSTAT_level) >= 6) and (get_pc_stat(PCSTAT_level) <= 1000)) then begin
                                             set_local_var(10, 22);
                                             set_local_var(11, 2);
                                             set_local_var(12, 1100 * LVar1);
@@ -511,7 +511,7 @@ begin
                                     end
                                     else begin
                                         if (LVar0 == 9) then begin
-                                            if ((get_pc_stat(1) >= 1) and (get_pc_stat(1) <= 3)) then begin
+                                            if ((get_pc_stat(PCSTAT_level) >= 1) and (get_pc_stat(PCSTAT_level) <= 3)) then begin
                                                 set_local_var(10, 19);
                                                 set_local_var(11, 3);
                                                 set_local_var(12, 1 * LVar1);
@@ -519,7 +519,7 @@ begin
                                         end
                                         else begin
                                             if (LVar0 == 10) then begin
-                                                if ((get_pc_stat(1) >= 1) and (get_pc_stat(1) <= 3)) then begin
+                                                if ((get_pc_stat(PCSTAT_level) >= 1) and (get_pc_stat(PCSTAT_level) <= 3)) then begin
                                                     set_local_var(10, 7);
                                                     set_local_var(11, 3);
                                                     set_local_var(12, 80 * LVar1);
@@ -527,7 +527,7 @@ begin
                                             end
                                             else begin
                                                 if (LVar0 == 11) then begin
-                                                    if ((get_pc_stat(1) >= 1) and (get_pc_stat(1) <= 10)) then begin
+                                                    if ((get_pc_stat(PCSTAT_level) >= 1) and (get_pc_stat(PCSTAT_level) <= 10)) then begin
                                                         set_local_var(10, 320);
                                                         set_local_var(11, 3);
                                                         set_local_var(12, 4 * LVar1);
@@ -535,7 +535,7 @@ begin
                                                 end
                                                 else begin
                                                     if (LVar0 == 12) then begin
-                                                        if ((get_pc_stat(1) >= 1) and (get_pc_stat(1) <= 1000)) then begin
+                                                        if ((get_pc_stat(PCSTAT_level) >= 1) and (get_pc_stat(PCSTAT_level) <= 1000)) then begin
                                                             set_local_var(10, 280);
                                                             set_local_var(11, 3);
                                                             set_local_var(12, 100 * LVar1);
@@ -543,7 +543,7 @@ begin
                                                     end
                                                     else begin
                                                         if (LVar0 == 13) then begin
-                                                            if ((get_pc_stat(1) >= 6) and (get_pc_stat(1) <= 1000)) then begin
+                                                            if ((get_pc_stat(PCSTAT_level) >= 6) and (get_pc_stat(PCSTAT_level) <= 1000)) then begin
                                                                 set_local_var(10, 160);
                                                                 set_local_var(11, 4);
                                                                 set_local_var(12, 600 * LVar1);
@@ -551,7 +551,7 @@ begin
                                                         end
                                                         else begin
                                                             if (LVar0 == 14) then begin
-                                                                if ((get_pc_stat(1) >= 1) and (get_pc_stat(1) <= 5)) then begin
+                                                                if ((get_pc_stat(PCSTAT_level) >= 1) and (get_pc_stat(PCSTAT_level) <= 5)) then begin
                                                                     set_local_var(10, 20);
                                                                     set_local_var(11, 4);
                                                                     set_local_var(12, 65 * LVar1);
@@ -559,7 +559,7 @@ begin
                                                             end
                                                             else begin
                                                                 if (LVar0 == 15) then begin
-                                                                    if ((get_pc_stat(1) >= 1) and (get_pc_stat(1) <= 5)) then begin
+                                                                    if ((get_pc_stat(PCSTAT_level) >= 1) and (get_pc_stat(PCSTAT_level) <= 5)) then begin
                                                                         set_local_var(10, 5);
                                                                         set_local_var(11, 4);
                                                                         set_local_var(12, 30 * LVar1);
@@ -567,7 +567,7 @@ begin
                                                                 end
                                                                 else begin
                                                                     if (LVar0 == 16) then begin
-                                                                        if ((get_pc_stat(1) >= 1) and (get_pc_stat(1) <= 1000)) then begin
+                                                                        if ((get_pc_stat(PCSTAT_level) >= 1) and (get_pc_stat(PCSTAT_level) <= 1000)) then begin
                                                                             set_local_var(10, 234);
                                                                             set_local_var(11, 4);
                                                                             set_local_var(12, 250 * LVar1);
@@ -575,7 +575,7 @@ begin
                                                                     end
                                                                     else begin
                                                                         if (LVar0 == 17) then begin
-                                                                            if ((get_pc_stat(1) >= 1) and (get_pc_stat(1) <= 5)) then begin
+                                                                            if ((get_pc_stat(PCSTAT_level) >= 1) and (get_pc_stat(PCSTAT_level) <= 5)) then begin
                                                                                 set_local_var(10, 21);
                                                                                 set_local_var(11, 4);
                                                                                 set_local_var(12, 40 * LVar1);
@@ -583,7 +583,7 @@ begin
                                                                         end
                                                                         else begin
                                                                             if (LVar0 == 18) then begin
-                                                                                if ((get_pc_stat(1) >= 10) and (get_pc_stat(1) <= 1000)) then begin
+                                                                                if ((get_pc_stat(PCSTAT_level) >= 10) and (get_pc_stat(PCSTAT_level) <= 1000)) then begin
                                                                                     set_local_var(10, 235);
                                                                                     set_local_var(11, 4);
                                                                                     set_local_var(12, 2200 * LVar1);
@@ -591,7 +591,7 @@ begin
                                                                             end
                                                                             else begin
                                                                                 if (LVar0 == 19) then begin
-                                                                                    if ((get_pc_stat(1) >= 3) and (get_pc_stat(1) <= 5)) then begin
+                                                                                    if ((get_pc_stat(PCSTAT_level) >= 3) and (get_pc_stat(PCSTAT_level) <= 5)) then begin
                                                                                         set_local_var(10, 94);
                                                                                         set_local_var(11, 5);
                                                                                         set_local_var(12, 800 * LVar1);
@@ -599,7 +599,7 @@ begin
                                                                                 end
                                                                                 else begin
                                                                                     if (LVar0 == 20) then begin
-                                                                                        if ((get_pc_stat(1) >= 3) and (get_pc_stat(1) <= 5)) then begin
+                                                                                        if ((get_pc_stat(PCSTAT_level) >= 3) and (get_pc_stat(PCSTAT_level) <= 5)) then begin
                                                                                             set_local_var(10, 10);
                                                                                             set_local_var(11, 5);
                                                                                             set_local_var(12, 1000 * LVar1);
@@ -607,7 +607,7 @@ begin
                                                                                     end
                                                                                     else begin
                                                                                         if (LVar0 == 21) then begin
-                                                                                            if ((get_pc_stat(1) >= 11) and (get_pc_stat(1) <= 15)) then begin
+                                                                                            if ((get_pc_stat(PCSTAT_level) >= 11) and (get_pc_stat(PCSTAT_level) <= 15)) then begin
                                                                                                 set_local_var(10, 23);
                                                                                                 set_local_var(11, 5);
                                                                                                 set_local_var(12, 1300 * LVar1);
@@ -615,7 +615,7 @@ begin
                                                                                         end
                                                                                         else begin
                                                                                             if (LVar0 == 22) then begin
-                                                                                                if ((get_pc_stat(1) >= 15) and (get_pc_stat(1) <= 20)) then begin
+                                                                                                if ((get_pc_stat(PCSTAT_level) >= 15) and (get_pc_stat(PCSTAT_level) <= 20)) then begin
                                                                                                     set_local_var(10, 12);
                                                                                                     set_local_var(11, 5);
                                                                                                     set_local_var(12, 3800 * LVar1);
@@ -623,7 +623,7 @@ begin
                                                                                             end
                                                                                             else begin
                                                                                                 if (LVar0 == 23) then begin
-                                                                                                    if ((get_pc_stat(1) >= 1) and (get_pc_stat(1) <= 5)) then begin
+                                                                                                    if ((get_pc_stat(PCSTAT_level) >= 1) and (get_pc_stat(PCSTAT_level) <= 5)) then begin
                                                                                                         set_local_var(10, 161);
                                                                                                         set_local_var(11, 5);
                                                                                                         set_local_var(12, 200 * LVar1);
@@ -631,7 +631,7 @@ begin
                                                                                                 end
                                                                                                 else begin
                                                                                                     if (LVar0 == 24) then begin
-                                                                                                        if ((get_pc_stat(1) >= 11) and (get_pc_stat(1) <= 15)) then begin
+                                                                                                        if ((get_pc_stat(PCSTAT_level) >= 11) and (get_pc_stat(PCSTAT_level) <= 15)) then begin
                                                                                                             set_local_var(10, 143);
                                                                                                             set_local_var(11, 5);
                                                                                                             set_local_var(12, 2200 * LVar1);
@@ -640,7 +640,7 @@ begin
                                                                                                     else begin
                                                                                                         if (LVar0 == 25) then begin
                                                                                                             if (is_success(do_check(dude_obj, 6, 0))) then begin
-                                                                                                                if ((get_pc_stat(1) >= 13) and (get_pc_stat(1) <= 1000)) then begin
+                                                                                                                if ((get_pc_stat(PCSTAT_level) >= 13) and (get_pc_stat(PCSTAT_level) <= 1000)) then begin
                                                                                                                     set_local_var(10, 120);
                                                                                                                     set_local_var(11, 6);
                                                                                                                     set_local_var(12, 10000 * LVar1);
@@ -683,9 +683,9 @@ end
 
 procedure Node002a
 begin
-    gfade_out(1);
+    fade_out;
     game_time_advance(60 * (60 * 10) * 3);
-    gfade_in(1);
+    fade_in;
     call Node002();
 end
 
@@ -698,4 +698,3 @@ begin
         call Node005();
     end
 end
-
