diff --git a/Mods/MegaMod/Original/ECTRVLR.ssl b/Mods/MegaMod/ReDefine/ECTRVLR.ssl
index 9b99393..fddc3f3 100644
--- a/Mods/MegaMod/Original/ECTRVLR.ssl
+++ b/Mods/MegaMod/ReDefine/ECTRVLR.ssl
@@ -177,7 +177,7 @@ procedure critter_p_proc
 begin
     if ((local_var(5) == 2) and obj_can_see_obj(self_obj, dude_obj)) then begin
         set_local_var(5, 1);
-        attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+        attack(dude_obj);
     end
 end
 
@@ -206,55 +206,55 @@ end
 procedure destroy_p_proc
 begin
     if (source_obj == dude_obj) then begin
-        set_global_var(5, global_var(5) + 1);
-        if (metarule(51, self_obj) == 2) then begin
+        set_global_var(GVAR_GOOD_MONSTER, global_var(GVAR_GOOD_MONSTER) + 1);
+        if (critter_kill_type(self_obj) == KILL_TYPE_children_kills) then begin
             if (source_obj == dude_obj) then begin
-                set_global_var(2, 0);
-                set_global_var(1, global_var(1) + 1);
-                set_global_var(0, global_var(0) + -15);
+                set_global_var(GVAR_CHAMPION_REPUTATION, 0);
+                set_global_var(GVAR_CHILDKILLER_REPUTATION, global_var(GVAR_CHILDKILLER_REPUTATION) + 1);
+                set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + -15);
                 debug_msg("Player gains " + -15 + " Karma Points.");
-                set_global_var(37, 0);
-                set_global_var(38, 0);
-                set_global_var(39, 0);
-                set_global_var(40, 0);
-                set_global_var(41, 0);
-                set_global_var(42, 0);
-                set_global_var(43, 0);
-                set_global_var(44, 0);
-                set_global_var(45, 0);
-                if (global_var(0) >= 1000) then begin
-                    set_global_var(37, 1);
+                set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+                set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+                set_global_var(GVAR_KARMA_DEFENDER, 0);
+                set_global_var(GVAR_KARMA_WANDERER, 0);
+                set_global_var(GVAR_KARMA_BETRAYER, 0);
+                set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+                set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+                set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+                    set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
                 end
                 else begin
-                    if (global_var(0) >= 750) then begin
-                        set_global_var(38, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
                     end
                     else begin
-                        if (global_var(0) >= 500) then begin
-                            set_global_var(39, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                            set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                         end
                         else begin
-                            if (global_var(0) >= 250) then begin
-                                set_global_var(40, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                                set_global_var(GVAR_KARMA_DEFENDER, 1);
                             end
                             else begin
-                                if (global_var(0) > -250) then begin
-                                    set_global_var(41, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                                    set_global_var(GVAR_KARMA_WANDERER, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -500) then begin
-                                        set_global_var(42, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                        set_global_var(GVAR_KARMA_BETRAYER, 1);
                                     end
                                     else begin
-                                        if (global_var(0) > -750) then begin
-                                            set_global_var(43, 1);
+                                        if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                            set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                         end
                                         else begin
-                                            if (global_var(0) > -1000) then begin
-                                                set_global_var(44, 1);
+                                            if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                                set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                             end
                                             else begin
-                                                set_global_var(45, 1);
+                                                set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                             end
                                         end
                                     end
@@ -265,58 +265,58 @@ begin
                 end
             end
         end
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(4) > (3 * global_var(5))) or (global_var(2) == 1)) and (global_var(1) == 0)) then begin
-            set_global_var(2, 1);
-            set_global_var(3, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(4) > (3 * global_var(GVAR_GOOD_MONSTER))) or (global_var(GVAR_CHAMPION_REPUTATION) == 1)) and (global_var(GVAR_CHILDKILLER_REPUTATION) == 0)) then begin
+            set_global_var(GVAR_CHAMPION_REPUTATION, 1);
+            set_global_var(GVAR_BERSERKER_REPUTATION, 0);
         end
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(5) > (2 * global_var(4))) or (global_var(3) == 1))) then begin
-            set_global_var(3, 1);
-            set_global_var(2, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
+            set_global_var(GVAR_BERSERKER_REPUTATION, 1);
+            set_global_var(GVAR_CHAMPION_REPUTATION, 0);
         end
-        set_global_var(0, global_var(0) + -10);
+        set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + -10);
         debug_msg("Player gains " + -10 + " Karma Points.");
-        set_global_var(37, 0);
-        set_global_var(38, 0);
-        set_global_var(39, 0);
-        set_global_var(40, 0);
-        set_global_var(41, 0);
-        set_global_var(42, 0);
-        set_global_var(43, 0);
-        set_global_var(44, 0);
-        set_global_var(45, 0);
-        if (global_var(0) >= 1000) then begin
-            set_global_var(37, 1);
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+        set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+        set_global_var(GVAR_KARMA_DEFENDER, 0);
+        set_global_var(GVAR_KARMA_WANDERER, 0);
+        set_global_var(GVAR_KARMA_BETRAYER, 0);
+        set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+        set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+            set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
         end
         else begin
-            if (global_var(0) >= 750) then begin
-                set_global_var(38, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
             end
             else begin
-                if (global_var(0) >= 500) then begin
-                    set_global_var(39, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                 end
                 else begin
-                    if (global_var(0) >= 250) then begin
-                        set_global_var(40, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                        set_global_var(GVAR_KARMA_DEFENDER, 1);
                     end
                     else begin
-                        if (global_var(0) > -250) then begin
-                            set_global_var(41, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                            set_global_var(GVAR_KARMA_WANDERER, 1);
                         end
                         else begin
-                            if (global_var(0) > -500) then begin
-                                set_global_var(42, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                set_global_var(GVAR_KARMA_BETRAYER, 1);
                             end
                             else begin
-                                if (global_var(0) > -750) then begin
-                                    set_global_var(43, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -1000) then begin
-                                        set_global_var(44, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                     end
                                     else begin
-                                        set_global_var(45, 1);
+                                        set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                     end
                                 end
                             end
@@ -331,7 +331,7 @@ end
 procedure look_at_p_proc
 begin
     script_overrides;
-    display_msg(message_str(739, 100));
+    display_msg(message_str(SCRIPT_ECTRAVLR, 100));
 end
 
 procedure description_p_proc
@@ -353,8 +353,8 @@ end
 procedure map_enter_p_proc
 begin
     Only_Once := 0;
-    critter_add_trait(self_obj, 1, 6, 163);
-    critter_add_trait(self_obj, 1, 5, 12);
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 163);
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_GENERIC_GUARDS);
     one := -1;
     two := -2;
 end
@@ -384,11 +384,11 @@ end
 
 procedure Node001
 begin
-    gsay_reply(739, 101);
-    giq_option(4, 739, 102, Node002, 50);
-    giq_option(4, 739, 103, Node010, 50);
-    giq_option(4, 739, 104, Node011, 50);
-    giq_option(-3, 739, 105, Node999, 50);
+    gsay_reply(SCRIPT_ECTRAVLR, 101);
+    giq_option(4, SCRIPT_ECTRAVLR, 102, Node002, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_ECTRAVLR, 103, Node010, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_ECTRAVLR, 104, Node011, NEUTRAL_REACTION);
+    giq_option(-3, SCRIPT_ECTRAVLR, 105, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node002
@@ -401,108 +401,108 @@ begin
         end
         Only_Once := 1;
     end
-    gsay_reply(739, 106);
-    giq_option(4, 739, 107 + one, checkChoiceOne, 50);
-    giq_option(4, 739, 107 + two, checkChoiceTwo, 50);
-    giq_option(4, 739, 114, Node999, 50);
+    gsay_reply(SCRIPT_ECTRAVLR, 106);
+    giq_option(4, SCRIPT_ECTRAVLR, 107 + one, checkChoiceOne, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_ECTRAVLR, 107 + two, checkChoiceTwo, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_ECTRAVLR, 114, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node003
 begin
-    if (metarule(17, 7) == 0) then begin
+    if (town_known(7) == 0) then begin
         debug_msg("  mark_on_map(" + 7 + ")");
         mark_area_known(0, 7, 1);
     end
-    gsay_reply(739, 115);
-    giq_option(4, 739, 150, Node002, 50);
-    giq_option(4, 739, 116, Node999, 50);
+    gsay_reply(SCRIPT_ECTRAVLR, 115);
+    giq_option(4, SCRIPT_ECTRAVLR, 150, Node002, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_ECTRAVLR, 116, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node004
 begin
-    if (metarule(17, 4) == 0) then begin
+    if (town_known(4) == 0) then begin
         debug_msg("  mark_on_map(" + 4 + ")");
         mark_area_known(0, 4, 1);
     end
-    gsay_reply(739, 117);
-    giq_option(4, 739, 150, Node002, 50);
-    giq_option(4, 739, 118, Node999, 50);
+    gsay_reply(SCRIPT_ECTRAVLR, 117);
+    giq_option(4, SCRIPT_ECTRAVLR, 150, Node002, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_ECTRAVLR, 118, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node005
 begin
-    if (metarule(17, 13) == 0) then begin
+    if (town_known(13) == 0) then begin
         debug_msg("  mark_on_map(" + 13 + ")");
         mark_area_known(0, 13, 1);
     end
-    gsay_reply(739, 119);
-    giq_option(4, 739, 150, Node002, 50);
-    giq_option(4, 739, 120, Node999, 50);
+    gsay_reply(SCRIPT_ECTRAVLR, 119);
+    giq_option(4, SCRIPT_ECTRAVLR, 150, Node002, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_ECTRAVLR, 120, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node006
 begin
-    if (metarule(17, 1) == 0) then begin
+    if (town_known(1) == 0) then begin
         debug_msg("  mark_on_map(" + 1 + ")");
         mark_area_known(0, 1, 1);
     end
-    gsay_reply(739, 121);
-    giq_option(4, 739, 150, Node002, 50);
-    giq_option(4, 739, 122, Node999, 50);
+    gsay_reply(SCRIPT_ECTRAVLR, 121);
+    giq_option(4, SCRIPT_ECTRAVLR, 150, Node002, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_ECTRAVLR, 122, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node007
 begin
-    if (metarule(17, 14) == 0) then begin
+    if (town_known(14) == 0) then begin
         debug_msg("  mark_on_map(" + 14 + ")");
         mark_area_known(0, 14, 1);
     end
-    gsay_reply(739, 123);
-    giq_option(4, 739, 150, Node002, 50);
-    giq_option(4, 739, 124, Node999, 50);
+    gsay_reply(SCRIPT_ECTRAVLR, 123);
+    giq_option(4, SCRIPT_ECTRAVLR, 150, Node002, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_ECTRAVLR, 124, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node008
 begin
-    if (metarule(17, 10) == 0) then begin
+    if (town_known(10) == 0) then begin
         debug_msg("  mark_on_map(" + 10 + ")");
         mark_area_known(0, 10, 1);
     end
-    gsay_reply(739, 125);
-    giq_option(4, 739, 150, Node002, 50);
-    giq_option(4, 739, 126, Node999, 50);
+    gsay_reply(SCRIPT_ECTRAVLR, 125);
+    giq_option(4, SCRIPT_ECTRAVLR, 150, Node002, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_ECTRAVLR, 126, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node009
 begin
-    if (metarule(17, 12) == 0) then begin
+    if (town_known(12) == 0) then begin
         debug_msg("  mark_on_map(" + 12 + ")");
         mark_area_known(0, 12, 1);
     end
-    gsay_reply(739, 127);
-    giq_option(4, 739, 150, Node002, 50);
-    giq_option(4, 739, 128, Node999, 50);
+    gsay_reply(SCRIPT_ECTRAVLR, 127);
+    giq_option(4, SCRIPT_ECTRAVLR, 150, Node002, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_ECTRAVLR, 128, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node010
 begin
-    gsay_reply(739, 129);
-    giq_option(4, 739, 130, Node002, 50);
-    giq_option(4, 739, 131, Node011, 50);
-    giq_option(4, 739, 132, Node999, 50);
+    gsay_reply(SCRIPT_ECTRAVLR, 129);
+    giq_option(4, SCRIPT_ECTRAVLR, 130, Node002, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_ECTRAVLR, 131, Node011, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_ECTRAVLR, 132, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node011
 begin
-    gsay_reply(739, 133);
-    giq_option(4, 739, 134, Node002, 50);
-    giq_option(4, 739, 135, Node010, 50);
-    giq_option(4, 739, 136, Node999, 50);
+    gsay_reply(SCRIPT_ECTRAVLR, 133);
+    giq_option(4, SCRIPT_ECTRAVLR, 134, Node002, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_ECTRAVLR, 135, Node010, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_ECTRAVLR, 136, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node012
 begin
-    float_msg(self_obj, message_str(739, 137 + n12count), 0);
+    float_msg(self_obj, message_str(SCRIPT_ECTRAVLR, 137 + n12count), FLOAT_MSG_NORMAL);
     n12count := n12count + 1;
     if (n12count > 3) then begin
         n12count := 0;
@@ -514,12 +514,12 @@ end
 
 procedure NodeFleeing
 begin
-    float_msg(self_obj, message_str(739, 105 + random(0, 3)), 8);
+    float_msg(self_obj, message_str(SCRIPT_ECTRAVLR, 105 + random(0, 3)), FLOAT_MSG_YELLOW);
 end
 
 procedure NodeFighting
 begin
-    float_msg(self_obj, message_str(739, 100 + random(0, 4)), 2);
+    float_msg(self_obj, message_str(SCRIPT_ECTRAVLR, 100 + random(0, 4)), FLOAT_MSG_RED);
 end
 
 procedure checkChoiceOne
@@ -595,4 +595,3 @@ begin
         end
     end
 end
-
