diff --git a/Mods/MegaMod/Original/EILV3DSK.ssl b/Mods/MegaMod/ReDefine/EILV3DSK.ssl
index 6fb7799..51e886c 100644
--- a/Mods/MegaMod/Original/EILV3DSK.ssl
+++ b/Mods/MegaMod/ReDefine/EILV3DSK.ssl
@@ -137,24 +137,23 @@ end
 
 procedure use_p_proc
 begin
-    if (global_var(223) != 1) then begin
-        set_global_var(223, 1);
-        display_msg(message_str(151, 102));
-        if ((global_var(220) == 1) and (global_var(221) == 1) and (global_var(222) == 3) and (global_var(223) == 1) and (global_var(224) == 1)) then begin
-            if (global_var(1076) == 2) then begin
-                display_msg(message_str(1671, 1201));
+    if (global_var(GVAR_HOLODISK_MB_LEVEL_3) != 1) then begin
+        set_global_var(GVAR_HOLODISK_MB_LEVEL_3, 1);
+        display_msg(message_str(SCRIPT_MISCTEXT, 102));
+        if ((global_var(GVAR_HOLODISK_MB_OUTSIDE) == 1) and (global_var(GVAR_HOLODISK_MB_LEVEL_1) == 1) and (global_var(GVAR_HOLODISK_MB_LEVEL_2) == 3) and (global_var(GVAR_HOLODISK_MB_LEVEL_3) == 1) and (global_var(GVAR_HOLODISK_MB_LEVEL_4) == 1)) then begin
+            if (global_var(GVAR_BRTHD_DISK_FROM_DEN) == 2) then begin
+                display_msg(message_str(SCRIPT_BOSBARTN, 1201));
             end
-            set_global_var(1076, 3);
+            set_global_var(GVAR_BRTHD_DISK_FROM_DEN, 3);
         end
     end
     else begin
-        display_msg(message_str(151, 101));
+        display_msg(message_str(SCRIPT_MISCTEXT, 101));
     end
 end
 
 procedure description_p_proc
 begin
     script_overrides;
-    display_msg(message_str(151, 100));
+    display_msg(message_str(SCRIPT_MISCTEXT, 100));
 end
-
