diff --git a/Mods/MegaMod/Original/ENCVERT.ssl b/Mods/MegaMod/ReDefine/ENCVERT.ssl
index 1af7856..7d25dfb 100644
--- a/Mods/MegaMod/Original/ENCVERT.ssl
+++ b/Mods/MegaMod/ReDefine/ENCVERT.ssl
@@ -216,13 +216,13 @@ begin
             end
         end
     end
-    set_global_var(27, 0);
-    if (global_var(1112) > 0) then begin
+    set_global_var(GVAR_LOAD_MAP_INDEX, 0);
+    if (global_var(GVAR_MOD_RUN_FLAG) > 0) then begin
         fadeIn(500);
-        set_global_var(623, 1);
-        set_global_var(624, 1);
-        set_global_var(625, 1);
-        set_global_var(626, 1);
+        set_global_var(GVAR_ENCLAVE_ENEMY_GUARD, 1);
+        set_global_var(GVAR_ENCLAVE_ENEMY_PRESIDENT, 1);
+        set_global_var(GVAR_ENCLAVE_ENEMY_TRAPS, 1);
+        set_global_var(GVAR_ENCLAVE_ENEMY_REACTOR, 1);
     end
 end
 
@@ -305,4 +305,3 @@ begin
         end
     end
 end
-
