diff --git a/Mods/MegaMod/Original/EPABABY.ssl b/Mods/MegaMod/ReDefine/EPABABY.ssl
index 4ff5985..fcf6cdc 100644
--- a/Mods/MegaMod/Original/EPABABY.ssl
+++ b/Mods/MegaMod/ReDefine/EPABABY.ssl
@@ -160,9 +160,9 @@ procedure use_obj_on_p_proc
 begin
     script_overrides;
     if (obj_pid(obj_being_used_with) == 584) then begin
-        if ((global_var(1070) > (1 - 1)) and (global_var(1070) < (2 + 1))) then begin
-            set_global_var(1070, 3);
-            display_msg(message_str(1700, 700));
+        if ((global_var(GVAR_EPA_LENNY_STATE) > (1 - 1)) and (global_var(GVAR_EPA_LENNY_STATE) < (2 + 1))) then begin
+            set_global_var(GVAR_EPA_LENNY_STATE, 3);
+            display_mstr(700);
         end
     end
 end
@@ -183,7 +183,7 @@ begin
         end_dialogue;
     end
     else begin
-        float_msg(self_obj, message_str(1700, random(235, 241)), 8);
+        float_msg(self_obj, mstr(random(235, 241)), FLOAT_MSG_YELLOW);
     end
 end
 
@@ -201,93 +201,92 @@ end
 
 procedure Node001
 begin
-    gsay_reply(1700, 200);
-    giq_option(4, 1700, 210, Node002, 50);
-    giq_option(4, 1700, 211, Node003, 50);
-    giq_option(4, 1700, 212, Node004, 50);
-    giq_option(-3, 1700, 220, Node007, 50);
-    if ((global_var(1070) > 0) and has_trait(0, dude_obj, 40) and (has_skill(dude_obj, 12) > 114)) then begin
-        giq_option(4, 1700, 213, Node005, 50);
+    Reply(200);
+    giq_option(4, NAME, 210, Node002, NEUTRAL_REACTION);
+    giq_option(4, NAME, 211, Node003, NEUTRAL_REACTION);
+    giq_option(4, NAME, 212, Node004, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 220, Node007, NEUTRAL_REACTION);
+    if ((global_var(GVAR_EPA_LENNY_STATE) > 0) and has_trait(TRAIT_PERK, dude_obj, PERK_scrounger) and (dude_skill(SKILL_SCIENCE) > 114)) then begin
+        giq_option(4, NAME, 213, Node005, NEUTRAL_REACTION);
     end
-    if ((global_var(1070) > 0) and has_trait(0, dude_obj, 40) and (has_skill(dude_obj, 12) < 115)) then begin
-        giq_option(4, 1700, 213, Node006, 50);
+    if ((global_var(GVAR_EPA_LENNY_STATE) > 0) and has_trait(TRAIT_PERK, dude_obj, PERK_scrounger) and (dude_skill(SKILL_SCIENCE) < 115)) then begin
+        giq_option(4, NAME, 213, Node006, NEUTRAL_REACTION);
     end
-    giq_option(4, 1700, 214, Node999, 50);
+    giq_option(4, NAME, 214, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node002
 begin
-    gsay_reply(1700, 300);
-    giq_option(4, 1700, 211, Node003, 50);
-    giq_option(4, 1700, 212, Node004, 50);
-    if ((global_var(1070) > 0) and has_trait(0, dude_obj, 40) and (has_skill(dude_obj, 12) > 114)) then begin
-        giq_option(4, 1700, 213, Node005, 50);
+    Reply(300);
+    giq_option(4, NAME, 211, Node003, NEUTRAL_REACTION);
+    giq_option(4, NAME, 212, Node004, NEUTRAL_REACTION);
+    if ((global_var(GVAR_EPA_LENNY_STATE) > 0) and has_trait(TRAIT_PERK, dude_obj, PERK_scrounger) and (dude_skill(SKILL_SCIENCE) > 114)) then begin
+        giq_option(4, NAME, 213, Node005, NEUTRAL_REACTION);
     end
-    if ((global_var(1070) > 0) and has_trait(0, dude_obj, 40) and (has_skill(dude_obj, 12) < 115)) then begin
-        giq_option(4, 1700, 213, Node006, 50);
+    if ((global_var(GVAR_EPA_LENNY_STATE) > 0) and has_trait(TRAIT_PERK, dude_obj, PERK_scrounger) and (dude_skill(SKILL_SCIENCE) < 115)) then begin
+        giq_option(4, NAME, 213, Node006, NEUTRAL_REACTION);
     end
-    giq_option(4, 1700, 214, Node999, 50);
+    giq_option(4, NAME, 214, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node003
 begin
-    gsay_reply(1700, 400);
-    giq_option(4, 1700, 211, Node003, 50);
-    giq_option(4, 1700, 212, Node004, 50);
-    if ((global_var(1070) > 0) and has_trait(0, dude_obj, 40) and (has_skill(dude_obj, 12) > 114)) then begin
-        giq_option(4, 1700, 213, Node005, 50);
+    Reply(400);
+    giq_option(4, NAME, 211, Node003, NEUTRAL_REACTION);
+    giq_option(4, NAME, 212, Node004, NEUTRAL_REACTION);
+    if ((global_var(GVAR_EPA_LENNY_STATE) > 0) and has_trait(TRAIT_PERK, dude_obj, PERK_scrounger) and (dude_skill(SKILL_SCIENCE) > 114)) then begin
+        giq_option(4, NAME, 213, Node005, NEUTRAL_REACTION);
     end
-    if ((global_var(1070) > 0) and has_trait(0, dude_obj, 40) and (has_skill(dude_obj, 12) < 115)) then begin
-        giq_option(4, 1700, 213, Node006, 50);
+    if ((global_var(GVAR_EPA_LENNY_STATE) > 0) and has_trait(TRAIT_PERK, dude_obj, PERK_scrounger) and (dude_skill(SKILL_SCIENCE) < 115)) then begin
+        giq_option(4, NAME, 213, Node006, NEUTRAL_REACTION);
     end
-    giq_option(4, 1700, 214, Node999, 50);
+    giq_option(4, NAME, 214, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node004
 begin
-    gsay_reply(1700, 500);
-    giq_option(4, 1700, 210, Node002, 50);
-    giq_option(4, 1700, 211, Node003, 50);
-    if ((global_var(1070) > 0) and has_trait(0, dude_obj, 40) and (has_skill(dude_obj, 12) > 114)) then begin
-        giq_option(4, 1700, 213, Node005, 50);
+    Reply(500);
+    giq_option(4, NAME, 210, Node002, NEUTRAL_REACTION);
+    giq_option(4, NAME, 211, Node003, NEUTRAL_REACTION);
+    if ((global_var(GVAR_EPA_LENNY_STATE) > 0) and has_trait(TRAIT_PERK, dude_obj, PERK_scrounger) and (dude_skill(SKILL_SCIENCE) > 114)) then begin
+        giq_option(4, NAME, 213, Node005, NEUTRAL_REACTION);
     end
-    if ((global_var(1070) > 0) and has_trait(0, dude_obj, 40) and (has_skill(dude_obj, 12) < 115)) then begin
-        giq_option(4, 1700, 213, Node006, 50);
+    if ((global_var(GVAR_EPA_LENNY_STATE) > 0) and has_trait(TRAIT_PERK, dude_obj, PERK_scrounger) and (dude_skill(SKILL_SCIENCE) < 115)) then begin
+        giq_option(4, NAME, 213, Node006, NEUTRAL_REACTION);
     end
-    giq_option(4, 1700, 214, Node999, 50);
+    giq_option(4, NAME, 214, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node005
 begin
-    set_global_var(1070, 3);
-    gsay_reply(1700, 700);
-    giq_option(4, 1700, 214, Node999, 50);
+    set_global_var(GVAR_EPA_LENNY_STATE, 3);
+    Reply(700);
+    giq_option(4, NAME, 214, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node006
 begin
-    set_global_var(1098, 1);
-    gsay_reply(1700, 600);
-    giq_option(4, 1700, 214, Node999, 51);
+    set_global_var(GVAR_PRIMIT_MONK_STATE, 1);
+    Reply(600);
+    giq_option(4, NAME, 214, Node999, BAD_REACTION);
 end
 
 procedure Node007
 begin
     set_local_var(2, 1);
-    gsay_reply(1700, 221);
-    giq_option(-3, 1700, 222, Node008, 51);
-    giq_option(-3, 1700, 223, Node999, 51);
-    giq_option(-3, 1700, 224, Node999, 51);
+    Reply(221);
+    giq_option(-3, NAME, 222, Node008, BAD_REACTION);
+    giq_option(-3, NAME, 223, Node999, BAD_REACTION);
+    giq_option(-3, NAME, 224, Node999, BAD_REACTION);
 end
 
 procedure Node008
 begin
-    gsay_reply(1700, 230);
-    giq_option(-3, 1700, 223, Node999, 51);
-    giq_option(-3, 1700, 224, Node999, 51);
+    Reply(230);
+    giq_option(-3, NAME, 223, Node999, BAD_REACTION);
+    giq_option(-3, NAME, 224, Node999, BAD_REACTION);
 end
 
 procedure Node999
 begin
 end
-
