diff --git a/Mods/MegaMod/Original/EPAC5.ssl b/Mods/MegaMod/ReDefine/EPAC5.ssl
index e3bb5df..a72a7a4 100644
--- a/Mods/MegaMod/Original/EPAC5.ssl
+++ b/Mods/MegaMod/ReDefine/EPAC5.ssl
@@ -48,10 +48,10 @@ begin
             if (anim_busy(self_obj) == 0) then begin
                 if ((tile_distance_objs(self_obj, dude_obj) < 8) or obj_can_see_obj(self_obj, dude_obj)) then begin
                     if (anim_busy(self_obj) == 0) then begin
-                        global_temp := rotation_to_tile(tile_num(dude_obj), tile_num(self_obj));
+                        global_temp := rotation_to_tile(dude_tile, tile_num(self_obj));
                         animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(self_obj), global_temp, random(3, 10)), 1);
                         global_temp := (global_temp + 1) % 6;
-                        while ((anim_busy(self_obj) == 0) and (global_temp != rotation_to_tile(tile_num(dude_obj), tile_num(self_obj)))) do begin
+                        while ((anim_busy(self_obj) == 0) and (global_temp != rotation_to_tile(dude_tile, tile_num(self_obj)))) do begin
                             debug_msg("flee loop: rot == " + global_temp);
                             animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(self_obj), global_temp, random(3, 10)), 1);
                             global_temp := (global_temp + 1) % 6;
@@ -61,10 +61,10 @@ begin
             end
         end
         else begin
-            attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+            attack(dude_obj);
         end
     end
-    if (global_var(740) == 1) then begin
+    if (global_var(GVAR_MADE_DRUGS) == 1) then begin
         destroy_object(self_obj);
     end
 end
@@ -78,8 +78,8 @@ end
 
 procedure talk_p_proc
 begin
-    if (global_var(750) == 0) then begin
-        float_msg(self_obj, message_str(1365, random(333, 335)), 8);
+    if (global_var(GVAR_FETCH_GECKOPELTS) == 0) then begin
+        float_msg(self_obj, message_str(SCRIPT_EPAC4, random(333, 335)), FLOAT_MSG_YELLOW);
     end
     else begin
         call talk_two_p_proc();
@@ -89,59 +89,59 @@ end
 procedure destroy_p_proc
 begin
     if (source_obj == dude_obj) then begin
-        set_global_var(5, global_var(5) + 1);
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(4) > (3 * global_var(5))) or (global_var(2) == 1)) and (global_var(1) == 0)) then begin
-            set_global_var(2, 1);
-            set_global_var(3, 0);
+        set_global_var(GVAR_GOOD_MONSTER, global_var(GVAR_GOOD_MONSTER) + 1);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(4) > (3 * global_var(GVAR_GOOD_MONSTER))) or (global_var(GVAR_CHAMPION_REPUTATION) == 1)) and (global_var(GVAR_CHILDKILLER_REPUTATION) == 0)) then begin
+            set_global_var(GVAR_CHAMPION_REPUTATION, 1);
+            set_global_var(GVAR_BERSERKER_REPUTATION, 0);
         end
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(5) > (2 * global_var(4))) or (global_var(3) == 1))) then begin
-            set_global_var(3, 1);
-            set_global_var(2, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
+            set_global_var(GVAR_BERSERKER_REPUTATION, 1);
+            set_global_var(GVAR_CHAMPION_REPUTATION, 0);
         end
-        set_global_var(0, global_var(0) + -10);
+        set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + -10);
         debug_msg("Player gains " + -10 + " Karma Points.");
-        set_global_var(37, 0);
-        set_global_var(38, 0);
-        set_global_var(39, 0);
-        set_global_var(40, 0);
-        set_global_var(41, 0);
-        set_global_var(42, 0);
-        set_global_var(43, 0);
-        set_global_var(44, 0);
-        set_global_var(45, 0);
-        if (global_var(0) >= 1000) then begin
-            set_global_var(37, 1);
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+        set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+        set_global_var(GVAR_KARMA_DEFENDER, 0);
+        set_global_var(GVAR_KARMA_WANDERER, 0);
+        set_global_var(GVAR_KARMA_BETRAYER, 0);
+        set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+        set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+            set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
         end
         else begin
-            if (global_var(0) >= 750) then begin
-                set_global_var(38, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
             end
             else begin
-                if (global_var(0) >= 500) then begin
-                    set_global_var(39, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                 end
                 else begin
-                    if (global_var(0) >= 250) then begin
-                        set_global_var(40, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                        set_global_var(GVAR_KARMA_DEFENDER, 1);
                     end
                     else begin
-                        if (global_var(0) > -250) then begin
-                            set_global_var(41, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                            set_global_var(GVAR_KARMA_WANDERER, 1);
                         end
                         else begin
-                            if (global_var(0) > -500) then begin
-                                set_global_var(42, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                set_global_var(GVAR_KARMA_BETRAYER, 1);
                             end
                             else begin
-                                if (global_var(0) > -750) then begin
-                                    set_global_var(43, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -1000) then begin
-                                        set_global_var(44, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                     end
                                     else begin
-                                        set_global_var(45, 1);
+                                        set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                     end
                                 end
                             end
@@ -157,17 +157,17 @@ procedure look_at_p_proc
 begin
     script_overrides;
     if (local_var(4) == 0) then begin
-        display_msg(message_str(1369, 100));
+        display_mstr(100);
     end
     else begin
-        display_msg(message_str(1369, 101));
+        display_mstr(101);
     end
 end
 
 procedure description_p_proc
 begin
     script_overrides;
-    display_msg(message_str(1369, 102));
+    display_mstr(102);
 end
 
 procedure use_skill_on_p_proc
@@ -190,7 +190,7 @@ procedure timed_event_p_proc
 begin
     if (fixed_param == 3) then begin
         if (((critter_state(self_obj) bwand 2) == 0) and not(combat_is_initialized)) then begin
-            float_msg(self_obj, message_str(1369, random(244, 248)), 8);
+            float_msg(self_obj, mstr(random(244, 248)), FLOAT_MSG_YELLOW);
             add_timer_event(self_obj, game_ticks(random(5, 6)), 3);
         end
     end
@@ -211,32 +211,32 @@ end
 
 procedure Node001
 begin
-    gsay_reply(1369, 200);
-    giq_option(4, 1369, 201, Node002, 50);
+    Reply(200);
+    giq_option(4, NAME, 201, Node002, NEUTRAL_REACTION);
 end
 
 procedure Node002
 begin
-    gsay_reply(1369, 210);
-    giq_option(4, 1369, 211, Node003, 50);
+    Reply(210);
+    giq_option(4, NAME, 211, Node003, NEUTRAL_REACTION);
 end
 
 procedure Node003
 begin
-    gsay_reply(1369, 220);
-    giq_option(4, 1369, 221, Node004, 50);
+    Reply(220);
+    giq_option(4, NAME, 221, Node004, NEUTRAL_REACTION);
 end
 
 procedure Node004
 begin
-    gsay_reply(1369, 230);
-    giq_option(4, 1369, 231, Node005, 50);
+    Reply(230);
+    giq_option(4, NAME, 231, Node005, NEUTRAL_REACTION);
 end
 
 procedure Node005
 begin
-    gsay_reply(1369, 240);
-    giq_option(4, 1369, 241, Node999, 50);
+    Reply(240);
+    giq_option(4, NAME, 241, Node999, NEUTRAL_REACTION);
 end
 
 procedure talk_two_p_proc
@@ -250,8 +250,7 @@ begin
     end
     else begin
         if (local_var(4) == 1) then begin
-            float_msg(self_obj, message_str(1369, random(242, 243)), 8);
+            float_msg(self_obj, mstr(random(242, 243)), FLOAT_MSG_YELLOW);
         end
     end
 end
-
