diff --git a/Mods/MegaMod/Original/EPAC17.ssl b/Mods/MegaMod/ReDefine/EPAC17.ssl
index 7d2b7f2..667a944 100644
--- a/Mods/MegaMod/Original/EPAC17.ssl
+++ b/Mods/MegaMod/ReDefine/EPAC17.ssl
@@ -231,31 +231,31 @@ end
 
 procedure critter_p_proc
 begin
-    if (global_var(753) == 1) then begin
+    if (global_var(GVAR_DYNAMITE_TRIGGER) == 1) then begin
         anim(self_obj, 1000, 5);
-        float_msg(self_obj, message_str(1400, 603), 2);
+        float_msg(self_obj, mstr(603), FLOAT_MSG_RED);
     end
 end
 
 procedure combat_p_proc
 begin
-    if (global_var(753) == 0) then begin
+    if (global_var(GVAR_DYNAMITE_TRIGGER) == 0) then begin
         script_overrides;
-        if (global_var(746) == 1) then begin
+        if (global_var(GVAR_SPRAY_KILLANTS) == 1) then begin
             critter_stop_attacking(self_obj);
             if (local_var(7) == 0) then begin
                 explosion(16104, 0, 1);
-                float_msg(self_obj, message_str(1400, 109), 12);
+                float_msg(self_obj, mstr(109), FLOAT_MSG_LIGHT_GREY);
                 set_local_var(7, 1);
                 terminate_combat;
             end
         end
     end
     else begin
-        if (global_var(753) == 1) then begin
+        if (global_var(GVAR_DYNAMITE_TRIGGER) == 1) then begin
             if (local_var(7) == 0) then begin
                 explosion(16104, 0, 1);
-                float_msg(self_obj, message_str(1400, 109), 12);
+                float_msg(self_obj, mstr(109), FLOAT_MSG_LIGHT_GREY);
                 set_local_var(7, 1);
                 terminate_combat;
             end
@@ -274,17 +274,17 @@ procedure look_at_p_proc
 begin
     script_overrides;
     if (local_var(4) == 0) then begin
-        display_msg(message_str(1400, 106));
+        display_mstr(106);
     end
     else begin
-        display_msg(message_str(1400, 107));
+        display_mstr(107);
     end
 end
 
 procedure description_p_proc
 begin
     script_overrides;
-    display_msg(message_str(1400, 108));
+    display_mstr(108);
 end
 
 procedure use_skill_on_p_proc
@@ -297,8 +297,8 @@ end
 
 procedure damage_p_proc
 begin
-    set_global_var(746, 1);
-    float_msg(self_obj, message_str(1400, 109), 12);
+    set_global_var(GVAR_SPRAY_KILLANTS, 1);
+    float_msg(self_obj, mstr(109), FLOAT_MSG_LIGHT_GREY);
 end
 
 procedure map_enter_p_proc
@@ -437,21 +437,21 @@ end
 
 procedure Node001
 begin
-    gsay_reply(1400, 120);
-    giq_option(4, 1400, message_str(1400, 121) + obj_name(dude_obj) + message_str(1400, 122), Node002, 50);
-    giq_option(4, 1400, 123, Node001a, 50);
+    Reply(120);
+    giq_option(4, NAME, mstr(121) + dude_name + mstr(122), Node002, NEUTRAL_REACTION);
+    giq_option(4, NAME, 123, Node001a, NEUTRAL_REACTION);
 end
 
 procedure Node002
 begin
-    gsay_reply(1400, 130);
-    giq_option(4, 1400, 131, Node002a, 50);
-    giq_option(4, 1400, 132, Node002b, 50);
+    Reply(130);
+    giq_option(4, NAME, 131, Node002a, NEUTRAL_REACTION);
+    giq_option(4, NAME, 132, Node002b, NEUTRAL_REACTION);
 end
 
 procedure Node002a
 begin
-    if (has_skill(dude_obj, 12) > 70) then begin
+    if (dude_skill(SKILL_SCIENCE) > 70) then begin
         call Node003();
     end
     else begin
@@ -461,189 +461,189 @@ end
 
 procedure Node003
 begin
-    gsay_reply(1400, 140);
-    giq_option(4, 1400, 141, Node005, 50);
-    giq_option(4, 1400, 142, Node006, 50);
+    Reply(140);
+    giq_option(4, NAME, 141, Node005, NEUTRAL_REACTION);
+    giq_option(4, NAME, 142, Node006, NEUTRAL_REACTION);
 end
 
 procedure Node004
 begin
-    gsay_reply(1400, 150);
-    if ((has_skill(dude_obj, 14) > 100) or (has_skill(dude_obj, 12) > 100)) then begin
-        giq_option(4, 1400, 151, Node007, 50);
+    Reply(150);
+    if ((dude_skill(SKILL_SPEECH) > 100) or (dude_skill(SKILL_SCIENCE) > 100)) then begin
+        giq_option(4, NAME, 151, Node007, NEUTRAL_REACTION);
     end
-    giq_option(4, 1400, 152, Node004a, 50);
+    giq_option(4, NAME, 152, Node004a, NEUTRAL_REACTION);
 end
 
 procedure Node005
 begin
-    gsay_reply(1400, 160);
-    giq_option(4, 1400, 161, Node009, 50);
-    giq_option(4, 1400, 700, Node050, 50);
+    Reply(160);
+    giq_option(4, NAME, 161, Node009, NEUTRAL_REACTION);
+    giq_option(4, NAME, 700, Node050, NEUTRAL_REACTION);
 end
 
 procedure Node006
 begin
-    gsay_reply(1400, 170);
-    giq_option(4, 1400, 171, Node013, 50);
-    giq_option(4, 1400, 172, Node014, 50);
+    Reply(170);
+    giq_option(4, NAME, 171, Node013, NEUTRAL_REACTION);
+    giq_option(4, NAME, 172, Node014, NEUTRAL_REACTION);
 end
 
 procedure Node007
 begin
-    gsay_reply(1400, 180);
-    giq_option(4, 1400, 181, Node008, 50);
+    Reply(180);
+    giq_option(4, NAME, 181, Node008, NEUTRAL_REACTION);
 end
 
 procedure Node008
 begin
-    gsay_reply(1400, 190);
-    giq_option(4, 1400, 191, Node005, 50);
-    giq_option(4, 1400, 192, Node006, 50);
+    Reply(190);
+    giq_option(4, NAME, 191, Node005, NEUTRAL_REACTION);
+    giq_option(4, NAME, 192, Node006, NEUTRAL_REACTION);
 end
 
 procedure Node009
 begin
-    gsay_reply(1400, 200);
-    giq_option(4, 1400, 201, Node010, 50);
-    giq_option(4, 1400, 202, Node011, 50);
-    giq_option(4, 1400, 203, Node012, 50);
+    Reply(200);
+    giq_option(4, NAME, 201, Node010, NEUTRAL_REACTION);
+    giq_option(4, NAME, 202, Node011, NEUTRAL_REACTION);
+    giq_option(4, NAME, 203, Node012, NEUTRAL_REACTION);
 end
 
 procedure Node010
 begin
-    gsay_reply(1400, 210);
-    if (has_skill(dude_obj, 14) > 90) then begin
-        giq_option(4, 1400, 211, Node015, 50);
+    Reply(210);
+    if (dude_skill(SKILL_SPEECH) > 90) then begin
+        giq_option(4, NAME, 211, Node015, NEUTRAL_REACTION);
     end
-    giq_option(4, 1400, 212, Node016, 50);
-    giq_option(4, 1400, 213, Node017, 50);
+    giq_option(4, NAME, 212, Node016, NEUTRAL_REACTION);
+    giq_option(4, NAME, 213, Node017, NEUTRAL_REACTION);
 end
 
 procedure Node011
 begin
-    gsay_reply(1400, 220);
-    giq_option(4, 1400, 221, Node013, 50);
-    giq_option(4, 1400, 222, Node012, 50);
+    Reply(220);
+    giq_option(4, NAME, 221, Node013, NEUTRAL_REACTION);
+    giq_option(4, NAME, 222, Node012, NEUTRAL_REACTION);
 end
 
 procedure Node012
 begin
-    gsay_reply(1400, 230);
-    giq_option(4, 1400, 231, Node017, 50);
+    Reply(230);
+    giq_option(4, NAME, 231, Node017, NEUTRAL_REACTION);
 end
 
 procedure Node013
 begin
-    gsay_reply(1400, 250);
-    giq_option(4, 1400, 251, Node018, 50);
-    if (get_critter_stat(dude_obj, 4) >= 7) then begin
-        giq_option(4, 1400, 252, Node017, 50);
+    Reply(250);
+    giq_option(4, NAME, 251, Node018, NEUTRAL_REACTION);
+    if (dude_iq >= 7) then begin
+        giq_option(4, NAME, 252, Node017, NEUTRAL_REACTION);
     end
 end
 
 procedure Node014
 begin
-    gsay_reply(1400, 240);
-    giq_option(4, 1400, 241, Node019, 50);
-    giq_option(4, 1400, 242, Node020, 50);
+    Reply(240);
+    giq_option(4, NAME, 241, Node019, NEUTRAL_REACTION);
+    giq_option(4, NAME, 242, Node020, NEUTRAL_REACTION);
     if (party_member_obj(16777376) != 0) then begin
-        giq_option(4, 1400, 243, Node026, 50);
+        giq_option(4, NAME, 243, Node026, NEUTRAL_REACTION);
     end
 end
 
 procedure Node015
 begin
-    gsay_reply(1400, 260);
-    giq_option(4, 1400, 261, Node021, 50);
-    giq_option(4, 1400, 262, Node006, 50);
-    giq_option(4, 1400, 263, Node017, 50);
+    Reply(260);
+    giq_option(4, NAME, 261, Node021, NEUTRAL_REACTION);
+    giq_option(4, NAME, 262, Node006, NEUTRAL_REACTION);
+    giq_option(4, NAME, 263, Node017, NEUTRAL_REACTION);
 end
 
 procedure Node016
 begin
-    gsay_reply(1400, 270);
-    giq_option(4, 1400, 271, Node010, 50);
-    giq_option(4, 1400, 272, Node011, 50);
-    giq_option(4, 1400, 273, Node012, 50);
+    Reply(270);
+    giq_option(4, NAME, 271, Node010, NEUTRAL_REACTION);
+    giq_option(4, NAME, 272, Node011, NEUTRAL_REACTION);
+    giq_option(4, NAME, 273, Node012, NEUTRAL_REACTION);
 end
 
 procedure Node017
 begin
-    gsay_reply(1400, 280);
-    giq_option(4, 1400, 281, Node032, 50);
-    giq_option(4, 1400, 282, Node017a, 50);
+    Reply(280);
+    giq_option(4, NAME, 281, Node032, NEUTRAL_REACTION);
+    giq_option(4, NAME, 282, Node017a, NEUTRAL_REACTION);
 end
 
 procedure Node018
 begin
-    gsay_reply(1400, 290);
-    giq_option(4, 1400, 291, Node022, 50);
-    giq_option(4, 1400, 292, Node018a, 50);
+    Reply(290);
+    giq_option(4, NAME, 291, Node022, NEUTRAL_REACTION);
+    giq_option(4, NAME, 292, Node018a, NEUTRAL_REACTION);
 end
 
 procedure Node019
 begin
-    gsay_reply(1400, 300);
-    giq_option(4, 1400, 301, Node023, 50);
-    giq_option(4, 1400, 302, Node024, 50);
+    Reply(300);
+    giq_option(4, NAME, 301, Node023, NEUTRAL_REACTION);
+    giq_option(4, NAME, 302, Node024, NEUTRAL_REACTION);
 end
 
 procedure Node020
 begin
-    gsay_reply(1400, 310);
-    giq_option(4, 1400, 311, Node019, 50);
+    Reply(310);
+    giq_option(4, NAME, 311, Node019, NEUTRAL_REACTION);
 end
 
 procedure Node021
 begin
-    gsay_reply(1400, 320);
-    giq_option(4, 1400, 321, Node025, 50);
+    Reply(320);
+    giq_option(4, NAME, 321, Node025, NEUTRAL_REACTION);
 end
 
 procedure Node022
 begin
-    gsay_reply(1400, 330);
-    giq_option(4, 1400, 331, Node022a, 50);
-    giq_option(4, 1400, 332, Node022b, 50);
+    Reply(330);
+    giq_option(4, NAME, 331, Node022a, NEUTRAL_REACTION);
+    giq_option(4, NAME, 332, Node022b, NEUTRAL_REACTION);
 end
 
 procedure Node022a
 begin
-    set_global_var(746, 1);
+    set_global_var(GVAR_SPRAY_KILLANTS, 1);
     set_local_var(4, 8);
     call combat_p_proc();
 end
 
 procedure Node023
 begin
-    gsay_reply(1400, 340);
-    giq_option(4, 1400, 341, Node024, 50);
+    Reply(340);
+    giq_option(4, NAME, 341, Node024, NEUTRAL_REACTION);
 end
 
 procedure Node024
 begin
-    gsay_reply(1400, 350);
-    giq_option(4, 1400, 351, Node017, 50);
+    Reply(350);
+    giq_option(4, NAME, 351, Node017, NEUTRAL_REACTION);
 end
 
 procedure Node025
 begin
-    gsay_reply(1400, 360);
-    giq_option(4, 1400, 361, Node022a, 50);
-    giq_option(4, 1400, 362, Node022b, 50);
+    Reply(360);
+    giq_option(4, NAME, 361, Node022a, NEUTRAL_REACTION);
+    giq_option(4, NAME, 362, Node022b, NEUTRAL_REACTION);
 end
 
 procedure Node026
 begin
-    gsay_reply(1400, 370);
-    giq_option(4, 1400, 371, Node027, 50);
+    Reply(370);
+    giq_option(4, NAME, 371, Node027, NEUTRAL_REACTION);
 end
 
 procedure Node027
 begin
-    gsay_reply(1400, 380);
-    giq_option(4, 1400, 381, Node027a, 50);
+    Reply(380);
+    giq_option(4, NAME, 381, Node027a, NEUTRAL_REACTION);
 end
 
 procedure Node027a
@@ -658,14 +658,14 @@ end
 
 procedure Node028
 begin
-    gsay_reply(1400, 390);
-    giq_option(4, 1400, 391, Node029, 50);
+    Reply(390);
+    giq_option(4, NAME, 391, Node029, NEUTRAL_REACTION);
 end
 
 procedure Node029
 begin
-    gsay_reply(1400, 400);
-    giq_option(4, 1400, 401, Node029a, 50);
+    Reply(400);
+    giq_option(4, NAME, 401, Node029a, NEUTRAL_REACTION);
 end
 
 procedure Node029a
@@ -676,71 +676,71 @@ end
 
 procedure Node030
 begin
-    gsay_reply(1400, 410);
-    giq_option(4, 1400, 411, Node031, 50);
+    Reply(410);
+    giq_option(4, NAME, 411, Node031, NEUTRAL_REACTION);
 end
 
 procedure Node031
 begin
-    gsay_reply(1400, 420);
-    giq_option(4, 1400, 421, Node032, 50);
-    giq_option(4, 1400, 422, Node017a, 50);
+    Reply(420);
+    giq_option(4, NAME, 421, Node032, NEUTRAL_REACTION);
+    giq_option(4, NAME, 422, Node017a, NEUTRAL_REACTION);
 end
 
 procedure Node032
 begin
-    gsay_reply(1400, 430);
+    Reply(430);
     if (obj_is_carrying_obj_pid(dude_obj, 277) >= 10) then begin
-        giq_option(4, 1400, 432, Node039, 50);
+        giq_option(4, NAME, 432, Node039, NEUTRAL_REACTION);
     end
     if (obj_is_carrying_obj_pid(dude_obj, 277) <= 9) then begin
-        giq_option(4, 1400, 431, Node032a, 50);
+        giq_option(4, NAME, 431, Node032a, NEUTRAL_REACTION);
     end
 end
 
 procedure Node033
 begin
-    gsay_reply(1400, 440);
-    giq_option(4, 1400, 441, Node002, 50);
+    Reply(440);
+    giq_option(4, NAME, 441, Node002, NEUTRAL_REACTION);
 end
 
 procedure Node034
 begin
-    gsay_reply(1400, 450);
-    giq_option(4, 1400, 451, Node002a, 50);
+    Reply(450);
+    giq_option(4, NAME, 451, Node002a, NEUTRAL_REACTION);
 end
 
 procedure Node035
 begin
-    gsay_reply(1400, 460);
-    giq_option(4, 1400, 461, Node031, 50);
-    giq_option(4, 1400, 462, Node999, 50);
+    Reply(460);
+    giq_option(4, NAME, 461, Node031, NEUTRAL_REACTION);
+    giq_option(4, NAME, 462, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node036
 begin
-    gsay_reply(1400, 470);
-    giq_option(4, 1400, 471, Node022a, 50);
-    giq_option(4, 1400, 472, Node037, 50);
+    Reply(470);
+    giq_option(4, NAME, 471, Node022a, NEUTRAL_REACTION);
+    giq_option(4, NAME, 472, Node037, NEUTRAL_REACTION);
 end
 
 procedure Node037
 begin
-    gsay_reply(1400, 480);
-    giq_option(4, 1400, 481, Node014, 50);
-    giq_option(4, 1400, 482, Node010, 50);
-    giq_option(4, 1400, 483, Node031, 50);
-    giq_option(4, 1400, 484, Node012, 50);
+    Reply(480);
+    giq_option(4, NAME, 481, Node014, NEUTRAL_REACTION);
+    giq_option(4, NAME, 482, Node010, NEUTRAL_REACTION);
+    giq_option(4, NAME, 483, Node031, NEUTRAL_REACTION);
+    giq_option(4, NAME, 484, Node012, NEUTRAL_REACTION);
 end
 
 procedure Node038
 begin
-    gsay_reply(1400, 490);
+    Reply(490);
     if (obj_is_carrying_obj_pid(dude_obj, 277) >= 10) then begin
-        giq_option(4, 1400, 491, Node039, 50);
+        giq_option(4, NAME, 491, Node039, NEUTRAL_REACTION);
     end
     if (obj_is_carrying_obj_pid(dude_obj, 277) <= 9) then begin
-        giq_option(4, 1400, 492, Node999, 50);
+        giq_option(4, NAME, 492, Node999, NEUTRAL_REACTION);
     end
 end
 
@@ -748,51 +748,51 @@ procedure Node039
 begin
     variable LVar0 := 0;
     variable LVar1 := 0;
-    gsay_reply(1400, 500);
-    giq_option(4, 1400, 501, Node040, 50);
-    giq_option(4, 1400, 502, Node039a, 50);
-    set_global_var(752, 2);
+    Reply(500);
+    giq_option(4, NAME, 501, Node040, NEUTRAL_REACTION);
+    giq_option(4, NAME, 502, Node039a, NEUTRAL_REACTION);
+    set_global_var(GVAR_DYNAMITE_SETTING, 2);
     LVar0 := obj_carrying_pid_obj(dude_obj, 277);
     LVar1 := rm_mult_objs_from_inven(dude_obj, LVar0, 10);
 end
 
 procedure Node040
 begin
-    gsay_reply(1400, 510);
-    if (global_var(745) != 4) then begin
-        giq_option(4, 1400, 511, Node040a, 50);
+    Reply(510);
+    if (global_var(GVAR_FREE_EPASLAVES) != 4) then begin
+        giq_option(4, NAME, 511, Node040a, NEUTRAL_REACTION);
     end
-    if (global_var(745) != 4) then begin
-        giq_option(4, 1400, 512, Node039a, 50);
+    if (global_var(GVAR_FREE_EPASLAVES) != 4) then begin
+        giq_option(4, NAME, 512, Node039a, NEUTRAL_REACTION);
     end
-    if (global_var(745) == 4) then begin
-        giq_option(4, 1400, 513, Node042, 50);
+    if (global_var(GVAR_FREE_EPASLAVES) == 4) then begin
+        giq_option(4, NAME, 513, Node042, NEUTRAL_REACTION);
     end
 end
 
 procedure Node041
 begin
-    gsay_reply(1400, 520);
-    giq_option(4, 1400, 521, Node017, 50);
-    giq_option(4, 1400, 522, Node999, 50);
+    Reply(520);
+    giq_option(4, NAME, 521, Node017, NEUTRAL_REACTION);
+    giq_option(4, NAME, 522, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node042
 begin
-    gsay_reply(1400, 531);
-    giq_option(4, 1400, 532, Node043, 50);
+    Reply(531);
+    giq_option(4, NAME, 532, Node043, NEUTRAL_REACTION);
 end
 
 procedure Node043
 begin
-    gsay_reply(1400, 540);
-    giq_option(4, 1400, 541, Node044, 50);
+    Reply(540);
+    giq_option(4, NAME, 541, Node044, NEUTRAL_REACTION);
 end
 
 procedure Node044
 begin
-    gsay_reply(1400, 550);
-    giq_option(4, 1400, 551, Node044a, 50);
+    Reply(550);
+    giq_option(4, NAME, 551, Node044a, NEUTRAL_REACTION);
 end
 
 procedure Node044a
@@ -802,27 +802,27 @@ begin
     game_ui_enable;
     gfade_in(10 * 6);
     game_time_advance(60 * (60 * 10) * 2);
-    set_critter_stat(dude_obj, 2, 1);
+    set_critter_stat(dude_obj, STAT_en, 1);
     call Node045();
 end
 
 procedure Node045
 begin
-    gsay_reply(1400, 570);
-    giq_option(4, 1400, 571, Node046, 50);
+    Reply(570);
+    giq_option(4, NAME, 571, Node046, NEUTRAL_REACTION);
 end
 
 procedure Node046
 begin
-    gsay_reply(1400, 580);
-    giq_option(4, 1400, 581, Node046a, 50);
+    Reply(580);
+    giq_option(4, NAME, 581, Node046a, NEUTRAL_REACTION);
 end
 
 procedure Node047
 begin
-    gsay_reply(1400, 590);
-    giq_option(4, 1400, 591, Node040a, 50);
-    giq_option(4, 1400, 592, Node999, 50);
+    Reply(590);
+    giq_option(4, NAME, 591, Node040a, NEUTRAL_REACTION);
+    giq_option(4, NAME, 592, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node001a
@@ -857,7 +857,7 @@ end
 
 procedure Node032a
 begin
-    set_global_var(752, 1);
+    set_global_var(GVAR_DYNAMITE_SETTING, 1);
     set_local_var(4, 7);
 end
 
@@ -869,7 +869,7 @@ end
 procedure Node040a
 begin
     set_local_var(4, 10);
-    set_global_var(745, 1);
+    set_global_var(GVAR_FREE_EPASLAVES, 1);
 end
 
 procedure Node046a
@@ -879,113 +879,113 @@ end
 
 procedure Node048
 begin
-    gsay_reply(1400, 600);
-    giq_option(4, 1400, 601, Node999, 50);
+    Reply(600);
+    giq_option(4, NAME, 601, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node049
 begin
-    gsay_reply(1400, 530);
-    if (global_var(745) != 4) then begin
-        giq_option(4, 1400, 533, Node999, 50);
+    Reply(530);
+    if (global_var(GVAR_FREE_EPASLAVES) != 4) then begin
+        giq_option(4, NAME, 533, Node999, NEUTRAL_REACTION);
     end
-    if (global_var(745) == 4) then begin
-        giq_option(4, 1400, 602, Node042, 50);
+    if (global_var(GVAR_FREE_EPASLAVES) == 4) then begin
+        giq_option(4, NAME, 602, Node042, NEUTRAL_REACTION);
     end
 end
 
 procedure Node050
 begin
-    gsay_reply(1400, 701);
-    giq_option(4, 1400, 702, Node051, 50);
+    Reply(701);
+    giq_option(4, NAME, 702, Node051, NEUTRAL_REACTION);
 end
 
 procedure Node051
 begin
-    gsay_reply(1400, 703);
-    giq_option(4, 1400, 704, Node052, 50);
+    Reply(703);
+    giq_option(4, NAME, 704, Node052, NEUTRAL_REACTION);
 end
 
 procedure Node052
 begin
-    gsay_reply(1400, 705);
-    giq_option(4, 1400, 706, Node053, 50);
+    Reply(705);
+    giq_option(4, NAME, 706, Node053, NEUTRAL_REACTION);
 end
 
 procedure Node053
 begin
-    gsay_reply(1400, 707);
-    giq_option(4, 1400, 708, Node054, 50);
-    giq_option(4, 1400, 709, Node055, 50);
-    giq_option(4, 1400, 710, Node055, 50);
-    giq_option(4, 1400, 715, Node999, 50);
+    Reply(707);
+    giq_option(4, NAME, 708, Node054, NEUTRAL_REACTION);
+    giq_option(4, NAME, 709, Node055, NEUTRAL_REACTION);
+    giq_option(4, NAME, 710, Node055, NEUTRAL_REACTION);
+    giq_option(4, NAME, 715, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node054
 begin
-    gsay_reply(1400, 711);
-    giq_option(4, 1400, 712, Node056, 50);
-    giq_option(4, 1400, 730, Node009, 50);
-    giq_option(4, 1400, 729, Node022a, 50);
-    giq_option(4, 1400, 715, Node999, 50);
+    Reply(711);
+    giq_option(4, NAME, 712, Node056, NEUTRAL_REACTION);
+    giq_option(4, NAME, 730, Node009, NEUTRAL_REACTION);
+    giq_option(4, NAME, 729, Node022a, NEUTRAL_REACTION);
+    giq_option(4, NAME, 715, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node055
 begin
-    gsay_reply(1400, 728);
-    giq_option(4, 1400, 712, Node056, 50);
-    giq_option(4, 1400, 729, Node022a, 50);
-    giq_option(4, 1400, 715, Node999, 50);
+    Reply(728);
+    giq_option(4, NAME, 712, Node056, NEUTRAL_REACTION);
+    giq_option(4, NAME, 729, Node022a, NEUTRAL_REACTION);
+    giq_option(4, NAME, 715, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node056
 begin
-    gsay_reply(1400, 713);
-    giq_option(4, 1400, 714, Node057, 50);
-    giq_option(4, 1400, 729, Node022a, 50);
-    giq_option(4, 1400, 715, Node999, 50);
+    Reply(713);
+    giq_option(4, NAME, 714, Node057, NEUTRAL_REACTION);
+    giq_option(4, NAME, 729, Node022a, NEUTRAL_REACTION);
+    giq_option(4, NAME, 715, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node057
 begin
-    gsay_reply(1400, 716);
-    giq_option(4, 1400, 717, Node058, 50);
-    giq_option(4, 1400, 715, Node999, 50);
+    Reply(716);
+    giq_option(4, NAME, 717, Node058, NEUTRAL_REACTION);
+    giq_option(4, NAME, 715, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node058
 begin
-    gsay_reply(1400, 718);
-    giq_option(4, 1400, 720, Node059, 50);
-    giq_option(4, 1400, 721, Node061, 50);
-    giq_option(4, 1400, 730, Node009, 50);
-    giq_option(4, 1400, 719, Node999, 50);
+    Reply(718);
+    giq_option(4, NAME, 720, Node059, NEUTRAL_REACTION);
+    giq_option(4, NAME, 721, Node061, NEUTRAL_REACTION);
+    giq_option(4, NAME, 730, Node009, NEUTRAL_REACTION);
+    giq_option(4, NAME, 719, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node059
 begin
-    gsay_reply(1400, 722);
-    giq_option(4, 1400, 723, Node060, 50);
-    giq_option(4, 1400, 721, Node061, 50);
-    giq_option(4, 1400, 715, Node999, 50);
+    Reply(722);
+    giq_option(4, NAME, 723, Node060, NEUTRAL_REACTION);
+    giq_option(4, NAME, 721, Node061, NEUTRAL_REACTION);
+    giq_option(4, NAME, 715, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node060
 begin
-    gsay_reply(1400, 724);
-    giq_option(4, 1400, 725, Node061, 50);
-    giq_option(4, 1400, 730, Node009, 50);
-    giq_option(4, 1400, 726, Node999, 50);
+    Reply(724);
+    giq_option(4, NAME, 725, Node061, NEUTRAL_REACTION);
+    giq_option(4, NAME, 730, Node009, NEUTRAL_REACTION);
+    giq_option(4, NAME, 726, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node061
 begin
-    gsay_reply(1400, 731);
-    giq_option(4, 1400, 730, Node009, 50);
-    giq_option(4, 1400, 192, Node006, 50);
-    giq_option(4, 1400, 201, Node010, 50);
-    giq_option(4, 1400, 202, Node011, 50);
-    giq_option(4, 1400, 203, Node012, 50);
+    Reply(731);
+    giq_option(4, NAME, 730, Node009, NEUTRAL_REACTION);
+    giq_option(4, NAME, 192, Node006, NEUTRAL_REACTION);
+    giq_option(4, NAME, 201, Node010, NEUTRAL_REACTION);
+    giq_option(4, NAME, 202, Node011, NEUTRAL_REACTION);
+    giq_option(4, NAME, 203, Node012, NEUTRAL_REACTION);
 end
 
 procedure Node998
@@ -1005,9 +1005,8 @@ procedure timed_event_p_proc
 begin
     if (fixed_param == 2) then begin
         if (((critter_state(self_obj) bwand 2) == 0) and not(combat_is_initialized)) then begin
-            float_msg(self_obj, message_str(1400, random(100, 105)), 12);
+            float_msg(self_obj, mstr(random(100, 105)), FLOAT_MSG_LIGHT_GREY);
             add_timer_event(self_obj, game_ticks(random(13, 16)), 2);
         end
     end
 end
-
