diff --git a/Mods/MegaMod/Original/EPAC11.ssl b/Mods/MegaMod/ReDefine/EPAC11.ssl
index 5982e37..413ad30 100644
--- a/Mods/MegaMod/Original/EPAC11.ssl
+++ b/Mods/MegaMod/ReDefine/EPAC11.ssl
@@ -181,17 +181,17 @@ procedure look_at_p_proc
 begin
     script_overrides;
     if (local_var(4) == 0) then begin
-        display_msg(message_str(1386, 100));
+        display_mstr(100);
     end
     else begin
-        display_msg(message_str(1386, 101));
+        display_mstr(101);
     end
 end
 
 procedure description_p_proc
 begin
     script_overrides;
-    display_msg(message_str(1386, 102));
+    display_mstr(102);
 end
 
 procedure use_skill_on_p_proc
@@ -236,60 +236,60 @@ end
 
 procedure Node001
 begin
-    gsay_reply(1386, 200);
-    giq_option(4, 1386, 201, Node002, 50);
+    Reply(200);
+    giq_option(4, NAME, 201, Node002, NEUTRAL_REACTION);
 end
 
 procedure Node002
 begin
-    gsay_reply(1386, 202);
-    if (get_critter_stat(dude_obj, 4) >= 7) then begin
-        giq_option(4, 1386, 203, Node003, 50);
+    Reply(202);
+    if (dude_iq >= 7) then begin
+        giq_option(4, NAME, 203, Node003, NEUTRAL_REACTION);
     end
-    if (get_critter_stat(dude_obj, 4) <= 6) then begin
-        giq_option(4, 1386, 204, Node003, 50);
+    if (dude_iq <= 6) then begin
+        giq_option(4, NAME, 204, Node003, NEUTRAL_REACTION);
     end
 end
 
 procedure Node003
 begin
-    gsay_reply(1386, 205);
-    giq_option(4, 1386, 206, Node004, 50);
-    giq_option(4, 1386, 207, Node999, 50);
+    Reply(205);
+    giq_option(4, NAME, 206, Node004, NEUTRAL_REACTION);
+    giq_option(4, NAME, 207, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node004
 begin
-    gsay_reply(1386, 208);
-    giq_option(4, 1386, 209, Node004a, 50);
+    Reply(208);
+    giq_option(4, NAME, 209, Node004a, NEUTRAL_REACTION);
 end
 
 procedure Node005
 begin
-    gsay_reply(1386, 210);
-    giq_option(4, 1386, 211, Node005a, 50);
+    Reply(210);
+    giq_option(4, NAME, 211, Node005a, NEUTRAL_REACTION);
 end
 
 procedure Node006
 begin
-    gsay_reply(1386, 212);
-    giq_option(4, 1386, 213, Node999, 50);
+    Reply(212);
+    giq_option(4, NAME, 213, Node999, NEUTRAL_REACTION);
     set_local_var(4, 2);
 end
 
 procedure Node007
 begin
-    gsay_reply(1386, 214);
-    giq_option(4, 1386, 215, Node004, 50);
+    Reply(214);
+    giq_option(4, NAME, 215, Node004, NEUTRAL_REACTION);
 end
 
 procedure Node004a
 begin
-    if (has_skill(dude_obj, 3) <= 50) then begin
+    if (has_skill(dude_obj, SKILL_UNARMED_COMBAT) <= 50) then begin
         call Node005();
     end
     else begin
-        if (has_skill(dude_obj, 3) > 50) then begin
+        if (has_skill(dude_obj, SKILL_UNARMED_COMBAT) > 50) then begin
             call Node006();
         end
     end
@@ -309,21 +309,21 @@ begin
     gfade_out(10);
     game_ui_enable;
     gfade_in(10);
-    critter_mod_skill(dude_obj, 3, 20);
+    critter_mod_skill(dude_obj, SKILL_UNARMED_COMBAT, 20);
     game_time_advance(60 * (60 * 10) * 3);
     call Node008();
 end
 
 procedure Node008
 begin
-    gsay_reply(1386, 216);
-    giq_option(4, 1386, 217, Node999, 50);
+    Reply(216);
+    giq_option(4, NAME, 217, Node999, NEUTRAL_REACTION);
     set_local_var(4, 2);
 end
 
 procedure Leave_p_proc
 begin
-    float_msg(self_obj, message_str(1386, 218), 8);
+    float_msg(self_obj, mstr(218), FLOAT_MSG_YELLOW);
     game_ui_disable;
     gfade_out(10);
     set_local_var(4, 3);
@@ -341,4 +341,3 @@ begin
     end
     add_timer_event(self_obj, game_ticks(random(3, 5)), 1);
 end
-
