diff --git a/Mods/MegaMod/Original/EPAC13.ssl b/Mods/MegaMod/ReDefine/EPAC13.ssl
index 6bb1c42..6d9c2f5 100644
--- a/Mods/MegaMod/Original/EPAC13.ssl
+++ b/Mods/MegaMod/ReDefine/EPAC13.ssl
@@ -147,7 +147,7 @@ end
 
 procedure start
 begin
-    float_msg(self_obj, message_str(1393, random(103, 106)), 2);
+    float_msg(self_obj, mstr(random(103, 106)), FLOAT_MSG_RED);
     add_timer_event(self_obj, game_ticks(random(13, 16)), 3);
     add_timer_event(self_obj, game_ticks(random(5, 7)), 1);
 end
@@ -159,12 +159,12 @@ end
 
 procedure critter_p_proc
 begin
-    if (global_var(746) == 1) then begin
+    if (global_var(GVAR_SPRAY_KILLANTS) == 1) then begin
         animate_move_obj_to_tile(self_obj, 18508, 1);
     end
     if (obj_can_see_obj(self_obj, dude_obj) and (pissed == 0)) then begin
         pissed := 1;
-        attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+        attack(dude_obj);
     end
     else begin
         pissed := 0;
@@ -179,17 +179,17 @@ procedure look_at_p_proc
 begin
     script_overrides;
     if (local_var(4) == 0) then begin
-        display_msg(message_str(1393, 100));
+        display_mstr(100);
     end
     else begin
-        display_msg(message_str(1393, 101));
+        display_mstr(101);
     end
 end
 
 procedure description_p_proc
 begin
     script_overrides;
-    display_msg(message_str(1393, 102));
+    display_mstr(102);
 end
 
 procedure use_skill_on_p_proc
@@ -202,13 +202,13 @@ end
 
 procedure damage_p_proc
 begin
-    attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+    attack(dude_obj);
 end
 
 procedure map_enter_p_proc
 begin
-    critter_add_trait(self_obj, 1, 6, 93);
-    critter_add_trait(self_obj, 1, 5, 1);
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 93);
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_ARROYO_WARRIOR);
 end
 
 procedure map_update_p_proc
@@ -217,13 +217,13 @@ end
 
 procedure talk_p_proc
 begin
-    float_msg(self_obj, message_str(1393, 107), 2);
+    float_msg(self_obj, mstr(107), FLOAT_MSG_RED);
 end
 
 procedure timed_event_p_proc
 begin
     if (fixed_param == 3) then begin
-        float_msg(self_obj, message_str(1393, random(103, 106)), 2);
+        float_msg(self_obj, mstr(random(103, 106)), FLOAT_MSG_RED);
         add_timer_event(self_obj, game_ticks(random(13, 16)), 3);
     end
     if (fixed_param == 1) then begin
@@ -240,4 +240,3 @@ end
 procedure combat_p_proc
 begin
 end
-
