diff --git a/Mods/MegaMod/Original/EPAENT.ssl b/Mods/MegaMod/ReDefine/EPAENT.ssl
index 6d71e57..35b6397 100644
--- a/Mods/MegaMod/Original/EPAENT.ssl
+++ b/Mods/MegaMod/ReDefine/EPAENT.ssl
@@ -146,9 +146,9 @@ end
 
 procedure map_enter_p_proc
 begin
-    if (global_var(766) == 4) then begin
+    if (global_var(GVAR_V23_HOLODISK) == 4) then begin
         override_map_start(20303 % 200, 20303 / 200, 0, 0);
-        set_global_var(766, 0);
+        set_global_var(GVAR_V23_HOLODISK, 0);
     end
     if ((get_month >= 3) and (get_month < 5)) then begin
         if ((game_time_hour >= 600) and (game_time_hour < (600 + 100))) then begin
@@ -307,4 +307,3 @@ begin
         end
     end
 end
-
