diff --git a/Mods/MegaMod/Original/EPAI9.ssl b/Mods/MegaMod/ReDefine/EPAI9.ssl
index e494d18..9150aab 100644
--- a/Mods/MegaMod/Original/EPAI9.ssl
+++ b/Mods/MegaMod/ReDefine/EPAI9.ssl
@@ -147,7 +147,7 @@ begin
     LVar0 := obj_pid(obj_being_used_with);
     if (LVar0 == 639) then begin
         script_overrides;
-        display_msg(message_str(1383, 103));
+        display_mstr(103);
         obj_unlock(self_obj);
         reg_anim_func(2, self_obj);
         reg_anim_func(1, 1);
@@ -162,24 +162,24 @@ end
 procedure look_at_p_proc
 begin
     script_overrides;
-    display_msg(message_str(1383, 102));
+    display_mstr(102);
 end
 
 procedure description_p_proc
 begin
     script_overrides;
-    display_msg(message_str(1383, 101));
+    display_mstr(101);
 end
 
 procedure use_p_proc
 begin
     if (local_var(4) == 0) then begin
         script_overrides;
-        display_msg(message_str(1383, 100));
+        display_mstr(100);
     end
     else begin
         if (local_var(4) == 1) then begin
-            display_msg(message_str(1383, 104));
+            display_mstr(104);
         end
     end
 end
@@ -195,4 +195,3 @@ end
 procedure drop_p_proc
 begin
 end
-
