diff --git a/Mods/MegaMod/Original/EPAI4.ssl b/Mods/MegaMod/ReDefine/EPAI4.ssl
index a8f34b0..bf3e35a 100644
--- a/Mods/MegaMod/Original/EPAI4.ssl
+++ b/Mods/MegaMod/ReDefine/EPAI4.ssl
@@ -146,17 +146,17 @@ end
 procedure description_p_proc
 begin
     script_overrides;
-    display_msg(message_str(1372, 108));
+    display_mstr(108);
 end
 
 procedure look_at_p_proc
 begin
     script_overrides;
     if (local_var(4) == 0) then begin
-        display_msg(message_str(1372, 106));
+        display_mstr(106);
     end
     else begin
-        display_msg(message_str(1372, 107));
+        display_mstr(107);
     end
 end
 
@@ -185,28 +185,28 @@ end
 
 procedure Node001
 begin
-    gsay_reply(1372, 100);
-    if (global_var(735) == 0) then begin
-        giq_option(4, 1372, 101, Node002, 50);
+    Reply(100);
+    if (global_var(GVAR_PLANT_QUEST) == 0) then begin
+        giq_option(4, NAME, 101, Node002, NEUTRAL_REACTION);
     end
-    if (global_var(735) == 1) then begin
-        giq_option(4, 1372, 102, Node003, 50);
+    if (global_var(GVAR_PLANT_QUEST) == 1) then begin
+        giq_option(4, NAME, 102, Node003, NEUTRAL_REACTION);
     end
-    giq_option(4, 1372, 105, Node999, 50);
+    giq_option(4, NAME, 105, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node002
 begin
-    gsay_reply(1372, 103);
-    giq_option(4, 1372, 105, Node999, 50);
-    set_global_var(735, 1);
+    Reply(103);
+    giq_option(4, NAME, 105, Node999, NEUTRAL_REACTION);
+    set_global_var(GVAR_PLANT_QUEST, 1);
 end
 
 procedure Node003
 begin
-    gsay_reply(1372, 104);
-    giq_option(4, 1372, 105, Node999, 50);
-    set_global_var(735, 0);
+    Reply(104);
+    giq_option(4, NAME, 105, Node999, NEUTRAL_REACTION);
+    set_global_var(GVAR_PLANT_QUEST, 0);
 end
 
 procedure Node999
@@ -216,4 +216,3 @@ begin
         set_local_var(4, 1);
     end
 end
-
