diff --git a/Mods/MegaMod/Original/EPAI7.ssl b/Mods/MegaMod/ReDefine/EPAI7.ssl
index c17ef9f..b15754e 100644
--- a/Mods/MegaMod/Original/EPAI7.ssl
+++ b/Mods/MegaMod/ReDefine/EPAI7.ssl
@@ -145,13 +145,13 @@ end
 procedure description_p_proc
 begin
     script_overrides;
-    display_msg(message_str(1380, 104));
+    display_mstr(104);
 end
 
 procedure look_at_p_proc
 begin
     script_overrides;
-    display_msg(message_str(1380, 103));
+    display_mstr(103);
 end
 
 procedure use_obj_on_p_proc
@@ -159,11 +159,11 @@ begin
     variable LVar0 := 0;
     LVar0 := obj_pid(obj_being_used_with);
     if (local_var(4) == 1) then begin
-        display_msg(message_str(1380, 102));
+        display_mstr(102);
     end
     if ((LVar0 == 627) and (local_var(4) != 1)) then begin
         script_overrides;
-        display_msg(message_str(1380, 101));
+        display_mstr(101);
         obj_unlock(self_obj);
         set_local_var(4, 1);
     end
@@ -176,7 +176,7 @@ begin
     if (local_var(4) == 0) then begin
         if (source_obj == dude_obj) then begin
             script_overrides;
-            display_msg(message_str(1380, 100));
+            display_mstr(100);
         end
     end
 end
@@ -196,4 +196,3 @@ end
 procedure Node999
 begin
 end
-
