diff --git a/Mods/MegaMod/Original/EPAI2.ssl b/Mods/MegaMod/ReDefine/EPAI2.ssl
index ddd774b..a28f56e 100644
--- a/Mods/MegaMod/Original/EPAI2.ssl
+++ b/Mods/MegaMod/ReDefine/EPAI2.ssl
@@ -148,7 +148,7 @@ end
 procedure timed_event_p_proc
 begin
     if (fixed_param == 3) then begin
-        float_msg(self_obj, message_str(1366, random(112, 114)), 9);
+        float_msg(self_obj, mstr(random(112, 114)), FLOAT_MSG_WHITE);
         add_timer_event(self_obj, game_ticks(random(5, 10)), 3);
     end
 end
@@ -157,15 +157,15 @@ procedure use_skill_on_p_proc
 begin
     script_overrides;
     if (action_being_used == 12) then begin
-        if ((has_skill(source_obj, 12) < 50) and (global_var(733) == 3)) then begin
-            display_msg(message_str(1366, 108));
+        if ((has_skill(source_obj, SKILL_SCIENCE) < 50) and (global_var(GVAR_ELECTRIC_CHAIR) == 3)) then begin
+            display_mstr(108);
         end
         else begin
-            if ((has_skill(source_obj, 12) > 75) and (global_var(733) == 3)) then begin
-                display_msg(message_str(1366, 109));
-                set_global_var(733, 4);
+            if ((has_skill(source_obj, SKILL_SCIENCE) > 75) and (global_var(GVAR_ELECTRIC_CHAIR) == 3)) then begin
+                display_mstr(109);
+                set_global_var(GVAR_ELECTRIC_CHAIR, 4);
                 give_exp_points(500);
-                display_msg(message_str(14, 100) + 500 + message_str(14, 101));
+                display_msg(g_mstr(100) + 500 + g_mstr(101));
             end
         end
     end
@@ -173,7 +173,7 @@ end
 
 procedure use_p_proc
 begin
-    if (global_var(733) == 4) then begin
+    if (global_var(GVAR_ELECTRIC_CHAIR) == 4) then begin
         start_gdialog(1366, self_obj, 4, -1, -1);
         gsay_start;
         call Node004();
@@ -203,31 +203,31 @@ end
 
 procedure Node001
 begin
-    gsay_reply(1366, 100);
-    giq_option(4, 1366, 101, Node002, 50);
-    if (global_var(733) == 2) then begin
-        giq_option(4, 1366, 102, Node003, 50);
+    Reply(100);
+    giq_option(4, NAME, 101, Node002, NEUTRAL_REACTION);
+    if (global_var(GVAR_ELECTRIC_CHAIR) == 2) then begin
+        giq_option(4, NAME, 102, Node003, NEUTRAL_REACTION);
     end
-    giq_option(4, 1366, 103, Node999, 50);
+    giq_option(4, NAME, 103, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node002
 begin
-    gsay_reply(1366, 104);
-    giq_option(4, 1366, 105, Node001, 50);
+    Reply(104);
+    giq_option(4, NAME, 105, Node001, NEUTRAL_REACTION);
 end
 
 procedure Node003
 begin
-    gsay_reply(1366, 106);
-    giq_option(4, 1366, 107, Node999, 50);
-    set_global_var(733, 3);
+    Reply(106);
+    giq_option(4, NAME, 107, Node999, NEUTRAL_REACTION);
+    set_global_var(GVAR_ELECTRIC_CHAIR, 3);
 end
 
 procedure Node004
 begin
-    gsay_reply(1366, 110);
-    giq_option(4, 1366, 111, Node999, 50);
+    Reply(110);
+    giq_option(4, NAME, 111, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node999
@@ -237,4 +237,3 @@ begin
         set_local_var(4, 1);
     end
 end
-
