diff --git a/Mods/MegaMod/Original/EPAI22.ssl b/Mods/MegaMod/ReDefine/EPAI22.ssl
index 5f37b97..54b5ee5 100644
--- a/Mods/MegaMod/Original/EPAI22.ssl
+++ b/Mods/MegaMod/ReDefine/EPAI22.ssl
@@ -142,13 +142,13 @@ end
 procedure look_at_p_proc
 begin
     script_overrides;
-    display_msg(message_str(1410, 101));
+    display_mstr(101);
 end
 
 procedure description_p_proc
 begin
     script_overrides;
-    display_msg(message_str(1410, 100));
+    display_mstr(100);
 end
 
 procedure use_p_proc
@@ -166,4 +166,3 @@ end
 procedure drop_p_proc
 begin
 end
-
