diff --git a/Mods/MegaMod/Original/EPAINLFT.ssl b/Mods/MegaMod/ReDefine/EPAINLFT.ssl
index dac613a..baa126a 100644
--- a/Mods/MegaMod/Original/EPAINLFT.ssl
+++ b/Mods/MegaMod/ReDefine/EPAINLFT.ssl
@@ -143,7 +143,7 @@ end
 
 procedure spatial_p_proc
 begin
-    metarule(15, 6);
+    elevator(ELEV_GLOW1);
     set_map_var(15, 0);
     set_map_var(16, 0);
 end
@@ -156,4 +156,3 @@ end
 procedure map_update_p_proc
 begin
 end
-
