diff --git a/Mods/MegaMod/Original/EPAI27.ssl b/Mods/MegaMod/ReDefine/EPAI27.ssl
index e3a1e39..5dea1e3 100644
--- a/Mods/MegaMod/Original/EPAI27.ssl
+++ b/Mods/MegaMod/ReDefine/EPAI27.ssl
@@ -140,16 +140,16 @@ end
 procedure use_p_proc
 begin
     script_overrides;
-    if (get_critter_stat(dude_obj, 1) < 7) then begin
-        display_msg(message_str(1417, 100));
+    if (dude_perception < 7) then begin
+        display_mstr(100);
     end
     else begin
-        if (get_critter_stat(dude_obj, 1) >= 7) then begin
+        if (dude_perception >= 7) then begin
             if (local_var(4) == 1) then begin
-                display_msg(message_str(1417, 101));
+                display_mstr(101);
             end
             else begin
-                display_msg(message_str(1417, 102));
+                display_mstr(102);
                 set_local_var(4, 1);
                 move_obj_inven_to_obj(self_obj, dude_obj);
             end
@@ -168,4 +168,3 @@ end
 procedure drop_p_proc
 begin
 end
-
