diff --git a/Mods/MegaMod/Original/EPAI18.ssl b/Mods/MegaMod/ReDefine/EPAI18.ssl
index 7ed69df..6e7fb18 100644
--- a/Mods/MegaMod/Original/EPAI18.ssl
+++ b/Mods/MegaMod/ReDefine/EPAI18.ssl
@@ -141,7 +141,7 @@ end
 procedure damage_p_proc
 begin
     variable LVar0 := 0;
-    if (metarule(49, target_obj) == 6) then begin
+    if (weapon_dmg_type(target_obj) == 6) then begin
         LVar0 := random(10, 20);
         explosion(tile_num(self_obj), elevation(self_obj), LVar0);
         destroy_object(self_obj);
@@ -163,4 +163,3 @@ end
 procedure drop_p_proc
 begin
 end
-
