diff --git a/Mods/MegaMod/Original/EPAI8.ssl b/Mods/MegaMod/ReDefine/EPAI8.ssl
index bc3ccab..92192cc 100644
--- a/Mods/MegaMod/Original/EPAI8.ssl
+++ b/Mods/MegaMod/ReDefine/EPAI8.ssl
@@ -143,10 +143,10 @@ procedure use_skill_on_p_proc
 begin
     script_overrides;
     if (action_being_used == 12) then begin
-        display_msg(message_str(1382, 103));
+        display_mstr(103);
     end
     if (action_being_used == 13) then begin
-        display_msg(message_str(1382, 104));
+        display_mstr(104);
     end
 end
 
@@ -154,7 +154,7 @@ procedure look_at_p_proc
 begin
     script_overrides;
     if (local_var(4) == 0) then begin
-        display_msg(message_str(1382, 102));
+        display_mstr(102);
     end
 end
 
@@ -162,13 +162,13 @@ procedure use_p_proc
 begin
     if (local_var(4) == 0) then begin
         script_overrides;
-        display_msg(message_str(1382, 100));
+        display_mstr(100);
         set_local_var(4, 1);
     end
     else begin
         if (local_var(4) == 1) then begin
             script_overrides;
-            display_msg(message_str(1382, 101));
+            display_mstr(101);
         end
     end
 end
@@ -184,4 +184,3 @@ end
 procedure drop_p_proc
 begin
 end
-
