diff --git a/Mods/MegaMod/Original/EPAI28.ssl b/Mods/MegaMod/ReDefine/EPAI28.ssl
index 2d2879f..0db5263 100644
--- a/Mods/MegaMod/Original/EPAI28.ssl
+++ b/Mods/MegaMod/ReDefine/EPAI28.ssl
@@ -145,7 +145,7 @@ begin
         animate_stand_obj(self_obj);
     end
     else begin
-        display_msg(message_str(1418, 100));
+        display_mstr(100);
     end
 end
 
@@ -160,4 +160,3 @@ end
 procedure drop_p_proc
 begin
 end
-
