diff --git a/Mods/MegaMod/Original/EPAI17.ssl b/Mods/MegaMod/ReDefine/EPAI17.ssl
index ae20160..508ce5d 100644
--- a/Mods/MegaMod/Original/EPAI17.ssl
+++ b/Mods/MegaMod/ReDefine/EPAI17.ssl
@@ -148,14 +148,14 @@ procedure use_skill_on_p_proc
 begin
     script_overrides;
     if (action_being_used == 12) then begin
-        if (has_skill(dude_obj, 12) > 40) then begin
+        if (dude_skill(SKILL_SCIENCE) > 40) then begin
             if (local_var(6) == 0) then begin
-                set_global_var(750, 1);
+                set_global_var(GVAR_FETCH_GECKOPELTS, 1);
                 set_local_var(6, 1);
-                float_msg(self_obj, message_str(1399, 108), 8);
+                float_msg(self_obj, mstr(108), FLOAT_MSG_YELLOW);
                 give_exp_points(100);
-                display_msg(message_str(14, 100) + 100 + message_str(14, 101));
-                display_msg(message_str(1399, 109));
+                display_msg(g_mstr(100) + 100 + g_mstr(101));
+                display_mstr(109);
             end
         end
     end
@@ -165,17 +165,17 @@ procedure look_at_p_proc
 begin
     script_overrides;
     if (local_var(4) == 0) then begin
-        display_msg(message_str(1399, 105));
+        display_mstr(105);
     end
     else begin
-        display_msg(message_str(1399, 106));
+        display_mstr(106);
     end
 end
 
 procedure description_p_proc
 begin
     script_overrides;
-    display_msg(message_str(1399, 107));
+    display_mstr(107);
 end
 
 procedure use_p_proc
@@ -201,17 +201,17 @@ end
 
 procedure Node001
 begin
-    gsay_reply(1399, 100);
+    Reply(100);
     if (local_var(5) != 1) then begin
-        giq_option(4, 1399, 101, Node002, 50);
+        giq_option(4, NAME, 101, Node002, NEUTRAL_REACTION);
     end
-    giq_option(4, 1399, 102, Node999, 50);
+    giq_option(4, NAME, 102, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node002
 begin
-    gsay_reply(1399, 103);
-    giq_option(4, 1399, 104, Node002a, 50);
+    Reply(103);
+    giq_option(4, NAME, 104, Node002a, NEUTRAL_REACTION);
 end
 
 procedure Node002a
@@ -226,4 +226,3 @@ begin
         set_local_var(4, 1);
     end
 end
-
