diff --git a/Mods/MegaMod/Original/EPAKURT.ssl b/Mods/MegaMod/ReDefine/EPAKURT.ssl
index f228a58..0d53948 100644
--- a/Mods/MegaMod/Original/EPAKURT.ssl
+++ b/Mods/MegaMod/ReDefine/EPAKURT.ssl
@@ -49,20 +49,20 @@ end
 procedure critter_p_proc
 begin
     if ((map_var(2) == 1) and (elevation(dude_obj) == 1)) then begin
-        float_msg(self_obj, message_str(1701, 600), 2);
+        float_msg(self_obj, mstr(600), FLOAT_MSG_RED);
         set_local_var(5, 2);
     end
-    if ((local_var(5) == 2) or (global_var(1098) == 1) and obj_can_see_obj(self_obj, dude_obj)) then begin
+    if ((local_var(5) == 2) or (global_var(GVAR_PRIMIT_MONK_STATE) == 1) and obj_can_see_obj(self_obj, dude_obj)) then begin
         set_local_var(5, 1);
-        attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+        attack(dude_obj);
     end
-    if ((tile_num(dude_obj) == 17315) or (tile_num(dude_obj) == 17515) or (tile_num(dude_obj) == 17715) and (elevation(dude_obj) == 1)) then begin
+    if ((dude_tile == 17315) or (dude_tile == 17515) or (dude_tile == 17715) and (elevation(dude_obj) == 1)) then begin
         if (map_var(17) == 2) then begin
-            float_msg(self_obj, 600, 2);
+            float_msg(self_obj, 600, FLOAT_MSG_RED);
             call Node998();
         end
         else begin
-            if ((map_var(15) == 0) and (global_var(1081) < 5)) then begin
+            if ((map_var(15) == 0) and (global_var(GVAR_BOS_TO_BH_QUEST) < 5)) then begin
                 move_to(dude_obj, 17525, elevation(self_obj));
                 start_gdialog(1701, self_obj, -1, -1, -1);
                 gsay_start;
@@ -73,7 +73,7 @@ begin
             else begin
                 if ((map_var(15) == 1) and ((game_time - map_var(16)) > (12 * (60 * (60 * 10))))) then begin
                     add_timer_event(self_obj, 30 * 10, 9);
-                    float_msg(self_obj, message_str(1701, 500), 8);
+                    float_msg(self_obj, mstr(500), FLOAT_MSG_YELLOW);
                 end
             end
         end
@@ -88,17 +88,17 @@ procedure look_at_p_proc
 begin
     script_overrides;
     if (local_var(4) == 0) then begin
-        display_msg(message_str(1701, 100));
+        display_mstr(100);
     end
     else begin
-        display_msg(message_str(1701, 101));
+        display_mstr(101);
     end
 end
 
 procedure description_p_proc
 begin
     script_overrides;
-    display_msg(message_str(1701, 102));
+    display_mstr(102);
 end
 
 procedure use_skill_on_p_proc
@@ -111,13 +111,13 @@ end
 
 procedure damage_p_proc
 begin
-    set_global_var(1098, 1);
+    set_global_var(GVAR_PRIMIT_MONK_STATE, 1);
 end
 
 procedure map_enter_p_proc
 begin
     set_local_var(0, tile_num(self_obj));
-    set_local_var(1, has_trait(1, self_obj, 10));
+    set_local_var(1, obj_get_rot(self_obj));
 end
 
 procedure map_update_p_proc
@@ -126,7 +126,7 @@ end
 
 procedure talk_p_proc
 begin
-    if ((map_var(15) == 0) and (global_var(1081) < 5)) then begin
+    if ((map_var(15) == 0) and (global_var(GVAR_BOS_TO_BH_QUEST) < 5)) then begin
         start_gdialog(1701, self_obj, -1, -1, -1);
         gsay_start;
         call Node001();
@@ -134,16 +134,16 @@ begin
         end_dialogue;
     end
     else begin
-        float_msg(self_obj, message_str(1701, 400), 8);
+        float_msg(self_obj, mstr(400), FLOAT_MSG_YELLOW);
     end
 end
 
 procedure Node001
 begin
-    gsay_reply(1701, 200);
-    gsay_option(1701, 210, Node002, -1);
-    if (has_skill(dude_obj, 14) > 120) then begin
-        gsay_option(1701, 211, Node003, -1);
+    Reply(200);
+    giq_option(1, NAME, 210, Node002, -1);
+    if (has_skill(dude_obj, SKILL_CONVERSANT) > 120) then begin
+        giq_option(1, NAME, 211, Node003, -1);
     end
 end
 
@@ -151,13 +151,13 @@ procedure Node002
 begin
     set_map_var(15, 1);
     set_map_var(16, game_time);
-    gsay_message(1701, 400, 49);
+    gsay_message(NAME, 400, GOOD_REACTION);
 end
 
 procedure Node003
 begin
     set_map_var(15, 5);
-    gsay_message(1701, 300, 49);
+    gsay_message(NAME, 300, GOOD_REACTION);
 end
 
 procedure Node998
@@ -168,4 +168,3 @@ end
 procedure Node999
 begin
 end
-
