diff --git a/Mods/MegaMod/Original/EPAS3.ssl b/Mods/MegaMod/ReDefine/EPAS3.ssl
index 00048ea..e0ef788 100644
--- a/Mods/MegaMod/Original/EPAS3.ssl
+++ b/Mods/MegaMod/ReDefine/EPAS3.ssl
@@ -138,7 +138,7 @@ end
 
 procedure spatial_p_proc
 begin
-    if (global_var(735) == 1) then begin
+    if (global_var(GVAR_PLANT_QUEST) == 1) then begin
         critter_dmg(source_obj, random(450, 550), 4);
     end
 end
@@ -150,4 +150,3 @@ end
 procedure map_update_p_proc
 begin
 end
-
