diff --git a/Mods/MegaMod/Original/EPAVTCOM.ssl b/Mods/MegaMod/ReDefine/EPAVTCOM.ssl
index 796cc65..943ead8 100644
--- a/Mods/MegaMod/Original/EPAVTCOM.ssl
+++ b/Mods/MegaMod/ReDefine/EPAVTCOM.ssl
@@ -210,7 +210,7 @@ procedure use_p_proc
 begin
     dialog_part := 0;
     call clear_all();
-    if ((global_var(1072) == 2) and (map_var(9) == 1)) then begin
+    if ((global_var(GVAR_BRTHD_NETWORK) == 2) and (map_var(9) == 1)) then begin
         start_gdialog(1684, self_obj, -1, -1, -1);
         gsay_start;
         call Node001();
@@ -253,27 +253,27 @@ end
 
 procedure Node001
 begin
-    gsay_reply(1684, 2000);
+    gsay_reply(SCRIPT_EPADAN, 2000);
     if (not((map_var(0) bwand 1) != 0)) then begin
-        gsay_option(1684, 2010, command1, -1);
+        giq_option(1, SCRIPT_EPADAN, 2010, command1, -1);
     end
     if (not((map_var(0) bwand 8) != 0)) then begin
-        gsay_option(1684, 2011, command4, -1);
+        giq_option(1, SCRIPT_EPADAN, 2011, command4, -1);
     end
     if (not((map_var(0) bwand 4) != 0)) then begin
-        gsay_option(1684, 2012, command3, -1);
+        giq_option(1, SCRIPT_EPADAN, 2012, command3, -1);
     end
     if (not((map_var(0) bwand 32) != 0)) then begin
-        gsay_option(1684, 2013, command6, -1);
+        giq_option(1, SCRIPT_EPADAN, 2013, command6, -1);
     end
     if (not((map_var(0) bwand 2) != 0)) then begin
-        gsay_option(1684, 2014, command2, -1);
+        giq_option(1, SCRIPT_EPADAN, 2014, command2, -1);
     end
     if (not((map_var(0) bwand 16) != 0)) then begin
-        gsay_option(1684, 2015, command5, -1);
+        giq_option(1, SCRIPT_EPADAN, 2015, command5, -1);
     end
-    gsay_option(1684, 2016, clear_all, -1);
-    gsay_option(1684, 3012, Node999, -1);
+    giq_option(1, SCRIPT_EPADAN, 2016, clear_all, -1);
+    giq_option(1, SCRIPT_EPADAN, 3012, Node999, -1);
 end
 
 procedure Node999
@@ -364,4 +364,3 @@ end
 procedure timed_event_p_proc
 begin
 end
-
