diff --git a/Mods/MegaMod/Original/ESMINCRT.ssl b/Mods/MegaMod/ReDefine/ESMINCRT.ssl
index b96ec51..6a2c668 100644
--- a/Mods/MegaMod/Original/ESMINCRT.ssl
+++ b/Mods/MegaMod/ReDefine/ESMINCRT.ssl
@@ -146,10 +146,10 @@ begin
     if (obj_pid(self_obj) == 33555586) then begin
         if (is_success(do_check(source_obj, 0, 0))) then begin
             gfade_out(10);
-            set_global_var(215, global_var(215) bwor 2);
-            display_msg(message_str(367, 107));
+            set_global_var(GVAR_MILITARY_BASE_FLAGS, global_var(GVAR_MILITARY_BASE_FLAGS) bwor 2);
+            display_mstr(107);
             give_exp_points(5000);
-            display_msg(message_str(14, 100) + 5000 + message_str(14, 101));
+            display_msg(g_mstr(100) + 5000 + g_mstr(101));
             destroy_object(tile_contains_pid_obj(21474, elevation(self_obj), 33554499));
             destroy_object(tile_contains_pid_obj(21274, elevation(self_obj), 33554499));
             destroy_object(tile_contains_pid_obj(21273, elevation(self_obj), 33554499));
@@ -168,33 +168,33 @@ begin
             Scenery_Creation_Count := 0;
             while (Scenery_Creation_Count < 1) do begin
                 Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 4, 1);
-                Scenery_Creation := create_object_sid(33554499, Scenery_Creation_Hex, elevation(self_obj), -1);
+                Scenery_Creation := create_object(33554499, Scenery_Creation_Hex, elevation(self_obj));
                 Scenery_Creation_Count := Scenery_Creation_Count + 1;
             end
             Scenery_Creation_Hex := 20877;
             Scenery_Creation_Count := 0;
             while (Scenery_Creation_Count < 1) do begin
                 Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 5, 1);
-                Scenery_Creation := create_object_sid(33554499, Scenery_Creation_Hex, elevation(self_obj), -1);
+                Scenery_Creation := create_object(33554499, Scenery_Creation_Hex, elevation(self_obj));
                 Scenery_Creation_Count := Scenery_Creation_Count + 1;
             end
-            move_to(create_object_sid(33556060, 20877, elevation(self_obj), -1), 20877, elevation(self_obj));
+            move_to(create_object(33556060, 20877, elevation(self_obj)), 20877, elevation(self_obj));
             set_obj_visibility(base_enterance_rocks, 1);
             explosion(tile_num(base_enterance_rocks), elevation(self_obj), 500);
-            move_to(create_object_sid(33556235, 20475, elevation(self_obj), -1), 20475, elevation(self_obj));
-            move_to(create_object_sid(33556236, 20674, elevation(self_obj), -1), 20674, elevation(self_obj));
-            move_to(create_object_sid(33556237, 21875, elevation(self_obj), -1), 21875, elevation(self_obj));
-            move_to(create_object_sid(33556238, 21678, elevation(self_obj), -1), 21678, elevation(self_obj));
+            move_to(create_object(33556235, 20475, elevation(self_obj)), 20475, elevation(self_obj));
+            move_to(create_object(33556236, 20674, elevation(self_obj)), 20674, elevation(self_obj));
+            move_to(create_object(33556237, 21875, elevation(self_obj)), 21875, elevation(self_obj));
+            move_to(create_object(33556238, 21678, elevation(self_obj)), 21678, elevation(self_obj));
             destroy_object(base_enterance_rocks);
             gfade_in(10);
             destroy_object(self_obj);
         end
         else begin
-            display_msg(message_str(367, 106));
+            display_mstr(106);
         end
     end
     else begin
-        display_msg(message_str(367, 108));
+        display_mstr(108);
     end
 end
 
@@ -211,9 +211,9 @@ begin
             script_overrides;
             LVar0 := rm_mult_objs_from_inven(source_obj, LVar1, 1);
             destroy_object(LVar1);
-            move_to(create_object_sid(33555585, tile_num(self_obj), elevation(self_obj), 367), tile_num(self_obj), elevation(self_obj));
+            move_to(create_object_sid(33555585, tile_num(self_obj), elevation(self_obj), NAME), tile_num(self_obj), elevation(self_obj));
             if (111 != 0) then begin
-                display_msg(message_str(367, 111));
+                display_mstr(111);
             end
             destroy_object(self_obj);
         end
@@ -229,9 +229,9 @@ begin
                 else begin
                     destroy_object(LVar1);
                 end
-                move_to(create_object_sid(33555586, tile_num(self_obj), elevation(self_obj), 367), tile_num(self_obj), elevation(self_obj));
+                move_to(create_object_sid(33555586, tile_num(self_obj), elevation(self_obj), NAME), tile_num(self_obj), elevation(self_obj));
                 if (112 != 0) then begin
-                    display_msg(message_str(367, 112));
+                    display_mstr(112);
                 end
                 destroy_object(self_obj);
             end
@@ -245,14 +245,14 @@ procedure look_at_p_proc
 begin
     script_overrides;
     if (obj_pid(self_obj) == 33555586) then begin
-        display_msg(message_str(367, 104));
+        display_mstr(104);
     end
     else begin
         if (obj_pid(self_obj) == 33555585) then begin
-            display_msg(message_str(367, 103));
+            display_mstr(103);
         end
         else begin
-            display_msg(message_str(367, 102));
+            display_mstr(102);
         end
     end
 end
@@ -261,14 +261,14 @@ procedure description_p_proc
 begin
     script_overrides;
     if (obj_pid(self_obj) == 33555586) then begin
-        display_msg(message_str(367, 105));
+        display_mstr(105);
     end
     else begin
         if (obj_pid(self_obj) == 33555585) then begin
-            display_msg(message_str(367, 103));
+            display_mstr(103);
         end
         else begin
-            display_msg(message_str(367, 110));
+            display_mstr(110);
         end
     end
 end
@@ -277,11 +277,10 @@ procedure damage_p_proc
 begin
     debug_msg("cart damage p proc");
     if (obj_pid(self_obj) == 33555586) then begin
-        move_to(create_object_sid(33555585, tile_num(self_obj), elevation(self_obj), 367), tile_num(self_obj), elevation(self_obj));
+        move_to(create_object_sid(33555585, tile_num(self_obj), elevation(self_obj), NAME), tile_num(self_obj), elevation(self_obj));
         if (0 != 0) then begin
-            display_msg(message_str(367, 0));
+            display_mstr(0);
         end
         destroy_object(self_obj);
     end
 end
-
