diff --git a/Mods/MegaMod/Original/FCDOKGRD.ssl b/Mods/MegaMod/ReDefine/FCDOKGRD.ssl
index 4b7a49c..dfcbb86 100644
--- a/Mods/MegaMod/Original/FCDOKGRD.ssl
+++ b/Mods/MegaMod/ReDefine/FCDOKGRD.ssl
@@ -49,17 +49,17 @@ end
 procedure look_at_p_proc
 begin
     script_overrides;
-    display_msg(message_str(2471, 100));
+    display_mstr(100);
 end
 
 procedure description_p_proc
 begin
     script_overrides;
-    if (global_var(362) == 0) then begin
-        display_msg(message_str(2471, 101));
+    if (global_var(GVAR_SAN_FRAN_SUB) == 0) then begin
+        display_mstr(101);
     end
     else begin
-        display_msg(message_str(2471, 102));
+        display_mstr(102);
     end
 end
 
@@ -76,14 +76,14 @@ begin
     if (source_obj == dude_obj) then begin
         set_local_var(5, 2);
         set_map_var(0, 0);
-        set_global_var(361, global_var(361) bwor 1);
+        set_global_var(GVAR_SAN_FRAN_FLAGS, global_var(GVAR_SAN_FRAN_FLAGS) bwor 1);
     end
 end
 
 procedure map_enter_p_proc
 begin
     boatGrd_ptr := self_obj;
-    if (metarule(14, 0)) then begin
+    if (map_first_run) then begin
         set_local_var(10, tile_num(self_obj));
     end
     add_timer_event(self_obj, game_ticks(random(10, 15)), 2);
@@ -96,7 +96,7 @@ end
 
 procedure talk_p_proc
 begin
-    if (global_var(362) >= 5) then begin
+    if (global_var(GVAR_SAN_FRAN_SUB) >= 5) then begin
         call Node010();
     end
     else begin
@@ -121,82 +121,82 @@ procedure Node001
 begin
     if (local_var(4) == 0) then begin
         set_local_var(4, 1);
-        gsay_reply(2471, 200);
+        Reply(200);
     end
     else begin
-        gsay_reply(2471, 201);
+        Reply(201);
     end
-    giq_option(4, 2471, 202, Node002, 50);
-    giq_option(4, 2471, 203, Node900, 50);
-    giq_option(4, 2471, 204, Node999, 50);
+    giq_option(4, NAME, 202, Node002, NEUTRAL_REACTION);
+    giq_option(4, NAME, 203, Node900, NEUTRAL_REACTION);
+    giq_option(4, NAME, 204, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node002
 begin
-    gsay_reply(2471, 210);
-    giq_option(4, 2471, 211, Node003, 50);
+    Reply(210);
+    giq_option(4, NAME, 211, Node003, NEUTRAL_REACTION);
 end
 
 procedure Node003
 begin
-    gsay_reply(2471, 220);
-    giq_option(4, 2471, 221, Node999, 50);
-    giq_option(4, 2471, 222, Node998, 51);
+    Reply(220);
+    giq_option(4, NAME, 221, Node999, NEUTRAL_REACTION);
+    giq_option(4, NAME, 222, Node998, BAD_REACTION);
 end
 
 procedure Node004
 begin
-    gsay_reply(2471, 230);
-    if ((global_var(362) < 3) and (has_skill(dude_obj, 14) > 95) and ((global_var(494) >= 4) and (global_var(494) < 8))) then begin
-        giq_option(4, 2471, 231, Node005, 50);
+    Reply(230);
+    if ((global_var(GVAR_SAN_FRAN_SUB) < 3) and (dude_skill(SKILL_SPEECH) > 95) and ((global_var(GVAR_FALLOUT_2) >= 4) and (global_var(GVAR_FALLOUT_2) < 8))) then begin
+        giq_option(4, NAME, 231, Node005, NEUTRAL_REACTION);
     end
-    if ((global_var(362) == 3) or (global_var(362) == 4)) then begin
-        giq_option(4, 2471, 232, Node007, 50);
+    if ((global_var(GVAR_SAN_FRAN_SUB) == 3) or (global_var(GVAR_SAN_FRAN_SUB) == 4)) then begin
+        giq_option(4, NAME, 232, Node007, NEUTRAL_REACTION);
     end
-    giq_option(4, 2471, 233, Node998, 51);
-    giq_option(4, 2471, 234, Node999, 50);
+    giq_option(4, NAME, 233, Node998, BAD_REACTION);
+    giq_option(4, NAME, 234, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node005
 begin
-    gsay_reply(2471, 240);
-    giq_option(4, 2471, 241, Node006, 50);
-    display_msg(message_str(2471, 500));
+    Reply(240);
+    giq_option(4, NAME, 241, Node006, NEUTRAL_REACTION);
+    display_mstr(500);
     give_exp_points(1000);
-    display_msg(message_str(14, 100) + 1000 + message_str(14, 101));
+    display_msg(g_mstr(100) + 1000 + g_mstr(101));
 end
 
 procedure Node006
 begin
-    gsay_reply(2471, 250);
-    giq_option(4, 2471, 251, Node008, 50);
-    giq_option(4, 2471, 252, Node990, 50);
+    Reply(250);
+    giq_option(4, NAME, 251, Node008, NEUTRAL_REACTION);
+    giq_option(4, NAME, 252, Node990, NEUTRAL_REACTION);
 end
 
 procedure Node007
 begin
-    gsay_reply(2471, 260);
-    giq_option(4, 2471, 261, Node993, 50);
-    giq_option(4, 2471, 262, Node991, 50);
+    Reply(260);
+    giq_option(4, NAME, 261, Node993, NEUTRAL_REACTION);
+    giq_option(4, NAME, 262, Node991, NEUTRAL_REACTION);
 end
 
 procedure Node008
 begin
-    gsay_reply(2471, 270);
-    giq_option(4, 2471, 271, Node992, 50);
-    giq_option(4, 2471, 272, Node992, 50);
+    Reply(270);
+    giq_option(4, NAME, 271, Node992, NEUTRAL_REACTION);
+    giq_option(4, NAME, 272, Node992, NEUTRAL_REACTION);
 end
 
 procedure Node009
 begin
-    gsay_reply(2471, 280);
-    giq_option(4, 2471, 281, Node993, 50);
-    giq_option(4, 2471, 282, Node991, 50);
+    Reply(280);
+    giq_option(4, NAME, 281, Node993, NEUTRAL_REACTION);
+    giq_option(4, NAME, 282, Node991, NEUTRAL_REACTION);
 end
 
 procedure Node010
 begin
-    float_msg(self_obj, message_str(2471, 290), 8);
+    float_msg(self_obj, mstr(290), FLOAT_MSG_YELLOW);
 end
 
 procedure Node999
@@ -206,7 +206,7 @@ end
 procedure Node998
 begin
     set_local_var(5, 2);
-    set_global_var(361, global_var(361) bwor 1);
+    set_global_var(GVAR_SAN_FRAN_FLAGS, global_var(GVAR_SAN_FRAN_FLAGS) bwor 1);
 end
 
 procedure Node900
@@ -239,8 +239,8 @@ procedure Node993
 begin
     set_local_var(7, 1);
     set_local_var(8, 0);
-    set_global_var(632, global_var(632) bwor 262144);
-    if (global_var(769)) then begin
+    set_global_var(GVAR_PARTY_MEMBERS_HIDDEN, global_var(GVAR_PARTY_MEMBERS_HIDDEN) bwor 262144);
+    if (global_var(GVAR_BOS_LOCATION_ELEVATORS)) then begin
         set_obj_visibility(party_member_obj(16777278), 1);
         set_obj_visibility(party_member_obj(16777376), 1);
         set_obj_visibility(party_member_obj(16777377), 1);
@@ -268,7 +268,7 @@ begin
         set_obj_visibility(party_member_obj(16777785), 1);
         set_obj_visibility(party_member_obj(16777801), 1);
         set_obj_visibility(party_member_obj(16777293), 1);
-        set_global_var(769, 1);
+        set_global_var(GVAR_BOS_LOCATION_ELEVATORS, 1);
     end
     game_time_advance((24 * (60 * (60 * 10)) * (game_time_hour > 0)) + (((0 / 100) - (game_time_hour / 100)) * (60 * (60 * 10))) + (((0 % 100) - (game_time_hour % 100)) * (60 * 10)));
     game_time_advance(15 * (60 * 10));
@@ -335,7 +335,7 @@ begin
             if (local_var(9) > 3) then begin
                 set_local_var(9, 3);
             end
-            float_msg(self_obj, message_str(2471, 149 + local_var(9)), 2);
+            float_msg(self_obj, mstr(149 + local_var(9)), FLOAT_MSG_RED);
             if (local_var(9) == 3) then begin
                 call Node998();
             end
@@ -348,63 +348,63 @@ end
 procedure destroy_p_proc
 begin
     if (source_obj == dude_obj) then begin
-        set_global_var(5, global_var(5) + 1);
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(4) > (3 * global_var(5))) or (global_var(2) == 1)) and (global_var(1) == 0)) then begin
-            set_global_var(2, 1);
-            set_global_var(3, 0);
+        set_global_var(GVAR_GOOD_MONSTER, global_var(GVAR_GOOD_MONSTER) + 1);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(4) > (3 * global_var(GVAR_GOOD_MONSTER))) or (global_var(GVAR_CHAMPION_REPUTATION) == 1)) and (global_var(GVAR_CHILDKILLER_REPUTATION) == 0)) then begin
+            set_global_var(GVAR_CHAMPION_REPUTATION, 1);
+            set_global_var(GVAR_BERSERKER_REPUTATION, 0);
         end
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(5) > (2 * global_var(4))) or (global_var(3) == 1))) then begin
-            set_global_var(3, 1);
-            set_global_var(2, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
+            set_global_var(GVAR_BERSERKER_REPUTATION, 1);
+            set_global_var(GVAR_CHAMPION_REPUTATION, 0);
         end
-        set_global_var(0, global_var(0) + -10);
+        set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + -10);
         debug_msg("Player gains " + -10 + " Karma Points.");
-        set_global_var(37, 0);
-        set_global_var(38, 0);
-        set_global_var(39, 0);
-        set_global_var(40, 0);
-        set_global_var(41, 0);
-        set_global_var(42, 0);
-        set_global_var(43, 0);
-        set_global_var(44, 0);
-        set_global_var(45, 0);
-        tmp_gen_rep := global_var(0);
-        if (has_trait(0, dude_obj, 95)) then begin
-            tmp_gen_rep := global_var(0) * 2;
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+        set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+        set_global_var(GVAR_KARMA_DEFENDER, 0);
+        set_global_var(GVAR_KARMA_WANDERER, 0);
+        set_global_var(GVAR_KARMA_BETRAYER, 0);
+        set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+        set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+        tmp_gen_rep := global_var(GVAR_PLAYER_REPUTATION);
+        if (has_trait(TRAIT_PERK, dude_obj, PERK_karma_beacon_perk)) then begin
+            tmp_gen_rep := global_var(GVAR_PLAYER_REPUTATION) * 2;
         end
         if (tmp_gen_rep >= 1000) then begin
-            set_global_var(37, 1);
+            set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
         end
         else begin
             if (tmp_gen_rep >= 750) then begin
-                set_global_var(38, 1);
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
             end
             else begin
                 if (tmp_gen_rep >= 500) then begin
-                    set_global_var(39, 1);
+                    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                 end
                 else begin
                     if (tmp_gen_rep >= 250) then begin
-                        set_global_var(40, 1);
+                        set_global_var(GVAR_KARMA_DEFENDER, 1);
                     end
                     else begin
                         if (tmp_gen_rep > -250) then begin
-                            set_global_var(41, 1);
+                            set_global_var(GVAR_KARMA_WANDERER, 1);
                         end
                         else begin
                             if (tmp_gen_rep > -500) then begin
-                                set_global_var(42, 1);
+                                set_global_var(GVAR_KARMA_BETRAYER, 1);
                             end
                             else begin
                                 if (tmp_gen_rep > -750) then begin
-                                    set_global_var(43, 1);
+                                    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                 end
                                 else begin
                                     if (tmp_gen_rep > -1000) then begin
-                                        set_global_var(44, 1);
+                                        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                     end
                                     else begin
-                                        set_global_var(45, 1);
+                                        set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                     end
                                 end
                             end
@@ -416,7 +416,7 @@ begin
     end
     boatGrd_ptr := 0;
     set_map_var(0, 0);
-    set_global_var(362, 2);
+    set_global_var(GVAR_SAN_FRAN_SUB, 2);
 end
 
 procedure pickup_p_proc
@@ -428,17 +428,16 @@ end
 
 procedure critter_p_proc
 begin
-    if ((tile_num(self_obj) == 15519) and (has_trait(1, self_obj, 10) != 4)) then begin
+    if ((tile_num(self_obj) == 15519) and (obj_get_rot(self_obj) != 4)) then begin
         reg_anim_func(2, self_obj);
         anim(self_obj, 1000, 4);
     end
-    if ((tile_num(self_obj) == 15532) and (has_trait(1, self_obj, 10) != 0)) then begin
+    if ((tile_num(self_obj) == 15532) and (obj_get_rot(self_obj) != 0)) then begin
         reg_anim_func(2, self_obj);
         anim(self_obj, 1000, 0);
     end
     if ((local_var(5) == 2) and obj_can_see_obj(self_obj, dude_obj)) then begin
         set_local_var(5, 1);
-        attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+        attack(dude_obj);
     end
 end
-
