diff --git a/Mods/MegaMod/Original/FCLAOCHO.ssl b/Mods/MegaMod/ReDefine/FCLAOCHO.ssl
index cce557b..73f38fb 100644
--- a/Mods/MegaMod/Original/FCLAOCHO.ssl
+++ b/Mods/MegaMod/ReDefine/FCLAOCHO.ssl
@@ -193,24 +193,24 @@ procedure pickup_p_proc
 begin
     if (source_obj == dude_obj) then begin
         if (3 == 1) then begin
-            attack_complex(source_obj, 0, 1, 0, 0, 30000, 0, 0);
+            attack(source_obj);
         end
         else begin
             if (3 == 2) then begin
-                float_msg(self_obj, message_str(14, random(3325, 3329)), 0);
+                float_msg(self_obj, g_mstr(random(3325, 3329)), FLOAT_MSG_NORMAL);
                 if (7 > -1) then begin
                     if (local_var(7) >= 1) then begin
-                        attack_complex(source_obj, 0, 1, 0, 0, 30000, 0, 0);
+                        attack(source_obj);
                     end
                     set_local_var(7, local_var(7) + 1);
                 end
             end
             else begin
                 if (3 == 3) then begin
-                    float_msg(self_obj, message_str(14, random(3050, 3064)), 0);
+                    float_msg(self_obj, g_mstr(random(3050, 3064)), FLOAT_MSG_NORMAL);
                     if (7 > -1) then begin
                         if (local_var(7) >= 1) then begin
-                            attack_complex(source_obj, 0, 1, 0, 0, 30000, 0, 0);
+                            attack(source_obj);
                         end
                         set_local_var(7, local_var(7) + 1);
                     end
@@ -237,7 +237,7 @@ procedure critter_p_proc
 begin
     if ((local_var(5) == 2) and obj_can_see_obj(self_obj, dude_obj)) then begin
         set_local_var(5, 1);
-        attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+        attack(dude_obj);
     end
     if (tile_num(self_obj) != 23720) then begin
         if (not(anim_busy(self_obj))) then begin
@@ -254,48 +254,48 @@ begin
     Evil_Critter := 0;
     Slavery_Tolerant := 1;
     Karma_Perception := 1;
-    set_global_var(37, 0);
-    set_global_var(38, 0);
-    set_global_var(39, 0);
-    set_global_var(40, 0);
-    set_global_var(41, 0);
-    set_global_var(42, 0);
-    set_global_var(43, 0);
-    set_global_var(44, 0);
-    set_global_var(45, 0);
-    if (global_var(0) >= 1000) then begin
-        set_global_var(37, 1);
+    set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+    set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+    set_global_var(GVAR_KARMA_DEFENDER, 0);
+    set_global_var(GVAR_KARMA_WANDERER, 0);
+    set_global_var(GVAR_KARMA_BETRAYER, 0);
+    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+    set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+    set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+    if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
     end
     else begin
-        if (global_var(0) >= 750) then begin
-            set_global_var(38, 1);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+            set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
         end
         else begin
-            if (global_var(0) >= 500) then begin
-                set_global_var(39, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
             end
             else begin
-                if (global_var(0) >= 250) then begin
-                    set_global_var(40, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                    set_global_var(GVAR_KARMA_DEFENDER, 1);
                 end
                 else begin
-                    if (global_var(0) > -250) then begin
-                        set_global_var(41, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                        set_global_var(GVAR_KARMA_WANDERER, 1);
                     end
                     else begin
-                        if (global_var(0) > -500) then begin
-                            set_global_var(42, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                            set_global_var(GVAR_KARMA_BETRAYER, 1);
                         end
                         else begin
-                            if (global_var(0) > -750) then begin
-                                set_global_var(43, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                             end
                             else begin
-                                if (global_var(0) > -1000) then begin
-                                    set_global_var(44, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                    set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                 end
                                 else begin
-                                    set_global_var(45, 1);
+                                    set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                 end
                             end
                         end
@@ -306,7 +306,7 @@ begin
     end
     if (local_var(2) == 0) then begin
         set_local_var(2, 1);
-        set_local_var(3, (get_critter_stat(dude_obj, 3) - get_critter_stat(self_obj, 3)) * 5);
+        set_local_var(3, (dude_charisma - self_charisma) * 5);
         debug_msg("Base Reaction == " + local_var(3));
     end
     Static_Reaction := 0;
@@ -318,14 +318,14 @@ begin
     else begin
         reaction_bonus_karma := -1 * (-1 * reaction_bonus_karma / 1000);
     end
-    if (global_var(61) >= 0) then begin
-        reaction_bonus_town_rep := global_var(61) / 2;
+    if (global_var(GVAR_TOWN_REP_SAN_FRANCISCO) >= 0) then begin
+        reaction_bonus_town_rep := global_var(GVAR_TOWN_REP_SAN_FRANCISCO) / 2;
     end
     else begin
-        reaction_bonus_town_rep := -1 * (-1 * global_var(61) / 2);
+        reaction_bonus_town_rep := -1 * (-1 * global_var(GVAR_TOWN_REP_SAN_FRANCISCO) / 2);
     end
-    if (has_trait(2, dude_obj, 13)) then begin
-        if (get_critter_stat(dude_obj, 34) == get_critter_stat(self_obj, 34)) then begin
+    if (has_trait(TRAIT_TRAIT, dude_obj, TRAIT_sex_appeal)) then begin
+        if (dude_gender == self_gender) then begin
             Static_Reaction := Static_Reaction + -20;
         end
         else begin
@@ -337,58 +337,58 @@ begin
         debug_msg("Sex Appeal Bonus == 0");
     end
     if (Evil_Critter == 1) then begin
-        Static_Reaction := Static_Reaction - reaction_bonus_karma + (reaction_bonus_karma * has_trait(0, dude_obj, 39));
-        if ((reaction_bonus_karma < 0) and (has_trait(0, dude_obj, 39) != 0)) then begin
+        Static_Reaction := Static_Reaction - reaction_bonus_karma + (reaction_bonus_karma * has_trait(TRAIT_PERK, dude_obj, PERK_cult_of_personality));
+        if ((reaction_bonus_karma < 0) and (has_trait(TRAIT_PERK, dude_obj, PERK_cult_of_personality) != 0)) then begin
             Static_Reaction := Static_Reaction - reaction_bonus_karma;
         end
         debug_msg("Karma Reaction Bonus == " + Static_Reaction);
-        Static_Reaction := Static_Reaction - reaction_bonus_town_rep + (reaction_bonus_town_rep * has_trait(0, dude_obj, 39));
-        if ((reaction_bonus_town_rep < 0) and (has_trait(0, dude_obj, 39) != 0)) then begin
+        Static_Reaction := Static_Reaction - reaction_bonus_town_rep + (reaction_bonus_town_rep * has_trait(TRAIT_PERK, dude_obj, PERK_cult_of_personality));
+        if ((reaction_bonus_town_rep < 0) and (has_trait(TRAIT_PERK, dude_obj, PERK_cult_of_personality) != 0)) then begin
             Static_Reaction := Static_Reaction - reaction_bonus_town_rep;
         end
         debug_msg("Town Rep Bonus == " + Static_Reaction);
-        set_global_var(37, 0);
-        set_global_var(38, 0);
-        set_global_var(39, 0);
-        set_global_var(40, 0);
-        set_global_var(41, 0);
-        set_global_var(42, 0);
-        set_global_var(43, 0);
-        set_global_var(44, 0);
-        set_global_var(45, 0);
-        if (global_var(0) >= 1000) then begin
-            set_global_var(37, 1);
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+        set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+        set_global_var(GVAR_KARMA_DEFENDER, 0);
+        set_global_var(GVAR_KARMA_WANDERER, 0);
+        set_global_var(GVAR_KARMA_BETRAYER, 0);
+        set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+        set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+            set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
         end
         else begin
-            if (global_var(0) >= 750) then begin
-                set_global_var(38, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
             end
             else begin
-                if (global_var(0) >= 500) then begin
-                    set_global_var(39, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                 end
                 else begin
-                    if (global_var(0) >= 250) then begin
-                        set_global_var(40, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                        set_global_var(GVAR_KARMA_DEFENDER, 1);
                     end
                     else begin
-                        if (global_var(0) > -250) then begin
-                            set_global_var(41, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                            set_global_var(GVAR_KARMA_WANDERER, 1);
                         end
                         else begin
-                            if (global_var(0) > -500) then begin
-                                set_global_var(42, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                set_global_var(GVAR_KARMA_BETRAYER, 1);
                             end
                             else begin
-                                if (global_var(0) > -750) then begin
-                                    set_global_var(43, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -1000) then begin
-                                        set_global_var(44, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                     end
                                     else begin
-                                        set_global_var(45, 1);
+                                        set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                     end
                                 end
                             end
@@ -397,35 +397,35 @@ begin
                 end
             end
         end
-        if (global_var(37) == 1) then begin
-            Static_Reaction := Static_Reaction - 20 + (40 * has_trait(0, dude_obj, 39));
+        if (global_var(GVAR_KARMA_HOLY_WARRIOR) == 1) then begin
+            Static_Reaction := Static_Reaction - 20 + (40 * has_trait(TRAIT_PERK, dude_obj, PERK_cult_of_personality));
         end
         else begin
-            if (global_var(38) == 1) then begin
-                Static_Reaction := Static_Reaction - 15 + (30 * has_trait(0, dude_obj, 39));
+            if (global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES) == 1) then begin
+                Static_Reaction := Static_Reaction - 15 + (30 * has_trait(TRAIT_PERK, dude_obj, PERK_cult_of_personality));
             end
             else begin
-                if (global_var(39) == 1) then begin
-                    Static_Reaction := Static_Reaction - 10 + (20 * has_trait(0, dude_obj, 39));
+                if (global_var(GVAR_KARMA_SHIELD_OF_HOPE) == 1) then begin
+                    Static_Reaction := Static_Reaction - 10 + (20 * has_trait(TRAIT_PERK, dude_obj, PERK_cult_of_personality));
                 end
                 else begin
-                    if (global_var(40) == 1) then begin
-                        Static_Reaction := Static_Reaction - 5 + (10 * has_trait(0, dude_obj, 39));
+                    if (global_var(GVAR_KARMA_DEFENDER) == 1) then begin
+                        Static_Reaction := Static_Reaction - 5 + (10 * has_trait(TRAIT_PERK, dude_obj, PERK_cult_of_personality));
                     end
                     else begin
-                        if (global_var(41) == 1) then begin
+                        if (global_var(GVAR_KARMA_WANDERER) == 1) then begin
                             Static_Reaction := Static_Reaction - 0;
                         end
                         else begin
-                            if (global_var(42) == 1) then begin
+                            if (global_var(GVAR_KARMA_BETRAYER) == 1) then begin
                                 Static_Reaction := Static_Reaction - -5;
                             end
                             else begin
-                                if (global_var(43) == 1) then begin
+                                if (global_var(GVAR_KARMA_SWORD_OF_DESPAIR) == 1) then begin
                                     Static_Reaction := Static_Reaction - -10;
                                 end
                                 else begin
-                                    if (global_var(44) == 1) then begin
+                                    if (global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE) == 1) then begin
                                         Static_Reaction := Static_Reaction - -15;
                                     end
                                     else begin
@@ -441,58 +441,58 @@ begin
         debug_msg("Bad Person Reaction Bonus == " + Static_Reaction);
     end
     else begin
-        Static_Reaction := Static_Reaction + reaction_bonus_karma - (reaction_bonus_karma * has_trait(0, dude_obj, 39));
-        if ((reaction_bonus_karma > 0) and (has_trait(0, dude_obj, 39) != 0)) then begin
+        Static_Reaction := Static_Reaction + reaction_bonus_karma - (reaction_bonus_karma * has_trait(TRAIT_PERK, dude_obj, PERK_cult_of_personality));
+        if ((reaction_bonus_karma > 0) and (has_trait(TRAIT_PERK, dude_obj, PERK_cult_of_personality) != 0)) then begin
             Static_Reaction := Static_Reaction + reaction_bonus_karma;
         end
         debug_msg("Karma Reaction Bonus == " + Static_Reaction);
-        Static_Reaction := Static_Reaction + reaction_bonus_town_rep - (reaction_bonus_town_rep * has_trait(0, dude_obj, 39));
-        if ((reaction_bonus_town_rep > 0) and (has_trait(0, dude_obj, 39) != 0)) then begin
+        Static_Reaction := Static_Reaction + reaction_bonus_town_rep - (reaction_bonus_town_rep * has_trait(TRAIT_PERK, dude_obj, PERK_cult_of_personality));
+        if ((reaction_bonus_town_rep > 0) and (has_trait(TRAIT_PERK, dude_obj, PERK_cult_of_personality) != 0)) then begin
             Static_Reaction := Static_Reaction + reaction_bonus_town_rep;
         end
         debug_msg("Town Rep Bonus == " + Static_Reaction);
-        set_global_var(37, 0);
-        set_global_var(38, 0);
-        set_global_var(39, 0);
-        set_global_var(40, 0);
-        set_global_var(41, 0);
-        set_global_var(42, 0);
-        set_global_var(43, 0);
-        set_global_var(44, 0);
-        set_global_var(45, 0);
-        if (global_var(0) >= 1000) then begin
-            set_global_var(37, 1);
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+        set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+        set_global_var(GVAR_KARMA_DEFENDER, 0);
+        set_global_var(GVAR_KARMA_WANDERER, 0);
+        set_global_var(GVAR_KARMA_BETRAYER, 0);
+        set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+        set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+            set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
         end
         else begin
-            if (global_var(0) >= 750) then begin
-                set_global_var(38, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
             end
             else begin
-                if (global_var(0) >= 500) then begin
-                    set_global_var(39, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                 end
                 else begin
-                    if (global_var(0) >= 250) then begin
-                        set_global_var(40, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                        set_global_var(GVAR_KARMA_DEFENDER, 1);
                     end
                     else begin
-                        if (global_var(0) > -250) then begin
-                            set_global_var(41, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                            set_global_var(GVAR_KARMA_WANDERER, 1);
                         end
                         else begin
-                            if (global_var(0) > -500) then begin
-                                set_global_var(42, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                set_global_var(GVAR_KARMA_BETRAYER, 1);
                             end
                             else begin
-                                if (global_var(0) > -750) then begin
-                                    set_global_var(43, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -1000) then begin
-                                        set_global_var(44, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                     end
                                     else begin
-                                        set_global_var(45, 1);
+                                        set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                     end
                                 end
                             end
@@ -501,39 +501,39 @@ begin
                 end
             end
         end
-        if (global_var(37) == 1) then begin
+        if (global_var(GVAR_KARMA_HOLY_WARRIOR) == 1) then begin
             Static_Reaction := Static_Reaction + 20;
         end
         else begin
-            if (global_var(38) == 1) then begin
+            if (global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES) == 1) then begin
                 Static_Reaction := Static_Reaction + 15;
             end
             else begin
-                if (global_var(39) == 1) then begin
+                if (global_var(GVAR_KARMA_SHIELD_OF_HOPE) == 1) then begin
                     Static_Reaction := Static_Reaction + 10;
                 end
                 else begin
-                    if (global_var(40) == 1) then begin
+                    if (global_var(GVAR_KARMA_DEFENDER) == 1) then begin
                         Static_Reaction := Static_Reaction + 5;
                     end
                     else begin
-                        if (global_var(41) == 1) then begin
+                        if (global_var(GVAR_KARMA_WANDERER) == 1) then begin
                             Static_Reaction := Static_Reaction + 0;
                         end
                         else begin
-                            if (global_var(42) == 1) then begin
-                                Static_Reaction := Static_Reaction + -5 + (10 * has_trait(0, dude_obj, 39));
+                            if (global_var(GVAR_KARMA_BETRAYER) == 1) then begin
+                                Static_Reaction := Static_Reaction + -5 + (10 * has_trait(TRAIT_PERK, dude_obj, PERK_cult_of_personality));
                             end
                             else begin
-                                if (global_var(43) == 1) then begin
-                                    Static_Reaction := Static_Reaction + -10 + (20 * has_trait(0, dude_obj, 39));
+                                if (global_var(GVAR_KARMA_SWORD_OF_DESPAIR) == 1) then begin
+                                    Static_Reaction := Static_Reaction + -10 + (20 * has_trait(TRAIT_PERK, dude_obj, PERK_cult_of_personality));
                                 end
                                 else begin
-                                    if (global_var(44) == 1) then begin
-                                        Static_Reaction := Static_Reaction + -15 + (30 * has_trait(0, dude_obj, 39));
+                                    if (global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE) == 1) then begin
+                                        Static_Reaction := Static_Reaction + -15 + (30 * has_trait(TRAIT_PERK, dude_obj, PERK_cult_of_personality));
                                     end
                                     else begin
-                                        Static_Reaction := Static_Reaction + -20 + (40 * has_trait(0, dude_obj, 39));
+                                        Static_Reaction := Static_Reaction + -20 + (40 * has_trait(TRAIT_PERK, dude_obj, PERK_cult_of_personality));
                                     end
                                 end
                             end
@@ -545,39 +545,39 @@ begin
         debug_msg("Good Person Reaction Bonus == " + Static_Reaction);
     end
     if (Slavery_Tolerant == 0) then begin
-        if (global_var(11) == 1) then begin
-            Static_Reaction := Static_Reaction + (2 * -25) + (2 * 25 * has_trait(0, dude_obj, 39));
+        if (global_var(GVAR_REPUTATION_SLAVER) == 1) then begin
+            Static_Reaction := Static_Reaction + (2 * -25) + (2 * 25 * has_trait(TRAIT_PERK, dude_obj, PERK_cult_of_personality));
         end
     end
     else begin
         if (Slavery_Tolerant == 1) then begin
-            if (global_var(11) == 1) then begin
-                Static_Reaction := Static_Reaction + -25 + (25 * has_trait(0, dude_obj, 39));
+            if (global_var(GVAR_REPUTATION_SLAVER) == 1) then begin
+                Static_Reaction := Static_Reaction + -25 + (25 * has_trait(TRAIT_PERK, dude_obj, PERK_cult_of_personality));
             end
         end
         else begin
             if (Slavery_Tolerant == 3) then begin
-                if (global_var(11) == 1) then begin
+                if (global_var(GVAR_REPUTATION_SLAVER) == 1) then begin
                     Static_Reaction := Static_Reaction - -25;
                 end
             end
         end
     end
     debug_msg("Slaver + Aligned Reaction == " + Static_Reaction);
-    Static_Reaction := Static_Reaction + (10 * has_trait(0, dude_obj, 10));
+    Static_Reaction := Static_Reaction + (10 * has_trait(TRAIT_PERK, dude_obj, PERK_presence));
     debug_msg("Presence Reaction == " + Static_Reaction);
-    if (global_var(1) >= 2) then begin
-        Static_Reaction := Static_Reaction + -30 + (30 * has_trait(0, dude_obj, 39));
+    if (global_var(GVAR_CHILDKILLER_REPUTATION) >= 2) then begin
+        Static_Reaction := Static_Reaction + -30 + (30 * has_trait(TRAIT_PERK, dude_obj, PERK_cult_of_personality));
     end
     debug_msg("Childkiller Reaction == " + Static_Reaction);
     set_local_var(0, local_var(3) + Static_Reaction);
     debug_msg("Initial Reaction == " + local_var(0));
     move_obj_inven_to_obj(i_lao_table_swap, self_obj);
-    if ((global_var(361) bwand 1) != 0) then begin
+    if ((global_var(GVAR_SAN_FRAN_FLAGS) bwand 1) != 0) then begin
         call Node020();
     end
     else begin
-        if (global_var(426) == 2) then begin
+        if (global_var(GVAR_SAN_FRAN_STRUGGLE) == 2) then begin
             start_gdialog(925, self_obj, 4, -1, -1);
             gsay_start;
             call Node021();
@@ -598,59 +598,59 @@ end
 procedure destroy_p_proc
 begin
     if (source_obj == dude_obj) then begin
-        set_global_var(5, global_var(5) + 1);
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(4) > (3 * global_var(5))) or (global_var(2) == 1)) and (global_var(1) == 0)) then begin
-            set_global_var(2, 1);
-            set_global_var(3, 0);
+        set_global_var(GVAR_GOOD_MONSTER, global_var(GVAR_GOOD_MONSTER) + 1);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(4) > (3 * global_var(GVAR_GOOD_MONSTER))) or (global_var(GVAR_CHAMPION_REPUTATION) == 1)) and (global_var(GVAR_CHILDKILLER_REPUTATION) == 0)) then begin
+            set_global_var(GVAR_CHAMPION_REPUTATION, 1);
+            set_global_var(GVAR_BERSERKER_REPUTATION, 0);
         end
-        if (metarule(51, self_obj) == 2) then begin
+        if (critter_kill_type(self_obj) == 2) then begin
             if (source_obj == dude_obj) then begin
-                set_global_var(2, 0);
-                set_global_var(1, global_var(1) + 1);
-                set_global_var(0, global_var(0) + -15);
+                set_global_var(GVAR_CHAMPION_REPUTATION, 0);
+                set_global_var(GVAR_CHILDKILLER_REPUTATION, global_var(GVAR_CHILDKILLER_REPUTATION) + 1);
+                set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + -15);
                 debug_msg("Player gains " + -15 + " Karma Points.");
-                set_global_var(37, 0);
-                set_global_var(38, 0);
-                set_global_var(39, 0);
-                set_global_var(40, 0);
-                set_global_var(41, 0);
-                set_global_var(42, 0);
-                set_global_var(43, 0);
-                set_global_var(44, 0);
-                set_global_var(45, 0);
-                if (global_var(0) >= 1000) then begin
-                    set_global_var(37, 1);
+                set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+                set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+                set_global_var(GVAR_KARMA_DEFENDER, 0);
+                set_global_var(GVAR_KARMA_WANDERER, 0);
+                set_global_var(GVAR_KARMA_BETRAYER, 0);
+                set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+                set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+                set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+                    set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
                 end
                 else begin
-                    if (global_var(0) >= 750) then begin
-                        set_global_var(38, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
                     end
                     else begin
-                        if (global_var(0) >= 500) then begin
-                            set_global_var(39, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                            set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                         end
                         else begin
-                            if (global_var(0) >= 250) then begin
-                                set_global_var(40, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                                set_global_var(GVAR_KARMA_DEFENDER, 1);
                             end
                             else begin
-                                if (global_var(0) > -250) then begin
-                                    set_global_var(41, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                                    set_global_var(GVAR_KARMA_WANDERER, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -500) then begin
-                                        set_global_var(42, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                        set_global_var(GVAR_KARMA_BETRAYER, 1);
                                     end
                                     else begin
-                                        if (global_var(0) > -750) then begin
-                                            set_global_var(43, 1);
+                                        if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                            set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                         end
                                         else begin
-                                            if (global_var(0) > -1000) then begin
-                                                set_global_var(44, 1);
+                                            if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                                set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                             end
                                             else begin
-                                                set_global_var(45, 1);
+                                                set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                             end
                                         end
                                     end
@@ -659,58 +659,58 @@ begin
                         end
                     end
                 end
-                set_global_var(61, global_var(61) + -8);
+                set_global_var(GVAR_TOWN_REP_SAN_FRANCISCO, global_var(GVAR_TOWN_REP_SAN_FRANCISCO) + -8);
                 debug_msg("Added " + -8 + " to Town Rep");
             end
         end
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(5) > (2 * global_var(4))) or (global_var(3) == 1))) then begin
-            set_global_var(3, 1);
-            set_global_var(2, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
+            set_global_var(GVAR_BERSERKER_REPUTATION, 1);
+            set_global_var(GVAR_CHAMPION_REPUTATION, 0);
         end
-        set_global_var(0, global_var(0) + -10);
+        set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + -10);
         debug_msg("Player gains " + -10 + " Karma Points.");
-        set_global_var(37, 0);
-        set_global_var(38, 0);
-        set_global_var(39, 0);
-        set_global_var(40, 0);
-        set_global_var(41, 0);
-        set_global_var(42, 0);
-        set_global_var(43, 0);
-        set_global_var(44, 0);
-        set_global_var(45, 0);
-        if (global_var(0) >= 1000) then begin
-            set_global_var(37, 1);
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+        set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+        set_global_var(GVAR_KARMA_DEFENDER, 0);
+        set_global_var(GVAR_KARMA_WANDERER, 0);
+        set_global_var(GVAR_KARMA_BETRAYER, 0);
+        set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+        set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+            set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
         end
         else begin
-            if (global_var(0) >= 750) then begin
-                set_global_var(38, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
             end
             else begin
-                if (global_var(0) >= 500) then begin
-                    set_global_var(39, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                 end
                 else begin
-                    if (global_var(0) >= 250) then begin
-                        set_global_var(40, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                        set_global_var(GVAR_KARMA_DEFENDER, 1);
                     end
                     else begin
-                        if (global_var(0) > -250) then begin
-                            set_global_var(41, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                            set_global_var(GVAR_KARMA_WANDERER, 1);
                         end
                         else begin
-                            if (global_var(0) > -500) then begin
-                                set_global_var(42, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                set_global_var(GVAR_KARMA_BETRAYER, 1);
                             end
                             else begin
-                                if (global_var(0) > -750) then begin
-                                    set_global_var(43, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -1000) then begin
-                                        set_global_var(44, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                     end
                                     else begin
-                                        set_global_var(45, 1);
+                                        set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                     end
                                 end
                             end
@@ -719,7 +719,7 @@ begin
                 end
             end
         end
-        set_global_var(61, global_var(61) + -5);
+        set_global_var(GVAR_TOWN_REP_SAN_FRANCISCO, global_var(GVAR_TOWN_REP_SAN_FRANCISCO) + -5);
         debug_msg("Added " + -5 + " to Town Rep");
     end
     i_lao_merchant := -1;
@@ -729,17 +729,17 @@ procedure look_at_p_proc
 begin
     script_overrides;
     if (local_var(4) == 0) then begin
-        display_msg(message_str(925, 100));
+        display_mstr(100);
     end
     else begin
-        display_msg(message_str(925, 101));
+        display_mstr(101);
     end
 end
 
 procedure description_p_proc
 begin
     script_overrides;
-    display_msg(message_str(925, 102));
+    display_mstr(102);
 end
 
 procedure use_skill_on_p_proc
@@ -756,8 +756,8 @@ end
 procedure map_enter_p_proc
 begin
     Only_Once := 0;
-    critter_add_trait(self_obj, 1, 6, 167);
-    critter_add_trait(self_obj, 1, 5, 153);
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 167);
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_SHI);
 end
 
 procedure timed_event_p_proc
@@ -780,198 +780,198 @@ procedure Node001
 begin
     if (n1 == 0) then begin
         n1 := 1;
-        gsay_reply(925, 103);
+        Reply(103);
     end
     else begin
-        gsay_reply(925, 104);
+        Reply(104);
     end
-    giq_option(-3, 925, 105, Node002, 50);
-    giq_option(-3, 925, 106, Node002, 50);
-    giq_option(4, 925, 107, Node005, 50);
-    giq_option(4, 925, 108, NodeBarter, 50);
-    giq_option(4, 925, 109, Node006, 50);
-    giq_option(1, 925, 110, Node999, 50);
+    giq_option(-3, NAME, 105, Node002, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 106, Node002, NEUTRAL_REACTION);
+    giq_option(4, NAME, 107, Node005, NEUTRAL_REACTION);
+    giq_option(4, NAME, 108, NodeBarter, NEUTRAL_REACTION);
+    giq_option(4, NAME, 109, Node006, NEUTRAL_REACTION);
+    giq_option(1, NAME, 110, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node002
 begin
-    gsay_reply(925, 111);
-    giq_option(1, 925, 112, Node003, 50);
-    giq_option(1, 925, 113, Node004, 50);
+    Reply(111);
+    giq_option(1, NAME, 112, Node003, NEUTRAL_REACTION);
+    giq_option(1, NAME, 113, Node004, NEUTRAL_REACTION);
 end
 
 procedure Node003
 begin
-    gsay_reply(925, 114);
-    giq_option(1, 925, 115, NodeBarter, 50);
+    Reply(114);
+    giq_option(1, NAME, 115, NodeBarter, NEUTRAL_REACTION);
 end
 
 procedure Node004
 begin
-    gsay_reply(925, 116);
-    giq_option(1, 925, 117, Node999, 50);
+    Reply(116);
+    giq_option(1, NAME, 117, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node005
 begin
-    gsay_reply(925, 118);
-    giq_option(4, 925, 119, Node006, 50);
-    giq_option(4, 925, 120, NodeBarter, 50);
+    Reply(118);
+    giq_option(4, NAME, 119, Node006, NEUTRAL_REACTION);
+    giq_option(4, NAME, 120, NodeBarter, NEUTRAL_REACTION);
 end
 
 procedure Node006
 begin
-    gsay_reply(925, 121);
-    giq_option(4, 925, 122, Node007, 50);
-    if ((global_var(519) == 2) or (global_var(519) == 4)) then begin
-        giq_option(4, 925, 123, Node822, 50);
+    Reply(121);
+    giq_option(4, NAME, 122, Node007, NEUTRAL_REACTION);
+    if ((global_var(GVAR_SAN_FRAN_SHI_WHIRLY) == 2) or (global_var(GVAR_SAN_FRAN_SHI_WHIRLY) == 4)) then begin
+        giq_option(4, NAME, 123, Node822, NEUTRAL_REACTION);
     end
-    if (((global_var(363) bwand 1) != 0) and not((global_var(363) bwand 8) != 0)) then begin
-        giq_option(4, 925, 124, Node01819, 50);
+    if (((global_var(GVAR_SAN_FRAN_TANKER) bwand 1) != 0) and not((global_var(GVAR_SAN_FRAN_TANKER) bwand 8) != 0)) then begin
+        giq_option(4, NAME, 124, Node01819, NEUTRAL_REACTION);
     end
-    if (((global_var(363) bwand 1) != 0) and not((global_var(363) bwand 2) != 0)) then begin
-        giq_option(4, 925, 125, Node008, 50);
+    if (((global_var(GVAR_SAN_FRAN_TANKER) bwand 1) != 0) and not((global_var(GVAR_SAN_FRAN_TANKER) bwand 2) != 0)) then begin
+        giq_option(4, NAME, 125, Node008, NEUTRAL_REACTION);
     end
-    if (((global_var(363) bwand 1) != 0) and not((global_var(363) bwand 4) != 0)) then begin
-        giq_option(4, 925, 126, Node009, 50);
+    if (((global_var(GVAR_SAN_FRAN_TANKER) bwand 1) != 0) and not((global_var(GVAR_SAN_FRAN_TANKER) bwand 4) != 0)) then begin
+        giq_option(4, NAME, 126, Node009, NEUTRAL_REACTION);
     end
-    if ((global_var(367) == 1) and ((game_time - global_var(448)) < (24 * (60 * (60 * 10)) * 5))) then begin
-        giq_option(4, 925, 127, Node010, 50);
+    if ((global_var(GVAR_SAN_FRAN_SPLEEN) == 1) and ((game_time - global_var(GVAR_SAN_FRAN_SPLEEN_TIME)) < (24 * (60 * (60 * 10)) * 5))) then begin
+        giq_option(4, NAME, 127, Node010, NEUTRAL_REACTION);
     end
-    if (global_var(494) == 4) then begin
-        giq_option(4, 925, 128, Node015, 50);
+    if (global_var(GVAR_FALLOUT_2) == 4) then begin
+        giq_option(4, NAME, 128, Node015, NEUTRAL_REACTION);
     end
-    giq_option(4, 925, 129, Node999, 50);
+    giq_option(4, NAME, 129, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node007
 begin
-    gsay_reply(925, 130);
-    giq_option(4, 925, 131, Node006, 50);
-    giq_option(4, 925, 132, Node999, 50);
+    Reply(130);
+    giq_option(4, NAME, 131, Node006, NEUTRAL_REACTION);
+    giq_option(4, NAME, 132, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node008
 begin
-    gsay_reply(925, 133);
-    giq_option(4, 925, 134, Node006, 50);
-    giq_option(4, 925, 135, Node999, 50);
+    Reply(133);
+    giq_option(4, NAME, 134, Node006, NEUTRAL_REACTION);
+    giq_option(4, NAME, 135, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node009
 begin
-    gsay_reply(925, 136);
-    giq_option(4, 925, 137, Node006, 50);
-    giq_option(4, 925, 138, Node999, 50);
+    Reply(136);
+    giq_option(4, NAME, 137, Node006, NEUTRAL_REACTION);
+    giq_option(4, NAME, 138, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node010
 begin
-    if (global_var(367) < 2) then begin
-        set_global_var(367, 2);
+    if (global_var(GVAR_SAN_FRAN_SPLEEN) < 2) then begin
+        set_global_var(GVAR_SAN_FRAN_SPLEEN, 2);
     end
-    gsay_reply(925, 139);
-    giq_option(4, 925, 140, Node011, 50);
-    giq_option(4, 925, 141, Node012, 50);
-    giq_option(4, 925, 142, Node001, 50);
-    giq_option(4, 925, 143, Node999, 50);
+    Reply(139);
+    giq_option(4, NAME, 140, Node011, NEUTRAL_REACTION);
+    giq_option(4, NAME, 141, Node012, NEUTRAL_REACTION);
+    giq_option(4, NAME, 142, Node001, NEUTRAL_REACTION);
+    giq_option(4, NAME, 143, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node011
 begin
-    gsay_reply(925, 144);
-    giq_option(4, 925, 145, Node012, 50);
-    giq_option(4, 925, 146, Node006, 50);
-    giq_option(4, 925, 147, Node999, 50);
+    Reply(144);
+    giq_option(4, NAME, 145, Node012, NEUTRAL_REACTION);
+    giq_option(4, NAME, 146, Node006, NEUTRAL_REACTION);
+    giq_option(4, NAME, 147, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node012
 begin
-    gsay_reply(925, 148);
-    giq_option(4, 925, 149, Node013, 50);
-    giq_option(4, 925, 150, Node014, 50);
+    Reply(148);
+    giq_option(4, NAME, 149, Node013, NEUTRAL_REACTION);
+    giq_option(4, NAME, 150, Node014, NEUTRAL_REACTION);
 end
 
 procedure Node013
 begin
-    gsay_reply(925, 151);
-    giq_option(4, 925, 152, Node006, 50);
-    giq_option(4, 925, 153, Node014a, 51);
-    giq_option(4, 925, 154, Node999, 50);
+    Reply(151);
+    giq_option(4, NAME, 152, Node006, NEUTRAL_REACTION);
+    giq_option(4, NAME, 153, Node014a, BAD_REACTION);
+    giq_option(4, NAME, 154, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node014
 begin
-    gsay_reply(925, 155);
-    giq_option(4, 925, 156, Node999, 50);
+    Reply(155);
+    giq_option(4, NAME, 156, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node015
 begin
-    gsay_reply(925, 157);
-    giq_option(4, 925, 158, Node016, 49);
-    giq_option(4, 925, 159, Node014a, 51);
+    Reply(157);
+    giq_option(4, NAME, 158, Node016, GOOD_REACTION);
+    giq_option(4, NAME, 159, Node014a, BAD_REACTION);
 end
 
 procedure Node016
 begin
-    gsay_reply(925, 160);
-    giq_option(4, 925, 161, Node017, 50);
-    giq_option(4, 925, 162, Node999, 51);
+    Reply(160);
+    giq_option(4, NAME, 161, Node017, NEUTRAL_REACTION);
+    giq_option(4, NAME, 162, Node999, BAD_REACTION);
 end
 
 procedure Node017
 begin
-    gsay_reply(925, 163);
-    giq_option(4, 925, 164, Node006, 50);
-    giq_option(4, 925, 165, Node999, 50);
+    Reply(163);
+    giq_option(4, NAME, 164, Node006, NEUTRAL_REACTION);
+    giq_option(4, NAME, 165, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node018
 begin
-    gsay_reply(925, 166);
-    giq_option(4, 925, 167, Node006, 50);
-    giq_option(4, 925, 168, Node999, 50);
+    Reply(166);
+    giq_option(4, NAME, 167, Node006, NEUTRAL_REACTION);
+    giq_option(4, NAME, 168, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node019
 begin
-    gsay_reply(925, 169);
-    giq_option(4, 925, 170, Node006, 50);
-    giq_option(4, 925, 171, Node999, 50);
+    Reply(169);
+    giq_option(4, NAME, 170, Node006, NEUTRAL_REACTION);
+    giq_option(4, NAME, 171, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node020
 begin
-    float_msg(self_obj, message_str(925, 172), 2);
+    float_msg(self_obj, mstr(172), FLOAT_MSG_RED);
     call Node998();
 end
 
 procedure Node021
 begin
-    gsay_reply(925, 173);
-    giq_option(4, 925, 174, Node005, 50);
-    giq_option(4, 925, 175, NodeBarter, 50);
-    giq_option(4, 925, 176, Node006, 50);
-    giq_option(4, 925, 177, Node999, 50);
+    Reply(173);
+    giq_option(4, NAME, 174, Node005, NEUTRAL_REACTION);
+    giq_option(4, NAME, 175, NodeBarter, NEUTRAL_REACTION);
+    giq_option(4, NAME, 176, Node006, NEUTRAL_REACTION);
+    giq_option(4, NAME, 177, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node022
 begin
-    gsay_reply(925, 178);
-    giq_option(4, 925, 179, Node006, 50);
-    giq_option(4, 925, 180, Node999, 50);
+    Reply(178);
+    giq_option(4, NAME, 179, Node006, NEUTRAL_REACTION);
+    giq_option(4, NAME, 180, Node999, NEUTRAL_REACTION);
 end
 
 procedure NodeBarter
 begin
-    gdialog_mod_barter(0);
+    gdialog_barter;
     call Node001();
 end
 
 procedure Node01819
 begin
-    if ((global_var(361) bwand 32768) != 0) then begin
+    if ((global_var(GVAR_SAN_FRAN_FLAGS) bwand 32768) != 0) then begin
         call Node018();
     end
     else begin
@@ -981,7 +981,7 @@ end
 
 procedure Node822
 begin
-    if (global_var(519) == 2) then begin
+    if (global_var(GVAR_SAN_FRAN_SHI_WHIRLY) == 2) then begin
         call Node008();
     end
     else begin
@@ -1036,4 +1036,3 @@ begin
     end
     call Node014();
 end
-
