diff --git a/Mods/MegaMod/Original/FCROCMAN.ssl b/Mods/MegaMod/ReDefine/FCROCMAN.ssl
index 017ea30..0507f5f 100644
--- a/Mods/MegaMod/Original/FCROCMAN.ssl
+++ b/Mods/MegaMod/ReDefine/FCROCMAN.ssl
@@ -189,24 +189,24 @@ procedure pickup_p_proc
 begin
     if (source_obj == dude_obj) then begin
         if (3 == 1) then begin
-            attack_complex(source_obj, 0, 1, 0, 0, 30000, 0, 0);
+            attack(source_obj);
         end
         else begin
             if (3 == 2) then begin
-                float_msg(self_obj, message_str(14, random(3325, 3329)), 0);
+                float_msg(self_obj, g_mstr(random(3325, 3329)), FLOAT_MSG_NORMAL);
                 if (7 > -1) then begin
                     if (local_var(7) >= 1) then begin
-                        attack_complex(source_obj, 0, 1, 0, 0, 30000, 0, 0);
+                        attack(source_obj);
                     end
                     set_local_var(7, local_var(7) + 1);
                 end
             end
             else begin
                 if (3 == 3) then begin
-                    float_msg(self_obj, message_str(14, random(3050, 3064)), 0);
+                    float_msg(self_obj, g_mstr(random(3050, 3064)), FLOAT_MSG_NORMAL);
                     if (7 > -1) then begin
                         if (local_var(7) >= 1) then begin
-                            attack_complex(source_obj, 0, 1, 0, 0, 30000, 0, 0);
+                            attack(source_obj);
                         end
                         set_local_var(7, local_var(7) + 1);
                     end
@@ -236,10 +236,10 @@ begin
             if (anim_busy(self_obj) == 0) then begin
                 if ((tile_distance_objs(self_obj, dude_obj) < 8) or obj_can_see_obj(self_obj, dude_obj)) then begin
                     if (anim_busy(self_obj) == 0) then begin
-                        global_temp := rotation_to_tile(tile_num(dude_obj), tile_num(self_obj));
+                        global_temp := rotation_to_tile(dude_tile, tile_num(self_obj));
                         animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(self_obj), global_temp, random(3, 10)), 1);
                         global_temp := (global_temp + 1) % 6;
-                        while ((anim_busy(self_obj) == 0) and (global_temp != rotation_to_tile(tile_num(dude_obj), tile_num(self_obj)))) do begin
+                        while ((anim_busy(self_obj) == 0) and (global_temp != rotation_to_tile(dude_tile, tile_num(self_obj)))) do begin
                             debug_msg("flee loop: rot == " + global_temp);
                             animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(self_obj), global_temp, random(3, 10)), 1);
                             global_temp := (global_temp + 1) % 6;
@@ -249,7 +249,7 @@ begin
             end
         end
         else begin
-            attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+            attack(dude_obj);
         end
     end
 end
@@ -259,48 +259,48 @@ begin
     Evil_Critter := 0;
     Slavery_Tolerant := 2;
     Karma_Perception := 10;
-    set_global_var(37, 0);
-    set_global_var(38, 0);
-    set_global_var(39, 0);
-    set_global_var(40, 0);
-    set_global_var(41, 0);
-    set_global_var(42, 0);
-    set_global_var(43, 0);
-    set_global_var(44, 0);
-    set_global_var(45, 0);
-    if (global_var(0) >= 1000) then begin
-        set_global_var(37, 1);
+    set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+    set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+    set_global_var(GVAR_KARMA_DEFENDER, 0);
+    set_global_var(GVAR_KARMA_WANDERER, 0);
+    set_global_var(GVAR_KARMA_BETRAYER, 0);
+    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+    set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+    set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+    if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
     end
     else begin
-        if (global_var(0) >= 750) then begin
-            set_global_var(38, 1);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+            set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
         end
         else begin
-            if (global_var(0) >= 500) then begin
-                set_global_var(39, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
             end
             else begin
-                if (global_var(0) >= 250) then begin
-                    set_global_var(40, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                    set_global_var(GVAR_KARMA_DEFENDER, 1);
                 end
                 else begin
-                    if (global_var(0) > -250) then begin
-                        set_global_var(41, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                        set_global_var(GVAR_KARMA_WANDERER, 1);
                     end
                     else begin
-                        if (global_var(0) > -500) then begin
-                            set_global_var(42, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                            set_global_var(GVAR_KARMA_BETRAYER, 1);
                         end
                         else begin
-                            if (global_var(0) > -750) then begin
-                                set_global_var(43, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                             end
                             else begin
-                                if (global_var(0) > -1000) then begin
-                                    set_global_var(44, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                    set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                 end
                                 else begin
-                                    set_global_var(45, 1);
+                                    set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                 end
                             end
                         end
@@ -311,7 +311,7 @@ begin
     end
     if (local_var(2) == 0) then begin
         set_local_var(2, 1);
-        set_local_var(3, (get_critter_stat(dude_obj, 3) - get_critter_stat(self_obj, 3)) * 5);
+        set_local_var(3, (dude_charisma - self_charisma) * 5);
         debug_msg("Base Reaction == " + local_var(3));
     end
     Static_Reaction := 0;
@@ -323,14 +323,14 @@ begin
     else begin
         reaction_bonus_karma := -1 * (-1 * reaction_bonus_karma / 1000);
     end
-    if (global_var(61) >= 0) then begin
-        reaction_bonus_town_rep := global_var(61) / 2;
+    if (global_var(GVAR_TOWN_REP_SAN_FRANCISCO) >= 0) then begin
+        reaction_bonus_town_rep := global_var(GVAR_TOWN_REP_SAN_FRANCISCO) / 2;
     end
     else begin
-        reaction_bonus_town_rep := -1 * (-1 * global_var(61) / 2);
+        reaction_bonus_town_rep := -1 * (-1 * global_var(GVAR_TOWN_REP_SAN_FRANCISCO) / 2);
     end
-    if (has_trait(2, dude_obj, 13)) then begin
-        if (get_critter_stat(dude_obj, 34) == get_critter_stat(self_obj, 34)) then begin
+    if (has_trait(TRAIT_TRAIT, dude_obj, TRAIT_sex_appeal)) then begin
+        if (dude_gender == self_gender) then begin
             Static_Reaction := Static_Reaction + -20;
         end
         else begin
@@ -346,48 +346,48 @@ begin
         debug_msg("Karma Reaction Bonus == " + Static_Reaction);
         Static_Reaction := Static_Reaction - reaction_bonus_town_rep;
         debug_msg("Town Rep Bonus == " + Static_Reaction);
-        set_global_var(37, 0);
-        set_global_var(38, 0);
-        set_global_var(39, 0);
-        set_global_var(40, 0);
-        set_global_var(41, 0);
-        set_global_var(42, 0);
-        set_global_var(43, 0);
-        set_global_var(44, 0);
-        set_global_var(45, 0);
-        if (global_var(0) >= 1000) then begin
-            set_global_var(37, 1);
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+        set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+        set_global_var(GVAR_KARMA_DEFENDER, 0);
+        set_global_var(GVAR_KARMA_WANDERER, 0);
+        set_global_var(GVAR_KARMA_BETRAYER, 0);
+        set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+        set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+            set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
         end
         else begin
-            if (global_var(0) >= 750) then begin
-                set_global_var(38, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
             end
             else begin
-                if (global_var(0) >= 500) then begin
-                    set_global_var(39, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                 end
                 else begin
-                    if (global_var(0) >= 250) then begin
-                        set_global_var(40, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                        set_global_var(GVAR_KARMA_DEFENDER, 1);
                     end
                     else begin
-                        if (global_var(0) > -250) then begin
-                            set_global_var(41, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                            set_global_var(GVAR_KARMA_WANDERER, 1);
                         end
                         else begin
-                            if (global_var(0) > -500) then begin
-                                set_global_var(42, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                set_global_var(GVAR_KARMA_BETRAYER, 1);
                             end
                             else begin
-                                if (global_var(0) > -750) then begin
-                                    set_global_var(43, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -1000) then begin
-                                        set_global_var(44, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                     end
                                     else begin
-                                        set_global_var(45, 1);
+                                        set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                     end
                                 end
                             end
@@ -396,35 +396,35 @@ begin
                 end
             end
         end
-        if (global_var(37) == 1) then begin
+        if (global_var(GVAR_KARMA_HOLY_WARRIOR) == 1) then begin
             Static_Reaction := Static_Reaction - 20;
         end
         else begin
-            if (global_var(38) == 1) then begin
+            if (global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES) == 1) then begin
                 Static_Reaction := Static_Reaction - 15;
             end
             else begin
-                if (global_var(39) == 1) then begin
+                if (global_var(GVAR_KARMA_SHIELD_OF_HOPE) == 1) then begin
                     Static_Reaction := Static_Reaction - 10;
                 end
                 else begin
-                    if (global_var(40) == 1) then begin
+                    if (global_var(GVAR_KARMA_DEFENDER) == 1) then begin
                         Static_Reaction := Static_Reaction - 5;
                     end
                     else begin
-                        if (global_var(41) == 1) then begin
+                        if (global_var(GVAR_KARMA_WANDERER) == 1) then begin
                             Static_Reaction := Static_Reaction - 0;
                         end
                         else begin
-                            if (global_var(42) == 1) then begin
+                            if (global_var(GVAR_KARMA_BETRAYER) == 1) then begin
                                 Static_Reaction := Static_Reaction - -5;
                             end
                             else begin
-                                if (global_var(43) == 1) then begin
+                                if (global_var(GVAR_KARMA_SWORD_OF_DESPAIR) == 1) then begin
                                     Static_Reaction := Static_Reaction - -10;
                                 end
                                 else begin
-                                    if (global_var(44) == 1) then begin
+                                    if (global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE) == 1) then begin
                                         Static_Reaction := Static_Reaction - -15;
                                     end
                                     else begin
@@ -444,48 +444,48 @@ begin
         debug_msg("Karma Reaction Bonus == " + Static_Reaction);
         Static_Reaction := Static_Reaction + reaction_bonus_town_rep;
         debug_msg("Town Rep Bonus == " + Static_Reaction);
-        set_global_var(37, 0);
-        set_global_var(38, 0);
-        set_global_var(39, 0);
-        set_global_var(40, 0);
-        set_global_var(41, 0);
-        set_global_var(42, 0);
-        set_global_var(43, 0);
-        set_global_var(44, 0);
-        set_global_var(45, 0);
-        if (global_var(0) >= 1000) then begin
-            set_global_var(37, 1);
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+        set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+        set_global_var(GVAR_KARMA_DEFENDER, 0);
+        set_global_var(GVAR_KARMA_WANDERER, 0);
+        set_global_var(GVAR_KARMA_BETRAYER, 0);
+        set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+        set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+            set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
         end
         else begin
-            if (global_var(0) >= 750) then begin
-                set_global_var(38, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
             end
             else begin
-                if (global_var(0) >= 500) then begin
-                    set_global_var(39, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                 end
                 else begin
-                    if (global_var(0) >= 250) then begin
-                        set_global_var(40, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                        set_global_var(GVAR_KARMA_DEFENDER, 1);
                     end
                     else begin
-                        if (global_var(0) > -250) then begin
-                            set_global_var(41, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                            set_global_var(GVAR_KARMA_WANDERER, 1);
                         end
                         else begin
-                            if (global_var(0) > -500) then begin
-                                set_global_var(42, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                set_global_var(GVAR_KARMA_BETRAYER, 1);
                             end
                             else begin
-                                if (global_var(0) > -750) then begin
-                                    set_global_var(43, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -1000) then begin
-                                        set_global_var(44, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                     end
                                     else begin
-                                        set_global_var(45, 1);
+                                        set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                     end
                                 end
                             end
@@ -494,35 +494,35 @@ begin
                 end
             end
         end
-        if (global_var(37) == 1) then begin
+        if (global_var(GVAR_KARMA_HOLY_WARRIOR) == 1) then begin
             Static_Reaction := Static_Reaction + 20;
         end
         else begin
-            if (global_var(38) == 1) then begin
+            if (global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES) == 1) then begin
                 Static_Reaction := Static_Reaction + 15;
             end
             else begin
-                if (global_var(39) == 1) then begin
+                if (global_var(GVAR_KARMA_SHIELD_OF_HOPE) == 1) then begin
                     Static_Reaction := Static_Reaction + 10;
                 end
                 else begin
-                    if (global_var(40) == 1) then begin
+                    if (global_var(GVAR_KARMA_DEFENDER) == 1) then begin
                         Static_Reaction := Static_Reaction + 5;
                     end
                     else begin
-                        if (global_var(41) == 1) then begin
+                        if (global_var(GVAR_KARMA_WANDERER) == 1) then begin
                             Static_Reaction := Static_Reaction + 0;
                         end
                         else begin
-                            if (global_var(42) == 1) then begin
+                            if (global_var(GVAR_KARMA_BETRAYER) == 1) then begin
                                 Static_Reaction := Static_Reaction + -5;
                             end
                             else begin
-                                if (global_var(43) == 1) then begin
+                                if (global_var(GVAR_KARMA_SWORD_OF_DESPAIR) == 1) then begin
                                     Static_Reaction := Static_Reaction + -10;
                                 end
                                 else begin
-                                    if (global_var(44) == 1) then begin
+                                    if (global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE) == 1) then begin
                                         Static_Reaction := Static_Reaction + -15;
                                     end
                                     else begin
@@ -538,38 +538,38 @@ begin
         debug_msg("Good Person Reaction Bonus == " + Static_Reaction);
     end
     if (Slavery_Tolerant == 0) then begin
-        if (global_var(11) == 1) then begin
+        if (global_var(GVAR_REPUTATION_SLAVER) == 1) then begin
             Static_Reaction := Static_Reaction + (2 * -25);
         end
     end
     else begin
         if (Slavery_Tolerant == 1) then begin
-            if (global_var(11) == 1) then begin
+            if (global_var(GVAR_REPUTATION_SLAVER) == 1) then begin
                 Static_Reaction := Static_Reaction + -25;
             end
         end
         else begin
             if (Slavery_Tolerant == 3) then begin
-                if (global_var(11) == 1) then begin
+                if (global_var(GVAR_REPUTATION_SLAVER) == 1) then begin
                     Static_Reaction := Static_Reaction - -25;
                 end
             end
         end
     end
     debug_msg("Slaver + Aligned Reaction == " + Static_Reaction);
-    Static_Reaction := Static_Reaction + (10 * has_trait(0, dude_obj, 10));
+    Static_Reaction := Static_Reaction + (10 * has_trait(TRAIT_PERK, dude_obj, PERK_presence));
     debug_msg("Presence Reaction == " + Static_Reaction);
-    if (global_var(1) >= 2) then begin
+    if (global_var(GVAR_CHILDKILLER_REPUTATION) >= 2) then begin
         Static_Reaction := Static_Reaction + -30;
     end
     debug_msg("Childkiller Reaction == " + Static_Reaction);
     set_local_var(0, local_var(3) + Static_Reaction);
     debug_msg("Initial Reaction == " + local_var(0));
-    if ((global_var(361) bwand 262144) != 0) then begin
+    if ((global_var(GVAR_SAN_FRAN_FLAGS) bwand 262144) != 0) then begin
         call Node001();
     end
     else begin
-        if (global_var(366) < 8) then begin
+        if (global_var(GVAR_SAN_FRAN_ELRON) < 8) then begin
             start_gdialog(967, self_obj, 4, -1, -1);
             gsay_start;
             call Node003();
@@ -577,7 +577,7 @@ begin
             end_dialogue;
         end
         else begin
-            if ((global_var(366) >= 8) and not((global_var(361) bwand 262144) != 0) and not((global_var(361) bwand 32768) != 0)) then begin
+            if ((global_var(GVAR_SAN_FRAN_ELRON) >= 8) and not((global_var(GVAR_SAN_FRAN_FLAGS) bwand 262144) != 0) and not((global_var(GVAR_SAN_FRAN_FLAGS) bwand 32768) != 0)) then begin
                 start_gdialog(967, self_obj, 4, -1, -1);
                 gsay_start;
                 call Node011();
@@ -585,7 +585,7 @@ begin
                 end_dialogue;
             end
             else begin
-                if ((global_var(366) >= 8) and not((global_var(361) bwand 262144) != 0) and ((global_var(361) bwand 32768) != 0)) then begin
+                if ((global_var(GVAR_SAN_FRAN_ELRON) >= 8) and not((global_var(GVAR_SAN_FRAN_FLAGS) bwand 262144) != 0) and ((global_var(GVAR_SAN_FRAN_FLAGS) bwand 32768) != 0)) then begin
                     start_gdialog(967, self_obj, 4, -1, -1);
                     gsay_start;
                     call Node016();
@@ -605,62 +605,62 @@ end
 procedure destroy_p_proc
 begin
     variable LVar0 := 0;
-    LVar0 := create_object_sid(1, 0, 0, -1);
+    LVar0 := create_object(1, 0, 0);
     add_obj_to_inven(self_obj, LVar0);
     if (source_obj == dude_obj) then begin
-        set_global_var(5, global_var(5) + 1);
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(4) > (3 * global_var(5))) or (global_var(2) == 1)) and (global_var(1) == 0)) then begin
-            set_global_var(2, 1);
-            set_global_var(3, 0);
+        set_global_var(GVAR_GOOD_MONSTER, global_var(GVAR_GOOD_MONSTER) + 1);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(4) > (3 * global_var(GVAR_GOOD_MONSTER))) or (global_var(GVAR_CHAMPION_REPUTATION) == 1)) and (global_var(GVAR_CHILDKILLER_REPUTATION) == 0)) then begin
+            set_global_var(GVAR_CHAMPION_REPUTATION, 1);
+            set_global_var(GVAR_BERSERKER_REPUTATION, 0);
         end
-        if (metarule(51, self_obj) == 2) then begin
+        if (critter_kill_type(self_obj) == KILL_TYPE_children_kills) then begin
             if (source_obj == dude_obj) then begin
-                set_global_var(2, 0);
-                set_global_var(1, global_var(1) + 1);
-                set_global_var(0, global_var(0) + -15);
+                set_global_var(GVAR_CHAMPION_REPUTATION, 0);
+                set_global_var(GVAR_CHILDKILLER_REPUTATION, global_var(GVAR_CHILDKILLER_REPUTATION) + 1);
+                set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + -15);
                 debug_msg("Player gains " + -15 + " Karma Points.");
-                set_global_var(37, 0);
-                set_global_var(38, 0);
-                set_global_var(39, 0);
-                set_global_var(40, 0);
-                set_global_var(41, 0);
-                set_global_var(42, 0);
-                set_global_var(43, 0);
-                set_global_var(44, 0);
-                set_global_var(45, 0);
-                if (global_var(0) >= 1000) then begin
-                    set_global_var(37, 1);
+                set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+                set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+                set_global_var(GVAR_KARMA_DEFENDER, 0);
+                set_global_var(GVAR_KARMA_WANDERER, 0);
+                set_global_var(GVAR_KARMA_BETRAYER, 0);
+                set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+                set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+                set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+                    set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
                 end
                 else begin
-                    if (global_var(0) >= 750) then begin
-                        set_global_var(38, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
                     end
                     else begin
-                        if (global_var(0) >= 500) then begin
-                            set_global_var(39, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                            set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                         end
                         else begin
-                            if (global_var(0) >= 250) then begin
-                                set_global_var(40, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                                set_global_var(GVAR_KARMA_DEFENDER, 1);
                             end
                             else begin
-                                if (global_var(0) > -250) then begin
-                                    set_global_var(41, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                                    set_global_var(GVAR_KARMA_WANDERER, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -500) then begin
-                                        set_global_var(42, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                        set_global_var(GVAR_KARMA_BETRAYER, 1);
                                     end
                                     else begin
-                                        if (global_var(0) > -750) then begin
-                                            set_global_var(43, 1);
+                                        if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                            set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                         end
                                         else begin
-                                            if (global_var(0) > -1000) then begin
-                                                set_global_var(44, 1);
+                                            if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                                set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                             end
                                             else begin
-                                                set_global_var(45, 1);
+                                                set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                             end
                                         end
                                     end
@@ -669,58 +669,58 @@ begin
                         end
                     end
                 end
-                set_global_var(61, global_var(61) + -8);
+                set_global_var(GVAR_TOWN_REP_SAN_FRANCISCO, global_var(GVAR_TOWN_REP_SAN_FRANCISCO) + -8);
                 debug_msg("Added " + -8 + " to Town Rep");
             end
         end
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(5) > (2 * global_var(4))) or (global_var(3) == 1))) then begin
-            set_global_var(3, 1);
-            set_global_var(2, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
+            set_global_var(GVAR_BERSERKER_REPUTATION, 1);
+            set_global_var(GVAR_CHAMPION_REPUTATION, 0);
         end
-        set_global_var(0, global_var(0) + -10);
+        set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + -10);
         debug_msg("Player gains " + -10 + " Karma Points.");
-        set_global_var(37, 0);
-        set_global_var(38, 0);
-        set_global_var(39, 0);
-        set_global_var(40, 0);
-        set_global_var(41, 0);
-        set_global_var(42, 0);
-        set_global_var(43, 0);
-        set_global_var(44, 0);
-        set_global_var(45, 0);
-        if (global_var(0) >= 1000) then begin
-            set_global_var(37, 1);
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+        set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+        set_global_var(GVAR_KARMA_DEFENDER, 0);
+        set_global_var(GVAR_KARMA_WANDERER, 0);
+        set_global_var(GVAR_KARMA_BETRAYER, 0);
+        set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+        set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+            set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
         end
         else begin
-            if (global_var(0) >= 750) then begin
-                set_global_var(38, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
             end
             else begin
-                if (global_var(0) >= 500) then begin
-                    set_global_var(39, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                 end
                 else begin
-                    if (global_var(0) >= 250) then begin
-                        set_global_var(40, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                        set_global_var(GVAR_KARMA_DEFENDER, 1);
                     end
                     else begin
-                        if (global_var(0) > -250) then begin
-                            set_global_var(41, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                            set_global_var(GVAR_KARMA_WANDERER, 1);
                         end
                         else begin
-                            if (global_var(0) > -500) then begin
-                                set_global_var(42, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                set_global_var(GVAR_KARMA_BETRAYER, 1);
                             end
                             else begin
-                                if (global_var(0) > -750) then begin
-                                    set_global_var(43, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -1000) then begin
-                                        set_global_var(44, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                     end
                                     else begin
-                                        set_global_var(45, 1);
+                                        set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                     end
                                 end
                             end
@@ -729,7 +729,7 @@ begin
                 end
             end
         end
-        set_global_var(61, global_var(61) + -5);
+        set_global_var(GVAR_TOWN_REP_SAN_FRANCISCO, global_var(GVAR_TOWN_REP_SAN_FRANCISCO) + -5);
         debug_msg("Added " + -5 + " to Town Rep");
     end
     return 0;
@@ -740,17 +740,17 @@ procedure look_at_p_proc
 begin
     script_overrides;
     if (local_var(4) == 0) then begin
-        display_msg(message_str(967, 100));
+        display_mstr(100);
     end
     else begin
-        display_msg(message_str(967, 101));
+        display_mstr(101);
     end
 end
 
 procedure description_p_proc
 begin
     script_overrides;
-    display_msg(message_str(967, 102));
+    display_mstr(102);
 end
 
 procedure use_skill_on_p_proc
@@ -761,15 +761,15 @@ procedure damage_p_proc
 begin
     if (party_member_obj(obj_pid(source_obj)) != 0) then begin
         set_local_var(6, 1);
-        set_global_var(361, global_var(361) bwor 262144);
+        set_global_var(GVAR_SAN_FRAN_FLAGS, global_var(GVAR_SAN_FRAN_FLAGS) bwor 262144);
     end
 end
 
 procedure map_enter_p_proc
 begin
     Only_Once := 0;
-    critter_add_trait(self_obj, 1, 6, 198);
-    critter_add_trait(self_obj, 1, 5, 157);
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 198);
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_ELRON);
     Node2Count := 0;
 end
 
@@ -784,171 +784,171 @@ end
 
 procedure Node001
 begin
-    float_msg(self_obj, message_str(967, 103), 2);
+    float_msg(self_obj, mstr(103), FLOAT_MSG_RED);
     callGuards := 1;
 end
 
 procedure Node003
 begin
-    gsay_reply(967, 107);
-    giq_option(-3, 967, 108, Node004, 50);
+    Reply(107);
+    giq_option(-3, NAME, 108, Node004, NEUTRAL_REACTION);
     if (obj_carrying_pid_obj(dude_obj, 457)) then begin
-        giq_option(4, 967, 109, Node005, 50);
+        giq_option(4, NAME, 109, Node005, NEUTRAL_REACTION);
     end
-    giq_option(4, 967, message_str(967, 110) + " " + obj_name(dude_obj) + message_str(967, 161), Node006, 50);
-    giq_option(4, 967, 111, Node007, 51);
-    giq_option(4, 967, 112, Node010, 50);
-    giq_option(4, 967, 113, Node008, 51);
-    giq_option(4, 967, 114, Node999, 50);
+    giq_option(4, NAME, mstr(110) + " " + dude_name + mstr(161), Node006, NEUTRAL_REACTION);
+    giq_option(4, NAME, 111, Node007, BAD_REACTION);
+    giq_option(4, NAME, 112, Node010, NEUTRAL_REACTION);
+    giq_option(4, NAME, 113, Node008, BAD_REACTION);
+    giq_option(4, NAME, 114, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node004
 begin
-    gsay_reply(967, 115);
-    giq_option(4, 967, 116, Node999, 50);
+    Reply(115);
+    giq_option(4, NAME, 116, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node005
 begin
-    gsay_reply(967, 117);
-    giq_option(4, 967, 118, Node999, 50);
+    Reply(117);
+    giq_option(4, NAME, 118, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node006
 begin
-    gsay_reply(967, 119);
-    giq_option(4, 967, 120, Node999, 50);
+    Reply(119);
+    giq_option(4, NAME, 120, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node007
 begin
-    gsay_reply(967, 121);
-    giq_option(4, 967, message_str(967, 122) + " " + obj_name(dude_obj) + message_str(967, 162), Node006, 50);
-    giq_option(4, 967, 123, Node009, 51);
-    giq_option(4, 967, 124, Node010, 50);
-    giq_option(4, 967, 125, Node008, 51);
-    giq_option(4, 967, 126, Node999, 50);
+    Reply(121);
+    giq_option(4, NAME, mstr(122) + " " + dude_name + mstr(162), Node006, NEUTRAL_REACTION);
+    giq_option(4, NAME, 123, Node009, BAD_REACTION);
+    giq_option(4, NAME, 124, Node010, NEUTRAL_REACTION);
+    giq_option(4, NAME, 125, Node008, BAD_REACTION);
+    giq_option(4, NAME, 126, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node008
 begin
-    gsay_reply(967, 127);
-    giq_option(4, 967, message_str(967, 128) + " " + obj_name(dude_obj) + message_str(967, 163), Node006, 50);
-    giq_option(4, 967, 129, Node009, 51);
-    giq_option(4, 967, 130, Node010, 50);
-    giq_option(4, 967, 131, Node999, 50);
+    Reply(127);
+    giq_option(4, NAME, mstr(128) + " " + dude_name + mstr(163), Node006, NEUTRAL_REACTION);
+    giq_option(4, NAME, 129, Node009, BAD_REACTION);
+    giq_option(4, NAME, 130, Node010, NEUTRAL_REACTION);
+    giq_option(4, NAME, 131, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node009
 begin
-    gsay_reply(967, 132);
-    giq_option(4, 967, 133, Node999a, 50);
+    Reply(132);
+    giq_option(4, NAME, 133, Node999a, NEUTRAL_REACTION);
 end
 
 procedure Node010
 begin
-    gsay_reply(967, 134);
-    giq_option(4, 967, 135, Node999, 50);
+    Reply(134);
+    giq_option(4, NAME, 135, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node011
 begin
-    if (global_var(427) == 8) then begin
-        gsay_reply(967, 171);
+    if (global_var(GVAR_SAN_FRAN_ELRON_WHIRLY) == 8) then begin
+        Reply(171);
     end
     else begin
-        gsay_reply(967, 170);
+        Reply(170);
     end
-    giq_option(4, 967, 172, Node012, 50);
-    giq_option(4, 967, 173, Node013, 50);
-    if (obj_carrying_pid_obj(dude_obj, 446) and (global_var(427) == 4)) then begin
-        giq_option(4, 967, 174, Node014, 50);
+    giq_option(4, NAME, 172, Node012, NEUTRAL_REACTION);
+    giq_option(4, NAME, 173, Node013, NEUTRAL_REACTION);
+    if (obj_carrying_pid_obj(dude_obj, 446) and (global_var(GVAR_SAN_FRAN_ELRON_WHIRLY) == 4)) then begin
+        giq_option(4, NAME, 174, Node014, NEUTRAL_REACTION);
     end
-    if (obj_carrying_pid_obj(dude_obj, 446) and ((global_var(427) == 1) or (global_var(427) == 2))) then begin
-        giq_option(4, 967, 175, Node014, 50);
+    if (obj_carrying_pid_obj(dude_obj, 446) and ((global_var(GVAR_SAN_FRAN_ELRON_WHIRLY) == 1) or (global_var(GVAR_SAN_FRAN_ELRON_WHIRLY) == 2))) then begin
+        giq_option(4, NAME, 175, Node014, NEUTRAL_REACTION);
     end
-    if ((global_var(519) == 8) and ((global_var(427) == 1) or (global_var(427) == 2))) then begin
-        giq_option(4, 967, 176, Node015, 50);
+    if ((global_var(GVAR_SAN_FRAN_SHI_WHIRLY) == 8) and ((global_var(GVAR_SAN_FRAN_ELRON_WHIRLY) == 1) or (global_var(GVAR_SAN_FRAN_ELRON_WHIRLY) == 2))) then begin
+        giq_option(4, NAME, 176, Node015, NEUTRAL_REACTION);
     end
-    if (not((global_var(427) == 8) and (((global_var(363) bwand 8) != 0) or ((global_var(361) bwand 32768) != 0)) and (global_var(519) == 4))) then begin
-        giq_option(4, 967, 177, Node1718, 50);
+    if (not((global_var(GVAR_SAN_FRAN_ELRON_WHIRLY) == 8) and (((global_var(GVAR_SAN_FRAN_TANKER) bwand 8) != 0) or ((global_var(GVAR_SAN_FRAN_FLAGS) bwand 32768) != 0)) and (global_var(GVAR_SAN_FRAN_SHI_WHIRLY) == 4))) then begin
+        giq_option(4, NAME, 177, Node1718, NEUTRAL_REACTION);
     end
-    giq_option(4, 967, 178, Node999, 50);
+    giq_option(4, NAME, 178, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node012
 begin
-    gsay_reply(967, 179);
-    giq_option(4, 967, 180, Node011, 50);
-    giq_option(4, 967, 181, Node999, 50);
+    Reply(179);
+    giq_option(4, NAME, 180, Node011, NEUTRAL_REACTION);
+    giq_option(4, NAME, 181, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node013
 begin
-    gsay_reply(967, 182);
-    giq_option(4, 967, 180, Node011, 50);
-    giq_option(4, 967, 181, Node999, 50);
+    Reply(182);
+    giq_option(4, NAME, 180, Node011, NEUTRAL_REACTION);
+    giq_option(4, NAME, 181, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node014
 begin
-    set_global_var(361, global_var(361) bwor 65536);
-    if (global_var(535) == 0) then begin
-        set_global_var(535, 1);
+    set_global_var(GVAR_SAN_FRAN_FLAGS, global_var(GVAR_SAN_FRAN_FLAGS) bwor 65536);
+    if (global_var(GVAR_SAN_FRAN_ELRON_GAS_QST) == 0) then begin
+        set_global_var(GVAR_SAN_FRAN_ELRON_GAS_QST, 1);
     end
-    if (global_var(427) == 4) then begin
-        set_global_var(557, 2);
+    if (global_var(GVAR_SAN_FRAN_ELRON_WHIRLY) == 4) then begin
+        set_global_var(GVAR_SAN_FRAN_VERTI_STEAL_SHI_QST, 2);
     end
     else begin
-        set_global_var(561, 2);
+        set_global_var(GVAR_SAN_FRAN_VERTI_ELE_QST, 2);
     end
-    set_global_var(427, 8);
+    set_global_var(GVAR_SAN_FRAN_ELRON_WHIRLY, 8);
     if (-20 != 0) then begin
-        set_global_var(0, global_var(0) + -20);
+        set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + -20);
     end
     debug_msg("Player gains " + -20 + " Karma Points.");
-    set_global_var(37, 0);
-    set_global_var(38, 0);
-    set_global_var(39, 0);
-    set_global_var(40, 0);
-    set_global_var(41, 0);
-    set_global_var(42, 0);
-    set_global_var(43, 0);
-    set_global_var(44, 0);
-    set_global_var(45, 0);
-    if (global_var(0) >= 1000) then begin
-        set_global_var(37, 1);
+    set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+    set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+    set_global_var(GVAR_KARMA_DEFENDER, 0);
+    set_global_var(GVAR_KARMA_WANDERER, 0);
+    set_global_var(GVAR_KARMA_BETRAYER, 0);
+    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+    set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+    set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+    if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
     end
     else begin
-        if (global_var(0) >= 750) then begin
-            set_global_var(38, 1);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+            set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
         end
         else begin
-            if (global_var(0) >= 500) then begin
-                set_global_var(39, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
             end
             else begin
-                if (global_var(0) >= 250) then begin
-                    set_global_var(40, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                    set_global_var(GVAR_KARMA_DEFENDER, 1);
                 end
                 else begin
-                    if (global_var(0) > -250) then begin
-                        set_global_var(41, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                        set_global_var(GVAR_KARMA_WANDERER, 1);
                     end
                     else begin
-                        if (global_var(0) > -500) then begin
-                            set_global_var(42, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                            set_global_var(GVAR_KARMA_BETRAYER, 1);
                         end
                         else begin
-                            if (global_var(0) > -750) then begin
-                                set_global_var(43, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                             end
                             else begin
-                                if (global_var(0) > -1000) then begin
-                                    set_global_var(44, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                    set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                 end
                                 else begin
-                                    set_global_var(45, 1);
+                                    set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                 end
                             end
                         end
@@ -958,70 +958,70 @@ begin
         end
     end
     if (-5 != 0) then begin
-        set_global_var(61, global_var(61) + 1);
+        set_global_var(GVAR_TOWN_REP_SAN_FRANCISCO, global_var(GVAR_TOWN_REP_SAN_FRANCISCO) + 1);
     end
     if (5000 != 0) then begin
         give_exp_points(5000);
     end
-    display_msg(message_str(14, 100) + 5000 + message_str(14, 101));
-    gsay_reply(967, 185);
-    giq_option(4, 967, 186, NodeEndTake, 50);
+    display_msg(g_mstr(100) + 5000 + g_mstr(101));
+    Reply(185);
+    giq_option(4, NAME, 186, NodeEndTake, NEUTRAL_REACTION);
 end
 
 procedure Node015
 begin
-    gsay_reply(967, 187);
+    Reply(187);
     if (obj_carrying_pid_obj(dude_obj, 446)) then begin
-        giq_option(4, 967, 188, Node014, 50);
+        giq_option(4, NAME, 188, Node014, NEUTRAL_REACTION);
     end
-    giq_option(4, 967, 189, Node999c, 50);
+    giq_option(4, NAME, 189, Node999c, NEUTRAL_REACTION);
 end
 
 procedure Node016
 begin
-    gsay_reply(967, 190);
-    giq_option(4, 967, 191, Node999, 50);
+    Reply(190);
+    giq_option(4, NAME, 191, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node017
 begin
-    if (global_var(427) == 0) then begin
-        gsay_reply(967, 192);
+    if (global_var(GVAR_SAN_FRAN_ELRON_WHIRLY) == 0) then begin
+        Reply(192);
     end
     else begin
-        if (global_var(427) == 2) then begin
-            gsay_reply(967, 193);
+        if (global_var(GVAR_SAN_FRAN_ELRON_WHIRLY) == 2) then begin
+            Reply(193);
         end
         else begin
-            if (global_var(427) == 1) then begin
-                gsay_reply(967, 194);
+            if (global_var(GVAR_SAN_FRAN_ELRON_WHIRLY) == 1) then begin
+                Reply(194);
             end
             else begin
-                if (global_var(427) == 4) then begin
-                    gsay_reply(967, 195);
+                if (global_var(GVAR_SAN_FRAN_ELRON_WHIRLY) == 4) then begin
+                    Reply(195);
                 end
             end
         end
     end
     if (obj_carrying_pid_obj(dude_obj, 446)) then begin
-        giq_option(4, 967, 196, Node014, 50);
+        giq_option(4, NAME, 196, Node014, NEUTRAL_REACTION);
     end
-    giq_option(4, 967, 197, Node999d, 50);
-    giq_option(4, 967, 198, Node999e, 50);
+    giq_option(4, NAME, 197, Node999d, NEUTRAL_REACTION);
+    giq_option(4, NAME, 198, Node999e, NEUTRAL_REACTION);
 end
 
 procedure Node018
 begin
-    if (not((global_var(361) bwand 65536) != 0) and not(((global_var(363) bwand 8) != 0) or ((global_var(361) bwand 32768) != 0))) then begin
-        set_global_var(361, global_var(361) bwor 65536);
-        gsay_reply(967, 199);
+    if (not((global_var(GVAR_SAN_FRAN_FLAGS) bwand 65536) != 0) and not(((global_var(GVAR_SAN_FRAN_TANKER) bwand 8) != 0) or ((global_var(GVAR_SAN_FRAN_FLAGS) bwand 32768) != 0))) then begin
+        set_global_var(GVAR_SAN_FRAN_FLAGS, global_var(GVAR_SAN_FRAN_FLAGS) bwor 65536);
+        Reply(199);
     end
     else begin
-        if ((global_var(361) bwand 65536) != 0) then begin
-            gsay_reply(967, 200);
+        if ((global_var(GVAR_SAN_FRAN_FLAGS) bwand 65536) != 0) then begin
+            Reply(200);
         end
     end
-    giq_option(4, 967, 201, Node999, 50);
+    giq_option(4, NAME, 201, Node999, NEUTRAL_REACTION);
 end
 
 procedure in_bitch_bounds
@@ -1031,8 +1031,8 @@ end
 
 procedure DoCallGuards
 begin
-    set_global_var(477, -1);
-    set_global_var(478, dude_obj);
+    set_global_var(GVAR_SAN_FRAN_EG_NOTIFY, -1);
+    set_global_var(GVAR_SAN_FRAN_EG_A_OBJ, dude_obj);
 end
 
 procedure Node999a
@@ -1043,7 +1043,7 @@ end
 
 procedure Node1718
 begin
-    if (not(global_var(427) == 8)) then begin
+    if (not(global_var(GVAR_SAN_FRAN_ELRON_WHIRLY) == 8)) then begin
         call Node017();
     end
     else begin
@@ -1114,41 +1114,40 @@ end
 
 procedure Node999c
 begin
-    set_global_var(427, 4);
-    if (global_var(557) == 0) then begin
-        set_global_var(557, 1);
+    set_global_var(GVAR_SAN_FRAN_ELRON_WHIRLY, 4);
+    if (global_var(GVAR_SAN_FRAN_VERTI_STEAL_SHI_QST) == 0) then begin
+        set_global_var(GVAR_SAN_FRAN_VERTI_STEAL_SHI_QST, 1);
     end
     call Node999();
 end
 
 procedure Node999d
 begin
-    set_global_var(427, 1);
+    set_global_var(GVAR_SAN_FRAN_ELRON_WHIRLY, 1);
     call Node999();
 end
 
 procedure Node999e
 begin
-    if (not(global_var(427) == 4) or not(global_var(427) == 2)) then begin
-        if (not(global_var(519) == 8)) then begin
-            set_global_var(427, 2);
-            if (metarule(17, 15) == 0) then begin
+    if (not(global_var(GVAR_SAN_FRAN_ELRON_WHIRLY) == 4) or not(global_var(GVAR_SAN_FRAN_ELRON_WHIRLY) == 2)) then begin
+        if (not(global_var(GVAR_SAN_FRAN_SHI_WHIRLY) == 8)) then begin
+            set_global_var(GVAR_SAN_FRAN_ELRON_WHIRLY, 2);
+            if (town_known(15) == 0) then begin
                 debug_msg("  mark_on_map(" + 15 + ")");
                 mark_area_known(0, 15, 1);
             end
-            if (global_var(561) == 0) then begin
-                set_global_var(561, 1);
+            if (global_var(GVAR_SAN_FRAN_VERTI_ELE_QST) == 0) then begin
+                set_global_var(GVAR_SAN_FRAN_VERTI_ELE_QST, 1);
             end
         end
         else begin
-            if (global_var(519) == 8) then begin
-                set_global_var(427, 4);
-                if (global_var(557) == 0) then begin
-                    set_global_var(557, 1);
+            if (global_var(GVAR_SAN_FRAN_SHI_WHIRLY) == 8) then begin
+                set_global_var(GVAR_SAN_FRAN_ELRON_WHIRLY, 4);
+                if (global_var(GVAR_SAN_FRAN_VERTI_STEAL_SHI_QST) == 0) then begin
+                    set_global_var(GVAR_SAN_FRAN_VERTI_STEAL_SHI_QST, 1);
                 end
             end
         end
     end
     call Node999();
 end
-
