diff --git a/Mods/MegaMod/Original/FCSKIDS.ssl b/Mods/MegaMod/ReDefine/FCSKIDS.ssl
index 9bebf50..3888c6e 100644
--- a/Mods/MegaMod/Original/FCSKIDS.ssl
+++ b/Mods/MegaMod/ReDefine/FCSKIDS.ssl
@@ -164,7 +164,7 @@ procedure critter_p_proc
 begin
     if ((local_var(5) == 2) and obj_can_see_obj(self_obj, dude_obj)) then begin
         set_local_var(5, 1);
-        attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+        attack(dude_obj);
     end
 end
 
@@ -177,16 +177,16 @@ end
 
 procedure talk_p_proc
 begin
-    debug_msg("FCKIDS: party_size:  " + metarule(16, 0));
-    if ((global_var(1) >= 2) or ((global_var(361) bwand 1) != 0)) then begin
+    debug_msg("FCKIDS: party_size:  " + party_member_count(DONT_LIST_HIDDEN_MEMBERS));
+    if ((global_var(GVAR_CHILDKILLER_REPUTATION) >= 2) or ((global_var(GVAR_SAN_FRAN_FLAGS) bwand 1) != 0)) then begin
         call Node004();
     end
     else begin
-        if ((obj_pid(critter_inven_obj(dude_obj, 0)) == 3) or (obj_pid(critter_inven_obj(dude_obj, 0)) == 348) or (obj_pid(critter_inven_obj(dude_obj, 0)) == 349) or (obj_pid(critter_inven_obj(dude_obj, 0)) == 232)) then begin
+        if ((obj_pid(critter_inven_obj(dude_obj, INVEN_TYPE_WORN)) == 3) or (obj_pid(critter_inven_obj(dude_obj, INVEN_TYPE_WORN)) == 348) or (obj_pid(critter_inven_obj(dude_obj, INVEN_TYPE_WORN)) == 349) or (obj_pid(critter_inven_obj(dude_obj, INVEN_TYPE_WORN)) == 232)) then begin
             call Node003();
         end
         else begin
-            if (metarule(16, 0) > 1) then begin
+            if (party_member_count(DONT_LIST_HIDDEN_MEMBERS) > 1) then begin
                 call Node002();
             end
             else begin
@@ -199,59 +199,59 @@ end
 procedure destroy_p_proc
 begin
     if (source_obj == dude_obj) then begin
-        set_global_var(5, global_var(5) + 1);
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(4) > (3 * global_var(5))) or (global_var(2) == 1)) and (global_var(1) == 0)) then begin
-            set_global_var(2, 1);
-            set_global_var(3, 0);
+        set_global_var(GVAR_GOOD_MONSTER, global_var(GVAR_GOOD_MONSTER) + 1);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(4) > (3 * global_var(GVAR_GOOD_MONSTER))) or (global_var(GVAR_CHAMPION_REPUTATION) == 1)) and (global_var(GVAR_CHILDKILLER_REPUTATION) == 0)) then begin
+            set_global_var(GVAR_CHAMPION_REPUTATION, 1);
+            set_global_var(GVAR_BERSERKER_REPUTATION, 0);
         end
-        if (metarule(51, self_obj) == 2) then begin
+        if (critter_kill_type(self_obj) == KILL_TYPE_children_kills) then begin
             if (source_obj == dude_obj) then begin
-                set_global_var(2, 0);
-                set_global_var(1, global_var(1) + 1);
-                set_global_var(0, global_var(0) + -15);
+                set_global_var(GVAR_CHAMPION_REPUTATION, 0);
+                set_global_var(GVAR_CHILDKILLER_REPUTATION, global_var(GVAR_CHILDKILLER_REPUTATION) + 1);
+                set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + -15);
                 debug_msg("Player gains " + -15 + " Karma Points.");
-                set_global_var(37, 0);
-                set_global_var(38, 0);
-                set_global_var(39, 0);
-                set_global_var(40, 0);
-                set_global_var(41, 0);
-                set_global_var(42, 0);
-                set_global_var(43, 0);
-                set_global_var(44, 0);
-                set_global_var(45, 0);
-                if (global_var(0) >= 1000) then begin
-                    set_global_var(37, 1);
+                set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+                set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+                set_global_var(GVAR_KARMA_DEFENDER, 0);
+                set_global_var(GVAR_KARMA_WANDERER, 0);
+                set_global_var(GVAR_KARMA_BETRAYER, 0);
+                set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+                set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+                set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+                    set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
                 end
                 else begin
-                    if (global_var(0) >= 750) then begin
-                        set_global_var(38, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
                     end
                     else begin
-                        if (global_var(0) >= 500) then begin
-                            set_global_var(39, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                            set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                         end
                         else begin
-                            if (global_var(0) >= 250) then begin
-                                set_global_var(40, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                                set_global_var(GVAR_KARMA_DEFENDER, 1);
                             end
                             else begin
-                                if (global_var(0) > -250) then begin
-                                    set_global_var(41, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                                    set_global_var(GVAR_KARMA_WANDERER, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -500) then begin
-                                        set_global_var(42, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                        set_global_var(GVAR_KARMA_BETRAYER, 1);
                                     end
                                     else begin
-                                        if (global_var(0) > -750) then begin
-                                            set_global_var(43, 1);
+                                        if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                            set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                         end
                                         else begin
-                                            if (global_var(0) > -1000) then begin
-                                                set_global_var(44, 1);
+                                            if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                                set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                             end
                                             else begin
-                                                set_global_var(45, 1);
+                                                set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                             end
                                         end
                                     end
@@ -260,58 +260,58 @@ begin
                         end
                     end
                 end
-                set_global_var(61, global_var(61) + -8);
+                set_global_var(GVAR_TOWN_REP_SAN_FRANCISCO, global_var(GVAR_TOWN_REP_SAN_FRANCISCO) + -8);
                 debug_msg("Added " + -8 + " to Town Rep");
             end
         end
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(5) > (2 * global_var(4))) or (global_var(3) == 1))) then begin
-            set_global_var(3, 1);
-            set_global_var(2, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
+            set_global_var(GVAR_BERSERKER_REPUTATION, 1);
+            set_global_var(GVAR_CHAMPION_REPUTATION, 0);
         end
-        set_global_var(0, global_var(0) + -10);
+        set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + -10);
         debug_msg("Player gains " + -10 + " Karma Points.");
-        set_global_var(37, 0);
-        set_global_var(38, 0);
-        set_global_var(39, 0);
-        set_global_var(40, 0);
-        set_global_var(41, 0);
-        set_global_var(42, 0);
-        set_global_var(43, 0);
-        set_global_var(44, 0);
-        set_global_var(45, 0);
-        if (global_var(0) >= 1000) then begin
-            set_global_var(37, 1);
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+        set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+        set_global_var(GVAR_KARMA_DEFENDER, 0);
+        set_global_var(GVAR_KARMA_WANDERER, 0);
+        set_global_var(GVAR_KARMA_BETRAYER, 0);
+        set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+        set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+            set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
         end
         else begin
-            if (global_var(0) >= 750) then begin
-                set_global_var(38, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
             end
             else begin
-                if (global_var(0) >= 500) then begin
-                    set_global_var(39, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                 end
                 else begin
-                    if (global_var(0) >= 250) then begin
-                        set_global_var(40, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                        set_global_var(GVAR_KARMA_DEFENDER, 1);
                     end
                     else begin
-                        if (global_var(0) > -250) then begin
-                            set_global_var(41, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                            set_global_var(GVAR_KARMA_WANDERER, 1);
                         end
                         else begin
-                            if (global_var(0) > -500) then begin
-                                set_global_var(42, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                set_global_var(GVAR_KARMA_BETRAYER, 1);
                             end
                             else begin
-                                if (global_var(0) > -750) then begin
-                                    set_global_var(43, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -1000) then begin
-                                        set_global_var(44, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                     end
                                     else begin
-                                        set_global_var(45, 1);
+                                        set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                     end
                                 end
                             end
@@ -320,7 +320,7 @@ begin
                 end
             end
         end
-        set_global_var(61, global_var(61) + -5);
+        set_global_var(GVAR_TOWN_REP_SAN_FRANCISCO, global_var(GVAR_TOWN_REP_SAN_FRANCISCO) + -5);
         debug_msg("Added " + -5 + " to Town Rep");
     end
 end
@@ -328,17 +328,17 @@ end
 procedure look_at_p_proc
 begin
     script_overrides;
-    display_msg(message_str(754, 100));
+    display_mstr(100);
 end
 
 procedure description_p_proc
 begin
     script_overrides;
     if (local_var(4) == 0) then begin
-        display_msg(message_str(754, 101));
+        display_mstr(101);
     end
     else begin
-        display_msg(message_str(754, 102));
+        display_mstr(102);
     end
 end
 
@@ -356,8 +356,8 @@ end
 procedure map_enter_p_proc
 begin
     Only_Once := 0;
-    critter_add_trait(self_obj, 1, 6, 167);
-    critter_add_trait(self_obj, 1, 5, 15);
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 167);
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_CHILD);
     add_timer_event(self_obj, game_ticks(random(1, 120)), 4);
     add_timer_event(self_obj, game_ticks(random(5, 100)), 1);
     starttile := tile_num(self_obj);
@@ -378,7 +378,7 @@ begin
     else begin
         if (fixed_param == 2) then begin
             if (local_var(7) == 3) then begin
-                LVar0 := tile_num_in_direction(tile_num(dude_obj), random(0, 5), random(1, 4));
+                LVar0 := tile_num_in_direction(dude_tile, random(0, 5), random(1, 4));
                 debug_msg("FCKIDS: tile: " + LVar0 + " self_tile: " + tile_num(self_obj));
                 if (tile_num(self_obj) != LVar0) then begin
                     debug_msg("FCKIDS: should be animating");
@@ -402,10 +402,10 @@ begin
                     if (tile_distance_objs(self_obj, dude_obj) < 8) then begin
                         if ((tile_distance_objs(self_obj, dude_obj) < 8) or obj_can_see_obj(self_obj, dude_obj)) then begin
                             if (anim_busy(self_obj) == 0) then begin
-                                global_temp := rotation_to_tile(tile_num(dude_obj), tile_num(self_obj));
+                                global_temp := rotation_to_tile(dude_tile, tile_num(self_obj));
                                 animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(self_obj), global_temp, random(3, 10)), 1);
                                 global_temp := (global_temp + 1) % 6;
-                                while ((anim_busy(self_obj) == 0) and (global_temp != rotation_to_tile(tile_num(dude_obj), tile_num(self_obj)))) do begin
+                                while ((anim_busy(self_obj) == 0) and (global_temp != rotation_to_tile(dude_tile, tile_num(self_obj)))) do begin
                                     debug_msg("flee loop: rot == " + global_temp);
                                     animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(self_obj), global_temp, random(3, 10)), 1);
                                     global_temp := (global_temp + 1) % 6;
@@ -449,7 +449,7 @@ end
 
 procedure Node001
 begin
-    float_msg(self_obj, message_str(754, 103 + random(0, 7)), 8);
+    float_msg(self_obj, mstr(103 + random(0, 7)), FLOAT_MSG_YELLOW);
 end
 
 procedure Node002
@@ -489,7 +489,7 @@ begin
     LVar0 := random(0, LVar1);
     if (party_member_obj(16777376) != 0) then begin
         if (LVar0 == 0) then begin
-            float_msg(self_obj, message_str(754, 111 + random(0, 1)), 0);
+            float_msg(self_obj, mstr(111 + random(0, 1)), FLOAT_MSG_NORMAL);
         end
         else begin
             LVar0 := LVar0 - 1;
@@ -497,7 +497,7 @@ begin
     end
     if (party_member_obj(16777377) != 0) then begin
         if (LVar0 == 0) then begin
-            float_msg(self_obj, message_str(754, 113 + random(0, 1)), 0);
+            float_msg(self_obj, mstr(113 + random(0, 1)), FLOAT_MSG_NORMAL);
         end
         else begin
             LVar0 := LVar0 - 1;
@@ -505,7 +505,7 @@ begin
     end
     if (party_member_obj(16777305) != 0) then begin
         if (LVar0 == 0) then begin
-            float_msg(self_obj, message_str(754, 115 + random(0, 1)), 0);
+            float_msg(self_obj, mstr(115 + random(0, 1)), FLOAT_MSG_NORMAL);
         end
         else begin
             LVar0 := LVar0 - 1;
@@ -513,7 +513,7 @@ begin
     end
     if (party_member_obj(16777313) != 0) then begin
         if (LVar0 == 0) then begin
-            float_msg(self_obj, message_str(754, 117 + random(0, 1)), 0);
+            float_msg(self_obj, mstr(117 + random(0, 1)), FLOAT_MSG_NORMAL);
         end
         else begin
             LVar0 := LVar0 - 1;
@@ -521,7 +521,7 @@ begin
     end
     if (party_member_obj(16777323) != 0) then begin
         if (LVar0 == 0) then begin
-            float_msg(self_obj, message_str(754, 119 + random(0, 1)), 0);
+            float_msg(self_obj, mstr(119 + random(0, 1)), FLOAT_MSG_NORMAL);
         end
         else begin
             LVar0 := LVar0 - 1;
@@ -529,7 +529,7 @@ begin
     end
     if (party_member_obj(16777352) != 0) then begin
         if (LVar0 == 0) then begin
-            float_msg(self_obj, message_str(754, 121 + random(0, 0)), 0);
+            float_msg(self_obj, mstr(121 + 0), FLOAT_MSG_NORMAL);
         end
         else begin
             LVar0 := LVar0 - 1;
@@ -537,7 +537,7 @@ begin
     end
     if (party_member_obj(16777368) != 0) then begin
         if (LVar0 == 0) then begin
-            float_msg(self_obj, message_str(754, 122 + random(0, 0)), 0);
+            float_msg(self_obj, mstr(122 + 0), FLOAT_MSG_NORMAL);
         end
         else begin
             LVar0 := LVar0 - 1;
@@ -545,7 +545,7 @@ begin
     end
     if (party_member_obj(16777379) != 0) then begin
         if (LVar0 == 0) then begin
-            float_msg(self_obj, message_str(754, 123 + random(0, 0)), 0);
+            float_msg(self_obj, mstr(123 + 0), FLOAT_MSG_NORMAL);
         end
         else begin
             LVar0 := LVar0 - 1;
@@ -553,7 +553,7 @@ begin
     end
     if (party_member_obj(16777380) != 0) then begin
         if (LVar0 == 0) then begin
-            float_msg(self_obj, message_str(754, 124 + random(0, 0)), 0);
+            float_msg(self_obj, mstr(124 + 0), FLOAT_MSG_NORMAL);
         end
         else begin
             LVar0 := LVar0 - 1;
@@ -561,7 +561,7 @@ begin
     end
     if (party_member_obj(16777295) != 0) then begin
         if (LVar0 == 0) then begin
-            float_msg(self_obj, message_str(754, 125 + random(0, 0)), 0);
+            float_msg(self_obj, mstr(125 + 0), FLOAT_MSG_NORMAL);
         end
         else begin
             LVar0 := LVar0 - 1;
@@ -583,13 +583,12 @@ begin
             add_timer_event(self_obj, 0, 2);
         end
     end
-    float_msg(self_obj, message_str(754, 126 + LVar0), 0);
+    float_msg(self_obj, mstr(126 + LVar0), FLOAT_MSG_NORMAL);
 end
 
 procedure Node004
 begin
     set_local_var(7, 2);
     add_timer_event(self_obj, 0, 3);
-    float_msg(self_obj, message_str(754, 130 + random(0, 3)), 2);
+    float_msg(self_obj, mstr(130 + random(0, 3)), FLOAT_MSG_RED);
 end
-
