diff --git a/Mods/MegaMod/Original/FCSUBGRD.ssl b/Mods/MegaMod/ReDefine/FCSUBGRD.ssl
index ecfd791..3cfe207 100644
--- a/Mods/MegaMod/Original/FCSUBGRD.ssl
+++ b/Mods/MegaMod/ReDefine/FCSUBGRD.ssl
@@ -24,13 +24,13 @@ end
 procedure look_at_p_proc
 begin
     script_overrides;
-    display_msg(message_str(2485, 101));
+    display_mstr(101);
 end
 
 procedure description_p_proc
 begin
     script_overrides;
-    display_msg(message_str(2485, 101));
+    display_mstr(101);
 end
 
 procedure use_skill_on_p_proc
@@ -45,14 +45,14 @@ procedure damage_p_proc
 begin
     if (source_obj == dude_obj) then begin
         set_local_var(5, 2);
-        set_global_var(361, global_var(361) bwor 1);
+        set_global_var(GVAR_SAN_FRAN_FLAGS, global_var(GVAR_SAN_FRAN_FLAGS) bwor 1);
         set_map_var(0, 1);
     end
 end
 
 procedure map_enter_p_proc
 begin
-    if (global_var(362) == 2) then begin
+    if (global_var(GVAR_SAN_FRAN_SUB) == 2) then begin
         destroy_object(self_obj);
     end
 end
@@ -63,7 +63,7 @@ end
 
 procedure talk_p_proc
 begin
-    float_msg(self_obj, message_str(2485, random(200, 201)), 8);
+    float_msg(self_obj, mstr(random(200, 201)), FLOAT_MSG_YELLOW);
 end
 
 procedure timed_event_p_proc
@@ -81,13 +81,12 @@ procedure critter_p_proc
 begin
     if ((local_var(5) == 2) or map_var(0) and obj_can_see_obj(self_obj, dude_obj)) then begin
         set_local_var(5, 1);
-        attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+        attack(dude_obj);
     end
     else begin
         if ((local_var(7) == 1) and obj_can_see_obj(self_obj, dude_obj) and using_skill(dude_obj, 8)) then begin
             set_local_var(7, 1);
-            float_msg(self_obj, message_str(2485, random(150, 151)), 8);
+            float_msg(self_obj, mstr(random(150, 151)), FLOAT_MSG_YELLOW);
         end
     end
 end
-
