diff --git a/Mods/MegaMod/Original/FCSUBTOR.ssl b/Mods/MegaMod/ReDefine/FCSUBTOR.ssl
index 5433b11..6d9ae7d 100644
--- a/Mods/MegaMod/Original/FCSUBTOR.ssl
+++ b/Mods/MegaMod/ReDefine/FCSUBTOR.ssl
@@ -26,9 +26,9 @@ begin
         add_timer_event(subTech3_ptr, 5, 2);
     end
     else begin
-        if (global_var(362) < 5) then begin
+        if (global_var(GVAR_SAN_FRAN_SUB) < 5) then begin
             script_overrides;
-            display_msg(message_str(2483, 200));
+            display_mstr(200);
         end
     end
 end
@@ -36,13 +36,13 @@ end
 procedure look_at_p_proc
 begin
     script_overrides;
-    display_msg(message_str(2483, 100));
+    display_mstr(100);
 end
 
 procedure description_p_proc
 begin
     script_overrides;
-    display_msg(message_str(2483, 101));
+    display_mstr(101);
 end
 
 procedure timed_event_p_proc
@@ -63,16 +63,16 @@ begin
         add_timer_event(subTech3_ptr, 5, 2);
     end
     else begin
-        if (global_var(362) < 5) then begin
+        if (global_var(GVAR_SAN_FRAN_SUB) < 5) then begin
             if (action_being_used == 13) then begin
                 if (is_success(roll_vs_skill(source_obj, 13, -50))) then begin
                     script_overrides;
-                    display_msg(message_str(2483, 210));
-                    set_global_var(362, 5);
+                    display_mstr(210);
+                    set_global_var(GVAR_SAN_FRAN_SUB, 5);
                 end
                 else begin
                     script_overrides;
-                    display_msg(message_str(2483, 211));
+                    display_mstr(211);
                 end
             end
         end
@@ -89,16 +89,16 @@ begin
         add_timer_event(subTech3_ptr, 5, 2);
     end
     else begin
-        if (global_var(362) < 5) then begin
+        if (global_var(GVAR_SAN_FRAN_SUB) < 5) then begin
             if ((LVar0 == 75) or (LVar0 == 308)) then begin
                 if (is_success(roll_vs_skill(source_obj, 13, -50 + 20))) then begin
                     script_overrides;
-                    display_msg(message_str(2483, 210));
-                    set_global_var(362, 5);
+                    display_mstr(210);
+                    set_global_var(GVAR_SAN_FRAN_SUB, 5);
                 end
                 else begin
                     script_overrides;
-                    display_msg(message_str(2483, 211));
+                    display_mstr(211);
                 end
             end
             else begin
@@ -124,7 +124,7 @@ begin
                                 end
                                 else begin
                                     if ((critter_inven_obj(dude_obj, 2) == restock_obj) or (critter_inven_obj(dude_obj, 1) == restock_obj)) then begin
-                                        metarule(43, dude_obj);
+                                        inven_unwield(dude_obj);
                                     end
                                 end
                             end
@@ -134,7 +134,7 @@ begin
                         restock_obj := 0;
                         restock_amt := 0;
                         set_map_var(2, 1);
-                        display_msg(message_str(2483, 220));
+                        display_mstr(220);
                         add_timer_event(self_obj, game_ticks(120), 1);
                     end
                     else begin
@@ -152,4 +152,3 @@ end
 procedure damage_p_proc
 begin
 end
-
