diff --git a/Mods/MegaMod/Original/FENCE.ssl b/Mods/MegaMod/ReDefine/FENCE.ssl
index 22862ee..2bf5271 100644
--- a/Mods/MegaMod/Original/FENCE.ssl
+++ b/Mods/MegaMod/ReDefine/FENCE.ssl
@@ -216,10 +216,10 @@ begin
             if (anim_busy(self_obj) == 0) then begin
                 if ((tile_distance_objs(self_obj, dude_obj) < 8) or obj_can_see_obj(self_obj, dude_obj)) then begin
                     if (anim_busy(self_obj) == 0) then begin
-                        global_temp := rotation_to_tile(tile_num(dude_obj), tile_num(self_obj));
+                        global_temp := rotation_to_tile(dude_tile, tile_num(self_obj));
                         animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(self_obj), global_temp, random(3, 10)), 1);
                         global_temp := (global_temp + 1) % 6;
-                        while ((anim_busy(self_obj) == 0) and (global_temp != rotation_to_tile(tile_num(dude_obj), tile_num(self_obj)))) do begin
+                        while ((anim_busy(self_obj) == 0) and (global_temp != rotation_to_tile(dude_tile, tile_num(self_obj)))) do begin
                             debug_msg("flee loop: rot == " + global_temp);
                             animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(self_obj), global_temp, random(3, 10)), 1);
                             global_temp := (global_temp + 1) % 6;
@@ -229,55 +229,55 @@ begin
             end
         end
         else begin
-            attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+            attack(dude_obj);
         end
     end
-    if (global_var(995) == 2) then begin
+    if (global_var(GVAR_KILL_THE_JUNKIE) == 2) then begin
         set_local_var(9, 3);
     end
-    if (global_var(992) == 1) then begin
+    if (global_var(GVAR_FREE_JIMMYTHESCREW) == 1) then begin
         set_local_var(9, 1);
     end
-    if (global_var(994) == 1) then begin
+    if (global_var(GVAR_KILL_TUBBY_CHILD) == 1) then begin
         set_local_var(9, 1);
     end
-    if (global_var(995) == 1) then begin
+    if (global_var(GVAR_KILL_THE_JUNKIE) == 1) then begin
         set_local_var(9, 1);
     end
-    if (global_var(1000) == 1) then begin
+    if (global_var(GVAR_TUBBY_TRIBUTE_MONEY) == 1) then begin
         set_local_var(9, 1);
     end
     if ((local_var(10) == 0) and (tile_distance_objs(dude_obj, self_obj) < 8)) then begin
         set_local_var(10, 1);
         dialogue_system_enter;
     end
-    if (global_var(1007) == 1) then begin
-        if (global_var(1000) == 3) then begin
+    if (global_var(GVAR_SENT_TO_GARDEN) == 1) then begin
+        if (global_var(GVAR_TUBBY_TRIBUTE_MONEY) == 3) then begin
             if (local_var(11) == 0) then begin
                 set_local_var(11, 1);
                 set_local_var(9, 4);
             end
         end
     end
-    if (global_var(1008) == 1) then begin
+    if (global_var(GVAR_LINK_THETOWNS) == 1) then begin
         set_local_var(9, 6);
     end
-    if (global_var(1008) == 5) then begin
+    if (global_var(GVAR_LINK_THETOWNS) == 5) then begin
         set_local_var(9, 7);
     end
-    if (global_var(1008) == 6) then begin
+    if (global_var(GVAR_LINK_THETOWNS) == 6) then begin
         if (local_var(13) == 0) then begin
             set_local_var(9, 8);
             set_local_var(13, local_var(13) + 1);
         end
     end
-    if (global_var(1008) == 13) then begin
+    if (global_var(GVAR_LINK_THETOWNS) == 13) then begin
         if (local_var(13) == 0) then begin
             set_local_var(13, local_var(13) + 1);
             set_local_var(9, 10);
         end
     end
-    if (global_var(1008) > 10) then begin
+    if (global_var(GVAR_LINK_THETOWNS) > 10) then begin
         animate_move_obj_to_tile(self_obj, 20450, 0);
     end
     if (tile_num(self_obj) == 20450) then begin
@@ -303,7 +303,7 @@ begin
     end
     else begin
         if (local_var(9) == 2) then begin
-            float_msg(self_obj, message_str(1833, 468), 8);
+            float_msg(self_obj, message_str(SCRIPT_HABVOIN, 468), FLOAT_MSG_YELLOW);
         end
         else begin
             if (local_var(9) == 3) then begin
@@ -331,7 +331,7 @@ begin
                     end
                     else begin
                         if (local_var(9) == 6) then begin
-                            float_msg(self_obj, message_str(1833, 472), 2);
+                            float_msg(self_obj, message_str(SCRIPT_HABVOIN, 472), FLOAT_MSG_RED);
                         end
                         else begin
                             if (local_var(9) == 7) then begin
@@ -351,15 +351,15 @@ begin
                                 end
                                 else begin
                                     if (local_var(9) == 9) then begin
-                                        float_msg(self_obj, message_str(1833, 480), 2);
+                                        float_msg(self_obj, message_str(SCRIPT_HABVOIN, 480), FLOAT_MSG_RED);
                                     end
                                     else begin
                                         if (local_var(9) == 10) then begin
-                                            float_msg(self_obj, message_str(1833, 450), 9);
+                                            float_msg(self_obj, message_str(SCRIPT_HABVOIN, 450), FLOAT_MSG_WHITE);
                                             set_local_var(9, 11);
                                             give_exp_points(2000);
-                                            display_msg(message_str(14, 100) + 2000 + message_str(14, 101));
-                                            display_msg(message_str(1833, 481));
+                                            display_msg(g_mstr(100) + 2000 + g_mstr(101));
+                                            display_msg(message_str(SCRIPT_HABVOIN, 481));
                                         end
                                         else begin
                                             if (local_var(9) == 11) then begin
@@ -371,11 +371,11 @@ begin
                                             end
                                             else begin
                                                 if (local_var(9) == 12) then begin
-                                                    float_msg(self_obj, message_str(1833, 490), 0);
+                                                    float_msg(self_obj, message_str(SCRIPT_HABVOIN, 490), FLOAT_MSG_NORMAL);
                                                 end
                                                 else begin
                                                     if (local_var(9) == 13) then begin
-                                                        float_msg(self_obj, message_str(1833, 502), 2);
+                                                        float_msg(self_obj, message_str(SCRIPT_HABVOIN, 502), FLOAT_MSG_RED);
                                                     end
                                                     else begin
                                                         start_gdialog(1833, self_obj, 4, -1, -1);
@@ -401,54 +401,54 @@ end
 procedure destroy_p_proc
 begin
     if (source_obj == dude_obj) then begin
-        set_global_var(5, global_var(5) + 1);
-        if (metarule(51, self_obj) == 2) then begin
+        set_global_var(GVAR_GOOD_MONSTER, global_var(GVAR_GOOD_MONSTER) + 1);
+        if (critter_kill_type(self_obj) == KILL_TYPE_children_kills) then begin
             if (source_obj == dude_obj) then begin
-                set_global_var(2, 0);
-                set_global_var(1, global_var(1) + 1);
-                set_global_var(0, global_var(0) - -15);
-                set_global_var(37, 0);
-                set_global_var(38, 0);
-                set_global_var(39, 0);
-                set_global_var(40, 0);
-                set_global_var(41, 0);
-                set_global_var(42, 0);
-                set_global_var(43, 0);
-                set_global_var(44, 0);
-                set_global_var(45, 0);
-                if (global_var(0) >= 1000) then begin
-                    set_global_var(37, 1);
+                set_global_var(GVAR_CHAMPION_REPUTATION, 0);
+                set_global_var(GVAR_CHILDKILLER_REPUTATION, global_var(GVAR_CHILDKILLER_REPUTATION) + 1);
+                set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) - -15);
+                set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+                set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+                set_global_var(GVAR_KARMA_DEFENDER, 0);
+                set_global_var(GVAR_KARMA_WANDERER, 0);
+                set_global_var(GVAR_KARMA_BETRAYER, 0);
+                set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+                set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+                set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+                    set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
                 end
                 else begin
-                    if (global_var(0) >= 750) then begin
-                        set_global_var(38, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
                     end
                     else begin
-                        if (global_var(0) >= 500) then begin
-                            set_global_var(39, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                            set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                         end
                         else begin
-                            if (global_var(0) >= 250) then begin
-                                set_global_var(40, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                                set_global_var(GVAR_KARMA_DEFENDER, 1);
                             end
                             else begin
-                                if (global_var(0) > -250) then begin
-                                    set_global_var(41, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                                    set_global_var(GVAR_KARMA_WANDERER, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -500) then begin
-                                        set_global_var(42, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                        set_global_var(GVAR_KARMA_BETRAYER, 1);
                                     end
                                     else begin
-                                        if (global_var(0) > -750) then begin
-                                            set_global_var(43, 1);
+                                        if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                            set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                         end
                                         else begin
-                                            if (global_var(0) > -1000) then begin
-                                                set_global_var(44, 1);
+                                            if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                                set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                             end
                                             else begin
-                                                set_global_var(45, 1);
+                                                set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                             end
                                         end
                                     end
@@ -459,58 +459,58 @@ begin
                 end
             end
         end
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(4) > (3 * global_var(5))) or (global_var(2) == 1)) and (global_var(1) == 0)) then begin
-            set_global_var(2, 1);
-            set_global_var(3, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(4) > (3 * global_var(GVAR_GOOD_MONSTER))) or (global_var(GVAR_CHAMPION_REPUTATION) == 1)) and (global_var(GVAR_CHILDKILLER_REPUTATION) == 0)) then begin
+            set_global_var(GVAR_CHAMPION_REPUTATION, 1);
+            set_global_var(GVAR_BERSERKER_REPUTATION, 0);
         end
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(5) > (2 * global_var(4))) or (global_var(3) == 1))) then begin
-            set_global_var(3, 1);
-            set_global_var(2, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
+            set_global_var(GVAR_BERSERKER_REPUTATION, 1);
+            set_global_var(GVAR_CHAMPION_REPUTATION, 0);
         end
-        set_global_var(0, global_var(0) + -10);
+        set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + -10);
         debug_msg("Player gains " + -10 + " Karma Points.");
-        set_global_var(37, 0);
-        set_global_var(38, 0);
-        set_global_var(39, 0);
-        set_global_var(40, 0);
-        set_global_var(41, 0);
-        set_global_var(42, 0);
-        set_global_var(43, 0);
-        set_global_var(44, 0);
-        set_global_var(45, 0);
-        if (global_var(0) >= 1000) then begin
-            set_global_var(37, 1);
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+        set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+        set_global_var(GVAR_KARMA_DEFENDER, 0);
+        set_global_var(GVAR_KARMA_WANDERER, 0);
+        set_global_var(GVAR_KARMA_BETRAYER, 0);
+        set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+        set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+            set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
         end
         else begin
-            if (global_var(0) >= 750) then begin
-                set_global_var(38, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
             end
             else begin
-                if (global_var(0) >= 500) then begin
-                    set_global_var(39, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                 end
                 else begin
-                    if (global_var(0) >= 250) then begin
-                        set_global_var(40, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                        set_global_var(GVAR_KARMA_DEFENDER, 1);
                     end
                     else begin
-                        if (global_var(0) > -250) then begin
-                            set_global_var(41, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                            set_global_var(GVAR_KARMA_WANDERER, 1);
                         end
                         else begin
-                            if (global_var(0) > -500) then begin
-                                set_global_var(42, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                set_global_var(GVAR_KARMA_BETRAYER, 1);
                             end
                             else begin
-                                if (global_var(0) > -750) then begin
-                                    set_global_var(43, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -1000) then begin
-                                        set_global_var(44, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                     end
                                     else begin
-                                        set_global_var(45, 1);
+                                        set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                     end
                                 end
                             end
@@ -526,17 +526,17 @@ procedure look_at_p_proc
 begin
     script_overrides;
     if (local_var(4) == 0) then begin
-        display_msg(message_str(1833, 100));
+        display_msg(message_str(SCRIPT_HABVOIN, 100));
     end
     else begin
-        display_msg(message_str(1833, 101));
+        display_msg(message_str(SCRIPT_HABVOIN, 101));
     end
 end
 
 procedure description_p_proc
 begin
     script_overrides;
-    display_msg(message_str(1833, 102));
+    display_msg(message_str(SCRIPT_HABVOIN, 102));
 end
 
 procedure use_skill_on_p_proc
@@ -545,7 +545,7 @@ end
 
 procedure damage_p_proc
 begin
-    if (party_member_obj(obj_pid(source_obj)) != 0) then begin
+    if (obj_in_party(source_obj)) then begin
         set_local_var(6, 1);
     end
 end
@@ -577,17 +577,17 @@ end
 
 procedure Node001
 begin
-    if (get_critter_stat(dude_obj, 34) == 0) then begin
-        gsay_reply(1833, 200);
+    if (dude_gender == 0) then begin
+        gsay_reply(SCRIPT_HABVOIN, 200);
     end
     else begin
-        if (get_critter_stat(dude_obj, 34) == 1) then begin
-            gsay_reply(1833, 201);
+        if (dude_gender == 1) then begin
+            gsay_reply(SCRIPT_HABVOIN, 201);
         end
     end
-    giq_option(4, 1833, 202, Node002, 51);
-    giq_option(4, 1833, message_str(1833, 203) + obj_name(dude_obj) + message_str(1833, 204), Node003, 50);
-    giq_option(-3, 1833, 205, Node998, 50);
+    giq_option(4, SCRIPT_HABVOIN, 202, Node002, BAD_REACTION);
+    giq_option(4, SCRIPT_HABVOIN, message_str(SCRIPT_HABVOIN, 203) + dude_name + message_str(SCRIPT_HABVOIN, 204), Node003, NEUTRAL_REACTION);
+    giq_option(-3, SCRIPT_HABVOIN, 205, Node998, NEUTRAL_REACTION);
     if (local_var(4) == 0) then begin
         set_local_var(4, 1);
     end
@@ -595,332 +595,332 @@ end
 
 procedure Node002
 begin
-    gsay_reply(1833, 210);
-    giq_option(4, 1833, 211, Node998, 50);
+    gsay_reply(SCRIPT_HABVOIN, 210);
+    giq_option(4, SCRIPT_HABVOIN, 211, Node998, NEUTRAL_REACTION);
 end
 
 procedure Node003
 begin
-    if (get_critter_stat(dude_obj, 34) == 0) then begin
-        gsay_reply(1833, 220);
+    if (dude_gender == 0) then begin
+        gsay_reply(SCRIPT_HABVOIN, 220);
     end
     else begin
-        if (get_critter_stat(dude_obj, 34) == 1) then begin
-            gsay_reply(1833, 221);
+        if (dude_gender == 1) then begin
+            gsay_reply(SCRIPT_HABVOIN, 221);
         end
     end
-    giq_option(4, 1833, 222, Node004, 50);
-    giq_option(4, 1833, 223, Node005, 50);
+    giq_option(4, SCRIPT_HABVOIN, 222, Node004, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_HABVOIN, 223, Node005, NEUTRAL_REACTION);
 end
 
 procedure Node004
 begin
-    if (get_critter_stat(dude_obj, 34) == 0) then begin
-        gsay_reply(1833, 230);
+    if (dude_gender == 0) then begin
+        gsay_reply(SCRIPT_HABVOIN, 230);
     end
     else begin
-        if (get_critter_stat(dude_obj, 34) == 1) then begin
-            gsay_reply(1833, 231);
+        if (dude_gender == 1) then begin
+            gsay_reply(SCRIPT_HABVOIN, 231);
         end
     end
-    giq_option(4, 1833, 232, Node006, 50);
-    giq_option(4, 1833, 233, Node998, 50);
+    giq_option(4, SCRIPT_HABVOIN, 232, Node006, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_HABVOIN, 233, Node998, NEUTRAL_REACTION);
 end
 
 procedure Node005
 begin
-    if (get_critter_stat(dude_obj, 34) == 0) then begin
-        gsay_reply(1833, 240);
+    if (dude_gender == 0) then begin
+        gsay_reply(SCRIPT_HABVOIN, 240);
     end
     else begin
-        if (get_critter_stat(dude_obj, 34) == 1) then begin
-            gsay_reply(1833, 241);
+        if (dude_gender == 1) then begin
+            gsay_reply(SCRIPT_HABVOIN, 241);
         end
     end
-    giq_option(4, 1833, 242, Node007, 50);
-    giq_option(4, 1833, 243, Node008, 50);
+    giq_option(4, SCRIPT_HABVOIN, 242, Node007, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_HABVOIN, 243, Node008, NEUTRAL_REACTION);
 end
 
 procedure Node006
 begin
-    if (get_critter_stat(dude_obj, 34) == 0) then begin
-        gsay_reply(1833, 250);
+    if (dude_gender == 0) then begin
+        gsay_reply(SCRIPT_HABVOIN, 250);
     end
     else begin
-        if (get_critter_stat(dude_obj, 34) == 1) then begin
-            gsay_reply(1833, 251);
+        if (dude_gender == 1) then begin
+            gsay_reply(SCRIPT_HABVOIN, 251);
         end
     end
-    giq_option(4, 1833, 233, Node998, 50);
-    giq_option(4, 1833, 252, Node998, 50);
+    giq_option(4, SCRIPT_HABVOIN, 233, Node998, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_HABVOIN, 252, Node998, NEUTRAL_REACTION);
 end
 
 procedure Node007
 begin
-    if (get_critter_stat(dude_obj, 34) == 0) then begin
-        gsay_reply(1833, 260);
+    if (dude_gender == 0) then begin
+        gsay_reply(SCRIPT_HABVOIN, 260);
     end
     else begin
-        if (get_critter_stat(dude_obj, 34) == 1) then begin
-            gsay_reply(1833, 261);
+        if (dude_gender == 1) then begin
+            gsay_reply(SCRIPT_HABVOIN, 261);
         end
     end
-    giq_option(4, 1833, 233, Node998, 50);
-    giq_option(4, 1833, 262, Node008, 50);
+    giq_option(4, SCRIPT_HABVOIN, 233, Node998, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_HABVOIN, 262, Node008, NEUTRAL_REACTION);
 end
 
 procedure Node008
 begin
-    if (get_critter_stat(dude_obj, 34) == 0) then begin
-        gsay_reply(1833, 270);
+    if (dude_gender == 0) then begin
+        gsay_reply(SCRIPT_HABVOIN, 270);
     end
     else begin
-        if (get_critter_stat(dude_obj, 34) == 1) then begin
-            gsay_reply(1833, 271);
+        if (dude_gender == 1) then begin
+            gsay_reply(SCRIPT_HABVOIN, 271);
         end
     end
-    giq_option(4, 1833, 272, Node002, 50);
-    giq_option(5, 1833, 273, Node009, 50);
-    giq_option(4, 1833, 274, Node002, 50);
-    giq_option(3, 1833, 275, Node002, 50);
+    giq_option(4, SCRIPT_HABVOIN, 272, Node002, NEUTRAL_REACTION);
+    giq_option(5, SCRIPT_HABVOIN, 273, Node009, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_HABVOIN, 274, Node002, NEUTRAL_REACTION);
+    giq_option(3, SCRIPT_HABVOIN, 275, Node002, NEUTRAL_REACTION);
 end
 
 procedure Node009
 begin
-    if (get_critter_stat(dude_obj, 34) == 0) then begin
-        gsay_reply(1833, 280);
+    if (dude_gender == 0) then begin
+        gsay_reply(SCRIPT_HABVOIN, 280);
     end
     else begin
-        if (get_critter_stat(dude_obj, 34) == 1) then begin
-            gsay_reply(1833, 281);
+        if (dude_gender == 1) then begin
+            gsay_reply(SCRIPT_HABVOIN, 281);
         end
     end
-    giq_option(4, 1833, 282, Node010, 50);
-    giq_option(4, 1833, 283, Node011, 50);
+    giq_option(4, SCRIPT_HABVOIN, 282, Node010, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_HABVOIN, 283, Node011, NEUTRAL_REACTION);
 end
 
 procedure Node010
 begin
-    if (not((global_var(315) bwand 512) != 0)) then begin
-        if (get_critter_stat(dude_obj, 34) == 0) then begin
-            gsay_reply(1833, 290);
+    if (not((global_var(GVAR_NEW_RENO_SALVATORE) bwand 512) != 0)) then begin
+        if (dude_gender == 0) then begin
+            gsay_reply(SCRIPT_HABVOIN, 290);
         end
         else begin
-            if (get_critter_stat(dude_obj, 34) == 1) then begin
-                gsay_reply(1833, 291);
+            if (dude_gender == 1) then begin
+                gsay_reply(SCRIPT_HABVOIN, 291);
             end
         end
     end
     else begin
-        if (get_critter_stat(dude_obj, 34) == 0) then begin
-            gsay_reply(1833, 500);
+        if (dude_gender == 0) then begin
+            gsay_reply(SCRIPT_HABVOIN, 500);
         end
         else begin
-            if (get_critter_stat(dude_obj, 34) == 1) then begin
-                gsay_reply(1833, 501);
+            if (dude_gender == 1) then begin
+                gsay_reply(SCRIPT_HABVOIN, 501);
             end
         end
     end
-    if (not((global_var(315) bwand 512) != 0)) then begin
-        giq_option(4, 1833, 292, Node012, 50);
+    if (not((global_var(GVAR_NEW_RENO_SALVATORE) bwand 512) != 0)) then begin
+        giq_option(4, SCRIPT_HABVOIN, 292, Node012, NEUTRAL_REACTION);
     end
-    if (not((global_var(315) bwand 512) != 0)) then begin
-        giq_option(4, 1833, 293, SALVATORE_QUEST_BEGIN, 50);
+    if (not((global_var(GVAR_NEW_RENO_SALVATORE) bwand 512) != 0)) then begin
+        giq_option(4, SCRIPT_HABVOIN, 293, SALVATORE_QUEST_BEGIN, NEUTRAL_REACTION);
     end
-    if ((global_var(315) bwand 512) != 0) then begin
-        giq_option(4, 1833, 496, Node010a, 50);
+    if ((global_var(GVAR_NEW_RENO_SALVATORE) bwand 512) != 0) then begin
+        giq_option(4, SCRIPT_HABVOIN, 496, Node010a, NEUTRAL_REACTION);
     end
 end
 
 procedure Node011
 begin
-    if (get_critter_stat(dude_obj, 34) == 0) then begin
-        gsay_reply(1833, 300);
+    if (dude_gender == 0) then begin
+        gsay_reply(SCRIPT_HABVOIN, 300);
     end
     else begin
-        if (get_critter_stat(dude_obj, 34) == 1) then begin
-            gsay_reply(1833, 301);
+        if (dude_gender == 1) then begin
+            gsay_reply(SCRIPT_HABVOIN, 301);
         end
     end
-    giq_option(4, 1833, 302, Node002, 50);
+    giq_option(4, SCRIPT_HABVOIN, 302, Node002, NEUTRAL_REACTION);
 end
 
 procedure Node012
 begin
-    gsay_reply(1833, 310);
-    giq_option(4, 1833, 311, Node013, 50);
-    giq_option(4, 1833, 312, SALVATORE_QUEST_BEGIN, 50);
+    gsay_reply(SCRIPT_HABVOIN, 310);
+    giq_option(4, SCRIPT_HABVOIN, 311, Node013, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_HABVOIN, 312, SALVATORE_QUEST_BEGIN, NEUTRAL_REACTION);
 end
 
 procedure Node013
 begin
-    gsay_reply(1833, 320);
-    giq_option(4, 1833, 312, SALVATORE_QUEST_BEGIN, 50);
-    giq_option(4, 1833, 321, Node014, 50);
-    giq_option(4, 1833, 322, Node015, 50);
+    gsay_reply(SCRIPT_HABVOIN, 320);
+    giq_option(4, SCRIPT_HABVOIN, 312, SALVATORE_QUEST_BEGIN, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_HABVOIN, 321, Node014, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_HABVOIN, 322, Node015, NEUTRAL_REACTION);
 end
 
 procedure Node014
 begin
-    if (get_critter_stat(dude_obj, 34) == 0) then begin
-        gsay_reply(1833, 330);
+    if (dude_gender == 0) then begin
+        gsay_reply(SCRIPT_HABVOIN, 330);
     end
     else begin
-        if (get_critter_stat(dude_obj, 34) == 1) then begin
-            gsay_reply(1833, 330);
+        if (dude_gender == 1) then begin
+            gsay_reply(SCRIPT_HABVOIN, 330);
         end
     end
-    giq_option(4, 1833, 322, Node015, 50);
-    giq_option(4, 1833, 331, SALVATORE_QUEST_BEGIN, 50);
+    giq_option(4, SCRIPT_HABVOIN, 322, Node015, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_HABVOIN, 331, SALVATORE_QUEST_BEGIN, NEUTRAL_REACTION);
 end
 
 procedure Node015
 begin
-    gsay_reply(1833, 340);
-    giq_option(4, 1833, 341, Node014, 50);
-    giq_option(4, 1833, 342, SALVATORE_QUEST_BEGIN, 50);
-    giq_option(4, 1833, 343, Node016, 50);
+    gsay_reply(SCRIPT_HABVOIN, 340);
+    giq_option(4, SCRIPT_HABVOIN, 341, Node014, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_HABVOIN, 342, SALVATORE_QUEST_BEGIN, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_HABVOIN, 343, Node016, NEUTRAL_REACTION);
 end
 
 procedure Node016
 begin
-    if (get_critter_stat(dude_obj, 34) == 0) then begin
-        gsay_reply(1833, 350);
+    if (dude_gender == 0) then begin
+        gsay_reply(SCRIPT_HABVOIN, 350);
     end
     else begin
-        if (get_critter_stat(dude_obj, 34) == 1) then begin
-            gsay_reply(1833, 350);
+        if (dude_gender == 1) then begin
+            gsay_reply(SCRIPT_HABVOIN, 350);
         end
     end
-    giq_option(4, 1833, 351, Node017, 50);
+    giq_option(4, SCRIPT_HABVOIN, 351, Node017, NEUTRAL_REACTION);
 end
 
 procedure Node017
 begin
-    if (get_critter_stat(dude_obj, 34) == 0) then begin
-        gsay_reply(1833, 360);
+    if (dude_gender == 0) then begin
+        gsay_reply(SCRIPT_HABVOIN, 360);
     end
     else begin
-        if (get_critter_stat(dude_obj, 34) == 1) then begin
-            gsay_reply(1833, 361);
+        if (dude_gender == 1) then begin
+            gsay_reply(SCRIPT_HABVOIN, 361);
         end
     end
-    giq_option(4, 1833, 362, Node014, 50);
-    giq_option(4, 1833, 363, SALVATORE_QUEST_BEGIN, 50);
+    giq_option(4, SCRIPT_HABVOIN, 362, Node014, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_HABVOIN, 363, SALVATORE_QUEST_BEGIN, NEUTRAL_REACTION);
 end
 
 procedure Node018
 begin
-    gsay_reply(1833, message_str(1833, 370) + obj_name(dude_obj) + message_str(1833, 371));
-    if (global_var(992) == 2) then begin
-        giq_option(4, 1833, 372, Node021, 50);
+    gsay_reply(SCRIPT_HABVOIN, message_str(SCRIPT_HABVOIN, 370) + dude_name + message_str(SCRIPT_HABVOIN, 371));
+    if (global_var(GVAR_FREE_JIMMYTHESCREW) == 2) then begin
+        giq_option(4, SCRIPT_HABVOIN, 372, Node021, NEUTRAL_REACTION);
     end
-    if (global_var(994) == 4) then begin
-        giq_option(4, 1833, 374, Node025, 50);
+    if (global_var(GVAR_KILL_TUBBY_CHILD) == 4) then begin
+        giq_option(4, SCRIPT_HABVOIN, 374, Node025, NEUTRAL_REACTION);
     end
-    if ((global_var(995) == 3) and (obj_is_carrying_obj_pid(dude_obj, 563) == 1)) then begin
-        giq_option(4, 1833, 375, Node027, 50);
+    if ((global_var(GVAR_KILL_THE_JUNKIE) == 3) and (obj_is_carrying_obj_pid(dude_obj, 563) == 1)) then begin
+        giq_option(4, SCRIPT_HABVOIN, 375, Node027, NEUTRAL_REACTION);
     end
-    if (global_var(1000) == 2) then begin
-        giq_option(4, 1833, 376, Node028, 50);
+    if (global_var(GVAR_TUBBY_TRIBUTE_MONEY) == 2) then begin
+        giq_option(4, SCRIPT_HABVOIN, 376, Node028, NEUTRAL_REACTION);
     end
-    if ((global_var(1007) == 1) and (global_var(1000) == 3)) then begin
-        giq_option(4, 1833, 377, Node030, 50);
+    if ((global_var(GVAR_SENT_TO_GARDEN) == 1) and (global_var(GVAR_TUBBY_TRIBUTE_MONEY) == 3)) then begin
+        giq_option(4, SCRIPT_HABVOIN, 377, Node030, NEUTRAL_REACTION);
     end
-    giq_option(4, 1833, 373, Node999, 50);
+    giq_option(4, SCRIPT_HABVOIN, 373, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node021
 begin
-    if (get_critter_stat(dude_obj, 34) == 0) then begin
-        gsay_reply(1833, 380);
+    if (dude_gender == 0) then begin
+        gsay_reply(SCRIPT_HABVOIN, 380);
     end
     else begin
-        if (get_critter_stat(dude_obj, 34) == 1) then begin
-            gsay_reply(1833, 381);
+        if (dude_gender == 1) then begin
+            gsay_reply(SCRIPT_HABVOIN, 381);
         end
     end
-    giq_option(4, 1833, 382, Node022, 50);
-    giq_option(4, 1833, 384, Node024, 50);
-    set_global_var(992, 3);
+    giq_option(4, SCRIPT_HABVOIN, 382, Node022, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_HABVOIN, 384, Node024, NEUTRAL_REACTION);
+    set_global_var(GVAR_FREE_JIMMYTHESCREW, 3);
     set_local_var(9, 0);
 end
 
 procedure Node022
 begin
-    gsay_reply(1833, 390);
-    giq_option(4, 1833, 391, Node024, 50);
+    gsay_reply(SCRIPT_HABVOIN, 390);
+    giq_option(4, SCRIPT_HABVOIN, 391, Node024, NEUTRAL_REACTION);
 end
 
 procedure Node023
 begin
-    gsay_reply(1833, 466);
-    giq_option(4, 1833, 467, Node024a, 50);
+    gsay_reply(SCRIPT_HABVOIN, 466);
+    giq_option(4, SCRIPT_HABVOIN, 467, Node024a, NEUTRAL_REACTION);
 end
 
 procedure Node024
 begin
-    gsay_reply(1833, 400);
-    giq_option(4, 1833, 401, Node023, 50);
+    gsay_reply(SCRIPT_HABVOIN, 400);
+    giq_option(4, SCRIPT_HABVOIN, 401, Node023, NEUTRAL_REACTION);
 end
 
 procedure Node024a
 begin
-    set_global_var(994, 1);
-    set_global_var(1001, 1);
+    set_global_var(GVAR_KILL_TUBBY_CHILD, 1);
+    set_global_var(GVAR_ENTER_BROKEN_HILLS_BASEMENT, 1);
     give_exp_points(100);
-    display_msg(message_str(14, 100) + 100 + message_str(14, 101));
-    if (metarule(17, 13) == 0) then begin
+    display_msg(g_mstr(100) + 100 + g_mstr(101));
+    if (town_known(13) == MARK_STATE_UNKNOWN) then begin
         debug_msg("  mark_on_map(" + 13 + ")");
-        mark_area_known(0, 13, 1);
+        mark_area_known(MARK_TYPE_TOWN, 13, MARK_STATE_KNOWN);
     end
 end
 
 procedure Node025
 begin
-    if (get_critter_stat(dude_obj, 34) == 0) then begin
-        gsay_reply(1833, 410);
+    if (dude_gender == 0) then begin
+        gsay_reply(SCRIPT_HABVOIN, 410);
     end
     else begin
-        if (get_critter_stat(dude_obj, 34) == 1) then begin
-            gsay_reply(1833, 411);
+        if (dude_gender == 1) then begin
+            gsay_reply(SCRIPT_HABVOIN, 411);
         end
     end
-    giq_option(4, 1833, 412, Node026, 50);
-    set_global_var(994, 5);
+    giq_option(4, SCRIPT_HABVOIN, 412, Node026, NEUTRAL_REACTION);
+    set_global_var(GVAR_KILL_TUBBY_CHILD, 5);
     set_local_var(9, 0);
 end
 
 procedure Node026
 begin
-    if (get_critter_stat(dude_obj, 34) == 0) then begin
-        gsay_reply(1833, 420);
+    if (dude_gender == 0) then begin
+        gsay_reply(SCRIPT_HABVOIN, 420);
     end
     else begin
-        if (get_critter_stat(dude_obj, 34) == 1) then begin
-            gsay_reply(1833, 421);
+        if (dude_gender == 1) then begin
+            gsay_reply(SCRIPT_HABVOIN, 421);
         end
     end
-    giq_option(4, 1833, 465, Node026a, 50);
+    giq_option(4, SCRIPT_HABVOIN, 465, Node026a, NEUTRAL_REACTION);
 end
 
 procedure Node026a
 begin
-    set_global_var(995, 1);
-    if (metarule(17, 75) == 0) then begin
+    set_global_var(GVAR_KILL_THE_JUNKIE, 1);
+    if (town_known(75) == MARK_STATE_UNKNOWN) then begin
         debug_msg("  mark_on_map(" + 75 + ")");
-        mark_area_known(0, 75, 1);
+        mark_area_known(MARK_TYPE_TOWN, 75, MARK_STATE_KNOWN);
     end
     give_exp_points(200);
-    display_msg(message_str(14, 100) + 200 + message_str(14, 101));
+    display_msg(g_mstr(100) + 200 + g_mstr(101));
 end
 
 procedure Node027
 begin
-    gsay_reply(1833, 430);
-    giq_option(4, 1833, 431, Node999, 50);
-    set_global_var(995, 4);
-    set_global_var(1000, 1);
+    gsay_reply(SCRIPT_HABVOIN, 430);
+    giq_option(4, SCRIPT_HABVOIN, 431, Node999, NEUTRAL_REACTION);
+    set_global_var(GVAR_KILL_THE_JUNKIE, 4);
+    set_global_var(GVAR_TUBBY_TRIBUTE_MONEY, 1);
     set_local_var(9, 0);
     removed_qty := obj_is_carrying_obj_pid(dude_obj, 563);
     if (1 < removed_qty) then begin
@@ -930,17 +930,17 @@ begin
         restock_obj := obj_carrying_pid_obj(dude_obj, 563);
         if (obj_type(self_obj) == 1) then begin
             restock_obj := obj_carrying_pid_obj(dude_obj, 563);
-            if (critter_inven_obj(self_obj, 0) == restock_obj) then begin
-                if (obj_item_subtype(critter_inven_obj(dude_obj, 0)) == 0) then begin
-                    restock_obj := critter_inven_obj(dude_obj, 0);
+            if (critter_inven_obj(self_obj, INVEN_TYPE_WORN) == restock_obj) then begin
+                if (obj_item_subtype(critter_inven_obj(dude_obj, INVEN_TYPE_WORN)) == 0) then begin
+                    restock_obj := critter_inven_obj(dude_obj, INVEN_TYPE_WORN);
                     debug_msg("armour pid == " + obj_pid(restock_obj));
                     rm_obj_from_inven(dude_obj, restock_obj);
                     add_obj_to_inven(dude_obj, restock_obj);
                 end
             end
             else begin
-                if ((critter_inven_obj(dude_obj, 2) == restock_obj) or (critter_inven_obj(dude_obj, 1) == restock_obj)) then begin
-                    metarule(43, dude_obj);
+                if ((critter_inven_obj(dude_obj, INVEN_TYPE_LEFT_HAND) == restock_obj) or (critter_inven_obj(dude_obj, INVEN_TYPE_RIGHT_HAND) == restock_obj)) then begin
+                    inven_unwield(dude_obj);
                 end
             end
         end
@@ -950,43 +950,43 @@ begin
     restock_obj := 0;
     restock_amt := 0;
     give_exp_points(400);
-    display_msg(message_str(14, 100) + 400 + message_str(14, 101));
+    display_msg(g_mstr(100) + 400 + g_mstr(101));
 end
 
 procedure Node028
 begin
-    gsay_reply(1833, 440);
-    giq_option(4, 1833, 465, Node999, 50);
+    gsay_reply(SCRIPT_HABVOIN, 440);
+    giq_option(4, SCRIPT_HABVOIN, 465, Node999, NEUTRAL_REACTION);
     set_local_var(9, 2);
-    set_global_var(1000, 3);
+    set_global_var(GVAR_TUBBY_TRIBUTE_MONEY, 3);
 end
 
 procedure Node029
 begin
-    gsay_reply(1833, 450);
-    giq_option(4, 1833, 100, Node999, 50);
+    gsay_reply(SCRIPT_HABVOIN, 450);
+    giq_option(4, SCRIPT_HABVOIN, 100, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node030
 begin
-    if (global_var(1006) == 2) then begin
-        gsay_reply(1833, 460);
+    if (global_var(GVAR_LAST_BATTLE_BEGIN) == 2) then begin
+        gsay_reply(SCRIPT_HABVOIN, 460);
     end
     else begin
-        if (global_var(1006) == 1) then begin
-            gsay_reply(1833, 469);
+        if (global_var(GVAR_LAST_BATTLE_BEGIN) == 1) then begin
+            gsay_reply(SCRIPT_HABVOIN, 469);
         end
     end
-    giq_option(4, 1833, 465, Node030a, 50);
+    giq_option(4, SCRIPT_HABVOIN, 465, Node030a, NEUTRAL_REACTION);
 end
 
 procedure Node030a
 begin
-    if (global_var(1006) == 2) then begin
-        set_global_var(1008, 1);
+    if (global_var(GVAR_LAST_BATTLE_BEGIN) == 2) then begin
+        set_global_var(GVAR_LINK_THETOWNS, 1);
     end
     else begin
-        if (global_var(1006) == 1) then begin
+        if (global_var(GVAR_LAST_BATTLE_BEGIN) == 1) then begin
             set_local_var(9, 5);
         end
     end
@@ -994,52 +994,52 @@ end
 
 procedure Node031
 begin
-    gsay_reply(1833, 461);
-    giq_option(4, 1833, 462, Node032, 50);
+    gsay_reply(SCRIPT_HABVOIN, 461);
+    giq_option(4, SCRIPT_HABVOIN, 462, Node032, NEUTRAL_REACTION);
 end
 
 procedure Node032
 begin
-    gsay_reply(1833, 463);
-    giq_option(4, 1833, 464, Node998, 50);
+    gsay_reply(SCRIPT_HABVOIN, 463);
+    giq_option(4, SCRIPT_HABVOIN, 464, Node998, NEUTRAL_REACTION);
 end
 
 procedure Node033
 begin
-    if (global_var(1006) == 2) then begin
-        gsay_reply(1833, 471);
+    if (global_var(GVAR_LAST_BATTLE_BEGIN) == 2) then begin
+        gsay_reply(SCRIPT_HABVOIN, 471);
     end
     else begin
-        gsay_reply(1833, 470);
+        gsay_reply(SCRIPT_HABVOIN, 470);
     end
-    giq_option(4, 1833, 465, Node030a, 50);
+    giq_option(4, SCRIPT_HABVOIN, 465, Node030a, NEUTRAL_REACTION);
 end
 
 procedure Node034
 begin
-    gsay_reply(1833, 473);
-    giq_option(4, 1833, 474, Node035, 50);
+    gsay_reply(SCRIPT_HABVOIN, 473);
+    giq_option(4, SCRIPT_HABVOIN, 474, Node035, NEUTRAL_REACTION);
 end
 
 procedure Node035
 begin
-    gsay_reply(1833, 475);
-    giq_option(4, 1833, 465, Node035a, 50);
-    set_global_var(1008, 7);
+    gsay_reply(SCRIPT_HABVOIN, 475);
+    giq_option(4, SCRIPT_HABVOIN, 465, Node035a, NEUTRAL_REACTION);
+    set_global_var(GVAR_LINK_THETOWNS, 7);
     set_local_var(9, 9);
-    critter_add_trait(self_obj, 1, 6, 0);
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 0);
 end
 
 procedure Node036
 begin
-    gsay_reply(1833, 476);
-    giq_option(4, 1833, 477, Node037, 50);
+    gsay_reply(SCRIPT_HABVOIN, 476);
+    giq_option(4, SCRIPT_HABVOIN, 477, Node037, NEUTRAL_REACTION);
 end
 
 procedure Node037
 begin
-    gsay_reply(1833, 478);
-    giq_option(4, 1833, 479, Node042, 50);
+    gsay_reply(SCRIPT_HABVOIN, 478);
+    giq_option(4, SCRIPT_HABVOIN, 479, Node042, NEUTRAL_REACTION);
 end
 
 procedure Node035a
@@ -1052,30 +1052,30 @@ begin
     variable LVar0 := 0;
     game_ui_disable;
     gfade_out(10 * 6);
-    LVar0 := create_object_sid(16777715, 14653, 2, 1345);
+    LVar0 := create_object_sid(16777715, 14653, 2, SCRIPT_KCBDNTN);
     critter_attempt_placement(LVar0, 14653, 2);
     anim(LVar0, 1000, 2);
-    restock_obj := create_object_sid(9, 0, 0, -1);
+    restock_obj := create_object(9, 0, 0);
     add_mult_objs_to_inven(LVar0, restock_obj, 1);
-    LVar0 := create_object_sid(16777715, 14253, 2, 1345);
+    LVar0 := create_object_sid(16777715, 14253, 2, SCRIPT_KCBDNTN);
     critter_attempt_placement(LVar0, 14253, 2);
     anim(LVar0, 1000, 2);
-    restock_obj := create_object_sid(16, 0, 0, -1);
+    restock_obj := create_object(16, 0, 0);
     add_mult_objs_to_inven(LVar0, restock_obj, 1);
-    LVar0 := create_object_sid(16777715, 14654, 2, 1345);
+    LVar0 := create_object_sid(16777715, 14654, 2, SCRIPT_KCBDNTN);
     critter_attempt_placement(LVar0, 14654, 2);
     anim(LVar0, 1000, 2);
-    restock_obj := create_object_sid(24, 0, 0, -1);
+    restock_obj := create_object(24, 0, 0);
     add_mult_objs_to_inven(LVar0, restock_obj, 1);
-    LVar0 := create_object_sid(16777713, 14455, 2, 1345);
+    LVar0 := create_object_sid(16777713, 14455, 2, SCRIPT_KCBDNTN);
     critter_attempt_placement(LVar0, 14455, 2);
     anim(LVar0, 1000, 2);
-    restock_obj := create_object_sid(94, 0, 0, -1);
+    restock_obj := create_object(94, 0, 0);
     add_mult_objs_to_inven(LVar0, restock_obj, 1);
-    LVar0 := create_object_sid(16777714, 14250, 2, 1346);
+    LVar0 := create_object_sid(16777714, 14250, 2, SCRIPT_KIDNDOOR);
     critter_attempt_placement(LVar0, 14250, 2);
     anim(LVar0, 1000, 2);
-    restock_obj := create_object_sid(21, 0, 0, -1);
+    restock_obj := create_object(21, 0, 0);
     add_mult_objs_to_inven(LVar0, restock_obj, 1);
     game_ui_enable;
     gfade_in(10 * 6);
@@ -1085,46 +1085,46 @@ end
 
 procedure Node038
 begin
-    gsay_reply(1833, 482);
-    giq_option(4, 1833, 483, Node039, 50);
+    gsay_reply(SCRIPT_HABVOIN, 482);
+    giq_option(4, SCRIPT_HABVOIN, 483, Node039, NEUTRAL_REACTION);
 end
 
 procedure Node039
 begin
-    gsay_reply(1833, 484);
-    giq_option(4, 1833, 485, Node040, 50);
+    gsay_reply(SCRIPT_HABVOIN, 484);
+    giq_option(4, SCRIPT_HABVOIN, 485, Node040, NEUTRAL_REACTION);
 end
 
 procedure Node040
 begin
-    gsay_reply(1833, 486);
-    giq_option(4, 1833, 487, Node041, 50);
+    gsay_reply(SCRIPT_HABVOIN, 486);
+    giq_option(4, SCRIPT_HABVOIN, 487, Node041, NEUTRAL_REACTION);
 end
 
 procedure Node041
 begin
-    gsay_reply(1833, 488);
-    giq_option(4, 1833, 489, Node041a, 50);
+    gsay_reply(SCRIPT_HABVOIN, 488);
+    giq_option(4, SCRIPT_HABVOIN, 489, Node041a, NEUTRAL_REACTION);
 end
 
 procedure Node041a
 begin
     set_local_var(9, 12);
-    set_global_var(1012, 1);
+    set_global_var(GVAR_HERBS_FOR_SHAMAN, 1);
 end
 
 procedure Node042
 begin
-    gsay_reply(1833, 495);
-    giq_option(4, 1833, 496, Node043, 50);
+    gsay_reply(SCRIPT_HABVOIN, 495);
+    giq_option(4, SCRIPT_HABVOIN, 496, Node043, NEUTRAL_REACTION);
 end
 
 procedure Node043
 begin
     set_local_var(9, 11);
     give_exp_points(2000);
-    display_msg(message_str(14, 100) + 2000 + message_str(14, 101));
-    display_msg(message_str(1833, 481));
+    display_msg(g_mstr(100) + 2000 + g_mstr(101));
+    display_msg(message_str(SCRIPT_HABVOIN, 481));
 end
 
 procedure Node010a
@@ -1134,12 +1134,12 @@ end
 
 procedure SALVATORE_QUEST_BEGIN
 begin
-    set_global_var(992, 1);
-    restock_obj := create_object_sid(561, 0, 0, -1);
+    set_global_var(GVAR_FREE_JIMMYTHESCREW, 1);
+    restock_obj := create_object(561, 0, 0);
     add_mult_objs_to_inven(dude_obj, restock_obj, 1);
-    if (metarule(17, 7) == 0) then begin
+    if (town_known(7) == MARK_STATE_UNKNOWN) then begin
         debug_msg("  mark_on_map(" + 7 + ")");
-        mark_area_known(0, 7, 1);
+        mark_area_known(MARK_TYPE_TOWN, 7, MARK_STATE_KNOWN);
     end
 end
 
@@ -1150,4 +1150,3 @@ end
 procedure Node020
 begin
 end
-
