diff --git a/Mods/MegaMod/Original/FITGUNTB.ssl b/Mods/MegaMod/ReDefine/FITGUNTB.ssl
index 6267eab..2624b8f 100644
--- a/Mods/MegaMod/Original/FITGUNTB.ssl
+++ b/Mods/MegaMod/ReDefine/FITGUNTB.ssl
@@ -190,7 +190,7 @@ begin
     if (validHandle(i_tgun_merchant)) then begin
         if (local_var(0) < game_time) then begin
             if (tile_num(self_obj) == 16293) then begin
-                LVar0 := create_object_sid(128, 5, 0, -1);
+                LVar0 := create_object(128, 5, 0);
                 if (obj_is_carrying_obj_pid(self_obj, 479) > 0) then begin
                     LVar1 := obj_carrying_pid_obj(self_obj, 479);
                     rm_obj_from_inven(self_obj, LVar1);
@@ -436,14 +436,14 @@ begin
                     rm_obj_from_inven(self_obj, LVar1);
                     add_obj_to_inven(LVar0, LVar1);
                 end
-                global_temp := create_object_sid(128, 0, 0, -1);
+                global_temp := create_object(128, 0, 0);
                 move_obj_inven_to_obj(self_obj, global_temp);
                 destroy_object(global_temp);
                 restock_amt := random(300, 436);
                 if (obj_is_carrying_obj_pid(self_obj, 41) < restock_amt) then begin
                     if (80 >= random(1, 100)) then begin
                         if (obj_is_carrying_obj_pid(self_obj, 41) < restock_amt) then begin
-                            restock_obj := create_object_sid(41, 0, 0, -1);
+                            restock_obj := create_object(41, 0, 0);
                             add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 41));
                         end
                         else begin
@@ -466,7 +466,7 @@ begin
                                         end
                                         else begin
                                             if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                                metarule(43, self_obj);
+                                                inven_unwield(self_obj);
                                             end
                                         end
                                     end
@@ -484,7 +484,7 @@ begin
                 end
                 else begin
                     if (obj_is_carrying_obj_pid(self_obj, 41) < restock_amt) then begin
-                        restock_obj := create_object_sid(41, 0, 0, -1);
+                        restock_obj := create_object(41, 0, 0);
                         add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 41));
                     end
                     else begin
@@ -507,7 +507,7 @@ begin
                                     end
                                     else begin
                                         if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                            metarule(43, self_obj);
+                                            inven_unwield(self_obj);
                                         end
                                     end
                                 end
@@ -526,7 +526,7 @@ begin
                 if (obj_is_carrying_obj_pid(self_obj, 234) < restock_amt) then begin
                     if (80 >= random(1, 100)) then begin
                         if (obj_is_carrying_obj_pid(self_obj, 234) < restock_amt) then begin
-                            restock_obj := create_object_sid(234, 0, 0, -1);
+                            restock_obj := create_object(234, 0, 0);
                             add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 234));
                         end
                         else begin
@@ -549,7 +549,7 @@ begin
                                         end
                                         else begin
                                             if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                                metarule(43, self_obj);
+                                                inven_unwield(self_obj);
                                             end
                                         end
                                     end
@@ -567,7 +567,7 @@ begin
                 end
                 else begin
                     if (obj_is_carrying_obj_pid(self_obj, 234) < restock_amt) then begin
-                        restock_obj := create_object_sid(234, 0, 0, -1);
+                        restock_obj := create_object(234, 0, 0);
                         add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 234));
                     end
                     else begin
@@ -590,7 +590,7 @@ begin
                                     end
                                     else begin
                                         if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                            metarule(43, self_obj);
+                                            inven_unwield(self_obj);
                                         end
                                     end
                                 end
@@ -609,7 +609,7 @@ begin
                 if (obj_is_carrying_obj_pid(self_obj, 407) < restock_amt) then begin
                     if (100 >= random(1, 100)) then begin
                         if (obj_is_carrying_obj_pid(self_obj, 407) < restock_amt) then begin
-                            restock_obj := create_object_sid(407, 0, 0, -1);
+                            restock_obj := create_object(407, 0, 0);
                             add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 407));
                         end
                         else begin
@@ -632,7 +632,7 @@ begin
                                         end
                                         else begin
                                             if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                                metarule(43, self_obj);
+                                                inven_unwield(self_obj);
                                             end
                                         end
                                     end
@@ -650,7 +650,7 @@ begin
                 end
                 else begin
                     if (obj_is_carrying_obj_pid(self_obj, 407) < restock_amt) then begin
-                        restock_obj := create_object_sid(407, 0, 0, -1);
+                        restock_obj := create_object(407, 0, 0);
                         add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 407));
                     end
                     else begin
@@ -673,7 +673,7 @@ begin
                                     end
                                     else begin
                                         if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                            metarule(43, self_obj);
+                                            inven_unwield(self_obj);
                                         end
                                     end
                                 end
@@ -692,7 +692,7 @@ begin
                 if (obj_is_carrying_obj_pid(self_obj, 20) < restock_amt) then begin
                     if (100 >= random(1, 100)) then begin
                         if (obj_is_carrying_obj_pid(self_obj, 20) < restock_amt) then begin
-                            restock_obj := create_object_sid(20, 0, 0, -1);
+                            restock_obj := create_object(20, 0, 0);
                             add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 20));
                         end
                         else begin
@@ -715,7 +715,7 @@ begin
                                         end
                                         else begin
                                             if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                                metarule(43, self_obj);
+                                                inven_unwield(self_obj);
                                             end
                                         end
                                     end
@@ -733,7 +733,7 @@ begin
                 end
                 else begin
                     if (obj_is_carrying_obj_pid(self_obj, 20) < restock_amt) then begin
-                        restock_obj := create_object_sid(20, 0, 0, -1);
+                        restock_obj := create_object(20, 0, 0);
                         add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 20));
                     end
                     else begin
@@ -756,7 +756,7 @@ begin
                                     end
                                     else begin
                                         if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                            metarule(43, self_obj);
+                                            inven_unwield(self_obj);
                                         end
                                     end
                                 end
@@ -775,7 +775,7 @@ begin
                 if (obj_is_carrying_obj_pid(self_obj, 383) < restock_amt) then begin
                     if (80 >= random(1, 100)) then begin
                         if (obj_is_carrying_obj_pid(self_obj, 383) < restock_amt) then begin
-                            restock_obj := create_object_sid(383, 0, 0, -1);
+                            restock_obj := create_object(383, 0, 0);
                             add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 383));
                         end
                         else begin
@@ -798,7 +798,7 @@ begin
                                         end
                                         else begin
                                             if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                                metarule(43, self_obj);
+                                                inven_unwield(self_obj);
                                             end
                                         end
                                     end
@@ -816,7 +816,7 @@ begin
                 end
                 else begin
                     if (obj_is_carrying_obj_pid(self_obj, 383) < restock_amt) then begin
-                        restock_obj := create_object_sid(383, 0, 0, -1);
+                        restock_obj := create_object(383, 0, 0);
                         add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 383));
                     end
                     else begin
@@ -839,7 +839,7 @@ begin
                                     end
                                     else begin
                                         if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                            metarule(43, self_obj);
+                                            inven_unwield(self_obj);
                                         end
                                     end
                                 end
@@ -858,7 +858,7 @@ begin
                 if (obj_is_carrying_obj_pid(self_obj, 384) < restock_amt) then begin
                     if (50 >= random(1, 100)) then begin
                         if (obj_is_carrying_obj_pid(self_obj, 384) < restock_amt) then begin
-                            restock_obj := create_object_sid(384, 0, 0, -1);
+                            restock_obj := create_object(384, 0, 0);
                             add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 384));
                         end
                         else begin
@@ -881,7 +881,7 @@ begin
                                         end
                                         else begin
                                             if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                                metarule(43, self_obj);
+                                                inven_unwield(self_obj);
                                             end
                                         end
                                     end
@@ -899,7 +899,7 @@ begin
                 end
                 else begin
                     if (obj_is_carrying_obj_pid(self_obj, 384) < restock_amt) then begin
-                        restock_obj := create_object_sid(384, 0, 0, -1);
+                        restock_obj := create_object(384, 0, 0);
                         add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 384));
                     end
                     else begin
@@ -922,7 +922,7 @@ begin
                                     end
                                     else begin
                                         if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                            metarule(43, self_obj);
+                                            inven_unwield(self_obj);
                                         end
                                     end
                                 end
@@ -941,7 +941,7 @@ begin
                 if (obj_is_carrying_obj_pid(self_obj, 40) < restock_amt) then begin
                     if (80 >= random(1, 100)) then begin
                         if (obj_is_carrying_obj_pid(self_obj, 40) < restock_amt) then begin
-                            restock_obj := create_object_sid(40, 0, 0, -1);
+                            restock_obj := create_object(40, 0, 0);
                             add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 40));
                         end
                         else begin
@@ -964,7 +964,7 @@ begin
                                         end
                                         else begin
                                             if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                                metarule(43, self_obj);
+                                                inven_unwield(self_obj);
                                             end
                                         end
                                     end
@@ -982,7 +982,7 @@ begin
                 end
                 else begin
                     if (obj_is_carrying_obj_pid(self_obj, 40) < restock_amt) then begin
-                        restock_obj := create_object_sid(40, 0, 0, -1);
+                        restock_obj := create_object(40, 0, 0);
                         add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 40));
                     end
                     else begin
@@ -1005,7 +1005,7 @@ begin
                                     end
                                     else begin
                                         if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                            metarule(43, self_obj);
+                                            inven_unwield(self_obj);
                                         end
                                     end
                                 end
@@ -1024,7 +1024,7 @@ begin
                 if (obj_is_carrying_obj_pid(self_obj, 47) < restock_amt) then begin
                     if (60 >= random(1, 100)) then begin
                         if (obj_is_carrying_obj_pid(self_obj, 47) < restock_amt) then begin
-                            restock_obj := create_object_sid(47, 0, 0, -1);
+                            restock_obj := create_object(47, 0, 0);
                             add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 47));
                         end
                         else begin
@@ -1047,7 +1047,7 @@ begin
                                         end
                                         else begin
                                             if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                                metarule(43, self_obj);
+                                                inven_unwield(self_obj);
                                             end
                                         end
                                     end
@@ -1065,7 +1065,7 @@ begin
                 end
                 else begin
                     if (obj_is_carrying_obj_pid(self_obj, 47) < restock_amt) then begin
-                        restock_obj := create_object_sid(47, 0, 0, -1);
+                        restock_obj := create_object(47, 0, 0);
                         add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 47));
                     end
                     else begin
@@ -1088,7 +1088,7 @@ begin
                                     end
                                     else begin
                                         if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                            metarule(43, self_obj);
+                                            inven_unwield(self_obj);
                                         end
                                     end
                                 end
@@ -1107,7 +1107,7 @@ begin
                 if (obj_is_carrying_obj_pid(self_obj, 48) < restock_amt) then begin
                     if (60 >= random(1, 100)) then begin
                         if (obj_is_carrying_obj_pid(self_obj, 48) < restock_amt) then begin
-                            restock_obj := create_object_sid(48, 0, 0, -1);
+                            restock_obj := create_object(48, 0, 0);
                             add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 48));
                         end
                         else begin
@@ -1130,7 +1130,7 @@ begin
                                         end
                                         else begin
                                             if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                                metarule(43, self_obj);
+                                                inven_unwield(self_obj);
                                             end
                                         end
                                     end
@@ -1148,7 +1148,7 @@ begin
                 end
                 else begin
                     if (obj_is_carrying_obj_pid(self_obj, 48) < restock_amt) then begin
-                        restock_obj := create_object_sid(48, 0, 0, -1);
+                        restock_obj := create_object(48, 0, 0);
                         add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 48));
                     end
                     else begin
@@ -1171,7 +1171,7 @@ begin
                                     end
                                     else begin
                                         if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                            metarule(43, self_obj);
+                                            inven_unwield(self_obj);
                                         end
                                     end
                                 end
@@ -1190,7 +1190,7 @@ begin
                 if (obj_is_carrying_obj_pid(self_obj, 53) < restock_amt) then begin
                     if (50 >= random(1, 100)) then begin
                         if (obj_is_carrying_obj_pid(self_obj, 53) < restock_amt) then begin
-                            restock_obj := create_object_sid(53, 0, 0, -1);
+                            restock_obj := create_object(53, 0, 0);
                             add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 53));
                         end
                         else begin
@@ -1213,7 +1213,7 @@ begin
                                         end
                                         else begin
                                             if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                                metarule(43, self_obj);
+                                                inven_unwield(self_obj);
                                             end
                                         end
                                     end
@@ -1231,7 +1231,7 @@ begin
                 end
                 else begin
                     if (obj_is_carrying_obj_pid(self_obj, 53) < restock_amt) then begin
-                        restock_obj := create_object_sid(53, 0, 0, -1);
+                        restock_obj := create_object(53, 0, 0);
                         add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 53));
                     end
                     else begin
@@ -1254,7 +1254,7 @@ begin
                                     end
                                     else begin
                                         if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                            metarule(43, self_obj);
+                                            inven_unwield(self_obj);
                                         end
                                     end
                                 end
@@ -1273,7 +1273,7 @@ begin
                 if (obj_is_carrying_obj_pid(self_obj, 71) < restock_amt) then begin
                     if (60 >= random(1, 100)) then begin
                         if (obj_is_carrying_obj_pid(self_obj, 71) < restock_amt) then begin
-                            restock_obj := create_object_sid(71, 0, 0, -1);
+                            restock_obj := create_object(71, 0, 0);
                             add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 71));
                         end
                         else begin
@@ -1296,7 +1296,7 @@ begin
                                         end
                                         else begin
                                             if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                                metarule(43, self_obj);
+                                                inven_unwield(self_obj);
                                             end
                                         end
                                     end
@@ -1314,7 +1314,7 @@ begin
                 end
                 else begin
                     if (obj_is_carrying_obj_pid(self_obj, 71) < restock_amt) then begin
-                        restock_obj := create_object_sid(71, 0, 0, -1);
+                        restock_obj := create_object(71, 0, 0);
                         add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 71));
                     end
                     else begin
@@ -1337,7 +1337,7 @@ begin
                                     end
                                     else begin
                                         if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                            metarule(43, self_obj);
+                                            inven_unwield(self_obj);
                                         end
                                     end
                                 end
@@ -1356,7 +1356,7 @@ begin
                 if (obj_is_carrying_obj_pid(self_obj, 87) < restock_amt) then begin
                     if (50 >= random(1, 100)) then begin
                         if (obj_is_carrying_obj_pid(self_obj, 87) < restock_amt) then begin
-                            restock_obj := create_object_sid(87, 0, 0, -1);
+                            restock_obj := create_object(87, 0, 0);
                             add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 87));
                         end
                         else begin
@@ -1379,7 +1379,7 @@ begin
                                         end
                                         else begin
                                             if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                                metarule(43, self_obj);
+                                                inven_unwield(self_obj);
                                             end
                                         end
                                     end
@@ -1397,7 +1397,7 @@ begin
                 end
                 else begin
                     if (obj_is_carrying_obj_pid(self_obj, 87) < restock_amt) then begin
-                        restock_obj := create_object_sid(87, 0, 0, -1);
+                        restock_obj := create_object(87, 0, 0);
                         add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 87));
                     end
                     else begin
@@ -1420,7 +1420,7 @@ begin
                                     end
                                     else begin
                                         if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                            metarule(43, self_obj);
+                                            inven_unwield(self_obj);
                                         end
                                     end
                                 end
@@ -1439,7 +1439,7 @@ begin
                 if (obj_is_carrying_obj_pid(self_obj, 110) < restock_amt) then begin
                     if (50 >= random(1, 100)) then begin
                         if (obj_is_carrying_obj_pid(self_obj, 110) < restock_amt) then begin
-                            restock_obj := create_object_sid(110, 0, 0, -1);
+                            restock_obj := create_object(110, 0, 0);
                             add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 110));
                         end
                         else begin
@@ -1462,7 +1462,7 @@ begin
                                         end
                                         else begin
                                             if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                                metarule(43, self_obj);
+                                                inven_unwield(self_obj);
                                             end
                                         end
                                     end
@@ -1480,7 +1480,7 @@ begin
                 end
                 else begin
                     if (obj_is_carrying_obj_pid(self_obj, 110) < restock_amt) then begin
-                        restock_obj := create_object_sid(110, 0, 0, -1);
+                        restock_obj := create_object(110, 0, 0);
                         add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 110));
                     end
                     else begin
@@ -1503,7 +1503,7 @@ begin
                                     end
                                     else begin
                                         if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                            metarule(43, self_obj);
+                                            inven_unwield(self_obj);
                                         end
                                     end
                                 end
@@ -1522,7 +1522,7 @@ begin
                 if (obj_is_carrying_obj_pid(self_obj, 144) < restock_amt) then begin
                     if (30 >= random(1, 100)) then begin
                         if (obj_is_carrying_obj_pid(self_obj, 144) < restock_amt) then begin
-                            restock_obj := create_object_sid(144, 0, 0, -1);
+                            restock_obj := create_object(144, 0, 0);
                             add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 144));
                         end
                         else begin
@@ -1545,7 +1545,7 @@ begin
                                         end
                                         else begin
                                             if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                                metarule(43, self_obj);
+                                                inven_unwield(self_obj);
                                             end
                                         end
                                     end
@@ -1563,7 +1563,7 @@ begin
                 end
                 else begin
                     if (obj_is_carrying_obj_pid(self_obj, 144) < restock_amt) then begin
-                        restock_obj := create_object_sid(144, 0, 0, -1);
+                        restock_obj := create_object(144, 0, 0);
                         add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 144));
                     end
                     else begin
@@ -1586,7 +1586,7 @@ begin
                                     end
                                     else begin
                                         if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                            metarule(43, self_obj);
+                                            inven_unwield(self_obj);
                                         end
                                     end
                                 end
@@ -1605,7 +1605,7 @@ begin
                 if (obj_is_carrying_obj_pid(self_obj, 260) < restock_amt) then begin
                     if (30 >= random(1, 100)) then begin
                         if (obj_is_carrying_obj_pid(self_obj, 260) < restock_amt) then begin
-                            restock_obj := create_object_sid(260, 0, 0, -1);
+                            restock_obj := create_object(260, 0, 0);
                             add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 260));
                         end
                         else begin
@@ -1628,7 +1628,7 @@ begin
                                         end
                                         else begin
                                             if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                                metarule(43, self_obj);
+                                                inven_unwield(self_obj);
                                             end
                                         end
                                     end
@@ -1646,7 +1646,7 @@ begin
                 end
                 else begin
                     if (obj_is_carrying_obj_pid(self_obj, 260) < restock_amt) then begin
-                        restock_obj := create_object_sid(260, 0, 0, -1);
+                        restock_obj := create_object(260, 0, 0);
                         add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 260));
                     end
                     else begin
@@ -1669,7 +1669,7 @@ begin
                                     end
                                     else begin
                                         if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                            metarule(43, self_obj);
+                                            inven_unwield(self_obj);
                                         end
                                     end
                                 end
@@ -1688,7 +1688,7 @@ begin
                 if (obj_is_carrying_obj_pid(self_obj, 284) < restock_amt) then begin
                     if (45 >= random(1, 100)) then begin
                         if (obj_is_carrying_obj_pid(self_obj, 284) < restock_amt) then begin
-                            restock_obj := create_object_sid(284, 0, 0, -1);
+                            restock_obj := create_object(284, 0, 0);
                             add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 284));
                         end
                         else begin
@@ -1711,7 +1711,7 @@ begin
                                         end
                                         else begin
                                             if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                                metarule(43, self_obj);
+                                                inven_unwield(self_obj);
                                             end
                                         end
                                     end
@@ -1729,7 +1729,7 @@ begin
                 end
                 else begin
                     if (obj_is_carrying_obj_pid(self_obj, 284) < restock_amt) then begin
-                        restock_obj := create_object_sid(284, 0, 0, -1);
+                        restock_obj := create_object(284, 0, 0);
                         add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 284));
                     end
                     else begin
@@ -1752,7 +1752,7 @@ begin
                                     end
                                     else begin
                                         if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                            metarule(43, self_obj);
+                                            inven_unwield(self_obj);
                                         end
                                     end
                                 end
@@ -1790,4 +1790,3 @@ end
 procedure timed_event_p_proc
 begin
 end
-
