diff --git a/Mods/MegaMod/Original/FRNKLEGS.ssl b/Mods/MegaMod/ReDefine/FRNKLEGS.ssl
index b136858..2680731 100644
--- a/Mods/MegaMod/Original/FRNKLEGS.ssl
+++ b/Mods/MegaMod/ReDefine/FRNKLEGS.ssl
@@ -144,8 +144,8 @@ end
 
 procedure critter_p_proc
 begin
-    if ((global_var(499) == 1) and (local_var(3) == 0)) then begin
-        item := create_object_sid(554, 0, 0, -1);
+    if ((global_var(GVAR_ENCLAVE_FRANK_DEAD) == 1) and (local_var(3) == 0)) then begin
+        item := create_object(554, 0, 0);
         add_obj_to_inven(self_obj, item);
         set_local_var(3, 1);
     end
@@ -158,13 +158,13 @@ end
 procedure look_at_p_proc
 begin
     script_overrides;
-    display_msg(message_str(1603, 100));
+    display_mstr(100);
 end
 
 procedure description_p_proc
 begin
     script_overrides;
-    display_msg(message_str(1603, 101));
+    display_mstr(101);
 end
 
 procedure use_skill_on_p_proc
@@ -178,4 +178,3 @@ end
 procedure damage_p_proc
 begin
 end
-
