diff --git a/Mods/MegaMod/Original/FSBRODOR.ssl b/Mods/MegaMod/ReDefine/FSBRODOR.ssl
index ae0fd23..37a0796 100644
--- a/Mods/MegaMod/Original/FSBRODOR.ssl
+++ b/Mods/MegaMod/ReDefine/FSBRODOR.ssl
@@ -130,10 +130,10 @@ end
 
 procedure use_p_proc
 begin
-    if (not(global_var(361) bwand 8388608)) then begin
+    if (not(global_var(GVAR_SAN_FRAN_FLAGS) bwand 8388608)) then begin
         script_overrides;
         if (source_obj == dude_obj) then begin
-            display_msg(message_str(1159, 100));
+            display_mstr(100);
         end
         if (validHandle(i_brother_obj)) then begin
             add_timer_event(i_brother_obj, game_ticks(0), 10);
@@ -191,7 +191,7 @@ end
 
 procedure map_enter_p_proc
 begin
-    if (not(metarule(22, 0))) then begin
+    if (not(is_loading_game)) then begin
         add_timer_event(self_obj, game_ticks(10), 1);
     end
 end
@@ -232,4 +232,3 @@ begin
         add_timer_event(self_obj, game_ticks(10), 1);
     end
 end
-
