diff --git a/Mods/MegaMod/Original/FSCHESTA.ssl b/Mods/MegaMod/ReDefine/FSCHESTA.ssl
index 5712891..6f906ad 100644
--- a/Mods/MegaMod/Original/FSCHESTA.ssl
+++ b/Mods/MegaMod/ReDefine/FSCHESTA.ssl
@@ -175,7 +175,7 @@ begin
         end_dialogue;
     end
     else begin
-        if (not((global_var(361) bwand 8192) != 0)) then begin
+        if (not((global_var(GVAR_SAN_FRAN_FLAGS) bwand 8192) != 0)) then begin
             start_gdialog(924, self_obj, 4, -1, -1);
             gsay_start;
             call Node009();
@@ -201,17 +201,17 @@ procedure look_at_p_proc
 begin
     script_overrides;
     if (local_var(4) == 0) then begin
-        display_msg(message_str(924, 100));
+        display_mstr(100);
     end
     else begin
-        display_msg(message_str(924, 101));
+        display_mstr(101);
     end
 end
 
 procedure description_p_proc
 begin
     script_overrides;
-    display_msg(message_str(924, 102));
+    display_mstr(102);
 end
 
 procedure use_skill_on_p_proc
@@ -221,7 +221,7 @@ end
 
 procedure use_obj_on_p_proc
 begin
-    display_msg(message_str(924, 135));
+    display_mstr(135);
 end
 
 procedure damage_p_proc
@@ -236,11 +236,11 @@ end
 procedure use_p_proc
 begin
     script_overrides;
-    if (not((global_var(361) bwand 1024) != 0)) then begin
+    if (not((global_var(GVAR_SAN_FRAN_FLAGS) bwand 1024) != 0)) then begin
         dialogue_system_enter;
     end
     else begin
-        display_msg(message_str(924, 134));
+        display_mstr(134);
     end
 end
 
@@ -254,100 +254,100 @@ end
 
 procedure Node001
 begin
-    set_global_var(361, global_var(361) bwor 8192);
-    gsay_reply(924, 103);
-    giq_option(-3, 924, 104, Node999, 50);
-    giq_option(4, 924, 136, NodeGuess, 50);
+    set_global_var(GVAR_SAN_FRAN_FLAGS, global_var(GVAR_SAN_FRAN_FLAGS) bwor 8192);
+    Reply(103);
+    giq_option(-3, NAME, 104, Node999, NEUTRAL_REACTION);
+    giq_option(4, NAME, 136, NodeGuess, NEUTRAL_REACTION);
     if (obj_carrying_pid_obj(dude_obj, 453)) then begin
-        giq_option(4, 924, 105, Node002, 50);
+        giq_option(4, NAME, 105, Node002, NEUTRAL_REACTION);
     end
-    giq_option(4, 924, 106, NodeHack, 50);
-    giq_option(4, 924, 107, Node999, 50);
+    giq_option(4, NAME, 106, NodeHack, NEUTRAL_REACTION);
+    giq_option(4, NAME, 107, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node002
 begin
-    gsay_reply(924, 108);
-    giq_option(1, 924, 109, Node005, 50);
-    giq_option(1, 924, 110, Node006, 50);
-    giq_option(1, 924, 111, Node009a, 50);
-    giq_option(1, 924, 112, Node999, 50);
+    Reply(108);
+    giq_option(1, NAME, 109, Node005, NEUTRAL_REACTION);
+    giq_option(1, NAME, 110, Node006, NEUTRAL_REACTION);
+    giq_option(1, NAME, 111, Node009a, NEUTRAL_REACTION);
+    giq_option(1, NAME, 112, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node003
 begin
-    gsay_reply(924, 113);
-    giq_option(1, 924, 114, NodeHack, 50);
-    giq_option(1, 924, 115, Node999, 50);
+    Reply(113);
+    giq_option(1, NAME, 114, NodeHack, NEUTRAL_REACTION);
+    giq_option(1, NAME, 115, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node004
 begin
     set_local_var(7, 1);
-    set_global_var(364, 1);
-    gsay_reply(924, 116);
-    giq_option(4, 924, 117, Node999, 50);
+    set_global_var(GVAR_SAN_FRAN_SHIHACKED, 1);
+    Reply(116);
+    giq_option(4, NAME, 117, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node005
 begin
     if (is_success(roll_vs_skill(dude_obj, 12, -100))) then begin
         game_time_advance(60 * (60 * 10));
-        gsay_reply(924, 118);
+        Reply(118);
     end
     else begin
         game_time_advance(60 * 10 * 15);
-        gsay_reply(924, 119);
+        Reply(119);
     end
-    if (not((global_var(363) bwand 8) != 0) and not((global_var(361) bwand 32768) != 0)) then begin
-        if ((global_var(363) bwand 1) != 0) then begin
-            giq_option(4, 924, 120, Node0708, 50);
+    if (not((global_var(GVAR_SAN_FRAN_TANKER) bwand 8) != 0) and not((global_var(GVAR_SAN_FRAN_FLAGS) bwand 32768) != 0)) then begin
+        if ((global_var(GVAR_SAN_FRAN_TANKER) bwand 1) != 0) then begin
+            giq_option(4, NAME, 120, Node0708, NEUTRAL_REACTION);
         end
-        if ((global_var(361) bwand 65536) != 0) then begin
-            giq_option(4, 924, 121, Node007a, 50);
+        if ((global_var(GVAR_SAN_FRAN_FLAGS) bwand 65536) != 0) then begin
+            giq_option(4, NAME, 121, Node007a, NEUTRAL_REACTION);
         end
     end
-    if (not(((global_var(363) bwand 1) != 0) and not((global_var(363) bwand 8) != 0)) and not(((global_var(361) bwand 65536) != 0) and not((global_var(361) bwand 32768) != 0))) then begin
-        giq_option(4, 924, 137, Node011, 50);
+    if (not(((global_var(GVAR_SAN_FRAN_TANKER) bwand 1) != 0) and not((global_var(GVAR_SAN_FRAN_TANKER) bwand 8) != 0)) and not(((global_var(GVAR_SAN_FRAN_FLAGS) bwand 65536) != 0) and not((global_var(GVAR_SAN_FRAN_FLAGS) bwand 32768) != 0))) then begin
+        giq_option(4, NAME, 137, Node011, NEUTRAL_REACTION);
     end
-    giq_option(4, 924, 122, Node002, 50);
+    giq_option(4, NAME, 122, Node002, NEUTRAL_REACTION);
 end
 
 procedure Node006
 begin
-    gsay_reply(924, 123);
-    giq_option(1, 924, 124, Node002, 50);
+    Reply(123);
+    giq_option(1, NAME, 124, Node002, NEUTRAL_REACTION);
 end
 
 procedure Node007
 begin
-    gsay_reply(924, 125);
-    giq_option(1, 924, 126, Node999a, 50);
+    Reply(125);
+    giq_option(1, NAME, 126, Node999a, NEUTRAL_REACTION);
 end
 
 procedure Node008
 begin
-    gsay_reply(924, 127);
-    giq_option(1, 924, 128, Node999, 50);
+    Reply(127);
+    giq_option(1, NAME, 128, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node009
 begin
-    gsay_reply(924, 129);
-    giq_option(1, 924, 130, Node001, 50);
-    giq_option(1, 924, 131, Node999, 50);
+    Reply(129);
+    giq_option(1, NAME, 130, Node001, NEUTRAL_REACTION);
+    giq_option(1, NAME, 131, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node010
 begin
-    gsay_reply(924, 132);
-    giq_option(1, 924, 133, Node002, 50);
+    Reply(132);
+    giq_option(1, NAME, 133, Node002, NEUTRAL_REACTION);
 end
 
 procedure Node011
 begin
-    gsay_reply(924, 138);
-    giq_option(1, 924, 124, Node002, 50);
+    Reply(138);
+    giq_option(1, NAME, 124, Node002, NEUTRAL_REACTION);
 end
 
 procedure NodeHack
@@ -357,7 +357,7 @@ begin
     end
     else begin
         if (is_success(roll_vs_skill(dude_obj, 12, 100))) then begin
-            set_global_var(364, 1);
+            set_global_var(GVAR_SAN_FRAN_SHIHACKED, 1);
             call Node002();
         end
         else begin
@@ -378,65 +378,65 @@ end
 
 procedure Node009a
 begin
-    set_global_var(361, global_var(361) bwand (-1 - 8192));
+    set_global_var(GVAR_SAN_FRAN_FLAGS, global_var(GVAR_SAN_FRAN_FLAGS) bwand (-1 - 8192));
     call Node009();
 end
 
 procedure Node0708
 begin
-    set_global_var(363, global_var(363) bwor 8);
-    if (((global_var(363) bwand 8) != 0) and ((global_var(363) bwand 4) != 0)) then begin
-        if (global_var(494) < 7) then begin
-            set_global_var(494, 7);
+    set_global_var(GVAR_SAN_FRAN_TANKER, global_var(GVAR_SAN_FRAN_TANKER) bwor 8);
+    if (((global_var(GVAR_SAN_FRAN_TANKER) bwand 8) != 0) and ((global_var(GVAR_SAN_FRAN_TANKER) bwand 4) != 0)) then begin
+        if (global_var(GVAR_FALLOUT_2) < 7) then begin
+            set_global_var(GVAR_FALLOUT_2, 7);
         end
     end
     if (924 != 950) then begin
         if (5 != 0) then begin
-            set_global_var(0, global_var(0) + 5);
+            set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + 5);
         end
         debug_msg("Player gains " + 5 + " Karma Points.");
-        set_global_var(37, 0);
-        set_global_var(38, 0);
-        set_global_var(39, 0);
-        set_global_var(40, 0);
-        set_global_var(41, 0);
-        set_global_var(42, 0);
-        set_global_var(43, 0);
-        set_global_var(44, 0);
-        set_global_var(45, 0);
-        if (global_var(0) >= 1000) then begin
-            set_global_var(37, 1);
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+        set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+        set_global_var(GVAR_KARMA_DEFENDER, 0);
+        set_global_var(GVAR_KARMA_WANDERER, 0);
+        set_global_var(GVAR_KARMA_BETRAYER, 0);
+        set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+        set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+            set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
         end
         else begin
-            if (global_var(0) >= 750) then begin
-                set_global_var(38, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
             end
             else begin
-                if (global_var(0) >= 500) then begin
-                    set_global_var(39, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                 end
                 else begin
-                    if (global_var(0) >= 250) then begin
-                        set_global_var(40, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                        set_global_var(GVAR_KARMA_DEFENDER, 1);
                     end
                     else begin
-                        if (global_var(0) > -250) then begin
-                            set_global_var(41, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                            set_global_var(GVAR_KARMA_WANDERER, 1);
                         end
                         else begin
-                            if (global_var(0) > -500) then begin
-                                set_global_var(42, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                set_global_var(GVAR_KARMA_BETRAYER, 1);
                             end
                             else begin
-                                if (global_var(0) > -750) then begin
-                                    set_global_var(43, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -1000) then begin
-                                        set_global_var(44, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                     end
                                     else begin
-                                        set_global_var(45, 1);
+                                        set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                     end
                                 end
                             end
@@ -446,15 +446,15 @@ begin
             end
         end
         if (0 != 0) then begin
-            set_global_var(61, global_var(61) + 0);
+            set_global_var(GVAR_TOWN_REP_SAN_FRANCISCO, global_var(GVAR_TOWN_REP_SAN_FRANCISCO) + 0);
         end
         if (5000 != 0) then begin
             give_exp_points(5000);
         end
-        display_msg(message_str(14, 100) + 5000 + message_str(14, 101));
+        display_msg(g_mstr(100) + 5000 + g_mstr(101));
     end
-    set_global_var(532, 2);
-    if ((global_var(361) bwand 2048) != 0) then begin
+    set_global_var(GVAR_SAN_FRAN_FUEL_TANK_QST, 2);
+    if ((global_var(GVAR_SAN_FRAN_FLAGS) bwand 2048) != 0) then begin
         call Node007();
     end
     else begin
@@ -464,50 +464,50 @@ end
 
 procedure Node007a
 begin
-    set_global_var(0, global_var(0) + -25);
+    set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + -25);
     debug_msg("Player gains " + -25 + " Karma Points.");
-    set_global_var(37, 0);
-    set_global_var(38, 0);
-    set_global_var(39, 0);
-    set_global_var(40, 0);
-    set_global_var(41, 0);
-    set_global_var(42, 0);
-    set_global_var(43, 0);
-    set_global_var(44, 0);
-    set_global_var(45, 0);
-    if (global_var(0) >= 1000) then begin
-        set_global_var(37, 1);
+    set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+    set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+    set_global_var(GVAR_KARMA_DEFENDER, 0);
+    set_global_var(GVAR_KARMA_WANDERER, 0);
+    set_global_var(GVAR_KARMA_BETRAYER, 0);
+    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+    set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+    set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+    if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
     end
     else begin
-        if (global_var(0) >= 750) then begin
-            set_global_var(38, 1);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+            set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
         end
         else begin
-            if (global_var(0) >= 500) then begin
-                set_global_var(39, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
             end
             else begin
-                if (global_var(0) >= 250) then begin
-                    set_global_var(40, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                    set_global_var(GVAR_KARMA_DEFENDER, 1);
                 end
                 else begin
-                    if (global_var(0) > -250) then begin
-                        set_global_var(41, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                        set_global_var(GVAR_KARMA_WANDERER, 1);
                     end
                     else begin
-                        if (global_var(0) > -500) then begin
-                            set_global_var(42, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                            set_global_var(GVAR_KARMA_BETRAYER, 1);
                         end
                         else begin
-                            if (global_var(0) > -750) then begin
-                                set_global_var(43, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                             end
                             else begin
-                                if (global_var(0) > -1000) then begin
-                                    set_global_var(44, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                    set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                 end
                                 else begin
-                                    set_global_var(45, 1);
+                                    set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                 end
                             end
                         end
@@ -517,16 +517,15 @@ begin
         end
     end
     give_exp_points(6000);
-    display_msg(message_str(14, 100) + 6000 + message_str(14, 101));
-    set_global_var(61, global_var(61) + -5);
-    set_global_var(361, global_var(361) bwor 32768);
-    set_global_var(535, 2);
+    display_msg(g_mstr(100) + 6000 + g_mstr(101));
+    set_global_var(GVAR_TOWN_REP_SAN_FRANCISCO, global_var(GVAR_TOWN_REP_SAN_FRANCISCO) + -5);
+    set_global_var(GVAR_SAN_FRAN_FLAGS, global_var(GVAR_SAN_FRAN_FLAGS) bwor 32768);
+    set_global_var(GVAR_SAN_FRAN_ELRON_GAS_QST, 2);
     call Node007();
 end
 
 procedure Node999a
 begin
-    set_global_var(364, 1);
+    set_global_var(GVAR_SAN_FRAN_SHIHACKED, 1);
     call Node999();
 end
-
