diff --git a/Mods/MegaMod/Original/FSFACE.ssl b/Mods/MegaMod/ReDefine/FSFACE.ssl
index cfb5718..9ef2e92 100644
--- a/Mods/MegaMod/Original/FSFACE.ssl
+++ b/Mods/MegaMod/ReDefine/FSFACE.ssl
@@ -247,281 +247,281 @@ end
 
 procedure Node001
 begin
-    gsay_reply(1162, 100);
-    giq_option(1, 1162, 101, Node002, 50);
-    giq_option(1, 1162, 102, Node999, 50);
-    giq_option(-3, 1162, 103, Node016, 50);
-    giq_option(4, 1162, 104, Node016, 50);
-    if (((global_var(582) bwand 4) != 0) and not((global_var(582) bwand 262144) != 0)) then begin
-        giq_option(4, 1162, 223, Node024, 50);
+    Reply(100);
+    giq_option(1, NAME, 101, Node002, NEUTRAL_REACTION);
+    giq_option(1, NAME, 102, Node999, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 103, Node016, NEUTRAL_REACTION);
+    giq_option(4, NAME, 104, Node016, NEUTRAL_REACTION);
+    if (((global_var(GVAR_VIGNETTE_SEQUENCE) bwand 4) != 0) and not((global_var(GVAR_VIGNETTE_SEQUENCE) bwand 262144) != 0)) then begin
+        giq_option(4, NAME, 223, Node024, NEUTRAL_REACTION);
     end
 end
 
 procedure Node002
 begin
-    gsay_reply(1162, 105);
-    if (get_critter_stat(dude_obj, 35) != get_critter_stat(dude_obj, 7)) then begin
-        giq_option(-3, 1162, 106, Node003, 50);
-        giq_option(4, 1162, 115, Node003, 50);
+    Reply(105);
+    if (get_critter_stat(dude_obj, STAT_current_hp) != get_critter_stat(dude_obj, STAT_max_hp)) then begin
+        giq_option(-3, NAME, 106, Node003, NEUTRAL_REACTION);
+        giq_option(4, NAME, 115, Node003, NEUTRAL_REACTION);
     end
     if (get_poison(dude_obj)) then begin
-        giq_option(-3, 1162, 107, Node004, 50);
-        giq_option(4, 1162, 116, Node004, 50);
+        giq_option(-3, NAME, 107, Node004, NEUTRAL_REACTION);
+        giq_option(4, NAME, 116, Node004, NEUTRAL_REACTION);
     end
-    if (get_critter_stat(dude_obj, 37)) then begin
-        giq_option(-3, 1162, 108, Node005, 50);
-        giq_option(4, 1162, 117, Node005, 50);
+    if (get_critter_stat(dude_obj, STAT_current_rad)) then begin
+        giq_option(-3, NAME, 108, Node005, NEUTRAL_REACTION);
+        giq_option(4, NAME, 117, Node005, NEUTRAL_REACTION);
     end
     if ((critter_state(dude_obj) bwand 4) or (critter_state(dude_obj) bwand 8) or (critter_state(dude_obj) bwand 16) or (critter_state(dude_obj) bwand 32)) then begin
-        giq_option(-3, 1162, 109, Node006, 50);
-        giq_option(4, 1162, 118, Node006, 50);
+        giq_option(-3, NAME, 109, Node006, NEUTRAL_REACTION);
+        giq_option(4, NAME, 118, Node006, NEUTRAL_REACTION);
     end
-    if (not((local_var(1) bwand 1) != 0) and (get_critter_stat(dude_obj, 0) < 10)) then begin
-        giq_option(-3, 1162, 110, Node007, 50);
-        giq_option(4, 1162, 119, Node007, 50);
+    if (not((local_var(1) bwand 1) != 0) and (dude_strength < 10)) then begin
+        giq_option(-3, NAME, 110, Node007, NEUTRAL_REACTION);
+        giq_option(4, NAME, 119, Node007, NEUTRAL_REACTION);
     end
-    if (not((local_var(1) bwand 2) != 0) and (get_critter_stat(dude_obj, 4) < 10)) then begin
-        giq_option(-3, 1162, 111, Node009, 50);
-        giq_option(4, 1162, 120, Node009, 50);
+    if (not((local_var(1) bwand 2) != 0) and (dude_iq < 10)) then begin
+        giq_option(-3, NAME, 111, Node009, NEUTRAL_REACTION);
+        giq_option(4, NAME, 120, Node009, NEUTRAL_REACTION);
     end
-    if (not((local_var(1) bwand 4) != 0) and (get_critter_stat(dude_obj, 1) < 10)) then begin
-        giq_option(-3, 1162, 112, Node011, 50);
-        giq_option(4, 1162, 121, Node011, 50);
+    if (not((local_var(1) bwand 4) != 0) and (dude_perception < 10)) then begin
+        giq_option(-3, NAME, 112, Node011, NEUTRAL_REACTION);
+        giq_option(4, NAME, 121, Node011, NEUTRAL_REACTION);
     end
-    if (not((local_var(1) bwand 8) != 0) and (get_critter_stat(dude_obj, 3) < 10)) then begin
-        giq_option(-3, 1162, 113, Node013, 50);
-        giq_option(4, 1162, 122, Node013, 50);
+    if (not((local_var(1) bwand 8) != 0) and (dude_charisma < 10)) then begin
+        giq_option(-3, NAME, 113, Node013, NEUTRAL_REACTION);
+        giq_option(4, NAME, 122, Node013, NEUTRAL_REACTION);
     end
-    giq_option(-3, 1162, 114, Node999, 50);
-    giq_option(4, 1162, 123, Node999, 50);
+    giq_option(-3, NAME, 114, Node999, NEUTRAL_REACTION);
+    giq_option(4, NAME, 123, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node003
 begin
-    gsay_reply(1162, 124);
-    giq_option(-3, 1162, 125, NodeHeal15, 50);
-    giq_option(-3, 1162, 126, Node002, 50);
-    giq_option(4, 1162, 127, NodeHeal15, 50);
-    giq_option(4, 1162, 128, Node002, 50);
+    Reply(124);
+    giq_option(-3, NAME, 125, NodeHeal15, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 126, Node002, NEUTRAL_REACTION);
+    giq_option(4, NAME, 127, NodeHeal15, NEUTRAL_REACTION);
+    giq_option(4, NAME, 128, Node002, NEUTRAL_REACTION);
 end
 
 procedure Node004
 begin
-    gsay_reply(1162, 129);
-    giq_option(-3, 1162, 130, NodePois15, 50);
-    giq_option(-3, 1162, 131, Node002, 50);
-    giq_option(4, 1162, 132, NodePois15, 50);
-    giq_option(4, 1162, 133, Node002, 50);
+    Reply(129);
+    giq_option(-3, NAME, 130, NodePois15, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 131, Node002, NEUTRAL_REACTION);
+    giq_option(4, NAME, 132, NodePois15, NEUTRAL_REACTION);
+    giq_option(4, NAME, 133, Node002, NEUTRAL_REACTION);
 end
 
 procedure Node005
 begin
-    gsay_reply(1162, 134);
-    giq_option(-3, 1162, 135, NodeRad15, 50);
-    giq_option(-3, 1162, 136, Node002, 50);
-    giq_option(4, 1162, 137, NodeRad15, 50);
-    giq_option(4, 1162, 138, Node002, 50);
+    Reply(134);
+    giq_option(-3, NAME, 135, NodeRad15, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 136, Node002, NEUTRAL_REACTION);
+    giq_option(4, NAME, 137, NodeRad15, NEUTRAL_REACTION);
+    giq_option(4, NAME, 138, Node002, NEUTRAL_REACTION);
 end
 
 procedure Node006
 begin
-    gsay_reply(1162, 139);
-    giq_option(-3, 1162, 140, NodeCrip15, 50);
-    giq_option(-3, 1162, 141, Node002, 50);
-    giq_option(4, 1162, 142, NodeCrip15, 50);
-    giq_option(4, 1162, 143, Node002, 50);
+    Reply(139);
+    giq_option(-3, NAME, 140, NodeCrip15, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 141, Node002, NEUTRAL_REACTION);
+    giq_option(4, NAME, 142, NodeCrip15, NEUTRAL_REACTION);
+    giq_option(4, NAME, 143, Node002, NEUTRAL_REACTION);
 end
 
 procedure Node007
 begin
     if (not((local_var(2) bwand 1) != 0)) then begin
-        gsay_reply(1162, 144);
+        Reply(144);
         if (obj_carrying_pid_obj(dude_obj, 505)) then begin
-            giq_option(-3, 1162, 145, Node007s, 50);
-            giq_option(4, 1162, 147, Node007s, 50);
+            giq_option(-3, NAME, 145, Node007s, NEUTRAL_REACTION);
+            giq_option(4, NAME, 147, Node007s, NEUTRAL_REACTION);
         end
         else begin
-            giq_option(-3, 1162, 146, Node999, 50);
-            giq_option(4, 1162, 148, Node999, 50);
+            giq_option(-3, NAME, 146, Node999, NEUTRAL_REACTION);
+            giq_option(4, NAME, 148, Node999, NEUTRAL_REACTION);
         end
     end
     else begin
-        gsay_reply(1162, 149);
-        giq_option(-3, 1162, 150, Node008, 50);
-        giq_option(-3, 1162, 151, Node002, 50);
-        giq_option(4, 1162, 152, Node008, 50);
-        giq_option(4, 1162, 153, Node002, 50);
+        Reply(149);
+        giq_option(-3, NAME, 150, Node008, NEUTRAL_REACTION);
+        giq_option(-3, NAME, 151, Node002, NEUTRAL_REACTION);
+        giq_option(4, NAME, 152, Node008, NEUTRAL_REACTION);
+        giq_option(4, NAME, 153, Node002, NEUTRAL_REACTION);
     end
 end
 
 procedure Node008
 begin
-    gsay_reply(1162, 154);
-    giq_option(1, 1162, 155, NodeStr15, 50);
+    Reply(154);
+    giq_option(1, NAME, 155, NodeStr15, NEUTRAL_REACTION);
 end
 
 procedure Node009
 begin
     if (not((local_var(2) bwand 2) != 0)) then begin
-        gsay_reply(1162, 156);
+        Reply(156);
         if (obj_carrying_pid_obj(dude_obj, 506)) then begin
-            giq_option(-3, 1162, 157, Node009s, 50);
-            giq_option(4, 1162, 159, Node009s, 50);
+            giq_option(-3, NAME, 157, Node009s, NEUTRAL_REACTION);
+            giq_option(4, NAME, 159, Node009s, NEUTRAL_REACTION);
         end
         else begin
-            giq_option(-3, 1162, 158, Node999, 50);
-            giq_option(4, 1162, 160, Node999, 50);
+            giq_option(-3, NAME, 158, Node999, NEUTRAL_REACTION);
+            giq_option(4, NAME, 160, Node999, NEUTRAL_REACTION);
         end
     end
     else begin
-        gsay_reply(1162, 161);
-        giq_option(-3, 1162, 162, Node010, 50);
-        giq_option(-3, 1162, 163, Node002, 50);
-        giq_option(4, 1162, 164, Node010, 50);
-        giq_option(4, 1162, 165, Node002, 50);
+        Reply(161);
+        giq_option(-3, NAME, 162, Node010, NEUTRAL_REACTION);
+        giq_option(-3, NAME, 163, Node002, NEUTRAL_REACTION);
+        giq_option(4, NAME, 164, Node010, NEUTRAL_REACTION);
+        giq_option(4, NAME, 165, Node002, NEUTRAL_REACTION);
     end
 end
 
 procedure Node010
 begin
-    gsay_reply(1162, 166);
-    giq_option(1, 1162, 167, NodeInt15, 50);
+    Reply(166);
+    giq_option(1, NAME, 167, NodeInt15, NEUTRAL_REACTION);
 end
 
 procedure Node011
 begin
     if (not((local_var(2) bwand 4) != 0)) then begin
-        gsay_reply(1162, 168);
+        Reply(168);
         if (obj_carrying_pid_obj(dude_obj, 504)) then begin
-            giq_option(-3, 1162, 169, Node011s, 50);
-            giq_option(4, 1162, 171, Node011s, 50);
+            giq_option(-3, NAME, 169, Node011s, NEUTRAL_REACTION);
+            giq_option(4, NAME, 171, Node011s, NEUTRAL_REACTION);
         end
         else begin
-            giq_option(-3, 1162, 170, Node999, 50);
-            giq_option(4, 1162, 172, Node999, 50);
+            giq_option(-3, NAME, 170, Node999, NEUTRAL_REACTION);
+            giq_option(4, NAME, 172, Node999, NEUTRAL_REACTION);
         end
     end
     else begin
-        gsay_reply(1162, 173);
-        giq_option(-3, 1162, 174, Node012, 50);
-        giq_option(-3, 1162, 175, Node002, 50);
-        giq_option(4, 1162, 176, Node012, 50);
-        giq_option(4, 1162, 177, Node002, 50);
+        Reply(173);
+        giq_option(-3, NAME, 174, Node012, NEUTRAL_REACTION);
+        giq_option(-3, NAME, 175, Node002, NEUTRAL_REACTION);
+        giq_option(4, NAME, 176, Node012, NEUTRAL_REACTION);
+        giq_option(4, NAME, 177, Node002, NEUTRAL_REACTION);
     end
 end
 
 procedure Node012
 begin
-    gsay_reply(1162, 178);
-    giq_option(1, 1162, 179, NodePer15, 50);
+    Reply(178);
+    giq_option(1, NAME, 179, NodePer15, NEUTRAL_REACTION);
 end
 
 procedure Node013
 begin
     if (not((local_var(2) bwand 8) != 0)) then begin
-        gsay_reply(1162, 180);
+        Reply(180);
         if (obj_carrying_pid_obj(dude_obj, 503)) then begin
-            giq_option(-3, 1162, 181, Node013s, 50);
-            giq_option(4, 1162, 183, Node013s, 50);
+            giq_option(-3, NAME, 181, Node013s, NEUTRAL_REACTION);
+            giq_option(4, NAME, 183, Node013s, NEUTRAL_REACTION);
         end
         else begin
-            giq_option(-3, 1162, 182, Node999, 50);
-            giq_option(4, 1162, 184, Node999, 50);
+            giq_option(-3, NAME, 182, Node999, NEUTRAL_REACTION);
+            giq_option(4, NAME, 184, Node999, NEUTRAL_REACTION);
         end
     end
     else begin
-        gsay_reply(1162, 185);
-        giq_option(-3, 1162, 186, Node014, 50);
-        giq_option(-3, 1162, 187, Node002, 50);
-        giq_option(4, 1162, 188, Node014, 50);
-        giq_option(4, 1162, 189, Node002, 50);
+        Reply(185);
+        giq_option(-3, NAME, 186, Node014, NEUTRAL_REACTION);
+        giq_option(-3, NAME, 187, Node002, NEUTRAL_REACTION);
+        giq_option(4, NAME, 188, Node014, NEUTRAL_REACTION);
+        giq_option(4, NAME, 189, Node002, NEUTRAL_REACTION);
     end
 end
 
 procedure Node014
 begin
-    gsay_reply(1162, 190);
-    giq_option(1, 1162, 191, NodeChr15, 50);
+    Reply(190);
+    giq_option(1, NAME, 191, NodeChr15, NEUTRAL_REACTION);
 end
 
 procedure Node015
 begin
-    gsay_reply(1162, 192);
-    giq_option(1, 1162, 193, Node999, 50);
+    Reply(192);
+    giq_option(1, NAME, 193, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node016
 begin
-    gsay_reply(1162, 194);
+    Reply(194);
     if (not(q1)) then begin
-        giq_option(-3, 1162, 195, Node017a, 50);
-        giq_option(4, 1162, 199, Node017a, 50);
+        giq_option(-3, NAME, 195, Node017a, NEUTRAL_REACTION);
+        giq_option(4, NAME, 199, Node017a, NEUTRAL_REACTION);
     end
     if (not(q2)) then begin
-        giq_option(-3, 1162, 196, Node022a, 50);
-        giq_option(4, 1162, 200, Node022a, 50);
+        giq_option(-3, NAME, 196, Node022a, NEUTRAL_REACTION);
+        giq_option(4, NAME, 200, Node022a, NEUTRAL_REACTION);
     end
     if (not(q3)) then begin
-        giq_option(-3, 1162, 197, Node023a, 50);
-        giq_option(4, 1162, 201, Node023a, 50);
+        giq_option(-3, NAME, 197, Node023a, NEUTRAL_REACTION);
+        giq_option(4, NAME, 201, Node023a, NEUTRAL_REACTION);
     end
-    giq_option(-3, 1162, 198, Node999, 50);
-    giq_option(4, 1162, 202, Node999, 50);
+    giq_option(-3, NAME, 198, Node999, NEUTRAL_REACTION);
+    giq_option(4, NAME, 202, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node017
 begin
-    gsay_reply(1162, 203);
-    giq_option(-3, 1162, 204, Node018, 50);
-    giq_option(4, 1162, 205, Node019, 50);
+    Reply(203);
+    giq_option(-3, NAME, 204, Node018, NEUTRAL_REACTION);
+    giq_option(4, NAME, 205, Node019, NEUTRAL_REACTION);
 end
 
 procedure Node018
 begin
-    gsay_reply(1162, 206);
-    giq_option(-3, 1162, 207, Node019, 50);
+    Reply(206);
+    giq_option(-3, NAME, 207, Node019, NEUTRAL_REACTION);
 end
 
 procedure Node019
 begin
-    gsay_reply(1162, 208);
-    giq_option(1, 1162, 209, Node020, 50);
+    Reply(208);
+    giq_option(1, NAME, 209, Node020, NEUTRAL_REACTION);
 end
 
 procedure Node020
 begin
-    gsay_reply(1162, 210);
-    giq_option(-3, 1162, 211, Node021, 50);
-    giq_option(-3, 1162, 212, Node016, 50);
-    giq_option(4, 1162, 213, Node021, 50);
-    giq_option(4, 1162, 214, Node016, 50);
+    Reply(210);
+    giq_option(-3, NAME, 211, Node021, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 212, Node016, NEUTRAL_REACTION);
+    giq_option(4, NAME, 213, Node021, NEUTRAL_REACTION);
+    giq_option(4, NAME, 214, Node016, NEUTRAL_REACTION);
 end
 
 procedure Node021
 begin
-    gsay_reply(1162, 215);
-    giq_option(-3, 1162, 216, Node016, 50);
-    giq_option(4, 1162, 217, Node016, 50);
+    Reply(215);
+    giq_option(-3, NAME, 216, Node016, NEUTRAL_REACTION);
+    giq_option(4, NAME, 217, Node016, NEUTRAL_REACTION);
 end
 
 procedure Node022
 begin
-    gsay_reply(1162, 218);
-    giq_option(-3, 1162, 219, Node016, 50);
-    giq_option(4, 1162, 220, Node016, 50);
+    Reply(218);
+    giq_option(-3, NAME, 219, Node016, NEUTRAL_REACTION);
+    giq_option(4, NAME, 220, Node016, NEUTRAL_REACTION);
 end
 
 procedure Node023
 begin
-    gsay_reply(1162, 221);
-    giq_option(1, 1162, 222, Node016, 50);
+    Reply(221);
+    giq_option(1, NAME, 222, Node016, NEUTRAL_REACTION);
 end
 
 procedure Node024
 begin
-    gsay_reply(1162, 224);
-    giq_option(1, 1162, 225, NodeSequence, 50);
-    giq_option(1, 1162, 226, Node001, 50);
+    Reply(224);
+    giq_option(1, NAME, 225, NodeSequence, NEUTRAL_REACTION);
+    giq_option(1, NAME, 226, Node001, NEUTRAL_REACTION);
 end
 
 procedure sequence
@@ -530,8 +530,8 @@ begin
     variable LVar1 := 0;
     if (seqCnt == 0) then begin
         gfade_out(500);
-        set_global_var(632, global_var(632) bwor 262144);
-        if (not(global_var(769))) then begin
+        set_global_var(GVAR_PARTY_MEMBERS_HIDDEN, global_var(GVAR_PARTY_MEMBERS_HIDDEN) bwor 262144);
+        if (not(global_var(GVAR_BOS_LOCATION_ELEVATORS))) then begin
             set_obj_visibility(party_member_obj(16777278), 1);
             set_obj_visibility(party_member_obj(16777376), 1);
             set_obj_visibility(party_member_obj(16777377), 1);
@@ -559,60 +559,60 @@ begin
             set_obj_visibility(party_member_obj(16777747), 1);
             set_obj_visibility(party_member_obj(16777785), 1);
             set_obj_visibility(party_member_obj(16777801), 1);
-            set_global_var(769, 1);
+            set_global_var(GVAR_BOS_LOCATION_ELEVATORS, 1);
         end
         game_ui_disable;
-        lastDudePos := tile_num(dude_obj);
-        lastDudeRot := has_trait(1, dude_obj, 10);
+        lastDudePos := dude_tile;
+        lastDudeRot := obj_get_rot(dude_obj);
         set_obj_visibility(dude_obj, 1);
-        guard1 := create_object_sid(16777466, 14324, 2, -1);
-        guard2 := create_object_sid(16777466, 14328, 2, -1);
-        body := create_object_sid(16777504, 14726, 2, -1);
-        matt := create_object_sid(16777256, 15326, 2, -1);
+        guard1 := create_object(16777466, 14324, 2);
+        guard2 := create_object(16777466, 14328, 2);
+        body := create_object(16777504, 14726, 2);
+        matt := create_object(16777256, 15326, 2);
         anim(guard1, 1000, rotation_to_tile(tile_num(guard1), tile_num(matt)));
         anim(guard2, 1000, rotation_to_tile(tile_num(guard2), tile_num(matt)));
         anim(body, 1000, rotation_to_tile(tile_num(body), tile_num(matt)));
         anim(matt, 1000, rotation_to_tile(tile_num(matt), tile_num(body)));
-        set_global_var(398, 1);
+        set_global_var(GVAR_PARTY_NO_FOLLOW, 1);
         move_to(dude_obj, 15125, 2);
         gfade_in(500);
-        float_msg(matt, message_str(1162, 227), 0);
+        float_msg(matt, mstr(227), FLOAT_MSG_NORMAL);
         seqCnt := seqCnt + 1;
         add_timer_event(self_obj, game_ticks(3), 1);
     end
     else begin
         if (seqCnt == 1) then begin
-            float_msg(body, message_str(1162, 228), 0);
+            float_msg(body, mstr(228), FLOAT_MSG_NORMAL);
             seqCnt := seqCnt + 1;
             add_timer_event(self_obj, game_ticks(3), 1);
         end
         else begin
             if (seqCnt == 2) then begin
-                float_msg(matt, message_str(1162, 229), 0);
+                float_msg(matt, mstr(229), FLOAT_MSG_NORMAL);
                 seqCnt := seqCnt + 1;
                 add_timer_event(self_obj, game_ticks(3), 1);
             end
             else begin
                 if (seqCnt == 3) then begin
-                    float_msg(body, message_str(1162, 230), 2);
+                    float_msg(body, mstr(230), FLOAT_MSG_RED);
                     seqCnt := seqCnt + 1;
                     add_timer_event(self_obj, game_ticks(5), 1);
                 end
                 else begin
                     if (seqCnt == 4) then begin
-                        float_msg(matt, message_str(1162, 231), 0);
+                        float_msg(matt, mstr(231), FLOAT_MSG_NORMAL);
                         seqCnt := seqCnt + 1;
                         add_timer_event(self_obj, game_ticks(3), 1);
                     end
                     else begin
                         if (seqCnt == 5) then begin
-                            float_msg(body, message_str(1162, 232), 3);
+                            float_msg(body, mstr(232), FLOAT_MSG_GREEN);
                             seqCnt := seqCnt + 1;
                             add_timer_event(self_obj, game_ticks(3), 1);
                         end
                         else begin
                             if (seqCnt == 6) then begin
-                                wep := create_object_sid(233, 0, 0, -1);
+                                wep := create_object(233, 0, 0);
                                 add_obj_to_inven(matt, wep);
                                 wield_obj_critter(matt, wep);
                                 seqCnt := seqCnt + 1;
@@ -620,7 +620,7 @@ begin
                             end
                             else begin
                                 if (seqCnt == 7) then begin
-                                    float_msg(body, message_str(1162, 233), 0);
+                                    float_msg(body, mstr(233), FLOAT_MSG_NORMAL);
                                     seqCnt := seqCnt + 1;
                                     add_timer_event(self_obj, game_ticks(1), 1);
                                 end
@@ -643,19 +643,19 @@ begin
                                     end
                                     else begin
                                         if (seqCnt == 10) then begin
-                                            float_msg(body, message_str(1162, 234), 3);
+                                            float_msg(body, mstr(234), FLOAT_MSG_GREEN);
                                             seqCnt := seqCnt + 1;
                                             add_timer_event(self_obj, game_ticks(3), 1);
                                         end
                                         else begin
                                             if (seqCnt == 11) then begin
-                                                float_msg(body, message_str(1162, 235), 3);
+                                                float_msg(body, mstr(235), FLOAT_MSG_GREEN);
                                                 seqCnt := seqCnt + 1;
                                                 add_timer_event(self_obj, game_ticks(3), 1);
                                             end
                                             else begin
                                                 if (seqCnt == 12) then begin
-                                                    float_msg(matt, message_str(1162, 236), 3);
+                                                    float_msg(matt, mstr(236), FLOAT_MSG_GREEN);
                                                     seqCnt := seqCnt + 1;
                                                     add_timer_event(self_obj, game_ticks(1), 1);
                                                 end
@@ -688,19 +688,19 @@ begin
                                                         else begin
                                                             if (seqCnt == 15) then begin
                                                                 anim(body, 1000, rotation_to_tile(tile_num(body), tile_num(guard1)));
-                                                                float_msg(matt, message_str(1162, 237), 0);
+                                                                float_msg(matt, mstr(237), FLOAT_MSG_NORMAL);
                                                                 seqCnt := seqCnt + 1;
                                                                 add_timer_event(self_obj, game_ticks(3), 1);
                                                             end
                                                             else begin
                                                                 if (seqCnt == 16) then begin
-                                                                    float_msg(guard1, message_str(1162, 238), 3);
+                                                                    float_msg(guard1, mstr(238), FLOAT_MSG_GREEN);
                                                                     seqCnt := seqCnt + 1;
                                                                     add_timer_event(self_obj, game_ticks(1), 1);
                                                                 end
                                                                 else begin
                                                                     if (seqCnt == 17) then begin
-                                                                        float_msg(guard2, message_str(1162, 239), 0);
+                                                                        float_msg(guard2, mstr(239), FLOAT_MSG_NORMAL);
                                                                         seqCnt := seqCnt + 1;
                                                                         add_timer_event(self_obj, game_ticks(3), 1);
                                                                     end
@@ -717,12 +717,12 @@ begin
                                                                             guard2 := 0;
                                                                             body := 0;
                                                                             move_to(dude_obj, 14547, 1);
-                                                                            set_global_var(398, 0);
+                                                                            set_global_var(GVAR_PARTY_NO_FOLLOW, 0);
                                                                             anim(dude_obj, 1000, lastDudeRot);
                                                                             set_obj_visibility(dude_obj, 0);
                                                                             game_ui_enable;
-                                                                            set_global_var(632, global_var(632) bwand (-1 - 262144));
-                                                                            if (global_var(769)) then begin
+                                                                            set_global_var(GVAR_PARTY_MEMBERS_HIDDEN, global_var(GVAR_PARTY_MEMBERS_HIDDEN) bwand (-1 - 262144));
+                                                                            if (global_var(GVAR_BOS_LOCATION_ELEVATORS)) then begin
                                                                                 set_obj_visibility(party_member_obj(16777278), 0);
                                                                                 set_obj_visibility(party_member_obj(16777376), 0);
                                                                                 set_obj_visibility(party_member_obj(16777377), 0);
@@ -750,9 +750,9 @@ begin
                                                                                 set_obj_visibility(party_member_obj(16777747), 0);
                                                                                 set_obj_visibility(party_member_obj(16777785), 0);
                                                                                 set_obj_visibility(party_member_obj(16777801), 0);
-                                                                                set_global_var(769, 0);
+                                                                                set_global_var(GVAR_BOS_LOCATION_ELEVATORS, 0);
                                                                             end
-                                                                            set_global_var(582, global_var(582) bwor 262144);
+                                                                            set_global_var(GVAR_VIGNETTE_SEQUENCE, global_var(GVAR_VIGNETTE_SEQUENCE) bwor 262144);
                                                                             gfade_in(500);
                                                                         end
                                                                     end
@@ -788,7 +788,7 @@ end
 procedure NodeHeal15
 begin
     gfade_out(500);
-    critter_heal(dude_obj, get_critter_stat(dude_obj, 7) - get_critter_stat(dude_obj, 35));
+    critter_heal(dude_obj, get_critter_stat(dude_obj, STAT_max_hp) - get_critter_stat(dude_obj, STAT_current_hp));
     game_time_advance(60 * 10 * 5);
     gfade_in(500);
     call Node015();
@@ -806,7 +806,7 @@ end
 procedure NodeRad15
 begin
     gfade_out(500);
-    radiation_dec(dude_obj, get_critter_stat(dude_obj, 37));
+    radiation_dec(dude_obj, get_critter_stat(dude_obj, STAT_current_rad));
     game_time_advance(60 * 10 * 60);
     gfade_in(500);
     call Node015();
@@ -824,7 +824,7 @@ end
 procedure NodeStr15
 begin
     gfade_out(500);
-    set_critter_stat(dude_obj, 0, 1);
+    set_critter_stat(dude_obj, STAT_st, 1);
     game_time_advance(7 * (24 * (60 * (60 * 10))) * 2);
     gfade_in(500);
     set_local_var(1, local_var(1) bwor 1);
@@ -834,7 +834,7 @@ end
 procedure NodeInt15
 begin
     gfade_out(500);
-    set_critter_stat(dude_obj, 4, 1);
+    set_critter_stat(dude_obj, STAT_iq, 1);
     game_time_advance(7 * (24 * (60 * (60 * 10))) * 4);
     gfade_in(500);
     set_local_var(1, local_var(1) bwor 2);
@@ -844,7 +844,7 @@ end
 procedure NodePer15
 begin
     gfade_out(500);
-    set_critter_stat(dude_obj, 1, 1);
+    set_critter_stat(dude_obj, STAT_pe, 1);
     game_time_advance(7 * (24 * (60 * (60 * 10))) * 2);
     gfade_in(500);
     set_local_var(1, local_var(1) bwor 4);
@@ -855,15 +855,15 @@ procedure NodeChr15
 begin
     gfade_out(500);
     game_time_advance(7 * (24 * (60 * (60 * 10))) * 3);
-    if (has_trait(0, dude_obj, 75) and has_trait(0, dude_obj, 77)) then begin
+    if (has_trait(TRAIT_PERK, dude_obj, PERK_dermal_enhancement_perk) and has_trait(TRAIT_PERK, dude_obj, PERK_phoenix_enhancement_perk)) then begin
         set_pc_extra_stat(3, get_pc_extra_stat(3) + 2);
     end
     else begin
-        if (has_trait(0, dude_obj, 75) or has_trait(0, dude_obj, 77)) then begin
+        if (has_trait(TRAIT_PERK, dude_obj, PERK_dermal_enhancement_perk) or has_trait(TRAIT_PERK, dude_obj, PERK_phoenix_enhancement_perk)) then begin
             set_pc_extra_stat(3, get_pc_extra_stat(3) + 1);
         end
         else begin
-            set_critter_stat(dude_obj, 3, 1);
+            set_critter_stat(dude_obj, STAT_ch, 1);
         end
     end
     gfade_in(500);
@@ -934,4 +934,3 @@ begin
     doSeq := 1;
     call Node999();
 end
-
