diff --git a/Mods/MegaMod/Original/FSPHYSTA.ssl b/Mods/MegaMod/ReDefine/FSPHYSTA.ssl
index ee66d44..3bcde05 100644
--- a/Mods/MegaMod/Original/FSPHYSTA.ssl
+++ b/Mods/MegaMod/ReDefine/FSPHYSTA.ssl
@@ -172,7 +172,7 @@ begin
         end_dialogue;
     end
     else begin
-        if (not((global_var(361) bwand 16384) != 0)) then begin
+        if (not((global_var(GVAR_SAN_FRAN_FLAGS) bwand 16384) != 0)) then begin
             start_gdialog(922, self_obj, 4, -1, -1);
             gsay_start;
             call Node008();
@@ -198,17 +198,17 @@ procedure look_at_p_proc
 begin
     script_overrides;
     if (local_var(4) == 0) then begin
-        display_msg(message_str(922, 100));
+        display_mstr(100);
     end
     else begin
-        display_msg(message_str(922, 101));
+        display_mstr(101);
     end
 end
 
 procedure description_p_proc
 begin
     script_overrides;
-    display_msg(message_str(922, 102));
+    display_mstr(102);
 end
 
 procedure use_skill_on_p_proc
@@ -218,7 +218,7 @@ end
 
 procedure use_obj_on_p_proc
 begin
-    display_msg(message_str(922, 131));
+    display_mstr(131);
 end
 
 procedure damage_p_proc
@@ -233,11 +233,11 @@ end
 procedure use_p_proc
 begin
     script_overrides;
-    if (not((global_var(361) bwand 1024) != 0)) then begin
+    if (not((global_var(GVAR_SAN_FRAN_FLAGS) bwand 1024) != 0)) then begin
         dialogue_system_enter;
     end
     else begin
-        display_msg(message_str(922, 130));
+        display_mstr(130);
     end
 end
 
@@ -251,78 +251,78 @@ end
 
 procedure Node001
 begin
-    gsay_reply(922, 103);
-    giq_option(-3, 922, 104, Node999, 50);
-    giq_option(4, 922, 105, NodeGuess, 50);
+    Reply(103);
+    giq_option(-3, NAME, 104, Node999, NEUTRAL_REACTION);
+    giq_option(4, NAME, 105, NodeGuess, NEUTRAL_REACTION);
     if (obj_carrying_pid_obj(dude_obj, 453)) then begin
-        giq_option(4, 922, 106, Node002, 50);
+        giq_option(4, NAME, 106, Node002, NEUTRAL_REACTION);
     end
-    giq_option(4, 922, 107, NodeHack, 50);
-    giq_option(4, 922, 108, Node999, 50);
+    giq_option(4, NAME, 107, NodeHack, NEUTRAL_REACTION);
+    giq_option(4, NAME, 108, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node002
 begin
-    gsay_reply(922, 109);
-    giq_option(4, 922, 110, Node56, 50);
-    giq_option(4, 922, 111, Node007, 50);
-    giq_option(4, 922, 112, Node008, 50);
+    Reply(109);
+    giq_option(4, NAME, 110, Node56, NEUTRAL_REACTION);
+    giq_option(4, NAME, 111, Node007, NEUTRAL_REACTION);
+    giq_option(4, NAME, 112, Node008, NEUTRAL_REACTION);
 end
 
 procedure Node003
 begin
-    gsay_reply(922, 113);
-    giq_option(4, 922, 114, NodeHack, 50);
-    giq_option(4, 922, 115, Node999, 50);
+    Reply(113);
+    giq_option(4, NAME, 114, NodeHack, NEUTRAL_REACTION);
+    giq_option(4, NAME, 115, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node004
 begin
     set_local_var(7, 1);
-    set_global_var(364, 1);
-    gsay_reply(922, 116);
-    giq_option(4, 922, 117, Node999, 50);
+    set_global_var(GVAR_SAN_FRAN_SHIHACKED, 1);
+    Reply(116);
+    giq_option(4, NAME, 117, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node005
 begin
-    gsay_reply(922, 118);
-    if ((global_var(427) == 4) or ((global_var(361) bwand 33554432) != 0) and (local_var(8) == 0)) then begin
-        giq_option(4, 922, 119, NodePlans, 50);
+    Reply(118);
+    if ((global_var(GVAR_SAN_FRAN_ELRON_WHIRLY) == 4) or ((global_var(GVAR_SAN_FRAN_FLAGS) bwand 33554432) != 0) and (local_var(8) == 0)) then begin
+        giq_option(4, NAME, 119, NodePlans, NEUTRAL_REACTION);
     end
-    giq_option(4, 922, 120, Node002, 50);
+    giq_option(4, NAME, 120, Node002, NEUTRAL_REACTION);
 end
 
 procedure Node006
 begin
-    gsay_reply(922, 121);
-    giq_option(4, 922, 122, Node002, 50);
+    Reply(121);
+    giq_option(4, NAME, 122, Node002, NEUTRAL_REACTION);
 end
 
 procedure Node007
 begin
-    gsay_reply(922, 123);
-    giq_option(4, 922, 124, Node002, 50);
+    Reply(123);
+    giq_option(4, NAME, 124, Node002, NEUTRAL_REACTION);
 end
 
 procedure Node008
 begin
-    gsay_reply(922, 125);
-    giq_option(4, 922, 126, Node001, 50);
-    giq_option(4, 922, 127, Node999, 50);
+    Reply(125);
+    giq_option(4, NAME, 126, Node001, NEUTRAL_REACTION);
+    giq_option(4, NAME, 127, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node009
 begin
-    gsay_reply(922, 128);
-    giq_option(4, 922, 129, Node002, 50);
+    Reply(128);
+    giq_option(4, NAME, 129, Node002, NEUTRAL_REACTION);
 end
 
 procedure Node010
 begin
-    set_global_var(361, global_var(361) bwand (-1 - 16384));
-    gsay_reply(922, 132);
-    giq_option(1, 922, 117, Node999, 50);
+    set_global_var(GVAR_SAN_FRAN_FLAGS, global_var(GVAR_SAN_FRAN_FLAGS) bwand (-1 - 16384));
+    Reply(132);
+    giq_option(1, NAME, 117, Node999, NEUTRAL_REACTION);
 end
 
 procedure NodeHack
@@ -332,7 +332,7 @@ begin
     end
     else begin
         if (is_success(roll_vs_skill(dude_obj, 12, 100))) then begin
-            set_global_var(364, 1);
+            set_global_var(GVAR_SAN_FRAN_SHIHACKED, 1);
             call Node002();
         end
         else begin
@@ -353,7 +353,7 @@ end
 
 procedure Node56
 begin
-    if (global_var(519) == 8) then begin
+    if (global_var(GVAR_SAN_FRAN_SHI_WHIRLY) == 8) then begin
         call Node005();
     end
     else begin
@@ -364,9 +364,8 @@ end
 procedure NodePlans
 begin
     variable LVar0 := 0;
-    LVar0 := create_object_sid(446, 0, 0, -1);
+    LVar0 := create_object(446, 0, 0);
     add_obj_to_inven(dude_obj, LVar0);
     set_local_var(8, 1);
     call Node010();
 end
-
