diff --git a/Mods/MegaMod/Original/FSSUBSP.ssl b/Mods/MegaMod/ReDefine/FSSUBSP.ssl
index 5ebea50..1cb6281 100644
--- a/Mods/MegaMod/Original/FSSUBSP.ssl
+++ b/Mods/MegaMod/ReDefine/FSSUBSP.ssl
@@ -14,11 +14,11 @@ begin
         if (local_var(1) == 0) then begin
             reg_anim_func(2, dude_obj);
             set_local_var(1, 1);
-            if (has_trait(0, dude_obj, 9)) then begin
-                display_msg(message_str(2487, 101));
+            if (has_trait(TRAIT_PERK, dude_obj, PERK_night_vision)) then begin
+                display_mstr(101);
             end
             else begin
-                display_msg(message_str(2487, 100));
+                display_mstr(100);
             end
         end
     end
@@ -31,4 +31,3 @@ end
 procedure map_update_p_proc
 begin
 end
-
