diff --git a/Mods/MegaMod/Original/GCENCLEN.ssl b/Mods/MegaMod/ReDefine/GCENCLEN.ssl
index 21b35e5..6bf4a23 100644
--- a/Mods/MegaMod/Original/GCENCLEN.ssl
+++ b/Mods/MegaMod/ReDefine/GCENCLEN.ssl
@@ -19,14 +19,14 @@ end
 
 procedure map_exit_p_proc
 begin
-    set_global_var(633, 0);
+    set_global_var(GVAR_CAR_PLACED_TILE, 0);
 end
 
 procedure map_enter_p_proc
 begin
-    set_global_var(633, 0);
-    set_global_var(1137, 2);
-    display_msg(message_str(2318, 100));
+    set_global_var(GVAR_CAR_PLACED_TILE, 0);
+    set_global_var(GVAR_ABBEY_LIGHTER, 2);
+    display_mstr(100);
     override_map_start(20123 % 200, 20123 / 200, 0, 0);
     call make_warriors_p_proc();
     if ((get_month >= 3) and (get_month < 5)) then begin
@@ -190,36 +190,36 @@ end
 procedure make_warriors_p_proc
 begin
     variable LVar0 := 0;
-    if (get_pc_stat(1) <= 7) then begin
-        LVar0 := create_object_sid(16777466, 18926, 0, 2320);
-        LVar0 := create_object_sid(16777466, 18521, 0, 2319);
+    if (get_pc_stat(PCSTAT_level) <= 7) then begin
+        LVar0 := create_object_sid(16777466, 18926, 0, SCRIPT_GCENC2);
+        LVar0 := create_object_sid(16777466, 18521, 0, SCRIPT_GCENC);
     end
     else begin
-        if ((get_pc_stat(1) <= 12) and (get_pc_stat(1) > 7)) then begin
-            LVar0 := create_object_sid(16777466, 18926, 0, 2320);
-            LVar0 := create_object_sid(16777466, 18521, 0, 2319);
-            LVar0 := create_object_sid(16777466, 19130, 0, 2319);
-            LVar0 := create_object_sid(16777466, 18516, 0, 2319);
+        if ((get_pc_stat(PCSTAT_level) <= 12) and (get_pc_stat(PCSTAT_level) > 7)) then begin
+            LVar0 := create_object_sid(16777466, 18926, 0, SCRIPT_GCENC2);
+            LVar0 := create_object_sid(16777466, 18521, 0, SCRIPT_GCENC);
+            LVar0 := create_object_sid(16777466, 19130, 0, SCRIPT_GCENC);
+            LVar0 := create_object_sid(16777466, 18516, 0, SCRIPT_GCENC);
         end
         else begin
-            if ((get_pc_stat(1) <= 18) and (get_pc_stat(1) > 12)) then begin
-                LVar0 := create_object_sid(16777466, 18926, 0, 2320);
-                LVar0 := create_object_sid(16777466, 18521, 0, 2319);
-                LVar0 := create_object_sid(16777466, 19130, 0, 2319);
-                LVar0 := create_object_sid(16777466, 18516, 0, 2319);
-                LVar0 := create_object_sid(16777466, 19313, 0, 2319);
-                LVar0 := create_object_sid(16777466, 18908, 0, 2319);
+            if ((get_pc_stat(PCSTAT_level) <= 18) and (get_pc_stat(PCSTAT_level) > 12)) then begin
+                LVar0 := create_object_sid(16777466, 18926, 0, SCRIPT_GCENC2);
+                LVar0 := create_object_sid(16777466, 18521, 0, SCRIPT_GCENC);
+                LVar0 := create_object_sid(16777466, 19130, 0, SCRIPT_GCENC);
+                LVar0 := create_object_sid(16777466, 18516, 0, SCRIPT_GCENC);
+                LVar0 := create_object_sid(16777466, 19313, 0, SCRIPT_GCENC);
+                LVar0 := create_object_sid(16777466, 18908, 0, SCRIPT_GCENC);
             end
             else begin
-                if (get_pc_stat(1) >= 18) then begin
-                    LVar0 := create_object_sid(16777466, 18926, 0, 2320);
-                    LVar0 := create_object_sid(16777466, 18521, 0, 2319);
-                    LVar0 := create_object_sid(16777466, 19130, 0, 2319);
-                    LVar0 := create_object_sid(16777466, 18516, 0, 2319);
-                    LVar0 := create_object_sid(16777466, 19313, 0, 2319);
-                    LVar0 := create_object_sid(16777466, 18908, 0, 2319);
-                    LVar0 := create_object_sid(16777466, 18312, 0, 2319);
-                    LVar0 := create_object_sid(16777466, 20315, 0, 2319);
+                if (get_pc_stat(PCSTAT_level) >= 18) then begin
+                    LVar0 := create_object_sid(16777466, 18926, 0, SCRIPT_GCENC2);
+                    LVar0 := create_object_sid(16777466, 18521, 0, SCRIPT_GCENC);
+                    LVar0 := create_object_sid(16777466, 19130, 0, SCRIPT_GCENC);
+                    LVar0 := create_object_sid(16777466, 18516, 0, SCRIPT_GCENC);
+                    LVar0 := create_object_sid(16777466, 19313, 0, SCRIPT_GCENC);
+                    LVar0 := create_object_sid(16777466, 18908, 0, SCRIPT_GCENC);
+                    LVar0 := create_object_sid(16777466, 18312, 0, SCRIPT_GCENC);
+                    LVar0 := create_object_sid(16777466, 20315, 0, SCRIPT_GCENC);
                 end
             end
         end
@@ -227,4 +227,3 @@ begin
     return 0;
     LVar0 := 0;
 end
-
