diff --git a/Mods/MegaMod/Original/GCENC2.ssl b/Mods/MegaMod/ReDefine/GCENC2.ssl
index ce3ac55..45a84b0 100644
--- a/Mods/MegaMod/Original/GCENC2.ssl
+++ b/Mods/MegaMod/ReDefine/GCENC2.ssl
@@ -38,7 +38,7 @@ begin
             if (not(anim_busy(self_obj)) and obj_on_screen(self_obj)) then begin
                 if (obj_can_see_obj(self_obj, dude_obj)) then begin
                     reg_anim_func(1, 1);
-                    anim(self_obj, 1000, rotation_to_tile(tile_num(self_obj), tile_num(dude_obj)));
+                    anim(self_obj, 1000, rotation_to_tile(tile_num(self_obj), dude_tile));
                     reg_anim_func(3, 0);
                 end
             end
@@ -61,11 +61,11 @@ begin
     end
     else begin
         if (count < 6) then begin
-            float_msg(self_obj, message_str(2320, random(150, 159)), 2);
+            float_msg(self_obj, mstr(random(150, 159)), FLOAT_MSG_RED);
             count := count + 1;
         end
         if (count > 5) then begin
-            float_msg(self_obj, message_str(2320, 160), 2);
+            float_msg(self_obj, mstr(160), FLOAT_MSG_RED);
             set_local_var(5, 2);
         end
     end
@@ -74,76 +74,76 @@ end
 procedure critter_p_proc
 begin
     if (local_var(5) == 2) then begin
-        attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+        attack(dude_obj);
     end
 end
 
 procedure pickup_p_proc
 begin
-    attack_complex(source_obj, 0, 1, 0, 0, 30000, 0, 0);
+    attack(source_obj);
     set_map_var(11, 1);
 end
 
 procedure destroy_p_proc
 begin
     if (source_obj == dude_obj) then begin
-        set_global_var(4, global_var(4) + 1);
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(5) > (2 * global_var(4))) or (global_var(3) == 1))) then begin
-            set_global_var(3, 1);
-            set_global_var(2, 0);
+        set_global_var(GVAR_BAD_MONSTER, global_var(GVAR_BAD_MONSTER) + 1);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
+            set_global_var(GVAR_BERSERKER_REPUTATION, 1);
+            set_global_var(GVAR_CHAMPION_REPUTATION, 0);
         end
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(4) > (3 * global_var(5))) or (global_var(2) == 1)) and (global_var(1) == 0)) then begin
-            set_global_var(2, 1);
-            set_global_var(3, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(4) > (3 * global_var(GVAR_GOOD_MONSTER))) or (global_var(GVAR_CHAMPION_REPUTATION) == 1)) and (global_var(GVAR_CHILDKILLER_REPUTATION) == 0)) then begin
+            set_global_var(GVAR_CHAMPION_REPUTATION, 1);
+            set_global_var(GVAR_BERSERKER_REPUTATION, 0);
         end
-        set_global_var(0, global_var(0) + 5);
+        set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + 5);
         debug_msg("Player gains " + 5 + " Karma Points.");
-        set_global_var(37, 0);
-        set_global_var(38, 0);
-        set_global_var(39, 0);
-        set_global_var(40, 0);
-        set_global_var(41, 0);
-        set_global_var(42, 0);
-        set_global_var(43, 0);
-        set_global_var(44, 0);
-        set_global_var(45, 0);
-        tmp_gen_rep := global_var(0);
-        if (has_trait(0, dude_obj, 95)) then begin
-            tmp_gen_rep := global_var(0) * 2;
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+        set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+        set_global_var(GVAR_KARMA_DEFENDER, 0);
+        set_global_var(GVAR_KARMA_WANDERER, 0);
+        set_global_var(GVAR_KARMA_BETRAYER, 0);
+        set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+        set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+        tmp_gen_rep := global_var(GVAR_PLAYER_REPUTATION);
+        if (has_trait(TRAIT_PERK, dude_obj, PERK_karma_beacon_perk)) then begin
+            tmp_gen_rep := global_var(GVAR_PLAYER_REPUTATION) * 2;
         end
         if (tmp_gen_rep >= 1000) then begin
-            set_global_var(37, 1);
+            set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
         end
         else begin
             if (tmp_gen_rep >= 750) then begin
-                set_global_var(38, 1);
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
             end
             else begin
                 if (tmp_gen_rep >= 500) then begin
-                    set_global_var(39, 1);
+                    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                 end
                 else begin
                     if (tmp_gen_rep >= 250) then begin
-                        set_global_var(40, 1);
+                        set_global_var(GVAR_KARMA_DEFENDER, 1);
                     end
                     else begin
                         if (tmp_gen_rep > -250) then begin
-                            set_global_var(41, 1);
+                            set_global_var(GVAR_KARMA_WANDERER, 1);
                         end
                         else begin
                             if (tmp_gen_rep > -500) then begin
-                                set_global_var(42, 1);
+                                set_global_var(GVAR_KARMA_BETRAYER, 1);
                             end
                             else begin
                                 if (tmp_gen_rep > -750) then begin
-                                    set_global_var(43, 1);
+                                    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                 end
                                 else begin
                                     if (tmp_gen_rep > -1000) then begin
-                                        set_global_var(44, 1);
+                                        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                     end
                                     else begin
-                                        set_global_var(45, 1);
+                                        set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                     end
                                 end
                             end
@@ -157,7 +157,7 @@ end
 
 procedure damage_p_proc
 begin
-    attack_complex(source_obj, 0, 1, 0, 0, 30000, 0, 0);
+    attack(source_obj);
     set_map_var(11, 1);
 end
 
@@ -165,15 +165,15 @@ procedure map_enter_p_proc
 begin
     variable LVar0 := 0;
     Only_Once := 0;
-    LVar0 := create_object_sid(394, 0, 0, -1);
+    LVar0 := create_object(394, 0, 0);
     add_obj_to_inven(self_obj, LVar0);
     wield_obj_critter(self_obj, LVar0);
-    LVar0 := create_object_sid(305, 0, 0, -1);
+    LVar0 := create_object(305, 0, 0);
     add_obj_to_inven(self_obj, LVar0);
-    critter_add_trait(self_obj, 1, 6, 190);
-    critter_add_trait(self_obj, 1, 5, 134);
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 190);
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_ENCLAVE_PATROL);
     add_timer_event(self_obj, 10, 1);
-    anim(self_obj, 1000, rotation_to_tile(tile_num(self_obj), tile_num(dude_obj)));
+    anim(self_obj, 1000, rotation_to_tile(tile_num(self_obj), dude_tile));
     return 0;
     LVar0 := 0;
 end
@@ -188,42 +188,42 @@ end
 
 procedure Node001
 begin
-    gsay_reply(2320, 100);
-    giq_option(4, 2320, 101, Node003, 51);
-    giq_option(4, 2320, 102, Node002, 50);
-    giq_option(-3, 2320, 201, Node003, 50);
+    Reply(100);
+    giq_option(4, NAME, 101, Node003, BAD_REACTION);
+    giq_option(4, NAME, 102, Node002, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 201, Node003, NEUTRAL_REACTION);
     set_local_var(7, 1);
 end
 
 procedure Node002
 begin
-    gsay_reply(2320, 105);
-    giq_option(4, 2320, 106, Node004, 50);
-    giq_option(4, 2320, 107, Node003, 51);
-    if (has_skill(dude_obj, 14) > 85) then begin
-        giq_option(4, 2320, 114, Node008, 50);
+    Reply(105);
+    giq_option(4, NAME, 106, Node004, NEUTRAL_REACTION);
+    giq_option(4, NAME, 107, Node003, BAD_REACTION);
+    if (has_skill(dude_obj, SKILL_CONVERSANT) > 85) then begin
+        giq_option(4, NAME, 114, Node008, NEUTRAL_REACTION);
     end
 end
 
 procedure Node003
 begin
-    if (get_critter_stat(dude_obj, 34) == 0) then begin
-        gsay_reply(2320, 103);
+    if (dude_gender == 0) then begin
+        Reply(103);
     end
     else begin
-        gsay_reply(2320, 120);
+        Reply(120);
     end
-    giq_option(4, 2320, 104, Node998, 51);
-    giq_option(-3, 2320, 202, Node998, 51);
+    giq_option(4, NAME, 104, Node998, BAD_REACTION);
+    giq_option(-3, NAME, 202, Node998, BAD_REACTION);
 end
 
 procedure Node004
 begin
-    gsay_reply(2320, 108);
-    giq_option(4, 2320, 109, Node003, 51);
-    giq_option(4, 2320, 110, Node005, 50);
+    Reply(108);
+    giq_option(4, NAME, 109, Node003, BAD_REACTION);
+    giq_option(4, NAME, 110, Node005, NEUTRAL_REACTION);
     if (party_member_obj(16777323) != 0) then begin
-        giq_option(4, 2320, 200, Node005, 50);
+        giq_option(4, NAME, 200, Node005, NEUTRAL_REACTION);
     end
 end
 
@@ -243,29 +243,29 @@ end
 
 procedure Node006
 begin
-    gsay_reply(2320, 111);
-    giq_option(4, 2320, 125, Node006a, 50);
-    giq_option(4, 2320, 126, Node006a, 50);
-    giq_option(4, 2320, 127, Node006a, 50);
+    Reply(111);
+    giq_option(4, NAME, 125, Node006a, NEUTRAL_REACTION);
+    giq_option(4, NAME, 126, Node006a, NEUTRAL_REACTION);
+    giq_option(4, NAME, 127, Node006a, NEUTRAL_REACTION);
 end
 
 procedure Node006a
 begin
-    set_global_var(758, 1);
+    set_global_var(GVAR_LETTER_DELIVER, 1);
 end
 
 procedure Node007
 begin
-    gsay_reply(2320, 112 + (get_critter_stat(dude_obj, 34) == 1));
-    giq_option(4, 2320, 128, Node998, 51);
-    giq_option(4, 2320, 129, Node998, 51);
-    giq_option(4, 2320, 130, Node998, 51);
+    Reply(112 + (dude_gender == 1));
+    giq_option(4, NAME, 128, Node998, BAD_REACTION);
+    giq_option(4, NAME, 129, Node998, BAD_REACTION);
+    giq_option(4, NAME, 130, Node998, BAD_REACTION);
 end
 
 procedure Node008
 begin
-    gsay_reply(2320, 115);
-    giq_option(4, 2320, 116, Node006, 50);
+    Reply(115);
+    giq_option(4, NAME, 116, Node006, NEUTRAL_REACTION);
 end
 
 procedure Node998
@@ -281,4 +281,3 @@ begin
         set_local_var(4, 1);
     end
 end
-
