diff --git a/Mods/MegaMod/Original/GIWELL.ssl b/Mods/MegaMod/ReDefine/GIWELL.ssl
index b2e4460..547094e 100644
--- a/Mods/MegaMod/Original/GIWELL.ssl
+++ b/Mods/MegaMod/ReDefine/GIWELL.ssl
@@ -143,25 +143,25 @@ end
 
 procedure use_p_proc
 begin
-    if (global_var(82) <= 7) then begin
-        display_msg(message_str(1480, 102));
+    if (global_var(GVAR_VAULT_GECKO_PLANT) <= 7) then begin
+        display_mstr(102);
         radiation_inc(dude_obj, 10);
     end
     else begin
-        display_msg(message_str(1480, 103));
+        display_mstr(103);
     end
 end
 
 procedure look_at_p_proc
 begin
     script_overrides;
-    display_msg(message_str(1480, 100));
+    display_mstr(100);
 end
 
 procedure description_p_proc
 begin
     script_overrides;
-    display_msg(message_str(1480, 101));
+    display_mstr(101);
 end
 
 procedure use_skill_on_p_proc
@@ -183,4 +183,3 @@ end
 procedure map_update_p_proc
 begin
 end
-
