diff --git a/Mods/MegaMod/Original/GL_EXPERIENCE_MASTER.ssl b/Mods/MegaMod/ReDefine/GL_EXPERIENCE_MASTER.ssl
index 8520404..94d02e7 100644
--- a/Mods/MegaMod/Original/GL_EXPERIENCE_MASTER.ssl
+++ b/Mods/MegaMod/ReDefine/GL_EXPERIENCE_MASTER.ssl
@@ -66,7 +66,7 @@ begin
             end
             if (not(has_fake_perk("Experience Master (Men)"))) then begin
                 if (get_sfall_global_int(2139) == get_sfall_global_int(1523)) then begin
-                    float_msg(dude_obj, "You are eligible for the Perk " + "Experience Master (Men)", 9);
+                    float_msg(dude_obj, "You are eligible for the Perk " + "Experience Master (Men)", FLOAT_MSG_WHITE);
                 end
             end
             else begin
@@ -87,7 +87,7 @@ begin
             end
             if (not(has_fake_perk("Experience Master (Women)"))) then begin
                 if (get_sfall_global_int(2140) == get_sfall_global_int(1522)) then begin
-                    float_msg(dude_obj, "You are eligible for the Perk " + "Experience Master (Women)", 9);
+                    float_msg(dude_obj, "You are eligible for the Perk " + "Experience Master (Women)", FLOAT_MSG_WHITE);
                 end
             end
             else begin
@@ -108,7 +108,7 @@ begin
             end
             if (not(has_fake_perk("Experience Master (Mutant)"))) then begin
                 if (get_sfall_global_int(2142) == get_sfall_global_int(1521)) then begin
-                    float_msg(dude_obj, "You are eligible for the Perk " + "Experience Master (Mutant)", 9);
+                    float_msg(dude_obj, "You are eligible for the Perk " + "Experience Master (Mutant)", FLOAT_MSG_WHITE);
                 end
             end
             else begin
@@ -129,7 +129,7 @@ begin
             end
             if (not(has_fake_perk("Experience Master (Ghoul)"))) then begin
                 if (get_sfall_global_int(2143) == get_sfall_global_int(1520)) then begin
-                    float_msg(dude_obj, "You are eligible for the Perk " + "Experience Master (Ghoul)", 9);
+                    float_msg(dude_obj, "You are eligible for the Perk " + "Experience Master (Ghoul)", FLOAT_MSG_WHITE);
                 end
             end
             else begin
@@ -150,7 +150,7 @@ begin
             end
             if (not(has_fake_perk("Experience Master (Scorpion)"))) then begin
                 if (get_sfall_global_int(2145) == get_sfall_global_int(1552)) then begin
-                    float_msg(dude_obj, "You are eligible for the Perk " + "Experience Master (Scorpion)", 9);
+                    float_msg(dude_obj, "You are eligible for the Perk " + "Experience Master (Scorpion)", FLOAT_MSG_WHITE);
                 end
             end
             else begin
@@ -171,7 +171,7 @@ begin
             end
             if (not(has_fake_perk("Experience Master (Rat)"))) then begin
                 if (get_sfall_global_int(2146) == get_sfall_global_int(1554)) then begin
-                    float_msg(dude_obj, "You are eligible for the Perk " + "Experience Master (Rat)", 9);
+                    float_msg(dude_obj, "You are eligible for the Perk " + "Experience Master (Rat)", FLOAT_MSG_WHITE);
                 end
             end
             else begin
@@ -192,7 +192,7 @@ begin
             end
             if (not(has_fake_perk("Experience Master (Floater)"))) then begin
                 if (get_sfall_global_int(2147) == get_sfall_global_int(1610)) then begin
-                    float_msg(dude_obj, "You are eligible for the Perk " + "Experience Master (Floater)", 9);
+                    float_msg(dude_obj, "You are eligible for the Perk " + "Experience Master (Floater)", FLOAT_MSG_WHITE);
                 end
             end
             else begin
@@ -213,7 +213,7 @@ begin
             end
             if (not(has_fake_perk("Experience Master (Centaur)"))) then begin
                 if (get_sfall_global_int(2148) == get_sfall_global_int(1639)) then begin
-                    float_msg(dude_obj, "You are eligible for the Perk " + "Experience Master (Centaur)", 9);
+                    float_msg(dude_obj, "You are eligible for the Perk " + "Experience Master (Centaur)", FLOAT_MSG_WHITE);
                 end
             end
             else begin
@@ -234,7 +234,7 @@ begin
             end
             if (not(has_fake_perk("Experience Master (Robot)"))) then begin
                 if (get_sfall_global_int(2149) == get_sfall_global_int(1658)) then begin
-                    float_msg(dude_obj, "You are eligible for the Perk " + "Experience Master (Robot)", 9);
+                    float_msg(dude_obj, "You are eligible for the Perk " + "Experience Master (Robot)", FLOAT_MSG_WHITE);
                 end
             end
             else begin
@@ -255,7 +255,7 @@ begin
             end
             if (not(has_fake_perk("Experience Master (Dog)"))) then begin
                 if (get_sfall_global_int(2150) == get_sfall_global_int(1699)) then begin
-                    float_msg(dude_obj, "You are eligible for the Perk " + "Experience Master (Dog)", 9);
+                    float_msg(dude_obj, "You are eligible for the Perk " + "Experience Master (Dog)", FLOAT_MSG_WHITE);
                 end
             end
             else begin
@@ -276,7 +276,7 @@ begin
             end
             if (not(has_fake_perk("Experience Master (Manti)"))) then begin
                 if (get_sfall_global_int(2151) == get_sfall_global_int(1726)) then begin
-                    float_msg(dude_obj, "You are eligible for the Perk " + "Experience Master (Manti)", 9);
+                    float_msg(dude_obj, "You are eligible for the Perk " + "Experience Master (Manti)", FLOAT_MSG_WHITE);
                 end
             end
             else begin
@@ -297,7 +297,7 @@ begin
             end
             if (not(has_fake_perk("Experience Master (DeathClaw)"))) then begin
                 if (get_sfall_global_int(2152) == get_sfall_global_int(1755)) then begin
-                    float_msg(dude_obj, "You are eligible for the Perk " + "Experience Master (DeathClaw)", 9);
+                    float_msg(dude_obj, "You are eligible for the Perk " + "Experience Master (DeathClaw)", FLOAT_MSG_WHITE);
                 end
             end
             else begin
@@ -318,7 +318,7 @@ begin
             end
             if (not(has_fake_perk("Experience Master (Plant)"))) then begin
                 if (get_sfall_global_int(2153) == get_sfall_global_int(1784)) then begin
-                    float_msg(dude_obj, "You are eligible for the Perk " + "Experience Master (Plant)", 9);
+                    float_msg(dude_obj, "You are eligible for the Perk " + "Experience Master (Plant)", FLOAT_MSG_WHITE);
                 end
             end
             else begin
@@ -339,7 +339,7 @@ begin
             end
             if (not(has_fake_perk("Experience Master (Gecko)"))) then begin
                 if (get_sfall_global_int(2154) == get_sfall_global_int(1813)) then begin
-                    float_msg(dude_obj, "You are eligible for the Perk " + "Experience Master (Gecko)", 9);
+                    float_msg(dude_obj, "You are eligible for the Perk " + "Experience Master (Gecko)", FLOAT_MSG_WHITE);
                 end
             end
             else begin
@@ -360,7 +360,7 @@ begin
             end
             if (not(has_fake_perk("Experience Master (Alien)"))) then begin
                 if (get_sfall_global_int(2155) == get_sfall_global_int(1842)) then begin
-                    float_msg(dude_obj, "You are eligible for the Perk " + "Experience Master (Alien)", 9);
+                    float_msg(dude_obj, "You are eligible for the Perk " + "Experience Master (Alien)", FLOAT_MSG_WHITE);
                 end
             end
             else begin
@@ -381,7 +381,7 @@ begin
             end
             if (not(has_fake_perk("Experience Master (Ant)"))) then begin
                 if (get_sfall_global_int(2156) == get_sfall_global_int(1871)) then begin
-                    float_msg(dude_obj, "You are eligible for the Perk " + "Experience Master (Ant)", 9);
+                    float_msg(dude_obj, "You are eligible for the Perk " + "Experience Master (Ant)", FLOAT_MSG_WHITE);
                 end
             end
             else begin
@@ -395,7 +395,7 @@ begin
             set_sfall_global(2158, get_kill_counter(19));
             if (not(has_fake_perk("Experience Master (Cockroach)"))) then begin
                 if (get_sfall_global_int(2158) == get_sfall_global_int(1900)) then begin
-                    float_msg(dude_obj, "You are eligible for the Perk " + "Experience Master (Cockroach)", 9);
+                    float_msg(dude_obj, "You are eligible for the Perk " + "Experience Master (Cockroach)", FLOAT_MSG_WHITE);
                 end
             end
             else begin
@@ -409,7 +409,7 @@ begin
             set_sfall_global(2159, get_kill_counter(20));
             if (not(has_fake_perk("Experience Master (Fruit Fly)"))) then begin
                 if (get_sfall_global_int(2159) == get_sfall_global_int(1929)) then begin
-                    float_msg(dude_obj, "You are eligible for the Perk " + "Experience Master (Fruit Fly)", 9);
+                    float_msg(dude_obj, "You are eligible for the Perk " + "Experience Master (Fruit Fly)", FLOAT_MSG_WHITE);
                 end
             end
             else begin
@@ -421,4 +421,3 @@ begin
         end
     end
 end
-
