diff --git a/Mods/MegaMod/Original/GL_REPLACE_WEAPONS.ssl b/Mods/MegaMod/ReDefine/GL_REPLACE_WEAPONS.ssl
index a3795c7..92a5b4f 100644
--- a/Mods/MegaMod/Original/GL_REPLACE_WEAPONS.ssl
+++ b/Mods/MegaMod/ReDefine/GL_REPLACE_WEAPONS.ssl
@@ -509,9 +509,9 @@ begin
     if (obj_is_carrying_obj_pid(arg0, arg1) > 0) then begin
         while (obj_is_carrying_obj_pid(arg0, arg1) > 0) do begin
             delete_item := obj_carrying_pid_obj(arg0, arg1);
-            if ((critter_inven_obj(arg0, 2) == arg1) or (critter_inven_obj(arg0, 1) == arg1)) then begin
+            if ((critter_inven_obj(arg0, INVEN_TYPE_LEFT_HAND) == arg1) or (critter_inven_obj(arg0, INVEN_TYPE_RIGHT_HAND) == arg1)) then begin
                 if (critter_state(arg0) == 0) then begin
-                    metarule(43, arg0);
+                    inven_unwield(arg0);
                     LVar3 := arg1;
                 end
             end
@@ -522,7 +522,7 @@ begin
             end
             number_items_remove := number_items_remove + 1;
         end
-        add_mult_objs_to_inven(arg0, create_object_sid(arg2, 0, 0, -1), number_items_remove);
+        add_mult_objs_to_inven(arg0, create_object(arg2, 0, 0), number_items_remove);
         if (LVar3 != 0) then begin
             if (LVar3 == arg1) then begin
                 wield_obj_critter(arg0, LVar3);
@@ -542,11 +542,10 @@ begin
     if (obj_pid(arg0) == arg1) then begin
         delete_item := tile_contains_pid_obj(tile_num(arg0), elevation(arg0), arg1);
         destroy_object(delete_item);
-        create_object_sid(arg2, tile_num(arg0), elevation(arg0), -1);
+        create_object(arg2, tile_num(arg0), elevation(arg0));
     end
     return 0;
     arg0 := 0;
     arg1 := 0;
     arg2 := 0;
 end
-
