diff --git a/Mods/MegaMod/Original/GL_SWIFT_LEARNER.ssl b/Mods/MegaMod/ReDefine/GL_SWIFT_LEARNER.ssl
index 392b9c5..38c0de8 100644
--- a/Mods/MegaMod/Original/GL_SWIFT_LEARNER.ssl
+++ b/Mods/MegaMod/ReDefine/GL_SWIFT_LEARNER.ssl
@@ -10,12 +10,12 @@ begin
     if (game_loaded) then begin
         set_global_script_repeat(60);
         set_global_script_type(1);
-        set_sfall_global("EXP___01", get_pc_stat(2));
-        set_sfall_global("EXP___02", get_critter_stat(dude_obj, 36));
-        set_sfall_global("EXP___03", get_critter_stat(dude_obj, 37));
+        set_sfall_global("EXP___01", get_pc_stat(PCSTAT_experience));
+        set_sfall_global("EXP___02", get_critter_stat(dude_obj, STAT_current_poison));
+        set_sfall_global("EXP___03", get_critter_stat(dude_obj, STAT_current_rad));
     end
     else begin
-        if ((get_pc_stat(2) != get_sfall_global_int("EXP___01")) and has_trait(0, dude_obj, 50)) then begin
+        if ((get_pc_stat(PCSTAT_experience) != get_sfall_global_int("EXP___01")) and has_trait(TRAIT_PERK, dude_obj, PERK_swift_learner)) then begin
             if (get_sfall_global_int(126) == 1) then begin
                 if (get_sfall_global_int(6004) == 1) then begin
                     LVar1 := 0.05000;
@@ -69,16 +69,16 @@ begin
             else begin
                 LVar1 := 0.05000;
             end
-            LVar3 := has_trait(0, dude_obj, 50) * LVar1;
+            LVar3 := has_trait(TRAIT_PERK, dude_obj, PERK_swift_learner) * LVar1;
             LVar2 := LVar3 + 1;
-            LVar0 := get_pc_stat(2) - get_sfall_global_int("EXP___01");
+            LVar0 := get_pc_stat(PCSTAT_experience) - get_sfall_global_int("EXP___01");
             LVar0 := LVar0 / LVar2;
             LVar0 := LVar0 * LVar3;
             display_msg("You gain " + floor(LVar0) + " experience from Swift Learner Perk.");
-            set_sfall_global("EXP___01", get_pc_stat(2));
+            set_sfall_global("EXP___01", get_pc_stat(PCSTAT_experience));
         end
-        if (get_critter_stat(dude_obj, 36) != get_sfall_global_int("EXP___02")) then begin
-            LVar0 := get_critter_stat(dude_obj, 36) - get_sfall_global_int("EXP___02");
+        if (get_critter_stat(dude_obj, STAT_current_poison) != get_sfall_global_int("EXP___02")) then begin
+            LVar0 := get_critter_stat(dude_obj, STAT_current_poison) - get_sfall_global_int("EXP___02");
             if (LVar0 > 0) then begin
                 display_msg("Your poison level has increased by " + LVar0 + ".");
             end
@@ -87,10 +87,10 @@ begin
                     display_msg("Your poison level has decreased by " + -LVar0 + ".");
                 end
             end
-            set_sfall_global("EXP___02", get_critter_stat(dude_obj, 36));
+            set_sfall_global("EXP___02", get_critter_stat(dude_obj, STAT_current_poison));
         end
-        if (get_critter_stat(dude_obj, 37) != get_sfall_global_int("EXP___03")) then begin
-            LVar0 := get_critter_stat(dude_obj, 37) - get_sfall_global_int("EXP___03");
+        if (get_critter_stat(dude_obj, STAT_current_rad) != get_sfall_global_int("EXP___03")) then begin
+            LVar0 := get_critter_stat(dude_obj, STAT_current_rad) - get_sfall_global_int("EXP___03");
             if (LVar0 > 0) then begin
                 display_msg("Your radiation level has increased by " + LVar0 + ".");
             end
@@ -99,8 +99,7 @@ begin
                     display_msg("Your radiation level has decreased by " + -LVar0 + ".");
                 end
             end
-            set_sfall_global("EXP___03", get_critter_stat(dude_obj, 37));
+            set_sfall_global("EXP___03", get_critter_stat(dude_obj, STAT_current_rad));
         end
     end
 end
-
