diff --git a/Mods/MegaMod/Original/HABMERGK.ssl b/Mods/MegaMod/ReDefine/HABMERGK.ssl
index 8a5d55d..7df24f3 100644
--- a/Mods/MegaMod/Original/HABMERGK.ssl
+++ b/Mods/MegaMod/ReDefine/HABMERGK.ssl
@@ -135,19 +135,19 @@ procedure spatial_p_proc
 begin
     if (source_obj == dude_obj) then begin
         if (elevation(dude_obj) == 0) then begin
-            if (tile_num(dude_obj) == 13280) then begin
+            if (dude_tile == 13280) then begin
                 move_to(dude_obj, 29076, 1);
             end
-            if (tile_num(dude_obj) == 15448) then begin
+            if (dude_tile == 15448) then begin
                 move_to(dude_obj, 30049, 1);
             end
         end
         else begin
             if (elevation(dude_obj) == 1) then begin
-                if (tile_num(dude_obj) == 30682) then begin
+                if (dude_tile == 30682) then begin
                     move_to(dude_obj, 14887, 0);
                 end
-                if (tile_num(dude_obj) == 32850) then begin
+                if (dude_tile == 32850) then begin
                     move_to(dude_obj, 17445, 0);
                 end
             end
@@ -162,4 +162,3 @@ end
 procedure start
 begin
 end
-
