diff --git a/Mods/MegaMod/Original/HCCHAD.ssl b/Mods/MegaMod/ReDefine/HCCHAD.ssl
index 9d2e46b..02ff242 100644
--- a/Mods/MegaMod/Original/HCCHAD.ssl
+++ b/Mods/MegaMod/ReDefine/HCCHAD.ssl
@@ -80,7 +80,7 @@ import variable Chad_Ptr;
 
 procedure start
 begin
-    if ((cur_map_index == 78) or (cur_map_index == 79)) then begin
+    if ((CUR_MAP_BROKEN1) or (CUR_MAP_BROKEN2)) then begin
         Chad_Ptr := self_obj;
     end
 end
@@ -90,8 +90,8 @@ begin
     variable LVar0 := 0;
     variable LVar1 := 0;
     if (local_var(7) == 2) then begin
-        LVar0 := get_critter_stat(self_obj, 7);
-        LVar1 := get_critter_stat(self_obj, 35);
+        LVar0 := get_critter_stat(self_obj, STAT_max_hp);
+        LVar1 := get_critter_stat(self_obj, STAT_current_hp);
         if (LVar1 < (LVar0 / 4)) then begin
             terminate_combat;
         end
@@ -105,11 +105,11 @@ procedure critter_p_proc
 begin
     variable LVar0 := 0;
     variable LVar1 := 0;
-    if ((global_var(309) == 1) or (local_var(5) == 2) and obj_can_see_obj(self_obj, dude_obj)) then begin
+    if ((global_var(GVAR_ENEMY_BROKEN_HILLS) == 1) or (local_var(5) == 2) and obj_can_see_obj(self_obj, dude_obj)) then begin
         set_local_var(5, 1);
-        attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+        attack(dude_obj);
     end
-    if (cur_map_index == 79) then begin
+    if (CUR_MAP_BROKEN2) then begin
         if (map_var(25) == 1) then begin
             set_map_var(25, 2);
             add_timer_event(self_obj, game_ticks(0), 4);
@@ -117,8 +117,8 @@ begin
         end
     end
     if (local_var(7) == 2) then begin
-        LVar0 := get_critter_stat(self_obj, 7);
-        LVar1 := get_critter_stat(self_obj, 35);
+        LVar0 := get_critter_stat(self_obj, STAT_max_hp);
+        LVar1 := get_critter_stat(self_obj, STAT_current_hp);
         if (LVar1 < (LVar0 / 4)) then begin
             terminate_combat;
             set_local_var(7, 0);
@@ -143,7 +143,7 @@ begin
     fromWTG := 1;
     if (local_var(2) == 0) then begin
         set_local_var(2, 1);
-        set_local_var(3, (get_critter_stat(dude_obj, 3) - get_critter_stat(self_obj, 3)) * 5);
+        set_local_var(3, (dude_charisma - self_charisma) * 5);
         debug_msg("Base Reaction == " + local_var(3));
     end
     Static_Reaction := 0;
@@ -155,14 +155,14 @@ begin
     else begin
         reaction_bonus_karma := -1 * (-1 * reaction_bonus_karma / 1000);
     end
-    if (global_var(54) >= 0) then begin
-        reaction_bonus_town_rep := global_var(54) / 2;
+    if (global_var(GVAR_TOWN_REP_BROKEN_HILLS) >= 0) then begin
+        reaction_bonus_town_rep := global_var(GVAR_TOWN_REP_BROKEN_HILLS) / 2;
     end
     else begin
-        reaction_bonus_town_rep := -1 * (-1 * global_var(54) / 2);
+        reaction_bonus_town_rep := -1 * (-1 * global_var(GVAR_TOWN_REP_BROKEN_HILLS) / 2);
     end
-    if (has_trait(2, dude_obj, 13)) then begin
-        if (get_critter_stat(dude_obj, 34) == get_critter_stat(self_obj, 34)) then begin
+    if (has_trait(TRAIT_TRAIT, dude_obj, TRAIT_sex_appeal)) then begin
+        if (dude_gender == self_gender) then begin
             Static_Reaction := Static_Reaction + -20;
         end
         else begin
@@ -188,52 +188,52 @@ begin
             Static_Reaction := Static_Reaction + reaction_bonus_town_rep;
         end
         debug_msg("Town Rep Bonus == " + Static_Reaction);
-        set_global_var(37, 0);
-        set_global_var(38, 0);
-        set_global_var(39, 0);
-        set_global_var(40, 0);
-        set_global_var(41, 0);
-        set_global_var(42, 0);
-        set_global_var(43, 0);
-        set_global_var(44, 0);
-        set_global_var(45, 0);
-        tmp_gen_rep := global_var(0);
-        if (has_trait(0, dude_obj, 95)) then begin
-            tmp_gen_rep := global_var(0) * 2;
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+        set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+        set_global_var(GVAR_KARMA_DEFENDER, 0);
+        set_global_var(GVAR_KARMA_WANDERER, 0);
+        set_global_var(GVAR_KARMA_BETRAYER, 0);
+        set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+        set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+        tmp_gen_rep := global_var(GVAR_PLAYER_REPUTATION);
+        if (has_trait(TRAIT_PERK, dude_obj, PERK_karma_beacon_perk)) then begin
+            tmp_gen_rep := global_var(GVAR_PLAYER_REPUTATION) * 2;
         end
         if (tmp_gen_rep >= 1000) then begin
-            set_global_var(37, 1);
+            set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
         end
         else begin
             if (tmp_gen_rep >= 750) then begin
-                set_global_var(38, 1);
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
             end
             else begin
                 if (tmp_gen_rep >= 500) then begin
-                    set_global_var(39, 1);
+                    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                 end
                 else begin
                     if (tmp_gen_rep >= 250) then begin
-                        set_global_var(40, 1);
+                        set_global_var(GVAR_KARMA_DEFENDER, 1);
                     end
                     else begin
                         if (tmp_gen_rep > -250) then begin
-                            set_global_var(41, 1);
+                            set_global_var(GVAR_KARMA_WANDERER, 1);
                         end
                         else begin
                             if (tmp_gen_rep > -500) then begin
-                                set_global_var(42, 1);
+                                set_global_var(GVAR_KARMA_BETRAYER, 1);
                             end
                             else begin
                                 if (tmp_gen_rep > -750) then begin
-                                    set_global_var(43, 1);
+                                    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                 end
                                 else begin
                                     if (tmp_gen_rep > -1000) then begin
-                                        set_global_var(44, 1);
+                                        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                     end
                                     else begin
-                                        set_global_var(45, 1);
+                                        set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                     end
                                 end
                             end
@@ -242,35 +242,35 @@ begin
                 end
             end
         end
-        if (global_var(37) == 1) then begin
+        if (global_var(GVAR_KARMA_HOLY_WARRIOR) == 1) then begin
             Static_Reaction := Static_Reaction - 20;
         end
         else begin
-            if (global_var(38) == 1) then begin
+            if (global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES) == 1) then begin
                 Static_Reaction := Static_Reaction - 15;
             end
             else begin
-                if (global_var(39) == 1) then begin
+                if (global_var(GVAR_KARMA_SHIELD_OF_HOPE) == 1) then begin
                     Static_Reaction := Static_Reaction - 10;
                 end
                 else begin
-                    if (global_var(40) == 1) then begin
+                    if (global_var(GVAR_KARMA_DEFENDER) == 1) then begin
                         Static_Reaction := Static_Reaction - 5;
                     end
                     else begin
-                        if (global_var(41) == 1) then begin
+                        if (global_var(GVAR_KARMA_WANDERER) == 1) then begin
                             Static_Reaction := Static_Reaction - 0;
                         end
                         else begin
-                            if (global_var(42) == 1) then begin
+                            if (global_var(GVAR_KARMA_BETRAYER) == 1) then begin
                                 Static_Reaction := Static_Reaction + -5;
                             end
                             else begin
-                                if (global_var(43) == 1) then begin
+                                if (global_var(GVAR_KARMA_SWORD_OF_DESPAIR) == 1) then begin
                                     Static_Reaction := Static_Reaction + -10;
                                 end
                                 else begin
-                                    if (global_var(44) == 1) then begin
+                                    if (global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE) == 1) then begin
                                         Static_Reaction := Static_Reaction + -15;
                                     end
                                     else begin
@@ -290,52 +290,52 @@ begin
         debug_msg("Karma Reaction Bonus == " + Static_Reaction);
         Static_Reaction := Static_Reaction + reaction_bonus_town_rep;
         debug_msg("Town Rep Bonus == " + Static_Reaction);
-        set_global_var(37, 0);
-        set_global_var(38, 0);
-        set_global_var(39, 0);
-        set_global_var(40, 0);
-        set_global_var(41, 0);
-        set_global_var(42, 0);
-        set_global_var(43, 0);
-        set_global_var(44, 0);
-        set_global_var(45, 0);
-        tmp_gen_rep := global_var(0);
-        if (has_trait(0, dude_obj, 95)) then begin
-            tmp_gen_rep := global_var(0) * 2;
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+        set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+        set_global_var(GVAR_KARMA_DEFENDER, 0);
+        set_global_var(GVAR_KARMA_WANDERER, 0);
+        set_global_var(GVAR_KARMA_BETRAYER, 0);
+        set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+        set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+        tmp_gen_rep := global_var(GVAR_PLAYER_REPUTATION);
+        if (has_trait(TRAIT_PERK, dude_obj, PERK_karma_beacon_perk)) then begin
+            tmp_gen_rep := global_var(GVAR_PLAYER_REPUTATION) * 2;
         end
         if (tmp_gen_rep >= 1000) then begin
-            set_global_var(37, 1);
+            set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
         end
         else begin
             if (tmp_gen_rep >= 750) then begin
-                set_global_var(38, 1);
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
             end
             else begin
                 if (tmp_gen_rep >= 500) then begin
-                    set_global_var(39, 1);
+                    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                 end
                 else begin
                     if (tmp_gen_rep >= 250) then begin
-                        set_global_var(40, 1);
+                        set_global_var(GVAR_KARMA_DEFENDER, 1);
                     end
                     else begin
                         if (tmp_gen_rep > -250) then begin
-                            set_global_var(41, 1);
+                            set_global_var(GVAR_KARMA_WANDERER, 1);
                         end
                         else begin
                             if (tmp_gen_rep > -500) then begin
-                                set_global_var(42, 1);
+                                set_global_var(GVAR_KARMA_BETRAYER, 1);
                             end
                             else begin
                                 if (tmp_gen_rep > -750) then begin
-                                    set_global_var(43, 1);
+                                    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                 end
                                 else begin
                                     if (tmp_gen_rep > -1000) then begin
-                                        set_global_var(44, 1);
+                                        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                     end
                                     else begin
-                                        set_global_var(45, 1);
+                                        set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                     end
                                 end
                             end
@@ -344,35 +344,35 @@ begin
                 end
             end
         end
-        if (global_var(37) == 1) then begin
+        if (global_var(GVAR_KARMA_HOLY_WARRIOR) == 1) then begin
             Static_Reaction := Static_Reaction + 20;
         end
         else begin
-            if (global_var(38) == 1) then begin
+            if (global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES) == 1) then begin
                 Static_Reaction := Static_Reaction + 15;
             end
             else begin
-                if (global_var(39) == 1) then begin
+                if (global_var(GVAR_KARMA_SHIELD_OF_HOPE) == 1) then begin
                     Static_Reaction := Static_Reaction + 10;
                 end
                 else begin
-                    if (global_var(40) == 1) then begin
+                    if (global_var(GVAR_KARMA_DEFENDER) == 1) then begin
                         Static_Reaction := Static_Reaction + 5;
                     end
                     else begin
-                        if (global_var(41) == 1) then begin
+                        if (global_var(GVAR_KARMA_WANDERER) == 1) then begin
                             Static_Reaction := Static_Reaction + 0;
                         end
                         else begin
-                            if (global_var(42) == 1) then begin
+                            if (global_var(GVAR_KARMA_BETRAYER) == 1) then begin
                                 Static_Reaction := Static_Reaction + -5;
                             end
                             else begin
-                                if (global_var(43) == 1) then begin
+                                if (global_var(GVAR_KARMA_SWORD_OF_DESPAIR) == 1) then begin
                                     Static_Reaction := Static_Reaction + -10;
                                 end
                                 else begin
-                                    if (global_var(44) == 1) then begin
+                                    if (global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE) == 1) then begin
                                         Static_Reaction := Static_Reaction + -15;
                                     end
                                     else begin
@@ -388,32 +388,32 @@ begin
         debug_msg("Good Person Reaction Bonus == " + Static_Reaction);
     end
     if (Slavery_Tolerant == 0) then begin
-        if (global_var(11) == 1) then begin
+        if (global_var(GVAR_REPUTATION_SLAVER) == 1) then begin
             Static_Reaction := Static_Reaction + (2 * -25);
         end
     end
     else begin
         if (Slavery_Tolerant == 1) then begin
-            if (global_var(11) == 1) then begin
+            if (global_var(GVAR_REPUTATION_SLAVER) == 1) then begin
                 Static_Reaction := Static_Reaction + -25;
             end
         end
         else begin
             if (Slavery_Tolerant == 3) then begin
-                if (global_var(11) == 1) then begin
+                if (global_var(GVAR_REPUTATION_SLAVER) == 1) then begin
                     Static_Reaction := Static_Reaction - -25;
                 end
             end
         end
     end
     debug_msg("Slaver + Aligned Reaction == " + Static_Reaction);
-    Static_Reaction := Static_Reaction + (10 * has_trait(0, dude_obj, 10));
+    Static_Reaction := Static_Reaction + (10 * has_trait(TRAIT_PERK, dude_obj, PERK_presence));
     debug_msg("Presence Reaction == " + Static_Reaction);
-    if (global_var(1) >= 2) then begin
+    if (global_var(GVAR_CHILDKILLER_REPUTATION) >= 2) then begin
         Static_Reaction := Static_Reaction + -30;
     end
     debug_msg("Childkiller Reaction == " + Static_Reaction);
-    if (has_trait(0, dude_obj, 39)) then begin
+    if (has_trait(TRAIT_PERK, dude_obj, PERK_cult_of_personality)) then begin
         if (Evil_Critter == 1) then begin
             if (Static_Reaction > 0) then begin
                 Static_Reaction := Static_Reaction * -1;
@@ -468,8 +468,8 @@ begin
             end
         end
     end
-    if (global_var(300) == 7) then begin
-        float_msg(self_obj, message_str(613, 400), 8);
+    if (global_var(GVAR_BH_CHAD) == 7) then begin
+        float_msg(self_obj, mstr(400), FLOAT_MSG_YELLOW);
     end
     else begin
         if (local_var(6) == 1) then begin
@@ -494,26 +494,26 @@ begin
                     end_dialogue;
                 end
                 else begin
-                    if (cur_map_index == 94) then begin
-                        float_msg(self_obj, message_str(613, random(173, 175)), 8);
+                    if (CUR_MAP_BHRNDDST) then begin
+                        float_msg(self_obj, mstr(random(173, 175)), FLOAT_MSG_YELLOW);
                     end
                     else begin
-                        if (global_var(562) == 1) then begin
-                            float_msg(self_obj, message_str(613, 172), 8);
+                        if (global_var(GVAR_BROKEN_HILLS_CARAVAN_POOCH_SCREW) == 1) then begin
+                            float_msg(self_obj, mstr(172), FLOAT_MSG_YELLOW);
                         end
                         else begin
-                            if (global_var(330) == 5) then begin
-                                set_global_var(562, 1);
-                                float_msg(self_obj, message_str(613, random(169, 170)), 8);
+                            if (global_var(GVAR_BH_CARAVAN) == 5) then begin
+                                set_global_var(GVAR_BROKEN_HILLS_CARAVAN_POOCH_SCREW, 1);
+                                float_msg(self_obj, mstr(random(169, 170)), FLOAT_MSG_YELLOW);
                             end
                             else begin
-                                if ((global_var(330) == 3) or (global_var(330) == 2)) then begin
-                                    set_global_var(330, 0);
+                                if ((global_var(GVAR_BH_CARAVAN) == 3) or (global_var(GVAR_BH_CARAVAN) == 2)) then begin
+                                    set_global_var(GVAR_BH_CARAVAN, 0);
                                     item_caps_adjust(dude_obj, 200);
                                     add_timer_event(self_obj, 1, 1);
                                 end
                                 else begin
-                                    if (global_var(309) == 1) then begin
+                                    if (global_var(GVAR_ENEMY_BROKEN_HILLS) == 1) then begin
                                         start_gdialog(613, self_obj, 4, -1, -1);
                                         gsay_start;
                                         call Node016();
@@ -527,11 +527,11 @@ begin
                                         else begin
                                             start_gdialog(613, self_obj, 4, -1, -1);
                                             gsay_start;
-                                            if (get_critter_stat(dude_obj, 4) == 1) then begin
+                                            if (dude_iq == 1) then begin
                                                 call Node005();
                                             end
                                             else begin
-                                                if ((get_critter_stat(dude_obj, 4) == 2) or (get_critter_stat(dude_obj, 4) == 3)) then begin
+                                                if ((dude_iq == 2) or (dude_iq == 3)) then begin
                                                     call Node006();
                                                 end
                                                 else begin
@@ -555,65 +555,65 @@ end
 
 procedure destroy_p_proc
 begin
-    set_global_var(563, 1);
+    set_global_var(GVAR_CHAD_DEAD, 1);
     if (source_obj == dude_obj) then begin
-        set_global_var(5, global_var(5) + 1);
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(4) > (3 * global_var(5))) or (global_var(2) == 1)) and (global_var(1) == 0)) then begin
-            set_global_var(2, 1);
-            set_global_var(3, 0);
+        set_global_var(GVAR_GOOD_MONSTER, global_var(GVAR_GOOD_MONSTER) + 1);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(4) > (3 * global_var(GVAR_GOOD_MONSTER))) or (global_var(GVAR_CHAMPION_REPUTATION) == 1)) and (global_var(GVAR_CHILDKILLER_REPUTATION) == 0)) then begin
+            set_global_var(GVAR_CHAMPION_REPUTATION, 1);
+            set_global_var(GVAR_BERSERKER_REPUTATION, 0);
         end
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(5) > (2 * global_var(4))) or (global_var(3) == 1))) then begin
-            set_global_var(3, 1);
-            set_global_var(2, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
+            set_global_var(GVAR_BERSERKER_REPUTATION, 1);
+            set_global_var(GVAR_CHAMPION_REPUTATION, 0);
         end
-        set_global_var(0, global_var(0) + -10);
+        set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + -10);
         debug_msg("Player gains " + -10 + " Karma Points.");
-        set_global_var(37, 0);
-        set_global_var(38, 0);
-        set_global_var(39, 0);
-        set_global_var(40, 0);
-        set_global_var(41, 0);
-        set_global_var(42, 0);
-        set_global_var(43, 0);
-        set_global_var(44, 0);
-        set_global_var(45, 0);
-        tmp_gen_rep := global_var(0);
-        if (has_trait(0, dude_obj, 95)) then begin
-            tmp_gen_rep := global_var(0) * 2;
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+        set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+        set_global_var(GVAR_KARMA_DEFENDER, 0);
+        set_global_var(GVAR_KARMA_WANDERER, 0);
+        set_global_var(GVAR_KARMA_BETRAYER, 0);
+        set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+        set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+        tmp_gen_rep := global_var(GVAR_PLAYER_REPUTATION);
+        if (has_trait(TRAIT_PERK, dude_obj, PERK_karma_beacon_perk)) then begin
+            tmp_gen_rep := global_var(GVAR_PLAYER_REPUTATION) * 2;
         end
         if (tmp_gen_rep >= 1000) then begin
-            set_global_var(37, 1);
+            set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
         end
         else begin
             if (tmp_gen_rep >= 750) then begin
-                set_global_var(38, 1);
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
             end
             else begin
                 if (tmp_gen_rep >= 500) then begin
-                    set_global_var(39, 1);
+                    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                 end
                 else begin
                     if (tmp_gen_rep >= 250) then begin
-                        set_global_var(40, 1);
+                        set_global_var(GVAR_KARMA_DEFENDER, 1);
                     end
                     else begin
                         if (tmp_gen_rep > -250) then begin
-                            set_global_var(41, 1);
+                            set_global_var(GVAR_KARMA_WANDERER, 1);
                         end
                         else begin
                             if (tmp_gen_rep > -500) then begin
-                                set_global_var(42, 1);
+                                set_global_var(GVAR_KARMA_BETRAYER, 1);
                             end
                             else begin
                                 if (tmp_gen_rep > -750) then begin
-                                    set_global_var(43, 1);
+                                    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                 end
                                 else begin
                                     if (tmp_gen_rep > -1000) then begin
-                                        set_global_var(44, 1);
+                                        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                     end
                                     else begin
-                                        set_global_var(45, 1);
+                                        set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                     end
                                 end
                             end
@@ -622,7 +622,7 @@ begin
                 end
             end
         end
-        set_global_var(54, global_var(54) + -5);
+        set_global_var(GVAR_TOWN_REP_BROKEN_HILLS, global_var(GVAR_TOWN_REP_BROKEN_HILLS) + -5);
         debug_msg("Added " + -5 + " to Town Rep");
     end
 end
@@ -630,13 +630,13 @@ end
 procedure look_at_p_proc
 begin
     script_overrides;
-    display_msg(message_str(613, 100));
+    display_mstr(100);
 end
 
 procedure description_p_proc
 begin
     script_overrides;
-    display_msg(message_str(613, 101));
+    display_mstr(101);
 end
 
 procedure use_skill_on_p_proc
@@ -648,45 +648,45 @@ begin
     if (source_obj == dude_obj) then begin
         set_local_var(6, 1);
         if (local_var(7) == 0) then begin
-            set_global_var(309, 1);
+            set_global_var(GVAR_ENEMY_BROKEN_HILLS, 1);
         end
     end
 end
 
 procedure map_enter_p_proc
 begin
-    if ((cur_map_index == 78) or (cur_map_index == 79)) then begin
+    if ((CUR_MAP_BROKEN1) or (CUR_MAP_BROKEN2)) then begin
         Chad_Ptr := self_obj;
     end
     Only_Once := 0;
-    if (cur_map_index == 94) then begin
-        critter_add_trait(self_obj, 1, 6, 0);
+    if (CUR_MAP_BHRNDDST) then begin
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 0);
     end
     else begin
-        critter_add_trait(self_obj, 1, 6, 120);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 120);
     end
-    critter_add_trait(self_obj, 1, 5, 24);
-    if ((global_var(563) == 1) or (map_var(25) == 2)) then begin
-        if (has_trait(1, self_obj, 666)) then begin
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_TOUGH_CITIZEN);
+    if ((global_var(GVAR_CHAD_DEAD) == 1) or (map_var(25) == 2)) then begin
+        if (obj_is_visible_flag(self_obj)) then begin
             set_obj_visibility(self_obj, 1);
         end
     end
     else begin
-        if (not(metarule(22, 0))) then begin
-            if (cur_map_index == 78) then begin
-                if (not(has_trait(1, self_obj, 666))) then begin
+        if (not(is_loading_game)) then begin
+            if (CUR_MAP_BROKEN1) then begin
+                if (not(obj_is_visible_flag(self_obj))) then begin
                     set_obj_visibility(self_obj, 0);
                 end
             end
             else begin
-                if ((cur_map_index == 31) or (cur_map_index == 46) or (cur_map_index == 15)) then begin
-                    if ((global_var(330) == 3) or (global_var(330) == 2) and (global_var(300) != 7) and (global_var(563) == 0)) then begin
-                        if (not(has_trait(1, self_obj, 666))) then begin
+                if ((CUR_MAP_GECKSETL) or (CUR_MAP_NCRENT) or (CUR_MAP_VCTYCTYD)) then begin
+                    if ((global_var(GVAR_BH_CARAVAN) == 3) or (global_var(GVAR_BH_CARAVAN) == 2) and (global_var(GVAR_BH_CHAD) != 7) and (global_var(GVAR_CHAD_DEAD) == 0)) then begin
+                        if (not(obj_is_visible_flag(self_obj))) then begin
                             set_obj_visibility(self_obj, 0);
                         end
                     end
                     else begin
-                        if (has_trait(1, self_obj, 666)) then begin
+                        if (obj_is_visible_flag(self_obj)) then begin
                             set_obj_visibility(self_obj, 1);
                         end
                     end
@@ -694,7 +694,7 @@ begin
             end
         end
     end
-    if (global_var(563) == 1) then begin
+    if (global_var(GVAR_CHAD_DEAD) == 1) then begin
         destroy_object(self_obj);
     end
 end
@@ -708,7 +708,7 @@ begin
     variable LVar0 := 0;
     if (fixed_param == 1) then begin
         game_ui_disable;
-        float_msg(self_obj, message_str(613, 171), 8);
+        float_msg(self_obj, mstr(171), FLOAT_MSG_YELLOW);
         add_timer_event(self_obj, game_ticks(2), 2);
     end
     else begin
@@ -723,57 +723,57 @@ begin
         end
         else begin
             if (fixed_param == 3) then begin
-                display_msg(message_str(613, 300));
+                display_mstr(300);
                 give_exp_points(750);
-                display_msg(message_str(14, 100) + 750 + message_str(14, 101));
-                set_global_var(0, global_var(0) + -5);
+                display_msg(g_mstr(100) + 750 + g_mstr(101));
+                set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + -5);
                 debug_msg("Player gains " + -5 + " Karma Points.");
-                set_global_var(37, 0);
-                set_global_var(38, 0);
-                set_global_var(39, 0);
-                set_global_var(40, 0);
-                set_global_var(41, 0);
-                set_global_var(42, 0);
-                set_global_var(43, 0);
-                set_global_var(44, 0);
-                set_global_var(45, 0);
-                tmp_gen_rep := global_var(0);
-                if (has_trait(0, dude_obj, 95)) then begin
-                    tmp_gen_rep := global_var(0) * 2;
+                set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+                set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+                set_global_var(GVAR_KARMA_DEFENDER, 0);
+                set_global_var(GVAR_KARMA_WANDERER, 0);
+                set_global_var(GVAR_KARMA_BETRAYER, 0);
+                set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+                set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+                set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+                tmp_gen_rep := global_var(GVAR_PLAYER_REPUTATION);
+                if (has_trait(TRAIT_PERK, dude_obj, PERK_karma_beacon_perk)) then begin
+                    tmp_gen_rep := global_var(GVAR_PLAYER_REPUTATION) * 2;
                 end
                 if (tmp_gen_rep >= 1000) then begin
-                    set_global_var(37, 1);
+                    set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
                 end
                 else begin
                     if (tmp_gen_rep >= 750) then begin
-                        set_global_var(38, 1);
+                        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
                     end
                     else begin
                         if (tmp_gen_rep >= 500) then begin
-                            set_global_var(39, 1);
+                            set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                         end
                         else begin
                             if (tmp_gen_rep >= 250) then begin
-                                set_global_var(40, 1);
+                                set_global_var(GVAR_KARMA_DEFENDER, 1);
                             end
                             else begin
                                 if (tmp_gen_rep > -250) then begin
-                                    set_global_var(41, 1);
+                                    set_global_var(GVAR_KARMA_WANDERER, 1);
                                 end
                                 else begin
                                     if (tmp_gen_rep > -500) then begin
-                                        set_global_var(42, 1);
+                                        set_global_var(GVAR_KARMA_BETRAYER, 1);
                                     end
                                     else begin
                                         if (tmp_gen_rep > -750) then begin
-                                            set_global_var(43, 1);
+                                            set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                         end
                                         else begin
                                             if (tmp_gen_rep > -1000) then begin
-                                                set_global_var(44, 1);
+                                                set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                             end
                                             else begin
-                                                set_global_var(45, 1);
+                                                set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                             end
                                         end
                                     end
@@ -785,7 +785,7 @@ begin
                 gfade_out(600);
                 set_obj_visibility(self_obj, 1);
                 gfade_in(600);
-                set_global_var(300, 9);
+                set_global_var(GVAR_BH_CHAD, 9);
             end
             else begin
                 if (fixed_param == 4) then begin
@@ -800,24 +800,24 @@ end
 
 procedure map_exit_p_proc
 begin
-    if ((cur_map_index == 31) or (cur_map_index == 46) or (cur_map_index == 15)) then begin
-        if ((global_var(330) == 3) or (global_var(330) == 2)) then begin
-            set_global_var(330, 0);
+    if ((CUR_MAP_GECKSETL) or (CUR_MAP_NCRENT) or (CUR_MAP_VCTYCTYD)) then begin
+        if ((global_var(GVAR_BH_CARAVAN) == 3) or (global_var(GVAR_BH_CARAVAN) == 2)) then begin
+            set_global_var(GVAR_BH_CARAVAN, 0);
         end
     end
-    if (global_var(300) == 9) then begin
+    if (global_var(GVAR_BH_CHAD) == 9) then begin
         destroy_object(self_obj);
     end
 end
 
 procedure map_update_p_proc
 begin
-    if ((cur_map_index == 78) or (cur_map_index == 79)) then begin
+    if ((CUR_MAP_BROKEN1) or (CUR_MAP_BROKEN2)) then begin
         Chad_Ptr := self_obj;
     end
-    if (cur_map_index == 94) then begin
+    if (CUR_MAP_BHRNDDST) then begin
         if (map_var(1) <= 0) then begin
-            float_msg(self_obj, message_str(14, 450), 8);
+            float_msg(self_obj, g_mstr(450), FLOAT_MSG_YELLOW);
         end
     end
 end
@@ -832,7 +832,7 @@ end
 
 procedure Node995
 begin
-    if (get_critter_stat(dude_obj, 3) >= 7) then begin
+    if (dude_charisma >= 7) then begin
         set_local_var(3, 100);
         set_local_var(1, 4);
     end
@@ -850,7 +850,7 @@ end
 
 procedure Node997
 begin
-    if (get_critter_stat(dude_obj, 3) >= 8) then begin
+    if (dude_charisma >= 8) then begin
         call Node004();
     end
     else begin
@@ -883,151 +883,151 @@ procedure Node004
 begin
     from4 := 1;
     if (fromWTG) then begin
-        gsay_reply(613, 120);
+        Reply(120);
     end
     else begin
-        gsay_reply(613, 121);
+        Reply(121);
     end
-    giq_option(-1, 613, 122, Node005, 50);
-    if ((get_critter_stat(dude_obj, 4) > 1) and (get_critter_stat(dude_obj, 4) < 3)) then begin
-        giq_option(4, 613, 123, Node006, 50);
+    giq_option(-1, NAME, 122, Node005, NEUTRAL_REACTION);
+    if ((dude_iq > 1) and (dude_iq < 3)) then begin
+        giq_option(4, NAME, 123, Node006, NEUTRAL_REACTION);
     end
-    if ((global_var(300) >= 1) and (global_var(300) <= 4)) then begin
-        giq_option(4, 613, 124, Node008, 50);
+    if ((global_var(GVAR_BH_CHAD) >= 1) and (global_var(GVAR_BH_CHAD) <= 4)) then begin
+        giq_option(4, NAME, 124, Node008, NEUTRAL_REACTION);
     end
-    if (global_var(300) == 5) then begin
-        giq_option(4, 613, 126, Node022, 50);
+    if (global_var(GVAR_BH_CHAD) == 5) then begin
+        giq_option(4, NAME, 126, Node022, NEUTRAL_REACTION);
     end
-    giq_option(4, 613, 127, Node006, 50);
-    giq_option(4, 613, 128, Node007, 50);
-    giq_option(4, 613, 129, Node999, 50);
+    giq_option(4, NAME, 127, Node006, NEUTRAL_REACTION);
+    giq_option(4, NAME, 128, Node007, NEUTRAL_REACTION);
+    giq_option(4, NAME, 129, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node005
 begin
-    gsay_reply(613, 130);
-    giq_option(1, 613, 131, Node999, 50);
+    Reply(130);
+    giq_option(1, NAME, 131, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node006
 begin
     fromWTG := 0;
     if (from4) then begin
-        gsay_reply(613, 132);
+        Reply(132);
     end
     else begin
-        if (get_critter_stat(dude_obj, 4) < 4) then begin
-            gsay_reply(613, 133);
+        if (dude_iq < 4) then begin
+            Reply(133);
         end
         else begin
-            gsay_reply(613, 134);
-            giq_option(4, 613, 137, Node013, 49);
-            giq_option(4, 613, 138, Node014, 51);
+            Reply(134);
+            giq_option(4, NAME, 137, Node013, GOOD_REACTION);
+            giq_option(4, NAME, 138, Node014, BAD_REACTION);
         end
     end
-    if ((get_critter_stat(dude_obj, 4) > 1) and (get_critter_stat(dude_obj, 4) < 3)) then begin
-        giq_option(2, 613, 135, Node999, 50);
+    if ((dude_iq > 1) and (dude_iq < 3)) then begin
+        giq_option(2, NAME, 135, Node999, NEUTRAL_REACTION);
     end
-    giq_option(4, 613, 136, Node012, 50);
-    giq_option(4, 613, 139, Node004, 50);
-    giq_option(4, 613, 140, Node999, 50);
+    giq_option(4, NAME, 136, Node012, NEUTRAL_REACTION);
+    giq_option(4, NAME, 139, Node004, NEUTRAL_REACTION);
+    giq_option(4, NAME, 140, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node007
 begin
     fromWTG := 0;
-    gsay_reply(613, 141);
-    giq_option(4, 613, 142, Node004, 50);
-    giq_option(4, 613, 143, Node999, 50);
+    Reply(141);
+    giq_option(4, NAME, 142, Node004, NEUTRAL_REACTION);
+    giq_option(4, NAME, 143, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node008
 begin
     fromWTG := 0;
-    gsay_reply(613, 144);
-    giq_option(4, 613, 145, Node009, 50);
-    giq_option(4, 613, 146, Node999, 50);
+    Reply(144);
+    giq_option(4, NAME, 145, Node009, NEUTRAL_REACTION);
+    giq_option(4, NAME, 146, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node009
 begin
-    gsay_reply(613, 147);
-    giq_option(4, 613, 150, Node999, 50);
+    Reply(147);
+    giq_option(4, NAME, 150, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node010
 begin
-    gsay_reply(613, 151);
-    giq_option(4, 613, 152, combat, 50);
+    Reply(151);
+    giq_option(4, NAME, 152, combat, NEUTRAL_REACTION);
 end
 
 procedure Node011
 begin
-    gsay_reply(613, 153);
-    giq_option(4, 613, 154, combat, 50);
-    giq_option(4, 613, 155, Node999, 50);
+    Reply(153);
+    giq_option(4, NAME, 154, combat, NEUTRAL_REACTION);
+    giq_option(4, NAME, 155, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node012
 begin
     fromWTG := 0;
-    gsay_reply(613, 156);
-    giq_option(4, 613, 157, Node004, 51);
-    giq_option(4, 613, 158, Node999, 50);
+    Reply(156);
+    giq_option(4, NAME, 157, Node004, BAD_REACTION);
+    giq_option(4, NAME, 158, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node013
 begin
-    gsay_reply(613, 159);
-    giq_option(4, 613, 160, Node012, 50);
-    giq_option(4, 613, 161, Node014, 51);
-    giq_option(4, 613, 163, Node999, 50);
+    Reply(159);
+    giq_option(4, NAME, 160, Node012, NEUTRAL_REACTION);
+    giq_option(4, NAME, 161, Node014, BAD_REACTION);
+    giq_option(4, NAME, 163, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node014
 begin
     fromWTG := 0;
-    gsay_reply(613, 164);
-    giq_option(4, 613, 165, Node004, 51);
-    giq_option(4, 613, 166, Node999, 50);
+    Reply(164);
+    giq_option(4, NAME, 165, Node004, BAD_REACTION);
+    giq_option(4, NAME, 166, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node015
 begin
-    float_msg(self_obj, message_str(613, 167), 8);
+    float_msg(self_obj, mstr(167), FLOAT_MSG_YELLOW);
 end
 
 procedure Node016
 begin
-    float_msg(self_obj, message_str(613, 168), 8);
+    float_msg(self_obj, mstr(168), FLOAT_MSG_YELLOW);
 end
 
 procedure Node017
 begin
-    gsay_reply(613, 200);
-    giq_option(4, 613, 201, Node023, 50);
-    giq_option(4, 613, 203, Node999, 50);
+    Reply(200);
+    giq_option(4, NAME, 201, Node023, NEUTRAL_REACTION);
+    giq_option(4, NAME, 203, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node018
 begin
-    gsay_reply(613, 202);
-    critter_add_trait(self_obj, 1, 5, 24);
+    Reply(202);
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_TOUGH_CITIZEN);
     set_local_var(6, 1);
     set_local_var(7, 2);
-    giq_option(4, 613, 205, Node998, 50);
-    float_msg(self_obj, message_str(613, 206), 8);
+    giq_option(4, NAME, 205, Node998, NEUTRAL_REACTION);
+    float_msg(self_obj, mstr(206), FLOAT_MSG_YELLOW);
 end
 
 procedure Node019
 begin
     if (not(obj_is_carrying_obj_pid(dude_obj, 72))) then begin
-        gsay_reply(613, 204);
+        Reply(204);
     end
     else begin
-        gsay_reply(613, 2040);
+        Reply(2040);
     end
-    giq_option(4, 613, 205, Node020, 50);
+    giq_option(4, NAME, 205, Node020, NEUTRAL_REACTION);
 end
 
 procedure Node020
@@ -1044,17 +1044,17 @@ begin
         restock_obj := obj_carrying_pid_obj(dude_obj, 72);
         if (obj_type(self_obj) == 1) then begin
             restock_obj := obj_carrying_pid_obj(dude_obj, 72);
-            if (critter_inven_obj(self_obj, 0) == restock_obj) then begin
-                if (obj_item_subtype(critter_inven_obj(dude_obj, 0)) == 0) then begin
-                    restock_obj := critter_inven_obj(dude_obj, 0);
+            if (critter_inven_obj(self_obj, INVEN_TYPE_WORN) == restock_obj) then begin
+                if (obj_item_subtype(critter_inven_obj(dude_obj, INVEN_TYPE_WORN)) == 0) then begin
+                    restock_obj := critter_inven_obj(dude_obj, INVEN_TYPE_WORN);
                     debug_msg("armour pid == " + obj_pid(restock_obj));
                     rm_obj_from_inven(dude_obj, restock_obj);
                     add_obj_to_inven(dude_obj, restock_obj);
                 end
             end
             else begin
-                if ((critter_inven_obj(dude_obj, 2) == restock_obj) or (critter_inven_obj(dude_obj, 1) == restock_obj)) then begin
-                    metarule(43, dude_obj);
+                if ((critter_inven_obj(dude_obj, INVEN_TYPE_LEFT_HAND) == restock_obj) or (critter_inven_obj(dude_obj, INVEN_TYPE_RIGHT_HAND) == restock_obj)) then begin
+                    inven_unwield(dude_obj);
                 end
             end
         end
@@ -1067,11 +1067,11 @@ end
 
 procedure Node021
 begin
-    float_msg(self_obj, message_str(613, 206), 8);
+    float_msg(self_obj, mstr(206), FLOAT_MSG_YELLOW);
     call Node992();
     set_local_var(5, 1);
     set_local_var(6, 1);
-    attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+    attack(dude_obj);
 end
 
 procedure Node022
@@ -1111,69 +1111,68 @@ end
 
 procedure Node026
 begin
-    gsay_reply(613, 208);
-    giq_option(4, 613, 209, Node027, 50);
+    Reply(208);
+    giq_option(4, NAME, 209, Node027, NEUTRAL_REACTION);
 end
 
 procedure Node027
 begin
-    gsay_reply(613, 210);
-    if (global_var(1132) == 1) then begin
-        giq_option(4, 613, 211, Node028, 50);
+    Reply(210);
+    if (global_var(GVAR_RESERVED) == 1) then begin
+        giq_option(4, NAME, 211, Node028, NEUTRAL_REACTION);
     end
     else begin
-        giq_option(4, 613, 219, Node032, 50);
+        giq_option(4, NAME, 219, Node032, NEUTRAL_REACTION);
     end
 end
 
 procedure Node028
 begin
-    gsay_reply(613, 212);
-    giq_option(4, 613, 213, Node029, 50);
+    Reply(212);
+    giq_option(4, NAME, 213, Node029, NEUTRAL_REACTION);
 end
 
 procedure Node029
 begin
-    gsay_reply(613, 214);
-    giq_option(4, 613, 215, Node030, 50);
+    Reply(214);
+    giq_option(4, NAME, 215, Node030, NEUTRAL_REACTION);
 end
 
 procedure Node030
 begin
-    gsay_reply(613, 216);
-    giq_option(4, 613, 217, Node031, 50);
+    Reply(216);
+    giq_option(4, NAME, 217, Node031, NEUTRAL_REACTION);
 end
 
 procedure Node031
 begin
-    gsay_reply(613, 218);
-    giq_option(4, 613, message_str(14, 350), Node991, 50);
+    Reply(218);
+    giq_option(4, NAME, g_mstr(350), Node991, NEUTRAL_REACTION);
 end
 
 procedure Node032
 begin
-    gsay_reply(613, message_str(613, 220) + " " + message_str(613, 216));
-    giq_option(4, 613, 217, Node031, 50);
+    Reply(mstr(220) + " " + mstr(216));
+    giq_option(4, NAME, 217, Node031, NEUTRAL_REACTION);
 end
 
 procedure Node033
 begin
-    set_global_var(300, 5);
-    gsay_reply(613, 221);
-    giq_option(4, 613, 222, Node999, 50);
+    set_global_var(GVAR_BH_CHAD, 5);
+    Reply(221);
+    giq_option(4, NAME, 222, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node991
 begin
-    set_global_var(300, 5);
+    set_global_var(GVAR_BH_CHAD, 5);
     set_local_var(5, 1);
     set_local_var(6, 1);
-    attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+    attack(dude_obj);
 end
 
 procedure Node992
 begin
-    critter_add_trait(self_obj, 1, 5, 24);
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_TOUGH_CITIZEN);
     set_local_var(7, 2);
 end
-
