diff --git a/Mods/MegaMod/Original/HCFRANK.ssl b/Mods/MegaMod/ReDefine/HCFRANK.ssl
index 07391f7..9dba05e 100644
--- a/Mods/MegaMod/Original/HCFRANK.ssl
+++ b/Mods/MegaMod/ReDefine/HCFRANK.ssl
@@ -60,75 +60,75 @@ begin
     variable LVar0 := 0;
     variable LVar1 := 0;
     variable LVar2 := 0;
-    if ((global_var(525) == 1) or (local_var(5) == 2) and obj_can_see_obj(self_obj, dude_obj)) then begin
+    if ((global_var(GVAR_ENEMY_CONSPIRATORS) == 1) or (local_var(5) == 2) and obj_can_see_obj(self_obj, dude_obj)) then begin
         set_local_var(5, 1);
-        attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+        attack(dude_obj);
     end
     else begin
-        if (global_var(443) == 1) then begin
-            if (global_var(443) != 2) then begin
-                set_global_var(443, 2);
-                if ((global_var(303) == 3) or (global_var(303) == 2)) then begin
-                    set_global_var(303, 4);
+        if (global_var(GVAR_JAIL_BREAK) == 1) then begin
+            if (global_var(GVAR_JAIL_BREAK) != 2) then begin
+                set_global_var(GVAR_JAIL_BREAK, 2);
+                if ((global_var(GVAR_BH_JAIL) == 3) or (global_var(GVAR_BH_JAIL) == 2)) then begin
+                    set_global_var(GVAR_BH_JAIL, 4);
                 end
                 else begin
-                    if (global_var(303) == 0) then begin
-                        set_global_var(303, 8);
+                    if (global_var(GVAR_BH_JAIL) == 0) then begin
+                        set_global_var(GVAR_BH_JAIL, 8);
                     end
                 end
             end
-            display_msg(message_str(595, 153));
+            display_mstr(153);
             give_exp_points(1500);
-            display_msg(message_str(14, 100) + 1500 + message_str(14, 101));
-            set_global_var(54, global_var(54) + -10);
-            set_global_var(0, global_var(0) + -10);
+            display_msg(g_mstr(100) + 1500 + g_mstr(101));
+            set_global_var(GVAR_TOWN_REP_BROKEN_HILLS, global_var(GVAR_TOWN_REP_BROKEN_HILLS) + -10);
+            set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + -10);
             debug_msg("Player gains " + -10 + " Karma Points.");
-            set_global_var(37, 0);
-            set_global_var(38, 0);
-            set_global_var(39, 0);
-            set_global_var(40, 0);
-            set_global_var(41, 0);
-            set_global_var(42, 0);
-            set_global_var(43, 0);
-            set_global_var(44, 0);
-            set_global_var(45, 0);
-            tmp_gen_rep := global_var(0);
-            if (has_trait(0, dude_obj, 95)) then begin
-                tmp_gen_rep := global_var(0) * 2;
+            set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+            set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+            set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+            set_global_var(GVAR_KARMA_DEFENDER, 0);
+            set_global_var(GVAR_KARMA_WANDERER, 0);
+            set_global_var(GVAR_KARMA_BETRAYER, 0);
+            set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+            set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+            set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+            tmp_gen_rep := global_var(GVAR_PLAYER_REPUTATION);
+            if (has_trait(TRAIT_PERK, dude_obj, PERK_karma_beacon_perk)) then begin
+                tmp_gen_rep := global_var(GVAR_PLAYER_REPUTATION) * 2;
             end
             if (tmp_gen_rep >= 1000) then begin
-                set_global_var(37, 1);
+                set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
             end
             else begin
                 if (tmp_gen_rep >= 750) then begin
-                    set_global_var(38, 1);
+                    set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
                 end
                 else begin
                     if (tmp_gen_rep >= 500) then begin
-                        set_global_var(39, 1);
+                        set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                     end
                     else begin
                         if (tmp_gen_rep >= 250) then begin
-                            set_global_var(40, 1);
+                            set_global_var(GVAR_KARMA_DEFENDER, 1);
                         end
                         else begin
                             if (tmp_gen_rep > -250) then begin
-                                set_global_var(41, 1);
+                                set_global_var(GVAR_KARMA_WANDERER, 1);
                             end
                             else begin
                                 if (tmp_gen_rep > -500) then begin
-                                    set_global_var(42, 1);
+                                    set_global_var(GVAR_KARMA_BETRAYER, 1);
                                 end
                                 else begin
                                     if (tmp_gen_rep > -750) then begin
-                                        set_global_var(43, 1);
+                                        set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                     end
                                     else begin
                                         if (tmp_gen_rep > -1000) then begin
-                                            set_global_var(44, 1);
+                                            set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                         end
                                         else begin
-                                            set_global_var(45, 1);
+                                            set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                         end
                                     end
                                 end
@@ -138,7 +138,7 @@ begin
                 end
             end
             game_ui_disable;
-            float_msg(self_obj, message_str(595, 152), 8);
+            float_msg(self_obj, mstr(152), FLOAT_MSG_YELLOW);
             add_timer_event(self_obj, game_ticks(3), 1);
         end
     end
@@ -159,16 +159,16 @@ procedure talk_p_proc
 begin
     if (local_var(2) == 0) then begin
         set_local_var(2, 1);
-        set_local_var(3, (get_critter_stat(dude_obj, 3) - get_critter_stat(self_obj, 3)) * 5);
+        set_local_var(3, (dude_charisma - self_charisma) * 5);
         debug_msg("Base Reaction == " + local_var(3));
     end
     Static_Reaction := 0;
     set_local_var(0, 0);
     debug_msg("Start Reaction == " + local_var(0));
     reaction_bonus_karma := floor(reaction_bonus_karma / 1000);
-    reaction_bonus_town_rep := floor(global_var(54) / 2);
-    if (has_trait(2, dude_obj, 13)) then begin
-        if (get_critter_stat(dude_obj, 34) == get_critter_stat(self_obj, 34)) then begin
+    reaction_bonus_town_rep := floor(global_var(GVAR_TOWN_REP_BROKEN_HILLS) / 2);
+    if (has_trait(TRAIT_TRAIT, dude_obj, TRAIT_sex_appeal)) then begin
+        if (dude_gender == self_gender) then begin
             Static_Reaction := Static_Reaction + -20;
         end
         else begin
@@ -194,52 +194,52 @@ begin
             Static_Reaction := Static_Reaction + reaction_bonus_town_rep;
         end
         debug_msg("Town Rep Bonus == " + Static_Reaction);
-        set_global_var(37, 0);
-        set_global_var(38, 0);
-        set_global_var(39, 0);
-        set_global_var(40, 0);
-        set_global_var(41, 0);
-        set_global_var(42, 0);
-        set_global_var(43, 0);
-        set_global_var(44, 0);
-        set_global_var(45, 0);
-        tmp_gen_rep := global_var(0);
-        if (has_trait(0, dude_obj, 95)) then begin
-            tmp_gen_rep := global_var(0) * 2;
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+        set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+        set_global_var(GVAR_KARMA_DEFENDER, 0);
+        set_global_var(GVAR_KARMA_WANDERER, 0);
+        set_global_var(GVAR_KARMA_BETRAYER, 0);
+        set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+        set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+        tmp_gen_rep := global_var(GVAR_PLAYER_REPUTATION);
+        if (has_trait(TRAIT_PERK, dude_obj, PERK_karma_beacon_perk)) then begin
+            tmp_gen_rep := global_var(GVAR_PLAYER_REPUTATION) * 2;
         end
         if (tmp_gen_rep >= 1000) then begin
-            set_global_var(37, 1);
+            set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
         end
         else begin
             if (tmp_gen_rep >= 750) then begin
-                set_global_var(38, 1);
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
             end
             else begin
                 if (tmp_gen_rep >= 500) then begin
-                    set_global_var(39, 1);
+                    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                 end
                 else begin
                     if (tmp_gen_rep >= 250) then begin
-                        set_global_var(40, 1);
+                        set_global_var(GVAR_KARMA_DEFENDER, 1);
                     end
                     else begin
                         if (tmp_gen_rep > -250) then begin
-                            set_global_var(41, 1);
+                            set_global_var(GVAR_KARMA_WANDERER, 1);
                         end
                         else begin
                             if (tmp_gen_rep > -500) then begin
-                                set_global_var(42, 1);
+                                set_global_var(GVAR_KARMA_BETRAYER, 1);
                             end
                             else begin
                                 if (tmp_gen_rep > -750) then begin
-                                    set_global_var(43, 1);
+                                    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                 end
                                 else begin
                                     if (tmp_gen_rep > -1000) then begin
-                                        set_global_var(44, 1);
+                                        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                     end
                                     else begin
-                                        set_global_var(45, 1);
+                                        set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                     end
                                 end
                             end
@@ -248,35 +248,35 @@ begin
                 end
             end
         end
-        if (global_var(37) == 1) then begin
+        if (global_var(GVAR_KARMA_HOLY_WARRIOR) == 1) then begin
             Static_Reaction := Static_Reaction - 20;
         end
         else begin
-            if (global_var(38) == 1) then begin
+            if (global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES) == 1) then begin
                 Static_Reaction := Static_Reaction - 15;
             end
             else begin
-                if (global_var(39) == 1) then begin
+                if (global_var(GVAR_KARMA_SHIELD_OF_HOPE) == 1) then begin
                     Static_Reaction := Static_Reaction - 10;
                 end
                 else begin
-                    if (global_var(40) == 1) then begin
+                    if (global_var(GVAR_KARMA_DEFENDER) == 1) then begin
                         Static_Reaction := Static_Reaction - 5;
                     end
                     else begin
-                        if (global_var(41) == 1) then begin
+                        if (global_var(GVAR_KARMA_WANDERER) == 1) then begin
                             Static_Reaction := Static_Reaction - 0;
                         end
                         else begin
-                            if (global_var(42) == 1) then begin
+                            if (global_var(GVAR_KARMA_BETRAYER) == 1) then begin
                                 Static_Reaction := Static_Reaction + 5;
                             end
                             else begin
-                                if (global_var(43) == 1) then begin
+                                if (global_var(GVAR_KARMA_SWORD_OF_DESPAIR) == 1) then begin
                                     Static_Reaction := Static_Reaction + 10;
                                 end
                                 else begin
-                                    if (global_var(44) == 1) then begin
+                                    if (global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE) == 1) then begin
                                         Static_Reaction := Static_Reaction + 15;
                                     end
                                     else begin
@@ -296,52 +296,52 @@ begin
         debug_msg("Karma Reaction Bonus == " + Static_Reaction);
         Static_Reaction := Static_Reaction + reaction_bonus_town_rep;
         debug_msg("Town Rep Bonus == " + Static_Reaction);
-        set_global_var(37, 0);
-        set_global_var(38, 0);
-        set_global_var(39, 0);
-        set_global_var(40, 0);
-        set_global_var(41, 0);
-        set_global_var(42, 0);
-        set_global_var(43, 0);
-        set_global_var(44, 0);
-        set_global_var(45, 0);
-        tmp_gen_rep := global_var(0);
-        if (has_trait(0, dude_obj, 95)) then begin
-            tmp_gen_rep := global_var(0) * 2;
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+        set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+        set_global_var(GVAR_KARMA_DEFENDER, 0);
+        set_global_var(GVAR_KARMA_WANDERER, 0);
+        set_global_var(GVAR_KARMA_BETRAYER, 0);
+        set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+        set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+        tmp_gen_rep := global_var(GVAR_PLAYER_REPUTATION);
+        if (has_trait(TRAIT_PERK, dude_obj, PERK_karma_beacon_perk)) then begin
+            tmp_gen_rep := global_var(GVAR_PLAYER_REPUTATION) * 2;
         end
         if (tmp_gen_rep >= 1000) then begin
-            set_global_var(37, 1);
+            set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
         end
         else begin
             if (tmp_gen_rep >= 750) then begin
-                set_global_var(38, 1);
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
             end
             else begin
                 if (tmp_gen_rep >= 500) then begin
-                    set_global_var(39, 1);
+                    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                 end
                 else begin
                     if (tmp_gen_rep >= 250) then begin
-                        set_global_var(40, 1);
+                        set_global_var(GVAR_KARMA_DEFENDER, 1);
                     end
                     else begin
                         if (tmp_gen_rep > -250) then begin
-                            set_global_var(41, 1);
+                            set_global_var(GVAR_KARMA_WANDERER, 1);
                         end
                         else begin
                             if (tmp_gen_rep > -500) then begin
-                                set_global_var(42, 1);
+                                set_global_var(GVAR_KARMA_BETRAYER, 1);
                             end
                             else begin
                                 if (tmp_gen_rep > -750) then begin
-                                    set_global_var(43, 1);
+                                    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                 end
                                 else begin
                                     if (tmp_gen_rep > -1000) then begin
-                                        set_global_var(44, 1);
+                                        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                     end
                                     else begin
-                                        set_global_var(45, 1);
+                                        set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                     end
                                 end
                             end
@@ -350,35 +350,35 @@ begin
                 end
             end
         end
-        if (global_var(37) == 1) then begin
+        if (global_var(GVAR_KARMA_HOLY_WARRIOR) == 1) then begin
             Static_Reaction := Static_Reaction + 20;
         end
         else begin
-            if (global_var(38) == 1) then begin
+            if (global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES) == 1) then begin
                 Static_Reaction := Static_Reaction + 15;
             end
             else begin
-                if (global_var(39) == 1) then begin
+                if (global_var(GVAR_KARMA_SHIELD_OF_HOPE) == 1) then begin
                     Static_Reaction := Static_Reaction + 10;
                 end
                 else begin
-                    if (global_var(40) == 1) then begin
+                    if (global_var(GVAR_KARMA_DEFENDER) == 1) then begin
                         Static_Reaction := Static_Reaction + 5;
                     end
                     else begin
-                        if (global_var(41) == 1) then begin
+                        if (global_var(GVAR_KARMA_WANDERER) == 1) then begin
                             Static_Reaction := Static_Reaction + 0;
                         end
                         else begin
-                            if (global_var(42) == 1) then begin
+                            if (global_var(GVAR_KARMA_BETRAYER) == 1) then begin
                                 Static_Reaction := Static_Reaction + -5;
                             end
                             else begin
-                                if (global_var(43) == 1) then begin
+                                if (global_var(GVAR_KARMA_SWORD_OF_DESPAIR) == 1) then begin
                                     Static_Reaction := Static_Reaction + -10;
                                 end
                                 else begin
-                                    if (global_var(44) == 1) then begin
+                                    if (global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE) == 1) then begin
                                         Static_Reaction := Static_Reaction + -15;
                                     end
                                     else begin
@@ -394,32 +394,32 @@ begin
         debug_msg("Good Person Reaction Bonus == " + Static_Reaction);
     end
     if (Slavery_Tolerant == 0) then begin
-        if (global_var(11) == 1) then begin
+        if (global_var(GVAR_REPUTATION_SLAVER) == 1) then begin
             Static_Reaction := Static_Reaction - 50;
         end
     end
     else begin
         if (Slavery_Tolerant == 1) then begin
-            if (global_var(11) == 1) then begin
+            if (global_var(GVAR_REPUTATION_SLAVER) == 1) then begin
                 Static_Reaction := Static_Reaction - 25;
             end
         end
         else begin
             if (Slavery_Tolerant == 3) then begin
-                if (global_var(11) == 1) then begin
+                if (global_var(GVAR_REPUTATION_SLAVER) == 1) then begin
                     Static_Reaction := Static_Reaction + 25;
                 end
             end
         end
     end
     debug_msg("Slaver + Aligned Reaction == " + Static_Reaction);
-    Static_Reaction := Static_Reaction + (10 * has_trait(0, dude_obj, 10));
+    Static_Reaction := Static_Reaction + (10 * has_trait(TRAIT_PERK, dude_obj, PERK_presence));
     debug_msg("Presence Reaction == " + Static_Reaction);
-    if (global_var(1) >= 2) then begin
+    if (global_var(GVAR_CHILDKILLER_REPUTATION) >= 2) then begin
         Static_Reaction := Static_Reaction + -30;
     end
     debug_msg("Childkiller Reaction == " + Static_Reaction);
-    if (has_trait(0, dude_obj, 39)) then begin
+    if (has_trait(TRAIT_PERK, dude_obj, PERK_cult_of_personality)) then begin
         if (Evil_Critter == 1) then begin
             if (Static_Reaction > 0) then begin
                 Static_Reaction := Static_Reaction * -1;
@@ -478,19 +478,19 @@ begin
         call Node998();
     end
     else begin
-        if (global_var(443) == 1) then begin
-            float_msg(self_obj, message_str(595, 148), 8);
+        if (global_var(GVAR_JAIL_BREAK) == 1) then begin
+            float_msg(self_obj, mstr(148), FLOAT_MSG_YELLOW);
         end
         else begin
-            if ((global_var(304) == 7) or (global_var(303) == 7)) then begin
+            if ((global_var(GVAR_BH_CONSPIRACY) == 7) or (global_var(GVAR_BH_JAIL) == 7)) then begin
                 call Node011();
             end
             else begin
-                if (global_var(304) == 3) then begin
+                if (global_var(GVAR_BH_CONSPIRACY) == 3) then begin
                     call Node011();
                 end
                 else begin
-                    if (global_var(304) == 5) then begin
+                    if (global_var(GVAR_BH_CONSPIRACY) == 5) then begin
                         start_gdialog(595, self_obj, 4, -1, -1);
                         gsay_start;
                         call Node017();
@@ -498,15 +498,15 @@ begin
                         end_dialogue;
                     end
                     else begin
-                        if (global_var(304) == 8) then begin
+                        if (global_var(GVAR_BH_CONSPIRACY) == 8) then begin
                             call Node019();
                         end
                         else begin
-                            if (global_var(304) == 4) then begin
+                            if (global_var(GVAR_BH_CONSPIRACY) == 4) then begin
                                 call Node016();
                             end
                             else begin
-                                if ((global_var(303) == 3) or ((global_var(303) == 6) and (global_var(443) == 0)) or ((global_var(303) == 5) and (global_var(443) == 0))) then begin
+                                if ((global_var(GVAR_BH_JAIL) == 3) or ((global_var(GVAR_BH_JAIL) == 6) and (global_var(GVAR_JAIL_BREAK) == 0)) or ((global_var(GVAR_BH_JAIL) == 5) and (global_var(GVAR_JAIL_BREAK) == 0))) then begin
                                     start_gdialog(595, self_obj, 4, -1, -1);
                                     gsay_start;
                                     call Node013();
@@ -514,7 +514,7 @@ begin
                                     end_dialogue;
                                 end
                                 else begin
-                                    if (global_var(303) == 8) then begin
+                                    if (global_var(GVAR_BH_JAIL) == 8) then begin
                                         start_gdialog(595, self_obj, 4, -1, -1);
                                         gsay_start;
                                         call Node020();
@@ -522,7 +522,7 @@ begin
                                         end_dialogue;
                                     end
                                     else begin
-                                        if (global_var(303) == 5) then begin
+                                        if (global_var(GVAR_BH_JAIL) == 5) then begin
                                             start_gdialog(595, self_obj, 4, -1, -1);
                                             gsay_start;
                                             call Node015();
@@ -530,7 +530,7 @@ begin
                                             end_dialogue;
                                         end
                                         else begin
-                                            if ((global_var(303) == 4) or (global_var(303) == 6)) then begin
+                                            if ((global_var(GVAR_BH_JAIL) == 4) or (global_var(GVAR_BH_JAIL) == 6)) then begin
                                                 start_gdialog(595, self_obj, 4, -1, -1);
                                                 gsay_start;
                                                 call Node014();
@@ -552,7 +552,7 @@ begin
                                                         call Node001();
                                                     end
                                                     else begin
-                                                        if (get_critter_stat(dude_obj, 4) <= 3) then begin
+                                                        if (dude_is_stupid) then begin
                                                             call Node002();
                                                         end
                                                         else begin
@@ -580,63 +580,63 @@ procedure destroy_p_proc
 begin
     set_map_var(27, 1);
     if (source_obj == dude_obj) then begin
-        set_global_var(4, global_var(4) + 1);
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(5) > (2 * global_var(4))) or (global_var(3) == 1))) then begin
-            set_global_var(3, 1);
-            set_global_var(2, 0);
+        set_global_var(GVAR_BAD_MONSTER, global_var(GVAR_BAD_MONSTER) + 1);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
+            set_global_var(GVAR_BERSERKER_REPUTATION, 1);
+            set_global_var(GVAR_CHAMPION_REPUTATION, 0);
         end
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(4) > (3 * global_var(5))) or (global_var(2) == 1)) and (global_var(1) == 0)) then begin
-            set_global_var(2, 1);
-            set_global_var(3, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(4) > (3 * global_var(GVAR_GOOD_MONSTER))) or (global_var(GVAR_CHAMPION_REPUTATION) == 1)) and (global_var(GVAR_CHILDKILLER_REPUTATION) == 0)) then begin
+            set_global_var(GVAR_CHAMPION_REPUTATION, 1);
+            set_global_var(GVAR_BERSERKER_REPUTATION, 0);
         end
-        set_global_var(0, global_var(0) + 5);
+        set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + 5);
         debug_msg("Player gains " + 5 + " Karma Points.");
-        set_global_var(37, 0);
-        set_global_var(38, 0);
-        set_global_var(39, 0);
-        set_global_var(40, 0);
-        set_global_var(41, 0);
-        set_global_var(42, 0);
-        set_global_var(43, 0);
-        set_global_var(44, 0);
-        set_global_var(45, 0);
-        tmp_gen_rep := global_var(0);
-        if (has_trait(0, dude_obj, 95)) then begin
-            tmp_gen_rep := global_var(0) * 2;
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+        set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+        set_global_var(GVAR_KARMA_DEFENDER, 0);
+        set_global_var(GVAR_KARMA_WANDERER, 0);
+        set_global_var(GVAR_KARMA_BETRAYER, 0);
+        set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+        set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+        tmp_gen_rep := global_var(GVAR_PLAYER_REPUTATION);
+        if (has_trait(TRAIT_PERK, dude_obj, PERK_karma_beacon_perk)) then begin
+            tmp_gen_rep := global_var(GVAR_PLAYER_REPUTATION) * 2;
         end
         if (tmp_gen_rep >= 1000) then begin
-            set_global_var(37, 1);
+            set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
         end
         else begin
             if (tmp_gen_rep >= 750) then begin
-                set_global_var(38, 1);
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
             end
             else begin
                 if (tmp_gen_rep >= 500) then begin
-                    set_global_var(39, 1);
+                    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                 end
                 else begin
                     if (tmp_gen_rep >= 250) then begin
-                        set_global_var(40, 1);
+                        set_global_var(GVAR_KARMA_DEFENDER, 1);
                     end
                     else begin
                         if (tmp_gen_rep > -250) then begin
-                            set_global_var(41, 1);
+                            set_global_var(GVAR_KARMA_WANDERER, 1);
                         end
                         else begin
                             if (tmp_gen_rep > -500) then begin
-                                set_global_var(42, 1);
+                                set_global_var(GVAR_KARMA_BETRAYER, 1);
                             end
                             else begin
                                 if (tmp_gen_rep > -750) then begin
-                                    set_global_var(43, 1);
+                                    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                 end
                                 else begin
                                     if (tmp_gen_rep > -1000) then begin
-                                        set_global_var(44, 1);
+                                        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                     end
                                     else begin
-                                        set_global_var(45, 1);
+                                        set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                     end
                                 end
                             end
@@ -645,7 +645,7 @@ begin
                 end
             end
         end
-        set_global_var(54, global_var(54) + 2);
+        set_global_var(GVAR_TOWN_REP_BROKEN_HILLS, global_var(GVAR_TOWN_REP_BROKEN_HILLS) + 2);
         debug_msg("Added " + 2 + " to Town Rep");
     end
 end
@@ -654,17 +654,17 @@ procedure look_at_p_proc
 begin
     script_overrides;
     if (local_var(4) == 0) then begin
-        display_msg(message_str(595, 100));
+        display_mstr(100);
     end
     else begin
-        display_msg(message_str(595, 101));
+        display_mstr(101);
     end
 end
 
 procedure description_p_proc
 begin
     script_overrides;
-    display_msg(message_str(595, 102));
+    display_mstr(102);
 end
 
 procedure use_skill_on_p_proc
@@ -675,17 +675,17 @@ procedure damage_p_proc
 begin
     if (source_obj == dude_obj) then begin
         set_local_var(5, 2);
-        set_global_var(525, 1);
-        set_global_var(309, 1);
+        set_global_var(GVAR_ENEMY_CONSPIRATORS, 1);
+        set_global_var(GVAR_ENEMY_BROKEN_HILLS, 1);
     end
 end
 
 procedure map_enter_p_proc
 begin
     Frank_Ptr := self_obj;
-    critter_add_trait(self_obj, 1, 6, 150);
-    critter_add_trait(self_obj, 1, 5, 24);
-    if ((global_var(304) == 5) or ((global_var(303) == 5) or ((global_var(303) == 6) and (global_var(443) > 0)) or (global_var(303) == 4) or (global_var(303) == 8) and (global_var(304) != 7) and (global_var(304) != 8))) then begin
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 150);
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_TOUGH_CITIZEN);
+    if ((global_var(GVAR_BH_CONSPIRACY) == 5) or ((global_var(GVAR_BH_JAIL) == 5) or ((global_var(GVAR_BH_JAIL) == 6) and (global_var(GVAR_JAIL_BREAK) > 0)) or (global_var(GVAR_BH_JAIL) == 4) or (global_var(GVAR_BH_JAIL) == 8) and (global_var(GVAR_BH_CONSPIRACY) != 7) and (global_var(GVAR_BH_CONSPIRACY) != 8))) then begin
         move_to(self_obj, 29077, 0);
         critter_attempt_placement(self_obj, 29077, 0);
     end
@@ -732,33 +732,33 @@ end
 
 procedure Node001
 begin
-    gsay_reply(595, 103);
-    if (get_critter_stat(dude_obj, 4) == 1) then begin
-        giq_option(1, 595, 104, Node002, 50);
+    Reply(103);
+    if (dude_iq == 1) then begin
+        giq_option(1, NAME, 104, Node002, NEUTRAL_REACTION);
     end
-    if ((get_critter_stat(dude_obj, 4) == 2) or (get_critter_stat(dude_obj, 4) == 3)) then begin
-        giq_option(2, 595, 105, Node002, 50);
+    if ((dude_iq == 2) or (dude_iq == 3)) then begin
+        giq_option(2, NAME, 105, Node002, NEUTRAL_REACTION);
     end
-    giq_option(4, 595, 106, Node003, 50);
-    giq_option(4, 595, 107, Node004, 50);
-    giq_option(4, 595, 108, Node005, 50);
-    giq_option(4, 595, 109, Node006, 50);
-    giq_option(-3, 595, 149, Node999, 50);
-    giq_option(4, 595, 150, Node999, 50);
+    giq_option(4, NAME, 106, Node003, NEUTRAL_REACTION);
+    giq_option(4, NAME, 107, Node004, NEUTRAL_REACTION);
+    giq_option(4, NAME, 108, Node005, NEUTRAL_REACTION);
+    giq_option(4, NAME, 109, Node006, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 149, Node999, NEUTRAL_REACTION);
+    giq_option(4, NAME, 150, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node002
 begin
-    gsay_reply(595, 110);
-    giq_option(-3, 595, 111, Node999, 50);
+    Reply(110);
+    giq_option(-3, NAME, 111, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node003
 begin
-    gsay_reply(595, 112);
-    giq_option(4, 595, 113, Node007, 50);
-    giq_option(4, 595, 114, Node003a, 51);
-    giq_option(4, 595, 115, Node999, 50);
+    Reply(112);
+    giq_option(4, NAME, 113, Node007, NEUTRAL_REACTION);
+    giq_option(4, NAME, 114, Node003a, BAD_REACTION);
+    giq_option(4, NAME, 115, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node003a
@@ -814,12 +814,12 @@ end
 
 procedure Node004
 begin
-    gsay_reply(595, 116);
-    giq_option(4, 595, 117, Node009, 50);
-    giq_option(4, 595, 118, Node004a, 51);
-    giq_option(4, 595, 119, Node005, 50);
-    giq_option(4, 595, 120, Node006, 50);
-    giq_option(4, 595, 121, Node999, 50);
+    Reply(116);
+    giq_option(4, NAME, 117, Node009, NEUTRAL_REACTION);
+    giq_option(4, NAME, 118, Node004a, BAD_REACTION);
+    giq_option(4, NAME, 119, Node005, NEUTRAL_REACTION);
+    giq_option(4, NAME, 120, Node006, NEUTRAL_REACTION);
+    giq_option(4, NAME, 121, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node004a
@@ -875,36 +875,36 @@ end
 
 procedure Node005
 begin
-    gsay_reply(595, 122);
-    giq_option(4, 595, 123, Node006, 50);
-    giq_option(4, 595, 124, Node999, 50);
+    Reply(122);
+    giq_option(4, NAME, 123, Node006, NEUTRAL_REACTION);
+    giq_option(4, NAME, 124, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node006
 begin
-    gsay_reply(595, 125);
-    giq_option(4, 595, 126, Node008, 51);
-    giq_option(4, 595, 127, Node009, 49);
+    Reply(125);
+    giq_option(4, NAME, 126, Node008, BAD_REACTION);
+    giq_option(4, NAME, 127, Node009, GOOD_REACTION);
 end
 
 procedure Node007
 begin
-    gsay_reply(595, 128);
-    giq_option(4, 595, 129, Node009, 50);
-    giq_option(4, 595, 130, Node008, 50);
+    Reply(128);
+    giq_option(4, NAME, 129, Node009, NEUTRAL_REACTION);
+    giq_option(4, NAME, 130, Node008, NEUTRAL_REACTION);
 end
 
 procedure Node008
 begin
-    gsay_reply(595, 131);
-    giq_option(4, 595, 132, Node999, 50);
+    Reply(131);
+    giq_option(4, NAME, 132, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node009
 begin
-    gsay_reply(595, 133);
-    giq_option(4, 595, 134, Node009a, 49);
-    giq_option(4, 595, 135, Node009b, 51);
+    Reply(133);
+    giq_option(4, NAME, 134, Node009a, GOOD_REACTION);
+    giq_option(4, NAME, 135, Node009b, BAD_REACTION);
 end
 
 procedure Node009a
@@ -955,7 +955,7 @@ begin
             end
         end
     end
-    set_global_var(303, 1);
+    set_global_var(GVAR_BH_JAIL, 1);
     call Node010();
 end
 
@@ -1012,66 +1012,65 @@ end
 
 procedure Node010
 begin
-    gsay_reply(595, 136);
-    giq_option(4, 595, 132, Node999, 50);
+    Reply(136);
+    giq_option(4, NAME, 132, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node011
 begin
-    float_msg(self_obj, message_str(595, 137), 8);
+    float_msg(self_obj, mstr(137), FLOAT_MSG_YELLOW);
 end
 
 procedure Node012
 begin
-    gsay_reply(595, 138);
-    giq_option(4, 595, 139, Node005, 50);
-    giq_option(4, 595, 140, Node006, 50);
-    giq_option(4, 595, 141, Node004, 50);
+    Reply(138);
+    giq_option(4, NAME, 139, Node005, NEUTRAL_REACTION);
+    giq_option(4, NAME, 140, Node006, NEUTRAL_REACTION);
+    giq_option(4, NAME, 141, Node004, NEUTRAL_REACTION);
 end
 
 procedure Node013
 begin
-    gsay_reply(595, 142);
-    giq_option(4, 595, 132, Node999, 50);
+    Reply(142);
+    giq_option(4, NAME, 132, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node014
 begin
-    gsay_reply(595, 143);
-    giq_option(4, 595, 132, Node999, 50);
+    Reply(143);
+    giq_option(4, NAME, 132, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node015
 begin
-    gsay_reply(595, 144);
-    giq_option(4, 595, 132, Node999, 50);
+    Reply(144);
+    giq_option(4, NAME, 132, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node016
 begin
-    float_msg(self_obj, message_str(595, 145), 8);
+    float_msg(self_obj, mstr(145), FLOAT_MSG_YELLOW);
 end
 
 procedure Node017
 begin
-    gsay_reply(595, 146);
-    giq_option(4, 595, 132, Node999, 50);
+    Reply(146);
+    giq_option(4, NAME, 132, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node018
 begin
-    float_msg(self_obj, message_str(595, 147), 8);
+    float_msg(self_obj, mstr(147), FLOAT_MSG_YELLOW);
     call Node998();
 end
 
 procedure Node019
 begin
-    float_msg(self_obj, message_str(595, 151), 8);
+    float_msg(self_obj, mstr(151), FLOAT_MSG_YELLOW);
 end
 
 procedure Node020
 begin
-    gsay_reply(595, 154);
-    giq_option(4, 595, 155, Node999, 50);
+    Reply(154);
+    giq_option(4, NAME, 155, Node999, NEUTRAL_REACTION);
 end
-
