diff --git a/Mods/MegaMod/Original/HCGUARD2.ssl b/Mods/MegaMod/ReDefine/HCGUARD2.ssl
index f920436..4cdaab8 100644
--- a/Mods/MegaMod/Original/HCGUARD2.ssl
+++ b/Mods/MegaMod/ReDefine/HCGUARD2.ssl
@@ -56,13 +56,13 @@ end
 procedure look_at_p_proc
 begin
     script_overrides;
-    display_msg(message_str(1815, 100));
+    display_mstr(100);
 end
 
 procedure description_p_proc
 begin
     script_overrides;
-    display_msg(message_str(1815, 101));
+    display_mstr(101);
 end
 
 procedure use_skill_on_p_proc
@@ -84,63 +84,63 @@ end
 procedure destroy_p_proc
 begin
     if (source_obj == dude_obj) then begin
-        set_global_var(4, global_var(4) + 1);
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(5) > (2 * global_var(4))) or (global_var(3) == 1))) then begin
-            set_global_var(3, 1);
-            set_global_var(2, 0);
+        set_global_var(GVAR_BAD_MONSTER, global_var(GVAR_BAD_MONSTER) + 1);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
+            set_global_var(GVAR_BERSERKER_REPUTATION, 1);
+            set_global_var(GVAR_CHAMPION_REPUTATION, 0);
         end
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(4) > (3 * global_var(5))) or (global_var(2) == 1)) and (global_var(1) == 0)) then begin
-            set_global_var(2, 1);
-            set_global_var(3, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(4) > (3 * global_var(GVAR_GOOD_MONSTER))) or (global_var(GVAR_CHAMPION_REPUTATION) == 1)) and (global_var(GVAR_CHILDKILLER_REPUTATION) == 0)) then begin
+            set_global_var(GVAR_CHAMPION_REPUTATION, 1);
+            set_global_var(GVAR_BERSERKER_REPUTATION, 0);
         end
-        set_global_var(0, global_var(0) + 5);
+        set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + 5);
         debug_msg("Player gains " + 5 + " Karma Points.");
-        set_global_var(37, 0);
-        set_global_var(38, 0);
-        set_global_var(39, 0);
-        set_global_var(40, 0);
-        set_global_var(41, 0);
-        set_global_var(42, 0);
-        set_global_var(43, 0);
-        set_global_var(44, 0);
-        set_global_var(45, 0);
-        tmp_gen_rep := global_var(0);
-        if (has_trait(0, dude_obj, 95)) then begin
-            tmp_gen_rep := global_var(0) * 2;
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+        set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+        set_global_var(GVAR_KARMA_DEFENDER, 0);
+        set_global_var(GVAR_KARMA_WANDERER, 0);
+        set_global_var(GVAR_KARMA_BETRAYER, 0);
+        set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+        set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+        tmp_gen_rep := global_var(GVAR_PLAYER_REPUTATION);
+        if (has_trait(TRAIT_PERK, dude_obj, PERK_karma_beacon_perk)) then begin
+            tmp_gen_rep := global_var(GVAR_PLAYER_REPUTATION) * 2;
         end
         if (tmp_gen_rep >= 1000) then begin
-            set_global_var(37, 1);
+            set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
         end
         else begin
             if (tmp_gen_rep >= 750) then begin
-                set_global_var(38, 1);
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
             end
             else begin
                 if (tmp_gen_rep >= 500) then begin
-                    set_global_var(39, 1);
+                    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                 end
                 else begin
                     if (tmp_gen_rep >= 250) then begin
-                        set_global_var(40, 1);
+                        set_global_var(GVAR_KARMA_DEFENDER, 1);
                     end
                     else begin
                         if (tmp_gen_rep > -250) then begin
-                            set_global_var(41, 1);
+                            set_global_var(GVAR_KARMA_WANDERER, 1);
                         end
                         else begin
                             if (tmp_gen_rep > -500) then begin
-                                set_global_var(42, 1);
+                                set_global_var(GVAR_KARMA_BETRAYER, 1);
                             end
                             else begin
                                 if (tmp_gen_rep > -750) then begin
-                                    set_global_var(43, 1);
+                                    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                 end
                                 else begin
                                     if (tmp_gen_rep > -1000) then begin
-                                        set_global_var(44, 1);
+                                        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                     end
                                     else begin
-                                        set_global_var(45, 1);
+                                        set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                     end
                                 end
                             end
@@ -156,7 +156,7 @@ end
 procedure map_enter_p_proc
 begin
     guard2_obj := self_obj;
-    critter_add_trait(self_obj, 1, 6, 217);
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 217);
 end
 
 procedure map_update_p_proc
@@ -252,91 +252,91 @@ end
 
 procedure Node001
 begin
-    gsay_reply(1815, 200);
-    if (has_skill(dude_obj, 14) > 100) then begin
-        giq_option(4, 1815, 201, Node990, 50);
+    Reply(200);
+    if (dude_skill(SKILL_SPEECH) > 100) then begin
+        giq_option(4, NAME, 201, Node990, NEUTRAL_REACTION);
     end
-    giq_option(4, 1815, 202, Node991, 50);
+    giq_option(4, NAME, 202, Node991, NEUTRAL_REACTION);
 end
 
 procedure Node002
 begin
-    gsay_reply(1815, 210);
-    giq_option(4, 1815, 211, Node992, 50);
-    giq_option(4, 1815, 212, Node998, 51);
+    Reply(210);
+    giq_option(4, NAME, 211, Node992, NEUTRAL_REACTION);
+    giq_option(4, NAME, 212, Node998, BAD_REACTION);
 end
 
 procedure Node003
 begin
-    gsay_reply(1815, 220);
-    giq_option(4, 1815, 221, Node004, 50);
+    Reply(220);
+    giq_option(4, NAME, 221, Node004, NEUTRAL_REACTION);
 end
 
 procedure Node004
 begin
-    gsay_reply(1815, 230);
-    giq_option(4, 1815, 231, Node800, 50);
+    Reply(230);
+    giq_option(4, NAME, 231, Node800, NEUTRAL_REACTION);
 end
 
 procedure Node005
 begin
-    gsay_reply(1815, 250);
-    giq_option(4, 1815, 251, Node801, 50);
+    Reply(250);
+    giq_option(4, NAME, 251, Node801, NEUTRAL_REACTION);
 end
 
 procedure Node006
 begin
-    gsay_reply(1815, 240);
+    Reply(240);
     if (From_Node == 991) then begin
-        giq_option(4, 1815, 241, Node003, 50);
+        giq_option(4, NAME, 241, Node003, NEUTRAL_REACTION);
     end
     else begin
         if (From_Node == 990) then begin
-            giq_option(4, 1815, 242, Node005, 50);
+            giq_option(4, NAME, 242, Node005, NEUTRAL_REACTION);
         end
     end
-    giq_option(4, 1815, 212, Node998, 51);
+    giq_option(4, NAME, 212, Node998, BAD_REACTION);
 end
 
 procedure Node007
 begin
-    gsay_reply(1815, 260);
-    giq_option(4, 1815, message_str(14, 350), Node998, 50);
+    Reply(260);
+    giq_option(4, NAME, g_mstr(350), Node998, NEUTRAL_REACTION);
 end
 
 procedure Node008
 begin
     set_local_var(9, 0);
-    gsay_reply(1815, 300);
-    giq_option(4, 1815, message_str(14, 350), Node998, 50);
+    Reply(300);
+    giq_option(4, NAME, g_mstr(350), Node998, NEUTRAL_REACTION);
 end
 
 procedure Node009
 begin
-    gsay_reply(1815, 400);
-    giq_option(4, 1815, message_str(14, 350), Node996, 50);
+    Reply(400);
+    giq_option(4, NAME, g_mstr(350), Node996, NEUTRAL_REACTION);
 end
 
 procedure Node010
 begin
-    gsay_reply(1815, 401);
-    giq_option(4, 1815, message_str(14, 350), Node999, 50);
+    Reply(401);
+    giq_option(4, NAME, g_mstr(350), Node999, NEUTRAL_REACTION);
 end
 
 procedure Node011
 begin
-    gsay_reply(1815, 402);
-    giq_option(4, 1815, message_str(14, 350), Node996, 50);
+    Reply(402);
+    giq_option(4, NAME, g_mstr(350), Node996, NEUTRAL_REACTION);
 end
 
 procedure critter_p_proc
 begin
-    if (((global_var(361) bwand 262144) != 0) or (map_var(0) == 1) and obj_can_see_obj(self_obj, dude_obj)) then begin
-        attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+    if (((global_var(GVAR_SAN_FRAN_FLAGS) bwand 262144) != 0) or (map_var(0) == 1) and obj_can_see_obj(self_obj, dude_obj)) then begin
+        attack(dude_obj);
     end
     if (map_var(0) != 1) then begin
         if ((local_var(9) == 1) and obj_can_see_obj(self_obj, dude_obj)) then begin
-            debug_msg("force_dialog_start(" + forced_node + "): " + obj_name(self_obj));
+            debug_msg("force_dialog_start(" + forced_node + "): " + self_name);
             forced_node := "Node008";
             dialogue_system_enter;
         end
@@ -344,11 +344,11 @@ begin
             if (local_var(7) == 1) then begin
                 if ((game_time - map_var(2)) >= (60 * 10 * 10)) then begin
                     set_local_var(7, 0);
-                    if (elevation(dude_obj) == 1) then begin
+                    if (dude_elevation == 1) then begin
                         move_to(self_obj, 22328, 1);
                         set_obj_visibility(self_obj, 0);
                         if (obj_can_see_obj(self_obj, dude_obj)) then begin
-                            debug_msg("force_dialog_start(" + forced_node + "): " + obj_name(self_obj));
+                            debug_msg("force_dialog_start(" + forced_node + "): " + self_name);
                             forced_node := "Node008";
                             dialogue_system_enter;
                         end
@@ -357,7 +357,7 @@ begin
                     else begin
                         set_obj_visibility(self_obj, 0);
                         if (obj_can_see_obj(self_obj, dude_obj)) then begin
-                            debug_msg("force_dialog_start(" + forced_node + "): " + obj_name(self_obj));
+                            debug_msg("force_dialog_start(" + forced_node + "): " + self_name);
                             forced_node := "Node008";
                             dialogue_system_enter;
                         end
@@ -385,13 +385,13 @@ begin
         end
         else begin
             if (fixed_param == 4) then begin
-                attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+                attack(dude_obj);
             end
             else begin
                 if (fixed_param == 5) then begin
                     if (object_tamper == 1) then begin
                         object_tamper := 0;
-                        debug_msg("force_dialog_start(" + forced_node + "): " + obj_name(self_obj));
+                        debug_msg("force_dialog_start(" + forced_node + "): " + self_name);
                         forced_node := "Node009";
                         dialogue_system_enter;
                     end
@@ -400,13 +400,13 @@ begin
                             if (local_var(10) == 0) then begin
                                 object_tamper := 0;
                                 set_local_var(10, 1);
-                                debug_msg("force_dialog_start(" + forced_node + "): " + obj_name(self_obj));
+                                debug_msg("force_dialog_start(" + forced_node + "): " + self_name);
                                 forced_node := "Node010";
                                 dialogue_system_enter;
                             end
                             else begin
                                 object_tamper := 0;
-                                debug_msg("force_dialog_start(" + forced_node + "): " + obj_name(self_obj));
+                                debug_msg("force_dialog_start(" + forced_node + "): " + self_name);
                                 forced_node := "Node011";
                                 dialogue_system_enter;
                             end
@@ -434,4 +434,3 @@ begin
     set_local_var(5, 1);
     add_timer_event(self_obj, game_ticks(5), 4);
 end
-
