diff --git a/Mods/MegaMod/Original/HCJACOB.ssl b/Mods/MegaMod/ReDefine/HCJACOB.ssl
index c447c12..eea2e40 100644
--- a/Mods/MegaMod/Original/HCJACOB.ssl
+++ b/Mods/MegaMod/ReDefine/HCJACOB.ssl
@@ -104,9 +104,9 @@ end
 
 procedure critter_p_proc
 begin
-    if ((global_var(525) == 1) or (local_var(5) == 2) and obj_can_see_obj(self_obj, dude_obj)) then begin
+    if ((global_var(GVAR_ENEMY_CONSPIRATORS) == 1) or (local_var(5) == 2) and obj_can_see_obj(self_obj, dude_obj)) then begin
         set_local_var(5, 1);
-        attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+        attack(dude_obj);
     end
 end
 
@@ -121,16 +121,16 @@ procedure talk_p_proc
 begin
     if (local_var(2) == 0) then begin
         set_local_var(2, 1);
-        set_local_var(3, (get_critter_stat(dude_obj, 3) - get_critter_stat(self_obj, 3)) * 5);
+        set_local_var(3, (dude_charisma - self_charisma) * 5);
         debug_msg("Base Reaction == " + local_var(3));
     end
     Static_Reaction := 0;
     set_local_var(0, 0);
     debug_msg("Start Reaction == " + local_var(0));
     reaction_bonus_karma := floor(reaction_bonus_karma / 1000);
-    reaction_bonus_town_rep := floor(global_var(54) / 2);
-    if (has_trait(2, dude_obj, 13)) then begin
-        if (get_critter_stat(dude_obj, 34) == get_critter_stat(self_obj, 34)) then begin
+    reaction_bonus_town_rep := floor(global_var(GVAR_TOWN_REP_BROKEN_HILLS) / 2);
+    if (has_trait(TRAIT_TRAIT, dude_obj, TRAIT_sex_appeal)) then begin
+        if (dude_gender == self_gender) then begin
             Static_Reaction := Static_Reaction + -20;
         end
         else begin
@@ -156,52 +156,52 @@ begin
             Static_Reaction := Static_Reaction + reaction_bonus_town_rep;
         end
         debug_msg("Town Rep Bonus == " + Static_Reaction);
-        set_global_var(37, 0);
-        set_global_var(38, 0);
-        set_global_var(39, 0);
-        set_global_var(40, 0);
-        set_global_var(41, 0);
-        set_global_var(42, 0);
-        set_global_var(43, 0);
-        set_global_var(44, 0);
-        set_global_var(45, 0);
-        tmp_gen_rep := global_var(0);
-        if (has_trait(0, dude_obj, 95)) then begin
-            tmp_gen_rep := global_var(0) * 2;
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+        set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+        set_global_var(GVAR_KARMA_DEFENDER, 0);
+        set_global_var(GVAR_KARMA_WANDERER, 0);
+        set_global_var(GVAR_KARMA_BETRAYER, 0);
+        set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+        set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+        tmp_gen_rep := global_var(GVAR_PLAYER_REPUTATION);
+        if (has_trait(TRAIT_PERK, dude_obj, PERK_karma_beacon_perk)) then begin
+            tmp_gen_rep := global_var(GVAR_PLAYER_REPUTATION) * 2;
         end
         if (tmp_gen_rep >= 1000) then begin
-            set_global_var(37, 1);
+            set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
         end
         else begin
             if (tmp_gen_rep >= 750) then begin
-                set_global_var(38, 1);
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
             end
             else begin
                 if (tmp_gen_rep >= 500) then begin
-                    set_global_var(39, 1);
+                    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                 end
                 else begin
                     if (tmp_gen_rep >= 250) then begin
-                        set_global_var(40, 1);
+                        set_global_var(GVAR_KARMA_DEFENDER, 1);
                     end
                     else begin
                         if (tmp_gen_rep > -250) then begin
-                            set_global_var(41, 1);
+                            set_global_var(GVAR_KARMA_WANDERER, 1);
                         end
                         else begin
                             if (tmp_gen_rep > -500) then begin
-                                set_global_var(42, 1);
+                                set_global_var(GVAR_KARMA_BETRAYER, 1);
                             end
                             else begin
                                 if (tmp_gen_rep > -750) then begin
-                                    set_global_var(43, 1);
+                                    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                 end
                                 else begin
                                     if (tmp_gen_rep > -1000) then begin
-                                        set_global_var(44, 1);
+                                        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                     end
                                     else begin
-                                        set_global_var(45, 1);
+                                        set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                     end
                                 end
                             end
@@ -210,35 +210,35 @@ begin
                 end
             end
         end
-        if (global_var(37) == 1) then begin
+        if (global_var(GVAR_KARMA_HOLY_WARRIOR) == 1) then begin
             Static_Reaction := Static_Reaction - 20;
         end
         else begin
-            if (global_var(38) == 1) then begin
+            if (global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES) == 1) then begin
                 Static_Reaction := Static_Reaction - 15;
             end
             else begin
-                if (global_var(39) == 1) then begin
+                if (global_var(GVAR_KARMA_SHIELD_OF_HOPE) == 1) then begin
                     Static_Reaction := Static_Reaction - 10;
                 end
                 else begin
-                    if (global_var(40) == 1) then begin
+                    if (global_var(GVAR_KARMA_DEFENDER) == 1) then begin
                         Static_Reaction := Static_Reaction - 5;
                     end
                     else begin
-                        if (global_var(41) == 1) then begin
+                        if (global_var(GVAR_KARMA_WANDERER) == 1) then begin
                             Static_Reaction := Static_Reaction - 0;
                         end
                         else begin
-                            if (global_var(42) == 1) then begin
+                            if (global_var(GVAR_KARMA_BETRAYER) == 1) then begin
                                 Static_Reaction := Static_Reaction + 5;
                             end
                             else begin
-                                if (global_var(43) == 1) then begin
+                                if (global_var(GVAR_KARMA_SWORD_OF_DESPAIR) == 1) then begin
                                     Static_Reaction := Static_Reaction + 10;
                                 end
                                 else begin
-                                    if (global_var(44) == 1) then begin
+                                    if (global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE) == 1) then begin
                                         Static_Reaction := Static_Reaction + 15;
                                     end
                                     else begin
@@ -258,52 +258,52 @@ begin
         debug_msg("Karma Reaction Bonus == " + Static_Reaction);
         Static_Reaction := Static_Reaction + reaction_bonus_town_rep;
         debug_msg("Town Rep Bonus == " + Static_Reaction);
-        set_global_var(37, 0);
-        set_global_var(38, 0);
-        set_global_var(39, 0);
-        set_global_var(40, 0);
-        set_global_var(41, 0);
-        set_global_var(42, 0);
-        set_global_var(43, 0);
-        set_global_var(44, 0);
-        set_global_var(45, 0);
-        tmp_gen_rep := global_var(0);
-        if (has_trait(0, dude_obj, 95)) then begin
-            tmp_gen_rep := global_var(0) * 2;
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+        set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+        set_global_var(GVAR_KARMA_DEFENDER, 0);
+        set_global_var(GVAR_KARMA_WANDERER, 0);
+        set_global_var(GVAR_KARMA_BETRAYER, 0);
+        set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+        set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+        tmp_gen_rep := global_var(GVAR_PLAYER_REPUTATION);
+        if (has_trait(TRAIT_PERK, dude_obj, PERK_karma_beacon_perk)) then begin
+            tmp_gen_rep := global_var(GVAR_PLAYER_REPUTATION) * 2;
         end
         if (tmp_gen_rep >= 1000) then begin
-            set_global_var(37, 1);
+            set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
         end
         else begin
             if (tmp_gen_rep >= 750) then begin
-                set_global_var(38, 1);
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
             end
             else begin
                 if (tmp_gen_rep >= 500) then begin
-                    set_global_var(39, 1);
+                    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                 end
                 else begin
                     if (tmp_gen_rep >= 250) then begin
-                        set_global_var(40, 1);
+                        set_global_var(GVAR_KARMA_DEFENDER, 1);
                     end
                     else begin
                         if (tmp_gen_rep > -250) then begin
-                            set_global_var(41, 1);
+                            set_global_var(GVAR_KARMA_WANDERER, 1);
                         end
                         else begin
                             if (tmp_gen_rep > -500) then begin
-                                set_global_var(42, 1);
+                                set_global_var(GVAR_KARMA_BETRAYER, 1);
                             end
                             else begin
                                 if (tmp_gen_rep > -750) then begin
-                                    set_global_var(43, 1);
+                                    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                 end
                                 else begin
                                     if (tmp_gen_rep > -1000) then begin
-                                        set_global_var(44, 1);
+                                        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                     end
                                     else begin
-                                        set_global_var(45, 1);
+                                        set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                     end
                                 end
                             end
@@ -312,35 +312,35 @@ begin
                 end
             end
         end
-        if (global_var(37) == 1) then begin
+        if (global_var(GVAR_KARMA_HOLY_WARRIOR) == 1) then begin
             Static_Reaction := Static_Reaction + 20;
         end
         else begin
-            if (global_var(38) == 1) then begin
+            if (global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES) == 1) then begin
                 Static_Reaction := Static_Reaction + 15;
             end
             else begin
-                if (global_var(39) == 1) then begin
+                if (global_var(GVAR_KARMA_SHIELD_OF_HOPE) == 1) then begin
                     Static_Reaction := Static_Reaction + 10;
                 end
                 else begin
-                    if (global_var(40) == 1) then begin
+                    if (global_var(GVAR_KARMA_DEFENDER) == 1) then begin
                         Static_Reaction := Static_Reaction + 5;
                     end
                     else begin
-                        if (global_var(41) == 1) then begin
+                        if (global_var(GVAR_KARMA_WANDERER) == 1) then begin
                             Static_Reaction := Static_Reaction + 0;
                         end
                         else begin
-                            if (global_var(42) == 1) then begin
+                            if (global_var(GVAR_KARMA_BETRAYER) == 1) then begin
                                 Static_Reaction := Static_Reaction + -5;
                             end
                             else begin
-                                if (global_var(43) == 1) then begin
+                                if (global_var(GVAR_KARMA_SWORD_OF_DESPAIR) == 1) then begin
                                     Static_Reaction := Static_Reaction + -10;
                                 end
                                 else begin
-                                    if (global_var(44) == 1) then begin
+                                    if (global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE) == 1) then begin
                                         Static_Reaction := Static_Reaction + -15;
                                     end
                                     else begin
@@ -356,32 +356,32 @@ begin
         debug_msg("Good Person Reaction Bonus == " + Static_Reaction);
     end
     if (Slavery_Tolerant == 0) then begin
-        if (global_var(11) == 1) then begin
+        if (global_var(GVAR_REPUTATION_SLAVER) == 1) then begin
             Static_Reaction := Static_Reaction - 50;
         end
     end
     else begin
         if (Slavery_Tolerant == 1) then begin
-            if (global_var(11) == 1) then begin
+            if (global_var(GVAR_REPUTATION_SLAVER) == 1) then begin
                 Static_Reaction := Static_Reaction - 25;
             end
         end
         else begin
             if (Slavery_Tolerant == 3) then begin
-                if (global_var(11) == 1) then begin
+                if (global_var(GVAR_REPUTATION_SLAVER) == 1) then begin
                     Static_Reaction := Static_Reaction + 25;
                 end
             end
         end
     end
     debug_msg("Slaver + Aligned Reaction == " + Static_Reaction);
-    Static_Reaction := Static_Reaction + (10 * has_trait(0, dude_obj, 10));
+    Static_Reaction := Static_Reaction + (10 * has_trait(TRAIT_PERK, dude_obj, PERK_presence));
     debug_msg("Presence Reaction == " + Static_Reaction);
-    if (global_var(1) >= 2) then begin
+    if (global_var(GVAR_CHILDKILLER_REPUTATION) >= 2) then begin
         Static_Reaction := Static_Reaction + -30;
     end
     debug_msg("Childkiller Reaction == " + Static_Reaction);
-    if (has_trait(0, dude_obj, 39)) then begin
+    if (has_trait(TRAIT_PERK, dude_obj, PERK_cult_of_personality)) then begin
         if (Evil_Critter == 1) then begin
             if (Static_Reaction > 0) then begin
                 Static_Reaction := Static_Reaction * -1;
@@ -438,7 +438,7 @@ begin
     end
     From_WTG := 1;
     From_13 := 0;
-    if (not(get_critter_stat(dude_obj, 4) == 2) or not(get_critter_stat(dude_obj, 4) == 3)) then begin
+    if (not(dude_iq == 2) or not(dude_iq == 3)) then begin
         gdialog_set_barter_mod(-20);
     end
     move_obj_inven_to_obj(self_obj, broken_hills_temp_box);
@@ -447,19 +447,19 @@ begin
         call Node998();
     end
     else begin
-        if ((global_var(303) == 7) or (global_var(304) == 7)) then begin
+        if ((global_var(GVAR_BH_JAIL) == 7) or (global_var(GVAR_BH_CONSPIRACY) == 7)) then begin
             call Node054();
         end
         else begin
-            if (global_var(304) == 3) then begin
+            if (global_var(GVAR_BH_CONSPIRACY) == 3) then begin
                 call Node055();
             end
             else begin
-                if (global_var(303) == 2) then begin
+                if (global_var(GVAR_BH_JAIL) == 2) then begin
                     call Node066();
                 end
                 else begin
-                    if (global_var(304) == 5) then begin
+                    if (global_var(GVAR_BH_CONSPIRACY) == 5) then begin
                         start_gdialog(608, self_obj, 4, -1, -1);
                         gsay_start;
                         call Node056();
@@ -467,11 +467,11 @@ begin
                         end_dialogue;
                     end
                     else begin
-                        if (global_var(304) == 8) then begin
+                        if (global_var(GVAR_BH_CONSPIRACY) == 8) then begin
                             call Node065();
                         end
                         else begin
-                            if (global_var(304) == 4) then begin
+                            if (global_var(GVAR_BH_CONSPIRACY) == 4) then begin
                                 start_gdialog(608, self_obj, 4, -1, -1);
                                 gsay_start;
                                 call Node052();
@@ -479,7 +479,7 @@ begin
                                 end_dialogue;
                             end
                             else begin
-                                if ((global_var(303) == 0) and (map_var(27) == 1) and (map_var(26) == 1)) then begin
+                                if ((global_var(GVAR_BH_JAIL) == 0) and (map_var(27) == 1) and (map_var(26) == 1)) then begin
                                     start_gdialog(608, self_obj, 4, -1, -1);
                                     gsay_start;
                                     call Node070();
@@ -487,7 +487,7 @@ begin
                                     end_dialogue;
                                 end
                                 else begin
-                                    if ((global_var(303) == 3) and (map_var(27) == 1) and (map_var(26) == 1)) then begin
+                                    if ((global_var(GVAR_BH_JAIL) == 3) and (map_var(27) == 1) and (map_var(26) == 1)) then begin
                                         start_gdialog(608, self_obj, 4, -1, -1);
                                         gsay_start;
                                         call Node068();
@@ -495,7 +495,7 @@ begin
                                         end_dialogue;
                                     end
                                     else begin
-                                        if (global_var(303) == 3) then begin
+                                        if (global_var(GVAR_BH_JAIL) == 3) then begin
                                             start_gdialog(608, self_obj, 4, -1, -1);
                                             gsay_start;
                                             call Node067();
@@ -503,7 +503,7 @@ begin
                                             end_dialogue;
                                         end
                                         else begin
-                                            if (global_var(303) == 4) then begin
+                                            if (global_var(GVAR_BH_JAIL) == 4) then begin
                                                 start_gdialog(608, self_obj, 4, -1, -1);
                                                 gsay_start;
                                                 call Node037();
@@ -511,7 +511,7 @@ begin
                                                 end_dialogue;
                                             end
                                             else begin
-                                                if (global_var(303) == 5) then begin
+                                                if (global_var(GVAR_BH_JAIL) == 5) then begin
                                                     start_gdialog(608, self_obj, 4, -1, -1);
                                                     gsay_start;
                                                     call Node038();
@@ -519,7 +519,7 @@ begin
                                                     end_dialogue;
                                                 end
                                                 else begin
-                                                    if (global_var(303) == 6) then begin
+                                                    if (global_var(GVAR_BH_JAIL) == 6) then begin
                                                         start_gdialog(608, self_obj, 4, -1, -1);
                                                         gsay_start;
                                                         call Node039();
@@ -527,7 +527,7 @@ begin
                                                         end_dialogue;
                                                     end
                                                     else begin
-                                                        if (global_var(303) == 8) then begin
+                                                        if (global_var(GVAR_BH_JAIL) == 8) then begin
                                                             start_gdialog(608, self_obj, 4, -1, -1);
                                                             gsay_start;
                                                             call Node075();
@@ -541,10 +541,10 @@ begin
                                                                 call Node001();
                                                             end
                                                             else begin
-                                                                if (get_critter_stat(dude_obj, 4) < 2) then begin
+                                                                if (dude_iq < 2) then begin
                                                                     call Node002();
                                                                 end
-                                                                if ((get_critter_stat(dude_obj, 4) == 2) or (get_critter_stat(dude_obj, 4) == 3)) then begin
+                                                                if ((dude_iq == 2) or (dude_iq == 3)) then begin
                                                                     call Node003();
                                                                 end
                                                                 else begin
@@ -575,63 +575,63 @@ procedure destroy_p_proc
 begin
     set_map_var(25, 1);
     if (source_obj == dude_obj) then begin
-        set_global_var(4, global_var(4) + 1);
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(5) > (2 * global_var(4))) or (global_var(3) == 1))) then begin
-            set_global_var(3, 1);
-            set_global_var(2, 0);
+        set_global_var(GVAR_BAD_MONSTER, global_var(GVAR_BAD_MONSTER) + 1);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
+            set_global_var(GVAR_BERSERKER_REPUTATION, 1);
+            set_global_var(GVAR_CHAMPION_REPUTATION, 0);
         end
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(4) > (3 * global_var(5))) or (global_var(2) == 1)) and (global_var(1) == 0)) then begin
-            set_global_var(2, 1);
-            set_global_var(3, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(4) > (3 * global_var(GVAR_GOOD_MONSTER))) or (global_var(GVAR_CHAMPION_REPUTATION) == 1)) and (global_var(GVAR_CHILDKILLER_REPUTATION) == 0)) then begin
+            set_global_var(GVAR_CHAMPION_REPUTATION, 1);
+            set_global_var(GVAR_BERSERKER_REPUTATION, 0);
         end
-        set_global_var(0, global_var(0) + 2);
+        set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + 2);
         debug_msg("Player gains " + 2 + " Karma Points.");
-        set_global_var(37, 0);
-        set_global_var(38, 0);
-        set_global_var(39, 0);
-        set_global_var(40, 0);
-        set_global_var(41, 0);
-        set_global_var(42, 0);
-        set_global_var(43, 0);
-        set_global_var(44, 0);
-        set_global_var(45, 0);
-        tmp_gen_rep := global_var(0);
-        if (has_trait(0, dude_obj, 95)) then begin
-            tmp_gen_rep := global_var(0) * 2;
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+        set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+        set_global_var(GVAR_KARMA_DEFENDER, 0);
+        set_global_var(GVAR_KARMA_WANDERER, 0);
+        set_global_var(GVAR_KARMA_BETRAYER, 0);
+        set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+        set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+        tmp_gen_rep := global_var(GVAR_PLAYER_REPUTATION);
+        if (has_trait(TRAIT_PERK, dude_obj, PERK_karma_beacon_perk)) then begin
+            tmp_gen_rep := global_var(GVAR_PLAYER_REPUTATION) * 2;
         end
         if (tmp_gen_rep >= 1000) then begin
-            set_global_var(37, 1);
+            set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
         end
         else begin
             if (tmp_gen_rep >= 750) then begin
-                set_global_var(38, 1);
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
             end
             else begin
                 if (tmp_gen_rep >= 500) then begin
-                    set_global_var(39, 1);
+                    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                 end
                 else begin
                     if (tmp_gen_rep >= 250) then begin
-                        set_global_var(40, 1);
+                        set_global_var(GVAR_KARMA_DEFENDER, 1);
                     end
                     else begin
                         if (tmp_gen_rep > -250) then begin
-                            set_global_var(41, 1);
+                            set_global_var(GVAR_KARMA_WANDERER, 1);
                         end
                         else begin
                             if (tmp_gen_rep > -500) then begin
-                                set_global_var(42, 1);
+                                set_global_var(GVAR_KARMA_BETRAYER, 1);
                             end
                             else begin
                                 if (tmp_gen_rep > -750) then begin
-                                    set_global_var(43, 1);
+                                    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                 end
                                 else begin
                                     if (tmp_gen_rep > -1000) then begin
-                                        set_global_var(44, 1);
+                                        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                     end
                                     else begin
-                                        set_global_var(45, 1);
+                                        set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                     end
                                 end
                             end
@@ -640,7 +640,7 @@ begin
                 end
             end
         end
-        set_global_var(54, global_var(54) + 2);
+        set_global_var(GVAR_TOWN_REP_BROKEN_HILLS, global_var(GVAR_TOWN_REP_BROKEN_HILLS) + 2);
         debug_msg("Added " + 2 + " to Town Rep");
     end
 end
@@ -649,17 +649,17 @@ procedure look_at_p_proc
 begin
     script_overrides;
     if (local_var(4) == 0) then begin
-        display_msg(message_str(608, 100));
+        display_mstr(100);
     end
     else begin
-        display_msg(message_str(608, 101));
+        display_mstr(101);
     end
 end
 
 procedure description_p_proc
 begin
     script_overrides;
-    display_msg(message_str(608, 101));
+    display_mstr(101);
 end
 
 procedure use_skill_on_p_proc
@@ -670,16 +670,16 @@ procedure damage_p_proc
 begin
     if (source_obj == dude_obj) then begin
         set_local_var(5, 2);
-        set_global_var(525, 1);
-        set_global_var(309, 1);
+        set_global_var(GVAR_ENEMY_CONSPIRATORS, 1);
+        set_global_var(GVAR_ENEMY_BROKEN_HILLS, 1);
     end
 end
 
 procedure map_enter_p_proc
 begin
     Jacob_Ptr := self_obj;
-    critter_add_trait(self_obj, 1, 6, 150);
-    critter_add_trait(self_obj, 1, 5, 24);
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 150);
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_TOUGH_CITIZEN);
 end
 
 procedure map_update_p_proc
@@ -704,34 +704,34 @@ procedure Node001
 begin
     if (local_var(4) == 0) then begin
         set_local_var(4, 1);
-        gsay_reply(608, 103);
+        Reply(103);
     end
     else begin
         if ((local_var(1) <= -2) or (From_WTG == 0)) then begin
-            gsay_reply(608, 104);
+            Reply(104);
         end
         else begin
             if (local_var(1) >= 2) then begin
-                gsay_reply(608, 105);
+                Reply(105);
             end
             else begin
-                gsay_reply(608, 106);
+                Reply(106);
             end
         end
     end
     From_WTG := 1;
-    if (get_critter_stat(dude_obj, 4) == 1) then begin
-        giq_option(1, 608, 107, Node002, 50);
+    if (dude_iq == 1) then begin
+        giq_option(1, NAME, 107, Node002, NEUTRAL_REACTION);
     end
-    if ((get_critter_stat(dude_obj, 4) == 2) or (get_critter_stat(dude_obj, 4) == 3)) then begin
-        giq_option(2, 608, 108, Node003, 50);
+    if ((dude_iq == 2) or (dude_iq == 3)) then begin
+        giq_option(2, NAME, 108, Node003, NEUTRAL_REACTION);
     end
-    giq_option(4, 608, 290, Node064, 50);
-    giq_option(4, 608, 110, Node001a, 50);
-    if (global_var(303) == 1) then begin
-        giq_option(4, 608, 112, Node001b, 50);
+    giq_option(4, NAME, 290, Node064, NEUTRAL_REACTION);
+    giq_option(4, NAME, 110, Node001a, NEUTRAL_REACTION);
+    if (global_var(GVAR_BH_JAIL) == 1) then begin
+        giq_option(4, NAME, 112, Node001b, NEUTRAL_REACTION);
     end
-    giq_option(4, 608, 113, Node999, 50);
+    giq_option(4, NAME, 113, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node001a
@@ -747,39 +747,39 @@ end
 procedure Node001b
 begin
     set_local_var(14, 1);
-    set_global_var(303, 3);
+    set_global_var(GVAR_BH_JAIL, 3);
     call Node025();
 end
 
 procedure Node002
 begin
-    gsay_message(608, 114, 51);
+    gsay_message(NAME, 114, BAD_REACTION);
 end
 
 procedure Node003
 begin
     if (From_WTG == 1) then begin
         From_WTG := 0;
-        gsay_reply(608, 117);
+        Reply(117);
     end
     else begin
-        gsay_reply(608, 116);
+        Reply(116);
     end
-    giq_option(-3, 608, 118, Node004, 50);
-    giq_option(-3, 608, 119, Node005, 50);
+    giq_option(-3, NAME, 118, Node004, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 119, Node005, NEUTRAL_REACTION);
 end
 
 procedure Node004
 begin
-    gsay_message(608, 120, 50);
-    gdialog_mod_barter(0);
-    gsay_reply(608, 313);
-    giq_option(-3, 608, 314, Node006, 50);
+    gsay_message(NAME, 120, NEUTRAL_REACTION);
+    gdialog_barter;
+    Reply(313);
+    giq_option(-3, NAME, 314, Node006, NEUTRAL_REACTION);
 end
 
 procedure Node005
 begin
-    gsay_message(608, 122, 50);
+    gsay_message(NAME, 122, NEUTRAL_REACTION);
 end
 
 procedure Node006
@@ -792,8 +792,8 @@ begin
     else begin
         item_caps_adjust(dude_obj, -item_caps_total(dude_obj));
     end
-    gsay_reply(608, 124);
-    giq_option(-3, 608, 125, Node999, 50);
+    Reply(124);
+    giq_option(-3, NAME, 125, Node999, NEUTRAL_REACTION);
     return 0;
     LVar0 := 0;
 end
@@ -802,10 +802,10 @@ procedure Node013
 begin
     From_13 := 1;
     set_local_var(14, 1);
-    gsay_reply(608, 148);
-    giq_option(4, 608, 149, Node013a, 50);
-    giq_option(4, 608, 150, Node013b, 50);
-    giq_option(4, 608, 151, Node013c, 50);
+    Reply(148);
+    giq_option(4, NAME, 149, Node013a, NEUTRAL_REACTION);
+    giq_option(4, NAME, 150, Node013b, NEUTRAL_REACTION);
+    giq_option(4, NAME, 151, Node013c, NEUTRAL_REACTION);
 end
 
 procedure Node013a
@@ -965,13 +965,13 @@ procedure Node014
 begin
     if (From_13 == 1) then begin
         From_13 := 0;
-        gsay_reply(608, message_str(608, 152) + " " + message_str(608, 153));
+        Reply(mstr(152) + " " + mstr(153));
     end
     else begin
-        gsay_reply(608, 153);
+        Reply(153);
     end
-    giq_option(4, 608, 154, Node014a, 50);
-    giq_option(4, 608, 155, Node018, 50);
+    giq_option(4, NAME, 154, Node014a, NEUTRAL_REACTION);
+    giq_option(4, NAME, 155, Node018, NEUTRAL_REACTION);
 end
 
 procedure Node014a
@@ -1026,168 +1026,168 @@ end
 
 procedure Node015
 begin
-    gsay_reply(608, 156);
-    giq_option(4, 608, 157, Node014, 50);
-    giq_option(4, 608, 158, Node013c, 50);
+    Reply(156);
+    giq_option(4, NAME, 157, Node014, NEUTRAL_REACTION);
+    giq_option(4, NAME, 158, Node013c, NEUTRAL_REACTION);
 end
 
 procedure Node016
 begin
-    gsay_reply(608, 159);
-    giq_option(4, 608, 160, Node001, 50);
-    giq_option(4, 608, 161, Node017, 50);
+    Reply(159);
+    giq_option(4, NAME, 160, Node001, NEUTRAL_REACTION);
+    giq_option(4, NAME, 161, Node017, NEUTRAL_REACTION);
 end
 
 procedure Node017
 begin
-    gsay_reply(608, 162);
-    giq_option(4, 608, 163, Node999, 50);
+    Reply(162);
+    giq_option(4, NAME, 163, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node018
 begin
-    gsay_reply(608, 164);
-    giq_option(4, 608, 165, Node019, 50);
-    giq_option(4, 608, 166, Node014a, 50);
+    Reply(164);
+    giq_option(4, NAME, 165, Node019, NEUTRAL_REACTION);
+    giq_option(4, NAME, 166, Node014a, NEUTRAL_REACTION);
 end
 
 procedure Node019
 begin
-    gsay_reply(608, 167);
-    giq_option(4, 608, 168, Node020, 50);
-    giq_option(4, 608, 169, Node021, 50);
+    Reply(167);
+    giq_option(4, NAME, 168, Node020, NEUTRAL_REACTION);
+    giq_option(4, NAME, 169, Node021, NEUTRAL_REACTION);
 end
 
 procedure Node020
 begin
-    gsay_reply(608, 170);
-    giq_option(4, 608, 171, Node021, 50);
-    giq_option(4, 608, 172, Node999, 50);
+    Reply(170);
+    giq_option(4, NAME, 171, Node021, NEUTRAL_REACTION);
+    giq_option(4, NAME, 172, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node021
 begin
-    gsay_reply(608, 173);
-    giq_option(4, 608, 174, Node021a, 50);
-    giq_option(4, 608, 175, Node021b, 50);
-    giq_option(4, 608, 176, Node021c, 50);
+    Reply(173);
+    giq_option(4, NAME, 174, Node021a, NEUTRAL_REACTION);
+    giq_option(4, NAME, 175, Node021b, NEUTRAL_REACTION);
+    giq_option(4, NAME, 176, Node021c, NEUTRAL_REACTION);
 end
 
 procedure Node021a
 begin
-    set_global_var(303, 2);
+    set_global_var(GVAR_BH_JAIL, 2);
     call Node022();
 end
 
 procedure Node021b
 begin
-    set_global_var(303, 2);
+    set_global_var(GVAR_BH_JAIL, 2);
     call Node024();
 end
 
 procedure Node021c
 begin
-    set_global_var(303, 3);
+    set_global_var(GVAR_BH_JAIL, 3);
     call Node025();
 end
 
 procedure Node022
 begin
-    gsay_reply(608, 177);
-    giq_option(4, 608, 178, Node023, 50);
-    giq_option(4, 608, 179, Node022a, 50);
+    Reply(177);
+    giq_option(4, NAME, 178, Node023, NEUTRAL_REACTION);
+    giq_option(4, NAME, 179, Node022a, NEUTRAL_REACTION);
 end
 
 procedure Node022a
 begin
-    set_global_var(303, 3);
+    set_global_var(GVAR_BH_JAIL, 3);
     call Node025();
 end
 
 procedure Node023
 begin
-    gsay_reply(608, 180);
-    giq_option(4, 608, 181, Node999, 50);
+    Reply(180);
+    giq_option(4, NAME, 181, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node024
 begin
-    gsay_reply(608, 182);
-    giq_option(4, 608, 183, Node998, 51);
+    Reply(182);
+    giq_option(4, NAME, 183, Node998, BAD_REACTION);
 end
 
 procedure Node025
 begin
-    gsay_reply(608, 184);
-    giq_option(4, 608, 185, Node026, 50);
-    giq_option(4, 608, 186, Node027, 50);
-    giq_option(4, 608, 187, Node025a, 50);
-    giq_option(4, 608, 188, Node999, 50);
+    Reply(184);
+    giq_option(4, NAME, 185, Node026, NEUTRAL_REACTION);
+    giq_option(4, NAME, 186, Node027, NEUTRAL_REACTION);
+    giq_option(4, NAME, 187, Node025a, NEUTRAL_REACTION);
+    giq_option(4, NAME, 188, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node025a
 begin
-    set_global_var(303, 2);
+    set_global_var(GVAR_BH_JAIL, 2);
     call Node024();
 end
 
 procedure Node026
 begin
-    gsay_reply(608, 189);
-    giq_option(4, 608, 190, Node027, 50);
-    giq_option(4, 608, 191, Node999, 50);
+    Reply(189);
+    giq_option(4, NAME, 190, Node027, NEUTRAL_REACTION);
+    giq_option(4, NAME, 191, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node027
 begin
-    gsay_reply(608, 192);
-    giq_option(4, 608, 194, Node001a, 50);
-    giq_option(4, 608, 196, Node999, 50);
+    Reply(192);
+    giq_option(4, NAME, 194, Node001a, NEUTRAL_REACTION);
+    giq_option(4, NAME, 196, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node036
 begin
-    gsay_reply(608, 222);
-    giq_option(4, 608, 223, Node999, 50);
+    Reply(222);
+    giq_option(4, NAME, 223, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node037
 begin
-    gsay_reply(608, 224);
-    giq_option(4, 608, 225, Node040, 50);
+    Reply(224);
+    giq_option(4, NAME, 225, Node040, NEUTRAL_REACTION);
 end
 
 procedure Node038
 begin
-    gsay_reply(608, 226);
-    giq_option(4, 608, 227, Node040, 50);
+    Reply(226);
+    giq_option(4, NAME, 227, Node040, NEUTRAL_REACTION);
 end
 
 procedure Node039
 begin
-    gsay_reply(608, 228);
-    giq_option(4, 608, 229, Node040, 50);
+    Reply(228);
+    giq_option(4, NAME, 229, Node040, NEUTRAL_REACTION);
 end
 
 procedure Node040
 begin
-    if (global_var(304) != 4) then begin
-        gsay_reply(608, 230);
+    if (global_var(GVAR_BH_CONSPIRACY) != 4) then begin
+        Reply(230);
     end
     else begin
-        gsay_reply(608, 231);
+        Reply(231);
     end
-    if (global_var(304) != 4) then begin
-        giq_option(4, 608, 232, Node048, 50);
+    if (global_var(GVAR_BH_CONSPIRACY) != 4) then begin
+        giq_option(4, NAME, 232, Node048, NEUTRAL_REACTION);
     end
-    giq_option(4, 608, 235, Node999, 50);
+    giq_option(4, NAME, 235, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node048
 begin
-    gsay_reply(608, 258);
-    giq_option(4, 608, 259, Node049, 50);
-    giq_option(4, 608, 260, Node048a, 50);
+    Reply(258);
+    giq_option(4, NAME, 259, Node049, NEUTRAL_REACTION);
+    giq_option(4, NAME, 260, Node048a, NEUTRAL_REACTION);
 end
 
 procedure Node048a
@@ -1244,38 +1244,38 @@ end
 procedure Node049
 begin
     variable LVar0 := 0;
-    if (global_var(304) == 4) then begin
-        gsay_reply(608, 262);
+    if (global_var(GVAR_BH_CONSPIRACY) == 4) then begin
+        Reply(262);
     end
     else begin
         if (made_trigger == 0) then begin
             made_trigger := 1;
-            LVar0 := create_object_sid(454, 0, 0, -1);
+            LVar0 := create_object(454, 0, 0);
             add_obj_to_inven(dude_obj, LVar0);
-            gsay_reply(608, message_str(608, 261) + " " + message_str(608, 262));
+            Reply(mstr(261) + " " + mstr(262));
         end
     end
-    giq_option(4, 608, 263, Node049a, 50);
-    giq_option(4, 608, 264, Node048a, 50);
+    giq_option(4, NAME, 263, Node049a, NEUTRAL_REACTION);
+    giq_option(4, NAME, 264, Node048a, NEUTRAL_REACTION);
     return 0;
     LVar0 := 0;
 end
 
 procedure Node049a
 begin
-    set_global_var(304, 4);
+    set_global_var(GVAR_BH_CONSPIRACY, 4);
 end
 
 procedure Node050
 begin
-    gsay_reply(608, 265);
-    giq_option(4, 608, 266, Node050a, 50);
-    giq_option(4, 608, 267, Node049, 50);
+    Reply(265);
+    giq_option(4, NAME, 266, Node050a, NEUTRAL_REACTION);
+    giq_option(4, NAME, 267, Node049, NEUTRAL_REACTION);
 end
 
 procedure Node050a
 begin
-    set_global_var(304, 3);
+    set_global_var(GVAR_BH_CONSPIRACY, 3);
     call Node051();
 end
 
@@ -1290,7 +1290,7 @@ begin
         LVar2 := rm_mult_objs_from_inven(dude_obj, LVar0, LVar1);
         destroy_object(LVar0);
     end
-    gsay_message(608, 268, 51);
+    gsay_message(NAME, 268, BAD_REACTION);
     return 0;
     LVar0 := 0;
     LVar1 := 0;
@@ -1299,43 +1299,43 @@ end
 
 procedure Node052
 begin
-    gsay_reply(608, 270);
-    giq_option(4, 608, 272, Node053, 50);
-    giq_option(4, 608, 273, Node048a, 50);
+    Reply(270);
+    giq_option(4, NAME, 272, Node053, NEUTRAL_REACTION);
+    giq_option(4, NAME, 273, Node048a, NEUTRAL_REACTION);
 end
 
 procedure Node053
 begin
-    gsay_reply(608, 274);
-    giq_option(4, 608, 275, Node999, 50);
-    giq_option(4, 608, 276, Node048a, 50);
+    Reply(274);
+    giq_option(4, NAME, 275, Node999, NEUTRAL_REACTION);
+    giq_option(4, NAME, 276, Node048a, NEUTRAL_REACTION);
 end
 
 procedure Node054
 begin
-    float_msg(self_obj, message_str(608, 277), 8);
+    float_msg(self_obj, mstr(277), FLOAT_MSG_YELLOW);
 end
 
 procedure Node055
 begin
-    float_msg(self_obj, message_str(608, 278), 8);
+    float_msg(self_obj, mstr(278), FLOAT_MSG_YELLOW);
 end
 
 procedure Node056
 begin
     set_map_var(17, 1);
     set_map_var(21, 1);
-    gsay_message(608, 279, 49);
+    gsay_message(NAME, 279, GOOD_REACTION);
 end
 
 procedure Node057
 begin
-    float_msg(self_obj, message_str(608, 280), 8);
+    float_msg(self_obj, mstr(280), FLOAT_MSG_YELLOW);
 end
 
 procedure Node058
 begin
-    float_msg(self_obj, message_str(608, 281), 8);
+    float_msg(self_obj, mstr(281), FLOAT_MSG_YELLOW);
 end
 
 procedure Node061
@@ -1354,82 +1354,81 @@ end
 procedure Node064
 begin
     gdialog_mod_barter(-20);
-    gsay_reply(608, 313);
-    giq_option(4, 608, 314, Node064a, 50);
+    Reply(313);
+    giq_option(4, NAME, 314, Node064a, NEUTRAL_REACTION);
 end
 
 procedure Node064a
 begin
-    gsay_reply(608, 291);
-    giq_option(4, 608, 292, Node001, 50);
-    giq_option(4, 608, 293, Node999, 50);
+    Reply(291);
+    giq_option(4, NAME, 292, Node001, NEUTRAL_REACTION);
+    giq_option(4, NAME, 293, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node065
 begin
-    float_msg(self_obj, message_str(608, 294), 8);
+    float_msg(self_obj, mstr(294), FLOAT_MSG_YELLOW);
 end
 
 procedure Node066
 begin
-    float_msg(self_obj, message_str(608, 295), 8);
+    float_msg(self_obj, mstr(295), FLOAT_MSG_YELLOW);
 end
 
 procedure Node067
 begin
-    gsay_reply(608, 296);
-    giq_option(4, 608, 297, Node999, 50);
+    Reply(296);
+    giq_option(4, NAME, 297, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node068
 begin
-    gsay_reply(608, 298);
-    giq_option(4, 608, 299, Node069, 50);
+    Reply(298);
+    giq_option(4, NAME, 299, Node069, NEUTRAL_REACTION);
 end
 
 procedure Node069
 begin
-    gsay_message(608, 300, 51);
+    gsay_message(NAME, 300, BAD_REACTION);
     call Node998();
 end
 
 procedure Node070
 begin
-    gsay_reply(608, 301);
-    giq_option(-3, 608, 302, Node999, 50);
-    giq_option(4, 608, 303, Node999, 50);
-    giq_option(4, 608, 304, Node071, 50);
+    Reply(301);
+    giq_option(-3, NAME, 302, Node999, NEUTRAL_REACTION);
+    giq_option(4, NAME, 303, Node999, NEUTRAL_REACTION);
+    giq_option(4, NAME, 304, Node071, NEUTRAL_REACTION);
 end
 
 procedure Node071
 begin
-    gsay_reply(608, 305);
-    giq_option(4, 608, 306, Node072, 50);
-    giq_option(4, 608, 307, Node073, 50);
+    Reply(305);
+    giq_option(4, NAME, 306, Node072, NEUTRAL_REACTION);
+    giq_option(4, NAME, 307, Node073, NEUTRAL_REACTION);
 end
 
 procedure Node072
 begin
-    gsay_message(608, 308, 51);
+    gsay_message(NAME, 308, BAD_REACTION);
     call Node998();
 end
 
 procedure Node073
 begin
-    gsay_message(608, 309, 50);
+    gsay_message(NAME, 309, NEUTRAL_REACTION);
 end
 
 procedure Node074
 begin
-    gsay_reply(608, 310);
-    giq_option(4, 608, 311, Node001, 50);
-    giq_option(4, 608, 312, Node999, 50);
+    Reply(310);
+    giq_option(4, NAME, 311, Node001, NEUTRAL_REACTION);
+    giq_option(4, NAME, 312, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node075
 begin
     set_local_var(14, 1);
-    gsay_reply(608, 315);
-    giq_option(4, 608, 316, Node040, 50);
+    Reply(315);
+    giq_option(4, NAME, 316, Node040, NEUTRAL_REACTION);
 end
-
