diff --git a/Mods/MegaMod/Original/HCLIZ.ssl b/Mods/MegaMod/ReDefine/HCLIZ.ssl
index 94171ab..0f669af 100644
--- a/Mods/MegaMod/Original/HCLIZ.ssl
+++ b/Mods/MegaMod/ReDefine/HCLIZ.ssl
@@ -73,9 +73,9 @@ end
 procedure critter_p_proc
 begin
     variable LVar0 := 0;
-    if ((global_var(525) == 1) or (local_var(5) == 2) and obj_can_see_obj(self_obj, dude_obj)) then begin
+    if ((global_var(GVAR_ENEMY_CONSPIRATORS) == 1) or (local_var(5) == 2) and obj_can_see_obj(self_obj, dude_obj)) then begin
         set_local_var(5, 1);
-        attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+        attack(dude_obj);
     end
     LVar0 := game_time_hour;
     if ((LVar0 >= 800) and (LVar0 < 2000)) then begin
@@ -93,19 +93,19 @@ begin
             set_map_var(18, 0);
             set_local_var(9, 1);
             if ((LVar0 >= 800) and (LVar0 < 2000)) then begin
-                float_msg(self_obj, message_str(597, 174), 8);
+                float_msg(self_obj, mstr(174), FLOAT_MSG_YELLOW);
             end
             else begin
-                float_msg(self_obj, message_str(597, 185), 8);
+                float_msg(self_obj, mstr(185), FLOAT_MSG_YELLOW);
             end
         end
         else begin
             set_map_var(18, 0);
             if ((LVar0 >= 800) and (LVar0 < 2000)) then begin
-                float_msg(self_obj, message_str(597, 175), 8);
+                float_msg(self_obj, mstr(175), FLOAT_MSG_YELLOW);
             end
             else begin
-                float_msg(self_obj, message_str(597, 186), 8);
+                float_msg(self_obj, mstr(186), FLOAT_MSG_YELLOW);
             end
             call Node998();
         end
@@ -115,15 +115,15 @@ begin
         if ((map_var(22) == 0) or (map_var(22) == 2)) then begin
             if (local_var(10) == 0) then begin
                 set_local_var(10, 1);
-                float_msg(self_obj, message_str(597, 179), 8);
+                float_msg(self_obj, mstr(179), FLOAT_MSG_YELLOW);
             end
             else begin
                 if (local_var(10) == 1) then begin
                     set_local_var(10, 2);
-                    float_msg(self_obj, message_str(597, 180), 8);
+                    float_msg(self_obj, mstr(180), FLOAT_MSG_YELLOW);
                 end
                 else begin
-                    float_msg(self_obj, message_str(597, 181), 8);
+                    float_msg(self_obj, mstr(181), FLOAT_MSG_YELLOW);
                     call Node998();
                 end
             end
@@ -147,7 +147,7 @@ begin
     From_WTG := 1;
     if (local_var(2) == 0) then begin
         set_local_var(2, 1);
-        set_local_var(3, (get_critter_stat(dude_obj, 3) - get_critter_stat(self_obj, 3)) * 5);
+        set_local_var(3, (dude_charisma - self_charisma) * 5);
         debug_msg("Base Reaction == " + local_var(3));
     end
     Static_Reaction := 0;
@@ -160,14 +160,14 @@ begin
     else begin
         reaction_bonus_karma := -1 * (-1 * reaction_bonus_karma / 1000);
     end
-    if (global_var(54) >= 0) then begin
-        reaction_bonus_town_rep := global_var(54) / 2;
+    if (global_var(GVAR_TOWN_REP_BROKEN_HILLS) >= 0) then begin
+        reaction_bonus_town_rep := global_var(GVAR_TOWN_REP_BROKEN_HILLS) / 2;
     end
     else begin
-        reaction_bonus_town_rep := -1 * (-1 * global_var(54) / 2);
+        reaction_bonus_town_rep := -1 * (-1 * global_var(GVAR_TOWN_REP_BROKEN_HILLS) / 2);
     end
-    if (has_trait(2, dude_obj, 13)) then begin
-        if (get_critter_stat(dude_obj, 34) == get_critter_stat(self_obj, 34)) then begin
+    if (has_trait(TRAIT_TRAIT, dude_obj, TRAIT_sex_appeal)) then begin
+        if (dude_gender == self_gender) then begin
             Static_Reaction := Static_Reaction + -20;
         end
         else begin
@@ -193,48 +193,48 @@ begin
             Static_Reaction := Static_Reaction + reaction_bonus_town_rep;
         end
         debug_msg("Town Rep Bonus == " + Static_Reaction);
-        set_global_var(37, 0);
-        set_global_var(38, 0);
-        set_global_var(39, 0);
-        set_global_var(40, 0);
-        set_global_var(41, 0);
-        set_global_var(42, 0);
-        set_global_var(43, 0);
-        set_global_var(44, 0);
-        set_global_var(45, 0);
-        if (global_var(0) >= 1000) then begin
-            set_global_var(37, 1);
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+        set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+        set_global_var(GVAR_KARMA_DEFENDER, 0);
+        set_global_var(GVAR_KARMA_WANDERER, 0);
+        set_global_var(GVAR_KARMA_BETRAYER, 0);
+        set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+        set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+            set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
         end
         else begin
-            if (global_var(0) >= 750) then begin
-                set_global_var(38, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
             end
             else begin
-                if (global_var(0) >= 500) then begin
-                    set_global_var(39, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                 end
                 else begin
-                    if (global_var(0) >= 250) then begin
-                        set_global_var(40, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                        set_global_var(GVAR_KARMA_DEFENDER, 1);
                     end
                     else begin
-                        if (global_var(0) > -250) then begin
-                            set_global_var(41, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                            set_global_var(GVAR_KARMA_WANDERER, 1);
                         end
                         else begin
-                            if (global_var(0) > -500) then begin
-                                set_global_var(42, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                set_global_var(GVAR_KARMA_BETRAYER, 1);
                             end
                             else begin
-                                if (global_var(0) > -750) then begin
-                                    set_global_var(43, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -1000) then begin
-                                        set_global_var(44, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                     end
                                     else begin
-                                        set_global_var(45, 1);
+                                        set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                     end
                                 end
                             end
@@ -243,35 +243,35 @@ begin
                 end
             end
         end
-        if (global_var(37) == 1) then begin
+        if (global_var(GVAR_KARMA_HOLY_WARRIOR) == 1) then begin
             Static_Reaction := Static_Reaction - 20;
         end
         else begin
-            if (global_var(38) == 1) then begin
+            if (global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES) == 1) then begin
                 Static_Reaction := Static_Reaction - 15;
             end
             else begin
-                if (global_var(39) == 1) then begin
+                if (global_var(GVAR_KARMA_SHIELD_OF_HOPE) == 1) then begin
                     Static_Reaction := Static_Reaction - 10;
                 end
                 else begin
-                    if (global_var(40) == 1) then begin
+                    if (global_var(GVAR_KARMA_DEFENDER) == 1) then begin
                         Static_Reaction := Static_Reaction - 5;
                     end
                     else begin
-                        if (global_var(41) == 1) then begin
+                        if (global_var(GVAR_KARMA_WANDERER) == 1) then begin
                             Static_Reaction := Static_Reaction - 0;
                         end
                         else begin
-                            if (global_var(42) == 1) then begin
+                            if (global_var(GVAR_KARMA_BETRAYER) == 1) then begin
                                 Static_Reaction := Static_Reaction + -5;
                             end
                             else begin
-                                if (global_var(43) == 1) then begin
+                                if (global_var(GVAR_KARMA_SWORD_OF_DESPAIR) == 1) then begin
                                     Static_Reaction := Static_Reaction + -10;
                                 end
                                 else begin
-                                    if (global_var(44) == 1) then begin
+                                    if (global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE) == 1) then begin
                                         Static_Reaction := Static_Reaction + -15;
                                     end
                                     else begin
@@ -291,48 +291,48 @@ begin
         debug_msg("Karma Reaction Bonus == " + Static_Reaction);
         Static_Reaction := Static_Reaction + reaction_bonus_town_rep;
         debug_msg("Town Rep Bonus == " + Static_Reaction);
-        set_global_var(37, 0);
-        set_global_var(38, 0);
-        set_global_var(39, 0);
-        set_global_var(40, 0);
-        set_global_var(41, 0);
-        set_global_var(42, 0);
-        set_global_var(43, 0);
-        set_global_var(44, 0);
-        set_global_var(45, 0);
-        if (global_var(0) >= 1000) then begin
-            set_global_var(37, 1);
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+        set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+        set_global_var(GVAR_KARMA_DEFENDER, 0);
+        set_global_var(GVAR_KARMA_WANDERER, 0);
+        set_global_var(GVAR_KARMA_BETRAYER, 0);
+        set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+        set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+            set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
         end
         else begin
-            if (global_var(0) >= 750) then begin
-                set_global_var(38, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
             end
             else begin
-                if (global_var(0) >= 500) then begin
-                    set_global_var(39, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                 end
                 else begin
-                    if (global_var(0) >= 250) then begin
-                        set_global_var(40, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                        set_global_var(GVAR_KARMA_DEFENDER, 1);
                     end
                     else begin
-                        if (global_var(0) > -250) then begin
-                            set_global_var(41, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                            set_global_var(GVAR_KARMA_WANDERER, 1);
                         end
                         else begin
-                            if (global_var(0) > -500) then begin
-                                set_global_var(42, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                set_global_var(GVAR_KARMA_BETRAYER, 1);
                             end
                             else begin
-                                if (global_var(0) > -750) then begin
-                                    set_global_var(43, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -1000) then begin
-                                        set_global_var(44, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                     end
                                     else begin
-                                        set_global_var(45, 1);
+                                        set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                     end
                                 end
                             end
@@ -341,38 +341,38 @@ begin
                 end
             end
         end
-        if (has_trait(0, dude_obj, 95)) then begin
-            karma_tmp := global_var(0) * 2;
+        if (has_trait(TRAIT_PERK, dude_obj, PERK_karma_beacon_perk)) then begin
+            karma_tmp := global_var(GVAR_PLAYER_REPUTATION) * 2;
         end
-        if ((global_var(37) == 1) or ((karma_tmp >= 1000) and (karma_tmp != 0))) then begin
+        if ((global_var(GVAR_KARMA_HOLY_WARRIOR) == 1) or ((karma_tmp >= 1000) and (karma_tmp != 0))) then begin
             Static_Reaction := Static_Reaction + 20;
         end
         else begin
-            if ((global_var(38) == 1) or ((karma_tmp >= 750) and (karma_tmp != 0))) then begin
+            if ((global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES) == 1) or ((karma_tmp >= 750) and (karma_tmp != 0))) then begin
                 Static_Reaction := Static_Reaction + 15;
             end
             else begin
-                if ((global_var(39) == 1) or ((karma_tmp >= 500) and (karma_tmp != 0))) then begin
+                if ((global_var(GVAR_KARMA_SHIELD_OF_HOPE) == 1) or ((karma_tmp >= 500) and (karma_tmp != 0))) then begin
                     Static_Reaction := Static_Reaction + 10;
                 end
                 else begin
-                    if ((global_var(40) == 1) or ((karma_tmp >= 250) and (karma_tmp != 0))) then begin
+                    if ((global_var(GVAR_KARMA_DEFENDER) == 1) or ((karma_tmp >= 250) and (karma_tmp != 0))) then begin
                         Static_Reaction := Static_Reaction + 5;
                     end
                     else begin
-                        if ((global_var(41) == 1) or ((karma_tmp >= 0) and (karma_tmp != 0))) then begin
+                        if ((global_var(GVAR_KARMA_WANDERER) == 1) or ((karma_tmp >= 0) and (karma_tmp != 0))) then begin
                             Static_Reaction := Static_Reaction + 0;
                         end
                         else begin
-                            if ((global_var(42) == 1) or ((karma_tmp >= -250) and (karma_tmp != 0))) then begin
+                            if ((global_var(GVAR_KARMA_BETRAYER) == 1) or ((karma_tmp >= -250) and (karma_tmp != 0))) then begin
                                 Static_Reaction := Static_Reaction + -5;
                             end
                             else begin
-                                if ((global_var(43) == 1) or ((karma_tmp >= -500) and (karma_tmp != 0))) then begin
+                                if ((global_var(GVAR_KARMA_SWORD_OF_DESPAIR) == 1) or ((karma_tmp >= -500) and (karma_tmp != 0))) then begin
                                     Static_Reaction := Static_Reaction + -10;
                                 end
                                 else begin
-                                    if ((global_var(44) == 1) or ((karma_tmp >= -750) and (karma_tmp != 0))) then begin
+                                    if ((global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE) == 1) or ((karma_tmp >= -750) and (karma_tmp != 0))) then begin
                                         Static_Reaction := Static_Reaction + -15;
                                     end
                                     else begin
@@ -388,32 +388,32 @@ begin
         debug_msg("Good Person Reaction Bonus == " + Static_Reaction);
     end
     if (Slavery_Tolerant == 0) then begin
-        if (global_var(11) == 1) then begin
+        if (global_var(GVAR_REPUTATION_SLAVER) == 1) then begin
             Static_Reaction := Static_Reaction + (2 * -25);
         end
     end
     else begin
         if (Slavery_Tolerant == 1) then begin
-            if (global_var(11) == 1) then begin
+            if (global_var(GVAR_REPUTATION_SLAVER) == 1) then begin
                 Static_Reaction := Static_Reaction + -25;
             end
         end
         else begin
             if (Slavery_Tolerant == 3) then begin
-                if (global_var(11) == 1) then begin
+                if (global_var(GVAR_REPUTATION_SLAVER) == 1) then begin
                     Static_Reaction := Static_Reaction - -25;
                 end
             end
         end
     end
     debug_msg("Slaver + Aligned Reaction == " + Static_Reaction);
-    Static_Reaction := Static_Reaction + (10 * has_trait(0, dude_obj, 10));
+    Static_Reaction := Static_Reaction + (10 * has_trait(TRAIT_PERK, dude_obj, PERK_presence));
     debug_msg("Presence Reaction == " + Static_Reaction);
-    if (global_var(1) >= 2) then begin
+    if (global_var(GVAR_CHILDKILLER_REPUTATION) >= 2) then begin
         Static_Reaction := Static_Reaction + -30;
     end
     debug_msg("Childkiller Reaction == " + Static_Reaction);
-    if (has_trait(0, dude_obj, 39)) then begin
+    if (has_trait(TRAIT_PERK, dude_obj, PERK_cult_of_personality)) then begin
         if (Evil_Critter == 1) then begin
             if (Static_Reaction > 0) then begin
                 Static_Reaction := Static_Reaction * -1;
@@ -495,11 +495,11 @@ begin
                     call Node001();
                 end
                 else begin
-                    if ((get_critter_stat(dude_obj, 4) == 2) or (get_critter_stat(dude_obj, 4) == 3)) then begin
+                    if ((dude_iq == 2) or (dude_iq == 3)) then begin
                         call Node003();
                     end
                     else begin
-                        if (get_critter_stat(dude_obj, 4) == 1) then begin
+                        if (dude_iq == 1) then begin
                             call Node002();
                         end
                         else begin
@@ -526,59 +526,59 @@ procedure destroy_p_proc
 begin
     set_map_var(23, 1);
     if (source_obj == dude_obj) then begin
-        set_global_var(5, global_var(5) + 1);
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(4) > (3 * global_var(5))) or (global_var(2) == 1)) and (global_var(1) == 0)) then begin
-            set_global_var(2, 1);
-            set_global_var(3, 0);
+        set_global_var(GVAR_GOOD_MONSTER, global_var(GVAR_GOOD_MONSTER) + 1);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(4) > (3 * global_var(GVAR_GOOD_MONSTER))) or (global_var(GVAR_CHAMPION_REPUTATION) == 1)) and (global_var(GVAR_CHILDKILLER_REPUTATION) == 0)) then begin
+            set_global_var(GVAR_CHAMPION_REPUTATION, 1);
+            set_global_var(GVAR_BERSERKER_REPUTATION, 0);
         end
-        if (metarule(51, self_obj) == 2) then begin
+        if (critter_kill_type(self_obj) == 2) then begin
             if (source_obj == dude_obj) then begin
-                set_global_var(2, 0);
-                set_global_var(1, global_var(1) + 1);
-                set_global_var(0, global_var(0) + -15);
+                set_global_var(GVAR_CHAMPION_REPUTATION, 0);
+                set_global_var(GVAR_CHILDKILLER_REPUTATION, global_var(GVAR_CHILDKILLER_REPUTATION) + 1);
+                set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + -15);
                 debug_msg("Player gains " + -15 + " Karma Points.");
-                set_global_var(37, 0);
-                set_global_var(38, 0);
-                set_global_var(39, 0);
-                set_global_var(40, 0);
-                set_global_var(41, 0);
-                set_global_var(42, 0);
-                set_global_var(43, 0);
-                set_global_var(44, 0);
-                set_global_var(45, 0);
-                if (global_var(0) >= 1000) then begin
-                    set_global_var(37, 1);
+                set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+                set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+                set_global_var(GVAR_KARMA_DEFENDER, 0);
+                set_global_var(GVAR_KARMA_WANDERER, 0);
+                set_global_var(GVAR_KARMA_BETRAYER, 0);
+                set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+                set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+                set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+                    set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
                 end
                 else begin
-                    if (global_var(0) >= 750) then begin
-                        set_global_var(38, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
                     end
                     else begin
-                        if (global_var(0) >= 500) then begin
-                            set_global_var(39, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                            set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                         end
                         else begin
-                            if (global_var(0) >= 250) then begin
-                                set_global_var(40, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                                set_global_var(GVAR_KARMA_DEFENDER, 1);
                             end
                             else begin
-                                if (global_var(0) > -250) then begin
-                                    set_global_var(41, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                                    set_global_var(GVAR_KARMA_WANDERER, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -500) then begin
-                                        set_global_var(42, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                        set_global_var(GVAR_KARMA_BETRAYER, 1);
                                     end
                                     else begin
-                                        if (global_var(0) > -750) then begin
-                                            set_global_var(43, 1);
+                                        if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                            set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                         end
                                         else begin
-                                            if (global_var(0) > -1000) then begin
-                                                set_global_var(44, 1);
+                                            if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                                set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                             end
                                             else begin
-                                                set_global_var(45, 1);
+                                                set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                             end
                                         end
                                     end
@@ -587,58 +587,58 @@ begin
                         end
                     end
                 end
-                set_global_var(54, global_var(54) + -8);
+                set_global_var(GVAR_TOWN_REP_BROKEN_HILLS, global_var(GVAR_TOWN_REP_BROKEN_HILLS) + -8);
                 debug_msg("Added " + -8 + " to Town Rep");
             end
         end
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(5) > (2 * global_var(4))) or (global_var(3) == 1))) then begin
-            set_global_var(3, 1);
-            set_global_var(2, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
+            set_global_var(GVAR_BERSERKER_REPUTATION, 1);
+            set_global_var(GVAR_CHAMPION_REPUTATION, 0);
         end
-        set_global_var(0, global_var(0) + -10);
+        set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + -10);
         debug_msg("Player gains " + -10 + " Karma Points.");
-        set_global_var(37, 0);
-        set_global_var(38, 0);
-        set_global_var(39, 0);
-        set_global_var(40, 0);
-        set_global_var(41, 0);
-        set_global_var(42, 0);
-        set_global_var(43, 0);
-        set_global_var(44, 0);
-        set_global_var(45, 0);
-        if (global_var(0) >= 1000) then begin
-            set_global_var(37, 1);
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+        set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+        set_global_var(GVAR_KARMA_DEFENDER, 0);
+        set_global_var(GVAR_KARMA_WANDERER, 0);
+        set_global_var(GVAR_KARMA_BETRAYER, 0);
+        set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+        set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+            set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
         end
         else begin
-            if (global_var(0) >= 750) then begin
-                set_global_var(38, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
             end
             else begin
-                if (global_var(0) >= 500) then begin
-                    set_global_var(39, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                 end
                 else begin
-                    if (global_var(0) >= 250) then begin
-                        set_global_var(40, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                        set_global_var(GVAR_KARMA_DEFENDER, 1);
                     end
                     else begin
-                        if (global_var(0) > -250) then begin
-                            set_global_var(41, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                            set_global_var(GVAR_KARMA_WANDERER, 1);
                         end
                         else begin
-                            if (global_var(0) > -500) then begin
-                                set_global_var(42, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                set_global_var(GVAR_KARMA_BETRAYER, 1);
                             end
                             else begin
-                                if (global_var(0) > -750) then begin
-                                    set_global_var(43, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -1000) then begin
-                                        set_global_var(44, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                     end
                                     else begin
-                                        set_global_var(45, 1);
+                                        set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                     end
                                 end
                             end
@@ -647,7 +647,7 @@ begin
                 end
             end
         end
-        set_global_var(54, global_var(54) + -5);
+        set_global_var(GVAR_TOWN_REP_BROKEN_HILLS, global_var(GVAR_TOWN_REP_BROKEN_HILLS) + -5);
         debug_msg("Added " + -5 + " to Town Rep");
     end
 end
@@ -655,13 +655,13 @@ end
 procedure look_at_p_proc
 begin
     script_overrides;
-    display_msg(message_str(597, 100));
+    display_mstr(100);
 end
 
 procedure description_p_proc
 begin
     script_overrides;
-    display_msg(message_str(597, 101));
+    display_mstr(101);
 end
 
 procedure use_skill_on_p_proc
@@ -672,8 +672,8 @@ procedure damage_p_proc
 begin
     if (source_obj == dude_obj) then begin
         set_local_var(6, 1);
-        set_global_var(525, 1);
-        set_global_var(309, 1);
+        set_global_var(GVAR_ENEMY_CONSPIRATORS, 1);
+        set_global_var(GVAR_ENEMY_BROKEN_HILLS, 1);
     end
 end
 
@@ -681,8 +681,8 @@ procedure map_enter_p_proc
 begin
     variable LVar0 := 0;
     Only_Once := 0;
-    critter_add_trait(self_obj, 1, 6, 150);
-    critter_add_trait(self_obj, 1, 5, 24);
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 150);
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_TOUGH_CITIZEN);
     LVar0 := game_time_hour;
     if ((LVar0 >= 800) and (LVar0 < 2000)) then begin
         move_to(self_obj, 12718, 0);
@@ -702,7 +702,7 @@ end
 
 procedure Node995
 begin
-    if (get_critter_stat(dude_obj, 3) >= 7) then begin
+    if (dude_charisma >= 7) then begin
         set_local_var(3, 100);
         set_local_var(1, 4);
     end
@@ -720,7 +720,7 @@ end
 
 procedure Node997
 begin
-    if (get_critter_stat(dude_obj, 3) >= 8) then begin
+    if (dude_charisma >= 8) then begin
         call Node004();
     end
     else begin
@@ -740,63 +740,63 @@ end
 procedure Node001
 begin
     if (local_var(4) == 1) then begin
-        gsay_reply(597, 104);
+        Reply(104);
     end
     else begin
         set_local_var(4, 1);
-        gsay_reply(597, 103);
+        Reply(103);
     end
-    if (get_critter_stat(dude_obj, 4) == 1) then begin
-        giq_option(1, 597, 105, Node002, 50);
+    if (dude_iq == 1) then begin
+        giq_option(1, NAME, 105, Node002, NEUTRAL_REACTION);
     end
-    if ((get_critter_stat(dude_obj, 4) == 2) or (get_critter_stat(dude_obj, 4) == 3)) then begin
-        giq_option(2, 597, 106, Node003, 50);
+    if ((dude_iq == 2) or (dude_iq == 3)) then begin
+        giq_option(2, NAME, 106, Node003, NEUTRAL_REACTION);
     end
-    giq_option(4, 597, 107, Node007, 50);
-    giq_option(4, 597, 108, Node008, 50);
-    if (global_var(0) <= -250) then begin
-        giq_option(4, 597, 109, Node009, 50);
+    giq_option(4, NAME, 107, Node007, NEUTRAL_REACTION);
+    giq_option(4, NAME, 108, Node008, NEUTRAL_REACTION);
+    if (global_var(GVAR_PLAYER_REPUTATION) <= -250) then begin
+        giq_option(4, NAME, 109, Node009, NEUTRAL_REACTION);
     end
     if ((map_var(17) == 1) and (map_var(22) == 0)) then begin
-        giq_option(4, 597, 110, Node010, 50);
+        giq_option(4, NAME, 110, Node010, NEUTRAL_REACTION);
     end
     if (local_var(11) == 0) then begin
-        giq_option(4, 597, 200, Node030, 50);
+        giq_option(4, NAME, 200, Node030, NEUTRAL_REACTION);
     end
-    giq_option(4, 597, 112, Node999, 50);
+    giq_option(4, NAME, 112, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node002
 begin
-    gsay_reply(597, 113);
-    giq_option(-3, 597, 114, Node999, 50);
+    Reply(113);
+    giq_option(-3, NAME, 114, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node003
 begin
-    gsay_reply(597, 115);
-    giq_option(2, 597, 116, Node004, 50);
-    giq_option(2, 597, 117, Node999, 50);
+    Reply(115);
+    giq_option(2, NAME, 116, Node004, NEUTRAL_REACTION);
+    giq_option(2, NAME, 117, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node004
 begin
-    gsay_reply(597, 118);
-    giq_option(2, 597, 119, Node005, 50);
-    giq_option(2, 597, 120, Node999, 50);
+    Reply(118);
+    giq_option(2, NAME, 119, Node005, NEUTRAL_REACTION);
+    giq_option(2, NAME, 120, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node005
 begin
-    gsay_reply(597, 121);
-    giq_option(2, 597, 122, Node006, 50);
+    Reply(121);
+    giq_option(2, NAME, 122, Node006, NEUTRAL_REACTION);
 end
 
 procedure Node006
 begin
     variable LVar0 := 0;
     variable LVar1 := 0;
-    gdialog_mod_barter(0);
+    gdialog_barter;
     LVar0 := item_caps_total(dude_obj);
     LVar1 := LVar0 / 10;
     if (LVar1 < 10) then begin
@@ -805,8 +805,8 @@ begin
     else begin
         item_caps_adjust(dude_obj, -LVar1);
     end
-    gsay_reply(597, 183);
-    giq_option(-3, 597, 184, Node006a, 50);
+    Reply(183);
+    giq_option(-3, NAME, 184, Node006a, NEUTRAL_REACTION);
     return 0;
     LVar0 := 0;
     LVar1 := 0;
@@ -814,36 +814,36 @@ end
 
 procedure Node006a
 begin
-    gsay_reply(597, 123);
-    giq_option(-3, 597, 124, Node999, 50);
+    Reply(123);
+    giq_option(-3, NAME, 124, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node007
 begin
     From_7 := 1;
-    gsay_message(597, 125, 50);
+    gsay_message(NAME, 125, NEUTRAL_REACTION);
     call Node007a();
 end
 
 procedure Node007a
 begin
-    gdialog_mod_barter(0);
-    gsay_reply(597, 183);
-    giq_option(4, 597, 184, Node007b, 50);
+    gdialog_barter;
+    Reply(183);
+    giq_option(4, NAME, 184, Node007b, NEUTRAL_REACTION);
 end
 
 procedure Node007b
 begin
-    gsay_reply(597, 182);
-    giq_option(4, 597, 126, Node001, 50);
-    giq_option(4, 597, 127, Node999, 50);
+    Reply(182);
+    giq_option(4, NAME, 126, Node001, NEUTRAL_REACTION);
+    giq_option(4, NAME, 127, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node008
 begin
-    gsay_reply(597, 128);
-    giq_option(4, 597, 129, Node008a, 50);
-    giq_option(4, 597, 130, Node999, 50);
+    Reply(128);
+    giq_option(4, NAME, 129, Node008a, NEUTRAL_REACTION);
+    giq_option(4, NAME, 130, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node008a
@@ -899,23 +899,23 @@ end
 
 procedure Node009
 begin
-    gsay_reply(597, 131);
-    giq_option(4, 597, 132, Node011, 50);
-    giq_option(4, 597, 133, Node012, 50);
-    giq_option(4, 597, 134, Node998, 51);
-    giq_option(4, 597, 135, Node013, 50);
-    giq_option(4, 597, 136, Node999, 50);
+    Reply(131);
+    giq_option(4, NAME, 132, Node011, NEUTRAL_REACTION);
+    giq_option(4, NAME, 133, Node012, NEUTRAL_REACTION);
+    giq_option(4, NAME, 134, Node998, BAD_REACTION);
+    giq_option(4, NAME, 135, Node013, NEUTRAL_REACTION);
+    giq_option(4, NAME, 136, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node010
 begin
-    gsay_reply(597, 137);
+    Reply(137);
     if (map_var(21) == 1) then begin
-        giq_option(4, 597, 138, Node014, 50);
+        giq_option(4, NAME, 138, Node014, NEUTRAL_REACTION);
     end
-    giq_option(4, 597, 139, Node015, 50);
+    giq_option(4, NAME, 139, Node015, NEUTRAL_REACTION);
     if (map_var(20) == 1) then begin
-        giq_option(4, 597, 140, Node010a, 50);
+        giq_option(4, NAME, 140, Node010a, NEUTRAL_REACTION);
     end
 end
 
@@ -931,44 +931,44 @@ end
 
 procedure Node011
 begin
-    gsay_reply(597, 141);
-    giq_option(4, 597, 142, Node999, 50);
+    Reply(141);
+    giq_option(4, NAME, 142, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node012
 begin
-    gsay_reply(597, 143);
-    giq_option(4, 597, 144, Node016, 51);
-    giq_option(4, 597, 145, Node017, 51);
-    giq_option(4, 597, 146, Node018, 49);
-    giq_option(4, 597, 147, Node019, 50);
+    Reply(143);
+    giq_option(4, NAME, 144, Node016, BAD_REACTION);
+    giq_option(4, NAME, 145, Node017, BAD_REACTION);
+    giq_option(4, NAME, 146, Node018, GOOD_REACTION);
+    giq_option(4, NAME, 147, Node019, NEUTRAL_REACTION);
 end
 
 procedure Node013
 begin
-    gsay_reply(597, 148);
-    giq_option(4, 597, 149, Node012, 50);
-    giq_option(4, 597, 150, Node018, 50);
-    giq_option(4, 597, 151, Node018, 50);
+    Reply(148);
+    giq_option(4, NAME, 149, Node012, NEUTRAL_REACTION);
+    giq_option(4, NAME, 150, Node018, NEUTRAL_REACTION);
+    giq_option(4, NAME, 151, Node018, NEUTRAL_REACTION);
 end
 
 procedure Node014
 begin
-    gsay_reply(597, 152);
-    giq_option(4, 597, 153, Node020, 50);
-    giq_option(4, 597, 154, Node021, 50);
+    Reply(152);
+    giq_option(4, NAME, 153, Node020, NEUTRAL_REACTION);
+    giq_option(4, NAME, 154, Node021, NEUTRAL_REACTION);
 end
 
 procedure Node015
 begin
-    gsay_reply(597, 155);
-    giq_option(4, 597, 156, Node998, 50);
+    Reply(155);
+    giq_option(4, NAME, 156, Node998, NEUTRAL_REACTION);
 end
 
 procedure Node016
 begin
-    gsay_reply(597, 157);
-    giq_option(4, 597, 158, Node016a, 50);
+    Reply(157);
+    giq_option(4, NAME, 158, Node016a, NEUTRAL_REACTION);
 end
 
 procedure Node016a
@@ -1024,67 +1024,67 @@ end
 
 procedure Node017
 begin
-    gsay_reply(597, 159);
-    giq_option(4, 597, 160, Node001, 50);
-    giq_option(4, 597, 161, Node999, 50);
+    Reply(159);
+    giq_option(4, NAME, 160, Node001, NEUTRAL_REACTION);
+    giq_option(4, NAME, 161, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node018
 begin
-    gsay_reply(597, 162);
-    giq_option(4, 597, 163, Node999, 50);
+    Reply(162);
+    giq_option(4, NAME, 163, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node019
 begin
-    gsay_reply(597, 164);
-    giq_option(4, 597, 165, Node999, 50);
+    Reply(164);
+    giq_option(4, NAME, 165, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node020
 begin
     set_map_var(22, 1);
-    gsay_reply(597, 166);
-    giq_option(4, 597, 167, Node999, 50);
+    Reply(166);
+    giq_option(4, NAME, 167, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node021
 begin
     set_map_var(22, 2);
-    gsay_reply(597, 168);
-    giq_option(4, 597, 169, Node001, 50);
-    giq_option(4, 597, 170, Node999, 50);
+    Reply(168);
+    giq_option(4, NAME, 169, Node001, NEUTRAL_REACTION);
+    giq_option(4, NAME, 170, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node022
 begin
     From_WTG := 0;
-    gsay_message(597, 171, 51);
+    gsay_message(NAME, 171, BAD_REACTION);
     call Node999();
 end
 
 procedure Node023
 begin
-    float_msg(self_obj, message_str(597, 173), 8);
+    float_msg(self_obj, mstr(173), FLOAT_MSG_YELLOW);
 end
 
 procedure Node024
 begin
     set_map_var(22, 2);
-    gsay_message(597, 178, 51);
+    gsay_message(NAME, 178, BAD_REACTION);
     call Node998();
 end
 
 procedure Node030
 begin
     if (party_member_obj(16777558) != 0) then begin
-        gsay_reply(597, 201);
-        giq_option(4, 597, 202, Node031, 50);
-        giq_option(4, 597, 203, Node007, 50);
+        Reply(201);
+        giq_option(4, NAME, 202, Node031, NEUTRAL_REACTION);
+        giq_option(4, NAME, 203, Node007, NEUTRAL_REACTION);
     end
     else begin
-        gsay_reply(597, 204);
-        giq_option(4, 597, 205, Node007, 50);
+        Reply(204);
+        giq_option(4, NAME, 205, Node007, NEUTRAL_REACTION);
     end
 end
 
@@ -1094,7 +1094,7 @@ begin
     if (item_caps_total(dude_obj) > 1999) then begin
         set_local_var(11, 1);
         item_caps_adjust(dude_obj, -2000);
-        LVar0 := create_object_sid(676, 0, 0, -1);
+        LVar0 := create_object(676, 0, 0);
         add_obj_to_inven(dude_obj, LVar0);
         call Node999();
     end
@@ -1107,8 +1107,7 @@ end
 
 procedure Node032
 begin
-    gsay_reply(597, 206);
-    giq_option(4, 597, 207, Node006, 50);
-    giq_option(4, 597, 208, Node999, 50);
+    Reply(206);
+    giq_option(4, NAME, 207, Node006, NEUTRAL_REACTION);
+    giq_option(4, NAME, 208, Node999, NEUTRAL_REACTION);
 end
-
