diff --git a/Mods/MegaMod/Original/HCMANSON.ssl b/Mods/MegaMod/ReDefine/HCMANSON.ssl
index f5a188f..172b35e 100644
--- a/Mods/MegaMod/Original/HCMANSON.ssl
+++ b/Mods/MegaMod/ReDefine/HCMANSON.ssl
@@ -259,76 +259,76 @@ end
 
 procedure critter_p_proc
 begin
-    if ((global_var(525) == 1) or (local_var(5) == 2) and obj_can_see_obj(self_obj, dude_obj)) then begin
+    if ((global_var(GVAR_ENEMY_CONSPIRATORS) == 1) or (local_var(5) == 2) and obj_can_see_obj(self_obj, dude_obj)) then begin
         set_local_var(5, 1);
-        attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+        attack(dude_obj);
     end
     else begin
-        if (global_var(443) == 1) then begin
+        if (global_var(GVAR_JAIL_BREAK) == 1) then begin
             if (map_var(27) == 1) then begin
-                if (global_var(443) != 2) then begin
-                    set_global_var(443, 2);
-                    if ((global_var(303) == 3) or (global_var(303) == 2)) then begin
-                        set_global_var(303, 4);
+                if (global_var(GVAR_JAIL_BREAK) != 2) then begin
+                    set_global_var(GVAR_JAIL_BREAK, 2);
+                    if ((global_var(GVAR_BH_JAIL) == 3) or (global_var(GVAR_BH_JAIL) == 2)) then begin
+                        set_global_var(GVAR_BH_JAIL, 4);
                     end
                     else begin
-                        if (global_var(303) == 0) then begin
-                            set_global_var(303, 8);
+                        if (global_var(GVAR_BH_JAIL) == 0) then begin
+                            set_global_var(GVAR_BH_JAIL, 8);
                         end
                     end
                 end
-                display_msg(message_str(599, 178));
+                display_mstr(178);
                 give_exp_points(1500);
-                display_msg(message_str(14, 100) + 1500 + message_str(14, 101));
-                set_global_var(54, global_var(54) + -10);
-                set_global_var(0, global_var(0) + -10);
+                display_msg(g_mstr(100) + 1500 + g_mstr(101));
+                set_global_var(GVAR_TOWN_REP_BROKEN_HILLS, global_var(GVAR_TOWN_REP_BROKEN_HILLS) + -10);
+                set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + -10);
                 debug_msg("Player gains " + -10 + " Karma Points.");
-                set_global_var(37, 0);
-                set_global_var(38, 0);
-                set_global_var(39, 0);
-                set_global_var(40, 0);
-                set_global_var(41, 0);
-                set_global_var(42, 0);
-                set_global_var(43, 0);
-                set_global_var(44, 0);
-                set_global_var(45, 0);
-                GVar10 := global_var(0);
-                if (has_trait(0, dude_obj, 95)) then begin
-                    GVar10 := global_var(0) * 2;
+                set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+                set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+                set_global_var(GVAR_KARMA_DEFENDER, 0);
+                set_global_var(GVAR_KARMA_WANDERER, 0);
+                set_global_var(GVAR_KARMA_BETRAYER, 0);
+                set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+                set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+                set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+                GVar10 := global_var(GVAR_PLAYER_REPUTATION);
+                if (has_trait(TRAIT_PERK, dude_obj, PERK_karma_beacon_perk)) then begin
+                    GVar10 := global_var(GVAR_PLAYER_REPUTATION) * 2;
                 end
                 if (GVar10 >= 1000) then begin
-                    set_global_var(37, 1);
+                    set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
                 end
                 else begin
                     if (GVar10 >= 750) then begin
-                        set_global_var(38, 1);
+                        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
                     end
                     else begin
                         if (GVar10 >= 500) then begin
-                            set_global_var(39, 1);
+                            set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                         end
                         else begin
                             if (GVar10 >= 250) then begin
-                                set_global_var(40, 1);
+                                set_global_var(GVAR_KARMA_DEFENDER, 1);
                             end
                             else begin
                                 if (GVar10 > -250) then begin
-                                    set_global_var(41, 1);
+                                    set_global_var(GVAR_KARMA_WANDERER, 1);
                                 end
                                 else begin
                                     if (GVar10 > -500) then begin
-                                        set_global_var(42, 1);
+                                        set_global_var(GVAR_KARMA_BETRAYER, 1);
                                     end
                                     else begin
                                         if (GVar10 > -750) then begin
-                                            set_global_var(43, 1);
+                                            set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                         end
                                         else begin
                                             if (GVar10 > -1000) then begin
-                                                set_global_var(44, 1);
+                                                set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                             end
                                             else begin
-                                                set_global_var(45, 1);
+                                                set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                             end
                                         end
                                     end
@@ -338,7 +338,7 @@ begin
                     end
                 end
                 game_ui_disable;
-                float_msg(self_obj, message_str(599, 177), 8);
+                float_msg(self_obj, mstr(177), FLOAT_MSG_YELLOW);
                 add_timer_event(self_obj, game_ticks(3), 1);
             end
         end
@@ -356,7 +356,7 @@ procedure talk_p_proc
 begin
     if (local_var(2) == 0) then begin
         set_local_var(2, 1);
-        set_local_var(3, (get_critter_stat(dude_obj, 3) - get_critter_stat(self_obj, 3)) * 5);
+        set_local_var(3, (dude_charisma - self_charisma) * 5);
         debug_msg("Base Reaction == " + local_var(3));
     end
     GVar20 := 0;
@@ -368,14 +368,14 @@ begin
     else begin
         GVar25 := -1 * (-1 * GVar25 / 1000);
     end
-    if (global_var(54) >= 0) then begin
-        GVar24 := global_var(54) / 2;
+    if (global_var(GVAR_TOWN_REP_BROKEN_HILLS) >= 0) then begin
+        GVar24 := global_var(GVAR_TOWN_REP_BROKEN_HILLS) / 2;
     end
     else begin
-        GVar24 := -1 * (-1 * global_var(54) / 2);
+        GVar24 := -1 * (-1 * global_var(GVAR_TOWN_REP_BROKEN_HILLS) / 2);
     end
-    if (has_trait(2, dude_obj, 13)) then begin
-        if (get_critter_stat(dude_obj, 34) == get_critter_stat(self_obj, 34)) then begin
+    if (has_trait(TRAIT_TRAIT, dude_obj, TRAIT_sex_appeal)) then begin
+        if (dude_gender == self_gender) then begin
             GVar20 := GVar20 + -20;
         end
         else begin
@@ -401,52 +401,52 @@ begin
             GVar20 := GVar20 + GVar24;
         end
         debug_msg("Town Rep Bonus == " + GVar20);
-        set_global_var(37, 0);
-        set_global_var(38, 0);
-        set_global_var(39, 0);
-        set_global_var(40, 0);
-        set_global_var(41, 0);
-        set_global_var(42, 0);
-        set_global_var(43, 0);
-        set_global_var(44, 0);
-        set_global_var(45, 0);
-        GVar10 := global_var(0);
-        if (has_trait(0, dude_obj, 95)) then begin
-            GVar10 := global_var(0) * 2;
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+        set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+        set_global_var(GVAR_KARMA_DEFENDER, 0);
+        set_global_var(GVAR_KARMA_WANDERER, 0);
+        set_global_var(GVAR_KARMA_BETRAYER, 0);
+        set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+        set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+        GVar10 := global_var(GVAR_PLAYER_REPUTATION);
+        if (has_trait(TRAIT_PERK, dude_obj, PERK_karma_beacon_perk)) then begin
+            GVar10 := global_var(GVAR_PLAYER_REPUTATION) * 2;
         end
         if (GVar10 >= 1000) then begin
-            set_global_var(37, 1);
+            set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
         end
         else begin
             if (GVar10 >= 750) then begin
-                set_global_var(38, 1);
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
             end
             else begin
                 if (GVar10 >= 500) then begin
-                    set_global_var(39, 1);
+                    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                 end
                 else begin
                     if (GVar10 >= 250) then begin
-                        set_global_var(40, 1);
+                        set_global_var(GVAR_KARMA_DEFENDER, 1);
                     end
                     else begin
                         if (GVar10 > -250) then begin
-                            set_global_var(41, 1);
+                            set_global_var(GVAR_KARMA_WANDERER, 1);
                         end
                         else begin
                             if (GVar10 > -500) then begin
-                                set_global_var(42, 1);
+                                set_global_var(GVAR_KARMA_BETRAYER, 1);
                             end
                             else begin
                                 if (GVar10 > -750) then begin
-                                    set_global_var(43, 1);
+                                    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                 end
                                 else begin
                                     if (GVar10 > -1000) then begin
-                                        set_global_var(44, 1);
+                                        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                     end
                                     else begin
-                                        set_global_var(45, 1);
+                                        set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                     end
                                 end
                             end
@@ -455,35 +455,35 @@ begin
                 end
             end
         end
-        if (global_var(37) == 1) then begin
+        if (global_var(GVAR_KARMA_HOLY_WARRIOR) == 1) then begin
             GVar20 := GVar20 - 20;
         end
         else begin
-            if (global_var(38) == 1) then begin
+            if (global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES) == 1) then begin
                 GVar20 := GVar20 - 15;
             end
             else begin
-                if (global_var(39) == 1) then begin
+                if (global_var(GVAR_KARMA_SHIELD_OF_HOPE) == 1) then begin
                     GVar20 := GVar20 - 10;
                 end
                 else begin
-                    if (global_var(40) == 1) then begin
+                    if (global_var(GVAR_KARMA_DEFENDER) == 1) then begin
                         GVar20 := GVar20 - 5;
                     end
                     else begin
-                        if (global_var(41) == 1) then begin
+                        if (global_var(GVAR_KARMA_WANDERER) == 1) then begin
                             GVar20 := GVar20 - 0;
                         end
                         else begin
-                            if (global_var(42) == 1) then begin
+                            if (global_var(GVAR_KARMA_BETRAYER) == 1) then begin
                                 GVar20 := GVar20 + -5;
                             end
                             else begin
-                                if (global_var(43) == 1) then begin
+                                if (global_var(GVAR_KARMA_SWORD_OF_DESPAIR) == 1) then begin
                                     GVar20 := GVar20 + -10;
                                 end
                                 else begin
-                                    if (global_var(44) == 1) then begin
+                                    if (global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE) == 1) then begin
                                         GVar20 := GVar20 + -15;
                                     end
                                     else begin
@@ -503,52 +503,52 @@ begin
         debug_msg("Karma Reaction Bonus == " + GVar20);
         GVar20 := GVar20 + GVar24;
         debug_msg("Town Rep Bonus == " + GVar20);
-        set_global_var(37, 0);
-        set_global_var(38, 0);
-        set_global_var(39, 0);
-        set_global_var(40, 0);
-        set_global_var(41, 0);
-        set_global_var(42, 0);
-        set_global_var(43, 0);
-        set_global_var(44, 0);
-        set_global_var(45, 0);
-        GVar10 := global_var(0);
-        if (has_trait(0, dude_obj, 95)) then begin
-            GVar10 := global_var(0) * 2;
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+        set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+        set_global_var(GVAR_KARMA_DEFENDER, 0);
+        set_global_var(GVAR_KARMA_WANDERER, 0);
+        set_global_var(GVAR_KARMA_BETRAYER, 0);
+        set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+        set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+        GVar10 := global_var(GVAR_PLAYER_REPUTATION);
+        if (has_trait(TRAIT_PERK, dude_obj, PERK_karma_beacon_perk)) then begin
+            GVar10 := global_var(GVAR_PLAYER_REPUTATION) * 2;
         end
         if (GVar10 >= 1000) then begin
-            set_global_var(37, 1);
+            set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
         end
         else begin
             if (GVar10 >= 750) then begin
-                set_global_var(38, 1);
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
             end
             else begin
                 if (GVar10 >= 500) then begin
-                    set_global_var(39, 1);
+                    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                 end
                 else begin
                     if (GVar10 >= 250) then begin
-                        set_global_var(40, 1);
+                        set_global_var(GVAR_KARMA_DEFENDER, 1);
                     end
                     else begin
                         if (GVar10 > -250) then begin
-                            set_global_var(41, 1);
+                            set_global_var(GVAR_KARMA_WANDERER, 1);
                         end
                         else begin
                             if (GVar10 > -500) then begin
-                                set_global_var(42, 1);
+                                set_global_var(GVAR_KARMA_BETRAYER, 1);
                             end
                             else begin
                                 if (GVar10 > -750) then begin
-                                    set_global_var(43, 1);
+                                    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                 end
                                 else begin
                                     if (GVar10 > -1000) then begin
-                                        set_global_var(44, 1);
+                                        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                     end
                                     else begin
-                                        set_global_var(45, 1);
+                                        set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                     end
                                 end
                             end
@@ -557,35 +557,35 @@ begin
                 end
             end
         end
-        if (global_var(37) == 1) then begin
+        if (global_var(GVAR_KARMA_HOLY_WARRIOR) == 1) then begin
             GVar20 := GVar20 + 20;
         end
         else begin
-            if (global_var(38) == 1) then begin
+            if (global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES) == 1) then begin
                 GVar20 := GVar20 + 15;
             end
             else begin
-                if (global_var(39) == 1) then begin
+                if (global_var(GVAR_KARMA_SHIELD_OF_HOPE) == 1) then begin
                     GVar20 := GVar20 + 10;
                 end
                 else begin
-                    if (global_var(40) == 1) then begin
+                    if (global_var(GVAR_KARMA_DEFENDER) == 1) then begin
                         GVar20 := GVar20 + 5;
                     end
                     else begin
-                        if (global_var(41) == 1) then begin
+                        if (global_var(GVAR_KARMA_WANDERER) == 1) then begin
                             GVar20 := GVar20 + 0;
                         end
                         else begin
-                            if (global_var(42) == 1) then begin
+                            if (global_var(GVAR_KARMA_BETRAYER) == 1) then begin
                                 GVar20 := GVar20 + -5;
                             end
                             else begin
-                                if (global_var(43) == 1) then begin
+                                if (global_var(GVAR_KARMA_SWORD_OF_DESPAIR) == 1) then begin
                                     GVar20 := GVar20 + -10;
                                 end
                                 else begin
-                                    if (global_var(44) == 1) then begin
+                                    if (global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE) == 1) then begin
                                         GVar20 := GVar20 + -15;
                                     end
                                     else begin
@@ -601,32 +601,32 @@ begin
         debug_msg("Good Person Reaction Bonus == " + GVar20);
     end
     if (GVar22 == 0) then begin
-        if (global_var(11) == 1) then begin
+        if (global_var(GVAR_REPUTATION_SLAVER) == 1) then begin
             GVar20 := GVar20 + (2 * -25);
         end
     end
     else begin
         if (GVar22 == 1) then begin
-            if (global_var(11) == 1) then begin
+            if (global_var(GVAR_REPUTATION_SLAVER) == 1) then begin
                 GVar20 := GVar20 + -25;
             end
         end
         else begin
             if (GVar22 == 3) then begin
-                if (global_var(11) == 1) then begin
+                if (global_var(GVAR_REPUTATION_SLAVER) == 1) then begin
                     GVar20 := GVar20 - -25;
                 end
             end
         end
     end
     debug_msg("Slaver + Aligned Reaction == " + GVar20);
-    GVar20 := GVar20 + (10 * has_trait(0, dude_obj, 10));
+    GVar20 := GVar20 + (10 * has_trait(TRAIT_PERK, dude_obj, PERK_presence));
     debug_msg("Presence Reaction == " + GVar20);
-    if (global_var(1) >= 2) then begin
+    if (global_var(GVAR_CHILDKILLER_REPUTATION) >= 2) then begin
         GVar20 := GVar20 + -30;
     end
     debug_msg("Childkiller Reaction == " + GVar20);
-    if (has_trait(0, dude_obj, 39)) then begin
+    if (has_trait(TRAIT_PERK, dude_obj, PERK_cult_of_personality)) then begin
         if (GVar21 == 1) then begin
             if (GVar20 > 0) then begin
                 GVar20 := GVar20 * -1;
@@ -685,15 +685,15 @@ begin
         call Node998();
     end
     else begin
-        if (global_var(443) == 1) then begin
-            float_msg(self_obj, message_str(599, 174), 8);
+        if (global_var(GVAR_JAIL_BREAK) == 1) then begin
+            float_msg(self_obj, mstr(174), FLOAT_MSG_YELLOW);
         end
         else begin
-            if ((global_var(304) == 7) or (global_var(303) == 7)) then begin
+            if ((global_var(GVAR_BH_CONSPIRACY) == 7) or (global_var(GVAR_BH_JAIL) == 7)) then begin
                 call Node020();
             end
             else begin
-                if (global_var(304) == 3) then begin
+                if (global_var(GVAR_BH_CONSPIRACY) == 3) then begin
                     start_gdialog(599, self_obj, 4, -1, -1);
                     gsay_start;
                     call Node019();
@@ -701,7 +701,7 @@ begin
                     end_dialogue;
                 end
                 else begin
-                    if (global_var(304) == 5) then begin
+                    if (global_var(GVAR_BH_CONSPIRACY) == 5) then begin
                         start_gdialog(599, self_obj, 4, -1, -1);
                         gsay_start;
                         call Node021();
@@ -709,15 +709,15 @@ begin
                         end_dialogue;
                     end
                     else begin
-                        if (global_var(304) == 8) then begin
+                        if (global_var(GVAR_BH_CONSPIRACY) == 8) then begin
                             call Node023();
                         end
                         else begin
-                            if (global_var(304) == 4) then begin
+                            if (global_var(GVAR_BH_CONSPIRACY) == 4) then begin
                                 call Node024();
                             end
                             else begin
-                                if ((global_var(303) == 3) or ((global_var(303) == 6) and (global_var(443) == 0)) or ((global_var(303) == 5) and (global_var(443) == 0))) then begin
+                                if ((global_var(GVAR_BH_JAIL) == 3) or ((global_var(GVAR_BH_JAIL) == 6) and (global_var(GVAR_JAIL_BREAK) == 0)) or ((global_var(GVAR_BH_JAIL) == 5) and (global_var(GVAR_JAIL_BREAK) == 0))) then begin
                                     start_gdialog(599, self_obj, 4, -1, -1);
                                     gsay_start;
                                     call Node013();
@@ -725,7 +725,7 @@ begin
                                     end_dialogue;
                                 end
                                 else begin
-                                    if ((global_var(303) == 5) or (global_var(303) == 4) or (global_var(303) == 6) or (global_var(303) == 8)) then begin
+                                    if ((global_var(GVAR_BH_JAIL) == 5) or (global_var(GVAR_BH_JAIL) == 4) or (global_var(GVAR_BH_JAIL) == 6) or (global_var(GVAR_BH_JAIL) == 8)) then begin
                                         start_gdialog(599, self_obj, 4, -1, -1);
                                         gsay_start;
                                         call Node014();
@@ -747,7 +747,7 @@ begin
                                                 call Node001();
                                             end
                                             else begin
-                                                if (get_critter_stat(dude_obj, 4) <= 3) then begin
+                                                if (dude_is_stupid) then begin
                                                     call Node002();
                                                 end
                                                 else begin
@@ -773,59 +773,59 @@ procedure destroy_p_proc
 begin
     set_map_var(26, 1);
     if (source_obj == dude_obj) then begin
-        set_global_var(4, global_var(4) + 1);
-        if (metarule(51, self_obj) == 2) then begin
+        set_global_var(GVAR_BAD_MONSTER, global_var(GVAR_BAD_MONSTER) + 1);
+        if (critter_kill_type(self_obj) == KILL_TYPE_children_kills) then begin
             if (source_obj == dude_obj) then begin
-                set_global_var(2, 0);
-                set_global_var(1, global_var(1) + 1);
-                set_global_var(0, global_var(0) + -15);
+                set_global_var(GVAR_CHAMPION_REPUTATION, 0);
+                set_global_var(GVAR_CHILDKILLER_REPUTATION, global_var(GVAR_CHILDKILLER_REPUTATION) + 1);
+                set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + -15);
                 debug_msg("Player gains " + -15 + " Karma Points.");
-                set_global_var(37, 0);
-                set_global_var(38, 0);
-                set_global_var(39, 0);
-                set_global_var(40, 0);
-                set_global_var(41, 0);
-                set_global_var(42, 0);
-                set_global_var(43, 0);
-                set_global_var(44, 0);
-                set_global_var(45, 0);
-                GVar10 := global_var(0);
-                if (has_trait(0, dude_obj, 95)) then begin
-                    GVar10 := global_var(0) * 2;
+                set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+                set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+                set_global_var(GVAR_KARMA_DEFENDER, 0);
+                set_global_var(GVAR_KARMA_WANDERER, 0);
+                set_global_var(GVAR_KARMA_BETRAYER, 0);
+                set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+                set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+                set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+                GVar10 := global_var(GVAR_PLAYER_REPUTATION);
+                if (has_trait(TRAIT_PERK, dude_obj, PERK_karma_beacon_perk)) then begin
+                    GVar10 := global_var(GVAR_PLAYER_REPUTATION) * 2;
                 end
                 if (GVar10 >= 1000) then begin
-                    set_global_var(37, 1);
+                    set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
                 end
                 else begin
                     if (GVar10 >= 750) then begin
-                        set_global_var(38, 1);
+                        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
                     end
                     else begin
                         if (GVar10 >= 500) then begin
-                            set_global_var(39, 1);
+                            set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                         end
                         else begin
                             if (GVar10 >= 250) then begin
-                                set_global_var(40, 1);
+                                set_global_var(GVAR_KARMA_DEFENDER, 1);
                             end
                             else begin
                                 if (GVar10 > -250) then begin
-                                    set_global_var(41, 1);
+                                    set_global_var(GVAR_KARMA_WANDERER, 1);
                                 end
                                 else begin
                                     if (GVar10 > -500) then begin
-                                        set_global_var(42, 1);
+                                        set_global_var(GVAR_KARMA_BETRAYER, 1);
                                     end
                                     else begin
                                         if (GVar10 > -750) then begin
-                                            set_global_var(43, 1);
+                                            set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                         end
                                         else begin
                                             if (GVar10 > -1000) then begin
-                                                set_global_var(44, 1);
+                                                set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                             end
                                             else begin
-                                                set_global_var(45, 1);
+                                                set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                             end
                                         end
                                     end
@@ -834,66 +834,66 @@ begin
                         end
                     end
                 end
-                set_global_var(54, global_var(54) + -8);
+                set_global_var(GVAR_TOWN_REP_BROKEN_HILLS, global_var(GVAR_TOWN_REP_BROKEN_HILLS) + -8);
                 debug_msg("Added " + -8 + " to Town Rep");
             end
         end
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(5) > (2 * global_var(4))) or (global_var(3) == 1))) then begin
-            set_global_var(3, 1);
-            set_global_var(2, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
+            set_global_var(GVAR_BERSERKER_REPUTATION, 1);
+            set_global_var(GVAR_CHAMPION_REPUTATION, 0);
         end
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(4) > (3 * global_var(5))) or (global_var(2) == 1)) and (global_var(1) == 0)) then begin
-            set_global_var(2, 1);
-            set_global_var(3, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(4) > (3 * global_var(GVAR_GOOD_MONSTER))) or (global_var(GVAR_CHAMPION_REPUTATION) == 1)) and (global_var(GVAR_CHILDKILLER_REPUTATION) == 0)) then begin
+            set_global_var(GVAR_CHAMPION_REPUTATION, 1);
+            set_global_var(GVAR_BERSERKER_REPUTATION, 0);
         end
-        set_global_var(0, global_var(0) + 5);
+        set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + 5);
         debug_msg("Player gains " + 5 + " Karma Points.");
-        set_global_var(37, 0);
-        set_global_var(38, 0);
-        set_global_var(39, 0);
-        set_global_var(40, 0);
-        set_global_var(41, 0);
-        set_global_var(42, 0);
-        set_global_var(43, 0);
-        set_global_var(44, 0);
-        set_global_var(45, 0);
-        GVar10 := global_var(0);
-        if (has_trait(0, dude_obj, 95)) then begin
-            GVar10 := global_var(0) * 2;
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+        set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+        set_global_var(GVAR_KARMA_DEFENDER, 0);
+        set_global_var(GVAR_KARMA_WANDERER, 0);
+        set_global_var(GVAR_KARMA_BETRAYER, 0);
+        set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+        set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+        GVar10 := global_var(GVAR_PLAYER_REPUTATION);
+        if (has_trait(TRAIT_PERK, dude_obj, PERK_karma_beacon_perk)) then begin
+            GVar10 := global_var(GVAR_PLAYER_REPUTATION) * 2;
         end
         if (GVar10 >= 1000) then begin
-            set_global_var(37, 1);
+            set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
         end
         else begin
             if (GVar10 >= 750) then begin
-                set_global_var(38, 1);
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
             end
             else begin
                 if (GVar10 >= 500) then begin
-                    set_global_var(39, 1);
+                    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                 end
                 else begin
                     if (GVar10 >= 250) then begin
-                        set_global_var(40, 1);
+                        set_global_var(GVAR_KARMA_DEFENDER, 1);
                     end
                     else begin
                         if (GVar10 > -250) then begin
-                            set_global_var(41, 1);
+                            set_global_var(GVAR_KARMA_WANDERER, 1);
                         end
                         else begin
                             if (GVar10 > -500) then begin
-                                set_global_var(42, 1);
+                                set_global_var(GVAR_KARMA_BETRAYER, 1);
                             end
                             else begin
                                 if (GVar10 > -750) then begin
-                                    set_global_var(43, 1);
+                                    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                 end
                                 else begin
                                     if (GVar10 > -1000) then begin
-                                        set_global_var(44, 1);
+                                        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                     end
                                     else begin
-                                        set_global_var(45, 1);
+                                        set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                     end
                                 end
                             end
@@ -902,7 +902,7 @@ begin
                 end
             end
         end
-        set_global_var(54, global_var(54) + 2);
+        set_global_var(GVAR_TOWN_REP_BROKEN_HILLS, global_var(GVAR_TOWN_REP_BROKEN_HILLS) + 2);
         debug_msg("Added " + 2 + " to Town Rep");
     end
 end
@@ -911,17 +911,17 @@ procedure look_at_p_proc
 begin
     script_overrides;
     if (local_var(4) == 0) then begin
-        display_msg(message_str(599, 100));
+        display_mstr(100);
     end
     else begin
-        display_msg(message_str(599, 101));
+        display_mstr(101);
     end
 end
 
 procedure description_p_proc
 begin
     script_overrides;
-    display_msg(message_str(599, 102));
+    display_mstr(102);
 end
 
 procedure use_skill_on_p_proc
@@ -932,8 +932,8 @@ procedure damage_p_proc
 begin
     if (source_obj == dude_obj) then begin
         set_local_var(6, 1);
-        set_global_var(525, 1);
-        set_global_var(309, 1);
+        set_global_var(GVAR_ENEMY_CONSPIRATORS, 1);
+        set_global_var(GVAR_ENEMY_BROKEN_HILLS, 1);
     end
 end
 
@@ -941,9 +941,9 @@ procedure map_enter_p_proc
 begin
     Manson_Ptr := self_obj;
     GVar26 := 0;
-    critter_add_trait(self_obj, 1, 6, 150);
-    critter_add_trait(self_obj, 1, 5, 24);
-    if ((global_var(304) == 5) or ((global_var(303) == 5) or ((global_var(303) == 6) and (global_var(443) > 0)) or (global_var(303) == 4) or (global_var(303) == 8) and (global_var(304) != 7) and (global_var(304) != 8))) then begin
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 150);
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_TOUGH_CITIZEN);
+    if ((global_var(GVAR_BH_CONSPIRACY) == 5) or ((global_var(GVAR_BH_JAIL) == 5) or ((global_var(GVAR_BH_JAIL) == 6) and (global_var(GVAR_JAIL_BREAK) > 0)) or (global_var(GVAR_BH_JAIL) == 4) or (global_var(GVAR_BH_JAIL) == 8) and (global_var(GVAR_BH_CONSPIRACY) != 7) and (global_var(GVAR_BH_CONSPIRACY) != 8))) then begin
         move_to(self_obj, 28879, 0);
         critter_attempt_placement(self_obj, 28879, 0);
     end
@@ -975,7 +975,7 @@ end
 
 procedure Node995
 begin
-    if (get_critter_stat(dude_obj, 3) >= 7) then begin
+    if (dude_charisma >= 7) then begin
         set_local_var(3, 100);
         set_local_var(1, 4);
     end
@@ -993,7 +993,7 @@ end
 
 procedure Node997
 begin
-    if (get_critter_stat(dude_obj, 3) >= 8) then begin
+    if (dude_charisma >= 8) then begin
         call Node004();
     end
     else begin
@@ -1012,49 +1012,49 @@ end
 
 procedure Node001
 begin
-    gsay_reply(599, 103);
-    if (get_critter_stat(dude_obj, 4) == 1) then begin
-        giq_option(1, 599, 104, Node002, 50);
+    Reply(103);
+    if (dude_iq == 1) then begin
+        giq_option(1, NAME, 104, Node002, NEUTRAL_REACTION);
     end
-    if ((get_critter_stat(dude_obj, 4) == 2) or (get_critter_stat(dude_obj, 4) == 3)) then begin
-        giq_option(2, 599, 105, Node002, 50);
+    if ((dude_iq == 2) or (dude_iq == 3)) then begin
+        giq_option(2, NAME, 105, Node002, NEUTRAL_REACTION);
     end
-    giq_option(4, 599, 106, Node003, 50);
-    giq_option(4, 599, 107, Node004, 50);
-    giq_option(4, 599, 108, Node005, 50);
-    giq_option(4, 599, 109, Node999, 50);
+    giq_option(4, NAME, 106, Node003, NEUTRAL_REACTION);
+    giq_option(4, NAME, 107, Node004, NEUTRAL_REACTION);
+    giq_option(4, NAME, 108, Node005, NEUTRAL_REACTION);
+    giq_option(4, NAME, 109, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node002
 begin
-    gsay_reply(599, 110);
-    giq_option(-3, 599, 111, Node999, 50);
+    Reply(110);
+    giq_option(-3, NAME, 111, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node003
 begin
-    gsay_reply(599, 112);
-    giq_option(4, 599, 113, Node006, 50);
-    giq_option(4, 599, 114, Node007, 49);
-    giq_option(4, 599, 115, Node008, 51);
-    giq_option(4, 599, 116, Node009, 50);
-    giq_option(4, 599, 117, Node999, 50);
+    Reply(112);
+    giq_option(4, NAME, 113, Node006, NEUTRAL_REACTION);
+    giq_option(4, NAME, 114, Node007, GOOD_REACTION);
+    giq_option(4, NAME, 115, Node008, BAD_REACTION);
+    giq_option(4, NAME, 116, Node009, NEUTRAL_REACTION);
+    giq_option(4, NAME, 117, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node004
 begin
-    gsay_reply(599, 118);
-    giq_option(4, 599, 119, Node006, 50);
-    giq_option(4, 599, 120, Node007, 50);
+    Reply(118);
+    giq_option(4, NAME, 119, Node006, NEUTRAL_REACTION);
+    giq_option(4, NAME, 120, Node007, NEUTRAL_REACTION);
 end
 
 procedure Node005
 begin
-    gsay_reply(599, 121);
-    giq_option(4, 599, 122, Node007, 50);
-    giq_option(4, 599, 123, Node005a, 50);
-    giq_option(7, 599, 124, Node005b, 50);
-    giq_option(4, 599, 125, Node999, 50);
+    Reply(121);
+    giq_option(4, NAME, 122, Node007, NEUTRAL_REACTION);
+    giq_option(4, NAME, 123, Node005a, NEUTRAL_REACTION);
+    giq_option(7, NAME, 124, Node005b, NEUTRAL_REACTION);
+    giq_option(4, NAME, 125, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node005a
@@ -1112,20 +1112,20 @@ procedure Node005b
 begin
     if (local_var(8) == 0) then begin
         set_local_var(8, 1);
-        display_msg(message_str(599, 175));
+        display_mstr(175);
         give_exp_points(300);
-        display_msg(message_str(14, 100) + 300 + message_str(14, 101));
-        set_global_var(54, global_var(54) + 2);
+        display_msg(g_mstr(100) + 300 + g_mstr(101));
+        set_global_var(GVAR_TOWN_REP_BROKEN_HILLS, global_var(GVAR_TOWN_REP_BROKEN_HILLS) + 2);
     end
     call Node010();
 end
 
 procedure Node006
 begin
-    gsay_reply(599, 126);
-    giq_option(4, 599, 127, Node011, 50);
-    giq_option(4, 599, 128, Node006a, 50);
-    giq_option(4, 599, 129, Node999, 50);
+    Reply(126);
+    giq_option(4, NAME, 127, Node011, NEUTRAL_REACTION);
+    giq_option(4, NAME, 128, Node006a, NEUTRAL_REACTION);
+    giq_option(4, NAME, 129, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node006a
@@ -1181,10 +1181,10 @@ end
 
 procedure Node007
 begin
-    gsay_reply(599, 130);
-    giq_option(4, 599, 131, Node007a, 51);
-    giq_option(4, 599, 132, Node011, 49);
-    giq_option(4, 599, 133, Node999, 50);
+    Reply(130);
+    giq_option(4, NAME, 131, Node007a, BAD_REACTION);
+    giq_option(4, NAME, 132, Node011, GOOD_REACTION);
+    giq_option(4, NAME, 133, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node007a
@@ -1240,16 +1240,16 @@ end
 
 procedure Node008
 begin
-    gsay_reply(599, 134);
-    giq_option(4, 599, 133, Node999, 50);
+    Reply(134);
+    giq_option(4, NAME, 133, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node009
 begin
-    gsay_reply(599, 135);
-    giq_option(4, 599, 136, Node008, 50);
-    giq_option(4, 599, 137, Node009a, 50);
-    giq_option(4, 599, 138, Node999, 50);
+    Reply(135);
+    giq_option(4, NAME, 136, Node008, NEUTRAL_REACTION);
+    giq_option(4, NAME, 137, Node009a, NEUTRAL_REACTION);
+    giq_option(4, NAME, 138, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node009a
@@ -1305,17 +1305,17 @@ end
 
 procedure Node010
 begin
-    gsay_reply(599, 139);
-    giq_option(4, 599, 140, Node012, 50);
+    Reply(139);
+    giq_option(4, NAME, 140, Node012, NEUTRAL_REACTION);
 end
 
 procedure Node011
 begin
-    gsay_reply(599, 141);
-    giq_option(4, 599, 142, Node011a, 51);
-    giq_option(4, 599, 143, Node011b, 49);
-    giq_option(4, 599, 144, Node012, 50);
-    giq_option(4, 599, 145, Node999, 50);
+    Reply(141);
+    giq_option(4, NAME, 142, Node011a, BAD_REACTION);
+    giq_option(4, NAME, 143, Node011b, GOOD_REACTION);
+    giq_option(4, NAME, 144, Node012, NEUTRAL_REACTION);
+    giq_option(4, NAME, 145, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node011a
@@ -1422,14 +1422,14 @@ end
 
 procedure Node012
 begin
-    gsay_reply(599, 146);
-    giq_option(4, 599, 147, Node003, 50);
-    giq_option(4, 599, 148, Node004, 50);
-    giq_option(4, 599, 149, Node005, 50);
-    giq_option(4, 599, 150, Node011, 50);
-    giq_option(4, 599, 151, Node009, 50);
-    giq_option(4, 599, 152, Node012a, 50);
-    giq_option(4, 599, 153, Node999, 50);
+    Reply(146);
+    giq_option(4, NAME, 147, Node003, NEUTRAL_REACTION);
+    giq_option(4, NAME, 148, Node004, NEUTRAL_REACTION);
+    giq_option(4, NAME, 149, Node005, NEUTRAL_REACTION);
+    giq_option(4, NAME, 150, Node011, NEUTRAL_REACTION);
+    giq_option(4, NAME, 151, Node009, NEUTRAL_REACTION);
+    giq_option(4, NAME, 152, Node012a, NEUTRAL_REACTION);
+    giq_option(4, NAME, 153, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node012a
@@ -1485,91 +1485,90 @@ end
 
 procedure Node013
 begin
-    gsay_reply(599, 154);
-    giq_option(4, 599, 155, Node999, 50);
+    Reply(154);
+    giq_option(4, NAME, 155, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node014
 begin
-    gsay_reply(599, 156);
-    giq_option(4, 599, 157, Node015, 50);
+    Reply(156);
+    giq_option(4, NAME, 157, Node015, NEUTRAL_REACTION);
 end
 
 procedure Node015
 begin
-    gsay_reply(599, 158);
-    giq_option(4, 599, 159, Node016, 50);
-    giq_option(4, 599, 160, Node017, 50);
-    giq_option(4, 599, 161, Node999, 50);
+    Reply(158);
+    giq_option(4, NAME, 159, Node016, NEUTRAL_REACTION);
+    giq_option(4, NAME, 160, Node017, NEUTRAL_REACTION);
+    giq_option(4, NAME, 161, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node016
 begin
-    gsay_reply(599, 162);
-    giq_option(4, 599, 163, Node016a, 50);
-    giq_option(4, 599, 164, Node016b, 50);
-    giq_option(4, 599, 165, Node016c, 50);
+    Reply(162);
+    giq_option(4, NAME, 163, Node016a, NEUTRAL_REACTION);
+    giq_option(4, NAME, 164, Node016b, NEUTRAL_REACTION);
+    giq_option(4, NAME, 165, Node016c, NEUTRAL_REACTION);
 end
 
 procedure Node016a
 begin
-    set_global_var(304, 2);
+    set_global_var(GVAR_BH_CONSPIRACY, 2);
     call Node018();
 end
 
 procedure Node016b
 begin
-    set_global_var(304, 3);
+    set_global_var(GVAR_BH_CONSPIRACY, 3);
     call Node019();
 end
 
 procedure Node016c
 begin
-    set_global_var(304, 2);
+    set_global_var(GVAR_BH_CONSPIRACY, 2);
 end
 
 procedure Node017
 begin
-    gsay_reply(599, 166);
-    giq_option(4, 599, 167, Node015, 50);
+    Reply(166);
+    giq_option(4, NAME, 167, Node015, NEUTRAL_REACTION);
 end
 
 procedure Node018
 begin
-    gsay_reply(599, 168);
-    giq_option(4, 599, 169, Node999, 50);
+    Reply(168);
+    giq_option(4, NAME, 169, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node019
 begin
-    gsay_reply(599, 170);
-    giq_option(4, 599, 133, Node999, 50);
+    Reply(170);
+    giq_option(4, NAME, 133, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node020
 begin
-    float_msg(self_obj, message_str(599, 171), 8);
+    float_msg(self_obj, mstr(171), FLOAT_MSG_YELLOW);
 end
 
 procedure Node021
 begin
-    gsay_reply(599, 172);
-    giq_option(4, 599, 133, Node999, 50);
+    Reply(172);
+    giq_option(4, NAME, 133, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node022
 begin
-    gsay_reply(599, 173);
-    giq_option(4, 599, 133, Node999, 50);
+    Reply(173);
+    giq_option(4, NAME, 133, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node023
 begin
-    float_msg(self_obj, message_str(599, 176), 8);
+    float_msg(self_obj, mstr(176), FLOAT_MSG_YELLOW);
 end
 
 procedure Node024
 begin
-    float_msg(self_obj, message_str(599, 179), 8);
+    float_msg(self_obj, mstr(179), FLOAT_MSG_YELLOW);
 end
-
