diff --git a/Mods/MegaMod/Original/HCOUTFIT.ssl b/Mods/MegaMod/ReDefine/HCOUTFIT.ssl
index 8cb129a..3ffc9b3 100644
--- a/Mods/MegaMod/Original/HCOUTFIT.ssl
+++ b/Mods/MegaMod/ReDefine/HCOUTFIT.ssl
@@ -63,9 +63,9 @@ end
 
 procedure critter_p_proc
 begin
-    if ((global_var(309) == 1) or (local_var(5) == 2) and obj_can_see_obj(self_obj, dude_obj)) then begin
+    if ((global_var(GVAR_ENEMY_BROKEN_HILLS) == 1) or (local_var(5) == 2) and obj_can_see_obj(self_obj, dude_obj)) then begin
         set_local_var(5, 1);
-        attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+        attack(dude_obj);
     end
 end
 
@@ -80,7 +80,7 @@ procedure talk_p_proc
 begin
     if (local_var(2) == 0) then begin
         set_local_var(2, 1);
-        set_local_var(3, (get_critter_stat(dude_obj, 3) - get_critter_stat(self_obj, 3)) * 5);
+        set_local_var(3, (dude_charisma - self_charisma) * 5);
         debug_msg("Base Reaction == " + local_var(3));
     end
     Static_Reaction := 0;
@@ -92,14 +92,14 @@ begin
     else begin
         reaction_bonus_karma := -1 * (-1 * reaction_bonus_karma / 1000);
     end
-    if (global_var(54) >= 0) then begin
-        reaction_bonus_town_rep := global_var(54) / 2;
+    if (global_var(GVAR_TOWN_REP_BROKEN_HILLS) >= 0) then begin
+        reaction_bonus_town_rep := global_var(GVAR_TOWN_REP_BROKEN_HILLS) / 2;
     end
     else begin
-        reaction_bonus_town_rep := -1 * (-1 * global_var(54) / 2);
+        reaction_bonus_town_rep := -1 * (-1 * global_var(GVAR_TOWN_REP_BROKEN_HILLS) / 2);
     end
-    if (has_trait(2, dude_obj, 13)) then begin
-        if (get_critter_stat(dude_obj, 34) == get_critter_stat(self_obj, 34)) then begin
+    if (has_trait(TRAIT_TRAIT, dude_obj, TRAIT_sex_appeal)) then begin
+        if (dude_gender == self_gender) then begin
             Static_Reaction := Static_Reaction + -20;
         end
         else begin
@@ -125,52 +125,52 @@ begin
             Static_Reaction := Static_Reaction + reaction_bonus_town_rep;
         end
         debug_msg("Town Rep Bonus == " + Static_Reaction);
-        set_global_var(37, 0);
-        set_global_var(38, 0);
-        set_global_var(39, 0);
-        set_global_var(40, 0);
-        set_global_var(41, 0);
-        set_global_var(42, 0);
-        set_global_var(43, 0);
-        set_global_var(44, 0);
-        set_global_var(45, 0);
-        tmp_gen_rep := global_var(0);
-        if (has_trait(0, dude_obj, 95)) then begin
-            tmp_gen_rep := global_var(0) * 2;
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+        set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+        set_global_var(GVAR_KARMA_DEFENDER, 0);
+        set_global_var(GVAR_KARMA_WANDERER, 0);
+        set_global_var(GVAR_KARMA_BETRAYER, 0);
+        set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+        set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+        tmp_gen_rep := global_var(GVAR_PLAYER_REPUTATION);
+        if (has_trait(TRAIT_PERK, dude_obj, PERK_karma_beacon_perk)) then begin
+            tmp_gen_rep := global_var(GVAR_PLAYER_REPUTATION) * 2;
         end
         if (tmp_gen_rep >= 1000) then begin
-            set_global_var(37, 1);
+            set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
         end
         else begin
             if (tmp_gen_rep >= 750) then begin
-                set_global_var(38, 1);
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
             end
             else begin
                 if (tmp_gen_rep >= 500) then begin
-                    set_global_var(39, 1);
+                    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                 end
                 else begin
                     if (tmp_gen_rep >= 250) then begin
-                        set_global_var(40, 1);
+                        set_global_var(GVAR_KARMA_DEFENDER, 1);
                     end
                     else begin
                         if (tmp_gen_rep > -250) then begin
-                            set_global_var(41, 1);
+                            set_global_var(GVAR_KARMA_WANDERER, 1);
                         end
                         else begin
                             if (tmp_gen_rep > -500) then begin
-                                set_global_var(42, 1);
+                                set_global_var(GVAR_KARMA_BETRAYER, 1);
                             end
                             else begin
                                 if (tmp_gen_rep > -750) then begin
-                                    set_global_var(43, 1);
+                                    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                 end
                                 else begin
                                     if (tmp_gen_rep > -1000) then begin
-                                        set_global_var(44, 1);
+                                        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                     end
                                     else begin
-                                        set_global_var(45, 1);
+                                        set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                     end
                                 end
                             end
@@ -179,35 +179,35 @@ begin
                 end
             end
         end
-        if (global_var(37) == 1) then begin
+        if (global_var(GVAR_KARMA_HOLY_WARRIOR) == 1) then begin
             Static_Reaction := Static_Reaction - 20;
         end
         else begin
-            if (global_var(38) == 1) then begin
+            if (global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES) == 1) then begin
                 Static_Reaction := Static_Reaction - 15;
             end
             else begin
-                if (global_var(39) == 1) then begin
+                if (global_var(GVAR_KARMA_SHIELD_OF_HOPE) == 1) then begin
                     Static_Reaction := Static_Reaction - 10;
                 end
                 else begin
-                    if (global_var(40) == 1) then begin
+                    if (global_var(GVAR_KARMA_DEFENDER) == 1) then begin
                         Static_Reaction := Static_Reaction - 5;
                     end
                     else begin
-                        if (global_var(41) == 1) then begin
+                        if (global_var(GVAR_KARMA_WANDERER) == 1) then begin
                             Static_Reaction := Static_Reaction - 0;
                         end
                         else begin
-                            if (global_var(42) == 1) then begin
+                            if (global_var(GVAR_KARMA_BETRAYER) == 1) then begin
                                 Static_Reaction := Static_Reaction + -5;
                             end
                             else begin
-                                if (global_var(43) == 1) then begin
+                                if (global_var(GVAR_KARMA_SWORD_OF_DESPAIR) == 1) then begin
                                     Static_Reaction := Static_Reaction + -10;
                                 end
                                 else begin
-                                    if (global_var(44) == 1) then begin
+                                    if (global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE) == 1) then begin
                                         Static_Reaction := Static_Reaction + -15;
                                     end
                                     else begin
@@ -227,52 +227,52 @@ begin
         debug_msg("Karma Reaction Bonus == " + Static_Reaction);
         Static_Reaction := Static_Reaction + reaction_bonus_town_rep;
         debug_msg("Town Rep Bonus == " + Static_Reaction);
-        set_global_var(37, 0);
-        set_global_var(38, 0);
-        set_global_var(39, 0);
-        set_global_var(40, 0);
-        set_global_var(41, 0);
-        set_global_var(42, 0);
-        set_global_var(43, 0);
-        set_global_var(44, 0);
-        set_global_var(45, 0);
-        tmp_gen_rep := global_var(0);
-        if (has_trait(0, dude_obj, 95)) then begin
-            tmp_gen_rep := global_var(0) * 2;
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+        set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+        set_global_var(GVAR_KARMA_DEFENDER, 0);
+        set_global_var(GVAR_KARMA_WANDERER, 0);
+        set_global_var(GVAR_KARMA_BETRAYER, 0);
+        set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+        set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+        tmp_gen_rep := global_var(GVAR_PLAYER_REPUTATION);
+        if (has_trait(TRAIT_PERK, dude_obj, PERK_karma_beacon_perk)) then begin
+            tmp_gen_rep := global_var(GVAR_PLAYER_REPUTATION) * 2;
         end
         if (tmp_gen_rep >= 1000) then begin
-            set_global_var(37, 1);
+            set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
         end
         else begin
             if (tmp_gen_rep >= 750) then begin
-                set_global_var(38, 1);
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
             end
             else begin
                 if (tmp_gen_rep >= 500) then begin
-                    set_global_var(39, 1);
+                    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                 end
                 else begin
                     if (tmp_gen_rep >= 250) then begin
-                        set_global_var(40, 1);
+                        set_global_var(GVAR_KARMA_DEFENDER, 1);
                     end
                     else begin
                         if (tmp_gen_rep > -250) then begin
-                            set_global_var(41, 1);
+                            set_global_var(GVAR_KARMA_WANDERER, 1);
                         end
                         else begin
                             if (tmp_gen_rep > -500) then begin
-                                set_global_var(42, 1);
+                                set_global_var(GVAR_KARMA_BETRAYER, 1);
                             end
                             else begin
                                 if (tmp_gen_rep > -750) then begin
-                                    set_global_var(43, 1);
+                                    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                 end
                                 else begin
                                     if (tmp_gen_rep > -1000) then begin
-                                        set_global_var(44, 1);
+                                        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                     end
                                     else begin
-                                        set_global_var(45, 1);
+                                        set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                     end
                                 end
                             end
@@ -281,35 +281,35 @@ begin
                 end
             end
         end
-        if (global_var(37) == 1) then begin
+        if (global_var(GVAR_KARMA_HOLY_WARRIOR) == 1) then begin
             Static_Reaction := Static_Reaction + 20;
         end
         else begin
-            if (global_var(38) == 1) then begin
+            if (global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES) == 1) then begin
                 Static_Reaction := Static_Reaction + 15;
             end
             else begin
-                if (global_var(39) == 1) then begin
+                if (global_var(GVAR_KARMA_SHIELD_OF_HOPE) == 1) then begin
                     Static_Reaction := Static_Reaction + 10;
                 end
                 else begin
-                    if (global_var(40) == 1) then begin
+                    if (global_var(GVAR_KARMA_DEFENDER) == 1) then begin
                         Static_Reaction := Static_Reaction + 5;
                     end
                     else begin
-                        if (global_var(41) == 1) then begin
+                        if (global_var(GVAR_KARMA_WANDERER) == 1) then begin
                             Static_Reaction := Static_Reaction + 0;
                         end
                         else begin
-                            if (global_var(42) == 1) then begin
+                            if (global_var(GVAR_KARMA_BETRAYER) == 1) then begin
                                 Static_Reaction := Static_Reaction + -5;
                             end
                             else begin
-                                if (global_var(43) == 1) then begin
+                                if (global_var(GVAR_KARMA_SWORD_OF_DESPAIR) == 1) then begin
                                     Static_Reaction := Static_Reaction + -10;
                                 end
                                 else begin
-                                    if (global_var(44) == 1) then begin
+                                    if (global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE) == 1) then begin
                                         Static_Reaction := Static_Reaction + -15;
                                     end
                                     else begin
@@ -325,32 +325,32 @@ begin
         debug_msg("Good Person Reaction Bonus == " + Static_Reaction);
     end
     if (Slavery_Tolerant == 0) then begin
-        if (global_var(11) == 1) then begin
+        if (global_var(GVAR_REPUTATION_SLAVER) == 1) then begin
             Static_Reaction := Static_Reaction + (2 * -25);
         end
     end
     else begin
         if (Slavery_Tolerant == 1) then begin
-            if (global_var(11) == 1) then begin
+            if (global_var(GVAR_REPUTATION_SLAVER) == 1) then begin
                 Static_Reaction := Static_Reaction + -25;
             end
         end
         else begin
             if (Slavery_Tolerant == 3) then begin
-                if (global_var(11) == 1) then begin
+                if (global_var(GVAR_REPUTATION_SLAVER) == 1) then begin
                     Static_Reaction := Static_Reaction - -25;
                 end
             end
         end
     end
     debug_msg("Slaver + Aligned Reaction == " + Static_Reaction);
-    Static_Reaction := Static_Reaction + (10 * has_trait(0, dude_obj, 10));
+    Static_Reaction := Static_Reaction + (10 * has_trait(TRAIT_PERK, dude_obj, PERK_presence));
     debug_msg("Presence Reaction == " + Static_Reaction);
-    if (global_var(1) >= 2) then begin
+    if (global_var(GVAR_CHILDKILLER_REPUTATION) >= 2) then begin
         Static_Reaction := Static_Reaction + -30;
     end
     debug_msg("Childkiller Reaction == " + Static_Reaction);
-    if (has_trait(0, dude_obj, 39)) then begin
+    if (has_trait(TRAIT_PERK, dude_obj, PERK_cult_of_personality)) then begin
         if (Evil_Critter == 1) then begin
             if (Static_Reaction > 0) then begin
                 Static_Reaction := Static_Reaction * -1;
@@ -370,7 +370,7 @@ begin
         call Node998();
     end
     else begin
-        if (global_var(330) == 5) then begin
+        if (global_var(GVAR_BH_CARAVAN) == 5) then begin
             start_gdialog(603, self_obj, 4, -1, -1);
             gsay_start;
             call Node033();
@@ -378,7 +378,7 @@ begin
             end_dialogue;
         end
         else begin
-            if ((local_var(7) == 1) or (global_var(309) == 1)) then begin
+            if ((local_var(7) == 1) or (global_var(GVAR_ENEMY_BROKEN_HILLS) == 1)) then begin
                 start_gdialog(603, self_obj, 4, -1, -1);
                 gsay_start;
                 call Node022();
@@ -386,7 +386,7 @@ begin
                 end_dialogue;
             end
             else begin
-                if (global_var(304) == 5) then begin
+                if (global_var(GVAR_BH_CONSPIRACY) == 5) then begin
                     start_gdialog(603, self_obj, 4, -1, -1);
                     gsay_start;
                     call Node032();
@@ -394,7 +394,7 @@ begin
                     end_dialogue;
                 end
                 else begin
-                    if ((local_var(9) == 1) and (get_day == 21) and (global_var(563) == 0) and ((global_var(300) != 7) and (global_var(300) != 9))) then begin
+                    if ((local_var(9) == 1) and (get_day == 21) and (global_var(GVAR_CHAD_DEAD) == 0) and ((global_var(GVAR_BH_CHAD) != 7) and (global_var(GVAR_BH_CHAD) != 9))) then begin
                         start_gdialog(603, self_obj, 4, -1, -1);
                         gsay_start;
                         call Node023();
@@ -402,7 +402,7 @@ begin
                         end_dialogue;
                     end
                     else begin
-                        if ((local_var(9) == 2) and (get_day == 1) and (global_var(563) == 0) and ((global_var(300) != 7) and (global_var(300) != 9))) then begin
+                        if ((local_var(9) == 2) and (get_day == 1) and (global_var(GVAR_CHAD_DEAD) == 0) and ((global_var(GVAR_BH_CHAD) != 7) and (global_var(GVAR_BH_CHAD) != 9))) then begin
                             start_gdialog(603, self_obj, 4, -1, -1);
                             gsay_start;
                             call Node025();
@@ -410,7 +410,7 @@ begin
                             end_dialogue;
                         end
                         else begin
-                            if ((local_var(9) == 3) and (get_day == 11) and (global_var(301) == 5) and (global_var(563) == 0) and ((global_var(300) != 7) and (global_var(300) != 9))) then begin
+                            if ((local_var(9) == 3) and (get_day == 11) and (global_var(GVAR_BH_FTM) == 5) and (global_var(GVAR_CHAD_DEAD) == 0) and ((global_var(GVAR_BH_CHAD) != 7) and (global_var(GVAR_BH_CHAD) != 9))) then begin
                                 start_gdialog(603, self_obj, 4, -1, -1);
                                 gsay_start;
                                 call Node026();
@@ -418,7 +418,7 @@ begin
                                 end_dialogue;
                             end
                             else begin
-                                if ((local_var(9) == 3) and (get_day == 11) and (global_var(563) == 0) and ((global_var(300) != 7) and (global_var(300) != 9))) then begin
+                                if ((local_var(9) == 3) and (get_day == 11) and (global_var(GVAR_CHAD_DEAD) == 0) and ((global_var(GVAR_BH_CHAD) != 7) and (global_var(GVAR_BH_CHAD) != 9))) then begin
                                     start_gdialog(603, self_obj, 4, -1, -1);
                                     gsay_start;
                                     call Node024();
@@ -452,63 +452,63 @@ end
 procedure destroy_p_proc
 begin
     if (source_obj == dude_obj) then begin
-        set_global_var(5, global_var(5) + 1);
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(4) > (3 * global_var(5))) or (global_var(2) == 1)) and (global_var(1) == 0)) then begin
-            set_global_var(2, 1);
-            set_global_var(3, 0);
+        set_global_var(GVAR_GOOD_MONSTER, global_var(GVAR_GOOD_MONSTER) + 1);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(4) > (3 * global_var(GVAR_GOOD_MONSTER))) or (global_var(GVAR_CHAMPION_REPUTATION) == 1)) and (global_var(GVAR_CHILDKILLER_REPUTATION) == 0)) then begin
+            set_global_var(GVAR_CHAMPION_REPUTATION, 1);
+            set_global_var(GVAR_BERSERKER_REPUTATION, 0);
         end
-        if (metarule(51, self_obj) == 2) then begin
+        if (critter_kill_type(self_obj) == 2) then begin
             if (source_obj == dude_obj) then begin
-                set_global_var(2, 0);
-                set_global_var(1, global_var(1) + 1);
-                set_global_var(0, global_var(0) + -15);
+                set_global_var(GVAR_CHAMPION_REPUTATION, 0);
+                set_global_var(GVAR_CHILDKILLER_REPUTATION, global_var(GVAR_CHILDKILLER_REPUTATION) + 1);
+                set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + -15);
                 debug_msg("Player gains " + -15 + " Karma Points.");
-                set_global_var(37, 0);
-                set_global_var(38, 0);
-                set_global_var(39, 0);
-                set_global_var(40, 0);
-                set_global_var(41, 0);
-                set_global_var(42, 0);
-                set_global_var(43, 0);
-                set_global_var(44, 0);
-                set_global_var(45, 0);
-                tmp_gen_rep := global_var(0);
-                if (has_trait(0, dude_obj, 95)) then begin
-                    tmp_gen_rep := global_var(0) * 2;
+                set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+                set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+                set_global_var(GVAR_KARMA_DEFENDER, 0);
+                set_global_var(GVAR_KARMA_WANDERER, 0);
+                set_global_var(GVAR_KARMA_BETRAYER, 0);
+                set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+                set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+                set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+                tmp_gen_rep := global_var(GVAR_PLAYER_REPUTATION);
+                if (has_trait(TRAIT_PERK, dude_obj, PERK_karma_beacon_perk)) then begin
+                    tmp_gen_rep := global_var(GVAR_PLAYER_REPUTATION) * 2;
                 end
                 if (tmp_gen_rep >= 1000) then begin
-                    set_global_var(37, 1);
+                    set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
                 end
                 else begin
                     if (tmp_gen_rep >= 750) then begin
-                        set_global_var(38, 1);
+                        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
                     end
                     else begin
                         if (tmp_gen_rep >= 500) then begin
-                            set_global_var(39, 1);
+                            set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                         end
                         else begin
                             if (tmp_gen_rep >= 250) then begin
-                                set_global_var(40, 1);
+                                set_global_var(GVAR_KARMA_DEFENDER, 1);
                             end
                             else begin
                                 if (tmp_gen_rep > -250) then begin
-                                    set_global_var(41, 1);
+                                    set_global_var(GVAR_KARMA_WANDERER, 1);
                                 end
                                 else begin
                                     if (tmp_gen_rep > -500) then begin
-                                        set_global_var(42, 1);
+                                        set_global_var(GVAR_KARMA_BETRAYER, 1);
                                     end
                                     else begin
                                         if (tmp_gen_rep > -750) then begin
-                                            set_global_var(43, 1);
+                                            set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                         end
                                         else begin
                                             if (tmp_gen_rep > -1000) then begin
-                                                set_global_var(44, 1);
+                                                set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                             end
                                             else begin
-                                                set_global_var(45, 1);
+                                                set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                             end
                                         end
                                     end
@@ -517,62 +517,62 @@ begin
                         end
                     end
                 end
-                set_global_var(54, global_var(54) + -8);
+                set_global_var(GVAR_TOWN_REP_BROKEN_HILLS, global_var(GVAR_TOWN_REP_BROKEN_HILLS) + -8);
                 debug_msg("Added " + -8 + " to Town Rep");
             end
         end
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(5) > (2 * global_var(4))) or (global_var(3) == 1))) then begin
-            set_global_var(3, 1);
-            set_global_var(2, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
+            set_global_var(GVAR_BERSERKER_REPUTATION, 1);
+            set_global_var(GVAR_CHAMPION_REPUTATION, 0);
         end
-        set_global_var(0, global_var(0) + -10);
+        set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + -10);
         debug_msg("Player gains " + -10 + " Karma Points.");
-        set_global_var(37, 0);
-        set_global_var(38, 0);
-        set_global_var(39, 0);
-        set_global_var(40, 0);
-        set_global_var(41, 0);
-        set_global_var(42, 0);
-        set_global_var(43, 0);
-        set_global_var(44, 0);
-        set_global_var(45, 0);
-        tmp_gen_rep := global_var(0);
-        if (has_trait(0, dude_obj, 95)) then begin
-            tmp_gen_rep := global_var(0) * 2;
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+        set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+        set_global_var(GVAR_KARMA_DEFENDER, 0);
+        set_global_var(GVAR_KARMA_WANDERER, 0);
+        set_global_var(GVAR_KARMA_BETRAYER, 0);
+        set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+        set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+        tmp_gen_rep := global_var(GVAR_PLAYER_REPUTATION);
+        if (has_trait(TRAIT_PERK, dude_obj, PERK_karma_beacon_perk)) then begin
+            tmp_gen_rep := global_var(GVAR_PLAYER_REPUTATION) * 2;
         end
         if (tmp_gen_rep >= 1000) then begin
-            set_global_var(37, 1);
+            set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
         end
         else begin
             if (tmp_gen_rep >= 750) then begin
-                set_global_var(38, 1);
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
             end
             else begin
                 if (tmp_gen_rep >= 500) then begin
-                    set_global_var(39, 1);
+                    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                 end
                 else begin
                     if (tmp_gen_rep >= 250) then begin
-                        set_global_var(40, 1);
+                        set_global_var(GVAR_KARMA_DEFENDER, 1);
                     end
                     else begin
                         if (tmp_gen_rep > -250) then begin
-                            set_global_var(41, 1);
+                            set_global_var(GVAR_KARMA_WANDERER, 1);
                         end
                         else begin
                             if (tmp_gen_rep > -500) then begin
-                                set_global_var(42, 1);
+                                set_global_var(GVAR_KARMA_BETRAYER, 1);
                             end
                             else begin
                                 if (tmp_gen_rep > -750) then begin
-                                    set_global_var(43, 1);
+                                    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                 end
                                 else begin
                                     if (tmp_gen_rep > -1000) then begin
-                                        set_global_var(44, 1);
+                                        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                     end
                                     else begin
-                                        set_global_var(45, 1);
+                                        set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                     end
                                 end
                             end
@@ -581,7 +581,7 @@ begin
                 end
             end
         end
-        set_global_var(54, global_var(54) + -5);
+        set_global_var(GVAR_TOWN_REP_BROKEN_HILLS, global_var(GVAR_TOWN_REP_BROKEN_HILLS) + -5);
         debug_msg("Added " + -5 + " to Town Rep");
     end
 end
@@ -590,17 +590,17 @@ procedure look_at_p_proc
 begin
     script_overrides;
     if (local_var(4) == 0) then begin
-        display_msg(message_str(603, 100));
+        display_mstr(100);
     end
     else begin
-        display_msg(message_str(603, 101));
+        display_mstr(101);
     end
 end
 
 procedure description_p_proc
 begin
     script_overrides;
-    display_msg(message_str(603, 102));
+    display_mstr(102);
 end
 
 procedure use_skill_on_p_proc
@@ -611,15 +611,15 @@ procedure damage_p_proc
 begin
     if (source_obj == dude_obj) then begin
         set_local_var(6, 1);
-        set_global_var(309, 1);
+        set_global_var(GVAR_ENEMY_BROKEN_HILLS, 1);
     end
 end
 
 procedure map_enter_p_proc
 begin
     Only_Once := 0;
-    critter_add_trait(self_obj, 1, 6, 120);
-    critter_add_trait(self_obj, 1, 5, 24);
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 120);
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_TOUGH_CITIZEN);
 end
 
 procedure combat
@@ -637,128 +637,128 @@ end
 
 procedure Node001
 begin
-    gsay_reply(603, 103);
-    if (get_critter_stat(dude_obj, 4) == 1) then begin
-        giq_option(1, 603, 104, Node002, 50);
+    Reply(103);
+    if (dude_iq == 1) then begin
+        giq_option(1, NAME, 104, Node002, NEUTRAL_REACTION);
     end
-    if ((get_critter_stat(dude_obj, 4) == 2) or (get_critter_stat(dude_obj, 4) == 3)) then begin
-        giq_option(2, 603, 105, Node003, 50);
+    if ((dude_iq == 2) or (dude_iq == 3)) then begin
+        giq_option(2, NAME, 105, Node003, NEUTRAL_REACTION);
     end
-    giq_option(4, 603, 106, Node003, 50);
-    giq_option(4, 603, 107, Node004, 50);
-    giq_option(4, 603, 111, Node006, 50);
+    giq_option(4, NAME, 106, Node003, NEUTRAL_REACTION);
+    giq_option(4, NAME, 107, Node004, NEUTRAL_REACTION);
+    giq_option(4, NAME, 111, Node006, NEUTRAL_REACTION);
 end
 
 procedure Node002
 begin
-    gsay_reply(603, 112);
-    giq_option(-3, 603, 113, Node999, 50);
+    Reply(112);
+    giq_option(-3, NAME, 113, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node003
 begin
-    if (global_var(563) == 1) then begin
-        gsay_reply(603, 212);
+    if (global_var(GVAR_CHAD_DEAD) == 1) then begin
+        Reply(212);
     end
     else begin
-        if ((global_var(300) == 7) or (global_var(300) == 9)) then begin
-            gsay_reply(603, 214);
+        if ((global_var(GVAR_BH_CHAD) == 7) or (global_var(GVAR_BH_CHAD) == 9)) then begin
+            Reply(214);
         end
         else begin
             if (local_var(8) < 5) then begin
-                if (get_critter_stat(dude_obj, 4) < 4) then begin
-                    gsay_reply(603, 114);
+                if (dude_iq < 4) then begin
+                    Reply(114);
                 end
                 else begin
-                    if (global_var(301) != 5) then begin
-                        gsay_reply(603, 115);
+                    if (global_var(GVAR_BH_FTM) != 5) then begin
+                        Reply(115);
                     end
                     else begin
-                        gsay_reply(603, 192);
+                        Reply(192);
                     end
                 end
             end
             else begin
-                if (get_critter_stat(dude_obj, 4) < 4) then begin
-                    gsay_reply(603, 188);
+                if (dude_iq < 4) then begin
+                    Reply(188);
                 end
                 else begin
-                    if (global_var(301) != 5) then begin
-                        gsay_reply(603, 189);
+                    if (global_var(GVAR_BH_FTM) != 5) then begin
+                        Reply(189);
                     end
                     else begin
-                        gsay_reply(603, 191);
+                        Reply(191);
                     end
                 end
             end
         end
     end
-    if ((global_var(563) == 1) or (global_var(300) == 7) or (global_var(300) == 9)) then begin
-        giq_option(1, 603, 213, Node999, 50);
+    if ((global_var(GVAR_CHAD_DEAD) == 1) or (global_var(GVAR_BH_CHAD) == 7) or (global_var(GVAR_BH_CHAD) == 9)) then begin
+        giq_option(1, NAME, 213, Node999, NEUTRAL_REACTION);
     end
     else begin
         if (local_var(8) < 5) then begin
-            giq_option(-3, 603, 116, Node007, 50);
-            giq_option(-3, 603, 117, Node008, 50);
+            giq_option(-3, NAME, 116, Node007, NEUTRAL_REACTION);
+            giq_option(-3, NAME, 117, Node008, NEUTRAL_REACTION);
         end
         else begin
-            giq_option(-3, 603, 190, Node999, 50);
+            giq_option(-3, NAME, 190, Node999, NEUTRAL_REACTION);
         end
         if (local_var(8) < 5) then begin
-            giq_option(4, 603, 118, Node007, 50);
+            giq_option(4, NAME, 118, Node007, NEUTRAL_REACTION);
         end
         if (local_var(9) == 0) then begin
-            giq_option(4, 603, 119, Node009, 50);
-            giq_option(4, 603, 121, Node011, 50);
+            giq_option(4, NAME, 119, Node009, NEUTRAL_REACTION);
+            giq_option(4, NAME, 121, Node011, NEUTRAL_REACTION);
         end
-        if ((local_var(9) == 0) and (global_var(301) != 5)) then begin
-            giq_option(4, 603, 120, Node010, 50);
+        if ((local_var(9) == 0) and (global_var(GVAR_BH_FTM) != 5)) then begin
+            giq_option(4, NAME, 120, Node010, NEUTRAL_REACTION);
         end
         if (local_var(9) != 0) then begin
-            giq_option(4, 603, 195, Node003a, 50);
+            giq_option(4, NAME, 195, Node003a, NEUTRAL_REACTION);
         end
-        giq_option(4, 603, 122, Node012, 50);
-        giq_option(4, 603, 123, Node999, 50);
+        giq_option(4, NAME, 122, Node012, NEUTRAL_REACTION);
+        giq_option(4, NAME, 123, Node999, NEUTRAL_REACTION);
     end
 end
 
 procedure Node003a
 begin
     if (local_var(9) == 2) then begin
-        gsay_reply(603, 196);
+        Reply(196);
     end
     else begin
         if (local_var(9) == 3) then begin
-            gsay_reply(603, 197);
+            Reply(197);
         end
         else begin
             if (local_var(9) == 1) then begin
-                gsay_reply(603, 198);
+                Reply(198);
             end
         end
     end
-    giq_option(4, 603, 200, Node029, 50);
-    giq_option(4, 603, 201, Node999, 50);
+    giq_option(4, NAME, 200, Node029, NEUTRAL_REACTION);
+    giq_option(4, NAME, 201, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node004
 begin
-    gsay_message(603, 124, 50);
+    gsay_message(NAME, 124, NEUTRAL_REACTION);
     gdialog_mod_barter(-50);
-    gsay_reply(603, 210);
-    giq_option(4, 603, 211, Node013, 50);
+    Reply(210);
+    giq_option(4, NAME, 211, Node013, NEUTRAL_REACTION);
 end
 
 procedure Node006
 begin
-    gsay_reply(603, 129);
-    giq_option(4, 603, 130, Node003, 50);
-    giq_option(4, 603, 131, Node999, 50);
+    Reply(129);
+    giq_option(4, NAME, 130, Node003, NEUTRAL_REACTION);
+    giq_option(4, NAME, 131, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node007
 begin
-    gsay_message(603, 132, 50);
+    gsay_message(NAME, 132, NEUTRAL_REACTION);
     call Node007a();
 end
 
@@ -768,11 +768,11 @@ begin
     game_time_advance(24 * (60 * (60 * 10)));
     set_local_var(8, local_var(8) + 1);
     if (local_var(8) == 5) then begin
-        display_msg(message_str(603, 193));
+        display_mstr(193);
         give_exp_points(500);
-        display_msg(message_str(14, 100) + 500 + message_str(14, 101));
-        set_global_var(54, global_var(54) + -5);
-        critter_add_trait(dude_obj, 0, 116, 1);
+        display_msg(g_mstr(100) + 500 + g_mstr(101));
+        set_global_var(GVAR_TOWN_REP_BROKEN_HILLS, global_var(GVAR_TOWN_REP_BROKEN_HILLS) + -5);
+        critter_add_trait(dude_obj, TRAIT_PERK, PERK_expert_excrement_expediter_perk, 1);
     end
     gfade_in(600);
     call Node014();
@@ -780,198 +780,198 @@ end
 
 procedure Node008
 begin
-    gsay_reply(603, 133);
-    giq_option(-3, 603, 134, Node999, 50);
-    giq_option(4, 603, 135, Node012, 50);
-    giq_option(4, 603, 136, Node999, 50);
+    Reply(133);
+    giq_option(-3, NAME, 134, Node999, NEUTRAL_REACTION);
+    giq_option(4, NAME, 135, Node012, NEUTRAL_REACTION);
+    giq_option(4, NAME, 136, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node009
 begin
     set_local_var(9, 2);
-    set_global_var(271, 7);
+    set_global_var(GVAR_CARAVAN_END, 7);
     if (get_day != 1) then begin
-        gsay_reply(603, 137);
+        Reply(137);
     end
     else begin
-        gsay_reply(603, 204);
+        Reply(204);
     end
     if (get_day != 1) then begin
-        giq_option(4, 603, 138, Node012, 50);
-        giq_option(4, 603, 139, Node999, 50);
+        giq_option(4, NAME, 138, Node012, NEUTRAL_REACTION);
+        giq_option(4, NAME, 139, Node999, NEUTRAL_REACTION);
     end
     else begin
-        giq_option(4, 603, 207, StartCaravan, 50);
-        giq_option(4, 603, 208, Node999, 50);
-        giq_option(4, 603, 209, Node012, 50);
+        giq_option(4, NAME, 207, StartCaravan, NEUTRAL_REACTION);
+        giq_option(4, NAME, 208, Node999, NEUTRAL_REACTION);
+        giq_option(4, NAME, 209, Node012, NEUTRAL_REACTION);
     end
 end
 
 procedure Node010
 begin
-    set_global_var(271, 4);
+    set_global_var(GVAR_CARAVAN_END, 4);
     set_local_var(9, 3);
     if (get_day != 11) then begin
-        gsay_reply(603, 140);
+        Reply(140);
     end
     else begin
-        gsay_reply(603, 205);
+        Reply(205);
     end
     if (get_day != 11) then begin
-        giq_option(4, 603, 138, Node012, 50);
-        giq_option(4, 603, 139, Node999, 50);
+        giq_option(4, NAME, 138, Node012, NEUTRAL_REACTION);
+        giq_option(4, NAME, 139, Node999, NEUTRAL_REACTION);
     end
     else begin
-        giq_option(4, 603, 207, StartCaravan, 50);
-        giq_option(4, 603, 208, Node999, 50);
-        giq_option(4, 603, 209, Node012, 50);
+        giq_option(4, NAME, 207, StartCaravan, NEUTRAL_REACTION);
+        giq_option(4, NAME, 208, Node999, NEUTRAL_REACTION);
+        giq_option(4, NAME, 209, Node012, NEUTRAL_REACTION);
     end
 end
 
 procedure Node011
 begin
-    set_global_var(271, 2);
+    set_global_var(GVAR_CARAVAN_END, 2);
     set_local_var(9, 1);
     if (get_day != 21) then begin
-        gsay_reply(603, 141);
+        Reply(141);
     end
     else begin
-        gsay_reply(603, 206);
+        Reply(206);
     end
     if (get_day != 21) then begin
-        giq_option(4, 603, 138, Node012, 50);
-        giq_option(4, 603, 139, Node999, 50);
+        giq_option(4, NAME, 138, Node012, NEUTRAL_REACTION);
+        giq_option(4, NAME, 139, Node999, NEUTRAL_REACTION);
     end
     else begin
-        giq_option(4, 603, 207, StartCaravan, 50);
-        giq_option(4, 603, 208, Node999, 50);
-        giq_option(4, 603, 209, Node012, 50);
+        giq_option(4, NAME, 207, StartCaravan, NEUTRAL_REACTION);
+        giq_option(4, NAME, 208, Node999, NEUTRAL_REACTION);
+        giq_option(4, NAME, 209, Node012, NEUTRAL_REACTION);
     end
 end
 
 procedure Node012
 begin
-    gsay_reply(603, 142);
-    giq_option(4, 603, 143, Node003, 50);
-    giq_option(4, 603, 144, Node004, 50);
-    giq_option(4, 603, 148, Node006, 50);
+    Reply(142);
+    giq_option(4, NAME, 143, Node003, NEUTRAL_REACTION);
+    giq_option(4, NAME, 144, Node004, NEUTRAL_REACTION);
+    giq_option(4, NAME, 148, Node006, NEUTRAL_REACTION);
 end
 
 procedure Node013
 begin
-    gsay_reply(603, 149);
-    giq_option(4, 603, 138, Node012, 50);
-    giq_option(4, 603, 151, Node999, 50);
+    Reply(149);
+    giq_option(4, NAME, 138, Node012, NEUTRAL_REACTION);
+    giq_option(4, NAME, 151, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node014
 begin
     item_caps_adjust(dude_obj, 100);
-    gsay_message(603, 152, 50);
+    gsay_message(NAME, 152, NEUTRAL_REACTION);
 end
 
 procedure Node015
 begin
-    gsay_reply(603, 153);
-    giq_option(4, 603, 138, Node012, 50);
-    giq_option(4, 603, 139, Node999, 50);
+    Reply(153);
+    giq_option(4, NAME, 138, Node012, NEUTRAL_REACTION);
+    giq_option(4, NAME, 139, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node019
 begin
-    float_msg(self_obj, message_str(603, 166), 8);
+    float_msg(self_obj, mstr(166), FLOAT_MSG_YELLOW);
 end
 
 procedure Node022
 begin
-    gsay_reply(603, 170);
-    giq_option(4, 603, 168, Node999, 50);
+    Reply(170);
+    giq_option(4, NAME, 168, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node023
 begin
-    set_global_var(271, 2);
-    gsay_reply(603, 171);
-    giq_option(4, 603, 138, StartCaravan, 50);
-    giq_option(4, 603, 139, Node027, 50);
+    set_global_var(GVAR_CARAVAN_END, 2);
+    Reply(171);
+    giq_option(4, NAME, 138, StartCaravan, NEUTRAL_REACTION);
+    giq_option(4, NAME, 139, Node027, NEUTRAL_REACTION);
 end
 
 procedure Node024
 begin
-    set_global_var(271, 4);
-    gsay_reply(603, 172);
-    giq_option(4, 603, 138, StartCaravan, 50);
-    giq_option(4, 603, 139, Node027, 50);
+    set_global_var(GVAR_CARAVAN_END, 4);
+    Reply(172);
+    giq_option(4, NAME, 138, StartCaravan, NEUTRAL_REACTION);
+    giq_option(4, NAME, 139, Node027, NEUTRAL_REACTION);
 end
 
 procedure Node025
 begin
-    set_global_var(271, 7);
-    gsay_reply(603, 173);
-    giq_option(4, 603, 138, StartCaravan, 50);
-    giq_option(4, 603, 139, Node027, 50);
+    set_global_var(GVAR_CARAVAN_END, 7);
+    Reply(173);
+    giq_option(4, NAME, 138, StartCaravan, NEUTRAL_REACTION);
+    giq_option(4, NAME, 139, Node027, NEUTRAL_REACTION);
 end
 
 procedure Node026
 begin
     set_local_var(9, 0);
-    gsay_reply(603, 174);
-    giq_option(4, 603, 138, Node012, 50);
-    giq_option(4, 603, 139, Node999, 50);
+    Reply(174);
+    giq_option(4, NAME, 138, Node012, NEUTRAL_REACTION);
+    giq_option(4, NAME, 139, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node027
 begin
-    gsay_reply(603, 175);
-    giq_option(4, 603, 176, Node028, 50);
-    giq_option(4, 603, 177, Node029, 50);
-    giq_option(4, 603, 138, Node030, 50);
-    giq_option(4, 603, 139, Node031, 50);
+    Reply(175);
+    giq_option(4, NAME, 176, Node028, NEUTRAL_REACTION);
+    giq_option(4, NAME, 177, Node029, NEUTRAL_REACTION);
+    giq_option(4, NAME, 138, Node030, NEUTRAL_REACTION);
+    giq_option(4, NAME, 139, Node031, NEUTRAL_REACTION);
 end
 
 procedure Node028
 begin
     set_local_var(9, 0);
-    gsay_reply(603, 178);
-    giq_option(4, 603, 168, Node999, 50);
+    Reply(178);
+    giq_option(4, NAME, 168, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node029
 begin
     set_local_var(9, 0);
-    gsay_reply(603, 179);
-    giq_option(4, 603, 180, Node010, 50);
-    giq_option(4, 603, 181, Node009, 50);
-    giq_option(4, 603, 182, Node011, 50);
-    giq_option(4, 603, 203, Node999, 50);
+    Reply(179);
+    giq_option(4, NAME, 180, Node010, NEUTRAL_REACTION);
+    giq_option(4, NAME, 181, Node009, NEUTRAL_REACTION);
+    giq_option(4, NAME, 182, Node011, NEUTRAL_REACTION);
+    giq_option(4, NAME, 203, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node030
 begin
     set_local_var(9, 0);
-    gsay_reply(603, 183);
-    giq_option(4, 603, 168, Node999, 50);
+    Reply(183);
+    giq_option(4, NAME, 168, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node031
 begin
-    gsay_reply(603, 184);
-    giq_option(4, 603, 185, Node012, 50);
-    giq_option(4, 603, 186, StartCaravan, 50);
+    Reply(184);
+    giq_option(4, NAME, 185, Node012, NEUTRAL_REACTION);
+    giq_option(4, NAME, 186, StartCaravan, NEUTRAL_REACTION);
 end
 
 procedure Node032
 begin
-    gsay_reply(603, 187);
-    giq_option(4, 603, 168, Node999, 50);
+    Reply(187);
+    giq_option(4, NAME, 168, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node033
 begin
-    set_global_var(562, 1);
-    set_global_var(330, 0);
+    set_global_var(GVAR_BROKEN_HILLS_CARAVAN_POOCH_SCREW, 1);
+    set_global_var(GVAR_BH_CARAVAN, 0);
     set_local_var(7, 1);
-    gsay_message(603, 202, 51);
+    gsay_message(NAME, 202, BAD_REACTION);
 end
 
 procedure StartCaravan
@@ -980,28 +980,28 @@ begin
     variable LVar1 := 0;
     variable LVar2 := 0;
     set_local_var(9, 0);
-    set_global_var(330, 330);
-    set_global_var(270, 6);
-    set_global_var(330, 2);
-    set_global_var(269, 2);
+    set_global_var(GVAR_BH_CARAVAN, 330);
+    set_global_var(GVAR_CARAVAN_START, 6);
+    set_global_var(GVAR_BH_CARAVAN, 2);
+    set_global_var(GVAR_CARAVAN_STATUS, 2);
     LVar1 := random(0, 4);
-    set_global_var(283, LVar1);
-    set_global_var(275, LVar1);
-    if (global_var(283) == 0) then begin
+    set_global_var(GVAR_CARAVAN_ENCOUNTERS_TOTAL, LVar1);
+    set_global_var(GVAR_CARAVAN_ENCOUNTERS, LVar1);
+    if (global_var(GVAR_CARAVAN_ENCOUNTERS_TOTAL) == 0) then begin
         LVar2 := random(2, 4);
-        set_global_var(279, LVar2);
-        set_global_var(273, LVar2);
-        if (global_var(271) == 2) then begin
+        set_global_var(GVAR_CARAVAN_GUARDS_TOTAL, LVar2);
+        set_global_var(GVAR_CARAVAN_GUARDS, LVar2);
+        if (global_var(GVAR_CARAVAN_END) == 2) then begin
             game_time_advance(14 * (24 * (60 * (60 * 10))));
             load_map("VCTYCTYD.map", 0);
         end
         else begin
-            if (global_var(271) == 4) then begin
+            if (global_var(GVAR_CARAVAN_END) == 4) then begin
                 game_time_advance(14 * (24 * (60 * (60 * 10))));
                 load_map("NCRENT.MAP", 0);
             end
             else begin
-                if (global_var(271) == 7) then begin
+                if (global_var(GVAR_CARAVAN_END) == 7) then begin
                     game_time_advance(14 * (24 * (60 * (60 * 10))));
                     load_map("GECKSETL.map", 0);
                 end
@@ -1011,16 +1011,16 @@ begin
     else begin
         LVar0 := random(1, 3);
         LVar2 := random(2, 4);
-        set_global_var(282, 1);
-        set_global_var(277, 1);
-        set_global_var(279, LVar2);
-        set_global_var(273, LVar2);
-        set_global_var(280, LVar0);
-        set_global_var(274, LVar0);
-        set_global_var(281, 2 * LVar0);
-        set_global_var(276, 2 * LVar0);
-        set_global_var(278, LVar0);
-        set_global_var(272, LVar0);
+        set_global_var(GVAR_CARAVAN_MASTERS_TOTAL, 1);
+        set_global_var(GVAR_CARAVAN_MASTERS, 1);
+        set_global_var(GVAR_CARAVAN_GUARDS_TOTAL, LVar2);
+        set_global_var(GVAR_CARAVAN_GUARDS, LVar2);
+        set_global_var(GVAR_CARAVAN_CARTS_TOTAL, LVar0);
+        set_global_var(GVAR_CARAVAN_CARTS, LVar0);
+        set_global_var(GVAR_CARAVAN_BRAHMIN_TOTAL, 2 * LVar0);
+        set_global_var(GVAR_CARAVAN_BRAHMIN, 2 * LVar0);
+        set_global_var(GVAR_CARAVAN_DRIVERS_TOTAL, LVar0);
+        set_global_var(GVAR_CARAVAN_DRIVERS, LVar0);
         load_map("BHRNDDST.MAP", 11);
     end
     return 0;
@@ -1028,4 +1028,3 @@ begin
     LVar1 := 0;
     LVar2 := 0;
 end
-
