diff --git a/Mods/MegaMod/Original/HICDSHLF.ssl b/Mods/MegaMod/ReDefine/HICDSHLF.ssl
index e44524a..1e6dca4 100644
--- a/Mods/MegaMod/Original/HICDSHLF.ssl
+++ b/Mods/MegaMod/ReDefine/HICDSHLF.ssl
@@ -24,7 +24,7 @@ end
 procedure pickup_p_proc
 begin
     if (local_var(1) == 0) then begin
-        if ((global_var(563) == 1) or (map_var(25) == 2)) then begin
+        if ((global_var(GVAR_CHAD_DEAD) == 1) or (map_var(25) == 2)) then begin
             script_overrides;
             call Node992();
         end
@@ -49,38 +49,38 @@ end
 
 procedure Node001
 begin
-    gsay_reply(2066, 100);
-    giq_option(4, 2066, 101, Node993, 50);
-    giq_option(4, 2066, 102, Node999, 50);
+    Reply(100);
+    giq_option(4, NAME, 101, Node993, NEUTRAL_REACTION);
+    giq_option(4, NAME, 102, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node002
 begin
-    gsay_reply(2066, 103);
-    giq_option(4, 2066, 104, Node992, 50);
-    giq_option(4, 2066, 105, Node004, 50);
+    Reply(103);
+    giq_option(4, NAME, 104, Node992, NEUTRAL_REACTION);
+    giq_option(4, NAME, 105, Node004, NEUTRAL_REACTION);
 end
 
 procedure Node003
 begin
-    gsay_reply(2066, 106);
+    Reply(106);
     game_time_advance(60 * 10 * 30);
-    giq_option(4, 2066, message_str(14, 350), Node991, 50);
+    giq_option(4, NAME, g_mstr(350), Node991, NEUTRAL_REACTION);
 end
 
 procedure Node004
 begin
     game_time_advance(60 * 10 * 30);
-    gsay_reply(2066, 110);
-    giq_option(4, 2066, 107, Node992, 50);
-    giq_option(4, 2066, 102, Node999, 50);
+    Reply(110);
+    giq_option(4, NAME, 107, Node992, NEUTRAL_REACTION);
+    giq_option(4, NAME, 102, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node991
 begin
     variable LVar0 := 0;
-    LVar0 := create_object_sid(16777250, 0, 0, 613);
-    critter_add_trait(self_obj, 1, 5, 24);
+    LVar0 := create_object_sid(16777250, 0, 0, SCRIPT_HCCHAD);
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_TOUGH_CITIZEN);
     move_to(LVar0, 22052, 0);
     set_map_var(25, 1);
     debug_msg("map var:" + map_var(25));
@@ -91,19 +91,19 @@ end
 procedure Node992
 begin
     variable LVar0 := 0;
-    display_msg(message_str(2066, 108));
-    LVar0 := create_object_sid(72, 0, 0, -1);
+    display_mstr(108);
+    LVar0 := create_object(72, 0, 0);
     add_obj_to_inven(dude_obj, LVar0);
-    float_msg(dude_obj, message_str(2066, 109), 8);
+    float_msg(dude_obj, mstr(109), FLOAT_MSG_YELLOW);
     set_local_var(1, 1);
-    set_global_var(300, 5);
+    set_global_var(GVAR_BH_CHAD, 5);
     return 0;
     LVar0 := 0;
 end
 
 procedure Node993
 begin
-    if ((get_critter_stat(dude_obj, 1) > 6) and (get_critter_stat(dude_obj, 4) >= 5)) then begin
+    if ((dude_perception > 6) and (dude_iq >= 5)) then begin
         call Node002();
     end
     else begin
@@ -114,4 +114,3 @@ end
 procedure Node999
 begin
 end
-
