diff --git a/Mods/MegaMod/Original/HIDEAD.ssl b/Mods/MegaMod/ReDefine/HIDEAD.ssl
index acee940..52f5153 100644
--- a/Mods/MegaMod/Original/HIDEAD.ssl
+++ b/Mods/MegaMod/ReDefine/HIDEAD.ssl
@@ -22,10 +22,10 @@ procedure look_at_p_proc
 begin
     script_overrides;
     if (tile_num(self_obj) == 18453) then begin
-        display_msg(message_str(1080, 102));
+        display_mstr(102);
     end
     else begin
-        display_msg(message_str(1080, 100));
+        display_mstr(100);
     end
 end
 
@@ -34,28 +34,27 @@ begin
     script_overrides;
     if (tile_num(self_obj) == 18453) then begin
         if (obj_is_carrying_obj_pid(self_obj, 487) > 0) then begin
-            display_msg(message_str(1080, 103));
+            display_mstr(103);
         end
         else begin
-            display_msg(message_str(1080, 104));
+            display_mstr(104);
         end
     end
     else begin
-        display_msg(message_str(1080, 101));
+        display_mstr(101);
     end
 end
 
 procedure map_enter_p_proc
 begin
-    if (global_var(305) >= 2) then begin
-        if (not(has_trait(1, self_obj, 666))) then begin
+    if (global_var(GVAR_BH_MISSING) >= 2) then begin
+        if (not(obj_is_visible_flag(self_obj))) then begin
             set_obj_visibility(self_obj, 0);
         end
     end
     else begin
-        if (has_trait(1, self_obj, 666)) then begin
+        if (obj_is_visible_flag(self_obj)) then begin
             set_obj_visibility(self_obj, 1);
         end
     end
 end
-
