diff --git a/Mods/MegaMod/Original/HIMNDOR.ssl b/Mods/MegaMod/ReDefine/HIMNDOR.ssl
index 5666739..a87971a 100644
--- a/Mods/MegaMod/Original/HIMNDOR.ssl
+++ b/Mods/MegaMod/ReDefine/HIMNDOR.ssl
@@ -164,17 +164,17 @@ end
 procedure look_at_p_proc
 begin
     script_overrides;
-    display_msg(message_str(2068, 100));
+    display_mstr(100);
 end
 
 procedure description_p_proc
 begin
     script_overrides;
     if (obj_is_locked(self_obj)) then begin
-        display_msg(message_str(2068, 101));
+        display_mstr(101);
     end
     else begin
-        display_msg(message_str(2068, 102));
+        display_mstr(102);
     end
 end
 
@@ -188,12 +188,12 @@ end
 
 procedure damage_p_proc
 begin
-    if (metarule(49, target_obj) == 6) then begin
+    if (weapon_dmg_type(target_obj) == 6) then begin
         if (local_var(5) == 1) then begin
-            display_msg(message_str(2068, 110));
+            display_mstr(110);
         end
         else begin
-            display_msg(message_str(2068, 111));
+            display_mstr(111);
             destroy_object(self_obj);
         end
     end
@@ -215,4 +215,3 @@ end
 procedure timed_event_p_proc
 begin
 end
-
