diff --git a/Mods/MegaMod/Original/HS_TOHIT.ssl b/Mods/MegaMod/ReDefine/HS_TOHIT.ssl
index 5c7a3c8..ccdd4ce 100644
--- a/Mods/MegaMod/Original/HS_TOHIT.ssl
+++ b/Mods/MegaMod/ReDefine/HS_TOHIT.ssl
@@ -43,7 +43,7 @@ begin
             end
         end
         if ((LVar0 > 0) and (LVar3 > 0)) then begin
-            if (((party_member_obj(obj_pid(LVar0)) != 0) == 0) and (LVar0 != dude_obj)) then begin
+            if (((obj_in_party(LVar0)) == 0) and (LVar0 != dude_obj)) then begin
                 if (get_sfall_global_int(2191) > 0) then begin
                     LVar2 := LVar1 + get_sfall_global_int(2191) + Calculate_Facing_Bonus(LVar3, LVar0);
                     if (LVar2 > 100) then begin
@@ -54,7 +54,7 @@ begin
                 end
             end
             else begin
-                if ((party_member_obj(obj_pid(LVar0)) != 0) or (LVar0 == dude_obj)) then begin
+                if ((obj_in_party(LVar0)) or (LVar0 == dude_obj)) then begin
                     if (LVar0 == dude_obj) then begin
                         if (has_fake_perk("True Aim")) then begin
                             LVar1 := LVar1 + Calculate_Facing_Bonus(LVar3, LVar0);
@@ -345,19 +345,19 @@ begin
     if ((arg0 == dude_obj) and has_fake_perk("Lightning Reflexes")) then begin
         return 1;
     end
-    if (has_trait(1, arg1, 10) == has_trait(1, arg0, 10)) then begin
+    if (obj_get_rot(arg1) == obj_get_rot(arg0)) then begin
         return 4;
     end
     else begin
-        if ((has_trait(1, arg1, 10) == (has_trait(1, arg0, 10) + 1)) or (has_trait(1, arg1, 10) == (has_trait(1, arg0, 10) - 1))) then begin
+        if ((obj_get_rot(arg1) == (obj_get_rot(arg0) + 1)) or (obj_get_rot(arg1) == (obj_get_rot(arg0) - 1))) then begin
             return 3;
         end
         else begin
-            if ((has_trait(1, arg1, 10) == (has_trait(1, arg0, 10) + 2)) or (has_trait(1, arg1, 10) == (has_trait(1, arg0, 10) - 2))) then begin
+            if ((obj_get_rot(arg1) == (obj_get_rot(arg0) + 2)) or (obj_get_rot(arg1) == (obj_get_rot(arg0) - 2))) then begin
                 return 2;
             end
             else begin
-                if ((has_trait(1, arg1, 10) == (has_trait(1, arg0, 10) + 3)) or (has_trait(1, arg1, 10) == (has_trait(1, arg0, 10) - 3))) then begin
+                if ((obj_get_rot(arg1) == (obj_get_rot(arg0) + 3)) or (obj_get_rot(arg1) == (obj_get_rot(arg0) - 3))) then begin
                     return 1;
                 end
             end
@@ -391,7 +391,7 @@ end
 
 procedure modify_critter(variable arg0)
 begin
-    if (((party_member_obj(obj_pid(arg0)) != 0) == 0) and (arg0 != dude_obj) and (obj_pid(arg0) != 16777376) and (obj_pid(arg0) != 16777377) and (obj_pid(arg0) != 16777305) and (obj_pid(arg0) != 16777313) and (obj_pid(arg0) != 16777323) and (obj_pid(arg0) != 16777352) and (obj_pid(arg0) != 16777368) and (obj_pid(arg0) != 16777291) and (obj_pid(arg0) != 16777278) and (obj_pid(arg0) != 16777379) and (obj_pid(arg0) != 16777380) and (obj_pid(arg0) != 16777859) and (obj_pid(arg0) != 16777558) and (obj_pid(arg0) != 16777687) and (obj_pid(arg0) != 16777766) and (obj_pid(arg0) != 16777762) and (obj_pid(arg0) != 16777785) and (obj_pid(arg0) != 16777720) and (obj_pid(arg0) != 16777718) and (obj_pid(arg0) != 16777719) and (get_sfall_global_int(4306) == 0) or ((arg0 != dude_obj) and (get_sfall_global_int(4306) == 1))) then begin
+    if (((obj_in_party(arg0)) == 0) and (arg0 != dude_obj) and (obj_pid(arg0) != 16777376) and (obj_pid(arg0) != 16777377) and (obj_pid(arg0) != 16777305) and (obj_pid(arg0) != 16777313) and (obj_pid(arg0) != 16777323) and (obj_pid(arg0) != 16777352) and (obj_pid(arg0) != 16777368) and (obj_pid(arg0) != 16777291) and (obj_pid(arg0) != 16777278) and (obj_pid(arg0) != 16777379) and (obj_pid(arg0) != 16777380) and (obj_pid(arg0) != 16777859) and (obj_pid(arg0) != 16777558) and (obj_pid(arg0) != 16777687) and (obj_pid(arg0) != 16777766) and (obj_pid(arg0) != 16777762) and (obj_pid(arg0) != 16777785) and (obj_pid(arg0) != 16777720) and (obj_pid(arg0) != 16777718) and (obj_pid(arg0) != 16777719) and (get_sfall_global_int(4306) == 0) or ((arg0 != dude_obj) and (get_sfall_global_int(4306) == 1))) then begin
         if (get_critter_base_stat(arg0, 29) < 10) then begin
             set_critter_base_stat(arg0, 29, 10);
             if (get_sfall_global_int(2197) > 0) then begin
@@ -403,16 +403,16 @@ begin
                 end
             end
             if (get_sfall_global_int(2190) > 0) then begin
-                if ((get_critter_extra_stat(arg0, 8) + get_critter_base_stat(arg0, 8) + get_sfall_global_int(2190)) > (get_pc_stat(1) + 5)) then begin
-                    set_critter_extra_stat(arg0, 8, get_pc_stat(1) + 5 - (get_critter_extra_stat(arg0, 8) + get_critter_base_stat(arg0, 8)));
+                if ((get_critter_extra_stat(arg0, 8) + get_critter_base_stat(arg0, 8) + get_sfall_global_int(2190)) > (get_pc_stat(PCSTAT_level) + 5)) then begin
+                    set_critter_extra_stat(arg0, 8, get_pc_stat(PCSTAT_level) + 5 - (get_critter_extra_stat(arg0, 8) + get_critter_base_stat(arg0, 8)));
                 end
                 else begin
                     set_critter_extra_stat(arg0, 8, get_critter_extra_stat(arg0, 8) + get_sfall_global_int(2190));
                 end
             end
             if (get_sfall_global_int(2196) > 0) then begin
-                if ((get_critter_extra_stat(arg0, 9) + get_critter_base_stat(arg0, 9) + get_sfall_global_int(2196)) > (get_pc_stat(1) + 5)) then begin
-                    set_critter_extra_stat(arg0, 9, get_pc_stat(1) + 5 - (get_critter_extra_stat(arg0, 9) + get_critter_base_stat(arg0, 9)));
+                if ((get_critter_extra_stat(arg0, 9) + get_critter_base_stat(arg0, 9) + get_sfall_global_int(2196)) > (get_pc_stat(PCSTAT_level) + 5)) then begin
+                    set_critter_extra_stat(arg0, 9, get_pc_stat(PCSTAT_level) + 5 - (get_critter_extra_stat(arg0, 9) + get_critter_base_stat(arg0, 9)));
                 end
                 else begin
                     set_critter_extra_stat(arg0, 9, get_critter_extra_stat(arg0, 9) + get_sfall_global_int(2196));
@@ -426,7 +426,7 @@ procedure CalculateBonuses
 begin
     math := 0;
     if (get_sfall_global_int(2161) > 0) then begin
-        math := get_pc_stat(1) / get_sfall_global_int(2161);
+        math := get_pc_stat(PCSTAT_level) / get_sfall_global_int(2161);
     end
     if ((get_sfall_global_int(2162) > 0) and (math > 0)) then begin
         math := math * get_sfall_global_int(2162);
@@ -447,13 +447,13 @@ begin
             math := 0;
         end
     end
-    if (math > (get_pc_stat(1) + 5)) then begin
-        math := get_pc_stat(1) + 5;
+    if (math > (get_pc_stat(PCSTAT_level) + 5)) then begin
+        math := get_pc_stat(PCSTAT_level) + 5;
     end
     set_sfall_global(2190, math);
     math := 0;
     if (get_sfall_global_int(2164) > 0) then begin
-        math := get_pc_stat(1) / get_sfall_global_int(2164);
+        math := get_pc_stat(PCSTAT_level) / get_sfall_global_int(2164);
     end
     if ((get_sfall_global_int(2165) > 0) and (math > 0)) then begin
         math := math * get_sfall_global_int(2165);
@@ -480,7 +480,7 @@ begin
     set_sfall_global(2191, math);
     math := 0;
     if (get_sfall_global_int(2167) > 0) then begin
-        math := get_pc_stat(1) / get_sfall_global_int(2167);
+        math := get_pc_stat(PCSTAT_level) / get_sfall_global_int(2167);
     end
     if ((get_sfall_global_int(2168) > 0) and (math > 0)) then begin
         math := math * get_sfall_global_int(2168);
@@ -507,7 +507,7 @@ begin
     set_sfall_global(2192, math);
     math := 0;
     if (get_sfall_global_int(2170) > 0) then begin
-        math := get_pc_stat(1) / get_sfall_global_int(2170);
+        math := get_pc_stat(PCSTAT_level) / get_sfall_global_int(2170);
     end
     if ((get_sfall_global_int(2171) > 0) and (math > 0)) then begin
         math := math * get_sfall_global_int(2171);
@@ -528,13 +528,13 @@ begin
             math := 0;
         end
     end
-    if (math > (get_pc_stat(1) + 5)) then begin
-        math := get_pc_stat(1) + 5;
+    if (math > (get_pc_stat(PCSTAT_level) + 5)) then begin
+        math := get_pc_stat(PCSTAT_level) + 5;
     end
     set_sfall_global(2193, math);
     math := 0;
     if (get_sfall_global_int(2173) > 0) then begin
-        math := get_pc_stat(1) / get_sfall_global_int(2173);
+        math := get_pc_stat(PCSTAT_level) / get_sfall_global_int(2173);
     end
     if ((get_sfall_global_int(2174) > 0) and (math > 0)) then begin
         math := math * get_sfall_global_int(2174);
@@ -561,7 +561,7 @@ begin
     set_sfall_global(2194, math);
     math := 0;
     if (get_sfall_global_int(2176) > 0) then begin
-        math := get_pc_stat(1) / get_sfall_global_int(2176);
+        math := get_pc_stat(PCSTAT_level) / get_sfall_global_int(2176);
     end
     if ((get_sfall_global_int(2177) > 0) and (math > 0)) then begin
         math := math * get_sfall_global_int(2177);
@@ -588,7 +588,7 @@ begin
     set_sfall_global(2195, math);
     math := 0;
     if (get_sfall_global_int(2179) > 0) then begin
-        math := get_pc_stat(1) / get_sfall_global_int(2179);
+        math := get_pc_stat(PCSTAT_level) / get_sfall_global_int(2179);
     end
     if ((get_sfall_global_int(2180) > 0) and (math > 0)) then begin
         math := math * get_sfall_global_int(2180);
@@ -609,13 +609,13 @@ begin
             math := 0;
         end
     end
-    if (math > (get_pc_stat(1) + 5)) then begin
-        math := get_pc_stat(1) + 5;
+    if (math > (get_pc_stat(PCSTAT_level) + 5)) then begin
+        math := get_pc_stat(PCSTAT_level) + 5;
     end
     set_sfall_global(2196, math);
     math := 0;
     if (get_sfall_global_int(2182) > 0) then begin
-        math := get_pc_stat(1) * get_sfall_global_int(2182);
+        math := get_pc_stat(PCSTAT_level) * get_sfall_global_int(2182);
     end
     if (get_sfall_global_int(2183) > 0) then begin
         if (math > 0) then begin
@@ -639,7 +639,7 @@ begin
     set_sfall_global(2197, math);
     math := 0;
     if (get_sfall_global_int(2184) > 0) then begin
-        math := get_pc_stat(1) / get_sfall_global_int(2184);
+        math := get_pc_stat(PCSTAT_level) / get_sfall_global_int(2184);
     end
     if ((get_sfall_global_int(2185) > 0) and (math > 0)) then begin
         math := math * get_sfall_global_int(2185);
@@ -666,7 +666,7 @@ begin
     set_sfall_global(2198, math);
     math := 0;
     if (get_sfall_global_int(2187) > 0) then begin
-        math := get_pc_stat(1) / get_sfall_global_int(2187);
+        math := get_pc_stat(PCSTAT_level) / get_sfall_global_int(2187);
     end
     if ((get_sfall_global_int(2188) > 0) and (math > 0)) then begin
         math := math * get_sfall_global_int(2188);
@@ -693,7 +693,7 @@ begin
     set_sfall_global(2199, math);
     math := 0;
     if (get_sfall_global_int(4259) > 0) then begin
-        math := get_pc_stat(1) / get_sfall_global_int(4259);
+        math := get_pc_stat(PCSTAT_level) / get_sfall_global_int(4259);
     end
     if ((get_sfall_global_int(4260) > 0) and (math > 0)) then begin
         math := math * get_sfall_global_int(4260);
@@ -720,7 +720,7 @@ begin
     set_sfall_global(4262, math);
     math := 0;
     if (get_sfall_global_int(4297) > 0) then begin
-        math := get_pc_stat(1) / get_sfall_global_int(4297);
+        math := get_pc_stat(PCSTAT_level) / get_sfall_global_int(4297);
     end
     if ((get_sfall_global_int(4298) > 0) and (math > 0)) then begin
         math := math * get_sfall_global_int(4298);
@@ -749,7 +749,7 @@ end
 
 procedure HealCritter(variable arg0)
 begin
-    if (((party_member_obj(obj_pid(arg0)) != 0) == 0) and (arg0 != dude_obj) or ((arg0 != dude_obj) and (get_sfall_global_int(4306) == 1)) and not(combat_is_initialized) and not(critter_state(arg0) bwand 2)) then begin
+    if (((obj_in_party(arg0)) == 0) and (arg0 != dude_obj) or ((arg0 != dude_obj) and (get_sfall_global_int(4306) == 1)) and not(combat_is_initialized) and not(critter_state(arg0) bwand 2)) then begin
         critter_heal(arg0, 2000);
         critter_injure(arg0, 4 bwor 8 bwor 16 bwor 32 bwor 64 bwor 8388608);
     end
@@ -757,8 +757,7 @@ end
 
 procedure MapEnterHealCritter(variable arg0)
 begin
-    if (((party_member_obj(obj_pid(arg0)) != 0) == 0) and (arg0 != dude_obj) or ((arg0 != dude_obj) and (get_sfall_global_int(4306) == 1)) and not(critter_state(arg0) bwand 2)) then begin
+    if (((obj_in_party(arg0)) == 0) and (arg0 != dude_obj) or ((arg0 != dude_obj) and (get_sfall_global_int(4306) == 1)) and not(critter_state(arg0) bwand 2)) then begin
         critter_heal(arg0, 2000);
     end
 end
-
