diff --git a/Mods/MegaMod/Original/HS_USEOBJON.ssl b/Mods/MegaMod/ReDefine/HS_USEOBJON.ssl
index 590ab47..d9b26bb 100644
--- a/Mods/MegaMod/Original/HS_USEOBJON.ssl
+++ b/Mods/MegaMod/ReDefine/HS_USEOBJON.ssl
@@ -24,23 +24,23 @@ begin
         LVar4 := get_sfall_arg;
         if ((LVar3 == dude_obj) and (LVar2 == dude_obj)) then begin
             if (obj_pid(LVar4) == 778) then begin
-                if (has_skill(dude_obj, 0) < 300) then begin
+                if (has_skill(dude_obj, SKILL_SMALL_GUNS) < 300) then begin
                     LVar0 := random(3, 6);
                     LVar1 := random(2, 4);
                     gfade_out(10);
-                    if (has_trait(0, dude_obj, 81)) then begin
-                        critter_mod_skill(dude_obj, 0, LVar0);
-                        display_msg(message_str(1, 160) + LVar0 + message_str(1, 178));
+                    if (has_trait(TRAIT_PERK, dude_obj, PERK_comprehension_perk)) then begin
+                        critter_mod_skill(dude_obj, SKILL_SMALL_GUNS, LVar0);
+                        display_msg(message_str(SCRIPT_OBJ_DUDE, 160) + LVar0 + message_str(SCRIPT_OBJ_DUDE, 178));
                     end
                     else begin
-                        critter_mod_skill(dude_obj, 0, LVar1);
-                        display_msg(message_str(1, 160) + LVar1 + message_str(1, 178));
+                        critter_mod_skill(dude_obj, SKILL_SMALL_GUNS, LVar1);
+                        display_msg(message_str(SCRIPT_OBJ_DUDE, 160) + LVar1 + message_str(SCRIPT_OBJ_DUDE, 178));
                     end
                     game_time_advance(60 * (60 * 10));
                     gfade_in(10);
                 end
                 else begin
-                    display_msg(message_str(1, 203) + message_str(1, 205) + message_str(1, 204));
+                    display_msg(message_str(SCRIPT_OBJ_DUDE, 203) + message_str(SCRIPT_OBJ_DUDE, 205) + message_str(SCRIPT_OBJ_DUDE, 204));
                     gfade_out(10);
                     game_time_advance(60 * (60 * 10));
                     gfade_in(10);
@@ -48,23 +48,23 @@ begin
             end
             else begin
                 if (obj_pid(LVar4) == 779) then begin
-                    if (has_skill(dude_obj, 1) < 300) then begin
+                    if (has_skill(dude_obj, SKILL_BIG_GUNS) < 300) then begin
                         LVar0 := random(3, 6);
                         LVar1 := random(2, 4);
                         gfade_out(10);
-                        if (has_trait(0, dude_obj, 81)) then begin
-                            critter_mod_skill(dude_obj, 1, LVar0);
-                            display_msg(message_str(1, 161) + LVar0 + message_str(1, 178));
+                        if (has_trait(TRAIT_PERK, dude_obj, PERK_comprehension_perk)) then begin
+                            critter_mod_skill(dude_obj, SKILL_BIG_GUNS, LVar0);
+                            display_msg(message_str(SCRIPT_OBJ_DUDE, 161) + LVar0 + message_str(SCRIPT_OBJ_DUDE, 178));
                         end
                         else begin
-                            critter_mod_skill(dude_obj, 1, LVar1);
-                            display_msg(message_str(1, 161) + LVar1 + message_str(1, 178));
+                            critter_mod_skill(dude_obj, SKILL_BIG_GUNS, LVar1);
+                            display_msg(message_str(SCRIPT_OBJ_DUDE, 161) + LVar1 + message_str(SCRIPT_OBJ_DUDE, 178));
                         end
                         game_time_advance(60 * (60 * 10));
                         gfade_in(10);
                     end
                     else begin
-                        display_msg(message_str(1, 203) + message_str(1, 206) + message_str(1, 204));
+                        display_msg(message_str(SCRIPT_OBJ_DUDE, 203) + message_str(SCRIPT_OBJ_DUDE, 206) + message_str(SCRIPT_OBJ_DUDE, 204));
                         gfade_out(10);
                         game_time_advance(60 * (60 * 10));
                         gfade_in(10);
@@ -72,23 +72,23 @@ begin
                 end
                 else begin
                     if (obj_pid(LVar4) == 780) then begin
-                        if (has_skill(dude_obj, 2) < 300) then begin
+                        if (has_skill(dude_obj, SKILL_ENERGY_WEAPONS) < 300) then begin
                             LVar0 := random(3, 6);
                             LVar1 := random(2, 4);
                             gfade_out(10);
-                            if (has_trait(0, dude_obj, 81)) then begin
-                                critter_mod_skill(dude_obj, 2, LVar0);
-                                display_msg(message_str(1, 162) + LVar0 + message_str(1, 178));
+                            if (has_trait(TRAIT_PERK, dude_obj, PERK_comprehension_perk)) then begin
+                                critter_mod_skill(dude_obj, SKILL_ENERGY_WEAPONS, LVar0);
+                                display_msg(message_str(SCRIPT_OBJ_DUDE, 162) + LVar0 + message_str(SCRIPT_OBJ_DUDE, 178));
                             end
                             else begin
-                                critter_mod_skill(dude_obj, 2, LVar1);
-                                display_msg(message_str(1, 162) + LVar1 + message_str(1, 178));
+                                critter_mod_skill(dude_obj, SKILL_ENERGY_WEAPONS, LVar1);
+                                display_msg(message_str(SCRIPT_OBJ_DUDE, 162) + LVar1 + message_str(SCRIPT_OBJ_DUDE, 178));
                             end
                             game_time_advance(60 * (60 * 10));
                             gfade_in(10);
                         end
                         else begin
-                            display_msg(message_str(1, 203) + message_str(1, 207) + message_str(1, 204));
+                            display_msg(message_str(SCRIPT_OBJ_DUDE, 203) + message_str(SCRIPT_OBJ_DUDE, 207) + message_str(SCRIPT_OBJ_DUDE, 204));
                             gfade_out(10);
                             game_time_advance(60 * (60 * 10));
                             gfade_in(10);
@@ -96,23 +96,23 @@ begin
                     end
                     else begin
                         if (obj_pid(LVar4) == 781) then begin
-                            if (has_skill(dude_obj, 3) < 300) then begin
+                            if (has_skill(dude_obj, SKILL_UNARMED_COMBAT) < 300) then begin
                                 LVar0 := random(3, 6);
                                 LVar1 := random(2, 4);
                                 gfade_out(10);
-                                if (has_trait(0, dude_obj, 81)) then begin
-                                    critter_mod_skill(dude_obj, 3, LVar0);
-                                    display_msg(message_str(1, 163) + LVar0 + message_str(1, 178));
+                                if (has_trait(TRAIT_PERK, dude_obj, PERK_comprehension_perk)) then begin
+                                    critter_mod_skill(dude_obj, SKILL_UNARMED_COMBAT, LVar0);
+                                    display_msg(message_str(SCRIPT_OBJ_DUDE, 163) + LVar0 + message_str(SCRIPT_OBJ_DUDE, 178));
                                 end
                                 else begin
-                                    critter_mod_skill(dude_obj, 3, LVar1);
-                                    display_msg(message_str(1, 163) + LVar1 + message_str(1, 178));
+                                    critter_mod_skill(dude_obj, SKILL_UNARMED_COMBAT, LVar1);
+                                    display_msg(message_str(SCRIPT_OBJ_DUDE, 163) + LVar1 + message_str(SCRIPT_OBJ_DUDE, 178));
                                 end
                                 game_time_advance(60 * (60 * 10));
                                 gfade_in(10);
                             end
                             else begin
-                                display_msg(message_str(1, 203) + message_str(1, 208) + message_str(1, 204));
+                                display_msg(message_str(SCRIPT_OBJ_DUDE, 203) + message_str(SCRIPT_OBJ_DUDE, 208) + message_str(SCRIPT_OBJ_DUDE, 204));
                                 gfade_out(10);
                                 game_time_advance(60 * (60 * 10));
                                 gfade_in(10);
@@ -120,23 +120,23 @@ begin
                         end
                         else begin
                             if (obj_pid(LVar4) == 782) then begin
-                                if (has_skill(dude_obj, 4) < 300) then begin
+                                if (has_skill(dude_obj, SKILL_MELEE) < 300) then begin
                                     LVar0 := random(3, 6);
                                     LVar1 := random(2, 4);
                                     gfade_out(10);
-                                    if (has_trait(0, dude_obj, 81)) then begin
-                                        critter_mod_skill(dude_obj, 4, LVar0);
-                                        display_msg(message_str(1, 164) + LVar0 + message_str(1, 178));
+                                    if (has_trait(TRAIT_PERK, dude_obj, PERK_comprehension_perk)) then begin
+                                        critter_mod_skill(dude_obj, SKILL_MELEE, LVar0);
+                                        display_msg(message_str(SCRIPT_OBJ_DUDE, 164) + LVar0 + message_str(SCRIPT_OBJ_DUDE, 178));
                                     end
                                     else begin
-                                        critter_mod_skill(dude_obj, 4, LVar1);
-                                        display_msg(message_str(1, 164) + LVar1 + message_str(1, 178));
+                                        critter_mod_skill(dude_obj, SKILL_MELEE, LVar1);
+                                        display_msg(message_str(SCRIPT_OBJ_DUDE, 164) + LVar1 + message_str(SCRIPT_OBJ_DUDE, 178));
                                     end
                                     game_time_advance(60 * (60 * 10));
                                     gfade_in(10);
                                 end
                                 else begin
-                                    display_msg(message_str(1, 203) + message_str(1, 209) + message_str(1, 204));
+                                    display_msg(message_str(SCRIPT_OBJ_DUDE, 203) + message_str(SCRIPT_OBJ_DUDE, 209) + message_str(SCRIPT_OBJ_DUDE, 204));
                                     gfade_out(10);
                                     game_time_advance(60 * (60 * 10));
                                     gfade_in(10);
@@ -144,23 +144,23 @@ begin
                             end
                             else begin
                                 if (obj_pid(LVar4) == 783) then begin
-                                    if (has_skill(dude_obj, 5) < 300) then begin
+                                    if (has_skill(dude_obj, SKILL_THROWING) < 300) then begin
                                         LVar0 := random(3, 6);
                                         LVar1 := random(2, 4);
                                         gfade_out(10);
-                                        if (has_trait(0, dude_obj, 81)) then begin
-                                            critter_mod_skill(dude_obj, 5, LVar0);
-                                            display_msg(message_str(1, 165) + LVar0 + message_str(1, 178));
+                                        if (has_trait(TRAIT_PERK, dude_obj, PERK_comprehension_perk)) then begin
+                                            critter_mod_skill(dude_obj, SKILL_THROWING, LVar0);
+                                            display_msg(message_str(SCRIPT_OBJ_DUDE, 165) + LVar0 + message_str(SCRIPT_OBJ_DUDE, 178));
                                         end
                                         else begin
-                                            critter_mod_skill(dude_obj, 5, LVar1);
-                                            display_msg(message_str(1, 165) + LVar1 + message_str(1, 178));
+                                            critter_mod_skill(dude_obj, SKILL_THROWING, LVar1);
+                                            display_msg(message_str(SCRIPT_OBJ_DUDE, 165) + LVar1 + message_str(SCRIPT_OBJ_DUDE, 178));
                                         end
                                         game_time_advance(60 * (60 * 10));
                                         gfade_in(10);
                                     end
                                     else begin
-                                        display_msg(message_str(1, 203) + message_str(1, 210) + message_str(1, 204));
+                                        display_msg(message_str(SCRIPT_OBJ_DUDE, 203) + message_str(SCRIPT_OBJ_DUDE, 210) + message_str(SCRIPT_OBJ_DUDE, 204));
                                         gfade_out(10);
                                         game_time_advance(60 * (60 * 10));
                                         gfade_in(10);
@@ -168,23 +168,23 @@ begin
                                 end
                                 else begin
                                     if (obj_pid(LVar4) == 784) then begin
-                                        if (has_skill(dude_obj, 6) < 300) then begin
+                                        if (has_skill(dude_obj, SKILL_FIRST_AID) < 300) then begin
                                             LVar0 := random(3, 6);
                                             LVar1 := random(2, 4);
                                             gfade_out(10);
-                                            if (has_trait(0, dude_obj, 81)) then begin
-                                                critter_mod_skill(dude_obj, 6, LVar0);
-                                                display_msg(message_str(1, 166) + LVar0 + message_str(1, 178));
+                                            if (has_trait(TRAIT_PERK, dude_obj, PERK_comprehension_perk)) then begin
+                                                critter_mod_skill(dude_obj, SKILL_FIRST_AID, LVar0);
+                                                display_msg(message_str(SCRIPT_OBJ_DUDE, 166) + LVar0 + message_str(SCRIPT_OBJ_DUDE, 178));
                                             end
                                             else begin
-                                                critter_mod_skill(dude_obj, 6, LVar1);
-                                                display_msg(message_str(1, 166) + LVar1 + message_str(1, 178));
+                                                critter_mod_skill(dude_obj, SKILL_FIRST_AID, LVar1);
+                                                display_msg(message_str(SCRIPT_OBJ_DUDE, 166) + LVar1 + message_str(SCRIPT_OBJ_DUDE, 178));
                                             end
                                             game_time_advance(60 * (60 * 10));
                                             gfade_in(10);
                                         end
                                         else begin
-                                            display_msg(message_str(1, 203) + message_str(1, 211) + message_str(1, 204));
+                                            display_msg(message_str(SCRIPT_OBJ_DUDE, 203) + message_str(SCRIPT_OBJ_DUDE, 211) + message_str(SCRIPT_OBJ_DUDE, 204));
                                             gfade_out(10);
                                             game_time_advance(60 * (60 * 10));
                                             gfade_in(10);
@@ -192,23 +192,23 @@ begin
                                     end
                                     else begin
                                         if (obj_pid(LVar4) == 785) then begin
-                                            if (has_skill(dude_obj, 7) < 300) then begin
+                                            if (has_skill(dude_obj, SKILL_DOCTOR) < 300) then begin
                                                 LVar0 := random(3, 6);
                                                 LVar1 := random(2, 4);
                                                 gfade_out(10);
-                                                if (has_trait(0, dude_obj, 81)) then begin
-                                                    critter_mod_skill(dude_obj, 7, LVar0);
-                                                    display_msg(message_str(1, 167) + LVar0 + message_str(1, 178));
+                                                if (has_trait(TRAIT_PERK, dude_obj, PERK_comprehension_perk)) then begin
+                                                    critter_mod_skill(dude_obj, SKILL_DOCTOR, LVar0);
+                                                    display_msg(message_str(SCRIPT_OBJ_DUDE, 167) + LVar0 + message_str(SCRIPT_OBJ_DUDE, 178));
                                                 end
                                                 else begin
-                                                    critter_mod_skill(dude_obj, 7, LVar1);
-                                                    display_msg(message_str(1, 167) + LVar1 + message_str(1, 178));
+                                                    critter_mod_skill(dude_obj, SKILL_DOCTOR, LVar1);
+                                                    display_msg(message_str(SCRIPT_OBJ_DUDE, 167) + LVar1 + message_str(SCRIPT_OBJ_DUDE, 178));
                                                 end
                                                 game_time_advance(60 * (60 * 10));
                                                 gfade_in(10);
                                             end
                                             else begin
-                                                display_msg(message_str(1, 203) + message_str(1, 212) + message_str(1, 204));
+                                                display_msg(message_str(SCRIPT_OBJ_DUDE, 203) + message_str(SCRIPT_OBJ_DUDE, 212) + message_str(SCRIPT_OBJ_DUDE, 204));
                                                 gfade_out(10);
                                                 game_time_advance(60 * (60 * 10));
                                                 gfade_in(10);
@@ -216,23 +216,23 @@ begin
                                         end
                                         else begin
                                             if (obj_pid(LVar4) == 786) then begin
-                                                if (has_skill(dude_obj, 8) < 300) then begin
+                                                if (has_skill(dude_obj, SKILL_SNEAK) < 300) then begin
                                                     LVar0 := random(3, 6);
                                                     LVar1 := random(2, 4);
                                                     gfade_out(10);
-                                                    if (has_trait(0, dude_obj, 81)) then begin
-                                                        critter_mod_skill(dude_obj, 8, LVar0);
-                                                        display_msg(message_str(1, 168) + LVar0 + message_str(1, 178));
+                                                    if (has_trait(TRAIT_PERK, dude_obj, PERK_comprehension_perk)) then begin
+                                                        critter_mod_skill(dude_obj, SKILL_SNEAK, LVar0);
+                                                        display_msg(message_str(SCRIPT_OBJ_DUDE, 168) + LVar0 + message_str(SCRIPT_OBJ_DUDE, 178));
                                                     end
                                                     else begin
-                                                        critter_mod_skill(dude_obj, 8, LVar1);
-                                                        display_msg(message_str(1, 168) + LVar1 + message_str(1, 178));
+                                                        critter_mod_skill(dude_obj, SKILL_SNEAK, LVar1);
+                                                        display_msg(message_str(SCRIPT_OBJ_DUDE, 168) + LVar1 + message_str(SCRIPT_OBJ_DUDE, 178));
                                                     end
                                                     game_time_advance(60 * (60 * 10));
                                                     gfade_in(10);
                                                 end
                                                 else begin
-                                                    display_msg(message_str(1, 203) + message_str(1, 213) + message_str(1, 204));
+                                                    display_msg(message_str(SCRIPT_OBJ_DUDE, 203) + message_str(SCRIPT_OBJ_DUDE, 213) + message_str(SCRIPT_OBJ_DUDE, 204));
                                                     gfade_out(10);
                                                     game_time_advance(60 * (60 * 10));
                                                     gfade_in(10);
@@ -240,23 +240,23 @@ begin
                                             end
                                             else begin
                                                 if (obj_pid(LVar4) == 787) then begin
-                                                    if (has_skill(dude_obj, 9) < 300) then begin
+                                                    if (has_skill(dude_obj, SKILL_LOCKPICK) < 300) then begin
                                                         LVar0 := random(3, 6);
                                                         LVar1 := random(2, 4);
                                                         gfade_out(10);
-                                                        if (has_trait(0, dude_obj, 81)) then begin
-                                                            critter_mod_skill(dude_obj, 9, LVar0);
-                                                            display_msg(message_str(1, 169) + LVar0 + message_str(1, 178));
+                                                        if (has_trait(TRAIT_PERK, dude_obj, PERK_comprehension_perk)) then begin
+                                                            critter_mod_skill(dude_obj, SKILL_LOCKPICK, LVar0);
+                                                            display_msg(message_str(SCRIPT_OBJ_DUDE, 169) + LVar0 + message_str(SCRIPT_OBJ_DUDE, 178));
                                                         end
                                                         else begin
-                                                            critter_mod_skill(dude_obj, 9, LVar1);
-                                                            display_msg(message_str(1, 169) + LVar1 + message_str(1, 178));
+                                                            critter_mod_skill(dude_obj, SKILL_LOCKPICK, LVar1);
+                                                            display_msg(message_str(SCRIPT_OBJ_DUDE, 169) + LVar1 + message_str(SCRIPT_OBJ_DUDE, 178));
                                                         end
                                                         game_time_advance(60 * (60 * 10));
                                                         gfade_in(10);
                                                     end
                                                     else begin
-                                                        display_msg(message_str(1, 203) + message_str(1, 214) + message_str(1, 204));
+                                                        display_msg(message_str(SCRIPT_OBJ_DUDE, 203) + message_str(SCRIPT_OBJ_DUDE, 214) + message_str(SCRIPT_OBJ_DUDE, 204));
                                                         gfade_out(10);
                                                         game_time_advance(60 * (60 * 10));
                                                         gfade_in(10);
@@ -264,23 +264,23 @@ begin
                                                 end
                                                 else begin
                                                     if (obj_pid(LVar4) == 788) then begin
-                                                        if (has_skill(dude_obj, 10) < 300) then begin
+                                                        if (has_skill(dude_obj, SKILL_STEAL) < 300) then begin
                                                             LVar0 := random(3, 6);
                                                             LVar1 := random(2, 4);
                                                             gfade_out(10);
-                                                            if (has_trait(0, dude_obj, 81)) then begin
-                                                                critter_mod_skill(dude_obj, 10, LVar0);
-                                                                display_msg(message_str(1, 170) + LVar0 + message_str(1, 178));
+                                                            if (has_trait(TRAIT_PERK, dude_obj, PERK_comprehension_perk)) then begin
+                                                                critter_mod_skill(dude_obj, SKILL_STEAL, LVar0);
+                                                                display_msg(message_str(SCRIPT_OBJ_DUDE, 170) + LVar0 + message_str(SCRIPT_OBJ_DUDE, 178));
                                                             end
                                                             else begin
-                                                                critter_mod_skill(dude_obj, 10, LVar1);
-                                                                display_msg(message_str(1, 170) + LVar1 + message_str(1, 178));
+                                                                critter_mod_skill(dude_obj, SKILL_STEAL, LVar1);
+                                                                display_msg(message_str(SCRIPT_OBJ_DUDE, 170) + LVar1 + message_str(SCRIPT_OBJ_DUDE, 178));
                                                             end
                                                             game_time_advance(60 * (60 * 10));
                                                             gfade_in(10);
                                                         end
                                                         else begin
-                                                            display_msg(message_str(1, 203) + message_str(1, 215) + message_str(1, 204));
+                                                            display_msg(message_str(SCRIPT_OBJ_DUDE, 203) + message_str(SCRIPT_OBJ_DUDE, 215) + message_str(SCRIPT_OBJ_DUDE, 204));
                                                             gfade_out(10);
                                                             game_time_advance(60 * (60 * 10));
                                                             gfade_in(10);
@@ -288,23 +288,23 @@ begin
                                                     end
                                                     else begin
                                                         if (obj_pid(LVar4) == 789) then begin
-                                                            if (has_skill(dude_obj, 11) < 300) then begin
+                                                            if (has_skill(dude_obj, SKILL_TRAPS) < 300) then begin
                                                                 LVar0 := random(3, 6);
                                                                 LVar1 := random(2, 4);
                                                                 gfade_out(10);
-                                                                if (has_trait(0, dude_obj, 81)) then begin
-                                                                    critter_mod_skill(dude_obj, 11, LVar0);
-                                                                    display_msg(message_str(1, 171) + LVar0 + message_str(1, 178));
+                                                                if (has_trait(TRAIT_PERK, dude_obj, PERK_comprehension_perk)) then begin
+                                                                    critter_mod_skill(dude_obj, SKILL_TRAPS, LVar0);
+                                                                    display_msg(message_str(SCRIPT_OBJ_DUDE, 171) + LVar0 + message_str(SCRIPT_OBJ_DUDE, 178));
                                                                 end
                                                                 else begin
-                                                                    critter_mod_skill(dude_obj, 11, LVar1);
-                                                                    display_msg(message_str(1, 171) + LVar1 + message_str(1, 178));
+                                                                    critter_mod_skill(dude_obj, SKILL_TRAPS, LVar1);
+                                                                    display_msg(message_str(SCRIPT_OBJ_DUDE, 171) + LVar1 + message_str(SCRIPT_OBJ_DUDE, 178));
                                                                 end
                                                                 game_time_advance(60 * (60 * 10));
                                                                 gfade_in(10);
                                                             end
                                                             else begin
-                                                                display_msg(message_str(1, 203) + message_str(1, 216) + message_str(1, 204));
+                                                                display_msg(message_str(SCRIPT_OBJ_DUDE, 203) + message_str(SCRIPT_OBJ_DUDE, 216) + message_str(SCRIPT_OBJ_DUDE, 204));
                                                                 gfade_out(10);
                                                                 game_time_advance(60 * (60 * 10));
                                                                 gfade_in(10);
@@ -312,23 +312,23 @@ begin
                                                         end
                                                         else begin
                                                             if (obj_pid(LVar4) == 790) then begin
-                                                                if (has_skill(dude_obj, 12) < 300) then begin
+                                                                if (has_skill(dude_obj, SKILL_SCIENCE) < 300) then begin
                                                                     LVar0 := random(3, 6);
                                                                     LVar1 := random(2, 4);
                                                                     gfade_out(10);
-                                                                    if (has_trait(0, dude_obj, 81)) then begin
-                                                                        critter_mod_skill(dude_obj, 12, LVar0);
-                                                                        display_msg(message_str(1, 172) + LVar0 + message_str(1, 178));
+                                                                    if (has_trait(TRAIT_PERK, dude_obj, PERK_comprehension_perk)) then begin
+                                                                        critter_mod_skill(dude_obj, SKILL_SCIENCE, LVar0);
+                                                                        display_msg(message_str(SCRIPT_OBJ_DUDE, 172) + LVar0 + message_str(SCRIPT_OBJ_DUDE, 178));
                                                                     end
                                                                     else begin
-                                                                        critter_mod_skill(dude_obj, 12, LVar1);
-                                                                        display_msg(message_str(1, 172) + LVar1 + message_str(1, 178));
+                                                                        critter_mod_skill(dude_obj, SKILL_SCIENCE, LVar1);
+                                                                        display_msg(message_str(SCRIPT_OBJ_DUDE, 172) + LVar1 + message_str(SCRIPT_OBJ_DUDE, 178));
                                                                     end
                                                                     game_time_advance(60 * (60 * 10));
                                                                     gfade_in(10);
                                                                 end
                                                                 else begin
-                                                                    display_msg(message_str(1, 203) + message_str(1, 217) + message_str(1, 204));
+                                                                    display_msg(message_str(SCRIPT_OBJ_DUDE, 203) + message_str(SCRIPT_OBJ_DUDE, 217) + message_str(SCRIPT_OBJ_DUDE, 204));
                                                                     gfade_out(10);
                                                                     game_time_advance(60 * (60 * 10));
                                                                     gfade_in(10);
@@ -336,23 +336,23 @@ begin
                                                             end
                                                             else begin
                                                                 if (obj_pid(LVar4) == 791) then begin
-                                                                    if (has_skill(dude_obj, 13) < 300) then begin
+                                                                    if (has_skill(dude_obj, SKILL_REPAIR) < 300) then begin
                                                                         LVar0 := random(3, 6);
                                                                         LVar1 := random(2, 4);
                                                                         gfade_out(10);
-                                                                        if (has_trait(0, dude_obj, 81)) then begin
-                                                                            critter_mod_skill(dude_obj, 13, LVar0);
-                                                                            display_msg(message_str(1, 173) + LVar0 + message_str(1, 178));
+                                                                        if (has_trait(TRAIT_PERK, dude_obj, PERK_comprehension_perk)) then begin
+                                                                            critter_mod_skill(dude_obj, SKILL_REPAIR, LVar0);
+                                                                            display_msg(message_str(SCRIPT_OBJ_DUDE, 173) + LVar0 + message_str(SCRIPT_OBJ_DUDE, 178));
                                                                         end
                                                                         else begin
-                                                                            critter_mod_skill(dude_obj, 13, LVar1);
-                                                                            display_msg(message_str(1, 173) + LVar1 + message_str(1, 178));
+                                                                            critter_mod_skill(dude_obj, SKILL_REPAIR, LVar1);
+                                                                            display_msg(message_str(SCRIPT_OBJ_DUDE, 173) + LVar1 + message_str(SCRIPT_OBJ_DUDE, 178));
                                                                         end
                                                                         game_time_advance(60 * (60 * 10));
                                                                         gfade_in(10);
                                                                     end
                                                                     else begin
-                                                                        display_msg(message_str(1, 203) + message_str(1, 218) + message_str(1, 204));
+                                                                        display_msg(message_str(SCRIPT_OBJ_DUDE, 203) + message_str(SCRIPT_OBJ_DUDE, 218) + message_str(SCRIPT_OBJ_DUDE, 204));
                                                                         gfade_out(10);
                                                                         game_time_advance(60 * (60 * 10));
                                                                         gfade_in(10);
@@ -360,23 +360,23 @@ begin
                                                                 end
                                                                 else begin
                                                                     if (obj_pid(LVar4) == 792) then begin
-                                                                        if (has_skill(dude_obj, 14) < 300) then begin
+                                                                        if (has_skill(dude_obj, SKILL_CONVERSANT) < 300) then begin
                                                                             LVar0 := random(3, 6);
                                                                             LVar1 := random(2, 4);
                                                                             gfade_out(10);
-                                                                            if (has_trait(0, dude_obj, 81)) then begin
-                                                                                critter_mod_skill(dude_obj, 14, LVar0);
-                                                                                display_msg(message_str(1, 174) + LVar0 + message_str(1, 178));
+                                                                            if (has_trait(TRAIT_PERK, dude_obj, PERK_comprehension_perk)) then begin
+                                                                                critter_mod_skill(dude_obj, SKILL_CONVERSANT, LVar0);
+                                                                                display_msg(message_str(SCRIPT_OBJ_DUDE, 174) + LVar0 + message_str(SCRIPT_OBJ_DUDE, 178));
                                                                             end
                                                                             else begin
-                                                                                critter_mod_skill(dude_obj, 14, LVar1);
-                                                                                display_msg(message_str(1, 174) + LVar1 + message_str(1, 178));
+                                                                                critter_mod_skill(dude_obj, SKILL_CONVERSANT, LVar1);
+                                                                                display_msg(message_str(SCRIPT_OBJ_DUDE, 174) + LVar1 + message_str(SCRIPT_OBJ_DUDE, 178));
                                                                             end
                                                                             game_time_advance(60 * (60 * 10));
                                                                             gfade_in(10);
                                                                         end
                                                                         else begin
-                                                                            display_msg(message_str(1, 203) + message_str(1, 219) + message_str(1, 204));
+                                                                            display_msg(message_str(SCRIPT_OBJ_DUDE, 203) + message_str(SCRIPT_OBJ_DUDE, 219) + message_str(SCRIPT_OBJ_DUDE, 204));
                                                                             gfade_out(10);
                                                                             game_time_advance(60 * (60 * 10));
                                                                             gfade_in(10);
@@ -384,23 +384,23 @@ begin
                                                                     end
                                                                     else begin
                                                                         if (obj_pid(LVar4) == 793) then begin
-                                                                            if (has_skill(dude_obj, 15) < 300) then begin
+                                                                            if (has_skill(dude_obj, SKILL_BARTER) < 300) then begin
                                                                                 LVar0 := random(3, 6);
                                                                                 LVar1 := random(2, 4);
                                                                                 gfade_out(10);
-                                                                                if (has_trait(0, dude_obj, 81)) then begin
-                                                                                    critter_mod_skill(dude_obj, 15, LVar0);
-                                                                                    display_msg(message_str(1, 175) + LVar0 + message_str(1, 178));
+                                                                                if (has_trait(TRAIT_PERK, dude_obj, PERK_comprehension_perk)) then begin
+                                                                                    critter_mod_skill(dude_obj, SKILL_BARTER, LVar0);
+                                                                                    display_msg(message_str(SCRIPT_OBJ_DUDE, 175) + LVar0 + message_str(SCRIPT_OBJ_DUDE, 178));
                                                                                 end
                                                                                 else begin
-                                                                                    critter_mod_skill(dude_obj, 15, LVar1);
-                                                                                    display_msg(message_str(1, 175) + LVar1 + message_str(1, 178));
+                                                                                    critter_mod_skill(dude_obj, SKILL_BARTER, LVar1);
+                                                                                    display_msg(message_str(SCRIPT_OBJ_DUDE, 175) + LVar1 + message_str(SCRIPT_OBJ_DUDE, 178));
                                                                                 end
                                                                                 game_time_advance(60 * (60 * 10));
                                                                                 gfade_in(10);
                                                                             end
                                                                             else begin
-                                                                                display_msg(message_str(1, 203) + message_str(1, 220) + message_str(1, 204));
+                                                                                display_msg(message_str(SCRIPT_OBJ_DUDE, 203) + message_str(SCRIPT_OBJ_DUDE, 220) + message_str(SCRIPT_OBJ_DUDE, 204));
                                                                                 gfade_out(10);
                                                                                 game_time_advance(60 * (60 * 10));
                                                                                 gfade_in(10);
@@ -408,23 +408,23 @@ begin
                                                                         end
                                                                         else begin
                                                                             if (obj_pid(LVar4) == 794) then begin
-                                                                                if (has_skill(dude_obj, 16) < 300) then begin
+                                                                                if (has_skill(dude_obj, SKILL_GAMBLING) < 300) then begin
                                                                                     LVar0 := random(3, 6);
                                                                                     LVar1 := random(2, 4);
                                                                                     gfade_out(10);
-                                                                                    if (has_trait(0, dude_obj, 81)) then begin
-                                                                                        critter_mod_skill(dude_obj, 16, LVar0);
-                                                                                        display_msg(message_str(1, 176) + LVar0 + message_str(1, 178));
+                                                                                    if (has_trait(TRAIT_PERK, dude_obj, PERK_comprehension_perk)) then begin
+                                                                                        critter_mod_skill(dude_obj, SKILL_GAMBLING, LVar0);
+                                                                                        display_msg(message_str(SCRIPT_OBJ_DUDE, 176) + LVar0 + message_str(SCRIPT_OBJ_DUDE, 178));
                                                                                     end
                                                                                     else begin
-                                                                                        critter_mod_skill(dude_obj, 16, LVar1);
-                                                                                        display_msg(message_str(1, 176) + LVar1 + message_str(1, 178));
+                                                                                        critter_mod_skill(dude_obj, SKILL_GAMBLING, LVar1);
+                                                                                        display_msg(message_str(SCRIPT_OBJ_DUDE, 176) + LVar1 + message_str(SCRIPT_OBJ_DUDE, 178));
                                                                                     end
                                                                                     game_time_advance(60 * (60 * 10));
                                                                                     gfade_in(10);
                                                                                 end
                                                                                 else begin
-                                                                                    display_msg(message_str(1, 203) + message_str(1, 221) + message_str(1, 204));
+                                                                                    display_msg(message_str(SCRIPT_OBJ_DUDE, 203) + message_str(SCRIPT_OBJ_DUDE, 221) + message_str(SCRIPT_OBJ_DUDE, 204));
                                                                                     gfade_out(10);
                                                                                     game_time_advance(60 * (60 * 10));
                                                                                     gfade_in(10);
@@ -432,23 +432,23 @@ begin
                                                                             end
                                                                             else begin
                                                                                 if (obj_pid(LVar4) == 795) then begin
-                                                                                    if (has_skill(dude_obj, 17) < 300) then begin
+                                                                                    if (has_skill(dude_obj, SKILL_OUTDOORSMAN) < 300) then begin
                                                                                         LVar0 := random(3, 6);
                                                                                         LVar1 := random(2, 4);
                                                                                         gfade_out(10);
-                                                                                        if (has_trait(0, dude_obj, 81)) then begin
-                                                                                            critter_mod_skill(dude_obj, 17, LVar0);
-                                                                                            display_msg(message_str(1, 177) + LVar0 + message_str(1, 178));
+                                                                                        if (has_trait(TRAIT_PERK, dude_obj, PERK_comprehension_perk)) then begin
+                                                                                            critter_mod_skill(dude_obj, SKILL_OUTDOORSMAN, LVar0);
+                                                                                            display_msg(message_str(SCRIPT_OBJ_DUDE, 177) + LVar0 + message_str(SCRIPT_OBJ_DUDE, 178));
                                                                                         end
                                                                                         else begin
-                                                                                            critter_mod_skill(dude_obj, 17, LVar1);
-                                                                                            display_msg(message_str(1, 177) + LVar1 + message_str(1, 178));
+                                                                                            critter_mod_skill(dude_obj, SKILL_OUTDOORSMAN, LVar1);
+                                                                                            display_msg(message_str(SCRIPT_OBJ_DUDE, 177) + LVar1 + message_str(SCRIPT_OBJ_DUDE, 178));
                                                                                         end
                                                                                         game_time_advance(60 * (60 * 10));
                                                                                         gfade_in(10);
                                                                                     end
                                                                                     else begin
-                                                                                        display_msg(message_str(1, 203) + message_str(1, 222) + message_str(1, 204));
+                                                                                        display_msg(message_str(SCRIPT_OBJ_DUDE, 203) + message_str(SCRIPT_OBJ_DUDE, 222) + message_str(SCRIPT_OBJ_DUDE, 204));
                                                                                         gfade_out(10);
                                                                                         game_time_advance(60 * (60 * 10));
                                                                                         gfade_in(10);
@@ -476,50 +476,50 @@ begin
             if (obj_pid(LVar4) == 127) then begin
                 if (Validate(LVar2)) then begin
                     if ((critter_state(LVar2) bwand 1) or (critter_state(LVar2) bwand 2) or (critter_state(LVar2) bwand 4) or (critter_state(LVar2) bwand 8)) then begin
-                        if (get_critter_stat(LVar2, 34) == 0) then begin
+                        if (get_critter_stat(LVar2, STAT_gender) == 0) then begin
                             rm_obj_from_inven(LVar3, LVar4);
                             call Delete_Removed_Items(127);
-                            set_global_var(0, global_var(0) + -45);
-                            set_global_var(757, global_var(757) + 1);
-                            add_obj_to_inven(dude_obj, create_object_sid(557, 0, 0, -1));
-                            if ((global_var(757) == 2) or (global_var(757) == 3)) then begin
-                                critter_add_trait(dude_obj, 0, 100, 1);
+                            set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + -45);
+                            set_global_var(GVAR_ELDRIDGE_TASK, global_var(GVAR_ELDRIDGE_TASK) + 1);
+                            add_obj_to_inven(dude_obj, create_object(557, 0, 0));
+                            if ((global_var(GVAR_ELDRIDGE_TASK) == 2) or (global_var(GVAR_ELDRIDGE_TASK) == 3)) then begin
+                                critter_add_trait(dude_obj, TRAIT_PERK, PERK_pack_rat_perk, 1);
                             end
-                            if ((global_var(757) == 4) or (global_var(757) == 5)) then begin
-                                critter_add_trait(dude_obj, 0, 13, 1);
-                                critter_rm_trait(dude_obj, 0, 100, 1);
+                            if ((global_var(GVAR_ELDRIDGE_TASK) == 4) or (global_var(GVAR_ELDRIDGE_TASK) == 5)) then begin
+                                critter_add_trait(dude_obj, TRAIT_PERK, PERK_strong_back, 1);
+                                critter_rm_trait(dude_obj, TRAIT_PERK, PERK_pack_rat_perk, 1);
                             end
-                            if ((global_var(757) == 6) or (global_var(757) == 7)) then begin
-                                critter_add_trait(dude_obj, 0, 13, 1);
-                                critter_rm_trait(dude_obj, 0, 100, 1);
+                            if ((global_var(GVAR_ELDRIDGE_TASK) == 6) or (global_var(GVAR_ELDRIDGE_TASK) == 7)) then begin
+                                critter_add_trait(dude_obj, TRAIT_PERK, PERK_strong_back, 1);
+                                critter_rm_trait(dude_obj, TRAIT_PERK, PERK_pack_rat_perk, 1);
                             end
-                            if (global_var(757) > 7) then begin
-                                critter_add_trait(dude_obj, 0, 100, 1);
-                                critter_add_trait(dude_obj, 0, 13, 1);
+                            if (global_var(GVAR_ELDRIDGE_TASK) > 7) then begin
+                                critter_add_trait(dude_obj, TRAIT_PERK, PERK_pack_rat_perk, 1);
+                                critter_add_trait(dude_obj, TRAIT_PERK, PERK_strong_back, 1);
                             end
                             destroy_object(LVar2);
                         end
                         else begin
-                            if (get_critter_stat(LVar2, 34) == 1) then begin
+                            if (get_critter_stat(LVar2, STAT_gender) == 1) then begin
                                 rm_obj_from_inven(LVar3, LVar4);
                                 call Delete_Removed_Items(127);
-                                set_global_var(0, global_var(0) + -45);
-                                set_global_var(757, global_var(757) + 1);
-                                add_obj_to_inven(dude_obj, create_object_sid(556, 0, 0, -1));
-                                if ((global_var(757) == 2) or (global_var(757) == 3)) then begin
-                                    critter_add_trait(dude_obj, 0, 100, 1);
+                                set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + -45);
+                                set_global_var(GVAR_ELDRIDGE_TASK, global_var(GVAR_ELDRIDGE_TASK) + 1);
+                                add_obj_to_inven(dude_obj, create_object(556, 0, 0));
+                                if ((global_var(GVAR_ELDRIDGE_TASK) == 2) or (global_var(GVAR_ELDRIDGE_TASK) == 3)) then begin
+                                    critter_add_trait(dude_obj, TRAIT_PERK, PERK_pack_rat_perk, 1);
                                 end
-                                if ((global_var(757) == 4) or (global_var(757) == 5)) then begin
-                                    critter_add_trait(dude_obj, 0, 13, 1);
-                                    critter_rm_trait(dude_obj, 0, 100, 1);
+                                if ((global_var(GVAR_ELDRIDGE_TASK) == 4) or (global_var(GVAR_ELDRIDGE_TASK) == 5)) then begin
+                                    critter_add_trait(dude_obj, TRAIT_PERK, PERK_strong_back, 1);
+                                    critter_rm_trait(dude_obj, TRAIT_PERK, PERK_pack_rat_perk, 1);
                                 end
-                                if ((global_var(757) == 6) or (global_var(757) == 7)) then begin
-                                    critter_add_trait(dude_obj, 0, 13, 1);
-                                    critter_rm_trait(dude_obj, 0, 100, 1);
+                                if ((global_var(GVAR_ELDRIDGE_TASK) == 6) or (global_var(GVAR_ELDRIDGE_TASK) == 7)) then begin
+                                    critter_add_trait(dude_obj, TRAIT_PERK, PERK_strong_back, 1);
+                                    critter_rm_trait(dude_obj, TRAIT_PERK, PERK_pack_rat_perk, 1);
                                 end
-                                if (global_var(757) > 7) then begin
-                                    critter_add_trait(dude_obj, 0, 100, 1);
-                                    critter_add_trait(dude_obj, 0, 13, 1);
+                                if (global_var(GVAR_ELDRIDGE_TASK) > 7) then begin
+                                    critter_add_trait(dude_obj, TRAIT_PERK, PERK_pack_rat_perk, 1);
+                                    critter_add_trait(dude_obj, TRAIT_PERK, PERK_strong_back, 1);
                                 end
                                 destroy_object(LVar2);
                             end
@@ -533,7 +533,7 @@ begin
             if (obj_pid(LVar4) == 144) then begin
                 if ((obj_pid(LVar2) != 16777376) and (obj_pid(LVar2) != 16777377) and (obj_pid(LVar2) != 16777305) and (obj_pid(LVar2) != 16777313) and (obj_pid(LVar2) != 16777323) and (obj_pid(LVar2) != 16777352) and (obj_pid(LVar2) != 16777368) and (obj_pid(LVar2) != 16777291) and (obj_pid(LVar2) != 16777278) and (obj_pid(LVar2) != 16777379) and (obj_pid(LVar2) != 16777380) and (obj_pid(LVar2) != 16777859) and (obj_pid(LVar2) != 16777558) and (obj_pid(LVar2) != 16777687) and (obj_pid(LVar2) != 16777766) and (obj_pid(LVar2) != 16777762) and (obj_pid(LVar2) != 16777785) and (LVar2 != dude_obj)) then begin
                     if (((obj_pid(LVar2) - 16777216) >= 0) and ((obj_pid(LVar2) - 33554431) < 0)) then begin
-                        if ((get_critter_stat(LVar2, 35) < get_critter_stat(LVar2, 7)) <= 0) then begin
+                        if ((get_critter_stat(LVar2, STAT_current_hp) < get_critter_stat(LVar2, STAT_max_hp)) <= 0) then begin
                             if (obj_is_carrying_obj_pid(dude_obj, 144) > 0) then begin
                                 while (obj_is_carrying_obj_pid(dude_obj, 144) > 0) do begin
                                     LVar5 := obj_carrying_pid_obj(dude_obj, 144);
@@ -588,4 +588,3 @@ begin
     return 0;
     arg0 := 0;
 end
-
