diff --git a/Mods/MegaMod/Original/HTCORPSE.ssl b/Mods/MegaMod/ReDefine/HTCORPSE.ssl
index 411ea22..5ae0b16 100644
--- a/Mods/MegaMod/Original/HTCORPSE.ssl
+++ b/Mods/MegaMod/ReDefine/HTCORPSE.ssl
@@ -4,22 +4,21 @@ procedure start;
 
 procedure spatial_p_proc
 begin
-    if ((global_var(305) == 5) or (global_var(305) == 6) and (source_obj == dude_obj) and (local_var(0) == 0)) then begin
+    if ((global_var(GVAR_BH_MISSING) == 5) or (global_var(GVAR_BH_MISSING) == 6) and (source_obj == dude_obj) and (local_var(0) == 0)) then begin
         set_local_var(0, local_var(0) + 1);
-        if (global_var(305) == 6) then begin
-            set_global_var(305, 8);
+        if (global_var(GVAR_BH_MISSING) == 6) then begin
+            set_global_var(GVAR_BH_MISSING, 8);
         end
         else begin
-            set_global_var(305, 7);
+            set_global_var(GVAR_BH_MISSING, 7);
         end
-        display_msg(message_str(877, 100));
+        display_mstr(100);
         give_exp_points(500);
-        display_msg(message_str(14, 100) + 500 + message_str(14, 101));
-        set_global_var(54, global_var(54) + 5);
+        display_msg(g_mstr(100) + 500 + g_mstr(101));
+        set_global_var(GVAR_TOWN_REP_BROKEN_HILLS, global_var(GVAR_TOWN_REP_BROKEN_HILLS) + 5);
     end
 end
 
 procedure start
 begin
 end
-
