diff --git a/Mods/MegaMod/Original/IIPIT.ssl b/Mods/MegaMod/ReDefine/IIPIT.ssl
index fd76197..c59887f 100644
--- a/Mods/MegaMod/Original/IIPIT.ssl
+++ b/Mods/MegaMod/ReDefine/IIPIT.ssl
@@ -144,7 +144,7 @@ begin
     variable LVar2 := 0;
     variable LVar3 := 0;
     if (source_obj == dude_obj) then begin
-        LVar0 := get_critter_stat(source_obj, 1) + -4;
+        LVar0 := get_critter_stat(source_obj, STAT_pe) + -4;
         LVar1 := LVar0 - (tile_distance_objs(self_obj, source_obj) * 2);
         if ((tile_distance_objs(self_obj, source_obj) <= LVar0) and (local_var(0) == 0)) then begin
             if (LVar1 > -4) then begin
@@ -154,24 +154,24 @@ begin
             if (is_success(LVar2)) then begin
                 set_local_var(0, 1);
                 reg_anim_func(2, source_obj);
-                LVar3 := create_object_sid(33556185, tile_num(self_obj), elevation(self_obj), 1101);
+                LVar3 := create_object_sid(33556185, tile_num(self_obj), elevation(self_obj), NAME);
                 if (is_critical(LVar2)) then begin
                     if (source_obj == dude_obj) then begin
-                        display_msg(message_str(1101, 101));
+                        display_mstr(101);
                     end
                     else begin
                         if ((proto_data(obj_pid(source_obj), 11) == 0) and obj_can_see_obj(source_obj, dude_obj)) then begin
-                            display_msg(obj_name(source_obj) + message_str(1101, 102));
+                            display_msg(obj_name(source_obj) + mstr(102));
                         end
                     end
                 end
                 else begin
                     if (source_obj == dude_obj) then begin
-                        display_msg(message_str(1101, 101));
+                        display_mstr(101);
                     end
                     else begin
                         if ((proto_data(obj_pid(source_obj), 11) == 0) and obj_can_see_obj(source_obj, dude_obj)) then begin
-                            display_msg(obj_name(source_obj) + message_str(1101, 102));
+                            display_msg(obj_name(source_obj) + mstr(102));
                         end
                     end
                 end
@@ -185,8 +185,8 @@ begin
                 destroy_object(tile_contains_pid_obj(tile_num(self_obj), elevation(self_obj), 33556185));
                 debug_msg("Destroyed Visible Trap");
             end
-            LVar3 := create_object_sid(33556185, tile_num(self_obj), elevation(self_obj), -1);
-            display_msg(message_str(1101, 100));
+            LVar3 := create_object(33556185, tile_num(self_obj), elevation(self_obj));
+            display_mstr(100);
             critter_dmg(source_obj, random(2, 10), 0 bwor 256 bwor 512);
             set_local_var(0, 1);
             call DumpPoorBastard();
@@ -197,7 +197,7 @@ end
 procedure look_at_p_proc
 begin
     script_overrides;
-    display_msg(message_str(1101, 103));
+    display_mstr(103);
     debug_msg("Looking At the Trap");
 end
 
@@ -212,14 +212,14 @@ begin
     end
     if (is_critical(local_var(3))) then begin
         if (is_success(local_var(3))) then begin
-            display_msg(message_str(1101, 101));
+            display_mstr(101);
         end
         else begin
-            display_msg(message_str(1101, 101));
+            display_mstr(101);
         end
     end
     else begin
-        display_msg(message_str(1101, 101));
+        display_mstr(101);
     end
 end
 
@@ -279,4 +279,3 @@ begin
     end
     gfade_in(10);
 end
-
