diff --git a/Mods/MegaMod/Original/JCKMKN.ssl b/Mods/MegaMod/ReDefine/JCKMKN.ssl
index 1d5dd82..f48f9a7 100644
--- a/Mods/MegaMod/Original/JCKMKN.ssl
+++ b/Mods/MegaMod/ReDefine/JCKMKN.ssl
@@ -165,20 +165,20 @@ begin
             animate_move_obj_to_tile(self_obj, dest_tile, 0);
         end
         if (fixed_param == 101) then begin
-            if (global_var(687) < 7) then begin
-                float_msg(self_obj, message_str(1319, random(102, 106)), 8);
+            if (global_var(GVAR_ENEMY_JENFARM) < 7) then begin
+                float_msg(self_obj, mstr(random(102, 106)), FLOAT_MSG_YELLOW);
             end
             else begin
-                if (not((global_var(445) bwand 1) != 0)) then begin
+                if (not((global_var(GVAR_DEN_FLAG_1) bwand 1) != 0)) then begin
                     if (map_var(3) != 0) then begin
-                        float_msg(self_obj, message_str(1319, random(120, 123)), 8);
+                        float_msg(self_obj, mstr(random(120, 123)), FLOAT_MSG_YELLOW);
                     end
                     else begin
-                        float_msg(self_obj, message_str(1319, random(122, 125)), 8);
+                        float_msg(self_obj, mstr(random(122, 125)), FLOAT_MSG_YELLOW);
                     end
                 end
                 else begin
-                    float_msg(self_obj, message_str(1319, random(125, 128)), 8);
+                    float_msg(self_obj, mstr(random(125, 128)), FLOAT_MSG_YELLOW);
                 end
             end
         end
@@ -190,7 +190,7 @@ procedure critter_p_proc
 begin
     if ((local_var(4) == 2) and obj_can_see_obj(self_obj, dude_obj)) then begin
         set_local_var(4, 1);
-        attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+        attack(dude_obj);
     end
     if (in_timed_event == 0) then begin
         in_timed_event := 1;
@@ -202,59 +202,59 @@ procedure destroy_p_proc
 begin
     if (source_obj == dude_obj) then begin
         if (source_obj == dude_obj) then begin
-            set_global_var(5, global_var(5) + 1);
-            if (((global_var(4) + global_var(5)) >= 25) and ((global_var(4) > (3 * global_var(5))) or (global_var(2) == 1)) and (global_var(1) == 0)) then begin
-                set_global_var(2, 1);
-                set_global_var(3, 0);
+            set_global_var(GVAR_GOOD_MONSTER, global_var(GVAR_GOOD_MONSTER) + 1);
+            if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(4) > (3 * global_var(GVAR_GOOD_MONSTER))) or (global_var(GVAR_CHAMPION_REPUTATION) == 1)) and (global_var(GVAR_CHILDKILLER_REPUTATION) == 0)) then begin
+                set_global_var(GVAR_CHAMPION_REPUTATION, 1);
+                set_global_var(GVAR_BERSERKER_REPUTATION, 0);
             end
-            if (metarule(51, self_obj) == 2) then begin
+            if (critter_kill_type(self_obj) == KILL_TYPE_children_kills) then begin
                 if (source_obj == dude_obj) then begin
-                    set_global_var(2, 0);
-                    set_global_var(1, global_var(1) + 1);
-                    set_global_var(0, global_var(0) + -15);
+                    set_global_var(GVAR_CHAMPION_REPUTATION, 0);
+                    set_global_var(GVAR_CHILDKILLER_REPUTATION, global_var(GVAR_CHILDKILLER_REPUTATION) + 1);
+                    set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + -15);
                     debug_msg("Player gains " + -15 + " Karma Points.");
-                    set_global_var(37, 0);
-                    set_global_var(38, 0);
-                    set_global_var(39, 0);
-                    set_global_var(40, 0);
-                    set_global_var(41, 0);
-                    set_global_var(42, 0);
-                    set_global_var(43, 0);
-                    set_global_var(44, 0);
-                    set_global_var(45, 0);
-                    if (global_var(0) >= 1000) then begin
-                        set_global_var(37, 1);
+                    set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+                    set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+                    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+                    set_global_var(GVAR_KARMA_DEFENDER, 0);
+                    set_global_var(GVAR_KARMA_WANDERER, 0);
+                    set_global_var(GVAR_KARMA_BETRAYER, 0);
+                    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+                    set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+                    set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+                        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
                     end
                     else begin
-                        if (global_var(0) >= 750) then begin
-                            set_global_var(38, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                            set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
                         end
                         else begin
-                            if (global_var(0) >= 500) then begin
-                                set_global_var(39, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                                set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                             end
                             else begin
-                                if (global_var(0) >= 250) then begin
-                                    set_global_var(40, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                                    set_global_var(GVAR_KARMA_DEFENDER, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -250) then begin
-                                        set_global_var(41, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                                        set_global_var(GVAR_KARMA_WANDERER, 1);
                                     end
                                     else begin
-                                        if (global_var(0) > -500) then begin
-                                            set_global_var(42, 1);
+                                        if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                            set_global_var(GVAR_KARMA_BETRAYER, 1);
                                         end
                                         else begin
-                                            if (global_var(0) > -750) then begin
-                                                set_global_var(43, 1);
+                                            if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                                set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                             end
                                             else begin
-                                                if (global_var(0) > -1000) then begin
-                                                    set_global_var(44, 1);
+                                                if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                                    set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                                 end
                                                 else begin
-                                                    set_global_var(45, 1);
+                                                    set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                                 end
                                             end
                                         end
@@ -263,58 +263,58 @@ begin
                             end
                         end
                     end
-                    set_global_var(63, global_var(63) + -8);
+                    set_global_var(GVAR_TOWN_REP_ABBEY, global_var(GVAR_TOWN_REP_ABBEY) + -8);
                     debug_msg("Added " + -8 + " to Town Rep");
                 end
             end
-            if (((global_var(4) + global_var(5)) >= 25) and ((global_var(5) > (2 * global_var(4))) or (global_var(3) == 1))) then begin
-                set_global_var(3, 1);
-                set_global_var(2, 0);
+            if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
+                set_global_var(GVAR_BERSERKER_REPUTATION, 1);
+                set_global_var(GVAR_CHAMPION_REPUTATION, 0);
             end
-            set_global_var(0, global_var(0) + -10);
+            set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + -10);
             debug_msg("Player gains " + -10 + " Karma Points.");
-            set_global_var(37, 0);
-            set_global_var(38, 0);
-            set_global_var(39, 0);
-            set_global_var(40, 0);
-            set_global_var(41, 0);
-            set_global_var(42, 0);
-            set_global_var(43, 0);
-            set_global_var(44, 0);
-            set_global_var(45, 0);
-            if (global_var(0) >= 1000) then begin
-                set_global_var(37, 1);
+            set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+            set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+            set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+            set_global_var(GVAR_KARMA_DEFENDER, 0);
+            set_global_var(GVAR_KARMA_WANDERER, 0);
+            set_global_var(GVAR_KARMA_BETRAYER, 0);
+            set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+            set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+            set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+                set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
             end
             else begin
-                if (global_var(0) >= 750) then begin
-                    set_global_var(38, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                    set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
                 end
                 else begin
-                    if (global_var(0) >= 500) then begin
-                        set_global_var(39, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                        set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                     end
                     else begin
-                        if (global_var(0) >= 250) then begin
-                            set_global_var(40, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                            set_global_var(GVAR_KARMA_DEFENDER, 1);
                         end
                         else begin
-                            if (global_var(0) > -250) then begin
-                                set_global_var(41, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                                set_global_var(GVAR_KARMA_WANDERER, 1);
                             end
                             else begin
-                                if (global_var(0) > -500) then begin
-                                    set_global_var(42, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                    set_global_var(GVAR_KARMA_BETRAYER, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -750) then begin
-                                        set_global_var(43, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                        set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                     end
                                     else begin
-                                        if (global_var(0) > -1000) then begin
-                                            set_global_var(44, 1);
+                                        if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                            set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                         end
                                         else begin
-                                            set_global_var(45, 1);
+                                            set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                         end
                                     end
                                 end
@@ -323,65 +323,65 @@ begin
                     end
                 end
             end
-            set_global_var(63, global_var(63) + -5);
+            set_global_var(GVAR_TOWN_REP_ABBEY, global_var(GVAR_TOWN_REP_ABBEY) + -5);
             debug_msg("Added " + -5 + " to Town Rep");
         end
-        set_global_var(689, 1);
+        set_global_var(GVAR_KCNED_DEAD, 1);
     end
 end
 
 procedure look_at_p_proc
 begin
     script_overrides;
-    display_msg(message_str(1319, 100));
+    display_mstr(100);
 end
 
 procedure description_p_proc
 begin
     script_overrides;
-    display_msg(message_str(1319, 101));
+    display_mstr(101);
 end
 
 procedure damage_p_proc
 begin
     if (source_obj == dude_obj) then begin
-        set_global_var(689, 1);
+        set_global_var(GVAR_KCNED_DEAD, 1);
     end
 end
 
 procedure map_enter_p_proc
 begin
-    if (not(metarule(22, 0))) then begin
-        if (metarule(14, 0)) then begin
+    if (not(is_loading_game)) then begin
+        if (map_first_run) then begin
             set_local_var(5, tile_num(self_obj));
         end
         else begin
-            if (((game_time - map_var(1)) >= (3 * (24 * (60 * (60 * 10))))) and ((global_var(687) >= 7) and not((global_var(445) bwand 1) != 0) or (global_var(687) < 4))) then begin
+            if (((game_time - map_var(1)) >= (3 * (24 * (60 * (60 * 10))))) and ((global_var(GVAR_ENEMY_JENFARM) >= 7) and not((global_var(GVAR_DEN_FLAG_1) bwand 1) != 0) or (global_var(GVAR_ENEMY_JENFARM) < 4))) then begin
                 destroy_object(self_obj);
             end
         end
-        critter_add_trait(self_obj, 1, 6, 211);
-        critter_add_trait(self_obj, 1, 5, 33);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 211);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_COWARD);
     end
 end
 
 procedure talk_p_proc
 begin
     in_dialog := 1;
-    if ((global_var(687) == 1) or (global_var(687) == 4)) then begin
-        float_msg(self_obj, message_str(1319, 107), 8);
+    if ((global_var(GVAR_ENEMY_JENFARM) == 1) or (global_var(GVAR_ENEMY_JENFARM) == 4)) then begin
+        float_msg(self_obj, mstr(107), FLOAT_MSG_YELLOW);
     end
     else begin
-        if ((global_var(687) == 2) or (global_var(687) == 3)) then begin
-            float_msg(self_obj, message_str(1319, 108), 8);
+        if ((global_var(GVAR_ENEMY_JENFARM) == 2) or (global_var(GVAR_ENEMY_JENFARM) == 3)) then begin
+            float_msg(self_obj, mstr(108), FLOAT_MSG_YELLOW);
         end
         else begin
-            if ((global_var(687) == 7) or (global_var(687) == 10)) then begin
-                float_msg(self_obj, message_str(1319, 109), 9);
+            if ((global_var(GVAR_ENEMY_JENFARM) == 7) or (global_var(GVAR_ENEMY_JENFARM) == 10)) then begin
+                float_msg(self_obj, mstr(109), FLOAT_MSG_WHITE);
             end
             else begin
-                if (global_var(687) == 9) then begin
-                    float_msg(self_obj, message_str(1319, 110), 12);
+                if (global_var(GVAR_ENEMY_JENFARM) == 9) then begin
+                    float_msg(self_obj, mstr(110), FLOAT_MSG_LIGHT_GREY);
                 end
             end
         end
@@ -406,4 +406,3 @@ end
 procedure push_p_proc
 begin
 end
-
