diff --git a/Mods/MegaMod/Original/KCCVRAT.ssl b/Mods/MegaMod/ReDefine/KCCVRAT.ssl
index 0e46279..3eecfa1 100644
--- a/Mods/MegaMod/Original/KCCVRAT.ssl
+++ b/Mods/MegaMod/ReDefine/KCCVRAT.ssl
@@ -146,7 +146,7 @@ end
 procedure critter_p_proc
 begin
     if (obj_can_see_obj(self_obj, dude_obj)) then begin
-        attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+        attack(dude_obj);
     end
 end
 
@@ -172,18 +172,18 @@ end
 procedure damage_p_proc
 begin
     if (source_obj == dude_obj) then begin
-        attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+        attack(dude_obj);
     end
 end
 
 procedure map_enter_p_proc
 begin
-    critter_add_trait(self_obj, 1, 6, 3);
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 3);
     if ((obj_pid(self_obj) >= 16777332) and (obj_pid(self_obj) <= 16777334)) then begin
-        critter_add_trait(self_obj, 1, 5, 48);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_MUTATED_RAT);
     end
     else begin
-        critter_add_trait(self_obj, 1, 5, 7);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_RAT);
     end
     add_timer_event(self_obj, game_ticks(random(1, 5)), 0);
 end
@@ -210,4 +210,3 @@ end
 procedure use_obj_on_p_proc
 begin
 end
-
