diff --git a/Mods/MegaMod/Original/KCDUNTON.ssl b/Mods/MegaMod/ReDefine/KCDUNTON.ssl
index 6228618..9b88201 100644
--- a/Mods/MegaMod/Original/KCDUNTON.ssl
+++ b/Mods/MegaMod/ReDefine/KCDUNTON.ssl
@@ -282,17 +282,17 @@ end
 procedure critter_p_proc
 begin
     if (active) then begin
-        if ((global_var(204) == 1) and obj_can_see_obj(self_obj, dude_obj)) then begin
-            attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+        if ((global_var(GVAR_KLAMATH_REP_ALIANCE) == 1) and obj_can_see_obj(self_obj, dude_obj)) then begin
+            attack(dude_obj);
         end
         else begin
-            if (cur_map_index == 9) then begin
+            if (CUR_MAP_KLADWTWN) then begin
                 call checkarea();
                 if ((area == 2) and (map_var(17) > 0) and obj_can_see_obj(self_obj, dude_obj)) then begin
                     set_map_var(17, 0);
-                    float_msg(self_obj, message_str(87, 600), 2);
-                    set_global_var(204, 1);
-                    attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+                    float_msg(self_obj, message_str(SCRIPT_KCCDNTN, 600), FLOAT_MSG_RED);
+                    set_global_var(GVAR_KLAMATH_REP_ALIANCE, 1);
+                    attack(dude_obj);
                 end
                 else begin
                     needtomove := 0;
@@ -311,8 +311,8 @@ begin
                 end
             end
             else begin
-                if ((global_var(76) == 1) or (global_var(75) == 1) or ((map_var(5) >= map_var(19)) and (global_var(77) != 1))) then begin
-                    set_global_var(184, 1);
+                if ((global_var(GVAR_RUSTLE_FAIL) == 1) or (global_var(GVAR_RUSTLE_FAIL_VIOLENT) == 1) or ((map_var(5) >= map_var(19)) and (global_var(GVAR_RUSTLE_SUCCESS) != 1))) then begin
+                    set_global_var(GVAR_RUSTLE_OVER, 1);
                     dest := tile_num_in_direction(tile_num(self_obj), 5, 12);
                     animate_move_obj_to_tile(self_obj, dest, 1);
                     add_timer_event(self_obj, game_ticks(2), 1);
@@ -345,7 +345,7 @@ procedure pickup_p_proc
 begin
     if (active) then begin
         if (source_obj == dude_obj) then begin
-            set_global_var(204, 1);
+            set_global_var(GVAR_KLAMATH_REP_ALIANCE, 1);
         end
     end
 end
@@ -356,7 +356,7 @@ begin
         script_overrides;
         if (local_var(2) == 0) then begin
             set_local_var(2, 1);
-            set_local_var(3, (get_critter_stat(dude_obj, 3) - get_critter_stat(self_obj, 3)) * 5);
+            set_local_var(3, (dude_charisma - self_charisma) * 5);
             debug_msg("Base Reaction == " + local_var(3));
         end
         Static_Reaction := 0;
@@ -368,14 +368,14 @@ begin
         else begin
             reaction_bonus_karma := -1 * (-1 * reaction_bonus_karma / 1000);
         end
-        if (global_var(48) >= 0) then begin
-            reaction_bonus_town_rep := global_var(48) / 2;
+        if (global_var(GVAR_TOWN_REP_KLAMATH) >= 0) then begin
+            reaction_bonus_town_rep := global_var(GVAR_TOWN_REP_KLAMATH) / 2;
         end
         else begin
-            reaction_bonus_town_rep := -1 * (-1 * global_var(48) / 2);
+            reaction_bonus_town_rep := -1 * (-1 * global_var(GVAR_TOWN_REP_KLAMATH) / 2);
         end
-        if (has_trait(2, dude_obj, 13)) then begin
-            if (get_critter_stat(dude_obj, 34) == get_critter_stat(self_obj, 34)) then begin
+        if (has_trait(TRAIT_TRAIT, dude_obj, TRAIT_sex_appeal)) then begin
+            if (dude_gender == self_gender) then begin
                 Static_Reaction := Static_Reaction + -20;
             end
             else begin
@@ -387,58 +387,58 @@ begin
             debug_msg("Sex Appeal Bonus == 0");
         end
         if (Evil_Critter == 1) then begin
-            Static_Reaction := Static_Reaction - reaction_bonus_karma + (reaction_bonus_karma * has_trait(0, dude_obj, 39));
-            if ((reaction_bonus_karma < 0) and (has_trait(0, dude_obj, 39) != 0)) then begin
+            Static_Reaction := Static_Reaction - reaction_bonus_karma + (reaction_bonus_karma * has_trait(TRAIT_PERK, dude_obj, PERK_cult_of_personality));
+            if ((reaction_bonus_karma < 0) and (has_trait(TRAIT_PERK, dude_obj, PERK_cult_of_personality) != 0)) then begin
                 Static_Reaction := Static_Reaction - reaction_bonus_karma;
             end
             debug_msg("Karma Reaction Bonus == " + Static_Reaction);
-            Static_Reaction := Static_Reaction - reaction_bonus_town_rep + (reaction_bonus_town_rep * has_trait(0, dude_obj, 39));
-            if ((reaction_bonus_town_rep < 0) and (has_trait(0, dude_obj, 39) != 0)) then begin
+            Static_Reaction := Static_Reaction - reaction_bonus_town_rep + (reaction_bonus_town_rep * has_trait(TRAIT_PERK, dude_obj, PERK_cult_of_personality));
+            if ((reaction_bonus_town_rep < 0) and (has_trait(TRAIT_PERK, dude_obj, PERK_cult_of_personality) != 0)) then begin
                 Static_Reaction := Static_Reaction - reaction_bonus_town_rep;
             end
             debug_msg("Town Rep Bonus == " + Static_Reaction);
-            set_global_var(37, 0);
-            set_global_var(38, 0);
-            set_global_var(39, 0);
-            set_global_var(40, 0);
-            set_global_var(41, 0);
-            set_global_var(42, 0);
-            set_global_var(43, 0);
-            set_global_var(44, 0);
-            set_global_var(45, 0);
-            if (global_var(0) >= 1000) then begin
-                set_global_var(37, 1);
+            set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+            set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+            set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+            set_global_var(GVAR_KARMA_DEFENDER, 0);
+            set_global_var(GVAR_KARMA_WANDERER, 0);
+            set_global_var(GVAR_KARMA_BETRAYER, 0);
+            set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+            set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+            set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+                set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
             end
             else begin
-                if (global_var(0) >= 750) then begin
-                    set_global_var(38, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                    set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
                 end
                 else begin
-                    if (global_var(0) >= 500) then begin
-                        set_global_var(39, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                        set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                     end
                     else begin
-                        if (global_var(0) >= 250) then begin
-                            set_global_var(40, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                            set_global_var(GVAR_KARMA_DEFENDER, 1);
                         end
                         else begin
-                            if (global_var(0) > -250) then begin
-                                set_global_var(41, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                                set_global_var(GVAR_KARMA_WANDERER, 1);
                             end
                             else begin
-                                if (global_var(0) > -500) then begin
-                                    set_global_var(42, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                    set_global_var(GVAR_KARMA_BETRAYER, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -750) then begin
-                                        set_global_var(43, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                        set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                     end
                                     else begin
-                                        if (global_var(0) > -1000) then begin
-                                            set_global_var(44, 1);
+                                        if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                            set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                         end
                                         else begin
-                                            set_global_var(45, 1);
+                                            set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                         end
                                     end
                                 end
@@ -447,35 +447,35 @@ begin
                     end
                 end
             end
-            if (global_var(37) == 1) then begin
-                Static_Reaction := Static_Reaction - 20 + (40 * has_trait(0, dude_obj, 39));
+            if (global_var(GVAR_KARMA_HOLY_WARRIOR) == 1) then begin
+                Static_Reaction := Static_Reaction - 20 + (40 * has_trait(TRAIT_PERK, dude_obj, PERK_cult_of_personality));
             end
             else begin
-                if (global_var(38) == 1) then begin
-                    Static_Reaction := Static_Reaction - 15 + (30 * has_trait(0, dude_obj, 39));
+                if (global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES) == 1) then begin
+                    Static_Reaction := Static_Reaction - 15 + (30 * has_trait(TRAIT_PERK, dude_obj, PERK_cult_of_personality));
                 end
                 else begin
-                    if (global_var(39) == 1) then begin
-                        Static_Reaction := Static_Reaction - 10 + (20 * has_trait(0, dude_obj, 39));
+                    if (global_var(GVAR_KARMA_SHIELD_OF_HOPE) == 1) then begin
+                        Static_Reaction := Static_Reaction - 10 + (20 * has_trait(TRAIT_PERK, dude_obj, PERK_cult_of_personality));
                     end
                     else begin
-                        if (global_var(40) == 1) then begin
-                            Static_Reaction := Static_Reaction - 5 + (10 * has_trait(0, dude_obj, 39));
+                        if (global_var(GVAR_KARMA_DEFENDER) == 1) then begin
+                            Static_Reaction := Static_Reaction - 5 + (10 * has_trait(TRAIT_PERK, dude_obj, PERK_cult_of_personality));
                         end
                         else begin
-                            if (global_var(41) == 1) then begin
+                            if (global_var(GVAR_KARMA_WANDERER) == 1) then begin
                                 Static_Reaction := Static_Reaction - 0;
                             end
                             else begin
-                                if (global_var(42) == 1) then begin
+                                if (global_var(GVAR_KARMA_BETRAYER) == 1) then begin
                                     Static_Reaction := Static_Reaction - -5;
                                 end
                                 else begin
-                                    if (global_var(43) == 1) then begin
+                                    if (global_var(GVAR_KARMA_SWORD_OF_DESPAIR) == 1) then begin
                                         Static_Reaction := Static_Reaction - -10;
                                     end
                                     else begin
-                                        if (global_var(44) == 1) then begin
+                                        if (global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE) == 1) then begin
                                             Static_Reaction := Static_Reaction - -15;
                                         end
                                         else begin
@@ -491,58 +491,58 @@ begin
             debug_msg("Bad Person Reaction Bonus == " + Static_Reaction);
         end
         else begin
-            Static_Reaction := Static_Reaction + reaction_bonus_karma - (reaction_bonus_karma * has_trait(0, dude_obj, 39));
-            if ((reaction_bonus_karma > 0) and (has_trait(0, dude_obj, 39) != 0)) then begin
+            Static_Reaction := Static_Reaction + reaction_bonus_karma - (reaction_bonus_karma * has_trait(TRAIT_PERK, dude_obj, PERK_cult_of_personality));
+            if ((reaction_bonus_karma > 0) and (has_trait(TRAIT_PERK, dude_obj, PERK_cult_of_personality) != 0)) then begin
                 Static_Reaction := Static_Reaction + reaction_bonus_karma;
             end
             debug_msg("Karma Reaction Bonus == " + Static_Reaction);
-            Static_Reaction := Static_Reaction + reaction_bonus_town_rep - (reaction_bonus_town_rep * has_trait(0, dude_obj, 39));
-            if ((reaction_bonus_town_rep > 0) and (has_trait(0, dude_obj, 39) != 0)) then begin
+            Static_Reaction := Static_Reaction + reaction_bonus_town_rep - (reaction_bonus_town_rep * has_trait(TRAIT_PERK, dude_obj, PERK_cult_of_personality));
+            if ((reaction_bonus_town_rep > 0) and (has_trait(TRAIT_PERK, dude_obj, PERK_cult_of_personality) != 0)) then begin
                 Static_Reaction := Static_Reaction + reaction_bonus_town_rep;
             end
             debug_msg("Town Rep Bonus == " + Static_Reaction);
-            set_global_var(37, 0);
-            set_global_var(38, 0);
-            set_global_var(39, 0);
-            set_global_var(40, 0);
-            set_global_var(41, 0);
-            set_global_var(42, 0);
-            set_global_var(43, 0);
-            set_global_var(44, 0);
-            set_global_var(45, 0);
-            if (global_var(0) >= 1000) then begin
-                set_global_var(37, 1);
+            set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+            set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+            set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+            set_global_var(GVAR_KARMA_DEFENDER, 0);
+            set_global_var(GVAR_KARMA_WANDERER, 0);
+            set_global_var(GVAR_KARMA_BETRAYER, 0);
+            set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+            set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+            set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+                set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
             end
             else begin
-                if (global_var(0) >= 750) then begin
-                    set_global_var(38, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                    set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
                 end
                 else begin
-                    if (global_var(0) >= 500) then begin
-                        set_global_var(39, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                        set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                     end
                     else begin
-                        if (global_var(0) >= 250) then begin
-                            set_global_var(40, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                            set_global_var(GVAR_KARMA_DEFENDER, 1);
                         end
                         else begin
-                            if (global_var(0) > -250) then begin
-                                set_global_var(41, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                                set_global_var(GVAR_KARMA_WANDERER, 1);
                             end
                             else begin
-                                if (global_var(0) > -500) then begin
-                                    set_global_var(42, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                    set_global_var(GVAR_KARMA_BETRAYER, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -750) then begin
-                                        set_global_var(43, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                        set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                     end
                                     else begin
-                                        if (global_var(0) > -1000) then begin
-                                            set_global_var(44, 1);
+                                        if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                            set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                         end
                                         else begin
-                                            set_global_var(45, 1);
+                                            set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                         end
                                     end
                                 end
@@ -551,39 +551,39 @@ begin
                     end
                 end
             end
-            if (global_var(37) == 1) then begin
+            if (global_var(GVAR_KARMA_HOLY_WARRIOR) == 1) then begin
                 Static_Reaction := Static_Reaction + 20;
             end
             else begin
-                if (global_var(38) == 1) then begin
+                if (global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES) == 1) then begin
                     Static_Reaction := Static_Reaction + 15;
                 end
                 else begin
-                    if (global_var(39) == 1) then begin
+                    if (global_var(GVAR_KARMA_SHIELD_OF_HOPE) == 1) then begin
                         Static_Reaction := Static_Reaction + 10;
                     end
                     else begin
-                        if (global_var(40) == 1) then begin
+                        if (global_var(GVAR_KARMA_DEFENDER) == 1) then begin
                             Static_Reaction := Static_Reaction + 5;
                         end
                         else begin
-                            if (global_var(41) == 1) then begin
+                            if (global_var(GVAR_KARMA_WANDERER) == 1) then begin
                                 Static_Reaction := Static_Reaction + 0;
                             end
                             else begin
-                                if (global_var(42) == 1) then begin
-                                    Static_Reaction := Static_Reaction + -5 + (10 * has_trait(0, dude_obj, 39));
+                                if (global_var(GVAR_KARMA_BETRAYER) == 1) then begin
+                                    Static_Reaction := Static_Reaction + -5 + (10 * has_trait(TRAIT_PERK, dude_obj, PERK_cult_of_personality));
                                 end
                                 else begin
-                                    if (global_var(43) == 1) then begin
-                                        Static_Reaction := Static_Reaction + -10 + (20 * has_trait(0, dude_obj, 39));
+                                    if (global_var(GVAR_KARMA_SWORD_OF_DESPAIR) == 1) then begin
+                                        Static_Reaction := Static_Reaction + -10 + (20 * has_trait(TRAIT_PERK, dude_obj, PERK_cult_of_personality));
                                     end
                                     else begin
-                                        if (global_var(44) == 1) then begin
-                                            Static_Reaction := Static_Reaction + -15 + (30 * has_trait(0, dude_obj, 39));
+                                        if (global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE) == 1) then begin
+                                            Static_Reaction := Static_Reaction + -15 + (30 * has_trait(TRAIT_PERK, dude_obj, PERK_cult_of_personality));
                                         end
                                         else begin
-                                            Static_Reaction := Static_Reaction + -20 + (40 * has_trait(0, dude_obj, 39));
+                                            Static_Reaction := Static_Reaction + -20 + (40 * has_trait(TRAIT_PERK, dude_obj, PERK_cult_of_personality));
                                         end
                                     end
                                 end
@@ -595,39 +595,39 @@ begin
             debug_msg("Good Person Reaction Bonus == " + Static_Reaction);
         end
         if (Slavery_Tolerant == 0) then begin
-            if (global_var(11) == 1) then begin
-                Static_Reaction := Static_Reaction + (2 * -25) + (2 * 25 * has_trait(0, dude_obj, 39));
+            if (global_var(GVAR_REPUTATION_SLAVER) == 1) then begin
+                Static_Reaction := Static_Reaction + (2 * -25) + (2 * 25 * has_trait(TRAIT_PERK, dude_obj, PERK_cult_of_personality));
             end
         end
         else begin
             if (Slavery_Tolerant == 1) then begin
-                if (global_var(11) == 1) then begin
-                    Static_Reaction := Static_Reaction + -25 + (25 * has_trait(0, dude_obj, 39));
+                if (global_var(GVAR_REPUTATION_SLAVER) == 1) then begin
+                    Static_Reaction := Static_Reaction + -25 + (25 * has_trait(TRAIT_PERK, dude_obj, PERK_cult_of_personality));
                 end
             end
             else begin
                 if (Slavery_Tolerant == 3) then begin
-                    if (global_var(11) == 1) then begin
+                    if (global_var(GVAR_REPUTATION_SLAVER) == 1) then begin
                         Static_Reaction := Static_Reaction - -25;
                     end
                 end
             end
         end
         debug_msg("Slaver + Aligned Reaction == " + Static_Reaction);
-        Static_Reaction := Static_Reaction + (10 * has_trait(0, dude_obj, 10));
+        Static_Reaction := Static_Reaction + (10 * has_trait(TRAIT_PERK, dude_obj, PERK_presence));
         debug_msg("Presence Reaction == " + Static_Reaction);
-        if (global_var(1) >= 2) then begin
-            Static_Reaction := Static_Reaction + -30 + (30 * has_trait(0, dude_obj, 39));
+        if (global_var(GVAR_CHILDKILLER_REPUTATION) >= 2) then begin
+            Static_Reaction := Static_Reaction + -30 + (30 * has_trait(TRAIT_PERK, dude_obj, PERK_cult_of_personality));
         end
         debug_msg("Childkiller Reaction == " + Static_Reaction);
         set_local_var(0, local_var(3) + Static_Reaction);
         debug_msg("Initial Reaction == " + local_var(0));
-        if (cur_map_index == 9) then begin
+        if (CUR_MAP_KLADWTWN) then begin
             move_obj_inven_to_obj(klam_dunton_box_obj, self_obj);
         end
         call checkarea();
         who_am_i := 1 + random(0, 1);
-        if (global_var(204) == 1) then begin
+        if (global_var(GVAR_KLAMATH_REP_ALIANCE) == 1) then begin
             call Node998();
         end
         else begin
@@ -635,7 +635,7 @@ begin
                 call Node035();
             end
             else begin
-                if (get_critter_stat(dude_obj, 4) <= (0 - -3)) then begin
+                if (dude_iq <= (0 - -3)) then begin
                     start_gdialog(87, self_obj, 4, -1, -1);
                     gsay_start;
                     call Node032();
@@ -644,25 +644,25 @@ begin
                 end
                 else begin
                     if (area == 3) then begin
-                        if (global_var(701) == 1) then begin
-                            float_msg(self_obj, message_str(87, 709), 2);
+                        if (global_var(GVAR_ENDGAME_MOVIE_MARCUS) == 1) then begin
+                            float_msg(self_obj, message_str(SCRIPT_KCCDNTN, 709), FLOAT_MSG_RED);
                         end
                         else begin
                             start_gdialog(87, self_obj, 4, -1, -1);
                             gsay_start;
-                            if (global_var(76) == 1) then begin
+                            if (global_var(GVAR_RUSTLE_FAIL) == 1) then begin
                                 call Node007();
                             end
                             else begin
-                                if (global_var(77) == 1) then begin
+                                if (global_var(GVAR_RUSTLE_SUCCESS) == 1) then begin
                                     call Node008();
                                 end
                                 else begin
-                                    if (global_var(180) == 1) then begin
+                                    if (global_var(GVAR_RUSTLE_REFUSE) == 1) then begin
                                         call Node012();
                                     end
                                     else begin
-                                        if (global_var(179) == 1) then begin
+                                        if (global_var(GVAR_RUSTLE_ACCEPT) == 1) then begin
                                             call Node025();
                                         end
                                         else begin
@@ -676,7 +676,7 @@ begin
                         end
                     end
                     else begin
-                        if (global_var(74) == 1) then begin
+                        if (global_var(GVAR_TORR_BRAHMIN_KILLED) == 1) then begin
                             start_gdialog(87, self_obj, 4, -1, -1);
                             gsay_start;
                             call Node001();
@@ -684,7 +684,7 @@ begin
                             end_dialogue;
                         end
                         else begin
-                            if (global_var(75) == 1) then begin
+                            if (global_var(GVAR_RUSTLE_FAIL_VIOLENT) == 1) then begin
                                 start_gdialog(87, self_obj, 4, -1, -1);
                                 gsay_start;
                                 call Node004();
@@ -692,9 +692,9 @@ begin
                                 end_dialogue;
                             end
                             else begin
-                                if (global_var(76) == 1) then begin
-                                    if (global_var(701) == 1) then begin
-                                        float_msg(self_obj, message_str(87, 709), 2);
+                                if (global_var(GVAR_RUSTLE_FAIL) == 1) then begin
+                                    if (global_var(GVAR_ENDGAME_MOVIE_MARCUS) == 1) then begin
+                                        float_msg(self_obj, message_str(SCRIPT_KCCDNTN, 709), FLOAT_MSG_RED);
                                     end
                                     else begin
                                         start_gdialog(87, self_obj, 4, -1, -1);
@@ -705,7 +705,7 @@ begin
                                     end
                                 end
                                 else begin
-                                    if (global_var(77) == 1) then begin
+                                    if (global_var(GVAR_RUSTLE_SUCCESS) == 1) then begin
                                         start_gdialog(87, self_obj, 4, -1, -1);
                                         gsay_start;
                                         call Node008();
@@ -713,7 +713,7 @@ begin
                                         end_dialogue;
                                     end
                                     else begin
-                                        if (global_var(180) == 1) then begin
+                                        if (global_var(GVAR_RUSTLE_REFUSE) == 1) then begin
                                             start_gdialog(87, self_obj, 4, -1, -1);
                                             gsay_start;
                                             call Node012();
@@ -721,7 +721,7 @@ begin
                                             end_dialogue;
                                         end
                                         else begin
-                                            if ((global_var(179) == 1) and (global_var(184) == 0)) then begin
+                                            if ((global_var(GVAR_RUSTLE_ACCEPT) == 1) and (global_var(GVAR_RUSTLE_OVER) == 0)) then begin
                                                 start_gdialog(87, self_obj, 4, -1, -1);
                                                 gsay_start;
                                                 call Node014();
@@ -729,7 +729,7 @@ begin
                                                 end_dialogue;
                                             end
                                             else begin
-                                                if ((global_var(179) == 1) and (global_var(184) == 1) and (local_var(10) == 0)) then begin
+                                                if ((global_var(GVAR_RUSTLE_ACCEPT) == 1) and (global_var(GVAR_RUSTLE_OVER) == 1) and (local_var(10) == 0)) then begin
                                                     set_local_var(10, 1);
                                                     start_gdialog(87, self_obj, 4, -1, -1);
                                                     gsay_start;
@@ -768,7 +768,7 @@ begin
                 end
             end
         end
-        if (cur_map_index == 9) then begin
+        if (CUR_MAP_KLADWTWN) then begin
             move_obj_inven_to_obj(self_obj, klam_dunton_box_obj);
         end
     end
@@ -777,59 +777,59 @@ end
 procedure destroy_p_proc
 begin
     if (active) then begin
-        if (cur_map_index == 9) then begin
+        if (CUR_MAP_KLADWTWN) then begin
             move_obj_inven_to_obj(klam_dunton_box_obj, self_obj);
         end
         if (source_obj == dude_obj) then begin
-            set_global_var(4, global_var(4) + 1);
-            if (metarule(51, self_obj) == 2) then begin
+            set_global_var(GVAR_BAD_MONSTER, global_var(GVAR_BAD_MONSTER) + 1);
+            if (critter_kill_type(self_obj) == KILL_TYPE_children_kills) then begin
                 if (source_obj == dude_obj) then begin
-                    set_global_var(2, 0);
-                    set_global_var(1, global_var(1) + 1);
-                    set_global_var(0, global_var(0) + -15);
+                    set_global_var(GVAR_CHAMPION_REPUTATION, 0);
+                    set_global_var(GVAR_CHILDKILLER_REPUTATION, global_var(GVAR_CHILDKILLER_REPUTATION) + 1);
+                    set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + -15);
                     debug_msg("Player gains " + -15 + " Karma Points.");
-                    set_global_var(37, 0);
-                    set_global_var(38, 0);
-                    set_global_var(39, 0);
-                    set_global_var(40, 0);
-                    set_global_var(41, 0);
-                    set_global_var(42, 0);
-                    set_global_var(43, 0);
-                    set_global_var(44, 0);
-                    set_global_var(45, 0);
-                    if (global_var(0) >= 1000) then begin
-                        set_global_var(37, 1);
+                    set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+                    set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+                    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+                    set_global_var(GVAR_KARMA_DEFENDER, 0);
+                    set_global_var(GVAR_KARMA_WANDERER, 0);
+                    set_global_var(GVAR_KARMA_BETRAYER, 0);
+                    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+                    set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+                    set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+                        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
                     end
                     else begin
-                        if (global_var(0) >= 750) then begin
-                            set_global_var(38, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                            set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
                         end
                         else begin
-                            if (global_var(0) >= 500) then begin
-                                set_global_var(39, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                                set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                             end
                             else begin
-                                if (global_var(0) >= 250) then begin
-                                    set_global_var(40, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                                    set_global_var(GVAR_KARMA_DEFENDER, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -250) then begin
-                                        set_global_var(41, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                                        set_global_var(GVAR_KARMA_WANDERER, 1);
                                     end
                                     else begin
-                                        if (global_var(0) > -500) then begin
-                                            set_global_var(42, 1);
+                                        if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                            set_global_var(GVAR_KARMA_BETRAYER, 1);
                                         end
                                         else begin
-                                            if (global_var(0) > -750) then begin
-                                                set_global_var(43, 1);
+                                            if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                                set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                             end
                                             else begin
-                                                if (global_var(0) > -1000) then begin
-                                                    set_global_var(44, 1);
+                                                if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                                    set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                                 end
                                                 else begin
-                                                    set_global_var(45, 1);
+                                                    set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                                 end
                                             end
                                         end
@@ -838,62 +838,62 @@ begin
                             end
                         end
                     end
-                    set_global_var(48, global_var(48) + -8);
+                    set_global_var(GVAR_TOWN_REP_KLAMATH, global_var(GVAR_TOWN_REP_KLAMATH) + -8);
                     debug_msg("Added " + -8 + " to Town Rep");
                 end
             end
-            if (((global_var(4) + global_var(5)) >= 25) and ((global_var(5) > (2 * global_var(4))) or (global_var(3) == 1))) then begin
-                set_global_var(3, 1);
-                set_global_var(2, 0);
+            if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
+                set_global_var(GVAR_BERSERKER_REPUTATION, 1);
+                set_global_var(GVAR_CHAMPION_REPUTATION, 0);
             end
-            if (((global_var(4) + global_var(5)) >= 25) and ((global_var(4) > (3 * global_var(5))) or (global_var(2) == 1)) and (global_var(1) == 0)) then begin
-                set_global_var(2, 1);
-                set_global_var(3, 0);
+            if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(4) > (3 * global_var(GVAR_GOOD_MONSTER))) or (global_var(GVAR_CHAMPION_REPUTATION) == 1)) and (global_var(GVAR_CHILDKILLER_REPUTATION) == 0)) then begin
+                set_global_var(GVAR_CHAMPION_REPUTATION, 1);
+                set_global_var(GVAR_BERSERKER_REPUTATION, 0);
             end
-            set_global_var(0, global_var(0) + 5);
+            set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + 5);
             debug_msg("Player gains " + 5 + " Karma Points.");
-            set_global_var(37, 0);
-            set_global_var(38, 0);
-            set_global_var(39, 0);
-            set_global_var(40, 0);
-            set_global_var(41, 0);
-            set_global_var(42, 0);
-            set_global_var(43, 0);
-            set_global_var(44, 0);
-            set_global_var(45, 0);
-            if (global_var(0) >= 1000) then begin
-                set_global_var(37, 1);
+            set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+            set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+            set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+            set_global_var(GVAR_KARMA_DEFENDER, 0);
+            set_global_var(GVAR_KARMA_WANDERER, 0);
+            set_global_var(GVAR_KARMA_BETRAYER, 0);
+            set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+            set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+            set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+                set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
             end
             else begin
-                if (global_var(0) >= 750) then begin
-                    set_global_var(38, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                    set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
                 end
                 else begin
-                    if (global_var(0) >= 500) then begin
-                        set_global_var(39, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                        set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                     end
                     else begin
-                        if (global_var(0) >= 250) then begin
-                            set_global_var(40, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                            set_global_var(GVAR_KARMA_DEFENDER, 1);
                         end
                         else begin
-                            if (global_var(0) > -250) then begin
-                                set_global_var(41, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                                set_global_var(GVAR_KARMA_WANDERER, 1);
                             end
                             else begin
-                                if (global_var(0) > -500) then begin
-                                    set_global_var(42, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                    set_global_var(GVAR_KARMA_BETRAYER, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -750) then begin
-                                        set_global_var(43, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                        set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                     end
                                     else begin
-                                        if (global_var(0) > -1000) then begin
-                                            set_global_var(44, 1);
+                                        if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                            set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                         end
                                         else begin
-                                            set_global_var(45, 1);
+                                            set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                         end
                                     end
                                 end
@@ -902,10 +902,10 @@ begin
                     end
                 end
             end
-            set_global_var(48, global_var(48) + 2);
+            set_global_var(GVAR_TOWN_REP_KLAMATH, global_var(GVAR_TOWN_REP_KLAMATH) + 2);
             debug_msg("Added " + 2 + " to Town Rep");
         end
-        set_global_var(188, 1);
+        set_global_var(GVAR_DUNTON_DEAD, 1);
     end
 end
 
@@ -913,7 +913,7 @@ procedure look_at_p_proc
 begin
     if (active) then begin
         script_overrides;
-        display_msg(message_str(87, 100));
+        display_msg(message_str(SCRIPT_KCCDNTN, 100));
     end
 end
 
@@ -921,7 +921,7 @@ procedure description_p_proc
 begin
     if (active) then begin
         script_overrides;
-        display_msg(message_str(87, 101));
+        display_msg(message_str(SCRIPT_KCCDNTN, 101));
     end
 end
 
@@ -933,7 +933,7 @@ procedure damage_p_proc
 begin
     if (active) then begin
         if (source_obj == dude_obj) then begin
-            set_global_var(204, 1);
+            set_global_var(GVAR_KLAMATH_REP_ALIANCE, 1);
         end
     end
 end
@@ -945,20 +945,20 @@ end
 procedure map_enter_p_proc
 begin
     Only_Once := 0;
-    critter_add_trait(self_obj, 1, 5, 126);
-    if (metarule(14, 0)) then begin
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_DUNTON);
+    if (map_first_run) then begin
         dunton_weapon := 234;
-        item := create_object_sid(dunton_weapon, 0, 0, -1);
+        item := create_object(dunton_weapon, 0, 0);
         add_obj_to_inven(self_obj, item);
     end
-    if (cur_map_index == 9) then begin
-        critter_add_trait(self_obj, 1, 6, 43);
+    if (CUR_MAP_KLADWTWN) then begin
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 43);
         set_local_var(5, 0);
         active := 1;
     end
     else begin
-        if (cur_map_index == 14) then begin
-            critter_add_trait(self_obj, 1, 6, 35);
+        if (CUR_MAP_KLAGRAZ) then begin
+            critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 35);
             call rustle_status();
             call should_i_be_here();
         end
@@ -974,7 +974,7 @@ procedure checkarea
 begin
     here := tile_num(self_obj);
     area := 0;
-    if (cur_map_index == 14) then begin
+    if (CUR_MAP_KLAGRAZ) then begin
         area := 3;
     end
     else begin
@@ -1021,36 +1021,36 @@ end
 
 procedure duntonbarter
 begin
-    gdialog_mod_barter(0);
+    gdialog_barter;
     set_local_var(6, 1);
-    gsay_message(87, "", 50);
+    gsay_message(SCRIPT_KCCDNTN, "", NEUTRAL_REACTION);
     call Node013();
 end
 
 procedure rustle_status
 begin
-    if (global_var(179) == 1) then begin
+    if (global_var(GVAR_RUSTLE_ACCEPT) == 1) then begin
         rustle_triggered := 1;
     end
-    if (global_var(180) == 1) then begin
+    if (global_var(GVAR_RUSTLE_REFUSE) == 1) then begin
         rustle_triggered := 1;
     end
-    if (global_var(182) == 1) then begin
+    if (global_var(GVAR_TORR_GUARD_STATUS) == 1) then begin
         rustle_triggered := 1;
     end
-    if (global_var(77) == 1) then begin
+    if (global_var(GVAR_RUSTLE_SUCCESS) == 1) then begin
         rustle_triggered := 0;
     end
-    if (global_var(76) == 1) then begin
+    if (global_var(GVAR_RUSTLE_FAIL) == 1) then begin
         rustle_triggered := 0;
     end
-    if (global_var(70) == 1) then begin
+    if (global_var(GVAR_TORR_DEAD) == 1) then begin
         rustle_triggered := 0;
     end
-    if (global_var(188) == 1) then begin
+    if (global_var(GVAR_DUNTON_DEAD) == 1) then begin
         rustle_triggered := 0;
     end
-    if (global_var(184) == 1) then begin
+    if (global_var(GVAR_RUSTLE_OVER) == 1) then begin
         rustle_triggered := 0;
     end
 end
@@ -1071,7 +1071,7 @@ end
 procedure combat
 begin
     debug_msg("COMBAT");
-    set_global_var(204, 1);
+    set_global_var(GVAR_KLAMATH_REP_ALIANCE, 1);
 end
 
 procedure scamper
@@ -1080,7 +1080,7 @@ begin
         set_map_var(3, 1);
         call rustle_status();
         gfade_out(900);
-        if ((global_var(184) == 1) and (cur_map_index == 14)) then begin
+        if ((global_var(GVAR_RUSTLE_OVER) == 1) and (CUR_MAP_KLAGRAZ)) then begin
             reg_anim_func(2, self_obj);
             move_to(self_obj, 0, 0);
         end
@@ -1095,35 +1095,35 @@ end
 
 procedure Node900
 begin
-    gsay_reply(87, 703);
-    giq_option(4, 87, 704, Node901, 50);
+    gsay_reply(SCRIPT_KCCDNTN, 703);
+    giq_option(4, SCRIPT_KCCDNTN, 704, Node901, NEUTRAL_REACTION);
 end
 
 procedure Node901
 begin
     if (is_success(roll_vs_skill(dude_obj, 14, 0))) then begin
-        gsay_reply(87, 706);
-        display_msg(message_str(87, 707));
-        giq_option(4, 87, 708, Node902, 50);
+        gsay_reply(SCRIPT_KCCDNTN, 706);
+        display_msg(message_str(SCRIPT_KCCDNTN, 707));
+        giq_option(4, SCRIPT_KCCDNTN, 708, Node902, NEUTRAL_REACTION);
     end
     else begin
-        gsay_reply(87, 705);
-        giq_option(4, 87, 708, Node998, 51);
+        gsay_reply(SCRIPT_KCCDNTN, 705);
+        giq_option(4, SCRIPT_KCCDNTN, 708, Node998, BAD_REACTION);
     end
 end
 
 procedure Node902
 begin
-    set_global_var(76, 1);
-    set_global_var(701, 1);
+    set_global_var(GVAR_RUSTLE_FAIL, 1);
+    set_global_var(GVAR_ENDGAME_MOVIE_MARCUS, 1);
     give_exp_points(250);
-    display_msg(message_str(14, 100) + 250 + message_str(14, 101));
+    display_msg(g_mstr(100) + 250 + g_mstr(101));
 end
 
 procedure Node903
 begin
-    gsay_reply(87, 710);
-    giq_option(4, 87, 708, Node999, 50);
+    gsay_reply(SCRIPT_KCCDNTN, 710);
+    giq_option(4, SCRIPT_KCCDNTN, 708, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node910
@@ -1596,14 +1596,14 @@ end
 
 procedure Node938
 begin
-    gsay_reply(87, 510);
-    giq_option(1, 87, 520, Node016, 50);
+    gsay_reply(SCRIPT_KCCDNTN, 510);
+    giq_option(1, SCRIPT_KCCDNTN, 520, Node016, NEUTRAL_REACTION);
 end
 
 procedure Node939
 begin
-    gsay_reply(87, 500);
-    giq_option(1, 87, 520, Node017, 50);
+    gsay_reply(SCRIPT_KCCDNTN, 500);
+    giq_option(1, SCRIPT_KCCDNTN, 520, Node017, NEUTRAL_REACTION);
 end
 
 procedure Node940
@@ -1753,7 +1753,7 @@ begin
         set_map_var(7, 1);
         if (local_var(8) == 0) then begin
             set_local_var(8, 1);
-            set_global_var(48, global_var(48) + -5);
+            set_global_var(GVAR_TOWN_REP_KLAMATH, global_var(GVAR_TOWN_REP_KLAMATH) + -5);
         end
         call Node938();
     end
@@ -1761,7 +1761,7 @@ begin
         set_map_var(8, 1);
         if (local_var(9) == 0) then begin
             set_local_var(9, 1);
-            set_global_var(48, global_var(48) + -2);
+            set_global_var(GVAR_TOWN_REP_KLAMATH, global_var(GVAR_TOWN_REP_KLAMATH) + -2);
         end
         call Node939();
     end
@@ -2513,15 +2513,15 @@ end
 
 procedure Node973
 begin
-    set_global_var(102, 1);
-    set_global_var(179, 1);
+    set_global_var(GVAR_QUEST_KILL_EVIL_PLANTS, 1);
+    set_global_var(GVAR_RUSTLE_ACCEPT, 1);
     call Node003();
 end
 
 procedure Node975
 begin
-    set_global_var(102, 1);
-    set_global_var(179, 1);
+    set_global_var(GVAR_QUEST_KILL_EVIL_PLANTS, 1);
+    set_global_var(GVAR_RUSTLE_ACCEPT, 1);
     if (area != 3) then begin
         gfade_out(600);
         game_time_advance(game_ticks(900));
@@ -2765,8 +2765,8 @@ begin
             end
         end
     end
-    set_global_var(102, 1);
-    set_global_var(179, 1);
+    set_global_var(GVAR_QUEST_KILL_EVIL_PLANTS, 1);
+    set_global_var(GVAR_RUSTLE_ACCEPT, 1);
 end
 
 procedure Node990
@@ -2819,22 +2819,22 @@ end
 
 procedure Node991
 begin
-    gsay_reply(87, 198);
-    giq_option(4, 87, 202, Node975, 49);
-    giq_option(4, 87, 204, Node993, 51);
+    gsay_reply(SCRIPT_KCCDNTN, 198);
+    giq_option(4, SCRIPT_KCCDNTN, 202, Node975, GOOD_REACTION);
+    giq_option(4, SCRIPT_KCCDNTN, 204, Node993, BAD_REACTION);
 end
 
 procedure Node992
 begin
-    gsay_reply(87, message_str(87, 198) + message_str(87, 601));
-    giq_option(4, 87, 202, Node975, 49);
-    giq_option(4, 87, 204, Node993, 51);
+    gsay_reply(SCRIPT_KCCDNTN, message_str(SCRIPT_KCCDNTN, 198) + message_str(SCRIPT_KCCDNTN, 601));
+    giq_option(4, SCRIPT_KCCDNTN, 202, Node975, GOOD_REACTION);
+    giq_option(4, SCRIPT_KCCDNTN, 204, Node993, BAD_REACTION);
 end
 
 procedure Node993
 begin
     if (from_14) then begin
-        set_global_var(180, 1);
+        set_global_var(GVAR_RUSTLE_REFUSE, 1);
     end
     else begin
         set_local_var(3, local_var(3) + -10);
@@ -2886,52 +2886,52 @@ end
 
 procedure Node994
 begin
-    set_global_var(76, 1);
-    set_global_var(102, 3);
-    set_global_var(0, global_var(0) + 50);
+    set_global_var(GVAR_RUSTLE_FAIL, 1);
+    set_global_var(GVAR_QUEST_KILL_EVIL_PLANTS, 3);
+    set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + 50);
     debug_msg("Player gains " + 50 + " Karma Points.");
-    set_global_var(37, 0);
-    set_global_var(38, 0);
-    set_global_var(39, 0);
-    set_global_var(40, 0);
-    set_global_var(41, 0);
-    set_global_var(42, 0);
-    set_global_var(43, 0);
-    set_global_var(44, 0);
-    set_global_var(45, 0);
-    if (global_var(0) >= 1000) then begin
-        set_global_var(37, 1);
-    end
-    else begin
-        if (global_var(0) >= 750) then begin
-            set_global_var(38, 1);
+    set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+    set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+    set_global_var(GVAR_KARMA_DEFENDER, 0);
+    set_global_var(GVAR_KARMA_WANDERER, 0);
+    set_global_var(GVAR_KARMA_BETRAYER, 0);
+    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+    set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+    set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+    if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
+    end
+    else begin
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+            set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
         end
         else begin
-            if (global_var(0) >= 500) then begin
-                set_global_var(39, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
             end
             else begin
-                if (global_var(0) >= 250) then begin
-                    set_global_var(40, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                    set_global_var(GVAR_KARMA_DEFENDER, 1);
                 end
                 else begin
-                    if (global_var(0) > -250) then begin
-                        set_global_var(41, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                        set_global_var(GVAR_KARMA_WANDERER, 1);
                     end
                     else begin
-                        if (global_var(0) > -500) then begin
-                            set_global_var(42, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                            set_global_var(GVAR_KARMA_BETRAYER, 1);
                         end
                         else begin
-                            if (global_var(0) > -750) then begin
-                                set_global_var(43, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                             end
                             else begin
-                                if (global_var(0) > -1000) then begin
-                                    set_global_var(44, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                    set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                 end
                                 else begin
-                                    set_global_var(45, 1);
+                                    set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                 end
                             end
                         end
@@ -2941,12 +2941,12 @@ begin
         end
     end
     give_exp_points(250);
-    display_msg(message_str(14, 100) + 250 + message_str(14, 101));
+    display_msg(g_mstr(100) + 250 + g_mstr(101));
 end
 
 procedure Node995
 begin
-    if (get_critter_stat(dude_obj, 3) >= 7) then begin
+    if (dude_charisma >= 7) then begin
         set_local_var(3, 100);
         set_local_var(1, 4);
     end
@@ -2964,7 +2964,7 @@ end
 
 procedure Node997
 begin
-    if (get_critter_stat(dude_obj, 3) >= 8) then begin
+    if (dude_charisma >= 8) then begin
         call Node004();
     end
     else begin
@@ -2974,7 +2974,7 @@ end
 
 procedure Node998
 begin
-    set_global_var(204, 1);
+    set_global_var(GVAR_KLAMATH_REP_ALIANCE, 1);
 end
 
 procedure Node999
@@ -2983,76 +2983,76 @@ end
 
 procedure Node001
 begin
-    gsay_reply(87, 110);
-    giq_option(4, 87, 111, Node936, 49);
-    giq_option(4, 87, 112, Node999, 50);
-    giq_option(4, 87, 113, Node932, 51);
-    giq_option(4, 87, 114, Node950, 51);
+    gsay_reply(SCRIPT_KCCDNTN, 110);
+    giq_option(4, SCRIPT_KCCDNTN, 111, Node936, GOOD_REACTION);
+    giq_option(4, SCRIPT_KCCDNTN, 112, Node999, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_KCCDNTN, 113, Node932, BAD_REACTION);
+    giq_option(4, SCRIPT_KCCDNTN, 114, Node950, BAD_REACTION);
 end
 
 procedure Node002
 begin
-    gsay_reply(87, 120);
-    giq_option(4, 87, 121, Node961, 49);
-    giq_option(4, 87, 122, Node999, 50);
-    giq_option(4, 87, 123, combat, 51);
+    gsay_reply(SCRIPT_KCCDNTN, 120);
+    giq_option(4, SCRIPT_KCCDNTN, 121, Node961, GOOD_REACTION);
+    giq_option(4, SCRIPT_KCCDNTN, 122, Node999, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_KCCDNTN, 123, combat, BAD_REACTION);
 end
 
 procedure Node003
 begin
     if (from_18) then begin
-        gsay_reply(87, 132);
+        gsay_reply(SCRIPT_KCCDNTN, 132);
     end
     else begin
-        if (global_var(48) < 5) then begin
-            gsay_reply(87, 130);
+        if (global_var(GVAR_TOWN_REP_KLAMATH) < 5) then begin
+            gsay_reply(SCRIPT_KCCDNTN, 130);
         end
         else begin
-            gsay_reply(87, 131);
+            gsay_reply(SCRIPT_KCCDNTN, 131);
         end
     end
-    giq_option(4, 87, 133, Node019, 50);
+    giq_option(4, SCRIPT_KCCDNTN, 133, Node019, NEUTRAL_REACTION);
     if (area == 2) then begin
-        giq_option(4, 87, 134, Node005, 50);
+        giq_option(4, SCRIPT_KCCDNTN, 134, Node005, NEUTRAL_REACTION);
     end
     if (local_var(7) == 0) then begin
         set_local_var(7, 1);
-        giq_option(4, 87, 135, Node006, 50);
+        giq_option(4, SCRIPT_KCCDNTN, 135, Node006, NEUTRAL_REACTION);
     end
-    giq_option(4, 87, 136, Node011, 50);
-    giq_option(4, 87, 137, Node020, 50);
-    giq_option(4, 87, 138, Node999, 50);
+    giq_option(4, SCRIPT_KCCDNTN, 136, Node011, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_KCCDNTN, 137, Node020, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_KCCDNTN, 138, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node004
 begin
-    gsay_reply(87, 140);
-    if (global_var(179) == 1) then begin
-        giq_option(4, 87, 141, Node999, 50);
-        giq_option(4, 87, 142, Node980, 51);
+    gsay_reply(SCRIPT_KCCDNTN, 140);
+    if (global_var(GVAR_RUSTLE_ACCEPT) == 1) then begin
+        giq_option(4, SCRIPT_KCCDNTN, 141, Node999, NEUTRAL_REACTION);
+        giq_option(4, SCRIPT_KCCDNTN, 142, Node980, BAD_REACTION);
     end
-    giq_option(4, 87, 143, Node990, 51);
+    giq_option(4, SCRIPT_KCCDNTN, 143, Node990, BAD_REACTION);
 end
 
 procedure Node005
 begin
-    gsay_reply(87, 150);
-    giq_option(4, 87, 151, Node937, 50);
-    giq_option(4, 87, 152, Node003, 50);
-    giq_option(4, 87, 153, Node999, 50);
+    gsay_reply(SCRIPT_KCCDNTN, 150);
+    giq_option(4, SCRIPT_KCCDNTN, 151, Node937, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_KCCDNTN, 152, Node003, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_KCCDNTN, 153, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node006
 begin
-    if (global_var(180) == 1) then begin
+    if (global_var(GVAR_RUSTLE_REFUSE) == 1) then begin
         call Node06a();
     end
     else begin
-        if ((global_var(184) == 1) and (has_trait(0, dude_obj, 39) == 0)) then begin
+        if ((global_var(GVAR_RUSTLE_OVER) == 1) and (has_trait(TRAIT_PERK, dude_obj, PERK_cult_of_personality) == 0)) then begin
             call Node06b();
         end
         else begin
-            if ((global_var(48) > 4) and (has_trait(0, dude_obj, 39) == 0)) then begin
+            if ((global_var(GVAR_TOWN_REP_KLAMATH) > 4) and (has_trait(TRAIT_PERK, dude_obj, PERK_cult_of_personality) == 0)) then begin
                 call Node06c();
             end
             else begin
@@ -3064,140 +3064,140 @@ end
 
 procedure Node06a
 begin
-    gsay_reply(87, 160);
-    if ((global_var(48) > 4) and (area == 3)) then begin
-        giq_option(4, 87, 531, Node999, 50);
+    gsay_reply(SCRIPT_KCCDNTN, 160);
+    if ((global_var(GVAR_TOWN_REP_KLAMATH) > 4) and (area == 3)) then begin
+        giq_option(4, SCRIPT_KCCDNTN, 531, Node999, NEUTRAL_REACTION);
     end
     if (area != 3) then begin
-        giq_option(4, 87, 166, Node003, 50);
+        giq_option(4, SCRIPT_KCCDNTN, 166, Node003, NEUTRAL_REACTION);
     end
-    if (global_var(184) == 0) then begin
-        giq_option(4, 87, 167, Node948, 49);
+    if (global_var(GVAR_RUSTLE_OVER) == 0) then begin
+        giq_option(4, SCRIPT_KCCDNTN, 167, Node948, GOOD_REACTION);
     end
-    giq_option(4, 87, 168, Node999, 50);
+    giq_option(4, SCRIPT_KCCDNTN, 168, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node06b
 begin
-    gsay_reply(87, 161);
-    giq_option(4, 87, 169, Node999, 50);
+    gsay_reply(SCRIPT_KCCDNTN, 161);
+    giq_option(4, SCRIPT_KCCDNTN, 169, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node06c
 begin
     if (area == 3) then begin
-        gsay_reply(87, 530);
+        gsay_reply(SCRIPT_KCCDNTN, 530);
     end
     else begin
-        gsay_reply(87, 161);
+        gsay_reply(SCRIPT_KCCDNTN, 161);
     end
-    giq_option(4, 87, 169, Node999, 50);
+    giq_option(4, SCRIPT_KCCDNTN, 169, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node06d
 begin
-    gsay_reply(87, 162);
+    gsay_reply(SCRIPT_KCCDNTN, 162);
     if (area != 3) then begin
-        giq_option(4, 87, 163, Node945, 50);
-        giq_option(4, 87, 164, Node946, 50);
+        giq_option(4, SCRIPT_KCCDNTN, 163, Node945, NEUTRAL_REACTION);
+        giq_option(4, SCRIPT_KCCDNTN, 164, Node946, NEUTRAL_REACTION);
     end
-    giq_option(4, 87, 165, Node010, 50);
+    giq_option(4, SCRIPT_KCCDNTN, 165, Node010, NEUTRAL_REACTION);
 end
 
 procedure Node007
 begin
-    if (global_var(701) == 1) then begin
-        float_msg(self_obj, message_str(87, 709), 2);
+    if (global_var(GVAR_ENDGAME_MOVIE_MARCUS) == 1) then begin
+        float_msg(self_obj, message_str(SCRIPT_KCCDNTN, 709), FLOAT_MSG_RED);
     end
     else begin
-        gsay_reply(87, 170);
-        giq_option(4, 87, 171, Node955, 49);
+        gsay_reply(SCRIPT_KCCDNTN, 170);
+        giq_option(4, SCRIPT_KCCDNTN, 171, Node955, GOOD_REACTION);
         if (area != 3) then begin
-            giq_option(4, 87, 172, Node003, 50);
+            giq_option(4, SCRIPT_KCCDNTN, 172, Node003, NEUTRAL_REACTION);
         end
-        giq_option(4, 87, 173, combat, 51);
+        giq_option(4, SCRIPT_KCCDNTN, 173, combat, BAD_REACTION);
     end
 end
 
 procedure Node008
 begin
-    if (global_var(181) == 1) then begin
-        gsay_reply(87, 180);
+    if (global_var(GVAR_RUSTLE_REWARD) == 1) then begin
+        gsay_reply(SCRIPT_KCCDNTN, 180);
     end
     else begin
-        gsay_reply(87, 181);
+        gsay_reply(SCRIPT_KCCDNTN, 181);
     end
-    if ((global_var(181) == 1) and (area != 3)) then begin
-        giq_option(4, 87, 182, Node983, 49);
+    if ((global_var(GVAR_RUSTLE_REWARD) == 1) and (area != 3)) then begin
+        giq_option(4, SCRIPT_KCCDNTN, 182, Node983, GOOD_REACTION);
     end
-    if (global_var(181) == 0) then begin
-        giq_option(4, 87, 183, Node985, 49);
+    if (global_var(GVAR_RUSTLE_REWARD) == 0) then begin
+        giq_option(4, SCRIPT_KCCDNTN, 183, Node985, GOOD_REACTION);
     end
-    giq_option(4, 87, 184, Node988, 51);
-    giq_option(4, 87, 185, Node999, 50);
+    giq_option(4, SCRIPT_KCCDNTN, 184, Node988, BAD_REACTION);
+    giq_option(4, SCRIPT_KCCDNTN, 185, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node009
 begin
-    item := create_object_sid(284, 0, 0, -1);
+    item := create_object(284, 0, 0);
     add_obj_to_inven(dude_obj, item);
-    item := create_object_sid(284, 0, 0, -1);
+    item := create_object(284, 0, 0);
     add_obj_to_inven(dude_obj, item);
-    item := create_object_sid(284, 0, 0, -1);
+    item := create_object(284, 0, 0);
     add_obj_to_inven(dude_obj, item);
-    item := create_object_sid(284, 0, 0, -1);
+    item := create_object(284, 0, 0);
     add_obj_to_inven(dude_obj, item);
-    item := create_object_sid(284, 0, 0, -1);
+    item := create_object(284, 0, 0);
     add_obj_to_inven(dude_obj, item);
     item_caps_adjust(dude_obj, 50);
-    set_global_var(181, 1);
-    display_msg(message_str(87, 700));
+    set_global_var(GVAR_RUSTLE_REWARD, 1);
+    display_msg(message_str(SCRIPT_KCCDNTN, 700));
     give_exp_points(150);
-    display_msg(message_str(14, 100) + 150 + message_str(14, 101));
-    set_global_var(0, global_var(0) + -60);
+    display_msg(g_mstr(100) + 150 + g_mstr(101));
+    set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + -60);
     debug_msg("Player gains " + -60 + " Karma Points.");
-    set_global_var(37, 0);
-    set_global_var(38, 0);
-    set_global_var(39, 0);
-    set_global_var(40, 0);
-    set_global_var(41, 0);
-    set_global_var(42, 0);
-    set_global_var(43, 0);
-    set_global_var(44, 0);
-    set_global_var(45, 0);
-    if (global_var(0) >= 1000) then begin
-        set_global_var(37, 1);
-    end
-    else begin
-        if (global_var(0) >= 750) then begin
-            set_global_var(38, 1);
+    set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+    set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+    set_global_var(GVAR_KARMA_DEFENDER, 0);
+    set_global_var(GVAR_KARMA_WANDERER, 0);
+    set_global_var(GVAR_KARMA_BETRAYER, 0);
+    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+    set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+    set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+    if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
+    end
+    else begin
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+            set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
         end
         else begin
-            if (global_var(0) >= 500) then begin
-                set_global_var(39, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
             end
             else begin
-                if (global_var(0) >= 250) then begin
-                    set_global_var(40, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                    set_global_var(GVAR_KARMA_DEFENDER, 1);
                 end
                 else begin
-                    if (global_var(0) > -250) then begin
-                        set_global_var(41, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                        set_global_var(GVAR_KARMA_WANDERER, 1);
                     end
                     else begin
-                        if (global_var(0) > -500) then begin
-                            set_global_var(42, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                            set_global_var(GVAR_KARMA_BETRAYER, 1);
                         end
                         else begin
-                            if (global_var(0) > -750) then begin
-                                set_global_var(43, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                             end
                             else begin
-                                if (global_var(0) > -1000) then begin
-                                    set_global_var(44, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                    set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                 end
                                 else begin
-                                    set_global_var(45, 1);
+                                    set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                 end
                             end
                         end
@@ -3206,12 +3206,12 @@ begin
             end
         end
     end
-    set_global_var(182, 3);
-    gsay_reply(87, 190);
+    set_global_var(GVAR_TORR_GUARD_STATUS, 3);
+    gsay_reply(SCRIPT_KCCDNTN, 190);
     if (area != 3) then begin
-        giq_option(4, 87, 191, Node003, 50);
+        giq_option(4, SCRIPT_KCCDNTN, 191, Node003, NEUTRAL_REACTION);
     end
-    giq_option(4, 87, 192, Node999, 50);
+    giq_option(4, SCRIPT_KCCDNTN, 192, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node010
@@ -3230,23 +3230,23 @@ begin
     if (who_am_i == 1) then begin
         k := random(1, 5);
         if (k == 1) then begin
-            gsay_reply(87, 208);
+            gsay_reply(SCRIPT_KCCDNTN, 208);
         end
         else begin
             if (k == 2) then begin
-                gsay_reply(87, 210);
+                gsay_reply(SCRIPT_KCCDNTN, 210);
             end
             else begin
                 if (k == 3) then begin
-                    gsay_reply(87, 212);
+                    gsay_reply(SCRIPT_KCCDNTN, 212);
                 end
                 else begin
                     if (k == 4) then begin
-                        gsay_reply(87, 214);
+                        gsay_reply(SCRIPT_KCCDNTN, 214);
                     end
                     else begin
                         if (k == 5) then begin
-                            gsay_reply(87, 216);
+                            gsay_reply(SCRIPT_KCCDNTN, 216);
                         end
                     end
                 end
@@ -3256,225 +3256,225 @@ begin
     else begin
         k := random(1, 5);
         if (k == 1) then begin
-            gsay_reply(87, 209);
+            gsay_reply(SCRIPT_KCCDNTN, 209);
         end
         else begin
             if (k == 2) then begin
-                gsay_reply(87, 211);
+                gsay_reply(SCRIPT_KCCDNTN, 211);
             end
             else begin
                 if (k == 3) then begin
-                    gsay_reply(87, 213);
+                    gsay_reply(SCRIPT_KCCDNTN, 213);
                 end
                 else begin
                     if (k == 4) then begin
-                        gsay_reply(87, 215);
+                        gsay_reply(SCRIPT_KCCDNTN, 215);
                     end
                     else begin
                         if (k == 5) then begin
-                            gsay_reply(87, 217);
+                            gsay_reply(SCRIPT_KCCDNTN, 217);
                         end
                     end
                 end
             end
         end
     end
-    giq_option(4, 87, 218, Node935, 49);
-    giq_option(4, 87, 219, Node003, 50);
-    giq_option(4, 87, 220, Node931, 51);
-    giq_option(4, 87, 221, Node934, 51);
+    giq_option(4, SCRIPT_KCCDNTN, 218, Node935, GOOD_REACTION);
+    giq_option(4, SCRIPT_KCCDNTN, 219, Node003, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_KCCDNTN, 220, Node931, BAD_REACTION);
+    giq_option(4, SCRIPT_KCCDNTN, 221, Node934, BAD_REACTION);
     if (from_10) then begin
-        giq_option(4, 87, 222, Node010, 50);
+        giq_option(4, SCRIPT_KCCDNTN, 222, Node010, NEUTRAL_REACTION);
     end
 end
 
 procedure Node012
 begin
-    set_global_var(180, 1);
-    gsay_reply(87, 230);
-    giq_option(4, 87, 231, Node960, 49);
+    set_global_var(GVAR_RUSTLE_REFUSE, 1);
+    gsay_reply(SCRIPT_KCCDNTN, 230);
+    giq_option(4, SCRIPT_KCCDNTN, 231, Node960, GOOD_REACTION);
     if (area != 3) then begin
-        giq_option(4, 87, 232, Node003, 50);
+        giq_option(4, SCRIPT_KCCDNTN, 232, Node003, NEUTRAL_REACTION);
     end
-    giq_option(4, 87, 233, Node941, 51);
+    giq_option(4, SCRIPT_KCCDNTN, 233, Node941, BAD_REACTION);
 end
 
 procedure Node013
 begin
-    gsay_reply(87, 240);
-    giq_option(4, 87, 241, Node937, 50);
-    giq_option(4, 87, 242, Node003, 50);
-    giq_option(4, 87, 243, Node981, 50);
-    giq_option(4, 87, 244, Node951, 50);
+    gsay_reply(SCRIPT_KCCDNTN, 240);
+    giq_option(4, SCRIPT_KCCDNTN, 241, Node937, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_KCCDNTN, 242, Node003, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_KCCDNTN, 243, Node981, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_KCCDNTN, 244, Node951, NEUTRAL_REACTION);
 end
 
 procedure Node014
 begin
     from_14 := 1;
-    gsay_reply(87, 250);
-    giq_option(4, 87, 251, Node971, 51);
+    gsay_reply(SCRIPT_KCCDNTN, 250);
+    giq_option(4, SCRIPT_KCCDNTN, 251, Node971, BAD_REACTION);
     if (area != 3) then begin
-        giq_option(4, 87, 252, Node003, 50);
+        giq_option(4, SCRIPT_KCCDNTN, 252, Node003, NEUTRAL_REACTION);
     end
-    giq_option(4, 87, 253, Node999, 50);
+    giq_option(4, SCRIPT_KCCDNTN, 253, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node015
 begin
     if (map_var(7) == 1) then begin
-        gsay_reply(87, 259);
+        gsay_reply(SCRIPT_KCCDNTN, 259);
     end
     else begin
         if (map_var(8) == 1) then begin
-            gsay_reply(87, 260);
+            gsay_reply(SCRIPT_KCCDNTN, 260);
         end
         else begin
-            if (get_critter_stat(dude_obj, 0) < 5) then begin
-                if (global_var(48) > 4) then begin
-                    gsay_reply(87, 261);
+            if (dude_strength < 5) then begin
+                if (global_var(GVAR_TOWN_REP_KLAMATH) > 4) then begin
+                    gsay_reply(SCRIPT_KCCDNTN, 261);
                 end
                 else begin
                     if (who_am_i == 1) then begin
-                        gsay_reply(87, 262);
+                        gsay_reply(SCRIPT_KCCDNTN, 262);
                     end
                     else begin
-                        gsay_reply(87, 263);
+                        gsay_reply(SCRIPT_KCCDNTN, 263);
                     end
                 end
             end
             else begin
                 if (who_am_i == 1) then begin
-                    if (global_var(48) > 4) then begin
-                        gsay_reply(87, 264);
+                    if (global_var(GVAR_TOWN_REP_KLAMATH) > 4) then begin
+                        gsay_reply(SCRIPT_KCCDNTN, 264);
                     end
                     else begin
-                        gsay_reply(87, 266);
+                        gsay_reply(SCRIPT_KCCDNTN, 266);
                     end
                 end
                 else begin
-                    if (global_var(48) > 4) then begin
-                        gsay_reply(87, 265);
+                    if (global_var(GVAR_TOWN_REP_KLAMATH) > 4) then begin
+                        gsay_reply(SCRIPT_KCCDNTN, 265);
                     end
                     else begin
-                        gsay_reply(87, 267);
+                        gsay_reply(SCRIPT_KCCDNTN, 267);
                     end
                 end
             end
         end
     end
-    giq_option(4, 87, 268, Node944, 49);
-    giq_option(4, 87, 269, Node018, 50);
-    giq_option(4, 87, 270, Node930, 51);
-    giq_option(4, 87, 271, Node999, 50);
+    giq_option(4, SCRIPT_KCCDNTN, 268, Node944, GOOD_REACTION);
+    giq_option(4, SCRIPT_KCCDNTN, 269, Node018, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_KCCDNTN, 270, Node930, BAD_REACTION);
+    giq_option(4, SCRIPT_KCCDNTN, 271, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node016
 begin
-    gsay_reply(87, 280);
-    giq_option(4, 87, 281, Node967, 49);
-    giq_option(4, 87, 282, Node999, 50);
-    giq_option(4, 87, 283, Node930, 51);
-    giq_option(4, 87, 284, Node941, 51);
+    gsay_reply(SCRIPT_KCCDNTN, 280);
+    giq_option(4, SCRIPT_KCCDNTN, 281, Node967, GOOD_REACTION);
+    giq_option(4, SCRIPT_KCCDNTN, 282, Node999, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_KCCDNTN, 283, Node930, BAD_REACTION);
+    giq_option(4, SCRIPT_KCCDNTN, 284, Node941, BAD_REACTION);
 end
 
 procedure Node017
 begin
     if (map_var(9) == 1) then begin
-        gsay_reply(87, 291);
+        gsay_reply(SCRIPT_KCCDNTN, 291);
     end
     else begin
         set_map_var(9, 1);
-        gsay_reply(87, 290);
+        gsay_reply(SCRIPT_KCCDNTN, 290);
     end
-    giq_option(4, 87, 292, Node944, 49);
-    giq_option(4, 87, 293, Node003, 50);
-    giq_option(4, 87, 294, Node999, 50);
+    giq_option(4, SCRIPT_KCCDNTN, 292, Node944, GOOD_REACTION);
+    giq_option(4, SCRIPT_KCCDNTN, 293, Node003, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_KCCDNTN, 294, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node018
 begin
     from_18 := 1;
-    gsay_reply(87, 300);
-    giq_option(4, 87, 301, Node935, 49);
-    giq_option(4, 87, 302, Node003, 50);
-    giq_option(4, 87, 303, Node941, 51);
-    giq_option(4, 87, 304, Node942, 51);
+    gsay_reply(SCRIPT_KCCDNTN, 300);
+    giq_option(4, SCRIPT_KCCDNTN, 301, Node935, GOOD_REACTION);
+    giq_option(4, SCRIPT_KCCDNTN, 302, Node003, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_KCCDNTN, 303, Node941, BAD_REACTION);
+    giq_option(4, SCRIPT_KCCDNTN, 304, Node942, BAD_REACTION);
 end
 
 procedure Node019
 begin
-    gsay_reply(87, 310);
-    giq_option(4, 87, 311, Node003, 50);
-    giq_option(4, 87, 312, Node999, 50);
+    gsay_reply(SCRIPT_KCCDNTN, 310);
+    giq_option(4, SCRIPT_KCCDNTN, 311, Node003, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_KCCDNTN, 312, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node020
 begin
-    gsay_reply(87, 320);
-    giq_option(4, 87, 321, Node003, 50);
-    giq_option(4, 87, 322, Node999, 50);
+    gsay_reply(SCRIPT_KCCDNTN, 320);
+    giq_option(4, SCRIPT_KCCDNTN, 321, Node003, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_KCCDNTN, 322, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node021
 begin
     if (map_var(10) == 0) then begin
         set_map_var(10, 1);
-        gsay_reply(87, 330);
+        gsay_reply(SCRIPT_KCCDNTN, 330);
     end
     else begin
-        gsay_reply(87, 331);
+        gsay_reply(SCRIPT_KCCDNTN, 331);
     end
-    giq_option(4, 87, 332, Node952, 49);
-    giq_option(4, 87, 333, Node964, 49);
-    giq_option(4, 87, 334, Node003, 50);
-    giq_option(4, 87, 335, Node999, 50);
+    giq_option(4, SCRIPT_KCCDNTN, 332, Node952, GOOD_REACTION);
+    giq_option(4, SCRIPT_KCCDNTN, 333, Node964, GOOD_REACTION);
+    giq_option(4, SCRIPT_KCCDNTN, 334, Node003, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_KCCDNTN, 335, Node999, NEUTRAL_REACTION);
     if (is_success(do_check(dude_obj, 1, 0))) then begin
-        giq_option(4, 87, 336, Node022, 51);
+        giq_option(4, SCRIPT_KCCDNTN, 336, Node022, BAD_REACTION);
     end
 end
 
 procedure Node022
 begin
-    gsay_reply(87, 340);
-    giq_option(4, 87, 341, Node972, 51);
-    giq_option(4, 87, 342, Node999, 50);
-    giq_option(4, 87, 343, Node021, 50);
+    gsay_reply(SCRIPT_KCCDNTN, 340);
+    giq_option(4, SCRIPT_KCCDNTN, 341, Node972, BAD_REACTION);
+    giq_option(4, SCRIPT_KCCDNTN, 342, Node999, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_KCCDNTN, 343, Node021, NEUTRAL_REACTION);
 end
 
 procedure Node023
 begin
     if (map_var(11) == 0) then begin
         set_map_var(11, 1);
-        gsay_reply(87, 350);
+        gsay_reply(SCRIPT_KCCDNTN, 350);
     end
     else begin
         k := random(1, 7);
         if (k == 1) then begin
-            gsay_reply(87, 351);
+            gsay_reply(SCRIPT_KCCDNTN, 351);
         end
         else begin
             if (k == 2) then begin
-                gsay_reply(87, 352);
+                gsay_reply(SCRIPT_KCCDNTN, 352);
             end
             else begin
                 if (k == 3) then begin
-                    gsay_reply(87, 353);
+                    gsay_reply(SCRIPT_KCCDNTN, 353);
                 end
                 else begin
                     if (k == 4) then begin
-                        gsay_reply(87, 354);
+                        gsay_reply(SCRIPT_KCCDNTN, 354);
                     end
                     else begin
                         if (k == 5) then begin
-                            gsay_reply(87, 355);
+                            gsay_reply(SCRIPT_KCCDNTN, 355);
                         end
                         else begin
                             if (k == 6) then begin
-                                gsay_reply(87, 356);
+                                gsay_reply(SCRIPT_KCCDNTN, 356);
                             end
                             else begin
                                 if (k == 7) then begin
-                                    gsay_reply(87, 357);
+                                    gsay_reply(SCRIPT_KCCDNTN, 357);
                                 end
                             end
                         end
@@ -3483,135 +3483,135 @@ begin
             end
         end
     end
-    giq_option(4, 87, 358, Node963, 49);
-    giq_option(4, 87, 359, Node023, 49);
-    giq_option(4, 87, 360, Node999, 50);
-    giq_option(4, 87, 361, Node966, 51);
+    giq_option(4, SCRIPT_KCCDNTN, 358, Node963, GOOD_REACTION);
+    giq_option(4, SCRIPT_KCCDNTN, 359, Node023, GOOD_REACTION);
+    giq_option(4, SCRIPT_KCCDNTN, 360, Node999, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_KCCDNTN, 361, Node966, BAD_REACTION);
 end
 
 procedure Node024
 begin
-    if (global_var(180) == 1) then begin
-        gsay_reply(87, 370);
+    if (global_var(GVAR_RUSTLE_REFUSE) == 1) then begin
+        gsay_reply(SCRIPT_KCCDNTN, 370);
     end
     else begin
-        gsay_reply(87, 371);
+        gsay_reply(SCRIPT_KCCDNTN, 371);
     end
-    giq_option(4, 87, 372, Node006, 50);
-    giq_option(4, 87, 373, Node941, 50);
+    giq_option(4, SCRIPT_KCCDNTN, 372, Node006, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_KCCDNTN, 373, Node941, NEUTRAL_REACTION);
     if (obj_is_carrying_obj_pid(dude_obj, 285) > 0) then begin
-        giq_option(4, 87, 374, Node026, 50);
+        giq_option(4, SCRIPT_KCCDNTN, 374, Node026, NEUTRAL_REACTION);
     end
 end
 
 procedure Node025
 begin
-    if (global_var(701) == 1) then begin
-        float_msg(self_obj, message_str(87, 709), 2);
+    if (global_var(GVAR_ENDGAME_MOVIE_MARCUS) == 1) then begin
+        float_msg(self_obj, message_str(SCRIPT_KCCDNTN, 709), FLOAT_MSG_RED);
     end
     else begin
-        gsay_reply(87, 380);
-        giq_option(4, 87, 381, Node999, 50);
-        giq_option(4, 87, 382, Node921, 50);
-        giq_option(4, 87, 383, Node027, 50);
+        gsay_reply(SCRIPT_KCCDNTN, 380);
+        giq_option(4, SCRIPT_KCCDNTN, 381, Node999, NEUTRAL_REACTION);
+        giq_option(4, SCRIPT_KCCDNTN, 382, Node921, NEUTRAL_REACTION);
+        giq_option(4, SCRIPT_KCCDNTN, 383, Node027, NEUTRAL_REACTION);
         if (obj_is_carrying_obj_pid(dude_obj, 285) > 0) then begin
-            giq_option(4, 87, 384, Node026, 50);
+            giq_option(4, SCRIPT_KCCDNTN, 384, Node026, NEUTRAL_REACTION);
         end
     end
 end
 
 procedure Node026
 begin
-    gsay_reply(87, 390);
-    giq_option(4, 87, 391, Node922, 51);
-    giq_option(4, 87, 392, Node999, 50);
+    gsay_reply(SCRIPT_KCCDNTN, 390);
+    giq_option(4, SCRIPT_KCCDNTN, 391, Node922, BAD_REACTION);
+    giq_option(4, SCRIPT_KCCDNTN, 392, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node027
 begin
     if (map_var(4) == 1) then begin
-        gsay_reply(87, message_str(87, 400) + message_str(87, 405));
+        gsay_reply(SCRIPT_KCCDNTN, message_str(SCRIPT_KCCDNTN, 400) + message_str(SCRIPT_KCCDNTN, 405));
     end
     else begin
-        gsay_reply(87, message_str(87, 404) + message_str(87, 405));
+        gsay_reply(SCRIPT_KCCDNTN, message_str(SCRIPT_KCCDNTN, 404) + message_str(SCRIPT_KCCDNTN, 405));
     end
-    giq_option(4, 87, 401, Node989, 49);
-    giq_option(4, 87, 402, Node923, 51);
-    giq_option(4, 87, 403, Node921, 50);
+    giq_option(4, SCRIPT_KCCDNTN, 401, Node989, GOOD_REACTION);
+    giq_option(4, SCRIPT_KCCDNTN, 402, Node923, BAD_REACTION);
+    giq_option(4, SCRIPT_KCCDNTN, 403, Node921, NEUTRAL_REACTION);
 end
 
 procedure Node028
 begin
-    gsay_reply(87, 410);
-    giq_option(4, 87, 411, Node994, 50);
+    gsay_reply(SCRIPT_KCCDNTN, 410);
+    giq_option(4, SCRIPT_KCCDNTN, 411, Node994, NEUTRAL_REACTION);
 end
 
 procedure Node029
 begin
     if (map_var(4) == 1) then begin
-        gsay_reply(87, message_str(87, 420) + message_str(87, 424));
+        gsay_reply(SCRIPT_KCCDNTN, message_str(SCRIPT_KCCDNTN, 420) + message_str(SCRIPT_KCCDNTN, 424));
     end
     else begin
-        gsay_reply(87, message_str(87, 420) + message_str(87, 533));
+        gsay_reply(SCRIPT_KCCDNTN, message_str(SCRIPT_KCCDNTN, 420) + message_str(SCRIPT_KCCDNTN, 533));
     end
-    giq_option(4, 87, 421, Node960, 49);
-    giq_option(4, 87, 422, Node999, 51);
-    giq_option(4, 87, 423, Node942, 50);
+    giq_option(4, SCRIPT_KCCDNTN, 421, Node960, GOOD_REACTION);
+    giq_option(4, SCRIPT_KCCDNTN, 422, Node999, BAD_REACTION);
+    giq_option(4, SCRIPT_KCCDNTN, 423, Node942, NEUTRAL_REACTION);
 end
 
 procedure Node030
 begin
-    gsay_reply(87, 430);
-    giq_option(4, 87, 431, Node924, 51);
-    giq_option(4, 87, 432, Node999, 50);
+    gsay_reply(SCRIPT_KCCDNTN, 430);
+    giq_option(4, SCRIPT_KCCDNTN, 431, Node924, BAD_REACTION);
+    giq_option(4, SCRIPT_KCCDNTN, 432, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node031
 begin
-    gsay_reply(87, 440);
-    if (global_var(184) == 0) then begin
-        giq_option(4, 87, 441, Node903, 49);
+    gsay_reply(SCRIPT_KCCDNTN, 440);
+    if (global_var(GVAR_RUSTLE_OVER) == 0) then begin
+        giq_option(4, SCRIPT_KCCDNTN, 441, Node903, GOOD_REACTION);
     end
-    if ((global_var(184) == 0) and (has_skill(dude_obj, 14) > 50)) then begin
-        giq_option(4, 87, 702, Node900, 51);
+    if ((global_var(GVAR_RUSTLE_OVER) == 0) and (dude_skill(SKILL_SPEECH) > 50)) then begin
+        giq_option(4, SCRIPT_KCCDNTN, 702, Node900, BAD_REACTION);
     end
-    giq_option(4, 87, 442, Node999, 50);
-    giq_option(4, 87, 443, Node940, 51);
-    giq_option(4, 87, 444, Node998, 50);
+    giq_option(4, SCRIPT_KCCDNTN, 442, Node999, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_KCCDNTN, 443, Node940, BAD_REACTION);
+    giq_option(4, SCRIPT_KCCDNTN, 444, Node998, NEUTRAL_REACTION);
 end
 
 procedure Node032
 begin
     if (who_am_i == 1) then begin
-        gsay_reply(87, message_str(87, 449) + message_str(87, 450));
+        gsay_reply(SCRIPT_KCCDNTN, message_str(SCRIPT_KCCDNTN, 449) + message_str(SCRIPT_KCCDNTN, 450));
     end
     else begin
-        gsay_reply(87, message_str(87, 449) + message_str(87, 451));
+        gsay_reply(SCRIPT_KCCDNTN, message_str(SCRIPT_KCCDNTN, 449) + message_str(SCRIPT_KCCDNTN, 451));
     end
-    giq_option(-3, 87, 452, Node033, 51);
-    giq_option(-3, 87, obj_name(dude_obj) + message_str(87, 453), Node034, 50);
+    giq_option(-3, SCRIPT_KCCDNTN, 452, Node033, BAD_REACTION);
+    giq_option(-3, SCRIPT_KCCDNTN, dude_name + message_str(SCRIPT_KCCDNTN, 453), Node034, NEUTRAL_REACTION);
 end
 
 procedure Node033
 begin
-    gsay_reply(87, 460);
-    giq_option(-3, 87, 461, Node999, 50);
+    gsay_reply(SCRIPT_KCCDNTN, 460);
+    giq_option(-3, SCRIPT_KCCDNTN, 461, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node034
 begin
-    gsay_reply(87, 470);
-    giq_option(-3, 87, message_str(87, 471) + obj_name(dude_obj) + message_str(87, 472), Node999, 50);
+    gsay_reply(SCRIPT_KCCDNTN, 470);
+    giq_option(-3, SCRIPT_KCCDNTN, message_str(SCRIPT_KCCDNTN, 471) + dude_name + message_str(SCRIPT_KCCDNTN, 472), Node999, NEUTRAL_REACTION);
 end
 
 procedure Node035
 begin
-    float_msg(self_obj, message_str(87, 490), 2);
+    float_msg(self_obj, message_str(SCRIPT_KCCDNTN, 490), FLOAT_MSG_RED);
 end
 
 procedure Node036
 begin
-    float_msg(self_obj, message_str(87, 530), 2);
+    float_msg(self_obj, message_str(SCRIPT_KCCDNTN, 530), FLOAT_MSG_RED);
 end
 
 procedure Node981
@@ -3620,4 +3620,3 @@ begin
         call Node011();
     end
 end
-
