diff --git a/Mods/MegaMod/Original/KCJENNY.ssl b/Mods/MegaMod/ReDefine/KCJENNY.ssl
index ba237ac..6048740 100644
--- a/Mods/MegaMod/Original/KCJENNY.ssl
+++ b/Mods/MegaMod/ReDefine/KCJENNY.ssl
@@ -178,15 +178,15 @@ end
 
 procedure critter_p_proc
 begin
-    if ((global_var(68) == 1) or (global_var(603) == 1) or (local_var(5) == 1) and obj_can_see_obj(self_obj, dude_obj)) then begin
+    if ((global_var(GVAR_ENEMY_KLAMATH) == 1) or (global_var(GVAR_BATH_HOUSE_REJECT) == 1) or (local_var(5) == 1) and obj_can_see_obj(self_obj, dude_obj)) then begin
         if (critter_is_fleeing(self_obj)) then begin
             if (anim_busy(self_obj) == 0) then begin
                 if ((tile_distance_objs(self_obj, dude_obj) < 8) or obj_can_see_obj(self_obj, dude_obj)) then begin
                     if (anim_busy(self_obj) == 0) then begin
-                        global_temp := rotation_to_tile(tile_num(dude_obj), tile_num(self_obj));
+                        global_temp := rotation_to_tile(dude_tile, tile_num(self_obj));
                         animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(self_obj), global_temp, random(3, 10)), 1);
                         global_temp := (global_temp + 1) % 6;
-                        while ((anim_busy(self_obj) == 0) and (global_temp != rotation_to_tile(tile_num(dude_obj), tile_num(self_obj)))) do begin
+                        while ((anim_busy(self_obj) == 0) and (global_temp != rotation_to_tile(dude_tile, tile_num(self_obj)))) do begin
                             debug_msg("flee loop: rot == " + global_temp);
                             animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(self_obj), global_temp, random(3, 10)), 1);
                             global_temp := (global_temp + 1) % 6;
@@ -196,23 +196,23 @@ begin
             end
         end
         else begin
-            attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+            attack(dude_obj);
         end
     end
-    if ((global_var(204) <= -1) and obj_can_see_obj(self_obj, dude_obj)) then begin
-        attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+    if ((global_var(GVAR_KLAMATH_REP_ALIANCE) <= -1) and obj_can_see_obj(self_obj, dude_obj)) then begin
+        attack(dude_obj);
     end
-    if ((global_var(696) != 0) and ((game_time - global_var(696)) < (10 * (24 * (60 * (60 * 10))))) and (local_var(6) == 0)) then begin
+    if ((global_var(GVAR_ACWARRIOR_TALKCOUNT) != 0) and ((game_time - global_var(GVAR_ACWARRIOR_TALKCOUNT)) < (10 * (24 * (60 * (60 * 10))))) and (local_var(6) == 0)) then begin
         set_local_var(6, 1);
         set_obj_visibility(self_obj, 1);
     end
     else begin
-        if ((global_var(696) != 0) and ((game_time - global_var(696)) >= (10 * (24 * (60 * (60 * 10))))) and (local_var(6) == 1)) then begin
+        if ((global_var(GVAR_ACWARRIOR_TALKCOUNT) != 0) and ((game_time - global_var(GVAR_ACWARRIOR_TALKCOUNT)) >= (10 * (24 * (60 * (60 * 10))))) and (local_var(6) == 1)) then begin
             set_local_var(6, 2);
             set_obj_visibility(self_obj, 0);
         end
         else begin
-            if ((map_var(16) == 1) and (tile_distance(tile_num(self_obj), tile_num(dude_obj)) < 3)) then begin
+            if ((map_var(16) == 1) and (tile_distance(tile_num(self_obj), dude_tile) < 3)) then begin
                 debug_msg("Attempting to enter dialogue");
                 dialogue_system_enter;
             end
@@ -223,9 +223,9 @@ end
 procedure pickup_p_proc
 begin
     if (source_obj == dude_obj) then begin
-        float_msg(self_obj, message_str(84, 500), 8);
+        float_msg(self_obj, mstr(500), FLOAT_MSG_YELLOW);
         set_local_var(5, 1);
-        set_global_var(603, 1);
+        set_global_var(GVAR_BATH_HOUSE_REJECT, 1);
     end
 end
 
@@ -233,7 +233,7 @@ procedure talk_p_proc
 begin
     if (local_var(2) == 0) then begin
         set_local_var(2, 1);
-        set_local_var(3, (get_critter_stat(dude_obj, 3) - get_critter_stat(self_obj, 3)) * 5);
+        set_local_var(3, (dude_charisma - self_charisma) * 5);
         debug_msg("Base Reaction == " + local_var(3));
     end
     Static_Reaction := 0;
@@ -245,14 +245,14 @@ begin
     else begin
         reaction_bonus_karma := -1 * (-1 * reaction_bonus_karma / 1000);
     end
-    if (global_var(48) >= 0) then begin
-        reaction_bonus_town_rep := global_var(48) / 2;
+    if (global_var(GVAR_TOWN_REP_KLAMATH) >= 0) then begin
+        reaction_bonus_town_rep := global_var(GVAR_TOWN_REP_KLAMATH) / 2;
     end
     else begin
-        reaction_bonus_town_rep := -1 * (-1 * global_var(48) / 2);
+        reaction_bonus_town_rep := -1 * (-1 * global_var(GVAR_TOWN_REP_KLAMATH) / 2);
     end
-    if (has_trait(2, dude_obj, 13)) then begin
-        if (get_critter_stat(dude_obj, 34) == get_critter_stat(self_obj, 34)) then begin
+    if (has_trait(TRAIT_TRAIT, dude_obj, TRAIT_sex_appeal)) then begin
+        if (dude_gender == self_gender) then begin
             Static_Reaction := Static_Reaction + -20;
         end
         else begin
@@ -268,48 +268,48 @@ begin
         debug_msg("Karma Reaction Bonus == " + Static_Reaction);
         Static_Reaction := Static_Reaction - reaction_bonus_town_rep;
         debug_msg("Town Rep Bonus == " + Static_Reaction);
-        set_global_var(37, 0);
-        set_global_var(38, 0);
-        set_global_var(39, 0);
-        set_global_var(40, 0);
-        set_global_var(41, 0);
-        set_global_var(42, 0);
-        set_global_var(43, 0);
-        set_global_var(44, 0);
-        set_global_var(45, 0);
-        if (global_var(0) >= 1000) then begin
-            set_global_var(37, 1);
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+        set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+        set_global_var(GVAR_KARMA_DEFENDER, 0);
+        set_global_var(GVAR_KARMA_WANDERER, 0);
+        set_global_var(GVAR_KARMA_BETRAYER, 0);
+        set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+        set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+            set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
         end
         else begin
-            if (global_var(0) >= 750) then begin
-                set_global_var(38, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
             end
             else begin
-                if (global_var(0) >= 500) then begin
-                    set_global_var(39, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                 end
                 else begin
-                    if (global_var(0) >= 250) then begin
-                        set_global_var(40, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                        set_global_var(GVAR_KARMA_DEFENDER, 1);
                     end
                     else begin
-                        if (global_var(0) > -250) then begin
-                            set_global_var(41, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                            set_global_var(GVAR_KARMA_WANDERER, 1);
                         end
                         else begin
-                            if (global_var(0) > -500) then begin
-                                set_global_var(42, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                set_global_var(GVAR_KARMA_BETRAYER, 1);
                             end
                             else begin
-                                if (global_var(0) > -750) then begin
-                                    set_global_var(43, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -1000) then begin
-                                        set_global_var(44, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                     end
                                     else begin
-                                        set_global_var(45, 1);
+                                        set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                     end
                                 end
                             end
@@ -318,35 +318,35 @@ begin
                 end
             end
         end
-        if (global_var(37) == 1) then begin
+        if (global_var(GVAR_KARMA_HOLY_WARRIOR) == 1) then begin
             Static_Reaction := Static_Reaction - 20;
         end
         else begin
-            if (global_var(38) == 1) then begin
+            if (global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES) == 1) then begin
                 Static_Reaction := Static_Reaction - 15;
             end
             else begin
-                if (global_var(39) == 1) then begin
+                if (global_var(GVAR_KARMA_SHIELD_OF_HOPE) == 1) then begin
                     Static_Reaction := Static_Reaction - 10;
                 end
                 else begin
-                    if (global_var(40) == 1) then begin
+                    if (global_var(GVAR_KARMA_DEFENDER) == 1) then begin
                         Static_Reaction := Static_Reaction - 5;
                     end
                     else begin
-                        if (global_var(41) == 1) then begin
+                        if (global_var(GVAR_KARMA_WANDERER) == 1) then begin
                             Static_Reaction := Static_Reaction - 0;
                         end
                         else begin
-                            if (global_var(42) == 1) then begin
+                            if (global_var(GVAR_KARMA_BETRAYER) == 1) then begin
                                 Static_Reaction := Static_Reaction - -5;
                             end
                             else begin
-                                if (global_var(43) == 1) then begin
+                                if (global_var(GVAR_KARMA_SWORD_OF_DESPAIR) == 1) then begin
                                     Static_Reaction := Static_Reaction - -10;
                                 end
                                 else begin
-                                    if (global_var(44) == 1) then begin
+                                    if (global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE) == 1) then begin
                                         Static_Reaction := Static_Reaction - -15;
                                     end
                                     else begin
@@ -366,48 +366,48 @@ begin
         debug_msg("Karma Reaction Bonus == " + Static_Reaction);
         Static_Reaction := Static_Reaction + reaction_bonus_town_rep;
         debug_msg("Town Rep Bonus == " + Static_Reaction);
-        set_global_var(37, 0);
-        set_global_var(38, 0);
-        set_global_var(39, 0);
-        set_global_var(40, 0);
-        set_global_var(41, 0);
-        set_global_var(42, 0);
-        set_global_var(43, 0);
-        set_global_var(44, 0);
-        set_global_var(45, 0);
-        if (global_var(0) >= 1000) then begin
-            set_global_var(37, 1);
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+        set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+        set_global_var(GVAR_KARMA_DEFENDER, 0);
+        set_global_var(GVAR_KARMA_WANDERER, 0);
+        set_global_var(GVAR_KARMA_BETRAYER, 0);
+        set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+        set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+            set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
         end
         else begin
-            if (global_var(0) >= 750) then begin
-                set_global_var(38, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
             end
             else begin
-                if (global_var(0) >= 500) then begin
-                    set_global_var(39, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                 end
                 else begin
-                    if (global_var(0) >= 250) then begin
-                        set_global_var(40, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                        set_global_var(GVAR_KARMA_DEFENDER, 1);
                     end
                     else begin
-                        if (global_var(0) > -250) then begin
-                            set_global_var(41, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                            set_global_var(GVAR_KARMA_WANDERER, 1);
                         end
                         else begin
-                            if (global_var(0) > -500) then begin
-                                set_global_var(42, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                set_global_var(GVAR_KARMA_BETRAYER, 1);
                             end
                             else begin
-                                if (global_var(0) > -750) then begin
-                                    set_global_var(43, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -1000) then begin
-                                        set_global_var(44, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                     end
                                     else begin
-                                        set_global_var(45, 1);
+                                        set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                     end
                                 end
                             end
@@ -416,35 +416,35 @@ begin
                 end
             end
         end
-        if (global_var(37) == 1) then begin
+        if (global_var(GVAR_KARMA_HOLY_WARRIOR) == 1) then begin
             Static_Reaction := Static_Reaction + 20;
         end
         else begin
-            if (global_var(38) == 1) then begin
+            if (global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES) == 1) then begin
                 Static_Reaction := Static_Reaction + 15;
             end
             else begin
-                if (global_var(39) == 1) then begin
+                if (global_var(GVAR_KARMA_SHIELD_OF_HOPE) == 1) then begin
                     Static_Reaction := Static_Reaction + 10;
                 end
                 else begin
-                    if (global_var(40) == 1) then begin
+                    if (global_var(GVAR_KARMA_DEFENDER) == 1) then begin
                         Static_Reaction := Static_Reaction + 5;
                     end
                     else begin
-                        if (global_var(41) == 1) then begin
+                        if (global_var(GVAR_KARMA_WANDERER) == 1) then begin
                             Static_Reaction := Static_Reaction + 0;
                         end
                         else begin
-                            if (global_var(42) == 1) then begin
+                            if (global_var(GVAR_KARMA_BETRAYER) == 1) then begin
                                 Static_Reaction := Static_Reaction + -5;
                             end
                             else begin
-                                if (global_var(43) == 1) then begin
+                                if (global_var(GVAR_KARMA_SWORD_OF_DESPAIR) == 1) then begin
                                     Static_Reaction := Static_Reaction + -10;
                                 end
                                 else begin
-                                    if (global_var(44) == 1) then begin
+                                    if (global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE) == 1) then begin
                                         Static_Reaction := Static_Reaction + -15;
                                     end
                                     else begin
@@ -460,34 +460,34 @@ begin
         debug_msg("Good Person Reaction Bonus == " + Static_Reaction);
     end
     if (Slavery_Tolerant == 0) then begin
-        if (global_var(11) == 1) then begin
+        if (global_var(GVAR_REPUTATION_SLAVER) == 1) then begin
             Static_Reaction := Static_Reaction + (2 * -25);
         end
     end
     else begin
         if (Slavery_Tolerant == 1) then begin
-            if (global_var(11) == 1) then begin
+            if (global_var(GVAR_REPUTATION_SLAVER) == 1) then begin
                 Static_Reaction := Static_Reaction + -25;
             end
         end
         else begin
             if (Slavery_Tolerant == 3) then begin
-                if (global_var(11) == 1) then begin
+                if (global_var(GVAR_REPUTATION_SLAVER) == 1) then begin
                     Static_Reaction := Static_Reaction - -25;
                 end
             end
         end
     end
     debug_msg("Slaver + Aligned Reaction == " + Static_Reaction);
-    Static_Reaction := Static_Reaction + (10 * has_trait(0, dude_obj, 10));
+    Static_Reaction := Static_Reaction + (10 * has_trait(TRAIT_PERK, dude_obj, PERK_presence));
     debug_msg("Presence Reaction == " + Static_Reaction);
-    if (global_var(1) >= 2) then begin
+    if (global_var(GVAR_CHILDKILLER_REPUTATION) >= 2) then begin
         Static_Reaction := Static_Reaction + -30;
     end
     debug_msg("Childkiller Reaction == " + Static_Reaction);
     set_local_var(0, local_var(3) + Static_Reaction);
     debug_msg("Initial Reaction == " + local_var(0));
-    if ((global_var(68) == 1) or (local_var(5) == 1)) then begin
+    if ((global_var(GVAR_ENEMY_KLAMATH) == 1) or (local_var(5) == 1)) then begin
         call Node998();
     end
     else begin
@@ -496,12 +496,12 @@ begin
             set_map_var(16, 0);
             start_gdialog(84, self_obj, 4, -1, -1);
             gsay_start;
-            if (get_critter_stat(dude_obj, 34) == 0) then begin
-                if (get_critter_stat(dude_obj, 2) < 5) then begin
+            if (dude_gender == 0) then begin
+                if (dude_endurance < 5) then begin
                     call Node001();
                 end
                 else begin
-                    if (get_critter_stat(dude_obj, 2) < 8) then begin
+                    if (dude_endurance < 8) then begin
                         call Node002();
                     end
                     else begin
@@ -510,7 +510,7 @@ begin
                 end
             end
             else begin
-                if (get_critter_stat(dude_obj, 3) < 5) then begin
+                if (dude_charisma < 5) then begin
                     call Node004();
                 end
                 else begin
@@ -523,15 +523,15 @@ begin
         else begin
             start_gdialog(84, self_obj, 4, -1, -1);
             gsay_start;
-            if (get_critter_stat(dude_obj, 4) <= (0 - -3)) then begin
+            if (dude_iq <= (0 - -3)) then begin
                 call Node008();
             end
             else begin
-                if (get_critter_stat(dude_obj, 34) == 0) then begin
+                if (dude_gender == 0) then begin
                     call Node016();
                 end
                 else begin
-                    if (get_critter_stat(dude_obj, 3) < 7) then begin
+                    if (dude_charisma < 7) then begin
                         call Node017();
                     end
                     else begin
@@ -548,59 +548,59 @@ end
 procedure destroy_p_proc
 begin
     if (source_obj == dude_obj) then begin
-        set_global_var(5, global_var(5) + 1);
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(4) > (3 * global_var(5))) or (global_var(2) == 1)) and (global_var(1) == 0)) then begin
-            set_global_var(2, 1);
-            set_global_var(3, 0);
+        set_global_var(GVAR_GOOD_MONSTER, global_var(GVAR_GOOD_MONSTER) + 1);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(4) > (3 * global_var(GVAR_GOOD_MONSTER))) or (global_var(GVAR_CHAMPION_REPUTATION) == 1)) and (global_var(GVAR_CHILDKILLER_REPUTATION) == 0)) then begin
+            set_global_var(GVAR_CHAMPION_REPUTATION, 1);
+            set_global_var(GVAR_BERSERKER_REPUTATION, 0);
         end
-        if (metarule(51, self_obj) == 2) then begin
+        if (critter_kill_type(self_obj) == 2) then begin
             if (source_obj == dude_obj) then begin
-                set_global_var(2, 0);
-                set_global_var(1, global_var(1) + 1);
-                set_global_var(0, global_var(0) + -15);
+                set_global_var(GVAR_CHAMPION_REPUTATION, 0);
+                set_global_var(GVAR_CHILDKILLER_REPUTATION, global_var(GVAR_CHILDKILLER_REPUTATION) + 1);
+                set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + -15);
                 debug_msg("Player gains " + -15 + " Karma Points.");
-                set_global_var(37, 0);
-                set_global_var(38, 0);
-                set_global_var(39, 0);
-                set_global_var(40, 0);
-                set_global_var(41, 0);
-                set_global_var(42, 0);
-                set_global_var(43, 0);
-                set_global_var(44, 0);
-                set_global_var(45, 0);
-                if (global_var(0) >= 1000) then begin
-                    set_global_var(37, 1);
+                set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+                set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+                set_global_var(GVAR_KARMA_DEFENDER, 0);
+                set_global_var(GVAR_KARMA_WANDERER, 0);
+                set_global_var(GVAR_KARMA_BETRAYER, 0);
+                set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+                set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+                set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+                    set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
                 end
                 else begin
-                    if (global_var(0) >= 750) then begin
-                        set_global_var(38, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
                     end
                     else begin
-                        if (global_var(0) >= 500) then begin
-                            set_global_var(39, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                            set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                         end
                         else begin
-                            if (global_var(0) >= 250) then begin
-                                set_global_var(40, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                                set_global_var(GVAR_KARMA_DEFENDER, 1);
                             end
                             else begin
-                                if (global_var(0) > -250) then begin
-                                    set_global_var(41, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                                    set_global_var(GVAR_KARMA_WANDERER, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -500) then begin
-                                        set_global_var(42, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                        set_global_var(GVAR_KARMA_BETRAYER, 1);
                                     end
                                     else begin
-                                        if (global_var(0) > -750) then begin
-                                            set_global_var(43, 1);
+                                        if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                            set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                         end
                                         else begin
-                                            if (global_var(0) > -1000) then begin
-                                                set_global_var(44, 1);
+                                            if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                                set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                             end
                                             else begin
-                                                set_global_var(45, 1);
+                                                set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                             end
                                         end
                                     end
@@ -609,58 +609,58 @@ begin
                         end
                     end
                 end
-                set_global_var(48, global_var(48) + -8);
+                set_global_var(GVAR_TOWN_REP_KLAMATH, global_var(GVAR_TOWN_REP_KLAMATH) + -8);
                 debug_msg("Added " + -8 + " to Town Rep");
             end
         end
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(5) > (2 * global_var(4))) or (global_var(3) == 1))) then begin
-            set_global_var(3, 1);
-            set_global_var(2, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
+            set_global_var(GVAR_BERSERKER_REPUTATION, 1);
+            set_global_var(GVAR_CHAMPION_REPUTATION, 0);
         end
-        set_global_var(0, global_var(0) + -10);
+        set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + -10);
         debug_msg("Player gains " + -10 + " Karma Points.");
-        set_global_var(37, 0);
-        set_global_var(38, 0);
-        set_global_var(39, 0);
-        set_global_var(40, 0);
-        set_global_var(41, 0);
-        set_global_var(42, 0);
-        set_global_var(43, 0);
-        set_global_var(44, 0);
-        set_global_var(45, 0);
-        if (global_var(0) >= 1000) then begin
-            set_global_var(37, 1);
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+        set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+        set_global_var(GVAR_KARMA_DEFENDER, 0);
+        set_global_var(GVAR_KARMA_WANDERER, 0);
+        set_global_var(GVAR_KARMA_BETRAYER, 0);
+        set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+        set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+            set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
         end
         else begin
-            if (global_var(0) >= 750) then begin
-                set_global_var(38, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
             end
             else begin
-                if (global_var(0) >= 500) then begin
-                    set_global_var(39, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                 end
                 else begin
-                    if (global_var(0) >= 250) then begin
-                        set_global_var(40, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                        set_global_var(GVAR_KARMA_DEFENDER, 1);
                     end
                     else begin
-                        if (global_var(0) > -250) then begin
-                            set_global_var(41, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                            set_global_var(GVAR_KARMA_WANDERER, 1);
                         end
                         else begin
-                            if (global_var(0) > -500) then begin
-                                set_global_var(42, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                set_global_var(GVAR_KARMA_BETRAYER, 1);
                             end
                             else begin
-                                if (global_var(0) > -750) then begin
-                                    set_global_var(43, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -1000) then begin
-                                        set_global_var(44, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                     end
                                     else begin
-                                        set_global_var(45, 1);
+                                        set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                     end
                                 end
                             end
@@ -669,7 +669,7 @@ begin
                 end
             end
         end
-        set_global_var(48, global_var(48) + -5);
+        set_global_var(GVAR_TOWN_REP_KLAMATH, global_var(GVAR_TOWN_REP_KLAMATH) + -5);
         debug_msg("Added " + -5 + " to Town Rep");
     end
 end
@@ -677,13 +677,13 @@ end
 procedure look_at_p_proc
 begin
     script_overrides;
-    display_msg(message_str(84, 100));
+    display_mstr(100);
 end
 
 procedure description_p_proc
 begin
     script_overrides;
-    display_msg(message_str(84, 101));
+    display_mstr(101);
 end
 
 procedure use_skill_on_p_proc
@@ -694,20 +694,20 @@ procedure damage_p_proc
 begin
     if (source_obj == dude_obj) then begin
         set_local_var(5, 1);
-        set_global_var(68, 1);
+        set_global_var(GVAR_ENEMY_KLAMATH, 1);
     end
 end
 
 procedure map_enter_p_proc
 begin
     Only_Once := 0;
-    critter_add_trait(self_obj, 1, 6, 35);
-    critter_add_trait(self_obj, 1, 5, 14);
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 35);
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_PEASANTS);
 end
 
 procedure Node995
 begin
-    if (get_critter_stat(dude_obj, 3) >= 7) then begin
+    if (dude_charisma >= 7) then begin
         set_local_var(3, 100);
         set_local_var(1, 4);
     end
@@ -725,7 +725,7 @@ end
 
 procedure Node997
 begin
-    if (get_critter_stat(dude_obj, 3) >= 8) then begin
+    if (dude_charisma >= 8) then begin
         call Node004();
     end
     else begin
@@ -744,156 +744,155 @@ end
 
 procedure Node001
 begin
-    gsay_reply(84, 200);
-    giq_option(4, 84, 201, Node006, 50);
+    Reply(200);
+    giq_option(4, NAME, 201, Node006, NEUTRAL_REACTION);
 end
 
 procedure Node002
 begin
-    gsay_reply(84, 210);
-    giq_option(4, 84, 211, Node006, 50);
+    Reply(210);
+    giq_option(4, NAME, 211, Node006, NEUTRAL_REACTION);
 end
 
 procedure Node003
 begin
-    gsay_reply(84, 220);
-    giq_option(4, 84, 221, Node006, 50);
+    Reply(220);
+    giq_option(4, NAME, 221, Node006, NEUTRAL_REACTION);
 end
 
 procedure Node004
 begin
-    gsay_reply(84, 230);
-    giq_option(4, 84, 231, Node006, 50);
+    Reply(230);
+    giq_option(4, NAME, 231, Node006, NEUTRAL_REACTION);
 end
 
 procedure Node005
 begin
-    gsay_reply(84, 240);
-    giq_option(4, 84, 241, Node006, 50);
+    Reply(240);
+    giq_option(4, NAME, 241, Node006, NEUTRAL_REACTION);
 end
 
 procedure Node006
 begin
-    gsay_reply(84, 250);
-    giq_option(4, 84, 251, Node010, 50);
-    giq_option(4, 84, 252, Node007, 50);
+    Reply(250);
+    giq_option(4, NAME, 251, Node010, NEUTRAL_REACTION);
+    giq_option(4, NAME, 252, Node007, NEUTRAL_REACTION);
 end
 
 procedure Node007
 begin
-    gsay_reply(84, 260);
-    giq_option(1, 84, 294, Node999, 50);
+    Reply(260);
+    giq_option(1, NAME, 294, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node008
 begin
-    gsay_reply(84, 270);
-    giq_option(-3, 84, 271, Node009, 50);
-    giq_option(-3, 84, message_str(84, 272) + obj_name(dude_obj) + message_str(84, 273), Node009, 50);
-    giq_option(-3, 84, 274, Node009, 50);
+    Reply(270);
+    giq_option(-3, NAME, 271, Node009, NEUTRAL_REACTION);
+    giq_option(-3, NAME, mstr(272) + dude_name + mstr(273), Node009, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 274, Node009, NEUTRAL_REACTION);
 end
 
 procedure Node009
 begin
-    gsay_message(84, 280, 50);
+    gsay_message(NAME, 280, NEUTRAL_REACTION);
 end
 
 procedure Node010
 begin
-    gsay_reply(84, 290);
-    giq_option(4, 84, 291, Node011, 50);
-    giq_option(4, 84, 292, Node012, 50);
-    giq_option(6, 84, 293, Node014, 50);
-    giq_option(4, 84, 301, Node007, 50);
-    giq_option(4, 84, 295, Node021, 50);
+    Reply(290);
+    giq_option(4, NAME, 291, Node011, NEUTRAL_REACTION);
+    giq_option(4, NAME, 292, Node012, NEUTRAL_REACTION);
+    giq_option(6, NAME, 293, Node014, NEUTRAL_REACTION);
+    giq_option(4, NAME, 301, Node007, NEUTRAL_REACTION);
+    giq_option(4, NAME, 295, Node021, NEUTRAL_REACTION);
 end
 
 procedure Node011
 begin
-    gsay_reply(84, 300);
-    giq_option(4, 84, 301, Node007, 50);
-    giq_option(4, 84, 302, Node010, 50);
+    Reply(300);
+    giq_option(4, NAME, 301, Node007, NEUTRAL_REACTION);
+    giq_option(4, NAME, 302, Node010, NEUTRAL_REACTION);
 end
 
 procedure Node012
 begin
-    gsay_reply(84, 310);
-    giq_option(4, 84, 311, Node007, 50);
-    giq_option(4, 84, 312, Node013, 50);
-    giq_option(6, 84, 313, Node010, 50);
+    Reply(310);
+    giq_option(4, NAME, 311, Node007, NEUTRAL_REACTION);
+    giq_option(4, NAME, 312, Node013, NEUTRAL_REACTION);
+    giq_option(6, NAME, 313, Node010, NEUTRAL_REACTION);
 end
 
 procedure Node013
 begin
-    gsay_reply(84, 320);
-    giq_option(4, 84, 321, Node015, 50);
-    giq_option(4, 84, 322, Node014, 50);
-    giq_option(7, 84, 323, Node020, 50);
-    giq_option(4, 84, 324, Node010, 50);
-    giq_option(4, 84, 325, Node007, 50);
+    Reply(320);
+    giq_option(4, NAME, 321, Node015, NEUTRAL_REACTION);
+    giq_option(4, NAME, 322, Node014, NEUTRAL_REACTION);
+    giq_option(7, NAME, 323, Node020, NEUTRAL_REACTION);
+    giq_option(4, NAME, 324, Node010, NEUTRAL_REACTION);
+    giq_option(4, NAME, 325, Node007, NEUTRAL_REACTION);
 end
 
 procedure Node014
 begin
-    if (metarule(17, 1) == 0) then begin
+    if (town_known(1) == 0) then begin
         debug_msg("  mark_on_map(" + 1 + ")");
         mark_area_known(0, 1, 1);
     end
-    gsay_reply(84, 330);
-    giq_option(4, 84, 331, Node010, 50);
-    giq_option(4, 84, 332, Node007, 50);
+    Reply(330);
+    giq_option(4, NAME, 331, Node010, NEUTRAL_REACTION);
+    giq_option(4, NAME, 332, Node007, NEUTRAL_REACTION);
 end
 
 procedure Node015
 begin
-    gsay_reply(84, 340);
-    giq_option(4, 84, 341, Node010, 50);
-    giq_option(4, 84, 342, Node007, 50);
+    Reply(340);
+    giq_option(4, NAME, 341, Node010, NEUTRAL_REACTION);
+    giq_option(4, NAME, 342, Node007, NEUTRAL_REACTION);
 end
 
 procedure Node016
 begin
-    gsay_reply(84, 350);
-    giq_option(4, 84, 351, Node007, 50);
-    giq_option(4, 84, 352, Node018, 50);
-    giq_option(4, 84, 353, Node010, 50);
-    giq_option(7, 84, 354, Node012, 50);
+    Reply(350);
+    giq_option(4, NAME, 351, Node007, NEUTRAL_REACTION);
+    giq_option(4, NAME, 352, Node018, NEUTRAL_REACTION);
+    giq_option(4, NAME, 353, Node010, NEUTRAL_REACTION);
+    giq_option(7, NAME, 354, Node012, NEUTRAL_REACTION);
 end
 
 procedure Node017
 begin
-    gsay_reply(84, 360);
-    giq_option(4, 84, 361, Node007, 50);
-    giq_option(4, 84, 362, Node018, 50);
-    giq_option(4, 84, 363, Node010, 50);
-    giq_option(7, 84, 364, Node012, 50);
+    Reply(360);
+    giq_option(4, NAME, 361, Node007, NEUTRAL_REACTION);
+    giq_option(4, NAME, 362, Node018, NEUTRAL_REACTION);
+    giq_option(4, NAME, 363, Node010, NEUTRAL_REACTION);
+    giq_option(7, NAME, 364, Node012, NEUTRAL_REACTION);
 end
 
 procedure Node018
 begin
-    gsay_message(84, 370, 50);
+    gsay_message(NAME, 370, NEUTRAL_REACTION);
 end
 
 procedure Node019
 begin
-    gsay_reply(84, 380);
-    giq_option(4, 84, 381, Node007, 50);
-    giq_option(4, 84, 382, Node018, 50);
-    giq_option(4, 84, 383, Node010, 50);
-    giq_option(7, 84, 384, Node012, 50);
+    Reply(380);
+    giq_option(4, NAME, 381, Node007, NEUTRAL_REACTION);
+    giq_option(4, NAME, 382, Node018, NEUTRAL_REACTION);
+    giq_option(4, NAME, 383, Node010, NEUTRAL_REACTION);
+    giq_option(7, NAME, 384, Node012, NEUTRAL_REACTION);
 end
 
 procedure Node020
 begin
-    gsay_reply(84, 390);
-    giq_option(4, 84, 391, Node010, 50);
-    giq_option(4, 84, 392, Node007, 50);
+    Reply(390);
+    giq_option(4, NAME, 391, Node010, NEUTRAL_REACTION);
+    giq_option(4, NAME, 392, Node007, NEUTRAL_REACTION);
 end
 
 procedure Node021
 begin
-    gsay_reply(84, 400);
-    giq_option(4, 84, 401, Node010, 50);
-    giq_option(4, 84, 402, Node007, 50);
+    Reply(400);
+    giq_option(4, NAME, 401, Node010, NEUTRAL_REACTION);
+    giq_option(4, NAME, 402, Node007, NEUTRAL_REACTION);
 end
-
