diff --git a/Mods/MegaMod/Original/KCNED.ssl b/Mods/MegaMod/ReDefine/KCNED.ssl
index f6861f0..b7a3852 100644
--- a/Mods/MegaMod/Original/KCNED.ssl
+++ b/Mods/MegaMod/ReDefine/KCNED.ssl
@@ -198,48 +198,48 @@ end
 
 procedure EvaluateKarma
 begin
-    set_global_var(37, 0);
-    set_global_var(38, 0);
-    set_global_var(39, 0);
-    set_global_var(40, 0);
-    set_global_var(41, 0);
-    set_global_var(42, 0);
-    set_global_var(43, 0);
-    set_global_var(44, 0);
-    set_global_var(45, 0);
-    if (global_var(0) >= 1000) then begin
-        set_global_var(37, 1);
+    set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+    set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+    set_global_var(GVAR_KARMA_DEFENDER, 0);
+    set_global_var(GVAR_KARMA_WANDERER, 0);
+    set_global_var(GVAR_KARMA_BETRAYER, 0);
+    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+    set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+    set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+    if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
     end
     else begin
-        if (global_var(0) >= 750) then begin
-            set_global_var(38, 1);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+            set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
         end
         else begin
-            if (global_var(0) >= 500) then begin
-                set_global_var(39, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
             end
             else begin
-                if (global_var(0) >= 250) then begin
-                    set_global_var(40, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                    set_global_var(GVAR_KARMA_DEFENDER, 1);
                 end
                 else begin
-                    if (global_var(0) > -250) then begin
-                        set_global_var(41, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                        set_global_var(GVAR_KARMA_WANDERER, 1);
                     end
                     else begin
-                        if (global_var(0) > -500) then begin
-                            set_global_var(42, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                            set_global_var(GVAR_KARMA_BETRAYER, 1);
                         end
                         else begin
-                            if (global_var(0) > -750) then begin
-                                set_global_var(43, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                             end
                             else begin
-                                if (global_var(0) > -1000) then begin
-                                    set_global_var(44, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                    set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                 end
                                 else begin
-                                    set_global_var(45, 1);
+                                    set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                 end
                             end
                         end
@@ -252,22 +252,22 @@ end
 
 procedure CheckRep
 begin
-    if (((global_var(4) + global_var(5)) >= 25) and ((global_var(4) > (3 * global_var(5))) or (global_var(2) == 1)) and (global_var(1) == 0)) then begin
-        set_global_var(2, 1);
-        set_global_var(3, 0);
+    if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(4) > (3 * global_var(GVAR_GOOD_MONSTER))) or (global_var(GVAR_CHAMPION_REPUTATION) == 1)) and (global_var(GVAR_CHILDKILLER_REPUTATION) == 0)) then begin
+        set_global_var(GVAR_CHAMPION_REPUTATION, 1);
+        set_global_var(GVAR_BERSERKER_REPUTATION, 0);
     end
-    if (((global_var(4) + global_var(5)) >= 25) and ((global_var(5) > (2 * global_var(4))) or (global_var(3) == 1))) then begin
-        set_global_var(3, 1);
-        set_global_var(2, 0);
+    if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
+        set_global_var(GVAR_BERSERKER_REPUTATION, 1);
+        set_global_var(GVAR_CHAMPION_REPUTATION, 0);
     end
 end
 
 procedure critter_p_proc
 begin
-    if ((cur_map_index == 9) and active and (map_var(19) == 1) and obj_can_see_obj(self_obj, dude_obj)) then begin
-        attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+    if ((CUR_MAP_KLADWTWN) and active and (map_var(19) == 1) and obj_can_see_obj(self_obj, dude_obj)) then begin
+        attack(dude_obj);
     end
-    if ((cur_map_index == 10) and (map_var(19) != 0) and (lock_out_event == 0)) then begin
+    if ((CUR_MAP_KLAMALL) and (map_var(19) != 0) and (lock_out_event == 0)) then begin
         lock_out_event := 1;
         add_timer_event(self_obj, 20, 0);
     end
@@ -281,8 +281,8 @@ begin
     end
     else begin
         if (fixed_param == 1) then begin
-            display_msg(message_str(1341, 800));
-            add_mult_objs_to_inven(dude_obj, create_object_sid(541, 0, 0, -1), 1);
+            display_mstr(800);
+            add_mult_objs_to_inven(dude_obj, create_object(541, 0, 0), 1);
             gfade_in(3);
             destroy_object(map_var(19));
             set_map_var(19, 0);
@@ -293,7 +293,7 @@ end
 
 procedure pickup_p_proc
 begin
-    if ((cur_map_index == 9) and (source_obj == dude_obj)) then begin
+    if ((CUR_MAP_KLADWTWN) and (source_obj == dude_obj)) then begin
         set_map_var(19, 1);
     end
 end
@@ -301,18 +301,18 @@ end
 procedure destroy_p_proc
 begin
     variable LVar0 := 0;
-    set_global_var(691, 1);
-    set_global_var(4, global_var(4) + 1);
+    set_global_var(GVAR_MODOC_JONNY_PIP, 1);
+    set_global_var(GVAR_BAD_MONSTER, global_var(GVAR_BAD_MONSTER) + 1);
     call CheckRep();
-    if ((cur_map_index == 9) and (global_var(692) == 0)) then begin
-        LVar0 := create_object_sid(541, 0, 0, -1);
+    if ((CUR_MAP_KLADWTWN) and (global_var(GVAR_NEW_RENO_FLAG_4) == 0)) then begin
+        LVar0 := create_object(541, 0, 0);
         add_mult_objs_to_inven(self_obj, LVar0, 1);
     end
 end
 
 procedure damage_p_proc
 begin
-    if ((cur_map_index == 9) and (source_obj == dude_obj)) then begin
+    if ((CUR_MAP_KLADWTWN) and (source_obj == dude_obj)) then begin
         set_map_var(19, 1);
     end
 end
@@ -324,7 +324,7 @@ end
 
 procedure map_enter_p_proc
 begin
-    if (cur_map_index == 9) then begin
+    if (CUR_MAP_KLADWTWN) then begin
         active := 1;
         set_obj_visibility(self_obj, 0);
     end
@@ -332,24 +332,24 @@ begin
         active := 0;
         set_obj_visibility(self_obj, 1);
     end
-    critter_add_trait(self_obj, 1, 6, 1);
-    critter_add_trait(self_obj, 1, 5, 183);
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 1);
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_PEASANT_KAMAKAZI);
 end
 
 procedure look_at_p_proc
 begin
     script_overrides;
-    display_msg(message_str(1341, 100));
+    display_mstr(100);
 end
 
 procedure description_p_proc
 begin
     script_overrides;
-    if (get_critter_stat(dude_obj, 34) == 0) then begin
-        display_msg(message_str(1341, 101));
+    if (dude_gender == 0) then begin
+        display_mstr(101);
     end
     else begin
-        display_msg(message_str(1341, 102));
+        display_mstr(102);
     end
 end
 
@@ -357,7 +357,7 @@ procedure combat_p_proc
 begin
     variable LVar0 := 0;
     if (local_var(13) == 0) then begin
-        LVar0 := create_object_sid(21, 0, 0, -1);
+        LVar0 := create_object(21, 0, 0);
         add_mult_objs_to_inven(self_obj, LVar0, 1);
         set_local_var(13, 1);
     end
@@ -366,395 +366,395 @@ end
 procedure Node000
 begin
     set_local_var(10, 1);
-    gsay_reply(1341, 200);
-    giq_option(2, 1341, 202, "Node001", 50);
-    if (get_critter_stat(dude_obj, 4) <= 3) then begin
-        giq_option(2, 1341, 204, "Node002", 50);
+    Reply(200);
+    giq_option(2, NAME, 202, "Node001", NEUTRAL_REACTION);
+    if (dude_is_stupid) then begin
+        giq_option(2, NAME, 204, "Node002", NEUTRAL_REACTION);
     end
-    giq_option(4, 1341, 206, "Node002", 50);
-    giq_option(2, 1341, 208, "Node003", 50);
-    if (get_critter_stat(dude_obj, 4) <= 3) then begin
-        giq_option(2, 1341, 210, "Node003", 50);
+    giq_option(4, NAME, 206, "Node002", NEUTRAL_REACTION);
+    giq_option(2, NAME, 208, "Node003", NEUTRAL_REACTION);
+    if (dude_is_stupid) then begin
+        giq_option(2, NAME, 210, "Node003", NEUTRAL_REACTION);
     end
-    giq_option(4, 1341, 212, "Node003", 50);
-    if ((get_critter_stat(dude_obj, 4) <= 3) and (global_var(667) == 11) and (global_var(692) == 0)) then begin
-        giq_option(2, 1341, 214, "Node004", 50);
+    giq_option(4, NAME, 212, "Node003", NEUTRAL_REACTION);
+    if ((dude_is_stupid) and (global_var(GVAR_KCSAJAG_SKINS) == 11) and (global_var(GVAR_NEW_RENO_FLAG_4) == 0)) then begin
+        giq_option(2, NAME, 214, "Node004", NEUTRAL_REACTION);
     end
-    if ((global_var(667) == 11) and (global_var(692) == 0)) then begin
-        giq_option(4, 1341, 216, "Node004", 50);
+    if ((global_var(GVAR_KCSAJAG_SKINS) == 11) and (global_var(GVAR_NEW_RENO_FLAG_4) == 0)) then begin
+        giq_option(4, NAME, 216, "Node004", NEUTRAL_REACTION);
     end
-    giq_option(-1, 1341, 218, "Node005", 50);
-    giq_option(2, 1341, 220, "Node999", 50);
+    giq_option(-1, NAME, 218, "Node005", NEUTRAL_REACTION);
+    giq_option(2, NAME, 220, "Node999", NEUTRAL_REACTION);
 end
 
 procedure Node001
 begin
-    gsay_reply(1341, 222);
-    giq_option(-3, 1341, 224, "Node002", 50);
-    giq_option(4, 1341, 226, "Node002", 50);
-    giq_option(1, 1341, 228, "Node003", 50);
-    giq_option(-3, 1341, 230, "Node003", 50);
-    giq_option(4, 1341, 232, "Node003", 50);
-    if ((global_var(667) == 11) and (global_var(692) == 0)) then begin
-        giq_option(-3, 1341, 234, "Node004", 50);
+    Reply(222);
+    giq_option(-3, NAME, 224, "Node002", NEUTRAL_REACTION);
+    giq_option(4, NAME, 226, "Node002", NEUTRAL_REACTION);
+    giq_option(1, NAME, 228, "Node003", NEUTRAL_REACTION);
+    giq_option(-3, NAME, 230, "Node003", NEUTRAL_REACTION);
+    giq_option(4, NAME, 232, "Node003", NEUTRAL_REACTION);
+    if ((global_var(GVAR_KCSAJAG_SKINS) == 11) and (global_var(GVAR_NEW_RENO_FLAG_4) == 0)) then begin
+        giq_option(-3, NAME, 234, "Node004", NEUTRAL_REACTION);
     end
-    if ((global_var(667) == 11) and (global_var(692) == 0)) then begin
-        giq_option(4, 1341, 236, "Node004", 50);
+    if ((global_var(GVAR_KCSAJAG_SKINS) == 11) and (global_var(GVAR_NEW_RENO_FLAG_4) == 0)) then begin
+        giq_option(4, NAME, 236, "Node004", NEUTRAL_REACTION);
     end
-    giq_option(1, 1341, 238, "Node999", 50);
+    giq_option(1, NAME, 238, "Node999", NEUTRAL_REACTION);
 end
 
 procedure Node002
 begin
     set_local_var(11, 1);
-    gsay_reply(1341, 240);
+    Reply(240);
     if (obj_is_carrying_obj_pid(dude_obj, 125) >= 1) then begin
-        giq_option(1, 1341, 242, "Node007", 50);
+        giq_option(1, NAME, 242, "Node007", NEUTRAL_REACTION);
     end
-    giq_option(1, 1341, 244, "Node999", 50);
-    giq_option(1, 1341, 246, "Node999", 50);
+    giq_option(1, NAME, 244, "Node999", NEUTRAL_REACTION);
+    giq_option(1, NAME, 246, "Node999", NEUTRAL_REACTION);
 end
 
 procedure Node003
 begin
     set_local_var(12, 1);
-    gsay_reply(1341, 248);
-    if ((global_var(667) == 11) and (global_var(692) == 0)) then begin
-        giq_option(4, 1341, 250, "Node039", 50);
+    Reply(248);
+    if ((global_var(GVAR_KCSAJAG_SKINS) == 11) and (global_var(GVAR_NEW_RENO_FLAG_4) == 0)) then begin
+        giq_option(4, NAME, 250, "Node039", NEUTRAL_REACTION);
     end
-    if ((global_var(667) == 11) and (global_var(692) == 0)) then begin
-        giq_option(-3, 1341, 252, "Node039", 50);
+    if ((global_var(GVAR_KCSAJAG_SKINS) == 11) and (global_var(GVAR_NEW_RENO_FLAG_4) == 0)) then begin
+        giq_option(-3, NAME, 252, "Node039", NEUTRAL_REACTION);
     end
-    giq_option(4, 1341, 254, "Node999", 50);
-    giq_option(-3, 1341, 256, "Node999", 50);
+    giq_option(4, NAME, 254, "Node999", NEUTRAL_REACTION);
+    giq_option(-3, NAME, 256, "Node999", NEUTRAL_REACTION);
 end
 
 procedure Node004
 begin
     set_local_var(12, 1);
-    gsay_reply(1341, 258);
-    if ((global_var(667) == 11) and (global_var(692) == 0)) then begin
-        giq_option(4, 1341, 260, "Node039", 50);
+    Reply(258);
+    if ((global_var(GVAR_KCSAJAG_SKINS) == 11) and (global_var(GVAR_NEW_RENO_FLAG_4) == 0)) then begin
+        giq_option(4, NAME, 260, "Node039", NEUTRAL_REACTION);
     end
-    if ((global_var(667) == 11) and (global_var(692) == 0)) then begin
-        giq_option(-3, 1341, 262, "Node039", 50);
+    if ((global_var(GVAR_KCSAJAG_SKINS) == 11) and (global_var(GVAR_NEW_RENO_FLAG_4) == 0)) then begin
+        giq_option(-3, NAME, 262, "Node039", NEUTRAL_REACTION);
     end
-    giq_option(4, 1341, 264, "Node999", 50);
-    giq_option(-3, 1341, 266, "Node999", 50);
+    giq_option(4, NAME, 264, "Node999", NEUTRAL_REACTION);
+    giq_option(-3, NAME, 266, "Node999", NEUTRAL_REACTION);
 end
 
 procedure Node005
 begin
-    gsay_reply(1341, 268);
-    giq_option(1, 1341, 270, "Node006", 50);
+    Reply(268);
+    giq_option(1, NAME, 270, "Node006", NEUTRAL_REACTION);
 end
 
 procedure Node006
 begin
     set_local_var(12, 1);
-    gsay_reply(1341, 272);
-    giq_option(1, 1341, 274, "Node999", 50);
+    Reply(272);
+    giq_option(1, NAME, 274, "Node999", NEUTRAL_REACTION);
 end
 
 procedure Node007
 begin
     set_local_var(11, 2);
-    gsay_reply(1341, 276);
-    giq_option(1, 1341, 278, "Node008", 50);
-    giq_option(1, 1341, 280, "Node999", 50);
+    Reply(276);
+    giq_option(1, NAME, 278, "Node008", NEUTRAL_REACTION);
+    giq_option(1, NAME, 280, "Node999", NEUTRAL_REACTION);
 end
 
 procedure Node008
 begin
-    gsay_reply(1341, 282);
-    giq_option(1, 1341, message_str(1341, 284) + obj_name(dude_obj) + message_str(1341, 285), "Node009", 50);
+    Reply(282);
+    giq_option(1, NAME, mstr(284) + dude_name + mstr(285), "Node009", NEUTRAL_REACTION);
 end
 
 procedure Node009
 begin
-    gsay_reply(1341, 288);
-    giq_option(-3, 1341, 290, "Node015", 50);
-    giq_option(4, 1341, 292, "Node015", 50);
-    if ((global_var(667) == 11) and (global_var(692) == 0)) then begin
-        giq_option(-3, 1341, 294, "Node010", 50);
+    Reply(288);
+    giq_option(-3, NAME, 290, "Node015", NEUTRAL_REACTION);
+    giq_option(4, NAME, 292, "Node015", NEUTRAL_REACTION);
+    if ((global_var(GVAR_KCSAJAG_SKINS) == 11) and (global_var(GVAR_NEW_RENO_FLAG_4) == 0)) then begin
+        giq_option(-3, NAME, 294, "Node010", NEUTRAL_REACTION);
     end
-    if ((global_var(667) == 11) and (global_var(692) == 0)) then begin
-        giq_option(4, 1341, 296, "Node010", 50);
+    if ((global_var(GVAR_KCSAJAG_SKINS) == 11) and (global_var(GVAR_NEW_RENO_FLAG_4) == 0)) then begin
+        giq_option(4, NAME, 296, "Node010", NEUTRAL_REACTION);
     end
-    giq_option(1, 1341, 297, "Node999", 50);
+    giq_option(1, NAME, 297, "Node999", NEUTRAL_REACTION);
 end
 
 procedure Node010
 begin
-    gsay_reply(1341, 298);
-    giq_option(4, 1341, 300, "Node011", 50);
-    giq_option(-3, 1341, 302, "Node011", 50);
+    Reply(298);
+    giq_option(4, NAME, 300, "Node011", NEUTRAL_REACTION);
+    giq_option(-3, NAME, 302, "Node011", NEUTRAL_REACTION);
 end
 
 procedure Node011
 begin
-    gsay_reply(1341, 304);
-    giq_option(4, 1341, 306, "Node016", 50);
-    giq_option(-3, 1341, 308, "Node016", 50);
-    giq_option(1, 1341, 310, "Node017", 50);
-    giq_option(1, 1341, 312, "Node012", 50);
+    Reply(304);
+    giq_option(4, NAME, 306, "Node016", NEUTRAL_REACTION);
+    giq_option(-3, NAME, 308, "Node016", NEUTRAL_REACTION);
+    giq_option(1, NAME, 310, "Node017", NEUTRAL_REACTION);
+    giq_option(1, NAME, 312, "Node012", NEUTRAL_REACTION);
 end
 
 procedure Node012
 begin
-    gsay_reply(1341, 314);
-    giq_option(4, 1341, 316, "Node013", 50);
-    giq_option(4, 1341, 318, "Node013", 50);
-    giq_option(-3, 1341, 320, "Node013", 50);
-    giq_option(1, 1341, 322, "Node999", 50);
+    Reply(314);
+    giq_option(4, NAME, 316, "Node013", NEUTRAL_REACTION);
+    giq_option(4, NAME, 318, "Node013", NEUTRAL_REACTION);
+    giq_option(-3, NAME, 320, "Node013", NEUTRAL_REACTION);
+    giq_option(1, NAME, 322, "Node999", NEUTRAL_REACTION);
 end
 
 procedure Node013
 begin
     set_local_var(12, 1);
-    gsay_reply(1341, 324);
-    giq_option(-3, 1341, 326, "Node014", 50);
-    giq_option(4, 1341, 328, "Node014", 50);
-    giq_option(1, 1341, 330, "Node999", 50);
+    Reply(324);
+    giq_option(-3, NAME, 326, "Node014", NEUTRAL_REACTION);
+    giq_option(4, NAME, 328, "Node014", NEUTRAL_REACTION);
+    giq_option(1, NAME, 330, "Node999", NEUTRAL_REACTION);
 end
 
 procedure Node014
 begin
-    if (has_trait(2, dude_obj, 2) or (get_critter_stat(dude_obj, 0) <= 3)) then begin
-        gsay_reply(1341, 332);
+    if (has_trait(TRAIT_TRAIT, dude_obj, TRAIT_small_frame) or (dude_strength <= 3)) then begin
+        Reply(332);
     end
     else begin
-        gsay_reply(1341, 334);
+        Reply(334);
     end
-    giq_option(4, 1341, 336, "NodeCombat", 50);
-    giq_option(-3, 1341, 338, "NodeCombat", 50);
-    giq_option(1, 1341, 340, "Node999", 50);
+    giq_option(4, NAME, 336, "NodeCombat", NEUTRAL_REACTION);
+    giq_option(-3, NAME, 338, "NodeCombat", NEUTRAL_REACTION);
+    giq_option(1, NAME, 340, "Node999", NEUTRAL_REACTION);
 end
 
 procedure Node015
 begin
-    gsay_reply(1341, 342);
-    if ((global_var(667) == 11) and (global_var(692) == 0)) then begin
-        giq_option(-3, 1341, 344, Node010, 50);
+    Reply(342);
+    if ((global_var(GVAR_KCSAJAG_SKINS) == 11) and (global_var(GVAR_NEW_RENO_FLAG_4) == 0)) then begin
+        giq_option(-3, NAME, 344, Node010, NEUTRAL_REACTION);
     end
-    if ((global_var(667) == 11) and (global_var(692) == 0)) then begin
-        giq_option(4, 1341, 346, Node010, 50);
+    if ((global_var(GVAR_KCSAJAG_SKINS) == 11) and (global_var(GVAR_NEW_RENO_FLAG_4) == 0)) then begin
+        giq_option(4, NAME, 346, Node010, NEUTRAL_REACTION);
     end
-    giq_option(1, 1341, 347, "Node999", 50);
+    giq_option(1, NAME, 347, "Node999", NEUTRAL_REACTION);
 end
 
 procedure Node016
 begin
-    gsay_reply(1341, 348);
-    giq_option(1, 1341, 350, Node012, 50);
-    giq_option(1, 1341, 352, "Node999", 50);
+    Reply(348);
+    giq_option(1, NAME, 350, Node012, NEUTRAL_REACTION);
+    giq_option(1, NAME, 352, "Node999", NEUTRAL_REACTION);
 end
 
 procedure Node017
 begin
     set_local_var(12, 1);
-    gsay_reply(1341, 354);
-    giq_option(4, 1341, 356, Node013, 50);
-    giq_option(-3, 1341, 358, Node013, 50);
-    giq_option(1, 1341, 360, "Node999", 50);
+    Reply(354);
+    giq_option(4, NAME, 356, Node013, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 358, Node013, NEUTRAL_REACTION);
+    giq_option(1, NAME, 360, "Node999", NEUTRAL_REACTION);
 end
 
 procedure Node018
 begin
-    gsay_reply(1341, 362);
+    Reply(362);
     if ((local_var(11) == 1) and (obj_is_carrying_obj_pid(dude_obj, 125) >= 1)) then begin
-        giq_option(1, 1341, 364, Node007, 50);
+        giq_option(1, NAME, 364, Node007, NEUTRAL_REACTION);
     end
-    if ((global_var(667) == 11) and (global_var(692) == 0)) then begin
-        giq_option(4, 1341, 366, "Node039", 50);
+    if ((global_var(GVAR_KCSAJAG_SKINS) == 11) and (global_var(GVAR_NEW_RENO_FLAG_4) == 0)) then begin
+        giq_option(4, NAME, 366, "Node039", NEUTRAL_REACTION);
     end
-    if ((global_var(667) == 11) and (global_var(692) == 0)) then begin
-        giq_option(-3, 1341, 368, "Node039", 50);
+    if ((global_var(GVAR_KCSAJAG_SKINS) == 11) and (global_var(GVAR_NEW_RENO_FLAG_4) == 0)) then begin
+        giq_option(-3, NAME, 368, "Node039", NEUTRAL_REACTION);
     end
-    giq_option(1, 1341, 370, "Node999", 50);
+    giq_option(1, NAME, 370, "Node999", NEUTRAL_REACTION);
 end
 
 procedure Node019
 begin
-    gsay_reply(1341, 372);
-    if ((global_var(667) == 11) and (global_var(692) == 0)) then begin
-        giq_option(4, 1341, 374, "Node020", 50);
+    Reply(372);
+    if ((global_var(GVAR_KCSAJAG_SKINS) == 11) and (global_var(GVAR_NEW_RENO_FLAG_4) == 0)) then begin
+        giq_option(4, NAME, 374, "Node020", NEUTRAL_REACTION);
     end
-    if ((get_critter_stat(dude_obj, 4) > 1) and (global_var(667) == 11) and (global_var(692) == 0)) then begin
-        giq_option(-3, 1341, 376, "Node020", 50);
+    if ((dude_iq > 1) and (global_var(GVAR_KCSAJAG_SKINS) == 11) and (global_var(GVAR_NEW_RENO_FLAG_4) == 0)) then begin
+        giq_option(-3, NAME, 376, "Node020", NEUTRAL_REACTION);
     end
-    giq_option(-1, 1341, 378, "Node999", 50);
-    giq_option(2, 1341, 380, "Node999", 50);
+    giq_option(-1, NAME, 378, "Node999", NEUTRAL_REACTION);
+    giq_option(2, NAME, 380, "Node999", NEUTRAL_REACTION);
 end
 
 procedure Node020
 begin
-    if (has_trait(2, dude_obj, 2) or (get_critter_stat(dude_obj, 0) <= 3)) then begin
-        gsay_reply(1341, 382);
+    if (has_trait(TRAIT_TRAIT, dude_obj, TRAIT_small_frame) or (dude_strength <= 3)) then begin
+        Reply(382);
     end
     else begin
-        gsay_reply(1341, 384);
+        Reply(384);
     end
-    giq_option(4, 1341, 386, "NodeCombat", 50);
-    giq_option(-3, 1341, 388, "NodeCombat", 50);
-    giq_option(1, 1341, 390, "Node999", 50);
+    giq_option(4, NAME, 386, "NodeCombat", NEUTRAL_REACTION);
+    giq_option(-3, NAME, 388, "NodeCombat", NEUTRAL_REACTION);
+    giq_option(1, NAME, 390, "Node999", NEUTRAL_REACTION);
 end
 
 procedure Node021
 begin
     set_local_var(10, 1);
-    gsay_reply(1341, 392);
-    giq_option(2, 1341, 394, "Node022", 50);
-    if ((global_var(667) == 11) and (global_var(692) == 0)) then begin
-        giq_option(-1, 1341, 396, "Node033", 50);
+    Reply(392);
+    giq_option(2, NAME, 394, "Node022", NEUTRAL_REACTION);
+    if ((global_var(GVAR_KCSAJAG_SKINS) == 11) and (global_var(GVAR_NEW_RENO_FLAG_4) == 0)) then begin
+        giq_option(-1, NAME, 396, "Node033", NEUTRAL_REACTION);
     end
-    giq_option(-1, 1341, 397, "Node999", 50);
+    giq_option(-1, NAME, 397, "Node999", NEUTRAL_REACTION);
 end
 
 procedure Node022
 begin
-    gsay_reply(1341, 398);
-    giq_option(1, 1341, message_str(1341, 400) + obj_name(dude_obj) + message_str(1341, 401), "Node023", 50);
+    Reply(398);
+    giq_option(1, NAME, mstr(400) + dude_name + mstr(401), "Node023", NEUTRAL_REACTION);
 end
 
 procedure Node023
 begin
-    gsay_reply(1341, 402);
-    if ((global_var(667) == 11) and (global_var(692) == 0)) then begin
-        giq_option(1, 1341, 404, "Node024", 50);
+    Reply(402);
+    if ((global_var(GVAR_KCSAJAG_SKINS) == 11) and (global_var(GVAR_NEW_RENO_FLAG_4) == 0)) then begin
+        giq_option(1, NAME, 404, "Node024", NEUTRAL_REACTION);
     end
-    giq_option(1, 1341, 406, "Node999", 50);
+    giq_option(1, NAME, 406, "Node999", NEUTRAL_REACTION);
 end
 
 procedure Node024
 begin
-    gsay_reply(1341, 408);
-    giq_option(4, 1341, 410, "Node025", 50);
-    giq_option(-3, 1341, 412, "Node025", 50);
+    Reply(408);
+    giq_option(4, NAME, 410, "Node025", NEUTRAL_REACTION);
+    giq_option(-3, NAME, 412, "Node025", NEUTRAL_REACTION);
 end
 
 procedure Node025
 begin
-    gsay_reply(1341, 414);
-    giq_option(4, 1341, 416, "Node026", 50);
-    giq_option(-3, 1341, 418, "Node026", 50);
+    Reply(414);
+    giq_option(4, NAME, 416, "Node026", NEUTRAL_REACTION);
+    giq_option(-3, NAME, 418, "Node026", NEUTRAL_REACTION);
 end
 
 procedure Node026
 begin
-    gsay_reply(1341, 420);
-    giq_option(4, 1341, 422, "Node027", 50);
-    giq_option(-3, 1341, 424, "Node027", 50);
+    Reply(420);
+    giq_option(4, NAME, 422, "Node027", NEUTRAL_REACTION);
+    giq_option(-3, NAME, 424, "Node027", NEUTRAL_REACTION);
 end
 
 procedure Node027
 begin
-    gsay_reply(1341, 426);
-    giq_option(4, 1341, 428, "Node028", 50);
-    giq_option(-3, 1341, 430, "Node028", 50);
+    Reply(426);
+    giq_option(4, NAME, 428, "Node028", NEUTRAL_REACTION);
+    giq_option(-3, NAME, 430, "Node028", NEUTRAL_REACTION);
 end
 
 procedure Node028
 begin
-    gsay_reply(1341, 432);
-    giq_option(4, 1341, 434, "Node029", 50);
-    giq_option(-3, 1341, 436, "Node029", 50);
+    Reply(432);
+    giq_option(4, NAME, 434, "Node029", NEUTRAL_REACTION);
+    giq_option(-3, NAME, 436, "Node029", NEUTRAL_REACTION);
 end
 
 procedure Node029
 begin
-    gsay_reply(1341, 438);
-    giq_option(1, 1341, 440, "Node030", 50);
+    Reply(438);
+    giq_option(1, NAME, 440, "Node030", NEUTRAL_REACTION);
 end
 
 procedure Node030
 begin
-    gsay_reply(1341, 442);
-    giq_option(1, 1341, 444, "Node031", 50);
+    Reply(442);
+    giq_option(1, NAME, 444, "Node031", NEUTRAL_REACTION);
 end
 
 procedure Node031
 begin
-    gsay_reply(1341, 446);
-    giq_option(4, 1341, 448, "Node032", 50);
-    giq_option(-3, 1341, 450, "Node032", 50);
+    Reply(446);
+    giq_option(4, NAME, 448, "Node032", NEUTRAL_REACTION);
+    giq_option(-3, NAME, 450, "Node032", NEUTRAL_REACTION);
 end
 
 procedure Node032
 begin
     set_map_var(19, 1);
-    gsay_reply(1341, 452);
-    giq_option(1, 1341, 454, "NodeCombat", 50);
+    Reply(452);
+    giq_option(1, NAME, 454, "NodeCombat", NEUTRAL_REACTION);
 end
 
 procedure Node033
 begin
-    gsay_reply(1341, 456);
-    giq_option(1, 1341, 458, "Node034", 50);
+    Reply(456);
+    giq_option(1, NAME, 458, "Node034", NEUTRAL_REACTION);
 end
 
 procedure Node034
 begin
     set_local_var(12, 1);
-    gsay_reply(1341, 460);
-    giq_option(1, 1341, 462, "Node035", 50);
-    giq_option(1, 1341, 464, "Node999", 50);
+    Reply(460);
+    giq_option(1, NAME, 462, "Node035", NEUTRAL_REACTION);
+    giq_option(1, NAME, 464, "Node999", NEUTRAL_REACTION);
 end
 
 procedure Node035
 begin
-    gsay_reply(1341, 466);
-    giq_option(1, 1341, 468, "Node036", 50);
-    giq_option(1, 1341, 470, "Node999", 50);
+    Reply(466);
+    giq_option(1, NAME, 468, "Node036", NEUTRAL_REACTION);
+    giq_option(1, NAME, 470, "Node999", NEUTRAL_REACTION);
 end
 
 procedure Node036
 begin
-    gsay_reply(1341, 472);
-    giq_option(1, 1341, 474, "NodeCombat", 50);
-    giq_option(1, 1341, 476, "Node999", 50);
+    Reply(472);
+    giq_option(1, NAME, 474, "NodeCombat", NEUTRAL_REACTION);
+    giq_option(1, NAME, 476, "Node999", NEUTRAL_REACTION);
 end
 
 procedure Node037
 begin
-    gsay_reply(1341, 478);
-    if ((global_var(667) == 11) and (global_var(692) == 0)) then begin
-        giq_option(2, 1341, 480, Node024, 50);
+    Reply(478);
+    if ((global_var(GVAR_KCSAJAG_SKINS) == 11) and (global_var(GVAR_NEW_RENO_FLAG_4) == 0)) then begin
+        giq_option(2, NAME, 480, Node024, NEUTRAL_REACTION);
     end
-    if ((global_var(667) == 11) and (global_var(692) == 0)) then begin
-        giq_option(-1, 1341, 396, Node033, 50);
+    if ((global_var(GVAR_KCSAJAG_SKINS) == 11) and (global_var(GVAR_NEW_RENO_FLAG_4) == 0)) then begin
+        giq_option(-1, NAME, 396, Node033, NEUTRAL_REACTION);
     end
-    giq_option(1, 1341, 482, "Node999", 50);
+    giq_option(1, NAME, 482, "Node999", NEUTRAL_REACTION);
 end
 
 procedure Node038
 begin
-    gsay_reply(1341, 484);
-    if ((global_var(667) == 11) and (global_var(692) == 0)) then begin
-        giq_option(1, 1341, 486, Node035, 50);
+    Reply(484);
+    if ((global_var(GVAR_KCSAJAG_SKINS) == 11) and (global_var(GVAR_NEW_RENO_FLAG_4) == 0)) then begin
+        giq_option(1, NAME, 486, Node035, NEUTRAL_REACTION);
     end
-    giq_option(1, 1341, 488, "Node999", 50);
+    giq_option(1, NAME, 488, "Node999", NEUTRAL_REACTION);
 end
 
 procedure Node039
 begin
     set_local_var(12, 1);
-    gsay_reply(1341, 490);
-    giq_option(1, 1341, 492, "Node040", 50);
+    Reply(490);
+    giq_option(1, NAME, 492, "Node040", NEUTRAL_REACTION);
 end
 
 procedure Node040
 begin
-    gsay_reply(1341, 494);
-    giq_option(4, 1341, 496, "Node041", 50);
-    giq_option(-3, 1341, 498, "Node041", 50);
+    Reply(494);
+    giq_option(4, NAME, 496, "Node041", NEUTRAL_REACTION);
+    giq_option(-3, NAME, 498, "Node041", NEUTRAL_REACTION);
 end
 
 procedure Node041
 begin
-    gsay_reply(1341, 500);
-    giq_option(4, 1341, 502, Node013, 50);
-    giq_option(-3, 1341, 504, Node013, 50);
+    Reply(500);
+    giq_option(4, NAME, 502, Node013, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 504, Node013, NEUTRAL_REACTION);
 end
 
 procedure Node999
@@ -768,7 +768,7 @@ end
 
 procedure talk_p_proc
 begin
-    if (get_critter_stat(dude_obj, 34) == 0) then begin
+    if (dude_gender == 0) then begin
         if (local_var(10) == 0) then begin
             start_gdialog(1341, self_obj, 4, -1, -1);
             gsay_start;
@@ -823,4 +823,3 @@ end
 procedure start
 begin
 end
-
