diff --git a/Mods/MegaMod/Original/KIVERT.ssl b/Mods/MegaMod/ReDefine/KIVERT.ssl
index 01850d3..2b8dbab 100644
--- a/Mods/MegaMod/Original/KIVERT.ssl
+++ b/Mods/MegaMod/ReDefine/KIVERT.ssl
@@ -141,15 +141,15 @@ end
 procedure description_p_proc
 begin
     script_overrides;
-    if (global_var(1109) == 0) then begin
-        display_msg(message_str(1718, 300));
+    if (global_var(GVAR_MARY_ANN_SON_QUEST) == 0) then begin
+        display_msg(message_str(SCRIPT_ITXCMAP, 300));
         give_exp_points(300);
-        display_msg(message_str(14, 100) + 300 + message_str(14, 101));
-        add_mult_objs_to_inven(dude_obj, create_object_sid(586, 0, 0, 1718), 1);
-        set_global_var(1109, 2);
+        display_msg(g_mstr(100) + 300 + g_mstr(101));
+        add_mult_objs_to_inven(dude_obj, create_object_sid(586, 0, 0, SCRIPT_ITXCMAP), 1);
+        set_global_var(GVAR_MARY_ANN_SON_QUEST, 2);
     end
     else begin
-        display_msg(message_str(1718, 301));
+        display_msg(message_str(SCRIPT_ITXCMAP, 301));
     end
 end
 
@@ -168,4 +168,3 @@ end
 procedure drop_p_proc
 begin
 end
-
