diff --git a/Mods/MegaMod/Original/KLADWTWN.ssl b/Mods/MegaMod/ReDefine/KLADWTWN.ssl
index a9a946c..55b3437 100644
--- a/Mods/MegaMod/Original/KLADWTWN.ssl
+++ b/Mods/MegaMod/ReDefine/KLADWTWN.ssl
@@ -149,54 +149,54 @@ end
 
 procedure map_enter_p_proc
 begin
-    if (metarule(14, 0)) then begin
-        display_msg(message_str(219, 4100));
-        k := get_critter_stat(dude_obj, 3);
+    if (map_first_run) then begin
+        display_mstr(4100);
+        k := dude_charisma;
         if (k > 5) then begin
             k := k - 5;
-            set_global_var(48, global_var(48) + k);
+            set_global_var(GVAR_TOWN_REP_KLAMATH, global_var(GVAR_TOWN_REP_KLAMATH) + k);
         end
     end
-    if (global_var(27) == 0) then begin
+    if (global_var(GVAR_LOAD_MAP_INDEX) == 0) then begin
         override_map_start(25075 % 200, 25075 / 200, 0, 3);
     end
-    if (global_var(27) == 2) then begin
+    if (global_var(GVAR_LOAD_MAP_INDEX) == 2) then begin
         override_map_start(23087 % 200, 23087 / 200, 0, 5);
     end
-    if (global_var(188) > 0) then begin
+    if (global_var(GVAR_DUNTON_DEAD) > 0) then begin
         kill_critter_type(16777276, 0);
     end
-    if (global_var(197) == 2) then begin
-        set_global_var(197, 3);
+    if (global_var(GVAR_SMILEY_STATUS) == 2) then begin
+        set_global_var(GVAR_SMILEY_STATUS, 3);
         debug_msg("Making Smiley");
-        smiley_ptr := create_object_sid(16777311, 0, 0, 83);
-        critter_add_trait(smiley_ptr, 1, 5, 24);
+        smiley_ptr := create_object_sid(16777311, 0, 0, SCRIPT_KCSMILEY);
+        critter_add_trait(smiley_ptr, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_TOUGH_CITIZEN);
         smiley_hex := 23686;
         critter_attempt_placement(smiley_ptr, smiley_hex, 0);
     end
-    if (global_var(77) == 1) then begin
-        set_global_var(77, 2);
-        k := 4 - global_var(202);
+    if (global_var(GVAR_RUSTLE_SUCCESS) == 1) then begin
+        set_global_var(GVAR_RUSTLE_SUCCESS, 2);
+        k := 4 - global_var(GVAR_TORR_BRAHMIN_KILLED);
         debug_msg("Making cows" + k);
-        cow_ptr := create_object_sid(16777226, 0, 0, 17);
-        critter_add_trait(cow_ptr, 1, 5, 6);
+        cow_ptr := create_object_sid(16777226, 0, 0, SCRIPT_ACBRAHMN);
+        critter_add_trait(cow_ptr, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_BRAHMIN);
         cow_hex := 12093;
         critter_attempt_placement(cow_ptr, cow_hex, 0);
         if (k > 1) then begin
-            cow_ptr := create_object_sid(16777226, 0, 0, 17);
-            critter_add_trait(cow_ptr, 1, 5, 6);
+            cow_ptr := create_object_sid(16777226, 0, 0, SCRIPT_ACBRAHMN);
+            critter_add_trait(cow_ptr, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_BRAHMIN);
             cow_hex := 11700;
             critter_attempt_placement(cow_ptr, cow_hex, 0);
         end
         if (k > 2) then begin
-            cow_ptr := create_object_sid(16777226, 0, 0, 17);
-            critter_add_trait(cow_ptr, 1, 5, 6);
+            cow_ptr := create_object_sid(16777226, 0, 0, SCRIPT_ACBRAHMN);
+            critter_add_trait(cow_ptr, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_BRAHMIN);
             cow_hex := 12483;
             critter_attempt_placement(cow_ptr, cow_hex, 0);
         end
         if (k > 3) then begin
-            cow_ptr := create_object_sid(16777226, 0, 0, 17);
-            critter_add_trait(cow_ptr, 1, 5, 6);
+            cow_ptr := create_object_sid(16777226, 0, 0, SCRIPT_ACBRAHMN);
+            critter_add_trait(cow_ptr, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_BRAHMIN);
             cow_hex := 13100;
             critter_attempt_placement(cow_ptr, cow_hex, 0);
         end
@@ -277,7 +277,7 @@ begin
             end
         end
     end
-    set_global_var(27, 0);
+    set_global_var(GVAR_LOAD_MAP_INDEX, 0);
 end
 
 procedure map_update_p_proc
@@ -359,4 +359,3 @@ begin
         end
     end
 end
-
