diff --git a/Mods/MegaMod/Original/KRUGER3.ssl b/Mods/MegaMod/ReDefine/KRUGER3.ssl
index 49d17b5..0da0678 100644
--- a/Mods/MegaMod/Original/KRUGER3.ssl
+++ b/Mods/MegaMod/ReDefine/KRUGER3.ssl
@@ -163,17 +163,17 @@ procedure look_at_p_proc
 begin
     script_overrides;
     if (local_var(4) == 0) then begin
-        display_msg(message_str(1634, 100));
+        display_msg(message_str(SCRIPT_KRUGER, 100));
     end
     else begin
-        display_msg(message_str(1634, 101));
+        display_msg(message_str(SCRIPT_KRUGER, 101));
     end
 end
 
 procedure description_p_proc
 begin
     script_overrides;
-    display_msg(message_str(1634, 102));
+    display_msg(message_str(SCRIPT_KRUGER, 102));
 end
 
 procedure use_skill_on_p_proc
@@ -190,15 +190,15 @@ end
 
 procedure map_enter_p_proc
 begin
-    if (not(metarule(22, 0))) then begin
-        if (global_var(1090) == 3) then begin
-            if (not(has_trait(1, self_obj, 666))) then begin
+    if (not(is_loading_game)) then begin
+        if (global_var(GVAR_CLOSE_TO_KRUGER) == 3) then begin
+            if (not(obj_is_visible_flag(self_obj))) then begin
                 add_timer_event(self_obj, game_ticks(2), 3);
                 set_obj_visibility(self_obj, 0);
             end
         end
         else begin
-            if (has_trait(1, self_obj, 666)) then begin
+            if (obj_is_visible_flag(self_obj)) then begin
                 set_obj_visibility(self_obj, 1);
             end
         end
@@ -218,23 +218,22 @@ begin
     if (elevation(self_obj) != elevation(dude_obj)) then begin
         move_to(self_obj, tile_num(self_obj), elevation(dude_obj));
     end
-    if (global_var(1090) == 3) then begin
+    if (global_var(GVAR_CLOSE_TO_KRUGER) == 3) then begin
         set_obj_visibility(self_obj, 0);
         add_timer_event(self_obj, game_ticks(2), 3);
     end
     if (obj_can_see_obj(self_obj, dude_obj) and im_ready) then begin
-        float_msg(self_obj, message_str(1634, 666), 8);
-        attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+        float_msg(self_obj, message_str(SCRIPT_KRUGER, 666), FLOAT_MSG_YELLOW);
+        attack(dude_obj);
     end
 end
 
 procedure destroy_p_proc
 begin
     variable LVar0 := 0;
-    LVar0 := create_object_sid(579, 0, 0, -1);
+    LVar0 := create_object(579, 0, 0);
     add_obj_to_inven(self_obj, LVar0);
-    set_global_var(1093, 5);
+    set_global_var(GVAR_PRIMIT_ATTACKED, 5);
     return 0;
     LVar0 := 0;
 end
-
