diff --git a/Mods/MegaMod/Original/KSBRLDTP.ssl b/Mods/MegaMod/ReDefine/KSBRLDTP.ssl
index 4aafcf4..eddc208 100644
--- a/Mods/MegaMod/Original/KSBRLDTP.ssl
+++ b/Mods/MegaMod/ReDefine/KSBRLDTP.ssl
@@ -26,7 +26,7 @@ begin
     end
     else begin
         if (local_var(0) == 0) then begin
-            display_msg(message_str(1334, 104));
+            display_mstr(104);
         end
     end
 end
@@ -41,12 +41,12 @@ begin
         Perception_Check := do_check(dude_obj, 1, 0);
         if (is_success(Perception_Check)) then begin
             script_overrides;
-            display_msg(message_str(1334, 100));
+            display_mstr(100);
         end
     end
     else begin
         script_overrides;
-        display_msg(message_str(1334, 102));
+        display_mstr(102);
     end
 end
 
@@ -65,8 +65,8 @@ begin
             LVar0 := obj_carrying_pid_obj(dude_obj, 127);
             rm_mult_objs_from_inven(dude_obj, LVar0, 1);
             destroy_object(LVar0);
-            display_msg(message_str(1334, 110));
-            anim(dude_obj, 1000, rotation_to_tile(tile_num(dude_obj), tile_num(self_obj)));
+            display_mstr(110);
+            anim(dude_obj, 1000, rotation_to_tile(dude_tile, tile_num(self_obj)));
             reg_anim_func(2, dude_obj);
             reg_anim_func(1, 1);
             reg_anim_animate(dude_obj, 10, -1);
@@ -86,4 +86,3 @@ procedure damage_p_proc
 begin
     script_overrides;
 end
-
