diff --git a/Mods/MegaMod/Original/KSBRSTIL.ssl b/Mods/MegaMod/ReDefine/KSBRSTIL.ssl
index 0866421..e8a2da1 100644
--- a/Mods/MegaMod/Original/KSBRSTIL.ssl
+++ b/Mods/MegaMod/ReDefine/KSBRSTIL.ssl
@@ -151,23 +151,23 @@ begin
             roll_result := roll_vs_skill(dude_obj, 13, 0);
             if (is_success(roll_result)) then begin
                 set_local_var(0, 2);
-                valve_ptr := create_object_sid(547, 0, 0, -1);
+                valve_ptr := create_object(547, 0, 0);
                 add_obj_to_inven(dude_obj, valve_ptr);
-                display_msg(message_str(1336, 124));
+                display_mstr(124);
             end
             else begin
                 if (is_critical(roll_result)) then begin
                     set_local_var(0, 3);
-                    display_msg(message_str(1336, 122));
+                    display_mstr(122);
                 end
                 else begin
-                    display_msg(message_str(1336, 126));
+                    display_mstr(126);
                 end
             end
         end
         else begin
             if (local_var(0) == 0) then begin
-                display_msg(message_str(1336, 120));
+                display_mstr(120);
             end
         end
     end
@@ -176,7 +176,7 @@ end
 procedure look_at_p_proc
 begin
     script_overrides;
-    display_msg(message_str(1336, 100));
+    display_mstr(100);
 end
 
 procedure description_p_proc
@@ -186,18 +186,18 @@ begin
         roll_result := do_check(dude_obj, 1, 0);
         if (is_success(roll_result)) then begin
             set_local_var(0, 1);
-            display_msg(message_str(1336, 110));
+            display_mstr(110);
         end
         else begin
-            display_msg(message_str(1336, 112));
+            display_mstr(112);
         end
     end
     else begin
         if (local_var(0) == 1) then begin
-            display_msg(message_str(1336, 104));
+            display_mstr(104);
         end
         else begin
-            display_msg(message_str(1336, 102));
+            display_mstr(102);
         end
     end
 end
@@ -206,7 +206,7 @@ procedure use_skill_on_p_proc
 begin
     script_overrides;
     if (action_being_used == 13) then begin
-        display_msg(message_str(1336, 130));
+        display_mstr(130);
     end
 end
 
@@ -224,17 +224,17 @@ begin
             roll_result := roll_vs_skill(dude_obj, 13, 20);
             if (is_success(roll_result)) then begin
                 set_local_var(0, 2);
-                valve_ptr := create_object_sid(547, 0, 0, -1);
+                valve_ptr := create_object(547, 0, 0);
                 add_obj_to_inven(dude_obj, valve_ptr);
-                display_msg(message_str(1336, 124));
+                display_mstr(124);
             end
             else begin
                 if (is_critical(roll_result)) then begin
                     set_local_var(0, 3);
-                    display_msg(message_str(1336, 122));
+                    display_mstr(122);
                 end
                 else begin
-                    display_msg(message_str(1336, 126));
+                    display_mstr(126);
                 end
             end
         end
@@ -243,21 +243,20 @@ begin
                 roll_result := roll_vs_skill(dude_obj, 13, 40);
                 if (is_success(roll_result)) then begin
                     set_local_var(0, 2);
-                    valve_ptr := create_object_sid(547, 0, 0, -1);
+                    valve_ptr := create_object(547, 0, 0);
                     add_obj_to_inven(dude_obj, valve_ptr);
-                    display_msg(message_str(1336, 124));
+                    display_mstr(124);
                 end
                 else begin
                     if (is_critical(roll_result)) then begin
                         set_local_var(0, 3);
-                        display_msg(message_str(1336, 122));
+                        display_mstr(122);
                     end
                     else begin
-                        display_msg(message_str(1336, 126));
+                        display_mstr(126);
                     end
                 end
             end
         end
     end
 end
-
