diff --git a/Mods/MegaMod/Original/KSCORVEG.ssl b/Mods/MegaMod/ReDefine/KSCORVEG.ssl
index e4c8d68..bb16eec 100644
--- a/Mods/MegaMod/Original/KSCORVEG.ssl
+++ b/Mods/MegaMod/ReDefine/KSCORVEG.ssl
@@ -152,18 +152,18 @@ procedure use_p_proc
 begin
     if (local_var(0) == 0) then begin
         set_local_var(0, 1);
-        item := create_object_sid(254, 0, 0, -1);
+        item := create_object(254, 0, 0);
         add_obj_to_inven(dude_obj, item);
-        display_msg(message_str(310, 130));
+        display_mstr(130);
         give_exp_points(200);
-        display_msg(message_str(14, 100) + 200 + message_str(14, 101));
+        display_msg(g_mstr(100) + 200 + g_mstr(101));
     end
 end
 
 procedure look_at_p_proc
 begin
     script_overrides;
-    display_msg(message_str(310, 100));
+    display_mstr(100);
 end
 
 procedure description_p_proc
@@ -195,18 +195,18 @@ begin
     repair_check := roll_vs_skill(dude_obj, 13, 0);
     if (is_success(repair_check)) then begin
         if (is_critical(repair_check)) then begin
-            display_msg(message_str(310, 120));
+            display_mstr(120);
         end
         else begin
-            display_msg(message_str(310, 121));
+            display_mstr(121);
         end
     end
     else begin
         if (is_critical(repair_check)) then begin
-            display_msg(message_str(310, 122));
+            display_mstr(122);
         end
         else begin
-            display_msg(message_str(310, 123));
+            display_mstr(123);
         end
     end
 end
@@ -216,18 +216,18 @@ begin
     per_check := do_check(dude_obj, 1, 0);
     if (is_success(per_check)) then begin
         if (is_critical(per_check)) then begin
-            display_msg(message_str(310, 110));
+            display_mstr(110);
         end
         else begin
-            display_msg(message_str(310, 111));
+            display_mstr(111);
         end
     end
     else begin
         if (is_critical(per_check)) then begin
-            display_msg(message_str(310, 113));
+            display_mstr(113);
         end
         else begin
-            display_msg(message_str(310, 112));
+            display_mstr(112);
         end
     end
 end
@@ -235,4 +235,3 @@ end
 procedure Node999
 begin
 end
-
