diff --git a/Mods/MegaMod/Original/KTCVENTR.ssl b/Mods/MegaMod/ReDefine/KTCVENTR.ssl
index bcf995e..dc8144a 100644
--- a/Mods/MegaMod/Original/KTCVENTR.ssl
+++ b/Mods/MegaMod/ReDefine/KTCVENTR.ssl
@@ -145,7 +145,7 @@ end
 procedure damage_p_proc
 begin
     variable LVar0 := 0;
-    if (metarule(49, target_obj) == 6) then begin
+    if (weapon_dmg_type(target_obj) == 6) then begin
         LVar0 := tile_contains_pid_obj(17139, 0, 33554513);
         if (LVar0 != 0) then begin
             game_ui_disable;
@@ -169,4 +169,3 @@ end
 procedure description_p_proc
 begin
 end
-
