diff --git a/Mods/MegaMod/Original/LARSSCRP.ssl b/Mods/MegaMod/ReDefine/LARSSCRP.ssl
index 7a49a91..e671a4b 100644
--- a/Mods/MegaMod/Original/LARSSCRP.ssl
+++ b/Mods/MegaMod/ReDefine/LARSSCRP.ssl
@@ -223,8 +223,7 @@ begin
         end
     end
     if (local_var(1) < 2) then begin
-        float_msg(dude_obj, message_str(1641, 2100), 8);
+        float_msg(dude_obj, message_str(SCRIPT_LCLILIAN, 2100), FLOAT_MSG_YELLOW);
         set_local_var(1, local_var(1) + 1);
     end
 end
-
