diff --git a/Mods/MegaMod/Original/LCBHNTRM.ssl b/Mods/MegaMod/ReDefine/LCBHNTRM.ssl
index 2bc6cd4..e52e645 100644
--- a/Mods/MegaMod/Original/LCBHNTRM.ssl
+++ b/Mods/MegaMod/ReDefine/LCBHNTRM.ssl
@@ -57,10 +57,10 @@ procedure critter_p_proc
 begin
     if ((local_var(5) == 2) and obj_can_see_obj(self_obj, dude_obj)) then begin
         set_local_var(5, 1);
-        attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+        attack(dude_obj);
     end
-    if ((global_var(1094) > 0) and obj_can_see_obj(self_obj, dude_obj)) then begin
-        attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+    if ((global_var(GVAR_LYDIA_WAITING_FOR_LEON) > 0) and obj_can_see_obj(self_obj, dude_obj)) then begin
+        attack(dude_obj);
     end
 end
 
@@ -73,11 +73,11 @@ end
 
 procedure talk_p_proc
 begin
-    if (global_var(1093) == 6) then begin
-        if (global_var(1083) < 3) then begin
+    if (global_var(GVAR_PRIMIT_ATTACKED) == 6) then begin
+        if (global_var(GVAR_UNKNOWN_MB_USED) < 3) then begin
             start_gdialog(1631, self_obj, -1, -1, -1);
             gsay_start;
-            if (global_var(1083) == 0) then begin
+            if (global_var(GVAR_UNKNOWN_MB_USED) == 0) then begin
                 call Node016();
             end
             else begin
@@ -87,13 +87,13 @@ begin
             end_dialogue;
         end
         else begin
-            float_msg(self_obj, message_str(1631, 201), 8);
+            float_msg(self_obj, mstr(201), FLOAT_MSG_YELLOW);
         end
     end
     else begin
         start_gdialog(1631, self_obj, -1, -1, -1);
         gsay_start;
-        if (global_var(1093) > 0) then begin
+        if (global_var(GVAR_PRIMIT_ATTACKED) > 0) then begin
             call Node015();
         end
         else begin
@@ -107,57 +107,57 @@ end
 procedure destroy_p_proc
 begin
     if (source_obj == dude_obj) then begin
-        set_global_var(792, -50);
+        set_global_var(GVAR_TOWN_REP_SLAVER_VAULT, -50);
     end
     if (source_obj == dude_obj) then begin
-        set_global_var(5, global_var(5) + 1);
-        if (metarule(51, self_obj) == 2) then begin
+        set_global_var(GVAR_GOOD_MONSTER, global_var(GVAR_GOOD_MONSTER) + 1);
+        if (critter_kill_type(self_obj) == KILL_TYPE_children_kills) then begin
             if (source_obj == dude_obj) then begin
-                set_global_var(2, 0);
-                set_global_var(1, global_var(1) + 1);
-                set_global_var(0, global_var(0) + -15);
-                set_global_var(37, 0);
-                set_global_var(38, 0);
-                set_global_var(39, 0);
-                set_global_var(40, 0);
-                set_global_var(41, 0);
-                set_global_var(42, 0);
-                set_global_var(43, 0);
-                set_global_var(44, 0);
-                set_global_var(45, 0);
-                if (global_var(0) >= 1000) then begin
-                    set_global_var(37, 1);
+                set_global_var(GVAR_CHAMPION_REPUTATION, 0);
+                set_global_var(GVAR_CHILDKILLER_REPUTATION, global_var(GVAR_CHILDKILLER_REPUTATION) + 1);
+                set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + -15);
+                set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+                set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+                set_global_var(GVAR_KARMA_DEFENDER, 0);
+                set_global_var(GVAR_KARMA_WANDERER, 0);
+                set_global_var(GVAR_KARMA_BETRAYER, 0);
+                set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+                set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+                set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+                    set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
                 end
                 else begin
-                    if (global_var(0) >= 750) then begin
-                        set_global_var(38, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
                     end
                     else begin
-                        if (global_var(0) >= 500) then begin
-                            set_global_var(39, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                            set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                         end
                         else begin
-                            if (global_var(0) >= 250) then begin
-                                set_global_var(40, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                                set_global_var(GVAR_KARMA_DEFENDER, 1);
                             end
                             else begin
-                                if (global_var(0) > -250) then begin
-                                    set_global_var(41, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                                    set_global_var(GVAR_KARMA_WANDERER, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -500) then begin
-                                        set_global_var(42, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                        set_global_var(GVAR_KARMA_BETRAYER, 1);
                                     end
                                     else begin
-                                        if (global_var(0) > -750) then begin
-                                            set_global_var(43, 1);
+                                        if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                            set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                         end
                                         else begin
-                                            if (global_var(0) > -1000) then begin
-                                                set_global_var(44, 1);
+                                            if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                                set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                             end
                                             else begin
-                                                set_global_var(45, 1);
+                                                set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                             end
                                         end
                                     end
@@ -168,58 +168,58 @@ begin
                 end
             end
         end
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(4) > (3 * global_var(5))) or (global_var(2) == 1)) and (global_var(1) == 0)) then begin
-            set_global_var(2, 1);
-            set_global_var(3, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(4) > (3 * global_var(GVAR_GOOD_MONSTER))) or (global_var(GVAR_CHAMPION_REPUTATION) == 1)) and (global_var(GVAR_CHILDKILLER_REPUTATION) == 0)) then begin
+            set_global_var(GVAR_CHAMPION_REPUTATION, 1);
+            set_global_var(GVAR_BERSERKER_REPUTATION, 0);
         end
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(5) > (2 * global_var(4))) or (global_var(3) == 1))) then begin
-            set_global_var(3, 1);
-            set_global_var(2, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
+            set_global_var(GVAR_BERSERKER_REPUTATION, 1);
+            set_global_var(GVAR_CHAMPION_REPUTATION, 0);
         end
-        set_global_var(0, global_var(0) + -10);
+        set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + -10);
         debug_msg("Player gains " + -10 + " Karma Points.");
-        set_global_var(37, 0);
-        set_global_var(38, 0);
-        set_global_var(39, 0);
-        set_global_var(40, 0);
-        set_global_var(41, 0);
-        set_global_var(42, 0);
-        set_global_var(43, 0);
-        set_global_var(44, 0);
-        set_global_var(45, 0);
-        if (global_var(0) >= 1000) then begin
-            set_global_var(37, 1);
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+        set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+        set_global_var(GVAR_KARMA_DEFENDER, 0);
+        set_global_var(GVAR_KARMA_WANDERER, 0);
+        set_global_var(GVAR_KARMA_BETRAYER, 0);
+        set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+        set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+            set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
         end
         else begin
-            if (global_var(0) >= 750) then begin
-                set_global_var(38, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
             end
             else begin
-                if (global_var(0) >= 500) then begin
-                    set_global_var(39, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                 end
                 else begin
-                    if (global_var(0) >= 250) then begin
-                        set_global_var(40, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                        set_global_var(GVAR_KARMA_DEFENDER, 1);
                     end
                     else begin
-                        if (global_var(0) > -250) then begin
-                            set_global_var(41, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                            set_global_var(GVAR_KARMA_WANDERER, 1);
                         end
                         else begin
-                            if (global_var(0) > -500) then begin
-                                set_global_var(42, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                set_global_var(GVAR_KARMA_BETRAYER, 1);
                             end
                             else begin
-                                if (global_var(0) > -750) then begin
-                                    set_global_var(43, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -1000) then begin
-                                        set_global_var(44, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                     end
                                     else begin
-                                        set_global_var(45, 1);
+                                        set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                     end
                                 end
                             end
@@ -234,13 +234,13 @@ end
 procedure look_at_p_proc
 begin
     script_overrides;
-    display_msg(message_str(1640, 150));
+    display_msg(message_str(SCRIPT_LCBHLEON, 150));
 end
 
 procedure description_p_proc
 begin
     script_overrides;
-    display_msg(message_str(1640, 150));
+    display_msg(message_str(SCRIPT_LCBHLEON, 150));
 end
 
 procedure use_skill_on_p_proc
@@ -251,16 +251,16 @@ procedure damage_p_proc
 begin
     if (source_obj == dude_obj) then begin
         set_local_var(6, 1);
-        set_global_var(792, -50);
-        set_global_var(1094, 1);
+        set_global_var(GVAR_TOWN_REP_SLAVER_VAULT, -50);
+        set_global_var(GVAR_LYDIA_WAITING_FOR_LEON, 1);
     end
 end
 
 procedure map_enter_p_proc
 begin
     Only_Once := 0;
-    critter_add_trait(self_obj, 1, 6, 214);
-    critter_add_trait(self_obj, 1, 5, 124);
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 214);
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_BOUNTY_HUNTER);
 end
 
 procedure combat_p_proc
@@ -288,121 +288,121 @@ end
 
 procedure Node001
 begin
-    gsay_reply(1631, 300);
-    gsay_option(1631, 310, Node002, -1);
-    if (global_var(1082) == 1) then begin
-        gsay_option(1631, 311, Node003, -1);
+    Reply(300);
+    giq_option(1, NAME, 310, Node002, -1);
+    if (global_var(GVAR_ATTACKED_ORDER) == 1) then begin
+        giq_option(1, NAME, 311, Node003, -1);
     end
-    if (global_var(1083) == 1) then begin
-        gsay_option(1631, 312, Node004, -1);
+    if (global_var(GVAR_UNKNOWN_MB_USED) == 1) then begin
+        giq_option(1, NAME, 312, Node004, -1);
     end
-    gsay_option(1631, 313, Node999, -1);
+    giq_option(1, NAME, 313, Node999, -1);
 end
 
 procedure Node002
 begin
-    gsay_reply(1631, 400);
-    gsay_option(1631, 410, Node005, -1);
-    gsay_option(1631, 413, Node999, -1);
+    Reply(400);
+    giq_option(1, NAME, 410, Node005, -1);
+    giq_option(1, NAME, 413, Node999, -1);
 end
 
 procedure Node003
 begin
-    gsay_reply(1631, 401);
-    gsay_option(1631, 411, Node006, -1);
-    gsay_option(1631, 413, Node999, -1);
+    Reply(401);
+    giq_option(1, NAME, 411, Node006, -1);
+    giq_option(1, NAME, 413, Node999, -1);
 end
 
 procedure Node004
 begin
-    gsay_reply(1631, 402);
-    gsay_option(1631, 412, Node007, -1);
-    gsay_option(1631, 413, Node999, -1);
+    Reply(402);
+    giq_option(1, NAME, 412, Node007, -1);
+    giq_option(1, NAME, 413, Node999, -1);
 end
 
 procedure Node005
 begin
-    gsay_reply(1631, 500);
-    gsay_option(1631, 510, Node008, -1);
-    gsay_option(1631, 513, Node999, -1);
+    Reply(500);
+    giq_option(1, NAME, 510, Node008, -1);
+    giq_option(1, NAME, 513, Node999, -1);
 end
 
 procedure Node006
 begin
-    gsay_reply(1631, 501);
-    gsay_option(1631, 511, Node009, -1);
-    gsay_option(1631, 413, Node999, -1);
+    Reply(501);
+    giq_option(1, NAME, 511, Node009, -1);
+    giq_option(1, NAME, 413, Node999, -1);
 end
 
 procedure Node007
 begin
-    gsay_reply(1631, 502);
-    gsay_option(1631, 512, Node010, -1);
-    gsay_option(1631, 513, Node999, -1);
+    Reply(502);
+    giq_option(1, NAME, 512, Node010, -1);
+    giq_option(1, NAME, 513, Node999, -1);
 end
 
 procedure Node008
 begin
-    gsay_reply(1631, 600);
-    gsay_option(1631, 610, Node011, -1);
-    gsay_option(1631, 513, Node999, -1);
+    Reply(600);
+    giq_option(1, NAME, 610, Node011, -1);
+    giq_option(1, NAME, 513, Node999, -1);
 end
 
 procedure Node009
 begin
-    gsay_reply(1631, 601);
-    gsay_option(1631, 611, Node012, -1);
-    gsay_option(1631, 614, accept_bh_quest, -1);
-    gsay_option(1631, 613, Node999, -1);
+    Reply(601);
+    giq_option(1, NAME, 611, Node012, -1);
+    giq_option(1, NAME, 614, accept_bh_quest, -1);
+    giq_option(1, NAME, 613, Node999, -1);
 end
 
 procedure Node010
 begin
-    gsay_reply(1631, 602);
-    gsay_option(1631, 612, Node013, -1);
-    gsay_option(1631, 513, Node999, -1);
+    Reply(602);
+    giq_option(1, NAME, 612, Node013, -1);
+    giq_option(1, NAME, 513, Node999, -1);
 end
 
 procedure Node011
 begin
-    gsay_reply(1631, 700);
-    gsay_option(1631, 710, Node014, -1);
-    gsay_option(1631, 413, Node999, -1);
+    Reply(700);
+    giq_option(1, NAME, 710, Node014, -1);
+    giq_option(1, NAME, 413, Node999, -1);
 end
 
 procedure Node012
 begin
-    gsay_reply(1631, 701);
-    gsay_option(1631, 614, accept_bh_quest, -1);
-    gsay_option(1631, 613, Node999, -1);
+    Reply(701);
+    giq_option(1, NAME, 614, accept_bh_quest, -1);
+    giq_option(1, NAME, 613, Node999, -1);
 end
 
 procedure Node013
 begin
-    gsay_reply(1631, 702);
-    gsay_option(1631, 611, Node012, -1);
-    gsay_option(1631, 614, accept_bh_quest, -1);
-    gsay_option(1631, 613, Node999, -1);
+    Reply(702);
+    giq_option(1, NAME, 611, Node012, -1);
+    giq_option(1, NAME, 614, accept_bh_quest, -1);
+    giq_option(1, NAME, 613, Node999, -1);
 end
 
 procedure Node014
 begin
-    gsay_reply(1631, 800);
-    gsay_option(1631, 511, Node009, -1);
-    gsay_option(1631, 413, Node999, -1);
+    Reply(800);
+    giq_option(1, NAME, 511, Node009, -1);
+    giq_option(1, NAME, 413, Node999, -1);
 end
 
 procedure Node015
 begin
     variable LVar0 := 0;
     LVar0 := obj_is_carrying_obj_pid(dude_obj, 579);
-    gsay_reply(1631, 900);
-    gsay_option(1631, 910, Node999, -1);
-    if ((global_var(1093) == 5) and (LVar0 > 0)) then begin
-        gsay_option(1631, 911, Node100, -1);
+    Reply(900);
+    giq_option(1, NAME, 910, Node999, -1);
+    if ((global_var(GVAR_PRIMIT_ATTACKED) == 5) and (LVar0 > 0)) then begin
+        giq_option(1, NAME, 911, Node100, -1);
     end
-    if ((global_var(1093) == 5) and (LVar0 == 0)) then begin
-        gsay_option(1631, 912, Node044, -1);
+    if ((global_var(GVAR_PRIMIT_ATTACKED) == 5) and (LVar0 == 0)) then begin
+        giq_option(1, NAME, 912, Node044, -1);
     end
     return 0;
     LVar0 := 0;
@@ -410,40 +410,40 @@ end
 
 procedure Node016
 begin
-    gsay_reply(1631, 3000);
-    gsay_option(1631, 3020, Node016a, -1);
-    gsay_option(1631, 3030, Node999, -1);
+    Reply(3000);
+    giq_option(1, NAME, 3020, Node016a, -1);
+    giq_option(1, NAME, 3030, Node999, -1);
 end
 
 procedure Node016a
 begin
-    gsay_reply(1631, 3100);
-    gsay_option(1631, 3120, Node016b, -1);
-    gsay_option(1631, 3130, Node999, -1);
+    Reply(3100);
+    giq_option(1, NAME, 3120, Node016b, -1);
+    giq_option(1, NAME, 3130, Node999, -1);
 end
 
 procedure Node016b
 begin
-    set_global_var(1083, 1);
+    set_global_var(GVAR_UNKNOWN_MB_USED, 1);
 end
 
 procedure Node017
 begin
-    gsay_reply(1631, 3200);
-    if (global_var(1083) == 2) then begin
-        gsay_option(1631, 3220, Node017a, -1);
+    Reply(3200);
+    if (global_var(GVAR_UNKNOWN_MB_USED) == 2) then begin
+        giq_option(1, NAME, 3220, Node017a, -1);
     end
-    gsay_option(1631, 3230, Node999, -1);
+    giq_option(1, NAME, 3230, Node999, -1);
 end
 
 procedure Node017a
 begin
-    gsay_message(1631, 3300, 49);
-    set_global_var(1083, 3);
-    set_global_var(792, global_var(792) + 15);
-    set_global_var(0, global_var(0) + 25);
+    gsay_message(NAME, 3300, GOOD_REACTION);
+    set_global_var(GVAR_UNKNOWN_MB_USED, 3);
+    set_global_var(GVAR_TOWN_REP_SLAVER_VAULT, global_var(GVAR_TOWN_REP_SLAVER_VAULT) + 15);
+    set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + 25);
     give_exp_points(1500);
-    display_msg(message_str(14, 100) + 1500 + message_str(14, 101));
+    display_msg(g_mstr(100) + 1500 + g_mstr(101));
 end
 
 procedure Node100
@@ -451,40 +451,39 @@ begin
     item := obj_carrying_pid_obj(dude_obj, 579);
     rm_obj_from_inven(dude_obj, item);
     add_obj_to_inven(self_obj, item);
-    gsay_reply(1631, 1000);
-    gsay_option(1631, 1003, give_quest_xp, -1);
+    Reply(1000);
+    giq_option(1, NAME, 1003, give_quest_xp, -1);
 end
 
 procedure give_quest_xp
 begin
     give_exp_points(4000);
-    set_global_var(1093, 6);
-    set_global_var(792, global_var(792) + 5);
-    set_global_var(0, global_var(0) + 35);
+    set_global_var(GVAR_PRIMIT_ATTACKED, 6);
+    set_global_var(GVAR_TOWN_REP_SLAVER_VAULT, global_var(GVAR_TOWN_REP_SLAVER_VAULT) + 5);
+    set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + 35);
 end
 
 procedure NodeFleeing
 begin
-    float_msg(self_obj, message_str(1631, random(105, 107)), 8);
+    float_msg(self_obj, mstr(random(105, 107)), FLOAT_MSG_YELLOW);
 end
 
 procedure NodeFighting
 begin
-    float_msg(self_obj, message_str(1631, random(100, 104)), 8);
+    float_msg(self_obj, mstr(random(100, 104)), FLOAT_MSG_YELLOW);
 end
 
 procedure accept_bh_quest
 begin
-    set_global_var(1093, 1);
-    set_global_var(1090, random(1, 3));
-    gsay_reply(1631, 615);
-    gsay_option(1631, 616, Node999, -1);
+    set_global_var(GVAR_PRIMIT_ATTACKED, 1);
+    set_global_var(GVAR_CLOSE_TO_KRUGER, random(1, 3));
+    Reply(615);
+    giq_option(1, NAME, 616, Node999, -1);
 end
 
 procedure Node044
 begin
-    gsay_reply(1631, 913);
-    gsay_option(1631, 914, Node999, -1);
-    gsay_option(1631, 915, Node999, -1);
+    Reply(913);
+    giq_option(1, NAME, 914, Node999, -1);
+    giq_option(1, NAME, 915, Node999, -1);
 end
-
