diff --git a/Mods/MegaMod/Original/LCMATEOS.ssl b/Mods/MegaMod/ReDefine/LCMATEOS.ssl
index b2b83e9..455f396 100644
--- a/Mods/MegaMod/Original/LCMATEOS.ssl
+++ b/Mods/MegaMod/ReDefine/LCMATEOS.ssl
@@ -141,28 +141,28 @@ begin
     end
     if (critter_inven_obj(self_obj, 0) != arg0) then begin
         if (critter_inven_obj(self_obj, 0) == 0) then begin
-            metarule3(107, self_obj, 16777216 + arg1, 0);
+            art_change_fid_num(self_obj, 16777216 + arg1);
         end
         if ((obj_pid(critter_inven_obj(self_obj, 0)) == 1) or (obj_pid(critter_inven_obj(self_obj, 0)) == 379)) then begin
-            metarule3(107, self_obj, 16777228, 0);
+            art_change_fid_num(self_obj, 16777228);
         end
         if ((obj_pid(critter_inven_obj(self_obj, 0)) == 232) or (obj_pid(critter_inven_obj(self_obj, 0)) == 3)) then begin
-            metarule3(107, self_obj, 16777217, 0);
+            art_change_fid_num(self_obj, 16777217);
         end
         if ((obj_pid(critter_inven_obj(self_obj, 0)) == 349) or (obj_pid(critter_inven_obj(self_obj, 0)) == 348)) then begin
-            metarule3(107, self_obj, 16777287, 0);
+            art_change_fid_num(self_obj, 16777287);
         end
         if ((obj_pid(critter_inven_obj(self_obj, 0)) == 240) or (obj_pid(critter_inven_obj(self_obj, 0)) == 380) or (obj_pid(critter_inven_obj(self_obj, 0)) == 2)) then begin
-            metarule3(107, self_obj, 16777230, 0);
+            art_change_fid_num(self_obj, 16777230);
         end
         if ((obj_pid(critter_inven_obj(self_obj, 0)) == 74) or (obj_pid(critter_inven_obj(self_obj, 0)) == 265)) then begin
-            metarule3(107, self_obj, 16777229, 0);
+            art_change_fid_num(self_obj, 16777229);
         end
         if ((obj_pid(critter_inven_obj(self_obj, 0)) == 239) or (obj_pid(critter_inven_obj(self_obj, 0)) == 17) or (obj_pid(critter_inven_obj(self_obj, 0)) == 381)) then begin
-            metarule3(107, self_obj, 16777226, 0);
+            art_change_fid_num(self_obj, 16777226);
         end
         if ((obj_pid(critter_inven_obj(self_obj, 0)) == 524) or (obj_pid(critter_inven_obj(self_obj, 0)) == 113)) then begin
-            metarule3(107, self_obj, 16777218, 0);
+            art_change_fid_num(self_obj, 16777218);
         end
         LVar2 := critter_inven_obj(self_obj, 1);
         debug_msg("armour pid == " + obj_pid(LVar2));
@@ -174,85 +174,85 @@ end
 
 procedure Node001
 begin
-    gsay_reply(1716, 200);
-    gsay_option(1716, 210, Node002, -1);
+    Reply(200);
+    giq_option(1, NAME, 210, Node002, -1);
     if (local_var(8) == 0) then begin
-        gsay_option(1716, 211, Node997, -1);
+        giq_option(1, NAME, 211, Node997, -1);
     end
-    gsay_option(1716, 212, Node999, -1);
+    giq_option(1, NAME, 212, Node999, -1);
 end
 
 procedure Node002
 begin
-    gsay_reply(1716, 300);
+    Reply(300);
     set_local_var(4, 1);
-    if ((map_var(2) == 2) or (map_var(2) == 3) and (global_var(1103) != 3)) then begin
-        gsay_option(1716, 310, Node003, -1);
+    if ((map_var(2) == 2) or (map_var(2) == 3) and (global_var(GVAR_DUDE_IS_SCRB) != 3)) then begin
+        giq_option(1, NAME, 310, Node003, -1);
     end
-    gsay_option(1716, 311, Node999, -1);
+    giq_option(1, NAME, 311, Node999, -1);
 end
 
 procedure Node003
 begin
-    gsay_reply(1716, 400);
-    gsay_option(1716, 410, Node004, -1);
-    gsay_option(1716, 411, Node005, -1);
-    gsay_option(1716, 412, Node999, -1);
+    Reply(400);
+    giq_option(1, NAME, 410, Node004, -1);
+    giq_option(1, NAME, 411, Node005, -1);
+    giq_option(1, NAME, 412, Node999, -1);
 end
 
 procedure Node004
 begin
-    gsay_reply(1716, 500);
-    gsay_option(1716, 510, Node008, -1);
-    gsay_option(1716, 511, Node996, -1);
+    Reply(500);
+    giq_option(1, NAME, 510, Node008, -1);
+    giq_option(1, NAME, 511, Node996, -1);
 end
 
 procedure Node005
 begin
-    gsay_reply(1716, 600);
-    gsay_option(1716, 610, Node006, -1);
-    gsay_option(1716, 612, Node999, -1);
+    Reply(600);
+    giq_option(1, NAME, 610, Node006, -1);
+    giq_option(1, NAME, 612, Node999, -1);
 end
 
 procedure Node006
 begin
-    gsay_reply(1716, 700);
-    gsay_option(1716, 710, Node007, -1);
-    gsay_option(1716, 711, Node008, -1);
+    Reply(700);
+    giq_option(1, NAME, 710, Node007, -1);
+    giq_option(1, NAME, 711, Node008, -1);
 end
 
 procedure Node007
 begin
-    gsay_reply(1716, 800);
+    Reply(800);
     set_local_var(9, 2);
-    gsay_option(1716, 810, Node008, -1);
-    gsay_option(1716, 811, Node999, -1);
+    giq_option(1, NAME, 810, Node008, -1);
+    giq_option(1, NAME, 811, Node999, -1);
 end
 
 procedure Node008
 begin
-    gsay_reply(1716, 1000);
+    Reply(1000);
     set_map_var(3, map_var(3) bwor 4);
-    gsay_option(1716, 1010, Node998, -1);
-    gsay_option(1716, 1011, Node999, -1);
+    giq_option(1, NAME, 1010, Node998, -1);
+    giq_option(1, NAME, 1011, Node999, -1);
 end
 
 procedure Node996
 begin
-    gsay_message(1716, 900, 49);
+    gsay_message(NAME, 900, GOOD_REACTION);
     set_local_var(9, 1);
 end
 
 procedure Node997
 begin
-    gsay_message(1716, 1100, 49);
+    gsay_message(NAME, 1100, GOOD_REACTION);
     set_local_var(8, get_day + (get_month * 100));
-    add_obj_to_inven(dude_obj, create_object_sid(124, 0, 0, -1));
+    add_obj_to_inven(dude_obj, create_object(124, 0, 0));
 end
 
 procedure Node998
 begin
-    set_global_var(1094, 1);
+    set_global_var(GVAR_LYDIA_WAITING_FOR_LEON, 1);
     set_local_var(5, 2);
 end
 
@@ -266,8 +266,8 @@ end
 
 procedure map_enter_p_proc
 begin
-    critter_add_trait(self_obj, 1, 6, 214);
-    critter_add_trait(self_obj, 1, 5, 124);
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 214);
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_BOUNTY_HUNTER);
 end
 
 procedure critter_p_proc
@@ -275,22 +275,22 @@ begin
     if ((local_var(5) == 2) and obj_can_see_obj(self_obj, dude_obj)) then begin
         set_local_var(5, 1);
         if ((map_var(3) bwand 3) != 0) then begin
-            critter_add_trait(self_obj, 1, 6, 3);
-            critter_add_trait(self_obj, 1, 5, 123);
+            critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 3);
+            critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_SCARED_BRAHMIN);
         end
-        attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+        attack(dude_obj);
     end
-    if (((map_var(3) bwand 4) != 0) and (tile_num(dude_obj) == 21292)) then begin
+    if (((map_var(3) bwand 4) != 0) and (dude_tile == 21292)) then begin
         destroy_object(self_obj);
     end
-    if ((global_var(1094) > 0) and obj_can_see_obj(self_obj, dude_obj)) then begin
-        attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+    if ((global_var(GVAR_LYDIA_WAITING_FOR_LEON) > 0) and obj_can_see_obj(self_obj, dude_obj)) then begin
+        attack(dude_obj);
     end
 end
 
 procedure damage_p_proc
 begin
-    set_global_var(1094, 1);
+    set_global_var(GVAR_LYDIA_WAITING_FOR_LEON, 1);
     if ((source_obj == dude_obj) and (Only_Once == 0)) then begin
         Only_Once := 1;
         set_local_var(6, 1);
@@ -307,20 +307,20 @@ end
 procedure talk_p_proc
 begin
     if ((map_var(3) bwand 4) != 0) then begin
-        float_msg(self_obj, message_str(1716, 800), 8);
+        float_msg(self_obj, mstr(800), FLOAT_MSG_YELLOW);
     end
     else begin
         if (local_var(8) == (get_day + (get_month * 100))) then begin
-            float_msg(self_obj, message_str(1716, 1200), 8);
+            float_msg(self_obj, mstr(1200), FLOAT_MSG_YELLOW);
         end
         else begin
             if (local_var(9) == 1) then begin
-                float_msg(self_obj, 1300, 2);
+                float_msg(self_obj, 1300, FLOAT_MSG_RED);
             end
             else begin
                 start_gdialog(1716, self_obj, -1, -1, -1);
                 gsay_start;
-                if ((local_var(9) == 2) and (global_var(1103) != 3)) then begin
+                if ((local_var(9) == 2) and (global_var(GVAR_DUDE_IS_SCRB) != 3)) then begin
                     call Node007();
                 end
                 else begin
@@ -341,20 +341,19 @@ procedure look_at_p_proc
 begin
     script_overrides;
     if (local_var(4) == 0) then begin
-        display_msg(message_str(1716, 100));
+        display_mstr(100);
     end
     else begin
-        display_msg(message_str(1716, 101));
+        display_mstr(101);
     end
 end
 
 procedure description_p_proc
 begin
     script_overrides;
-    display_msg(message_str(1716, 102));
+    display_mstr(102);
 end
 
 procedure use_skill_on_p_proc
 begin
 end
-
