diff --git a/Mods/MegaMod/Original/LENNY.ssl b/Mods/MegaMod/ReDefine/LENNY.ssl
index d234595..96cf352 100644
--- a/Mods/MegaMod/Original/LENNY.ssl
+++ b/Mods/MegaMod/ReDefine/LENNY.ssl
@@ -66,7 +66,7 @@ procedure start
 begin
     if (local_var(0) == 0) then begin
         set_local_var(0, 1);
-        critter_add_trait(self_obj, 1, 6, 184);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 184);
         set_local_var(1, tile_num(self_obj));
     end
 end
@@ -74,62 +74,62 @@ end
 procedure damage_p_proc
 begin
     if (source_obj == dude_obj) then begin
-        attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+        attack(dude_obj);
     end
 end
 
 procedure destroy_p_proc
 begin
     if (source_obj == dude_obj) then begin
-        set_global_var(5, global_var(5) + 1);
-        if (metarule(51, self_obj) == 2) then begin
+        set_global_var(GVAR_GOOD_MONSTER, global_var(GVAR_GOOD_MONSTER) + 1);
+        if (critter_kill_type(self_obj) == 2) then begin
             if (source_obj == dude_obj) then begin
-                set_global_var(2, 0);
-                set_global_var(1, global_var(1) + 1);
-                set_global_var(0, global_var(0) + -15);
+                set_global_var(GVAR_CHAMPION_REPUTATION, 0);
+                set_global_var(GVAR_CHILDKILLER_REPUTATION, global_var(GVAR_CHILDKILLER_REPUTATION) + 1);
+                set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + -15);
                 debug_msg("Player gains " + -15 + " Karma Points.");
-                set_global_var(37, 0);
-                set_global_var(38, 0);
-                set_global_var(39, 0);
-                set_global_var(40, 0);
-                set_global_var(41, 0);
-                set_global_var(42, 0);
-                set_global_var(43, 0);
-                set_global_var(44, 0);
-                set_global_var(45, 0);
-                if (global_var(0) >= 1000) then begin
-                    set_global_var(37, 1);
+                set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+                set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+                set_global_var(GVAR_KARMA_DEFENDER, 0);
+                set_global_var(GVAR_KARMA_WANDERER, 0);
+                set_global_var(GVAR_KARMA_BETRAYER, 0);
+                set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+                set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+                set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+                    set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
                 end
                 else begin
-                    if (global_var(0) >= 750) then begin
-                        set_global_var(38, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
                     end
                     else begin
-                        if (global_var(0) >= 500) then begin
-                            set_global_var(39, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                            set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                         end
                         else begin
-                            if (global_var(0) >= 250) then begin
-                                set_global_var(40, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                                set_global_var(GVAR_KARMA_DEFENDER, 1);
                             end
                             else begin
-                                if (global_var(0) > -250) then begin
-                                    set_global_var(41, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                                    set_global_var(GVAR_KARMA_WANDERER, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -500) then begin
-                                        set_global_var(42, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                        set_global_var(GVAR_KARMA_BETRAYER, 1);
                                     end
                                     else begin
-                                        if (global_var(0) > -750) then begin
-                                            set_global_var(43, 1);
+                                        if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                            set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                         end
                                         else begin
-                                            if (global_var(0) > -1000) then begin
-                                                set_global_var(44, 1);
+                                            if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                                set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                             end
                                             else begin
-                                                set_global_var(45, 1);
+                                                set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                             end
                                         end
                                     end
@@ -138,62 +138,62 @@ begin
                         end
                     end
                 end
-                set_global_var(808, global_var(808) + -5);
+                set_global_var(GVAR_GANG_2, global_var(GVAR_GANG_2) + -5);
                 debug_msg("Added " + -5 + " to Town Rep");
             end
         end
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(4) > (3 * global_var(5))) or (global_var(2) == 1)) and (global_var(1) == 0)) then begin
-            set_global_var(2, 1);
-            set_global_var(3, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(4) > (3 * global_var(GVAR_GOOD_MONSTER))) or (global_var(GVAR_CHAMPION_REPUTATION) == 1)) and (global_var(GVAR_CHILDKILLER_REPUTATION) == 0)) then begin
+            set_global_var(GVAR_CHAMPION_REPUTATION, 1);
+            set_global_var(GVAR_BERSERKER_REPUTATION, 0);
         end
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(5) > (2 * global_var(4))) or (global_var(3) == 1))) then begin
-            set_global_var(3, 1);
-            set_global_var(2, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
+            set_global_var(GVAR_BERSERKER_REPUTATION, 1);
+            set_global_var(GVAR_CHAMPION_REPUTATION, 0);
         end
-        set_global_var(0, global_var(0) + -5);
+        set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + -5);
         debug_msg("Player gains " + -5 + " Karma Points.");
-        set_global_var(37, 0);
-        set_global_var(38, 0);
-        set_global_var(39, 0);
-        set_global_var(40, 0);
-        set_global_var(41, 0);
-        set_global_var(42, 0);
-        set_global_var(43, 0);
-        set_global_var(44, 0);
-        set_global_var(45, 0);
-        if (global_var(0) >= 1000) then begin
-            set_global_var(37, 1);
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+        set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+        set_global_var(GVAR_KARMA_DEFENDER, 0);
+        set_global_var(GVAR_KARMA_WANDERER, 0);
+        set_global_var(GVAR_KARMA_BETRAYER, 0);
+        set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+        set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+            set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
         end
         else begin
-            if (global_var(0) >= 750) then begin
-                set_global_var(38, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
             end
             else begin
-                if (global_var(0) >= 500) then begin
-                    set_global_var(39, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                 end
                 else begin
-                    if (global_var(0) >= 250) then begin
-                        set_global_var(40, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                        set_global_var(GVAR_KARMA_DEFENDER, 1);
                     end
                     else begin
-                        if (global_var(0) > -250) then begin
-                            set_global_var(41, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                            set_global_var(GVAR_KARMA_WANDERER, 1);
                         end
                         else begin
-                            if (global_var(0) > -500) then begin
-                                set_global_var(42, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                set_global_var(GVAR_KARMA_BETRAYER, 1);
                             end
                             else begin
-                                if (global_var(0) > -750) then begin
-                                    set_global_var(43, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -1000) then begin
-                                        set_global_var(44, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                     end
                                     else begin
-                                        set_global_var(45, 1);
+                                        set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                     end
                                 end
                             end
@@ -202,7 +202,7 @@ begin
                 end
             end
         end
-        set_global_var(808, global_var(808) + -3);
+        set_global_var(GVAR_GANG_2, global_var(GVAR_GANG_2) + -3);
         debug_msg("Added " + -3 + " to Town Rep");
     end
 end
@@ -228,17 +228,17 @@ procedure look_at_p_proc
 begin
     script_overrides;
     if (known == 1) then begin
-        display_msg(message_str(1498, 101));
+        display_mstr(101);
     end
     else begin
-        display_msg(message_str(1498, 100));
+        display_mstr(100);
     end
 end
 
 procedure description_p_proc
 begin
     script_overrides;
-    display_msg(message_str(1498, 100));
+    display_mstr(100);
 end
 
 procedure do_dialogue
@@ -267,39 +267,39 @@ end
 procedure Node000
 begin
     known := 1;
-    gsay_reply(1498, 104);
-    giq_option(4, 1498, 103, Node999, 50);
+    Reply(104);
+    giq_option(4, NAME, 103, Node999, NEUTRAL_REACTION);
     call LennyCombat();
 end
 
 procedure Node001
 begin
-    gsay_reply(1498, 105);
-    giq_option(4, 1498, 106, Node015, 50);
-    giq_option(4, 1498, 107, Node003, 50);
-    giq_option(4, 1498, 108, Node002, 50);
+    Reply(105);
+    giq_option(4, NAME, 106, Node015, NEUTRAL_REACTION);
+    giq_option(4, NAME, 107, Node003, NEUTRAL_REACTION);
+    giq_option(4, NAME, 108, Node002, NEUTRAL_REACTION);
 end
 
 procedure Node002
 begin
-    gsay_reply(1498, 109);
-    giq_option(4, 1498, 106, Node003, 50);
-    sayReply(0, message_str(1498, 109));
+    Reply(109);
+    giq_option(4, NAME, 106, Node003, NEUTRAL_REACTION);
+    sayReply(0, mstr(109));
 end
 
 procedure Node003
 begin
-    gsay_reply(1498, 110);
-    giq_option(4, 1498, 111, Node008, 50);
-    giq_option(4, 1498, 112, Node004, 50);
+    Reply(110);
+    giq_option(4, NAME, 111, Node008, NEUTRAL_REACTION);
+    giq_option(4, NAME, 112, Node004, NEUTRAL_REACTION);
 end
 
 procedure Node004
 begin
-    gsay_reply(1498, 113);
-    giq_option(4, 1498, 114, Node004a, 50);
-    giq_option(4, 1498, 115, Node004b, 50);
-    giq_option(4, 1498, 116, Node006, 50);
+    Reply(113);
+    giq_option(4, NAME, 114, Node004a, NEUTRAL_REACTION);
+    giq_option(4, NAME, 115, Node004b, NEUTRAL_REACTION);
+    giq_option(4, NAME, 116, Node006, NEUTRAL_REACTION);
 end
 
 procedure Node004a
@@ -324,14 +324,14 @@ end
 
 procedure Node005
 begin
-    gsay_reply(1498, 117);
-    giq_option(4, 1498, 118, Node999, 50);
+    Reply(117);
+    giq_option(4, NAME, 118, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node006
 begin
-    gsay_reply(1498, 119);
-    giq_option(4, 1498, 120, Node006a, 50);
+    Reply(119);
+    giq_option(4, NAME, 120, Node006a, NEUTRAL_REACTION);
 end
 
 procedure Node006a
@@ -342,104 +342,104 @@ end
 
 procedure Node007
 begin
-    gsay_reply(1498, 121);
-    giq_option(4, 1498, 122, Node999, 50);
+    Reply(121);
+    giq_option(4, NAME, 122, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node008
 begin
-    gsay_reply(1498, 123);
-    giq_option(4, 1498, 122, Node999, 50);
+    Reply(123);
+    giq_option(4, NAME, 122, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node009
 begin
-    gsay_reply(1498, 124);
+    Reply(124);
     if (know_fools) then begin
-        giq_option(4, 1498, 125, Node010, 50);
+        giq_option(4, NAME, 125, Node010, NEUTRAL_REACTION);
     end
     if (know_crypts) then begin
-        giq_option(4, 1498, 126, Node011, 50);
+        giq_option(4, NAME, 126, Node011, NEUTRAL_REACTION);
     end
     if (know_power) then begin
-        giq_option(4, 1498, 127, Node012, 50);
+        giq_option(4, NAME, 127, Node012, NEUTRAL_REACTION);
     end
-    giq_option(4, 1498, 128, Node999, 50);
+    giq_option(4, NAME, 128, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node010
 begin
-    gsay_reply(1498, 129);
-    giq_option(4, 1498, 130, Node999, 50);
-    giq_option(4, 1498, 126, Node011, 50);
-    giq_option(4, 1498, 127, Node012, 50);
+    Reply(129);
+    giq_option(4, NAME, 130, Node999, NEUTRAL_REACTION);
+    giq_option(4, NAME, 126, Node011, NEUTRAL_REACTION);
+    giq_option(4, NAME, 127, Node012, NEUTRAL_REACTION);
 end
 
 procedure Node011
 begin
-    gsay_reply(1498, 131);
-    giq_option(4, 1498, 130, Node999, 50);
-    giq_option(4, 1498, 132, Node013, 50);
-    giq_option(4, 1498, 125, Node010, 50);
-    giq_option(4, 1498, 127, Node012, 50);
+    Reply(131);
+    giq_option(4, NAME, 130, Node999, NEUTRAL_REACTION);
+    giq_option(4, NAME, 132, Node013, NEUTRAL_REACTION);
+    giq_option(4, NAME, 125, Node010, NEUTRAL_REACTION);
+    giq_option(4, NAME, 127, Node012, NEUTRAL_REACTION);
 end
 
 procedure Node012
 begin
-    gsay_reply(1498, 133);
-    giq_option(4, 1498, 130, Node999, 50);
-    giq_option(4, 1498, 125, Node010, 50);
-    giq_option(4, 1498, 126, Node011, 50);
+    Reply(133);
+    giq_option(4, NAME, 130, Node999, NEUTRAL_REACTION);
+    giq_option(4, NAME, 125, Node010, NEUTRAL_REACTION);
+    giq_option(4, NAME, 126, Node011, NEUTRAL_REACTION);
 end
 
 procedure Node013
 begin
-    gsay_reply(1498, 134);
-    giq_option(4, 1498, 130, Node999, 50);
-    giq_option(4, 1498, 125, Node010, 50);
-    giq_option(4, 1498, 126, Node011, 50);
-    giq_option(4, 1498, 127, Node012, 50);
+    Reply(134);
+    giq_option(4, NAME, 130, Node999, NEUTRAL_REACTION);
+    giq_option(4, NAME, 125, Node010, NEUTRAL_REACTION);
+    giq_option(4, NAME, 126, Node011, NEUTRAL_REACTION);
+    giq_option(4, NAME, 127, Node012, NEUTRAL_REACTION);
 end
 
 procedure Node014
 begin
-    gsay_reply(1498, 135);
-    giq_option(-4, 1498, 103, Node999, 50);
+    Reply(135);
+    giq_option(-4, NAME, 103, Node999, NEUTRAL_REACTION);
     call Node999();
 end
 
 procedure Node015
 begin
-    gsay_reply(1498, 136);
-    giq_option(4, 1498, 137, Node016, 50);
-    giq_option(4, 1498, 138, Node017, 50);
+    Reply(136);
+    giq_option(4, NAME, 137, Node016, NEUTRAL_REACTION);
+    giq_option(4, NAME, 138, Node017, NEUTRAL_REACTION);
 end
 
 procedure Node016
 begin
-    gsay_reply(1498, 139);
-    giq_option(4, 1498, 140, Node003, 50);
-    giq_option(4, 1498, 141, Node018, 50);
+    Reply(139);
+    giq_option(4, NAME, 140, Node003, NEUTRAL_REACTION);
+    giq_option(4, NAME, 141, Node018, NEUTRAL_REACTION);
 end
 
 procedure Node017
 begin
-    gsay_reply(1498, 142);
-    giq_option(4, 1498, 130, Node999, 50);
-    giq_option(4, 1498, 143, Node003, 50);
+    Reply(142);
+    giq_option(4, NAME, 130, Node999, NEUTRAL_REACTION);
+    giq_option(4, NAME, 143, Node003, NEUTRAL_REACTION);
 end
 
 procedure Node018
 begin
-    gsay_reply(1498, 144);
-    giq_option(4, 1498, 145, Node003, 50);
-    giq_option(4, 1498, 146, Node019, 50);
+    Reply(144);
+    giq_option(4, NAME, 145, Node003, NEUTRAL_REACTION);
+    giq_option(4, NAME, 146, Node019, NEUTRAL_REACTION);
 end
 
 procedure Node019
 begin
-    gsay_reply(1498, 147);
-    giq_option(4, 1498, 103, LennyCombat, 50);
+    Reply(147);
+    giq_option(4, NAME, 103, LennyCombat, NEUTRAL_REACTION);
 end
 
 procedure LennyCombat
@@ -451,20 +451,20 @@ procedure get_reaction
 begin
     if (not(got_reaction)) then begin
         got_reaction := 1;
-        reaction := reaction + ((get_critter_stat(dude_obj, 3) - 5) * 5);
+        reaction := reaction + ((dude_charisma - 5) * 5);
         if (badguy) then begin
             reaction := reaction - map_var(8);
         end
         else begin
             reaction := reaction + map_var(8);
         end
-        if (global_var(1) > 2) then begin
+        if (global_var(GVAR_CHILDKILLER_REPUTATION) > 2) then begin
             reaction := reaction - 30;
         end
-        if ((global_var(4) > (3 * global_var(5))) or (global_var(2) == 1)) then begin
+        if ((global_var(GVAR_BAD_MONSTER) > (3 * global_var(GVAR_GOOD_MONSTER))) or (global_var(GVAR_CHAMPION_REPUTATION) == 1)) then begin
             reaction := reaction + 20;
         end
-        if ((global_var(159) > (2 * global_var(4))) or (global_var(3) == 1)) then begin
+        if ((global_var(GVAR_GECKO_REQ_FORM) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1)) then begin
             reaction := reaction - 20;
         end
         call ReactToLevel();
@@ -580,4 +580,3 @@ begin
     return 0;
     LVar0 := 0;
 end
-
