diff --git a/Mods/MegaMod/Original/LEYLA.ssl b/Mods/MegaMod/ReDefine/LEYLA.ssl
index 8941659..ca30336 100644
--- a/Mods/MegaMod/Original/LEYLA.ssl
+++ b/Mods/MegaMod/ReDefine/LEYLA.ssl
@@ -194,17 +194,17 @@ begin
     if ((local_var(5) == 2) and obj_can_see_obj(self_obj, dude_obj)) then begin
         set_local_var(5, 1);
     end
-    if ((local_var(5) == 1) or (global_var(829) == 1) and obj_can_see_obj(self_obj, dude_obj)) then begin
-        attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+    if ((local_var(5) == 1) or (global_var(GVAR_COLLY_FLAGS_2) == 1) and obj_can_see_obj(self_obj, dude_obj)) then begin
+        attack(dude_obj);
     end
     if (((local_var(8) bwand 2048) != 0) and (((local_var(8) bwand 4096) != 0) == 0)) then begin
-        gfade_out(1);
+        fade_out;
         critter_attempt_placement(self_obj, 15084, elevation(self_obj));
         anim(self_obj, 1000, 2);
         critter_attempt_placement(dude_obj, 15284, elevation(self_obj));
         anim(dude_obj, 1000, 5);
         metarule3(108, 16685, 0, 0);
-        gfade_in(1);
+        fade_in;
         set_local_var(8, local_var(8) bwor 4096);
         if (in_dialog == 0) then begin
             in_dialog := 1;
@@ -220,9 +220,9 @@ begin
         end
     end
     if (((local_var(8) bwand 4096) != 0) and (((local_var(8) bwand 8192) != 0) == 0)) then begin
-        gfade_out(1);
+        fade_out;
         game_time_advance(600 * random(15, 40));
-        gfade_in(1);
+        fade_in;
         if (in_dialog == 0) then begin
             in_dialog := 1;
             start_gdialog(28, self_obj, 4, -1, -1);
@@ -236,20 +236,20 @@ begin
             call Node013();
         end
     end
-    if (((global_var(830) bwand 1048576) != 0) and (tile_num(self_obj) == 15084)) then begin
-        gfade_out(1);
+    if (((global_var(GVAR_TOWN_REP_COLLY) bwand 1048576) != 0) and (tile_num(self_obj) == 15084)) then begin
+        fade_out;
         critter_attempt_placement(self_obj, 20672, elevation(self_obj));
         anim(self_obj, 1000, 2);
-        set_global_var(829, 1);
-        gfade_in(1);
+        set_global_var(GVAR_COLLY_FLAGS_2, 1);
+        fade_in;
     end
     if (((local_var(8) bwand 65536) != 0) and (tile_num(self_obj) == 15084)) then begin
-        gfade_out(1);
+        fade_out;
         critter_attempt_placement(self_obj, 20672, elevation(self_obj));
         anim(self_obj, 1000, 2);
-        gfade_in(1);
+        fade_in;
     end
-    if ((global_var(830) bwand 536870912) != 0) then begin
+    if ((global_var(GVAR_TOWN_REP_COLLY) bwand 536870912) != 0) then begin
         kill_critter(self_obj, 49);
     end
 end
@@ -263,11 +263,11 @@ end
 
 procedure talk_p_proc
 begin
-    if ((global_var(830) bwand 8) != 0) then begin
+    if ((global_var(GVAR_TOWN_REP_COLLY) bwand 8) != 0) then begin
         call Node126();
     end
     else begin
-        if (global_var(834) > 0) then begin
+        if (global_var(GVAR_BERK_BOOK) > 0) then begin
             call final();
         end
         else begin
@@ -320,7 +320,7 @@ begin
                             end
                         end
                         else begin
-                            if ((global_var(830) bwand 4) != 0) then begin
+                            if ((global_var(GVAR_TOWN_REP_COLLY) bwand 4) != 0) then begin
                                 if (local_var(10) > 0) then begin
                                     if (in_dialog == 0) then begin
                                         in_dialog := 1;
@@ -390,60 +390,60 @@ end
 
 procedure destroy_p_proc
 begin
-    set_global_var(830, global_var(830) bwor 536870912);
+    set_global_var(GVAR_TOWN_REP_COLLY, global_var(GVAR_TOWN_REP_COLLY) bwor 536870912);
     if (source_obj == dude_obj) then begin
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(4) > (3 * global_var(5))) or (global_var(2) == 1)) and (global_var(1) == 0)) then begin
-            set_global_var(2, 1);
-            set_global_var(3, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(4) > (3 * global_var(GVAR_GOOD_MONSTER))) or (global_var(GVAR_CHAMPION_REPUTATION) == 1)) and (global_var(GVAR_CHILDKILLER_REPUTATION) == 0)) then begin
+            set_global_var(GVAR_CHAMPION_REPUTATION, 1);
+            set_global_var(GVAR_BERSERKER_REPUTATION, 0);
         end
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(5) > (2 * global_var(4))) or (global_var(3) == 1))) then begin
-            set_global_var(3, 1);
-            set_global_var(2, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
+            set_global_var(GVAR_BERSERKER_REPUTATION, 1);
+            set_global_var(GVAR_CHAMPION_REPUTATION, 0);
         end
-        set_global_var(0, global_var(0) + 0);
+        set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + 0);
         debug_msg("Player gains " + 0 + " Karma Points.");
-        set_global_var(37, 0);
-        set_global_var(38, 0);
-        set_global_var(39, 0);
-        set_global_var(40, 0);
-        set_global_var(41, 0);
-        set_global_var(42, 0);
-        set_global_var(43, 0);
-        set_global_var(44, 0);
-        set_global_var(45, 0);
-        if (global_var(0) >= 1000) then begin
-            set_global_var(37, 1);
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+        set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+        set_global_var(GVAR_KARMA_DEFENDER, 0);
+        set_global_var(GVAR_KARMA_WANDERER, 0);
+        set_global_var(GVAR_KARMA_BETRAYER, 0);
+        set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+        set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+            set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
         end
         else begin
-            if (global_var(0) >= 750) then begin
-                set_global_var(38, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
             end
             else begin
-                if (global_var(0) >= 500) then begin
-                    set_global_var(39, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                 end
                 else begin
-                    if (global_var(0) >= 250) then begin
-                        set_global_var(40, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                        set_global_var(GVAR_KARMA_DEFENDER, 1);
                     end
                     else begin
-                        if (global_var(0) > -250) then begin
-                            set_global_var(41, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                            set_global_var(GVAR_KARMA_WANDERER, 1);
                         end
                         else begin
-                            if (global_var(0) > -500) then begin
-                                set_global_var(42, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                set_global_var(GVAR_KARMA_BETRAYER, 1);
                             end
                             else begin
-                                if (global_var(0) > -750) then begin
-                                    set_global_var(43, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -1000) then begin
-                                        set_global_var(44, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                     end
                                     else begin
-                                        set_global_var(45, 1);
+                                        set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                     end
                                 end
                             end
@@ -460,17 +460,17 @@ begin
     script_overrides;
     if (local_var(6) == 0) then begin
         set_local_var(6, 1);
-        display_msg(message_str(28, 100));
+        display_msg(message_str(SCRIPT_ARBRIDGE, 100));
     end
     else begin
-        display_msg(message_str(28, 101));
+        display_msg(message_str(SCRIPT_ARBRIDGE, 101));
     end
 end
 
 procedure description_p_proc
 begin
     script_overrides;
-    display_msg(message_str(28, 105));
+    display_msg(message_str(SCRIPT_ARBRIDGE, 105));
 end
 
 procedure use_skill_on_p_proc
@@ -481,7 +481,7 @@ procedure damage_p_proc
 begin
     if (source_obj == dude_obj) then begin
         if (((local_var(8) bwand 32768) != 0) == 0) then begin
-            set_global_var(829, 1);
+            set_global_var(GVAR_COLLY_FLAGS_2, 1);
         end
         set_local_var(5, 1);
     end
@@ -489,11 +489,11 @@ end
 
 procedure map_enter_p_proc
 begin
-    if (metarule(14, 0)) then begin
-        if (metarule(22, 0)) then begin
-            critter_add_trait(self_obj, 1, 5, 22);
-            critter_mod_skill(self_obj, 4, 20);
-            critter_mod_skill(self_obj, 3, 20);
+    if (map_first_run) then begin
+        if (is_loading_game) then begin
+            critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_TOUGH_GUARD);
+            critter_mod_skill(self_obj, SKILL_MELEE, 20);
+            critter_mod_skill(self_obj, SKILL_UNARMED_COMBAT, 20);
         end
     end
     if (((local_var(8) bwand 8192) != 0) and (((local_var(8) bwand 65536) != 0) == 0)) then begin
@@ -502,7 +502,7 @@ begin
         if (tile_num(self_obj) != 20672) then begin
             critter_attempt_placement(self_obj, 20672, elevation(self_obj));
             anim(self_obj, 1000, 2);
-            critter_add_trait(self_obj, 1, 6, 5);
+            critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 5);
             set_local_var(8, local_var(8) bwand (-1 - 32768));
         end
     end
@@ -526,64 +526,64 @@ procedure Node001
 begin
     set_local_var(10, 1);
     set_local_var(7, 1);
-    if ((global_var(830) bwand 2) != 0) then begin
-        gsay_reply(28, message_str(28, 201));
+    if ((global_var(GVAR_TOWN_REP_COLLY) bwand 2) != 0) then begin
+        gsay_reply(SCRIPT_ARBRIDGE, message_str(SCRIPT_ARBRIDGE, 201));
     end
     else begin
-        gsay_reply(28, message_str(28, 200));
+        gsay_reply(SCRIPT_ARBRIDGE, message_str(SCRIPT_ARBRIDGE, 200));
     end
-    giq_option(4, 28, 202, Node002, 50);
-    giq_option(4, 28, message_str(28, 203) + obj_name(dude_obj) + message_str(28, 204), Node022, 50);
-    if ((global_var(830) bwand 2) != 0) then begin
-        giq_option(4, 28, 205, Node014, 50);
+    giq_option(4, SCRIPT_ARBRIDGE, 202, Node002, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_ARBRIDGE, message_str(SCRIPT_ARBRIDGE, 203) + dude_name + message_str(SCRIPT_ARBRIDGE, 204), Node022, NEUTRAL_REACTION);
+    if ((global_var(GVAR_TOWN_REP_COLLY) bwand 2) != 0) then begin
+        giq_option(4, SCRIPT_ARBRIDGE, 205, Node014, NEUTRAL_REACTION);
     end
-    giq_option(4, 28, 206, Node027, 50);
+    giq_option(4, SCRIPT_ARBRIDGE, 206, Node027, NEUTRAL_REACTION);
 end
 
 procedure Node002
 begin
-    gsay_reply(28, message_str(28, 210));
-    giq_option(4, 28, 211, Node003, 50);
-    giq_option(4, 28, 212, Node999, 50);
+    gsay_reply(SCRIPT_ARBRIDGE, message_str(SCRIPT_ARBRIDGE, 210));
+    giq_option(4, SCRIPT_ARBRIDGE, 211, Node003, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_ARBRIDGE, 212, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node003
 begin
-    gsay_reply(28, message_str(28, 220));
-    giq_option(4, 28, 221, Node004, 50);
-    giq_option(4, 28, 222, Node004, 50);
-    giq_option(4, 28, 223, Node023, 50);
-    giq_option(4, 28, 224, Node999, 50);
+    gsay_reply(SCRIPT_ARBRIDGE, message_str(SCRIPT_ARBRIDGE, 220));
+    giq_option(4, SCRIPT_ARBRIDGE, 221, Node004, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_ARBRIDGE, 222, Node004, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_ARBRIDGE, 223, Node023, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_ARBRIDGE, 224, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node004
 begin
-    gsay_reply(28, message_str(28, 230));
-    giq_option(4, 28, 231, Node005, 50);
+    gsay_reply(SCRIPT_ARBRIDGE, message_str(SCRIPT_ARBRIDGE, 230));
+    giq_option(4, SCRIPT_ARBRIDGE, 231, Node005, NEUTRAL_REACTION);
     if (obj_is_carrying_obj_pid(dude_obj, 314) + obj_is_carrying_obj_pid(dude_obj, 315) + obj_is_carrying_obj_pid(dude_obj, 316)) then begin
-        giq_option(4, 28, 232, Node007, 50);
+        giq_option(4, SCRIPT_ARBRIDGE, 232, Node007, NEUTRAL_REACTION);
     end
-    giq_option(4, 28, 233, Node999, 50);
+    giq_option(4, SCRIPT_ARBRIDGE, 233, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node005
 begin
-    gsay_message(28, message_str(28, 240), 50);
+    gsay_message(SCRIPT_ARBRIDGE, message_str(SCRIPT_ARBRIDGE, 240), NEUTRAL_REACTION);
     call Node006();
 end
 
 procedure Node006
 begin
-    gsay_reply(28, message_str(28, 250));
-    giq_option(4, 28, 251, Node030, 50);
-    giq_option(4, 28, 252, Node999, 50);
+    gsay_reply(SCRIPT_ARBRIDGE, message_str(SCRIPT_ARBRIDGE, 250));
+    giq_option(4, SCRIPT_ARBRIDGE, 251, Node030, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_ARBRIDGE, 252, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node007
 begin
-    gsay_reply(28, message_str(28, 260));
-    giq_option(4, 28, 261, Node008, 50);
-    giq_option(4, 28, 262, Node017, 50);
+    gsay_reply(SCRIPT_ARBRIDGE, message_str(SCRIPT_ARBRIDGE, 260));
+    giq_option(4, SCRIPT_ARBRIDGE, 261, Node008, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_ARBRIDGE, 262, Node017, NEUTRAL_REACTION);
 end
 
 procedure Node008
@@ -608,24 +608,24 @@ begin
             amt_ower := amt_ower - 1;
         end
     end
-    gsay_reply(28, message_str(28, 270));
-    giq_option(4, 28, 271, Node009, 50);
-    giq_option(4, 28, 272, Node009, 50);
+    gsay_reply(SCRIPT_ARBRIDGE, message_str(SCRIPT_ARBRIDGE, 270));
+    giq_option(4, SCRIPT_ARBRIDGE, 271, Node009, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_ARBRIDGE, 272, Node009, NEUTRAL_REACTION);
 end
 
 procedure Node009
 begin
-    gsay_reply(28, message_str(28, 280));
-    giq_option(4, 28, 281, Node010, 50);
-    giq_option(4, 28, 282, Node010, 50);
-    giq_option(4, 28, 283, Node010, 50);
+    gsay_reply(SCRIPT_ARBRIDGE, message_str(SCRIPT_ARBRIDGE, 280));
+    giq_option(4, SCRIPT_ARBRIDGE, 281, Node010, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_ARBRIDGE, 282, Node010, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_ARBRIDGE, 283, Node010, NEUTRAL_REACTION);
 end
 
 procedure Node010
 begin
-    gsay_reply(28, message_str(28, 290));
-    giq_option(4, 28, 291, Node010a, 50);
-    giq_option(4, 28, 292, Node010a, 50);
+    gsay_reply(SCRIPT_ARBRIDGE, message_str(SCRIPT_ARBRIDGE, 290));
+    giq_option(4, SCRIPT_ARBRIDGE, 291, Node010a, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_ARBRIDGE, 292, Node010a, NEUTRAL_REACTION);
 end
 
 procedure Node010a
@@ -635,34 +635,34 @@ end
 
 procedure Node011
 begin
-    gsay_reply(28, message_str(28, 300));
-    giq_option(4, 28, 301, Node012, 50);
-    giq_option(4, 28, 302, Node012, 50);
+    gsay_reply(SCRIPT_ARBRIDGE, message_str(SCRIPT_ARBRIDGE, 300));
+    giq_option(4, SCRIPT_ARBRIDGE, 301, Node012, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_ARBRIDGE, 302, Node012, NEUTRAL_REACTION);
 end
 
 procedure Node012
 begin
     if ((obj_is_carrying_obj_pid(self_obj, 314) + obj_is_carrying_obj_pid(self_obj, 315) + obj_is_carrying_obj_pid(self_obj, 316)) > 0) then begin
-        gsay_message(28, message_str(28, 310) + message_str(28, 311) + message_str(28, 312), 50);
+        gsay_message(SCRIPT_ARBRIDGE, message_str(SCRIPT_ARBRIDGE, 310) + message_str(SCRIPT_ARBRIDGE, 311) + message_str(SCRIPT_ARBRIDGE, 312), NEUTRAL_REACTION);
     end
     else begin
-        gsay_message(28, message_str(28, 310) + message_str(28, 312), 50);
+        gsay_message(SCRIPT_ARBRIDGE, message_str(SCRIPT_ARBRIDGE, 310) + message_str(SCRIPT_ARBRIDGE, 312), NEUTRAL_REACTION);
     end
 end
 
 procedure Node013
 begin
-    display_msg(message_str(28, 3000));
+    display_msg(message_str(SCRIPT_ARBRIDGE, 3000));
     give_exp_points(350);
     set_local_var(8, local_var(8) bwor 8192);
-    if (((global_var(830) bwand 2) != 0) or ((global_var(830) bwand 256) != 0) or (get_critter_stat(dude_obj, 3) > 6)) then begin
+    if (((global_var(GVAR_TOWN_REP_COLLY) bwand 2) != 0) or ((global_var(GVAR_TOWN_REP_COLLY) bwand 256) != 0) or (dude_charisma > 6)) then begin
         set_local_var(8, local_var(8) bwor 131072);
     end
-    gsay_reply(28, message_str(28, 320));
-    giq_option(4, 28, 321, Node053, 50);
-    giq_option(4, 28, 324, Node052, 50);
-    giq_option(4, 28, 322, Node033, 50);
-    giq_option(4, 28, 323, leyla_gone, 50);
+    gsay_reply(SCRIPT_ARBRIDGE, message_str(SCRIPT_ARBRIDGE, 320));
+    giq_option(4, SCRIPT_ARBRIDGE, 321, Node053, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_ARBRIDGE, 324, Node052, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_ARBRIDGE, 322, Node033, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_ARBRIDGE, 323, leyla_gone, NEUTRAL_REACTION);
 end
 
 procedure leyla_gone
@@ -672,223 +672,223 @@ end
 
 procedure Node014
 begin
-    gsay_reply(28, message_str(28, 330));
-    giq_option(4, 28, 331, Node015, 50);
-    giq_option(4, 28, 332, Node017, 50);
+    gsay_reply(SCRIPT_ARBRIDGE, message_str(SCRIPT_ARBRIDGE, 330));
+    giq_option(4, SCRIPT_ARBRIDGE, 331, Node015, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_ARBRIDGE, 332, Node017, NEUTRAL_REACTION);
 end
 
 procedure Node015
 begin
-    gsay_reply(28, message_str(28, 340));
-    giq_option(4, 28, 341, Node016, 50);
+    gsay_reply(SCRIPT_ARBRIDGE, message_str(SCRIPT_ARBRIDGE, 340));
+    giq_option(4, SCRIPT_ARBRIDGE, 341, Node016, NEUTRAL_REACTION);
 end
 
 procedure Node016
 begin
-    gsay_reply(28, message_str(28, 350));
-    giq_option(4, 28, 351, Node007, 50);
-    giq_option(4, 28, 352, Node999, 50);
+    gsay_reply(SCRIPT_ARBRIDGE, message_str(SCRIPT_ARBRIDGE, 350));
+    giq_option(4, SCRIPT_ARBRIDGE, 351, Node007, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_ARBRIDGE, 352, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node017
 begin
-    gsay_reply(28, message_str(28, 360));
-    giq_option(4, 28, 361, Node020, 50);
-    giq_option(4, 28, 362, Node019, 50);
-    giq_option(4, 28, 363, Node018, 50);
+    gsay_reply(SCRIPT_ARBRIDGE, message_str(SCRIPT_ARBRIDGE, 360));
+    giq_option(4, SCRIPT_ARBRIDGE, 361, Node020, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_ARBRIDGE, 362, Node019, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_ARBRIDGE, 363, Node018, NEUTRAL_REACTION);
 end
 
 procedure Node018
 begin
-    gsay_reply(28, message_str(28, 370));
-    giq_option(4, 28, 371, Node999, 50);
+    gsay_reply(SCRIPT_ARBRIDGE, message_str(SCRIPT_ARBRIDGE, 370));
+    giq_option(4, SCRIPT_ARBRIDGE, 371, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node019
 begin
-    gsay_reply(28, message_str(28, 380));
-    giq_option(4, 28, 381, set_holy_anger, 50);
-    giq_option(4, 28, 382, set_holy_anger, 50);
+    gsay_reply(SCRIPT_ARBRIDGE, message_str(SCRIPT_ARBRIDGE, 380));
+    giq_option(4, SCRIPT_ARBRIDGE, 381, set_holy_anger, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_ARBRIDGE, 382, set_holy_anger, NEUTRAL_REACTION);
 end
 
 procedure Node020
 begin
-    gsay_reply(28, message_str(28, 390));
-    giq_option(4, 28, 391, Node021, 50);
-    giq_option(4, 28, 392, set_holy_anger, 50);
+    gsay_reply(SCRIPT_ARBRIDGE, message_str(SCRIPT_ARBRIDGE, 390));
+    giq_option(4, SCRIPT_ARBRIDGE, 391, Node021, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_ARBRIDGE, 392, set_holy_anger, NEUTRAL_REACTION);
 end
 
 procedure Node021
 begin
-    gsay_message(28, message_str(28, 400), 50);
+    gsay_message(SCRIPT_ARBRIDGE, message_str(SCRIPT_ARBRIDGE, 400), NEUTRAL_REACTION);
     call Node998();
 end
 
 procedure Node022
 begin
-    if ((global_var(830) bwand 2) != 0) then begin
-        gsay_reply(28, message_str(28, 411));
+    if ((global_var(GVAR_TOWN_REP_COLLY) bwand 2) != 0) then begin
+        gsay_reply(SCRIPT_ARBRIDGE, message_str(SCRIPT_ARBRIDGE, 411));
     end
     else begin
-        gsay_reply(28, message_str(28, 410));
+        gsay_reply(SCRIPT_ARBRIDGE, message_str(SCRIPT_ARBRIDGE, 410));
     end
-    giq_option(4, 28, 412, Node027, 50);
-    giq_option(4, 28, 413, Node023, 50);
-    giq_option(4, 28, 414, Node999, 50);
+    giq_option(4, SCRIPT_ARBRIDGE, 412, Node027, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_ARBRIDGE, 413, Node023, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_ARBRIDGE, 414, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node023
 begin
-    gsay_reply(28, message_str(28, 420));
-    giq_option(4, 28, 421, Node024, 50);
-    giq_option(4, 28, 422, Node999, 50);
+    gsay_reply(SCRIPT_ARBRIDGE, message_str(SCRIPT_ARBRIDGE, 420));
+    giq_option(4, SCRIPT_ARBRIDGE, 421, Node024, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_ARBRIDGE, 422, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node024
 begin
-    gsay_reply(28, message_str(28, 430));
-    giq_option(4, 28, 431, Node025, 50);
-    giq_option(4, 28, 432, Node999, 50);
+    gsay_reply(SCRIPT_ARBRIDGE, message_str(SCRIPT_ARBRIDGE, 430));
+    giq_option(4, SCRIPT_ARBRIDGE, 431, Node025, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_ARBRIDGE, 432, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node025
 begin
-    gsay_reply(28, message_str(28, 440));
-    giq_option(4, 28, 441, Node026, 50);
-    giq_option(4, 28, 442, Node999, 50);
+    gsay_reply(SCRIPT_ARBRIDGE, message_str(SCRIPT_ARBRIDGE, 440));
+    giq_option(4, SCRIPT_ARBRIDGE, 441, Node026, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_ARBRIDGE, 442, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node026
 begin
-    gsay_reply(28, message_str(28, 450));
-    giq_option(4, 28, 451, Node999, 50);
-    giq_option(4, 28, 452, Node999, 50);
-    giq_option(4, 28, 453, Node999, 50);
+    gsay_reply(SCRIPT_ARBRIDGE, message_str(SCRIPT_ARBRIDGE, 450));
+    giq_option(4, SCRIPT_ARBRIDGE, 451, Node999, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_ARBRIDGE, 452, Node999, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_ARBRIDGE, 453, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node027
 begin
-    gsay_reply(28, message_str(28, 460));
-    giq_option(4, 28, 461, Node028, 50);
+    gsay_reply(SCRIPT_ARBRIDGE, message_str(SCRIPT_ARBRIDGE, 460));
+    giq_option(4, SCRIPT_ARBRIDGE, 461, Node028, NEUTRAL_REACTION);
     if (obj_is_carrying_obj_pid(dude_obj, 314) + obj_is_carrying_obj_pid(dude_obj, 315) + obj_is_carrying_obj_pid(dude_obj, 316)) then begin
-        giq_option(4, 28, 462, Node008, 50);
+        giq_option(4, SCRIPT_ARBRIDGE, 462, Node008, NEUTRAL_REACTION);
     end
-    giq_option(4, 28, 463, Node023, 50);
+    giq_option(4, SCRIPT_ARBRIDGE, 463, Node023, NEUTRAL_REACTION);
 end
 
 procedure Node028
 begin
-    gsay_reply(28, message_str(28, 470));
-    giq_option(4, 28, 471, Node029, 50);
+    gsay_reply(SCRIPT_ARBRIDGE, message_str(SCRIPT_ARBRIDGE, 470));
+    giq_option(4, SCRIPT_ARBRIDGE, 471, Node029, NEUTRAL_REACTION);
 end
 
 procedure Node029
 begin
-    gsay_reply(28, message_str(28, 480));
-    giq_option(4, 28, 481, Node999, 50);
-    giq_option(4, 28, 482, Node999, 50);
+    gsay_reply(SCRIPT_ARBRIDGE, message_str(SCRIPT_ARBRIDGE, 480));
+    giq_option(4, SCRIPT_ARBRIDGE, 481, Node999, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_ARBRIDGE, 482, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node030
 begin
-    gsay_reply(28, message_str(28, 490));
-    giq_option(4, 28, 491, Node031, 50);
-    giq_option(4, 28, 492, Node017, 50);
-    giq_option(4, 28, 493, Node999, 50);
+    gsay_reply(SCRIPT_ARBRIDGE, message_str(SCRIPT_ARBRIDGE, 490));
+    giq_option(4, SCRIPT_ARBRIDGE, 491, Node031, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_ARBRIDGE, 492, Node017, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_ARBRIDGE, 493, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node031
 begin
-    gsay_reply(28, message_str(28, 500));
-    giq_option(4, 28, 501, Node999, 50);
+    gsay_reply(SCRIPT_ARBRIDGE, message_str(SCRIPT_ARBRIDGE, 500));
+    giq_option(4, SCRIPT_ARBRIDGE, 501, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node032
 begin
-    gsay_reply(28, message_str(28, random(510, 513)));
-    giq_option(4, 28, random(514, 517), Node999, 50);
+    gsay_reply(SCRIPT_ARBRIDGE, message_str(SCRIPT_ARBRIDGE, random(510, 513)));
+    giq_option(4, SCRIPT_ARBRIDGE, random(514, 517), Node999, NEUTRAL_REACTION);
 end
 
 procedure Node033
 begin
-    gsay_reply(28, message_str(28, 520));
-    giq_option(4, 28, 521, Node036, 50);
-    giq_option(4, 28, 522, Node034, 50);
+    gsay_reply(SCRIPT_ARBRIDGE, message_str(SCRIPT_ARBRIDGE, 520));
+    giq_option(4, SCRIPT_ARBRIDGE, 521, Node036, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_ARBRIDGE, 522, Node034, NEUTRAL_REACTION);
 end
 
 procedure Node034
 begin
-    gsay_reply(28, message_str(28, 530));
-    giq_option(4, 28, 531, Node035, 50);
+    gsay_reply(SCRIPT_ARBRIDGE, message_str(SCRIPT_ARBRIDGE, 530));
+    giq_option(4, SCRIPT_ARBRIDGE, 531, Node035, NEUTRAL_REACTION);
 end
 
 procedure Node035
 begin
     set_local_var(8, local_var(8) bwor 65536);
-    gsay_reply(28, message_str(28, 540));
-    giq_option(4, 28, 541, Node999, 50);
-    giq_option(4, 28, 542, Node999, 50);
+    gsay_reply(SCRIPT_ARBRIDGE, message_str(SCRIPT_ARBRIDGE, 540));
+    giq_option(4, SCRIPT_ARBRIDGE, 541, Node999, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_ARBRIDGE, 542, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node036
 begin
     set_local_var(5, 1);
-    gsay_reply(28, message_str(28, 550));
-    giq_option(4, 28, 551, Node049, 50);
-    giq_option(4, 28, 552, Node042, 50);
-    giq_option(4, 28, 553, Node037, 50);
+    gsay_reply(SCRIPT_ARBRIDGE, message_str(SCRIPT_ARBRIDGE, 550));
+    giq_option(4, SCRIPT_ARBRIDGE, 551, Node049, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_ARBRIDGE, 552, Node042, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_ARBRIDGE, 553, Node037, NEUTRAL_REACTION);
 end
 
 procedure Node037
 begin
-    gsay_reply(28, message_str(28, 560));
-    giq_option(4, 28, 561, Node038, 50);
-    giq_option(4, 28, 562, revenge, 50);
+    gsay_reply(SCRIPT_ARBRIDGE, message_str(SCRIPT_ARBRIDGE, 560));
+    giq_option(4, SCRIPT_ARBRIDGE, 561, Node038, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_ARBRIDGE, 562, revenge, NEUTRAL_REACTION);
 end
 
 procedure Node038
 begin
-    display_msg(message_str(28, 3002));
+    display_msg(message_str(SCRIPT_ARBRIDGE, 3002));
     give_exp_points(300);
-    gsay_reply(28, message_str(28, 570));
-    giq_option(4, 28, 571, Node039a, 50);
-    giq_option(4, 28, 572, Node039b, 50);
+    gsay_reply(SCRIPT_ARBRIDGE, message_str(SCRIPT_ARBRIDGE, 570));
+    giq_option(4, SCRIPT_ARBRIDGE, 571, Node039a, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_ARBRIDGE, 572, Node039b, NEUTRAL_REACTION);
 end
 
 procedure Node039a
 begin
-    gsay_reply(28, message_str(28, 580));
-    giq_option(4, 28, 582, Node040, 50);
-    if (get_critter_stat(dude_obj, 1) > 5) then begin
-        giq_option(4, 28, 583, Node041, 50);
+    gsay_reply(SCRIPT_ARBRIDGE, message_str(SCRIPT_ARBRIDGE, 580));
+    giq_option(4, SCRIPT_ARBRIDGE, 582, Node040, NEUTRAL_REACTION);
+    if (dude_perception > 5) then begin
+        giq_option(4, SCRIPT_ARBRIDGE, 583, Node041, NEUTRAL_REACTION);
     end
 end
 
 procedure Node039b
 begin
-    gsay_reply(28, message_str(28, 581));
-    giq_option(4, 28, 582, Node040, 50);
-    if (get_critter_stat(dude_obj, 1) > 5) then begin
-        giq_option(4, 28, 583, Node041, 50);
+    gsay_reply(SCRIPT_ARBRIDGE, message_str(SCRIPT_ARBRIDGE, 581));
+    giq_option(4, SCRIPT_ARBRIDGE, 582, Node040, NEUTRAL_REACTION);
+    if (dude_perception > 5) then begin
+        giq_option(4, SCRIPT_ARBRIDGE, 583, Node041, NEUTRAL_REACTION);
     end
 end
 
 procedure Node040
 begin
-    critter_dmg(dude_obj, get_critter_stat(dude_obj, 35) - 6, 0);
-    gsay_reply(28, message_str(28, 590));
-    giq_option(4, 28, 591, solo_fight, 50);
+    critter_dmg(dude_obj, get_critter_stat(dude_obj, STAT_current_hp) - 6, 0);
+    gsay_reply(SCRIPT_ARBRIDGE, message_str(SCRIPT_ARBRIDGE, 590));
+    giq_option(4, SCRIPT_ARBRIDGE, 591, solo_fight, NEUTRAL_REACTION);
 end
 
 procedure Node041
 begin
-    gsay_reply(28, message_str(28, 600));
-    giq_option(4, 28, 601, solo_fight, 50);
+    gsay_reply(SCRIPT_ARBRIDGE, message_str(SCRIPT_ARBRIDGE, 600));
+    giq_option(4, SCRIPT_ARBRIDGE, 601, solo_fight, NEUTRAL_REACTION);
 end
 
 procedure Node042
 begin
-    if (((global_var(830) bwand 4) != 0) or (has_skill(dude_obj, 14) >= 47)) then begin
+    if (((global_var(GVAR_TOWN_REP_COLLY) bwand 4) != 0) or (has_skill(dude_obj, SKILL_CONVERSANT) >= 47)) then begin
         call Node042a();
     end
     else begin
@@ -898,56 +898,56 @@ end
 
 procedure Node042a
 begin
-    gsay_reply(28, message_str(28, 610));
-    giq_option(4, 28, 611, Node043, 50);
+    gsay_reply(SCRIPT_ARBRIDGE, message_str(SCRIPT_ARBRIDGE, 610));
+    giq_option(4, SCRIPT_ARBRIDGE, 611, Node043, NEUTRAL_REACTION);
 end
 
 procedure Node043
 begin
-    set_global_var(0, global_var(0) + -15);
+    set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + -15);
     debug_msg("Player gains " + -15 + " Karma Points.");
-    set_global_var(37, 0);
-    set_global_var(38, 0);
-    set_global_var(39, 0);
-    set_global_var(40, 0);
-    set_global_var(41, 0);
-    set_global_var(42, 0);
-    set_global_var(43, 0);
-    set_global_var(44, 0);
-    set_global_var(45, 0);
-    if (global_var(0) >= 1000) then begin
-        set_global_var(37, 1);
+    set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+    set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+    set_global_var(GVAR_KARMA_DEFENDER, 0);
+    set_global_var(GVAR_KARMA_WANDERER, 0);
+    set_global_var(GVAR_KARMA_BETRAYER, 0);
+    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+    set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+    set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+    if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
     end
     else begin
-        if (global_var(0) >= 750) then begin
-            set_global_var(38, 1);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+            set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
         end
         else begin
-            if (global_var(0) >= 500) then begin
-                set_global_var(39, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
             end
             else begin
-                if (global_var(0) >= 250) then begin
-                    set_global_var(40, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                    set_global_var(GVAR_KARMA_DEFENDER, 1);
                 end
                 else begin
-                    if (global_var(0) > -250) then begin
-                        set_global_var(41, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                        set_global_var(GVAR_KARMA_WANDERER, 1);
                     end
                     else begin
-                        if (global_var(0) > -500) then begin
-                            set_global_var(42, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                            set_global_var(GVAR_KARMA_BETRAYER, 1);
                         end
                         else begin
-                            if (global_var(0) > -750) then begin
-                                set_global_var(43, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                             end
                             else begin
-                                if (global_var(0) > -1000) then begin
-                                    set_global_var(44, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                    set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                 end
                                 else begin
-                                    set_global_var(45, 1);
+                                    set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                 end
                             end
                         end
@@ -956,60 +956,60 @@ begin
             end
         end
     end
-    display_msg(message_str(28, 3001));
+    display_msg(message_str(SCRIPT_ARBRIDGE, 3001));
     give_exp_points(300);
-    set_global_var(830, global_var(830) bwor 536870912);
-    gsay_reply(28, message_str(28, 620));
-    giq_option(4, 28, 621, Node997, 50);
-    giq_option(4, 28, 622, Node997, 50);
-    giq_option(4, 28, 623, Node997, 50);
+    set_global_var(GVAR_TOWN_REP_COLLY, global_var(GVAR_TOWN_REP_COLLY) bwor 536870912);
+    gsay_reply(SCRIPT_ARBRIDGE, message_str(SCRIPT_ARBRIDGE, 620));
+    giq_option(4, SCRIPT_ARBRIDGE, 621, Node997, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_ARBRIDGE, 622, Node997, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_ARBRIDGE, 623, Node997, NEUTRAL_REACTION);
 end
 
 procedure Node044
 begin
-    gsay_reply(28, message_str(28, 630));
-    giq_option(4, 28, 631, Node046, 50);
-    giq_option(4, 28, 632, Node045, 50);
+    gsay_reply(SCRIPT_ARBRIDGE, message_str(SCRIPT_ARBRIDGE, 630));
+    giq_option(4, SCRIPT_ARBRIDGE, 631, Node046, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_ARBRIDGE, 632, Node045, NEUTRAL_REACTION);
 end
 
 procedure Node045
 begin
-    gsay_reply(28, message_str(28, 640));
-    giq_option(4, 28, 641, revenge, 50);
-    giq_option(4, 28, 642, revenge, 50);
+    gsay_reply(SCRIPT_ARBRIDGE, message_str(SCRIPT_ARBRIDGE, 640));
+    giq_option(4, SCRIPT_ARBRIDGE, 641, revenge, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_ARBRIDGE, 642, revenge, NEUTRAL_REACTION);
 end
 
 procedure Node046
 begin
-    gsay_reply(28, message_str(28, 650));
-    giq_option(4, 28, 651, Node047, 50);
-    giq_option(4, 28, 652, Node048, 50);
+    gsay_reply(SCRIPT_ARBRIDGE, message_str(SCRIPT_ARBRIDGE, 650));
+    giq_option(4, SCRIPT_ARBRIDGE, 651, Node047, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_ARBRIDGE, 652, Node048, NEUTRAL_REACTION);
 end
 
 procedure Node047
 begin
-    gsay_reply(28, message_str(28, 660));
-    giq_option(4, 28, 661, solo_fight, 50);
+    gsay_reply(SCRIPT_ARBRIDGE, message_str(SCRIPT_ARBRIDGE, 660));
+    giq_option(4, SCRIPT_ARBRIDGE, 661, solo_fight, NEUTRAL_REACTION);
 end
 
 procedure Node048
 begin
-    gsay_reply(28, message_str(28, 670));
-    giq_option(4, 28, 671, solo_fight, 50);
+    gsay_reply(SCRIPT_ARBRIDGE, message_str(SCRIPT_ARBRIDGE, 670));
+    giq_option(4, SCRIPT_ARBRIDGE, 671, solo_fight, NEUTRAL_REACTION);
 end
 
 procedure Node049
 begin
-    gsay_reply(28, message_str(28, 680));
-    giq_option(4, 28, 681, Node049a, 50);
-    giq_option(4, 28, 682, Node045, 50);
+    gsay_reply(SCRIPT_ARBRIDGE, message_str(SCRIPT_ARBRIDGE, 680));
+    giq_option(4, SCRIPT_ARBRIDGE, 681, Node049a, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_ARBRIDGE, 682, Node045, NEUTRAL_REACTION);
 end
 
 procedure Node049a
 begin
     variable LVar0 := 0;
     LVar0 := 2 + random(1, 8);
-    if (LVar0 > get_critter_stat(dude_obj, 0)) then begin
+    if (LVar0 > dude_strength) then begin
         call Node051();
     end
     else begin
@@ -1021,16 +1021,16 @@ end
 
 procedure Node050
 begin
-    gsay_reply(28, message_str(28, 690));
-    giq_option(4, 28, 691, Node050a, 50);
-    giq_option(4, 28, 692, Node045, 50);
+    gsay_reply(SCRIPT_ARBRIDGE, message_str(SCRIPT_ARBRIDGE, 690));
+    giq_option(4, SCRIPT_ARBRIDGE, 691, Node050a, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_ARBRIDGE, 692, Node045, NEUTRAL_REACTION);
 end
 
 procedure Node050a
 begin
     variable LVar0 := 0;
     LVar0 := 3 + random(1, 6);
-    if (LVar0 > get_critter_stat(dude_obj, 0)) then begin
+    if (LVar0 > dude_strength) then begin
         call Node038();
     end
     else begin
@@ -1042,32 +1042,32 @@ end
 
 procedure Node051
 begin
-    gsay_reply(28, message_str(28, 700));
-    giq_option(4, 28, 701, solo_fight, 50);
+    gsay_reply(SCRIPT_ARBRIDGE, message_str(SCRIPT_ARBRIDGE, 700));
+    giq_option(4, SCRIPT_ARBRIDGE, 701, solo_fight, NEUTRAL_REACTION);
 end
 
 procedure Node052
 begin
     set_local_var(8, local_var(8) bwand (-1 - 131072));
     set_local_var(8, local_var(8) bwor 65536);
-    gsay_reply(28, message_str(28, 710));
-    giq_option(4, 28, 711, Node999, 50);
-    giq_option(4, 28, 712, Node999, 50);
-    giq_option(4, 28, 713, Node999, 50);
+    gsay_reply(SCRIPT_ARBRIDGE, message_str(SCRIPT_ARBRIDGE, 710));
+    giq_option(4, SCRIPT_ARBRIDGE, 711, Node999, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_ARBRIDGE, 712, Node999, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_ARBRIDGE, 713, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node053
 begin
     set_local_var(8, local_var(8) bwor 65536);
-    gsay_reply(28, message_str(28, 720));
-    giq_option(4, 28, 721, Node054, 50);
-    giq_option(4, 28, 722, Node053a, 50);
-    giq_option(4, 28, 723, Node999, 50);
+    gsay_reply(SCRIPT_ARBRIDGE, message_str(SCRIPT_ARBRIDGE, 720));
+    giq_option(4, SCRIPT_ARBRIDGE, 721, Node054, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_ARBRIDGE, 722, Node053a, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_ARBRIDGE, 723, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node053a
 begin
-    if (((local_var(8) bwand 131072) != 0) or (has_skill(dude_obj, 14) >= 48)) then begin
+    if (((local_var(8) bwand 131072) != 0) or (has_skill(dude_obj, SKILL_CONVERSANT) >= 48)) then begin
         call Node057();
     end
     else begin
@@ -1077,273 +1077,273 @@ end
 
 procedure Node054
 begin
-    gsay_reply(28, message_str(28, 730));
-    giq_option(4, 28, 731, Node055, 50);
-    giq_option(4, 28, 732, Node999, 50);
+    gsay_reply(SCRIPT_ARBRIDGE, message_str(SCRIPT_ARBRIDGE, 730));
+    giq_option(4, SCRIPT_ARBRIDGE, 731, Node055, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_ARBRIDGE, 732, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node055
 begin
-    gsay_reply(28, message_str(28, 740));
-    giq_option(4, 28, 741, Node999, 50);
-    giq_option(4, 28, 742, Node999, 50);
-    giq_option(4, 28, 743, Node999, 50);
+    gsay_reply(SCRIPT_ARBRIDGE, message_str(SCRIPT_ARBRIDGE, 740));
+    giq_option(4, SCRIPT_ARBRIDGE, 741, Node999, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_ARBRIDGE, 742, Node999, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_ARBRIDGE, 743, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node056
 begin
-    gsay_reply(28, message_str(28, 750));
-    giq_option(4, 28, 751, Node999, 50);
+    gsay_reply(SCRIPT_ARBRIDGE, message_str(SCRIPT_ARBRIDGE, 750));
+    giq_option(4, SCRIPT_ARBRIDGE, 751, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node057
 begin
-    gsay_reply(28, message_str(28, 760));
-    giq_option(4, 28, 761, Node200, 50);
+    gsay_reply(SCRIPT_ARBRIDGE, message_str(SCRIPT_ARBRIDGE, 760));
+    giq_option(4, SCRIPT_ARBRIDGE, 761, Node200, NEUTRAL_REACTION);
 end
 
 procedure Node058
 begin
     if (local_var(9) == 2) then begin
-        gsay_reply(28, message_str(28, 769));
+        gsay_reply(SCRIPT_ARBRIDGE, message_str(SCRIPT_ARBRIDGE, 769));
     end
     else begin
         if ((local_var(8) bwand 131072) != 0) then begin
-            gsay_reply(28, message_str(28, 773));
+            gsay_reply(SCRIPT_ARBRIDGE, message_str(SCRIPT_ARBRIDGE, 773));
         end
         else begin
-            gsay_reply(28, message_str(28, random(770, 772)));
+            gsay_reply(SCRIPT_ARBRIDGE, message_str(SCRIPT_ARBRIDGE, random(770, 772)));
         end
     end
-    giq_option(4, 28, 774, Node200, 50);
+    giq_option(4, SCRIPT_ARBRIDGE, 774, Node200, NEUTRAL_REACTION);
     if (local_var(9) == 0) then begin
-        giq_option(4, 28, 775, Node059, 50);
+        giq_option(4, SCRIPT_ARBRIDGE, 775, Node059, NEUTRAL_REACTION);
     end
     if (local_var(9) == 1) then begin
         if ((local_var(8) bwand 131072) != 0) then begin
-            giq_option(4, 28, 776, Node066, 50);
+            giq_option(4, SCRIPT_ARBRIDGE, 776, Node066, NEUTRAL_REACTION);
         end
         else begin
-            giq_option(4, 28, 776, Node068, 50);
+            giq_option(4, SCRIPT_ARBRIDGE, 776, Node068, NEUTRAL_REACTION);
         end
     end
     if (local_var(9) == 2) then begin
-        giq_option(4, 28, 777, Node071, 50);
+        giq_option(4, SCRIPT_ARBRIDGE, 777, Node071, NEUTRAL_REACTION);
     end
-    giq_option(4, 28, 778, Node999, 50);
+    giq_option(4, SCRIPT_ARBRIDGE, 778, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node059
 begin
     set_local_var(9, 1);
-    gsay_reply(28, message_str(28, 780));
-    giq_option(4, 28, 781, Node060, 50);
-    giq_option(4, 28, 782, Node064, 50);
+    gsay_reply(SCRIPT_ARBRIDGE, message_str(SCRIPT_ARBRIDGE, 780));
+    giq_option(4, SCRIPT_ARBRIDGE, 781, Node060, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_ARBRIDGE, 782, Node064, NEUTRAL_REACTION);
 end
 
 procedure Node060
 begin
-    gsay_reply(28, message_str(28, 790));
+    gsay_reply(SCRIPT_ARBRIDGE, message_str(SCRIPT_ARBRIDGE, 790));
     if ((local_var(8) bwand 131072) != 0) then begin
-        giq_option(4, 28, 791, Node062, 50);
+        giq_option(4, SCRIPT_ARBRIDGE, 791, Node062, NEUTRAL_REACTION);
     end
-    giq_option(4, 28, 792, Node061, 50);
-    giq_option(4, 28, 793, Node999, 50);
+    giq_option(4, SCRIPT_ARBRIDGE, 792, Node061, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_ARBRIDGE, 793, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node061
 begin
-    gsay_reply(28, message_str(28, 800));
-    giq_option(4, 28, 801, Node999, 50);
+    gsay_reply(SCRIPT_ARBRIDGE, message_str(SCRIPT_ARBRIDGE, 800));
+    giq_option(4, SCRIPT_ARBRIDGE, 801, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node062
 begin
-    gsay_reply(28, message_str(28, 810));
-    giq_option(4, 28, 811, Node063, 50);
-    giq_option(4, 28, 812, Node999, 50);
+    gsay_reply(SCRIPT_ARBRIDGE, message_str(SCRIPT_ARBRIDGE, 810));
+    giq_option(4, SCRIPT_ARBRIDGE, 811, Node063, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_ARBRIDGE, 812, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node063
 begin
-    gsay_reply(28, message_str(28, 820));
-    giq_option(4, 28, 821, Node999, 50);
+    gsay_reply(SCRIPT_ARBRIDGE, message_str(SCRIPT_ARBRIDGE, 820));
+    giq_option(4, SCRIPT_ARBRIDGE, 821, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node064
 begin
     set_local_var(8, local_var(8) bwor 16384);
-    gsay_reply(28, message_str(28, 830));
-    giq_option(4, 28, 831, Node999, 50);
-    giq_option(4, 28, 832, Node065, 50);
+    gsay_reply(SCRIPT_ARBRIDGE, message_str(SCRIPT_ARBRIDGE, 830));
+    giq_option(4, SCRIPT_ARBRIDGE, 831, Node999, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_ARBRIDGE, 832, Node065, NEUTRAL_REACTION);
 end
 
 procedure Node065
 begin
-    gsay_reply(28, message_str(28, 840));
-    if ((global_var(830) bwand 4) != 0) then begin
-        giq_option(7, 28, 841, Node072, 50);
+    gsay_reply(SCRIPT_ARBRIDGE, message_str(SCRIPT_ARBRIDGE, 840));
+    if ((global_var(GVAR_TOWN_REP_COLLY) bwand 4) != 0) then begin
+        giq_option(7, SCRIPT_ARBRIDGE, 841, Node072, NEUTRAL_REACTION);
     end
-    if ((global_var(830) bwand 4) != 0) then begin
-        giq_option(4, 28, 842, Node072, 50);
+    if ((global_var(GVAR_TOWN_REP_COLLY) bwand 4) != 0) then begin
+        giq_option(4, SCRIPT_ARBRIDGE, 842, Node072, NEUTRAL_REACTION);
     end
     else begin
-        giq_option(4, 28, 842, Node075, 50);
+        giq_option(4, SCRIPT_ARBRIDGE, 842, Node075, NEUTRAL_REACTION);
     end
-    giq_option(4, 28, 843, Node998, 50);
+    giq_option(4, SCRIPT_ARBRIDGE, 843, Node998, NEUTRAL_REACTION);
 end
 
 procedure Node066
 begin
     set_local_var(8, local_var(8) bwand (-1 - 131072));
-    gsay_reply(28, message_str(28, 850));
-    giq_option(4, 28, 851, Node067, 50);
-    giq_option(4, 28, 852, Node999, 50);
+    gsay_reply(SCRIPT_ARBRIDGE, message_str(SCRIPT_ARBRIDGE, 850));
+    giq_option(4, SCRIPT_ARBRIDGE, 851, Node067, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_ARBRIDGE, 852, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node067
 begin
-    gsay_reply(28, message_str(28, 860));
-    giq_option(4, 28, 861, Node999, 50);
+    gsay_reply(SCRIPT_ARBRIDGE, message_str(SCRIPT_ARBRIDGE, 860));
+    giq_option(4, SCRIPT_ARBRIDGE, 861, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node068
 begin
     set_local_var(9, 2);
-    gsay_reply(28, message_str(28, 870));
-    giq_option(4, 28, 871, Node069, 50);
-    giq_option(4, 28, 872, Node999, 50);
-    giq_option(4, 28, 873, Node999, 50);
+    gsay_reply(SCRIPT_ARBRIDGE, message_str(SCRIPT_ARBRIDGE, 870));
+    giq_option(4, SCRIPT_ARBRIDGE, 871, Node069, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_ARBRIDGE, 872, Node999, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_ARBRIDGE, 873, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node069
 begin
-    gsay_reply(28, message_str(28, 880));
-    giq_option(4, 28, 881, Node070, 50);
-    giq_option(4, 28, 882, Node999, 50);
+    gsay_reply(SCRIPT_ARBRIDGE, message_str(SCRIPT_ARBRIDGE, 880));
+    giq_option(4, SCRIPT_ARBRIDGE, 881, Node070, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_ARBRIDGE, 882, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node070
 begin
-    gsay_reply(28, message_str(28, 890));
-    giq_option(4, 28, 891, Node999, 50);
-    giq_option(4, 28, 892, Node999, 50);
+    gsay_reply(SCRIPT_ARBRIDGE, message_str(SCRIPT_ARBRIDGE, 890));
+    giq_option(4, SCRIPT_ARBRIDGE, 891, Node999, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_ARBRIDGE, 892, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node071
 begin
-    gsay_reply(28, message_str(28, 900));
-    giq_option(4, 28, 901, set_holy_anger, 50);
-    giq_option(4, 28, 902, Node065, 50);
+    gsay_reply(SCRIPT_ARBRIDGE, message_str(SCRIPT_ARBRIDGE, 900));
+    giq_option(4, SCRIPT_ARBRIDGE, 901, set_holy_anger, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_ARBRIDGE, 902, Node065, NEUTRAL_REACTION);
 end
 
 procedure Node072
 begin
     set_local_var(5, 1);
     set_local_var(8, local_var(8) bwor 16384);
-    gsay_reply(28, message_str(28, 910));
-    if (has_skill(dude_obj, 14) >= 56) then begin
-        giq_option(6, 28, 911, Node076, 50);
+    gsay_reply(SCRIPT_ARBRIDGE, message_str(SCRIPT_ARBRIDGE, 910));
+    if (has_skill(dude_obj, SKILL_CONVERSANT) >= 56) then begin
+        giq_option(6, SCRIPT_ARBRIDGE, 911, Node076, NEUTRAL_REACTION);
     end
-    giq_option(5, 28, 912, Node077, 50);
-    giq_option(4, 28, 913, Node073, 50);
+    giq_option(5, SCRIPT_ARBRIDGE, 912, Node077, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_ARBRIDGE, 913, Node073, NEUTRAL_REACTION);
 end
 
 procedure Node073
 begin
-    gsay_reply(28, message_str(28, 920));
-    giq_option(4, 28, 921, Node074, 50);
+    gsay_reply(SCRIPT_ARBRIDGE, message_str(SCRIPT_ARBRIDGE, 920));
+    giq_option(4, SCRIPT_ARBRIDGE, 921, Node074, NEUTRAL_REACTION);
 end
 
 procedure Node074
 begin
-    gsay_reply(28, message_str(28, 930));
-    giq_option(4, 28, 931, Node998, 50);
+    gsay_reply(SCRIPT_ARBRIDGE, message_str(SCRIPT_ARBRIDGE, 930));
+    giq_option(4, SCRIPT_ARBRIDGE, 931, Node998, NEUTRAL_REACTION);
 end
 
 procedure Node075
 begin
-    gsay_reply(28, message_str(28, 940));
-    giq_option(4, 28, 941, Node998, 50);
+    gsay_reply(SCRIPT_ARBRIDGE, message_str(SCRIPT_ARBRIDGE, 940));
+    giq_option(4, SCRIPT_ARBRIDGE, 941, Node998, NEUTRAL_REACTION);
 end
 
 procedure Node076
 begin
-    gsay_reply(28, message_str(28, 950));
-    giq_option(4, 28, 951, Node082, 50);
+    gsay_reply(SCRIPT_ARBRIDGE, message_str(SCRIPT_ARBRIDGE, 950));
+    giq_option(4, SCRIPT_ARBRIDGE, 951, Node082, NEUTRAL_REACTION);
 end
 
 procedure Node077
 begin
-    gsay_reply(28, message_str(28, 960));
-    giq_option(4, 28, 961, Node078, 50);
+    gsay_reply(SCRIPT_ARBRIDGE, message_str(SCRIPT_ARBRIDGE, 960));
+    giq_option(4, SCRIPT_ARBRIDGE, 961, Node078, NEUTRAL_REACTION);
 end
 
 procedure Node078
 begin
-    gsay_reply(28, message_str(28, 970));
-    if (has_skill(dude_obj, 14) >= 36) then begin
-        giq_option(4, 28, 971, Node079, 50);
+    gsay_reply(SCRIPT_ARBRIDGE, message_str(SCRIPT_ARBRIDGE, 970));
+    if (has_skill(dude_obj, SKILL_CONVERSANT) >= 36) then begin
+        giq_option(4, SCRIPT_ARBRIDGE, 971, Node079, NEUTRAL_REACTION);
     end
-    giq_option(4, 28, 972, Node080, 50);
-    giq_option(4, 28, 973, Node081, 50);
+    giq_option(4, SCRIPT_ARBRIDGE, 972, Node080, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_ARBRIDGE, 973, Node081, NEUTRAL_REACTION);
 end
 
 procedure Node079
 begin
-    gsay_reply(28, message_str(28, 980));
-    giq_option(4, 28, 921, Node082, 50);
+    gsay_reply(SCRIPT_ARBRIDGE, message_str(SCRIPT_ARBRIDGE, 980));
+    giq_option(4, SCRIPT_ARBRIDGE, 921, Node082, NEUTRAL_REACTION);
 end
 
 procedure Node080
 begin
-    gsay_reply(28, message_str(28, 990));
-    giq_option(4, 28, 921, Node081, 50);
+    gsay_reply(SCRIPT_ARBRIDGE, message_str(SCRIPT_ARBRIDGE, 990));
+    giq_option(4, SCRIPT_ARBRIDGE, 921, Node081, NEUTRAL_REACTION);
 end
 
 procedure Node081
 begin
-    gsay_reply(28, message_str(28, 1000));
-    giq_option(4, 28, 1001, Node998, 50);
+    gsay_reply(SCRIPT_ARBRIDGE, message_str(SCRIPT_ARBRIDGE, 1000));
+    giq_option(4, SCRIPT_ARBRIDGE, 1001, Node998, NEUTRAL_REACTION);
 end
 
 procedure Node082
 begin
-    display_msg(message_str(28, 3003));
+    display_msg(message_str(SCRIPT_ARBRIDGE, 3003));
     give_exp_points(200);
-    gsay_reply(28, message_str(28, 1010));
-    giq_option(4, 28, 1011, Node999, 50);
+    gsay_reply(SCRIPT_ARBRIDGE, message_str(SCRIPT_ARBRIDGE, 1010));
+    giq_option(4, SCRIPT_ARBRIDGE, 1011, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node083
 begin
     set_local_var(10, 2);
-    gsay_reply(28, message_str(28, 1020));
-    if (has_skill(dude_obj, 14) >= 42) then begin
-        giq_option(5, 28, 1021, Node090, 50);
+    gsay_reply(SCRIPT_ARBRIDGE, message_str(SCRIPT_ARBRIDGE, 1020));
+    if (has_skill(dude_obj, SKILL_CONVERSANT) >= 42) then begin
+        giq_option(5, SCRIPT_ARBRIDGE, 1021, Node090, NEUTRAL_REACTION);
     end
     if (obj_is_carrying_obj_pid(dude_obj, 314) + obj_is_carrying_obj_pid(dude_obj, 315) + obj_is_carrying_obj_pid(dude_obj, 316)) then begin
-        giq_option(4, 28, 1022, Node008, 50);
+        giq_option(4, SCRIPT_ARBRIDGE, 1022, Node008, NEUTRAL_REACTION);
     end
-    giq_option(4, 28, 1023, Node084, 50);
-    giq_option(4, 28, 1024, Node017, 50);
+    giq_option(4, SCRIPT_ARBRIDGE, 1023, Node084, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_ARBRIDGE, 1024, Node017, NEUTRAL_REACTION);
 end
 
 procedure Node084
 begin
-    gsay_reply(28, message_str(28, 1030));
+    gsay_reply(SCRIPT_ARBRIDGE, message_str(SCRIPT_ARBRIDGE, 1030));
     if ((obj_is_carrying_obj_pid(dude_obj, 106) + obj_is_carrying_obj_pid(dude_obj, 124) + obj_is_carrying_obj_pid(dude_obj, 125) + obj_is_carrying_obj_pid(dude_obj, 310) + obj_is_carrying_obj_pid(dude_obj, 311) + obj_is_carrying_obj_pid(dude_obj, 469)) > 2) then begin
-        giq_option(4, 28, 1031, Node086, 50);
+        giq_option(4, SCRIPT_ARBRIDGE, 1031, Node086, NEUTRAL_REACTION);
     end
-    giq_option(4, 28, 1032, Node085, 50);
-    giq_option(4, 28, 1033, Node999, 50);
+    giq_option(4, SCRIPT_ARBRIDGE, 1032, Node085, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_ARBRIDGE, 1033, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node085
 begin
-    gsay_reply(28, message_str(28, 1040));
-    giq_option(4, 28, 1041, Node999, 50);
-    giq_option(4, 28, 1042, Node999, 50);
+    gsay_reply(SCRIPT_ARBRIDGE, message_str(SCRIPT_ARBRIDGE, 1040));
+    giq_option(4, SCRIPT_ARBRIDGE, 1041, Node999, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_ARBRIDGE, 1042, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node086
@@ -1384,228 +1384,228 @@ begin
             amt_ower := amt_ower - 1;
         end
     end
-    gsay_reply(28, message_str(28, 1050));
-    giq_option(4, 28, 1051, Node087, 50);
-    giq_option(4, 28, 1052, Node088, 50);
-    giq_option(4, 28, 1053, Node999, 50);
+    gsay_reply(SCRIPT_ARBRIDGE, message_str(SCRIPT_ARBRIDGE, 1050));
+    giq_option(4, SCRIPT_ARBRIDGE, 1051, Node087, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_ARBRIDGE, 1052, Node088, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_ARBRIDGE, 1053, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node087
 begin
     set_local_var(8, local_var(8) bwor 131072);
-    gsay_reply(28, message_str(28, 1060));
-    giq_option(4, 28, 1061, Node094, 50);
-    giq_option(4, 28, 1062, Node999, 50);
+    gsay_reply(SCRIPT_ARBRIDGE, message_str(SCRIPT_ARBRIDGE, 1060));
+    giq_option(4, SCRIPT_ARBRIDGE, 1061, Node094, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_ARBRIDGE, 1062, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node088
 begin
-    gsay_reply(28, message_str(28, 1070));
-    giq_option(4, 28, 1071, Node089, 50);
-    giq_option(4, 28, 1072, Node999, 50);
+    gsay_reply(SCRIPT_ARBRIDGE, message_str(SCRIPT_ARBRIDGE, 1070));
+    giq_option(4, SCRIPT_ARBRIDGE, 1071, Node089, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_ARBRIDGE, 1072, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node089
 begin
-    gsay_reply(28, message_str(28, 1080));
-    giq_option(4, 28, 1081, set_holy_anger, 50);
+    gsay_reply(SCRIPT_ARBRIDGE, message_str(SCRIPT_ARBRIDGE, 1080));
+    giq_option(4, SCRIPT_ARBRIDGE, 1081, set_holy_anger, NEUTRAL_REACTION);
 end
 
 procedure Node090
 begin
-    gsay_reply(28, message_str(28, 1090));
-    giq_option(4, 28, 1091, Node092, 50);
-    giq_option(4, 28, 1092, Node091, 50);
+    gsay_reply(SCRIPT_ARBRIDGE, message_str(SCRIPT_ARBRIDGE, 1090));
+    giq_option(4, SCRIPT_ARBRIDGE, 1091, Node092, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_ARBRIDGE, 1092, Node091, NEUTRAL_REACTION);
 end
 
 procedure Node091
 begin
-    gsay_reply(28, message_str(28, 1100));
-    giq_option(4, 28, 1101, Node999, 50);
+    gsay_reply(SCRIPT_ARBRIDGE, message_str(SCRIPT_ARBRIDGE, 1100));
+    giq_option(4, SCRIPT_ARBRIDGE, 1101, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node092
 begin
-    gsay_reply(28, message_str(28, 1110));
-    giq_option(4, 28, 1111, Node094, 50);
+    gsay_reply(SCRIPT_ARBRIDGE, message_str(SCRIPT_ARBRIDGE, 1110));
+    giq_option(4, SCRIPT_ARBRIDGE, 1111, Node094, NEUTRAL_REACTION);
 end
 
 procedure Node093
 begin
     set_local_var(10, 2);
-    gsay_reply(28, message_str(28, 1120 + ((global_var(830) bwand 2) != 0)));
-    giq_option(4, 28, 1122, Node094, 50);
+    gsay_reply(SCRIPT_ARBRIDGE, message_str(SCRIPT_ARBRIDGE, 1120 + ((global_var(GVAR_TOWN_REP_COLLY) bwand 2) != 0)));
+    giq_option(4, SCRIPT_ARBRIDGE, 1122, Node094, NEUTRAL_REACTION);
     if ((obj_is_carrying_obj_pid(dude_obj, 314) + obj_is_carrying_obj_pid(dude_obj, 315) + obj_is_carrying_obj_pid(dude_obj, 316)) > 0) then begin
-        giq_option(4, 28, 1123, Node008, 50);
+        giq_option(4, SCRIPT_ARBRIDGE, 1123, Node008, NEUTRAL_REACTION);
     end
 end
 
 procedure Node094
 begin
-    gsay_reply(28, message_str(28, 1130));
-    giq_option(4, 28, 1131, Node104, 50);
-    giq_option(4, 28, 1133, Node095, 50);
-    if ((get_critter_stat(dude_obj, 0) + get_critter_stat(dude_obj, 2)) > 12) then begin
-        giq_option(4, 28, 1134, Node110, 50);
+    gsay_reply(SCRIPT_ARBRIDGE, message_str(SCRIPT_ARBRIDGE, 1130));
+    giq_option(4, SCRIPT_ARBRIDGE, 1131, Node104, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_ARBRIDGE, 1133, Node095, NEUTRAL_REACTION);
+    if ((dude_strength + dude_endurance) > 12) then begin
+        giq_option(4, SCRIPT_ARBRIDGE, 1134, Node110, NEUTRAL_REACTION);
     end
     else begin
-        giq_option(4, 28, 1134, Node111, 50);
+        giq_option(4, SCRIPT_ARBRIDGE, 1134, Node111, NEUTRAL_REACTION);
     end
 end
 
 procedure Node095
 begin
-    gsay_reply(28, message_str(28, 1140));
-    giq_option(4, 28, 1141, Node096, 50);
-    giq_option(5, 28, 1142, Node099, 50);
-    giq_option(4, 28, 1143, Node999, 50);
+    gsay_reply(SCRIPT_ARBRIDGE, message_str(SCRIPT_ARBRIDGE, 1140));
+    giq_option(4, SCRIPT_ARBRIDGE, 1141, Node096, NEUTRAL_REACTION);
+    giq_option(5, SCRIPT_ARBRIDGE, 1142, Node099, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_ARBRIDGE, 1143, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node096
 begin
-    gsay_reply(28, message_str(28, 1150));
-    giq_option(4, 28, 1151, Node097, 50);
+    gsay_reply(SCRIPT_ARBRIDGE, message_str(SCRIPT_ARBRIDGE, 1150));
+    giq_option(4, SCRIPT_ARBRIDGE, 1151, Node097, NEUTRAL_REACTION);
 end
 
 procedure Node097
 begin
-    gsay_reply(28, message_str(28, 1160));
-    giq_option(4, 28, 1161, Node098, 50);
+    gsay_reply(SCRIPT_ARBRIDGE, message_str(SCRIPT_ARBRIDGE, 1160));
+    giq_option(4, SCRIPT_ARBRIDGE, 1161, Node098, NEUTRAL_REACTION);
 end
 
 procedure Node098
 begin
-    gsay_reply(28, message_str(28, 1170));
-    giq_option(4, 28, 1171, Node999, 50);
+    gsay_reply(SCRIPT_ARBRIDGE, message_str(SCRIPT_ARBRIDGE, 1170));
+    giq_option(4, SCRIPT_ARBRIDGE, 1171, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node099
 begin
-    gsay_reply(28, message_str(28, 1180));
-    giq_option(6, 28, 1181, Node100, 50);
-    giq_option(4, 28, 1182, Node103, 50);
+    gsay_reply(SCRIPT_ARBRIDGE, message_str(SCRIPT_ARBRIDGE, 1180));
+    giq_option(6, SCRIPT_ARBRIDGE, 1181, Node100, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_ARBRIDGE, 1182, Node103, NEUTRAL_REACTION);
 end
 
 procedure Node100
 begin
-    gsay_reply(28, message_str(28, 1190));
-    giq_option(4, 28, 1191, Node101, 50);
+    gsay_reply(SCRIPT_ARBRIDGE, message_str(SCRIPT_ARBRIDGE, 1190));
+    giq_option(4, SCRIPT_ARBRIDGE, 1191, Node101, NEUTRAL_REACTION);
 end
 
 procedure Node101
 begin
-    gsay_reply(28, message_str(28, 1200));
-    if (((local_var(8) bwand 131072) != 0) and (has_skill(dude_obj, 14) >= 40) or (has_skill(dude_obj, 14) >= 55) or (get_critter_stat(dude_obj, 3) > 7)) then begin
-        giq_option(4, 28, 1201, Node010, 50);
+    gsay_reply(SCRIPT_ARBRIDGE, message_str(SCRIPT_ARBRIDGE, 1200));
+    if (((local_var(8) bwand 131072) != 0) and (has_skill(dude_obj, SKILL_CONVERSANT) >= 40) or (has_skill(dude_obj, SKILL_CONVERSANT) >= 55) or (dude_charisma > 7)) then begin
+        giq_option(4, SCRIPT_ARBRIDGE, 1201, Node010, NEUTRAL_REACTION);
     end
     else begin
-        giq_option(4, 28, 1201, Node102, 50);
+        giq_option(4, SCRIPT_ARBRIDGE, 1201, Node102, NEUTRAL_REACTION);
     end
-    giq_option(4, 28, 1202, Node999, 50);
+    giq_option(4, SCRIPT_ARBRIDGE, 1202, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node102
 begin
-    gsay_reply(28, message_str(28, 1210));
-    giq_option(4, 28, 1211, Node999, 50);
+    gsay_reply(SCRIPT_ARBRIDGE, message_str(SCRIPT_ARBRIDGE, 1210));
+    giq_option(4, SCRIPT_ARBRIDGE, 1211, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node103
 begin
-    gsay_reply(28, message_str(28, 1220));
-    giq_option(4, 28, 1221, Node999, 50);
+    gsay_reply(SCRIPT_ARBRIDGE, message_str(SCRIPT_ARBRIDGE, 1220));
+    giq_option(4, SCRIPT_ARBRIDGE, 1221, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node104
 begin
-    gsay_reply(28, message_str(28, 1230));
-    if ((((global_var(830) bwand 2) != 0) + ((global_var(830) bwand 256) != 0) + get_critter_stat(dude_obj, 3)) > 7) then begin
-        giq_option(4, 28, 1231, Node010, 50);
+    gsay_reply(SCRIPT_ARBRIDGE, message_str(SCRIPT_ARBRIDGE, 1230));
+    if ((((global_var(GVAR_TOWN_REP_COLLY) bwand 2) != 0) + ((global_var(GVAR_TOWN_REP_COLLY) bwand 256) != 0) + dude_charisma) > 7) then begin
+        giq_option(4, SCRIPT_ARBRIDGE, 1231, Node010, NEUTRAL_REACTION);
     end
-    giq_option(4, 28, 1232, Node108, 50);
-    if ((global_var(830) bwand 256) != 0) then begin
-        giq_option(4, 28, 1233, Node106, 50);
+    giq_option(4, SCRIPT_ARBRIDGE, 1232, Node108, NEUTRAL_REACTION);
+    if ((global_var(GVAR_TOWN_REP_COLLY) bwand 256) != 0) then begin
+        giq_option(4, SCRIPT_ARBRIDGE, 1233, Node106, NEUTRAL_REACTION);
     end
     else begin
-        giq_option(4, 28, 1233, Node105, 50);
+        giq_option(4, SCRIPT_ARBRIDGE, 1233, Node105, NEUTRAL_REACTION);
     end
 end
 
 procedure Node105
 begin
-    gsay_reply(28, message_str(28, 1240));
-    giq_option(4, 28, 1241, set_holy_anger, 50);
+    gsay_reply(SCRIPT_ARBRIDGE, message_str(SCRIPT_ARBRIDGE, 1240));
+    giq_option(4, SCRIPT_ARBRIDGE, 1241, set_holy_anger, NEUTRAL_REACTION);
 end
 
 procedure Node106
 begin
-    gsay_reply(28, message_str(28, 1250));
-    giq_option(4, 28, 1251, Node107, 50);
+    gsay_reply(SCRIPT_ARBRIDGE, message_str(SCRIPT_ARBRIDGE, 1250));
+    giq_option(4, SCRIPT_ARBRIDGE, 1251, Node107, NEUTRAL_REACTION);
 end
 
 procedure Node107
 begin
-    gsay_reply(28, message_str(28, 1260));
-    giq_option(4, 28, 1261, Node011, 50);
+    gsay_reply(SCRIPT_ARBRIDGE, message_str(SCRIPT_ARBRIDGE, 1260));
+    giq_option(4, SCRIPT_ARBRIDGE, 1261, Node011, NEUTRAL_REACTION);
 end
 
 procedure Node108
 begin
-    gsay_reply(28, message_str(28, 1270));
-    giq_option(4, 28, 1271, Node109, 50);
-    giq_option(4, 28, 1272, Node999, 50);
+    gsay_reply(SCRIPT_ARBRIDGE, message_str(SCRIPT_ARBRIDGE, 1270));
+    giq_option(4, SCRIPT_ARBRIDGE, 1271, Node109, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_ARBRIDGE, 1272, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node109
 begin
-    gsay_reply(28, message_str(28, 1280));
-    giq_option(4, 28, 1281, Node999, 50);
+    gsay_reply(SCRIPT_ARBRIDGE, message_str(SCRIPT_ARBRIDGE, 1280));
+    giq_option(4, SCRIPT_ARBRIDGE, 1281, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node110
 begin
-    gsay_reply(28, message_str(28, 1290));
-    giq_option(4, 28, 1291, Node011, 50);
+    gsay_reply(SCRIPT_ARBRIDGE, message_str(SCRIPT_ARBRIDGE, 1290));
+    giq_option(4, SCRIPT_ARBRIDGE, 1291, Node011, NEUTRAL_REACTION);
 end
 
 procedure Node111
 begin
-    gsay_reply(28, message_str(28, 1300));
-    giq_option(4, 28, 1301, Node112, 50);
-    giq_option(4, 28, 1302, Node999, 50);
+    gsay_reply(SCRIPT_ARBRIDGE, message_str(SCRIPT_ARBRIDGE, 1300));
+    giq_option(4, SCRIPT_ARBRIDGE, 1301, Node112, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_ARBRIDGE, 1302, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node112
 begin
-    gsay_reply(28, message_str(28, 1310));
-    giq_option(4, 28, 1311, Node999, 50);
+    gsay_reply(SCRIPT_ARBRIDGE, message_str(SCRIPT_ARBRIDGE, 1310));
+    giq_option(4, SCRIPT_ARBRIDGE, 1311, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node113
 begin
-    gsay_reply(28, message_str(28, 1320));
-    giq_option(4, 28, 1321, Node200, 50);
-    giq_option(4, 28, 1322, Node114, 50);
-    giq_option(4, 28, 1323, Node999, 50);
+    gsay_reply(SCRIPT_ARBRIDGE, message_str(SCRIPT_ARBRIDGE, 1320));
+    giq_option(4, SCRIPT_ARBRIDGE, 1321, Node200, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_ARBRIDGE, 1322, Node114, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_ARBRIDGE, 1323, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node114
 begin
-    gsay_reply(28, message_str(28, random(1330, 1332)));
-    giq_option(4, 28, 1333, Node999, 50);
+    gsay_reply(SCRIPT_ARBRIDGE, message_str(SCRIPT_ARBRIDGE, random(1330, 1332)));
+    giq_option(4, SCRIPT_ARBRIDGE, 1333, Node999, NEUTRAL_REACTION);
 end
 
 procedure final
 begin
-    if (global_var(834) == 5) then begin
+    if (global_var(GVAR_BERK_BOOK) == 5) then begin
         call Node115();
     end
     else begin
-        if (global_var(834) == 1) then begin
+        if (global_var(GVAR_BERK_BOOK) == 1) then begin
             call Node116();
         end
         else begin
-            if ((global_var(834) == 2) or (global_var(834) == 3)) then begin
+            if ((global_var(GVAR_BERK_BOOK) == 2) or (global_var(GVAR_BERK_BOOK) == 3)) then begin
                 if (in_dialog == 0) then begin
                     in_dialog := 1;
                     start_gdialog(28, self_obj, 4, -1, -1);
@@ -1639,330 +1639,330 @@ end
 
 procedure Node115
 begin
-    float_msg(self_obj, message_str(28, random(1340, 1345)), 2);
+    float_msg(self_obj, message_str(SCRIPT_ARBRIDGE, random(1340, 1345)), FLOAT_MSG_RED);
 end
 
 procedure Node116
 begin
-    float_msg(self_obj, message_str(28, random(1350, 1355)), 2);
+    float_msg(self_obj, message_str(SCRIPT_ARBRIDGE, random(1350, 1355)), FLOAT_MSG_RED);
 end
 
 procedure Node117
 begin
-    gsay_reply(28, message_str(28, 1360));
-    giq_option(4, 28, 1361, Node118, 50);
-    giq_option(4, 28, 1362, Node119, 50);
-    giq_option(4, 28, 1363, Node120, 50);
-    giq_option(4, 28, 1364, Node999, 50);
+    gsay_reply(SCRIPT_ARBRIDGE, message_str(SCRIPT_ARBRIDGE, 1360));
+    giq_option(4, SCRIPT_ARBRIDGE, 1361, Node118, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_ARBRIDGE, 1362, Node119, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_ARBRIDGE, 1363, Node120, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_ARBRIDGE, 1364, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node118
 begin
-    gsay_reply(28, message_str(28, 1370));
-    giq_option(4, 28, 1373, Node124, 50);
+    gsay_reply(SCRIPT_ARBRIDGE, message_str(SCRIPT_ARBRIDGE, 1370));
+    giq_option(4, SCRIPT_ARBRIDGE, 1373, Node124, NEUTRAL_REACTION);
 end
 
 procedure Node119
 begin
-    gsay_reply(28, message_str(28, 1371));
-    giq_option(4, 28, 1373, Node124, 50);
+    gsay_reply(SCRIPT_ARBRIDGE, message_str(SCRIPT_ARBRIDGE, 1371));
+    giq_option(4, SCRIPT_ARBRIDGE, 1373, Node124, NEUTRAL_REACTION);
 end
 
 procedure Node120
 begin
-    gsay_reply(28, message_str(28, 1372));
-    giq_option(4, 28, 1374, Node124, 50);
+    gsay_reply(SCRIPT_ARBRIDGE, message_str(SCRIPT_ARBRIDGE, 1372));
+    giq_option(4, SCRIPT_ARBRIDGE, 1374, Node124, NEUTRAL_REACTION);
 end
 
 procedure Node121
 begin
-    gsay_reply(28, message_str(28, 1380));
-    giq_option(4, 28, 1381, Node122, 50);
+    gsay_reply(SCRIPT_ARBRIDGE, message_str(SCRIPT_ARBRIDGE, 1380));
+    giq_option(4, SCRIPT_ARBRIDGE, 1381, Node122, NEUTRAL_REACTION);
 end
 
 procedure Node122
 begin
-    gsay_reply(28, message_str(28, 1390));
-    giq_option(4, 28, 1391, Node123, 50);
+    gsay_reply(SCRIPT_ARBRIDGE, message_str(SCRIPT_ARBRIDGE, 1390));
+    giq_option(4, SCRIPT_ARBRIDGE, 1391, Node123, NEUTRAL_REACTION);
 end
 
 procedure Node123
 begin
-    gsay_reply(28, message_str(28, 1400));
-    giq_option(4, 28, 1401, Node124, 50);
-    giq_option(4, 28, 1402, Node125, 50);
+    gsay_reply(SCRIPT_ARBRIDGE, message_str(SCRIPT_ARBRIDGE, 1400));
+    giq_option(4, SCRIPT_ARBRIDGE, 1401, Node124, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_ARBRIDGE, 1402, Node125, NEUTRAL_REACTION);
 end
 
 procedure Node124
 begin
-    gsay_reply(28, message_str(28, random(1410, 1412)));
-    giq_option(4, 28, 1413, Node999, 50);
-    giq_option(4, 28, 1414, Node999, 50);
+    gsay_reply(SCRIPT_ARBRIDGE, message_str(SCRIPT_ARBRIDGE, random(1410, 1412)));
+    giq_option(4, SCRIPT_ARBRIDGE, 1413, Node999, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_ARBRIDGE, 1414, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node125
 begin
-    gsay_message(28, message_str(28, 1420), 50);
+    gsay_message(SCRIPT_ARBRIDGE, message_str(SCRIPT_ARBRIDGE, 1420), NEUTRAL_REACTION);
 end
 
 procedure Node126
 begin
-    float_msg(self_obj, message_str(28, random(150, 152)), 2);
+    float_msg(self_obj, message_str(SCRIPT_ARBRIDGE, random(150, 152)), FLOAT_MSG_RED);
 end
 
 procedure Node200
 begin
-    gsay_reply(28, message_str(28, random(2000, 2001)));
-    if ((local_var(9) < 2) and (((local_var(8) bwand 4096) != 0) or ((global_var(830) bwand 2) != 0) or ((global_var(830) bwand 256) != 0))) then begin
-        giq_option(4, 28, 2003, Node203, 50);
-        giq_option(4, 28, 2004, Node208, 50);
-        giq_option(4, 28, 2005, Node211, 50);
+    gsay_reply(SCRIPT_ARBRIDGE, message_str(SCRIPT_ARBRIDGE, random(2000, 2001)));
+    if ((local_var(9) < 2) and (((local_var(8) bwand 4096) != 0) or ((global_var(GVAR_TOWN_REP_COLLY) bwand 2) != 0) or ((global_var(GVAR_TOWN_REP_COLLY) bwand 256) != 0))) then begin
+        giq_option(4, SCRIPT_ARBRIDGE, 2003, Node203, NEUTRAL_REACTION);
+        giq_option(4, SCRIPT_ARBRIDGE, 2004, Node208, NEUTRAL_REACTION);
+        giq_option(4, SCRIPT_ARBRIDGE, 2005, Node211, NEUTRAL_REACTION);
     end
     else begin
-        giq_option(4, 28, 2003, Node202, 50);
-        giq_option(4, 28, 2004, Node207, 50);
-        giq_option(4, 28, 2005, Node210, 50);
+        giq_option(4, SCRIPT_ARBRIDGE, 2003, Node202, NEUTRAL_REACTION);
+        giq_option(4, SCRIPT_ARBRIDGE, 2004, Node207, NEUTRAL_REACTION);
+        giq_option(4, SCRIPT_ARBRIDGE, 2005, Node210, NEUTRAL_REACTION);
     end
     if ((local_var(8) bwand 4096) != 0) then begin
-        if ((local_var(9) < 2) and (has_skill(dude_obj, 14) >= 53)) then begin
-            giq_option(4, 28, 2006, Node214, 50);
+        if ((local_var(9) < 2) and (has_skill(dude_obj, SKILL_CONVERSANT) >= 53)) then begin
+            giq_option(4, SCRIPT_ARBRIDGE, 2006, Node214, NEUTRAL_REACTION);
         end
         else begin
-            giq_option(4, 28, 2006, Node201, 50);
+            giq_option(4, SCRIPT_ARBRIDGE, 2006, Node201, NEUTRAL_REACTION);
         end
         if ((local_var(8) bwand 262144) != 0) then begin
-            if ((local_var(9) < 2) and (has_skill(dude_obj, 14) >= 78)) then begin
-                giq_option(4, 28, 2007, Node219, 50);
+            if ((local_var(9) < 2) and (has_skill(dude_obj, SKILL_CONVERSANT) >= 78)) then begin
+                giq_option(4, SCRIPT_ARBRIDGE, 2007, Node219, NEUTRAL_REACTION);
             end
             else begin
-                giq_option(4, 28, 2007, Node201, 50);
+                giq_option(4, SCRIPT_ARBRIDGE, 2007, Node201, NEUTRAL_REACTION);
             end
         end
-        if ((global_var(831) bwand 128) != 0) then begin
+        if ((global_var(GVAR_RED_ACCEPT_TO_GANG) bwand 128) != 0) then begin
             if (local_var(9) < 2) then begin
-                giq_option(4, 28, 2008, Node221, 50);
-                giq_option(4, 28, 2009, Node222, 50);
+                giq_option(4, SCRIPT_ARBRIDGE, 2008, Node221, NEUTRAL_REACTION);
+                giq_option(4, SCRIPT_ARBRIDGE, 2009, Node222, NEUTRAL_REACTION);
             end
             else begin
-                giq_option(4, 28, 2008, Node201, 50);
-                giq_option(4, 28, 2009, Node201, 50);
+                giq_option(4, SCRIPT_ARBRIDGE, 2008, Node201, NEUTRAL_REACTION);
+                giq_option(4, SCRIPT_ARBRIDGE, 2009, Node201, NEUTRAL_REACTION);
             end
         end
     end
-    giq_option(4, 28, 2011, Node999, 50);
+    giq_option(4, SCRIPT_ARBRIDGE, 2011, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node201
 begin
-    gsay_reply(28, message_str(28, random(2020, 2021)));
-    giq_option(4, 28, 2022, Node200, 50);
-    giq_option(4, 28, 2023, Node999, 50);
+    gsay_reply(SCRIPT_ARBRIDGE, message_str(SCRIPT_ARBRIDGE, random(2020, 2021)));
+    giq_option(4, SCRIPT_ARBRIDGE, 2022, Node200, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_ARBRIDGE, 2023, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node202
 begin
-    gsay_reply(28, message_str(28, 2030));
-    giq_option(4, 28, 2031, Node200, 50);
-    giq_option(4, 28, 2032, Node999, 50);
+    gsay_reply(SCRIPT_ARBRIDGE, message_str(SCRIPT_ARBRIDGE, 2030));
+    giq_option(4, SCRIPT_ARBRIDGE, 2031, Node200, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_ARBRIDGE, 2032, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node203
 begin
-    gsay_reply(28, message_str(28, 2040));
-    giq_option(4, 28, 2041, Node204, 50);
-    giq_option(4, 28, 2042, Node205, 50);
-    giq_option(4, 28, 2043, Node206, 50);
-    giq_option(4, 28, 2044, Node200, 50);
-    giq_option(4, 28, 2045, Node999, 50);
+    gsay_reply(SCRIPT_ARBRIDGE, message_str(SCRIPT_ARBRIDGE, 2040));
+    giq_option(4, SCRIPT_ARBRIDGE, 2041, Node204, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_ARBRIDGE, 2042, Node205, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_ARBRIDGE, 2043, Node206, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_ARBRIDGE, 2044, Node200, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_ARBRIDGE, 2045, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node204
 begin
-    if (global_var(837) == 3) then begin
-        gsay_reply(28, message_str(28, 2050) + message_str(28, 2051));
+    if (global_var(GVAR_COLLY_QUEST_2) == 3) then begin
+        gsay_reply(SCRIPT_ARBRIDGE, message_str(SCRIPT_ARBRIDGE, 2050) + message_str(SCRIPT_ARBRIDGE, 2051));
     end
     else begin
-        gsay_reply(28, message_str(28, 2051));
+        gsay_reply(SCRIPT_ARBRIDGE, message_str(SCRIPT_ARBRIDGE, 2051));
     end
-    giq_option(4, 28, 2042, Node205, 50);
-    giq_option(4, 28, 2043, Node206, 50);
-    giq_option(4, 28, 2044, Node200, 50);
-    giq_option(4, 28, 2045, Node999, 50);
+    giq_option(4, SCRIPT_ARBRIDGE, 2042, Node205, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_ARBRIDGE, 2043, Node206, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_ARBRIDGE, 2044, Node200, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_ARBRIDGE, 2045, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node205
 begin
-    gsay_reply(28, message_str(28, 2060));
-    giq_option(4, 28, 2041, Node204, 50);
-    giq_option(4, 28, 2043, Node206, 50);
-    giq_option(4, 28, 2061, Node200, 50);
-    giq_option(4, 28, 2045, Node999, 50);
+    gsay_reply(SCRIPT_ARBRIDGE, message_str(SCRIPT_ARBRIDGE, 2060));
+    giq_option(4, SCRIPT_ARBRIDGE, 2041, Node204, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_ARBRIDGE, 2043, Node206, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_ARBRIDGE, 2061, Node200, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_ARBRIDGE, 2045, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node206
 begin
-    gsay_reply(28, message_str(28, 2070));
-    giq_option(4, 28, 2041, Node204, 50);
-    giq_option(4, 28, 2042, Node205, 50);
-    giq_option(4, 28, 2044, Node200, 50);
-    giq_option(4, 28, 2045, Node999, 50);
+    gsay_reply(SCRIPT_ARBRIDGE, message_str(SCRIPT_ARBRIDGE, 2070));
+    giq_option(4, SCRIPT_ARBRIDGE, 2041, Node204, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_ARBRIDGE, 2042, Node205, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_ARBRIDGE, 2044, Node200, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_ARBRIDGE, 2045, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node207
 begin
-    gsay_reply(28, message_str(28, 2080));
-    giq_option(4, 28, 2081, Node200, 50);
-    giq_option(4, 28, 2082, Node999, 50);
+    gsay_reply(SCRIPT_ARBRIDGE, message_str(SCRIPT_ARBRIDGE, 2080));
+    giq_option(4, SCRIPT_ARBRIDGE, 2081, Node200, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_ARBRIDGE, 2082, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node208
 begin
-    gsay_reply(28, message_str(28, 2090));
-    giq_option(4, 28, 2091, Node209, 50);
-    giq_option(4, 28, 2044, Node200, 50);
-    giq_option(4, 28, 2045, Node999, 50);
+    gsay_reply(SCRIPT_ARBRIDGE, message_str(SCRIPT_ARBRIDGE, 2090));
+    giq_option(4, SCRIPT_ARBRIDGE, 2091, Node209, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_ARBRIDGE, 2044, Node200, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_ARBRIDGE, 2045, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node209
 begin
-    gsay_reply(28, message_str(28, 2100));
-    giq_option(4, 28, 2044, Node200, 50);
-    giq_option(4, 28, 2045, Node999, 50);
+    gsay_reply(SCRIPT_ARBRIDGE, message_str(SCRIPT_ARBRIDGE, 2100));
+    giq_option(4, SCRIPT_ARBRIDGE, 2044, Node200, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_ARBRIDGE, 2045, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node210
 begin
-    gsay_reply(28, message_str(28, 2110));
-    giq_option(4, 28, 2111, Node200, 50);
-    giq_option(4, 28, 2045, Node999, 50);
+    gsay_reply(SCRIPT_ARBRIDGE, message_str(SCRIPT_ARBRIDGE, 2110));
+    giq_option(4, SCRIPT_ARBRIDGE, 2111, Node200, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_ARBRIDGE, 2045, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node211
 begin
-    gsay_reply(28, message_str(28, 2120));
-    giq_option(4, 28, 2121, Node212, 50);
-    giq_option(4, 28, 2122, Node200, 50);
-    giq_option(4, 28, 2123, Node999, 50);
+    gsay_reply(SCRIPT_ARBRIDGE, message_str(SCRIPT_ARBRIDGE, 2120));
+    giq_option(4, SCRIPT_ARBRIDGE, 2121, Node212, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_ARBRIDGE, 2122, Node200, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_ARBRIDGE, 2123, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node212
 begin
-    gsay_reply(28, message_str(28, 2130));
-    giq_option(4, 28, 2131, Node213, 50);
-    giq_option(4, 28, 2044, Node200, 50);
-    giq_option(4, 28, 2045, Node999, 50);
+    gsay_reply(SCRIPT_ARBRIDGE, message_str(SCRIPT_ARBRIDGE, 2130));
+    giq_option(4, SCRIPT_ARBRIDGE, 2131, Node213, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_ARBRIDGE, 2044, Node200, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_ARBRIDGE, 2045, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node213
 begin
-    gsay_reply(28, message_str(28, 2140));
-    giq_option(4, 28, 2141, Node200, 50);
-    giq_option(4, 28, 2142, Node999, 50);
+    gsay_reply(SCRIPT_ARBRIDGE, message_str(SCRIPT_ARBRIDGE, 2140));
+    giq_option(4, SCRIPT_ARBRIDGE, 2141, Node200, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_ARBRIDGE, 2142, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node214
 begin
-    if (((global_var(831) bwand 64) != 0) == 0) then begin
-        set_global_var(831, global_var(831) bwor 64);
-        display_msg(message_str(28, 3004));
+    if (((global_var(GVAR_RED_ACCEPT_TO_GANG) bwand 64) != 0) == 0) then begin
+        set_global_var(GVAR_RED_ACCEPT_TO_GANG, global_var(GVAR_RED_ACCEPT_TO_GANG) bwor 64);
+        display_msg(message_str(SCRIPT_ARBRIDGE, 3004));
         give_exp_points(100);
     end
-    gsay_reply(28, message_str(28, 2150));
-    giq_option(4, 28, 2151, Node215, 50);
-    giq_option(4, 28, 2152, Node200, 50);
-    giq_option(4, 28, 2153, Node999, 50);
+    gsay_reply(SCRIPT_ARBRIDGE, message_str(SCRIPT_ARBRIDGE, 2150));
+    giq_option(4, SCRIPT_ARBRIDGE, 2151, Node215, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_ARBRIDGE, 2152, Node200, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_ARBRIDGE, 2153, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node215
 begin
-    gsay_reply(28, message_str(28, 2160));
-    giq_option(4, 28, 2161, Node216, 50);
+    gsay_reply(SCRIPT_ARBRIDGE, message_str(SCRIPT_ARBRIDGE, 2160));
+    giq_option(4, SCRIPT_ARBRIDGE, 2161, Node216, NEUTRAL_REACTION);
 end
 
 procedure Node216
 begin
-    gsay_reply(28, message_str(28, 2170));
-    if (has_skill(dude_obj, 14) >= 78) then begin
-        giq_option(4, 28, 2171, Node219, 50);
+    gsay_reply(SCRIPT_ARBRIDGE, message_str(SCRIPT_ARBRIDGE, 2170));
+    if (has_skill(dude_obj, SKILL_CONVERSANT) >= 78) then begin
+        giq_option(4, SCRIPT_ARBRIDGE, 2171, Node219, NEUTRAL_REACTION);
     end
     else begin
-        giq_option(4, 28, 2171, Node217, 50);
+        giq_option(4, SCRIPT_ARBRIDGE, 2171, Node217, NEUTRAL_REACTION);
     end
-    if ((has_skill(dude_obj, 14) >= 88) or (get_critter_stat(dude_obj, 6) > 8)) then begin
-        giq_option(6, 28, 2172, Node218, 50);
+    if ((has_skill(dude_obj, SKILL_CONVERSANT) >= 88) or (dude_luck > 8)) then begin
+        giq_option(6, SCRIPT_ARBRIDGE, 2172, Node218, NEUTRAL_REACTION);
     end
 end
 
 procedure Node217
 begin
     set_local_var(8, local_var(8) bwor 262144);
-    gsay_reply(28, message_str(28, 2180));
-    giq_option(4, 28, 2181, Node200, 50);
-    giq_option(4, 28, 2182, Node999, 50);
+    gsay_reply(SCRIPT_ARBRIDGE, message_str(SCRIPT_ARBRIDGE, 2180));
+    giq_option(4, SCRIPT_ARBRIDGE, 2181, Node200, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_ARBRIDGE, 2182, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node218
 begin
-    gsay_reply(28, message_str(28, 2190));
-    giq_option(4, 28, 2191, Node220, 50);
+    gsay_reply(SCRIPT_ARBRIDGE, message_str(SCRIPT_ARBRIDGE, 2190));
+    giq_option(4, SCRIPT_ARBRIDGE, 2191, Node220, NEUTRAL_REACTION);
 end
 
 procedure Node219
 begin
-    gsay_reply(28, message_str(28, 2200));
-    giq_option(4, 28, 2201, Node220, 50);
+    gsay_reply(SCRIPT_ARBRIDGE, message_str(SCRIPT_ARBRIDGE, 2200));
+    giq_option(4, SCRIPT_ARBRIDGE, 2201, Node220, NEUTRAL_REACTION);
 end
 
 procedure Node220
 begin
-    display_msg(message_str(28, 3005));
+    display_msg(message_str(SCRIPT_ARBRIDGE, 3005));
     give_exp_points(200);
-    set_global_var(831, global_var(831) bwor 128);
+    set_global_var(GVAR_RED_ACCEPT_TO_GANG, global_var(GVAR_RED_ACCEPT_TO_GANG) bwor 128);
     set_local_var(8, local_var(8) bwand (-1 - 262144));
-    gsay_reply(28, message_str(28, 2210));
-    giq_option(4, 28, 2211, Node221, 50);
+    gsay_reply(SCRIPT_ARBRIDGE, message_str(SCRIPT_ARBRIDGE, 2210));
+    giq_option(4, SCRIPT_ARBRIDGE, 2211, Node221, NEUTRAL_REACTION);
 end
 
 procedure Node221
 begin
-    gsay_reply(28, message_str(28, 2220));
-    giq_option(4, 28, 2221, Node200, 50);
-    giq_option(4, 28, 2222, Node999, 50);
+    gsay_reply(SCRIPT_ARBRIDGE, message_str(SCRIPT_ARBRIDGE, 2220));
+    giq_option(4, SCRIPT_ARBRIDGE, 2221, Node200, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_ARBRIDGE, 2222, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node222
 begin
-    gsay_reply(28, message_str(28, 2230));
-    giq_option(4, 28, 2231, Node223, 50);
-    giq_option(4, 28, 2232, Node224, 50);
-    giq_option(4, 28, 2233, Node200, 50);
-    giq_option(4, 28, 2234, Node999, 50);
+    gsay_reply(SCRIPT_ARBRIDGE, message_str(SCRIPT_ARBRIDGE, 2230));
+    giq_option(4, SCRIPT_ARBRIDGE, 2231, Node223, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_ARBRIDGE, 2232, Node224, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_ARBRIDGE, 2233, Node200, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_ARBRIDGE, 2234, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node223
 begin
-    gsay_reply(28, message_str(28, 2240));
-    giq_option(4, 28, 2241, Node200, 50);
-    giq_option(4, 28, 2242, Node999, 50);
+    gsay_reply(SCRIPT_ARBRIDGE, message_str(SCRIPT_ARBRIDGE, 2240));
+    giq_option(4, SCRIPT_ARBRIDGE, 2241, Node200, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_ARBRIDGE, 2242, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node224
 begin
-    gsay_reply(28, message_str(28, 2250));
-    giq_option(4, 28, 2251, Node200, 50);
-    giq_option(4, 28, 2252, Node999, 50);
+    gsay_reply(SCRIPT_ARBRIDGE, message_str(SCRIPT_ARBRIDGE, 2250));
+    giq_option(4, SCRIPT_ARBRIDGE, 2251, Node200, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_ARBRIDGE, 2252, Node999, NEUTRAL_REACTION);
 end
 
 procedure solo_fight
 begin
     set_local_var(8, local_var(8) bwor 32768);
-    critter_add_trait(self_obj, 1, 6, 16);
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 16);
     call Node998();
 end
 
 procedure revenge
 begin
-    set_global_var(830, global_var(830) bwor 1048576);
+    set_global_var(GVAR_TOWN_REP_COLLY, global_var(GVAR_TOWN_REP_COLLY) bwor 1048576);
     call Node997();
 end
 
@@ -1973,6 +1973,5 @@ end
 
 procedure holy_anger
 begin
-    float_msg(self_obj, message_str(28, 1989 + random(1, 5)), 2);
+    float_msg(self_obj, message_str(SCRIPT_ARBRIDGE, 1989 + random(1, 5)), FLOAT_MSG_RED);
 end
-
