diff --git a/Mods/MegaMod/Original/MBASE12.ssl b/Mods/MegaMod/ReDefine/MBASE12.ssl
index 2e360e9..e56667b 100644
--- a/Mods/MegaMod/Original/MBASE12.ssl
+++ b/Mods/MegaMod/ReDefine/MBASE12.ssl
@@ -137,10 +137,10 @@ end
 
 procedure map_enter_p_proc
 begin
-    if (metarule(14, 0)) then begin
-        display_msg(message_str(277, 100));
+    if (map_first_run) then begin
+        display_mstr(100);
     end
-    if ((global_var(215) bwand 1) != 0) then begin
+    if ((global_var(GVAR_MILITARY_BASE_FLAGS) bwand 1) != 0) then begin
         set_light_level(75);
     end
     else begin
@@ -150,11 +150,10 @@ end
 
 procedure map_update_p_proc
 begin
-    if ((global_var(215) bwand 1) != 0) then begin
+    if ((global_var(GVAR_MILITARY_BASE_FLAGS) bwand 1) != 0) then begin
         set_light_level(75);
     end
     else begin
         set_light_level(50);
     end
 end
-
