diff --git a/Mods/MegaMod/Original/MBASE34.ssl b/Mods/MegaMod/ReDefine/MBASE34.ssl
index cf77226..76c1415 100644
--- a/Mods/MegaMod/Original/MBASE34.ssl
+++ b/Mods/MegaMod/ReDefine/MBASE34.ssl
@@ -141,7 +141,7 @@ end
 
 procedure map_enter_p_proc
 begin
-    if ((global_var(215) bwand 1) != 0) then begin
+    if ((global_var(GVAR_MILITARY_BASE_FLAGS) bwand 1) != 0) then begin
         set_light_level(75);
     end
     else begin
@@ -151,11 +151,10 @@ end
 
 procedure map_update_p_proc
 begin
-    if ((global_var(215) bwand 1) != 0) then begin
+    if ((global_var(GVAR_MILITARY_BASE_FLAGS) bwand 1) != 0) then begin
         set_light_level(75);
     end
     else begin
         set_light_level(50);
     end
 end
-
