diff --git a/Mods/MegaMod/Original/MBCOMPTR.ssl b/Mods/MegaMod/ReDefine/MBCOMPTR.ssl
index c2bccf2..c691903 100644
--- a/Mods/MegaMod/Original/MBCOMPTR.ssl
+++ b/Mods/MegaMod/ReDefine/MBCOMPTR.ssl
@@ -239,13 +239,13 @@ end
 procedure look_at_p_proc
 begin
     script_overrides;
-    display_msg(message_str(1581, 101));
+    display_mstr(101);
 end
 
 procedure description_p_proc
 begin
     script_overrides;
-    display_msg(message_str(1581, 102));
+    display_mstr(102);
 end
 
 procedure use_skill_on_p_proc
@@ -277,7 +277,7 @@ end
 
 procedure talk_p_proc
 begin
-    if ((local_var(0) == 0) and ((global_var(762) == 1) or (global_var(762) == 2))) then begin
+    if ((local_var(0) == 0) and ((global_var(GVAR_KAGA_KILLED) == 1) or (global_var(GVAR_KAGA_KILLED) == 2))) then begin
         start_gdialog(1581, self_obj, 4, -1, -1);
         gsay_start;
         call Node001();
@@ -297,24 +297,24 @@ procedure Node001
 begin
     if (is_success(roll_vs_skill(dude_obj, 12, -4))) then begin
         set_local_var(0, 1);
-        if (metarule(17, 72) == 0) then begin
+        if (town_known(72) == MARK_STATE_UNKNOWN) then begin
             debug_msg("  mark_on_map(" + 72 + ")");
-            mark_area_known(0, 72, 1);
+            mark_area_known(MARK_TYPE_TOWN, 72, MARK_STATE_KNOWN);
         end
         give_exp_points(500);
-        display_msg(message_str(1581, 105));
-        gsay_reply(1581, 103);
+        display_mstr(105);
+        Reply(103);
     end
     else begin
-        gsay_reply(1581, 104);
+        Reply(104);
     end
-    giq_option(0, 1581, message_str(14, 350), Node999, 50);
+    giq_option(0, NAME, g_mstr(350), Node999, NEUTRAL_REACTION);
 end
 
 procedure Node002
 begin
-    gsay_reply(1581, 106);
-    giq_option(0, 1581, message_str(14, 350), Node999, 50);
+    Reply(106);
+    giq_option(0, NAME, g_mstr(350), Node999, NEUTRAL_REACTION);
 end
 
 procedure Node999
@@ -324,4 +324,3 @@ end
 procedure timed_event_p_proc
 begin
 end
-
