diff --git a/Mods/MegaMod/Original/MCATKDOG.ssl b/Mods/MegaMod/ReDefine/MCATKDOG.ssl
index 9b91759..4b09a42 100644
--- a/Mods/MegaMod/Original/MCATKDOG.ssl
+++ b/Mods/MegaMod/ReDefine/MCATKDOG.ssl
@@ -160,8 +160,8 @@ end
 procedure critter_p_proc
 begin
     variable LVar0 := 0;
-    if (cur_map_index == 20) then begin
-        if ((global_var(115) < 0) == 0) then begin
+    if (CUR_MAP_MODBRAH) then begin
+        if ((global_var(GVAR_JONNY_TILE) < 0) == 0) then begin
             if (((local_var(3) bwand 2) != 0) == 0) then begin
                 if (dog_object == 0) then begin
                     dog_object := self_obj;
@@ -170,7 +170,7 @@ begin
                     if (brahmin_object == 0) then begin
                     end
                     else begin
-                        if ((global_var(402) bwand 32768) != 0) then begin
+                        if ((global_var(GVAR_MODOC_GENERIC_FLAG_2) bwand 32768) != 0) then begin
                             if (tile_distance_objs(self_obj, brahmin_object) < tile_distance_objs(dog_object, brahmin_object)) then begin
                                 brahmin_object := self_obj;
                             end
@@ -190,7 +190,7 @@ begin
             end
             if (brahmin_object != 0) then begin
                 if (obj_can_see_obj(self_obj, brahmin_object)) then begin
-                    attack_complex(brahmin_object, 0, 1, 0, 0, 30000, 0, 0);
+                    attack(brahmin_object);
                 end
                 else begin
                     if ((tile_num(self_obj) != tile_num(brahmin_object)) and (tile_num(brahmin_object) != 0)) then begin
@@ -207,12 +207,12 @@ begin
                                     animate_move_obj_to_tile(self_obj, step_tile, 1);
                                 end
                                 if (anim_busy(self_obj) == 0) then begin
-                                    if (metarule3(106, tile_num_in_direction(tile_num(self_obj), has_trait(1, self_obj, 10), 1), elevation(self_obj), 0) != 0) then begin
+                                    if (tile_get_next_critter(tile_num_in_direction(tile_num(self_obj), obj_get_rot(self_obj), 1), elevation(self_obj), 0) != 0) then begin
                                         if (anim_busy(self_obj) == 0) then begin
-                                            global_temp := rotation_to_tile(tile_num_in_direction(tile_num(self_obj), rotation_to_tile(tile_num_in_direction(tile_num(self_obj), has_trait(1, self_obj, 10), 1), tile_num(self_obj)), 1), tile_num(self_obj));
+                                            global_temp := rotation_to_tile(tile_num_in_direction(tile_num(self_obj), rotation_to_tile(tile_num_in_direction(tile_num(self_obj), obj_get_rot(self_obj), 1), tile_num(self_obj)), 1), tile_num(self_obj));
                                             animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(self_obj), global_temp, random(2, 5)), 0);
                                             global_temp := (global_temp + 1) % 6;
-                                            while ((anim_busy(self_obj) == 0) and (global_temp != rotation_to_tile(tile_num_in_direction(tile_num(self_obj), rotation_to_tile(tile_num_in_direction(tile_num(self_obj), has_trait(1, self_obj, 10), 1), tile_num(self_obj)), 1), tile_num(self_obj)))) do begin
+                                            while ((anim_busy(self_obj) == 0) and (global_temp != rotation_to_tile(tile_num_in_direction(tile_num(self_obj), rotation_to_tile(tile_num_in_direction(tile_num(self_obj), obj_get_rot(self_obj), 1), tile_num(self_obj)), 1), tile_num(self_obj)))) do begin
                                                 debug_msg("flee loop: rot == " + global_temp);
                                                 animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(self_obj), global_temp, random(2, 5)), 0);
                                                 global_temp := (global_temp + 1) % 6;
@@ -240,10 +240,10 @@ begin
                     if (anim_busy(self_obj) == 0) then begin
                         if ((tile_distance_objs(self_obj, dude_obj) < 8) or obj_can_see_obj(self_obj, dude_obj)) then begin
                             if (anim_busy(self_obj) == 0) then begin
-                                global_temp := rotation_to_tile(tile_num(dude_obj), tile_num(self_obj));
+                                global_temp := rotation_to_tile(dude_tile, tile_num(self_obj));
                                 animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(self_obj), global_temp, random(3, 10)), 1);
                                 global_temp := (global_temp + 1) % 6;
-                                while ((anim_busy(self_obj) == 0) and (global_temp != rotation_to_tile(tile_num(dude_obj), tile_num(self_obj)))) do begin
+                                while ((anim_busy(self_obj) == 0) and (global_temp != rotation_to_tile(dude_tile, tile_num(self_obj)))) do begin
                                     debug_msg("flee loop: rot == " + global_temp);
                                     animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(self_obj), global_temp, random(3, 10)), 1);
                                     global_temp := (global_temp + 1) % 6;
@@ -253,7 +253,7 @@ begin
                     end
                 end
                 else begin
-                    attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+                    attack(dude_obj);
                 end
             end
             else begin
@@ -271,12 +271,12 @@ begin
                                 animate_move_obj_to_tile(self_obj, step_tile, 1);
                             end
                             if (anim_busy(self_obj) == 0) then begin
-                                if (metarule3(106, tile_num_in_direction(tile_num(self_obj), has_trait(1, self_obj, 10), 1), elevation(self_obj), 0) != 0) then begin
+                                if (tile_get_next_critter(tile_num_in_direction(tile_num(self_obj), obj_get_rot(self_obj), 1), elevation(self_obj), 0) != 0) then begin
                                     if (anim_busy(self_obj) == 0) then begin
-                                        global_temp := rotation_to_tile(tile_num_in_direction(tile_num(self_obj), rotation_to_tile(tile_num_in_direction(tile_num(self_obj), has_trait(1, self_obj, 10), 1), tile_num(self_obj)), 1), tile_num(self_obj));
+                                        global_temp := rotation_to_tile(tile_num_in_direction(tile_num(self_obj), rotation_to_tile(tile_num_in_direction(tile_num(self_obj), obj_get_rot(self_obj), 1), tile_num(self_obj)), 1), tile_num(self_obj));
                                         animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(self_obj), global_temp, random(2, 5)), 0);
                                         global_temp := (global_temp + 1) % 6;
-                                        while ((anim_busy(self_obj) == 0) and (global_temp != rotation_to_tile(tile_num_in_direction(tile_num(self_obj), rotation_to_tile(tile_num_in_direction(tile_num(self_obj), has_trait(1, self_obj, 10), 1), tile_num(self_obj)), 1), tile_num(self_obj)))) do begin
+                                        while ((anim_busy(self_obj) == 0) and (global_temp != rotation_to_tile(tile_num_in_direction(tile_num(self_obj), rotation_to_tile(tile_num_in_direction(tile_num(self_obj), obj_get_rot(self_obj), 1), tile_num(self_obj)), 1), tile_num(self_obj)))) do begin
                                             debug_msg("flee loop: rot == " + global_temp);
                                             animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(self_obj), global_temp, random(2, 5)), 0);
                                             global_temp := (global_temp + 1) % 6;
@@ -309,8 +309,8 @@ begin
                 dog_that_can_see_dude := 0;
             end
         end
-        if (tile_in_tile_rect(12335, 12313, 16535, 16513, tile_num(dude_obj))) then begin
-            if ((elevation(self_obj) != elevation(dude_obj)) or (has_trait(1, self_obj, 666) == 0)) then begin
+        if (tile_in_tile_rect(12335, 12313, 16535, 16513, dude_tile)) then begin
+            if ((elevation(self_obj) != dude_elevation) or (obj_is_visible_flag(self_obj) == 0)) then begin
                 destroy_object(self_obj);
             end
             else begin
@@ -320,10 +320,10 @@ begin
                             if (anim_busy(self_obj) == 0) then begin
                                 if ((tile_distance_objs(self_obj, dude_obj) < 8) or obj_can_see_obj(self_obj, dude_obj)) then begin
                                     if (anim_busy(self_obj) == 0) then begin
-                                        global_temp := rotation_to_tile(tile_num(dude_obj), tile_num(self_obj));
+                                        global_temp := rotation_to_tile(dude_tile, tile_num(self_obj));
                                         animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(self_obj), global_temp, random(3, 10)), 1);
                                         global_temp := (global_temp + 1) % 6;
-                                        while ((anim_busy(self_obj) == 0) and (global_temp != rotation_to_tile(tile_num(dude_obj), tile_num(self_obj)))) do begin
+                                        while ((anim_busy(self_obj) == 0) and (global_temp != rotation_to_tile(dude_tile, tile_num(self_obj)))) do begin
                                             debug_msg("flee loop: rot == " + global_temp);
                                             animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(self_obj), global_temp, random(3, 10)), 1);
                                             global_temp := (global_temp + 1) % 6;
@@ -333,20 +333,20 @@ begin
                             end
                         end
                         else begin
-                            attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+                            attack(dude_obj);
                         end
                     end
                 end
                 else begin
                     if (tile_distance_objs(self_obj, dude_obj) > 1) then begin
-                        LVar0 := tile_num_in_direction(tile_num(dude_obj), rotation_to_tile(tile_num(dude_obj), 15127), random(1, 2));
+                        LVar0 := tile_num_in_direction(dude_tile, rotation_to_tile(dude_tile, 15127), random(1, 2));
                         if (tile_in_tile_rect(12335, 12313, 16535, 16513, LVar0)) then begin
                             animate_move_obj_to_tile(self_obj, LVar0, 1 bwor 16);
                         end
                     end
-                    if (((global_var(297) bwand 4) != 0) == 0) then begin
-                        set_global_var(297, global_var(297) bwor 4);
-                        display_msg(message_str(206, 210));
+                    if (((global_var(GVAR_MODOC_GENERIC_FLAG_1) bwand 4) != 0) == 0) then begin
+                        set_global_var(GVAR_MODOC_GENERIC_FLAG_1, global_var(GVAR_MODOC_GENERIC_FLAG_1) bwor 4);
+                        display_mstr(210);
                         if (random(0, 2) != 1) then begin
                             play_sfx("maddogaq");
                         end
@@ -373,57 +373,57 @@ end
 
 procedure destroy_p_proc
 begin
-    if (cur_map_index == 20) then begin
+    if (CUR_MAP_MODBRAH) then begin
         if (source_obj == dude_obj) then begin
-            set_global_var(4, global_var(4) + 1);
-            if (metarule(51, self_obj) == 2) then begin
+            set_global_var(GVAR_BAD_MONSTER, global_var(GVAR_BAD_MONSTER) + 1);
+            if (critter_kill_type(self_obj) == KILL_TYPE_children_kills) then begin
                 if (source_obj == dude_obj) then begin
-                    set_global_var(2, 0);
-                    set_global_var(1, global_var(1) + 1);
-                    set_global_var(0, global_var(0) + -15);
+                    set_global_var(GVAR_CHAMPION_REPUTATION, 0);
+                    set_global_var(GVAR_CHILDKILLER_REPUTATION, global_var(GVAR_CHILDKILLER_REPUTATION) + 1);
+                    set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + -15);
                     debug_msg("Player gains " + -15 + " Karma Points.");
-                    set_global_var(37, 0);
-                    set_global_var(38, 0);
-                    set_global_var(39, 0);
-                    set_global_var(40, 0);
-                    set_global_var(41, 0);
-                    set_global_var(42, 0);
-                    set_global_var(43, 0);
-                    set_global_var(44, 0);
-                    set_global_var(45, 0);
-                    if (global_var(0) >= 1000) then begin
-                        set_global_var(37, 1);
+                    set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+                    set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+                    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+                    set_global_var(GVAR_KARMA_DEFENDER, 0);
+                    set_global_var(GVAR_KARMA_WANDERER, 0);
+                    set_global_var(GVAR_KARMA_BETRAYER, 0);
+                    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+                    set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+                    set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+                        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
                     end
                     else begin
-                        if (global_var(0) >= 750) then begin
-                            set_global_var(38, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                            set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
                         end
                         else begin
-                            if (global_var(0) >= 500) then begin
-                                set_global_var(39, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                                set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                             end
                             else begin
-                                if (global_var(0) >= 250) then begin
-                                    set_global_var(40, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                                    set_global_var(GVAR_KARMA_DEFENDER, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -250) then begin
-                                        set_global_var(41, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                                        set_global_var(GVAR_KARMA_WANDERER, 1);
                                     end
                                     else begin
-                                        if (global_var(0) > -500) then begin
-                                            set_global_var(42, 1);
+                                        if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                            set_global_var(GVAR_KARMA_BETRAYER, 1);
                                         end
                                         else begin
-                                            if (global_var(0) > -750) then begin
-                                                set_global_var(43, 1);
+                                            if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                                set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                             end
                                             else begin
-                                                if (global_var(0) > -1000) then begin
-                                                    set_global_var(44, 1);
+                                                if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                                    set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                                 end
                                                 else begin
-                                                    set_global_var(45, 1);
+                                                    set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                                 end
                                             end
                                         end
@@ -432,62 +432,62 @@ begin
                             end
                         end
                     end
-                    set_global_var(52, global_var(52) + -8);
+                    set_global_var(GVAR_TOWN_REP_MODOC, global_var(GVAR_TOWN_REP_MODOC) + -8);
                     debug_msg("Added " + -8 + " to Town Rep");
                 end
             end
-            if (((global_var(4) + global_var(5)) >= 25) and ((global_var(5) > (2 * global_var(4))) or (global_var(3) == 1))) then begin
-                set_global_var(3, 1);
-                set_global_var(2, 0);
+            if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
+                set_global_var(GVAR_BERSERKER_REPUTATION, 1);
+                set_global_var(GVAR_CHAMPION_REPUTATION, 0);
             end
-            if (((global_var(4) + global_var(5)) >= 25) and ((global_var(4) > (3 * global_var(5))) or (global_var(2) == 1)) and (global_var(1) == 0)) then begin
-                set_global_var(2, 1);
-                set_global_var(3, 0);
+            if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(4) > (3 * global_var(GVAR_GOOD_MONSTER))) or (global_var(GVAR_CHAMPION_REPUTATION) == 1)) and (global_var(GVAR_CHILDKILLER_REPUTATION) == 0)) then begin
+                set_global_var(GVAR_CHAMPION_REPUTATION, 1);
+                set_global_var(GVAR_BERSERKER_REPUTATION, 0);
             end
-            set_global_var(0, global_var(0) + 5);
+            set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + 5);
             debug_msg("Player gains " + 5 + " Karma Points.");
-            set_global_var(37, 0);
-            set_global_var(38, 0);
-            set_global_var(39, 0);
-            set_global_var(40, 0);
-            set_global_var(41, 0);
-            set_global_var(42, 0);
-            set_global_var(43, 0);
-            set_global_var(44, 0);
-            set_global_var(45, 0);
-            if (global_var(0) >= 1000) then begin
-                set_global_var(37, 1);
+            set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+            set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+            set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+            set_global_var(GVAR_KARMA_DEFENDER, 0);
+            set_global_var(GVAR_KARMA_WANDERER, 0);
+            set_global_var(GVAR_KARMA_BETRAYER, 0);
+            set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+            set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+            set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+                set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
             end
             else begin
-                if (global_var(0) >= 750) then begin
-                    set_global_var(38, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                    set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
                 end
                 else begin
-                    if (global_var(0) >= 500) then begin
-                        set_global_var(39, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                        set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                     end
                     else begin
-                        if (global_var(0) >= 250) then begin
-                            set_global_var(40, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                            set_global_var(GVAR_KARMA_DEFENDER, 1);
                         end
                         else begin
-                            if (global_var(0) > -250) then begin
-                                set_global_var(41, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                                set_global_var(GVAR_KARMA_WANDERER, 1);
                             end
                             else begin
-                                if (global_var(0) > -500) then begin
-                                    set_global_var(42, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                    set_global_var(GVAR_KARMA_BETRAYER, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -750) then begin
-                                        set_global_var(43, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                        set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                     end
                                     else begin
-                                        if (global_var(0) > -1000) then begin
-                                            set_global_var(44, 1);
+                                        if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                            set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                         end
                                         else begin
-                                            set_global_var(45, 1);
+                                            set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                         end
                                     end
                                 end
@@ -496,17 +496,17 @@ begin
                     end
                 end
             end
-            set_global_var(52, global_var(52) + 2);
+            set_global_var(GVAR_TOWN_REP_MODOC, global_var(GVAR_TOWN_REP_MODOC) + 2);
             debug_msg("Added " + 2 + " to Town Rep");
         end
         if (dog_object == self_obj) then begin
             dog_object := 0;
         end
-        set_global_var(116, global_var(116) - 1);
-        if (global_var(116) <= 0) then begin
-            set_global_var(116, -1);
+        set_global_var(GVAR_MODOC_BRAHMIN_ALIVE, global_var(GVAR_MODOC_BRAHMIN_ALIVE) - 1);
+        if (global_var(GVAR_MODOC_BRAHMIN_ALIVE) <= 0) then begin
+            set_global_var(GVAR_MODOC_BRAHMIN_ALIVE, -1);
         end
-        debug_msg("current dogs status == " + global_var(116));
+        debug_msg("current dogs status == " + global_var(GVAR_MODOC_BRAHMIN_ALIVE));
     end
     else begin
         if (dog_that_can_see_dude == self_obj) then begin
@@ -517,12 +517,12 @@ end
 
 procedure look_at_p_proc
 begin
-    display_msg(message_str(206, 100 + (cur_map_index == 20)));
+    display_mstr(100 + (CUR_MAP_MODBRAH));
 end
 
 procedure description_p_proc
 begin
-    display_msg(message_str(206, 120 + (cur_map_index == 20)));
+    display_mstr(120 + (CUR_MAP_MODBRAH));
 end
 
 procedure use_skill_on_p_proc
@@ -541,17 +541,17 @@ end
 
 procedure map_enter_p_proc
 begin
-    if ((cur_map_index == 19) and (elevation(self_obj) != ((global_var(297) bwand 1) != 0))) then begin
-        move_to(self_obj, tile_num(self_obj), (global_var(297) bwand 1) != 0);
+    if ((CUR_MAP_MODINN) and (elevation(self_obj) != ((global_var(GVAR_MODOC_GENERIC_FLAG_1) bwand 1) != 0))) then begin
+        move_to(self_obj, tile_num(self_obj), (global_var(GVAR_MODOC_GENERIC_FLAG_1) bwand 1) != 0);
     end
-    critter_add_trait(self_obj, 1, 5, 186);
-    if (cur_map_index == 20) then begin
-        critter_add_trait(self_obj, 1, 6, 33);
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_DOG_TOUGH);
+    if (CUR_MAP_MODBRAH) then begin
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 33);
         last_update := game_time;
         if (obj_pid(self_obj) != 16777225) then begin
             if (local_var(0) == 0) then begin
                 set_local_var(0, tile_num(self_obj));
-                set_global_var(116, global_var(116) + 1);
+                set_global_var(GVAR_MODOC_BRAHMIN_ALIVE, global_var(GVAR_MODOC_BRAHMIN_ALIVE) + 1);
             end
         end
         else begin
@@ -559,7 +559,7 @@ begin
         end
     end
     else begin
-        critter_add_trait(self_obj, 1, 6, 28);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 28);
         if (local_var(0) == 0) then begin
             set_local_var(0, tile_num(self_obj));
             set_local_var(2, tile_num(self_obj));
@@ -567,7 +567,7 @@ begin
             add_timer_event(self_obj, game_ticks(random(15, 30)), 2);
         end
         else begin
-            if (metarule(22, 0) == 0) then begin
+            if (is_loading_game == 0) then begin
                 if (tile_num(self_obj) != local_var(2)) then begin
                     critter_attempt_placement(self_obj, local_var(2), 0);
                 end
@@ -578,21 +578,21 @@ end
 
 procedure map_update_p_proc
 begin
-    if (cur_map_index == 20) then begin
+    if (CUR_MAP_MODBRAH) then begin
         if (((game_time - last_update) > (15 * (60 * 10))) and (last_update != 0)) then begin
-            critter_attempt_placement(self_obj, local_var(0), elevation(dude_obj));
-            if (local_var(1) and ((global_var(115) < 0) == 0)) then begin
-                if (((global_var(116) < 0) == 0) and (global_var(99) != 0)) then begin
-                    if ((global_var(115) < 0) == 0) then begin
-                        set_global_var(115, -1);
+            critter_attempt_placement(self_obj, local_var(0), dude_elevation);
+            if (local_var(1) and ((global_var(GVAR_JONNY_TILE) < 0) == 0)) then begin
+                if (((global_var(GVAR_MODOC_BRAHMIN_ALIVE) < 0) == 0) and (global_var(GVAR_MODOC_KILL_ALL_BRAHMIN_TIME) != 0)) then begin
+                    if ((global_var(GVAR_JONNY_TILE) < 0) == 0) then begin
+                        set_global_var(GVAR_JONNY_TILE, -1);
                     end
-                    if (cur_map_index == 20) then begin
+                    if (CUR_MAP_MODBRAH) then begin
                         kill_critter_type(16777226, random(61, 62));
                         if (map_var(0) == 0) then begin
-                            if (global_var(115) < 0) then begin
-                                set_map_var(0, random(0, global_var(116) / 2));
+                            if (global_var(GVAR_JONNY_TILE) < 0) then begin
+                                set_map_var(0, random(0, global_var(GVAR_MODOC_BRAHMIN_ALIVE) / 2));
                                 while (map_var(0) > 0) do begin
-                                    temp_dog := create_object_sid(16777225, 0, 0, 206);
+                                    temp_dog := create_object_sid(16777225, 0, 0, NAME);
                                     critter_attempt_placement(temp_dog, tile_num_in_direction(16915, random(0, 5), random(2, 10)), 0);
                                     set_map_var(0, map_var(0) - 1);
                                 end
@@ -612,8 +612,8 @@ end
 
 procedure map_exit_p_proc
 begin
-    if (cur_map_index == 20) then begin
-        critter_attempt_placement(self_obj, local_var(0), elevation(dude_obj));
+    if (CUR_MAP_MODBRAH) then begin
+        critter_attempt_placement(self_obj, local_var(0), dude_elevation);
     end
     else begin
         if (cage_door_obj != 0) then begin
@@ -630,13 +630,13 @@ begin
     if (fixed_param == 1) then begin
         if (combat_is_initialized == 0) then begin
             if (obj_can_see_obj(self_obj, dude_obj)) then begin
-                if (tile_in_tile_rect(12335, 12313, 16535, 16513, tile_num(dude_obj))) then begin
+                if (tile_in_tile_rect(12335, 12313, 16535, 16513, dude_tile)) then begin
                     if (anim_busy(self_obj) == 0) then begin
                         if (tile_distance_objs(self_obj, dude_obj) <= 3) then begin
-                            anim(self_obj, 1000, rotation_to_tile(tile_num(self_obj), tile_num(dude_obj)));
+                            anim(self_obj, 1000, rotation_to_tile(tile_num(self_obj), dude_tile));
                         end
                     end
-                    float_msg(self_obj, message_str(206, random(200, 202)), 8 + ((2 - 8) * has_trait(0, dude_obj, 22)));
+                    float_msg(self_obj, mstr(random(200, 202)), 8 + ((2 - 8) * has_trait(TRAIT_PERK, dude_obj, PERK_empathy)));
                 end
             end
         end
@@ -645,7 +645,7 @@ begin
     if (fixed_param == 2) then begin
         if (combat_is_initialized == 0) then begin
             debug_msg("telling dog to switch tile");
-            if ((anim_busy(self_obj) == 0) and (tile_in_tile_rect(12335, 12313, 16535, 16513, tile_num(dude_obj)) == 0)) then begin
+            if ((anim_busy(self_obj) == 0) and (tile_in_tile_rect(12335, 12313, 16535, 16513, dude_tile) == 0)) then begin
                 if (tile_num(self_obj) == local_var(0)) then begin
                     LVar0 := tile_num_in_direction(local_var(0), random(0, 5), random(3, 10));
                     if (tile_in_tile_rect(12335, 12313, 16535, 16513, LVar0)) then begin
@@ -671,7 +671,7 @@ begin
             if (brahmin_object == 0) then begin
             end
             else begin
-                if ((global_var(402) bwand 32768) != 0) then begin
+                if ((global_var(GVAR_MODOC_GENERIC_FLAG_2) bwand 32768) != 0) then begin
                     if (tile_distance_objs(self_obj, brahmin_object) < tile_distance_objs(dog_object, brahmin_object)) then begin
                         brahmin_object := self_obj;
                     end
@@ -690,10 +690,10 @@ begin
         end
     end
     if (((local_var(3) bwand 2) != 0) == 0) then begin
-        if (cur_map_index == 20) then begin
+        if (CUR_MAP_MODBRAH) then begin
             if (brahmin_object != 0) then begin
                 set_local_var(3, local_var(3) bwor 512);
-                attack_complex(brahmin_object, 0, 1, 0, 0, 30000, 0, 0);
+                attack(brahmin_object);
             end
         end
     end
@@ -702,4 +702,3 @@ end
 procedure use_obj_on_p_proc
 begin
 end
-
