diff --git a/Mods/MegaMod/Original/MCBOB3.ssl b/Mods/MegaMod/ReDefine/MCBOB3.ssl
index aa825d5..feda983 100644
--- a/Mods/MegaMod/Original/MCBOB3.ssl
+++ b/Mods/MegaMod/ReDefine/MCBOB3.ssl
@@ -169,23 +169,23 @@ end
 procedure critter_p_proc
 begin
     if (elevation(self_obj) == 2) then begin
-        if (global_var(1008) < 12) then begin
-            if (global_var(1008) == 4) then begin
+        if (global_var(GVAR_LINK_THETOWNS) < 12) then begin
+            if (global_var(GVAR_LINK_THETOWNS) == 4) then begin
                 if (tile_distance_objs(dude_obj, self_obj) > 2) then begin
                     animate_move_obj_to_tile(dude_obj, tile_num(self_obj), 0);
                 end
             end
             else begin
-                if (not(global_var(1008) == 8)) then begin
-                    animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(dude_obj), random(0, 5), 5), 1);
+                if (not(global_var(GVAR_LINK_THETOWNS) == 8)) then begin
+                    animate_move_obj_to_tile(self_obj, tile_num_in_direction(dude_tile, random(0, 5), 5), 1);
                 end
             end
-            if (not(global_var(1008) == 4)) then begin
-                if (global_var(1008) == 8) then begin
+            if (not(global_var(GVAR_LINK_THETOWNS) == 4)) then begin
+                if (global_var(GVAR_LINK_THETOWNS) == 8) then begin
                 end
                 else begin
                     if (obj_can_see_obj(self_obj, dude_obj)) then begin
-                        attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+                        attack(dude_obj);
                     end
                 end
             end
@@ -202,7 +202,7 @@ begin
         end
     end
     else begin
-        if (global_var(1008) == 9) then begin
+        if (global_var(GVAR_LINK_THETOWNS) == 9) then begin
             if (local_var(9) == 0) then begin
                 set_local_var(9, 1);
                 add_timer_event(self_obj, 35, 5);
@@ -245,7 +245,7 @@ end
 
 procedure talk_p_proc
 begin
-    if (global_var(1008) == 11) then begin
+    if (global_var(GVAR_LINK_THETOWNS) == 11) then begin
         start_gdialog(1569, self_obj, 4, -1, -1);
         gsay_start;
         call Node007();
@@ -253,7 +253,7 @@ begin
         end_dialogue;
     end
     else begin
-        if (global_var(1008) == 12) then begin
+        if (global_var(GVAR_LINK_THETOWNS) == 12) then begin
             start_gdialog(1569, self_obj, 4, -1, -1);
             gsay_start;
             call Node008();
@@ -270,7 +270,7 @@ begin
                 end_dialogue;
             end
             else begin
-                float_msg(self_obj, message_str(1569, random(100, 101)), 8);
+                float_msg(self_obj, mstr(random(100, 101)), FLOAT_MSG_YELLOW);
             end
         end
     end
@@ -278,14 +278,14 @@ end
 
 procedure Node001
 begin
-    gsay_reply(1569, 200);
-    giq_option(4, 1569, 201, Node002, 50);
+    Reply(200);
+    giq_option(4, NAME, 201, Node002, NEUTRAL_REACTION);
 end
 
 procedure Node002
 begin
-    gsay_reply(1569, 202);
-    giq_option(4, 1569, 203, Node003, 50);
+    Reply(202);
+    giq_option(4, NAME, 203, Node003, NEUTRAL_REACTION);
 end
 
 procedure Node003
@@ -299,7 +299,7 @@ end
 procedure Node004
 begin
     game_ui_disable;
-    float_msg(self_obj, message_str(1569, 102), 2);
+    float_msg(self_obj, mstr(102), FLOAT_MSG_RED);
     add_timer_event(self_obj, 35, 1);
 end
 
@@ -318,10 +318,10 @@ begin
         anim(self_obj, 1000, 4);
         move_to(dude_obj, 17274, 0);
         anim(dude_obj, 1000, 5);
-        LVar0 := create_object_sid(33556371, 16877, 0, 1570);
+        LVar0 := create_object_sid(33556371, 16877, 0, SCRIPT_MODOCT41);
         destroy_object(fence_obj);
-        set_global_var(1008, 9);
-        LVar1 := create_object_sid(289, 17476, 0, -1);
+        set_global_var(GVAR_LINK_THETOWNS, 9);
+        LVar1 := create_object(289, 17476, 0);
         gfade_in(10 * 6);
     end
     else begin
@@ -334,33 +334,33 @@ end
 
 procedure Node007
 begin
-    gsay_reply(1569, message_str(1569, 204) + obj_name(dude_obj) + message_str(1569, 206));
-    giq_option(4, 1569, 205, Node007a, 50);
+    Reply(mstr(204) + dude_name + mstr(206));
+    giq_option(4, NAME, 205, Node007a, NEUTRAL_REACTION);
     give_exp_points(500);
-    display_msg(message_str(14, 100) + 500 + message_str(14, 101));
+    display_msg(g_mstr(100) + 500 + g_mstr(101));
 end
 
 procedure Node007a
 begin
-    set_global_var(1008, 12);
+    set_global_var(GVAR_LINK_THETOWNS, 12);
     gfade_out(10 * 6);
     move_to(self_obj, 20450, 2);
     anim(self_obj, 1000, 4);
     gfade_in(10 * 6);
-    display_msg(message_str(1569, 210));
+    display_mstr(210);
 end
 
 procedure Node008
 begin
-    gsay_reply(1569, 207);
-    giq_option(4, 1569, 208, Node009, 50);
-    giq_option(4, 1569, 203, Node999, 50);
+    Reply(207);
+    giq_option(4, NAME, 208, Node009, NEUTRAL_REACTION);
+    giq_option(4, NAME, 203, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node009
 begin
-    gsay_reply(1569, 209);
-    giq_option(4, 1569, 203, Node999, 50);
+    Reply(209);
+    giq_option(4, NAME, 203, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node999
@@ -374,13 +374,13 @@ end
 procedure timed_event_p_proc
 begin
     if (fixed_param == 1) then begin
-        float_msg(fence_obj, message_str(1569, 103), 7);
+        float_msg(fence_obj, mstr(103), FLOAT_MSG_LIGHT_RED);
         add_timer_event(self_obj, 35, 2);
     end
     else begin
         if (fixed_param == 2) then begin
-            float_msg(self_obj, message_str(1569, 104), 2);
-            set_global_var(1008, 8);
+            float_msg(self_obj, mstr(104), FLOAT_MSG_RED);
+            set_global_var(GVAR_LINK_THETOWNS, 8);
             add_timer_event(self_obj, 10, 3);
         end
         else begin
@@ -394,12 +394,12 @@ begin
                 end
                 else begin
                     if (fixed_param == 5) then begin
-                        float_msg(dude_obj, message_str(1569, 211), 7);
+                        float_msg(dude_obj, mstr(211), FLOAT_MSG_LIGHT_RED);
                         add_timer_event(self_obj, 35, 6);
                     end
                     else begin
                         if (fixed_param == 6) then begin
-                            float_msg(self_obj, message_str(1569, 111), 2);
+                            float_msg(self_obj, mstr(111), FLOAT_MSG_RED);
                             game_ui_enable;
                         end
                     end
@@ -408,4 +408,3 @@ begin
         end
     end
 end
-
