diff --git a/Mods/MegaMod/Original/MCKILDOG.ssl b/Mods/MegaMod/ReDefine/MCKILDOG.ssl
index f40845b..01516e5 100644
--- a/Mods/MegaMod/Original/MCKILDOG.ssl
+++ b/Mods/MegaMod/ReDefine/MCKILDOG.ssl
@@ -191,14 +191,14 @@ begin
         end
         if (LVar0 != 0) then begin
             if (tile_distance_objs(self_obj, LVar0) > 2) then begin
-                animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(LVar0), (has_trait(1, LVar0, 10) + 3) % 6, 2), 1);
+                animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(LVar0), (obj_get_rot(LVar0) + 3) % 6, 2), 1);
             end
             if (game_time >= local_var(0)) then begin
                 set_local_var(0, game_time + (random(1, 5) * 10));
                 if (random(0, 2) != 1) then begin
                     play_sfx("dogbark2");
                 end
-                float_msg(self_obj, message_str(578, random(100, 102)), 8 + ((2 - 8) * has_trait(0, dude_obj, 22)));
+                float_msg(self_obj, mstr(random(100, 102)), 8 + ((2 - 8) * has_trait(TRAIT_PERK, dude_obj, PERK_empathy)));
             end
         end
         else begin
@@ -239,8 +239,8 @@ end
 
 procedure map_enter_p_proc
 begin
-    critter_add_trait(self_obj, 1, 6, 28);
-    critter_add_trait(self_obj, 1, 5, 21);
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 28);
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_GENERIC_DOG);
     if (local_var(1) == 0) then begin
         set_local_var(1, tile_num(self_obj));
         add_timer_event(self_obj, game_ticks(random(90, 120)), 1);
@@ -272,4 +272,3 @@ end
 procedure use_obj_on_p_proc
 begin
 end
-
