diff --git a/Mods/MegaMod/Original/MCMEGREW.ssl b/Mods/MegaMod/ReDefine/MCMEGREW.ssl
index c910d51..a303bd0 100644
--- a/Mods/MegaMod/Original/MCMEGREW.ssl
+++ b/Mods/MegaMod/ReDefine/MCMEGREW.ssl
@@ -49,17 +49,17 @@ procedure look_at_p_proc
 begin
     script_overrides;
     if (local_var(4) == 0) then begin
-        display_msg(message_str(1573, 100));
+        display_mstr(100);
     end
     else begin
-        display_msg(message_str(1573, 101));
+        display_mstr(101);
     end
 end
 
 procedure description_p_proc
 begin
     script_overrides;
-    display_msg(message_str(1573, 102));
+    display_mstr(102);
 end
 
 procedure use_skill_on_p_proc
@@ -72,7 +72,7 @@ end
 
 procedure damage_p_proc
 begin
-    attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+    attack(dude_obj);
 end
 
 procedure map_enter_p_proc
@@ -85,9 +85,9 @@ end
 
 procedure talk_p_proc
 begin
-    if (global_var(1010) == 0) then begin
+    if (global_var(GVAR_ROOM_OPEN) == 0) then begin
         if (local_var(4) == 2) then begin
-            float_msg(self_obj, message_str(1573, 105), 0);
+            float_msg(self_obj, mstr(105), FLOAT_MSG_NORMAL);
         end
         else begin
             start_gdialog(1573, self_obj, 4, -1, -1);
@@ -126,61 +126,61 @@ end
 
 procedure Node001
 begin
-    gsay_reply(1573, 106);
-    giq_option(4, 1573, 107, Node002, 50);
-    giq_option(4, 1573, 145, Node003a, 50);
+    Reply(106);
+    giq_option(4, NAME, 107, Node002, NEUTRAL_REACTION);
+    giq_option(4, NAME, 145, Node003a, NEUTRAL_REACTION);
 end
 
 procedure Node002
 begin
-    gsay_reply(1573, 108);
-    giq_option(4, 1573, 109, Node003, 50);
-    giq_option(4, 1573, 110, Node003a, 50);
+    Reply(108);
+    giq_option(4, NAME, 109, Node003, NEUTRAL_REACTION);
+    giq_option(4, NAME, 110, Node003a, NEUTRAL_REACTION);
 end
 
 procedure Node003
 begin
-    gsay_reply(1573, 111);
-    giq_option(4, 1573, 112, Node003a, 50);
+    Reply(111);
+    giq_option(4, NAME, 112, Node003a, NEUTRAL_REACTION);
 end
 
 procedure Node004
 begin
-    gsay_reply(1573, 113);
-    giq_option(4, 1573, 114, Node005, 50);
-    giq_option(4, 1573, 145, Node003a, 50);
+    Reply(113);
+    giq_option(4, NAME, 114, Node005, NEUTRAL_REACTION);
+    giq_option(4, NAME, 145, Node003a, NEUTRAL_REACTION);
 end
 
 procedure Node005
 begin
-    gsay_reply(1573, 115);
-    giq_option(4, 1573, 116, Node006, 50);
+    Reply(115);
+    giq_option(4, NAME, 116, Node006, NEUTRAL_REACTION);
 end
 
 procedure Node006
 begin
-    gsay_reply(1573, 117);
-    giq_option(4, 1573, 118, Node007, 50);
+    Reply(117);
+    giq_option(4, NAME, 118, Node007, NEUTRAL_REACTION);
 end
 
 procedure Node007
 begin
-    gsay_reply(1573, 119);
-    if (has_skill(dude_obj, 14) >= 100) then begin
-        giq_option(4, 1573, 120, Node008, 50);
+    Reply(119);
+    if (has_skill(dude_obj, SKILL_CONVERSANT) >= 100) then begin
+        giq_option(4, NAME, 120, Node008, NEUTRAL_REACTION);
     end
-    if (has_skill(dude_obj, 14) < 100) then begin
-        giq_option(4, 1573, 121, Node009, 50);
+    if (has_skill(dude_obj, SKILL_CONVERSANT) < 100) then begin
+        giq_option(4, NAME, 121, Node009, NEUTRAL_REACTION);
     end
 end
 
 procedure Node008
 begin
-    gsay_reply(1573, 122);
+    Reply(122);
     if (item_caps_total(dude_obj) > 1500) then begin
-        giq_option(4, 1573, 123, Node008b, 50);
+        giq_option(4, NAME, 123, Node008b, NEUTRAL_REACTION);
     end
-    giq_option(4, 1573, 124, Node008a, 50);
+    giq_option(4, NAME, 124, Node008a, NEUTRAL_REACTION);
 end
 
 procedure Node008a
@@ -190,15 +190,15 @@ end
 
 procedure Node008b
 begin
-    gsay_reply(1573, 125);
-    giq_option(4, 1573, 126, Node008c, 50);
+    Reply(125);
+    giq_option(4, NAME, 126, Node008c, NEUTRAL_REACTION);
     item_caps_adjust(self_obj, 1500);
     item_caps_adjust(dude_obj, -1500);
 end
 
 procedure Node008c
 begin
-    set_global_var(1010, 2);
+    set_global_var(GVAR_ROOM_OPEN, 2);
     gfade_out(10 * 6);
     game_ui_disable;
     move_to(self_obj, 16674, 2);
@@ -208,8 +208,8 @@ end
 
 procedure Node009a
 begin
-    gsay_reply(1573, 125);
-    giq_option(4, 1573, 126, Node008c, 50);
+    Reply(125);
+    giq_option(4, NAME, 126, Node008c, NEUTRAL_REACTION);
     item_caps_adjust(self_obj, 2000);
     item_caps_adjust(dude_obj, -2000);
 end
@@ -221,49 +221,49 @@ end
 
 procedure Node010
 begin
-    gsay_reply(1573, 130);
-    giq_option(4, 1573, message_str(1573, 131) + obj_name(dude_obj) + message_str(1573, 132), Node011, 50);
-    giq_option(4, 1573, 145, Node999, 50);
+    Reply(130);
+    giq_option(4, NAME, mstr(131) + dude_name + mstr(132), Node011, NEUTRAL_REACTION);
+    giq_option(4, NAME, 145, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node011
 begin
-    gsay_reply(1573, 133);
-    giq_option(4, 1573, 134, Node012, 50);
-    giq_option(4, 1573, 146, Node999, 50);
-    giq_option(4, 1573, 147, Node999, 51);
+    Reply(133);
+    giq_option(4, NAME, 134, Node012, NEUTRAL_REACTION);
+    giq_option(4, NAME, 146, Node999, NEUTRAL_REACTION);
+    giq_option(4, NAME, 147, Node999, BAD_REACTION);
 end
 
 procedure Node012
 begin
-    gsay_reply(1573, 135);
-    if (has_skill(dude_obj, 14) >= 100) then begin
-        giq_option(4, 1573, 136, Node008, 50);
+    Reply(135);
+    if (has_skill(dude_obj, SKILL_CONVERSANT) >= 100) then begin
+        giq_option(4, NAME, 136, Node008, NEUTRAL_REACTION);
     end
-    if (has_skill(dude_obj, 14) < 100) then begin
-        giq_option(4, 1573, 137, Node009, 50);
+    if (has_skill(dude_obj, SKILL_CONVERSANT) < 100) then begin
+        giq_option(4, NAME, 137, Node009, NEUTRAL_REACTION);
     end
 end
 
 procedure Node013
 begin
-    gsay_reply(1573, 138);
-    if ((has_skill(dude_obj, 14) < 100) and (item_caps_total(dude_obj) > 2000)) then begin
-        giq_option(4, 1573, 139, Node009a, 50);
+    Reply(138);
+    if ((has_skill(dude_obj, SKILL_CONVERSANT) < 100) and (item_caps_total(dude_obj) > 2000)) then begin
+        giq_option(4, NAME, 139, Node009a, NEUTRAL_REACTION);
     end
-    if ((has_skill(dude_obj, 14) >= 100) and (item_caps_total(dude_obj) > 1500)) then begin
-        giq_option(4, 1573, 140, Node012, 50);
+    if ((has_skill(dude_obj, SKILL_CONVERSANT) >= 100) and (item_caps_total(dude_obj) > 1500)) then begin
+        giq_option(4, NAME, 140, Node012, NEUTRAL_REACTION);
     end
-    giq_option(4, 1573, 141, Node999, 50);
+    giq_option(4, NAME, 141, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node009
 begin
-    gsay_reply(1573, 127);
+    Reply(127);
     if (item_caps_total(dude_obj) > 2000) then begin
-        giq_option(4, 1573, 128, Node009a, 50);
+        giq_option(4, NAME, 128, Node009a, NEUTRAL_REACTION);
     end
-    giq_option(4, 1573, 129, Node008a, 50);
+    giq_option(4, NAME, 129, Node008a, NEUTRAL_REACTION);
 end
 
 procedure timed_event_p_proc
@@ -282,4 +282,3 @@ begin
         set_local_var(4, 1);
     end
 end
-
