diff --git a/Mods/MegaMod/Original/MCSLGGRD.ssl b/Mods/MegaMod/ReDefine/MCSLGGRD.ssl
index 209adb2..02d84cc 100644
--- a/Mods/MegaMod/Original/MCSLGGRD.ssl
+++ b/Mods/MegaMod/ReDefine/MCSLGGRD.ssl
@@ -159,17 +159,17 @@ end
 
 procedure critter_p_proc
 begin
-    if (has_trait(1, self_obj, 666)) then begin
+    if (obj_is_visible_flag(self_obj)) then begin
         if (obj_can_see_obj(self_obj, dude_obj)) then begin
-            if (((local_var(0) bwand 2) != 0) or ((global_var(297) bwand 512) != 0)) then begin
+            if (((local_var(0) bwand 2) != 0) or ((global_var(GVAR_MODOC_GENERIC_FLAG_1) bwand 512) != 0)) then begin
                 if (critter_is_fleeing(self_obj)) then begin
                     if (anim_busy(self_obj) == 0) then begin
                         if ((tile_distance_objs(self_obj, dude_obj) < 8) or obj_can_see_obj(self_obj, dude_obj)) then begin
                             if (anim_busy(self_obj) == 0) then begin
-                                global_temp := rotation_to_tile(tile_num(dude_obj), tile_num(self_obj));
+                                global_temp := rotation_to_tile(dude_tile, tile_num(self_obj));
                                 animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(self_obj), global_temp, random(3, 10)), 1);
                                 global_temp := (global_temp + 1) % 6;
-                                while ((anim_busy(self_obj) == 0) and (global_temp != rotation_to_tile(tile_num(dude_obj), tile_num(self_obj)))) do begin
+                                while ((anim_busy(self_obj) == 0) and (global_temp != rotation_to_tile(dude_tile, tile_num(self_obj)))) do begin
                                     debug_msg("flee loop: rot == " + global_temp);
                                     animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(self_obj), global_temp, random(3, 10)), 1);
                                     global_temp := (global_temp + 1) % 6;
@@ -179,7 +179,7 @@ begin
                     end
                 end
                 else begin
-                    attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+                    attack(dude_obj);
                 end
             end
             else begin
@@ -206,7 +206,7 @@ end
 procedure talk_p_proc
 begin
     script_overrides;
-    if ((global_var(297) bwand 536870912) != 0) then begin
+    if ((global_var(GVAR_MODOC_GENERIC_FLAG_1) bwand 536870912) != 0) then begin
         call Node007();
     end
     else begin
@@ -225,65 +225,65 @@ end
 procedure destroy_p_proc
 begin
     variable LVar0 := 0;
-    LVar0 := create_object_sid(1, 0, 0, -1);
+    LVar0 := create_object(1, 0, 0);
     add_obj_to_inven(self_obj, LVar0);
     if (slag_guard_obj == self_obj) then begin
         slag_guard_obj := 0;
     end
     if (source_obj == dude_obj) then begin
-        set_global_var(5, global_var(5) + 1);
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(4) > (3 * global_var(5))) or (global_var(2) == 1)) and (global_var(1) == 0)) then begin
-            set_global_var(2, 1);
-            set_global_var(3, 0);
+        set_global_var(GVAR_GOOD_MONSTER, global_var(GVAR_GOOD_MONSTER) + 1);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(4) > (3 * global_var(GVAR_GOOD_MONSTER))) or (global_var(GVAR_CHAMPION_REPUTATION) == 1)) and (global_var(GVAR_CHILDKILLER_REPUTATION) == 0)) then begin
+            set_global_var(GVAR_CHAMPION_REPUTATION, 1);
+            set_global_var(GVAR_BERSERKER_REPUTATION, 0);
         end
-        if (metarule(51, self_obj) == 2) then begin
+        if (critter_kill_type(self_obj) == KILL_TYPE_children_kills) then begin
             if (source_obj == dude_obj) then begin
-                set_global_var(2, 0);
-                set_global_var(1, global_var(1) + 1);
-                set_global_var(0, global_var(0) + -15);
+                set_global_var(GVAR_CHAMPION_REPUTATION, 0);
+                set_global_var(GVAR_CHILDKILLER_REPUTATION, global_var(GVAR_CHILDKILLER_REPUTATION) + 1);
+                set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + -15);
                 debug_msg("Player gains " + -15 + " Karma Points.");
-                set_global_var(37, 0);
-                set_global_var(38, 0);
-                set_global_var(39, 0);
-                set_global_var(40, 0);
-                set_global_var(41, 0);
-                set_global_var(42, 0);
-                set_global_var(43, 0);
-                set_global_var(44, 0);
-                set_global_var(45, 0);
-                if (global_var(0) >= 1000) then begin
-                    set_global_var(37, 1);
+                set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+                set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+                set_global_var(GVAR_KARMA_DEFENDER, 0);
+                set_global_var(GVAR_KARMA_WANDERER, 0);
+                set_global_var(GVAR_KARMA_BETRAYER, 0);
+                set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+                set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+                set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+                    set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
                 end
                 else begin
-                    if (global_var(0) >= 750) then begin
-                        set_global_var(38, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
                     end
                     else begin
-                        if (global_var(0) >= 500) then begin
-                            set_global_var(39, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                            set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                         end
                         else begin
-                            if (global_var(0) >= 250) then begin
-                                set_global_var(40, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                                set_global_var(GVAR_KARMA_DEFENDER, 1);
                             end
                             else begin
-                                if (global_var(0) > -250) then begin
-                                    set_global_var(41, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                                    set_global_var(GVAR_KARMA_WANDERER, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -500) then begin
-                                        set_global_var(42, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                        set_global_var(GVAR_KARMA_BETRAYER, 1);
                                     end
                                     else begin
-                                        if (global_var(0) > -750) then begin
-                                            set_global_var(43, 1);
+                                        if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                            set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                         end
                                         else begin
-                                            if (global_var(0) > -1000) then begin
-                                                set_global_var(44, 1);
+                                            if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                                set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                             end
                                             else begin
-                                                set_global_var(45, 1);
+                                                set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                             end
                                         end
                                     end
@@ -292,58 +292,58 @@ begin
                         end
                     end
                 end
-                set_global_var(308, global_var(308) + -8);
+                set_global_var(GVAR_TOWN_REP_GHOST_FARM, global_var(GVAR_TOWN_REP_GHOST_FARM) + -8);
                 debug_msg("Added " + -8 + " to Town Rep");
             end
         end
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(5) > (2 * global_var(4))) or (global_var(3) == 1))) then begin
-            set_global_var(3, 1);
-            set_global_var(2, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
+            set_global_var(GVAR_BERSERKER_REPUTATION, 1);
+            set_global_var(GVAR_CHAMPION_REPUTATION, 0);
         end
-        set_global_var(0, global_var(0) + -10);
+        set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + -10);
         debug_msg("Player gains " + -10 + " Karma Points.");
-        set_global_var(37, 0);
-        set_global_var(38, 0);
-        set_global_var(39, 0);
-        set_global_var(40, 0);
-        set_global_var(41, 0);
-        set_global_var(42, 0);
-        set_global_var(43, 0);
-        set_global_var(44, 0);
-        set_global_var(45, 0);
-        if (global_var(0) >= 1000) then begin
-            set_global_var(37, 1);
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+        set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+        set_global_var(GVAR_KARMA_DEFENDER, 0);
+        set_global_var(GVAR_KARMA_WANDERER, 0);
+        set_global_var(GVAR_KARMA_BETRAYER, 0);
+        set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+        set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+            set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
         end
         else begin
-            if (global_var(0) >= 750) then begin
-                set_global_var(38, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
             end
             else begin
-                if (global_var(0) >= 500) then begin
-                    set_global_var(39, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                 end
                 else begin
-                    if (global_var(0) >= 250) then begin
-                        set_global_var(40, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                        set_global_var(GVAR_KARMA_DEFENDER, 1);
                     end
                     else begin
-                        if (global_var(0) > -250) then begin
-                            set_global_var(41, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                            set_global_var(GVAR_KARMA_WANDERER, 1);
                         end
                         else begin
-                            if (global_var(0) > -500) then begin
-                                set_global_var(42, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                set_global_var(GVAR_KARMA_BETRAYER, 1);
                             end
                             else begin
-                                if (global_var(0) > -750) then begin
-                                    set_global_var(43, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -1000) then begin
-                                        set_global_var(44, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                     end
                                     else begin
-                                        set_global_var(45, 1);
+                                        set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                     end
                                 end
                             end
@@ -352,7 +352,7 @@ begin
                 end
             end
         end
-        set_global_var(308, global_var(308) + -5);
+        set_global_var(GVAR_TOWN_REP_GHOST_FARM, global_var(GVAR_TOWN_REP_GHOST_FARM) + -5);
         debug_msg("Added " + -5 + " to Town Rep");
     end
     return 0;
@@ -362,14 +362,14 @@ end
 procedure look_at_p_proc
 begin
     script_overrides;
-    display_msg(message_str(819, 100 + ((local_var(0) bwand 4) != 0)));
+    display_mstr(100 + ((local_var(0) bwand 4) != 0));
     set_local_var(0, local_var(0) bwor 4);
 end
 
 procedure description_p_proc
 begin
     script_overrides;
-    display_msg(message_str(819, 150));
+    display_mstr(150);
 end
 
 procedure use_skill_on_p_proc
@@ -380,99 +380,99 @@ procedure damage_p_proc
 begin
     if (source_obj == dude_obj) then begin
         call Node998();
-        set_global_var(297, global_var(297) bwor 512);
+        set_global_var(GVAR_MODOC_GENERIC_FLAG_1, global_var(GVAR_MODOC_GENERIC_FLAG_1) bwor 512);
         debug_msg("HEY: I SET ENEMY SLAGS!!!! I'M SCRIPT #" + 819);
     end
 end
 
 procedure map_enter_p_proc
 begin
-    if ((cur_map_index == 19) and (elevation(self_obj) != ((global_var(297) bwand 1) != 0))) then begin
-        move_to(self_obj, tile_num(self_obj), (global_var(297) bwand 1) != 0);
+    if ((CUR_MAP_MODINN) and (elevation(self_obj) != ((global_var(GVAR_MODOC_GENERIC_FLAG_1) bwand 1) != 0))) then begin
+        move_to(self_obj, tile_num(self_obj), (global_var(GVAR_MODOC_GENERIC_FLAG_1) bwand 1) != 0);
     end
-    if (metarule(22, 0) == 0) then begin
-        if (((global_var(402) bwand 2048) != 0) or ((global_var(112) == 2) or (((global_var(403) - (game_time / (10 * 60 * 60 * 24))) <= 0) or ((global_var(403) - (game_time / (10 * 60 * 60 * 24))) > 31) and (global_var(403) > 0)))) then begin
-            if (cur_map_index == 24) then begin
-                if (((global_var(402) bwand 1024) != 0) and (global_var(310) <= 0) or ((global_var(112) == 2) or (((global_var(403) - (game_time / (10 * 60 * 60 * 24))) <= 0) or ((global_var(403) - (game_time / (10 * 60 * 60 * 24))) > 31) and (global_var(403) > 0)))) then begin
+    if (is_loading_game == 0) then begin
+        if (((global_var(GVAR_MODOC_GENERIC_FLAG_2) bwand 2048) != 0) or ((global_var(GVAR_MODOC_GHOST_FARM_SEED) == 2) or (((global_var(GVAR_SLAUGHTER_SLAG_TIME) - (game_time / (10 * 60 * 60 * 24))) <= 0) or ((global_var(GVAR_SLAUGHTER_SLAG_TIME) - (game_time / (10 * 60 * 60 * 24))) > 31) and (global_var(GVAR_SLAUGHTER_SLAG_TIME) > 0)))) then begin
+            if (CUR_MAP_GSTFARM) then begin
+                if (((global_var(GVAR_MODOC_GENERIC_FLAG_2) bwand 1024) != 0) and (global_var(GVAR_SLAG_CNT) <= 0) or ((global_var(GVAR_MODOC_GHOST_FARM_SEED) == 2) or (((global_var(GVAR_SLAUGHTER_SLAG_TIME) - (game_time / (10 * 60 * 60 * 24))) <= 0) or ((global_var(GVAR_SLAUGHTER_SLAG_TIME) - (game_time / (10 * 60 * 60 * 24))) > 31) and (global_var(GVAR_SLAUGHTER_SLAG_TIME) > 0)))) then begin
                     debug_msg("KILLING SLAG IN FARM");
-                    set_global_var(310, global_var(310) - 1);
-                    if (((global_var(402) bwand 1024) != 0) and (global_var(310) <= 0) or ((global_var(112) == 2) or (((global_var(403) - (game_time / (10 * 60 * 60 * 24))) <= 0) or ((global_var(403) - (game_time / (10 * 60 * 60 * 24))) > 31) and (global_var(403) > 0)))) then begin
-                        if (global_var(112) != 0) then begin
-                            if (global_var(111) >= 4) then begin
-                                if (global_var(631) < 3) then begin
-                                    set_global_var(631, 3);
+                    set_global_var(GVAR_SLAG_CNT, global_var(GVAR_SLAG_CNT) - 1);
+                    if (((global_var(GVAR_MODOC_GENERIC_FLAG_2) bwand 1024) != 0) and (global_var(GVAR_SLAG_CNT) <= 0) or ((global_var(GVAR_MODOC_GHOST_FARM_SEED) == 2) or (((global_var(GVAR_SLAUGHTER_SLAG_TIME) - (game_time / (10 * 60 * 60 * 24))) <= 0) or ((global_var(GVAR_SLAUGHTER_SLAG_TIME) - (game_time / (10 * 60 * 60 * 24))) > 31) and (global_var(GVAR_SLAUGHTER_SLAG_TIME) > 0)))) then begin
+                        if (global_var(GVAR_MODOC_GHOST_FARM_SEED) != 0) then begin
+                            if (global_var(GVAR_MODOC_VERMIN_HUNTER_SEED) >= 4) then begin
+                                if (global_var(GVAR_MODOC_GHOST_SEED_PIP) < 3) then begin
+                                    set_global_var(GVAR_MODOC_GHOST_SEED_PIP, 3);
                                 end
-                                debug_msg("ghost pip == " + global_var(631));
+                                debug_msg("ghost pip == " + global_var(GVAR_MODOC_GHOST_SEED_PIP));
                             end
-                            if (global_var(388) < 2) then begin
-                                set_global_var(388, 2);
+                            if (global_var(GVAR_MODOC_ENDINGS) < 2) then begin
+                                set_global_var(GVAR_MODOC_ENDINGS, 2);
                             end
                             debug_msg("set_modoc_ending_wipeout_1");
                         end
                         else begin
-                            if (global_var(111) >= 4) then begin
-                                if (global_var(631) < 3) then begin
-                                    set_global_var(631, 3);
+                            if (global_var(GVAR_MODOC_VERMIN_HUNTER_SEED) >= 4) then begin
+                                if (global_var(GVAR_MODOC_GHOST_SEED_PIP) < 3) then begin
+                                    set_global_var(GVAR_MODOC_GHOST_SEED_PIP, 3);
                                 end
-                                debug_msg("ghost pip == " + global_var(631));
+                                debug_msg("ghost pip == " + global_var(GVAR_MODOC_GHOST_SEED_PIP));
                             end
-                            if (global_var(388) < 3) then begin
-                                set_global_var(388, 3);
+                            if (global_var(GVAR_MODOC_ENDINGS) < 3) then begin
+                                set_global_var(GVAR_MODOC_ENDINGS, 3);
                             end
                             debug_msg("set_modoc_ending_wipeout_2");
                         end
                     end
-                    debug_msg("current slag count == " + global_var(310));
+                    debug_msg("current slag count == " + global_var(GVAR_SLAG_CNT));
                     kill_critter(self_obj, random(62, 63));
                 end
             end
             else begin
                 debug_msg("DESTROYING SLAG");
-                set_global_var(310, global_var(310) - 1);
-                if (((global_var(402) bwand 1024) != 0) and (global_var(310) <= 0) or ((global_var(112) == 2) or (((global_var(403) - (game_time / (10 * 60 * 60 * 24))) <= 0) or ((global_var(403) - (game_time / (10 * 60 * 60 * 24))) > 31) and (global_var(403) > 0)))) then begin
-                    if (global_var(112) != 0) then begin
-                        if (global_var(111) >= 4) then begin
-                            if (global_var(631) < 3) then begin
-                                set_global_var(631, 3);
+                set_global_var(GVAR_SLAG_CNT, global_var(GVAR_SLAG_CNT) - 1);
+                if (((global_var(GVAR_MODOC_GENERIC_FLAG_2) bwand 1024) != 0) and (global_var(GVAR_SLAG_CNT) <= 0) or ((global_var(GVAR_MODOC_GHOST_FARM_SEED) == 2) or (((global_var(GVAR_SLAUGHTER_SLAG_TIME) - (game_time / (10 * 60 * 60 * 24))) <= 0) or ((global_var(GVAR_SLAUGHTER_SLAG_TIME) - (game_time / (10 * 60 * 60 * 24))) > 31) and (global_var(GVAR_SLAUGHTER_SLAG_TIME) > 0)))) then begin
+                    if (global_var(GVAR_MODOC_GHOST_FARM_SEED) != 0) then begin
+                        if (global_var(GVAR_MODOC_VERMIN_HUNTER_SEED) >= 4) then begin
+                            if (global_var(GVAR_MODOC_GHOST_SEED_PIP) < 3) then begin
+                                set_global_var(GVAR_MODOC_GHOST_SEED_PIP, 3);
                             end
-                            debug_msg("ghost pip == " + global_var(631));
+                            debug_msg("ghost pip == " + global_var(GVAR_MODOC_GHOST_SEED_PIP));
                         end
-                        if (global_var(388) < 2) then begin
-                            set_global_var(388, 2);
+                        if (global_var(GVAR_MODOC_ENDINGS) < 2) then begin
+                            set_global_var(GVAR_MODOC_ENDINGS, 2);
                         end
                         debug_msg("set_modoc_ending_wipeout_1");
                     end
                     else begin
-                        if (global_var(111) >= 4) then begin
-                            if (global_var(631) < 3) then begin
-                                set_global_var(631, 3);
+                        if (global_var(GVAR_MODOC_VERMIN_HUNTER_SEED) >= 4) then begin
+                            if (global_var(GVAR_MODOC_GHOST_SEED_PIP) < 3) then begin
+                                set_global_var(GVAR_MODOC_GHOST_SEED_PIP, 3);
                             end
-                            debug_msg("ghost pip == " + global_var(631));
+                            debug_msg("ghost pip == " + global_var(GVAR_MODOC_GHOST_SEED_PIP));
                         end
-                        if (global_var(388) < 3) then begin
-                            set_global_var(388, 3);
+                        if (global_var(GVAR_MODOC_ENDINGS) < 3) then begin
+                            set_global_var(GVAR_MODOC_ENDINGS, 3);
                         end
                         debug_msg("set_modoc_ending_wipeout_2");
                     end
                 end
-                debug_msg("current slag count == " + global_var(310));
+                debug_msg("current slag count == " + global_var(GVAR_SLAG_CNT));
                 destroy_object(self_obj);
             end
         end
-        if (metarule(14, 0)) then begin
-            set_local_var(3, (global_var(297) bwand 268435456) != 0);
-            if (cur_map_index == 24) then begin
-                set_global_var(310, global_var(310) + 1);
+        if (map_first_run) then begin
+            set_local_var(3, (global_var(GVAR_MODOC_GENERIC_FLAG_1) bwand 268435456) != 0);
+            if (CUR_MAP_GSTFARM) then begin
+                set_global_var(GVAR_SLAG_CNT, global_var(GVAR_SLAG_CNT) + 1);
             end
-            debug_msg("current slag count == " + global_var(310));
-            set_global_var(297, global_var(297) bwor 268435456);
+            debug_msg("current slag count == " + global_var(GVAR_SLAG_CNT));
+            set_global_var(GVAR_MODOC_GENERIC_FLAG_1, global_var(GVAR_MODOC_GENERIC_FLAG_1) bwor 268435456);
             set_local_var(1, tile_num(self_obj));
-            set_local_var(2, has_trait(1, self_obj, 10));
-            critter_add_trait(self_obj, 1, 6, 29);
-            critter_add_trait(self_obj, 1, 5, 22);
+            set_local_var(2, obj_get_rot(self_obj));
+            critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 29);
+            critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_TOUGH_GUARD);
         end
-        if ((global_var(27) == 777) and (((local_var(0) bwand 2) != 0) == 0) and (((global_var(297) bwand 512) != 0) == 0)) then begin
-            metarule3(100, self_obj, 777, 0);
+        if ((global_var(GVAR_LOAD_MAP_INDEX) == 777) and (((local_var(0) bwand 2) != 0) == 0) and (((global_var(GVAR_MODOC_GENERIC_FLAG_1) bwand 512) != 0) == 0)) then begin
+            rm_fixed_timer_event(self_obj, 777);
             add_timer_event(self_obj, game_ticks(0), 777);
         end
     end
@@ -486,7 +486,7 @@ end
 
 procedure map_update_p_proc
 begin
-    if (local_var(2) != has_trait(1, self_obj, 10)) then begin
+    if (local_var(2) != obj_get_rot(self_obj)) then begin
         if (anim_busy(self_obj) == 0) then begin
             anim(self_obj, 1000, local_var(2));
         end
@@ -500,18 +500,18 @@ end
 procedure Node002
 begin
     if (local_var(4) == 0) then begin
-        float_msg(self_obj, message_str(819, 200 + local_var(3)), 8);
+        float_msg(self_obj, mstr(200 + local_var(3)), FLOAT_MSG_YELLOW);
     end
     else begin
         if (local_var(4) == 1) then begin
-            float_msg(self_obj, message_str(819, 210 + local_var(3)), 8);
+            float_msg(self_obj, mstr(210 + local_var(3)), FLOAT_MSG_YELLOW);
         end
         else begin
             if (local_var(4) == 2) then begin
-                float_msg(self_obj, message_str(819, 212 + local_var(3)), 8);
+                float_msg(self_obj, mstr(212 + local_var(3)), FLOAT_MSG_YELLOW);
             end
             else begin
-                float_msg(self_obj, message_str(819, 214 + local_var(3)), 8);
+                float_msg(self_obj, mstr(214 + local_var(3)), FLOAT_MSG_YELLOW);
                 if (local_var(3) == 1) then begin
                     if (random(0, 1) == 1) then begin
                         reg_anim_animate(self_obj, 48, -1);
@@ -528,40 +528,40 @@ end
 
 procedure Node003
 begin
-    gsay_reply(819, 220);
-    giq_option(4, 819, 221, Node004, 51);
-    giq_option(4, 819, 222, Node005, 50);
-    giq_option(-3, 819, 223, Node006, 50);
+    Reply(220);
+    giq_option(4, NAME, 221, Node004, BAD_REACTION);
+    giq_option(4, NAME, 222, Node005, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 223, Node006, NEUTRAL_REACTION);
 end
 
 procedure Node004
 begin
-    gsay_reply(819, 224);
-    giq_option(4, 819, 226, Node990, 50);
-    giq_option(4, 819, 225, Node998, 51);
+    Reply(224);
+    giq_option(4, NAME, 226, Node990, NEUTRAL_REACTION);
+    giq_option(4, NAME, 225, Node998, BAD_REACTION);
 end
 
 procedure Node005
 begin
-    gsay_reply(819, 230);
-    giq_option(4, 819, 231, Node990, 50);
-    giq_option(4, 819, 232, Node004, 51);
+    Reply(230);
+    giq_option(4, NAME, 231, Node990, NEUTRAL_REACTION);
+    giq_option(4, NAME, 232, Node004, BAD_REACTION);
 end
 
 procedure Node006
 begin
-    gsay_reply(819, 240);
-    giq_option(1, 819, 241, Node990, 50);
-    giq_option(1, 819, 242, Node998, 51);
+    Reply(240);
+    giq_option(1, NAME, 241, Node990, NEUTRAL_REACTION);
+    giq_option(1, NAME, 242, Node998, BAD_REACTION);
 end
 
 procedure Node007
 begin
     if (local_var(5) >= 3) then begin
-        float_msg(self_obj, message_str(819, 253), 8);
+        float_msg(self_obj, mstr(253), FLOAT_MSG_YELLOW);
     end
     else begin
-        float_msg(self_obj, message_str(819, 250 + local_var(5)), 8);
+        float_msg(self_obj, mstr(250 + local_var(5)), FLOAT_MSG_YELLOW);
     end
     set_local_var(5, local_var(5) + 1);
 end
@@ -578,7 +578,7 @@ end
 procedure timed_event_p_proc
 begin
     if (fixed_param == 777) then begin
-        if ((combat_is_initialized == 0) and ((global_var(402) bwand 8) != 0)) then begin
+        if ((combat_is_initialized == 0) and ((global_var(GVAR_MODOC_GENERIC_FLAG_2) bwand 8) != 0)) then begin
             call Node002();
         end
     end
@@ -591,133 +591,132 @@ procedure Node990
 begin
     gfade_out(5);
     animate_stand_obj(dude_obj);
-    move_to(dude_obj, 25702, elevation(dude_obj));
-    critter_attempt_placement(dude_obj, 25702, elevation(dude_obj));
+    move_to(dude_obj, 25702, dude_elevation);
+    critter_attempt_placement(dude_obj, 25702, dude_elevation);
     if (party_member_obj(16777278) != 0) then begin
-        if (has_trait(1, party_member_obj(16777278), 666)) then begin
-            critter_attempt_placement(party_member_obj(16777278), 25702, elevation(dude_obj));
+        if (obj_is_visible_flag(party_member_obj(16777278))) then begin
+            critter_attempt_placement(party_member_obj(16777278), 25702, dude_elevation);
         end
     end
     if (party_member_obj(16777376) != 0) then begin
-        if (has_trait(1, party_member_obj(16777376), 666)) then begin
-            critter_attempt_placement(party_member_obj(16777376), 25702, elevation(dude_obj));
+        if (obj_is_visible_flag(party_member_obj(16777376))) then begin
+            critter_attempt_placement(party_member_obj(16777376), 25702, dude_elevation);
         end
     end
     if (party_member_obj(16777377) != 0) then begin
-        if (has_trait(1, party_member_obj(16777377), 666)) then begin
-            critter_attempt_placement(party_member_obj(16777377), 25702, elevation(dude_obj));
+        if (obj_is_visible_flag(party_member_obj(16777377))) then begin
+            critter_attempt_placement(party_member_obj(16777377), 25702, dude_elevation);
         end
     end
     if (party_member_obj(16777305) != 0) then begin
-        if (has_trait(1, party_member_obj(16777305), 666)) then begin
-            critter_attempt_placement(party_member_obj(16777305), 25702, elevation(dude_obj));
+        if (obj_is_visible_flag(party_member_obj(16777305))) then begin
+            critter_attempt_placement(party_member_obj(16777305), 25702, dude_elevation);
         end
     end
     if (party_member_obj(16777313) != 0) then begin
-        if (has_trait(1, party_member_obj(16777313), 666)) then begin
-            critter_attempt_placement(party_member_obj(16777313), 25702, elevation(dude_obj));
+        if (obj_is_visible_flag(party_member_obj(16777313))) then begin
+            critter_attempt_placement(party_member_obj(16777313), 25702, dude_elevation);
         end
     end
     if (party_member_obj(16777323) != 0) then begin
-        if (has_trait(1, party_member_obj(16777323), 666)) then begin
-            critter_attempt_placement(party_member_obj(16777323), 25702, elevation(dude_obj));
+        if (obj_is_visible_flag(party_member_obj(16777323))) then begin
+            critter_attempt_placement(party_member_obj(16777323), 25702, dude_elevation);
         end
     end
     if (party_member_obj(16777352) != 0) then begin
-        if (has_trait(1, party_member_obj(16777352), 666)) then begin
-            critter_attempt_placement(party_member_obj(16777352), 25702, elevation(dude_obj));
+        if (obj_is_visible_flag(party_member_obj(16777352))) then begin
+            critter_attempt_placement(party_member_obj(16777352), 25702, dude_elevation);
         end
     end
     if (party_member_obj(16777378) != 0) then begin
-        if (has_trait(1, party_member_obj(16777378), 666)) then begin
-            critter_attempt_placement(party_member_obj(16777378), 25702, elevation(dude_obj));
+        if (obj_is_visible_flag(party_member_obj(16777378))) then begin
+            critter_attempt_placement(party_member_obj(16777378), 25702, dude_elevation);
         end
     end
     if (party_member_obj(16777368) != 0) then begin
-        if (has_trait(1, party_member_obj(16777368), 666)) then begin
-            critter_attempt_placement(party_member_obj(16777368), 25702, elevation(dude_obj));
+        if (obj_is_visible_flag(party_member_obj(16777368))) then begin
+            critter_attempt_placement(party_member_obj(16777368), 25702, dude_elevation);
         end
     end
     if (party_member_obj(16777379) != 0) then begin
-        if (has_trait(1, party_member_obj(16777379), 666)) then begin
-            critter_attempt_placement(party_member_obj(16777379), 25702, elevation(dude_obj));
+        if (obj_is_visible_flag(party_member_obj(16777379))) then begin
+            critter_attempt_placement(party_member_obj(16777379), 25702, dude_elevation);
         end
     end
     if (party_member_obj(16777380) != 0) then begin
-        if (has_trait(1, party_member_obj(16777380), 666)) then begin
-            critter_attempt_placement(party_member_obj(16777380), 25702, elevation(dude_obj));
+        if (obj_is_visible_flag(party_member_obj(16777380))) then begin
+            critter_attempt_placement(party_member_obj(16777380), 25702, dude_elevation);
         end
     end
     if (party_member_obj(16777295) != 0) then begin
-        if (has_trait(1, party_member_obj(16777295), 666)) then begin
-            critter_attempt_placement(party_member_obj(16777295), 25702, elevation(dude_obj));
+        if (obj_is_visible_flag(party_member_obj(16777295))) then begin
+            critter_attempt_placement(party_member_obj(16777295), 25702, dude_elevation);
         end
     end
     if (party_member_obj(455) != 0) then begin
-        if (has_trait(1, party_member_obj(455), 666)) then begin
-            critter_attempt_placement(party_member_obj(455), 25702, elevation(dude_obj));
+        if (obj_is_visible_flag(party_member_obj(455))) then begin
+            critter_attempt_placement(party_member_obj(455), 25702, dude_elevation);
         end
     end
     if (party_member_obj(16777558) != 0) then begin
-        if (has_trait(1, party_member_obj(16777558), 666)) then begin
-            critter_attempt_placement(party_member_obj(16777558), 25702, elevation(dude_obj));
+        if (obj_is_visible_flag(party_member_obj(16777558))) then begin
+            critter_attempt_placement(party_member_obj(16777558), 25702, dude_elevation);
         end
     end
     if (party_member_obj(16777600) != 0) then begin
-        if (has_trait(1, party_member_obj(16777600), 666)) then begin
-            critter_attempt_placement(party_member_obj(16777600), 25702, elevation(dude_obj));
+        if (obj_is_visible_flag(party_member_obj(16777600))) then begin
+            critter_attempt_placement(party_member_obj(16777600), 25702, dude_elevation);
         end
     end
     if (party_member_obj(16777687) != 0) then begin
-        if (has_trait(1, party_member_obj(16777687), 666)) then begin
-            critter_attempt_placement(party_member_obj(16777687), 25702, elevation(dude_obj));
+        if (obj_is_visible_flag(party_member_obj(16777687))) then begin
+            critter_attempt_placement(party_member_obj(16777687), 25702, dude_elevation);
         end
     end
     if (party_member_obj(16777597) != 0) then begin
-        if (has_trait(1, party_member_obj(16777597), 666)) then begin
-            critter_attempt_placement(party_member_obj(16777597), 25702, elevation(dude_obj));
+        if (obj_is_visible_flag(party_member_obj(16777597))) then begin
+            critter_attempt_placement(party_member_obj(16777597), 25702, dude_elevation);
         end
     end
     if (party_member_obj(16777596) != 0) then begin
-        if (has_trait(1, party_member_obj(16777596), 666)) then begin
-            critter_attempt_placement(party_member_obj(16777596), 25702, elevation(dude_obj));
+        if (obj_is_visible_flag(party_member_obj(16777596))) then begin
+            critter_attempt_placement(party_member_obj(16777596), 25702, dude_elevation);
         end
     end
     if (party_member_obj(16777595) != 0) then begin
-        if (has_trait(1, party_member_obj(16777595), 666)) then begin
-            critter_attempt_placement(party_member_obj(16777595), 25702, elevation(dude_obj));
+        if (obj_is_visible_flag(party_member_obj(16777595))) then begin
+            critter_attempt_placement(party_member_obj(16777595), 25702, dude_elevation);
         end
     end
     if (party_member_obj(16777381) != 0) then begin
-        if (has_trait(1, party_member_obj(16777381), 666)) then begin
-            critter_attempt_placement(party_member_obj(16777381), 25702, elevation(dude_obj));
+        if (obj_is_visible_flag(party_member_obj(16777381))) then begin
+            critter_attempt_placement(party_member_obj(16777381), 25702, dude_elevation);
         end
     end
     if (party_member_obj(16777407) != 0) then begin
-        if (has_trait(1, party_member_obj(16777407), 666)) then begin
-            critter_attempt_placement(party_member_obj(16777407), 25702, elevation(dude_obj));
+        if (obj_is_visible_flag(party_member_obj(16777407))) then begin
+            critter_attempt_placement(party_member_obj(16777407), 25702, dude_elevation);
         end
     end
     if (party_member_obj(16777411) != 0) then begin
-        if (has_trait(1, party_member_obj(16777411), 666)) then begin
-            critter_attempt_placement(party_member_obj(16777411), 25702, elevation(dude_obj));
+        if (obj_is_visible_flag(party_member_obj(16777411))) then begin
+            critter_attempt_placement(party_member_obj(16777411), 25702, dude_elevation);
         end
     end
     if (party_member_obj(16777412) != 0) then begin
-        if (has_trait(1, party_member_obj(16777412), 666)) then begin
-            critter_attempt_placement(party_member_obj(16777412), 25702, elevation(dude_obj));
+        if (obj_is_visible_flag(party_member_obj(16777412))) then begin
+            critter_attempt_placement(party_member_obj(16777412), 25702, dude_elevation);
         end
     end
     if (party_member_obj(16777413) != 0) then begin
-        if (has_trait(1, party_member_obj(16777413), 666)) then begin
-            critter_attempt_placement(party_member_obj(16777413), 25702, elevation(dude_obj));
+        if (obj_is_visible_flag(party_member_obj(16777413))) then begin
+            critter_attempt_placement(party_member_obj(16777413), 25702, dude_elevation);
         end
     end
     if (party_member_obj(16777481) != 0) then begin
-        if (has_trait(1, party_member_obj(16777481), 666)) then begin
-            critter_attempt_placement(party_member_obj(16777481), 25702, elevation(dude_obj));
+        if (obj_is_visible_flag(party_member_obj(16777481))) then begin
+            critter_attempt_placement(party_member_obj(16777481), 25702, dude_elevation);
         end
     end
-    set_global_var(297, global_var(297) bwor 67108864);
+    set_global_var(GVAR_MODOC_GENERIC_FLAG_1, global_var(GVAR_MODOC_GENERIC_FLAG_1) bwor 67108864);
 end
-
