diff --git a/Mods/MegaMod/Original/MODOCT43.ssl b/Mods/MegaMod/ReDefine/MODOCT43.ssl
index a29d5d6..ae82f20 100644
--- a/Mods/MegaMod/Original/MODOCT43.ssl
+++ b/Mods/MegaMod/ReDefine/MODOCT43.ssl
@@ -157,10 +157,10 @@ procedure critter_p_proc
 begin
     if (tile_distance_objs(self_obj, dude_obj) > 10) then begin
         if (tile_distance_objs(self_obj, dude_obj) > (10 * 2)) then begin
-            animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(dude_obj), (has_trait(1, dude_obj, 10) + 3) % 6, 5), 1);
+            animate_move_obj_to_tile(self_obj, tile_num_in_direction(dude_tile, (obj_get_rot(dude_obj) + 3) % 6, 5), 1);
         end
         else begin
-            animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(dude_obj), (has_trait(1, dude_obj, 10) + 3) % 6, 5), 0);
+            animate_move_obj_to_tile(self_obj, tile_num_in_direction(dude_tile, (obj_get_rot(dude_obj) + 3) % 6, 5), 0);
         end
     end
 end
@@ -173,17 +173,17 @@ procedure look_at_p_proc
 begin
     script_overrides;
     if (local_var(4) == 0) then begin
-        display_msg(message_str(1572, 100));
+        display_mstr(100);
     end
     else begin
-        display_msg(message_str(1572, 101));
+        display_mstr(101);
     end
 end
 
 procedure description_p_proc
 begin
     script_overrides;
-    display_msg(message_str(1572, 102));
+    display_mstr(102);
 end
 
 procedure use_skill_on_p_proc
@@ -200,7 +200,7 @@ end
 
 procedure map_enter_p_proc
 begin
-    critter_add_trait(self_obj, 1, 6, 0);
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 0);
 end
 
 procedure map_update_p_proc
@@ -210,14 +210,13 @@ end
 procedure talk_p_proc
 begin
     if (map_var(0) == 0) then begin
-        float_msg(self_obj, message_str(1572, 103), 2);
+        float_msg(self_obj, mstr(103), FLOAT_MSG_RED);
     end
     else begin
-        float_msg(self_obj, message_str(1572, 104), 2);
+        float_msg(self_obj, mstr(104), FLOAT_MSG_RED);
     end
 end
 
 procedure timed_event_p_proc
 begin
 end
-
