diff --git a/Mods/MegaMod/Original/MODOCT29.ssl b/Mods/MegaMod/ReDefine/MODOCT29.ssl
index 894ff69..e3febb3 100644
--- a/Mods/MegaMod/Original/MODOCT29.ssl
+++ b/Mods/MegaMod/ReDefine/MODOCT29.ssl
@@ -162,17 +162,17 @@ procedure look_at_p_proc
 begin
     script_overrides;
     if (local_var(4) == 0) then begin
-        display_msg(message_str(1785, 100));
+        display_msg(message_str(SCRIPT_WOLF1, 100));
     end
     else begin
-        display_msg(message_str(1785, 101));
+        display_msg(message_str(SCRIPT_WOLF1, 101));
     end
 end
 
 procedure description_p_proc
 begin
     script_overrides;
-    display_msg(message_str(1785, 102));
+    display_msg(message_str(SCRIPT_WOLF1, 102));
 end
 
 procedure use_skill_on_p_proc
@@ -197,7 +197,7 @@ end
 
 procedure talk_p_proc
 begin
-    if (global_var(999) == 1) then begin
+    if (global_var(GVAR_ALLOWED_TO_SPEAK_TO_JACK) == 1) then begin
         start_gdialog(1785, self_obj, 4, -1, -1);
         gsay_start;
         call Node001();
@@ -205,52 +205,52 @@ begin
         end_dialogue;
     end
     else begin
-        float_msg(self_obj, message_str(1785, random(103, 107)), 0);
+        float_msg(self_obj, message_str(SCRIPT_WOLF1, random(103, 107)), FLOAT_MSG_NORMAL);
     end
 end
 
 procedure Node001
 begin
-    gsay_reply(1785, 108);
-    giq_option(4, 1785, 109, Node002, 50);
+    gsay_reply(SCRIPT_WOLF1, 108);
+    giq_option(4, SCRIPT_WOLF1, 109, Node002, NEUTRAL_REACTION);
 end
 
 procedure Node002
 begin
-    gsay_reply(1785, 110);
-    giq_option(4, 1785, 111, Node003, 50);
-    giq_option(4, 1785, 112, Node004, 50);
-    giq_option(4, 1785, 113, Node005, 50);
+    gsay_reply(SCRIPT_WOLF1, 110);
+    giq_option(4, SCRIPT_WOLF1, 111, Node003, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_WOLF1, 112, Node004, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_WOLF1, 113, Node005, NEUTRAL_REACTION);
 end
 
 procedure Node003
 begin
-    gsay_reply(1785, 114);
-    giq_option(4, 1785, 115, Node006, 50);
+    gsay_reply(SCRIPT_WOLF1, 114);
+    giq_option(4, SCRIPT_WOLF1, 115, Node006, NEUTRAL_REACTION);
 end
 
 procedure Node004
 begin
-    gsay_reply(1785, 118);
-    giq_option(4, 1785, 119, Node007, 50);
+    gsay_reply(SCRIPT_WOLF1, 118);
+    giq_option(4, SCRIPT_WOLF1, 119, Node007, NEUTRAL_REACTION);
 end
 
 procedure Node005
 begin
-    gsay_reply(1785, 122);
-    giq_option(4, 1785, 123, Node008, 50);
+    gsay_reply(SCRIPT_WOLF1, 122);
+    giq_option(4, SCRIPT_WOLF1, 123, Node008, NEUTRAL_REACTION);
 end
 
 procedure Node006
 begin
-    gsay_reply(1785, 116);
-    giq_option(4, 1785, 117, Node008, 50);
+    gsay_reply(SCRIPT_WOLF1, 116);
+    giq_option(4, SCRIPT_WOLF1, 117, Node008, NEUTRAL_REACTION);
 end
 
 procedure Node007
 begin
-    gsay_reply(1785, 120);
-    giq_option(4, 1785, 121, Node008, 50);
+    gsay_reply(SCRIPT_WOLF1, 120);
+    giq_option(4, SCRIPT_WOLF1, 121, Node008, NEUTRAL_REACTION);
 end
 
 procedure Node008
@@ -260,10 +260,10 @@ begin
     game_ui_disable;
     gfade_out(10 * 6);
     move_to(self_obj, local_var(7), 1);
-    create_object_sid(LVar0, local_var(7), 0, -1);
+    create_object(LVar0, local_var(7), 0);
     game_ui_enable;
     gfade_in(10 * 6);
-    set_global_var(999, 2);
+    set_global_var(GVAR_ALLOWED_TO_SPEAK_TO_JACK, 2);
     return 0;
     LVar0 := 0;
 end
@@ -271,4 +271,3 @@ end
 procedure timed_event_p_proc
 begin
 end
-
