diff --git a/Mods/MegaMod/Original/MODOCT37.ssl b/Mods/MegaMod/ReDefine/MODOCT37.ssl
index bce09b0..8d22f2b 100644
--- a/Mods/MegaMod/Original/MODOCT37.ssl
+++ b/Mods/MegaMod/ReDefine/MODOCT37.ssl
@@ -156,9 +156,9 @@ end
 
 procedure critter_p_proc
 begin
-    if (global_var(1008) == 2) then begin
+    if (global_var(GVAR_LINK_THETOWNS) == 2) then begin
         if (obj_can_see_obj(self_obj, dude_obj)) then begin
-            attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+            attack(dude_obj);
         end
     end
 end
@@ -171,17 +171,17 @@ procedure look_at_p_proc
 begin
     script_overrides;
     if (local_var(4) == 0) then begin
-        display_msg(message_str(1566, 101));
+        display_mstr(101);
     end
     else begin
-        display_msg(message_str(1566, 102));
+        display_mstr(102);
     end
 end
 
 procedure description_p_proc
 begin
     script_overrides;
-    display_msg(message_str(1566, 103));
+    display_mstr(103);
 end
 
 procedure use_skill_on_p_proc
@@ -198,7 +198,7 @@ end
 
 procedure map_enter_p_proc
 begin
-    if (global_var(1008) == 14) then begin
+    if (global_var(GVAR_LINK_THETOWNS) == 14) then begin
         destroy_object(self_obj);
     end
 end
@@ -214,4 +214,3 @@ end
 procedure timed_event_p_proc
 begin
 end
-
