diff --git a/Mods/MegaMod/Original/MODOCT24.ssl b/Mods/MegaMod/ReDefine/MODOCT24.ssl
index 72c2ae3..83b1218 100644
--- a/Mods/MegaMod/Original/MODOCT24.ssl
+++ b/Mods/MegaMod/ReDefine/MODOCT24.ssl
@@ -197,10 +197,10 @@ begin
             if (anim_busy(self_obj) == 0) then begin
                 if ((tile_distance_objs(self_obj, dude_obj) < 8) or obj_can_see_obj(self_obj, dude_obj)) then begin
                     if (anim_busy(self_obj) == 0) then begin
-                        global_temp := rotation_to_tile(tile_num(dude_obj), tile_num(self_obj));
+                        global_temp := rotation_to_tile(dude_tile, tile_num(self_obj));
                         animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(self_obj), global_temp, random(3, 10)), 1);
                         global_temp := (global_temp + 1) % 6;
-                        while ((anim_busy(self_obj) == 0) and (global_temp != rotation_to_tile(tile_num(dude_obj), tile_num(self_obj)))) do begin
+                        while ((anim_busy(self_obj) == 0) and (global_temp != rotation_to_tile(dude_tile, tile_num(self_obj)))) do begin
                             debug_msg("flee loop: rot == " + global_temp);
                             animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(self_obj), global_temp, random(3, 10)), 1);
                             global_temp := (global_temp + 1) % 6;
@@ -210,26 +210,26 @@ begin
             end
         end
         else begin
-            attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+            attack(dude_obj);
         end
     end
-    if ((global_var(996) == 1) or (global_var(996) == 2)) then begin
+    if ((global_var(GVAR_KILL_TREASURE_HUNTER) == 1) or (global_var(GVAR_KILL_TREASURE_HUNTER) == 2)) then begin
         set_local_var(8, 1);
     end
-    if ((global_var(997) == 1) or (global_var(997) == 2) or (global_var(997) == 3)) then begin
+    if ((global_var(GVAR_SELL_METZGER_MUM) == 1) or (global_var(GVAR_SELL_METZGER_MUM) == 2) or (global_var(GVAR_SELL_METZGER_MUM) == 3)) then begin
         set_local_var(8, 1);
     end
-    if ((global_var(998) == 1) or (global_var(998) == 2) or (global_var(998) == 3)) then begin
+    if ((global_var(GVAR_ESCORT_JIMMY_TO_RENO) == 1) or (global_var(GVAR_ESCORT_JIMMY_TO_RENO) == 2) or (global_var(GVAR_ESCORT_JIMMY_TO_RENO) == 3)) then begin
         set_local_var(8, 1);
     end
-    if ((global_var(999) == 1) or (global_var(999) == 2)) then begin
+    if ((global_var(GVAR_ALLOWED_TO_SPEAK_TO_JACK) == 1) or (global_var(GVAR_ALLOWED_TO_SPEAK_TO_JACK) == 2)) then begin
         set_local_var(8, 1);
     end
-    if (global_var(1008) == 1) then begin
+    if (global_var(GVAR_LINK_THETOWNS) == 1) then begin
         animate_move_obj_to_tile(self_obj, 19277, 0);
     end
     if (tile_num(self_obj) == local_var(9)) then begin
-        set_global_var(1008, 2);
+        set_global_var(GVAR_LINK_THETOWNS, 2);
         move_to(self_obj, 11665, 0);
     end
 end
@@ -243,7 +243,7 @@ end
 
 procedure talk_p_proc
 begin
-    if (global_var(1006) == 1) then begin
+    if (global_var(GVAR_LAST_BATTLE_BEGIN) == 1) then begin
         if (local_var(8) == 1) then begin
             start_gdialog(1781, self_obj, 4, -1, -1);
             gsay_start;
@@ -260,8 +260,8 @@ begin
         end
     end
     else begin
-        if (global_var(1006) == 2) then begin
-            if (global_var(1008) == 1) then begin
+        if (global_var(GVAR_LAST_BATTLE_BEGIN) == 2) then begin
+            if (global_var(GVAR_LINK_THETOWNS) == 1) then begin
                 start_gdialog(1781, self_obj, 4, -1, -1);
                 gsay_start;
                 call Node021();
@@ -289,55 +289,55 @@ end
 procedure destroy_p_proc
 begin
     if (source_obj == dude_obj) then begin
-        set_global_var(4, global_var(4) + 1);
-        if (metarule(51, self_obj) == 2) then begin
+        set_global_var(GVAR_BAD_MONSTER, global_var(GVAR_BAD_MONSTER) + 1);
+        if (critter_kill_type(self_obj) == KILL_TYPE_children_kills) then begin
             if (source_obj == dude_obj) then begin
-                set_global_var(2, 0);
-                set_global_var(1, global_var(1) + 1);
-                set_global_var(0, global_var(0) + -15);
+                set_global_var(GVAR_CHAMPION_REPUTATION, 0);
+                set_global_var(GVAR_CHILDKILLER_REPUTATION, global_var(GVAR_CHILDKILLER_REPUTATION) + 1);
+                set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + -15);
                 debug_msg("Player gains " + -15 + " Karma Points.");
-                set_global_var(37, 0);
-                set_global_var(38, 0);
-                set_global_var(39, 0);
-                set_global_var(40, 0);
-                set_global_var(41, 0);
-                set_global_var(42, 0);
-                set_global_var(43, 0);
-                set_global_var(44, 0);
-                set_global_var(45, 0);
-                if (global_var(0) >= 1000) then begin
-                    set_global_var(37, 1);
+                set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+                set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+                set_global_var(GVAR_KARMA_DEFENDER, 0);
+                set_global_var(GVAR_KARMA_WANDERER, 0);
+                set_global_var(GVAR_KARMA_BETRAYER, 0);
+                set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+                set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+                set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+                    set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
                 end
                 else begin
-                    if (global_var(0) >= 750) then begin
-                        set_global_var(38, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
                     end
                     else begin
-                        if (global_var(0) >= 500) then begin
-                            set_global_var(39, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                            set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                         end
                         else begin
-                            if (global_var(0) >= 250) then begin
-                                set_global_var(40, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                                set_global_var(GVAR_KARMA_DEFENDER, 1);
                             end
                             else begin
-                                if (global_var(0) > -250) then begin
-                                    set_global_var(41, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                                    set_global_var(GVAR_KARMA_WANDERER, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -500) then begin
-                                        set_global_var(42, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                        set_global_var(GVAR_KARMA_BETRAYER, 1);
                                     end
                                     else begin
-                                        if (global_var(0) > -750) then begin
-                                            set_global_var(43, 1);
+                                        if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                            set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                         end
                                         else begin
-                                            if (global_var(0) > -1000) then begin
-                                                set_global_var(44, 1);
+                                            if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                                set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                             end
                                             else begin
-                                                set_global_var(45, 1);
+                                                set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                             end
                                         end
                                     end
@@ -346,62 +346,62 @@ begin
                         end
                     end
                 end
-                set_global_var(0, global_var(0) + -8);
+                set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + -8);
                 debug_msg("Added " + -8 + " to Town Rep");
             end
         end
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(5) > (2 * global_var(4))) or (global_var(3) == 1))) then begin
-            set_global_var(3, 1);
-            set_global_var(2, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
+            set_global_var(GVAR_BERSERKER_REPUTATION, 1);
+            set_global_var(GVAR_CHAMPION_REPUTATION, 0);
         end
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(4) > (3 * global_var(5))) or (global_var(2) == 1)) and (global_var(1) == 0)) then begin
-            set_global_var(2, 1);
-            set_global_var(3, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(4) > (3 * global_var(GVAR_GOOD_MONSTER))) or (global_var(GVAR_CHAMPION_REPUTATION) == 1)) and (global_var(GVAR_CHILDKILLER_REPUTATION) == 0)) then begin
+            set_global_var(GVAR_CHAMPION_REPUTATION, 1);
+            set_global_var(GVAR_BERSERKER_REPUTATION, 0);
         end
-        set_global_var(0, global_var(0) + 5);
+        set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + 5);
         debug_msg("Player gains " + 5 + " Karma Points.");
-        set_global_var(37, 0);
-        set_global_var(38, 0);
-        set_global_var(39, 0);
-        set_global_var(40, 0);
-        set_global_var(41, 0);
-        set_global_var(42, 0);
-        set_global_var(43, 0);
-        set_global_var(44, 0);
-        set_global_var(45, 0);
-        if (global_var(0) >= 1000) then begin
-            set_global_var(37, 1);
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+        set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+        set_global_var(GVAR_KARMA_DEFENDER, 0);
+        set_global_var(GVAR_KARMA_WANDERER, 0);
+        set_global_var(GVAR_KARMA_BETRAYER, 0);
+        set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+        set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+            set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
         end
         else begin
-            if (global_var(0) >= 750) then begin
-                set_global_var(38, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
             end
             else begin
-                if (global_var(0) >= 500) then begin
-                    set_global_var(39, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                 end
                 else begin
-                    if (global_var(0) >= 250) then begin
-                        set_global_var(40, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                        set_global_var(GVAR_KARMA_DEFENDER, 1);
                     end
                     else begin
-                        if (global_var(0) > -250) then begin
-                            set_global_var(41, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                            set_global_var(GVAR_KARMA_WANDERER, 1);
                         end
                         else begin
-                            if (global_var(0) > -500) then begin
-                                set_global_var(42, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                set_global_var(GVAR_KARMA_BETRAYER, 1);
                             end
                             else begin
-                                if (global_var(0) > -750) then begin
-                                    set_global_var(43, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -1000) then begin
-                                        set_global_var(44, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                     end
                                     else begin
-                                        set_global_var(45, 1);
+                                        set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                     end
                                 end
                             end
@@ -410,7 +410,7 @@ begin
                 end
             end
         end
-        set_global_var(0, global_var(0) + 2);
+        set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + 2);
         debug_msg("Added " + 2 + " to Town Rep");
     end
 end
@@ -429,7 +429,7 @@ end
 
 procedure damage_p_proc
 begin
-    if (party_member_obj(obj_pid(source_obj)) != 0) then begin
+    if (obj_in_party(source_obj)) then begin
         set_local_var(6, 1);
     end
 end
@@ -475,13 +475,13 @@ end
 
 procedure Node001
 begin
-    gsay_reply(1781, 200);
-    giq_option(4, 1781, 201, Node999, 50);
+    gsay_reply(SCRIPT_RAFAEL, 200);
+    giq_option(4, SCRIPT_RAFAEL, 201, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node003
 begin
-    if (global_var(1002) == 6) then begin
+    if (global_var(GVAR_ENTER_ELDRIDGE_BASEMENT) == 6) then begin
         call Node003b();
     end
     else begin
@@ -491,85 +491,85 @@ end
 
 procedure Node003a
 begin
-    gsay_reply(1781, message_str(1781, 209) + obj_name(dude_obj) + message_str(1781, 210));
-    giq_option(4, 1781, 211, Node004, 50);
-    giq_option(4, 1781, 212, Node005, 50);
+    gsay_reply(SCRIPT_RAFAEL, message_str(SCRIPT_RAFAEL, 209) + dude_name + message_str(SCRIPT_RAFAEL, 210));
+    giq_option(4, SCRIPT_RAFAEL, 211, Node004, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_RAFAEL, 212, Node005, NEUTRAL_REACTION);
 end
 
 procedure Node003b
 begin
-    gsay_reply(1781, 410);
-    giq_option(4, 1781, 411, Node005, 50);
+    gsay_reply(SCRIPT_RAFAEL, 410);
+    giq_option(4, SCRIPT_RAFAEL, 411, Node005, NEUTRAL_REACTION);
 end
 
 procedure Node002
 begin
-    gsay_reply(1781, message_str(1781, 209) + obj_name(dude_obj) + message_str(1781, 220));
-    if (global_var(996) == 2) then begin
-        giq_option(4, 1781, 221, Node011, 50);
+    gsay_reply(SCRIPT_RAFAEL, message_str(SCRIPT_RAFAEL, 209) + dude_name + message_str(SCRIPT_RAFAEL, 220));
+    if (global_var(GVAR_KILL_TREASURE_HUNTER) == 2) then begin
+        giq_option(4, SCRIPT_RAFAEL, 221, Node011, NEUTRAL_REACTION);
     end
-    if (global_var(997) == 3) then begin
-        giq_option(4, 1781, 222, Node015, 50);
+    if (global_var(GVAR_SELL_METZGER_MUM) == 3) then begin
+        giq_option(4, SCRIPT_RAFAEL, 222, Node015, NEUTRAL_REACTION);
     end
-    if (global_var(998) == 3) then begin
-        giq_option(4, 1781, 223, Node016, 50);
+    if (global_var(GVAR_ESCORT_JIMMY_TO_RENO) == 3) then begin
+        giq_option(4, SCRIPT_RAFAEL, 223, Node016, NEUTRAL_REACTION);
     end
-    if (global_var(998) == 2) then begin
-        giq_option(4, 1781, 224, Node018, 50);
+    if (global_var(GVAR_ESCORT_JIMMY_TO_RENO) == 2) then begin
+        giq_option(4, SCRIPT_RAFAEL, 224, Node018, NEUTRAL_REACTION);
     end
-    if (global_var(999) == 3) then begin
-        giq_option(4, 1781, 225, Node019, 50);
+    if (global_var(GVAR_ALLOWED_TO_SPEAK_TO_JACK) == 3) then begin
+        giq_option(4, SCRIPT_RAFAEL, 225, Node019, NEUTRAL_REACTION);
     end
-    if (global_var(999) == 4) then begin
-        giq_option(4, 1781, 409, Node029, 50);
+    if (global_var(GVAR_ALLOWED_TO_SPEAK_TO_JACK) == 4) then begin
+        giq_option(4, SCRIPT_RAFAEL, 409, Node029, NEUTRAL_REACTION);
     end
-    giq_option(4, 1781, 401, Node999, 50);
+    giq_option(4, SCRIPT_RAFAEL, 401, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node004
 begin
-    gsay_reply(1781, 230);
-    giq_option(4, 1781, 231, Node005, 50);
+    gsay_reply(SCRIPT_RAFAEL, 230);
+    giq_option(4, SCRIPT_RAFAEL, 231, Node005, NEUTRAL_REACTION);
 end
 
 procedure Node005
 begin
-    gsay_reply(1781, 240);
-    giq_option(4, 1781, 241, Node006, 50);
-    giq_option(4, 1781, 242, Node007, 50);
+    gsay_reply(SCRIPT_RAFAEL, 240);
+    giq_option(4, SCRIPT_RAFAEL, 241, Node006, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_RAFAEL, 242, Node007, NEUTRAL_REACTION);
 end
 
 procedure Node006
 begin
-    gsay_reply(1781, 250);
-    giq_option(4, 1781, 251, Node007, 50);
+    gsay_reply(SCRIPT_RAFAEL, 250);
+    giq_option(4, SCRIPT_RAFAEL, 251, Node007, NEUTRAL_REACTION);
 end
 
 procedure Node007
 begin
-    gsay_reply(1781, 260);
-    if (not((global_var(445) bwand 4) != 0)) then begin
-        giq_option(4, 1781, 261, Node008, 50);
+    gsay_reply(SCRIPT_RAFAEL, 260);
+    if (not((global_var(GVAR_DEN_FLAG_1) bwand 4) != 0)) then begin
+        giq_option(4, SCRIPT_RAFAEL, 261, Node008, NEUTRAL_REACTION);
     end
-    if (not((global_var(445) bwand 4) != 0)) then begin
-        giq_option(4, 1781, 262, Node007a, 50);
+    if (not((global_var(GVAR_DEN_FLAG_1) bwand 4) != 0)) then begin
+        giq_option(4, SCRIPT_RAFAEL, 262, Node007a, NEUTRAL_REACTION);
     end
-    if ((global_var(445) bwand 4) != 0) then begin
-        giq_option(4, 1781, 407, Node007b, 50);
+    if ((global_var(GVAR_DEN_FLAG_1) bwand 4) != 0) then begin
+        giq_option(4, SCRIPT_RAFAEL, 407, Node007b, NEUTRAL_REACTION);
     end
 end
 
 procedure Node008
 begin
-    gsay_reply(1781, 270);
-    giq_option(4, 1781, 271, Node009, 50);
+    gsay_reply(SCRIPT_RAFAEL, 270);
+    giq_option(4, SCRIPT_RAFAEL, 271, Node009, NEUTRAL_REACTION);
 end
 
 procedure Node009
 begin
-    gsay_reply(1781, 280);
-    giq_option(4, 1781, 281, Node999, 50);
-    restock_obj := create_object_sid(569, 0, 0, -1);
+    gsay_reply(SCRIPT_RAFAEL, 280);
+    giq_option(4, SCRIPT_RAFAEL, 281, Node999, NEUTRAL_REACTION);
+    restock_obj := create_object(569, 0, 0);
     add_mult_objs_to_inven(dude_obj, restock_obj, 1);
 end
 
@@ -585,15 +585,15 @@ end
 
 procedure Node012
 begin
-    gsay_reply(1781, 300);
-    giq_option(4, 1781, 301, Node013, 50);
+    gsay_reply(SCRIPT_RAFAEL, 300);
+    giq_option(4, SCRIPT_RAFAEL, 301, Node013, NEUTRAL_REACTION);
 end
 
 procedure Node013
 begin
-    gsay_reply(1781, 310);
-    giq_option(4, 1781, 311, Node013a, 50);
-    giq_option(4, 1781, 312, Node008, 50);
+    gsay_reply(SCRIPT_RAFAEL, 310);
+    giq_option(4, SCRIPT_RAFAEL, 311, Node013a, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_RAFAEL, 312, Node008, NEUTRAL_REACTION);
 end
 
 procedure Node015
@@ -608,51 +608,51 @@ end
 
 procedure Node016
 begin
-    gsay_reply(1781, 330);
-    giq_option(4, 1781, 331, Node017, 50);
+    gsay_reply(SCRIPT_RAFAEL, 330);
+    giq_option(4, SCRIPT_RAFAEL, 331, Node017, NEUTRAL_REACTION);
 end
 
 procedure Node017
 begin
-    gsay_reply(1781, message_str(1781, 340) + obj_name(dude_obj) + message_str(1781, 341));
-    giq_option(4, 1781, 342, Node009, 50);
+    gsay_reply(SCRIPT_RAFAEL, message_str(SCRIPT_RAFAEL, 340) + dude_name + message_str(SCRIPT_RAFAEL, 341));
+    giq_option(4, SCRIPT_RAFAEL, 342, Node009, NEUTRAL_REACTION);
 end
 
 procedure Node018
 begin
-    gsay_reply(1781, 350);
-    if (not((global_var(445) bwand 1) != 0)) then begin
-        giq_option(4, 1781, 351, Node008, 50);
+    gsay_reply(SCRIPT_RAFAEL, 350);
+    if (not((global_var(GVAR_DEN_FLAG_1) bwand 1) != 0)) then begin
+        giq_option(4, SCRIPT_RAFAEL, 351, Node008, NEUTRAL_REACTION);
     end
-    if (not((global_var(445) bwand 1) != 0)) then begin
-        giq_option(4, 1781, 352, Node018a, 50);
+    if (not((global_var(GVAR_DEN_FLAG_1) bwand 1) != 0)) then begin
+        giq_option(4, SCRIPT_RAFAEL, 352, Node018a, NEUTRAL_REACTION);
     end
-    if ((global_var(445) bwand 1) != 0) then begin
-        giq_option(4, 1781, 404, Node028, 50);
+    if ((global_var(GVAR_DEN_FLAG_1) bwand 1) != 0) then begin
+        giq_option(4, SCRIPT_RAFAEL, 404, Node028, NEUTRAL_REACTION);
     end
-    set_global_var(998, 4);
+    set_global_var(GVAR_ESCORT_JIMMY_TO_RENO, 4);
     give_exp_points(200);
-    display_msg(message_str(14, 100) + 200 + message_str(14, 101));
+    display_msg(g_mstr(100) + 200 + g_mstr(101));
 end
 
 procedure Node019
 begin
-    gsay_message(1781, 360, 50);
+    gsay_message(SCRIPT_RAFAEL, 360, NEUTRAL_REACTION);
     set_local_var(4, 1);
-    set_global_var(1006, 2);
+    set_global_var(GVAR_LAST_BATTLE_BEGIN, 2);
     give_exp_points(200);
-    display_msg(message_str(14, 100) + 200 + message_str(14, 101));
-    set_global_var(999, 6);
+    display_msg(g_mstr(100) + 200 + g_mstr(101));
+    set_global_var(GVAR_ALLOWED_TO_SPEAK_TO_JACK, 6);
 end
 
 procedure Node021
 begin
-    if ((global_var(315) bwand 512) != 0) then begin
+    if ((global_var(GVAR_NEW_RENO_SALVATORE) bwand 512) != 0) then begin
         call Node030();
     end
     else begin
-        if (((global_var(445) bwand 1) != 0) and not((global_var(315) bwand 512) != 0)) then begin
-            float_msg(self_obj, 103, 2);
+        if (((global_var(GVAR_DEN_FLAG_1) bwand 1) != 0) and not((global_var(GVAR_NEW_RENO_SALVATORE) bwand 512) != 0)) then begin
+            float_msg(self_obj, 103, FLOAT_MSG_RED);
         end
         else begin
             call Node021a();
@@ -662,135 +662,134 @@ end
 
 procedure Node021a
 begin
-    gsay_reply(1781, 370);
-    giq_option(4, 1781, 371, Node022, 50);
+    gsay_reply(SCRIPT_RAFAEL, 370);
+    giq_option(4, SCRIPT_RAFAEL, 371, Node022, NEUTRAL_REACTION);
 end
 
 procedure Node022
 begin
-    gsay_reply(1781, 380);
-    giq_option(4, 1781, 381, Node022b, 50);
-    giq_option(4, 1781, 382, Node022a, 50);
+    gsay_reply(SCRIPT_RAFAEL, 380);
+    giq_option(4, SCRIPT_RAFAEL, 381, Node022b, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_RAFAEL, 382, Node022a, NEUTRAL_REACTION);
 end
 
 procedure Node022a
 begin
-    set_global_var(1008, 2);
+    set_global_var(GVAR_LINK_THETOWNS, 2);
     add_timer_event(self_obj, 10, 1);
 end
 
 procedure Node022b
 begin
-    set_global_var(1008, 3);
+    set_global_var(GVAR_LINK_THETOWNS, 3);
     add_timer_event(self_obj, 10, 2);
 end
 
 procedure Node025
 begin
-    gsay_message(1781, 390, 50);
+    gsay_message(SCRIPT_RAFAEL, 390, NEUTRAL_REACTION);
     set_local_var(4, 1);
 end
 
 procedure Node026
 begin
-    gsay_message(1781, 400, 50);
+    gsay_message(SCRIPT_RAFAEL, 400, NEUTRAL_REACTION);
     set_local_var(4, 1);
 end
 
 procedure Node015a
 begin
-    set_global_var(997, 4);
-    if (global_var(998) != 2) then begin
-        set_global_var(998, 1);
+    set_global_var(GVAR_SELL_METZGER_MUM, 4);
+    if (global_var(GVAR_ESCORT_JIMMY_TO_RENO) != 2) then begin
+        set_global_var(GVAR_ESCORT_JIMMY_TO_RENO, 1);
     end
-    restock_obj := create_object_sid(280, 0, 0, -1);
+    restock_obj := create_object(280, 0, 0);
     add_mult_objs_to_inven(dude_obj, restock_obj, 1);
-    restock_obj := create_object_sid(25, 0, 0, -1);
+    restock_obj := create_object(25, 0, 0);
     add_mult_objs_to_inven(dude_obj, restock_obj, 1);
-    restock_obj := create_object_sid(51, 0, 0, -1);
+    restock_obj := create_object(51, 0, 0);
     add_mult_objs_to_inven(dude_obj, restock_obj, 1);
     give_exp_points(100);
-    display_msg(message_str(14, 100) + 100 + message_str(14, 101));
+    display_msg(g_mstr(100) + 100 + g_mstr(101));
 end
 
 procedure Node015b
 begin
-    gsay_reply(1781, 320);
-    giq_option(4, 1781, 321, Node015a, 50);
-    restock_obj := create_object_sid(51, 0, 0, -1);
+    gsay_reply(SCRIPT_RAFAEL, 320);
+    giq_option(4, SCRIPT_RAFAEL, 321, Node015a, NEUTRAL_REACTION);
+    restock_obj := create_object(51, 0, 0);
     add_mult_objs_to_inven(dude_obj, restock_obj, 1);
 end
 
 procedure Node018a
 begin
-    set_global_var(999, 1);
-    if (metarule(17, 10) == 0) then begin
+    set_global_var(GVAR_ALLOWED_TO_SPEAK_TO_JACK, 1);
+    if (town_known(10) == MARK_STATE_UNKNOWN) then begin
         debug_msg("  mark_on_map(" + 10 + ")");
-        mark_area_known(0, 10, 1);
+        mark_area_known(MARK_TYPE_TOWN, 10, MARK_STATE_KNOWN);
     end
 end
 
 procedure Node007a
 begin
-    set_global_var(996, 1);
-    restock_obj := create_object_sid(567, 0, 0, -1);
+    set_global_var(GVAR_KILL_TREASURE_HUNTER, 1);
+    restock_obj := create_object(567, 0, 0);
     add_mult_objs_to_inven(dude_obj, restock_obj, 1);
-    restock_obj := create_object_sid(568, 0, 0, -1);
+    restock_obj := create_object(568, 0, 0);
     add_mult_objs_to_inven(dude_obj, restock_obj, 1);
 end
 
 procedure Node013a
 begin
-    set_global_var(997, 1);
-    restock_obj := create_object_sid(570, 0, 0, -1);
+    set_global_var(GVAR_SELL_METZGER_MUM, 1);
+    restock_obj := create_object(570, 0, 0);
     add_mult_objs_to_inven(dude_obj, restock_obj, 1);
-    restock_obj := create_object_sid(568, 0, 0, -1);
+    restock_obj := create_object(568, 0, 0);
     add_mult_objs_to_inven(dude_obj, restock_obj, 1);
 end
 
 procedure Node027
 begin
-    gsay_reply(1781, 402);
-    giq_option(4, 1781, 403, Node009, 50);
+    gsay_reply(SCRIPT_RAFAEL, 402);
+    giq_option(4, SCRIPT_RAFAEL, 403, Node009, NEUTRAL_REACTION);
 end
 
 procedure Node011a
 begin
-    gsay_reply(1781, 290);
-    giq_option(4, 1781, 291, Node012, 50);
-    set_global_var(996, 3);
+    gsay_reply(SCRIPT_RAFAEL, 290);
+    giq_option(4, SCRIPT_RAFAEL, 291, Node012, NEUTRAL_REACTION);
+    set_global_var(GVAR_KILL_TREASURE_HUNTER, 3);
     give_exp_points(100);
-    display_msg(message_str(14, 100) + 100 + message_str(14, 101));
+    display_msg(g_mstr(100) + 100 + g_mstr(101));
 end
 
 procedure Node028
 begin
-    gsay_reply(1781, 405);
-    giq_option(4, 1781, 406, Node999, 50);
+    gsay_reply(SCRIPT_RAFAEL, 405);
+    giq_option(4, SCRIPT_RAFAEL, 406, Node999, NEUTRAL_REACTION);
     set_local_var(8, 0);
 end
 
 procedure Node007b
 begin
-    gsay_reply(1781, 408);
-    giq_option(4, 1781, 261, Node008, 50);
-    giq_option(4, 1781, 262, Node007a, 50);
+    gsay_reply(SCRIPT_RAFAEL, 408);
+    giq_option(4, SCRIPT_RAFAEL, 261, Node008, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_RAFAEL, 262, Node007a, NEUTRAL_REACTION);
 end
 
 procedure Node029
 begin
     set_local_var(8, 0);
-    gsay_reply(1781, 405);
-    giq_option(4, 1781, 406, Node999, 50);
-    set_global_var(1006, 2);
-    set_global_var(999, 6);
+    gsay_reply(SCRIPT_RAFAEL, 405);
+    giq_option(4, SCRIPT_RAFAEL, 406, Node999, NEUTRAL_REACTION);
+    set_global_var(GVAR_LAST_BATTLE_BEGIN, 2);
+    set_global_var(GVAR_ALLOWED_TO_SPEAK_TO_JACK, 6);
 end
 
 procedure Node030
 begin
-    gsay_reply(1781, 413);
-    giq_option(4, 1781, 406, Node999, 50);
+    gsay_reply(SCRIPT_RAFAEL, 413);
+    giq_option(4, SCRIPT_RAFAEL, 406, Node999, NEUTRAL_REACTION);
     add_timer_event(self_obj, 10, 1);
-    set_global_var(1008, 14);
+    set_global_var(GVAR_LINK_THETOWNS, 14);
 end
-
