diff --git a/Mods/MegaMod/Original/MODOCT18.ssl b/Mods/MegaMod/ReDefine/MODOCT18.ssl
index cea835f..1c0bde5 100644
--- a/Mods/MegaMod/Original/MODOCT18.ssl
+++ b/Mods/MegaMod/ReDefine/MODOCT18.ssl
@@ -141,12 +141,12 @@ end
 procedure use_p_proc
 begin
     script_overrides;
-    if (global_var(994) == 3) then begin
-        display_msg(message_str(1546, 100));
-        float_msg(self_obj, message_str(1546, 101), 2);
+    if (global_var(GVAR_KILL_TUBBY_CHILD) == 3) then begin
+        display_mstr(100);
+        float_msg(self_obj, mstr(101), FLOAT_MSG_RED);
     end
     else begin
-        display_msg(message_str(1546, 102));
+        display_mstr(102);
     end
 end
 
@@ -161,4 +161,3 @@ end
 procedure drop_p_proc
 begin
 end
-
