diff --git a/Mods/MegaMod/Original/MODOCT48.ssl b/Mods/MegaMod/ReDefine/MODOCT48.ssl
index ecd900d..7632b20 100644
--- a/Mods/MegaMod/Original/MODOCT48.ssl
+++ b/Mods/MegaMod/ReDefine/MODOCT48.ssl
@@ -161,17 +161,17 @@ procedure look_at_p_proc
 begin
     script_overrides;
     if (local_var(4) == 0) then begin
-        display_msg(message_str(1577, 100));
+        display_mstr(100);
     end
     else begin
-        display_msg(message_str(1577, 101));
+        display_mstr(101);
     end
 end
 
 procedure description_p_proc
 begin
     script_overrides;
-    display_msg(message_str(1577, 102));
+    display_mstr(102);
 end
 
 procedure use_skill_on_p_proc
@@ -188,7 +188,7 @@ end
 
 procedure map_enter_p_proc
 begin
-    if (global_var(1000) == 1) then begin
+    if (global_var(GVAR_TUBBY_TRIBUTE_MONEY) == 1) then begin
         set_obj_visibility(self_obj, 0);
     end
     else begin
@@ -211,14 +211,14 @@ end
 
 procedure Node001
 begin
-    gsay_reply(1577, 103);
-    giq_option(4, 1577, 104, Node002, 50);
+    Reply(103);
+    giq_option(4, NAME, 104, Node002, NEUTRAL_REACTION);
 end
 
 procedure Node002
 begin
-    gsay_reply(1577, 105);
-    giq_option(4, 1577, 106, Node003, 50);
+    Reply(105);
+    giq_option(4, NAME, 106, Node003, NEUTRAL_REACTION);
 end
 
 procedure Node003
@@ -245,4 +245,3 @@ begin
         load_map(225, 0);
     end
 end
-
