diff --git a/Mods/MegaMod/Original/MODOCT38.ssl b/Mods/MegaMod/ReDefine/MODOCT38.ssl
index 0c8d02e..1a7e550 100644
--- a/Mods/MegaMod/Original/MODOCT38.ssl
+++ b/Mods/MegaMod/ReDefine/MODOCT38.ssl
@@ -157,17 +157,17 @@ procedure look_at_p_proc
 begin
     script_overrides;
     if (local_var(4) == 0) then begin
-        display_msg(message_str(1567, 100));
+        display_mstr(100);
     end
     else begin
-        display_msg(message_str(1567, 101));
+        display_mstr(101);
     end
 end
 
 procedure description_p_proc
 begin
     script_overrides;
-    display_msg(message_str(1567, 102));
+    display_mstr(102);
 end
 
 procedure use_skill_on_p_proc
@@ -184,7 +184,7 @@ end
 
 procedure map_enter_p_proc
 begin
-    if (global_var(1008) == 14) then begin
+    if (global_var(GVAR_LINK_THETOWNS) == 14) then begin
         destroy_object(self_obj);
     end
 end
@@ -200,4 +200,3 @@ end
 procedure timed_event_p_proc
 begin
 end
-
