diff --git a/Mods/MegaMod/Original/MODOCT44.ssl b/Mods/MegaMod/ReDefine/MODOCT44.ssl
index 096db03..2ca7e3f 100644
--- a/Mods/MegaMod/Original/MODOCT44.ssl
+++ b/Mods/MegaMod/ReDefine/MODOCT44.ssl
@@ -171,17 +171,17 @@ procedure look_at_p_proc
 begin
     script_overrides;
     if (local_var(4) == 0) then begin
-        display_msg(message_str(1802, 100));
+        display_msg(message_str(SCRIPT_V14GARD2, 100));
     end
     else begin
-        display_msg(message_str(1802, 101));
+        display_msg(message_str(SCRIPT_V14GARD2, 101));
     end
 end
 
 procedure description_p_proc
 begin
     script_overrides;
-    display_msg(message_str(1802, 102));
+    display_msg(message_str(SCRIPT_V14GARD2, 102));
 end
 
 procedure use_skill_on_p_proc
@@ -194,7 +194,7 @@ end
 
 procedure damage_p_proc
 begin
-    attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+    attack(dude_obj);
 end
 
 procedure map_enter_p_proc
@@ -207,9 +207,9 @@ end
 
 procedure talk_p_proc
 begin
-    if (global_var(1010) == 0) then begin
+    if (global_var(GVAR_ROOM_OPEN) == 0) then begin
         if (local_var(4) == 2) then begin
-            float_msg(self_obj, message_str(1802, 105), 0);
+            float_msg(self_obj, message_str(SCRIPT_V14GARD2, 105), FLOAT_MSG_NORMAL);
         end
         else begin
             start_gdialog(1802, self_obj, 4, -1, -1);
@@ -248,59 +248,59 @@ end
 
 procedure Node001
 begin
-    gsay_reply(1802, 106);
-    giq_option(4, 1802, 107, Node002, 50);
+    gsay_reply(SCRIPT_V14GARD2, 106);
+    giq_option(4, SCRIPT_V14GARD2, 107, Node002, NEUTRAL_REACTION);
 end
 
 procedure Node002
 begin
-    gsay_reply(1802, 108);
-    giq_option(4, 1802, 109, Node003, 50);
-    giq_option(4, 1802, 110, Node003a, 50);
+    gsay_reply(SCRIPT_V14GARD2, 108);
+    giq_option(4, SCRIPT_V14GARD2, 109, Node003, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_V14GARD2, 110, Node003a, NEUTRAL_REACTION);
 end
 
 procedure Node003
 begin
-    gsay_reply(1802, 111);
-    giq_option(4, 1802, 112, Node003a, 50);
+    gsay_reply(SCRIPT_V14GARD2, 111);
+    giq_option(4, SCRIPT_V14GARD2, 112, Node003a, NEUTRAL_REACTION);
 end
 
 procedure Node004
 begin
-    gsay_reply(1802, 113);
-    giq_option(4, 1802, 114, Node005, 50);
+    gsay_reply(SCRIPT_V14GARD2, 113);
+    giq_option(4, SCRIPT_V14GARD2, 114, Node005, NEUTRAL_REACTION);
 end
 
 procedure Node005
 begin
-    gsay_reply(1802, 115);
-    giq_option(4, 1802, 116, Node006, 50);
+    gsay_reply(SCRIPT_V14GARD2, 115);
+    giq_option(4, SCRIPT_V14GARD2, 116, Node006, NEUTRAL_REACTION);
 end
 
 procedure Node006
 begin
-    gsay_reply(1802, 117);
-    giq_option(4, 1802, 118, Node007, 50);
+    gsay_reply(SCRIPT_V14GARD2, 117);
+    giq_option(4, SCRIPT_V14GARD2, 118, Node007, NEUTRAL_REACTION);
 end
 
 procedure Node007
 begin
-    gsay_reply(1802, 119);
-    if (has_skill(dude_obj, 14) >= 100) then begin
-        giq_option(4, 1802, 120, Node008, 50);
+    gsay_reply(SCRIPT_V14GARD2, 119);
+    if (dude_skill(SKILL_SPEECH) >= 100) then begin
+        giq_option(4, SCRIPT_V14GARD2, 120, Node008, NEUTRAL_REACTION);
     end
-    if (has_skill(dude_obj, 14) < 100) then begin
-        giq_option(4, 1802, 121, Node009, 50);
+    if (dude_skill(SKILL_SPEECH) < 100) then begin
+        giq_option(4, SCRIPT_V14GARD2, 121, Node009, NEUTRAL_REACTION);
     end
 end
 
 procedure Node008
 begin
-    gsay_reply(1802, 122);
+    gsay_reply(SCRIPT_V14GARD2, 122);
     if (item_caps_total(dude_obj) > 1500) then begin
-        giq_option(4, 1802, 123, Node008b, 50);
+        giq_option(4, SCRIPT_V14GARD2, 123, Node008b, NEUTRAL_REACTION);
     end
-    giq_option(4, 1802, 124, Node008a, 50);
+    giq_option(4, SCRIPT_V14GARD2, 124, Node008a, NEUTRAL_REACTION);
 end
 
 procedure Node008a
@@ -310,15 +310,15 @@ end
 
 procedure Node008b
 begin
-    gsay_reply(1802, 125);
-    giq_option(4, 1802, 126, Node008c, 50);
+    gsay_reply(SCRIPT_V14GARD2, 125);
+    giq_option(4, SCRIPT_V14GARD2, 126, Node008c, NEUTRAL_REACTION);
     item_caps_adjust(self_obj, 1500);
     item_caps_adjust(dude_obj, -1500);
 end
 
 procedure Node008c
 begin
-    set_global_var(1010, 2);
+    set_global_var(GVAR_ROOM_OPEN, 2);
     gfade_out(10 * 6);
     game_ui_disable;
     move_to(self_obj, 16674, 2);
@@ -328,8 +328,8 @@ end
 
 procedure Node009a
 begin
-    gsay_reply(1802, 125);
-    giq_option(4, 1802, 126, Node008c, 50);
+    gsay_reply(SCRIPT_V14GARD2, 125);
+    giq_option(4, SCRIPT_V14GARD2, 126, Node008c, NEUTRAL_REACTION);
     item_caps_adjust(self_obj, 2000);
     item_caps_adjust(dude_obj, -2000);
 end
@@ -341,46 +341,46 @@ end
 
 procedure Node010
 begin
-    gsay_reply(1802, 130);
-    giq_option(4, 1802, message_str(1802, 131) + obj_name(dude_obj) + message_str(1802, 132), Node011, 50);
+    gsay_reply(SCRIPT_V14GARD2, 130);
+    giq_option(4, SCRIPT_V14GARD2, message_str(SCRIPT_V14GARD2, 131) + dude_name + message_str(SCRIPT_V14GARD2, 132), Node011, NEUTRAL_REACTION);
 end
 
 procedure Node011
 begin
-    gsay_reply(1802, 133);
-    giq_option(4, 1802, 134, Node012, 50);
+    gsay_reply(SCRIPT_V14GARD2, 133);
+    giq_option(4, SCRIPT_V14GARD2, 134, Node012, NEUTRAL_REACTION);
 end
 
 procedure Node012
 begin
-    gsay_reply(1802, 135);
-    if (has_skill(dude_obj, 14) >= 100) then begin
-        giq_option(4, 1802, 136, Node008, 50);
+    gsay_reply(SCRIPT_V14GARD2, 135);
+    if (dude_skill(SKILL_SPEECH) >= 100) then begin
+        giq_option(4, SCRIPT_V14GARD2, 136, Node008, NEUTRAL_REACTION);
     end
-    if (has_skill(dude_obj, 14) < 100) then begin
-        giq_option(4, 1802, 137, Node009, 50);
+    if (dude_skill(SKILL_SPEECH) < 100) then begin
+        giq_option(4, SCRIPT_V14GARD2, 137, Node009, NEUTRAL_REACTION);
     end
 end
 
 procedure Node013
 begin
-    gsay_reply(1802, 138);
-    if ((has_skill(dude_obj, 14) < 100) and (item_caps_total(dude_obj) > 2000)) then begin
-        giq_option(4, 1802, 139, Node009a, 50);
+    gsay_reply(SCRIPT_V14GARD2, 138);
+    if ((dude_skill(SKILL_SPEECH) < 100) and (item_caps_total(dude_obj) > 2000)) then begin
+        giq_option(4, SCRIPT_V14GARD2, 139, Node009a, NEUTRAL_REACTION);
     end
-    if ((has_skill(dude_obj, 14) >= 100) and (item_caps_total(dude_obj) > 1500)) then begin
-        giq_option(4, 1802, 140, Node012, 50);
+    if ((dude_skill(SKILL_SPEECH) >= 100) and (item_caps_total(dude_obj) > 1500)) then begin
+        giq_option(4, SCRIPT_V14GARD2, 140, Node012, NEUTRAL_REACTION);
     end
-    giq_option(4, 1802, 141, Node999, 50);
+    giq_option(4, SCRIPT_V14GARD2, 141, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node009
 begin
-    gsay_reply(1802, 127);
+    gsay_reply(SCRIPT_V14GARD2, 127);
     if (item_caps_total(dude_obj) > 2000) then begin
-        giq_option(4, 1802, 128, Node009a, 50);
+        giq_option(4, SCRIPT_V14GARD2, 128, Node009a, NEUTRAL_REACTION);
     end
-    giq_option(4, 1802, 129, Node008a, 50);
+    giq_option(4, SCRIPT_V14GARD2, 129, Node008a, NEUTRAL_REACTION);
 end
 
 procedure timed_event_p_proc
@@ -399,4 +399,3 @@ begin
         set_local_var(4, 1);
     end
 end
-
