diff --git a/Mods/MegaMod/Original/MODOCT50.ssl b/Mods/MegaMod/ReDefine/MODOCT50.ssl
index 781c96e..bbc073f 100644
--- a/Mods/MegaMod/Original/MODOCT50.ssl
+++ b/Mods/MegaMod/ReDefine/MODOCT50.ssl
@@ -143,13 +143,13 @@ procedure use_skill_on_p_proc
 begin
     variable LVar0 := 0;
     LVar0 := action_being_used;
-    if (global_var(1012) == 0) then begin
+    if (global_var(GVAR_HERBS_FOR_SHAMAN) == 0) then begin
         if (LVar0 == 9) then begin
-            display_msg(message_str(1807, 100));
+            display_msg(message_str(SCRIPT_V14LKR, 100));
         end
     end
     else begin
-        display_msg(message_str(1807, 101));
+        display_msg(message_str(SCRIPT_V14LKR, 101));
     end
     return 0;
     LVar0 := 0;
@@ -157,12 +157,12 @@ end
 
 procedure use_p_proc
 begin
-    if (global_var(1012) == 1) then begin
+    if (global_var(GVAR_HERBS_FOR_SHAMAN) == 1) then begin
         obj_unlock(self_obj);
     end
     else begin
         script_overrides;
-        display_msg(message_str(1807, 100));
+        display_msg(message_str(SCRIPT_V14LKR, 100));
     end
 end
 
@@ -177,4 +177,3 @@ end
 procedure drop_p_proc
 begin
 end
-
