diff --git a/Mods/MegaMod/Original/NCBAND.ssl b/Mods/MegaMod/ReDefine/NCBAND.ssl
index ef6d2c4..e3f6180 100644
--- a/Mods/MegaMod/Original/NCBAND.ssl
+++ b/Mods/MegaMod/ReDefine/NCBAND.ssl
@@ -182,53 +182,53 @@ end
 procedure ChooseFromList
 begin
     variable LVar0 := 0;
-    gsay_reply(1120, stored_msg);
-    if (message_str(353, top_msg - 1) != "") then begin
-        giq_option(1, 1120, message_str(353, 150), CLICK_PREV, 50);
+    Reply(stored_msg);
+    if (message_str(SCRIPT_NEWR1, top_msg - 1) != "") then begin
+        giq_option(1, NAME, message_str(SCRIPT_NEWR1, 150), CLICK_PREV, NEUTRAL_REACTION);
     end
     if (LVar0 == 0) then begin
-        if (message_str(353, top_msg + 0) != "") then begin
-            giq_option(1, 1120, message_str(353, top_msg + 0), CLICK_0, 50);
+        if (message_str(SCRIPT_NEWR1, top_msg + 0) != "") then begin
+            giq_option(1, NAME, message_str(SCRIPT_NEWR1, top_msg + 0), CLICK_0, NEUTRAL_REACTION);
         end
         else begin
             LVar0 := 1;
         end
     end
     if (LVar0 == 0) then begin
-        if (message_str(353, top_msg + 1) != "") then begin
-            giq_option(1, 1120, message_str(353, top_msg + 1), CLICK_1, 50);
+        if (message_str(SCRIPT_NEWR1, top_msg + 1) != "") then begin
+            giq_option(1, NAME, message_str(SCRIPT_NEWR1, top_msg + 1), CLICK_1, NEUTRAL_REACTION);
         end
         else begin
             LVar0 := 1;
         end
     end
     if (LVar0 == 0) then begin
-        if (message_str(353, top_msg + 2) != "") then begin
-            giq_option(1, 1120, message_str(353, top_msg + 2), CLICK_2, 50);
+        if (message_str(SCRIPT_NEWR1, top_msg + 2) != "") then begin
+            giq_option(1, NAME, message_str(SCRIPT_NEWR1, top_msg + 2), CLICK_2, NEUTRAL_REACTION);
         end
         else begin
             LVar0 := 1;
         end
     end
     if (LVar0 == 0) then begin
-        if (message_str(353, top_msg + 3) != "") then begin
-            giq_option(1, 1120, message_str(353, top_msg + 3), CLICK_3, 50);
+        if (message_str(SCRIPT_NEWR1, top_msg + 3) != "") then begin
+            giq_option(1, NAME, message_str(SCRIPT_NEWR1, top_msg + 3), CLICK_3, NEUTRAL_REACTION);
         end
         else begin
             LVar0 := 1;
         end
     end
     if (LVar0 == 0) then begin
-        if (message_str(353, top_msg + 4) != "") then begin
-            giq_option(1, 1120, message_str(353, top_msg + 4), CLICK_4, 50);
+        if (message_str(SCRIPT_NEWR1, top_msg + 4) != "") then begin
+            giq_option(1, NAME, message_str(SCRIPT_NEWR1, top_msg + 4), CLICK_4, NEUTRAL_REACTION);
         end
         else begin
             LVar0 := 1;
         end
     end
     if (LVar0 == 0) then begin
-        if (message_str(353, top_msg + 5) != "") then begin
-            giq_option(1, 1120, message_str(353, 151), CLICK_MORE, 50);
+        if (message_str(SCRIPT_NEWR1, top_msg + 5) != "") then begin
+            giq_option(1, NAME, message_str(SCRIPT_NEWR1, 151), CLICK_MORE, NEUTRAL_REACTION);
         end
     end
 end
@@ -287,10 +287,10 @@ begin
                 if (anim_busy(self_obj) == 0) then begin
                     if ((tile_distance_objs(self_obj, dude_obj) < 8) or obj_can_see_obj(self_obj, dude_obj)) then begin
                         if (anim_busy(self_obj) == 0) then begin
-                            global_temp := rotation_to_tile(tile_num(dude_obj), tile_num(self_obj));
+                            global_temp := rotation_to_tile(dude_tile, tile_num(self_obj));
                             animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(self_obj), global_temp, random(3, 10)), 1);
                             global_temp := (global_temp + 1) % 6;
-                            while ((anim_busy(self_obj) == 0) and (global_temp != rotation_to_tile(tile_num(dude_obj), tile_num(self_obj)))) do begin
+                            while ((anim_busy(self_obj) == 0) and (global_temp != rotation_to_tile(dude_tile, tile_num(self_obj)))) do begin
                                 debug_msg("flee loop: rot == " + global_temp);
                                 animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(self_obj), global_temp, random(3, 10)), 1);
                                 global_temp := (global_temp + 1) % 6;
@@ -300,7 +300,7 @@ begin
                 end
             end
             else begin
-                attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+                attack(dude_obj);
             end
         end
     end
@@ -309,7 +309,7 @@ begin
             animate_move_obj_to_tile(self_obj, local_var(4), 0);
         end
         else begin
-            if (has_trait(1, self_obj, 10) != local_var(5)) then begin
+            if (obj_get_rot(self_obj) != local_var(5)) then begin
                 anim(self_obj, 1000, local_var(5));
             end
         end
@@ -318,7 +318,7 @@ end
 
 procedure pickup_p_proc
 begin
-    if ((source_obj == dude_obj) or (party_member_obj(obj_pid(source_obj)) != 0)) then begin
+    if ((source_obj == dude_obj) or (obj_in_party(source_obj))) then begin
         call Node998();
     end
 end
@@ -345,59 +345,59 @@ end
 procedure destroy_p_proc
 begin
     if (source_obj == dude_obj) then begin
-        set_global_var(5, global_var(5) + 1);
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(4) > (3 * global_var(5))) or (global_var(2) == 1)) and (global_var(1) == 0)) then begin
-            set_global_var(2, 1);
-            set_global_var(3, 0);
+        set_global_var(GVAR_GOOD_MONSTER, global_var(GVAR_GOOD_MONSTER) + 1);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(4) > (3 * global_var(GVAR_GOOD_MONSTER))) or (global_var(GVAR_CHAMPION_REPUTATION) == 1)) and (global_var(GVAR_CHILDKILLER_REPUTATION) == 0)) then begin
+            set_global_var(GVAR_CHAMPION_REPUTATION, 1);
+            set_global_var(GVAR_BERSERKER_REPUTATION, 0);
         end
-        if (metarule(51, self_obj) == 2) then begin
+        if (critter_kill_type(self_obj) == KILL_TYPE_children_kills) then begin
             if (source_obj == dude_obj) then begin
-                set_global_var(2, 0);
-                set_global_var(1, global_var(1) + 1);
-                set_global_var(0, global_var(0) + -15);
+                set_global_var(GVAR_CHAMPION_REPUTATION, 0);
+                set_global_var(GVAR_CHILDKILLER_REPUTATION, global_var(GVAR_CHILDKILLER_REPUTATION) + 1);
+                set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + -15);
                 debug_msg("Player gains " + -15 + " Karma Points.");
-                set_global_var(37, 0);
-                set_global_var(38, 0);
-                set_global_var(39, 0);
-                set_global_var(40, 0);
-                set_global_var(41, 0);
-                set_global_var(42, 0);
-                set_global_var(43, 0);
-                set_global_var(44, 0);
-                set_global_var(45, 0);
-                if (global_var(0) >= 1000) then begin
-                    set_global_var(37, 1);
+                set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+                set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+                set_global_var(GVAR_KARMA_DEFENDER, 0);
+                set_global_var(GVAR_KARMA_WANDERER, 0);
+                set_global_var(GVAR_KARMA_BETRAYER, 0);
+                set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+                set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+                set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+                    set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
                 end
                 else begin
-                    if (global_var(0) >= 750) then begin
-                        set_global_var(38, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
                     end
                     else begin
-                        if (global_var(0) >= 500) then begin
-                            set_global_var(39, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                            set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                         end
                         else begin
-                            if (global_var(0) >= 250) then begin
-                                set_global_var(40, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                                set_global_var(GVAR_KARMA_DEFENDER, 1);
                             end
                             else begin
-                                if (global_var(0) > -250) then begin
-                                    set_global_var(41, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                                    set_global_var(GVAR_KARMA_WANDERER, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -500) then begin
-                                        set_global_var(42, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                        set_global_var(GVAR_KARMA_BETRAYER, 1);
                                     end
                                     else begin
-                                        if (global_var(0) > -750) then begin
-                                            set_global_var(43, 1);
+                                        if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                            set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                         end
                                         else begin
-                                            if (global_var(0) > -1000) then begin
-                                                set_global_var(44, 1);
+                                            if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                                set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                             end
                                             else begin
-                                                set_global_var(45, 1);
+                                                set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                             end
                                         end
                                     end
@@ -406,58 +406,58 @@ begin
                         end
                     end
                 end
-                set_global_var(55, global_var(55) + -8);
+                set_global_var(GVAR_TOWN_REP_NEW_RENO, global_var(GVAR_TOWN_REP_NEW_RENO) + -8);
                 debug_msg("Added " + -8 + " to Town Rep");
             end
         end
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(5) > (2 * global_var(4))) or (global_var(3) == 1))) then begin
-            set_global_var(3, 1);
-            set_global_var(2, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
+            set_global_var(GVAR_BERSERKER_REPUTATION, 1);
+            set_global_var(GVAR_CHAMPION_REPUTATION, 0);
         end
-        set_global_var(0, global_var(0) + -10);
+        set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + -10);
         debug_msg("Player gains " + -10 + " Karma Points.");
-        set_global_var(37, 0);
-        set_global_var(38, 0);
-        set_global_var(39, 0);
-        set_global_var(40, 0);
-        set_global_var(41, 0);
-        set_global_var(42, 0);
-        set_global_var(43, 0);
-        set_global_var(44, 0);
-        set_global_var(45, 0);
-        if (global_var(0) >= 1000) then begin
-            set_global_var(37, 1);
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+        set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+        set_global_var(GVAR_KARMA_DEFENDER, 0);
+        set_global_var(GVAR_KARMA_WANDERER, 0);
+        set_global_var(GVAR_KARMA_BETRAYER, 0);
+        set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+        set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+            set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
         end
         else begin
-            if (global_var(0) >= 750) then begin
-                set_global_var(38, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
             end
             else begin
-                if (global_var(0) >= 500) then begin
-                    set_global_var(39, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                 end
                 else begin
-                    if (global_var(0) >= 250) then begin
-                        set_global_var(40, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                        set_global_var(GVAR_KARMA_DEFENDER, 1);
                     end
                     else begin
-                        if (global_var(0) > -250) then begin
-                            set_global_var(41, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                            set_global_var(GVAR_KARMA_WANDERER, 1);
                         end
                         else begin
-                            if (global_var(0) > -500) then begin
-                                set_global_var(42, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                set_global_var(GVAR_KARMA_BETRAYER, 1);
                             end
                             else begin
-                                if (global_var(0) > -750) then begin
-                                    set_global_var(43, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -1000) then begin
-                                        set_global_var(44, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                     end
                                     else begin
-                                        set_global_var(45, 1);
+                                        set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                     end
                                 end
                             end
@@ -466,10 +466,10 @@ begin
                 end
             end
         end
-        set_global_var(55, global_var(55) + -5);
+        set_global_var(GVAR_TOWN_REP_NEW_RENO, global_var(GVAR_TOWN_REP_NEW_RENO) + -5);
         debug_msg("Added " + -5 + " to Town Rep");
     end
-    set_global_var(495, global_var(495) bwor 16777216);
+    set_global_var(GVAR_NEW_RENO_FLAG_3, global_var(GVAR_NEW_RENO_FLAG_3) bwor 16777216);
     if (local_var(1) == 1) then begin
         trash_can_singer_1_obj := 0;
     end
@@ -488,13 +488,13 @@ end
 procedure look_at_p_proc
 begin
     script_overrides;
-    display_msg(message_str(1120, 100 + local_var(1) - 1));
+    display_mstr(100 + local_var(1) - 1);
 end
 
 procedure description_p_proc
 begin
     script_overrides;
-    display_msg(message_str(1120, 100 + ((local_var(0) bwand 4) != 0)));
+    display_mstr(100 + ((local_var(0) bwand 4) != 0));
     set_local_var(0, local_var(0) bwor 4);
 end
 
@@ -518,10 +518,10 @@ begin
         set_local_var(3, 400);
         last_timed_obj := self_obj;
         last_timed_param := last_timed_param;
-        metarule3(100, self_obj, 100, 0);
+        rm_fixed_timer_event(self_obj, 100);
         add_timer_event(self_obj, game_ticks(6), 100);
     end
-    if ((message_str(403, 100) == "You see one of the Reno townsfolk.") == 0) then begin
+    if ((message_str(SCRIPT_NCPERSON, 100) == "You see one of the Reno townsfolk.") == 0) then begin
         destroy_object(self_obj);
     end
     else begin
@@ -556,16 +556,16 @@ begin
                 end
             end
         end
-        if (metarule(22, 0) == 0) then begin
-            if ((global_var(495) bwand 16777216) != 0) then begin
+        if (is_loading_game == 0) then begin
+            if ((global_var(GVAR_NEW_RENO_FLAG_3) bwand 16777216) != 0) then begin
                 debug_msg("the band's breaking up, the lead is dead");
                 destroy_object(self_obj);
             end
-            if (metarule(14, 0)) then begin
+            if (map_first_run) then begin
                 set_local_var(4, tile_num(self_obj));
-                set_local_var(5, has_trait(1, self_obj, 10));
-                critter_add_trait(self_obj, 1, 6, 204);
-                critter_add_trait(self_obj, 1, 5, 24);
+                set_local_var(5, obj_get_rot(self_obj));
+                critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 204);
+                critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_TOUGH_CITIZEN);
             end
         end
     end
@@ -579,7 +579,7 @@ procedure timed_event_p_proc
 begin
     debug_msg("band timed_event_p_proc: fixed_param == " + fixed_param);
     if (fixed_param == 100) then begin
-        if ((combat_is_initialized == 0) and has_trait(1, self_obj, 666)) then begin
+        if ((combat_is_initialized == 0) and obj_is_visible_flag(self_obj)) then begin
             the_line := local_var(3);
             if ((the_line == 600) or (the_line == 601) or (the_line == 604)) then begin
                 if (the_line == 604) then begin
@@ -603,10 +603,10 @@ begin
                     end
                 end
                 if (last_singer_obj != 0) then begin
-                    float_msg(last_singer_obj, " ", 8);
+                    float_msg(last_singer_obj, " ", FLOAT_MSG_YELLOW);
                 end
                 last_singer_obj := new_singer;
-                float_msg(new_singer, "" + local_var(2) + message_str(1120, the_line), 8);
+                float_msg(new_singer, "" + local_var(2) + mstr(the_line), FLOAT_MSG_YELLOW);
                 if (local_var(3) != 604) then begin
                     set_local_var(3, local_var(3) + 1);
                 end
@@ -666,7 +666,7 @@ begin
                                                                 the_line := 915;
                                                             end
                                                             else begin
-                                                                if (message_str(1120, the_line) == "") then begin
+                                                                if (mstr(the_line) == "") then begin
                                                                     the_line := -1;
                                                                 end
                                                             end
@@ -699,10 +699,10 @@ begin
                         end
                     end
                     if (last_singer_obj != 0) then begin
-                        float_msg(last_singer_obj, " ", 8);
+                        float_msg(last_singer_obj, " ", FLOAT_MSG_YELLOW);
                     end
                     last_singer_obj := new_singer;
-                    float_msg(new_singer, message_str(1120, the_line), 8);
+                    float_msg(new_singer, mstr(the_line), FLOAT_MSG_YELLOW);
                     set_local_var(3, local_var(3) + 1);
                 end
                 else begin
@@ -712,7 +712,7 @@ begin
         end
         last_timed_obj := self_obj;
         last_timed_param := last_timed_param;
-        metarule3(100, self_obj, 100, 0);
+        rm_fixed_timer_event(self_obj, 100);
         add_timer_event(self_obj, game_ticks(3) + random(0, game_ticks(5 - 3)), 100);
     end
     else begin
@@ -720,7 +720,7 @@ begin
             new_track := fixed_param;
         end
         else begin
-            if (global_var(494) == 8) then begin
+            if (global_var(GVAR_FALLOUT_2) == 8) then begin
                 new_track := random(4, 9);
             end
             else begin
@@ -750,10 +750,10 @@ end
 
 procedure map_update_p_proc
 begin
-    if ((global_var(495) bwand 536870912) != 0) then begin
-        if (has_trait(1, self_obj, 666)) then begin
+    if ((global_var(GVAR_NEW_RENO_FLAG_3) bwand 536870912) != 0) then begin
+        if (obj_is_visible_flag(self_obj)) then begin
             if (tile_in_tile_rect(22298, 22266, 29498, 29466, tile_num(self_obj)) == 0) then begin
-                if (has_trait(1, self_obj, 666) == 1) then begin
+                if (obj_is_visible_flag(self_obj) == 1) then begin
                     set_obj_visibility(self_obj, 1);
                 end
                 set_local_var(0, local_var(0) bwor 512);
@@ -762,7 +762,7 @@ begin
     end
     else begin
         if ((local_var(0) bwand 512) != 0) then begin
-            if (has_trait(1, self_obj, 666) == 0) then begin
+            if (obj_is_visible_flag(self_obj) == 0) then begin
                 set_obj_visibility(self_obj, 0);
             end
             set_local_var(0, local_var(0) bwand (-1 - 512));
@@ -781,37 +781,37 @@ end
 
 procedure Node001
 begin
-    if (global_var(494) == 8) then begin
-        gsay_reply(1120, message_str(1120, 200) + obj_name(dude_obj) + message_str(1120, 1200));
-        giq_option(4, 1120, 208, Node988, 50);
+    if (global_var(GVAR_FALLOUT_2) == 8) then begin
+        Reply(mstr(200) + dude_name + mstr(1200));
+        giq_option(4, NAME, 208, Node988, NEUTRAL_REACTION);
     end
     else begin
-        if (global_var(230) > 0) then begin
-            gsay_reply(1120, message_str(1120, 200) + message_str(353, 300 + (global_var(230) % 100)) + message_str(1120, 1201));
+        if (global_var(GVAR_NEW_RENO_MADE_MAN) > 0) then begin
+            Reply(mstr(200) + message_str(SCRIPT_NEWR1, 300 + (global_var(GVAR_NEW_RENO_MADE_MAN) % 100)) + mstr(1201));
         end
         else begin
-            if (global_var(259) > 0) then begin
-                gsay_reply(1120, message_str(1120, 200) + message_str(353, 400 + global_var(231)) + message_str(1120, 1202));
+            if (global_var(GVAR_NEW_RENO_HAS_REP_PRIZEFIGHTER) > 0) then begin
+                Reply(mstr(200) + message_str(SCRIPT_NEWR1, 400 + global_var(GVAR_NEW_RENO_PRIZEFIGHTER)) + mstr(1202));
             end
             else begin
-                gsay_reply(1120, 203 + (get_critter_stat(dude_obj, 34) == 1));
+                Reply(203 + (dude_gender == 1));
             end
         end
         if (item_caps_total(dude_obj) >= 2) then begin
-            giq_option(4, 1120, 209, Node988, 50);
+            giq_option(4, NAME, 209, Node988, NEUTRAL_REACTION);
         end
     end
-    giq_option(-3, 1120, 205, Node990, 50);
-    giq_option(-3, 1120, 206, Node989, 50);
-    giq_option(-3, 1120, 207, Node999, 50);
-    giq_option(4, 1120, 210, Node999, 50);
+    giq_option(-3, NAME, 205, Node990, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 206, Node989, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 207, Node999, NEUTRAL_REACTION);
+    giq_option(4, NAME, 210, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node002
 begin
-    gsay_reply(1120, 300 + (get_critter_stat(dude_obj, 34) == 1));
-    giq_option(-3, 1120, 302, Node999, 50);
-    giq_option(4, 1120, 303, Node999, 50);
+    Reply(300 + (dude_gender == 1));
+    giq_option(-3, NAME, 302, Node999, NEUTRAL_REACTION);
+    giq_option(4, NAME, 303, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node990
@@ -828,10 +828,9 @@ end
 
 procedure Node988
 begin
-    if ((global_var(494) == 8) == 0) then begin
+    if ((global_var(GVAR_FALLOUT_2) == 8) == 0) then begin
         item_caps_adjust(dude_obj, -2);
     end
     add_timer_event(trash_can_singer_1_obj, 0, 0);
     call Node002();
 end
-
