diff --git a/Mods/MegaMod/Original/NCCHEAT.ssl b/Mods/MegaMod/ReDefine/NCCHEAT.ssl
index a88c59d..8d856f1 100644
--- a/Mods/MegaMod/Original/NCCHEAT.ssl
+++ b/Mods/MegaMod/ReDefine/NCCHEAT.ssl
@@ -187,53 +187,53 @@ end
 procedure ChooseFromList
 begin
     variable LVar0 := 0;
-    gsay_reply(424, stored_msg);
-    if (message_str(353, top_msg - 1) != "") then begin
-        giq_option(1, 424, message_str(353, 150), CLICK_PREV, 50);
+    gsay_reply(SCRIPT_NCCODY, stored_msg);
+    if (message_str(SCRIPT_NEWR1, top_msg - 1) != "") then begin
+        giq_option(1, SCRIPT_NCCODY, message_str(SCRIPT_NEWR1, 150), CLICK_PREV, NEUTRAL_REACTION);
     end
     if (LVar0 == 0) then begin
-        if (message_str(353, top_msg + 0) != "") then begin
-            giq_option(1, 424, message_str(353, top_msg + 0), CLICK_0, 50);
+        if (message_str(SCRIPT_NEWR1, top_msg + 0) != "") then begin
+            giq_option(1, SCRIPT_NCCODY, message_str(SCRIPT_NEWR1, top_msg + 0), CLICK_0, NEUTRAL_REACTION);
         end
         else begin
             LVar0 := 1;
         end
     end
     if (LVar0 == 0) then begin
-        if (message_str(353, top_msg + 1) != "") then begin
-            giq_option(1, 424, message_str(353, top_msg + 1), CLICK_1, 50);
+        if (message_str(SCRIPT_NEWR1, top_msg + 1) != "") then begin
+            giq_option(1, SCRIPT_NCCODY, message_str(SCRIPT_NEWR1, top_msg + 1), CLICK_1, NEUTRAL_REACTION);
         end
         else begin
             LVar0 := 1;
         end
     end
     if (LVar0 == 0) then begin
-        if (message_str(353, top_msg + 2) != "") then begin
-            giq_option(1, 424, message_str(353, top_msg + 2), CLICK_2, 50);
+        if (message_str(SCRIPT_NEWR1, top_msg + 2) != "") then begin
+            giq_option(1, SCRIPT_NCCODY, message_str(SCRIPT_NEWR1, top_msg + 2), CLICK_2, NEUTRAL_REACTION);
         end
         else begin
             LVar0 := 1;
         end
     end
     if (LVar0 == 0) then begin
-        if (message_str(353, top_msg + 3) != "") then begin
-            giq_option(1, 424, message_str(353, top_msg + 3), CLICK_3, 50);
+        if (message_str(SCRIPT_NEWR1, top_msg + 3) != "") then begin
+            giq_option(1, SCRIPT_NCCODY, message_str(SCRIPT_NEWR1, top_msg + 3), CLICK_3, NEUTRAL_REACTION);
         end
         else begin
             LVar0 := 1;
         end
     end
     if (LVar0 == 0) then begin
-        if (message_str(353, top_msg + 4) != "") then begin
-            giq_option(1, 424, message_str(353, top_msg + 4), CLICK_4, 50);
+        if (message_str(SCRIPT_NEWR1, top_msg + 4) != "") then begin
+            giq_option(1, SCRIPT_NCCODY, message_str(SCRIPT_NEWR1, top_msg + 4), CLICK_4, NEUTRAL_REACTION);
         end
         else begin
             LVar0 := 1;
         end
     end
     if (LVar0 == 0) then begin
-        if (message_str(353, top_msg + 5) != "") then begin
-            giq_option(1, 424, message_str(353, 151), CLICK_MORE, 50);
+        if (message_str(SCRIPT_NEWR1, top_msg + 5) != "") then begin
+            giq_option(1, SCRIPT_NCCODY, message_str(SCRIPT_NEWR1, 151), CLICK_MORE, NEUTRAL_REACTION);
         end
     end
 end
@@ -345,8 +345,8 @@ end
 procedure map_enter_p_proc
 begin
     destroy_object(self_obj);
-    if (metarule(22, 0) == 0) then begin
-        if (metarule(14, 0)) then begin
+    if (is_loading_game == 0) then begin
+        if (map_first_run) then begin
         end
     end
 end
@@ -365,281 +365,281 @@ end
 
 procedure Node001
 begin
-    gsay_reply(424, "WHAT CAN I CHEAT FOR YOU");
-    if ((global_var(1) >= 2) == 0) then begin
-        giq_option(1, 424, "CHILDKILLER", Node002, 50);
+    gsay_reply(SCRIPT_NCCODY, "WHAT CAN I CHEAT FOR YOU");
+    if ((global_var(GVAR_CHILDKILLER_REPUTATION) >= 2) == 0) then begin
+        giq_option(1, SCRIPT_NCCODY, "CHILDKILLER", Node002, NEUTRAL_REACTION);
     end
-    if ((((global_var(4) + global_var(5)) >= 25) and ((global_var(5) > (2 * global_var(4))) or (global_var(3) == 1))) == 0) then begin
-        giq_option(1, 424, "BESERKER", Node003, 50);
+    if ((((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) == 0) then begin
+        giq_option(1, SCRIPT_NCCODY, "BESERKER", Node003, NEUTRAL_REACTION);
     end
-    if ((global_var(11) == 1) == 0) then begin
-        giq_option(1, 424, "SLAVER", Node004, 50);
+    if ((global_var(GVAR_REPUTATION_SLAVER) == 1) == 0) then begin
+        giq_option(1, SCRIPT_NCCODY, "SLAVER", Node004, NEUTRAL_REACTION);
     end
-    if ((global_var(232) > 0) == 0) then begin
-        giq_option(1, 424, "PORN STAR", Node005, 50);
+    if ((global_var(GVAR_NEW_RENO_PORN_STAR) > 0) == 0) then begin
+        giq_option(1, SCRIPT_NCCODY, "PORN STAR", Node005, NEUTRAL_REACTION);
     end
-    if ((global_var(259) > 0) == 0) then begin
-        giq_option(1, 424, "PRIZEFIGHTER", Node013, 50);
+    if ((global_var(GVAR_NEW_RENO_HAS_REP_PRIZEFIGHTER) > 0) == 0) then begin
+        giq_option(1, SCRIPT_NCCODY, "PRIZEFIGHTER", Node013, NEUTRAL_REACTION);
     end
-    giq_option(1, 424, "MADE MAN", Node006, 50);
-    giq_option(1, 424, "KARMA", Node007, 50);
-    giq_option(1, 424, "FALLOUT 2 FLAGS", Node008, 50);
-    giq_option(1, 424, "EXIT", Node999, 50);
+    giq_option(1, SCRIPT_NCCODY, "MADE MAN", Node006, NEUTRAL_REACTION);
+    giq_option(1, SCRIPT_NCCODY, "KARMA", Node007, NEUTRAL_REACTION);
+    giq_option(1, SCRIPT_NCCODY, "FALLOUT 2 FLAGS", Node008, NEUTRAL_REACTION);
+    giq_option(1, SCRIPT_NCCODY, "EXIT", Node999, NEUTRAL_REACTION);
 end
 
 procedure Node002
 begin
-    set_global_var(1, 2);
-    gsay_reply(424, "CHILDKILLER");
-    giq_option(1, 424, "SOMETHING ELSE", Node001, 50);
-    giq_option(1, 424, "EXIT", Node999, 50);
+    set_global_var(GVAR_CHILDKILLER_REPUTATION, 2);
+    gsay_reply(SCRIPT_NCCODY, "CHILDKILLER");
+    giq_option(1, SCRIPT_NCCODY, "SOMETHING ELSE", Node001, NEUTRAL_REACTION);
+    giq_option(1, SCRIPT_NCCODY, "EXIT", Node999, NEUTRAL_REACTION);
 end
 
 procedure Node003
 begin
-    set_global_var(3, 1);
-    gsay_reply(424, "BESERKER");
-    giq_option(1, 424, "SOMETHING ELSE", Node001, 50);
-    giq_option(1, 424, "EXIT", Node999, 50);
+    set_global_var(GVAR_BERSERKER_REPUTATION, 1);
+    gsay_reply(SCRIPT_NCCODY, "BESERKER");
+    giq_option(1, SCRIPT_NCCODY, "SOMETHING ELSE", Node001, NEUTRAL_REACTION);
+    giq_option(1, SCRIPT_NCCODY, "EXIT", Node999, NEUTRAL_REACTION);
 end
 
 procedure Node004
 begin
-    set_global_var(11, 1);
-    gsay_reply(424, "SLAVER");
-    giq_option(1, 424, "SOMETHING ELSE", Node001, 50);
-    giq_option(1, 424, "EXIT", Node999, 50);
+    set_global_var(GVAR_REPUTATION_SLAVER, 1);
+    gsay_reply(SCRIPT_NCCODY, "SLAVER");
+    giq_option(1, SCRIPT_NCCODY, "SOMETHING ELSE", Node001, NEUTRAL_REACTION);
+    giq_option(1, SCRIPT_NCCODY, "EXIT", Node999, NEUTRAL_REACTION);
 end
 
 procedure Node005
 begin
-    set_global_var(232, random(1, 24));
-    gsay_reply(424, "PORN STAR == " + message_str(353, 500 + global_var(232)));
-    giq_option(1, 424, "SOMETHING ELSE", Node001, 50);
-    giq_option(1, 424, "EXIT", Node999, 50);
+    set_global_var(GVAR_NEW_RENO_PORN_STAR, random(1, 24));
+    gsay_reply(SCRIPT_NCCODY, "PORN STAR == " + message_str(SCRIPT_NEWR1, 500 + global_var(GVAR_NEW_RENO_PORN_STAR)));
+    giq_option(1, SCRIPT_NCCODY, "SOMETHING ELSE", Node001, NEUTRAL_REACTION);
+    giq_option(1, SCRIPT_NCCODY, "EXIT", Node999, NEUTRAL_REACTION);
 end
 
 procedure Node006
 begin
-    gsay_reply(424, "MADE MAN");
-    if ((global_var(230) / 100) != 1) then begin
-        giq_option(1, 424, "SALVATORE", Node009, 50);
+    gsay_reply(SCRIPT_NCCODY, "MADE MAN");
+    if ((global_var(GVAR_NEW_RENO_MADE_MAN) / 100) != 1) then begin
+        giq_option(1, SCRIPT_NCCODY, "SALVATORE", Node009, NEUTRAL_REACTION);
     end
-    if ((global_var(230) / 100) != 2) then begin
-        giq_option(1, 424, "BISHOP", Node010, 50);
+    if ((global_var(GVAR_NEW_RENO_MADE_MAN) / 100) != 2) then begin
+        giq_option(1, SCRIPT_NCCODY, "BISHOP", Node010, NEUTRAL_REACTION);
     end
-    if ((global_var(230) / 100) != 3) then begin
-        giq_option(1, 424, "MORDINO", Node012, 50);
+    if ((global_var(GVAR_NEW_RENO_MADE_MAN) / 100) != 3) then begin
+        giq_option(1, SCRIPT_NCCODY, "MORDINO", Node012, NEUTRAL_REACTION);
     end
-    if ((global_var(230) / 100) != 4) then begin
-        giq_option(1, 424, "WRIGHT", Node011, 50);
+    if ((global_var(GVAR_NEW_RENO_MADE_MAN) / 100) != 4) then begin
+        giq_option(1, SCRIPT_NCCODY, "WRIGHT", Node011, NEUTRAL_REACTION);
     end
-    giq_option(1, 424, "SOMETHING ELSE", Node001, 50);
-    giq_option(1, 424, "EXIT", Node999, 50);
+    giq_option(1, SCRIPT_NCCODY, "SOMETHING ELSE", Node001, NEUTRAL_REACTION);
+    giq_option(1, SCRIPT_NCCODY, "EXIT", Node999, NEUTRAL_REACTION);
 end
 
 procedure Node007
 begin
-    gsay_reply(424, "KARMA == " + global_var(0));
-    giq_option(1, 424, "-100", Node023, 50);
-    giq_option(1, 424, "+100", Node024, 50);
-    giq_option(1, 424, "SOMETHING ELSE", Node001, 50);
-    giq_option(1, 424, "EXIT", Node999, 50);
+    gsay_reply(SCRIPT_NCCODY, "KARMA == " + global_var(GVAR_PLAYER_REPUTATION));
+    giq_option(1, SCRIPT_NCCODY, "-100", Node023, NEUTRAL_REACTION);
+    giq_option(1, SCRIPT_NCCODY, "+100", Node024, NEUTRAL_REACTION);
+    giq_option(1, SCRIPT_NCCODY, "SOMETHING ELSE", Node001, NEUTRAL_REACTION);
+    giq_option(1, SCRIPT_NCCODY, "EXIT", Node999, NEUTRAL_REACTION);
 end
 
 procedure Node008
 begin
-    gsay_reply(424, "FALLOUT 2 FLAGS");
-    if ((global_var(494) == 0) == 0) then begin
-        giq_option(1, 424, "Fallout2_vault13_looking", Node014, 50);
+    gsay_reply(SCRIPT_NCCODY, "FALLOUT 2 FLAGS");
+    if ((global_var(GVAR_FALLOUT_2) == 0) == 0) then begin
+        giq_option(1, SCRIPT_NCCODY, "Fallout2_vault13_looking", Node014, NEUTRAL_REACTION);
     end
-    if ((global_var(494) == 1) == 0) then begin
-        giq_option(1, 424, "Fallout2_vault13_location_known", Node015, 50);
+    if ((global_var(GVAR_FALLOUT_2) == 1) == 0) then begin
+        giq_option(1, SCRIPT_NCCODY, "Fallout2_vault13_location_known", Node015, NEUTRAL_REACTION);
     end
-    if ((global_var(494) == 2) == 0) then begin
-        giq_option(1, 424, "Fallout2_vault13_found", Node016, 50);
+    if ((global_var(GVAR_FALLOUT_2) == 2) == 0) then begin
+        giq_option(1, SCRIPT_NCCODY, "Fallout2_vault13_found", Node016, NEUTRAL_REACTION);
     end
-    if ((global_var(494) == 3) == 0) then begin
-        giq_option(1, 424, "Fallout2_vault13_entered", Node017, 50);
+    if ((global_var(GVAR_FALLOUT_2) == 3) == 0) then begin
+        giq_option(1, SCRIPT_NCCODY, "Fallout2_vault13_entered", Node017, NEUTRAL_REACTION);
     end
-    if ((global_var(494) == 4) == 0) then begin
-        giq_option(1, 424, "Fallout2_tribe_kidnapped", Node018, 50);
+    if ((global_var(GVAR_FALLOUT_2) == 4) == 0) then begin
+        giq_option(1, SCRIPT_NCCODY, "Fallout2_tribe_kidnapped", Node018, NEUTRAL_REACTION);
     end
-    if ((global_var(494) == 5) == 0) then begin
-        giq_option(1, 424, "Fallout2_fix_tanker", Node019, 50);
+    if ((global_var(GVAR_FALLOUT_2) == 5) == 0) then begin
+        giq_option(1, SCRIPT_NCCODY, "Fallout2_fix_tanker", Node019, NEUTRAL_REACTION);
     end
-    if ((global_var(494) == 6) == 0) then begin
-        giq_option(1, 424, "Fallout2_tanker_known", Node020, 50);
+    if ((global_var(GVAR_FALLOUT_2) == 6) == 0) then begin
+        giq_option(1, SCRIPT_NCCODY, "Fallout2_tanker_known", Node020, NEUTRAL_REACTION);
     end
-    if ((global_var(494) == 7) == 0) then begin
-        giq_option(1, 424, "Fallout2_tanker_fixed", Node021, 50);
+    if ((global_var(GVAR_FALLOUT_2) == 7) == 0) then begin
+        giq_option(1, SCRIPT_NCCODY, "Fallout2_tanker_fixed", Node021, NEUTRAL_REACTION);
     end
-    if ((global_var(494) == 8) == 0) then begin
-        giq_option(1, 424, "Fallout2_enclave_destroyed", Node022, 50);
+    if ((global_var(GVAR_FALLOUT_2) == 8) == 0) then begin
+        giq_option(1, SCRIPT_NCCODY, "Fallout2_enclave_destroyed", Node022, NEUTRAL_REACTION);
     end
-    giq_option(1, 424, "SOMETHING ELSE", Node001, 50);
-    giq_option(1, 424, "EXIT", Node999, 50);
+    giq_option(1, SCRIPT_NCCODY, "SOMETHING ELSE", Node001, NEUTRAL_REACTION);
+    giq_option(1, SCRIPT_NCCODY, "EXIT", Node999, NEUTRAL_REACTION);
 end
 
 procedure Node009
 begin
-    set_global_var(230, (1 * 100) + random(1, 15));
-    gsay_reply(424, "SALVATORE == " + message_str(353, 200 + (global_var(230) / 100)) + " / " + message_str(353, 300 + (global_var(230) % 100)));
-    giq_option(1, 424, "SOMETHING ELSE", Node001, 50);
-    giq_option(1, 424, "EXIT", Node999, 50);
+    set_global_var(GVAR_NEW_RENO_MADE_MAN, (1 * 100) + random(1, 15));
+    gsay_reply(SCRIPT_NCCODY, "SALVATORE == " + message_str(SCRIPT_NEWR1, 200 + (global_var(GVAR_NEW_RENO_MADE_MAN) / 100)) + " / " + message_str(353, 300 + (global_var(GVAR_NEW_RENO_MADE_MAN) % 100)));
+    giq_option(1, SCRIPT_NCCODY, "SOMETHING ELSE", Node001, NEUTRAL_REACTION);
+    giq_option(1, SCRIPT_NCCODY, "EXIT", Node999, NEUTRAL_REACTION);
 end
 
 procedure Node010
 begin
-    set_global_var(230, (2 * 100) + random(1, 15));
-    gsay_reply(424, "BISHOP == " + message_str(353, 200 + (global_var(230) / 100)) + " / " + message_str(353, 300 + (global_var(230) % 100)));
-    giq_option(1, 424, "SOMETHING ELSE", Node001, 50);
-    giq_option(1, 424, "EXIT", Node999, 50);
+    set_global_var(GVAR_NEW_RENO_MADE_MAN, (2 * 100) + random(1, 15));
+    gsay_reply(SCRIPT_NCCODY, "BISHOP == " + message_str(SCRIPT_NEWR1, 200 + (global_var(GVAR_NEW_RENO_MADE_MAN) / 100)) + " / " + message_str(353, 300 + (global_var(GVAR_NEW_RENO_MADE_MAN) % 100)));
+    giq_option(1, SCRIPT_NCCODY, "SOMETHING ELSE", Node001, NEUTRAL_REACTION);
+    giq_option(1, SCRIPT_NCCODY, "EXIT", Node999, NEUTRAL_REACTION);
 end
 
 procedure Node011
 begin
-    set_global_var(230, (4 * 100) + random(1, 15));
-    gsay_reply(424, "WRIGHT == " + message_str(353, 200 + (global_var(230) / 100)) + " / " + message_str(353, 300 + (global_var(230) % 100)));
-    giq_option(1, 424, "SOMETHING ELSE", Node001, 50);
-    giq_option(1, 424, "EXIT", Node999, 50);
+    set_global_var(GVAR_NEW_RENO_MADE_MAN, (4 * 100) + random(1, 15));
+    gsay_reply(SCRIPT_NCCODY, "WRIGHT == " + message_str(SCRIPT_NEWR1, 200 + (global_var(GVAR_NEW_RENO_MADE_MAN) / 100)) + " / " + message_str(353, 300 + (global_var(GVAR_NEW_RENO_MADE_MAN) % 100)));
+    giq_option(1, SCRIPT_NCCODY, "SOMETHING ELSE", Node001, NEUTRAL_REACTION);
+    giq_option(1, SCRIPT_NCCODY, "EXIT", Node999, NEUTRAL_REACTION);
 end
 
 procedure Node012
 begin
-    set_global_var(230, (3 * 100) + random(1, 15));
-    gsay_reply(424, "MORDINO == " + message_str(353, 200 + (global_var(230) / 100)) + " / " + message_str(353, 300 + (global_var(230) % 100)));
-    giq_option(1, 424, "SOMETHING ELSE", Node001, 50);
-    giq_option(1, 424, "EXIT", Node999, 50);
+    set_global_var(GVAR_NEW_RENO_MADE_MAN, (3 * 100) + random(1, 15));
+    gsay_reply(SCRIPT_NCCODY, "MORDINO == " + message_str(SCRIPT_NEWR1, 200 + (global_var(GVAR_NEW_RENO_MADE_MAN) / 100)) + " / " + message_str(353, 300 + (global_var(GVAR_NEW_RENO_MADE_MAN) % 100)));
+    giq_option(1, SCRIPT_NCCODY, "SOMETHING ELSE", Node001, NEUTRAL_REACTION);
+    giq_option(1, SCRIPT_NCCODY, "EXIT", Node999, NEUTRAL_REACTION);
 end
 
 procedure Node013
 begin
-    set_global_var(231, random(1, 15));
-    set_global_var(259, 1);
-    gsay_reply(424, "PRIZEFIGHTER == " + message_str(353, 400 + global_var(231)));
-    giq_option(1, 424, "SOMETHING ELSE", Node001, 50);
-    giq_option(1, 424, "EXIT", Node999, 50);
+    set_global_var(GVAR_NEW_RENO_PRIZEFIGHTER, random(1, 15));
+    set_global_var(GVAR_NEW_RENO_HAS_REP_PRIZEFIGHTER, 1);
+    gsay_reply(SCRIPT_NCCODY, "PRIZEFIGHTER == " + message_str(SCRIPT_NEWR1, 400 + global_var(GVAR_NEW_RENO_PRIZEFIGHTER)));
+    giq_option(1, SCRIPT_NCCODY, "SOMETHING ELSE", Node001, NEUTRAL_REACTION);
+    giq_option(1, SCRIPT_NCCODY, "EXIT", Node999, NEUTRAL_REACTION);
 end
 
 procedure Node014
 begin
-    set_global_var(494, 0);
-    gsay_reply(424, "f2_state_v13_looking");
-    giq_option(1, 424, "SOMETHING ELSE", Node001, 50);
-    giq_option(1, 424, "EXIT", Node999, 50);
+    set_global_var(GVAR_FALLOUT_2, 0);
+    gsay_reply(SCRIPT_NCCODY, "f2_state_v13_looking");
+    giq_option(1, SCRIPT_NCCODY, "SOMETHING ELSE", Node001, NEUTRAL_REACTION);
+    giq_option(1, SCRIPT_NCCODY, "EXIT", Node999, NEUTRAL_REACTION);
 end
 
 procedure Node015
 begin
-    set_global_var(494, 1);
-    gsay_reply(424, "f2_state_v13_known");
-    giq_option(1, 424, "SOMETHING ELSE", Node001, 50);
-    giq_option(1, 424, "EXIT", Node999, 50);
+    set_global_var(GVAR_FALLOUT_2, 1);
+    gsay_reply(SCRIPT_NCCODY, "f2_state_v13_known");
+    giq_option(1, SCRIPT_NCCODY, "SOMETHING ELSE", Node001, NEUTRAL_REACTION);
+    giq_option(1, SCRIPT_NCCODY, "EXIT", Node999, NEUTRAL_REACTION);
 end
 
 procedure Node016
 begin
-    set_global_var(494, 2);
-    gsay_reply(424, "f2_state_v13_found");
-    giq_option(1, 424, "SOMETHING ELSE", Node001, 50);
-    giq_option(1, 424, "EXIT", Node999, 50);
+    set_global_var(GVAR_FALLOUT_2, 2);
+    gsay_reply(SCRIPT_NCCODY, "f2_state_v13_found");
+    giq_option(1, SCRIPT_NCCODY, "SOMETHING ELSE", Node001, NEUTRAL_REACTION);
+    giq_option(1, SCRIPT_NCCODY, "EXIT", Node999, NEUTRAL_REACTION);
 end
 
 procedure Node017
 begin
-    set_global_var(494, 3);
-    gsay_reply(424, "f2_state_v13_entered");
-    giq_option(1, 424, "SOMETHING ELSE", Node001, 50);
-    giq_option(1, 424, "EXIT", Node999, 50);
+    set_global_var(GVAR_FALLOUT_2, 3);
+    gsay_reply(SCRIPT_NCCODY, "f2_state_v13_entered");
+    giq_option(1, SCRIPT_NCCODY, "SOMETHING ELSE", Node001, NEUTRAL_REACTION);
+    giq_option(1, SCRIPT_NCCODY, "EXIT", Node999, NEUTRAL_REACTION);
 end
 
 procedure Node018
 begin
-    set_global_var(494, 4);
-    gsay_reply(424, "f2_state_tribe_kidnapped");
-    giq_option(1, 424, "SOMETHING ELSE", Node001, 50);
-    giq_option(1, 424, "EXIT", Node999, 50);
+    set_global_var(GVAR_FALLOUT_2, 4);
+    gsay_reply(SCRIPT_NCCODY, "f2_state_tribe_kidnapped");
+    giq_option(1, SCRIPT_NCCODY, "SOMETHING ELSE", Node001, NEUTRAL_REACTION);
+    giq_option(1, SCRIPT_NCCODY, "EXIT", Node999, NEUTRAL_REACTION);
 end
 
 procedure Node019
 begin
-    set_global_var(494, 5);
-    gsay_reply(424, "f2_state_fix_tanker");
-    giq_option(1, 424, "SOMETHING ELSE", Node001, 50);
-    giq_option(1, 424, "EXIT", Node999, 50);
+    set_global_var(GVAR_FALLOUT_2, 5);
+    gsay_reply(SCRIPT_NCCODY, "f2_state_fix_tanker");
+    giq_option(1, SCRIPT_NCCODY, "SOMETHING ELSE", Node001, NEUTRAL_REACTION);
+    giq_option(1, SCRIPT_NCCODY, "EXIT", Node999, NEUTRAL_REACTION);
 end
 
 procedure Node020
 begin
-    set_global_var(494, 6);
-    gsay_reply(424, "f2_state_tanker_fix_known");
-    giq_option(1, 424, "SOMETHING ELSE", Node001, 50);
-    giq_option(1, 424, "EXIT", Node999, 50);
+    set_global_var(GVAR_FALLOUT_2, 6);
+    gsay_reply(SCRIPT_NCCODY, "f2_state_tanker_fix_known");
+    giq_option(1, SCRIPT_NCCODY, "SOMETHING ELSE", Node001, NEUTRAL_REACTION);
+    giq_option(1, SCRIPT_NCCODY, "EXIT", Node999, NEUTRAL_REACTION);
 end
 
 procedure Node021
 begin
-    set_global_var(494, 7);
-    gsay_reply(424, "f2_state_tanker_fixed");
-    giq_option(1, 424, "SOMETHING ELSE", Node001, 50);
-    giq_option(1, 424, "EXIT", Node999, 50);
+    set_global_var(GVAR_FALLOUT_2, 7);
+    gsay_reply(SCRIPT_NCCODY, "f2_state_tanker_fixed");
+    giq_option(1, SCRIPT_NCCODY, "SOMETHING ELSE", Node001, NEUTRAL_REACTION);
+    giq_option(1, SCRIPT_NCCODY, "EXIT", Node999, NEUTRAL_REACTION);
 end
 
 procedure Node022
 begin
-    set_global_var(494, 8);
-    gsay_reply(424, "f2_state_enclave_destroyed");
-    giq_option(1, 424, "SOMETHING ELSE", Node001, 50);
-    giq_option(1, 424, "EXIT", Node999, 50);
+    set_global_var(GVAR_FALLOUT_2, 8);
+    gsay_reply(SCRIPT_NCCODY, "f2_state_enclave_destroyed");
+    giq_option(1, SCRIPT_NCCODY, "SOMETHING ELSE", Node001, NEUTRAL_REACTION);
+    giq_option(1, SCRIPT_NCCODY, "EXIT", Node999, NEUTRAL_REACTION);
 end
 
 procedure Node023
 begin
-    set_global_var(0, global_var(0) + -100);
+    set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + -100);
     debug_msg("Player gains " + -100 + " Karma Points.");
-    set_global_var(37, 0);
-    set_global_var(38, 0);
-    set_global_var(39, 0);
-    set_global_var(40, 0);
-    set_global_var(41, 0);
-    set_global_var(42, 0);
-    set_global_var(43, 0);
-    set_global_var(44, 0);
-    set_global_var(45, 0);
-    if (global_var(0) >= 1000) then begin
-        set_global_var(37, 1);
+    set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+    set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+    set_global_var(GVAR_KARMA_DEFENDER, 0);
+    set_global_var(GVAR_KARMA_WANDERER, 0);
+    set_global_var(GVAR_KARMA_BETRAYER, 0);
+    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+    set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+    set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+    if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
     end
     else begin
-        if (global_var(0) >= 750) then begin
-            set_global_var(38, 1);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+            set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
         end
         else begin
-            if (global_var(0) >= 500) then begin
-                set_global_var(39, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
             end
             else begin
-                if (global_var(0) >= 250) then begin
-                    set_global_var(40, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                    set_global_var(GVAR_KARMA_DEFENDER, 1);
                 end
                 else begin
-                    if (global_var(0) > -250) then begin
-                        set_global_var(41, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                        set_global_var(GVAR_KARMA_WANDERER, 1);
                     end
                     else begin
-                        if (global_var(0) > -500) then begin
-                            set_global_var(42, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                            set_global_var(GVAR_KARMA_BETRAYER, 1);
                         end
                         else begin
-                            if (global_var(0) > -750) then begin
-                                set_global_var(43, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                             end
                             else begin
-                                if (global_var(0) > -1000) then begin
-                                    set_global_var(44, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                    set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                 end
                                 else begin
-                                    set_global_var(45, 1);
+                                    set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                 end
                             end
                         end
@@ -653,50 +653,50 @@ end
 
 procedure Node024
 begin
-    set_global_var(0, global_var(0) + 100);
+    set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + 100);
     debug_msg("Player gains " + 100 + " Karma Points.");
-    set_global_var(37, 0);
-    set_global_var(38, 0);
-    set_global_var(39, 0);
-    set_global_var(40, 0);
-    set_global_var(41, 0);
-    set_global_var(42, 0);
-    set_global_var(43, 0);
-    set_global_var(44, 0);
-    set_global_var(45, 0);
-    if (global_var(0) >= 1000) then begin
-        set_global_var(37, 1);
+    set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+    set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+    set_global_var(GVAR_KARMA_DEFENDER, 0);
+    set_global_var(GVAR_KARMA_WANDERER, 0);
+    set_global_var(GVAR_KARMA_BETRAYER, 0);
+    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+    set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+    set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+    if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
     end
     else begin
-        if (global_var(0) >= 750) then begin
-            set_global_var(38, 1);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+            set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
         end
         else begin
-            if (global_var(0) >= 500) then begin
-                set_global_var(39, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
             end
             else begin
-                if (global_var(0) >= 250) then begin
-                    set_global_var(40, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                    set_global_var(GVAR_KARMA_DEFENDER, 1);
                 end
                 else begin
-                    if (global_var(0) > -250) then begin
-                        set_global_var(41, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                        set_global_var(GVAR_KARMA_WANDERER, 1);
                     end
                     else begin
-                        if (global_var(0) > -500) then begin
-                            set_global_var(42, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                            set_global_var(GVAR_KARMA_BETRAYER, 1);
                         end
                         else begin
-                            if (global_var(0) > -750) then begin
-                                set_global_var(43, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                             end
                             else begin
-                                if (global_var(0) > -1000) then begin
-                                    set_global_var(44, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                    set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                 end
                                 else begin
-                                    set_global_var(45, 1);
+                                    set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                 end
                             end
                         end
@@ -707,4 +707,3 @@ begin
     end
     call Node007();
 end
-
