diff --git a/Mods/MegaMod/Original/NCRESEAR.ssl b/Mods/MegaMod/ReDefine/NCRESEAR.ssl
index 2b6f9d3..9161a9d 100644
--- a/Mods/MegaMod/Original/NCRESEAR.ssl
+++ b/Mods/MegaMod/ReDefine/NCRESEAR.ssl
@@ -196,53 +196,53 @@ end
 procedure ChooseFromList
 begin
     variable LVar0 := 0;
-    gsay_reply(442, stored_msg);
-    if (message_str(353, top_msg - 1) != "") then begin
-        giq_option(1, 442, message_str(353, 150), CLICK_PREV, 50);
+    Reply(stored_msg);
+    if (message_str(SCRIPT_NEWR1, top_msg - 1) != "") then begin
+        giq_option(1, NAME, message_str(SCRIPT_NEWR1, 150), CLICK_PREV, NEUTRAL_REACTION);
     end
     if (LVar0 == 0) then begin
-        if (message_str(353, top_msg + 0) != "") then begin
-            giq_option(1, 442, message_str(353, top_msg + 0), CLICK_0, 50);
+        if (message_str(SCRIPT_NEWR1, top_msg + 0) != "") then begin
+            giq_option(1, NAME, message_str(SCRIPT_NEWR1, top_msg + 0), CLICK_0, NEUTRAL_REACTION);
         end
         else begin
             LVar0 := 1;
         end
     end
     if (LVar0 == 0) then begin
-        if (message_str(353, top_msg + 1) != "") then begin
-            giq_option(1, 442, message_str(353, top_msg + 1), CLICK_1, 50);
+        if (message_str(SCRIPT_NEWR1, top_msg + 1) != "") then begin
+            giq_option(1, NAME, message_str(SCRIPT_NEWR1, top_msg + 1), CLICK_1, NEUTRAL_REACTION);
         end
         else begin
             LVar0 := 1;
         end
     end
     if (LVar0 == 0) then begin
-        if (message_str(353, top_msg + 2) != "") then begin
-            giq_option(1, 442, message_str(353, top_msg + 2), CLICK_2, 50);
+        if (message_str(SCRIPT_NEWR1, top_msg + 2) != "") then begin
+            giq_option(1, NAME, message_str(SCRIPT_NEWR1, top_msg + 2), CLICK_2, NEUTRAL_REACTION);
         end
         else begin
             LVar0 := 1;
         end
     end
     if (LVar0 == 0) then begin
-        if (message_str(353, top_msg + 3) != "") then begin
-            giq_option(1, 442, message_str(353, top_msg + 3), CLICK_3, 50);
+        if (message_str(SCRIPT_NEWR1, top_msg + 3) != "") then begin
+            giq_option(1, NAME, message_str(SCRIPT_NEWR1, top_msg + 3), CLICK_3, NEUTRAL_REACTION);
         end
         else begin
             LVar0 := 1;
         end
     end
     if (LVar0 == 0) then begin
-        if (message_str(353, top_msg + 4) != "") then begin
-            giq_option(1, 442, message_str(353, top_msg + 4), CLICK_4, 50);
+        if (message_str(SCRIPT_NEWR1, top_msg + 4) != "") then begin
+            giq_option(1, NAME, message_str(SCRIPT_NEWR1, top_msg + 4), CLICK_4, NEUTRAL_REACTION);
         end
         else begin
             LVar0 := 1;
         end
     end
     if (LVar0 == 0) then begin
-        if (message_str(353, top_msg + 5) != "") then begin
-            giq_option(1, 442, message_str(353, 151), CLICK_MORE, 50);
+        if (message_str(SCRIPT_NEWR1, top_msg + 5) != "") then begin
+            giq_option(1, NAME, message_str(SCRIPT_NEWR1, 151), CLICK_MORE, NEUTRAL_REACTION);
         end
     end
 end
@@ -302,10 +302,10 @@ begin
                 if (anim_busy(self_obj) == 0) then begin
                     if ((tile_distance_objs(self_obj, dude_obj) < 8) or obj_can_see_obj(self_obj, dude_obj)) then begin
                         if (anim_busy(self_obj) == 0) then begin
-                            global_temp := rotation_to_tile(tile_num(dude_obj), tile_num(self_obj));
+                            global_temp := rotation_to_tile(dude_tile, tile_num(self_obj));
                             animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(self_obj), global_temp, random(3, 10)), 1);
                             global_temp := (global_temp + 1) % 6;
-                            while ((anim_busy(self_obj) == 0) and (global_temp != rotation_to_tile(tile_num(dude_obj), tile_num(self_obj)))) do begin
+                            while ((anim_busy(self_obj) == 0) and (global_temp != rotation_to_tile(dude_tile, tile_num(self_obj)))) do begin
                                 debug_msg("flee loop: rot == " + global_temp);
                                 animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(self_obj), global_temp, random(3, 10)), 1);
                                 global_temp := (global_temp + 1) % 6;
@@ -315,17 +315,17 @@ begin
                 end
             end
             else begin
-                attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+                attack(dude_obj);
             end
-            if (((global_var(314) bwand 2) != 0) == 0) then begin
-                set_global_var(314, global_var(314) bwor 2);
+            if (((global_var(GVAR_NEW_RENO_FLAG_1) bwand 2) != 0) == 0) then begin
+                set_global_var(GVAR_NEW_RENO_FLAG_1, global_var(GVAR_NEW_RENO_FLAG_1) bwor 2);
                 last_timed_obj := self_obj;
                 last_timed_param := last_timed_param;
-                metarule3(100, self_obj, 19, 0);
+                rm_fixed_timer_event(self_obj, 19);
                 add_timer_event(self_obj, game_ticks(0), 19);
                 last_timed_obj := self_obj;
                 last_timed_param := last_timed_param;
-                metarule3(100, self_obj, 1, 0);
+                rm_fixed_timer_event(self_obj, 1);
                 add_timer_event(self_obj, game_ticks(5), 1);
             end
         end
@@ -345,7 +345,7 @@ begin
         call Node003();
     end
     else begin
-        if ((global_var(284) bwand 1024) != 0) then begin
+        if ((global_var(GVAR_NEW_RENO_MYRON) bwand 1024) != 0) then begin
             call Node002();
         end
         else begin
@@ -358,62 +358,62 @@ begin
     end
     last_timed_obj := self_obj;
     last_timed_param := last_timed_param;
-    metarule3(100, self_obj, 1, 0);
+    rm_fixed_timer_event(self_obj, 1);
     add_timer_event(self_obj, game_ticks(25) + random(0, game_ticks(35 - 25)), 1);
 end
 
 procedure destroy_p_proc
 begin
     if (source_obj == dude_obj) then begin
-        set_global_var(4, global_var(4) + 1);
-        if (metarule(51, self_obj) == 2) then begin
+        set_global_var(GVAR_BAD_MONSTER, global_var(GVAR_BAD_MONSTER) + 1);
+        if (critter_kill_type(self_obj) == KILL_TYPE_children_kills) then begin
             if (source_obj == dude_obj) then begin
-                set_global_var(2, 0);
-                set_global_var(1, global_var(1) + 1);
-                set_global_var(0, global_var(0) + -15);
+                set_global_var(GVAR_CHAMPION_REPUTATION, 0);
+                set_global_var(GVAR_CHILDKILLER_REPUTATION, global_var(GVAR_CHILDKILLER_REPUTATION) + 1);
+                set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + -15);
                 debug_msg("Player gains " + -15 + " Karma Points.");
-                set_global_var(37, 0);
-                set_global_var(38, 0);
-                set_global_var(39, 0);
-                set_global_var(40, 0);
-                set_global_var(41, 0);
-                set_global_var(42, 0);
-                set_global_var(43, 0);
-                set_global_var(44, 0);
-                set_global_var(45, 0);
-                if (global_var(0) >= 1000) then begin
-                    set_global_var(37, 1);
+                set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+                set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+                set_global_var(GVAR_KARMA_DEFENDER, 0);
+                set_global_var(GVAR_KARMA_WANDERER, 0);
+                set_global_var(GVAR_KARMA_BETRAYER, 0);
+                set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+                set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+                set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+                    set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
                 end
                 else begin
-                    if (global_var(0) >= 750) then begin
-                        set_global_var(38, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
                     end
                     else begin
-                        if (global_var(0) >= 500) then begin
-                            set_global_var(39, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                            set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                         end
                         else begin
-                            if (global_var(0) >= 250) then begin
-                                set_global_var(40, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                                set_global_var(GVAR_KARMA_DEFENDER, 1);
                             end
                             else begin
-                                if (global_var(0) > -250) then begin
-                                    set_global_var(41, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                                    set_global_var(GVAR_KARMA_WANDERER, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -500) then begin
-                                        set_global_var(42, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                        set_global_var(GVAR_KARMA_BETRAYER, 1);
                                     end
                                     else begin
-                                        if (global_var(0) > -750) then begin
-                                            set_global_var(43, 1);
+                                        if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                            set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                         end
                                         else begin
-                                            if (global_var(0) > -1000) then begin
-                                                set_global_var(44, 1);
+                                            if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                                set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                             end
                                             else begin
-                                                set_global_var(45, 1);
+                                                set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                             end
                                         end
                                     end
@@ -422,62 +422,62 @@ begin
                         end
                     end
                 end
-                set_global_var(55, global_var(55) + -8);
+                set_global_var(GVAR_TOWN_REP_NEW_RENO, global_var(GVAR_TOWN_REP_NEW_RENO) + -8);
                 debug_msg("Added " + -8 + " to Town Rep");
             end
         end
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(5) > (2 * global_var(4))) or (global_var(3) == 1))) then begin
-            set_global_var(3, 1);
-            set_global_var(2, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
+            set_global_var(GVAR_BERSERKER_REPUTATION, 1);
+            set_global_var(GVAR_CHAMPION_REPUTATION, 0);
         end
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(4) > (3 * global_var(5))) or (global_var(2) == 1)) and (global_var(1) == 0)) then begin
-            set_global_var(2, 1);
-            set_global_var(3, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(4) > (3 * global_var(GVAR_GOOD_MONSTER))) or (global_var(GVAR_CHAMPION_REPUTATION) == 1)) and (global_var(GVAR_CHILDKILLER_REPUTATION) == 0)) then begin
+            set_global_var(GVAR_CHAMPION_REPUTATION, 1);
+            set_global_var(GVAR_BERSERKER_REPUTATION, 0);
         end
-        set_global_var(0, global_var(0) + 5);
+        set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + 5);
         debug_msg("Player gains " + 5 + " Karma Points.");
-        set_global_var(37, 0);
-        set_global_var(38, 0);
-        set_global_var(39, 0);
-        set_global_var(40, 0);
-        set_global_var(41, 0);
-        set_global_var(42, 0);
-        set_global_var(43, 0);
-        set_global_var(44, 0);
-        set_global_var(45, 0);
-        if (global_var(0) >= 1000) then begin
-            set_global_var(37, 1);
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+        set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+        set_global_var(GVAR_KARMA_DEFENDER, 0);
+        set_global_var(GVAR_KARMA_WANDERER, 0);
+        set_global_var(GVAR_KARMA_BETRAYER, 0);
+        set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+        set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+            set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
         end
         else begin
-            if (global_var(0) >= 750) then begin
-                set_global_var(38, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
             end
             else begin
-                if (global_var(0) >= 500) then begin
-                    set_global_var(39, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                 end
                 else begin
-                    if (global_var(0) >= 250) then begin
-                        set_global_var(40, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                        set_global_var(GVAR_KARMA_DEFENDER, 1);
                     end
                     else begin
-                        if (global_var(0) > -250) then begin
-                            set_global_var(41, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                            set_global_var(GVAR_KARMA_WANDERER, 1);
                         end
                         else begin
-                            if (global_var(0) > -500) then begin
-                                set_global_var(42, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                set_global_var(GVAR_KARMA_BETRAYER, 1);
                             end
                             else begin
-                                if (global_var(0) > -750) then begin
-                                    set_global_var(43, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -1000) then begin
-                                        set_global_var(44, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                     end
                                     else begin
-                                        set_global_var(45, 1);
+                                        set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                     end
                                 end
                             end
@@ -486,7 +486,7 @@ begin
                 end
             end
         end
-        set_global_var(55, global_var(55) + 2);
+        set_global_var(GVAR_TOWN_REP_NEW_RENO, global_var(GVAR_TOWN_REP_NEW_RENO) + 2);
         debug_msg("Added " + 2 + " to Town Rep");
     end
 end
@@ -494,13 +494,13 @@ end
 procedure look_at_p_proc
 begin
     script_overrides;
-    display_msg(message_str(442, 100));
+    display_mstr(100);
 end
 
 procedure description_p_proc
 begin
     script_overrides;
-    display_msg(message_str(442, 150 + ((local_var(5) bwand 4) != 0)));
+    display_mstr(150 + ((local_var(5) bwand 4) != 0));
     set_local_var(5, local_var(5) bwor 4);
 end
 
@@ -517,19 +517,19 @@ end
 
 procedure map_enter_p_proc
 begin
-    if (metarule(22, 0) == 0) then begin
-        if (metarule(14, 0)) then begin
-            critter_add_trait(self_obj, 1, 6, 58);
-            critter_add_trait(self_obj, 1, 5, 33);
+    if (is_loading_game == 0) then begin
+        if (map_first_run) then begin
+            critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 58);
+            critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_COWARD);
         end
-        if (((global_var(284) bwand 8) != 0) or critter_is_fleeing(self_obj)) then begin
-            if (has_trait(1, self_obj, 666) == 0) then begin
+        if (((global_var(GVAR_NEW_RENO_MYRON) bwand 8) != 0) or critter_is_fleeing(self_obj)) then begin
+            if (obj_is_visible_flag(self_obj) == 0) then begin
                 set_obj_visibility(self_obj, 0);
             end
         end
         last_timed_obj := self_obj;
         last_timed_param := last_timed_param;
-        metarule3(100, self_obj, 1, 0);
+        rm_fixed_timer_event(self_obj, 1);
         add_timer_event(self_obj, game_ticks(25) + random(0, game_ticks(35 - 25)), 1);
     end
 end
@@ -541,8 +541,8 @@ begin
         call Node008();
     end
     else begin
-        if ((global_var(314) bwand 2) != 0) then begin
-            set_global_var(314, global_var(314) bwand (-1 - 2));
+        if ((global_var(GVAR_NEW_RENO_FLAG_1) bwand 2) != 0) then begin
+            set_global_var(GVAR_NEW_RENO_FLAG_1, global_var(GVAR_NEW_RENO_FLAG_1) bwand (-1 - 2));
         end
         else begin
             if (combat_is_initialized == 0) then begin
@@ -557,7 +557,7 @@ begin
             end
             last_timed_obj := self_obj;
             last_timed_param := last_timed_param;
-            metarule3(100, self_obj, 1, 0);
+            rm_fixed_timer_event(self_obj, 1);
             add_timer_event(self_obj, game_ticks(25) + random(0, game_ticks(35 - 25)), 1);
         end
     end
@@ -574,233 +574,232 @@ end
 
 procedure Node001
 begin
-    float_msg(self_obj, message_str(442, random(200, 204)), 8);
+    float_msg(self_obj, mstr(random(200, 204)), FLOAT_MSG_YELLOW);
 end
 
 procedure Node002
 begin
-    float_msg(self_obj, message_str(442, random(215, 228)), 8);
+    float_msg(self_obj, mstr(random(215, 228)), FLOAT_MSG_YELLOW);
 end
 
 procedure Node003
 begin
-    float_msg(self_obj, message_str(442, random(235, 246)), 8);
+    float_msg(self_obj, mstr(random(235, 246)), FLOAT_MSG_YELLOW);
 end
 
 procedure Node004
 begin
-    gsay_reply(442, 250);
-    giq_option(-3, 442, 251, Node005, 51);
-    giq_option(-3, 442, 252, Node005, 51);
-    giq_option(-3, 442, 253, Node005, 51);
-    giq_option(-3, 442, 254, Node005, 51);
-    giq_option(-3, 442, 255, Node999, 50);
-    giq_option(4, 442, 256, Node006, 50);
-    giq_option(4, 442, 257, Node999, 50);
+    Reply(250);
+    giq_option(-3, NAME, 251, Node005, BAD_REACTION);
+    giq_option(-3, NAME, 252, Node005, BAD_REACTION);
+    giq_option(-3, NAME, 253, Node005, BAD_REACTION);
+    giq_option(-3, NAME, 254, Node005, BAD_REACTION);
+    giq_option(-3, NAME, 255, Node999, NEUTRAL_REACTION);
+    giq_option(4, NAME, 256, Node006, NEUTRAL_REACTION);
+    giq_option(4, NAME, 257, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node005
 begin
-    gsay_reply(442, 265);
-    giq_option(-3, 442, 266, Node998, 50);
-    giq_option(-3, 442, 255, Node998, 50);
+    Reply(265);
+    giq_option(-3, NAME, 266, Node998, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 255, Node998, NEUTRAL_REACTION);
 end
 
 procedure Node006
 begin
     if ((local_var(5) bwand 512) != 0) then begin
-        gsay_reply(442, 276);
+        Reply(276);
     end
     else begin
-        gsay_reply(442, 275);
-        giq_option(4, 442, 277, Node998, 51);
+        Reply(275);
+        giq_option(4, NAME, 277, Node998, BAD_REACTION);
     end
-    if (get_critter_stat(dude_obj, 3) > 5) then begin
-        giq_option(4, 442, 278, Node009, 51);
+    if (dude_charisma > 5) then begin
+        giq_option(4, NAME, 278, Node009, BAD_REACTION);
     end
-    if ((global_var(284) bwand 1) != 0) then begin
-        giq_option(4, 442, 279, Node026, 50);
+    if ((global_var(GVAR_NEW_RENO_MYRON) bwand 1) != 0) then begin
+        giq_option(4, NAME, 279, Node026, NEUTRAL_REACTION);
     end
-    if ((get_critter_stat(dude_obj, 4) < 7) and ((has_skill(dude_obj, 7) > 95) or (has_skill(dude_obj, 12) > 95)) or ((get_critter_stat(dude_obj, 4) >= 7) and ((has_skill(dude_obj, 7) > 75) or (has_skill(dude_obj, 12) > 75)))) then begin
-        giq_option(4, 442, 280, Node010, 49);
+    if ((dude_iq < 7) and ((dude_skill(SKILL_DOCTOR) > 95) or (dude_skill(SKILL_SCIENCE) > 95)) or ((dude_iq >= 7) and ((dude_skill(SKILL_DOCTOR) > 75) or (dude_skill(SKILL_SCIENCE) > 75)))) then begin
+        giq_option(4, NAME, 280, Node010, GOOD_REACTION);
     end
-    giq_option(4, 442, 282, Node999, 49);
+    giq_option(4, NAME, 282, Node999, GOOD_REACTION);
     set_local_var(5, local_var(5) bwor 512);
 end
 
 procedure Node007
 begin
-    gsay_reply(442, 295);
-    giq_option(4, 442, 296, Node998, 51);
-    if (get_critter_stat(dude_obj, 3) > 5) then begin
-        giq_option(4, 442, 278, Node009, 51);
+    Reply(295);
+    giq_option(4, NAME, 296, Node998, BAD_REACTION);
+    if (dude_charisma > 5) then begin
+        giq_option(4, NAME, 278, Node009, BAD_REACTION);
     end
-    if ((get_critter_stat(dude_obj, 4) < 7) and ((has_skill(dude_obj, 7) > 95) or (has_skill(dude_obj, 12) > 95)) or ((get_critter_stat(dude_obj, 4) >= 7) and ((has_skill(dude_obj, 7) > 75) or (has_skill(dude_obj, 12) > 75)))) then begin
-        giq_option(4, 442, 280, Node010, 49);
+    if ((dude_iq < 7) and ((dude_skill(SKILL_DOCTOR) > 95) or (dude_skill(SKILL_SCIENCE) > 95)) or ((dude_iq >= 7) and ((dude_skill(SKILL_DOCTOR) > 75) or (dude_skill(SKILL_SCIENCE) > 75)))) then begin
+        giq_option(4, NAME, 280, Node010, GOOD_REACTION);
     end
-    giq_option(4, 442, 300, Node999, 50);
+    giq_option(4, NAME, 300, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node008
 begin
-    float_msg(self_obj, message_str(442, 310), 8);
+    float_msg(self_obj, mstr(310), FLOAT_MSG_YELLOW);
 end
 
 procedure Node009
 begin
-    gsay_reply(442, 315);
-    giq_option(5, 442, 316, Node010, 50);
-    if ((get_critter_stat(dude_obj, 3) > 6) or (has_skill(dude_obj, 14) > 65) or has_trait(0, dude_obj, 49)) then begin
-        giq_option(4, 442, 317, Node990, 50);
+    Reply(315);
+    giq_option(5, NAME, 316, Node010, NEUTRAL_REACTION);
+    if ((dude_charisma > 6) or (dude_skill(SKILL_SPEECH) > 65) or has_trait(TRAIT_PERK, dude_obj, PERK_smooth_talker)) then begin
+        giq_option(4, NAME, 317, Node990, NEUTRAL_REACTION);
     end
-    giq_option(4, 442, 318, Node999, 50);
+    giq_option(4, NAME, 318, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node010
 begin
-    gsay_reply(442, 325);
-    giq_option(4, 442, 326, Node011, 50);
-    if (has_skill(dude_obj, 12) > 75) then begin
-        giq_option(7, 442, 327, Node012, 50);
+    Reply(325);
+    giq_option(4, NAME, 326, Node011, NEUTRAL_REACTION);
+    if (dude_skill(SKILL_SCIENCE) > 75) then begin
+        giq_option(7, NAME, 327, Node012, NEUTRAL_REACTION);
     end
-    giq_option(4, 442, 328, Node999, 50);
+    giq_option(4, NAME, 328, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node011
 begin
-    gsay_reply(442, 335);
-    giq_option(4, 442, 336 + (global_var(0) < 0), Node999, 50);
+    Reply(335);
+    giq_option(4, NAME, 336 + (global_var(GVAR_PLAYER_REPUTATION) < 0), Node999, NEUTRAL_REACTION);
 end
 
 procedure Node012
 begin
-    gsay_reply(442, 345);
-    giq_option(4, 442, 346, Node013, 50);
-    giq_option(4, 442, 347, Node999, 50);
+    Reply(345);
+    giq_option(4, NAME, 346, Node013, NEUTRAL_REACTION);
+    giq_option(4, NAME, 347, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node013
 begin
-    gsay_reply(442, 355);
-    giq_option(4, 442, 356, Node015, 50);
-    giq_option(4, 442, 357, Node014, 50);
-    giq_option(4, 442, 358, Node999, 50);
+    Reply(355);
+    giq_option(4, NAME, 356, Node015, NEUTRAL_REACTION);
+    giq_option(4, NAME, 357, Node014, NEUTRAL_REACTION);
+    giq_option(4, NAME, 358, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node014
 begin
-    if (global_var(84) < 2) then begin
-        set_global_var(84, 2);
+    if (global_var(GVAR_VAULT_REDDING_PROBLEM) < 2) then begin
+        set_global_var(GVAR_VAULT_REDDING_PROBLEM, 2);
     end
-    gsay_reply(442, 365);
-    giq_option(4, 442, 366, Node016, 50);
-    giq_option(4, 442, 367, Node999, 50);
+    Reply(365);
+    giq_option(4, NAME, 366, Node016, NEUTRAL_REACTION);
+    giq_option(4, NAME, 367, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node015
 begin
-    gsay_reply(442, 375);
-    giq_option(4, 442, 366, Node014, 50);
-    giq_option(4, 442, 377, Node999, 50);
+    Reply(375);
+    giq_option(4, NAME, 366, Node014, NEUTRAL_REACTION);
+    giq_option(4, NAME, 377, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node016
 begin
-    gsay_reply(442, 385);
-    giq_option(4, 442, 386, Node017, 50);
-    giq_option(4, 442, 377, Node999, 50);
+    Reply(385);
+    giq_option(4, NAME, 386, Node017, NEUTRAL_REACTION);
+    giq_option(4, NAME, 377, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node017
 begin
-    set_global_var(314, global_var(314) bwor 8);
-    set_global_var(284, global_var(284) bwor 1);
-    gsay_reply(442, 395);
-    giq_option(4, 442, 396, Node018, 50);
-    giq_option(4, 442, 377, Node999, 50);
+    set_global_var(GVAR_NEW_RENO_FLAG_1, global_var(GVAR_NEW_RENO_FLAG_1) bwor 8);
+    set_global_var(GVAR_NEW_RENO_MYRON, global_var(GVAR_NEW_RENO_MYRON) bwor 1);
+    Reply(395);
+    giq_option(4, NAME, 396, Node018, NEUTRAL_REACTION);
+    giq_option(4, NAME, 377, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node018
 begin
-    set_global_var(284, global_var(284) bwor 2);
-    gsay_reply(442, 405);
-    if ((obj_is_carrying_obj_pid(self_obj, 413) > 0) and (((global_var(284) bwand 1024) != 0) == 0)) then begin
-        giq_option(4, 442, 406, Node019, 50);
+    set_global_var(GVAR_NEW_RENO_MYRON, global_var(GVAR_NEW_RENO_MYRON) bwor 2);
+    Reply(405);
+    if ((obj_is_carrying_obj_pid(self_obj, 413) > 0) and (((global_var(GVAR_NEW_RENO_MYRON) bwand 1024) != 0) == 0)) then begin
+        giq_option(4, NAME, 406, Node019, NEUTRAL_REACTION);
     end
-    giq_option(4, 442, 407, Node999, 50);
+    giq_option(4, NAME, 407, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node019
 begin
-    set_global_var(284, global_var(284) bwor 1024);
-    restock_obj := create_object_sid(413, 0, 0, -1);
+    set_global_var(GVAR_NEW_RENO_MYRON, global_var(GVAR_NEW_RENO_MYRON) bwor 1024);
+    restock_obj := create_object(413, 0, 0);
     add_mult_objs_to_inven(dude_obj, restock_obj, 1);
-    display_msg(message_str(442, 1000));
+    display_mstr(1000);
     give_exp_points(500);
-    display_msg(message_str(14, 100) + 500 + message_str(14, 101));
-    gsay_reply(442, 415 + ((get_critter_stat(dude_obj, 34) == 1) and (get_critter_stat(dude_obj, 3) > 5)));
-    giq_option(4, 442, 417, Node999, 50);
+    display_msg(g_mstr(100) + 500 + g_mstr(101));
+    Reply(415 + ((dude_gender == 1) and (dude_charisma > 5)));
+    giq_option(4, NAME, 417, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node020
 begin
-    gsay_reply(442, 425 + (get_critter_stat(dude_obj, 34) == 1));
-    giq_option(4, 442, 427, Node999, 50);
-    giq_option(4, 442, 428 + (global_var(0) < 0), Node999, 50);
+    Reply(425 + (dude_gender == 1));
+    giq_option(4, NAME, 427, Node999, NEUTRAL_REACTION);
+    giq_option(4, NAME, 428 + (global_var(GVAR_PLAYER_REPUTATION) < 0), Node999, NEUTRAL_REACTION);
 end
 
 procedure Node021
 begin
-    gsay_reply(442, 435);
-    if ((get_critter_stat(dude_obj, 3) > 6) or (has_skill(dude_obj, 14) > 75) or has_trait(0, dude_obj, 49)) then begin
-        giq_option(4, 442, 436, Node022, 50);
+    Reply(435);
+    if ((dude_charisma > 6) or (dude_skill(SKILL_SPEECH) > 75) or has_trait(TRAIT_PERK, dude_obj, PERK_smooth_talker)) then begin
+        giq_option(4, NAME, 436, Node022, NEUTRAL_REACTION);
     end
-    giq_option(4, 442, 437, Node999, 50);
+    giq_option(4, NAME, 437, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node022
 begin
-    gsay_reply(442, 445);
-    giq_option(4, 442, 446, Node023, 51);
-    giq_option(4, 442, 447, Node025, 49);
-    giq_option(4, 442, 448, Node013, 50);
-    giq_option(4, 442, 449, Node999, 50);
+    Reply(445);
+    giq_option(4, NAME, 446, Node023, BAD_REACTION);
+    giq_option(4, NAME, 447, Node025, GOOD_REACTION);
+    giq_option(4, NAME, 448, Node013, NEUTRAL_REACTION);
+    giq_option(4, NAME, 449, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node023
 begin
-    gsay_reply(442, 455);
-    giq_option(4, 442, 456, Node024, 51);
-    giq_option(4, 442, 457, Node999, 50);
+    Reply(455);
+    giq_option(4, NAME, 456, Node024, BAD_REACTION);
+    giq_option(4, NAME, 457, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node024
 begin
-    gsay_reply(442, 465);
-    giq_option(4, 442, 466, Node998, 51);
-    giq_option(4, 442, 467, Node999, 50);
+    Reply(465);
+    giq_option(4, NAME, 466, Node998, BAD_REACTION);
+    giq_option(4, NAME, 467, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node025
 begin
-    gsay_reply(442, 475);
-    giq_option(4, 442, 476, Node013, 50);
-    giq_option(4, 442, 477, Node999, 50);
+    Reply(475);
+    giq_option(4, NAME, 476, Node013, NEUTRAL_REACTION);
+    giq_option(4, NAME, 477, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node026
 begin
-    gsay_reply(442, 485);
-    giq_option(4, 442, 318, Node999, 50);
+    Reply(485);
+    giq_option(4, NAME, 318, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node990
 begin
-    if ((get_critter_stat(dude_obj, 34) == 1) or (get_critter_stat(dude_obj, 3) < 6)) then begin
+    if ((dude_gender == 1) or (dude_charisma < 6)) then begin
         call Node020();
     end
     else begin
         call Node021();
     end
 end
-
