diff --git a/Mods/MegaMod/Original/NCRNGSH2.ssl b/Mods/MegaMod/ReDefine/NCRNGSH2.ssl
index 937a290..82b47e1 100644
--- a/Mods/MegaMod/Original/NCRNGSH2.ssl
+++ b/Mods/MegaMod/ReDefine/NCRNGSH2.ssl
@@ -18,39 +18,39 @@ procedure map_enter_p_proc
 begin
     variable LVar0 := 0;
     variable LVar1 := 0;
-    if (metarule(14, 0)) then begin
-        display_msg(message_str(2072, 200));
+    if (map_first_run) then begin
+        display_msg(message_str(SCRIPT_NCRNGSH1, 200));
     end
-    if (global_var(27) == 5) then begin
+    if (global_var(GVAR_LOAD_MAP_INDEX) == 5) then begin
         set_map_var(2, 1);
         move_to(dude_obj, 19694, 0);
-        if (global_var(856) >= 1) then begin
-            LVar0 := create_object_sid(16777433, 20299, elevation(dude_obj), 2076);
-            critter_add_trait(LVar0, 1, 6, 0);
-            LVar1 := create_object_sid(10, 0, 0, -1);
+        if (global_var(GVAR_EASTER_EGG_ENCOUNTER) >= 1) then begin
+            LVar0 := create_object_sid(16777433, 20299, dude_elevation, SCRIPT_SCATKSLV);
+            critter_add_trait(LVar0, TRAIT_OBJECT, OBJECT_TEAM_NUM, 0);
+            LVar1 := create_object(10, 0, 0);
             add_obj_to_inven(LVar0, LVar1);
             wield_obj_critter(LVar0, LVar1);
-            LVar1 := create_object_sid(34, 0, 0, -1);
+            LVar1 := create_object(34, 0, 0);
             add_obj_to_inven(LVar0, LVar1);
-            LVar1 := create_object_sid(40, 0, 0, -1);
+            LVar1 := create_object(40, 0, 0);
             add_obj_to_inven(LVar0, LVar1);
         end
-        if (global_var(856) == 2) then begin
-            LVar0 := create_object_sid(16777433, 19890, elevation(dude_obj), 2076);
-            critter_add_trait(LVar0, 1, 6, 0);
-            LVar1 := create_object_sid(10, 0, 0, -1);
+        if (global_var(GVAR_EASTER_EGG_ENCOUNTER) == 2) then begin
+            LVar0 := create_object_sid(16777433, 19890, dude_elevation, SCRIPT_SCATKSLV);
+            critter_add_trait(LVar0, TRAIT_OBJECT, OBJECT_TEAM_NUM, 0);
+            LVar1 := create_object(10, 0, 0);
             add_obj_to_inven(LVar0, LVar1);
             wield_obj_critter(LVar0, LVar1);
-            LVar1 := create_object_sid(34, 0, 0, -1);
+            LVar1 := create_object(34, 0, 0);
             add_obj_to_inven(LVar0, LVar1);
-            LVar1 := create_object_sid(40, 0, 0, -1);
+            LVar1 := create_object(40, 0, 0);
             add_obj_to_inven(LVar0, LVar1);
         end
         set_map_var(3, 1);
     end
     else begin
         set_exit_grids(0, -2, 0, 20100, 0);
-        set_global_var(27, 0);
+        set_global_var(GVAR_LOAD_MAP_INDEX, 0);
     end
     if ((get_month >= 3) and (get_month < 5)) then begin
         if ((game_time_hour >= 600) and (game_time_hour < (600 + 100))) then begin
@@ -212,4 +212,3 @@ begin
         end
     end
 end
-
