diff --git a/Mods/MegaMod/Original/NCTRAY.ssl b/Mods/MegaMod/ReDefine/NCTRAY.ssl
index 38227e9..a32b9ac 100644
--- a/Mods/MegaMod/Original/NCTRAY.ssl
+++ b/Mods/MegaMod/ReDefine/NCTRAY.ssl
@@ -280,53 +280,53 @@ end
 procedure ChooseFromList
 begin
     variable LVar0 := 0;
-    gsay_reply(445, stored_msg);
-    if (message_str(353, top_msg - 1) != "") then begin
-        giq_option(1, 445, message_str(353, 150), CLICK_PREV, 50);
+    Reply(stored_msg);
+    if (message_str(SCRIPT_NEWR1, top_msg - 1) != "") then begin
+        giq_option(1, NAME, message_str(SCRIPT_NEWR1, 150), CLICK_PREV, NEUTRAL_REACTION);
     end
     if (LVar0 == 0) then begin
-        if (message_str(353, top_msg + 0) != "") then begin
-            giq_option(1, 445, message_str(353, top_msg + 0), CLICK_0, 50);
+        if (message_str(SCRIPT_NEWR1, top_msg + 0) != "") then begin
+            giq_option(1, NAME, message_str(SCRIPT_NEWR1, top_msg + 0), CLICK_0, NEUTRAL_REACTION);
         end
         else begin
             LVar0 := 1;
         end
     end
     if (LVar0 == 0) then begin
-        if (message_str(353, top_msg + 1) != "") then begin
-            giq_option(1, 445, message_str(353, top_msg + 1), CLICK_1, 50);
+        if (message_str(SCRIPT_NEWR1, top_msg + 1) != "") then begin
+            giq_option(1, NAME, message_str(SCRIPT_NEWR1, top_msg + 1), CLICK_1, NEUTRAL_REACTION);
         end
         else begin
             LVar0 := 1;
         end
     end
     if (LVar0 == 0) then begin
-        if (message_str(353, top_msg + 2) != "") then begin
-            giq_option(1, 445, message_str(353, top_msg + 2), CLICK_2, 50);
+        if (message_str(SCRIPT_NEWR1, top_msg + 2) != "") then begin
+            giq_option(1, NAME, message_str(SCRIPT_NEWR1, top_msg + 2), CLICK_2, NEUTRAL_REACTION);
         end
         else begin
             LVar0 := 1;
         end
     end
     if (LVar0 == 0) then begin
-        if (message_str(353, top_msg + 3) != "") then begin
-            giq_option(1, 445, message_str(353, top_msg + 3), CLICK_3, 50);
+        if (message_str(SCRIPT_NEWR1, top_msg + 3) != "") then begin
+            giq_option(1, NAME, message_str(SCRIPT_NEWR1, top_msg + 3), CLICK_3, NEUTRAL_REACTION);
         end
         else begin
             LVar0 := 1;
         end
     end
     if (LVar0 == 0) then begin
-        if (message_str(353, top_msg + 4) != "") then begin
-            giq_option(1, 445, message_str(353, top_msg + 4), CLICK_4, 50);
+        if (message_str(SCRIPT_NEWR1, top_msg + 4) != "") then begin
+            giq_option(1, NAME, message_str(SCRIPT_NEWR1, top_msg + 4), CLICK_4, NEUTRAL_REACTION);
         end
         else begin
             LVar0 := 1;
         end
     end
     if (LVar0 == 0) then begin
-        if (message_str(353, top_msg + 5) != "") then begin
-            giq_option(1, 445, message_str(353, 151), CLICK_MORE, 50);
+        if (message_str(SCRIPT_NEWR1, top_msg + 5) != "") then begin
+            giq_option(1, NAME, message_str(SCRIPT_NEWR1, 151), CLICK_MORE, NEUTRAL_REACTION);
         end
     end
     return 0;
@@ -384,15 +384,15 @@ end
 procedure critter_p_proc
 begin
     if (obj_can_see_obj(self_obj, dude_obj)) then begin
-        if (((local_var(4) bwand 2) != 0) or ((global_var(694) bwand 4) != 0)) then begin
+        if (((local_var(4) bwand 2) != 0) or ((global_var(GVAR_EMILY_SUBST_TIME) bwand 4) != 0)) then begin
             if (critter_is_fleeing(self_obj)) then begin
                 if (anim_busy(self_obj) == 0) then begin
                     if ((tile_distance_objs(self_obj, dude_obj) < 8) or obj_can_see_obj(self_obj, dude_obj)) then begin
                         if (anim_busy(self_obj) == 0) then begin
-                            global_temp := rotation_to_tile(tile_num(dude_obj), tile_num(self_obj));
+                            global_temp := rotation_to_tile(dude_tile, tile_num(self_obj));
                             animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(self_obj), global_temp, random(3, 10)), 1);
                             global_temp := (global_temp + 1) % 6;
-                            while ((anim_busy(self_obj) == 0) and (global_temp != rotation_to_tile(tile_num(dude_obj), tile_num(self_obj)))) do begin
+                            while ((anim_busy(self_obj) == 0) and (global_temp != rotation_to_tile(dude_tile, tile_num(self_obj)))) do begin
                                 debug_msg("flee loop: rot == " + global_temp);
                                 animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(self_obj), global_temp, random(3, 10)), 1);
                                 global_temp := (global_temp + 1) % 6;
@@ -402,11 +402,11 @@ begin
                 end
             end
             else begin
-                attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+                attack(dude_obj);
             end
         end
         else begin
-            if (has_trait(1, self_obj, 666) and (elevation(self_obj) == elevation(dude_obj))) then begin
+            if (obj_is_visible_flag(self_obj) and (elevation(self_obj) == dude_elevation)) then begin
                 if (new_reno_t_ray_obj == 0) then begin
                     new_reno_t_ray_obj := self_obj;
                 end
@@ -434,16 +434,16 @@ begin
         call Node999();
     end
     else begin
-        if ((global_var(348) == 2) or (global_var(348) == 3)) then begin
+        if ((global_var(GVAR_NEW_RENO_STOLEN_CAR) == 2) or (global_var(GVAR_NEW_RENO_STOLEN_CAR) == 3)) then begin
             if (((local_var(4) bwand 512) != 0) == 0) then begin
-                display_msg(message_str(445, 1000));
+                display_mstr(1000);
                 give_exp_points(750);
-                display_msg(message_str(14, 100) + 750 + message_str(14, 101));
-                if ((46 <= prev_node) and (prev_node <= 53) and ((global_var(230) / 100) != 2)) then begin
-                    display_msg(message_str(445, 1001));
-                    set_global_var(55, global_var(55) + 2);
+                display_msg(g_mstr(100) + 750 + g_mstr(101));
+                if ((46 <= prev_node) and (prev_node <= 53) and ((global_var(GVAR_NEW_RENO_MADE_MAN) / 100) != 2)) then begin
+                    display_mstr(1001);
+                    set_global_var(GVAR_TOWN_REP_NEW_RENO, global_var(GVAR_TOWN_REP_NEW_RENO) + 2);
                     give_exp_points(750);
-                    display_msg(message_str(14, 100) + 750 + message_str(14, 101));
+                    display_msg(g_mstr(100) + 750 + g_mstr(101));
                 end
             end
             set_local_var(4, local_var(4) bwor 512);
@@ -451,7 +451,7 @@ begin
     end
     move_obj_inven_to_obj(self_obj, new_reno_t_ray_box);
     move_obj_inven_to_obj(new_reno_t_ray_box, self_obj);
-    if ((global_var(494) == 8) and (global_var(348) == 2)) then begin
+    if ((global_var(GVAR_FALLOUT_2) == 8) and (global_var(GVAR_NEW_RENO_STOLEN_CAR) == 2)) then begin
         if (in_dialog == 0) then begin
             in_dialog := 1;
             start_gdialog(445, self_obj, 4, -1, -1);
@@ -484,7 +484,7 @@ begin
     set_local_var(4, local_var(4) bwor 1);
     last_timed_obj := self_obj;
     last_timed_param := last_timed_param;
-    metarule3(100, self_obj, 1, 0);
+    rm_fixed_timer_event(self_obj, 1);
     add_timer_event(self_obj, game_ticks(25) + random(0, game_ticks(35 - 25)), 1);
     gdialog_set_barter_mod(0);
 end
@@ -492,55 +492,55 @@ end
 procedure destroy_p_proc
 begin
     if (source_obj == dude_obj) then begin
-        set_global_var(4, global_var(4) + 1);
-        if (metarule(51, self_obj) == 2) then begin
+        set_global_var(GVAR_BAD_MONSTER, global_var(GVAR_BAD_MONSTER) + 1);
+        if (critter_kill_type(self_obj) == KILL_TYPE_children_kills) then begin
             if (source_obj == dude_obj) then begin
-                set_global_var(2, 0);
-                set_global_var(1, global_var(1) + 1);
-                set_global_var(0, global_var(0) + -15);
+                set_global_var(GVAR_CHAMPION_REPUTATION, 0);
+                set_global_var(GVAR_CHILDKILLER_REPUTATION, global_var(GVAR_CHILDKILLER_REPUTATION) + 1);
+                set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + -15);
                 debug_msg("Player gains " + -15 + " Karma Points.");
-                set_global_var(37, 0);
-                set_global_var(38, 0);
-                set_global_var(39, 0);
-                set_global_var(40, 0);
-                set_global_var(41, 0);
-                set_global_var(42, 0);
-                set_global_var(43, 0);
-                set_global_var(44, 0);
-                set_global_var(45, 0);
-                if (global_var(0) >= 1000) then begin
-                    set_global_var(37, 1);
+                set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+                set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+                set_global_var(GVAR_KARMA_DEFENDER, 0);
+                set_global_var(GVAR_KARMA_WANDERER, 0);
+                set_global_var(GVAR_KARMA_BETRAYER, 0);
+                set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+                set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+                set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+                    set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
                 end
                 else begin
-                    if (global_var(0) >= 750) then begin
-                        set_global_var(38, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
                     end
                     else begin
-                        if (global_var(0) >= 500) then begin
-                            set_global_var(39, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                            set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                         end
                         else begin
-                            if (global_var(0) >= 250) then begin
-                                set_global_var(40, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                                set_global_var(GVAR_KARMA_DEFENDER, 1);
                             end
                             else begin
-                                if (global_var(0) > -250) then begin
-                                    set_global_var(41, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                                    set_global_var(GVAR_KARMA_WANDERER, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -500) then begin
-                                        set_global_var(42, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                        set_global_var(GVAR_KARMA_BETRAYER, 1);
                                     end
                                     else begin
-                                        if (global_var(0) > -750) then begin
-                                            set_global_var(43, 1);
+                                        if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                            set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                         end
                                         else begin
-                                            if (global_var(0) > -1000) then begin
-                                                set_global_var(44, 1);
+                                            if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                                set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                             end
                                             else begin
-                                                set_global_var(45, 1);
+                                                set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                             end
                                         end
                                     end
@@ -549,62 +549,62 @@ begin
                         end
                     end
                 end
-                set_global_var(55, global_var(55) + -8);
+                set_global_var(GVAR_TOWN_REP_NEW_RENO, global_var(GVAR_TOWN_REP_NEW_RENO) + -8);
                 debug_msg("Added " + -8 + " to Town Rep");
             end
         end
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(5) > (2 * global_var(4))) or (global_var(3) == 1))) then begin
-            set_global_var(3, 1);
-            set_global_var(2, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
+            set_global_var(GVAR_BERSERKER_REPUTATION, 1);
+            set_global_var(GVAR_CHAMPION_REPUTATION, 0);
         end
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(4) > (3 * global_var(5))) or (global_var(2) == 1)) and (global_var(1) == 0)) then begin
-            set_global_var(2, 1);
-            set_global_var(3, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(4) > (3 * global_var(GVAR_GOOD_MONSTER))) or (global_var(GVAR_CHAMPION_REPUTATION) == 1)) and (global_var(GVAR_CHILDKILLER_REPUTATION) == 0)) then begin
+            set_global_var(GVAR_CHAMPION_REPUTATION, 1);
+            set_global_var(GVAR_BERSERKER_REPUTATION, 0);
         end
-        set_global_var(0, global_var(0) + 5);
+        set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + 5);
         debug_msg("Player gains " + 5 + " Karma Points.");
-        set_global_var(37, 0);
-        set_global_var(38, 0);
-        set_global_var(39, 0);
-        set_global_var(40, 0);
-        set_global_var(41, 0);
-        set_global_var(42, 0);
-        set_global_var(43, 0);
-        set_global_var(44, 0);
-        set_global_var(45, 0);
-        if (global_var(0) >= 1000) then begin
-            set_global_var(37, 1);
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+        set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+        set_global_var(GVAR_KARMA_DEFENDER, 0);
+        set_global_var(GVAR_KARMA_WANDERER, 0);
+        set_global_var(GVAR_KARMA_BETRAYER, 0);
+        set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+        set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+            set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
         end
         else begin
-            if (global_var(0) >= 750) then begin
-                set_global_var(38, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
             end
             else begin
-                if (global_var(0) >= 500) then begin
-                    set_global_var(39, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                 end
                 else begin
-                    if (global_var(0) >= 250) then begin
-                        set_global_var(40, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                        set_global_var(GVAR_KARMA_DEFENDER, 1);
                     end
                     else begin
-                        if (global_var(0) > -250) then begin
-                            set_global_var(41, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                            set_global_var(GVAR_KARMA_WANDERER, 1);
                         end
                         else begin
-                            if (global_var(0) > -500) then begin
-                                set_global_var(42, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                set_global_var(GVAR_KARMA_BETRAYER, 1);
                             end
                             else begin
-                                if (global_var(0) > -750) then begin
-                                    set_global_var(43, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -1000) then begin
-                                        set_global_var(44, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                     end
                                     else begin
-                                        set_global_var(45, 1);
+                                        set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                     end
                                 end
                             end
@@ -613,7 +613,7 @@ begin
                 end
             end
         end
-        set_global_var(55, global_var(55) + 2);
+        set_global_var(GVAR_TOWN_REP_NEW_RENO, global_var(GVAR_TOWN_REP_NEW_RENO) + 2);
         debug_msg("Added " + 2 + " to Town Rep");
     end
 end
@@ -621,13 +621,13 @@ end
 procedure look_at_p_proc
 begin
     script_overrides;
-    display_msg(message_str(445, 100));
+    display_mstr(100);
 end
 
 procedure description_p_proc
 begin
     script_overrides;
-    display_msg(message_str(445, random(150, 150 + (3 * ((local_var(4) bwand 4) != 0)))));
+    display_mstr(random(150, 150 + (3 * ((local_var(4) bwand 4) != 0))));
     set_local_var(4, local_var(4) bwor 4);
 end
 
@@ -640,20 +640,20 @@ begin
     if (source_obj == dude_obj) then begin
         set_local_var(4, local_var(4) bwor 8);
         set_local_var(4, local_var(4) bwor 2);
-        set_global_var(694, global_var(694) bwor 4);
+        set_global_var(GVAR_EMILY_SUBST_TIME, global_var(GVAR_EMILY_SUBST_TIME) bwor 4);
     end
 end
 
 procedure map_enter_p_proc
 begin
-    if (metarule(22, 0) == 0) then begin
-        if (metarule(14, 0)) then begin
-            critter_add_trait(self_obj, 1, 6, 191);
-            critter_add_trait(self_obj, 1, 5, 13);
+    if (is_loading_game == 0) then begin
+        if (map_first_run) then begin
+            critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 191);
+            critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_THUGS);
         end
         last_timed_obj := self_obj;
         last_timed_param := last_timed_param;
-        metarule3(100, self_obj, 1, 0);
+        rm_fixed_timer_event(self_obj, 1);
         add_timer_event(self_obj, game_ticks(25) + random(0, game_ticks(35 - 25)), 1);
     end
     if (local_var(6) < game_time) then begin
@@ -661,7 +661,7 @@ begin
         if (obj_is_carrying_obj_pid(self_obj, 38) < restock_amt) then begin
             if (80 >= random(1, 100)) then begin
                 if (obj_is_carrying_obj_pid(self_obj, 38) < restock_amt) then begin
-                    restock_obj := create_object_sid(38, 0, 0, -1);
+                    restock_obj := create_object(38, 0, 0);
                     add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 38));
                 end
                 else begin
@@ -684,7 +684,7 @@ begin
                                 end
                                 else begin
                                     if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                        metarule(43, self_obj);
+                                        inven_unwield(self_obj);
                                     end
                                 end
                             end
@@ -702,7 +702,7 @@ begin
         end
         else begin
             if (obj_is_carrying_obj_pid(self_obj, 38) < restock_amt) then begin
-                restock_obj := create_object_sid(38, 0, 0, -1);
+                restock_obj := create_object(38, 0, 0);
                 add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 38));
             end
             else begin
@@ -725,7 +725,7 @@ begin
                             end
                             else begin
                                 if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                    metarule(43, self_obj);
+                                    inven_unwield(self_obj);
                                 end
                             end
                         end
@@ -744,7 +744,7 @@ begin
         if (obj_is_carrying_obj_pid(self_obj, 39) < restock_amt) then begin
             if (60 >= random(1, 100)) then begin
                 if (obj_is_carrying_obj_pid(self_obj, 39) < restock_amt) then begin
-                    restock_obj := create_object_sid(39, 0, 0, -1);
+                    restock_obj := create_object(39, 0, 0);
                     add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 39));
                 end
                 else begin
@@ -767,7 +767,7 @@ begin
                                 end
                                 else begin
                                     if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                        metarule(43, self_obj);
+                                        inven_unwield(self_obj);
                                     end
                                 end
                             end
@@ -785,7 +785,7 @@ begin
         end
         else begin
             if (obj_is_carrying_obj_pid(self_obj, 39) < restock_amt) then begin
-                restock_obj := create_object_sid(39, 0, 0, -1);
+                restock_obj := create_object(39, 0, 0);
                 add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 39));
             end
             else begin
@@ -808,7 +808,7 @@ begin
                             end
                             else begin
                                 if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                    metarule(43, self_obj);
+                                    inven_unwield(self_obj);
                                 end
                             end
                         end
@@ -841,7 +841,7 @@ begin
     else begin
         if (combat_is_initialized == 0) then begin
             if (obj_can_see_obj(self_obj, dude_obj)) then begin
-                if (global_var(494) == 8) then begin
+                if (global_var(GVAR_FALLOUT_2) == 8) then begin
                     if (local_var(5) > 0) then begin
                         call Node001();
                     end
@@ -867,7 +867,7 @@ begin
     end
     last_timed_obj := self_obj;
     last_timed_param := last_timed_param;
-    metarule3(100, self_obj, 1, 0);
+    rm_fixed_timer_event(self_obj, 1);
     add_timer_event(self_obj, game_ticks(25) + random(0, game_ticks(35 - 25)), 1);
 end
 
@@ -882,422 +882,422 @@ end
 
 procedure Node001
 begin
-    float_msg(self_obj, message_str(445, random(200, 202)), 8);
+    float_msg(self_obj, mstr(random(200, 202)), FLOAT_MSG_YELLOW);
 end
 
 procedure Node002
 begin
-    float_msg(self_obj, message_str(445, random(210, 212)), 8);
+    float_msg(self_obj, mstr(random(210, 212)), FLOAT_MSG_YELLOW);
 end
 
 procedure Node003
 begin
-    float_msg(self_obj, message_str(445, random(220, 222)), 8);
+    float_msg(self_obj, mstr(random(220, 222)), FLOAT_MSG_YELLOW);
 end
 
 procedure Node004
 begin
-    float_msg(self_obj, message_str(445, random(230, 232)), 8);
+    float_msg(self_obj, mstr(random(230, 232)), FLOAT_MSG_YELLOW);
 end
 
 procedure Node005
 begin
-    float_msg(self_obj, message_str(445, random(251 - (get_critter_stat(dude_obj, 34) == 0), 267 + (get_critter_stat(dude_obj, 34) == 1))), 8);
+    float_msg(self_obj, mstr(random(251 - (dude_gender == 0), 267 + (dude_gender == 1))), FLOAT_MSG_YELLOW);
 end
 
 procedure Node006
 begin
     if (((local_var(4) bwand 2048) != 0) == 0) then begin
-        gsay_reply(445, 275 + ((get_critter_stat(dude_obj, 34) == 1) and ((global_var(232) > 0) or (get_critter_stat(dude_obj, 3) > 6))));
+        Reply(275 + ((dude_gender == 1) and ((global_var(GVAR_NEW_RENO_PORN_STAR) > 0) or (dude_charisma > 6))));
     end
     else begin
         set_local_var(4, local_var(4) bwor 2048);
-        gsay_reply(445, 277 + (local_var(5) > 0));
-    end
-    giq_option(-3, 445, obj_name(dude_obj) + message_str(445, 280) + obj_name(dude_obj) + message_str(445, 281), Node007, 50);
-    giq_option(-3, 445, 282, Node999, 50);
-    if (global_var(348) == 1) then begin
-        giq_option(4, 445, 283, Node010, 50);
-        giq_option(4, 445, 284, Node019, 51);
-        giq_option(4, 445, 285, Node019, 51);
-        if ((global_var(314) bwand --2147483648) != 0) then begin
-            if ((global_var(230) / 100) == 2) then begin
-                giq_option(5, 445, message_str(445, 286) + message_str(353, 300 + (global_var(230) % 100)) + message_str(445, 5286), Node039, 50);
+        Reply(277 + (local_var(5) > 0));
+    end
+    giq_option(-3, NAME, dude_name + mstr(280) + dude_name + mstr(281), Node007, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 282, Node999, NEUTRAL_REACTION);
+    if (global_var(GVAR_NEW_RENO_STOLEN_CAR) == 1) then begin
+        giq_option(4, NAME, 283, Node010, NEUTRAL_REACTION);
+        giq_option(4, NAME, 284, Node019, BAD_REACTION);
+        giq_option(4, NAME, 285, Node019, BAD_REACTION);
+        if ((global_var(GVAR_NEW_RENO_FLAG_1) bwand --2147483648) != 0) then begin
+            if ((global_var(GVAR_NEW_RENO_MADE_MAN) / 100) == 2) then begin
+                giq_option(5, NAME, mstr(286) + message_str(SCRIPT_NEWR1, 300 + (global_var(GVAR_NEW_RENO_MADE_MAN) % 100)) + mstr(5286), Node039, NEUTRAL_REACTION);
             end
             else begin
-                if ((global_var(314) bwand --2147483648) != 0) then begin
-                    giq_option(7, 445, 287, Node990, 50);
+                if ((global_var(GVAR_NEW_RENO_FLAG_1) bwand --2147483648) != 0) then begin
+                    giq_option(7, NAME, 287, Node990, NEUTRAL_REACTION);
                 end
             end
         end
     end
     else begin
-        if (global_var(348) == 2) then begin
+        if (global_var(GVAR_NEW_RENO_STOLEN_CAR) == 2) then begin
             if ((local_var(4) bwand 1024) != 0) then begin
-                giq_option(4, 445, 288, Node989, 51);
+                giq_option(4, NAME, 288, Node989, BAD_REACTION);
             end
             else begin
-                giq_option(4, 445, 290, Node988, 50);
+                giq_option(4, NAME, 290, Node988, NEUTRAL_REACTION);
             end
-            if ((global_var(455) != 0) == 0) then begin
-                giq_option(4, 445, 289, Node024, 50);
+            if ((global_var(GVAR_NEW_RENO_CAR_UPGRADE) != 0) == 0) then begin
+                giq_option(4, NAME, 289, Node024, NEUTRAL_REACTION);
             end
-            if ((global_var(1644) == 0) and (obj_is_carrying_obj_pid(dude_obj, 837) > 0)) then begin
-                giq_option(4, 445, 2100, Node059, 50);
+            if ((global_var(GVAR_CAR_CURRENT_ELEVATION) == 0) and (obj_is_carrying_obj_pid(dude_obj, 837) > 0)) then begin
+                giq_option(4, NAME, 2100, Node059, NEUTRAL_REACTION);
             end
-            if ((global_var(1644) == 1) and (obj_is_carrying_obj_pid(dude_obj, 838) > 0)) then begin
-                giq_option(4, 445, 2101, Node060, 50);
+            if ((global_var(GVAR_CAR_CURRENT_ELEVATION) == 1) and (obj_is_carrying_obj_pid(dude_obj, 838) > 0)) then begin
+                giq_option(4, NAME, 2101, Node060, NEUTRAL_REACTION);
             end
         end
     end
     if (obj_is_carrying_obj_pid(dude_obj, 851) or obj_is_carrying_obj_pid(dude_obj, 852) or obj_is_carrying_obj_pid(dude_obj, 853) or obj_is_carrying_obj_pid(dude_obj, 854) or obj_is_carrying_obj_pid(dude_obj, 855)) then begin
-        giq_option(4, 445, 1200, Node120, 50);
+        giq_option(4, NAME, 1200, Node120, NEUTRAL_REACTION);
     end
-    giq_option(4, 445, 291, Node999, 50);
+    giq_option(4, NAME, 291, Node999, NEUTRAL_REACTION);
     prev_node := 6;
 end
 
 procedure Node007
 begin
-    if ((get_critter_stat(dude_obj, 34) == 1) and ((get_critter_stat(dude_obj, 3) > 6) or (global_var(232) > 0))) then begin
-        gsay_reply(445, 302);
-        giq_option(-3, 445, 303, Node008, 49);
-        giq_option(-3, 445, 304, Node009, 51);
+    if ((dude_gender == 1) and ((dude_charisma > 6) or (global_var(GVAR_NEW_RENO_PORN_STAR) > 0))) then begin
+        Reply(302);
+        giq_option(-3, NAME, 303, Node008, GOOD_REACTION);
+        giq_option(-3, NAME, 304, Node009, BAD_REACTION);
     end
     else begin
-        gsay_reply(445, 300 + (get_critter_stat(dude_obj, 34) == 1));
+        Reply(300 + (dude_gender == 1));
     end
-    giq_option(-3, 445, 282, Node999, 50);
-    giq_option(4, 445, 306, Node987, 49);
-    giq_option(4, 445, 307, Node009, 50);
+    giq_option(-3, NAME, 282, Node999, NEUTRAL_REACTION);
+    giq_option(4, NAME, 306, Node987, GOOD_REACTION);
+    giq_option(4, NAME, 307, Node009, NEUTRAL_REACTION);
 end
 
 procedure Node008
 begin
-    gsay_reply(445, 320);
-    giq_option(1, 445, 321, Node986, 49);
+    Reply(320);
+    giq_option(1, NAME, 321, Node986, GOOD_REACTION);
     prev_node := 8;
 end
 
 procedure Node009
 begin
-    gsay_reply(445, 330);
-    giq_option(-3, 445, 331, Node999, 50);
-    giq_option(-3, 445, 282, Node999, 50);
-    giq_option(4, 445, 333 + (get_critter_stat(dude_obj, 4) >= 6), Node999, 51);
+    Reply(330);
+    giq_option(-3, NAME, 331, Node999, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 282, Node999, NEUTRAL_REACTION);
+    giq_option(4, NAME, 333 + (dude_iq >= 6), Node999, BAD_REACTION);
     prev_node := 9;
 end
 
 procedure Node010
 begin
-    set_global_var(314, global_var(314) bwor --2147483648);
-    set_global_var(255, global_var(255) bwor 1);
-    gsay_reply(445, 340);
-    giq_option(4, 445, 341, Node011, 50);
-    giq_option(4, 445, 342 + (((global_var(255) bwand 512) != 0) == 0), Node012, 50);
-    if ((global_var(230) / 100) == 2) then begin
-        giq_option(5, 445, message_str(445, 344) + message_str(353, 300 + (global_var(230) % 100)) + message_str(445, 5286), Node039, 50);
+    set_global_var(GVAR_NEW_RENO_FLAG_1, global_var(GVAR_NEW_RENO_FLAG_1) bwor --2147483648);
+    set_global_var(GVAR_NEW_RENO_BISHOP, global_var(GVAR_NEW_RENO_BISHOP) bwor 1);
+    Reply(340);
+    giq_option(4, NAME, 341, Node011, NEUTRAL_REACTION);
+    giq_option(4, NAME, 342 + (((global_var(GVAR_NEW_RENO_BISHOP) bwand 512) != 0) == 0), Node012, NEUTRAL_REACTION);
+    if ((global_var(GVAR_NEW_RENO_MADE_MAN) / 100) == 2) then begin
+        giq_option(5, NAME, mstr(344) + message_str(SCRIPT_NEWR1, 300 + (global_var(GVAR_NEW_RENO_MADE_MAN) % 100)) + mstr(5286), Node039, NEUTRAL_REACTION);
     end
     else begin
-        if ((global_var(314) bwand --2147483648) != 0) then begin
-            giq_option(7, 445, 287, Node990, 50);
+        if ((global_var(GVAR_NEW_RENO_FLAG_1) bwand --2147483648) != 0) then begin
+            giq_option(7, NAME, 287, Node990, NEUTRAL_REACTION);
         end
     end
-    giq_option(4, 445, 346, Node019, 51);
-    giq_option(4, 445, 347, Node999, 50);
+    giq_option(4, NAME, 346, Node019, BAD_REACTION);
+    giq_option(4, NAME, 347, Node999, NEUTRAL_REACTION);
     prev_node := 10;
 end
 
 procedure Node011
 begin
-    gsay_reply(445, 360);
-    giq_option(4, 445, 361, Node012, 50);
-    if ((global_var(230) / 100) == 2) then begin
-        giq_option(5, 445, message_str(445, 362) + message_str(353, 300 + (global_var(230) % 100)) + message_str(445, 5286), Node039, 50);
+    Reply(360);
+    giq_option(4, NAME, 361, Node012, NEUTRAL_REACTION);
+    if ((global_var(GVAR_NEW_RENO_MADE_MAN) / 100) == 2) then begin
+        giq_option(5, NAME, mstr(362) + message_str(SCRIPT_NEWR1, 300 + (global_var(GVAR_NEW_RENO_MADE_MAN) % 100)) + mstr(5286), Node039, NEUTRAL_REACTION);
     end
     else begin
-        if ((global_var(314) bwand --2147483648) != 0) then begin
-            giq_option(7, 445, 287, Node990, 50);
+        if ((global_var(GVAR_NEW_RENO_FLAG_1) bwand --2147483648) != 0) then begin
+            giq_option(7, NAME, 287, Node990, NEUTRAL_REACTION);
         end
     end
-    giq_option(4, 445, 346, Node019, 51);
-    giq_option(4, 445, 365, Node999, 50);
+    giq_option(4, NAME, 346, Node019, BAD_REACTION);
+    giq_option(4, NAME, 365, Node999, NEUTRAL_REACTION);
     prev_node := 11;
 end
 
 procedure Node012
 begin
     if ((local_var(4) bwand 8192) != 0) then begin
-        gsay_reply(445, 377);
+        Reply(377);
     end
     else begin
         set_local_var(4, local_var(4) bwor 8192);
-        gsay_reply(445, 375 + (get_critter_stat(dude_obj, 34) == 1));
+        Reply(375 + (dude_gender == 1));
     end
     if (item_caps_total(dude_obj) >= 500) then begin
-        giq_option(4, 445, 380, Node983, 51);
+        giq_option(4, NAME, 380, Node983, BAD_REACTION);
         if (item_caps_total(dude_obj) >= 750) then begin
-            giq_option(4, 445, 379, Node984, 50);
+            giq_option(4, NAME, 379, Node984, NEUTRAL_REACTION);
             if (item_caps_total(dude_obj) >= 1000) then begin
-                giq_option(4, 445, 378, Node985, 49);
+                giq_option(4, NAME, 378, Node985, GOOD_REACTION);
             end
         end
     end
     else begin
-        giq_option(4, 445, 381, Node018, 50);
+        giq_option(4, NAME, 381, Node018, NEUTRAL_REACTION);
     end
-    if (get_critter_stat(dude_obj, 34) == 1) then begin
-        giq_option(4, 445, 382, Node982, 50);
+    if (dude_gender == 1) then begin
+        giq_option(4, NAME, 382, Node982, NEUTRAL_REACTION);
     end
-    giq_option(4, 445, 383, Node019, 51);
-    giq_option(4, 445, 384, Node999, 50);
+    giq_option(4, NAME, 383, Node019, BAD_REACTION);
+    giq_option(4, NAME, 384, Node999, NEUTRAL_REACTION);
     prev_node := 12;
 end
 
 procedure Node013
 begin
-    gsay_reply(445, message_str(445, 395 + (get_critter_stat(dude_obj, 34) == 1)) + message_str(445, 397));
-    giq_option(4, 445, 398, Node981, 50);
-    giq_option(4, 445, 399, Node018, 50);
-    if (get_critter_stat(dude_obj, 34) == 1) then begin
-        giq_option(4, 445, 382, Node982, 50);
+    Reply(mstr(395 + (dude_gender == 1)) + mstr(397));
+    giq_option(4, NAME, 398, Node981, NEUTRAL_REACTION);
+    giq_option(4, NAME, 399, Node018, NEUTRAL_REACTION);
+    if (dude_gender == 1) then begin
+        giq_option(4, NAME, 382, Node982, NEUTRAL_REACTION);
     end
-    giq_option(4, 445, 401, Node019, 51);
-    giq_option(4, 445, 402, Node999, 50);
+    giq_option(4, NAME, 401, Node019, BAD_REACTION);
+    giq_option(4, NAME, 402, Node999, NEUTRAL_REACTION);
     prev_node := 13;
 end
 
 procedure Node014
 begin
     return_node := "Node014";
-    gsay_reply(445, 420);
-    giq_option(4, 445, 421, Node980, 50);
-    giq_option(4, 445, 422, Node998, 51);
-    giq_option(4, 445, 423, Node999, 50);
+    Reply(420);
+    giq_option(4, NAME, 421, Node980, NEUTRAL_REACTION);
+    giq_option(4, NAME, 422, Node998, BAD_REACTION);
+    giq_option(4, NAME, 423, Node999, NEUTRAL_REACTION);
     prev_node := 14;
 end
 
 procedure Node015
 begin
-    gsay_reply(445, 420);
-    giq_option(4, 445, 436, Node019, 51);
-    giq_option(4, 445, 437, Node998, 51);
-    giq_option(4, 445, 423, Node999, 50);
+    Reply(420);
+    giq_option(4, NAME, 436, Node019, BAD_REACTION);
+    giq_option(4, NAME, 437, Node998, BAD_REACTION);
+    giq_option(4, NAME, 423, Node999, NEUTRAL_REACTION);
     prev_node := 15;
 end
 
 procedure Node016
 begin
-    gsay_reply(445, 445);
-    giq_option(4, 445, 446, Node017, 50);
-    giq_option(4, 445, 436, Node019, 51);
-    giq_option(4, 445, 437, Node998, 51);
-    giq_option(4, 445, 423, Node999, 50);
+    Reply(445);
+    giq_option(4, NAME, 446, Node017, NEUTRAL_REACTION);
+    giq_option(4, NAME, 436, Node019, BAD_REACTION);
+    giq_option(4, NAME, 437, Node998, BAD_REACTION);
+    giq_option(4, NAME, 423, Node999, NEUTRAL_REACTION);
     prev_node := 16;
 end
 
 procedure Node017
 begin
-    gsay_reply(445, 455);
-    giq_option(4, 445, 456, Node007, 50);
-    giq_option(4, 445, 457, Node019, 51);
-    giq_option(4, 445, 458, Node998, 51);
-    giq_option(4, 445, 459, Node009, 50);
+    Reply(455);
+    giq_option(4, NAME, 456, Node007, NEUTRAL_REACTION);
+    giq_option(4, NAME, 457, Node019, BAD_REACTION);
+    giq_option(4, NAME, 458, Node998, BAD_REACTION);
+    giq_option(4, NAME, 459, Node009, NEUTRAL_REACTION);
     prev_node := 17;
 end
 
 procedure Node018
 begin
-    gsay_reply(445, 470);
-    giq_option(4, 445, 471, Node019, 51);
-    giq_option(4, 445, 472, Node998, 51);
-    giq_option(4, 445, 423, Node999, 50);
+    Reply(470);
+    giq_option(4, NAME, 471, Node019, BAD_REACTION);
+    giq_option(4, NAME, 472, Node998, BAD_REACTION);
+    giq_option(4, NAME, 423, Node999, NEUTRAL_REACTION);
     prev_node := 18;
 end
 
 procedure Node019
 begin
-    gsay_reply(445, message_str(445, 485) + message_str(445, 486 + (get_critter_stat(dude_obj, 34) == 1)) + message_str(445, 488));
-    giq_option(4, 445, 489, Node998, 51);
-    giq_option(4, 445, 490, Node011, 50);
-    giq_option(6, 445, 491, Node020, 51);
-    giq_option(4, 445, 492, Node021, 50);
+    Reply(mstr(485) + mstr(486 + (dude_gender == 1)) + mstr(488));
+    giq_option(4, NAME, 489, Node998, BAD_REACTION);
+    giq_option(4, NAME, 490, Node011, NEUTRAL_REACTION);
+    giq_option(6, NAME, 491, Node020, BAD_REACTION);
+    giq_option(4, NAME, 492, Node021, NEUTRAL_REACTION);
     prev_node := 19;
 end
 
 procedure Node020
 begin
-    set_global_var(255, global_var(255) bwor 1);
-    set_global_var(314, global_var(314) bwor --2147483648);
-    gsay_reply(445, message_str(445, 500) + message_str(445, 501 + (get_critter_stat(dude_obj, 34) == 1)) + message_str(445, 503));
-    giq_option(4, 445, 504, Node998, 51);
-    giq_option(4, 445, 505, Node011, 50);
-    giq_option(4, 445, 506, Node999, 50);
+    set_global_var(GVAR_NEW_RENO_BISHOP, global_var(GVAR_NEW_RENO_BISHOP) bwor 1);
+    set_global_var(GVAR_NEW_RENO_FLAG_1, global_var(GVAR_NEW_RENO_FLAG_1) bwor --2147483648);
+    Reply(mstr(500) + mstr(501 + (dude_gender == 1)) + mstr(503));
+    giq_option(4, NAME, 504, Node998, BAD_REACTION);
+    giq_option(4, NAME, 505, Node011, NEUTRAL_REACTION);
+    giq_option(4, NAME, 506, Node999, NEUTRAL_REACTION);
     prev_node := 20;
 end
 
 procedure Node021
 begin
-    gsay_reply(445, 520);
-    giq_option(4, 445, 521, Node998, 51);
-    giq_option(4, 445, 522, Node999, 50);
+    Reply(520);
+    giq_option(4, NAME, 521, Node998, BAD_REACTION);
+    giq_option(4, NAME, 522, Node999, NEUTRAL_REACTION);
     prev_node := 21;
 end
 
 procedure Node022
 begin
-    if (global_var(348) < 2) then begin
-        set_global_var(348, 2);
+    if (global_var(GVAR_NEW_RENO_STOLEN_CAR) < 2) then begin
+        set_global_var(GVAR_NEW_RENO_STOLEN_CAR, 2);
     end
-    debug_msg("stolen_car state == " + global_var(348));
-    gsay_reply(445, 530);
-    if ((global_var(455) != 0) == 0) then begin
-        giq_option(4, 445, 531, Node024, 50);
+    debug_msg("stolen_car state == " + global_var(GVAR_NEW_RENO_STOLEN_CAR));
+    Reply(530);
+    if ((global_var(GVAR_NEW_RENO_CAR_UPGRADE) != 0) == 0) then begin
+        giq_option(4, NAME, 531, Node024, NEUTRAL_REACTION);
     end
-    giq_option(4, 445, 532, Node998, 51);
-    giq_option(4, 445, 533, Node999, 50);
+    giq_option(4, NAME, 532, Node998, BAD_REACTION);
+    giq_option(4, NAME, 533, Node999, NEUTRAL_REACTION);
     prev_node := 22;
 end
 
 procedure Node023
 begin
-    if (global_var(348) < 2) then begin
-        set_global_var(348, 2);
+    if (global_var(GVAR_NEW_RENO_STOLEN_CAR) < 2) then begin
+        set_global_var(GVAR_NEW_RENO_STOLEN_CAR, 2);
     end
-    debug_msg("stolen_car state == " + global_var(348));
-    gsay_reply(445, 540);
-    giq_option(4, 445, 541, Node024, 50);
-    giq_option(4, 445, 542, Node998, 51);
-    giq_option(4, 445, 543, Node999, 50);
+    debug_msg("stolen_car state == " + global_var(GVAR_NEW_RENO_STOLEN_CAR));
+    Reply(540);
+    giq_option(4, NAME, 541, Node024, NEUTRAL_REACTION);
+    giq_option(4, NAME, 542, Node998, BAD_REACTION);
+    giq_option(4, NAME, 543, Node999, NEUTRAL_REACTION);
     prev_node := 23;
 end
 
 procedure Node024
 begin
-    gsay_reply(445, 550 + (prev_node != 6));
+    Reply(550 + (prev_node != 6));
     if (item_caps_total(dude_obj) >= 300) then begin
-        giq_option(4, 445, 553, Node978, 50);
+        giq_option(4, NAME, 553, Node978, NEUTRAL_REACTION);
         if (item_caps_total(dude_obj) >= 500) then begin
-            giq_option(4, 445, 552, Node979, 50);
+            giq_option(4, NAME, 552, Node979, NEUTRAL_REACTION);
         end
     end
-    giq_option(4, 445, 554, Node027, 50);
+    giq_option(4, NAME, 554, Node027, NEUTRAL_REACTION);
     if (local_var(5) > 0) then begin
-        giq_option(4, 445, 555, Node034, 50);
+        giq_option(4, NAME, 555, Node034, NEUTRAL_REACTION);
     end
     else begin
-        if (get_critter_stat(dude_obj, 34) == 1) then begin
-            if ((get_critter_stat(dude_obj, 3) > 6) or (global_var(232) > 0)) then begin
-                giq_option(4, 445, 556, Node034, 49);
+        if (dude_gender == 1) then begin
+            if ((dude_charisma > 6) or (global_var(GVAR_NEW_RENO_PORN_STAR) > 0)) then begin
+                giq_option(4, NAME, 556, Node034, GOOD_REACTION);
             end
             else begin
-                giq_option(4, 445, 556, Node030, 51);
+                giq_option(4, NAME, 556, Node030, BAD_REACTION);
             end
         end
     end
-    giq_option(4, 445, 558, Node998, 51);
-    giq_option(4, 445, 559, Node999, 51);
+    giq_option(4, NAME, 558, Node998, BAD_REACTION);
+    giq_option(4, NAME, 559, Node999, BAD_REACTION);
     prev_node := 24;
 end
 
 procedure Node025
 begin
-    if ((global_var(455) != 0) == 0) then begin
-        metarule(52, 110 * metarule(53, 0) / 100);
+    if ((global_var(GVAR_NEW_RENO_CAR_UPGRADE) != 0) == 0) then begin
+        set_car_carry_amount(110 * car_carry_amount / 100);
     end
-    set_global_var(455, 1);
+    set_global_var(GVAR_NEW_RENO_CAR_UPGRADE, 1);
     gfade_out(10);
     game_time_advance(60 * (60 * 10));
     gfade_in(10);
     if (do_once == 0) then begin
-        display_msg(message_str(445, 2010));
+        display_mstr(2010);
         do_once := 1;
     end
     return_node := "Node025";
-    gsay_reply(445, 575);
-    giq_option(4, 445, 576, Node980, 50);
-    giq_option(4, 445, 577, Node999, 50);
+    Reply(575);
+    giq_option(4, NAME, 576, Node980, NEUTRAL_REACTION);
+    giq_option(4, NAME, 577, Node999, NEUTRAL_REACTION);
     prev_node := 25;
 end
 
 procedure Node026
 begin
-    gsay_reply(445, 590);
+    Reply(590);
     if (item_caps_total(dude_obj) >= 500) then begin
-        giq_option(4, 445, 552, Node979, 50);
+        giq_option(4, NAME, 552, Node979, NEUTRAL_REACTION);
     end
-    giq_option(4, 445, 554, Node027, 50);
+    giq_option(4, NAME, 554, Node027, NEUTRAL_REACTION);
     if (local_var(5) > 0) then begin
-        giq_option(4, 445, 555, Node034, 50);
+        giq_option(4, NAME, 555, Node034, NEUTRAL_REACTION);
     end
     else begin
-        if (get_critter_stat(dude_obj, 34) == 1) then begin
-            if ((get_critter_stat(dude_obj, 3) > 6) or (global_var(232) > 0)) then begin
-                giq_option(4, 445, 556, Node034, 49);
+        if (dude_gender == 1) then begin
+            if ((dude_charisma > 6) or (global_var(GVAR_NEW_RENO_PORN_STAR) > 0)) then begin
+                giq_option(4, NAME, 556, Node034, GOOD_REACTION);
             end
             else begin
-                giq_option(4, 445, 556, Node030, 51);
+                giq_option(4, NAME, 556, Node030, BAD_REACTION);
             end
         end
     end
-    giq_option(4, 445, 558, Node998, 51);
-    giq_option(4, 445, 559, Node999, 51);
-    giq_option(4, 445, 577, Node999, 50);
+    giq_option(4, NAME, 558, Node998, BAD_REACTION);
+    giq_option(4, NAME, 559, Node999, BAD_REACTION);
+    giq_option(4, NAME, 577, Node999, NEUTRAL_REACTION);
     prev_node := 26;
 end
 
 procedure Node027
 begin
     return_node := "Node027";
-    gsay_reply(445, 615);
-    giq_option(4, 445, 616, Node980, 50);
-    if (get_critter_stat(dude_obj, 34) == 1) then begin
-        if ((get_critter_stat(dude_obj, 3) > 6) or (global_var(232) > 0)) then begin
-            giq_option(4, 445, 617, Node967, 49);
+    Reply(615);
+    giq_option(4, NAME, 616, Node980, NEUTRAL_REACTION);
+    if (dude_gender == 1) then begin
+        if ((dude_charisma > 6) or (global_var(GVAR_NEW_RENO_PORN_STAR) > 0)) then begin
+            giq_option(4, NAME, 617, Node967, GOOD_REACTION);
         end
         else begin
-            giq_option(4, 445, 617, Node014, 51);
+            giq_option(4, NAME, 617, Node014, BAD_REACTION);
         end
     end
-    giq_option(4, 445, 619, Node999, 50);
+    giq_option(4, NAME, 619, Node999, NEUTRAL_REACTION);
     prev_node := 27;
 end
 
 procedure Node028
 begin
-    gsay_reply(445, 630);
-    giq_option(4, 445, 631, Node986, 49);
-    giq_option(4, 445, 632, Node999, 50);
+    Reply(630);
+    giq_option(4, NAME, 631, Node986, GOOD_REACTION);
+    giq_option(4, NAME, 632, Node999, NEUTRAL_REACTION);
     prev_node := 28;
 end
 
 procedure Node029
 begin
-    gsay_reply(445, 640 + (((local_var(4) bwand 16384) != 0) == 0));
-    giq_option(4, 445, 642, Node031, 50);
-    giq_option(4, 445, 643, Node976, 50);
+    Reply(640 + (((local_var(4) bwand 16384) != 0) == 0));
+    giq_option(4, NAME, 642, Node031, NEUTRAL_REACTION);
+    giq_option(4, NAME, 643, Node976, NEUTRAL_REACTION);
     prev_node := 29;
 end
 
 procedure Node030
 begin
-    gsay_reply(445, 650);
+    Reply(650);
     if (item_caps_total(dude_obj) >= 500) then begin
-        giq_option(4, 445, 651, Node979, 50);
+        giq_option(4, NAME, 651, Node979, NEUTRAL_REACTION);
     end
     else begin
-        giq_option(4, 445, 554, Node027, 50);
+        giq_option(4, NAME, 554, Node027, NEUTRAL_REACTION);
     end
-    giq_option(4, 445, 653, Node998, 51);
-    giq_option(4, 445, 423, Node999, 50);
+    giq_option(4, NAME, 653, Node998, BAD_REACTION);
+    giq_option(4, NAME, 423, Node999, NEUTRAL_REACTION);
     prev_node := 30;
 end
 
 procedure Node031
 begin
-    gsay_reply(445, 660 + (hold_on == 0));
-    giq_option(4, 445, 662, Node975, 49);
-    giq_option(4, 445, 663, Node007, 51);
+    Reply(660 + (hold_on == 0));
+    giq_option(4, NAME, 662, Node975, GOOD_REACTION);
+    giq_option(4, NAME, 663, Node007, BAD_REACTION);
     prev_node := 31;
     hold_on := 0;
 end
@@ -1305,116 +1305,116 @@ end
 procedure Node033
 begin
     set_local_var(4, local_var(4) bwor 2048);
-    if ((global_var(455) != 0) and (global_var(456) != 0)) then begin
-        gsay_reply(445, 670 + (local_var(5) > 0));
+    if ((global_var(GVAR_NEW_RENO_CAR_UPGRADE) != 0) and (global_var(GVAR_NEW_RENO_SUPER_CAR) != 0)) then begin
+        Reply(670 + (local_var(5) > 0));
     end
     else begin
-        gsay_reply(445, message_str(445, 670 + (local_var(5) > 0)) + message_str(445, 672));
-        giq_option(-3, 445, 673, Node054, 50);
-        if ((global_var(455) != 0) == 0) then begin
-            giq_option(4, 445, 289, Node024, 50);
+        Reply(mstr(670 + (local_var(5) > 0)) + mstr(672));
+        giq_option(-3, NAME, 673, Node054, NEUTRAL_REACTION);
+        if ((global_var(GVAR_NEW_RENO_CAR_UPGRADE) != 0) == 0) then begin
+            giq_option(4, NAME, 289, Node024, NEUTRAL_REACTION);
         end
-        if ((global_var(456) != 0) == 0) then begin
-            giq_option(4, 445, 675, Node054, 50);
+        if ((global_var(GVAR_NEW_RENO_SUPER_CAR) != 0) == 0) then begin
+            giq_option(4, NAME, 675, Node054, NEUTRAL_REACTION);
         end
     end
-    giq_option(-3, 445, 282, Node999, 50);
-    if (global_var(348) == 2) then begin
+    giq_option(-3, NAME, 282, Node999, NEUTRAL_REACTION);
+    if (global_var(GVAR_NEW_RENO_STOLEN_CAR) == 2) then begin
         if ((local_var(4) bwand 1024) != 0) then begin
-            giq_option(4, 445, 288, Node989, 51);
+            giq_option(4, NAME, 288, Node989, BAD_REACTION);
         end
         else begin
-            giq_option(4, 445, 290, Node988, 50);
+            giq_option(4, NAME, 290, Node988, NEUTRAL_REACTION);
         end
     end
-    giq_option(4, 445, 291, Node999, 50);
+    giq_option(4, NAME, 291, Node999, NEUTRAL_REACTION);
     prev_node := 33;
 end
 
 procedure Node034
 begin
     if ((local_var(4) bwand 16384) != 0) then begin
-        gsay_reply(445, 690);
+        Reply(690);
     end
     else begin
-        gsay_reply(445, 691 + (local_var(5) > 0));
+        Reply(691 + (local_var(5) > 0));
     end
-    giq_option(4, 445, 693, Node035, 50);
-    giq_option(4, 445, 643, Node972, 50);
+    giq_option(4, NAME, 693, Node035, NEUTRAL_REACTION);
+    giq_option(4, NAME, 643, Node972, NEUTRAL_REACTION);
     prev_node := 34;
 end
 
 procedure Node035
 begin
-    gsay_reply(445, 660 + (hold_on == 0));
-    giq_option(4, 445, 662, Node971, 49);
-    giq_option(4, 445, 663, Node009, 51);
+    Reply(660 + (hold_on == 0));
+    giq_option(4, NAME, 662, Node971, GOOD_REACTION);
+    giq_option(4, NAME, 663, Node009, BAD_REACTION);
     prev_node := 35;
     hold_on := 0;
 end
 
 procedure Node037
 begin
-    gsay_reply(445, 710);
-    giq_option(4, 445, 711, Node038, 51);
-    giq_option(4, 445, 712, Node999, 50);
+    Reply(710);
+    giq_option(4, NAME, 711, Node038, BAD_REACTION);
+    giq_option(4, NAME, 712, Node999, NEUTRAL_REACTION);
     prev_node := 37;
 end
 
 procedure Node038
 begin
-    gsay_reply(445, 720);
-    giq_option(4, 445, 721, Node998, 51);
+    Reply(720);
+    giq_option(4, NAME, 721, Node998, BAD_REACTION);
     prev_node := 38;
 end
 
 procedure Node039
 begin
-    if ((global_var(230) / 100) != 2) then begin
-        gsay_reply(445, 730);
-        giq_option(4, 445, 732, Node040, 50);
+    if ((global_var(GVAR_NEW_RENO_MADE_MAN) / 100) != 2) then begin
+        Reply(730);
+        giq_option(4, NAME, 732, Node040, NEUTRAL_REACTION);
     end
     else begin
-        gsay_reply(445, message_str(445, 731) + message_str(353, 300 + (global_var(230) % 100)) + message_str(445, 5731) + message_str(353, 300 + (global_var(230) % 100)) + message_str(445, 6731));
-        giq_option(4, 445, message_str(445, 733) + message_str(353, 300 + (global_var(230) % 100)) + message_str(445, 5733), Node040, 50);
+        Reply(mstr(731) + message_str(SCRIPT_NEWR1, 300 + (global_var(GVAR_NEW_RENO_MADE_MAN) % 100)) + mstr(5731) + message_str(SCRIPT_NEWR1, 300 + (global_var(GVAR_NEW_RENO_MADE_MAN) % 100)) + mstr(6731));
+        giq_option(4, NAME, mstr(733) + message_str(SCRIPT_NEWR1, 300 + (global_var(GVAR_NEW_RENO_MADE_MAN) % 100)) + mstr(5733), Node040, NEUTRAL_REACTION);
     end
-    giq_option(4, 445, 734, Node041, 50);
-    giq_option(4, 445, 735, Node999, 50);
+    giq_option(4, NAME, 734, Node041, NEUTRAL_REACTION);
+    giq_option(4, NAME, 735, Node999, NEUTRAL_REACTION);
     prev_node := 39;
 end
 
 procedure Node040
 begin
-    gsay_reply(445, 745);
-    giq_option(4, 445, 746, Node041, 50);
-    giq_option(4, 445, 747, Node999, 50);
+    Reply(745);
+    giq_option(4, NAME, 746, Node041, NEUTRAL_REACTION);
+    giq_option(4, NAME, 747, Node999, NEUTRAL_REACTION);
     prev_node := 40;
 end
 
 procedure Node041
 begin
-    gsay_reply(445, 755);
-    giq_option(4, 445, 756, Node042, 50);
-    giq_option(4, 445, 757, Node042, 51);
-    giq_option(6, 445, 758, Node046, 50);
-    giq_option(4, 445, 759, Node999, 50);
+    Reply(755);
+    giq_option(4, NAME, 756, Node042, NEUTRAL_REACTION);
+    giq_option(4, NAME, 757, Node042, BAD_REACTION);
+    giq_option(6, NAME, 758, Node046, NEUTRAL_REACTION);
+    giq_option(4, NAME, 759, Node999, NEUTRAL_REACTION);
     prev_node := 41;
 end
 
 procedure Node042
 begin
-    gsay_reply(445, 770);
+    Reply(770);
     if (item_caps_total(dude_obj) >= 900) then begin
         if (item_caps_total(dude_obj) >= 1500) then begin
             if (item_caps_total(dude_obj) >= 2700) then begin
-                giq_option(4, 445, 771, Node043, 50);
+                giq_option(4, NAME, 771, Node043, NEUTRAL_REACTION);
             end
-            giq_option(4, 445, 772, Node044, 51);
+            giq_option(4, NAME, 772, Node044, BAD_REACTION);
         end
-        giq_option(4, 445, 773, Node045, 51);
+        giq_option(4, NAME, 773, Node045, BAD_REACTION);
     end
-    giq_option(6, 445, 774, Node046, 50);
-    giq_option(4, 445, 775, Node999, 50);
+    giq_option(6, NAME, 774, Node046, NEUTRAL_REACTION);
+    giq_option(4, NAME, 775, Node999, NEUTRAL_REACTION);
     prev_node := 42;
 end
 
@@ -1422,112 +1422,112 @@ procedure Node043
 begin
     item_caps_adjust(dude_obj, -2700);
     item_caps_adjust(self_obj, 2700);
-    if (global_var(348) < 2) then begin
-        set_global_var(348, 2);
-    end
-    debug_msg("stolen_car state == " + global_var(348));
-    gsay_reply(445, 780);
-    giq_option(7, 445, 781, Node048, 51);
-    giq_option(4, 445, 782, Node049, 50);
-    giq_option(4, 445, 783, Node049, 50);
+    if (global_var(GVAR_NEW_RENO_STOLEN_CAR) < 2) then begin
+        set_global_var(GVAR_NEW_RENO_STOLEN_CAR, 2);
+    end
+    debug_msg("stolen_car state == " + global_var(GVAR_NEW_RENO_STOLEN_CAR));
+    Reply(780);
+    giq_option(7, NAME, 781, Node048, BAD_REACTION);
+    giq_option(4, NAME, 782, Node049, NEUTRAL_REACTION);
+    giq_option(4, NAME, 783, Node049, NEUTRAL_REACTION);
     prev_node := 43;
 end
 
 procedure Node044
 begin
     cap_amt := 1500;
-    gsay_reply(445, message_str(445, 790 + (get_critter_stat(dude_obj, 34) == 1)) + message_str(445, 792));
-    giq_option(4, 445, 793, Node966, 51);
-    giq_option(4, 445, 794, Node965, 51);
-    if ((has_skill(dude_obj, 14) > 50) and (get_critter_stat(dude_obj, 4) < 6)) then begin
-        giq_option(4, 445, 795, Node045, 51);
+    Reply(mstr(790 + (dude_gender == 1)) + mstr(792));
+    giq_option(4, NAME, 793, Node966, BAD_REACTION);
+    giq_option(4, NAME, 794, Node965, BAD_REACTION);
+    if ((dude_skill(SKILL_SPEECH) > 50) and (dude_iq < 6)) then begin
+        giq_option(4, NAME, 795, Node045, BAD_REACTION);
     end
     else begin
-        giq_option(4, 445, 795, Node045, 51);
+        giq_option(4, NAME, 795, Node045, BAD_REACTION);
     end
-    giq_option(4, 445, 797, Node046, 50);
-    giq_option(4, 445, 798, Node999, 50);
+    giq_option(4, NAME, 797, Node046, NEUTRAL_REACTION);
+    giq_option(4, NAME, 798, Node999, NEUTRAL_REACTION);
     prev_node := 44;
 end
 
 procedure Node045
 begin
     cap_amt := 900;
-    gsay_reply(445, message_str(445, 810 + (get_critter_stat(dude_obj, 34) == 1)) + message_str(445, 812));
-    giq_option(4, 445, 793, Node966, 51);
-    giq_option(6, 445, 797, Node046, 51);
-    giq_option(6, 445, 816, Node966, 50);
-    giq_option(4, 445, 818, Node999, 50);
+    Reply(mstr(810 + (dude_gender == 1)) + mstr(812));
+    giq_option(4, NAME, 793, Node966, BAD_REACTION);
+    giq_option(6, NAME, 797, Node046, BAD_REACTION);
+    giq_option(6, NAME, 816, Node966, NEUTRAL_REACTION);
+    giq_option(4, NAME, 818, Node999, NEUTRAL_REACTION);
     prev_node := 45;
 end
 
 procedure Node046
 begin
-    if (global_var(348) < 2) then begin
-        set_global_var(348, 2);
+    if (global_var(GVAR_NEW_RENO_STOLEN_CAR) < 2) then begin
+        set_global_var(GVAR_NEW_RENO_STOLEN_CAR, 2);
     end
-    debug_msg("stolen_car state == " + global_var(348));
-    gsay_reply(445, message_str(445, 830 + (get_critter_stat(dude_obj, 34) == 1)) + message_str(445, 832));
-    giq_option(4, 445, 793, Node047, 51);
-    giq_option(7, 445, 834, Node048, 51);
-    giq_option(4, 445, 835, Node049, 50);
+    debug_msg("stolen_car state == " + global_var(GVAR_NEW_RENO_STOLEN_CAR));
+    Reply(mstr(830 + (dude_gender == 1)) + mstr(832));
+    giq_option(4, NAME, 793, Node047, BAD_REACTION);
+    giq_option(7, NAME, 834, Node048, BAD_REACTION);
+    giq_option(4, NAME, 835, Node049, NEUTRAL_REACTION);
     prev_node := 46;
 end
 
 procedure Node047
 begin
-    if (global_var(348) < 2) then begin
-        set_global_var(348, 2);
+    if (global_var(GVAR_NEW_RENO_STOLEN_CAR) < 2) then begin
+        set_global_var(GVAR_NEW_RENO_STOLEN_CAR, 2);
     end
-    debug_msg("stolen_car state == " + global_var(348));
-    gsay_reply(445, 845);
-    giq_option(7, 445, 846, Node048, 51);
-    giq_option(4, 445, 847, Node049, 50);
-    giq_option(4, 445, 783, Node049, 50);
+    debug_msg("stolen_car state == " + global_var(GVAR_NEW_RENO_STOLEN_CAR));
+    Reply(845);
+    giq_option(7, NAME, 846, Node048, BAD_REACTION);
+    giq_option(4, NAME, 847, Node049, NEUTRAL_REACTION);
+    giq_option(4, NAME, 783, Node049, NEUTRAL_REACTION);
     prev_node := 47;
 end
 
 procedure Node048
 begin
     if (((local_var(4) bwand 1024) != 0) == 0) then begin
-        display_msg(message_str(445, 1002));
+        display_mstr(1002);
         give_exp_points(750);
-        display_msg(message_str(14, 100) + 750 + message_str(14, 101));
+        display_msg(g_mstr(100) + 750 + g_mstr(101));
     end
     set_local_var(4, local_var(4) bwor 1024);
-    gsay_reply(445, 855);
-    giq_option(4, 445, 856, Node989, 50);
-    giq_option(4, 445, 857, Node999, 50);
+    Reply(855);
+    giq_option(4, NAME, 856, Node989, NEUTRAL_REACTION);
+    giq_option(4, NAME, 857, Node999, NEUTRAL_REACTION);
     prev_node := 48;
 end
 
 procedure Node049
 begin
     return_node := "Node049";
-    gsay_reply(445, 865);
-    giq_option(4, 445, 866, Node980, 50);
-    giq_option(4, 445, 857, Node999, 50);
+    Reply(865);
+    giq_option(4, NAME, 866, Node980, NEUTRAL_REACTION);
+    giq_option(4, NAME, 857, Node999, NEUTRAL_REACTION);
     prev_node := 49;
 end
 
 procedure Node050
 begin
-    gsay_reply(445, 875);
-    giq_option(4, 445, 876, Node999, 50);
+    Reply(875);
+    giq_option(4, NAME, 876, Node999, NEUTRAL_REACTION);
     prev_node := 50;
 end
 
 procedure Node051
 begin
-    gsay_reply(445, 885);
-    giq_option(4, 445, 886, Node999, 50);
+    Reply(885);
+    giq_option(4, NAME, 886, Node999, NEUTRAL_REACTION);
     prev_node := 51;
 end
 
 procedure Node052
 begin
-    gsay_reply(445, 875);
-    giq_option(4, 445, 896, Node999, 50);
+    Reply(875);
+    giq_option(4, NAME, 896, Node999, NEUTRAL_REACTION);
     prev_node := 52;
 end
 
@@ -1535,115 +1535,115 @@ procedure Node053
 begin
     return_node := "Node053";
     hold_on := 0;
-    gsay_reply(445, 905);
-    giq_option(4, 445, 906, Node980, 49);
-    if (get_critter_stat(dude_obj, 34) == 1) then begin
-        if ((get_critter_stat(dude_obj, 3) > 6) or (global_var(232) > 0)) then begin
-            giq_option(4, 445, 907, Node031, 49);
+    Reply(905);
+    giq_option(4, NAME, 906, Node980, GOOD_REACTION);
+    if (dude_gender == 1) then begin
+        if ((dude_charisma > 6) or (global_var(GVAR_NEW_RENO_PORN_STAR) > 0)) then begin
+            giq_option(4, NAME, 907, Node031, GOOD_REACTION);
         end
         else begin
-            giq_option(4, 445, 907, Node014, 51);
+            giq_option(4, NAME, 907, Node014, BAD_REACTION);
         end
     end
-    giq_option(4, 445, 909, Node999, 50);
+    giq_option(4, NAME, 909, Node999, NEUTRAL_REACTION);
     prev_node := 53;
     hold_on := 0;
 end
 
 procedure Node054
 begin
-    gsay_reply(445, 915);
-    giq_option(-3, 445, 916, Node055, 50);
-    giq_option(-3, 445, 282, Node999, 50);
-    if (global_var(348) == 2) then begin
-        if ((global_var(455) != 0) == 0) then begin
-            giq_option(4, 445, 920, Node024, 50);
+    Reply(915);
+    giq_option(-3, NAME, 916, Node055, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 282, Node999, NEUTRAL_REACTION);
+    if (global_var(GVAR_NEW_RENO_STOLEN_CAR) == 2) then begin
+        if ((global_var(GVAR_NEW_RENO_CAR_UPGRADE) != 0) == 0) then begin
+            giq_option(4, NAME, 920, Node024, NEUTRAL_REACTION);
         end
-        if ((global_var(456) != 0) == 0) then begin
-            giq_option(4, 445, 918, Node055, 50);
+        if ((global_var(GVAR_NEW_RENO_SUPER_CAR) != 0) == 0) then begin
+            giq_option(4, NAME, 918, Node055, NEUTRAL_REACTION);
         end
         if ((local_var(4) bwand 1024) != 0) then begin
-            giq_option(4, 445, 919, Node989, 51);
+            giq_option(4, NAME, 919, Node989, BAD_REACTION);
         end
         else begin
-            giq_option(4, 445, 921, Node988, 50);
+            giq_option(4, NAME, 921, Node988, NEUTRAL_REACTION);
         end
     end
-    giq_option(4, 445, 922, Node999, 50);
+    giq_option(4, NAME, 922, Node999, NEUTRAL_REACTION);
     prev_node := 54;
 end
 
 procedure Node055
 begin
-    if ((global_var(456) != 0) == 0) then begin
+    if ((global_var(GVAR_NEW_RENO_SUPER_CAR) != 0) == 0) then begin
         gfade_out(10);
         game_time_advance(60 * (60 * 10) * 2);
-        metarule(52, 190 * metarule(53, 0) / 100);
+        set_car_carry_amount(190 * car_carry_amount / 100);
         gfade_in(10);
     end
-    set_global_var(456, 1);
-    gsay_reply(445, 930);
-    giq_option(4, 445, 931, Node999, 50);
+    set_global_var(GVAR_NEW_RENO_SUPER_CAR, 1);
+    Reply(930);
+    giq_option(4, NAME, 931, Node999, NEUTRAL_REACTION);
     prev_node := 55;
 end
 
 procedure Node056
 begin
-    float_msg(self_obj, message_str(445, random(940, 942)), 8);
+    float_msg(self_obj, mstr(random(940, 942)), FLOAT_MSG_YELLOW);
 end
 
 procedure Node057
 begin
-    float_msg(self_obj, message_str(445, random(950, 951)), 8);
+    float_msg(self_obj, mstr(random(950, 951)), FLOAT_MSG_YELLOW);
 end
 
 procedure Node058
 begin
-    float_msg(self_obj, message_str(445, 1100), 8);
+    float_msg(self_obj, mstr(1100), FLOAT_MSG_YELLOW);
 end
 
 procedure Node059
 begin
-    if ((global_var(1644) == 0) and (obj_is_carrying_obj_pid(dude_obj, 837) >= 3)) then begin
-        gsay_reply(445, 2102);
-        giq_option(4, 445, 2103, Node061, 50);
+    if ((global_var(GVAR_CAR_CURRENT_ELEVATION) == 0) and (obj_is_carrying_obj_pid(dude_obj, 837) >= 3)) then begin
+        Reply(2102);
+        giq_option(4, NAME, 2103, Node061, NEUTRAL_REACTION);
     end
     else begin
-        gsay_reply(445, message_str(445, 2112) + obj_is_carrying_obj_pid(dude_obj, 837) + message_str(445, 2113));
-        giq_option(4, 445, 2114, Node999, 50);
+        Reply(mstr(2112) + obj_is_carrying_obj_pid(dude_obj, 837) + mstr(2113));
+        giq_option(4, NAME, 2114, Node999, NEUTRAL_REACTION);
     end
 end
 
 procedure Node060
 begin
-    if ((global_var(1644) == 1) and (obj_is_carrying_obj_pid(dude_obj, 838) >= 3)) then begin
-        gsay_reply(445, 2102);
-        giq_option(4, 445, 2103, Node064, 50);
+    if ((global_var(GVAR_CAR_CURRENT_ELEVATION) == 1) and (obj_is_carrying_obj_pid(dude_obj, 838) >= 3)) then begin
+        Reply(2102);
+        giq_option(4, NAME, 2103, Node064, NEUTRAL_REACTION);
     end
     else begin
-        gsay_reply(445, message_str(445, 2112) + obj_is_carrying_obj_pid(dude_obj, 838) + message_str(445, 2113));
-        giq_option(4, 445, 2114, Node999, 50);
+        Reply(mstr(2112) + obj_is_carrying_obj_pid(dude_obj, 838) + mstr(2113));
+        giq_option(4, NAME, 2114, Node999, NEUTRAL_REACTION);
     end
 end
 
 procedure Node061
 begin
-    gsay_reply(445, 2104);
+    Reply(2104);
     if (item_caps_total(dude_obj) >= 2000) then begin
-        giq_option(4, 445, 2105, Node062, 50);
+        giq_option(4, NAME, 2105, Node062, NEUTRAL_REACTION);
     end
     else begin
-        giq_option(4, 445, 2106, Node063, 50);
+        giq_option(4, NAME, 2106, Node063, NEUTRAL_REACTION);
     end
-    giq_option(4, 445, 2107, Node999, 50);
+    giq_option(4, NAME, 2107, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node062
 begin
     variable LVar0 := 0;
     variable LVar1 := 0;
-    gsay_reply(445, 2108);
-    giq_option(4, 445, 2109, Node999, 50);
+    Reply(2108);
+    giq_option(4, NAME, 2109, Node999, NEUTRAL_REACTION);
     set_map_var(0, game_time);
     item_caps_adjust(dude_obj, -2000);
     while (LVar0 <= 3) do begin
@@ -1658,28 +1658,28 @@ end
 
 procedure Node063
 begin
-    gsay_reply(445, 2110);
-    giq_option(4, 445, 2111, Node999, 50);
+    Reply(2110);
+    giq_option(4, NAME, 2111, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node064
 begin
-    gsay_reply(445, 2104);
+    Reply(2104);
     if (item_caps_total(dude_obj) >= 2000) then begin
-        giq_option(4, 445, 2105, Node065, 50);
+        giq_option(4, NAME, 2105, Node065, NEUTRAL_REACTION);
     end
     else begin
-        giq_option(4, 445, 2106, Node066, 50);
+        giq_option(4, NAME, 2106, Node066, NEUTRAL_REACTION);
     end
-    giq_option(4, 445, 2107, Node999, 50);
+    giq_option(4, NAME, 2107, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node065
 begin
     variable LVar0 := 0;
     variable LVar1 := 0;
-    gsay_reply(445, 2108);
-    giq_option(4, 445, 2109, Node999, 50);
+    Reply(2108);
+    giq_option(4, NAME, 2109, Node999, NEUTRAL_REACTION);
     set_map_var(0, game_time);
     item_caps_adjust(dude_obj, -2000);
     while (LVar0 <= 3) do begin
@@ -1694,13 +1694,13 @@ end
 
 procedure Node066
 begin
-    gsay_reply(445, 2110);
-    giq_option(4, 445, 2111, Node999, 50);
+    Reply(2110);
+    giq_option(4, NAME, 2111, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node990
 begin
-    if (is_success(roll_vs_skill(dude_obj, 14, 0)) or (get_critter_stat(dude_obj, 3) > 7)) then begin
+    if (is_success(roll_vs_skill(dude_obj, 14, 0)) or (dude_charisma > 7)) then begin
         call Node039();
     end
     else begin
@@ -1744,20 +1744,20 @@ procedure Node987
 begin
     gfade_out(60 * 10);
     set_local_var(4, local_var(4) bwor 16384);
-    game_time_advance(get_critter_stat(dude_obj, 2) * random(4, 8) * (60 * 10));
-    set_global_var(194, global_var(194) + (((((get_critter_stat(dude_obj, 3) * 50) + (get_critter_stat(dude_obj, 2) * 25) + (get_critter_stat(dude_obj, 5) * 13) + (get_critter_stat(dude_obj, 0) * 12)) / 100) + (has_trait(0, dude_obj, 94) * 2) + has_trait(2, dude_obj, 13) + (((global_var(194) >= 100) or (global_var(232) > 0)) * 2 * (dude_obj == dude_obj))) * 1));
-    if (global_var(358) >= 9) then begin
-        set_global_var(588, 1);
+    game_time_advance(dude_endurance * random(4, 8) * (60 * 10));
+    set_global_var(GVAR_PLAYER_SEX_LEVEL, global_var(194) + (((((dude_charisma * 50) + (dude_endurance * 25) + (get_critter_stat(dude_obj, STAT_ag) * 13) + (dude_strength * 12)) / 100) + (has_trait(TRAIT_PERK, dude_obj, PERK_kama_sutra_perk) * 2) + has_trait(2, dude_obj, 13) + (((global_var(GVAR_PLAYER_SEX_LEVEL) >= 100) or (global_var(GVAR_NEW_RENO_PORN_STAR) > 0)) * 2 * (dude_obj == dude_obj))) * 1));
+    if (global_var(GVAR_SEX_COUNTER) >= 9) then begin
+        set_global_var(GVAR_SEXPERT, 1);
     end
-    if (((((get_critter_stat(dude_obj, 3) * 50) + (get_critter_stat(dude_obj, 2) * 25) + (get_critter_stat(dude_obj, 5) * 13) + (get_critter_stat(dude_obj, 0) * 12)) / 100) + (has_trait(0, dude_obj, 94) * 2) + has_trait(2, dude_obj, 13) + (((global_var(194) >= 100) or (global_var(232) > 0)) * 2 * (dude_obj == dude_obj))) >= 9) then begin
-        set_global_var(589, 1);
+    if (((((dude_charisma * 50) + (dude_endurance * 25) + (get_critter_stat(dude_obj, STAT_ag) * 13) + (dude_strength * 12)) / 100) + (has_trait(TRAIT_PERK, dude_obj, PERK_kama_sutra_perk) * 2) + has_trait(2, dude_obj, 13) + (((global_var(GVAR_PLAYER_SEX_LEVEL) >= 100) or (global_var(GVAR_NEW_RENO_PORN_STAR) > 0)) * 2 * (dude_obj == dude_obj))) >= 9) then begin
+        set_global_var(GVAR_GIGALO, 1);
     end
-    set_global_var(358, global_var(358) + 1);
-    set_local_var(5, local_var(5) + ((((get_critter_stat(dude_obj, 3) * 50) + (get_critter_stat(dude_obj, 2) * 25) + (get_critter_stat(dude_obj, 5) * 13) + (get_critter_stat(dude_obj, 0) * 12)) / 100) + (has_trait(0, dude_obj, 94) * 2) + has_trait(2, dude_obj, 13) + (((global_var(194) >= 100) or (global_var(232) > 0)) * 2 * (dude_obj == dude_obj))));
+    set_global_var(GVAR_SEX_COUNTER, global_var(GVAR_SEX_COUNTER) + 1);
+    set_local_var(5, local_var(5) + ((((dude_charisma * 50) + (dude_endurance * 25) + (get_critter_stat(dude_obj, STAT_ag) * 13) + (dude_strength * 12)) / 100) + (has_trait(TRAIT_PERK, dude_obj, PERK_kama_sutra_perk) * 2) + has_trait(2, dude_obj, 13) + (((global_var(GVAR_PLAYER_SEX_LEVEL) >= 100) or (global_var(GVAR_NEW_RENO_PORN_STAR) > 0)) * 2 * (dude_obj == dude_obj))));
     if (local_var(5) >= 50) then begin
         last_timed_obj := self_obj;
         last_timed_param := last_timed_param;
-        metarule3(100, self_obj, 20, 0);
+        rm_fixed_timer_event(self_obj, 20);
         add_timer_event(self_obj, game_ticks(5), 20);
     end
     else begin
@@ -1770,20 +1770,20 @@ procedure Node986
 begin
     gfade_out(60 * 10);
     set_local_var(4, local_var(4) bwor 16384);
-    game_time_advance(get_critter_stat(dude_obj, 2) * random(4, 8) * (60 * 10));
-    set_global_var(194, global_var(194) + (((((get_critter_stat(dude_obj, 3) * 50) + (get_critter_stat(dude_obj, 2) * 25) + (get_critter_stat(dude_obj, 5) * 13) + (get_critter_stat(dude_obj, 0) * 12)) / 100) + (has_trait(0, dude_obj, 94) * 2) + has_trait(2, dude_obj, 13) + (((global_var(194) >= 100) or (global_var(232) > 0)) * 2 * (dude_obj == dude_obj))) * 1));
-    if (global_var(358) >= 9) then begin
-        set_global_var(588, 1);
+    game_time_advance(dude_endurance * random(4, 8) * (60 * 10));
+    set_global_var(GVAR_PLAYER_SEX_LEVEL, global_var(194) + (((((dude_charisma * 50) + (dude_endurance * 25) + (get_critter_stat(dude_obj, STAT_ag) * 13) + (dude_strength * 12)) / 100) + (has_trait(TRAIT_PERK, dude_obj, PERK_kama_sutra_perk) * 2) + has_trait(2, dude_obj, 13) + (((global_var(GVAR_PLAYER_SEX_LEVEL) >= 100) or (global_var(GVAR_NEW_RENO_PORN_STAR) > 0)) * 2 * (dude_obj == dude_obj))) * 1));
+    if (global_var(GVAR_SEX_COUNTER) >= 9) then begin
+        set_global_var(GVAR_SEXPERT, 1);
     end
-    if (((((get_critter_stat(dude_obj, 3) * 50) + (get_critter_stat(dude_obj, 2) * 25) + (get_critter_stat(dude_obj, 5) * 13) + (get_critter_stat(dude_obj, 0) * 12)) / 100) + (has_trait(0, dude_obj, 94) * 2) + has_trait(2, dude_obj, 13) + (((global_var(194) >= 100) or (global_var(232) > 0)) * 2 * (dude_obj == dude_obj))) >= 9) then begin
-        set_global_var(589, 1);
+    if (((((dude_charisma * 50) + (dude_endurance * 25) + (get_critter_stat(dude_obj, STAT_ag) * 13) + (dude_strength * 12)) / 100) + (has_trait(TRAIT_PERK, dude_obj, PERK_kama_sutra_perk) * 2) + has_trait(2, dude_obj, 13) + (((global_var(GVAR_PLAYER_SEX_LEVEL) >= 100) or (global_var(GVAR_NEW_RENO_PORN_STAR) > 0)) * 2 * (dude_obj == dude_obj))) >= 9) then begin
+        set_global_var(GVAR_GIGALO, 1);
     end
-    set_global_var(358, global_var(358) + 1);
-    set_local_var(5, local_var(5) + ((((get_critter_stat(dude_obj, 3) * 50) + (get_critter_stat(dude_obj, 2) * 25) + (get_critter_stat(dude_obj, 5) * 13) + (get_critter_stat(dude_obj, 0) * 12)) / 100) + (has_trait(0, dude_obj, 94) * 2) + has_trait(2, dude_obj, 13) + (((global_var(194) >= 100) or (global_var(232) > 0)) * 2 * (dude_obj == dude_obj))));
+    set_global_var(GVAR_SEX_COUNTER, global_var(GVAR_SEX_COUNTER) + 1);
+    set_local_var(5, local_var(5) + ((((dude_charisma * 50) + (dude_endurance * 25) + (get_critter_stat(dude_obj, STAT_ag) * 13) + (dude_strength * 12)) / 100) + (has_trait(TRAIT_PERK, dude_obj, PERK_kama_sutra_perk) * 2) + has_trait(2, dude_obj, 13) + (((global_var(GVAR_PLAYER_SEX_LEVEL) >= 100) or (global_var(GVAR_NEW_RENO_PORN_STAR) > 0)) * 2 * (dude_obj == dude_obj))));
     if (local_var(5) >= 50) then begin
         last_timed_obj := self_obj;
         last_timed_param := last_timed_param;
-        metarule3(100, self_obj, 20, 0);
+        rm_fixed_timer_event(self_obj, 20);
         add_timer_event(self_obj, game_ticks(5), 20);
     end
     else begin
@@ -1825,7 +1825,7 @@ end
 
 procedure Node982
 begin
-    if ((get_critter_stat(dude_obj, 3) > 6) or (global_var(232) > 0)) then begin
+    if ((dude_charisma > 6) or (global_var(GVAR_NEW_RENO_PORN_STAR) > 0)) then begin
         call Node016();
     end
     else begin
@@ -1841,7 +1841,7 @@ end
 
 procedure Node980
 begin
-    gdialog_mod_barter(0);
+    gdialog_barter;
     call return_node();
 end
 
@@ -1898,27 +1898,27 @@ begin
     end
     gfade_out(60 * 10);
     set_local_var(4, local_var(4) bwor 16384);
-    game_time_advance(get_critter_stat(dude_obj, 2) * random(4, 8) * (60 * 10));
-    set_global_var(194, global_var(194) + (((((get_critter_stat(dude_obj, 3) * 50) + (get_critter_stat(dude_obj, 2) * 25) + (get_critter_stat(dude_obj, 5) * 13) + (get_critter_stat(dude_obj, 0) * 12)) / 100) + (has_trait(0, dude_obj, 94) * 2) + has_trait(2, dude_obj, 13) + (((global_var(194) >= 100) or (global_var(232) > 0)) * 2 * (dude_obj == dude_obj))) * 1));
-    if (global_var(358) >= 9) then begin
-        set_global_var(588, 1);
+    game_time_advance(dude_endurance * random(4, 8) * (60 * 10));
+    set_global_var(GVAR_PLAYER_SEX_LEVEL, global_var(194) + (((((dude_charisma * 50) + (dude_endurance * 25) + (get_critter_stat(dude_obj, STAT_ag) * 13) + (dude_strength * 12)) / 100) + (has_trait(TRAIT_PERK, dude_obj, PERK_kama_sutra_perk) * 2) + has_trait(2, dude_obj, 13) + (((global_var(GVAR_PLAYER_SEX_LEVEL) >= 100) or (global_var(GVAR_NEW_RENO_PORN_STAR) > 0)) * 2 * (dude_obj == dude_obj))) * 1));
+    if (global_var(GVAR_SEX_COUNTER) >= 9) then begin
+        set_global_var(GVAR_SEXPERT, 1);
     end
-    if (((((get_critter_stat(dude_obj, 3) * 50) + (get_critter_stat(dude_obj, 2) * 25) + (get_critter_stat(dude_obj, 5) * 13) + (get_critter_stat(dude_obj, 0) * 12)) / 100) + (has_trait(0, dude_obj, 94) * 2) + has_trait(2, dude_obj, 13) + (((global_var(194) >= 100) or (global_var(232) > 0)) * 2 * (dude_obj == dude_obj))) >= 9) then begin
-        set_global_var(589, 1);
+    if (((((dude_charisma * 50) + (dude_endurance * 25) + (get_critter_stat(dude_obj, STAT_ag) * 13) + (dude_strength * 12)) / 100) + (has_trait(TRAIT_PERK, dude_obj, PERK_kama_sutra_perk) * 2) + has_trait(2, dude_obj, 13) + (((global_var(GVAR_PLAYER_SEX_LEVEL) >= 100) or (global_var(GVAR_NEW_RENO_PORN_STAR) > 0)) * 2 * (dude_obj == dude_obj))) >= 9) then begin
+        set_global_var(GVAR_GIGALO, 1);
     end
-    set_global_var(358, global_var(358) + 1);
-    set_local_var(5, local_var(5) + ((((get_critter_stat(dude_obj, 3) * 50) + (get_critter_stat(dude_obj, 2) * 25) + (get_critter_stat(dude_obj, 5) * 13) + (get_critter_stat(dude_obj, 0) * 12)) / 100) + (has_trait(0, dude_obj, 94) * 2) + has_trait(2, dude_obj, 13) + (((global_var(194) >= 100) or (global_var(232) > 0)) * 2 * (dude_obj == dude_obj))));
+    set_global_var(GVAR_SEX_COUNTER, global_var(GVAR_SEX_COUNTER) + 1);
+    set_local_var(5, local_var(5) + ((((dude_charisma * 50) + (dude_endurance * 25) + (get_critter_stat(dude_obj, STAT_ag) * 13) + (dude_strength * 12)) / 100) + (has_trait(TRAIT_PERK, dude_obj, PERK_kama_sutra_perk) * 2) + has_trait(2, dude_obj, 13) + (((global_var(GVAR_PLAYER_SEX_LEVEL) >= 100) or (global_var(GVAR_NEW_RENO_PORN_STAR) > 0)) * 2 * (dude_obj == dude_obj))));
     if (local_var(5) >= 50) then begin
         last_timed_obj := self_obj;
         last_timed_param := last_timed_param;
-        metarule3(100, self_obj, 20, 0);
+        rm_fixed_timer_event(self_obj, 20);
         add_timer_event(self_obj, game_ticks(5), 20);
     end
     else begin
         call Node999();
     end
     gfade_in(60 * 10);
-    display_msg(message_str(445, 2000));
+    display_mstr(2000);
 end
 
 procedure Node972
@@ -1931,39 +1931,39 @@ procedure Node971
 begin
     gfade_out(60 * 10);
     set_local_var(4, local_var(4) bwor 16384);
-    game_time_advance(get_critter_stat(dude_obj, 2) * random(4, 8) * (60 * 10));
-    set_global_var(194, global_var(194) + (((((get_critter_stat(dude_obj, 3) * 50) + (get_critter_stat(dude_obj, 2) * 25) + (get_critter_stat(dude_obj, 5) * 13) + (get_critter_stat(dude_obj, 0) * 12)) / 100) + (has_trait(0, dude_obj, 94) * 2) + has_trait(2, dude_obj, 13) + (((global_var(194) >= 100) or (global_var(232) > 0)) * 2 * (dude_obj == dude_obj))) * 1));
-    if (global_var(358) >= 9) then begin
-        set_global_var(588, 1);
+    game_time_advance(dude_endurance * random(4, 8) * (60 * 10));
+    set_global_var(GVAR_PLAYER_SEX_LEVEL, global_var(194) + (((((dude_charisma * 50) + (dude_endurance * 25) + (get_critter_stat(dude_obj, STAT_ag) * 13) + (dude_strength * 12)) / 100) + (has_trait(TRAIT_PERK, dude_obj, PERK_kama_sutra_perk) * 2) + has_trait(2, dude_obj, 13) + (((global_var(GVAR_PLAYER_SEX_LEVEL) >= 100) or (global_var(GVAR_NEW_RENO_PORN_STAR) > 0)) * 2 * (dude_obj == dude_obj))) * 1));
+    if (global_var(GVAR_SEX_COUNTER) >= 9) then begin
+        set_global_var(GVAR_SEXPERT, 1);
     end
-    if (((((get_critter_stat(dude_obj, 3) * 50) + (get_critter_stat(dude_obj, 2) * 25) + (get_critter_stat(dude_obj, 5) * 13) + (get_critter_stat(dude_obj, 0) * 12)) / 100) + (has_trait(0, dude_obj, 94) * 2) + has_trait(2, dude_obj, 13) + (((global_var(194) >= 100) or (global_var(232) > 0)) * 2 * (dude_obj == dude_obj))) >= 9) then begin
-        set_global_var(589, 1);
+    if (((((dude_charisma * 50) + (dude_endurance * 25) + (get_critter_stat(dude_obj, STAT_ag) * 13) + (dude_strength * 12)) / 100) + (has_trait(TRAIT_PERK, dude_obj, PERK_kama_sutra_perk) * 2) + has_trait(2, dude_obj, 13) + (((global_var(GVAR_PLAYER_SEX_LEVEL) >= 100) or (global_var(GVAR_NEW_RENO_PORN_STAR) > 0)) * 2 * (dude_obj == dude_obj))) >= 9) then begin
+        set_global_var(GVAR_GIGALO, 1);
     end
-    set_global_var(358, global_var(358) + 1);
-    set_local_var(5, local_var(5) + ((((get_critter_stat(dude_obj, 3) * 50) + (get_critter_stat(dude_obj, 2) * 25) + (get_critter_stat(dude_obj, 5) * 13) + (get_critter_stat(dude_obj, 0) * 12)) / 100) + (has_trait(0, dude_obj, 94) * 2) + has_trait(2, dude_obj, 13) + (((global_var(194) >= 100) or (global_var(232) > 0)) * 2 * (dude_obj == dude_obj))));
+    set_global_var(GVAR_SEX_COUNTER, global_var(GVAR_SEX_COUNTER) + 1);
+    set_local_var(5, local_var(5) + ((((dude_charisma * 50) + (dude_endurance * 25) + (get_critter_stat(dude_obj, STAT_ag) * 13) + (dude_strength * 12)) / 100) + (has_trait(TRAIT_PERK, dude_obj, PERK_kama_sutra_perk) * 2) + has_trait(2, dude_obj, 13) + (((global_var(GVAR_PLAYER_SEX_LEVEL) >= 100) or (global_var(GVAR_NEW_RENO_PORN_STAR) > 0)) * 2 * (dude_obj == dude_obj))));
     if (local_var(5) >= 50) then begin
         last_timed_obj := self_obj;
         last_timed_param := last_timed_param;
-        metarule3(100, self_obj, 20, 0);
+        rm_fixed_timer_event(self_obj, 20);
         add_timer_event(self_obj, game_ticks(5), 20);
     end
     else begin
         call Node999();
     end
     gfade_in(60 * 10);
-    if ((global_var(455) != 0) == 0) then begin
-        metarule(52, 110 * metarule(53, 0) / 100);
+    if ((global_var(GVAR_NEW_RENO_CAR_UPGRADE) != 0) == 0) then begin
+        set_car_carry_amount(110 * car_carry_amount / 100);
     end
-    set_global_var(455, 1);
+    set_global_var(GVAR_NEW_RENO_CAR_UPGRADE, 1);
     gfade_out(10);
     game_time_advance(60 * (60 * 10));
     gfade_in(10);
-    display_msg(message_str(445, 2010));
+    display_mstr(2010);
 end
 
 procedure Node967
 begin
-    if ((((obj_is_carrying_obj_pid(self_obj, 38) <= 0) and (obj_is_carrying_obj_pid(self_obj, 39) <= 0)) == 0) and ((get_critter_stat(dude_obj, 3) > 6) or (global_var(232) > 0))) then begin
+    if ((((obj_is_carrying_obj_pid(self_obj, 38) <= 0) and (obj_is_carrying_obj_pid(self_obj, 39) <= 0)) == 0) and ((dude_charisma > 6) or (global_var(GVAR_NEW_RENO_PORN_STAR) > 0))) then begin
         call Node029();
     end
     else begin
@@ -1987,103 +1987,103 @@ end
 
 procedure Node120
 begin
-    gsay_reply(445, 1201);
+    Reply(1201);
     if (obj_is_carrying_obj_pid(dude_obj, 851)) then begin
-        giq_option(4, 445, 1202, Node121, 50);
+        giq_option(4, NAME, 1202, Node121, NEUTRAL_REACTION);
     end
     if (obj_is_carrying_obj_pid(dude_obj, 852)) then begin
-        giq_option(4, 445, 1202, Node122, 50);
+        giq_option(4, NAME, 1202, Node122, NEUTRAL_REACTION);
     end
     if (obj_is_carrying_obj_pid(dude_obj, 853)) then begin
-        giq_option(4, 445, 1203, Node123, 50);
+        giq_option(4, NAME, 1203, Node123, NEUTRAL_REACTION);
     end
     if (obj_is_carrying_obj_pid(dude_obj, 854)) then begin
-        giq_option(4, 445, 1202, Node124, 50);
+        giq_option(4, NAME, 1202, Node124, NEUTRAL_REACTION);
     end
     if (obj_is_carrying_obj_pid(dude_obj, 855)) then begin
-        giq_option(4, 445, 1202, Node125, 50);
+        giq_option(4, NAME, 1202, Node125, NEUTRAL_REACTION);
     end
-    giq_option(4, 445, 1216, Node999, 50);
+    giq_option(4, NAME, 1216, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node121
 begin
-    gsay_reply(445, 1204);
+    Reply(1204);
     if (item_caps_total(dude_obj) < 1500) then begin
-        giq_option(4, 445, 1217, Node999, 50);
+        giq_option(4, NAME, 1217, Node999, NEUTRAL_REACTION);
     end
     if (item_caps_total(dude_obj) > 1499) then begin
-        giq_option(4, 445, 1210, Node126, 50);
+        giq_option(4, NAME, 1210, Node126, NEUTRAL_REACTION);
     end
-    giq_option(4, 445, 1209, Node131, 50);
+    giq_option(4, NAME, 1209, Node131, NEUTRAL_REACTION);
     if (item_caps_total(dude_obj) > 1499) then begin
-        giq_option(4, 445, 1213, Node126, 50);
+        giq_option(4, NAME, 1213, Node126, NEUTRAL_REACTION);
     end
-    giq_option(4, 445, 1211, Node999, 50);
+    giq_option(4, NAME, 1211, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node122
 begin
-    gsay_reply(445, 1205);
+    Reply(1205);
     if (item_caps_total(dude_obj) < 1500) then begin
-        giq_option(4, 445, 1217, Node999, 50);
+        giq_option(4, NAME, 1217, Node999, NEUTRAL_REACTION);
     end
     if (item_caps_total(dude_obj) > 1499) then begin
-        giq_option(4, 445, 1210, Node127, 50);
+        giq_option(4, NAME, 1210, Node127, NEUTRAL_REACTION);
     end
-    giq_option(4, 445, 1209, Node132, 50);
+    giq_option(4, NAME, 1209, Node132, NEUTRAL_REACTION);
     if (item_caps_total(dude_obj) > 1499) then begin
-        giq_option(4, 445, 1213, Node127, 50);
+        giq_option(4, NAME, 1213, Node127, NEUTRAL_REACTION);
     end
-    giq_option(4, 445, 1216, Node999, 50);
+    giq_option(4, NAME, 1216, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node123
 begin
-    gsay_reply(445, 1208);
+    Reply(1208);
     if (item_caps_total(dude_obj) < 500) then begin
-        giq_option(4, 445, 1217, Node999, 50);
+        giq_option(4, NAME, 1217, Node999, NEUTRAL_REACTION);
     end
     if (item_caps_total(dude_obj) > 499) then begin
-        giq_option(4, 445, 1210, Node128, 50);
+        giq_option(4, NAME, 1210, Node128, NEUTRAL_REACTION);
     end
-    giq_option(4, 445, 1209, Node133, 50);
+    giq_option(4, NAME, 1209, Node133, NEUTRAL_REACTION);
     if (item_caps_total(dude_obj) > 499) then begin
-        giq_option(4, 445, 1213, Node128, 50);
+        giq_option(4, NAME, 1213, Node128, NEUTRAL_REACTION);
     end
-    giq_option(4, 445, 1211, Node999, 50);
+    giq_option(4, NAME, 1211, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node124
 begin
-    gsay_reply(445, 1206);
+    Reply(1206);
     if (item_caps_total(dude_obj) < 1800) then begin
-        giq_option(4, 445, 1217, Node999, 50);
+        giq_option(4, NAME, 1217, Node999, NEUTRAL_REACTION);
     end
     if (item_caps_total(dude_obj) > 1799) then begin
-        giq_option(4, 445, 1210, Node129, 50);
+        giq_option(4, NAME, 1210, Node129, NEUTRAL_REACTION);
     end
-    giq_option(4, 445, 1209, Node134, 50);
+    giq_option(4, NAME, 1209, Node134, NEUTRAL_REACTION);
     if (item_caps_total(dude_obj) > 1799) then begin
-        giq_option(4, 445, 1213, Node129, 50);
+        giq_option(4, NAME, 1213, Node129, NEUTRAL_REACTION);
     end
-    giq_option(4, 445, 1216, Node999, 50);
+    giq_option(4, NAME, 1216, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node125
 begin
-    gsay_reply(445, 1207);
+    Reply(1207);
     if (item_caps_total(dude_obj) < 1800) then begin
-        giq_option(4, 445, 1217, Node999, 50);
+        giq_option(4, NAME, 1217, Node999, NEUTRAL_REACTION);
     end
     if (item_caps_total(dude_obj) > 1799) then begin
-        giq_option(4, 445, 1210, Node130, 50);
+        giq_option(4, NAME, 1210, Node130, NEUTRAL_REACTION);
     end
-    giq_option(4, 445, 1209, Node135, 50);
+    giq_option(4, NAME, 1209, Node135, NEUTRAL_REACTION);
     if (item_caps_total(dude_obj) > 1799) then begin
-        giq_option(4, 445, 1213, Node130, 50);
+        giq_option(4, NAME, 1213, Node130, NEUTRAL_REACTION);
     end
-    giq_option(4, 445, 1211, Node999, 50);
+    giq_option(4, NAME, 1211, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node126
@@ -2094,7 +2094,7 @@ begin
     game_time_advance(3 * (60 * (60 * 10)));
     item_caps_adjust(dude_obj, -1500);
     item_caps_adjust(self_obj, 1500);
-    LVar0 := create_object_sid(857, 0, 0, -1);
+    LVar0 := create_object(857, 0, 0);
     add_obj_to_inven(dude_obj, LVar0);
     LVar1 := obj_carrying_pid_obj(dude_obj, 851);
     rm_obj_from_inven(dude_obj, LVar1);
@@ -2116,7 +2116,7 @@ begin
     game_time_advance(3 * (60 * (60 * 10)));
     item_caps_adjust(dude_obj, -1500);
     item_caps_adjust(self_obj, 1500);
-    LVar0 := create_object_sid(858, 0, 0, -1);
+    LVar0 := create_object(858, 0, 0);
     add_obj_to_inven(dude_obj, LVar0);
     LVar1 := obj_carrying_pid_obj(dude_obj, 852);
     rm_obj_from_inven(dude_obj, LVar1);
@@ -2138,7 +2138,7 @@ begin
     game_time_advance(60 * (60 * 10));
     item_caps_adjust(dude_obj, -500);
     item_caps_adjust(self_obj, 500);
-    LVar0 := create_object_sid(856, 0, 0, -1);
+    LVar0 := create_object(856, 0, 0);
     add_obj_to_inven(dude_obj, LVar0);
     LVar1 := obj_carrying_pid_obj(dude_obj, 853);
     rm_obj_from_inven(dude_obj, LVar1);
@@ -2160,7 +2160,7 @@ begin
     game_time_advance(3 * (60 * (60 * 10)));
     item_caps_adjust(dude_obj, -1800);
     item_caps_adjust(self_obj, 1800);
-    LVar0 := create_object_sid(859, 0, 0, -1);
+    LVar0 := create_object(859, 0, 0);
     add_obj_to_inven(dude_obj, LVar0);
     LVar1 := obj_carrying_pid_obj(dude_obj, 854);
     rm_obj_from_inven(dude_obj, LVar1);
@@ -2182,7 +2182,7 @@ begin
     game_time_advance(3 * (60 * (60 * 10)));
     item_caps_adjust(dude_obj, -1800);
     item_caps_adjust(self_obj, 1800);
-    LVar0 := create_object_sid(860, 0, 0, -1);
+    LVar0 := create_object(860, 0, 0);
     add_obj_to_inven(dude_obj, LVar0);
     LVar1 := obj_carrying_pid_obj(dude_obj, 855);
     rm_obj_from_inven(dude_obj, LVar1);
@@ -2198,76 +2198,75 @@ end
 
 procedure Node131
 begin
-    gsay_reply(445, 1212);
+    Reply(1212);
     if (item_caps_total(dude_obj) < 1500) then begin
-        giq_option(4, 445, 1217, Node999, 50);
+        giq_option(4, NAME, 1217, Node999, NEUTRAL_REACTION);
     end
     if (item_caps_total(dude_obj) > 1499) then begin
-        giq_option(4, 445, 1210, Node126, 50);
+        giq_option(4, NAME, 1210, Node126, NEUTRAL_REACTION);
     end
     if (item_caps_total(dude_obj) > 1499) then begin
-        giq_option(4, 445, 1213, Node126, 50);
+        giq_option(4, NAME, 1213, Node126, NEUTRAL_REACTION);
     end
-    giq_option(4, 445, 1211, Node999, 50);
+    giq_option(4, NAME, 1211, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node132
 begin
-    gsay_reply(445, 1212);
+    Reply(1212);
     if (item_caps_total(dude_obj) < 1500) then begin
-        giq_option(4, 445, 1217, Node999, 50);
+        giq_option(4, NAME, 1217, Node999, NEUTRAL_REACTION);
     end
     if (item_caps_total(dude_obj) > 1499) then begin
-        giq_option(4, 445, 1210, Node127, 50);
+        giq_option(4, NAME, 1210, Node127, NEUTRAL_REACTION);
     end
     if (item_caps_total(dude_obj) > 1499) then begin
-        giq_option(4, 445, 1213, Node127, 50);
+        giq_option(4, NAME, 1213, Node127, NEUTRAL_REACTION);
     end
-    giq_option(4, 445, 1216, Node999, 50);
+    giq_option(4, NAME, 1216, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node133
 begin
-    gsay_reply(445, 1212);
+    Reply(1212);
     if (item_caps_total(dude_obj) < 500) then begin
-        giq_option(4, 445, 1217, Node999, 50);
+        giq_option(4, NAME, 1217, Node999, NEUTRAL_REACTION);
     end
     if (item_caps_total(dude_obj) > 499) then begin
-        giq_option(4, 445, 1210, Node128, 50);
+        giq_option(4, NAME, 1210, Node128, NEUTRAL_REACTION);
     end
     if (item_caps_total(dude_obj) > 499) then begin
-        giq_option(4, 445, 1213, Node128, 50);
+        giq_option(4, NAME, 1213, Node128, NEUTRAL_REACTION);
     end
-    giq_option(4, 445, 1211, Node999, 50);
+    giq_option(4, NAME, 1211, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node134
 begin
-    gsay_reply(445, 1212);
+    Reply(1212);
     if (item_caps_total(dude_obj) < 1800) then begin
-        giq_option(4, 445, 1217, Node999, 50);
+        giq_option(4, NAME, 1217, Node999, NEUTRAL_REACTION);
     end
     if (item_caps_total(dude_obj) > 1799) then begin
-        giq_option(4, 445, 1210, Node129, 50);
+        giq_option(4, NAME, 1210, Node129, NEUTRAL_REACTION);
     end
     if (item_caps_total(dude_obj) > 1799) then begin
-        giq_option(4, 445, 1213, Node129, 50);
+        giq_option(4, NAME, 1213, Node129, NEUTRAL_REACTION);
     end
-    giq_option(4, 445, 1216, Node999, 50);
+    giq_option(4, NAME, 1216, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node135
 begin
-    gsay_reply(445, 1212);
+    Reply(1212);
     if (item_caps_total(dude_obj) < 1800) then begin
-        giq_option(4, 445, 1217, Node999, 50);
+        giq_option(4, NAME, 1217, Node999, NEUTRAL_REACTION);
     end
     if (item_caps_total(dude_obj) > 1799) then begin
-        giq_option(4, 445, 1210, Node130, 50);
+        giq_option(4, NAME, 1210, Node130, NEUTRAL_REACTION);
     end
     if (item_caps_total(dude_obj) > 1799) then begin
-        giq_option(4, 445, 1213, Node130, 50);
+        giq_option(4, NAME, 1213, Node130, NEUTRAL_REACTION);
     end
-    giq_option(4, 445, 1211, Node999, 50);
+    giq_option(4, NAME, 1211, Node999, NEUTRAL_REACTION);
 end
-
