diff --git a/Mods/MegaMod/Original/NEWR1.ssl b/Mods/MegaMod/ReDefine/NEWR1.ssl
index 21b7129..222b78d 100644
--- a/Mods/MegaMod/Original/NEWR1.ssl
+++ b/Mods/MegaMod/ReDefine/NEWR1.ssl
@@ -164,53 +164,53 @@ end
 procedure ChooseFromList
 begin
     variable LVar0 := 0;
-    gsay_reply(353, stored_msg);
-    if (message_str(353, top_msg - 1) != "") then begin
-        giq_option(1, 353, message_str(353, 150), CLICK_PREV, 50);
+    Reply(stored_msg);
+    if (mstr(top_msg - 1) != "") then begin
+        giq_option(1, NAME, mstr(150), CLICK_PREV, NEUTRAL_REACTION);
     end
     if (LVar0 == 0) then begin
-        if (message_str(353, top_msg + 0) != "") then begin
-            giq_option(1, 353, message_str(353, top_msg + 0), CLICK_0, 50);
+        if (mstr(top_msg + 0) != "") then begin
+            giq_option(1, NAME, mstr(top_msg + 0), CLICK_0, NEUTRAL_REACTION);
         end
         else begin
             LVar0 := 1;
         end
     end
     if (LVar0 == 0) then begin
-        if (message_str(353, top_msg + 1) != "") then begin
-            giq_option(1, 353, message_str(353, top_msg + 1), CLICK_1, 50);
+        if (mstr(top_msg + 1) != "") then begin
+            giq_option(1, NAME, mstr(top_msg + 1), CLICK_1, NEUTRAL_REACTION);
         end
         else begin
             LVar0 := 1;
         end
     end
     if (LVar0 == 0) then begin
-        if (message_str(353, top_msg + 2) != "") then begin
-            giq_option(1, 353, message_str(353, top_msg + 2), CLICK_2, 50);
+        if (mstr(top_msg + 2) != "") then begin
+            giq_option(1, NAME, mstr(top_msg + 2), CLICK_2, NEUTRAL_REACTION);
         end
         else begin
             LVar0 := 1;
         end
     end
     if (LVar0 == 0) then begin
-        if (message_str(353, top_msg + 3) != "") then begin
-            giq_option(1, 353, message_str(353, top_msg + 3), CLICK_3, 50);
+        if (mstr(top_msg + 3) != "") then begin
+            giq_option(1, NAME, mstr(top_msg + 3), CLICK_3, NEUTRAL_REACTION);
         end
         else begin
             LVar0 := 1;
         end
     end
     if (LVar0 == 0) then begin
-        if (message_str(353, top_msg + 4) != "") then begin
-            giq_option(1, 353, message_str(353, top_msg + 4), CLICK_4, 50);
+        if (mstr(top_msg + 4) != "") then begin
+            giq_option(1, NAME, mstr(top_msg + 4), CLICK_4, NEUTRAL_REACTION);
         end
         else begin
             LVar0 := 1;
         end
     end
     if (LVar0 == 0) then begin
-        if (message_str(353, top_msg + 5) != "") then begin
-            giq_option(1, 353, message_str(353, 151), CLICK_MORE, 50);
+        if (mstr(top_msg + 5) != "") then begin
+            giq_option(1, NAME, mstr(151), CLICK_MORE, NEUTRAL_REACTION);
         end
     end
 end
@@ -264,29 +264,29 @@ end
 procedure map_enter_p_proc
 begin
     variable LVar0 := 0;
-    if (metarule(22, 0) == 0) then begin
-        if (((global_var(495) bwand 131072) != 0) == 0) then begin
-            set_global_var(495, global_var(495) bwor 131072);
-            if ((global_var(518) == 0) and (global_var(383) == 1)) then begin
-                LVar0 := create_object_sid(16777252, 0, 0, 811);
+    if (is_loading_game == 0) then begin
+        if (((global_var(GVAR_NEW_RENO_FLAG_3) bwand 131072) != 0) == 0) then begin
+            set_global_var(GVAR_NEW_RENO_FLAG_3, global_var(GVAR_NEW_RENO_FLAG_3) bwor 131072);
+            if ((global_var(GVAR_SAVINE_PAYOUT) == 0) and (global_var(GVAR_SAVINELLI_STATUS) == 1)) then begin
+                LVar0 := create_object_sid(16777252, 0, 0, SCRIPT_RCNORD);
                 critter_attempt_placement(LVar0, 27507, 0);
             end
         end
-        if (metarule(14, 0)) then begin
-            display_msg(message_str(353, 100));
+        if (map_first_run) then begin
+            display_mstr(100);
         end
-        if ((global_var(495) bwand 67108864) != 0) then begin
-            set_global_var(495, global_var(495) bwand (-1 - 67108864));
-            debug_msg("stolen_car state == " + global_var(348));
+        if ((global_var(GVAR_NEW_RENO_FLAG_3) bwand 67108864) != 0) then begin
+            set_global_var(GVAR_NEW_RENO_FLAG_3, global_var(GVAR_NEW_RENO_FLAG_3) bwand (-1 - 67108864));
+            debug_msg("stolen_car state == " + global_var(GVAR_NEW_RENO_STOLEN_CAR));
         end
     end
-    set_global_var(46, 33107);
-    set_global_var(67, 6105);
-    if (elevation(dude_obj) == 1) then begin
+    set_global_var(GVAR_MAP_EXIT_TILE, 33107);
+    set_global_var(GVAR_MAP_NEXT_TILE, 6105);
+    if (dude_elevation == 1) then begin
         set_light_level(50);
     end
     else begin
-        if (elevation(dude_obj) == 2) then begin
+        if (dude_elevation == 2) then begin
             set_light_level(60);
         end
         else begin
@@ -372,20 +372,20 @@ end
 
 procedure map_exit_p_proc
 begin
-    if ((global_var(18) != 0) and (metarule(30, 0) == 7) and (global_var(1176) == 0) and (global_var(1645) == cur_map_index) and (local_var(0) != 1) and (global_var(348) == 0)) then begin
-        set_global_var(348, 1);
-        set_global_var(633, -1);
+    if ((global_var(GVAR_PLAYER_GOT_CAR) != 0) and (car_current_town == 7) and (global_var(GVAR_BREAKDOWN_TIMER) == 0) and (global_var(GVAR_CAR_CURRENT_TOWN) == cur_map_index) and (local_var(0) != 1) and (global_var(GVAR_NEW_RENO_STOLEN_CAR) == 0)) then begin
+        set_global_var(GVAR_NEW_RENO_STOLEN_CAR, 1);
+        set_global_var(GVAR_CAR_PLACED_TILE, -1);
         set_local_var(0, 1);
     end
 end
 
 procedure map_update_p_proc
 begin
-    if (elevation(dude_obj) == 1) then begin
+    if (dude_elevation == 1) then begin
         set_light_level(50);
     end
     else begin
-        if (elevation(dude_obj) == 2) then begin
+        if (dude_elevation == 2) then begin
             set_light_level(60);
         end
         else begin
@@ -468,4 +468,3 @@ begin
         end
     end
 end
-
