diff --git a/Mods/MegaMod/Original/NEWRGO.ssl b/Mods/MegaMod/ReDefine/NEWRGO.ssl
index 0f3ba2f..3b1d6d6 100644
--- a/Mods/MegaMod/Original/NEWRGO.ssl
+++ b/Mods/MegaMod/ReDefine/NEWRGO.ssl
@@ -155,53 +155,53 @@ end
 procedure ChooseFromList
 begin
     variable LVar0 := 0;
-    gsay_reply(359, stored_msg);
-    if (message_str(353, top_msg - 1) != "") then begin
-        giq_option(1, 359, message_str(353, 150), CLICK_PREV, 50);
+    Reply(stored_msg);
+    if (message_str(SCRIPT_NEWR1, top_msg - 1) != "") then begin
+        giq_option(1, NAME, message_str(SCRIPT_NEWR1, 150), CLICK_PREV, NEUTRAL_REACTION);
     end
     if (LVar0 == 0) then begin
-        if (message_str(353, top_msg + 0) != "") then begin
-            giq_option(1, 359, message_str(353, top_msg + 0), CLICK_0, 50);
+        if (message_str(SCRIPT_NEWR1, top_msg + 0) != "") then begin
+            giq_option(1, NAME, message_str(SCRIPT_NEWR1, top_msg + 0), CLICK_0, NEUTRAL_REACTION);
         end
         else begin
             LVar0 := 1;
         end
     end
     if (LVar0 == 0) then begin
-        if (message_str(353, top_msg + 1) != "") then begin
-            giq_option(1, 359, message_str(353, top_msg + 1), CLICK_1, 50);
+        if (message_str(SCRIPT_NEWR1, top_msg + 1) != "") then begin
+            giq_option(1, NAME, message_str(SCRIPT_NEWR1, top_msg + 1), CLICK_1, NEUTRAL_REACTION);
         end
         else begin
             LVar0 := 1;
         end
     end
     if (LVar0 == 0) then begin
-        if (message_str(353, top_msg + 2) != "") then begin
-            giq_option(1, 359, message_str(353, top_msg + 2), CLICK_2, 50);
+        if (message_str(SCRIPT_NEWR1, top_msg + 2) != "") then begin
+            giq_option(1, NAME, message_str(SCRIPT_NEWR1, top_msg + 2), CLICK_2, NEUTRAL_REACTION);
         end
         else begin
             LVar0 := 1;
         end
     end
     if (LVar0 == 0) then begin
-        if (message_str(353, top_msg + 3) != "") then begin
-            giq_option(1, 359, message_str(353, top_msg + 3), CLICK_3, 50);
+        if (message_str(SCRIPT_NEWR1, top_msg + 3) != "") then begin
+            giq_option(1, NAME, message_str(SCRIPT_NEWR1, top_msg + 3), CLICK_3, NEUTRAL_REACTION);
         end
         else begin
             LVar0 := 1;
         end
     end
     if (LVar0 == 0) then begin
-        if (message_str(353, top_msg + 4) != "") then begin
-            giq_option(1, 359, message_str(353, top_msg + 4), CLICK_4, 50);
+        if (message_str(SCRIPT_NEWR1, top_msg + 4) != "") then begin
+            giq_option(1, NAME, message_str(SCRIPT_NEWR1, top_msg + 4), CLICK_4, NEUTRAL_REACTION);
         end
         else begin
             LVar0 := 1;
         end
     end
     if (LVar0 == 0) then begin
-        if (message_str(353, top_msg + 5) != "") then begin
-            giq_option(1, 359, message_str(353, 151), CLICK_MORE, 50);
+        if (message_str(SCRIPT_NEWR1, top_msg + 5) != "") then begin
+            giq_option(1, NAME, message_str(SCRIPT_NEWR1, 151), CLICK_MORE, NEUTRAL_REACTION);
         end
     end
 end
@@ -255,25 +255,25 @@ end
 procedure map_enter_p_proc
 begin
     debug_msg("map_enter_p_proc");
-    debug_msg("  get_escape(escape_follow) == " + (global_var(318) == 3));
+    debug_msg("  get_escape(escape_follow) == " + (global_var(GVAR_NEW_RENO_ESCAPE) == 3));
     debug_msg("  Lloyd_In_Party == " + (party_member_obj(16777481) != 0));
-    if (metarule(22, 0) == 0) then begin
-        if (metarule(14, 0)) then begin
-            if (metarule(17, 30) == 0) then begin
+    if (is_loading_game == 0) then begin
+        if (map_first_run) then begin
+            if (town_known(30) == 0) then begin
                 debug_msg("  mark_on_map(" + 30 + ")");
                 mark_area_known(0, 30, 1);
             end
-            display_msg(message_str(353, 101));
+            display_msg(message_str(SCRIPT_NEWR1, 101));
         end
-        if (((global_var(318) == 3) == 0) and (party_member_obj(16777481) != 0)) then begin
+        if (((global_var(GVAR_NEW_RENO_ESCAPE) == 3) == 0) and (party_member_obj(16777481) != 0)) then begin
             override_map_start(13467 % 200, 13467 / 200, 0, 5);
         end
         else begin
             override_map_start(28301 % 200, 28301 / 200, 0, 5);
         end
     end
-    set_global_var(46, 33709);
-    set_global_var(67, 6296);
+    set_global_var(GVAR_MAP_EXIT_TILE, 33709);
+    set_global_var(GVAR_MAP_NEXT_TILE, 6296);
     if (elevation(dude_obj) == 1) then begin
         set_light_level(50);
     end
@@ -441,4 +441,3 @@ begin
         end
     end
 end
-
