diff --git a/Mods/MegaMod/Original/NEWR2.ssl b/Mods/MegaMod/ReDefine/NEWR2.ssl
index 6271aef..7c216ba 100644
--- a/Mods/MegaMod/Original/NEWR2.ssl
+++ b/Mods/MegaMod/ReDefine/NEWR2.ssl
@@ -180,8 +180,8 @@ end
 
 procedure map_enter_p_proc
 begin
-    set_global_var(46, 34103);
-    set_global_var(67, 6106);
+    set_global_var(GVAR_MAP_EXIT_TILE, 34103);
+    set_global_var(GVAR_MAP_NEXT_TILE, 6106);
     if (elevation(dude_obj) == 1) then begin
         set_light_level(60);
     end
@@ -348,11 +348,11 @@ begin
             end
         end
     end
-    if ((global_var(632) bwand 262144) != 0) then begin
+    if ((global_var(GVAR_PARTY_MEMBERS_HIDDEN) bwand 262144) != 0) then begin
         debug_msg("all_party_hidden: elevation == " + elevation(dude_obj));
         if (elevation(dude_obj) == 1) then begin
-            set_global_var(632, global_var(632) bwand (-1 - 262144));
-            if (global_var(769)) then begin
+            set_global_var(GVAR_PARTY_MEMBERS_HIDDEN, global_var(GVAR_PARTY_MEMBERS_HIDDEN) bwand (-1 - 262144));
+            if (global_var(GVAR_BOS_LOCATION_ELEVATORS)) then begin
                 set_obj_visibility(party_member_obj(16777278), 0);
                 set_obj_visibility(party_member_obj(16777376), 0);
                 set_obj_visibility(party_member_obj(16777377), 0);
@@ -380,204 +380,203 @@ begin
                 set_obj_visibility(party_member_obj(16777747), 0);
                 set_obj_visibility(party_member_obj(16777785), 0);
                 set_obj_visibility(party_member_obj(16777801), 0);
-                set_global_var(769, 0);
+                set_global_var(GVAR_BOS_LOCATION_ELEVATORS, 0);
             end
             if (party_member_obj(16777278) != 0) then begin
-                if (has_trait(1, party_member_obj(16777278), 666)) then begin
-                    critter_attempt_placement(party_member_obj(16777278), tile_num(dude_obj), elevation(dude_obj));
+                if (obj_is_visible_flag(party_member_obj(16777278))) then begin
+                    critter_attempt_placement(party_member_obj(16777278), dude_tile, elevation(dude_obj));
                 end
             end
             if (party_member_obj(16777376) != 0) then begin
-                if (has_trait(1, party_member_obj(16777376), 666)) then begin
-                    critter_attempt_placement(party_member_obj(16777376), tile_num(dude_obj), elevation(dude_obj));
+                if (obj_is_visible_flag(party_member_obj(16777376))) then begin
+                    critter_attempt_placement(party_member_obj(16777376), dude_tile, elevation(dude_obj));
                 end
             end
             if (party_member_obj(16777377) != 0) then begin
-                if (has_trait(1, party_member_obj(16777377), 666)) then begin
-                    critter_attempt_placement(party_member_obj(16777377), tile_num(dude_obj), elevation(dude_obj));
+                if (obj_is_visible_flag(party_member_obj(16777377))) then begin
+                    critter_attempt_placement(party_member_obj(16777377), dude_tile, elevation(dude_obj));
                 end
             end
             if (party_member_obj(16777305) != 0) then begin
-                if (has_trait(1, party_member_obj(16777305), 666)) then begin
-                    critter_attempt_placement(party_member_obj(16777305), tile_num(dude_obj), elevation(dude_obj));
+                if (obj_is_visible_flag(party_member_obj(16777305))) then begin
+                    critter_attempt_placement(party_member_obj(16777305), dude_tile, elevation(dude_obj));
                 end
             end
             if (party_member_obj(16777313) != 0) then begin
-                if (has_trait(1, party_member_obj(16777313), 666)) then begin
-                    critter_attempt_placement(party_member_obj(16777313), tile_num(dude_obj), elevation(dude_obj));
+                if (obj_is_visible_flag(party_member_obj(16777313))) then begin
+                    critter_attempt_placement(party_member_obj(16777313), dude_tile, elevation(dude_obj));
                 end
             end
             if (party_member_obj(16777323) != 0) then begin
-                if (has_trait(1, party_member_obj(16777323), 666)) then begin
-                    critter_attempt_placement(party_member_obj(16777323), tile_num(dude_obj), elevation(dude_obj));
+                if (obj_is_visible_flag(party_member_obj(16777323))) then begin
+                    critter_attempt_placement(party_member_obj(16777323), dude_tile, elevation(dude_obj));
                 end
             end
             if (party_member_obj(16777352) != 0) then begin
-                if (has_trait(1, party_member_obj(16777352), 666)) then begin
-                    critter_attempt_placement(party_member_obj(16777352), tile_num(dude_obj), elevation(dude_obj));
+                if (obj_is_visible_flag(party_member_obj(16777352))) then begin
+                    critter_attempt_placement(party_member_obj(16777352), dude_tile, elevation(dude_obj));
                 end
             end
             if (party_member_obj(16777378) != 0) then begin
-                if (has_trait(1, party_member_obj(16777378), 666)) then begin
-                    critter_attempt_placement(party_member_obj(16777378), tile_num(dude_obj), elevation(dude_obj));
+                if (obj_is_visible_flag(party_member_obj(16777378))) then begin
+                    critter_attempt_placement(party_member_obj(16777378), dude_tile, elevation(dude_obj));
                 end
             end
             if (party_member_obj(16777368) != 0) then begin
-                if (has_trait(1, party_member_obj(16777368), 666)) then begin
-                    critter_attempt_placement(party_member_obj(16777368), tile_num(dude_obj), elevation(dude_obj));
+                if (obj_is_visible_flag(party_member_obj(16777368))) then begin
+                    critter_attempt_placement(party_member_obj(16777368), dude_tile, elevation(dude_obj));
                 end
             end
             if (party_member_obj(16777379) != 0) then begin
-                if (has_trait(1, party_member_obj(16777379), 666)) then begin
-                    critter_attempt_placement(party_member_obj(16777379), tile_num(dude_obj), elevation(dude_obj));
+                if (obj_is_visible_flag(party_member_obj(16777379))) then begin
+                    critter_attempt_placement(party_member_obj(16777379), dude_tile, elevation(dude_obj));
                 end
             end
             if (party_member_obj(16777380) != 0) then begin
-                if (has_trait(1, party_member_obj(16777380), 666)) then begin
-                    critter_attempt_placement(party_member_obj(16777380), tile_num(dude_obj), elevation(dude_obj));
+                if (obj_is_visible_flag(party_member_obj(16777380))) then begin
+                    critter_attempt_placement(party_member_obj(16777380), dude_tile, elevation(dude_obj));
                 end
             end
             if (party_member_obj(16777295) != 0) then begin
-                if (has_trait(1, party_member_obj(16777295), 666)) then begin
-                    critter_attempt_placement(party_member_obj(16777295), tile_num(dude_obj), elevation(dude_obj));
+                if (obj_is_visible_flag(party_member_obj(16777295))) then begin
+                    critter_attempt_placement(party_member_obj(16777295), dude_tile, elevation(dude_obj));
                 end
             end
             if (party_member_obj(16777558) != 0) then begin
-                if (has_trait(1, party_member_obj(16777558), 666)) then begin
-                    critter_attempt_placement(party_member_obj(16777558), tile_num(dude_obj), elevation(dude_obj));
+                if (obj_is_visible_flag(party_member_obj(16777558))) then begin
+                    critter_attempt_placement(party_member_obj(16777558), dude_tile, elevation(dude_obj));
                 end
             end
             if (party_member_obj(16777600) != 0) then begin
-                if (has_trait(1, party_member_obj(16777600), 666)) then begin
-                    critter_attempt_placement(party_member_obj(16777600), tile_num(dude_obj), elevation(dude_obj));
+                if (obj_is_visible_flag(party_member_obj(16777600))) then begin
+                    critter_attempt_placement(party_member_obj(16777600), dude_tile, elevation(dude_obj));
                 end
             end
             if (party_member_obj(16777687) != 0) then begin
-                if (has_trait(1, party_member_obj(16777687), 666)) then begin
-                    critter_attempt_placement(party_member_obj(16777687), tile_num(dude_obj), elevation(dude_obj));
+                if (obj_is_visible_flag(party_member_obj(16777687))) then begin
+                    critter_attempt_placement(party_member_obj(16777687), dude_tile, elevation(dude_obj));
                 end
             end
             if (party_member_obj(16777597) != 0) then begin
-                if (has_trait(1, party_member_obj(16777597), 666)) then begin
-                    critter_attempt_placement(party_member_obj(16777597), tile_num(dude_obj), elevation(dude_obj));
+                if (obj_is_visible_flag(party_member_obj(16777597))) then begin
+                    critter_attempt_placement(party_member_obj(16777597), dude_tile, elevation(dude_obj));
                 end
             end
             if (party_member_obj(16777596) != 0) then begin
-                if (has_trait(1, party_member_obj(16777596), 666)) then begin
-                    critter_attempt_placement(party_member_obj(16777596), tile_num(dude_obj), elevation(dude_obj));
+                if (obj_is_visible_flag(party_member_obj(16777596))) then begin
+                    critter_attempt_placement(party_member_obj(16777596), dude_tile, elevation(dude_obj));
                 end
             end
             if (party_member_obj(16777595) != 0) then begin
-                if (has_trait(1, party_member_obj(16777595), 666)) then begin
-                    critter_attempt_placement(party_member_obj(16777595), tile_num(dude_obj), elevation(dude_obj));
+                if (obj_is_visible_flag(party_member_obj(16777595))) then begin
+                    critter_attempt_placement(party_member_obj(16777595), dude_tile, elevation(dude_obj));
                 end
             end
             if (party_member_obj(16777381) != 0) then begin
-                if (has_trait(1, party_member_obj(16777381), 666)) then begin
-                    critter_attempt_placement(party_member_obj(16777381), tile_num(dude_obj), elevation(dude_obj));
+                if (obj_is_visible_flag(party_member_obj(16777381))) then begin
+                    critter_attempt_placement(party_member_obj(16777381), dude_tile, elevation(dude_obj));
                 end
             end
             if (party_member_obj(16777407) != 0) then begin
-                if (has_trait(1, party_member_obj(16777407), 666)) then begin
-                    critter_attempt_placement(party_member_obj(16777407), tile_num(dude_obj), elevation(dude_obj));
+                if (obj_is_visible_flag(party_member_obj(16777407))) then begin
+                    critter_attempt_placement(party_member_obj(16777407), dude_tile, elevation(dude_obj));
                 end
             end
             if (party_member_obj(16777411) != 0) then begin
-                if (has_trait(1, party_member_obj(16777411), 666)) then begin
-                    critter_attempt_placement(party_member_obj(16777411), tile_num(dude_obj), elevation(dude_obj));
+                if (obj_is_visible_flag(party_member_obj(16777411))) then begin
+                    critter_attempt_placement(party_member_obj(16777411), dude_tile, elevation(dude_obj));
                 end
             end
             if (party_member_obj(16777412) != 0) then begin
-                if (has_trait(1, party_member_obj(16777412), 666)) then begin
-                    critter_attempt_placement(party_member_obj(16777412), tile_num(dude_obj), elevation(dude_obj));
+                if (obj_is_visible_flag(party_member_obj(16777412))) then begin
+                    critter_attempt_placement(party_member_obj(16777412), dude_tile, elevation(dude_obj));
                 end
             end
             if (party_member_obj(16777413) != 0) then begin
-                if (has_trait(1, party_member_obj(16777413), 666)) then begin
-                    critter_attempt_placement(party_member_obj(16777413), tile_num(dude_obj), elevation(dude_obj));
+                if (obj_is_visible_flag(party_member_obj(16777413))) then begin
+                    critter_attempt_placement(party_member_obj(16777413), dude_tile, elevation(dude_obj));
                 end
             end
             if (party_member_obj(16777481) != 0) then begin
-                if (has_trait(1, party_member_obj(16777481), 666)) then begin
-                    critter_attempt_placement(party_member_obj(16777481), tile_num(dude_obj), elevation(dude_obj));
+                if (obj_is_visible_flag(party_member_obj(16777481))) then begin
+                    critter_attempt_placement(party_member_obj(16777481), dude_tile, elevation(dude_obj));
                 end
             end
             if (party_member_obj(16777762) != 0) then begin
-                if (has_trait(1, party_member_obj(16777762), 666)) then begin
-                    critter_attempt_placement(party_member_obj(16777762), tile_num(dude_obj), elevation(dude_obj));
+                if (obj_is_visible_flag(party_member_obj(16777762))) then begin
+                    critter_attempt_placement(party_member_obj(16777762), dude_tile, elevation(dude_obj));
                 end
             end
             if (party_member_obj(16777859) != 0) then begin
-                if (has_trait(1, party_member_obj(16777859), 666)) then begin
-                    critter_attempt_placement(party_member_obj(16777859), tile_num(dude_obj), elevation(dude_obj));
+                if (obj_is_visible_flag(party_member_obj(16777859))) then begin
+                    critter_attempt_placement(party_member_obj(16777859), dude_tile, elevation(dude_obj));
                 end
             end
             if (party_member_obj(16777293) != 0) then begin
-                if (has_trait(1, party_member_obj(16777293), 666)) then begin
-                    critter_attempt_placement(party_member_obj(16777293), tile_num(dude_obj), elevation(dude_obj));
+                if (obj_is_visible_flag(party_member_obj(16777293))) then begin
+                    critter_attempt_placement(party_member_obj(16777293), dude_tile, elevation(dude_obj));
                 end
             end
             if (party_member_obj(16777749) != 0) then begin
-                if (has_trait(1, party_member_obj(16777749), 666)) then begin
-                    critter_attempt_placement(party_member_obj(16777749), tile_num(dude_obj), elevation(dude_obj));
+                if (obj_is_visible_flag(party_member_obj(16777749))) then begin
+                    critter_attempt_placement(party_member_obj(16777749), dude_tile, elevation(dude_obj));
                 end
             end
             if (party_member_obj(16777748) != 0) then begin
-                if (has_trait(1, party_member_obj(16777748), 666)) then begin
-                    critter_attempt_placement(party_member_obj(16777748), tile_num(dude_obj), elevation(dude_obj));
+                if (obj_is_visible_flag(party_member_obj(16777748))) then begin
+                    critter_attempt_placement(party_member_obj(16777748), dude_tile, elevation(dude_obj));
                 end
             end
             if (party_member_obj(16777747) != 0) then begin
-                if (has_trait(1, party_member_obj(16777747), 666)) then begin
-                    critter_attempt_placement(party_member_obj(16777747), tile_num(dude_obj), elevation(dude_obj));
+                if (obj_is_visible_flag(party_member_obj(16777747))) then begin
+                    critter_attempt_placement(party_member_obj(16777747), dude_tile, elevation(dude_obj));
                 end
             end
             if (party_member_obj(16777785) != 0) then begin
-                if (has_trait(1, party_member_obj(16777785), 666)) then begin
-                    critter_attempt_placement(party_member_obj(16777785), tile_num(dude_obj), elevation(dude_obj));
+                if (obj_is_visible_flag(party_member_obj(16777785))) then begin
+                    critter_attempt_placement(party_member_obj(16777785), dude_tile, elevation(dude_obj));
                 end
             end
             if (party_member_obj(16777801) != 0) then begin
-                if (has_trait(1, party_member_obj(16777801), 666)) then begin
-                    critter_attempt_placement(party_member_obj(16777801), tile_num(dude_obj), elevation(dude_obj));
+                if (obj_is_visible_flag(party_member_obj(16777801))) then begin
+                    critter_attempt_placement(party_member_obj(16777801), dude_tile, elevation(dude_obj));
                 end
             end
             if (party_member_obj(832) != 0) then begin
-                if (has_trait(1, party_member_obj(832), 666)) then begin
-                    critter_attempt_placement(party_member_obj(832), tile_num(dude_obj), elevation(dude_obj));
+                if (obj_is_visible_flag(party_member_obj(832))) then begin
+                    critter_attempt_placement(party_member_obj(832), dude_tile, elevation(dude_obj));
                 end
             end
             if (party_member_obj(455) != 0) then begin
-                if (has_trait(1, party_member_obj(455), 666)) then begin
-                    critter_attempt_placement(party_member_obj(455), tile_num(dude_obj), elevation(dude_obj));
+                if (obj_is_visible_flag(party_member_obj(455))) then begin
+                    critter_attempt_placement(party_member_obj(455), dude_tile, elevation(dude_obj));
                 end
             end
             if (party_member_obj(831) != 0) then begin
-                if (has_trait(1, party_member_obj(831), 666)) then begin
-                    critter_attempt_placement(party_member_obj(831), tile_num(dude_obj), elevation(dude_obj));
+                if (obj_is_visible_flag(party_member_obj(831))) then begin
+                    critter_attempt_placement(party_member_obj(831), dude_tile, elevation(dude_obj));
                 end
             end
             if (party_member_obj(833) != 0) then begin
-                if (has_trait(1, party_member_obj(833), 666)) then begin
-                    critter_attempt_placement(party_member_obj(833), tile_num(dude_obj), elevation(dude_obj));
+                if (obj_is_visible_flag(party_member_obj(833))) then begin
+                    critter_attempt_placement(party_member_obj(833), dude_tile, elevation(dude_obj));
                 end
             end
             if (party_member_obj(834) != 0) then begin
-                if (has_trait(1, party_member_obj(834), 666)) then begin
-                    critter_attempt_placement(party_member_obj(834), tile_num(dude_obj), elevation(dude_obj));
+                if (obj_is_visible_flag(party_member_obj(834))) then begin
+                    critter_attempt_placement(party_member_obj(834), dude_tile, elevation(dude_obj));
                 end
             end
             if (party_member_obj(835) != 0) then begin
-                if (has_trait(1, party_member_obj(835), 666)) then begin
-                    critter_attempt_placement(party_member_obj(835), tile_num(dude_obj), elevation(dude_obj));
+                if (obj_is_visible_flag(party_member_obj(835))) then begin
+                    critter_attempt_placement(party_member_obj(835), dude_tile, elevation(dude_obj));
                 end
             end
             if (party_member_obj(836) != 0) then begin
-                if (has_trait(1, party_member_obj(836), 666)) then begin
-                    critter_attempt_placement(party_member_obj(836), tile_num(dude_obj), elevation(dude_obj));
+                if (obj_is_visible_flag(party_member_obj(836))) then begin
+                    critter_attempt_placement(party_member_obj(836), dude_tile, elevation(dude_obj));
                 end
             end
         end
     end
 end
-
