diff --git a/Mods/MegaMod/Original/NICMPTR1.ssl b/Mods/MegaMod/ReDefine/NICMPTR1.ssl
index 258e670..c882671 100644
--- a/Mods/MegaMod/Original/NICMPTR1.ssl
+++ b/Mods/MegaMod/ReDefine/NICMPTR1.ssl
@@ -156,7 +156,7 @@ begin
         end_dialogue;
     end
     else begin
-        display_msg(message_str(1704, 212));
+        display_mstr(212);
     end
 end
 
@@ -174,49 +174,49 @@ end
 
 procedure Node001
 begin
-    gsay_reply(1704, 150);
-    if (global_var(1067) == 1) then begin
-        if (has_trait(0, dude_obj, 40)) then begin
-            gsay_option(1704, 200, Node002, -1);
+    Reply(150);
+    if (global_var(GVAR_EPA_IS_PREVIOUS) == 1) then begin
+        if (has_trait(TRAIT_PERK, dude_obj, PERK_scrounger)) then begin
+            giq_option(1, NAME, 200, Node002, -1);
         end
         else begin
-            gsay_option(1704, 201, Node002, -1);
+            giq_option(1, NAME, 201, Node002, -1);
         end
     end
-    gsay_option(1704, 210, Node004, -1);
-    gsay_option(1704, 211, Node999, -1);
+    giq_option(1, NAME, 210, Node004, -1);
+    giq_option(1, NAME, 211, Node999, -1);
 end
 
 procedure Node002
 begin
     if (map_var(12) == 1) then begin
-        set_global_var(1067, 2);
+        set_global_var(GVAR_EPA_IS_PREVIOUS, 2);
         give_exp_points(500);
-        gsay_message(1704, 301, 49);
+        gsay_message(NAME, 301, GOOD_REACTION);
     end
     else begin
-        gsay_reply(1704, 300);
-        if (has_skill(dude_obj, 12) > 150) then begin
-            gsay_option(1704, 310, Node003, -1);
+        Reply(300);
+        if (has_skill(dude_obj, SKILL_SCIENCE) > 150) then begin
+            giq_option(1, NAME, 310, Node003, -1);
         end
-        gsay_option(1704, 211, Node999, -1);
+        giq_option(1, NAME, 211, Node999, -1);
     end
 end
 
 procedure Node003
 begin
     give_exp_points(1500);
-    set_global_var(1067, 2);
-    gsay_message(1704, 301, 49);
+    set_global_var(GVAR_EPA_IS_PREVIOUS, 2);
+    gsay_message(NAME, 301, GOOD_REACTION);
 end
 
 procedure Node004
 begin
-    gsay_reply(1704, 400);
-    if (has_trait(0, dude_obj, 40) and (global_var(1067) == 1)) then begin
-        gsay_option(1704, 200, Node002, -1);
+    Reply(400);
+    if (has_trait(TRAIT_PERK, dude_obj, PERK_scrounger) and (global_var(GVAR_EPA_IS_PREVIOUS) == 1)) then begin
+        giq_option(1, NAME, 200, Node002, -1);
     end
-    gsay_option(1704, 211, Node999, -1);
+    giq_option(1, NAME, 211, Node999, -1);
 end
 
 procedure Node999
@@ -227,17 +227,16 @@ procedure use_obj_on_p_proc
 begin
     if (obj_pid(obj_being_used_with) == 585) then begin
         set_map_var(12, 1);
-        float_msg(self_obj, message_str(1704, 500), 8);
+        float_msg(self_obj, mstr(500), FLOAT_MSG_YELLOW);
         destroy_object(obj_being_used_with);
     end
-    if ((obj_pid(obj_being_used_with) == 584) and (global_var(1067) == 1)) then begin
+    if ((obj_pid(obj_being_used_with) == 584) and (global_var(GVAR_EPA_IS_PREVIOUS) == 1)) then begin
         if (map_var(12) == 1) then begin
-            set_global_var(1067, 2);
-            float_msg(self_obj, message_str(1704, 301), 8);
+            set_global_var(GVAR_EPA_IS_PREVIOUS, 2);
+            float_msg(self_obj, mstr(301), FLOAT_MSG_YELLOW);
         end
         else begin
-            float_msg(self_obj, message_str(1704, 300), 8);
+            float_msg(self_obj, mstr(300), FLOAT_MSG_YELLOW);
         end
     end
 end
-
