diff --git a/Mods/MegaMod/Original/NIFASSRD.ssl b/Mods/MegaMod/ReDefine/NIFASSRD.ssl
index 1602676..1f32c00 100644
--- a/Mods/MegaMod/Original/NIFASSRD.ssl
+++ b/Mods/MegaMod/ReDefine/NIFASSRD.ssl
@@ -145,8 +145,8 @@ end
 
 procedure use_p_proc
 begin
-    if (global_var(709) == 0) then begin
-        set_global_var(709, 1);
+    if (global_var(GVAR_IN_TROUBLE_WITH_MARIA) == 0) then begin
+        set_global_var(GVAR_IN_TROUBLE_WITH_MARIA, 1);
     end
 end
 
@@ -164,8 +164,7 @@ end
 
 procedure description_p_proc
 begin
-    if (global_var(709) == 0) then begin
-        set_global_var(709, 1);
+    if (global_var(GVAR_IN_TROUBLE_WITH_MARIA) == 0) then begin
+        set_global_var(GVAR_IN_TROUBLE_WITH_MARIA, 1);
     end
 end
-
