diff --git a/Mods/MegaMod/Original/NIMCGSLT.ssl b/Mods/MegaMod/ReDefine/NIMCGSLT.ssl
index 934c828..aa616ea 100644
--- a/Mods/MegaMod/Original/NIMCGSLT.ssl
+++ b/Mods/MegaMod/ReDefine/NIMCGSLT.ssl
@@ -193,53 +193,53 @@ end
 procedure ChooseFromList
 begin
     variable LVar0 := 0;
-    gsay_reply(1505, stored_msg);
-    if (message_str(353, top_msg - 1) != "") then begin
-        giq_option(1, 1505, message_str(353, 150), CLICK_PREV, 50);
+    Reply(stored_msg);
+    if (message_str(SCRIPT_NEWR1, top_msg - 1) != "") then begin
+        giq_option(1, NAME, message_str(SCRIPT_NEWR1, 150), CLICK_PREV, NEUTRAL_REACTION);
     end
     if (LVar0 == 0) then begin
-        if (message_str(353, top_msg + 0) != "") then begin
-            giq_option(1, 1505, message_str(353, top_msg + 0), CLICK_0, 50);
+        if (message_str(SCRIPT_NEWR1, top_msg + 0) != "") then begin
+            giq_option(1, NAME, message_str(SCRIPT_NEWR1, top_msg + 0), CLICK_0, NEUTRAL_REACTION);
         end
         else begin
             LVar0 := 1;
         end
     end
     if (LVar0 == 0) then begin
-        if (message_str(353, top_msg + 1) != "") then begin
-            giq_option(1, 1505, message_str(353, top_msg + 1), CLICK_1, 50);
+        if (message_str(SCRIPT_NEWR1, top_msg + 1) != "") then begin
+            giq_option(1, NAME, message_str(SCRIPT_NEWR1, top_msg + 1), CLICK_1, NEUTRAL_REACTION);
         end
         else begin
             LVar0 := 1;
         end
     end
     if (LVar0 == 0) then begin
-        if (message_str(353, top_msg + 2) != "") then begin
-            giq_option(1, 1505, message_str(353, top_msg + 2), CLICK_2, 50);
+        if (message_str(SCRIPT_NEWR1, top_msg + 2) != "") then begin
+            giq_option(1, NAME, message_str(SCRIPT_NEWR1, top_msg + 2), CLICK_2, NEUTRAL_REACTION);
         end
         else begin
             LVar0 := 1;
         end
     end
     if (LVar0 == 0) then begin
-        if (message_str(353, top_msg + 3) != "") then begin
-            giq_option(1, 1505, message_str(353, top_msg + 3), CLICK_3, 50);
+        if (message_str(SCRIPT_NEWR1, top_msg + 3) != "") then begin
+            giq_option(1, NAME, message_str(SCRIPT_NEWR1, top_msg + 3), CLICK_3, NEUTRAL_REACTION);
         end
         else begin
             LVar0 := 1;
         end
     end
     if (LVar0 == 0) then begin
-        if (message_str(353, top_msg + 4) != "") then begin
-            giq_option(1, 1505, message_str(353, top_msg + 4), CLICK_4, 50);
+        if (message_str(SCRIPT_NEWR1, top_msg + 4) != "") then begin
+            giq_option(1, NAME, message_str(SCRIPT_NEWR1, top_msg + 4), CLICK_4, NEUTRAL_REACTION);
         end
         else begin
             LVar0 := 1;
         end
     end
     if (LVar0 == 0) then begin
-        if (message_str(353, top_msg + 5) != "") then begin
-            giq_option(1, 1505, message_str(353, 151), CLICK_MORE, 50);
+        if (message_str(SCRIPT_NEWR1, top_msg + 5) != "") then begin
+            giq_option(1, NAME, message_str(SCRIPT_NEWR1, 151), CLICK_MORE, NEUTRAL_REACTION);
         end
     end
     return 0;
@@ -316,11 +316,11 @@ end
 procedure look_at_p_proc
 begin
     script_overrides;
-    if ((get_critter_stat(dude_obj, 4) > 3) and (has_skill(dude_obj, 16) >= 25)) then begin
-        display_msg(message_str(1505, 150));
+    if ((dude_iq > 3) and (dude_skill(SKILL_GAMBLING) >= 25)) then begin
+        display_mstr(150);
     end
     else begin
-        display_msg(message_str(377, 100));
+        display_msg(message_str(SCRIPT_ZISLOTS, 100));
     end
 end
 
@@ -328,11 +328,11 @@ procedure description_p_proc
 begin
     variable LVar0 := 0;
     script_overrides;
-    if ((get_critter_stat(dude_obj, 4) > 3) and (has_skill(dude_obj, 16) >= 25)) then begin
-        display_msg(message_str(1505, 150 + ((local_var(4) bwand 4) != 0)));
+    if ((dude_iq > 3) and (dude_skill(SKILL_GAMBLING) >= 25)) then begin
+        display_mstr(150 + ((local_var(4) bwand 4) != 0));
     end
     else begin
-        display_msg(message_str(377, 100) + message_str(377, random(130, 130)));
+        display_msg(message_str(SCRIPT_ZISLOTS, 100) + message_str(SCRIPT_ZISLOTS, 130));
     end
     set_local_var(4, local_var(4) bwor 4);
     return 0;
@@ -349,51 +349,51 @@ end
 procedure use_skill_on_p_proc
 begin
     variable LVar0 := 0;
-    if (not(global_var(246) == 3)) then begin
-        set_global_var(246, 1);
+    if (not(global_var(GVAR_NEW_RENO_MCGEE_SEED) == 3)) then begin
+        set_global_var(GVAR_NEW_RENO_MCGEE_SEED, 1);
     end
-    if ((action_being_used == 16) and ((get_critter_stat(dude_obj, 4) > 3) and (has_skill(dude_obj, 16) >= 25))) then begin
+    if ((action_being_used == 16) and ((dude_iq > 3) and (dude_skill(SKILL_GAMBLING) >= 25))) then begin
         script_overrides;
-        display_msg(message_str(1505, 320));
+        display_mstr(320);
     end
     else begin
         if (action_being_used == 13) then begin
             script_overrides;
             LVar0 := roll_vs_skill(dude_obj, action_being_used, 0);
             if (tile_distance_objs(mcGeeOBJ, mcGeeSlotMachineOBJ) <= 2) then begin
-                display_msg(message_str(1505, 321));
+                display_mstr(321);
             end
             else begin
-                if ((global_var(246) == 2) or (global_var(246) == 3)) then begin
-                    display_msg(message_str(1505, 322));
+                if ((global_var(GVAR_NEW_RENO_MCGEE_SEED) == 2) or (global_var(GVAR_NEW_RENO_MCGEE_SEED) == 3)) then begin
+                    display_mstr(322);
                 end
                 else begin
                     if (local_var(3) != 0) then begin
-                        display_msg(message_str(1505, 323));
+                        display_mstr(323);
                     end
                     else begin
                         LVar0 := roll_vs_skill(dude_obj, 13, 0);
                         if (not(is_success(LVar0))) then begin
                             if (is_critical(LVar0)) then begin
-                                display_msg(message_str(1505, 324));
+                                display_mstr(324);
                                 set_local_var(3, 1);
                             end
                             else begin
-                                display_msg(message_str(1505, 325));
+                                display_mstr(325);
                                 set_local_var(3, 2);
                             end
                         end
                         else begin
-                            display_msg(message_str(1505, 326));
+                            display_mstr(326);
                             if (is_critical(LVar0)) then begin
                                 set_local_var(3, 4);
                                 give_exp_points(100);
-                                display_msg(message_str(14, 100) + 100 + message_str(14, 101));
+                                display_msg(g_mstr(100) + 100 + g_mstr(101));
                             end
                             else begin
                                 set_local_var(3, 3);
                                 give_exp_points(50);
-                                display_msg(message_str(14, 100) + 50 + message_str(14, 101));
+                                display_msg(g_mstr(100) + 50 + g_mstr(101));
                             end
                         end
                     end
@@ -413,14 +413,14 @@ procedure map_enter_p_proc
 begin
     variable LVar0 := 0;
     mcGeeSlotMachineOBJ := self_obj;
-    if (metarule(14, 0)) then begin
+    if (map_first_run) then begin
         if (local_var(0) == 0) then begin
             set_local_var(0, 3);
         end
     end
     last_timed_obj := self_obj;
     last_timed_param := last_timed_param;
-    metarule3(100, self_obj, 1, 0);
+    rm_fixed_timer_event(self_obj, 1);
     add_timer_event(self_obj, game_ticks(5) + random(0, game_ticks(30 - 5)), 1);
     return 0;
     LVar0 := 0;
@@ -428,12 +428,12 @@ end
 
 procedure timed_event_p_proc
 begin
-    if ((combat_is_initialized == 0) and (tile_distance(tile_num(self_obj), tile_num(dude_obj)) <= 5)) then begin
-        float_msg(self_obj, message_str(1505, random(203 - (3 * (get_critter_stat(dude_obj, 34) == 0) * metarule(18, dude_obj)), 212 + (4 * (get_critter_stat(dude_obj, 34) == 1) * metarule(18, dude_obj)))), 12);
+    if ((combat_is_initialized == 0) and (tile_distance(tile_num(self_obj), dude_tile) <= 5)) then begin
+        float_msg(self_obj, mstr(random(203 - (3 * (dude_gender == 0) * drug_influence(dude_obj)), 212 + (4 * (dude_gender == 1) * drug_influence(dude_obj)))), FLOAT_MSG_LIGHT_GREY);
     end
     last_timed_obj := self_obj;
     last_timed_param := last_timed_param;
-    metarule3(100, self_obj, 1, 0);
+    rm_fixed_timer_event(self_obj, 1);
     add_timer_event(self_obj, game_ticks(15) + random(0, game_ticks(30 - 15)), 1);
 end
 
@@ -447,7 +447,7 @@ end
 
 procedure Node000
 begin
-    if (((get_critter_stat(dude_obj, 4) > 3) and (has_skill(dude_obj, 16) >= 25)) == 0) then begin
+    if (((dude_iq > 3) and (dude_skill(SKILL_GAMBLING) >= 25)) == 0) then begin
         call Node004();
     end
     else begin
@@ -458,88 +458,88 @@ begin
             call Node001();
         end
     end
-    if (not(global_var(246) == 3)) then begin
-        set_global_var(246, 1);
+    if (not(global_var(GVAR_NEW_RENO_MCGEE_SEED) == 3)) then begin
+        set_global_var(GVAR_NEW_RENO_MCGEE_SEED, 1);
     end
 end
 
 procedure Node001
 begin
-    gsay_reply(1505, 245 + metarule(18, dude_obj));
+    Reply(245 + drug_influence(dude_obj));
     if (local_var(1) == 1) then begin
-        giq_option(1, 1505, 311, Node003, 50);
+        giq_option(1, NAME, 311, Node003, NEUTRAL_REACTION);
     end
     else begin
         if ((((5 * (local_var(0) == 1)) + (5 * (local_var(0) == 2)) + (5 * (local_var(0) == 3))) >= 1) and (item_caps_total(dude_obj) >= (1 * ((1 * (local_var(0) == 1)) + (1 * (local_var(0) == 2)) + (1 * (local_var(0) == 3)))))) then begin
             if (item_caps_total(dude_obj) == ((1 * (local_var(0) == 1)) + (1 * (local_var(0) == 2)) + (1 * (local_var(0) == 3)))) then begin
-                giq_option(1, 377, message_str(377, 252) + (1 * ((1 * (local_var(0) == 1)) + (1 * (local_var(0) == 2)) + (1 * (local_var(0) == 3)))) + message_str(377, 253), Node981, 50);
+                giq_option(1, SCRIPT_ZISLOTS, message_str(SCRIPT_ZISLOTS, 252) + (1 * ((1 * (local_var(0) == 1)) + (1 * (local_var(0) == 2)) + (1 * (local_var(0) == 3)))) + message_str(SCRIPT_ZISLOTS, 253), Node981, NEUTRAL_REACTION);
             end
             else begin
-                if ((1 == ((5 * (local_var(0) == 1)) + (5 * (local_var(0) == 2)) + (5 * (local_var(0) == 3)))) and (has_skill(dude_obj, 16) >= 50)) then begin
-                    giq_option(1, 377, message_str(377, 250) + (1 * ((1 * (local_var(0) == 1)) + (1 * (local_var(0) == 2)) + (1 * (local_var(0) == 3)))) + message_str(377, 251) + (message_str(377, 3000) + message_str(377, random(1105, 1107)) + message_str(377, random(3001, 3002))), Node981, 50);
+                if ((1 == ((5 * (local_var(0) == 1)) + (5 * (local_var(0) == 2)) + (5 * (local_var(0) == 3)))) and (dude_skill(SKILL_GAMBLING) >= 50)) then begin
+                    giq_option(1, SCRIPT_ZISLOTS, message_str(SCRIPT_ZISLOTS, 250) + (1 * ((1 * (local_var(0) == 1)) + (1 * (local_var(0) == 2)) + (1 * (local_var(0) == 3)))) + message_str(SCRIPT_ZISLOTS, 251) + (message_str(SCRIPT_ZISLOTS, 3000) + message_str(SCRIPT_ZISLOTS, random(1105, 1107)) + message_str(SCRIPT_ZISLOTS, random(3001, 3002))), Node981, NEUTRAL_REACTION);
                 end
                 else begin
-                    giq_option(1, 377, message_str(377, 250) + (1 * ((1 * (local_var(0) == 1)) + (1 * (local_var(0) == 2)) + (1 * (local_var(0) == 3)))) + message_str(377, 251), Node981, 50);
+                    giq_option(1, SCRIPT_ZISLOTS, message_str(SCRIPT_ZISLOTS, 250) + (1 * ((1 * (local_var(0) == 1)) + (1 * (local_var(0) == 2)) + (1 * (local_var(0) == 3)))) + message_str(SCRIPT_ZISLOTS, 251), Node981, NEUTRAL_REACTION);
                 end
             end
         end
         if ((((5 * (local_var(0) == 1)) + (5 * (local_var(0) == 2)) + (5 * (local_var(0) == 3))) >= 2) and (item_caps_total(dude_obj) >= (2 * ((1 * (local_var(0) == 1)) + (1 * (local_var(0) == 2)) + (1 * (local_var(0) == 3)))))) then begin
             if (item_caps_total(dude_obj) == ((1 * (local_var(0) == 1)) + (1 * (local_var(0) == 2)) + (1 * (local_var(0) == 3)))) then begin
-                giq_option(1, 377, message_str(377, 252) + (2 * ((1 * (local_var(0) == 1)) + (1 * (local_var(0) == 2)) + (1 * (local_var(0) == 3)))) + message_str(377, 253), Node982, 50);
+                giq_option(1, SCRIPT_ZISLOTS, message_str(SCRIPT_ZISLOTS, 252) + (2 * ((1 * (local_var(0) == 1)) + (1 * (local_var(0) == 2)) + (1 * (local_var(0) == 3)))) + message_str(SCRIPT_ZISLOTS, 253), Node982, NEUTRAL_REACTION);
             end
             else begin
-                if ((2 == ((5 * (local_var(0) == 1)) + (5 * (local_var(0) == 2)) + (5 * (local_var(0) == 3)))) and (has_skill(dude_obj, 16) >= 50)) then begin
-                    giq_option(1, 377, message_str(377, 250) + (2 * ((1 * (local_var(0) == 1)) + (1 * (local_var(0) == 2)) + (1 * (local_var(0) == 3)))) + message_str(377, 251) + (message_str(377, 3000) + message_str(377, random(1105, 1107)) + message_str(377, random(3001, 3002))), Node982, 50);
+                if ((2 == ((5 * (local_var(0) == 1)) + (5 * (local_var(0) == 2)) + (5 * (local_var(0) == 3)))) and (dude_skill(SKILL_GAMBLING) >= 50)) then begin
+                    giq_option(1, SCRIPT_ZISLOTS, message_str(SCRIPT_ZISLOTS, 250) + (2 * ((1 * (local_var(0) == 1)) + (1 * (local_var(0) == 2)) + (1 * (local_var(0) == 3)))) + message_str(SCRIPT_ZISLOTS, 251) + (message_str(SCRIPT_ZISLOTS, 3000) + message_str(SCRIPT_ZISLOTS, random(1105, 1107)) + message_str(SCRIPT_ZISLOTS, random(3001, 3002))), Node982, NEUTRAL_REACTION);
                 end
                 else begin
-                    giq_option(1, 377, message_str(377, 250) + (2 * ((1 * (local_var(0) == 1)) + (1 * (local_var(0) == 2)) + (1 * (local_var(0) == 3)))) + message_str(377, 251), Node982, 50);
+                    giq_option(1, SCRIPT_ZISLOTS, message_str(SCRIPT_ZISLOTS, 250) + (2 * ((1 * (local_var(0) == 1)) + (1 * (local_var(0) == 2)) + (1 * (local_var(0) == 3)))) + message_str(SCRIPT_ZISLOTS, 251), Node982, NEUTRAL_REACTION);
                 end
             end
         end
         if ((((5 * (local_var(0) == 1)) + (5 * (local_var(0) == 2)) + (5 * (local_var(0) == 3))) >= 3) and (item_caps_total(dude_obj) >= (3 * ((1 * (local_var(0) == 1)) + (1 * (local_var(0) == 2)) + (1 * (local_var(0) == 3)))))) then begin
             if (item_caps_total(dude_obj) == ((1 * (local_var(0) == 1)) + (1 * (local_var(0) == 2)) + (1 * (local_var(0) == 3)))) then begin
-                giq_option(1, 377, message_str(377, 252) + (3 * ((1 * (local_var(0) == 1)) + (1 * (local_var(0) == 2)) + (1 * (local_var(0) == 3)))) + message_str(377, 253), Node983, 50);
+                giq_option(1, SCRIPT_ZISLOTS, message_str(SCRIPT_ZISLOTS, 252) + (3 * ((1 * (local_var(0) == 1)) + (1 * (local_var(0) == 2)) + (1 * (local_var(0) == 3)))) + message_str(SCRIPT_ZISLOTS, 253), Node983, NEUTRAL_REACTION);
             end
             else begin
-                if ((3 == ((5 * (local_var(0) == 1)) + (5 * (local_var(0) == 2)) + (5 * (local_var(0) == 3)))) and (has_skill(dude_obj, 16) >= 50)) then begin
-                    giq_option(1, 377, message_str(377, 250) + (3 * ((1 * (local_var(0) == 1)) + (1 * (local_var(0) == 2)) + (1 * (local_var(0) == 3)))) + message_str(377, 251) + (message_str(377, 3000) + message_str(377, random(1105, 1107)) + message_str(377, random(3001, 3002))), Node983, 50);
+                if ((3 == ((5 * (local_var(0) == 1)) + (5 * (local_var(0) == 2)) + (5 * (local_var(0) == 3)))) and (dude_skill(SKILL_GAMBLING) >= 50)) then begin
+                    giq_option(1, SCRIPT_ZISLOTS, message_str(SCRIPT_ZISLOTS, 250) + (3 * ((1 * (local_var(0) == 1)) + (1 * (local_var(0) == 2)) + (1 * (local_var(0) == 3)))) + message_str(SCRIPT_ZISLOTS, 251) + (message_str(SCRIPT_ZISLOTS, 3000) + message_str(SCRIPT_ZISLOTS, random(1105, 1107)) + message_str(SCRIPT_ZISLOTS, random(3001, 3002))), Node983, NEUTRAL_REACTION);
                 end
                 else begin
-                    giq_option(1, 377, message_str(377, 250) + (3 * ((1 * (local_var(0) == 1)) + (1 * (local_var(0) == 2)) + (1 * (local_var(0) == 3)))) + message_str(377, 251), Node983, 50);
+                    giq_option(1, SCRIPT_ZISLOTS, message_str(SCRIPT_ZISLOTS, 250) + (3 * ((1 * (local_var(0) == 1)) + (1 * (local_var(0) == 2)) + (1 * (local_var(0) == 3)))) + message_str(SCRIPT_ZISLOTS, 251), Node983, NEUTRAL_REACTION);
                 end
             end
         end
         if ((((5 * (local_var(0) == 1)) + (5 * (local_var(0) == 2)) + (5 * (local_var(0) == 3))) >= 4) and (item_caps_total(dude_obj) >= (4 * ((1 * (local_var(0) == 1)) + (1 * (local_var(0) == 2)) + (1 * (local_var(0) == 3)))))) then begin
             if (item_caps_total(dude_obj) == ((1 * (local_var(0) == 1)) + (1 * (local_var(0) == 2)) + (1 * (local_var(0) == 3)))) then begin
-                giq_option(1, 377, message_str(377, 252) + (4 * ((1 * (local_var(0) == 1)) + (1 * (local_var(0) == 2)) + (1 * (local_var(0) == 3)))) + message_str(377, 253), Node984, 50);
+                giq_option(1, SCRIPT_ZISLOTS, message_str(SCRIPT_ZISLOTS, 252) + (4 * ((1 * (local_var(0) == 1)) + (1 * (local_var(0) == 2)) + (1 * (local_var(0) == 3)))) + message_str(SCRIPT_ZISLOTS, 253), Node984, NEUTRAL_REACTION);
             end
             else begin
-                if ((4 == ((5 * (local_var(0) == 1)) + (5 * (local_var(0) == 2)) + (5 * (local_var(0) == 3)))) and (has_skill(dude_obj, 16) >= 50)) then begin
-                    giq_option(1, 377, message_str(377, 250) + (4 * ((1 * (local_var(0) == 1)) + (1 * (local_var(0) == 2)) + (1 * (local_var(0) == 3)))) + message_str(377, 251) + (message_str(377, 3000) + message_str(377, random(1105, 1107)) + message_str(377, random(3001, 3002))), Node984, 50);
+                if ((4 == ((5 * (local_var(0) == 1)) + (5 * (local_var(0) == 2)) + (5 * (local_var(0) == 3)))) and (dude_skill(SKILL_GAMBLING) >= 50)) then begin
+                    giq_option(1, SCRIPT_ZISLOTS, message_str(SCRIPT_ZISLOTS, 250) + (4 * ((1 * (local_var(0) == 1)) + (1 * (local_var(0) == 2)) + (1 * (local_var(0) == 3)))) + message_str(SCRIPT_ZISLOTS, 251) + (message_str(SCRIPT_ZISLOTS, 3000) + message_str(SCRIPT_ZISLOTS, random(1105, 1107)) + message_str(SCRIPT_ZISLOTS, random(3001, 3002))), Node984, NEUTRAL_REACTION);
                 end
                 else begin
-                    giq_option(1, 377, message_str(377, 250) + (4 * ((1 * (local_var(0) == 1)) + (1 * (local_var(0) == 2)) + (1 * (local_var(0) == 3)))) + message_str(377, 251), Node984, 50);
+                    giq_option(1, SCRIPT_ZISLOTS, message_str(SCRIPT_ZISLOTS, 250) + (4 * ((1 * (local_var(0) == 1)) + (1 * (local_var(0) == 2)) + (1 * (local_var(0) == 3)))) + message_str(SCRIPT_ZISLOTS, 251), Node984, NEUTRAL_REACTION);
                 end
             end
         end
         if ((((5 * (local_var(0) == 1)) + (5 * (local_var(0) == 2)) + (5 * (local_var(0) == 3))) >= 5) and (item_caps_total(dude_obj) >= (5 * ((1 * (local_var(0) == 1)) + (1 * (local_var(0) == 2)) + (1 * (local_var(0) == 3)))))) then begin
             if (item_caps_total(dude_obj) == ((1 * (local_var(0) == 1)) + (1 * (local_var(0) == 2)) + (1 * (local_var(0) == 3)))) then begin
-                giq_option(1, 377, message_str(377, 252) + (5 * ((1 * (local_var(0) == 1)) + (1 * (local_var(0) == 2)) + (1 * (local_var(0) == 3)))) + message_str(377, 253), Node985, 50);
+                giq_option(1, SCRIPT_ZISLOTS, message_str(SCRIPT_ZISLOTS, 252) + (5 * ((1 * (local_var(0) == 1)) + (1 * (local_var(0) == 2)) + (1 * (local_var(0) == 3)))) + message_str(SCRIPT_ZISLOTS, 253), Node985, NEUTRAL_REACTION);
             end
             else begin
-                if ((5 == ((5 * (local_var(0) == 1)) + (5 * (local_var(0) == 2)) + (5 * (local_var(0) == 3)))) and (has_skill(dude_obj, 16) >= 50)) then begin
-                    giq_option(1, 377, message_str(377, 250) + (5 * ((1 * (local_var(0) == 1)) + (1 * (local_var(0) == 2)) + (1 * (local_var(0) == 3)))) + message_str(377, 251) + (message_str(377, 3000) + message_str(377, random(1105, 1107)) + message_str(377, random(3001, 3002))), Node985, 50);
+                if ((5 == ((5 * (local_var(0) == 1)) + (5 * (local_var(0) == 2)) + (5 * (local_var(0) == 3)))) and (dude_skill(SKILL_GAMBLING) >= 50)) then begin
+                    giq_option(1, SCRIPT_ZISLOTS, message_str(SCRIPT_ZISLOTS, 250) + (5 * ((1 * (local_var(0) == 1)) + (1 * (local_var(0) == 2)) + (1 * (local_var(0) == 3)))) + message_str(SCRIPT_ZISLOTS, 251) + (message_str(SCRIPT_ZISLOTS, 3000) + message_str(SCRIPT_ZISLOTS, random(1105, 1107)) + message_str(SCRIPT_ZISLOTS, random(3001, 3002))), Node985, NEUTRAL_REACTION);
                 end
                 else begin
-                    giq_option(1, 377, message_str(377, 250) + (5 * ((1 * (local_var(0) == 1)) + (1 * (local_var(0) == 2)) + (1 * (local_var(0) == 3)))) + message_str(377, 251), Node985, 50);
+                    giq_option(1, SCRIPT_ZISLOTS, message_str(SCRIPT_ZISLOTS, 250) + (5 * ((1 * (local_var(0) == 1)) + (1 * (local_var(0) == 2)) + (1 * (local_var(0) == 3)))) + message_str(SCRIPT_ZISLOTS, 251), Node985, NEUTRAL_REACTION);
                 end
             end
         end
-        if (metarule(18, dude_obj) == 0) then begin
-            giq_option(1, 377, 254 + (item_caps_total(dude_obj) == 0) + (((game_time_hour <= 600) or (game_time_hour >= 1800)) * 2), Node999, 50);
+        if (drug_influence(dude_obj) == 0) then begin
+            giq_option(1, SCRIPT_ZISLOTS, 254 + (item_caps_total(dude_obj) == 0) + (((game_time_hour <= 600) or (game_time_hour >= 1800)) * 2), Node999, NEUTRAL_REACTION);
         end
         else begin
-            giq_option(1, 377, 258 + (get_critter_stat(dude_obj, 34) == 0), Node999, 50);
+            giq_option(1, SCRIPT_ZISLOTS, 258 + (dude_gender == 0), Node999, NEUTRAL_REACTION);
         end
     end
 end
@@ -550,7 +550,7 @@ begin
     item_caps_adjust(dude_obj, -coin_mult * ((1 * (local_var(0) == 1)) + (1 * (local_var(0) == 2)) + (1 * (local_var(0) == 3))));
     if (((-coin_mult * ((1 * (local_var(0) == 1)) + (1 * (local_var(0) == 2)) + (1 * (local_var(0) == 3)))) >= (100 * ((1 * (local_var(0) == 1)) + (1 * (local_var(0) == 2)) + (1 * (local_var(0) == 3))) * coin_mult)) and (coin_mult == ((5 * (local_var(0) == 1)) + (5 * (local_var(0) == 2)) + (5 * (local_var(0) == 3))))) then begin
         set_local_var(1, 1);
-        set_global_var(246, 2);
+        set_global_var(GVAR_NEW_RENO_MCGEE_SEED, 2);
     end
     if ((-coin_mult * ((1 * (local_var(0) == 1)) + (1 * (local_var(0) == 2)) + (1 * (local_var(0) == 3)))) > 0) then begin
         play_sfx("winner");
@@ -735,32 +735,32 @@ begin
     end
     if (win_payoff == 500) then begin
         set_local_var(1, 1);
-        set_global_var(246, 2);
-        LVar0 := message_str(1505, 310);
+        set_global_var(GVAR_NEW_RENO_MCGEE_SEED, 2);
+        LVar0 := mstr(310);
     end
     else begin
-        LVar0 := message_str(1505, 260) + message_str(377, reel_1) + message_str(377, 599) + message_str(377, reel_2) + message_str(377, 599) + message_str(377, reel_3) + message_str(377, 599);
+        LVar0 := mstr(260) + message_str(SCRIPT_ZISLOTS, reel_1) + message_str(SCRIPT_ZISLOTS, 599) + message_str(SCRIPT_ZISLOTS, reel_2) + message_str(SCRIPT_ZISLOTS, 599) + message_str(SCRIPT_ZISLOTS, reel_3) + message_str(SCRIPT_ZISLOTS, 599);
         if ((200 <= win_payoff) and (coin_mult == ((5 * (local_var(0) == 1)) + (5 * (local_var(0) == 2)) + (5 * (local_var(0) == 3))))) then begin
-            LVar0 := LVar0 + message_str(377, 605);
+            LVar0 := LVar0 + message_str(SCRIPT_ZISLOTS, 605);
         end
         else begin
             if ((100 <= win_payoff) and (coin_mult == ((5 * (local_var(0) == 1)) + (5 * (local_var(0) == 2)) + (5 * (local_var(0) == 3))))) then begin
-                LVar0 := LVar0 + message_str(377, 604);
+                LVar0 := LVar0 + message_str(SCRIPT_ZISLOTS, 604);
             end
             else begin
                 if (10 <= win_payoff) then begin
-                    LVar0 := LVar0 + (message_str(377, 603) + message_str(377, (reel_1 * (reel_1 != 1005)) + (reel_2 * (reel_1 == 1005)) + 100) + message_str(377, 613));
+                    LVar0 := LVar0 + (message_str(SCRIPT_ZISLOTS, 603) + message_str(SCRIPT_ZISLOTS, (reel_1 * (reel_1 != 1005)) + (reel_2 * (reel_1 == 1005)) + 100) + message_str(SCRIPT_ZISLOTS, 613));
                 end
                 else begin
                     if (5 <= win_payoff) then begin
-                        LVar0 := LVar0 + message_str(377, 602);
+                        LVar0 := LVar0 + message_str(SCRIPT_ZISLOTS, 602);
                     end
                     else begin
                         if (2 <= win_payoff) then begin
-                            LVar0 := LVar0 + message_str(377, 601);
+                            LVar0 := LVar0 + message_str(SCRIPT_ZISLOTS, 601);
                         end
                         else begin
-                            LVar0 := LVar0 + message_str(377, 600);
+                            LVar0 := LVar0 + message_str(SCRIPT_ZISLOTS, 600);
                         end
                     end
                 end
@@ -768,22 +768,22 @@ begin
         end
         win_payoff := win_payoff * (coin_mult * ((1 * (local_var(0) == 1)) + (1 * (local_var(0) == 2)) + (1 * (local_var(0) == 3))));
         if (win_payoff > 0) then begin
-            LVar0 := LVar0 + (message_str(377, 700) + win_payoff + message_str(377, 701));
+            LVar0 := LVar0 + (message_str(SCRIPT_ZISLOTS, 700) + win_payoff + message_str(SCRIPT_ZISLOTS, 701));
         end
-        if (metarule(18, dude_obj)) then begin
+        if (drug_influence(dude_obj)) then begin
             if (win_payoff == 0) then begin
-                LVar0 := LVar0 + message_str(1505, 251);
+                LVar0 := LVar0 + mstr(251);
             end
             else begin
                 if ((win_payoff / (coin_mult * ((1 * (local_var(0) == 1)) + (1 * (local_var(0) == 2)) + (1 * (local_var(0) == 3))))) >= 10) then begin
-                    LVar0 := LVar0 + message_str(1505, 265);
+                    LVar0 := LVar0 + mstr(265);
                 end
                 else begin
                     if ((win_payoff / (coin_mult * ((1 * (local_var(0) == 1)) + (1 * (local_var(0) == 2)) + (1 * (local_var(0) == 3))))) >= 5) then begin
-                        LVar0 := LVar0 + message_str(1505, 261);
+                        LVar0 := LVar0 + mstr(261);
                     end
                     else begin
-                        LVar0 := LVar0 + message_str(1505, 255);
+                        LVar0 := LVar0 + mstr(255);
                     end
                 end
             end
@@ -795,7 +795,7 @@ begin
     item_caps_adjust(dude_obj, win_payoff);
     if ((win_payoff >= (100 * ((1 * (local_var(0) == 1)) + (1 * (local_var(0) == 2)) + (1 * (local_var(0) == 3))) * coin_mult)) and (coin_mult == ((5 * (local_var(0) == 1)) + (5 * (local_var(0) == 2)) + (5 * (local_var(0) == 3))))) then begin
         set_local_var(1, 1);
-        set_global_var(246, 2);
+        set_global_var(GVAR_NEW_RENO_MCGEE_SEED, 2);
     end
     if (win_payoff > 0) then begin
         play_sfx("winner");
@@ -806,81 +806,81 @@ begin
         end
     end
     debug_msg("\nSlot money remaining == " + local_var(2));
-    gsay_reply(377, LVar0);
+    gsay_reply(SCRIPT_ZISLOTS, LVar0);
     if (local_var(1) == 1) then begin
-        giq_option(1, 1505, 311, Node003, 50);
+        giq_option(1, NAME, 311, Node003, NEUTRAL_REACTION);
     end
     else begin
         if ((((5 * (local_var(0) == 1)) + (5 * (local_var(0) == 2)) + (5 * (local_var(0) == 3))) >= 1) and (item_caps_total(dude_obj) >= (1 * ((1 * (local_var(0) == 1)) + (1 * (local_var(0) == 2)) + (1 * (local_var(0) == 3)))))) then begin
             if (item_caps_total(dude_obj) == ((1 * (local_var(0) == 1)) + (1 * (local_var(0) == 2)) + (1 * (local_var(0) == 3)))) then begin
-                giq_option(1, 377, message_str(377, 252) + (1 * ((1 * (local_var(0) == 1)) + (1 * (local_var(0) == 2)) + (1 * (local_var(0) == 3)))) + message_str(377, 253), Node981, 50);
+                giq_option(1, SCRIPT_ZISLOTS, message_str(SCRIPT_ZISLOTS, 252) + (1 * ((1 * (local_var(0) == 1)) + (1 * (local_var(0) == 2)) + (1 * (local_var(0) == 3)))) + message_str(SCRIPT_ZISLOTS, 253), Node981, NEUTRAL_REACTION);
             end
             else begin
-                if ((1 == ((5 * (local_var(0) == 1)) + (5 * (local_var(0) == 2)) + (5 * (local_var(0) == 3)))) and (has_skill(dude_obj, 16) >= 50)) then begin
-                    giq_option(1, 377, message_str(377, 250) + (1 * ((1 * (local_var(0) == 1)) + (1 * (local_var(0) == 2)) + (1 * (local_var(0) == 3)))) + message_str(377, 251) + (message_str(377, 3000) + message_str(377, random(1105, 1107)) + message_str(377, random(3001, 3002))), Node981, 50);
+                if ((1 == ((5 * (local_var(0) == 1)) + (5 * (local_var(0) == 2)) + (5 * (local_var(0) == 3)))) and (dude_skill(SKILL_GAMBLING) >= 50)) then begin
+                    giq_option(1, SCRIPT_ZISLOTS, message_str(SCRIPT_ZISLOTS, 250) + (1 * ((1 * (local_var(0) == 1)) + (1 * (local_var(0) == 2)) + (1 * (local_var(0) == 3)))) + message_str(SCRIPT_ZISLOTS, 251) + (message_str(SCRIPT_ZISLOTS, 3000) + message_str(SCRIPT_ZISLOTS, random(1105, 1107)) + message_str(SCRIPT_ZISLOTS, random(3001, 3002))), Node981, NEUTRAL_REACTION);
                 end
                 else begin
-                    giq_option(1, 377, message_str(377, 250) + (1 * ((1 * (local_var(0) == 1)) + (1 * (local_var(0) == 2)) + (1 * (local_var(0) == 3)))) + message_str(377, 251), Node981, 50);
+                    giq_option(1, SCRIPT_ZISLOTS, message_str(SCRIPT_ZISLOTS, 250) + (1 * ((1 * (local_var(0) == 1)) + (1 * (local_var(0) == 2)) + (1 * (local_var(0) == 3)))) + message_str(SCRIPT_ZISLOTS, 251), Node981, NEUTRAL_REACTION);
                 end
             end
         end
         if ((((5 * (local_var(0) == 1)) + (5 * (local_var(0) == 2)) + (5 * (local_var(0) == 3))) >= 2) and (item_caps_total(dude_obj) >= (2 * ((1 * (local_var(0) == 1)) + (1 * (local_var(0) == 2)) + (1 * (local_var(0) == 3)))))) then begin
             if (item_caps_total(dude_obj) == ((1 * (local_var(0) == 1)) + (1 * (local_var(0) == 2)) + (1 * (local_var(0) == 3)))) then begin
-                giq_option(1, 377, message_str(377, 252) + (2 * ((1 * (local_var(0) == 1)) + (1 * (local_var(0) == 2)) + (1 * (local_var(0) == 3)))) + message_str(377, 253), Node982, 50);
+                giq_option(1, SCRIPT_ZISLOTS, message_str(SCRIPT_ZISLOTS, 252) + (2 * ((1 * (local_var(0) == 1)) + (1 * (local_var(0) == 2)) + (1 * (local_var(0) == 3)))) + message_str(SCRIPT_ZISLOTS, 253), Node982, NEUTRAL_REACTION);
             end
             else begin
-                if ((2 == ((5 * (local_var(0) == 1)) + (5 * (local_var(0) == 2)) + (5 * (local_var(0) == 3)))) and (has_skill(dude_obj, 16) >= 50)) then begin
-                    giq_option(1, 377, message_str(377, 250) + (2 * ((1 * (local_var(0) == 1)) + (1 * (local_var(0) == 2)) + (1 * (local_var(0) == 3)))) + message_str(377, 251) + (message_str(377, 3000) + message_str(377, random(1105, 1107)) + message_str(377, random(3001, 3002))), Node982, 50);
+                if ((2 == ((5 * (local_var(0) == 1)) + (5 * (local_var(0) == 2)) + (5 * (local_var(0) == 3)))) and (dude_skill(SKILL_GAMBLING) >= 50)) then begin
+                    giq_option(1, SCRIPT_ZISLOTS, message_str(SCRIPT_ZISLOTS, 250) + (2 * ((1 * (local_var(0) == 1)) + (1 * (local_var(0) == 2)) + (1 * (local_var(0) == 3)))) + message_str(SCRIPT_ZISLOTS, 251) + (message_str(SCRIPT_ZISLOTS, 3000) + message_str(SCRIPT_ZISLOTS, random(1105, 1107)) + message_str(SCRIPT_ZISLOTS, random(3001, 3002))), Node982, NEUTRAL_REACTION);
                 end
                 else begin
-                    giq_option(1, 377, message_str(377, 250) + (2 * ((1 * (local_var(0) == 1)) + (1 * (local_var(0) == 2)) + (1 * (local_var(0) == 3)))) + message_str(377, 251), Node982, 50);
+                    giq_option(1, SCRIPT_ZISLOTS, message_str(SCRIPT_ZISLOTS, 250) + (2 * ((1 * (local_var(0) == 1)) + (1 * (local_var(0) == 2)) + (1 * (local_var(0) == 3)))) + message_str(SCRIPT_ZISLOTS, 251), Node982, NEUTRAL_REACTION);
                 end
             end
         end
         if ((((5 * (local_var(0) == 1)) + (5 * (local_var(0) == 2)) + (5 * (local_var(0) == 3))) >= 3) and (item_caps_total(dude_obj) >= (3 * ((1 * (local_var(0) == 1)) + (1 * (local_var(0) == 2)) + (1 * (local_var(0) == 3)))))) then begin
             if (item_caps_total(dude_obj) == ((1 * (local_var(0) == 1)) + (1 * (local_var(0) == 2)) + (1 * (local_var(0) == 3)))) then begin
-                giq_option(1, 377, message_str(377, 252) + (3 * ((1 * (local_var(0) == 1)) + (1 * (local_var(0) == 2)) + (1 * (local_var(0) == 3)))) + message_str(377, 253), Node983, 50);
+                giq_option(1, SCRIPT_ZISLOTS, message_str(SCRIPT_ZISLOTS, 252) + (3 * ((1 * (local_var(0) == 1)) + (1 * (local_var(0) == 2)) + (1 * (local_var(0) == 3)))) + message_str(SCRIPT_ZISLOTS, 253), Node983, NEUTRAL_REACTION);
             end
             else begin
-                if ((3 == ((5 * (local_var(0) == 1)) + (5 * (local_var(0) == 2)) + (5 * (local_var(0) == 3)))) and (has_skill(dude_obj, 16) >= 50)) then begin
-                    giq_option(1, 377, message_str(377, 250) + (3 * ((1 * (local_var(0) == 1)) + (1 * (local_var(0) == 2)) + (1 * (local_var(0) == 3)))) + message_str(377, 251) + (message_str(377, 3000) + message_str(377, random(1105, 1107)) + message_str(377, random(3001, 3002))), Node983, 50);
+                if ((3 == ((5 * (local_var(0) == 1)) + (5 * (local_var(0) == 2)) + (5 * (local_var(0) == 3)))) and (dude_skill(SKILL_GAMBLING) >= 50)) then begin
+                    giq_option(1, SCRIPT_ZISLOTS, message_str(SCRIPT_ZISLOTS, 250) + (3 * ((1 * (local_var(0) == 1)) + (1 * (local_var(0) == 2)) + (1 * (local_var(0) == 3)))) + message_str(SCRIPT_ZISLOTS, 251) + (message_str(SCRIPT_ZISLOTS, 3000) + message_str(SCRIPT_ZISLOTS, random(1105, 1107)) + message_str(SCRIPT_ZISLOTS, random(3001, 3002))), Node983, NEUTRAL_REACTION);
                 end
                 else begin
-                    giq_option(1, 377, message_str(377, 250) + (3 * ((1 * (local_var(0) == 1)) + (1 * (local_var(0) == 2)) + (1 * (local_var(0) == 3)))) + message_str(377, 251), Node983, 50);
+                    giq_option(1, SCRIPT_ZISLOTS, message_str(SCRIPT_ZISLOTS, 250) + (3 * ((1 * (local_var(0) == 1)) + (1 * (local_var(0) == 2)) + (1 * (local_var(0) == 3)))) + message_str(SCRIPT_ZISLOTS, 251), Node983, NEUTRAL_REACTION);
                 end
             end
         end
         if ((((5 * (local_var(0) == 1)) + (5 * (local_var(0) == 2)) + (5 * (local_var(0) == 3))) >= 4) and (item_caps_total(dude_obj) >= (4 * ((1 * (local_var(0) == 1)) + (1 * (local_var(0) == 2)) + (1 * (local_var(0) == 3)))))) then begin
             if (item_caps_total(dude_obj) == ((1 * (local_var(0) == 1)) + (1 * (local_var(0) == 2)) + (1 * (local_var(0) == 3)))) then begin
-                giq_option(1, 377, message_str(377, 252) + (4 * ((1 * (local_var(0) == 1)) + (1 * (local_var(0) == 2)) + (1 * (local_var(0) == 3)))) + message_str(377, 253), Node984, 50);
+                giq_option(1, SCRIPT_ZISLOTS, message_str(SCRIPT_ZISLOTS, 252) + (4 * ((1 * (local_var(0) == 1)) + (1 * (local_var(0) == 2)) + (1 * (local_var(0) == 3)))) + message_str(SCRIPT_ZISLOTS, 253), Node984, NEUTRAL_REACTION);
             end
             else begin
-                if ((4 == ((5 * (local_var(0) == 1)) + (5 * (local_var(0) == 2)) + (5 * (local_var(0) == 3)))) and (has_skill(dude_obj, 16) >= 50)) then begin
-                    giq_option(1, 377, message_str(377, 250) + (4 * ((1 * (local_var(0) == 1)) + (1 * (local_var(0) == 2)) + (1 * (local_var(0) == 3)))) + message_str(377, 251) + (message_str(377, 3000) + message_str(377, random(1105, 1107)) + message_str(377, random(3001, 3002))), Node984, 50);
+                if ((4 == ((5 * (local_var(0) == 1)) + (5 * (local_var(0) == 2)) + (5 * (local_var(0) == 3)))) and (dude_skill(SKILL_GAMBLING) >= 50)) then begin
+                    giq_option(1, SCRIPT_ZISLOTS, message_str(SCRIPT_ZISLOTS, 250) + (4 * ((1 * (local_var(0) == 1)) + (1 * (local_var(0) == 2)) + (1 * (local_var(0) == 3)))) + message_str(SCRIPT_ZISLOTS, 251) + (message_str(SCRIPT_ZISLOTS, 3000) + message_str(SCRIPT_ZISLOTS, random(1105, 1107)) + message_str(SCRIPT_ZISLOTS, random(3001, 3002))), Node984, NEUTRAL_REACTION);
                 end
                 else begin
-                    giq_option(1, 377, message_str(377, 250) + (4 * ((1 * (local_var(0) == 1)) + (1 * (local_var(0) == 2)) + (1 * (local_var(0) == 3)))) + message_str(377, 251), Node984, 50);
+                    giq_option(1, SCRIPT_ZISLOTS, message_str(SCRIPT_ZISLOTS, 250) + (4 * ((1 * (local_var(0) == 1)) + (1 * (local_var(0) == 2)) + (1 * (local_var(0) == 3)))) + message_str(SCRIPT_ZISLOTS, 251), Node984, NEUTRAL_REACTION);
                 end
             end
         end
         if ((((5 * (local_var(0) == 1)) + (5 * (local_var(0) == 2)) + (5 * (local_var(0) == 3))) >= 5) and (item_caps_total(dude_obj) >= (5 * ((1 * (local_var(0) == 1)) + (1 * (local_var(0) == 2)) + (1 * (local_var(0) == 3)))))) then begin
             if (item_caps_total(dude_obj) == ((1 * (local_var(0) == 1)) + (1 * (local_var(0) == 2)) + (1 * (local_var(0) == 3)))) then begin
-                giq_option(1, 377, message_str(377, 252) + (5 * ((1 * (local_var(0) == 1)) + (1 * (local_var(0) == 2)) + (1 * (local_var(0) == 3)))) + message_str(377, 253), Node985, 50);
+                giq_option(1, SCRIPT_ZISLOTS, message_str(SCRIPT_ZISLOTS, 252) + (5 * ((1 * (local_var(0) == 1)) + (1 * (local_var(0) == 2)) + (1 * (local_var(0) == 3)))) + message_str(SCRIPT_ZISLOTS, 253), Node985, NEUTRAL_REACTION);
             end
             else begin
-                if ((5 == ((5 * (local_var(0) == 1)) + (5 * (local_var(0) == 2)) + (5 * (local_var(0) == 3)))) and (has_skill(dude_obj, 16) >= 50)) then begin
-                    giq_option(1, 377, message_str(377, 250) + (5 * ((1 * (local_var(0) == 1)) + (1 * (local_var(0) == 2)) + (1 * (local_var(0) == 3)))) + message_str(377, 251) + (message_str(377, 3000) + message_str(377, random(1105, 1107)) + message_str(377, random(3001, 3002))), Node985, 50);
+                if ((5 == ((5 * (local_var(0) == 1)) + (5 * (local_var(0) == 2)) + (5 * (local_var(0) == 3)))) and (dude_skill(SKILL_GAMBLING) >= 50)) then begin
+                    giq_option(1, SCRIPT_ZISLOTS, message_str(SCRIPT_ZISLOTS, 250) + (5 * ((1 * (local_var(0) == 1)) + (1 * (local_var(0) == 2)) + (1 * (local_var(0) == 3)))) + message_str(SCRIPT_ZISLOTS, 251) + (message_str(SCRIPT_ZISLOTS, 3000) + message_str(SCRIPT_ZISLOTS, random(1105, 1107)) + message_str(SCRIPT_ZISLOTS, random(3001, 3002))), Node985, NEUTRAL_REACTION);
                 end
                 else begin
-                    giq_option(1, 377, message_str(377, 250) + (5 * ((1 * (local_var(0) == 1)) + (1 * (local_var(0) == 2)) + (1 * (local_var(0) == 3)))) + message_str(377, 251), Node985, 50);
+                    giq_option(1, SCRIPT_ZISLOTS, message_str(SCRIPT_ZISLOTS, 250) + (5 * ((1 * (local_var(0) == 1)) + (1 * (local_var(0) == 2)) + (1 * (local_var(0) == 3)))) + message_str(SCRIPT_ZISLOTS, 251), Node985, NEUTRAL_REACTION);
                 end
             end
         end
-        if (metarule(18, dude_obj) == 0) then begin
-            giq_option(1, 377, 254 + (item_caps_total(dude_obj) == 0) + (((game_time_hour <= 600) or (game_time_hour >= 1800)) * 2), Node999, 50);
+        if (drug_influence(dude_obj) == 0) then begin
+            giq_option(1, SCRIPT_ZISLOTS, 254 + (item_caps_total(dude_obj) == 0) + (((game_time_hour <= 600) or (game_time_hour >= 1800)) * 2), Node999, NEUTRAL_REACTION);
         end
         else begin
-            giq_option(1, 377, 258 + (get_critter_stat(dude_obj, 34) == 0), Node999, 50);
+            giq_option(1, SCRIPT_ZISLOTS, 258 + (dude_gender == 0), Node999, NEUTRAL_REACTION);
         end
     end
     return 0;
@@ -890,128 +890,128 @@ end
 procedure Node003
 begin
     if ((item_caps_total(dude_obj) < ((1 * (local_var(0) == 1)) + (1 * (local_var(0) == 2)) + (1 * (local_var(0) == 3)))) and ((local_var(1) == 1) == 0)) then begin
-        gsay_reply(377, 2175);
-        giq_option(1, 377, 2176, Node999, 50);
+        gsay_reply(SCRIPT_ZISLOTS, 2175);
+        giq_option(1, SCRIPT_ZISLOTS, 2176, Node999, NEUTRAL_REACTION);
     end
     else begin
-        gsay_reply(1505, 315);
-        giq_option(1, 377, 2186, Node999, 50);
+        Reply(315);
+        giq_option(1, SCRIPT_ZISLOTS, 2186, Node999, NEUTRAL_REACTION);
     end
 end
 
 procedure Node004
 begin
-    gsay_reply(1505, 230);
-    giq_option(-3, 377, message_str(377, 2101) + obj_name(dude_obj) + message_str(377, 2102), Node005, 50);
-    giq_option(-3, 377, message_str(377, 2103) + obj_name(dude_obj) + message_str(377, 2104), Node005, 50);
-    giq_option(-3, 377, 2105, Node006, 50);
-    giq_option(-3, 377, 2106, Node005, 50);
-    giq_option(-3, 377, 2107, Node005, 50);
-    giq_option(-3, 377, 2108, Node005, 50);
-    giq_option(-3, 377, 2109, Node999, 50);
-    giq_option(4, 377, 2110, Node999, 50);
+    Reply(230);
+    giq_option(-3, SCRIPT_ZISLOTS, message_str(SCRIPT_ZISLOTS, 2101) + dude_name + message_str(SCRIPT_ZISLOTS, 2102), Node005, NEUTRAL_REACTION);
+    giq_option(-3, SCRIPT_ZISLOTS, message_str(SCRIPT_ZISLOTS, 2103) + dude_name + message_str(SCRIPT_ZISLOTS, 2104), Node005, NEUTRAL_REACTION);
+    giq_option(-3, SCRIPT_ZISLOTS, 2105, Node006, NEUTRAL_REACTION);
+    giq_option(-3, SCRIPT_ZISLOTS, 2106, Node005, NEUTRAL_REACTION);
+    giq_option(-3, SCRIPT_ZISLOTS, 2107, Node005, NEUTRAL_REACTION);
+    giq_option(-3, SCRIPT_ZISLOTS, 2108, Node005, NEUTRAL_REACTION);
+    giq_option(-3, SCRIPT_ZISLOTS, 2109, Node999, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_ZISLOTS, 2110, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node005
 begin
-    gsay_reply(1505, 235);
-    giq_option(-3, 377, message_str(377, 2131) + obj_name(dude_obj) + message_str(377, 2132), Node005, 50);
-    giq_option(-3, 377, 2133, Node006, 50);
-    giq_option(-3, 377, 2134, Node005, 50);
-    giq_option(-3, 377, 2135, Node005, 50);
-    giq_option(-3, 377, 2136, Node005, 50);
-    giq_option(-3, 377, 2137, Node999, 50);
-    giq_option(4, 377, 2138, Node999, 50);
+    Reply(235);
+    giq_option(-3, SCRIPT_ZISLOTS, message_str(SCRIPT_ZISLOTS, 2131) + dude_name + message_str(SCRIPT_ZISLOTS, 2132), Node005, NEUTRAL_REACTION);
+    giq_option(-3, SCRIPT_ZISLOTS, 2133, Node006, NEUTRAL_REACTION);
+    giq_option(-3, SCRIPT_ZISLOTS, 2134, Node005, NEUTRAL_REACTION);
+    giq_option(-3, SCRIPT_ZISLOTS, 2135, Node005, NEUTRAL_REACTION);
+    giq_option(-3, SCRIPT_ZISLOTS, 2136, Node005, NEUTRAL_REACTION);
+    giq_option(-3, SCRIPT_ZISLOTS, 2137, Node999, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_ZISLOTS, 2138, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node006
 begin
-    gsay_reply(1505, 240);
-    giq_option(1, 377, 2152, Node999, 50);
-    giq_option(4, 377, 2153, Node999, 50);
+    Reply(240);
+    giq_option(1, SCRIPT_ZISLOTS, 2152, Node999, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_ZISLOTS, 2153, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node013
 begin
-    gsay_reply(1505, 270);
-    giq_option(1, 377, 2282, Node014, 50);
-    giq_option(1, 377, 2283, Node014, 50);
-    giq_option(1, 377, 2284, Node999, 50);
+    Reply(270);
+    giq_option(1, SCRIPT_ZISLOTS, 2282, Node014, NEUTRAL_REACTION);
+    giq_option(1, SCRIPT_ZISLOTS, 2283, Node014, NEUTRAL_REACTION);
+    giq_option(1, SCRIPT_ZISLOTS, 2284, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node014
 begin
-    gsay_reply(1505, 275);
-    giq_option(1, 377, 2302, Node015, 50);
-    giq_option(1, 377, 2303, Node001, 50);
-    if (get_critter_stat(dude_obj, 34) == 0) then begin
-        giq_option(1, 377, 2304, Node999, 50);
+    Reply(275);
+    giq_option(1, SCRIPT_ZISLOTS, 2302, Node015, NEUTRAL_REACTION);
+    giq_option(1, SCRIPT_ZISLOTS, 2303, Node001, NEUTRAL_REACTION);
+    if (dude_gender == 0) then begin
+        giq_option(1, SCRIPT_ZISLOTS, 2304, Node999, NEUTRAL_REACTION);
     end
-    giq_option(1, 377, 2305, Node999, 50);
+    giq_option(1, SCRIPT_ZISLOTS, 2305, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node015
 begin
-    gsay_reply(1505, 280);
-    giq_option(1, 377, 2322, Node016, 50);
-    giq_option(1, 377, 2303, Node001, 50);
-    if (get_critter_stat(dude_obj, 34) == 0) then begin
-        giq_option(1, 377, 2304, Node999, 50);
+    Reply(280);
+    giq_option(1, SCRIPT_ZISLOTS, 2322, Node016, NEUTRAL_REACTION);
+    giq_option(1, SCRIPT_ZISLOTS, 2303, Node001, NEUTRAL_REACTION);
+    if (dude_gender == 0) then begin
+        giq_option(1, SCRIPT_ZISLOTS, 2304, Node999, NEUTRAL_REACTION);
     end
-    giq_option(1, 377, 2305, Node999, 50);
+    giq_option(1, SCRIPT_ZISLOTS, 2305, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node016
 begin
-    gsay_reply(1505, 285);
-    if (global_var(494) == 8) then begin
-        giq_option(1, 377, 2345, Node017, 50);
-        giq_option(1, 377, 2346, Node017, 50);
-        giq_option(1, 377, 2347, Node017, 50);
+    Reply(285);
+    if (global_var(GVAR_FALLOUT_2) == 8) then begin
+        giq_option(1, SCRIPT_ZISLOTS, 2345, Node017, NEUTRAL_REACTION);
+        giq_option(1, SCRIPT_ZISLOTS, 2346, Node017, NEUTRAL_REACTION);
+        giq_option(1, SCRIPT_ZISLOTS, 2347, Node017, NEUTRAL_REACTION);
     end
     else begin
-        if (global_var(494) == 5) then begin
-            giq_option(1, 377, 2344, Node017, 50);
+        if (global_var(GVAR_FALLOUT_2) == 5) then begin
+            giq_option(1, SCRIPT_ZISLOTS, 2344, Node017, NEUTRAL_REACTION);
         end
         else begin
-            if (global_var(494) == 4) then begin
-                giq_option(1, 377, 2343, Node017, 50);
+            if (global_var(GVAR_FALLOUT_2) == 4) then begin
+                giq_option(1, SCRIPT_ZISLOTS, 2343, Node017, NEUTRAL_REACTION);
             end
             else begin
-                if (global_var(494) == 0) then begin
-                    giq_option(1, 377, 2342, Node017, 50);
+                if (global_var(GVAR_FALLOUT_2) == 0) then begin
+                    giq_option(1, SCRIPT_ZISLOTS, 2342, Node017, NEUTRAL_REACTION);
                 end
             end
         end
     end
-    giq_option(1, 377, 2348, Node017, 50);
-    giq_option(1, 377, 2305, Node999, 50);
+    giq_option(1, SCRIPT_ZISLOTS, 2348, Node017, NEUTRAL_REACTION);
+    giq_option(1, SCRIPT_ZISLOTS, 2305, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node017
 begin
-    giq_option(1, 377, 2372, Node018, 50);
-    giq_option(1, 377, 2373, Node018, 50);
-    giq_option(1, 377, 2305, Node999, 50);
+    giq_option(1, SCRIPT_ZISLOTS, 2372, Node018, NEUTRAL_REACTION);
+    giq_option(1, SCRIPT_ZISLOTS, 2373, Node018, NEUTRAL_REACTION);
+    giq_option(1, SCRIPT_ZISLOTS, 2305, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node018
 begin
-    gsay_reply(1505, 300);
-    giq_option(1, 1505, 301, Node019, 50);
+    Reply(300);
+    giq_option(1, NAME, 301, Node019, NEUTRAL_REACTION);
 end
 
 procedure Node019
 begin
-    gsay_reply(1505, 305);
-    giq_option(1, 1505, 306, Node999, 50);
+    Reply(305);
+    giq_option(1, NAME, 306, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node997
 begin
     variable LVar0 := 0;
     variable LVar1 := 0;
-    LVar1 := (-145 * (get_critter_stat(dude_obj, 6) < 9)) + (75 * (get_critter_stat(dude_obj, 6) >= 8)) + ((-50 * (local_var(3) == 1)) + (-25 * (local_var(3) == 2)) + (5 * (local_var(3) == 3)) + (10 * (local_var(3) == 4))) + (coin_mult * 10);
+    LVar1 := (-145 * (dude_luck < 9)) + (75 * (dude_luck >= 8)) + ((-50 * (local_var(3) == 1)) + (-25 * (local_var(3) == 2)) + (5 * (local_var(3) == 3)) + (10 * (local_var(3) == 4))) + (coin_mult * 10);
     debug_msg("machine bonus" + ((-50 * (local_var(3) == 1)) + (-25 * (local_var(3) == 2)) + (5 * (local_var(3) == 3)) + (10 * (local_var(3) == 4))));
     debug_msg("total penalty" + LVar1);
     LVar0 := roll_vs_skill(dude_obj, 16, LVar1);
@@ -1057,4 +1057,3 @@ begin
     coin_mult := 5;
     call Node997();
 end
-
