diff --git a/Mods/MegaMod/Original/NITULBOX.ssl b/Mods/MegaMod/ReDefine/NITULBOX.ssl
index 6f81415..fbb68f7 100644
--- a/Mods/MegaMod/Original/NITULBOX.ssl
+++ b/Mods/MegaMod/ReDefine/NITULBOX.ssl
@@ -155,53 +155,53 @@ end
 procedure ChooseFromList
 begin
     variable LVar0 := 0;
-    gsay_reply(1079, stored_msg);
-    if (message_str(353, top_msg - 1) != "") then begin
-        giq_option(1, 1079, message_str(353, 150), CLICK_PREV, 50);
+    Reply(stored_msg);
+    if (message_str(SCRIPT_NEWR1, top_msg - 1) != "") then begin
+        giq_option(1, NAME, message_str(SCRIPT_NEWR1, 150), CLICK_PREV, NEUTRAL_REACTION);
     end
     if (LVar0 == 0) then begin
-        if (message_str(353, top_msg + 0) != "") then begin
-            giq_option(1, 1079, message_str(353, top_msg + 0), CLICK_0, 50);
+        if (message_str(SCRIPT_NEWR1, top_msg + 0) != "") then begin
+            giq_option(1, NAME, message_str(SCRIPT_NEWR1, top_msg + 0), CLICK_0, NEUTRAL_REACTION);
         end
         else begin
             LVar0 := 1;
         end
     end
     if (LVar0 == 0) then begin
-        if (message_str(353, top_msg + 1) != "") then begin
-            giq_option(1, 1079, message_str(353, top_msg + 1), CLICK_1, 50);
+        if (message_str(SCRIPT_NEWR1, top_msg + 1) != "") then begin
+            giq_option(1, NAME, message_str(SCRIPT_NEWR1, top_msg + 1), CLICK_1, NEUTRAL_REACTION);
         end
         else begin
             LVar0 := 1;
         end
     end
     if (LVar0 == 0) then begin
-        if (message_str(353, top_msg + 2) != "") then begin
-            giq_option(1, 1079, message_str(353, top_msg + 2), CLICK_2, 50);
+        if (message_str(SCRIPT_NEWR1, top_msg + 2) != "") then begin
+            giq_option(1, NAME, message_str(SCRIPT_NEWR1, top_msg + 2), CLICK_2, NEUTRAL_REACTION);
         end
         else begin
             LVar0 := 1;
         end
     end
     if (LVar0 == 0) then begin
-        if (message_str(353, top_msg + 3) != "") then begin
-            giq_option(1, 1079, message_str(353, top_msg + 3), CLICK_3, 50);
+        if (message_str(SCRIPT_NEWR1, top_msg + 3) != "") then begin
+            giq_option(1, NAME, message_str(SCRIPT_NEWR1, top_msg + 3), CLICK_3, NEUTRAL_REACTION);
         end
         else begin
             LVar0 := 1;
         end
     end
     if (LVar0 == 0) then begin
-        if (message_str(353, top_msg + 4) != "") then begin
-            giq_option(1, 1079, message_str(353, top_msg + 4), CLICK_4, 50);
+        if (message_str(SCRIPT_NEWR1, top_msg + 4) != "") then begin
+            giq_option(1, NAME, message_str(SCRIPT_NEWR1, top_msg + 4), CLICK_4, NEUTRAL_REACTION);
         end
         else begin
             LVar0 := 1;
         end
     end
     if (LVar0 == 0) then begin
-        if (message_str(353, top_msg + 5) != "") then begin
-            giq_option(1, 1079, message_str(353, 151), CLICK_MORE, 50);
+        if (message_str(SCRIPT_NEWR1, top_msg + 5) != "") then begin
+            giq_option(1, NAME, message_str(SCRIPT_NEWR1, 151), CLICK_MORE, NEUTRAL_REACTION);
         end
     end
 end
@@ -256,8 +256,8 @@ end
 procedure map_enter_p_proc
 begin
     new_reno_poor_box_obj := self_obj;
-    if (metarule(22, 0) == 0) then begin
-        if (((global_var(495) bwand 256) != 0) == 0) then begin
+    if (is_loading_game == 0) then begin
+        if (((global_var(GVAR_NEW_RENO_FLAG_3) bwand 256) != 0) == 0) then begin
             if (local_var(0) < game_time) then begin
                 item_caps_adjust(self_obj, random(1, 50) - item_caps_total(self_obj));
                 set_local_var(1, item_caps_total(self_obj));
@@ -272,49 +272,49 @@ begin
     if (local_var(1) > item_caps_total(self_obj)) then begin
         set_local_var(1, item_caps_total(self_obj));
         if (((local_var(2) bwand 1024) != 0) == 0) then begin
-            set_global_var(0, global_var(0) - 3);
-            set_global_var(37, 0);
-            set_global_var(38, 0);
-            set_global_var(39, 0);
-            set_global_var(40, 0);
-            set_global_var(41, 0);
-            set_global_var(42, 0);
-            set_global_var(43, 0);
-            set_global_var(44, 0);
-            set_global_var(45, 0);
-            if (global_var(0) >= 1000) then begin
-                set_global_var(37, 1);
+            set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) - 3);
+            set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+            set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+            set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+            set_global_var(GVAR_KARMA_DEFENDER, 0);
+            set_global_var(GVAR_KARMA_WANDERER, 0);
+            set_global_var(GVAR_KARMA_BETRAYER, 0);
+            set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+            set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+            set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+                set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
             end
             else begin
-                if (global_var(0) >= 750) then begin
-                    set_global_var(38, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                    set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
                 end
                 else begin
-                    if (global_var(0) >= 500) then begin
-                        set_global_var(39, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                        set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                     end
                     else begin
-                        if (global_var(0) >= 250) then begin
-                            set_global_var(40, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                            set_global_var(GVAR_KARMA_DEFENDER, 1);
                         end
                         else begin
-                            if (global_var(0) > -250) then begin
-                                set_global_var(41, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                                set_global_var(GVAR_KARMA_WANDERER, 1);
                             end
                             else begin
-                                if (global_var(0) > -500) then begin
-                                    set_global_var(42, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                    set_global_var(GVAR_KARMA_BETRAYER, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -750) then begin
-                                        set_global_var(43, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                        set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                     end
                                     else begin
-                                        if (global_var(0) > -1000) then begin
-                                            set_global_var(44, 1);
+                                        if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                            set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                         end
                                         else begin
-                                            set_global_var(45, 1);
+                                            set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                         end
                                     end
                                 end
@@ -323,7 +323,7 @@ begin
                     end
                 end
             end
-            set_global_var(55, global_var(55) + 1);
+            set_global_var(GVAR_TOWN_REP_NEW_RENO, global_var(GVAR_TOWN_REP_NEW_RENO) + 1);
         end
         set_local_var(2, local_var(2) bwor 1024);
     end
@@ -331,50 +331,50 @@ begin
         if (local_var(1) < item_caps_total(self_obj)) then begin
             set_local_var(1, item_caps_total(self_obj));
             if (((local_var(2) bwand 512) != 0) == 0) then begin
-                set_global_var(0, global_var(0) + 3);
+                set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + 3);
                 debug_msg("Player gains " + 3 + " Karma Points.");
-                set_global_var(37, 0);
-                set_global_var(38, 0);
-                set_global_var(39, 0);
-                set_global_var(40, 0);
-                set_global_var(41, 0);
-                set_global_var(42, 0);
-                set_global_var(43, 0);
-                set_global_var(44, 0);
-                set_global_var(45, 0);
-                if (global_var(0) >= 1000) then begin
-                    set_global_var(37, 1);
+                set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+                set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+                set_global_var(GVAR_KARMA_DEFENDER, 0);
+                set_global_var(GVAR_KARMA_WANDERER, 0);
+                set_global_var(GVAR_KARMA_BETRAYER, 0);
+                set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+                set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+                set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+                    set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
                 end
                 else begin
-                    if (global_var(0) >= 750) then begin
-                        set_global_var(38, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
                     end
                     else begin
-                        if (global_var(0) >= 500) then begin
-                            set_global_var(39, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                            set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                         end
                         else begin
-                            if (global_var(0) >= 250) then begin
-                                set_global_var(40, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                                set_global_var(GVAR_KARMA_DEFENDER, 1);
                             end
                             else begin
-                                if (global_var(0) > -250) then begin
-                                    set_global_var(41, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                                    set_global_var(GVAR_KARMA_WANDERER, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -500) then begin
-                                        set_global_var(42, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                        set_global_var(GVAR_KARMA_BETRAYER, 1);
                                     end
                                     else begin
-                                        if (global_var(0) > -750) then begin
-                                            set_global_var(43, 1);
+                                        if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                            set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                         end
                                         else begin
-                                            if (global_var(0) > -1000) then begin
-                                                set_global_var(44, 1);
+                                            if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                                set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                             end
                                             else begin
-                                                set_global_var(45, 1);
+                                                set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                             end
                                         end
                                     end
@@ -383,10 +383,9 @@ begin
                         end
                     end
                 end
-                set_global_var(55, global_var(55) + 1);
+                set_global_var(GVAR_TOWN_REP_NEW_RENO, global_var(GVAR_TOWN_REP_NEW_RENO) + 1);
             end
             set_local_var(2, local_var(2) bwor 512);
         end
     end
 end
-
