diff --git a/Mods/MegaMod/Original/NIWRISTI.ssl b/Mods/MegaMod/ReDefine/NIWRISTI.ssl
index d083477..1bd9d7a 100644
--- a/Mods/MegaMod/Original/NIWRISTI.ssl
+++ b/Mods/MegaMod/ReDefine/NIWRISTI.ssl
@@ -165,53 +165,53 @@ end
 procedure ChooseFromList
 begin
     variable LVar0 := 0;
-    gsay_reply(655, stored_msg);
-    if (message_str(353, top_msg - 1) != "") then begin
-        giq_option(1, 655, message_str(353, 150), CLICK_PREV, 50);
+    Reply(stored_msg);
+    if (message_str(SCRIPT_NEWR1, top_msg - 1) != "") then begin
+        giq_option(1, NAME, message_str(SCRIPT_NEWR1, 150), CLICK_PREV, NEUTRAL_REACTION);
     end
     if (LVar0 == 0) then begin
-        if (message_str(353, top_msg + 0) != "") then begin
-            giq_option(1, 655, message_str(353, top_msg + 0), CLICK_0, 50);
+        if (message_str(SCRIPT_NEWR1, top_msg + 0) != "") then begin
+            giq_option(1, NAME, message_str(SCRIPT_NEWR1, top_msg + 0), CLICK_0, NEUTRAL_REACTION);
         end
         else begin
             LVar0 := 1;
         end
     end
     if (LVar0 == 0) then begin
-        if (message_str(353, top_msg + 1) != "") then begin
-            giq_option(1, 655, message_str(353, top_msg + 1), CLICK_1, 50);
+        if (message_str(SCRIPT_NEWR1, top_msg + 1) != "") then begin
+            giq_option(1, NAME, message_str(SCRIPT_NEWR1, top_msg + 1), CLICK_1, NEUTRAL_REACTION);
         end
         else begin
             LVar0 := 1;
         end
     end
     if (LVar0 == 0) then begin
-        if (message_str(353, top_msg + 2) != "") then begin
-            giq_option(1, 655, message_str(353, top_msg + 2), CLICK_2, 50);
+        if (message_str(SCRIPT_NEWR1, top_msg + 2) != "") then begin
+            giq_option(1, NAME, message_str(SCRIPT_NEWR1, top_msg + 2), CLICK_2, NEUTRAL_REACTION);
         end
         else begin
             LVar0 := 1;
         end
     end
     if (LVar0 == 0) then begin
-        if (message_str(353, top_msg + 3) != "") then begin
-            giq_option(1, 655, message_str(353, top_msg + 3), CLICK_3, 50);
+        if (message_str(SCRIPT_NEWR1, top_msg + 3) != "") then begin
+            giq_option(1, NAME, message_str(SCRIPT_NEWR1, top_msg + 3), CLICK_3, NEUTRAL_REACTION);
         end
         else begin
             LVar0 := 1;
         end
     end
     if (LVar0 == 0) then begin
-        if (message_str(353, top_msg + 4) != "") then begin
-            giq_option(1, 655, message_str(353, top_msg + 4), CLICK_4, 50);
+        if (message_str(SCRIPT_NEWR1, top_msg + 4) != "") then begin
+            giq_option(1, NAME, message_str(SCRIPT_NEWR1, top_msg + 4), CLICK_4, NEUTRAL_REACTION);
         end
         else begin
             LVar0 := 1;
         end
     end
     if (LVar0 == 0) then begin
-        if (message_str(353, top_msg + 5) != "") then begin
-            giq_option(1, 655, message_str(353, 151), CLICK_MORE, 50);
+        if (message_str(SCRIPT_NEWR1, top_msg + 5) != "") then begin
+            giq_option(1, NAME, message_str(SCRIPT_NEWR1, 151), CLICK_MORE, NEUTRAL_REACTION);
         end
     end
     return 0;
@@ -270,34 +270,34 @@ end
 
 procedure look_at_p_proc
 begin
-    set_global_var(285, global_var(285) bwor 128);
+    set_global_var(GVAR_NEW_RENO_WRIGHT_FLAGS, global_var(GVAR_NEW_RENO_WRIGHT_FLAGS) bwor 128);
     script_overrides;
-    if (global_var(344) == 2) then begin
-        display_msg(message_str(655, 120));
+    if (global_var(GVAR_NEW_RENO_WRIGHT_STILL) == 2) then begin
+        display_mstr(120);
     end
     else begin
-        if (global_var(344) == 1) then begin
-            display_msg(message_str(655, 110));
+        if (global_var(GVAR_NEW_RENO_WRIGHT_STILL) == 1) then begin
+            display_mstr(110);
         end
         else begin
-            display_msg(message_str(655, 100));
+            display_mstr(100);
         end
     end
 end
 
 procedure description_p_proc
 begin
-    set_global_var(285, global_var(285) bwor 128);
+    set_global_var(GVAR_NEW_RENO_WRIGHT_FLAGS, global_var(GVAR_NEW_RENO_WRIGHT_FLAGS) bwor 128);
     script_overrides;
-    if (global_var(344) == 2) then begin
-        display_msg(message_str(655, 120));
+    if (global_var(GVAR_NEW_RENO_WRIGHT_STILL) == 2) then begin
+        display_mstr(120);
     end
     else begin
-        if (global_var(344) == 1) then begin
-            display_msg(message_str(655, 110));
+        if (global_var(GVAR_NEW_RENO_WRIGHT_STILL) == 1) then begin
+            display_mstr(110);
         end
         else begin
-            display_msg(message_str(655, 100));
+            display_mstr(100);
         end
     end
 end
@@ -305,94 +305,94 @@ end
 procedure damage_p_proc
 begin
     tampered := 0;
-    if ((global_var(344) == 2) == 0) then begin
+    if ((global_var(GVAR_NEW_RENO_WRIGHT_STILL) == 2) == 0) then begin
         if (source_obj != 0) then begin
-            if (metarule(49, source_obj) == 6) then begin
+            if (weapon_dmg_type(source_obj) == 6) then begin
                 if (2 != 1) then begin
                     if (2 == 2) then begin
-                        if ((global_var(344) == 2) == 0) then begin
-                            set_global_var(55, global_var(55) + -5);
-                            display_msg(message_str(655, 200));
-                            if (global_var(346) == 1) then begin
+                        if ((global_var(GVAR_NEW_RENO_WRIGHT_STILL) == 2) == 0) then begin
+                            set_global_var(GVAR_TOWN_REP_NEW_RENO, global_var(GVAR_TOWN_REP_NEW_RENO) + -5);
+                            display_mstr(200);
+                            if (global_var(GVAR_NEW_RENO_WRIGHT_STILL_MISSION) == 1) then begin
                                 give_exp_points(500);
-                                display_msg(message_str(14, 100) + 500 + message_str(14, 101));
+                                display_msg(g_mstr(100) + 500 + g_mstr(101));
                             end
-                            if (((4 == 1) and (global_var(134) <= -1) or ((4 == 2) and (global_var(135) <= -1)) or ((4 == 3) and (global_var(136) <= -1)) or ((4 == 4) and (global_var(216) <= -1))) == 0) then begin
+                            if (((4 == 1) and (global_var(GVAR_SALVATORE_FAMILY_COUNTER) <= -1) or ((4 == 2) and (global_var(GVAR_BISHOP_FAMILY_COUNTER) <= -1)) or ((4 == 3) and (global_var(GVAR_MORDINO_FAMILY_COUNTER) <= -1)) or ((4 == 4) and (global_var(GVAR_WRIGHT_FAMILY_COUNTER) <= -1))) == 0) then begin
                                 if (4 == 1) then begin
-                                    set_global_var(134, global_var(134) - 20);
+                                    set_global_var(GVAR_SALVATORE_FAMILY_COUNTER, global_var(GVAR_SALVATORE_FAMILY_COUNTER) - 20);
                                 end
                                 else begin
                                     if (4 == 2) then begin
-                                        set_global_var(135, global_var(135) - 20);
+                                        set_global_var(GVAR_BISHOP_FAMILY_COUNTER, global_var(GVAR_BISHOP_FAMILY_COUNTER) - 20);
                                     end
                                     else begin
                                         if (4 == 3) then begin
-                                            set_global_var(136, global_var(136) - 20);
+                                            set_global_var(GVAR_MORDINO_FAMILY_COUNTER, global_var(GVAR_MORDINO_FAMILY_COUNTER) - 20);
                                         end
                                         else begin
                                             if (4 == 4) then begin
-                                                set_global_var(216, global_var(216) - 20);
+                                                set_global_var(GVAR_WRIGHT_FAMILY_COUNTER, global_var(GVAR_WRIGHT_FAMILY_COUNTER) - 20);
                                             end
                                         end
                                     end
                                 end
                             end
                             debug_msg("\nfamily standings: ");
-                            debug_msg("\n  Salvatore == " + global_var(134));
-                            debug_msg("\n  Bishop == " + global_var(135));
-                            debug_msg("\n  Mordino == " + global_var(136));
-                            debug_msg("\n  Wright == " + global_var(216));
+                            debug_msg("\n  Salvatore == " + global_var(GVAR_SALVATORE_FAMILY_COUNTER));
+                            debug_msg("\n  Bishop == " + global_var(GVAR_BISHOP_FAMILY_COUNTER));
+                            debug_msg("\n  Mordino == " + global_var(GVAR_MORDINO_FAMILY_COUNTER));
+                            debug_msg("\n  Wright == " + global_var(GVAR_WRIGHT_FAMILY_COUNTER));
                         end
                     end
-                    set_global_var(552, 0);
+                    set_global_var(GVAR_NEW_RENO_POISON_STILL_TIME, 0);
                 end
                 else begin
-                    if (global_var(552) == 0) then begin
-                        if (global_var(344) == 0) then begin
-                            set_global_var(0, global_var(0) + -30);
+                    if (global_var(GVAR_NEW_RENO_POISON_STILL_TIME) == 0) then begin
+                        if (global_var(GVAR_NEW_RENO_WRIGHT_STILL) == 0) then begin
+                            set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + -30);
                             debug_msg("Player gains " + -30 + " Karma Points.");
-                            set_global_var(37, 0);
-                            set_global_var(38, 0);
-                            set_global_var(39, 0);
-                            set_global_var(40, 0);
-                            set_global_var(41, 0);
-                            set_global_var(42, 0);
-                            set_global_var(43, 0);
-                            set_global_var(44, 0);
-                            set_global_var(45, 0);
-                            if (global_var(0) >= 1000) then begin
-                                set_global_var(37, 1);
+                            set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+                            set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+                            set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+                            set_global_var(GVAR_KARMA_DEFENDER, 0);
+                            set_global_var(GVAR_KARMA_WANDERER, 0);
+                            set_global_var(GVAR_KARMA_BETRAYER, 0);
+                            set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+                            set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+                            set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+                            if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+                                set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
                             end
                             else begin
-                                if (global_var(0) >= 750) then begin
-                                    set_global_var(38, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                                    set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
                                 end
                                 else begin
-                                    if (global_var(0) >= 500) then begin
-                                        set_global_var(39, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                                        set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                                     end
                                     else begin
-                                        if (global_var(0) >= 250) then begin
-                                            set_global_var(40, 1);
+                                        if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                                            set_global_var(GVAR_KARMA_DEFENDER, 1);
                                         end
                                         else begin
-                                            if (global_var(0) > -250) then begin
-                                                set_global_var(41, 1);
+                                            if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                                                set_global_var(GVAR_KARMA_WANDERER, 1);
                                             end
                                             else begin
-                                                if (global_var(0) > -500) then begin
-                                                    set_global_var(42, 1);
+                                                if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                                    set_global_var(GVAR_KARMA_BETRAYER, 1);
                                                 end
                                                 else begin
-                                                    if (global_var(0) > -750) then begin
-                                                        set_global_var(43, 1);
+                                                    if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                                        set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                                     end
                                                     else begin
-                                                        if (global_var(0) > -1000) then begin
-                                                            set_global_var(44, 1);
+                                                        if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                                            set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                                         end
                                                         else begin
-                                                            set_global_var(45, 1);
+                                                            set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                                         end
                                                     end
                                                 end
@@ -402,108 +402,108 @@ begin
                                 end
                             end
                         end
-                        set_global_var(552, game_time + (3 * (24 * (60 * (60 * 10)))));
+                        set_global_var(GVAR_NEW_RENO_POISON_STILL_TIME, game_time + (3 * (24 * (60 * (60 * 10)))));
                     end
                 end
-                set_global_var(344, 2);
-                display_msg(message_str(655, 300));
+                set_global_var(GVAR_NEW_RENO_WRIGHT_STILL, 2);
+                display_mstr(300);
                 if (new_reno_wright_teen_obj != 0) then begin
                     tampered := 1;
                     last_timed_obj := new_reno_wright_teen_obj;
                     last_timed_param := last_timed_param;
-                    metarule3(100, new_reno_wright_teen_obj, 19, 0);
+                    rm_fixed_timer_event(new_reno_wright_teen_obj, 19);
                     add_timer_event(new_reno_wright_teen_obj, 0, 19);
                 end
             end
         end
-        if ((global_var(344) == 2) == 0) then begin
+        if ((global_var(GVAR_NEW_RENO_WRIGHT_STILL) == 2) == 0) then begin
             if (target_obj != 0) then begin
-                if (metarule(49, target_obj) == 6) then begin
+                if (weapon_dmg_type(target_obj) == 6) then begin
                     if (2 != 1) then begin
                         if (2 == 2) then begin
-                            if ((global_var(344) == 2) == 0) then begin
-                                set_global_var(55, global_var(55) + -5);
-                                display_msg(message_str(655, 200));
-                                if (global_var(346) == 1) then begin
+                            if ((global_var(GVAR_NEW_RENO_WRIGHT_STILL) == 2) == 0) then begin
+                                set_global_var(GVAR_TOWN_REP_NEW_RENO, global_var(GVAR_TOWN_REP_NEW_RENO) + -5);
+                                display_mstr(200);
+                                if (global_var(GVAR_NEW_RENO_WRIGHT_STILL_MISSION) == 1) then begin
                                     give_exp_points(500);
-                                    display_msg(message_str(14, 100) + 500 + message_str(14, 101));
+                                    display_msg(g_mstr(100) + 500 + g_mstr(101));
                                 end
-                                if (((4 == 1) and (global_var(134) <= -1) or ((4 == 2) and (global_var(135) <= -1)) or ((4 == 3) and (global_var(136) <= -1)) or ((4 == 4) and (global_var(216) <= -1))) == 0) then begin
+                                if (((4 == 1) and (global_var(GVAR_SALVATORE_FAMILY_COUNTER) <= -1) or ((4 == 2) and (global_var(GVAR_BISHOP_FAMILY_COUNTER) <= -1)) or ((4 == 3) and (global_var(GVAR_MORDINO_FAMILY_COUNTER) <= -1)) or ((4 == 4) and (global_var(GVAR_WRIGHT_FAMILY_COUNTER) <= -1))) == 0) then begin
                                     if (4 == 1) then begin
-                                        set_global_var(134, global_var(134) - 20);
+                                        set_global_var(GVAR_SALVATORE_FAMILY_COUNTER, global_var(GVAR_SALVATORE_FAMILY_COUNTER) - 20);
                                     end
                                     else begin
                                         if (4 == 2) then begin
-                                            set_global_var(135, global_var(135) - 20);
+                                            set_global_var(GVAR_BISHOP_FAMILY_COUNTER, global_var(GVAR_BISHOP_FAMILY_COUNTER) - 20);
                                         end
                                         else begin
                                             if (4 == 3) then begin
-                                                set_global_var(136, global_var(136) - 20);
+                                                set_global_var(GVAR_MORDINO_FAMILY_COUNTER, global_var(GVAR_MORDINO_FAMILY_COUNTER) - 20);
                                             end
                                             else begin
                                                 if (4 == 4) then begin
-                                                    set_global_var(216, global_var(216) - 20);
+                                                    set_global_var(GVAR_WRIGHT_FAMILY_COUNTER, global_var(GVAR_WRIGHT_FAMILY_COUNTER) - 20);
                                                 end
                                             end
                                         end
                                     end
                                 end
                                 debug_msg("\nfamily standings: ");
-                                debug_msg("\n  Salvatore == " + global_var(134));
-                                debug_msg("\n  Bishop == " + global_var(135));
-                                debug_msg("\n  Mordino == " + global_var(136));
-                                debug_msg("\n  Wright == " + global_var(216));
+                                debug_msg("\n  Salvatore == " + global_var(GVAR_SALVATORE_FAMILY_COUNTER));
+                                debug_msg("\n  Bishop == " + global_var(GVAR_BISHOP_FAMILY_COUNTER));
+                                debug_msg("\n  Mordino == " + global_var(GVAR_MORDINO_FAMILY_COUNTER));
+                                debug_msg("\n  Wright == " + global_var(GVAR_WRIGHT_FAMILY_COUNTER));
                             end
                         end
-                        set_global_var(552, 0);
+                        set_global_var(GVAR_NEW_RENO_POISON_STILL_TIME, 0);
                     end
                     else begin
-                        if (global_var(552) == 0) then begin
-                            if (global_var(344) == 0) then begin
-                                set_global_var(0, global_var(0) + -30);
+                        if (global_var(GVAR_NEW_RENO_POISON_STILL_TIME) == 0) then begin
+                            if (global_var(GVAR_NEW_RENO_WRIGHT_STILL) == 0) then begin
+                                set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + -30);
                                 debug_msg("Player gains " + -30 + " Karma Points.");
-                                set_global_var(37, 0);
-                                set_global_var(38, 0);
-                                set_global_var(39, 0);
-                                set_global_var(40, 0);
-                                set_global_var(41, 0);
-                                set_global_var(42, 0);
-                                set_global_var(43, 0);
-                                set_global_var(44, 0);
-                                set_global_var(45, 0);
-                                if (global_var(0) >= 1000) then begin
-                                    set_global_var(37, 1);
+                                set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+                                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+                                set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+                                set_global_var(GVAR_KARMA_DEFENDER, 0);
+                                set_global_var(GVAR_KARMA_WANDERER, 0);
+                                set_global_var(GVAR_KARMA_BETRAYER, 0);
+                                set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+                                set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+                                set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+                                if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+                                    set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
                                 end
                                 else begin
-                                    if (global_var(0) >= 750) then begin
-                                        set_global_var(38, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                                        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
                                     end
                                     else begin
-                                        if (global_var(0) >= 500) then begin
-                                            set_global_var(39, 1);
+                                        if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                                            set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                                         end
                                         else begin
-                                            if (global_var(0) >= 250) then begin
-                                                set_global_var(40, 1);
+                                            if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                                                set_global_var(GVAR_KARMA_DEFENDER, 1);
                                             end
                                             else begin
-                                                if (global_var(0) > -250) then begin
-                                                    set_global_var(41, 1);
+                                                if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                                                    set_global_var(GVAR_KARMA_WANDERER, 1);
                                                 end
                                                 else begin
-                                                    if (global_var(0) > -500) then begin
-                                                        set_global_var(42, 1);
+                                                    if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                                        set_global_var(GVAR_KARMA_BETRAYER, 1);
                                                     end
                                                     else begin
-                                                        if (global_var(0) > -750) then begin
-                                                            set_global_var(43, 1);
+                                                        if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                                            set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                                         end
                                                         else begin
-                                                            if (global_var(0) > -1000) then begin
-                                                                set_global_var(44, 1);
+                                                            if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                                                set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                                             end
                                                             else begin
-                                                                set_global_var(45, 1);
+                                                                set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                                             end
                                                         end
                                                     end
@@ -513,16 +513,16 @@ begin
                                     end
                                 end
                             end
-                            set_global_var(552, game_time + (3 * (24 * (60 * (60 * 10)))));
+                            set_global_var(GVAR_NEW_RENO_POISON_STILL_TIME, game_time + (3 * (24 * (60 * (60 * 10)))));
                         end
                     end
-                    set_global_var(344, 2);
-                    display_msg(message_str(655, 300));
+                    set_global_var(GVAR_NEW_RENO_WRIGHT_STILL, 2);
+                    display_mstr(300);
                     if (new_reno_wright_teen_obj != 0) then begin
                         tampered := 1;
                         last_timed_obj := new_reno_wright_teen_obj;
                         last_timed_param := last_timed_param;
-                        metarule3(100, new_reno_wright_teen_obj, 19, 0);
+                        rm_fixed_timer_event(new_reno_wright_teen_obj, 19);
                         add_timer_event(new_reno_wright_teen_obj, 0, 19);
                     end
                 end
@@ -533,7 +533,7 @@ begin
         if (new_reno_wright_teen_obj != 0) then begin
             last_timed_obj := new_reno_wright_teen_obj;
             last_timed_param := last_timed_param;
-            metarule3(100, new_reno_wright_teen_obj, 19, 0);
+            rm_fixed_timer_event(new_reno_wright_teen_obj, 19);
             add_timer_event(new_reno_wright_teen_obj, 0, 19);
         end
     end
@@ -541,10 +541,10 @@ end
 
 procedure map_enter_p_proc
 begin
-    if (metarule(14, 0)) then begin
+    if (map_first_run) then begin
         last_timed_obj := self_obj;
         last_timed_param := last_timed_param;
-        metarule3(100, self_obj, 1, 0);
+        rm_fixed_timer_event(self_obj, 1);
         add_timer_event(self_obj, game_ticks(5) + random(0, game_ticks(15 - 5)), 1);
     end
 end
@@ -558,7 +558,7 @@ begin
             script_overrides;
             last_timed_obj := new_reno_wright_teen_obj;
             last_timed_param := last_timed_param;
-            metarule3(100, new_reno_wright_teen_obj, 19, 0);
+            rm_fixed_timer_event(new_reno_wright_teen_obj, 19);
             add_timer_event(new_reno_wright_teen_obj, 0, 19);
         end
         else begin
@@ -570,92 +570,92 @@ begin
             the_roll := roll_vs_skill(dude_obj, action_being_used, 0);
             if (is_success(the_roll)) then begin
                 script_overrides;
-                if ((global_var(344) == 2) == 0) then begin
+                if ((global_var(GVAR_NEW_RENO_WRIGHT_STILL) == 2) == 0) then begin
                     if (2 != 1) then begin
                         if (2 == 2) then begin
-                            if ((global_var(344) == 2) == 0) then begin
-                                set_global_var(55, global_var(55) + -5);
-                                display_msg(message_str(655, 200));
-                                if (global_var(346) == 1) then begin
+                            if ((global_var(GVAR_NEW_RENO_WRIGHT_STILL) == 2) == 0) then begin
+                                set_global_var(GVAR_TOWN_REP_NEW_RENO, global_var(GVAR_TOWN_REP_NEW_RENO) + -5);
+                                display_mstr(200);
+                                if (global_var(GVAR_NEW_RENO_WRIGHT_STILL_MISSION) == 1) then begin
                                     give_exp_points(500);
-                                    display_msg(message_str(14, 100) + 500 + message_str(14, 101));
+                                    display_msg(g_mstr(100) + 500 + g_mstr(101));
                                 end
-                                if (((4 == 1) and (global_var(134) <= -1) or ((4 == 2) and (global_var(135) <= -1)) or ((4 == 3) and (global_var(136) <= -1)) or ((4 == 4) and (global_var(216) <= -1))) == 0) then begin
+                                if (((4 == 1) and (global_var(GVAR_SALVATORE_FAMILY_COUNTER) <= -1) or ((4 == 2) and (global_var(GVAR_BISHOP_FAMILY_COUNTER) <= -1)) or ((4 == 3) and (global_var(GVAR_MORDINO_FAMILY_COUNTER) <= -1)) or ((4 == 4) and (global_var(GVAR_WRIGHT_FAMILY_COUNTER) <= -1))) == 0) then begin
                                     if (4 == 1) then begin
-                                        set_global_var(134, global_var(134) - 20);
+                                        set_global_var(GVAR_SALVATORE_FAMILY_COUNTER, global_var(GVAR_SALVATORE_FAMILY_COUNTER) - 20);
                                     end
                                     else begin
                                         if (4 == 2) then begin
-                                            set_global_var(135, global_var(135) - 20);
+                                            set_global_var(GVAR_BISHOP_FAMILY_COUNTER, global_var(GVAR_BISHOP_FAMILY_COUNTER) - 20);
                                         end
                                         else begin
                                             if (4 == 3) then begin
-                                                set_global_var(136, global_var(136) - 20);
+                                                set_global_var(GVAR_MORDINO_FAMILY_COUNTER, global_var(GVAR_MORDINO_FAMILY_COUNTER) - 20);
                                             end
                                             else begin
                                                 if (4 == 4) then begin
-                                                    set_global_var(216, global_var(216) - 20);
+                                                    set_global_var(GVAR_WRIGHT_FAMILY_COUNTER, global_var(GVAR_WRIGHT_FAMILY_COUNTER) - 20);
                                                 end
                                             end
                                         end
                                     end
                                 end
                                 debug_msg("\nfamily standings: ");
-                                debug_msg("\n  Salvatore == " + global_var(134));
-                                debug_msg("\n  Bishop == " + global_var(135));
-                                debug_msg("\n  Mordino == " + global_var(136));
-                                debug_msg("\n  Wright == " + global_var(216));
+                                debug_msg("\n  Salvatore == " + global_var(GVAR_SALVATORE_FAMILY_COUNTER));
+                                debug_msg("\n  Bishop == " + global_var(GVAR_BISHOP_FAMILY_COUNTER));
+                                debug_msg("\n  Mordino == " + global_var(GVAR_MORDINO_FAMILY_COUNTER));
+                                debug_msg("\n  Wright == " + global_var(GVAR_WRIGHT_FAMILY_COUNTER));
                             end
                         end
-                        set_global_var(552, 0);
+                        set_global_var(GVAR_NEW_RENO_POISON_STILL_TIME, 0);
                     end
                     else begin
-                        if (global_var(552) == 0) then begin
-                            if (global_var(344) == 0) then begin
-                                set_global_var(0, global_var(0) + -30);
+                        if (global_var(GVAR_NEW_RENO_POISON_STILL_TIME) == 0) then begin
+                            if (global_var(GVAR_NEW_RENO_WRIGHT_STILL) == 0) then begin
+                                set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + -30);
                                 debug_msg("Player gains " + -30 + " Karma Points.");
-                                set_global_var(37, 0);
-                                set_global_var(38, 0);
-                                set_global_var(39, 0);
-                                set_global_var(40, 0);
-                                set_global_var(41, 0);
-                                set_global_var(42, 0);
-                                set_global_var(43, 0);
-                                set_global_var(44, 0);
-                                set_global_var(45, 0);
-                                if (global_var(0) >= 1000) then begin
-                                    set_global_var(37, 1);
+                                set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+                                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+                                set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+                                set_global_var(GVAR_KARMA_DEFENDER, 0);
+                                set_global_var(GVAR_KARMA_WANDERER, 0);
+                                set_global_var(GVAR_KARMA_BETRAYER, 0);
+                                set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+                                set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+                                set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+                                if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+                                    set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
                                 end
                                 else begin
-                                    if (global_var(0) >= 750) then begin
-                                        set_global_var(38, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                                        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
                                     end
                                     else begin
-                                        if (global_var(0) >= 500) then begin
-                                            set_global_var(39, 1);
+                                        if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                                            set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                                         end
                                         else begin
-                                            if (global_var(0) >= 250) then begin
-                                                set_global_var(40, 1);
+                                            if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                                                set_global_var(GVAR_KARMA_DEFENDER, 1);
                                             end
                                             else begin
-                                                if (global_var(0) > -250) then begin
-                                                    set_global_var(41, 1);
+                                                if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                                                    set_global_var(GVAR_KARMA_WANDERER, 1);
                                                 end
                                                 else begin
-                                                    if (global_var(0) > -500) then begin
-                                                        set_global_var(42, 1);
+                                                    if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                                        set_global_var(GVAR_KARMA_BETRAYER, 1);
                                                     end
                                                     else begin
-                                                        if (global_var(0) > -750) then begin
-                                                            set_global_var(43, 1);
+                                                        if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                                            set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                                         end
                                                         else begin
-                                                            if (global_var(0) > -1000) then begin
-                                                                set_global_var(44, 1);
+                                                            if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                                                set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                                             end
                                                             else begin
-                                                                set_global_var(45, 1);
+                                                                set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                                             end
                                                         end
                                                     end
@@ -665,15 +665,15 @@ begin
                                     end
                                 end
                             end
-                            set_global_var(552, game_time + (3 * (24 * (60 * (60 * 10)))));
+                            set_global_var(GVAR_NEW_RENO_POISON_STILL_TIME, game_time + (3 * (24 * (60 * (60 * 10)))));
                         end
                     end
-                    set_global_var(344, 2);
-                    display_msg(message_str(655, 130));
-                    display_msg(message_str(655, 300));
+                    set_global_var(GVAR_NEW_RENO_WRIGHT_STILL, 2);
+                    display_mstr(130);
+                    display_mstr(300);
                 end
                 else begin
-                    display_msg(message_str(655, 150));
+                    display_mstr(150);
                 end
             end
             else begin
@@ -681,7 +681,7 @@ begin
                     if (new_reno_wright_teen_obj != 0) then begin
                         last_timed_obj := new_reno_wright_teen_obj;
                         last_timed_param := last_timed_param;
-                        metarule3(100, new_reno_wright_teen_obj, 19, 0);
+                        rm_fixed_timer_event(new_reno_wright_teen_obj, 19);
                         add_timer_event(new_reno_wright_teen_obj, 0, 19);
                     end
                 end
@@ -699,7 +699,7 @@ begin
             script_overrides;
             last_timed_obj := new_reno_wright_teen_obj;
             last_timed_param := last_timed_param;
-            metarule3(100, new_reno_wright_teen_obj, 19, 0);
+            rm_fixed_timer_event(new_reno_wright_teen_obj, 19);
             add_timer_event(new_reno_wright_teen_obj, 0, 19);
         end
         else begin
@@ -708,93 +708,93 @@ begin
     end
     if (tampered == 0) then begin
         if (obj_pid(obj_being_used_with) == 92) then begin
-            if (global_var(344) == 0) then begin
+            if (global_var(GVAR_NEW_RENO_WRIGHT_STILL) == 0) then begin
                 script_overrides;
                 if (1 != 1) then begin
                     if (1 == 2) then begin
-                        if ((global_var(344) == 2) == 0) then begin
-                            set_global_var(55, global_var(55) + -5);
-                            display_msg(message_str(655, 200));
-                            if (global_var(346) == 1) then begin
+                        if ((global_var(GVAR_NEW_RENO_WRIGHT_STILL) == 2) == 0) then begin
+                            set_global_var(GVAR_TOWN_REP_NEW_RENO, global_var(GVAR_TOWN_REP_NEW_RENO) + -5);
+                            display_mstr(200);
+                            if (global_var(GVAR_NEW_RENO_WRIGHT_STILL_MISSION) == 1) then begin
                                 give_exp_points(500);
-                                display_msg(message_str(14, 100) + 500 + message_str(14, 101));
+                                display_msg(g_mstr(100) + 500 + g_mstr(101));
                             end
-                            if (((4 == 1) and (global_var(134) <= -1) or ((4 == 2) and (global_var(135) <= -1)) or ((4 == 3) and (global_var(136) <= -1)) or ((4 == 4) and (global_var(216) <= -1))) == 0) then begin
+                            if (((4 == 1) and (global_var(GVAR_SALVATORE_FAMILY_COUNTER) <= -1) or ((4 == 2) and (global_var(GVAR_BISHOP_FAMILY_COUNTER) <= -1)) or ((4 == 3) and (global_var(GVAR_MORDINO_FAMILY_COUNTER) <= -1)) or ((4 == 4) and (global_var(GVAR_WRIGHT_FAMILY_COUNTER) <= -1))) == 0) then begin
                                 if (4 == 1) then begin
-                                    set_global_var(134, global_var(134) - 20);
+                                    set_global_var(GVAR_SALVATORE_FAMILY_COUNTER, global_var(GVAR_SALVATORE_FAMILY_COUNTER) - 20);
                                 end
                                 else begin
                                     if (4 == 2) then begin
-                                        set_global_var(135, global_var(135) - 20);
+                                        set_global_var(GVAR_BISHOP_FAMILY_COUNTER, global_var(GVAR_BISHOP_FAMILY_COUNTER) - 20);
                                     end
                                     else begin
                                         if (4 == 3) then begin
-                                            set_global_var(136, global_var(136) - 20);
+                                            set_global_var(GVAR_MORDINO_FAMILY_COUNTER, global_var(GVAR_MORDINO_FAMILY_COUNTER) - 20);
                                         end
                                         else begin
                                             if (4 == 4) then begin
-                                                set_global_var(216, global_var(216) - 20);
+                                                set_global_var(GVAR_WRIGHT_FAMILY_COUNTER, global_var(GVAR_WRIGHT_FAMILY_COUNTER) - 20);
                                             end
                                         end
                                     end
                                 end
                             end
                             debug_msg("\nfamily standings: ");
-                            debug_msg("\n  Salvatore == " + global_var(134));
-                            debug_msg("\n  Bishop == " + global_var(135));
-                            debug_msg("\n  Mordino == " + global_var(136));
-                            debug_msg("\n  Wright == " + global_var(216));
+                            debug_msg("\n  Salvatore == " + global_var(GVAR_SALVATORE_FAMILY_COUNTER));
+                            debug_msg("\n  Bishop == " + global_var(GVAR_BISHOP_FAMILY_COUNTER));
+                            debug_msg("\n  Mordino == " + global_var(GVAR_MORDINO_FAMILY_COUNTER));
+                            debug_msg("\n  Wright == " + global_var(GVAR_WRIGHT_FAMILY_COUNTER));
                         end
                     end
-                    set_global_var(552, 0);
+                    set_global_var(GVAR_NEW_RENO_POISON_STILL_TIME, 0);
                 end
                 else begin
-                    if (global_var(552) == 0) then begin
-                        if (global_var(344) == 0) then begin
-                            set_global_var(0, global_var(0) + -30);
+                    if (global_var(GVAR_NEW_RENO_POISON_STILL_TIME) == 0) then begin
+                        if (global_var(GVAR_NEW_RENO_WRIGHT_STILL) == 0) then begin
+                            set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + -30);
                             debug_msg("Player gains " + -30 + " Karma Points.");
-                            set_global_var(37, 0);
-                            set_global_var(38, 0);
-                            set_global_var(39, 0);
-                            set_global_var(40, 0);
-                            set_global_var(41, 0);
-                            set_global_var(42, 0);
-                            set_global_var(43, 0);
-                            set_global_var(44, 0);
-                            set_global_var(45, 0);
-                            if (global_var(0) >= 1000) then begin
-                                set_global_var(37, 1);
+                            set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+                            set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+                            set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+                            set_global_var(GVAR_KARMA_DEFENDER, 0);
+                            set_global_var(GVAR_KARMA_WANDERER, 0);
+                            set_global_var(GVAR_KARMA_BETRAYER, 0);
+                            set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+                            set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+                            set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+                            if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+                                set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
                             end
                             else begin
-                                if (global_var(0) >= 750) then begin
-                                    set_global_var(38, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                                    set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
                                 end
                                 else begin
-                                    if (global_var(0) >= 500) then begin
-                                        set_global_var(39, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                                        set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                                     end
                                     else begin
-                                        if (global_var(0) >= 250) then begin
-                                            set_global_var(40, 1);
+                                        if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                                            set_global_var(GVAR_KARMA_DEFENDER, 1);
                                         end
                                         else begin
-                                            if (global_var(0) > -250) then begin
-                                                set_global_var(41, 1);
+                                            if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                                                set_global_var(GVAR_KARMA_WANDERER, 1);
                                             end
                                             else begin
-                                                if (global_var(0) > -500) then begin
-                                                    set_global_var(42, 1);
+                                                if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                                    set_global_var(GVAR_KARMA_BETRAYER, 1);
                                                 end
                                                 else begin
-                                                    if (global_var(0) > -750) then begin
-                                                        set_global_var(43, 1);
+                                                    if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                                        set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                                     end
                                                     else begin
-                                                        if (global_var(0) > -1000) then begin
-                                                            set_global_var(44, 1);
+                                                        if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                                            set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                                         end
                                                         else begin
-                                                            set_global_var(45, 1);
+                                                            set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                                         end
                                                     end
                                                 end
@@ -804,11 +804,11 @@ begin
                                 end
                             end
                         end
-                        set_global_var(552, game_time + (3 * (24 * (60 * (60 * 10)))));
+                        set_global_var(GVAR_NEW_RENO_POISON_STILL_TIME, game_time + (3 * (24 * (60 * (60 * 10)))));
                     end
                 end
-                set_global_var(344, 1);
-                display_msg(message_str(655, 140));
+                set_global_var(GVAR_NEW_RENO_WRIGHT_STILL, 1);
+                display_mstr(140);
                 removed_qty := obj_is_carrying_obj_pid(source_obj, 92);
                 if (1 < removed_qty) then begin
                     removed_qty := 1;
@@ -817,17 +817,17 @@ begin
                     restock_obj := obj_carrying_pid_obj(source_obj, 92);
                     if (obj_type(self_obj) == 1) then begin
                         restock_obj := obj_carrying_pid_obj(source_obj, 92);
-                        if (critter_inven_obj(self_obj, 0) == restock_obj) then begin
-                            if (obj_item_subtype(critter_inven_obj(source_obj, 0)) == 0) then begin
-                                restock_obj := critter_inven_obj(source_obj, 0);
+                        if (critter_inven_obj(self_obj, INVEN_TYPE_WORN) == restock_obj) then begin
+                            if (obj_item_subtype(critter_inven_obj(source_obj, INVEN_TYPE_WORN)) == 0) then begin
+                                restock_obj := critter_inven_obj(source_obj, INVEN_TYPE_WORN);
                                 debug_msg("armour pid == " + obj_pid(restock_obj));
                                 rm_obj_from_inven(source_obj, restock_obj);
                                 add_obj_to_inven(source_obj, restock_obj);
                             end
                         end
                         else begin
-                            if ((critter_inven_obj(source_obj, 2) == restock_obj) or (critter_inven_obj(source_obj, 1) == restock_obj)) then begin
-                                metarule(43, source_obj);
+                            if ((critter_inven_obj(source_obj, INVEN_TYPE_LEFT_HAND) == restock_obj) or (critter_inven_obj(source_obj, INVEN_TYPE_RIGHT_HAND) == restock_obj)) then begin
+                                inven_unwield(source_obj);
                             end
                         end
                     end
@@ -840,93 +840,93 @@ begin
         end
         else begin
             if (obj_pid(obj_being_used_with) == 545) then begin
-                if (global_var(344) == 0) then begin
+                if (global_var(GVAR_NEW_RENO_WRIGHT_STILL) == 0) then begin
                     script_overrides;
                     if (1 != 1) then begin
                         if (1 == 2) then begin
-                            if ((global_var(344) == 2) == 0) then begin
-                                set_global_var(55, global_var(55) + -5);
-                                display_msg(message_str(655, 200));
-                                if (global_var(346) == 1) then begin
+                            if ((global_var(GVAR_NEW_RENO_WRIGHT_STILL) == 2) == 0) then begin
+                                set_global_var(GVAR_TOWN_REP_NEW_RENO, global_var(GVAR_TOWN_REP_NEW_RENO) + -5);
+                                display_mstr(200);
+                                if (global_var(GVAR_NEW_RENO_WRIGHT_STILL_MISSION) == 1) then begin
                                     give_exp_points(500);
-                                    display_msg(message_str(14, 100) + 500 + message_str(14, 101));
+                                    display_msg(g_mstr(100) + 500 + g_mstr(101));
                                 end
-                                if (((4 == 1) and (global_var(134) <= -1) or ((4 == 2) and (global_var(135) <= -1)) or ((4 == 3) and (global_var(136) <= -1)) or ((4 == 4) and (global_var(216) <= -1))) == 0) then begin
+                                if (((4 == 1) and (global_var(GVAR_SALVATORE_FAMILY_COUNTER) <= -1) or ((4 == 2) and (global_var(GVAR_BISHOP_FAMILY_COUNTER) <= -1)) or ((4 == 3) and (global_var(GVAR_MORDINO_FAMILY_COUNTER) <= -1)) or ((4 == 4) and (global_var(GVAR_WRIGHT_FAMILY_COUNTER) <= -1))) == 0) then begin
                                     if (4 == 1) then begin
-                                        set_global_var(134, global_var(134) - 20);
+                                        set_global_var(GVAR_SALVATORE_FAMILY_COUNTER, global_var(GVAR_SALVATORE_FAMILY_COUNTER) - 20);
                                     end
                                     else begin
                                         if (4 == 2) then begin
-                                            set_global_var(135, global_var(135) - 20);
+                                            set_global_var(GVAR_BISHOP_FAMILY_COUNTER, global_var(GVAR_BISHOP_FAMILY_COUNTER) - 20);
                                         end
                                         else begin
                                             if (4 == 3) then begin
-                                                set_global_var(136, global_var(136) - 20);
+                                                set_global_var(GVAR_MORDINO_FAMILY_COUNTER, global_var(GVAR_MORDINO_FAMILY_COUNTER) - 20);
                                             end
                                             else begin
                                                 if (4 == 4) then begin
-                                                    set_global_var(216, global_var(216) - 20);
+                                                    set_global_var(GVAR_WRIGHT_FAMILY_COUNTER, global_var(GVAR_WRIGHT_FAMILY_COUNTER) - 20);
                                                 end
                                             end
                                         end
                                     end
                                 end
                                 debug_msg("\nfamily standings: ");
-                                debug_msg("\n  Salvatore == " + global_var(134));
-                                debug_msg("\n  Bishop == " + global_var(135));
-                                debug_msg("\n  Mordino == " + global_var(136));
-                                debug_msg("\n  Wright == " + global_var(216));
+                                debug_msg("\n  Salvatore == " + global_var(GVAR_SALVATORE_FAMILY_COUNTER));
+                                debug_msg("\n  Bishop == " + global_var(GVAR_BISHOP_FAMILY_COUNTER));
+                                debug_msg("\n  Mordino == " + global_var(GVAR_MORDINO_FAMILY_COUNTER));
+                                debug_msg("\n  Wright == " + global_var(GVAR_WRIGHT_FAMILY_COUNTER));
                             end
                         end
-                        set_global_var(552, 0);
+                        set_global_var(GVAR_NEW_RENO_POISON_STILL_TIME, 0);
                     end
                     else begin
-                        if (global_var(552) == 0) then begin
-                            if (global_var(344) == 0) then begin
-                                set_global_var(0, global_var(0) + -30);
+                        if (global_var(GVAR_NEW_RENO_POISON_STILL_TIME) == 0) then begin
+                            if (global_var(GVAR_NEW_RENO_WRIGHT_STILL) == 0) then begin
+                                set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + -30);
                                 debug_msg("Player gains " + -30 + " Karma Points.");
-                                set_global_var(37, 0);
-                                set_global_var(38, 0);
-                                set_global_var(39, 0);
-                                set_global_var(40, 0);
-                                set_global_var(41, 0);
-                                set_global_var(42, 0);
-                                set_global_var(43, 0);
-                                set_global_var(44, 0);
-                                set_global_var(45, 0);
-                                if (global_var(0) >= 1000) then begin
-                                    set_global_var(37, 1);
+                                set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+                                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+                                set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+                                set_global_var(GVAR_KARMA_DEFENDER, 0);
+                                set_global_var(GVAR_KARMA_WANDERER, 0);
+                                set_global_var(GVAR_KARMA_BETRAYER, 0);
+                                set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+                                set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+                                set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+                                if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+                                    set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
                                 end
                                 else begin
-                                    if (global_var(0) >= 750) then begin
-                                        set_global_var(38, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                                        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
                                     end
                                     else begin
-                                        if (global_var(0) >= 500) then begin
-                                            set_global_var(39, 1);
+                                        if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                                            set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                                         end
                                         else begin
-                                            if (global_var(0) >= 250) then begin
-                                                set_global_var(40, 1);
+                                            if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                                                set_global_var(GVAR_KARMA_DEFENDER, 1);
                                             end
                                             else begin
-                                                if (global_var(0) > -250) then begin
-                                                    set_global_var(41, 1);
+                                                if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                                                    set_global_var(GVAR_KARMA_WANDERER, 1);
                                                 end
                                                 else begin
-                                                    if (global_var(0) > -500) then begin
-                                                        set_global_var(42, 1);
+                                                    if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                                        set_global_var(GVAR_KARMA_BETRAYER, 1);
                                                     end
                                                     else begin
-                                                        if (global_var(0) > -750) then begin
-                                                            set_global_var(43, 1);
+                                                        if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                                            set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                                         end
                                                         else begin
-                                                            if (global_var(0) > -1000) then begin
-                                                                set_global_var(44, 1);
+                                                            if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                                                set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                                             end
                                                             else begin
-                                                                set_global_var(45, 1);
+                                                                set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                                             end
                                                         end
                                                     end
@@ -936,11 +936,11 @@ begin
                                     end
                                 end
                             end
-                            set_global_var(552, game_time + (3 * (24 * (60 * (60 * 10)))));
+                            set_global_var(GVAR_NEW_RENO_POISON_STILL_TIME, game_time + (3 * (24 * (60 * (60 * 10)))));
                         end
                     end
-                    set_global_var(344, 1);
-                    display_msg(message_str(655, 140));
+                    set_global_var(GVAR_NEW_RENO_WRIGHT_STILL, 1);
+                    display_mstr(140);
                     removed_qty := obj_is_carrying_obj_pid(source_obj, 545);
                     if (1 < removed_qty) then begin
                         removed_qty := 1;
@@ -949,17 +949,17 @@ begin
                         restock_obj := obj_carrying_pid_obj(source_obj, 545);
                         if (obj_type(self_obj) == 1) then begin
                             restock_obj := obj_carrying_pid_obj(source_obj, 545);
-                            if (critter_inven_obj(self_obj, 0) == restock_obj) then begin
-                                if (obj_item_subtype(critter_inven_obj(source_obj, 0)) == 0) then begin
-                                    restock_obj := critter_inven_obj(source_obj, 0);
+                            if (critter_inven_obj(self_obj, INVEN_TYPE_WORN) == restock_obj) then begin
+                                if (obj_item_subtype(critter_inven_obj(source_obj, INVEN_TYPE_WORN)) == 0) then begin
+                                    restock_obj := critter_inven_obj(source_obj, INVEN_TYPE_WORN);
                                     debug_msg("armour pid == " + obj_pid(restock_obj));
                                     rm_obj_from_inven(source_obj, restock_obj);
                                     add_obj_to_inven(source_obj, restock_obj);
                                 end
                             end
                             else begin
-                                if ((critter_inven_obj(source_obj, 2) == restock_obj) or (critter_inven_obj(source_obj, 1) == restock_obj)) then begin
-                                    metarule(43, source_obj);
+                                if ((critter_inven_obj(source_obj, INVEN_TYPE_LEFT_HAND) == restock_obj) or (critter_inven_obj(source_obj, INVEN_TYPE_RIGHT_HAND) == restock_obj)) then begin
+                                    inven_unwield(source_obj);
                                 end
                             end
                         end
@@ -972,93 +972,93 @@ begin
             end
             else begin
                 if (obj_pid(obj_being_used_with) == 542) then begin
-                    if (global_var(344) == 0) then begin
+                    if (global_var(GVAR_NEW_RENO_WRIGHT_STILL) == 0) then begin
                         script_overrides;
                         if (1 != 1) then begin
                             if (1 == 2) then begin
-                                if ((global_var(344) == 2) == 0) then begin
-                                    set_global_var(55, global_var(55) + -5);
-                                    display_msg(message_str(655, 200));
-                                    if (global_var(346) == 1) then begin
+                                if ((global_var(GVAR_NEW_RENO_WRIGHT_STILL) == 2) == 0) then begin
+                                    set_global_var(GVAR_TOWN_REP_NEW_RENO, global_var(GVAR_TOWN_REP_NEW_RENO) + -5);
+                                    display_mstr(200);
+                                    if (global_var(GVAR_NEW_RENO_WRIGHT_STILL_MISSION) == 1) then begin
                                         give_exp_points(500);
-                                        display_msg(message_str(14, 100) + 500 + message_str(14, 101));
+                                        display_msg(g_mstr(100) + 500 + g_mstr(101));
                                     end
-                                    if (((4 == 1) and (global_var(134) <= -1) or ((4 == 2) and (global_var(135) <= -1)) or ((4 == 3) and (global_var(136) <= -1)) or ((4 == 4) and (global_var(216) <= -1))) == 0) then begin
+                                    if (((4 == 1) and (global_var(GVAR_SALVATORE_FAMILY_COUNTER) <= -1) or ((4 == 2) and (global_var(GVAR_BISHOP_FAMILY_COUNTER) <= -1)) or ((4 == 3) and (global_var(GVAR_MORDINO_FAMILY_COUNTER) <= -1)) or ((4 == 4) and (global_var(GVAR_WRIGHT_FAMILY_COUNTER) <= -1))) == 0) then begin
                                         if (4 == 1) then begin
-                                            set_global_var(134, global_var(134) - 20);
+                                            set_global_var(GVAR_SALVATORE_FAMILY_COUNTER, global_var(GVAR_SALVATORE_FAMILY_COUNTER) - 20);
                                         end
                                         else begin
                                             if (4 == 2) then begin
-                                                set_global_var(135, global_var(135) - 20);
+                                                set_global_var(GVAR_BISHOP_FAMILY_COUNTER, global_var(GVAR_BISHOP_FAMILY_COUNTER) - 20);
                                             end
                                             else begin
                                                 if (4 == 3) then begin
-                                                    set_global_var(136, global_var(136) - 20);
+                                                    set_global_var(GVAR_MORDINO_FAMILY_COUNTER, global_var(GVAR_MORDINO_FAMILY_COUNTER) - 20);
                                                 end
                                                 else begin
                                                     if (4 == 4) then begin
-                                                        set_global_var(216, global_var(216) - 20);
+                                                        set_global_var(GVAR_WRIGHT_FAMILY_COUNTER, global_var(GVAR_WRIGHT_FAMILY_COUNTER) - 20);
                                                     end
                                                 end
                                             end
                                         end
                                     end
                                     debug_msg("\nfamily standings: ");
-                                    debug_msg("\n  Salvatore == " + global_var(134));
-                                    debug_msg("\n  Bishop == " + global_var(135));
-                                    debug_msg("\n  Mordino == " + global_var(136));
-                                    debug_msg("\n  Wright == " + global_var(216));
+                                    debug_msg("\n  Salvatore == " + global_var(GVAR_SALVATORE_FAMILY_COUNTER));
+                                    debug_msg("\n  Bishop == " + global_var(GVAR_BISHOP_FAMILY_COUNTER));
+                                    debug_msg("\n  Mordino == " + global_var(GVAR_MORDINO_FAMILY_COUNTER));
+                                    debug_msg("\n  Wright == " + global_var(GVAR_WRIGHT_FAMILY_COUNTER));
                                 end
                             end
-                            set_global_var(552, 0);
+                            set_global_var(GVAR_NEW_RENO_POISON_STILL_TIME, 0);
                         end
                         else begin
-                            if (global_var(552) == 0) then begin
-                                if (global_var(344) == 0) then begin
-                                    set_global_var(0, global_var(0) + -30);
+                            if (global_var(GVAR_NEW_RENO_POISON_STILL_TIME) == 0) then begin
+                                if (global_var(GVAR_NEW_RENO_WRIGHT_STILL) == 0) then begin
+                                    set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + -30);
                                     debug_msg("Player gains " + -30 + " Karma Points.");
-                                    set_global_var(37, 0);
-                                    set_global_var(38, 0);
-                                    set_global_var(39, 0);
-                                    set_global_var(40, 0);
-                                    set_global_var(41, 0);
-                                    set_global_var(42, 0);
-                                    set_global_var(43, 0);
-                                    set_global_var(44, 0);
-                                    set_global_var(45, 0);
-                                    if (global_var(0) >= 1000) then begin
-                                        set_global_var(37, 1);
+                                    set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+                                    set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+                                    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+                                    set_global_var(GVAR_KARMA_DEFENDER, 0);
+                                    set_global_var(GVAR_KARMA_WANDERER, 0);
+                                    set_global_var(GVAR_KARMA_BETRAYER, 0);
+                                    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+                                    set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+                                    set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+                                        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
                                     end
                                     else begin
-                                        if (global_var(0) >= 750) then begin
-                                            set_global_var(38, 1);
+                                        if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                                            set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
                                         end
                                         else begin
-                                            if (global_var(0) >= 500) then begin
-                                                set_global_var(39, 1);
+                                            if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                                                set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                                             end
                                             else begin
-                                                if (global_var(0) >= 250) then begin
-                                                    set_global_var(40, 1);
+                                                if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                                                    set_global_var(GVAR_KARMA_DEFENDER, 1);
                                                 end
                                                 else begin
-                                                    if (global_var(0) > -250) then begin
-                                                        set_global_var(41, 1);
+                                                    if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                                                        set_global_var(GVAR_KARMA_WANDERER, 1);
                                                     end
                                                     else begin
-                                                        if (global_var(0) > -500) then begin
-                                                            set_global_var(42, 1);
+                                                        if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                                            set_global_var(GVAR_KARMA_BETRAYER, 1);
                                                         end
                                                         else begin
-                                                            if (global_var(0) > -750) then begin
-                                                                set_global_var(43, 1);
+                                                            if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                                                set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                                             end
                                                             else begin
-                                                                if (global_var(0) > -1000) then begin
-                                                                    set_global_var(44, 1);
+                                                                if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                                                    set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                                                 end
                                                                 else begin
-                                                                    set_global_var(45, 1);
+                                                                    set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                                                 end
                                                             end
                                                         end
@@ -1068,11 +1068,11 @@ begin
                                         end
                                     end
                                 end
-                                set_global_var(552, game_time + (3 * (24 * (60 * (60 * 10)))));
+                                set_global_var(GVAR_NEW_RENO_POISON_STILL_TIME, game_time + (3 * (24 * (60 * (60 * 10)))));
                             end
                         end
-                        set_global_var(344, 1);
-                        display_msg(message_str(655, 140));
+                        set_global_var(GVAR_NEW_RENO_WRIGHT_STILL, 1);
+                        display_mstr(140);
                         removed_qty := obj_is_carrying_obj_pid(source_obj, 542);
                         if (1 < removed_qty) then begin
                             removed_qty := 1;
@@ -1081,17 +1081,17 @@ begin
                             restock_obj := obj_carrying_pid_obj(source_obj, 542);
                             if (obj_type(self_obj) == 1) then begin
                                 restock_obj := obj_carrying_pid_obj(source_obj, 542);
-                                if (critter_inven_obj(self_obj, 0) == restock_obj) then begin
-                                    if (obj_item_subtype(critter_inven_obj(source_obj, 0)) == 0) then begin
-                                        restock_obj := critter_inven_obj(source_obj, 0);
+                                if (critter_inven_obj(self_obj, INVEN_TYPE_WORN) == restock_obj) then begin
+                                    if (obj_item_subtype(critter_inven_obj(source_obj, INVEN_TYPE_WORN)) == 0) then begin
+                                        restock_obj := critter_inven_obj(source_obj, INVEN_TYPE_WORN);
                                         debug_msg("armour pid == " + obj_pid(restock_obj));
                                         rm_obj_from_inven(source_obj, restock_obj);
                                         add_obj_to_inven(source_obj, restock_obj);
                                     end
                                 end
                                 else begin
-                                    if ((critter_inven_obj(source_obj, 2) == restock_obj) or (critter_inven_obj(source_obj, 1) == restock_obj)) then begin
-                                        metarule(43, source_obj);
+                                    if ((critter_inven_obj(source_obj, INVEN_TYPE_LEFT_HAND) == restock_obj) or (critter_inven_obj(source_obj, INVEN_TYPE_RIGHT_HAND) == restock_obj)) then begin
+                                        inven_unwield(source_obj);
                                     end
                                 end
                             end
@@ -1103,94 +1103,94 @@ begin
                     end
                 end
                 else begin
-                    if ((global_var(344) == 2) == 0) then begin
+                    if ((global_var(GVAR_NEW_RENO_WRIGHT_STILL) == 2) == 0) then begin
                         if ((obj_pid(obj_being_used_with) == 384) or (obj_pid(obj_being_used_with) == 20) or (obj_pid(obj_being_used_with) == 75)) then begin
                             script_overrides;
                             if (2 != 1) then begin
                                 if (2 == 2) then begin
-                                    if ((global_var(344) == 2) == 0) then begin
-                                        set_global_var(55, global_var(55) + -5);
-                                        display_msg(message_str(655, 200));
-                                        if (global_var(346) == 1) then begin
+                                    if ((global_var(GVAR_NEW_RENO_WRIGHT_STILL) == 2) == 0) then begin
+                                        set_global_var(GVAR_TOWN_REP_NEW_RENO, global_var(GVAR_TOWN_REP_NEW_RENO) + -5);
+                                        display_mstr(200);
+                                        if (global_var(GVAR_NEW_RENO_WRIGHT_STILL_MISSION) == 1) then begin
                                             give_exp_points(500);
-                                            display_msg(message_str(14, 100) + 500 + message_str(14, 101));
+                                            display_msg(g_mstr(100) + 500 + g_mstr(101));
                                         end
-                                        if (((4 == 1) and (global_var(134) <= -1) or ((4 == 2) and (global_var(135) <= -1)) or ((4 == 3) and (global_var(136) <= -1)) or ((4 == 4) and (global_var(216) <= -1))) == 0) then begin
+                                        if (((4 == 1) and (global_var(GVAR_SALVATORE_FAMILY_COUNTER) <= -1) or ((4 == 2) and (global_var(GVAR_BISHOP_FAMILY_COUNTER) <= -1)) or ((4 == 3) and (global_var(GVAR_MORDINO_FAMILY_COUNTER) <= -1)) or ((4 == 4) and (global_var(GVAR_WRIGHT_FAMILY_COUNTER) <= -1))) == 0) then begin
                                             if (4 == 1) then begin
-                                                set_global_var(134, global_var(134) - 20);
+                                                set_global_var(GVAR_SALVATORE_FAMILY_COUNTER, global_var(GVAR_SALVATORE_FAMILY_COUNTER) - 20);
                                             end
                                             else begin
                                                 if (4 == 2) then begin
-                                                    set_global_var(135, global_var(135) - 20);
+                                                    set_global_var(GVAR_BISHOP_FAMILY_COUNTER, global_var(GVAR_BISHOP_FAMILY_COUNTER) - 20);
                                                 end
                                                 else begin
                                                     if (4 == 3) then begin
-                                                        set_global_var(136, global_var(136) - 20);
+                                                        set_global_var(GVAR_MORDINO_FAMILY_COUNTER, global_var(GVAR_MORDINO_FAMILY_COUNTER) - 20);
                                                     end
                                                     else begin
                                                         if (4 == 4) then begin
-                                                            set_global_var(216, global_var(216) - 20);
+                                                            set_global_var(GVAR_WRIGHT_FAMILY_COUNTER, global_var(GVAR_WRIGHT_FAMILY_COUNTER) - 20);
                                                         end
                                                     end
                                                 end
                                             end
                                         end
                                         debug_msg("\nfamily standings: ");
-                                        debug_msg("\n  Salvatore == " + global_var(134));
-                                        debug_msg("\n  Bishop == " + global_var(135));
-                                        debug_msg("\n  Mordino == " + global_var(136));
-                                        debug_msg("\n  Wright == " + global_var(216));
+                                        debug_msg("\n  Salvatore == " + global_var(GVAR_SALVATORE_FAMILY_COUNTER));
+                                        debug_msg("\n  Bishop == " + global_var(GVAR_BISHOP_FAMILY_COUNTER));
+                                        debug_msg("\n  Mordino == " + global_var(GVAR_MORDINO_FAMILY_COUNTER));
+                                        debug_msg("\n  Wright == " + global_var(GVAR_WRIGHT_FAMILY_COUNTER));
                                     end
                                 end
-                                set_global_var(552, 0);
+                                set_global_var(GVAR_NEW_RENO_POISON_STILL_TIME, 0);
                             end
                             else begin
-                                if (global_var(552) == 0) then begin
-                                    if (global_var(344) == 0) then begin
-                                        set_global_var(0, global_var(0) + -30);
+                                if (global_var(GVAR_NEW_RENO_POISON_STILL_TIME) == 0) then begin
+                                    if (global_var(GVAR_NEW_RENO_WRIGHT_STILL) == 0) then begin
+                                        set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + -30);
                                         debug_msg("Player gains " + -30 + " Karma Points.");
-                                        set_global_var(37, 0);
-                                        set_global_var(38, 0);
-                                        set_global_var(39, 0);
-                                        set_global_var(40, 0);
-                                        set_global_var(41, 0);
-                                        set_global_var(42, 0);
-                                        set_global_var(43, 0);
-                                        set_global_var(44, 0);
-                                        set_global_var(45, 0);
-                                        if (global_var(0) >= 1000) then begin
-                                            set_global_var(37, 1);
+                                        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+                                        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+                                        set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+                                        set_global_var(GVAR_KARMA_DEFENDER, 0);
+                                        set_global_var(GVAR_KARMA_WANDERER, 0);
+                                        set_global_var(GVAR_KARMA_BETRAYER, 0);
+                                        set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+                                        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+                                        set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+                                        if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+                                            set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
                                         end
                                         else begin
-                                            if (global_var(0) >= 750) then begin
-                                                set_global_var(38, 1);
+                                            if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                                                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
                                             end
                                             else begin
-                                                if (global_var(0) >= 500) then begin
-                                                    set_global_var(39, 1);
+                                                if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                                                    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                                                 end
                                                 else begin
-                                                    if (global_var(0) >= 250) then begin
-                                                        set_global_var(40, 1);
+                                                    if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                                                        set_global_var(GVAR_KARMA_DEFENDER, 1);
                                                     end
                                                     else begin
-                                                        if (global_var(0) > -250) then begin
-                                                            set_global_var(41, 1);
+                                                        if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                                                            set_global_var(GVAR_KARMA_WANDERER, 1);
                                                         end
                                                         else begin
-                                                            if (global_var(0) > -500) then begin
-                                                                set_global_var(42, 1);
+                                                            if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                                                set_global_var(GVAR_KARMA_BETRAYER, 1);
                                                             end
                                                             else begin
-                                                                if (global_var(0) > -750) then begin
-                                                                    set_global_var(43, 1);
+                                                                if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                                                    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                                                 end
                                                                 else begin
-                                                                    if (global_var(0) > -1000) then begin
-                                                                        set_global_var(44, 1);
+                                                                    if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                                                        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                                                     end
                                                                     else begin
-                                                                        set_global_var(45, 1);
+                                                                        set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                                                     end
                                                                 end
                                                             end
@@ -1200,18 +1200,18 @@ begin
                                             end
                                         end
                                     end
-                                    set_global_var(552, game_time + (3 * (24 * (60 * (60 * 10)))));
+                                    set_global_var(GVAR_NEW_RENO_POISON_STILL_TIME, game_time + (3 * (24 * (60 * (60 * 10)))));
                                 end
                             end
-                            set_global_var(344, 2);
-                            display_msg(message_str(655, 130));
-                            display_msg(message_str(655, 300));
+                            set_global_var(GVAR_NEW_RENO_WRIGHT_STILL, 2);
+                            display_mstr(130);
+                            display_mstr(300);
                             the_roll := roll_vs_skill(dude_obj, 13, 0);
                             if ((is_success(the_roll) == 0) and is_critical(the_roll)) then begin
                                 if (new_reno_wright_teen_obj != 0) then begin
                                     last_timed_obj := new_reno_wright_teen_obj;
                                     last_timed_param := last_timed_param;
-                                    metarule3(100, new_reno_wright_teen_obj, 19, 0);
+                                    rm_fixed_timer_event(new_reno_wright_teen_obj, 19);
                                     add_timer_event(new_reno_wright_teen_obj, 0, 19);
                                 end
                             end
@@ -1225,15 +1225,14 @@ end
 
 procedure timed_event_p_proc
 begin
-    if (global_var(344) == 2) then begin
-        float_msg(self_obj, message_str(655, random(200, 204)), 9);
+    if (global_var(GVAR_NEW_RENO_WRIGHT_STILL) == 2) then begin
+        float_msg(self_obj, mstr(random(200, 204)), FLOAT_MSG_WHITE);
     end
     else begin
-        float_msg(self_obj, message_str(655, random(250, 251)), 9);
+        float_msg(self_obj, mstr(random(250, 251)), FLOAT_MSG_WHITE);
     end
     last_timed_obj := self_obj;
     last_timed_param := last_timed_param;
-    metarule3(100, self_obj, 1, 0);
+    rm_fixed_timer_event(self_obj, 1);
     add_timer_event(self_obj, game_ticks(5) + random(0, game_ticks(15 - 5)), 1);
 end
-
