diff --git a/Mods/MegaMod/Original/PALADINA.ssl b/Mods/MegaMod/ReDefine/PALADINA.ssl
index 2f74c39..99b9808 100644
--- a/Mods/MegaMod/Original/PALADINA.ssl
+++ b/Mods/MegaMod/ReDefine/PALADINA.ssl
@@ -16,32 +16,32 @@ procedure start
 begin
     if (only_once) then begin
         only_once := 0;
-        critter_add_trait(self_obj, 1, 6, 44);
-        critter_add_trait(self_obj, 1, 5, 65);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 44);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_PARTY_MARCUS_DEFENSIVE);
         add_timer_event(self_obj, game_ticks(300), 1);
     end
-    if (script_action == 21) then begin
+    if (script_action == look_at_proc) then begin
         call look_at_p_proc();
     end
     else begin
-        if (script_action == 4) then begin
+        if (script_action == pickup_proc) then begin
             call pickup_p_proc();
         end
         else begin
-            if (script_action == 11) then begin
+            if (script_action == talk_proc) then begin
                 call talk_p_proc();
             end
         end
     end
-    if (script_action == 22) then begin
+    if (script_action == timed_event_proc) then begin
         call time_p_proc();
     end
     else begin
-        if (script_action == 12) then begin
+        if (script_action == critter_proc) then begin
             call critter_p_proc();
         end
         else begin
-            if (script_action == 18) then begin
+            if (script_action == destroy_proc) then begin
                 call destroy_p_proc();
             end
         end
@@ -58,37 +58,37 @@ begin
     if (LVar0 > 10) then begin
         LVar0 := 1;
     end
-    LVar1 := message_str(325, 101);
+    LVar1 := message_str(SCRIPT_SCMERK, 101);
     if (LVar0 == 2) then begin
-        LVar1 := message_str(325, 102);
+        LVar1 := message_str(SCRIPT_SCMERK, 102);
     end
     else begin
         if (LVar0 == 3) then begin
-            LVar1 := message_str(325, 103);
+            LVar1 := message_str(SCRIPT_SCMERK, 103);
         end
         else begin
             if (LVar0 == 4) then begin
-                LVar1 := message_str(325, 104) + proto_data(obj_pid(dude_obj), 1) + message_str(325, 105);
+                LVar1 := message_str(SCRIPT_SCMERK, 104) + proto_data(obj_pid(dude_obj), 1) + message_str(SCRIPT_SCMERK, 105);
             end
             else begin
                 if (LVar0 == 5) then begin
-                    LVar1 := message_str(325, 106);
+                    LVar1 := message_str(SCRIPT_SCMERK, 106);
                 end
                 else begin
                     if (LVar0 == 6) then begin
-                        LVar1 := message_str(325, 107);
+                        LVar1 := message_str(SCRIPT_SCMERK, 107);
                     end
                     else begin
                         if (LVar0 == 7) then begin
-                            LVar1 := message_str(325, 108);
+                            LVar1 := message_str(SCRIPT_SCMERK, 108);
                         end
                         else begin
                             if (LVar0 == 8) then begin
-                                LVar1 := message_str(325, 109);
+                                LVar1 := message_str(SCRIPT_SCMERK, 109);
                             end
                             else begin
                                 if (LVar0 == 9) then begin
-                                    LVar1 := message_str(325, 110);
+                                    LVar1 := message_str(SCRIPT_SCMERK, 110);
                                 end
                                 else begin
                                     LVar0 := 1;
@@ -101,7 +101,7 @@ begin
         end
     end
     LVar0 := LVar0 + 1;
-    float_msg(self_obj, LVar1, 0);
+    float_msg(self_obj, LVar1, FLOAT_MSG_NORMAL);
 end
 
 procedure PaladinABackground
@@ -109,26 +109,26 @@ begin
     variable LVar0 := 0;
     add_timer_event(self_obj, game_ticks(300), 1);
     if (random(0, 1)) then begin
-        LVar0 := message_str(325, 112);
+        LVar0 := message_str(SCRIPT_SCMERK, 112);
     end
     else begin
-        LVar0 := message_str(325, 113);
+        LVar0 := message_str(SCRIPT_SCMERK, 113);
     end
-    float_msg(self_obj, LVar0, 0);
+    float_msg(self_obj, LVar0, FLOAT_MSG_NORMAL);
 end
 
 procedure critter_p_proc
 begin
-    if (global_var(250)) then begin
+    if (global_var(GVAR_GECKO_BRAIN_FRIEND)) then begin
         hostile := 1;
     end
     if (tile_distance_objs(self_obj, dude_obj) > 12) then begin
         hostile := 0;
     end
     if (hostile) then begin
-        set_global_var(250, 1);
+        set_global_var(GVAR_GECKO_BRAIN_FRIEND, 1);
         hostile := 0;
-        attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+        attack(dude_obj);
     end
 end
 
@@ -146,20 +146,20 @@ end
 
 procedure destroy_p_proc
 begin
-    set_global_var(250, 1);
+    set_global_var(GVAR_GECKO_BRAIN_FRIEND, 1);
     rm_timer_event(self_obj);
     if (source_obj == dude_obj) then begin
-        if (((global_var(160) + global_var(159)) >= 25) and ((global_var(159) > (2 * global_var(160))) or (global_var(156) == 1))) then begin
-            set_global_var(156, 1);
-            set_global_var(157, 0);
+        if (((global_var(GVAR_GECKO_SKEETER_PART) + global_var(GVAR_GECKO_REQ_FORM)) >= 25) and ((global_var(GVAR_GECKO_REQ_FORM) > (2 * global_var(GVAR_GECKO_SKEETER_PART))) or (global_var(GVAR_SIERRA_LOCKOUT) == 1))) then begin
+            set_global_var(GVAR_SIERRA_LOCKOUT, 1);
+            set_global_var(GVAR_SIERRA_PASSWORD, 0);
         end
-        if (((global_var(160) + global_var(159)) >= 25) and ((global_var(160) > (3 * global_var(159))) or (global_var(157) == 1))) then begin
-            set_global_var(157, 1);
-            set_global_var(156, 0);
+        if (((global_var(GVAR_GECKO_SKEETER_PART) + global_var(GVAR_GECKO_REQ_FORM)) >= 25) and ((global_var(160) > (3 * global_var(GVAR_GECKO_REQ_FORM))) or (global_var(GVAR_SIERRA_PASSWORD) == 1))) then begin
+            set_global_var(GVAR_SIERRA_PASSWORD, 1);
+            set_global_var(GVAR_SIERRA_LOCKOUT, 0);
         end
-        set_global_var(159, global_var(159) + 1);
-        if ((global_var(159) % 2) == 0) then begin
-            set_global_var(155, global_var(155) - 1);
+        set_global_var(GVAR_GECKO_REQ_FORM, global_var(GVAR_GECKO_REQ_FORM) + 1);
+        if ((global_var(GVAR_GECKO_REQ_FORM) % 2) == 0) then begin
+            set_global_var(GVAR_BRAIN_BOT_BRAIN, global_var(GVAR_BRAIN_BOT_BRAIN) - 1);
         end
     end
 end
@@ -167,11 +167,10 @@ end
 procedure look_at_p_proc
 begin
     script_overrides;
-    display_msg(message_str(325, 100));
+    display_msg(message_str(SCRIPT_SCMERK, 100));
 end
 
 procedure time_p_proc
 begin
     call PaladinABackground();
 end
-
