diff --git a/Mods/MegaMod/Original/PEZ.ssl b/Mods/MegaMod/ReDefine/PEZ.ssl
index d1b517d..0c17341 100644
--- a/Mods/MegaMod/Original/PEZ.ssl
+++ b/Mods/MegaMod/ReDefine/PEZ.ssl
@@ -103,7 +103,7 @@ procedure Goodbyes;
 procedure start
 begin
     if (local_var(0) == 0) then begin
-        critter_add_trait(self_obj, 1, 6, 184);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 184);
         set_local_var(0, 1);
         set_local_var(1, tile_num(self_obj));
     end
@@ -121,7 +121,7 @@ begin
         display_msg("You see a Scraptown child.");
     end
     else begin
-        display_msg(message_str(1500, 100));
+        display_mstr(100);
     end
 end
 
@@ -132,62 +132,62 @@ begin
         display_msg("You see a Scraptown child.");
     end
     else begin
-        display_msg(message_str(1500, 100));
+        display_mstr(100);
     end
 end
 
 procedure destroy_p_proc
 begin
     if (source_obj == dude_obj) then begin
-        set_global_var(5, global_var(5) + 1);
-        if (metarule(51, self_obj) == 2) then begin
+        set_global_var(GVAR_GOOD_MONSTER, global_var(GVAR_GOOD_MONSTER) + 1);
+        if (critter_kill_type(self_obj) == 2) then begin
             if (source_obj == dude_obj) then begin
-                set_global_var(2, 0);
-                set_global_var(1, global_var(1) + 1);
-                set_global_var(0, global_var(0) + -15);
+                set_global_var(GVAR_CHAMPION_REPUTATION, 0);
+                set_global_var(GVAR_CHILDKILLER_REPUTATION, global_var(GVAR_CHILDKILLER_REPUTATION) + 1);
+                set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + -15);
                 debug_msg("Player gains " + -15 + " Karma Points.");
-                set_global_var(37, 0);
-                set_global_var(38, 0);
-                set_global_var(39, 0);
-                set_global_var(40, 0);
-                set_global_var(41, 0);
-                set_global_var(42, 0);
-                set_global_var(43, 0);
-                set_global_var(44, 0);
-                set_global_var(45, 0);
-                if (global_var(0) >= 1000) then begin
-                    set_global_var(37, 1);
+                set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+                set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+                set_global_var(GVAR_KARMA_DEFENDER, 0);
+                set_global_var(GVAR_KARMA_WANDERER, 0);
+                set_global_var(GVAR_KARMA_BETRAYER, 0);
+                set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+                set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+                set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+                    set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
                 end
                 else begin
-                    if (global_var(0) >= 750) then begin
-                        set_global_var(38, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
                     end
                     else begin
-                        if (global_var(0) >= 500) then begin
-                            set_global_var(39, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                            set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                         end
                         else begin
-                            if (global_var(0) >= 250) then begin
-                                set_global_var(40, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                                set_global_var(GVAR_KARMA_DEFENDER, 1);
                             end
                             else begin
-                                if (global_var(0) > -250) then begin
-                                    set_global_var(41, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                                    set_global_var(GVAR_KARMA_WANDERER, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -500) then begin
-                                        set_global_var(42, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                        set_global_var(GVAR_KARMA_BETRAYER, 1);
                                     end
                                     else begin
-                                        if (global_var(0) > -750) then begin
-                                            set_global_var(43, 1);
+                                        if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                            set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                         end
                                         else begin
-                                            if (global_var(0) > -1000) then begin
-                                                set_global_var(44, 1);
+                                            if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                                set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                             end
                                             else begin
-                                                set_global_var(45, 1);
+                                                set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                             end
                                         end
                                     end
@@ -196,62 +196,62 @@ begin
                         end
                     end
                 end
-                set_global_var(808, global_var(808) + -5);
+                set_global_var(GVAR_GANG_2, global_var(GVAR_GANG_2) + -5);
                 debug_msg("Added " + -5 + " to Town Rep");
             end
         end
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(4) > (3 * global_var(5))) or (global_var(2) == 1)) and (global_var(1) == 0)) then begin
-            set_global_var(2, 1);
-            set_global_var(3, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(4) > (3 * global_var(GVAR_GOOD_MONSTER))) or (global_var(GVAR_CHAMPION_REPUTATION) == 1)) and (global_var(GVAR_CHILDKILLER_REPUTATION) == 0)) then begin
+            set_global_var(GVAR_CHAMPION_REPUTATION, 1);
+            set_global_var(GVAR_BERSERKER_REPUTATION, 0);
         end
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(5) > (2 * global_var(4))) or (global_var(3) == 1))) then begin
-            set_global_var(3, 1);
-            set_global_var(2, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
+            set_global_var(GVAR_BERSERKER_REPUTATION, 1);
+            set_global_var(GVAR_CHAMPION_REPUTATION, 0);
         end
-        set_global_var(0, global_var(0) + -5);
+        set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + -5);
         debug_msg("Player gains " + -5 + " Karma Points.");
-        set_global_var(37, 0);
-        set_global_var(38, 0);
-        set_global_var(39, 0);
-        set_global_var(40, 0);
-        set_global_var(41, 0);
-        set_global_var(42, 0);
-        set_global_var(43, 0);
-        set_global_var(44, 0);
-        set_global_var(45, 0);
-        if (global_var(0) >= 1000) then begin
-            set_global_var(37, 1);
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+        set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+        set_global_var(GVAR_KARMA_DEFENDER, 0);
+        set_global_var(GVAR_KARMA_WANDERER, 0);
+        set_global_var(GVAR_KARMA_BETRAYER, 0);
+        set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+        set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+            set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
         end
         else begin
-            if (global_var(0) >= 750) then begin
-                set_global_var(38, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
             end
             else begin
-                if (global_var(0) >= 500) then begin
-                    set_global_var(39, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                 end
                 else begin
-                    if (global_var(0) >= 250) then begin
-                        set_global_var(40, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                        set_global_var(GVAR_KARMA_DEFENDER, 1);
                     end
                     else begin
-                        if (global_var(0) > -250) then begin
-                            set_global_var(41, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                            set_global_var(GVAR_KARMA_WANDERER, 1);
                         end
                         else begin
-                            if (global_var(0) > -500) then begin
-                                set_global_var(42, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                set_global_var(GVAR_KARMA_BETRAYER, 1);
                             end
                             else begin
-                                if (global_var(0) > -750) then begin
-                                    set_global_var(43, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -1000) then begin
-                                        set_global_var(44, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                     end
                                     else begin
-                                        set_global_var(45, 1);
+                                        set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                     end
                                 end
                             end
@@ -260,7 +260,7 @@ begin
                 end
             end
         end
-        set_global_var(808, global_var(808) + -3);
+        set_global_var(GVAR_GANG_2, global_var(GVAR_GANG_2) + -3);
         debug_msg("Added " + -3 + " to Town Rep");
     end
 end
@@ -329,29 +329,29 @@ end
 
 procedure pez00
 begin
-    float_msg(self_obj, message_str(1500, random(101, 104)), 8);
+    float_msg(self_obj, mstr(random(101, 104)), FLOAT_MSG_YELLOW);
 end
 
 procedure pez01
 begin
-    float_msg(self_obj, message_str(1500, random(105, 109)), 8);
+    float_msg(self_obj, mstr(random(105, 109)), FLOAT_MSG_YELLOW);
 end
 
 procedure pez02
 begin
-    float_msg(self_obj, message_str(1500, random(110, 113)), 7);
+    float_msg(self_obj, mstr(random(110, 113)), FLOAT_MSG_LIGHT_RED);
 end
 
 procedure pez03
 begin
-    float_msg(self_obj, message_str(1500, random(114, 118)), 7);
+    float_msg(self_obj, mstr(random(114, 118)), FLOAT_MSG_LIGHT_RED);
 end
 
 procedure pez04
 begin
     GVar12 := random(1, 3);
     if (GVar12 == 1) then begin
-        float_msg(self_obj, message_str(1500, random(119, 123)), 0);
+        float_msg(self_obj, mstr(random(119, 123)), FLOAT_MSG_NORMAL);
     end
     else begin
         if (GVar12 == 2) then begin
@@ -359,7 +359,7 @@ begin
         end
         else begin
             if (GVar12 == 3) then begin
-                float_msg(self_obj, message_str(1500, 124), 0);
+                float_msg(self_obj, mstr(124), FLOAT_MSG_NORMAL);
             end
         end
     end
@@ -369,16 +369,16 @@ procedure pez05
 begin
     GVar12 := random(1, 10);
     if (GVar12 < 3) then begin
-        float_msg(self_obj, message_str(1500, 125), 3);
+        float_msg(self_obj, mstr(125), FLOAT_MSG_GREEN);
     end
     else begin
-        float_msg(self_obj, message_str(1500, 126), 0);
+        float_msg(self_obj, mstr(126), FLOAT_MSG_NORMAL);
     end
 end
 
 procedure pezdone
 begin
-    sayOption(message_str(1500, 127), pezend);
+    sayOption(mstr(127), pezend);
     sayOption("", pezend);
 end
 
@@ -406,20 +406,20 @@ procedure get_reaction
 begin
     if (not(GVar16)) then begin
         GVar16 := 1;
-        GVar14 := GVar14 + ((get_critter_stat(dude_obj, 3) - 5) * 5);
+        GVar14 := GVar14 + ((dude_charisma - 5) * 5);
         if (GVar17) then begin
             GVar14 := GVar14 - map_var(8);
         end
         else begin
             GVar14 := GVar14 + map_var(8);
         end
-        if (global_var(1) > 2) then begin
+        if (global_var(GVAR_CHILDKILLER_REPUTATION) > 2) then begin
             GVar14 := GVar14 - 30;
         end
-        if ((global_var(4) > (3 * global_var(5))) or (global_var(2) == 1)) then begin
+        if ((global_var(GVAR_BAD_MONSTER) > (3 * global_var(GVAR_GOOD_MONSTER))) or (global_var(GVAR_CHAMPION_REPUTATION) == 1)) then begin
             GVar14 := GVar14 + 20;
         end
-        if ((global_var(5) > (2 * global_var(4))) or (global_var(3) == 1)) then begin
+        if ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1)) then begin
             GVar14 := GVar14 - 20;
         end
         call ReactToLevel();
@@ -535,4 +535,3 @@ begin
     return 0;
     LVar0 := 0;
 end
-
