diff --git a/Mods/MegaMod/Original/PHRAX.ssl b/Mods/MegaMod/ReDefine/PHRAX.ssl
index aa1cae3..85f1328 100644
--- a/Mods/MegaMod/Original/PHRAX.ssl
+++ b/Mods/MegaMod/ReDefine/PHRAX.ssl
@@ -97,8 +97,8 @@ begin
     set_local_var(2, 2);
     Phrax_Pointer := self_obj;
     if (local_var(0) == 0) then begin
-        critter_add_trait(self_obj, 1, 6, 136);
-        critter_add_trait(self_obj, 1, 5, 22);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 136);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_TOUGH_GUARD);
         set_map_var(7, map_var(7) + 1);
         set_local_var(0, 1);
     end
@@ -107,12 +107,12 @@ end
 procedure damage_p_proc
 begin
     if (source_obj == dude_obj) then begin
-        if (has_trait(1, self_obj, 6) == 0) then begin
+        if (has_trait(TRAIT_OBJECT, self_obj, OBJECT_TEAM_NUM) == 0) then begin
             party_remove(self_obj);
         end
-        critter_add_trait(self_obj, 1, 6, 136);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 136);
         set_map_var(10, 1);
-        attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+        attack(dude_obj);
     end
 end
 
@@ -121,11 +121,11 @@ begin
     if (map_var(6) != 0) then begin
         script_overrides;
         if (map_var(13)) then begin
-            float_msg(self_obj, message_str(1494, 119), 0);
+            float_msg(self_obj, mstr(119), FLOAT_MSG_NORMAL);
         end
         else begin
             if (map_var(12)) then begin
-                float_msg(self_obj, message_str(1494, 111), 0);
+                float_msg(self_obj, mstr(111), FLOAT_MSG_NORMAL);
             end
             else begin
                 if ((map_var(3) == 1) and (GVar4 == 0)) then begin
@@ -146,21 +146,21 @@ begin
                         end_dialogue;
                     end
                     else begin
-                        float_msg(self_obj, message_str(1494, 113), 0);
+                        float_msg(self_obj, mstr(113), FLOAT_MSG_NORMAL);
                     end
                 end
             end
         end
     end
     else begin
-        float_msg(self_obj, message_str(1496, random(122, 124)), 0);
+        float_msg(self_obj, message_str(SCRIPT_DEMOCRYP, random(122, 124)), FLOAT_MSG_NORMAL);
     end
 end
 
 procedure look_at_p_proc
 begin
     script_overrides;
-    display_msg(message_str(1494, 100));
+    display_mstr(100);
 end
 
 procedure critter_p_proc
@@ -168,22 +168,22 @@ begin
     variable LVar0 := 0;
     variable LVar1 := 0;
     if (map_var(10) == 1) then begin
-        if (has_trait(1, self_obj, 6) == 0) then begin
+        if (has_trait(TRAIT_OBJECT, self_obj, OBJECT_TEAM_NUM) == 0) then begin
             party_remove(self_obj);
         end
-        critter_add_trait(self_obj, 1, 6, 136);
-        critter_add_trait(self_obj, 1, 5, 22);
-        attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 136);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_TOUGH_GUARD);
+        attack(dude_obj);
     end
     if (map_var(5) == 3) then begin
-        attack_complex(Fool_Pointer, 0, 1, 0, 0, 30000, 0, 0);
+        attack(Fool_Pointer);
     end
     Crypt_Pointer := self_obj;
     LVar0 := (self_obj == Fool_Pointer) or (self_obj == Crypt_Pointer) or (self_obj == Phrax_Pointer) or (self_obj == Rock_Pointer);
     if (Fool_Taunt) then begin
         Fool_Taunt := 0;
         if (random(1, 5) == 1) then begin
-            float_msg(self_obj, message_str(964, random(10200, 10218)), -1);
+            float_msg(self_obj, message_str(SCRIPT_KLATRAP, random(10200, 10218)), FLOAT_MSG_SEQUENTIAL);
         end
     end
     else begin
@@ -196,7 +196,7 @@ begin
             end
             else begin
                 if (GVar2) then begin
-                    attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+                    attack(dude_obj);
                 end
                 else begin
                     if (not(LVar0)) then begin
@@ -231,7 +231,7 @@ begin
                 else begin
                     if ((map_var(3) > 3) or (map_var(1) > 1)) then begin
                         if (is_success(do_check(self_obj, 1, 3))) then begin
-                            attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+                            attack(dude_obj);
                         end
                     end
                 end
@@ -253,8 +253,8 @@ begin
             if ((map_var(2) >= 1) and (tile_num(self_obj) != 27504)) then begin
                 if (GVar1 == 0) then begin
                     GVar1 := 1;
-                    float_msg(self_obj, message_str(964, random(10300, 10304)), 6);
-                    if (has_trait(1, self_obj, 6) == 0) then begin
+                    float_msg(self_obj, message_str(SCRIPT_KLATRAP, random(10300, 10304)), FLOAT_MSG_NEAR_WHITE);
+                    if (has_trait(TRAIT_OBJECT, self_obj, OBJECT_TEAM_NUM) == 0) then begin
                         party_remove(self_obj);
                     end
                 end
@@ -272,7 +272,7 @@ begin
     variable LVar0 := 0;
     if (fixed_param == 1) then begin
         Crypt_Taunt := 1;
-        float_msg(self_obj, message_str(964, random(10200, 10218)), -1);
+        float_msg(self_obj, message_str(SCRIPT_KLATRAP, random(10200, 10218)), FLOAT_MSG_SEQUENTIAL);
         if (LVar0 == 0) then begin
             add_timer_event(self_obj, 100, 2);
         end
@@ -295,55 +295,55 @@ begin
         set_map_var(5, 2);
     end
     if (source_obj == dude_obj) then begin
-        set_global_var(4, global_var(4) + 1);
-        if (metarule(51, self_obj) == 2) then begin
+        set_global_var(GVAR_BAD_MONSTER, global_var(GVAR_BAD_MONSTER) + 1);
+        if (critter_kill_type(self_obj) == KILL_TYPE_children_kills) then begin
             if (source_obj == dude_obj) then begin
-                set_global_var(2, 0);
-                set_global_var(1, global_var(1) + 1);
-                set_global_var(0, global_var(0) + -15);
+                set_global_var(GVAR_CHAMPION_REPUTATION, 0);
+                set_global_var(GVAR_CHILDKILLER_REPUTATION, global_var(GVAR_CHILDKILLER_REPUTATION) + 1);
+                set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + -15);
                 debug_msg("Player gains " + -15 + " Karma Points.");
-                set_global_var(37, 0);
-                set_global_var(38, 0);
-                set_global_var(39, 0);
-                set_global_var(40, 0);
-                set_global_var(41, 0);
-                set_global_var(42, 0);
-                set_global_var(43, 0);
-                set_global_var(44, 0);
-                set_global_var(45, 0);
-                if (global_var(0) >= 1000) then begin
-                    set_global_var(37, 1);
+                set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+                set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+                set_global_var(GVAR_KARMA_DEFENDER, 0);
+                set_global_var(GVAR_KARMA_WANDERER, 0);
+                set_global_var(GVAR_KARMA_BETRAYER, 0);
+                set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+                set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+                set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+                    set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
                 end
                 else begin
-                    if (global_var(0) >= 750) then begin
-                        set_global_var(38, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
                     end
                     else begin
-                        if (global_var(0) >= 500) then begin
-                            set_global_var(39, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                            set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                         end
                         else begin
-                            if (global_var(0) >= 250) then begin
-                                set_global_var(40, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                                set_global_var(GVAR_KARMA_DEFENDER, 1);
                             end
                             else begin
-                                if (global_var(0) > -250) then begin
-                                    set_global_var(41, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                                    set_global_var(GVAR_KARMA_WANDERER, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -500) then begin
-                                        set_global_var(42, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                        set_global_var(GVAR_KARMA_BETRAYER, 1);
                                     end
                                     else begin
-                                        if (global_var(0) > -750) then begin
-                                            set_global_var(43, 1);
+                                        if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                            set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                         end
                                         else begin
-                                            if (global_var(0) > -1000) then begin
-                                                set_global_var(44, 1);
+                                            if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                                set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                             end
                                             else begin
-                                                set_global_var(45, 1);
+                                                set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                             end
                                         end
                                     end
@@ -352,62 +352,62 @@ begin
                         end
                     end
                 end
-                set_global_var(808, global_var(808) + -5);
+                set_global_var(GVAR_GANG_2, global_var(GVAR_GANG_2) + -5);
                 debug_msg("Added " + -5 + " to Town Rep");
             end
         end
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(4) > (3 * global_var(5))) or (global_var(2) == 1)) and (global_var(1) == 0)) then begin
-            set_global_var(2, 1);
-            set_global_var(3, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(4) > (3 * global_var(GVAR_GOOD_MONSTER))) or (global_var(GVAR_CHAMPION_REPUTATION) == 1)) and (global_var(GVAR_CHILDKILLER_REPUTATION) == 0)) then begin
+            set_global_var(GVAR_CHAMPION_REPUTATION, 1);
+            set_global_var(GVAR_BERSERKER_REPUTATION, 0);
         end
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(5) > (2 * global_var(4))) or (global_var(3) == 1))) then begin
-            set_global_var(3, 1);
-            set_global_var(2, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
+            set_global_var(GVAR_BERSERKER_REPUTATION, 1);
+            set_global_var(GVAR_CHAMPION_REPUTATION, 0);
         end
-        set_global_var(0, global_var(0) + 5);
+        set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + 5);
         debug_msg("Player gains " + 5 + " Karma Points.");
-        set_global_var(37, 0);
-        set_global_var(38, 0);
-        set_global_var(39, 0);
-        set_global_var(40, 0);
-        set_global_var(41, 0);
-        set_global_var(42, 0);
-        set_global_var(43, 0);
-        set_global_var(44, 0);
-        set_global_var(45, 0);
-        if (global_var(0) >= 1000) then begin
-            set_global_var(37, 1);
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+        set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+        set_global_var(GVAR_KARMA_DEFENDER, 0);
+        set_global_var(GVAR_KARMA_WANDERER, 0);
+        set_global_var(GVAR_KARMA_BETRAYER, 0);
+        set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+        set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+            set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
         end
         else begin
-            if (global_var(0) >= 750) then begin
-                set_global_var(38, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
             end
             else begin
-                if (global_var(0) >= 500) then begin
-                    set_global_var(39, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                 end
                 else begin
-                    if (global_var(0) >= 250) then begin
-                        set_global_var(40, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                        set_global_var(GVAR_KARMA_DEFENDER, 1);
                     end
                     else begin
-                        if (global_var(0) > -250) then begin
-                            set_global_var(41, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                            set_global_var(GVAR_KARMA_WANDERER, 1);
                         end
                         else begin
-                            if (global_var(0) > -500) then begin
-                                set_global_var(42, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                set_global_var(GVAR_KARMA_BETRAYER, 1);
                             end
                             else begin
-                                if (global_var(0) > -750) then begin
-                                    set_global_var(43, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -1000) then begin
-                                        set_global_var(44, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                     end
                                     else begin
-                                        set_global_var(45, 1);
+                                        set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                     end
                                 end
                             end
@@ -416,7 +416,7 @@ begin
                 end
             end
         end
-        set_global_var(808, global_var(808) + 3);
+        set_global_var(GVAR_GANG_2, global_var(GVAR_GANG_2) + 3);
         debug_msg("Added " + 3 + " to Town Rep");
     end
 end
@@ -424,7 +424,7 @@ end
 procedure description_p_proc
 begin
     script_overrides;
-    display_msg(message_str(1494, 100));
+    display_mstr(100);
 end
 
 procedure Node999
@@ -433,55 +433,55 @@ end
 
 procedure Node000
 begin
-    gsay_reply(1494, 101);
-    giq_option(4, 1494, 102, Node002, 50);
-    giq_option(4, 1494, 103, Node001, 50);
+    Reply(101);
+    giq_option(4, NAME, 102, Node002, NEUTRAL_REACTION);
+    giq_option(4, NAME, 103, Node001, NEUTRAL_REACTION);
 end
 
 procedure Node001
 begin
-    gsay_reply(1494, 104);
-    giq_option(4, 1494, 105, combat, 50);
+    Reply(104);
+    giq_option(4, NAME, 105, combat, NEUTRAL_REACTION);
 end
 
 procedure Node002
 begin
-    gsay_reply(1494, 106);
-    giq_option(4, 1494, 107, Node003, 50);
-    giq_option(4, 1494, 108, Node999, 50);
+    Reply(106);
+    giq_option(4, NAME, 107, Node003, NEUTRAL_REACTION);
+    giq_option(4, NAME, 108, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node003
 begin
-    gsay_reply(1494, 109);
-    giq_option(4, 1494, 114, Node999, 50);
+    Reply(109);
+    giq_option(4, NAME, 114, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node004
 begin
-    gsay_reply(1494, 120);
-    giq_option(4, 1494, 121, Node006, 50);
-    giq_option(4, 1494, 122, Node005, 50);
+    Reply(120);
+    giq_option(4, NAME, 121, Node006, NEUTRAL_REACTION);
+    giq_option(4, NAME, 122, Node005, NEUTRAL_REACTION);
 end
 
 procedure Node005
 begin
-    gsay_reply(1494, 123);
-    giq_option(4, 1494, 124, combat, 50);
+    Reply(123);
+    giq_option(4, NAME, 124, combat, NEUTRAL_REACTION);
 end
 
 procedure Node006
 begin
-    gsay_reply(1494, 125);
-    giq_option(4, 1494, 126, Node007, 50);
-    giq_option(4, 1494, 129, Node999, 50);
+    Reply(125);
+    giq_option(4, NAME, 126, Node007, NEUTRAL_REACTION);
+    giq_option(4, NAME, 129, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node007
 begin
-    gsay_reply(1494, 127);
-    giq_option(4, 1494, 128, combat, 50);
-    giq_option(4, 1494, 130, Node999, 50);
+    Reply(127);
+    giq_option(4, NAME, 128, combat, NEUTRAL_REACTION);
+    giq_option(4, NAME, 130, Node999, NEUTRAL_REACTION);
 end
 
 procedure combat
@@ -497,12 +497,12 @@ procedure Start_War
 begin
     if (fixed_param == 1) then begin
         Crypt_Taunt := 1;
-        float_msg(self_obj, message_str(964, random(10200, 10218)), -1);
+        float_msg(self_obj, message_str(SCRIPT_KLATRAP, random(10200, 10218)), FLOAT_MSG_SEQUENTIAL);
         add_timer_event(self_obj, 200, 2);
     end
     else begin
         if (fixed_param == 2) then begin
-            attack_complex(Fool_Pointer, 0, 1, 0, 0, 30000, 0, 0);
+            attack(Fool_Pointer);
         end
     end
 end
@@ -510,4 +510,3 @@ end
 procedure phraxend
 begin
 end
-
