diff --git a/Mods/MegaMod/Original/POKER.ssl b/Mods/MegaMod/ReDefine/POKER.ssl
index 53192d6..ba9a03e 100644
--- a/Mods/MegaMod/Original/POKER.ssl
+++ b/Mods/MegaMod/ReDefine/POKER.ssl
@@ -299,8 +299,8 @@ begin
         call aquire_cash(NPC1, LVar0, 150);
     end
     total_cash_on_map_enter := get_cash(dealer) + get_cash(NPC1);
-    critter_add_trait(self_obj, 1, 6, 35);
-    critter_add_trait(self_obj, 1, 5, 24);
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 35);
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_TOUGH_CITIZEN);
 end
 
 procedure pow(variable arg0, variable arg1)
@@ -645,67 +645,67 @@ end
 procedure Start_Game_Dialog(variable arg0)
 begin
     if (NPC1 == 0) then begin
-        gsay_reply(1360, 250);
-        gsay_option(1360, 108, Node999, -1);
+        Reply(250);
+        giq_option(1, NAME, 108, Node999, -1);
     end
     else begin
         if ((get_cash(dealer) > bet) and (get_cash(NPC1) > bet)) then begin
             if (((total_cash_on_map_enter - (get_cash(dealer) + get_cash(NPC1))) >= 200) or (get_total_win() >= 200)) then begin
                 if (days_since_visited < 1) then begin
-                    gsay_reply(1360, 248);
+                    Reply(248);
                 end
                 else begin
-                    gsay_reply(1360, message_str(1360, 249) + generate_days());
+                    Reply(mstr(249) + generate_days());
                 end
             end
             else begin
                 if (not(arg0)) then begin
-                    gsay_reply(1360, message_str(1360, random(235, 237)) + pl_gender() + message_str(1360, random(261, 263)) + generate_ending(bet, 139) + message_str(1360, random(238, 240)) + generate_ending(bet / bet, 139) + message_str(1360, random(241, 243)) + (2 * 10 * bet) + message_str(1360, 113));
+                    Reply(mstr(random(235, 237)) + pl_gender() + mstr(random(261, 263)) + generate_ending(bet, 139) + mstr(random(238, 240)) + generate_ending(bet / bet, 139) + mstr(random(241, 243)) + (2 * 10 * bet) + mstr(113));
                 end
                 else begin
-                    gsay_reply(1360, message_str(1360, random(244, 246)));
+                    Reply(mstr(random(244, 246)));
                 end
-                if (get_critter_stat(dude_obj, 4) < 4) then begin
-                    gsay_option(1360, 251, Node_Dumb, -1);
-                    gsay_option(1360, 252, Node_Dumb, -1);
+                if (dude_iq < 4) then begin
+                    giq_option(1, NAME, 251, Node_Dumb, -1);
+                    giq_option(1, NAME, 252, Node_Dumb, -1);
                 end
                 else begin
                     if (get_cash(dude_obj) > bet) then begin
                         if (not(arg0)) then begin
-                            gsay_option(1360, 303, NodePokerStart, -1);
+                            giq_option(1, NAME, 303, NodePokerStart, -1);
                         end
                         else begin
-                            gsay_option(1360, 106, NodePokerStart, -1);
+                            giq_option(1, NAME, 106, NodePokerStart, -1);
                         end
                         if ((bet == 5) and (item_caps_total(dude_obj) >= 200)) then begin
-                            gsay_option(1360, 330, NodeP_BigGame, -1);
+                            giq_option(1, NAME, 330, NodeP_BigGame, -1);
                         end
                         if (bet == 10) then begin
-                            gsay_option(1360, 331, NodeP_NormalGame, -1);
+                            giq_option(1, NAME, 331, NodeP_NormalGame, -1);
                         end
                     end
                     else begin
-                        gsay_option(1360, 302, Node999, -1);
+                        giq_option(1, NAME, 302, Node999, -1);
                     end
                 end
             end
-            if (get_critter_stat(dude_obj, 4) > 3) then begin
+            if (dude_iq > 3) then begin
                 if (((total_cash_on_map_enter - (get_cash(dealer) + get_cash(NPC1))) >= 200) or (get_total_win() >= 200)) then begin
-                    gsay_option(1360, 304, Node999, -1);
+                    giq_option(1, NAME, 304, Node999, -1);
                 end
                 else begin
                     if (not(arg0)) then begin
-                        gsay_option(1360, 301, Node999, -1);
+                        giq_option(1, NAME, 301, Node999, -1);
                     end
                     else begin
-                        gsay_option(1360, 309, Poker_End_Seq, -1);
+                        giq_option(1, NAME, 309, Poker_End_Seq, -1);
                     end
                 end
             end
         end
         else begin
-            gsay_reply(1360, 207);
-            gsay_option(1360, 222, Node999, -1);
+            Reply(207);
+            giq_option(1, NAME, 222, Node999, -1);
         end
     end
 end
@@ -726,15 +726,15 @@ end
 procedure NodeP_BigGame
 begin
     bet := 10;
-    gsay_reply(1360, 219);
-    gsay_option(1360, 108, NodeP_Begin, -1);
+    Reply(219);
+    giq_option(1, NAME, 108, NodeP_Begin, -1);
 end
 
 procedure NodeP_NormalGame
 begin
     bet := 5;
-    gsay_reply(1360, 220);
-    gsay_option(1360, 108, NodeP_Begin, -1);
+    Reply(220);
+    giq_option(1, NAME, 108, NodeP_Begin, -1);
 end
 
 procedure NodePokerStart
@@ -744,30 +744,30 @@ begin
     call mod_cash(dude_obj, -bet);
     call mod_cash(dealer, -bet);
     call mod_cash(NPC1, -bet);
-    gsay_reply(1360, "[" + obj_name(dealer) + message_str(1360, 101));
+    Reply("[" + obj_name(dealer) + mstr(101));
     call Shuffle_Deck();
-    gsay_option(1360, 105, NodeP_PlayerViewCards, -1);
+    giq_option(1, NAME, 105, NodeP_PlayerViewCards, -1);
 end
 
 procedure NodeP_PlayerViewCards
 begin
     call setDeck(dude_obj);
     call Sort_By_Color();
-    gsay_reply(1360, message_str(1360, 305) + message_str(1360, get_hand(1)) + ", " + message_str(1360, get_hand(2)) + ", " + message_str(1360, get_hand(3)) + ", " + message_str(1360, get_hand(4)) + message_str(1360, 118) + message_str(1360, get_hand(5)) + "]");
+    Reply(mstr(305) + mstr(get_hand(1)) + ", " + mstr(get_hand(2)) + ", " + mstr(get_hand(3)) + ", " + mstr(get_hand(4)) + mstr(118) + mstr(get_hand(5)) + "]");
     call Sort();
-    gsay_option(1360, "[" + message_str(1360, 112) + showComb(dude_obj) + "]", NodeP_DealerChangeCards, -1);
+    giq_option(1, NAME, "[" + mstr(112) + showComb(dude_obj) + "]", NodeP_DealerChangeCards, -1);
     if (not(pl_change_done)) then begin
-        gsay_option(1360, 306, NodeP_PlayerChangeCard, -1);
+        giq_option(1, NAME, 306, NodeP_PlayerChangeCard, -1);
     end
 end
 
 procedure NodeP_PlayerChangeCard
 begin
-    gsay_reply(1360, pr_npc(dealer) + message_str(1360, 247));
-    gsay_option(1360, message_str(1360, 132) + message_str(1360, 130), Player_Change_1_Card, -1);
-    gsay_option(1360, message_str(1360, 133) + message_str(1360, 131), Player_Change_2_Card, -1);
-    gsay_option(1360, message_str(1360, 134) + message_str(1360, 131), Player_Change_3_Card, -1);
-    gsay_option(1360, message_str(1360, 135), NodeP_PlayerViewCards, -1);
+    Reply(pr_npc(dealer) + mstr(247));
+    giq_option(1, NAME, mstr(132) + mstr(130), Player_Change_1_Card, -1);
+    giq_option(1, NAME, mstr(133) + mstr(131), Player_Change_2_Card, -1);
+    giq_option(1, NAME, mstr(134) + mstr(131), Player_Change_3_Card, -1);
+    giq_option(1, NAME, mstr(135), NodeP_PlayerViewCards, -1);
 end
 
 procedure Done_change_cards
@@ -778,7 +778,7 @@ begin
     if (pl_change_done == 1) then begin
         while (LVar0 <= 5) do begin
             if (get_was_changed(LVar0)) then begin
-                gsay_reply(1360, message_str(1360, 310) + message_str(1360, get_hand(LVar0)) + message_str(1360, 113) + "]");
+                Reply(mstr(310) + mstr(get_hand(LVar0)) + mstr(113) + "]");
                 LVar0 := 6;
             end
             LVar0 := LVar0 + 1;
@@ -790,7 +790,7 @@ begin
                 LVar1 := LVar0 + 1;
                 while (LVar1 <= 5) do begin
                     if (get_was_changed(LVar0) and get_was_changed(LVar1)) then begin
-                        gsay_reply(1360, message_str(1360, 310) + message_str(1360, get_hand(LVar0)) + message_str(1360, 118) + message_str(1360, get_hand(LVar1)) + message_str(1360, 113) + "]");
+                        Reply(mstr(310) + mstr(get_hand(LVar0)) + mstr(118) + mstr(get_hand(LVar1)) + mstr(113) + "]");
                         LVar0 := 6;
                         LVar1 := 6;
                     end
@@ -806,7 +806,7 @@ begin
                     LVar2 := LVar1 + 1;
                     while (LVar2 <= 5) do begin
                         if (get_was_changed(LVar0) and get_was_changed(LVar1) and get_was_changed(LVar2)) then begin
-                            gsay_reply(1360, message_str(1360, 310) + message_str(1360, get_hand(LVar0)) + ", " + message_str(1360, get_hand(LVar1)) + message_str(1360, 118) + message_str(1360, get_hand(LVar2)) + message_str(1360, 113) + "]");
+                            Reply(mstr(310) + mstr(get_hand(LVar0)) + ", " + mstr(get_hand(LVar1)) + mstr(118) + mstr(get_hand(LVar2)) + mstr(113) + "]");
                             LVar0 := 6;
                             LVar1 := 6;
                             LVar2 := 6;
@@ -819,7 +819,7 @@ begin
             end
         end
     end
-    gsay_option(1360, 108, NodeP_PlayerViewCards, -1);
+    giq_option(1, NAME, 108, NodeP_PlayerViewCards, -1);
 end
 
 procedure Player_Change_1_Card
@@ -842,21 +842,21 @@ end
 
 procedure Show_change_cards
 begin
-    gsay_reply(1360, 221);
+    Reply(221);
     if (not(get_was_changed(1))) then begin
-        gsay_option(1360, message_str(1360, 307) + message_str(1360, get_hand(1)), NodeChange1, -1);
+        giq_option(1, NAME, mstr(307) + mstr(get_hand(1)), NodeChange1, -1);
     end
     if (not(get_was_changed(2))) then begin
-        gsay_option(1360, message_str(1360, 307) + message_str(1360, get_hand(2)), NodeChange2, -1);
+        giq_option(1, NAME, mstr(307) + mstr(get_hand(2)), NodeChange2, -1);
     end
     if (not(get_was_changed(3))) then begin
-        gsay_option(1360, message_str(1360, 307) + message_str(1360, get_hand(3)), NodeChange3, -1);
+        giq_option(1, NAME, mstr(307) + mstr(get_hand(3)), NodeChange3, -1);
     end
     if (not(get_was_changed(4))) then begin
-        gsay_option(1360, message_str(1360, 307) + message_str(1360, get_hand(4)), NodeChange4, -1);
+        giq_option(1, NAME, mstr(307) + mstr(get_hand(4)), NodeChange4, -1);
     end
     if (not(get_was_changed(5))) then begin
-        gsay_option(1360, message_str(1360, 307) + message_str(1360, get_hand(5)), NodeChange5, -1);
+        giq_option(1, NAME, mstr(307) + mstr(get_hand(5)), NodeChange5, -1);
     end
 end
 
@@ -868,12 +868,12 @@ begin
     end
     else begin
         if (pl_change_done == 1) then begin
-            gsay_reply(1360, message_str(1360, 334) + pl_change_done + message_str(1360, 130) + message_str(1360, 118) + obj_name(dealer) + message_str(1360, 335));
+            Reply(mstr(334) + pl_change_done + mstr(130) + mstr(118) + obj_name(dealer) + mstr(335));
         end
         else begin
-            gsay_reply(1360, message_str(1360, 334) + pl_change_done + message_str(1360, 131) + message_str(1360, 118) + obj_name(dealer) + message_str(1360, 335));
+            Reply(mstr(334) + pl_change_done + mstr(131) + mstr(118) + obj_name(dealer) + mstr(335));
         end
-        gsay_option(1360, 308, Done_change_cards, -1);
+        giq_option(1, NAME, 308, Done_change_cards, -1);
     end
 end
 
@@ -924,47 +924,47 @@ begin
         call set_try_open(dealer, 0);
         call set_try_open(NPC1, 0);
         if (((2 * 10 * bet) - stash) > (bet / bet * 2)) then begin
-            gsay_reply(1360, "[" + message_str(1360, 208) + generate_ending(stash, 139) + message_str(1360, 209) + " " + message_str(1360, 112) + showComb(dude_obj) + "]");
+            Reply("[" + mstr(208) + generate_ending(stash, 139) + mstr(209) + " " + mstr(112) + showComb(dude_obj) + "]");
         end
         else begin
-            gsay_reply(1360, "[" + message_str(1360, 103) + generate_ending(stash, 139) + message_str(1360, 104) + " " + message_str(1360, 112) + showComb(dude_obj) + "]");
+            Reply("[" + mstr(103) + generate_ending(stash, 139) + mstr(104) + " " + mstr(112) + showComb(dude_obj) + "]");
         end
         if (not(dealer_check or dealer_pass)) then begin
-            gsay_option(1360, message_str(1360, 325) + obj_name(dealer) + message_str(1360, 326), NodeCheckDealer, -1);
+            giq_option(1, NAME, mstr(325) + obj_name(dealer) + mstr(326), NodeCheckDealer, -1);
         end
         if (not(npc1_check or npc1_pass)) then begin
-            gsay_option(1360, message_str(1360, 325) + obj_name(NPC1) + message_str(1360, 326), NodeCheckNPC1, -1);
+            giq_option(1, NAME, mstr(325) + obj_name(NPC1) + mstr(326), NodeCheckNPC1, -1);
         end
         if (not(try_open)) then begin
-            gsay_option(1360, 129, NodeP_Open, -1);
+            giq_option(1, NAME, 129, NodeP_Open, -1);
         end
         if (stash < (2 * 10 * bet)) then begin
             if ((get_cash(dude_obj) > (bet / bet)) and ((bet / bet) <= (((2 * 10 * bet) - stash) / 2))) then begin
-                gsay_option(1360, message_str(1360, 328) + generate_ending(bet / bet, 139) + message_str(1360, 329) + (stash + (bet / bet)), Add10, -1);
+                giq_option(1, NAME, mstr(328) + generate_ending(bet / bet, 139) + mstr(329) + (stash + (bet / bet)), Add10, -1);
             end
             if ((get_cash(dude_obj) > (bet / 2)) and ((bet / 2) <= (((2 * 10 * bet) - stash) / 2))) then begin
-                gsay_option(1360, message_str(1360, 328) + generate_ending(bet / 2, 139) + message_str(1360, 329) + (stash + (bet / 2)), Add20, -1);
+                giq_option(1, NAME, mstr(328) + generate_ending(bet / 2, 139) + mstr(329) + (stash + (bet / 2)), Add20, -1);
             end
             if ((get_cash(dude_obj) > bet) and (bet <= (((2 * 10 * bet) - stash) / 2))) then begin
-                gsay_option(1360, message_str(1360, 328) + generate_ending(bet, 139) + message_str(1360, 329) + (stash + bet), Add50, -1);
+                giq_option(1, NAME, mstr(328) + generate_ending(bet, 139) + mstr(329) + (stash + bet), Add50, -1);
             end
             if ((get_cash(dude_obj) > (2 * bet)) and ((2 * bet) <= (((2 * 10 * bet) - stash) / 2))) then begin
-                gsay_option(1360, message_str(1360, 328) + generate_ending(2 * bet, 139) + message_str(1360, 329) + (stash + (2 * bet)), Add100, -1);
+                giq_option(1, NAME, mstr(328) + generate_ending(2 * bet, 139) + mstr(329) + (stash + (2 * bet)), Add100, -1);
             end
         end
-        gsay_option(1360, 324, NodeP_PlayerPass, -1);
+        giq_option(1, NAME, 324, NodeP_PlayerPass, -1);
     end
     else begin
         if (step == 1) then begin
             step := 2;
-            gsay_reply(1360, message_str(1360, 227) + obj_name(dealer) + message_str(1360, 228));
-            gsay_option(1360, 108, NodeP_BarterDealer, -1);
+            Reply(mstr(227) + obj_name(dealer) + mstr(228));
+            giq_option(1, NAME, 108, NodeP_BarterDealer, -1);
         end
         else begin
             if (step == 2) then begin
                 step := 0;
-                gsay_reply(1360, message_str(1360, 227) + obj_name(NPC1) + message_str(1360, 228));
-                gsay_option(1360, 108, NodeP_BarterNPC1, -1);
+                Reply(mstr(227) + obj_name(NPC1) + mstr(228));
+                giq_option(1, NAME, 108, NodeP_BarterNPC1, -1);
             end
         end
     end
@@ -986,25 +986,25 @@ begin
     LVar0 := stash;
     if (get_pass(dealer)) then begin
         call mod_cash(NPC1, stash);
-        gsay_reply(1360, message_str(1360, 116) + obj_name(NPC1) + message_str(1360, 229) + generate_ending(LVar0, 139) + message_str(1360, 230));
+        Reply(mstr(116) + obj_name(NPC1) + mstr(229) + generate_ending(LVar0, 139) + mstr(230));
     end
     else begin
         if (get_pass(NPC1)) then begin
             call mod_cash(dealer, stash);
-            gsay_reply(1360, message_str(1360, 116) + obj_name(dealer) + message_str(1360, 229) + generate_ending(LVar0, 139) + message_str(1360, 230));
+            Reply(mstr(116) + obj_name(dealer) + mstr(229) + generate_ending(LVar0, 139) + mstr(230));
         end
         else begin
             if (checkComb(dealer) >= checkComb(NPC1)) then begin
                 call mod_cash(dealer, stash);
-                gsay_reply(1360, message_str(1360, 116) + obj_name(dealer) + message_str(1360, 118) + obj_name(NPC1) + message_str(1360, 117) + obj_name(dealer) + message_str(1360, 229) + generate_ending(LVar0, 139) + message_str(1360, 230));
+                Reply(mstr(116) + obj_name(dealer) + mstr(118) + obj_name(NPC1) + mstr(117) + obj_name(dealer) + mstr(229) + generate_ending(LVar0, 139) + mstr(230));
             end
             else begin
                 call mod_cash(NPC1, stash);
-                gsay_reply(1360, message_str(1360, 116) + obj_name(dealer) + message_str(1360, 118) + obj_name(NPC1) + message_str(1360, 117) + obj_name(NPC1) + message_str(1360, 229) + generate_ending(LVar0, 139) + message_str(1360, 230));
+                Reply(mstr(116) + obj_name(dealer) + mstr(118) + obj_name(NPC1) + mstr(117) + obj_name(NPC1) + mstr(229) + generate_ending(LVar0, 139) + mstr(230));
             end
         end
     end
-    gsay_option(1360, 108, NodeP_Again, -1);
+    giq_option(1, NAME, 108, NodeP_Again, -1);
 end
 
 procedure NodeP_PlayerAccept
@@ -1023,23 +1023,23 @@ begin
     end
     else begin
         if (NPCBarter(LVar0) and (get_cash(LVar0) > get_bet(other_npc(LVar0)))) then begin
-            gsay_reply(1360, pr_npc(LVar0) + message_str(1360, 115));
+            Reply(pr_npc(LVar0) + mstr(115));
             call mod_cash(LVar0, -get_bet(other_npc(LVar0)));
-            gsay_option(1360, 108, NodeP_Barter, -1);
+            giq_option(1, NAME, 108, NodeP_Barter, -1);
         end
         else begin
-            gsay_reply(1360, pr_npc(LVar0) + message_str(1360, 114));
+            Reply(pr_npc(LVar0) + mstr(114));
             call set_pass(LVar0);
-            gsay_option(1360, 108, NodeP_Barter, -1);
+            giq_option(1, NAME, 108, NodeP_Barter, -1);
         end
     end
 end
 
 procedure NodeP_NPCPass
 begin
-    gsay_reply(1360, 213);
+    Reply(213);
     item_caps_adjust(dude_obj, stash);
-    gsay_option(1360, message_str(1360, 322) + generate_ending(item_caps_total(dude_obj) - pl_cash, 139) + message_str(1360, 323), NodeP_Again, -1);
+    giq_option(1, NAME, mstr(322) + generate_ending(item_caps_total(dude_obj) - pl_cash, 139) + mstr(323), NodeP_Again, -1);
 end
 
 procedure NodeCheckDealer
@@ -1059,31 +1059,31 @@ begin
     variable LVar1 := 0;
     LVar1 := check_partner_comb(dude_obj, arg0);
     if (LVar1 == 5) then begin
-        gsay_reply(1360, 321);
+        Reply(321);
     end
     else begin
         if (LVar1 == 6) then begin
-            gsay_reply(1360, 332);
+            Reply(332);
         end
         else begin
             if (LVar1 == 7) then begin
-                gsay_reply(1360, 333);
+                Reply(333);
             end
             else begin
                 if (LVar1 == 4) then begin
-                    gsay_reply(1360, 317);
+                    Reply(317);
                 end
                 else begin
                     if (LVar1 == 3) then begin
-                        gsay_reply(1360, 318);
+                        Reply(318);
                     end
                     else begin
                         if (LVar1 == 2) then begin
-                            gsay_reply(1360, 319);
+                            Reply(319);
                         end
                         else begin
                             if (LVar1 <= 1) then begin
-                                gsay_reply(1360, 320);
+                                Reply(320);
                             end
                         end
                     end
@@ -1091,24 +1091,24 @@ begin
             end
         end
     end
-    gsay_option(1360, 108, NodeP_Barter, -1);
+    giq_option(1, NAME, 108, NodeP_Barter, -1);
 end
 
 procedure Show_Dealer_cards
 begin
-    gsay_reply(1360, pr_npc(dealer) + message_str(1360, 112) + showComb(dealer) + message_str(1360, 113));
+    Reply(pr_npc(dealer) + mstr(112) + showComb(dealer) + mstr(113));
     if (get_pass(NPC1)) then begin
-        gsay_option(1360, 108, NodeP_CompareEnd, -1);
+        giq_option(1, NAME, 108, NodeP_CompareEnd, -1);
     end
     else begin
-        gsay_option(1360, 108, Show_NPC1_cards, -1);
+        giq_option(1, NAME, 108, Show_NPC1_cards, -1);
     end
 end
 
 procedure Show_NPC1_cards
 begin
-    gsay_reply(1360, pr_npc(NPC1) + message_str(1360, 112) + showComb(NPC1) + message_str(1360, 113));
-    gsay_option(1360, 108, NodeP_CompareEnd, -1);
+    Reply(pr_npc(NPC1) + mstr(112) + showComb(NPC1) + mstr(113));
+    giq_option(1, NAME, 108, NodeP_CompareEnd, -1);
 end
 
 procedure NodeP_CompareEnd
@@ -1137,21 +1137,21 @@ begin
         LVar5 := checkComb(LVar0);
         if (LVar3 < LVar5) then begin
             call mod_cash(LVar0, stash);
-            gsay_reply(1360, pr_npc(dealer) + message_str(1360, 231) + obj_name(LVar0) + message_str(1360, 113));
-            gsay_option(1360, message_str(1360, 315) + message_str(1360, 110) + generate_ending(LVar6, 139) + message_str(1360, 111), NodeP_Again, -1);
+            Reply(pr_npc(dealer) + mstr(231) + obj_name(LVar0) + mstr(113));
+            giq_option(1, NAME, mstr(315) + mstr(110) + generate_ending(LVar6, 139) + mstr(111), NodeP_Again, -1);
         end
         else begin
             if (LVar3 > LVar5) then begin
                 call mod_cash(dude_obj, stash);
-                gsay_reply(1360, pr_npc(dealer) + message_str(1360, 231) + obj_name(dude_obj) + message_str(1360, 113));
-                gsay_option(1360, message_str(1360, 313) + message_str(1360, 110) + generate_ending(LVar6, 139) + message_str(1360, 111), NodeP_Again, -1);
+                Reply(pr_npc(dealer) + mstr(231) + dude_name + mstr(113));
+                giq_option(1, NAME, mstr(313) + mstr(110) + generate_ending(LVar6, 139) + mstr(111), NodeP_Again, -1);
             end
             else begin
                 item_caps_adjust(dude_obj, stash / 2);
                 call mod_cash(LVar0, stash / 2);
                 item_caps_adjust(dude_obj, stash - (stash / 2 * 2));
-                gsay_reply(1360, pr_npc(dealer) + obj_name(dude_obj) + message_str(1360, 118) + obj_name(LVar0) + message_str(1360, 119) + message_str(1360, 121) + message_str(1360, 120));
-                gsay_option(1360, message_str(1360, 314) + message_str(1360, 110) + generate_ending(stash, 139) + message_str(1360, 111), NodeP_Again, -1);
+                Reply(pr_npc(dealer) + dude_name + mstr(118) + obj_name(LVar0) + mstr(119) + mstr(121) + mstr(120));
+                giq_option(1, NAME, mstr(314) + mstr(110) + generate_ending(stash, 139) + mstr(111), NodeP_Again, -1);
             end
         end
     end
@@ -1163,8 +1163,8 @@ begin
             call mod_cash(NPC1, stash / 3);
             call mod_cash(dealer, stash / 3);
             item_caps_adjust(dude_obj, stash);
-            gsay_reply(1360, pr_npc(dealer) + message_str(1360, 232) + message_str(1360, 119) + message_str(1360, 122) + message_str(1360, 120));
-            gsay_option(1360, message_str(1360, 314) + message_str(1360, 110) + generate_ending(stash, 139) + message_str(1360, 111), NodeP_Again, -1);
+            Reply(pr_npc(dealer) + mstr(232) + mstr(119) + mstr(122) + mstr(120));
+            giq_option(1, NAME, mstr(314) + mstr(110) + generate_ending(stash, 139) + mstr(111), NodeP_Again, -1);
         end
         else begin
             if ((LVar3 > LVar4) and (LVar3 > LVar5)) then begin
@@ -1197,24 +1197,24 @@ begin
                 end
             end
             if (LVar0) then begin
-                gsay_reply(1360, pr_npc(dealer) + message_str(1360, 231) + obj_name(LVar0) + message_str(1360, 113));
+                Reply(pr_npc(dealer) + mstr(231) + obj_name(LVar0) + mstr(113));
                 call mod_cash(LVar0, stash);
                 if (LVar0 == dude_obj) then begin
-                    gsay_option(1360, message_str(1360, 313) + message_str(1360, 110) + generate_ending(LVar6, 139) + message_str(1360, 111), NodeP_Again, -1);
+                    giq_option(1, NAME, mstr(313) + mstr(110) + generate_ending(LVar6, 139) + mstr(111), NodeP_Again, -1);
                 end
                 else begin
-                    gsay_option(1360, message_str(1360, 315) + message_str(1360, 110) + generate_ending(LVar6, 139) + message_str(1360, 111), NodeP_Again, -1);
+                    giq_option(1, NAME, mstr(315) + mstr(110) + generate_ending(LVar6, 139) + mstr(111), NodeP_Again, -1);
                 end
             end
             else begin
-                gsay_reply(1360, pr_npc(dealer) + message_str(1360, 142) + obj_name(LVar1) + message_str(1360, 118) + obj_name(LVar2) + message_str(1360, 119) + message_str(1360, 121) + message_str(1360, 120));
+                Reply(pr_npc(dealer) + mstr(142) + obj_name(LVar1) + mstr(118) + obj_name(LVar2) + mstr(119) + mstr(121) + mstr(120));
                 call mod_cash(LVar1, (stash / 2) + (stash % 2));
                 call mod_cash(LVar2, stash / 2);
                 if ((LVar1 == dude_obj) or (LVar2 == dude_obj)) then begin
-                    gsay_option(1360, message_str(1360, 314) + message_str(1360, 110) + generate_ending(LVar6, 139) + message_str(1360, 111), NodeP_Again, -1);
+                    giq_option(1, NAME, mstr(314) + mstr(110) + generate_ending(LVar6, 139) + mstr(111), NodeP_Again, -1);
                 end
                 else begin
-                    gsay_option(1360, message_str(1360, 315) + message_str(1360, 110) + generate_ending(LVar6, 139) + message_str(1360, 111), NodeP_Again, -1);
+                    giq_option(1, NAME, mstr(315) + mstr(110) + generate_ending(LVar6, 139) + mstr(111), NodeP_Again, -1);
                 end
             end
         end
@@ -1321,17 +1321,17 @@ begin
     end
     else begin
         if (check_open(dealer)) then begin
-            gsay_reply(1360, pr_npc(dealer) + message_str(1360, random(123, 125)));
+            Reply(pr_npc(dealer) + mstr(random(123, 125)));
             if (not(get_pass(NPC1))) then begin
-                gsay_option(1360, 108, NodeP_NPC1Open, -1);
+                giq_option(1, NAME, 108, NodeP_NPC1Open, -1);
             end
             else begin
-                gsay_option(1360, 108, NodePokerDoCompare, -1);
+                giq_option(1, NAME, 108, NodePokerDoCompare, -1);
             end
         end
         else begin
-            gsay_reply(1360, pr_npc(dealer) + message_str(1360, random(126, 128)));
-            gsay_option(1360, 108, NodeP_Barter, -1);
+            Reply(pr_npc(dealer) + mstr(random(126, 128)));
+            giq_option(1, NAME, 108, NodeP_Barter, -1);
         end
     end
 end
@@ -1342,19 +1342,19 @@ begin
         call NodeP_Barter();
     end
     if (check_open(NPC1)) then begin
-        gsay_reply(1360, pr_npc(NPC1) + message_str(1360, random(123, 125)));
-        gsay_option(1360, 108, NodePokerDoCompare, -1);
+        Reply(pr_npc(NPC1) + mstr(random(123, 125)));
+        giq_option(1, NAME, 108, NodePokerDoCompare, -1);
     end
     else begin
-        gsay_reply(1360, pr_npc(NPC1) + message_str(1360, random(126, 128)));
-        gsay_option(1360, 108, NodeP_Barter, -1);
+        Reply(pr_npc(NPC1) + mstr(random(126, 128)));
+        giq_option(1, NAME, 108, NodeP_Barter, -1);
     end
 end
 
 procedure Node_Dumb
 begin
-    gsay_reply(1360, message_str(1360, 253));
-    gsay_option(1360, 254, Node999, -1);
+    Reply(mstr(253));
+    giq_option(1, NAME, 254, Node999, -1);
 end
 
 procedure Node999
@@ -1686,10 +1686,10 @@ end
 
 procedure CheckLuck(variable arg0)
 begin
-    if (random(1, 200) <= (get_critter_stat(arg0, 6) * get_critter_stat(arg0, 6))) then begin
+    if (random(1, 200) <= (get_critter_stat(arg0, STAT_lu) * get_critter_stat(arg0, STAT_lu))) then begin
         return 2;
     end
-    if (random(1, 20) <= get_critter_stat(arg0, 6)) then begin
+    if (random(1, 20) <= get_critter_stat(arg0, STAT_lu)) then begin
         return 1;
     end
     return 0;
@@ -1702,31 +1702,31 @@ begin
     call setDeck(arg0);
     LVar2 := CheckLuck(arg0);
     if (LVar2 == 2) then begin
-        if (random(1, 1000000) <= (50000 + (50000 / 10 * (get_critter_stat(arg0, 6) - 5)))) then begin
+        if (random(1, 1000000) <= (50000 + (50000 / 10 * (get_critter_stat(arg0, STAT_lu) - 5)))) then begin
             call give_straight_flesh();
         end
         else begin
-            if (random(1, 1000000) <= (100000 + (100000 / 10 * (get_critter_stat(arg0, 6) - 5)))) then begin
+            if (random(1, 1000000) <= (100000 + (100000 / 10 * (get_critter_stat(arg0, STAT_lu) - 5)))) then begin
                 call give_four();
             end
             else begin
-                if (random(1, 1000000) <= (200000 + (200000 / 10 * (get_critter_stat(arg0, 6) - 5)))) then begin
+                if (random(1, 1000000) <= (200000 + (200000 / 10 * (get_critter_stat(arg0, STAT_lu) - 5)))) then begin
                     call give_full_house();
                 end
                 else begin
-                    if (random(1, 1000000) <= (300000 + (300000 / 10 * (get_critter_stat(arg0, 6) - 5)))) then begin
+                    if (random(1, 1000000) <= (300000 + (300000 / 10 * (get_critter_stat(arg0, STAT_lu) - 5)))) then begin
                         call give_flesh();
                     end
                     else begin
-                        if (random(1, 1000000) <= (400000 + (400000 / 10 * (get_critter_stat(arg0, 6) - 5)))) then begin
+                        if (random(1, 1000000) <= (400000 + (400000 / 10 * (get_critter_stat(arg0, STAT_lu) - 5)))) then begin
                             call give_straight();
                         end
                         else begin
-                            if (random(1, 1000000) <= (500000 + (500000 / 10 * (get_critter_stat(arg0, 6) - 5)))) then begin
+                            if (random(1, 1000000) <= (500000 + (500000 / 10 * (get_critter_stat(arg0, STAT_lu) - 5)))) then begin
                                 call give_three();
                             end
                             else begin
-                                if (random(1, 1000000) <= (750000 + (750000 / 10 * (get_critter_stat(arg0, 6) - 5)))) then begin
+                                if (random(1, 1000000) <= (750000 + (750000 / 10 * (get_critter_stat(arg0, STAT_lu) - 5)))) then begin
                                     call give_two_pairs();
                                 end
                                 else begin
@@ -1741,31 +1741,31 @@ begin
     end
     else begin
         if (LVar2 == 1) then begin
-            if (random(1, 1000000) <= (30 + (30 / 10 * (get_critter_stat(arg0, 6) - 5)))) then begin
+            if (random(1, 1000000) <= (30 + (30 / 10 * (get_critter_stat(arg0, STAT_lu) - 5)))) then begin
                 call give_straight_flesh();
             end
             else begin
-                if (random(1, 1000000) <= (480 + (480 / 10 * (get_critter_stat(arg0, 6) - 5)))) then begin
+                if (random(1, 1000000) <= (480 + (480 / 10 * (get_critter_stat(arg0, STAT_lu) - 5)))) then begin
                     call give_four();
                 end
                 else begin
-                    if (random(1, 1000000) <= (2800 + (2800 / 10 * (get_critter_stat(arg0, 6) - 5)))) then begin
+                    if (random(1, 1000000) <= (2800 + (2800 / 10 * (get_critter_stat(arg0, STAT_lu) - 5)))) then begin
                         call give_full_house();
                     end
                     else begin
-                        if (random(1, 1000000) <= (4000 + (4000 / 10 * (get_critter_stat(arg0, 6) - 5)))) then begin
+                        if (random(1, 1000000) <= (4000 + (4000 / 10 * (get_critter_stat(arg0, STAT_lu) - 5)))) then begin
                             call give_flesh();
                         end
                         else begin
-                            if (random(1, 1000000) <= (7800 + (7800 / 10 * (get_critter_stat(arg0, 6) - 5)))) then begin
+                            if (random(1, 1000000) <= (7800 + (7800 / 10 * (get_critter_stat(arg0, STAT_lu) - 5)))) then begin
                                 call give_straight();
                             end
                             else begin
-                                if (random(1, 1000000) <= (42200 + (42200 / 10 * (get_critter_stat(arg0, 6) - 5)))) then begin
+                                if (random(1, 1000000) <= (42200 + (42200 / 10 * (get_critter_stat(arg0, STAT_lu) - 5)))) then begin
                                     call give_three();
                                 end
                                 else begin
-                                    if (random(1, 1000000) <= (95000 + (95000 / 10 * (get_critter_stat(arg0, 6) - 5)))) then begin
+                                    if (random(1, 1000000) <= (95000 + (95000 / 10 * (get_critter_stat(arg0, STAT_lu) - 5)))) then begin
                                         call give_two_pairs();
                                     end
                                     else begin
@@ -1893,30 +1893,30 @@ begin
     if (has_pairs()) then begin
         if (has_three()) then begin
             if (has_full_house()) then begin
-                return message_str(1360, 64);
+                return mstr(64);
             end
             if (Has_four()) then begin
-                return message_str(1360, 65) + message_str(1360, get_hand(get_comb(21))) + ", " + message_str(1360, get_hand(get_comb(23))) + ", " + message_str(1360, get_hand(get_comb(25))) + message_str(1360, 118) + message_str(1360, get_hand(get_comb(28)));
+                return mstr(65) + mstr(get_hand(get_comb(21))) + ", " + mstr(get_hand(get_comb(23))) + ", " + mstr(get_hand(get_comb(25))) + mstr(118) + mstr(get_hand(get_comb(28)));
             end
-            return message_str(1360, 62) + message_str(1360, get_hand(get_comb(13))) + ", " + message_str(1360, get_hand(get_comb(15))) + message_str(1360, 118) + message_str(1360, get_hand(get_comb(18)));
+            return mstr(62) + mstr(get_hand(get_comb(13))) + ", " + mstr(get_hand(get_comb(15))) + mstr(118) + mstr(get_hand(get_comb(18)));
         end
         if (has_pairs() == 2) then begin
-            return message_str(1360, 61) + message_str(1360, get_hand(get_comb(1))) + message_str(1360, 118) + message_str(1360, get_hand(get_comb(4))) + ", " + message_str(1360, get_hand(get_comb(7))) + message_str(1360, 118) + message_str(1360, get_hand(get_comb(10)));
+            return mstr(61) + mstr(get_hand(get_comb(1))) + mstr(118) + mstr(get_hand(get_comb(4))) + ", " + mstr(get_hand(get_comb(7))) + mstr(118) + mstr(get_hand(get_comb(10)));
         end
-        return message_str(1360, 60) + message_str(1360, get_hand(get_comb(1))) + message_str(1360, 118) + message_str(1360, get_hand(get_comb(4)));
+        return mstr(60) + mstr(get_hand(get_comb(1))) + mstr(118) + mstr(get_hand(get_comb(4)));
     end
     if (has_straight()) then begin
         if (Has_flesh()) then begin
-            return message_str(1360, 67) + message_str(1360, 159) + message_str(1360, get_hand(1)) + message_str(1360, 160) + message_str(1360, get_hand(5));
+            return mstr(67) + mstr(159) + mstr(get_hand(1)) + mstr(160) + mstr(get_hand(5));
         end
         else begin
-            return message_str(1360, 63) + message_str(1360, 159) + message_str(1360, get_hand(1)) + message_str(1360, 160) + message_str(1360, get_hand(5));
+            return mstr(63) + mstr(159) + mstr(get_hand(1)) + mstr(160) + mstr(get_hand(5));
         end
     end
     if (Has_flesh()) then begin
-        return message_str(1360, Has_flesh()) + message_str(1360, 66) + message_str(1360, 159) + message_str(1360, get_hand(1));
+        return mstr(Has_flesh()) + mstr(66) + mstr(159) + mstr(get_hand(1));
     end
-    return message_str(1360, 68) + message_str(1360, get_hand(5));
+    return mstr(68) + mstr(get_hand(5));
 end
 
 procedure ChangeCard(variable arg0, variable arg1)
@@ -2029,9 +2029,9 @@ procedure check_partner_comb(variable arg0, variable arg1)
 begin
     variable LVar2 := 0;
     variable LVar3 := 0;
-    LVar2 := has_skill(arg0, 16);
-    LVar3 := has_skill(arg1, 16);
-    if (get_critter_stat(arg0, 1) < 4) then begin
+    LVar2 := has_skill(arg0, SKILL_GAMBLING);
+    LVar3 := has_skill(arg1, SKILL_GAMBLING);
+    if (get_critter_stat(arg0, STAT_pe) < 4) then begin
         return 6;
     end
     if (LVar2 < (LVar3 / 2)) then begin
@@ -2082,32 +2082,32 @@ end
 procedure bart_with_player(variable arg0)
 begin
     if (NPCBarter(arg0) and (get_cash(arg0) > get_bet(dude_obj))) then begin
-        gsay_reply(1360, pr_npc(arg0) + message_str(1360, 115));
+        Reply(pr_npc(arg0) + mstr(115));
         call mod_cash(arg0, -get_bet(dude_obj));
         if (arg0 == dealer) then begin
             if (get_pass(NPC1)) then begin
-                gsay_option(1360, 108, NodeP_Barter, -1);
+                giq_option(1, NAME, 108, NodeP_Barter, -1);
             end
             else begin
-                gsay_option(1360, 108, player_barter_NPC1, -1);
+                giq_option(1, NAME, 108, player_barter_NPC1, -1);
             end
         end
         else begin
-            gsay_option(1360, 108, NodeP_Barter, -1);
+            giq_option(1, NAME, 108, NodeP_Barter, -1);
         end
     end
     else begin
-        gsay_reply(1360, pr_npc(arg0) + message_str(1360, 114));
+        Reply(pr_npc(arg0) + mstr(114));
         call set_pass(arg0);
         if (get_pass(other_npc(arg0))) then begin
-            gsay_option(1360, 108, NodeP_NPCPass, -1);
+            giq_option(1, NAME, 108, NodeP_NPCPass, -1);
         end
         else begin
             if (arg0 == dealer) then begin
-                gsay_option(1360, 108, player_barter_NPC1, -1);
+                giq_option(1, NAME, 108, player_barter_NPC1, -1);
             end
             else begin
-                gsay_option(1360, 108, NodeP_Barter, -1);
+                giq_option(1, NAME, 108, NodeP_Barter, -1);
             end
         end
     end
@@ -2330,16 +2330,16 @@ begin
     end
     else begin
         if (check_open(other_npc(arg0))) then begin
-            gsay_reply(1360, pr_npc(other_npc(arg0)) + message_str(1360, random(123, 125)));
-            gsay_option(1360, 108, NodePokerDoCompare, -1);
+            Reply(pr_npc(other_npc(arg0)) + mstr(random(123, 125)));
+            giq_option(1, NAME, 108, NodePokerDoCompare, -1);
         end
         else begin
-            gsay_reply(1360, pr_npc(other_npc(arg0)) + message_str(1360, random(126, 128)));
+            Reply(pr_npc(other_npc(arg0)) + mstr(random(126, 128)));
             if (arg0 == dealer) then begin
-                gsay_option(1360, 108, player_barter_NPC1, -1);
+                giq_option(1, NAME, 108, player_barter_NPC1, -1);
             end
             else begin
-                gsay_option(1360, 108, NodeP_Barter, -1);
+                giq_option(1, NAME, 108, NodeP_Barter, -1);
             end
         end
     end
@@ -2355,29 +2355,29 @@ begin
     if (check_open(arg0) and not(get_try_open(arg0))) then begin
         call set_try_open(arg0, 1);
         if (((2 * 10 * bet) - stash) < (2 * bet / bet)) then begin
-            gsay_reply(1360, pr_npc(dealer) + message_str(1360, 136) + generate_ending(stash, 139) + message_str(1360, 137) + generate_ending(2 * 10 * bet, 139) + message_str(1360, 138));
+            Reply(pr_npc(dealer) + mstr(136) + generate_ending(stash, 139) + mstr(137) + generate_ending(2 * 10 * bet, 139) + mstr(138));
         end
         else begin
-            gsay_reply(1360, pr_npc(arg0) + message_str(1360, 129));
+            Reply(pr_npc(arg0) + mstr(129));
         end
         if (arg0 == dealer) then begin
-            gsay_option(1360, 108, NodeP_DealerWantsOpen, -1);
-            gsay_option(1360, 107, NodeP_BarterDealer, -1);
+            giq_option(1, NAME, 108, NodeP_DealerWantsOpen, -1);
+            giq_option(1, NAME, 107, NodeP_BarterDealer, -1);
         end
         else begin
-            gsay_option(1360, 108, NodeP_NPC1WantsOpen, -1);
-            gsay_option(1360, 107, NodeP_BarterNPC1, -1);
+            giq_option(1, NAME, 108, NodeP_NPC1WantsOpen, -1);
+            giq_option(1, NAME, 107, NodeP_BarterNPC1, -1);
         end
     end
     else begin
         if (not(NPCBarter(arg0))) then begin
             call set_pass(arg0);
-            gsay_reply(1360, pr_npc(arg0) + message_str(1360, 114));
+            Reply(pr_npc(arg0) + mstr(114));
             if (get_pass(other_npc(arg0))) then begin
-                gsay_option(1360, 108, NodeP_NPCPass, -1);
+                giq_option(1, NAME, 108, NodeP_NPCPass, -1);
             end
             else begin
-                gsay_option(1360, 108, NodeP_Barter, -1);
+                giq_option(1, NAME, 108, NodeP_Barter, -1);
             end
         end
         else begin
@@ -2400,20 +2400,20 @@ begin
             else begin
                 call set_bet(arg0, generate_bet(arg0));
             end
-            if (know_player_cards(arg0) and (get_critter_stat(dude_obj, 1) > get_critter_stat(arg0, 1))) then begin
-                gsay_reply(1360, pr_npc(arg0) + message_str(1360, 233) + generate_ending(get_bet(arg0), 139) + message_str(1360, 234) + message_str(1360, 336));
+            if (know_player_cards(arg0) and (dude_perception > get_critter_stat(arg0, STAT_pe))) then begin
+                Reply(pr_npc(arg0) + mstr(233) + generate_ending(get_bet(arg0), 139) + mstr(234) + mstr(336));
             end
             else begin
-                gsay_reply(1360, pr_npc(arg0) + message_str(1360, 233) + generate_ending(get_bet(arg0), 139) + message_str(1360, 234));
+                Reply(pr_npc(arg0) + mstr(233) + generate_ending(get_bet(arg0), 139) + mstr(234));
             end
             if (bet > LVar4) then begin
                 call set_bet(arg0, LVar4);
             end
             call mod_cash(arg0, -get_bet(arg0));
             if (get_cash(dude_obj) >= get_bet(arg0)) then begin
-                gsay_option(1360, 115, NodeP_PlayerAccept, -1);
+                giq_option(1, NAME, 115, NodeP_PlayerAccept, -1);
             end
-            gsay_option(1360, 324, NodeP_PlayerPass, -1);
+            giq_option(1, NAME, 324, NodeP_PlayerPass, -1);
         end
     end
 end
@@ -2423,21 +2423,21 @@ begin
     num_changes := 0;
     call NPCThink(arg0);
     if (num_changes > 1) then begin
-        gsay_reply(1360, pr_npc(arg0) + message_str(1360, 224) + num_changes + message_str(1360, 225));
+        Reply(pr_npc(arg0) + mstr(224) + num_changes + mstr(225));
     end
     else begin
         if (num_changes == 1) then begin
-            gsay_reply(1360, pr_npc(arg0) + message_str(1360, 223));
+            Reply(pr_npc(arg0) + mstr(223));
         end
         else begin
-            gsay_reply(1360, pr_npc(arg0) + message_str(1360, 226));
+            Reply(pr_npc(arg0) + mstr(226));
         end
     end
     if (arg0 == dealer) then begin
-        gsay_option(1360, 108, NodeP_NPC1ChangeCards, -1);
+        giq_option(1, NAME, 108, NodeP_NPC1ChangeCards, -1);
     end
     else begin
-        gsay_option(1360, 108, NodeP_Barter, -1);
+        giq_option(1, NAME, 108, NodeP_Barter, -1);
     end
 end
 
@@ -2508,86 +2508,86 @@ begin
         arg0 := 1;
     end
     if ((LVar3 == 1) and ((LVar4 > 1) or (LVar4 == 0))) then begin
-        return arg0 + message_str(1360, arg1);
+        return arg0 + mstr(arg1);
     end
     else begin
         if ((LVar3 >= 2) and (LVar3 <= 4) and ((LVar4 > 1) or (LVar4 == 0))) then begin
-            return arg0 + message_str(1360, arg1 + 1);
+            return arg0 + mstr(arg1 + 1);
         end
         else begin
-            return arg0 + message_str(1360, arg1 + 2);
+            return arg0 + mstr(arg1 + 2);
         end
     end
 end
 
 procedure pl_gender
 begin
-    if (get_critter_stat(dude_obj, 34) == 0) then begin
-        return message_str(1360, random(200, 201));
+    if (dude_gender == 0) then begin
+        return mstr(random(200, 201));
     end
     else begin
-        return message_str(1360, random(203, 204));
+        return mstr(random(203, 204));
     end
 end
 
 procedure damage_p_proc
 begin
     if (source_obj == dude_obj) then begin
-        set_global_var(204, -1);
+        set_global_var(GVAR_KLAMATH_REP_ALIANCE, -1);
     end
 end
 
 procedure destroy_p_proc
 begin
     if (source_obj == dude_obj) then begin
-        set_global_var(5, global_var(5) + 1);
-        if (metarule(51, self_obj) == 2) then begin
+        set_global_var(GVAR_GOOD_MONSTER, global_var(GVAR_GOOD_MONSTER) + 1);
+        if (critter_kill_type(self_obj) == KILL_TYPE_children_kills) then begin
             if (source_obj == dude_obj) then begin
-                set_global_var(2, 0);
-                set_global_var(1, global_var(1) + 1);
-                set_global_var(0, global_var(0) - -15);
-                set_global_var(37, 0);
-                set_global_var(38, 0);
-                set_global_var(39, 0);
-                set_global_var(40, 0);
-                set_global_var(41, 0);
-                set_global_var(42, 0);
-                set_global_var(43, 0);
-                set_global_var(44, 0);
-                set_global_var(45, 0);
-                if (global_var(0) >= 1000) then begin
-                    set_global_var(37, 1);
+                set_global_var(GVAR_CHAMPION_REPUTATION, 0);
+                set_global_var(GVAR_CHILDKILLER_REPUTATION, global_var(GVAR_CHILDKILLER_REPUTATION) + 1);
+                set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) - -15);
+                set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+                set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+                set_global_var(GVAR_KARMA_DEFENDER, 0);
+                set_global_var(GVAR_KARMA_WANDERER, 0);
+                set_global_var(GVAR_KARMA_BETRAYER, 0);
+                set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+                set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+                set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+                    set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
                 end
                 else begin
-                    if (global_var(0) >= 750) then begin
-                        set_global_var(38, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
                     end
                     else begin
-                        if (global_var(0) >= 500) then begin
-                            set_global_var(39, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                            set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                         end
                         else begin
-                            if (global_var(0) >= 250) then begin
-                                set_global_var(40, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                                set_global_var(GVAR_KARMA_DEFENDER, 1);
                             end
                             else begin
-                                if (global_var(0) > -250) then begin
-                                    set_global_var(41, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                                    set_global_var(GVAR_KARMA_WANDERER, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -500) then begin
-                                        set_global_var(42, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                        set_global_var(GVAR_KARMA_BETRAYER, 1);
                                     end
                                     else begin
-                                        if (global_var(0) > -750) then begin
-                                            set_global_var(43, 1);
+                                        if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                            set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                         end
                                         else begin
-                                            if (global_var(0) > -1000) then begin
-                                                set_global_var(44, 1);
+                                            if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                                set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                             end
                                             else begin
-                                                set_global_var(45, 1);
+                                                set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                             end
                                         end
                                     end
@@ -2598,58 +2598,58 @@ begin
                 end
             end
         end
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(4) > (3 * global_var(5))) or (global_var(2) == 1)) and (global_var(1) == 0)) then begin
-            set_global_var(2, 1);
-            set_global_var(3, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(4) > (3 * global_var(GVAR_GOOD_MONSTER))) or (global_var(GVAR_CHAMPION_REPUTATION) == 1)) and (global_var(GVAR_CHILDKILLER_REPUTATION) == 0)) then begin
+            set_global_var(GVAR_CHAMPION_REPUTATION, 1);
+            set_global_var(GVAR_BERSERKER_REPUTATION, 0);
         end
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(5) > (2 * global_var(4))) or (global_var(3) == 1))) then begin
-            set_global_var(3, 1);
-            set_global_var(2, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
+            set_global_var(GVAR_BERSERKER_REPUTATION, 1);
+            set_global_var(GVAR_CHAMPION_REPUTATION, 0);
         end
-        set_global_var(0, global_var(0) + -10);
+        set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + -10);
         debug_msg("Player gains " + -10 + " Karma Points.");
-        set_global_var(37, 0);
-        set_global_var(38, 0);
-        set_global_var(39, 0);
-        set_global_var(40, 0);
-        set_global_var(41, 0);
-        set_global_var(42, 0);
-        set_global_var(43, 0);
-        set_global_var(44, 0);
-        set_global_var(45, 0);
-        if (global_var(0) >= 1000) then begin
-            set_global_var(37, 1);
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+        set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+        set_global_var(GVAR_KARMA_DEFENDER, 0);
+        set_global_var(GVAR_KARMA_WANDERER, 0);
+        set_global_var(GVAR_KARMA_BETRAYER, 0);
+        set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+        set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+            set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
         end
         else begin
-            if (global_var(0) >= 750) then begin
-                set_global_var(38, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
             end
             else begin
-                if (global_var(0) >= 500) then begin
-                    set_global_var(39, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                 end
                 else begin
-                    if (global_var(0) >= 250) then begin
-                        set_global_var(40, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                        set_global_var(GVAR_KARMA_DEFENDER, 1);
                     end
                     else begin
-                        if (global_var(0) > -250) then begin
-                            set_global_var(41, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                            set_global_var(GVAR_KARMA_WANDERER, 1);
                         end
                         else begin
-                            if (global_var(0) > -500) then begin
-                                set_global_var(42, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                set_global_var(GVAR_KARMA_BETRAYER, 1);
                             end
                             else begin
-                                if (global_var(0) > -750) then begin
-                                    set_global_var(43, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -1000) then begin
-                                        set_global_var(44, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                     end
                                     else begin
-                                        set_global_var(45, 1);
+                                        set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                     end
                                 end
                             end
@@ -2659,9 +2659,9 @@ begin
             end
         end
     end
-    set_global_var(204, -1);
-    set_global_var(0, global_var(0) - 5);
-    set_global_var(0, global_var(48) - 2);
+    set_global_var(GVAR_KLAMATH_REP_ALIANCE, -1);
+    set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) - 5);
+    set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_TOWN_REP_KLAMATH) - 2);
     item_caps_adjust(self_obj, get_cash1(self_obj));
     if (NPC1 != 0) then begin
         item_caps_adjust(NPC1, get_cash1(NPC1));
@@ -2671,8 +2671,8 @@ end
 
 procedure critter_p_proc
 begin
-    if ((global_var(204) == -1) and obj_can_see_obj(dude_obj, self_obj)) then begin
-        attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+    if ((global_var(GVAR_KLAMATH_REP_ALIANCE) == -1) and obj_can_see_obj(dude_obj, self_obj)) then begin
+        attack(dude_obj);
     end
     else begin
         if (obj_can_see_obj(self_obj, dude_obj) and not(say_once)) then begin
@@ -2688,10 +2688,10 @@ begin
     if ((fixed_param == 1) and not(combat_is_initialized)) then begin
         LVar0 := random(143, 145);
         if (LVar0 == 143) then begin
-            float_msg(self_obj, message_str(1360, 143) + pl_gender() + message_str(1360, 161), 8);
+            float_msg(self_obj, mstr(143) + pl_gender() + mstr(161), FLOAT_MSG_YELLOW);
         end
         else begin
-            float_msg(self_obj, message_str(1360, LVar0), 8);
+            float_msg(self_obj, mstr(LVar0), FLOAT_MSG_YELLOW);
         end
     end
     else begin
@@ -2699,34 +2699,34 @@ begin
             if (get_total_win() < 200) then begin
                 LVar0 := random(143, 145);
                 if (LVar0 == 143) then begin
-                    float_msg(self_obj, message_str(1360, 143) + pl_gender() + message_str(1360, 161), 8);
+                    float_msg(self_obj, mstr(143) + pl_gender() + mstr(161), FLOAT_MSG_YELLOW);
                 end
                 else begin
-                    float_msg(self_obj, message_str(1360, LVar0), 8);
+                    float_msg(self_obj, mstr(LVar0), FLOAT_MSG_YELLOW);
                 end
             end
             else begin
                 LVar0 := random(146, 148);
                 if (LVar0 == 146) then begin
-                    if (get_critter_stat(dude_obj, 34) == 0) then begin
-                        float_msg(self_obj, message_str(1360, 146) + ".", 8);
+                    if (dude_gender == 0) then begin
+                        float_msg(self_obj, mstr(146) + ".", FLOAT_MSG_YELLOW);
                     end
                     else begin
-                        float_msg(self_obj, message_str(1360, 146) + " .", 0);
+                        float_msg(self_obj, mstr(146) + " .", FLOAT_MSG_NORMAL);
                     end
                 end
                 else begin
-                    float_msg(self_obj, message_str(1360, LVar0), 8);
+                    float_msg(self_obj, mstr(LVar0), FLOAT_MSG_YELLOW);
                 end
             end
         end
         else begin
             if (fixed_param == 7) then begin
                 if (get_total_win() < 200) then begin
-                    float_msg(NPC1, message_str(1360, random(149, 151)), 8);
+                    float_msg(NPC1, mstr(random(149, 151)), FLOAT_MSG_YELLOW);
                 end
                 else begin
-                    float_msg(NPC1, message_str(1360, random(152, 154)), 0);
+                    float_msg(NPC1, mstr(random(152, 154)), FLOAT_MSG_NORMAL);
                 end
             end
             else begin
@@ -2748,19 +2748,18 @@ begin
     LVar1 := (LVar0 / 3) + (LVar0 % 3);
     LVar2 := LVar1 / 7;
     if (LVar1 < 7) then begin
-        return message_str(1360, 158);
+        return mstr(158);
     end
     if (LVar2 == 1) then begin
-        return message_str(1360, 155);
+        return mstr(155);
     end
     if ((LVar2 >= 2) and (LVar2 <= 4)) then begin
-        return message_str(1360, 156);
+        return mstr(156);
     end
-    return message_str(1360, 157);
+    return mstr(157);
 end
 
 procedure NodeP_Again
 begin
     call Start_Game_Dialog(1);
 end
-
