diff --git a/Mods/MegaMod/Original/PRIMIT.ssl b/Mods/MegaMod/ReDefine/PRIMIT.ssl
index 6b4344c..45f878c 100644
--- a/Mods/MegaMod/Original/PRIMIT.ssl
+++ b/Mods/MegaMod/ReDefine/PRIMIT.ssl
@@ -143,28 +143,28 @@ begin
     end
     if (critter_inven_obj(self_obj, 0) != arg0) then begin
         if (critter_inven_obj(self_obj, 0) == 0) then begin
-            metarule3(107, self_obj, 16777216 + arg1, 0);
+            art_change_fid_num(self_obj, 16777216 + arg1);
         end
         if ((obj_pid(critter_inven_obj(self_obj, 0)) == 1) or (obj_pid(critter_inven_obj(self_obj, 0)) == 379)) then begin
-            metarule3(107, self_obj, 16777228, 0);
+            art_change_fid_num(self_obj, 16777228);
         end
         if ((obj_pid(critter_inven_obj(self_obj, 0)) == 232) or (obj_pid(critter_inven_obj(self_obj, 0)) == 3)) then begin
-            metarule3(107, self_obj, 16777217, 0);
+            art_change_fid_num(self_obj, 16777217);
         end
         if ((obj_pid(critter_inven_obj(self_obj, 0)) == 349) or (obj_pid(critter_inven_obj(self_obj, 0)) == 348)) then begin
-            metarule3(107, self_obj, 16777287, 0);
+            art_change_fid_num(self_obj, 16777287);
         end
         if ((obj_pid(critter_inven_obj(self_obj, 0)) == 240) or (obj_pid(critter_inven_obj(self_obj, 0)) == 380) or (obj_pid(critter_inven_obj(self_obj, 0)) == 2)) then begin
-            metarule3(107, self_obj, 16777230, 0);
+            art_change_fid_num(self_obj, 16777230);
         end
         if ((obj_pid(critter_inven_obj(self_obj, 0)) == 74) or (obj_pid(critter_inven_obj(self_obj, 0)) == 265)) then begin
-            metarule3(107, self_obj, 16777229, 0);
+            art_change_fid_num(self_obj, 16777229);
         end
         if ((obj_pid(critter_inven_obj(self_obj, 0)) == 239) or (obj_pid(critter_inven_obj(self_obj, 0)) == 17) or (obj_pid(critter_inven_obj(self_obj, 0)) == 381)) then begin
-            metarule3(107, self_obj, 16777226, 0);
+            art_change_fid_num(self_obj, 16777226);
         end
         if ((obj_pid(critter_inven_obj(self_obj, 0)) == 524) or (obj_pid(critter_inven_obj(self_obj, 0)) == 113)) then begin
-            metarule3(107, self_obj, 16777218, 0);
+            art_change_fid_num(self_obj, 16777218);
         end
         LVar2 := critter_inven_obj(self_obj, 1);
         debug_msg("armour pid == " + obj_pid(LVar2));
@@ -189,140 +189,140 @@ end
 
 procedure map_enter_p_proc
 begin
-    if ((global_var(1107) == 2) and (local_var(6) != 1)) then begin
-        anim(create_object_sid(16777220, 22493, 0, 1321), 1000, 3);
+    if ((global_var(GVAR_NAVARRO_APA_QUEST) == 2) and (local_var(6) != 1)) then begin
+        anim(create_object_sid(16777220, 22493, 0, SCRIPT_JCDOGS), 1000, 3);
         set_map_var(6, 1);
     end
-    debug_msg("Check_Create_Car(" + 29116 + "," + elevation(dude_obj) + ")");
-    if (not(metarule(22, 0))) then begin
-        if (global_var(18) != 0) then begin
-            debug_msg("car_current_town == " + metarule(30, 0) + " / map_get_load_area == " + metarule3(111, 0, 0, 0));
-            if ((metarule(30, 0) == metarule3(111, 0, 0, 0)) or ((26 <= metarule(30, 0)) and (metarule(30, 0) <= 29) and (metarule3(111, 0, 0, 0) == -1))) then begin
-                debug_msg("Place_Car(" + 29116 + "," + elevation(dude_obj) + ")");
-                debug_msg("   GVAR_CAR_PLACED_TILE == " + global_var(633));
-                debug_msg("   Car_At_Loc == " + (tile_contains_pid_obj(29116, elevation(dude_obj), 33555441) != 0));
-                if ((global_var(633) <= 0) or (29116 == global_var(633))) then begin
-                    set_global_var(633, 29116);
-                    if (not(tile_contains_pid_obj(29116, elevation(dude_obj), 33555441) != 0)) then begin
-                        Scenery_Creation := create_object_sid(33555441, 29116, elevation(dude_obj), 304);
+    debug_msg("Check_Create_Car(" + 29116 + "," + dude_elevation + ")");
+    if (not(is_loading_game)) then begin
+        if (global_var(GVAR_PLAYER_GOT_CAR) != 0) then begin
+            debug_msg("car_current_town == " + car_current_town + " / map_get_load_area == " + map_get_load_area);
+            if ((car_current_town == map_get_load_area) or ((26 <= car_current_town) and (car_current_town <= 29) and (map_get_load_area == -1))) then begin
+                debug_msg("Place_Car(" + 29116 + "," + dude_elevation + ")");
+                debug_msg("   GVAR_CAR_PLACED_TILE == " + global_var(GVAR_CAR_PLACED_TILE));
+                debug_msg("   Car_At_Loc == " + (tile_contains_pid_obj(29116, dude_elevation, 33555441) != 0));
+                if ((global_var(GVAR_CAR_PLACED_TILE) <= 0) or (29116 == global_var(GVAR_CAR_PLACED_TILE))) then begin
+                    set_global_var(GVAR_CAR_PLACED_TILE, 29116);
+                    if (not(tile_contains_pid_obj(29116, dude_elevation, 33555441) != 0)) then begin
+                        Scenery_Creation := create_object_sid(33555441, 29116, dude_elevation, SCRIPT_ZSDRVCAR);
                         Scenery_Creation_Hex := 29116;
                         Scenery_Creation_Count := 0;
                         while (Scenery_Creation_Count < 2) do begin
                             Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 1, 1);
-                            Scenery_Creation := create_object_sid(33554499, Scenery_Creation_Hex, elevation(dude_obj), -1);
+                            Scenery_Creation := create_object(33554499, Scenery_Creation_Hex, dude_elevation);
                             Scenery_Creation_Count := Scenery_Creation_Count + 1;
                         end
                         Scenery_Creation_Count := 0;
                         while (Scenery_Creation_Count < 2) do begin
                             Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 2, 1);
-                            Scenery_Creation := create_object_sid(33554499, Scenery_Creation_Hex, elevation(dude_obj), -1);
+                            Scenery_Creation := create_object(33554499, Scenery_Creation_Hex, dude_elevation);
                             Scenery_Creation_Count := Scenery_Creation_Count + 1;
                         end
                         Scenery_Creation_Count := 0;
                         while (Scenery_Creation_Count < 2) do begin
                             Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 3, 1);
-                            Scenery_Creation := create_object_sid(33554499, Scenery_Creation_Hex, elevation(dude_obj), -1);
+                            Scenery_Creation := create_object(33554499, Scenery_Creation_Hex, dude_elevation);
                             Scenery_Creation_Count := Scenery_Creation_Count + 1;
                         end
                         Scenery_Creation_Count := 0;
                         while (Scenery_Creation_Count < 2) do begin
                             Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 4, 1);
-                            Scenery_Creation := create_object_sid(33554499, Scenery_Creation_Hex, elevation(dude_obj), -1);
+                            Scenery_Creation := create_object(33554499, Scenery_Creation_Hex, dude_elevation);
                             Scenery_Creation_Count := Scenery_Creation_Count + 1;
                         end
                         Scenery_Creation_Count := 0;
                         while (Scenery_Creation_Count < 1) do begin
                             Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 5, 1);
-                            Scenery_Creation := create_object_sid(33554499, Scenery_Creation_Hex, elevation(dude_obj), -1);
+                            Scenery_Creation := create_object(33554499, Scenery_Creation_Hex, dude_elevation);
                             Scenery_Creation_Count := Scenery_Creation_Count + 1;
                         end
                         Scenery_Creation_Count := 0;
                         while (Scenery_Creation_Count < 1) do begin
                             Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 4, 1);
-                            Scenery_Creation := create_object_sid(33554499, Scenery_Creation_Hex, elevation(dude_obj), -1);
+                            Scenery_Creation := create_object(33554499, Scenery_Creation_Hex, dude_elevation);
                             Scenery_Creation_Count := Scenery_Creation_Count + 1;
                         end
                         Scenery_Creation_Count := 0;
                         while (Scenery_Creation_Count < 1) do begin
                             Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 5, 1);
-                            Scenery_Creation := create_object_sid(33554499, Scenery_Creation_Hex, elevation(dude_obj), -1);
+                            Scenery_Creation := create_object(33554499, Scenery_Creation_Hex, dude_elevation);
                             Scenery_Creation_Count := Scenery_Creation_Count + 1;
                         end
                         Scenery_Creation_Count := 0;
                         while (Scenery_Creation_Count < 1) do begin
                             Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 4, 1);
-                            Scenery_Creation := create_object_sid(33554499, Scenery_Creation_Hex, elevation(dude_obj), -1);
+                            Scenery_Creation := create_object(33554499, Scenery_Creation_Hex, dude_elevation);
                             Scenery_Creation_Count := Scenery_Creation_Count + 1;
                         end
                         Scenery_Creation_Count := 0;
                         while (Scenery_Creation_Count < 1) do begin
                             Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 5, 1);
-                            Scenery_Creation := create_object_sid(33554499, Scenery_Creation_Hex, elevation(dude_obj), -1);
+                            Scenery_Creation := create_object(33554499, Scenery_Creation_Hex, dude_elevation);
                             Scenery_Creation_Count := Scenery_Creation_Count + 1;
                         end
                         Scenery_Creation_Count := 0;
                         while (Scenery_Creation_Count < 1) do begin
                             Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 4, 1);
-                            Scenery_Creation := create_object_sid(33554499, Scenery_Creation_Hex, elevation(dude_obj), -1);
+                            Scenery_Creation := create_object(33554499, Scenery_Creation_Hex, dude_elevation);
                             Scenery_Creation_Count := Scenery_Creation_Count + 1;
                         end
                         Scenery_Creation_Count := 0;
                         while (Scenery_Creation_Count < 1) do begin
                             Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 0, 1);
-                            Scenery_Creation := create_object_sid(33554499, Scenery_Creation_Hex, elevation(dude_obj), -1);
+                            Scenery_Creation := create_object(33554499, Scenery_Creation_Hex, dude_elevation);
                             Scenery_Creation_Count := Scenery_Creation_Count + 1;
                         end
                         Scenery_Creation_Count := 0;
                         while (Scenery_Creation_Count < 1) do begin
                             Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 5, 1);
-                            Scenery_Creation := create_object_sid(33554499, Scenery_Creation_Hex, elevation(dude_obj), -1);
+                            Scenery_Creation := create_object(33554499, Scenery_Creation_Hex, dude_elevation);
                             Scenery_Creation_Count := Scenery_Creation_Count + 1;
                         end
                         Scenery_Creation_Count := 0;
                         while (Scenery_Creation_Count < 2) do begin
                             Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 0, 1);
-                            Scenery_Creation := create_object_sid(33554499, Scenery_Creation_Hex, elevation(dude_obj), -1);
+                            Scenery_Creation := create_object(33554499, Scenery_Creation_Hex, dude_elevation);
                             Scenery_Creation_Count := Scenery_Creation_Count + 1;
                         end
                         Scenery_Creation_Count := 0;
                         while (Scenery_Creation_Count < 1) do begin
                             Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 1, 1);
-                            Scenery_Creation := create_object_sid(33554499, Scenery_Creation_Hex, elevation(dude_obj), -1);
+                            Scenery_Creation := create_object(33554499, Scenery_Creation_Hex, dude_elevation);
                             Scenery_Creation_Count := Scenery_Creation_Count + 1;
                         end
                         Scenery_Creation_Count := 0;
                         while (Scenery_Creation_Count < 1) do begin
                             Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 2, 1);
-                            Scenery_Creation := create_object_sid(33554499, Scenery_Creation_Hex, elevation(dude_obj), -1);
+                            Scenery_Creation := create_object(33554499, Scenery_Creation_Hex, dude_elevation);
                             Scenery_Creation_Count := Scenery_Creation_Count + 1;
                         end
                     end
                 end
-                if (tile_contains_pid_obj(29116, elevation(dude_obj), 33555441) != 0) then begin
-                    if (not(tile_contains_pid_obj(tile_num_in_direction(tile_num_in_direction(29116, 5, 2), 4, 1), elevation(dude_obj), 455) != 0)) then begin
+                if (tile_contains_pid_obj(29116, dude_elevation, 33555441) != 0) then begin
+                    if (not(tile_contains_pid_obj(tile_num_in_direction(tile_num_in_direction(29116, 5, 2), 4, 1), dude_elevation, 455) != 0)) then begin
                         debug_msg("placing trunk:");
                         Scenery_Creation_Ptr := 0;
                         if (party_member_obj(455) != 0) then begin
                             Scenery_Creation_Ptr := party_member_obj(455);
                         end
                         else begin
-                            if (global_var(18) == 0) then begin
-                                Scenery_Creation_Ptr := create_object_sid(455, 0, 0, 920);
+                            if (global_var(GVAR_PLAYER_GOT_CAR) == 0) then begin
+                                Scenery_Creation_Ptr := create_object_sid(455, 0, 0, SCRIPT_ZICRTRNK);
                             end
                         end
                         if (Scenery_Creation_Ptr != 0) then begin
                             Scenery_Creation_Hex := tile_num_in_direction(tile_num_in_direction(29116, 5, 2), 4, 1);
-                            move_to(Scenery_Creation_Ptr, Scenery_Creation_Hex, elevation(dude_obj));
+                            move_to(Scenery_Creation_Ptr, Scenery_Creation_Hex, dude_elevation);
                             Scenery_Creation_Count := 0;
                             while (Scenery_Creation_Count < 1) do begin
                                 Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 1, 1);
-                                Scenery_Creation := create_object_sid(33554499, Scenery_Creation_Hex, elevation(dude_obj), -1);
+                                Scenery_Creation := create_object(33554499, Scenery_Creation_Hex, dude_elevation);
                                 Scenery_Creation_Count := Scenery_Creation_Count + 1;
                             end
                             Scenery_Creation_Count := 0;
                             while (Scenery_Creation_Count < 1) do begin
                                 Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 2, 1);
-                                Scenery_Creation := create_object_sid(33554499, Scenery_Creation_Hex, elevation(dude_obj), -1);
+                                Scenery_Creation := create_object(33554499, Scenery_Creation_Hex, dude_elevation);
                                 Scenery_Creation_Count := Scenery_Creation_Count + 1;
                             end
                         end
@@ -333,133 +333,133 @@ begin
                 end
             end
             else begin
-                if (tile_contains_pid_obj(29116, elevation(dude_obj), 33555441) != 0) then begin
-                    debug_msg("Dest_Car_Loc(" + 29116 + "," + elevation(dude_obj) + ")");
-                    Scenery_Creation_Ptr := tile_contains_pid_obj(29116, elevation(dude_obj), 33555441);
+                if (tile_contains_pid_obj(29116, dude_elevation, 33555441) != 0) then begin
+                    debug_msg("Dest_Car_Loc(" + 29116 + "," + dude_elevation + ")");
+                    Scenery_Creation_Ptr := tile_contains_pid_obj(29116, dude_elevation, 33555441);
                     if (Scenery_Creation_Ptr != 0) then begin
                         destroy_object(Scenery_Creation_Ptr);
                         Scenery_Creation_Hex := 29116;
                         Scenery_Creation_Count := 0;
                         while (Scenery_Creation_Count < 2) do begin
                             Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 1, 1);
-                            Scenery_Creation_Ptr := tile_contains_pid_obj(Scenery_Creation_Hex, elevation(dude_obj), 33554499);
+                            Scenery_Creation_Ptr := tile_contains_pid_obj(Scenery_Creation_Hex, dude_elevation, 33554499);
                             destroy_object(Scenery_Creation_Ptr);
                             Scenery_Creation_Count := Scenery_Creation_Count + 1;
                         end
                         Scenery_Creation_Count := 0;
                         while (Scenery_Creation_Count < 2) do begin
                             Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 2, 1);
-                            Scenery_Creation_Ptr := tile_contains_pid_obj(Scenery_Creation_Hex, elevation(dude_obj), 33554499);
+                            Scenery_Creation_Ptr := tile_contains_pid_obj(Scenery_Creation_Hex, dude_elevation, 33554499);
                             destroy_object(Scenery_Creation_Ptr);
                             Scenery_Creation_Count := Scenery_Creation_Count + 1;
                         end
                         Scenery_Creation_Count := 0;
                         while (Scenery_Creation_Count < 2) do begin
                             Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 3, 1);
-                            Scenery_Creation_Ptr := tile_contains_pid_obj(Scenery_Creation_Hex, elevation(dude_obj), 33554499);
+                            Scenery_Creation_Ptr := tile_contains_pid_obj(Scenery_Creation_Hex, dude_elevation, 33554499);
                             destroy_object(Scenery_Creation_Ptr);
                             Scenery_Creation_Count := Scenery_Creation_Count + 1;
                         end
                         Scenery_Creation_Count := 0;
                         while (Scenery_Creation_Count < 2) do begin
                             Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 4, 1);
-                            Scenery_Creation_Ptr := tile_contains_pid_obj(Scenery_Creation_Hex, elevation(dude_obj), 33554499);
+                            Scenery_Creation_Ptr := tile_contains_pid_obj(Scenery_Creation_Hex, dude_elevation, 33554499);
                             destroy_object(Scenery_Creation_Ptr);
                             Scenery_Creation_Count := Scenery_Creation_Count + 1;
                         end
                         Scenery_Creation_Count := 0;
                         while (Scenery_Creation_Count < 1) do begin
                             Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 5, 1);
-                            Scenery_Creation_Ptr := tile_contains_pid_obj(Scenery_Creation_Hex, elevation(dude_obj), 33554499);
+                            Scenery_Creation_Ptr := tile_contains_pid_obj(Scenery_Creation_Hex, dude_elevation, 33554499);
                             destroy_object(Scenery_Creation_Ptr);
                             Scenery_Creation_Count := Scenery_Creation_Count + 1;
                         end
                         Scenery_Creation_Count := 0;
                         while (Scenery_Creation_Count < 1) do begin
                             Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 4, 1);
-                            Scenery_Creation_Ptr := tile_contains_pid_obj(Scenery_Creation_Hex, elevation(dude_obj), 33554499);
+                            Scenery_Creation_Ptr := tile_contains_pid_obj(Scenery_Creation_Hex, dude_elevation, 33554499);
                             destroy_object(Scenery_Creation_Ptr);
                             Scenery_Creation_Count := Scenery_Creation_Count + 1;
                         end
                         Scenery_Creation_Count := 0;
                         while (Scenery_Creation_Count < 1) do begin
                             Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 5, 1);
-                            Scenery_Creation_Ptr := tile_contains_pid_obj(Scenery_Creation_Hex, elevation(dude_obj), 33554499);
+                            Scenery_Creation_Ptr := tile_contains_pid_obj(Scenery_Creation_Hex, dude_elevation, 33554499);
                             destroy_object(Scenery_Creation_Ptr);
                             Scenery_Creation_Count := Scenery_Creation_Count + 1;
                         end
                         Scenery_Creation_Count := 0;
                         while (Scenery_Creation_Count < 1) do begin
                             Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 4, 1);
-                            Scenery_Creation_Ptr := tile_contains_pid_obj(Scenery_Creation_Hex, elevation(dude_obj), 33554499);
+                            Scenery_Creation_Ptr := tile_contains_pid_obj(Scenery_Creation_Hex, dude_elevation, 33554499);
                             destroy_object(Scenery_Creation_Ptr);
                             Scenery_Creation_Count := Scenery_Creation_Count + 1;
                         end
                         Scenery_Creation_Count := 0;
                         while (Scenery_Creation_Count < 1) do begin
                             Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 5, 1);
-                            Scenery_Creation_Ptr := tile_contains_pid_obj(Scenery_Creation_Hex, elevation(dude_obj), 33554499);
+                            Scenery_Creation_Ptr := tile_contains_pid_obj(Scenery_Creation_Hex, dude_elevation, 33554499);
                             destroy_object(Scenery_Creation_Ptr);
                             Scenery_Creation_Count := Scenery_Creation_Count + 1;
                         end
                         Scenery_Creation_Count := 0;
                         while (Scenery_Creation_Count < 1) do begin
                             Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 4, 1);
-                            Scenery_Creation_Ptr := tile_contains_pid_obj(Scenery_Creation_Hex, elevation(dude_obj), 33554499);
+                            Scenery_Creation_Ptr := tile_contains_pid_obj(Scenery_Creation_Hex, dude_elevation, 33554499);
                             destroy_object(Scenery_Creation_Ptr);
                             Scenery_Creation_Count := Scenery_Creation_Count + 1;
                         end
                         Scenery_Creation_Count := 0;
                         while (Scenery_Creation_Count < 1) do begin
                             Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 0, 1);
-                            Scenery_Creation_Ptr := tile_contains_pid_obj(Scenery_Creation_Hex, elevation(dude_obj), 33554499);
+                            Scenery_Creation_Ptr := tile_contains_pid_obj(Scenery_Creation_Hex, dude_elevation, 33554499);
                             destroy_object(Scenery_Creation_Ptr);
                             Scenery_Creation_Count := Scenery_Creation_Count + 1;
                         end
                         Scenery_Creation_Count := 0;
                         while (Scenery_Creation_Count < 1) do begin
                             Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 5, 1);
-                            Scenery_Creation_Ptr := tile_contains_pid_obj(Scenery_Creation_Hex, elevation(dude_obj), 33554499);
+                            Scenery_Creation_Ptr := tile_contains_pid_obj(Scenery_Creation_Hex, dude_elevation, 33554499);
                             destroy_object(Scenery_Creation_Ptr);
                             Scenery_Creation_Count := Scenery_Creation_Count + 1;
                         end
                         Scenery_Creation_Count := 0;
                         while (Scenery_Creation_Count < 2) do begin
                             Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 0, 1);
-                            Scenery_Creation_Ptr := tile_contains_pid_obj(Scenery_Creation_Hex, elevation(dude_obj), 33554499);
+                            Scenery_Creation_Ptr := tile_contains_pid_obj(Scenery_Creation_Hex, dude_elevation, 33554499);
                             destroy_object(Scenery_Creation_Ptr);
                             Scenery_Creation_Count := Scenery_Creation_Count + 1;
                         end
                         Scenery_Creation_Count := 0;
                         while (Scenery_Creation_Count < 1) do begin
                             Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 1, 1);
-                            Scenery_Creation_Ptr := tile_contains_pid_obj(Scenery_Creation_Hex, elevation(dude_obj), 33554499);
+                            Scenery_Creation_Ptr := tile_contains_pid_obj(Scenery_Creation_Hex, dude_elevation, 33554499);
                             destroy_object(Scenery_Creation_Ptr);
                             Scenery_Creation_Count := Scenery_Creation_Count + 1;
                         end
                         Scenery_Creation_Count := 0;
                         while (Scenery_Creation_Count < 1) do begin
                             Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 2, 1);
-                            Scenery_Creation_Ptr := tile_contains_pid_obj(Scenery_Creation_Hex, elevation(dude_obj), 33554499);
+                            Scenery_Creation_Ptr := tile_contains_pid_obj(Scenery_Creation_Hex, dude_elevation, 33554499);
                             destroy_object(Scenery_Creation_Ptr);
                             Scenery_Creation_Count := Scenery_Creation_Count + 1;
                         end
                         Scenery_Creation_Count := 0;
                         while (Scenery_Creation_Count < 1) do begin
                             Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 1, 1);
-                            Scenery_Creation_Ptr := tile_contains_pid_obj(Scenery_Creation_Hex, elevation(dude_obj), 33554499);
+                            Scenery_Creation_Ptr := tile_contains_pid_obj(Scenery_Creation_Hex, dude_elevation, 33554499);
                             destroy_object(Scenery_Creation_Ptr);
                             Scenery_Creation_Count := Scenery_Creation_Count + 1;
                         end
                         Scenery_Creation_Count := 0;
                         while (Scenery_Creation_Count < 1) do begin
                             Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 2, 1);
-                            Scenery_Creation_Ptr := tile_contains_pid_obj(Scenery_Creation_Hex, elevation(dude_obj), 33554499);
+                            Scenery_Creation_Ptr := tile_contains_pid_obj(Scenery_Creation_Hex, dude_elevation, 33554499);
                             destroy_object(Scenery_Creation_Ptr);
                             Scenery_Creation_Count := Scenery_Creation_Count + 1;
                         end
                     end
-                    Scenery_Creation_Ptr := tile_contains_pid_obj(tile_num_in_direction(tile_num_in_direction(29116, 5, 2), 4, 1), elevation(dude_obj), 455);
+                    Scenery_Creation_Ptr := tile_contains_pid_obj(tile_num_in_direction(tile_num_in_direction(29116, 5, 2), 4, 1), dude_elevation, 455);
                     if (Scenery_Creation_Ptr != 0) then begin
                         if (Scenery_Creation_Ptr == party_member_obj(455)) then begin
                             set_obj_visibility(Scenery_Creation_Ptr, 1);
@@ -635,4 +635,3 @@ begin
         end
     end
 end
-
