diff --git a/Mods/MegaMod/Original/QCCURLNG.ssl b/Mods/MegaMod/ReDefine/QCCURLNG.ssl
index 477992b..b20baa9 100644
--- a/Mods/MegaMod/Original/QCCURLNG.ssl
+++ b/Mods/MegaMod/ReDefine/QCCURLNG.ssl
@@ -208,7 +208,7 @@ procedure critter_p_proc
 begin
     if ((local_var(5) == 2) and obj_can_see_obj(self_obj, dude_obj)) then begin
         set_local_var(5, 1);
-        attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+        attack(dude_obj);
     end
 end
 
@@ -224,48 +224,48 @@ begin
     Evil_Critter := 0;
     Slavery_Tolerant := 2;
     Karma_Perception := 10;
-    set_global_var(37, 0);
-    set_global_var(38, 0);
-    set_global_var(39, 0);
-    set_global_var(40, 0);
-    set_global_var(41, 0);
-    set_global_var(42, 0);
-    set_global_var(43, 0);
-    set_global_var(44, 0);
-    set_global_var(45, 0);
-    if (global_var(0) >= 1000) then begin
-        set_global_var(37, 1);
+    set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+    set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+    set_global_var(GVAR_KARMA_DEFENDER, 0);
+    set_global_var(GVAR_KARMA_WANDERER, 0);
+    set_global_var(GVAR_KARMA_BETRAYER, 0);
+    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+    set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+    set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+    if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
     end
     else begin
-        if (global_var(0) >= 750) then begin
-            set_global_var(38, 1);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+            set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
         end
         else begin
-            if (global_var(0) >= 500) then begin
-                set_global_var(39, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
             end
             else begin
-                if (global_var(0) >= 250) then begin
-                    set_global_var(40, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                    set_global_var(GVAR_KARMA_DEFENDER, 1);
                 end
                 else begin
-                    if (global_var(0) > -250) then begin
-                        set_global_var(41, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                        set_global_var(GVAR_KARMA_WANDERER, 1);
                     end
                     else begin
-                        if (global_var(0) > -500) then begin
-                            set_global_var(42, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                            set_global_var(GVAR_KARMA_BETRAYER, 1);
                         end
                         else begin
-                            if (global_var(0) > -750) then begin
-                                set_global_var(43, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                             end
                             else begin
-                                if (global_var(0) > -1000) then begin
-                                    set_global_var(44, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                    set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                 end
                                 else begin
-                                    set_global_var(45, 1);
+                                    set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                 end
                             end
                         end
@@ -276,7 +276,7 @@ begin
     end
     if (local_var(2) == 0) then begin
         set_local_var(2, 1);
-        set_local_var(3, (get_critter_stat(dude_obj, 3) - get_critter_stat(self_obj, 3)) * 5);
+        set_local_var(3, (dude_charisma - self_charisma) * 5);
         debug_msg("Base Reaction == " + local_var(3));
     end
     Static_Reaction := 0;
@@ -288,14 +288,14 @@ begin
     else begin
         reaction_bonus_karma := -1 * (-1 * reaction_bonus_karma / 1000);
     end
-    if (global_var(62) >= 0) then begin
-        reaction_bonus_town_rep := global_var(62) / 2;
+    if (global_var(GVAR_TOWN_REP_ENCLAVE) >= 0) then begin
+        reaction_bonus_town_rep := global_var(GVAR_TOWN_REP_ENCLAVE) / 2;
     end
     else begin
-        reaction_bonus_town_rep := -1 * (-1 * global_var(62) / 2);
+        reaction_bonus_town_rep := -1 * (-1 * global_var(GVAR_TOWN_REP_ENCLAVE) / 2);
     end
-    if (has_trait(2, dude_obj, 13)) then begin
-        if (get_critter_stat(dude_obj, 34) == get_critter_stat(self_obj, 34)) then begin
+    if (has_trait(TRAIT_TRAIT, dude_obj, TRAIT_sex_appeal)) then begin
+        if (dude_gender == self_gender) then begin
             Static_Reaction := Static_Reaction + -20;
         end
         else begin
@@ -311,48 +311,48 @@ begin
         debug_msg("Karma Reaction Bonus == " + Static_Reaction);
         Static_Reaction := Static_Reaction - reaction_bonus_town_rep;
         debug_msg("Town Rep Bonus == " + Static_Reaction);
-        set_global_var(37, 0);
-        set_global_var(38, 0);
-        set_global_var(39, 0);
-        set_global_var(40, 0);
-        set_global_var(41, 0);
-        set_global_var(42, 0);
-        set_global_var(43, 0);
-        set_global_var(44, 0);
-        set_global_var(45, 0);
-        if (global_var(0) >= 1000) then begin
-            set_global_var(37, 1);
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+        set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+        set_global_var(GVAR_KARMA_DEFENDER, 0);
+        set_global_var(GVAR_KARMA_WANDERER, 0);
+        set_global_var(GVAR_KARMA_BETRAYER, 0);
+        set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+        set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+            set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
         end
         else begin
-            if (global_var(0) >= 750) then begin
-                set_global_var(38, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
             end
             else begin
-                if (global_var(0) >= 500) then begin
-                    set_global_var(39, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                 end
                 else begin
-                    if (global_var(0) >= 250) then begin
-                        set_global_var(40, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                        set_global_var(GVAR_KARMA_DEFENDER, 1);
                     end
                     else begin
-                        if (global_var(0) > -250) then begin
-                            set_global_var(41, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                            set_global_var(GVAR_KARMA_WANDERER, 1);
                         end
                         else begin
-                            if (global_var(0) > -500) then begin
-                                set_global_var(42, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                set_global_var(GVAR_KARMA_BETRAYER, 1);
                             end
                             else begin
-                                if (global_var(0) > -750) then begin
-                                    set_global_var(43, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -1000) then begin
-                                        set_global_var(44, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                     end
                                     else begin
-                                        set_global_var(45, 1);
+                                        set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                     end
                                 end
                             end
@@ -361,35 +361,35 @@ begin
                 end
             end
         end
-        if (global_var(37) == 1) then begin
+        if (global_var(GVAR_KARMA_HOLY_WARRIOR) == 1) then begin
             Static_Reaction := Static_Reaction - 20;
         end
         else begin
-            if (global_var(38) == 1) then begin
+            if (global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES) == 1) then begin
                 Static_Reaction := Static_Reaction - 15;
             end
             else begin
-                if (global_var(39) == 1) then begin
+                if (global_var(GVAR_KARMA_SHIELD_OF_HOPE) == 1) then begin
                     Static_Reaction := Static_Reaction - 10;
                 end
                 else begin
-                    if (global_var(40) == 1) then begin
+                    if (global_var(GVAR_KARMA_DEFENDER) == 1) then begin
                         Static_Reaction := Static_Reaction - 5;
                     end
                     else begin
-                        if (global_var(41) == 1) then begin
+                        if (global_var(GVAR_KARMA_WANDERER) == 1) then begin
                             Static_Reaction := Static_Reaction - 0;
                         end
                         else begin
-                            if (global_var(42) == 1) then begin
+                            if (global_var(GVAR_KARMA_BETRAYER) == 1) then begin
                                 Static_Reaction := Static_Reaction - -5;
                             end
                             else begin
-                                if (global_var(43) == 1) then begin
+                                if (global_var(GVAR_KARMA_SWORD_OF_DESPAIR) == 1) then begin
                                     Static_Reaction := Static_Reaction - -10;
                                 end
                                 else begin
-                                    if (global_var(44) == 1) then begin
+                                    if (global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE) == 1) then begin
                                         Static_Reaction := Static_Reaction - -15;
                                     end
                                     else begin
@@ -409,48 +409,48 @@ begin
         debug_msg("Karma Reaction Bonus == " + Static_Reaction);
         Static_Reaction := Static_Reaction + reaction_bonus_town_rep;
         debug_msg("Town Rep Bonus == " + Static_Reaction);
-        set_global_var(37, 0);
-        set_global_var(38, 0);
-        set_global_var(39, 0);
-        set_global_var(40, 0);
-        set_global_var(41, 0);
-        set_global_var(42, 0);
-        set_global_var(43, 0);
-        set_global_var(44, 0);
-        set_global_var(45, 0);
-        if (global_var(0) >= 1000) then begin
-            set_global_var(37, 1);
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+        set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+        set_global_var(GVAR_KARMA_DEFENDER, 0);
+        set_global_var(GVAR_KARMA_WANDERER, 0);
+        set_global_var(GVAR_KARMA_BETRAYER, 0);
+        set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+        set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+            set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
         end
         else begin
-            if (global_var(0) >= 750) then begin
-                set_global_var(38, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
             end
             else begin
-                if (global_var(0) >= 500) then begin
-                    set_global_var(39, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                 end
                 else begin
-                    if (global_var(0) >= 250) then begin
-                        set_global_var(40, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                        set_global_var(GVAR_KARMA_DEFENDER, 1);
                     end
                     else begin
-                        if (global_var(0) > -250) then begin
-                            set_global_var(41, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                            set_global_var(GVAR_KARMA_WANDERER, 1);
                         end
                         else begin
-                            if (global_var(0) > -500) then begin
-                                set_global_var(42, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                set_global_var(GVAR_KARMA_BETRAYER, 1);
                             end
                             else begin
-                                if (global_var(0) > -750) then begin
-                                    set_global_var(43, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -1000) then begin
-                                        set_global_var(44, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                     end
                                     else begin
-                                        set_global_var(45, 1);
+                                        set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                     end
                                 end
                             end
@@ -459,35 +459,35 @@ begin
                 end
             end
         end
-        if (global_var(37) == 1) then begin
+        if (global_var(GVAR_KARMA_HOLY_WARRIOR) == 1) then begin
             Static_Reaction := Static_Reaction + 20;
         end
         else begin
-            if (global_var(38) == 1) then begin
+            if (global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES) == 1) then begin
                 Static_Reaction := Static_Reaction + 15;
             end
             else begin
-                if (global_var(39) == 1) then begin
+                if (global_var(GVAR_KARMA_SHIELD_OF_HOPE) == 1) then begin
                     Static_Reaction := Static_Reaction + 10;
                 end
                 else begin
-                    if (global_var(40) == 1) then begin
+                    if (global_var(GVAR_KARMA_DEFENDER) == 1) then begin
                         Static_Reaction := Static_Reaction + 5;
                     end
                     else begin
-                        if (global_var(41) == 1) then begin
+                        if (global_var(GVAR_KARMA_WANDERER) == 1) then begin
                             Static_Reaction := Static_Reaction + 0;
                         end
                         else begin
-                            if (global_var(42) == 1) then begin
+                            if (global_var(GVAR_KARMA_BETRAYER) == 1) then begin
                                 Static_Reaction := Static_Reaction + -5;
                             end
                             else begin
-                                if (global_var(43) == 1) then begin
+                                if (global_var(GVAR_KARMA_SWORD_OF_DESPAIR) == 1) then begin
                                     Static_Reaction := Static_Reaction + -10;
                                 end
                                 else begin
-                                    if (global_var(44) == 1) then begin
+                                    if (global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE) == 1) then begin
                                         Static_Reaction := Static_Reaction + -15;
                                     end
                                     else begin
@@ -503,28 +503,28 @@ begin
         debug_msg("Good Person Reaction Bonus == " + Static_Reaction);
     end
     if (Slavery_Tolerant == 0) then begin
-        if (global_var(11) == 1) then begin
+        if (global_var(GVAR_REPUTATION_SLAVER) == 1) then begin
             Static_Reaction := Static_Reaction + (2 * -25);
         end
     end
     else begin
         if (Slavery_Tolerant == 1) then begin
-            if (global_var(11) == 1) then begin
+            if (global_var(GVAR_REPUTATION_SLAVER) == 1) then begin
                 Static_Reaction := Static_Reaction + -25;
             end
         end
         else begin
             if (Slavery_Tolerant == 3) then begin
-                if (global_var(11) == 1) then begin
+                if (global_var(GVAR_REPUTATION_SLAVER) == 1) then begin
                     Static_Reaction := Static_Reaction - -25;
                 end
             end
         end
     end
     debug_msg("Slaver + Aligned Reaction == " + Static_Reaction);
-    Static_Reaction := Static_Reaction + (10 * has_trait(0, dude_obj, 10));
+    Static_Reaction := Static_Reaction + (10 * has_trait(TRAIT_PERK, dude_obj, PERK_presence));
     debug_msg("Presence Reaction == " + Static_Reaction);
-    if (global_var(1) >= 2) then begin
+    if (global_var(GVAR_CHILDKILLER_REPUTATION) >= 2) then begin
         Static_Reaction := Static_Reaction + -30;
     end
     debug_msg("Childkiller Reaction == " + Static_Reaction);
@@ -534,15 +534,15 @@ begin
         call Node046();
     end
     else begin
-        if (global_var(433) != 0) then begin
+        if (global_var(GVAR_ENCLAVE_ALARM) != 0) then begin
             call Node004();
         end
         else begin
-            if ((obj_pid(critter_inven_obj(dude_obj, 0)) == 3) or (obj_pid(critter_inven_obj(dude_obj, 0)) == 348) or (obj_pid(critter_inven_obj(dude_obj, 0)) == 349) or (obj_pid(critter_inven_obj(dude_obj, 0)) == 232) and (get_critter_stat(dude_obj, 4) < 4) and (metarule(16, 0) < 2)) then begin
+            if ((obj_pid(critter_inven_obj(dude_obj, 0)) == 3) or (obj_pid(critter_inven_obj(dude_obj, 0)) == 348) or (obj_pid(critter_inven_obj(dude_obj, 0)) == 349) or (obj_pid(critter_inven_obj(dude_obj, 0)) == 232) and (dude_iq < 4) and (party_member_count(0) < 2)) then begin
                 call Node002();
             end
             else begin
-                if (get_critter_stat(dude_obj, 4) < 4) then begin
+                if (dude_iq < 4) then begin
                     call Node001();
                 end
                 else begin
@@ -569,62 +569,62 @@ end
 procedure destroy_p_proc
 begin
     variable LVar0 := 0;
-    LVar0 := create_object_sid(1, 0, 0, -1);
+    LVar0 := create_object(1, 0, 0);
     add_obj_to_inven(self_obj, LVar0);
     if (source_obj == dude_obj) then begin
-        set_global_var(5, global_var(5) + 1);
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(4) > (3 * global_var(5))) or (global_var(2) == 1)) and (global_var(1) == 0)) then begin
-            set_global_var(2, 1);
-            set_global_var(3, 0);
+        set_global_var(GVAR_GOOD_MONSTER, global_var(GVAR_GOOD_MONSTER) + 1);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(4) > (3 * global_var(GVAR_GOOD_MONSTER))) or (global_var(GVAR_CHAMPION_REPUTATION) == 1)) and (global_var(GVAR_CHILDKILLER_REPUTATION) == 0)) then begin
+            set_global_var(GVAR_CHAMPION_REPUTATION, 1);
+            set_global_var(GVAR_BERSERKER_REPUTATION, 0);
         end
-        if (metarule(51, self_obj) == 2) then begin
+        if (critter_kill_type(self_obj) == KILL_TYPE_children_kills) then begin
             if (source_obj == dude_obj) then begin
-                set_global_var(2, 0);
-                set_global_var(1, global_var(1) + 1);
-                set_global_var(0, global_var(0) + -15);
+                set_global_var(GVAR_CHAMPION_REPUTATION, 0);
+                set_global_var(GVAR_CHILDKILLER_REPUTATION, global_var(GVAR_CHILDKILLER_REPUTATION) + 1);
+                set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + -15);
                 debug_msg("Player gains " + -15 + " Karma Points.");
-                set_global_var(37, 0);
-                set_global_var(38, 0);
-                set_global_var(39, 0);
-                set_global_var(40, 0);
-                set_global_var(41, 0);
-                set_global_var(42, 0);
-                set_global_var(43, 0);
-                set_global_var(44, 0);
-                set_global_var(45, 0);
-                if (global_var(0) >= 1000) then begin
-                    set_global_var(37, 1);
+                set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+                set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+                set_global_var(GVAR_KARMA_DEFENDER, 0);
+                set_global_var(GVAR_KARMA_WANDERER, 0);
+                set_global_var(GVAR_KARMA_BETRAYER, 0);
+                set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+                set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+                set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+                    set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
                 end
                 else begin
-                    if (global_var(0) >= 750) then begin
-                        set_global_var(38, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
                     end
                     else begin
-                        if (global_var(0) >= 500) then begin
-                            set_global_var(39, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                            set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                         end
                         else begin
-                            if (global_var(0) >= 250) then begin
-                                set_global_var(40, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                                set_global_var(GVAR_KARMA_DEFENDER, 1);
                             end
                             else begin
-                                if (global_var(0) > -250) then begin
-                                    set_global_var(41, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                                    set_global_var(GVAR_KARMA_WANDERER, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -500) then begin
-                                        set_global_var(42, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                        set_global_var(GVAR_KARMA_BETRAYER, 1);
                                     end
                                     else begin
-                                        if (global_var(0) > -750) then begin
-                                            set_global_var(43, 1);
+                                        if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                            set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                         end
                                         else begin
-                                            if (global_var(0) > -1000) then begin
-                                                set_global_var(44, 1);
+                                            if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                                set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                             end
                                             else begin
-                                                set_global_var(45, 1);
+                                                set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                             end
                                         end
                                     end
@@ -633,58 +633,58 @@ begin
                         end
                     end
                 end
-                set_global_var(62, global_var(62) + -8);
+                set_global_var(GVAR_TOWN_REP_ENCLAVE, global_var(GVAR_TOWN_REP_ENCLAVE) + -8);
                 debug_msg("Added " + -8 + " to Town Rep");
             end
         end
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(5) > (2 * global_var(4))) or (global_var(3) == 1))) then begin
-            set_global_var(3, 1);
-            set_global_var(2, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
+            set_global_var(GVAR_BERSERKER_REPUTATION, 1);
+            set_global_var(GVAR_CHAMPION_REPUTATION, 0);
         end
-        set_global_var(0, global_var(0) + -10);
+        set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + -10);
         debug_msg("Player gains " + -10 + " Karma Points.");
-        set_global_var(37, 0);
-        set_global_var(38, 0);
-        set_global_var(39, 0);
-        set_global_var(40, 0);
-        set_global_var(41, 0);
-        set_global_var(42, 0);
-        set_global_var(43, 0);
-        set_global_var(44, 0);
-        set_global_var(45, 0);
-        if (global_var(0) >= 1000) then begin
-            set_global_var(37, 1);
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+        set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+        set_global_var(GVAR_KARMA_DEFENDER, 0);
+        set_global_var(GVAR_KARMA_WANDERER, 0);
+        set_global_var(GVAR_KARMA_BETRAYER, 0);
+        set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+        set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+            set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
         end
         else begin
-            if (global_var(0) >= 750) then begin
-                set_global_var(38, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
             end
             else begin
-                if (global_var(0) >= 500) then begin
-                    set_global_var(39, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                 end
                 else begin
-                    if (global_var(0) >= 250) then begin
-                        set_global_var(40, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                        set_global_var(GVAR_KARMA_DEFENDER, 1);
                     end
                     else begin
-                        if (global_var(0) > -250) then begin
-                            set_global_var(41, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                            set_global_var(GVAR_KARMA_WANDERER, 1);
                         end
                         else begin
-                            if (global_var(0) > -500) then begin
-                                set_global_var(42, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                set_global_var(GVAR_KARMA_BETRAYER, 1);
                             end
                             else begin
-                                if (global_var(0) > -750) then begin
-                                    set_global_var(43, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -1000) then begin
-                                        set_global_var(44, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                     end
                                     else begin
-                                        set_global_var(45, 1);
+                                        set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                     end
                                 end
                             end
@@ -693,7 +693,7 @@ begin
                 end
             end
         end
-        set_global_var(62, global_var(62) + -5);
+        set_global_var(GVAR_TOWN_REP_ENCLAVE, global_var(GVAR_TOWN_REP_ENCLAVE) + -5);
         debug_msg("Added " + -5 + " to Town Rep");
     end
     return 0;
@@ -705,17 +705,17 @@ begin
     script_overrides;
     if (local_var(7) == 0) then begin
         set_local_var(7, local_var(7) + 1);
-        display_msg(message_str(866, 100));
+        display_mstr(100);
     end
     else begin
-        display_msg(message_str(866, 101));
+        display_mstr(101);
     end
 end
 
 procedure description_p_proc
 begin
     script_overrides;
-    display_msg(message_str(866, 102));
+    display_mstr(102);
 end
 
 procedure use_skill_on_p_proc
@@ -732,8 +732,8 @@ end
 procedure map_enter_p_proc
 begin
     Only_Once := 0;
-    critter_add_trait(self_obj, 1, 6, 192);
-    critter_add_trait(self_obj, 1, 5, 19);
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 192);
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_DOCTOR);
 end
 
 procedure Node998
@@ -747,338 +747,338 @@ end
 
 procedure Node001
 begin
-    float_msg(self_obj, message_str(866, random(103, 109)), 8);
+    float_msg(self_obj, mstr(random(103, 109)), FLOAT_MSG_YELLOW);
 end
 
 procedure Node002
 begin
-    float_msg(self_obj, message_str(866, random(110, 114)), 8);
+    float_msg(self_obj, mstr(random(110, 114)), FLOAT_MSG_YELLOW);
 end
 
 procedure Node003
 begin
-    gsay_reply(866, 115);
-    giq_option(4, 866, message_str(866, 116) + obj_name(dude_obj) + message_str(866, 117), Node005, 51);
-    giq_option(4, 866, 118, Node999, 50);
+    Reply(115);
+    giq_option(4, NAME, mstr(116) + dude_name + mstr(117), Node005, BAD_REACTION);
+    giq_option(4, NAME, 118, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node004
 begin
-    float_msg(self_obj, message_str(866, random(119, 126)), 8);
+    float_msg(self_obj, mstr(random(119, 126)), FLOAT_MSG_YELLOW);
 end
 
 procedure Node005
 begin
-    gsay_reply(866, message_str(866, 127) + obj_name(dude_obj) + message_str(866, 128));
-    giq_option(4, 866, 129, Node007, 50);
-    giq_option(4, 866, 130, Node008, 51);
-    giq_option(4, 866, 131, Node999, 50);
+    Reply(mstr(127) + dude_name + mstr(128));
+    giq_option(4, NAME, 129, Node007, NEUTRAL_REACTION);
+    giq_option(4, NAME, 130, Node008, BAD_REACTION);
+    giq_option(4, NAME, 131, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node006
 begin
-    gsay_reply(866, 132);
-    giq_option(4, 866, 133, Node008, 51);
-    giq_option(4, 866, message_str(866, 134) + obj_name(dude_obj) + message_str(866, 135), Node005, 50);
-    giq_option(4, 866, 136, Node007, 50);
+    Reply(132);
+    giq_option(4, NAME, 133, Node008, BAD_REACTION);
+    giq_option(4, NAME, mstr(134) + dude_name + mstr(135), Node005, NEUTRAL_REACTION);
+    giq_option(4, NAME, 136, Node007, NEUTRAL_REACTION);
 end
 
 procedure Node007
 begin
-    gsay_reply(866, 137);
-    giq_option(4, 866, 138, Node009, 50);
-    giq_option(4, 866, message_str(866, 139) + obj_name(dude_obj) + message_str(866, 140), Node008, 50);
-    giq_option(4, 866, 141, Node008, 50);
-    giq_option(4, 866, 142, Node999, 50);
+    Reply(137);
+    giq_option(4, NAME, 138, Node009, NEUTRAL_REACTION);
+    giq_option(4, NAME, mstr(139) + dude_name + mstr(140), Node008, NEUTRAL_REACTION);
+    giq_option(4, NAME, 141, Node008, NEUTRAL_REACTION);
+    giq_option(4, NAME, 142, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node008
 begin
-    gsay_reply(866, 143);
-    giq_option(4, 866, 144, Node011, 50);
-    giq_option(4, 866, 145, Node010, 51);
+    Reply(143);
+    giq_option(4, NAME, 144, Node011, NEUTRAL_REACTION);
+    giq_option(4, NAME, 145, Node010, BAD_REACTION);
 end
 
 procedure Node009
 begin
-    gsay_reply(866, 146);
-    giq_option(4, 866, 147, Node013, 50);
-    giq_option(4, 866, 148, Node008, 51);
+    Reply(146);
+    giq_option(4, NAME, 147, Node013, NEUTRAL_REACTION);
+    giq_option(4, NAME, 148, Node008, BAD_REACTION);
 end
 
 procedure Node010
 begin
-    gsay_message(866, random(149, 152), 51);
+    gsay_message(NAME, random(149, 152), BAD_REACTION);
     call Node998();
 end
 
 procedure Node011
 begin
-    gsay_reply(866, 153);
-    giq_option(7, 866, 154, Node013, 50);
-    giq_option(4, 866, 155, Node013, 50);
+    Reply(153);
+    giq_option(7, NAME, 154, Node013, NEUTRAL_REACTION);
+    giq_option(4, NAME, 155, Node013, NEUTRAL_REACTION);
 end
 
 procedure Node012
 begin
-    gsay_reply(866, 156);
-    giq_option(7, 866, 157, Node013, 50);
-    giq_option(4, 866, 158, Node013, 50);
+    Reply(156);
+    giq_option(7, NAME, 157, Node013, NEUTRAL_REACTION);
+    giq_option(4, NAME, 158, Node013, NEUTRAL_REACTION);
 end
 
 procedure Node013
 begin
     if (local_var(9) == 0) then begin
         set_local_var(9, local_var(9) + 1);
-        gsay_reply(866, 159);
+        Reply(159);
     end
     else begin
-        gsay_reply(866, 160);
+        Reply(160);
     end
-    giq_option(6, 866, 161, Node014, 50);
-    giq_option(4, 866, 162, Node024, 50);
-    giq_option(4, 866, 163, Node017, 50);
-    giq_option(4, 866, 164, Node008, 50);
+    giq_option(6, NAME, 161, Node014, NEUTRAL_REACTION);
+    giq_option(4, NAME, 162, Node024, NEUTRAL_REACTION);
+    giq_option(4, NAME, 163, Node017, NEUTRAL_REACTION);
+    giq_option(4, NAME, 164, Node008, NEUTRAL_REACTION);
 end
 
 procedure Node014
 begin
-    gsay_reply(866, 165);
-    giq_option(4, 866, 166, Node015, 50);
-    giq_option(4, 866, 167, Node013, 50);
+    Reply(165);
+    giq_option(4, NAME, 166, Node015, NEUTRAL_REACTION);
+    giq_option(4, NAME, 167, Node013, NEUTRAL_REACTION);
 end
 
 procedure Node015
 begin
-    gsay_reply(866, 168);
-    giq_option(4, 866, 169, Node016, 50);
+    Reply(168);
+    giq_option(4, NAME, 169, Node016, NEUTRAL_REACTION);
 end
 
 procedure Node016
 begin
-    gsay_reply(866, 170);
-    giq_option(4, 866, 171, Node017, 50);
+    Reply(170);
+    giq_option(4, NAME, 171, Node017, NEUTRAL_REACTION);
 end
 
 procedure Node017
 begin
-    gsay_reply(866, 172);
-    giq_option(4, 866, 173, Node018, 50);
-    giq_option(4, 866, 174, Node013, 50);
+    Reply(172);
+    giq_option(4, NAME, 173, Node018, NEUTRAL_REACTION);
+    giq_option(4, NAME, 174, Node013, NEUTRAL_REACTION);
 end
 
 procedure Node018
 begin
-    gsay_reply(866, 175);
-    giq_option(4, 866, 176, Node019, 50);
+    Reply(175);
+    giq_option(4, NAME, 176, Node019, NEUTRAL_REACTION);
 end
 
 procedure Node019
 begin
-    gsay_reply(866, 177);
-    giq_option(4, 866, 178, Node020, 51);
-    giq_option(4, 866, 179, Node025, 50);
+    Reply(177);
+    giq_option(4, NAME, 178, Node020, BAD_REACTION);
+    giq_option(4, NAME, 179, Node025, NEUTRAL_REACTION);
 end
 
 procedure Node020
 begin
-    gsay_reply(866, 180);
-    giq_option(7, 866, 181, Node020a, 50);
-    giq_option(4, 866, 182, Node020b, 50);
-    giq_option(4, 866, 183, Node010, 50);
+    Reply(180);
+    giq_option(7, NAME, 181, Node020a, NEUTRAL_REACTION);
+    giq_option(4, NAME, 182, Node020b, NEUTRAL_REACTION);
+    giq_option(4, NAME, 183, Node010, NEUTRAL_REACTION);
 end
 
 procedure Node021
 begin
-    gsay_reply(866, 184);
-    giq_option(4, 866, 185, Node036, 49);
-    giq_option(4, 866, 186, Node010, 51);
+    Reply(184);
+    giq_option(4, NAME, 185, Node036, GOOD_REACTION);
+    giq_option(4, NAME, 186, Node010, BAD_REACTION);
 end
 
 procedure Node022
 begin
-    gsay_reply(866, 187);
-    giq_option(7, 866, 188, Node022a, 50);
-    giq_option(4, 866, 189, Node022a, 50);
-    giq_option(4, 866, 190, Node010, 51);
+    Reply(187);
+    giq_option(7, NAME, 188, Node022a, NEUTRAL_REACTION);
+    giq_option(4, NAME, 189, Node022a, NEUTRAL_REACTION);
+    giq_option(4, NAME, 190, Node010, BAD_REACTION);
 end
 
 procedure Node023
 begin
-    gsay_reply(866, 191);
-    giq_option(4, 866, 192, Node010, 51);
-    giq_option(4, 866, 193, Node999, 51);
+    Reply(191);
+    giq_option(4, NAME, 192, Node010, BAD_REACTION);
+    giq_option(4, NAME, 193, Node999, BAD_REACTION);
 end
 
 procedure Node024
 begin
-    gsay_reply(866, 194);
-    giq_option(4, 866, 195, Node026, 50);
-    giq_option(4, 866, 196, Node013, 50);
+    Reply(194);
+    giq_option(4, NAME, 195, Node026, NEUTRAL_REACTION);
+    giq_option(4, NAME, 196, Node013, NEUTRAL_REACTION);
 end
 
 procedure Node025
 begin
-    gsay_reply(866, 197);
-    giq_option(4, 866, 198, Node026, 50);
-    giq_option(4, 866, 199, Node013, 50);
+    Reply(197);
+    giq_option(4, NAME, 198, Node026, NEUTRAL_REACTION);
+    giq_option(4, NAME, 199, Node013, NEUTRAL_REACTION);
 end
 
 procedure Node026
 begin
-    gsay_reply(866, 200);
-    giq_option(4, 866, 201, Node027, 50);
+    Reply(200);
+    giq_option(4, NAME, 201, Node027, NEUTRAL_REACTION);
 end
 
 procedure Node027
 begin
-    gsay_reply(866, 202);
-    giq_option(4, 866, 203, Node028, 50);
+    Reply(202);
+    giq_option(4, NAME, 203, Node028, NEUTRAL_REACTION);
 end
 
 procedure Node028
 begin
-    gsay_reply(866, 204);
-    giq_option(4, 866, 205, Node029, 50);
+    Reply(204);
+    giq_option(4, NAME, 205, Node029, NEUTRAL_REACTION);
 end
 
 procedure Node029
 begin
-    gsay_reply(866, 206);
-    giq_option(4, 866, 207, Node030, 50);
+    Reply(206);
+    giq_option(4, NAME, 207, Node030, NEUTRAL_REACTION);
 end
 
 procedure Node030
 begin
-    gsay_reply(866, 208);
-    giq_option(4, 866, 209, Node031, 50);
-    giq_option(4, 866, 210, Node013, 50);
+    Reply(208);
+    giq_option(4, NAME, 209, Node031, NEUTRAL_REACTION);
+    giq_option(4, NAME, 210, Node013, NEUTRAL_REACTION);
 end
 
 procedure Node031
 begin
-    gsay_reply(866, 211);
-    giq_option(4, 866, 212, Node032, 50);
-    giq_option(4, 866, 213, Node010, 51);
+    Reply(211);
+    giq_option(4, NAME, 212, Node032, NEUTRAL_REACTION);
+    giq_option(4, NAME, 213, Node010, BAD_REACTION);
 end
 
 procedure Node032
 begin
-    gsay_reply(866, 214);
-    giq_option(4, 866, 215, Node033, 50);
+    Reply(214);
+    giq_option(4, NAME, 215, Node033, NEUTRAL_REACTION);
 end
 
 procedure Node033
 begin
-    gsay_reply(866, 216);
-    giq_option(4, 866, 217, Node034, 50);
-    giq_option(4, 866, 218, Node999, 51);
-    giq_option(4, 866, 219, Node010, 51);
+    Reply(216);
+    giq_option(4, NAME, 217, Node034, NEUTRAL_REACTION);
+    giq_option(4, NAME, 218, Node999, BAD_REACTION);
+    giq_option(4, NAME, 219, Node010, BAD_REACTION);
 end
 
 procedure Node034
 begin
-    gsay_reply(866, 220);
-    giq_option(4, 866, 221, Node020b, 50);
-    giq_option(4, 866, 222, Node010, 50);
+    Reply(220);
+    giq_option(4, NAME, 221, Node020b, NEUTRAL_REACTION);
+    giq_option(4, NAME, 222, Node010, NEUTRAL_REACTION);
 end
 
 procedure Node035
 begin
-    gsay_message(866, random(223, 227), 50);
+    gsay_message(NAME, random(223, 227), NEUTRAL_REACTION);
 end
 
 procedure Node036
 begin
     gfade_out(600);
     gfade_in(600);
-    gsay_reply(866, 228);
-    giq_option(4, 866, 229, Node037, 50);
+    Reply(228);
+    giq_option(4, NAME, 229, Node037, NEUTRAL_REACTION);
 end
 
 procedure Node037
 begin
-    gsay_reply(866, 230);
-    giq_option(4, 866, 231, Node038, 50);
+    Reply(230);
+    giq_option(4, NAME, 231, Node038, NEUTRAL_REACTION);
 end
 
 procedure Node038
 begin
-    gsay_reply(866, 232);
-    giq_option(4, 866, 233, Node039, 50);
+    Reply(232);
+    giq_option(4, NAME, 233, Node039, NEUTRAL_REACTION);
 end
 
 procedure Node039
 begin
-    gsay_reply(866, 234);
-    giq_option(4, 866, 235, Node040, 50);
+    Reply(234);
+    giq_option(4, NAME, 235, Node040, NEUTRAL_REACTION);
 end
 
 procedure Node040
 begin
-    gsay_reply(866, 236);
-    giq_option(4, 866, 237, Node041, 50);
+    Reply(236);
+    giq_option(4, NAME, 237, Node041, NEUTRAL_REACTION);
 end
 
 procedure Node041
 begin
-    gsay_reply(866, 238);
-    giq_option(4, 866, 239, Node042, 50);
-    giq_option(4, 866, 240, Node047, 51);
+    Reply(238);
+    giq_option(4, NAME, 239, Node042, NEUTRAL_REACTION);
+    giq_option(4, NAME, 240, Node047, BAD_REACTION);
 end
 
 procedure Node042
 begin
-    gsay_reply(866, 241);
-    giq_option(4, 866, 242, Node043, 50);
+    Reply(241);
+    giq_option(4, NAME, 242, Node043, NEUTRAL_REACTION);
 end
 
 procedure Node043
 begin
-    gsay_reply(866, 243);
-    giq_option(4, 866, 244, Node044, 50);
-    giq_option(4, 866, 245, Node044, 51);
+    Reply(243);
+    giq_option(4, NAME, 244, Node044, NEUTRAL_REACTION);
+    giq_option(4, NAME, 245, Node044, BAD_REACTION);
 end
 
 procedure Node044
 begin
-    gsay_reply(866, message_str(866, 246) + " " + message_str(866, 247) + " " + message_str(866, 248) + " " + message_str(866, 249));
-    giq_option(4, 866, 250, Node045, 50);
+    Reply(mstr(246) + " " + mstr(247) + " " + mstr(248) + " " + mstr(249));
+    giq_option(4, NAME, 250, Node045, NEUTRAL_REACTION);
 end
 
 procedure Node045
 begin
     set_local_var(8, local_var(8) + 1);
-    if (global_var(433) != 1) then begin
+    if (global_var(GVAR_ENCLAVE_ALARM) != 1) then begin
         debug_msg("Set Toxin Alarm");
-        set_global_var(433, 2);
+        set_global_var(GVAR_ENCLAVE_ALARM, 2);
         give_exp_points(5000);
-        display_msg(message_str(14, 100) + 5000 + message_str(14, 101));
+        display_msg(g_mstr(100) + 5000 + g_mstr(101));
     end
-    if (global_var(434) == 0) then begin
-        set_global_var(434, game_time);
+    if (global_var(GVAR_ENCLAVE_TIMER) == 0) then begin
+        set_global_var(GVAR_ENCLAVE_TIMER, game_time);
     end
-    gsay_reply(866, 251);
-    giq_option(4, 866, 252, Node999, 50);
-    giq_option(4, 866, 253, Node999, 50);
+    Reply(251);
+    giq_option(4, NAME, 252, Node999, NEUTRAL_REACTION);
+    giq_option(4, NAME, 253, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node046
 begin
-    float_msg(self_obj, message_str(866, random(254, 258)), 8);
+    float_msg(self_obj, mstr(random(254, 258)), FLOAT_MSG_YELLOW);
 end
 
 procedure Node047
 begin
-    gsay_reply(866, 259);
-    giq_option(4, 866, 260, Node042, 50);
-    giq_option(4, 866, 261, Node048, 50);
+    Reply(259);
+    giq_option(4, NAME, 260, Node042, NEUTRAL_REACTION);
+    giq_option(4, NAME, 261, Node048, NEUTRAL_REACTION);
 end
 
 procedure Node048
 begin
-    gsay_reply(866, 262);
-    giq_option(4, 866, 263, Node042, 50);
+    Reply(262);
+    giq_option(4, NAME, 263, Node042, NEUTRAL_REACTION);
 end
 
 procedure Node020a
@@ -1120,4 +1120,3 @@ begin
         call Node023();
     end
 end
-
