diff --git a/Mods/MegaMod/Original/QCGRANIT.ssl b/Mods/MegaMod/ReDefine/QCGRANIT.ssl
index 8b11bcc..4249158 100644
--- a/Mods/MegaMod/Original/QCGRANIT.ssl
+++ b/Mods/MegaMod/ReDefine/QCGRANIT.ssl
@@ -176,10 +176,10 @@ procedure critter_p_proc
 begin
     if ((local_var(5) == 2) and obj_can_see_obj(self_obj, dude_obj)) then begin
         set_local_var(5, 1);
-        attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+        attack(dude_obj);
     end
-    if (global_var(432) == 1) then begin
-        critter_add_trait(self_obj, 1, 6, 0);
+    if (global_var(GVAR_ENCLAVE_GRANITE_JOINED) == 1) then begin
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 0);
         if ((tile_distance(tile_num(self_obj), 19693) > 4) and (local_var(8) == 0)) then begin
             set_local_var(8, local_var(8) + 1);
             add_timer_event(self_obj, game_ticks(1), 1);
@@ -199,48 +199,48 @@ begin
     Evil_Critter := 0;
     Slavery_Tolerant := 2;
     Karma_Perception := 10;
-    set_global_var(37, 0);
-    set_global_var(38, 0);
-    set_global_var(39, 0);
-    set_global_var(40, 0);
-    set_global_var(41, 0);
-    set_global_var(42, 0);
-    set_global_var(43, 0);
-    set_global_var(44, 0);
-    set_global_var(45, 0);
-    if (global_var(0) >= 1000) then begin
-        set_global_var(37, 1);
+    set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+    set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+    set_global_var(GVAR_KARMA_DEFENDER, 0);
+    set_global_var(GVAR_KARMA_WANDERER, 0);
+    set_global_var(GVAR_KARMA_BETRAYER, 0);
+    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+    set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+    set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+    if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
     end
     else begin
-        if (global_var(0) >= 750) then begin
-            set_global_var(38, 1);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+            set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
         end
         else begin
-            if (global_var(0) >= 500) then begin
-                set_global_var(39, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
             end
             else begin
-                if (global_var(0) >= 250) then begin
-                    set_global_var(40, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                    set_global_var(GVAR_KARMA_DEFENDER, 1);
                 end
                 else begin
-                    if (global_var(0) > -250) then begin
-                        set_global_var(41, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                        set_global_var(GVAR_KARMA_WANDERER, 1);
                     end
                     else begin
-                        if (global_var(0) > -500) then begin
-                            set_global_var(42, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                            set_global_var(GVAR_KARMA_BETRAYER, 1);
                         end
                         else begin
-                            if (global_var(0) > -750) then begin
-                                set_global_var(43, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                             end
                             else begin
-                                if (global_var(0) > -1000) then begin
-                                    set_global_var(44, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                    set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                 end
                                 else begin
-                                    set_global_var(45, 1);
+                                    set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                 end
                             end
                         end
@@ -251,7 +251,7 @@ begin
     end
     if (local_var(2) == 0) then begin
         set_local_var(2, 1);
-        set_local_var(3, (get_critter_stat(dude_obj, 3) - get_critter_stat(self_obj, 3)) * 5);
+        set_local_var(3, (dude_charisma - self_charisma) * 5);
         debug_msg("Base Reaction == " + local_var(3));
     end
     Static_Reaction := 0;
@@ -263,14 +263,14 @@ begin
     else begin
         reaction_bonus_karma := -1 * (-1 * reaction_bonus_karma / 1000);
     end
-    if (global_var(62) >= 0) then begin
-        reaction_bonus_town_rep := global_var(62) / 2;
+    if (global_var(GVAR_TOWN_REP_ENCLAVE) >= 0) then begin
+        reaction_bonus_town_rep := global_var(GVAR_TOWN_REP_ENCLAVE) / 2;
     end
     else begin
-        reaction_bonus_town_rep := -1 * (-1 * global_var(62) / 2);
+        reaction_bonus_town_rep := -1 * (-1 * global_var(GVAR_TOWN_REP_ENCLAVE) / 2);
     end
-    if (has_trait(2, dude_obj, 13)) then begin
-        if (get_critter_stat(dude_obj, 34) == get_critter_stat(self_obj, 34)) then begin
+    if (has_trait(TRAIT_TRAIT, dude_obj, TRAIT_sex_appeal)) then begin
+        if (dude_gender == self_gender) then begin
             Static_Reaction := Static_Reaction + -20;
         end
         else begin
@@ -286,48 +286,48 @@ begin
         debug_msg("Karma Reaction Bonus == " + Static_Reaction);
         Static_Reaction := Static_Reaction - reaction_bonus_town_rep;
         debug_msg("Town Rep Bonus == " + Static_Reaction);
-        set_global_var(37, 0);
-        set_global_var(38, 0);
-        set_global_var(39, 0);
-        set_global_var(40, 0);
-        set_global_var(41, 0);
-        set_global_var(42, 0);
-        set_global_var(43, 0);
-        set_global_var(44, 0);
-        set_global_var(45, 0);
-        if (global_var(0) >= 1000) then begin
-            set_global_var(37, 1);
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+        set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+        set_global_var(GVAR_KARMA_DEFENDER, 0);
+        set_global_var(GVAR_KARMA_WANDERER, 0);
+        set_global_var(GVAR_KARMA_BETRAYER, 0);
+        set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+        set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+            set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
         end
         else begin
-            if (global_var(0) >= 750) then begin
-                set_global_var(38, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
             end
             else begin
-                if (global_var(0) >= 500) then begin
-                    set_global_var(39, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                 end
                 else begin
-                    if (global_var(0) >= 250) then begin
-                        set_global_var(40, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                        set_global_var(GVAR_KARMA_DEFENDER, 1);
                     end
                     else begin
-                        if (global_var(0) > -250) then begin
-                            set_global_var(41, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                            set_global_var(GVAR_KARMA_WANDERER, 1);
                         end
                         else begin
-                            if (global_var(0) > -500) then begin
-                                set_global_var(42, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                set_global_var(GVAR_KARMA_BETRAYER, 1);
                             end
                             else begin
-                                if (global_var(0) > -750) then begin
-                                    set_global_var(43, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -1000) then begin
-                                        set_global_var(44, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                     end
                                     else begin
-                                        set_global_var(45, 1);
+                                        set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                     end
                                 end
                             end
@@ -336,35 +336,35 @@ begin
                 end
             end
         end
-        if (global_var(37) == 1) then begin
+        if (global_var(GVAR_KARMA_HOLY_WARRIOR) == 1) then begin
             Static_Reaction := Static_Reaction - 20;
         end
         else begin
-            if (global_var(38) == 1) then begin
+            if (global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES) == 1) then begin
                 Static_Reaction := Static_Reaction - 15;
             end
             else begin
-                if (global_var(39) == 1) then begin
+                if (global_var(GVAR_KARMA_SHIELD_OF_HOPE) == 1) then begin
                     Static_Reaction := Static_Reaction - 10;
                 end
                 else begin
-                    if (global_var(40) == 1) then begin
+                    if (global_var(GVAR_KARMA_DEFENDER) == 1) then begin
                         Static_Reaction := Static_Reaction - 5;
                     end
                     else begin
-                        if (global_var(41) == 1) then begin
+                        if (global_var(GVAR_KARMA_WANDERER) == 1) then begin
                             Static_Reaction := Static_Reaction - 0;
                         end
                         else begin
-                            if (global_var(42) == 1) then begin
+                            if (global_var(GVAR_KARMA_BETRAYER) == 1) then begin
                                 Static_Reaction := Static_Reaction - -5;
                             end
                             else begin
-                                if (global_var(43) == 1) then begin
+                                if (global_var(GVAR_KARMA_SWORD_OF_DESPAIR) == 1) then begin
                                     Static_Reaction := Static_Reaction - -10;
                                 end
                                 else begin
-                                    if (global_var(44) == 1) then begin
+                                    if (global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE) == 1) then begin
                                         Static_Reaction := Static_Reaction - -15;
                                     end
                                     else begin
@@ -384,48 +384,48 @@ begin
         debug_msg("Karma Reaction Bonus == " + Static_Reaction);
         Static_Reaction := Static_Reaction + reaction_bonus_town_rep;
         debug_msg("Town Rep Bonus == " + Static_Reaction);
-        set_global_var(37, 0);
-        set_global_var(38, 0);
-        set_global_var(39, 0);
-        set_global_var(40, 0);
-        set_global_var(41, 0);
-        set_global_var(42, 0);
-        set_global_var(43, 0);
-        set_global_var(44, 0);
-        set_global_var(45, 0);
-        if (global_var(0) >= 1000) then begin
-            set_global_var(37, 1);
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+        set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+        set_global_var(GVAR_KARMA_DEFENDER, 0);
+        set_global_var(GVAR_KARMA_WANDERER, 0);
+        set_global_var(GVAR_KARMA_BETRAYER, 0);
+        set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+        set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+            set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
         end
         else begin
-            if (global_var(0) >= 750) then begin
-                set_global_var(38, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
             end
             else begin
-                if (global_var(0) >= 500) then begin
-                    set_global_var(39, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                 end
                 else begin
-                    if (global_var(0) >= 250) then begin
-                        set_global_var(40, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                        set_global_var(GVAR_KARMA_DEFENDER, 1);
                     end
                     else begin
-                        if (global_var(0) > -250) then begin
-                            set_global_var(41, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                            set_global_var(GVAR_KARMA_WANDERER, 1);
                         end
                         else begin
-                            if (global_var(0) > -500) then begin
-                                set_global_var(42, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                set_global_var(GVAR_KARMA_BETRAYER, 1);
                             end
                             else begin
-                                if (global_var(0) > -750) then begin
-                                    set_global_var(43, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -1000) then begin
-                                        set_global_var(44, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                     end
                                     else begin
-                                        set_global_var(45, 1);
+                                        set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                     end
                                 end
                             end
@@ -434,35 +434,35 @@ begin
                 end
             end
         end
-        if (global_var(37) == 1) then begin
+        if (global_var(GVAR_KARMA_HOLY_WARRIOR) == 1) then begin
             Static_Reaction := Static_Reaction + 20;
         end
         else begin
-            if (global_var(38) == 1) then begin
+            if (global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES) == 1) then begin
                 Static_Reaction := Static_Reaction + 15;
             end
             else begin
-                if (global_var(39) == 1) then begin
+                if (global_var(GVAR_KARMA_SHIELD_OF_HOPE) == 1) then begin
                     Static_Reaction := Static_Reaction + 10;
                 end
                 else begin
-                    if (global_var(40) == 1) then begin
+                    if (global_var(GVAR_KARMA_DEFENDER) == 1) then begin
                         Static_Reaction := Static_Reaction + 5;
                     end
                     else begin
-                        if (global_var(41) == 1) then begin
+                        if (global_var(GVAR_KARMA_WANDERER) == 1) then begin
                             Static_Reaction := Static_Reaction + 0;
                         end
                         else begin
-                            if (global_var(42) == 1) then begin
+                            if (global_var(GVAR_KARMA_BETRAYER) == 1) then begin
                                 Static_Reaction := Static_Reaction + -5;
                             end
                             else begin
-                                if (global_var(43) == 1) then begin
+                                if (global_var(GVAR_KARMA_SWORD_OF_DESPAIR) == 1) then begin
                                     Static_Reaction := Static_Reaction + -10;
                                 end
                                 else begin
-                                    if (global_var(44) == 1) then begin
+                                    if (global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE) == 1) then begin
                                         Static_Reaction := Static_Reaction + -15;
                                     end
                                     else begin
@@ -478,38 +478,38 @@ begin
         debug_msg("Good Person Reaction Bonus == " + Static_Reaction);
     end
     if (Slavery_Tolerant == 0) then begin
-        if (global_var(11) == 1) then begin
+        if (global_var(GVAR_REPUTATION_SLAVER) == 1) then begin
             Static_Reaction := Static_Reaction + (2 * -25);
         end
     end
     else begin
         if (Slavery_Tolerant == 1) then begin
-            if (global_var(11) == 1) then begin
+            if (global_var(GVAR_REPUTATION_SLAVER) == 1) then begin
                 Static_Reaction := Static_Reaction + -25;
             end
         end
         else begin
             if (Slavery_Tolerant == 3) then begin
-                if (global_var(11) == 1) then begin
+                if (global_var(GVAR_REPUTATION_SLAVER) == 1) then begin
                     Static_Reaction := Static_Reaction - -25;
                 end
             end
         end
     end
     debug_msg("Slaver + Aligned Reaction == " + Static_Reaction);
-    Static_Reaction := Static_Reaction + (10 * has_trait(0, dude_obj, 10));
+    Static_Reaction := Static_Reaction + (10 * has_trait(TRAIT_PERK, dude_obj, PERK_presence));
     debug_msg("Presence Reaction == " + Static_Reaction);
-    if (global_var(1) >= 2) then begin
+    if (global_var(GVAR_CHILDKILLER_REPUTATION) >= 2) then begin
         Static_Reaction := Static_Reaction + -30;
     end
     debug_msg("Childkiller Reaction == " + Static_Reaction);
     set_local_var(0, local_var(3) + Static_Reaction);
     debug_msg("Initial Reaction == " + local_var(0));
-    if (global_var(432) == 1) then begin
+    if (global_var(GVAR_ENCLAVE_GRANITE_JOINED) == 1) then begin
         call Node001();
     end
     else begin
-        if (get_critter_stat(dude_obj, 4) < 4) then begin
+        if (dude_iq < 4) then begin
             call Node002();
         end
         else begin
@@ -525,59 +525,59 @@ end
 procedure destroy_p_proc
 begin
     if (source_obj == dude_obj) then begin
-        set_global_var(5, global_var(5) + 1);
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(4) > (3 * global_var(5))) or (global_var(2) == 1)) and (global_var(1) == 0)) then begin
-            set_global_var(2, 1);
-            set_global_var(3, 0);
+        set_global_var(GVAR_GOOD_MONSTER, global_var(GVAR_GOOD_MONSTER) + 1);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(4) > (3 * global_var(GVAR_GOOD_MONSTER))) or (global_var(GVAR_CHAMPION_REPUTATION) == 1)) and (global_var(GVAR_CHILDKILLER_REPUTATION) == 0)) then begin
+            set_global_var(GVAR_CHAMPION_REPUTATION, 1);
+            set_global_var(GVAR_BERSERKER_REPUTATION, 0);
         end
-        if (metarule(51, self_obj) == 2) then begin
+        if (critter_kill_type(self_obj) == KILL_TYPE_children_kills) then begin
             if (source_obj == dude_obj) then begin
-                set_global_var(2, 0);
-                set_global_var(1, global_var(1) + 1);
-                set_global_var(0, global_var(0) + -15);
+                set_global_var(GVAR_CHAMPION_REPUTATION, 0);
+                set_global_var(GVAR_CHILDKILLER_REPUTATION, global_var(GVAR_CHILDKILLER_REPUTATION) + 1);
+                set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + -15);
                 debug_msg("Player gains " + -15 + " Karma Points.");
-                set_global_var(37, 0);
-                set_global_var(38, 0);
-                set_global_var(39, 0);
-                set_global_var(40, 0);
-                set_global_var(41, 0);
-                set_global_var(42, 0);
-                set_global_var(43, 0);
-                set_global_var(44, 0);
-                set_global_var(45, 0);
-                if (global_var(0) >= 1000) then begin
-                    set_global_var(37, 1);
+                set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+                set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+                set_global_var(GVAR_KARMA_DEFENDER, 0);
+                set_global_var(GVAR_KARMA_WANDERER, 0);
+                set_global_var(GVAR_KARMA_BETRAYER, 0);
+                set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+                set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+                set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+                    set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
                 end
                 else begin
-                    if (global_var(0) >= 750) then begin
-                        set_global_var(38, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
                     end
                     else begin
-                        if (global_var(0) >= 500) then begin
-                            set_global_var(39, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                            set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                         end
                         else begin
-                            if (global_var(0) >= 250) then begin
-                                set_global_var(40, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                                set_global_var(GVAR_KARMA_DEFENDER, 1);
                             end
                             else begin
-                                if (global_var(0) > -250) then begin
-                                    set_global_var(41, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                                    set_global_var(GVAR_KARMA_WANDERER, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -500) then begin
-                                        set_global_var(42, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                        set_global_var(GVAR_KARMA_BETRAYER, 1);
                                     end
                                     else begin
-                                        if (global_var(0) > -750) then begin
-                                            set_global_var(43, 1);
+                                        if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                            set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                         end
                                         else begin
-                                            if (global_var(0) > -1000) then begin
-                                                set_global_var(44, 1);
+                                            if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                                set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                             end
                                             else begin
-                                                set_global_var(45, 1);
+                                                set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                             end
                                         end
                                     end
@@ -586,58 +586,58 @@ begin
                         end
                     end
                 end
-                set_global_var(62, global_var(62) + -8);
+                set_global_var(GVAR_TOWN_REP_ENCLAVE, global_var(GVAR_TOWN_REP_ENCLAVE) + -8);
                 debug_msg("Added " + -8 + " to Town Rep");
             end
         end
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(5) > (2 * global_var(4))) or (global_var(3) == 1))) then begin
-            set_global_var(3, 1);
-            set_global_var(2, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
+            set_global_var(GVAR_BERSERKER_REPUTATION, 1);
+            set_global_var(GVAR_CHAMPION_REPUTATION, 0);
         end
-        set_global_var(0, global_var(0) + -10);
+        set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + -10);
         debug_msg("Player gains " + -10 + " Karma Points.");
-        set_global_var(37, 0);
-        set_global_var(38, 0);
-        set_global_var(39, 0);
-        set_global_var(40, 0);
-        set_global_var(41, 0);
-        set_global_var(42, 0);
-        set_global_var(43, 0);
-        set_global_var(44, 0);
-        set_global_var(45, 0);
-        if (global_var(0) >= 1000) then begin
-            set_global_var(37, 1);
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+        set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+        set_global_var(GVAR_KARMA_DEFENDER, 0);
+        set_global_var(GVAR_KARMA_WANDERER, 0);
+        set_global_var(GVAR_KARMA_BETRAYER, 0);
+        set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+        set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+            set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
         end
         else begin
-            if (global_var(0) >= 750) then begin
-                set_global_var(38, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
             end
             else begin
-                if (global_var(0) >= 500) then begin
-                    set_global_var(39, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                 end
                 else begin
-                    if (global_var(0) >= 250) then begin
-                        set_global_var(40, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                        set_global_var(GVAR_KARMA_DEFENDER, 1);
                     end
                     else begin
-                        if (global_var(0) > -250) then begin
-                            set_global_var(41, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                            set_global_var(GVAR_KARMA_WANDERER, 1);
                         end
                         else begin
-                            if (global_var(0) > -500) then begin
-                                set_global_var(42, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                set_global_var(GVAR_KARMA_BETRAYER, 1);
                             end
                             else begin
-                                if (global_var(0) > -750) then begin
-                                    set_global_var(43, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -1000) then begin
-                                        set_global_var(44, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                     end
                                     else begin
-                                        set_global_var(45, 1);
+                                        set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                     end
                                 end
                             end
@@ -646,7 +646,7 @@ begin
                 end
             end
         end
-        set_global_var(62, global_var(62) + -5);
+        set_global_var(GVAR_TOWN_REP_ENCLAVE, global_var(GVAR_TOWN_REP_ENCLAVE) + -5);
         debug_msg("Added " + -5 + " to Town Rep");
     end
 end
@@ -656,17 +656,17 @@ begin
     script_overrides;
     if (local_var(7) == 0) then begin
         set_local_var(7, local_var(7) + 1);
-        display_msg(message_str(857, 100));
+        display_mstr(100);
     end
     else begin
-        display_msg(message_str(857, 101));
+        display_mstr(101);
     end
 end
 
 procedure description_p_proc
 begin
     script_overrides;
-    display_msg(message_str(857, 102));
+    display_mstr(102);
 end
 
 procedure use_skill_on_p_proc
@@ -683,14 +683,14 @@ end
 procedure map_enter_p_proc
 begin
     Only_Once := 0;
-    if (global_var(432) == 0) then begin
-        critter_add_trait(self_obj, 1, 6, 192);
+    if (global_var(GVAR_ENCLAVE_GRANITE_JOINED) == 0) then begin
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 192);
     end
     else begin
-        critter_add_trait(self_obj, 1, 6, 0);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 0);
     end
-    critter_add_trait(self_obj, 1, 5, 22);
-    if (global_var(435) == 1) then begin
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_TOUGH_GUARD);
+    if (global_var(GVAR_ENCLAVE_REACTOR) == 1) then begin
         set_obj_visibility(self_obj, 0);
     end
     else begin
@@ -715,147 +715,147 @@ end
 
 procedure Node001
 begin
-    float_msg(self_obj, message_str(857, random(103, 111)), 8);
+    float_msg(self_obj, mstr(random(103, 111)), FLOAT_MSG_YELLOW);
 end
 
 procedure Node002
 begin
-    float_msg(self_obj, message_str(857, random(112, 121)), 8);
+    float_msg(self_obj, mstr(random(112, 121)), FLOAT_MSG_YELLOW);
 end
 
 procedure Node003
 begin
-    gsay_reply(857, 122);
-    giq_option(4, 857, 123, Node004, 49);
-    giq_option(4, 857, 124, Node005, 51);
-    giq_option(4, 857, 125, Node006, 51);
-    giq_option(4, 857, 126, Node016, 50);
+    Reply(122);
+    giq_option(4, NAME, 123, Node004, GOOD_REACTION);
+    giq_option(4, NAME, 124, Node005, BAD_REACTION);
+    giq_option(4, NAME, 125, Node006, BAD_REACTION);
+    giq_option(4, NAME, 126, Node016, NEUTRAL_REACTION);
 end
 
 procedure Node004
 begin
-    gsay_reply(857, 127);
-    giq_option(4, 857, 128, Node008, 50);
+    Reply(127);
+    giq_option(4, NAME, 128, Node008, NEUTRAL_REACTION);
 end
 
 procedure Node005
 begin
-    gsay_reply(857, 129);
-    giq_option(4, 857, 130, Node004, 49);
+    Reply(129);
+    giq_option(4, NAME, 130, Node004, GOOD_REACTION);
 end
 
 procedure Node006
 begin
-    gsay_reply(857, 131);
-    giq_option(4, 857, 132, Node004, 50);
-    giq_option(4, 857, 133, Node007, 51);
+    Reply(131);
+    giq_option(4, NAME, 132, Node004, NEUTRAL_REACTION);
+    giq_option(4, NAME, 133, Node007, BAD_REACTION);
 end
 
 procedure Node007
 begin
-    gsay_reply(857, 134);
-    giq_option(4, 857, 135, Node004, 50);
-    giq_option(4, 857, 136, Node998, 51);
+    Reply(134);
+    giq_option(4, NAME, 135, Node004, NEUTRAL_REACTION);
+    giq_option(4, NAME, 136, Node998, BAD_REACTION);
 end
 
 procedure Node008
 begin
-    gsay_reply(857, 137);
-    giq_option(4, 857, 138, Node009, 50);
-    giq_option(4, 857, 139, Node010, 51);
-    giq_option(4, 857, 140, Node007, 51);
+    Reply(137);
+    giq_option(4, NAME, 138, Node009, NEUTRAL_REACTION);
+    giq_option(4, NAME, 139, Node010, BAD_REACTION);
+    giq_option(4, NAME, 140, Node007, BAD_REACTION);
 end
 
 procedure Node009
 begin
-    gsay_reply(857, 141);
-    giq_option(4, 857, 142, Node010, 50);
+    Reply(141);
+    giq_option(4, NAME, 142, Node010, NEUTRAL_REACTION);
 end
 
 procedure Node010
 begin
-    gsay_reply(857, 143);
-    giq_option(4, 857, 144, Node011, 50);
+    Reply(143);
+    giq_option(4, NAME, 144, Node011, NEUTRAL_REACTION);
 end
 
 procedure Node011
 begin
-    gsay_reply(857, 145);
-    giq_option(4, 857, 146, Node011a, 50);
-    giq_option(4, 857, 147, Node998, 51);
-    giq_option(4, 857, 148, Node999, 50);
+    Reply(145);
+    giq_option(4, NAME, 146, Node011a, NEUTRAL_REACTION);
+    giq_option(4, NAME, 147, Node998, BAD_REACTION);
+    giq_option(4, NAME, 148, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node012
 begin
-    gsay_reply(857, 149);
-    giq_option(4, 857, 150, Node015, 49);
-    giq_option(4, 857, 151, Node999, 51);
+    Reply(149);
+    giq_option(4, NAME, 150, Node015, GOOD_REACTION);
+    giq_option(4, NAME, 151, Node999, BAD_REACTION);
 end
 
 procedure Node013
 begin
-    gsay_reply(857, 152);
-    giq_option(4, 857, 153, Node013a, 50);
-    giq_option(4, 857, 154, Node999, 51);
+    Reply(152);
+    giq_option(4, NAME, 153, Node013a, NEUTRAL_REACTION);
+    giq_option(4, NAME, 154, Node999, BAD_REACTION);
 end
 
 procedure Node014
 begin
-    gsay_reply(857, 155);
-    giq_option(4, 857, 156, Node998, 51);
-    giq_option(4, 857, 157, Node999, 50);
+    Reply(155);
+    giq_option(4, NAME, 156, Node998, BAD_REACTION);
+    giq_option(4, NAME, 157, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node015
 begin
-    set_global_var(432, 1);
+    set_global_var(GVAR_ENCLAVE_GRANITE_JOINED, 1);
     give_exp_points(1500);
-    display_msg(message_str(14, 100) + 1500 + message_str(14, 101));
-    set_global_var(0, global_var(0) + 25);
+    display_msg(g_mstr(100) + 1500 + g_mstr(101));
+    set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + 25);
     debug_msg("Player gains " + 25 + " Karma Points.");
-    set_global_var(37, 0);
-    set_global_var(38, 0);
-    set_global_var(39, 0);
-    set_global_var(40, 0);
-    set_global_var(41, 0);
-    set_global_var(42, 0);
-    set_global_var(43, 0);
-    set_global_var(44, 0);
-    set_global_var(45, 0);
-    if (global_var(0) >= 1000) then begin
-        set_global_var(37, 1);
+    set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+    set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+    set_global_var(GVAR_KARMA_DEFENDER, 0);
+    set_global_var(GVAR_KARMA_WANDERER, 0);
+    set_global_var(GVAR_KARMA_BETRAYER, 0);
+    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+    set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+    set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+    if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
     end
     else begin
-        if (global_var(0) >= 750) then begin
-            set_global_var(38, 1);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+            set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
         end
         else begin
-            if (global_var(0) >= 500) then begin
-                set_global_var(39, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
             end
             else begin
-                if (global_var(0) >= 250) then begin
-                    set_global_var(40, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                    set_global_var(GVAR_KARMA_DEFENDER, 1);
                 end
                 else begin
-                    if (global_var(0) > -250) then begin
-                        set_global_var(41, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                        set_global_var(GVAR_KARMA_WANDERER, 1);
                     end
                     else begin
-                        if (global_var(0) > -500) then begin
-                            set_global_var(42, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                            set_global_var(GVAR_KARMA_BETRAYER, 1);
                         end
                         else begin
-                            if (global_var(0) > -750) then begin
-                                set_global_var(43, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                             end
                             else begin
-                                if (global_var(0) > -1000) then begin
-                                    set_global_var(44, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                    set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                 end
                                 else begin
-                                    set_global_var(45, 1);
+                                    set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                 end
                             end
                         end
@@ -864,20 +864,20 @@ begin
             end
         end
     end
-    gsay_reply(857, 158);
-    giq_option(4, 857, 159, Node999, 49);
+    Reply(158);
+    giq_option(4, NAME, 159, Node999, GOOD_REACTION);
 end
 
 procedure Node016
 begin
-    gsay_reply(857, 160);
-    giq_option(4, 857, 161, Node017, 50);
+    Reply(160);
+    giq_option(4, NAME, 161, Node017, NEUTRAL_REACTION);
 end
 
 procedure Node017
 begin
-    gsay_reply(857, 162);
-    giq_option(4, 857, 163, Node004, 50);
+    Reply(162);
+    giq_option(4, NAME, 163, Node004, NEUTRAL_REACTION);
 end
 
 procedure Node011a
@@ -899,4 +899,3 @@ begin
         call Node014();
     end
 end
-
