diff --git a/Mods/MegaMod/Original/QITURTRM.ssl b/Mods/MegaMod/ReDefine/QITURTRM.ssl
index cc7272f..2a49c4b 100644
--- a/Mods/MegaMod/Original/QITURTRM.ssl
+++ b/Mods/MegaMod/ReDefine/QITURTRM.ssl
@@ -187,13 +187,13 @@ end
 procedure use_p_proc
 begin
     script_overrides;
-    if (metarule(40, 8) or is_success(roll_vs_skill(dude_obj, 8, -70))) then begin
+    if (is_skill_tagged(SKILL_SNEAK) or is_success(roll_vs_skill(dude_obj, 8, -70))) then begin
         Needed_IQ := 4;
     end
     else begin
         Needed_IQ := 6;
     end
-    if ((local_var(1) == 1) or (get_critter_stat(dude_obj, 4) < Needed_IQ)) then begin
+    if ((local_var(1) == 1) or (dude_iq < Needed_IQ)) then begin
         call Node002();
     end
     else begin
@@ -227,14 +227,14 @@ begin
     script_overrides;
     if (local_var(2) == 0) then begin
         set_local_var(2, local_var(2) + 1);
-        display_msg(message_str(1163, 100));
+        display_mstr(100);
     end
     else begin
         if (local_var(1) == 1) then begin
             call Node002();
         end
         else begin
-            display_msg(message_str(1163, 101));
+            display_mstr(101);
         end
     end
 end
@@ -243,10 +243,10 @@ procedure description_p_proc
 begin
     script_overrides;
     if (local_var(1) == 1) then begin
-        display_msg(message_str(1163, 102));
+        display_mstr(102);
     end
     else begin
-        display_msg(message_str(1163, 103));
+        display_mstr(103);
     end
 end
 
@@ -259,316 +259,316 @@ end
 
 procedure Node001
 begin
-    gsay_reply(1163, 104);
-    giq_option(Needed_IQ, 1163, 105, Node003, 50);
-    giq_option(Needed_IQ, 1163, 106, Node017, 50);
-    giq_option(Needed_IQ, 1163, 107, NodeEnd, 50);
+    Reply(104);
+    giq_option(Needed_IQ, NAME, 105, Node003, NEUTRAL_REACTION);
+    giq_option(Needed_IQ, NAME, 106, Node017, NEUTRAL_REACTION);
+    giq_option(Needed_IQ, NAME, 107, NodeEnd, NEUTRAL_REACTION);
 end
 
 procedure Node002
 begin
-    display_msg(message_str(1163, random(108, 117)));
+    display_mstr(random(108, 117));
 end
 
 procedure Node003
 begin
-    gsay_reply(1163, 118);
-    if (has_skill(dude_obj, 12) >= 100) then begin
-        giq_option(Needed_IQ, 1163, 119, Node004, 50);
+    Reply(118);
+    if (dude_skill(SKILL_SCIENCE) >= 100) then begin
+        giq_option(Needed_IQ, NAME, 119, Node004, NEUTRAL_REACTION);
     end
-    if (get_critter_stat(dude_obj, 6) >= 8) then begin
-        giq_option(Needed_IQ, 1163, 120, Node008, 50);
+    if (dude_luck >= 8) then begin
+        giq_option(Needed_IQ, NAME, 120, Node008, NEUTRAL_REACTION);
     end
-    giq_option(Needed_IQ, 1163, 121, Node001, 50);
-    giq_option(Needed_IQ, 1163, 123, NodeEnd, 50);
+    giq_option(Needed_IQ, NAME, 121, Node001, NEUTRAL_REACTION);
+    giq_option(Needed_IQ, NAME, 123, NodeEnd, NEUTRAL_REACTION);
 end
 
 procedure Node004
 begin
     if (local_var(3) == 0) then begin
         set_local_var(3, local_var(3) + 1);
-        if (get_critter_stat(dude_obj, 4) >= 7) then begin
-            gsay_reply(1163, 124);
+        if (dude_iq >= 7) then begin
+            Reply(124);
         end
         else begin
-            gsay_reply(1163, 125);
+            Reply(125);
         end
     end
     else begin
-        gsay_reply(1163, 126);
+        Reply(126);
     end
-    giq_option(Needed_IQ, 1163, 127, Node008, 50);
-    giq_option(Needed_IQ, 1163, 128, Node020, 50);
-    if (has_skill(dude_obj, 12) >= 125) then begin
-        giq_option(Needed_IQ, 1163, 129, Node005, 50);
+    giq_option(Needed_IQ, NAME, 127, Node008, NEUTRAL_REACTION);
+    giq_option(Needed_IQ, NAME, 128, Node020, NEUTRAL_REACTION);
+    if (dude_skill(SKILL_SCIENCE) >= 125) then begin
+        giq_option(Needed_IQ, NAME, 129, Node005, NEUTRAL_REACTION);
     end
-    if (has_skill(dude_obj, 12) >= 150) then begin
-        giq_option(Needed_IQ, 1163, 130, Node006, 50);
+    if (dude_skill(SKILL_SCIENCE) >= 150) then begin
+        giq_option(Needed_IQ, NAME, 130, Node006, NEUTRAL_REACTION);
     end
-    if (has_skill(dude_obj, 12) >= 175) then begin
-        giq_option(Needed_IQ, 1163, 131, Node007, 50);
+    if (dude_skill(SKILL_SCIENCE) >= 175) then begin
+        giq_option(Needed_IQ, NAME, 131, Node007, NEUTRAL_REACTION);
     end
-    giq_option(Needed_IQ, 1163, 132, NodeEnd, 50);
+    giq_option(Needed_IQ, NAME, 132, NodeEnd, NEUTRAL_REACTION);
 end
 
 procedure Node005
 begin
-    gsay_reply(1163, message_str(1163, 133) + message_str(1163, 134) + message_str(1163, 135));
-    if (has_skill(dude_obj, 12) >= 125) then begin
-        giq_option(Needed_IQ, 1163, 136, Node014, 50);
+    Reply(mstr(133) + mstr(134) + mstr(135));
+    if (dude_skill(SKILL_SCIENCE) >= 125) then begin
+        giq_option(Needed_IQ, NAME, 136, Node014, NEUTRAL_REACTION);
     end
-    if (has_skill(dude_obj, 12) >= 100) then begin
-        giq_option(Needed_IQ, 1163, 137, Node009, 50);
+    if (dude_skill(SKILL_SCIENCE) >= 100) then begin
+        giq_option(Needed_IQ, NAME, 137, Node009, NEUTRAL_REACTION);
     end
-    giq_option(Needed_IQ, 1163, 138, Node004, 50);
+    giq_option(Needed_IQ, NAME, 138, Node004, NEUTRAL_REACTION);
 end
 
 procedure Node006
 begin
-    gsay_reply(1163, message_str(1163, 139) + message_str(1163, 140) + message_str(1163, 141) + message_str(1163, 142) + message_str(1163, 143));
-    if (has_skill(dude_obj, 12) >= 150) then begin
-        giq_option(Needed_IQ, 1163, 144, Node012, 50);
+    Reply(mstr(139) + mstr(140) + mstr(141) + mstr(142) + mstr(143));
+    if (dude_skill(SKILL_SCIENCE) >= 150) then begin
+        giq_option(Needed_IQ, NAME, 144, Node012, NEUTRAL_REACTION);
     end
-    if (has_skill(dude_obj, 12) >= 150) then begin
-        giq_option(Needed_IQ, 1163, 145, Node013, 50);
+    if (dude_skill(SKILL_SCIENCE) >= 150) then begin
+        giq_option(Needed_IQ, NAME, 145, Node013, NEUTRAL_REACTION);
     end
-    if (has_skill(dude_obj, 12) >= 125) then begin
-        giq_option(Needed_IQ, 1163, 146, Node014, 50);
+    if (dude_skill(SKILL_SCIENCE) >= 125) then begin
+        giq_option(Needed_IQ, NAME, 146, Node014, NEUTRAL_REACTION);
     end
-    if (has_skill(dude_obj, 12) >= 100) then begin
-        giq_option(Needed_IQ, 1163, 147, Node009, 50);
+    if (dude_skill(SKILL_SCIENCE) >= 100) then begin
+        giq_option(Needed_IQ, NAME, 147, Node009, NEUTRAL_REACTION);
     end
-    giq_option(Needed_IQ, 1163, 148, Node004, 50);
+    giq_option(Needed_IQ, NAME, 148, Node004, NEUTRAL_REACTION);
 end
 
 procedure Node007
 begin
-    gsay_reply(1163, message_str(1163, 149) + message_str(1163, 150) + message_str(1163, 151) + message_str(1163, 152) + message_str(1163, 153) + message_str(1163, 154) + message_str(1163, 155));
-    if (has_skill(dude_obj, 12) >= 175) then begin
-        giq_option(Needed_IQ, 1163, 156, Node010, 50);
+    Reply(mstr(149) + mstr(150) + mstr(151) + mstr(152) + mstr(153) + mstr(154) + mstr(155));
+    if (dude_skill(SKILL_SCIENCE) >= 175) then begin
+        giq_option(Needed_IQ, NAME, 156, Node010, NEUTRAL_REACTION);
     end
-    if (has_skill(dude_obj, 12) >= 175) then begin
-        giq_option(Needed_IQ, 1163, 157, Node011, 50);
+    if (dude_skill(SKILL_SCIENCE) >= 175) then begin
+        giq_option(Needed_IQ, NAME, 157, Node011, NEUTRAL_REACTION);
     end
-    if (has_skill(dude_obj, 12) >= 150) then begin
-        giq_option(Needed_IQ, 1163, 158, Node012, 50);
+    if (dude_skill(SKILL_SCIENCE) >= 150) then begin
+        giq_option(Needed_IQ, NAME, 158, Node012, NEUTRAL_REACTION);
     end
-    if (has_skill(dude_obj, 12) >= 150) then begin
-        giq_option(Needed_IQ, 1163, 159, Node013, 50);
+    if (dude_skill(SKILL_SCIENCE) >= 150) then begin
+        giq_option(Needed_IQ, NAME, 159, Node013, NEUTRAL_REACTION);
     end
-    if (has_skill(dude_obj, 12) >= 125) then begin
-        giq_option(Needed_IQ, 1163, 160, Node014, 50);
+    if (dude_skill(SKILL_SCIENCE) >= 125) then begin
+        giq_option(Needed_IQ, NAME, 160, Node014, NEUTRAL_REACTION);
     end
-    if (has_skill(dude_obj, 12) >= 100) then begin
-        giq_option(Needed_IQ, 1163, 161, Node009, 50);
+    if (dude_skill(SKILL_SCIENCE) >= 100) then begin
+        giq_option(Needed_IQ, NAME, 161, Node009, NEUTRAL_REACTION);
     end
-    giq_option(Needed_IQ, 1163, 162, Node004, 50);
+    giq_option(Needed_IQ, NAME, 162, Node004, NEUTRAL_REACTION);
 end
 
 procedure Node008
 begin
-    gsay_reply(1163, 163);
-    if (has_skill(dude_obj, 12) >= 100) then begin
-        giq_option(Needed_IQ, 1163, 164, Node009, 50);
+    Reply(163);
+    if (dude_skill(SKILL_SCIENCE) >= 100) then begin
+        giq_option(Needed_IQ, NAME, 164, Node009, NEUTRAL_REACTION);
     end
-    giq_option(Needed_IQ, 1163, 165, Node004, 50);
+    giq_option(Needed_IQ, NAME, 165, Node004, NEUTRAL_REACTION);
 end
 
 procedure Node009
 begin
-    set_global_var(573, 1);
-    gsay_reply(1163, 166);
-    if (has_skill(dude_obj, 12) >= 175) then begin
-        giq_option(Needed_IQ, 1163, 167, Node010, 50);
+    set_global_var(GVAR_HOLODISK_ENCLAVE_SECURITY, 1);
+    Reply(166);
+    if (dude_skill(SKILL_SCIENCE) >= 175) then begin
+        giq_option(Needed_IQ, NAME, 167, Node010, NEUTRAL_REACTION);
     end
-    if (has_skill(dude_obj, 12) >= 175) then begin
-        giq_option(Needed_IQ, 1163, 168, Node011, 50);
+    if (dude_skill(SKILL_SCIENCE) >= 175) then begin
+        giq_option(Needed_IQ, NAME, 168, Node011, NEUTRAL_REACTION);
     end
-    if (has_skill(dude_obj, 12) >= 150) then begin
-        giq_option(Needed_IQ, 1163, 169, Node012, 50);
+    if (dude_skill(SKILL_SCIENCE) >= 150) then begin
+        giq_option(Needed_IQ, NAME, 169, Node012, NEUTRAL_REACTION);
     end
-    if (has_skill(dude_obj, 12) >= 150) then begin
-        giq_option(Needed_IQ, 1163, 170, Node013, 50);
+    if (dude_skill(SKILL_SCIENCE) >= 150) then begin
+        giq_option(Needed_IQ, NAME, 170, Node013, NEUTRAL_REACTION);
     end
-    if (has_skill(dude_obj, 12) >= 125) then begin
-        giq_option(Needed_IQ, 1163, 171, Node014, 50);
+    if (dude_skill(SKILL_SCIENCE) >= 125) then begin
+        giq_option(Needed_IQ, NAME, 171, Node014, NEUTRAL_REACTION);
     end
-    giq_option(Needed_IQ, 1163, 172, Node004, 50);
+    giq_option(Needed_IQ, NAME, 172, Node004, NEUTRAL_REACTION);
 end
 
 procedure Node010
 begin
-    set_global_var(574, 1);
-    gsay_reply(1163, 173);
-    if (has_skill(dude_obj, 12) >= 175) then begin
-        giq_option(Needed_IQ, 1163, 174, Node011, 50);
+    set_global_var(GVAR_HOLODISK_ENCLAVE_STATE, 1);
+    Reply(173);
+    if (dude_skill(SKILL_SCIENCE) >= 175) then begin
+        giq_option(Needed_IQ, NAME, 174, Node011, NEUTRAL_REACTION);
     end
-    if (has_skill(dude_obj, 12) >= 150) then begin
-        giq_option(Needed_IQ, 1163, 175, Node012, 50);
-        giq_option(Needed_IQ, 1163, 176, Node013, 50);
+    if (dude_skill(SKILL_SCIENCE) >= 150) then begin
+        giq_option(Needed_IQ, NAME, 175, Node012, NEUTRAL_REACTION);
+        giq_option(Needed_IQ, NAME, 176, Node013, NEUTRAL_REACTION);
     end
-    if (has_skill(dude_obj, 12) >= 125) then begin
-        giq_option(Needed_IQ, 1163, 177, Node014, 50);
+    if (dude_skill(SKILL_SCIENCE) >= 125) then begin
+        giq_option(Needed_IQ, NAME, 177, Node014, NEUTRAL_REACTION);
     end
-    if (has_skill(dude_obj, 12) >= 100) then begin
-        giq_option(Needed_IQ, 1163, 178, Node004, 50);
+    if (dude_skill(SKILL_SCIENCE) >= 100) then begin
+        giq_option(Needed_IQ, NAME, 178, Node004, NEUTRAL_REACTION);
     end
-    giq_option(Needed_IQ, 1163, 179, Node004, 50);
+    giq_option(Needed_IQ, NAME, 179, Node004, NEUTRAL_REACTION);
 end
 
 procedure Node011
 begin
-    set_global_var(575, 1);
-    gsay_reply(1163, 180);
-    if (has_skill(dude_obj, 12) >= 175) then begin
-        giq_option(Needed_IQ, 1163, 181, Node010, 50);
+    set_global_var(GVAR_HOLODISK_ENCLAVE_WORD, 1);
+    Reply(180);
+    if (dude_skill(SKILL_SCIENCE) >= 175) then begin
+        giq_option(Needed_IQ, NAME, 181, Node010, NEUTRAL_REACTION);
     end
-    if (has_skill(dude_obj, 12) >= 150) then begin
-        giq_option(Needed_IQ, 1163, 182, Node012, 50);
-        giq_option(Needed_IQ, 1163, 183, Node013, 50);
+    if (dude_skill(SKILL_SCIENCE) >= 150) then begin
+        giq_option(Needed_IQ, NAME, 182, Node012, NEUTRAL_REACTION);
+        giq_option(Needed_IQ, NAME, 183, Node013, NEUTRAL_REACTION);
     end
-    if (has_skill(dude_obj, 12) >= 125) then begin
-        giq_option(Needed_IQ, 1163, 184, Node014, 50);
+    if (dude_skill(SKILL_SCIENCE) >= 125) then begin
+        giq_option(Needed_IQ, NAME, 184, Node014, NEUTRAL_REACTION);
     end
-    if (has_skill(dude_obj, 12) >= 100) then begin
-        giq_option(Needed_IQ, 1163, 185, Node009, 50);
+    if (dude_skill(SKILL_SCIENCE) >= 100) then begin
+        giq_option(Needed_IQ, NAME, 185, Node009, NEUTRAL_REACTION);
     end
-    giq_option(Needed_IQ, 1163, 186, Node004, 50);
+    giq_option(Needed_IQ, NAME, 186, Node004, NEUTRAL_REACTION);
 end
 
 procedure Node012
 begin
-    if (cur_map_index == 131) then begin
-        if (global_var(568) == 1) then begin
-            gsay_reply(1163, 187);
+    if (CUR_MAP_ENCGD) then begin
+        if (global_var(GVAR_ENCLAVE_TURRET_GUARD) == 1) then begin
+            Reply(187);
         end
         else begin
-            if (global_var(568) == 0) then begin
-                gsay_reply(1163, 188);
+            if (global_var(GVAR_ENCLAVE_TURRET_GUARD) == 0) then begin
+                Reply(188);
             end
             else begin
-                gsay_reply(1163, 189);
+                Reply(189);
             end
         end
     end
     else begin
-        if (cur_map_index == 128) then begin
-            if (global_var(569) == 1) then begin
-                gsay_reply(1163, 190);
+        if (CUR_MAP_ENCDET) then begin
+            if (global_var(GVAR_ENCLAVE_TURRET_DETENTION) == 1) then begin
+                Reply(190);
             end
             else begin
-                if (global_var(569) == 0) then begin
-                    gsay_reply(1163, 191);
+                if (global_var(GVAR_ENCLAVE_TURRET_DETENTION) == 0) then begin
+                    Reply(191);
                 end
                 else begin
-                    gsay_reply(1163, 192);
+                    Reply(192);
                 end
             end
         end
         else begin
-            if (cur_map_index == 133) then begin
-                if (global_var(570) == 1) then begin
-                    gsay_reply(1163, 193);
+            if (CUR_MAP_ENCRCTR) then begin
+                if (global_var(GVAR_ENCLAVE_TURRET_SCIENCE) == 1) then begin
+                    Reply(193);
                 end
                 else begin
-                    if (global_var(570) == 0) then begin
-                        gsay_reply(1163, 194);
+                    if (global_var(GVAR_ENCLAVE_TURRET_SCIENCE) == 0) then begin
+                        Reply(194);
                     end
                     else begin
-                        gsay_reply(1163, 195);
+                        Reply(195);
                     end
                 end
             end
             else begin
-                if (cur_map_index == 132) then begin
-                    if (global_var(571) == 1) then begin
-                        gsay_reply(1163, 196);
+                if (CUR_MAP_ENCPRES) then begin
+                    if (global_var(GVAR_ENCLAVE_TURRET_PRESIDENT) == 1) then begin
+                        Reply(196);
                     end
                     else begin
-                        if (global_var(571) == 0) then begin
-                            gsay_reply(1163, 197);
+                        if (global_var(GVAR_ENCLAVE_TURRET_PRESIDENT) == 0) then begin
+                            Reply(197);
                         end
                         else begin
-                            gsay_reply(1163, 198);
+                            Reply(198);
                         end
                     end
                 end
                 else begin
-                    if (global_var(572) == 1) then begin
-                        gsay_reply(1163, 199);
+                    if (global_var(GVAR_ENCLAVE_TURRET_MAIN) == 1) then begin
+                        Reply(199);
                     end
                     else begin
-                        if (global_var(572) == 0) then begin
-                            gsay_reply(1163, 200);
+                        if (global_var(GVAR_ENCLAVE_TURRET_MAIN) == 0) then begin
+                            Reply(200);
                         end
                         else begin
-                            gsay_reply(1163, 201);
+                            Reply(201);
                         end
                     end
                 end
             end
         end
     end
-    if (has_skill(dude_obj, 12) >= 125) then begin
-        giq_option(Needed_IQ, 1163, 202, Node020, 50);
+    if (dude_skill(SKILL_SCIENCE) >= 125) then begin
+        giq_option(Needed_IQ, NAME, 202, Node020, NEUTRAL_REACTION);
     end
-    if (has_skill(dude_obj, 12) >= 175) then begin
-        giq_option(Needed_IQ, 1163, 203, Node010, 50);
-        giq_option(Needed_IQ, 1163, 204, Node011, 50);
+    if (dude_skill(SKILL_SCIENCE) >= 175) then begin
+        giq_option(Needed_IQ, NAME, 203, Node010, NEUTRAL_REACTION);
+        giq_option(Needed_IQ, NAME, 204, Node011, NEUTRAL_REACTION);
     end
-    if (has_skill(dude_obj, 12) >= 150) then begin
-        giq_option(Needed_IQ, 1163, 205, Node013, 50);
+    if (dude_skill(SKILL_SCIENCE) >= 150) then begin
+        giq_option(Needed_IQ, NAME, 205, Node013, NEUTRAL_REACTION);
     end
-    if (has_skill(dude_obj, 12) >= 125) then begin
-        giq_option(Needed_IQ, 1163, 206, Node014, 50);
+    if (dude_skill(SKILL_SCIENCE) >= 125) then begin
+        giq_option(Needed_IQ, NAME, 206, Node014, NEUTRAL_REACTION);
     end
-    if (has_skill(dude_obj, 12) >= 100) then begin
-        giq_option(Needed_IQ, 1163, 207, Node009, 50);
+    if (dude_skill(SKILL_SCIENCE) >= 100) then begin
+        giq_option(Needed_IQ, NAME, 207, Node009, NEUTRAL_REACTION);
     end
-    giq_option(Needed_IQ, 1163, 208, Node004, 50);
+    giq_option(Needed_IQ, NAME, 208, Node004, NEUTRAL_REACTION);
 end
 
 procedure Node013
 begin
-    set_global_var(576, 1);
-    gsay_reply(1163, 209);
-    if (has_skill(dude_obj, 12) >= 175) then begin
-        giq_option(Needed_IQ, 1163, 210, Node010, 50);
-        giq_option(Needed_IQ, 1163, 211, Node011, 50);
+    set_global_var(GVAR_HOLODISK_ENCLAVE_CHEMICAL, 1);
+    Reply(209);
+    if (dude_skill(SKILL_SCIENCE) >= 175) then begin
+        giq_option(Needed_IQ, NAME, 210, Node010, NEUTRAL_REACTION);
+        giq_option(Needed_IQ, NAME, 211, Node011, NEUTRAL_REACTION);
     end
-    if (has_skill(dude_obj, 12) >= 150) then begin
-        giq_option(Needed_IQ, 1163, 212, Node012, 50);
+    if (dude_skill(SKILL_SCIENCE) >= 150) then begin
+        giq_option(Needed_IQ, NAME, 212, Node012, NEUTRAL_REACTION);
     end
-    if (has_skill(dude_obj, 12) >= 125) then begin
-        giq_option(Needed_IQ, 1163, 213, Node014, 50);
+    if (dude_skill(SKILL_SCIENCE) >= 125) then begin
+        giq_option(Needed_IQ, NAME, 213, Node014, NEUTRAL_REACTION);
     end
-    if (has_skill(dude_obj, 12) >= 100) then begin
-        giq_option(Needed_IQ, 1163, 215, Node009, 50);
+    if (dude_skill(SKILL_SCIENCE) >= 100) then begin
+        giq_option(Needed_IQ, NAME, 215, Node009, NEUTRAL_REACTION);
     end
-    giq_option(Needed_IQ, 1163, 216, Node004, 50);
+    giq_option(Needed_IQ, NAME, 216, Node004, NEUTRAL_REACTION);
 end
 
 procedure Node014
 begin
-    set_global_var(577, 1);
-    gsay_reply(1163, 217);
-    if (has_skill(dude_obj, 12) >= 175) then begin
-        giq_option(Needed_IQ, 1163, 218, Node010, 50);
-        giq_option(Needed_IQ, 1163, 219, Node011, 50);
+    set_global_var(GVAR_HOLODISK_ENCLAVE_ATOMIC, 1);
+    Reply(217);
+    if (dude_skill(SKILL_SCIENCE) >= 175) then begin
+        giq_option(Needed_IQ, NAME, 218, Node010, NEUTRAL_REACTION);
+        giq_option(Needed_IQ, NAME, 219, Node011, NEUTRAL_REACTION);
     end
-    if (has_skill(dude_obj, 12) >= 150) then begin
-        giq_option(Needed_IQ, 1163, 220, Node012, 50);
-        giq_option(Needed_IQ, 1163, 221, Node013, 50);
+    if (dude_skill(SKILL_SCIENCE) >= 150) then begin
+        giq_option(Needed_IQ, NAME, 220, Node012, NEUTRAL_REACTION);
+        giq_option(Needed_IQ, NAME, 221, Node013, NEUTRAL_REACTION);
     end
-    if (has_skill(dude_obj, 12) >= 100) then begin
-        giq_option(Needed_IQ, 1163, 222, Node009, 50);
+    if (dude_skill(SKILL_SCIENCE) >= 100) then begin
+        giq_option(Needed_IQ, NAME, 222, Node009, NEUTRAL_REACTION);
     end
-    giq_option(Needed_IQ, 1163, 223, Node004, 50);
+    giq_option(Needed_IQ, NAME, 223, Node004, NEUTRAL_REACTION);
 end
 
 procedure Node015
 begin
-    display_msg(message_str(1163, 224));
+    display_mstr(224);
 end
 
 procedure Node016
@@ -577,9 +577,9 @@ end
 
 procedure Node017
 begin
-    gsay_reply(1163, 225);
-    giq_option(Needed_IQ, 1163, 226, Node019, 50);
-    giq_option(Needed_IQ, 1163, 227, Node001, 50);
+    Reply(225);
+    giq_option(Needed_IQ, NAME, 226, Node019, NEUTRAL_REACTION);
+    giq_option(Needed_IQ, NAME, 227, Node001, NEUTRAL_REACTION);
 end
 
 procedure Node018
@@ -588,121 +588,121 @@ end
 
 procedure Node019
 begin
-    gsay_reply(1163, 228);
-    giq_option(Needed_IQ, 1163, 229, Node004, 50);
-    giq_option(Needed_IQ, 1163, 230, NodeEnd, 50);
+    Reply(228);
+    giq_option(Needed_IQ, NAME, 229, Node004, NEUTRAL_REACTION);
+    giq_option(Needed_IQ, NAME, 230, NodeEnd, NEUTRAL_REACTION);
 end
 
 procedure Node020
 begin
-    gsay_reply(1163, 231);
-    giq_option(Needed_IQ, 1163, 232, Node021, 50);
-    giq_option(Needed_IQ, 1163, 233, Node025, 50);
-    giq_option(Needed_IQ, 1163, 234, Node004, 50);
-    giq_option(Needed_IQ, 1163, 235, NodeEnd, 50);
+    Reply(231);
+    giq_option(Needed_IQ, NAME, 232, Node021, NEUTRAL_REACTION);
+    giq_option(Needed_IQ, NAME, 233, Node025, NEUTRAL_REACTION);
+    giq_option(Needed_IQ, NAME, 234, Node004, NEUTRAL_REACTION);
+    giq_option(Needed_IQ, NAME, 235, NodeEnd, NEUTRAL_REACTION);
 end
 
 procedure Node021
 begin
-    gsay_reply(1163, 236);
-    giq_option(Needed_IQ, 1163, 237, Node021a, 50);
+    Reply(236);
+    giq_option(Needed_IQ, NAME, 237, Node021a, NEUTRAL_REACTION);
     if (hacked_system) then begin
-        giq_option(Needed_IQ, 1163, 238, Node020a, 50);
+        giq_option(Needed_IQ, NAME, 238, Node020a, NEUTRAL_REACTION);
     end
     else begin
-        giq_option(Needed_IQ, 1163, 238, Node020, 50);
+        giq_option(Needed_IQ, NAME, 238, Node020, NEUTRAL_REACTION);
     end
-    giq_option(Needed_IQ, 1163, 239, NodeEnd, 50);
+    giq_option(Needed_IQ, NAME, 239, NodeEnd, NEUTRAL_REACTION);
 end
 
 procedure Node022
 begin
-    set_global_var(578, 1);
-    gsay_reply(1163, 240);
-    giq_option(Needed_IQ, 1163, 241, NodeEnd, 50);
+    set_global_var(GVAR_ENCLAVE_TURRET_HELP_PLAYER, 1);
+    Reply(240);
+    giq_option(Needed_IQ, NAME, 241, NodeEnd, NEUTRAL_REACTION);
 end
 
 procedure Node023
 begin
-    gsay_reply(1163, 242);
-    giq_option(Needed_IQ, 1163, 243, Node004, 50);
+    Reply(242);
+    giq_option(Needed_IQ, NAME, 243, Node004, NEUTRAL_REACTION);
     if (hacked_system) then begin
-        giq_option(Needed_IQ, 1163, 244, Node020a, 50);
+        giq_option(Needed_IQ, NAME, 244, Node020a, NEUTRAL_REACTION);
     end
     else begin
-        giq_option(Needed_IQ, 1163, 244, Node020, 50);
+        giq_option(Needed_IQ, NAME, 244, Node020, NEUTRAL_REACTION);
     end
 end
 
 procedure Node024
 begin
-    gsay_reply(1163, 245);
-    if (has_skill(dude_obj, 12) >= 75) then begin
-        giq_option(Needed_IQ, 1163, 255, Node027a, 50);
+    Reply(245);
+    if (dude_skill(SKILL_SCIENCE) >= 75) then begin
+        giq_option(Needed_IQ, NAME, 255, Node027a, NEUTRAL_REACTION);
     end
     else begin
-        giq_option(Needed_IQ, 1163, 256, Node020, 50);
+        giq_option(Needed_IQ, NAME, 256, Node020, NEUTRAL_REACTION);
     end
 end
 
 procedure Node025
 begin
-    gsay_reply(1163, 247);
-    giq_option(Needed_IQ, 1163, 248, Node025a, 50);
+    Reply(247);
+    giq_option(Needed_IQ, NAME, 248, Node025a, NEUTRAL_REACTION);
     if (hacked_system) then begin
-        giq_option(Needed_IQ, 1163, 249, Node020a, 50);
+        giq_option(Needed_IQ, NAME, 249, Node020a, NEUTRAL_REACTION);
     end
     else begin
-        giq_option(Needed_IQ, 1163, 249, Node020, 50);
+        giq_option(Needed_IQ, NAME, 249, Node020, NEUTRAL_REACTION);
     end
-    giq_option(Needed_IQ, 1163, 250, NodeEnd, 50);
+    giq_option(Needed_IQ, NAME, 250, NodeEnd, NEUTRAL_REACTION);
 end
 
 procedure Node026
 begin
-    gsay_reply(1163, 251);
+    Reply(251);
     if (hacked_system) then begin
-        giq_option(Needed_IQ, 1163, 252, Node020a, 50);
+        giq_option(Needed_IQ, NAME, 252, Node020a, NEUTRAL_REACTION);
     end
     else begin
-        giq_option(Needed_IQ, 1163, 252, Node020, 50);
+        giq_option(Needed_IQ, NAME, 252, Node020, NEUTRAL_REACTION);
     end
-    giq_option(Needed_IQ, 1163, 253, Node026a, 50);
+    giq_option(Needed_IQ, NAME, 253, Node026a, NEUTRAL_REACTION);
 end
 
 procedure Node027
 begin
-    gsay_reply(1163, 254);
-    if (has_skill(dude_obj, 12) >= 75) then begin
-        giq_option(Needed_IQ, 1163, 255, Node027a, 50);
+    Reply(254);
+    if (dude_skill(SKILL_SCIENCE) >= 75) then begin
+        giq_option(Needed_IQ, NAME, 255, Node027a, NEUTRAL_REACTION);
     end
     else begin
-        giq_option(Needed_IQ, 1163, 256, Node020, 50);
+        giq_option(Needed_IQ, NAME, 256, Node020, NEUTRAL_REACTION);
     end
 end
 
 procedure Node028
 begin
-    gsay_reply(1163, 257);
-    if (has_skill(dude_obj, 12) >= 125) then begin
-        giq_option(Needed_IQ, 1163, 258, Node028a, 50);
+    Reply(257);
+    if (dude_skill(SKILL_SCIENCE) >= 125) then begin
+        giq_option(Needed_IQ, NAME, 258, Node028a, NEUTRAL_REACTION);
     end
     else begin
-        giq_option(Needed_IQ, 1163, 259, Node020, 50);
+        giq_option(Needed_IQ, NAME, 259, Node020, NEUTRAL_REACTION);
     end
 end
 
 procedure Node029
 begin
-    gsay_reply(1163, 260);
-    giq_option(Needed_IQ, 1163, 261, Node029a, 50);
+    Reply(260);
+    giq_option(Needed_IQ, NAME, 261, Node029a, NEUTRAL_REACTION);
 end
 
 procedure Node030
 begin
     set_local_var(1, 1);
-    gsay_reply(1163, 262);
-    giq_option(Needed_IQ, 1163, 263, NodeEnd, 50);
+    Reply(262);
+    giq_option(Needed_IQ, NAME, 263, NodeEnd, NEUTRAL_REACTION);
 end
 
 procedure NodeEnd
@@ -712,7 +712,7 @@ end
 procedure Node021a
 begin
     if (obj_is_carrying_obj_pid(dude_obj, 495)) then begin
-        if (cur_map_index == 130) then begin
+        if (CUR_MAP_ENCFITE) then begin
             call Node022();
         end
         else begin
@@ -736,27 +736,27 @@ end
 
 procedure Node026a
 begin
-    if (cur_map_index == 128) then begin
-        set_global_var(569, 2);
+    if (CUR_MAP_ENCDET) then begin
+        set_global_var(GVAR_ENCLAVE_TURRET_DETENTION, 2);
     end
     else begin
-        if (cur_map_index == 130) then begin
-            set_global_var(572, 2);
+        if (CUR_MAP_ENCFITE) then begin
+            set_global_var(GVAR_ENCLAVE_TURRET_MAIN, 2);
         end
         else begin
-            if (cur_map_index == 131) then begin
-                set_global_var(568, 2);
+            if (CUR_MAP_ENCGD) then begin
+                set_global_var(GVAR_ENCLAVE_TURRET_GUARD, 2);
             end
             else begin
-                if (cur_map_index == 132) then begin
-                    set_global_var(571, 2);
+                if (CUR_MAP_ENCPRES) then begin
+                    set_global_var(GVAR_ENCLAVE_TURRET_PRESIDENT, 2);
                 end
                 else begin
-                    if (cur_map_index == 133) then begin
-                        set_global_var(570, 2);
+                    if (CUR_MAP_ENCRCTR) then begin
+                        set_global_var(GVAR_ENCLAVE_TURRET_SCIENCE, 2);
                     end
                     else begin
-                        set_global_var(572, 2);
+                        set_global_var(GVAR_ENCLAVE_TURRET_MAIN, 2);
                     end
                 end
             end
@@ -797,17 +797,17 @@ end
 
 procedure Node020a
 begin
-    gsay_reply(1163, 264 + here_before);
-    giq_option(Needed_IQ, 1163, 237, Node033, 50);
-    giq_option(Needed_IQ, 1163, 248, Node026, 50);
-    giq_option(Needed_IQ, 1163, 234, Node004, 50);
-    giq_option(Needed_IQ, 1163, 235, NodeEnd, 50);
+    Reply(264 + here_before);
+    giq_option(Needed_IQ, NAME, 237, Node033, NEUTRAL_REACTION);
+    giq_option(Needed_IQ, NAME, 248, Node026, NEUTRAL_REACTION);
+    giq_option(Needed_IQ, NAME, 234, Node004, NEUTRAL_REACTION);
+    giq_option(Needed_IQ, NAME, 235, NodeEnd, NEUTRAL_REACTION);
     here_before := 1;
 end
 
 procedure Node030a
 begin
-    if (has_skill(dude_obj, 12) >= 200) then begin
+    if (dude_skill(SKILL_SCIENCE) >= 200) then begin
         call Node020a();
     end
     else begin
@@ -818,7 +818,7 @@ end
 
 procedure Node033
 begin
-    if (cur_map_index == 130) then begin
+    if (CUR_MAP_ENCFITE) then begin
         call Node022();
     end
     else begin
@@ -828,31 +828,30 @@ end
 
 procedure Alarm_Reactive
 begin
-    if (cur_map_index == 128) then begin
-        set_global_var(569, 0);
+    if (CUR_MAP_ENCDET) then begin
+        set_global_var(GVAR_ENCLAVE_TURRET_DETENTION, 0);
     end
     else begin
-        if (cur_map_index == 130) then begin
-            set_global_var(572, 0);
+        if (CUR_MAP_ENCFITE) then begin
+            set_global_var(GVAR_ENCLAVE_TURRET_MAIN, 0);
         end
         else begin
-            if (cur_map_index == 131) then begin
-                set_global_var(568, 0);
+            if (CUR_MAP_ENCGD) then begin
+                set_global_var(GVAR_ENCLAVE_TURRET_GUARD, 0);
             end
             else begin
-                if (cur_map_index == 132) then begin
-                    set_global_var(571, 0);
+                if (CUR_MAP_ENCPRES) then begin
+                    set_global_var(GVAR_ENCLAVE_TURRET_PRESIDENT, 0);
                 end
                 else begin
-                    if (cur_map_index == 133) then begin
-                        set_global_var(570, 0);
+                    if (CUR_MAP_ENCRCTR) then begin
+                        set_global_var(GVAR_ENCLAVE_TURRET_SCIENCE, 0);
                     end
                     else begin
-                        set_global_var(572, 0);
+                        set_global_var(GVAR_ENCLAVE_TURRET_MAIN, 0);
                     end
                 end
             end
         end
     end
 end
-
