diff --git a/Mods/MegaMod/Original/RAYZE.ssl b/Mods/MegaMod/ReDefine/RAYZE.ssl
index c06c74c..7cb9769 100644
--- a/Mods/MegaMod/Original/RAYZE.ssl
+++ b/Mods/MegaMod/ReDefine/RAYZE.ssl
@@ -98,8 +98,8 @@ begin
     set_local_var(2, 2);
     Crypt_Pointer := self_obj;
     if (local_var(0) == 0) then begin
-        critter_add_trait(self_obj, 1, 6, 136);
-        critter_add_trait(self_obj, 1, 5, 22);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 136);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_TOUGH_GUARD);
         set_map_var(7, map_var(7) + 1);
         set_local_var(0, 1);
     end
@@ -143,22 +143,22 @@ procedure damage_p_proc
 begin
     if (source_obj == dude_obj) then begin
         party_remove(self_obj);
-        critter_add_trait(self_obj, 1, 6, 136);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 136);
         set_map_var(10, 1);
-        attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+        attack(dude_obj);
     end
 end
 
 procedure look_at_p_proc
 begin
     script_overrides;
-    display_msg(message_str(1503, 100));
+    display_mstr(100);
 end
 
 procedure description_p_proc
 begin
     script_overrides;
-    display_msg(message_str(1503, 100));
+    display_mstr(100);
 end
 
 procedure critter_p_proc
@@ -166,29 +166,29 @@ begin
     variable LVar0 := 0;
     variable LVar1 := 0;
     if (local_var(6) == 0) then begin
-        if (has_trait(1, self_obj, 6) == 0) then begin
+        if (has_trait(TRAIT_OBJECT, self_obj, OBJECT_TEAM_NUM) == 0) then begin
             party_remove(self_obj);
         end
-        critter_add_trait(self_obj, 1, 6, 136);
-        critter_add_trait(self_obj, 1, 5, 22);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 136);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_TOUGH_GUARD);
     end
     if (map_var(10) == 1) then begin
-        if (has_trait(1, self_obj, 6) == 0) then begin
+        if (has_trait(TRAIT_OBJECT, self_obj, OBJECT_TEAM_NUM) == 0) then begin
             party_remove(self_obj);
         end
-        critter_add_trait(self_obj, 1, 6, 136);
-        critter_add_trait(self_obj, 1, 5, 22);
-        attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 136);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_TOUGH_GUARD);
+        attack(dude_obj);
     end
     if (map_var(5) == 3) then begin
-        attack_complex(Fool_Pointer, 0, 1, 0, 0, 30000, 0, 0);
+        attack(Fool_Pointer);
     end
     Crypt_Pointer := self_obj;
     LVar0 := (self_obj == Fool_Pointer) or (self_obj == Crypt_Pointer) or (self_obj == Phrax_Pointer) or (self_obj == Rock_Pointer);
     if (Fool_Taunt) then begin
         Fool_Taunt := 0;
         if (random(1, 5) == 1) then begin
-            float_msg(self_obj, message_str(964, random(10200, 10218)), -1);
+            float_msg(self_obj, message_str(SCRIPT_KLATRAP, random(10200, 10218)), FLOAT_MSG_SEQUENTIAL);
         end
     end
     else begin
@@ -198,7 +198,7 @@ begin
         else begin
             if (GVar3) then begin
                 GVar3 := 0;
-                attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+                attack(dude_obj);
             end
             else begin
                 if (not(LVar0)) then begin
@@ -211,8 +211,8 @@ begin
         end
     end
     if ((map_var(5) == 1) and (tile_num(self_obj) != 20725)) then begin
-        if (global_var(804) == 1) then begin
-            critter_add_trait(self_obj, 1, 6, 0);
+        if (global_var(GVAR_DOG_EMPATHY) == 1) then begin
+            critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 0);
         end
         animate_move_obj_to_tile(self_obj, 20725, 0);
         if (GVar1 == 0) then begin
@@ -228,16 +228,16 @@ begin
             if ((map_var(2) >= 1) and (tile_num(self_obj) != 27504)) then begin
                 if (GVar0 == 0) then begin
                     GVar0 := 1;
-                    float_msg(self_obj, message_str(964, random(10300, 1310)), 6);
+                    float_msg(self_obj, message_str(SCRIPT_KLATRAP, random(10300, 1310)), FLOAT_MSG_NEAR_WHITE);
                 end
                 animate_move_obj_to_tile(self_obj, 27504, 0);
             end
             else begin
                 if ((map_var(2) >= 1) and (tile_num(self_obj) == 27504)) then begin
-                    if (has_trait(1, self_obj, 6) == 0) then begin
+                    if (has_trait(TRAIT_OBJECT, self_obj, OBJECT_TEAM_NUM) == 0) then begin
                         party_remove(self_obj);
                     end
-                    critter_add_trait(self_obj, 1, 6, 136);
+                    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 136);
                     set_obj_visibility(self_obj, 1);
                 end
             end
@@ -253,7 +253,7 @@ begin
     variable LVar0 := 0;
     if (fixed_param == 1) then begin
         Crypt_Taunt := 1;
-        float_msg(self_obj, message_str(964, random(10200, 10218)), -1);
+        float_msg(self_obj, message_str(SCRIPT_KLATRAP, random(10200, 10218)), FLOAT_MSG_SEQUENTIAL);
         if (LVar0 == 0) then begin
             add_timer_event(self_obj, 100, 2);
         end
@@ -276,55 +276,55 @@ begin
         set_map_var(5, 2);
     end
     if (source_obj == dude_obj) then begin
-        set_global_var(4, global_var(4) + 1);
-        if (metarule(51, self_obj) == 2) then begin
+        set_global_var(GVAR_BAD_MONSTER, global_var(GVAR_BAD_MONSTER) + 1);
+        if (critter_kill_type(self_obj) == 2) then begin
             if (source_obj == dude_obj) then begin
-                set_global_var(2, 0);
-                set_global_var(1, global_var(1) + 1);
-                set_global_var(0, global_var(0) + -15);
+                set_global_var(GVAR_CHAMPION_REPUTATION, 0);
+                set_global_var(GVAR_CHILDKILLER_REPUTATION, global_var(GVAR_CHILDKILLER_REPUTATION) + 1);
+                set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + -15);
                 debug_msg("Player gains " + -15 + " Karma Points.");
-                set_global_var(37, 0);
-                set_global_var(38, 0);
-                set_global_var(39, 0);
-                set_global_var(40, 0);
-                set_global_var(41, 0);
-                set_global_var(42, 0);
-                set_global_var(43, 0);
-                set_global_var(44, 0);
-                set_global_var(45, 0);
-                if (global_var(0) >= 1000) then begin
-                    set_global_var(37, 1);
+                set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+                set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+                set_global_var(GVAR_KARMA_DEFENDER, 0);
+                set_global_var(GVAR_KARMA_WANDERER, 0);
+                set_global_var(GVAR_KARMA_BETRAYER, 0);
+                set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+                set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+                set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+                    set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
                 end
                 else begin
-                    if (global_var(0) >= 750) then begin
-                        set_global_var(38, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
                     end
                     else begin
-                        if (global_var(0) >= 500) then begin
-                            set_global_var(39, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                            set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                         end
                         else begin
-                            if (global_var(0) >= 250) then begin
-                                set_global_var(40, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                                set_global_var(GVAR_KARMA_DEFENDER, 1);
                             end
                             else begin
-                                if (global_var(0) > -250) then begin
-                                    set_global_var(41, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                                    set_global_var(GVAR_KARMA_WANDERER, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -500) then begin
-                                        set_global_var(42, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                        set_global_var(GVAR_KARMA_BETRAYER, 1);
                                     end
                                     else begin
-                                        if (global_var(0) > -750) then begin
-                                            set_global_var(43, 1);
+                                        if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                            set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                         end
                                         else begin
-                                            if (global_var(0) > -1000) then begin
-                                                set_global_var(44, 1);
+                                            if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                                set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                             end
                                             else begin
-                                                set_global_var(45, 1);
+                                                set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                             end
                                         end
                                     end
@@ -333,62 +333,62 @@ begin
                         end
                     end
                 end
-                set_global_var(808, global_var(808) + -5);
+                set_global_var(GVAR_GANG_2, global_var(GVAR_GANG_2) + -5);
                 debug_msg("Added " + -5 + " to Town Rep");
             end
         end
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(4) > (3 * global_var(5))) or (global_var(2) == 1)) and (global_var(1) == 0)) then begin
-            set_global_var(2, 1);
-            set_global_var(3, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(4) > (3 * global_var(GVAR_GOOD_MONSTER))) or (global_var(GVAR_CHAMPION_REPUTATION) == 1)) and (global_var(GVAR_CHILDKILLER_REPUTATION) == 0)) then begin
+            set_global_var(GVAR_CHAMPION_REPUTATION, 1);
+            set_global_var(GVAR_BERSERKER_REPUTATION, 0);
         end
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(5) > (2 * global_var(4))) or (global_var(3) == 1))) then begin
-            set_global_var(3, 1);
-            set_global_var(2, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
+            set_global_var(GVAR_BERSERKER_REPUTATION, 1);
+            set_global_var(GVAR_CHAMPION_REPUTATION, 0);
         end
-        set_global_var(0, global_var(0) + 5);
+        set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + 5);
         debug_msg("Player gains " + 5 + " Karma Points.");
-        set_global_var(37, 0);
-        set_global_var(38, 0);
-        set_global_var(39, 0);
-        set_global_var(40, 0);
-        set_global_var(41, 0);
-        set_global_var(42, 0);
-        set_global_var(43, 0);
-        set_global_var(44, 0);
-        set_global_var(45, 0);
-        if (global_var(0) >= 1000) then begin
-            set_global_var(37, 1);
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+        set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+        set_global_var(GVAR_KARMA_DEFENDER, 0);
+        set_global_var(GVAR_KARMA_WANDERER, 0);
+        set_global_var(GVAR_KARMA_BETRAYER, 0);
+        set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+        set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+            set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
         end
         else begin
-            if (global_var(0) >= 750) then begin
-                set_global_var(38, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
             end
             else begin
-                if (global_var(0) >= 500) then begin
-                    set_global_var(39, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                 end
                 else begin
-                    if (global_var(0) >= 250) then begin
-                        set_global_var(40, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                        set_global_var(GVAR_KARMA_DEFENDER, 1);
                     end
                     else begin
-                        if (global_var(0) > -250) then begin
-                            set_global_var(41, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                            set_global_var(GVAR_KARMA_WANDERER, 1);
                         end
                         else begin
-                            if (global_var(0) > -500) then begin
-                                set_global_var(42, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                set_global_var(GVAR_KARMA_BETRAYER, 1);
                             end
                             else begin
-                                if (global_var(0) > -750) then begin
-                                    set_global_var(43, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -1000) then begin
-                                        set_global_var(44, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                     end
                                     else begin
-                                        set_global_var(45, 1);
+                                        set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                     end
                                 end
                             end
@@ -397,7 +397,7 @@ begin
                 end
             end
         end
-        set_global_var(808, global_var(808) + 3);
+        set_global_var(GVAR_GANG_2, global_var(GVAR_GANG_2) + 3);
         debug_msg("Added " + 3 + " to Town Rep");
     end
 end
@@ -408,76 +408,76 @@ end
 
 procedure Node000
 begin
-    float_msg(self_obj, message_str(1503, 101), 0);
+    float_msg(self_obj, mstr(101), FLOAT_MSG_NORMAL);
     call Combat();
 end
 
 procedure Node001
 begin
-    gsay_reply(1503, 102);
+    Reply(102);
     if (Rock_Invite) then begin
-        giq_option(4, 1503, 103, Node002, 50);
+        giq_option(4, NAME, 103, Node002, NEUTRAL_REACTION);
     end
-    giq_option(4, 1503, 104, Node007, 50);
-    giq_option(4, 1503, 105, Node014, 50);
-    giq_option(4, 1503, 106, Node015, 50);
+    giq_option(4, NAME, 104, Node007, NEUTRAL_REACTION);
+    giq_option(4, NAME, 105, Node014, NEUTRAL_REACTION);
+    giq_option(4, NAME, 106, Node015, NEUTRAL_REACTION);
 end
 
 procedure Node002
 begin
-    gsay_reply(1503, 107);
-    giq_option(4, 1503, 108, Node003, 50);
-    giq_option(4, 1503, 109, Node004, 50);
+    Reply(107);
+    giq_option(4, NAME, 108, Node003, NEUTRAL_REACTION);
+    giq_option(4, NAME, 109, Node004, NEUTRAL_REACTION);
 end
 
 procedure Node003
 begin
-    gsay_reply(1503, 110);
-    giq_option(4, 1503, 136, node999, 50);
+    Reply(110);
+    giq_option(4, NAME, 136, node999, NEUTRAL_REACTION);
 end
 
 procedure Node004
 begin
-    gsay_reply(1503, 111);
-    giq_option(4, 1503, 112, Node005, 50);
-    giq_option(4, 1503, 113, Node006, 50);
+    Reply(111);
+    giq_option(4, NAME, 112, Node005, NEUTRAL_REACTION);
+    giq_option(4, NAME, 113, Node006, NEUTRAL_REACTION);
 end
 
 procedure Node005
 begin
-    gsay_reply(1503, 114);
-    giq_option(4, 1503, 115, Node005a, 50);
-    giq_option(4, 1503, 116, Node006, 50);
+    Reply(114);
+    giq_option(4, NAME, 115, Node005a, NEUTRAL_REACTION);
+    giq_option(4, NAME, 116, Node006, NEUTRAL_REACTION);
 end
 
 procedure Node005a
 begin
     set_map_var(12, 1);
-    set_global_var(804, 1);
+    set_global_var(GVAR_DOG_EMPATHY, 1);
 end
 
 procedure Node006
 begin
-    gsay_reply(1503, 117);
-    giq_option(4, 1503, 136, node999, 50);
+    Reply(117);
+    giq_option(4, NAME, 136, node999, NEUTRAL_REACTION);
 end
 
 procedure Node007
 begin
-    gsay_reply(1503, 118);
-    giq_option(4, 1503, 119, Node008, 50);
+    Reply(118);
+    giq_option(4, NAME, 119, Node008, NEUTRAL_REACTION);
 end
 
 procedure Node008
 begin
-    gsay_reply(1503, 120);
-    giq_option(4, 1503, 121, Node005, 50);
-    giq_option(4, 1503, 122, Node006, 50);
+    Reply(120);
+    giq_option(4, NAME, 121, Node005, NEUTRAL_REACTION);
+    giq_option(4, NAME, 122, Node006, NEUTRAL_REACTION);
 end
 
 procedure Node009
 begin
-    float_msg(self_obj, message_str(1503, 123), 0);
+    float_msg(self_obj, mstr(123), FLOAT_MSG_NORMAL);
 end
 
 procedure Node010
@@ -487,14 +487,14 @@ begin
     end
     else begin
         if (map_var(9) == 2) then begin
-            gsay_reply(1503, 124);
-            giq_option(4, 1503, 125, Node010a, 50);
-            giq_option(4, 1503, 126, Node010b, 50);
-            giq_option(4, 1503, 127, Node013, 50);
+            Reply(124);
+            giq_option(4, NAME, 125, Node010a, NEUTRAL_REACTION);
+            giq_option(4, NAME, 126, Node010b, NEUTRAL_REACTION);
+            giq_option(4, NAME, 127, Node013, NEUTRAL_REACTION);
         end
         else begin
-            gsay_reply(1503, 149);
-            giq_option(4, 1503, 150, node999, 50);
+            Reply(149);
+            giq_option(4, NAME, 150, node999, NEUTRAL_REACTION);
         end
     end
 end
@@ -532,70 +532,70 @@ end
 procedure Node011
 begin
     set_map_var(5, 1);
-    gsay_reply(1503, 128);
-    giq_option(4, 1503, 136, node999, 50);
+    Reply(128);
+    giq_option(4, NAME, 136, node999, NEUTRAL_REACTION);
 end
 
 procedure Node012
 begin
     set_map_var(5, 1);
-    gsay_reply(1503, 129);
-    giq_option(4, 1503, 136, node999, 50);
+    Reply(129);
+    giq_option(4, NAME, 136, node999, NEUTRAL_REACTION);
 end
 
 procedure Node012a
 begin
     set_local_var(4, 1);
-    gsay_reply(1503, 146);
-    giq_option(4, 1503, 147, node999, 50);
-    giq_option(4, 1503, 148, node999, 50);
+    Reply(146);
+    giq_option(4, NAME, 147, node999, NEUTRAL_REACTION);
+    giq_option(4, NAME, 148, node999, NEUTRAL_REACTION);
 end
 
 procedure Node013
 begin
-    gsay_reply(1503, 130);
-    giq_option(4, 1503, 136, node999, 50);
+    Reply(130);
+    giq_option(4, NAME, 136, node999, NEUTRAL_REACTION);
 end
 
 procedure Node014
 begin
-    gsay_reply(1503, 131);
+    Reply(131);
     if (Rock_Invite) then begin
-        giq_option(4, 1503, 132, Node002, 50);
+        giq_option(4, NAME, 132, Node002, NEUTRAL_REACTION);
     end
-    giq_option(4, 1503, 133, Node007, 50);
-    giq_option(4, 1503, 134, Node015, 50);
+    giq_option(4, NAME, 133, Node007, NEUTRAL_REACTION);
+    giq_option(4, NAME, 134, Node015, NEUTRAL_REACTION);
 end
 
 procedure Node015
 begin
-    gsay_reply(1503, 135);
-    giq_option(4, 1503, 136, node999, 50);
+    Reply(135);
+    giq_option(4, NAME, 136, node999, NEUTRAL_REACTION);
 end
 
 procedure Node016
 begin
     GVar6 := 1;
-    gsay_reply(1503, 142);
-    giq_option(4, 1503, 136, node999, 50);
+    Reply(142);
+    giq_option(4, NAME, 136, node999, NEUTRAL_REACTION);
 end
 
 procedure Node017
 begin
-    gsay_reply(1503, 143);
-    giq_option(4, 1503, 136, node999, 50);
+    Reply(143);
+    giq_option(4, NAME, 136, node999, NEUTRAL_REACTION);
     call Combat();
 end
 
 procedure Node018
 begin
-    float_msg(self_obj, message_str(1503, 144), 0);
+    float_msg(self_obj, mstr(144), FLOAT_MSG_NORMAL);
 end
 
 procedure Node019
 begin
-    gsay_reply(1503, 145);
-    giq_option(4, 1503, 136, node999, 50);
+    Reply(145);
+    giq_option(4, NAME, 136, node999, NEUTRAL_REACTION);
 end
 
 procedure Combat
@@ -611,13 +611,12 @@ procedure Start_War
 begin
     if (fixed_param == 1) then begin
         Crypt_Taunt := 1;
-        float_msg(self_obj, message_str(964, random(10200, 10218)), -1);
+        float_msg(self_obj, message_str(SCRIPT_KLATRAP, random(10200, 10218)), FLOAT_MSG_SEQUENTIAL);
         add_timer_event(self_obj, 200, 2);
     end
     else begin
         if (fixed_param == 2) then begin
-            attack_complex(Fool_Pointer, 0, 1, 0, 0, 30000, 0, 0);
+            attack(Fool_Pointer);
         end
     end
 end
-
