diff --git a/Mods/MegaMod/Original/RCHORTON.ssl b/Mods/MegaMod/ReDefine/RCHORTON.ssl
index 48255b3..38aa82f 100644
--- a/Mods/MegaMod/Original/RCHORTON.ssl
+++ b/Mods/MegaMod/ReDefine/RCHORTON.ssl
@@ -170,7 +170,7 @@ procedure critter_p_proc
 begin
     if ((local_var(5) == 2) and obj_can_see_obj(self_obj, dude_obj)) then begin
         set_local_var(5, 1);
-        attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+        attack(dude_obj);
     end
 end
 
@@ -184,8 +184,8 @@ end
 procedure talk_p_proc
 begin
     in_dialog := 1;
-    if (global_var(684) == 5) then begin
-        float_msg(self_obj, message_str(1347, 400), 10);
+    if (global_var(GVAR_DUDE_WARRIOR) == 5) then begin
+        float_msg(self_obj, mstr(400), FLOAT_MSG_GREY);
     end
     else begin
         if (local_var(4) == 0) then begin
@@ -196,7 +196,7 @@ begin
             end_dialogue;
         end
         else begin
-            if (global_var(684) < 2) then begin
+            if (global_var(GVAR_DUDE_WARRIOR) < 2) then begin
                 start_gdialog(1347, self_obj, 4, -1, -1);
                 gsay_start;
                 call Node008();
@@ -218,59 +218,59 @@ end
 procedure destroy_p_proc
 begin
     if (source_obj == dude_obj) then begin
-        set_global_var(5, global_var(5) + 1);
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(4) > (3 * global_var(5))) or (global_var(2) == 1)) and (global_var(1) == 0)) then begin
-            set_global_var(2, 1);
-            set_global_var(3, 0);
+        set_global_var(GVAR_GOOD_MONSTER, global_var(GVAR_GOOD_MONSTER) + 1);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(4) > (3 * global_var(GVAR_GOOD_MONSTER))) or (global_var(GVAR_CHAMPION_REPUTATION) == 1)) and (global_var(GVAR_CHILDKILLER_REPUTATION) == 0)) then begin
+            set_global_var(GVAR_CHAMPION_REPUTATION, 1);
+            set_global_var(GVAR_BERSERKER_REPUTATION, 0);
         end
-        if (metarule(51, self_obj) == 2) then begin
+        if (critter_kill_type(self_obj) == 2) then begin
             if (source_obj == dude_obj) then begin
-                set_global_var(2, 0);
-                set_global_var(1, global_var(1) + 1);
-                set_global_var(0, global_var(0) + -15);
+                set_global_var(GVAR_CHAMPION_REPUTATION, 0);
+                set_global_var(GVAR_CHILDKILLER_REPUTATION, global_var(GVAR_CHILDKILLER_REPUTATION) + 1);
+                set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + -15);
                 debug_msg("Player gains " + -15 + " Karma Points.");
-                set_global_var(37, 0);
-                set_global_var(38, 0);
-                set_global_var(39, 0);
-                set_global_var(40, 0);
-                set_global_var(41, 0);
-                set_global_var(42, 0);
-                set_global_var(43, 0);
-                set_global_var(44, 0);
-                set_global_var(45, 0);
-                if (global_var(0) >= 1000) then begin
-                    set_global_var(37, 1);
+                set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+                set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+                set_global_var(GVAR_KARMA_DEFENDER, 0);
+                set_global_var(GVAR_KARMA_WANDERER, 0);
+                set_global_var(GVAR_KARMA_BETRAYER, 0);
+                set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+                set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+                set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+                    set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
                 end
                 else begin
-                    if (global_var(0) >= 750) then begin
-                        set_global_var(38, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
                     end
                     else begin
-                        if (global_var(0) >= 500) then begin
-                            set_global_var(39, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                            set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                         end
                         else begin
-                            if (global_var(0) >= 250) then begin
-                                set_global_var(40, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                                set_global_var(GVAR_KARMA_DEFENDER, 1);
                             end
                             else begin
-                                if (global_var(0) > -250) then begin
-                                    set_global_var(41, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                                    set_global_var(GVAR_KARMA_WANDERER, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -500) then begin
-                                        set_global_var(42, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                        set_global_var(GVAR_KARMA_BETRAYER, 1);
                                     end
                                     else begin
-                                        if (global_var(0) > -750) then begin
-                                            set_global_var(43, 1);
+                                        if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                            set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                         end
                                         else begin
-                                            if (global_var(0) > -1000) then begin
-                                                set_global_var(44, 1);
+                                            if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                                set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                             end
                                             else begin
-                                                set_global_var(45, 1);
+                                                set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                             end
                                         end
                                     end
@@ -279,58 +279,58 @@ begin
                         end
                     end
                 end
-                set_global_var(56, global_var(56) + -8);
+                set_global_var(GVAR_TOWN_REP_REDDING, global_var(GVAR_TOWN_REP_REDDING) + -8);
                 debug_msg("Added " + -8 + " to Town Rep");
             end
         end
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(5) > (2 * global_var(4))) or (global_var(3) == 1))) then begin
-            set_global_var(3, 1);
-            set_global_var(2, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
+            set_global_var(GVAR_BERSERKER_REPUTATION, 1);
+            set_global_var(GVAR_CHAMPION_REPUTATION, 0);
         end
-        set_global_var(0, global_var(0) + -10);
+        set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + -10);
         debug_msg("Player gains " + -10 + " Karma Points.");
-        set_global_var(37, 0);
-        set_global_var(38, 0);
-        set_global_var(39, 0);
-        set_global_var(40, 0);
-        set_global_var(41, 0);
-        set_global_var(42, 0);
-        set_global_var(43, 0);
-        set_global_var(44, 0);
-        set_global_var(45, 0);
-        if (global_var(0) >= 1000) then begin
-            set_global_var(37, 1);
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+        set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+        set_global_var(GVAR_KARMA_DEFENDER, 0);
+        set_global_var(GVAR_KARMA_WANDERER, 0);
+        set_global_var(GVAR_KARMA_BETRAYER, 0);
+        set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+        set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+            set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
         end
         else begin
-            if (global_var(0) >= 750) then begin
-                set_global_var(38, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
             end
             else begin
-                if (global_var(0) >= 500) then begin
-                    set_global_var(39, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                 end
                 else begin
-                    if (global_var(0) >= 250) then begin
-                        set_global_var(40, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                        set_global_var(GVAR_KARMA_DEFENDER, 1);
                     end
                     else begin
-                        if (global_var(0) > -250) then begin
-                            set_global_var(41, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                            set_global_var(GVAR_KARMA_WANDERER, 1);
                         end
                         else begin
-                            if (global_var(0) > -500) then begin
-                                set_global_var(42, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                set_global_var(GVAR_KARMA_BETRAYER, 1);
                             end
                             else begin
-                                if (global_var(0) > -750) then begin
-                                    set_global_var(43, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -1000) then begin
-                                        set_global_var(44, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                     end
                                     else begin
-                                        set_global_var(45, 1);
+                                        set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                     end
                                 end
                             end
@@ -339,7 +339,7 @@ begin
                 end
             end
         end
-        set_global_var(56, global_var(56) + -5);
+        set_global_var(GVAR_TOWN_REP_REDDING, global_var(GVAR_TOWN_REP_REDDING) + -5);
         debug_msg("Added " + -5 + " to Town Rep");
     end
 end
@@ -347,18 +347,18 @@ end
 procedure look_at_p_proc
 begin
     script_overrides;
-    if (global_var(684) >= 1) then begin
-        display_msg(message_str(1347, 101));
+    if (global_var(GVAR_DUDE_WARRIOR) >= 1) then begin
+        display_mstr(101);
     end
     else begin
-        display_msg(message_str(1347, 100));
+        display_mstr(100);
     end
 end
 
 procedure description_p_proc
 begin
     script_overrides;
-    display_msg(message_str(1347, 102));
+    display_mstr(102);
 end
 
 procedure damage_p_proc
@@ -368,8 +368,8 @@ end
 procedure map_enter_p_proc
 begin
     Only_Once := 0;
-    critter_add_trait(self_obj, 1, 6, 152);
-    critter_add_trait(self_obj, 1, 5, 14);
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 152);
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_PEASANTS);
 end
 
 procedure push_p_proc
@@ -388,155 +388,154 @@ end
 
 procedure Node001
 begin
-    if (get_critter_stat(dude_obj, 34) == 0) then begin
-        gsay_reply(1347, 110);
+    if (dude_gender == 0) then begin
+        Reply(110);
     end
     else begin
-        gsay_reply(1347, 111);
+        Reply(111);
     end
-    giq_option(4, 1347, 112, Node002, 50);
-    giq_option(-3, 1347, 114, Node002, 50);
-    giq_option(4, 1347, 116, Node999, 50);
-    giq_option(-3, 1347, 118, Node999, 50);
+    giq_option(4, NAME, 112, Node002, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 114, Node002, NEUTRAL_REACTION);
+    giq_option(4, NAME, 116, Node999, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 118, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node002
 begin
-    gsay_reply(1347, 120);
-    giq_option(4, 1347, 122, Node003, 50);
-    giq_option(-3, 1347, 124, Node003, 50);
-    giq_option(4, 1347, 126, Node999, 50);
-    giq_option(-3, 1347, 128, Node999, 50);
+    Reply(120);
+    giq_option(4, NAME, 122, Node003, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 124, Node003, NEUTRAL_REACTION);
+    giq_option(4, NAME, 126, Node999, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 128, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node003
 begin
-    set_global_var(684, 1);
-    gsay_reply(1347, 130);
-    giq_option(4, 1347, message_str(1347, 132) + obj_name(dude_obj) + message_str(1347, 133), Node004, 50);
-    giq_option(-3, 1347, message_str(1347, 136) + obj_name(dude_obj) + message_str(1347, 134), Node004, 50);
-    giq_option(4, 1347, message_str(1347, 132) + obj_name(dude_obj) + message_str(1347, 137), Node005, 50);
-    giq_option(-3, 1347, message_str(1347, 136) + obj_name(dude_obj) + message_str(1347, 138), Node005, 50);
-    giq_option(4, 1347, 140, Node999, 50);
-    giq_option(-3, 1347, 142, Node999, 50);
+    set_global_var(GVAR_DUDE_WARRIOR, 1);
+    Reply(130);
+    giq_option(4, NAME, mstr(132) + dude_name + mstr(133), Node004, NEUTRAL_REACTION);
+    giq_option(-3, NAME, mstr(136) + dude_name + mstr(134), Node004, NEUTRAL_REACTION);
+    giq_option(4, NAME, mstr(132) + dude_name + mstr(137), Node005, NEUTRAL_REACTION);
+    giq_option(-3, NAME, mstr(136) + dude_name + mstr(138), Node005, NEUTRAL_REACTION);
+    giq_option(4, NAME, 140, Node999, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 142, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node004
 begin
     set_local_var(7, 1);
-    set_global_var(684, 2);
-    gsay_reply(1347, 144);
+    set_global_var(GVAR_DUDE_WARRIOR, 2);
+    Reply(144);
     if (local_var(6) == 0) then begin
-        giq_option(4, 1347, 146, Node005, 50);
-        giq_option(-3, 1347, 148, Node005, 50);
+        giq_option(4, NAME, 146, Node005, NEUTRAL_REACTION);
+        giq_option(-3, NAME, 148, Node005, NEUTRAL_REACTION);
     end
     else begin
-        giq_option(4, 1347, 150, Node006, 50);
-        giq_option(-3, 1347, 152, Node006, 50);
+        giq_option(4, NAME, 150, Node006, NEUTRAL_REACTION);
+        giq_option(-3, NAME, 152, Node006, NEUTRAL_REACTION);
     end
-    giq_option(4, 1347, 154, Node999, 50);
-    giq_option(-3, 1347, 156, Node999, 50);
+    giq_option(4, NAME, 154, Node999, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 156, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node005
 begin
-    gsay_reply(1347, 158);
-    giq_option(4, 1347, 160, Node005a, 50);
-    giq_option(-3, 1347, 162, Node005a, 50);
-    giq_option(4, 1347, 164, Node999, 50);
-    giq_option(-3, 1347, 166, Node999, 50);
+    Reply(158);
+    giq_option(4, NAME, 160, Node005a, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 162, Node005a, NEUTRAL_REACTION);
+    giq_option(4, NAME, 164, Node999, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 166, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node005a
 begin
-    gsay_reply(1347, 168);
-    giq_option(4, 1347, 170, Node005b, 50);
-    giq_option(-3, 1347, 172, Node005b, 50);
-    giq_option(4, 1347, 174, Node999, 50);
-    giq_option(-3, 1347, 176, Node999, 50);
+    Reply(168);
+    giq_option(4, NAME, 170, Node005b, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 172, Node005b, NEUTRAL_REACTION);
+    giq_option(4, NAME, 174, Node999, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 176, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node005b
 begin
     set_local_var(6, 1);
-    gsay_reply(1347, 178);
+    Reply(178);
     if (local_var(7) == 0) then begin
-        giq_option(4, 1347, 180, Node004, 50);
-        giq_option(-3, 1347, 182, Node004, 50);
+        giq_option(4, NAME, 180, Node004, NEUTRAL_REACTION);
+        giq_option(-3, NAME, 182, Node004, NEUTRAL_REACTION);
     end
     else begin
-        giq_option(4, 1347, 184, Node006, 50);
-        giq_option(-3, 1347, 186, Node006, 50);
+        giq_option(4, NAME, 184, Node006, NEUTRAL_REACTION);
+        giq_option(-3, NAME, 186, Node006, NEUTRAL_REACTION);
     end
-    giq_option(-3, 1347, 190, Node007, 50);
-    giq_option(4, 1347, 192, Node999, 50);
-    giq_option(-3, 1347, 194, Node999, 50);
+    giq_option(-3, NAME, 190, Node007, NEUTRAL_REACTION);
+    giq_option(4, NAME, 192, Node999, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 194, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node006
 begin
-    if (get_critter_stat(dude_obj, 34) == 0) then begin
-        gsay_reply(1347, 196);
+    if (dude_gender == 0) then begin
+        Reply(196);
     end
     else begin
-        gsay_reply(1347, 198);
+        Reply(198);
     end
-    giq_option(4, 1347, 200, Node999, 50);
-    giq_option(-3, 1347, 202, Node999, 50);
-    giq_option(4, 1347, 204, Node007, 50);
-    giq_option(-3, 1347, 206, Node007, 50);
+    giq_option(4, NAME, 200, Node999, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 202, Node999, NEUTRAL_REACTION);
+    giq_option(4, NAME, 204, Node007, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 206, Node007, NEUTRAL_REACTION);
 end
 
 procedure Node007
 begin
-    gsay_reply(1347, 208);
-    giq_option(4, 1347, 210, Node999, 50);
-    giq_option(-3, 1347, 212, Node999, 50);
-    giq_option(4, 1347, 214, Node999, 50);
-    giq_option(-3, 1347, 216, Node999, 50);
+    Reply(208);
+    giq_option(4, NAME, 210, Node999, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 212, Node999, NEUTRAL_REACTION);
+    giq_option(4, NAME, 214, Node999, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 216, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node008
 begin
-    gsay_reply(1347, 218);
-    giq_option(4, 1347, 220, Node002, 50);
-    giq_option(-3, 1347, 222, Node002, 50);
-    giq_option(4, 1347, 224, Node999, 50);
-    giq_option(-3, 1347, 226, Node999, 50);
+    Reply(218);
+    giq_option(4, NAME, 220, Node002, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 222, Node002, NEUTRAL_REACTION);
+    giq_option(4, NAME, 224, Node999, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 226, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node009
 begin
-    gsay_reply(1347, 300);
-    if (global_var(684) != 4) then begin
-        giq_option(4, 1347, 302, Node999, 50);
-        giq_option(-3, 1347, 304, Node999, 50);
+    Reply(300);
+    if (global_var(GVAR_DUDE_WARRIOR) != 4) then begin
+        giq_option(4, NAME, 302, Node999, NEUTRAL_REACTION);
+        giq_option(-3, NAME, 304, Node999, NEUTRAL_REACTION);
     end
     else begin
-        giq_option(4, 1347, 306, Node010, 50);
-        giq_option(-3, 1347, 308, Node010, 50);
+        giq_option(4, NAME, 306, Node010, NEUTRAL_REACTION);
+        giq_option(-3, NAME, 308, Node010, NEUTRAL_REACTION);
     end
 end
 
 procedure Node010
 begin
-    gsay_reply(1347, 310);
-    giq_option(4, 1347, 312, Node999, 50);
-    giq_option(4, 1347, 314, Node011, 50);
-    giq_option(-3, 1347, 316, Node011, 50);
+    Reply(310);
+    giq_option(4, NAME, 312, Node999, NEUTRAL_REACTION);
+    giq_option(4, NAME, 314, Node011, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 316, Node011, NEUTRAL_REACTION);
 end
 
 procedure Node011
 begin
-    set_global_var(684, 5);
-    gsay_reply(1347, 318);
-    giq_option(4, 1347, 320, Node012, 50);
-    giq_option(-3, 1347, 322, Node012, 50);
+    set_global_var(GVAR_DUDE_WARRIOR, 5);
+    Reply(318);
+    giq_option(4, NAME, 320, Node012, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 322, Node012, NEUTRAL_REACTION);
 end
 
 procedure Node012
 begin
-    gsay_reply(1347, 324);
-    giq_option(1, 1347, 326, Node999, 50);
+    Reply(324);
+    giq_option(1, NAME, 326, Node999, NEUTRAL_REACTION);
 end
-
