diff --git a/Mods/MegaMod/Original/REDMTUN.ssl b/Mods/MegaMod/ReDefine/REDMTUN.ssl
index 090b209..8b4c9a5 100644
--- a/Mods/MegaMod/Original/REDMTUN.ssl
+++ b/Mods/MegaMod/ReDefine/REDMTUN.ssl
@@ -138,19 +138,18 @@ end
 procedure map_enter_p_proc
 begin
     set_light_level(50);
-    if (global_var(27) == 1) then begin
+    if (global_var(GVAR_LOAD_MAP_INDEX) == 1) then begin
         override_map_start(16874 % 200, 16874 / 200, 0, 5);
     end
     else begin
-        if (global_var(27) == 2) then begin
+        if (global_var(GVAR_LOAD_MAP_INDEX) == 2) then begin
             override_map_start(13075 % 200, 13075 / 200, 0, 2);
         end
     end
-    set_global_var(27, 0);
+    set_global_var(GVAR_LOAD_MAP_INDEX, 0);
 end
 
 procedure map_update_p_proc
 begin
     set_light_level(50);
 end
-
