diff --git a/Mods/MegaMod/Original/RTRIBC5A.ssl b/Mods/MegaMod/ReDefine/RTRIBC5A.ssl
index d6c50aa..4c6e597 100644
--- a/Mods/MegaMod/Original/RTRIBC5A.ssl
+++ b/Mods/MegaMod/ReDefine/RTRIBC5A.ssl
@@ -22,13 +22,13 @@ end
 procedure look_at_p_proc
 begin
     script_overrides;
-    display_msg(message_str(2300, 100));
+    display_mstr(100);
 end
 
 procedure description_p_proc
 begin
     script_overrides;
-    display_msg(message_str(2300, 101));
+    display_mstr(101);
     if (map_var(1) == 0) then begin
         set_map_var(1, 1);
     end
@@ -61,4 +61,3 @@ end
 procedure timed_event_p_proc
 begin
 end
-
