diff --git a/Mods/MegaMod/Original/RTRIBEM1.ssl b/Mods/MegaMod/ReDefine/RTRIBEM1.ssl
index 3794467..02f9d6f 100644
--- a/Mods/MegaMod/Original/RTRIBEM1.ssl
+++ b/Mods/MegaMod/ReDefine/RTRIBEM1.ssl
@@ -15,10 +15,10 @@ end
 
 procedure map_exit_p_proc
 begin
-    if (global_var(1123) == 4) then begin
-        set_global_var(1123, 5);
+    if (global_var(GVAR_SULIK_DEAD) == 4) then begin
+        set_global_var(GVAR_SULIK_DEAD, 5);
     end
-    if (global_var(1124) >= 3) then begin
+    if (global_var(GVAR_FIND_SULIK) >= 3) then begin
         if (tile_contains_obj_pid(21648, 0, 16777750)) then begin
             destroy_object(tile_contains_pid_obj(21648, 0, 16777750));
         end
@@ -38,28 +38,28 @@ begin
             call Make_Plants();
         end
     end
-    if ((global_var(860) > 1) and ((global_var(860) < 6) or (global_var(27) == 901))) then begin
+    if ((global_var(GVAR_ABBEY_FORMAT_DATABASES) > 1) and ((global_var(GVAR_ABBEY_FORMAT_DATABASES) < 6) or (global_var(GVAR_LOAD_MAP_INDEX) == 901))) then begin
         move_to(dude_obj, 22887, 1);
     end
-    if (global_var(1127) == 4) then begin
-        LVar0 := create_object_sid(16777418, 18517, 1, 2294);
+    if (global_var(GVAR_WIPED_OUT_RANGER_SAFE_HOUSES) == 4) then begin
+        LVar0 := create_object_sid(16777418, 18517, 1, SCRIPT_RTRIBEC9);
         move_to(dude_obj, 20518, 1);
         anim(dude_obj, 1000, 5);
-        display_msg(message_str(2292, 102));
+        display_mstr(102);
         give_exp_points(600);
-        display_msg(message_str(14, 100) + 600 + message_str(14, 101));
-        set_global_var(65, global_var(65) + 15);
+        display_msg(g_mstr(100) + 600 + g_mstr(101));
+        set_global_var(GVAR_TOWN_REP_PRIMITIVE_TRIBE, global_var(GVAR_TOWN_REP_PRIMITIVE_TRIBE) + 15);
     end
-    if (metarule(14, 0)) then begin
-        display_msg(message_str(2292, 100));
+    if (map_first_run) then begin
+        display_mstr(100);
     end
-    if (global_var(27) == 900) then begin
-        set_global_var(1123, 3);
-        display_msg(message_str(2292, 200));
+    if (global_var(GVAR_LOAD_MAP_INDEX) == 900) then begin
+        set_global_var(GVAR_SULIK_DEAD, 3);
+        display_mstr(200);
         give_exp_points(1000);
-        display_msg(message_str(14, 100) + 1000 + message_str(14, 101));
-        set_global_var(65, global_var(65) + 15);
-        LVar0 := create_object_sid(16777754, 22101, 1, 2290);
+        display_msg(g_mstr(100) + 1000 + g_mstr(101));
+        set_global_var(GVAR_TOWN_REP_PRIMITIVE_TRIBE, global_var(GVAR_TOWN_REP_PRIMITIVE_TRIBE) + 15);
+        LVar0 := create_object_sid(16777754, 22101, 1, SCRIPT_RTRIBEC7);
         move_to(dude_obj, 22104, 1);
         anim(LVar0, 1000, 4);
         anim(dude_obj, 1000, 1);
@@ -140,7 +140,7 @@ begin
             end
         end
     end
-    set_global_var(27, 0);
+    set_global_var(GVAR_LOAD_MAP_INDEX, 0);
     return 0;
     LVar0 := 0;
 end
@@ -229,18 +229,17 @@ procedure Make_Plants
 begin
     variable LVar0 := 0;
     if (not(tile_contains_obj_pid(16081, 0, 272))) then begin
-        LVar0 := create_object_sid(272, 16081, 0, -1);
+        LVar0 := create_object(272, 16081, 0);
     end
     if (not(tile_contains_obj_pid(21251, 0, 272))) then begin
-        LVar0 := create_object_sid(272, 21251, 0, -1);
+        LVar0 := create_object(272, 21251, 0);
     end
     if (not(tile_contains_obj_pid(16850, 0, 271))) then begin
-        LVar0 := create_object_sid(271, 16850, 0, -1);
+        LVar0 := create_object(271, 16850, 0);
     end
     if (not(tile_contains_obj_pid(21134, 0, 271))) then begin
-        LVar0 := create_object_sid(271, 21134, 0, -1);
+        LVar0 := create_object(271, 21134, 0);
     end
     return 0;
     LVar0 := 0;
 end
-
