diff --git a/Mods/MegaMod/Original/RTRIBES2.ssl b/Mods/MegaMod/ReDefine/RTRIBES2.ssl
index 626a632..0a6abc7 100644
--- a/Mods/MegaMod/Original/RTRIBES2.ssl
+++ b/Mods/MegaMod/ReDefine/RTRIBES2.ssl
@@ -20,7 +20,7 @@ end
 
 procedure spatial_p_proc
 begin
-    if (not(global_var(1121) == 1)) then begin
+    if (not(global_var(GVAR_SULIK_SISTER) == 1)) then begin
         if (source_obj == dude_obj) then begin
             if (local_var(4) == 0) then begin
                 set_local_var(4, 1);
@@ -47,57 +47,57 @@ end
 
 procedure Node004
 begin
-    gsay_reply(2292, 300);
-    giq_option(4, 2292, 301, Node006, 50);
-    giq_option(4, 2292, 302, Node005, 50);
+    gsay_reply(SCRIPT_RTRIBEM1, 300);
+    giq_option(4, SCRIPT_RTRIBEM1, 301, Node006, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_RTRIBEM1, 302, Node005, NEUTRAL_REACTION);
 end
 
 procedure Node005
 begin
     add_timer_event(self_obj, game_ticks(0), 1);
-    set_global_var(0, global_var(0) - 10);
-    set_global_var(37, 0);
-    set_global_var(38, 0);
-    set_global_var(39, 0);
-    set_global_var(40, 0);
-    set_global_var(41, 0);
-    set_global_var(42, 0);
-    set_global_var(43, 0);
-    set_global_var(44, 0);
-    set_global_var(45, 0);
-    if (global_var(0) >= 1000) then begin
-        set_global_var(37, 1);
+    set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) - 10);
+    set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+    set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+    set_global_var(GVAR_KARMA_DEFENDER, 0);
+    set_global_var(GVAR_KARMA_WANDERER, 0);
+    set_global_var(GVAR_KARMA_BETRAYER, 0);
+    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+    set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+    set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+    if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
     end
     else begin
-        if (global_var(0) >= 750) then begin
-            set_global_var(38, 1);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+            set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
         end
         else begin
-            if (global_var(0) >= 500) then begin
-                set_global_var(39, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
             end
             else begin
-                if (global_var(0) >= 250) then begin
-                    set_global_var(40, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                    set_global_var(GVAR_KARMA_DEFENDER, 1);
                 end
                 else begin
-                    if (global_var(0) > -250) then begin
-                        set_global_var(41, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                        set_global_var(GVAR_KARMA_WANDERER, 1);
                     end
                     else begin
-                        if (global_var(0) > -500) then begin
-                            set_global_var(42, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                            set_global_var(GVAR_KARMA_BETRAYER, 1);
                         end
                         else begin
-                            if (global_var(0) > -750) then begin
-                                set_global_var(43, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                             end
                             else begin
-                                if (global_var(0) > -1000) then begin
-                                    set_global_var(44, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                    set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                 end
                                 else begin
-                                    set_global_var(45, 1);
+                                    set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                 end
                             end
                         end
@@ -131,12 +131,12 @@ begin
         end
         else begin
             if (fixed_param == 3) then begin
-                float_msg(WARRIOR_obj, message_str(2292, 400), 8);
+                float_msg(WARRIOR_obj, message_str(SCRIPT_RTRIBEM1, 400), FLOAT_MSG_YELLOW);
                 add_timer_event(self_obj, game_ticks(5), 4);
             end
             else begin
                 if (fixed_param == 4) then begin
-                    float_msg(WARRIOR_obj, message_str(2292, 401), 8);
+                    float_msg(WARRIOR_obj, message_str(SCRIPT_RTRIBEM1, 401), FLOAT_MSG_YELLOW);
                     add_timer_event(self_obj, game_ticks(1), 5);
                 end
                 else begin
@@ -150,4 +150,3 @@ begin
     return 0;
     LVar0 := 0;
 end
-
