diff --git a/Mods/MegaMod/Original/RTRIBEI2.ssl b/Mods/MegaMod/ReDefine/RTRIBEI2.ssl
index 8e3f1ed..c87fad8 100644
--- a/Mods/MegaMod/Original/RTRIBEI2.ssl
+++ b/Mods/MegaMod/ReDefine/RTRIBEI2.ssl
@@ -56,12 +56,12 @@ end
 
 procedure Node900
 begin
-    if (message_str(1517, 100) != 0) then begin
+    if (message_str(SCRIPT_TRIBEC7, 100) != 0) then begin
         if (party_member_obj(16777376) != 0) then begin
             call Node004();
         end
         else begin
-            if ((get_critter_stat(dude_obj, 1) > 8) or has_trait(0, dude_obj, 107) or has_trait(0, dude_obj, 17) or has_trait(0, dude_obj, 32) or (has_skill(dude_obj, 12) > 69) or (has_skill(dude_obj, 7) > 84) or (has_skill(dude_obj, 6) > 99)) then begin
+            if ((dude_perception > 8) or has_trait(TRAIT_PERK, dude_obj, PERK_vault_city_training_perk) or has_trait(TRAIT_PERK, dude_obj, PERK_master_trader) or has_trait(TRAIT_PERK, dude_obj, PERK_medic) or (has_skill(dude_obj, SKILL_SCIENCE) > 69) or (has_skill(dude_obj, SKILL_DOCTOR) > 84) or (has_skill(dude_obj, SKILL_FIRST_AID) > 99)) then begin
                 call Node003();
             end
             else begin
@@ -73,40 +73,39 @@ end
 
 procedure Node001
 begin
-    gsay_reply(2457, 200);
-    giq_option(4, 2457, 201, Node900, 50);
-    giq_option(4, 2457, 202, Node999, 50);
+    Reply(200);
+    giq_option(4, NAME, 201, Node900, NEUTRAL_REACTION);
+    giq_option(4, NAME, 202, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node002
 begin
-    gsay_reply(2457, 203);
-    giq_option(4, 2457, 204, Node999, 50);
+    Reply(203);
+    giq_option(4, NAME, 204, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node003
 begin
     set_map_var(2, 1);
-    gsay_reply(2457, 205);
-    giq_option(4, 2457, 206, Node999, 50);
+    Reply(205);
+    giq_option(4, NAME, 206, Node999, NEUTRAL_REACTION);
 end
 
 procedure look_at_p_proc
 begin
     script_overrides;
-    display_msg(message_str(2457, 100));
+    display_mstr(100);
 end
 
 procedure description_p_proc
 begin
     script_overrides;
-    display_msg(message_str(2457, 101 + map_var(2)));
+    display_mstr(101 + map_var(2));
 end
 
 procedure Node004
 begin
     set_map_var(2, 1);
-    gsay_reply(1517, 207);
-    giq_option(4, 1517, 206, Node999, 50);
+    gsay_reply(SCRIPT_TRIBEC7, 207);
+    giq_option(4, SCRIPT_TRIBEC7, 206, Node999, NEUTRAL_REACTION);
 end
-
