diff --git a/Mods/MegaMod/Original/SCDOROTH.ssl b/Mods/MegaMod/ReDefine/SCDOROTH.ssl
index 2067d19..c54c90f 100644
--- a/Mods/MegaMod/Original/SCDOROTH.ssl
+++ b/Mods/MegaMod/ReDefine/SCDOROTH.ssl
@@ -196,7 +196,7 @@ procedure critter_p_proc
 begin
     if ((local_var(5) == 2) and obj_can_see_obj(self_obj, dude_obj)) then begin
         set_local_var(5, 1);
-        attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+        attack(dude_obj);
     end
     if (not(combat_is_initialized)) then begin
         if (i_toto_obj != -1) then begin
@@ -229,48 +229,48 @@ begin
     Evil_Critter := 0;
     Slavery_Tolerant := 1;
     Karma_Perception := 20;
-    set_global_var(37, 0);
-    set_global_var(38, 0);
-    set_global_var(39, 0);
-    set_global_var(40, 0);
-    set_global_var(41, 0);
-    set_global_var(42, 0);
-    set_global_var(43, 0);
-    set_global_var(44, 0);
-    set_global_var(45, 0);
-    if (global_var(0) >= 1000) then begin
-        set_global_var(37, 1);
+    set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+    set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+    set_global_var(GVAR_KARMA_DEFENDER, 0);
+    set_global_var(GVAR_KARMA_WANDERER, 0);
+    set_global_var(GVAR_KARMA_BETRAYER, 0);
+    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+    set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+    set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+    if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
     end
     else begin
-        if (global_var(0) >= 750) then begin
-            set_global_var(38, 1);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+            set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
         end
         else begin
-            if (global_var(0) >= 500) then begin
-                set_global_var(39, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
             end
             else begin
-                if (global_var(0) >= 250) then begin
-                    set_global_var(40, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                    set_global_var(GVAR_KARMA_DEFENDER, 1);
                 end
                 else begin
-                    if (global_var(0) > -250) then begin
-                        set_global_var(41, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                        set_global_var(GVAR_KARMA_WANDERER, 1);
                     end
                     else begin
-                        if (global_var(0) > -500) then begin
-                            set_global_var(42, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                            set_global_var(GVAR_KARMA_BETRAYER, 1);
                         end
                         else begin
-                            if (global_var(0) > -750) then begin
-                                set_global_var(43, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                             end
                             else begin
-                                if (global_var(0) > -1000) then begin
-                                    set_global_var(44, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                    set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                 end
                                 else begin
-                                    set_global_var(45, 1);
+                                    set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                 end
                             end
                         end
@@ -281,7 +281,7 @@ begin
     end
     if (local_var(2) == 0) then begin
         set_local_var(2, 1);
-        set_local_var(3, (get_critter_stat(dude_obj, 3) - get_critter_stat(self_obj, 3)) * 5);
+        set_local_var(3, (dude_charisma - self_charisma) * 5);
         debug_msg("Base Reaction == " + local_var(3));
     end
     Static_Reaction := 0;
@@ -293,14 +293,14 @@ begin
     else begin
         reaction_bonus_karma := -1 * (-1 * reaction_bonus_karma / 1000);
     end
-    if (global_var(57) >= 0) then begin
-        reaction_bonus_town_rep := global_var(57) / 2;
+    if (global_var(GVAR_TOWN_REP_NCR) >= 0) then begin
+        reaction_bonus_town_rep := global_var(GVAR_TOWN_REP_NCR) / 2;
     end
     else begin
-        reaction_bonus_town_rep := -1 * (-1 * global_var(57) / 2);
+        reaction_bonus_town_rep := -1 * (-1 * global_var(GVAR_TOWN_REP_NCR) / 2);
     end
-    if (has_trait(2, dude_obj, 13)) then begin
-        if (get_critter_stat(dude_obj, 34) == get_critter_stat(self_obj, 34)) then begin
+    if (has_trait(TRAIT_TRAIT, dude_obj, TRAIT_sex_appeal)) then begin
+        if (dude_gender == self_gender) then begin
             Static_Reaction := Static_Reaction + -20;
         end
         else begin
@@ -316,48 +316,48 @@ begin
         debug_msg("Karma Reaction Bonus == " + Static_Reaction);
         Static_Reaction := Static_Reaction - reaction_bonus_town_rep;
         debug_msg("Town Rep Bonus == " + Static_Reaction);
-        set_global_var(37, 0);
-        set_global_var(38, 0);
-        set_global_var(39, 0);
-        set_global_var(40, 0);
-        set_global_var(41, 0);
-        set_global_var(42, 0);
-        set_global_var(43, 0);
-        set_global_var(44, 0);
-        set_global_var(45, 0);
-        if (global_var(0) >= 1000) then begin
-            set_global_var(37, 1);
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+        set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+        set_global_var(GVAR_KARMA_DEFENDER, 0);
+        set_global_var(GVAR_KARMA_WANDERER, 0);
+        set_global_var(GVAR_KARMA_BETRAYER, 0);
+        set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+        set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+            set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
         end
         else begin
-            if (global_var(0) >= 750) then begin
-                set_global_var(38, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
             end
             else begin
-                if (global_var(0) >= 500) then begin
-                    set_global_var(39, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                 end
                 else begin
-                    if (global_var(0) >= 250) then begin
-                        set_global_var(40, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                        set_global_var(GVAR_KARMA_DEFENDER, 1);
                     end
                     else begin
-                        if (global_var(0) > -250) then begin
-                            set_global_var(41, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                            set_global_var(GVAR_KARMA_WANDERER, 1);
                         end
                         else begin
-                            if (global_var(0) > -500) then begin
-                                set_global_var(42, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                set_global_var(GVAR_KARMA_BETRAYER, 1);
                             end
                             else begin
-                                if (global_var(0) > -750) then begin
-                                    set_global_var(43, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -1000) then begin
-                                        set_global_var(44, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                     end
                                     else begin
-                                        set_global_var(45, 1);
+                                        set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                     end
                                 end
                             end
@@ -366,35 +366,35 @@ begin
                 end
             end
         end
-        if (global_var(37) == 1) then begin
+        if (global_var(GVAR_KARMA_HOLY_WARRIOR) == 1) then begin
             Static_Reaction := Static_Reaction - 20;
         end
         else begin
-            if (global_var(38) == 1) then begin
+            if (global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES) == 1) then begin
                 Static_Reaction := Static_Reaction - 15;
             end
             else begin
-                if (global_var(39) == 1) then begin
+                if (global_var(GVAR_KARMA_SHIELD_OF_HOPE) == 1) then begin
                     Static_Reaction := Static_Reaction - 10;
                 end
                 else begin
-                    if (global_var(40) == 1) then begin
+                    if (global_var(GVAR_KARMA_DEFENDER) == 1) then begin
                         Static_Reaction := Static_Reaction - 5;
                     end
                     else begin
-                        if (global_var(41) == 1) then begin
+                        if (global_var(GVAR_KARMA_WANDERER) == 1) then begin
                             Static_Reaction := Static_Reaction - 0;
                         end
                         else begin
-                            if (global_var(42) == 1) then begin
+                            if (global_var(GVAR_KARMA_BETRAYER) == 1) then begin
                                 Static_Reaction := Static_Reaction - -5;
                             end
                             else begin
-                                if (global_var(43) == 1) then begin
+                                if (global_var(GVAR_KARMA_SWORD_OF_DESPAIR) == 1) then begin
                                     Static_Reaction := Static_Reaction - -10;
                                 end
                                 else begin
-                                    if (global_var(44) == 1) then begin
+                                    if (global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE) == 1) then begin
                                         Static_Reaction := Static_Reaction - -15;
                                     end
                                     else begin
@@ -414,48 +414,48 @@ begin
         debug_msg("Karma Reaction Bonus == " + Static_Reaction);
         Static_Reaction := Static_Reaction + reaction_bonus_town_rep;
         debug_msg("Town Rep Bonus == " + Static_Reaction);
-        set_global_var(37, 0);
-        set_global_var(38, 0);
-        set_global_var(39, 0);
-        set_global_var(40, 0);
-        set_global_var(41, 0);
-        set_global_var(42, 0);
-        set_global_var(43, 0);
-        set_global_var(44, 0);
-        set_global_var(45, 0);
-        if (global_var(0) >= 1000) then begin
-            set_global_var(37, 1);
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+        set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+        set_global_var(GVAR_KARMA_DEFENDER, 0);
+        set_global_var(GVAR_KARMA_WANDERER, 0);
+        set_global_var(GVAR_KARMA_BETRAYER, 0);
+        set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+        set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+            set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
         end
         else begin
-            if (global_var(0) >= 750) then begin
-                set_global_var(38, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
             end
             else begin
-                if (global_var(0) >= 500) then begin
-                    set_global_var(39, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                 end
                 else begin
-                    if (global_var(0) >= 250) then begin
-                        set_global_var(40, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                        set_global_var(GVAR_KARMA_DEFENDER, 1);
                     end
                     else begin
-                        if (global_var(0) > -250) then begin
-                            set_global_var(41, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                            set_global_var(GVAR_KARMA_WANDERER, 1);
                         end
                         else begin
-                            if (global_var(0) > -500) then begin
-                                set_global_var(42, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                set_global_var(GVAR_KARMA_BETRAYER, 1);
                             end
                             else begin
-                                if (global_var(0) > -750) then begin
-                                    set_global_var(43, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -1000) then begin
-                                        set_global_var(44, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                     end
                                     else begin
-                                        set_global_var(45, 1);
+                                        set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                     end
                                 end
                             end
@@ -464,35 +464,35 @@ begin
                 end
             end
         end
-        if (global_var(37) == 1) then begin
+        if (global_var(GVAR_KARMA_HOLY_WARRIOR) == 1) then begin
             Static_Reaction := Static_Reaction + 20;
         end
         else begin
-            if (global_var(38) == 1) then begin
+            if (global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES) == 1) then begin
                 Static_Reaction := Static_Reaction + 15;
             end
             else begin
-                if (global_var(39) == 1) then begin
+                if (global_var(GVAR_KARMA_SHIELD_OF_HOPE) == 1) then begin
                     Static_Reaction := Static_Reaction + 10;
                 end
                 else begin
-                    if (global_var(40) == 1) then begin
+                    if (global_var(GVAR_KARMA_DEFENDER) == 1) then begin
                         Static_Reaction := Static_Reaction + 5;
                     end
                     else begin
-                        if (global_var(41) == 1) then begin
+                        if (global_var(GVAR_KARMA_WANDERER) == 1) then begin
                             Static_Reaction := Static_Reaction + 0;
                         end
                         else begin
-                            if (global_var(42) == 1) then begin
+                            if (global_var(GVAR_KARMA_BETRAYER) == 1) then begin
                                 Static_Reaction := Static_Reaction + -5;
                             end
                             else begin
-                                if (global_var(43) == 1) then begin
+                                if (global_var(GVAR_KARMA_SWORD_OF_DESPAIR) == 1) then begin
                                     Static_Reaction := Static_Reaction + -10;
                                 end
                                 else begin
-                                    if (global_var(44) == 1) then begin
+                                    if (global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE) == 1) then begin
                                         Static_Reaction := Static_Reaction + -15;
                                     end
                                     else begin
@@ -508,34 +508,34 @@ begin
         debug_msg("Good Person Reaction Bonus == " + Static_Reaction);
     end
     if (Slavery_Tolerant == 0) then begin
-        if (global_var(11) == 1) then begin
+        if (global_var(GVAR_REPUTATION_SLAVER) == 1) then begin
             Static_Reaction := Static_Reaction + (2 * -25);
         end
     end
     else begin
         if (Slavery_Tolerant == 1) then begin
-            if (global_var(11) == 1) then begin
+            if (global_var(GVAR_REPUTATION_SLAVER) == 1) then begin
                 Static_Reaction := Static_Reaction + -25;
             end
         end
         else begin
             if (Slavery_Tolerant == 3) then begin
-                if (global_var(11) == 1) then begin
+                if (global_var(GVAR_REPUTATION_SLAVER) == 1) then begin
                     Static_Reaction := Static_Reaction - -25;
                 end
             end
         end
     end
     debug_msg("Slaver + Aligned Reaction == " + Static_Reaction);
-    Static_Reaction := Static_Reaction + (10 * has_trait(0, dude_obj, 10));
+    Static_Reaction := Static_Reaction + (10 * has_trait(TRAIT_PERK, dude_obj, PERK_presence));
     debug_msg("Presence Reaction == " + Static_Reaction);
-    if (global_var(1) >= 2) then begin
+    if (global_var(GVAR_CHILDKILLER_REPUTATION) >= 2) then begin
         Static_Reaction := Static_Reaction + -30;
     end
     debug_msg("Childkiller Reaction == " + Static_Reaction);
     set_local_var(0, local_var(3) + Static_Reaction);
     debug_msg("Initial Reaction == " + local_var(0));
-    if (global_var(242) == 1) then begin
+    if (global_var(GVAR_NCR_JACK_STATE) == 1) then begin
         start_gdialog(402, self_obj, 4, -1, -1);
         gsay_start;
         call Node020();
@@ -543,15 +543,15 @@ begin
         end_dialogue;
     end
     else begin
-        if (global_var(240) == 5) then begin
+        if (global_var(GVAR_NCR_POWERPLANT) == 5) then begin
             call Node022();
         end
         else begin
-            if (global_var(240) == 7) then begin
+            if (global_var(GVAR_NCR_POWERPLANT) == 7) then begin
                 call Node023();
             end
             else begin
-                if (global_var(240) == 4) then begin
+                if (global_var(GVAR_NCR_POWERPLANT) == 4) then begin
                     start_gdialog(402, self_obj, 4, -1, -1);
                     gsay_start;
                     call Node006();
@@ -562,7 +562,7 @@ begin
                     end
                 end
                 else begin
-                    if (global_var(240) == 2) then begin
+                    if (global_var(GVAR_NCR_POWERPLANT) == 2) then begin
                         start_gdialog(402, self_obj, 4, -1, -1);
                         gsay_start;
                         call Node015();
@@ -570,11 +570,11 @@ begin
                         end_dialogue;
                     end
                     else begin
-                        if (global_var(240) == 3) then begin
+                        if (global_var(GVAR_NCR_POWERPLANT) == 3) then begin
                             call Node014();
                         end
                         else begin
-                            if (global_var(240) == 8) then begin
+                            if (global_var(GVAR_NCR_POWERPLANT) == 8) then begin
                                 start_gdialog(402, self_obj, 4, -1, -1);
                                 gsay_start;
                                 call Node010();
@@ -582,15 +582,15 @@ begin
                                 end_dialogue;
                             end
                             else begin
-                                if (global_var(240) == 9) then begin
+                                if (global_var(GVAR_NCR_POWERPLANT) == 9) then begin
                                     call Node021();
                                 end
                                 else begin
-                                    if (global_var(240) == 6) then begin
+                                    if (global_var(GVAR_NCR_POWERPLANT) == 6) then begin
                                         call Node013();
                                     end
                                     else begin
-                                        if (global_var(240) == 1) then begin
+                                        if (global_var(GVAR_NCR_POWERPLANT) == 1) then begin
                                             call Node018();
                                         end
                                         else begin
@@ -601,12 +601,12 @@ begin
                                                     call Node001();
                                                     gsay_end;
                                                     end_dialogue;
-                                                    set_global_var(242, 4);
+                                                    set_global_var(GVAR_NCR_JACK_STATE, 4);
                                                     if (response == 4) then begin
-                                                        gfade_out(1);
+                                                        fade_out;
                                                         move_to(self_obj, 15462, 0);
                                                         move_to(dude_obj, 15266, 0);
-                                                        gfade_in(1);
+                                                        fade_in;
                                                     end
                                                 end
                                                 else begin
@@ -614,7 +614,7 @@ begin
                                                 end
                                             end
                                             else begin
-                                                if (global_var(240) == 0) then begin
+                                                if (global_var(GVAR_NCR_POWERPLANT) == 0) then begin
                                                     call Node019();
                                                 end
                                             end
@@ -633,59 +633,59 @@ end
 procedure destroy_p_proc
 begin
     if (source_obj == dude_obj) then begin
-        set_global_var(5, global_var(5) + 1);
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(4) > (3 * global_var(5))) or (global_var(2) == 1)) and (global_var(1) == 0)) then begin
-            set_global_var(2, 1);
-            set_global_var(3, 0);
+        set_global_var(GVAR_GOOD_MONSTER, global_var(GVAR_GOOD_MONSTER) + 1);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(4) > (3 * global_var(GVAR_GOOD_MONSTER))) or (global_var(GVAR_CHAMPION_REPUTATION) == 1)) and (global_var(GVAR_CHILDKILLER_REPUTATION) == 0)) then begin
+            set_global_var(GVAR_CHAMPION_REPUTATION, 1);
+            set_global_var(GVAR_BERSERKER_REPUTATION, 0);
         end
-        if (metarule(51, self_obj) == 2) then begin
+        if (critter_kill_type(self_obj) == 2) then begin
             if (source_obj == dude_obj) then begin
-                set_global_var(2, 0);
-                set_global_var(1, global_var(1) + 1);
-                set_global_var(0, global_var(0) + -15);
+                set_global_var(GVAR_CHAMPION_REPUTATION, 0);
+                set_global_var(GVAR_CHILDKILLER_REPUTATION, global_var(GVAR_CHILDKILLER_REPUTATION) + 1);
+                set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + -15);
                 debug_msg("Player gains " + -15 + " Karma Points.");
-                set_global_var(37, 0);
-                set_global_var(38, 0);
-                set_global_var(39, 0);
-                set_global_var(40, 0);
-                set_global_var(41, 0);
-                set_global_var(42, 0);
-                set_global_var(43, 0);
-                set_global_var(44, 0);
-                set_global_var(45, 0);
-                if (global_var(0) >= 1000) then begin
-                    set_global_var(37, 1);
+                set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+                set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+                set_global_var(GVAR_KARMA_DEFENDER, 0);
+                set_global_var(GVAR_KARMA_WANDERER, 0);
+                set_global_var(GVAR_KARMA_BETRAYER, 0);
+                set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+                set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+                set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+                    set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
                 end
                 else begin
-                    if (global_var(0) >= 750) then begin
-                        set_global_var(38, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
                     end
                     else begin
-                        if (global_var(0) >= 500) then begin
-                            set_global_var(39, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                            set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                         end
                         else begin
-                            if (global_var(0) >= 250) then begin
-                                set_global_var(40, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                                set_global_var(GVAR_KARMA_DEFENDER, 1);
                             end
                             else begin
-                                if (global_var(0) > -250) then begin
-                                    set_global_var(41, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                                    set_global_var(GVAR_KARMA_WANDERER, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -500) then begin
-                                        set_global_var(42, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                        set_global_var(GVAR_KARMA_BETRAYER, 1);
                                     end
                                     else begin
-                                        if (global_var(0) > -750) then begin
-                                            set_global_var(43, 1);
+                                        if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                            set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                         end
                                         else begin
-                                            if (global_var(0) > -1000) then begin
-                                                set_global_var(44, 1);
+                                            if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                                set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                             end
                                             else begin
-                                                set_global_var(45, 1);
+                                                set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                             end
                                         end
                                     end
@@ -694,58 +694,58 @@ begin
                         end
                     end
                 end
-                set_global_var(57, global_var(57) + -8);
+                set_global_var(GVAR_TOWN_REP_NCR, global_var(GVAR_TOWN_REP_NCR) + -8);
                 debug_msg("Added " + -8 + " to Town Rep");
             end
         end
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(5) > (2 * global_var(4))) or (global_var(3) == 1))) then begin
-            set_global_var(3, 1);
-            set_global_var(2, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
+            set_global_var(GVAR_BERSERKER_REPUTATION, 1);
+            set_global_var(GVAR_CHAMPION_REPUTATION, 0);
         end
-        set_global_var(0, global_var(0) + -10);
+        set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + -10);
         debug_msg("Player gains " + -10 + " Karma Points.");
-        set_global_var(37, 0);
-        set_global_var(38, 0);
-        set_global_var(39, 0);
-        set_global_var(40, 0);
-        set_global_var(41, 0);
-        set_global_var(42, 0);
-        set_global_var(43, 0);
-        set_global_var(44, 0);
-        set_global_var(45, 0);
-        if (global_var(0) >= 1000) then begin
-            set_global_var(37, 1);
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+        set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+        set_global_var(GVAR_KARMA_DEFENDER, 0);
+        set_global_var(GVAR_KARMA_WANDERER, 0);
+        set_global_var(GVAR_KARMA_BETRAYER, 0);
+        set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+        set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+            set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
         end
         else begin
-            if (global_var(0) >= 750) then begin
-                set_global_var(38, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
             end
             else begin
-                if (global_var(0) >= 500) then begin
-                    set_global_var(39, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                 end
                 else begin
-                    if (global_var(0) >= 250) then begin
-                        set_global_var(40, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                        set_global_var(GVAR_KARMA_DEFENDER, 1);
                     end
                     else begin
-                        if (global_var(0) > -250) then begin
-                            set_global_var(41, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                            set_global_var(GVAR_KARMA_WANDERER, 1);
                         end
                         else begin
-                            if (global_var(0) > -500) then begin
-                                set_global_var(42, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                set_global_var(GVAR_KARMA_BETRAYER, 1);
                             end
                             else begin
-                                if (global_var(0) > -750) then begin
-                                    set_global_var(43, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -1000) then begin
-                                        set_global_var(44, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                     end
                                     else begin
-                                        set_global_var(45, 1);
+                                        set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                     end
                                 end
                             end
@@ -754,7 +754,7 @@ begin
                 end
             end
         end
-        set_global_var(57, global_var(57) + -5);
+        set_global_var(GVAR_TOWN_REP_NCR, global_var(GVAR_TOWN_REP_NCR) + -5);
         debug_msg("Added " + -5 + " to Town Rep");
     end
 end
@@ -763,17 +763,17 @@ procedure look_at_p_proc
 begin
     script_overrides;
     if (local_var(4) == 0) then begin
-        display_msg(message_str(402, 100));
+        display_mstr(100);
     end
     else begin
-        display_msg(message_str(402, 101));
+        display_mstr(101);
     end
 end
 
 procedure description_p_proc
 begin
     script_overrides;
-    display_msg(message_str(402, 102));
+    display_mstr(102);
 end
 
 procedure use_skill_on_p_proc
@@ -784,20 +784,20 @@ procedure damage_p_proc
 begin
     if (source_obj == dude_obj) then begin
         set_local_var(6, 1);
-        set_global_var(438, global_var(438) bwor 1);
+        set_global_var(GVAR_NCR_GEN_FLAGS, global_var(GVAR_NCR_GEN_FLAGS) bwor 1);
     end
 end
 
 procedure map_enter_p_proc
 begin
     Only_Once := 0;
-    critter_add_trait(self_obj, 1, 6, 80);
-    critter_add_trait(self_obj, 1, 5, 24);
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 80);
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_TOUGH_CITIZEN);
     add_timer_event(self_obj, game_ticks(1), 1);
-    if ((i_jack_obj != -1) and (global_var(242) != 1)) then begin
+    if ((i_jack_obj != -1) and (global_var(GVAR_NCR_JACK_STATE) != 1)) then begin
         add_timer_event(self_obj, 1, 2);
     end
-    if (metarule(22, 0) == 0) then begin
+    if (is_loading_game == 0) then begin
         call checkHere();
     end
 end
@@ -813,96 +813,96 @@ end
 
 procedure Node001
 begin
-    gsay_reply(402, 103);
-    giq_option(-3, 402, 104, Node002, 50);
-    giq_option(4, 402, 105, Node003, 50);
+    Reply(103);
+    giq_option(-3, NAME, 104, Node002, NEUTRAL_REACTION);
+    giq_option(4, NAME, 105, Node003, NEUTRAL_REACTION);
 end
 
 procedure Node002
 begin
-    gsay_reply(402, 106);
-    giq_option(1, 402, 107, NodeWarpA, 49);
-    giq_option(1, 402, 108, NodeWarpB, 51);
+    Reply(106);
+    giq_option(1, NAME, 107, NodeWarpA, GOOD_REACTION);
+    giq_option(1, NAME, 108, NodeWarpB, BAD_REACTION);
 end
 
 procedure Node003
 begin
-    gsay_reply(402, 109);
-    giq_option(4, 402, 110, NodeWarpA, 49);
-    giq_option(4, 402, 111, NodeWarpB, 51);
+    Reply(109);
+    giq_option(4, NAME, 110, NodeWarpA, GOOD_REACTION);
+    giq_option(4, NAME, 111, NodeWarpB, BAD_REACTION);
 end
 
 procedure Node004
 begin
-    gsay_reply(402, 112);
-    giq_option(4, 402, 113, Node005, 50);
-    giq_option(4, 402, 114, Node999, 50);
+    Reply(112);
+    giq_option(4, NAME, 113, Node005, NEUTRAL_REACTION);
+    giq_option(4, NAME, 114, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node005
 begin
-    gsay_reply(402, 115);
-    giq_option(0, 402, message_str(14, 350), Node999, 50);
+    Reply(115);
+    giq_option(0, NAME, g_mstr(350), Node999, NEUTRAL_REACTION);
 end
 
 procedure Node006
 begin
-    gsay_reply(402, 116);
-    giq_option(-3, 402, 117, Node007, 50);
-    if (has_skill(dude_obj, 13) > 64) then begin
-        giq_option(4, 402, 118, Node008, 50);
+    Reply(116);
+    giq_option(-3, NAME, 117, Node007, NEUTRAL_REACTION);
+    if (has_skill(dude_obj, SKILL_REPAIR) > 64) then begin
+        giq_option(4, NAME, 118, Node008, NEUTRAL_REACTION);
     end
-    giq_option(4, 402, 119, Node009, 50);
+    giq_option(4, NAME, 119, Node009, NEUTRAL_REACTION);
 end
 
 procedure Node007
 begin
-    set_global_var(243, global_var(243) bwor 16);
-    gsay_reply(402, 120);
-    giq_option(0, 402, message_str(14, 350), Node999, 50);
+    set_global_var(GVAR_8_BALL_TOILET_SECRET, global_var(GVAR_8_BALL_TOILET_SECRET) bwor 16);
+    Reply(120);
+    giq_option(0, NAME, g_mstr(350), Node999, NEUTRAL_REACTION);
 end
 
 procedure Node008
 begin
-    set_global_var(243, global_var(243) bwor 16);
-    gsay_reply(402, 121);
-    giq_option(0, 402, message_str(14, 350), Node999, 50);
+    set_global_var(GVAR_8_BALL_TOILET_SECRET, global_var(GVAR_8_BALL_TOILET_SECRET) bwor 16);
+    Reply(121);
+    giq_option(0, NAME, g_mstr(350), Node999, NEUTRAL_REACTION);
     response := 2;
 end
 
 procedure Node009
 begin
-    gsay_reply(402, 122);
-    giq_option(0, 402, message_str(14, 350), Node999, 50);
+    Reply(122);
+    giq_option(0, NAME, g_mstr(350), Node999, NEUTRAL_REACTION);
     response := 3;
 end
 
 procedure Node010
 begin
-    gsay_reply(402, 123);
-    giq_option(-3, 402, 124, Node011, 50);
-    giq_option(4, 402, 125, Node012, 50);
+    Reply(123);
+    giq_option(-3, NAME, 124, Node011, NEUTRAL_REACTION);
+    giq_option(4, NAME, 125, Node012, NEUTRAL_REACTION);
 end
 
 procedure Node011
 begin
-    gsay_reply(402, 126);
-    giq_option(1, 402, message_str(14, 350), NodePowerDone, 50);
+    Reply(126);
+    giq_option(1, NAME, g_mstr(350), NodePowerDone, NEUTRAL_REACTION);
 end
 
 procedure Node012
 begin
-    gsay_reply(402, 127);
-    giq_option(1, 402, message_str(14, 350), NodePowerDone, 50);
+    Reply(127);
+    giq_option(1, NAME, g_mstr(350), NodePowerDone, NEUTRAL_REACTION);
 end
 
 procedure Node013
 begin
-    if (get_critter_stat(dude_obj, 34) == 0) then begin
-        float_msg(self_obj, message_str(402, 128), 0);
+    if (dude_gender == 0) then begin
+        float_msg(self_obj, mstr(128), FLOAT_MSG_NORMAL);
     end
     else begin
-        float_msg(self_obj, message_str(402, 142), 0);
+        float_msg(self_obj, mstr(142), FLOAT_MSG_NORMAL);
     end
 end
 
@@ -910,78 +910,78 @@ procedure Node014
 begin
     Node14Rep := Node14Rep + 1;
     if (Node14Rep == 1) then begin
-        float_msg(self_obj, message_str(402, 129), 0);
+        float_msg(self_obj, mstr(129), FLOAT_MSG_NORMAL);
     end
     else begin
-        float_msg(self_obj, message_str(402, 130), 0);
+        float_msg(self_obj, mstr(130), FLOAT_MSG_NORMAL);
     end
 end
 
 procedure Node015
 begin
-    gsay_reply(402, 131);
-    giq_option(1, 402, 132, Node016, 50);
-    giq_option(1, 402, 133, Node017, 50);
+    Reply(131);
+    giq_option(1, NAME, 132, Node016, NEUTRAL_REACTION);
+    giq_option(1, NAME, 133, Node017, NEUTRAL_REACTION);
 end
 
 procedure Node016
 begin
-    gsay_reply(402, 134);
-    giq_option(1, 402, message_str(14, 350), NodeMurder, 50);
+    Reply(134);
+    giq_option(1, NAME, g_mstr(350), NodeMurder, NEUTRAL_REACTION);
 end
 
 procedure Node017
 begin
-    gsay_reply(402, 135);
-    giq_option(1, 402, message_str(14, 350), NodeMurderButOk, 50);
+    Reply(135);
+    giq_option(1, NAME, g_mstr(350), NodeMurderButOk, NEUTRAL_REACTION);
 end
 
 procedure Node018
 begin
-    float_msg(self_obj, message_str(402, 136), 0);
+    float_msg(self_obj, mstr(136), FLOAT_MSG_NORMAL);
 end
 
 procedure Node019
 begin
     Node19Rep := Node19Rep + 1;
     if (Node19Rep == 1) then begin
-        float_msg(self_obj, message_str(402, 137), 0);
+        float_msg(self_obj, mstr(137), FLOAT_MSG_NORMAL);
     end
     else begin
-        float_msg(self_obj, message_str(402, 138), 0);
+        float_msg(self_obj, mstr(138), FLOAT_MSG_NORMAL);
     end
 end
 
 procedure Node020
 begin
-    if (global_var(240) != 6) then begin
-        if (get_critter_stat(dude_obj, 34) == 0) then begin
-            gsay_reply(402, message_str(402, 139));
+    if (global_var(GVAR_NCR_POWERPLANT) != 6) then begin
+        if (dude_gender == 0) then begin
+            Reply(mstr(139));
         end
         else begin
-            gsay_reply(402, message_str(402, 143));
+            Reply(mstr(143));
         end
-        giq_option(1, 402, message_str(14, 350), NodePowerDone, 50);
+        giq_option(1, NAME, g_mstr(350), NodePowerDone, NEUTRAL_REACTION);
     end
     else begin
-        gsay_reply(402, 140);
-        giq_option(0, 402, message_str(14, 350), Node999, 50);
+        Reply(140);
+        giq_option(0, NAME, g_mstr(350), Node999, NEUTRAL_REACTION);
     end
 end
 
 procedure Node021
 begin
-    float_msg(self_obj, message_str(402, 141), 0);
+    float_msg(self_obj, mstr(141), FLOAT_MSG_NORMAL);
 end
 
 procedure Node022
 begin
-    float_msg(self_obj, message_str(402, 144 + random(0, 2)), 0);
+    float_msg(self_obj, mstr(144 + random(0, 2)), FLOAT_MSG_NORMAL);
 end
 
 procedure Node023
 begin
-    float_msg(self_obj, message_str(402, 147 + random(0, 2)), 2);
+    float_msg(self_obj, mstr(147 + random(0, 2)), FLOAT_MSG_RED);
 end
 
 procedure walkToPos
@@ -1079,7 +1079,7 @@ end
 procedure checkHere
 begin
     if (getTimeBit() bwand (128 bwor 256 bwor 512 bwor 1024) bwor (2048 bwor 4096 bwor 8192 bwor 16384)) then begin
-        if ((i_jack_obj != -1) and (global_var(242) != 1)) then begin
+        if ((i_jack_obj != -1) and (global_var(GVAR_NCR_JACK_STATE) != 1)) then begin
             move_to(self_obj, 16274, 0);
         end
         else begin
@@ -1096,7 +1096,7 @@ end
 procedure timed_event_p_proc
 begin
     if (fixed_param == 1) then begin
-        if ((global_var(240) == 4) and (i_jack_obj == -1)) then begin
+        if ((global_var(GVAR_NCR_POWERPLANT) == 4) and (i_jack_obj == -1)) then begin
             if (waitTime == 0) then begin
                 waitTime := 1;
                 add_timer_event(self_obj, game_ticks(1), 1);
@@ -1112,7 +1112,7 @@ begin
             end
         end
         else begin
-            if (global_var(240) != 6) then begin
+            if (global_var(GVAR_NCR_POWERPLANT) != 6) then begin
                 add_timer_event(self_obj, game_ticks(1), 1);
             end
         end
@@ -1120,7 +1120,7 @@ begin
     else begin
         if (fixed_param == 2) then begin
             if (i_jack_obj != -1) then begin
-                if ((tile_distance_objs(self_obj, dude_obj) <= 5) and has_trait(1, self_obj, 666)) then begin
+                if ((tile_distance_objs(self_obj, dude_obj) <= 5) and obj_is_visible_flag(self_obj)) then begin
                     if (tile_num(i_jack_obj) == 14866) then begin
                         dialogue_system_enter;
                     end
@@ -1139,66 +1139,66 @@ end
 
 procedure NodeWarpA
 begin
-    set_global_var(240, 5);
+    set_global_var(GVAR_NCR_POWERPLANT, 5);
     response := 4;
 end
 
 procedure NodeWarpB
 begin
-    set_global_var(240, 7);
+    set_global_var(GVAR_NCR_POWERPLANT, 7);
 end
 
 procedure NodePowerDone
 begin
     move_obj_inven_to_obj(self_obj, dude_obj);
-    add_obj_to_inven(dude_obj, create_object_sid(73, tile_num(self_obj), 0, -1));
-    add_obj_to_inven(dude_obj, create_object_sid(80, tile_num(self_obj), 0, -1));
+    add_obj_to_inven(dude_obj, create_object(73, tile_num(self_obj), 0));
+    add_obj_to_inven(dude_obj, create_object(80, tile_num(self_obj), 0));
     give_exp_points(3000);
-    display_msg(message_str(14, 100) + 3000 + message_str(14, 101));
-    set_global_var(0, global_var(0) + 20);
+    display_msg(g_mstr(100) + 3000 + g_mstr(101));
+    set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + 20);
     debug_msg("Player gains " + 20 + " Karma Points.");
-    set_global_var(37, 0);
-    set_global_var(38, 0);
-    set_global_var(39, 0);
-    set_global_var(40, 0);
-    set_global_var(41, 0);
-    set_global_var(42, 0);
-    set_global_var(43, 0);
-    set_global_var(44, 0);
-    set_global_var(45, 0);
-    if (global_var(0) >= 1000) then begin
-        set_global_var(37, 1);
+    set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+    set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+    set_global_var(GVAR_KARMA_DEFENDER, 0);
+    set_global_var(GVAR_KARMA_WANDERER, 0);
+    set_global_var(GVAR_KARMA_BETRAYER, 0);
+    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+    set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+    set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+    if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
     end
     else begin
-        if (global_var(0) >= 750) then begin
-            set_global_var(38, 1);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+            set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
         end
         else begin
-            if (global_var(0) >= 500) then begin
-                set_global_var(39, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
             end
             else begin
-                if (global_var(0) >= 250) then begin
-                    set_global_var(40, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                    set_global_var(GVAR_KARMA_DEFENDER, 1);
                 end
                 else begin
-                    if (global_var(0) > -250) then begin
-                        set_global_var(41, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                        set_global_var(GVAR_KARMA_WANDERER, 1);
                     end
                     else begin
-                        if (global_var(0) > -500) then begin
-                            set_global_var(42, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                            set_global_var(GVAR_KARMA_BETRAYER, 1);
                         end
                         else begin
-                            if (global_var(0) > -750) then begin
-                                set_global_var(43, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                             end
                             else begin
-                                if (global_var(0) > -1000) then begin
-                                    set_global_var(44, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                    set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                 end
                                 else begin
-                                    set_global_var(45, 1);
+                                    set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                 end
                             end
                         end
@@ -1207,57 +1207,57 @@ begin
             end
         end
     end
-    set_global_var(240, 6);
+    set_global_var(GVAR_NCR_POWERPLANT, 6);
     call Node999();
 end
 
 procedure NodeMurder
 begin
-    set_global_var(240, 1);
-    set_global_var(0, global_var(0) + -20);
+    set_global_var(GVAR_NCR_POWERPLANT, 1);
+    set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + -20);
     debug_msg("Player gains " + -20 + " Karma Points.");
-    set_global_var(37, 0);
-    set_global_var(38, 0);
-    set_global_var(39, 0);
-    set_global_var(40, 0);
-    set_global_var(41, 0);
-    set_global_var(42, 0);
-    set_global_var(43, 0);
-    set_global_var(44, 0);
-    set_global_var(45, 0);
-    if (global_var(0) >= 1000) then begin
-        set_global_var(37, 1);
+    set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+    set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+    set_global_var(GVAR_KARMA_DEFENDER, 0);
+    set_global_var(GVAR_KARMA_WANDERER, 0);
+    set_global_var(GVAR_KARMA_BETRAYER, 0);
+    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+    set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+    set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+    if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
     end
     else begin
-        if (global_var(0) >= 750) then begin
-            set_global_var(38, 1);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+            set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
         end
         else begin
-            if (global_var(0) >= 500) then begin
-                set_global_var(39, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
             end
             else begin
-                if (global_var(0) >= 250) then begin
-                    set_global_var(40, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                    set_global_var(GVAR_KARMA_DEFENDER, 1);
                 end
                 else begin
-                    if (global_var(0) > -250) then begin
-                        set_global_var(41, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                        set_global_var(GVAR_KARMA_WANDERER, 1);
                     end
                     else begin
-                        if (global_var(0) > -500) then begin
-                            set_global_var(42, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                            set_global_var(GVAR_KARMA_BETRAYER, 1);
                         end
                         else begin
-                            if (global_var(0) > -750) then begin
-                                set_global_var(43, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                             end
                             else begin
-                                if (global_var(0) > -1000) then begin
-                                    set_global_var(44, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                    set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                 end
                                 else begin
-                                    set_global_var(45, 1);
+                                    set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                 end
                             end
                         end
@@ -1272,54 +1272,54 @@ end
 procedure NodeMurderButOk
 begin
     move_obj_inven_to_obj(self_obj, dude_obj);
-    add_obj_to_inven(dude_obj, create_object_sid(73, tile_num(self_obj), 0, -1));
-    add_obj_to_inven(dude_obj, create_object_sid(80, tile_num(self_obj), 0, -1));
+    add_obj_to_inven(dude_obj, create_object(73, tile_num(self_obj), 0));
+    add_obj_to_inven(dude_obj, create_object(80, tile_num(self_obj), 0));
     give_exp_points(3000);
-    display_msg(message_str(14, 100) + 3000 + message_str(14, 101));
-    set_global_var(0, global_var(0) + 20);
+    display_msg(g_mstr(100) + 3000 + g_mstr(101));
+    set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + 20);
     debug_msg("Player gains " + 20 + " Karma Points.");
-    set_global_var(37, 0);
-    set_global_var(38, 0);
-    set_global_var(39, 0);
-    set_global_var(40, 0);
-    set_global_var(41, 0);
-    set_global_var(42, 0);
-    set_global_var(43, 0);
-    set_global_var(44, 0);
-    set_global_var(45, 0);
-    if (global_var(0) >= 1000) then begin
-        set_global_var(37, 1);
+    set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+    set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+    set_global_var(GVAR_KARMA_DEFENDER, 0);
+    set_global_var(GVAR_KARMA_WANDERER, 0);
+    set_global_var(GVAR_KARMA_BETRAYER, 0);
+    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+    set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+    set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+    if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
     end
     else begin
-        if (global_var(0) >= 750) then begin
-            set_global_var(38, 1);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+            set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
         end
         else begin
-            if (global_var(0) >= 500) then begin
-                set_global_var(39, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
             end
             else begin
-                if (global_var(0) >= 250) then begin
-                    set_global_var(40, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                    set_global_var(GVAR_KARMA_DEFENDER, 1);
                 end
                 else begin
-                    if (global_var(0) > -250) then begin
-                        set_global_var(41, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                        set_global_var(GVAR_KARMA_WANDERER, 1);
                     end
                     else begin
-                        if (global_var(0) > -500) then begin
-                            set_global_var(42, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                            set_global_var(GVAR_KARMA_BETRAYER, 1);
                         end
                         else begin
-                            if (global_var(0) > -750) then begin
-                                set_global_var(43, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                             end
                             else begin
-                                if (global_var(0) > -1000) then begin
-                                    set_global_var(44, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                    set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                 end
                                 else begin
-                                    set_global_var(45, 1);
+                                    set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                 end
                             end
                         end
@@ -1328,7 +1328,6 @@ begin
             end
         end
     end
-    set_global_var(240, 6);
+    set_global_var(GVAR_NCR_POWERPLANT, 6);
     call Node999();
 end
-
