diff --git a/Mods/MegaMod/Original/SCGGCAP.ssl b/Mods/MegaMod/ReDefine/SCGGCAP.ssl
index 01d66db..f89d50b 100644
--- a/Mods/MegaMod/Original/SCGGCAP.ssl
+++ b/Mods/MegaMod/ReDefine/SCGGCAP.ssl
@@ -157,10 +157,10 @@ procedure critter_p_proc
 begin
     if ((local_var(5) == 2) and obj_can_see_obj(self_obj, dude_obj)) then begin
         set_local_var(5, 1);
-        attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+        attack(dude_obj);
     end
     else begin
-        if (global_var(225) == 1) then begin
+        if (global_var(GVAR_NCR_GTEGRD_ATTACK) == 1) then begin
             if (not(combat_is_initialized) and canSeePlayer()) then begin
                 set_local_var(5, 2);
             end
@@ -201,63 +201,63 @@ end
 procedure destroy_p_proc
 begin
     if (source_obj == dude_obj) then begin
-        set_global_var(5, global_var(5) + 1);
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(4) > (3 * global_var(5))) or (global_var(2) == 1)) and (global_var(1) == 0)) then begin
-            set_global_var(2, 1);
-            set_global_var(3, 0);
+        set_global_var(GVAR_GOOD_MONSTER, global_var(GVAR_GOOD_MONSTER) + 1);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(4) > (3 * global_var(GVAR_GOOD_MONSTER))) or (global_var(GVAR_CHAMPION_REPUTATION) == 1)) and (global_var(GVAR_CHILDKILLER_REPUTATION) == 0)) then begin
+            set_global_var(GVAR_CHAMPION_REPUTATION, 1);
+            set_global_var(GVAR_BERSERKER_REPUTATION, 0);
         end
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(5) > (2 * global_var(4))) or (global_var(3) == 1))) then begin
-            set_global_var(3, 1);
-            set_global_var(2, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
+            set_global_var(GVAR_BERSERKER_REPUTATION, 1);
+            set_global_var(GVAR_CHAMPION_REPUTATION, 0);
         end
-        set_global_var(0, global_var(0) + -10);
+        set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + -10);
         debug_msg("Player gains " + -10 + " Karma Points.");
-        set_global_var(37, 0);
-        set_global_var(38, 0);
-        set_global_var(39, 0);
-        set_global_var(40, 0);
-        set_global_var(41, 0);
-        set_global_var(42, 0);
-        set_global_var(43, 0);
-        set_global_var(44, 0);
-        set_global_var(45, 0);
-        tmp_gen_rep := global_var(0);
-        if (has_trait(0, dude_obj, 95)) then begin
-            tmp_gen_rep := global_var(0) * 2;
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+        set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+        set_global_var(GVAR_KARMA_DEFENDER, 0);
+        set_global_var(GVAR_KARMA_WANDERER, 0);
+        set_global_var(GVAR_KARMA_BETRAYER, 0);
+        set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+        set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+        tmp_gen_rep := global_var(GVAR_PLAYER_REPUTATION);
+        if (has_trait(TRAIT_PERK, dude_obj, PERK_karma_beacon_perk)) then begin
+            tmp_gen_rep := global_var(GVAR_PLAYER_REPUTATION) * 2;
         end
         if (tmp_gen_rep >= 1000) then begin
-            set_global_var(37, 1);
+            set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
         end
         else begin
             if (tmp_gen_rep >= 750) then begin
-                set_global_var(38, 1);
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
             end
             else begin
                 if (tmp_gen_rep >= 500) then begin
-                    set_global_var(39, 1);
+                    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                 end
                 else begin
                     if (tmp_gen_rep >= 250) then begin
-                        set_global_var(40, 1);
+                        set_global_var(GVAR_KARMA_DEFENDER, 1);
                     end
                     else begin
                         if (tmp_gen_rep > -250) then begin
-                            set_global_var(41, 1);
+                            set_global_var(GVAR_KARMA_WANDERER, 1);
                         end
                         else begin
                             if (tmp_gen_rep > -500) then begin
-                                set_global_var(42, 1);
+                                set_global_var(GVAR_KARMA_BETRAYER, 1);
                             end
                             else begin
                                 if (tmp_gen_rep > -750) then begin
-                                    set_global_var(43, 1);
+                                    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                 end
                                 else begin
                                     if (tmp_gen_rep > -1000) then begin
-                                        set_global_var(44, 1);
+                                        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                     end
                                     else begin
-                                        set_global_var(45, 1);
+                                        set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                     end
                                 end
                             end
@@ -266,7 +266,7 @@ begin
                 end
             end
         end
-        set_global_var(57, global_var(57) + -5);
+        set_global_var(GVAR_TOWN_REP_NCR, global_var(GVAR_TOWN_REP_NCR) + -5);
         debug_msg("Added " + -5 + " to Town Rep");
     end
     i_guard_captain_obj := -1;
@@ -276,17 +276,17 @@ procedure look_at_p_proc
 begin
     script_overrides;
     if (local_var(4) == 0) then begin
-        display_msg(message_str(371, 100));
+        display_mstr(100);
     end
     else begin
-        display_msg(message_str(371, 101));
+        display_mstr(101);
     end
 end
 
 procedure description_p_proc
 begin
     script_overrides;
-    display_msg(message_str(371, 102));
+    display_mstr(102);
 end
 
 procedure use_skill_on_p_proc
@@ -297,14 +297,14 @@ procedure damage_p_proc
 begin
     if (source_obj == dude_obj) then begin
         set_local_var(5, 2);
-        set_global_var(225, 1);
+        set_global_var(GVAR_NCR_GTEGRD_ATTACK, 1);
     end
 end
 
 procedure map_enter_p_proc
 begin
-    critter_add_trait(self_obj, 1, 6, 69);
-    critter_add_trait(self_obj, 1, 5, 22);
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 69);
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_TOUGH_GUARD);
     i_guard_captain_obj := self_obj;
     debug_msg("SCGGCAP: i_guard_captain_obj: " + self_obj);
     if (local_var(7) == 0) then begin
@@ -323,20 +323,20 @@ end
 
 procedure Node001
 begin
-    float_msg(self_obj, message_str(371, 103 + random(0, 3)), 0);
+    float_msg(self_obj, mstr(103 + random(0, 3)), FLOAT_MSG_NORMAL);
 end
 
 procedure Node002
 begin
     if (local_var(4) == 0) then begin
-        gsay_reply(371, 107);
+        Reply(107);
     end
     else begin
-        gsay_reply(371, 108);
+        Reply(108);
     end
-    giq_option(-3, 371, 109, Node003, 50);
-    giq_option(4, 371, 110, Node003, 50);
-    giq_option(1, 371, 111, Node999, 50);
+    giq_option(-3, NAME, 109, Node003, NEUTRAL_REACTION);
+    giq_option(4, NAME, 110, Node003, NEUTRAL_REACTION);
+    giq_option(1, NAME, 111, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node003
@@ -421,32 +421,32 @@ begin
         end
     end
     if ((obj_item_subtype(critter_inven_obj(dude_obj, 1)) == 3) or (obj_item_subtype(critter_inven_obj(dude_obj, 2)) == 3)) then begin
-        gsay_reply(371, 112);
-        giq_option(0, 371, message_str(14, 350), Node999, 50);
+        Reply(112);
+        giq_option(0, NAME, g_mstr(350), Node999, NEUTRAL_REACTION);
     end
     else begin
         if (How_Many_Party_Members_Armed > 0) then begin
-            gsay_reply(371, 113);
-            giq_option(0, 371, message_str(14, 350), Node999, 50);
+            Reply(113);
+            giq_option(0, NAME, g_mstr(350), Node999, NEUTRAL_REACTION);
         end
         else begin
-            gsay_reply(371, 114);
-            giq_option(1, 371, obj_name(dude_obj) + message_str(371, 120), Node004, 50);
+            Reply(114);
+            giq_option(1, NAME, dude_name + mstr(120), Node004, NEUTRAL_REACTION);
         end
     end
 end
 
 procedure Node004
 begin
-    set_global_var(226, 1);
+    set_global_var(GVAR_NCR_GATE_NIGHT, 1);
     set_map_var(6, 2);
-    gsay_reply(371, 116);
-    giq_option(0, 371, message_str(14, 350), Node999, 50);
+    Reply(116);
+    giq_option(0, NAME, g_mstr(350), Node999, NEUTRAL_REACTION);
 end
 
 procedure Node005
 begin
-    float_msg(self_obj, message_str(371, 117), 2);
+    float_msg(self_obj, mstr(117), FLOAT_MSG_RED);
 end
 
 procedure canSeePlayer
@@ -478,11 +478,10 @@ begin
     end
     if (fixed_param == 20) then begin
         metarule3(100, self_obj, 20, 0);
-        float_msg(self_obj, message_str(371, 117), 2);
+        float_msg(self_obj, mstr(117), FLOAT_MSG_RED);
     end
     if (fixed_param == 21) then begin
         metarule3(100, self_obj, 21, 0);
         set_local_var(5, 2);
     end
 end
-
