diff --git a/Mods/MegaMod/Original/SCMERK.ssl b/Mods/MegaMod/ReDefine/SCMERK.ssl
index 4d035ce..081fffa 100644
--- a/Mods/MegaMod/Original/SCMERK.ssl
+++ b/Mods/MegaMod/ReDefine/SCMERK.ssl
@@ -271,21 +271,21 @@ end
 
 procedure critter_p_proc
 begin
-    if ((do_once != 1) and has_trait(1, self_obj, 666)) then begin
+    if ((do_once != 1) and obj_is_visible_flag(self_obj)) then begin
         do_once := 1;
         merk_visible := 1;
         debug_msg("merk is visible.");
     end
-    if ((global_var(207) != 0) and obj_can_see_obj(self_obj, dude_obj)) then begin
-        set_global_var(207, 1);
+    if ((global_var(GVAR_NCR_SCMERK_HOSTILE) != 0) and obj_can_see_obj(self_obj, dude_obj)) then begin
+        set_global_var(GVAR_NCR_SCMERK_HOSTILE, 1);
         if (critter_is_fleeing(self_obj)) then begin
             if (anim_busy(self_obj) == 0) then begin
                 if ((tile_distance_objs(self_obj, dude_obj) < 8) or obj_can_see_obj(self_obj, dude_obj)) then begin
                     if (anim_busy(self_obj) == 0) then begin
-                        global_temp := rotation_to_tile(tile_num(dude_obj), tile_num(self_obj));
+                        global_temp := rotation_to_tile(dude_tile, tile_num(self_obj));
                         animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(self_obj), global_temp, random(3, 10)), 1);
                         global_temp := (global_temp + 1) % 6;
-                        while ((anim_busy(self_obj) == 0) and (global_temp != rotation_to_tile(tile_num(dude_obj), tile_num(self_obj)))) do begin
+                        while ((anim_busy(self_obj) == 0) and (global_temp != rotation_to_tile(dude_tile, tile_num(self_obj)))) do begin
                             debug_msg("flee loop: rot == " + global_temp);
                             animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(self_obj), global_temp, random(3, 10)), 1);
                             global_temp := (global_temp + 1) % 6;
@@ -295,18 +295,18 @@ begin
             end
         end
         else begin
-            attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+            attack(dude_obj);
         end
     end
-    if ((local_var(4) == 1) and (map_var(2) == 1) or (((global_var(438) bwand 32768) != 0) and not((global_var(438) bwand 65536) != 0) and (local_var(4) == 1))) then begin
+    if ((local_var(4) == 1) and (map_var(2) == 1) or (((global_var(GVAR_NCR_GEN_FLAGS) bwand 32768) != 0) and not((global_var(GVAR_NCR_GEN_FLAGS) bwand 65536) != 0) and (local_var(4) == 1))) then begin
         debug_msg("MERK: should end fight");
         call EndFight();
     end
-    if (((global_var(438) bwand 32768) != 0) and not((global_var(438) bwand 65536) != 0) and (local_var(4) == 0)) then begin
-        if (obj_can_see_obj(self_obj, dude_obj) and has_trait(1, self_obj, 666)) then begin
-            set_global_var(438, global_var(438) bwor 65536);
-            float_msg(self_obj, message_str(325, 265), 2);
-            set_global_var(207, 1);
+    if (((global_var(GVAR_NCR_GEN_FLAGS) bwand 32768) != 0) and not((global_var(GVAR_NCR_GEN_FLAGS) bwand 65536) != 0) and (local_var(4) == 0)) then begin
+        if (obj_can_see_obj(self_obj, dude_obj) and obj_is_visible_flag(self_obj)) then begin
+            set_global_var(GVAR_NCR_GEN_FLAGS, global_var(GVAR_NCR_GEN_FLAGS) bwor 65536);
+            float_msg(self_obj, mstr(265), FLOAT_MSG_RED);
+            set_global_var(GVAR_NCR_SCMERK_HOSTILE, 1);
         end
     end
 end
@@ -314,7 +314,7 @@ end
 procedure pickup_p_proc
 begin
     if (source_obj == dude_obj) then begin
-        set_global_var(207, 2);
+        set_global_var(GVAR_NCR_SCMERK_HOSTILE, 2);
     end
 end
 
@@ -323,48 +323,48 @@ begin
     Evil_Critter := 1;
     Slavery_Tolerant := 2;
     Karma_Perception := 10;
-    set_global_var(37, 0);
-    set_global_var(38, 0);
-    set_global_var(39, 0);
-    set_global_var(40, 0);
-    set_global_var(41, 0);
-    set_global_var(42, 0);
-    set_global_var(43, 0);
-    set_global_var(44, 0);
-    set_global_var(45, 0);
-    if (global_var(0) >= 1000) then begin
-        set_global_var(37, 1);
+    set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+    set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+    set_global_var(GVAR_KARMA_DEFENDER, 0);
+    set_global_var(GVAR_KARMA_WANDERER, 0);
+    set_global_var(GVAR_KARMA_BETRAYER, 0);
+    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+    set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+    set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+    if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
     end
     else begin
-        if (global_var(0) >= 750) then begin
-            set_global_var(38, 1);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+            set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
         end
         else begin
-            if (global_var(0) >= 500) then begin
-                set_global_var(39, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
             end
             else begin
-                if (global_var(0) >= 250) then begin
-                    set_global_var(40, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                    set_global_var(GVAR_KARMA_DEFENDER, 1);
                 end
                 else begin
-                    if (global_var(0) > -250) then begin
-                        set_global_var(41, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                        set_global_var(GVAR_KARMA_WANDERER, 1);
                     end
                     else begin
-                        if (global_var(0) > -500) then begin
-                            set_global_var(42, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                            set_global_var(GVAR_KARMA_BETRAYER, 1);
                         end
                         else begin
-                            if (global_var(0) > -750) then begin
-                                set_global_var(43, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                             end
                             else begin
-                                if (global_var(0) > -1000) then begin
-                                    set_global_var(44, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                    set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                 end
                                 else begin
-                                    set_global_var(45, 1);
+                                    set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                 end
                             end
                         end
@@ -375,7 +375,7 @@ begin
     end
     if (local_var(2) == 0) then begin
         set_local_var(2, 1);
-        set_local_var(3, (get_critter_stat(dude_obj, 3) - get_critter_stat(self_obj, 3)) * 5);
+        set_local_var(3, (dude_charisma - self_charisma) * 5);
         debug_msg("Base Reaction == " + local_var(3));
     end
     Static_Reaction := 0;
@@ -387,14 +387,14 @@ begin
     else begin
         reaction_bonus_karma := -1 * (-1 * reaction_bonus_karma / 1000);
     end
-    if (global_var(57) >= 0) then begin
-        reaction_bonus_town_rep := global_var(57) / 2;
+    if (global_var(GVAR_TOWN_REP_NCR) >= 0) then begin
+        reaction_bonus_town_rep := global_var(GVAR_TOWN_REP_NCR) / 2;
     end
     else begin
-        reaction_bonus_town_rep := -1 * (-1 * global_var(57) / 2);
+        reaction_bonus_town_rep := -1 * (-1 * global_var(GVAR_TOWN_REP_NCR) / 2);
     end
-    if (has_trait(2, dude_obj, 13)) then begin
-        if (get_critter_stat(dude_obj, 34) == get_critter_stat(self_obj, 34)) then begin
+    if (has_trait(TRAIT_TRAIT, dude_obj, TRAIT_sex_appeal)) then begin
+        if (dude_gender == self_gender) then begin
             Static_Reaction := Static_Reaction + -20;
         end
         else begin
@@ -406,58 +406,58 @@ begin
         debug_msg("Sex Appeal Bonus == 0");
     end
     if (Evil_Critter == 1) then begin
-        Static_Reaction := Static_Reaction - reaction_bonus_karma + (reaction_bonus_karma * has_trait(0, dude_obj, 39));
-        if ((reaction_bonus_karma < 0) and (has_trait(0, dude_obj, 39) != 0)) then begin
+        Static_Reaction := Static_Reaction - reaction_bonus_karma + (reaction_bonus_karma * has_trait(TRAIT_PERK, dude_obj, PERK_cult_of_personality));
+        if ((reaction_bonus_karma < 0) and (has_trait(TRAIT_PERK, dude_obj, PERK_cult_of_personality) != 0)) then begin
             Static_Reaction := Static_Reaction - reaction_bonus_karma;
         end
         debug_msg("Karma Reaction Bonus == " + Static_Reaction);
-        Static_Reaction := Static_Reaction - reaction_bonus_town_rep + (reaction_bonus_town_rep * has_trait(0, dude_obj, 39));
-        if ((reaction_bonus_town_rep < 0) and (has_trait(0, dude_obj, 39) != 0)) then begin
+        Static_Reaction := Static_Reaction - reaction_bonus_town_rep + (reaction_bonus_town_rep * has_trait(TRAIT_PERK, dude_obj, PERK_cult_of_personality));
+        if ((reaction_bonus_town_rep < 0) and (has_trait(TRAIT_PERK, dude_obj, PERK_cult_of_personality) != 0)) then begin
             Static_Reaction := Static_Reaction - reaction_bonus_town_rep;
         end
         debug_msg("Town Rep Bonus == " + Static_Reaction);
-        set_global_var(37, 0);
-        set_global_var(38, 0);
-        set_global_var(39, 0);
-        set_global_var(40, 0);
-        set_global_var(41, 0);
-        set_global_var(42, 0);
-        set_global_var(43, 0);
-        set_global_var(44, 0);
-        set_global_var(45, 0);
-        if (global_var(0) >= 1000) then begin
-            set_global_var(37, 1);
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+        set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+        set_global_var(GVAR_KARMA_DEFENDER, 0);
+        set_global_var(GVAR_KARMA_WANDERER, 0);
+        set_global_var(GVAR_KARMA_BETRAYER, 0);
+        set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+        set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+            set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
         end
         else begin
-            if (global_var(0) >= 750) then begin
-                set_global_var(38, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
             end
             else begin
-                if (global_var(0) >= 500) then begin
-                    set_global_var(39, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                 end
                 else begin
-                    if (global_var(0) >= 250) then begin
-                        set_global_var(40, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                        set_global_var(GVAR_KARMA_DEFENDER, 1);
                     end
                     else begin
-                        if (global_var(0) > -250) then begin
-                            set_global_var(41, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                            set_global_var(GVAR_KARMA_WANDERER, 1);
                         end
                         else begin
-                            if (global_var(0) > -500) then begin
-                                set_global_var(42, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                set_global_var(GVAR_KARMA_BETRAYER, 1);
                             end
                             else begin
-                                if (global_var(0) > -750) then begin
-                                    set_global_var(43, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -1000) then begin
-                                        set_global_var(44, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                     end
                                     else begin
-                                        set_global_var(45, 1);
+                                        set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                     end
                                 end
                             end
@@ -466,35 +466,35 @@ begin
                 end
             end
         end
-        if (global_var(37) == 1) then begin
-            Static_Reaction := Static_Reaction - 20 + (40 * has_trait(0, dude_obj, 39));
+        if (global_var(GVAR_KARMA_HOLY_WARRIOR) == 1) then begin
+            Static_Reaction := Static_Reaction - 20 + (40 * has_trait(TRAIT_PERK, dude_obj, PERK_cult_of_personality));
         end
         else begin
-            if (global_var(38) == 1) then begin
-                Static_Reaction := Static_Reaction - 15 + (30 * has_trait(0, dude_obj, 39));
+            if (global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES) == 1) then begin
+                Static_Reaction := Static_Reaction - 15 + (30 * has_trait(TRAIT_PERK, dude_obj, PERK_cult_of_personality));
             end
             else begin
-                if (global_var(39) == 1) then begin
-                    Static_Reaction := Static_Reaction - 10 + (20 * has_trait(0, dude_obj, 39));
+                if (global_var(GVAR_KARMA_SHIELD_OF_HOPE) == 1) then begin
+                    Static_Reaction := Static_Reaction - 10 + (20 * has_trait(TRAIT_PERK, dude_obj, PERK_cult_of_personality));
                 end
                 else begin
-                    if (global_var(40) == 1) then begin
-                        Static_Reaction := Static_Reaction - 5 + (10 * has_trait(0, dude_obj, 39));
+                    if (global_var(GVAR_KARMA_DEFENDER) == 1) then begin
+                        Static_Reaction := Static_Reaction - 5 + (10 * has_trait(TRAIT_PERK, dude_obj, PERK_cult_of_personality));
                     end
                     else begin
-                        if (global_var(41) == 1) then begin
+                        if (global_var(GVAR_KARMA_WANDERER) == 1) then begin
                             Static_Reaction := Static_Reaction - 0;
                         end
                         else begin
-                            if (global_var(42) == 1) then begin
+                            if (global_var(GVAR_KARMA_BETRAYER) == 1) then begin
                                 Static_Reaction := Static_Reaction - -5;
                             end
                             else begin
-                                if (global_var(43) == 1) then begin
+                                if (global_var(GVAR_KARMA_SWORD_OF_DESPAIR) == 1) then begin
                                     Static_Reaction := Static_Reaction - -10;
                                 end
                                 else begin
-                                    if (global_var(44) == 1) then begin
+                                    if (global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE) == 1) then begin
                                         Static_Reaction := Static_Reaction - -15;
                                     end
                                     else begin
@@ -510,58 +510,58 @@ begin
         debug_msg("Bad Person Reaction Bonus == " + Static_Reaction);
     end
     else begin
-        Static_Reaction := Static_Reaction + reaction_bonus_karma - (reaction_bonus_karma * has_trait(0, dude_obj, 39));
-        if ((reaction_bonus_karma > 0) and (has_trait(0, dude_obj, 39) != 0)) then begin
+        Static_Reaction := Static_Reaction + reaction_bonus_karma - (reaction_bonus_karma * has_trait(TRAIT_PERK, dude_obj, PERK_cult_of_personality));
+        if ((reaction_bonus_karma > 0) and (has_trait(TRAIT_PERK, dude_obj, PERK_cult_of_personality) != 0)) then begin
             Static_Reaction := Static_Reaction + reaction_bonus_karma;
         end
         debug_msg("Karma Reaction Bonus == " + Static_Reaction);
-        Static_Reaction := Static_Reaction + reaction_bonus_town_rep - (reaction_bonus_town_rep * has_trait(0, dude_obj, 39));
-        if ((reaction_bonus_town_rep > 0) and (has_trait(0, dude_obj, 39) != 0)) then begin
+        Static_Reaction := Static_Reaction + reaction_bonus_town_rep - (reaction_bonus_town_rep * has_trait(TRAIT_PERK, dude_obj, PERK_cult_of_personality));
+        if ((reaction_bonus_town_rep > 0) and (has_trait(TRAIT_PERK, dude_obj, PERK_cult_of_personality) != 0)) then begin
             Static_Reaction := Static_Reaction + reaction_bonus_town_rep;
         end
         debug_msg("Town Rep Bonus == " + Static_Reaction);
-        set_global_var(37, 0);
-        set_global_var(38, 0);
-        set_global_var(39, 0);
-        set_global_var(40, 0);
-        set_global_var(41, 0);
-        set_global_var(42, 0);
-        set_global_var(43, 0);
-        set_global_var(44, 0);
-        set_global_var(45, 0);
-        if (global_var(0) >= 1000) then begin
-            set_global_var(37, 1);
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+        set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+        set_global_var(GVAR_KARMA_DEFENDER, 0);
+        set_global_var(GVAR_KARMA_WANDERER, 0);
+        set_global_var(GVAR_KARMA_BETRAYER, 0);
+        set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+        set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+            set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
         end
         else begin
-            if (global_var(0) >= 750) then begin
-                set_global_var(38, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
             end
             else begin
-                if (global_var(0) >= 500) then begin
-                    set_global_var(39, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                 end
                 else begin
-                    if (global_var(0) >= 250) then begin
-                        set_global_var(40, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                        set_global_var(GVAR_KARMA_DEFENDER, 1);
                     end
                     else begin
-                        if (global_var(0) > -250) then begin
-                            set_global_var(41, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                            set_global_var(GVAR_KARMA_WANDERER, 1);
                         end
                         else begin
-                            if (global_var(0) > -500) then begin
-                                set_global_var(42, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                set_global_var(GVAR_KARMA_BETRAYER, 1);
                             end
                             else begin
-                                if (global_var(0) > -750) then begin
-                                    set_global_var(43, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -1000) then begin
-                                        set_global_var(44, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                     end
                                     else begin
-                                        set_global_var(45, 1);
+                                        set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                     end
                                 end
                             end
@@ -570,39 +570,39 @@ begin
                 end
             end
         end
-        if (global_var(37) == 1) then begin
+        if (global_var(GVAR_KARMA_HOLY_WARRIOR) == 1) then begin
             Static_Reaction := Static_Reaction + 20;
         end
         else begin
-            if (global_var(38) == 1) then begin
+            if (global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES) == 1) then begin
                 Static_Reaction := Static_Reaction + 15;
             end
             else begin
-                if (global_var(39) == 1) then begin
+                if (global_var(GVAR_KARMA_SHIELD_OF_HOPE) == 1) then begin
                     Static_Reaction := Static_Reaction + 10;
                 end
                 else begin
-                    if (global_var(40) == 1) then begin
+                    if (global_var(GVAR_KARMA_DEFENDER) == 1) then begin
                         Static_Reaction := Static_Reaction + 5;
                     end
                     else begin
-                        if (global_var(41) == 1) then begin
+                        if (global_var(GVAR_KARMA_WANDERER) == 1) then begin
                             Static_Reaction := Static_Reaction + 0;
                         end
                         else begin
-                            if (global_var(42) == 1) then begin
-                                Static_Reaction := Static_Reaction + -5 + (10 * has_trait(0, dude_obj, 39));
+                            if (global_var(GVAR_KARMA_BETRAYER) == 1) then begin
+                                Static_Reaction := Static_Reaction + -5 + (10 * has_trait(TRAIT_PERK, dude_obj, PERK_cult_of_personality));
                             end
                             else begin
-                                if (global_var(43) == 1) then begin
-                                    Static_Reaction := Static_Reaction + -10 + (20 * has_trait(0, dude_obj, 39));
+                                if (global_var(GVAR_KARMA_SWORD_OF_DESPAIR) == 1) then begin
+                                    Static_Reaction := Static_Reaction + -10 + (20 * has_trait(TRAIT_PERK, dude_obj, PERK_cult_of_personality));
                                 end
                                 else begin
-                                    if (global_var(44) == 1) then begin
-                                        Static_Reaction := Static_Reaction + -15 + (30 * has_trait(0, dude_obj, 39));
+                                    if (global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE) == 1) then begin
+                                        Static_Reaction := Static_Reaction + -15 + (30 * has_trait(TRAIT_PERK, dude_obj, PERK_cult_of_personality));
                                     end
                                     else begin
-                                        Static_Reaction := Static_Reaction + -20 + (40 * has_trait(0, dude_obj, 39));
+                                        Static_Reaction := Static_Reaction + -20 + (40 * has_trait(TRAIT_PERK, dude_obj, PERK_cult_of_personality));
                                     end
                                 end
                             end
@@ -614,29 +614,29 @@ begin
         debug_msg("Good Person Reaction Bonus == " + Static_Reaction);
     end
     if (Slavery_Tolerant == 0) then begin
-        if (global_var(11) == 1) then begin
-            Static_Reaction := Static_Reaction + (2 * -25) + (2 * 25 * has_trait(0, dude_obj, 39));
+        if (global_var(GVAR_REPUTATION_SLAVER) == 1) then begin
+            Static_Reaction := Static_Reaction + (2 * -25) + (2 * 25 * has_trait(TRAIT_PERK, dude_obj, PERK_cult_of_personality));
         end
     end
     else begin
         if (Slavery_Tolerant == 1) then begin
-            if (global_var(11) == 1) then begin
-                Static_Reaction := Static_Reaction + -25 + (25 * has_trait(0, dude_obj, 39));
+            if (global_var(GVAR_REPUTATION_SLAVER) == 1) then begin
+                Static_Reaction := Static_Reaction + -25 + (25 * has_trait(TRAIT_PERK, dude_obj, PERK_cult_of_personality));
             end
         end
         else begin
             if (Slavery_Tolerant == 3) then begin
-                if (global_var(11) == 1) then begin
+                if (global_var(GVAR_REPUTATION_SLAVER) == 1) then begin
                     Static_Reaction := Static_Reaction - -25;
                 end
             end
         end
     end
     debug_msg("Slaver + Aligned Reaction == " + Static_Reaction);
-    Static_Reaction := Static_Reaction + (10 * has_trait(0, dude_obj, 10));
+    Static_Reaction := Static_Reaction + (10 * has_trait(TRAIT_PERK, dude_obj, PERK_presence));
     debug_msg("Presence Reaction == " + Static_Reaction);
-    if (global_var(1) >= 2) then begin
-        Static_Reaction := Static_Reaction + -30 + (30 * has_trait(0, dude_obj, 39));
+    if (global_var(GVAR_CHILDKILLER_REPUTATION) >= 2) then begin
+        Static_Reaction := Static_Reaction + -30 + (30 * has_trait(TRAIT_PERK, dude_obj, PERK_cult_of_personality));
     end
     debug_msg("Childkiller Reaction == " + Static_Reaction);
     set_local_var(0, local_var(3) + Static_Reaction);
@@ -683,28 +683,28 @@ begin
         end
     end
     set_local_var(12, whereAmI());
-    if (global_var(207) == 1) then begin
+    if (global_var(GVAR_NCR_SCMERK_HOSTILE) == 1) then begin
         call Node998();
     end
     else begin
         if (local_var(5) == 1) then begin
-            set_global_var(209, 4);
+            set_global_var(GVAR_NCR_SCMERK_STATUS, 4);
             set_local_var(5, 0);
             start_gdialog(325, self_obj, 4, -1, -1);
             gsay_start;
             call Node019();
             gsay_end;
             end_dialogue;
-            gfade_out(1);
+            fade_out;
             move_to(dude_obj, 23752, 0);
             call ReturnWeapons();
             call unHideParty();
-            gfade_in(1);
+            fade_in;
         end
         else begin
             debug_msg("MERK: Current Status: " + local_var(12));
             if (local_var(12) == 1) then begin
-                if (global_var(209) == 0) then begin
+                if (global_var(GVAR_NCR_SCMERK_STATUS) == 0) then begin
                     call Node002();
                 end
                 else begin
@@ -717,7 +717,7 @@ begin
                 end
                 else begin
                     if (local_var(12) == 3) then begin
-                        if ((global_var(210) bwand 1) and not(global_var(210) bwand 2)) then begin
+                        if ((global_var(GVAR_NCR_SCMERK_SEED_STATUS) bwand 1) and not(global_var(GVAR_NCR_SCMERK_SEED_STATUS) bwand 2)) then begin
                             start_gdialog(325, self_obj, 4, -1, -1);
                             gsay_start;
                             call Node048();
@@ -725,7 +725,7 @@ begin
                             end_dialogue;
                         end
                         else begin
-                            if (global_var(209) == 3) then begin
+                            if (global_var(GVAR_NCR_SCMERK_STATUS) == 3) then begin
                                 start_gdialog(325, self_obj, 4, -1, -1);
                                 gsay_start;
                                 call Node043();
@@ -733,7 +733,7 @@ begin
                                 end_dialogue;
                             end
                             else begin
-                                if ((global_var(209) == 4) and not(global_var(210) bwand 2)) then begin
+                                if ((global_var(GVAR_NCR_SCMERK_STATUS) == 4) and not(global_var(GVAR_NCR_SCMERK_SEED_STATUS) bwand 2)) then begin
                                     start_gdialog(325, self_obj, 4, -1, -1);
                                     gsay_start;
                                     call Node042();
@@ -741,7 +741,7 @@ begin
                                     end_dialogue;
                                 end
                                 else begin
-                                    if ((global_var(210) bwand 16) and ((global_var(210) bwand 32) == 0) or ((global_var(210) bwand 64) and ((global_var(210) bwand 128) == 0))) then begin
+                                    if ((global_var(GVAR_NCR_SCMERK_SEED_STATUS) bwand 16) and ((global_var(GVAR_NCR_SCMERK_SEED_STATUS) bwand 32) == 0) or ((global_var(GVAR_NCR_SCMERK_SEED_STATUS) bwand 64) and ((global_var(GVAR_NCR_SCMERK_SEED_STATUS) bwand 128) == 0))) then begin
                                         start_gdialog(325, self_obj, 4, -1, -1);
                                         gsay_start;
                                         call Node006();
@@ -754,7 +754,7 @@ begin
                                         call Node005();
                                         gsay_end;
                                         end_dialogue;
-                                        set_global_var(206, 1);
+                                        set_global_var(GVAR_NCR_SCMERK_HEREBEFORE, 1);
                                     end
                                 end
                             end
@@ -767,9 +767,9 @@ begin
                     end
                 end
             end
-            if (global_var(189) == 2) then begin
+            if (global_var(GVAR_NCR_CAR_JACKED) == 2) then begin
                 move_obj_inven_to_obj(playerTmpBox2, party_member_obj(455));
-                set_global_var(189, 0);
+                set_global_var(GVAR_NCR_CAR_JACKED, 0);
             end
             if (local_var(4) == 1) then begin
                 call LennyFight();
@@ -780,58 +780,58 @@ end
 
 procedure destroy_p_proc
 begin
-    set_global_var(209, 1);
+    set_global_var(GVAR_NCR_SCMERK_STATUS, 1);
     debug_msg("MERK: should have set merk to be dead. Arg");
     if (source_obj == dude_obj) then begin
-        set_global_var(4, global_var(4) + 1);
-        if (metarule(51, self_obj) == 2) then begin
+        set_global_var(GVAR_BAD_MONSTER, global_var(GVAR_BAD_MONSTER) + 1);
+        if (critter_kill_type(self_obj) == KILL_TYPE_children_kills) then begin
             if (source_obj == dude_obj) then begin
-                set_global_var(2, 0);
-                set_global_var(1, global_var(1) + 1);
-                set_global_var(0, global_var(0) + -15);
+                set_global_var(GVAR_CHAMPION_REPUTATION, 0);
+                set_global_var(GVAR_CHILDKILLER_REPUTATION, global_var(GVAR_CHILDKILLER_REPUTATION) + 1);
+                set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + -15);
                 debug_msg("Player gains " + -15 + " Karma Points.");
-                set_global_var(37, 0);
-                set_global_var(38, 0);
-                set_global_var(39, 0);
-                set_global_var(40, 0);
-                set_global_var(41, 0);
-                set_global_var(42, 0);
-                set_global_var(43, 0);
-                set_global_var(44, 0);
-                set_global_var(45, 0);
-                if (global_var(0) >= 1000) then begin
-                    set_global_var(37, 1);
+                set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+                set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+                set_global_var(GVAR_KARMA_DEFENDER, 0);
+                set_global_var(GVAR_KARMA_WANDERER, 0);
+                set_global_var(GVAR_KARMA_BETRAYER, 0);
+                set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+                set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+                set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+                    set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
                 end
                 else begin
-                    if (global_var(0) >= 750) then begin
-                        set_global_var(38, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
                     end
                     else begin
-                        if (global_var(0) >= 500) then begin
-                            set_global_var(39, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                            set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                         end
                         else begin
-                            if (global_var(0) >= 250) then begin
-                                set_global_var(40, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                                set_global_var(GVAR_KARMA_DEFENDER, 1);
                             end
                             else begin
-                                if (global_var(0) > -250) then begin
-                                    set_global_var(41, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                                    set_global_var(GVAR_KARMA_WANDERER, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -500) then begin
-                                        set_global_var(42, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                        set_global_var(GVAR_KARMA_BETRAYER, 1);
                                     end
                                     else begin
-                                        if (global_var(0) > -750) then begin
-                                            set_global_var(43, 1);
+                                        if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                            set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                         end
                                         else begin
-                                            if (global_var(0) > -1000) then begin
-                                                set_global_var(44, 1);
+                                            if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                                set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                             end
                                             else begin
-                                                set_global_var(45, 1);
+                                                set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                             end
                                         end
                                     end
@@ -840,62 +840,62 @@ begin
                         end
                     end
                 end
-                set_global_var(57, global_var(57) + -8);
+                set_global_var(GVAR_TOWN_REP_NCR, global_var(GVAR_TOWN_REP_NCR) + -8);
                 debug_msg("Added " + -8 + " to Town Rep");
             end
         end
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(5) > (2 * global_var(4))) or (global_var(3) == 1))) then begin
-            set_global_var(3, 1);
-            set_global_var(2, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
+            set_global_var(GVAR_BERSERKER_REPUTATION, 1);
+            set_global_var(GVAR_CHAMPION_REPUTATION, 0);
         end
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(4) > (3 * global_var(5))) or (global_var(2) == 1)) and (global_var(1) == 0)) then begin
-            set_global_var(2, 1);
-            set_global_var(3, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(4) > (3 * global_var(GVAR_GOOD_MONSTER))) or (global_var(GVAR_CHAMPION_REPUTATION) == 1)) and (global_var(GVAR_CHILDKILLER_REPUTATION) == 0)) then begin
+            set_global_var(GVAR_CHAMPION_REPUTATION, 1);
+            set_global_var(GVAR_BERSERKER_REPUTATION, 0);
         end
-        set_global_var(0, global_var(0) + 5);
+        set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + 5);
         debug_msg("Player gains " + 5 + " Karma Points.");
-        set_global_var(37, 0);
-        set_global_var(38, 0);
-        set_global_var(39, 0);
-        set_global_var(40, 0);
-        set_global_var(41, 0);
-        set_global_var(42, 0);
-        set_global_var(43, 0);
-        set_global_var(44, 0);
-        set_global_var(45, 0);
-        if (global_var(0) >= 1000) then begin
-            set_global_var(37, 1);
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+        set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+        set_global_var(GVAR_KARMA_DEFENDER, 0);
+        set_global_var(GVAR_KARMA_WANDERER, 0);
+        set_global_var(GVAR_KARMA_BETRAYER, 0);
+        set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+        set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+            set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
         end
         else begin
-            if (global_var(0) >= 750) then begin
-                set_global_var(38, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
             end
             else begin
-                if (global_var(0) >= 500) then begin
-                    set_global_var(39, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                 end
                 else begin
-                    if (global_var(0) >= 250) then begin
-                        set_global_var(40, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                        set_global_var(GVAR_KARMA_DEFENDER, 1);
                     end
                     else begin
-                        if (global_var(0) > -250) then begin
-                            set_global_var(41, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                            set_global_var(GVAR_KARMA_WANDERER, 1);
                         end
                         else begin
-                            if (global_var(0) > -500) then begin
-                                set_global_var(42, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                set_global_var(GVAR_KARMA_BETRAYER, 1);
                             end
                             else begin
-                                if (global_var(0) > -750) then begin
-                                    set_global_var(43, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -1000) then begin
-                                        set_global_var(44, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                     end
                                     else begin
-                                        set_global_var(45, 1);
+                                        set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                     end
                                 end
                             end
@@ -904,7 +904,7 @@ begin
                 end
             end
         end
-        set_global_var(57, global_var(57) + 2);
+        set_global_var(GVAR_TOWN_REP_NCR, global_var(GVAR_TOWN_REP_NCR) + 2);
         debug_msg("Added " + 2 + " to Town Rep");
     end
 end
@@ -912,18 +912,18 @@ end
 procedure look_at_p_proc
 begin
     script_overrides;
-    if (global_var(206) == 0) then begin
-        display_msg(message_str(325, 100));
+    if (global_var(GVAR_NCR_SCMERK_HEREBEFORE) == 0) then begin
+        display_mstr(100);
     end
     else begin
-        display_msg(message_str(325, 101));
+        display_mstr(101);
     end
 end
 
 procedure description_p_proc
 begin
     script_overrides;
-    display_msg(message_str(325, 102));
+    display_mstr(102);
 end
 
 procedure use_skill_on_p_proc
@@ -933,7 +933,7 @@ end
 procedure damage_p_proc
 begin
     if (source_obj == dude_obj) then begin
-        set_global_var(208, 1);
+        set_global_var(GVAR_NCR_SCMERK_PERSONAL_ENEMY, 1);
     end
 end
 
@@ -943,17 +943,17 @@ begin
     Only_Once := 0;
     merk_visible := 1;
     set_local_var(14, 0);
-    if (global_var(209) == 1) then begin
+    if (global_var(GVAR_NCR_SCMERK_STATUS) == 1) then begin
         debug_msg("MERK: Killing merk because he was killed on another map.");
         move_to(self_obj, 7000, 0);
         destroy_object(self_obj);
     end
     else begin
-        critter_add_trait(self_obj, 1, 6, 56);
-        critter_add_trait(self_obj, 1, 5, 24);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 56);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_TOUGH_CITIZEN);
         set_local_var(6, loadMapPosition());
         if (local_var(6)) then begin
-            if (metarule(22, 0) == 0) then begin
+            if (is_loading_game == 0) then begin
                 debug_msg("MERK: Should be visible");
                 set_obj_visibility(self_obj, 0);
                 merk_visible := 1;
@@ -962,7 +962,7 @@ begin
         end
         else begin
             debug_msg("MERK: Should be invisible");
-            if (metarule(22, 0) == 0) then begin
+            if (is_loading_game == 0) then begin
                 set_obj_visibility(self_obj, 1);
             end
             merk_visible := 0;
@@ -1012,7 +1012,7 @@ begin
         end
         else begin
             if (fixed_param == 3) then begin
-                if (cur_map_index == 46) then begin
+                if (CUR_MAP_NCRENT) then begin
                     if (tile_num(self_obj) == 30134) then begin
                         set_obj_visibility(self_obj, 1);
                         merk_visible := 0;
@@ -1022,7 +1022,7 @@ begin
                     end
                 end
                 else begin
-                    if (cur_map_index == 43) then begin
+                    if (CUR_MAP_NCR2) then begin
                         if (tile_num(self_obj) == 14096) then begin
                             set_obj_visibility(self_obj, 1);
                             merk_visible := 0;
@@ -1072,8 +1072,8 @@ end
 
 procedure Node998
 begin
-    set_global_var(207, 2);
-    set_global_var(209, 3);
+    set_global_var(GVAR_NCR_SCMERK_HOSTILE, 2);
+    set_global_var(GVAR_NCR_SCMERK_STATUS, 3);
 end
 
 procedure Node999
@@ -1082,11 +1082,11 @@ end
 
 procedure Node001
 begin
-    if (global_var(232) > 0) then begin
-        float_msg(self_obj, message_str(325, 103), 0);
+    if (global_var(GVAR_NEW_RENO_PORN_STAR) > 0) then begin
+        float_msg(self_obj, mstr(103), FLOAT_MSG_NORMAL);
     end
     else begin
-        float_msg(self_obj, message_str(325, 104), 8);
+        float_msg(self_obj, mstr(104), FLOAT_MSG_YELLOW);
         add_timer_event(self_obj, game_ticks(5), 5);
     end
 end
@@ -1095,22 +1095,22 @@ procedure Node002
 begin
     variable LVar0 := 105;
     LVar0 := LVar0 + random(0, 8);
-    if ((LVar0 == 107) and not(global_var(232) > 0)) then begin
+    if ((LVar0 == 107) and not(global_var(GVAR_NEW_RENO_PORN_STAR) > 0)) then begin
         LVar0 := 105;
     end
-    if ((LVar0 == 108) and not(global_var(1) >= 2)) then begin
+    if ((LVar0 == 108) and not(global_var(GVAR_CHILDKILLER_REPUTATION) >= 2)) then begin
         LVar0 := 105;
     end
     else begin
-        set_global_var(209, 2);
+        set_global_var(GVAR_NCR_SCMERK_STATUS, 2);
     end
-    if ((LVar0 == 109) and (global_var(0) < 250)) then begin
+    if ((LVar0 == 109) and (global_var(GVAR_PLAYER_REPUTATION) < 250)) then begin
         LVar0 := 105;
     end
-    if ((LVar0 == 113) and (get_critter_stat(dude_obj, 34) == 0)) then begin
+    if ((LVar0 == 113) and (dude_gender == 0)) then begin
         LVar0 := 105;
     end
-    float_msg(self_obj, message_str(325, LVar0), 0);
+    float_msg(self_obj, mstr(LVar0), FLOAT_MSG_NORMAL);
     return 0;
     LVar0 := 0;
 end
@@ -1119,177 +1119,177 @@ procedure Node003
 begin
     variable LVar0 := 114;
     LVar0 := LVar0 + random(0, 6);
-    if ((LVar0 == 115) and not(global_var(1) >= 2)) then begin
+    if ((LVar0 == 115) and not(global_var(GVAR_CHILDKILLER_REPUTATION) >= 2)) then begin
         LVar0 := 114;
     end
-    if ((LVar0 == 116) and (global_var(0) < 250)) then begin
+    if ((LVar0 == 116) and (global_var(GVAR_PLAYER_REPUTATION) < 250)) then begin
         LVar0 := 114;
     end
-    if ((LVar0 == 117) and (get_critter_stat(dude_obj, 34) == 0)) then begin
+    if ((LVar0 == 117) and (dude_gender == 0)) then begin
         LVar0 := 114;
     end
-    float_msg(self_obj, message_str(325, LVar0), 0);
+    float_msg(self_obj, mstr(LVar0), FLOAT_MSG_NORMAL);
     return 0;
     LVar0 := 0;
 end
 
 procedure Node004
 begin
-    float_msg(self_obj, message_str(325, 121 + random(0, 3)), 0);
+    float_msg(self_obj, mstr(121 + random(0, 3)), FLOAT_MSG_NORMAL);
 end
 
 procedure Node005
 begin
-    if ((global_var(209) == 2) or (global_var(206) == 1)) then begin
-        gsay_reply(325, 126);
+    if ((global_var(GVAR_NCR_SCMERK_STATUS) == 2) or (global_var(GVAR_NCR_SCMERK_HEREBEFORE) == 1)) then begin
+        Reply(126);
     end
     else begin
-        if ((local_var(1) <= -2) or (global_var(209) == 3)) then begin
-            gsay_reply(325, 127);
+        if ((local_var(1) <= -2) or (global_var(GVAR_NCR_SCMERK_STATUS) == 3)) then begin
+            Reply(127);
         end
         else begin
-            gsay_reply(325, 125);
+            Reply(125);
         end
     end
-    giq_option(-3, 325, 128, Node007, 50);
-    if (not(global_var(209) == 4)) then begin
-        giq_option(4, 325, 129, Node008, 50);
+    giq_option(-3, NAME, 128, Node007, NEUTRAL_REACTION);
+    if (not(global_var(GVAR_NCR_SCMERK_STATUS) == 4)) then begin
+        giq_option(4, NAME, 129, Node008, NEUTRAL_REACTION);
     end
-    if (global_var(190) == 1) then begin
-        giq_option(4, 325, 130, Node009, 50);
+    if (global_var(GVAR_NCR_MERK_WORK) == 1) then begin
+        giq_option(4, NAME, 130, Node009, NEUTRAL_REACTION);
     end
-    if (global_var(189) == 1) then begin
-        giq_option(4, 325, 131, Node1011a, 51);
+    if (global_var(GVAR_NCR_CAR_JACKED) == 1) then begin
+        giq_option(4, NAME, 131, Node1011a, BAD_REACTION);
     end
-    giq_option(4, 325, 132, Node020, 50);
-    if (global_var(189) == 1) then begin
-        giq_option(7, 325, 133, Node1211a, 50);
+    giq_option(4, NAME, 132, Node020, NEUTRAL_REACTION);
+    if (global_var(GVAR_NCR_CAR_JACKED) == 1) then begin
+        giq_option(7, NAME, 133, Node1211a, NEUTRAL_REACTION);
     end
-    if (global_var(213) == 2) then begin
-        giq_option(4, 325, 266, Node037, 50);
+    if (global_var(GVAR_NCR_FAKE_VAULT13_MAP) == 2) then begin
+        giq_option(4, NAME, 266, Node037, NEUTRAL_REACTION);
     end
 end
 
 procedure Node006
 begin
-    gsay_reply(325, 134);
-    giq_option(-3, 325, 135, Node044, 50);
-    giq_option(4, 325, 136, Node045, 50);
-    giq_option(4, 325, 137, Node046, 50);
+    Reply(134);
+    giq_option(-3, NAME, 135, Node044, NEUTRAL_REACTION);
+    giq_option(4, NAME, 136, Node045, NEUTRAL_REACTION);
+    giq_option(4, NAME, 137, Node046, NEUTRAL_REACTION);
 end
 
 procedure Node007
 begin
-    if (get_critter_stat(dude_obj, 0) > 6) then begin
-        gsay_reply(325, 138);
+    if (dude_strength > 6) then begin
+        Reply(138);
     end
     else begin
-        gsay_reply(325, 139);
+        Reply(139);
     end
-    giq_option(0, 325, message_str(14, 350), Node999, 50);
+    giq_option(0, NAME, g_mstr(350), Node999, NEUTRAL_REACTION);
 end
 
 procedure Node008
 begin
-    debug_msg("SCMERK: " + (global_var(210) bwand 128));
-    gsay_reply(325, 140);
-    if ((global_var(211) == 1) and (map_var(2) == 0)) then begin
-        giq_option(4, 325, 141, Node014, 51);
+    debug_msg("SCMERK: " + (global_var(GVAR_NCR_SCMERK_SEED_STATUS) bwand 128));
+    Reply(140);
+    if ((global_var(GVAR_NCR_LENNY_MET) == 1) and (map_var(2) == 0)) then begin
+        giq_option(4, NAME, 141, Node014, BAD_REACTION);
     end
-    if (not(global_var(210) bwand 128)) then begin
-        giq_option(4, 325, 142, Node016, 50);
+    if (not(global_var(GVAR_NCR_SCMERK_SEED_STATUS) bwand 128)) then begin
+        giq_option(4, NAME, 142, Node016, NEUTRAL_REACTION);
     end
     if (party_member_obj(16777376) != 0) then begin
-        giq_option(4, 325, 143, Node017, 50);
+        giq_option(4, NAME, 143, Node017, NEUTRAL_REACTION);
     end
-    giq_option(4, 325, message_str(325, 144) + " " + obj_name(dude_obj) + message_str(325, 258), Node2122a, 50);
-    giq_option(4, 325, 145, Node999a, 50);
+    giq_option(4, NAME, mstr(144) + " " + dude_name + mstr(258), Node2122a, NEUTRAL_REACTION);
+    giq_option(4, NAME, 145, Node999a, NEUTRAL_REACTION);
 end
 
 procedure Node009
 begin
-    gsay_reply(325, 146);
-    giq_option(4, 325, 147, Node999b, 50);
-    giq_option(4, 325, 148, Node022, 50);
+    Reply(146);
+    giq_option(4, NAME, 147, Node999b, NEUTRAL_REACTION);
+    giq_option(4, NAME, 148, Node022, NEUTRAL_REACTION);
 end
 
 procedure Node010
 begin
-    gsay_reply(325, 149);
-    giq_option(0, 325, message_str(14, 350), Node998, 50);
+    Reply(149);
+    giq_option(0, NAME, g_mstr(350), Node998, NEUTRAL_REACTION);
 end
 
 procedure Node011
 begin
-    gsay_reply(325, 150);
-    giq_option(4, 325, 151, Node2413a, 51);
-    giq_option(4, 325, 152, Node1312a, 50);
-    giq_option(4, 325, 153, Node999, 50);
+    Reply(150);
+    giq_option(4, NAME, 151, Node2413a, BAD_REACTION);
+    giq_option(4, NAME, 152, Node1312a, NEUTRAL_REACTION);
+    giq_option(4, NAME, 153, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node012
 begin
-    gsay_reply(325, 154);
-    if (get_critter_stat(dude_obj, 15) > 60) then begin
-        giq_option(4, 325, 155, Node25a, 50);
+    Reply(154);
+    if (get_critter_stat(dude_obj, STAT_crit_chance) > 60) then begin
+        giq_option(4, NAME, 155, Node25a, NEUTRAL_REACTION);
     end
-    giq_option(4, 325, 156, Node2413a, 51);
-    giq_option(4, 325, 157, Node027, 50);
+    giq_option(4, NAME, 156, Node2413a, BAD_REACTION);
+    giq_option(4, NAME, 157, Node027, NEUTRAL_REACTION);
 end
 
 procedure Node013
 begin
-    gsay_reply(325, 158);
-    giq_option(4, 325, message_str(14, 350), Node998, 50);
+    Reply(158);
+    giq_option(4, NAME, g_mstr(350), Node998, NEUTRAL_REACTION);
 end
 
 procedure Node014
 begin
-    gsay_reply(325, 159);
-    giq_option(4, 325, 160, Node015, 50);
-    giq_option(4, 325, 161, Node999, 50);
+    Reply(159);
+    giq_option(4, NAME, 160, Node015, NEUTRAL_REACTION);
+    giq_option(4, NAME, 161, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node015
 begin
     variable LVar0 := 0;
-    if (get_critter_stat(dude_obj, 34) == 0) then begin
-        LVar0 := message_str(325, 259);
+    if (dude_gender == 0) then begin
+        LVar0 := mstr(259);
     end
     else begin
-        LVar0 := message_str(325, 260);
+        LVar0 := mstr(260);
     end
-    gsay_reply(325, message_str(325, 162) + LVar0);
-    giq_option(4, 325, message_str(14, 350), Node997, 50);
+    Reply(mstr(162) + LVar0);
+    giq_option(4, NAME, g_mstr(350), Node997, NEUTRAL_REACTION);
     return 0;
     LVar0 := 0;
 end
 
 procedure Node016
 begin
-    gsay_reply(325, 163);
-    giq_option(4, 325, 164, Node028, 50);
+    Reply(163);
+    giq_option(4, NAME, 164, Node028, NEUTRAL_REACTION);
 end
 
 procedure Node017
 begin
-    gsay_reply(325, 165);
-    giq_option(4, 325, 166, Node018, 49);
-    giq_option(4, 325, 167, Node023, 50);
+    Reply(165);
+    giq_option(4, NAME, 166, Node018, GOOD_REACTION);
+    giq_option(4, NAME, 167, Node023, NEUTRAL_REACTION);
 end
 
 procedure Node018
 begin
-    gsay_reply(325, 168);
-    giq_option(4, 325, 169, Node19a, 50);
-    giq_option(4, 325, 170, Node999, 50);
+    Reply(168);
+    giq_option(4, NAME, 169, Node19a, NEUTRAL_REACTION);
+    giq_option(4, NAME, 170, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node019
 begin
-    gsay_reply(325, 171);
-    giq_option(4, 325, 172, Node39a, 50);
-    giq_option(4, 325, 173, Node023, 50);
+    Reply(171);
+    giq_option(4, NAME, 172, Node39a, NEUTRAL_REACTION);
+    giq_option(4, NAME, 173, Node023, NEUTRAL_REACTION);
 end
 
 procedure Node020
@@ -1299,315 +1299,315 @@ begin
     end
     else begin
         if (Node020Repeat == 0) then begin
-            gsay_reply(325, 174);
+            Reply(174);
         end
         else begin
-            gsay_reply(325, 175);
+            Reply(175);
         end
         if (item_caps_total(dude_obj) >= 50) then begin
-            giq_option(4, 325, 176, Node20Rep, 50);
+            giq_option(4, NAME, 176, Node20Rep, NEUTRAL_REACTION);
         end
-        giq_option(4, 325, 177, Node999, 50);
+        giq_option(4, NAME, 177, Node999, NEUTRAL_REACTION);
     end
 end
 
 procedure Node021
 begin
-    gsay_reply(325, 178);
-    giq_option(0, 325, message_str(14, 352), Node019, 50);
+    Reply(178);
+    giq_option(0, NAME, g_mstr(352), Node019, NEUTRAL_REACTION);
 end
 
 procedure Node022
 begin
-    set_global_var(209, 3);
-    gsay_reply(325, 180);
-    giq_option(0, 325, message_str(14, 350), Node999, 50);
+    set_global_var(GVAR_NCR_SCMERK_STATUS, 3);
+    Reply(180);
+    giq_option(0, NAME, g_mstr(350), Node999, NEUTRAL_REACTION);
 end
 
 procedure Node023
 begin
-    set_global_var(209, 3);
-    gsay_reply(325, 181);
-    giq_option(0, 325, message_str(14, 350), Node999, 50);
+    set_global_var(GVAR_NCR_SCMERK_STATUS, 3);
+    Reply(181);
+    giq_option(0, NAME, g_mstr(350), Node999, NEUTRAL_REACTION);
 end
 
 procedure Node024
 begin
-    gsay_reply(325, 182);
-    giq_option(4, 325, 183, Node999c, 50);
-    giq_option(4, 325, 184, Node998, 51);
+    Reply(182);
+    giq_option(4, NAME, 183, Node999c, NEUTRAL_REACTION);
+    giq_option(4, NAME, 184, Node998, BAD_REACTION);
 end
 
 procedure Node025
 begin
-    gsay_reply(325, 185);
-    giq_option(0, 325, message_str(14, 352), NodeBarter, 50);
+    Reply(185);
+    giq_option(0, NAME, g_mstr(352), NodeBarter, NEUTRAL_REACTION);
 end
 
 procedure Node026
 begin
-    gsay_reply(325, 186);
-    giq_option(0, 325, message_str(14, 350), Node999, 50);
+    Reply(186);
+    giq_option(0, NAME, g_mstr(350), Node999, NEUTRAL_REACTION);
 end
 
 procedure Node027
 begin
-    gsay_reply(325, 187);
-    giq_option(4, 325, 188, Node4041a, 50);
-    giq_option(4, 325, 189, Node026, 50);
+    Reply(187);
+    giq_option(4, NAME, 188, Node4041a, NEUTRAL_REACTION);
+    giq_option(4, NAME, 189, Node026, NEUTRAL_REACTION);
 end
 
 procedure Node028
 begin
-    gsay_reply(325, 190);
-    giq_option(4, 325, 191, Node055d, 49);
-    if (global_var(212) == 1) then begin
-        giq_option(4, 325, 192, Node998, 51);
+    Reply(190);
+    giq_option(4, NAME, 191, Node055d, GOOD_REACTION);
+    if (global_var(GVAR_NCR_ELRON_ADJUST) == 1) then begin
+        giq_option(4, NAME, 192, Node998, BAD_REACTION);
     end
-    giq_option(4, 325, 193, Node023, 50);
+    giq_option(4, NAME, 193, Node023, NEUTRAL_REACTION);
 end
 
 procedure Node029
 begin
     if (Node029Repeat == 0) then begin
-        gsay_reply(325, 194);
+        Reply(194);
     end
     else begin
-        gsay_reply(325, 195);
+        Reply(195);
     end
-    if (not(metarule(17, 11) != 0)) then begin
-        giq_option(4, 325, 196, Node3731a, 50);
+    if (not(town_known(11) != 0)) then begin
+        giq_option(4, NAME, 196, Node3731a, NEUTRAL_REACTION);
     end
-    if (global_var(233) == 0) then begin
-        giq_option(4, 325, 197, Node32a, 50);
+    if (global_var(GVAR_VAULT13_FOUND_GECK) == 0) then begin
+        giq_option(4, NAME, 197, Node32a, NEUTRAL_REACTION);
     end
-    if ((global_var(243) bwand 1) or (global_var(243) bwand 4) or (global_var(243) bwand 8) or global_var(196)) then begin
-        giq_option(4, 325, 198, Node33a, 50);
+    if ((global_var(GVAR_8_BALL_TOILET_SECRET) bwand 1) or (global_var(GVAR_8_BALL_TOILET_SECRET) bwand 4) or (global_var(GVAR_8_BALL_TOILET_SECRET) bwand 8) or global_var(GVAR_NCR_RANGERS_KNOWN)) then begin
+        giq_option(4, NAME, 198, Node33a, NEUTRAL_REACTION);
     end
-    if (((global_var(174) bwand 1) != 0) and not((global_var(174) bwand 2) != 0)) then begin
-        giq_option(4, 325, 199, Node34a, 50);
+    if (((global_var(GVAR_NEWRENO_SNUFF_WESTIN) bwand 1) != 0) and not((global_var(GVAR_NEWRENO_SNUFF_WESTIN) bwand 2) != 0)) then begin
+        giq_option(4, NAME, 199, Node34a, NEUTRAL_REACTION);
     end
-    if (((global_var(175) bwand 1) != 0) and not((global_var(175) bwand 2) != 0)) then begin
-        giq_option(4, 325, 200, Node35a, 50);
+    if (((global_var(GVAR_NEWRENO_SNUFF_CARLSON) bwand 1) != 0) and not((global_var(GVAR_NEWRENO_SNUFF_CARLSON) bwand 2) != 0)) then begin
+        giq_option(4, NAME, 200, Node35a, NEUTRAL_REACTION);
     end
-    giq_option(4, 325, 201, Node999, 50);
+    giq_option(4, NAME, 201, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node030
 begin
-    gsay_reply(325, 202);
+    Reply(202);
     if (item_caps_total(dude_obj) >= 50) then begin
-        giq_option(4, 325, 203, Node29a, 50);
+        giq_option(4, NAME, 203, Node29a, NEUTRAL_REACTION);
     end
-    giq_option(4, 325, 204, Node999, 50);
+    giq_option(4, NAME, 204, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node031
 begin
-    gsay_reply(325, 205);
+    Reply(205);
     if (item_caps_total(dude_obj) > 999) then begin
-        giq_option(4, 325, 206, Node36a, 50);
+        giq_option(4, NAME, 206, Node36a, NEUTRAL_REACTION);
     end
-    giq_option(7, 325, 207, Node037, 50);
-    if (get_critter_stat(dude_obj, 4) <= 6) then begin
-        giq_option(1, 325, 208, Node036, 50);
+    giq_option(7, NAME, 207, Node037, NEUTRAL_REACTION);
+    if (dude_iq <= 6) then begin
+        giq_option(1, NAME, 208, Node036, NEUTRAL_REACTION);
     end
 end
 
 procedure Node032
 begin
-    gsay_reply(325, 209);
-    giq_option(4, 325, 210, Node030, 50);
-    giq_option(4, 325, 211, Node999, 50);
+    Reply(209);
+    giq_option(4, NAME, 210, Node030, NEUTRAL_REACTION);
+    giq_option(4, NAME, 211, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node033
 begin
-    gsay_reply(325, 212);
-    giq_option(4, 325, 213, Node030, 50);
-    giq_option(4, 325, 214, Node999, 50);
+    Reply(212);
+    giq_option(4, NAME, 213, Node030, NEUTRAL_REACTION);
+    giq_option(4, NAME, 214, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node034
 begin
-    gsay_reply(325, 215);
-    giq_option(4, 325, 216, Node030, 50);
-    giq_option(4, 325, 217, Node999, 50);
+    Reply(215);
+    giq_option(4, NAME, 216, Node030, NEUTRAL_REACTION);
+    giq_option(4, NAME, 217, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node035
 begin
-    gsay_reply(325, 218);
-    if (not(local_var(1) <= -2) and (global_var(209) != 3)) then begin
-        giq_option(4, 325, 219, Node3836a, 50);
+    Reply(218);
+    if (not(local_var(1) <= -2) and (global_var(GVAR_NCR_SCMERK_STATUS) != 3)) then begin
+        giq_option(4, NAME, 219, Node3836a, NEUTRAL_REACTION);
     end
-    giq_option(4, 325, 220, Node999, 50);
+    giq_option(4, NAME, 220, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node036
 begin
-    gsay_reply(325, 221);
-    giq_option(4, 325, 222, Node030, 50);
-    giq_option(4, 325, 223, Node999, 50);
+    Reply(221);
+    giq_option(4, NAME, 222, Node030, NEUTRAL_REACTION);
+    giq_option(4, NAME, 223, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node037
 begin
-    if (global_var(213) == 0) then begin
-        gsay_reply(325, 224);
+    if (global_var(GVAR_NCR_FAKE_VAULT13_MAP) == 0) then begin
+        Reply(224);
     end
     else begin
-        if (global_var(213) == 2) then begin
-            gsay_reply(325, 226);
+        if (global_var(GVAR_NCR_FAKE_VAULT13_MAP) == 2) then begin
+            Reply(226);
         end
         else begin
-            if (global_var(213) == 1) then begin
-                gsay_reply(325, 225);
+            if (global_var(GVAR_NCR_FAKE_VAULT13_MAP) == 1) then begin
+                Reply(225);
             end
         end
     end
-    giq_option(4, 325, 227, Node030, 50);
-    giq_option(4, 325, 228, Node999, 50);
-    if (global_var(213) == 2) then begin
-        giq_option(4, 325, 267, Node37a, 51);
+    giq_option(4, NAME, 227, Node030, NEUTRAL_REACTION);
+    giq_option(4, NAME, 228, Node999, NEUTRAL_REACTION);
+    if (global_var(GVAR_NCR_FAKE_VAULT13_MAP) == 2) then begin
+        giq_option(4, NAME, 267, Node37a, BAD_REACTION);
     end
 end
 
 procedure Node038
 begin
-    gsay_reply(325, 229);
-    giq_option(4, 325, 230, NodeCarlsonSec, 50);
-    giq_option(4, 325, 231, Node999, 50);
+    Reply(229);
+    giq_option(4, NAME, 230, NodeCarlsonSec, NEUTRAL_REACTION);
+    giq_option(4, NAME, 231, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node039
 begin
-    gsay_reply(325, 232);
-    giq_option(0, 325, message_str(14, 350), Node999, 50);
+    Reply(232);
+    giq_option(0, NAME, g_mstr(350), Node999, NEUTRAL_REACTION);
 end
 
 procedure Node040
 begin
-    gsay_reply(325, 233);
-    giq_option(0, 325, message_str(14, 350), Node998, 50);
+    Reply(233);
+    giq_option(0, NAME, g_mstr(350), Node998, NEUTRAL_REACTION);
 end
 
 procedure Node041
 begin
-    set_global_var(209, 3);
-    gsay_reply(325, 234);
-    giq_option(0, 325, message_str(14, 350), Node999, 50);
+    set_global_var(GVAR_NCR_SCMERK_STATUS, 3);
+    Reply(234);
+    giq_option(0, NAME, g_mstr(350), Node999, NEUTRAL_REACTION);
 end
 
 procedure Node042
 begin
-    gsay_reply(325, 235);
-    giq_option(4, 325, 236, Node019, 50);
-    giq_option(4, 325, 237, Node047, 50);
-    giq_option(4, 325, 264, Node020, 50);
-    set_global_var(209, 4);
+    Reply(235);
+    giq_option(4, NAME, 236, Node019, NEUTRAL_REACTION);
+    giq_option(4, NAME, 237, Node047, NEUTRAL_REACTION);
+    giq_option(4, NAME, 264, Node020, NEUTRAL_REACTION);
+    set_global_var(GVAR_NCR_SCMERK_STATUS, 4);
 end
 
 procedure Node043
 begin
-    gsay_reply(325, 238);
-    giq_option(4, 325, 239, Node522a, 50);
-    giq_option(4, 325, 240, Node013, 50);
+    Reply(238);
+    giq_option(4, NAME, 239, Node522a, NEUTRAL_REACTION);
+    giq_option(4, NAME, 240, Node013, NEUTRAL_REACTION);
 end
 
 procedure Node044
 begin
-    gsay_reply(325, 241);
-    giq_option(0, 325, message_str(14, 350), Node999, 50);
+    Reply(241);
+    giq_option(0, NAME, g_mstr(350), Node999, NEUTRAL_REACTION);
 end
 
 procedure Node045
 begin
-    gsay_reply(325, 242);
-    giq_option(0, 325, message_str(14, 350), Node999, 50);
+    Reply(242);
+    giq_option(0, NAME, g_mstr(350), Node999, NEUTRAL_REACTION);
 end
 
 procedure Node046
 begin
-    if ((global_var(210) bwand 1) and not(global_var(210) bwand 2)) then begin
-        gsay_reply(325, 263);
+    if ((global_var(GVAR_NCR_SCMERK_SEED_STATUS) bwand 1) and not(global_var(GVAR_NCR_SCMERK_SEED_STATUS) bwand 2)) then begin
+        Reply(263);
     end
     else begin
-        gsay_reply(325, 243);
+        Reply(243);
     end
-    giq_option(0, 325, message_str(14, 350), Node999, 50);
+    giq_option(0, NAME, g_mstr(350), Node999, NEUTRAL_REACTION);
 end
 
 procedure Node047
 begin
-    set_global_var(209, 3);
-    gsay_reply(325, 244);
-    giq_option(0, 325, message_str(14, 350), Node999, 50);
+    set_global_var(GVAR_NCR_SCMERK_STATUS, 3);
+    Reply(244);
+    giq_option(0, NAME, g_mstr(350), Node999, NEUTRAL_REACTION);
 end
 
 procedure Node048
 begin
-    gsay_reply(325, 245);
+    Reply(245);
     if (obj_is_carrying_obj_pid(dude_obj, 340)) then begin
-        giq_option(4, 325, 246, Node049, 50);
+        giq_option(4, NAME, 246, Node049, NEUTRAL_REACTION);
     end
-    giq_option(4, 325, 247, Node046, 50);
+    giq_option(4, NAME, 247, Node046, NEUTRAL_REACTION);
 end
 
 procedure Node049
 begin
-    gsay_reply(325, 248);
-    giq_option(0, 325, message_str(14, 352), Node5051a, 50);
+    Reply(248);
+    giq_option(0, NAME, g_mstr(352), Node5051a, NEUTRAL_REACTION);
 end
 
 procedure Node050
 begin
-    gsay_reply(325, 249);
-    giq_option(4, 325, 250, Node5253a, 50);
-    giq_option(4, 325, 251, Node053, 50);
+    Reply(249);
+    giq_option(4, NAME, 250, Node5253a, NEUTRAL_REACTION);
+    giq_option(4, NAME, 251, Node053, NEUTRAL_REACTION);
 end
 
 procedure Node051
 begin
-    gsay_reply(325, 252);
-    giq_option(4, 325, 253, Node999, 50);
+    Reply(252);
+    giq_option(4, NAME, 253, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node052
 begin
     item_caps_adjust(dude_obj, -500);
     item_caps_adjust(self_obj, 500);
-    gsay_reply(325, 254);
-    giq_option(0, 325, message_str(14, 350), Node999, 50);
+    Reply(254);
+    giq_option(0, NAME, g_mstr(350), Node999, NEUTRAL_REACTION);
 end
 
 procedure Node053
 begin
-    gsay_reply(325, 255);
-    giq_option(0, 325, message_str(14, 350), Node998, 50);
+    Reply(255);
+    giq_option(0, NAME, g_mstr(350), Node998, NEUTRAL_REACTION);
 end
 
 procedure Node054
 begin
-    if (global_var(210) bwand 4) then begin
-        float_msg(self_obj, message_str(325, 256), 8);
+    if (global_var(GVAR_NCR_SCMERK_SEED_STATUS) bwand 4) then begin
+        float_msg(self_obj, mstr(256), FLOAT_MSG_YELLOW);
         call doMerkAttack();
     end
     else begin
-        float_msg(self_obj, message_str(325, 257), 0);
+        float_msg(self_obj, mstr(257), FLOAT_MSG_NORMAL);
     end
 end
 
 procedure Node055
 begin
-    gsay_reply(325, 261);
-    giq_option(4, 325, 262, Node999, 50);
+    Reply(261);
+    giq_option(4, NAME, 262, Node999, NEUTRAL_REACTION);
 end
 
 procedure loadMapPosition
 begin
     set_local_var(9, 0);
-    if (cur_map_index == 46) then begin
+    if (CUR_MAP_NCRENT) then begin
         if (checktime() == 46) then begin
             move_to(self_obj, 22154, 0);
             anim(self_obj, 1000, 1);
@@ -1615,7 +1615,7 @@ begin
         end
     end
     else begin
-        if (cur_map_index == 43) then begin
+        if (CUR_MAP_NCR2) then begin
             if (checktime() == 42) then begin
                 move_to(self_obj, 16337, 0);
                 anim(self_obj, 1000, 1);
@@ -1637,7 +1637,7 @@ begin
     debug_msg("MERK: enter map called");
     set_obj_visibility(self_obj, 0);
     merk_visible := 1;
-    if (cur_map_index == 43) then begin
+    if (CUR_MAP_NCR2) then begin
         if (local_var(14) == 19518) then begin
             debug_msg("MERK: enter council map");
             reg_anim_func(2, self_obj);
@@ -1676,7 +1676,7 @@ begin
         end
     end
     else begin
-        if (cur_map_index == 46) then begin
+        if (CUR_MAP_NCRENT) then begin
             debug_msg("MERK: should be moving to bazaar pos");
             reg_anim_func(2, self_obj);
             reg_anim_func(1, 1);
@@ -1693,7 +1693,7 @@ end
 procedure exitMap
 begin
     debug_msg("MERK: Should be exiting map");
-    if (cur_map_index == 46) then begin
+    if (CUR_MAP_NCRENT) then begin
         debug_msg("MERK: leave bazaar map");
         reg_anim_func(2, self_obj);
         reg_anim_func(1, 1);
@@ -1703,7 +1703,7 @@ begin
         set_local_var(7, 30134);
     end
     else begin
-        if (cur_map_index == 43) then begin
+        if (CUR_MAP_NCR2) then begin
             if (local_var(14) == 19508) then begin
                 debug_msg("MERK: leave council map");
                 reg_anim_func(2, self_obj);
@@ -1733,12 +1733,12 @@ procedure shouldBeHere
 begin
     set_local_var(9, 0);
     set_local_var(10, checktime());
-    if (cur_map_index == 43) then begin
+    if (CUR_MAP_NCR2) then begin
         if ((local_var(10) == 43) or (local_var(10) == 42)) then begin
             set_local_var(9, 1);
         end
     end
-    if (cur_map_index == 46) then begin
+    if (CUR_MAP_NCRENT) then begin
         if (local_var(10) == 46) then begin
             set_local_var(9, 1);
         end
@@ -1750,13 +1750,13 @@ end
 procedure onValidMap
 begin
     set_local_var(9, 0);
-    if (cur_map_index == 46) then begin
+    if (CUR_MAP_NCRENT) then begin
         set_local_var(9, 1);
     end
-    if (cur_map_index == 43) then begin
+    if (CUR_MAP_NCR2) then begin
         set_local_var(9, 1);
     end
-    if (cur_map_index == 42) then begin
+    if (CUR_MAP_NCR1) then begin
         set_local_var(9, 1);
     end
     return local_var(9);
@@ -1772,7 +1772,7 @@ begin
                 debug_msg("MERK: told to animate");
             end
             else begin
-                if (cur_map_index == 43) then begin
+                if (CUR_MAP_NCR2) then begin
                     if (local_var(7) == 19518) then begin
                         debug_msg("I should be moving to my standing point location.");
                         if (local_var(13) != 1) then begin
@@ -1830,7 +1830,7 @@ begin
     set_local_var(11, 0);
     set_local_var(9, 0);
     set_local_var(11, checktime());
-    if (cur_map_index == 43) then begin
+    if (CUR_MAP_NCR2) then begin
         if (tile_num(merk_obj) == 18551) then begin
             set_local_var(9, 1);
         end
@@ -1844,7 +1844,7 @@ begin
         end
     end
     else begin
-        if (cur_map_index == 46) then begin
+        if (CUR_MAP_NCRENT) then begin
             if (tile_num(merk_obj) == 22154) then begin
                 set_local_var(9, 3);
             end
@@ -1864,14 +1864,14 @@ end
 procedure LennyFight
 begin
     call StripPlayer();
-    gfade_out(1);
+    fade_out;
     call HideParty();
     move_to(lenny_obj, 23747, 1);
     move_to(dude_obj, 21752, 1);
     anim(dude_obj, 1000, 2);
     anim(lenny_obj, 1000, 5);
     attack_setup(lenny_obj, dude_obj);
-    gfade_in(1);
+    fade_in;
     debug_msg("MERK: Lenny Should attack");
 end
 
@@ -1879,8 +1879,8 @@ procedure EndFight
 begin
     set_local_var(4, 0);
     set_local_var(5, 1);
-    if ((global_var(438) bwand 32768) != 0) then begin
-        set_global_var(438, global_var(438) bwor 65536);
+    if ((global_var(GVAR_NCR_GEN_FLAGS) bwand 32768) != 0) then begin
+        set_global_var(GVAR_NCR_GEN_FLAGS, global_var(GVAR_NCR_GEN_FLAGS) bwor 65536);
         debug_msg("MERK: Leeny killed by serum");
     end
     else begin
@@ -1899,8 +1899,8 @@ end
 
 procedure unHideParty
 begin
-    set_global_var(632, global_var(632) bwand (-1 - 262144));
-    if (global_var(769)) then begin
+    set_global_var(GVAR_PARTY_MEMBERS_HIDDEN, global_var(GVAR_PARTY_MEMBERS_HIDDEN) bwand (-1 - 262144));
+    if (global_var(GVAR_BOS_LOCATION_ELEVATORS)) then begin
         set_obj_visibility(party_member_obj(16777278), 0);
         set_obj_visibility(party_member_obj(16777376), 0);
         set_obj_visibility(party_member_obj(16777377), 0);
@@ -1928,7 +1928,7 @@ begin
         set_obj_visibility(party_member_obj(16777801), 0);
         set_obj_visibility(party_member_obj(16777872), 0);
         set_obj_visibility(party_member_obj(16777859), 0);
-        set_global_var(769, 0);
+        set_global_var(GVAR_BOS_LOCATION_ELEVATORS, 0);
     end
 end
 
@@ -1963,7 +1963,7 @@ end
 
 procedure Node1011a
 begin
-    if ((local_var(1) <= -2) or (global_var(209) == 3)) then begin
+    if ((local_var(1) <= -2) or (global_var(GVAR_NCR_SCMERK_STATUS) == 3)) then begin
         call Node010();
     end
     else begin
@@ -1983,7 +1983,7 @@ end
 
 procedure Node2122a
 begin
-    if ((global_var(1) >= 2) or (((global_var(4) + global_var(5)) >= 25) and ((global_var(5) > (2 * global_var(4))) or (global_var(3) == 1))) or (has_trait(0, dude_obj, 39) != 0)) then begin
+    if ((global_var(GVAR_CHILDKILLER_REPUTATION) >= 2) or (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) or (has_trait(TRAIT_PERK, dude_obj, PERK_cult_of_personality) != 0)) then begin
         call Node021();
     end
     else begin
@@ -1993,55 +1993,55 @@ end
 
 procedure Node999a
 begin
-    set_global_var(209, 3);
+    set_global_var(GVAR_NCR_SCMERK_STATUS, 3);
 end
 
 procedure Node999b
 begin
-    set_global_var(0, global_var(0) + -10);
+    set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + -10);
     debug_msg("Player gains " + -10 + " Karma Points.");
-    set_global_var(37, 0);
-    set_global_var(38, 0);
-    set_global_var(39, 0);
-    set_global_var(40, 0);
-    set_global_var(41, 0);
-    set_global_var(42, 0);
-    set_global_var(43, 0);
-    set_global_var(44, 0);
-    set_global_var(45, 0);
-    if (global_var(0) >= 1000) then begin
-        set_global_var(37, 1);
+    set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+    set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+    set_global_var(GVAR_KARMA_DEFENDER, 0);
+    set_global_var(GVAR_KARMA_WANDERER, 0);
+    set_global_var(GVAR_KARMA_BETRAYER, 0);
+    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+    set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+    set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+    if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
     end
     else begin
-        if (global_var(0) >= 750) then begin
-            set_global_var(38, 1);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+            set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
         end
         else begin
-            if (global_var(0) >= 500) then begin
-                set_global_var(39, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
             end
             else begin
-                if (global_var(0) >= 250) then begin
-                    set_global_var(40, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                    set_global_var(GVAR_KARMA_DEFENDER, 1);
                 end
                 else begin
-                    if (global_var(0) > -250) then begin
-                        set_global_var(41, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                        set_global_var(GVAR_KARMA_WANDERER, 1);
                     end
                     else begin
-                        if (global_var(0) > -500) then begin
-                            set_global_var(42, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                            set_global_var(GVAR_KARMA_BETRAYER, 1);
                         end
                         else begin
-                            if (global_var(0) > -750) then begin
-                                set_global_var(43, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                             end
                             else begin
-                                if (global_var(0) > -1000) then begin
-                                    set_global_var(44, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                    set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                 end
                                 else begin
-                                    set_global_var(45, 1);
+                                    set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                 end
                             end
                         end
@@ -2050,14 +2050,14 @@ begin
             end
         end
     end
-    set_global_var(210, global_var(210) bwor 16);
+    set_global_var(GVAR_NCR_SCMERK_SEED_STATUS, global_var(GVAR_NCR_SCMERK_SEED_STATUS) bwor 16);
     call Node999();
 end
 
 procedure Node2413a
 begin
-    set_global_var(209, 3);
-    if (get_critter_stat(dude_obj, 0) >= 8) then begin
+    set_global_var(GVAR_NCR_SCMERK_STATUS, 3);
+    if (dude_strength >= 8) then begin
         call Node024();
     end
     else begin
@@ -2067,11 +2067,11 @@ end
 
 procedure Node1312a
 begin
-    if (global_var(209) == 3) then begin
+    if (global_var(GVAR_NCR_SCMERK_STATUS) == 3) then begin
         call Node013();
     end
     else begin
-        set_global_var(209, 3);
+        set_global_var(GVAR_NCR_SCMERK_STATUS, 3);
         call Node012();
     end
 end
@@ -2129,9 +2129,9 @@ begin
     if (party_member_obj(16777376) != 0) then begin
         gfade_out(10);
         item_caps_adjust(dude_obj, 1000);
-        set_global_var(209, 4);
+        set_global_var(GVAR_NCR_SCMERK_STATUS, 4);
         if ((obj_item_subtype(critter_inven_obj(party_member_obj(16777376), 1)) == 3) or (obj_item_subtype(critter_inven_obj(party_member_obj(16777376), 2)) == 3)) then begin
-            metarule(43, party_member_obj(16777376));
+            inven_unwield(party_member_obj(16777376));
         end
         move_obj_inven_to_obj(party_member_obj(16777376), dude_obj);
         move_to(party_member_obj(16777376), 0, 0);
@@ -2144,8 +2144,8 @@ end
 
 procedure Node39a
 begin
-    set_global_var(210, global_var(210) bwor 1);
-    set_global_var(482, 1);
+    set_global_var(GVAR_NCR_SCMERK_SEED_STATUS, global_var(GVAR_NCR_SCMERK_SEED_STATUS) bwor 1);
+    set_global_var(GVAR_NCR_DRPAPR_QST, 1);
     call Node039();
 end
 
@@ -2160,19 +2160,19 @@ end
 
 procedure Node999c
 begin
-    set_global_var(189, 2);
+    set_global_var(GVAR_NCR_CAR_JACKED, 2);
     call Node999();
 end
 
 procedure NodeBarter
 begin
-    gdialog_mod_barter(0);
+    gdialog_barter;
     call Node026();
 end
 
 procedure Node4041a
 begin
-    if (get_critter_stat(dude_obj, 14) >= 90) then begin
+    if (get_critter_stat(dude_obj, STAT_heal_rate) >= 90) then begin
         call Node040();
     end
     else begin
@@ -2182,15 +2182,15 @@ end
 
 procedure Node055d
 begin
-    set_global_var(210, global_var(210) bwor 64);
-    set_global_var(486, 1);
+    set_global_var(GVAR_NCR_SCMERK_SEED_STATUS, global_var(GVAR_NCR_SCMERK_SEED_STATUS) bwor 64);
+    set_global_var(GVAR_NCR_KILL_ELRON_QST, 1);
     call Node055();
 end
 
 procedure Node3731a
 begin
     Merk_Question := 0;
-    if ((global_var(213) == 1) or (global_var(213) == 2)) then begin
+    if ((global_var(GVAR_NCR_FAKE_VAULT13_MAP) == 1) or (global_var(GVAR_NCR_FAKE_VAULT13_MAP) == 2)) then begin
         call Node037();
     end
     else begin
@@ -2233,14 +2233,14 @@ end
 procedure Node36a
 begin
     variable LVar0 := 0;
-    if (global_var(213) == 0) then begin
+    if (global_var(GVAR_NCR_FAKE_VAULT13_MAP) == 0) then begin
         item_caps_adjust(dude_obj, -1000);
         item_caps_adjust(self_obj, 1000);
-        LVar0 := create_object_sid(294, tile_num(self_obj), 0, 334);
+        LVar0 := create_object_sid(294, tile_num(self_obj), 0, SCRIPT_SIFAKHLO);
         if (LVar0) then begin
             add_obj_to_inven(dude_obj, LVar0);
-            set_global_var(213, 1);
-            if (metarule(17, 45) == 0) then begin
+            set_global_var(GVAR_NCR_FAKE_VAULT13_MAP, 1);
+            if (town_known(45) == 0) then begin
                 debug_msg("  mark_on_map(" + 45 + ")");
                 mark_area_known(0, 45, 1);
             end
@@ -2256,7 +2256,7 @@ end
 
 procedure Node3836a
 begin
-    if (global_var(210) bwand 2) then begin
+    if (global_var(GVAR_NCR_SCMERK_SEED_STATUS) bwand 2) then begin
         call Node038();
     end
     else begin
@@ -2266,20 +2266,20 @@ end
 
 procedure Node37a
 begin
-    gsay_reply(325, 268);
-    giq_option(4, 325, 269, Node998, 51);
-    giq_option(4, 325, 270, Node999, 50);
+    Reply(268);
+    giq_option(4, NAME, 269, Node998, BAD_REACTION);
+    giq_option(4, NAME, 270, Node999, NEUTRAL_REACTION);
 end
 
 procedure NodeCarlsonSec
 begin
-    set_global_var(175, global_var(175) bwor 4);
+    set_global_var(GVAR_NEWRENO_SNUFF_CARLSON, global_var(GVAR_NEWRENO_SNUFF_CARLSON) bwor 4);
     call Node999();
 end
 
 procedure Node522a
 begin
-    if (get_critter_stat(dude_obj, 3) >= 7) then begin
+    if (dude_charisma >= 7) then begin
         call Node005();
     end
     else begin
@@ -2295,12 +2295,12 @@ begin
         item_caps_adjust(dude_obj, 1000);
         rm_obj_from_inven(dude_obj, LVar0);
         add_obj_to_inven(self_obj, LVar0);
-        set_global_var(210, global_var(210) bwor 2);
-        set_global_var(482, 2);
+        set_global_var(GVAR_NCR_SCMERK_SEED_STATUS, global_var(GVAR_NCR_SCMERK_SEED_STATUS) bwor 2);
+        set_global_var(GVAR_NCR_DRPAPR_QST, 2);
         give_exp_points(1000);
-        display_msg(message_str(325, 271) + 1000 + message_str(325, 272));
+        display_msg(mstr(271) + 1000 + mstr(272));
     end
-    if (global_var(210) bwand 4) then begin
+    if (global_var(GVAR_NCR_SCMERK_SEED_STATUS) bwand 4) then begin
         call Node050();
     end
     else begin
@@ -2312,7 +2312,7 @@ end
 
 procedure Node5253a
 begin
-    if (get_critter_stat(dude_obj, 3) >= 9) then begin
+    if (dude_charisma >= 9) then begin
         call Node052();
     end
     else begin
@@ -2431,8 +2431,8 @@ end
 
 procedure HideParty
 begin
-    set_global_var(632, global_var(632) bwor 262144);
-    if (not(global_var(769))) then begin
+    set_global_var(GVAR_PARTY_MEMBERS_HIDDEN, global_var(GVAR_PARTY_MEMBERS_HIDDEN) bwor 262144);
+    if (not(global_var(GVAR_BOS_LOCATION_ELEVATORS))) then begin
         set_obj_visibility(party_member_obj(16777278), 1);
         set_obj_visibility(party_member_obj(16777376), 1);
         set_obj_visibility(party_member_obj(16777377), 1);
@@ -2460,7 +2460,6 @@ begin
         set_obj_visibility(party_member_obj(16777801), 1);
         set_obj_visibility(party_member_obj(16777872), 1);
         set_obj_visibility(party_member_obj(16777859), 1);
-        set_global_var(769, 1);
+        set_global_var(GVAR_BOS_LOCATION_ELEVATORS, 1);
     end
 end
-
