diff --git a/Mods/MegaMod/Original/SFDOCK.ssl b/Mods/MegaMod/ReDefine/SFDOCK.ssl
index c07cb45..ef4d7e2 100644
--- a/Mods/MegaMod/Original/SFDOCK.ssl
+++ b/Mods/MegaMod/ReDefine/SFDOCK.ssl
@@ -87,14 +87,14 @@ begin
             end
         end
     end
-    if (metarule(14, 0)) then begin
-        display_msg(message_str(1274, 100));
+    if (map_first_run) then begin
+        display_mstr(100);
     end
-    if (global_var(27) == 900) then begin
+    if (global_var(GVAR_LOAD_MAP_INDEX) == 900) then begin
         game_time_advance(15 * (60 * 10));
         move_to(dude_obj, 14921, 0);
-        set_global_var(632, global_var(632) bwand (-1 - 262144));
-        if (global_var(769)) then begin
+        set_global_var(GVAR_PARTY_MEMBERS_HIDDEN, global_var(GVAR_PARTY_MEMBERS_HIDDEN) bwand (-1 - 262144));
+        if (global_var(GVAR_BOS_LOCATION_ELEVATORS)) then begin
             set_obj_visibility(party_member_obj(16777278), 0);
             set_obj_visibility(party_member_obj(16777376), 0);
             set_obj_visibility(party_member_obj(16777377), 0);
@@ -122,196 +122,196 @@ begin
             set_obj_visibility(party_member_obj(16777785), 0);
             set_obj_visibility(party_member_obj(16777801), 0);
             set_obj_visibility(party_member_obj(16777293), 0);
-            set_global_var(769, 0);
+            set_global_var(GVAR_BOS_LOCATION_ELEVATORS, 0);
         end
         if (party_member_obj(16777278) != 0) then begin
-            if (has_trait(1, party_member_obj(16777278), 666)) then begin
-                critter_attempt_placement(party_member_obj(16777278), tile_num(dude_obj), elevation(dude_obj));
+            if (obj_is_visible_flag(party_member_obj(16777278))) then begin
+                critter_attempt_placement(party_member_obj(16777278), dude_tile, elevation(dude_obj));
             end
         end
         if (party_member_obj(16777376) != 0) then begin
-            if (has_trait(1, party_member_obj(16777376), 666)) then begin
-                critter_attempt_placement(party_member_obj(16777376), tile_num(dude_obj), elevation(dude_obj));
+            if (obj_is_visible_flag(party_member_obj(16777376))) then begin
+                critter_attempt_placement(party_member_obj(16777376), dude_tile, elevation(dude_obj));
             end
         end
         if (party_member_obj(16777377) != 0) then begin
-            if (has_trait(1, party_member_obj(16777377), 666)) then begin
-                critter_attempt_placement(party_member_obj(16777377), tile_num(dude_obj), elevation(dude_obj));
+            if (obj_is_visible_flag(party_member_obj(16777377))) then begin
+                critter_attempt_placement(party_member_obj(16777377), dude_tile, elevation(dude_obj));
             end
         end
         if (party_member_obj(16777305) != 0) then begin
-            if (has_trait(1, party_member_obj(16777305), 666)) then begin
-                critter_attempt_placement(party_member_obj(16777305), tile_num(dude_obj), elevation(dude_obj));
+            if (obj_is_visible_flag(party_member_obj(16777305))) then begin
+                critter_attempt_placement(party_member_obj(16777305), dude_tile, elevation(dude_obj));
             end
         end
         if (party_member_obj(16777313) != 0) then begin
-            if (has_trait(1, party_member_obj(16777313), 666)) then begin
-                critter_attempt_placement(party_member_obj(16777313), tile_num(dude_obj), elevation(dude_obj));
+            if (obj_is_visible_flag(party_member_obj(16777313))) then begin
+                critter_attempt_placement(party_member_obj(16777313), dude_tile, elevation(dude_obj));
             end
         end
         if (party_member_obj(16777323) != 0) then begin
-            if (has_trait(1, party_member_obj(16777323), 666)) then begin
-                critter_attempt_placement(party_member_obj(16777323), tile_num(dude_obj), elevation(dude_obj));
+            if (obj_is_visible_flag(party_member_obj(16777323))) then begin
+                critter_attempt_placement(party_member_obj(16777323), dude_tile, elevation(dude_obj));
             end
         end
         if (party_member_obj(16777352) != 0) then begin
-            if (has_trait(1, party_member_obj(16777352), 666)) then begin
-                critter_attempt_placement(party_member_obj(16777352), tile_num(dude_obj), elevation(dude_obj));
+            if (obj_is_visible_flag(party_member_obj(16777352))) then begin
+                critter_attempt_placement(party_member_obj(16777352), dude_tile, elevation(dude_obj));
             end
         end
         if (party_member_obj(16777378) != 0) then begin
-            if (has_trait(1, party_member_obj(16777378), 666)) then begin
-                critter_attempt_placement(party_member_obj(16777378), tile_num(dude_obj), elevation(dude_obj));
+            if (obj_is_visible_flag(party_member_obj(16777378))) then begin
+                critter_attempt_placement(party_member_obj(16777378), dude_tile, elevation(dude_obj));
             end
         end
         if (party_member_obj(16777368) != 0) then begin
-            if (has_trait(1, party_member_obj(16777368), 666)) then begin
-                critter_attempt_placement(party_member_obj(16777368), tile_num(dude_obj), elevation(dude_obj));
+            if (obj_is_visible_flag(party_member_obj(16777368))) then begin
+                critter_attempt_placement(party_member_obj(16777368), dude_tile, elevation(dude_obj));
             end
         end
         if (party_member_obj(16777379) != 0) then begin
-            if (has_trait(1, party_member_obj(16777379), 666)) then begin
-                critter_attempt_placement(party_member_obj(16777379), tile_num(dude_obj), elevation(dude_obj));
+            if (obj_is_visible_flag(party_member_obj(16777379))) then begin
+                critter_attempt_placement(party_member_obj(16777379), dude_tile, elevation(dude_obj));
             end
         end
         if (party_member_obj(16777380) != 0) then begin
-            if (has_trait(1, party_member_obj(16777380), 666)) then begin
-                critter_attempt_placement(party_member_obj(16777380), tile_num(dude_obj), elevation(dude_obj));
+            if (obj_is_visible_flag(party_member_obj(16777380))) then begin
+                critter_attempt_placement(party_member_obj(16777380), dude_tile, elevation(dude_obj));
             end
         end
         if (party_member_obj(16777295) != 0) then begin
-            if (has_trait(1, party_member_obj(16777295), 666)) then begin
-                critter_attempt_placement(party_member_obj(16777295), tile_num(dude_obj), elevation(dude_obj));
+            if (obj_is_visible_flag(party_member_obj(16777295))) then begin
+                critter_attempt_placement(party_member_obj(16777295), dude_tile, elevation(dude_obj));
             end
         end
         if (party_member_obj(455) != 0) then begin
-            if (has_trait(1, party_member_obj(455), 666)) then begin
-                critter_attempt_placement(party_member_obj(455), tile_num(dude_obj), elevation(dude_obj));
+            if (obj_is_visible_flag(party_member_obj(455))) then begin
+                critter_attempt_placement(party_member_obj(455), dude_tile, elevation(dude_obj));
             end
         end
         if (party_member_obj(16777558) != 0) then begin
-            if (has_trait(1, party_member_obj(16777558), 666)) then begin
-                critter_attempt_placement(party_member_obj(16777558), tile_num(dude_obj), elevation(dude_obj));
+            if (obj_is_visible_flag(party_member_obj(16777558))) then begin
+                critter_attempt_placement(party_member_obj(16777558), dude_tile, elevation(dude_obj));
             end
         end
         if (party_member_obj(16777600) != 0) then begin
-            if (has_trait(1, party_member_obj(16777600), 666)) then begin
-                critter_attempt_placement(party_member_obj(16777600), tile_num(dude_obj), elevation(dude_obj));
+            if (obj_is_visible_flag(party_member_obj(16777600))) then begin
+                critter_attempt_placement(party_member_obj(16777600), dude_tile, elevation(dude_obj));
             end
         end
         if (party_member_obj(16777687) != 0) then begin
-            if (has_trait(1, party_member_obj(16777687), 666)) then begin
-                critter_attempt_placement(party_member_obj(16777687), tile_num(dude_obj), elevation(dude_obj));
+            if (obj_is_visible_flag(party_member_obj(16777687))) then begin
+                critter_attempt_placement(party_member_obj(16777687), dude_tile, elevation(dude_obj));
             end
         end
         if (party_member_obj(16777597) != 0) then begin
-            if (has_trait(1, party_member_obj(16777597), 666)) then begin
-                critter_attempt_placement(party_member_obj(16777597), tile_num(dude_obj), elevation(dude_obj));
+            if (obj_is_visible_flag(party_member_obj(16777597))) then begin
+                critter_attempt_placement(party_member_obj(16777597), dude_tile, elevation(dude_obj));
             end
         end
         if (party_member_obj(16777596) != 0) then begin
-            if (has_trait(1, party_member_obj(16777596), 666)) then begin
-                critter_attempt_placement(party_member_obj(16777596), tile_num(dude_obj), elevation(dude_obj));
+            if (obj_is_visible_flag(party_member_obj(16777596))) then begin
+                critter_attempt_placement(party_member_obj(16777596), dude_tile, elevation(dude_obj));
             end
         end
         if (party_member_obj(16777595) != 0) then begin
-            if (has_trait(1, party_member_obj(16777595), 666)) then begin
-                critter_attempt_placement(party_member_obj(16777595), tile_num(dude_obj), elevation(dude_obj));
+            if (obj_is_visible_flag(party_member_obj(16777595))) then begin
+                critter_attempt_placement(party_member_obj(16777595), dude_tile, elevation(dude_obj));
             end
         end
         if (party_member_obj(16777381) != 0) then begin
-            if (has_trait(1, party_member_obj(16777381), 666)) then begin
-                critter_attempt_placement(party_member_obj(16777381), tile_num(dude_obj), elevation(dude_obj));
+            if (obj_is_visible_flag(party_member_obj(16777381))) then begin
+                critter_attempt_placement(party_member_obj(16777381), dude_tile, elevation(dude_obj));
             end
         end
         if (party_member_obj(16777407) != 0) then begin
-            if (has_trait(1, party_member_obj(16777407), 666)) then begin
-                critter_attempt_placement(party_member_obj(16777407), tile_num(dude_obj), elevation(dude_obj));
+            if (obj_is_visible_flag(party_member_obj(16777407))) then begin
+                critter_attempt_placement(party_member_obj(16777407), dude_tile, elevation(dude_obj));
             end
         end
         if (party_member_obj(16777411) != 0) then begin
-            if (has_trait(1, party_member_obj(16777411), 666)) then begin
-                critter_attempt_placement(party_member_obj(16777411), tile_num(dude_obj), elevation(dude_obj));
+            if (obj_is_visible_flag(party_member_obj(16777411))) then begin
+                critter_attempt_placement(party_member_obj(16777411), dude_tile, elevation(dude_obj));
             end
         end
         if (party_member_obj(16777412) != 0) then begin
-            if (has_trait(1, party_member_obj(16777412), 666)) then begin
-                critter_attempt_placement(party_member_obj(16777412), tile_num(dude_obj), elevation(dude_obj));
+            if (obj_is_visible_flag(party_member_obj(16777412))) then begin
+                critter_attempt_placement(party_member_obj(16777412), dude_tile, elevation(dude_obj));
             end
         end
         if (party_member_obj(16777413) != 0) then begin
-            if (has_trait(1, party_member_obj(16777413), 666)) then begin
-                critter_attempt_placement(party_member_obj(16777413), tile_num(dude_obj), elevation(dude_obj));
+            if (obj_is_visible_flag(party_member_obj(16777413))) then begin
+                critter_attempt_placement(party_member_obj(16777413), dude_tile, elevation(dude_obj));
             end
         end
         if (party_member_obj(16777859) != 0) then begin
-            if (has_trait(1, party_member_obj(16777859), 666)) then begin
-                critter_attempt_placement(party_member_obj(16777859), tile_num(dude_obj), elevation(dude_obj));
+            if (obj_is_visible_flag(party_member_obj(16777859))) then begin
+                critter_attempt_placement(party_member_obj(16777859), dude_tile, elevation(dude_obj));
             end
         end
         if (party_member_obj(16777872) != 0) then begin
-            if (has_trait(1, party_member_obj(16777872), 666)) then begin
-                critter_attempt_placement(party_member_obj(16777872), tile_num(dude_obj), elevation(dude_obj));
+            if (obj_is_visible_flag(party_member_obj(16777872))) then begin
+                critter_attempt_placement(party_member_obj(16777872), dude_tile, elevation(dude_obj));
             end
         end
         if (party_member_obj(16777874) != 0) then begin
-            if (has_trait(1, party_member_obj(16777874), 666)) then begin
-                critter_attempt_placement(party_member_obj(16777874), tile_num(dude_obj), elevation(dude_obj));
+            if (obj_is_visible_flag(party_member_obj(16777874))) then begin
+                critter_attempt_placement(party_member_obj(16777874), dude_tile, elevation(dude_obj));
             end
         end
         if (party_member_obj(16777481) != 0) then begin
-            if (has_trait(1, party_member_obj(16777481), 666)) then begin
-                critter_attempt_placement(party_member_obj(16777481), tile_num(dude_obj), elevation(dude_obj));
+            if (obj_is_visible_flag(party_member_obj(16777481))) then begin
+                critter_attempt_placement(party_member_obj(16777481), dude_tile, elevation(dude_obj));
             end
         end
         if (party_member_obj(16777875) != 0) then begin
-            if (has_trait(1, party_member_obj(16777875), 666)) then begin
-                critter_attempt_placement(party_member_obj(16777875), tile_num(dude_obj), elevation(dude_obj));
+            if (obj_is_visible_flag(party_member_obj(16777875))) then begin
+                critter_attempt_placement(party_member_obj(16777875), dude_tile, elevation(dude_obj));
             end
         end
         if (party_member_obj(16777876) != 0) then begin
-            if (has_trait(1, party_member_obj(16777876), 666)) then begin
-                critter_attempt_placement(party_member_obj(16777876), tile_num(dude_obj), elevation(dude_obj));
+            if (obj_is_visible_flag(party_member_obj(16777876))) then begin
+                critter_attempt_placement(party_member_obj(16777876), dude_tile, elevation(dude_obj));
             end
         end
         if (party_member_obj(16777785) != 0) then begin
-            if (has_trait(1, party_member_obj(16777785), 666)) then begin
-                critter_attempt_placement(party_member_obj(16777785), tile_num(dude_obj), elevation(dude_obj));
+            if (obj_is_visible_flag(party_member_obj(16777785))) then begin
+                critter_attempt_placement(party_member_obj(16777785), dude_tile, elevation(dude_obj));
             end
         end
         if (party_member_obj(16777749) != 0) then begin
-            if (has_trait(1, party_member_obj(16777749), 666)) then begin
-                critter_attempt_placement(party_member_obj(16777749), tile_num(dude_obj), elevation(dude_obj));
+            if (obj_is_visible_flag(party_member_obj(16777749))) then begin
+                critter_attempt_placement(party_member_obj(16777749), dude_tile, elevation(dude_obj));
             end
         end
         if (party_member_obj(16777748) != 0) then begin
-            if (has_trait(1, party_member_obj(16777748), 666)) then begin
-                critter_attempt_placement(party_member_obj(16777748), tile_num(dude_obj), elevation(dude_obj));
+            if (obj_is_visible_flag(party_member_obj(16777748))) then begin
+                critter_attempt_placement(party_member_obj(16777748), dude_tile, elevation(dude_obj));
             end
         end
         if (party_member_obj(16777747) != 0) then begin
-            if (has_trait(1, party_member_obj(16777747), 666)) then begin
-                critter_attempt_placement(party_member_obj(16777747), tile_num(dude_obj), elevation(dude_obj));
+            if (obj_is_visible_flag(party_member_obj(16777747))) then begin
+                critter_attempt_placement(party_member_obj(16777747), dude_tile, elevation(dude_obj));
             end
         end
         if (party_member_obj(16777762) != 0) then begin
-            if (has_trait(1, party_member_obj(16777762), 666)) then begin
-                critter_attempt_placement(party_member_obj(16777762), tile_num(dude_obj), elevation(dude_obj));
+            if (obj_is_visible_flag(party_member_obj(16777762))) then begin
+                critter_attempt_placement(party_member_obj(16777762), dude_tile, elevation(dude_obj));
             end
         end
         if (party_member_obj(16777801) != 0) then begin
-            if (has_trait(1, party_member_obj(16777801), 666)) then begin
-                critter_attempt_placement(party_member_obj(16777801), tile_num(dude_obj), elevation(dude_obj));
+            if (obj_is_visible_flag(party_member_obj(16777801))) then begin
+                critter_attempt_placement(party_member_obj(16777801), dude_tile, elevation(dude_obj));
             end
         end
-        display_msg(message_str(1274, 200));
+        display_mstr(200);
         give_exp_points(1000);
-        display_msg(message_str(14, 100) + 1000 + message_str(14, 101));
+        display_msg(g_mstr(100) + 1000 + g_mstr(101));
     end
-    if (global_var(499) == 2) then begin
-        set_global_var(499, 3);
+    if (global_var(GVAR_ENCLAVE_FRANK_DEAD) == 2) then begin
+        set_global_var(GVAR_ENCLAVE_FRANK_DEAD, 3);
     end
-    set_global_var(27, 0);
+    set_global_var(GVAR_LOAD_MAP_INDEX, 0);
 end
 
 procedure map_update_p_proc
@@ -393,4 +393,3 @@ begin
         end
     end
 end
-
