diff --git a/Mods/MegaMod/Original/SICMPWES.ssl b/Mods/MegaMod/ReDefine/SICMPWES.ssl
index de5e294..9663354 100644
--- a/Mods/MegaMod/Original/SICMPWES.ssl
+++ b/Mods/MegaMod/ReDefine/SICMPWES.ssl
@@ -178,19 +178,19 @@ begin
                 add_timer_event(i_westin_guard_obj, 0, 4);
             end
         end
-        display_msg(message_str(852, 101));
+        display_mstr(101);
     end
 end
 
 procedure use_obj_on_p_proc
 begin
-    display_msg(message_str(852, 100));
+    display_mstr(100);
 end
 
 procedure damage_p_proc
 begin
-    if (metarule(49, target_obj) == 6) then begin
-        display_msg(message_str(852, 102));
+    if (weapon_dmg_type(target_obj) == 6) then begin
+        display_mstr(102);
         explosion(tile_num(self_obj), 0, 25);
         add_timer_event(self_obj, 3, 1);
     end
@@ -216,10 +216,9 @@ begin
             add_timer_event(i_westin_emitter_obj, 0, 3);
             explosion(tile_num(i_westin_emitter_obj), 0, 0);
         end
-        LVar0 := create_object_sid(33555233, tile_num(self_obj), elevation(self_obj), -1);
+        LVar0 := create_object(33555233, tile_num(self_obj), elevation(self_obj));
         if (LVar0) then begin
             destroy_object(self_obj);
         end
     end
 end
-
