diff --git a/Mods/MegaMod/Original/SLVCO2.ssl b/Mods/MegaMod/ReDefine/SLVCO2.ssl
index 28469ba..746fde7 100644
--- a/Mods/MegaMod/Original/SLVCO2.ssl
+++ b/Mods/MegaMod/ReDefine/SLVCO2.ssl
@@ -143,7 +143,7 @@ begin
             script_overrides;
             obj_lock(self_obj);
         end
-        if ((global_var(1123) < 4) and (local_var(0) == 0)) then begin
+        if ((global_var(GVAR_SULIK_DEAD) < 4) and (local_var(0) == 0)) then begin
             start_gdialog(964, self_obj, 4, -1, -1);
             gsay_start;
             call Node001();
@@ -176,35 +176,35 @@ end
 procedure description_p_proc
 begin
     script_overrides;
-    display_msg(message_str(13, 101));
+    display_msg(message_str(SCRIPT_DOOR, 101));
 end
 
 procedure look_at_p_proc
 begin
     script_overrides;
-    display_msg(message_str(13, 100));
+    display_msg(message_str(SCRIPT_DOOR, 100));
 end
 
 procedure Node001
 begin
-    if ((map_var(0) >= 5) and (global_var(961) == 0)) then begin
-        gsay_reply(964, 9501);
-        giq_option(4, 964, 9502, Node999, 50);
+    if ((map_var(0) >= 5) and (global_var(GVAR_ABBEY_FORMAT_DATABASES) == 0)) then begin
+        gsay_reply(SCRIPT_KLATRAP, 9501);
+        giq_option(4, SCRIPT_KLATRAP, 9502, Node999, NEUTRAL_REACTION);
     end
     else begin
-        if ((map_var(0) == 0) or (global_var(961) == 1)) then begin
-            gsay_reply(964, 9500);
+        if ((map_var(0) == 0) or (global_var(GVAR_ABBEY_FORMAT_DATABASES) == 1)) then begin
+            gsay_reply(SCRIPT_KLATRAP, 9500);
             if (obj_is_carrying_obj_pid(dude_obj, 83) >= 1) then begin
-                giq_option(4, 964, 9503, Node002, 50);
+                giq_option(4, SCRIPT_KLATRAP, 9503, Node002, NEUTRAL_REACTION);
             end
             else begin
-                giq_option(4, 964, 9502, Node003, 50);
+                giq_option(4, SCRIPT_KLATRAP, 9502, Node003, NEUTRAL_REACTION);
             end
         end
         else begin
             if ((map_var(0) == 1) or (map_var(0) == 2) or (map_var(0) == 3) or (map_var(0) == 4)) then begin
-                gsay_reply(964, 9505);
-                giq_option(4, 964, 9502, Node999, 50);
+                gsay_reply(SCRIPT_KLATRAP, 9505);
+                giq_option(4, SCRIPT_KLATRAP, 9502, Node999, NEUTRAL_REACTION);
             end
         end
     end
@@ -214,7 +214,7 @@ procedure Node002
 begin
     obj_unlock(self_obj);
     set_local_var(0, 1);
-    display_msg(message_str(964, 9504));
+    display_msg(message_str(SCRIPT_KLATRAP, 9504));
     call Node999();
 end
 
@@ -244,4 +244,3 @@ end
 procedure Tamper_Proc
 begin
 end
-
