diff --git a/Mods/MegaMod/Original/SSCOMP.ssl b/Mods/MegaMod/ReDefine/SSCOMP.ssl
index 46c78e0..6e0e21a 100644
--- a/Mods/MegaMod/Original/SSCOMP.ssl
+++ b/Mods/MegaMod/ReDefine/SSCOMP.ssl
@@ -157,17 +157,17 @@ end
 procedure look_at_p_proc
 begin
     script_overrides;
-    display_msg(message_str(520, 100));
+    display_mstr(100);
 end
 
 procedure description_p_proc
 begin
     script_overrides;
-    if ((global_var(240) != 4) and (global_var(240) != 3)) then begin
-        display_msg(message_str(520, 101));
+    if ((global_var(GVAR_NCR_POWERPLANT) != 4) and (global_var(GVAR_NCR_POWERPLANT) != 3)) then begin
+        display_mstr(101);
     end
     else begin
-        display_msg(message_str(520, 102));
+        display_mstr(102);
     end
 end
 
@@ -199,7 +199,7 @@ end
 
 procedure map_enter_p_proc
 begin
-    if (global_var(234) != 1) then begin
+    if (global_var(GVAR_NCR_POWER_ON) != 1) then begin
         add_timer_event(self_obj, 1, 1);
         set_local_var(0, 1);
         set_local_var(1, 0);
@@ -215,32 +215,32 @@ end
 
 procedure use_fail
 begin
-    if ((global_var(240) != 6) and (global_var(240) != 8)) then begin
-        set_global_var(240, 9);
-        display_msg(message_str(520, 104));
+    if ((global_var(GVAR_NCR_POWERPLANT) != 6) and (global_var(GVAR_NCR_POWERPLANT) != 8)) then begin
+        set_global_var(GVAR_NCR_POWERPLANT, 9);
+        display_mstr(104);
     end
 end
 
 procedure use_success
 begin
-    if ((global_var(240) != 6) and (global_var(240) != 8)) then begin
-        set_global_var(240, 8);
-        display_msg(message_str(520, 103));
+    if ((global_var(GVAR_NCR_POWERPLANT) != 6) and (global_var(GVAR_NCR_POWERPLANT) != 8)) then begin
+        set_global_var(GVAR_NCR_POWERPLANT, 8);
+        display_mstr(103);
     end
 end
 
 procedure loadDestroyed
 begin
     variable LVar0 := 0;
-    create_object_sid(33555233, tile_num(self_obj), 0, 520);
+    create_object_sid(33555233, tile_num(self_obj), 0, NAME);
     destroy_object(self_obj);
 end
 
 procedure doExplosion
 begin
     set_local_var(0, 2);
-    set_global_var(240, 3);
-    set_global_var(234, 1);
+    set_global_var(GVAR_NCR_POWERPLANT, 3);
+    set_global_var(GVAR_NCR_POWER_ON, 1);
     explosion(tile_num(self_obj), 0, 200);
     add_timer_event(self_obj, game_ticks(2), 3);
 end
@@ -249,7 +249,7 @@ procedure timed_event_p_proc
 begin
     variable LVar0 := 0;
     if (fixed_param == 1) then begin
-        if ((global_var(240) == 4) and (local_var(1) == 0)) then begin
+        if ((global_var(GVAR_NCR_POWERPLANT) == 4) and (local_var(1) == 0)) then begin
             set_local_var(1, 1);
             add_timer_event(self_obj, game_ticks(1), 2);
         end
@@ -257,7 +257,7 @@ begin
     end
     else begin
         if (fixed_param == 2) then begin
-            if (global_var(240) != 8) then begin
+            if (global_var(GVAR_NCR_POWERPLANT) != 8) then begin
                 set_local_var(1, local_var(1) + 1);
                 if (local_var(1) >= 30) then begin
                     call doExplosion();
@@ -285,4 +285,3 @@ begin
         end
     end
 end
-
