diff --git a/Mods/MegaMod/Original/TDLIFT.ssl b/Mods/MegaMod/ReDefine/TDLIFT.ssl
index f422772..7ba034b 100644
--- a/Mods/MegaMod/Original/TDLIFT.ssl
+++ b/Mods/MegaMod/ReDefine/TDLIFT.ssl
@@ -166,13 +166,13 @@ begin
         addButton("tdelk3", 10, 160, 52, 48);
         addButtonGfx("tdelk3", "tdlift\\k3in.pcx", "tdlift\\k3out.pcx", "tdlift\\k3out.pcx");
         addButtonProc("tdelk3", k_not, k_not, k_not, k3_release);
-        if (elevation(dude_obj) == 0) then begin
+        if (dude_elevation == 0) then begin
             displayGfx("tdlift\\tind00.pcx", 121, 41, 92, 55);
         end
-        if (elevation(dude_obj) == 1) then begin
+        if (dude_elevation == 1) then begin
             displayGfx("tdlift\\tind06.pcx", 121, 41, 92, 55);
         end
-        if (elevation(dude_obj) == 2) then begin
+        if (dude_elevation == 2) then begin
             displayGfx("tdlift\\tind12.pcx", 121, 41, 92, 55);
         end
         showWin;
@@ -184,9 +184,9 @@ procedure k1_release
 begin
     hideMouse;
     reg_anim_func(2, dude_obj);
-    if (elevation(dude_obj) != 0) then begin
+    if (dude_elevation != 0) then begin
         set_map_music(cur_map_index, "");
-        if (elevation(dude_obj) == 2) then begin
+        if (dude_elevation == 2) then begin
             play_sfx("ELV2_3");
             call lift_goto_3_1();
         end
@@ -214,9 +214,9 @@ procedure k2_release
 begin
     hideMouse;
     reg_anim_func(2, dude_obj);
-    if (elevation(dude_obj) != 1) then begin
+    if (dude_elevation != 1) then begin
         set_map_music(cur_map_index, "");
-        if (elevation(dude_obj) == 0) then begin
+        if (dude_elevation == 0) then begin
             play_sfx("ELV1_2");
             call lift_goto_1_2();
         end
@@ -244,9 +244,9 @@ procedure k3_release
 begin
     hideMouse;
     reg_anim_func(2, dude_obj);
-    if (elevation(dude_obj) != 2) then begin
+    if (dude_elevation != 2) then begin
         set_map_music(cur_map_index, "");
-        if (elevation(dude_obj) == 0) then begin
+        if (dude_elevation == 0) then begin
             play_sfx("ELV2_3");
             call lift_goto_1_3();
         end
@@ -373,4 +373,3 @@ begin
     call lift_goto_3_2();
     call lift_goto_2_1();
 end
-
