diff --git a/Mods/MegaMod/Original/TONY.ssl b/Mods/MegaMod/ReDefine/TONY.ssl
index c48998e..271efe4 100644
--- a/Mods/MegaMod/Original/TONY.ssl
+++ b/Mods/MegaMod/ReDefine/TONY.ssl
@@ -213,10 +213,10 @@ procedure critter_p_proc
 begin
     if ((local_var(5) == 2) and obj_can_see_obj(self_obj, dude_obj)) then begin
         set_local_var(5, 1);
-        attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+        attack(dude_obj);
     end
     if ((local_var(5) == 1) and obj_can_see_obj(self_obj, dude_obj)) then begin
-        attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+        attack(dude_obj);
     end
 end
 
@@ -266,60 +266,60 @@ end
 
 procedure destroy_p_proc
 begin
-    set_global_var(830, global_var(830) bwor 262144);
+    set_global_var(GVAR_TOWN_REP_COLLY, global_var(GVAR_TOWN_REP_COLLY) bwor 262144);
     if (source_obj == dude_obj) then begin
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(4) > (3 * global_var(5))) or (global_var(2) == 1)) and (global_var(1) == 0)) then begin
-            set_global_var(2, 1);
-            set_global_var(3, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(4) > (3 * global_var(GVAR_GOOD_MONSTER))) or (global_var(GVAR_CHAMPION_REPUTATION) == 1)) and (global_var(GVAR_CHILDKILLER_REPUTATION) == 0)) then begin
+            set_global_var(GVAR_CHAMPION_REPUTATION, 1);
+            set_global_var(GVAR_BERSERKER_REPUTATION, 0);
         end
-        if (metarule(51, self_obj) == 2) then begin
+        if (critter_kill_type(self_obj) == 2) then begin
             if (source_obj == dude_obj) then begin
-                set_global_var(2, 0);
-                set_global_var(1, global_var(1) + 1);
-                set_global_var(0, global_var(0) + -15);
+                set_global_var(GVAR_CHAMPION_REPUTATION, 0);
+                set_global_var(GVAR_CHILDKILLER_REPUTATION, global_var(GVAR_CHILDKILLER_REPUTATION) + 1);
+                set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + -15);
                 debug_msg("Player gains " + -15 + " Karma Points.");
-                set_global_var(37, 0);
-                set_global_var(38, 0);
-                set_global_var(39, 0);
-                set_global_var(40, 0);
-                set_global_var(41, 0);
-                set_global_var(42, 0);
-                set_global_var(43, 0);
-                set_global_var(44, 0);
-                set_global_var(45, 0);
-                if (global_var(0) >= 1000) then begin
-                    set_global_var(37, 1);
+                set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+                set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+                set_global_var(GVAR_KARMA_DEFENDER, 0);
+                set_global_var(GVAR_KARMA_WANDERER, 0);
+                set_global_var(GVAR_KARMA_BETRAYER, 0);
+                set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+                set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+                set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+                    set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
                 end
                 else begin
-                    if (global_var(0) >= 750) then begin
-                        set_global_var(38, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
                     end
                     else begin
-                        if (global_var(0) >= 500) then begin
-                            set_global_var(39, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                            set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                         end
                         else begin
-                            if (global_var(0) >= 250) then begin
-                                set_global_var(40, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                                set_global_var(GVAR_KARMA_DEFENDER, 1);
                             end
                             else begin
-                                if (global_var(0) > -250) then begin
-                                    set_global_var(41, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                                    set_global_var(GVAR_KARMA_WANDERER, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -500) then begin
-                                        set_global_var(42, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                        set_global_var(GVAR_KARMA_BETRAYER, 1);
                                     end
                                     else begin
-                                        if (global_var(0) > -750) then begin
-                                            set_global_var(43, 1);
+                                        if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                            set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                         end
                                         else begin
-                                            if (global_var(0) > -1000) then begin
-                                                set_global_var(44, 1);
+                                            if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                                set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                             end
                                             else begin
-                                                set_global_var(45, 1);
+                                                set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                             end
                                         end
                                     end
@@ -328,58 +328,58 @@ begin
                         end
                     end
                 end
-                set_global_var(832, global_var(832) + -8);
+                set_global_var(GVAR_GANG_OUT, global_var(GVAR_GANG_OUT) + -8);
                 debug_msg("Added " + -8 + " to Town Rep");
             end
         end
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(5) > (2 * global_var(4))) or (global_var(3) == 1))) then begin
-            set_global_var(3, 1);
-            set_global_var(2, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
+            set_global_var(GVAR_BERSERKER_REPUTATION, 1);
+            set_global_var(GVAR_CHAMPION_REPUTATION, 0);
         end
-        set_global_var(0, global_var(0) + 0);
+        set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + 0);
         debug_msg("Player gains " + 0 + " Karma Points.");
-        set_global_var(37, 0);
-        set_global_var(38, 0);
-        set_global_var(39, 0);
-        set_global_var(40, 0);
-        set_global_var(41, 0);
-        set_global_var(42, 0);
-        set_global_var(43, 0);
-        set_global_var(44, 0);
-        set_global_var(45, 0);
-        if (global_var(0) >= 1000) then begin
-            set_global_var(37, 1);
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+        set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+        set_global_var(GVAR_KARMA_DEFENDER, 0);
+        set_global_var(GVAR_KARMA_WANDERER, 0);
+        set_global_var(GVAR_KARMA_BETRAYER, 0);
+        set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+        set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+            set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
         end
         else begin
-            if (global_var(0) >= 750) then begin
-                set_global_var(38, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
             end
             else begin
-                if (global_var(0) >= 500) then begin
-                    set_global_var(39, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                 end
                 else begin
-                    if (global_var(0) >= 250) then begin
-                        set_global_var(40, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                        set_global_var(GVAR_KARMA_DEFENDER, 1);
                     end
                     else begin
-                        if (global_var(0) > -250) then begin
-                            set_global_var(41, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                            set_global_var(GVAR_KARMA_WANDERER, 1);
                         end
                         else begin
-                            if (global_var(0) > -500) then begin
-                                set_global_var(42, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                set_global_var(GVAR_KARMA_BETRAYER, 1);
                             end
                             else begin
-                                if (global_var(0) > -750) then begin
-                                    set_global_var(43, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -1000) then begin
-                                        set_global_var(44, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                     end
                                     else begin
-                                        set_global_var(45, 1);
+                                        set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                     end
                                 end
                             end
@@ -395,17 +395,17 @@ procedure look_at_p_proc
 begin
     script_overrides;
     if (((local_var(6) bwand 1) != 0) == 0) then begin
-        display_msg(message_str(1309, 1100));
+        display_msg(message_str(SCRIPT_AITEM6, 1100));
     end
     else begin
-        display_msg(message_str(1309, 1101));
+        display_msg(message_str(SCRIPT_AITEM6, 1101));
     end
 end
 
 procedure description_p_proc
 begin
     script_overrides;
-    display_msg(message_str(1309, 1105));
+    display_msg(message_str(SCRIPT_AITEM6, 1105));
 end
 
 procedure use_skill_on_p_proc
@@ -421,7 +421,7 @@ end
 
 procedure map_enter_p_proc
 begin
-    if (not(metarule(22, 0))) then begin
+    if (not(is_loading_game)) then begin
         critter_attempt_placement(self_obj, 15323, elevation(self_obj));
         anim(self_obj, 1000, 3);
     end
@@ -443,155 +443,155 @@ end
 
 procedure Node001
 begin
-    gsay_reply(1309, message_str(1309, 1200));
-    if (((global_var(830) bwand 131072) != 0) and (get_critter_stat(dude_obj, 1) > 6)) then begin
-        giq_option(4, 1309, 1201, Node010, 50);
+    gsay_reply(SCRIPT_AITEM6, message_str(SCRIPT_AITEM6, 1200));
+    if (((global_var(GVAR_TOWN_REP_COLLY) bwand 131072) != 0) and (dude_perception > 6)) then begin
+        giq_option(4, SCRIPT_AITEM6, 1201, Node010, NEUTRAL_REACTION);
     end
-    giq_option(4, 1309, 1203, Node002, 50);
-    giq_option(4, 1309, 1204, Node999, 50);
+    giq_option(4, SCRIPT_AITEM6, 1203, Node002, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_AITEM6, 1204, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node002
 begin
-    set_global_var(830, global_var(830) bwor 131072);
+    set_global_var(GVAR_TOWN_REP_COLLY, global_var(GVAR_TOWN_REP_COLLY) bwor 131072);
     set_local_var(6, local_var(6) bwor 1);
-    gsay_reply(1309, message_str(1309, 1210));
-    giq_option(4, 1309, 1211, Node003, 50);
-    if ((((global_var(830) bwand 16) != 0) == 1) and (((global_var(830) bwand 4) != 0) == 0) and (((global_var(830) bwand 64) != 0) == 0)) then begin
-        giq_option(4, 1309, 1212, Node998, 50);
+    gsay_reply(SCRIPT_AITEM6, message_str(SCRIPT_AITEM6, 1210));
+    giq_option(4, SCRIPT_AITEM6, 1211, Node003, NEUTRAL_REACTION);
+    if ((((global_var(GVAR_TOWN_REP_COLLY) bwand 16) != 0) == 1) and (((global_var(GVAR_TOWN_REP_COLLY) bwand 4) != 0) == 0) and (((global_var(GVAR_TOWN_REP_COLLY) bwand 64) != 0) == 0)) then begin
+        giq_option(4, SCRIPT_AITEM6, 1212, Node998, NEUTRAL_REACTION);
     end
-    giq_option(4, 1309, 1213, Node999, 50);
-    giq_option(4, 1309, 1214, Node999, 50);
+    giq_option(4, SCRIPT_AITEM6, 1213, Node999, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_AITEM6, 1214, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node003
 begin
-    if ((global_var(830) bwand 2097152) != 0) then begin
-        gsay_reply(1309, message_str(1309, random(1217, 1219)));
+    if ((global_var(GVAR_TOWN_REP_COLLY) bwand 2097152) != 0) then begin
+        gsay_reply(SCRIPT_AITEM6, message_str(SCRIPT_AITEM6, random(1217, 1219)));
     end
     else begin
-        gsay_reply(1309, message_str(1309, 1220));
-        set_global_var(830, global_var(830) bwor 2097152);
+        gsay_reply(SCRIPT_AITEM6, message_str(SCRIPT_AITEM6, 1220));
+        set_global_var(GVAR_TOWN_REP_COLLY, global_var(GVAR_TOWN_REP_COLLY) bwor 2097152);
     end
     if (((local_var(6) bwand 2048) != 0) == 0) then begin
-        if ((global_var(830) bwand 8388608) != 0) then begin
-            giq_option(4, 1309, 1222, Node004, 50);
+        if ((global_var(GVAR_TOWN_REP_COLLY) bwand 8388608) != 0) then begin
+            giq_option(4, SCRIPT_AITEM6, 1222, Node004, NEUTRAL_REACTION);
         end
     end
-    if (((global_var(830) bwand 524288) != 0) and ((local_var(6) bwand 131072) != 0)) then begin
-        giq_option(4, 1309, 1232, Node039, 50);
+    if (((global_var(GVAR_TOWN_REP_COLLY) bwand 524288) != 0) and ((local_var(6) bwand 131072) != 0)) then begin
+        giq_option(4, SCRIPT_AITEM6, 1232, Node039, NEUTRAL_REACTION);
     end
     if ((local_var(6) bwand 131072) != 0) then begin
         if (((local_var(6) bwand 65536) != 0) or ((local_var(6) bwand 32768) != 0)) then begin
-            giq_option(4, 1309, 1227 - ((global_var(830) bwand 524288) != 0), go_secret_place, 50);
+            giq_option(4, SCRIPT_AITEM6, 1227 - ((global_var(GVAR_TOWN_REP_COLLY) bwand 524288) != 0), go_secret_place, NEUTRAL_REACTION);
         end
         else begin
-            giq_option(4, 1309, 1227, Node036, 50);
+            giq_option(4, SCRIPT_AITEM6, 1227, Node036, NEUTRAL_REACTION);
         end
     end
     else begin
         if ((local_var(6) bwand 16384) != 0) then begin
-            giq_option(4, 1309, 1225, Node022, 50);
+            giq_option(4, SCRIPT_AITEM6, 1225, Node022, NEUTRAL_REACTION);
         end
         else begin
-            giq_option(4, 1309, 1224, Node015, 50);
+            giq_option(4, SCRIPT_AITEM6, 1224, Node015, NEUTRAL_REACTION);
         end
     end
     if (((local_var(6) bwand 131072) != 0) == 0) then begin
-        giq_option(4, 1309, 1223, Node007, 50);
+        giq_option(4, SCRIPT_AITEM6, 1223, Node007, NEUTRAL_REACTION);
     end
-    if ((((global_var(830) bwand 16) != 0) == 1) and (((global_var(830) bwand 4) != 0) == 0) and (((global_var(830) bwand 64) != 0) == 0)) then begin
+    if ((((global_var(GVAR_TOWN_REP_COLLY) bwand 16) != 0) == 1) and (((global_var(GVAR_TOWN_REP_COLLY) bwand 4) != 0) == 0) and (((global_var(GVAR_TOWN_REP_COLLY) bwand 64) != 0) == 0)) then begin
         if (((local_var(6) bwand 131072) != 0) == 0) then begin
             if ((local_var(6) bwand 65536) != 0) then begin
-                giq_option(4, 1309, 1229, Node998, 50);
+                giq_option(4, SCRIPT_AITEM6, 1229, Node998, NEUTRAL_REACTION);
             end
             else begin
-                giq_option(4, 1309, 1228, Node011, 50);
+                giq_option(4, SCRIPT_AITEM6, 1228, Node011, NEUTRAL_REACTION);
             end
         end
         else begin
-            giq_option(4, 1309, 1228 + ((local_var(6) bwand 65536) != 0), Node998, 50);
+            giq_option(4, SCRIPT_AITEM6, 1228 + ((local_var(6) bwand 65536) != 0), Node998, NEUTRAL_REACTION);
         end
     end
-    giq_option(4, 1309, 1221, Node999, 50);
+    giq_option(4, SCRIPT_AITEM6, 1221, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node004
 begin
     set_local_var(6, local_var(6) bwor 2048);
-    gsay_reply(1309, message_str(1309, 1242));
-    giq_option(4, 1309, 1243, Node005, 50);
-    giq_option(4, 1309, 1244, Node999, 50);
+    gsay_reply(SCRIPT_AITEM6, message_str(SCRIPT_AITEM6, 1242));
+    giq_option(4, SCRIPT_AITEM6, 1243, Node005, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_AITEM6, 1244, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node005
 begin
-    gsay_reply(1309, message_str(1309, 1245));
-    giq_option(4, 1309, 1246, Node006, 50);
-    giq_option(4, 1309, 1247, Node999, 50);
+    gsay_reply(SCRIPT_AITEM6, message_str(SCRIPT_AITEM6, 1245));
+    giq_option(4, SCRIPT_AITEM6, 1246, Node006, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_AITEM6, 1247, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node006
 begin
-    gsay_reply(1309, message_str(1309, 1250));
-    giq_option(4, 1309, 1251, Node998, 50);
-    giq_option(4, 1309, 1252, Node007, 50);
-    giq_option(4, 1309, 1253, Node999, 50);
+    gsay_reply(SCRIPT_AITEM6, message_str(SCRIPT_AITEM6, 1250));
+    giq_option(4, SCRIPT_AITEM6, 1251, Node998, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_AITEM6, 1252, Node007, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_AITEM6, 1253, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node007
 begin
-    gsay_reply(1309, message_str(1309, 1260));
-    giq_option(4, 1309, 1261, Node008, 50);
-    if ((has_skill(dude_obj, 7) > 21) or (has_skill(dude_obj, 6) > 31)) then begin
+    gsay_reply(SCRIPT_AITEM6, message_str(SCRIPT_AITEM6, 1260));
+    giq_option(4, SCRIPT_AITEM6, 1261, Node008, NEUTRAL_REACTION);
+    if ((has_skill(dude_obj, SKILL_DOCTOR) > 21) or (has_skill(dude_obj, SKILL_FIRST_AID) > 31)) then begin
         if ((local_var(6) bwand 4096) != 0) then begin
-            giq_option(4, 1309, 1264, Node018, 50);
+            giq_option(4, SCRIPT_AITEM6, 1264, Node018, NEUTRAL_REACTION);
         end
     end
-    giq_option(4, 1309, 1262, Node009, 50);
-    giq_option(4, 1309, 1263, Node999, 50);
+    giq_option(4, SCRIPT_AITEM6, 1262, Node009, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_AITEM6, 1263, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node008
 begin
     set_local_var(6, local_var(6) bwor 4096);
-    gsay_reply(1309, message_str(1309, 1270));
-    giq_option(4, 1309, 1271, Node999, 50);
-    if ((has_skill(dude_obj, 7) > 21) or (has_skill(dude_obj, 6) > 31)) then begin
-        giq_option(4, 1309, 1273, Node018, 50);
+    gsay_reply(SCRIPT_AITEM6, message_str(SCRIPT_AITEM6, 1270));
+    giq_option(4, SCRIPT_AITEM6, 1271, Node999, NEUTRAL_REACTION);
+    if ((has_skill(dude_obj, SKILL_DOCTOR) > 21) or (has_skill(dude_obj, SKILL_FIRST_AID) > 31)) then begin
+        giq_option(4, SCRIPT_AITEM6, 1273, Node018, NEUTRAL_REACTION);
     end
-    giq_option(4, 1309, 1272, Node009, 50);
+    giq_option(4, SCRIPT_AITEM6, 1272, Node009, NEUTRAL_REACTION);
 end
 
 procedure Node009
 begin
-    gsay_reply(1309, message_str(1309, 1280));
-    giq_option(4, 1309, 1281, Node999, 50);
+    gsay_reply(SCRIPT_AITEM6, message_str(SCRIPT_AITEM6, 1280));
+    giq_option(4, SCRIPT_AITEM6, 1281, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node010
 begin
     set_local_var(6, local_var(6) bwor 1);
-    set_global_var(830, global_var(830) bwor 2097152);
-    gsay_reply(1309, message_str(1309, 1290));
-    giq_option(4, 1309, 1291, Node020, 50);
-    giq_option(4, 1309, 1292, Node021, 50);
-    if ((((global_var(830) bwand 16) != 0) == 1) and (((global_var(830) bwand 4) != 0) == 0) and (((global_var(830) bwand 64) != 0) == 0)) then begin
-        giq_option(4, 1309, 1293, Node011, 50);
+    set_global_var(GVAR_TOWN_REP_COLLY, global_var(GVAR_TOWN_REP_COLLY) bwor 2097152);
+    gsay_reply(SCRIPT_AITEM6, message_str(SCRIPT_AITEM6, 1290));
+    giq_option(4, SCRIPT_AITEM6, 1291, Node020, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_AITEM6, 1292, Node021, NEUTRAL_REACTION);
+    if ((((global_var(GVAR_TOWN_REP_COLLY) bwand 16) != 0) == 1) and (((global_var(GVAR_TOWN_REP_COLLY) bwand 4) != 0) == 0) and (((global_var(GVAR_TOWN_REP_COLLY) bwand 64) != 0) == 0)) then begin
+        giq_option(4, SCRIPT_AITEM6, 1293, Node011, NEUTRAL_REACTION);
     end
 end
 
 procedure Node011
 begin
-    gsay_reply(1309, message_str(1309, 1300));
-    giq_option(4, 1309, 1301, Node012, 50);
-    giq_option(4, 1309, 1302, Node011a, 50);
-    giq_option(4, 1309, 1303, Node998, 50);
+    gsay_reply(SCRIPT_AITEM6, message_str(SCRIPT_AITEM6, 1300));
+    giq_option(4, SCRIPT_AITEM6, 1301, Node012, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_AITEM6, 1302, Node011a, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_AITEM6, 1303, Node998, NEUTRAL_REACTION);
 end
 
 procedure Node011a
 begin
-    if (((get_critter_stat(dude_obj, 0) + get_critter_stat(dude_obj, 2)) > 12) or (get_pc_stat(1) > 7)) then begin
+    if (((dude_strength + dude_endurance) > 12) or (get_pc_stat(PCSTAT_level) > 7)) then begin
         call Node024();
     end
     else begin
@@ -648,20 +648,20 @@ begin
             end
         end
     end
-    gsay_reply(1309, message_str(1309, 1310));
-    giq_option(4, 1309, 1311, Node999, 50);
-    giq_option(4, 1309, 1312, Node012b, 50);
+    gsay_reply(SCRIPT_AITEM6, message_str(SCRIPT_AITEM6, 1310));
+    giq_option(4, SCRIPT_AITEM6, 1311, Node999, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_AITEM6, 1312, Node012b, NEUTRAL_REACTION);
     if (LVar0 != 0) then begin
-        giq_option(4, 1309, 1313, Node013, 50);
+        giq_option(4, SCRIPT_AITEM6, 1313, Node013, NEUTRAL_REACTION);
     end
-    giq_option(4, 1309, 1314, Node012a, 50);
+    giq_option(4, SCRIPT_AITEM6, 1314, Node012a, NEUTRAL_REACTION);
     return 0;
     LVar0 := 0;
 end
 
 procedure Node012a
 begin
-    if (((get_critter_stat(dude_obj, 0) + get_critter_stat(dude_obj, 2)) > 12) or (get_pc_stat(1) > 7)) then begin
+    if (((dude_strength + dude_endurance) > 12) or (get_pc_stat(PCSTAT_level) > 7)) then begin
         call Node026();
     end
     else begin
@@ -671,93 +671,93 @@ end
 
 procedure Node012b
 begin
-    gsay_message(1309, message_str(1309, 1315), 51);
+    gsay_message(SCRIPT_AITEM6, message_str(SCRIPT_AITEM6, 1315), BAD_REACTION);
     call Node998();
 end
 
 procedure Node013
 begin
-    gsay_reply(1309, message_str(1309, 1320));
-    giq_option(4, 1309, 1321, Node014, 50);
-    giq_option(4, 1309, 1322, Node999, 50);
+    gsay_reply(SCRIPT_AITEM6, message_str(SCRIPT_AITEM6, 1320));
+    giq_option(4, SCRIPT_AITEM6, 1321, Node014, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_AITEM6, 1322, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node014
 begin
-    gsay_reply(1309, message_str(1309, 1330));
-    giq_option(4, 1309, 1331, Node998, 50);
-    if (has_skill(dude_obj, 0) > 59) then begin
-        giq_option(5, 1309, 1332, Node031, 50);
+    gsay_reply(SCRIPT_AITEM6, message_str(SCRIPT_AITEM6, 1330));
+    giq_option(4, SCRIPT_AITEM6, 1331, Node998, NEUTRAL_REACTION);
+    if (has_skill(dude_obj, SKILL_SMALL_GUNS) > 59) then begin
+        giq_option(5, SCRIPT_AITEM6, 1332, Node031, NEUTRAL_REACTION);
     end
-    giq_option(4, 1309, 1333, Node030, 50);
+    giq_option(4, SCRIPT_AITEM6, 1333, Node030, NEUTRAL_REACTION);
 end
 
 procedure Node015
 begin
-    gsay_reply(1309, message_str(1309, 1340));
-    giq_option(4, 1309, 1341, Node016, 50);
-    giq_option(4, 1309, 1342, Node007, 50);
-    giq_option(4, 1309, 1343, Node999, 50);
+    gsay_reply(SCRIPT_AITEM6, message_str(SCRIPT_AITEM6, 1340));
+    giq_option(4, SCRIPT_AITEM6, 1341, Node016, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_AITEM6, 1342, Node007, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_AITEM6, 1343, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node016
 begin
-    gsay_reply(1309, message_str(1309, 1350));
-    giq_option(4, 1309, 1351, Node017, 50);
-    giq_option(4, 1309, 1352, Node998, 50);
+    gsay_reply(SCRIPT_AITEM6, message_str(SCRIPT_AITEM6, 1350));
+    giq_option(4, SCRIPT_AITEM6, 1351, Node017, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_AITEM6, 1352, Node998, NEUTRAL_REACTION);
 end
 
 procedure Node017
 begin
-    gsay_reply(1309, message_str(1309, 1360));
-    giq_option(4, 1309, 1361, Node022, 50);
-    giq_option(4, 1309, 1362, Node999, 50);
+    gsay_reply(SCRIPT_AITEM6, message_str(SCRIPT_AITEM6, 1360));
+    giq_option(4, SCRIPT_AITEM6, 1361, Node022, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_AITEM6, 1362, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node018
 begin
     set_local_var(6, local_var(6) bwor 8192);
-    gsay_reply(1309, message_str(1309, 1370));
-    giq_option(4, 1309, 1371, Node019, 50);
-    giq_option(4, 1309, 1372, Node999, 50);
+    gsay_reply(SCRIPT_AITEM6, message_str(SCRIPT_AITEM6, 1370));
+    giq_option(4, SCRIPT_AITEM6, 1371, Node019, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_AITEM6, 1372, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node019
 begin
-    gsay_reply(1309, message_str(1309, 1380));
-    giq_option(4, 1309, 1381, Node999, 50);
+    gsay_reply(SCRIPT_AITEM6, message_str(SCRIPT_AITEM6, 1380));
+    giq_option(4, SCRIPT_AITEM6, 1381, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node020
 begin
-    gsay_reply(1309, message_str(1309, 1390));
-    giq_option(4, 1309, 1391, Node003, 50);
-    giq_option(4, 1309, 1392, Node999, 50);
+    gsay_reply(SCRIPT_AITEM6, message_str(SCRIPT_AITEM6, 1390));
+    giq_option(4, SCRIPT_AITEM6, 1391, Node003, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_AITEM6, 1392, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node021
 begin
-    gsay_reply(1309, message_str(1309, 1400));
-    giq_option(4, 1309, 1401, Node017, 50);
+    gsay_reply(SCRIPT_AITEM6, message_str(SCRIPT_AITEM6, 1400));
+    giq_option(4, SCRIPT_AITEM6, 1401, Node017, NEUTRAL_REACTION);
 end
 
 procedure Node022
 begin
     set_local_var(6, local_var(6) bwor 16384);
-    gsay_reply(1309, message_str(1309, 1410));
-    giq_option(4, 1309, 1411, Node029, 50);
-    giq_option(4, 1309, 1412, Node999, 50);
+    gsay_reply(SCRIPT_AITEM6, message_str(SCRIPT_AITEM6, 1410));
+    giq_option(4, SCRIPT_AITEM6, 1411, Node029, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_AITEM6, 1412, Node999, NEUTRAL_REACTION);
     if ((local_var(6) bwand 8192) != 0) then begin
-        giq_option(7, 1309, 1413, Node025, 50);
+        giq_option(7, SCRIPT_AITEM6, 1413, Node025, NEUTRAL_REACTION);
     end
-    if ((((global_var(830) bwand 16) != 0) == 1) and (((global_var(830) bwand 4) != 0) == 0) and (((global_var(830) bwand 64) != 0) == 0)) then begin
-        giq_option(4, 1309, 1414, Node022a, 50);
+    if ((((global_var(GVAR_TOWN_REP_COLLY) bwand 16) != 0) == 1) and (((global_var(GVAR_TOWN_REP_COLLY) bwand 4) != 0) == 0) and (((global_var(GVAR_TOWN_REP_COLLY) bwand 64) != 0) == 0)) then begin
+        giq_option(4, SCRIPT_AITEM6, 1414, Node022a, NEUTRAL_REACTION);
     end
 end
 
 procedure Node022a
 begin
-    if (((get_critter_stat(dude_obj, 0) + get_critter_stat(dude_obj, 2)) > 12) or (get_pc_stat(1) > 7)) then begin
+    if (((dude_strength + dude_endurance) > 12) or (get_pc_stat(PCSTAT_level) > 7)) then begin
         call Node026();
     end
     else begin
@@ -767,128 +767,128 @@ end
 
 procedure Node023
 begin
-    gsay_message(1309, message_str(1309, 1420), 51);
+    gsay_message(SCRIPT_AITEM6, message_str(SCRIPT_AITEM6, 1420), BAD_REACTION);
     call Node998();
 end
 
 procedure Node024
 begin
-    gsay_reply(1309, message_str(1309, 1430));
-    giq_option(4, 1309, 1431, Node026, 50);
-    giq_option(4, 1309, 1432, Node998, 50);
+    gsay_reply(SCRIPT_AITEM6, message_str(SCRIPT_AITEM6, 1430));
+    giq_option(4, SCRIPT_AITEM6, 1431, Node026, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_AITEM6, 1432, Node998, NEUTRAL_REACTION);
 end
 
 procedure Node025
 begin
-    gsay_reply(1309, message_str(1309, 1440));
-    giq_option(4, 1309, 1441, Node033, 50);
+    gsay_reply(SCRIPT_AITEM6, message_str(SCRIPT_AITEM6, 1440));
+    giq_option(4, SCRIPT_AITEM6, 1441, Node033, NEUTRAL_REACTION);
 end
 
 procedure Node026
 begin
     set_local_var(6, local_var(6) bwor 65536);
-    gsay_reply(1309, message_str(1309, 1450));
-    giq_option(4, 1309, 1451, Node027, 50);
-    giq_option(4, 1309, 1452, Node998, 50);
+    gsay_reply(SCRIPT_AITEM6, message_str(SCRIPT_AITEM6, 1450));
+    giq_option(4, SCRIPT_AITEM6, 1451, Node027, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_AITEM6, 1452, Node998, NEUTRAL_REACTION);
 end
 
 procedure Node027
 begin
     set_local_var(6, local_var(6) bwor 131072);
-    if (((global_var(830) bwand 524288) != 0) == 0) then begin
+    if (((global_var(GVAR_TOWN_REP_COLLY) bwand 524288) != 0) == 0) then begin
         give_exp_points(200);
-        display_msg(message_str(1309, 1700));
+        display_msg(message_str(SCRIPT_AITEM6, 1700));
     end
-    gsay_reply(1309, message_str(1309, 1460));
-    giq_option(4, 1309, 1461, go_secret_place, 50);
-    giq_option(4, 1309, 1462, Node999, 50);
+    gsay_reply(SCRIPT_AITEM6, message_str(SCRIPT_AITEM6, 1460));
+    giq_option(4, SCRIPT_AITEM6, 1461, go_secret_place, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_AITEM6, 1462, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node028
 begin
-    gsay_reply(1309, message_str(1309, 1470));
-    if (has_skill(dude_obj, 14) > 36) then begin
-        giq_option(5, 1309, 1471, Node032, 50);
+    gsay_reply(SCRIPT_AITEM6, message_str(SCRIPT_AITEM6, 1470));
+    if (has_skill(dude_obj, SKILL_CONVERSANT) > 36) then begin
+        giq_option(5, SCRIPT_AITEM6, 1471, Node032, NEUTRAL_REACTION);
     end
-    giq_option(4, 1309, 1472, Node998, 50);
-    giq_option(4, 1309, 1473, Node999, 50);
+    giq_option(4, SCRIPT_AITEM6, 1472, Node998, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_AITEM6, 1473, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node029
 begin
-    gsay_reply(1309, message_str(1309, 1480));
-    if (has_skill(dude_obj, 14) > 36) then begin
-        giq_option(5, 1309, 1481, Node032, 50);
+    gsay_reply(SCRIPT_AITEM6, message_str(SCRIPT_AITEM6, 1480));
+    if (has_skill(dude_obj, SKILL_CONVERSANT) > 36) then begin
+        giq_option(5, SCRIPT_AITEM6, 1481, Node032, NEUTRAL_REACTION);
     end
-    giq_option(4, 1309, 1482, Node998, 50);
-    giq_option(4, 1309, 1483, Node999, 50);
+    giq_option(4, SCRIPT_AITEM6, 1482, Node998, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_AITEM6, 1483, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node030
 begin
-    gsay_message(1309, message_str(1309, 1490), 51);
+    gsay_message(SCRIPT_AITEM6, message_str(SCRIPT_AITEM6, 1490), BAD_REACTION);
     call Node999();
 end
 
 procedure Node031
 begin
-    gsay_reply(1309, message_str(1309, 1500));
-    giq_option(4, 1309, 1501, Node026, 50);
-    giq_option(4, 1309, 1502, Node998, 50);
+    gsay_reply(SCRIPT_AITEM6, message_str(SCRIPT_AITEM6, 1500));
+    giq_option(4, SCRIPT_AITEM6, 1501, Node026, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_AITEM6, 1502, Node998, NEUTRAL_REACTION);
 end
 
 procedure Node032
 begin
-    gsay_reply(1309, message_str(1309, 1510));
-    giq_option(4, 1309, 1511, Node026, 50);
-    giq_option(4, 1309, 1512, Node998, 50);
+    gsay_reply(SCRIPT_AITEM6, message_str(SCRIPT_AITEM6, 1510));
+    giq_option(4, SCRIPT_AITEM6, 1511, Node026, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_AITEM6, 1512, Node998, NEUTRAL_REACTION);
 end
 
 procedure Node033
 begin
-    gsay_reply(1309, message_str(1309, 1520));
-    giq_option(4, 1309, 1521, Node034, 50);
+    gsay_reply(SCRIPT_AITEM6, message_str(SCRIPT_AITEM6, 1520));
+    giq_option(4, SCRIPT_AITEM6, 1521, Node034, NEUTRAL_REACTION);
 end
 
 procedure Node034
 begin
-    gsay_reply(1309, message_str(1309, 1530));
-    giq_option(4, 1309, 1531, Node035, 50);
+    gsay_reply(SCRIPT_AITEM6, message_str(SCRIPT_AITEM6, 1530));
+    giq_option(4, SCRIPT_AITEM6, 1531, Node035, NEUTRAL_REACTION);
 end
 
 procedure Node035
 begin
     set_local_var(6, local_var(6) bwor 131072);
-    if (((global_var(830) bwand 524288) != 0) == 0) then begin
+    if (((global_var(GVAR_TOWN_REP_COLLY) bwand 524288) != 0) == 0) then begin
         give_exp_points(250);
-        display_msg(message_str(1309, 1701));
+        display_msg(message_str(SCRIPT_AITEM6, 1701));
     end
-    gsay_reply(1309, message_str(1309, 1540));
-    giq_option(4, 1309, 1541, Node036, 50);
-    if ((((global_var(830) bwand 16) != 0) == 1) and (((global_var(830) bwand 4) != 0) == 0)) then begin
-        giq_option(5, 1309, 1543, Node038, 50);
+    gsay_reply(SCRIPT_AITEM6, message_str(SCRIPT_AITEM6, 1540));
+    giq_option(4, SCRIPT_AITEM6, 1541, Node036, NEUTRAL_REACTION);
+    if ((((global_var(GVAR_TOWN_REP_COLLY) bwand 16) != 0) == 1) and (((global_var(GVAR_TOWN_REP_COLLY) bwand 4) != 0) == 0)) then begin
+        giq_option(5, SCRIPT_AITEM6, 1543, Node038, NEUTRAL_REACTION);
     end
-    giq_option(4, 1309, 1544, Node999, 50);
+    giq_option(4, SCRIPT_AITEM6, 1544, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node036
 begin
-    gsay_reply(1309, message_str(1309, 1550));
-    giq_option(4, 1309, 1551, Node037, 50);
-    giq_option(4, 1309, 1552, Node038, 50);
-    giq_option(4, 1309, 1553, Node999, 50);
+    gsay_reply(SCRIPT_AITEM6, message_str(SCRIPT_AITEM6, 1550));
+    giq_option(4, SCRIPT_AITEM6, 1551, Node037, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_AITEM6, 1552, Node038, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_AITEM6, 1553, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node037
 begin
-    gsay_reply(1309, message_str(1309, 1560));
+    gsay_reply(SCRIPT_AITEM6, message_str(SCRIPT_AITEM6, 1560));
     if (has_skill(dude_obj, 14 > 31)) then begin
-        giq_option(4, 1309, 1561, Node038, 50);
+        giq_option(4, SCRIPT_AITEM6, 1561, Node038, NEUTRAL_REACTION);
     end
     if ((obj_is_carrying_obj_pid(dude_obj, 106) + obj_is_carrying_obj_pid(dude_obj, 124) + obj_is_carrying_obj_pid(dude_obj, 125) + obj_is_carrying_obj_pid(dude_obj, 310) + obj_is_carrying_obj_pid(dude_obj, 311) + obj_is_carrying_obj_pid(dude_obj, 469)) > 0) then begin
-        giq_option(4, 1309, 1562, Node037a, 50);
+        giq_option(4, SCRIPT_AITEM6, 1562, Node037a, NEUTRAL_REACTION);
     end
-    giq_option(4, 1309, 1563, Node999, 50);
+    giq_option(4, SCRIPT_AITEM6, 1563, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node037a
@@ -929,94 +929,93 @@ begin
             amt_ower := amt_ower - 1;
         end
     end
-    gsay_reply(1309, message_str(1309, 1565));
-    giq_option(4, 1309, 1566, Node038, 50);
+    gsay_reply(SCRIPT_AITEM6, message_str(SCRIPT_AITEM6, 1565));
+    giq_option(4, SCRIPT_AITEM6, 1566, Node038, NEUTRAL_REACTION);
 end
 
 procedure Node038
 begin
     set_local_var(6, local_var(6) bwor 32768);
-    gsay_reply(1309, message_str(1309, 1570));
-    giq_option(4, 1309, 1571, go_secret_place, 50);
-    giq_option(4, 1309, 1572, Node999, 50);
+    gsay_reply(SCRIPT_AITEM6, message_str(SCRIPT_AITEM6, 1570));
+    giq_option(4, SCRIPT_AITEM6, 1571, go_secret_place, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_AITEM6, 1572, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node039
 begin
-    gsay_reply(1309, message_str(1309, 1580));
-    giq_option(4, 1309, 1581, Node041, 50);
-    giq_option(4, 1309, 1582, Node040, 50);
-    giq_option(4, 1309, 1583, Node040, 50);
+    gsay_reply(SCRIPT_AITEM6, message_str(SCRIPT_AITEM6, 1580));
+    giq_option(4, SCRIPT_AITEM6, 1581, Node041, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_AITEM6, 1582, Node040, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_AITEM6, 1583, Node040, NEUTRAL_REACTION);
 end
 
 procedure Node040
 begin
-    gsay_reply(1309, message_str(1309, 1590));
-    giq_option(4, 1309, 1591, Node042, 50);
-    giq_option(4, 1309, 1592, Node999, 50);
+    gsay_reply(SCRIPT_AITEM6, message_str(SCRIPT_AITEM6, 1590));
+    giq_option(4, SCRIPT_AITEM6, 1591, Node042, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_AITEM6, 1592, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node041
 begin
-    gsay_reply(1309, message_str(1309, 1600));
-    giq_option(4, 1309, 1601, Node042, 50);
-    giq_option(4, 1309, 1602, Node999, 50);
+    gsay_reply(SCRIPT_AITEM6, message_str(SCRIPT_AITEM6, 1600));
+    giq_option(4, SCRIPT_AITEM6, 1601, Node042, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_AITEM6, 1602, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node042
 begin
-    gsay_reply(1309, message_str(1309, 1610));
-    giq_option(4, 1309, 1611, Node043, 50);
+    gsay_reply(SCRIPT_AITEM6, message_str(SCRIPT_AITEM6, 1610));
+    giq_option(4, SCRIPT_AITEM6, 1611, Node043, NEUTRAL_REACTION);
 end
 
 procedure Node043
 begin
-    gsay_reply(1309, message_str(1309, 1620));
-    giq_option(4, 1309, 1621, Node044, 50);
-    giq_option(4, 1309, 1622, Node999, 50);
+    gsay_reply(SCRIPT_AITEM6, message_str(SCRIPT_AITEM6, 1620));
+    giq_option(4, SCRIPT_AITEM6, 1621, Node044, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_AITEM6, 1622, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node044
 begin
-    gsay_reply(1309, message_str(1309, 1630));
-    giq_option(4, 1309, 1631, Node045, 50);
-    giq_option(4, 1309, 1632, Node999, 50);
+    gsay_reply(SCRIPT_AITEM6, message_str(SCRIPT_AITEM6, 1630));
+    giq_option(4, SCRIPT_AITEM6, 1631, Node045, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_AITEM6, 1632, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node045
 begin
-    gsay_reply(1309, message_str(1309, 1640));
-    giq_option(4, 1309, 1641, Node046, 50);
+    gsay_reply(SCRIPT_AITEM6, message_str(SCRIPT_AITEM6, 1640));
+    giq_option(4, SCRIPT_AITEM6, 1641, Node046, NEUTRAL_REACTION);
 end
 
 procedure Node046
 begin
-    gsay_reply(1309, message_str(1309, 1650));
-    giq_option(4, 1309, 1651, Node047, 50);
-    giq_option(4, 1309, 1652, Node999, 50);
+    gsay_reply(SCRIPT_AITEM6, message_str(SCRIPT_AITEM6, 1650));
+    giq_option(4, SCRIPT_AITEM6, 1651, Node047, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_AITEM6, 1652, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node047
 begin
-    gsay_reply(1309, message_str(1309, 1660));
-    giq_option(4, 1309, 1661, Node048, 50);
+    gsay_reply(SCRIPT_AITEM6, message_str(SCRIPT_AITEM6, 1660));
+    giq_option(4, SCRIPT_AITEM6, 1661, Node048, NEUTRAL_REACTION);
 end
 
 procedure Node048
 begin
-    gsay_reply(1309, message_str(1309, 1670));
-    giq_option(4, 1309, 1671, Node049, 50);
+    gsay_reply(SCRIPT_AITEM6, message_str(SCRIPT_AITEM6, 1670));
+    giq_option(4, SCRIPT_AITEM6, 1671, Node049, NEUTRAL_REACTION);
 end
 
 procedure Node049
 begin
-    gsay_reply(1309, message_str(1309, 1680));
-    giq_option(4, 1309, 1681, Node050, 50);
+    gsay_reply(SCRIPT_AITEM6, message_str(SCRIPT_AITEM6, 1680));
+    giq_option(4, SCRIPT_AITEM6, 1681, Node050, NEUTRAL_REACTION);
 end
 
 procedure Node050
 begin
-    gsay_reply(1309, message_str(1309, 1690));
-    giq_option(4, 1309, 1691, Node999, 50);
+    gsay_reply(SCRIPT_AITEM6, message_str(SCRIPT_AITEM6, 1690));
+    giq_option(4, SCRIPT_AITEM6, 1691, Node999, NEUTRAL_REACTION);
 end
-
