diff --git a/Mods/MegaMod/Original/TRIBEM3.ssl b/Mods/MegaMod/ReDefine/TRIBEM3.ssl
index c5d6f89..f59f6f6 100644
--- a/Mods/MegaMod/Original/TRIBEM3.ssl
+++ b/Mods/MegaMod/ReDefine/TRIBEM3.ssl
@@ -142,8 +142,8 @@ end
 
 procedure map_enter_p_proc
 begin
-    if (metarule(14, 0)) then begin
-        display_msg(message_str(1510, 4100));
+    if (map_first_run) then begin
+        display_msg(message_str(SCRIPT_TRIBEC1, 4100));
         game_ui_enable;
     end
     if (elevation(dude_obj) == 0) then begin
@@ -313,4 +313,3 @@ begin
         set_light_level(50);
     end
 end
-
