diff --git a/Mods/MegaMod/Original/TRIBEC5.ssl b/Mods/MegaMod/ReDefine/TRIBEC5.ssl
index a0eda28..7472d3e 100644
--- a/Mods/MegaMod/Original/TRIBEC5.ssl
+++ b/Mods/MegaMod/ReDefine/TRIBEC5.ssl
@@ -159,7 +159,7 @@ end
 
 procedure critter_p_proc
 begin
-    if (global_var(824) == 3) then begin
+    if (global_var(GVAR_FIND_SULIK) == 3) then begin
         add_timer_event(self_obj, 15, 1);
     end
 end
@@ -172,17 +172,17 @@ procedure look_at_p_proc
 begin
     script_overrides;
     if (local_var(4) == 0) then begin
-        display_msg(message_str(1515, 100));
+        display_mstr(100);
     end
     else begin
-        display_msg(message_str(1515, 101));
+        display_mstr(101);
     end
 end
 
 procedure description_p_proc
 begin
     script_overrides;
-    display_msg(message_str(1515, 102));
+    display_mstr(102);
 end
 
 procedure use_skill_on_p_proc
@@ -205,13 +205,13 @@ procedure map_update_p_proc
 begin
     if (game_time_hour <= 400) then begin
         if (map_var(1) == 1) then begin
-            if (has_trait(1, self_obj, 666) == 0) then begin
+            if (obj_is_visible_flag(self_obj) == 0) then begin
                 set_obj_visibility(self_obj, 0);
             end
         end
     end
     else begin
-        if (has_trait(1, self_obj, 666) == 1) then begin
+        if (obj_is_visible_flag(self_obj) == 1) then begin
             set_obj_visibility(self_obj, 1);
         end
     end
@@ -220,10 +220,10 @@ end
 procedure talk_p_proc
 begin
     if (local_var(7) == 1) then begin
-        display_msg(message_str(1515, 200));
+        display_mstr(200);
     end
     else begin
-        if (global_var(824) == 1) then begin
+        if (global_var(GVAR_FIND_SULIK) == 1) then begin
             start_gdialog(1515, self_obj, 4, -1, -1);
             gsay_start;
             call Node001();
@@ -231,7 +231,7 @@ begin
             end_dialogue;
         end
         else begin
-            if (global_var(824) == 2) then begin
+            if (global_var(GVAR_FIND_SULIK) == 2) then begin
                 start_gdialog(1515, self_obj, 4, -1, -1);
                 gsay_start;
                 call Node006();
@@ -239,7 +239,7 @@ begin
                 end_dialogue;
             end
             else begin
-                float_msg(self_obj, message_str(1515, random(103, 105)), 8);
+                float_msg(self_obj, mstr(random(103, 105)), FLOAT_MSG_YELLOW);
             end
         end
     end
@@ -247,40 +247,40 @@ end
 
 procedure Node001
 begin
-    gsay_reply(1515, 110);
-    giq_option(4, 1515, 111, Node002, 50);
+    Reply(110);
+    giq_option(4, NAME, 111, Node002, NEUTRAL_REACTION);
 end
 
 procedure Node002
 begin
-    gsay_reply(1515, 120);
-    giq_option(4, 1515, 121, Node003, 50);
-    giq_option(4, 1515, 122, Node991, 51);
+    Reply(120);
+    giq_option(4, NAME, 121, Node003, NEUTRAL_REACTION);
+    giq_option(4, NAME, 122, Node991, BAD_REACTION);
 end
 
 procedure Node003
 begin
-    gsay_reply(1515, 130);
-    giq_option(4, 1515, 131, Node004, 50);
+    Reply(130);
+    giq_option(4, NAME, 131, Node004, NEUTRAL_REACTION);
 end
 
 procedure Node004
 begin
-    gsay_reply(1515, 140);
-    giq_option(4, 1515, 141, Node005, 50);
+    Reply(140);
+    giq_option(4, NAME, 141, Node005, NEUTRAL_REACTION);
 end
 
 procedure Node005
 begin
-    set_global_var(824, 2);
-    gsay_reply(1515, 150);
-    giq_option(4, 1515, 151, Node999, 50);
+    set_global_var(GVAR_FIND_SULIK, 2);
+    Reply(150);
+    giq_option(4, NAME, 151, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node006
 begin
-    gsay_reply(1515, 160);
-    giq_option(4, 1515, 161, Node999, 50);
+    Reply(160);
+    giq_option(4, NAME, 161, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node998
@@ -307,4 +307,3 @@ begin
         destroy_object(self_obj);
     end
 end
-
