diff --git a/Mods/MegaMod/Original/TRIBESP1.ssl b/Mods/MegaMod/ReDefine/TRIBESP1.ssl
index 35ca467..571c8bc 100644
--- a/Mods/MegaMod/Original/TRIBESP1.ssl
+++ b/Mods/MegaMod/ReDefine/TRIBESP1.ssl
@@ -24,7 +24,7 @@ end
 procedure spatial_p_proc
 begin
     if (source_obj == dude_obj) then begin
-        if (global_var(1127) == 2) then begin
+        if (global_var(GVAR_WIPED_OUT_RANGER_SAFE_HOUSES) == 2) then begin
             if (not(WARRIOR_obj == -1)) then begin
                 if (map_var(0) == 1) then begin
                     if (local_var(4) == 0) then begin
@@ -43,60 +43,60 @@ end
 
 procedure Node001
 begin
-    gsay_reply(1510, 5200);
-    giq_option(4, 1510, 5201, Node002, 50);
+    gsay_reply(SCRIPT_TRIBEC1, 5200);
+    giq_option(4, SCRIPT_TRIBEC1, 5201, Node002, NEUTRAL_REACTION);
 end
 
 procedure Node002
 begin
-    gsay_reply(1510, 5202);
-    giq_option(4, 1510, 5203, Node003, 50);
+    gsay_reply(SCRIPT_TRIBEC1, 5202);
+    giq_option(4, SCRIPT_TRIBEC1, 5203, Node003, NEUTRAL_REACTION);
 end
 
 procedure Node003
 begin
-    gsay_reply(1510, 5204);
-    giq_option(4, 1510, 5205, Node004, 50);
+    gsay_reply(SCRIPT_TRIBEC1, 5204);
+    giq_option(4, SCRIPT_TRIBEC1, 5205, Node004, NEUTRAL_REACTION);
 end
 
 procedure Node004
 begin
-    gsay_reply(1510, 5206);
-    giq_option(4, 1510, 5207, Node005, 50);
+    gsay_reply(SCRIPT_TRIBEC1, 5206);
+    giq_option(4, SCRIPT_TRIBEC1, 5207, Node005, NEUTRAL_REACTION);
 end
 
 procedure Node005
 begin
-    gsay_reply(1510, 5208);
-    giq_option(4, 1510, 5209, Node006, 50);
+    gsay_reply(SCRIPT_TRIBEC1, 5208);
+    giq_option(4, SCRIPT_TRIBEC1, 5209, Node006, NEUTRAL_REACTION);
 end
 
 procedure Node006
 begin
-    gsay_reply(1510, 5210);
-    giq_option(4, 1510, 5211, Node007, 50);
+    gsay_reply(SCRIPT_TRIBEC1, 5210);
+    giq_option(4, SCRIPT_TRIBEC1, 5211, Node007, NEUTRAL_REACTION);
 end
 
 procedure Node007
 begin
-    gsay_reply(1510, 5212);
-    if (has_skill(dude_obj, 7) > 100) then begin
-        giq_option(4, 1510, 5213, Node008, 50);
+    gsay_reply(SCRIPT_TRIBEC1, 5212);
+    if (has_skill(dude_obj, SKILL_DOCTOR) > 100) then begin
+        giq_option(4, SCRIPT_TRIBEC1, 5213, Node008, NEUTRAL_REACTION);
     end
     else begin
-        if (get_critter_stat(dude_obj, 0) > 5) then begin
-            giq_option(4, 1510, 5214, Node011, 50);
+        if (dude_strength > 5) then begin
+            giq_option(4, SCRIPT_TRIBEC1, 5214, Node011, NEUTRAL_REACTION);
         end
         else begin
             if (party_member_obj(16777377) != 0) then begin
-                giq_option(4, 1510, 5217, Node011, 50);
+                giq_option(4, SCRIPT_TRIBEC1, 5217, Node011, NEUTRAL_REACTION);
             end
             else begin
                 if (party_member_obj(16777313) != 0) then begin
-                    giq_option(4, 1510, 5216, Node011, 50);
+                    giq_option(4, SCRIPT_TRIBEC1, 5216, Node011, NEUTRAL_REACTION);
                 end
                 else begin
-                    giq_option(4, 1510, 5215, Node012, 50);
+                    giq_option(4, SCRIPT_TRIBEC1, 5215, Node012, NEUTRAL_REACTION);
                 end
             end
         end
@@ -106,19 +106,19 @@ end
 procedure Node008
 begin
     critter_injure(self_obj, 8 bwor 8388608);
-    display_msg(message_str(1510, 5600));
+    display_msg(message_str(SCRIPT_TRIBEC1, 5600));
     give_exp_points(25);
-    display_msg(message_str(14, 100) + 25 + message_str(14, 101));
-    gsay_reply(1510, 5218);
-    giq_option(4, 1510, 5209, Node009, 50);
+    display_msg(g_mstr(100) + 25 + g_mstr(101));
+    gsay_reply(SCRIPT_TRIBEC1, 5218);
+    giq_option(4, SCRIPT_TRIBEC1, 5209, Node009, NEUTRAL_REACTION);
 end
 
 procedure Node009
 begin
     critter_injure(self_obj, 8 bwor 8388608);
     critter_heal(self_obj, 10);
-    gsay_reply(1510, 5219);
-    giq_option(4, 1510, 5220, Node010, 50);
+    gsay_reply(SCRIPT_TRIBEC1, 5219);
+    giq_option(4, SCRIPT_TRIBEC1, 5220, Node010, NEUTRAL_REACTION);
 end
 
 procedure Node010
@@ -126,8 +126,8 @@ begin
     gfade_out(600);
     gfade_in(600);
     game_time_advance(60 * 10 * 15);
-    gsay_reply(1510, 5221);
-    set_global_var(1127, 4);
+    gsay_reply(SCRIPT_TRIBEC1, 5221);
+    set_global_var(GVAR_WIPED_OUT_RANGER_SAFE_HOUSES, 4);
     kill_critter_type(16777752, 0);
     load_map(163, 0);
 end
@@ -136,11 +136,10 @@ procedure Node011
 begin
     kill_critter_type(16777752, 0);
     load_map(163, 0);
-    set_global_var(1127, 4);
+    set_global_var(GVAR_WIPED_OUT_RANGER_SAFE_HOUSES, 4);
 end
 
 procedure Node012
 begin
-    set_global_var(1127, 3);
+    set_global_var(GVAR_WIPED_OUT_RANGER_SAFE_HOUSES, 3);
 end
-
