diff --git a/Mods/MegaMod/Original/V13ENT.ssl b/Mods/MegaMod/ReDefine/V13ENT.ssl
index 8eb7513..eec040a 100644
--- a/Mods/MegaMod/Original/V13ENT.ssl
+++ b/Mods/MegaMod/ReDefine/V13ENT.ssl
@@ -141,14 +141,14 @@ procedure map_enter_p_proc
 begin
     set_light_level(50);
     i_vault13_door := -1;
-    if (global_var(494) < 2) then begin
-        set_global_var(494, 2);
+    if (global_var(GVAR_FALLOUT_2) < 2) then begin
+        set_global_var(GVAR_FALLOUT_2, 2);
     end
-    if (not(metarule(22, 0))) then begin
+    if (not(is_loading_game)) then begin
         debug_msg("V13ENT: FIXED_PRAM " + fixed_param);
-        if (global_var(27) == 5) then begin
+        if (global_var(GVAR_LOAD_MAP_INDEX) == 5) then begin
             override_map_start(17290 % 200, 17290 / 200, 0, 2);
-            set_global_var(27, 0);
+            set_global_var(GVAR_LOAD_MAP_INDEX, 0);
         end
         else begin
             override_map_start(26148 % 200, 26148 / 200, 0, 1);
@@ -160,4 +160,3 @@ procedure map_update_p_proc
 begin
     set_light_level(50);
 end
-
