diff --git a/Mods/MegaMod/Original/V14COMP2.ssl b/Mods/MegaMod/ReDefine/V14COMP2.ssl
index fa65741..44ebe93 100644
--- a/Mods/MegaMod/Original/V14COMP2.ssl
+++ b/Mods/MegaMod/ReDefine/V14COMP2.ssl
@@ -108,17 +108,17 @@ procedure look_at_p_proc
 begin
     script_overrides;
     if (local_var(0) == 0) then begin
-        display_msg(message_str(635, 200));
+        display_msg(message_str(SCRIPT_BSCOMP4, 200));
     end
     else begin
-        display_msg(message_str(635, 201));
+        display_msg(message_str(SCRIPT_BSCOMP4, 201));
     end
 end
 
 procedure description_p_proc
 begin
     script_overrides;
-    display_msg(message_str(635, 202));
+    display_msg(message_str(SCRIPT_BSCOMP4, 202));
 end
 
 procedure use_skill_on_p_proc
@@ -164,38 +164,38 @@ procedure Node001
 begin
     if (map_var(4) == 2) then begin
         if (map_var(4) bwand 4) then begin
-            gsay_reply(635, 212);
+            gsay_reply(SCRIPT_BSCOMP4, 212);
         end
         else begin
             if (map_var(4) bwand 1) then begin
-                gsay_reply(635, 203);
+                gsay_reply(SCRIPT_BSCOMP4, 203);
             end
             else begin
                 if (map_var(4) bwand 8) then begin
-                    gsay_reply(635, 213);
+                    gsay_reply(SCRIPT_BSCOMP4, 213);
                 end
                 else begin
                     if (map_var(4) bwand 2) then begin
-                        gsay_reply(635, 214);
+                        gsay_reply(SCRIPT_BSCOMP4, 214);
                     end
                 end
             end
         end
     end
     else begin
-        gsay_reply(635, 104);
+        gsay_reply(SCRIPT_BSCOMP4, 104);
     end
     if (map_var(4) == 2) then begin
         if (map_var(4) bwand 1) then begin
-            giq_option(1, 635, 205, Node002, 50);
+            giq_option(1, SCRIPT_BSCOMP4, 205, Node002, NEUTRAL_REACTION);
         end
         else begin
             if (map_var(4) bwand 2) then begin
-                giq_option(1, 635, 206, Node003, 50);
+                giq_option(1, SCRIPT_BSCOMP4, 206, Node003, NEUTRAL_REACTION);
             end
         end
     end
-    giq_option(1, 635, 207, Node999, 50);
+    giq_option(1, SCRIPT_BSCOMP4, 207, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node002
@@ -203,12 +203,12 @@ begin
     if (is_success(roll_vs_skill(dude_obj, 12, 0))) then begin
         set_map_var(4, map_var(4) bwand (-1 - 1));
         set_map_var(4, map_var(4) bwor 2);
-        gsay_reply(635, 208);
+        gsay_reply(SCRIPT_BSCOMP4, 208);
     end
     else begin
-        gsay_reply(635, 220);
+        gsay_reply(SCRIPT_BSCOMP4, 220);
     end
-    giq_option(1, 635, 211, Node999, 50);
+    giq_option(1, SCRIPT_BSCOMP4, 211, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node003
@@ -216,15 +216,14 @@ begin
     if (is_success(roll_vs_skill(dude_obj, 12, 0))) then begin
         set_map_var(3, map_var(4) bwand (-1 - 2));
         set_map_var(3, map_var(4) bwor 1);
-        gsay_reply(635, 210);
+        gsay_reply(SCRIPT_BSCOMP4, 210);
     end
     else begin
-        gsay_reply(635, 221);
+        gsay_reply(SCRIPT_BSCOMP4, 221);
     end
-    giq_option(1, 635, 211, Node999, 50);
+    giq_option(1, SCRIPT_BSCOMP4, 211, Node999, NEUTRAL_REACTION);
 end
 
 procedure timed_event_p_proc
 begin
 end
-
