diff --git a/Mods/MegaMod/Original/V14DOOR.ssl b/Mods/MegaMod/ReDefine/V14DOOR.ssl
index a0253dc..1a49d5b 100644
--- a/Mods/MegaMod/Original/V14DOOR.ssl
+++ b/Mods/MegaMod/ReDefine/V14DOOR.ssl
@@ -150,13 +150,13 @@ end
 procedure look_at_p_proc
 begin
     script_overrides;
-    display_msg(message_str(1582, 104));
+    display_mstr(104);
 end
 
 procedure description_p_proc
 begin
     script_overrides;
-    display_msg(message_str(1582, 100));
+    display_mstr(100);
 end
 
 procedure use_p_proc
@@ -173,13 +173,13 @@ begin
     LVar0 := obj_pid(obj_being_used_with);
     if (LVar0 != 653) then begin
         script_overrides;
-        display_msg(message_str(1582, 100));
+        display_mstr(100);
     end
     else begin
         script_overrides;
         Allow_Open := 1;
         obj_unlock(self_obj);
-        display_msg(message_str(1582, 101));
+        display_mstr(101);
     end
     return 0;
     LVar0 := 0;
@@ -188,7 +188,7 @@ end
 procedure use_skill_on_p_proc
 begin
     script_overrides;
-    display_msg(message_str(1582, 100));
+    display_mstr(100);
     call Damage_Critter();
 end
 
@@ -205,13 +205,12 @@ begin
     LVar0 := random(20, 40);
     if (source_obj == dude_obj) then begin
         critter_dmg(dude_obj, LVar0, 6 bwor 256);
-        display_msg(message_str(1582, 166));
+        display_mstr(166);
     end
     else begin
         critter_dmg(source_obj, LVar0, 6 bwor 256);
-        display_msg(obj_name(source_obj) + message_str(1582, 168));
+        display_msg(obj_name(source_obj) + mstr(168));
     end
     return 0;
     LVar0 := 0;
 end
-
