diff --git a/Mods/MegaMod/Original/V14DOWSON.ssl b/Mods/MegaMod/ReDefine/V14DOWSON.ssl
index 48be5eb..9a00e96 100644
--- a/Mods/MegaMod/Original/V14DOWSON.ssl
+++ b/Mods/MegaMod/ReDefine/V14DOWSON.ssl
@@ -277,7 +277,7 @@ end
 procedure critter_p_proc
 begin
     if ((local_var(1) == 2) and (tile_distance_objs(self_obj, dude_obj) <= 4)) then begin
-        attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+        attack(dude_obj);
     end
 end
 
@@ -345,17 +345,17 @@ procedure look_at_p_proc
 begin
     script_overrides;
     if (local_var(0) == 0) then begin
-        display_msg(message_str(1594, 100));
+        display_msg(message_str(SCRIPT_V14DOWSN, 100));
     end
     else begin
-        display_msg(message_str(1594, 101));
+        display_msg(message_str(SCRIPT_V14DOWSN, 101));
     end
 end
 
 procedure description_p_proc
 begin
     script_overrides;
-    display_msg(message_str(1594, 102));
+    display_msg(message_str(SCRIPT_V14DOWSN, 102));
 end
 
 procedure timed_event_p_proc
@@ -386,138 +386,138 @@ end
 
 procedure Node001
 begin
-    gsay_reply(1594, 103);
-    giq_option(3, 1594, "" + obj_name(dude_obj) + ".", Node002, 50);
-    giq_option(3, 1594, 104, Node999, 50);
-    giq_option(3, 1594, 105, Node998, 51);
-    giq_option(-3, 1594, 106, Node999, 51);
+    gsay_reply(SCRIPT_V14DOWSN, 103);
+    giq_option(3, SCRIPT_V14DOWSN, "" + dude_name + ".", Node002, NEUTRAL_REACTION);
+    giq_option(3, SCRIPT_V14DOWSN, 104, Node999, NEUTRAL_REACTION);
+    giq_option(3, SCRIPT_V14DOWSN, 105, Node998, BAD_REACTION);
+    giq_option(-3, SCRIPT_V14DOWSN, 106, Node999, BAD_REACTION);
 end
 
 procedure Node002
 begin
     set_local_var(0, 1);
-    gsay_reply(1594, 107);
-    if (get_critter_stat(dude_obj, 34) == 0) then begin
-        giq_option(3, 1594, 108, Node003, 50);
+    gsay_reply(SCRIPT_V14DOWSN, 107);
+    if (dude_gender == 0) then begin
+        giq_option(3, SCRIPT_V14DOWSN, 108, Node003, NEUTRAL_REACTION);
     end
     else begin
-        giq_option(3, 1594, 201, Node003, 50);
+        giq_option(3, SCRIPT_V14DOWSN, 201, Node003, NEUTRAL_REACTION);
     end
-    giq_option(3, 1594, 109, Node998, 51);
+    giq_option(3, SCRIPT_V14DOWSN, 109, Node998, BAD_REACTION);
 end
 
 procedure Node003
 begin
-    if (get_critter_stat(dude_obj, 34) == 0) then begin
-        gsay_reply(1594, 110);
+    if (dude_gender == 0) then begin
+        gsay_reply(SCRIPT_V14DOWSN, 110);
     end
     else begin
-        gsay_reply(1594, 200);
+        gsay_reply(SCRIPT_V14DOWSN, 200);
     end
-    giq_option(3, 1594, 111, Node004, 50);
+    giq_option(3, SCRIPT_V14DOWSN, 111, Node004, NEUTRAL_REACTION);
 end
 
 procedure Node004
 begin
-    gsay_reply(1594, 112);
-    giq_option(3, 1594, 113, Node005, 50);
-    giq_option(3, 1594, 114, Node998, 51);
+    gsay_reply(SCRIPT_V14DOWSN, 112);
+    giq_option(3, SCRIPT_V14DOWSN, 113, Node005, NEUTRAL_REACTION);
+    giq_option(3, SCRIPT_V14DOWSN, 114, Node998, BAD_REACTION);
 end
 
 procedure Node005
 begin
     set_local_var(0, 2);
-    gsay_reply(1594, 116);
-    giq_option(3, 1594, 117, Node006, 51);
-    giq_option(3, 1594, 118, Node007, 50);
-    giq_option(3, 1594, 119, Node008, 50);
-    giq_option(3, 1594, 125, Node010, 50);
-    giq_option(3, 1594, 115, Node999, 50);
+    gsay_reply(SCRIPT_V14DOWSN, 116);
+    giq_option(3, SCRIPT_V14DOWSN, 117, Node006, BAD_REACTION);
+    giq_option(3, SCRIPT_V14DOWSN, 118, Node007, NEUTRAL_REACTION);
+    giq_option(3, SCRIPT_V14DOWSN, 119, Node008, NEUTRAL_REACTION);
+    giq_option(3, SCRIPT_V14DOWSN, 125, Node010, NEUTRAL_REACTION);
+    giq_option(3, SCRIPT_V14DOWSN, 115, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node006
 begin
-    gsay_reply(1594, 121);
-    giq_option(3, 1594, 123, Node009, 50);
-    giq_option(3, 1594, 130, Node999, 50);
+    gsay_reply(SCRIPT_V14DOWSN, 121);
+    giq_option(3, SCRIPT_V14DOWSN, 123, Node009, NEUTRAL_REACTION);
+    giq_option(3, SCRIPT_V14DOWSN, 130, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node007
 begin
-    gsay_reply(1594, 120);
-    giq_option(3, 1594, 123, Node009, 50);
-    giq_option(3, 1594, 130, Node999, 50);
+    gsay_reply(SCRIPT_V14DOWSN, 120);
+    giq_option(3, SCRIPT_V14DOWSN, 123, Node009, NEUTRAL_REACTION);
+    giq_option(3, SCRIPT_V14DOWSN, 130, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node008
 begin
-    gsay_reply(1594, 122);
-    giq_option(3, 1594, 123, Node009, 50);
-    giq_option(3, 1594, 130, Node999, 50);
+    gsay_reply(SCRIPT_V14DOWSN, 122);
+    giq_option(3, SCRIPT_V14DOWSN, 123, Node009, NEUTRAL_REACTION);
+    giq_option(3, SCRIPT_V14DOWSN, 130, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node009
 begin
-    gsay_reply(1594, 124);
-    giq_option(3, 1594, 117, Node006, 51);
-    giq_option(3, 1594, 118, Node007, 50);
-    giq_option(3, 1594, 119, Node008, 50);
-    giq_option(3, 1594, 125, Node010, 50);
-    giq_option(3, 1594, 115, Node999, 50);
+    gsay_reply(SCRIPT_V14DOWSN, 124);
+    giq_option(3, SCRIPT_V14DOWSN, 117, Node006, BAD_REACTION);
+    giq_option(3, SCRIPT_V14DOWSN, 118, Node007, NEUTRAL_REACTION);
+    giq_option(3, SCRIPT_V14DOWSN, 119, Node008, NEUTRAL_REACTION);
+    giq_option(3, SCRIPT_V14DOWSN, 125, Node010, NEUTRAL_REACTION);
+    giq_option(3, SCRIPT_V14DOWSN, 115, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node010
 begin
     set_local_var(0, 3);
-    gsay_reply(1594, 126);
-    giq_option(3, 1594, 127, Node999, 50);
+    gsay_reply(SCRIPT_V14DOWSN, 126);
+    giq_option(3, SCRIPT_V14DOWSN, 127, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node011
 begin
-    gsay_reply(1594, 124);
-    giq_option(3, 1594, 117, Node006, 51);
-    giq_option(3, 1594, 118, Node007, 50);
-    giq_option(3, 1594, 119, Node008, 50);
-    giq_option(3, 1594, 125, Node010, 50);
-    giq_option(3, 1594, 115, Node999, 50);
+    gsay_reply(SCRIPT_V14DOWSN, 124);
+    giq_option(3, SCRIPT_V14DOWSN, 117, Node006, BAD_REACTION);
+    giq_option(3, SCRIPT_V14DOWSN, 118, Node007, NEUTRAL_REACTION);
+    giq_option(3, SCRIPT_V14DOWSN, 119, Node008, NEUTRAL_REACTION);
+    giq_option(3, SCRIPT_V14DOWSN, 125, Node010, NEUTRAL_REACTION);
+    giq_option(3, SCRIPT_V14DOWSN, 115, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node012
 begin
-    gsay_reply(1594, 129);
-    giq_option(3, 1594, 128, Node013, 50);
-    giq_option(3, 1594, 104, Node999, 50);
+    gsay_reply(SCRIPT_V14DOWSN, 129);
+    giq_option(3, SCRIPT_V14DOWSN, 128, Node013, NEUTRAL_REACTION);
+    giq_option(3, SCRIPT_V14DOWSN, 104, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node013
 begin
-    gsay_reply(1594, 124);
-    if (get_critter_stat(dude_obj, 35) < get_critter_stat(dude_obj, 7)) then begin
-        giq_option(3, 1594, 149, Node014, 50);
+    gsay_reply(SCRIPT_V14DOWSN, 124);
+    if (get_critter_stat(dude_obj, STAT_current_hp) < get_critter_stat(dude_obj, STAT_max_hp)) then begin
+        giq_option(3, SCRIPT_V14DOWSN, 149, Node014, NEUTRAL_REACTION);
     end
     if (get_poison(dude_obj) > 1) then begin
-        giq_option(3, 1594, 154, Node024, 50);
+        giq_option(3, SCRIPT_V14DOWSN, 154, Node024, NEUTRAL_REACTION);
     end
-    giq_option(3, 1594, 155, Node025, 50);
+    giq_option(3, SCRIPT_V14DOWSN, 155, Node025, NEUTRAL_REACTION);
     if ((local_var(4) == 0) and ((party_member_obj(16777377) != 0) == 1)) then begin
-        giq_option(3, 1594, 163, Node030, 50);
+        giq_option(3, SCRIPT_V14DOWSN, 163, Node030, NEUTRAL_REACTION);
     end
     if ((obj_is_carrying_obj_pid(dude_obj, 2) > 2) and (obj_is_carrying_obj_pid(dude_obj, 308) > 0) and (local_var(4) == 1) and ((party_member_obj(16777377) != 0) == 1)) then begin
-        giq_option(3, 1594, 171, Node033, 50);
+        giq_option(3, SCRIPT_V14DOWSN, 171, Node033, NEUTRAL_REACTION);
     end
     if (((game_time - local_var(5)) > (7 * (24 * (60 * (60 * 10))))) and (local_var(4) == 2) and ((party_member_obj(16777377) != 0) == 1)) then begin
-        giq_option(3, 1594, 182, Node038, 50);
+        giq_option(3, SCRIPT_V14DOWSN, 182, Node038, NEUTRAL_REACTION);
     end
     if (local_var(2) == 0) then begin
-        giq_option(3, 1594, 135, Node016, 50);
+        giq_option(3, SCRIPT_V14DOWSN, 135, Node016, NEUTRAL_REACTION);
     end
-    giq_option(3, 1594, 104, Node999, 50);
+    giq_option(3, SCRIPT_V14DOWSN, 104, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node014
 begin
-    gsay_reply(1594, 131);
+    gsay_reply(SCRIPT_V14DOWSN, 131);
     gfade_out(600);
     game_time_advance(8 * (60 * (60 * 10)));
     critter_heal(dude_obj, 999);
@@ -527,36 +527,36 @@ end
 
 procedure Node015
 begin
-    gsay_reply(1594, 132);
-    giq_option(3, 1594, 133, Node013, 50);
-    giq_option(3, 1594, 134, Node999, 50);
+    gsay_reply(SCRIPT_V14DOWSN, 132);
+    giq_option(3, SCRIPT_V14DOWSN, 133, Node013, NEUTRAL_REACTION);
+    giq_option(3, SCRIPT_V14DOWSN, 134, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node016
 begin
-    gsay_reply(1594, 136);
-    giq_option(3, 1594, 198, Node022, 50);
-    giq_option(3, 1594, 151, Node023, 50);
+    gsay_reply(SCRIPT_V14DOWSN, 136);
+    giq_option(3, SCRIPT_V14DOWSN, 198, Node022, NEUTRAL_REACTION);
+    giq_option(3, SCRIPT_V14DOWSN, 151, Node023, NEUTRAL_REACTION);
 end
 
 procedure Node017
 begin
-    gsay_reply(1594, 138);
-    giq_option(3, 1594, 139, Node018, 50);
+    gsay_reply(SCRIPT_V14DOWSN, 138);
+    giq_option(3, SCRIPT_V14DOWSN, 139, Node018, NEUTRAL_REACTION);
 end
 
 procedure Node018
 begin
-    gsay_reply(1594, 140);
-    giq_option(3, 1594, 141, Node013, 50);
-    giq_option(3, 1594, 142, Node019, 50);
+    gsay_reply(SCRIPT_V14DOWSN, 140);
+    giq_option(3, SCRIPT_V14DOWSN, 141, Node013, NEUTRAL_REACTION);
+    giq_option(3, SCRIPT_V14DOWSN, 142, Node019, NEUTRAL_REACTION);
 end
 
 procedure Node019
 begin
-    gsay_reply(1594, 143);
-    giq_option(3, 1594, 144, Node020, 50);
-    giq_option(3, 1594, 145, Node013, 50);
+    gsay_reply(SCRIPT_V14DOWSN, 143);
+    giq_option(3, SCRIPT_V14DOWSN, 144, Node020, NEUTRAL_REACTION);
+    giq_option(3, SCRIPT_V14DOWSN, 145, Node013, NEUTRAL_REACTION);
 end
 
 procedure Node020
@@ -565,7 +565,7 @@ begin
     game_ui_disable;
     set_local_var(2, 1);
     gfade_out(600);
-    set_global_var(398, 1);
+    set_global_var(GVAR_PARTY_NO_FOLLOW, 1);
     move_to(dude_obj, 23926, 1);
     critter_attempt_placement(dude_obj, 23926, 1);
     move_to(meris1, 23321, 1);
@@ -583,13 +583,13 @@ procedure Node021
 begin
     meris2 := tile_contains_pid_obj(23321, 1, 16777762);
     if (Complex_Float_Stage == 0) then begin
-        float_msg(self_obj, message_str(1594, 147), 3);
+        float_msg(self_obj, message_str(SCRIPT_V14DOWSN, 147), FLOAT_MSG_GREEN);
         Complex_Float_Stage := Complex_Float_Stage + 1;
         add_timer_event(self_obj, game_ticks(3), 3);
     end
     else begin
         if (Complex_Float_Stage == 1) then begin
-            float_msg(meris2, message_str(1594, 148), 3);
+            float_msg(meris2, message_str(SCRIPT_V14DOWSN, 148), FLOAT_MSG_GREEN);
             Complex_Float_Stage := Complex_Float_Stage + 1;
             add_timer_event(self_obj, game_ticks(3), 3);
         end
@@ -598,13 +598,13 @@ begin
                 gfade_out(600);
                 rm_timer_event(self_obj);
                 Complex_Float_Stage := 0;
-                set_global_var(398, 0);
+                set_global_var(GVAR_PARTY_NO_FOLLOW, 0);
                 game_ui_enable;
                 anim(self_obj, 1000, 3);
                 gfade_in(600);
                 animate_stand_obj(meris2);
                 party_add(meris2);
-                critter_add_trait(meris2, 1, 6, 0);
+                critter_add_trait(meris2, TRAIT_OBJECT, OBJECT_TEAM_NUM, 0);
             end
         end
     end
@@ -612,19 +612,19 @@ end
 
 procedure Node022
 begin
-    gsay_reply(1594, 150);
-    giq_option(3, 1594, 137, Node017, 50);
+    gsay_reply(SCRIPT_V14DOWSN, 150);
+    giq_option(3, SCRIPT_V14DOWSN, 137, Node017, NEUTRAL_REACTION);
 end
 
 procedure Node023
 begin
-    gsay_reply(1594, 152);
-    giq_option(3, 1594, 153, Node017, 50);
+    gsay_reply(SCRIPT_V14DOWSN, 152);
+    giq_option(3, SCRIPT_V14DOWSN, 153, Node017, NEUTRAL_REACTION);
 end
 
 procedure Node024
 begin
-    gsay_reply(1594, 131);
+    gsay_reply(SCRIPT_V14DOWSN, 131);
     gfade_out(600);
     poison(dude_obj, -999);
     gfade_in(600);
@@ -650,52 +650,52 @@ end
 
 procedure Node026
 begin
-    gsay_reply(1594, 162);
-    giq_option(0, 1594, message_str(14, 350), Node999, 50);
+    gsay_reply(SCRIPT_V14DOWSN, 162);
+    giq_option(0, SCRIPT_V14DOWSN, g_mstr(350), Node999, NEUTRAL_REACTION);
 end
 
 procedure Node027
 begin
-    gsay_reply(1594, 156);
-    giq_option(3, 1594, 159, Node999, 50);
+    gsay_reply(SCRIPT_V14DOWSN, 156);
+    giq_option(3, SCRIPT_V14DOWSN, 159, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node028
 begin
-    gsay_reply(1594, 157);
-    giq_option(3, 1594, 158, Node999, 50);
+    gsay_reply(SCRIPT_V14DOWSN, 157);
+    giq_option(3, SCRIPT_V14DOWSN, 158, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node029
 begin
-    add_obj_to_inven(dude_obj, create_object_sid(232, 0, 0, -1));
+    add_obj_to_inven(dude_obj, create_object(232, 0, 0));
     set_local_var(3, 0);
-    gsay_reply(1594, 160);
-    giq_option(3, 1594, 161, Node999, 50);
+    gsay_reply(SCRIPT_V14DOWSN, 160);
+    giq_option(3, SCRIPT_V14DOWSN, 161, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node030
 begin
-    gsay_reply(1594, 164);
-    giq_option(3, 1594, 165, Node031, 50);
-    giq_option(3, 1594, 166, Node999, 50);
+    gsay_reply(SCRIPT_V14DOWSN, 164);
+    giq_option(3, SCRIPT_V14DOWSN, 165, Node031, NEUTRAL_REACTION);
+    giq_option(3, SCRIPT_V14DOWSN, 166, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node031
 begin
-    gsay_reply(1594, 167);
-    giq_option(3, 1594, 168, Node032, 50);
+    gsay_reply(SCRIPT_V14DOWSN, 167);
+    giq_option(3, SCRIPT_V14DOWSN, 168, Node032, NEUTRAL_REACTION);
 end
 
 procedure Node032
 begin
     set_local_var(4, 1);
-    gsay_reply(1594, 169);
-    if (get_critter_stat(dude_obj, 34) == 0) then begin
-        giq_option(3, 1594, 170, Node999, 50);
+    gsay_reply(SCRIPT_V14DOWSN, 169);
+    if (dude_gender == 0) then begin
+        giq_option(3, SCRIPT_V14DOWSN, 170, Node999, NEUTRAL_REACTION);
     end
     else begin
-        giq_option(3, 1594, 202, Node999, 50);
+        giq_option(3, SCRIPT_V14DOWSN, 202, Node999, NEUTRAL_REACTION);
     end
 end
 
@@ -705,21 +705,21 @@ begin
     destroy_object(obj_carrying_pid_obj(dude_obj, 2));
     destroy_object(obj_carrying_pid_obj(dude_obj, 2));
     destroy_object(obj_carrying_pid_obj(dude_obj, 2));
-    gsay_reply(1594, 172);
-    giq_option(3, 1594, 173, Node034, 50);
+    gsay_reply(SCRIPT_V14DOWSN, 172);
+    giq_option(3, SCRIPT_V14DOWSN, 173, Node034, NEUTRAL_REACTION);
 end
 
 procedure Node034
 begin
-    gsay_reply(1594, 174);
-    giq_option(3, 1594, 175, Node035, 50);
+    gsay_reply(SCRIPT_V14DOWSN, 174);
+    giq_option(3, SCRIPT_V14DOWSN, 175, Node035, NEUTRAL_REACTION);
 end
 
 procedure Node035
 begin
     game_ui_disable;
     gfade_out(600);
-    set_global_var(398, 1);
+    set_global_var(GVAR_PARTY_NO_FOLLOW, 1);
     move_to(party_member_obj(16777377), 24125, 1);
     critter_attempt_placement(party_member_obj(16777377), 24125, 1);
     anim(party_member_obj(16777377), 1000, rotation_to_tile(tile_num(party_member_obj(16777377)), tile_num(self_obj)));
@@ -734,26 +734,26 @@ end
 procedure Node036
 begin
     if (Complex_Float_Stage == 3) then begin
-        float_msg(self_obj, message_str(1594, 176), 0);
+        float_msg(self_obj, message_str(SCRIPT_V14DOWSN, 176), FLOAT_MSG_NORMAL);
         Complex_Float_Stage := Complex_Float_Stage + 1;
         add_timer_event(self_obj, game_ticks(3), 2);
     end
     else begin
         if (Complex_Float_Stage == 4) then begin
             anim(party_member_obj(16777377), 1000, 2);
-            float_msg(party_member_obj(16777377), message_str(1594, 177), 7);
+            float_msg(party_member_obj(16777377), message_str(SCRIPT_V14DOWSN, 177), FLOAT_MSG_LIGHT_RED);
             Complex_Float_Stage := Complex_Float_Stage + 1;
             add_timer_event(self_obj, game_ticks(3), 2);
         end
         else begin
             if (Complex_Float_Stage == 5) then begin
-                float_msg(self_obj, message_str(1594, 178), 0);
+                float_msg(self_obj, message_str(SCRIPT_V14DOWSN, 178), FLOAT_MSG_NORMAL);
                 Complex_Float_Stage := Complex_Float_Stage + 1;
                 add_timer_event(self_obj, game_ticks(3), 2);
             end
             else begin
                 if (Complex_Float_Stage == 6) then begin
-                    float_msg(self_obj, message_str(1594, 179), 0);
+                    float_msg(self_obj, message_str(SCRIPT_V14DOWSN, 179), FLOAT_MSG_NORMAL);
                     Complex_Float_Stage := Complex_Float_Stage + 1;
                     add_timer_event(self_obj, game_ticks(3), 2);
                 end
@@ -762,7 +762,7 @@ begin
                         gfade_out(600);
                         rm_timer_event(self_obj);
                         Complex_Float_Stage := 0;
-                        set_global_var(398, 0);
+                        set_global_var(GVAR_PARTY_NO_FOLLOW, 0);
                         anim(self_obj, 1000, 3);
                         anim(party_member_obj(16777377), 1000, 3);
                         gfade_in(600);
@@ -783,36 +783,36 @@ procedure Node037
 begin
     set_local_var(4, 2);
     set_local_var(5, game_time);
-    gsay_reply(1594, 180);
-    giq_option(3, 1594, 181, Node999, 50);
+    gsay_reply(SCRIPT_V14DOWSN, 180);
+    giq_option(3, SCRIPT_V14DOWSN, 181, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node038
 begin
     set_local_var(4, 3);
     set_local_var(5, 0);
-    gsay_reply(1594, 183);
-    giq_option(3, 1594, 184, Node039, 50);
+    gsay_reply(SCRIPT_V14DOWSN, 183);
+    giq_option(3, SCRIPT_V14DOWSN, 184, Node039, NEUTRAL_REACTION);
 end
 
 procedure Node039
 begin
-    gsay_reply(1594, 185);
-    giq_option(3, 1594, 186, Node040, 50);
+    gsay_reply(SCRIPT_V14DOWSN, 185);
+    giq_option(3, SCRIPT_V14DOWSN, 186, Node040, NEUTRAL_REACTION);
 end
 
 procedure Node040
 begin
     gfade_out(600);
     gfade_in(600);
-    gsay_reply(1594, 187);
-    giq_option(3, 1594, 188, Node041, 50);
+    gsay_reply(SCRIPT_V14DOWSN, 187);
+    giq_option(3, SCRIPT_V14DOWSN, 188, Node041, NEUTRAL_REACTION);
 end
 
 procedure Node041
 begin
     game_ui_disable;
-    set_global_var(398, 1);
+    set_global_var(GVAR_PARTY_NO_FOLLOW, 1);
     move_to(party_member_obj(16777377), 24325, 1);
     critter_attempt_placement(party_member_obj(16777377), 24325, 1);
     anim(party_member_obj(16777377), 1000, rotation_to_tile(tile_num(party_member_obj(16777377)), tile_num(self_obj)));
@@ -821,7 +821,7 @@ begin
     critter_attempt_placement(dude_obj, 24329, 1);
     reg_anim_func(2, dude_obj);
     anim(dude_obj, 1000, 0);
-    create_object_sid(232, 23924, 1, -1);
+    create_object(232, 23924, 1);
     Complex_Float_Stage := 8;
     add_timer_event(self_obj, game_ticks(3), 4);
 end
@@ -829,25 +829,25 @@ end
 procedure Node042
 begin
     if (Complex_Float_Stage == 8) then begin
-        float_msg(self_obj, message_str(1594, 189), 0);
+        float_msg(self_obj, message_str(SCRIPT_V14DOWSN, 189), FLOAT_MSG_NORMAL);
         Complex_Float_Stage := Complex_Float_Stage + 1;
         add_timer_event(self_obj, game_ticks(3), 4);
     end
     else begin
         if (Complex_Float_Stage == 9) then begin
-            float_msg(party_member_obj(16777377), message_str(1594, 190), 0);
+            float_msg(party_member_obj(16777377), message_str(SCRIPT_V14DOWSN, 190), FLOAT_MSG_NORMAL);
             Complex_Float_Stage := Complex_Float_Stage + 1;
             add_timer_event(self_obj, game_ticks(4), 4);
         end
         else begin
             if (Complex_Float_Stage == 10) then begin
-                float_msg(dude_obj, message_str(1594, 191), 0);
+                float_msg(dude_obj, message_str(SCRIPT_V14DOWSN, 191), FLOAT_MSG_NORMAL);
                 Complex_Float_Stage := Complex_Float_Stage + 1;
                 add_timer_event(self_obj, game_ticks(4), 4);
             end
             else begin
                 if (Complex_Float_Stage == 11) then begin
-                    float_msg(party_member_obj(16777377), message_str(1594, 192), 0);
+                    float_msg(party_member_obj(16777377), message_str(SCRIPT_V14DOWSN, 192), FLOAT_MSG_NORMAL);
                     animate_move_obj_to_tile(party_member_obj(16777377), 24124, 0);
                     Complex_Float_Stage := Complex_Float_Stage + 1;
                     add_timer_event(self_obj, game_ticks(4), 4);
@@ -857,31 +857,31 @@ begin
                         gfade_out(600);
                         destroy_object(tile_contains_pid_obj(23924, 1, 232));
                         reg_anim_func(2, party_member_obj(16777377));
-                        metarule3(107, party_member_obj(16777377), 16777325, 0);
+                        art_change_fid_num(party_member_obj(16777377), 16777325);
                         anim(party_member_obj(16777377), 1000, 3);
-                        critter_add_trait(party_member_obj(16777377), 0, 75, 1);
-                        critter_add_trait(party_member_obj(16777377), 0, 104, 1);
-                        critter_add_trait(party_member_obj(16777377), 0, 74, 1);
-                        critter_add_trait(party_member_obj(16777377), 0, 76, 1);
-                        critter_add_trait(party_member_obj(16777377), 0, 77, 1);
-                        critter_add_trait(party_member_obj(16777377), 0, 13, 1);
-                        critter_add_trait(party_member_obj(16777377), 0, 29, 2);
-                        critter_add_trait(party_member_obj(16777377), 0, 12, 2);
-                        critter_add_trait(party_member_obj(16777377), 0, 102, 1);
-                        critter_add_trait(party_member_obj(16777377), 0, 117, 1);
-                        critter_add_trait(party_member_obj(16777377), 0, 78, 1);
-                        critter_add_trait(party_member_obj(16777377), 0, 30, 1);
-                        critter_add_trait(party_member_obj(16777377), 0, 11, 2);
-                        critter_add_trait(party_member_obj(16777377), 0, 23, 1);
-                        set_global_var(762, 4);
+                        critter_add_trait(party_member_obj(16777377), TRAIT_PERK, PERK_dermal_enhancement_perk, 1);
+                        critter_add_trait(party_member_obj(16777377), TRAIT_PERK, PERK_stonewall_perk, 1);
+                        critter_add_trait(party_member_obj(16777377), TRAIT_PERK, PERK_dermal_armor_perk, 1);
+                        critter_add_trait(party_member_obj(16777377), TRAIT_PERK, PERK_phoenix_armor_perk, 1);
+                        critter_add_trait(party_member_obj(16777377), TRAIT_PERK, PERK_phoenix_enhancement_perk, 1);
+                        critter_add_trait(party_member_obj(16777377), TRAIT_PERK, PERK_strong_back, 1);
+                        critter_add_trait(party_member_obj(16777377), TRAIT_PERK, PERK_dodger, 2);
+                        critter_add_trait(party_member_obj(16777377), TRAIT_PERK, PERK_toughness, 2);
+                        critter_add_trait(party_member_obj(16777377), TRAIT_PERK, PERK_quick_recovery_perk, 1);
+                        critter_add_trait(party_member_obj(16777377), TRAIT_PERK, PERK_weapon_knockout_perk, 1);
+                        critter_add_trait(party_member_obj(16777377), TRAIT_PERK, PERK_vault_city_inoculations_perk, 1);
+                        critter_add_trait(party_member_obj(16777377), TRAIT_PERK, PERK_snakeater, 1);
+                        critter_add_trait(party_member_obj(16777377), TRAIT_PERK, PERK_rad_resistance, 2);
+                        critter_add_trait(party_member_obj(16777377), TRAIT_PERK, PERK_slayer, 1);
+                        set_global_var(GVAR_KAGA_KILLED, 4);
                         gfade_in(600);
-                        float_msg(party_member_obj(16777377), message_str(1594, 193), 0);
+                        float_msg(party_member_obj(16777377), message_str(SCRIPT_V14DOWSN, 193), FLOAT_MSG_NORMAL);
                         Complex_Float_Stage := Complex_Float_Stage + 1;
                         add_timer_event(self_obj, game_ticks(4), 4);
                     end
                     else begin
                         if (Complex_Float_Stage == 13) then begin
-                            float_msg(self_obj, message_str(1594, 194), 0);
+                            float_msg(self_obj, message_str(SCRIPT_V14DOWSN, 194), FLOAT_MSG_NORMAL);
                             Complex_Float_Stage := Complex_Float_Stage + 1;
                             add_timer_event(self_obj, game_ticks(4), 4);
                         end
@@ -890,7 +890,7 @@ begin
                                 gfade_out(600);
                                 rm_timer_event(self_obj);
                                 Complex_Float_Stage := 0;
-                                set_global_var(398, 0);
+                                set_global_var(GVAR_PARTY_NO_FOLLOW, 0);
                                 anim(self_obj, 1000, 3);
                                 gfade_in(600);
                                 animate_move_obj_to_tile(party_member_obj(16777377), 24326, 0);
@@ -913,8 +913,7 @@ end
 procedure Node043
 begin
     give_exp_points(1000);
-    display_msg(message_str(1594, 197));
-    gsay_reply(1594, 195);
-    giq_option(3, 1594, 196, Node999, 50);
+    display_msg(message_str(SCRIPT_V14DOWSN, 197));
+    gsay_reply(SCRIPT_V14DOWSN, 195);
+    giq_option(3, SCRIPT_V14DOWSN, 196, Node999, NEUTRAL_REACTION);
 end
-
