diff --git a/Mods/MegaMod/Original/V14ENT.ssl b/Mods/MegaMod/ReDefine/V14ENT.ssl
index 50aebeb..ee68c7e 100644
--- a/Mods/MegaMod/Original/V14ENT.ssl
+++ b/Mods/MegaMod/ReDefine/V14ENT.ssl
@@ -219,25 +219,24 @@ begin
             end
         end
     end
-    set_global_var(27, 0);
-    if (metarule(14, 0) == 1) then begin
-        display_msg(message_str(1597, 100));
+    set_global_var(GVAR_LOAD_MAP_INDEX, 0);
+    if (map_first_run == 1) then begin
+        display_mstr(100);
     end
     else begin
-        if ((global_var(762) == 5) and (metarule(22, 0) == 0)) then begin
-            set_global_var(762, 6);
+        if ((global_var(GVAR_KAGA_KILLED) == 5) and (is_loading_game == 0)) then begin
+            set_global_var(GVAR_KAGA_KILLED, 6);
             game_ui_disable;
             explosion(16274, 0, 70);
             gfade_out(600);
             destroy_object(tile_contains_pid_obj(16474, 0, 33554902));
-            create_object_sid(33554513, 16274, 0, -1);
-            create_object_sid(33554499, 16673, 0, -1);
-            create_object_sid(33554499, 16674, 0, -1);
-            create_object_sid(33554499, 16675, 0, -1);
+            create_object(33554513, 16274, 0);
+            create_object(33554499, 16673, 0);
+            create_object(33554499, 16674, 0);
+            create_object(33554499, 16675, 0);
             gfade_in(600);
             game_ui_enable;
-            display_msg(message_str(1509, 101));
+            display_msg(message_str(SCRIPT_TRIBEM1, 101));
         end
     end
 end
-
