diff --git a/Mods/MegaMod/Original/V14GEN.ssl b/Mods/MegaMod/ReDefine/V14GEN.ssl
index a54ffb4..f194cb8 100644
--- a/Mods/MegaMod/Original/V14GEN.ssl
+++ b/Mods/MegaMod/ReDefine/V14GEN.ssl
@@ -149,32 +149,32 @@ begin
     if (LVar0 == 13) then begin
         if (not(map_var(4) == 2)) then begin
             script_overrides;
-            if (has_skill(source_obj, 13) < 110) then begin
-                display_msg(message_str(1585, 219));
+            if (has_skill(source_obj, SKILL_REPAIR) < 110) then begin
+                display_msg(message_str(SCRIPT_V14SLDR1, 219));
             end
             else begin
                 if (is_success(roll_vs_skill(source_obj, 13, -30))) then begin
-                    display_msg(message_str(1585, 223));
+                    display_msg(message_str(SCRIPT_V14SLDR1, 223));
                     set_light_level(50);
                     set_map_var(4, 2);
                     if (local_var(0) == 0) then begin
-                        display_msg(message_str(1585, 225));
+                        display_msg(message_str(SCRIPT_V14SLDR1, 225));
                         give_exp_points(1500);
-                        display_msg(message_str(14, 100) + 1500 + message_str(14, 101));
+                        display_msg(g_mstr(100) + 1500 + g_mstr(101));
                         set_local_var(0, 1);
                     end
                 end
                 else begin
                     set_local_var(1, local_var(1) + 1);
                     if (local_var(1) == 1) then begin
-                        display_msg(message_str(1585, 220));
+                        display_msg(message_str(SCRIPT_V14SLDR1, 220));
                     end
                     else begin
                         if (local_var(1) == 2) then begin
-                            display_msg(message_str(1585, 221));
+                            display_msg(message_str(SCRIPT_V14SLDR1, 221));
                         end
                         else begin
-                            display_msg(message_str(1585, 222));
+                            display_msg(message_str(SCRIPT_V14SLDR1, 222));
                         end
                     end
                 end
@@ -189,32 +189,32 @@ procedure use_obj_on_p_proc
 begin
     if (not(map_var(4) == 2) and (obj_pid(obj_being_used_with) == 75)) then begin
         script_overrides;
-        if (has_skill(source_obj, 13) < 110) then begin
-            display_msg(message_str(1585, 219));
+        if (has_skill(source_obj, SKILL_REPAIR) < 110) then begin
+            display_msg(message_str(SCRIPT_V14SLDR1, 219));
         end
         else begin
             if (is_success(roll_vs_skill(source_obj, 13, -30 + 20))) then begin
-                display_msg(message_str(1585, 223));
+                display_msg(message_str(SCRIPT_V14SLDR1, 223));
                 set_light_level(50);
                 set_map_var(4, 2);
                 if (local_var(0) == 0) then begin
-                    display_msg(message_str(1585, 225));
+                    display_msg(message_str(SCRIPT_V14SLDR1, 225));
                     give_exp_points(1500);
-                    display_msg(message_str(14, 100) + 1500 + message_str(14, 101));
+                    display_msg(g_mstr(100) + 1500 + g_mstr(101));
                     set_local_var(0, 1);
                 end
             end
             else begin
                 set_local_var(1, local_var(1) + 1);
                 if (local_var(1) == 1) then begin
-                    display_msg(message_str(1585, 220));
+                    display_msg(message_str(SCRIPT_V14SLDR1, 220));
                 end
                 else begin
                     if (local_var(1) == 2) then begin
-                        display_msg(message_str(1585, 221));
+                        display_msg(message_str(SCRIPT_V14SLDR1, 221));
                     end
                     else begin
-                        display_msg(message_str(1585, 222));
+                        display_msg(message_str(SCRIPT_V14SLDR1, 222));
                     end
                 end
             end
@@ -226,10 +226,10 @@ procedure look_at_p_proc
 begin
     script_overrides;
     if (map_var(4) == 2) then begin
-        display_msg(message_str(1585, 217));
+        display_msg(message_str(SCRIPT_V14SLDR1, 217));
     end
     else begin
-        display_msg(message_str(1585, 215));
+        display_msg(message_str(SCRIPT_V14SLDR1, 215));
     end
 end
 
@@ -237,10 +237,9 @@ procedure description_p_proc
 begin
     script_overrides;
     if (map_var(4) == 2) then begin
-        display_msg(message_str(1585, 218));
+        display_msg(message_str(SCRIPT_V14SLDR1, 218));
     end
     else begin
-        display_msg(message_str(1585, 216));
+        display_msg(message_str(SCRIPT_V14SLDR1, 216));
     end
 end
-
