diff --git a/Mods/MegaMod/Original/V14LKR.ssl b/Mods/MegaMod/ReDefine/V14LKR.ssl
index 78a9051..b124350 100644
--- a/Mods/MegaMod/Original/V14LKR.ssl
+++ b/Mods/MegaMod/ReDefine/V14LKR.ssl
@@ -147,7 +147,7 @@ procedure use_p_proc
 begin
     if ((map_var(3) == 0) or (map_var(3) == 1)) then begin
         script_overrides;
-        display_msg(message_str(1585, 210));
+        display_msg(message_str(SCRIPT_V14SLDR1, 210));
         call signal_critter();
     end
 end
@@ -156,7 +156,7 @@ procedure use_skill_on_p_proc
 begin
     if ((map_var(3) == 0) or (map_var(3) == 1)) then begin
         script_overrides;
-        display_msg(message_str(1585, 210));
+        display_msg(message_str(SCRIPT_V14SLDR1, 210));
         call signal_critter();
     end
 end
@@ -165,7 +165,7 @@ procedure use_obj_on_p_proc
 begin
     if ((map_var(3) == 0) or (map_var(3) == 1)) then begin
         script_overrides;
-        display_msg(message_str(1585, 210));
+        display_msg(message_str(SCRIPT_V14SLDR1, 210));
         call signal_critter();
     end
 end
@@ -180,8 +180,7 @@ procedure pickup_p_proc
 begin
     if ((map_var(3) == 0) or (map_var(3) == 1)) then begin
         script_overrides;
-        display_msg(message_str(1585, 210));
+        display_msg(message_str(SCRIPT_V14SLDR1, 210));
         call signal_critter();
     end
 end
-
