diff --git a/Mods/MegaMod/Original/V14LBMN2.ssl b/Mods/MegaMod/ReDefine/V14LBMN2.ssl
index 7e149ad..1d349ae 100644
--- a/Mods/MegaMod/Original/V14LBMN2.ssl
+++ b/Mods/MegaMod/ReDefine/V14LBMN2.ssl
@@ -149,27 +149,26 @@ end
 
 procedure talk_p_proc
 begin
-    if (global_var(762) != 2) then begin
-        float_msg(self_obj, message_str(1593, random(125, 126)), 8);
+    if (global_var(GVAR_KAGA_KILLED) != 2) then begin
+        float_msg(self_obj, mstr(random(125, 126)), FLOAT_MSG_YELLOW);
     end
     else begin
-        float_msg(self_obj, message_str(1593, random(148, 158)), 8);
+        float_msg(self_obj, mstr(random(148, 158)), FLOAT_MSG_YELLOW);
     end
 end
 
 procedure look_at_p_proc
 begin
     script_overrides;
-    display_msg(message_str(1593, 102));
+    display_mstr(102);
 end
 
 procedure description_p_proc
 begin
     script_overrides;
-    display_msg(message_str(1593, 101));
+    display_mstr(101);
 end
 
 procedure Node999
 begin
 end
-
