diff --git a/Mods/MegaMod/Original/V14TRAP.ssl b/Mods/MegaMod/ReDefine/V14TRAP.ssl
index c209bc0..3d51114 100644
--- a/Mods/MegaMod/Original/V14TRAP.ssl
+++ b/Mods/MegaMod/ReDefine/V14TRAP.ssl
@@ -151,12 +151,12 @@ end
 
 procedure critter_p_proc
 begin
-    if ((local_var(1) == 0) and (tile_distance_objs(self_obj, dude_obj) <= 4) and (has_trait(1, self_obj, 666) == 1)) then begin
+    if ((local_var(1) == 0) and (tile_distance_objs(self_obj, dude_obj) <= 4) and (obj_is_visible_flag(self_obj) == 1)) then begin
         call talk_p_proc();
     end
     else begin
-        if ((local_var(1) == 2) and (tile_distance_objs(self_obj, dude_obj) <= 4) and (has_trait(1, self_obj, 666) == 1)) then begin
-            attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+        if ((local_var(1) == 2) and (tile_distance_objs(self_obj, dude_obj) <= 4) and (obj_is_visible_flag(self_obj) == 1)) then begin
+            attack(dude_obj);
         end
     end
 end
@@ -175,17 +175,17 @@ procedure look_at_p_proc
 begin
     script_overrides;
     if (local_var(0) == 0) then begin
-        display_msg(message_str(1585, 360));
+        display_msg(message_str(SCRIPT_V14SLDR1, 360));
     end
     else begin
-        display_msg(message_str(1585, 361));
+        display_msg(message_str(SCRIPT_V14SLDR1, 361));
     end
 end
 
 procedure description_p_proc
 begin
     script_overrides;
-    display_msg(message_str(1585, 362));
+    display_msg(message_str(SCRIPT_V14SLDR1, 362));
 end
 
 procedure Node998
@@ -199,80 +199,80 @@ end
 
 procedure Node001
 begin
-    gsay_reply(1585, 363);
-    giq_option(4, 1585, 364, Node002, 50);
-    giq_option(4, 1585, 367, Node998, 51);
-    giq_option(-3, 1585, 365, Node999, 51);
+    gsay_reply(SCRIPT_V14SLDR1, 363);
+    giq_option(4, SCRIPT_V14SLDR1, 364, Node002, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_V14SLDR1, 367, Node998, BAD_REACTION);
+    giq_option(-3, SCRIPT_V14SLDR1, 365, Node999, BAD_REACTION);
 end
 
 procedure Node002
 begin
-    gsay_reply(1585, 366);
-    giq_option(4, 1585, 369, Node003, 50);
-    giq_option(4, 1585, 368, Node998, 51);
+    gsay_reply(SCRIPT_V14SLDR1, 366);
+    giq_option(4, SCRIPT_V14SLDR1, 369, Node003, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_V14SLDR1, 368, Node998, BAD_REACTION);
 end
 
 procedure Node003
 begin
-    gsay_reply(1585, 370);
-    giq_option(4, 1585, 371, Node004, 50);
-    giq_option(4, 1585, 372, Node998, 51);
+    gsay_reply(SCRIPT_V14SLDR1, 370);
+    giq_option(4, SCRIPT_V14SLDR1, 371, Node004, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_V14SLDR1, 372, Node998, BAD_REACTION);
 end
 
 procedure Node004
 begin
-    gsay_reply(1585, 373);
-    giq_option(4, 1585, 374, Node005, 50);
+    gsay_reply(SCRIPT_V14SLDR1, 373);
+    giq_option(4, SCRIPT_V14SLDR1, 374, Node005, NEUTRAL_REACTION);
 end
 
 procedure Node005
 begin
     gfade_out(600);
     set_obj_visibility(tile_contains_pid_obj(19316, 1, 16777430), 1);
-    set_global_var(0, global_var(0) + 5);
+    set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + 5);
     debug_msg("Player gains " + 5 + " Karma Points.");
-    set_global_var(37, 0);
-    set_global_var(38, 0);
-    set_global_var(39, 0);
-    set_global_var(40, 0);
-    set_global_var(41, 0);
-    set_global_var(42, 0);
-    set_global_var(43, 0);
-    set_global_var(44, 0);
-    set_global_var(45, 0);
-    if (global_var(0) >= 1000) then begin
-        set_global_var(37, 1);
+    set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+    set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+    set_global_var(GVAR_KARMA_DEFENDER, 0);
+    set_global_var(GVAR_KARMA_WANDERER, 0);
+    set_global_var(GVAR_KARMA_BETRAYER, 0);
+    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+    set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+    set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+    if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
     end
     else begin
-        if (global_var(0) >= 750) then begin
-            set_global_var(38, 1);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+            set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
         end
         else begin
-            if (global_var(0) >= 500) then begin
-                set_global_var(39, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
             end
             else begin
-                if (global_var(0) >= 250) then begin
-                    set_global_var(40, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                    set_global_var(GVAR_KARMA_DEFENDER, 1);
                 end
                 else begin
-                    if (global_var(0) > -250) then begin
-                        set_global_var(41, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                        set_global_var(GVAR_KARMA_WANDERER, 1);
                     end
                     else begin
-                        if (global_var(0) > -500) then begin
-                            set_global_var(42, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                            set_global_var(GVAR_KARMA_BETRAYER, 1);
                         end
                         else begin
-                            if (global_var(0) > -750) then begin
-                                set_global_var(43, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                             end
                             else begin
-                                if (global_var(0) > -1000) then begin
-                                    set_global_var(44, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                    set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                 end
                                 else begin
-                                    set_global_var(45, 1);
+                                    set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                 end
                             end
                         end
@@ -282,7 +282,6 @@ begin
         end
     end
     give_exp_points(500);
-    display_msg(message_str(1585, 375));
+    display_msg(message_str(SCRIPT_V14SLDR1, 375));
     gfade_in(600);
 end
-
