diff --git a/Mods/MegaMod/Original/V23BOSS.ssl b/Mods/MegaMod/ReDefine/V23BOSS.ssl
index 015a2f3..24eb874 100644
--- a/Mods/MegaMod/Original/V23BOSS.ssl
+++ b/Mods/MegaMod/ReDefine/V23BOSS.ssl
@@ -16,7 +16,7 @@ procedure start
 begin
     variable LVar0 := 0;
     if (local_var(6) == 0) then begin
-        LVar0 := create_object_sid(675, 0, 0, -1);
+        LVar0 := create_object(675, 0, 0);
         add_mult_objs_to_inven(self_obj, LVar0, 1);
         wield_obj_critter(self_obj, obj_carrying_pid_obj(self_obj, 675));
         set_local_var(6, 1);
@@ -28,7 +28,7 @@ end
 procedure critter_p_proc
 begin
     if (obj_can_see_obj(self_obj, dude_obj)) then begin
-        attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+        attack(dude_obj);
     end
 end
 
@@ -50,7 +50,7 @@ end
 
 procedure destroy_p_proc
 begin
-    set_global_var(767, 2);
+    set_global_var(GVAR_PARTY_NPCS_HIDDEN, 2);
     destroy_object(critter_inven_obj(self_obj, 1));
     destroy_object(critter_inven_obj(self_obj, 2));
     destroy_object(obj_carrying_pid_obj(self_obj, 675));
@@ -68,4 +68,3 @@ end
 procedure combat_p_proc
 begin
 end
-
