diff --git a/Mods/MegaMod/Original/V23COMP.ssl b/Mods/MegaMod/ReDefine/V23COMP.ssl
index 4149afc..0b53372 100644
--- a/Mods/MegaMod/Original/V23COMP.ssl
+++ b/Mods/MegaMod/ReDefine/V23COMP.ssl
@@ -13,13 +13,13 @@ procedure ToggleElevator;
 procedure description_p_proc
 begin
     script_overrides;
-    display_msg(message_str(964, 11201));
+    display_msg(message_str(SCRIPT_KLATRAP, 11201));
 end
 
 procedure look_at_proc
 begin
     script_overrides;
-    display_msg(message_str(964, 11200));
+    display_msg(message_str(SCRIPT_KLATRAP, 11200));
 end
 
 procedure start_p_proc
@@ -39,40 +39,40 @@ end
 
 procedure Node000
 begin
-    gsay_reply(964, 11202);
-    gsay_option(964, 11209, Node001, 50);
-    gsay_option(964, 11210, Node002, 50);
-    gsay_option(964, 11208, Node999, 50);
+    gsay_reply(SCRIPT_KLATRAP, 11202);
+    giq_option(1, SCRIPT_KLATRAP, 11209, Node001, NEUTRAL_REACTION);
+    giq_option(1, SCRIPT_KLATRAP, 11210, Node002, NEUTRAL_REACTION);
+    giq_option(1, SCRIPT_KLATRAP, 11208, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node001
 begin
     if (map_var(0) == 0) then begin
-        gsay_reply(964, 11203);
-        gsay_option(964, 11211, ToggleElevator, 50);
+        gsay_reply(SCRIPT_KLATRAP, 11203);
+        giq_option(1, SCRIPT_KLATRAP, 11211, ToggleElevator, NEUTRAL_REACTION);
     end
     else begin
         if (map_var(0) == 1) then begin
-            gsay_reply(964, 11204);
-            gsay_option(964, 11212, ToggleElevator, 50);
+            gsay_reply(SCRIPT_KLATRAP, 11204);
+            giq_option(1, SCRIPT_KLATRAP, 11212, ToggleElevator, NEUTRAL_REACTION);
         end
     end
-    gsay_option(964, 11207, Node000, 50);
+    giq_option(1, SCRIPT_KLATRAP, 11207, Node000, NEUTRAL_REACTION);
 end
 
 procedure Node002
 begin
     if (map_var(1) == 0) then begin
-        gsay_reply(964, 11205);
-        gsay_option(964, 11211, ToggleLights, 50);
+        gsay_reply(SCRIPT_KLATRAP, 11205);
+        giq_option(1, SCRIPT_KLATRAP, 11211, ToggleLights, NEUTRAL_REACTION);
     end
     else begin
         if (map_var(1) == 1) then begin
-            gsay_reply(964, 11206);
-            gsay_option(964, 11212, ToggleLights, 50);
+            gsay_reply(SCRIPT_KLATRAP, 11206);
+            giq_option(1, SCRIPT_KLATRAP, 11212, ToggleLights, NEUTRAL_REACTION);
         end
     end
-    gsay_option(964, 11207, Node000, 50);
+    giq_option(1, SCRIPT_KLATRAP, 11207, Node000, NEUTRAL_REACTION);
 end
 
 procedure Node999
@@ -104,4 +104,3 @@ begin
     end
     call Node001();
 end
-
