diff --git a/Mods/MegaMod/Original/V23TNL.ssl b/Mods/MegaMod/ReDefine/V23TNL.ssl
index 4eba097..a0507be 100644
--- a/Mods/MegaMod/Original/V23TNL.ssl
+++ b/Mods/MegaMod/ReDefine/V23TNL.ssl
@@ -11,10 +11,10 @@ procedure use_p_proc
 begin
     if (map_var(0) == 0) then begin
         script_overrides;
-        display_msg(message_str(1748, 10));
+        display_mstr(10);
     end
     if (map_var(0) == 1) then begin
-        display_msg(message_str(1748, 11));
+        display_mstr(11);
     end
 end
 
@@ -23,19 +23,18 @@ begin
     variable LVar0 := 0;
     LVar0 := obj_pid(obj_being_used_with);
     if ((LVar0 == 127) and (map_var(0) == 0)) then begin
-        display_msg(message_str(1748, 20));
+        display_mstr(20);
         set_map_var(0, 1);
         rm_obj_from_inven(dude_obj, obj_carrying_pid_obj(dude_obj, 127));
     end
     else begin
         if (map_var(0) == 1) then begin
-            display_msg(message_str(1748, 12));
+            display_mstr(12);
         end
         else begin
-            display_msg(message_str(1748, 10));
+            display_mstr(10);
         end
     end
     return 0;
     LVar0 := 0;
 end
-
