diff --git a/Mods/MegaMod/Original/V5HAROLD.ssl b/Mods/MegaMod/ReDefine/V5HAROLD.ssl
index f9a348a..dda6879 100644
--- a/Mods/MegaMod/Original/V5HAROLD.ssl
+++ b/Mods/MegaMod/ReDefine/V5HAROLD.ssl
@@ -158,13 +158,13 @@ end
 procedure look_at_p_proc
 begin
     script_overrides;
-    display_msg(message_str(1328, 50));
+    display_msg(message_str(SCRIPT_KCCVRAT, 50));
 end
 
 procedure description_p_proc
 begin
     script_overrides;
-    display_msg(message_str(1328, 60));
+    display_msg(message_str(SCRIPT_KCCVRAT, 60));
 end
 
 procedure talk_p_proc
@@ -192,75 +192,75 @@ end
 
 procedure Node001
 begin
-    gsay_reply(1328, 100);
-    giq_option(2, 1328, 110, Node007, 50);
-    giq_option(2, 1328, 120, Node005, 50);
-    giq_option(2, 1328, 130, Node006, 50);
-    giq_option(2, 1328, 140, Node002, 50);
-    giq_option(2, 1328, 150, Node003, 50);
+    gsay_reply(SCRIPT_KCCVRAT, 100);
+    giq_option(2, SCRIPT_KCCVRAT, 110, Node007, NEUTRAL_REACTION);
+    giq_option(2, SCRIPT_KCCVRAT, 120, Node005, NEUTRAL_REACTION);
+    giq_option(2, SCRIPT_KCCVRAT, 130, Node006, NEUTRAL_REACTION);
+    giq_option(2, SCRIPT_KCCVRAT, 140, Node002, NEUTRAL_REACTION);
+    giq_option(2, SCRIPT_KCCVRAT, 150, Node003, NEUTRAL_REACTION);
 end
 
 procedure Node002
 begin
-    gsay_reply(1328, 200);
-    giq_option(2, 1328, 150, Node003, 50);
+    gsay_reply(SCRIPT_KCCVRAT, 200);
+    giq_option(2, SCRIPT_KCCVRAT, 150, Node003, NEUTRAL_REACTION);
 end
 
 procedure Node003
 begin
-    gsay_reply(1328, 230);
-    giq_option(2, 1328, 210, Node004, 50);
+    gsay_reply(SCRIPT_KCCVRAT, 230);
+    giq_option(2, SCRIPT_KCCVRAT, 210, Node004, NEUTRAL_REACTION);
 end
 
 procedure Node004
 begin
-    gsay_reply(1328, 220);
-    giq_option(2, 1328, 110, Node007, 50);
-    giq_option(2, 1328, 120, Node005, 50);
-    giq_option(2, 1328, 130, Node006, 50);
-    giq_option(2, 1328, 140, Node002, 50);
-    giq_option(2, 1328, 150, Node003, 50);
+    gsay_reply(SCRIPT_KCCVRAT, 220);
+    giq_option(2, SCRIPT_KCCVRAT, 110, Node007, NEUTRAL_REACTION);
+    giq_option(2, SCRIPT_KCCVRAT, 120, Node005, NEUTRAL_REACTION);
+    giq_option(2, SCRIPT_KCCVRAT, 130, Node006, NEUTRAL_REACTION);
+    giq_option(2, SCRIPT_KCCVRAT, 140, Node002, NEUTRAL_REACTION);
+    giq_option(2, SCRIPT_KCCVRAT, 150, Node003, NEUTRAL_REACTION);
 end
 
 procedure Node005
 begin
-    gsay_reply(1328, 240);
-    giq_option(2, 1328, 210, Node004, 50);
+    gsay_reply(SCRIPT_KCCVRAT, 240);
+    giq_option(2, SCRIPT_KCCVRAT, 210, Node004, NEUTRAL_REACTION);
 end
 
 procedure Node006
 begin
-    gsay_reply(1328, 250);
-    giq_option(2, 1328, 260, Node999, 50);
+    gsay_reply(SCRIPT_KCCVRAT, 250);
+    giq_option(2, SCRIPT_KCCVRAT, 260, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node007
 begin
-    gsay_reply(1328, 300);
-    giq_option(2, 1328, 310, Node008, 50);
-    giq_option(2, 1328, 320, Node999, 50);
-    giq_option(2, 1328, 330, Node009, 50);
+    gsay_reply(SCRIPT_KCCVRAT, 300);
+    giq_option(2, SCRIPT_KCCVRAT, 310, Node008, NEUTRAL_REACTION);
+    giq_option(2, SCRIPT_KCCVRAT, 320, Node999, NEUTRAL_REACTION);
+    giq_option(2, SCRIPT_KCCVRAT, 330, Node009, NEUTRAL_REACTION);
 end
 
 procedure Node008
 begin
-    gsay_reply(1328, 340);
-    giq_option(2, 1328, 350, Node999, 50);
+    gsay_reply(SCRIPT_KCCVRAT, 340);
+    giq_option(2, SCRIPT_KCCVRAT, 350, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node009
 begin
-    gsay_reply(1328, 400);
+    gsay_reply(SCRIPT_KCCVRAT, 400);
     set_local_var(0, 1);
-    giq_option(2, 1328, 410, Node999, 50);
+    giq_option(2, SCRIPT_KCCVRAT, 410, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node010
 begin
     variable LVar0 := 0;
     if (obj_is_carrying_obj_pid(dude_obj, 586) >= 3) then begin
-        gsay_reply(1328, 500);
-        LVar0 := create_object_sid(587, 0, 0, -1);
+        gsay_reply(SCRIPT_KCCVRAT, 500);
+        LVar0 := create_object(587, 0, 0);
         add_obj_to_inven(dude_obj, LVar0);
         set_local_var(0, 2);
         removed_qty := obj_is_carrying_obj_pid(dude_obj, 586);
@@ -281,7 +281,7 @@ begin
                 end
                 else begin
                     if ((critter_inven_obj(dude_obj, 2) == restock_obj) or (critter_inven_obj(dude_obj, 1) == restock_obj)) then begin
-                        metarule(43, dude_obj);
+                        inven_unwield(dude_obj);
                     end
                 end
             end
@@ -291,62 +291,61 @@ begin
         restock_obj := 0;
         restock_amt := 0;
         give_exp_points(700);
-        display_msg(message_str(1328, 630));
-        giq_option(2, 1328, 520, Node011, 50);
+        display_msg(message_str(SCRIPT_KCCVRAT, 630));
+        giq_option(2, SCRIPT_KCCVRAT, 520, Node011, NEUTRAL_REACTION);
     end
     else begin
-        gsay_reply(1328, 500);
-        giq_option(2, 1328, 510, Node999, 50);
+        gsay_reply(SCRIPT_KCCVRAT, 500);
+        giq_option(2, SCRIPT_KCCVRAT, 510, Node999, NEUTRAL_REACTION);
     end
 end
 
 procedure Node011
 begin
-    gsay_reply(1328, 600);
-    giq_option(2, 1328, 610, Node012, 50);
-    giq_option(2, 1328, 615, Node999, 50);
+    gsay_reply(SCRIPT_KCCVRAT, 600);
+    giq_option(2, SCRIPT_KCCVRAT, 610, Node012, NEUTRAL_REACTION);
+    giq_option(2, SCRIPT_KCCVRAT, 615, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node012
 begin
-    gsay_reply(1328, 700);
-    giq_option(2, 1328, 690, Node013, 50);
+    gsay_reply(SCRIPT_KCCVRAT, 700);
+    giq_option(2, SCRIPT_KCCVRAT, 690, Node013, NEUTRAL_REACTION);
 end
 
 procedure Node013
 begin
-    gsay_reply(1328, 710);
-    giq_option(2, 1328, 690, Node014, 50);
+    gsay_reply(SCRIPT_KCCVRAT, 710);
+    giq_option(2, SCRIPT_KCCVRAT, 690, Node014, NEUTRAL_REACTION);
 end
 
 procedure Node014
 begin
-    gsay_reply(1328, 720);
-    giq_option(2, 1328, 690, Node015, 50);
+    gsay_reply(SCRIPT_KCCVRAT, 720);
+    giq_option(2, SCRIPT_KCCVRAT, 690, Node015, NEUTRAL_REACTION);
 end
 
 procedure Node015
 begin
-    gsay_reply(1328, 730);
-    giq_option(2, 1328, 900, Node999, 50);
+    gsay_reply(SCRIPT_KCCVRAT, 730);
+    giq_option(2, SCRIPT_KCCVRAT, 900, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node016
 begin
-    gsay_reply(1328, 800);
-    set_global_var(695, global_var(695) + 1);
-    giq_option(2, 1328, 810, Node999, 50);
+    gsay_reply(SCRIPT_KCCVRAT, 800);
+    set_global_var(GVAR_FLINT_FLAG, global_var(GVAR_FLINT_FLAG) + 1);
+    giq_option(2, SCRIPT_KCCVRAT, 810, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node017
 begin
-    gsay_reply(1328, 810);
-    giq_option(2, 1328, 690, Node018, 50);
+    gsay_reply(SCRIPT_KCCVRAT, 810);
+    giq_option(2, SCRIPT_KCCVRAT, 690, Node018, NEUTRAL_REACTION);
 end
 
 procedure Node018
 begin
-    gsay_reply(1328, 820);
-    giq_option(2, 1328, 900, Node999, 50);
+    gsay_reply(SCRIPT_KCCVRAT, 820);
+    giq_option(2, SCRIPT_KCCVRAT, 900, Node999, NEUTRAL_REACTION);
 end
-
