diff --git a/Mods/MegaMod/Original/V5INHAB.ssl b/Mods/MegaMod/ReDefine/V5INHAB.ssl
index 1328f77..3d5468f 100644
--- a/Mods/MegaMod/Original/V5INHAB.ssl
+++ b/Mods/MegaMod/ReDefine/V5INHAB.ssl
@@ -139,16 +139,15 @@ end
 procedure look_at_p_proc
 begin
     script_overrides;
-    display_msg(message_str(1332, 50));
+    display_msg(message_str(SCRIPT_KCRTGGRD, 50));
 end
 
 procedure description_p_proc
 begin
-    float_msg(self_obj, message_str(1332, random(60, 65)), 8);
+    float_msg(self_obj, message_str(SCRIPT_KCRTGGRD, random(60, 65)), FLOAT_MSG_YELLOW);
 end
 
 procedure talk_p_proc
 begin
-    float_msg(self_obj, message_str(1332, random(100, 150)), 8);
+    float_msg(self_obj, message_str(SCRIPT_KCRTGGRD, random(100, 150)), FLOAT_MSG_YELLOW);
 end
-
