diff --git a/Mods/MegaMod/Original/VAULT23.ssl b/Mods/MegaMod/ReDefine/VAULT23.ssl
index f09ee6a..14b10ce 100644
--- a/Mods/MegaMod/Original/VAULT23.ssl
+++ b/Mods/MegaMod/ReDefine/VAULT23.ssl
@@ -15,8 +15,8 @@ begin
     if (map_var(1) == 1) then begin
         set_light_level(100);
     end
-    if (global_var(767) < 2) then begin
-        display_msg(message_str(1309, 890));
+    if (global_var(GVAR_PARTY_NPCS_HIDDEN) < 2) then begin
+        display_msg(message_str(SCRIPT_AITEM6, 890));
     end
 end
 
@@ -29,4 +29,3 @@ begin
         set_light_level(100);
     end
 end
-
