diff --git a/Mods/MegaMod/Original/VCDWNBAR.ssl b/Mods/MegaMod/ReDefine/VCDWNBAR.ssl
index d1ef392..5e8f481 100644
--- a/Mods/MegaMod/Original/VCDWNBAR.ssl
+++ b/Mods/MegaMod/ReDefine/VCDWNBAR.ssl
@@ -86,9 +86,9 @@ end
 
 procedure critter_p_proc
 begin
-    if ((local_var(5) == 2) or (global_var(137) == 1) and obj_can_see_obj(self_obj, dude_obj)) then begin
+    if ((local_var(5) == 2) or (global_var(GVAR_ENEMY_VAULT_CITY) == 1) and obj_can_see_obj(self_obj, dude_obj)) then begin
         set_local_var(5, 1);
-        attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+        attack(dude_obj);
     end
 end
 
@@ -104,52 +104,52 @@ begin
     Evil_Critter := 0;
     Slavery_Tolerant := 2;
     Karma_Perception := 10;
-    set_global_var(37, 0);
-    set_global_var(38, 0);
-    set_global_var(39, 0);
-    set_global_var(40, 0);
-    set_global_var(41, 0);
-    set_global_var(42, 0);
-    set_global_var(43, 0);
-    set_global_var(44, 0);
-    set_global_var(45, 0);
-    tmp_gen_rep := global_var(0);
-    if (has_trait(0, dude_obj, 95)) then begin
-        tmp_gen_rep := global_var(0) * 2;
+    set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+    set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+    set_global_var(GVAR_KARMA_DEFENDER, 0);
+    set_global_var(GVAR_KARMA_WANDERER, 0);
+    set_global_var(GVAR_KARMA_BETRAYER, 0);
+    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+    set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+    set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+    tmp_gen_rep := global_var(GVAR_PLAYER_REPUTATION);
+    if (has_trait(TRAIT_PERK, dude_obj, PERK_karma_beacon_perk)) then begin
+        tmp_gen_rep := global_var(GVAR_PLAYER_REPUTATION) * 2;
     end
-    if (global_var(0) >= 1000) then begin
-        set_global_var(37, 1);
+    if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
     end
     else begin
-        if (global_var(0) >= 750) then begin
-            set_global_var(38, 1);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+            set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
         end
         else begin
-            if (global_var(0) >= 500) then begin
-                set_global_var(39, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
             end
             else begin
-                if (global_var(0) >= 250) then begin
-                    set_global_var(40, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                    set_global_var(GVAR_KARMA_DEFENDER, 1);
                 end
                 else begin
-                    if (global_var(0) > -250) then begin
-                        set_global_var(41, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                        set_global_var(GVAR_KARMA_WANDERER, 1);
                     end
                     else begin
-                        if (global_var(0) > -500) then begin
-                            set_global_var(42, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                            set_global_var(GVAR_KARMA_BETRAYER, 1);
                         end
                         else begin
-                            if (global_var(0) > -750) then begin
-                                set_global_var(43, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                             end
                             else begin
-                                if (global_var(0) > -1000) then begin
-                                    set_global_var(44, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                    set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                 end
                                 else begin
-                                    set_global_var(45, 1);
+                                    set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                 end
                             end
                         end
@@ -160,7 +160,7 @@ begin
     end
     if (local_var(2) == 0) then begin
         set_local_var(2, 1);
-        set_local_var(3, (get_critter_stat(dude_obj, 3) - get_critter_stat(self_obj, 3)) * 5);
+        set_local_var(3, (dude_charisma - self_charisma) * 5);
         debug_msg("Base Reaction == " + local_var(3));
     end
     Static_Reaction := 0;
@@ -172,14 +172,14 @@ begin
     else begin
         reaction_bonus_karma := -1 * (-1 * reaction_bonus_karma / 1000);
     end
-    if (global_var(50) >= 0) then begin
-        reaction_bonus_town_rep := global_var(50) / 2;
+    if (global_var(GVAR_TOWN_REP_VAULT_CITY) >= 0) then begin
+        reaction_bonus_town_rep := global_var(GVAR_TOWN_REP_VAULT_CITY) / 2;
     end
     else begin
-        reaction_bonus_town_rep := -1 * (-1 * global_var(50) / 2);
+        reaction_bonus_town_rep := -1 * (-1 * global_var(GVAR_TOWN_REP_VAULT_CITY) / 2);
     end
-    if (has_trait(2, dude_obj, 13)) then begin
-        if (get_critter_stat(dude_obj, 34) == get_critter_stat(self_obj, 34)) then begin
+    if (has_trait(TRAIT_TRAIT, dude_obj, TRAIT_sex_appeal)) then begin
+        if (dude_gender == self_gender) then begin
             Static_Reaction := Static_Reaction + -20;
         end
         else begin
@@ -195,52 +195,52 @@ begin
         debug_msg("Karma Reaction Bonus == " + Static_Reaction);
         Static_Reaction := Static_Reaction - reaction_bonus_town_rep;
         debug_msg("Town Rep Bonus == " + Static_Reaction);
-        set_global_var(37, 0);
-        set_global_var(38, 0);
-        set_global_var(39, 0);
-        set_global_var(40, 0);
-        set_global_var(41, 0);
-        set_global_var(42, 0);
-        set_global_var(43, 0);
-        set_global_var(44, 0);
-        set_global_var(45, 0);
-        tmp_gen_rep := global_var(0);
-        if (has_trait(0, dude_obj, 95)) then begin
-            tmp_gen_rep := global_var(0) * 2;
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+        set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+        set_global_var(GVAR_KARMA_DEFENDER, 0);
+        set_global_var(GVAR_KARMA_WANDERER, 0);
+        set_global_var(GVAR_KARMA_BETRAYER, 0);
+        set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+        set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+        tmp_gen_rep := global_var(GVAR_PLAYER_REPUTATION);
+        if (has_trait(TRAIT_PERK, dude_obj, PERK_karma_beacon_perk)) then begin
+            tmp_gen_rep := global_var(GVAR_PLAYER_REPUTATION) * 2;
         end
-        if (global_var(0) >= 1000) then begin
-            set_global_var(37, 1);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+            set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
         end
         else begin
-            if (global_var(0) >= 750) then begin
-                set_global_var(38, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
             end
             else begin
-                if (global_var(0) >= 500) then begin
-                    set_global_var(39, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                 end
                 else begin
-                    if (global_var(0) >= 250) then begin
-                        set_global_var(40, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                        set_global_var(GVAR_KARMA_DEFENDER, 1);
                     end
                     else begin
-                        if (global_var(0) > -250) then begin
-                            set_global_var(41, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                            set_global_var(GVAR_KARMA_WANDERER, 1);
                         end
                         else begin
-                            if (global_var(0) > -500) then begin
-                                set_global_var(42, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                set_global_var(GVAR_KARMA_BETRAYER, 1);
                             end
                             else begin
-                                if (global_var(0) > -750) then begin
-                                    set_global_var(43, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -1000) then begin
-                                        set_global_var(44, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                     end
                                     else begin
-                                        set_global_var(45, 1);
+                                        set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                     end
                                 end
                             end
@@ -249,35 +249,35 @@ begin
                 end
             end
         end
-        if (global_var(37) == 1) then begin
+        if (global_var(GVAR_KARMA_HOLY_WARRIOR) == 1) then begin
             Static_Reaction := Static_Reaction - 20;
         end
         else begin
-            if (global_var(38) == 1) then begin
+            if (global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES) == 1) then begin
                 Static_Reaction := Static_Reaction - 15;
             end
             else begin
-                if (global_var(39) == 1) then begin
+                if (global_var(GVAR_KARMA_SHIELD_OF_HOPE) == 1) then begin
                     Static_Reaction := Static_Reaction - 10;
                 end
                 else begin
-                    if (global_var(40) == 1) then begin
+                    if (global_var(GVAR_KARMA_DEFENDER) == 1) then begin
                         Static_Reaction := Static_Reaction - 5;
                     end
                     else begin
-                        if (global_var(41) == 1) then begin
+                        if (global_var(GVAR_KARMA_WANDERER) == 1) then begin
                             Static_Reaction := Static_Reaction - 0;
                         end
                         else begin
-                            if (global_var(42) == 1) then begin
+                            if (global_var(GVAR_KARMA_BETRAYER) == 1) then begin
                                 Static_Reaction := Static_Reaction - -5;
                             end
                             else begin
-                                if (global_var(43) == 1) then begin
+                                if (global_var(GVAR_KARMA_SWORD_OF_DESPAIR) == 1) then begin
                                     Static_Reaction := Static_Reaction - -10;
                                 end
                                 else begin
-                                    if (global_var(44) == 1) then begin
+                                    if (global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE) == 1) then begin
                                         Static_Reaction := Static_Reaction - -15;
                                     end
                                     else begin
@@ -297,52 +297,52 @@ begin
         debug_msg("Karma Reaction Bonus == " + Static_Reaction);
         Static_Reaction := Static_Reaction + reaction_bonus_town_rep;
         debug_msg("Town Rep Bonus == " + Static_Reaction);
-        set_global_var(37, 0);
-        set_global_var(38, 0);
-        set_global_var(39, 0);
-        set_global_var(40, 0);
-        set_global_var(41, 0);
-        set_global_var(42, 0);
-        set_global_var(43, 0);
-        set_global_var(44, 0);
-        set_global_var(45, 0);
-        tmp_gen_rep := global_var(0);
-        if (has_trait(0, dude_obj, 95)) then begin
-            tmp_gen_rep := global_var(0) * 2;
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+        set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+        set_global_var(GVAR_KARMA_DEFENDER, 0);
+        set_global_var(GVAR_KARMA_WANDERER, 0);
+        set_global_var(GVAR_KARMA_BETRAYER, 0);
+        set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+        set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+        tmp_gen_rep := global_var(GVAR_PLAYER_REPUTATION);
+        if (has_trait(TRAIT_PERK, dude_obj, PERK_karma_beacon_perk)) then begin
+            tmp_gen_rep := global_var(GVAR_PLAYER_REPUTATION) * 2;
         end
-        if (global_var(0) >= 1000) then begin
-            set_global_var(37, 1);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+            set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
         end
         else begin
-            if (global_var(0) >= 750) then begin
-                set_global_var(38, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
             end
             else begin
-                if (global_var(0) >= 500) then begin
-                    set_global_var(39, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                 end
                 else begin
-                    if (global_var(0) >= 250) then begin
-                        set_global_var(40, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                        set_global_var(GVAR_KARMA_DEFENDER, 1);
                     end
                     else begin
-                        if (global_var(0) > -250) then begin
-                            set_global_var(41, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                            set_global_var(GVAR_KARMA_WANDERER, 1);
                         end
                         else begin
-                            if (global_var(0) > -500) then begin
-                                set_global_var(42, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                set_global_var(GVAR_KARMA_BETRAYER, 1);
                             end
                             else begin
-                                if (global_var(0) > -750) then begin
-                                    set_global_var(43, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -1000) then begin
-                                        set_global_var(44, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                     end
                                     else begin
-                                        set_global_var(45, 1);
+                                        set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                     end
                                 end
                             end
@@ -351,35 +351,35 @@ begin
                 end
             end
         end
-        if (global_var(37) == 1) then begin
+        if (global_var(GVAR_KARMA_HOLY_WARRIOR) == 1) then begin
             Static_Reaction := Static_Reaction + 20;
         end
         else begin
-            if (global_var(38) == 1) then begin
+            if (global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES) == 1) then begin
                 Static_Reaction := Static_Reaction + 15;
             end
             else begin
-                if (global_var(39) == 1) then begin
+                if (global_var(GVAR_KARMA_SHIELD_OF_HOPE) == 1) then begin
                     Static_Reaction := Static_Reaction + 10;
                 end
                 else begin
-                    if (global_var(40) == 1) then begin
+                    if (global_var(GVAR_KARMA_DEFENDER) == 1) then begin
                         Static_Reaction := Static_Reaction + 5;
                     end
                     else begin
-                        if (global_var(41) == 1) then begin
+                        if (global_var(GVAR_KARMA_WANDERER) == 1) then begin
                             Static_Reaction := Static_Reaction + 0;
                         end
                         else begin
-                            if (global_var(42) == 1) then begin
+                            if (global_var(GVAR_KARMA_BETRAYER) == 1) then begin
                                 Static_Reaction := Static_Reaction + -5;
                             end
                             else begin
-                                if (global_var(43) == 1) then begin
+                                if (global_var(GVAR_KARMA_SWORD_OF_DESPAIR) == 1) then begin
                                     Static_Reaction := Static_Reaction + -10;
                                 end
                                 else begin
-                                    if (global_var(44) == 1) then begin
+                                    if (global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE) == 1) then begin
                                         Static_Reaction := Static_Reaction + -15;
                                     end
                                     else begin
@@ -395,28 +395,28 @@ begin
         debug_msg("Good Person Reaction Bonus == " + Static_Reaction);
     end
     if (Slavery_Tolerant == 0) then begin
-        if (global_var(11) == 1) then begin
+        if (global_var(GVAR_REPUTATION_SLAVER) == 1) then begin
             Static_Reaction := Static_Reaction + (2 * -25);
         end
     end
     else begin
         if (Slavery_Tolerant == 1) then begin
-            if (global_var(11) == 1) then begin
+            if (global_var(GVAR_REPUTATION_SLAVER) == 1) then begin
                 Static_Reaction := Static_Reaction + -25;
             end
         end
         else begin
             if (Slavery_Tolerant == 3) then begin
-                if (global_var(11) == 1) then begin
+                if (global_var(GVAR_REPUTATION_SLAVER) == 1) then begin
                     Static_Reaction := Static_Reaction - -25;
                 end
             end
         end
     end
     debug_msg("Slaver + Aligned Reaction == " + Static_Reaction);
-    Static_Reaction := Static_Reaction + (10 * has_trait(0, dude_obj, 10));
+    Static_Reaction := Static_Reaction + (10 * has_trait(TRAIT_PERK, dude_obj, PERK_presence));
     debug_msg("Presence Reaction == " + Static_Reaction);
-    if (global_var(1) >= 2) then begin
+    if (global_var(GVAR_CHILDKILLER_REPUTATION) >= 2) then begin
         Static_Reaction := Static_Reaction + -30;
     end
     debug_msg("Childkiller Reaction == " + Static_Reaction);
@@ -425,18 +425,18 @@ begin
     From_15 := 0;
     From_33 := 0;
     if ((game_time_hour > 200) and (game_time_hour < 700)) then begin
-        float_msg(self_obj, message_str(965, 258), 8);
+        float_msg(self_obj, mstr(258), FLOAT_MSG_YELLOW);
     end
     else begin
-        if (global_var(494) == 8) then begin
+        if (global_var(GVAR_FALLOUT_2) == 8) then begin
             call Node008();
         end
         else begin
-            if ((party_member_obj(16777377) != 0) and (global_var(79) != 5)) then begin
+            if ((party_member_obj(16777377) != 0) and (global_var(GVAR_VAULT_CITIZEN) != 5)) then begin
                 call Node005();
             end
             else begin
-                if ((party_member_obj(16777323) != 0) and (global_var(79) != 5)) then begin
+                if ((party_member_obj(16777323) != 0) and (global_var(GVAR_VAULT_CITIZEN) != 5)) then begin
                     call Node006();
                 end
                 else begin
@@ -459,63 +459,63 @@ end
 procedure destroy_p_proc
 begin
     if (source_obj == dude_obj) then begin
-        set_global_var(5, global_var(5) + 1);
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(4) > (3 * global_var(5))) or (global_var(2) == 1)) and (global_var(1) == 0)) then begin
-            set_global_var(2, 1);
-            set_global_var(3, 0);
+        set_global_var(GVAR_GOOD_MONSTER, global_var(GVAR_GOOD_MONSTER) + 1);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(4) > (3 * global_var(GVAR_GOOD_MONSTER))) or (global_var(GVAR_CHAMPION_REPUTATION) == 1)) and (global_var(GVAR_CHILDKILLER_REPUTATION) == 0)) then begin
+            set_global_var(GVAR_CHAMPION_REPUTATION, 1);
+            set_global_var(GVAR_BERSERKER_REPUTATION, 0);
         end
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(5) > (2 * global_var(4))) or (global_var(3) == 1))) then begin
-            set_global_var(3, 1);
-            set_global_var(2, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
+            set_global_var(GVAR_BERSERKER_REPUTATION, 1);
+            set_global_var(GVAR_CHAMPION_REPUTATION, 0);
         end
-        set_global_var(0, global_var(0) + -10);
+        set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + -10);
         debug_msg("Player gains " + -10 + " Karma Points.");
-        set_global_var(37, 0);
-        set_global_var(38, 0);
-        set_global_var(39, 0);
-        set_global_var(40, 0);
-        set_global_var(41, 0);
-        set_global_var(42, 0);
-        set_global_var(43, 0);
-        set_global_var(44, 0);
-        set_global_var(45, 0);
-        tmp_gen_rep := global_var(0);
-        if (has_trait(0, dude_obj, 95)) then begin
-            tmp_gen_rep := global_var(0) * 2;
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+        set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+        set_global_var(GVAR_KARMA_DEFENDER, 0);
+        set_global_var(GVAR_KARMA_WANDERER, 0);
+        set_global_var(GVAR_KARMA_BETRAYER, 0);
+        set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+        set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+        tmp_gen_rep := global_var(GVAR_PLAYER_REPUTATION);
+        if (has_trait(TRAIT_PERK, dude_obj, PERK_karma_beacon_perk)) then begin
+            tmp_gen_rep := global_var(GVAR_PLAYER_REPUTATION) * 2;
         end
-        if (global_var(0) >= 1000) then begin
-            set_global_var(37, 1);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+            set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
         end
         else begin
-            if (global_var(0) >= 750) then begin
-                set_global_var(38, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
             end
             else begin
-                if (global_var(0) >= 500) then begin
-                    set_global_var(39, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                 end
                 else begin
-                    if (global_var(0) >= 250) then begin
-                        set_global_var(40, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                        set_global_var(GVAR_KARMA_DEFENDER, 1);
                     end
                     else begin
-                        if (global_var(0) > -250) then begin
-                            set_global_var(41, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                            set_global_var(GVAR_KARMA_WANDERER, 1);
                         end
                         else begin
-                            if (global_var(0) > -500) then begin
-                                set_global_var(42, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                set_global_var(GVAR_KARMA_BETRAYER, 1);
                             end
                             else begin
-                                if (global_var(0) > -750) then begin
-                                    set_global_var(43, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -1000) then begin
-                                        set_global_var(44, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                     end
                                     else begin
-                                        set_global_var(45, 1);
+                                        set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                     end
                                 end
                             end
@@ -524,7 +524,7 @@ begin
                 end
             end
         end
-        set_global_var(50, global_var(50) + -5);
+        set_global_var(GVAR_TOWN_REP_VAULT_CITY, global_var(GVAR_TOWN_REP_VAULT_CITY) + -5);
         debug_msg("Added " + -5 + " to Town Rep");
     end
 end
@@ -532,13 +532,13 @@ end
 procedure look_at_p_proc
 begin
     script_overrides;
-    display_msg(message_str(965, 100));
+    display_mstr(100);
 end
 
 procedure description_p_proc
 begin
     script_overrides;
-    display_msg(message_str(965, 101));
+    display_mstr(101);
 end
 
 procedure use_skill_on_p_proc
@@ -549,10 +549,10 @@ procedure damage_p_proc
 begin
     if (source_obj == dude_obj) then begin
         set_local_var(6, 1);
-        set_global_var(137, 1);
-        set_global_var(79, 6);
-        set_global_var(50, global_var(50) + -30);
-        set_global_var(81, 0);
+        set_global_var(GVAR_ENEMY_VAULT_CITY, 1);
+        set_global_var(GVAR_VAULT_CITIZEN, 6);
+        set_global_var(GVAR_TOWN_REP_VAULT_CITY, global_var(GVAR_TOWN_REP_VAULT_CITY) + -30);
+        set_global_var(GVAR_VAULT_CITIZENSHIP, 0);
     end
 end
 
@@ -561,9 +561,9 @@ begin
     Only_Once := 0;
     From_15 := 0;
     From_33 := 0;
-    critter_add_trait(self_obj, 1, 6, 26);
-    critter_add_trait(self_obj, 1, 5, 24);
-    if (not(metarule(22, 0))) then begin
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 26);
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_TOUGH_CITIZEN);
+    if (not(is_loading_game)) then begin
         if ((game_time_hour > 200) and (game_time_hour < 700)) then begin
             move_to(self_obj, 25488, 0);
             critter_attempt_placement(self_obj, 25488, 0);
@@ -584,14 +584,14 @@ begin
     if (fixed_param == 1) then begin
         if (not(combat_is_initialized) and obj_can_see_obj(self_obj, dude_obj)) then begin
             if ((game_time_hour > 200) and (game_time_hour < 700)) then begin
-                float_msg(self_obj, message_str(965, 258), 8);
+                float_msg(self_obj, mstr(258), FLOAT_MSG_YELLOW);
             end
             else begin
-                if (global_var(494) == 8) then begin
+                if (global_var(GVAR_FALLOUT_2) == 8) then begin
                     call Node002();
                 end
                 else begin
-                    if ((game_time_hour > 600) and (game_time_hour < 1800) and (global_var(79) == 5)) then begin
+                    if ((game_time_hour > 600) and (game_time_hour < 1800) and (global_var(GVAR_VAULT_CITIZEN) == 5)) then begin
                         call Node003();
                     end
                     else begin
@@ -603,7 +603,7 @@ begin
                                 call Node006();
                             end
                             else begin
-                                if ((game_time_hour > 600) and (game_time_hour < 1800) and (global_var(79) == 4)) then begin
+                                if ((game_time_hour > 600) and (game_time_hour < 1800) and (global_var(GVAR_VAULT_CITIZEN) == 4)) then begin
                                     call Node004();
                                 end
                             end
@@ -631,101 +631,101 @@ end
 
 procedure Node002
 begin
-    float_msg(self_obj, message_str(965, random(104, 107)), 8);
+    float_msg(self_obj, mstr(random(104, 107)), FLOAT_MSG_YELLOW);
 end
 
 procedure Node003
 begin
-    float_msg(self_obj, message_str(965, random(108, 109)), 8);
+    float_msg(self_obj, mstr(random(108, 109)), FLOAT_MSG_YELLOW);
 end
 
 procedure Node004
 begin
-    float_msg(self_obj, message_str(965, random(110, 111)), 8);
+    float_msg(self_obj, mstr(random(110, 111)), FLOAT_MSG_YELLOW);
 end
 
 procedure Node005
 begin
-    float_msg(self_obj, message_str(965, 112), 8);
+    float_msg(self_obj, mstr(112), FLOAT_MSG_YELLOW);
     call Node998();
 end
 
 procedure Node006
 begin
-    float_msg(self_obj, message_str(965, 113), 8);
+    float_msg(self_obj, mstr(113), FLOAT_MSG_YELLOW);
     call Node998();
 end
 
 procedure Node007
 begin
-    float_msg(self_obj, message_str(965, random(114, 116)), 8);
+    float_msg(self_obj, mstr(random(114, 116)), FLOAT_MSG_YELLOW);
 end
 
 procedure Node008
 begin
-    gsay_reply(965, 117);
-    giq_option(-3, 965, 118, Node009, 51);
-    giq_option(-3, 965, 119, Node999, 50);
-    giq_option(4, 965, 120, Node014, 50);
-    giq_option(4, 965, 121, Node017, 50);
-    if ((global_var(497) == 1) and ((obj_is_carrying_obj_pid(dude_obj, 124) >= 10) and (obj_is_carrying_obj_pid(dude_obj, 125) >= 10))) then begin
-        giq_option(4, 965, 122, Node032, 50);
+    Reply(117);
+    giq_option(-3, NAME, 118, Node009, BAD_REACTION);
+    giq_option(-3, NAME, 119, Node999, NEUTRAL_REACTION);
+    giq_option(4, NAME, 120, Node014, NEUTRAL_REACTION);
+    giq_option(4, NAME, 121, Node017, NEUTRAL_REACTION);
+    if ((global_var(GVAR_VAULT_BOOZE_SMUGGLING) == 1) and ((obj_is_carrying_obj_pid(dude_obj, 124) >= 10) and (obj_is_carrying_obj_pid(dude_obj, 125) >= 10))) then begin
+        giq_option(4, NAME, 122, Node032, NEUTRAL_REACTION);
     end
-    giq_option(4, 965, 123, Node999, 50);
+    giq_option(4, NAME, 123, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node009
 begin
-    gsay_reply(965, 124);
-    giq_option(-3, 965, 125, Node010, 51);
-    giq_option(-3, 965, 126, Node999, 50);
+    Reply(124);
+    giq_option(-3, NAME, 125, Node010, BAD_REACTION);
+    giq_option(-3, NAME, 126, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node010
 begin
-    gsay_reply(965, 127);
-    giq_option(-3, 965, 128, Node011, 51);
-    giq_option(-3, 965, 129, Node999, 50);
+    Reply(127);
+    giq_option(-3, NAME, 128, Node011, BAD_REACTION);
+    giq_option(-3, NAME, 129, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node011
 begin
     variable LVar0 := 0;
-    LVar0 := create_object_sid(378, 0, 0, -1);
+    LVar0 := create_object(378, 0, 0);
     add_mult_objs_to_inven(dude_obj, LVar0, 100);
-    gsay_reply(965, 130);
-    giq_option(-3, 965, 131, Node999, 51);
-    giq_option(-3, 965, 132, Node999, 50);
+    Reply(130);
+    giq_option(-3, NAME, 131, Node999, BAD_REACTION);
+    giq_option(-3, NAME, 132, Node999, NEUTRAL_REACTION);
     return 0;
     LVar0 := 0;
 end
 
 procedure Node012
 begin
-    if (global_var(79) == 5) then begin
-        gsay_reply(965, 133);
+    if (global_var(GVAR_VAULT_CITIZEN) == 5) then begin
+        Reply(133);
     end
     else begin
-        if (global_var(79) == 4) then begin
-            gsay_reply(965, 134);
+        if (global_var(GVAR_VAULT_CITIZEN) == 4) then begin
+            Reply(134);
         end
         else begin
-            gsay_reply(965, 135);
+            Reply(135);
         end
     end
-    giq_option(-3, 965, 136, Node012a, 51);
-    giq_option(-3, 965, 137, Node999, 50);
-    giq_option(4, 965, 138, Node014, 50);
-    giq_option(4, 965, 139, Node017, 50);
-    if ((global_var(497) == 1) and ((obj_is_carrying_obj_pid(dude_obj, 124) >= 10) and (obj_is_carrying_obj_pid(dude_obj, 125) >= 10))) then begin
-        giq_option(4, 965, 140, Node032, 50);
+    giq_option(-3, NAME, 136, Node012a, BAD_REACTION);
+    giq_option(-3, NAME, 137, Node999, NEUTRAL_REACTION);
+    giq_option(4, NAME, 138, Node014, NEUTRAL_REACTION);
+    giq_option(4, NAME, 139, Node017, NEUTRAL_REACTION);
+    if ((global_var(GVAR_VAULT_BOOZE_SMUGGLING) == 1) and ((obj_is_carrying_obj_pid(dude_obj, 124) >= 10) and (obj_is_carrying_obj_pid(dude_obj, 125) >= 10))) then begin
+        giq_option(4, NAME, 140, Node032, NEUTRAL_REACTION);
     end
-    giq_option(4, 965, 141, Node999, 50);
+    giq_option(4, NAME, 141, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node012a
 begin
-    if ((global_var(79) == 5) or (global_var(79) == 4)) then begin
+    if ((global_var(GVAR_VAULT_CITIZEN) == 5) or (global_var(GVAR_VAULT_CITIZEN) == 4)) then begin
         call Node035();
     end
     else begin
@@ -735,16 +735,16 @@ end
 
 procedure Node013
 begin
-    gsay_reply(965, 142);
-    giq_option(-3, 965, 143, Node013a, 50);
-    giq_option(-3, 965, 144, Node999, 50);
+    Reply(142);
+    giq_option(-3, NAME, 143, Node013a, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 144, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node013a
 begin
-    set_global_var(79, 6);
-    set_global_var(81, 0);
-    set_global_var(86, 0);
+    set_global_var(GVAR_VAULT_CITIZEN, 6);
+    set_global_var(GVAR_VAULT_CITIZENSHIP, 0);
+    set_global_var(GVAR_DAY_PASS_SHOWN, 0);
     DayPass_KickOutCount := obj_is_carrying_obj_pid(dude_obj, 255);
     if (DayPass_KickOutCount > 0) then begin
         DayPass_KickOutObj := obj_carrying_pid_obj(dude_obj, 255);
@@ -757,7 +757,7 @@ begin
         KickOutTemp := rm_mult_objs_from_inven(dude_obj, CitizenPapers_KickOutObj, CitizenPapers_KickOutCount);
         destroy_object(CitizenPapers_KickOutObj);
     end
-    set_global_var(50, global_var(50) + -30);
+    set_global_var(GVAR_TOWN_REP_VAULT_CITY, global_var(GVAR_TOWN_REP_VAULT_CITY) + -30);
     load_map(15, 0);
 end
 
@@ -766,35 +766,35 @@ begin
     if ((From_15 == 1) or (From_33 == 1)) then begin
         From_15 := 0;
         From_33 := 0;
-        gsay_reply(965, 146);
+        Reply(146);
     end
     else begin
-        gsay_reply(965, 145);
+        Reply(145);
     end
-    if (global_var(497) == 0) then begin
-        giq_option(4, 965, 147, Node014a, 50);
+    if (global_var(GVAR_VAULT_BOOZE_SMUGGLING) == 0) then begin
+        giq_option(4, NAME, 147, Node014a, NEUTRAL_REACTION);
     end
     if (item_caps_total(dude_obj) >= 10) then begin
-        giq_option(4, 965, 148, Node014b, 50);
-        giq_option(4, 965, 149, Node014b, 50);
-        giq_option(4, 965, 150, Node014b, 50);
+        giq_option(4, NAME, 148, Node014b, NEUTRAL_REACTION);
+        giq_option(4, NAME, 149, Node014b, NEUTRAL_REACTION);
+        giq_option(4, NAME, 150, Node014b, NEUTRAL_REACTION);
     end
     if (item_caps_total(dude_obj) >= 3) then begin
-        giq_option(4, 965, 151, Node014c, 50);
-        giq_option(4, 965, 152, Node015, 50);
+        giq_option(4, NAME, 151, Node014c, NEUTRAL_REACTION);
+        giq_option(4, NAME, 152, Node015, NEUTRAL_REACTION);
     end
-    giq_option(4, 965, 153, Node017, 50);
+    giq_option(4, NAME, 153, Node017, NEUTRAL_REACTION);
     if (item_caps_total(dude_obj) < 3) then begin
-        giq_option(4, 965, 154, Node999, 50);
+        giq_option(4, NAME, 154, Node999, NEUTRAL_REACTION);
     end
     else begin
-        giq_option(4, 965, 155, Node999, 50);
+        giq_option(4, NAME, 155, Node999, NEUTRAL_REACTION);
     end
 end
 
 procedure Node014a
 begin
-    if (global_var(79) == 5) then begin
+    if (global_var(GVAR_VAULT_CITIZEN) == 5) then begin
         call Node033();
     end
     else begin
@@ -823,11 +823,11 @@ end
 procedure Node015
 begin
     From_15 := 1;
-    gsay_reply(965, 156);
-    giq_option(4, 965, 157, Node015a, 50);
-    giq_option(4, 965, 158, Node014, 50);
-    giq_option(4, 965, 159, Node017, 50);
-    giq_option(4, 965, 160, Node999, 50);
+    Reply(156);
+    giq_option(4, NAME, 157, Node015a, NEUTRAL_REACTION);
+    giq_option(4, NAME, 158, Node014, NEUTRAL_REACTION);
+    giq_option(4, NAME, 159, Node017, NEUTRAL_REACTION);
+    giq_option(4, NAME, 160, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node015a
@@ -841,223 +841,223 @@ end
 
 procedure Node016
 begin
-    gsay_reply(965, 161);
-    if (global_var(497) == 0) then begin
-        giq_option(4, 965, 162, Node014a, 50);
+    Reply(161);
+    if (global_var(GVAR_VAULT_BOOZE_SMUGGLING) == 0) then begin
+        giq_option(4, NAME, 162, Node014a, NEUTRAL_REACTION);
     end
     if (item_caps_total(dude_obj) >= 10) then begin
-        giq_option(4, 965, 163, Node014b, 50);
-        giq_option(4, 965, 164, Node014b, 50);
-        giq_option(4, 965, 165, Node014b, 50);
+        giq_option(4, NAME, 163, Node014b, NEUTRAL_REACTION);
+        giq_option(4, NAME, 164, Node014b, NEUTRAL_REACTION);
+        giq_option(4, NAME, 165, Node014b, NEUTRAL_REACTION);
     end
     if (item_caps_total(dude_obj) >= 3) then begin
-        giq_option(4, 965, 166, Node014c, 50);
-        giq_option(4, 965, 167, Node015, 50);
+        giq_option(4, NAME, 166, Node014c, NEUTRAL_REACTION);
+        giq_option(4, NAME, 167, Node015, NEUTRAL_REACTION);
     end
-    giq_option(4, 965, 168, Node017, 50);
+    giq_option(4, NAME, 168, Node017, NEUTRAL_REACTION);
     if (item_caps_total(dude_obj) < 3) then begin
-        giq_option(4, 965, 169, Node999, 50);
+        giq_option(4, NAME, 169, Node999, NEUTRAL_REACTION);
     end
     else begin
-        giq_option(4, 965, 170, Node999, 50);
+        giq_option(4, NAME, 170, Node999, NEUTRAL_REACTION);
     end
 end
 
 procedure Node017
 begin
-    gsay_reply(965, 171);
+    Reply(171);
     if ((local_var(6) bwand 1) == 0) then begin
-        giq_option(4, 965, 172, Node018, 50);
+        giq_option(4, NAME, 172, Node018, NEUTRAL_REACTION);
     end
-    giq_option(4, 965, 173, Node019, 50);
+    giq_option(4, NAME, 173, Node019, NEUTRAL_REACTION);
     if ((local_var(6) bwand 2) == 0) then begin
-        giq_option(4, 965, 174, Node021, 50);
+        giq_option(4, NAME, 174, Node021, NEUTRAL_REACTION);
     end
-    giq_option(4, 965, 175, Node022, 50);
-    giq_option(4, 965, 176, Node014, 50);
-    if (((local_var(6) bwand 4) == 0) and get_critter_stat(dude_obj, 34) and (local_var(0) > 0) and (global_var(891) bwand 16777216) and ((global_var(79) == 4) or (global_var(79) == 5))) then begin
-        giq_option(6, 965, 300, Node100, 50);
+    giq_option(4, NAME, 175, Node022, NEUTRAL_REACTION);
+    giq_option(4, NAME, 176, Node014, NEUTRAL_REACTION);
+    if (((local_var(6) bwand 4) == 0) and dude_gender and (local_var(0) > 0) and (global_var(GVAR_ENEMY_COUNT) bwand 16777216) and ((global_var(GVAR_VAULT_CITIZEN) == 4) or (global_var(GVAR_VAULT_CITIZEN) == 5))) then begin
+        giq_option(6, NAME, 300, Node100, NEUTRAL_REACTION);
     end
-    if ((global_var(497) == 1) and ((obj_is_carrying_obj_pid(dude_obj, 124) >= 10) and (obj_is_carrying_obj_pid(dude_obj, 125) >= 10))) then begin
-        giq_option(4, 965, 177, Node032, 50);
+    if ((global_var(GVAR_VAULT_BOOZE_SMUGGLING) == 1) and ((obj_is_carrying_obj_pid(dude_obj, 124) >= 10) and (obj_is_carrying_obj_pid(dude_obj, 125) >= 10))) then begin
+        giq_option(4, NAME, 177, Node032, NEUTRAL_REACTION);
     end
-    giq_option(4, 965, 178, Node999, 50);
+    giq_option(4, NAME, 178, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node018
 begin
     set_local_var(6, local_var(6) bwor 1);
-    if ((global_var(79) == 5) or (global_var(79) == 4)) then begin
-        gsay_reply(965, 179);
+    if ((global_var(GVAR_VAULT_CITIZEN) == 5) or (global_var(GVAR_VAULT_CITIZEN) == 4)) then begin
+        Reply(179);
     end
     else begin
-        gsay_reply(965, 180);
+        Reply(180);
     end
-    giq_option(4, 965, 181, Node017, 50);
-    giq_option(4, 965, 182, Node999, 50);
+    giq_option(4, NAME, 181, Node017, NEUTRAL_REACTION);
+    giq_option(4, NAME, 182, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node019
 begin
-    if ((global_var(79) == 5) or (global_var(79) == 4)) then begin
-        gsay_reply(965, 183);
+    if ((global_var(GVAR_VAULT_CITIZEN) == 5) or (global_var(GVAR_VAULT_CITIZEN) == 4)) then begin
+        Reply(183);
     end
     else begin
-        gsay_reply(965, 184);
+        Reply(184);
     end
-    giq_option(4, 965, 185, Node017, 50);
-    if (global_var(79) == 3) then begin
-        giq_option(4, 965, 186, Node020, 50);
+    giq_option(4, NAME, 185, Node017, NEUTRAL_REACTION);
+    if (global_var(GVAR_VAULT_CITIZEN) == 3) then begin
+        giq_option(4, NAME, 186, Node020, NEUTRAL_REACTION);
     end
     else begin
-        if ((global_var(79) == 5) or (global_var(79) == 4)) then begin
-            giq_option(4, 965, 187, Node999, 50);
+        if ((global_var(GVAR_VAULT_CITIZEN) == 5) or (global_var(GVAR_VAULT_CITIZEN) == 4)) then begin
+            giq_option(4, NAME, 187, Node999, NEUTRAL_REACTION);
         end
         else begin
-            giq_option(4, 965, 188, Node999, 50);
+            giq_option(4, NAME, 188, Node999, NEUTRAL_REACTION);
         end
     end
 end
 
 procedure Node020
 begin
-    gsay_reply(965, 189);
-    giq_option(4, 965, 190, Node017, 50);
-    giq_option(4, 965, 191, Node999, 50);
+    Reply(189);
+    giq_option(4, NAME, 190, Node017, NEUTRAL_REACTION);
+    giq_option(4, NAME, 191, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node021
 begin
     set_local_var(6, local_var(6) bwor 2);
-    gsay_reply(965, 192);
-    giq_option(4, 965, 193, Node017, 50);
-    giq_option(4, 965, 194, Node999, 50);
+    Reply(192);
+    giq_option(4, NAME, 193, Node017, NEUTRAL_REACTION);
+    giq_option(4, NAME, 194, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node022
 begin
-    gsay_reply(965, 195);
-    giq_option(4, 965, 196, Node023, 50);
-    giq_option(4, 965, 197, Node024, 50);
-    giq_option(4, 965, 198, Node025, 50);
-    giq_option(4, 965, 199, Node026, 50);
-    giq_option(4, 965, 200, Node017, 50);
-    giq_option(4, 965, 201, Node999, 50);
+    Reply(195);
+    giq_option(4, NAME, 196, Node023, NEUTRAL_REACTION);
+    giq_option(4, NAME, 197, Node024, NEUTRAL_REACTION);
+    giq_option(4, NAME, 198, Node025, NEUTRAL_REACTION);
+    giq_option(4, NAME, 199, Node026, NEUTRAL_REACTION);
+    giq_option(4, NAME, 200, Node017, NEUTRAL_REACTION);
+    giq_option(4, NAME, 201, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node023
 begin
-    if (metarule(17, 6) == 0) then begin
+    if (town_known(6) == 0) then begin
         debug_msg("  mark_on_map(" + 6 + ")");
         mark_area_known(0, 6, 1);
     end
-    gsay_reply(965, 202);
-    giq_option(4, 965, 203, Node024, 50);
-    giq_option(4, 965, 204, Node025, 50);
-    giq_option(4, 965, 205, Node026, 50);
-    giq_option(4, 965, 206, Node017, 50);
-    giq_option(4, 965, 207, Node999, 50);
+    Reply(202);
+    giq_option(4, NAME, 203, Node024, NEUTRAL_REACTION);
+    giq_option(4, NAME, 204, Node025, NEUTRAL_REACTION);
+    giq_option(4, NAME, 205, Node026, NEUTRAL_REACTION);
+    giq_option(4, NAME, 206, Node017, NEUTRAL_REACTION);
+    giq_option(4, NAME, 207, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node024
 begin
-    if (metarule(17, 7) == 0) then begin
+    if (town_known(7) == 0) then begin
         debug_msg("  mark_on_map(" + 7 + ")");
         mark_area_known(0, 7, 1);
     end
-    gsay_reply(965, 208);
-    giq_option(4, 965, 209, Node023, 50);
-    giq_option(4, 965, 210, Node025, 50);
-    giq_option(4, 965, 211, Node026, 50);
-    giq_option(4, 965, 212, Node017, 50);
-    giq_option(4, 965, 213, Node999, 50);
+    Reply(208);
+    giq_option(4, NAME, 209, Node023, NEUTRAL_REACTION);
+    giq_option(4, NAME, 210, Node025, NEUTRAL_REACTION);
+    giq_option(4, NAME, 211, Node026, NEUTRAL_REACTION);
+    giq_option(4, NAME, 212, Node017, NEUTRAL_REACTION);
+    giq_option(4, NAME, 213, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node025
 begin
-    if (metarule(17, 13) == 0) then begin
+    if (town_known(13) == 0) then begin
         debug_msg("  mark_on_map(" + 13 + ")");
         mark_area_known(0, 13, 1);
     end
-    gsay_reply(965, 214);
-    giq_option(4, 965, 215, Node023, 50);
-    giq_option(4, 965, 216, Node024, 50);
-    giq_option(4, 965, 217, Node026, 50);
-    giq_option(4, 965, 218, Node017, 50);
-    giq_option(4, 965, 219, Node999, 50);
+    Reply(214);
+    giq_option(4, NAME, 215, Node023, NEUTRAL_REACTION);
+    giq_option(4, NAME, 216, Node024, NEUTRAL_REACTION);
+    giq_option(4, NAME, 217, Node026, NEUTRAL_REACTION);
+    giq_option(4, NAME, 218, Node017, NEUTRAL_REACTION);
+    giq_option(4, NAME, 219, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node026
 begin
-    if (metarule(17, 5) == 0) then begin
+    if (town_known(5) == 0) then begin
         debug_msg("  mark_on_map(" + 5 + ")");
         mark_area_known(0, 5, 1);
     end
-    gsay_reply(965, 220);
-    giq_option(4, 965, 221, Node023, 50);
-    giq_option(4, 965, 222, Node024, 50);
-    giq_option(4, 965, 223, Node025, 50);
-    giq_option(4, 965, 224, Node017, 50);
-    giq_option(4, 965, 225, Node999, 50);
+    Reply(220);
+    giq_option(4, NAME, 221, Node023, NEUTRAL_REACTION);
+    giq_option(4, NAME, 222, Node024, NEUTRAL_REACTION);
+    giq_option(4, NAME, 223, Node025, NEUTRAL_REACTION);
+    giq_option(4, NAME, 224, Node017, NEUTRAL_REACTION);
+    giq_option(4, NAME, 225, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node027
 begin
-    gsay_reply(965, 226);
-    giq_option(4, 965, 227, Node028, 50);
-    giq_option(4, 965, 228, Node017, 50);
-    giq_option(4, 965, 229, Node999, 50);
+    Reply(226);
+    giq_option(4, NAME, 227, Node028, NEUTRAL_REACTION);
+    giq_option(4, NAME, 228, Node017, NEUTRAL_REACTION);
+    giq_option(4, NAME, 229, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node028
 begin
-    gsay_reply(965, 230);
-    giq_option(4, 965, 231, Node029, 50);
-    giq_option(4, 965, 232, Node017, 50);
-    giq_option(4, 965, 233, Node999, 50);
+    Reply(230);
+    giq_option(4, NAME, 231, Node029, NEUTRAL_REACTION);
+    giq_option(4, NAME, 232, Node017, NEUTRAL_REACTION);
+    giq_option(4, NAME, 233, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node029
 begin
-    gsay_reply(965, 234);
-    giq_option(4, 965, 235, Node030, 50);
-    giq_option(4, 965, 236, Node017, 50);
-    giq_option(4, 965, 237, Node999, 50);
+    Reply(234);
+    giq_option(4, NAME, 235, Node030, NEUTRAL_REACTION);
+    giq_option(4, NAME, 236, Node017, NEUTRAL_REACTION);
+    giq_option(4, NAME, 237, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node030
 begin
-    gsay_reply(965, 238);
-    giq_option(4, 965, 239, Node031, 50);
+    Reply(238);
+    giq_option(4, NAME, 239, Node031, NEUTRAL_REACTION);
 end
 
 procedure Node031
 begin
-    set_global_var(497, 1);
-    gsay_reply(965, 240);
-    if ((global_var(497) == 1) and ((obj_is_carrying_obj_pid(dude_obj, 124) >= 10) and (obj_is_carrying_obj_pid(dude_obj, 125) >= 10))) then begin
-        giq_option(4, 965, 241, Node032, 50);
+    set_global_var(GVAR_VAULT_BOOZE_SMUGGLING, 1);
+    Reply(240);
+    if ((global_var(GVAR_VAULT_BOOZE_SMUGGLING) == 1) and ((obj_is_carrying_obj_pid(dude_obj, 124) >= 10) and (obj_is_carrying_obj_pid(dude_obj, 125) >= 10))) then begin
+        giq_option(4, NAME, 241, Node032, NEUTRAL_REACTION);
     end
-    giq_option(4, 965, 242, Node017, 50);
-    giq_option(4, 965, 243, Node999, 50);
+    giq_option(4, NAME, 242, Node017, NEUTRAL_REACTION);
+    giq_option(4, NAME, 243, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node032
 begin
     variable LVar0 := 0;
     variable LVar1 := 0;
-    set_global_var(497, 2);
+    set_global_var(GVAR_VAULT_BOOZE_SMUGGLING, 2);
     give_exp_points(250);
-    display_msg(message_str(14, 100) + 250 + message_str(14, 101));
-    set_global_var(50, global_var(50) + 3);
+    display_msg(g_mstr(100) + 250 + g_mstr(101));
+    set_global_var(GVAR_TOWN_REP_VAULT_CITY, global_var(GVAR_TOWN_REP_VAULT_CITY) + 3);
     item_caps_adjust(dude_obj, 300);
     LVar0 := obj_carrying_pid_obj(dude_obj, 124);
     LVar1 := rm_mult_objs_from_inven(dude_obj, LVar0, 10);
     LVar0 := obj_carrying_pid_obj(dude_obj, 125);
     LVar1 := rm_mult_objs_from_inven(dude_obj, LVar0, 10);
-    gsay_reply(965, 244);
-    giq_option(4, 965, 245, Node017, 50);
-    giq_option(4, 965, 246, Node999, 50);
+    Reply(244);
+    giq_option(4, NAME, 245, Node017, NEUTRAL_REACTION);
+    giq_option(4, NAME, 246, Node999, NEUTRAL_REACTION);
     return 0;
     LVar0 := 0;
     LVar1 := 0;
@@ -1066,86 +1066,85 @@ end
 procedure Node033
 begin
     From_33 := 1;
-    gsay_reply(965, 247);
-    giq_option(4, 965, 248, Node034, 50);
-    giq_option(4, 965, 249, Node014, 50);
-    giq_option(4, 965, 250, Node017, 50);
-    giq_option(4, 965, 251, Node999, 50);
+    Reply(247);
+    giq_option(4, NAME, 248, Node034, NEUTRAL_REACTION);
+    giq_option(4, NAME, 249, Node014, NEUTRAL_REACTION);
+    giq_option(4, NAME, 250, Node017, NEUTRAL_REACTION);
+    giq_option(4, NAME, 251, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node034
 begin
-    gsay_reply(965, 252);
-    giq_option(4, 965, 253, Node017, 50);
-    giq_option(4, 965, 254, Node999, 50);
+    Reply(252);
+    giq_option(4, NAME, 253, Node017, NEUTRAL_REACTION);
+    giq_option(4, NAME, 254, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node035
 begin
-    gsay_reply(965, 255);
-    giq_option(-3, 965, 256, Node999, 51);
-    giq_option(-3, 965, 257, Node999, 50);
+    Reply(255);
+    giq_option(-3, NAME, 256, Node999, BAD_REACTION);
+    giq_option(-3, NAME, 257, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node100
 begin
-    gsay_reply(965, 310);
-    if (global_var(6) == 1) then begin
-        giq_option(4, 965, 312, Node101, 50);
-        giq_option(4, 965, 314, Node101, 50);
+    Reply(310);
+    if (global_var(GVAR_PLAYER_MARRIED) == 1) then begin
+        giq_option(4, NAME, 312, Node101, NEUTRAL_REACTION);
+        giq_option(4, NAME, 314, Node101, NEUTRAL_REACTION);
     end
     else begin
-        giq_option(4, 965, 311, Node101, 50);
-        giq_option(4, 965, 313, Node101, 50);
+        giq_option(4, NAME, 311, Node101, NEUTRAL_REACTION);
+        giq_option(4, NAME, 313, Node101, NEUTRAL_REACTION);
     end
-    giq_option(4, 965, 315, Node101, 50);
+    giq_option(4, NAME, 315, Node101, NEUTRAL_REACTION);
 end
 
 procedure Node101
 begin
-    gsay_reply(965, 320);
-    giq_option(4, 965, 321, Node102, 50);
+    Reply(320);
+    giq_option(4, NAME, 321, Node102, NEUTRAL_REACTION);
 end
 
 procedure Node102
 begin
-    gsay_reply(965, 325);
-    giq_option(4, 965, 326, Node103, 50);
-    giq_option(4, 965, 327, Node103, 50);
+    Reply(325);
+    giq_option(4, NAME, 326, Node103, NEUTRAL_REACTION);
+    giq_option(4, NAME, 327, Node103, NEUTRAL_REACTION);
 end
 
 procedure Node103
 begin
     set_local_var(6, local_var(6) bwor 4);
-    gsay_reply(965, 330);
-    giq_option(7, 965, 331, Node104, 50);
-    giq_option(4, 965, 332, Node105, 50);
-    giq_option(4, 965, 333, Node999, 50);
+    Reply(330);
+    giq_option(7, NAME, 331, Node104, NEUTRAL_REACTION);
+    giq_option(4, NAME, 332, Node105, NEUTRAL_REACTION);
+    giq_option(4, NAME, 333, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node104
 begin
-    gsay_reply(965, 335);
-    giq_option(4, 965, 332, Node105, 50);
-    giq_option(4, 965, 336, Node017, 50);
-    giq_option(4, 965, 337, Node017, 50);
+    Reply(335);
+    giq_option(4, NAME, 332, Node105, NEUTRAL_REACTION);
+    giq_option(4, NAME, 336, Node017, NEUTRAL_REACTION);
+    giq_option(4, NAME, 337, Node017, NEUTRAL_REACTION);
 end
 
 procedure Node105
 begin
-    gsay_reply(965, 340);
-    giq_option(8, 965, 341, Node106, 50);
-    giq_option(4, 965, 344, Node104, 50);
-    giq_option(4, 965, 342, Node017, 50);
-    giq_option(4, 965, 343, Node017, 50);
+    Reply(340);
+    giq_option(8, NAME, 341, Node106, NEUTRAL_REACTION);
+    giq_option(4, NAME, 344, Node104, NEUTRAL_REACTION);
+    giq_option(4, NAME, 342, Node017, NEUTRAL_REACTION);
+    giq_option(4, NAME, 343, Node017, NEUTRAL_REACTION);
 end
 
 procedure Node106
 begin
-    set_global_var(891, global_var(891) bwor 67108864);
-    gsay_reply(965, 345);
-    giq_option(4, 965, 344, Node104, 50);
-    giq_option(4, 965, 346, Node999, 50);
-    giq_option(4, 965, 347, Node999, 50);
+    set_global_var(GVAR_ENEMY_COUNT, global_var(GVAR_ENEMY_COUNT) bwor 67108864);
+    Reply(345);
+    giq_option(4, NAME, 344, Node104, NEUTRAL_REACTION);
+    giq_option(4, NAME, 346, Node999, NEUTRAL_REACTION);
+    giq_option(4, NAME, 347, Node999, NEUTRAL_REACTION);
 end
-
