diff --git a/Mods/MegaMod/Original/VCINGRD.ssl b/Mods/MegaMod/ReDefine/VCINGRD.ssl
index 2053651..ca5c79a 100644
--- a/Mods/MegaMod/Original/VCINGRD.ssl
+++ b/Mods/MegaMod/ReDefine/VCINGRD.ssl
@@ -182,35 +182,35 @@ end
 
 procedure critter_p_proc
 begin
-    if ((local_var(5) == 2) or (global_var(137) == 1) and obj_can_see_obj(self_obj, dude_obj)) then begin
+    if ((local_var(5) == 2) or (global_var(GVAR_ENEMY_VAULT_CITY) == 1) and obj_can_see_obj(self_obj, dude_obj)) then begin
         set_local_var(5, 1);
-        attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+        attack(dude_obj);
     end
     else begin
-        if (cur_map_index == 16) then begin
+        if (CUR_MAP_VCTYDWTN) then begin
             if (map_var(3) == 1) then begin
                 set_map_var(3, 0);
                 set_local_var(5, 1);
-                if ((tile_num(self_obj) != tile_num(dude_obj)) and (tile_num(dude_obj) != 0)) then begin
-                    if ((dest_tile == tile_num(dude_obj)) and (step_tile < 0)) then begin
+                if ((tile_num(self_obj) != dude_tile) and (dude_tile != 0)) then begin
+                    if ((dest_tile == dude_tile) and (step_tile < 0)) then begin
                         step_tile := step_tile + 1;
                     end
                     else begin
                         if (anim_busy(self_obj) == 0) then begin
-                            dest_tile := tile_num(dude_obj);
-                            step_tile := tile_num(dude_obj);
+                            dest_tile := dude_tile;
+                            step_tile := dude_tile;
                             animate_move_obj_to_tile(self_obj, step_tile, 1);
-                            while ((anim_busy(self_obj) == 0) and (tile_distance(tile_num(self_obj), tile_num(dude_obj)) >= tile_distance(step_tile, tile_num(dude_obj)))) do begin
+                            while ((anim_busy(self_obj) == 0) and (tile_distance(tile_num(self_obj), dude_tile) >= tile_distance(step_tile, dude_tile))) do begin
                                 step_tile := tile_num_in_direction(step_tile, rotation_to_tile(step_tile, tile_num(self_obj)), (5 % tile_distance(step_tile, tile_num(self_obj))) + 1);
                                 animate_move_obj_to_tile(self_obj, step_tile, 1);
                             end
                             if (anim_busy(self_obj) == 0) then begin
-                                if (metarule3(106, tile_num_in_direction(tile_num(self_obj), has_trait(1, self_obj, 10), 1), elevation(self_obj), 0) != 0) then begin
+                                if (metarule3(106, tile_num_in_direction(tile_num(self_obj), obj_get_rot(self_obj), 1), elevation(self_obj), 0) != 0) then begin
                                     if (anim_busy(self_obj) == 0) then begin
-                                        global_temp := rotation_to_tile(tile_num_in_direction(tile_num(self_obj), rotation_to_tile(tile_num_in_direction(tile_num(self_obj), has_trait(1, self_obj, 10), 1), tile_num(self_obj)), 1), tile_num(self_obj));
+                                        global_temp := rotation_to_tile(tile_num_in_direction(tile_num(self_obj), rotation_to_tile(tile_num_in_direction(tile_num(self_obj), obj_get_rot(self_obj), 1), tile_num(self_obj)), 1), tile_num(self_obj));
                                         animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(self_obj), global_temp, random(2, 5)), 0);
                                         global_temp := (global_temp + 1) % 6;
-                                        while ((anim_busy(self_obj) == 0) and (global_temp != rotation_to_tile(tile_num_in_direction(tile_num(self_obj), rotation_to_tile(tile_num_in_direction(tile_num(self_obj), has_trait(1, self_obj, 10), 1), tile_num(self_obj)), 1), tile_num(self_obj)))) do begin
+                                        while ((anim_busy(self_obj) == 0) and (global_temp != rotation_to_tile(tile_num_in_direction(tile_num(self_obj), rotation_to_tile(tile_num_in_direction(tile_num(self_obj), obj_get_rot(self_obj), 1), tile_num(self_obj)), 1), tile_num(self_obj)))) do begin
                                             debug_msg("flee loop: rot == " + global_temp);
                                             animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(self_obj), global_temp, random(2, 5)), 0);
                                             global_temp := (global_temp + 1) % 6;
@@ -223,17 +223,17 @@ begin
                             end
                         end
                         else begin
-                            if (tile_distance(tile_num(self_obj), tile_num(dude_obj)) < tile_distance(dest_tile, tile_num(dude_obj))) then begin
+                            if (tile_distance(tile_num(self_obj), dude_tile) < tile_distance(dest_tile, dude_tile)) then begin
                                 reg_anim_func(2, self_obj);
                             end
                         end
                     end
                 end
-                attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+                attack(dude_obj);
             end
             else begin
                 if (obj_can_see_obj(self_obj, dude_obj)) then begin
-                    if (has_trait(1, self_obj, 666)) then begin
+                    if (obj_is_visible_flag(self_obj)) then begin
                         if (Locker_Watcher == 0) then begin
                             Locker_Watcher := self_obj;
                         end
@@ -247,9 +247,9 @@ begin
             end
         end
         else begin
-            if (cur_map_index == 17) then begin
+            if (CUR_MAP_VCTYCOCL) then begin
                 if (obj_can_see_obj(self_obj, dude_obj)) then begin
-                    if (has_trait(1, self_obj, 666)) then begin
+                    if (obj_is_visible_flag(self_obj)) then begin
                         if (Shack_Shelf_Watcher == 0) then begin
                             Shack_Shelf_Watcher := self_obj;
                         end
@@ -259,7 +259,7 @@ begin
                             end
                         end
                     end
-                    if (has_trait(1, self_obj, 666)) then begin
+                    if (obj_is_visible_flag(self_obj)) then begin
                         if (Shack_Locker_Watcher == 0) then begin
                             Shack_Locker_Watcher := self_obj;
                         end
@@ -288,48 +288,48 @@ begin
     Evil_Critter := 0;
     Slavery_Tolerant := 2;
     Karma_Perception := 1;
-    set_global_var(37, 0);
-    set_global_var(38, 0);
-    set_global_var(39, 0);
-    set_global_var(40, 0);
-    set_global_var(41, 0);
-    set_global_var(42, 0);
-    set_global_var(43, 0);
-    set_global_var(44, 0);
-    set_global_var(45, 0);
-    if (global_var(0) >= 1000) then begin
-        set_global_var(37, 1);
+    set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+    set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+    set_global_var(GVAR_KARMA_DEFENDER, 0);
+    set_global_var(GVAR_KARMA_WANDERER, 0);
+    set_global_var(GVAR_KARMA_BETRAYER, 0);
+    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+    set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+    set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+    if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
     end
     else begin
-        if (global_var(0) >= 750) then begin
-            set_global_var(38, 1);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+            set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
         end
         else begin
-            if (global_var(0) >= 500) then begin
-                set_global_var(39, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
             end
             else begin
-                if (global_var(0) >= 250) then begin
-                    set_global_var(40, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                    set_global_var(GVAR_KARMA_DEFENDER, 1);
                 end
                 else begin
-                    if (global_var(0) > -250) then begin
-                        set_global_var(41, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                        set_global_var(GVAR_KARMA_WANDERER, 1);
                     end
                     else begin
-                        if (global_var(0) > -500) then begin
-                            set_global_var(42, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                            set_global_var(GVAR_KARMA_BETRAYER, 1);
                         end
                         else begin
-                            if (global_var(0) > -750) then begin
-                                set_global_var(43, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                             end
                             else begin
-                                if (global_var(0) > -1000) then begin
-                                    set_global_var(44, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                    set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                 end
                                 else begin
-                                    set_global_var(45, 1);
+                                    set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                 end
                             end
                         end
@@ -340,7 +340,7 @@ begin
     end
     if (local_var(2) == 0) then begin
         set_local_var(2, 1);
-        set_local_var(3, (get_critter_stat(dude_obj, 3) - get_critter_stat(self_obj, 3)) * 5);
+        set_local_var(3, (dude_charisma - self_charisma) * 5);
         debug_msg("Base Reaction == " + local_var(3));
     end
     Static_Reaction := 0;
@@ -352,14 +352,14 @@ begin
     else begin
         reaction_bonus_karma := -1 * (-1 * reaction_bonus_karma / 1000);
     end
-    if (global_var(50) >= 0) then begin
-        reaction_bonus_town_rep := global_var(50) / 2;
+    if (global_var(GVAR_TOWN_REP_VAULT_CITY) >= 0) then begin
+        reaction_bonus_town_rep := global_var(GVAR_TOWN_REP_VAULT_CITY) / 2;
     end
     else begin
-        reaction_bonus_town_rep := -1 * (-1 * global_var(50) / 2);
+        reaction_bonus_town_rep := -1 * (-1 * global_var(GVAR_TOWN_REP_VAULT_CITY) / 2);
     end
-    if (has_trait(2, dude_obj, 13)) then begin
-        if (get_critter_stat(dude_obj, 34) == get_critter_stat(self_obj, 34)) then begin
+    if (has_trait(TRAIT_TRAIT, dude_obj, TRAIT_sex_appeal)) then begin
+        if (dude_gender == self_gender) then begin
             Static_Reaction := Static_Reaction + -20;
         end
         else begin
@@ -375,48 +375,48 @@ begin
         debug_msg("Karma Reaction Bonus == " + Static_Reaction);
         Static_Reaction := Static_Reaction - reaction_bonus_town_rep;
         debug_msg("Town Rep Bonus == " + Static_Reaction);
-        set_global_var(37, 0);
-        set_global_var(38, 0);
-        set_global_var(39, 0);
-        set_global_var(40, 0);
-        set_global_var(41, 0);
-        set_global_var(42, 0);
-        set_global_var(43, 0);
-        set_global_var(44, 0);
-        set_global_var(45, 0);
-        if (global_var(0) >= 1000) then begin
-            set_global_var(37, 1);
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+        set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+        set_global_var(GVAR_KARMA_DEFENDER, 0);
+        set_global_var(GVAR_KARMA_WANDERER, 0);
+        set_global_var(GVAR_KARMA_BETRAYER, 0);
+        set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+        set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+            set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
         end
         else begin
-            if (global_var(0) >= 750) then begin
-                set_global_var(38, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
             end
             else begin
-                if (global_var(0) >= 500) then begin
-                    set_global_var(39, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                 end
                 else begin
-                    if (global_var(0) >= 250) then begin
-                        set_global_var(40, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                        set_global_var(GVAR_KARMA_DEFENDER, 1);
                     end
                     else begin
-                        if (global_var(0) > -250) then begin
-                            set_global_var(41, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                            set_global_var(GVAR_KARMA_WANDERER, 1);
                         end
                         else begin
-                            if (global_var(0) > -500) then begin
-                                set_global_var(42, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                set_global_var(GVAR_KARMA_BETRAYER, 1);
                             end
                             else begin
-                                if (global_var(0) > -750) then begin
-                                    set_global_var(43, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -1000) then begin
-                                        set_global_var(44, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                     end
                                     else begin
-                                        set_global_var(45, 1);
+                                        set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                     end
                                 end
                             end
@@ -425,35 +425,35 @@ begin
                 end
             end
         end
-        if (global_var(37) == 1) then begin
+        if (global_var(GVAR_KARMA_HOLY_WARRIOR) == 1) then begin
             Static_Reaction := Static_Reaction - 20;
         end
         else begin
-            if (global_var(38) == 1) then begin
+            if (global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES) == 1) then begin
                 Static_Reaction := Static_Reaction - 15;
             end
             else begin
-                if (global_var(39) == 1) then begin
+                if (global_var(GVAR_KARMA_SHIELD_OF_HOPE) == 1) then begin
                     Static_Reaction := Static_Reaction - 10;
                 end
                 else begin
-                    if (global_var(40) == 1) then begin
+                    if (global_var(GVAR_KARMA_DEFENDER) == 1) then begin
                         Static_Reaction := Static_Reaction - 5;
                     end
                     else begin
-                        if (global_var(41) == 1) then begin
+                        if (global_var(GVAR_KARMA_WANDERER) == 1) then begin
                             Static_Reaction := Static_Reaction - 0;
                         end
                         else begin
-                            if (global_var(42) == 1) then begin
+                            if (global_var(GVAR_KARMA_BETRAYER) == 1) then begin
                                 Static_Reaction := Static_Reaction - -5;
                             end
                             else begin
-                                if (global_var(43) == 1) then begin
+                                if (global_var(GVAR_KARMA_SWORD_OF_DESPAIR) == 1) then begin
                                     Static_Reaction := Static_Reaction - -10;
                                 end
                                 else begin
-                                    if (global_var(44) == 1) then begin
+                                    if (global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE) == 1) then begin
                                         Static_Reaction := Static_Reaction - -15;
                                     end
                                     else begin
@@ -473,48 +473,48 @@ begin
         debug_msg("Karma Reaction Bonus == " + Static_Reaction);
         Static_Reaction := Static_Reaction + reaction_bonus_town_rep;
         debug_msg("Town Rep Bonus == " + Static_Reaction);
-        set_global_var(37, 0);
-        set_global_var(38, 0);
-        set_global_var(39, 0);
-        set_global_var(40, 0);
-        set_global_var(41, 0);
-        set_global_var(42, 0);
-        set_global_var(43, 0);
-        set_global_var(44, 0);
-        set_global_var(45, 0);
-        if (global_var(0) >= 1000) then begin
-            set_global_var(37, 1);
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+        set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+        set_global_var(GVAR_KARMA_DEFENDER, 0);
+        set_global_var(GVAR_KARMA_WANDERER, 0);
+        set_global_var(GVAR_KARMA_BETRAYER, 0);
+        set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+        set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+            set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
         end
         else begin
-            if (global_var(0) >= 750) then begin
-                set_global_var(38, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
             end
             else begin
-                if (global_var(0) >= 500) then begin
-                    set_global_var(39, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                 end
                 else begin
-                    if (global_var(0) >= 250) then begin
-                        set_global_var(40, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                        set_global_var(GVAR_KARMA_DEFENDER, 1);
                     end
                     else begin
-                        if (global_var(0) > -250) then begin
-                            set_global_var(41, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                            set_global_var(GVAR_KARMA_WANDERER, 1);
                         end
                         else begin
-                            if (global_var(0) > -500) then begin
-                                set_global_var(42, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                set_global_var(GVAR_KARMA_BETRAYER, 1);
                             end
                             else begin
-                                if (global_var(0) > -750) then begin
-                                    set_global_var(43, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -1000) then begin
-                                        set_global_var(44, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                     end
                                     else begin
-                                        set_global_var(45, 1);
+                                        set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                     end
                                 end
                             end
@@ -523,35 +523,35 @@ begin
                 end
             end
         end
-        if (global_var(37) == 1) then begin
+        if (global_var(GVAR_KARMA_HOLY_WARRIOR) == 1) then begin
             Static_Reaction := Static_Reaction + 20;
         end
         else begin
-            if (global_var(38) == 1) then begin
+            if (global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES) == 1) then begin
                 Static_Reaction := Static_Reaction + 15;
             end
             else begin
-                if (global_var(39) == 1) then begin
+                if (global_var(GVAR_KARMA_SHIELD_OF_HOPE) == 1) then begin
                     Static_Reaction := Static_Reaction + 10;
                 end
                 else begin
-                    if (global_var(40) == 1) then begin
+                    if (global_var(GVAR_KARMA_DEFENDER) == 1) then begin
                         Static_Reaction := Static_Reaction + 5;
                     end
                     else begin
-                        if (global_var(41) == 1) then begin
+                        if (global_var(GVAR_KARMA_WANDERER) == 1) then begin
                             Static_Reaction := Static_Reaction + 0;
                         end
                         else begin
-                            if (global_var(42) == 1) then begin
+                            if (global_var(GVAR_KARMA_BETRAYER) == 1) then begin
                                 Static_Reaction := Static_Reaction + -5;
                             end
                             else begin
-                                if (global_var(43) == 1) then begin
+                                if (global_var(GVAR_KARMA_SWORD_OF_DESPAIR) == 1) then begin
                                     Static_Reaction := Static_Reaction + -10;
                                 end
                                 else begin
-                                    if (global_var(44) == 1) then begin
+                                    if (global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE) == 1) then begin
                                         Static_Reaction := Static_Reaction + -15;
                                     end
                                     else begin
@@ -567,38 +567,38 @@ begin
         debug_msg("Good Person Reaction Bonus == " + Static_Reaction);
     end
     if (Slavery_Tolerant == 0) then begin
-        if (global_var(11) == 1) then begin
+        if (global_var(GVAR_REPUTATION_SLAVER) == 1) then begin
             Static_Reaction := Static_Reaction + (2 * -25);
         end
     end
     else begin
         if (Slavery_Tolerant == 1) then begin
-            if (global_var(11) == 1) then begin
+            if (global_var(GVAR_REPUTATION_SLAVER) == 1) then begin
                 Static_Reaction := Static_Reaction + -25;
             end
         end
         else begin
             if (Slavery_Tolerant == 3) then begin
-                if (global_var(11) == 1) then begin
+                if (global_var(GVAR_REPUTATION_SLAVER) == 1) then begin
                     Static_Reaction := Static_Reaction - -25;
                 end
             end
         end
     end
     debug_msg("Slaver + Aligned Reaction == " + Static_Reaction);
-    Static_Reaction := Static_Reaction + (10 * has_trait(0, dude_obj, 10));
+    Static_Reaction := Static_Reaction + (10 * has_trait(TRAIT_PERK, dude_obj, PERK_presence));
     debug_msg("Presence Reaction == " + Static_Reaction);
-    if (global_var(1) >= 2) then begin
+    if (global_var(GVAR_CHILDKILLER_REPUTATION) >= 2) then begin
         Static_Reaction := Static_Reaction + -30;
     end
     debug_msg("Childkiller Reaction == " + Static_Reaction);
     set_local_var(0, local_var(3) + Static_Reaction);
     debug_msg("Initial Reaction == " + local_var(0));
-    if (global_var(494) == 8) then begin
+    if (global_var(GVAR_FALLOUT_2) == 8) then begin
         call Node014();
     end
     else begin
-        if (global_var(79) == 5) then begin
+        if (global_var(GVAR_VAULT_CITIZEN) == 5) then begin
             call Node008();
         end
         else begin
@@ -618,7 +618,7 @@ begin
                     end_dialogue;
                 end
                 else begin
-                    if (global_var(79) == 4) then begin
+                    if (global_var(GVAR_VAULT_CITIZEN) == 4) then begin
                         call Node006();
                     end
                     else begin
@@ -630,7 +630,7 @@ begin
                             end_dialogue;
                         end
                         else begin
-                            if ((get_critter_stat(dude_obj, 3) < 6) or (get_critter_stat(dude_obj, 4) < 4)) then begin
+                            if ((dude_charisma < 6) or (dude_iq < 4)) then begin
                                 call Node001();
                             end
                             else begin
@@ -649,59 +649,59 @@ end
 procedure destroy_p_proc
 begin
     if (source_obj == dude_obj) then begin
-        set_global_var(5, global_var(5) + 1);
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(4) > (3 * global_var(5))) or (global_var(2) == 1)) and (global_var(1) == 0)) then begin
-            set_global_var(2, 1);
-            set_global_var(3, 0);
+        set_global_var(GVAR_GOOD_MONSTER, global_var(GVAR_GOOD_MONSTER) + 1);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(4) > (3 * global_var(GVAR_GOOD_MONSTER))) or (global_var(GVAR_CHAMPION_REPUTATION) == 1)) and (global_var(GVAR_CHILDKILLER_REPUTATION) == 0)) then begin
+            set_global_var(GVAR_CHAMPION_REPUTATION, 1);
+            set_global_var(GVAR_BERSERKER_REPUTATION, 0);
         end
-        if (metarule(51, self_obj) == 2) then begin
+        if (critter_kill_type(self_obj) == 2) then begin
             if (source_obj == dude_obj) then begin
-                set_global_var(2, 0);
-                set_global_var(1, global_var(1) + 1);
-                set_global_var(0, global_var(0) + -15);
+                set_global_var(GVAR_CHAMPION_REPUTATION, 0);
+                set_global_var(GVAR_CHILDKILLER_REPUTATION, global_var(GVAR_CHILDKILLER_REPUTATION) + 1);
+                set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + -15);
                 debug_msg("Player gains " + -15 + " Karma Points.");
-                set_global_var(37, 0);
-                set_global_var(38, 0);
-                set_global_var(39, 0);
-                set_global_var(40, 0);
-                set_global_var(41, 0);
-                set_global_var(42, 0);
-                set_global_var(43, 0);
-                set_global_var(44, 0);
-                set_global_var(45, 0);
-                if (global_var(0) >= 1000) then begin
-                    set_global_var(37, 1);
+                set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+                set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+                set_global_var(GVAR_KARMA_DEFENDER, 0);
+                set_global_var(GVAR_KARMA_WANDERER, 0);
+                set_global_var(GVAR_KARMA_BETRAYER, 0);
+                set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+                set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+                set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+                    set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
                 end
                 else begin
-                    if (global_var(0) >= 750) then begin
-                        set_global_var(38, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
                     end
                     else begin
-                        if (global_var(0) >= 500) then begin
-                            set_global_var(39, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                            set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                         end
                         else begin
-                            if (global_var(0) >= 250) then begin
-                                set_global_var(40, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                                set_global_var(GVAR_KARMA_DEFENDER, 1);
                             end
                             else begin
-                                if (global_var(0) > -250) then begin
-                                    set_global_var(41, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                                    set_global_var(GVAR_KARMA_WANDERER, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -500) then begin
-                                        set_global_var(42, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                        set_global_var(GVAR_KARMA_BETRAYER, 1);
                                     end
                                     else begin
-                                        if (global_var(0) > -750) then begin
-                                            set_global_var(43, 1);
+                                        if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                            set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                         end
                                         else begin
-                                            if (global_var(0) > -1000) then begin
-                                                set_global_var(44, 1);
+                                            if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                                set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                             end
                                             else begin
-                                                set_global_var(45, 1);
+                                                set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                             end
                                         end
                                     end
@@ -710,58 +710,58 @@ begin
                         end
                     end
                 end
-                set_global_var(50, global_var(50) + -8);
+                set_global_var(GVAR_TOWN_REP_VAULT_CITY, global_var(GVAR_TOWN_REP_VAULT_CITY) + -8);
                 debug_msg("Added " + -8 + " to Town Rep");
             end
         end
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(5) > (2 * global_var(4))) or (global_var(3) == 1))) then begin
-            set_global_var(3, 1);
-            set_global_var(2, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
+            set_global_var(GVAR_BERSERKER_REPUTATION, 1);
+            set_global_var(GVAR_CHAMPION_REPUTATION, 0);
         end
-        set_global_var(0, global_var(0) + -10);
+        set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + -10);
         debug_msg("Player gains " + -10 + " Karma Points.");
-        set_global_var(37, 0);
-        set_global_var(38, 0);
-        set_global_var(39, 0);
-        set_global_var(40, 0);
-        set_global_var(41, 0);
-        set_global_var(42, 0);
-        set_global_var(43, 0);
-        set_global_var(44, 0);
-        set_global_var(45, 0);
-        if (global_var(0) >= 1000) then begin
-            set_global_var(37, 1);
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+        set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+        set_global_var(GVAR_KARMA_DEFENDER, 0);
+        set_global_var(GVAR_KARMA_WANDERER, 0);
+        set_global_var(GVAR_KARMA_BETRAYER, 0);
+        set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+        set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+            set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
         end
         else begin
-            if (global_var(0) >= 750) then begin
-                set_global_var(38, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
             end
             else begin
-                if (global_var(0) >= 500) then begin
-                    set_global_var(39, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                 end
                 else begin
-                    if (global_var(0) >= 250) then begin
-                        set_global_var(40, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                        set_global_var(GVAR_KARMA_DEFENDER, 1);
                     end
                     else begin
-                        if (global_var(0) > -250) then begin
-                            set_global_var(41, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                            set_global_var(GVAR_KARMA_WANDERER, 1);
                         end
                         else begin
-                            if (global_var(0) > -500) then begin
-                                set_global_var(42, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                set_global_var(GVAR_KARMA_BETRAYER, 1);
                             end
                             else begin
-                                if (global_var(0) > -750) then begin
-                                    set_global_var(43, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -1000) then begin
-                                        set_global_var(44, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                     end
                                     else begin
-                                        set_global_var(45, 1);
+                                        set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                     end
                                 end
                             end
@@ -770,7 +770,7 @@ begin
                 end
             end
         end
-        set_global_var(50, global_var(50) + -5);
+        set_global_var(GVAR_TOWN_REP_VAULT_CITY, global_var(GVAR_TOWN_REP_VAULT_CITY) + -5);
         debug_msg("Added " + -5 + " to Town Rep");
     end
 end
@@ -778,13 +778,13 @@ end
 procedure look_at_p_proc
 begin
     script_overrides;
-    display_msg(message_str(90, 100));
+    display_msg(message_str(SCRIPT_VCCRTGRD, 100));
 end
 
 procedure description_p_proc
 begin
     script_overrides;
-    display_msg(message_str(90, 101));
+    display_msg(message_str(SCRIPT_VCCRTGRD, 101));
 end
 
 procedure use_skill_on_p_proc
@@ -795,18 +795,18 @@ procedure damage_p_proc
 begin
     if (source_obj == dude_obj) then begin
         set_local_var(6, 1);
-        set_global_var(137, 1);
-        set_global_var(79, 6);
-        set_global_var(50, global_var(50) + -30);
-        set_global_var(81, 0);
+        set_global_var(GVAR_ENEMY_VAULT_CITY, 1);
+        set_global_var(GVAR_VAULT_CITIZEN, 6);
+        set_global_var(GVAR_TOWN_REP_VAULT_CITY, global_var(GVAR_TOWN_REP_VAULT_CITY) + -30);
+        set_global_var(GVAR_VAULT_CITIZENSHIP, 0);
     end
 end
 
 procedure map_enter_p_proc
 begin
     Only_Once := 0;
-    critter_add_trait(self_obj, 1, 6, 26);
-    critter_add_trait(self_obj, 1, 5, 18);
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 26);
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_VAULT_GUARD);
     add_timer_event(self_obj, game_ticks(random(20, 40)), 1);
 end
 
@@ -815,11 +815,11 @@ begin
     if (fixed_param == 1) then begin
         if (not(combat_is_initialized)) then begin
             if (obj_can_see_obj(self_obj, dude_obj)) then begin
-                if (global_var(494) == 8) then begin
+                if (global_var(GVAR_FALLOUT_2) == 8) then begin
                     call Node013();
                 end
                 else begin
-                    if (global_var(79) == 5) then begin
+                    if (global_var(GVAR_VAULT_CITIZEN) == 5) then begin
                         call Node008();
                     end
                     else begin
@@ -831,7 +831,7 @@ begin
                                 dialogue_system_enter;
                             end
                             else begin
-                                if (global_var(79) == 4) then begin
+                                if (global_var(GVAR_VAULT_CITIZEN) == 4) then begin
                                     call Node006();
                                 end
                                 else begin
@@ -863,139 +863,138 @@ end
 
 procedure Node001
 begin
-    float_msg(self_obj, message_str(91, random(100, 103)), 8);
+    float_msg(self_obj, mstr(random(100, 103)), FLOAT_MSG_YELLOW);
 end
 
 procedure Node002
 begin
-    gsay_reply(91, 104);
-    giq_option(4, 91, 105, Node003, 50);
-    if (global_var(79) == 3) then begin
-        giq_option(4, 91, 127, Node015, 50);
-    end
-    giq_option(4, 91, 106, Node003, 50);
-    giq_option(4, 91, 107, Node004, 50);
-    giq_option(-3, 91, 108, Node003, 50);
+    Reply(104);
+    giq_option(4, NAME, 105, Node003, NEUTRAL_REACTION);
+    if (global_var(GVAR_VAULT_CITIZEN) == 3) then begin
+        giq_option(4, NAME, 127, Node015, NEUTRAL_REACTION);
+    end
+    giq_option(4, NAME, 106, Node003, NEUTRAL_REACTION);
+    giq_option(4, NAME, 107, Node004, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 108, Node003, NEUTRAL_REACTION);
 end
 
 procedure Node003
 begin
-    gsay_reply(91, 109);
-    giq_option(4, 91, 110, Node997, 50);
-    giq_option(4, 91, 112, Node004, 50);
-    giq_option(-3, 91, 111, Node997, 50);
+    Reply(109);
+    giq_option(4, NAME, 110, Node997, NEUTRAL_REACTION);
+    giq_option(4, NAME, 112, Node004, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 111, Node997, NEUTRAL_REACTION);
 end
 
 procedure Node004
 begin
-    gsay_reply(91, 113);
-    giq_option(4, 91, 114, Node997, 50);
-    giq_option(4, 91, 115, Node005, 50);
+    Reply(113);
+    giq_option(4, NAME, 114, Node997, NEUTRAL_REACTION);
+    giq_option(4, NAME, 115, Node005, NEUTRAL_REACTION);
 end
 
 procedure Node005
 begin
-    gsay_reply(91, 116);
-    giq_option(4, 91, 117, Node997, 50);
-    giq_option(4, 91, 118, Node998, 51);
+    Reply(116);
+    giq_option(4, NAME, 117, Node997, NEUTRAL_REACTION);
+    giq_option(4, NAME, 118, Node998, BAD_REACTION);
 end
 
 procedure Node006
 begin
-    float_msg(self_obj, message_str(91, random(119, 122)), 8);
+    float_msg(self_obj, mstr(random(119, 122)), FLOAT_MSG_YELLOW);
 end
 
 procedure Node007
 begin
-    float_msg(self_obj, message_str(91, random(123, 126)), 8);
+    float_msg(self_obj, mstr(random(123, 126)), FLOAT_MSG_YELLOW);
 end
 
 procedure Node008
 begin
-    float_msg(self_obj, message_str(91, random(128, 131)), 8);
+    float_msg(self_obj, mstr(random(128, 131)), FLOAT_MSG_YELLOW);
 end
 
 procedure Node009
 begin
     if (party_member_obj(16777323) != 0) then begin
-        gsay_reply(91, 132);
+        Reply(132);
     end
     else begin
         if (party_member_obj(16777377) != 0) then begin
-            gsay_reply(91, 133);
+            Reply(133);
         end
     end
-    giq_option(-3, 91, 134, Node011, 50);
+    giq_option(-3, NAME, 134, Node011, NEUTRAL_REACTION);
     if (party_member_obj(16777323) != 0) then begin
-        giq_option(4, 91, 135, Node011, 50);
+        giq_option(4, NAME, 135, Node011, NEUTRAL_REACTION);
     end
     if (party_member_obj(16777377) != 0) then begin
-        giq_option(4, 91, 136, Node011, 50);
+        giq_option(4, NAME, 136, Node011, NEUTRAL_REACTION);
     end
-    giq_option(4, 91, 137, Node010, 50);
-    giq_option(4, 91, 138, Node012, 50);
+    giq_option(4, NAME, 137, Node010, NEUTRAL_REACTION);
+    giq_option(4, NAME, 138, Node012, NEUTRAL_REACTION);
 end
 
 procedure Node010
 begin
-    gsay_reply(91, 139);
-    giq_option(4, 91, 140, Node997, 50);
-    giq_option(4, 91, 141, Node012, 51);
+    Reply(139);
+    giq_option(4, NAME, 140, Node997, NEUTRAL_REACTION);
+    giq_option(4, NAME, 141, Node012, BAD_REACTION);
 end
 
 procedure Node011
 begin
-    if (get_critter_stat(dude_obj, 4) < 4) then begin
-        gsay_reply(91, message_str(91, 142) + " " + message_str(91, 144));
+    if (dude_iq < 4) then begin
+        Reply(mstr(142) + " " + mstr(144));
     end
     else begin
-        gsay_reply(91, message_str(91, 143) + " " + message_str(91, 144));
+        Reply(mstr(143) + " " + mstr(144));
     end
-    giq_option(-3, 91, 145, Node012, 51);
-    giq_option(-3, 91, 146, Node997, 50);
-    giq_option(4, 91, 147, Node997, 50);
-    giq_option(4, 91, 148, Node012, 50);
+    giq_option(-3, NAME, 145, Node012, BAD_REACTION);
+    giq_option(-3, NAME, 146, Node997, NEUTRAL_REACTION);
+    giq_option(4, NAME, 147, Node997, NEUTRAL_REACTION);
+    giq_option(4, NAME, 148, Node012, NEUTRAL_REACTION);
 end
 
 procedure Node012
 begin
-    set_global_var(79, 6);
-    set_global_var(50, global_var(50) + -30);
-    set_global_var(81, 0);
-    set_global_var(86, 0);
-    gsay_reply(91, 149);
-    giq_option(-3, 91, 150, Node998, 51);
-    giq_option(4, 91, 151, Node998, 51);
+    set_global_var(GVAR_VAULT_CITIZEN, 6);
+    set_global_var(GVAR_TOWN_REP_VAULT_CITY, global_var(GVAR_TOWN_REP_VAULT_CITY) + -30);
+    set_global_var(GVAR_VAULT_CITIZENSHIP, 0);
+    set_global_var(GVAR_DAY_PASS_SHOWN, 0);
+    Reply(149);
+    giq_option(-3, NAME, 150, Node998, BAD_REACTION);
+    giq_option(4, NAME, 151, Node998, BAD_REACTION);
 end
 
 procedure Node013
 begin
-    float_msg(self_obj, message_str(91, random(152, 154)), 8);
+    float_msg(self_obj, mstr(random(152, 154)), FLOAT_MSG_YELLOW);
 end
 
 procedure Node014
 begin
-    float_msg(self_obj, message_str(91, random(155, 160)), 8);
+    float_msg(self_obj, mstr(random(155, 160)), FLOAT_MSG_YELLOW);
 end
 
 procedure Node015
 begin
-    gsay_reply(91, 161);
+    Reply(161);
     if (obj_is_carrying_obj_pid(dude_obj, 256) > 0) then begin
-        giq_option(4, 91, 162, Node016, 50);
+        giq_option(4, NAME, 162, Node016, NEUTRAL_REACTION);
     end
-    giq_option(4, 91, 163, Node005, 50);
+    giq_option(4, NAME, 163, Node005, NEUTRAL_REACTION);
 end
 
 procedure Node016
 begin
-    gsay_reply(91, 164);
-    giq_option(4, 91, 165, Node999, 50);
-    giq_option(4, 91, 166, Node999, 51);
+    Reply(164);
+    giq_option(4, NAME, 165, Node999, NEUTRAL_REACTION);
+    giq_option(4, NAME, 166, Node999, BAD_REACTION);
 end
 
 procedure Node997
 begin
     load_map(15, 0);
 end
-
