diff --git a/Mods/MegaMod/Original/VCPRISON.ssl b/Mods/MegaMod/ReDefine/VCPRISON.ssl
index 3780549..f89806e 100644
--- a/Mods/MegaMod/Original/VCPRISON.ssl
+++ b/Mods/MegaMod/ReDefine/VCPRISON.ssl
@@ -18,26 +18,26 @@ procedure quest_complete_;
 procedure look_at_p_proc
 begin
     script_overrides;
-    display_msg(message_str(1610, 101));
+    display_mstr(101);
 end
 
 procedure description_p_proc
 begin
     script_overrides;
-    display_msg(message_str(1610, 101));
+    display_mstr(101);
 end
 
 procedure critter_p_proc
 begin
-    if (global_var(774) == 4) then begin
+    if (global_var(GVAR_MARDOK) == 4) then begin
         animate_move_obj_to_tile(self_obj, 19493, 1);
-        float_msg(self_obj, message_str(1610, random(700, 702)), 8);
+        float_msg(self_obj, mstr(random(700, 702)), FLOAT_MSG_YELLOW);
     end
     if (tile_num(self_obj) == 19493) then begin
-        set_global_var(774, 6);
+        set_global_var(GVAR_MARDOK, 6);
     end
-    if ((global_var(774) == 0) and (tile_distance(tile_num(self_obj), tile_num(dude_obj)) < 12) and (local_var(0) == 0)) then begin
-        float_msg(self_obj, message_str(1610, 600), 3);
+    if ((global_var(GVAR_MARDOK) == 0) and (tile_distance(tile_num(self_obj), dude_tile) < 12) and (local_var(0) == 0)) then begin
+        float_msg(self_obj, mstr(600), FLOAT_MSG_GREEN);
         set_local_var(0, 1);
         add_timer_event(self_obj, 100, 1);
     end
@@ -60,7 +60,7 @@ end
 
 procedure talk_p_proc
 begin
-    if ((global_var(774) == 0) and (map_var(12) == 1)) then begin
+    if ((global_var(GVAR_MARDOK) == 0) and (map_var(12) == 1)) then begin
         start_gdialog(1610, self_obj, -1, -1, -1);
         gsay_start;
         call Node001();
@@ -71,13 +71,13 @@ begin
         if (map_var(10) == 0) then begin
             item_caps_adjust(dude_obj, 200);
             give_exp_points(500);
-            display_msg(message_str(14, 100) + 500 + message_str(14, 101));
-            float_msg(self_obj, message_str(1610, 601), 3);
+            display_msg(g_mstr(100) + 500 + g_mstr(101));
+            float_msg(self_obj, mstr(601), FLOAT_MSG_GREEN);
             set_map_var(10, 1);
         end
         else begin
             if (map_var(10) == 1) then begin
-                float_msg(self_obj, message_str(1610, random(670, 674)), 8);
+                float_msg(self_obj, mstr(random(670, 674)), FLOAT_MSG_YELLOW);
             end
         end
     end
@@ -85,30 +85,30 @@ end
 
 procedure Node001
 begin
-    gsay_reply(1610, 100);
-    gsay_option(1610, 110, Node002, -1);
-    gsay_option(1610, 111, Node999, -1);
+    Reply(100);
+    giq_option(1, NAME, 110, Node002, -1);
+    giq_option(1, NAME, 111, Node999, -1);
 end
 
 procedure Node002
 begin
-    gsay_reply(1610, 200);
-    gsay_option(1610, 210, Node003, -1);
+    Reply(200);
+    giq_option(1, NAME, 210, Node003, -1);
 end
 
 procedure Node003
 begin
-    gsay_reply(1610, 300);
-    gsay_option(1610, 310, Node004, -1);
-    gsay_option(1610, 311, accept_quest, -1);
-    gsay_option(1610, 312, Node999, -1);
+    Reply(300);
+    giq_option(1, NAME, 310, Node004, -1);
+    giq_option(1, NAME, 311, accept_quest, -1);
+    giq_option(1, NAME, 312, Node999, -1);
 end
 
 procedure Node004
 begin
-    gsay_reply(1610, 400);
-    gsay_option(1610, 410, accept_quest, -1);
-    gsay_option(1610, 411, Node999, -1);
+    Reply(400);
+    giq_option(1, NAME, 410, accept_quest, -1);
+    giq_option(1, NAME, 411, Node999, -1);
 end
 
 procedure Node999
@@ -121,13 +121,12 @@ end
 
 procedure accept_quest
 begin
-    set_global_var(774, 1);
+    set_global_var(GVAR_MARDOK, 1);
 end
 
 procedure quest_complete_
 begin
-    if (global_var(774) < 4) then begin
-        set_global_var(774, 4);
+    if (global_var(GVAR_MARDOK) < 4) then begin
+        set_global_var(GVAR_MARDOK, 4);
     end
 end
-
