diff --git a/Mods/MegaMod/Original/VIDARBOX.ssl b/Mods/MegaMod/ReDefine/VIDARBOX.ssl
index 536ee98..0589821 100644
--- a/Mods/MegaMod/Original/VIDARBOX.ssl
+++ b/Mods/MegaMod/ReDefine/VIDARBOX.ssl
@@ -149,9 +149,9 @@ begin
     variable LVar0 := 0;
     variable LVar1 := 0;
     vault_city_darrow_box := self_obj;
-    if (metarule(22, 0) == 0) then begin
+    if (is_loading_game == 0) then begin
         if (local_var(0) < game_time) then begin
-            LVar0 := create_object_sid(128, 5, 0, -1);
+            LVar0 := create_object(128, 5, 0);
             if (obj_is_carrying_obj_pid(self_obj, 479) > 0) then begin
                 LVar1 := obj_carrying_pid_obj(self_obj, 479);
                 rm_obj_from_inven(self_obj, LVar1);
@@ -397,14 +397,14 @@ begin
                 rm_obj_from_inven(self_obj, LVar1);
                 add_obj_to_inven(LVar0, LVar1);
             end
-            global_temp := create_object_sid(128, 0, 0, -1);
+            global_temp := create_object(128, 0, 0);
             move_obj_inven_to_obj(self_obj, global_temp);
             destroy_object(global_temp);
             restock_amt := random(100, 500);
             if (obj_is_carrying_obj_pid(self_obj, 41) < restock_amt) then begin
                 if (100 >= random(1, 100)) then begin
                     if (obj_is_carrying_obj_pid(self_obj, 41) < restock_amt) then begin
-                        restock_obj := create_object_sid(41, 0, 0, -1);
+                        restock_obj := create_object(41, 0, 0);
                         add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 41));
                     end
                     else begin
@@ -417,17 +417,17 @@ begin
                                 restock_obj := obj_carrying_pid_obj(self_obj, 41);
                                 if (obj_type(self_obj) == 1) then begin
                                     restock_obj := obj_carrying_pid_obj(self_obj, 41);
-                                    if (critter_inven_obj(self_obj, 0) == restock_obj) then begin
-                                        if (obj_item_subtype(critter_inven_obj(self_obj, 0)) == 0) then begin
-                                            restock_obj := critter_inven_obj(self_obj, 0);
+                                    if (critter_inven_obj(self_obj, INVEN_TYPE_WORN) == restock_obj) then begin
+                                        if (obj_item_subtype(critter_inven_obj(self_obj, INVEN_TYPE_WORN)) == 0) then begin
+                                            restock_obj := critter_inven_obj(self_obj, INVEN_TYPE_WORN);
                                             debug_msg("armour pid == " + obj_pid(restock_obj));
                                             rm_obj_from_inven(self_obj, restock_obj);
                                             add_obj_to_inven(self_obj, restock_obj);
                                         end
                                     end
                                     else begin
-                                        if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                            metarule(43, self_obj);
+                                        if ((critter_inven_obj(self_obj, INVEN_TYPE_LEFT_HAND) == restock_obj) or (critter_inven_obj(self_obj, INVEN_TYPE_RIGHT_HAND) == restock_obj)) then begin
+                                            inven_unwield(self_obj);
                                         end
                                     end
                                 end
@@ -445,7 +445,7 @@ begin
             end
             else begin
                 if (obj_is_carrying_obj_pid(self_obj, 41) < restock_amt) then begin
-                    restock_obj := create_object_sid(41, 0, 0, -1);
+                    restock_obj := create_object(41, 0, 0);
                     add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 41));
                 end
                 else begin
@@ -458,17 +458,17 @@ begin
                             restock_obj := obj_carrying_pid_obj(self_obj, 41);
                             if (obj_type(self_obj) == 1) then begin
                                 restock_obj := obj_carrying_pid_obj(self_obj, 41);
-                                if (critter_inven_obj(self_obj, 0) == restock_obj) then begin
-                                    if (obj_item_subtype(critter_inven_obj(self_obj, 0)) == 0) then begin
-                                        restock_obj := critter_inven_obj(self_obj, 0);
+                                if (critter_inven_obj(self_obj, INVEN_TYPE_WORN) == restock_obj) then begin
+                                    if (obj_item_subtype(critter_inven_obj(self_obj, INVEN_TYPE_WORN)) == 0) then begin
+                                        restock_obj := critter_inven_obj(self_obj, INVEN_TYPE_WORN);
                                         debug_msg("armour pid == " + obj_pid(restock_obj));
                                         rm_obj_from_inven(self_obj, restock_obj);
                                         add_obj_to_inven(self_obj, restock_obj);
                                     end
                                 end
                                 else begin
-                                    if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                        metarule(43, self_obj);
+                                    if ((critter_inven_obj(self_obj, INVEN_TYPE_LEFT_HAND) == restock_obj) or (critter_inven_obj(self_obj, INVEN_TYPE_RIGHT_HAND) == restock_obj)) then begin
+                                        inven_unwield(self_obj);
                                     end
                                 end
                             end
@@ -483,11 +483,11 @@ begin
                     end
                 end
             end
-            restock_amt := random(1, 1);
+            restock_amt := 1;
             if (obj_is_carrying_obj_pid(self_obj, 20) < restock_amt) then begin
                 if (100 >= random(1, 100)) then begin
                     if (obj_is_carrying_obj_pid(self_obj, 20) < restock_amt) then begin
-                        restock_obj := create_object_sid(20, 0, 0, -1);
+                        restock_obj := create_object(20, 0, 0);
                         add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 20));
                     end
                     else begin
@@ -500,17 +500,17 @@ begin
                                 restock_obj := obj_carrying_pid_obj(self_obj, 20);
                                 if (obj_type(self_obj) == 1) then begin
                                     restock_obj := obj_carrying_pid_obj(self_obj, 20);
-                                    if (critter_inven_obj(self_obj, 0) == restock_obj) then begin
-                                        if (obj_item_subtype(critter_inven_obj(self_obj, 0)) == 0) then begin
-                                            restock_obj := critter_inven_obj(self_obj, 0);
+                                    if (critter_inven_obj(self_obj, INVEN_TYPE_WORN) == restock_obj) then begin
+                                        if (obj_item_subtype(critter_inven_obj(self_obj, INVEN_TYPE_WORN)) == 0) then begin
+                                            restock_obj := critter_inven_obj(self_obj, INVEN_TYPE_WORN);
                                             debug_msg("armour pid == " + obj_pid(restock_obj));
                                             rm_obj_from_inven(self_obj, restock_obj);
                                             add_obj_to_inven(self_obj, restock_obj);
                                         end
                                     end
                                     else begin
-                                        if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                            metarule(43, self_obj);
+                                        if ((critter_inven_obj(self_obj, INVEN_TYPE_LEFT_HAND) == restock_obj) or (critter_inven_obj(self_obj, INVEN_TYPE_RIGHT_HAND) == restock_obj)) then begin
+                                            inven_unwield(self_obj);
                                         end
                                     end
                                 end
@@ -528,7 +528,7 @@ begin
             end
             else begin
                 if (obj_is_carrying_obj_pid(self_obj, 20) < restock_amt) then begin
-                    restock_obj := create_object_sid(20, 0, 0, -1);
+                    restock_obj := create_object(20, 0, 0);
                     add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 20));
                 end
                 else begin
@@ -541,17 +541,17 @@ begin
                             restock_obj := obj_carrying_pid_obj(self_obj, 20);
                             if (obj_type(self_obj) == 1) then begin
                                 restock_obj := obj_carrying_pid_obj(self_obj, 20);
-                                if (critter_inven_obj(self_obj, 0) == restock_obj) then begin
-                                    if (obj_item_subtype(critter_inven_obj(self_obj, 0)) == 0) then begin
-                                        restock_obj := critter_inven_obj(self_obj, 0);
+                                if (critter_inven_obj(self_obj, INVEN_TYPE_WORN) == restock_obj) then begin
+                                    if (obj_item_subtype(critter_inven_obj(self_obj, INVEN_TYPE_WORN)) == 0) then begin
+                                        restock_obj := critter_inven_obj(self_obj, INVEN_TYPE_WORN);
                                         debug_msg("armour pid == " + obj_pid(restock_obj));
                                         rm_obj_from_inven(self_obj, restock_obj);
                                         add_obj_to_inven(self_obj, restock_obj);
                                     end
                                 end
                                 else begin
-                                    if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                        metarule(43, self_obj);
+                                    if ((critter_inven_obj(self_obj, INVEN_TYPE_LEFT_HAND) == restock_obj) or (critter_inven_obj(self_obj, INVEN_TYPE_RIGHT_HAND) == restock_obj)) then begin
+                                        inven_unwield(self_obj);
                                     end
                                 end
                             end
@@ -566,11 +566,11 @@ begin
                     end
                 end
             end
-            restock_amt := random(1, 1);
+            restock_amt := 1;
             if (obj_is_carrying_obj_pid(self_obj, 6) < restock_amt) then begin
                 if (100 >= random(1, 100)) then begin
                     if (obj_is_carrying_obj_pid(self_obj, 6) < restock_amt) then begin
-                        restock_obj := create_object_sid(6, 0, 0, -1);
+                        restock_obj := create_object(6, 0, 0);
                         add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 6));
                     end
                     else begin
@@ -583,17 +583,17 @@ begin
                                 restock_obj := obj_carrying_pid_obj(self_obj, 6);
                                 if (obj_type(self_obj) == 1) then begin
                                     restock_obj := obj_carrying_pid_obj(self_obj, 6);
-                                    if (critter_inven_obj(self_obj, 0) == restock_obj) then begin
-                                        if (obj_item_subtype(critter_inven_obj(self_obj, 0)) == 0) then begin
-                                            restock_obj := critter_inven_obj(self_obj, 0);
+                                    if (critter_inven_obj(self_obj, INVEN_TYPE_WORN) == restock_obj) then begin
+                                        if (obj_item_subtype(critter_inven_obj(self_obj, INVEN_TYPE_WORN)) == 0) then begin
+                                            restock_obj := critter_inven_obj(self_obj, INVEN_TYPE_WORN);
                                             debug_msg("armour pid == " + obj_pid(restock_obj));
                                             rm_obj_from_inven(self_obj, restock_obj);
                                             add_obj_to_inven(self_obj, restock_obj);
                                         end
                                     end
                                     else begin
-                                        if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                            metarule(43, self_obj);
+                                        if ((critter_inven_obj(self_obj, INVEN_TYPE_LEFT_HAND) == restock_obj) or (critter_inven_obj(self_obj, INVEN_TYPE_RIGHT_HAND) == restock_obj)) then begin
+                                            inven_unwield(self_obj);
                                         end
                                     end
                                 end
@@ -611,7 +611,7 @@ begin
             end
             else begin
                 if (obj_is_carrying_obj_pid(self_obj, 6) < restock_amt) then begin
-                    restock_obj := create_object_sid(6, 0, 0, -1);
+                    restock_obj := create_object(6, 0, 0);
                     add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 6));
                 end
                 else begin
@@ -624,17 +624,17 @@ begin
                             restock_obj := obj_carrying_pid_obj(self_obj, 6);
                             if (obj_type(self_obj) == 1) then begin
                                 restock_obj := obj_carrying_pid_obj(self_obj, 6);
-                                if (critter_inven_obj(self_obj, 0) == restock_obj) then begin
-                                    if (obj_item_subtype(critter_inven_obj(self_obj, 0)) == 0) then begin
-                                        restock_obj := critter_inven_obj(self_obj, 0);
+                                if (critter_inven_obj(self_obj, INVEN_TYPE_WORN) == restock_obj) then begin
+                                    if (obj_item_subtype(critter_inven_obj(self_obj, INVEN_TYPE_WORN)) == 0) then begin
+                                        restock_obj := critter_inven_obj(self_obj, INVEN_TYPE_WORN);
                                         debug_msg("armour pid == " + obj_pid(restock_obj));
                                         rm_obj_from_inven(self_obj, restock_obj);
                                         add_obj_to_inven(self_obj, restock_obj);
                                     end
                                 end
                                 else begin
-                                    if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                        metarule(43, self_obj);
+                                    if ((critter_inven_obj(self_obj, INVEN_TYPE_LEFT_HAND) == restock_obj) or (critter_inven_obj(self_obj, INVEN_TYPE_RIGHT_HAND) == restock_obj)) then begin
+                                        inven_unwield(self_obj);
                                     end
                                 end
                             end
@@ -649,11 +649,11 @@ begin
                     end
                 end
             end
-            restock_amt := random(1, 1);
+            restock_amt := 1;
             if (obj_is_carrying_obj_pid(self_obj, 75) < restock_amt) then begin
                 if (100 >= random(1, 100)) then begin
                     if (obj_is_carrying_obj_pid(self_obj, 75) < restock_amt) then begin
-                        restock_obj := create_object_sid(75, 0, 0, -1);
+                        restock_obj := create_object(75, 0, 0);
                         add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 75));
                     end
                     else begin
@@ -666,17 +666,17 @@ begin
                                 restock_obj := obj_carrying_pid_obj(self_obj, 75);
                                 if (obj_type(self_obj) == 1) then begin
                                     restock_obj := obj_carrying_pid_obj(self_obj, 75);
-                                    if (critter_inven_obj(self_obj, 0) == restock_obj) then begin
-                                        if (obj_item_subtype(critter_inven_obj(self_obj, 0)) == 0) then begin
-                                            restock_obj := critter_inven_obj(self_obj, 0);
+                                    if (critter_inven_obj(self_obj, INVEN_TYPE_WORN) == restock_obj) then begin
+                                        if (obj_item_subtype(critter_inven_obj(self_obj, INVEN_TYPE_WORN)) == 0) then begin
+                                            restock_obj := critter_inven_obj(self_obj, INVEN_TYPE_WORN);
                                             debug_msg("armour pid == " + obj_pid(restock_obj));
                                             rm_obj_from_inven(self_obj, restock_obj);
                                             add_obj_to_inven(self_obj, restock_obj);
                                         end
                                     end
                                     else begin
-                                        if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                            metarule(43, self_obj);
+                                        if ((critter_inven_obj(self_obj, INVEN_TYPE_LEFT_HAND) == restock_obj) or (critter_inven_obj(self_obj, INVEN_TYPE_RIGHT_HAND) == restock_obj)) then begin
+                                            inven_unwield(self_obj);
                                         end
                                     end
                                 end
@@ -694,7 +694,7 @@ begin
             end
             else begin
                 if (obj_is_carrying_obj_pid(self_obj, 75) < restock_amt) then begin
-                    restock_obj := create_object_sid(75, 0, 0, -1);
+                    restock_obj := create_object(75, 0, 0);
                     add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 75));
                 end
                 else begin
@@ -707,17 +707,17 @@ begin
                             restock_obj := obj_carrying_pid_obj(self_obj, 75);
                             if (obj_type(self_obj) == 1) then begin
                                 restock_obj := obj_carrying_pid_obj(self_obj, 75);
-                                if (critter_inven_obj(self_obj, 0) == restock_obj) then begin
-                                    if (obj_item_subtype(critter_inven_obj(self_obj, 0)) == 0) then begin
-                                        restock_obj := critter_inven_obj(self_obj, 0);
+                                if (critter_inven_obj(self_obj, INVEN_TYPE_WORN) == restock_obj) then begin
+                                    if (obj_item_subtype(critter_inven_obj(self_obj, INVEN_TYPE_WORN)) == 0) then begin
+                                        restock_obj := critter_inven_obj(self_obj, INVEN_TYPE_WORN);
                                         debug_msg("armour pid == " + obj_pid(restock_obj));
                                         rm_obj_from_inven(self_obj, restock_obj);
                                         add_obj_to_inven(self_obj, restock_obj);
                                     end
                                 end
                                 else begin
-                                    if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                        metarule(43, self_obj);
+                                    if ((critter_inven_obj(self_obj, INVEN_TYPE_LEFT_HAND) == restock_obj) or (critter_inven_obj(self_obj, INVEN_TYPE_RIGHT_HAND) == restock_obj)) then begin
+                                        inven_unwield(self_obj);
                                     end
                                 end
                             end
@@ -736,7 +736,7 @@ begin
             if (obj_is_carrying_obj_pid(self_obj, 79) < restock_amt) then begin
                 if (100 >= random(1, 100)) then begin
                     if (obj_is_carrying_obj_pid(self_obj, 79) < restock_amt) then begin
-                        restock_obj := create_object_sid(79, 0, 0, -1);
+                        restock_obj := create_object(79, 0, 0);
                         add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 79));
                     end
                     else begin
@@ -749,17 +749,17 @@ begin
                                 restock_obj := obj_carrying_pid_obj(self_obj, 79);
                                 if (obj_type(self_obj) == 1) then begin
                                     restock_obj := obj_carrying_pid_obj(self_obj, 79);
-                                    if (critter_inven_obj(self_obj, 0) == restock_obj) then begin
-                                        if (obj_item_subtype(critter_inven_obj(self_obj, 0)) == 0) then begin
-                                            restock_obj := critter_inven_obj(self_obj, 0);
+                                    if (critter_inven_obj(self_obj, INVEN_TYPE_WORN) == restock_obj) then begin
+                                        if (obj_item_subtype(critter_inven_obj(self_obj, INVEN_TYPE_WORN)) == 0) then begin
+                                            restock_obj := critter_inven_obj(self_obj, INVEN_TYPE_WORN);
                                             debug_msg("armour pid == " + obj_pid(restock_obj));
                                             rm_obj_from_inven(self_obj, restock_obj);
                                             add_obj_to_inven(self_obj, restock_obj);
                                         end
                                     end
                                     else begin
-                                        if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                            metarule(43, self_obj);
+                                        if ((critter_inven_obj(self_obj, INVEN_TYPE_LEFT_HAND) == restock_obj) or (critter_inven_obj(self_obj, INVEN_TYPE_RIGHT_HAND) == restock_obj)) then begin
+                                            inven_unwield(self_obj);
                                         end
                                     end
                                 end
@@ -777,7 +777,7 @@ begin
             end
             else begin
                 if (obj_is_carrying_obj_pid(self_obj, 79) < restock_amt) then begin
-                    restock_obj := create_object_sid(79, 0, 0, -1);
+                    restock_obj := create_object(79, 0, 0);
                     add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 79));
                 end
                 else begin
@@ -790,17 +790,17 @@ begin
                             restock_obj := obj_carrying_pid_obj(self_obj, 79);
                             if (obj_type(self_obj) == 1) then begin
                                 restock_obj := obj_carrying_pid_obj(self_obj, 79);
-                                if (critter_inven_obj(self_obj, 0) == restock_obj) then begin
-                                    if (obj_item_subtype(critter_inven_obj(self_obj, 0)) == 0) then begin
-                                        restock_obj := critter_inven_obj(self_obj, 0);
+                                if (critter_inven_obj(self_obj, INVEN_TYPE_WORN) == restock_obj) then begin
+                                    if (obj_item_subtype(critter_inven_obj(self_obj, INVEN_TYPE_WORN)) == 0) then begin
+                                        restock_obj := critter_inven_obj(self_obj, INVEN_TYPE_WORN);
                                         debug_msg("armour pid == " + obj_pid(restock_obj));
                                         rm_obj_from_inven(self_obj, restock_obj);
                                         add_obj_to_inven(self_obj, restock_obj);
                                     end
                                 end
                                 else begin
-                                    if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                        metarule(43, self_obj);
+                                    if ((critter_inven_obj(self_obj, INVEN_TYPE_LEFT_HAND) == restock_obj) or (critter_inven_obj(self_obj, INVEN_TYPE_RIGHT_HAND) == restock_obj)) then begin
+                                        inven_unwield(self_obj);
                                     end
                                 end
                             end
@@ -815,11 +815,11 @@ begin
                     end
                 end
             end
-            restock_amt := random(1, 1);
+            restock_amt := 1;
             if (obj_is_carrying_obj_pid(self_obj, 4) < restock_amt) then begin
                 if (100 >= random(1, 100)) then begin
                     if (obj_is_carrying_obj_pid(self_obj, 4) < restock_amt) then begin
-                        restock_obj := create_object_sid(4, 0, 0, -1);
+                        restock_obj := create_object(4, 0, 0);
                         add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 4));
                     end
                     else begin
@@ -832,17 +832,17 @@ begin
                                 restock_obj := obj_carrying_pid_obj(self_obj, 4);
                                 if (obj_type(self_obj) == 1) then begin
                                     restock_obj := obj_carrying_pid_obj(self_obj, 4);
-                                    if (critter_inven_obj(self_obj, 0) == restock_obj) then begin
-                                        if (obj_item_subtype(critter_inven_obj(self_obj, 0)) == 0) then begin
-                                            restock_obj := critter_inven_obj(self_obj, 0);
+                                    if (critter_inven_obj(self_obj, INVEN_TYPE_WORN) == restock_obj) then begin
+                                        if (obj_item_subtype(critter_inven_obj(self_obj, INVEN_TYPE_WORN)) == 0) then begin
+                                            restock_obj := critter_inven_obj(self_obj, INVEN_TYPE_WORN);
                                             debug_msg("armour pid == " + obj_pid(restock_obj));
                                             rm_obj_from_inven(self_obj, restock_obj);
                                             add_obj_to_inven(self_obj, restock_obj);
                                         end
                                     end
                                     else begin
-                                        if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                            metarule(43, self_obj);
+                                        if ((critter_inven_obj(self_obj, INVEN_TYPE_LEFT_HAND) == restock_obj) or (critter_inven_obj(self_obj, INVEN_TYPE_RIGHT_HAND) == restock_obj)) then begin
+                                            inven_unwield(self_obj);
                                         end
                                     end
                                 end
@@ -860,7 +860,7 @@ begin
             end
             else begin
                 if (obj_is_carrying_obj_pid(self_obj, 4) < restock_amt) then begin
-                    restock_obj := create_object_sid(4, 0, 0, -1);
+                    restock_obj := create_object(4, 0, 0);
                     add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 4));
                 end
                 else begin
@@ -873,17 +873,17 @@ begin
                             restock_obj := obj_carrying_pid_obj(self_obj, 4);
                             if (obj_type(self_obj) == 1) then begin
                                 restock_obj := obj_carrying_pid_obj(self_obj, 4);
-                                if (critter_inven_obj(self_obj, 0) == restock_obj) then begin
-                                    if (obj_item_subtype(critter_inven_obj(self_obj, 0)) == 0) then begin
-                                        restock_obj := critter_inven_obj(self_obj, 0);
+                                if (critter_inven_obj(self_obj, INVEN_TYPE_WORN) == restock_obj) then begin
+                                    if (obj_item_subtype(critter_inven_obj(self_obj, INVEN_TYPE_WORN)) == 0) then begin
+                                        restock_obj := critter_inven_obj(self_obj, INVEN_TYPE_WORN);
                                         debug_msg("armour pid == " + obj_pid(restock_obj));
                                         rm_obj_from_inven(self_obj, restock_obj);
                                         add_obj_to_inven(self_obj, restock_obj);
                                     end
                                 end
                                 else begin
-                                    if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                        metarule(43, self_obj);
+                                    if ((critter_inven_obj(self_obj, INVEN_TYPE_LEFT_HAND) == restock_obj) or (critter_inven_obj(self_obj, INVEN_TYPE_RIGHT_HAND) == restock_obj)) then begin
+                                        inven_unwield(self_obj);
                                     end
                                 end
                             end
@@ -898,11 +898,11 @@ begin
                     end
                 end
             end
-            restock_amt := random(1, 1);
+            restock_amt := 1;
             if (obj_is_carrying_obj_pid(self_obj, 127) < restock_amt) then begin
                 if (50 >= random(1, 100)) then begin
                     if (obj_is_carrying_obj_pid(self_obj, 127) < restock_amt) then begin
-                        restock_obj := create_object_sid(127, 0, 0, -1);
+                        restock_obj := create_object(127, 0, 0);
                         add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 127));
                     end
                     else begin
@@ -915,17 +915,17 @@ begin
                                 restock_obj := obj_carrying_pid_obj(self_obj, 127);
                                 if (obj_type(self_obj) == 1) then begin
                                     restock_obj := obj_carrying_pid_obj(self_obj, 127);
-                                    if (critter_inven_obj(self_obj, 0) == restock_obj) then begin
-                                        if (obj_item_subtype(critter_inven_obj(self_obj, 0)) == 0) then begin
-                                            restock_obj := critter_inven_obj(self_obj, 0);
+                                    if (critter_inven_obj(self_obj, INVEN_TYPE_WORN) == restock_obj) then begin
+                                        if (obj_item_subtype(critter_inven_obj(self_obj, INVEN_TYPE_WORN)) == 0) then begin
+                                            restock_obj := critter_inven_obj(self_obj, INVEN_TYPE_WORN);
                                             debug_msg("armour pid == " + obj_pid(restock_obj));
                                             rm_obj_from_inven(self_obj, restock_obj);
                                             add_obj_to_inven(self_obj, restock_obj);
                                         end
                                     end
                                     else begin
-                                        if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                            metarule(43, self_obj);
+                                        if ((critter_inven_obj(self_obj, INVEN_TYPE_LEFT_HAND) == restock_obj) or (critter_inven_obj(self_obj, INVEN_TYPE_RIGHT_HAND) == restock_obj)) then begin
+                                            inven_unwield(self_obj);
                                         end
                                     end
                                 end
@@ -943,7 +943,7 @@ begin
             end
             else begin
                 if (obj_is_carrying_obj_pid(self_obj, 127) < restock_amt) then begin
-                    restock_obj := create_object_sid(127, 0, 0, -1);
+                    restock_obj := create_object(127, 0, 0);
                     add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 127));
                 end
                 else begin
@@ -956,17 +956,17 @@ begin
                             restock_obj := obj_carrying_pid_obj(self_obj, 127);
                             if (obj_type(self_obj) == 1) then begin
                                 restock_obj := obj_carrying_pid_obj(self_obj, 127);
-                                if (critter_inven_obj(self_obj, 0) == restock_obj) then begin
-                                    if (obj_item_subtype(critter_inven_obj(self_obj, 0)) == 0) then begin
-                                        restock_obj := critter_inven_obj(self_obj, 0);
+                                if (critter_inven_obj(self_obj, INVEN_TYPE_WORN) == restock_obj) then begin
+                                    if (obj_item_subtype(critter_inven_obj(self_obj, INVEN_TYPE_WORN)) == 0) then begin
+                                        restock_obj := critter_inven_obj(self_obj, INVEN_TYPE_WORN);
                                         debug_msg("armour pid == " + obj_pid(restock_obj));
                                         rm_obj_from_inven(self_obj, restock_obj);
                                         add_obj_to_inven(self_obj, restock_obj);
                                     end
                                 end
                                 else begin
-                                    if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                        metarule(43, self_obj);
+                                    if ((critter_inven_obj(self_obj, INVEN_TYPE_LEFT_HAND) == restock_obj) or (critter_inven_obj(self_obj, INVEN_TYPE_RIGHT_HAND) == restock_obj)) then begin
+                                        inven_unwield(self_obj);
                                     end
                                 end
                             end
@@ -981,11 +981,11 @@ begin
                     end
                 end
             end
-            restock_amt := random(1, 1);
+            restock_amt := 1;
             if (obj_is_carrying_obj_pid(self_obj, 101) < restock_amt) then begin
                 if (25 >= random(1, 100)) then begin
                     if (obj_is_carrying_obj_pid(self_obj, 101) < restock_amt) then begin
-                        restock_obj := create_object_sid(101, 0, 0, -1);
+                        restock_obj := create_object(101, 0, 0);
                         add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 101));
                     end
                     else begin
@@ -998,17 +998,17 @@ begin
                                 restock_obj := obj_carrying_pid_obj(self_obj, 101);
                                 if (obj_type(self_obj) == 1) then begin
                                     restock_obj := obj_carrying_pid_obj(self_obj, 101);
-                                    if (critter_inven_obj(self_obj, 0) == restock_obj) then begin
-                                        if (obj_item_subtype(critter_inven_obj(self_obj, 0)) == 0) then begin
-                                            restock_obj := critter_inven_obj(self_obj, 0);
+                                    if (critter_inven_obj(self_obj, INVEN_TYPE_WORN) == restock_obj) then begin
+                                        if (obj_item_subtype(critter_inven_obj(self_obj, INVEN_TYPE_WORN)) == 0) then begin
+                                            restock_obj := critter_inven_obj(self_obj, INVEN_TYPE_WORN);
                                             debug_msg("armour pid == " + obj_pid(restock_obj));
                                             rm_obj_from_inven(self_obj, restock_obj);
                                             add_obj_to_inven(self_obj, restock_obj);
                                         end
                                     end
                                     else begin
-                                        if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                            metarule(43, self_obj);
+                                        if ((critter_inven_obj(self_obj, INVEN_TYPE_LEFT_HAND) == restock_obj) or (critter_inven_obj(self_obj, INVEN_TYPE_RIGHT_HAND) == restock_obj)) then begin
+                                            inven_unwield(self_obj);
                                         end
                                     end
                                 end
@@ -1026,7 +1026,7 @@ begin
             end
             else begin
                 if (obj_is_carrying_obj_pid(self_obj, 101) < restock_amt) then begin
-                    restock_obj := create_object_sid(101, 0, 0, -1);
+                    restock_obj := create_object(101, 0, 0);
                     add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 101));
                 end
                 else begin
@@ -1039,17 +1039,17 @@ begin
                             restock_obj := obj_carrying_pid_obj(self_obj, 101);
                             if (obj_type(self_obj) == 1) then begin
                                 restock_obj := obj_carrying_pid_obj(self_obj, 101);
-                                if (critter_inven_obj(self_obj, 0) == restock_obj) then begin
-                                    if (obj_item_subtype(critter_inven_obj(self_obj, 0)) == 0) then begin
-                                        restock_obj := critter_inven_obj(self_obj, 0);
+                                if (critter_inven_obj(self_obj, INVEN_TYPE_WORN) == restock_obj) then begin
+                                    if (obj_item_subtype(critter_inven_obj(self_obj, INVEN_TYPE_WORN)) == 0) then begin
+                                        restock_obj := critter_inven_obj(self_obj, INVEN_TYPE_WORN);
                                         debug_msg("armour pid == " + obj_pid(restock_obj));
                                         rm_obj_from_inven(self_obj, restock_obj);
                                         add_obj_to_inven(self_obj, restock_obj);
                                     end
                                 end
                                 else begin
-                                    if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                        metarule(43, self_obj);
+                                    if ((critter_inven_obj(self_obj, INVEN_TYPE_LEFT_HAND) == restock_obj) or (critter_inven_obj(self_obj, INVEN_TYPE_RIGHT_HAND) == restock_obj)) then begin
+                                        inven_unwield(self_obj);
                                     end
                                 end
                             end
@@ -1085,4 +1085,3 @@ begin
         end
     end
 end
-
