diff --git a/Mods/MegaMod/Original/VIVLTDOR.ssl b/Mods/MegaMod/ReDefine/VIVLTDOR.ssl
index 3b572f9..fc893ae 100644
--- a/Mods/MegaMod/Original/VIVLTDOR.ssl
+++ b/Mods/MegaMod/ReDefine/VIVLTDOR.ssl
@@ -153,7 +153,7 @@ begin
         call Node008();
     end
     else begin
-        if (get_critter_stat(dude_obj, 0) < 8) then begin
+        if (dude_strength < 8) then begin
             call Node002();
         end
         else begin
@@ -165,13 +165,13 @@ end
 procedure look_at_p_proc
 begin
     script_overrides;
-    display_msg(message_str(982, 100));
+    display_mstr(100);
 end
 
 procedure description_p_proc
 begin
     script_overrides;
-    display_msg(message_str(982, 101));
+    display_mstr(101);
 end
 
 procedure use_skill_on_p_proc
@@ -200,7 +200,7 @@ begin
     else begin
         script_overrides;
         if ((LVar0 == 20) or (LVar0 == 412)) then begin
-            if (get_critter_stat(dude_obj, 0) > 6) then begin
+            if (dude_strength > 6) then begin
                 if (LVar0 == 20) then begin
                     call Node004();
                 end
@@ -233,7 +233,7 @@ end
 procedure damage_p_proc
 begin
     script_overrides;
-    if (metarule(49, target_obj) == 6) then begin
+    if (weapon_dmg_type(target_obj) == 6) then begin
         call Node006();
     end
 end
@@ -265,75 +265,75 @@ end
 
 procedure Node002
 begin
-    if (get_critter_stat(dude_obj, 4) < 4) then begin
-        float_msg(dude_obj, message_str(982, 102), 8);
+    if (dude_iq < 4) then begin
+        float_msg(dude_obj, mstr(102), FLOAT_MSG_YELLOW);
     end
     else begin
-        float_msg(dude_obj, message_str(982, 103), 8);
+        float_msg(dude_obj, mstr(103), FLOAT_MSG_YELLOW);
     end
 end
 
 procedure Node003
 begin
-    float_msg(dude_obj, message_str(982, 104), 8);
-    display_msg(message_str(982, 105));
+    float_msg(dude_obj, mstr(104), FLOAT_MSG_YELLOW);
+    display_mstr(105);
 end
 
 procedure Node004
 begin
-    float_msg(dude_obj, message_str(982, 106), 8);
-    display_msg(message_str(982, 107));
+    float_msg(dude_obj, mstr(106), FLOAT_MSG_YELLOW);
+    display_mstr(107);
     call NodeOpen();
 end
 
 procedure Node005
 begin
-    float_msg(dude_obj, message_str(982, 108), 8);
-    display_msg(message_str(982, 109));
+    float_msg(dude_obj, mstr(108), FLOAT_MSG_YELLOW);
+    display_mstr(109);
     call NodeOpen();
 end
 
 procedure Node006
 begin
-    if (get_critter_stat(dude_obj, 4) < 4) then begin
-        float_msg(dude_obj, message_str(982, 111), 8);
+    if (dude_iq < 4) then begin
+        float_msg(dude_obj, mstr(111), FLOAT_MSG_YELLOW);
     end
     else begin
-        float_msg(dude_obj, message_str(982, 110), 8);
+        float_msg(dude_obj, mstr(110), FLOAT_MSG_YELLOW);
     end
 end
 
 procedure Node007
 begin
-    if (get_critter_stat(dude_obj, 4) < 4) then begin
-        float_msg(dude_obj, message_str(982, 113), 8);
+    if (dude_iq < 4) then begin
+        float_msg(dude_obj, mstr(113), FLOAT_MSG_YELLOW);
     end
     else begin
-        float_msg(dude_obj, message_str(982, 112), 8);
+        float_msg(dude_obj, mstr(112), FLOAT_MSG_YELLOW);
     end
 end
 
 procedure Node008
 begin
-    if (get_critter_stat(dude_obj, 4) < 4) then begin
-        float_msg(dude_obj, message_str(982, 115), 8);
+    if (dude_iq < 4) then begin
+        float_msg(dude_obj, mstr(115), FLOAT_MSG_YELLOW);
     end
     else begin
-        float_msg(dude_obj, message_str(982, 114), 8);
+        float_msg(dude_obj, mstr(114), FLOAT_MSG_YELLOW);
     end
 end
 
 procedure Node009
 begin
-    float_msg(dude_obj, message_str(982, 106), 8);
-    display_msg(message_str(982, 117));
+    float_msg(dude_obj, mstr(106), FLOAT_MSG_YELLOW);
+    display_mstr(117);
     call NodeOpen();
 end
 
 procedure Node010
 begin
-    float_msg(dude_obj, message_str(982, 104), 8);
-    display_msg(message_str(982, 116));
+    float_msg(dude_obj, mstr(104), FLOAT_MSG_YELLOW);
+    display_mstr(116);
 end
 
 procedure NodeOpen
@@ -341,4 +341,3 @@ begin
     set_local_var(0, 1);
     add_timer_event(self_obj, game_ticks(2), 1);
 end
-
