diff --git a/Mods/MegaMod/Original/WCBRNBOT.ssl b/Mods/MegaMod/ReDefine/WCBRNBOT.ssl
index 310e658..7fc5710 100644
--- a/Mods/MegaMod/Original/WCBRNBOT.ssl
+++ b/Mods/MegaMod/ReDefine/WCBRNBOT.ssl
@@ -65,22 +65,22 @@ begin
     variable LVar5 := 0;
     if (local_var(15) == 0) then begin
         set_local_var(15, 1);
-        critter_add_trait(self_obj, 1, 6, 52);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 52);
         add_timer_event(self_obj, game_ticks(75), 321);
-        if (global_var(765) == 0) then begin
-            set_global_var(765, 1);
+        if (global_var(GVAR_V23_QUEST) == 0) then begin
+            set_global_var(GVAR_V23_QUEST, 1);
         end
         if (proto_data(obj_pid(self_obj), 3) == 16777282) then begin
-            if (global_var(155) > 1) then begin
-                LVar0 := create_object_sid(274, 0, 0, -1);
-                LVar1 := create_object_sid(36, 0, 0, -1);
+            if (global_var(GVAR_BRAIN_BOT_BRAIN) > 1) then begin
+                LVar0 := create_object(274, 0, 0);
+                LVar1 := create_object(36, 0, 0);
                 LVar2 := random(2, 3);
                 LVar3 := random(1, 3);
                 add_mult_objs_to_inven(self_obj, LVar0, LVar2);
                 add_mult_objs_to_inven(self_obj, LVar1, LVar3);
             end
-            LVar4 := create_object_sid(12, 0, 0, -1);
-            LVar5 := create_object_sid(270, 0, 0, -1);
+            LVar4 := create_object(12, 0, 0);
+            LVar5 := create_object(270, 0, 0);
             add_mult_objs_to_inven(self_obj, LVar4, 1);
             add_mult_objs_to_inven(self_obj, LVar5, 1);
         end
@@ -96,14 +96,14 @@ end
 
 procedure critter_p_proc
 begin
-    if ((cur_map_index == 54) and (self_obj == party_member_obj(16777295)) and (map_var(2) == party_member_obj(16777295))) then begin
+    if ((CUR_MAP_NEWR1) and (self_obj == party_member_obj(16777295)) and (map_var(2) == party_member_obj(16777295))) then begin
         set_map_var(2, 0);
         waiting := 1;
-        if (get_critter_stat(self_obj, 0) < 10) then begin
-            use_obj_on_obj(create_object_sid(482, 0, 0, -1), self_obj);
+        if (self_strength < 10) then begin
+            use_obj_on_obj(create_object(482, 0, 0), self_obj);
         end
     end
-    if ((cur_map_index == 27) and (global_var(152) == 1) and (local_var(5) == 0)) then begin
+    if ((CUR_MAP_DEPOLV1) and (global_var(GVAR_SIERRA_BASE_ENEMY) == 1) and (local_var(5) == 0)) then begin
         if (local_var(13) == 0) then begin
             set_local_var(13, local_var(13) + 1);
             call Kill_The_Party();
@@ -111,26 +111,26 @@ begin
     end
     if (local_var(5) == 2) then begin
         if (obj_can_see_obj(self_obj, dude_obj)) then begin
-            float_msg(self_obj, message_str(167, 173), 8);
-            attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+            float_msg(self_obj, mstr(173), FLOAT_MSG_YELLOW);
+            attack(dude_obj);
         end
         else begin
-            animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(dude_obj), random(0, 5), random(1, 4)), 0);
+            animate_move_obj_to_tile(self_obj, tile_num_in_direction(dude_tile, random(0, 5), random(1, 4)), 0);
         end
     end
-    if ((cur_map_index == 28) and ((local_var(9) == 0) or (local_var(10) == 0) or (local_var(11) == 0))) then begin
+    if ((CUR_MAP_DEPOLVA) and ((local_var(9) == 0) or (local_var(10) == 0) or (local_var(11) == 0))) then begin
         if (local_var(5) != 2) then begin
             call Node024();
         end
     end
     if (waiting != 1) then begin
-        if ((local_var(8) == 0) and not(global_var(398) != 0)) then begin
+        if ((local_var(8) == 0) and not(global_var(GVAR_PARTY_NO_FOLLOW) != 0)) then begin
             if (tile_distance_objs(self_obj, dude_obj) > (2 * local_var(7))) then begin
-                animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(dude_obj), random(0, 5), 3), 1);
+                animate_move_obj_to_tile(self_obj, tile_num_in_direction(dude_tile, random(0, 5), 3), 1);
             end
             else begin
                 if (tile_distance_objs(self_obj, dude_obj) > local_var(7)) then begin
-                    animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(dude_obj), random(0, 5), 6), 1);
+                    animate_move_obj_to_tile(self_obj, tile_num_in_direction(dude_tile, random(0, 5), 6), 1);
                 end
             end
         end
@@ -145,7 +145,7 @@ end
 
 procedure pickup_p_proc
 begin
-    if ((source_obj == dude_obj) and (party_member_obj(obj_pid(self_obj)) != 0)) then begin
+    if ((source_obj == dude_obj) and (obj_in_party(self_obj))) then begin
         set_local_var(5, 2);
     end
 end
@@ -176,7 +176,7 @@ begin
                 end_dialogue;
             end
             else begin
-                if (global_var(155) == 1) then begin
+                if (global_var(GVAR_BRAIN_BOT_BRAIN) == 1) then begin
                     start_gdialog(167, self_obj, 4, -1, -1);
                     gsay_start;
                     call Node002();
@@ -184,7 +184,7 @@ begin
                     end_dialogue;
                 end
                 else begin
-                    if (global_var(155) == 2) then begin
+                    if (global_var(GVAR_BRAIN_BOT_BRAIN) == 2) then begin
                         start_gdialog(167, self_obj, 4, -1, -1);
                         gsay_start;
                         call Node003();
@@ -192,7 +192,7 @@ begin
                         end_dialogue;
                     end
                     else begin
-                        if (global_var(155) == 3) then begin
+                        if (global_var(GVAR_BRAIN_BOT_BRAIN) == 3) then begin
                             start_gdialog(167, self_obj, 4, -1, -1);
                             gsay_start;
                             call Node004();
@@ -200,7 +200,7 @@ begin
                             end_dialogue;
                         end
                         else begin
-                            if ((global_var(155) == 4) and (global_var(152) == 1) and (local_var(5) == 0)) then begin
+                            if ((global_var(GVAR_BRAIN_BOT_BRAIN) == 4) and (global_var(GVAR_SIERRA_BASE_ENEMY) == 1) and (local_var(5) == 0)) then begin
                                 start_gdialog(167, self_obj, 4, -1, -1);
                                 gsay_start;
                                 call Node005();
@@ -208,7 +208,7 @@ begin
                                 end_dialogue;
                             end
                             else begin
-                                if (global_var(155) == 4) then begin
+                                if (global_var(GVAR_BRAIN_BOT_BRAIN) == 4) then begin
                                     start_gdialog(167, self_obj, 4, -1, -1);
                                     gsay_start;
                                     call Node006();
@@ -227,63 +227,63 @@ end
 procedure destroy_p_proc
 begin
     if (source_obj == dude_obj) then begin
-        set_global_var(5, global_var(5) + 1);
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(4) > (3 * global_var(5))) or (global_var(2) == 1)) and (global_var(1) == 0)) then begin
-            set_global_var(2, 1);
-            set_global_var(3, 0);
+        set_global_var(GVAR_GOOD_MONSTER, global_var(GVAR_GOOD_MONSTER) + 1);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(4) > (3 * global_var(GVAR_GOOD_MONSTER))) or (global_var(GVAR_CHAMPION_REPUTATION) == 1)) and (global_var(GVAR_CHILDKILLER_REPUTATION) == 0)) then begin
+            set_global_var(GVAR_CHAMPION_REPUTATION, 1);
+            set_global_var(GVAR_BERSERKER_REPUTATION, 0);
         end
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(5) > (2 * global_var(4))) or (global_var(3) == 1))) then begin
-            set_global_var(3, 1);
-            set_global_var(2, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
+            set_global_var(GVAR_BERSERKER_REPUTATION, 1);
+            set_global_var(GVAR_CHAMPION_REPUTATION, 0);
         end
-        set_global_var(0, global_var(0) + -10);
+        set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + -10);
         debug_msg("Player gains " + -10 + " Karma Points.");
-        set_global_var(37, 0);
-        set_global_var(38, 0);
-        set_global_var(39, 0);
-        set_global_var(40, 0);
-        set_global_var(41, 0);
-        set_global_var(42, 0);
-        set_global_var(43, 0);
-        set_global_var(44, 0);
-        set_global_var(45, 0);
-        tmp_gen_rep := global_var(0);
-        if (has_trait(0, dude_obj, 95)) then begin
-            tmp_gen_rep := global_var(0) * 2;
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+        set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+        set_global_var(GVAR_KARMA_DEFENDER, 0);
+        set_global_var(GVAR_KARMA_WANDERER, 0);
+        set_global_var(GVAR_KARMA_BETRAYER, 0);
+        set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+        set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+        tmp_gen_rep := global_var(GVAR_PLAYER_REPUTATION);
+        if (has_trait(TRAIT_PERK, dude_obj, PERK_karma_beacon_perk)) then begin
+            tmp_gen_rep := global_var(GVAR_PLAYER_REPUTATION) * 2;
         end
         if (tmp_gen_rep >= 1000) then begin
-            set_global_var(37, 1);
+            set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
         end
         else begin
             if (tmp_gen_rep >= 750) then begin
-                set_global_var(38, 1);
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
             end
             else begin
                 if (tmp_gen_rep >= 500) then begin
-                    set_global_var(39, 1);
+                    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                 end
                 else begin
                     if (tmp_gen_rep >= 250) then begin
-                        set_global_var(40, 1);
+                        set_global_var(GVAR_KARMA_DEFENDER, 1);
                     end
                     else begin
                         if (tmp_gen_rep > -250) then begin
-                            set_global_var(41, 1);
+                            set_global_var(GVAR_KARMA_WANDERER, 1);
                         end
                         else begin
                             if (tmp_gen_rep > -500) then begin
-                                set_global_var(42, 1);
+                                set_global_var(GVAR_KARMA_BETRAYER, 1);
                             end
                             else begin
                                 if (tmp_gen_rep > -750) then begin
-                                    set_global_var(43, 1);
+                                    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                 end
                                 else begin
                                     if (tmp_gen_rep > -1000) then begin
-                                        set_global_var(44, 1);
+                                        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                     end
                                     else begin
-                                        set_global_var(45, 1);
+                                        set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                     end
                                 end
                             end
@@ -298,20 +298,20 @@ end
 procedure look_at_p_proc
 begin
     script_overrides;
-    display_msg(message_str(167, 1000));
+    display_mstr(1000);
 end
 
 procedure description_p_proc
 begin
     script_overrides;
-    display_msg(message_str(167, 101));
+    display_mstr(101);
 end
 
 procedure use_skill_on_p_proc
 begin
     if (action_being_used == 13) then begin
         script_overrides;
-        display_msg(message_str(167, 610));
+        display_mstr(610);
     end
 end
 
@@ -325,10 +325,10 @@ end
 procedure map_update_p_proc
 begin
     if ((local_var(12) == 0) or (local_var(8) != 0)) then begin
-        critter_add_trait(self_obj, 1, 6, 52);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 52);
     end
     else begin
-        critter_add_trait(self_obj, 1, 6, 0);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 0);
     end
     if (waiting == 1) then begin
         waiting := 0;
@@ -337,15 +337,15 @@ end
 
 procedure map_enter_p_proc
 begin
-    if ((cur_map_index == 19) or (cur_map_index == 37) or (cur_map_index == 38)) then begin
-        if ((elevation(self_obj) != elevation(dude_obj)) and (local_var(8) != 0)) then begin
-            move_to(self_obj, tile_num(self_obj), elevation(dude_obj));
+    if ((CUR_MAP_MODINN) or (CUR_MAP_V15ENT) or (CUR_MAP_V15ENTB)) then begin
+        if ((elevation(self_obj) != dude_elevation) and (local_var(8) != 0)) then begin
+            move_to(self_obj, tile_num(self_obj), dude_elevation);
         end
     end
-    if (global_var(155) == 4) then begin
-        if (party_member_obj(obj_pid(self_obj)) != 0) then begin
+    if (global_var(GVAR_BRAIN_BOT_BRAIN) == 4) then begin
+        if (obj_in_party(self_obj)) then begin
             float_now := 1;
-            metarule3(100, self_obj, 321, 0);
+            rm_fixed_timer_event(self_obj, 321);
             add_timer_event(self_obj, game_ticks(random(40, 60)), 321);
         end
         else begin
@@ -360,13 +360,13 @@ begin
         call Kill_The_Party();
     end
     if (fixed_param == 321) then begin
-        if (not(combat_is_initialized) and has_trait(1, self_obj, 666)) then begin
-            if (party_member_obj(obj_pid(self_obj)) != 0) then begin
-                float_msg(self_obj, message_str(167, random(534, 582)), 8);
+        if (not(combat_is_initialized) and obj_is_visible_flag(self_obj)) then begin
+            if (obj_in_party(self_obj)) then begin
+                float_msg(self_obj, mstr(random(534, 582)), FLOAT_MSG_YELLOW);
             end
         end
-        if (global_var(155) == 4) then begin
-            metarule3(100, self_obj, 321, 0);
+        if (global_var(GVAR_BRAIN_BOT_BRAIN) == 4) then begin
+            rm_fixed_timer_event(self_obj, 321);
             add_timer_event(self_obj, game_ticks(random(40, 60)), 321);
         end
     end
@@ -380,15 +380,15 @@ end
 
 procedure push_p_proc
 begin
-    if ((global_var(155) == 1) or (global_var(155) == 2)) then begin
-        float_msg(self_obj, message_str(167, 410), 8);
+    if ((global_var(GVAR_BRAIN_BOT_BRAIN) == 1) or (global_var(GVAR_BRAIN_BOT_BRAIN) == 2)) then begin
+        float_msg(self_obj, mstr(410), FLOAT_MSG_YELLOW);
     end
     else begin
-        if (global_var(155) == 3) then begin
-            float_msg(self_obj, message_str(167, 420), 8);
+        if (global_var(GVAR_BRAIN_BOT_BRAIN) == 3) then begin
+            float_msg(self_obj, mstr(420), FLOAT_MSG_YELLOW);
         end
         else begin
-            float_msg(self_obj, message_str(167, random(430, 432)), 8);
+            float_msg(self_obj, mstr(random(430, 432)), FLOAT_MSG_YELLOW);
         end
     end
 end
@@ -399,7 +399,7 @@ begin
     variable LVar1 := 0;
     if ((obj_pid(obj_being_used_with) == 39) or (obj_pid(obj_being_used_with) == 38)) then begin
         script_overrides;
-        display_msg(message_str(167, 600));
+        display_mstr(600);
     end
     if ((obj_pid(obj_being_used_with) == 14) or (obj_pid(obj_being_used_with) == 37) or (obj_pid(obj_being_used_with) == 35) or (obj_pid(obj_being_used_with) == 36)) then begin
         if (proto_data(obj_pid(self_obj), 3) == 16777282) then begin
@@ -407,45 +407,45 @@ begin
             rm_obj_from_inven(dude_obj, obj_being_used_with);
             destroy_object(obj_being_used_with);
             if ((obj_pid(obj_being_used_with) == 14) or (obj_pid(obj_being_used_with) == 37)) then begin
-                LVar0 := create_object_sid(274, 0, 0, -1);
+                LVar0 := create_object(274, 0, 0);
                 add_mult_objs_to_inven(self_obj, LVar0, 1);
-                display_msg(message_str(167, 216));
+                display_mstr(216);
             end
             if ((obj_pid(obj_being_used_with) == 35) or (obj_pid(obj_being_used_with) == 36)) then begin
-                LVar1 := create_object_sid(36, 0, 0, -1);
+                LVar1 := create_object(36, 0, 0);
                 add_mult_objs_to_inven(self_obj, LVar1, 1);
-                display_msg(message_str(167, 217));
+                display_mstr(217);
             end
         end
     end
     if (obj_pid(obj_being_used_with) == 412) then begin
         script_overrides;
-        if (global_var(155) > 3) then begin
-            float_msg(self_obj, message_str(167, 601), 3);
+        if (global_var(GVAR_BRAIN_BOT_BRAIN) > 3) then begin
+            float_msg(self_obj, mstr(601), FLOAT_MSG_GREEN);
         end
-        if (global_var(155) < 3) then begin
-            float_msg(self_obj, message_str(167, 603), 3);
+        if (global_var(GVAR_BRAIN_BOT_BRAIN) < 3) then begin
+            float_msg(self_obj, mstr(603), FLOAT_MSG_GREEN);
         end
-        if (global_var(155) == 3) then begin
-            float_msg(self_obj, message_str(167, 604), 3);
+        if (global_var(GVAR_BRAIN_BOT_BRAIN) == 3) then begin
+            float_msg(self_obj, mstr(604), FLOAT_MSG_GREEN);
         end
-        display_msg(message_str(167, 602));
+        display_mstr(602);
     end
     if ((obj_pid(obj_being_used_with) == 40) or (obj_pid(obj_being_used_with) == 47) or (obj_pid(obj_being_used_with) == 71) or (obj_pid(obj_being_used_with) == 91) or (obj_pid(obj_being_used_with) == 144) or (obj_pid(obj_being_used_with) == 259) or (obj_pid(obj_being_used_with) == 87) or (obj_pid(obj_being_used_with) == 110) or (obj_pid(obj_being_used_with) == 273)) then begin
         script_overrides;
-        display_msg(message_str(167, 611));
+        display_mstr(611);
     end
     if ((obj_pid(obj_being_used_with) == 75) or (obj_pid(obj_being_used_with) == 308)) then begin
         script_overrides;
-        display_msg(message_str(167, 610));
+        display_mstr(610);
     end
     if ((obj_pid(obj_being_used_with) == 81) or (obj_pid(obj_being_used_with) == 103) or (obj_pid(obj_being_used_with) == 284) or (obj_pid(obj_being_used_with) == 378) or (obj_pid(obj_being_used_with) == 617)) then begin
         script_overrides;
-        display_msg(message_str(167, 613));
+        display_mstr(613);
     end
     if ((obj_pid(obj_being_used_with) == 106) or (obj_pid(obj_being_used_with) == 124) or (obj_pid(obj_being_used_with) == 125) or (obj_pid(obj_being_used_with) == 310) or (obj_pid(obj_being_used_with) == 311) or (obj_pid(obj_being_used_with) == 469) or (obj_pid(obj_being_used_with) == 662)) then begin
         script_overrides;
-        display_msg(message_str(167, 612));
+        display_mstr(612);
     end
     return 0;
     LVar0 := 0;
@@ -463,27 +463,27 @@ end
 
 procedure Node001
 begin
-    if ((global_var(152) == 1) and (global_var(155) == 4)) then begin
-        gsay_reply(167, 102);
-        giq_option(1, 167, 109, Node001b, 50);
+    if ((global_var(GVAR_SIERRA_BASE_ENEMY) == 1) and (global_var(GVAR_BRAIN_BOT_BRAIN) == 4)) then begin
+        Reply(102);
+        giq_option(1, NAME, 109, Node001b, NEUTRAL_REACTION);
     end
     else begin
-        if (global_var(155) == 4) then begin
-            gsay_reply(167, message_str(167, 103) + message_str(167, 104) + message_str(167, 105) + message_str(167, 106));
+        if (global_var(GVAR_BRAIN_BOT_BRAIN) == 4) then begin
+            Reply(mstr(103) + mstr(104) + mstr(105) + mstr(106));
         end
         else begin
-            gsay_reply(167, 306);
+            Reply(306);
         end
-        giq_option(4, 167, 107, Node001a, 50);
-        giq_option(4, 167, 108, Node011, 50);
-        giq_option(-3, 167, 110, Node999, 50);
+        giq_option(4, NAME, 107, Node001a, NEUTRAL_REACTION);
+        giq_option(4, NAME, 108, Node011, NEUTRAL_REACTION);
+        giq_option(-3, NAME, 110, Node999, NEUTRAL_REACTION);
     end
 end
 
 procedure Node001a
 begin
     debug_msg("Checking Party Size");
-    if (((metarule(16, 0) - 1) >= (floor(get_critter_stat(dude_obj, 3) / 2) + has_trait(0, dude_obj, 98))) or ((metarule(16, 0) - 1) >= 5)) then begin
+    if (((party_member_count(0) - 1) >= (floor(dude_charisma / 2) + has_trait(TRAIT_PERK, dude_obj, PERK_magnetic_personality_perk))) or ((party_member_count(0) - 1) >= 5)) then begin
         call Node012();
     end
     else begin
@@ -504,55 +504,55 @@ end
 
 procedure Node002
 begin
-    gsay_reply(167, 111);
-    giq_option(4, 167, 114, Node999, 50);
-    giq_option(-3, 167, 115, Node999, 50);
+    Reply(111);
+    giq_option(4, NAME, 114, Node999, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 115, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node003
 begin
-    gsay_reply(167, 116);
-    giq_option(4, 167, 118, Node021, 50);
-    giq_option(4, 167, 120, Node999, 50);
-    giq_option(-3, 167, 121, Node999, 50);
+    Reply(116);
+    giq_option(4, NAME, 118, Node021, NEUTRAL_REACTION);
+    giq_option(4, NAME, 120, Node999, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 121, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node004
 begin
-    gsay_reply(167, 122);
-    giq_option(4, 167, 124, Node021, 50);
-    giq_option(4, 167, 125, Node022, 50);
-    giq_option(4, 167, 126, Node013, 50);
-    giq_option(4, 167, 128, Node999, 50);
-    giq_option(-3, 167, 129, Node999, 50);
+    Reply(122);
+    giq_option(4, NAME, 124, Node021, NEUTRAL_REACTION);
+    giq_option(4, NAME, 125, Node022, NEUTRAL_REACTION);
+    giq_option(4, NAME, 126, Node013, NEUTRAL_REACTION);
+    giq_option(4, NAME, 128, Node999, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 129, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node005
 begin
-    gsay_reply(167, message_str(167, 130) + message_str(167, 131));
-    giq_option(4, 167, 132, Node999, 50);
-    giq_option(-3, 167, 133, Node999, 50);
+    Reply(mstr(130) + mstr(131));
+    giq_option(4, NAME, 132, Node999, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 133, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node006
 begin
-    gsay_reply(167, 134);
-    giq_option(4, 167, 136, Node021, 50);
-    giq_option(4, 167, 137, Node022, 50);
-    giq_option(4, 167, 138, Node017, 50);
-    if (get_critter_stat(self_obj, 7) != get_critter_stat(self_obj, 35)) then begin
-        giq_option(4, 167, 140, Node006a, 50);
-    end
-    giq_option(4, 167, 141, Node020, 50);
-    giq_option(4, 167, 142, Node019, 50);
-    giq_option(4, 167, 143, Node999, 50);
-    giq_option(-3, 167, 144, Node999, 50);
+    Reply(134);
+    giq_option(4, NAME, 136, Node021, NEUTRAL_REACTION);
+    giq_option(4, NAME, 137, Node022, NEUTRAL_REACTION);
+    giq_option(4, NAME, 138, Node017, NEUTRAL_REACTION);
+    if (get_critter_stat(self_obj, STAT_max_hp) != get_critter_stat(self_obj, STAT_current_hp)) then begin
+        giq_option(4, NAME, 140, Node006a, NEUTRAL_REACTION);
+    end
+    giq_option(4, NAME, 141, Node020, NEUTRAL_REACTION);
+    giq_option(4, NAME, 142, Node019, NEUTRAL_REACTION);
+    giq_option(4, NAME, 143, Node999, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 144, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node006a
 begin
     variable LVar0 := 0;
-    LVar0 := get_critter_stat(self_obj, 7) - get_critter_stat(self_obj, 35);
+    LVar0 := get_critter_stat(self_obj, STAT_max_hp) - get_critter_stat(self_obj, STAT_current_hp);
     critter_heal(self_obj, LVar0);
     gfade_out(600);
     gfade_in(600);
@@ -564,96 +564,96 @@ end
 
 procedure Node007
 begin
-    if (global_var(155) == 4) then begin
-        gsay_reply(167, message_str(167, 145) + message_str(167, 147));
+    if (global_var(GVAR_BRAIN_BOT_BRAIN) == 4) then begin
+        Reply(mstr(145) + mstr(147));
     end
     else begin
-        gsay_reply(167, message_str(167, 146) + message_str(167, 147));
+        Reply(mstr(146) + mstr(147));
     end
-    giq_option(4, 167, 148, Node010, 50);
-    giq_option(4, 167, 149, Node021, 50);
-    giq_option(-3, 167, 144, Node999, 50);
+    giq_option(4, NAME, 148, Node010, NEUTRAL_REACTION);
+    giq_option(4, NAME, 149, Node021, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 144, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node008
 begin
-    if (global_var(155) == 4) then begin
-        gsay_reply(167, 150);
-        giq_option(4, 167, 615, Node010, 50);
+    if (global_var(GVAR_BRAIN_BOT_BRAIN) == 4) then begin
+        Reply(150);
+        giq_option(4, NAME, 615, Node010, NEUTRAL_REACTION);
     end
     else begin
-        gsay_reply(167, 146);
+        Reply(146);
     end
-    giq_option(4, 167, 152, Node010, 50);
-    giq_option(4, 167, 153, Node999, 50);
-    giq_option(-3, 167, 144, Node999, 50);
+    giq_option(4, NAME, 152, Node010, NEUTRAL_REACTION);
+    giq_option(4, NAME, 153, Node999, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 144, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node010
 begin
-    if (((metarule(16, 0) - 1) >= (floor(get_critter_stat(dude_obj, 3) / 2) + has_trait(0, dude_obj, 98))) or ((metarule(16, 0) - 1) >= 5)) then begin
-        gsay_reply(167, 650);
-        giq_option(4, 167, 651, Node999, 50);
+    if (((party_member_count(0) - 1) >= (floor(dude_charisma / 2) + has_trait(TRAIT_PERK, dude_obj, PERK_magnetic_personality_perk))) or ((party_member_count(0) - 1) >= 5)) then begin
+        Reply(650);
+        giq_option(4, NAME, 651, Node999, NEUTRAL_REACTION);
     end
     else begin
         if (local_var(12) == 0) then begin
             set_local_var(12, 1);
         end
         party_add(self_obj);
-        if (has_trait(1, self_obj, 6) != 0) then begin
-            critter_add_trait(self_obj, 1, 6, 0);
+        if (has_trait(TRAIT_OBJECT, self_obj, OBJECT_TEAM_NUM) != 0) then begin
+            critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 0);
         end
         set_local_var(8, 0);
-        if (global_var(155) == 4) then begin
-            gsay_reply(167, 154);
+        if (global_var(GVAR_BRAIN_BOT_BRAIN) == 4) then begin
+            Reply(154);
         end
         else begin
-            if (global_var(155) == 3) then begin
-                gsay_reply(167, 155);
+            if (global_var(GVAR_BRAIN_BOT_BRAIN) == 3) then begin
+                Reply(155);
             end
             else begin
-                gsay_reply(167, 156);
+                Reply(156);
             end
         end
-        giq_option(4, 167, 157, Node999, 50);
-        giq_option(-3, 167, 144, Node999, 50);
+        giq_option(4, NAME, 157, Node999, NEUTRAL_REACTION);
+        giq_option(-3, NAME, 144, Node999, NEUTRAL_REACTION);
     end
 end
 
 procedure Node011
 begin
-    if (global_var(155) == 4) then begin
-        gsay_reply(167, 158);
+    if (global_var(GVAR_BRAIN_BOT_BRAIN) == 4) then begin
+        Reply(158);
     end
     else begin
-        gsay_reply(167, 159);
+        Reply(159);
     end
-    giq_option(4, 167, 160, Node999, 50);
+    giq_option(4, NAME, 160, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node012
 begin
-    if (((metarule(16, 0) - 1) >= (floor(get_critter_stat(dude_obj, 3) / 2) + has_trait(0, dude_obj, 98))) or ((metarule(16, 0) - 1) >= 5)) then begin
-        gsay_reply(167, 162);
+    if (((party_member_count(0) - 1) >= (floor(dude_charisma / 2) + has_trait(TRAIT_PERK, dude_obj, PERK_magnetic_personality_perk))) or ((party_member_count(0) - 1) >= 5)) then begin
+        Reply(162);
     end
     else begin
-        gsay_reply(167, 162);
+        Reply(162);
     end
-    giq_option(4, 167, 163, Node999, 50);
+    giq_option(4, NAME, 163, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node013
 begin
-    gsay_reply(167, message_str(167, 164) + message_str(167, 165) + message_str(167, 166));
-    giq_option(4, 167, 400, Node013b, 50);
-    giq_option(4, 167, 168, Node004, 50);
-    giq_option(4, 167, 169, Node999, 50);
+    Reply(mstr(164) + mstr(165) + mstr(166));
+    giq_option(4, NAME, 400, Node013b, NEUTRAL_REACTION);
+    giq_option(4, NAME, 168, Node004, NEUTRAL_REACTION);
+    giq_option(4, NAME, 169, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node013b
 begin
-    metarule(43, self_obj);
-    if (global_var(155) != 4) then begin
+    inven_unwield(self_obj);
+    if (global_var(GVAR_BRAIN_BOT_BRAIN) != 4) then begin
         call Node014();
     end
     else begin
@@ -663,9 +663,9 @@ end
 
 procedure Node014
 begin
-    gsay_reply(167, 170);
-    giq_option(4, 167, 171, Node004, 50);
-    giq_option(4, 167, 172, Node999, 50);
+    Reply(170);
+    giq_option(4, NAME, 171, Node004, NEUTRAL_REACTION);
+    giq_option(4, NAME, 172, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node015
@@ -674,73 +674,73 @@ begin
     set_local_var(5, 2);
     party_remove(self_obj);
     set_local_var(12, 0);
-    critter_add_trait(self_obj, 1, 6, 52);
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 52);
 end
 
 procedure Node016
 begin
-    float_msg(self_obj, message_str(167, random(175, 177)), 7);
+    float_msg(self_obj, mstr(random(175, 177)), FLOAT_MSG_LIGHT_RED);
 end
 
 procedure Node017
 begin
     if (proto_data(obj_pid(self_obj), 3) == 16777282) then begin
-        gsay_reply(167, 210);
+        Reply(210);
     end
     else begin
-        gsay_reply(167, 178);
+        Reply(178);
     end
     if (proto_data(obj_pid(self_obj), 3) == 16777282) then begin
-        giq_option(4, 167, 211, Node017a, 50);
+        giq_option(4, NAME, 211, Node017a, NEUTRAL_REACTION);
     end
     if ((obj_item_subtype(critter_inven_obj(self_obj, 1)) == 3) or (obj_item_subtype(critter_inven_obj(self_obj, 2)) == 3)) then begin
-        giq_option(4, 167, 400, Node013b, 50);
+        giq_option(4, NAME, 400, Node013b, NEUTRAL_REACTION);
     end
-    giq_option(4, 167, 180, Node006, 50);
-    giq_option(4, 167, 181, Node999, 50);
+    giq_option(4, NAME, 180, Node006, NEUTRAL_REACTION);
+    giq_option(4, NAME, 181, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node017a
 begin
     variable LVar0 := 0;
-    LVar0 := create_object_sid(274, 0, 0, -1);
+    LVar0 := create_object(274, 0, 0);
     if (obj_is_carrying_obj_pid(self_obj, 274) == 0) then begin
-        gsay_reply(167, 214);
+        Reply(214);
     end
     if (obj_is_carrying_obj_pid(self_obj, 274) == 1) then begin
-        gsay_reply(167, 215);
+        Reply(215);
     end
     if (obj_is_carrying_obj_pid(self_obj, 274) > 1) then begin
-        gsay_reply(167, message_str(167, 212) + obj_is_carrying_obj_pid(self_obj, 274) + message_str(167, 213));
+        Reply(mstr(212) + obj_is_carrying_obj_pid(self_obj, 274) + mstr(213));
     end
-    giq_option(4, 167, 180, Node006, 50);
-    giq_option(4, 167, 181, Node999, 50);
+    giq_option(4, NAME, 180, Node006, NEUTRAL_REACTION);
+    giq_option(4, NAME, 181, Node999, NEUTRAL_REACTION);
     return 0;
     LVar0 := 0;
 end
 
 procedure Node018
 begin
-    gsay_reply(167, 182);
-    giq_option(4, 167, 183, Node006, 50);
-    giq_option(4, 167, 184, Node999, 50);
+    Reply(182);
+    giq_option(4, NAME, 183, Node006, NEUTRAL_REACTION);
+    giq_option(4, NAME, 184, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node019
 begin
-    gsay_reply(167, 185);
-    giq_option(4, 167, 186, Node006, 50);
-    giq_option(4, 167, 187, Node999, 50);
+    Reply(185);
+    giq_option(4, NAME, 186, Node006, NEUTRAL_REACTION);
+    giq_option(4, NAME, 187, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node020
 begin
-    gsay_reply(167, 188);
-    giq_option(4, 167, 189, Node020a, 50);
-    giq_option(4, 167, 190, Node020b, 50);
-    giq_option(4, 167, 191, Node020c, 50);
-    giq_option(4, 167, 192, Node006, 50);
-    giq_option(4, 167, 193, Node999, 50);
+    Reply(188);
+    giq_option(4, NAME, 189, Node020a, NEUTRAL_REACTION);
+    giq_option(4, NAME, 190, Node020b, NEUTRAL_REACTION);
+    giq_option(4, NAME, 191, Node020c, NEUTRAL_REACTION);
+    giq_option(4, NAME, 192, Node006, NEUTRAL_REACTION);
+    giq_option(4, NAME, 193, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node020a
@@ -765,31 +765,31 @@ procedure Node021
 begin
     set_local_var(8, 2);
     party_remove(self_obj);
-    if (global_var(155) == 4) then begin
-        gsay_reply(167, 194);
+    if (global_var(GVAR_BRAIN_BOT_BRAIN) == 4) then begin
+        Reply(194);
     end
     else begin
-        if (global_var(155) == 3) then begin
-            gsay_reply(167, 195);
+        if (global_var(GVAR_BRAIN_BOT_BRAIN) == 3) then begin
+            Reply(195);
         end
         else begin
-            gsay_reply(167, 196);
+            Reply(196);
         end
     end
-    giq_option(4, 167, 197, Node999, 50);
+    giq_option(4, NAME, 197, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node022
 begin
     party_remove(self_obj);
     set_local_var(8, 1);
-    if (global_var(155) == 4) then begin
-        gsay_reply(167, 198);
+    if (global_var(GVAR_BRAIN_BOT_BRAIN) == 4) then begin
+        Reply(198);
     end
     else begin
-        gsay_reply(167, 199);
+        Reply(199);
     end
-    giq_option(4, 167, 200, Node999, 50);
+    giq_option(4, NAME, 200, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node023
@@ -797,47 +797,47 @@ begin
     set_local_var(8, 0);
     set_local_var(12, 1);
     party_add(self_obj);
-    if (global_var(155) == 4) then begin
-        gsay_reply(167, 201);
+    if (global_var(GVAR_BRAIN_BOT_BRAIN) == 4) then begin
+        Reply(201);
     end
     else begin
-        gsay_reply(167, 202);
+        Reply(202);
     end
-    giq_option(4, 167, 203, Node999, 50);
+    giq_option(4, NAME, 203, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node024
 begin
-    if (cur_map_index == 28) then begin
+    if (CUR_MAP_DEPOLVA) then begin
         if ((map_var(49) == 1) and (elevation(self_obj) == 0)) then begin
-            set_global_var(151, 0);
-            set_global_var(154, 0);
-            float_msg(self_obj, message_str(167, 204), 8);
+            set_global_var(GVAR_SIERRA_BASE_ALERT, 0);
+            set_global_var(GVAR_SIERRA_BASE_SECURITY, 0);
+            float_msg(self_obj, mstr(204), FLOAT_MSG_YELLOW);
             set_map_var(49, 0);
         end
         else begin
             if ((map_var(50) == 1) and (elevation(self_obj) == 1)) then begin
-                set_global_var(151, 0);
-                set_global_var(154, 0);
-                float_msg(self_obj, message_str(167, 205), 8);
+                set_global_var(GVAR_SIERRA_BASE_ALERT, 0);
+                set_global_var(GVAR_SIERRA_BASE_SECURITY, 0);
+                float_msg(self_obj, mstr(205), FLOAT_MSG_YELLOW);
                 set_map_var(50, 0);
             end
             else begin
                 if ((map_var(51) == 1) and (elevation(self_obj) == 2)) then begin
-                    set_global_var(151, 0);
-                    set_global_var(154, 0);
-                    float_msg(self_obj, message_str(167, 206), 8);
+                    set_global_var(GVAR_SIERRA_BASE_ALERT, 0);
+                    set_global_var(GVAR_SIERRA_BASE_SECURITY, 0);
+                    float_msg(self_obj, mstr(206), FLOAT_MSG_YELLOW);
                     set_map_var(51, 0);
                 end
             end
         end
     end
     else begin
-        if (cur_map_index == 29) then begin
+        if (CUR_MAP_DEPOLVB) then begin
             if (map_var(133) == 1) then begin
-                set_global_var(151, 0);
-                set_global_var(154, 0);
-                float_msg(self_obj, message_str(167, 207), 8);
+                set_global_var(GVAR_SIERRA_BASE_ALERT, 0);
+                set_global_var(GVAR_SIERRA_BASE_SECURITY, 0);
+                float_msg(self_obj, mstr(207), FLOAT_MSG_YELLOW);
                 set_map_var(133, 0);
             end
         end
@@ -993,7 +993,7 @@ begin
                                                             call Node015();
                                                         end
                                                         else begin
-                                                            animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(dude_obj), random(0, 5), 1), 1);
+                                                            animate_move_obj_to_tile(self_obj, tile_num_in_direction(dude_tile, random(0, 5), 1), 1);
                                                             set_local_var(14, 14);
                                                             add_timer_event(self_obj, game_ticks(3), 2);
                                                         end
@@ -1015,136 +1015,136 @@ end
 procedure local_floats
 begin
     if (75 >= random(1, 100)) then begin
-        if (metarule(46, 0) == 0) then begin
-            float_msg(self_obj, message_str(167, 502), 8);
+        if (cur_town == 0) then begin
+            float_msg(self_obj, mstr(502), FLOAT_MSG_YELLOW);
         end
         else begin
-            if (metarule(46, 0) == 1) then begin
-                float_msg(self_obj, message_str(167, 503), 8);
+            if (cur_town == 1) then begin
+                float_msg(self_obj, mstr(503), FLOAT_MSG_YELLOW);
             end
             else begin
-                if (metarule(46, 0) == 2) then begin
-                    float_msg(self_obj, message_str(167, 504), 8);
+                if (cur_town == 2) then begin
+                    float_msg(self_obj, mstr(504), FLOAT_MSG_YELLOW);
                 end
                 else begin
-                    if (metarule(46, 0) == 3) then begin
-                        float_msg(self_obj, message_str(167, 505), 8);
+                    if (cur_town == 3) then begin
+                        float_msg(self_obj, mstr(505), FLOAT_MSG_YELLOW);
                     end
                     else begin
-                        if (metarule(46, 0) == 4) then begin
-                            float_msg(self_obj, message_str(167, 506), 8);
+                        if (cur_town == 4) then begin
+                            float_msg(self_obj, mstr(506), FLOAT_MSG_YELLOW);
                         end
                         else begin
-                            if (metarule(46, 0) == 5) then begin
-                                float_msg(self_obj, message_str(167, 507), 8);
+                            if (cur_town == 5) then begin
+                                float_msg(self_obj, mstr(507), FLOAT_MSG_YELLOW);
                             end
                             else begin
-                                if (metarule(46, 0) == 6) then begin
-                                    float_msg(self_obj, message_str(167, 508), 8);
+                                if (cur_town == 6) then begin
+                                    float_msg(self_obj, mstr(508), FLOAT_MSG_YELLOW);
                                 end
                                 else begin
-                                    if (metarule(46, 0) == 7) then begin
-                                        float_msg(self_obj, message_str(167, 509), 8);
+                                    if (cur_town == 7) then begin
+                                        float_msg(self_obj, mstr(509), FLOAT_MSG_YELLOW);
                                     end
                                     else begin
-                                        if (metarule(46, 0) == 8) then begin
-                                            float_msg(self_obj, message_str(167, 510), 8);
+                                        if (cur_town == 8) then begin
+                                            float_msg(self_obj, mstr(510), FLOAT_MSG_YELLOW);
                                         end
                                         else begin
-                                            if (metarule(46, 0) == 9) then begin
-                                                float_msg(self_obj, message_str(167, 511), 8);
+                                            if (cur_town == 9) then begin
+                                                float_msg(self_obj, mstr(511), FLOAT_MSG_YELLOW);
                                             end
                                             else begin
-                                                if (metarule(46, 0) == 10) then begin
-                                                    float_msg(self_obj, message_str(167, 512), 8);
+                                                if (cur_town == 10) then begin
+                                                    float_msg(self_obj, mstr(512), FLOAT_MSG_YELLOW);
                                                 end
                                                 else begin
-                                                    if (metarule(46, 0) == 11) then begin
-                                                        float_msg(self_obj, message_str(167, 513), 8);
+                                                    if (cur_town == 11) then begin
+                                                        float_msg(self_obj, mstr(513), FLOAT_MSG_YELLOW);
                                                     end
                                                     else begin
-                                                        if (metarule(46, 0) == 12) then begin
-                                                            float_msg(self_obj, message_str(167, 514), 8);
+                                                        if (cur_town == 12) then begin
+                                                            float_msg(self_obj, mstr(514), FLOAT_MSG_YELLOW);
                                                         end
                                                         else begin
-                                                            if (metarule(46, 0) == 13) then begin
-                                                                float_msg(self_obj, message_str(167, 515), 8);
+                                                            if (cur_town == 13) then begin
+                                                                float_msg(self_obj, mstr(515), FLOAT_MSG_YELLOW);
                                                             end
                                                             else begin
-                                                                if (metarule(46, 0) == 14) then begin
-                                                                    float_msg(self_obj, message_str(167, 516), 8);
+                                                                if (cur_town == 14) then begin
+                                                                    float_msg(self_obj, mstr(516), FLOAT_MSG_YELLOW);
                                                                 end
                                                                 else begin
-                                                                    if (metarule(46, 0) == 15) then begin
-                                                                        float_msg(self_obj, message_str(167, 517), 8);
+                                                                    if (cur_town == 15) then begin
+                                                                        float_msg(self_obj, mstr(517), FLOAT_MSG_YELLOW);
                                                                     end
                                                                     else begin
-                                                                        if (metarule(46, 0) == 16) then begin
-                                                                            float_msg(self_obj, message_str(167, 518), 8);
+                                                                        if (cur_town == 16) then begin
+                                                                            float_msg(self_obj, mstr(518), FLOAT_MSG_YELLOW);
                                                                         end
                                                                         else begin
-                                                                            if (metarule(46, 0) == 34) then begin
-                                                                                float_msg(self_obj, message_str(167, 499), 8);
+                                                                            if (cur_town == 34) then begin
+                                                                                float_msg(self_obj, mstr(499), FLOAT_MSG_YELLOW);
                                                                             end
                                                                             else begin
-                                                                                if (metarule(46, 0) == 40) then begin
-                                                                                    float_msg(self_obj, message_str(167, 500), 8);
+                                                                                if (cur_town == 40) then begin
+                                                                                    float_msg(self_obj, mstr(500), FLOAT_MSG_YELLOW);
                                                                                 end
                                                                                 else begin
-                                                                                    if (metarule(46, 0) == 49) then begin
-                                                                                        float_msg(self_obj, message_str(167, 519), 8);
+                                                                                    if (cur_town == 49) then begin
+                                                                                        float_msg(self_obj, mstr(519), FLOAT_MSG_YELLOW);
                                                                                     end
                                                                                     else begin
-                                                                                        if (metarule(46, 0) == 51) then begin
-                                                                                            float_msg(self_obj, message_str(167, 520), 8);
+                                                                                        if (cur_town == 51) then begin
+                                                                                            float_msg(self_obj, mstr(520), FLOAT_MSG_YELLOW);
                                                                                         end
                                                                                         else begin
-                                                                                            if (metarule(46, 0) == 52) then begin
-                                                                                                float_msg(self_obj, message_str(167, 521), 8);
+                                                                                            if (cur_town == 52) then begin
+                                                                                                float_msg(self_obj, mstr(521), FLOAT_MSG_YELLOW);
                                                                                             end
                                                                                             else begin
-                                                                                                if (metarule(46, 0) == 53) then begin
-                                                                                                    float_msg(self_obj, message_str(167, 522), 8);
+                                                                                                if (cur_town == 53) then begin
+                                                                                                    float_msg(self_obj, mstr(522), FLOAT_MSG_YELLOW);
                                                                                                 end
                                                                                                 else begin
-                                                                                                    if (metarule(46, 0) == 54) then begin
-                                                                                                        float_msg(self_obj, message_str(167, 523), 8);
+                                                                                                    if (cur_town == 54) then begin
+                                                                                                        float_msg(self_obj, mstr(523), FLOAT_MSG_YELLOW);
                                                                                                     end
                                                                                                     else begin
-                                                                                                        if (metarule(46, 0) == 56) then begin
-                                                                                                            float_msg(self_obj, message_str(167, 524), 8);
+                                                                                                        if (cur_town == 56) then begin
+                                                                                                            float_msg(self_obj, mstr(524), FLOAT_MSG_YELLOW);
                                                                                                         end
                                                                                                         else begin
-                                                                                                            if (metarule(46, 0) == 58) then begin
-                                                                                                                float_msg(self_obj, message_str(167, 525), 8);
+                                                                                                            if (cur_town == 58) then begin
+                                                                                                                float_msg(self_obj, mstr(525), FLOAT_MSG_YELLOW);
                                                                                                             end
                                                                                                             else begin
-                                                                                                                if (metarule(46, 0) == 72) then begin
-                                                                                                                    float_msg(self_obj, message_str(167, 526), 8);
+                                                                                                                if (cur_town == 72) then begin
+                                                                                                                    float_msg(self_obj, mstr(526), FLOAT_MSG_YELLOW);
                                                                                                                 end
                                                                                                                 else begin
-                                                                                                                    if (metarule(46, 0) == 73) then begin
-                                                                                                                        float_msg(self_obj, message_str(167, 527), 8);
+                                                                                                                    if (cur_town == 73) then begin
+                                                                                                                        float_msg(self_obj, mstr(527), FLOAT_MSG_YELLOW);
                                                                                                                     end
                                                                                                                     else begin
-                                                                                                                        if (metarule(46, 0) == 78) then begin
-                                                                                                                            float_msg(self_obj, message_str(167, 528), 8);
+                                                                                                                        if (cur_town == 78) then begin
+                                                                                                                            float_msg(self_obj, mstr(528), FLOAT_MSG_YELLOW);
                                                                                                                         end
                                                                                                                         else begin
-                                                                                                                            if (metarule(46, 0) == 81) then begin
-                                                                                                                                float_msg(self_obj, message_str(167, 529), 8);
+                                                                                                                            if (cur_town == 81) then begin
+                                                                                                                                float_msg(self_obj, mstr(529), FLOAT_MSG_YELLOW);
                                                                                                                             end
                                                                                                                             else begin
-                                                                                                                                if (metarule(46, 0) == 83) then begin
-                                                                                                                                    float_msg(self_obj, message_str(167, 530), 8);
+                                                                                                                                if (cur_town == 83) then begin
+                                                                                                                                    float_msg(self_obj, mstr(530), FLOAT_MSG_YELLOW);
                                                                                                                                 end
                                                                                                                                 else begin
-                                                                                                                                    if (metarule(46, 0) == 87) then begin
-                                                                                                                                        float_msg(self_obj, message_str(167, 532), 8);
+                                                                                                                                    if (cur_town == 87) then begin
+                                                                                                                                        float_msg(self_obj, mstr(532), FLOAT_MSG_YELLOW);
                                                                                                                                     end
                                                                                                                                     else begin
-                                                                                                                                        if (metarule(46, 0) == 89) then begin
-                                                                                                                                            float_msg(self_obj, message_str(167, 533), 8);
+                                                                                                                                        if (cur_town == 89) then begin
+                                                                                                                                            float_msg(self_obj, mstr(533), FLOAT_MSG_YELLOW);
                                                                                                                                         end
                                                                                                                                     end
                                                                                                                                 end
@@ -1180,4 +1180,3 @@ begin
         end
     end
 end
-
