diff --git a/Mods/MegaMod/Original/WIBRKBRN.ssl b/Mods/MegaMod/ReDefine/WIBRKBRN.ssl
index 6609480..31f848c 100644
--- a/Mods/MegaMod/Original/WIBRKBRN.ssl
+++ b/Mods/MegaMod/ReDefine/WIBRKBRN.ssl
@@ -12,7 +12,7 @@ procedure use_skill_on_p_proc
 begin
     if ((action_being_used == 12) or (action_being_used == 13)) then begin
         script_overrides;
-        display_msg(message_str(897, 100));
+        display_mstr(100);
     end
 end
 
@@ -24,7 +24,7 @@ begin
         if (map_var(73) == 0) then begin
             script_overrides;
             set_map_var(73, map_var(73) + 1);
-            display_msg(message_str(897, 101));
+            display_mstr(101);
             rm_obj_from_inven(dude_obj, obj_being_used_with);
             destroy_object(obj_being_used_with);
         end
@@ -34,7 +34,7 @@ begin
             if (map_var(74) == 0) then begin
                 script_overrides;
                 set_map_var(74, map_var(74) + 1);
-                display_msg(message_str(897, 103));
+                display_mstr(103);
                 rm_obj_from_inven(dude_obj, obj_being_used_with);
                 destroy_object(obj_being_used_with);
             end
@@ -44,25 +44,25 @@ begin
                 if (map_var(72) == 0) then begin
                     script_overrides;
                     give_exp_points(500);
-                    display_msg(message_str(14, 100) + 500 + message_str(14, 101));
+                    display_msg(g_mstr(100) + 500 + g_mstr(101));
                     set_map_var(72, map_var(72) + 1);
                     if (LVar0 == 321) then begin
-                        set_global_var(155, 4);
+                        set_global_var(GVAR_BRAIN_BOT_BRAIN, 4);
                     end
                     else begin
                         if (LVar0 == 322) then begin
-                            set_global_var(155, 3);
+                            set_global_var(GVAR_BRAIN_BOT_BRAIN, 3);
                         end
                         else begin
                             if (LVar0 == 323) then begin
-                                set_global_var(155, 2);
+                                set_global_var(GVAR_BRAIN_BOT_BRAIN, 2);
                             end
                             else begin
-                                set_global_var(155, 1);
+                                set_global_var(GVAR_BRAIN_BOT_BRAIN, 1);
                             end
                         end
                     end
-                    display_msg(message_str(897, 102));
+                    display_mstr(102);
                     rm_obj_from_inven(dude_obj, obj_being_used_with);
                     destroy_object(obj_being_used_with);
                 end
@@ -80,4 +80,3 @@ begin
         destroy_object(self_obj);
     end
 end
-
