diff --git a/Mods/MegaMod/Original/WIELVDOR.ssl b/Mods/MegaMod/ReDefine/WIELVDOR.ssl
index 390e15f..23fcfca 100644
--- a/Mods/MegaMod/Original/WIELVDOR.ssl
+++ b/Mods/MegaMod/ReDefine/WIELVDOR.ssl
@@ -141,7 +141,7 @@ end
 
 procedure use_p_proc
 begin
-    if (cur_map_index == 28) then begin
+    if (CUR_MAP_DEPOLVA) then begin
         if ((elevation(self_obj) == map_var(49)) and (tile_num(self_obj) != 19918)) then begin
             debug_msg("Clearance Granted");
         end
@@ -151,14 +151,14 @@ begin
             end
             else begin
                 script_overrides;
-                display_msg(message_str(896, 100));
+                display_mstr(100);
             end
         end
     end
     else begin
         if (map_var(1) != elevation(self_obj)) then begin
             script_overrides;
-            display_msg(message_str(896, 100));
+            display_mstr(100);
         end
     end
 end
@@ -167,7 +167,7 @@ procedure use_skill_p_proc
 begin
     if (action_being_used == 9) then begin
         script_overrides;
-        display_msg(message_str(896, 101));
+        display_mstr(101);
     end
 end
 
@@ -175,25 +175,25 @@ procedure use_obj_on_p_proc
 begin
     if ((obj_pid(obj_being_used_with) == 411) or (obj_pid(obj_being_used_with) == 77)) then begin
         script_overrides;
-        display_msg(message_str(896, 101));
+        display_mstr(101);
     end
 end
 
 procedure look_p_proc
 begin
     script_overrides;
-    display_msg(message_str(896, 102));
+    display_mstr(102);
 end
 
 procedure description_p_proc
 begin
     script_overrides;
-    display_msg(message_str(896, 103));
+    display_mstr(103);
 end
 
 procedure map_update_p_proc
 begin
-    if (cur_map_index == 28) then begin
+    if (CUR_MAP_DEPOLVA) then begin
         if ((elevation(self_obj) == map_var(49)) and (tile_num(self_obj) != 19918)) then begin
             obj_open(self_obj);
         end
@@ -204,4 +204,3 @@ begin
         end
     end
 end
-
