diff --git a/Mods/MegaMod/Original/WISCRTDR.ssl b/Mods/MegaMod/ReDefine/WISCRTDR.ssl
index da1a5cd..5dd5a24 100644
--- a/Mods/MegaMod/Original/WISCRTDR.ssl
+++ b/Mods/MegaMod/ReDefine/WISCRTDR.ssl
@@ -125,7 +125,7 @@ end
 
 procedure map_update_p_proc
 begin
-    if (global_var(154) == 1) then begin
+    if (global_var(GVAR_SIERRA_BASE_SECURITY) == 1) then begin
         if ((elevation(self_obj) == 0) and (map_var(25) == 1)) then begin
             animate_stand_obj(self_obj);
             if (tile_contains_obj_pid(tile_num(self_obj), elevation(self_obj), 50332269)) then begin
@@ -138,13 +138,13 @@ begin
                 destroy_object(tile_contains_pid_obj(tile_num(self_obj), elevation(self_obj), 50332269));
             end
         end
-        if ((elevation(self_obj) == 2) and (global_var(152) == 1)) then begin
+        if ((elevation(self_obj) == 2) and (global_var(GVAR_SIERRA_BASE_ENEMY) == 1)) then begin
             animate_stand_obj(self_obj);
             if (tile_contains_obj_pid(tile_num(self_obj), elevation(self_obj), 50332269)) then begin
                 destroy_object(tile_contains_pid_obj(tile_num(self_obj), elevation(self_obj), 50332269));
             end
         end
-        if ((cur_map_index == 29) and (global_var(151) == 1)) then begin
+        if ((CUR_MAP_DEPOLVB) and (global_var(GVAR_SIERRA_BASE_ALERT) == 1)) then begin
             animate_stand_obj(self_obj);
             if (tile_contains_obj_pid(tile_num(self_obj), elevation(self_obj), 50332269)) then begin
                 destroy_object(tile_contains_pid_obj(tile_num(self_obj), elevation(self_obj), 50332269));
@@ -154,8 +154,7 @@ begin
     else begin
         animate_stand_reverse_obj(self_obj);
         if (not(tile_contains_obj_pid(tile_num(self_obj), elevation(self_obj), 50332269))) then begin
-            create_object_sid(50332269, tile_num(self_obj), elevation(self_obj), -1);
+            create_object(50332269, tile_num(self_obj), elevation(self_obj));
         end
     end
 end
-
