diff --git a/Mods/MegaMod/Original/WITRPBX1.ssl b/Mods/MegaMod/ReDefine/WITRPBX1.ssl
index b49cf96..b5d1a8d 100644
--- a/Mods/MegaMod/Original/WITRPBX1.ssl
+++ b/Mods/MegaMod/ReDefine/WITRPBX1.ssl
@@ -170,16 +170,16 @@ begin
             set_local_var(2, 1);
             reg_anim_func(2, source_obj);
             if (source_obj == dude_obj) then begin
-                display_msg(message_str(16, 204));
+                display_msg(message_str(SCRIPT_CONTAINR, 204));
             end
             else begin
-                display_msg(message_str(16, 205));
+                display_msg(message_str(SCRIPT_CONTAINR, 205));
             end
         end
         else begin
             if (obj_is_locked(self_obj)) then begin
                 script_overrides;
-                display_msg(message_str(16, 203));
+                display_msg(message_str(SCRIPT_CONTAINR, 203));
                 call Damage_Critter();
             end
             else begin
@@ -190,7 +190,7 @@ begin
     else begin
         if (obj_is_locked(self_obj)) then begin
             script_overrides;
-            display_msg(message_str(16, 203));
+            display_msg(message_str(SCRIPT_CONTAINR, 203));
         end
     end
 end
@@ -204,14 +204,14 @@ procedure look_at_p_proc
 begin
     script_overrides;
     if (1 == 0) then begin
-        display_msg(message_str(16, 100));
+        display_msg(message_str(SCRIPT_CONTAINR, 100));
     end
     else begin
         if (1 == 2) then begin
-            display_msg(message_str(16, 212));
+            display_msg(message_str(SCRIPT_CONTAINR, 212));
         end
         else begin
-            display_msg(message_str(16, 101));
+            display_msg(message_str(SCRIPT_CONTAINR, 101));
         end
     end
 end
@@ -232,14 +232,14 @@ begin
             end
             else begin
                 if (1 == 0) then begin
-                    display_msg(message_str(16, 100));
+                    display_msg(message_str(SCRIPT_CONTAINR, 100));
                 end
                 else begin
                     if (1 == 2) then begin
-                        display_msg(message_str(16, 212));
+                        display_msg(message_str(SCRIPT_CONTAINR, 212));
                     end
                     else begin
-                        display_msg(message_str(16, 101));
+                        display_msg(message_str(SCRIPT_CONTAINR, 101));
                     end
                 end
             end
@@ -268,10 +268,10 @@ begin
             end
             else begin
                 if (source_obj == dude_obj) then begin
-                    display_msg(message_str(16, 195));
+                    display_msg(message_str(SCRIPT_CONTAINR, 195));
                 end
                 else begin
-                    display_msg(obj_name(source_obj) + message_str(16, 200));
+                    display_msg(obj_name(source_obj) + message_str(SCRIPT_CONTAINR, 200));
                 end
             end
         end
@@ -345,7 +345,7 @@ end
 procedure damage_p_proc
 begin
     variable LVar0 := 0;
-    if (metarule(49, target_obj) == 6) then begin
+    if (weapon_dmg_type(target_obj) == 6) then begin
         if (1 == 0) then begin
             if (local_var(1) == 0) then begin
                 set_local_var(1, 1);
@@ -380,7 +380,7 @@ begin
         end
     end
     else begin
-        display_msg(message_str(16, 193));
+        display_msg(message_str(SCRIPT_CONTAINR, 193));
     end
 end
 
@@ -424,36 +424,36 @@ begin
                 if (is_critical(LVar1)) then begin
                     if (is_success(LVar2)) then begin
                         if (is_critical(LVar2)) then begin
-                            display_msg(message_str(16, 102));
+                            display_msg(message_str(SCRIPT_CONTAINR, 102));
                         end
                         else begin
-                            display_msg(message_str(16, 103));
+                            display_msg(message_str(SCRIPT_CONTAINR, 103));
                         end
                     end
                     else begin
                         if (is_critical(LVar2)) then begin
-                            display_msg(message_str(16, 104));
+                            display_msg(message_str(SCRIPT_CONTAINR, 104));
                         end
                         else begin
-                            display_msg(message_str(16, 105));
+                            display_msg(message_str(SCRIPT_CONTAINR, 105));
                         end
                     end
                 end
                 else begin
                     if (is_success(LVar2)) then begin
                         if (is_critical(LVar2)) then begin
-                            display_msg(message_str(16, 106));
+                            display_msg(message_str(SCRIPT_CONTAINR, 106));
                         end
                         else begin
-                            display_msg(message_str(16, 107));
+                            display_msg(message_str(SCRIPT_CONTAINR, 107));
                         end
                     end
                     else begin
                         if (is_critical(LVar2)) then begin
-                            display_msg(message_str(16, 108));
+                            display_msg(message_str(SCRIPT_CONTAINR, 108));
                         end
                         else begin
-                            display_msg(message_str(16, 109));
+                            display_msg(message_str(SCRIPT_CONTAINR, 109));
                         end
                     end
                 end
@@ -462,36 +462,36 @@ begin
                 if (is_critical(LVar1)) then begin
                     if (is_success(LVar2)) then begin
                         if (is_critical(LVar2)) then begin
-                            display_msg(message_str(16, 110));
+                            display_msg(message_str(SCRIPT_CONTAINR, 110));
                         end
                         else begin
-                            display_msg(message_str(16, 111));
+                            display_msg(message_str(SCRIPT_CONTAINR, 111));
                         end
                     end
                     else begin
                         if (is_critical(LVar2)) then begin
-                            display_msg(message_str(16, 112));
+                            display_msg(message_str(SCRIPT_CONTAINR, 112));
                         end
                         else begin
-                            display_msg(message_str(16, 113));
+                            display_msg(message_str(SCRIPT_CONTAINR, 113));
                         end
                     end
                 end
                 else begin
                     if (is_success(LVar2)) then begin
                         if (is_critical(LVar2)) then begin
-                            display_msg(message_str(16, 114));
+                            display_msg(message_str(SCRIPT_CONTAINR, 114));
                         end
                         else begin
-                            display_msg(message_str(16, 115));
+                            display_msg(message_str(SCRIPT_CONTAINR, 115));
                         end
                     end
                     else begin
                         if (is_critical(LVar2)) then begin
-                            display_msg(message_str(16, 116));
+                            display_msg(message_str(SCRIPT_CONTAINR, 116));
                         end
                         else begin
-                            display_msg(message_str(16, 117));
+                            display_msg(message_str(SCRIPT_CONTAINR, 117));
                         end
                     end
                 end
@@ -503,36 +503,36 @@ begin
                 if (is_critical(LVar1)) then begin
                     if (is_success(LVar2)) then begin
                         if (is_critical(LVar2)) then begin
-                            display_msg(message_str(16, 118));
+                            display_msg(message_str(SCRIPT_CONTAINR, 118));
                         end
                         else begin
-                            display_msg(message_str(16, 119));
+                            display_msg(message_str(SCRIPT_CONTAINR, 119));
                         end
                     end
                     else begin
                         if (is_critical(LVar2)) then begin
-                            display_msg(message_str(16, 120));
+                            display_msg(message_str(SCRIPT_CONTAINR, 120));
                         end
                         else begin
-                            display_msg(message_str(16, 121));
+                            display_msg(message_str(SCRIPT_CONTAINR, 121));
                         end
                     end
                 end
                 else begin
                     if (is_success(LVar2)) then begin
                         if (is_critical(LVar2)) then begin
-                            display_msg(message_str(16, 122));
+                            display_msg(message_str(SCRIPT_CONTAINR, 122));
                         end
                         else begin
-                            display_msg(message_str(16, 123));
+                            display_msg(message_str(SCRIPT_CONTAINR, 123));
                         end
                     end
                     else begin
                         if (is_critical(LVar2)) then begin
-                            display_msg(message_str(16, 124));
+                            display_msg(message_str(SCRIPT_CONTAINR, 124));
                         end
                         else begin
-                            display_msg(message_str(16, 125));
+                            display_msg(message_str(SCRIPT_CONTAINR, 125));
                         end
                     end
                 end
@@ -541,36 +541,36 @@ begin
                 if (is_critical(LVar1)) then begin
                     if (is_success(LVar2)) then begin
                         if (is_critical(LVar2)) then begin
-                            display_msg(message_str(16, 126));
+                            display_msg(message_str(SCRIPT_CONTAINR, 126));
                         end
                         else begin
-                            display_msg(message_str(16, 127));
+                            display_msg(message_str(SCRIPT_CONTAINR, 127));
                         end
                     end
                     else begin
                         if (is_critical(LVar2)) then begin
-                            display_msg(message_str(16, 128));
+                            display_msg(message_str(SCRIPT_CONTAINR, 128));
                         end
                         else begin
-                            display_msg(message_str(16, 129));
+                            display_msg(message_str(SCRIPT_CONTAINR, 129));
                         end
                     end
                 end
                 else begin
                     if (is_success(LVar2)) then begin
                         if (is_critical(LVar2)) then begin
-                            display_msg(message_str(16, 130));
+                            display_msg(message_str(SCRIPT_CONTAINR, 130));
                         end
                         else begin
-                            display_msg(message_str(16, 131));
+                            display_msg(message_str(SCRIPT_CONTAINR, 131));
                         end
                     end
                     else begin
                         if (is_critical(LVar2)) then begin
-                            display_msg(message_str(16, 132));
+                            display_msg(message_str(SCRIPT_CONTAINR, 132));
                         end
                         else begin
-                            display_msg(message_str(16, 133));
+                            display_msg(message_str(SCRIPT_CONTAINR, 133));
                         end
                     end
                 end
@@ -584,36 +584,36 @@ begin
                 if (is_critical(LVar1)) then begin
                     if (is_success(LVar2)) then begin
                         if (is_critical(LVar2)) then begin
-                            display_msg(message_str(16, 134));
+                            display_msg(message_str(SCRIPT_CONTAINR, 134));
                         end
                         else begin
-                            display_msg(message_str(16, 135));
+                            display_msg(message_str(SCRIPT_CONTAINR, 135));
                         end
                     end
                     else begin
                         if (is_critical(LVar2)) then begin
-                            display_msg(message_str(16, 136));
+                            display_msg(message_str(SCRIPT_CONTAINR, 136));
                         end
                         else begin
-                            display_msg(message_str(16, 137));
+                            display_msg(message_str(SCRIPT_CONTAINR, 137));
                         end
                     end
                 end
                 else begin
                     if (is_success(LVar2)) then begin
                         if (is_critical(LVar2)) then begin
-                            display_msg(message_str(16, 138));
+                            display_msg(message_str(SCRIPT_CONTAINR, 138));
                         end
                         else begin
-                            display_msg(message_str(16, 139));
+                            display_msg(message_str(SCRIPT_CONTAINR, 139));
                         end
                     end
                     else begin
                         if (is_critical(LVar2)) then begin
-                            display_msg(message_str(16, 140));
+                            display_msg(message_str(SCRIPT_CONTAINR, 140));
                         end
                         else begin
-                            display_msg(message_str(16, 141));
+                            display_msg(message_str(SCRIPT_CONTAINR, 141));
                         end
                     end
                 end
@@ -622,36 +622,36 @@ begin
                 if (is_critical(LVar1)) then begin
                     if (is_success(LVar2)) then begin
                         if (is_critical(LVar2)) then begin
-                            display_msg(message_str(16, 142));
+                            display_msg(message_str(SCRIPT_CONTAINR, 142));
                         end
                         else begin
-                            display_msg(message_str(16, 143));
+                            display_msg(message_str(SCRIPT_CONTAINR, 143));
                         end
                     end
                     else begin
                         if (is_critical(LVar2)) then begin
-                            display_msg(message_str(16, 144));
+                            display_msg(message_str(SCRIPT_CONTAINR, 144));
                         end
                         else begin
-                            display_msg(message_str(16, 145));
+                            display_msg(message_str(SCRIPT_CONTAINR, 145));
                         end
                     end
                 end
                 else begin
                     if (is_success(LVar2)) then begin
                         if (is_critical(LVar2)) then begin
-                            display_msg(message_str(16, 146));
+                            display_msg(message_str(SCRIPT_CONTAINR, 146));
                         end
                         else begin
-                            display_msg(message_str(16, 147));
+                            display_msg(message_str(SCRIPT_CONTAINR, 147));
                         end
                     end
                     else begin
                         if (is_critical(LVar2)) then begin
-                            display_msg(message_str(16, 148));
+                            display_msg(message_str(SCRIPT_CONTAINR, 148));
                         end
                         else begin
-                            display_msg(message_str(16, 149));
+                            display_msg(message_str(SCRIPT_CONTAINR, 149));
                         end
                     end
                 end
@@ -663,36 +663,36 @@ begin
                 if (is_critical(LVar1)) then begin
                     if (is_success(LVar2)) then begin
                         if (is_critical(LVar2)) then begin
-                            display_msg(message_str(16, 150));
+                            display_msg(message_str(SCRIPT_CONTAINR, 150));
                         end
                         else begin
-                            display_msg(message_str(16, 151));
+                            display_msg(message_str(SCRIPT_CONTAINR, 151));
                         end
                     end
                     else begin
                         if (is_critical(LVar2)) then begin
-                            display_msg(message_str(16, 152));
+                            display_msg(message_str(SCRIPT_CONTAINR, 152));
                         end
                         else begin
-                            display_msg(message_str(16, 153));
+                            display_msg(message_str(SCRIPT_CONTAINR, 153));
                         end
                     end
                 end
                 else begin
                     if (is_success(LVar2)) then begin
                         if (is_critical(LVar2)) then begin
-                            display_msg(message_str(16, 154));
+                            display_msg(message_str(SCRIPT_CONTAINR, 154));
                         end
                         else begin
-                            display_msg(message_str(16, 155));
+                            display_msg(message_str(SCRIPT_CONTAINR, 155));
                         end
                     end
                     else begin
                         if (is_critical(LVar2)) then begin
-                            display_msg(message_str(16, 156));
+                            display_msg(message_str(SCRIPT_CONTAINR, 156));
                         end
                         else begin
-                            display_msg(message_str(16, 157));
+                            display_msg(message_str(SCRIPT_CONTAINR, 157));
                         end
                     end
                 end
@@ -701,36 +701,36 @@ begin
                 if (is_critical(LVar1)) then begin
                     if (is_success(LVar2)) then begin
                         if (is_critical(LVar2)) then begin
-                            display_msg(message_str(16, 158));
+                            display_msg(message_str(SCRIPT_CONTAINR, 158));
                         end
                         else begin
-                            display_msg(message_str(16, 159));
+                            display_msg(message_str(SCRIPT_CONTAINR, 159));
                         end
                     end
                     else begin
                         if (is_critical(LVar2)) then begin
-                            display_msg(message_str(16, 160));
+                            display_msg(message_str(SCRIPT_CONTAINR, 160));
                         end
                         else begin
-                            display_msg(message_str(16, 161));
+                            display_msg(message_str(SCRIPT_CONTAINR, 161));
                         end
                     end
                 end
                 else begin
                     if (is_success(LVar2)) then begin
                         if (is_critical(LVar2)) then begin
-                            display_msg(message_str(16, 162));
+                            display_msg(message_str(SCRIPT_CONTAINR, 162));
                         end
                         else begin
-                            display_msg(message_str(16, 163));
+                            display_msg(message_str(SCRIPT_CONTAINR, 163));
                         end
                     end
                     else begin
                         if (is_critical(LVar2)) then begin
-                            display_msg(message_str(16, 164));
+                            display_msg(message_str(SCRIPT_CONTAINR, 164));
                         end
                         else begin
-                            display_msg(message_str(16, 165));
+                            display_msg(message_str(SCRIPT_CONTAINR, 165));
                         end
                     end
                 end
@@ -749,36 +749,36 @@ begin
         if (is_critical(LVar0)) then begin
             if (is_success(LVar1)) then begin
                 if (is_critical(LVar1)) then begin
-                    display_msg(message_str(16, 104));
+                    display_msg(message_str(SCRIPT_CONTAINR, 104));
                 end
                 else begin
-                    display_msg(message_str(16, 108));
+                    display_msg(message_str(SCRIPT_CONTAINR, 108));
                 end
             end
             else begin
                 if (is_critical(LVar1)) then begin
-                    display_msg(message_str(16, 112));
+                    display_msg(message_str(SCRIPT_CONTAINR, 112));
                 end
                 else begin
-                    display_msg(message_str(16, 116));
+                    display_msg(message_str(SCRIPT_CONTAINR, 116));
                 end
             end
         end
         else begin
             if (is_success(LVar1)) then begin
                 if (is_critical(LVar1)) then begin
-                    display_msg(message_str(16, 120));
+                    display_msg(message_str(SCRIPT_CONTAINR, 120));
                 end
                 else begin
-                    display_msg(message_str(16, 124));
+                    display_msg(message_str(SCRIPT_CONTAINR, 124));
                 end
             end
             else begin
                 if (is_critical(LVar1)) then begin
-                    display_msg(message_str(16, 128));
+                    display_msg(message_str(SCRIPT_CONTAINR, 128));
                 end
                 else begin
-                    display_msg(message_str(16, 132));
+                    display_msg(message_str(SCRIPT_CONTAINR, 132));
                 end
             end
         end
@@ -787,36 +787,36 @@ begin
         if (is_critical(LVar0)) then begin
             if (is_success(LVar1)) then begin
                 if (is_critical(LVar1)) then begin
-                    display_msg(message_str(16, 136));
+                    display_msg(message_str(SCRIPT_CONTAINR, 136));
                 end
                 else begin
-                    display_msg(message_str(16, 140));
+                    display_msg(message_str(SCRIPT_CONTAINR, 140));
                 end
             end
             else begin
                 if (is_critical(LVar1)) then begin
-                    display_msg(message_str(16, 144));
+                    display_msg(message_str(SCRIPT_CONTAINR, 144));
                 end
                 else begin
-                    display_msg(message_str(16, 148));
+                    display_msg(message_str(SCRIPT_CONTAINR, 148));
                 end
             end
         end
         else begin
             if (is_success(LVar1)) then begin
                 if (is_critical(LVar1)) then begin
-                    display_msg(message_str(16, 152));
+                    display_msg(message_str(SCRIPT_CONTAINR, 152));
                 end
                 else begin
-                    display_msg(message_str(16, 156));
+                    display_msg(message_str(SCRIPT_CONTAINR, 156));
                 end
             end
             else begin
                 if (is_critical(LVar1)) then begin
-                    display_msg(message_str(16, 160));
+                    display_msg(message_str(SCRIPT_CONTAINR, 160));
                 end
                 else begin
-                    display_msg(message_str(16, 164));
+                    display_msg(message_str(SCRIPT_CONTAINR, 164));
                 end
             end
         end
@@ -833,36 +833,36 @@ begin
         if (is_critical(LVar0)) then begin
             if (is_success(LVar1)) then begin
                 if (is_critical(LVar1)) then begin
-                    display_msg(message_str(16, 114));
+                    display_msg(message_str(SCRIPT_CONTAINR, 114));
                 end
                 else begin
-                    display_msg(message_str(16, 115));
+                    display_msg(message_str(SCRIPT_CONTAINR, 115));
                 end
             end
             else begin
                 if (is_critical(LVar1)) then begin
-                    display_msg(message_str(16, 116));
+                    display_msg(message_str(SCRIPT_CONTAINR, 116));
                 end
                 else begin
-                    display_msg(message_str(16, 117));
+                    display_msg(message_str(SCRIPT_CONTAINR, 117));
                 end
             end
         end
         else begin
             if (is_success(LVar1)) then begin
                 if (is_critical(LVar1)) then begin
-                    display_msg(message_str(16, 130));
+                    display_msg(message_str(SCRIPT_CONTAINR, 130));
                 end
                 else begin
-                    display_msg(message_str(16, 131));
+                    display_msg(message_str(SCRIPT_CONTAINR, 131));
                 end
             end
             else begin
                 if (is_critical(LVar1)) then begin
-                    display_msg(message_str(16, 132));
+                    display_msg(message_str(SCRIPT_CONTAINR, 132));
                 end
                 else begin
-                    display_msg(message_str(16, 133));
+                    display_msg(message_str(SCRIPT_CONTAINR, 133));
                 end
             end
         end
@@ -871,36 +871,36 @@ begin
         if (is_critical(LVar0)) then begin
             if (is_success(LVar1)) then begin
                 if (is_critical(LVar1)) then begin
-                    display_msg(message_str(16, 146));
+                    display_msg(message_str(SCRIPT_CONTAINR, 146));
                 end
                 else begin
-                    display_msg(message_str(16, 147));
+                    display_msg(message_str(SCRIPT_CONTAINR, 147));
                 end
             end
             else begin
                 if (is_critical(LVar1)) then begin
-                    display_msg(message_str(16, 148));
+                    display_msg(message_str(SCRIPT_CONTAINR, 148));
                 end
                 else begin
-                    display_msg(message_str(16, 149));
+                    display_msg(message_str(SCRIPT_CONTAINR, 149));
                 end
             end
         end
         else begin
             if (is_success(LVar1)) then begin
                 if (is_critical(LVar1)) then begin
-                    display_msg(message_str(16, 162));
+                    display_msg(message_str(SCRIPT_CONTAINR, 162));
                 end
                 else begin
-                    display_msg(message_str(16, 163));
+                    display_msg(message_str(SCRIPT_CONTAINR, 163));
                 end
             end
             else begin
                 if (is_critical(LVar1)) then begin
-                    display_msg(message_str(16, 164));
+                    display_msg(message_str(SCRIPT_CONTAINR, 164));
                 end
                 else begin
-                    display_msg(message_str(16, 165));
+                    display_msg(message_str(SCRIPT_CONTAINR, 165));
                 end
             end
         end
@@ -919,24 +919,24 @@ begin
         if (local_var(2) == 0) then begin
             set_local_var(2, 1);
             if (source_obj == dude_obj) then begin
-                display_msg(message_str(16, 194));
+                display_msg(message_str(SCRIPT_CONTAINR, 194));
             end
             else begin
-                display_msg(message_str(16, 199));
+                display_msg(message_str(SCRIPT_CONTAINR, 199));
             end
             if (local_var(6) == 0) then begin
                 set_local_var(6, 1);
                 give_exp_points(25);
-                display_msg(message_str(14, 100) + 25 + message_str(14, 101));
+                display_msg(g_mstr(100) + 25 + g_mstr(101));
             end
         end
         else begin
             set_local_var(1, 1);
             if (source_obj == dude_obj) then begin
-                display_msg(message_str(16, 196));
+                display_msg(message_str(SCRIPT_CONTAINR, 196));
             end
             else begin
-                display_msg(obj_name(source_obj) + message_str(16, 201));
+                display_msg(obj_name(source_obj) + message_str(SCRIPT_CONTAINR, 201));
             end
         end
     end
@@ -947,14 +947,14 @@ begin
         else begin
             if (local_var(2) == 0) then begin
                 if (source_obj == dude_obj) then begin
-                    display_msg(message_str(16, 195));
+                    display_msg(message_str(SCRIPT_CONTAINR, 195));
                 end
                 else begin
-                    display_msg(obj_name(source_obj) + message_str(16, 200));
+                    display_msg(obj_name(source_obj) + message_str(SCRIPT_CONTAINR, 200));
                 end
             end
             else begin
-                display_msg(obj_name(source_obj) + message_str(16, 202));
+                display_msg(obj_name(source_obj) + message_str(SCRIPT_CONTAINR, 202));
             end
         end
     end
@@ -978,33 +978,33 @@ begin
         set_local_var(0, 1);
         obj_unlock(self_obj);
         if (source_obj == dude_obj) then begin
-            display_msg(message_str(16, 170));
+            display_msg(message_str(SCRIPT_CONTAINR, 170));
         end
         else begin
-            display_msg(obj_name(source_obj) + message_str(16, 173));
+            display_msg(obj_name(source_obj) + message_str(SCRIPT_CONTAINR, 173));
         end
         if (local_var(5) == 0) then begin
             set_local_var(5, 1);
             give_exp_points(45);
-            display_msg(message_str(14, 100) + 45 + message_str(14, 101));
+            display_msg(g_mstr(100) + 45 + g_mstr(101));
         end
     end
     else begin
         if (is_critical(Locks_Roll)) then begin
             jam_lock(self_obj);
             if (source_obj == dude_obj) then begin
-                display_msg(message_str(16, 171));
+                display_msg(message_str(SCRIPT_CONTAINR, 171));
             end
             else begin
-                display_msg(obj_name(source_obj) + message_str(16, 174));
+                display_msg(obj_name(source_obj) + message_str(SCRIPT_CONTAINR, 174));
             end
         end
         else begin
             if (source_obj == dude_obj) then begin
-                display_msg(message_str(16, 172));
+                display_msg(message_str(SCRIPT_CONTAINR, 172));
             end
             else begin
-                display_msg(obj_name(source_obj) + message_str(16, 175));
+                display_msg(obj_name(source_obj) + message_str(SCRIPT_CONTAINR, 175));
             end
         end
     end
@@ -1016,23 +1016,23 @@ begin
         set_local_var(0, 0);
         obj_lock(self_obj);
         if (source_obj == dude_obj) then begin
-            display_msg(message_str(16, 189));
+            display_msg(message_str(SCRIPT_CONTAINR, 189));
         end
         else begin
-            display_msg(obj_name(source_obj) + message_str(16, 191));
+            display_msg(obj_name(source_obj) + message_str(SCRIPT_CONTAINR, 191));
         end
         if (local_var(5) == 0) then begin
             set_local_var(5, 1);
             give_exp_points(45);
-            display_msg(message_str(14, 100) + 45 + message_str(14, 101));
+            display_msg(g_mstr(100) + 45 + g_mstr(101));
         end
     end
     else begin
         if (source_obj == dude_obj) then begin
-            display_msg(message_str(16, 190));
+            display_msg(message_str(SCRIPT_CONTAINR, 190));
         end
         else begin
-            display_msg(obj_name(source_obj) + message_str(16, 192));
+            display_msg(obj_name(source_obj) + message_str(SCRIPT_CONTAINR, 192));
         end
     end
 end
@@ -1058,10 +1058,10 @@ begin
     if (obj_is_open(self_obj)) then begin
         script_overrides;
         if (source_obj == dude_obj) then begin
-            display_msg(message_str(16, 206));
+            display_msg(message_str(SCRIPT_CONTAINR, 206));
         end
         else begin
-            display_msg(obj_name(source_obj) + message_str(16, 207));
+            display_msg(obj_name(source_obj) + message_str(SCRIPT_CONTAINR, 207));
         end
     end
     else begin
@@ -1070,10 +1070,10 @@ begin
         if (is_success(Traps_Roll)) then begin
             set_local_var(1, 0);
             if (source_obj == dude_obj) then begin
-                display_msg(message_str(16, 208));
+                display_msg(message_str(SCRIPT_CONTAINR, 208));
             end
             else begin
-                display_msg(obj_name(source_obj) + message_str(16, 210));
+                display_msg(obj_name(source_obj) + message_str(SCRIPT_CONTAINR, 210));
             end
         end
         else begin
@@ -1082,10 +1082,10 @@ begin
             end
             else begin
                 if (source_obj == dude_obj) then begin
-                    display_msg(message_str(16, 209));
+                    display_msg(message_str(SCRIPT_CONTAINR, 209));
                 end
                 else begin
-                    display_msg(obj_name(source_obj) + message_str(16, 211));
+                    display_msg(obj_name(source_obj) + message_str(SCRIPT_CONTAINR, 211));
                 end
             end
         end
@@ -1100,10 +1100,10 @@ begin
         set_local_var(0, 1);
         obj_unlock(self_obj);
         if (source_obj == dude_obj) then begin
-            display_msg(message_str(16, 176));
+            display_msg(message_str(SCRIPT_CONTAINR, 176));
         end
         else begin
-            display_msg(message_str(16, 181));
+            display_msg(message_str(SCRIPT_CONTAINR, 181));
         end
     end
     else begin
@@ -1111,37 +1111,37 @@ begin
             critter_dmg(source_obj, 2, 0 bwor 256);
             if (source_obj == dude_obj) then begin
                 if (2 == 1) then begin
-                    display_msg(message_str(16, 177));
+                    display_msg(message_str(SCRIPT_CONTAINR, 177));
                 end
                 else begin
-                    display_msg(message_str(16, 178) + 2 + message_str(16, 179));
+                    display_msg(message_str(SCRIPT_CONTAINR, 178) + 2 + message_str(SCRIPT_CONTAINR, 179));
                 end
             end
             else begin
-                if (get_critter_stat(source_obj, 34) == 0) then begin
+                if (get_critter_stat(source_obj, STAT_gender) == 0) then begin
                     if (2 == 1) then begin
-                        display_msg(message_str(16, 182));
+                        display_msg(message_str(SCRIPT_CONTAINR, 182));
                     end
                     else begin
-                        display_msg(message_str(16, 183) + 2 + message_str(16, 184));
+                        display_msg(message_str(SCRIPT_CONTAINR, 183) + 2 + message_str(SCRIPT_CONTAINR, 184));
                     end
                 end
                 else begin
                     if (2 == 1) then begin
-                        display_msg(message_str(16, 186));
+                        display_msg(message_str(SCRIPT_CONTAINR, 186));
                     end
                     else begin
-                        display_msg(message_str(16, 187) + 2 + message_str(16, 188));
+                        display_msg(message_str(SCRIPT_CONTAINR, 187) + 2 + message_str(SCRIPT_CONTAINR, 188));
                     end
                 end
             end
         end
         else begin
             if (source_obj == dude_obj) then begin
-                display_msg(message_str(16, 180));
+                display_msg(message_str(SCRIPT_CONTAINR, 180));
             end
             else begin
-                display_msg(message_str(16, 185));
+                display_msg(message_str(SCRIPT_CONTAINR, 185));
             end
         end
     end
@@ -1153,11 +1153,11 @@ begin
     LVar0 := random(20, 30);
     if (source_obj == dude_obj) then begin
         critter_dmg(dude_obj, LVar0, 6 bwor 256);
-        display_msg(message_str(16, 166) + LVar0 + message_str(16, 167));
+        display_msg(message_str(SCRIPT_CONTAINR, 166) + LVar0 + message_str(SCRIPT_CONTAINR, 167));
     end
     else begin
         critter_dmg(source_obj, LVar0, 6 bwor 256);
-        display_msg(obj_name(source_obj) + message_str(16, 168) + LVar0 + message_str(16, 169));
+        display_msg(obj_name(source_obj) + message_str(SCRIPT_CONTAINR, 168) + LVar0 + message_str(SCRIPT_CONTAINR, 169));
     end
     set_local_var(1, 1);
 end
@@ -1173,4 +1173,3 @@ begin
     Locks_Roll := roll_vs_skill(source_obj, 9, -40 + 40);
     call Lock_Door();
 end
-
