diff --git a/Mods/MegaMod/Original/WSTERM3C.ssl b/Mods/MegaMod/ReDefine/WSTERM3C.ssl
index ef81fdb..e630921 100644
--- a/Mods/MegaMod/Original/WSTERM3C.ssl
+++ b/Mods/MegaMod/ReDefine/WSTERM3C.ssl
@@ -150,12 +150,12 @@ end
 procedure use_p_proc
 begin
     variable LVar0 := 0;
-    if (get_critter_stat(source_obj, 4) < 4) then begin
+    if (get_critter_stat(source_obj, STAT_iq) < 4) then begin
         if (source_obj == dude_obj) then begin
-            display_msg(message_str(161, 102));
+            display_mstr(102);
         end
         else begin
-            display_msg(obj_name(source_obj) + message_str(161, 200));
+            display_msg(obj_name(source_obj) + mstr(200));
         end
     end
     else begin
@@ -186,13 +186,13 @@ end
 procedure look_p_proc
 begin
     script_overrides;
-    display_msg(message_str(161, 100));
+    display_mstr(100);
 end
 
 procedure description_p_proc
 begin
     script_overrides;
-    display_msg(message_str(161, 101));
+    display_mstr(101);
 end
 
 procedure push_p_proc
@@ -201,16 +201,16 @@ end
 
 procedure Node001
 begin
-    gsay_reply(161, message_str(161, 103) + message_str(161, 104));
-    giq_option(4, 161, 600, Node003, 50);
-    giq_option(4, 161, 601, Node004, 50);
-    giq_option(4, 161, 107, Node999, 50);
+    Reply(mstr(103) + mstr(104));
+    giq_option(4, NAME, 600, Node003, NEUTRAL_REACTION);
+    giq_option(4, NAME, 601, Node004, NEUTRAL_REACTION);
+    giq_option(4, NAME, 107, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node002
 begin
-    gsay_reply(161, 110);
-    giq_option(4, 161, 111, Node999, 50);
+    Reply(110);
+    giq_option(4, NAME, 111, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node010
@@ -218,41 +218,41 @@ begin
     if (map_var(72) == 1) then begin
         if (map_var(73) == 1) then begin
             if (map_var(74) == 1) then begin
-                gsay_reply(161, 113);
+                Reply(113);
             end
             else begin
-                gsay_reply(161, 117);
+                Reply(117);
             end
         end
         else begin
             if (map_var(74) == 1) then begin
-                gsay_reply(161, 115);
+                Reply(115);
             end
             else begin
-                gsay_reply(161, 119);
+                Reply(119);
             end
         end
     end
     else begin
         if (map_var(73) == 1) then begin
             if (map_var(74) == 1) then begin
-                gsay_reply(161, 114);
+                Reply(114);
             end
             else begin
-                gsay_reply(161, 118);
+                Reply(118);
             end
         end
         else begin
             if (map_var(74) == 1) then begin
-                gsay_reply(161, 116);
+                Reply(116);
             end
             else begin
-                gsay_reply(161, 120);
+                Reply(120);
             end
         end
     end
-    giq_option(4, 161, 153, Node011, 50);
-    giq_option(4, 161, 154, Node999, 50);
+    giq_option(4, NAME, 153, Node011, NEUTRAL_REACTION);
+    giq_option(4, NAME, 154, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node011
@@ -262,7 +262,7 @@ begin
     set_local_var(0, 1);
     if ((map_var(73) == 0) or (map_var(74) == 0) or (map_var(72) == 0)) then begin
         set_map_var(75, 1);
-        gsay_reply(161, message_str(161, 155) + message_str(161, 156));
+        Reply(mstr(155) + mstr(156));
     end
     else begin
         gfade_out(600);
@@ -270,49 +270,49 @@ begin
         if (tile_contains_obj_pid(8908, elevation(self_obj), 33555662)) then begin
             LVar0 := tile_contains_pid_obj(8908, elevation(self_obj), 33555662);
             destroy_object(LVar0);
-            if (global_var(155) == 4) then begin
-                LVar1 := create_object_sid(16777295, 8908, elevation(self_obj), 167);
+            if (global_var(GVAR_BRAIN_BOT_BRAIN) == 4) then begin
+                LVar1 := create_object_sid(16777295, 8908, elevation(self_obj), SCRIPT_WCBRNBOT);
                 debug_msg("Made Cybernetic Robobrain " + obj_pid(LVar1));
             end
             else begin
-                if (global_var(155) == 3) then begin
-                    LVar1 := create_object_sid(16777597, 8908, elevation(self_obj), 167);
+                if (global_var(GVAR_BRAIN_BOT_BRAIN) == 3) then begin
+                    LVar1 := create_object_sid(16777597, 8908, elevation(self_obj), SCRIPT_WCBRNBOT);
                     debug_msg("Made Human Robobrain " + obj_pid(LVar1));
                 end
                 else begin
-                    if (global_var(155) == 2) then begin
-                        LVar1 := create_object_sid(16777595, 8908, elevation(self_obj), 167);
+                    if (global_var(GVAR_BRAIN_BOT_BRAIN) == 2) then begin
+                        LVar1 := create_object_sid(16777595, 8908, elevation(self_obj), SCRIPT_WCBRNBOT);
                         debug_msg("Made Chimp Robobrain " + obj_pid(LVar1));
                     end
                     else begin
-                        if (global_var(155) == 1) then begin
-                            LVar1 := create_object_sid(16777596, 8908, elevation(self_obj), 167);
+                        if (global_var(GVAR_BRAIN_BOT_BRAIN) == 1) then begin
+                            LVar1 := create_object_sid(16777596, 8908, elevation(self_obj), SCRIPT_WCBRNBOT);
                             debug_msg("Made Abnormal Robobrain " + obj_pid(LVar1));
                         end
                     end
                 end
             end
         end
-        if (global_var(155) == 4) then begin
-            gsay_reply(161, 500);
+        if (global_var(GVAR_BRAIN_BOT_BRAIN) == 4) then begin
+            Reply(500);
         end
         else begin
-            if (global_var(155) == 3) then begin
-                gsay_reply(161, message_str(161, 164) + message_str(161, 165) + message_str(161, 166) + message_str(161, 167));
+            if (global_var(GVAR_BRAIN_BOT_BRAIN) == 3) then begin
+                Reply(mstr(164) + mstr(165) + mstr(166) + mstr(167));
             end
             else begin
-                if (global_var(155) == 2) then begin
-                    gsay_reply(161, message_str(161, 164) + message_str(161, 165) + message_str(161, 170) + message_str(161, 167));
+                if (global_var(GVAR_BRAIN_BOT_BRAIN) == 2) then begin
+                    Reply(mstr(164) + mstr(165) + mstr(170) + mstr(167));
                 end
                 else begin
-                    if (global_var(155) == 1) then begin
-                        gsay_reply(161, message_str(161, 164) + message_str(161, 165) + message_str(161, 174) + message_str(161, 167));
+                    if (global_var(GVAR_BRAIN_BOT_BRAIN) == 1) then begin
+                        Reply(mstr(164) + mstr(165) + mstr(174) + mstr(167));
                     end
                 end
             end
         end
     end
-    giq_option(4, 161, 176, Node999, 50);
+    giq_option(4, NAME, 176, Node999, NEUTRAL_REACTION);
     return 0;
     LVar0 := 0;
     LVar1 := 0;
@@ -324,18 +324,18 @@ end
 
 procedure Node003
 begin
-    gsay_reply(161, 603);
-    giq_option(4, 161, 105, Node010, 50);
-    giq_option(4, 161, 106, Node011, 50);
-    giq_option(4, 161, 107, Node999, 50);
+    Reply(603);
+    giq_option(4, NAME, 105, Node010, NEUTRAL_REACTION);
+    giq_option(4, NAME, 106, Node011, NEUTRAL_REACTION);
+    giq_option(4, NAME, 107, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node004
 begin
-    gsay_reply(161, 660);
-    giq_option(4, 161, 661, Node012, 50);
-    giq_option(4, 161, 662, Node013, 50);
-    giq_option(4, 161, 107, Node999, 50);
+    Reply(660);
+    giq_option(4, NAME, 661, Node012, NEUTRAL_REACTION);
+    giq_option(4, NAME, 662, Node013, NEUTRAL_REACTION);
+    giq_option(4, NAME, 107, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node012
@@ -343,41 +343,41 @@ begin
     if (map_var(92) == 1) then begin
         if (map_var(93) == 1) then begin
             if (map_var(94) == 1) then begin
-                gsay_reply(161, 613);
+                Reply(613);
             end
             else begin
-                gsay_reply(161, 617);
+                Reply(617);
             end
         end
         else begin
             if (map_var(94) == 1) then begin
-                gsay_reply(161, 615);
+                Reply(615);
             end
             else begin
-                gsay_reply(161, 619);
+                Reply(619);
             end
         end
     end
     else begin
         if (map_var(93) == 1) then begin
             if (map_var(94) == 1) then begin
-                gsay_reply(161, 614);
+                Reply(614);
             end
             else begin
-                gsay_reply(161, 618);
+                Reply(618);
             end
         end
         else begin
             if (map_var(94) == 1) then begin
-                gsay_reply(161, 616);
+                Reply(616);
             end
             else begin
-                gsay_reply(161, 620);
+                Reply(620);
             end
         end
     end
-    giq_option(4, 161, 653, Node013, 50);
-    giq_option(4, 161, 654, Node999, 50);
+    giq_option(4, NAME, 653, Node013, NEUTRAL_REACTION);
+    giq_option(4, NAME, 654, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node013
@@ -387,7 +387,7 @@ begin
     set_local_var(0, 1);
     if ((map_var(93) == 0) or (map_var(94) == 0) or (map_var(92) == 0)) then begin
         set_map_var(95, 1);
-        gsay_reply(161, message_str(161, 655) + message_str(161, 656));
+        Reply(mstr(655) + mstr(656));
     end
     else begin
         gfade_out(600);
@@ -395,51 +395,50 @@ begin
         if (tile_contains_obj_pid(9296, elevation(self_obj), 33556326)) then begin
             LVar0 := tile_contains_pid_obj(9296, elevation(self_obj), 33556326);
             destroy_object(LVar0);
-            if (global_var(155) == 4) then begin
-                LVar1 := create_object_sid(16777785, 9296, elevation(self_obj), 167);
+            if (global_var(GVAR_BRAIN_BOT_BRAIN) == 4) then begin
+                LVar1 := create_object_sid(16777785, 9296, elevation(self_obj), SCRIPT_WCBRNBOT);
                 debug_msg("Made Cybernetic Robobrain " + obj_pid(LVar1));
             end
             else begin
-                if (global_var(155) == 3) then begin
-                    LVar1 := create_object_sid(16777749, 9296, elevation(self_obj), 167);
+                if (global_var(GVAR_BRAIN_BOT_BRAIN) == 3) then begin
+                    LVar1 := create_object_sid(16777749, 9296, elevation(self_obj), SCRIPT_WCBRNBOT);
                     debug_msg("Made Human Robobrain " + obj_pid(LVar1));
                 end
                 else begin
-                    if (global_var(155) == 2) then begin
-                        LVar1 := create_object_sid(16777747, 9296, elevation(self_obj), 167);
+                    if (global_var(GVAR_BRAIN_BOT_BRAIN) == 2) then begin
+                        LVar1 := create_object_sid(16777747, 9296, elevation(self_obj), SCRIPT_WCBRNBOT);
                         debug_msg("Made Chimp Robobrain " + obj_pid(LVar1));
                     end
                     else begin
-                        if (global_var(155) == 1) then begin
-                            LVar1 := create_object_sid(16777748, 9296, elevation(self_obj), 167);
+                        if (global_var(GVAR_BRAIN_BOT_BRAIN) == 1) then begin
+                            LVar1 := create_object_sid(16777748, 9296, elevation(self_obj), SCRIPT_WCBRNBOT);
                             debug_msg("Made Abnormal Robobrain " + obj_pid(LVar1));
                         end
                     end
                 end
             end
         end
-        if (global_var(155) == 4) then begin
-            gsay_reply(161, 500);
+        if (global_var(GVAR_BRAIN_BOT_BRAIN) == 4) then begin
+            Reply(500);
         end
         else begin
-            if (global_var(155) == 3) then begin
-                gsay_reply(161, message_str(161, 164) + message_str(161, 165) + message_str(161, 166) + message_str(161, 167));
+            if (global_var(GVAR_BRAIN_BOT_BRAIN) == 3) then begin
+                Reply(mstr(164) + mstr(165) + mstr(166) + mstr(167));
             end
             else begin
-                if (global_var(155) == 2) then begin
-                    gsay_reply(161, message_str(161, 164) + message_str(161, 165) + message_str(161, 170) + message_str(161, 167));
+                if (global_var(GVAR_BRAIN_BOT_BRAIN) == 2) then begin
+                    Reply(mstr(164) + mstr(165) + mstr(170) + mstr(167));
                 end
                 else begin
-                    if (global_var(155) == 1) then begin
-                        gsay_reply(161, message_str(161, 164) + message_str(161, 165) + message_str(161, 174) + message_str(161, 167));
+                    if (global_var(GVAR_BRAIN_BOT_BRAIN) == 1) then begin
+                        Reply(mstr(164) + mstr(165) + mstr(174) + mstr(167));
                     end
                 end
             end
         end
     end
-    giq_option(4, 161, 176, Node999, 50);
+    giq_option(4, NAME, 176, Node999, NEUTRAL_REACTION);
     return 0;
     LVar0 := 0;
     LVar1 := 0;
 end
-
