diff --git a/Mods/MegaMod/Original/WSTERM2B.ssl b/Mods/MegaMod/ReDefine/WSTERM2B.ssl
index 851889b..ac9f6b9 100644
--- a/Mods/MegaMod/Original/WSTERM2B.ssl
+++ b/Mods/MegaMod/ReDefine/WSTERM2B.ssl
@@ -180,20 +180,20 @@ begin
     end_dialogue;
     if (Injure_Player == 1) then begin
         Injure_Player := Injure_Player + 1;
-        critter_dmg(source_obj, 20, 6);
+        critter_dmg(source_obj, 20, DMG_explosion);
     end
 end
 
 procedure look_at_p_proc
 begin
     script_overrides;
-    display_msg(message_str(157, 100));
+    display_mstr(100);
 end
 
 procedure description_p_proc
 begin
     script_overrides;
-    display_msg(message_str(157, 101));
+    display_mstr(101);
 end
 
 procedure use_skill_on_p_proc
@@ -205,15 +205,15 @@ begin
     variable LVar0 := 0;
     script_overrides;
     if (local_var(0) == 1) then begin
-        display_msg(message_str(157, 200));
+        display_mstr(200);
     end
     else begin
-        if (get_critter_stat(source_obj, 4) < 4) then begin
+        if (get_critter_stat(source_obj, STAT_iq) < 4) then begin
             if (source_obj == dude_obj) then begin
-                display_msg(message_str(157, 102));
+                display_mstr(102);
             end
             else begin
-                display_msg(message_str(157, 400));
+                display_mstr(400);
             end
         end
         else begin
@@ -233,49 +233,49 @@ end
 procedure Node001
 begin
     if ((map_var(56) == 1) and (map_var(41) == 1)) then begin
-        gsay_reply(157, 103);
+        Reply(103);
     end
     else begin
         if ((map_var(56) == 0) and (map_var(41) == 1)) then begin
-            gsay_reply(157, 104);
+            Reply(104);
         end
         else begin
             if ((map_var(56) == 1) and (map_var(41) == 0)) then begin
-                gsay_reply(157, 105);
+                Reply(105);
             end
             else begin
-                gsay_reply(157, 106);
+                Reply(106);
             end
         end
     end
     if (map_var(56) == 0) then begin
-        giq_option(4, 157, 107, Node010, 50);
+        giq_option(4, NAME, 107, Node010, NEUTRAL_REACTION);
     end
     if (map_var(41) == 0) then begin
-        giq_option(4, 157, 108, Node013, 50);
+        giq_option(4, NAME, 108, Node013, NEUTRAL_REACTION);
     end
-    giq_option(4, 157, 109, Node016, 50);
-    giq_option(4, 157, 110, Node999, 50);
+    giq_option(4, NAME, 109, Node016, NEUTRAL_REACTION);
+    giq_option(4, NAME, 110, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node010
 begin
-    gsay_reply(157, 112);
+    Reply(112);
     if (local_var(1) == 0) then begin
-        giq_option(8, 157, 113, Node010a, 50);
+        giq_option(8, NAME, 113, Node010a, NEUTRAL_REACTION);
     end
     if (local_var(2) == 0) then begin
-        giq_option(4, 157, 114, Node010b, 50);
+        giq_option(4, NAME, 114, Node010b, NEUTRAL_REACTION);
     end
-    giq_option(4, 157, 115, Node001, 50);
-    giq_option(4, 157, 116, Node999, 50);
+    giq_option(4, NAME, 115, Node001, NEUTRAL_REACTION);
+    giq_option(4, NAME, 116, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node010a
 begin
     variable LVar0 := 0;
     set_local_var(1, local_var(1) + 1);
-    if (has_skill(dude_obj, 12) > 89) then begin
+    if (dude_skill(SKILL_SCIENCE) > 89) then begin
         call Node012();
     end
     else begin
@@ -313,9 +313,9 @@ begin
         set_obj_visibility(LVar0, 0);
     end
     set_map_var(56, 0);
-    gsay_reply(157, 117);
-    giq_option(4, 157, 118, Node001, 50);
-    giq_option(4, 157, 119, Node999, 50);
+    Reply(117);
+    giq_option(4, NAME, 118, Node001, NEUTRAL_REACTION);
+    giq_option(4, NAME, 119, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node012
@@ -328,33 +328,33 @@ begin
     set_map_var(56, 1);
     if (local_var(10) == 0) then begin
         set_local_var(10, local_var(10) + 1);
-        display_msg(message_str(14, 1019));
+        display_msg(g_mstr(1019));
         give_exp_points(200);
-        display_msg(message_str(14, 100) + 200 + message_str(14, 101));
+        display_msg(g_mstr(100) + 200 + g_mstr(101));
     end
-    gsay_reply(157, 120);
-    giq_option(4, 157, 121, Node001, 50);
-    giq_option(4, 157, 122, Node999, 50);
+    Reply(120);
+    giq_option(4, NAME, 121, Node001, NEUTRAL_REACTION);
+    giq_option(4, NAME, 122, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node013
 begin
-    gsay_reply(157, 123);
+    Reply(123);
     if (local_var(3) == 0) then begin
-        giq_option(4, 157, 124, Node013a, 50);
+        giq_option(4, NAME, 124, Node013a, NEUTRAL_REACTION);
     end
     if (local_var(4) == 0) then begin
-        giq_option(4, 157, 125, Node013b, 50);
+        giq_option(4, NAME, 125, Node013b, NEUTRAL_REACTION);
     end
-    giq_option(4, 157, 126, Node001, 50);
-    giq_option(4, 157, 127, Node999, 50);
+    giq_option(4, NAME, 126, Node001, NEUTRAL_REACTION);
+    giq_option(4, NAME, 127, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node013a
 begin
     variable LVar0 := 0;
     set_local_var(3, local_var(3) + 1);
-    if (has_skill(dude_obj, 12) > 89) then begin
+    if (dude_skill(SKILL_SCIENCE) > 89) then begin
         call Node015();
     end
     else begin
@@ -402,36 +402,36 @@ begin
     if (local_var(11) == 0) then begin
         set_local_var(11, local_var(11) + 1);
         give_exp_points(200);
-        display_msg(message_str(14, 100) + 200 + message_str(14, 101));
+        display_msg(g_mstr(100) + 200 + g_mstr(101));
     end
-    gsay_reply(157, 131);
-    giq_option(4, 157, 132, Node001, 50);
-    giq_option(4, 157, 133, Node999, 50);
+    Reply(131);
+    giq_option(4, NAME, 132, Node001, NEUTRAL_REACTION);
+    giq_option(4, NAME, 133, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node016
 begin
     if (local_var(5) == 0) then begin
         set_local_var(5, 1);
-        gsay_reply(157, 134);
+        Reply(134);
     end
     else begin
-        gsay_reply(157, 135);
+        Reply(135);
     end
     if (local_var(6) == 0) then begin
-        giq_option(4, 157, 136, Node016a, 50);
+        giq_option(4, NAME, 136, Node016a, NEUTRAL_REACTION);
     end
     if (local_var(7) == 0) then begin
-        giq_option(4, 157, 137, Node016b, 50);
+        giq_option(4, NAME, 137, Node016b, NEUTRAL_REACTION);
     end
     if (local_var(8) == 0) then begin
-        giq_option(4, 157, 138, Node016c, 50);
+        giq_option(4, NAME, 138, Node016c, NEUTRAL_REACTION);
     end
     if (local_var(9) == 0) then begin
-        giq_option(4, 157, 139, Node016d, 50);
+        giq_option(4, NAME, 139, Node016d, NEUTRAL_REACTION);
     end
-    giq_option(4, 157, 140, Node001, 50);
-    giq_option(4, 157, 141, Node999, 50);
+    giq_option(4, NAME, 140, Node001, NEUTRAL_REACTION);
+    giq_option(4, NAME, 141, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node016a
@@ -460,16 +460,16 @@ end
 
 procedure Node017
 begin
-    gsay_reply(157, 142);
-    giq_option(4, 157, 143, Node016, 50);
-    giq_option(4, 157, 144, Node999, 50);
+    Reply(142);
+    giq_option(4, NAME, 143, Node016, NEUTRAL_REACTION);
+    giq_option(4, NAME, 144, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node018
 begin
-    gsay_reply(157, 145);
-    giq_option(4, 157, 143, Node016, 50);
-    giq_option(4, 157, 144, Node999, 50);
+    Reply(145);
+    giq_option(4, NAME, 143, Node016, NEUTRAL_REACTION);
+    giq_option(4, NAME, 144, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node019
@@ -481,15 +481,15 @@ begin
     end
     set_local_var(0, 1);
     set_map_var(56, 1);
-    gsay_reply(157, 148);
-    giq_option(4, 157, 144, Node999, 50);
+    Reply(148);
+    giq_option(4, NAME, 144, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node020
 begin
-    gsay_reply(157, 150);
-    giq_option(4, 157, 143, Node016, 50);
-    giq_option(4, 157, 144, Node999, 50);
+    Reply(150);
+    giq_option(4, NAME, 143, Node016, NEUTRAL_REACTION);
+    giq_option(4, NAME, 144, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node021
@@ -500,9 +500,9 @@ begin
         set_obj_visibility(LVar0, 1);
     end
     set_map_var(56, 1);
-    gsay_reply(157, 153);
-    giq_option(4, 157, 154, Node001, 50);
-    giq_option(4, 157, 155, Node999, 50);
+    Reply(153);
+    giq_option(4, NAME, 154, Node001, NEUTRAL_REACTION);
+    giq_option(4, NAME, 155, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node022
@@ -513,9 +513,9 @@ begin
         set_obj_visibility(LVar0, 0);
     end
     set_map_var(56, 0);
-    gsay_reply(157, 156);
-    giq_option(4, 157, 157, Node001, 50);
-    giq_option(4, 157, 158, Node999, 50);
+    Reply(156);
+    giq_option(4, NAME, 157, Node001, NEUTRAL_REACTION);
+    giq_option(4, NAME, 158, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node023
@@ -533,9 +533,9 @@ begin
     Field2b_Ptr := 0;
     Field2c_Ptr := 0;
     Field2d_Ptr := 0;
-    gsay_reply(157, 159);
-    giq_option(4, 157, 160, Node001, 50);
-    giq_option(4, 157, 161, Node999, 50);
+    Reply(159);
+    giq_option(4, NAME, 160, Node001, NEUTRAL_REACTION);
+    giq_option(4, NAME, 161, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node024
@@ -545,20 +545,19 @@ begin
     set_map_var(51, 0);
     set_map_var(52, 0);
     set_map_var(53, 0);
-    Field2a_Ptr := create_object_sid(33554921, local_var(12), elevation(self_obj), 245);
-    Field2b_Ptr := create_object_sid(33554921, local_var(13), elevation(self_obj), 246);
-    Field2c_Ptr := create_object_sid(33554921, local_var(14), elevation(self_obj), 247);
-    Field2d_Ptr := create_object_sid(33554921, local_var(15), elevation(self_obj), 248);
-    gsay_reply(157, 162);
-    giq_option(4, 157, 163, Node001, 50);
-    giq_option(4, 157, 164, Node999, 50);
+    Field2a_Ptr := create_object_sid(33554921, local_var(12), elevation(self_obj), SCRIPT_WSFELD2A);
+    Field2b_Ptr := create_object_sid(33554921, local_var(13), elevation(self_obj), SCRIPT_WSFELD2B);
+    Field2c_Ptr := create_object_sid(33554921, local_var(14), elevation(self_obj), SCRIPT_WSFELD2C);
+    Field2d_Ptr := create_object_sid(33554921, local_var(15), elevation(self_obj), SCRIPT_WSFELD2D);
+    Reply(162);
+    giq_option(4, NAME, 163, Node001, NEUTRAL_REACTION);
+    giq_option(4, NAME, 164, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node025
 begin
     set_local_var(0, 1);
     Injure_Player := 1;
-    gsay_reply(157, 165);
-    giq_option(4, 157, 166, Node999, 50);
+    Reply(165);
+    giq_option(4, NAME, 166, Node999, NEUTRAL_REACTION);
 end
-
