diff --git a/Mods/MegaMod/Original/ZCCORPSE.ssl b/Mods/MegaMod/ReDefine/ZCCORPSE.ssl
index 724733a..5718db5 100644
--- a/Mods/MegaMod/Original/ZCCORPSE.ssl
+++ b/Mods/MegaMod/ReDefine/ZCCORPSE.ssl
@@ -145,16 +145,15 @@ end
 procedure look_at_p_proc
 begin
     script_overrides;
-    display_msg(message_str(1483, 100));
+    display_mstr(100);
 end
 
 procedure description_p_proc
 begin
     script_overrides;
-    display_msg(message_str(1483, 101));
+    display_mstr(101);
 end
 
 procedure map_enter_p_proc
 begin
 end
-
