diff --git a/Mods/MegaMod/Original/ZIDCEGRL.ssl b/Mods/MegaMod/ReDefine/ZIDCEGRL.ssl
index 5f8f0e8..8d28a5a 100644
--- a/Mods/MegaMod/Original/ZIDCEGRL.ssl
+++ b/Mods/MegaMod/ReDefine/ZIDCEGRL.ssl
@@ -185,53 +185,53 @@ end
 procedure ChooseFromList
 begin
     variable LVar0 := 0;
-    gsay_reply(410, stored_msg);
-    if (message_str(353, top_msg - 1) != "") then begin
-        giq_option(1, 410, message_str(353, 150), CLICK_PREV, 50);
+    Reply(stored_msg);
+    if (message_str(SCRIPT_NEWR1, top_msg - 1) != "") then begin
+        giq_option(1, NAME, message_str(SCRIPT_NEWR1, 150), CLICK_PREV, NEUTRAL_REACTION);
     end
     if (LVar0 == 0) then begin
-        if (message_str(353, top_msg + 0) != "") then begin
-            giq_option(1, 410, message_str(353, top_msg + 0), CLICK_0, 50);
+        if (message_str(SCRIPT_NEWR1, top_msg + 0) != "") then begin
+            giq_option(1, NAME, message_str(SCRIPT_NEWR1, top_msg + 0), CLICK_0, NEUTRAL_REACTION);
         end
         else begin
             LVar0 := 1;
         end
     end
     if (LVar0 == 0) then begin
-        if (message_str(353, top_msg + 1) != "") then begin
-            giq_option(1, 410, message_str(353, top_msg + 1), CLICK_1, 50);
+        if (message_str(SCRIPT_NEWR1, top_msg + 1) != "") then begin
+            giq_option(1, NAME, message_str(SCRIPT_NEWR1, top_msg + 1), CLICK_1, NEUTRAL_REACTION);
         end
         else begin
             LVar0 := 1;
         end
     end
     if (LVar0 == 0) then begin
-        if (message_str(353, top_msg + 2) != "") then begin
-            giq_option(1, 410, message_str(353, top_msg + 2), CLICK_2, 50);
+        if (message_str(SCRIPT_NEWR1, top_msg + 2) != "") then begin
+            giq_option(1, NAME, message_str(SCRIPT_NEWR1, top_msg + 2), CLICK_2, NEUTRAL_REACTION);
         end
         else begin
             LVar0 := 1;
         end
     end
     if (LVar0 == 0) then begin
-        if (message_str(353, top_msg + 3) != "") then begin
-            giq_option(1, 410, message_str(353, top_msg + 3), CLICK_3, 50);
+        if (message_str(SCRIPT_NEWR1, top_msg + 3) != "") then begin
+            giq_option(1, NAME, message_str(SCRIPT_NEWR1, top_msg + 3), CLICK_3, NEUTRAL_REACTION);
         end
         else begin
             LVar0 := 1;
         end
     end
     if (LVar0 == 0) then begin
-        if (message_str(353, top_msg + 4) != "") then begin
-            giq_option(1, 410, message_str(353, top_msg + 4), CLICK_4, 50);
+        if (message_str(SCRIPT_NEWR1, top_msg + 4) != "") then begin
+            giq_option(1, NAME, message_str(SCRIPT_NEWR1, top_msg + 4), CLICK_4, NEUTRAL_REACTION);
         end
         else begin
             LVar0 := 1;
         end
     end
     if (LVar0 == 0) then begin
-        if (message_str(353, top_msg + 5) != "") then begin
-            giq_option(1, 410, message_str(353, 151), CLICK_MORE, 50);
+        if (message_str(SCRIPT_NEWR1, top_msg + 5) != "") then begin
+            giq_option(1, NAME, message_str(SCRIPT_NEWR1, 151), CLICK_MORE, NEUTRAL_REACTION);
         end
     end
 end
@@ -289,10 +289,10 @@ begin
             if (anim_busy(self_obj) == 0) then begin
                 if ((tile_distance_objs(self_obj, dude_obj) < 8) or obj_can_see_obj(self_obj, dude_obj)) then begin
                     if (anim_busy(self_obj) == 0) then begin
-                        global_temp := rotation_to_tile(tile_num(dude_obj), tile_num(self_obj));
+                        global_temp := rotation_to_tile(dude_tile, tile_num(self_obj));
                         animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(self_obj), global_temp, random(3, 10)), 1);
                         global_temp := (global_temp + 1) % 6;
-                        while ((anim_busy(self_obj) == 0) and (global_temp != rotation_to_tile(tile_num(dude_obj), tile_num(self_obj)))) do begin
+                        while ((anim_busy(self_obj) == 0) and (global_temp != rotation_to_tile(dude_tile, tile_num(self_obj)))) do begin
                             debug_msg("flee loop: rot == " + global_temp);
                             animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(self_obj), global_temp, random(3, 10)), 1);
                             global_temp := (global_temp + 1) % 6;
@@ -302,7 +302,7 @@ begin
             end
         end
         else begin
-            attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+            attack(dude_obj);
         end
     end
 end
@@ -341,55 +341,55 @@ end
 procedure destroy_p_proc
 begin
     if (source_obj == dude_obj) then begin
-        set_global_var(5, global_var(5) + 1);
-        if (metarule(51, self_obj) == 2) then begin
+        set_global_var(GVAR_GOOD_MONSTER, global_var(GVAR_GOOD_MONSTER) + 1);
+        if (critter_kill_type(self_obj) == KILL_TYPE_children_kills) then begin
             if (source_obj == dude_obj) then begin
-                set_global_var(2, 0);
-                set_global_var(1, global_var(1) + 1);
-                set_global_var(0, global_var(0) + -15);
+                set_global_var(GVAR_CHAMPION_REPUTATION, 0);
+                set_global_var(GVAR_CHILDKILLER_REPUTATION, global_var(GVAR_CHILDKILLER_REPUTATION) + 1);
+                set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + -15);
                 debug_msg("Player gains " + -15 + " Karma Points.");
-                set_global_var(37, 0);
-                set_global_var(38, 0);
-                set_global_var(39, 0);
-                set_global_var(40, 0);
-                set_global_var(41, 0);
-                set_global_var(42, 0);
-                set_global_var(43, 0);
-                set_global_var(44, 0);
-                set_global_var(45, 0);
-                if (global_var(0) >= 1000) then begin
-                    set_global_var(37, 1);
+                set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+                set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+                set_global_var(GVAR_KARMA_DEFENDER, 0);
+                set_global_var(GVAR_KARMA_WANDERER, 0);
+                set_global_var(GVAR_KARMA_BETRAYER, 0);
+                set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+                set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+                set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+                    set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
                 end
                 else begin
-                    if (global_var(0) >= 750) then begin
-                        set_global_var(38, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
                     end
                     else begin
-                        if (global_var(0) >= 500) then begin
-                            set_global_var(39, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                            set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                         end
                         else begin
-                            if (global_var(0) >= 250) then begin
-                                set_global_var(40, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                                set_global_var(GVAR_KARMA_DEFENDER, 1);
                             end
                             else begin
-                                if (global_var(0) > -250) then begin
-                                    set_global_var(41, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                                    set_global_var(GVAR_KARMA_WANDERER, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -500) then begin
-                                        set_global_var(42, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                        set_global_var(GVAR_KARMA_BETRAYER, 1);
                                     end
                                     else begin
-                                        if (global_var(0) > -750) then begin
-                                            set_global_var(43, 1);
+                                        if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                            set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                         end
                                         else begin
-                                            if (global_var(0) > -1000) then begin
-                                                set_global_var(44, 1);
+                                            if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                                set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                             end
                                             else begin
-                                                set_global_var(45, 1);
+                                                set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                             end
                                         end
                                     end
@@ -400,58 +400,58 @@ begin
                 end
             end
         end
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(4) > (3 * global_var(5))) or (global_var(2) == 1)) and (global_var(1) == 0)) then begin
-            set_global_var(2, 1);
-            set_global_var(3, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(4) > (3 * global_var(GVAR_GOOD_MONSTER))) or (global_var(GVAR_CHAMPION_REPUTATION) == 1)) and (global_var(GVAR_CHILDKILLER_REPUTATION) == 0)) then begin
+            set_global_var(GVAR_CHAMPION_REPUTATION, 1);
+            set_global_var(GVAR_BERSERKER_REPUTATION, 0);
         end
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(5) > (2 * global_var(4))) or (global_var(3) == 1))) then begin
-            set_global_var(3, 1);
-            set_global_var(2, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
+            set_global_var(GVAR_BERSERKER_REPUTATION, 1);
+            set_global_var(GVAR_CHAMPION_REPUTATION, 0);
         end
-        set_global_var(0, global_var(0) + -10);
+        set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + -10);
         debug_msg("Player gains " + -10 + " Karma Points.");
-        set_global_var(37, 0);
-        set_global_var(38, 0);
-        set_global_var(39, 0);
-        set_global_var(40, 0);
-        set_global_var(41, 0);
-        set_global_var(42, 0);
-        set_global_var(43, 0);
-        set_global_var(44, 0);
-        set_global_var(45, 0);
-        if (global_var(0) >= 1000) then begin
-            set_global_var(37, 1);
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+        set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+        set_global_var(GVAR_KARMA_DEFENDER, 0);
+        set_global_var(GVAR_KARMA_WANDERER, 0);
+        set_global_var(GVAR_KARMA_BETRAYER, 0);
+        set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+        set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+            set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
         end
         else begin
-            if (global_var(0) >= 750) then begin
-                set_global_var(38, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
             end
             else begin
-                if (global_var(0) >= 500) then begin
-                    set_global_var(39, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                 end
                 else begin
-                    if (global_var(0) >= 250) then begin
-                        set_global_var(40, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                        set_global_var(GVAR_KARMA_DEFENDER, 1);
                     end
                     else begin
-                        if (global_var(0) > -250) then begin
-                            set_global_var(41, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                            set_global_var(GVAR_KARMA_WANDERER, 1);
                         end
                         else begin
-                            if (global_var(0) > -500) then begin
-                                set_global_var(42, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                set_global_var(GVAR_KARMA_BETRAYER, 1);
                             end
                             else begin
-                                if (global_var(0) > -750) then begin
-                                    set_global_var(43, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -1000) then begin
-                                        set_global_var(44, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                     end
                                     else begin
-                                        set_global_var(45, 1);
+                                        set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                     end
                                 end
                             end
@@ -466,17 +466,17 @@ end
 procedure look_at_p_proc
 begin
     script_overrides;
-    display_msg(message_str(410, 100 + (obj_type(self_obj) != 1) + (10 * (((get_critter_stat(dude_obj, 4) > 3) and (has_skill(dude_obj, 16) >= 25)) == 0))));
+    display_mstr(100 + (obj_type(self_obj) != 1) + (10 * (((dude_iq > 3) and (dude_skill(SKILL_GAMBLING) >= 25)) == 0)));
 end
 
 procedure description_p_proc
 begin
     script_overrides;
     if (obj_type(self_obj) == 1) then begin
-        display_msg(message_str(410, random(150, 150 + (((local_var(0) bwand 4) != 0) * (3 + (3 * (get_critter_stat(dude_obj, 34) == 0))))) + (10 * (((get_critter_stat(dude_obj, 4) > 3) and (has_skill(dude_obj, 16) >= 25)) == 0))));
+        display_mstr(random(150, 150 + (((local_var(0) bwand 4) != 0) * (3 + (3 * (dude_gender == 0))))) + (10 * (((dude_iq > 3) and (dude_skill(SKILL_GAMBLING) >= 25)) == 0)));
     end
     else begin
-        display_msg(message_str(410, 140 + (((get_critter_stat(dude_obj, 4) > 3) and (has_skill(dude_obj, 16) >= 25)) == 0)));
+        display_mstr(140 + (((dude_iq > 3) and (dude_skill(SKILL_GAMBLING) >= 25)) == 0));
     end
     set_local_var(0, local_var(0) bwor 4);
 end
@@ -484,46 +484,46 @@ end
 procedure use_skill_on_p_proc
 begin
     variable LVar0 := 0;
-    if ((action_being_used == 16) and ((get_critter_stat(dude_obj, 4) > 3) and (has_skill(dude_obj, 16) >= 25)) and (obj_type(self_obj) == 1)) then begin
+    if ((action_being_used == 16) and ((dude_iq > 3) and (dude_skill(SKILL_GAMBLING) >= 25)) and (obj_type(self_obj) == 1)) then begin
         script_overrides;
         LVar0 := roll_vs_skill(dude_obj, action_being_used, 0);
         if (is_success(LVar0)) then begin
             if ((obj_is_carrying_obj_pid(self_obj, 325) + obj_is_carrying_obj_pid(self_obj, 326)) < 1) then begin
-                display_msg(message_str(410, 175));
+                display_mstr(175);
             end
             else begin
                 if ((obj_is_carrying_obj_pid(self_obj, 325) + obj_is_carrying_obj_pid(self_obj, 326)) > 1) then begin
-                    display_msg(message_str(410, 176));
+                    display_mstr(176);
                 end
                 else begin
                     if (obj_is_carrying_obj_pid(self_obj, 326) > 0) then begin
-                        display_msg(message_str(410, 177));
+                        display_mstr(177);
                     end
                     else begin
-                        display_msg(message_str(410, 187));
+                        display_mstr(187);
                     end
                 end
             end
         end
         else begin
             if ((obj_is_carrying_obj_pid(self_obj, 325) + obj_is_carrying_obj_pid(self_obj, 326)) != 1) then begin
-                display_msg(message_str(410, 185));
+                display_mstr(185);
             end
             else begin
                 if (obj_is_carrying_obj_pid(self_obj, 326) > 0) then begin
                     if (is_critical(LVar0)) then begin
-                        display_msg(message_str(410, 178));
+                        display_mstr(178);
                     end
                     else begin
-                        display_msg(message_str(410, 179));
+                        display_mstr(179);
                     end
                 end
                 else begin
                     if (is_critical(LVar0)) then begin
-                        display_msg(message_str(410, 185));
+                        display_mstr(185);
                     end
                     else begin
-                        display_msg(message_str(410, 186));
+                        display_mstr(186);
                     end
                 end
             end
@@ -543,14 +543,14 @@ end
 procedure map_enter_p_proc
 begin
     Only_Once := 0;
-    if (metarule(14, 0)) then begin
+    if (map_first_run) then begin
         set_local_var(1, tile_num(self_obj));
         if (obj_is_carrying_obj_pid(self_obj, 325) < 1) then begin
-            restock_amt := random(1, 1);
+            restock_amt := 1;
             if (obj_is_carrying_obj_pid(self_obj, 325) < restock_amt) then begin
                 if (100 >= random(1, 100)) then begin
                     if (obj_is_carrying_obj_pid(self_obj, 325) < restock_amt) then begin
-                        restock_obj := create_object_sid(325, 0, 0, -1);
+                        restock_obj := create_object(325, 0, 0);
                         add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 325));
                     end
                     else begin
@@ -563,17 +563,17 @@ begin
                                 restock_obj := obj_carrying_pid_obj(self_obj, 325);
                                 if (obj_type(self_obj) == 1) then begin
                                     restock_obj := obj_carrying_pid_obj(self_obj, 325);
-                                    if (critter_inven_obj(self_obj, 0) == restock_obj) then begin
-                                        if (obj_item_subtype(critter_inven_obj(self_obj, 0)) == 0) then begin
-                                            restock_obj := critter_inven_obj(self_obj, 0);
+                                    if (critter_inven_obj(self_obj, INVEN_TYPE_WORN) == restock_obj) then begin
+                                        if (obj_item_subtype(critter_inven_obj(self_obj, INVEN_TYPE_WORN)) == 0) then begin
+                                            restock_obj := critter_inven_obj(self_obj, INVEN_TYPE_WORN);
                                             debug_msg("armour pid == " + obj_pid(restock_obj));
                                             rm_obj_from_inven(self_obj, restock_obj);
                                             add_obj_to_inven(self_obj, restock_obj);
                                         end
                                     end
                                     else begin
-                                        if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                            metarule(43, self_obj);
+                                        if ((critter_inven_obj(self_obj, INVEN_TYPE_LEFT_HAND) == restock_obj) or (critter_inven_obj(self_obj, INVEN_TYPE_RIGHT_HAND) == restock_obj)) then begin
+                                            inven_unwield(self_obj);
                                         end
                                     end
                                 end
@@ -591,7 +591,7 @@ begin
             end
             else begin
                 if (obj_is_carrying_obj_pid(self_obj, 325) < restock_amt) then begin
-                    restock_obj := create_object_sid(325, 0, 0, -1);
+                    restock_obj := create_object(325, 0, 0);
                     add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 325));
                 end
                 else begin
@@ -604,17 +604,17 @@ begin
                             restock_obj := obj_carrying_pid_obj(self_obj, 325);
                             if (obj_type(self_obj) == 1) then begin
                                 restock_obj := obj_carrying_pid_obj(self_obj, 325);
-                                if (critter_inven_obj(self_obj, 0) == restock_obj) then begin
-                                    if (obj_item_subtype(critter_inven_obj(self_obj, 0)) == 0) then begin
-                                        restock_obj := critter_inven_obj(self_obj, 0);
+                                if (critter_inven_obj(self_obj, INVEN_TYPE_WORN) == restock_obj) then begin
+                                    if (obj_item_subtype(critter_inven_obj(self_obj, INVEN_TYPE_WORN)) == 0) then begin
+                                        restock_obj := critter_inven_obj(self_obj, INVEN_TYPE_WORN);
                                         debug_msg("armour pid == " + obj_pid(restock_obj));
                                         rm_obj_from_inven(self_obj, restock_obj);
                                         add_obj_to_inven(self_obj, restock_obj);
                                     end
                                 end
                                 else begin
-                                    if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                        metarule(43, self_obj);
+                                    if ((critter_inven_obj(self_obj, INVEN_TYPE_LEFT_HAND) == restock_obj) or (critter_inven_obj(self_obj, INVEN_TYPE_RIGHT_HAND) == restock_obj)) then begin
+                                        inven_unwield(self_obj);
                                     end
                                 end
                             end
@@ -635,7 +635,7 @@ begin
             if (obj_is_carrying_obj_pid(self_obj, 41) < restock_amt) then begin
                 if (100 >= random(1, 100)) then begin
                     if (obj_is_carrying_obj_pid(self_obj, 41) < restock_amt) then begin
-                        restock_obj := create_object_sid(41, 0, 0, -1);
+                        restock_obj := create_object(41, 0, 0);
                         add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 41));
                     end
                     else begin
@@ -648,17 +648,17 @@ begin
                                 restock_obj := obj_carrying_pid_obj(self_obj, 41);
                                 if (obj_type(self_obj) == 1) then begin
                                     restock_obj := obj_carrying_pid_obj(self_obj, 41);
-                                    if (critter_inven_obj(self_obj, 0) == restock_obj) then begin
-                                        if (obj_item_subtype(critter_inven_obj(self_obj, 0)) == 0) then begin
-                                            restock_obj := critter_inven_obj(self_obj, 0);
+                                    if (critter_inven_obj(self_obj, INVEN_TYPE_WORN) == restock_obj) then begin
+                                        if (obj_item_subtype(critter_inven_obj(self_obj, INVEN_TYPE_WORN)) == 0) then begin
+                                            restock_obj := critter_inven_obj(self_obj, INVEN_TYPE_WORN);
                                             debug_msg("armour pid == " + obj_pid(restock_obj));
                                             rm_obj_from_inven(self_obj, restock_obj);
                                             add_obj_to_inven(self_obj, restock_obj);
                                         end
                                     end
                                     else begin
-                                        if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                            metarule(43, self_obj);
+                                        if ((critter_inven_obj(self_obj, INVEN_TYPE_LEFT_HAND) == restock_obj) or (critter_inven_obj(self_obj, INVEN_TYPE_RIGHT_HAND) == restock_obj)) then begin
+                                            inven_unwield(self_obj);
                                         end
                                     end
                                 end
@@ -676,7 +676,7 @@ begin
             end
             else begin
                 if (obj_is_carrying_obj_pid(self_obj, 41) < restock_amt) then begin
-                    restock_obj := create_object_sid(41, 0, 0, -1);
+                    restock_obj := create_object(41, 0, 0);
                     add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 41));
                 end
                 else begin
@@ -689,17 +689,17 @@ begin
                             restock_obj := obj_carrying_pid_obj(self_obj, 41);
                             if (obj_type(self_obj) == 1) then begin
                                 restock_obj := obj_carrying_pid_obj(self_obj, 41);
-                                if (critter_inven_obj(self_obj, 0) == restock_obj) then begin
-                                    if (obj_item_subtype(critter_inven_obj(self_obj, 0)) == 0) then begin
-                                        restock_obj := critter_inven_obj(self_obj, 0);
+                                if (critter_inven_obj(self_obj, INVEN_TYPE_WORN) == restock_obj) then begin
+                                    if (obj_item_subtype(critter_inven_obj(self_obj, INVEN_TYPE_WORN)) == 0) then begin
+                                        restock_obj := critter_inven_obj(self_obj, INVEN_TYPE_WORN);
                                         debug_msg("armour pid == " + obj_pid(restock_obj));
                                         rm_obj_from_inven(self_obj, restock_obj);
                                         add_obj_to_inven(self_obj, restock_obj);
                                     end
                                 end
                                 else begin
-                                    if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                        metarule(43, self_obj);
+                                    if ((critter_inven_obj(self_obj, INVEN_TYPE_LEFT_HAND) == restock_obj) or (critter_inven_obj(self_obj, INVEN_TYPE_RIGHT_HAND) == restock_obj)) then begin
+                                        inven_unwield(self_obj);
                                     end
                                 end
                             end
@@ -721,37 +721,37 @@ begin
             critter_attempt_placement(self_obj, local_var(1), elevation(self_obj));
         end
     end
-    critter_add_trait(self_obj, 1, 6, 83);
-    critter_add_trait(self_obj, 1, 5, 33);
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 83);
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_COWARD);
     add_timer_event(self_obj, game_ticks(random(1, 30)), 1);
 end
 
 procedure timed_event_p_proc
 begin
     if (combat_is_initialized == 0) then begin
-        if (tile_distance(tile_num(self_obj), tile_num(dude_obj)) <= 4) then begin
+        if (tile_distance(tile_num(self_obj), dude_tile) <= 4) then begin
             if (random(0, 2) == 0) then begin
                 play_sfx("dice1");
             end
             else begin
                 if (obj_can_see_obj(self_obj, dude_obj)) then begin
-                    if (global_var(494) == 8) then begin
+                    if (global_var(GVAR_FALLOUT_2) == 8) then begin
                         call Node002();
                     end
                     else begin
-                        if (((global_var(230) / 100) == 0) and (0 != 0)) then begin
+                        if (((global_var(GVAR_NEW_RENO_MADE_MAN) / 100) == 0) and (0 != 0)) then begin
                             call Node003();
                         end
                         else begin
-                            if (((global_var(230) / 100) != 0) and (0 != 0)) then begin
+                            if (((global_var(GVAR_NEW_RENO_MADE_MAN) / 100) != 0) and (0 != 0)) then begin
                                 call Node005();
                             end
                             else begin
-                                if (global_var(232) > 0) then begin
+                                if (global_var(GVAR_NEW_RENO_PORN_STAR) > 0) then begin
                                     call Node007();
                                 end
                                 else begin
-                                    if (global_var(259) > 0) then begin
+                                    if (global_var(GVAR_NEW_RENO_HAS_REP_PRIZEFIGHTER) > 0) then begin
                                         call Node009();
                                     end
                                     else begin
@@ -779,88 +779,88 @@ end
 
 procedure Node002
 begin
-    call floater_rand_with_check_cust(200, 204, obj_name(dude_obj));
+    call floater_rand_with_check_cust(200, 204, dude_name);
 end
 
 procedure Node003
 begin
-    call floater_rand_with_check_cust(219 - (4 * (get_critter_stat(dude_obj, 34) == 0)), 220 + (3 * (get_critter_stat(dude_obj, 34) == 1)), message_str(353, 300 + (global_var(230) % 100)));
+    call floater_rand_with_check_cust(219 - (4 * (dude_gender == 0)), 220 + (3 * (dude_gender == 1)), message_str(SCRIPT_NEWR1, 300 + (global_var(GVAR_NEW_RENO_MADE_MAN) % 100)));
 end
 
 procedure Node005
 begin
-    float_msg(self_obj, message_str(410, random(238 - (3 * (get_critter_stat(dude_obj, 34) == 0)), 238 + (3 * (get_critter_stat(dude_obj, 34) == 1)))), 2);
+    float_msg(self_obj, mstr(random(238 - (3 * (dude_gender == 0)), 238 + (3 * (dude_gender == 1)))), FLOAT_MSG_RED);
 end
 
 procedure Node007
 begin
-    call floater_rand_with_check_cust(255 - (5 * (get_critter_stat(dude_obj, 34) == 0)), 260 + (3 * (get_critter_stat(dude_obj, 34) == 1)), message_str(353, 400 + global_var(231)));
+    call floater_rand_with_check_cust(255 - (5 * (dude_gender == 0)), 260 + (3 * (dude_gender == 1)), message_str(SCRIPT_NEWR1, 400 + global_var(GVAR_NEW_RENO_PRIZEFIGHTER)));
 end
 
 procedure Node009
 begin
-    call floater_rand_with_check_cust(271 - (get_critter_stat(dude_obj, 34) == 0), 277 + (2 * (get_critter_stat(dude_obj, 34) == 1)), message_str(353, 500 + global_var(232)));
+    call floater_rand_with_check_cust(271 - (dude_gender == 0), 277 + (2 * (dude_gender == 1)), message_str(SCRIPT_NEWR1, 500 + global_var(GVAR_NEW_RENO_PORN_STAR)));
 end
 
 procedure Node011
 begin
-    float_msg(self_obj, message_str(410, random(293 - (3 * (get_critter_stat(dude_obj, 34) == 0)), 298 + (3 * (get_critter_stat(dude_obj, 34) == 1)))), 2);
+    float_msg(self_obj, mstr(random(293 - (3 * (dude_gender == 0)), 298 + (3 * (dude_gender == 1)))), FLOAT_MSG_RED);
 end
 
 procedure Node013
 begin
-    if (((global_var(230) / 100) != 0) and (0 != 0) and ((global_var(494) == 8) == 0)) then begin
-        gsay_reply(410, message_str(410, 320) + 5 + message_str(410, 2321));
+    if (((global_var(GVAR_NEW_RENO_MADE_MAN) / 100) != 0) and (0 != 0) and ((global_var(GVAR_FALLOUT_2) == 8) == 0)) then begin
+        Reply(mstr(320) + 5 + mstr(2321));
     end
     else begin
-        if ((global_var(494) == 8) or (global_var(232) > 0) or (global_var(259) > 0) or (global_var(230) > 0)) then begin
-            gsay_reply(410, message_str(410, 321 + (get_critter_stat(dude_obj, 34) == 1)) + 5 + message_str(410, 2321));
+        if ((global_var(GVAR_FALLOUT_2) == 8) or (global_var(GVAR_NEW_RENO_PORN_STAR) > 0) or (global_var(GVAR_NEW_RENO_HAS_REP_PRIZEFIGHTER) > 0) or (global_var(GVAR_NEW_RENO_MADE_MAN) > 0)) then begin
+            Reply(mstr(321 + (dude_gender == 1)) + 5 + mstr(2321));
         end
         else begin
-            gsay_reply(410, message_str(410, 323) + 5 + message_str(410, 2321));
+            Reply(mstr(323) + 5 + mstr(2321));
         end
     end
-    giq_option(-3, 410, 324, Node014, 50);
-    giq_option(-3, 410, 325, Node014, 50);
-    giq_option(-3, 410, 326, Node999, 50);
+    giq_option(-3, NAME, 324, Node014, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 325, Node014, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 326, Node999, NEUTRAL_REACTION);
     if (item_caps_total(dude_obj) >= 5) then begin
-        giq_option(4, 410, message_str(410, 328) + 5 + message_str(410, 2328), Node991, 50);
+        giq_option(4, NAME, mstr(328) + 5 + mstr(2328), Node991, NEUTRAL_REACTION);
         if (item_caps_total(dude_obj) >= 15) then begin
-            giq_option(4, 410, message_str(410, 328) + 15 + message_str(410, 2328), Node992, 50);
+            giq_option(4, NAME, mstr(328) + 15 + mstr(2328), Node992, NEUTRAL_REACTION);
             if (item_caps_total(dude_obj) >= 25) then begin
-                giq_option(4, 410, message_str(410, 328) + 25 + message_str(410, 2328), Node993, 50);
+                giq_option(4, NAME, mstr(328) + 25 + mstr(2328), Node993, NEUTRAL_REACTION);
                 if (item_caps_total(dude_obj) >= 50) then begin
-                    giq_option(4, 410, message_str(410, 328) + 50 + message_str(410, 2328), Node994, 50);
+                    giq_option(4, NAME, mstr(328) + 50 + mstr(2328), Node994, NEUTRAL_REACTION);
                     if (item_caps_total(dude_obj) >= 100) then begin
-                        giq_option(4, 410, message_str(410, 328) + 100 + message_str(410, 2328), Node995, 50);
+                        giq_option(4, NAME, mstr(328) + 100 + mstr(2328), Node995, NEUTRAL_REACTION);
                     end
                 end
             end
         end
     end
     else begin
-        giq_option(4, 410, 329, Node015, 50);
+        giq_option(4, NAME, 329, Node015, NEUTRAL_REACTION);
     end
-    giq_option(4, 410, 330, Node999, 50);
+    giq_option(4, NAME, 330, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node014
 begin
-    gsay_reply(410, 340 + ((global_var(230) > 0) * ((get_critter_stat(dude_obj, 34) == 0) + (2 * (get_critter_stat(dude_obj, 34) == 1)))));
-    giq_option(1, 410, 343, Node999, 50);
-    giq_option(1, 410, message_str(410, 344) + obj_name(dude_obj) + message_str(410, 2344), Node999, 50);
-    giq_option(1, 410, 345, Node999, 50);
+    Reply(340 + ((global_var(GVAR_NEW_RENO_MADE_MAN) > 0) * ((dude_gender == 0) + (2 * (dude_gender == 1)))));
+    giq_option(1, NAME, 343, Node999, NEUTRAL_REACTION);
+    giq_option(1, NAME, mstr(344) + dude_name + mstr(2344), Node999, NEUTRAL_REACTION);
+    giq_option(1, NAME, 345, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node015
 begin
-    if ((global_var(494) == 8) or (global_var(232) > 0) or (global_var(259) > 0) or (global_var(230) > 0)) then begin
-        gsay_reply(410, 350 + (get_critter_stat(dude_obj, 34) == 1));
-        giq_option(1, 410, 354, Node999, 50);
+    if ((global_var(GVAR_FALLOUT_2) == 8) or (global_var(GVAR_NEW_RENO_PORN_STAR) > 0) or (global_var(GVAR_NEW_RENO_HAS_REP_PRIZEFIGHTER) > 0) or (global_var(GVAR_NEW_RENO_MADE_MAN) > 0)) then begin
+        Reply(350 + (dude_gender == 1));
+        giq_option(1, NAME, 354, Node999, NEUTRAL_REACTION);
     end
     else begin
-        gsay_reply(410, message_str(410, 352) + 5 + message_str(410, 2352 + (get_critter_stat(dude_obj, 34) == 1)));
-        giq_option(1, 410, 355, Node999, 50);
+        Reply(mstr(352) + 5 + mstr(2352 + (dude_gender == 1)));
+        giq_option(1, NAME, 355, Node999, NEUTRAL_REACTION);
     end
 end
 
@@ -889,30 +889,30 @@ begin
         end
     end
     item_caps_adjust(dude_obj, LVar2 * bet_amt);
-    gsay_reply(410, 400 + (((global_var(494) == 8) or (global_var(232) > 0) or (global_var(259) > 0) or (global_var(230) > 0)) * ((get_critter_stat(dude_obj, 34) == 0) + (2 * (get_critter_stat(dude_obj, 34) == 1)))) + (10 * LVar2));
-    giq_option(-3, 410, 324, Node014, 50);
-    giq_option(-3, 410, 325, Node014, 50);
-    giq_option(-3, 410, 326, Node999, 50);
+    Reply(400 + (((global_var(GVAR_FALLOUT_2) == 8) or (global_var(GVAR_NEW_RENO_PORN_STAR) > 0) or (global_var(GVAR_NEW_RENO_HAS_REP_PRIZEFIGHTER) > 0) or (global_var(GVAR_NEW_RENO_MADE_MAN) > 0)) * ((dude_gender == 0) + (2 * (dude_gender == 1)))) + (10 * LVar2));
+    giq_option(-3, NAME, 324, Node014, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 325, Node014, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 326, Node999, NEUTRAL_REACTION);
     if (item_caps_total(dude_obj) >= 5) then begin
-        giq_option(4, 410, message_str(410, 328) + 5 + message_str(410, 2328), Node991, 50);
+        giq_option(4, NAME, mstr(328) + 5 + mstr(2328), Node991, NEUTRAL_REACTION);
         if (item_caps_total(dude_obj) >= 15) then begin
-            giq_option(4, 410, message_str(410, 328) + 15 + message_str(410, 2328), Node992, 50);
+            giq_option(4, NAME, mstr(328) + 15 + mstr(2328), Node992, NEUTRAL_REACTION);
             if (item_caps_total(dude_obj) >= 25) then begin
-                giq_option(4, 410, message_str(410, 328) + 25 + message_str(410, 2328), Node993, 50);
+                giq_option(4, NAME, mstr(328) + 25 + mstr(2328), Node993, NEUTRAL_REACTION);
                 if (item_caps_total(dude_obj) >= 50) then begin
-                    giq_option(4, 410, message_str(410, 328) + 50 + message_str(410, 2328), Node994, 50);
+                    giq_option(4, NAME, mstr(328) + 50 + mstr(2328), Node994, NEUTRAL_REACTION);
                     if (item_caps_total(dude_obj) >= 100) then begin
-                        giq_option(4, 410, message_str(410, 328) + 100 + message_str(410, 2328), Node995, 50);
+                        giq_option(4, NAME, mstr(328) + 100 + mstr(2328), Node995, NEUTRAL_REACTION);
                     end
                 end
             end
         end
     end
     else begin
-        giq_option(4, 410, 329, Node015, 50);
+        giq_option(4, NAME, 329, Node015, NEUTRAL_REACTION);
     end
-    giq_option(4, 410, 330, Node999, 50);
-    giq_option(1, 410, 440 + ((game_time_hour <= 600) or (game_time_hour >= 1800)), Node999, 50);
+    giq_option(4, NAME, 330, Node999, NEUTRAL_REACTION);
+    giq_option(1, NAME, 440 + ((game_time_hour <= 600) or (game_time_hour >= 1800)), Node999, NEUTRAL_REACTION);
 end
 
 procedure Node017
@@ -928,17 +928,17 @@ begin
             restock_obj := obj_carrying_pid_obj(self_obj, 325);
             if (obj_type(self_obj) == 1) then begin
                 restock_obj := obj_carrying_pid_obj(self_obj, 325);
-                if (critter_inven_obj(self_obj, 0) == restock_obj) then begin
-                    if (obj_item_subtype(critter_inven_obj(self_obj, 0)) == 0) then begin
-                        restock_obj := critter_inven_obj(self_obj, 0);
+                if (critter_inven_obj(self_obj, INVEN_TYPE_WORN) == restock_obj) then begin
+                    if (obj_item_subtype(critter_inven_obj(self_obj, INVEN_TYPE_WORN)) == 0) then begin
+                        restock_obj := critter_inven_obj(self_obj, INVEN_TYPE_WORN);
                         debug_msg("armour pid == " + obj_pid(restock_obj));
                         rm_obj_from_inven(self_obj, restock_obj);
                         add_obj_to_inven(self_obj, restock_obj);
                     end
                 end
                 else begin
-                    if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                        metarule(43, self_obj);
+                    if ((critter_inven_obj(self_obj, INVEN_TYPE_LEFT_HAND) == restock_obj) or (critter_inven_obj(self_obj, INVEN_TYPE_RIGHT_HAND) == restock_obj)) then begin
+                        inven_unwield(self_obj);
                     end
                 end
             end
@@ -955,17 +955,17 @@ begin
             restock_obj := obj_carrying_pid_obj(self_obj, 326);
             if (obj_type(self_obj) == 1) then begin
                 restock_obj := obj_carrying_pid_obj(self_obj, 326);
-                if (critter_inven_obj(self_obj, 0) == restock_obj) then begin
-                    if (obj_item_subtype(critter_inven_obj(self_obj, 0)) == 0) then begin
-                        restock_obj := critter_inven_obj(self_obj, 0);
+                if (critter_inven_obj(self_obj, INVEN_TYPE_WORN) == restock_obj) then begin
+                    if (obj_item_subtype(critter_inven_obj(self_obj, INVEN_TYPE_WORN)) == 0) then begin
+                        restock_obj := critter_inven_obj(self_obj, INVEN_TYPE_WORN);
                         debug_msg("armour pid == " + obj_pid(restock_obj));
                         rm_obj_from_inven(self_obj, restock_obj);
                         add_obj_to_inven(self_obj, restock_obj);
                     end
                 end
                 else begin
-                    if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                        metarule(43, self_obj);
+                    if ((critter_inven_obj(self_obj, INVEN_TYPE_LEFT_HAND) == restock_obj) or (critter_inven_obj(self_obj, INVEN_TYPE_RIGHT_HAND) == restock_obj)) then begin
+                        inven_unwield(self_obj);
                     end
                 end
             end
@@ -976,7 +976,7 @@ begin
         restock_amt := 0;
     end
     debug_msg("   a");
-    float_msg(self_obj, message_str(410, 450 + (((global_var(494) == 8) or (global_var(232) > 0) or (global_var(259) > 0) or (global_var(230) > 0)) * ((get_critter_stat(dude_obj, 34) == 0) + (2 * (get_critter_stat(dude_obj, 34) == 1))))), 2);
+    float_msg(self_obj, mstr(450 + (((global_var(GVAR_FALLOUT_2) == 8) or (global_var(GVAR_NEW_RENO_PORN_STAR) > 0) or (global_var(GVAR_NEW_RENO_HAS_REP_PRIZEFIGHTER) > 0) or (global_var(GVAR_NEW_RENO_MADE_MAN) > 0)) * ((dude_gender == 0) + (2 * (dude_gender == 1))))), FLOAT_MSG_RED);
     debug_msg("   b");
 end
 
@@ -1015,10 +1015,9 @@ begin
     variable LVar3 := 0;
     LVar3 := random(arg0, arg1);
     if ((LVar3 == 203) or (LVar3 == 215) or (LVar3 == 219) or (LVar3 == 255) or (LVar3 == 258) or ((260 <= LVar3) and (LVar3 <= 263)) or (LVar3 == 261) or (LVar3 == 262)) then begin
-        float_msg(self_obj, "" + message_str(410, LVar3) + arg2 + message_str(410, LVar3 + 2000), 8);
+        float_msg(self_obj, "" + mstr(LVar3) + arg2 + mstr(LVar3 + 2000), FLOAT_MSG_YELLOW);
     end
     else begin
-        float_msg(self_obj, message_str(410, LVar3), 2);
+        float_msg(self_obj, mstr(LVar3), FLOAT_MSG_RED);
     end
 end
-
