diff --git a/Mods/MegaMod/Original/ZINUKE.ssl b/Mods/MegaMod/ReDefine/ZINUKE.ssl
index 34eb779..5e5ce33 100644
--- a/Mods/MegaMod/Original/ZINUKE.ssl
+++ b/Mods/MegaMod/ReDefine/ZINUKE.ssl
@@ -43,7 +43,7 @@ end
 procedure description_p_proc
 begin
     script_overrides;
-    display_msg(message_str(1273, 100));
+    display_mstr(100);
 end
 
 procedure use_skill_on_p_proc
@@ -60,18 +60,18 @@ begin
         if (local_var(0) > 0) then begin
             set_local_var(0, local_var(0) - 1);
             item_caps_adjust(source_obj, -1);
-            display_msg(message_str(1273, 101));
+            display_mstr(101);
             if (not(is_success(do_check(source_obj, 6, 0)))) then begin
-                projectile := create_object_sid(83886086, tile_num(self_obj), elevation(self_obj), -1);
+                projectile := create_object(83886086, tile_num(self_obj), elevation(self_obj));
                 call checkHit();
             end
             else begin
-                display_msg(message_str(1273, 102));
-                add_obj_to_inven(source_obj, create_object_sid(106, 0, 0, -1));
+                display_mstr(102);
+                add_obj_to_inven(source_obj, create_object(106, 0, 0));
             end
         end
         else begin
-            display_msg(message_str(1273, 106));
+            display_mstr(106);
         end
     end
     return 0;
@@ -108,10 +108,10 @@ procedure checkHit
 begin
     if (is_success(do_check(source_obj, 5, 0))) then begin
         if (source_obj == dude_obj) then begin
-            display_msg(message_str(1273, 104));
+            display_mstr(104);
         end
         else begin
-            display_msg(obj_name(source_obj) + message_str(1273, 105));
+            display_msg(obj_name(source_obj) + mstr(105));
         end
         Final_Hex := tile_num_in_direction(tile_num(source_obj), rotation_to_tile(tile_num(self_obj), tile_num(source_obj)), 3);
         call Missile_Fired();
@@ -119,7 +119,7 @@ begin
     else begin
         Final_Hex := tile_num(source_obj);
         call Missile_Fired();
-        float_msg(dude_obj, message_str(1273, 103), 2);
+        float_msg(dude_obj, mstr(103), FLOAT_MSG_RED);
         critter_dmg(source_obj, random(1, 2), 256 bwor 512);
     end
     add_timer_event(self_obj, game_ticks(0), 1);
@@ -154,9 +154,8 @@ begin
     if (fixed_param == 1) then begin
         destroy_object(projectile);
         if (106 != 0) then begin
-            projectile := create_object_sid(106, Final_Hex, elevation(self_obj), -1);
+            projectile := create_object(106, Final_Hex, elevation(self_obj));
             anim(projectile, 1000, rotation_to_tile(tile_num(self_obj), Final_Hex));
         end
     end
 end
-
