diff --git a/Games/Fallout1.Steam/Original/ABEL.ssl b/Games/Fallout1.Steam/ReDefine/ABEL.ssl
index f93b8dd..d7683ab 100644
--- a/Games/Fallout1.Steam/Original/ABEL.ssl
+++ b/Games/Fallout1.Steam/ReDefine/ABEL.ssl
@@ -39,23 +39,23 @@ variable exit_line;
 procedure start
 begin
     if (not(initialized)) then begin
-        critter_add_trait(self_obj, 1, 6, 34);
-        critter_add_trait(self_obj, 1, 5, 48);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 34);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_SUPER_MUTANT_AGGRESSIVE);
         initialized := 1;
     end
-    if (script_action == 12) then begin
+    if (script_action == critter_proc) then begin
         call critter_p_proc();
     end
     else begin
-        if (script_action == 18) then begin
+        if (script_action == destroy_proc) then begin
             call destroy_p_proc();
         end
         else begin
-            if (script_action == 4) then begin
+            if (script_action == pickup_proc) then begin
                 call pickup_p_proc();
             end
             else begin
-                if (script_action == 11) then begin
+                if (script_action == talk_proc) then begin
                     call talk_p_proc();
                 end
             end
@@ -67,7 +67,7 @@ procedure critter_p_proc
 begin
     if (hostile) then begin
         hostile := 0;
-        attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+        attack(dude_obj);
     end
 end
 
@@ -103,17 +103,17 @@ end
 procedure destroy_p_proc
 begin
     if (source_obj == dude_obj) then begin
-        if (((global_var(160) + global_var(159)) >= 25) and ((global_var(159) > (2 * global_var(160))) or (global_var(156) == 1))) then begin
-            set_global_var(156, 1);
-            set_global_var(157, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
+            set_global_var(GVAR_BERSERKER_REPUTATION, 1);
+            set_global_var(GVAR_CHAMPION_REPUTATION, 0);
         end
-        if (((global_var(160) + global_var(159)) >= 25) and ((global_var(160) > (3 * global_var(159))) or (global_var(157) == 1))) then begin
-            set_global_var(157, 1);
-            set_global_var(156, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(160) > (3 * global_var(GVAR_GOOD_MONSTER))) or (global_var(GVAR_CHAMPION_REPUTATION) == 1))) then begin
+            set_global_var(GVAR_CHAMPION_REPUTATION, 1);
+            set_global_var(GVAR_BERSERKER_REPUTATION, 0);
         end
-        set_global_var(160, global_var(160) + 1);
-        if ((global_var(160) % 6) == 0) then begin
-            set_global_var(155, global_var(155) + 1);
+        set_global_var(GVAR_BAD_MONSTER, global_var(GVAR_BAD_MONSTER) + 1);
+        if ((global_var(GVAR_BAD_MONSTER) % 6) == 0) then begin
+            set_global_var(GVAR_PLAYER_REPUATION, global_var(GVAR_PLAYER_REPUATION) + 1);
         end
     end
 end
@@ -121,57 +121,57 @@ end
 procedure Abel00
 begin
     set_local_var(4, 1);
-    gsay_reply(359, 102);
-    giq_option(-3, 359, 103, Abel01, 0);
-    giq_option(4, 359, 104, Abel03, 0);
-    giq_option(5, 359, 105, Abel04, 0);
+    Reply(102);
+    giq_option(-3, NAME, 103, Abel01, 0);
+    giq_option(4, NAME, 104, Abel03, 0);
+    giq_option(5, NAME, 105, Abel04, 0);
 end
 
 procedure Abel01
 begin
-    gsay_reply(359, 106);
-    giq_option(-3, 359, 107, Abel02, 0);
+    Reply(106);
+    giq_option(-3, NAME, 107, Abel02, 0);
 end
 
 procedure Abel02
 begin
-    gsay_reply(359, 108);
-    giq_option(-3, 359, 109, Abel03, 0);
+    Reply(108);
+    giq_option(-3, NAME, 109, Abel03, 0);
 end
 
 procedure Abel03
 begin
-    gsay_message(359, 110, 0);
+    gsay_message(NAME, 110, 0);
     call AbelCombat();
 end
 
 procedure Abel04
 begin
-    gsay_reply(359, 111);
-    giq_option(5, 359, 112, Abel05, 0);
+    Reply(111);
+    giq_option(5, NAME, 112, Abel05, 0);
 end
 
 procedure Abel05
 begin
-    gsay_message(359, 113, 0);
+    gsay_message(NAME, 113, 0);
     call AbelCombat();
 end
 
 procedure Abel06
 begin
-    gsay_reply(359, 114);
-    giq_option(-3, 359, 115, Abel01, 0);
-    giq_option(4, 359, 116, Abel07, 0);
+    Reply(114);
+    giq_option(-3, NAME, 115, Abel01, 0);
+    giq_option(4, NAME, 116, Abel07, 0);
 end
 
 procedure Abel07
 begin
-    gsay_message(359, 117, 0);
+    gsay_message(NAME, 117, 0);
 end
 
 procedure Abel08
 begin
-    gsay_message(359, 118, 0);
+    gsay_message(NAME, 118, 0);
     call AbelCombat();
 end
 
@@ -189,31 +189,31 @@ begin
         set_local_var(0, 50);
         set_local_var(1, 2);
         set_local_var(2, 1);
-        set_local_var(0, local_var(0) + (5 * get_critter_stat(dude_obj, 3)) - 25);
-        set_local_var(0, local_var(0) + (10 * has_trait(0, dude_obj, 10)));
-        if (has_trait(0, dude_obj, 39)) then begin
-            if (global_var(155) > 0) then begin
-                set_local_var(0, local_var(0) + global_var(155));
+        set_local_var(0, local_var(0) + (5 * dude_charisma) - 25);
+        set_local_var(0, local_var(0) + (10 * has_trait(TRAIT_PERK, dude_obj, PERK_presence)));
+        if (has_trait(TRAIT_PERK, dude_obj, PERK_cult_of_personality)) then begin
+            if (global_var(GVAR_PLAYER_REPUATION) > 0) then begin
+                set_local_var(0, local_var(0) + global_var(GVAR_PLAYER_REPUATION));
             end
             else begin
-                set_local_var(0, local_var(0) - global_var(155));
+                set_local_var(0, local_var(0) - global_var(GVAR_PLAYER_REPUATION));
             end
         end
         else begin
             if (local_var(3) == 1) then begin
-                set_local_var(0, local_var(0) - global_var(155));
+                set_local_var(0, local_var(0) - global_var(GVAR_PLAYER_REPUATION));
             end
             else begin
-                set_local_var(0, local_var(0) + global_var(155));
+                set_local_var(0, local_var(0) + global_var(GVAR_PLAYER_REPUATION));
             end
         end
-        if (global_var(158) > 2) then begin
+        if (global_var(GVAR_CHILDKILLER_REPUATION) > 2) then begin
             set_local_var(0, local_var(0) - 30);
         end
-        if (((global_var(160) + global_var(159)) >= 25) and ((global_var(160) > (3 * global_var(159))) or (global_var(157) == 1))) then begin
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(160) > (3 * global_var(GVAR_GOOD_MONSTER))) or (global_var(GVAR_CHAMPION_REPUTATION) == 1))) then begin
             set_local_var(0, local_var(0) + 20);
         end
-        if (((global_var(160) + global_var(159)) >= 25) and ((global_var(159) > (2 * global_var(160))) or (global_var(156) == 1))) then begin
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
             set_local_var(0, local_var(0) - 20);
         end
         call ReactToLevel();
@@ -306,6 +306,5 @@ end
 
 procedure Goodbyes
 begin
-    exit_line := message_str(634, random(100, 105));
+    exit_line := message_str(SCRIPT_MODREACT, random(100, 105));
 end
-
