diff --git a/Games/Fallout1.Steam/Original/AGATHA.ssl b/Games/Fallout1.Steam/ReDefine/AGATHA.ssl
index d7a78d8..f0258eb 100644
--- a/Games/Fallout1.Steam/Original/AGATHA.ssl
+++ b/Games/Fallout1.Steam/ReDefine/AGATHA.ssl
@@ -49,31 +49,31 @@ variable exit_line;
 procedure start
 begin
     if (not(initialized)) then begin
-        critter_add_trait(self_obj, 1, 6, 2);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 2);
         initialized := 1;
     end
     else begin
-        if (script_action == 12) then begin
+        if (script_action == critter_proc) then begin
             call critter_p_proc();
         end
         else begin
-            if (script_action == 3) then begin
+            if (script_action == description_proc) then begin
                 call description_p_proc();
             end
             else begin
-                if (script_action == 18) then begin
+                if (script_action == destroy_proc) then begin
                     call destroy_p_proc();
                 end
                 else begin
-                    if (script_action == 21) then begin
+                    if (script_action == look_at_proc) then begin
                         call look_at_p_proc();
                     end
                     else begin
-                        if (script_action == 11) then begin
+                        if (script_action == talk_proc) then begin
                             call talk_p_proc();
                         end
                         else begin
-                            if (script_action == 22) then begin
+                            if (script_action == timed_event_proc) then begin
                                 call timed_event_p_proc();
                             end
                         end
@@ -90,26 +90,26 @@ end
 
 procedure description_p_proc
 begin
-    if (is_success(do_check(dude_obj, 1, has_trait(0, dude_obj, 0)))) then begin
+    if (is_success(do_check(dude_obj, 1, has_trait(TRAIT_PERK, dude_obj, PERK_bonus_awareness)))) then begin
         script_overrides;
-        display_msg(message_str(663, 102));
+        display_mstr(102);
     end
 end
 
 procedure destroy_p_proc
 begin
     if (source_obj == dude_obj) then begin
-        if (((global_var(160) + global_var(159)) >= 25) and ((global_var(159) > (2 * global_var(160))) or (global_var(156) == 1))) then begin
-            set_global_var(156, 1);
-            set_global_var(157, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
+            set_global_var(GVAR_BERSERKER_REPUTATION, 1);
+            set_global_var(GVAR_CHAMPION_REPUTATION, 0);
         end
-        if (((global_var(160) + global_var(159)) >= 25) and ((global_var(160) > (3 * global_var(159))) or (global_var(157) == 1))) then begin
-            set_global_var(157, 1);
-            set_global_var(156, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(160) > (3 * global_var(GVAR_GOOD_MONSTER))) or (global_var(GVAR_CHAMPION_REPUTATION) == 1))) then begin
+            set_global_var(GVAR_CHAMPION_REPUTATION, 1);
+            set_global_var(GVAR_BERSERKER_REPUTATION, 0);
         end
-        set_global_var(159, global_var(159) + 1);
-        if ((global_var(159) % 2) == 0) then begin
-            set_global_var(155, global_var(155) - 1);
+        set_global_var(GVAR_GOOD_MONSTER, global_var(GVAR_GOOD_MONSTER) + 1);
+        if ((global_var(GVAR_GOOD_MONSTER) % 2) == 0) then begin
+            set_global_var(GVAR_PLAYER_REPUATION, global_var(GVAR_PLAYER_REPUATION) - 1);
         end
     end
 end
@@ -118,10 +118,10 @@ procedure look_at_p_proc
 begin
     script_overrides;
     if (local_var(4)) then begin
-        display_msg(message_str(663, 101));
+        display_mstr(101);
     end
     else begin
-        display_msg(message_str(663, 100));
+        display_mstr(100);
     end
 end
 
@@ -155,135 +155,135 @@ end
 
 procedure Agatha00
 begin
-    float_msg(self_obj, message_str(663, 103), 0);
+    float_msg(self_obj, mstr(103), FLOAT_MSG_NORMAL);
 end
 
 procedure Agatha01
 begin
-    gsay_reply(663, 104);
-    giq_option(-3, 663, 105, Agatha02, 49);
-    giq_option(4, 663, 106, AgathaEnd, 50);
-    giq_option(5, 663, 107, Agatha03, 50);
-    giq_option(4, 663, 108, Agatha04, 50);
+    Reply(104);
+    giq_option(-3, NAME, 105, Agatha02, GOOD_REACTION);
+    giq_option(4, NAME, 106, AgathaEnd, NEUTRAL_REACTION);
+    giq_option(5, NAME, 107, Agatha03, NEUTRAL_REACTION);
+    giq_option(4, NAME, 108, Agatha04, NEUTRAL_REACTION);
 end
 
 procedure Agatha02
 begin
-    gsay_message(663, 109, 49);
+    gsay_message(NAME, 109, GOOD_REACTION);
 end
 
 procedure Agatha03
 begin
-    gsay_reply(663, 110);
-    giq_option(5, 663, 111, AgathaEnd, 49);
-    giq_option(6, 663, 112, Agatha04, 50);
-    giq_option(5, 663, 113, AgathaEnd, 50);
+    Reply(110);
+    giq_option(5, NAME, 111, AgathaEnd, GOOD_REACTION);
+    giq_option(6, NAME, 112, Agatha04, NEUTRAL_REACTION);
+    giq_option(5, NAME, 113, AgathaEnd, NEUTRAL_REACTION);
 end
 
 procedure Agatha04
 begin
-    gsay_reply(663, 114);
-    giq_option(5, 663, 115, AgathaEnd, 49);
+    Reply(114);
+    giq_option(5, NAME, 115, AgathaEnd, GOOD_REACTION);
 end
 
 procedure Agatha05
 begin
-    gsay_message(663, 116, 50);
+    gsay_message(NAME, 116, NEUTRAL_REACTION);
 end
 
 procedure Agatha06
 begin
     call UpReact();
-    gsay_message(663, 117, 49);
+    gsay_message(NAME, 117, GOOD_REACTION);
 end
 
 procedure Agatha07
 begin
-    gsay_message(663, 118, 50);
+    gsay_message(NAME, 118, NEUTRAL_REACTION);
 end
 
 procedure Agatha08
 begin
-    gsay_reply(663, 119);
-    giq_option(-3, 663, 120, Agatha09, 50);
-    giq_option(4, 663, 121, Agatha10, 50);
-    if (get_critter_stat(dude_obj, 34) == 1) then begin
-        giq_option(5, 663, 122, Agatha13, 50);
+    Reply(119);
+    giq_option(-3, NAME, 120, Agatha09, NEUTRAL_REACTION);
+    giq_option(4, NAME, 121, Agatha10, NEUTRAL_REACTION);
+    if (dude_gender == 1) then begin
+        giq_option(5, NAME, 122, Agatha13, NEUTRAL_REACTION);
     end
     else begin
-        giq_option(7, 663, 123, Agatha13, 50);
+        giq_option(7, NAME, 123, Agatha13, NEUTRAL_REACTION);
     end
-    giq_option(4, 663, 124, Agatha09, 51);
+    giq_option(4, NAME, 124, Agatha09, BAD_REACTION);
 end
 
 procedure Agatha09
 begin
-    gsay_message(663, 125, 50);
+    gsay_message(NAME, 125, NEUTRAL_REACTION);
 end
 
 procedure Agatha10
 begin
-    gsay_reply(663, 126);
-    giq_option(4, 663, 127, Agatha11, 50);
-    giq_option(4, 663, 128, AgathaEnd, 50);
-    giq_option(6, 663, 129, Agatha12, 49);
+    Reply(126);
+    giq_option(4, NAME, 127, Agatha11, NEUTRAL_REACTION);
+    giq_option(4, NAME, 128, AgathaEnd, NEUTRAL_REACTION);
+    giq_option(6, NAME, 129, Agatha12, GOOD_REACTION);
 end
 
 procedure Agatha11
 begin
-    gsay_message(663, 130, 50);
+    gsay_message(NAME, 130, NEUTRAL_REACTION);
 end
 
 procedure Agatha12
 begin
     call UpReact();
-    gsay_message(663, 131, 50);
+    gsay_message(NAME, 131, NEUTRAL_REACTION);
 end
 
 procedure Agatha13
 begin
-    gsay_message(663, 132, 50);
+    gsay_message(NAME, 132, NEUTRAL_REACTION);
 end
 
 procedure Agatha14
 begin
-    gsay_reply(663, 133);
-    giq_option(-3, 663, 134, AgathaEnd, 50);
-    giq_option(4, 663, 135, AgathaEnd, 50);
-    giq_option(6, 663, 136, Agatha15, 50);
+    Reply(133);
+    giq_option(-3, NAME, 134, AgathaEnd, NEUTRAL_REACTION);
+    giq_option(4, NAME, 135, AgathaEnd, NEUTRAL_REACTION);
+    giq_option(6, NAME, 136, Agatha15, NEUTRAL_REACTION);
 end
 
 procedure Agatha15
 begin
-    gsay_message(663, 137, 50);
+    gsay_message(NAME, 137, NEUTRAL_REACTION);
 end
 
 procedure Agatha16
 begin
-    gsay_reply(663, 138);
-    giq_option(-3, 663, 139, Agatha17, 50);
-    giq_option(4, 663, 140, Agatha20, 50);
-    giq_option(6, 663, 141, Agatha18, 50);
-    giq_option(4, 663, 142, AgathaEnd, 50);
+    Reply(138);
+    giq_option(-3, NAME, 139, Agatha17, NEUTRAL_REACTION);
+    giq_option(4, NAME, 140, Agatha20, NEUTRAL_REACTION);
+    giq_option(6, NAME, 141, Agatha18, NEUTRAL_REACTION);
+    giq_option(4, NAME, 142, AgathaEnd, NEUTRAL_REACTION);
 end
 
 procedure Agatha17
 begin
-    gsay_message(663, 143, 50);
+    gsay_message(NAME, 143, NEUTRAL_REACTION);
 end
 
 procedure Agatha18
 begin
-    gsay_reply(663, 144);
-    giq_option(5, 663, 145, Agatha20, 50);
-    giq_option(5, 663, 146, Agatha19, 51);
+    Reply(144);
+    giq_option(5, NAME, 145, Agatha20, NEUTRAL_REACTION);
+    giq_option(5, NAME, 146, Agatha19, BAD_REACTION);
 end
 
 procedure Agatha19
 begin
     call DownReactLevel();
     call DownReactLevel();
-    gsay_message(663, 147, 51);
+    gsay_message(NAME, 147, BAD_REACTION);
 end
 
 procedure Agatha20
@@ -300,31 +300,31 @@ begin
         set_local_var(0, 50);
         set_local_var(1, 2);
         set_local_var(2, 1);
-        set_local_var(0, local_var(0) + (5 * get_critter_stat(dude_obj, 3)) - 25);
-        set_local_var(0, local_var(0) + (10 * has_trait(0, dude_obj, 10)));
-        if (has_trait(0, dude_obj, 39)) then begin
-            if (global_var(155) > 0) then begin
-                set_local_var(0, local_var(0) + global_var(155));
+        set_local_var(0, local_var(0) + (5 * dude_charisma) - 25);
+        set_local_var(0, local_var(0) + (10 * has_trait(TRAIT_PERK, dude_obj, PERK_presence)));
+        if (has_trait(TRAIT_PERK, dude_obj, PERK_cult_of_personality)) then begin
+            if (global_var(GVAR_PLAYER_REPUATION) > 0) then begin
+                set_local_var(0, local_var(0) + global_var(GVAR_PLAYER_REPUATION));
             end
             else begin
-                set_local_var(0, local_var(0) - global_var(155));
+                set_local_var(0, local_var(0) - global_var(GVAR_PLAYER_REPUATION));
             end
         end
         else begin
             if (local_var(3) == 1) then begin
-                set_local_var(0, local_var(0) - global_var(155));
+                set_local_var(0, local_var(0) - global_var(GVAR_PLAYER_REPUATION));
             end
             else begin
-                set_local_var(0, local_var(0) + global_var(155));
+                set_local_var(0, local_var(0) + global_var(GVAR_PLAYER_REPUATION));
             end
         end
-        if (global_var(158) > 2) then begin
+        if (global_var(GVAR_CHILDKILLER_REPUATION) > 2) then begin
             set_local_var(0, local_var(0) - 30);
         end
-        if (((global_var(160) + global_var(159)) >= 25) and ((global_var(160) > (3 * global_var(159))) or (global_var(157) == 1))) then begin
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(160) > (3 * global_var(GVAR_GOOD_MONSTER))) or (global_var(GVAR_CHAMPION_REPUTATION) == 1))) then begin
             set_local_var(0, local_var(0) + 20);
         end
-        if (((global_var(160) + global_var(159)) >= 25) and ((global_var(159) > (2 * global_var(160))) or (global_var(156) == 1))) then begin
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
             set_local_var(0, local_var(0) - 20);
         end
         call ReactToLevel();
@@ -417,6 +417,5 @@ end
 
 procedure Goodbyes
 begin
-    exit_line := message_str(634, random(100, 105));
+    exit_line := message_str(SCRIPT_MODREACT, random(100, 105));
 end
-
