diff --git a/Games/Fallout1.Steam/Original/ARADESH.ssl b/Games/Fallout1.Steam/ReDefine/ARADESH.ssl
index af977d6..44b43b9 100644
--- a/Games/Fallout1.Steam/Original/ARADESH.ssl
+++ b/Games/Fallout1.Steam/ReDefine/ARADESH.ssl
@@ -137,16 +137,16 @@ procedure aradesh73;
 
 procedure start
 begin
-    critter_add_trait(self_obj, 1, 6, 2);
-    critter_add_trait(self_obj, 1, 5, 6);
-    if (script_action == 11) then begin
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 2);
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_PEASANT);
+    if (script_action == talk_proc) then begin
         if (local_var(9) > 2) then begin
-            display_msg(message_str(766, 170));
+            display_msg(message_str(SCRIPT_GENCHAT, 170));
             hostile := 1;
         end
         else begin
-            if (global_var(246) == 1) then begin
-                float_msg(self_obj, message_str(33, 277), 2);
+            if (global_var(GVAR_ENEMY_SHADY_SANDS) == 1) then begin
+                float_msg(self_obj, mstr(277), FLOAT_MSG_RED);
             end
             else begin
                 call do_dialogue();
@@ -157,48 +157,48 @@ begin
         end
     end
     else begin
-        if (script_action == 12) then begin
+        if (script_action == critter_proc) then begin
             if (obj_can_see_obj(self_obj, dude_obj)) then begin
-                if (global_var(246) == 1) then begin
+                if (global_var(GVAR_ENEMY_SHADY_SANDS) == 1) then begin
                     hostile := 1;
                 end
             end
             if (hostile) then begin
                 hostile := 0;
-                attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+                attack(dude_obj);
             end
         end
         else begin
-            if (script_action == 14) then begin
+            if (script_action == damage_proc) then begin
                 if (source_obj == dude_obj) then begin
-                    set_global_var(246, 1);
+                    set_global_var(GVAR_ENEMY_SHADY_SANDS, 1);
                 end
             end
             else begin
-                if (script_action == 21) then begin
+                if (script_action == look_at_proc) then begin
                     script_overrides;
-                    display_msg(message_str(33, 100));
+                    display_mstr(100);
                 end
                 else begin
-                    if (script_action == 18) then begin
-                        set_global_var(604, 1);
+                    if (script_action == destroy_proc) then begin
+                        set_global_var(GVAR_ARADESH_STATUS, 1);
                         if (source_obj == dude_obj) then begin
-                            if (((global_var(160) + global_var(159)) >= 25) and ((global_var(159) > (2 * global_var(160))) or (global_var(156) == 1))) then begin
-                                set_global_var(156, 1);
-                                set_global_var(157, 0);
+                            if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
+                                set_global_var(GVAR_BERSERKER_REPUTATION, 1);
+                                set_global_var(GVAR_CHAMPION_REPUTATION, 0);
                             end
-                            if (((global_var(160) + global_var(159)) >= 25) and ((global_var(160) > (3 * global_var(159))) or (global_var(157) == 1))) then begin
-                                set_global_var(157, 1);
-                                set_global_var(156, 0);
+                            if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(160) > (3 * global_var(GVAR_GOOD_MONSTER))) or (global_var(GVAR_CHAMPION_REPUTATION) == 1))) then begin
+                                set_global_var(GVAR_CHAMPION_REPUTATION, 1);
+                                set_global_var(GVAR_BERSERKER_REPUTATION, 0);
                             end
-                            set_global_var(159, global_var(159) + 1);
-                            if ((global_var(159) % 2) == 0) then begin
-                                set_global_var(155, global_var(155) - 1);
+                            set_global_var(GVAR_GOOD_MONSTER, global_var(GVAR_GOOD_MONSTER) + 1);
+                            if ((global_var(GVAR_GOOD_MONSTER) % 2) == 0) then begin
+                                set_global_var(GVAR_PLAYER_REPUATION, global_var(GVAR_PLAYER_REPUATION) - 1);
                             end
                         end
                     end
                     else begin
-                        if (script_action == 4) then begin
+                        if (script_action == pickup_proc) then begin
                             hostile := 1;
                         end
                     end
@@ -210,21 +210,21 @@ end
 
 procedure do_dialogue
 begin
-    if (global_var(246) == 1) then begin
-        float_msg(self_obj, message_str(33, 277), 2);
+    if (global_var(GVAR_ENEMY_SHADY_SANDS) == 1) then begin
+        float_msg(self_obj, mstr(277), FLOAT_MSG_RED);
     end
     else begin
         if (local_var(9) > 2) then begin
-            display_msg(message_str(766, 170));
+            display_msg(message_str(SCRIPT_GENCHAT, 170));
             hostile := 1;
         end
         else begin
             call get_reaction();
             start_gdialog(33, self_obj, 4, 13, 9);
             gsay_start;
-            if (global_var(26) == 1) then begin
+            if (global_var(GVAR_TANDI_STATUS) == 1) then begin
                 set_local_var(4, 1);
-                if (global_var(103) == 0) then begin
+                if (global_var(GVAR_RESCUE_TANDI) == 0) then begin
                     call aradesh39();
                 end
                 else begin
@@ -237,17 +237,17 @@ begin
                         call aradesh46();
                     end
                     else begin
-                        if ((global_var(26) == 2) and (local_var(8) == 0)) then begin
+                        if ((global_var(GVAR_TANDI_STATUS) == 2) and (local_var(8) == 0)) then begin
                             call aradesh43();
                         end
                         else begin
-                            if ((global_var(43) == 2) and (local_var(10) == 0)) then begin
+                            if ((global_var(GVAR_RADSCORPION_SEED) == 2) and (local_var(10) == 0)) then begin
                                 set_local_var(10, 1);
                                 call aradesh31();
                             end
                             else begin
-                                if (global_var(26) == 1) then begin
-                                    if (global_var(103) == 0) then begin
+                                if (global_var(GVAR_TANDI_STATUS) == 1) then begin
+                                    if (global_var(GVAR_RESCUE_TANDI) == 0) then begin
                                         call aradesh39();
                                     end
                                     else begin
@@ -255,11 +255,11 @@ begin
                                     end
                                 end
                                 else begin
-                                    if ((global_var(26) == 2) and (global_var(43) != 2)) then begin
+                                    if ((global_var(GVAR_TANDI_STATUS) == 2) and (global_var(GVAR_RADSCORPION_SEED) != 2)) then begin
                                         call aradesh30a();
                                     end
                                     else begin
-                                        if (global_var(26) == 2) then begin
+                                        if (global_var(GVAR_TANDI_STATUS) == 2) then begin
                                             call aradesh35();
                                         end
                                         else begin
@@ -289,7 +289,7 @@ end
 
 procedure aradeshleave
 begin
-    gsay_message(33, 276, 50);
+    gsay_message(NAME, 276, NEUTRAL_REACTION);
 end
 
 procedure aradeshend
@@ -305,22 +305,22 @@ procedure aradesh01
 begin
     if (local_var(1) >= 2) then begin
         if (local_var(1) >= 3) then begin
-            gsay_reply(33, 101);
+            Reply(101);
         end
         else begin
-            gsay_reply(33, 102);
+            Reply(102);
         end
     end
     else begin
-        gsay_reply(33, 103);
+        Reply(103);
     end
-    giq_option(5, 33, 104, aradesh02, 50);
-    giq_option(4, 33, 105, aradesh01_2, 51);
-    giq_option(4, 33, 106, aradesh10, 51);
-    giq_option(4, 33, 107, aradesh01_3, 50);
-    giq_option(4, 33, 108, aradesh14, 51);
-    giq_option(5, 33, 109, aradesh07, 51);
-    giq_option(-3, 33, 110, aradesh15, 50);
+    giq_option(5, NAME, 104, aradesh02, NEUTRAL_REACTION);
+    giq_option(4, NAME, 105, aradesh01_2, BAD_REACTION);
+    giq_option(4, NAME, 106, aradesh10, BAD_REACTION);
+    giq_option(4, NAME, 107, aradesh01_3, NEUTRAL_REACTION);
+    giq_option(4, NAME, 108, aradesh14, BAD_REACTION);
+    giq_option(5, NAME, 109, aradesh07, BAD_REACTION);
+    giq_option(-3, NAME, 110, aradesh15, NEUTRAL_REACTION);
 end
 
 procedure aradesh01_2
@@ -347,11 +347,11 @@ end
 
 procedure aradesh02
 begin
-    gsay_reply(33, 111);
-    giq_option(4, 33, 112, aradesh02_2, 50);
-    giq_option(4, 33, 113, aradesh01_2, 51);
-    giq_option(5, 33, 114, aradesh02_4, 51);
-    giq_option(5, 33, 115, aradesh02_5, 50);
+    Reply(111);
+    giq_option(4, NAME, 112, aradesh02_2, NEUTRAL_REACTION);
+    giq_option(4, NAME, 113, aradesh01_2, BAD_REACTION);
+    giq_option(5, NAME, 114, aradesh02_4, BAD_REACTION);
+    giq_option(5, NAME, 115, aradesh02_5, NEUTRAL_REACTION);
 end
 
 procedure aradesh02_2
@@ -386,46 +386,46 @@ end
 
 procedure aradesh03
 begin
-    gsay_reply(33, 116);
-    giq_option(4, 33, 117, aradesh04, 50);
-    giq_option(4, 33, 118, aradesh01_2, 51);
-    giq_option(5, 33, 119, aradesh07, 51);
-    giq_option(4, 33, 120, aradesh07, 51);
+    Reply(116);
+    giq_option(4, NAME, 117, aradesh04, NEUTRAL_REACTION);
+    giq_option(4, NAME, 118, aradesh01_2, BAD_REACTION);
+    giq_option(5, NAME, 119, aradesh07, BAD_REACTION);
+    giq_option(4, NAME, 120, aradesh07, BAD_REACTION);
 end
 
 procedure aradesh04
 begin
-    gsay_message(33, 121, 50);
+    gsay_message(NAME, 121, NEUTRAL_REACTION);
     call aradeshend();
 end
 
 procedure aradesh05
 begin
-    gsay_reply(33, 122);
-    giq_option(5, 33, 123, aradesh06, 51);
-    giq_option(4, 33, 124, aradesh04, 50);
-    giq_option(4, 33, 125, aradesh07, 51);
+    Reply(122);
+    giq_option(5, NAME, 123, aradesh06, BAD_REACTION);
+    giq_option(4, NAME, 124, aradesh04, NEUTRAL_REACTION);
+    giq_option(4, NAME, 125, aradesh07, BAD_REACTION);
 end
 
 procedure aradesh06
 begin
     set_local_var(9, local_var(9) + 1);
-    gsay_message(33, 126, 51);
+    gsay_message(NAME, 126, BAD_REACTION);
     call aradeshx1();
 end
 
 procedure aradesh07
 begin
     set_local_var(9, local_var(9) + 1);
-    gsay_message(33, 127, 51);
+    gsay_message(NAME, 127, BAD_REACTION);
     call aradeshx2();
 end
 
 procedure aradesh08
 begin
-    gsay_reply(33, 128);
-    giq_option(4, 33, 129, aradesh08a, 49);
-    giq_option(5, 33, 130, aradesh08b, 51);
+    Reply(128);
+    giq_option(4, NAME, 129, aradesh08a, GOOD_REACTION);
+    giq_option(5, NAME, 130, aradesh08b, BAD_REACTION);
 end
 
 procedure aradesh08a
@@ -443,28 +443,28 @@ end
 
 procedure aradesh09
 begin
-    gsay_reply(33, 131);
-    giq_option(0, 33, 132, aradesh09a, 50);
+    Reply(131);
+    giq_option(0, NAME, 132, aradesh09a, NEUTRAL_REACTION);
 end
 
 procedure aradesh09a
 begin
-    gsay_message(33, 133, 50);
+    gsay_message(NAME, 133, NEUTRAL_REACTION);
     call aradeshx();
 end
 
 procedure aradesh10
 begin
-    gsay_message(33, 134, 50);
+    gsay_message(NAME, 134, NEUTRAL_REACTION);
     call aradeshx();
 end
 
 procedure aradesh11
 begin
-    gsay_reply(33, 135);
-    giq_option(4, 33, 136, aradesh12, 50);
-    giq_option(5, 33, 137, aradesh11a, 50);
-    giq_option(4, 33, 138, aradesh11b, 51);
+    Reply(135);
+    giq_option(4, NAME, 136, aradesh12, NEUTRAL_REACTION);
+    giq_option(5, NAME, 137, aradesh11a, NEUTRAL_REACTION);
+    giq_option(4, NAME, 138, aradesh11b, BAD_REACTION);
 end
 
 procedure aradesh11a
@@ -486,57 +486,57 @@ end
 
 procedure aradesh12
 begin
-    gsay_reply(33, 139);
-    giq_option(5, 33, 140, aradesh02, 50);
-    giq_option(4, 33, 141, aradesh13, 50);
-    giq_option(4, 33, 142, aradesh05, 50);
+    Reply(139);
+    giq_option(5, NAME, 140, aradesh02, NEUTRAL_REACTION);
+    giq_option(4, NAME, 141, aradesh13, NEUTRAL_REACTION);
+    giq_option(4, NAME, 142, aradesh05, NEUTRAL_REACTION);
 end
 
 procedure aradesh13
 begin
-    gsay_reply(33, 143);
-    giq_option(4, 33, 160, aradesh19, 50);
-    giq_option(4, 33, 151, aradesh24c, 50);
-    giq_option(-3, 33, 156, aradesh36, 50);
+    Reply(143);
+    giq_option(4, NAME, 160, aradesh19, NEUTRAL_REACTION);
+    giq_option(4, NAME, 151, aradesh24c, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 156, aradesh36, NEUTRAL_REACTION);
 end
 
 procedure aradesh14
 begin
-    gsay_reply(33, 144);
-    giq_option(5, 33, 145, aradesh06, 51);
-    giq_option(4, 33, 146, aradesh04, 50);
-    giq_option(4, 33, 147, aradesh07, 51);
+    Reply(144);
+    giq_option(5, NAME, 145, aradesh06, BAD_REACTION);
+    giq_option(4, NAME, 146, aradesh04, NEUTRAL_REACTION);
+    giq_option(4, NAME, 147, aradesh07, BAD_REACTION);
 end
 
 procedure aradesh15
 begin
-    gsay_message(33, 148, 50);
+    gsay_message(NAME, 148, NEUTRAL_REACTION);
     call aradeshx();
 end
 
 procedure aradesh16
 begin
     gsay_reply(0, 0);
-    giq_option(8, 33, 149, aradesh25, 50);
-    giq_option(5, 33, 150, aradesh16_2, 50);
-    giq_option(5, 33, 151, aradesh24c, 50);
+    giq_option(8, NAME, 149, aradesh25, NEUTRAL_REACTION);
+    giq_option(5, NAME, 150, aradesh16_2, NEUTRAL_REACTION);
+    giq_option(5, NAME, 151, aradesh24c, NEUTRAL_REACTION);
 end
 
 procedure aradesh16_2
 begin
-    if (global_var(43) == 0) then begin
+    if (global_var(GVAR_RADSCORPION_SEED) == 0) then begin
         call aradesh28();
     end
     else begin
-        if (global_var(43) == 1) then begin
+        if (global_var(GVAR_RADSCORPION_SEED) == 1) then begin
             call aradesh30a();
         end
         else begin
-            if ((global_var(43) == 2) and not(global_var(26) == 2)) then begin
+            if ((global_var(GVAR_RADSCORPION_SEED) == 2) and not(global_var(GVAR_TANDI_STATUS) == 2)) then begin
                 call aradesh31();
             end
             else begin
-                if (global_var(26) == 2) then begin
+                if (global_var(GVAR_TANDI_STATUS) == 2) then begin
                     call aradesh35();
                 end
                 else begin
@@ -549,16 +549,16 @@ end
 
 procedure aradesh17
 begin
-    gsay_reply(33, 152);
-    giq_option(4, 33, 153, aradesh19, 50);
-    giq_option(4, 33, 154, aradesh17_2, 50);
-    giq_option(5, 33, 155, aradesh17_3, 51);
-    giq_option(-3, 33, 156, aradesh36, 50);
+    Reply(152);
+    giq_option(4, NAME, 153, aradesh19, NEUTRAL_REACTION);
+    giq_option(4, NAME, 154, aradesh17_2, NEUTRAL_REACTION);
+    giq_option(5, NAME, 155, aradesh17_3, BAD_REACTION);
+    giq_option(-3, NAME, 156, aradesh36, NEUTRAL_REACTION);
 end
 
 procedure aradesh17_2
 begin
-    if (global_var(26) == 2) then begin
+    if (global_var(GVAR_TANDI_STATUS) == 2) then begin
         call aradesh44();
     end
     else begin
@@ -570,7 +570,7 @@ procedure aradesh17_3
 begin
     set_local_var(9, local_var(9) + 1);
     call DownReact();
-    if (global_var(26) == 2) then begin
+    if (global_var(GVAR_TANDI_STATUS) == 2) then begin
         call aradesh27();
     end
     else begin
@@ -591,41 +591,41 @@ end
 procedure aradesh18
 begin
     if (local_var(1) >= 2) then begin
-        gsay_reply(33, 157);
+        Reply(157);
     end
     else begin
-        gsay_reply(33, 158);
+        Reply(158);
     end
-    giq_option(5, 33, 159, aradesh23, 50);
-    giq_option(4, 33, 160, aradesh19, 50);
-    giq_option(4, 33, 161, aradesh17_2, 50);
-    giq_option(-3, 33, 162, aradesh36, 50);
+    giq_option(5, NAME, 159, aradesh23, NEUTRAL_REACTION);
+    giq_option(4, NAME, 160, aradesh19, NEUTRAL_REACTION);
+    giq_option(4, NAME, 161, aradesh17_2, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 162, aradesh36, NEUTRAL_REACTION);
 end
 
 procedure aradesh19
 begin
     if (local_var(1) >= 2) then begin
-        gsay_reply(33, 163);
+        Reply(163);
     end
     else begin
-        gsay_reply(33, 164);
+        Reply(164);
     end
-    giq_option(5, 33, 165, aradesh25, 50);
-    giq_option(4, 33, 166, aradesh16_2, 50);
-    giq_option(4, 33, 167, aradesh24c, 50);
+    giq_option(5, NAME, 165, aradesh25, NEUTRAL_REACTION);
+    giq_option(4, NAME, 166, aradesh16_2, NEUTRAL_REACTION);
+    giq_option(4, NAME, 167, aradesh24c, NEUTRAL_REACTION);
 end
 
 procedure aradesh20
 begin
     if (local_var(1) >= 3) then begin
-        gsay_message(33, 168, 50);
+        gsay_message(NAME, 168, NEUTRAL_REACTION);
     end
     else begin
         if (local_var(1) >= 2) then begin
-            gsay_message(33, 169, 50);
+            gsay_message(NAME, 169, NEUTRAL_REACTION);
         end
         else begin
-            gsay_message(33, 170, 50);
+            gsay_message(NAME, 170, NEUTRAL_REACTION);
         end
     end
     call aradeshx();
@@ -633,27 +633,27 @@ end
 
 procedure aradesh21
 begin
-    gsay_message(33, 171, 50);
+    gsay_message(NAME, 171, NEUTRAL_REACTION);
     call aradeshx();
 end
 
 procedure aradesh22
 begin
-    gsay_message(33, 172, 50);
+    gsay_message(NAME, 172, NEUTRAL_REACTION);
     call aradeshx1();
 end
 
 procedure aradesh23
 begin
     if (local_var(1) >= 2) then begin
-        gsay_reply(33, 173);
+        Reply(173);
     end
     else begin
-        gsay_reply(33, 174);
+        Reply(174);
     end
-    giq_option(4, 33, 175, aradesh23_2, 50);
-    giq_option(5, 33, 176, aradesh25, 50);
-    giq_option(4, 33, 177, aradesh17_2, 50);
+    giq_option(4, NAME, 175, aradesh23_2, NEUTRAL_REACTION);
+    giq_option(5, NAME, 176, aradesh25, NEUTRAL_REACTION);
+    giq_option(4, NAME, 177, aradesh17_2, NEUTRAL_REACTION);
 end
 
 procedure aradesh23_2
@@ -668,26 +668,26 @@ end
 
 procedure aradesh24
 begin
-    gsay_reply(33, 178);
-    giq_option(0, 33, 179, aradesh24a, 50);
+    Reply(178);
+    giq_option(0, NAME, 179, aradesh24a, NEUTRAL_REACTION);
 end
 
 procedure aradesh24a
 begin
-    gsay_reply(33, 180);
-    giq_option(4, 33, 181, aradesh19, 50);
-    giq_option(4, 33, 182, aradesh24c, 50);
+    Reply(180);
+    giq_option(4, NAME, 181, aradesh19, NEUTRAL_REACTION);
+    giq_option(4, NAME, 182, aradesh24c, NEUTRAL_REACTION);
 end
 
 procedure aradesh24b
 begin
-    gsay_reply(33, 183);
-    giq_option(4, 33, 184, aradesh19, 50);
+    Reply(183);
+    giq_option(4, NAME, 184, aradesh19, NEUTRAL_REACTION);
 end
 
 procedure aradesh24c
 begin
-    if (global_var(26) == 2) then begin
+    if (global_var(GVAR_TANDI_STATUS) == 2) then begin
         call aradesh44();
     end
     else begin
@@ -698,42 +698,42 @@ end
 procedure aradesh25
 begin
     if (local_var(0) < 2) then begin
-        gsay_reply(33, 185);
+        Reply(185);
     end
     else begin
-        gsay_reply(33, 186);
+        Reply(186);
     end
-    giq_option(4, 33, 187, aradesh19, 50);
+    giq_option(4, NAME, 187, aradesh19, NEUTRAL_REACTION);
 end
 
 procedure aradesh26
 begin
-    gsay_reply(33, 188);
-    giq_option(7, 33, 189, aradesh27, 50);
-    giq_option(6, 33, 190, aradesh19, 50);
-    giq_option(6, 33, 191, aradesh24c, 50);
+    Reply(188);
+    giq_option(7, NAME, 189, aradesh27, NEUTRAL_REACTION);
+    giq_option(6, NAME, 190, aradesh19, NEUTRAL_REACTION);
+    giq_option(6, NAME, 191, aradesh24c, NEUTRAL_REACTION);
 end
 
 procedure aradesh27
 begin
-    gsay_reply(33, 192);
-    giq_option(8, 33, 193, aradesh25, 50);
-    giq_option(5, 33, 194, aradesh16_2, 50);
-    giq_option(5, 33, 195, aradesh24c, 50);
+    Reply(192);
+    giq_option(8, NAME, 193, aradesh25, NEUTRAL_REACTION);
+    giq_option(5, NAME, 194, aradesh16_2, NEUTRAL_REACTION);
+    giq_option(5, NAME, 195, aradesh24c, NEUTRAL_REACTION);
 end
 
 procedure aradesh28
 begin
-    gsay_reply(33, 196);
-    giq_option(4, 33, 197, aradesh28a, 50);
+    Reply(196);
+    giq_option(4, NAME, 197, aradesh28a, NEUTRAL_REACTION);
 end
 
 procedure aradesh28a
 begin
-    gsay_reply(33, 198);
-    giq_option(4, 33, 199, aradesh29, 49);
-    giq_option(4, 33, 200, aradesh28b, 51);
-    giq_option(4, 33, 201, aradesh28c, 51);
+    Reply(198);
+    giq_option(4, NAME, 199, aradesh29, GOOD_REACTION);
+    giq_option(4, NAME, 200, aradesh28b, BAD_REACTION);
+    giq_option(4, NAME, 201, aradesh28c, BAD_REACTION);
 end
 
 procedure aradesh28b
@@ -752,21 +752,21 @@ end
 
 procedure aradesh29
 begin
-    gsay_message(33, 202, 50);
-    set_global_var(43, 1);
+    gsay_message(NAME, 202, NEUTRAL_REACTION);
+    set_global_var(GVAR_RADSCORPION_SEED, 1);
     call aradeshx();
 end
 
 procedure aradesh30
 begin
-    gsay_reply(33, 203);
-    giq_option(5, 33, 204, aradesh30b, 49);
-    giq_option(4, 33, 205, aradesh19, 50);
+    Reply(203);
+    giq_option(5, NAME, 204, aradesh30b, GOOD_REACTION);
+    giq_option(4, NAME, 205, aradesh19, NEUTRAL_REACTION);
 end
 
 procedure aradesh30a
 begin
-    gsay_message(33, 206, 50);
+    gsay_message(NAME, 206, NEUTRAL_REACTION);
     call aradeshx();
 end
 
@@ -778,86 +778,86 @@ end
 
 procedure aradesh31
 begin
-    gsay_reply(33, 207);
-    giq_option(4, 33, 208, aradesh19, 50);
-    giq_option(4, 33, 209, aradesh32, 50);
+    Reply(207);
+    giq_option(4, NAME, 208, aradesh19, NEUTRAL_REACTION);
+    giq_option(4, NAME, 209, aradesh32, NEUTRAL_REACTION);
 end
 
 procedure aradesh32
 begin
-    gsay_reply(33, 210);
-    giq_option(4, 33, 211, aradesh19, 50);
-    giq_option(4, 33, 212, aradesh17_2, 50);
+    Reply(210);
+    giq_option(4, NAME, 211, aradesh19, NEUTRAL_REACTION);
+    giq_option(4, NAME, 212, aradesh17_2, NEUTRAL_REACTION);
 end
 
 procedure aradesh33
 begin
-    gsay_message(33, 213, 50);
-    set_global_var(103, 1);
+    gsay_message(NAME, 213, NEUTRAL_REACTION);
+    set_global_var(GVAR_RESCUE_TANDI, 1);
     call aradeshx();
 end
 
 procedure aradesh35
 begin
-    gsay_reply(33, 214);
-    giq_option(5, 33, 215, aradesh25, 50);
-    giq_option(4, 33, 216, aradesh16_2, 50);
-    giq_option(4, 33, 217, aradesh24c, 50);
+    Reply(214);
+    giq_option(5, NAME, 215, aradesh25, NEUTRAL_REACTION);
+    giq_option(4, NAME, 216, aradesh16_2, NEUTRAL_REACTION);
+    giq_option(4, NAME, 217, aradesh24c, NEUTRAL_REACTION);
 end
 
 procedure aradesh36
 begin
-    gsay_message(33, 218, 50);
+    gsay_message(NAME, 218, NEUTRAL_REACTION);
     call aradeshx();
 end
 
 procedure aradesh37
 begin
-    gsay_reply(33, 219);
-    if (global_var(26) == 3) then begin
-        giq_option(5, 33, 220, aradesh38, 50);
-        giq_option(4, 33, 221, aradesh37a, 50);
-        giq_option(-3, 33, 222, aradesh37a, 50);
+    Reply(219);
+    if (global_var(GVAR_TANDI_STATUS) == 3) then begin
+        giq_option(5, NAME, 220, aradesh38, NEUTRAL_REACTION);
+        giq_option(4, NAME, 221, aradesh37a, NEUTRAL_REACTION);
+        giq_option(-3, NAME, 222, aradesh37a, NEUTRAL_REACTION);
     end
     else begin
-        giq_option(4, 33, 223, aradesh38, 50);
-        giq_option(5, 33, 224, aradesh19, 50);
-        giq_option(-3, 33, 225, aradesh38, 50);
+        giq_option(4, NAME, 223, aradesh38, NEUTRAL_REACTION);
+        giq_option(5, NAME, 224, aradesh19, NEUTRAL_REACTION);
+        giq_option(-3, NAME, 225, aradesh38, NEUTRAL_REACTION);
     end
 end
 
 procedure aradesh37a
 begin
-    gsay_message(33, 226, 50);
+    gsay_message(NAME, 226, NEUTRAL_REACTION);
     call aradeshx5();
 end
 
 procedure aradesh38
 begin
-    gsay_message(33, 227, 50);
+    gsay_message(NAME, 227, NEUTRAL_REACTION);
     call aradeshx();
 end
 
 procedure aradesh39
 begin
-    set_global_var(218, 1);
-    gsay_reply(33, 228);
-    giq_option(4, 33, 229, aradesh39a, 50);
-    giq_option(-3, 33, 230, aradesh33, 50);
+    set_global_var(GVAR_TALKED_ABOUT_TANDI, 1);
+    Reply(228);
+    giq_option(4, NAME, 229, aradesh39a, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 230, aradesh33, NEUTRAL_REACTION);
 end
 
 procedure aradesh39a
 begin
-    gsay_reply(33, 231);
-    giq_option(4, 33, 232, aradesh39b, 50);
-    giq_option(4, 33, 233, aradesh41, 50);
-    giq_option(4, 33, 234, aradesh39c, 50);
+    Reply(231);
+    giq_option(4, NAME, 232, aradesh39b, NEUTRAL_REACTION);
+    giq_option(4, NAME, 233, aradesh41, NEUTRAL_REACTION);
+    giq_option(4, NAME, 234, aradesh39c, NEUTRAL_REACTION);
 end
 
 procedure aradesh39b
 begin
     call UpReact();
-    set_global_var(103, 1);
+    set_global_var(GVAR_RESCUE_TANDI, 1);
     call aradesh40();
 end
 
@@ -871,29 +871,29 @@ end
 procedure aradesh40
 begin
     variable LVar0 := 0;
-    gsay_reply(33, 235);
-    set_global_var(103, 1);
-    LVar0 := create_object_sid(7, 0, 0, -1);
+    Reply(235);
+    set_global_var(GVAR_RESCUE_TANDI, 1);
+    LVar0 := create_object(7, 0, 0);
     add_obj_to_inven(dude_obj, LVar0);
-    giq_option(4, 33, 236, aradesh40a, 50);
+    giq_option(4, NAME, 236, aradesh40a, NEUTRAL_REACTION);
 end
 
 procedure aradesh40a
 begin
-    gsay_message(33, 237, 50);
+    gsay_message(NAME, 237, NEUTRAL_REACTION);
     call aradeshx();
 end
 
 procedure aradesh41
 begin
-    gsay_reply(33, 238);
-    giq_option(4, 33, 239, aradesh40, 50);
-    giq_option(4, 33, 240, aradesh42, 50);
+    Reply(238);
+    giq_option(4, NAME, 239, aradesh40, NEUTRAL_REACTION);
+    giq_option(4, NAME, 240, aradesh42, NEUTRAL_REACTION);
 end
 
 procedure aradesh42
 begin
-    gsay_message(33, 241, 50);
+    gsay_message(NAME, 241, NEUTRAL_REACTION);
     call aradeshx3();
 end
 
@@ -902,19 +902,19 @@ begin
     variable LVar0 := 0;
     if (local_var(8) == 0) then begin
         set_local_var(8, 1);
-        LVar0 := create_object_sid(41, 0, 0, -1);
+        LVar0 := create_object(41, 0, 0);
         add_mult_objs_to_inven(dude_obj, LVar0, 500);
     end
     call TopReact();
-    set_global_var(103, 2);
-    gsay_reply(33, 242);
-    giq_option(4, 33, 243, aradesh19, 50);
-    giq_option(4, 33, 244, aradesh44, 50);
+    set_global_var(GVAR_RESCUE_TANDI, 2);
+    Reply(242);
+    giq_option(4, NAME, 243, aradesh19, NEUTRAL_REACTION);
+    giq_option(4, NAME, 244, aradesh44, NEUTRAL_REACTION);
 end
 
 procedure aradesh44
 begin
-    gsay_message(33, 245, 50);
+    gsay_message(NAME, 245, NEUTRAL_REACTION);
     call aradeshx();
 end
 
@@ -922,7 +922,7 @@ procedure aradesh45
 begin
     set_local_var(9, local_var(9) + 1);
     call BigDownReact();
-    float_msg(self_obj, message_str(33, 246), 7);
+    float_msg(self_obj, mstr(246), FLOAT_MSG_LIGHT_RED);
     call aradeshx();
 end
 
@@ -930,7 +930,7 @@ procedure aradesh46
 begin
     set_local_var(9, local_var(9) + 1);
     call BigDownReact();
-    float_msg(self_obj, message_str(33, 247), 7);
+    float_msg(self_obj, mstr(247), FLOAT_MSG_LIGHT_RED);
     call aradeshx();
 end
 
@@ -938,7 +938,7 @@ procedure aradesh47
 begin
     set_local_var(9, local_var(9) + 1);
     call DownReact();
-    gsay_message(33, 248, 50);
+    gsay_message(NAME, 248, NEUTRAL_REACTION);
     call aradeshx();
 end
 
@@ -946,7 +946,7 @@ procedure aradesh48
 begin
     set_local_var(9, local_var(9) + 1);
     call DownReact();
-    gsay_message(33, 249, 50);
+    gsay_message(NAME, 249, NEUTRAL_REACTION);
 end
 
 procedure aradeshx
@@ -978,31 +978,31 @@ begin
         set_local_var(0, 50);
         set_local_var(1, 2);
         set_local_var(2, 1);
-        set_local_var(0, local_var(0) + (5 * get_critter_stat(dude_obj, 3)) - 25);
-        set_local_var(0, local_var(0) + (10 * has_trait(0, dude_obj, 10)));
-        if (has_trait(0, dude_obj, 39)) then begin
-            if (global_var(155) > 0) then begin
-                set_local_var(0, local_var(0) + global_var(155));
+        set_local_var(0, local_var(0) + (5 * dude_charisma) - 25);
+        set_local_var(0, local_var(0) + (10 * has_trait(TRAIT_PERK, dude_obj, PERK_presence)));
+        if (has_trait(TRAIT_PERK, dude_obj, PERK_cult_of_personality)) then begin
+            if (global_var(GVAR_PLAYER_REPUATION) > 0) then begin
+                set_local_var(0, local_var(0) + global_var(GVAR_PLAYER_REPUATION));
             end
             else begin
-                set_local_var(0, local_var(0) - global_var(155));
+                set_local_var(0, local_var(0) - global_var(GVAR_PLAYER_REPUATION));
             end
         end
         else begin
             if (local_var(3) == 1) then begin
-                set_local_var(0, local_var(0) - global_var(155));
+                set_local_var(0, local_var(0) - global_var(GVAR_PLAYER_REPUATION));
             end
             else begin
-                set_local_var(0, local_var(0) + global_var(155));
+                set_local_var(0, local_var(0) + global_var(GVAR_PLAYER_REPUATION));
             end
         end
-        if (global_var(158) > 2) then begin
+        if (global_var(GVAR_CHILDKILLER_REPUATION) > 2) then begin
             set_local_var(0, local_var(0) - 30);
         end
-        if (((global_var(160) + global_var(159)) >= 25) and ((global_var(160) > (3 * global_var(159))) or (global_var(157) == 1))) then begin
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(160) > (3 * global_var(GVAR_GOOD_MONSTER))) or (global_var(GVAR_CHAMPION_REPUTATION) == 1))) then begin
             set_local_var(0, local_var(0) + 20);
         end
-        if (((global_var(160) + global_var(159)) >= 25) and ((global_var(159) > (2 * global_var(160))) or (global_var(156) == 1))) then begin
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
             set_local_var(0, local_var(0) - 20);
         end
         call ReactToLevel();
@@ -1095,136 +1095,135 @@ end
 
 procedure Goodbyes
 begin
-    exit_line := message_str(634, random(100, 105));
+    exit_line := message_str(SCRIPT_MODREACT, random(100, 105));
 end
 
 procedure aradesh49
 begin
-    gsay_message(33, 250, 50);
+    gsay_message(NAME, 250, NEUTRAL_REACTION);
 end
 
 procedure aradesh49a
 begin
-    gsay_message(33, 251, 50);
+    gsay_message(NAME, 251, NEUTRAL_REACTION);
 end
 
 procedure aradesh50
 begin
-    gsay_message(33, 252, 50);
+    gsay_message(NAME, 252, NEUTRAL_REACTION);
 end
 
 procedure aradesh51
 begin
-    gsay_message(33, 253, 50);
+    gsay_message(NAME, 253, NEUTRAL_REACTION);
 end
 
 procedure aradesh52
 begin
-    gsay_message(33, 254, 50);
+    gsay_message(NAME, 254, NEUTRAL_REACTION);
 end
 
 procedure aradesh53
 begin
-    gsay_message(33, 255, 50);
+    gsay_message(NAME, 255, NEUTRAL_REACTION);
 end
 
 procedure aradesh54
 begin
-    gsay_message(33, 256, 50);
+    gsay_message(NAME, 256, NEUTRAL_REACTION);
 end
 
 procedure aradesh55
 begin
-    gsay_message(33, 257, 50);
+    gsay_message(NAME, 257, NEUTRAL_REACTION);
 end
 
 procedure aradesh56
 begin
-    gsay_message(33, 258, 50);
+    gsay_message(NAME, 258, NEUTRAL_REACTION);
 end
 
 procedure aradesh57
 begin
-    gsay_message(33, 259, 50);
+    gsay_message(NAME, 259, NEUTRAL_REACTION);
 end
 
 procedure aradesh58
 begin
-    gsay_message(33, 260, 50);
+    gsay_message(NAME, 260, NEUTRAL_REACTION);
 end
 
 procedure aradesh59
 begin
-    gsay_message(33, 261, 50);
+    gsay_message(NAME, 261, NEUTRAL_REACTION);
 end
 
 procedure aradesh60
 begin
-    gsay_message(33, 262, 50);
+    gsay_message(NAME, 262, NEUTRAL_REACTION);
 end
 
 procedure aradesh61
 begin
-    gsay_message(33, 263, 50);
+    gsay_message(NAME, 263, NEUTRAL_REACTION);
 end
 
 procedure aradesh62
 begin
-    gsay_message(33, 264, 50);
+    gsay_message(NAME, 264, NEUTRAL_REACTION);
 end
 
 procedure aradesh63
 begin
-    gsay_message(33, 265, 50);
+    gsay_message(NAME, 265, NEUTRAL_REACTION);
 end
 
 procedure aradesh64
 begin
-    gsay_message(33, 266, 50);
+    gsay_message(NAME, 266, NEUTRAL_REACTION);
 end
 
 procedure aradesh65
 begin
-    gsay_message(33, 267, 50);
+    gsay_message(NAME, 267, NEUTRAL_REACTION);
 end
 
 procedure aradesh66
 begin
-    gsay_message(33, 268, 50);
+    gsay_message(NAME, 268, NEUTRAL_REACTION);
 end
 
 procedure aradesh67
 begin
-    gsay_message(33, 269, 50);
+    gsay_message(NAME, 269, NEUTRAL_REACTION);
 end
 
 procedure aradesh68
 begin
-    gsay_message(33, 270, 50);
+    gsay_message(NAME, 270, NEUTRAL_REACTION);
 end
 
 procedure aradesh69
 begin
-    gsay_message(33, 271, 50);
+    gsay_message(NAME, 271, NEUTRAL_REACTION);
 end
 
 procedure aradesh70
 begin
-    gsay_message(33, 272, 50);
+    gsay_message(NAME, 272, NEUTRAL_REACTION);
 end
 
 procedure aradesh71
 begin
-    gsay_message(33, 273, 50);
+    gsay_message(NAME, 273, NEUTRAL_REACTION);
 end
 
 procedure aradesh72
 begin
-    gsay_message(33, 274, 50);
+    gsay_message(NAME, 274, NEUTRAL_REACTION);
 end
 
 procedure aradesh73
 begin
-    gsay_message(33, 275, 50);
+    gsay_message(NAME, 275, NEUTRAL_REACTION);
 end
-
