diff --git a/Games/Fallout1.Steam/Original/BARFLY.ssl b/Games/Fallout1.Steam/ReDefine/BARFLY.ssl
index 4ae7ce8..24e9d59 100644
--- a/Games/Fallout1.Steam/Original/BARFLY.ssl
+++ b/Games/Fallout1.Steam/ReDefine/BARFLY.ssl
@@ -17,19 +17,19 @@ variable hostile;
 
 procedure start
 begin
-    if (script_action == 12) then begin
+    if (script_action == critter_proc) then begin
         call critter_p_proc();
     end
     else begin
-        if (script_action == 15) then begin
+        if (script_action == map_enter_proc) then begin
             call map_enter_p_proc();
         end
         else begin
-            if (script_action == 4) then begin
+            if (script_action == pickup_proc) then begin
                 call pickup_p_proc();
             end
             else begin
-                if (script_action == 11) then begin
+                if (script_action == talk_proc) then begin
                     call talk_p_proc();
                 end
             end
@@ -41,7 +41,7 @@ procedure critter_p_proc
 begin
     if (hostile) then begin
         hostile := 0;
-        attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+        attack(dude_obj);
     end
     call sleeping();
     if ((game_time_hour > (wake_time + 20)) or (game_time_hour < sleep_time)) then begin
@@ -49,16 +49,16 @@ begin
             animate_move_obj_to_tile(self_obj, home_tile, 0);
         end
     end
-    if (global_var(247) == 1) then begin
+    if (global_var(GVAR_ENEMY_JUNKTOWN) == 1) then begin
         if (obj_can_see_obj(self_obj, dude_obj)) then begin
-            attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+            attack(dude_obj);
         end
     end
 end
 
 procedure map_enter_p_proc
 begin
-    if (global_var(15) == 1) then begin
+    if (global_var(GVAR_JUNKTOWN_INVADED) == 1) then begin
         destroy_object(self_obj);
     end
     night_person := 1;
@@ -66,8 +66,8 @@ begin
     sleep_time := random(310, 340);
     sleep_tile := 7000;
     home_tile := 20675;
-    critter_add_trait(self_obj, 1, 6, 26);
-    critter_add_trait(self_obj, 1, 5, 5);
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 26);
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_LOSER);
     if (map_var(2) == 1) then begin
         destroy_object(self_obj);
     end
@@ -80,32 +80,32 @@ end
 
 procedure talk_p_proc
 begin
-    float_msg(self_obj, message_str(473, random(100, 106)), 0);
+    float_msg(self_obj, mstr(random(100, 106)), FLOAT_MSG_NORMAL);
 end
 
 procedure damage_p_proc
 begin
     if (source_obj == dude_obj) then begin
-        set_global_var(247, 1);
+        set_global_var(GVAR_ENEMY_JUNKTOWN, 1);
     end
 end
 
 procedure destroy_p_proc
 begin
     if (source_obj == dude_obj) then begin
-        set_global_var(247, 1);
+        set_global_var(GVAR_ENEMY_JUNKTOWN, 1);
         if (source_obj == dude_obj) then begin
-            if (((global_var(160) + global_var(159)) >= 25) and ((global_var(159) > (2 * global_var(160))) or (global_var(156) == 1))) then begin
-                set_global_var(156, 1);
-                set_global_var(157, 0);
+            if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
+                set_global_var(GVAR_BERSERKER_REPUTATION, 1);
+                set_global_var(GVAR_CHAMPION_REPUTATION, 0);
             end
-            if (((global_var(160) + global_var(159)) >= 25) and ((global_var(160) > (3 * global_var(159))) or (global_var(157) == 1))) then begin
-                set_global_var(157, 1);
-                set_global_var(156, 0);
+            if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(160) > (3 * global_var(GVAR_GOOD_MONSTER))) or (global_var(GVAR_CHAMPION_REPUTATION) == 1))) then begin
+                set_global_var(GVAR_CHAMPION_REPUTATION, 1);
+                set_global_var(GVAR_BERSERKER_REPUTATION, 0);
             end
-            set_global_var(159, global_var(159) + 1);
-            if ((global_var(159) % 2) == 0) then begin
-                set_global_var(155, global_var(155) - 1);
+            set_global_var(GVAR_GOOD_MONSTER, global_var(GVAR_GOOD_MONSTER) + 1);
+            if ((global_var(GVAR_GOOD_MONSTER) % 2) == 0) then begin
+                set_global_var(GVAR_PLAYER_REPUATION, global_var(GVAR_PLAYER_REPUATION) - 1);
             end
         end
     end
@@ -152,4 +152,3 @@ begin
         end
     end
 end
-
