diff --git a/Games/Fallout1.Steam/Original/BARSTOW.ssl b/Games/Fallout1.Steam/ReDefine/BARSTOW.ssl
index 38ee565..6f00d55 100644
--- a/Games/Fallout1.Steam/Original/BARSTOW.ssl
+++ b/Games/Fallout1.Steam/ReDefine/BARSTOW.ssl
@@ -13,22 +13,22 @@ variable HEREBEFORE;
 procedure start
 begin
     variable LVar0 := 0;
-    if (script_action == 11) then begin
+    if (script_action == talk_proc) then begin
         call do_dialogue();
         detach;
     end
     else begin
-        if (script_action == 21) then begin
+        if (script_action == look_at_proc) then begin
             script_overrides;
             display_msg("You see Zack Barstow, an Initiate of the Childred of the Cathedral.");
             detach;
         end
         else begin
-            if (script_action == 12) then begin
+            if (script_action == critter_proc) then begin
                 detach;
             end
             else begin
-                if (script_action == 13) then begin
+                if (script_action == combat_proc) then begin
                     detach;
                 end
             end
@@ -38,7 +38,7 @@ end
 
 procedure do_dialogue
 begin
-    start_gdialog(-1, self_obj, 4, -1, -1);
+    start_dialogue(self_obj, 4);
     sayStart;
     if (HEREBEFORE) then begin
         call zack05();
@@ -54,19 +54,19 @@ end
 procedure zack00
 begin
     sayReply(0, "Welcome to the Children of the Cathedral. How might I help you today?");
-    if (-get_critter_stat(dude_obj, 4) >= -5) then begin
+    if (-dude_iq >= -5) then begin
         sayOption("Urrrooo?", zack04);
     end
-    if (get_critter_stat(dude_obj, 4) >= 14) then begin
+    if (dude_iq >= 14) then begin
         sayOption("What is the meaning of the universe?", zack04);
     end
-    if (get_critter_stat(dude_obj, 4) >= 6) then begin
+    if (dude_iq >= 6) then begin
         sayOption("I'm hurt pretty bad. Think you can help me?", zack03);
     end
-    if (get_critter_stat(dude_obj, 4) >= 10) then begin
+    if (dude_iq >= 10) then begin
         sayOption("How do you fix the water pump?", zack02);
     end
-    if (get_critter_stat(dude_obj, 4) >= 6) then begin
+    if (dude_iq >= 6) then begin
         sayOption("Can you tell me something about this place?", zack01);
     end
 end
@@ -95,4 +95,3 @@ procedure zack05
 begin
     sayMessage(0, "Please consult with Laura. I have told you all I know.");
 end
-
