diff --git a/Games/Fallout1.Steam/Original/BARTER.ssl b/Games/Fallout1.Steam/ReDefine/BARTER.ssl
index 2565713..d92b10e 100644
--- a/Games/Fallout1.Steam/Original/BARTER.ssl
+++ b/Games/Fallout1.Steam/ReDefine/BARTER.ssl
@@ -22,22 +22,22 @@ procedure elder07;
 
 procedure start
 begin
-    if (script_action == 12) then begin
+    if (script_action == critter_proc) then begin
         if (hostile) then begin
             hostile := 0;
-            attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+            attack(dude_obj);
         end
     end
     else begin
-        if (script_action == 4) then begin
+        if (script_action == pickup_proc) then begin
             hostile := 1;
         end
         else begin
-            if (script_action == 11) then begin
+            if (script_action == talk_proc) then begin
                 call do_dialogue();
             end
             else begin
-                if (script_action == 3) then begin
+                if (script_action == description_proc) then begin
                     display_msg(message_str(0, 100));
                     script_overrides;
                 end
@@ -48,7 +48,7 @@ end
 
 procedure do_dialogue
 begin
-    start_gdialog(-1, self_obj, 4, -1, -1);
+    start_dialogue(self_obj, 4);
     sayStart;
     sayReply(0, message_str(0, 101));
     sayOption(message_str(0, 102), do_nothing);
@@ -109,4 +109,3 @@ procedure elder07
 begin
     sayMessage(0, message_str(0, 110));
 end
-
