diff --git a/Games/Fallout1.Steam/Original/BGUARD.ssl b/Games/Fallout1.Steam/ReDefine/BGUARD.ssl
index 9c8a5ea..9053e17 100644
--- a/Games/Fallout1.Steam/Original/BGUARD.ssl
+++ b/Games/Fallout1.Steam/ReDefine/BGUARD.ssl
@@ -7,24 +7,24 @@ variable hostile;
 
 procedure start
 begin
-    if (script_action == 11) then begin
+    if (script_action == talk_proc) then begin
         call do_dialogue();
     end
     else begin
-        if (script_action == 12) then begin
+        if (script_action == critter_proc) then begin
             if (hostile) then begin
                 hostile := 0;
-                attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+                attack(dude_obj);
             end
         end
         else begin
-            if (script_action == 4) then begin
+            if (script_action == pickup_proc) then begin
                 hostile := 1;
             end
             else begin
-                if ((script_action == 21) or (script_action == 3)) then begin
+                if ((script_action == look_at_proc) or (script_action == description_proc)) then begin
                     script_overrides;
-                    display_msg(message_str(4, 100));
+                    display_mstr(100);
                 end
             end
         end
@@ -36,26 +36,25 @@ begin
     variable LVar0 := 0;
     rndx := random(0, 6);
     if (rndx == 0) then begin
-        LVar0 := message_str(4, 101);
+        LVar0 := mstr(101);
     end
     if (rndx == 1) then begin
-        LVar0 := message_str(4, 102);
+        LVar0 := mstr(102);
     end
     if (rndx == 2) then begin
-        LVar0 := message_str(4, 103);
+        LVar0 := mstr(103);
     end
     if (rndx == 3) then begin
-        LVar0 := message_str(4, 104);
+        LVar0 := mstr(104);
     end
     if (rndx == 4) then begin
-        LVar0 := message_str(4, 105);
+        LVar0 := mstr(105);
     end
     if (rndx == 5) then begin
-        LVar0 := message_str(4, 106);
+        LVar0 := mstr(106);
     end
     if (rndx == 6) then begin
-        LVar0 := message_str(4, 107);
+        LVar0 := mstr(107);
     end
-    float_msg(self_obj, LVar0, 0);
+    float_msg(self_obj, LVar0, FLOAT_MSG_NORMAL);
 end
-
