diff --git a/Games/Fallout1.Steam/Original/BHDOOR.ssl b/Games/Fallout1.Steam/ReDefine/BHDOOR.ssl
index ff692f8..9339be1 100644
--- a/Games/Fallout1.Steam/Original/BHDOOR.ssl
+++ b/Games/Fallout1.Steam/ReDefine/BHDOOR.ssl
@@ -13,31 +13,31 @@ import variable Cabbot_Ptr;
 
 procedure start
 begin
-    if (script_action == 15) then begin
+    if (script_action == map_enter_proc) then begin
         call map_enter_p_proc();
     end
     else begin
-        if (script_action == 23) then begin
+        if (script_action == map_update_proc) then begin
             call map_update_p_proc();
         end
         else begin
-            if (script_action == 21) then begin
+            if (script_action == look_at_proc) then begin
                 call look_at_p_proc();
             end
             else begin
-                if (script_action == 6) then begin
+                if (script_action == use_proc) then begin
                     call use_p_proc();
                 end
                 else begin
-                    if (script_action == 7) then begin
+                    if (script_action == use_obj_on_proc) then begin
                         call use_obj_on_p_proc();
                     end
                     else begin
-                        if (script_action == 8) then begin
+                        if (script_action == use_skill_on_proc) then begin
                             call use_skill_on_p_proc();
                         end
                         else begin
-                            if (script_action == 22) then begin
+                            if (script_action == timed_event_proc) then begin
                                 call timed_event_p_proc();
                             end
                         end
@@ -63,21 +63,21 @@ end
 
 procedure look_at_p_proc
 begin
-    if ((global_var(108) == 2) and (local_var(0) == 0)) then begin
-        display_msg(message_str(526, 102));
+    if ((global_var(GVAR_BECOME_AN_INITIATE) == 2) and (local_var(0) == 0)) then begin
+        display_msg(message_str(SCRIPT_MBOUT2IN, 102));
     end
     else begin
-        display_msg(message_str(526, 100));
+        display_msg(message_str(SCRIPT_MBOUT2IN, 100));
     end
 end
 
 procedure use_p_proc
 begin
-    if ((global_var(108) != 2) and (local_var(0) == 0)) then begin
+    if ((global_var(GVAR_BECOME_AN_INITIATE) != 2) and (local_var(0) == 0)) then begin
         if (source_obj != Cabbot_Ptr) then begin
             script_overrides;
             set_map_var(0, map_var(0) + 1);
-            display_msg(message_str(526, 104));
+            display_msg(message_str(SCRIPT_MBOUT2IN, 104));
         end
     end
 end
@@ -88,18 +88,18 @@ begin
     variable LVar1 := 0;
     LVar0 := obj_being_used_with;
     LVar1 := roll_vs_skill(dude_obj, 9, -50);
-    if (global_var(108) != 2) then begin
+    if (global_var(GVAR_BECOME_AN_INITIATE) != 2) then begin
         set_map_var(0, map_var(0) + 1);
     end
     if (obj_pid(LVar0) == 77) then begin
         script_overrides;
         if (is_success(LVar1)) then begin
-            if (global_var(108) != 2) then begin
+            if (global_var(GVAR_BECOME_AN_INITIATE) != 2) then begin
                 set_map_var(0, map_var(0) + 2);
             end
             set_local_var(0, 1);
-            display_msg(message_str(63, 100));
-            display_msg(message_str(766, 103) + "75" + message_str(766, 104));
+            display_msg(message_str(SCRIPT_DOOR, 100));
+            display_msg(message_str(SCRIPT_GENCHAT, 103) + "75" + message_str(SCRIPT_GENCHAT, 104));
             give_exp_points(75);
         end
         else begin
@@ -107,10 +107,10 @@ begin
                 rm_obj_from_inven(dude_obj, LVar0);
                 destroy_object(LVar0);
                 jam_lock(self_obj);
-                display_msg(message_str(63, 101));
+                display_msg(message_str(SCRIPT_DOOR, 101));
             end
             else begin
-                display_msg(message_str(63, 103));
+                display_msg(message_str(SCRIPT_DOOR, 103));
             end
         end
     end
@@ -119,7 +119,7 @@ end
 procedure use_skill_on_p_proc
 begin
     variable LVar0 := 0;
-    if (global_var(108) != 2) then begin
+    if (global_var(GVAR_BECOME_AN_INITIATE) != 2) then begin
         set_map_var(0, map_var(0) + 1);
     end
     if (local_var(0) == 0) then begin
@@ -128,27 +128,27 @@ begin
             LVar0 := roll_vs_skill(dude_obj, 9, -70);
             if (is_success(LVar0)) then begin
                 set_local_var(0, 1);
-                if (global_var(108) != 2) then begin
+                if (global_var(GVAR_BECOME_AN_INITIATE) != 2) then begin
                     set_map_var(0, map_var(0) + 2);
-                    set_global_var(250, 1);
+                    set_global_var(GVAR_ENEMY_BROTHERHOOD, 1);
                 end
-                display_msg(message_str(63, 100));
+                display_msg(message_str(SCRIPT_DOOR, 100));
                 give_exp_points(75);
-                display_msg(message_str(766, 103) + "75" + message_str(766, 104));
+                display_msg(message_str(SCRIPT_GENCHAT, 103) + "75" + message_str(SCRIPT_GENCHAT, 104));
             end
             else begin
                 if (is_critical(LVar0)) then begin
                     jam_lock(self_obj);
-                    display_msg(message_str(63, 110));
+                    display_msg(message_str(SCRIPT_DOOR, 110));
                 end
                 else begin
-                    display_msg(message_str(63, 103));
+                    display_msg(message_str(SCRIPT_DOOR, 103));
                 end
             end
         end
     end
     else begin
-        display_msg(message_str(63, 109));
+        display_msg(message_str(SCRIPT_DOOR, 109));
     end
 end
 
@@ -157,4 +157,3 @@ begin
     set_local_var(0, 0);
     obj_close(self_obj);
 end
-
