diff --git a/Games/Fallout1.Steam/Original/BITSBOB.ssl b/Games/Fallout1.Steam/ReDefine/BITSBOB.ssl
index 7614e48..6398b54 100644
--- a/Games/Fallout1.Steam/Original/BITSBOB.ssl
+++ b/Games/Fallout1.Steam/ReDefine/BITSBOB.ssl
@@ -95,26 +95,26 @@ procedure start
 begin
     if (Only_Once) then begin
         Only_Once := 0;
-        critter_add_trait(self_obj, 1, 6, 19);
-        critter_add_trait(self_obj, 1, 5, 50);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 19);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_MERCHANT);
     end
-    if (script_action == 21) then begin
+    if (script_action == look_at_proc) then begin
         call look_at_p_proc();
     end
     else begin
-        if (script_action == 4) then begin
+        if (script_action == pickup_proc) then begin
             call pickup_p_proc();
         end
         else begin
-            if (script_action == 11) then begin
+            if (script_action == talk_proc) then begin
                 call talk_p_proc();
             end
             else begin
-                if (script_action == 12) then begin
+                if (script_action == critter_proc) then begin
                     call critter_p_proc();
                 end
                 else begin
-                    if (script_action == 18) then begin
+                    if (script_action == destroy_proc) then begin
                         call destroy_p_proc();
                     end
                 end
@@ -132,7 +132,7 @@ procedure critter_p_proc
 begin
     if (hostile) then begin
         hostile := 0;
-        attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+        attack(dude_obj);
     end
 end
 
@@ -197,17 +197,17 @@ end
 procedure destroy_p_proc
 begin
     if (source_obj == dude_obj) then begin
-        if (((global_var(160) + global_var(159)) >= 25) and ((global_var(159) > (2 * global_var(160))) or (global_var(156) == 1))) then begin
-            set_global_var(156, 1);
-            set_global_var(157, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
+            set_global_var(GVAR_BERSERKER_REPUTATION, 1);
+            set_global_var(GVAR_CHAMPION_REPUTATION, 0);
         end
-        if (((global_var(160) + global_var(159)) >= 25) and ((global_var(160) > (3 * global_var(159))) or (global_var(157) == 1))) then begin
-            set_global_var(157, 1);
-            set_global_var(156, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(160) > (3 * global_var(GVAR_GOOD_MONSTER))) or (global_var(GVAR_CHAMPION_REPUTATION) == 1))) then begin
+            set_global_var(GVAR_CHAMPION_REPUTATION, 1);
+            set_global_var(GVAR_BERSERKER_REPUTATION, 0);
         end
-        set_global_var(159, global_var(159) + 1);
-        if ((global_var(159) % 2) == 0) then begin
-            set_global_var(155, global_var(155) - 1);
+        set_global_var(GVAR_GOOD_MONSTER, global_var(GVAR_GOOD_MONSTER) + 1);
+        if ((global_var(GVAR_GOOD_MONSTER) % 2) == 0) then begin
+            set_global_var(GVAR_PLAYER_REPUATION, global_var(GVAR_PLAYER_REPUATION) - 1);
         end
     end
 end
@@ -217,57 +217,57 @@ begin
     variable LVar0 := 0;
     LVar0 := obj_pid(source_obj);
     if (party_member_obj(LVar0) != 0) then begin
-        set_global_var(248, 1);
+        set_global_var(GVAR_ENEMY_HUB, 1);
     end
 end
 
 procedure look_at_p_proc
 begin
     script_overrides;
-    display_msg(message_str(891, 100));
+    display_mstr(100);
 end
 
 procedure BobStandardQuestions
 begin
-    giq_option(4, 891, 231, Bob11, 50);
-    if (global_var(101) != 2) then begin
-        giq_option(4, 891, 232, Bob18, 50);
+    giq_option(4, NAME, 231, Bob11, NEUTRAL_REACTION);
+    if (global_var(GVAR_FIND_WATER_CHIP) != 2) then begin
+        giq_option(4, NAME, 232, Bob18, NEUTRAL_REACTION);
     end
-    if (global_var(219) == 1) then begin
-        if (global_var(203) != 0) then begin
-            giq_option(4, 891, 233, Bob23, 50);
+    if (global_var(GVAR_DECKER_KNOWN) == 1) then begin
+        if (global_var(GVAR_DECKER_STATUS) != 0) then begin
+            giq_option(4, NAME, 233, Bob23, NEUTRAL_REACTION);
         end
         else begin
-            giq_option(4, 891, 233, Bob19, 50);
+            giq_option(4, NAME, 233, Bob19, NEUTRAL_REACTION);
         end
     end
-    if (global_var(305) == 1) then begin
-        giq_option(4, 891, 234, Bob20, 50);
+    if (global_var(GVAR_DESTROY_MASTER_1) == 1) then begin
+        giq_option(4, NAME, 234, Bob20, NEUTRAL_REACTION);
     end
-    if (global_var(106) == 1) then begin
-        giq_option(4, 891, 235, Bob21, 50);
+    if (global_var(GVAR_MISSING_CARAVAN) == 1) then begin
+        giq_option(4, NAME, 235, Bob21, NEUTRAL_REACTION);
     end
-    giq_option(4, 891, 237, Bob25, 50);
-    giq_option(4, 891, 240, BobEnd, 50);
-    giq_option(-3, 891, 238, Bob11, 50);
-    giq_option(-3, 891, 241, BobEnd, 50);
+    giq_option(4, NAME, 237, Bob25, NEUTRAL_REACTION);
+    giq_option(4, NAME, 240, BobEnd, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 238, Bob11, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 241, BobEnd, NEUTRAL_REACTION);
 end
 
 procedure BobBlackmailQuestions
 begin
-    giq_option(4, 891, 102, BBQa, 50);
-    giq_option(4, 891, 103, Bob06, 51);
-    giq_option(4, 891, 104, Bob07, 50);
-    if (global_var(219) == 1) then begin
-        if (global_var(203) != 0) then begin
-            giq_option(4, 891, 233, Bob23, 50);
+    giq_option(4, NAME, 102, BBQa, NEUTRAL_REACTION);
+    giq_option(4, NAME, 103, Bob06, BAD_REACTION);
+    giq_option(4, NAME, 104, Bob07, NEUTRAL_REACTION);
+    if (global_var(GVAR_DECKER_KNOWN) == 1) then begin
+        if (global_var(GVAR_DECKER_STATUS) != 0) then begin
+            giq_option(4, NAME, 233, Bob23, NEUTRAL_REACTION);
         end
         else begin
-            giq_option(4, 891, 233, Bob08, 50);
+            giq_option(4, NAME, 233, Bob08, NEUTRAL_REACTION);
         end
     end
-    giq_option(4, 891, 106, Bob10, 50);
-    giq_option(4, 891, 240, BobEnd, 50);
+    giq_option(4, NAME, 106, Bob10, NEUTRAL_REACTION);
+    giq_option(4, NAME, 240, BobEnd, NEUTRAL_REACTION);
 end
 
 procedure BBQa
@@ -285,19 +285,19 @@ begin
     variable LVar0 := 0;
     LVar0 := item_caps_adjust(dude_obj, local_var(7));
     set_local_var(6, (game_time / (10 * 60 * 60 * 24)) + 5);
-    gsay_reply(891, 101);
+    Reply(101);
     call BobBlackmailQuestions();
 end
 
 procedure Bob01
 begin
-    gsay_reply(891, 108);
-    giq_option(6, 891, 109, Bob01a, 50);
-    giq_option(4, 891, 110, Bob01a, 50);
-    giq_option(4, 891, 111, Bob14, 50);
-    giq_option(4, 891, 112, Bob15, 50);
-    giq_option(4, 891, 113, Bob16, 50);
-    giq_option(4, 891, 114, BobCombat, 50);
+    Reply(108);
+    giq_option(6, NAME, 109, Bob01a, NEUTRAL_REACTION);
+    giq_option(4, NAME, 110, Bob01a, NEUTRAL_REACTION);
+    giq_option(4, NAME, 111, Bob14, NEUTRAL_REACTION);
+    giq_option(4, NAME, 112, Bob15, NEUTRAL_REACTION);
+    giq_option(4, NAME, 113, Bob16, NEUTRAL_REACTION);
+    giq_option(4, NAME, 114, BobCombat, NEUTRAL_REACTION);
 end
 
 procedure Bob01a
@@ -317,54 +317,54 @@ end
 
 procedure Bob02
 begin
-    gsay_reply(891, 115);
+    Reply(115);
     call BobBlackmailQuestions();
 end
 
 procedure Bob03
 begin
-    float_msg(self_obj, message_str(891, 116), 2);
+    float_msg(self_obj, mstr(116), FLOAT_MSG_RED);
 end
 
 procedure Bob04
 begin
-    gsay_reply(891, 117);
-    giq_option(4, 891, 236, Bob22, 50);
+    Reply(117);
+    giq_option(4, NAME, 236, Bob22, NEUTRAL_REACTION);
     call BobStandardQuestions();
 end
 
 procedure Bob05
 begin
-    gsay_reply(891, 118);
+    Reply(118);
     call BobStandardQuestions();
 end
 
 procedure Bob06
 begin
-    gsay_message(891, 119, 51);
+    gsay_message(NAME, 119, BAD_REACTION);
     call BobCombat();
 end
 
 procedure Bob07
 begin
     set_local_var(5, 3);
-    gsay_message(891, 120, 50);
+    gsay_message(NAME, 120, NEUTRAL_REACTION);
 end
 
 procedure Bob08
 begin
-    gsay_reply(891, 121);
-    giq_option(4, 891, 106, Bob10, 50);
-    giq_option(4, 891, 107, Bob11, 50);
+    Reply(121);
+    giq_option(4, NAME, 106, Bob10, NEUTRAL_REACTION);
+    giq_option(4, NAME, 107, Bob11, NEUTRAL_REACTION);
 end
 
 procedure Bob09
 begin
-    gsay_reply(891, 122);
-    giq_option(4, 891, message_str(891, 123) + (local_var(7) + 25) + message_str(891, 124), Bob09a, 50);
-    giq_option(4, 891, message_str(891, 123) + (local_var(7) + 50) + message_str(891, 124), Bob09b, 50);
-    giq_option(4, 891, message_str(891, 123) + (local_var(7) + 75) + message_str(891, 124), Bob09c, 50);
-    giq_option(4, 891, message_str(891, 123) + (local_var(7) + 100) + message_str(891, 124), Bob09d, 50);
+    Reply(122);
+    giq_option(4, NAME, mstr(123) + (local_var(7) + 25) + mstr(124), Bob09a, NEUTRAL_REACTION);
+    giq_option(4, NAME, mstr(123) + (local_var(7) + 50) + mstr(124), Bob09b, NEUTRAL_REACTION);
+    giq_option(4, NAME, mstr(123) + (local_var(7) + 75) + mstr(124), Bob09c, NEUTRAL_REACTION);
+    giq_option(4, NAME, mstr(123) + (local_var(7) + 100) + mstr(124), Bob09d, NEUTRAL_REACTION);
 end
 
 procedure Bob09a
@@ -413,18 +413,18 @@ end
 
 procedure Bob10
 begin
-    gsay_message(891, 125, 50);
+    gsay_message(NAME, 125, NEUTRAL_REACTION);
 end
 
 procedure Bob11
 begin
-    gsay_reply(891, 142);
-    giq_option(4, 891, 129, Bob27, 50);
-    giq_option(4, 891, 131, Bob28, 50);
-    giq_option(4, 891, 130, Bob30, 50);
-    giq_option(-3, 891, 203, Bob27, 50);
-    giq_option(-3, 891, 204, Bob28, 50);
-    giq_option(-3, 891, 133, Bob30, 50);
+    Reply(142);
+    giq_option(4, NAME, 129, Bob27, NEUTRAL_REACTION);
+    giq_option(4, NAME, 131, Bob28, NEUTRAL_REACTION);
+    giq_option(4, NAME, 130, Bob30, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 203, Bob27, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 204, Bob28, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 133, Bob30, NEUTRAL_REACTION);
 end
 
 procedure Bob12
@@ -432,55 +432,55 @@ begin
     variable LVar0 := 0;
     LVar0 := item_caps_adjust(dude_obj, local_var(7));
     set_local_var(6, (game_time / (10 * 60 * 60 * 24)) + 5);
-    gsay_message(891, 134, 51);
+    gsay_message(NAME, 134, BAD_REACTION);
 end
 
 procedure Bob13
 begin
     set_local_var(5, 2);
-    gsay_message(891, 135, 51);
+    gsay_message(NAME, 135, BAD_REACTION);
 end
 
 procedure Bob14
 begin
     set_local_var(5, 2);
-    gsay_message(891, 136, 51);
+    gsay_message(NAME, 136, BAD_REACTION);
 end
 
 procedure Bob15
 begin
     set_local_var(5, 2);
-    gsay_message(891, 137, 51);
+    gsay_message(NAME, 137, BAD_REACTION);
 end
 
 procedure Bob16
 begin
     set_local_var(5, 2);
-    gsay_message(891, 138, 51);
+    gsay_message(NAME, 138, BAD_REACTION);
 end
 
 procedure Bob18
 begin
-    gsay_reply(891, 148);
-    giq_option(4, 891, 149, Bob32, 50);
+    Reply(148);
+    giq_option(4, NAME, 149, Bob32, NEUTRAL_REACTION);
     call BobStandardQuestions();
 end
 
 procedure Bob19
 begin
-    set_global_var(219, 1);
-    gsay_reply(891, 150);
-    giq_option(4, 891, 152, Bob33, 50);
+    set_global_var(GVAR_DECKER_KNOWN, 1);
+    Reply(150);
+    giq_option(4, NAME, 152, Bob33, NEUTRAL_REACTION);
     call BobStandardQuestions();
 end
 
 procedure Bob20
 begin
-    gsay_reply(891, 153);
-    giq_option(4, 891, 154, Bob20a, 50);
-    giq_option(4, 891, 155, Bob34, 50);
-    giq_option(4, 891, 156, Bob35, 50);
-    giq_option(4, 891, 157, Bob36, 50);
+    Reply(153);
+    giq_option(4, NAME, 154, Bob20a, NEUTRAL_REACTION);
+    giq_option(4, NAME, 155, Bob34, NEUTRAL_REACTION);
+    giq_option(4, NAME, 156, Bob35, NEUTRAL_REACTION);
+    giq_option(4, NAME, 157, Bob36, NEUTRAL_REACTION);
 end
 
 procedure Bob20a
@@ -495,57 +495,57 @@ end
 
 procedure Bob21
 begin
-    gsay_reply(891, 239);
+    Reply(239);
     call BobStandardQuestions();
 end
 
 procedure Bob22
 begin
-    gsay_reply(891, 158);
+    Reply(158);
     call BobStandardQuestions();
 end
 
 procedure Bob23
 begin
-    gsay_reply(891, 159);
+    Reply(159);
     call BobStandardQuestions();
 end
 
 procedure Bob25
 begin
-    if ((global_var(203) != 0) and (ToldRumor1 == 0)) then begin
+    if ((global_var(GVAR_DECKER_STATUS) != 0) and (ToldRumor1 == 0)) then begin
         ToldRumor1 := 1;
-        set_global_var(219, 1);
-        gsay_message(891, 160, 50);
+        set_global_var(GVAR_DECKER_KNOWN, 1);
+        gsay_message(NAME, 160, NEUTRAL_REACTION);
     end
     else begin
-        if ((global_var(221) == 1) and (ToldRumor2 == 0)) then begin
+        if ((global_var(GVAR_GREENE_DEAD) == 1) and (ToldRumor2 == 0)) then begin
             ToldRumor2 := 1;
-            gsay_message(891, 161, 50);
+            gsay_message(NAME, 161, NEUTRAL_REACTION);
         end
         else begin
-            if ((global_var(111) == 2) and (ToldRumor3 == 0)) then begin
+            if ((global_var(GVAR_KILL_MERCHANT) == 2) and (ToldRumor3 == 0)) then begin
                 ToldRumor3 := 1;
-                gsay_message(891, 162, 50);
+                gsay_message(NAME, 162, NEUTRAL_REACTION);
             end
             else begin
-                if ((global_var(107) == 2) and (ToldRumor4 == 0)) then begin
+                if ((global_var(GVAR_STEAL_NECKLACE) == 2) and (ToldRumor4 == 0)) then begin
                     ToldRumor4 := 1;
-                    gsay_message(891, 163, 50);
+                    gsay_message(NAME, 163, NEUTRAL_REACTION);
                 end
                 else begin
-                    if ((global_var(112) == 2) and (ToldRumor5 == 0)) then begin
+                    if ((global_var(GVAR_KILL_JAIN) == 2) and (ToldRumor5 == 0)) then begin
                         ToldRumor5 := 1;
-                        gsay_message(891, 164, 50);
+                        gsay_message(NAME, 164, NEUTRAL_REACTION);
                     end
                     else begin
-                        gsay_message(891, 165, 50);
+                        gsay_message(NAME, 165, NEUTRAL_REACTION);
                     end
                 end
             end
         end
     end
-    gsay_message(891, 166, 50);
+    gsay_message(NAME, 166, NEUTRAL_REACTION);
     call BobStandardQuestions();
 end
 
@@ -554,13 +554,13 @@ begin
     variable LVar0 := 0;
     variable LVar1 := 0;
     if (item_caps_total(dude_obj) < 20) then begin
-        gsay_message(891, 169, 50);
+        gsay_message(NAME, 169, NEUTRAL_REACTION);
     end
     else begin
         LVar1 := item_caps_adjust(dude_obj, -20);
-        LVar0 := create_object_sid(81, 0, 0, -1);
+        LVar0 := create_object(81, 0, 0);
         add_obj_to_inven(dude_obj, LVar0);
-        gsay_message(891, 170, 50);
+        gsay_message(NAME, 170, NEUTRAL_REACTION);
     end
     call BobStandardQuestions();
 end
@@ -570,18 +570,18 @@ begin
     variable LVar0 := 0;
     variable LVar1 := 0;
     if (item_caps_total(dude_obj) < 8) then begin
-        gsay_message(891, 173, 50);
+        gsay_message(NAME, 173, NEUTRAL_REACTION);
     end
     else begin
         LVar1 := item_caps_adjust(dude_obj, -8);
-        LVar0 := create_object_sid(103, 0, 0, -1);
+        LVar0 := create_object(103, 0, 0);
         add_obj_to_inven(dude_obj, LVar0);
         if (local_var(5) > 0) then begin
-            gsay_message(891, 171, 50);
-            giq_option(4, 891, 172, Bob38, 50);
+            gsay_message(NAME, 171, NEUTRAL_REACTION);
+            giq_option(4, NAME, 172, Bob38, NEUTRAL_REACTION);
         end
         else begin
-            gsay_message(891, 174, 50);
+            gsay_message(NAME, 174, NEUTRAL_REACTION);
         end
     end
     call BobStandardQuestions();
@@ -589,84 +589,84 @@ end
 
 procedure Bob30
 begin
-    gsay_message(891, 175, 50);
+    gsay_message(NAME, 175, NEUTRAL_REACTION);
 end
 
 procedure Bob31
 begin
-    gsay_reply(891, 176);
-    giq_option(4, 891, 177, Bob42, 50);
-    giq_option(4, 891, 178, Bob43, 50);
-    giq_option(4, 891, 179, Bob44, 50);
+    Reply(176);
+    giq_option(4, NAME, 177, Bob42, NEUTRAL_REACTION);
+    giq_option(4, NAME, 178, Bob43, NEUTRAL_REACTION);
+    giq_option(4, NAME, 179, Bob44, NEUTRAL_REACTION);
 end
 
 procedure Bob32
 begin
-    gsay_reply(891, 180);
+    Reply(180);
     call BobStandardQuestions();
 end
 
 procedure Bob33
 begin
-    set_global_var(219, 1);
-    gsay_reply(891, 182);
+    set_global_var(GVAR_DECKER_KNOWN, 1);
+    Reply(182);
     call BobStandardQuestions();
 end
 
 procedure Bob34
 begin
     set_local_var(5, 2);
-    gsay_message(891, 183, 51);
+    gsay_message(NAME, 183, BAD_REACTION);
 end
 
 procedure Bob35
 begin
-    gsay_reply(891, 184);
-    giq_option(4, 891, 185, Bob42, 50);
-    giq_option(4, 891, 186, Bob43, 50);
-    giq_option(4, 891, 187, Bob44, 50);
+    Reply(184);
+    giq_option(4, NAME, 185, Bob42, NEUTRAL_REACTION);
+    giq_option(4, NAME, 186, Bob43, NEUTRAL_REACTION);
+    giq_option(4, NAME, 187, Bob44, NEUTRAL_REACTION);
 end
 
 procedure Bob36
 begin
-    gsay_reply(891, 188);
-    giq_option(4, 891, 189, Bob42, 50);
-    giq_option(4, 891, 190, Bob43, 50);
-    giq_option(4, 891, 191, Bob44, 50);
+    Reply(188);
+    giq_option(4, NAME, 189, Bob42, NEUTRAL_REACTION);
+    giq_option(4, NAME, 190, Bob43, NEUTRAL_REACTION);
+    giq_option(4, NAME, 191, Bob44, NEUTRAL_REACTION);
 end
 
 procedure Bob38
 begin
-    gsay_message(891, 194, 51);
+    gsay_message(NAME, 194, BAD_REACTION);
 end
 
 procedure Bob39
 begin
-    gsay_message(891, 195, 50);
+    gsay_message(NAME, 195, NEUTRAL_REACTION);
 end
 
 procedure Bob42
 begin
-    gsay_message(891, 207, 50);
+    gsay_message(NAME, 207, NEUTRAL_REACTION);
 end
 
 procedure Bob43
 begin
-    gsay_reply(891, 208);
-    giq_option(4, 891, 209, BobCombat, 50);
-    giq_option(4, 891, 210, Bob45, 50);
-    giq_option(4, 891, 211, Bob44, 50);
+    Reply(208);
+    giq_option(4, NAME, 209, BobCombat, NEUTRAL_REACTION);
+    giq_option(4, NAME, 210, Bob45, NEUTRAL_REACTION);
+    giq_option(4, NAME, 211, Bob44, NEUTRAL_REACTION);
 end
 
 procedure Bob44
 begin
-    gsay_reply(891, 212);
-    giq_option(4, 891, message_str(891, 213) + message_str(891, 214) + message_str(891, 219), Bob44a, 50);
-    giq_option(4, 891, message_str(891, 213) + message_str(891, 215) + message_str(891, 219), Bob44b, 50);
-    giq_option(4, 891, message_str(891, 213) + message_str(891, 216) + message_str(891, 219), Bob44c, 50);
-    giq_option(4, 891, message_str(891, 213) + message_str(891, 217) + message_str(891, 219), Bob44d, 50);
-    giq_option(4, 891, message_str(891, 213) + message_str(891, 218) + message_str(891, 219), Bob44e, 50);
-    giq_option(4, 891, 220, Bob48, 50);
+    Reply(212);
+    giq_option(4, NAME, mstr(213) + mstr(214) + mstr(219), Bob44a, NEUTRAL_REACTION);
+    giq_option(4, NAME, mstr(213) + mstr(215) + mstr(219), Bob44b, NEUTRAL_REACTION);
+    giq_option(4, NAME, mstr(213) + mstr(216) + mstr(219), Bob44c, NEUTRAL_REACTION);
+    giq_option(4, NAME, mstr(213) + mstr(217) + mstr(219), Bob44d, NEUTRAL_REACTION);
+    giq_option(4, NAME, mstr(213) + mstr(218) + mstr(219), Bob44e, NEUTRAL_REACTION);
+    giq_option(4, NAME, 220, Bob48, NEUTRAL_REACTION);
 end
 
 procedure Bob44a
@@ -726,42 +726,42 @@ end
 
 procedure Bob45
 begin
-    gsay_message(891, 221, 51);
+    gsay_message(NAME, 221, BAD_REACTION);
 end
 
 procedure Bob46
 begin
     if (local_var(5) != 1) then begin
         give_exp_points(500);
-        display_msg(message_str(766, 103) + 500 + message_str(766, 104));
+        display_msg(message_str(SCRIPT_GENCHAT, 103) + 500 + message_str(SCRIPT_GENCHAT, 104));
     end
     set_local_var(5, 1);
     set_local_var(6, (game_time / (10 * 60 * 60 * 24)) + 5);
-    set_global_var(155, global_var(155) - 2);
-    gsay_message(891, message_str(891, 222) + local_var(7) + message_str(891, 223), 50);
+    set_global_var(GVAR_PLAYER_REPUATION, global_var(GVAR_PLAYER_REPUATION) - 2);
+    gsay_message(NAME, mstr(222) + local_var(7) + mstr(223), NEUTRAL_REACTION);
 end
 
 procedure Bob47
 begin
-    gsay_reply(891, 224);
-    giq_option(4, 891, 225, Bob49, 50);
-    giq_option(4, 891, 226, Bob50, 50);
-    giq_option(4, 891, 227, Bob13, 50);
+    Reply(224);
+    giq_option(4, NAME, 225, Bob49, NEUTRAL_REACTION);
+    giq_option(4, NAME, 226, Bob50, NEUTRAL_REACTION);
+    giq_option(4, NAME, 227, Bob13, NEUTRAL_REACTION);
 end
 
 procedure Bob48
 begin
-    gsay_message(891, 228, 51);
+    gsay_message(NAME, 228, BAD_REACTION);
 end
 
 procedure Bob49
 begin
-    gsay_message(891, 229, 51);
+    gsay_message(NAME, 229, BAD_REACTION);
 end
 
 procedure Bob50
 begin
-    gsay_message(891, 230, 50);
+    gsay_message(NAME, 230, NEUTRAL_REACTION);
 end
 
 procedure BobEnd
@@ -779,31 +779,31 @@ begin
         set_local_var(0, 50);
         set_local_var(1, 2);
         set_local_var(2, 1);
-        set_local_var(0, local_var(0) + (5 * get_critter_stat(dude_obj, 3)) - 25);
-        set_local_var(0, local_var(0) + (10 * has_trait(0, dude_obj, 10)));
-        if (has_trait(0, dude_obj, 39)) then begin
-            if (global_var(155) > 0) then begin
-                set_local_var(0, local_var(0) + global_var(155));
+        set_local_var(0, local_var(0) + (5 * dude_charisma) - 25);
+        set_local_var(0, local_var(0) + (10 * has_trait(TRAIT_PERK, dude_obj, PERK_presence)));
+        if (has_trait(TRAIT_PERK, dude_obj, PERK_cult_of_personality)) then begin
+            if (global_var(GVAR_PLAYER_REPUATION) > 0) then begin
+                set_local_var(0, local_var(0) + global_var(GVAR_PLAYER_REPUATION));
             end
             else begin
-                set_local_var(0, local_var(0) - global_var(155));
+                set_local_var(0, local_var(0) - global_var(GVAR_PLAYER_REPUATION));
             end
         end
         else begin
             if (local_var(3) == 1) then begin
-                set_local_var(0, local_var(0) - global_var(155));
+                set_local_var(0, local_var(0) - global_var(GVAR_PLAYER_REPUATION));
             end
             else begin
-                set_local_var(0, local_var(0) + global_var(155));
+                set_local_var(0, local_var(0) + global_var(GVAR_PLAYER_REPUATION));
             end
         end
-        if (global_var(158) > 2) then begin
+        if (global_var(GVAR_CHILDKILLER_REPUATION) > 2) then begin
             set_local_var(0, local_var(0) - 30);
         end
-        if (((global_var(160) + global_var(159)) >= 25) and ((global_var(160) > (3 * global_var(159))) or (global_var(157) == 1))) then begin
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(160) > (3 * global_var(GVAR_GOOD_MONSTER))) or (global_var(GVAR_CHAMPION_REPUTATION) == 1))) then begin
             set_local_var(0, local_var(0) + 20);
         end
-        if (((global_var(160) + global_var(159)) >= 25) and ((global_var(159) > (2 * global_var(160))) or (global_var(156) == 1))) then begin
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
             set_local_var(0, local_var(0) - 20);
         end
         call ReactToLevel();
@@ -896,6 +896,5 @@ end
 
 procedure Goodbyes
 begin
-    exit_line := message_str(634, random(100, 105));
+    exit_line := message_str(SCRIPT_MODREACT, random(100, 105));
 end
-
