diff --git a/Games/Fallout1.Steam/Original/BOXGUARD.ssl b/Games/Fallout1.Steam/ReDefine/BOXGUARD.ssl
index c8e8ec2..1a6c237 100644
--- a/Games/Fallout1.Steam/Original/BOXGUARD.ssl
+++ b/Games/Fallout1.Steam/ReDefine/BOXGUARD.ssl
@@ -17,19 +17,19 @@ import variable shot_challenger;
 
 procedure start
 begin
-    if (script_action == 12) then begin
+    if (script_action == critter_proc) then begin
         call critter_p_proc();
     end
     else begin
-        if (script_action == 18) then begin
+        if (script_action == destroy_proc) then begin
             call destroy_p_proc();
         end
         else begin
-            if (script_action == 11) then begin
+            if (script_action == talk_proc) then begin
                 call talk_p_proc();
             end
             else begin
-                if (script_action == 22) then begin
+                if (script_action == timed_event_proc) then begin
                     call timed_event_p_proc();
                 end
             end
@@ -41,7 +41,7 @@ procedure critter_p_proc
 begin
     if (hostile) then begin
         hostile := 0;
-        attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+        attack(dude_obj);
     end
     else begin
         if (Saul_loses) then begin
@@ -49,7 +49,7 @@ begin
                 if (shot_challenger == 0) then begin
                     Saul_loses := 0;
                     anim(self_obj, 1000, 5);
-                    float_msg(self_obj, message_str(835, random(100, 103)), 0);
+                    float_msg(self_obj, mstr(random(100, 103)), FLOAT_MSG_NORMAL);
                     reg_anim_func(2, self_obj);
                     reg_anim_func(1, 1);
                     reg_anim_animate(self_obj, 43, -1);
@@ -83,31 +83,31 @@ end
 procedure destroy_p_proc
 begin
     if (source_obj == dude_obj) then begin
-        if (((global_var(160) + global_var(159)) >= 25) and ((global_var(159) > (2 * global_var(160))) or (global_var(156) == 1))) then begin
-            set_global_var(156, 1);
-            set_global_var(157, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
+            set_global_var(GVAR_BERSERKER_REPUTATION, 1);
+            set_global_var(GVAR_CHAMPION_REPUTATION, 0);
         end
-        if (((global_var(160) + global_var(159)) >= 25) and ((global_var(160) > (3 * global_var(159))) or (global_var(157) == 1))) then begin
-            set_global_var(157, 1);
-            set_global_var(156, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(160) > (3 * global_var(GVAR_GOOD_MONSTER))) or (global_var(GVAR_CHAMPION_REPUTATION) == 1))) then begin
+            set_global_var(GVAR_CHAMPION_REPUTATION, 1);
+            set_global_var(GVAR_BERSERKER_REPUTATION, 0);
         end
-        set_global_var(160, global_var(160) + 1);
-        if ((global_var(160) % 6) == 0) then begin
-            set_global_var(155, global_var(155) + 1);
+        set_global_var(GVAR_BAD_MONSTER, global_var(GVAR_BAD_MONSTER) + 1);
+        if ((global_var(GVAR_BAD_MONSTER) % 6) == 0) then begin
+            set_global_var(GVAR_PLAYER_REPUATION, global_var(GVAR_PLAYER_REPUATION) + 1);
         end
     end
 end
 
 procedure map_enter_p_proc
 begin
-    if (global_var(15) == 1) then begin
+    if (global_var(GVAR_JUNKTOWN_INVADED) == 1) then begin
         kill_critter(self_obj, 49);
     end
     if (local_var(0) == 0) then begin
         set_local_var(0, tile_num(self_obj));
     end
     home_tile := local_var(0);
-    critter_add_trait(self_obj, 1, 6, 25);
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 25);
     if (item_caps_total(self_obj) == 0) then begin
         item_caps_adjust(self_obj, random(10, 25));
     end
@@ -115,7 +115,7 @@ end
 
 procedure talk_p_proc
 begin
-    float_msg(self_obj, message_str(835, random(106, 107)), 0);
+    float_msg(self_obj, mstr(random(106, 107)), FLOAT_MSG_NORMAL);
 end
 
 procedure timed_event_p_proc
@@ -123,4 +123,3 @@ begin
     Saul_loses := 1;
     shot_challenger := 1;
 end
-
