diff --git a/Games/Fallout1.Steam/Original/BRAHMIN.ssl b/Games/Fallout1.Steam/ReDefine/BRAHMIN.ssl
index 75c1595..98c04b2 100644
--- a/Games/Fallout1.Steam/Original/BRAHMIN.ssl
+++ b/Games/Fallout1.Steam/ReDefine/BRAHMIN.ssl
@@ -14,27 +14,27 @@ variable initialized;
 
 procedure start
 begin
-    if (script_action == 12) then begin
+    if (script_action == critter_proc) then begin
         call critter_p_proc();
     end
     else begin
-        if (script_action == 3) then begin
+        if (script_action == description_proc) then begin
             call description_p_proc();
         end
         else begin
-            if (script_action == 11) then begin
+            if (script_action == talk_proc) then begin
                 call talk_p_proc();
             end
             else begin
-                if (script_action == 22) then begin
+                if (script_action == timed_event_proc) then begin
                     call timed_event_p_proc();
                 end
                 else begin
-                    if (script_action == 7) then begin
+                    if (script_action == use_obj_on_proc) then begin
                         call use_obj_on_p_proc();
                     end
                     else begin
-                        if (script_action == 8) then begin
+                        if (script_action == use_skill_on_proc) then begin
                             call use_skill_on_p_proc();
                         end
                     end
@@ -54,12 +54,12 @@ end
 
 procedure description_p_proc
 begin
-    display_msg(message_str(34, 100));
+    display_mstr(100);
 end
 
 procedure talk_p_proc
 begin
-    float_msg(self_obj, message_str(34, 103), 0);
+    float_msg(self_obj, mstr(103), FLOAT_MSG_NORMAL);
 end
 
 procedure timed_event_p_proc
@@ -97,10 +97,10 @@ begin
     if (action_being_used == 12) then begin
         script_overrides;
         if (is_success(roll_vs_skill(dude_obj, action_being_used, 0))) then begin
-            display_msg(message_str(34, 101));
+            display_mstr(101);
         end
         else begin
-            display_msg(message_str(34, 102));
+            display_mstr(102);
         end
     end
 end
@@ -108,14 +108,13 @@ end
 procedure damage_p_proc
 begin
     if (source_obj == dude_obj) then begin
-        set_global_var(247, 1);
+        set_global_var(GVAR_ENEMY_JUNKTOWN, 1);
     end
 end
 
 procedure destroy_p_proc
 begin
     if (source_obj == dude_obj) then begin
-        set_global_var(247, 1);
+        set_global_var(GVAR_ENEMY_JUNKTOWN, 1);
     end
 end
-
