diff --git a/Games/Fallout1.Steam/Original/BROELD.ssl b/Games/Fallout1.Steam/ReDefine/BROELD.ssl
index 880ef6b..093d4c9 100644
--- a/Games/Fallout1.Steam/Original/BROELD.ssl
+++ b/Games/Fallout1.Steam/ReDefine/BROELD.ssl
@@ -13,26 +13,26 @@ procedure start
 begin
     if (Only_Once) then begin
         Only_Once := 0;
-        critter_add_trait(self_obj, 1, 6, 44);
-        critter_add_trait(self_obj, 1, 5, 80);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 44);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_BOS_ELDERS);
     end
-    if (script_action == 11) then begin
+    if (script_action == talk_proc) then begin
         call talk_p_proc();
     end
     else begin
-        if (script_action == 12) then begin
+        if (script_action == critter_proc) then begin
             call critter_p_proc();
         end
         else begin
-            if (script_action == 4) then begin
+            if (script_action == pickup_proc) then begin
                 call pickup_p_proc();
             end
             else begin
-                if (script_action == 21) then begin
+                if (script_action == look_at_proc) then begin
                     call look_at_p_proc();
                 end
                 else begin
-                    if (script_action == 18) then begin
+                    if (script_action == destroy_proc) then begin
                         call destroy_p_proc();
                     end
                 end
@@ -43,22 +43,22 @@ end
 
 procedure critter_p_proc
 begin
-    if (global_var(250)) then begin
+    if (global_var(GVAR_ENEMY_BROTHERHOOD)) then begin
         hostile := 1;
     end
     if (tile_distance_objs(self_obj, dude_obj) > 12) then begin
         hostile := 0;
     end
     if (hostile) then begin
-        set_global_var(250, 1);
+        set_global_var(GVAR_ENEMY_BROTHERHOOD, 1);
         hostile := 0;
-        attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+        attack(dude_obj);
     end
 end
 
 procedure talk_p_proc
 begin
-    float_msg(self_obj, message_str(944, random(101, 105)), 2);
+    float_msg(self_obj, mstr(random(101, 105)), FLOAT_MSG_RED);
 end
 
 procedure pickup_p_proc
@@ -68,11 +68,10 @@ end
 
 procedure look_at_p_proc
 begin
-    display_msg(message_str(944, 100));
+    display_mstr(100);
 end
 
 procedure destroy_p_proc
 begin
-    set_global_var(250, 1);
+    set_global_var(GVAR_ENEMY_BROTHERHOOD, 1);
 end
-
