diff --git a/Games/Fallout1.Steam/Original/BROHDENT.ssl b/Games/Fallout1.Steam/ReDefine/BROHDENT.ssl
index bf7924c..a7b7bf6 100644
--- a/Games/Fallout1.Steam/Original/BROHDENT.ssl
+++ b/Games/Fallout1.Steam/ReDefine/BROHDENT.ssl
@@ -24,23 +24,23 @@ procedure Invasion;
 
 procedure start
 begin
-    if (script_action == 15) then begin
-        if (metarule(14, 0)) then begin
-            display_msg(message_str(194, 107));
+    if (script_action == map_enter_proc) then begin
+        if (map_first_run) then begin
+            display_msg(message_str(SCRIPT_SHADYWST, 107));
         end
-        set_global_var(582, 1);
-        if ((game_time / (10 * 60 * 60 * 24)) >= global_var(151)) then begin
-            set_global_var(16, 1);
+        set_global_var(GVAR_MARK_BROTHER_1, 1);
+        if ((game_time / (10 * 60 * 60 * 24)) >= global_var(GVAR_BROTHERHOOD_INVADED_DATE)) then begin
+            set_global_var(GVAR_BROTHERHOOD_INVADED, 1);
         end
         call lighting();
         call add_party();
     end
-    if (script_action == 23) then begin
+    if (script_action == map_update_proc) then begin
         call lighting();
         call update_party();
     end
     else begin
-        if (script_action == 16) then begin
+        if (script_action == map_exit_proc) then begin
             call remove_party();
         end
     end
@@ -72,11 +72,11 @@ procedure add_party
 begin
     variable LVar0 := 0;
     variable LVar1 := 0;
-    party_elevation := elevation(dude_obj);
-    if (global_var(26) == 5) then begin
+    party_elevation := dude_elevation;
+    if (global_var(GVAR_TANDI_STATUS) == 5) then begin
         if (Tandi_ptr == 0) then begin
         end
-        critter_add_trait(Tandi_ptr, 1, 6, 0);
+        critter_add_trait(Tandi_ptr, TRAIT_OBJECT, OBJECT_TEAM_NUM, 0);
     end
 end
 
@@ -84,41 +84,41 @@ procedure update_party
 begin
     variable LVar0 := 0;
     variable LVar1 := 0;
-    if (elevation(dude_obj) != party_elevation) then begin
-        party_elevation := elevation(dude_obj);
-        if (global_var(118) == 2) then begin
+    if (dude_elevation != party_elevation) then begin
+        party_elevation := dude_elevation;
+        if (global_var(GVAR_IAN_STATUS) == 2) then begin
             if (Ian_ptr == 0) then begin
             end
             else begin
-                move_to(Ian_ptr, tile_num_in_direction(tile_num(dude_obj), 1, 2), elevation(dude_obj));
+                move_to(Ian_ptr, tile_num_in_direction(dude_tile, 1, 2), dude_elevation);
             end
         end
-        if (global_var(5)) then begin
+        if (global_var(GVAR_DOG_EMPATHY)) then begin
             if (Dog_ptr == 0) then begin
             end
             else begin
-                move_to(Dog_ptr, tile_num_in_direction(tile_num(dude_obj), 2, 1), elevation(dude_obj));
+                move_to(Dog_ptr, tile_num_in_direction(dude_tile, 2, 1), dude_elevation);
             end
         end
-        if (global_var(121) == 2) then begin
+        if (global_var(GVAR_HERNANDEZ_STATUS) == 2) then begin
             if (Tycho_ptr == 0) then begin
             end
             else begin
-                move_to(Tycho_ptr, tile_num_in_direction(tile_num(dude_obj), 3, 2), elevation(dude_obj));
+                move_to(Tycho_ptr, tile_num_in_direction(dude_tile, 3, 2), dude_elevation);
             end
         end
-        if (global_var(244) == 2) then begin
+        if (global_var(GVAR_KATJA_STATUS) == 2) then begin
             if (Katja_ptr == 0) then begin
             end
             else begin
-                move_to(Katja_ptr, tile_num_in_direction(tile_num(dude_obj), 4, 2), elevation(dude_obj));
+                move_to(Katja_ptr, tile_num_in_direction(dude_tile, 4, 2), dude_elevation);
             end
         end
-        if (global_var(26) == 5) then begin
+        if (global_var(GVAR_TANDI_STATUS) == 5) then begin
             if (Tandi_ptr == 0) then begin
             end
             else begin
-                move_to(Tandi_ptr, tile_num_in_direction(tile_num(dude_obj), 4, 4), elevation(dude_obj));
+                move_to(Tandi_ptr, tile_num_in_direction(dude_tile, 4, 4), dude_elevation);
             end
         end
     end
@@ -126,19 +126,19 @@ end
 
 procedure remove_party
 begin
-    if (global_var(118) == 2) then begin
-        set_global_var(118, 2);
+    if (global_var(GVAR_IAN_STATUS) == 2) then begin
+        set_global_var(GVAR_IAN_STATUS, 2);
     end
-    if (global_var(5)) then begin
-        set_global_var(5, 1);
+    if (global_var(GVAR_DOG_EMPATHY)) then begin
+        set_global_var(GVAR_DOG_EMPATHY, 1);
     end
-    if (global_var(121) == 2) then begin
-        set_global_var(121, 2);
+    if (global_var(GVAR_HERNANDEZ_STATUS) == 2) then begin
+        set_global_var(GVAR_HERNANDEZ_STATUS, 2);
     end
-    if (global_var(244) == 2) then begin
-        set_global_var(244, 2);
+    if (global_var(GVAR_KATJA_STATUS) == 2) then begin
+        set_global_var(GVAR_KATJA_STATUS, 2);
     end
-    if (global_var(26) == 5) then begin
+    if (global_var(GVAR_TANDI_STATUS) == 5) then begin
     end
 end
 
@@ -149,28 +149,27 @@ end
 
 procedure Invasion
 begin
-    if (not(global_var(18) == 2)) then begin
-        if (global_var(149) > (game_time / (10 * 60 * 60 * 24))) then begin
-            set_global_var(13, 1);
+    if (not(global_var(GVAR_MASTER_BLOWN) == 2)) then begin
+        if (global_var(GVAR_NECROPOLIS_INVADED_DATE) > (game_time / (10 * 60 * 60 * 24))) then begin
+            set_global_var(GVAR_NECROPOLIS_INVADED, 1);
         end
-        if (global_var(150) > (game_time / (10 * 60 * 60 * 24))) then begin
-            set_global_var(14, 1);
+        if (global_var(GVAR_THE_HUB_INVADED_DATE) > (game_time / (10 * 60 * 60 * 24))) then begin
+            set_global_var(GVAR_HUB_INVADED, 1);
         end
-        if (global_var(151) > (game_time / (10 * 60 * 60 * 24))) then begin
-            set_global_var(16, 1);
+        if (global_var(GVAR_BROTHERHOOD_INVADED_DATE) > (game_time / (10 * 60 * 60 * 24))) then begin
+            set_global_var(GVAR_BROTHERHOOD_INVADED, 1);
         end
-        if (global_var(152) > (game_time / (10 * 60 * 60 * 24))) then begin
-            set_global_var(15, 1);
+        if (global_var(GVAR_JUNKTOWN_INVADED_DATE) > (game_time / (10 * 60 * 60 * 24))) then begin
+            set_global_var(GVAR_JUNKTOWN_INVADED, 1);
         end
-        if (global_var(153) > (game_time / (10 * 60 * 60 * 24))) then begin
-            set_global_var(12, 1);
+        if (global_var(GVAR_SHADY_SANDS_INVADED_DATE) > (game_time / (10 * 60 * 60 * 24))) then begin
+            set_global_var(GVAR_SHADY_SANDS_INVADED, 1);
         end
-        if (global_var(154) <= 0) then begin
-            set_global_var(11, 1);
+        if (global_var(GVAR_VAULT_13_INVADED_DATE) <= 0) then begin
+            set_global_var(GVAR_VAULT_INVADED, 1);
         end
-        if (global_var(148) > (game_time / (10 * 60 * 60 * 24))) then begin
-            set_global_var(7, 1);
+        if (global_var(GVAR_FOLLOWERS_INVADED_DATE) > (game_time / (10 * 60 * 60 * 24))) then begin
+            set_global_var(GVAR_FOLLOWERS_INVADED, 1);
         end
     end
 end
-
