diff --git a/Games/Fallout1.Steam/Original/BULLBORD.ssl b/Games/Fallout1.Steam/ReDefine/BULLBORD.ssl
index b5ad381..c99744e 100644
--- a/Games/Fallout1.Steam/Original/BULLBORD.ssl
+++ b/Games/Fallout1.Steam/ReDefine/BULLBORD.ssl
@@ -10,15 +10,15 @@ procedure bull04;
 
 procedure start
 begin
-    if ((script_action == 21) or (script_action == 3)) then begin
+    if ((script_action == look_at_proc) or (script_action == description_proc)) then begin
         call look_at_p_proc();
     end
     else begin
-        if (script_action == 6) then begin
+        if (script_action == use_proc) then begin
             call use_p_proc();
         end
         else begin
-            if (script_action == 11) then begin
+            if (script_action == talk_proc) then begin
                 call talk_p_proc();
             end
         end
@@ -33,7 +33,7 @@ end
 procedure look_at_p_proc
 begin
     script_overrides;
-    display_msg(message_str(839, 100));
+    display_mstr(100);
 end
 
 procedure talk_p_proc
@@ -48,24 +48,23 @@ end
 
 procedure bull01
 begin
-    gsay_reply(839, 101);
-    giq_option(0, 839, 105, bull02, 50);
+    Reply(101);
+    giq_option(0, NAME, 105, bull02, NEUTRAL_REACTION);
 end
 
 procedure bull02
 begin
-    gsay_reply(839, 102);
-    giq_option(0, 839, 105, bull03, 50);
+    Reply(102);
+    giq_option(0, NAME, 105, bull03, NEUTRAL_REACTION);
 end
 
 procedure bull03
 begin
-    gsay_reply(839, 103);
-    giq_option(0, 839, 105, bull04, 50);
+    Reply(103);
+    giq_option(0, NAME, 105, bull04, NEUTRAL_REACTION);
 end
 
 procedure bull04
 begin
-    gsay_message(839, 104, 50);
+    gsay_message(NAME, 104, NEUTRAL_REACTION);
 end
-
