diff --git a/Games/Fallout1.Steam/Original/BVENT.ssl b/Games/Fallout1.Steam/ReDefine/BVENT.ssl
index abd850b..b288afe 100644
--- a/Games/Fallout1.Steam/Original/BVENT.ssl
+++ b/Games/Fallout1.Steam/ReDefine/BVENT.ssl
@@ -3,18 +3,17 @@ procedure start;
 
 procedure start
 begin
-    if (script_action == 2) then begin
+    if (script_action == spatial_proc) then begin
         if (source_obj == dude_obj) then begin
             if (local_var(0) == 0) then begin
                 set_local_var(0, 1);
                 if (is_success(do_check(dude_obj, 4, 0))) then begin
-                    display_msg(message_str(192, 100));
+                    display_mstr(100);
                 end
                 if (is_success(roll_vs_skill(dude_obj, 11, 0))) then begin
-                    display_msg(message_str(192, 101));
+                    display_mstr(101);
                 end
             end
         end
     end
 end
-
