diff --git a/Games/Fallout1.Steam/Original/CATHSHOP.ssl b/Games/Fallout1.Steam/ReDefine/CATHSHOP.ssl
index adb9d6c..d4fa55f 100644
--- a/Games/Fallout1.Steam/Original/CATHSHOP.ssl
+++ b/Games/Fallout1.Steam/ReDefine/CATHSHOP.ssl
@@ -54,31 +54,31 @@ procedure start
 begin
     if (Only_Once) then begin
         Only_Once := 0;
-        critter_add_trait(self_obj, 1, 6, 20);
-        critter_add_trait(self_obj, 1, 5, 69);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 20);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_CHILDREN_TRUE);
         Shopkepper_Ptr := self_obj;
     end
-    if (script_action == 21) then begin
+    if (script_action == look_at_proc) then begin
         call look_at_p_proc();
     end
     else begin
-        if (script_action == 22) then begin
+        if (script_action == timed_event_proc) then begin
             call timed_event_p_proc();
         end
         else begin
-            if (script_action == 4) then begin
+            if (script_action == pickup_proc) then begin
                 call pickup_p_proc();
             end
             else begin
-                if (script_action == 11) then begin
+                if (script_action == talk_proc) then begin
                     call talk_p_proc();
                 end
                 else begin
-                    if (script_action == 12) then begin
+                    if (script_action == critter_proc) then begin
                         call critter_p_proc();
                     end
                     else begin
-                        if (script_action == 18) then begin
+                        if (script_action == destroy_proc) then begin
                             call destroy_p_proc();
                         end
                     end
@@ -97,7 +97,7 @@ procedure critter_p_proc
 begin
     if (hostile) then begin
         hostile := 0;
-        attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+        attack(dude_obj);
     end
 end
 
@@ -122,7 +122,7 @@ begin
         call BarterGuy00();
     end
     else begin
-        if ((obj_item_subtype(critter_inven_obj(dude_obj, 1)) == 3) or (obj_item_subtype(critter_inven_obj(dude_obj, 2)) == 3) or (global_var(195) == 1)) then begin
+        if ((obj_item_subtype(critter_inven_obj(dude_obj, 1)) == 3) or (obj_item_subtype(critter_inven_obj(dude_obj, 2)) == 3) or (global_var(GVAR_CATHEDRAL_ENEMY) == 1)) then begin
             call BarterGuy02();
         end
         else begin
@@ -140,7 +140,7 @@ begin
     add_obj_to_inven(self_obj, LVar1);
     wield_obj_critter(self_obj, LVar1);
     if (Go_Balistic) then begin
-        LVar0 := create_object_sid(12, 0, 0, -1);
+        LVar0 := create_object(12, 0, 0);
         add_obj_to_inven(self_obj, LVar0);
         wield_obj_critter(self_obj, LVar0);
     end
@@ -149,17 +149,17 @@ end
 procedure destroy_p_proc
 begin
     if (source_obj == dude_obj) then begin
-        if (((global_var(160) + global_var(159)) >= 25) and ((global_var(159) > (2 * global_var(160))) or (global_var(156) == 1))) then begin
-            set_global_var(156, 1);
-            set_global_var(157, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
+            set_global_var(GVAR_BERSERKER_REPUTATION, 1);
+            set_global_var(GVAR_CHAMPION_REPUTATION, 0);
         end
-        if (((global_var(160) + global_var(159)) >= 25) and ((global_var(160) > (3 * global_var(159))) or (global_var(157) == 1))) then begin
-            set_global_var(157, 1);
-            set_global_var(156, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(160) > (3 * global_var(GVAR_GOOD_MONSTER))) or (global_var(GVAR_CHAMPION_REPUTATION) == 1))) then begin
+            set_global_var(GVAR_CHAMPION_REPUTATION, 1);
+            set_global_var(GVAR_BERSERKER_REPUTATION, 0);
         end
-        set_global_var(159, global_var(159) + 1);
-        if ((global_var(159) % 2) == 0) then begin
-            set_global_var(155, global_var(155) - 1);
+        set_global_var(GVAR_GOOD_MONSTER, global_var(GVAR_GOOD_MONSTER) + 1);
+        if ((global_var(GVAR_GOOD_MONSTER) % 2) == 0) then begin
+            set_global_var(GVAR_PLAYER_REPUATION, global_var(GVAR_PLAYER_REPUATION) - 1);
         end
     end
 end
@@ -167,7 +167,7 @@ end
 procedure look_at_p_proc
 begin
     script_overrides;
-    display_msg(message_str(843, 100));
+    display_mstr(100);
 end
 
 procedure timed_event_p_proc
@@ -186,10 +186,10 @@ end
 
 procedure BarterGuy00
 begin
-    gsay_reply(843, 101);
-    giq_option(-3, 843, 102, BarterGuy01, 50);
-    giq_option(4, 843, 103, BarterGuy01, 50);
-    giq_option(5, 843, 104, BarterGuy00a, 50);
+    Reply(101);
+    giq_option(-3, NAME, 102, BarterGuy01, NEUTRAL_REACTION);
+    giq_option(4, NAME, 103, BarterGuy01, NEUTRAL_REACTION);
+    giq_option(5, NAME, 104, BarterGuy00a, NEUTRAL_REACTION);
 end
 
 procedure BarterGuy00a
@@ -204,21 +204,21 @@ end
 
 procedure BarterGuy01
 begin
-    gsay_message(843, 105, 51);
+    gsay_message(NAME, 105, BAD_REACTION);
     call combat();
 end
 
 procedure BarterGuy02
 begin
-    gsay_reply(843, 106);
-    giq_option(-3, 843, 107, BarterGuyEnd, 50);
-    giq_option(4, 843, 108, BarterGuy03, 50);
-    giq_option(6, 843, 109, BarterGuy02a, 50);
+    Reply(106);
+    giq_option(-3, NAME, 107, BarterGuyEnd, NEUTRAL_REACTION);
+    giq_option(4, NAME, 108, BarterGuy03, NEUTRAL_REACTION);
+    giq_option(6, NAME, 109, BarterGuy02a, NEUTRAL_REACTION);
 end
 
 procedure BarterGuy02a
 begin
-    if (global_var(195) == 1) then begin
+    if (global_var(GVAR_CATHEDRAL_ENEMY) == 1) then begin
         call BarterGuy04();
     end
     else begin
@@ -228,79 +228,79 @@ end
 
 procedure BarterGuy03
 begin
-    gsay_message(843, 110, 50);
+    gsay_message(NAME, 110, NEUTRAL_REACTION);
 end
 
 procedure BarterGuy04
 begin
-    gsay_reply(843, 111);
-    giq_option(6, 843, 112, BarterGuy01, 50);
-    giq_option(6, 843, 113, BarterGuyEnd, 50);
+    Reply(111);
+    giq_option(6, NAME, 112, BarterGuy01, NEUTRAL_REACTION);
+    giq_option(6, NAME, 113, BarterGuyEnd, NEUTRAL_REACTION);
 end
 
 procedure BarterGuy05
 begin
-    gsay_reply(843, 114);
-    giq_option(6, 843, 115, BarterGuy00a, 50);
-    giq_option(6, 843, 116, BarterGuy06, 50);
+    Reply(114);
+    giq_option(6, NAME, 115, BarterGuy00a, NEUTRAL_REACTION);
+    giq_option(6, NAME, 116, BarterGuy06, NEUTRAL_REACTION);
 end
 
 procedure BarterGuy06
 begin
-    gsay_message(843, 117, 50);
+    gsay_message(NAME, 117, NEUTRAL_REACTION);
 end
 
 procedure BarterGuy07
 begin
-    gsay_reply(843, 118);
-    giq_option(-3, 843, 119, BarterGuy08, 50);
-    giq_option(4, 843, 120, BarterGuy12, 50);
-    giq_option(4, 843, 121, BarterGuy09, 50);
+    Reply(118);
+    giq_option(-3, NAME, 119, BarterGuy08, NEUTRAL_REACTION);
+    giq_option(4, NAME, 120, BarterGuy12, NEUTRAL_REACTION);
+    giq_option(4, NAME, 121, BarterGuy09, NEUTRAL_REACTION);
 end
 
 procedure BarterGuy08
 begin
-    gsay_reply(843, 122);
-    giq_option(-3, 843, 123, BarterGuyEnd, 50);
+    Reply(122);
+    giq_option(-3, NAME, 123, BarterGuyEnd, NEUTRAL_REACTION);
 end
 
 procedure BarterGuy09
 begin
-    gsay_message(843, 124, 50);
+    gsay_message(NAME, 124, NEUTRAL_REACTION);
 end
 
 procedure BarterGuy10
 begin
-    gsay_reply(843, 125);
-    giq_option(-3, 843, 126, BarterGuy08, 50);
-    giq_option(4, 843, 127, BarterGuy13, 50);
-    giq_option(5, 843, 128, BarterGuy11, 50);
+    Reply(125);
+    giq_option(-3, NAME, 126, BarterGuy08, NEUTRAL_REACTION);
+    giq_option(4, NAME, 127, BarterGuy13, NEUTRAL_REACTION);
+    giq_option(5, NAME, 128, BarterGuy11, NEUTRAL_REACTION);
 end
 
 procedure BarterGuy11
 begin
-    gsay_reply(843, 129);
-    giq_option(4, 843, 130, BarterGuyEnd, 50);
-    giq_option(4, 843, 131, BarterGuy01, 50);
-    giq_option(4, 843, 132, BarterGuy13, 50);
+    Reply(129);
+    giq_option(4, NAME, 130, BarterGuyEnd, NEUTRAL_REACTION);
+    giq_option(4, NAME, 131, BarterGuy01, NEUTRAL_REACTION);
+    giq_option(4, NAME, 132, BarterGuy13, NEUTRAL_REACTION);
 end
 
 procedure BarterGuy12
 begin
-    gsay_reply(843, 133);
-    giq_option(4, 843, 134, BarterGuyEnd, 50);
+    Reply(133);
+    giq_option(4, NAME, 134, BarterGuyEnd, NEUTRAL_REACTION);
 end
 
 procedure BarterGuy13
 begin
-    gsay_reply(843, 135);
-    giq_option(4, 843, 136, BarterGuy14, 50);
+    Reply(135);
+    giq_option(4, NAME, 136, BarterGuy14, NEUTRAL_REACTION);
 end
 
 procedure BarterGuy14
 begin
     gdialog_mod_barter(-10);
-    gsay_message(843, 137, 50);
+    gsay_message(NAME, 137, NEUTRAL_REACTION);
 end
 
 procedure BarterGuyEnd
@@ -313,31 +313,31 @@ begin
         set_local_var(0, 50);
         set_local_var(1, 2);
         set_local_var(2, 1);
-        set_local_var(0, local_var(0) + (5 * get_critter_stat(dude_obj, 3)) - 25);
-        set_local_var(0, local_var(0) + (10 * has_trait(0, dude_obj, 10)));
-        if (has_trait(0, dude_obj, 39)) then begin
-            if (global_var(155) > 0) then begin
-                set_local_var(0, local_var(0) + global_var(155));
+        set_local_var(0, local_var(0) + (5 * dude_charisma) - 25);
+        set_local_var(0, local_var(0) + (10 * has_trait(TRAIT_PERK, dude_obj, PERK_presence)));
+        if (has_trait(TRAIT_PERK, dude_obj, PERK_cult_of_personality)) then begin
+            if (global_var(GVAR_PLAYER_REPUATION) > 0) then begin
+                set_local_var(0, local_var(0) + global_var(GVAR_PLAYER_REPUATION));
             end
             else begin
-                set_local_var(0, local_var(0) - global_var(155));
+                set_local_var(0, local_var(0) - global_var(GVAR_PLAYER_REPUATION));
             end
         end
         else begin
             if (local_var(3) == 1) then begin
-                set_local_var(0, local_var(0) - global_var(155));
+                set_local_var(0, local_var(0) - global_var(GVAR_PLAYER_REPUATION));
             end
             else begin
-                set_local_var(0, local_var(0) + global_var(155));
+                set_local_var(0, local_var(0) + global_var(GVAR_PLAYER_REPUATION));
             end
         end
-        if (global_var(158) > 2) then begin
+        if (global_var(GVAR_CHILDKILLER_REPUATION) > 2) then begin
             set_local_var(0, local_var(0) - 30);
         end
-        if (((global_var(160) + global_var(159)) >= 25) and ((global_var(160) > (3 * global_var(159))) or (global_var(157) == 1))) then begin
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(160) > (3 * global_var(GVAR_GOOD_MONSTER))) or (global_var(GVAR_CHAMPION_REPUTATION) == 1))) then begin
             set_local_var(0, local_var(0) + 20);
         end
-        if (((global_var(160) + global_var(159)) >= 25) and ((global_var(159) > (2 * global_var(160))) or (global_var(156) == 1))) then begin
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
             set_local_var(0, local_var(0) - 20);
         end
         call ReactToLevel();
@@ -430,6 +430,5 @@ end
 
 procedure Goodbyes
 begin
-    exit_line := message_str(634, random(100, 105));
+    exit_line := message_str(SCRIPT_MODREACT, random(100, 105));
 end
-
