diff --git a/Games/Fallout1.Steam/Original/CAVE2V13.ssl b/Games/Fallout1.Steam/ReDefine/CAVE2V13.ssl
index 11aa7d2..d9ed032 100644
--- a/Games/Fallout1.Steam/Original/CAVE2V13.ssl
+++ b/Games/Fallout1.Steam/ReDefine/CAVE2V13.ssl
@@ -4,7 +4,7 @@ procedure spatial_p_proc;
 
 procedure start
 begin
-    if (script_action == 2) then begin
+    if (script_action == spatial_proc) then begin
         call spatial_p_proc();
     end
 end
@@ -15,4 +15,3 @@ begin
         load_map(6, 12);
     end
 end
-
