diff --git a/Games/Fallout1.Steam/Original/CHEATER.ssl b/Games/Fallout1.Steam/ReDefine/CHEATER.ssl
index 809ad92..ab80298 100644
--- a/Games/Fallout1.Steam/Original/CHEATER.ssl
+++ b/Games/Fallout1.Steam/ReDefine/CHEATER.ssl
@@ -56,26 +56,26 @@ procedure start
 begin
     if (GVar1) then begin
         GVar1 := 0;
-        critter_add_trait(self_obj, 1, 6, 84);
-        critter_add_trait(self_obj, 1, 5, 0);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 84);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_PLAYER);
     end
-    if (script_action == 21) then begin
+    if (script_action == look_at_proc) then begin
         call look_at_p_proc();
     end
     else begin
-        if (script_action == 4) then begin
+        if (script_action == pickup_proc) then begin
             call pickup_p_proc();
         end
         else begin
-            if (script_action == 11) then begin
+            if (script_action == talk_proc) then begin
                 call talk_p_proc();
             end
             else begin
-                if (script_action == 12) then begin
+                if (script_action == critter_proc) then begin
                     call critter_p_proc();
                 end
                 else begin
-                    if (script_action == 18) then begin
+                    if (script_action == destroy_proc) then begin
                         call destroy_p_proc();
                     end
                 end
@@ -93,7 +93,7 @@ procedure critter_p_proc
 begin
     if (GVar0) then begin
         GVar0 := 0;
-        attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+        attack(dude_obj);
     end
 end
 
@@ -117,9 +117,9 @@ end
 procedure destroy_p_proc
 begin
     if (source_obj == dude_obj) then begin
-        set_global_var(159, global_var(159) + 1);
-        if ((global_var(159) % 7) == 0) then begin
-            set_global_var(155, global_var(155) - 1);
+        set_global_var(GVAR_GOOD_MONSTER, global_var(GVAR_GOOD_MONSTER) + 1);
+        if ((global_var(GVAR_GOOD_MONSTER) % 7) == 0) then begin
+            set_global_var(GVAR_PLAYER_REPUATION, global_var(GVAR_PLAYER_REPUATION) - 1);
         end
     end
 end
@@ -132,117 +132,117 @@ end
 
 procedure Cheater00a
 begin
-    gsay_reply(621, "How may I Cheat you today?");
-    giq_option(1, 621, "Experience Points.", Cheater00, 50);
-    giq_option(1, 621, "Damage.", Cheater06, 50);
-    giq_option(1, 621, "Poison", Cheater07, 50);
-    giq_option(1, 621, "Radiation", Cheater08, 50);
-    giq_option(1, 621, "Nothing.", CheaterEnd, 50);
+    Reply("How may I Cheat you today?");
+    giq_option(1, NAME, "Experience Points.", Cheater00, NEUTRAL_REACTION);
+    giq_option(1, NAME, "Damage.", Cheater06, NEUTRAL_REACTION);
+    giq_option(1, NAME, "Poison", Cheater07, NEUTRAL_REACTION);
+    giq_option(1, NAME, "Radiation", Cheater08, NEUTRAL_REACTION);
+    giq_option(1, NAME, "Nothing.", CheaterEnd, NEUTRAL_REACTION);
 end
 
 procedure Cheater00
 begin
-    gsay_reply(621, "How much Exp Points do you want?");
-    giq_option(1, 621, "500", Cheater01, 50);
-    giq_option(1, 621, "1000", Cheater02, 50);
-    giq_option(1, 621, "5000", Cheater03, 50);
-    giq_option(1, 621, "10000", Cheater04, 50);
-    giq_option(1, 621, "20000", Cheater05, 50);
-    giq_option(1, 621, "Nevermind", CheaterEnd, 50);
+    Reply("How much Exp Points do you want?");
+    giq_option(1, NAME, "500", Cheater01, NEUTRAL_REACTION);
+    giq_option(1, NAME, "1000", Cheater02, NEUTRAL_REACTION);
+    giq_option(1, NAME, "5000", Cheater03, NEUTRAL_REACTION);
+    giq_option(1, NAME, "10000", Cheater04, NEUTRAL_REACTION);
+    giq_option(1, NAME, "20000", Cheater05, NEUTRAL_REACTION);
+    giq_option(1, NAME, "Nevermind", CheaterEnd, NEUTRAL_REACTION);
 end
 
 procedure Cheater01
 begin
     give_exp_points(500);
-    gsay_message(621, "Ok, you have them.", 50);
+    gsay_message(NAME, "Ok, you have them.", NEUTRAL_REACTION);
     call Cheater00();
 end
 
 procedure Cheater02
 begin
     give_exp_points(1000);
-    gsay_message(621, "Ok, you have them.", 50);
+    gsay_message(NAME, "Ok, you have them.", NEUTRAL_REACTION);
     call Cheater00();
 end
 
 procedure Cheater03
 begin
     give_exp_points(5000);
-    gsay_message(621, "Ok, you have them.", 50);
+    gsay_message(NAME, "Ok, you have them.", NEUTRAL_REACTION);
     call Cheater00();
 end
 
 procedure Cheater04
 begin
     give_exp_points(10000);
-    gsay_message(621, "Ok, you have them.", 50);
+    gsay_message(NAME, "Ok, you have them.", NEUTRAL_REACTION);
     call Cheater00();
 end
 
 procedure Cheater05
 begin
     give_exp_points(20000);
-    gsay_message(621, "Ok, you have them.", 50);
+    gsay_message(NAME, "Ok, you have them.", NEUTRAL_REACTION);
     call Cheater00();
 end
 
 procedure Cheater06
 begin
-    gsay_reply(621, "How might I abuse you today?");
-    giq_option(1, 621, "Cripple Left Leg", Cheater06a, 50);
-    giq_option(1, 621, "Cripple Right Leg", Cheater06b, 50);
-    giq_option(1, 621, "Cripple Left Arm", Cheater06c, 50);
-    giq_option(1, 621, "Cripple Right Arm", Cheater06d, 50);
-    giq_option(1, 621, "Blind Me ", Cheater06e, 50);
-    giq_option(1, 621, "Don't hurt me.", Cheater00a, 50);
+    Reply("How might I abuse you today?");
+    giq_option(1, NAME, "Cripple Left Leg", Cheater06a, NEUTRAL_REACTION);
+    giq_option(1, NAME, "Cripple Right Leg", Cheater06b, NEUTRAL_REACTION);
+    giq_option(1, NAME, "Cripple Left Arm", Cheater06c, NEUTRAL_REACTION);
+    giq_option(1, NAME, "Cripple Right Arm", Cheater06d, NEUTRAL_REACTION);
+    giq_option(1, NAME, "Blind Me ", Cheater06e, NEUTRAL_REACTION);
+    giq_option(1, NAME, "Don't hurt me.", Cheater00a, NEUTRAL_REACTION);
 end
 
 procedure Cheater06a
 begin
     critter_injure(dude_obj, 4);
-    gsay_message(621, "[Grinning Visciously] I was happy to Help you", 50);
+    gsay_message(NAME, "[Grinning Visciously] I was happy to Help you", NEUTRAL_REACTION);
     call Cheater00a();
 end
 
 procedure Cheater06b
 begin
     critter_injure(dude_obj, 8);
-    gsay_message(621, "[Grinning Visciously] I was happy to Help you", 50);
+    gsay_message(NAME, "[Grinning Visciously] I was happy to Help you", NEUTRAL_REACTION);
     call Cheater00a();
 end
 
 procedure Cheater06c
 begin
     critter_injure(dude_obj, 4);
-    gsay_message(621, "[Grinning Visciously] I was happy to Help you", 50);
+    gsay_message(NAME, "[Grinning Visciously] I was happy to Help you", NEUTRAL_REACTION);
     call Cheater00a();
 end
 
 procedure Cheater06d
 begin
     critter_injure(dude_obj, 16);
-    gsay_message(621, "[Grinning Visciously] I was happy to Help you", 50);
+    gsay_message(NAME, "[Grinning Visciously] I was happy to Help you", NEUTRAL_REACTION);
     call Cheater00a();
 end
 
 procedure Cheater06e
 begin
     critter_injure(dude_obj, 32);
-    gsay_message(621, "[Grinning Visciously] I was happy to Help you", 50);
+    gsay_message(NAME, "[Grinning Visciously] I was happy to Help you", NEUTRAL_REACTION);
     call Cheater00a();
 end
 
 procedure Cheater07
 begin
     poison(dude_obj, 10);
-    gsay_message(621, "[Hands you a Burrito Supreme] Enjoy your Grade D Beef.", 50);
+    gsay_message(NAME, "[Hands you a Burrito Supreme] Enjoy your Grade D Beef.", NEUTRAL_REACTION);
     call Cheater00a();
 end
 
 procedure Cheater08
 begin
     radiation_inc(dude_obj, 50);
-    gsay_message(621, "Now you glow like a 50 watt light bulb.", 50);
+    gsay_message(NAME, "Now you glow like a 50 watt light bulb.", NEUTRAL_REACTION);
     call Cheater00a();
 end
 
@@ -256,31 +256,31 @@ begin
         set_local_var(0, 50);
         set_local_var(1, 2);
         set_local_var(2, 1);
-        set_local_var(0, local_var(0) + (5 * get_critter_stat(dude_obj, 3)) - 25);
-        set_local_var(0, local_var(0) + (10 * has_trait(0, dude_obj, 10)));
-        if (has_trait(0, dude_obj, 39)) then begin
-            if (global_var(155) > 0) then begin
-                set_local_var(0, local_var(0) + global_var(155));
+        set_local_var(0, local_var(0) + (5 * dude_charisma) - 25);
+        set_local_var(0, local_var(0) + (10 * has_trait(TRAIT_PERK, dude_obj, PERK_presence)));
+        if (has_trait(TRAIT_PERK, dude_obj, PERK_cult_of_personality)) then begin
+            if (global_var(GVAR_PLAYER_REPUATION) > 0) then begin
+                set_local_var(0, local_var(0) + global_var(GVAR_PLAYER_REPUATION));
             end
             else begin
-                set_local_var(0, local_var(0) - global_var(155));
+                set_local_var(0, local_var(0) - global_var(GVAR_PLAYER_REPUATION));
             end
         end
         else begin
             if (local_var(3) == 1) then begin
-                set_local_var(0, local_var(0) - global_var(155));
+                set_local_var(0, local_var(0) - global_var(GVAR_PLAYER_REPUATION));
             end
             else begin
-                set_local_var(0, local_var(0) + global_var(155));
+                set_local_var(0, local_var(0) + global_var(GVAR_PLAYER_REPUATION));
             end
         end
-        if (global_var(158) > 2) then begin
+        if (global_var(GVAR_CHILDKILLER_REPUATION) > 2) then begin
             set_local_var(0, local_var(0) - 30);
         end
-        if (((global_var(160) + global_var(159)) > 34) and ((global_var(160) > (3 * global_var(159))) or (global_var(157) == 1))) then begin
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) > 34) and ((global_var(160) > (3 * global_var(GVAR_GOOD_MONSTER))) or (global_var(GVAR_CHAMPION_REPUTATION) == 1))) then begin
             set_local_var(0, local_var(0) + 20);
         end
-        if (((global_var(160) + global_var(159)) > 34) and ((global_var(159) > (2 * global_var(160))) or (global_var(156) == 1))) then begin
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) > 34) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
             set_local_var(0, local_var(0) - 20);
         end
         call ReactToLevel();
@@ -373,6 +373,5 @@ end
 
 procedure Goodbyes
 begin
-    GVar2 := message_str(634, random(100, 105));
+    GVar2 := message_str(SCRIPT_MODREACT, random(100, 105));
 end
-
