diff --git a/Games/Fallout1.Steam/Original/CHILDR~1.ssl b/Games/Fallout1.Steam/ReDefine/CHILDR~1.ssl
index 8e62f3c..a1b0708 100644
--- a/Games/Fallout1.Steam/Original/CHILDR~1.ssl
+++ b/Games/Fallout1.Steam/ReDefine/CHILDR~1.ssl
@@ -18,18 +18,18 @@ procedure Invasion;
 
 procedure start
 begin
-    if (script_action == 15) then begin
+    if (script_action == map_enter_proc) then begin
         set_light_level(75);
-        Player_Elevation := elevation(dude_obj);
-        if (global_var(32) == 0) then begin
+        Player_Elevation := dude_elevation;
+        if (global_var(GVAR_LOAD_MAP_INDEX) == 0) then begin
             override_map_start(137, 103, 0, 3);
         end
         else begin
-            if (global_var(32) == 4) then begin
+            if (global_var(GVAR_LOAD_MAP_INDEX) == 4) then begin
                 override_map_start(98, 60, 0, 3);
             end
             else begin
-                if (global_var(32) == 1) then begin
+                if (global_var(GVAR_LOAD_MAP_INDEX) == 1) then begin
                     override_map_start(94, 98, 1, 3);
                 end
                 else begin
@@ -37,35 +37,35 @@ begin
                 end
             end
         end
-        if (global_var(55) != 0) then begin
-            display_msg(message_str(446, 100) + (240 - ((game_time / 10) - global_var(55))) + message_str(446, 101));
-            if (((game_time / 10) - global_var(55)) > 240) then begin
-                display_msg(message_str(446, 102));
+        if (global_var(GVAR_COUNTDOWN_TO_DESTRUCTION) != 0) then begin
+            display_msg(message_str(SCRIPT_MASTER1, 100) + (240 - ((game_time / 10) - global_var(GVAR_COUNTDOWN_TO_DESTRUCTION))) + message_str(SCRIPT_MASTER1, 101));
+            if (((game_time / 10) - global_var(GVAR_COUNTDOWN_TO_DESTRUCTION)) > 240) then begin
+                display_msg(message_str(SCRIPT_MASTER1, 102));
                 play_gmovie(6);
-                metarule(13, 0);
+                signal_end_game;
             end
         end
         call add_party();
     end
-    if (script_action == 23) then begin
+    if (script_action == map_update_proc) then begin
         set_light_level(75);
-        if (global_var(55) != 0) then begin
-            display_msg(message_str(446, 104) + (240 - ((game_time / 10) - global_var(55))) + message_str(446, 105));
-            if (((game_time / 10) - global_var(55)) > 240) then begin
-                display_msg(message_str(446, 106));
+        if (global_var(GVAR_COUNTDOWN_TO_DESTRUCTION) != 0) then begin
+            display_msg(message_str(SCRIPT_MASTER1, 104) + (240 - ((game_time / 10) - global_var(GVAR_COUNTDOWN_TO_DESTRUCTION))) + message_str(SCRIPT_MASTER1, 105));
+            if (((game_time / 10) - global_var(GVAR_COUNTDOWN_TO_DESTRUCTION)) > 240) then begin
+                display_msg(message_str(SCRIPT_MASTER1, 106));
                 play_gmovie(6);
-                metarule(13, 0);
+                signal_end_game;
             end
         end
         call update_party();
     end
-    if (script_action == 16) then begin
-        if (global_var(55) != 0) then begin
-            display_msg(message_str(446, 108) + (240 - ((game_time / 10) - global_var(55))) + message_str(446, 109));
-            if (((game_time / 10) - global_var(55)) > 240) then begin
-                display_msg(message_str(446, 110));
+    if (script_action == map_exit_proc) then begin
+        if (global_var(GVAR_COUNTDOWN_TO_DESTRUCTION) != 0) then begin
+            display_msg(message_str(SCRIPT_MASTER1, 108) + (240 - ((game_time / 10) - global_var(GVAR_COUNTDOWN_TO_DESTRUCTION))) + message_str(SCRIPT_MASTER1, 109));
+            if (((game_time / 10) - global_var(GVAR_COUNTDOWN_TO_DESTRUCTION)) > 240) then begin
+                display_msg(message_str(SCRIPT_MASTER1, 110));
                 play_gmovie(6);
-                metarule(13, 0);
+                signal_end_game;
             end
         end
         call remove_party();
@@ -102,96 +102,95 @@ end
 procedure add_party
 begin
     variable LVar0 := 0;
-    party_elevation := elevation(dude_obj);
-    if (global_var(118) == 2) then begin
+    party_elevation := dude_elevation;
+    if (global_var(GVAR_IAN_STATUS) == 2) then begin
         if (not(Ian_ptr)) then begin
-            Ian_ptr := create_object_sid(16777233, 0, 0, 235);
-            critter_attempt_placement(Ian_ptr, tile_num_in_direction(tile_num(dude_obj), 1, 2), elevation(dude_obj));
+            Ian_ptr := create_object_sid(16777233, 0, 0, SCRIPT_IAN);
+            critter_attempt_placement(Ian_ptr, tile_num_in_direction(dude_tile, 1, 2), dude_elevation);
         end
-        critter_add_trait(Ian_ptr, 1, 6, 0);
-        LVar0 := create_object_sid(74, 0, 0, -1);
+        critter_add_trait(Ian_ptr, TRAIT_OBJECT, OBJECT_TEAM_NUM, 0);
+        LVar0 := create_object(74, 0, 0);
         add_obj_to_inven(Ian_ptr, LVar0);
         wield_obj_critter(Ian_ptr, LVar0);
-        LVar0 := create_object_sid(8, 0, 0, -1);
+        LVar0 := create_object(8, 0, 0);
         add_obj_to_inven(Ian_ptr, LVar0);
     end
-    if (global_var(5)) then begin
+    if (global_var(GVAR_DOG_EMPATHY)) then begin
         if (not(Dog_ptr)) then begin
-            Dog_ptr := create_object_sid(16777252, 0, 0, 229);
-            critter_attempt_placement(Dog_ptr, tile_num_in_direction(tile_num(dude_obj), 2, 2), elevation(dude_obj));
+            Dog_ptr := create_object_sid(16777252, 0, 0, SCRIPT_DOG);
+            critter_attempt_placement(Dog_ptr, tile_num_in_direction(dude_tile, 2, 2), dude_elevation);
         end
-        critter_add_trait(Dog_ptr, 1, 6, 0);
+        critter_add_trait(Dog_ptr, TRAIT_OBJECT, OBJECT_TEAM_NUM, 0);
     end
-    if (global_var(121) == 2) then begin
+    if (global_var(GVAR_HERNANDEZ_STATUS) == 2) then begin
         if (not(Tycho_ptr)) then begin
-            Tycho_ptr := create_object_sid(16777218, 0, 0, 389);
-            critter_attempt_placement(Tycho_ptr, tile_num_in_direction(tile_num(dude_obj), 3, 2), elevation(dude_obj));
+            Tycho_ptr := create_object_sid(16777218, 0, 0, SCRIPT_TYCHO);
+            critter_attempt_placement(Tycho_ptr, tile_num_in_direction(dude_tile, 3, 2), dude_elevation);
         end
-        critter_add_trait(Tycho_ptr, 1, 6, 0);
-        LVar0 := create_object_sid(1, 0, 0, -1);
+        critter_add_trait(Tycho_ptr, TRAIT_OBJECT, OBJECT_TEAM_NUM, 0);
+        LVar0 := create_object(1, 0, 0);
         add_obj_to_inven(Tycho_ptr, LVar0);
         wield_obj_critter(Tycho_ptr, LVar0);
-        LVar0 := create_object_sid(94, 0, 0, -1);
+        LVar0 := create_object(94, 0, 0);
         add_obj_to_inven(Tycho_ptr, LVar0);
     end
 end
 
 procedure update_party
 begin
-    if (elevation(dude_obj) != party_elevation) then begin
-        party_elevation := elevation(dude_obj);
-        if (global_var(118) == 2) then begin
-            move_to(Ian_ptr, tile_num_in_direction(tile_num(dude_obj), 1, 2), elevation(dude_obj));
+    if (dude_elevation != party_elevation) then begin
+        party_elevation := dude_elevation;
+        if (global_var(GVAR_IAN_STATUS) == 2) then begin
+            move_to(Ian_ptr, tile_num_in_direction(dude_tile, 1, 2), dude_elevation);
         end
-        if (global_var(5)) then begin
-            move_to(Dog_ptr, tile_num_in_direction(tile_num(dude_obj), 2, 1), elevation(dude_obj));
+        if (global_var(GVAR_DOG_EMPATHY)) then begin
+            move_to(Dog_ptr, tile_num_in_direction(dude_tile, 2, 1), dude_elevation);
         end
-        if (global_var(121) == 2) then begin
-            move_to(Tycho_ptr, tile_num_in_direction(tile_num(dude_obj), 3, 2), elevation(dude_obj));
+        if (global_var(GVAR_HERNANDEZ_STATUS) == 2) then begin
+            move_to(Tycho_ptr, tile_num_in_direction(dude_tile, 3, 2), dude_elevation);
         end
     end
 end
 
 procedure remove_party
 begin
-    if (global_var(118) == 2) then begin
+    if (global_var(GVAR_IAN_STATUS) == 2) then begin
         destroy_object(Ian_ptr);
-        set_global_var(118, 2);
+        set_global_var(GVAR_IAN_STATUS, 2);
     end
-    if (global_var(5)) then begin
+    if (global_var(GVAR_DOG_EMPATHY)) then begin
         destroy_object(Dog_ptr);
-        set_global_var(5, 1);
+        set_global_var(GVAR_DOG_EMPATHY, 1);
     end
-    if (global_var(121) == 2) then begin
+    if (global_var(GVAR_HERNANDEZ_STATUS) == 2) then begin
         destroy_object(Tycho_ptr);
-        set_global_var(121, 2);
+        set_global_var(GVAR_HERNANDEZ_STATUS, 2);
     end
 end
 
 procedure Invasion
 begin
-    if (not(global_var(18) == 2)) then begin
-        if (global_var(149) > (game_time / (10 * 60 * 60 * 24))) then begin
-            set_global_var(13, 1);
+    if (not(global_var(GVAR_MASTER_BLOWN) == 2)) then begin
+        if (global_var(GVAR_NECROPOLIS_INVADED_DATE) > (game_time / (10 * 60 * 60 * 24))) then begin
+            set_global_var(GVAR_NECROPOLIS_INVADED, 1);
         end
-        if (global_var(150) > (game_time / (10 * 60 * 60 * 24))) then begin
-            set_global_var(14, 1);
+        if (global_var(GVAR_THE_HUB_INVADED_DATE) > (game_time / (10 * 60 * 60 * 24))) then begin
+            set_global_var(GVAR_HUB_INVADED, 1);
         end
-        if (global_var(151) > (game_time / (10 * 60 * 60 * 24))) then begin
-            set_global_var(16, 1);
+        if (global_var(GVAR_BROTHERHOOD_INVADED_DATE) > (game_time / (10 * 60 * 60 * 24))) then begin
+            set_global_var(GVAR_BROTHERHOOD_INVADED, 1);
         end
-        if (global_var(152) > (game_time / (10 * 60 * 60 * 24))) then begin
-            set_global_var(15, 1);
+        if (global_var(GVAR_JUNKTOWN_INVADED_DATE) > (game_time / (10 * 60 * 60 * 24))) then begin
+            set_global_var(GVAR_JUNKTOWN_INVADED, 1);
         end
-        if (global_var(153) > (game_time / (10 * 60 * 60 * 24))) then begin
-            set_global_var(12, 1);
+        if (global_var(GVAR_SHADY_SANDS_INVADED_DATE) > (game_time / (10 * 60 * 60 * 24))) then begin
+            set_global_var(GVAR_SHADY_SANDS_INVADED, 1);
         end
-        if (global_var(154) > (game_time / (10 * 60 * 60 * 24))) then begin
-            set_global_var(11, 1);
+        if (global_var(GVAR_VAULT_13_INVADED_DATE) > (game_time / (10 * 60 * 60 * 24))) then begin
+            set_global_var(GVAR_VAULT_INVADED, 1);
         end
-        if (global_var(148) > (game_time / (10 * 60 * 60 * 24))) then begin
-            set_global_var(7, 1);
+        if (global_var(GVAR_FOLLOWERS_INVADED_DATE) > (game_time / (10 * 60 * 60 * 24))) then begin
+            set_global_var(GVAR_FOLLOWERS_INVADED, 1);
         end
     end
 end
-
