diff --git a/Games/Fallout1.Steam/Original/CHUCK.ssl b/Games/Fallout1.Steam/ReDefine/CHUCK.ssl
index 7a15e5d..16fb199 100644
--- a/Games/Fallout1.Steam/Original/CHUCK.ssl
+++ b/Games/Fallout1.Steam/ReDefine/CHUCK.ssl
@@ -56,27 +56,27 @@ begin
     variable LVar0 := 0;
     if (not(initialized)) then begin
         if (obj_is_carrying_obj_pid(self_obj, 41) == 0) then begin
-            LVar0 := create_object_sid(41, 0, 0, -1);
+            LVar0 := create_object(41, 0, 0);
             add_mult_objs_to_inven(self_obj, LVar0, random(0, 20));
-            if ((global_var(613) == 9103) or (global_var(613) == 9102)) then begin
-                critter_add_trait(self_obj, 1, 6, 0);
+            if ((global_var(GVAR_BLADES) == 9103) or (global_var(GVAR_BLADES) == 9102)) then begin
+                critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 0);
             end
             else begin
-                critter_add_trait(self_obj, 1, 6, 49);
+                critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 49);
             end
         end
         initialized := 1;
     end
     else begin
-        if (script_action == 18) then begin
+        if (script_action == destroy_proc) then begin
             call destroy_p_proc();
         end
         else begin
-            if (script_action == 21) then begin
+            if (script_action == look_at_proc) then begin
                 call look_at_p_proc();
             end
             else begin
-                if (script_action == 11) then begin
+                if (script_action == talk_proc) then begin
                     call talk_p_proc();
                 end
             end
@@ -87,30 +87,30 @@ end
 procedure damage_p_proc
 begin
     if (source_obj == dude_obj) then begin
-        set_global_var(251, 1);
-        critter_add_trait(self_obj, 1, 6, 49);
+        set_global_var(GVAR_ENEMY_ADYTUM, 1);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 49);
     end
 end
 
 procedure destroy_p_proc
 begin
     if (source_obj == dude_obj) then begin
-        set_global_var(251, 1);
+        set_global_var(GVAR_ENEMY_ADYTUM, 1);
         if (source_obj == dude_obj) then begin
-            if (((global_var(160) + global_var(159)) >= 25) and ((global_var(159) > (2 * global_var(160))) or (global_var(156) == 1))) then begin
-                set_global_var(156, 1);
-                set_global_var(157, 0);
+            if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
+                set_global_var(GVAR_BERSERKER_REPUTATION, 1);
+                set_global_var(GVAR_CHAMPION_REPUTATION, 0);
             end
-            if (((global_var(160) + global_var(159)) >= 25) and ((global_var(160) > (3 * global_var(159))) or (global_var(157) == 1))) then begin
-                set_global_var(157, 1);
-                set_global_var(156, 0);
+            if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(160) > (3 * global_var(GVAR_GOOD_MONSTER))) or (global_var(GVAR_CHAMPION_REPUTATION) == 1))) then begin
+                set_global_var(GVAR_CHAMPION_REPUTATION, 1);
+                set_global_var(GVAR_BERSERKER_REPUTATION, 0);
             end
-            set_global_var(159, global_var(159) + 1);
-            if ((global_var(159) % 2) == 0) then begin
-                set_global_var(155, global_var(155) - 1);
+            set_global_var(GVAR_GOOD_MONSTER, global_var(GVAR_GOOD_MONSTER) + 1);
+            if ((global_var(GVAR_GOOD_MONSTER) % 2) == 0) then begin
+                set_global_var(GVAR_PLAYER_REPUATION, global_var(GVAR_PLAYER_REPUATION) - 1);
             end
         end
-        critter_add_trait(self_obj, 1, 6, 49);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 49);
     end
 end
 
@@ -118,10 +118,10 @@ procedure look_at_p_proc
 begin
     script_overrides;
     if (local_var(4)) then begin
-        display_msg(message_str(664, 101));
+        display_mstr(101);
     end
     else begin
-        display_msg(message_str(664, 100));
+        display_mstr(100);
     end
 end
 
@@ -143,235 +143,235 @@ end
 procedure critter_p_proc
 begin
     if (obj_can_see_obj(self_obj, dude_obj)) then begin
-        if (global_var(251) == 1) then begin
-            attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+        if (global_var(GVAR_ENEMY_ADYTUM) == 1) then begin
+            attack(dude_obj);
         end
     end
 end
 
 procedure pickup_p_proc
 begin
-    set_global_var(251, 1);
+    set_global_var(GVAR_ENEMY_ADYTUM, 1);
 end
 
 procedure Chuck01
 begin
     set_local_var(4, 1);
-    gsay_reply(664, 102);
-    giq_option(-3, 664, 103, Chuck02, 50);
-    giq_option(4, 664, 104, Chuck03, 50);
-    giq_option(4, 664, 105, Chuck04, 50);
-    giq_option(6, 664, 106, Chuck05, 50);
+    Reply(102);
+    giq_option(-3, NAME, 103, Chuck02, NEUTRAL_REACTION);
+    giq_option(4, NAME, 104, Chuck03, NEUTRAL_REACTION);
+    giq_option(4, NAME, 105, Chuck04, NEUTRAL_REACTION);
+    giq_option(6, NAME, 106, Chuck05, NEUTRAL_REACTION);
 end
 
 procedure Chuck02
 begin
-    gsay_message(664, 107, 50);
+    gsay_message(NAME, 107, NEUTRAL_REACTION);
 end
 
 procedure Chuck03
 begin
-    gsay_reply(664, 108);
-    giq_option(4, 664, 109, Chuck09, 50);
-    giq_option(4, 664, 110, Chuck14, 50);
-    giq_option(6, 664, 111, Chuck15, 49);
+    Reply(108);
+    giq_option(4, NAME, 109, Chuck09, NEUTRAL_REACTION);
+    giq_option(4, NAME, 110, Chuck14, NEUTRAL_REACTION);
+    giq_option(6, NAME, 111, Chuck15, GOOD_REACTION);
 end
 
 procedure Chuck04
 begin
-    gsay_reply(664, 112);
-    giq_option(4, 664, 113, Chuck14, 50);
-    giq_option(4, 664, 114, Chuck18, 50);
-    giq_option(4, 664, 115, ChuckBye, 50);
+    Reply(112);
+    giq_option(4, NAME, 113, Chuck14, NEUTRAL_REACTION);
+    giq_option(4, NAME, 114, Chuck18, NEUTRAL_REACTION);
+    giq_option(4, NAME, 115, ChuckBye, NEUTRAL_REACTION);
 end
 
 procedure Chuck05
 begin
-    gsay_reply(664, 117);
-    giq_option(4, 664, 118, ChuckBye, 50);
-    giq_option(4, 664, 119, ChuckCards, 50);
+    Reply(117);
+    giq_option(4, NAME, 118, ChuckBye, NEUTRAL_REACTION);
+    giq_option(4, NAME, 119, ChuckCards, NEUTRAL_REACTION);
 end
 
 procedure Chuck06
 begin
     call DownReact();
-    gsay_reply(664, 128);
-    giq_option(4, 664, 129, Chuck07, 51);
-    giq_option(4, 664, 130, Chuck08, 50);
+    Reply(128);
+    giq_option(4, NAME, 129, Chuck07, BAD_REACTION);
+    giq_option(4, NAME, 130, Chuck08, NEUTRAL_REACTION);
 end
 
 procedure Chuck07
 begin
     call DownReact();
     set_local_var(5, 4);
-    gsay_message(664, 131, 51);
+    gsay_message(NAME, 131, BAD_REACTION);
 end
 
 procedure Chuck08
 begin
     call UpReact();
-    gsay_message(664, 132, 50);
+    gsay_message(NAME, 132, NEUTRAL_REACTION);
 end
 
 procedure Chuck09
 begin
-    gsay_reply(664, 133);
-    giq_option(4, 664, 134, ChuckBye, 50);
-    giq_option(4, 664, 135, Chuck10, 50);
-    giq_option(6, 664, 136, Chuck19, 50);
+    Reply(133);
+    giq_option(4, NAME, 134, ChuckBye, NEUTRAL_REACTION);
+    giq_option(4, NAME, 135, Chuck10, NEUTRAL_REACTION);
+    giq_option(6, NAME, 136, Chuck19, NEUTRAL_REACTION);
 end
 
 procedure Chuck10
 begin
-    gsay_reply(664, 137);
-    giq_option(4, 664, 138, ChuckBye, 50);
-    giq_option(4, 664, 139, Chuck11, 50);
+    Reply(137);
+    giq_option(4, NAME, 138, ChuckBye, NEUTRAL_REACTION);
+    giq_option(4, NAME, 139, Chuck11, NEUTRAL_REACTION);
 end
 
 procedure Chuck11
 begin
-    gsay_reply(664, 140);
-    giq_option(4, 664, 134, ChuckBye, 50);
-    giq_option(4, 664, 141, Chuck12, 51);
-    giq_option(4, 664, 142, Chuck13, 50);
+    Reply(140);
+    giq_option(4, NAME, 134, ChuckBye, NEUTRAL_REACTION);
+    giq_option(4, NAME, 141, Chuck12, BAD_REACTION);
+    giq_option(4, NAME, 142, Chuck13, NEUTRAL_REACTION);
 end
 
 procedure Chuck12
 begin
-    gsay_reply(664, 143);
-    giq_option(4, 664, 144, Chuck07, 51);
-    giq_option(5, 664, 145, UpReact, 50);
-    giq_option(4, 664, 146, ChuckBye, 50);
+    Reply(143);
+    giq_option(4, NAME, 144, Chuck07, BAD_REACTION);
+    giq_option(5, NAME, 145, UpReact, NEUTRAL_REACTION);
+    giq_option(4, NAME, 146, ChuckBye, NEUTRAL_REACTION);
 end
 
 procedure Chuck13
 begin
-    gsay_reply(664, 147);
-    giq_option(4, 664, 148, ChuckCards, 49);
-    giq_option(4, 664, 144, Chuck07, 51);
+    Reply(147);
+    giq_option(4, NAME, 148, ChuckCards, GOOD_REACTION);
+    giq_option(4, NAME, 144, Chuck07, BAD_REACTION);
 end
 
 procedure Chuck14
 begin
-    gsay_reply(664, 149);
-    giq_option(4, 664, 150, ChuckEnd, 50);
-    giq_option(4, 664, 151, Chuck10, 50);
+    Reply(149);
+    giq_option(4, NAME, 150, ChuckEnd, NEUTRAL_REACTION);
+    giq_option(4, NAME, 151, Chuck10, NEUTRAL_REACTION);
 end
 
 procedure Chuck15
 begin
-    gsay_reply(664, 152);
-    giq_option(4, 664, 153, ChuckBye, 50);
-    giq_option(4, 664, 150, ChuckEnd, 50);
-    giq_option(6, 664, 154, Chuck16, 50);
+    Reply(152);
+    giq_option(4, NAME, 153, ChuckBye, NEUTRAL_REACTION);
+    giq_option(4, NAME, 150, ChuckEnd, NEUTRAL_REACTION);
+    giq_option(6, NAME, 154, Chuck16, NEUTRAL_REACTION);
 end
 
 procedure Chuck16
 begin
-    gsay_reply(664, 155);
-    giq_option(4, 664, 156, ChuckBye, 50);
-    giq_option(6, 664, 157, Chuck17, 50);
+    Reply(155);
+    giq_option(4, NAME, 156, ChuckBye, NEUTRAL_REACTION);
+    giq_option(6, NAME, 157, Chuck17, NEUTRAL_REACTION);
 end
 
 procedure Chuck17
 begin
-    gsay_reply(664, 158);
-    giq_option(4, 664, 159, ChuckBye, 50);
-    giq_option(4, 664, 150, DownReact, 51);
+    Reply(158);
+    giq_option(4, NAME, 159, ChuckBye, NEUTRAL_REACTION);
+    giq_option(4, NAME, 150, DownReact, BAD_REACTION);
 end
 
 procedure Chuck18
 begin
     set_local_var(6, 1);
-    gsay_reply(664, 160);
-    giq_option(4, 664, 146, ChuckBye, 50);
-    giq_option(4, 664, 161, ChuckCards, 50);
+    Reply(160);
+    giq_option(4, NAME, 146, ChuckBye, NEUTRAL_REACTION);
+    giq_option(4, NAME, 161, ChuckCards, NEUTRAL_REACTION);
 end
 
 procedure Chuck19
 begin
-    gsay_reply(664, 162);
-    giq_option(4, 664, 146, ChuckBye, 50);
-    giq_option(6, 664, 163, Chuck20, 50);
-    giq_option(6, 664, 164, Chuck21, 50);
+    Reply(162);
+    giq_option(4, NAME, 146, ChuckBye, NEUTRAL_REACTION);
+    giq_option(6, NAME, 163, Chuck20, NEUTRAL_REACTION);
+    giq_option(6, NAME, 164, Chuck21, NEUTRAL_REACTION);
 end
 
 procedure Chuck20
 begin
-    gsay_reply(664, 165);
-    giq_option(4, 664, 146, ChuckBye, 50);
-    giq_option(6, 664, 164, Chuck21, 50);
+    Reply(165);
+    giq_option(4, NAME, 146, ChuckBye, NEUTRAL_REACTION);
+    giq_option(6, NAME, 164, Chuck21, NEUTRAL_REACTION);
 end
 
 procedure Chuck21
 begin
-    gsay_reply(664, 166);
-    giq_option(0, 664, 125, ChuckBye, 50);
+    Reply(166);
+    giq_option(0, NAME, 125, ChuckBye, NEUTRAL_REACTION);
 end
 
 procedure Chuck22
 begin
-    gsay_reply(664, random(167, 169));
-    giq_option(-3, 664, 103, Chuck02, 50);
-    giq_option(4, 664, 170, Chuck23, 50);
+    Reply(random(167, 169));
+    giq_option(-3, NAME, 103, Chuck02, NEUTRAL_REACTION);
+    giq_option(4, NAME, 170, Chuck23, NEUTRAL_REACTION);
     if (local_var(5) < 4) then begin
-        giq_option(4, 664, 171, ChuckCards, 50);
+        giq_option(4, NAME, 171, ChuckCards, NEUTRAL_REACTION);
     end
     if (local_var(6) == 0) then begin
-        giq_option(4, 664, 172, Chuck18, 50);
+        giq_option(4, NAME, 172, Chuck18, NEUTRAL_REACTION);
     end
 end
 
 procedure Chuck23
 begin
-    gsay_reply(664, 173);
-    giq_option(0, 664, 125, ChuckBye, 50);
+    Reply(173);
+    giq_option(0, NAME, 125, ChuckBye, NEUTRAL_REACTION);
 end
 
 procedure ChuckCards
 begin
-    gsay_reply(664, 120);
-    giq_option(0, 664, 125, ChuckCards2, 50);
+    Reply(120);
+    giq_option(0, NAME, 125, ChuckCards2, NEUTRAL_REACTION);
 end
 
 procedure ChuckCards2
 begin
     if (local_var(5) == 0) then begin
-        if (global_var(74) < 1) then begin
-            gsay_reply(664, 121);
+        if (global_var(GVAR_BROTHERHOOD_OF_STEEL) < 1) then begin
+            Reply(121);
         end
         else begin
             set_local_var(5, 1);
         end
     end
     if (local_var(5) == 1) then begin
-        if (global_var(78) < 1) then begin
-            gsay_reply(664, 122);
+        if (global_var(GVAR_THE_VATS) < 1) then begin
+            Reply(122);
         end
         else begin
             set_local_var(5, 2);
         end
     end
     if (local_var(5) == 2) then begin
-        if (global_var(79) < 1) then begin
-            gsay_reply(664, 123);
+        if (global_var(GVAR_MASTERS_LAIR) < 1) then begin
+            Reply(123);
         end
         else begin
             set_local_var(5, 3);
         end
     end
     if (local_var(5) >= 3) then begin
-        gsay_reply(664, 124);
-        set_critter_stat(dude_obj, 6, 1);
+        Reply(124);
+        set_critter_stat(dude_obj, STAT_lu, 1);
     end
     set_local_var(5, local_var(5) + 1);
-    giq_option(4, 664, 126, ChuckBye, 50);
-    giq_option(4, 664, 127, Chuck06, 50);
+    giq_option(4, NAME, 126, ChuckBye, NEUTRAL_REACTION);
+    giq_option(4, NAME, 127, Chuck06, NEUTRAL_REACTION);
 end
 
 procedure ChuckBye
 begin
-    gsay_message(664, 116, 50);
+    gsay_message(NAME, 116, NEUTRAL_REACTION);
 end
 
 procedure ChuckEnd
@@ -384,31 +384,31 @@ begin
         set_local_var(0, 50);
         set_local_var(1, 2);
         set_local_var(2, 1);
-        set_local_var(0, local_var(0) + (5 * get_critter_stat(dude_obj, 3)) - 25);
-        set_local_var(0, local_var(0) + (10 * has_trait(0, dude_obj, 10)));
-        if (has_trait(0, dude_obj, 39)) then begin
-            if (global_var(155) > 0) then begin
-                set_local_var(0, local_var(0) + global_var(155));
+        set_local_var(0, local_var(0) + (5 * dude_charisma) - 25);
+        set_local_var(0, local_var(0) + (10 * has_trait(TRAIT_PERK, dude_obj, PERK_presence)));
+        if (has_trait(TRAIT_PERK, dude_obj, PERK_cult_of_personality)) then begin
+            if (global_var(GVAR_PLAYER_REPUATION) > 0) then begin
+                set_local_var(0, local_var(0) + global_var(GVAR_PLAYER_REPUATION));
             end
             else begin
-                set_local_var(0, local_var(0) - global_var(155));
+                set_local_var(0, local_var(0) - global_var(GVAR_PLAYER_REPUATION));
             end
         end
         else begin
             if (local_var(3) == 1) then begin
-                set_local_var(0, local_var(0) - global_var(155));
+                set_local_var(0, local_var(0) - global_var(GVAR_PLAYER_REPUATION));
             end
             else begin
-                set_local_var(0, local_var(0) + global_var(155));
+                set_local_var(0, local_var(0) + global_var(GVAR_PLAYER_REPUATION));
             end
         end
-        if (global_var(158) > 2) then begin
+        if (global_var(GVAR_CHILDKILLER_REPUATION) > 2) then begin
             set_local_var(0, local_var(0) - 30);
         end
-        if (((global_var(160) + global_var(159)) >= 25) and ((global_var(160) > (3 * global_var(159))) or (global_var(157) == 1))) then begin
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(160) > (3 * global_var(GVAR_GOOD_MONSTER))) or (global_var(GVAR_CHAMPION_REPUTATION) == 1))) then begin
             set_local_var(0, local_var(0) + 20);
         end
-        if (((global_var(160) + global_var(159)) >= 25) and ((global_var(159) > (2 * global_var(160))) or (global_var(156) == 1))) then begin
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
             set_local_var(0, local_var(0) - 20);
         end
         call ReactToLevel();
@@ -501,6 +501,5 @@ end
 
 procedure Goodbyes
 begin
-    exit_line := message_str(634, random(100, 105));
+    exit_line := message_str(SCRIPT_MODREACT, random(100, 105));
 end
-
