diff --git a/Games/Fallout1.Steam/Original/CITIZEN.ssl b/Games/Fallout1.Steam/ReDefine/CITIZEN.ssl
index c9f2f35..992957b 100644
--- a/Games/Fallout1.Steam/Original/CITIZEN.ssl
+++ b/Games/Fallout1.Steam/ReDefine/CITIZEN.ssl
@@ -40,41 +40,41 @@ variable exit_line;
 procedure start
 begin
     if (not(initialized)) then begin
-        critter_add_trait(self_obj, 1, 6, 2);
-        critter_add_trait(self_obj, 1, 5, 6);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 2);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_PEASANT);
         add_timer_event(self_obj, game_ticks(random(3, 5)), 1);
         initialized := 1;
     end
     else begin
-        if (script_action == 13) then begin
+        if (script_action == combat_proc) then begin
             call combat_p_proc();
         end
         else begin
-            if (script_action == 12) then begin
+            if (script_action == critter_proc) then begin
                 call critter_p_proc();
             end
             else begin
-                if (script_action == 3) then begin
+                if (script_action == description_proc) then begin
                     call description_p_proc();
                 end
                 else begin
-                    if (script_action == 18) then begin
+                    if (script_action == destroy_proc) then begin
                         call destroy_p_proc();
                     end
                     else begin
-                        if (script_action == 21) then begin
+                        if (script_action == look_at_proc) then begin
                             call look_at_p_proc();
                         end
                         else begin
-                            if (script_action == 4) then begin
+                            if (script_action == pickup_proc) then begin
                                 call pickup_p_proc();
                             end
                             else begin
-                                if (script_action == 11) then begin
+                                if (script_action == talk_proc) then begin
                                     call talk_p_proc();
                                 end
                                 else begin
-                                    if (script_action == 22) then begin
+                                    if (script_action == timed_event_proc) then begin
                                         call timed_event_p_proc();
                                     end
                                 end
@@ -92,9 +92,9 @@ begin
     if (fixed_param == 4) then begin
         round_counter := round_counter + 1;
         if (round_counter > 3) then begin
-            if (global_var(246) == 0) then begin
-                set_global_var(246, 1);
-                set_global_var(155, global_var(155) - 5);
+            if (global_var(GVAR_ENEMY_SHADY_SANDS) == 0) then begin
+                set_global_var(GVAR_ENEMY_SHADY_SANDS, 1);
+                set_global_var(GVAR_PLAYER_REPUATION, global_var(GVAR_PLAYER_REPUATION) - 5);
             end
         end
     end
@@ -105,17 +105,17 @@ begin
     variable LVar0 := 0;
     if (hostile) then begin
         hostile := 1;
-        attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+        attack(dude_obj);
     end
     else begin
         if (local_var(4) and (tile_distance_objs(dude_obj, self_obj) < 8)) then begin
             call flee_dude();
         end
-        LVar0 := global_var(343);
-        if ((((game_time / 10) - LVar0) >= 10) and (tile_distance_objs(self_obj, dude_obj) <= 4) and (global_var(246) == 0)) then begin
-            float_msg(self_obj, message_str(115, random(110, 114)), 0);
+        LVar0 := global_var(GVAR_CURE_JARVIS_4);
+        if ((((game_time / 10) - LVar0) >= 10) and (tile_distance_objs(self_obj, dude_obj) <= 4) and (global_var(GVAR_ENEMY_SHADY_SANDS) == 0)) then begin
+            float_msg(self_obj, mstr(random(110, 114)), FLOAT_MSG_NORMAL);
             LVar0 := game_time / 10;
-            set_global_var(343, LVar0);
+            set_global_var(GVAR_CURE_JARVIS_4, LVar0);
         end
     end
 end
@@ -130,26 +130,26 @@ end
 procedure description_p_proc
 begin
     script_overrides;
-    display_msg(message_str(115, 100));
+    display_mstr(100);
 end
 
 procedure destroy_p_proc
 begin
     if (source_obj == dude_obj) then begin
-        set_global_var(246, 1);
+        set_global_var(GVAR_ENEMY_SHADY_SANDS, 1);
     end
     if (source_obj == dude_obj) then begin
-        if (((global_var(160) + global_var(159)) >= 25) and ((global_var(159) > (2 * global_var(160))) or (global_var(156) == 1))) then begin
-            set_global_var(156, 1);
-            set_global_var(157, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
+            set_global_var(GVAR_BERSERKER_REPUTATION, 1);
+            set_global_var(GVAR_CHAMPION_REPUTATION, 0);
         end
-        if (((global_var(160) + global_var(159)) >= 25) and ((global_var(160) > (3 * global_var(159))) or (global_var(157) == 1))) then begin
-            set_global_var(157, 1);
-            set_global_var(156, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(160) > (3 * global_var(GVAR_GOOD_MONSTER))) or (global_var(GVAR_CHAMPION_REPUTATION) == 1))) then begin
+            set_global_var(GVAR_CHAMPION_REPUTATION, 1);
+            set_global_var(GVAR_BERSERKER_REPUTATION, 0);
         end
-        set_global_var(159, global_var(159) + 1);
-        if ((global_var(159) % 2) == 0) then begin
-            set_global_var(155, global_var(155) - 1);
+        set_global_var(GVAR_GOOD_MONSTER, global_var(GVAR_GOOD_MONSTER) + 1);
+        if ((global_var(GVAR_GOOD_MONSTER) % 2) == 0) then begin
+            set_global_var(GVAR_PLAYER_REPUATION, global_var(GVAR_PLAYER_REPUATION) - 1);
         end
     end
 end
@@ -157,7 +157,7 @@ end
 procedure look_at_p_proc
 begin
     script_overrides;
-    display_msg(message_str(115, 100));
+    display_mstr(100);
 end
 
 procedure pickup_p_proc
@@ -167,8 +167,8 @@ end
 
 procedure talk_p_proc
 begin
-    if (local_var(4) or global_var(246)) then begin
-        float_msg(self_obj, message_str(669, random(100, 105)), 0);
+    if (local_var(4) or global_var(GVAR_ENEMY_SHADY_SANDS)) then begin
+        float_msg(self_obj, message_str(SCRIPT_ENEMY, random(100, 105)), FLOAT_MSG_NORMAL);
     end
     else begin
         start_gdialog(115, self_obj, 4, -1, -1);
@@ -190,19 +190,19 @@ end
 procedure Citizen01
 begin
     DisplayMessage := 105;
-    gsay_reply(115, 101);
-    giq_option(4, 115, 103, Citizen02, 50);
-    giq_option(4, 115, 102, CitizenEnd, 50);
-    giq_option(-3, 115, 104, CitizenEnd, 50);
+    Reply(101);
+    giq_option(4, NAME, 103, Citizen02, NEUTRAL_REACTION);
+    giq_option(4, NAME, 102, CitizenEnd, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 104, CitizenEnd, NEUTRAL_REACTION);
 end
 
 procedure Citizen02
 begin
-    gsay_reply(115, DisplayMessage);
-    giq_option(4, 115, 106, Citizen03, 50);
-    giq_option(4, 115, 107, Citizen04, 50);
+    Reply(DisplayMessage);
+    giq_option(4, NAME, 106, Citizen03, NEUTRAL_REACTION);
+    giq_option(4, NAME, 107, Citizen04, NEUTRAL_REACTION);
     call Goodbyes();
-    giq_option(1, 115, exit_line, CitizenEnd, 50);
+    giq_option(1, NAME, exit_line, CitizenEnd, NEUTRAL_REACTION);
 end
 
 procedure Citizen03
@@ -227,9 +227,9 @@ begin
     variable LVar1 := 0;
     variable LVar2 := 0;
     while (LVar1 < 5) do begin
-        if (tile_distance(tile_num(dude_obj), tile_num_in_direction(tile_num(self_obj), LVar1, 3)) > LVar2) then begin
+        if (tile_distance(dude_tile, tile_num_in_direction(tile_num(self_obj), LVar1, 3)) > LVar2) then begin
             LVar0 := tile_num_in_direction(tile_num(self_obj), LVar1, 3);
-            LVar2 := tile_distance(tile_num(dude_obj), LVar0);
+            LVar2 := tile_distance(dude_tile, LVar0);
         end
         LVar1 := LVar1 + 1;
     end
@@ -242,31 +242,31 @@ begin
         set_local_var(0, 50);
         set_local_var(1, 2);
         set_local_var(2, 1);
-        set_local_var(0, local_var(0) + (5 * get_critter_stat(dude_obj, 3)) - 25);
-        set_local_var(0, local_var(0) + (10 * has_trait(0, dude_obj, 10)));
-        if (has_trait(0, dude_obj, 39)) then begin
-            if (global_var(155) > 0) then begin
-                set_local_var(0, local_var(0) + global_var(155));
+        set_local_var(0, local_var(0) + (5 * dude_charisma) - 25);
+        set_local_var(0, local_var(0) + (10 * has_trait(TRAIT_PERK, dude_obj, PERK_presence)));
+        if (has_trait(TRAIT_PERK, dude_obj, PERK_cult_of_personality)) then begin
+            if (global_var(GVAR_PLAYER_REPUATION) > 0) then begin
+                set_local_var(0, local_var(0) + global_var(GVAR_PLAYER_REPUATION));
             end
             else begin
-                set_local_var(0, local_var(0) - global_var(155));
+                set_local_var(0, local_var(0) - global_var(GVAR_PLAYER_REPUATION));
             end
         end
         else begin
             if (local_var(3) == 1) then begin
-                set_local_var(0, local_var(0) - global_var(155));
+                set_local_var(0, local_var(0) - global_var(GVAR_PLAYER_REPUATION));
             end
             else begin
-                set_local_var(0, local_var(0) + global_var(155));
+                set_local_var(0, local_var(0) + global_var(GVAR_PLAYER_REPUATION));
             end
         end
-        if (global_var(158) > 2) then begin
+        if (global_var(GVAR_CHILDKILLER_REPUATION) > 2) then begin
             set_local_var(0, local_var(0) - 30);
         end
-        if (((global_var(160) + global_var(159)) >= 25) and ((global_var(160) > (3 * global_var(159))) or (global_var(157) == 1))) then begin
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(160) > (3 * global_var(GVAR_GOOD_MONSTER))) or (global_var(GVAR_CHAMPION_REPUTATION) == 1))) then begin
             set_local_var(0, local_var(0) + 20);
         end
-        if (((global_var(160) + global_var(159)) >= 25) and ((global_var(159) > (2 * global_var(160))) or (global_var(156) == 1))) then begin
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
             set_local_var(0, local_var(0) - 20);
         end
         call ReactToLevel();
@@ -359,6 +359,5 @@ end
 
 procedure Goodbyes
 begin
-    exit_line := message_str(634, random(100, 105));
+    exit_line := message_str(SCRIPT_MODREACT, random(100, 105));
 end
-
