diff --git a/Games/Fallout1.Steam/Original/COKE.ssl b/Games/Fallout1.Steam/ReDefine/COKE.ssl
index 934964a..88b4652 100644
--- a/Games/Fallout1.Steam/Original/COKE.ssl
+++ b/Games/Fallout1.Steam/ReDefine/COKE.ssl
@@ -8,7 +8,7 @@ variable caps;
 
 procedure start
 begin
-    if (script_action == 6) then begin
+    if (script_action == use_proc) then begin
         call use_p_proc();
     end
 end
@@ -26,16 +26,15 @@ end
 
 procedure Coke00
 begin
-    display_msg(message_str(22, 100));
+    display_mstr(100);
 end
 
 procedure Coke01
 begin
     set_local_var(0, 1);
     if (is_success(do_check(dude_obj, 6, 0))) then begin
-        display_msg(message_str(22, 101));
+        display_mstr(101);
         caps := random(1, 6) + random(1, 6) + random(1, 6);
         item_caps_adjust(dude_obj, caps);
     end
 end
-
