diff --git a/Games/Fallout1.Steam/Original/COLATRUK.ssl b/Games/Fallout1.Steam/ReDefine/COLATRUK.ssl
index 6575a5d..48ccff5 100644
--- a/Games/Fallout1.Steam/Original/COLATRUK.ssl
+++ b/Games/Fallout1.Steam/ReDefine/COLATRUK.ssl
@@ -7,16 +7,16 @@ procedure Invasion;
 
 procedure start
 begin
-    if (script_action == 15) then begin
+    if (script_action == map_enter_proc) then begin
         call Lighting();
         call add_caps();
-        if (metarule(14, 0)) then begin
+        if (map_first_run) then begin
             override_map_start(130, 107, 0, 0);
-            display_msg(message_str(112, 315));
+            display_msg(message_str(SCRIPT_RNDDESRT, 315));
         end
     end
     else begin
-        if (script_action == 23) then begin
+        if (script_action == map_update_proc) then begin
             call Lighting();
         end
     end
@@ -49,45 +49,45 @@ begin
     variable LVar0 := 0;
     variable LVar1 := 0;
     variable LVar2 := 0;
-    LVar0 := create_object_sid(180, 19919, 0, -1);
-    LVar1 := create_object_sid(41, 0, 0, -1);
-    if (get_critter_stat(dude_obj, 6) == 1) then begin
+    LVar0 := create_object(180, 19919, 0);
+    LVar1 := create_object(41, 0, 0);
+    if (dude_luck == 1) then begin
         LVar2 := 1;
     end
     else begin
-        if (get_critter_stat(dude_obj, 6) == 2) then begin
+        if (dude_luck == 2) then begin
             LVar2 := 32;
         end
         else begin
-            if (get_critter_stat(dude_obj, 6) == 3) then begin
+            if (dude_luck == 3) then begin
                 LVar2 := 105;
             end
             else begin
-                if (get_critter_stat(dude_obj, 6) == 4) then begin
+                if (dude_luck == 4) then begin
                     LVar2 := 298;
                 end
                 else begin
-                    if (get_critter_stat(dude_obj, 6) == 5) then begin
+                    if (dude_luck == 5) then begin
                         LVar2 := 730;
                     end
                     else begin
-                        if (get_critter_stat(dude_obj, 6) == 6) then begin
+                        if (dude_luck == 6) then begin
                             LVar2 := 1645;
                         end
                         else begin
-                            if (get_critter_stat(dude_obj, 6) == 7) then begin
+                            if (dude_luck == 7) then begin
                                 LVar2 := 2976;
                             end
                             else begin
-                                if (get_critter_stat(dude_obj, 6) == 8) then begin
+                                if (dude_luck == 8) then begin
                                     LVar2 := 5709;
                                 end
                                 else begin
-                                    if (get_critter_stat(dude_obj, 6) == 9) then begin
+                                    if (dude_luck == 9) then begin
                                         LVar2 := 8443;
                                     end
                                     else begin
-                                        if (get_critter_stat(dude_obj, 6) == 10) then begin
+                                        if (dude_luck == 10) then begin
                                             LVar2 := 10765;
                                         end
                                     end
@@ -99,7 +99,7 @@ begin
             end
         end
     end
-    LVar2 := LVar2 * (has_trait(0, dude_obj, 20) + 1);
+    LVar2 := LVar2 * (has_trait(TRAIT_PERK, dude_obj, PERK_fortune_finder) + 1);
     add_mult_objs_to_inven(LVar0, LVar1, LVar2);
 end
 
@@ -110,28 +110,27 @@ end
 
 procedure Invasion
 begin
-    if (not(global_var(18) == 2)) then begin
-        if (global_var(149) > (game_time / (10 * 60 * 60 * 24))) then begin
-            set_global_var(13, 1);
+    if (not(global_var(GVAR_MASTER_BLOWN) == 2)) then begin
+        if (global_var(GVAR_NECROPOLIS_INVADED_DATE) > (game_time / (10 * 60 * 60 * 24))) then begin
+            set_global_var(GVAR_NECROPOLIS_INVADED, 1);
         end
-        if (global_var(150) > (game_time / (10 * 60 * 60 * 24))) then begin
-            set_global_var(14, 1);
+        if (global_var(GVAR_THE_HUB_INVADED_DATE) > (game_time / (10 * 60 * 60 * 24))) then begin
+            set_global_var(GVAR_HUB_INVADED, 1);
         end
-        if (global_var(151) > (game_time / (10 * 60 * 60 * 24))) then begin
-            set_global_var(16, 1);
+        if (global_var(GVAR_BROTHERHOOD_INVADED_DATE) > (game_time / (10 * 60 * 60 * 24))) then begin
+            set_global_var(GVAR_BROTHERHOOD_INVADED, 1);
         end
-        if (global_var(152) > (game_time / (10 * 60 * 60 * 24))) then begin
-            set_global_var(15, 1);
+        if (global_var(GVAR_JUNKTOWN_INVADED_DATE) > (game_time / (10 * 60 * 60 * 24))) then begin
+            set_global_var(GVAR_JUNKTOWN_INVADED, 1);
         end
-        if (global_var(153) > (game_time / (10 * 60 * 60 * 24))) then begin
-            set_global_var(12, 1);
+        if (global_var(GVAR_SHADY_SANDS_INVADED_DATE) > (game_time / (10 * 60 * 60 * 24))) then begin
+            set_global_var(GVAR_SHADY_SANDS_INVADED, 1);
         end
-        if (global_var(154) <= 0) then begin
-            set_global_var(11, 1);
+        if (global_var(GVAR_VAULT_13_INVADED_DATE) <= 0) then begin
+            set_global_var(GVAR_VAULT_INVADED, 1);
         end
-        if (global_var(148) > (game_time / (10 * 60 * 60 * 24))) then begin
-            set_global_var(7, 1);
+        if (global_var(GVAR_FOLLOWERS_INVADED_DATE) > (game_time / (10 * 60 * 60 * 24))) then begin
+            set_global_var(GVAR_FOLLOWERS_INVADED, 1);
         end
     end
 end
-
