diff --git a/Games/Fallout1.Steam/Original/COMPUTER.ssl b/Games/Fallout1.Steam/ReDefine/COMPUTER.ssl
index d4fb8f6..ef237db 100644
--- a/Games/Fallout1.Steam/Original/COMPUTER.ssl
+++ b/Games/Fallout1.Steam/ReDefine/COMPUTER.ssl
@@ -6,38 +6,37 @@ variable test;
 
 procedure start
 begin
-    if (script_action == 4) then begin
+    if (script_action == pickup_proc) then begin
         script_overrides;
         call computer00();
     end
     else begin
-        if (script_action == 21) then begin
+        if (script_action == look_at_proc) then begin
             script_overrides;
-            display_msg(message_str(23, 100));
+            display_mstr(100);
         end
     end
 end
 
 procedure computer00
 begin
-    if (has_skill(dude_obj, 12) > has_skill(dude_obj, 13)) then begin
+    if (dude_skill(SKILL_SCIENCE) > dude_skill(SKILL_REPAIR)) then begin
         test := roll_vs_skill(dude_obj, 12, 0);
     end
     else begin
         test := roll_vs_skill(dude_obj, 13, 0);
     end
     if (is_success(test)) then begin
-        display_msg(message_str(23, 101));
+        display_mstr(101);
         if (is_critical(test)) then begin
-            display_msg(message_str(23, 102));
+            display_mstr(102);
         end
     end
     else begin
-        display_msg(message_str(23, 103));
+        display_mstr(103);
         if (is_critical(test)) then begin
-            display_msg(message_str(23, 104));
-            critter_dmg(dude_obj, 1, 0);
+            display_mstr(104);
+            critter_damage(dude_obj, 1);
         end
     end
 end
-
