diff --git a/Games/Fallout1.Steam/Original/COOKTABL.ssl b/Games/Fallout1.Steam/ReDefine/COOKTABL.ssl
index 51690d5..2736f24 100644
--- a/Games/Fallout1.Steam/Original/COOKTABL.ssl
+++ b/Games/Fallout1.Steam/ReDefine/COOKTABL.ssl
@@ -6,11 +6,11 @@ procedure description_p_proc;
 
 procedure start
 begin
-    if (script_action == 4) then begin
+    if (script_action == pickup_proc) then begin
         call pickup_p_proc();
     end
     else begin
-        if ((script_action == 3) or (script_action == 21)) then begin
+        if ((script_action == description_proc) or (script_action == look_at_proc)) then begin
             call look_at_p_proc();
         end
     end
@@ -19,18 +19,17 @@ end
 procedure pickup_p_proc
 begin
     script_overrides;
-    display_msg(message_str(126, 101));
+    display_mstr(101);
 end
 
 procedure look_at_p_proc
 begin
     script_overrides;
-    display_msg(message_str(126, 100));
+    display_mstr(100);
 end
 
 procedure description_p_proc
 begin
     script_overrides;
-    display_msg(message_str(126, 100));
+    display_mstr(100);
 end
-
