diff --git a/Games/Fallout1.Steam/Original/CORECOMP.ssl b/Games/Fallout1.Steam/ReDefine/CORECOMP.ssl
index 6f96bb9..f895670 100644
--- a/Games/Fallout1.Steam/Original/CORECOMP.ssl
+++ b/Games/Fallout1.Steam/ReDefine/CORECOMP.ssl
@@ -6,15 +6,15 @@ procedure use_skill_on_p_proc;
 
 procedure start
 begin
-    if (script_action == 3) then begin
+    if (script_action == description_proc) then begin
         call description_p_proc();
     end
     else begin
-        if (script_action == 6) then begin
+        if (script_action == use_proc) then begin
             call use_p_proc();
         end
         else begin
-            if (script_action == 8) then begin
+            if (script_action == use_skill_on_proc) then begin
                 call use_skill_on_p_proc();
             end
         end
@@ -24,13 +24,13 @@ end
 procedure description_p_proc
 begin
     script_overrides;
-    display_msg(message_str(133, 100));
+    display_mstr(100);
 end
 
 procedure use_p_proc
 begin
     script_overrides;
-    display_msg(message_str(793, 100));
+    display_msg(message_str(SCRIPT_VALTCOMP, 100));
 end
 
 procedure use_skill_on_p_proc
@@ -39,15 +39,14 @@ begin
         script_overrides;
         if (is_success(roll_vs_skill(dude_obj, action_being_used, 0))) then begin
             if ((game_time / (10 * 60 * 60 * 24)) >= 90) then begin
-                display_msg(message_str(133, 101));
+                display_mstr(101);
             end
             else begin
-                display_msg(message_str(133, 102));
+                display_mstr(102);
             end
         end
         else begin
-            display_msg(message_str(133, 103));
+            display_mstr(103);
         end
     end
 end
-
