diff --git a/Games/Fallout1.Steam/Original/CTOWER2.ssl b/Games/Fallout1.Steam/ReDefine/CTOWER2.ssl
index fa13cd0..17dd895 100644
--- a/Games/Fallout1.Steam/Original/CTOWER2.ssl
+++ b/Games/Fallout1.Steam/ReDefine/CTOWER2.ssl
@@ -3,9 +3,8 @@ procedure start;
 
 procedure start
 begin
-    if (script_action == 6) then begin
+    if (script_action == use_proc) then begin
         script_overrides;
         move_to(dude_obj, 19101, 1);
     end
 end
-
