diff --git a/Games/Fallout1.Steam/Original/DISKD.ssl b/Games/Fallout1.Steam/ReDefine/DISKD.ssl
index 6355c95..ce44ef9 100644
--- a/Games/Fallout1.Steam/Original/DISKD.ssl
+++ b/Games/Fallout1.Steam/ReDefine/DISKD.ssl
@@ -3,25 +3,24 @@ procedure start;
 
 procedure start
 begin
-    if ((script_action == 21) or (script_action == 3)) then begin
+    if ((script_action == look_at_proc) or (script_action == description_proc)) then begin
         script_overrides;
-        display_msg(message_str(346, 100));
+        display_mstr(100);
     end
     else begin
-        if (script_action == 6) then begin
+        if (script_action == use_proc) then begin
             script_overrides;
             if (local_var(0) == 1) then begin
-                display_msg(message_str(346, 101));
+                display_mstr(101);
             end
             else begin
-                display_msg(message_str(346, 102));
-                set_global_var(165, 1);
+                display_mstr(102);
+                set_global_var(GVAR_DELTA_DISK, 1);
                 set_local_var(0, 1);
                 rm_obj_from_inven(dude_obj, self_obj);
-                display_msg(message_str(766, 103) + "100" + message_str(766, 104));
+                display_msg(message_str(SCRIPT_GENCHAT, 103) + "100" + message_str(SCRIPT_GENCHAT, 104));
                 give_exp_points(100);
             end
         end
     end
 end
-
