diff --git a/Games/Fallout1.Steam/Original/DRWHOMAP.ssl b/Games/Fallout1.Steam/ReDefine/DRWHOMAP.ssl
index 538d1e7..64dbfb6 100644
--- a/Games/Fallout1.Steam/Original/DRWHOMAP.ssl
+++ b/Games/Fallout1.Steam/ReDefine/DRWHOMAP.ssl
@@ -11,11 +11,11 @@ procedure Invasion;
 
 procedure start
 begin
-    if (script_action == 15) then begin
+    if (script_action == map_enter_proc) then begin
         call map_enter_p_proc();
     end
     else begin
-        if (script_action == 23) then begin
+        if (script_action == map_update_proc) then begin
             call map_update_p_proc();
         end
     end
@@ -60,28 +60,27 @@ end
 
 procedure Invasion
 begin
-    if (not(global_var(18) == 2)) then begin
-        if (global_var(149) > (game_time / (10 * 60 * 60 * 24))) then begin
-            set_global_var(13, 1);
+    if (not(global_var(GVAR_MASTER_BLOWN) == 2)) then begin
+        if (global_var(GVAR_NECROPOLIS_INVADED_DATE) > (game_time / (10 * 60 * 60 * 24))) then begin
+            set_global_var(GVAR_NECROPOLIS_INVADED, 1);
         end
-        if (global_var(150) > (game_time / (10 * 60 * 60 * 24))) then begin
-            set_global_var(14, 1);
+        if (global_var(GVAR_THE_HUB_INVADED_DATE) > (game_time / (10 * 60 * 60 * 24))) then begin
+            set_global_var(GVAR_HUB_INVADED, 1);
         end
-        if (global_var(151) > (game_time / (10 * 60 * 60 * 24))) then begin
-            set_global_var(16, 1);
+        if (global_var(GVAR_BROTHERHOOD_INVADED_DATE) > (game_time / (10 * 60 * 60 * 24))) then begin
+            set_global_var(GVAR_BROTHERHOOD_INVADED, 1);
         end
-        if (global_var(152) > (game_time / (10 * 60 * 60 * 24))) then begin
-            set_global_var(15, 1);
+        if (global_var(GVAR_JUNKTOWN_INVADED_DATE) > (game_time / (10 * 60 * 60 * 24))) then begin
+            set_global_var(GVAR_JUNKTOWN_INVADED, 1);
         end
-        if (global_var(153) > (game_time / (10 * 60 * 60 * 24))) then begin
-            set_global_var(12, 1);
+        if (global_var(GVAR_SHADY_SANDS_INVADED_DATE) > (game_time / (10 * 60 * 60 * 24))) then begin
+            set_global_var(GVAR_SHADY_SANDS_INVADED, 1);
         end
-        if (global_var(154) <= 0) then begin
-            set_global_var(11, 1);
+        if (global_var(GVAR_VAULT_13_INVADED_DATE) <= 0) then begin
+            set_global_var(GVAR_VAULT_INVADED, 1);
         end
-        if (global_var(148) > (game_time / (10 * 60 * 60 * 24))) then begin
-            set_global_var(7, 1);
+        if (global_var(GVAR_FOLLOWERS_INVADED_DATE) > (game_time / (10 * 60 * 60 * 24))) then begin
+            set_global_var(GVAR_FOLLOWERS_INVADED, 1);
         end
     end
 end
-
