diff --git a/Games/Fallout1.Steam/Original/EDEBRIS.ssl b/Games/Fallout1.Steam/ReDefine/EDEBRIS.ssl
index ab7210c..26c861f 100644
--- a/Games/Fallout1.Steam/Original/EDEBRIS.ssl
+++ b/Games/Fallout1.Steam/ReDefine/EDEBRIS.ssl
@@ -3,19 +3,18 @@ procedure start;
 
 procedure start
 begin
-    if (script_action == 2) then begin
+    if (script_action == spatial_proc) then begin
         if (source_obj == dude_obj) then begin
             if (map_var(2) == 0) then begin
                 set_map_var(2, 1);
                 set_map_var(4, map_var(4) + 1);
                 if (map_var(4) > 3) then begin
-                    display_msg(message_str(824, 104));
+                    display_msg(message_str(SCRIPT_NDEBRIS, 104));
                 end
                 else begin
-                    display_msg(message_str(824, 102));
+                    display_msg(message_str(SCRIPT_NDEBRIS, 102));
                 end
             end
         end
     end
 end
-
