diff --git a/Games/Fallout1.Steam/Original/ELEVATR.ssl b/Games/Fallout1.Steam/ReDefine/ELEVATR.ssl
index df7f8f3..39b66a9 100644
--- a/Games/Fallout1.Steam/Original/ELEVATR.ssl
+++ b/Games/Fallout1.Steam/ReDefine/ELEVATR.ssl
@@ -3,7 +3,7 @@ procedure start;
 
 procedure start
 begin
-    if (script_action == 2) then begin
+    if (script_action == spatial_proc) then begin
         if (obj_type(source_obj) != 1) then begin
             detach;
         end
@@ -17,4 +17,3 @@ begin
     end
     detach;
 end
-
