diff --git a/Games/Fallout1.Steam/Original/ELEV6_M1.ssl b/Games/Fallout1.Steam/ReDefine/ELEV6_M1.ssl
index a0fe050..8077dcc 100644
--- a/Games/Fallout1.Steam/Original/ELEV6_M1.ssl
+++ b/Games/Fallout1.Steam/ReDefine/ELEV6_M1.ssl
@@ -3,10 +3,9 @@ procedure start;
 
 procedure start
 begin
-    if (script_action == 2) then begin
+    if (script_action == spatial_proc) then begin
         if (source_obj == dude_obj) then begin
-            metarule(15, 7);
+            elevator(7);
         end
     end
 end
-
