diff --git a/Games/Fallout1.Steam/Original/FALCON1.ssl b/Games/Fallout1.Steam/ReDefine/FALCON1.ssl
index d455c87..5e19ecd 100644
--- a/Games/Fallout1.Steam/Original/FALCON1.ssl
+++ b/Games/Fallout1.Steam/ReDefine/FALCON1.ssl
@@ -50,26 +50,26 @@ procedure start
 begin
     if (Only_Once) then begin
         Only_Once := 0;
-        critter_add_trait(self_obj, 1, 6, 38);
-        critter_add_trait(self_obj, 1, 5, 50);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 38);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_MERCHANT);
     end
-    if (script_action == 21) then begin
+    if (script_action == look_at_proc) then begin
         call look_at_p_proc();
     end
     else begin
-        if (script_action == 4) then begin
+        if (script_action == pickup_proc) then begin
             call pickup_p_proc();
         end
         else begin
-            if (script_action == 11) then begin
+            if (script_action == talk_proc) then begin
                 call talk_p_proc();
             end
             else begin
-                if (script_action == 12) then begin
+                if (script_action == critter_proc) then begin
                     call critter_p_proc();
                 end
                 else begin
-                    if (script_action == 18) then begin
+                    if (script_action == destroy_proc) then begin
                         call destroy_p_proc();
                     end
                 end
@@ -87,7 +87,7 @@ procedure critter_p_proc
 begin
     if (hostile) then begin
         hostile := 0;
-        attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+        attack(dude_obj);
     end
 end
 
@@ -111,17 +111,17 @@ end
 procedure destroy_p_proc
 begin
     if (source_obj == dude_obj) then begin
-        if (((global_var(160) + global_var(159)) >= 25) and ((global_var(159) > (2 * global_var(160))) or (global_var(156) == 1))) then begin
-            set_global_var(156, 1);
-            set_global_var(157, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
+            set_global_var(GVAR_BERSERKER_REPUTATION, 1);
+            set_global_var(GVAR_CHAMPION_REPUTATION, 0);
         end
-        if (((global_var(160) + global_var(159)) >= 25) and ((global_var(160) > (3 * global_var(159))) or (global_var(157) == 1))) then begin
-            set_global_var(157, 1);
-            set_global_var(156, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(160) > (3 * global_var(GVAR_GOOD_MONSTER))) or (global_var(GVAR_CHAMPION_REPUTATION) == 1))) then begin
+            set_global_var(GVAR_CHAMPION_REPUTATION, 1);
+            set_global_var(GVAR_BERSERKER_REPUTATION, 0);
         end
-        set_global_var(159, global_var(159) + 1);
-        if ((global_var(159) % 2) == 0) then begin
-            set_global_var(155, global_var(155) - 1);
+        set_global_var(GVAR_GOOD_MONSTER, global_var(GVAR_GOOD_MONSTER) + 1);
+        if ((global_var(GVAR_GOOD_MONSTER) % 2) == 0) then begin
+            set_global_var(GVAR_PLAYER_REPUATION, global_var(GVAR_PLAYER_REPUATION) - 1);
         end
     end
 end
@@ -129,49 +129,49 @@ end
 procedure look_at_p_proc
 begin
     script_overrides;
-    display_msg(message_str(697, 100));
+    display_mstr(100);
 end
 
 procedure Falcon00
 begin
-    gsay_reply(697, 101);
-    giq_option(4, 697, 102, Falcon05, 50);
-    giq_option(4, 697, 103, Falcon06, 50);
-    giq_option(4, 697, 104, Falcon07, 50);
-    giq_option(4, 697, 106, Falcon04, 50);
-    giq_option(-3, 697, 107, Falcon03, 50);
+    Reply(101);
+    giq_option(4, NAME, 102, Falcon05, NEUTRAL_REACTION);
+    giq_option(4, NAME, 103, Falcon06, NEUTRAL_REACTION);
+    giq_option(4, NAME, 104, Falcon07, NEUTRAL_REACTION);
+    giq_option(4, NAME, 106, Falcon04, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 107, Falcon03, NEUTRAL_REACTION);
 end
 
 procedure Falcon01
 begin
-    gsay_message(697, 108, 50);
+    gsay_message(NAME, 108, NEUTRAL_REACTION);
 end
 
 procedure Falcon02
 begin
-    gsay_message(697, 109, 50);
+    gsay_message(NAME, 109, NEUTRAL_REACTION);
 end
 
 procedure Falcon03
 begin
-    if (get_critter_stat(dude_obj, 34) == 0) then begin
-        gsay_message(697, 110, 50);
+    if (dude_gender == 0) then begin
+        gsay_message(NAME, 110, NEUTRAL_REACTION);
     end
     else begin
-        gsay_message(697, 111, 50);
+        gsay_message(NAME, 111, NEUTRAL_REACTION);
     end
 end
 
 procedure Falcon04
 begin
-    gsay_message(697, 112, 50);
+    gsay_message(NAME, 112, NEUTRAL_REACTION);
 end
 
 procedure Falcon05
 begin
-    gsay_reply(697, 113);
-    giq_option(4, 697, 114, Falcon05a, 50);
-    giq_option(4, 697, 115, Falcon11, 50);
+    Reply(113);
+    giq_option(4, NAME, 114, Falcon05a, NEUTRAL_REACTION);
+    giq_option(4, NAME, 115, Falcon11, NEUTRAL_REACTION);
 end
 
 procedure Falcon05a
@@ -186,9 +186,9 @@ end
 
 procedure Falcon06
 begin
-    gsay_reply(697, 116);
-    giq_option(4, 697, 117, Falcon06a, 50);
-    giq_option(4, 697, 115, Falcon11, 50);
+    Reply(116);
+    giq_option(4, NAME, 117, Falcon06a, NEUTRAL_REACTION);
+    giq_option(4, NAME, 115, Falcon11, NEUTRAL_REACTION);
 end
 
 procedure Falcon06a
@@ -203,9 +203,9 @@ end
 
 procedure Falcon07
 begin
-    gsay_reply(697, 118);
-    giq_option(4, 697, 117, Falcon07a, 50);
-    giq_option(4, 697, 115, Falcon11, 50);
+    Reply(118);
+    giq_option(4, NAME, 117, Falcon07a, NEUTRAL_REACTION);
+    giq_option(4, NAME, 115, Falcon11, NEUTRAL_REACTION);
 end
 
 procedure Falcon07a
@@ -221,10 +221,10 @@ end
 procedure Falcon09
 begin
     if (First_Water) then begin
-        gsay_reply(697, message_str(697, 139) + message_str(697, 140) + message_str(697, 141));
+        Reply(mstr(139) + mstr(140) + mstr(141));
     end
     else begin
-        gsay_reply(697, message_str(697, 136) + message_str(697, 137) + message_str(697, 138));
+        Reply(mstr(136) + mstr(137) + mstr(138));
     end
     Money_Pack := obj_carrying_pid_obj(dude_obj, 41);
     Money_Count := rm_mult_objs_from_inven(dude_obj, Money_Pack, 2);
@@ -236,10 +236,10 @@ end
 procedure Falcon11
 begin
     if (random(0, 1)) then begin
-        gsay_message(697, message_str(697, 151) + message_str(697, 152), 50);
+        gsay_message(NAME, mstr(151) + mstr(152), NEUTRAL_REACTION);
     end
     else begin
-        gsay_message(697, message_str(697, 153) + message_str(697, 154), 50);
+        gsay_message(NAME, mstr(153) + mstr(154), NEUTRAL_REACTION);
     end
 end
 
@@ -248,19 +248,19 @@ begin
     Money_Pack := obj_carrying_pid_obj(dude_obj, 41);
     Money_Count := rm_mult_objs_from_inven(dude_obj, Money_Pack, 5);
     destroy_object(Money_Pack);
-    item := create_object_sid(124, 0, 0, -1);
+    item := create_object(124, 0, 0);
     add_obj_to_inven(dude_obj, item);
-    gsay_reply(697, message_str(697, 160) + message_str(697, 161));
+    Reply(mstr(160) + mstr(161));
     call Falcon15();
 end
 
 procedure Falcon15
 begin
-    giq_option(4, 697, 102, Falcon05, 50);
-    giq_option(4, 697, 103, Falcon06, 50);
-    giq_option(4, 697, 104, Falcon07, 50);
-    giq_option(4, 697, 106, Falcon04, 50);
-    giq_option(-3, 697, 107, Falcon03, 50);
+    giq_option(4, NAME, 102, Falcon05, NEUTRAL_REACTION);
+    giq_option(4, NAME, 103, Falcon06, NEUTRAL_REACTION);
+    giq_option(4, NAME, 104, Falcon07, NEUTRAL_REACTION);
+    giq_option(4, NAME, 106, Falcon04, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 107, Falcon03, NEUTRAL_REACTION);
 end
 
 procedure Falcon16
@@ -268,9 +268,9 @@ begin
     Money_Pack := obj_carrying_pid_obj(dude_obj, 41);
     Money_Count := rm_mult_objs_from_inven(dude_obj, Money_Pack, 10);
     destroy_object(Money_Pack);
-    item := create_object_sid(125, 0, 0, -1);
+    item := create_object(125, 0, 0);
     add_obj_to_inven(dude_obj, item);
-    gsay_reply(697, message_str(697, 163) + message_str(697, 164));
+    Reply(mstr(163) + mstr(164));
     call Falcon15();
 end
 
@@ -284,31 +284,31 @@ begin
         set_local_var(0, 50);
         set_local_var(1, 2);
         set_local_var(2, 1);
-        set_local_var(0, local_var(0) + (5 * get_critter_stat(dude_obj, 3)) - 25);
-        set_local_var(0, local_var(0) + (10 * has_trait(0, dude_obj, 10)));
-        if (has_trait(0, dude_obj, 39)) then begin
-            if (global_var(155) > 0) then begin
-                set_local_var(0, local_var(0) + global_var(155));
+        set_local_var(0, local_var(0) + (5 * dude_charisma) - 25);
+        set_local_var(0, local_var(0) + (10 * has_trait(TRAIT_PERK, dude_obj, PERK_presence)));
+        if (has_trait(TRAIT_PERK, dude_obj, PERK_cult_of_personality)) then begin
+            if (global_var(GVAR_PLAYER_REPUATION) > 0) then begin
+                set_local_var(0, local_var(0) + global_var(GVAR_PLAYER_REPUATION));
             end
             else begin
-                set_local_var(0, local_var(0) - global_var(155));
+                set_local_var(0, local_var(0) - global_var(GVAR_PLAYER_REPUATION));
             end
         end
         else begin
             if (local_var(3) == 1) then begin
-                set_local_var(0, local_var(0) - global_var(155));
+                set_local_var(0, local_var(0) - global_var(GVAR_PLAYER_REPUATION));
             end
             else begin
-                set_local_var(0, local_var(0) + global_var(155));
+                set_local_var(0, local_var(0) + global_var(GVAR_PLAYER_REPUATION));
             end
         end
-        if (global_var(158) > 2) then begin
+        if (global_var(GVAR_CHILDKILLER_REPUATION) > 2) then begin
             set_local_var(0, local_var(0) - 30);
         end
-        if (((global_var(160) + global_var(159)) >= 25) and ((global_var(160) > (3 * global_var(159))) or (global_var(157) == 1))) then begin
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(160) > (3 * global_var(GVAR_GOOD_MONSTER))) or (global_var(GVAR_CHAMPION_REPUTATION) == 1))) then begin
             set_local_var(0, local_var(0) + 20);
         end
-        if (((global_var(160) + global_var(159)) >= 25) and ((global_var(159) > (2 * global_var(160))) or (global_var(156) == 1))) then begin
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
             set_local_var(0, local_var(0) - 20);
         end
         call ReactToLevel();
@@ -401,6 +401,5 @@ end
 
 procedure Goodbyes
 begin
-    exit_line := message_str(634, random(100, 105));
+    exit_line := message_str(SCRIPT_MODREACT, random(100, 105));
 end
-
