diff --git a/Games/Fallout1.Steam/Original/FGTCARVN.ssl b/Games/Fallout1.Steam/ReDefine/FGTCARVN.ssl
index 33f6155..703f491 100644
--- a/Games/Fallout1.Steam/Original/FGTCARVN.ssl
+++ b/Games/Fallout1.Steam/ReDefine/FGTCARVN.ssl
@@ -66,26 +66,26 @@ procedure start
 begin
     if (Only_Once) then begin
         Only_Once := 0;
-        critter_add_trait(self_obj, 1, 6, 37);
-        critter_add_trait(self_obj, 1, 5, 50);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 37);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_MERCHANT);
     end
-    if (script_action == 21) then begin
+    if (script_action == look_at_proc) then begin
         call look_at_p_proc();
     end
     else begin
-        if (script_action == 4) then begin
+        if (script_action == pickup_proc) then begin
             call pickup_p_proc();
         end
         else begin
-            if (script_action == 11) then begin
+            if (script_action == talk_proc) then begin
                 call talk_p_proc();
             end
             else begin
-                if (script_action == 12) then begin
+                if (script_action == critter_proc) then begin
                     call critter_p_proc();
                 end
                 else begin
-                    if (script_action == 18) then begin
+                    if (script_action == destroy_proc) then begin
                         call destroy_p_proc();
                     end
                 end
@@ -103,7 +103,7 @@ procedure critter_p_proc
 begin
     if (hostile) then begin
         hostile := 0;
-        attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+        attack(dude_obj);
     end
 end
 
@@ -141,7 +141,7 @@ begin
             call FGTDriver15();
         end
         else begin
-            if (global_var(206) == 3) then begin
+            if (global_var(GVAR_FARGO_JOB) == 3) then begin
                 start_gdialog(849, self_obj, 4, -1, -1);
                 gsay_start;
                 call FGTDriver19();
@@ -149,7 +149,7 @@ begin
                 end_dialogue;
             end
             else begin
-                if (global_var(206) == 4) then begin
+                if (global_var(GVAR_FARGO_JOB) == 4) then begin
                     start_gdialog(849, self_obj, 4, -1, -1);
                     gsay_start;
                     call FGTDriver27();
@@ -202,17 +202,17 @@ end
 procedure destroy_p_proc
 begin
     if (source_obj == dude_obj) then begin
-        if (((global_var(160) + global_var(159)) >= 25) and ((global_var(159) > (2 * global_var(160))) or (global_var(156) == 1))) then begin
-            set_global_var(156, 1);
-            set_global_var(157, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
+            set_global_var(GVAR_BERSERKER_REPUTATION, 1);
+            set_global_var(GVAR_CHAMPION_REPUTATION, 0);
         end
-        if (((global_var(160) + global_var(159)) >= 25) and ((global_var(160) > (3 * global_var(159))) or (global_var(157) == 1))) then begin
-            set_global_var(157, 1);
-            set_global_var(156, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(160) > (3 * global_var(GVAR_GOOD_MONSTER))) or (global_var(GVAR_CHAMPION_REPUTATION) == 1))) then begin
+            set_global_var(GVAR_CHAMPION_REPUTATION, 1);
+            set_global_var(GVAR_BERSERKER_REPUTATION, 0);
         end
-        set_global_var(159, global_var(159) + 1);
-        if ((global_var(159) % 2) == 0) then begin
-            set_global_var(155, global_var(155) - 1);
+        set_global_var(GVAR_GOOD_MONSTER, global_var(GVAR_GOOD_MONSTER) + 1);
+        if ((global_var(GVAR_GOOD_MONSTER) % 2) == 0) then begin
+            set_global_var(GVAR_PLAYER_REPUATION, global_var(GVAR_PLAYER_REPUATION) - 1);
         end
     end
 end
@@ -220,7 +220,7 @@ end
 procedure look_at_p_proc
 begin
     script_overrides;
-    display_msg(message_str(849, 100));
+    display_mstr(100);
 end
 
 procedure RecalcDate
@@ -275,51 +275,51 @@ begin
     LVar0 := local_var(6);
     LVar1 := local_var(5);
     if (LVar0 == 1) then begin
-        DayString := message_str(849, 204);
+        DayString := mstr(204);
     end
     else begin
         if (LVar0 == 2) then begin
-            DayString := message_str(849, 205);
+            DayString := mstr(205);
         end
         else begin
             if (LVar0 == 3) then begin
-                DayString := message_str(849, 206);
+                DayString := mstr(206);
             end
             else begin
                 if (LVar0 == 4) then begin
-                    DayString := message_str(849, 207);
+                    DayString := mstr(207);
                 end
                 else begin
                     if (LVar0 == 5) then begin
-                        DayString := message_str(849, 208);
+                        DayString := mstr(208);
                     end
                     else begin
                         if (LVar0 == 6) then begin
-                            DayString := message_str(849, 209);
+                            DayString := mstr(209);
                         end
                         else begin
                             if (LVar0 == 7) then begin
-                                DayString := message_str(849, 210);
+                                DayString := mstr(210);
                             end
                             else begin
                                 if (LVar0 == 8) then begin
-                                    DayString := message_str(849, 211);
+                                    DayString := mstr(211);
                                 end
                                 else begin
                                     if (LVar0 == 9) then begin
-                                        DayString := message_str(849, 212);
+                                        DayString := mstr(212);
                                     end
                                     else begin
                                         if (LVar0 == 10) then begin
-                                            DayString := message_str(849, 213);
+                                            DayString := mstr(213);
                                         end
                                         else begin
                                             if (LVar0 == 11) then begin
-                                                DayString := message_str(849, 214);
+                                                DayString := mstr(214);
                                             end
                                             else begin
                                                 if (LVar0 == 12) then begin
-                                                    DayString := message_str(849, 215);
+                                                    DayString := mstr(215);
                                                 end
                                             end
                                         end
@@ -333,15 +333,15 @@ begin
         end
     end
     if (LVar1 == 8) then begin
-        DayString := DayString + message_str(849, 197);
+        DayString := DayString + mstr(197);
     end
     else begin
         if (LVar1 == 18) then begin
-            DayString := DayString + message_str(849, 198);
+            DayString := DayString + mstr(198);
         end
         else begin
             if (LVar1 == 28) then begin
-                DayString := DayString + message_str(849, 199);
+                DayString := DayString + mstr(199);
             end
         end
     end
@@ -352,35 +352,35 @@ begin
     variable LVar0 := 0;
     LVar0 := obj_pid(source_obj);
     if (party_member_obj(LVar0) != 0) then begin
-        set_global_var(248, 1);
+        set_global_var(GVAR_ENEMY_HUB, 1);
     end
 end
 
 procedure FGTDriver00
 begin
     if (local_var(4) == 0) then begin
-        float_msg(self_obj, message_str(849, 101), 2);
+        float_msg(self_obj, mstr(101), FLOAT_MSG_RED);
         set_local_var(4, 1);
     end
     else begin
-        float_msg(self_obj, message_str(849, 102), 2);
+        float_msg(self_obj, mstr(102), FLOAT_MSG_RED);
     end
 end
 
 procedure FGTDriver01
 begin
-    if (global_var(71) == 0) then begin
-        set_global_var(71, 1);
+    if (global_var(GVAR_JUNKTOWN) == 0) then begin
+        set_global_var(GVAR_JUNKTOWN, 1);
     end
-    if (global_var(75) == 0) then begin
-        set_global_var(75, 1);
+    if (global_var(GVAR_ANGELS_BONEYARD) == 0) then begin
+        set_global_var(GVAR_ANGELS_BONEYARD, 1);
     end
-    if (global_var(74) == 0) then begin
-        set_global_var(74, 1);
+    if (global_var(GVAR_BROTHERHOOD_OF_STEEL) == 0) then begin
+        set_global_var(GVAR_BROTHERHOOD_OF_STEEL, 1);
     end
-    gsay_reply(849, 103);
-    giq_option(4, 849, 104, FGTDriver02, 50);
-    giq_option(-3, 849, 105, FGTDriver02, 50);
+    Reply(103);
+    giq_option(4, NAME, 104, FGTDriver02, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 105, FGTDriver02, NEUTRAL_REACTION);
 end
 
 procedure FGTDriver02
@@ -389,10 +389,10 @@ begin
     call RecalcDateString();
     LVar0 := local_var(5);
     if (LVar0 == get_day) then begin
-        gsay_reply(849, message_str(849, 106) + message_str(849, 107));
+        Reply(mstr(106) + mstr(107));
     end
     else begin
-        gsay_reply(849, message_str(849, 106) + message_str(849, 108) + DayString + message_str(849, 111));
+        Reply(mstr(106) + mstr(108) + DayString + mstr(111));
     end
     call FGTDriver04();
 end
@@ -404,26 +404,26 @@ begin
         call RecalcDateString();
         LVar0 := local_var(5);
         if (LVar0 == get_day) then begin
-            gsay_reply(849, message_str(849, 107));
+            Reply(mstr(107));
         end
         else begin
-            gsay_reply(849, message_str(849, 108) + DayString + message_str(849, 111));
+            Reply(mstr(108) + DayString + mstr(111));
         end
         call FGTDriver04();
     end
     else begin
-        gsay_reply(849, 112);
+        Reply(112);
         call FGTDriver04();
     end
 end
 
 procedure FGTDriver04
 begin
-    giq_option(4, 849, 113, FGTDriver04a, 50);
-    giq_option(4, 849, 114, FGTDriver05, 50);
-    giq_option(4, 849, 115, FGTDriver06, 50);
-    giq_option(-3, 849, 116, FGTDriver07, 50);
-    giq_option(-3, 849, 117, FGTDriver07, 50);
+    giq_option(4, NAME, 113, FGTDriver04a, NEUTRAL_REACTION);
+    giq_option(4, NAME, 114, FGTDriver05, NEUTRAL_REACTION);
+    giq_option(4, NAME, 115, FGTDriver06, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 116, FGTDriver07, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 117, FGTDriver07, NEUTRAL_REACTION);
 end
 
 procedure FGTDriver04a
@@ -443,23 +443,23 @@ end
 procedure FGTDriver05
 begin
     set_local_var(10, 0);
-    gsay_message(849, 118, 50);
+    gsay_message(NAME, 118, NEUTRAL_REACTION);
 end
 
 procedure FGTDriver06
 begin
     set_local_var(10, 0);
-    gsay_message(849, 120, 50);
+    gsay_message(NAME, 120, NEUTRAL_REACTION);
 end
 
 procedure FGTDriver07
 begin
-    gsay_message(849, 121, 50);
+    gsay_message(NAME, 121, NEUTRAL_REACTION);
 end
 
 procedure FGTDriver08
 begin
-    gsay_message(849, 123, 50);
+    gsay_message(NAME, 123, NEUTRAL_REACTION);
 end
 
 procedure FGTDriver10
@@ -468,14 +468,14 @@ begin
         set_local_var(8, 1);
         set_map_var(35, 1);
         set_map_var(10, 0);
-        gsay_reply(849, 125);
-        giq_option(4, 849, 126, FGTDriver08, 50);
-        giq_option(4, 849, 127, FGTDriver10a, 50);
-        giq_option(4, 849, 128, FGTDriver31, 51);
-        giq_option(-3, 849, 130, FGTDriver31, 51);
+        Reply(125);
+        giq_option(4, NAME, 126, FGTDriver08, NEUTRAL_REACTION);
+        giq_option(4, NAME, 127, FGTDriver10a, NEUTRAL_REACTION);
+        giq_option(4, NAME, 128, FGTDriver31, BAD_REACTION);
+        giq_option(-3, NAME, 130, FGTDriver31, BAD_REACTION);
     end
     else begin
-        gsay_message(849, 131, 51);
+        gsay_message(NAME, 131, BAD_REACTION);
         call combat();
     end
 end
@@ -495,27 +495,27 @@ begin
     set_map_var(36, 0);
     set_map_var(35, 1);
     set_map_var(10, 0);
-    gsay_message(849, 132, 50);
+    gsay_message(NAME, 132, NEUTRAL_REACTION);
 end
 
 procedure FGTDriver12
 begin
-    gsay_message(849, 133, 51);
+    gsay_message(NAME, 133, BAD_REACTION);
     call combat();
 end
 
 procedure FGTDriver13
 begin
-    float_msg(self_obj, message_str(849, 137), 2);
+    float_msg(self_obj, mstr(137), FLOAT_MSG_RED);
 end
 
 procedure FGTDriver14
 begin
     if (local_var(8) > 0) then begin
-        float_msg(self_obj, message_str(849, 138), 2);
+        float_msg(self_obj, mstr(138), FLOAT_MSG_RED);
     end
     else begin
-        float_msg(self_obj, message_str(849, 139), 2);
+        float_msg(self_obj, mstr(139), FLOAT_MSG_RED);
         call FGTDriver31();
     end
 end
@@ -525,19 +525,19 @@ begin
     variable LVar0 := 0;
     LVar0 := random(1, 4);
     if (LVar0 == 1) then begin
-        float_msg(self_obj, message_str(849, 140), 2);
+        float_msg(self_obj, mstr(140), FLOAT_MSG_RED);
     end
     else begin
         if (LVar0 == 2) then begin
-            float_msg(self_obj, message_str(849, 141), 2);
+            float_msg(self_obj, mstr(141), FLOAT_MSG_RED);
         end
         else begin
             if (LVar0 == 3) then begin
-                float_msg(self_obj, message_str(849, 142), 2);
+                float_msg(self_obj, mstr(142), FLOAT_MSG_RED);
             end
             else begin
                 if (LVar0 == 4) then begin
-                    float_msg(self_obj, message_str(849, 143), 2);
+                    float_msg(self_obj, mstr(143), FLOAT_MSG_RED);
                 end
             end
         end
@@ -546,7 +546,7 @@ end
 
 procedure FGTDriver17
 begin
-    gsay_reply(849, 145);
+    Reply(145);
     call FGTDriver33();
     call FGTDriver20();
 end
@@ -554,58 +554,58 @@ end
 procedure FGTDriver18
 begin
     call RecalcDateString();
-    gsay_reply(849, message_str(849, 146) + DayString + message_str(849, 153));
-    giq_option(4, 849, 154, FGTDriver30, 50);
-    giq_option(4, 849, 155, FGTDriver34, 50);
+    Reply(mstr(146) + DayString + mstr(153));
+    giq_option(4, NAME, 154, FGTDriver30, NEUTRAL_REACTION);
+    giq_option(4, NAME, 155, FGTDriver34, NEUTRAL_REACTION);
 end
 
 procedure FGTDriver19
 begin
     set_map_var(35, 1);
     set_map_var(10, 0);
-    set_global_var(206, 0);
-    gsay_message(849, 156, 51);
+    set_global_var(GVAR_FARGO_JOB, 0);
+    gsay_message(NAME, 156, BAD_REACTION);
 end
 
 procedure FGTDriver20
 begin
-    giq_option(4, 849, 157, FGTDriver21a, 50);
-    giq_option(4, 849, 158, FGTDriver21b, 50);
-    giq_option(4, 849, 152, FGTDriver21c, 50);
-    giq_option(4, 849, 159, FGTDriver34, 50);
+    giq_option(4, NAME, 157, FGTDriver21a, NEUTRAL_REACTION);
+    giq_option(4, NAME, 158, FGTDriver21b, NEUTRAL_REACTION);
+    giq_option(4, NAME, 152, FGTDriver21c, NEUTRAL_REACTION);
+    giq_option(4, NAME, 159, FGTDriver34, NEUTRAL_REACTION);
 end
 
 procedure FGTDriver21a
 begin
     variable LVar0 := 0;
-    set_global_var(201, 1);
+    set_global_var(GVAR_FARGO_TRADERS_STATUS, 1);
     LVar0 := random(1, 7);
     if (LVar0 == 1) then begin
-        gsay_message(849, 160, 50);
+        gsay_message(NAME, 160, NEUTRAL_REACTION);
     end
     else begin
         if (LVar0 == 2) then begin
-            gsay_message(849, 161, 50);
+            gsay_message(NAME, 161, NEUTRAL_REACTION);
         end
         else begin
             if (LVar0 == 3) then begin
-                gsay_message(849, 162, 50);
+                gsay_message(NAME, 162, NEUTRAL_REACTION);
             end
             else begin
                 if (LVar0 == 4) then begin
-                    gsay_message(849, 163, 50);
+                    gsay_message(NAME, 163, NEUTRAL_REACTION);
                 end
                 else begin
                     if (LVar0 == 5) then begin
-                        gsay_message(849, 167, 50);
+                        gsay_message(NAME, 167, NEUTRAL_REACTION);
                     end
                     else begin
                         if (LVar0 == 6) then begin
-                            gsay_message(849, 168, 50);
+                            gsay_message(NAME, 168, NEUTRAL_REACTION);
                         end
                         else begin
                             if (LVar0 == 7) then begin
-                                gsay_message(849, 169, 50);
+                                gsay_message(NAME, 169, NEUTRAL_REACTION);
                             end
                         end
                     end
@@ -625,34 +625,34 @@ end
 procedure FGTDriver21b
 begin
     variable LVar0 := 0;
-    set_global_var(201, 1);
+    set_global_var(GVAR_FARGO_TRADERS_STATUS, 1);
     LVar0 := random(1, 7);
     if (LVar0 == 1) then begin
-        gsay_message(849, 160, 50);
+        gsay_message(NAME, 160, NEUTRAL_REACTION);
     end
     else begin
         if (LVar0 == 2) then begin
-            gsay_message(849, 161, 50);
+            gsay_message(NAME, 161, NEUTRAL_REACTION);
         end
         else begin
             if (LVar0 == 3) then begin
-                gsay_message(849, 162, 50);
+                gsay_message(NAME, 162, NEUTRAL_REACTION);
             end
             else begin
                 if (LVar0 == 4) then begin
-                    gsay_message(849, 163, 50);
+                    gsay_message(NAME, 163, NEUTRAL_REACTION);
                 end
                 else begin
                     if (LVar0 == 5) then begin
-                        gsay_message(849, 164, 50);
+                        gsay_message(NAME, 164, NEUTRAL_REACTION);
                     end
                     else begin
                         if (LVar0 == 6) then begin
-                            gsay_message(849, 165, 50);
+                            gsay_message(NAME, 165, NEUTRAL_REACTION);
                         end
                         else begin
                             if (LVar0 == 7) then begin
-                                gsay_message(849, 166, 50);
+                                gsay_message(NAME, 166, NEUTRAL_REACTION);
                             end
                         end
                     end
@@ -672,38 +672,38 @@ end
 procedure FGTDriver21c
 begin
     variable LVar0 := 0;
-    set_global_var(201, 1);
+    set_global_var(GVAR_FARGO_TRADERS_STATUS, 1);
     LVar0 := random(1, 8);
     if (LVar0 == 1) then begin
-        gsay_message(849, 160, 50);
+        gsay_message(NAME, 160, NEUTRAL_REACTION);
     end
     else begin
         if (LVar0 == 2) then begin
-            gsay_message(849, 161, 50);
+            gsay_message(NAME, 161, NEUTRAL_REACTION);
         end
         else begin
             if (LVar0 == 3) then begin
-                gsay_message(849, 162, 50);
+                gsay_message(NAME, 162, NEUTRAL_REACTION);
             end
             else begin
                 if (LVar0 == 4) then begin
-                    gsay_message(849, 163, 50);
+                    gsay_message(NAME, 163, NEUTRAL_REACTION);
                 end
                 else begin
                     if (LVar0 == 5) then begin
-                        gsay_message(849, 216, 50);
+                        gsay_message(NAME, 216, NEUTRAL_REACTION);
                     end
                     else begin
                         if (LVar0 == 6) then begin
-                            gsay_message(849, 217, 50);
+                            gsay_message(NAME, 217, NEUTRAL_REACTION);
                         end
                         else begin
                             if (LVar0 == 7) then begin
-                                gsay_message(849, 218, 50);
+                                gsay_message(NAME, 218, NEUTRAL_REACTION);
                             end
                             else begin
                                 if (LVar0 == 8) then begin
-                                    gsay_message(849, 219, 50);
+                                    gsay_message(NAME, 219, NEUTRAL_REACTION);
                                 end
                             end
                         end
@@ -723,20 +723,20 @@ end
 
 procedure FGTDriver27
 begin
-    gsay_reply(849, 185);
-    giq_option(4, 849, 186, FGTDriver04a, 50);
-    giq_option(4, 849, 187, FGTDriver05, 50);
-    giq_option(4, 849, 188, FGTDriver06, 50);
-    giq_option(-3, 849, 189, FGTDriver07, 50);
-    giq_option(-3, 849, 190, FGTDriver07, 50);
+    Reply(185);
+    giq_option(4, NAME, 186, FGTDriver04a, NEUTRAL_REACTION);
+    giq_option(4, NAME, 187, FGTDriver05, NEUTRAL_REACTION);
+    giq_option(4, NAME, 188, FGTDriver06, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 189, FGTDriver07, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 190, FGTDriver07, NEUTRAL_REACTION);
 end
 
 procedure FGTDriver28
 begin
-    gsay_reply(849, 191);
-    giq_option(4, 849, 192, FGTDriver17, 50);
-    giq_option(4, 849, 193, FGTDriver35, 50);
-    giq_option(-3, 849, 194, FGTDriver07, 50);
+    Reply(191);
+    giq_option(4, NAME, 192, FGTDriver17, NEUTRAL_REACTION);
+    giq_option(4, NAME, 193, FGTDriver35, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 194, FGTDriver07, NEUTRAL_REACTION);
 end
 
 procedure FGTDriver29
@@ -744,7 +744,7 @@ begin
     variable LVar0 := 0;
     call RecalcDateString();
     LVar0 := local_var(5);
-    gsay_message(849, message_str(849, 196) + DayString + message_str(849, 203) + message_str(849, 195), 50);
+    gsay_message(NAME, mstr(196) + DayString + mstr(203) + mstr(195), NEUTRAL_REACTION);
 end
 
 procedure FGTDriver30
@@ -765,7 +765,7 @@ procedure FGTDriver33
 begin
     variable LVar0 := 0;
     set_local_var(10, 0);
-    set_global_var(206, 1);
+    set_global_var(GVAR_FARGO_JOB, 1);
     LVar0 := local_var(9);
     set_local_var(9, LVar0 + 1);
 end
@@ -785,31 +785,31 @@ begin
         set_local_var(0, 50);
         set_local_var(1, 2);
         set_local_var(2, 1);
-        set_local_var(0, local_var(0) + (5 * get_critter_stat(dude_obj, 3)) - 25);
-        set_local_var(0, local_var(0) + (10 * has_trait(0, dude_obj, 10)));
-        if (has_trait(0, dude_obj, 39)) then begin
-            if (global_var(155) > 0) then begin
-                set_local_var(0, local_var(0) + global_var(155));
+        set_local_var(0, local_var(0) + (5 * dude_charisma) - 25);
+        set_local_var(0, local_var(0) + (10 * has_trait(TRAIT_PERK, dude_obj, PERK_presence)));
+        if (has_trait(TRAIT_PERK, dude_obj, PERK_cult_of_personality)) then begin
+            if (global_var(GVAR_PLAYER_REPUATION) > 0) then begin
+                set_local_var(0, local_var(0) + global_var(GVAR_PLAYER_REPUATION));
             end
             else begin
-                set_local_var(0, local_var(0) - global_var(155));
+                set_local_var(0, local_var(0) - global_var(GVAR_PLAYER_REPUATION));
             end
         end
         else begin
             if (local_var(3) == 1) then begin
-                set_local_var(0, local_var(0) - global_var(155));
+                set_local_var(0, local_var(0) - global_var(GVAR_PLAYER_REPUATION));
             end
             else begin
-                set_local_var(0, local_var(0) + global_var(155));
+                set_local_var(0, local_var(0) + global_var(GVAR_PLAYER_REPUATION));
             end
         end
-        if (global_var(158) > 2) then begin
+        if (global_var(GVAR_CHILDKILLER_REPUATION) > 2) then begin
             set_local_var(0, local_var(0) - 30);
         end
-        if (((global_var(160) + global_var(159)) >= 25) and ((global_var(160) > (3 * global_var(159))) or (global_var(157) == 1))) then begin
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(160) > (3 * global_var(GVAR_GOOD_MONSTER))) or (global_var(GVAR_CHAMPION_REPUTATION) == 1))) then begin
             set_local_var(0, local_var(0) + 20);
         end
-        if (((global_var(160) + global_var(159)) >= 25) and ((global_var(159) > (2 * global_var(160))) or (global_var(156) == 1))) then begin
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
             set_local_var(0, local_var(0) - 20);
         end
         call ReactToLevel();
@@ -902,6 +902,5 @@ end
 
 procedure Goodbyes
 begin
-    exit_line := message_str(634, random(100, 105));
+    exit_line := message_str(SCRIPT_MODREACT, random(100, 105));
 end
-
