diff --git a/Games/Fallout1.Steam/Original/FIELDA.ssl b/Games/Fallout1.Steam/ReDefine/FIELDA.ssl
index 7a6c407..09bdfda 100644
--- a/Games/Fallout1.Steam/Original/FIELDA.ssl
+++ b/Games/Fallout1.Steam/ReDefine/FIELDA.ssl
@@ -37,7 +37,7 @@ begin
         initialized := 1;
     end
     else begin
-        if (script_action == 6) then begin
+        if (script_action == use_proc) then begin
             call use_p_proc();
         end
     end
@@ -45,7 +45,7 @@ end
 
 procedure use_p_proc
 begin
-    if (global_var(609)) then begin
+    if (global_var(GVAR_FORCE_FIELDS_OFF)) then begin
         field_change := "off";
     end
     if (source_obj != dude_obj) then begin
@@ -86,4 +86,3 @@ begin
         call turn_field_off();
     end
 end
-
