diff --git a/Games/Fallout1.Steam/Original/FIELDH.ssl b/Games/Fallout1.Steam/ReDefine/FIELDH.ssl
index 3d3f031..c4f179b 100644
--- a/Games/Fallout1.Steam/Original/FIELDH.ssl
+++ b/Games/Fallout1.Steam/ReDefine/FIELDH.ssl
@@ -22,7 +22,7 @@ begin
         initialized := 1;
     end
     else begin
-        if (script_action == 6) then begin
+        if (script_action == use_proc) then begin
             call use_p_proc();
         end
         else begin
@@ -30,7 +30,7 @@ begin
                 call map_enter_p_proc();
             end
             else begin
-                if (script_action == 23) then begin
+                if (script_action == map_update_proc) then begin
                     call map_update_p_proc();
                 end
             end
@@ -45,7 +45,7 @@ end
 
 procedure use_p_proc
 begin
-    if (global_var(609)) then begin
+    if (global_var(GVAR_FORCE_FIELDS_OFF)) then begin
         field_change := "off";
     end
     if (source_obj != dude_obj) then begin
@@ -96,4 +96,3 @@ begin
         call turn_field_off();
     end
 end
-
