diff --git a/Games/Fallout1.Steam/Original/FIELD34.ssl b/Games/Fallout1.Steam/ReDefine/FIELD34.ssl
index 192093a..b8c859e 100644
--- a/Games/Fallout1.Steam/Original/FIELD34.ssl
+++ b/Games/Fallout1.Steam/ReDefine/FIELD34.ssl
@@ -20,12 +20,12 @@ end
 procedure map_init_p_proc
 begin
     field3a_Ptr := self_obj;
-    if (global_var(146) == 1) then begin
+    if (global_var(GVAR_VATS_ALERT) == 1) then begin
         set_obj_visibility(self_obj, 0);
-        set_global_var(262, 1);
+        set_global_var(GVAR_WATER_CHIP_5, 1);
         set_map_var(18, 1);
     end
-    if ((global_var(262) == 0) or (map_var(18) == 0)) then begin
+    if ((global_var(GVAR_WATER_CHIP_5) == 0) or (map_var(18) == 0)) then begin
         set_obj_visibility(self_obj, 1);
     end
 end
@@ -33,15 +33,14 @@ end
 procedure map_update_p_proc
 begin
     field3a_Ptr := self_obj;
-    if ((global_var(262) == 0) or (map_var(18) == 0)) then begin
+    if ((global_var(GVAR_WATER_CHIP_5) == 0) or (map_var(18) == 0)) then begin
         set_obj_visibility(self_obj, 1);
     end
     else begin
-        if (global_var(146) == 1) then begin
+        if (global_var(GVAR_VATS_ALERT) == 1) then begin
             set_obj_visibility(self_obj, 0);
-            set_global_var(262, 1);
+            set_global_var(GVAR_WATER_CHIP_5, 1);
             set_map_var(18, 1);
         end
     end
 end
-
