diff --git a/Games/Fallout1.Steam/Original/FOLDN2UP.ssl b/Games/Fallout1.Steam/ReDefine/FOLDN2UP.ssl
index 63fb931..fea06ea 100644
--- a/Games/Fallout1.Steam/Original/FOLDN2UP.ssl
+++ b/Games/Fallout1.Steam/ReDefine/FOLDN2UP.ssl
@@ -4,7 +4,7 @@ procedure use_p_proc;
 
 procedure start
 begin
-    if (script_action == 6) then begin
+    if (script_action == use_proc) then begin
         call use_p_proc();
     end
 end
@@ -13,4 +13,3 @@ procedure use_p_proc
 begin
     move_to(dude_obj, 15268, 0);
 end
-
