diff --git a/Games/Fallout1.Steam/Original/FOLINVAD.ssl b/Games/Fallout1.Steam/ReDefine/FOLINVAD.ssl
index 1706b34..7beaef2 100644
--- a/Games/Fallout1.Steam/Original/FOLINVAD.ssl
+++ b/Games/Fallout1.Steam/ReDefine/FOLINVAD.ssl
@@ -14,11 +14,11 @@ begin
         call Initialize_p_proc();
     end
     else begin
-        if (script_action == 11) then begin
+        if (script_action == talk_proc) then begin
             call talk_p_proc();
         end
         else begin
-            if (script_action == 12) then begin
+            if (script_action == critter_proc) then begin
                 call critter_p_proc();
             end
             else begin
@@ -33,21 +33,21 @@ end
 procedure talk_p_proc
 begin
     if (local_var(3) == 0) then begin
-        float_msg(self_obj, message_str(820, random(100, 107)), 9);
+        float_msg(self_obj, mstr(random(100, 107)), FLOAT_MSG_WHITE);
     end
 end
 
 procedure Initialize_p_proc
 begin
-    critter_add_trait(self_obj, 1, 6, 0);
-    critter_add_trait(self_obj, 1, 5, 65);
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 0);
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_BROTHERHOOD_PALADIN);
     Only_Once := 0;
 end
 
 procedure critter_p_proc
 begin
     variable LVar0 := 0;
-    if (global_var(63) == 1) then begin
+    if (global_var(GVAR_FOLLOWER_MACHO_HELP) == 1) then begin
         if (local_var(4) == 0) then begin
             if (Hex_Number == 0) then begin
                 LVar0 := random(1, 10);
@@ -109,11 +109,11 @@ begin
         end
         else begin
             if ((tile_distance_objs(dude_obj, self_obj) > 10) and (tile_distance_objs(dude_obj, self_obj) < 15)) then begin
-                animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(dude_obj), random(0, 5), random(3, 6)), 0);
+                animate_move_obj_to_tile(self_obj, tile_num_in_direction(dude_tile, random(0, 5), random(3, 6)), 0);
             end
             else begin
                 if (tile_distance_objs(dude_obj, self_obj) >= 15) then begin
-                    animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(dude_obj), random(0, 5), random(3, 6)), 1);
+                    animate_move_obj_to_tile(self_obj, tile_num_in_direction(dude_tile, random(0, 5), random(3, 6)), 1);
                 end
                 else begin
                     if (random(1, 100) == 1) then begin
@@ -127,7 +127,7 @@ begin
                 if (local_var(0) == 0) then begin
                     set_local_var(0, 1);
                     party_add(self_obj);
-                    float_msg(self_obj, message_str(820, 112), 9);
+                    float_msg(self_obj, mstr(112), FLOAT_MSG_WHITE);
                 end
             end
             else begin
@@ -135,7 +135,7 @@ begin
                     if (local_var(1) == 0) then begin
                         if (elevation(self_obj) == 1) then begin
                             set_local_var(1, 1);
-                            float_msg(self_obj, message_str(820, 109), 9);
+                            float_msg(self_obj, mstr(109), FLOAT_MSG_WHITE);
                         end
                     end
                 end
@@ -143,16 +143,16 @@ begin
                     if (cur_map_index == 34) then begin
                         if (local_var(2) == 0) then begin
                             set_local_var(2, 1);
-                            float_msg(self_obj, message_str(820, 110), 9);
+                            float_msg(self_obj, mstr(110), FLOAT_MSG_WHITE);
                         end
                     end
                     else begin
                         if (local_var(3) == 0) then begin
                             set_local_var(3, 1);
-                            float_msg(self_obj, message_str(820, 111), 9);
+                            float_msg(self_obj, mstr(111), FLOAT_MSG_WHITE);
                             party_remove(self_obj);
                             animate_move_obj_to_tile(self_obj, 26000, 0);
-                            set_global_var(63, 5);
+                            set_global_var(GVAR_FOLLOWER_MACHO_HELP, 5);
                         end
                     end
                 end
@@ -166,8 +166,7 @@ end
 
 procedure map_enter_p_proc
 begin
-    if (global_var(63) != 1) then begin
+    if (global_var(GVAR_FOLLOWER_MACHO_HELP) != 1) then begin
         set_obj_visibility(self_obj, 1);
     end
 end
-
