diff --git a/Games/Fallout1.Steam/Original/FREEZER.ssl b/Games/Fallout1.Steam/ReDefine/FREEZER.ssl
index 37b08ae..9bfb2f6 100644
--- a/Games/Fallout1.Steam/Original/FREEZER.ssl
+++ b/Games/Fallout1.Steam/ReDefine/FREEZER.ssl
@@ -3,15 +3,14 @@ procedure start;
 
 procedure start
 begin
-    if (script_action == 21) then begin
+    if (script_action == look_at_proc) then begin
         script_overrides;
-        display_msg(message_str(314, 100));
+        display_mstr(100);
     end
     else begin
-        if (script_action == 3) then begin
+        if (script_action == description_proc) then begin
             script_overrides;
-            display_msg(message_str(314, 101));
+            display_mstr(101);
         end
     end
 end
-
