diff --git a/Games/Fallout1.Steam/Original/GENGAMBL.ssl b/Games/Fallout1.Steam/ReDefine/GENGAMBL.ssl
index 3298b18..2745ccb 100644
--- a/Games/Fallout1.Steam/Original/GENGAMBL.ssl
+++ b/Games/Fallout1.Steam/ReDefine/GENGAMBL.ssl
@@ -20,27 +20,27 @@ procedure pickup_p_proc;
 
 procedure start
 begin
-    if (script_action == 12) then begin
+    if (script_action == critter_proc) then begin
         call critter_p_proc();
     end
     else begin
-        if (script_action == 18) then begin
+        if (script_action == destroy_proc) then begin
             call destroy_p_proc();
         end
         else begin
-            if (script_action == 21) then begin
+            if (script_action == look_at_proc) then begin
                 call look_at_p_proc();
             end
             else begin
-                if (script_action == 15) then begin
+                if (script_action == map_enter_proc) then begin
                     call map_enter_p_proc();
                 end
                 else begin
-                    if (script_action == 11) then begin
+                    if (script_action == talk_proc) then begin
                         call talk_p_proc();
                     end
                     else begin
-                        if (script_action == 22) then begin
+                        if (script_action == timed_event_proc) then begin
                             call timed_event_p_proc();
                         end
                     end
@@ -52,7 +52,7 @@ end
 
 procedure critter_p_proc
 begin
-    if ((obj_item_subtype(critter_inven_obj(dude_obj, 1)) == 3) or (obj_item_subtype(critter_inven_obj(dude_obj, 2)) == 3)) then begin
+    if ((obj_item_subtype(critter_inven_obj(dude_obj, INVEN_TYPE_RIGHT_HAND)) == 3) or (obj_item_subtype(critter_inven_obj(dude_obj, INVEN_TYPE_LEFT_HAND)) == 3)) then begin
         if (obj_can_see_obj(self_obj, dude_obj)) then begin
             if (tile_distance_objs(self_obj, dude_obj) < 12) then begin
                 if (not(weapon_check)) then begin
@@ -63,15 +63,15 @@ begin
         end
     end
     if (hostile) then begin
-        attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+        attack(dude_obj);
     end
-    if (global_var(247) == 1) then begin
+    if (global_var(GVAR_ENEMY_JUNKTOWN) == 1) then begin
         if (obj_can_see_obj(self_obj, dude_obj)) then begin
-            attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+            attack(dude_obj);
         end
     end
     if (cur_map_index == 11) then begin
-        if ((global_var(104) == 2) or (global_var(38) == 1)) then begin
+        if ((global_var(GVAR_CAPTURE_GIZMO) == 2) or (global_var(GVAR_GIZMO_DEAD) == 1)) then begin
             destroy_object(self_obj);
         end
     end
@@ -80,19 +80,19 @@ end
 procedure destroy_p_proc
 begin
     if (source_obj == dude_obj) then begin
-        set_global_var(247, 1);
+        set_global_var(GVAR_ENEMY_JUNKTOWN, 1);
         if (source_obj == dude_obj) then begin
-            if (((global_var(160) + global_var(159)) >= 25) and ((global_var(159) > (2 * global_var(160))) or (global_var(156) == 1))) then begin
-                set_global_var(156, 1);
-                set_global_var(157, 0);
+            if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
+                set_global_var(GVAR_BERSERKER_REPUTATION, 1);
+                set_global_var(GVAR_CHAMPION_REPUTATION, 0);
             end
-            if (((global_var(160) + global_var(159)) >= 25) and ((global_var(160) > (3 * global_var(159))) or (global_var(157) == 1))) then begin
-                set_global_var(157, 1);
-                set_global_var(156, 0);
+            if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(160) > (3 * global_var(GVAR_GOOD_MONSTER))) or (global_var(GVAR_CHAMPION_REPUTATION) == 1))) then begin
+                set_global_var(GVAR_CHAMPION_REPUTATION, 1);
+                set_global_var(GVAR_BERSERKER_REPUTATION, 0);
             end
-            set_global_var(159, global_var(159) + 1);
-            if ((global_var(159) % 2) == 0) then begin
-                set_global_var(155, global_var(155) - 1);
+            set_global_var(GVAR_GOOD_MONSTER, global_var(GVAR_GOOD_MONSTER) + 1);
+            if ((global_var(GVAR_GOOD_MONSTER) % 2) == 0) then begin
+                set_global_var(GVAR_PLAYER_REPUATION, global_var(GVAR_PLAYER_REPUATION) - 1);
             end
         end
     end
@@ -101,7 +101,7 @@ end
 procedure look_at_p_proc
 begin
     script_overrides;
-    display_msg(message_str(351, 100));
+    display_mstr(100);
 end
 
 procedure map_enter_p_proc
@@ -113,8 +113,8 @@ begin
         item_caps_adjust(self_obj, random(0, 200));
     end
     if (cur_map_index == 11) then begin
-        critter_add_trait(self_obj, 1, 6, 24);
-        if ((global_var(104) == 2) or (global_var(38) == 1)) then begin
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 24);
+        if ((global_var(GVAR_CAPTURE_GIZMO) == 2) or (global_var(GVAR_GIZMO_DEAD) == 1)) then begin
             destroy_object(self_obj);
         end
     end
@@ -123,19 +123,19 @@ end
 procedure talk_p_proc
 begin
     script_overrides;
-    if ((obj_item_subtype(critter_inven_obj(dude_obj, 1)) == 3) or (obj_item_subtype(critter_inven_obj(dude_obj, 2)) == 3)) then begin
+    if ((obj_item_subtype(critter_inven_obj(dude_obj, INVEN_TYPE_RIGHT_HAND)) == 3) or (obj_item_subtype(critter_inven_obj(dude_obj, INVEN_TYPE_LEFT_HAND)) == 3)) then begin
         call gambler1();
     end
     else begin
-        if (global_var(158) > 2) then begin
+        if (global_var(GVAR_CHILDKILLER_REPUATION) > 2) then begin
             call gambler3();
         end
         else begin
-            if (global_var(105) == 2) then begin
+            if (global_var(GVAR_KILL_KILLIAN) == 2) then begin
                 call gambler4();
             end
             else begin
-                if (global_var(155) < -10) then begin
+                if (global_var(GVAR_PLAYER_REPUATION) < -10) then begin
                     call gambler2();
                 end
                 else begin
@@ -154,7 +154,7 @@ end
 procedure damage_p_proc
 begin
     if (source_obj == dude_obj) then begin
-        set_global_var(247, 1);
+        set_global_var(GVAR_ENEMY_JUNKTOWN, 1);
     end
 end
 
@@ -164,19 +164,19 @@ begin
     weapon_check := 1;
     LVar0 := random(1, 3);
     if (LVar0 == 1) then begin
-        float_msg(self_obj, message_str(351, 101), 0);
+        float_msg(self_obj, mstr(101), FLOAT_MSG_NORMAL);
     end
     else begin
         if (LVar0 == 2) then begin
-            float_msg(self_obj, message_str(351, 102), 0);
+            float_msg(self_obj, mstr(102), FLOAT_MSG_NORMAL);
         end
         else begin
             if (LVar0 == 3) then begin
-                if (get_critter_stat(dude_obj, 34) == 0) then begin
-                    float_msg(self_obj, message_str(351, 103), 0);
+                if (dude_gender == 0) then begin
+                    float_msg(self_obj, mstr(103), FLOAT_MSG_NORMAL);
                 end
                 else begin
-                    float_msg(self_obj, message_str(351, 104), 0);
+                    float_msg(self_obj, mstr(104), FLOAT_MSG_NORMAL);
                 end
             end
         end
@@ -185,17 +185,17 @@ end
 
 procedure gambler2
 begin
-    float_msg(self_obj, message_str(351, random(105, 107)), 0);
+    float_msg(self_obj, mstr(random(105, 107)), FLOAT_MSG_NORMAL);
 end
 
 procedure gambler3
 begin
-    float_msg(self_obj, message_str(351, random(108, 110)), 0);
+    float_msg(self_obj, mstr(random(108, 110)), FLOAT_MSG_NORMAL);
 end
 
 procedure gambler4
 begin
-    float_msg(self_obj, message_str(351, random(111, 113)), 0);
+    float_msg(self_obj, mstr(random(111, 113)), FLOAT_MSG_NORMAL);
 end
 
 procedure gambler5
@@ -204,14 +204,14 @@ begin
     LVar0 := random(1, 6);
     if (LVar0 == 1) then begin
         if ((game_time_hour >= 700) and (game_time_hour < 1800)) then begin
-            float_msg(self_obj, message_str(351, 114), 0);
+            float_msg(self_obj, mstr(114), FLOAT_MSG_NORMAL);
         end
         else begin
-            float_msg(self_obj, message_str(351, 115), 0);
+            float_msg(self_obj, mstr(115), FLOAT_MSG_NORMAL);
         end
     end
     else begin
-        float_msg(self_obj, message_str(351, 114 + LVar0), 0);
+        float_msg(self_obj, mstr(114 + LVar0), FLOAT_MSG_NORMAL);
     end
 end
 
@@ -219,4 +219,3 @@ procedure pickup_p_proc
 begin
     hostile := 1;
 end
-
