diff --git a/Games/Fallout1.Steam/Original/GENMERCA.ssl b/Games/Fallout1.Steam/ReDefine/GENMERCA.ssl
index 56158ec..abbe5ea 100644
--- a/Games/Fallout1.Steam/Original/GENMERCA.ssl
+++ b/Games/Fallout1.Steam/ReDefine/GENMERCA.ssl
@@ -21,27 +21,27 @@ variable job_index;
 procedure start
 begin
     if (not(initialized)) then begin
-        critter_add_trait(self_obj, 1, 6, 2);
-        critter_add_trait(self_obj, 1, 5, 50);
-        name_index := 105 + global_var(335);
-        job_index := 109 + global_var(335);
-        hostile := global_var(334);
-        set_global_var(335, global_var(335) + 1);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 2);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_MERCHANT);
+        name_index := 105 + global_var(GVAR_SAVE_TANDI_6);
+        job_index := 109 + global_var(GVAR_SAVE_TANDI_6);
+        hostile := global_var(GVAR_SAVE_TANDI_5);
+        set_global_var(GVAR_SAVE_TANDI_6, global_var(GVAR_SAVE_TANDI_6) + 1);
         initialized := 1;
     end
-    if (script_action == 12) then begin
+    if (script_action == critter_proc) then begin
         call critter_p_proc();
     end
     else begin
-        if (script_action == 18) then begin
+        if (script_action == destroy_proc) then begin
             call destroy_p_proc();
         end
         else begin
-            if (script_action == 4) then begin
+            if (script_action == pickup_proc) then begin
                 call pickup_p_proc();
             end
             else begin
-                if (script_action == 11) then begin
+                if (script_action == talk_proc) then begin
                     call talk_p_proc();
                 end
             end
@@ -58,7 +58,7 @@ begin
         if (hostile) then begin
             hostile := 0;
             scared := 1;
-            attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+            attack(dude_obj);
         end
     end
 end
@@ -66,19 +66,19 @@ end
 procedure destroy_p_proc
 begin
     if (source_obj == dude_obj) then begin
-        set_global_var(334, 1);
+        set_global_var(GVAR_SAVE_TANDI_5, 1);
         if (source_obj == dude_obj) then begin
-            if (((global_var(160) + global_var(159)) >= 25) and ((global_var(159) > (2 * global_var(160))) or (global_var(156) == 1))) then begin
-                set_global_var(156, 1);
-                set_global_var(157, 0);
+            if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
+                set_global_var(GVAR_BERSERKER_REPUTATION, 1);
+                set_global_var(GVAR_CHAMPION_REPUTATION, 0);
             end
-            if (((global_var(160) + global_var(159)) >= 25) and ((global_var(160) > (3 * global_var(159))) or (global_var(157) == 1))) then begin
-                set_global_var(157, 1);
-                set_global_var(156, 0);
+            if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(160) > (3 * global_var(GVAR_GOOD_MONSTER))) or (global_var(GVAR_CHAMPION_REPUTATION) == 1))) then begin
+                set_global_var(GVAR_CHAMPION_REPUTATION, 1);
+                set_global_var(GVAR_BERSERKER_REPUTATION, 0);
             end
-            set_global_var(159, global_var(159) + 1);
-            if ((global_var(159) % 2) == 0) then begin
-                set_global_var(155, global_var(155) - 1);
+            set_global_var(GVAR_GOOD_MONSTER, global_var(GVAR_GOOD_MONSTER) + 1);
+            if ((global_var(GVAR_GOOD_MONSTER) % 2) == 0) then begin
+                set_global_var(GVAR_PLAYER_REPUATION, global_var(GVAR_PLAYER_REPUATION) - 1);
             end
         end
     end
@@ -88,18 +88,18 @@ procedure pickup_p_proc
 begin
     if (scared == 0) then begin
         hostile := 1;
-        set_global_var(334, 1);
+        set_global_var(GVAR_SAVE_TANDI_5, 1);
     end
 end
 
 procedure talk_p_proc
 begin
-    if (((global_var(160) + global_var(159)) >= 25) and ((global_var(159) > (2 * global_var(160))) or (global_var(156) == 1))) then begin
-        set_global_var(156, 1);
-        set_global_var(157, 0);
+    if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
+        set_global_var(GVAR_BERSERKER_REPUTATION, 1);
+        set_global_var(GVAR_CHAMPION_REPUTATION, 0);
     end
     if (scared) then begin
-        float_msg(self_obj, message_str(669, random(100, 105)), 2);
+        float_msg(self_obj, message_str(SCRIPT_ENEMY, random(100, 105)), FLOAT_MSG_RED);
     end
     else begin
         start_gdialog(752, self_obj, 4, -1, -1);
@@ -112,47 +112,47 @@ end
 
 procedure GenMercA00
 begin
-    gsay_reply(752, 100);
-    giq_option(4, 752, 101, GenMercA01, 50);
-    giq_option(4, 752, 102, GenMercA02, 50);
-    if (((global_var(160) + global_var(159)) >= 25) and ((global_var(159) > (2 * global_var(160))) or (global_var(156) == 1))) then begin
-        giq_option(4, 752, 103, GenMercA03, 51);
+    Reply(100);
+    giq_option(4, NAME, 101, GenMercA01, NEUTRAL_REACTION);
+    giq_option(4, NAME, 102, GenMercA02, NEUTRAL_REACTION);
+    if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
+        giq_option(4, NAME, 103, GenMercA03, BAD_REACTION);
     end
-    giq_option(-3, 752, 104, GenMercAEnd, 50);
-    giq_option(4, 634, 103, GenMercAEnd, 50);
+    giq_option(-3, NAME, 104, GenMercAEnd, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_MODREACT, 103, GenMercAEnd, NEUTRAL_REACTION);
 end
 
 procedure GenMercA01
 begin
-    gsay_reply(752, name_index);
-    giq_option(4, 752, 102, GenMercA02, 50);
-    giq_option(4, 634, 100, GenMercAEnd, 50);
+    Reply(name_index);
+    giq_option(4, NAME, 102, GenMercA02, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_MODREACT, 100, GenMercAEnd, NEUTRAL_REACTION);
 end
 
 procedure GenMercA02
 begin
-    gsay_reply(752, job_index);
-    giq_option(4, 752, 114, GenMercA04, 50);
-    giq_option(4, 634, 100, GenMercAEnd, 50);
+    Reply(job_index);
+    giq_option(4, NAME, 114, GenMercA04, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_MODREACT, 100, GenMercAEnd, NEUTRAL_REACTION);
 end
 
 procedure GenMercA03
 begin
     scared := 1;
-    gsay_message(752, 113, 51);
+    gsay_message(NAME, 113, BAD_REACTION);
 end
 
 procedure GenMercA04
 begin
-    if (((global_var(160) + global_var(159)) >= 25) and ((global_var(159) > (2 * global_var(160))) or (global_var(156) == 1)) or (global_var(158) > 2) or (global_var(155) < -20)) then begin
-        gsay_message(752, 117, 51);
+    if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1)) or (global_var(GVAR_CHILDKILLER_REPUATION) > 2) or (global_var(GVAR_PLAYER_REPUATION) < -20)) then begin
+        gsay_message(NAME, 117, BAD_REACTION);
     end
     else begin
         if (is_success(roll_vs_skill(dude_obj, 14, 0))) then begin
-            gsay_message(752, 116, 50);
+            gsay_message(NAME, 116, NEUTRAL_REACTION);
         end
         else begin
-            gsay_message(752, 116, 50);
+            gsay_message(NAME, 116, NEUTRAL_REACTION);
         end
     end
 end
@@ -167,12 +167,11 @@ begin
     variable LVar1 := 0;
     variable LVar2 := 0;
     while (LVar1 < 5) do begin
-        if (tile_distance(tile_num(dude_obj), tile_num_in_direction(tile_num(self_obj), LVar1, 3)) > LVar2) then begin
+        if (tile_distance(dude_tile, tile_num_in_direction(tile_num(self_obj), LVar1, 3)) > LVar2) then begin
             LVar0 := tile_num_in_direction(tile_num(self_obj), LVar1, 3);
-            LVar2 := tile_distance(tile_num(dude_obj), LVar0);
+            LVar2 := tile_distance(dude_tile, LVar0);
         end
         LVar1 := LVar1 + 1;
     end
     animate_move_obj_to_tile(self_obj, LVar0, 1);
 end
-
