diff --git a/Games/Fallout1.Steam/Original/GENSKULZ.ssl b/Games/Fallout1.Steam/ReDefine/GENSKULZ.ssl
index b61beb5..7a51753 100644
--- a/Games/Fallout1.Steam/Original/GENSKULZ.ssl
+++ b/Games/Fallout1.Steam/ReDefine/GENSKULZ.ssl
@@ -27,35 +27,35 @@ import variable Trish_ptr;
 
 procedure start
 begin
-    if (script_action == 12) then begin
+    if (script_action == critter_proc) then begin
         call critter_p_proc();
     end
     else begin
-        if (script_action == 18) then begin
+        if (script_action == destroy_proc) then begin
             call destroy_p_proc();
         end
         else begin
-            if (script_action == 21) then begin
+            if (script_action == look_at_proc) then begin
                 call look_at_p_proc();
             end
             else begin
-                if (script_action == 15) then begin
+                if (script_action == map_enter_proc) then begin
                     call map_enter_p_proc();
                 end
                 else begin
-                    if (script_action == 16) then begin
+                    if (script_action == map_exit_proc) then begin
                         call map_exit_p_proc();
                     end
                     else begin
-                        if (script_action == 4) then begin
+                        if (script_action == pickup_proc) then begin
                             call pickup_p_proc();
                         end
                         else begin
-                            if (script_action == 11) then begin
+                            if (script_action == talk_proc) then begin
                                 call talk_p_proc();
                             end
                             else begin
-                                if (script_action == 22) then begin
+                                if (script_action == timed_event_proc) then begin
                                     call timed_event_p_proc();
                                 end
                             end
@@ -71,7 +71,7 @@ procedure critter_p_proc
 begin
     if (hostile) then begin
         hostile := 0;
-        attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+        attack(dude_obj);
     end
     else begin
         if (cur_map_index == 11) then begin
@@ -81,7 +81,7 @@ begin
                 sleeping_disabled := 1;
             end
             else begin
-                if ((global_var(282) == 1) and (waypoint == 0)) then begin
+                if ((global_var(GVAR_DESTROY_VATS_10) == 1) and (waypoint == 0)) then begin
                     waypoint := 1;
                     sleeping_disabled := 1;
                     add_timer_event(self_obj, game_ticks(8), 3);
@@ -109,7 +109,7 @@ begin
         end
         else begin
             if (cur_map_index == 12) then begin
-                if ((global_var(555) != 2) and (sleeping_disabled == 0)) then begin
+                if ((global_var(GVAR_BUST_SKULZ) != 2) and (sleeping_disabled == 0)) then begin
                     call sleeping();
                 end
             end
@@ -120,17 +120,17 @@ end
 procedure destroy_p_proc
 begin
     if (source_obj == dude_obj) then begin
-        if (((global_var(160) + global_var(159)) >= 25) and ((global_var(159) > (2 * global_var(160))) or (global_var(156) == 1))) then begin
-            set_global_var(156, 1);
-            set_global_var(157, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
+            set_global_var(GVAR_BERSERKER_REPUTATION, 1);
+            set_global_var(GVAR_CHAMPION_REPUTATION, 0);
         end
-        if (((global_var(160) + global_var(159)) >= 25) and ((global_var(160) > (3 * global_var(159))) or (global_var(157) == 1))) then begin
-            set_global_var(157, 1);
-            set_global_var(156, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(160) > (3 * global_var(GVAR_GOOD_MONSTER))) or (global_var(GVAR_CHAMPION_REPUTATION) == 1))) then begin
+            set_global_var(GVAR_CHAMPION_REPUTATION, 1);
+            set_global_var(GVAR_BERSERKER_REPUTATION, 0);
         end
-        set_global_var(160, global_var(160) + 1);
-        if ((global_var(160) % 6) == 0) then begin
-            set_global_var(155, global_var(155) + 1);
+        set_global_var(GVAR_BAD_MONSTER, global_var(GVAR_BAD_MONSTER) + 1);
+        if ((global_var(GVAR_BAD_MONSTER) % 6) == 0) then begin
+            set_global_var(GVAR_PLAYER_REPUATION, global_var(GVAR_PLAYER_REPUATION) + 1);
         end
     end
     if (home_tile == 20082) then begin
@@ -144,7 +144,7 @@ end
 procedure look_at_p_proc
 begin
     script_overrides;
-    display_msg(message_str(390, 100));
+    display_mstr(100);
 end
 
 procedure map_enter_p_proc
@@ -154,7 +154,7 @@ begin
     end
     home_tile := local_var(5);
     if (cur_map_index == 10) then begin
-        if (global_var(555) == 2) then begin
+        if (global_var(GVAR_BUST_SKULZ) == 2) then begin
             set_obj_visibility(self_obj, 0);
         end
         else begin
@@ -163,7 +163,7 @@ begin
     end
     if (cur_map_index != 11) then begin
         if (cur_map_index == 10) then begin
-            if (global_var(555) == 2) then begin
+            if (global_var(GVAR_BUST_SKULZ) == 2) then begin
                 add_timer_event(self_obj, game_ticks(random(30, 60)), 1);
             end
         end
@@ -174,13 +174,13 @@ begin
         end
     end
     else begin
-        if ((global_var(282) == 1) and (map_var(2) == 0)) then begin
+        if ((global_var(GVAR_DESTROY_VATS_10) == 1) and (map_var(2) == 0)) then begin
             destroy_object(self_obj);
         end
     end
-    critter_add_trait(self_obj, 1, 6, 14);
-    critter_add_trait(self_obj, 1, 5, 59);
-    if (global_var(555) == 2) then begin
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 14);
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_SKULZ);
+    if (global_var(GVAR_BUST_SKULZ) == 2) then begin
         if (cur_map_index == 10) then begin
             set_obj_visibility(self_obj, 0);
         end
@@ -197,7 +197,7 @@ end
 procedure map_exit_p_proc
 begin
     if (cur_map_index == 11) then begin
-        if ((global_var(282) == 1) and (map_var(2) == 0)) then begin
+        if ((global_var(GVAR_DESTROY_VATS_10) == 1) and (map_var(2) == 0)) then begin
             destroy_object(self_obj);
         end
     end
@@ -205,7 +205,7 @@ end
 
 procedure pickup_p_proc
 begin
-    float_msg(self_obj, message_str(390, 101), 2);
+    float_msg(self_obj, mstr(101), FLOAT_MSG_RED);
     hostile := 1;
 end
 
@@ -225,11 +225,11 @@ begin
             end
         end
     end
-    LVar0 := message_str(390, local_var(6));
+    LVar0 := mstr(local_var(6));
     if (local_var(4) == 1) then begin
-        LVar0 := message_str(390, 106);
+        LVar0 := mstr(106);
     end
-    float_msg(self_obj, LVar0, 0);
+    float_msg(self_obj, LVar0, FLOAT_MSG_NORMAL);
 end
 
 procedure timed_event_p_proc
@@ -237,7 +237,7 @@ begin
     if (fixed_param == 1) then begin
         if ((local_var(4) == 0) and (combat_is_initialized == 0)) then begin
             if (random(0, 1)) then begin
-                float_msg(self_obj, message_str(390, random(107, 112)), 0);
+                float_msg(self_obj, mstr(random(107, 112)), FLOAT_MSG_NORMAL);
             end
             else begin
                 animate_move_obj_to_tile(self_obj, tile_num_in_direction(local_var(5), random(0, 3), random(0, 5)), 0);
@@ -247,7 +247,7 @@ begin
     end
     else begin
         if ((fixed_param == 2) and (Trish_ptr != 0) and (Neal_ptr != 0)) then begin
-            float_msg(self_obj, message_str(387, 102), 2);
+            float_msg(self_obj, message_str(SCRIPT_SHARK, 102), FLOAT_MSG_RED);
             reg_anim_func(2, self_obj);
             reg_anim_func(1, 1);
             reg_anim_obj_move_to_obj(self_obj, Trish_ptr, -1);
@@ -345,4 +345,3 @@ begin
         end
     end
 end
-
