diff --git a/Games/Fallout1.Steam/Original/GIZGUARD.ssl b/Games/Fallout1.Steam/ReDefine/GIZGUARD.ssl
index c298baa..593cf70 100644
--- a/Games/Fallout1.Steam/Original/GIZGUARD.ssl
+++ b/Games/Fallout1.Steam/ReDefine/GIZGUARD.ssl
@@ -27,27 +27,27 @@ import variable weapon_checked;
 
 procedure start
 begin
-    if (script_action == 12) then begin
+    if (script_action == critter_proc) then begin
         call critter_p_proc();
     end
     else begin
-        if (script_action == 18) then begin
+        if (script_action == destroy_proc) then begin
             call destroy_p_proc();
         end
         else begin
-            if (script_action == 15) then begin
+            if (script_action == map_enter_proc) then begin
                 call map_enter_p_proc();
             end
             else begin
-                if (script_action == 4) then begin
+                if (script_action == pickup_proc) then begin
                     call pickup_p_proc();
                 end
                 else begin
-                    if (script_action == 11) then begin
+                    if (script_action == talk_proc) then begin
                         call talk_p_proc();
                     end
                     else begin
-                        if (script_action == 22) then begin
+                        if (script_action == timed_event_proc) then begin
                             call timed_event_p_proc();
                         end
                     end
@@ -62,13 +62,13 @@ begin
     if (hostile) then begin
         if (obj_can_see_obj(self_obj, dude_obj)) then begin
             hostile := 0;
-            attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+            attack(dude_obj);
         end
     end
     else begin
         if (obj_can_see_obj(self_obj, dude_obj)) then begin
             if (tile_distance_objs(self_obj, dude_obj) < 12) then begin
-                if ((obj_item_subtype(critter_inven_obj(dude_obj, 1)) == 3) or (obj_item_subtype(critter_inven_obj(dude_obj, 2)) == 3) and not(weapon_checked)) then begin
+                if ((obj_item_subtype(critter_inven_obj(dude_obj, INVEN_TYPE_RIGHT_HAND)) == 3) or (obj_item_subtype(critter_inven_obj(dude_obj, INVEN_TYPE_LEFT_HAND)) == 3) and not(weapon_checked)) then begin
                     dialogue_system_enter;
                 end
                 else begin
@@ -79,9 +79,9 @@ begin
             end
         end
     end
-    if (global_var(347) == 1) then begin
+    if (global_var(GVAR_CURE_JARVIS_8) == 1) then begin
         if (obj_can_see_obj(self_obj, dude_obj)) then begin
-            attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+            attack(dude_obj);
         end
     end
 end
@@ -89,19 +89,19 @@ end
 procedure destroy_p_proc
 begin
     if (source_obj == dude_obj) then begin
-        set_global_var(347, 1);
+        set_global_var(GVAR_CURE_JARVIS_8, 1);
         if (source_obj == dude_obj) then begin
-            if (((global_var(160) + global_var(159)) >= 25) and ((global_var(159) > (2 * global_var(160))) or (global_var(156) == 1))) then begin
-                set_global_var(156, 1);
-                set_global_var(157, 0);
+            if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
+                set_global_var(GVAR_BERSERKER_REPUTATION, 1);
+                set_global_var(GVAR_CHAMPION_REPUTATION, 0);
             end
-            if (((global_var(160) + global_var(159)) >= 25) and ((global_var(160) > (3 * global_var(159))) or (global_var(157) == 1))) then begin
-                set_global_var(157, 1);
-                set_global_var(156, 0);
+            if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(160) > (3 * global_var(GVAR_GOOD_MONSTER))) or (global_var(GVAR_CHAMPION_REPUTATION) == 1))) then begin
+                set_global_var(GVAR_CHAMPION_REPUTATION, 1);
+                set_global_var(GVAR_BERSERKER_REPUTATION, 0);
             end
-            set_global_var(160, global_var(160) + 1);
-            if ((global_var(160) % 6) == 0) then begin
-                set_global_var(155, global_var(155) + 1);
+            set_global_var(GVAR_BAD_MONSTER, global_var(GVAR_BAD_MONSTER) + 1);
+            if ((global_var(GVAR_BAD_MONSTER) % 6) == 0) then begin
+                set_global_var(GVAR_PLAYER_REPUATION, global_var(GVAR_PLAYER_REPUATION) + 1);
             end
         end
     end
@@ -112,13 +112,13 @@ begin
     if (not(obj_is_carrying_obj_pid(self_obj, 41))) then begin
         item_caps_adjust(self_obj, random(0, 10));
     end
-    if (global_var(32) == 4) then begin
-        critter_add_trait(self_obj, 1, 6, 87);
+    if (global_var(GVAR_LOAD_MAP_INDEX) == 4) then begin
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 87);
     end
     else begin
-        critter_add_trait(self_obj, 1, 6, 13);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 13);
     end
-    if ((global_var(38) == 1) or (global_var(104) == 2)) then begin
+    if ((global_var(GVAR_GIZMO_DEAD) == 1) or (global_var(GVAR_CAPTURE_GIZMO) == 2)) then begin
         set_obj_visibility(self_obj, 1);
         move_to(self_obj, 7000, 0);
         removal_ptr := self_obj;
@@ -132,7 +132,7 @@ end
 
 procedure talk_p_proc
 begin
-    if ((obj_item_subtype(critter_inven_obj(dude_obj, 1)) == 3) or (obj_item_subtype(critter_inven_obj(dude_obj, 2)) == 3)) then begin
+    if ((obj_item_subtype(critter_inven_obj(dude_obj, INVEN_TYPE_RIGHT_HAND)) == 3) or (obj_item_subtype(critter_inven_obj(dude_obj, INVEN_TYPE_LEFT_HAND)) == 3)) then begin
         call GizGuard00();
     end
     else begin
@@ -157,8 +157,8 @@ end
 procedure timed_event_p_proc
 begin
     if (fixed_param == 1) then begin
-        if ((obj_item_subtype(critter_inven_obj(dude_obj, 1)) == 3) or (obj_item_subtype(critter_inven_obj(dude_obj, 2)) == 3)) then begin
-            float_msg(self_obj, message_str(639, random(100, 103)), 2);
+        if ((obj_item_subtype(critter_inven_obj(dude_obj, INVEN_TYPE_RIGHT_HAND)) == 3) or (obj_item_subtype(critter_inven_obj(dude_obj, INVEN_TYPE_LEFT_HAND)) == 3)) then begin
+            float_msg(self_obj, mstr(random(100, 103)), FLOAT_MSG_RED);
             hostile := 1;
         end
         else begin
@@ -180,20 +180,20 @@ end
 procedure damage_p_proc
 begin
     if (source_obj == dude_obj) then begin
-        set_global_var(347, 1);
+        set_global_var(GVAR_CURE_JARVIS_8, 1);
     end
 end
 
 procedure GizGuard00
 begin
-    float_msg(self_obj, message_str(639, random(132, 133)), 2);
+    float_msg(self_obj, mstr(random(132, 133)), FLOAT_MSG_RED);
     weapon_checked := 1;
     add_timer_event(self_obj, game_ticks(5), 1);
 end
 
 procedure GizGuard01
 begin
-    float_msg(self_obj, message_str(639, random(104, 106)), 2);
+    float_msg(self_obj, mstr(random(104, 106)), FLOAT_MSG_RED);
     sneak_checked := 1;
     add_timer_event(self_obj, game_ticks(5), 2);
 end
@@ -201,51 +201,50 @@ end
 procedure GizGuard02
 begin
     set_local_var(0, 1);
-    gsay_reply(639, 107);
-    giq_option(4, 639, 108, GizGuard03, 50);
-    giq_option(4, 639, 109, GizGuard04, 50);
-    giq_option(4, 639, 110, GizGuardEnd, 50);
-    giq_option(-3, 639, 111, GizGuard08, 51);
+    Reply(107);
+    giq_option(4, NAME, 108, GizGuard03, NEUTRAL_REACTION);
+    giq_option(4, NAME, 109, GizGuard04, NEUTRAL_REACTION);
+    giq_option(4, NAME, 110, GizGuardEnd, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 111, GizGuard08, BAD_REACTION);
 end
 
 procedure GizGuard03
 begin
     if (is_success(roll_vs_skill(dude_obj, 14, 0))) then begin
-        gsay_reply(639, 112);
-        giq_option(4, 639, 114, GizGuard05, 50);
-        giq_option(4, 639, 115, GizGuard06, 50);
+        Reply(112);
+        giq_option(4, NAME, 114, GizGuard05, NEUTRAL_REACTION);
+        giq_option(4, NAME, 115, GizGuard06, NEUTRAL_REACTION);
     end
     else begin
-        gsay_message(639, 113, 51);
+        gsay_message(NAME, 113, BAD_REACTION);
     end
 end
 
 procedure GizGuard04
 begin
-    gsay_message(639, random(116, 119), 50);
+    gsay_message(NAME, random(116, 119), NEUTRAL_REACTION);
 end
 
 procedure GizGuard05
 begin
-    gsay_message(639, 120, 50);
+    gsay_message(NAME, 120, NEUTRAL_REACTION);
 end
 
 procedure GizGuard06
 begin
-    gsay_message(639, random(121, 124), 50);
+    gsay_message(NAME, random(121, 124), NEUTRAL_REACTION);
 end
 
 procedure GizGuard07
 begin
-    float_msg(self_obj, message_str(639, random(125, 130)), 0);
+    float_msg(self_obj, mstr(random(125, 130)), FLOAT_MSG_NORMAL);
 end
 
 procedure GizGuard08
 begin
-    gsay_message(639, 131, 50);
+    gsay_message(NAME, 131, NEUTRAL_REACTION);
 end
 
 procedure GizGuardEnd
 begin
 end
-
