diff --git a/Games/Fallout1.Steam/Original/GLOWGN.ssl b/Games/Fallout1.Steam/ReDefine/GLOWGN.ssl
index 2f98f73..24de181 100644
--- a/Games/Fallout1.Steam/Original/GLOWGN.ssl
+++ b/Games/Fallout1.Steam/ReDefine/GLOWGN.ssl
@@ -26,35 +26,35 @@ variable Per;
 
 procedure start
 begin
-    if (script_action == 6) then begin
+    if (script_action == use_proc) then begin
         call use_p_proc();
     end
     else begin
-        if (script_action == 15) then begin
+        if (script_action == map_enter_proc) then begin
             call map_enter_p_proc();
         end
         else begin
-            if (script_action == 23) then begin
+            if (script_action == map_update_proc) then begin
                 call map_update_p_proc();
             end
             else begin
-                if (script_action == 7) then begin
+                if (script_action == use_obj_on_proc) then begin
                     call use_obj_on_p_proc();
                 end
                 else begin
-                    if (script_action == 21) then begin
+                    if (script_action == look_at_proc) then begin
                         call look_at_p_proc();
                     end
                     else begin
-                        if (script_action == 3) then begin
+                        if (script_action == description_proc) then begin
                             call description_p_proc();
                         end
                         else begin
-                            if (script_action == 8) then begin
+                            if (script_action == use_skill_on_proc) then begin
                                 call use_skill_on_p_proc();
                             end
                             else begin
-                                if (script_action == 14) then begin
+                                if (script_action == damage_proc) then begin
                                     call damage_p_proc();
                                 end
                             end
@@ -73,7 +73,7 @@ begin
     if (local_var(2) == 0) then begin
         if (is_success(LVar0)) then begin
             script_overrides;
-            display_msg(message_str(306, 204));
+            display_msg(message_str(SCRIPT_GLOBLDOR, 204));
             reg_anim_func(2, source_obj);
         end
         else begin
@@ -92,7 +92,7 @@ begin
     else begin
         if (local_var(3) == 0) then begin
             script_overrides;
-            display_msg(message_str(306, 105));
+            display_msg(message_str(SCRIPT_GLOBLDOR, 105));
         end
     end
 end
@@ -138,16 +138,16 @@ begin
             if (is_success(LVar2)) then begin
                 obj_unlock(self_obj);
                 set_local_var(3, 1);
-                display_msg(message_str(306, 109));
+                display_msg(message_str(SCRIPT_GLOBLDOR, 109));
             end
             else begin
                 if (is_critical(LVar2)) then begin
-                    display_msg(message_str(306, 121));
+                    display_msg(message_str(SCRIPT_GLOBLDOR, 121));
                     rm_obj_from_inven(source_obj, LVar0);
                     destroy_object(LVar0);
                 end
                 else begin
-                    display_msg(message_str(306, 122));
+                    display_msg(message_str(SCRIPT_GLOBLDOR, 122));
                 end
             end
         end
@@ -158,7 +158,7 @@ begin
             set_local_var(2, 1);
             obj_unlock(self_obj);
             set_local_var(3, 1);
-            display_msg(message_str(306, 106));
+            display_msg(message_str(SCRIPT_GLOBLDOR, 106));
         end
     end
 end
@@ -166,7 +166,7 @@ end
 procedure look_at_p_proc
 begin
     script_overrides;
-    display_msg(message_str(306, 123));
+    display_msg(message_str(SCRIPT_GLOBLDOR, 123));
 end
 
 procedure description_p_proc
@@ -187,7 +187,7 @@ begin
                 call Display_Locked();
             end
             else begin
-                display_msg(message_str(306, 123));
+                display_msg(message_str(SCRIPT_GLOBLDOR, 123));
             end
         end
     end
@@ -228,12 +228,12 @@ begin
         script_overrides;
         LVar0 := roll_vs_skill(source_obj, 11, -20);
         if (is_success(LVar0)) then begin
-            display_msg(message_str(306, 113));
+            display_msg(message_str(SCRIPT_GLOBLDOR, 113));
             set_local_var(2, 1);
         end
         else begin
             if (is_critical(LVar0)) then begin
-                display_msg(message_str(306, 114));
+                display_msg(message_str(SCRIPT_GLOBLDOR, 114));
                 if (local_var(0) == 0) then begin
                     set_local_var(0, 4);
                 end
@@ -244,7 +244,7 @@ begin
             end
             else begin
                 set_local_var(0, local_var(0) + 1);
-                display_msg(message_str(306, 115));
+                display_msg(message_str(SCRIPT_GLOBLDOR, 115));
                 call Damage_Dude();
             end
         end
@@ -258,24 +258,24 @@ begin
                 if (is_success(LVar0)) then begin
                     obj_unlock(self_obj);
                     set_local_var(3, 1);
-                    display_msg(message_str(306, 116));
+                    display_msg(message_str(SCRIPT_GLOBLDOR, 116));
                 end
                 else begin
                     if (is_critical(LVar0)) then begin
                         set_local_var(0, local_var(0) + 2);
-                        display_msg(message_str(306, 117));
+                        display_msg(message_str(SCRIPT_GLOBLDOR, 117));
                         call Damage_Dude();
                     end
                     else begin
                         set_local_var(0, local_var(0) + 1);
-                        display_msg(message_str(306, 118));
+                        display_msg(message_str(SCRIPT_GLOBLDOR, 118));
                         call Damage_Dude();
                     end
                 end
             end
             else begin
                 if (is_critical(LVar1)) then begin
-                    display_msg(message_str(306, 120));
+                    display_msg(message_str(SCRIPT_GLOBLDOR, 120));
                     if (local_var(0) == 0) then begin
                         set_local_var(0, 4);
                     end
@@ -286,7 +286,7 @@ begin
                 end
                 else begin
                     set_local_var(0, local_var(0) + 1);
-                    display_msg(message_str(306, 119));
+                    display_msg(message_str(SCRIPT_GLOBLDOR, 119));
                     call Damage_Dude();
                 end
             end
@@ -303,15 +303,15 @@ begin
         if (is_success(LVar0)) then begin
             obj_unlock(self_obj);
             set_local_var(3, 1);
-            display_msg(message_str(306, 116));
+            display_msg(message_str(SCRIPT_GLOBLDOR, 116));
         end
         else begin
             if (is_critical(LVar0)) then begin
-                display_msg(message_str(306, 121));
+                display_msg(message_str(SCRIPT_GLOBLDOR, 121));
                 game_time_advance(game_ticks(600));
             end
             else begin
-                display_msg(message_str(306, 122));
+                display_msg(message_str(SCRIPT_GLOBLDOR, 122));
             end
         end
     end
@@ -324,12 +324,12 @@ begin
         script_overrides;
         LVar0 := roll_vs_skill(source_obj, 11, -20);
         if (is_success(LVar0)) then begin
-            display_msg(message_str(306, 113));
+            display_msg(message_str(SCRIPT_GLOBLDOR, 113));
             set_local_var(2, 1);
         end
         else begin
             if (is_critical(LVar0)) then begin
-                display_msg(message_str(306, 114));
+                display_msg(message_str(SCRIPT_GLOBLDOR, 114));
                 if (local_var(0) == 0) then begin
                     set_local_var(0, 4);
                 end
@@ -340,7 +340,7 @@ begin
             end
             else begin
                 set_local_var(0, local_var(0) + 1);
-                display_msg(message_str(306, 115));
+                display_msg(message_str(SCRIPT_GLOBLDOR, 115));
                 call Damage_Dude();
             end
         end
@@ -349,12 +349,12 @@ end
 
 procedure Damage_Dude
 begin
-    critter_dmg(source_obj, local_var(0), 0);
+    critter_damage(source_obj, local_var(0));
     if (local_var(0) == 1) then begin
-        display_msg(message_str(306, 101));
+        display_msg(message_str(SCRIPT_GLOBLDOR, 101));
     end
     else begin
-        display_msg(message_str(306, 102) + local_var(0) + message_str(306, 103));
+        display_msg(message_str(SCRIPT_GLOBLDOR, 102) + local_var(0) + message_str(SCRIPT_GLOBLDOR, 103));
     end
 end
 
@@ -443,18 +443,18 @@ begin
     if (Per == 0) then begin
         if (Locks == 0) then begin
             if (Traps == 0) then begin
-                display_msg(message_str(306, 124));
+                display_msg(message_str(SCRIPT_GLOBLDOR, 124));
             end
             else begin
                 if (Traps == 1) then begin
-                    display_msg(message_str(306, 125));
+                    display_msg(message_str(SCRIPT_GLOBLDOR, 125));
                 end
                 else begin
                     if (Traps == 2) then begin
-                        display_msg(message_str(306, 126));
+                        display_msg(message_str(SCRIPT_GLOBLDOR, 126));
                     end
                     else begin
-                        display_msg(message_str(306, 127));
+                        display_msg(message_str(SCRIPT_GLOBLDOR, 127));
                     end
                 end
             end
@@ -462,18 +462,18 @@ begin
         else begin
             if (Locks == 1) then begin
                 if (Traps == 0) then begin
-                    display_msg(message_str(306, 128));
+                    display_msg(message_str(SCRIPT_GLOBLDOR, 128));
                 end
                 else begin
                     if (Traps == 1) then begin
-                        display_msg(message_str(306, 129));
+                        display_msg(message_str(SCRIPT_GLOBLDOR, 129));
                     end
                     else begin
                         if (Traps == 2) then begin
-                            display_msg(message_str(306, 130));
+                            display_msg(message_str(SCRIPT_GLOBLDOR, 130));
                         end
                         else begin
-                            display_msg(message_str(306, 131));
+                            display_msg(message_str(SCRIPT_GLOBLDOR, 131));
                         end
                     end
                 end
@@ -481,36 +481,36 @@ begin
             else begin
                 if (Locks == 2) then begin
                     if (Traps == 0) then begin
-                        display_msg(message_str(306, 132));
+                        display_msg(message_str(SCRIPT_GLOBLDOR, 132));
                     end
                     else begin
                         if (Traps == 1) then begin
-                            display_msg(message_str(306, 133));
+                            display_msg(message_str(SCRIPT_GLOBLDOR, 133));
                         end
                         else begin
                             if (Traps == 2) then begin
-                                display_msg(message_str(306, 134));
+                                display_msg(message_str(SCRIPT_GLOBLDOR, 134));
                             end
                             else begin
-                                display_msg(message_str(306, 135));
+                                display_msg(message_str(SCRIPT_GLOBLDOR, 135));
                             end
                         end
                     end
                 end
                 else begin
                     if (Traps == 0) then begin
-                        display_msg(message_str(306, 136));
+                        display_msg(message_str(SCRIPT_GLOBLDOR, 136));
                     end
                     else begin
                         if (Traps == 1) then begin
-                            display_msg(message_str(306, 137));
+                            display_msg(message_str(SCRIPT_GLOBLDOR, 137));
                         end
                         else begin
                             if (Traps == 2) then begin
-                                display_msg(message_str(306, 138));
+                                display_msg(message_str(SCRIPT_GLOBLDOR, 138));
                             end
                             else begin
-                                display_msg(message_str(306, 139));
+                                display_msg(message_str(SCRIPT_GLOBLDOR, 139));
                             end
                         end
                     end
@@ -522,18 +522,18 @@ begin
         if (Per == 1) then begin
             if (Locks == 0) then begin
                 if (Traps == 0) then begin
-                    display_msg(message_str(306, 140));
+                    display_msg(message_str(SCRIPT_GLOBLDOR, 140));
                 end
                 else begin
                     if (Traps == 1) then begin
-                        display_msg(message_str(306, 141));
+                        display_msg(message_str(SCRIPT_GLOBLDOR, 141));
                     end
                     else begin
                         if (Traps == 2) then begin
-                            display_msg(message_str(306, 142));
+                            display_msg(message_str(SCRIPT_GLOBLDOR, 142));
                         end
                         else begin
-                            display_msg(message_str(306, 143));
+                            display_msg(message_str(SCRIPT_GLOBLDOR, 143));
                         end
                     end
                 end
@@ -541,18 +541,18 @@ begin
             else begin
                 if (Locks == 1) then begin
                     if (Traps == 0) then begin
-                        display_msg(message_str(306, 144));
+                        display_msg(message_str(SCRIPT_GLOBLDOR, 144));
                     end
                     else begin
                         if (Traps == 1) then begin
-                            display_msg(message_str(306, 145));
+                            display_msg(message_str(SCRIPT_GLOBLDOR, 145));
                         end
                         else begin
                             if (Traps == 2) then begin
-                                display_msg(message_str(306, 146));
+                                display_msg(message_str(SCRIPT_GLOBLDOR, 146));
                             end
                             else begin
-                                display_msg(message_str(306, 147));
+                                display_msg(message_str(SCRIPT_GLOBLDOR, 147));
                             end
                         end
                     end
@@ -560,36 +560,36 @@ begin
                 else begin
                     if (Locks == 2) then begin
                         if (Traps == 0) then begin
-                            display_msg(message_str(306, 148));
+                            display_msg(message_str(SCRIPT_GLOBLDOR, 148));
                         end
                         else begin
                             if (Traps == 1) then begin
-                                display_msg(message_str(306, 149));
+                                display_msg(message_str(SCRIPT_GLOBLDOR, 149));
                             end
                             else begin
                                 if (Traps == 2) then begin
-                                    display_msg(message_str(306, 150));
+                                    display_msg(message_str(SCRIPT_GLOBLDOR, 150));
                                 end
                                 else begin
-                                    display_msg(message_str(306, 151));
+                                    display_msg(message_str(SCRIPT_GLOBLDOR, 151));
                                 end
                             end
                         end
                     end
                     else begin
                         if (Traps == 0) then begin
-                            display_msg(message_str(306, 152));
+                            display_msg(message_str(SCRIPT_GLOBLDOR, 152));
                         end
                         else begin
                             if (Traps == 1) then begin
-                                display_msg(message_str(306, 153));
+                                display_msg(message_str(SCRIPT_GLOBLDOR, 153));
                             end
                             else begin
                                 if (Traps == 2) then begin
-                                    display_msg(message_str(306, 154));
+                                    display_msg(message_str(SCRIPT_GLOBLDOR, 154));
                                 end
                                 else begin
-                                    display_msg(message_str(306, 155));
+                                    display_msg(message_str(SCRIPT_GLOBLDOR, 155));
                                 end
                             end
                         end
@@ -601,18 +601,18 @@ begin
             if (Per == 2) then begin
                 if (Locks == 0) then begin
                     if (Traps == 0) then begin
-                        display_msg(message_str(306, 156));
+                        display_msg(message_str(SCRIPT_GLOBLDOR, 156));
                     end
                     else begin
                         if (Traps == 1) then begin
-                            display_msg(message_str(306, 157));
+                            display_msg(message_str(SCRIPT_GLOBLDOR, 157));
                         end
                         else begin
                             if (Traps == 2) then begin
-                                display_msg(message_str(306, 158));
+                                display_msg(message_str(SCRIPT_GLOBLDOR, 158));
                             end
                             else begin
-                                display_msg(message_str(306, 159));
+                                display_msg(message_str(SCRIPT_GLOBLDOR, 159));
                             end
                         end
                     end
@@ -620,18 +620,18 @@ begin
                 else begin
                     if (Locks == 1) then begin
                         if (Traps == 0) then begin
-                            display_msg(message_str(306, 160));
+                            display_msg(message_str(SCRIPT_GLOBLDOR, 160));
                         end
                         else begin
                             if (Traps == 1) then begin
-                                display_msg(message_str(306, 161));
+                                display_msg(message_str(SCRIPT_GLOBLDOR, 161));
                             end
                             else begin
                                 if (Traps == 2) then begin
-                                    display_msg(message_str(306, 162));
+                                    display_msg(message_str(SCRIPT_GLOBLDOR, 162));
                                 end
                                 else begin
-                                    display_msg(message_str(306, 163));
+                                    display_msg(message_str(SCRIPT_GLOBLDOR, 163));
                                 end
                             end
                         end
@@ -639,36 +639,36 @@ begin
                     else begin
                         if (Locks == 2) then begin
                             if (Traps == 0) then begin
-                                display_msg(message_str(306, 164));
+                                display_msg(message_str(SCRIPT_GLOBLDOR, 164));
                             end
                             else begin
                                 if (Traps == 1) then begin
-                                    display_msg(message_str(306, 165));
+                                    display_msg(message_str(SCRIPT_GLOBLDOR, 165));
                                 end
                                 else begin
                                     if (Traps == 2) then begin
-                                        display_msg(message_str(306, 166));
+                                        display_msg(message_str(SCRIPT_GLOBLDOR, 166));
                                     end
                                     else begin
-                                        display_msg(message_str(306, 167));
+                                        display_msg(message_str(SCRIPT_GLOBLDOR, 167));
                                     end
                                 end
                             end
                         end
                         else begin
                             if (Traps == 0) then begin
-                                display_msg(message_str(306, 168));
+                                display_msg(message_str(SCRIPT_GLOBLDOR, 168));
                             end
                             else begin
                                 if (Traps == 1) then begin
-                                    display_msg(message_str(306, 169));
+                                    display_msg(message_str(SCRIPT_GLOBLDOR, 169));
                                 end
                                 else begin
                                     if (Traps == 2) then begin
-                                        display_msg(message_str(306, 170));
+                                        display_msg(message_str(SCRIPT_GLOBLDOR, 170));
                                     end
                                     else begin
-                                        display_msg(message_str(306, 171));
+                                        display_msg(message_str(SCRIPT_GLOBLDOR, 171));
                                     end
                                 end
                             end
@@ -679,18 +679,18 @@ begin
             else begin
                 if (Locks == 0) then begin
                     if (Traps == 0) then begin
-                        display_msg(message_str(306, 172));
+                        display_msg(message_str(SCRIPT_GLOBLDOR, 172));
                     end
                     else begin
                         if (Traps == 1) then begin
-                            display_msg(message_str(306, 173));
+                            display_msg(message_str(SCRIPT_GLOBLDOR, 173));
                         end
                         else begin
                             if (Traps == 2) then begin
-                                display_msg(message_str(306, 174));
+                                display_msg(message_str(SCRIPT_GLOBLDOR, 174));
                             end
                             else begin
-                                display_msg(message_str(306, 175));
+                                display_msg(message_str(SCRIPT_GLOBLDOR, 175));
                             end
                         end
                     end
@@ -698,18 +698,18 @@ begin
                 else begin
                     if (Locks == 1) then begin
                         if (Traps == 0) then begin
-                            display_msg(message_str(306, 176));
+                            display_msg(message_str(SCRIPT_GLOBLDOR, 176));
                         end
                         else begin
                             if (Traps == 1) then begin
-                                display_msg(message_str(306, 177));
+                                display_msg(message_str(SCRIPT_GLOBLDOR, 177));
                             end
                             else begin
                                 if (Traps == 2) then begin
-                                    display_msg(message_str(306, 178));
+                                    display_msg(message_str(SCRIPT_GLOBLDOR, 178));
                                 end
                                 else begin
-                                    display_msg(message_str(306, 179));
+                                    display_msg(message_str(SCRIPT_GLOBLDOR, 179));
                                 end
                             end
                         end
@@ -717,36 +717,36 @@ begin
                     else begin
                         if (Locks == 2) then begin
                             if (Traps == 0) then begin
-                                display_msg(message_str(306, 180));
+                                display_msg(message_str(SCRIPT_GLOBLDOR, 180));
                             end
                             else begin
                                 if (Traps == 1) then begin
-                                    display_msg(message_str(306, 181));
+                                    display_msg(message_str(SCRIPT_GLOBLDOR, 181));
                                 end
                                 else begin
                                     if (Traps == 2) then begin
-                                        display_msg(message_str(306, 182));
+                                        display_msg(message_str(SCRIPT_GLOBLDOR, 182));
                                     end
                                     else begin
-                                        display_msg(message_str(306, 183));
+                                        display_msg(message_str(SCRIPT_GLOBLDOR, 183));
                                     end
                                 end
                             end
                         end
                         else begin
                             if (Traps == 0) then begin
-                                display_msg(message_str(306, 184));
+                                display_msg(message_str(SCRIPT_GLOBLDOR, 184));
                             end
                             else begin
                                 if (Traps == 1) then begin
-                                    display_msg(message_str(306, 185));
+                                    display_msg(message_str(SCRIPT_GLOBLDOR, 185));
                                 end
                                 else begin
                                     if (Traps == 2) then begin
-                                        display_msg(message_str(306, 186));
+                                        display_msg(message_str(SCRIPT_GLOBLDOR, 186));
                                     end
                                     else begin
-                                        display_msg(message_str(306, 187));
+                                        display_msg(message_str(SCRIPT_GLOBLDOR, 187));
                                     end
                                 end
                             end
@@ -762,18 +762,18 @@ procedure Display_Locked
 begin
     if (Per == 0) then begin
         if (Locks == 0) then begin
-            display_msg(message_str(306, 124));
+            display_msg(message_str(SCRIPT_GLOBLDOR, 124));
         end
         else begin
             if (Locks == 1) then begin
-                display_msg(message_str(306, 128));
+                display_msg(message_str(SCRIPT_GLOBLDOR, 128));
             end
             else begin
                 if (Locks == 2) then begin
-                    display_msg(message_str(306, 132));
+                    display_msg(message_str(SCRIPT_GLOBLDOR, 132));
                 end
                 else begin
-                    display_msg(message_str(306, 136));
+                    display_msg(message_str(SCRIPT_GLOBLDOR, 136));
                 end
             end
         end
@@ -781,18 +781,18 @@ begin
     else begin
         if (Per == 1) then begin
             if (Locks == 0) then begin
-                display_msg(message_str(306, 140));
+                display_msg(message_str(SCRIPT_GLOBLDOR, 140));
             end
             else begin
                 if (Locks == 1) then begin
-                    display_msg(message_str(306, 144));
+                    display_msg(message_str(SCRIPT_GLOBLDOR, 144));
                 end
                 else begin
                     if (Locks == 2) then begin
-                        display_msg(message_str(306, 148));
+                        display_msg(message_str(SCRIPT_GLOBLDOR, 148));
                     end
                     else begin
-                        display_msg(message_str(306, 152));
+                        display_msg(message_str(SCRIPT_GLOBLDOR, 152));
                     end
                 end
             end
@@ -800,36 +800,36 @@ begin
         else begin
             if (Per == 2) then begin
                 if (Locks == 0) then begin
-                    display_msg(message_str(306, 156));
+                    display_msg(message_str(SCRIPT_GLOBLDOR, 156));
                 end
                 else begin
                     if (Locks == 1) then begin
-                        display_msg(message_str(306, 160));
+                        display_msg(message_str(SCRIPT_GLOBLDOR, 160));
                     end
                     else begin
                         if (Locks == 2) then begin
-                            display_msg(message_str(306, 164));
+                            display_msg(message_str(SCRIPT_GLOBLDOR, 164));
                         end
                         else begin
-                            display_msg(message_str(306, 168));
+                            display_msg(message_str(SCRIPT_GLOBLDOR, 168));
                         end
                     end
                 end
             end
             else begin
                 if (Locks == 0) then begin
-                    display_msg(message_str(306, 172));
+                    display_msg(message_str(SCRIPT_GLOBLDOR, 172));
                 end
                 else begin
                     if (Locks == 1) then begin
-                        display_msg(message_str(306, 176));
+                        display_msg(message_str(SCRIPT_GLOBLDOR, 176));
                     end
                     else begin
                         if (Locks == 2) then begin
-                            display_msg(message_str(306, 180));
+                            display_msg(message_str(SCRIPT_GLOBLDOR, 180));
                         end
                         else begin
-                            display_msg(message_str(306, 184));
+                            display_msg(message_str(SCRIPT_GLOBLDOR, 184));
                         end
                     end
                 end
@@ -842,18 +842,18 @@ procedure Display_Trapped
 begin
     if (Per == 0) then begin
         if (Traps == 0) then begin
-            display_msg(message_str(306, 188));
+            display_msg(message_str(SCRIPT_GLOBLDOR, 188));
         end
         else begin
             if (Traps == 1) then begin
-                display_msg(message_str(306, 189));
+                display_msg(message_str(SCRIPT_GLOBLDOR, 189));
             end
             else begin
                 if (Traps == 2) then begin
-                    display_msg(message_str(306, 190));
+                    display_msg(message_str(SCRIPT_GLOBLDOR, 190));
                 end
                 else begin
-                    display_msg(message_str(306, 191));
+                    display_msg(message_str(SCRIPT_GLOBLDOR, 191));
                 end
             end
         end
@@ -861,18 +861,18 @@ begin
     else begin
         if (Per == 1) then begin
             if (Traps == 0) then begin
-                display_msg(message_str(306, 192));
+                display_msg(message_str(SCRIPT_GLOBLDOR, 192));
             end
             else begin
                 if (Traps == 1) then begin
-                    display_msg(message_str(306, 193));
+                    display_msg(message_str(SCRIPT_GLOBLDOR, 193));
                 end
                 else begin
                     if (Traps == 2) then begin
-                        display_msg(message_str(306, 194));
+                        display_msg(message_str(SCRIPT_GLOBLDOR, 194));
                     end
                     else begin
-                        display_msg(message_str(306, 195));
+                        display_msg(message_str(SCRIPT_GLOBLDOR, 195));
                     end
                 end
             end
@@ -880,36 +880,36 @@ begin
         else begin
             if (Per == 2) then begin
                 if (Traps == 0) then begin
-                    display_msg(message_str(306, 196));
+                    display_msg(message_str(SCRIPT_GLOBLDOR, 196));
                 end
                 else begin
                     if (Traps == 1) then begin
-                        display_msg(message_str(306, 197));
+                        display_msg(message_str(SCRIPT_GLOBLDOR, 197));
                     end
                     else begin
                         if (Traps == 2) then begin
-                            display_msg(message_str(306, 198));
+                            display_msg(message_str(SCRIPT_GLOBLDOR, 198));
                         end
                         else begin
-                            display_msg(message_str(306, 199));
+                            display_msg(message_str(SCRIPT_GLOBLDOR, 199));
                         end
                     end
                 end
             end
             else begin
                 if (Traps == 0) then begin
-                    display_msg(message_str(306, 200));
+                    display_msg(message_str(SCRIPT_GLOBLDOR, 200));
                 end
                 else begin
                     if (Traps == 1) then begin
-                        display_msg(message_str(306, 201));
+                        display_msg(message_str(SCRIPT_GLOBLDOR, 201));
                     end
                     else begin
                         if (Traps == 2) then begin
-                            display_msg(message_str(306, 202));
+                            display_msg(message_str(SCRIPT_GLOBLDOR, 202));
                         end
                         else begin
-                            display_msg(message_str(306, 203));
+                            display_msg(message_str(SCRIPT_GLOBLDOR, 203));
                         end
                     end
                 end
@@ -917,4 +917,3 @@ begin
         end
     end
 end
-
