diff --git a/Games/Fallout1.Steam/Original/GLOWDESK.ssl b/Games/Fallout1.Steam/ReDefine/GLOWDESK.ssl
index 57e2c4e..6382a38 100644
--- a/Games/Fallout1.Steam/Original/GLOWDESK.ssl
+++ b/Games/Fallout1.Steam/ReDefine/GLOWDESK.ssl
@@ -3,27 +3,26 @@ procedure start;
 
 procedure start
 begin
-    if (script_action == 4) then begin
+    if (script_action == pickup_proc) then begin
         if (local_var(0) == 0) then begin
             script_overrides;
-            display_msg(message_str(331, 100));
+            display_mstr(100);
         end
     end
     else begin
-        if (script_action == 8) then begin
+        if (script_action == use_skill_on_proc) then begin
             if (action_being_used == 9) then begin
                 if (is_success(roll_vs_skill(dude_obj, 9, 0))) then begin
                     set_local_var(0, 1);
-                    display_msg(message_str(331, 101));
+                    display_mstr(101);
                 end
                 else begin
-                    display_msg(message_str(331, 102));
+                    display_mstr(102);
                 end
             end
             else begin
-                display_msg(message_str(331, 103));
+                display_mstr(103);
             end
         end
     end
 end
-
