diff --git a/Games/Fallout1.Steam/Original/GLOWGEN.ssl b/Games/Fallout1.Steam/ReDefine/GLOWGEN.ssl
index ae1810e..bcb4171 100644
--- a/Games/Fallout1.Steam/Original/GLOWGEN.ssl
+++ b/Games/Fallout1.Steam/ReDefine/GLOWGEN.ssl
@@ -3,83 +3,83 @@ procedure start;
 
 procedure start
 begin
-    if (script_action == 21) then begin
+    if (script_action == look_at_proc) then begin
         script_overrides;
-        display_msg(message_str(301, 100));
+        display_mstr(100);
     end
     else begin
-        if (script_action == 3) then begin
+        if (script_action == description_proc) then begin
             script_overrides;
-            if (global_var(139) == 2) then begin
-                display_msg(message_str(301, 101));
+            if (global_var(GVAR_START_POWER) == 2) then begin
+                display_mstr(101);
             end
             else begin
-                display_msg(message_str(301, 102));
+                display_mstr(102);
             end
         end
         else begin
-            if (script_action == 8) then begin
+            if (script_action == use_skill_on_proc) then begin
                 if (action_being_used == 13) then begin
                     script_overrides;
-                    if (has_skill(dude_obj, 13) > 35) then begin
+                    if (dude_skill(SKILL_REPAIR) > 35) then begin
                         set_local_var(0, 0);
                     end
-                    if (global_var(139) == 2) then begin
-                        display_msg(message_str(301, 103));
+                    if (global_var(GVAR_START_POWER) == 2) then begin
+                        display_mstr(103);
                     end
                     else begin
                         if (local_var(0) == 0) then begin
                             if (is_success(roll_vs_skill(dude_obj, 13, -10))) then begin
-                                display_msg(message_str(301, 104));
-                                set_global_var(139, 2);
+                                display_mstr(104);
+                                set_global_var(GVAR_START_POWER, 2);
                                 give_exp_points(1000);
-                                display_msg(message_str(766, 103) + "1000" + message_str(766, 104));
+                                display_msg(message_str(SCRIPT_GENCHAT, 103) + "1000" + message_str(SCRIPT_GENCHAT, 104));
                             end
                             else begin
-                                if (has_skill(dude_obj, 13) < 35) then begin
+                                if (dude_skill(SKILL_REPAIR) < 35) then begin
                                     set_local_var(0, 1);
                                 end
-                                display_msg(message_str(301, 105));
+                                display_mstr(105);
                                 game_time_advance(game_ticks(1200));
                             end
                         end
                         else begin
                             if (local_var(0) == 1) then begin
-                                display_msg(message_str(301, 106));
+                                display_mstr(106);
                             end
                         end
                     end
                 end
             end
             else begin
-                if (script_action == 7) then begin
+                if (script_action == use_obj_on_proc) then begin
                     if (obj_pid(obj_being_used_with) == 75) then begin
                         script_overrides;
-                        if (has_skill(dude_obj, 13) > 35) then begin
+                        if (dude_skill(SKILL_REPAIR) > 35) then begin
                             set_local_var(0, 0);
                         end
-                        if (global_var(139) == 2) then begin
-                            display_msg(message_str(301, 103));
+                        if (global_var(GVAR_START_POWER) == 2) then begin
+                            display_mstr(103);
                         end
                         else begin
                             if (local_var(0) == 0) then begin
                                 if (is_success(roll_vs_skill(dude_obj, 13, 10))) then begin
-                                    display_msg(message_str(301, 104));
-                                    set_global_var(139, 2);
-                                    display_msg(message_str(766, 103) + "1000" + message_str(766, 104));
+                                    display_mstr(104);
+                                    set_global_var(GVAR_START_POWER, 2);
+                                    display_msg(message_str(SCRIPT_GENCHAT, 103) + "1000" + message_str(SCRIPT_GENCHAT, 104));
                                     give_exp_points(1000);
                                 end
                                 else begin
-                                    if (has_skill(dude_obj, 13) < 35) then begin
+                                    if (dude_skill(SKILL_REPAIR) < 35) then begin
                                         set_local_var(0, 1);
                                     end
-                                    display_msg(message_str(301, 105));
+                                    display_mstr(105);
                                     game_time_advance(game_ticks(1200));
                                 end
                             end
                             else begin
                                 if (local_var(0) == 1) then begin
-                                    display_msg(message_str(301, 106));
+                                    display_mstr(106);
                                 end
                             end
                         end
@@ -89,4 +89,3 @@ begin
         end
     end
 end
-
