diff --git a/Games/Fallout1.Steam/Original/GRAFFITI.ssl b/Games/Fallout1.Steam/ReDefine/GRAFFITI.ssl
index 39f4c92..01bcf87 100644
--- a/Games/Fallout1.Steam/Original/GRAFFITI.ssl
+++ b/Games/Fallout1.Steam/ReDefine/GRAFFITI.ssl
@@ -5,13 +5,12 @@ variable HEREBEFORE;
 
 procedure start
 begin
-    if (script_action == 2) then begin
+    if (script_action == spatial_proc) then begin
         if (obj_type(source_obj) == dude_obj) then begin
             if (not(HEREBEFORE)) then begin
                 HEREBEFORE := 1;
-                display_msg(message_str(91, 100));
+                display_mstr(100);
             end
         end
     end
 end
-
