diff --git a/Games/Fallout1.Steam/Original/GUNCASE.ssl b/Games/Fallout1.Steam/ReDefine/GUNCASE.ssl
index f9a9b42..73b82a1 100644
--- a/Games/Fallout1.Steam/Original/GUNCASE.ssl
+++ b/Games/Fallout1.Steam/ReDefine/GUNCASE.ssl
@@ -5,11 +5,11 @@ procedure pickup_p_proc;
 
 procedure start
 begin
-    if (script_action == 6) then begin
+    if (script_action == use_proc) then begin
         call use_p_proc();
     end
     else begin
-        if (script_action == 4) then begin
+        if (script_action == pickup_proc) then begin
             call pickup_p_proc();
         end
     end
@@ -18,12 +18,11 @@ end
 procedure use_p_proc
 begin
     script_overrides;
-    display_msg(message_str(766, 176));
+    display_msg(message_str(SCRIPT_GENCHAT, 176));
 end
 
 procedure pickup_p_proc
 begin
     script_overrides;
-    display_msg(message_str(766, 176));
+    display_msg(message_str(SCRIPT_GENCHAT, 176));
 end
-
