diff --git a/Games/Fallout1.Steam/Original/GUSTOFER.ssl b/Games/Fallout1.Steam/ReDefine/GUSTOFER.ssl
index 3ebd718..7c8fa7b 100644
--- a/Games/Fallout1.Steam/Original/GUSTOFER.ssl
+++ b/Games/Fallout1.Steam/ReDefine/GUSTOFER.ssl
@@ -56,27 +56,27 @@ variable sleep_tile;
 
 procedure start
 begin
-    if (script_action == 13) then begin
+    if (script_action == combat_proc) then begin
         call combat_p_proc();
     end
     else begin
-        if (script_action == 12) then begin
+        if (script_action == critter_proc) then begin
             call critter_p_proc();
         end
         else begin
-            if (script_action == 18) then begin
+            if (script_action == destroy_proc) then begin
                 call destroy_p_proc();
             end
             else begin
-                if (script_action == 15) then begin
+                if (script_action == map_enter_proc) then begin
                     call map_enter_p_proc();
                 end
                 else begin
-                    if (script_action == 4) then begin
+                    if (script_action == pickup_proc) then begin
                         call pickup_p_proc();
                     end
                     else begin
-                        if (script_action == 11) then begin
+                        if (script_action == talk_proc) then begin
                             call talk_p_proc();
                         end
                     end
@@ -92,9 +92,9 @@ begin
         round_counter := round_counter + 1;
     end
     if (round_counter > 3) then begin
-        if (global_var(247) == 0) then begin
-            set_global_var(247, 1);
-            set_global_var(155, global_var(155) - 5);
+        if (global_var(GVAR_ENEMY_JUNKTOWN) == 0) then begin
+            set_global_var(GVAR_ENEMY_JUNKTOWN, 1);
+            set_global_var(GVAR_PLAYER_REPUATION, global_var(GVAR_PLAYER_REPUATION) - 5);
         end
     end
 end
@@ -109,7 +109,7 @@ begin
             if ((line148flag == 0) and ((game_time / (10 * 60 * 60 * 24) % 3) == 0)) then begin
                 line148flag := 1;
                 fight := 1;
-                float_msg(self_obj, message_str(529, 148), 0);
+                float_msg(self_obj, mstr(148), FLOAT_MSG_NORMAL);
             end
         end
         else begin
@@ -130,24 +130,24 @@ end
 procedure destroy_p_proc
 begin
     if (source_obj == dude_obj) then begin
-        if (((global_var(160) + global_var(159)) >= 25) and ((global_var(159) > (2 * global_var(160))) or (global_var(156) == 1))) then begin
-            set_global_var(156, 1);
-            set_global_var(157, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
+            set_global_var(GVAR_BERSERKER_REPUTATION, 1);
+            set_global_var(GVAR_CHAMPION_REPUTATION, 0);
         end
-        if (((global_var(160) + global_var(159)) >= 25) and ((global_var(160) > (3 * global_var(159))) or (global_var(157) == 1))) then begin
-            set_global_var(157, 1);
-            set_global_var(156, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(160) > (3 * global_var(GVAR_GOOD_MONSTER))) or (global_var(GVAR_CHAMPION_REPUTATION) == 1))) then begin
+            set_global_var(GVAR_CHAMPION_REPUTATION, 1);
+            set_global_var(GVAR_BERSERKER_REPUTATION, 0);
         end
-        set_global_var(160, global_var(160) + 1);
-        if ((global_var(160) % 6) == 0) then begin
-            set_global_var(155, global_var(155) + 1);
+        set_global_var(GVAR_BAD_MONSTER, global_var(GVAR_BAD_MONSTER) + 1);
+        if ((global_var(GVAR_BAD_MONSTER) % 6) == 0) then begin
+            set_global_var(GVAR_PLAYER_REPUATION, global_var(GVAR_PLAYER_REPUATION) + 1);
         end
     end
 end
 
 procedure map_enter_p_proc
 begin
-    if (global_var(15) == 1) then begin
+    if (global_var(GVAR_JUNKTOWN_INVADED) == 1) then begin
         destroy_object(self_obj);
     end
     if (local_var(0) == 0) then begin
@@ -156,7 +156,7 @@ begin
     if (obj_is_carrying_obj_pid(self_obj, 41) == 0) then begin
         item_caps_adjust(self_obj, random(0, 10));
     end
-    critter_add_trait(self_obj, 1, 6, 25);
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 25);
     sleep_time := 1920;
     wake_time := 530;
     home_tile := 17096;
@@ -205,7 +205,7 @@ begin
                                 call Gustofer06();
                             end
                             else begin
-                                gsay_message(529, 152, 50);
+                                gsay_message(NAME, 152, NEUTRAL_REACTION);
                             end
                         end
                         else begin
@@ -232,10 +232,10 @@ end
 procedure Gustofer00
 begin
     set_local_var(2, 1);
-    gsay_reply(529, 100);
-    giq_option(-3, 529, 101, Gustofer01, 50);
-    giq_option(4, 529, 102, GustoferEnd, 50);
-    giq_option(6, 529, 103, Gustofer00a, 50);
+    Reply(100);
+    giq_option(-3, NAME, 101, Gustofer01, NEUTRAL_REACTION);
+    giq_option(4, NAME, 102, GustoferEnd, NEUTRAL_REACTION);
+    giq_option(6, NAME, 103, Gustofer00a, NEUTRAL_REACTION);
 end
 
 procedure Gustofer00a
@@ -250,50 +250,50 @@ end
 
 procedure Gustofer01
 begin
-    gsay_message(529, 104, 50);
+    gsay_message(NAME, 104, NEUTRAL_REACTION);
 end
 
 procedure Gustofer02
 begin
-    gsay_reply(529, 105);
-    giq_option(4, 529, 107, GustoferEnd, 50);
-    giq_option(4, 529, 108, Gustofer04, 50);
+    Reply(105);
+    giq_option(4, NAME, 107, GustoferEnd, NEUTRAL_REACTION);
+    giq_option(4, NAME, 108, Gustofer04, NEUTRAL_REACTION);
 end
 
 procedure Gustofer03
 begin
-    gsay_reply(529, 106);
-    giq_option(4, 529, 107, GustoferEnd, 50);
-    giq_option(4, 529, 108, Gustofer04, 50);
+    Reply(106);
+    giq_option(4, NAME, 107, GustoferEnd, NEUTRAL_REACTION);
+    giq_option(4, NAME, 108, Gustofer04, NEUTRAL_REACTION);
 end
 
 procedure Gustofer04
 begin
-    gsay_message(529, 109, 50);
+    gsay_message(NAME, 109, NEUTRAL_REACTION);
 end
 
 procedure Gustofer05
 begin
     variable LVar0 := 0;
-    LVar0 := message_str(529, 110);
+    LVar0 := mstr(110);
     if ((game_time / (10 * 60 * 60 * 24) % 3) == 1) then begin
-        LVar0 := LVar0 + message_str(529, 112);
+        LVar0 := LVar0 + mstr(112);
     end
     else begin
         if ((game_time / (10 * 60 * 60 * 24) % 3) == 2) then begin
-            LVar0 := LVar0 + message_str(529, 111);
+            LVar0 := LVar0 + mstr(111);
         end
     end
-    float_msg(self_obj, LVar0, 0);
+    float_msg(self_obj, LVar0, FLOAT_MSG_NORMAL);
 end
 
 procedure Gustofer06
 begin
-    gsay_reply(529, 113);
-    giq_option(-3, 529, 101, Gustofer01, 50);
-    giq_option(4, 529, 114, GustoferEnd, 50);
-    giq_option(4, 529, 115, Gustofer08, 50);
-    giq_option(4, 529, 116, Gustofer09, 50);
+    Reply(113);
+    giq_option(-3, NAME, 101, Gustofer01, NEUTRAL_REACTION);
+    giq_option(4, NAME, 114, GustoferEnd, NEUTRAL_REACTION);
+    giq_option(4, NAME, 115, Gustofer08, NEUTRAL_REACTION);
+    giq_option(4, NAME, 116, Gustofer09, NEUTRAL_REACTION);
 end
 
 procedure Gustofer07
@@ -301,79 +301,79 @@ begin
     Saul_wins := 0;
     Saul_loses := 0;
     dude_bet := 0;
-    if (get_critter_stat(dude_obj, 34) == 0) then begin
-        gsay_message(529, 150, 50);
+    if (dude_gender == 0) then begin
+        gsay_message(NAME, 150, NEUTRAL_REACTION);
     end
     else begin
-        gsay_message(529, 151, 50);
+        gsay_message(NAME, 151, NEUTRAL_REACTION);
     end
 end
 
 procedure Gustofer08
 begin
-    gsay_reply(529, 120);
-    giq_option(4, 529, 121, GustoferBet1, 50);
-    giq_option(4, 529, 122, GustoferBet2, 50);
+    Reply(120);
+    giq_option(4, NAME, 121, GustoferBet1, NEUTRAL_REACTION);
+    giq_option(4, NAME, 122, GustoferBet2, NEUTRAL_REACTION);
 end
 
 procedure Gustofer09
 begin
     if (is_success(roll_vs_skill(dude_obj, 14, 0))) then begin
-        gsay_reply(529, 123);
-        if (global_var(169) < 1) then begin
-            giq_option(4, 529, 124, Gustofer10, 50);
+        Reply(123);
+        if (global_var(GVAR_SAUL_STATUS) < 1) then begin
+            giq_option(4, NAME, 124, Gustofer10, NEUTRAL_REACTION);
         end
-        giq_option(4, 529, 125, Gustofer11, 50);
-        giq_option(4, 529, 126, GustoferEnd, 50);
+        giq_option(4, NAME, 125, Gustofer11, NEUTRAL_REACTION);
+        giq_option(4, NAME, 126, GustoferEnd, NEUTRAL_REACTION);
     end
     else begin
-        gsay_message(529, 127, 50);
+        gsay_message(NAME, 127, NEUTRAL_REACTION);
     end
 end
 
 procedure Gustofer10
 begin
-    gsay_reply(529, 128);
-    giq_option(0, 634, 106, Gustofer10a, 50);
+    Reply(128);
+    giq_option(0, SCRIPT_MODREACT, 106, Gustofer10a, NEUTRAL_REACTION);
 end
 
 procedure Gustofer10a
 begin
-    gsay_reply(529, 129);
-    giq_option(4, 529, 130, Gustofer11, 50);
-    giq_option(4, 529, 131, GustoferEnd, 50);
+    Reply(129);
+    giq_option(4, NAME, 130, Gustofer11, NEUTRAL_REACTION);
+    giq_option(4, NAME, 131, GustoferEnd, NEUTRAL_REACTION);
 end
 
 procedure Gustofer11
 begin
-    gsay_reply(529, 132);
-    giq_option(4, 529, 133, Gustofer12, 50);
-    giq_option(4, 529, 134, GustoferEnd, 50);
+    Reply(132);
+    giq_option(4, NAME, 133, Gustofer12, NEUTRAL_REACTION);
+    giq_option(4, NAME, 134, GustoferEnd, NEUTRAL_REACTION);
 end
 
 procedure Gustofer12
 begin
-    gsay_reply(529, 135);
-    giq_option(4, 529, 136, GustoferEnd, 50);
+    Reply(135);
+    giq_option(4, NAME, 136, GustoferEnd, NEUTRAL_REACTION);
 end
 
 procedure Gustofer13
 begin
     robbed := 0;
-    gsay_reply(529, 137);
-    giq_option(4, 529, 138, Gustofer14, 51);
-    giq_option(4, 529, 139, Gustofer15, 51);
-    giq_option(-3, 529, 101, Gustofer15, 51);
+    Reply(137);
+    giq_option(4, NAME, 138, Gustofer14, BAD_REACTION);
+    giq_option(4, NAME, 139, Gustofer15, BAD_REACTION);
+    giq_option(-3, NAME, 101, Gustofer15, BAD_REACTION);
 end
 
 procedure Gustofer14
 begin
-    gsay_message(529, 140, 51);
+    gsay_message(NAME, 140, BAD_REACTION);
 end
 
 procedure Gustofer15
 begin
-    gsay_message(529, 141, 51);
+    gsay_message(NAME, 141, BAD_REACTION);
 end
 
 procedure Gustofer16
@@ -387,25 +387,25 @@ begin
     end
     Saul_wins := 0;
     Saul_loses := 0;
-    gsay_message(529, 142, 50);
+    gsay_message(NAME, 142, NEUTRAL_REACTION);
     call destroy_challenger();
 end
 
 procedure Gustofer17
 begin
-    gsay_message(529, 147, 50);
+    gsay_message(NAME, 147, NEUTRAL_REACTION);
 end
 
 procedure Gustofer18
 begin
-    gsay_reply(529, 118);
-    giq_option(4, 529, 119, GustoferEnd, 50);
-    giq_option(-3, 529, 101, GustoferEnd, 50);
+    Reply(118);
+    giq_option(4, NAME, 119, GustoferEnd, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 101, GustoferEnd, NEUTRAL_REACTION);
 end
 
 procedure Gustofer19
 begin
-    gsay_message(529, 149, 50);
+    gsay_message(NAME, 149, NEUTRAL_REACTION);
 end
 
 procedure GustoferEnd
@@ -424,12 +424,12 @@ begin
     if (item_caps_total(dude_obj) >= 25) then begin
         item_caps_adjust(dude_obj, -25);
         dude_bet := 1;
-        gsay_reply(529, 144);
-        giq_option(4, 529, 145, GustoferWait, 50);
-        giq_option(4, 529, 146, Gustofer17, 50);
+        Reply(144);
+        giq_option(4, NAME, 145, GustoferWait, NEUTRAL_REACTION);
+        giq_option(4, NAME, 146, Gustofer17, NEUTRAL_REACTION);
     end
     else begin
-        gsay_message(529, 143, 50);
+        gsay_message(NAME, 143, NEUTRAL_REACTION);
     end
 end
 
@@ -438,12 +438,12 @@ begin
     if (item_caps_total(dude_obj) >= 25) then begin
         item_caps_adjust(dude_obj, -25);
         dude_bet := 2;
-        gsay_reply(529, 144);
-        giq_option(4, 529, 145, GustoferWait, 50);
-        giq_option(4, 529, 146, Gustofer17, 50);
+        Reply(144);
+        giq_option(4, NAME, 145, GustoferWait, NEUTRAL_REACTION);
+        giq_option(4, NAME, 146, Gustofer17, NEUTRAL_REACTION);
     end
     else begin
-        gsay_message(529, 143, 50);
+        gsay_message(NAME, 143, NEUTRAL_REACTION);
     end
 end
 
@@ -453,9 +453,9 @@ begin
     variable LVar1 := 0;
     variable LVar2 := 0;
     while (LVar1 < 5) do begin
-        if (tile_distance(tile_num(dude_obj), tile_num_in_direction(tile_num(self_obj), LVar1, 3)) > LVar2) then begin
+        if (tile_distance(dude_tile, tile_num_in_direction(tile_num(self_obj), LVar1, 3)) > LVar2) then begin
             LVar0 := tile_num_in_direction(tile_num(self_obj), LVar1, 3);
-            LVar2 := tile_distance(tile_num(dude_obj), LVar0);
+            LVar2 := tile_distance(dude_tile, LVar0);
         end
         LVar1 := LVar1 + 1;
     end
@@ -510,4 +510,3 @@ begin
         end
     end
 end
-
