diff --git a/Games/Fallout1.Steam/Original/HALL2SHD.ssl b/Games/Fallout1.Steam/ReDefine/HALL2SHD.ssl
index 4625dbd..429bc4c 100644
--- a/Games/Fallout1.Steam/Original/HALL2SHD.ssl
+++ b/Games/Fallout1.Steam/ReDefine/HALL2SHD.ssl
@@ -3,8 +3,7 @@ procedure start;
 
 procedure start
 begin
-    if (script_action == 2) then begin
+    if (script_action == spatial_proc) then begin
         load_map(5, 12);
     end
 end
-
