diff --git a/Games/Fallout1.Steam/Original/HAROLD.ssl b/Games/Fallout1.Steam/ReDefine/HAROLD.ssl
index a5f15a6..c0c9283 100644
--- a/Games/Fallout1.Steam/Original/HAROLD.ssl
+++ b/Games/Fallout1.Steam/ReDefine/HAROLD.ssl
@@ -99,26 +99,26 @@ procedure start
 begin
     if (Only_Once) then begin
         Only_Once := 0;
-        critter_add_trait(self_obj, 1, 6, 41);
-        critter_add_trait(self_obj, 1, 5, 53);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 41);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_HUB_MUTANTS);
     end
-    if (script_action == 21) then begin
+    if (script_action == look_at_proc) then begin
         call look_at_p_proc();
     end
     else begin
-        if (script_action == 4) then begin
+        if (script_action == pickup_proc) then begin
             call pickup_p_proc();
         end
         else begin
-            if (script_action == 11) then begin
+            if (script_action == talk_proc) then begin
                 call talk_p_proc();
             end
             else begin
-                if (script_action == 12) then begin
+                if (script_action == critter_proc) then begin
                     call critter_p_proc();
                 end
                 else begin
-                    if (script_action == 18) then begin
+                    if (script_action == destroy_proc) then begin
                         call destroy_p_proc();
                     end
                 end
@@ -136,7 +136,7 @@ procedure critter_p_proc
 begin
     if (hostile) then begin
         hostile := 0;
-        attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+        attack(dude_obj);
     end
 end
 
@@ -189,17 +189,17 @@ end
 procedure destroy_p_proc
 begin
     if (source_obj == dude_obj) then begin
-        if (((global_var(160) + global_var(159)) >= 25) and ((global_var(159) > (2 * global_var(160))) or (global_var(156) == 1))) then begin
-            set_global_var(156, 1);
-            set_global_var(157, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
+            set_global_var(GVAR_BERSERKER_REPUTATION, 1);
+            set_global_var(GVAR_CHAMPION_REPUTATION, 0);
         end
-        if (((global_var(160) + global_var(159)) >= 25) and ((global_var(160) > (3 * global_var(159))) or (global_var(157) == 1))) then begin
-            set_global_var(157, 1);
-            set_global_var(156, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(160) > (3 * global_var(GVAR_GOOD_MONSTER))) or (global_var(GVAR_CHAMPION_REPUTATION) == 1))) then begin
+            set_global_var(GVAR_CHAMPION_REPUTATION, 1);
+            set_global_var(GVAR_BERSERKER_REPUTATION, 0);
         end
-        set_global_var(159, global_var(159) + 1);
-        if ((global_var(159) % 2) == 0) then begin
-            set_global_var(155, global_var(155) - 1);
+        set_global_var(GVAR_GOOD_MONSTER, global_var(GVAR_GOOD_MONSTER) + 1);
+        if ((global_var(GVAR_GOOD_MONSTER) % 2) == 0) then begin
+            set_global_var(GVAR_PLAYER_REPUATION, global_var(GVAR_PLAYER_REPUATION) - 1);
         end
     end
 end
@@ -207,25 +207,25 @@ end
 procedure look_at_p_proc
 begin
     script_overrides;
-    display_msg(message_str(45, 100));
+    display_mstr(100);
 end
 
 procedure Harold00
 begin
-    if (get_critter_stat(dude_obj, 34) == 0) then begin
-        gsay_reply(45, 266);
+    if (dude_gender == 0) then begin
+        Reply(266);
     end
     else begin
-        gsay_reply(45, 267);
+        Reply(267);
     end
-    giq_option(4, 45, 102, Harold00a, 50);
-    giq_option(4, 45, 103, Harold00b, 50);
-    giq_option(4, 45, 104, Harold38, 50);
-    if (global_var(226) == 2) then begin
-        giq_option(4, 45, 225, Harold45, 50);
+    giq_option(4, NAME, 102, Harold00a, NEUTRAL_REACTION);
+    giq_option(4, NAME, 103, Harold00b, NEUTRAL_REACTION);
+    giq_option(4, NAME, 104, Harold38, NEUTRAL_REACTION);
+    if (global_var(GVAR_HUB_FILLER_29) == 2) then begin
+        giq_option(4, NAME, 225, Harold45, NEUTRAL_REACTION);
     end
-    giq_option(4, 45, 105, Harold37, 51);
-    giq_option(-3, 45, 106, Harold01, 50);
+    giq_option(4, NAME, 105, Harold37, BAD_REACTION);
+    giq_option(-3, NAME, 106, Harold01, NEUTRAL_REACTION);
 end
 
 procedure Harold00a
@@ -254,124 +254,124 @@ end
 
 procedure Harold01
 begin
-    gsay_message(45, 107, 50);
+    gsay_message(NAME, 107, NEUTRAL_REACTION);
 end
 
 procedure Harold02
 begin
-    if (get_critter_stat(dude_obj, 34) == 0) then begin
-        gsay_reply(45, 268);
+    if (dude_gender == 0) then begin
+        Reply(268);
     end
     else begin
-        gsay_reply(45, 269);
+        Reply(269);
     end
-    giq_option(4, 45, 109, Harold03, 50);
-    if (global_var(226) == 2) then begin
-        giq_option(4, 45, 225, Harold45, 50);
+    giq_option(4, NAME, 109, Harold03, NEUTRAL_REACTION);
+    if (global_var(GVAR_HUB_FILLER_29) == 2) then begin
+        giq_option(4, NAME, 225, Harold45, NEUTRAL_REACTION);
     end
-    giq_option(4, 45, 110, HaroldEnd, 50);
+    giq_option(4, NAME, 110, HaroldEnd, NEUTRAL_REACTION);
 end
 
 procedure Harold03
 begin
-    gsay_reply(45, 111);
-    giq_option(4, 45, 112, Harold04, 50);
-    giq_option(4, 45, 113, Harold36, 50);
-    giq_option(4, 45, 114, HaroldEnd, 50);
+    Reply(111);
+    giq_option(4, NAME, 112, Harold04, NEUTRAL_REACTION);
+    giq_option(4, NAME, 113, Harold36, NEUTRAL_REACTION);
+    giq_option(4, NAME, 114, HaroldEnd, NEUTRAL_REACTION);
 end
 
 procedure Harold04
 begin
-    gsay_reply(45, 115);
-    giq_option(4, 45, 116, Harold05, 50);
-    giq_option(4, 45, 117, Harold32, 50);
+    Reply(115);
+    giq_option(4, NAME, 116, Harold05, NEUTRAL_REACTION);
+    giq_option(4, NAME, 117, Harold32, NEUTRAL_REACTION);
 end
 
 procedure Harold05
 begin
-    gsay_reply(45, 118);
-    giq_option(4, 45, 119, Harold07, 50);
-    giq_option(4, 45, 120, Harold06, 50);
-    giq_option(4, 45, 121, Harold30, 50);
+    Reply(118);
+    giq_option(4, NAME, 119, Harold07, NEUTRAL_REACTION);
+    giq_option(4, NAME, 120, Harold06, NEUTRAL_REACTION);
+    giq_option(4, NAME, 121, Harold30, NEUTRAL_REACTION);
 end
 
 procedure Harold06
 begin
-    gsay_reply(45, 122);
-    giq_option(4, 45, 123, Harold07, 50);
+    Reply(122);
+    giq_option(4, NAME, 123, Harold07, NEUTRAL_REACTION);
 end
 
 procedure Harold07
 begin
-    gsay_reply(45, 124);
-    giq_option(4, 45, 125, Harold08, 50);
+    Reply(124);
+    giq_option(4, NAME, 125, Harold08, NEUTRAL_REACTION);
 end
 
 procedure Harold08
 begin
-    gsay_reply(45, 126);
-    giq_option(4, 45, 127, Harold09, 50);
-    giq_option(4, 45, 128, Harold10, 50);
+    Reply(126);
+    giq_option(4, NAME, 127, Harold09, NEUTRAL_REACTION);
+    giq_option(4, NAME, 128, Harold10, NEUTRAL_REACTION);
 end
 
 procedure Harold09
 begin
-    gsay_reply(45, 129);
-    giq_option(4, 45, 130, Harold09a, 50);
+    Reply(129);
+    giq_option(4, NAME, 130, Harold09a, NEUTRAL_REACTION);
 end
 
 procedure Harold09a
 begin
-    gsay_reply(45, 131);
-    giq_option(4, 45, 132, Harold11, 50);
+    Reply(131);
+    giq_option(4, NAME, 132, Harold11, NEUTRAL_REACTION);
 end
 
 procedure Harold10
 begin
-    gsay_reply(45, 133);
-    giq_option(4, 45, 134, Harold11, 50);
-    giq_option(4, 45, 135, Harold12, 50);
-    giq_option(4, 45, 136, Harold10a, 50);
+    Reply(133);
+    giq_option(4, NAME, 134, Harold11, NEUTRAL_REACTION);
+    giq_option(4, NAME, 135, Harold12, NEUTRAL_REACTION);
+    giq_option(4, NAME, 136, Harold10a, NEUTRAL_REACTION);
 end
 
 procedure Harold10a
 begin
-    gsay_message(45, 137, 50);
+    gsay_message(NAME, 137, NEUTRAL_REACTION);
     call Harold11();
 end
 
 procedure Harold11
 begin
-    gsay_reply(45, 138);
-    giq_option(4, 45, 139, Harold14, 50);
-    giq_option(4, 45, 140, Harold13, 50);
+    Reply(138);
+    giq_option(4, NAME, 139, Harold14, NEUTRAL_REACTION);
+    giq_option(4, NAME, 140, Harold13, NEUTRAL_REACTION);
 end
 
 procedure Harold12
 begin
-    gsay_reply(45, 141);
-    giq_option(4, 45, 142, Harold11, 50);
+    Reply(141);
+    giq_option(4, NAME, 142, Harold11, NEUTRAL_REACTION);
 end
 
 procedure Harold13
 begin
-    gsay_reply(45, 143);
-    giq_option(4, 45, 144, Harold15, 50);
-    giq_option(4, 45, 145, Harold14, 50);
+    Reply(143);
+    giq_option(4, NAME, 144, Harold15, NEUTRAL_REACTION);
+    giq_option(4, NAME, 145, Harold14, NEUTRAL_REACTION);
 end
 
 procedure Harold14
 begin
-    gsay_reply(45, 146);
-    giq_option(4, 45, 147, Harold18, 50);
+    Reply(146);
+    giq_option(4, NAME, 147, Harold18, NEUTRAL_REACTION);
 end
 
 procedure Harold15
 begin
-    gsay_reply(45, 148);
-    giq_option(4, 45, 149, Harold14, 50);
-    giq_option(4, 45, 150, Harold15a, 50);
-    giq_option(4, 45, 152, HaroldEnd, 50);
+    Reply(148);
+    giq_option(4, NAME, 149, Harold14, NEUTRAL_REACTION);
+    giq_option(4, NAME, 150, Harold15a, NEUTRAL_REACTION);
+    giq_option(4, NAME, 152, HaroldEnd, NEUTRAL_REACTION);
 end
 
 procedure Harold15a
@@ -389,13 +389,13 @@ end
 
 procedure Harold16
 begin
-    gsay_message(45, 153, 50);
-    giq_option(4, 45, 154, Harold14, 50);
+    gsay_message(NAME, 153, NEUTRAL_REACTION);
+    giq_option(4, NAME, 154, Harold14, NEUTRAL_REACTION);
 end
 
 procedure Harold17
 begin
-    gsay_message(45, 156, 50);
+    gsay_message(NAME, 156, NEUTRAL_REACTION);
     if (local_var(1) >= 3) then begin
         call Harold16();
     end
@@ -406,173 +406,173 @@ end
 
 procedure Harold17a
 begin
-    gsay_reply(45, 157);
-    giq_option(4, 45, 158, HaroldEnd, 50);
-    giq_option(4, 45, 159, Harold14, 50);
+    Reply(157);
+    giq_option(4, NAME, 158, HaroldEnd, NEUTRAL_REACTION);
+    giq_option(4, NAME, 159, Harold14, NEUTRAL_REACTION);
 end
 
 procedure Harold18
 begin
-    gsay_reply(45, 160);
-    giq_option(4, 45, 161, Harold20, 50);
-    giq_option(4, 45, 162, Harold19, 50);
+    Reply(160);
+    giq_option(4, NAME, 161, Harold20, NEUTRAL_REACTION);
+    giq_option(4, NAME, 162, Harold19, NEUTRAL_REACTION);
 end
 
 procedure Harold19
 begin
-    gsay_reply(45, 163);
-    giq_option(4, 45, 164, Harold21, 50);
+    Reply(163);
+    giq_option(4, NAME, 164, Harold21, NEUTRAL_REACTION);
 end
 
 procedure Harold20
 begin
-    gsay_reply(45, 165);
-    giq_option(4, 45, 166, Harold21, 50);
+    Reply(165);
+    giq_option(4, NAME, 166, Harold21, NEUTRAL_REACTION);
 end
 
 procedure Harold21
 begin
-    gsay_reply(45, 167);
-    giq_option(4, 45, 168, Harold22, 50);
+    Reply(167);
+    giq_option(4, NAME, 168, Harold22, NEUTRAL_REACTION);
 end
 
 procedure Harold22
 begin
-    gsay_reply(45, 169);
-    giq_option(4, 45, 170, Harold23, 50);
-    giq_option(4, 45, 171, Harold24, 50);
+    Reply(169);
+    giq_option(4, NAME, 170, Harold23, NEUTRAL_REACTION);
+    giq_option(4, NAME, 171, Harold24, NEUTRAL_REACTION);
 end
 
 procedure Harold23
 begin
-    gsay_reply(45, 172);
-    giq_option(4, 45, 173, Harold24, 50);
+    Reply(172);
+    giq_option(4, NAME, 173, Harold24, NEUTRAL_REACTION);
 end
 
 procedure Harold24
 begin
-    gsay_reply(45, 174);
-    giq_option(4, 45, 175, Harold25, 50);
+    Reply(174);
+    giq_option(4, NAME, 175, Harold25, NEUTRAL_REACTION);
 end
 
 procedure Harold25
 begin
-    gsay_reply(45, 176);
-    giq_option(4, 45, 177, Harold25a, 50);
-    giq_option(4, 45, 178, Harold27, 50);
+    Reply(176);
+    giq_option(4, NAME, 177, Harold25a, NEUTRAL_REACTION);
+    giq_option(4, NAME, 178, Harold27, NEUTRAL_REACTION);
 end
 
 procedure Harold25a
 begin
-    gsay_reply(45, 179);
-    giq_option(4, 45, 180, Harold26, 50);
+    Reply(179);
+    giq_option(4, NAME, 180, Harold26, NEUTRAL_REACTION);
 end
 
 procedure Harold26
 begin
-    gsay_reply(45, 181);
-    giq_option(4, 45, 182, Harold28, 50);
-    giq_option(4, 45, 183, Harold26a, 50);
+    Reply(181);
+    giq_option(4, NAME, 182, Harold28, NEUTRAL_REACTION);
+    giq_option(4, NAME, 183, Harold26a, NEUTRAL_REACTION);
 end
 
 procedure Harold26a
 begin
     set_local_var(5, 1);
     if (local_var(6) == 1) then begin
-        gsay_reply(45, 184);
-        giq_option(4, 45, 185, Harold26b, 50);
+        Reply(184);
+        giq_option(4, NAME, 185, Harold26b, NEUTRAL_REACTION);
     end
     else begin
-        gsay_message(45, 184, 50);
+        gsay_message(NAME, 184, NEUTRAL_REACTION);
     end
 end
 
 procedure Harold26b
 begin
-    gsay_message(45, 187, 50);
+    gsay_message(NAME, 187, NEUTRAL_REACTION);
 end
 
 procedure Harold27
 begin
-    gsay_reply(45, 188);
-    giq_option(4, 45, 189, Harold26, 50);
+    Reply(188);
+    giq_option(4, NAME, 189, Harold26, NEUTRAL_REACTION);
 end
 
 procedure Harold28
 begin
-    gsay_reply(45, 190);
-    giq_option(4, 45, 191, Harold29, 50);
-    giq_option(4, 45, 192, Harold26a, 50);
+    Reply(190);
+    giq_option(4, NAME, 191, Harold29, NEUTRAL_REACTION);
+    giq_option(4, NAME, 192, Harold26a, NEUTRAL_REACTION);
 end
 
 procedure Harold29
 begin
-    gsay_reply(45, 193);
-    giq_option(4, 45, 194, Harold26a, 50);
+    Reply(193);
+    giq_option(4, NAME, 194, Harold26a, NEUTRAL_REACTION);
 end
 
 procedure Harold30
 begin
-    gsay_reply(45, 195);
-    giq_option(4, 45, 196, Harold31, 50);
+    Reply(195);
+    giq_option(4, NAME, 196, Harold31, NEUTRAL_REACTION);
 end
 
 procedure Harold31
 begin
-    gsay_reply(45, 197);
-    giq_option(4, 45, 198, Harold07, 50);
+    Reply(197);
+    giq_option(4, NAME, 198, Harold07, NEUTRAL_REACTION);
 end
 
 procedure Harold32
 begin
-    gsay_reply(45, 199);
-    giq_option(4, 45, 200, Harold11, 50);
+    Reply(199);
+    giq_option(4, NAME, 200, Harold11, NEUTRAL_REACTION);
 end
 
 procedure Harold33
 begin
-    gsay_reply(45, 201);
-    giq_option(4, 45, 202, Harold34, 50);
-    giq_option(4, 45, 203, HaroldEnd, 50);
+    Reply(201);
+    giq_option(4, NAME, 202, Harold34, NEUTRAL_REACTION);
+    giq_option(4, NAME, 203, HaroldEnd, NEUTRAL_REACTION);
 end
 
 procedure Harold34
 begin
-    gsay_reply(45, 204);
-    giq_option(4, 45, 205, Harold05, 50);
+    Reply(204);
+    giq_option(4, NAME, 205, Harold05, NEUTRAL_REACTION);
 end
 
 procedure Harold36
 begin
-    gsay_message(45, 207, 50);
+    gsay_message(NAME, 207, NEUTRAL_REACTION);
 end
 
 procedure Harold37
 begin
     call BigDownReact();
-    gsay_message(45, 208, 51);
+    gsay_message(NAME, 208, BAD_REACTION);
 end
 
 procedure Harold38
 begin
-    gsay_message(45, 209, 50);
+    gsay_message(NAME, 209, NEUTRAL_REACTION);
 end
 
 procedure Harold39
 begin
-    if (get_critter_stat(dude_obj, 34) == 0) then begin
-        gsay_reply(45, 272);
+    if (dude_gender == 0) then begin
+        Reply(272);
     end
     else begin
-        gsay_reply(45, 273);
+        Reply(273);
     end
-    giq_option(4, 45, 211, Harold39a, 50);
-    giq_option(4, 45, 212, Harold39b, 50);
-    if (global_var(226) == 2) then begin
-        giq_option(4, 45, 225, Harold45, 50);
+    giq_option(4, NAME, 211, Harold39a, NEUTRAL_REACTION);
+    giq_option(4, NAME, 212, Harold39b, NEUTRAL_REACTION);
+    if (global_var(GVAR_HUB_FILLER_29) == 2) then begin
+        giq_option(4, NAME, 225, Harold45, NEUTRAL_REACTION);
     end
-    giq_option(4, 45, 213, Harold38, 50);
-    giq_option(-3, 45, 214, Harold01, 50);
+    giq_option(4, NAME, 213, Harold38, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 214, Harold01, NEUTRAL_REACTION);
 end
 
 procedure Harold39a
@@ -601,17 +601,17 @@ end
 
 procedure Harold40
 begin
-    gsay_reply(45, 215);
-    giq_option(4, 45, 216, Harold03, 50);
-    giq_option(4, 45, 217, HaroldEnd, 50);
+    Reply(215);
+    giq_option(4, NAME, 216, Harold03, NEUTRAL_REACTION);
+    giq_option(4, NAME, 217, HaroldEnd, NEUTRAL_REACTION);
 end
 
 procedure Harold41
 begin
-    gsay_reply(45, 218);
-    giq_option(4, 45, 219, Harold41a, 50);
-    giq_option(4, 45, 220, Harold43, 50);
-    giq_option(-3, 45, 221, HaroldEnd, 50);
+    Reply(218);
+    giq_option(4, NAME, 219, Harold41a, NEUTRAL_REACTION);
+    giq_option(4, NAME, 220, Harold43, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 221, HaroldEnd, NEUTRAL_REACTION);
 end
 
 procedure Harold41a
@@ -629,37 +629,37 @@ end
 procedure Harold42
 begin
     call DownReact();
-    gsay_message(45, 222, 51);
+    gsay_message(NAME, 222, BAD_REACTION);
 end
 
 procedure Harold43
 begin
-    gsay_reply(45, 223);
-    if (global_var(226) == 2) then begin
-        giq_option(4, 45, 225, Harold45, 50);
+    Reply(223);
+    if (global_var(GVAR_HUB_FILLER_29) == 2) then begin
+        giq_option(4, NAME, 225, Harold45, NEUTRAL_REACTION);
     end
-    if (((global_var(150) - (game_time / (10 * 60 * 60 * 24))) < 10) or (global_var(109) == 1)) then begin
-        giq_option(4, 45, 224, Harold43b, 50);
+    if (((global_var(GVAR_THE_HUB_INVADED_DATE) - (game_time / (10 * 60 * 60 * 24))) < 10) or (global_var(GVAR_FIND_LOST_INITIATE) == 1)) then begin
+        giq_option(4, NAME, 224, Harold43b, NEUTRAL_REACTION);
     end
     if (local_var(5) == 0) then begin
-        giq_option(4, 45, 226, Harold43a, 50);
+        giq_option(4, NAME, 226, Harold43a, NEUTRAL_REACTION);
     end
-    giq_option(4, 45, 265, HaroldEnd, 50);
+    giq_option(4, NAME, 265, HaroldEnd, NEUTRAL_REACTION);
 end
 
 procedure Harold43a
 begin
-    gsay_message(45, 228, 50);
+    gsay_message(NAME, 228, NEUTRAL_REACTION);
     call Harold04();
 end
 
 procedure Harold43b
 begin
-    if (global_var(109) == 1) then begin
+    if (global_var(GVAR_FIND_LOST_INITIATE) == 1) then begin
         call Harold53();
     end
     else begin
-        if ((global_var(150) - (game_time / (10 * 60 * 60 * 24))) < 10) then begin
+        if ((global_var(GVAR_THE_HUB_INVADED_DATE) - (game_time / (10 * 60 * 60 * 24))) < 10) then begin
             call Harold54();
         end
     end
@@ -667,74 +667,74 @@ end
 
 procedure Harold44
 begin
-    gsay_message(45, 229, 50);
+    gsay_message(NAME, 229, NEUTRAL_REACTION);
 end
 
 procedure Harold45
 begin
-    set_global_var(226, 3);
-    gsay_reply(45, 230);
-    giq_option(4, 45, 231, Harold47, 50);
-    giq_option(4, 45, 232, Harold47, 50);
+    set_global_var(GVAR_HUB_FILLER_29, 3);
+    Reply(230);
+    giq_option(4, NAME, 231, Harold47, NEUTRAL_REACTION);
+    giq_option(4, NAME, 232, Harold47, NEUTRAL_REACTION);
 end
 
 procedure Harold47
 begin
-    gsay_reply(45, 233);
-    giq_option(4, 45, 234, Harold49, 50);
+    Reply(233);
+    giq_option(4, NAME, 234, Harold49, NEUTRAL_REACTION);
 end
 
 procedure Harold49
 begin
-    gsay_reply(45, 235);
-    giq_option(4, 45, 236, Harold50, 50);
+    Reply(235);
+    giq_option(4, NAME, 236, Harold50, NEUTRAL_REACTION);
 end
 
 procedure Harold50
 begin
-    gsay_reply(45, 237);
-    giq_option(4, 45, 238, Harold51, 50);
+    Reply(237);
+    giq_option(4, NAME, 238, Harold51, NEUTRAL_REACTION);
 end
 
 procedure Harold51
 begin
-    gsay_reply(45, 239);
-    giq_option(4, 45, 240, HaroldEnd, 50);
+    Reply(239);
+    giq_option(4, NAME, 240, HaroldEnd, NEUTRAL_REACTION);
 end
 
 procedure Harold52
 begin
-    gsay_message(45, 241, 50);
+    gsay_message(NAME, 241, NEUTRAL_REACTION);
 end
 
 procedure Harold53
 begin
-    set_global_var(219, 1);
-    gsay_reply(45, 242);
-    giq_option(4, 45, 243, HaroldEnd, 50);
+    set_global_var(GVAR_DECKER_KNOWN, 1);
+    Reply(242);
+    giq_option(4, NAME, 243, HaroldEnd, NEUTRAL_REACTION);
 end
 
 procedure Harold54
 begin
-    gsay_reply(45, 244);
-    giq_option(4, 45, 245, HaroldEnd, 50);
+    Reply(244);
+    giq_option(4, NAME, 245, HaroldEnd, NEUTRAL_REACTION);
 end
 
 procedure Harold55
 begin
-    gsay_reply(45, 246);
-    giq_option(4, 45, 247, Harold56, 50);
-    giq_option(4, 45, 248, Harold57, 50);
+    Reply(246);
+    giq_option(4, NAME, 247, Harold56, NEUTRAL_REACTION);
+    giq_option(4, NAME, 248, Harold57, NEUTRAL_REACTION);
 end
 
 procedure Harold56
 begin
-    gsay_message(45, 249, 51);
+    gsay_message(NAME, 249, BAD_REACTION);
 end
 
 procedure Harold57
 begin
-    gsay_message(45, 250, 51);
+    gsay_message(NAME, 250, BAD_REACTION);
 end
 
 procedure HaroldEnd
@@ -747,31 +747,31 @@ begin
         set_local_var(0, 50);
         set_local_var(1, 2);
         set_local_var(2, 1);
-        set_local_var(0, local_var(0) + (5 * get_critter_stat(dude_obj, 3)) - 25);
-        set_local_var(0, local_var(0) + (10 * has_trait(0, dude_obj, 10)));
-        if (has_trait(0, dude_obj, 39)) then begin
-            if (global_var(155) > 0) then begin
-                set_local_var(0, local_var(0) + global_var(155));
+        set_local_var(0, local_var(0) + (5 * dude_charisma) - 25);
+        set_local_var(0, local_var(0) + (10 * has_trait(TRAIT_PERK, dude_obj, PERK_presence)));
+        if (has_trait(TRAIT_PERK, dude_obj, PERK_cult_of_personality)) then begin
+            if (global_var(GVAR_PLAYER_REPUATION) > 0) then begin
+                set_local_var(0, local_var(0) + global_var(GVAR_PLAYER_REPUATION));
             end
             else begin
-                set_local_var(0, local_var(0) - global_var(155));
+                set_local_var(0, local_var(0) - global_var(GVAR_PLAYER_REPUATION));
             end
         end
         else begin
             if (local_var(3) == 1) then begin
-                set_local_var(0, local_var(0) - global_var(155));
+                set_local_var(0, local_var(0) - global_var(GVAR_PLAYER_REPUATION));
             end
             else begin
-                set_local_var(0, local_var(0) + global_var(155));
+                set_local_var(0, local_var(0) + global_var(GVAR_PLAYER_REPUATION));
             end
         end
-        if (global_var(158) > 2) then begin
+        if (global_var(GVAR_CHILDKILLER_REPUATION) > 2) then begin
             set_local_var(0, local_var(0) - 30);
         end
-        if (((global_var(160) + global_var(159)) >= 25) and ((global_var(160) > (3 * global_var(159))) or (global_var(157) == 1))) then begin
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(160) > (3 * global_var(GVAR_GOOD_MONSTER))) or (global_var(GVAR_CHAMPION_REPUTATION) == 1))) then begin
             set_local_var(0, local_var(0) + 20);
         end
-        if (((global_var(160) + global_var(159)) >= 25) and ((global_var(159) > (2 * global_var(160))) or (global_var(156) == 1))) then begin
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
             set_local_var(0, local_var(0) - 20);
         end
         call ReactToLevel();
@@ -864,6 +864,5 @@ end
 
 procedure Goodbyes
 begin
-    exit_line := message_str(634, random(100, 105));
+    exit_line := message_str(SCRIPT_MODREACT, random(100, 105));
 end
-
