diff --git a/Games/Fallout1.Steam/Original/HIGHTOWR.ssl b/Games/Fallout1.Steam/ReDefine/HIGHTOWR.ssl
index 25c1669..2931904 100644
--- a/Games/Fallout1.Steam/Original/HIGHTOWR.ssl
+++ b/Games/Fallout1.Steam/ReDefine/HIGHTOWR.ssl
@@ -42,38 +42,38 @@ procedure start
 begin
     if (Only_Once) then begin
         Only_Once := 0;
-        if (global_var(111) == 5) then begin
+        if (global_var(GVAR_KILL_MERCHANT) == 5) then begin
             set_obj_visibility(self_obj, 1);
             visible := 0;
         end
-        critter_add_trait(self_obj, 1, 6, 42);
-        critter_add_trait(self_obj, 1, 5, 5);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 42);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_LOSER);
     end
-    if (script_action == 21) then begin
+    if (script_action == look_at_proc) then begin
         call look_at_p_proc();
     end
     else begin
-        if (script_action == 4) then begin
+        if (script_action == pickup_proc) then begin
             call pickup_p_proc();
         end
         else begin
-            if (script_action == 11) then begin
+            if (script_action == talk_proc) then begin
                 call talk_p_proc();
             end
             else begin
-                if (script_action == 12) then begin
+                if (script_action == critter_proc) then begin
                     call critter_p_proc();
                 end
                 else begin
-                    if (script_action == 18) then begin
+                    if (script_action == destroy_proc) then begin
                         call destroy_p_proc();
                     end
                     else begin
-                        if (script_action == 14) then begin
+                        if (script_action == damage_proc) then begin
                             call damage_p_proc();
                         end
                         else begin
-                            if (script_action == 13) then begin
+                            if (script_action == combat_proc) then begin
                                 call combat_p_proc();
                             end
                         end
@@ -97,7 +97,7 @@ begin
     else begin
         if (hostile) then begin
             hostile := 0;
-            attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+            attack(dude_obj);
         end
         if (map_var(5) == 1) then begin
             if ((obj_can_hear_obj(self_obj, dude_obj) == 1) or (obj_can_see_obj(self_obj, dude_obj) == 1)) then begin
@@ -110,7 +110,7 @@ begin
                     animate_move_obj_to_tile(self_obj, 25125, 0);
                 end
                 else begin
-                    if ((tile_num(dude_obj) == 24522) and (map_var(1) == 0)) then begin
+                    if ((dude_tile == 24522) and (map_var(1) == 0)) then begin
                         dialogue_system_enter;
                     end
                 end
@@ -135,12 +135,12 @@ procedure talk_p_proc
 begin
     call get_reaction();
     if ((game_time_hour >= 1900) or (game_time_hour < 600)) then begin
-        float_msg(self_obj, message_str(582, 127), 2);
+        float_msg(self_obj, mstr(127), FLOAT_MSG_RED);
         call combat();
     end
     else begin
-        if (global_var(248) == 1) then begin
-            float_msg(self_obj, message_str(582, 127), 2);
+        if (global_var(GVAR_ENEMY_HUB) == 1) then begin
+            float_msg(self_obj, mstr(127), FLOAT_MSG_RED);
             call combat();
         end
         else begin
@@ -153,7 +153,7 @@ begin
                 end_dialogue;
             end
             else begin
-                float_msg(self_obj, message_str(582, 127), 2);
+                float_msg(self_obj, mstr(127), FLOAT_MSG_RED);
                 call combat();
             end
         end
@@ -163,35 +163,35 @@ end
 procedure destroy_p_proc
 begin
     if (source_obj == dude_obj) then begin
-        if (((global_var(160) + global_var(159)) >= 25) and ((global_var(159) > (2 * global_var(160))) or (global_var(156) == 1))) then begin
-            set_global_var(156, 1);
-            set_global_var(157, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
+            set_global_var(GVAR_BERSERKER_REPUTATION, 1);
+            set_global_var(GVAR_CHAMPION_REPUTATION, 0);
         end
-        if (((global_var(160) + global_var(159)) >= 25) and ((global_var(160) > (3 * global_var(159))) or (global_var(157) == 1))) then begin
-            set_global_var(157, 1);
-            set_global_var(156, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(160) > (3 * global_var(GVAR_GOOD_MONSTER))) or (global_var(GVAR_CHAMPION_REPUTATION) == 1))) then begin
+            set_global_var(GVAR_CHAMPION_REPUTATION, 1);
+            set_global_var(GVAR_BERSERKER_REPUTATION, 0);
         end
-        set_global_var(159, global_var(159) + 1);
-        if ((global_var(159) % 2) == 0) then begin
-            set_global_var(155, global_var(155) - 1);
+        set_global_var(GVAR_GOOD_MONSTER, global_var(GVAR_GOOD_MONSTER) + 1);
+        if ((global_var(GVAR_GOOD_MONSTER) % 2) == 0) then begin
+            set_global_var(GVAR_PLAYER_REPUATION, global_var(GVAR_PLAYER_REPUATION) - 1);
         end
     end
     set_map_var(0, map_var(0) + 1);
     if (map_var(0) > 1) then begin
-        set_global_var(111, 2);
+        set_global_var(GVAR_KILL_MERCHANT, 2);
     end
 end
 
 procedure look_at_p_proc
 begin
     script_overrides;
-    display_msg(message_str(582, 100));
+    display_mstr(100);
 end
 
 procedure damage_p_proc
 begin
     set_map_var(5, 1);
-    set_global_var(111, 5);
+    set_global_var(GVAR_KILL_MERCHANT, 5);
 end
 
 procedure combat_p_proc
@@ -201,70 +201,70 @@ end
 
 procedure Daren01
 begin
-    gsay_reply(582, 101);
-    giq_option(4, 582, 102, Daren02, 50);
-    if (global_var(111) == 1) then begin
-        giq_option(4, 582, 103, Daren03, 50);
+    Reply(101);
+    giq_option(4, NAME, 102, Daren02, NEUTRAL_REACTION);
+    if (global_var(GVAR_KILL_MERCHANT) == 1) then begin
+        giq_option(4, NAME, 103, Daren03, NEUTRAL_REACTION);
     end
-    giq_option(4, 582, message_str(582, 104) + proto_data(obj_pid(dude_obj), 1) + message_str(582, 105), Daren04, 50);
-    giq_option(4, 582, 106, Daren07, 50);
-    giq_option(4, 582, 107, Daren06, 50);
-    giq_option(-3, 582, 108, Daren06, 50);
+    giq_option(4, NAME, mstr(104) + proto_data(obj_pid(dude_obj), 1) + mstr(105), Daren04, NEUTRAL_REACTION);
+    giq_option(4, NAME, 106, Daren07, NEUTRAL_REACTION);
+    giq_option(4, NAME, 107, Daren06, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 108, Daren06, NEUTRAL_REACTION);
 end
 
 procedure Daren02
 begin
-    gsay_reply(582, 109);
-    if (global_var(111) == 1) then begin
-        giq_option(4, 582, 110, DarenCombat, 51);
+    Reply(109);
+    if (global_var(GVAR_KILL_MERCHANT) == 1) then begin
+        giq_option(4, NAME, 110, DarenCombat, BAD_REACTION);
     end
-    giq_option(4, 582, 111, Daren05, 50);
-    giq_option(4, 582, 112, Daren06, 50);
-    giq_option(4, 582, 113, Daren06, 50);
+    giq_option(4, NAME, 111, Daren05, NEUTRAL_REACTION);
+    giq_option(4, NAME, 112, Daren06, NEUTRAL_REACTION);
+    giq_option(4, NAME, 113, Daren06, NEUTRAL_REACTION);
 end
 
 procedure Daren03
 begin
-    gsay_reply(582, 114);
-    giq_option(4, 582, 115, Daren06, 50);
-    giq_option(4, 582, 116, Daren07, 50);
-    giq_option(4, 582, 117, DarenCombat, 51);
-    giq_option(4, 582, 118, Daren06, 50);
+    Reply(114);
+    giq_option(4, NAME, 115, Daren06, NEUTRAL_REACTION);
+    giq_option(4, NAME, 116, Daren07, NEUTRAL_REACTION);
+    giq_option(4, NAME, 117, DarenCombat, BAD_REACTION);
+    giq_option(4, NAME, 118, Daren06, NEUTRAL_REACTION);
 end
 
 procedure Daren04
 begin
-    gsay_message(582, message_str(582, 119) + proto_data(obj_pid(dude_obj), 1) + message_str(582, 120), 51);
+    gsay_message(NAME, mstr(119) + proto_data(obj_pid(dude_obj), 1) + mstr(120), BAD_REACTION);
     call DarenEndLeon();
 end
 
 procedure Daren05
 begin
-    gsay_reply(582, 121);
-    giq_option(4, 582, 122, Daren07, 50);
+    Reply(121);
+    giq_option(4, NAME, 122, Daren07, NEUTRAL_REACTION);
     if (item_caps_total(dude_obj) >= 100) then begin
-        giq_option(4, 582, 123, Daren05a, 50);
+        giq_option(4, NAME, 123, Daren05a, NEUTRAL_REACTION);
     end
-    giq_option(4, 582, 124, Daren06, 50);
+    giq_option(4, NAME, 124, Daren06, NEUTRAL_REACTION);
 end
 
 procedure Daren05a
 begin
     variable LVar0 := 0;
     LVar0 := item_caps_adjust(dude_obj, -100);
-    gsay_message(582, 125, 51);
+    gsay_message(NAME, 125, BAD_REACTION);
     call DarenEndLeon();
 end
 
 procedure Daren06
 begin
-    gsay_message(582, 126, 51);
+    gsay_message(NAME, 126, BAD_REACTION);
     call DarenEndLeon();
 end
 
 procedure Daren07
 begin
-    gsay_message(582, 127, 51);
+    gsay_message(NAME, 127, BAD_REACTION);
     call DarenCombat();
 end
 
@@ -284,31 +284,31 @@ begin
         set_local_var(0, 50);
         set_local_var(1, 2);
         set_local_var(2, 1);
-        set_local_var(0, local_var(0) + (5 * get_critter_stat(dude_obj, 3)) - 25);
-        set_local_var(0, local_var(0) + (10 * has_trait(0, dude_obj, 10)));
-        if (has_trait(0, dude_obj, 39)) then begin
-            if (global_var(155) > 0) then begin
-                set_local_var(0, local_var(0) + global_var(155));
+        set_local_var(0, local_var(0) + (5 * dude_charisma) - 25);
+        set_local_var(0, local_var(0) + (10 * has_trait(TRAIT_PERK, dude_obj, PERK_presence)));
+        if (has_trait(TRAIT_PERK, dude_obj, PERK_cult_of_personality)) then begin
+            if (global_var(GVAR_PLAYER_REPUATION) > 0) then begin
+                set_local_var(0, local_var(0) + global_var(GVAR_PLAYER_REPUATION));
             end
             else begin
-                set_local_var(0, local_var(0) - global_var(155));
+                set_local_var(0, local_var(0) - global_var(GVAR_PLAYER_REPUATION));
             end
         end
         else begin
             if (local_var(3) == 1) then begin
-                set_local_var(0, local_var(0) - global_var(155));
+                set_local_var(0, local_var(0) - global_var(GVAR_PLAYER_REPUATION));
             end
             else begin
-                set_local_var(0, local_var(0) + global_var(155));
+                set_local_var(0, local_var(0) + global_var(GVAR_PLAYER_REPUATION));
             end
         end
-        if (global_var(158) > 2) then begin
+        if (global_var(GVAR_CHILDKILLER_REPUATION) > 2) then begin
             set_local_var(0, local_var(0) - 30);
         end
-        if (((global_var(160) + global_var(159)) >= 25) and ((global_var(160) > (3 * global_var(159))) or (global_var(157) == 1))) then begin
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(160) > (3 * global_var(GVAR_GOOD_MONSTER))) or (global_var(GVAR_CHAMPION_REPUTATION) == 1))) then begin
             set_local_var(0, local_var(0) + 20);
         end
-        if (((global_var(160) + global_var(159)) >= 25) and ((global_var(159) > (2 * global_var(160))) or (global_var(156) == 1))) then begin
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
             set_local_var(0, local_var(0) - 20);
         end
         call ReactToLevel();
@@ -401,6 +401,5 @@ end
 
 procedure Goodbyes
 begin
-    exit_line := message_str(634, random(100, 105));
+    exit_line := message_str(SCRIPT_MODREACT, random(100, 105));
 end
-
