diff --git a/Games/Fallout1.Steam/Original/HOTSPOT.ssl b/Games/Fallout1.Steam/ReDefine/HOTSPOT.ssl
index 08ab1d3..b76a6b0 100644
--- a/Games/Fallout1.Steam/Original/HOTSPOT.ssl
+++ b/Games/Fallout1.Steam/ReDefine/HOTSPOT.ssl
@@ -5,9 +5,8 @@ variable Rads;
 
 procedure start
 begin
-    if (script_action == 2) then begin
+    if (script_action == spatial_proc) then begin
         Rads := random(20, 50);
         radiation_inc(dude_obj, Rads);
     end
 end
-
