diff --git a/Games/Fallout1.Steam/Original/HTWRBOX.ssl b/Games/Fallout1.Steam/ReDefine/HTWRBOX.ssl
index 403f3e1..e62cca5 100644
--- a/Games/Fallout1.Steam/Original/HTWRBOX.ssl
+++ b/Games/Fallout1.Steam/ReDefine/HTWRBOX.ssl
@@ -10,19 +10,19 @@ variable test;
 
 procedure start
 begin
-    if (script_action == 21) then begin
+    if (script_action == look_at_proc) then begin
         call look_at_p_proc();
     end
     else begin
-        if (script_action == 6) then begin
+        if (script_action == use_proc) then begin
             call use_p_proc();
         end
         else begin
-            if (script_action == 7) then begin
+            if (script_action == use_obj_on_proc) then begin
                 call use_obj_on_p_proc();
             end
             else begin
-                if (script_action == 8) then begin
+                if (script_action == use_skill_on_proc) then begin
                     call use_skill_on_p_proc();
                 end
             end
@@ -34,10 +34,10 @@ procedure look_at_p_proc
 begin
     script_overrides;
     if (local_var(0) == 0) then begin
-        display_msg(message_str(872, 100));
+        display_mstr(100);
     end
     else begin
-        display_msg(message_str(872, 101));
+        display_mstr(101);
     end
 end
 
@@ -45,7 +45,7 @@ procedure use_p_proc
 begin
     if (local_var(0) == 0) then begin
         script_overrides;
-        display_msg(message_str(872, 102));
+        display_mstr(102);
     end
     else begin
         if (local_var(1) == 0) then begin
@@ -58,19 +58,19 @@ begin
                     if (is_success(test)) then begin
                         set_local_var(2, 1);
                         set_local_var(1, 1);
-                        display_msg(message_str(872, 103));
+                        display_mstr(103);
                     end
                     else begin
                         if (is_critical(test)) then begin
                             set_local_var(2, 1);
                             set_local_var(1, 1);
-                            display_msg(message_str(872, 104));
+                            display_mstr(104);
                             call set_off_trap();
                         end
                         else begin
                             set_local_var(2, 1);
                             set_local_var(1, 0);
-                            display_msg(message_str(872, 105));
+                            display_mstr(105);
                         end
                     end
                 end
@@ -95,10 +95,10 @@ begin
             test := roll_vs_skill(dude_obj, 9, 0);
             if (is_success(test)) then begin
                 set_local_var(0, 1);
-                display_msg(message_str(872, 107));
+                display_mstr(107);
                 set_local_var(3, local_var(3) + 1);
                 if ((local_var(3) >= 3) and (local_var(1) == 0)) then begin
-                    display_msg(message_str(872, 108));
+                    display_mstr(108);
                     set_local_var(1, 1);
                     call set_off_trap();
                 end
@@ -106,19 +106,19 @@ begin
             else begin
                 if (is_critical(test)) then begin
                     jam_lock(self_obj);
-                    display_msg(message_str(872, 109));
+                    display_mstr(109);
                     set_local_var(3, local_var(3) + 1);
                     if ((local_var(3) >= 3) and (local_var(1) == 0)) then begin
-                        display_msg(message_str(872, 108));
+                        display_mstr(108);
                         set_local_var(1, 1);
                         call set_off_trap();
                     end
                 end
                 else begin
-                    display_msg(message_str(872, 110));
+                    display_mstr(110);
                     set_local_var(3, local_var(3) + 1);
                     if ((local_var(3) >= 3) and (local_var(1) == 0)) then begin
-                        display_msg(message_str(872, 108));
+                        display_mstr(108);
                         set_local_var(1, 1);
                         call set_off_trap();
                     end
@@ -126,7 +126,7 @@ begin
             end
         end
         else begin
-            display_msg(message_str(872, 106));
+            display_mstr(106);
         end
     end
 end
@@ -139,10 +139,10 @@ begin
             test := roll_vs_skill(dude_obj, 9, -20);
             if (is_success(test)) then begin
                 set_local_var(0, 1);
-                display_msg(message_str(872, 107));
+                display_mstr(107);
                 set_local_var(3, local_var(3) + 1);
                 if ((local_var(3) >= 3) and (local_var(1) == 0)) then begin
-                    display_msg(message_str(872, 108));
+                    display_mstr(108);
                     set_local_var(1, 1);
                     call set_off_trap();
                 end
@@ -150,19 +150,19 @@ begin
             else begin
                 if (is_critical(test)) then begin
                     jam_lock(self_obj);
-                    display_msg(message_str(872, 109));
+                    display_mstr(109);
                     set_local_var(3, local_var(3) + 1);
                     if ((local_var(3) >= 3) and (local_var(1) == 0)) then begin
-                        display_msg(message_str(872, 108));
+                        display_mstr(108);
                         set_local_var(1, 1);
                         call set_off_trap();
                     end
                 end
                 else begin
-                    display_msg(message_str(872, 110));
+                    display_mstr(110);
                     set_local_var(3, local_var(3) + 1);
                     if ((local_var(3) >= 3) and (local_var(1) == 0)) then begin
-                        display_msg(message_str(872, 108));
+                        display_mstr(108);
                         set_local_var(1, 1);
                         call set_off_trap();
                     end
@@ -170,7 +170,7 @@ begin
             end
         end
         else begin
-            display_msg(message_str(872, 106));
+            display_mstr(106);
         end
     end
     else begin
@@ -180,26 +180,26 @@ begin
                 if (is_success(test)) then begin
                     if (is_critical(test)) then begin
                         script_overrides;
-                        display_msg(message_str(872, 111));
+                        display_mstr(111);
                         set_local_var(2, 1);
                         set_local_var(1, 1);
                     end
                     else begin
                         script_overrides;
-                        display_msg(message_str(872, 112));
+                        display_mstr(112);
                         set_local_var(2, 1);
                     end
                 end
                 else begin
                     if (is_critical(test)) then begin
                         script_overrides;
-                        display_msg(message_str(872, 113));
+                        display_mstr(113);
                         set_local_var(1, 1);
                         call set_off_trap();
                     end
                     else begin
                         script_overrides;
-                        display_msg(message_str(872, 114));
+                        display_mstr(114);
                     end
                 end
             end
@@ -207,17 +207,17 @@ begin
                 script_overrides;
                 test := roll_vs_skill(dude_obj, 11, 0);
                 if (is_success(test)) then begin
-                    display_msg(message_str(872, 115));
+                    display_mstr(115);
                     set_local_var(1, 1);
                 end
                 else begin
                     if (is_critical(test)) then begin
-                        display_msg(message_str(872, 116));
+                        display_mstr(116);
                         set_local_var(1, 1);
                         call set_off_trap();
                     end
                     else begin
-                        display_msg(message_str(872, 117));
+                        display_mstr(117);
                     end
                 end
             end
@@ -233,4 +233,3 @@ begin
     obj_unlock(self_obj);
     obj_open(self_obj);
 end
-
