diff --git a/Games/Fallout1.Steam/Original/HUBJAKE.ssl b/Games/Fallout1.Steam/ReDefine/HUBJAKE.ssl
index 68a78fe..e12e72a 100644
--- a/Games/Fallout1.Steam/Original/HUBJAKE.ssl
+++ b/Games/Fallout1.Steam/ReDefine/HUBJAKE.ssl
@@ -54,26 +54,26 @@ begin
     gdialog_set_barter_mod(-15);
     if (Only_Once) then begin
         Only_Once := 0;
-        critter_add_trait(self_obj, 1, 6, 48);
-        critter_add_trait(self_obj, 1, 5, 50);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 48);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_MERCHANT);
     end
-    if (script_action == 21) then begin
+    if (script_action == look_at_proc) then begin
         call look_at_p_proc();
     end
     else begin
-        if (script_action == 4) then begin
+        if (script_action == pickup_proc) then begin
             call pickup_p_proc();
         end
         else begin
-            if (script_action == 11) then begin
+            if (script_action == talk_proc) then begin
                 call talk_p_proc();
             end
             else begin
-                if (script_action == 12) then begin
+                if (script_action == critter_proc) then begin
                     call critter_p_proc();
                 end
                 else begin
-                    if (script_action == 18) then begin
+                    if (script_action == destroy_proc) then begin
                         call destroy_p_proc();
                     end
                 end
@@ -92,7 +92,7 @@ procedure critter_p_proc
 begin
     if (hostile) then begin
         hostile := 0;
-        attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+        attack(dude_obj);
     end
 end
 
@@ -120,7 +120,7 @@ begin
         call combat();
     end
     else begin
-        if (global_var(248) == 1) then begin
+        if (global_var(GVAR_ENEMY_HUB) == 1) then begin
             start_gdialog(589, self_obj, 4, -1, -1);
             gsay_start;
             call Jake10();
@@ -162,17 +162,17 @@ procedure destroy_p_proc
 begin
     move_obj_inven_to_obj(Jake_Desk_Ptr, self_obj);
     if (source_obj == dude_obj) then begin
-        if (((global_var(160) + global_var(159)) >= 25) and ((global_var(159) > (2 * global_var(160))) or (global_var(156) == 1))) then begin
-            set_global_var(156, 1);
-            set_global_var(157, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
+            set_global_var(GVAR_BERSERKER_REPUTATION, 1);
+            set_global_var(GVAR_CHAMPION_REPUTATION, 0);
         end
-        if (((global_var(160) + global_var(159)) >= 25) and ((global_var(160) > (3 * global_var(159))) or (global_var(157) == 1))) then begin
-            set_global_var(157, 1);
-            set_global_var(156, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(160) > (3 * global_var(GVAR_GOOD_MONSTER))) or (global_var(GVAR_CHAMPION_REPUTATION) == 1))) then begin
+            set_global_var(GVAR_CHAMPION_REPUTATION, 1);
+            set_global_var(GVAR_BERSERKER_REPUTATION, 0);
         end
-        set_global_var(159, global_var(159) + 1);
-        if ((global_var(159) % 2) == 0) then begin
-            set_global_var(155, global_var(155) - 1);
+        set_global_var(GVAR_GOOD_MONSTER, global_var(GVAR_GOOD_MONSTER) + 1);
+        if ((global_var(GVAR_GOOD_MONSTER) % 2) == 0) then begin
+            set_global_var(GVAR_PLAYER_REPUATION, global_var(GVAR_PLAYER_REPUATION) - 1);
         end
     end
 end
@@ -180,7 +180,7 @@ end
 procedure look_at_p_proc
 begin
     script_overrides;
-    display_msg(message_str(589, 100));
+    display_mstr(100);
 end
 
 procedure damage_p_proc
@@ -188,127 +188,127 @@ begin
     variable LVar0 := 0;
     LVar0 := obj_pid(source_obj);
     if (party_member_obj(LVar0) != 0) then begin
-        set_global_var(248, 1);
+        set_global_var(GVAR_ENEMY_HUB, 1);
     end
 end
 
 procedure Jake01
 begin
-    gsay_reply(589, 101);
-    giq_option(4, 589, 102, Barter, 50);
-    giq_option(4, 589, 103, Jake02, 50);
-    if (global_var(101) != 2) then begin
-        giq_option(4, 589, 104, Jake03, 50);
+    Reply(101);
+    giq_option(4, NAME, 102, Barter, NEUTRAL_REACTION);
+    giq_option(4, NAME, 103, Jake02, NEUTRAL_REACTION);
+    if (global_var(GVAR_FIND_WATER_CHIP) != 2) then begin
+        giq_option(4, NAME, 104, Jake03, NEUTRAL_REACTION);
     end
-    giq_option(4, 589, 105, Jake04, 50);
-    giq_option(4, 589, 106, Jake13, 50);
-    giq_option(-3, 589, 107, Barter, 50);
-    giq_option(-3, 589, 108, Jake13, 50);
+    giq_option(4, NAME, 105, Jake04, NEUTRAL_REACTION);
+    giq_option(4, NAME, 106, Jake13, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 107, Barter, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 108, Jake13, NEUTRAL_REACTION);
 end
 
 procedure Jake02
 begin
-    gsay_reply(589, 109);
-    giq_option(4, 589, 110, Barter, 50);
-    giq_option(4, 589, 111, Jake05, 50);
-    giq_option(4, 589, 112, Jake06, 50);
-    giq_option(4, 589, 113, Jake13, 50);
+    Reply(109);
+    giq_option(4, NAME, 110, Barter, NEUTRAL_REACTION);
+    giq_option(4, NAME, 111, Jake05, NEUTRAL_REACTION);
+    giq_option(4, NAME, 112, Jake06, NEUTRAL_REACTION);
+    giq_option(4, NAME, 113, Jake13, NEUTRAL_REACTION);
 end
 
 procedure Jake03
 begin
-    gsay_reply(589, 114);
-    giq_option(4, 589, 115, Barter, 50);
-    giq_option(4, 589, 116, Jake02, 50);
-    giq_option(4, 589, 117, Jake04, 50);
+    Reply(114);
+    giq_option(4, NAME, 115, Barter, NEUTRAL_REACTION);
+    giq_option(4, NAME, 116, Jake02, NEUTRAL_REACTION);
+    giq_option(4, NAME, 117, Jake04, NEUTRAL_REACTION);
 end
 
 procedure Jake04
 begin
-    gsay_message(589, 118, 50);
-    gsay_reply(589, 162);
-    giq_option(4, 589, 119, Barter, 50);
-    giq_option(4, 589, 120, Jake02, 50);
-    if (global_var(101) != 2) then begin
-        giq_option(4, 589, 121, Jake03, 50);
+    gsay_message(NAME, 118, NEUTRAL_REACTION);
+    Reply(162);
+    giq_option(4, NAME, 119, Barter, NEUTRAL_REACTION);
+    giq_option(4, NAME, 120, Jake02, NEUTRAL_REACTION);
+    if (global_var(GVAR_FIND_WATER_CHIP) != 2) then begin
+        giq_option(4, NAME, 121, Jake03, NEUTRAL_REACTION);
     end
-    giq_option(4, 589, 122, Jake13, 50);
+    giq_option(4, NAME, 122, Jake13, NEUTRAL_REACTION);
 end
 
 procedure Jake05
 begin
-    gsay_reply(589, 123);
-    giq_option(4, 589, 124, Barter, 50);
-    giq_option(4, 589, 125, Jake06, 50);
-    giq_option(4, 589, 126, Jake13, 50);
+    Reply(123);
+    giq_option(4, NAME, 124, Barter, NEUTRAL_REACTION);
+    giq_option(4, NAME, 125, Jake06, NEUTRAL_REACTION);
+    giq_option(4, NAME, 126, Jake13, NEUTRAL_REACTION);
 end
 
 procedure Jake06
 begin
-    gsay_reply(589, 127);
-    giq_option(4, 589, 128, Barter, 50);
-    giq_option(4, 589, 129, Jake07, 50);
-    giq_option(4, 589, 130, Jake08, 50);
-    giq_option(4, 589, 131, Jake14, 51);
+    Reply(127);
+    giq_option(4, NAME, 128, Barter, NEUTRAL_REACTION);
+    giq_option(4, NAME, 129, Jake07, NEUTRAL_REACTION);
+    giq_option(4, NAME, 130, Jake08, NEUTRAL_REACTION);
+    giq_option(4, NAME, 131, Jake14, BAD_REACTION);
 end
 
 procedure Jake07
 begin
-    gsay_reply(589, 132);
-    giq_option(4, 589, 133, Barter, 50);
-    giq_option(4, 589, 134, Jake08, 50);
-    giq_option(4, 589, 135, Jake12, 50);
-    giq_option(4, 589, 136, Jake04, 50);
-    giq_option(4, 589, 137, Jake13, 50);
+    Reply(132);
+    giq_option(4, NAME, 133, Barter, NEUTRAL_REACTION);
+    giq_option(4, NAME, 134, Jake08, NEUTRAL_REACTION);
+    giq_option(4, NAME, 135, Jake12, NEUTRAL_REACTION);
+    giq_option(4, NAME, 136, Jake04, NEUTRAL_REACTION);
+    giq_option(4, NAME, 137, Jake13, NEUTRAL_REACTION);
 end
 
 procedure Jake08
 begin
-    gsay_reply(589, 138);
-    giq_option(4, 589, 139, Barter, 50);
-    giq_option(4, 589, 140, Jake07, 50);
-    giq_option(4, 589, 141, Jake14, 51);
-    giq_option(4, 589, 142, Jake13, 50);
+    Reply(138);
+    giq_option(4, NAME, 139, Barter, NEUTRAL_REACTION);
+    giq_option(4, NAME, 140, Jake07, NEUTRAL_REACTION);
+    giq_option(4, NAME, 141, Jake14, BAD_REACTION);
+    giq_option(4, NAME, 142, Jake13, NEUTRAL_REACTION);
 end
 
 procedure Jake09
 begin
-    if (get_critter_stat(dude_obj, 34) == 0) then begin
-        gsay_reply(589, 163);
+    if (dude_gender == 0) then begin
+        Reply(163);
     end
     else begin
-        gsay_reply(589, 164);
+        Reply(164);
     end
-    giq_option(-3, 589, 144, Barter, 50);
-    giq_option(-3, 589, 145, Jake13, 50);
-    giq_option(4, 589, 146, Barter, 50);
-    giq_option(4, 589, 165, Jake16, 50);
-    giq_option(4, 589, 147, Jake13, 50);
+    giq_option(-3, NAME, 144, Barter, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 145, Jake13, NEUTRAL_REACTION);
+    giq_option(4, NAME, 146, Barter, NEUTRAL_REACTION);
+    giq_option(4, NAME, 165, Jake16, NEUTRAL_REACTION);
+    giq_option(4, NAME, 147, Jake13, NEUTRAL_REACTION);
 end
 
 procedure Jake10
 begin
-    gsay_reply(589, 148);
-    giq_option(-3, 589, 149, Barter, 50);
-    giq_option(-3, 589, 150, Jake13, 50);
-    giq_option(4, 589, 151, Barter, 50);
-    giq_option(4, 589, 165, Jake16, 50);
-    giq_option(4, 589, 152, Jake15, 50);
-    giq_option(4, 589, 153, Jake13, 50);
+    Reply(148);
+    giq_option(-3, NAME, 149, Barter, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 150, Jake13, NEUTRAL_REACTION);
+    giq_option(4, NAME, 151, Barter, NEUTRAL_REACTION);
+    giq_option(4, NAME, 165, Jake16, NEUTRAL_REACTION);
+    giq_option(4, NAME, 152, Jake15, NEUTRAL_REACTION);
+    giq_option(4, NAME, 153, Jake13, NEUTRAL_REACTION);
 end
 
 procedure Jake11
 begin
-    float_msg(self_obj, message_str(589, 154), 2);
+    float_msg(self_obj, mstr(154), FLOAT_MSG_RED);
     call combat();
 end
 
 procedure Jake12
 begin
-    set_global_var(236, 1);
-    gsay_reply(589, 155);
-    giq_option(4, 589, 156, Jake08, 50);
-    giq_option(4, 589, 157, Jake13, 50);
+    set_global_var(GVAR_MASTER_FILLER_9, 1);
+    Reply(155);
+    giq_option(4, NAME, 156, Jake08, NEUTRAL_REACTION);
+    giq_option(4, NAME, 157, Jake13, NEUTRAL_REACTION);
 end
 
 procedure Jake13
@@ -328,13 +328,13 @@ end
 
 procedure Jake16
 begin
-    if (global_var(74) == 0) then begin
-        set_global_var(74, 1);
+    if (global_var(GVAR_BROTHERHOOD_OF_STEEL) == 0) then begin
+        set_global_var(GVAR_BROTHERHOOD_OF_STEEL, 1);
     end
-    if (global_var(75) == 0) then begin
-        set_global_var(75, 1);
+    if (global_var(GVAR_ANGELS_BONEYARD) == 0) then begin
+        set_global_var(GVAR_ANGELS_BONEYARD, 1);
     end
-    gsay_message(589, 166, 50);
+    gsay_message(NAME, 166, NEUTRAL_REACTION);
 end
 
 procedure JakeEnd
@@ -343,11 +343,11 @@ end
 
 procedure Barter
 begin
-    gsay_message(589, 158, 50);
+    gsay_message(NAME, 158, NEUTRAL_REACTION);
     gdialog_mod_barter(-15);
-    gsay_reply(589, 159);
-    giq_option(4, 589, 160, JakeEnd, 50);
-    giq_option(-3, 589, 161, JakeEnd, 50);
+    Reply(159);
+    giq_option(4, NAME, 160, JakeEnd, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 161, JakeEnd, NEUTRAL_REACTION);
 end
 
 procedure Get_Stuff
@@ -366,31 +366,31 @@ begin
         set_local_var(0, 50);
         set_local_var(1, 2);
         set_local_var(2, 1);
-        set_local_var(0, local_var(0) + (5 * get_critter_stat(dude_obj, 3)) - 25);
-        set_local_var(0, local_var(0) + (10 * has_trait(0, dude_obj, 10)));
-        if (has_trait(0, dude_obj, 39)) then begin
-            if (global_var(155) > 0) then begin
-                set_local_var(0, local_var(0) + global_var(155));
+        set_local_var(0, local_var(0) + (5 * dude_charisma) - 25);
+        set_local_var(0, local_var(0) + (10 * has_trait(TRAIT_PERK, dude_obj, PERK_presence)));
+        if (has_trait(TRAIT_PERK, dude_obj, PERK_cult_of_personality)) then begin
+            if (global_var(GVAR_PLAYER_REPUATION) > 0) then begin
+                set_local_var(0, local_var(0) + global_var(GVAR_PLAYER_REPUATION));
             end
             else begin
-                set_local_var(0, local_var(0) - global_var(155));
+                set_local_var(0, local_var(0) - global_var(GVAR_PLAYER_REPUATION));
             end
         end
         else begin
             if (local_var(3) == 1) then begin
-                set_local_var(0, local_var(0) - global_var(155));
+                set_local_var(0, local_var(0) - global_var(GVAR_PLAYER_REPUATION));
             end
             else begin
-                set_local_var(0, local_var(0) + global_var(155));
+                set_local_var(0, local_var(0) + global_var(GVAR_PLAYER_REPUATION));
             end
         end
-        if (global_var(158) > 2) then begin
+        if (global_var(GVAR_CHILDKILLER_REPUATION) > 2) then begin
             set_local_var(0, local_var(0) - 30);
         end
-        if (((global_var(160) + global_var(159)) >= 25) and ((global_var(160) > (3 * global_var(159))) or (global_var(157) == 1))) then begin
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(160) > (3 * global_var(GVAR_GOOD_MONSTER))) or (global_var(GVAR_CHAMPION_REPUTATION) == 1))) then begin
             set_local_var(0, local_var(0) + 20);
         end
-        if (((global_var(160) + global_var(159)) >= 25) and ((global_var(159) > (2 * global_var(160))) or (global_var(156) == 1))) then begin
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
             set_local_var(0, local_var(0) - 20);
         end
         call ReactToLevel();
@@ -483,6 +483,5 @@ end
 
 procedure Goodbyes
 begin
-    exit_line := message_str(634, random(100, 105));
+    exit_line := message_str(SCRIPT_MODREACT, random(100, 105));
 end
-
