diff --git a/Games/Fallout1.Steam/Original/HUBSAFE.ssl b/Games/Fallout1.Steam/ReDefine/HUBSAFE.ssl
index 283dead..2c9750e 100644
--- a/Games/Fallout1.Steam/Original/HUBSAFE.ssl
+++ b/Games/Fallout1.Steam/ReDefine/HUBSAFE.ssl
@@ -7,11 +7,11 @@ import variable Lorenzo_Safe_Ptr;
 
 procedure start
 begin
-    if (script_action == 15) then begin
+    if (script_action == map_enter_proc) then begin
         call map_enter_p_proc();
     end
     else begin
-        if (script_action == 23) then begin
+        if (script_action == map_update_proc) then begin
             call map_update_p_proc();
         end
     end
@@ -26,4 +26,3 @@ procedure map_update_p_proc
 begin
     Lorenzo_Safe_Ptr := self_obj;
 end
-
