diff --git a/Games/Fallout1.Steam/Original/ICECHEST.ssl b/Games/Fallout1.Steam/ReDefine/ICECHEST.ssl
index df426f7..9720786 100644
--- a/Games/Fallout1.Steam/Original/ICECHEST.ssl
+++ b/Games/Fallout1.Steam/ReDefine/ICECHEST.ssl
@@ -7,7 +7,7 @@ import variable Garls_Inven_Ptr;
 
 procedure start
 begin
-    if (script_action == 15) then begin
+    if (script_action == map_enter_proc) then begin
         call map_enter_p_proc();
     end
     else begin
@@ -26,4 +26,3 @@ procedure map_update_p_proc
 begin
     Garls_Inven_Ptr := self_obj;
 end
-
