diff --git a/Games/Fallout1.Steam/Original/IZO.ssl b/Games/Fallout1.Steam/ReDefine/IZO.ssl
index d3f3ca8..bdb6ed1 100644
--- a/Games/Fallout1.Steam/Original/IZO.ssl
+++ b/Games/Fallout1.Steam/ReDefine/IZO.ssl
@@ -18,26 +18,26 @@ import variable payment;
 procedure start
 begin
     if (not(initialized)) then begin
-        critter_add_trait(self_obj, 1, 6, 13);
-        if (global_var(104) == 2) then begin
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 13);
+        if (global_var(GVAR_CAPTURE_GIZMO) == 2) then begin
             kill_critter(self_obj, 62);
         end
         initialized := 1;
     end
     else begin
-        if (script_action == 12) then begin
+        if (script_action == critter_proc) then begin
             call critter_p_proc();
         end
         else begin
-            if (script_action == 3) then begin
+            if (script_action == description_proc) then begin
                 call description_p_proc();
             end
             else begin
-                if (script_action == 18) then begin
+                if (script_action == destroy_proc) then begin
                     call destroy_p_proc();
                 end
                 else begin
-                    if (script_action == 11) then begin
+                    if (script_action == talk_proc) then begin
                         call talk_p_proc();
                     end
                 end
@@ -50,7 +50,7 @@ procedure critter_p_proc
 begin
     if (hostile) then begin
         hostile := 0;
-        attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+        attack(dude_obj);
     end
     else begin
         if (Gizmo_is_angry) then begin
@@ -76,7 +76,7 @@ begin
                     else begin
                         if (not(line103flag)) then begin
                             line103flag := 1;
-                            display_msg(message_str(622, 103));
+                            display_mstr(103);
                         end
                     end
                 end
@@ -88,25 +88,25 @@ end
 procedure destroy_p_proc
 begin
     if (source_obj == dude_obj) then begin
-        if (((global_var(160) + global_var(159)) >= 25) and ((global_var(159) > (2 * global_var(160))) or (global_var(156) == 1))) then begin
-            set_global_var(156, 1);
-            set_global_var(157, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
+            set_global_var(GVAR_BERSERKER_REPUTATION, 1);
+            set_global_var(GVAR_CHAMPION_REPUTATION, 0);
         end
-        if (((global_var(160) + global_var(159)) >= 25) and ((global_var(160) > (3 * global_var(159))) or (global_var(157) == 1))) then begin
-            set_global_var(157, 1);
-            set_global_var(156, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(160) > (3 * global_var(GVAR_GOOD_MONSTER))) or (global_var(GVAR_CHAMPION_REPUTATION) == 1))) then begin
+            set_global_var(GVAR_CHAMPION_REPUTATION, 1);
+            set_global_var(GVAR_BERSERKER_REPUTATION, 0);
         end
-        set_global_var(160, global_var(160) + 1);
-        if ((global_var(160) % 6) == 0) then begin
-            set_global_var(155, global_var(155) + 1);
+        set_global_var(GVAR_BAD_MONSTER, global_var(GVAR_BAD_MONSTER) + 1);
+        if ((global_var(GVAR_BAD_MONSTER) % 6) == 0) then begin
+            set_global_var(GVAR_PLAYER_REPUATION, global_var(GVAR_PLAYER_REPUATION) + 1);
         end
     end
 end
 
 procedure description_p_proc
 begin
-    if (is_success(do_check(dude_obj, 1, has_trait(0, dude_obj, 0)))) then begin
-        display_msg(message_str(622, 100));
+    if (is_success(do_check(dude_obj, 1, has_trait(TRAIT_PERK, dude_obj, PERK_bonus_awareness)))) then begin
+        display_mstr(100);
     end
 end
 
@@ -118,13 +118,12 @@ end
 procedure talk_p_proc
 begin
     variable LVar0 := 0;
-    display_msg(message_str(622, 101));
+    display_mstr(101);
     if (payment > 0) then begin
         use_obj(dude_obj);
-        LVar0 := create_object_sid(41, 0, 0, -1);
+        LVar0 := create_object(41, 0, 0);
         add_mult_objs_to_inven(dude_obj, LVar0, payment);
-        display_msg(message_str(622, 102));
+        display_mstr(102);
         payment := 0;
     end
 end
-
