diff --git a/Games/Fallout1.Steam/Original/JAIN.ssl b/Games/Fallout1.Steam/ReDefine/JAIN.ssl
index ae8bb87..907cf8e 100644
--- a/Games/Fallout1.Steam/Original/JAIN.ssl
+++ b/Games/Fallout1.Steam/ReDefine/JAIN.ssl
@@ -72,26 +72,26 @@ procedure start
 begin
     if (Only_Once) then begin
         Only_Once := 0;
-        critter_add_trait(self_obj, 1, 6, 72);
-        critter_add_trait(self_obj, 1, 5, 77);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 72);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_COC_HUB_BRAINWASHED);
     end
-    if (script_action == 21) then begin
+    if (script_action == look_at_proc) then begin
         call look_at_p_proc();
     end
     else begin
-        if (script_action == 4) then begin
+        if (script_action == pickup_proc) then begin
             call pickup_p_proc();
         end
         else begin
-            if (script_action == 11) then begin
+            if (script_action == talk_proc) then begin
                 call talk_p_proc();
             end
             else begin
-                if (script_action == 12) then begin
+                if (script_action == critter_proc) then begin
                     call critter_p_proc();
                 end
                 else begin
-                    if (script_action == 18) then begin
+                    if (script_action == destroy_proc) then begin
                         call destroy_p_proc();
                     end
                 end
@@ -109,7 +109,7 @@ procedure critter_p_proc
 begin
     if (hostile) then begin
         hostile := 0;
-        attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+        attack(dude_obj);
     end
     if ((map_var(6) == 1) and (obj_can_see_obj(self_obj, dude_obj) == 1)) then begin
         call combat();
@@ -134,7 +134,7 @@ begin
         end_dialogue;
     end
     else begin
-        if (global_var(18) == 1) then begin
+        if (global_var(GVAR_MASTER_BLOWN) == 1) then begin
             start_gdialog(46, self_obj, 4, 18, 3);
             gsay_start;
             call Jain30();
@@ -173,26 +173,26 @@ end
 procedure destroy_p_proc
 begin
     if (source_obj == dude_obj) then begin
-        if (((global_var(160) + global_var(159)) >= 25) and ((global_var(159) > (2 * global_var(160))) or (global_var(156) == 1))) then begin
-            set_global_var(156, 1);
-            set_global_var(157, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
+            set_global_var(GVAR_BERSERKER_REPUTATION, 1);
+            set_global_var(GVAR_CHAMPION_REPUTATION, 0);
         end
-        if (((global_var(160) + global_var(159)) >= 25) and ((global_var(160) > (3 * global_var(159))) or (global_var(157) == 1))) then begin
-            set_global_var(157, 1);
-            set_global_var(156, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(160) > (3 * global_var(GVAR_GOOD_MONSTER))) or (global_var(GVAR_CHAMPION_REPUTATION) == 1))) then begin
+            set_global_var(GVAR_CHAMPION_REPUTATION, 1);
+            set_global_var(GVAR_BERSERKER_REPUTATION, 0);
         end
-        set_global_var(160, global_var(160) + 1);
-        if ((global_var(160) % 6) == 0) then begin
-            set_global_var(155, global_var(155) + 1);
+        set_global_var(GVAR_BAD_MONSTER, global_var(GVAR_BAD_MONSTER) + 1);
+        if ((global_var(GVAR_BAD_MONSTER) % 6) == 0) then begin
+            set_global_var(GVAR_PLAYER_REPUATION, global_var(GVAR_PLAYER_REPUATION) + 1);
         end
     end
-    set_global_var(112, 2);
+    set_global_var(GVAR_KILL_JAIN, 2);
 end
 
 procedure look_at_p_proc
 begin
     script_overrides;
-    display_msg(message_str(46, 100));
+    display_mstr(100);
 end
 
 procedure damage_p_proc
@@ -206,43 +206,43 @@ end
 
 procedure Jain00
 begin
-    gsay_reply(46, 101);
-    giq_option(4, 46, 102, Jain04, 50);
-    giq_option(4, 46, 103, Jain07, 50);
-    giq_option(4, 46, 104, Jain10, 50);
-    giq_option(-3, 46, 105, Jain02, 50);
+    Reply(101);
+    giq_option(4, NAME, 102, Jain04, NEUTRAL_REACTION);
+    giq_option(4, NAME, 103, Jain07, NEUTRAL_REACTION);
+    giq_option(4, NAME, 104, Jain10, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 105, Jain02, NEUTRAL_REACTION);
 end
 
 procedure Jain01
 begin
-    gsay_reply(46, 106);
-    giq_option(-3, 46, 107, Jain03, 50);
-    giq_option(-3, 46, 108, Jain02, 50);
+    Reply(106);
+    giq_option(-3, NAME, 107, Jain03, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 108, Jain02, NEUTRAL_REACTION);
 end
 
 procedure Jain02
 begin
-    gsay_message(46, 109, 51);
+    gsay_message(NAME, 109, BAD_REACTION);
     call JainEndCombat();
 end
 
 procedure Jain03
 begin
     set_map_var(0, 1);
-    gsay_message(46, 110, 50);
+    gsay_message(NAME, 110, NEUTRAL_REACTION);
 end
 
 procedure Jain04
 begin
-    gsay_reply(46, 111);
-    giq_option(4, 46, 112, Jain05, 50);
-    giq_option(4, 46, 113, Jain03, 50);
+    Reply(111);
+    giq_option(4, NAME, 112, Jain05, NEUTRAL_REACTION);
+    giq_option(4, NAME, 113, Jain03, NEUTRAL_REACTION);
 end
 
 procedure Jain05
 begin
     set_map_var(0, 1);
-    gsay_message(46, 114, 51);
+    gsay_message(NAME, 114, BAD_REACTION);
 end
 
 procedure Jain06
@@ -251,70 +251,70 @@ end
 
 procedure Jain07
 begin
-    gsay_reply(46, 115);
-    giq_option(4, 46, 116, Jain10, 50);
-    giq_option(4, 46, 117, Jain09, 50);
-    giq_option(4, 46, 118, Jain08, 51);
-    giq_option(4, 46, 119, JainEnd, 50);
+    Reply(115);
+    giq_option(4, NAME, 116, Jain10, NEUTRAL_REACTION);
+    giq_option(4, NAME, 117, Jain09, NEUTRAL_REACTION);
+    giq_option(4, NAME, 118, Jain08, BAD_REACTION);
+    giq_option(4, NAME, 119, JainEnd, NEUTRAL_REACTION);
 end
 
 procedure Jain08
 begin
-    gsay_message(46, 120, 51);
+    gsay_message(NAME, 120, BAD_REACTION);
     call JainEndCombat();
 end
 
 procedure Jain09
 begin
     call BigDownReact();
-    gsay_message(46, 121, 51);
+    gsay_message(NAME, 121, BAD_REACTION);
     call JainEndCombat();
 end
 
 procedure Jain10
 begin
-    gsay_reply(46, 122);
-    giq_option(4, 46, 123, Jain11, 49);
-    giq_option(4, 46, 124, Jain26, 51);
-    giq_option(4, 46, 125, Jain20, 50);
+    Reply(122);
+    giq_option(4, NAME, 123, Jain11, GOOD_REACTION);
+    giq_option(4, NAME, 124, Jain26, BAD_REACTION);
+    giq_option(4, NAME, 125, Jain20, NEUTRAL_REACTION);
 end
 
 procedure Jain11
 begin
     call UpReact();
-    gsay_reply(46, 126);
-    giq_option(4, 46, 127, Jain17, 51);
-    giq_option(4, 46, 128, Jain13, 50);
-    giq_option(4, 46, 129, Jain12, 50);
+    Reply(126);
+    giq_option(4, NAME, 127, Jain17, BAD_REACTION);
+    giq_option(4, NAME, 128, Jain13, NEUTRAL_REACTION);
+    giq_option(4, NAME, 129, Jain12, NEUTRAL_REACTION);
 end
 
 procedure Jain12
 begin
     set_map_var(0, 1);
-    gsay_reply(46, 130);
-    giq_option(4, 46, 131, JainEnd, 50);
-    giq_option(4, 46, 132, JainEndCombat, 51);
+    Reply(130);
+    giq_option(4, NAME, 131, JainEnd, NEUTRAL_REACTION);
+    giq_option(4, NAME, 132, JainEndCombat, BAD_REACTION);
 end
 
 procedure Jain13
 begin
-    gsay_reply(46, 133);
-    giq_option(4, 46, 134, Jain14, 50);
-    giq_option(4, 46, 135, Jain15, 51);
+    Reply(133);
+    giq_option(4, NAME, 134, Jain14, NEUTRAL_REACTION);
+    giq_option(4, NAME, 135, Jain15, BAD_REACTION);
 end
 
 procedure Jain14
 begin
-    gsay_reply(46, 136);
-    giq_option(4, 46, 137, Jain20, 49);
-    giq_option(4, 46, 138, Jain09, 51);
-    giq_option(4, 46, 139, Jain15, 51);
+    Reply(136);
+    giq_option(4, NAME, 137, Jain20, GOOD_REACTION);
+    giq_option(4, NAME, 138, Jain09, BAD_REACTION);
+    giq_option(4, NAME, 139, Jain15, BAD_REACTION);
 end
 
 procedure Jain15
 begin
     call DownReact();
-    gsay_message(46, 140, 50);
+    gsay_message(NAME, 140, NEUTRAL_REACTION);
 end
 
 procedure Jain16
@@ -323,47 +323,47 @@ end
 
 procedure Jain17
 begin
-    gsay_reply(46, 141);
-    giq_option(4, 46, 142, Jain18, 51);
-    giq_option(4, 46, 143, Jain19, 50);
+    Reply(141);
+    giq_option(4, NAME, 142, Jain18, BAD_REACTION);
+    giq_option(4, NAME, 143, Jain19, NEUTRAL_REACTION);
 end
 
 procedure Jain18
 begin
     call DownReact();
-    gsay_message(46, 144, 51);
+    gsay_message(NAME, 144, BAD_REACTION);
 end
 
 procedure Jain19
 begin
-    gsay_reply(46, 145);
-    giq_option(4, 46, 146, Jain13, 50);
-    giq_option(4, 46, 147, JainEnd, 50);
+    Reply(145);
+    giq_option(4, NAME, 146, Jain13, NEUTRAL_REACTION);
+    giq_option(4, NAME, 147, JainEnd, NEUTRAL_REACTION);
 end
 
 procedure Jain20
 begin
     call UpReact();
-    gsay_reply(46, 148);
-    giq_option(4, 46, 149, Jain21, 50);
-    giq_option(4, 46, 150, Jain22, 50);
+    Reply(148);
+    giq_option(4, NAME, 149, Jain21, NEUTRAL_REACTION);
+    giq_option(4, NAME, 150, Jain22, NEUTRAL_REACTION);
 end
 
 procedure Jain21
 begin
-    gsay_reply(46, 151);
-    giq_option(4, 46, 152, Jain38, 50);
-    giq_option(4, 46, 153, JainEndCombat, 51);
-    giq_option(4, 46, 154, Jain37, 50);
-    giq_option(4, 46, 155, JainEnd, 50);
+    Reply(151);
+    giq_option(4, NAME, 152, Jain38, NEUTRAL_REACTION);
+    giq_option(4, NAME, 153, JainEndCombat, BAD_REACTION);
+    giq_option(4, NAME, 154, Jain37, NEUTRAL_REACTION);
+    giq_option(4, NAME, 155, JainEnd, NEUTRAL_REACTION);
 end
 
 procedure Jain22
 begin
-    gsay_reply(46, 156);
-    giq_option(4, 46, 157, Jain37, 50);
-    giq_option(4, 46, 158, Jain15, 51);
-    giq_option(4, 46, 159, JainEnd, 50);
+    Reply(156);
+    giq_option(4, NAME, 157, Jain37, NEUTRAL_REACTION);
+    giq_option(4, NAME, 158, Jain15, BAD_REACTION);
+    giq_option(4, NAME, 159, JainEnd, NEUTRAL_REACTION);
 end
 
 procedure Jain23
@@ -381,93 +381,93 @@ end
 procedure Jain26
 begin
     call BigDownReact();
-    gsay_reply(46, 161);
-    giq_option(4, 46, 161, Jain19, 50);
-    giq_option(4, 46, 162, JainEndCombat, 50);
+    Reply(161);
+    giq_option(4, NAME, 161, Jain19, NEUTRAL_REACTION);
+    giq_option(4, NAME, 162, JainEndCombat, NEUTRAL_REACTION);
 end
 
 procedure Jain27
 begin
-    gsay_reply(46, 163);
-    giq_option(4, 46, 164, Jain36, 50);
-    giq_option(4, 46, 165, Jain28, 51);
-    giq_option(4, 46, 166, JainEnd, 50);
-    giq_option(-3, 46, 167, Jain39, 50);
-    giq_option(-3, 46, 168, Jain34, 50);
+    Reply(163);
+    giq_option(4, NAME, 164, Jain36, NEUTRAL_REACTION);
+    giq_option(4, NAME, 165, Jain28, BAD_REACTION);
+    giq_option(4, NAME, 166, JainEnd, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 167, Jain39, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 168, Jain34, NEUTRAL_REACTION);
 end
 
 procedure Jain28
 begin
     call BigDownReact();
     set_map_var(0, 1);
-    gsay_message(46, 169, 50);
+    gsay_message(NAME, 169, NEUTRAL_REACTION);
 end
 
 procedure Jain29
 begin
-    gsay_message(46, 170, 50);
+    gsay_message(NAME, 170, NEUTRAL_REACTION);
 end
 
 procedure Jain30
 begin
-    gsay_reply(46, 171);
-    giq_option(4, 46, 172, Jain31, 50);
-    giq_option(4, 46, 173, Jain32, 50);
-    giq_option(-3, 46, 174, Jain39, 50);
-    giq_option(-3, 46, 175, JainEnd, 50);
+    Reply(171);
+    giq_option(4, NAME, 172, Jain31, NEUTRAL_REACTION);
+    giq_option(4, NAME, 173, Jain32, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 174, Jain39, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 175, JainEnd, NEUTRAL_REACTION);
 end
 
 procedure Jain31
 begin
-    gsay_message(46, 176, 51);
+    gsay_message(NAME, 176, BAD_REACTION);
     call JainEndCombat();
 end
 
 procedure Jain32
 begin
-    gsay_reply(46, 177);
-    giq_option(4, 46, 178, Jain33, 50);
-    giq_option(4, 46, 179, JainEnd, 50);
+    Reply(177);
+    giq_option(4, NAME, 178, Jain33, NEUTRAL_REACTION);
+    giq_option(4, NAME, 179, JainEnd, NEUTRAL_REACTION);
 end
 
 procedure Jain33
 begin
-    gsay_message(46, 180, 50);
+    gsay_message(NAME, 180, NEUTRAL_REACTION);
 end
 
 procedure Jain34
 begin
     set_map_var(0, 1);
-    gsay_message(46, 181, 51);
+    gsay_message(NAME, 181, BAD_REACTION);
 end
 
 procedure Jain35
 begin
     set_map_var(0, 1);
-    gsay_message(46, 182, 51);
+    gsay_message(NAME, 182, BAD_REACTION);
 end
 
 procedure Jain36
 begin
-    gsay_reply(46, 183);
-    giq_option(4, 46, 184, Jain14, 50);
-    giq_option(4, 46, 185, JainEnd, 50);
+    Reply(183);
+    giq_option(4, NAME, 184, Jain14, NEUTRAL_REACTION);
+    giq_option(4, NAME, 185, JainEnd, NEUTRAL_REACTION);
 end
 
 procedure Jain37
 begin
     call UpReact();
-    gsay_message(46, 186, 49);
+    gsay_message(NAME, 186, GOOD_REACTION);
 end
 
 procedure Jain38
 begin
-    gsay_message(46, 187, 50);
+    gsay_message(NAME, 187, NEUTRAL_REACTION);
 end
 
 procedure Jain39
 begin
-    gsay_message(46, 188, 50);
+    gsay_message(NAME, 188, NEUTRAL_REACTION);
 end
 
 procedure JainEnd
@@ -487,31 +487,31 @@ begin
         set_local_var(0, 50);
         set_local_var(1, 2);
         set_local_var(2, 1);
-        set_local_var(0, local_var(0) + (5 * get_critter_stat(dude_obj, 3)) - 25);
-        set_local_var(0, local_var(0) + (10 * has_trait(0, dude_obj, 10)));
-        if (has_trait(0, dude_obj, 39)) then begin
-            if (global_var(155) > 0) then begin
-                set_local_var(0, local_var(0) + global_var(155));
+        set_local_var(0, local_var(0) + (5 * dude_charisma) - 25);
+        set_local_var(0, local_var(0) + (10 * has_trait(TRAIT_PERK, dude_obj, PERK_presence)));
+        if (has_trait(TRAIT_PERK, dude_obj, PERK_cult_of_personality)) then begin
+            if (global_var(GVAR_PLAYER_REPUATION) > 0) then begin
+                set_local_var(0, local_var(0) + global_var(GVAR_PLAYER_REPUATION));
             end
             else begin
-                set_local_var(0, local_var(0) - global_var(155));
+                set_local_var(0, local_var(0) - global_var(GVAR_PLAYER_REPUATION));
             end
         end
         else begin
             if (local_var(3) == 1) then begin
-                set_local_var(0, local_var(0) - global_var(155));
+                set_local_var(0, local_var(0) - global_var(GVAR_PLAYER_REPUATION));
             end
             else begin
-                set_local_var(0, local_var(0) + global_var(155));
+                set_local_var(0, local_var(0) + global_var(GVAR_PLAYER_REPUATION));
             end
         end
-        if (global_var(158) > 2) then begin
+        if (global_var(GVAR_CHILDKILLER_REPUATION) > 2) then begin
             set_local_var(0, local_var(0) - 30);
         end
-        if (((global_var(160) + global_var(159)) >= 25) and ((global_var(160) > (3 * global_var(159))) or (global_var(157) == 1))) then begin
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(160) > (3 * global_var(GVAR_GOOD_MONSTER))) or (global_var(GVAR_CHAMPION_REPUTATION) == 1))) then begin
             set_local_var(0, local_var(0) + 20);
         end
-        if (((global_var(160) + global_var(159)) >= 25) and ((global_var(159) > (2 * global_var(160))) or (global_var(156) == 1))) then begin
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
             set_local_var(0, local_var(0) - 20);
         end
         call ReactToLevel();
@@ -604,6 +604,5 @@ end
 
 procedure Goodbyes
 begin
-    exit_line := message_str(634, random(100, 105));
+    exit_line := message_str(SCRIPT_MODREACT, random(100, 105));
 end
-
