diff --git a/Games/Fallout1.Steam/Original/JARVIS.ssl b/Games/Fallout1.Steam/ReDefine/JARVIS.ssl
index 39b4655..811f03e 100644
--- a/Games/Fallout1.Steam/Original/JARVIS.ssl
+++ b/Games/Fallout1.Steam/ReDefine/JARVIS.ssl
@@ -26,51 +26,51 @@ variable hostile;
 
 procedure start
 begin
-    if (script_action == 11) then begin
+    if (script_action == talk_proc) then begin
         call talk_p_proc();
     end
     else begin
-        if (script_action == 16) then begin
+        if (script_action == map_exit_proc) then begin
             call map_exit_p_proc();
         end
         else begin
-            if (script_action == 15) then begin
+            if (script_action == map_enter_proc) then begin
                 call map_enter_p_proc();
             end
             else begin
-                if ((script_action == 21) or (script_action == 3)) then begin
+                if ((script_action == look_at_proc) or (script_action == description_proc)) then begin
                     call look_at_p_proc();
                 end
                 else begin
-                    if (script_action == 7) then begin
+                    if (script_action == use_obj_on_proc) then begin
                         call use_obj_on_p_proc();
                     end
                     else begin
-                        if (script_action == 4) then begin
+                        if (script_action == pickup_proc) then begin
                             call pickup_p_proc();
                         end
                         else begin
-                            if (script_action == 18) then begin
+                            if (script_action == destroy_proc) then begin
                                 call destroy_p_proc();
                             end
                             else begin
-                                if (script_action == 13) then begin
+                                if (script_action == combat_proc) then begin
                                     call combat_p_proc();
                                 end
                                 else begin
-                                    if (script_action == 8) then begin
+                                    if (script_action == use_skill_on_proc) then begin
                                         call use_skill_on_p_proc();
                                     end
                                     else begin
-                                        if (script_action == 18) then begin
+                                        if (script_action == destroy_proc) then begin
                                             call destroy_p_proc();
                                         end
                                         else begin
-                                            if (script_action == 12) then begin
+                                            if (script_action == critter_proc) then begin
                                                 call critter_p_proc();
                                             end
                                             else begin
-                                                if (script_action == 22) then begin
+                                                if (script_action == timed_event_proc) then begin
                                                     call timed_event_p_proc();
                                                 end
                                             end
@@ -90,10 +90,10 @@ procedure look_at_p_proc
 begin
     script_overrides;
     if (local_var(4) == 0) then begin
-        display_msg(message_str(440, 102));
+        display_mstr(102);
     end
     else begin
-        display_msg(message_str(440, 103));
+        display_mstr(103);
     end
 end
 
@@ -103,8 +103,8 @@ begin
         anim(self_obj, 48, 0);
         add_timer_event(self_obj, game_ticks(3600), 1);
     end
-    critter_add_trait(self_obj, 1, 6, 2);
-    critter_add_trait(self_obj, 1, 5, 6);
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 2);
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_PEASANT);
 end
 
 procedure talk_p_proc
@@ -134,23 +134,23 @@ begin
         script_overrides;
         if (action_being_used == 6) then begin
             if (is_success(LVar0)) then begin
-                display_msg(message_str(440, 112));
+                display_mstr(112);
             end
             else begin
-                display_msg(message_str(440, 113));
+                display_mstr(113);
             end
         end
         else begin
             if (action_being_used == 7) then begin
                 if (is_success(LVar0)) then begin
-                    display_msg(message_str(440, 114));
+                    display_mstr(114);
                 end
                 else begin
-                    display_msg(message_str(440, 115));
+                    display_mstr(115);
                 end
             end
             else begin
-                display_msg(message_str(440, 116));
+                display_mstr(116);
             end
         end
     end
@@ -164,30 +164,30 @@ begin
         set_local_var(5, 1);
         rm_obj_from_inven(dude_obj, obj_being_used_with);
         destroy_object(obj_being_used_with);
-        display_msg(message_str(440, 117));
+        display_mstr(117);
         give_exp_points(400);
-        set_global_var(155, global_var(155) + 1);
-        display_msg(message_str(440, 200));
+        set_global_var(GVAR_PLAYER_REPUATION, global_var(GVAR_PLAYER_REPUATION) + 1);
+        display_mstr(200);
     end
 end
 
 procedure destroy_p_proc
 begin
     if (source_obj == dude_obj) then begin
-        set_global_var(246, 1);
+        set_global_var(GVAR_ENEMY_SHADY_SANDS, 1);
     end
     if (source_obj == dude_obj) then begin
-        if (((global_var(160) + global_var(159)) >= 25) and ((global_var(159) > (2 * global_var(160))) or (global_var(156) == 1))) then begin
-            set_global_var(156, 1);
-            set_global_var(157, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
+            set_global_var(GVAR_BERSERKER_REPUTATION, 1);
+            set_global_var(GVAR_CHAMPION_REPUTATION, 0);
         end
-        if (((global_var(160) + global_var(159)) >= 25) and ((global_var(160) > (3 * global_var(159))) or (global_var(157) == 1))) then begin
-            set_global_var(157, 1);
-            set_global_var(156, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(160) > (3 * global_var(GVAR_GOOD_MONSTER))) or (global_var(GVAR_CHAMPION_REPUTATION) == 1))) then begin
+            set_global_var(GVAR_CHAMPION_REPUTATION, 1);
+            set_global_var(GVAR_BERSERKER_REPUTATION, 0);
         end
-        set_global_var(159, global_var(159) + 1);
-        if ((global_var(159) % 2) == 0) then begin
-            set_global_var(155, global_var(155) - 1);
+        set_global_var(GVAR_GOOD_MONSTER, global_var(GVAR_GOOD_MONSTER) + 1);
+        if ((global_var(GVAR_GOOD_MONSTER) % 2) == 0) then begin
+            set_global_var(GVAR_PLAYER_REPUATION, global_var(GVAR_PLAYER_REPUATION) - 1);
         end
     end
     rm_timer_event(self_obj);
@@ -202,41 +202,41 @@ end
 
 procedure Jarvis01
 begin
-    float_msg(self_obj, message_str(440, 104), 3);
+    float_msg(self_obj, mstr(104), FLOAT_MSG_GREEN);
 end
 
 procedure Jarvis02
 begin
-    gsay_reply(440, 105);
+    Reply(105);
     set_local_var(4, 1);
     call Jarvis04();
 end
 
 procedure Jarvis03
 begin
-    gsay_reply(440, 106);
+    Reply(106);
     call Jarvis04();
 end
 
 procedure Jarvis04
 begin
-    giq_option(4, 440, 107, Jarvis05, 50);
-    giq_option(4, 440, 108, Jarvis06, 50);
-    giq_option(4, 440, 109, Jarvisend, 50);
-    giq_option(-3, 440, 118, Jarvis05, 50);
-    giq_option(-3, 440, 119, Jarvis06, 50);
-    giq_option(-3, 440, 120, Jarvisend, 50);
+    giq_option(4, NAME, 107, Jarvis05, NEUTRAL_REACTION);
+    giq_option(4, NAME, 108, Jarvis06, NEUTRAL_REACTION);
+    giq_option(4, NAME, 109, Jarvisend, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 118, Jarvis05, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 119, Jarvis06, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 120, Jarvisend, NEUTRAL_REACTION);
 end
 
 procedure Jarvis05
 begin
-    gsay_reply(440, 110);
+    Reply(110);
     call Jarvis04();
 end
 
 procedure Jarvis06
 begin
-    gsay_reply(440, 111);
+    Reply(111);
     call Jarvis04();
 end
 
@@ -248,10 +248,10 @@ procedure description_p_proc
 begin
     script_overrides;
     if (local_var(4) == 0) then begin
-        display_msg(message_str(440, 102));
+        display_mstr(102);
     end
     else begin
-        display_msg(message_str(440, 103));
+        display_mstr(103);
     end
 end
 
@@ -259,7 +259,7 @@ procedure critter_p_proc
 begin
     if (hostile) then begin
         hostile := 0;
-        attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+        attack(dude_obj);
     end
 end
 
@@ -273,9 +273,9 @@ begin
     if (local_var(5) == 0) then begin
         add_timer_event(self_obj, game_ticks(7200), 1);
     end
-    if (tile_distance_objs(dude_obj, self_obj) < get_critter_stat(dude_obj, 1)) then begin
-        display_msg(message_str(440, 100));
-        float_msg(self_obj, message_str(440, 101), 0);
+    if (tile_distance_objs(dude_obj, self_obj) < dude_perception) then begin
+        display_mstr(100);
+        float_msg(self_obj, mstr(101), FLOAT_MSG_NORMAL);
     end
 end
 
@@ -283,4 +283,3 @@ procedure map_exit_p_proc
 begin
     rm_timer_event(self_obj);
 end
-
