diff --git a/Games/Fallout1.Steam/Original/JENNIFER.ssl b/Games/Fallout1.Steam/ReDefine/JENNIFER.ssl
index c029c7b..f683e07 100644
--- a/Games/Fallout1.Steam/Original/JENNIFER.ssl
+++ b/Games/Fallout1.Steam/ReDefine/JENNIFER.ssl
@@ -75,26 +75,26 @@ procedure start
 begin
     if (only_once) then begin
         only_once := 0;
-        critter_add_trait(self_obj, 1, 6, 44);
-        critter_add_trait(self_obj, 1, 5, 65);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 44);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_BROTHERHOOD_PALADIN);
     end
-    if (script_action == 21) then begin
+    if (script_action == look_at_proc) then begin
         call look_at_p_proc();
     end
     else begin
-        if (script_action == 4) then begin
+        if (script_action == pickup_proc) then begin
             call pickup_p_proc();
         end
         else begin
-            if (script_action == 11) then begin
+            if (script_action == talk_proc) then begin
                 call talk_p_proc();
             end
             else begin
-                if (script_action == 12) then begin
+                if (script_action == critter_proc) then begin
                     call critter_p_proc();
                 end
                 else begin
-                    if (script_action == 18) then begin
+                    if (script_action == destroy_proc) then begin
                         call destroy_p_proc();
                     end
                 end
@@ -122,7 +122,7 @@ end
 
 procedure do_dialogue
 begin
-    start_gdialog(-1, self_obj, 4, -1, -1);
+    start_dialogue(self_obj, 4);
     gsay_start;
     call Jennifer01();
     gsay_end;
@@ -131,7 +131,7 @@ end
 
 procedure weapon_check
 begin
-    if ((obj_item_subtype(critter_inven_obj(dude_obj, 1)) == 3) or (obj_item_subtype(critter_inven_obj(dude_obj, 2)) == 3)) then begin
+    if ((obj_item_subtype(critter_inven_obj(dude_obj, INVEN_TYPE_RIGHT_HAND)) == 3) or (obj_item_subtype(critter_inven_obj(dude_obj, INVEN_TYPE_LEFT_HAND)) == 3)) then begin
         armed := 1;
     end
     else begin
@@ -142,24 +142,24 @@ end
 procedure Jennifer00
 begin
     warned := 1;
-    float_msg(self_obj, message_str(462, 102), 0);
-    float_msg(self_obj, message_str(462, 103), 0);
+    float_msg(self_obj, mstr(102), FLOAT_MSG_NORMAL);
+    float_msg(self_obj, mstr(103), FLOAT_MSG_NORMAL);
 end
 
 procedure Jennifer01
 begin
-    gsay_reply(462, 104);
-    if (get_critter_stat(dude_obj, 34) == 1) then begin
-        giq_option(5, 462, 105, Jennifer14, 50);
+    Reply(104);
+    if (dude_gender == 1) then begin
+        giq_option(5, NAME, 105, Jennifer14, NEUTRAL_REACTION);
     end
     else begin
-        giq_option(5, 462, 106, JenniferCharm, 50);
-        giq_option(-3, 462, 107, Jennifer11, 50);
+        giq_option(5, NAME, 106, JenniferCharm, NEUTRAL_REACTION);
+        giq_option(-3, NAME, 107, Jennifer11, NEUTRAL_REACTION);
     end
-    giq_option(5, 462, 108, Jennifer01a, 49);
-    giq_option(4, 462, 109, Jennifer01b, 51);
-    giq_option(4, 462, 110, Jennifer02, 50);
-    giq_option(-3, 462, 111, Jennifer13, 50);
+    giq_option(5, NAME, 108, Jennifer01a, GOOD_REACTION);
+    giq_option(4, NAME, 109, Jennifer01b, BAD_REACTION);
+    giq_option(4, NAME, 110, Jennifer02, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 111, Jennifer13, NEUTRAL_REACTION);
 end
 
 procedure Jennifer01a
@@ -177,132 +177,132 @@ end
 procedure Jennifer02
 begin
     line2flag := 1;
-    gsay_message(462, 112, 50);
-    gsay_reply(462, 113);
-    giq_option(5, 462, 114, UpReact, 50);
+    gsay_message(NAME, 112, NEUTRAL_REACTION);
+    Reply(113);
+    giq_option(5, NAME, 114, UpReact, NEUTRAL_REACTION);
     if (not(line5flag)) then begin
-        giq_option(4, 462, 115, Jennifer05, 50);
+        giq_option(4, NAME, 115, Jennifer05, NEUTRAL_REACTION);
     end
     call Goodbyes();
-    giq_option(4, 0, exit_line, JenniferEnd, 50);
+    giq_option(4, 0, exit_line, JenniferEnd, NEUTRAL_REACTION);
 end
 
 procedure Jennifer05
 begin
     line5flag := 1;
-    gsay_message(462, 116, 50);
+    gsay_message(NAME, 116, NEUTRAL_REACTION);
     call Jennifer07();
 end
 
 procedure Jennifer07
 begin
-    gsay_reply(462, 117);
-    giq_option(4, 462, 118, UpReact, 49);
-    giq_option(4, 462, 119, Jennifer08, 50);
-    giq_option(4, 462, 120, DownReact, 51);
+    Reply(117);
+    giq_option(4, NAME, 118, UpReact, GOOD_REACTION);
+    giq_option(4, NAME, 119, Jennifer08, NEUTRAL_REACTION);
+    giq_option(4, NAME, 120, DownReact, BAD_REACTION);
 end
 
 procedure Jennifer08
 begin
-    gsay_reply(462, 121);
-    giq_option(5, 462, 122, UpReact, 50);
-    giq_option(6, 462, 123, Jennifer29, 50);
-    if (get_critter_stat(dude_obj, 34) == 1) then begin
+    Reply(121);
+    giq_option(5, NAME, 122, UpReact, NEUTRAL_REACTION);
+    giq_option(6, NAME, 123, Jennifer29, NEUTRAL_REACTION);
+    if (dude_gender == 1) then begin
         if (not(line14flag)) then begin
-            giq_option(5, 462, 124, Jennifer14, 50);
+            giq_option(5, NAME, 124, Jennifer14, NEUTRAL_REACTION);
         end
     end
     else begin
-        if (not(flag1) and not(flag2) and (get_critter_stat(dude_obj, 34) == 0)) then begin
-            giq_option(5, 462, 125, JenniferCharm, 50);
+        if (not(flag1) and not(flag2) and (dude_gender == 0)) then begin
+            giq_option(5, NAME, 125, JenniferCharm, NEUTRAL_REACTION);
         end
     end
-    giq_option(4, 462, 126, JenniferEnd, 50);
+    giq_option(4, NAME, 126, JenniferEnd, NEUTRAL_REACTION);
 end
 
 procedure Jennifer11
 begin
-    gsay_reply(462, 127);
-    giq_option(-3, 462, 128, Jennifer13, 50);
+    Reply(127);
+    giq_option(-3, NAME, 128, Jennifer13, NEUTRAL_REACTION);
 end
 
 procedure Jennifer13
 begin
-    gsay_message(462, 129, 50);
+    gsay_message(NAME, 129, NEUTRAL_REACTION);
 end
 
 procedure Jennifer14
 begin
     line14flag := 1;
     call TopReact();
-    gsay_message(462, 130, 50);
-    gsay_reply(462, 131);
-    giq_option(8, 462, 132, Jennifer16, 50);
-    giq_option(4, 462, 133, Jennifer15, 50);
-    giq_option(4, 462, 134, Jennifer17, 50);
+    gsay_message(NAME, 130, NEUTRAL_REACTION);
+    Reply(131);
+    giq_option(8, NAME, 132, Jennifer16, NEUTRAL_REACTION);
+    giq_option(4, NAME, 133, Jennifer15, NEUTRAL_REACTION);
+    giq_option(4, NAME, 134, Jennifer17, NEUTRAL_REACTION);
 end
 
 procedure Jennifer15
 begin
-    gsay_reply(462, 135);
-    giq_option(4, 462, 136, Jennifer17, 50);
-    giq_option(4, 462, 137, Jennifer17, 50);
+    Reply(135);
+    giq_option(4, NAME, 136, Jennifer17, NEUTRAL_REACTION);
+    giq_option(4, NAME, 137, Jennifer17, NEUTRAL_REACTION);
 end
 
 procedure Jennifer16
 begin
-    gsay_reply(462, 138);
-    giq_option(8, 462, 139, Jennifer18, 50);
-    giq_option(4, 462, 140, Jennifer17, 50);
+    Reply(138);
+    giq_option(8, NAME, 139, Jennifer18, NEUTRAL_REACTION);
+    giq_option(4, NAME, 140, Jennifer17, NEUTRAL_REACTION);
 end
 
 procedure Jennifer17
 begin
-    gsay_reply(462, 141);
+    Reply(141);
     if (not(line2flag)) then begin
-        giq_option(4, 462, 142, Jennifer02, 50);
+        giq_option(4, NAME, 142, Jennifer02, NEUTRAL_REACTION);
     end
     if (not(line5flag)) then begin
-        giq_option(4, 462, 143, Jennifer05, 50);
+        giq_option(4, NAME, 143, Jennifer05, NEUTRAL_REACTION);
     end
-    giq_option(4, 462, 144, JenniferEnd, 50);
+    giq_option(4, NAME, 144, JenniferEnd, NEUTRAL_REACTION);
 end
 
 procedure Jennifer18
 begin
-    gsay_message(462, 145, 50);
-    gsay_reply(462, 146);
-    giq_option(4, 462, 147, Jennifer19, 50);
+    gsay_message(NAME, 145, NEUTRAL_REACTION);
+    Reply(146);
+    giq_option(4, NAME, 147, Jennifer19, NEUTRAL_REACTION);
     call Goodbyes();
-    giq_option(4, 0, exit_line, JenniferEnd, 50);
+    giq_option(4, 0, exit_line, JenniferEnd, NEUTRAL_REACTION);
 end
 
 procedure Jennifer19
 begin
-    gsay_message(462, 148, 50);
-    gsay_reply(462, 149);
+    gsay_message(NAME, 148, NEUTRAL_REACTION);
+    Reply(149);
     call Goodbyes();
-    giq_option(4, 0, exit_line, JenniferEnd, 50);
+    giq_option(4, 0, exit_line, JenniferEnd, NEUTRAL_REACTION);
 end
 
 procedure Jennifer20
 begin
-    gsay_reply(462, 150);
-    giq_option(5, 462, 151, BottomReact, 51);
-    giq_option(4, 462, 152, UpReact, 49);
-    giq_option(4, 462, 153, DownReact, 51);
+    Reply(150);
+    giq_option(5, NAME, 151, BottomReact, BAD_REACTION);
+    giq_option(4, NAME, 152, UpReact, GOOD_REACTION);
+    giq_option(4, NAME, 153, DownReact, BAD_REACTION);
 end
 
 procedure Jennifer21
 begin
-    gsay_reply(462, 154);
-    giq_option(4, 462, 155, Jennifer22, 50);
-    giq_option(4, 462, 156, JenniferEnd, 50);
+    Reply(154);
+    giq_option(4, NAME, 155, Jennifer22, NEUTRAL_REACTION);
+    giq_option(4, NAME, 156, JenniferEnd, NEUTRAL_REACTION);
 end
 
 procedure Jennifer22
 begin
-    gsay_message(462, 157, 50);
+    gsay_message(NAME, 157, NEUTRAL_REACTION);
 end
 
 procedure Jennifer23
@@ -311,89 +311,89 @@ begin
     flag1 := 1;
     flag2 := 0;
     last_seen := game_time;
-    gsay_message(462, 163, 49);
-    gsay_reply(462, 164);
+    gsay_message(NAME, 163, GOOD_REACTION);
+    Reply(164);
     if (line2flag == 0) then begin
-        giq_option(4, 462, 165, Jennifer02, 50);
+        giq_option(4, NAME, 165, Jennifer02, NEUTRAL_REACTION);
     end
     if (line5flag == 0) then begin
-        giq_option(4, 462, 166, Jennifer05, 50);
+        giq_option(4, NAME, 166, Jennifer05, NEUTRAL_REACTION);
     end
-    giq_option(4, 462, 167, JenniferEnd, 50);
-    giq_option(4, 462, 168, JenniferEnd, 50);
+    giq_option(4, NAME, 167, JenniferEnd, NEUTRAL_REACTION);
+    giq_option(4, NAME, 168, JenniferEnd, NEUTRAL_REACTION);
 end
 
 procedure Jennifer25
 begin
     flag1 := 0;
     flag2 := 1;
-    gsay_message(462, 163, 50);
-    gsay_reply(462, 164);
+    gsay_message(NAME, 163, NEUTRAL_REACTION);
+    Reply(164);
     if (not(line2flag)) then begin
-        giq_option(4, 462, 165, Jennifer02, 50);
+        giq_option(4, NAME, 165, Jennifer02, NEUTRAL_REACTION);
     end
     if (not(line5flag)) then begin
-        giq_option(4, 462, 166, Jennifer05, 50);
+        giq_option(4, NAME, 166, Jennifer05, NEUTRAL_REACTION);
     end
-    giq_option(4, 462, 167, JenniferEnd, 50);
-    giq_option(4, 462, 168, JenniferEnd, 50);
+    giq_option(4, NAME, 167, JenniferEnd, NEUTRAL_REACTION);
+    giq_option(4, NAME, 168, JenniferEnd, NEUTRAL_REACTION);
 end
 
 procedure Jennifer26
 begin
-    gsay_reply(462, 169);
-    giq_option(4, 462, 170, BigDownReact, 51);
-    giq_option(4, 462, 171, Jennifer27, 50);
-    giq_option(4, 462, 172, JenniferEnd, 50);
+    Reply(169);
+    giq_option(4, NAME, 170, BigDownReact, BAD_REACTION);
+    giq_option(4, NAME, 171, Jennifer27, NEUTRAL_REACTION);
+    giq_option(4, NAME, 172, JenniferEnd, NEUTRAL_REACTION);
 end
 
 procedure Jennifer27
 begin
-    gsay_reply(462, 173);
+    Reply(173);
     if (not(line2flag)) then begin
-        giq_option(4, 462, 174, Jennifer02, 50);
+        giq_option(4, NAME, 174, Jennifer02, NEUTRAL_REACTION);
     end
     if (not(line5flag)) then begin
-        giq_option(4, 462, 175, Jennifer05, 50);
+        giq_option(4, NAME, 175, Jennifer05, NEUTRAL_REACTION);
     end
-    giq_option(4, 462, 176, JenniferEnd, 50);
+    giq_option(4, NAME, 176, JenniferEnd, NEUTRAL_REACTION);
 end
 
 procedure Jennifer28
 begin
-    gsay_reply(462, 177);
-    giq_option(4, 462, 178, Jennifer27, 50);
-    giq_option(4, 462, 179, Jennifer27, 51);
-    giq_option(4, 462, 180, JenniferEnd, 50);
+    Reply(177);
+    giq_option(4, NAME, 178, Jennifer27, NEUTRAL_REACTION);
+    giq_option(4, NAME, 179, Jennifer27, BAD_REACTION);
+    giq_option(4, NAME, 180, JenniferEnd, NEUTRAL_REACTION);
 end
 
 procedure Jennifer29
 begin
-    gsay_reply(462, 181);
-    giq_option(4, 462, 182, Jennifer30, 50);
-    giq_option(4, 462, 183, BottomReact, 51);
+    Reply(181);
+    giq_option(4, NAME, 182, Jennifer30, NEUTRAL_REACTION);
+    giq_option(4, NAME, 183, BottomReact, BAD_REACTION);
 end
 
 procedure Jennifer30
 begin
-    gsay_message(462, 184, 50);
-    gsay_message(462, 185, 50);
-    gsay_reply(462, 186);
-    giq_option(7, 462, 187, Jennifer31, 50);
-    giq_option(4, 462, 188, UpReact, 49);
-    giq_option(4, 462, 189, DownReact, 51);
+    gsay_message(NAME, 184, NEUTRAL_REACTION);
+    gsay_message(NAME, 185, NEUTRAL_REACTION);
+    Reply(186);
+    giq_option(7, NAME, 187, Jennifer31, NEUTRAL_REACTION);
+    giq_option(4, NAME, 188, UpReact, GOOD_REACTION);
+    giq_option(4, NAME, 189, DownReact, BAD_REACTION);
     if (not(line2flag)) then begin
-        giq_option(4, 462, 190, Jennifer02, 50);
+        giq_option(4, NAME, 190, Jennifer02, NEUTRAL_REACTION);
     end
 end
 
 procedure Jennifer31
 begin
-    gsay_reply(462, 191);
-    giq_option(4, 462, 192, JenniferEnd, 50);
-    giq_option(4, 462, 193, DownReact, 51);
+    Reply(191);
+    giq_option(4, NAME, 192, JenniferEnd, NEUTRAL_REACTION);
+    giq_option(4, NAME, 193, DownReact, BAD_REACTION);
     if (not(line2flag)) then begin
-        giq_option(4, 462, 194, Jennifer02, 50);
+        giq_option(4, NAME, 194, Jennifer02, NEUTRAL_REACTION);
     end
 end
 
@@ -436,40 +436,40 @@ begin
         if (LVar0 > 10) then begin
             LVar0 := 1;
         end
-        LVar1 := message_str(462, 195);
+        LVar1 := mstr(195);
         if (LVar0 == 2) then begin
-            LVar1 := message_str(462, 196);
+            LVar1 := mstr(196);
         end
         else begin
             if (LVar0 == 3) then begin
-                LVar1 := message_str(462, 197);
+                LVar1 := mstr(197);
             end
             else begin
                 if (LVar0 == 4) then begin
-                    LVar1 := message_str(462, 198);
+                    LVar1 := mstr(198);
                 end
                 else begin
                     if (LVar0 == 5) then begin
-                        LVar1 := message_str(462, 199);
+                        LVar1 := mstr(199);
                     end
                     else begin
                         if (LVar0 == 6) then begin
-                            LVar1 := message_str(462, 200);
+                            LVar1 := mstr(200);
                         end
                         else begin
                             if (LVar0 == 7) then begin
-                                LVar1 := message_str(462, 201);
+                                LVar1 := mstr(201);
                             end
                             else begin
                                 if ((last_seen - game_time) > 86400) then begin
                                     if (LVar0 == 8) then begin
-                                        LVar1 := message_str(462, 202);
+                                        LVar1 := mstr(202);
                                     end
                                     if (LVar0 == 9) then begin
-                                        LVar1 := message_str(462, 203);
+                                        LVar1 := mstr(203);
                                     end
                                     if (LVar0 == 10) then begin
-                                        LVar1 := message_str(462, 204);
+                                        LVar1 := mstr(204);
                                     end
                                 end
                                 else begin
@@ -486,25 +486,25 @@ begin
         if (LVar0 > 6) then begin
             LVar0 := 1;
         end
-        LVar1 := message_str(462, 205);
+        LVar1 := mstr(205);
         if (LVar0 == 2) then begin
-            LVar1 := message_str(462, 206);
+            LVar1 := mstr(206);
         end
         else begin
             if (LVar0 == 3) then begin
-                LVar1 := message_str(462, 207);
+                LVar1 := mstr(207);
             end
             else begin
                 if (LVar0 == 4) then begin
-                    LVar1 := message_str(462, 208);
+                    LVar1 := mstr(208);
                 end
                 else begin
                     if (LVar0 == 5) then begin
-                        LVar1 := message_str(462, 209);
+                        LVar1 := mstr(209);
                     end
                     else begin
                         if (LVar0 == 6) then begin
-                            LVar1 := message_str(462, 210);
+                            LVar1 := mstr(210);
                         end
                     end
                 end
@@ -513,7 +513,7 @@ begin
     end
     LVar0 := LVar0 + 1;
     last_seen := game_time;
-    float_msg(self_obj, LVar1, 0);
+    float_msg(self_obj, LVar1, FLOAT_MSG_NORMAL);
 end
 
 procedure JenniferRandom2
@@ -526,27 +526,27 @@ begin
     if (LVar0 > 5) then begin
         LVar0 := 1;
     end
-    LVar1 := message_str(462, 211);
+    LVar1 := mstr(211);
     if (LVar0 == 2) then begin
-        LVar1 := message_str(462, 212) + proto_data(obj_pid(dude_obj), 1) + message_str(462, 213);
+        LVar1 := mstr(212) + proto_data(obj_pid(dude_obj), 1) + mstr(213);
     end
     else begin
         if (LVar0 == 3) then begin
-            LVar1 := message_str(462, 214);
+            LVar1 := mstr(214);
         end
         else begin
             if (LVar0 == 4) then begin
-                LVar1 := message_str(462, 215) + proto_data(obj_pid(dude_obj), 1) + message_str(462, 216);
+                LVar1 := mstr(215) + proto_data(obj_pid(dude_obj), 1) + mstr(216);
             end
             else begin
                 if (LVar0 == 5) then begin
-                    LVar1 := message_str(462, 217);
+                    LVar1 := mstr(217);
                 end
             end
         end
     end
     LVar0 := LVar0 + 1;
-    float_msg(self_obj, LVar1, 0);
+    float_msg(self_obj, LVar1, FLOAT_MSG_NORMAL);
 end
 
 procedure JenniferRandom3
@@ -559,33 +559,33 @@ begin
     if (LVar0 > 8) then begin
         LVar0 := 1;
     end
-    LVar1 := message_str(462, 218);
+    LVar1 := mstr(218);
     if (LVar0 == 2) then begin
-        LVar1 := message_str(462, 219);
+        LVar1 := mstr(219);
     end
     else begin
         if (LVar0 == 3) then begin
-            LVar1 := message_str(462, 220);
+            LVar1 := mstr(220);
         end
         else begin
             if (LVar0 == 4) then begin
-                LVar1 := message_str(462, 221);
+                LVar1 := mstr(221);
             end
             else begin
                 if (LVar0 == 5) then begin
-                    LVar1 := message_str(462, 222);
+                    LVar1 := mstr(222);
                 end
                 else begin
                     if (LVar0 == 6) then begin
-                        LVar1 := message_str(462, 223);
+                        LVar1 := mstr(223);
                     end
                     else begin
                         if (LVar0 == 7) then begin
-                            LVar1 := message_str(462, 224);
+                            LVar1 := mstr(224);
                         end
                         else begin
                             if (LVar0 == 8) then begin
-                                LVar1 := message_str(462, 225);
+                                LVar1 := mstr(225);
                             end
                         end
                     end
@@ -594,7 +594,7 @@ begin
         end
     end
     LVar0 := LVar0 + 1;
-    float_msg(self_obj, LVar1, 0);
+    float_msg(self_obj, LVar1, FLOAT_MSG_NORMAL);
 end
 
 procedure JenniferBackground1
@@ -609,47 +609,47 @@ begin
             LVar0 := 1;
         end
         if (LVar0 == 1) then begin
-            LVar1 := message_str(462, 226);
+            LVar1 := mstr(226);
         end
         else begin
             if (LVar0 == 2) then begin
-                LVar1 := message_str(462, 227);
+                LVar1 := mstr(227);
             end
             else begin
                 if (LVar0 == 3) then begin
-                    LVar1 := proto_data(obj_pid(dude_obj), 1) + message_str(462, 228);
+                    LVar1 := proto_data(obj_pid(dude_obj), 1) + mstr(228);
                 end
                 else begin
                     if (LVar0 == 4) then begin
-                        LVar1 := message_str(462, 229);
+                        LVar1 := mstr(229);
                     end
                     else begin
                         if (LVar0 == 5) then begin
-                            LVar1 := message_str(462, 230);
+                            LVar1 := mstr(230);
                         end
                         else begin
                             if (LVar0 == 6) then begin
-                                LVar1 := message_str(462, 231);
+                                LVar1 := mstr(231);
                             end
                             else begin
                                 if (LVar0 == 7) then begin
-                                    LVar1 := message_str(462, 232);
+                                    LVar1 := mstr(232);
                                 end
                                 else begin
                                     if (LVar0 == 8) then begin
-                                        LVar1 := message_str(462, 233);
+                                        LVar1 := mstr(233);
                                     end
                                     else begin
                                         if (LVar0 == 9) then begin
-                                            LVar1 := message_str(462, 234);
+                                            LVar1 := mstr(234);
                                         end
                                         else begin
                                             if (LVar0 == 10) then begin
-                                                LVar1 := message_str(462, 235);
+                                                LVar1 := mstr(235);
                                             end
                                             else begin
                                                 if (LVar0 == 11) then begin
-                                                    LVar1 := message_str(462, 236);
+                                                    LVar1 := mstr(236);
                                                 end
                                             end
                                         end
@@ -664,23 +664,23 @@ begin
     end
     else begin
         if (LVar0 == 1) then begin
-            LVar1 := message_str(462, 237);
+            LVar1 := mstr(237);
         end
         else begin
             if (LVar0 == 2) then begin
-                LVar1 := message_str(462, 238);
+                LVar1 := mstr(238);
             end
             else begin
                 if (LVar0 == 3) then begin
-                    LVar1 := message_str(462, 239);
+                    LVar1 := mstr(239);
                 end
                 else begin
                     if (LVar0 == 4) then begin
-                        LVar1 := message_str(462, 240);
+                        LVar1 := mstr(240);
                     end
                     else begin
                         if (LVar0 == 5) then begin
-                            LVar1 := message_str(462, 241);
+                            LVar1 := mstr(241);
                         end
                     end
                 end
@@ -688,7 +688,7 @@ begin
         end
     end
     LVar0 := LVar0 + 1;
-    float_msg(self_obj, LVar1, 0);
+    float_msg(self_obj, LVar1, FLOAT_MSG_NORMAL);
     last_seen := game_time;
 end
 
@@ -699,16 +699,16 @@ end
 
 procedure critter_p_proc
 begin
-    if (global_var(250)) then begin
+    if (global_var(GVAR_ENEMY_BROTHERHOOD)) then begin
         hostile := 1;
     end
     if (tile_distance_objs(self_obj, dude_obj) > 12) then begin
         hostile := 0;
     end
     if (hostile) then begin
-        set_global_var(250, 1);
+        set_global_var(GVAR_ENEMY_BROTHERHOOD, 1);
         hostile := 0;
-        attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+        attack(dude_obj);
     end
     else begin
         if (warned == 0) then begin
@@ -738,20 +738,20 @@ end
 
 procedure destroy_p_proc
 begin
-    set_global_var(250, 1);
+    set_global_var(GVAR_ENEMY_BROTHERHOOD, 1);
     rm_timer_event(self_obj);
     if (source_obj == dude_obj) then begin
-        if (((global_var(160) + global_var(159)) >= 25) and ((global_var(159) > (2 * global_var(160))) or (global_var(156) == 1))) then begin
-            set_global_var(156, 1);
-            set_global_var(157, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
+            set_global_var(GVAR_BERSERKER_REPUTATION, 1);
+            set_global_var(GVAR_CHAMPION_REPUTATION, 0);
         end
-        if (((global_var(160) + global_var(159)) >= 25) and ((global_var(160) > (3 * global_var(159))) or (global_var(157) == 1))) then begin
-            set_global_var(157, 1);
-            set_global_var(156, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(160) > (3 * global_var(GVAR_GOOD_MONSTER))) or (global_var(GVAR_CHAMPION_REPUTATION) == 1))) then begin
+            set_global_var(GVAR_CHAMPION_REPUTATION, 1);
+            set_global_var(GVAR_BERSERKER_REPUTATION, 0);
         end
-        set_global_var(159, global_var(159) + 1);
-        if ((global_var(159) % 2) == 0) then begin
-            set_global_var(155, global_var(155) - 1);
+        set_global_var(GVAR_GOOD_MONSTER, global_var(GVAR_GOOD_MONSTER) + 1);
+        if ((global_var(GVAR_GOOD_MONSTER) % 2) == 0) then begin
+            set_global_var(GVAR_PLAYER_REPUATION, global_var(GVAR_PLAYER_REPUATION) - 1);
         end
     end
 end
@@ -760,10 +760,10 @@ procedure look_at_p_proc
 begin
     script_overrides;
     if (known) then begin
-        display_msg(message_str(462, 100));
+        display_mstr(100);
     end
     else begin
-        display_msg(message_str(462, 101));
+        display_mstr(101);
     end
 end
 
@@ -773,31 +773,31 @@ begin
         set_local_var(0, 50);
         set_local_var(1, 2);
         set_local_var(2, 1);
-        set_local_var(0, local_var(0) + (5 * get_critter_stat(dude_obj, 3)) - 25);
-        set_local_var(0, local_var(0) + (10 * has_trait(0, dude_obj, 10)));
-        if (has_trait(0, dude_obj, 39)) then begin
-            if (global_var(155) > 0) then begin
-                set_local_var(0, local_var(0) + global_var(155));
+        set_local_var(0, local_var(0) + (5 * dude_charisma) - 25);
+        set_local_var(0, local_var(0) + (10 * has_trait(TRAIT_PERK, dude_obj, PERK_presence)));
+        if (has_trait(TRAIT_PERK, dude_obj, PERK_cult_of_personality)) then begin
+            if (global_var(GVAR_PLAYER_REPUATION) > 0) then begin
+                set_local_var(0, local_var(0) + global_var(GVAR_PLAYER_REPUATION));
             end
             else begin
-                set_local_var(0, local_var(0) - global_var(155));
+                set_local_var(0, local_var(0) - global_var(GVAR_PLAYER_REPUATION));
             end
         end
         else begin
             if (local_var(3) == 1) then begin
-                set_local_var(0, local_var(0) - global_var(155));
+                set_local_var(0, local_var(0) - global_var(GVAR_PLAYER_REPUATION));
             end
             else begin
-                set_local_var(0, local_var(0) + global_var(155));
+                set_local_var(0, local_var(0) + global_var(GVAR_PLAYER_REPUATION));
             end
         end
-        if (global_var(158) > 2) then begin
+        if (global_var(GVAR_CHILDKILLER_REPUATION) > 2) then begin
             set_local_var(0, local_var(0) - 30);
         end
-        if (((global_var(160) + global_var(159)) >= 25) and ((global_var(160) > (3 * global_var(159))) or (global_var(157) == 1))) then begin
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(160) > (3 * global_var(GVAR_GOOD_MONSTER))) or (global_var(GVAR_CHAMPION_REPUTATION) == 1))) then begin
             set_local_var(0, local_var(0) + 20);
         end
-        if (((global_var(160) + global_var(159)) >= 25) and ((global_var(159) > (2 * global_var(160))) or (global_var(156) == 1))) then begin
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
             set_local_var(0, local_var(0) - 20);
         end
         call ReactToLevel();
@@ -890,6 +890,5 @@ end
 
 procedure Goodbyes
 begin
-    exit_line := message_str(634, random(100, 105));
+    exit_line := message_str(SCRIPT_MODREACT, random(100, 105));
 end
-
