diff --git a/Games/Fallout1.Steam/Original/JEREM.ssl b/Games/Fallout1.Steam/ReDefine/JEREM.ssl
index ce9474a..fd1397f 100644
--- a/Games/Fallout1.Steam/Original/JEREM.ssl
+++ b/Games/Fallout1.Steam/ReDefine/JEREM.ssl
@@ -74,14 +74,14 @@ begin
     if (only_once) then begin
         only_once := 0;
         home_tile := 12098;
-        critter_add_trait(self_obj, 1, 6, 34);
-        critter_add_trait(self_obj, 1, 5, 69);
-        if (global_var(233) == 1) then begin
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 34);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_CHILDREN_TRUE);
+        if (global_var(GVAR_MASTER_FILLER_6) == 1) then begin
             set_obj_visibility(self_obj, 1);
             moving := 0;
         end
         else begin
-            if (global_var(232) == 1) then begin
+            if (global_var(GVAR_MASTER_FILLER_5) == 1) then begin
                 set_obj_visibility(self_obj, 1);
                 moving := 0;
                 add_timer_event(self_obj, game_ticks(300), 1);
@@ -89,15 +89,15 @@ begin
         end
     end
     else begin
-        if (global_var(233) != 1) then begin
-            if (script_action == 11) then begin
+        if (global_var(GVAR_MASTER_FILLER_6) != 1) then begin
+            if (script_action == talk_proc) then begin
                 call pre_dialogue();
             end
             else begin
-                if (script_action == 22) then begin
+                if (script_action == timed_event_proc) then begin
                     if (fixed_param == 1) then begin
                         moving := 1;
-                        set_global_var(232, 0);
+                        set_global_var(GVAR_MASTER_FILLER_5, 0);
                         if (local_var(1) == 0) then begin
                             move_to(self_obj, home_tile, 0);
                             set_local_var(3, 7);
@@ -121,19 +121,19 @@ begin
                     end
                 end
                 else begin
-                    if (script_action == 4) then begin
+                    if (script_action == pickup_proc) then begin
                         HOSTILE := 1;
                     end
                 end
             end
-            if (script_action == 12) then begin
+            if (script_action == critter_proc) then begin
                 set_map_var(1, 0);
                 if (tile_distance_objs(dude_obj, self_obj) < 8) then begin
                     set_map_var(1, 1);
                 end
                 if (HOSTILE) then begin
                     HOSTILE := 0;
-                    attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+                    attack(dude_obj);
                 end
                 if (moving) then begin
                     my_hex := tile_num(self_obj);
@@ -149,7 +149,7 @@ begin
                             set_local_var(1, 0);
                             set_local_var(3, 6);
                             set_obj_visibility(self_obj, 1);
-                            set_global_var(232, 1);
+                            set_global_var(GVAR_MASTER_FILLER_5, 1);
                             move_to(self_obj, 7000, 0);
                             add_timer_event(self_obj, game_ticks(300), 1);
                         end
@@ -217,7 +217,7 @@ begin
                     if (obj_can_see_obj(self_obj, dude_obj)) then begin
                         DISGUISED := 0;
                         if (obj_pid(critter_inven_obj(dude_obj, 0)) == 113) then begin
-                            if (metarule(16, 0) > 1) then begin
+                            if (party_member_count(0) > 1) then begin
                                 DISGUISED := 0;
                             end
                             else begin
@@ -242,25 +242,25 @@ begin
                 end
             end
             else begin
-                if ((script_action == 21) or (script_action == 3)) then begin
+                if ((script_action == look_at_proc) or (script_action == description_proc)) then begin
                     script_overrides;
-                    display_msg(message_str(672, 100));
+                    display_mstr(100);
                 end
                 else begin
-                    if (script_action == 18) then begin
-                        set_global_var(233, 1);
+                    if (script_action == destroy_proc) then begin
+                        set_global_var(GVAR_MASTER_FILLER_6, 1);
                         if (source_obj == dude_obj) then begin
-                            if (((global_var(160) + global_var(159)) >= 25) and ((global_var(159) > (2 * global_var(160))) or (global_var(156) == 1))) then begin
-                                set_global_var(156, 1);
-                                set_global_var(157, 0);
+                            if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
+                                set_global_var(GVAR_BERSERKER_REPUTATION, 1);
+                                set_global_var(GVAR_CHAMPION_REPUTATION, 0);
                             end
-                            if (((global_var(160) + global_var(159)) >= 25) and ((global_var(160) > (3 * global_var(159))) or (global_var(157) == 1))) then begin
-                                set_global_var(157, 1);
-                                set_global_var(156, 0);
+                            if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(160) > (3 * global_var(GVAR_GOOD_MONSTER))) or (global_var(GVAR_CHAMPION_REPUTATION) == 1))) then begin
+                                set_global_var(GVAR_CHAMPION_REPUTATION, 1);
+                                set_global_var(GVAR_BERSERKER_REPUTATION, 0);
                             end
-                            set_global_var(160, global_var(160) + 1);
-                            if ((global_var(160) % 6) == 0) then begin
-                                set_global_var(155, global_var(155) + 1);
+                            set_global_var(GVAR_BAD_MONSTER, global_var(GVAR_BAD_MONSTER) + 1);
+                            if ((global_var(GVAR_BAD_MONSTER) % 6) == 0) then begin
+                                set_global_var(GVAR_PLAYER_REPUATION, global_var(GVAR_PLAYER_REPUATION) + 1);
                             end
                         end
                     end
@@ -274,7 +274,7 @@ procedure pre_dialogue
 begin
     again := again + 1;
     if (obj_pid(critter_inven_obj(dude_obj, 0)) == 113) then begin
-        if (metarule(16, 0) > 1) then begin
+        if (party_member_count(0) > 1) then begin
             DISGUISED := 0;
         end
         else begin
@@ -297,7 +297,7 @@ begin
     gsay_start;
     again := again + 1;
     if (obj_pid(critter_inven_obj(dude_obj, 0)) == 113) then begin
-        if (metarule(16, 0) > 1) then begin
+        if (party_member_count(0) > 1) then begin
             DISGUISED := 0;
         end
         else begin
@@ -311,7 +311,7 @@ begin
                 call jerem26();
             end
             else begin
-                if ((again == 2) and (global_var(231) == 1)) then begin
+                if ((again == 2) and (global_var(GVAR_MASTER_FILLER_4) == 1)) then begin
                     call jerem23();
                 end
                 else begin
@@ -356,7 +356,7 @@ end
 
 procedure checkarea
 begin
-    here := tile_num(dude_obj);
+    here := dude_tile;
     area := 2;
     if (tile_distance(here, 13475) < 15) then begin
         area := 1;
@@ -395,14 +395,14 @@ end
 
 procedure jerem00
 begin
-    if (get_critter_stat(dude_obj, 34) == 0) then begin
-        gsay_reply(672, 102);
+    if (dude_gender == 0) then begin
+        Reply(102);
     end
     else begin
-        gsay_reply(672, 192);
+        Reply(192);
     end
-    giq_option(-3, 672, 103, jerem01, 50);
-    giq_option(4, 672, 104, jerem00a, 50);
+    giq_option(-3, NAME, 103, jerem01, NEUTRAL_REACTION);
+    giq_option(4, NAME, 104, jerem00a, NEUTRAL_REACTION);
 end
 
 procedure jerem00a
@@ -417,17 +417,17 @@ end
 
 procedure jerem01
 begin
-    gsay_message(672, 105, 50);
+    gsay_message(NAME, 105, NEUTRAL_REACTION);
 end
 
 procedure jerem02
 begin
-    gsay_reply(672, 106);
-    giq_option(7, 672, 107, jerem02a, 50);
-    giq_option(4, 672, 108, jerem04, 50);
-    giq_option(4, 672, 109, jerem04, 50);
-    if (((global_var(160) + global_var(159)) >= 25) and ((global_var(159) > (2 * global_var(160))) or (global_var(156) == 1))) then begin
-        giq_option(4, 672, 110, jerem02b, 50);
+    Reply(106);
+    giq_option(7, NAME, 107, jerem02a, NEUTRAL_REACTION);
+    giq_option(4, NAME, 108, jerem04, NEUTRAL_REACTION);
+    giq_option(4, NAME, 109, jerem04, NEUTRAL_REACTION);
+    if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
+        giq_option(4, NAME, 110, jerem02b, NEUTRAL_REACTION);
     end
 end
 
@@ -453,10 +453,10 @@ end
 
 procedure jerem03
 begin
-    gsay_reply(672, 111);
-    giq_option(4, 672, 112, jerem03a, 50);
-    giq_option(4, 672, 113, jerem07, 50);
-    giq_option(4, 672, 114, jeremend, 50);
+    Reply(111);
+    giq_option(4, NAME, 112, jerem03a, NEUTRAL_REACTION);
+    giq_option(4, NAME, 113, jerem07, NEUTRAL_REACTION);
+    giq_option(4, NAME, 114, jeremend, NEUTRAL_REACTION);
 end
 
 procedure jerem03a
@@ -471,16 +471,16 @@ end
 
 procedure jerem04
 begin
-    gsay_message(672, 115, 50);
+    gsay_message(NAME, 115, NEUTRAL_REACTION);
 end
 
 procedure jerem05
 begin
-    gsay_reply(672, 116);
-    giq_option(4, 672, 117, jerem03, 50);
-    giq_option(4, 672, 118, jerem05a, 50);
-    if (((global_var(160) + global_var(159)) >= 25) and ((global_var(159) > (2 * global_var(160))) or (global_var(156) == 1))) then begin
-        giq_option(4, 672, 119, jeremcbt, 51);
+    Reply(116);
+    giq_option(4, NAME, 117, jerem03, NEUTRAL_REACTION);
+    giq_option(4, NAME, 118, jerem05a, NEUTRAL_REACTION);
+    if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
+        giq_option(4, NAME, 119, jeremcbt, BAD_REACTION);
     end
 end
 
@@ -496,24 +496,24 @@ end
 
 procedure jerem06
 begin
-    gsay_message(672, 120, 51);
+    gsay_message(NAME, 120, BAD_REACTION);
     call jeremcbt();
 end
 
 procedure jerem07
 begin
-    gsay_reply(672, 121);
-    giq_option(4, 672, 122, jerem06, 51);
-    giq_option(-4, 672, 103, jerem01, 50);
-    giq_option(7, 672, 124, jerem10, 50);
-    if (((global_var(160) + global_var(159)) >= 25) and ((global_var(159) > (2 * global_var(160))) or (global_var(156) == 1))) then begin
-        giq_option(4, 672, 125, jeremcbt, 51);
+    Reply(121);
+    giq_option(4, NAME, 122, jerem06, BAD_REACTION);
+    giq_option(-4, NAME, 103, jerem01, NEUTRAL_REACTION);
+    giq_option(7, NAME, 124, jerem10, NEUTRAL_REACTION);
+    if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
+        giq_option(4, NAME, 125, jeremcbt, BAD_REACTION);
     end
 end
 
 procedure jerem08
 begin
-    gsay_reply(672, 126);
+    Reply(126);
     call jerem09a();
 end
 
@@ -539,52 +539,52 @@ end
 
 procedure jerem09
 begin
-    gsay_reply(672, 190);
+    Reply(190);
     call jerem09a();
 end
 
 procedure jerem09a
 begin
-    giq_option(4, 672, 127, jerem08a, 50);
-    giq_option(4, 672, 128, jerem12, 51);
-    giq_option(8, 672, 129, jerem11, 50);
-    if (((global_var(160) + global_var(159)) >= 25) and ((global_var(159) > (2 * global_var(160))) or (global_var(156) == 1))) then begin
-        giq_option(4, 672, 130, jerem08b, 50);
+    giq_option(4, NAME, 127, jerem08a, NEUTRAL_REACTION);
+    giq_option(4, NAME, 128, jerem12, BAD_REACTION);
+    giq_option(8, NAME, 129, jerem11, NEUTRAL_REACTION);
+    if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
+        giq_option(4, NAME, 130, jerem08b, NEUTRAL_REACTION);
     end
 end
 
 procedure jerem10
 begin
-    gsay_reply(672, 131);
-    giq_option(4, 672, 127, jerem08a, 50);
-    giq_option(8, 672, 133, jerem11, 50);
+    Reply(131);
+    giq_option(4, NAME, 127, jerem08a, NEUTRAL_REACTION);
+    giq_option(8, NAME, 133, jerem11, NEUTRAL_REACTION);
 end
 
 procedure jerem11
 begin
-    gsay_message(672, 134, 50);
-    set_global_var(231, 1);
+    gsay_message(NAME, 134, NEUTRAL_REACTION);
+    set_global_var(GVAR_MASTER_FILLER_4, 1);
 end
 
 procedure jerem12
 begin
-    gsay_message(672, 135, 51);
+    gsay_message(NAME, 135, BAD_REACTION);
     call jeremcbt();
 end
 
 procedure jerem13
 begin
-    gsay_reply(672, 136);
+    Reply(136);
     call jerem19a();
 end
 
 procedure jerem14
 begin
-    gsay_reply(672, 141);
-    giq_option(4, 672, 142, jeremret, 50);
-    giq_option(4, 672, 143, jerem14a, 50);
-    if (((global_var(160) + global_var(159)) >= 25) and ((global_var(159) > (2 * global_var(160))) or (global_var(156) == 1))) then begin
-        giq_option(4, 672, 144, jeremcbt, 51);
+    Reply(141);
+    giq_option(4, NAME, 142, jeremret, NEUTRAL_REACTION);
+    giq_option(4, NAME, 143, jerem14a, NEUTRAL_REACTION);
+    if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
+        giq_option(4, NAME, 144, jeremcbt, BAD_REACTION);
     end
 end
 
@@ -600,56 +600,56 @@ end
 
 procedure jerem15
 begin
-    gsay_message(672, 145, 51);
+    gsay_message(NAME, 145, BAD_REACTION);
     call jeremcbt();
 end
 
 procedure jerem17
 begin
-    gsay_reply(672, 150);
+    Reply(150);
     call jerem18a();
 end
 
 procedure jerem18
 begin
-    gsay_reply(672, 156);
+    Reply(156);
     call jerem18a();
 end
 
 procedure jerem18a
 begin
-    giq_option(4, 672, 151, jeremret, 50);
-    giq_option(4, 672, 152, jerem12, 51);
-    if (((global_var(160) + global_var(159)) >= 25) and ((global_var(159) > (2 * global_var(160))) or (global_var(156) == 1))) then begin
-        giq_option(4, 672, 153, jeremcbt, 51);
+    giq_option(4, NAME, 151, jeremret, NEUTRAL_REACTION);
+    giq_option(4, NAME, 152, jerem12, BAD_REACTION);
+    if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
+        giq_option(4, NAME, 153, jeremcbt, BAD_REACTION);
     end
 end
 
 procedure jerem19
 begin
-    gsay_reply(672, 154);
+    Reply(154);
     call jerem19a();
 end
 
 procedure jerem19a
 begin
-    giq_option(4, 672, 137, jerem14, 50);
-    giq_option(4, 672, 138, jerem15, 51);
-    if (((global_var(160) + global_var(159)) >= 25) and ((global_var(159) > (2 * global_var(160))) or (global_var(156) == 1))) then begin
-        giq_option(4, 672, 139, jeremcbt, 51);
+    giq_option(4, NAME, 137, jerem14, NEUTRAL_REACTION);
+    giq_option(4, NAME, 138, jerem15, BAD_REACTION);
+    if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
+        giq_option(4, NAME, 139, jeremcbt, BAD_REACTION);
     end
-    giq_option(-3, 672, 140, jerem16, 50);
+    giq_option(-3, NAME, 140, jerem16, NEUTRAL_REACTION);
 end
 
 procedure jerem20
 begin
-    gsay_reply(672, 157);
-    giq_option(-3, 672, 158, jerem01, 50);
-    giq_option(4, 672, 159, jerem12, 51);
-    giq_option(4, 672, 160, jerem15, 51);
-    giq_option(7, 672, 161, jerem20a, 50);
-    if (((global_var(160) + global_var(159)) >= 25) and ((global_var(159) > (2 * global_var(160))) or (global_var(156) == 1))) then begin
-        giq_option(4, 672, 162, jeremcbt, 51);
+    Reply(157);
+    giq_option(-3, NAME, 158, jerem01, NEUTRAL_REACTION);
+    giq_option(4, NAME, 159, jerem12, BAD_REACTION);
+    giq_option(4, NAME, 160, jerem15, BAD_REACTION);
+    giq_option(7, NAME, 161, jerem20a, NEUTRAL_REACTION);
+    if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
+        giq_option(4, NAME, 162, jeremcbt, BAD_REACTION);
     end
 end
 
@@ -665,12 +665,12 @@ end
 
 procedure jerem21
 begin
-    gsay_reply(672, 163);
-    giq_option(4, 672, 164, jerem12, 51);
-    giq_option(4, 672, 165, jerem12, 51);
-    giq_option(4, 672, 166, jerem21a, 50);
-    if (((global_var(160) + global_var(159)) >= 25) and ((global_var(159) > (2 * global_var(160))) or (global_var(156) == 1))) then begin
-        giq_option(4, 672, 167, jeremcbt, 51);
+    Reply(163);
+    giq_option(4, NAME, 164, jerem12, BAD_REACTION);
+    giq_option(4, NAME, 165, jerem12, BAD_REACTION);
+    giq_option(4, NAME, 166, jerem21a, NEUTRAL_REACTION);
+    if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
+        giq_option(4, NAME, 167, jeremcbt, BAD_REACTION);
     end
 end
 
@@ -686,75 +686,75 @@ end
 
 procedure jerem22
 begin
-    gsay_reply(672, 168);
-    giq_option(4, 672, 169, jeremret, 50);
-    giq_option(4, 672, 170, jerem12, 51);
-    giq_option(4, 672, 171, jerem18, 50);
+    Reply(168);
+    giq_option(4, NAME, 169, jeremret, NEUTRAL_REACTION);
+    giq_option(4, NAME, 170, jerem12, BAD_REACTION);
+    giq_option(4, NAME, 171, jerem18, NEUTRAL_REACTION);
 end
 
 procedure jerem23
 begin
-    if (get_critter_stat(dude_obj, 34) == 0) then begin
-        gsay_reply(672, 172);
+    if (dude_gender == 0) then begin
+        Reply(172);
     end
     else begin
-        gsay_reply(672, 173);
+        Reply(173);
     end
-    giq_option(4, 672, 174, jerem24, 50);
-    giq_option(4, 672, 175, jeremend, 50);
-    giq_option(4, 672, 176, jeremend, 50);
-    giq_option(4, 672, 177, jeremcbt, 51);
+    giq_option(4, NAME, 174, jerem24, NEUTRAL_REACTION);
+    giq_option(4, NAME, 175, jeremend, NEUTRAL_REACTION);
+    giq_option(4, NAME, 176, jeremend, NEUTRAL_REACTION);
+    giq_option(4, NAME, 177, jeremcbt, BAD_REACTION);
 end
 
 procedure jerem24
 begin
-    gsay_reply(672, 178);
-    giq_option(4, 672, 179, jeremend, 50);
-    giq_option(4, 672, 180, jerem26, 50);
-    giq_option(4, 672, 181, jerem25, 50);
+    Reply(178);
+    giq_option(4, NAME, 179, jeremend, NEUTRAL_REACTION);
+    giq_option(4, NAME, 180, jerem26, NEUTRAL_REACTION);
+    giq_option(4, NAME, 181, jerem25, NEUTRAL_REACTION);
 end
 
 procedure jerem25
 begin
-    gsay_reply(672, 182);
-    giq_option(4, 672, 183, jeremend, 50);
-    giq_option(4, 672, 184, jerem26, 50);
-    if (((global_var(160) + global_var(159)) >= 25) and ((global_var(159) > (2 * global_var(160))) or (global_var(156) == 1))) then begin
-        giq_option(4, 672, 185, jerem26, 50);
+    Reply(182);
+    giq_option(4, NAME, 183, jeremend, NEUTRAL_REACTION);
+    giq_option(4, NAME, 184, jerem26, NEUTRAL_REACTION);
+    if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
+        giq_option(4, NAME, 185, jerem26, NEUTRAL_REACTION);
     end
 end
 
 procedure jerem26
 begin
-    gsay_message(672, 187, 51);
+    gsay_message(NAME, 187, BAD_REACTION);
     call jeremcbt();
 end
 
 procedure jerem27
 begin
-    if (get_critter_stat(dude_obj, 34) == 0) then begin
-        gsay_reply(672, 172);
+    if (dude_gender == 0) then begin
+        Reply(172);
     end
     else begin
-        gsay_reply(672, 173);
+        Reply(173);
     end
-    giq_option(4, 672, 174, jerem28, 51);
-    giq_option(4, 672, 175, jeremend, 50);
-    giq_option(4, 672, 176, jeremend, 50);
-    if (((global_var(160) + global_var(159)) >= 25) and ((global_var(159) > (2 * global_var(160))) or (global_var(156) == 1))) then begin
-        giq_option(4, 672, 177, jeremcbt, 51);
+    giq_option(4, NAME, 174, jerem28, BAD_REACTION);
+    giq_option(4, NAME, 175, jeremend, NEUTRAL_REACTION);
+    giq_option(4, NAME, 176, jeremend, NEUTRAL_REACTION);
+    if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
+        giq_option(4, NAME, 177, jeremcbt, BAD_REACTION);
     end
 end
 
 procedure jerem28
 begin
-    gsay_message(672, 188, 51);
+    gsay_message(NAME, 188, BAD_REACTION);
     call jeremcbt();
 end
 
 procedure jerem29
 begin
-    float_msg(self_obj, message_str(672, 189), 2);
+    float_msg(self_obj, mstr(189), FLOAT_MSG_RED);
     call jeremcbt();
 end
 
@@ -775,9 +775,8 @@ end
 
 procedure jerem16
 begin
-    gsay_reply(672, 146);
-    giq_option(-3, 672, 147, jeremret, 50);
-    giq_option(-3, 672, 148, jerem19, 50);
-    giq_option(-3, 672, 149, jeremcbt, 51);
+    Reply(146);
+    giq_option(-3, NAME, 147, jeremret, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 148, jerem19, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 149, jeremcbt, BAD_REACTION);
 end
-
