diff --git a/Games/Fallout1.Steam/Original/JNKSCOUT.ssl b/Games/Fallout1.Steam/ReDefine/JNKSCOUT.ssl
index 6f25a07..c5349ed 100644
--- a/Games/Fallout1.Steam/Original/JNKSCOUT.ssl
+++ b/Games/Fallout1.Steam/ReDefine/JNKSCOUT.ssl
@@ -17,23 +17,23 @@ variable hostile;
 
 procedure start
 begin
-    if (script_action == 12) then begin
+    if (script_action == critter_proc) then begin
         call critter_p_proc();
     end
     else begin
-        if (script_action == 18) then begin
+        if (script_action == destroy_proc) then begin
             call destroy_p_proc();
         end
         else begin
-            if (script_action == 21) then begin
+            if (script_action == look_at_proc) then begin
                 call look_at_p_proc();
             end
             else begin
-                if (script_action == 4) then begin
+                if (script_action == pickup_proc) then begin
                     call pickup_p_proc();
                 end
                 else begin
-                    if (script_action == 11) then begin
+                    if (script_action == talk_proc) then begin
                         call talk_p_proc();
                     end
                 end
@@ -45,11 +45,11 @@ end
 procedure critter_p_proc
 begin
     if (hostile) then begin
-        attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+        attack(dude_obj);
     end
-    if (global_var(247) == 1) then begin
+    if (global_var(GVAR_ENEMY_JUNKTOWN) == 1) then begin
         if (obj_can_see_obj(self_obj, dude_obj)) then begin
-            attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+            attack(dude_obj);
         end
     end
 end
@@ -57,26 +57,26 @@ end
 procedure damage_p_proc
 begin
     if (source_obj == dude_obj) then begin
-        set_global_var(247, 1);
+        set_global_var(GVAR_ENEMY_JUNKTOWN, 1);
     end
 end
 
 procedure destroy_p_proc
 begin
     if (source_obj == dude_obj) then begin
-        set_global_var(247, 1);
+        set_global_var(GVAR_ENEMY_JUNKTOWN, 1);
         if (source_obj == dude_obj) then begin
-            if (((global_var(160) + global_var(159)) >= 25) and ((global_var(159) > (2 * global_var(160))) or (global_var(156) == 1))) then begin
-                set_global_var(156, 1);
-                set_global_var(157, 0);
+            if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
+                set_global_var(GVAR_BERSERKER_REPUTATION, 1);
+                set_global_var(GVAR_CHAMPION_REPUTATION, 0);
             end
-            if (((global_var(160) + global_var(159)) >= 25) and ((global_var(160) > (3 * global_var(159))) or (global_var(157) == 1))) then begin
-                set_global_var(157, 1);
-                set_global_var(156, 0);
+            if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(160) > (3 * global_var(GVAR_GOOD_MONSTER))) or (global_var(GVAR_CHAMPION_REPUTATION) == 1))) then begin
+                set_global_var(GVAR_CHAMPION_REPUTATION, 1);
+                set_global_var(GVAR_BERSERKER_REPUTATION, 0);
             end
-            set_global_var(159, global_var(159) + 1);
-            if ((global_var(159) % 2) == 0) then begin
-                set_global_var(155, global_var(155) - 1);
+            set_global_var(GVAR_GOOD_MONSTER, global_var(GVAR_GOOD_MONSTER) + 1);
+            if ((global_var(GVAR_GOOD_MONSTER) % 2) == 0) then begin
+                set_global_var(GVAR_PLAYER_REPUATION, global_var(GVAR_PLAYER_REPUATION) - 1);
             end
         end
     end
@@ -84,7 +84,7 @@ end
 
 procedure look_at_p_proc
 begin
-    display_msg(message_str(437, 100));
+    display_mstr(100);
 end
 
 procedure pickup_p_proc
@@ -103,38 +103,37 @@ end
 
 procedure Scout01
 begin
-    gsay_reply(437, 101);
+    Reply(101);
     call Scout02();
 end
 
 procedure Scout02
 begin
-    giq_option(4, 437, 102, Scout03, 50);
-    giq_option(4, 437, 103, Scout04, 50);
-    giq_option(4, 437, 104, Scout05, 50);
-    giq_option(4, 437, 105, ScoutEnd, 50);
-    giq_option(-3, 437, 106, ScoutEnd, 50);
+    giq_option(4, NAME, 102, Scout03, NEUTRAL_REACTION);
+    giq_option(4, NAME, 103, Scout04, NEUTRAL_REACTION);
+    giq_option(4, NAME, 104, Scout05, NEUTRAL_REACTION);
+    giq_option(4, NAME, 105, ScoutEnd, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 106, ScoutEnd, NEUTRAL_REACTION);
 end
 
 procedure Scout03
 begin
-    gsay_reply(437, 107);
+    Reply(107);
     call Scout02();
 end
 
 procedure Scout04
 begin
-    gsay_reply(437, 108);
+    Reply(108);
     call Scout02();
 end
 
 procedure Scout05
 begin
-    gsay_reply(437, 109);
+    Reply(109);
     call Scout02();
 end
 
 procedure ScoutEnd
 begin
 end
-
