diff --git a/Games/Fallout1.Steam/Original/JTUP2DN.ssl b/Games/Fallout1.Steam/ReDefine/JTUP2DN.ssl
index 2ff3759..fbb6203 100644
--- a/Games/Fallout1.Steam/Original/JTUP2DN.ssl
+++ b/Games/Fallout1.Steam/ReDefine/JTUP2DN.ssl
@@ -13,7 +13,7 @@ begin
         initialized := 1;
     end
     else begin
-        if (script_action == 6) then begin
+        if (script_action == use_proc) then begin
             call use_p_proc();
         end
     end
@@ -26,4 +26,3 @@ begin
         critter_stop_attacking(source_obj);
     end
 end
-
