diff --git a/Games/Fallout1.Steam/Original/JUNKJAIL.ssl b/Games/Fallout1.Steam/ReDefine/JUNKJAIL.ssl
index 8916b7c..f0ae249 100644
--- a/Games/Fallout1.Steam/Original/JUNKJAIL.ssl
+++ b/Games/Fallout1.Steam/ReDefine/JUNKJAIL.ssl
@@ -17,27 +17,27 @@ procedure use_p_proc;
 
 procedure start
 begin
-    if (script_action == 14) then begin
+    if (script_action == damage_proc) then begin
         call damage_p_proc();
     end
     else begin
-        if (script_action == 3) then begin
+        if (script_action == description_proc) then begin
             call description_p_proc();
         end
         else begin
-            if (script_action == 15) then begin
+            if (script_action == map_enter_proc) then begin
                 call map_enter_p_proc();
             end
             else begin
-                if (script_action == 2) then begin
+                if (script_action == spatial_proc) then begin
                     call spatial_p_proc();
                 end
                 else begin
-                    if (script_action == 7) then begin
+                    if (script_action == use_obj_on_proc) then begin
                         call use_obj_on_p_proc();
                     end
                     else begin
-                        if (script_action == 8) then begin
+                        if (script_action == use_skill_on_proc) then begin
                             call use_skill_on_p_proc();
                         end
                     end
@@ -55,7 +55,7 @@ end
 procedure description_p_proc
 begin
     script_overrides;
-    display_msg(message_str(789, 101));
+    display_mstr(101);
 end
 
 procedure map_enter_p_proc
@@ -70,14 +70,14 @@ end
 procedure spatial_p_proc
 begin
     if ((source_obj == dude_obj) and (triggered == 0) and (map_var(1) == 1)) then begin
-        display_msg(message_str(789, 100));
+        display_mstr(100);
         set_map_var(1, 0);
-        set_global_var(155, global_var(155) - 3);
-        display_msg(message_str(789, 107));
+        set_global_var(GVAR_PLAYER_REPUATION, global_var(GVAR_PLAYER_REPUATION) - 3);
+        display_mstr(107);
         give_exp_points(250);
-        if (global_var(247) == 0) then begin
-            set_global_var(247, 1);
-            set_global_var(155, global_var(155) - 5);
+        if (global_var(GVAR_ENEMY_JUNKTOWN) == 0) then begin
+            set_global_var(GVAR_ENEMY_JUNKTOWN, 1);
+            set_global_var(GVAR_PLAYER_REPUATION, global_var(GVAR_PLAYER_REPUATION) - 5);
         end
         triggered := 1;
         destroy_object(self_obj);
@@ -107,19 +107,19 @@ begin
     if (is_success(skill_test)) then begin
         if (obj_is_locked(self_obj)) then begin
             obj_unlock(self_obj);
-            display_msg(message_str(789, 103));
+            display_mstr(103);
         end
         else begin
             obj_lock(self_obj);
-            display_msg(message_str(789, 104));
+            display_mstr(104);
         end
     end
     else begin
         if (obj_is_locked(self_obj)) then begin
-            display_msg(message_str(789, 105));
+            display_mstr(105);
         end
         else begin
-            display_msg(message_str(789, 106));
+            display_mstr(106);
         end
     end
 end
@@ -128,7 +128,6 @@ procedure use_p_proc
 begin
     if (obj_is_locked(self_obj)) then begin
         script_overrides;
-        display_msg(message_str(789, 102));
+        display_mstr(102);
     end
 end
-
