diff --git a/Games/Fallout1.Steam/Original/JUSTIN.ssl b/Games/Fallout1.Steam/ReDefine/JUSTIN.ssl
index 9a9fb97..60708b0 100644
--- a/Games/Fallout1.Steam/Original/JUSTIN.ssl
+++ b/Games/Fallout1.Steam/ReDefine/JUSTIN.ssl
@@ -82,33 +82,33 @@ begin
     if (Only_Once) then begin
         Only_Once := 0;
         Justin_Ptr := self_obj;
-        if (global_var(221) == 1) then begin
+        if (global_var(GVAR_GREENE_DEAD) == 1) then begin
             set_obj_visibility(self_obj, 0);
         end
-        critter_add_trait(self_obj, 1, 6, 40);
-        critter_add_trait(self_obj, 1, 5, 86);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 40);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_HUB_POLICE);
     end
-    if (script_action == 21) then begin
+    if (script_action == look_at_proc) then begin
         call look_at_p_proc();
     end
     else begin
-        if (script_action == 4) then begin
+        if (script_action == pickup_proc) then begin
             call pickup_p_proc();
         end
         else begin
-            if (script_action == 11) then begin
+            if (script_action == talk_proc) then begin
                 call talk_p_proc();
             end
             else begin
-                if (script_action == 12) then begin
+                if (script_action == critter_proc) then begin
                     call critter_p_proc();
                 end
                 else begin
-                    if (script_action == 18) then begin
+                    if (script_action == destroy_proc) then begin
                         call destroy_p_proc();
                     end
                     else begin
-                        if (script_action == 13) then begin
+                        if (script_action == combat_proc) then begin
                             call combat_p_proc();
                         end
                     end
@@ -127,7 +127,7 @@ procedure critter_p_proc
 begin
     if (hostile) then begin
         hostile := 0;
-        attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+        attack(dude_obj);
     end
     if (travel == 1) then begin
         travel := 0;
@@ -162,7 +162,7 @@ begin
             end_dialogue;
         end
         else begin
-            if (global_var(203) == 1) then begin
+            if (global_var(GVAR_DECKER_STATUS) == 1) then begin
                 start_gdialog(696, self_obj, 4, -1, -1);
                 gsay_start;
                 call Justin27();
@@ -195,7 +195,7 @@ begin
                             end_dialogue;
                         end
                         else begin
-                            if (global_var(158) > 2) then begin
+                            if (global_var(GVAR_CHILDKILLER_REPUATION) > 2) then begin
                                 call Justin03();
                             end
                             else begin
@@ -236,26 +236,26 @@ end
 procedure destroy_p_proc
 begin
     if (source_obj == dude_obj) then begin
-        if (((global_var(160) + global_var(159)) >= 25) and ((global_var(159) > (2 * global_var(160))) or (global_var(156) == 1))) then begin
-            set_global_var(156, 1);
-            set_global_var(157, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
+            set_global_var(GVAR_BERSERKER_REPUTATION, 1);
+            set_global_var(GVAR_CHAMPION_REPUTATION, 0);
         end
-        if (((global_var(160) + global_var(159)) >= 25) and ((global_var(160) > (3 * global_var(159))) or (global_var(157) == 1))) then begin
-            set_global_var(157, 1);
-            set_global_var(156, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(160) > (3 * global_var(GVAR_GOOD_MONSTER))) or (global_var(GVAR_CHAMPION_REPUTATION) == 1))) then begin
+            set_global_var(GVAR_CHAMPION_REPUTATION, 1);
+            set_global_var(GVAR_BERSERKER_REPUTATION, 0);
         end
-        set_global_var(159, global_var(159) + 1);
-        if ((global_var(159) % 2) == 0) then begin
-            set_global_var(155, global_var(155) - 1);
+        set_global_var(GVAR_GOOD_MONSTER, global_var(GVAR_GOOD_MONSTER) + 1);
+        if ((global_var(GVAR_GOOD_MONSTER) % 2) == 0) then begin
+            set_global_var(GVAR_PLAYER_REPUATION, global_var(GVAR_PLAYER_REPUATION) - 1);
         end
     end
-    set_global_var(221, 1);
+    set_global_var(GVAR_GREENE_DEAD, 1);
 end
 
 procedure look_at_p_proc
 begin
     script_overrides;
-    display_msg(message_str(696, 100));
+    display_mstr(100);
 end
 
 procedure combat_p_proc
@@ -270,21 +270,21 @@ begin
     variable LVar0 := 0;
     LVar0 := obj_pid(source_obj);
     if ((party_member_obj(LVar0) != 0) and (map_var(52) == 0)) then begin
-        set_global_var(248, 1);
+        set_global_var(GVAR_ENEMY_HUB, 1);
     end
 end
 
 procedure Justin00
 begin
-    if (global_var(158) > 2) then begin
-        gsay_message(696, 101, 50);
+    if (global_var(GVAR_CHILDKILLER_REPUATION) > 2) then begin
+        gsay_message(NAME, 101, NEUTRAL_REACTION);
     end
     else begin
-        if (get_critter_stat(dude_obj, 34) == 0) then begin
-            gsay_reply(696, 102);
+        if (dude_gender == 0) then begin
+            Reply(102);
         end
         else begin
-            gsay_reply(696, 103);
+            Reply(103);
         end
         call Justin30();
     end
@@ -292,20 +292,20 @@ end
 
 procedure Justin01
 begin
-    gsay_reply(696, 104);
-    if ((global_var(203) == 0) and ((map_var(46) > 0) and (map_var(46) != 6))) then begin
-        giq_option(4, 696, 105, Justin01a, 50);
+    Reply(104);
+    if ((global_var(GVAR_DECKER_STATUS) == 0) and ((map_var(46) > 0) and (map_var(46) != 6))) then begin
+        giq_option(4, NAME, 105, Justin01a, NEUTRAL_REACTION);
     end
-    if ((global_var(203) == 0) and ((map_var(47) > 0) and (map_var(47) != 7))) then begin
-        giq_option(4, 696, 106, Justin01b, 50);
+    if ((global_var(GVAR_DECKER_STATUS) == 0) and ((map_var(47) > 0) and (map_var(47) != 7))) then begin
+        giq_option(4, NAME, 106, Justin01b, NEUTRAL_REACTION);
     end
-    giq_option(4, 696, 107, JustinEnd, 50);
-    giq_option(4, 696, 108, Justin04, 50);
+    giq_option(4, NAME, 107, JustinEnd, NEUTRAL_REACTION);
+    giq_option(4, NAME, 108, Justin04, NEUTRAL_REACTION);
 end
 
 procedure Justin01a
 begin
-    if (global_var(111) != 2) then begin
+    if (global_var(GVAR_KILL_MERCHANT) != 2) then begin
         call Justin05a();
     end
     else begin
@@ -320,7 +320,7 @@ end
 
 procedure Justin01b
 begin
-    if (global_var(112) != 2) then begin
+    if (global_var(GVAR_KILL_JAIN) != 2) then begin
         call Justin05a();
     end
     else begin
@@ -338,65 +338,65 @@ begin
     if (local_var(5) == 0) then begin
         call UpReact();
         set_local_var(5, 1);
-        gsay_message(696, 109, 49);
+        gsay_message(NAME, 109, GOOD_REACTION);
     end
     else begin
-        gsay_message(696, 110, 49);
+        gsay_message(NAME, 110, GOOD_REACTION);
     end
 end
 
 procedure Justin03
 begin
-    float_msg(self_obj, message_str(696, 111), 2);
+    float_msg(self_obj, mstr(111), FLOAT_MSG_RED);
 end
 
 procedure Justin04
 begin
-    gsay_message(696, 112, 50);
+    gsay_message(NAME, 112, NEUTRAL_REACTION);
 end
 
 procedure Justin05a
 begin
-    gsay_reply(696, 170);
-    giq_option(4, 696, 171, Justin05b, 50);
+    Reply(170);
+    giq_option(4, NAME, 171, Justin05b, NEUTRAL_REACTION);
 end
 
 procedure Justin05b
 begin
-    gsay_reply(696, 113);
+    Reply(113);
     call Justin09();
 end
 
 procedure Justin06
 begin
-    gsay_reply(696, 114);
+    Reply(114);
     call Justin09();
 end
 
 procedure Justin07
 begin
-    gsay_reply(696, 115);
+    Reply(115);
     call Justin15();
 end
 
 procedure Justin08
 begin
-    gsay_reply(696, 116);
+    Reply(116);
     call Justin15();
 end
 
 procedure Justin09
 begin
-    giq_option(4, 696, 117, Justin11, 50);
-    giq_option(4, 696, 118, Justin10, 50);
-    giq_option(4, 696, 119, Justin12, 50);
+    giq_option(4, NAME, 117, Justin11, NEUTRAL_REACTION);
+    giq_option(4, NAME, 118, Justin10, NEUTRAL_REACTION);
+    giq_option(4, NAME, 119, Justin12, NEUTRAL_REACTION);
 end
 
 procedure Justin10
 begin
-    gsay_reply(696, 120);
-    giq_option(4, 696, 121, Justin11, 50);
-    giq_option(4, 696, 122, Justin12, 50);
+    Reply(120);
+    giq_option(4, NAME, 121, Justin11, NEUTRAL_REACTION);
+    giq_option(4, NAME, 122, Justin12, NEUTRAL_REACTION);
 end
 
 procedure Justin11
@@ -404,19 +404,19 @@ begin
     variable LVar0 := 0;
     set_local_var(6, 1);
     LVar0 := item_caps_adjust(dude_obj, 300);
-    gsay_message(696, 123, 50);
-    gsay_reply(696, 174);
-    giq_option(4, 696, 124, Justin20, 50);
-    giq_option(4, 696, 125, JustinEnd, 50);
+    gsay_message(NAME, 123, NEUTRAL_REACTION);
+    Reply(174);
+    giq_option(4, NAME, 124, Justin20, NEUTRAL_REACTION);
+    giq_option(4, NAME, 125, JustinEnd, NEUTRAL_REACTION);
 end
 
 procedure Justin12
 begin
     set_local_var(6, 2);
-    gsay_reply(696, 126);
-    giq_option(4, 696, 127, Justin13, 50);
-    giq_option(4, 696, 128, Justin14, 50);
-    giq_option(4, 696, 129, JustinEnd, 50);
+    Reply(126);
+    giq_option(4, NAME, 127, Justin13, NEUTRAL_REACTION);
+    giq_option(4, NAME, 128, Justin14, NEUTRAL_REACTION);
+    giq_option(4, NAME, 129, JustinEnd, NEUTRAL_REACTION);
 end
 
 procedure Justin13
@@ -424,22 +424,22 @@ begin
     variable LVar0 := 0;
     set_local_var(6, 1);
     LVar0 := item_caps_adjust(dude_obj, 300);
-    gsay_message(696, 130, 49);
-    gsay_message(696, 175, 50);
+    gsay_message(NAME, 130, GOOD_REACTION);
+    gsay_message(NAME, 175, NEUTRAL_REACTION);
 end
 
 procedure Justin14
 begin
-    gsay_message(696, 131, 50);
+    gsay_message(NAME, 131, NEUTRAL_REACTION);
 end
 
 procedure Justin15
 begin
-    giq_option(4, 696, 132, JustinCombat, 50);
-    giq_option(4, 696, 133, Justin15a, 50);
-    giq_option(4, 696, 134, Justin15b, 50);
-    giq_option(4, 696, 135, Justin15c, 50);
-    giq_option(4, 696, 136, Justin15d, 50);
+    giq_option(4, NAME, 132, JustinCombat, NEUTRAL_REACTION);
+    giq_option(4, NAME, 133, Justin15a, NEUTRAL_REACTION);
+    giq_option(4, NAME, 134, Justin15b, NEUTRAL_REACTION);
+    giq_option(4, NAME, 135, Justin15c, NEUTRAL_REACTION);
+    giq_option(4, NAME, 136, Justin15d, NEUTRAL_REACTION);
 end
 
 procedure Justin15a
@@ -484,105 +484,105 @@ end
 
 procedure Justin16
 begin
-    gsay_reply(696, 137);
+    Reply(137);
     call Justin09();
 end
 
 procedure Justin17
 begin
     set_local_var(7, 1);
-    gsay_message(696, 138, 51);
+    gsay_message(NAME, 138, BAD_REACTION);
 end
 
 procedure Justin18
 begin
-    gsay_reply(696, 139);
+    Reply(139);
     call Justin09();
 end
 
 procedure Justin19
 begin
     set_local_var(7, 1);
-    gsay_message(696, 140, 51);
+    gsay_message(NAME, 140, BAD_REACTION);
 end
 
 procedure Justin20
 begin
-    gsay_message(696, 141, 50);
+    gsay_message(NAME, 141, NEUTRAL_REACTION);
     call JustinTravel();
 end
 
 procedure Justin26
 begin
-    gsay_reply(696, 151);
-    giq_option(4, 696, 152, JustinTravel, 50);
-    giq_option(4, 696, 153, JustinEnd, 50);
-    giq_option(-3, 696, 154, Justin34, 50);
+    Reply(151);
+    giq_option(4, NAME, 152, JustinTravel, NEUTRAL_REACTION);
+    giq_option(4, NAME, 153, JustinEnd, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 154, Justin34, NEUTRAL_REACTION);
 end
 
 procedure Justin27
 begin
-    gsay_message(696, 155, 49);
+    gsay_message(NAME, 155, GOOD_REACTION);
 end
 
 procedure Justin28
 begin
-    gsay_reply(696, 156);
-    giq_option(4, 696, 157, Justin11, 50);
-    giq_option(4, 696, 158, JustinEnd, 50);
-    giq_option(-3, 696, 159, Justin34, 50);
+    Reply(156);
+    giq_option(4, NAME, 157, Justin11, NEUTRAL_REACTION);
+    giq_option(4, NAME, 158, JustinEnd, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 159, Justin34, NEUTRAL_REACTION);
 end
 
 procedure Justin29
 begin
-    gsay_message(696, 160, 50);
+    gsay_message(NAME, 160, NEUTRAL_REACTION);
 end
 
 procedure Justin30
 begin
-    giq_option(4, 696, 161, Justin01, 50);
-    giq_option(4, 696, 162, Justin02, 50);
-    giq_option(4, 696, 163, JustinEnd, 50);
-    giq_option(-3, 696, 164, Justin34, 50);
+    giq_option(4, NAME, 161, Justin01, NEUTRAL_REACTION);
+    giq_option(4, NAME, 162, Justin02, NEUTRAL_REACTION);
+    giq_option(4, NAME, 163, JustinEnd, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 164, Justin34, NEUTRAL_REACTION);
 end
 
 procedure Justin31
 begin
-    if (get_critter_stat(dude_obj, 34) == 0) then begin
-        gsay_reply(696, 165);
+    if (dude_gender == 0) then begin
+        Reply(165);
     end
     else begin
-        gsay_reply(696, 166);
+        Reply(166);
     end
     call Justin30();
 end
 
 procedure Justin32
 begin
-    gsay_reply(696, 167);
+    Reply(167);
     call Justin30();
 end
 
 procedure Justin33
 begin
-    gsay_reply(696, 168);
+    Reply(168);
     call Justin30();
 end
 
 procedure Justin34
 begin
-    gsay_message(696, 169, 50);
+    gsay_message(NAME, 169, NEUTRAL_REACTION);
 end
 
 procedure Justin35
 begin
     variable LVar0 := 0;
     LVar0 := item_caps_adjust(dude_obj, 1000);
-    set_global_var(155, global_var(155) + 4);
+    set_global_var(GVAR_PLAYER_REPUATION, global_var(GVAR_PLAYER_REPUATION) + 4);
     give_exp_points(1400);
-    display_msg(message_str(766, 103) + 1400 + message_str(766, 104));
-    gsay_message(696, 172, 49);
-    gsay_message(696, 173, 49);
+    display_msg(message_str(SCRIPT_GENCHAT, 103) + 1400 + message_str(SCRIPT_GENCHAT, 104));
+    gsay_message(NAME, 172, GOOD_REACTION);
+    gsay_message(NAME, 173, GOOD_REACTION);
 end
 
 procedure JustinEnd
@@ -606,11 +606,11 @@ begin
     set_obj_visibility(Fry_Stub_Ptr, 0);
     move_to(self_obj, 23926, 1);
     anim(self_obj, 1000, rotation_to_tile(tile_num(self_obj), tile_num(Decker_Ptr)));
-    critter_add_trait(self_obj, 1, 6, 0);
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 0);
     move_to(Kane_Ptr, 22526, 1);
     anim(Kane_Ptr, 1000, rotation_to_tile(tile_num(Kane_Ptr), tile_num(Decker_Ptr)));
     move_to(dude_obj, 23924, 1);
-    anim(dude_obj, 1000, rotation_to_tile(tile_num(dude_obj), tile_num(Decker_Ptr)));
+    anim(dude_obj, 1000, rotation_to_tile(dude_tile, tile_num(Decker_Ptr)));
     gfade_in(1000);
 end
 
@@ -620,31 +620,31 @@ begin
         set_local_var(0, 50);
         set_local_var(1, 2);
         set_local_var(2, 1);
-        set_local_var(0, local_var(0) + (5 * get_critter_stat(dude_obj, 3)) - 25);
-        set_local_var(0, local_var(0) + (10 * has_trait(0, dude_obj, 10)));
-        if (has_trait(0, dude_obj, 39)) then begin
-            if (global_var(155) > 0) then begin
-                set_local_var(0, local_var(0) + global_var(155));
+        set_local_var(0, local_var(0) + (5 * dude_charisma) - 25);
+        set_local_var(0, local_var(0) + (10 * has_trait(TRAIT_PERK, dude_obj, PERK_presence)));
+        if (has_trait(TRAIT_PERK, dude_obj, PERK_cult_of_personality)) then begin
+            if (global_var(GVAR_PLAYER_REPUATION) > 0) then begin
+                set_local_var(0, local_var(0) + global_var(GVAR_PLAYER_REPUATION));
             end
             else begin
-                set_local_var(0, local_var(0) - global_var(155));
+                set_local_var(0, local_var(0) - global_var(GVAR_PLAYER_REPUATION));
             end
         end
         else begin
             if (local_var(3) == 1) then begin
-                set_local_var(0, local_var(0) - global_var(155));
+                set_local_var(0, local_var(0) - global_var(GVAR_PLAYER_REPUATION));
             end
             else begin
-                set_local_var(0, local_var(0) + global_var(155));
+                set_local_var(0, local_var(0) + global_var(GVAR_PLAYER_REPUATION));
             end
         end
-        if (global_var(158) > 2) then begin
+        if (global_var(GVAR_CHILDKILLER_REPUATION) > 2) then begin
             set_local_var(0, local_var(0) - 30);
         end
-        if (((global_var(160) + global_var(159)) >= 25) and ((global_var(160) > (3 * global_var(159))) or (global_var(157) == 1))) then begin
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(160) > (3 * global_var(GVAR_GOOD_MONSTER))) or (global_var(GVAR_CHAMPION_REPUTATION) == 1))) then begin
             set_local_var(0, local_var(0) + 20);
         end
-        if (((global_var(160) + global_var(159)) >= 25) and ((global_var(159) > (2 * global_var(160))) or (global_var(156) == 1))) then begin
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
             set_local_var(0, local_var(0) - 20);
         end
         call ReactToLevel();
@@ -737,10 +737,9 @@ end
 
 procedure Goodbyes
 begin
-    exit_line := message_str(634, random(100, 105));
+    exit_line := message_str(SCRIPT_MODREACT, random(100, 105));
 end
 
 procedure Justin22
 begin
 end
-
