diff --git a/Games/Fallout1.Steam/Original/KATJA.ssl b/Games/Fallout1.Steam/ReDefine/KATJA.ssl
index b3db241..3cb32e6 100644
--- a/Games/Fallout1.Steam/Original/KATJA.ssl
+++ b/Games/Fallout1.Steam/ReDefine/KATJA.ssl
@@ -66,27 +66,27 @@ import variable Tycho_ptr;
 
 procedure start
 begin
-    if (script_action == 12) then begin
+    if (script_action == critter_proc) then begin
         call critter_p_proc();
     end
     else begin
-        if (script_action == 18) then begin
+        if (script_action == destroy_proc) then begin
             call destroy_p_proc();
         end
         else begin
-            if (script_action == 15) then begin
+            if (script_action == map_enter_proc) then begin
                 call map_enter_p_proc();
             end
             else begin
-                if (script_action == 7) then begin
+                if (script_action == use_obj_on_proc) then begin
                     call use_obj_on_p_proc();
                 end
                 else begin
-                    if (script_action == 11) then begin
+                    if (script_action == talk_proc) then begin
                         call talk_p_proc();
                     end
                     else begin
-                        if (script_action == 22) then begin
+                        if (script_action == timed_event_proc) then begin
                             call timed_event_p_proc();
                         end
                     end
@@ -100,7 +100,7 @@ procedure critter_p_proc
 begin
     if (hostile) then begin
         hostile := 0;
-        attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+        attack(dude_obj);
     end
     else begin
         call map_commentary();
@@ -109,30 +109,30 @@ end
 
 procedure destroy_p_proc
 begin
-    set_global_var(244, 3);
+    set_global_var(GVAR_KATJA_STATUS, 3);
     if (source_obj == dude_obj) then begin
-        if (((global_var(160) + global_var(159)) >= 25) and ((global_var(159) > (2 * global_var(160))) or (global_var(156) == 1))) then begin
-            set_global_var(156, 1);
-            set_global_var(157, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
+            set_global_var(GVAR_BERSERKER_REPUTATION, 1);
+            set_global_var(GVAR_CHAMPION_REPUTATION, 0);
         end
-        if (((global_var(160) + global_var(159)) >= 25) and ((global_var(160) > (3 * global_var(159))) or (global_var(157) == 1))) then begin
-            set_global_var(157, 1);
-            set_global_var(156, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(160) > (3 * global_var(GVAR_GOOD_MONSTER))) or (global_var(GVAR_CHAMPION_REPUTATION) == 1))) then begin
+            set_global_var(GVAR_CHAMPION_REPUTATION, 1);
+            set_global_var(GVAR_BERSERKER_REPUTATION, 0);
         end
-        set_global_var(159, global_var(159) + 1);
-        if ((global_var(159) % 2) == 0) then begin
-            set_global_var(155, global_var(155) - 1);
+        set_global_var(GVAR_GOOD_MONSTER, global_var(GVAR_GOOD_MONSTER) + 1);
+        if ((global_var(GVAR_GOOD_MONSTER) % 2) == 0) then begin
+            set_global_var(GVAR_PLAYER_REPUATION, global_var(GVAR_PLAYER_REPUATION) - 1);
         end
     end
 end
 
 procedure map_enter_p_proc
 begin
-    if (global_var(244) < 2) then begin
-        critter_add_trait(self_obj, 1, 6, 49);
+    if (global_var(GVAR_KATJA_STATUS) < 2) then begin
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 49);
     end
     else begin
-        critter_add_trait(self_obj, 1, 6, 0);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 0);
         add_timer_event(self_obj, game_ticks(2), 1);
     end
     Katja_ptr := self_obj;
@@ -140,7 +140,7 @@ end
 
 procedure use_obj_on_p_proc
 begin
-    if (global_var(244) == 2) then begin
+    if (global_var(GVAR_KATJA_STATUS) == 2) then begin
         item := obj_pid(obj_being_used_with);
         if ((item != 40) and (item != 47) and (item != 91)) then begin
             script_overrides;
@@ -150,7 +150,7 @@ begin
                     add_obj_to_inven(self_obj, obj_being_used_with);
                 end
                 else begin
-                    float_msg(self_obj, message_str(634, 109), 5);
+                    float_msg(self_obj, message_str(SCRIPT_MODREACT, 109), FLOAT_MSG_PURPLE);
                 end
             end
             else begin
@@ -165,15 +165,15 @@ procedure talk_p_proc
 begin
     start_gdialog(623, self_obj, 4, -1, -1);
     gsay_start;
-    if (global_var(244) == 3) then begin
+    if (global_var(GVAR_KATJA_STATUS) == 3) then begin
         call Katja28();
     end
     else begin
-        if (global_var(244) == 2) then begin
+        if (global_var(GVAR_KATJA_STATUS) == 2) then begin
             call Katja24();
         end
         else begin
-            if (global_var(244) == 1) then begin
+            if (global_var(GVAR_KATJA_STATUS) == 1) then begin
                 call Katja25();
             end
             else begin
@@ -190,252 +190,252 @@ begin
     if (lock) then begin
         call pick_lock();
     end
-    if (global_var(244) == 2) then begin
-        if (global_var(315) == 0) then begin
-            set_global_var(315, 1);
+    if (global_var(GVAR_KATJA_STATUS) == 2) then begin
+        if (global_var(GVAR_DESTROY_MASTER_11) == 0) then begin
+            set_global_var(GVAR_DESTROY_MASTER_11, 1);
             give_exp_points(200);
-            display_msg(message_str(623, 180));
+            display_mstr(180);
         end
     end
 end
 
 procedure timed_event_p_proc
 begin
-    if (global_var(244) == 2) then begin
+    if (global_var(GVAR_KATJA_STATUS) == 2) then begin
         call follow_player();
     end
 end
 
 procedure Katja01
 begin
-    set_global_var(244, 1);
-    gsay_reply(623, 102);
-    giq_option(4, 623, message_str(623, 103) + proto_data(obj_pid(dude_obj), 1) + message_str(623, 104), Katja02, 50);
-    giq_option(4, 623, 105, Katja02, 50);
-    giq_option(5, 623, 106, Katja05, 50);
-    giq_option(-3, 623, 189, KatjaExit, 50);
+    set_global_var(GVAR_KATJA_STATUS, 1);
+    Reply(102);
+    giq_option(4, NAME, mstr(103) + proto_data(obj_pid(dude_obj), 1) + mstr(104), Katja02, NEUTRAL_REACTION);
+    giq_option(4, NAME, 105, Katja02, NEUTRAL_REACTION);
+    giq_option(5, NAME, 106, Katja05, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 189, KatjaExit, NEUTRAL_REACTION);
 end
 
 procedure Katja02
 begin
-    gsay_reply(623, 107);
-    giq_option(5, 623, 108, Katja03, 50);
-    giq_option(5, 623, 109, Katja04, 50);
-    giq_option(4, 623, 110, Katja07, 50);
-    if ((global_var(101) != 2) and (obj_is_carrying_obj_pid(dude_obj, 55) == 0)) then begin
-        giq_option(4, 623, 111, Katja06, 50);
+    Reply(107);
+    giq_option(5, NAME, 108, Katja03, NEUTRAL_REACTION);
+    giq_option(5, NAME, 109, Katja04, NEUTRAL_REACTION);
+    giq_option(4, NAME, 110, Katja07, NEUTRAL_REACTION);
+    if ((global_var(GVAR_FIND_WATER_CHIP) != 2) and (obj_is_carrying_obj_pid(dude_obj, 55) == 0)) then begin
+        giq_option(4, NAME, 111, Katja06, NEUTRAL_REACTION);
     end
 end
 
 procedure Katja03
 begin
-    gsay_reply(623, 112);
-    giq_option(5, 623, 109, Katja04, 50);
-    giq_option(4, 623, 110, Katja07, 50);
-    if ((global_var(101) != 2) and (obj_is_carrying_obj_pid(dude_obj, 55) == 0)) then begin
-        giq_option(4, 623, 111, Katja06, 50);
+    Reply(112);
+    giq_option(5, NAME, 109, Katja04, NEUTRAL_REACTION);
+    giq_option(4, NAME, 110, Katja07, NEUTRAL_REACTION);
+    if ((global_var(GVAR_FIND_WATER_CHIP) != 2) and (obj_is_carrying_obj_pid(dude_obj, 55) == 0)) then begin
+        giq_option(4, NAME, 111, Katja06, NEUTRAL_REACTION);
     end
 end
 
 procedure Katja04
 begin
-    gsay_reply(623, 113);
-    giq_option(5, 623, 114, Katja21, 50);
-    giq_option(4, 623, 116, Katja09, 50);
-    giq_option(4, 623, 115, KatjaEnd, 50);
+    Reply(113);
+    giq_option(5, NAME, 114, Katja21, NEUTRAL_REACTION);
+    giq_option(4, NAME, 116, Katja09, NEUTRAL_REACTION);
+    giq_option(4, NAME, 115, KatjaEnd, NEUTRAL_REACTION);
 end
 
 procedure Katja05
 begin
-    gsay_reply(623, 117);
-    giq_option(4, 623, message_str(623, 103) + proto_data(obj_pid(dude_obj), 1) + message_str(623, 104), Katja02, 50);
-    giq_option(4, 623, 118, Katja07, 50);
-    giq_option(4, 623, 115, KatjaEnd, 50);
+    Reply(117);
+    giq_option(4, NAME, mstr(103) + proto_data(obj_pid(dude_obj), 1) + mstr(104), Katja02, NEUTRAL_REACTION);
+    giq_option(4, NAME, 118, Katja07, NEUTRAL_REACTION);
+    giq_option(4, NAME, 115, KatjaEnd, NEUTRAL_REACTION);
 end
 
 procedure Katja06
 begin
-    gsay_reply(623, 119);
-    giq_option(5, 623, 121, Katja11, 50);
-    giq_option(4, 623, 122, Katja10, 50);
-    giq_option(4, 623, 120, KatjaEnd, 50);
+    Reply(119);
+    giq_option(5, NAME, 121, Katja11, NEUTRAL_REACTION);
+    giq_option(4, NAME, 122, Katja10, NEUTRAL_REACTION);
+    giq_option(4, NAME, 120, KatjaEnd, NEUTRAL_REACTION);
 end
 
 procedure Katja07
 begin
-    gsay_reply(623, 123);
-    giq_option(4, 623, 124, Katja06, 50);
-    giq_option(4, 623, 125, Katja08, 50);
+    Reply(123);
+    giq_option(4, NAME, 124, Katja06, NEUTRAL_REACTION);
+    giq_option(4, NAME, 125, Katja08, NEUTRAL_REACTION);
 end
 
 procedure Katja08
 begin
-    gsay_message(623, 126, 51);
+    gsay_message(NAME, 126, BAD_REACTION);
 end
 
 procedure Katja09
 begin
-    gsay_reply(623, 127);
-    if ((global_var(101) != 2) and (obj_is_carrying_obj_pid(dude_obj, 55) == 0)) then begin
-        giq_option(4, 623, 128, Katja06, 50);
+    Reply(127);
+    if ((global_var(GVAR_FIND_WATER_CHIP) != 2) and (obj_is_carrying_obj_pid(dude_obj, 55) == 0)) then begin
+        giq_option(4, NAME, 128, Katja06, NEUTRAL_REACTION);
     end
-    giq_option(4, 623, 129, Katja10, 50);
-    giq_option(4, 623, 130, Katja17, 50);
+    giq_option(4, NAME, 129, Katja10, NEUTRAL_REACTION);
+    giq_option(4, NAME, 130, Katja17, NEUTRAL_REACTION);
 end
 
 procedure Katja10
 begin
-    giq_option(4, 623, 131, Katja13, 50);
-    giq_option(4, 623, 133, Katja15, 50);
-    giq_option(4, 623, 134, Katja16, 50);
-    giq_option(4, 623, 135, Katja12, 50);
-    giq_option(4, 623, 130, Katja17, 50);
-    giq_option(4, 623, 136, KatjaEnd, 50);
+    giq_option(4, NAME, 131, Katja13, NEUTRAL_REACTION);
+    giq_option(4, NAME, 133, Katja15, NEUTRAL_REACTION);
+    giq_option(4, NAME, 134, Katja16, NEUTRAL_REACTION);
+    giq_option(4, NAME, 135, Katja12, NEUTRAL_REACTION);
+    giq_option(4, NAME, 130, Katja17, NEUTRAL_REACTION);
+    giq_option(4, NAME, 136, KatjaEnd, NEUTRAL_REACTION);
 end
 
 procedure Katja11
 begin
-    gsay_reply(623, 137);
-    giq_option(4, 623, 135, Katja12, 50);
-    giq_option(4, 623, 138, KatjaEnd, 50);
+    Reply(137);
+    giq_option(4, NAME, 135, Katja12, NEUTRAL_REACTION);
+    giq_option(4, NAME, 138, KatjaEnd, NEUTRAL_REACTION);
 end
 
 procedure Katja12
 begin
-    gsay_reply(623, 139);
+    Reply(139);
     call Katja10();
 end
 
 procedure Katja13
 begin
-    gsay_reply(623, 140);
+    Reply(140);
     call Katja10();
 end
 
 procedure Katja14
 begin
-    gsay_reply(623, 141);
+    Reply(141);
     call Katja10();
 end
 
 procedure Katja15
 begin
-    gsay_reply(623, 142);
+    Reply(142);
     call Katja10();
 end
 
 procedure Katja16
 begin
-    gsay_reply(623, 143);
+    Reply(143);
     call Katja10();
 end
 
 procedure Katja17
 begin
-    gsay_reply(623, 144);
-    if (not(global_var(77))) then begin
-        set_global_var(77, 1);
+    Reply(144);
+    if (not(global_var(GVAR_CHILDREN_OF_CATHEDRAL))) then begin
+        set_global_var(GVAR_CHILDREN_OF_CATHEDRAL, 1);
     end
-    giq_option(4, 623, 145, Katja18, 50);
-    giq_option(4, 623, 147, Katja10, 50);
-    giq_option(4, 623, 146, KatjaEnd, 50);
+    giq_option(4, NAME, 145, Katja18, NEUTRAL_REACTION);
+    giq_option(4, NAME, 147, Katja10, NEUTRAL_REACTION);
+    giq_option(4, NAME, 146, KatjaEnd, NEUTRAL_REACTION);
 end
 
 procedure Katja18
 begin
-    gsay_reply(623, 148);
-    giq_option(4, 623, 149, Katja20, 50);
-    giq_option(4, 623, 151, Katja10, 50);
-    giq_option(4, 623, 150, KatjaEnd, 50);
+    Reply(148);
+    giq_option(4, NAME, 149, Katja20, NEUTRAL_REACTION);
+    giq_option(4, NAME, 151, Katja10, NEUTRAL_REACTION);
+    giq_option(4, NAME, 150, KatjaEnd, NEUTRAL_REACTION);
 end
 
 procedure Katja19
 begin
-    gsay_reply(623, 152);
+    Reply(152);
     call Katja10();
 end
 
 procedure Katja20
 begin
-    set_global_var(244, 2);
+    set_global_var(GVAR_KATJA_STATUS, 2);
     party_add(self_obj);
     add_timer_event(self_obj, game_ticks(1), 1);
-    critter_add_trait(self_obj, 1, 6, 0);
-    gsay_message(623, 153, 50);
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 0);
+    gsay_message(NAME, 153, NEUTRAL_REACTION);
 end
 
 procedure Katja21
 begin
-    gsay_reply(623, 154);
-    giq_option(4, 623, 156, Katja22, 50);
-    giq_option(6, 623, 157, Katja23, 50);
-    giq_option(4, 623, 155, KatjaEnd, 50);
+    Reply(154);
+    giq_option(4, NAME, 156, Katja22, NEUTRAL_REACTION);
+    giq_option(6, NAME, 157, Katja23, NEUTRAL_REACTION);
+    giq_option(4, NAME, 155, KatjaEnd, NEUTRAL_REACTION);
 end
 
 procedure Katja22
 begin
-    gsay_reply(623, 158);
+    Reply(158);
     call Katja10();
 end
 
 procedure Katja23
 begin
-    gsay_reply(623, 159);
-    giq_option(4, 623, 160, Katja09, 50);
-    if ((global_var(101) == 0) and (obj_is_carrying_obj_pid(dude_obj, 55) == 0)) then begin
-        giq_option(4, 623, 161, Katja06, 50);
+    Reply(159);
+    giq_option(4, NAME, 160, Katja09, NEUTRAL_REACTION);
+    if ((global_var(GVAR_FIND_WATER_CHIP) == 0) and (obj_is_carrying_obj_pid(dude_obj, 55) == 0)) then begin
+        giq_option(4, NAME, 161, Katja06, NEUTRAL_REACTION);
     end
-    giq_option(4, 623, 162, KatjaEnd, 50);
+    giq_option(4, NAME, 162, KatjaEnd, NEUTRAL_REACTION);
 end
 
 procedure Katja24
 begin
-    gsay_reply(623, 163);
+    Reply(163);
     if (cur_map_index == 30) then begin
         if (map_var(3)) then begin
-            giq_option(4, 623, 164, set_lock, 50);
+            giq_option(4, NAME, 164, set_lock, NEUTRAL_REACTION);
         end
     end
-    if ((cur_map_index == 10) and (elevation(dude_obj) == 1)) then begin
+    if ((cur_map_index == 10) and (dude_elevation == 1)) then begin
         if (map_var(4)) then begin
-            giq_option(4, 623, 164, set_lock, 50);
+            giq_option(4, NAME, 164, set_lock, NEUTRAL_REACTION);
         end
     end
-    giq_option(4, 389, 159, KatjaTactics, 49);
-    giq_option(4, 623, 177, Katja26, 50);
-    giq_option(4, 623, 174, Katja27, 50);
-    giq_option(4, 623, 167, KatjaLeave, 50);
-    giq_option(4, 623, 168, KatjaEnd, 50);
-    giq_option(-3, 623, 168, KatjaEnd, 50);
-    giq_option(-3, 623, 190, KatjaLeave, 50);
+    giq_option(4, SCRIPT_TYCHO, 159, KatjaTactics, GOOD_REACTION);
+    giq_option(4, NAME, 177, Katja26, NEUTRAL_REACTION);
+    giq_option(4, NAME, 174, Katja27, NEUTRAL_REACTION);
+    giq_option(4, NAME, 167, KatjaLeave, NEUTRAL_REACTION);
+    giq_option(4, NAME, 168, KatjaEnd, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 168, KatjaEnd, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 190, KatjaLeave, NEUTRAL_REACTION);
 end
 
 procedure Katja25
 begin
-    gsay_reply(623, 169);
-    giq_option(4, 623, 171, Katja10, 50);
-    giq_option(4, 623, 170, KatjaEnd, 50);
-    giq_option(-3, 623, 189, KatjaExit, 50);
+    Reply(169);
+    giq_option(4, NAME, 171, Katja10, NEUTRAL_REACTION);
+    giq_option(4, NAME, 170, KatjaEnd, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 189, KatjaExit, NEUTRAL_REACTION);
 end
 
 procedure Katja26
 begin
-    gsay_message(623, 178, 50);
+    gsay_message(NAME, 178, NEUTRAL_REACTION);
 end
 
 procedure Katja27
 begin
     unwield_flag := 1;
-    gsay_message(623, 179, 50);
+    gsay_message(NAME, 179, NEUTRAL_REACTION);
     call Katja24();
 end
 
 procedure Katja28
 begin
-    gsay_reply(623, 181);
-    giq_option(4, 623, 182, Katja20, 50);
-    giq_option(4, 623, 183, KatjaEnd, 50);
-    giq_option(-3, 623, 189, KatjaExit, 50);
+    Reply(181);
+    giq_option(4, NAME, 182, Katja20, NEUTRAL_REACTION);
+    giq_option(4, NAME, 183, KatjaEnd, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 189, KatjaExit, NEUTRAL_REACTION);
 end
 
 procedure KatjaEnd
@@ -444,47 +444,47 @@ end
 
 procedure KatjaLeave
 begin
-    set_global_var(244, 3);
+    set_global_var(GVAR_KATJA_STATUS, 3);
     party_remove(self_obj);
     rm_timer_event(self_obj);
 end
 
 procedure KatjaTactics
 begin
-    gsay_reply(389, 160);
-    giq_option(4, 389, 161, KatjaClose, 49);
-    giq_option(4, 389, 162, KatjaModerate, 49);
-    giq_option(4, 389, 163, KatjaFar, 49);
+    gsay_reply(SCRIPT_TYCHO, 160);
+    giq_option(4, SCRIPT_TYCHO, 161, KatjaClose, GOOD_REACTION);
+    giq_option(4, SCRIPT_TYCHO, 162, KatjaModerate, GOOD_REACTION);
+    giq_option(4, SCRIPT_TYCHO, 163, KatjaFar, GOOD_REACTION);
 end
 
 procedure KatjaClose
 begin
-    set_global_var(279, 0);
+    set_global_var(GVAR_DESTROY_VATS_7, 0);
 end
 
 procedure KatjaModerate
 begin
-    set_global_var(279, 1);
+    set_global_var(GVAR_DESTROY_VATS_7, 1);
 end
 
 procedure KatjaFar
 begin
-    set_global_var(279, 2);
+    set_global_var(GVAR_DESTROY_VATS_7, 2);
 end
 
 procedure KatjaExit
 begin
-    gsay_message(623, 191, 50);
+    gsay_message(NAME, 191, NEUTRAL_REACTION);
 end
 
 procedure follow_player
 begin
     variable LVar0 := 0;
     prev_tile := dest_tile;
-    LVar0 := (has_trait(1, dude_obj, 10) + random(2, 4)) % 6;
-    dest_tile := tile_num_in_direction(tile_num(dude_obj), LVar0, (global_var(279) * 2) + random(1, 2));
-    if ((tile_distance_objs(self_obj, dude_obj) > ((global_var(279) + 1) * 2)) or (random(0, 3) == 3)) then begin
-        if (tile_distance(prev_tile, tile_num(dude_obj)) > tile_distance(dest_tile, tile_num(dude_obj))) then begin
+    LVar0 := (obj_get_rot(dude_obj) + random(2, 4)) % 6;
+    dest_tile := tile_num_in_direction(dude_tile, LVar0, (global_var(GVAR_DESTROY_VATS_7) * 2) + random(1, 2));
+    if ((tile_distance_objs(self_obj, dude_obj) > ((global_var(GVAR_DESTROY_VATS_7) + 1) * 2)) or (random(0, 3) == 3)) then begin
+        if (tile_distance(prev_tile, dude_tile) > tile_distance(dest_tile, dude_tile)) then begin
             if (tile_distance_objs(self_obj, dude_obj) > 8) then begin
                 animate_move_obj_to_tile(self_obj, dest_tile, 1 bwor 16);
             end
@@ -508,7 +508,7 @@ procedure map_commentary
 begin
     if (cur_map_index == 33) then begin
         if ((tile_distance(tile_num(self_obj), 27317) < 4) and (line173flag == 0)) then begin
-            float_msg(self_obj, message_str(623, 173), 5);
+            float_msg(self_obj, mstr(173), FLOAT_MSG_PURPLE);
             line173flag := 1;
         end
     end
@@ -516,8 +516,8 @@ begin
         if (cur_map_index == 6) then begin
             if (elevation(self_obj) == 1) then begin
                 if (tile_distance(tile_num(self_obj), 19111) < 4) then begin
-                    if ((line174flag == 0) and (global_var(238) != 2) and (Ian_ptr != 0)) then begin
-                        float_msg(self_obj, message_str(623, 172), 5);
+                    if ((line174flag == 0) and (global_var(GVAR_CALM_REBELS) != 2) and (Ian_ptr != 0)) then begin
+                        float_msg(self_obj, mstr(172), FLOAT_MSG_PURPLE);
                         line174flag := 1;
                     end
                 end
@@ -526,9 +526,9 @@ begin
         else begin
             if (cur_map_index == 38) then begin
                 if ((tile_distance(tile_num(self_obj), 28104) < 5) and (line176flag == 0)) then begin
-                    float_msg(self_obj, message_str(623, 176), 5);
-                    if (global_var(121) == 2) then begin
-                        float_msg(Tycho_ptr, message_str(389, 167), 12);
+                    float_msg(self_obj, mstr(176), FLOAT_MSG_PURPLE);
+                    if (global_var(GVAR_HERNANDEZ_STATUS) == 2) then begin
+                        float_msg(Tycho_ptr, message_str(SCRIPT_TYCHO, 167), FLOAT_MSG_LIGHT_GREY);
                     end
                     line176flag := 1;
                 end
@@ -536,9 +536,9 @@ begin
             else begin
                 if (cur_map_index == 32) then begin
                     if (line184flag == 0) then begin
-                        float_msg(self_obj, message_str(623, 184), 5);
-                        if (global_var(118) == 2) then begin
-                            float_msg(Ian_ptr, message_str(235, 175), 0);
+                        float_msg(self_obj, mstr(184), FLOAT_MSG_PURPLE);
+                        if (global_var(GVAR_IAN_STATUS) == 2) then begin
+                            float_msg(Ian_ptr, message_str(SCRIPT_IAN, 175), FLOAT_MSG_NORMAL);
                         end
                         line184flag := 1;
                     end
@@ -546,21 +546,21 @@ begin
                 else begin
                     if (cur_map_index == 34) then begin
                         if (line186flag == 0) then begin
-                            float_msg(self_obj, message_str(623, 186), 5);
+                            float_msg(self_obj, mstr(186), FLOAT_MSG_PURPLE);
                             line186flag := 1;
                         end
                     end
                     else begin
                         if (cur_map_index == 3) then begin
                             if ((tile_distance(tile_num(self_obj), 17274) < 4) and (line187flag == 0)) then begin
-                                float_msg(self_obj, message_str(623, 187), 5);
+                                float_msg(self_obj, mstr(187), FLOAT_MSG_PURPLE);
                                 line187flag := 1;
                             end
                         end
                         else begin
                             if (cur_map_index == 45) then begin
                                 if (line188flag == 0) then begin
-                                    float_msg(self_obj, message_str(623, 188), 5);
+                                    float_msg(self_obj, mstr(188), FLOAT_MSG_PURPLE);
                                     line188flag := 1;
                                 end
                             end
@@ -595,4 +595,3 @@ begin
         end
     end
 end
-
