diff --git a/Games/Fallout1.Steam/Original/KERI.ssl b/Games/Fallout1.Steam/ReDefine/KERI.ssl
index bcf93f4..9882603 100644
--- a/Games/Fallout1.Steam/Original/KERI.ssl
+++ b/Games/Fallout1.Steam/ReDefine/KERI.ssl
@@ -76,31 +76,31 @@ procedure start
 begin
     if (Only_Once) then begin
         Only_Once := 0;
-        critter_add_trait(self_obj, 1, 6, 36);
-        critter_add_trait(self_obj, 1, 5, 50);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 36);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_MERCHANT);
         LastMove := 12127;
     end
-    if (script_action == 21) then begin
+    if (script_action == look_at_proc) then begin
         call look_at_p_proc();
     end
     else begin
-        if (script_action == 4) then begin
+        if (script_action == pickup_proc) then begin
             call pickup_p_proc();
         end
         else begin
-            if (script_action == 11) then begin
+            if (script_action == talk_proc) then begin
                 call talk_p_proc();
             end
             else begin
-                if (script_action == 12) then begin
+                if (script_action == critter_proc) then begin
                     call critter_p_proc();
                 end
                 else begin
-                    if (script_action == 18) then begin
+                    if (script_action == destroy_proc) then begin
                         call destroy_p_proc();
                     end
                     else begin
-                        if (script_action == 22) then begin
+                        if (script_action == timed_event_proc) then begin
                             call timed_event_p_proc();
                         end
                     end
@@ -120,7 +120,7 @@ begin
     variable LVar0 := 0;
     if (hostile) then begin
         hostile := 0;
-        attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+        attack(dude_obj);
     end
     if (SetDayNight == 0) then begin
         add_timer_event(self_obj, game_ticks(random(3, 8)), 1);
@@ -129,7 +129,7 @@ begin
     LVar0 := tile_num(self_obj);
     if (LVar0 == 12126) then begin
         if (anim_busy(self_obj) == 0) then begin
-            if (has_trait(1, self_obj, 10) != 0) then begin
+            if (obj_get_rot(self_obj) != 0) then begin
                 anim(self_obj, 1000, 0);
             end
         end
@@ -137,7 +137,7 @@ begin
     else begin
         if (LVar0 == 11927) then begin
             if (anim_busy(self_obj) == 0) then begin
-                if (has_trait(1, self_obj, 10) != 5) then begin
+                if (obj_get_rot(self_obj) != 5) then begin
                     anim(self_obj, 1000, 5);
                 end
             end
@@ -145,7 +145,7 @@ begin
         else begin
             if (LVar0 == 12127) then begin
                 if (anim_busy(self_obj) == 0) then begin
-                    if (has_trait(1, self_obj, 10) != 2) then begin
+                    if (obj_get_rot(self_obj) != 2) then begin
                         anim(self_obj, 1000, 2);
                     end
                 end
@@ -209,7 +209,7 @@ begin
                         call Keri00();
                     end
                     else begin
-                        if (global_var(204) == 4) then begin
+                        if (global_var(GVAR_CC_JOB) == 4) then begin
                             start_gdialog(595, self_obj, 4, -1, -1);
                             gsay_start;
                             call Keri27();
@@ -258,17 +258,17 @@ end
 procedure destroy_p_proc
 begin
     if (source_obj == dude_obj) then begin
-        if (((global_var(160) + global_var(159)) >= 25) and ((global_var(159) > (2 * global_var(160))) or (global_var(156) == 1))) then begin
-            set_global_var(156, 1);
-            set_global_var(157, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
+            set_global_var(GVAR_BERSERKER_REPUTATION, 1);
+            set_global_var(GVAR_CHAMPION_REPUTATION, 0);
         end
-        if (((global_var(160) + global_var(159)) >= 25) and ((global_var(160) > (3 * global_var(159))) or (global_var(157) == 1))) then begin
-            set_global_var(157, 1);
-            set_global_var(156, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(160) > (3 * global_var(GVAR_GOOD_MONSTER))) or (global_var(GVAR_CHAMPION_REPUTATION) == 1))) then begin
+            set_global_var(GVAR_CHAMPION_REPUTATION, 1);
+            set_global_var(GVAR_BERSERKER_REPUTATION, 0);
         end
-        set_global_var(159, global_var(159) + 1);
-        if ((global_var(159) % 2) == 0) then begin
-            set_global_var(155, global_var(155) - 1);
+        set_global_var(GVAR_GOOD_MONSTER, global_var(GVAR_GOOD_MONSTER) + 1);
+        if ((global_var(GVAR_GOOD_MONSTER) % 2) == 0) then begin
+            set_global_var(GVAR_PLAYER_REPUATION, global_var(GVAR_PLAYER_REPUATION) - 1);
         end
     end
     set_map_var(28, 1);
@@ -277,7 +277,7 @@ end
 procedure look_at_p_proc
 begin
     script_overrides;
-    display_msg(message_str(595, 100));
+    display_mstr(100);
 end
 
 procedure timed_event_p_proc
@@ -290,7 +290,7 @@ begin
     variable LVar0 := 0;
     LVar0 := obj_pid(source_obj);
     if (party_member_obj(LVar0) != 0) then begin
-        set_global_var(248, 1);
+        set_global_var(GVAR_ENEMY_HUB, 1);
     end
 end
 
@@ -298,35 +298,35 @@ procedure Keri00
 begin
     if (local_var(7) == 0) then begin
         set_local_var(7, 1);
-        float_msg(self_obj, message_str(595, 101), 2);
+        float_msg(self_obj, mstr(101), FLOAT_MSG_RED);
     end
     else begin
-        float_msg(self_obj, message_str(595, 102), 2);
+        float_msg(self_obj, mstr(102), FLOAT_MSG_RED);
     end
 end
 
 procedure Keri01
 begin
-    if (global_var(71) == 0) then begin
-        set_global_var(71, 1);
+    if (global_var(GVAR_JUNKTOWN) == 0) then begin
+        set_global_var(GVAR_JUNKTOWN, 1);
     end
-    if (global_var(74) == 0) then begin
-        set_global_var(74, 1);
+    if (global_var(GVAR_BROTHERHOOD_OF_STEEL) == 0) then begin
+        set_global_var(GVAR_BROTHERHOOD_OF_STEEL, 1);
     end
-    if (global_var(72) == 0) then begin
-        set_global_var(72, 1);
+    if (global_var(GVAR_NECROPOLIS) == 0) then begin
+        set_global_var(GVAR_NECROPOLIS, 1);
     end
-    if (global_var(75) == 0) then begin
-        set_global_var(75, 1);
+    if (global_var(GVAR_ANGELS_BONEYARD) == 0) then begin
+        set_global_var(GVAR_ANGELS_BONEYARD, 1);
     end
-    if (get_critter_stat(dude_obj, 34) == 1) then begin
-        gsay_reply(595, 103);
+    if (dude_gender == 1) then begin
+        Reply(103);
     end
     else begin
-        gsay_reply(595, 104);
+        Reply(104);
     end
-    giq_option(4, 595, 105, Keri02, 50);
-    giq_option(-3, 595, 106, Keri02, 50);
+    giq_option(4, NAME, 105, Keri02, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 106, Keri02, NEUTRAL_REACTION);
 end
 
 procedure Keri02
@@ -335,10 +335,10 @@ begin
     call RecalcDateString();
     LVar0 := local_var(9);
     if (LVar0 == get_day) then begin
-        gsay_reply(595, message_str(595, 107) + message_str(595, 108));
+        Reply(mstr(107) + mstr(108));
     end
     else begin
-        gsay_reply(595, message_str(595, 107) + message_str(595, 109) + DayString + message_str(595, 112));
+        Reply(mstr(107) + mstr(109) + DayString + mstr(112));
     end
     call Keri04();
 end
@@ -350,28 +350,28 @@ begin
         call RecalcDateString();
         LVar0 := local_var(9);
         if (LVar0 == get_day) then begin
-            gsay_reply(595, message_str(595, 113));
+            Reply(mstr(113));
         end
         else begin
-            gsay_reply(595, message_str(595, 114) + DayString + message_str(595, 117));
+            Reply(mstr(114) + DayString + mstr(117));
         end
         call Keri04();
     end
     else begin
-        gsay_reply(595, 118);
+        Reply(118);
         call Keri04();
     end
 end
 
 procedure Keri04
 begin
-    giq_option(4, 595, 119, Keri04a, 50);
-    giq_option(4, 595, 120, Keri05, 50);
-    if ((get_critter_stat(dude_obj, 34) == 0) and (local_var(14) == 0)) then begin
-        giq_option(4, 595, 121, Keri04b, 50);
+    giq_option(4, NAME, 119, Keri04a, NEUTRAL_REACTION);
+    giq_option(4, NAME, 120, Keri05, NEUTRAL_REACTION);
+    if ((dude_gender == 0) and (local_var(14) == 0)) then begin
+        giq_option(4, NAME, 121, Keri04b, NEUTRAL_REACTION);
     end
-    giq_option(4, 595, 122, Keri06, 50);
-    giq_option(-3, 595, 123, Keri07, 50);
+    giq_option(4, NAME, 122, Keri06, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 123, Keri07, NEUTRAL_REACTION);
 end
 
 procedure Keri04a
@@ -391,11 +391,11 @@ end
 procedure Keri04b
 begin
     variable LVar0 := 0;
-    if (((global_var(160) + global_var(159)) >= 25) and ((global_var(159) > (2 * global_var(160))) or (global_var(156) == 1))) then begin
+    if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
         call Keri08();
     end
     else begin
-        LVar0 := get_critter_stat(dude_obj, 3);
+        LVar0 := dude_charisma;
         if (LVar0 <= 6) then begin
             call Keri16();
         end
@@ -413,56 +413,56 @@ end
 procedure Keri05
 begin
     set_local_var(13, 0);
-    gsay_message(595, 124, 50);
+    gsay_message(NAME, 124, NEUTRAL_REACTION);
 end
 
 procedure Keri06
 begin
     set_local_var(13, 0);
-    gsay_message(595, 126, 50);
+    gsay_message(NAME, 126, NEUTRAL_REACTION);
 end
 
 procedure Keri07
 begin
-    gsay_message(595, 127, 50);
+    gsay_message(NAME, 127, NEUTRAL_REACTION);
 end
 
 procedure Keri08
 begin
     variable LVar0 := 0;
     set_local_var(14, 1);
-    gsay_message(595, 130, 50);
+    gsay_message(NAME, 130, NEUTRAL_REACTION);
     gfade_out(600);
     game_time_advance(game_ticks(60 * 60 * 2));
     gfade_in(600);
-    LVar0 := create_object_sid(53, 0, 0, -1);
+    LVar0 := create_object(53, 0, 0);
     add_obj_to_inven(dude_obj, LVar0);
-    LVar0 := create_object_sid(87, 0, 0, -1);
+    LVar0 := create_object(87, 0, 0);
     add_obj_to_inven(dude_obj, LVar0);
-    LVar0 := create_object_sid(110, 0, 0, -1);
+    LVar0 := create_object(110, 0, 0);
     add_obj_to_inven(dude_obj, LVar0);
-    display_msg(message_str(595, 229));
-    gsay_message(595, 131, 50);
-    gsay_message(595, 132, 50);
+    display_mstr(229);
+    gsay_message(NAME, 131, NEUTRAL_REACTION);
+    gsay_message(NAME, 132, NEUTRAL_REACTION);
 end
 
 procedure Keri09
 begin
     variable LVar0 := 0;
     set_local_var(14, 1);
-    gsay_message(595, 133, 50);
+    gsay_message(NAME, 133, NEUTRAL_REACTION);
     gfade_out(600);
     game_time_advance(game_ticks(60 * 60 * 2));
     gfade_in(600);
-    LVar0 := create_object_sid(53, 0, 0, -1);
+    LVar0 := create_object(53, 0, 0);
     add_obj_to_inven(dude_obj, LVar0);
-    LVar0 := create_object_sid(87, 0, 0, -1);
+    LVar0 := create_object(87, 0, 0);
     add_obj_to_inven(dude_obj, LVar0);
-    LVar0 := create_object_sid(110, 0, 0, -1);
+    LVar0 := create_object(110, 0, 0);
     add_obj_to_inven(dude_obj, LVar0);
-    display_msg(message_str(595, 229));
-    gsay_message(595, 131, 50);
-    gsay_message(595, 132, 50);
+    display_mstr(229);
+    gsay_message(NAME, 131, NEUTRAL_REACTION);
+    gsay_message(NAME, 132, NEUTRAL_REACTION);
 end
 
 procedure Keri10
@@ -470,18 +470,18 @@ begin
     if (local_var(5) == 0) then begin
         set_local_var(5, 1);
         set_map_var(25, 1);
-        gsay_reply(595, 136);
-        giq_option(4, 595, 137, Keri10a, 50);
-        giq_option(4, 595, 138, KeriCombat, 51);
-        giq_option(4, 595, 139, KeriCombat, 51);
-        giq_option(-3, 595, 141, KeriCombat, 51);
+        Reply(136);
+        giq_option(4, NAME, 137, Keri10a, NEUTRAL_REACTION);
+        giq_option(4, NAME, 138, KeriCombat, BAD_REACTION);
+        giq_option(4, NAME, 139, KeriCombat, BAD_REACTION);
+        giq_option(-3, NAME, 141, KeriCombat, BAD_REACTION);
     end
     else begin
-        if (get_critter_stat(dude_obj, 34) == 0) then begin
-            float_msg(self_obj, message_str(595, 143), 2);
+        if (dude_gender == 0) then begin
+            float_msg(self_obj, mstr(143), FLOAT_MSG_RED);
         end
         else begin
-            float_msg(self_obj, message_str(595, 142), 2);
+            float_msg(self_obj, mstr(142), FLOAT_MSG_RED);
         end
         call KeriCombat();
     end
@@ -499,46 +499,46 @@ end
 
 procedure Keri11
 begin
-    gsay_message(595, 144, 50);
+    gsay_message(NAME, 144, NEUTRAL_REACTION);
 end
 
 procedure Keri12
 begin
-    gsay_reply(595, 145);
+    Reply(145);
     call KeriCombat();
 end
 
 procedure Keri13
 begin
-    float_msg(self_obj, message_str(595, 148), 2);
+    float_msg(self_obj, mstr(148), FLOAT_MSG_RED);
     call KeriCombat();
 end
 
 procedure Keri14
 begin
-    float_msg(self_obj, message_str(595, random(149, 150)), 2);
+    float_msg(self_obj, mstr(random(149, 150)), FLOAT_MSG_RED);
     call KeriCombat();
 end
 
 procedure Keri15
 begin
     if (local_var(6) != 3) then begin
-        float_msg(self_obj, message_str(595, 151 + local_var(6)), 2);
+        float_msg(self_obj, mstr(151 + local_var(6)), FLOAT_MSG_RED);
         set_local_var(6, local_var(6) + 1);
     end
     else begin
-        float_msg(self_obj, message_str(595, 154), 2);
+        float_msg(self_obj, mstr(154), FLOAT_MSG_RED);
     end
 end
 
 procedure Keri16
 begin
-    gsay_message(595, 155, 50);
+    gsay_message(NAME, 155, NEUTRAL_REACTION);
 end
 
 procedure Keri17
 begin
-    gsay_reply(595, 156);
+    Reply(156);
     call Keri33();
     call Keri20();
 end
@@ -546,67 +546,67 @@ end
 procedure Keri18
 begin
     call RecalcDateString();
-    gsay_reply(595, message_str(595, 157) + DayString + message_str(595, 158));
-    giq_option(4, 595, 159, Keri30, 50);
-    giq_option(4, 595, 160, Keri34, 50);
+    Reply(mstr(157) + DayString + mstr(158));
+    giq_option(4, NAME, 159, Keri30, NEUTRAL_REACTION);
+    giq_option(4, NAME, 160, Keri34, NEUTRAL_REACTION);
 end
 
 procedure Keri19
 begin
     set_map_var(25, 1);
-    set_global_var(204, 0);
-    if (get_critter_stat(dude_obj, 34) == 1) then begin
-        float_msg(self_obj, message_str(595, 161), 2);
+    set_global_var(GVAR_CC_JOB, 0);
+    if (dude_gender == 1) then begin
+        float_msg(self_obj, mstr(161), FLOAT_MSG_RED);
     end
     else begin
-        float_msg(self_obj, message_str(595, 162), 2);
+        float_msg(self_obj, mstr(162), FLOAT_MSG_RED);
     end
 end
 
 procedure Keri20
 begin
-    giq_option(4, 595, 163, Keri21a, 50);
-    giq_option(4, 595, 164, Keri21b, 50);
-    giq_option(4, 595, 165, Keri21c, 50);
-    giq_option(4, 595, 166, Keri21d, 50);
-    giq_option(4, 595, 167, Keri34, 50);
+    giq_option(4, NAME, 163, Keri21a, NEUTRAL_REACTION);
+    giq_option(4, NAME, 164, Keri21b, NEUTRAL_REACTION);
+    giq_option(4, NAME, 165, Keri21c, NEUTRAL_REACTION);
+    giq_option(4, NAME, 166, Keri21d, NEUTRAL_REACTION);
+    giq_option(4, NAME, 167, Keri34, NEUTRAL_REACTION);
 end
 
 procedure Keri21a
 begin
     variable LVar0 := 0;
-    set_global_var(199, 1);
+    set_global_var(GVAR_CRIMSON_CARAVANS_STATUS, 1);
     LVar0 := random(1, 8);
     if (LVar0 == 1) then begin
-        gsay_message(595, 168, 50);
+        gsay_message(NAME, 168, NEUTRAL_REACTION);
     end
     else begin
         if (LVar0 == 2) then begin
-            gsay_message(595, 169, 50);
+            gsay_message(NAME, 169, NEUTRAL_REACTION);
         end
         else begin
             if (LVar0 == 3) then begin
-                gsay_message(595, 170, 50);
+                gsay_message(NAME, 170, NEUTRAL_REACTION);
             end
             else begin
                 if (LVar0 == 4) then begin
-                    gsay_message(595, 171, 50);
+                    gsay_message(NAME, 171, NEUTRAL_REACTION);
                 end
                 else begin
                     if (LVar0 == 5) then begin
-                        gsay_message(595, 172, 50);
+                        gsay_message(NAME, 172, NEUTRAL_REACTION);
                     end
                     else begin
                         if (LVar0 == 6) then begin
-                            gsay_message(595, 173, 50);
+                            gsay_message(NAME, 173, NEUTRAL_REACTION);
                         end
                         else begin
                             if (LVar0 == 7) then begin
-                                gsay_message(595, 174, 50);
+                                gsay_message(NAME, 174, NEUTRAL_REACTION);
                             end
                             else begin
                                 if (LVar0 == 8) then begin
-                                    gsay_message(595, 175, 50);
+                                    gsay_message(NAME, 175, NEUTRAL_REACTION);
                                 end
                             end
                         end
@@ -621,34 +621,34 @@ end
 procedure Keri21b
 begin
     variable LVar0 := 0;
-    set_global_var(199, 1);
+    set_global_var(GVAR_CRIMSON_CARAVANS_STATUS, 1);
     LVar0 := random(1, 7);
     if (LVar0 == 1) then begin
-        gsay_message(595, 168, 50);
+        gsay_message(NAME, 168, NEUTRAL_REACTION);
     end
     else begin
         if (LVar0 == 2) then begin
-            gsay_message(595, 169, 50);
+            gsay_message(NAME, 169, NEUTRAL_REACTION);
         end
         else begin
             if (LVar0 == 3) then begin
-                gsay_message(595, 170, 50);
+                gsay_message(NAME, 170, NEUTRAL_REACTION);
             end
             else begin
                 if (LVar0 == 4) then begin
-                    gsay_message(595, 171, 50);
+                    gsay_message(NAME, 171, NEUTRAL_REACTION);
                 end
                 else begin
                     if (LVar0 == 5) then begin
-                        gsay_message(595, 176, 50);
+                        gsay_message(NAME, 176, NEUTRAL_REACTION);
                     end
                     else begin
                         if (LVar0 == 6) then begin
-                            gsay_message(595, 177, 50);
+                            gsay_message(NAME, 177, NEUTRAL_REACTION);
                         end
                         else begin
                             if (LVar0 == 7) then begin
-                                gsay_message(595, 178, 50);
+                                gsay_message(NAME, 178, NEUTRAL_REACTION);
                             end
                         end
                     end
@@ -663,34 +663,34 @@ end
 procedure Keri21c
 begin
     variable LVar0 := 0;
-    set_global_var(199, 1);
+    set_global_var(GVAR_CRIMSON_CARAVANS_STATUS, 1);
     LVar0 := random(1, 7);
     if (LVar0 == 1) then begin
-        gsay_message(595, 168, 50);
+        gsay_message(NAME, 168, NEUTRAL_REACTION);
     end
     else begin
         if (LVar0 == 2) then begin
-            gsay_message(595, 169, 50);
+            gsay_message(NAME, 169, NEUTRAL_REACTION);
         end
         else begin
             if (LVar0 == 3) then begin
-                gsay_message(595, 170, 50);
+                gsay_message(NAME, 170, NEUTRAL_REACTION);
             end
             else begin
                 if (LVar0 == 4) then begin
-                    gsay_message(595, 171, 50);
+                    gsay_message(NAME, 171, NEUTRAL_REACTION);
                 end
                 else begin
                     if (LVar0 == 5) then begin
-                        gsay_message(595, 179, 50);
+                        gsay_message(NAME, 179, NEUTRAL_REACTION);
                     end
                     else begin
                         if (LVar0 == 6) then begin
-                            gsay_message(595, 180, 50);
+                            gsay_message(NAME, 180, NEUTRAL_REACTION);
                         end
                         else begin
                             if (LVar0 == 7) then begin
-                                gsay_message(595, 181, 50);
+                                gsay_message(NAME, 181, NEUTRAL_REACTION);
                             end
                         end
                     end
@@ -704,34 +704,34 @@ end
 procedure Keri21d
 begin
     variable LVar0 := 0;
-    set_global_var(199, 1);
+    set_global_var(GVAR_CRIMSON_CARAVANS_STATUS, 1);
     LVar0 := random(1, 7);
     if (LVar0 == 1) then begin
-        gsay_message(595, 168, 50);
+        gsay_message(NAME, 168, NEUTRAL_REACTION);
     end
     else begin
         if (LVar0 == 2) then begin
-            gsay_message(595, 169, 50);
+            gsay_message(NAME, 169, NEUTRAL_REACTION);
         end
         else begin
             if (LVar0 == 3) then begin
-                gsay_message(595, 170, 50);
+                gsay_message(NAME, 170, NEUTRAL_REACTION);
             end
             else begin
                 if (LVar0 == 4) then begin
-                    gsay_message(595, 171, 50);
+                    gsay_message(NAME, 171, NEUTRAL_REACTION);
                 end
                 else begin
                     if (LVar0 == 5) then begin
-                        gsay_message(595, 182, 50);
+                        gsay_message(NAME, 182, NEUTRAL_REACTION);
                     end
                     else begin
                         if (LVar0 == 6) then begin
-                            gsay_message(595, 183, 50);
+                            gsay_message(NAME, 183, NEUTRAL_REACTION);
                         end
                         else begin
                             if (LVar0 == 7) then begin
-                                gsay_message(595, 184, 50);
+                                gsay_message(NAME, 184, NEUTRAL_REACTION);
                             end
                         end
                     end
@@ -744,20 +744,20 @@ end
 
 procedure Keri27
 begin
-    gsay_reply(595, 198);
-    giq_option(4, 595, 199, Keri04a, 50);
-    giq_option(4, 595, 200, Keri05, 50);
-    giq_option(4, 595, 201, Keri06, 50);
-    giq_option(-3, 595, 202, Keri07, 50);
-    giq_option(-3, 595, 203, Keri07, 50);
+    Reply(198);
+    giq_option(4, NAME, 199, Keri04a, NEUTRAL_REACTION);
+    giq_option(4, NAME, 200, Keri05, NEUTRAL_REACTION);
+    giq_option(4, NAME, 201, Keri06, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 202, Keri07, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 203, Keri07, NEUTRAL_REACTION);
 end
 
 procedure Keri28
 begin
-    gsay_reply(595, 204);
-    giq_option(4, 595, 205, Keri17, 50);
-    giq_option(4, 595, 206, KeriEnd, 50);
-    giq_option(-3, 595, 207, Keri07, 50);
+    Reply(204);
+    giq_option(4, NAME, 205, Keri17, NEUTRAL_REACTION);
+    giq_option(4, NAME, 206, KeriEnd, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 207, Keri07, NEUTRAL_REACTION);
 end
 
 procedure Keri29
@@ -765,7 +765,7 @@ begin
     variable LVar0 := 0;
     call RecalcDateString();
     LVar0 := local_var(9);
-    gsay_message(595, message_str(595, 209) + DayString + message_str(595, 212), 50);
+    gsay_message(NAME, mstr(209) + DayString + mstr(212), NEUTRAL_REACTION);
 end
 
 procedure Keri30
@@ -777,7 +777,7 @@ procedure Keri33
 begin
     variable LVar0 := 0;
     set_local_var(13, 0);
-    set_global_var(204, 1);
+    set_global_var(GVAR_CC_JOB, 1);
     LVar0 := local_var(12);
     set_local_var(12, LVar0 + 1);
 end
@@ -851,51 +851,51 @@ begin
     LVar0 := local_var(10);
     LVar1 := local_var(9);
     if (LVar0 == 1) then begin
-        DayString := message_str(595, 217);
+        DayString := mstr(217);
     end
     else begin
         if (LVar0 == 2) then begin
-            DayString := message_str(595, 218);
+            DayString := mstr(218);
         end
         else begin
             if (LVar0 == 3) then begin
-                DayString := message_str(595, 219);
+                DayString := mstr(219);
             end
             else begin
                 if (LVar0 == 4) then begin
-                    DayString := message_str(595, 220);
+                    DayString := mstr(220);
                 end
                 else begin
                     if (LVar0 == 5) then begin
-                        DayString := message_str(595, 221);
+                        DayString := mstr(221);
                     end
                     else begin
                         if (LVar0 == 6) then begin
-                            DayString := message_str(595, 222);
+                            DayString := mstr(222);
                         end
                         else begin
                             if (LVar0 == 7) then begin
-                                DayString := message_str(595, 223);
+                                DayString := mstr(223);
                             end
                             else begin
                                 if (LVar0 == 8) then begin
-                                    DayString := message_str(595, 224);
+                                    DayString := mstr(224);
                                 end
                                 else begin
                                     if (LVar0 == 9) then begin
-                                        DayString := message_str(595, 225);
+                                        DayString := mstr(225);
                                     end
                                     else begin
                                         if (LVar0 == 10) then begin
-                                            DayString := message_str(595, 226);
+                                            DayString := mstr(226);
                                         end
                                         else begin
                                             if (LVar0 == 11) then begin
-                                                DayString := message_str(595, 227);
+                                                DayString := mstr(227);
                                             end
                                             else begin
                                                 if (LVar0 == 12) then begin
-                                                    DayString := message_str(595, 228);
+                                                    DayString := mstr(228);
                                                 end
                                             end
                                         end
@@ -909,11 +909,11 @@ begin
         end
     end
     if (LVar1 == 3) then begin
-        DayString := DayString + message_str(595, 210);
+        DayString := DayString + mstr(210);
     end
     else begin
         if (LVar1 == 17) then begin
-            DayString := DayString + message_str(595, 211);
+            DayString := DayString + mstr(211);
         end
     end
 end
@@ -957,31 +957,31 @@ begin
         set_local_var(0, 50);
         set_local_var(1, 2);
         set_local_var(2, 1);
-        set_local_var(0, local_var(0) + (5 * get_critter_stat(dude_obj, 3)) - 25);
-        set_local_var(0, local_var(0) + (10 * has_trait(0, dude_obj, 10)));
-        if (has_trait(0, dude_obj, 39)) then begin
-            if (global_var(155) > 0) then begin
-                set_local_var(0, local_var(0) + global_var(155));
+        set_local_var(0, local_var(0) + (5 * dude_charisma) - 25);
+        set_local_var(0, local_var(0) + (10 * has_trait(TRAIT_PERK, dude_obj, PERK_presence)));
+        if (has_trait(TRAIT_PERK, dude_obj, PERK_cult_of_personality)) then begin
+            if (global_var(GVAR_PLAYER_REPUATION) > 0) then begin
+                set_local_var(0, local_var(0) + global_var(GVAR_PLAYER_REPUATION));
             end
             else begin
-                set_local_var(0, local_var(0) - global_var(155));
+                set_local_var(0, local_var(0) - global_var(GVAR_PLAYER_REPUATION));
             end
         end
         else begin
             if (local_var(3) == 1) then begin
-                set_local_var(0, local_var(0) - global_var(155));
+                set_local_var(0, local_var(0) - global_var(GVAR_PLAYER_REPUATION));
             end
             else begin
-                set_local_var(0, local_var(0) + global_var(155));
+                set_local_var(0, local_var(0) + global_var(GVAR_PLAYER_REPUATION));
             end
         end
-        if (global_var(158) > 2) then begin
+        if (global_var(GVAR_CHILDKILLER_REPUATION) > 2) then begin
             set_local_var(0, local_var(0) - 30);
         end
-        if (((global_var(160) + global_var(159)) >= 25) and ((global_var(160) > (3 * global_var(159))) or (global_var(157) == 1))) then begin
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(160) > (3 * global_var(GVAR_GOOD_MONSTER))) or (global_var(GVAR_CHAMPION_REPUTATION) == 1))) then begin
             set_local_var(0, local_var(0) + 20);
         end
-        if (((global_var(160) + global_var(159)) >= 25) and ((global_var(159) > (2 * global_var(160))) or (global_var(156) == 1))) then begin
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
             set_local_var(0, local_var(0) - 20);
         end
         call ReactToLevel();
@@ -1074,6 +1074,5 @@ end
 
 procedure Goodbyes
 begin
-    exit_line := message_str(634, random(100, 105));
+    exit_line := message_str(SCRIPT_MODREACT, random(100, 105));
 end
-
