diff --git a/Games/Fallout1.Steam/Original/KILDOOR1.ssl b/Games/Fallout1.Steam/ReDefine/KILDOOR1.ssl
index 9c056ee..d739092 100644
--- a/Games/Fallout1.Steam/Original/KILDOOR1.ssl
+++ b/Games/Fallout1.Steam/ReDefine/KILDOOR1.ssl
@@ -20,27 +20,27 @@ procedure use_skill_on_p_proc;
 
 procedure start
 begin
-    if (script_action == 14) then begin
+    if (script_action == damage_proc) then begin
         call damage_p_proc();
     end
     else begin
-        if (script_action == 15) then begin
+        if (script_action == map_enter_proc) then begin
             call map_enter_p_proc();
         end
         else begin
-            if (script_action == 23) then begin
+            if (script_action == map_update_proc) then begin
                 call map_update_p_proc();
             end
             else begin
-                if (script_action == 6) then begin
+                if (script_action == use_proc) then begin
                     call use_p_proc();
                 end
                 else begin
-                    if (script_action == 7) then begin
+                    if (script_action == use_obj_on_proc) then begin
                         call use_obj_on_p_proc();
                     end
                     else begin
-                        if (script_action == 8) then begin
+                        if (script_action == use_skill_on_proc) then begin
                             call use_skill_on_p_proc();
                         end
                     end
@@ -83,7 +83,7 @@ procedure use_p_proc
 begin
     if (obj_is_locked(self_obj)) then begin
         script_overrides;
-        display_msg(message_str(873, 100));
+        display_mstr(100);
     end
     else begin
         if ((game_time_hour >= 1900) or (game_time_hour < 600)) then begin
@@ -98,22 +98,22 @@ begin
         script_overrides;
         set_map_var(8, 1);
         if (not(obj_is_locked(self_obj))) then begin
-            display_msg(message_str(873, 104));
+            display_mstr(104);
         end
         else begin
             GVar0 := roll_vs_skill(source_obj, 9, 0);
             if (is_success(GVar0)) then begin
-                display_msg(message_str(873, 101));
+                display_mstr(101);
                 obj_unlock(self_obj);
                 set_local_var(0, 1);
             end
             else begin
                 if (is_critical(GVar0)) then begin
-                    display_msg(message_str(873, 103));
+                    display_mstr(103);
                     jam_lock(self_obj);
                 end
                 else begin
-                    display_msg(message_str(873, 102));
+                    display_mstr(102);
                 end
             end
         end
@@ -126,25 +126,24 @@ begin
         script_overrides;
         set_map_var(8, 1);
         if (not(obj_is_locked(self_obj))) then begin
-            display_msg(message_str(873, 104));
+            display_mstr(104);
         end
         else begin
             GVar0 := roll_vs_skill(source_obj, 9, -20);
             if (is_success(GVar0)) then begin
-                display_msg(message_str(873, 101));
+                display_mstr(101);
                 obj_unlock(self_obj);
                 set_local_var(0, 1);
             end
             else begin
                 if (is_critical(GVar0)) then begin
-                    display_msg(message_str(873, 103));
+                    display_mstr(103);
                     jam_lock(self_obj);
                 end
                 else begin
-                    display_msg(message_str(873, 102));
+                    display_mstr(102);
                 end
             end
         end
     end
 end
-
