diff --git a/Games/Fallout1.Steam/Original/KNIGHTA.ssl b/Games/Fallout1.Steam/ReDefine/KNIGHTA.ssl
index 83fc22f..50d8021 100644
--- a/Games/Fallout1.Steam/Original/KNIGHTA.ssl
+++ b/Games/Fallout1.Steam/ReDefine/KNIGHTA.ssl
@@ -14,33 +14,33 @@ procedure start
 begin
     if (only_once) then begin
         only_once := 0;
-        critter_add_trait(self_obj, 1, 6, 44);
-        critter_add_trait(self_obj, 1, 5, 64);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 44);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_BROTHERHOOD_KNIGHT);
     end
-    if (script_action == 21) then begin
+    if (script_action == look_at_proc) then begin
         call look_at_p_proc();
     end
     else begin
-        if (script_action == 4) then begin
+        if (script_action == pickup_proc) then begin
             call pickup_p_proc();
         end
         else begin
-            if (script_action == 11) then begin
+            if (script_action == talk_proc) then begin
                 call talk_p_proc();
             end
             else begin
-                if (script_action == 12) then begin
+                if (script_action == critter_proc) then begin
                     call critter_p_proc();
                 end
                 else begin
-                    if (script_action == 18) then begin
+                    if (script_action == destroy_proc) then begin
                         call destroy_p_proc();
                     end
                 end
             end
         end
     end
-    if (script_action == 11) then begin
+    if (script_action == talk_proc) then begin
         call KnightARandom();
     end
 end
@@ -55,33 +55,33 @@ begin
     if (LVar0 > 8) then begin
         LVar0 := 1;
     end
-    LVar1 := message_str(329, 101);
+    LVar1 := mstr(101);
     if (LVar0 == 2) then begin
-        LVar1 := message_str(329, 102);
+        LVar1 := mstr(102);
     end
     else begin
         if (LVar0 == 3) then begin
-            LVar1 := message_str(329, 103);
+            LVar1 := mstr(103);
         end
         else begin
             if (LVar0 == 4) then begin
-                LVar1 := message_str(329, 104);
+                LVar1 := mstr(104);
             end
             else begin
                 if (LVar0 == 5) then begin
-                    LVar1 := message_str(329, 105);
+                    LVar1 := mstr(105);
                 end
                 else begin
                     if (LVar0 == 6) then begin
-                        LVar1 := message_str(329, 106);
+                        LVar1 := mstr(106);
                     end
                     else begin
                         if (LVar0 == 7) then begin
-                            LVar1 := message_str(329, 107);
+                            LVar1 := mstr(107);
                         end
                         else begin
                             if (LVar0 == 8) then begin
-                                LVar1 := message_str(329, 108);
+                                LVar1 := mstr(108);
                             end
                             else begin
                                 LVar0 := 1;
@@ -93,21 +93,21 @@ begin
         end
     end
     LVar0 := LVar0 + 1;
-    float_msg(self_obj, LVar1, 0);
+    float_msg(self_obj, LVar1, FLOAT_MSG_NORMAL);
 end
 
 procedure critter_p_proc
 begin
-    if (global_var(250)) then begin
+    if (global_var(GVAR_ENEMY_BROTHERHOOD)) then begin
         hostile := 1;
     end
     if (tile_distance_objs(self_obj, dude_obj) > 12) then begin
         hostile := 0;
     end
     if (hostile) then begin
-        set_global_var(250, 1);
+        set_global_var(GVAR_ENEMY_BROTHERHOOD, 1);
         hostile := 0;
-        attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+        attack(dude_obj);
     end
 end
 
@@ -125,20 +125,20 @@ end
 
 procedure destroy_p_proc
 begin
-    set_global_var(250, 1);
+    set_global_var(GVAR_ENEMY_BROTHERHOOD, 1);
     rm_timer_event(self_obj);
     if (source_obj == dude_obj) then begin
-        if (((global_var(160) + global_var(159)) >= 25) and ((global_var(159) > (2 * global_var(160))) or (global_var(156) == 1))) then begin
-            set_global_var(156, 1);
-            set_global_var(157, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
+            set_global_var(GVAR_BERSERKER_REPUTATION, 1);
+            set_global_var(GVAR_CHAMPION_REPUTATION, 0);
         end
-        if (((global_var(160) + global_var(159)) >= 25) and ((global_var(160) > (3 * global_var(159))) or (global_var(157) == 1))) then begin
-            set_global_var(157, 1);
-            set_global_var(156, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(160) > (3 * global_var(GVAR_GOOD_MONSTER))) or (global_var(GVAR_CHAMPION_REPUTATION) == 1))) then begin
+            set_global_var(GVAR_CHAMPION_REPUTATION, 1);
+            set_global_var(GVAR_BERSERKER_REPUTATION, 0);
         end
-        set_global_var(159, global_var(159) + 1);
-        if ((global_var(159) % 2) == 0) then begin
-            set_global_var(155, global_var(155) - 1);
+        set_global_var(GVAR_GOOD_MONSTER, global_var(GVAR_GOOD_MONSTER) + 1);
+        if ((global_var(GVAR_GOOD_MONSTER) % 2) == 0) then begin
+            set_global_var(GVAR_PLAYER_REPUATION, global_var(GVAR_PLAYER_REPUATION) - 1);
         end
     end
 end
@@ -146,6 +146,5 @@ end
 procedure look_at_p_proc
 begin
     script_overrides;
-    display_msg(message_str(329, 100));
+    display_mstr(100);
 end
-
