diff --git a/Games/Fallout1.Steam/Original/KYLE.ssl b/Games/Fallout1.Steam/ReDefine/KYLE.ssl
index a77c790..e4e1208 100644
--- a/Games/Fallout1.Steam/Original/KYLE.ssl
+++ b/Games/Fallout1.Steam/ReDefine/KYLE.ssl
@@ -68,26 +68,26 @@ procedure start
 begin
     if (only_once) then begin
         only_once := 0;
-        critter_add_trait(self_obj, 1, 6, 44);
-        critter_add_trait(self_obj, 1, 5, 64);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 44);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_BROTHERHOOD_KNIGHT);
     end
-    if (script_action == 21) then begin
+    if (script_action == look_at_proc) then begin
         call look_at_p_proc();
     end
     else begin
-        if (script_action == 4) then begin
+        if (script_action == pickup_proc) then begin
             call pickup_p_proc();
         end
         else begin
-            if (script_action == 11) then begin
+            if (script_action == talk_proc) then begin
                 call talk_p_proc();
             end
             else begin
-                if (script_action == 12) then begin
+                if (script_action == critter_proc) then begin
                     call critter_p_proc();
                 end
                 else begin
-                    if (script_action == 18) then begin
+                    if (script_action == destroy_proc) then begin
                         call destroy_p_proc();
                     end
                 end
@@ -107,11 +107,11 @@ begin
         call goto01();
     end
     else begin
-        if (global_var(304) == 2) then begin
+        if (global_var(GVAR_CALM_REBELS_7) == 2) then begin
             call goto26();
         end
         else begin
-            if (global_var(304) == 3) then begin
+            if (global_var(GVAR_CALM_REBELS_7) == 3) then begin
                 call goto26();
             end
             else begin
@@ -119,7 +119,7 @@ begin
                     call goto23();
                 end
                 else begin
-                    if (global_var(304)) then begin
+                    if (global_var(GVAR_CALM_REBELS_7)) then begin
                         call goto17();
                     end
                     else begin
@@ -135,146 +135,146 @@ end
 
 procedure goto01
 begin
-    gsay_reply(845, 102);
-    giq_option(-3, 845, 103, goto02, 50);
-    giq_option(4, 845, 103, goto02, 50);
-    giq_option(4, 845, 104, goto13, 50);
+    Reply(102);
+    giq_option(-3, NAME, 103, goto02, NEUTRAL_REACTION);
+    giq_option(4, NAME, 103, goto02, NEUTRAL_REACTION);
+    giq_option(4, NAME, 104, goto13, NEUTRAL_REACTION);
 end
 
 procedure goto02
 begin
-    gsay_reply(845, 105);
-    giq_option(4, 845, 106, goto04, 50);
-    giq_option(4, 845, 107, goto03, 50);
-    giq_option(4, 845, 108, goto14, 50);
-    giq_option(-3, 845, 103, gotoend, 50);
+    Reply(105);
+    giq_option(4, NAME, 106, goto04, NEUTRAL_REACTION);
+    giq_option(4, NAME, 107, goto03, NEUTRAL_REACTION);
+    giq_option(4, NAME, 108, goto14, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 103, gotoend, NEUTRAL_REACTION);
 end
 
 procedure goto03
 begin
-    gsay_reply(845, 109);
-    giq_option(4, 845, 110, goto04, 50);
-    giq_option(4, 845, 111, goto06, 50);
-    giq_option(4, 845, 112, goto09, 51);
+    Reply(109);
+    giq_option(4, NAME, 110, goto04, NEUTRAL_REACTION);
+    giq_option(4, NAME, 111, goto06, NEUTRAL_REACTION);
+    giq_option(4, NAME, 112, goto09, BAD_REACTION);
 end
 
 procedure goto04
 begin
-    gsay_reply(845, 113);
-    if (global_var(30) != 1) then begin
-        giq_option(4, 845, 114, goto12, 50);
+    Reply(113);
+    if (global_var(GVAR_NECROP_WATER_CHIP_TAKEN) != 1) then begin
+        giq_option(4, NAME, 114, goto12, NEUTRAL_REACTION);
     end
-    giq_option(4, 845, 115, goto05, 50);
-    giq_option(4, 845, 116, gotoend, 50);
+    giq_option(4, NAME, 115, goto05, NEUTRAL_REACTION);
+    giq_option(4, NAME, 116, gotoend, NEUTRAL_REACTION);
 end
 
 procedure goto05
 begin
-    gsay_reply(845, 117);
-    giq_option(4, 845, 118, goto06, 50);
-    if (global_var(30) != 1) then begin
-        giq_option(4, 845, 114, goto12, 50);
+    Reply(117);
+    giq_option(4, NAME, 118, goto06, NEUTRAL_REACTION);
+    if (global_var(GVAR_NECROP_WATER_CHIP_TAKEN) != 1) then begin
+        giq_option(4, NAME, 114, goto12, NEUTRAL_REACTION);
     end
 end
 
 procedure goto06
 begin
-    gsay_reply(845, 119);
-    giq_option(4, 845, 120, goto07, 50);
-    giq_option(4, 845, 121, gotoend, 50);
+    Reply(119);
+    giq_option(4, NAME, 120, goto07, NEUTRAL_REACTION);
+    giq_option(4, NAME, 121, gotoend, NEUTRAL_REACTION);
 end
 
 procedure goto07
 begin
-    gsay_reply(845, 122);
-    giq_option(4, 845, 123, goto08, 50);
-    giq_option(4, 845, 124, goto10, 50);
+    Reply(122);
+    giq_option(4, NAME, 123, goto08, NEUTRAL_REACTION);
+    giq_option(4, NAME, 124, goto10, NEUTRAL_REACTION);
 end
 
 procedure goto08
 begin
-    gsay_reply(845, 125);
-    giq_option(4, 845, 124, goto10, 50);
-    giq_option(4, 845, 126, goto09, 51);
+    Reply(125);
+    giq_option(4, NAME, 124, goto10, NEUTRAL_REACTION);
+    giq_option(4, NAME, 126, goto09, BAD_REACTION);
 end
 
 procedure goto09
 begin
     call BottomReact();
-    gsay_message(845, 127, 51);
+    gsay_message(NAME, 127, BAD_REACTION);
 end
 
 procedure goto10
 begin
-    gsay_reply(845, 128);
-    giq_option(4, 845, 129, goto11, 50);
-    giq_option(4, 845, 130, gotopart, 50);
+    Reply(128);
+    giq_option(4, NAME, 129, goto11, NEUTRAL_REACTION);
+    giq_option(4, NAME, 130, gotopart, NEUTRAL_REACTION);
 end
 
 procedure goto11
 begin
-    gsay_reply(845, 131);
-    giq_option(4, 845, 132, gotopart, 50);
+    Reply(131);
+    giq_option(4, NAME, 132, gotopart, NEUTRAL_REACTION);
 end
 
 procedure goto12
 begin
-    gsay_reply(845, 133);
-    giq_option(4, 845, 134, gotoend, 50);
-    giq_option(4, 845, 135, goto09, 51);
-    giq_option(4, 845, 136, goto06, 50);
+    Reply(133);
+    giq_option(4, NAME, 134, gotoend, NEUTRAL_REACTION);
+    giq_option(4, NAME, 135, goto09, BAD_REACTION);
+    giq_option(4, NAME, 136, goto06, NEUTRAL_REACTION);
 end
 
 procedure goto13
 begin
-    gsay_reply(845, 137);
-    giq_option(4, 845, 138, goto02, 50);
-    giq_option(4, 845, 139, goto04, 51);
-    giq_option(4, 845, 140, gotoend, 50);
+    Reply(137);
+    giq_option(4, NAME, 138, goto02, NEUTRAL_REACTION);
+    giq_option(4, NAME, 139, goto04, BAD_REACTION);
+    giq_option(4, NAME, 140, gotoend, NEUTRAL_REACTION);
 end
 
 procedure goto14
 begin
-    gsay_reply(845, 141);
-    giq_option(4, 845, 142, goto15, 50);
+    Reply(141);
+    giq_option(4, NAME, 142, goto15, NEUTRAL_REACTION);
 end
 
 procedure goto15
 begin
-    gsay_reply(845, 143);
-    giq_option(4, 845, message_str(845, 144) + proto_data(obj_pid(dude_obj), 1) + message_str(845, 145), goto13, 50);
-    giq_option(4, 845, 146, goto16, 50);
+    Reply(143);
+    giq_option(4, NAME, mstr(144) + proto_data(obj_pid(dude_obj), 1) + mstr(145), goto13, NEUTRAL_REACTION);
+    giq_option(4, NAME, 146, goto16, NEUTRAL_REACTION);
 end
 
 procedure goto16
 begin
-    gsay_reply(845, 147);
-    giq_option(4, 845, 148, goto06, 50);
+    Reply(147);
+    giq_option(4, NAME, 148, goto06, NEUTRAL_REACTION);
 end
 
 procedure goto17
 begin
-    gsay_reply(845, 149);
-    giq_option(4, 845, 150, goto18, 50);
-    giq_option(4, 845, 151, gotoend, 50);
-    giq_option(4, 845, 152, gotoend, 50);
-    giq_option(-3, 845, 103, gotoend, 50);
+    Reply(149);
+    giq_option(4, NAME, 150, goto18, NEUTRAL_REACTION);
+    giq_option(4, NAME, 151, gotoend, NEUTRAL_REACTION);
+    giq_option(4, NAME, 152, gotoend, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 103, gotoend, NEUTRAL_REACTION);
 end
 
 procedure goto18
 begin
-    gsay_reply(845, 153);
-    giq_option(4, 845, 154, goto09, 51);
-    giq_option(4, 845, 155, goto19, 50);
-    giq_option(4, 845, 156, goto11, 50);
-    giq_option(4, 845, 157, gotoend, 50);
+    Reply(153);
+    giq_option(4, NAME, 154, goto09, BAD_REACTION);
+    giq_option(4, NAME, 155, goto19, NEUTRAL_REACTION);
+    giq_option(4, NAME, 156, goto11, NEUTRAL_REACTION);
+    giq_option(4, NAME, 157, gotoend, NEUTRAL_REACTION);
 end
 
 procedure goto19
 begin
-    gsay_reply(845, 158);
-    giq_option(4, 845, 159, gotoend, 50);
-    giq_option(4, 845, 160, goto11, 50);
+    Reply(158);
+    giq_option(4, NAME, 159, gotoend, NEUTRAL_REACTION);
+    giq_option(4, NAME, 160, goto11, NEUTRAL_REACTION);
 end
 
 procedure goto20
@@ -283,61 +283,61 @@ end
 
 procedure goto21
 begin
-    float_msg(self_obj, message_str(845, 161), 2);
+    float_msg(self_obj, mstr(161), FLOAT_MSG_RED);
 end
 
 procedure goto22
 begin
-    float_msg(self_obj, message_str(845, 162), 2);
+    float_msg(self_obj, mstr(162), FLOAT_MSG_RED);
 end
 
 procedure goto23
 begin
     loot := 0;
-    gsay_reply(845, 163);
+    Reply(163);
     loot := obj_carrying_pid_obj(dude_obj, 229);
     if (loot) then begin
-        giq_option(4, 845, 164, goto24, 50);
+        giq_option(4, NAME, 164, goto24, NEUTRAL_REACTION);
     end
-    giq_option(4, 845, 165, goto25, 50);
-    giq_option(-3, 845, 103, gotoend, 50);
+    giq_option(4, NAME, 165, goto25, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 103, gotoend, NEUTRAL_REACTION);
 end
 
 procedure goto24
 begin
     rm_obj_from_inven(dude_obj, loot);
-    gsay_reply(845, 166);
-    giq_option(4, 845, 300, goto24a, 50);
+    Reply(166);
+    giq_option(4, NAME, 300, goto24a, NEUTRAL_REACTION);
 end
 
 procedure goto24a
 begin
     gfade_out(600);
     gfade_in(600);
-    stuff := create_object_sid(76, 0, 0, -1);
+    stuff := create_object(76, 0, 0);
     add_obj_to_inven(dude_obj, stuff);
-    set_global_var(304, 2);
-    gsay_message(845, 167, 50);
+    set_global_var(GVAR_CALM_REBELS_7, 2);
+    gsay_message(NAME, 167, NEUTRAL_REACTION);
 end
 
 procedure goto25
 begin
-    gsay_message(845, 168, 50);
+    gsay_message(NAME, 168, NEUTRAL_REACTION);
 end
 
 procedure goto26
 begin
-    gsay_reply(845, 169);
-    giq_option(4, 845, 170, gotoend, 50);
-    giq_option(-3, 845, 103, gotoend, 50);
+    Reply(169);
+    giq_option(4, NAME, 170, gotoend, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 103, gotoend, NEUTRAL_REACTION);
 end
 
 procedure goto27
 begin
-    gsay_reply(845, 171);
-    giq_option(4, 845, 106, goto04, 50);
-    giq_option(4, 845, 104, goto13, 50);
-    giq_option(-3, 845, 103, gotoend, 50);
+    Reply(171);
+    giq_option(4, NAME, 106, goto04, NEUTRAL_REACTION);
+    giq_option(4, NAME, 104, goto13, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 103, gotoend, NEUTRAL_REACTION);
 end
 
 procedure goto75
@@ -350,7 +350,7 @@ end
 
 procedure gotopart
 begin
-    set_global_var(304, 1);
+    set_global_var(GVAR_CALM_REBELS_7, 1);
 end
 
 procedure combat
@@ -360,16 +360,16 @@ end
 
 procedure critter_p_proc
 begin
-    if (global_var(250)) then begin
+    if (global_var(GVAR_ENEMY_BROTHERHOOD)) then begin
         hostile := 1;
     end
     if (tile_distance_objs(self_obj, dude_obj) > 12) then begin
         hostile := 0;
     end
     if (hostile) then begin
-        set_global_var(250, 1);
+        set_global_var(GVAR_ENEMY_BROTHERHOOD, 1);
         hostile := 0;
-        attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+        attack(dude_obj);
     end
 end
 
@@ -382,8 +382,8 @@ end
 
 procedure talk_p_proc
 begin
-    debug_msg("global_var(PART_RUN) == " + global_var(304));
-    if (global_var(304) == 4) then begin
+    debug_msg("global_var(PART_RUN) == " + global_var(GVAR_CALM_REBELS_7));
+    if (global_var(GVAR_CALM_REBELS_7) == 4) then begin
         call smalltalk();
     end
     else begin
@@ -399,11 +399,11 @@ begin
                 call goto01();
             end
             else begin
-                if (global_var(304) == 2) then begin
+                if (global_var(GVAR_CALM_REBELS_7) == 2) then begin
                     call goto26();
                 end
                 else begin
-                    if (global_var(304) == 3) then begin
+                    if (global_var(GVAR_CALM_REBELS_7) == 3) then begin
                         call goto26();
                     end
                     else begin
@@ -411,7 +411,7 @@ begin
                             call goto23();
                         end
                         else begin
-                            if (global_var(304)) then begin
+                            if (global_var(GVAR_CALM_REBELS_7)) then begin
                                 call goto17();
                             end
                             else begin
@@ -429,19 +429,19 @@ end
 
 procedure destroy_p_proc
 begin
-    set_global_var(250, 1);
+    set_global_var(GVAR_ENEMY_BROTHERHOOD, 1);
     if (source_obj == dude_obj) then begin
-        if (((global_var(160) + global_var(159)) >= 25) and ((global_var(159) > (2 * global_var(160))) or (global_var(156) == 1))) then begin
-            set_global_var(156, 1);
-            set_global_var(157, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
+            set_global_var(GVAR_BERSERKER_REPUTATION, 1);
+            set_global_var(GVAR_CHAMPION_REPUTATION, 0);
         end
-        if (((global_var(160) + global_var(159)) >= 25) and ((global_var(160) > (3 * global_var(159))) or (global_var(157) == 1))) then begin
-            set_global_var(157, 1);
-            set_global_var(156, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(160) > (3 * global_var(GVAR_GOOD_MONSTER))) or (global_var(GVAR_CHAMPION_REPUTATION) == 1))) then begin
+            set_global_var(GVAR_CHAMPION_REPUTATION, 1);
+            set_global_var(GVAR_BERSERKER_REPUTATION, 0);
         end
-        set_global_var(159, global_var(159) + 1);
-        if ((global_var(159) % 2) == 0) then begin
-            set_global_var(155, global_var(155) - 1);
+        set_global_var(GVAR_GOOD_MONSTER, global_var(GVAR_GOOD_MONSTER) + 1);
+        if ((global_var(GVAR_GOOD_MONSTER) % 2) == 0) then begin
+            set_global_var(GVAR_PLAYER_REPUATION, global_var(GVAR_PLAYER_REPUATION) - 1);
         end
     end
 end
@@ -450,10 +450,10 @@ procedure look_at_p_proc
 begin
     script_overrides;
     if (known) then begin
-        display_msg(message_str(845, 100));
+        display_mstr(100);
     end
     else begin
-        display_msg(message_str(845, 101));
+        display_mstr(101);
     end
 end
 
@@ -461,18 +461,18 @@ procedure smalltalk
 begin
     ccount := ccount + 1;
     if (ccount < 6) then begin
-        float_msg(self_obj, message_str(723, 200 + random(0, 2)), 2);
+        float_msg(self_obj, message_str(SCRIPT_RD1SCRB1, 200 + random(0, 2)), FLOAT_MSG_RED);
     end
     else begin
         if (ccount < 9) then begin
-            float_msg(self_obj, message_str(723, 203 + random(0, 4)), 2);
+            float_msg(self_obj, message_str(SCRIPT_RD1SCRB1, 203 + random(0, 4)), FLOAT_MSG_RED);
         end
         else begin
             if (ccount < 12) then begin
-                float_msg(self_obj, message_str(723, 209 + random(0, 3)), 2);
+                float_msg(self_obj, message_str(SCRIPT_RD1SCRB1, 209 + random(0, 3)), FLOAT_MSG_RED);
             end
             else begin
-                display_msg(message_str(723, 213));
+                display_msg(message_str(SCRIPT_RD1SCRB1, 213));
             end
         end
     end
@@ -481,7 +481,7 @@ end
 procedure pre_dialogue
 begin
     display_msg("in pre_dialogue");
-    if (global_var(304) == 4) then begin
+    if (global_var(GVAR_CALM_REBELS_7) == 4) then begin
         call smalltalk();
     end
     else begin
@@ -500,31 +500,31 @@ begin
         set_local_var(0, 50);
         set_local_var(1, 2);
         set_local_var(2, 1);
-        set_local_var(0, local_var(0) + (5 * get_critter_stat(dude_obj, 3)) - 25);
-        set_local_var(0, local_var(0) + (10 * has_trait(0, dude_obj, 10)));
-        if (has_trait(0, dude_obj, 39)) then begin
-            if (global_var(155) > 0) then begin
-                set_local_var(0, local_var(0) + global_var(155));
+        set_local_var(0, local_var(0) + (5 * dude_charisma) - 25);
+        set_local_var(0, local_var(0) + (10 * has_trait(TRAIT_PERK, dude_obj, PERK_presence)));
+        if (has_trait(TRAIT_PERK, dude_obj, PERK_cult_of_personality)) then begin
+            if (global_var(GVAR_PLAYER_REPUATION) > 0) then begin
+                set_local_var(0, local_var(0) + global_var(GVAR_PLAYER_REPUATION));
             end
             else begin
-                set_local_var(0, local_var(0) - global_var(155));
+                set_local_var(0, local_var(0) - global_var(GVAR_PLAYER_REPUATION));
             end
         end
         else begin
             if (local_var(3) == 1) then begin
-                set_local_var(0, local_var(0) - global_var(155));
+                set_local_var(0, local_var(0) - global_var(GVAR_PLAYER_REPUATION));
             end
             else begin
-                set_local_var(0, local_var(0) + global_var(155));
+                set_local_var(0, local_var(0) + global_var(GVAR_PLAYER_REPUATION));
             end
         end
-        if (global_var(158) > 2) then begin
+        if (global_var(GVAR_CHILDKILLER_REPUATION) > 2) then begin
             set_local_var(0, local_var(0) - 30);
         end
-        if (((global_var(160) + global_var(159)) >= 25) and ((global_var(160) > (3 * global_var(159))) or (global_var(157) == 1))) then begin
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(160) > (3 * global_var(GVAR_GOOD_MONSTER))) or (global_var(GVAR_CHAMPION_REPUTATION) == 1))) then begin
             set_local_var(0, local_var(0) + 20);
         end
-        if (((global_var(160) + global_var(159)) >= 25) and ((global_var(159) > (2 * global_var(160))) or (global_var(156) == 1))) then begin
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
             set_local_var(0, local_var(0) - 20);
         end
         call ReactToLevel();
@@ -617,6 +617,5 @@ end
 
 procedure Goodbyes
 begin
-    exit_line := message_str(634, random(100, 105));
+    exit_line := message_str(SCRIPT_MODREACT, random(100, 105));
 end
-
