diff --git a/Games/Fallout1.Steam/Original/LAURA.ssl b/Games/Fallout1.Steam/ReDefine/LAURA.ssl
index 616d6e9..fed0482 100644
--- a/Games/Fallout1.Steam/Original/LAURA.ssl
+++ b/Games/Fallout1.Steam/ReDefine/LAURA.ssl
@@ -182,37 +182,37 @@ procedure lauraend;
 
 procedure start
 begin
-    if (script_action == 11) then begin
+    if (script_action == talk_proc) then begin
         call do_dialogue();
     end
     else begin
-        if (script_action == 18) then begin
+        if (script_action == destroy_proc) then begin
             if (source_obj == dude_obj) then begin
-                if (((global_var(160) + global_var(159)) >= 25) and ((global_var(159) > (2 * global_var(160))) or (global_var(156) == 1))) then begin
-                    set_global_var(156, 1);
-                    set_global_var(157, 0);
+                if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
+                    set_global_var(GVAR_BERSERKER_REPUTATION, 1);
+                    set_global_var(GVAR_CHAMPION_REPUTATION, 0);
                 end
-                if (((global_var(160) + global_var(159)) >= 25) and ((global_var(160) > (3 * global_var(159))) or (global_var(157) == 1))) then begin
-                    set_global_var(157, 1);
-                    set_global_var(156, 0);
+                if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(160) > (3 * global_var(GVAR_GOOD_MONSTER))) or (global_var(GVAR_CHAMPION_REPUTATION) == 1))) then begin
+                    set_global_var(GVAR_CHAMPION_REPUTATION, 1);
+                    set_global_var(GVAR_BERSERKER_REPUTATION, 0);
                 end
-                set_global_var(159, global_var(159) + 1);
-                if ((global_var(159) % 2) == 0) then begin
-                    set_global_var(155, global_var(155) - 1);
+                set_global_var(GVAR_GOOD_MONSTER, global_var(GVAR_GOOD_MONSTER) + 1);
+                if ((global_var(GVAR_GOOD_MONSTER) % 2) == 0) then begin
+                    set_global_var(GVAR_PLAYER_REPUATION, global_var(GVAR_PLAYER_REPUATION) - 1);
                 end
             end
         end
         else begin
-            if ((script_action == 21) or (script_action == 3)) then begin
+            if ((script_action == look_at_proc) or (script_action == description_proc)) then begin
                 script_overrides;
-                display_msg(message_str(48, 100));
+                display_mstr(100);
             end
             else begin
-                if (script_action == 4) then begin
+                if (script_action == pickup_proc) then begin
                     GVar5 := 1;
                 end
                 else begin
-                    if (script_action == 12) then begin
+                    if (script_action == critter_proc) then begin
                         Laura_Ptr := self_obj;
                         if (local_var(4) == 1) then begin
                             set_local_var(4, 2);
@@ -241,7 +241,7 @@ begin
                                         end
                                         else begin
                                             if (local_var(4) == 4) then begin
-                                                float_msg(self_obj, message_str(766, 178), 8);
+                                                float_msg(self_obj, message_str(SCRIPT_GENCHAT, 178), FLOAT_MSG_YELLOW);
                                                 set_local_var(4, 5);
                                                 set_local_var(9, 2);
                                             end
@@ -314,11 +314,11 @@ begin
                         end
                         if (GVar5) then begin
                             GVar5 := 0;
-                            attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+                            attack(dude_obj);
                         end
                     end
                     else begin
-                        if (script_action == 22) then begin
+                        if (script_action == timed_event_proc) then begin
                             GVar5 := 1;
                         end
                     end
@@ -332,16 +332,16 @@ procedure do_dialogue
 begin
     call get_reaction();
     if (local_var(9) == 1) then begin
-        float_msg(self_obj, message_str(766, 177), 8);
+        float_msg(self_obj, message_str(SCRIPT_GENCHAT, 177), FLOAT_MSG_YELLOW);
     end
     else begin
         if (local_var(9) == 2) then begin
-            float_msg(self_obj, message_str(766, 179), 8);
+            float_msg(self_obj, message_str(SCRIPT_GENCHAT, 179), FLOAT_MSG_YELLOW);
         end
         else begin
             start_gdialog(48, self_obj, 4, 3, 8);
             gsay_start;
-            GVar4 := get_critter_stat(dude_obj, 34) == 0;
+            GVar4 := dude_gender == 0;
             if (GVar10) then begin
                 if (GVar11) then begin
                     call laura63();
@@ -389,177 +389,177 @@ end
 procedure laura01
 begin
     if (GVar4) then begin
-        gsay_reply(48, 102);
+        Reply(102);
     end
     else begin
-        gsay_reply(48, 104);
+        Reply(104);
     end
-    if (global_var(61) == 1) then begin
-        giq_option(4, 48, 105, laura02, 50);
+    if (global_var(GVAR_SHADOW_PASSWORD) == 1) then begin
+        giq_option(4, NAME, 105, laura02, NEUTRAL_REACTION);
     end
-    giq_option(5, 48, 106, laura14, 50);
-    giq_option(4, 48, 107, laura15, 51);
-    giq_option(-3, 48, 108, laura01a, 50);
+    giq_option(5, NAME, 106, laura14, NEUTRAL_REACTION);
+    giq_option(4, NAME, 107, laura15, BAD_REACTION);
+    giq_option(-3, NAME, 108, laura01a, NEUTRAL_REACTION);
 end
 
 procedure laura01a
 begin
-    gsay_message(48, 110, 50);
+    gsay_message(NAME, 110, NEUTRAL_REACTION);
     call laurax();
 end
 
 procedure laura02
 begin
-    gsay_reply(48, 112);
-    if (global_var(61) == 1) then begin
-        giq_option(4, 48, 113, laura03, 50);
+    Reply(112);
+    if (global_var(GVAR_SHADOW_PASSWORD) == 1) then begin
+        giq_option(4, NAME, 113, laura03, NEUTRAL_REACTION);
     end
-    giq_option(4, 48, 114, laura10, 50);
+    giq_option(4, NAME, 114, laura10, NEUTRAL_REACTION);
 end
 
 procedure laura03
 begin
-    gsay_reply(48, 116);
-    if (global_var(61) == 1) then begin
-        giq_option(4, 48, 117, laura04, 50);
+    Reply(116);
+    if (global_var(GVAR_SHADOW_PASSWORD) == 1) then begin
+        giq_option(4, NAME, 117, laura04, NEUTRAL_REACTION);
     end
-    giq_option(4, 48, 118, laura09, 50);
+    giq_option(4, NAME, 118, laura09, NEUTRAL_REACTION);
 end
 
 procedure laura04
 begin
-    gsay_message(48, 120, 50);
+    gsay_message(NAME, 120, NEUTRAL_REACTION);
     set_local_var(7, 1);
     call laurax1();
 end
 
 procedure laura09
 begin
-    gsay_reply(48, 122);
+    Reply(122);
     call laurax();
 end
 
 procedure laura10
 begin
-    gsay_reply(48, 124);
-    giq_option(4, 48, 125, laura03, 50);
-    giq_option(4, 48, 126, laura11, 50);
+    Reply(124);
+    giq_option(4, NAME, 125, laura03, NEUTRAL_REACTION);
+    giq_option(4, NAME, 126, laura11, NEUTRAL_REACTION);
 end
 
 procedure laura11
 begin
     call DownReact();
-    gsay_message(48, 128, 50);
+    gsay_message(NAME, 128, NEUTRAL_REACTION);
     call laurax();
 end
 
 procedure laura14
 begin
-    gsay_message(48, 130, 50);
+    gsay_message(NAME, 130, NEUTRAL_REACTION);
     call laurax();
 end
 
 procedure laura15
 begin
     call DownReact();
-    gsay_message(48, 132, 50);
+    gsay_message(NAME, 132, NEUTRAL_REACTION);
     call laurax();
 end
 
 procedure laura16
 begin
     if (local_var(1) > 1) then begin
-        gsay_reply(48, 134);
+        Reply(134);
     end
     else begin
-        gsay_reply(48, 136);
+        Reply(136);
     end
-    if (global_var(61) == 1) then begin
-        giq_option(4, 48, 137, laura02, 50);
+    if (global_var(GVAR_SHADOW_PASSWORD) == 1) then begin
+        giq_option(4, NAME, 137, laura02, NEUTRAL_REACTION);
     end
-    giq_option(4, 48, 138, laura14, 50);
-    giq_option(4, 48, 139, laurax, 50);
-    giq_option(-3, 48, 140, laura16a, 50);
+    giq_option(4, NAME, 138, laura14, NEUTRAL_REACTION);
+    giq_option(4, NAME, 139, laurax, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 140, laura16a, NEUTRAL_REACTION);
 end
 
 procedure laura16a
 begin
-    gsay_message(48, 142, 50);
+    gsay_message(NAME, 142, NEUTRAL_REACTION);
     call laurax();
 end
 
 procedure laura21
 begin
-    gsay_message(48, 144, 50);
-    giq_option(4, 48, 145, laura22, 50);
+    gsay_message(NAME, 144, NEUTRAL_REACTION);
+    giq_option(4, NAME, 145, laura22, NEUTRAL_REACTION);
 end
 
 procedure laura22
 begin
-    gsay_message(48, 147, 50);
+    gsay_message(NAME, 147, NEUTRAL_REACTION);
     call laura23();
 end
 
 procedure laura23
 begin
-    giq_option(5, 48, 148, laura24, 50);
-    giq_option(4, 48, 149, laura33, 50);
-    giq_option(4, 48, 150, laura23_1, 50);
+    giq_option(5, NAME, 148, laura24, NEUTRAL_REACTION);
+    giq_option(4, NAME, 149, laura33, NEUTRAL_REACTION);
+    giq_option(4, NAME, 150, laura23_1, NEUTRAL_REACTION);
     if (GVar4) then begin
-        giq_option(4, 48, 151, laura60, 50);
+        giq_option(4, NAME, 151, laura60, NEUTRAL_REACTION);
     end
-    giq_option(4, 48, 152, laura59, 50);
+    giq_option(4, NAME, 152, laura59, NEUTRAL_REACTION);
 end
 
 procedure laura23_1
 begin
-    if (global_var(52) == 0) then begin
+    if (global_var(GVAR_CHILDREN_STATUS) == 0) then begin
         call laura51();
     end
-    if (global_var(52) == 1) then begin
+    if (global_var(GVAR_CHILDREN_STATUS) == 1) then begin
         call laura38();
     end
-    if (global_var(52) == 2) then begin
+    if (global_var(GVAR_CHILDREN_STATUS) == 2) then begin
         call laura46();
     end
 end
 
 procedure laura24
 begin
-    gsay_reply(48, 154);
-    giq_option(4, 48, 155, laura25, 50);
+    Reply(154);
+    giq_option(4, NAME, 155, laura25, NEUTRAL_REACTION);
 end
 
 procedure laura25
 begin
-    gsay_reply(48, 157);
-    giq_option(4, 48, 158, laura26, 50);
-    giq_option(4, 48, 159, laura28, 50);
+    Reply(157);
+    giq_option(4, NAME, 158, laura26, NEUTRAL_REACTION);
+    giq_option(4, NAME, 159, laura28, NEUTRAL_REACTION);
 end
 
 procedure laura26
 begin
-    gsay_reply(48, 161);
-    giq_option(4, 48, 162, laura27, 50);
-    giq_option(4, 48, 163, laura31, 50);
+    Reply(161);
+    giq_option(4, NAME, 162, laura27, NEUTRAL_REACTION);
+    giq_option(4, NAME, 163, laura31, NEUTRAL_REACTION);
 end
 
 procedure laura27
 begin
-    gsay_reply(48, 165);
-    giq_option(4, 48, 166, laura28, 50);
+    Reply(165);
+    giq_option(4, NAME, 166, laura28, NEUTRAL_REACTION);
 end
 
 procedure laura28
 begin
-    gsay_reply(48, 168);
-    giq_option(4, 48, 169, laura29, 50);
-    giq_option(4, 48, 170, laura30, 50);
+    Reply(168);
+    giq_option(4, NAME, 169, laura29, NEUTRAL_REACTION);
+    giq_option(4, NAME, 170, laura30, NEUTRAL_REACTION);
 end
 
 procedure laura29
 begin
-    gsay_reply(48, 172);
+    Reply(172);
     call laurax4();
 end
 
@@ -567,27 +567,27 @@ procedure laura30
 begin
     GVar3 := random(1, 2);
     if (GVar3 == 1) then begin
-        gsay_reply(48, 174);
+        Reply(174);
     end
     if (GVar3 == 2) then begin
-        gsay_reply(48, 176);
+        Reply(176);
     end
     call laura23();
 end
 
 procedure laura31
 begin
-    gsay_reply(48, 178);
-    giq_option(4, 48, 179, laura32a, 51);
-    giq_option(4, 48, 180, laura30, 50);
+    Reply(178);
+    giq_option(4, NAME, 179, laura32a, BAD_REACTION);
+    giq_option(4, NAME, 180, laura30, NEUTRAL_REACTION);
 end
 
 procedure laura32
 begin
-    gsay_reply(48, 182);
-    giq_option(4, 48, 183, laurax2, 51);
-    giq_option(4, 48, 184, laurax8, 50);
-    giq_option(4, 48, 185, laura30, 50);
+    Reply(182);
+    giq_option(4, NAME, 183, laurax2, BAD_REACTION);
+    giq_option(4, NAME, 184, laurax8, NEUTRAL_REACTION);
+    giq_option(4, NAME, 185, laura30, NEUTRAL_REACTION);
 end
 
 procedure laura32a
@@ -598,183 +598,183 @@ end
 
 procedure laura33
 begin
-    gsay_reply(48, 187);
-    giq_option(4, 48, 188, laura34, 50);
-    giq_option(4, 48, 189, laura35, 50);
+    Reply(187);
+    giq_option(4, NAME, 188, laura34, NEUTRAL_REACTION);
+    giq_option(4, NAME, 189, laura35, NEUTRAL_REACTION);
 end
 
 procedure laura34
 begin
-    gsay_reply(48, 191);
-    giq_option(6, 48, 192, laura36, 50);
-    giq_option(4, 48, 193, laura36, 50);
+    Reply(191);
+    giq_option(6, NAME, 192, laura36, NEUTRAL_REACTION);
+    giq_option(4, NAME, 193, laura36, NEUTRAL_REACTION);
 end
 
 procedure laura35
 begin
-    gsay_reply(48, 195);
-    giq_option(6, 48, 196, laura36, 50);
-    giq_option(4, 48, 197, laura36, 50);
+    Reply(195);
+    giq_option(6, NAME, 196, laura36, NEUTRAL_REACTION);
+    giq_option(4, NAME, 197, laura36, NEUTRAL_REACTION);
 end
 
 procedure laura36
 begin
-    gsay_reply(48, 199);
-    giq_option(4, 48, 200, laura37, 50);
+    Reply(199);
+    giq_option(4, NAME, 200, laura37, NEUTRAL_REACTION);
 end
 
 procedure laura37
 begin
-    gsay_reply(48, 202);
-    giq_option(4, 48, 203, laura28, 50);
-    giq_option(4, 48, 204, laura27, 50);
+    Reply(202);
+    giq_option(4, NAME, 203, laura28, NEUTRAL_REACTION);
+    giq_option(4, NAME, 204, laura27, NEUTRAL_REACTION);
 end
 
 procedure laura38
 begin
-    gsay_reply(48, 206);
-    giq_option(4, 48, 207, laura39, 50);
-    giq_option(4, 48, 208, laura43, 50);
+    Reply(206);
+    giq_option(4, NAME, 207, laura39, NEUTRAL_REACTION);
+    giq_option(4, NAME, 208, laura43, NEUTRAL_REACTION);
 end
 
 procedure laura39
 begin
-    gsay_reply(48, 210);
-    giq_option(4, 48, 211, laura40, 50);
-    giq_option(4, 48, 212, laura42, 50);
+    Reply(210);
+    giq_option(4, NAME, 211, laura40, NEUTRAL_REACTION);
+    giq_option(4, NAME, 212, laura42, NEUTRAL_REACTION);
 end
 
 procedure laura40
 begin
-    gsay_reply(48, 214);
-    giq_option(4, 48, 215, laura30, 50);
-    giq_option(4, 48, 216, laura41, 50);
+    Reply(214);
+    giq_option(4, NAME, 215, laura30, NEUTRAL_REACTION);
+    giq_option(4, NAME, 216, laura41, NEUTRAL_REACTION);
 end
 
 procedure laura41
 begin
-    gsay_message(48, 218, 50);
+    gsay_message(NAME, 218, NEUTRAL_REACTION);
     call laurax6();
 end
 
 procedure laura42
 begin
-    gsay_reply(48, 220);
-    giq_option(4, 48, 221, laura30, 50);
-    giq_option(4, 48, 222, laura41, 50);
+    Reply(220);
+    giq_option(4, NAME, 221, laura30, NEUTRAL_REACTION);
+    giq_option(4, NAME, 222, laura41, NEUTRAL_REACTION);
 end
 
 procedure laura43
 begin
-    gsay_reply(48, 224);
-    giq_option(4, 48, 225, laura44, 50);
-    giq_option(4, 48, 226, laura45, 50);
+    Reply(224);
+    giq_option(4, NAME, 225, laura44, NEUTRAL_REACTION);
+    giq_option(4, NAME, 226, laura45, NEUTRAL_REACTION);
 end
 
 procedure laura44
 begin
-    gsay_reply(48, 228);
-    giq_option(4, 48, 229, laura42, 50);
-    giq_option(4, 48, 230, laura45, 50);
+    Reply(228);
+    giq_option(4, NAME, 229, laura42, NEUTRAL_REACTION);
+    giq_option(4, NAME, 230, laura45, NEUTRAL_REACTION);
 end
 
 procedure laura45
 begin
-    gsay_reply(48, 232);
-    giq_option(4, 48, 233, laura37, 50);
-    giq_option(4, 48, 234, laura30, 50);
-    giq_option(4, 48, 235, laura41, 50);
+    Reply(232);
+    giq_option(4, NAME, 233, laura37, NEUTRAL_REACTION);
+    giq_option(4, NAME, 234, laura30, NEUTRAL_REACTION);
+    giq_option(4, NAME, 235, laura41, NEUTRAL_REACTION);
 end
 
 procedure laura46
 begin
-    gsay_reply(48, 237);
-    giq_option(4, 48, 238, laura46_1, 50);
+    Reply(237);
+    giq_option(4, NAME, 238, laura46_1, NEUTRAL_REACTION);
 end
 
 procedure laura46_1
 begin
-    if (global_var(7) == 1) then begin
+    if (global_var(GVAR_FOLLOWERS_INVADED) == 1) then begin
         call laura46();
     end
-    if (global_var(14) == 1) then begin
+    if (global_var(GVAR_HUB_INVADED) == 1) then begin
         call laura49();
     end
 end
 
 procedure laura47
 begin
-    gsay_reply(48, 240);
-    giq_option(4, 48, 241, laura48, 50);
-    giq_option(4, 48, 242, laura45, 50);
+    Reply(240);
+    giq_option(4, NAME, 241, laura48, NEUTRAL_REACTION);
+    giq_option(4, NAME, 242, laura45, NEUTRAL_REACTION);
 end
 
 procedure laura48
 begin
-    gsay_reply(48, 244);
-    giq_option(4, 48, 245, laura45, 50);
+    Reply(244);
+    giq_option(4, NAME, 245, laura45, NEUTRAL_REACTION);
 end
 
 procedure laura49
 begin
-    gsay_reply(48, 247);
-    giq_option(4, 48, 248, laura50, 50);
-    giq_option(4, 48, 249, laura45, 50);
+    Reply(247);
+    giq_option(4, NAME, 248, laura50, NEUTRAL_REACTION);
+    giq_option(4, NAME, 249, laura45, NEUTRAL_REACTION);
 end
 
 procedure laura50
 begin
-    gsay_reply(48, 251);
-    giq_option(4, 48, 252, laura45, 50);
+    Reply(251);
+    giq_option(4, NAME, 252, laura45, NEUTRAL_REACTION);
 end
 
 procedure laura51
 begin
-    gsay_reply(48, 254);
-    giq_option(4, 48, 255, laura52, 50);
-    giq_option(4, 48, 256, laura55, 50);
-    giq_option(4, 48, 257, laura57, 50);
+    Reply(254);
+    giq_option(4, NAME, 255, laura52, NEUTRAL_REACTION);
+    giq_option(4, NAME, 256, laura55, NEUTRAL_REACTION);
+    giq_option(4, NAME, 257, laura57, NEUTRAL_REACTION);
 end
 
 procedure laura52
 begin
-    gsay_reply(48, 259);
-    giq_option(4, 48, 260, laura53, 50);
-    giq_option(4, 48, 261, laura55, 50);
+    Reply(259);
+    giq_option(4, NAME, 260, laura53, NEUTRAL_REACTION);
+    giq_option(4, NAME, 261, laura55, NEUTRAL_REACTION);
 end
 
 procedure laura53
 begin
-    gsay_reply(48, 263);
-    giq_option(4, 48, 264, laura54, 50);
-    giq_option(4, 48, 265, laura30, 50);
+    Reply(263);
+    giq_option(4, NAME, 264, laura54, NEUTRAL_REACTION);
+    giq_option(4, NAME, 265, laura30, NEUTRAL_REACTION);
 end
 
 procedure laura54
 begin
-    gsay_reply(48, 267);
+    Reply(267);
     call laura23();
 end
 
 procedure laura55
 begin
-    gsay_reply(48, 269);
-    giq_option(4, 48, 270, laura33, 50);
-    giq_option(4, 48, 271, laura56, 50);
+    Reply(269);
+    giq_option(4, NAME, 270, laura33, NEUTRAL_REACTION);
+    giq_option(4, NAME, 271, laura56, NEUTRAL_REACTION);
 end
 
 procedure laura56
 begin
-    gsay_reply(48, 273);
+    Reply(273);
     call laura23();
 end
 
 procedure laura57
 begin
-    gsay_reply(48, 275);
-    giq_option(4, 48, 276, laura57_01, 50);
-    giq_option(4, 48, 277, laura56, 50);
-    giq_option(4, 48, 278, laura55, 50);
+    Reply(275);
+    giq_option(4, NAME, 276, laura57_01, NEUTRAL_REACTION);
+    giq_option(4, NAME, 277, laura56, NEUTRAL_REACTION);
+    giq_option(4, NAME, 278, laura55, NEUTRAL_REACTION);
 end
 
 procedure laura57_01
@@ -790,126 +790,126 @@ end
 
 procedure laura58
 begin
-    gsay_reply(48, 280);
-    giq_option(6, 48, 281, laura32, 50);
-    giq_option(4, 48, 282, laura23, 50);
+    Reply(280);
+    giq_option(6, NAME, 281, laura32, NEUTRAL_REACTION);
+    giq_option(4, NAME, 282, laura23, NEUTRAL_REACTION);
 end
 
 procedure laura59
 begin
-    gsay_reply(48, 284);
+    Reply(284);
     call laurax7();
 end
 
 procedure laura60
 begin
-    gsay_reply(48, 286);
+    Reply(286);
     call laura23();
 end
 
 procedure laura61
 begin
     if (local_var(1) > 1) then begin
-        gsay_reply(48, 288);
+        Reply(288);
     end
     else begin
-        gsay_reply(48, 290);
+        Reply(290);
     end
-    giq_option(4, 48, 291, laura30, 50);
-    giq_option(4, 48, 292, laura41, 50);
+    giq_option(4, NAME, 291, laura30, NEUTRAL_REACTION);
+    giq_option(4, NAME, 292, laura41, NEUTRAL_REACTION);
 end
 
 procedure laura62
 begin
     call DownReact();
-    gsay_message(48, 294, 50);
+    gsay_message(NAME, 294, NEUTRAL_REACTION);
 end
 
 procedure laura63
 begin
     call DownReact();
-    gsay_reply(48, 296);
-    giq_option(4, 48, 297, laurax2, 50);
-    giq_option(4, 48, 298, laurax5, 50);
+    Reply(296);
+    giq_option(4, NAME, 297, laurax2, NEUTRAL_REACTION);
+    giq_option(4, NAME, 298, laurax5, NEUTRAL_REACTION);
 end
 
 procedure laura64
 begin
     call DownReact();
-    gsay_message(48, 300, 50);
+    gsay_message(NAME, 300, NEUTRAL_REACTION);
 end
 
 procedure laura65
 begin
     call DownReact();
-    gsay_reply(48, 302);
-    giq_option(4, 48, 303, laurax2, 50);
-    giq_option(4, 48, 304, laurax5, 50);
+    Reply(302);
+    giq_option(4, NAME, 303, laurax2, NEUTRAL_REACTION);
+    giq_option(4, NAME, 304, laurax5, NEUTRAL_REACTION);
 end
 
 procedure laura67
 begin
-    gsay_message(48, 306, 50);
+    gsay_message(NAME, 306, NEUTRAL_REACTION);
 end
 
 procedure laura68
 begin
-    gsay_message(48, 309, 50);
+    gsay_message(NAME, 309, NEUTRAL_REACTION);
 end
 
 procedure laura69
 begin
-    gsay_message(48, 311, 50);
+    gsay_message(NAME, 311, NEUTRAL_REACTION);
 end
 
 procedure laura70
 begin
-    gsay_message(48, 313, 50);
+    gsay_message(NAME, 313, NEUTRAL_REACTION);
 end
 
 procedure laura71
 begin
-    gsay_message(48, 315, 50);
+    gsay_message(NAME, 315, NEUTRAL_REACTION);
 end
 
 procedure laura72
 begin
-    gsay_message(48, 317, 50);
+    gsay_message(NAME, 317, NEUTRAL_REACTION);
 end
 
 procedure laura73
 begin
-    gsay_message(48, 319, 50);
+    gsay_message(NAME, 319, NEUTRAL_REACTION);
 end
 
 procedure laura74
 begin
-    gsay_message(48, 321, 50);
+    gsay_message(NAME, 321, NEUTRAL_REACTION);
 end
 
 procedure laura75
 begin
-    gsay_message(48, 323, 50);
+    gsay_message(NAME, 323, NEUTRAL_REACTION);
 end
 
 procedure laura76
 begin
-    gsay_message(48, 325, 50);
+    gsay_message(NAME, 325, NEUTRAL_REACTION);
 end
 
 procedure laura77
 begin
-    gsay_message(48, 327, 50);
+    gsay_message(NAME, 327, NEUTRAL_REACTION);
 end
 
 procedure laura78
 begin
-    gsay_message(48, 329, 50);
+    gsay_message(NAME, 329, NEUTRAL_REACTION);
 end
 
 procedure laura79
 begin
-    gsay_message(48, 331, 50);
+    gsay_message(NAME, 331, NEUTRAL_REACTION);
 end
 
 procedure lauraxx
@@ -968,31 +968,31 @@ begin
         set_local_var(0, 50);
         set_local_var(1, 2);
         set_local_var(2, 1);
-        set_local_var(0, local_var(0) + (5 * get_critter_stat(dude_obj, 3)) - 25);
-        set_local_var(0, local_var(0) + (10 * has_trait(0, dude_obj, 10)));
-        if (has_trait(0, dude_obj, 39)) then begin
-            if (global_var(155) > 0) then begin
-                set_local_var(0, local_var(0) + global_var(155));
+        set_local_var(0, local_var(0) + (5 * dude_charisma) - 25);
+        set_local_var(0, local_var(0) + (10 * has_trait(TRAIT_PERK, dude_obj, PERK_presence)));
+        if (has_trait(TRAIT_PERK, dude_obj, PERK_cult_of_personality)) then begin
+            if (global_var(GVAR_PLAYER_REPUATION) > 0) then begin
+                set_local_var(0, local_var(0) + global_var(GVAR_PLAYER_REPUATION));
             end
             else begin
-                set_local_var(0, local_var(0) - global_var(155));
+                set_local_var(0, local_var(0) - global_var(GVAR_PLAYER_REPUATION));
             end
         end
         else begin
             if (local_var(3) == 1) then begin
-                set_local_var(0, local_var(0) - global_var(155));
+                set_local_var(0, local_var(0) - global_var(GVAR_PLAYER_REPUATION));
             end
             else begin
-                set_local_var(0, local_var(0) + global_var(155));
+                set_local_var(0, local_var(0) + global_var(GVAR_PLAYER_REPUATION));
             end
         end
-        if (global_var(158) > 2) then begin
+        if (global_var(GVAR_CHILDKILLER_REPUATION) > 2) then begin
             set_local_var(0, local_var(0) - 30);
         end
-        if (((global_var(160) + global_var(159)) >= 25) and ((global_var(160) > (3 * global_var(159))) or (global_var(157) == 1))) then begin
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(160) > (3 * global_var(GVAR_GOOD_MONSTER))) or (global_var(GVAR_CHAMPION_REPUTATION) == 1))) then begin
             set_local_var(0, local_var(0) + 20);
         end
-        if (((global_var(160) + global_var(159)) >= 25) and ((global_var(159) > (2 * global_var(160))) or (global_var(156) == 1))) then begin
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
             set_local_var(0, local_var(0) - 20);
         end
         call ReactToLevel();
@@ -1085,165 +1085,164 @@ end
 
 procedure Goodbyes
 begin
-    GVar15 := message_str(634, random(100, 105));
+    GVar15 := message_str(SCRIPT_MODREACT, random(100, 105));
 end
 
 procedure laura67a
 begin
-    gsay_message(48, 308, 50);
+    gsay_message(NAME, 308, NEUTRAL_REACTION);
 end
 
 procedure laura68a
 begin
-    gsay_message(48, 310, 50);
+    gsay_message(NAME, 310, NEUTRAL_REACTION);
 end
 
 procedure laura69a
 begin
-    gsay_message(48, 312, 50);
+    gsay_message(NAME, 312, NEUTRAL_REACTION);
 end
 
 procedure laura70a
 begin
-    gsay_message(48, 314, 50);
+    gsay_message(NAME, 314, NEUTRAL_REACTION);
 end
 
 procedure laura71a
 begin
-    gsay_message(48, 316, 50);
+    gsay_message(NAME, 316, NEUTRAL_REACTION);
 end
 
 procedure laura72a
 begin
-    gsay_message(48, 318, 50);
+    gsay_message(NAME, 318, NEUTRAL_REACTION);
 end
 
 procedure laura73a
 begin
-    gsay_message(48, 320, 50);
+    gsay_message(NAME, 320, NEUTRAL_REACTION);
 end
 
 procedure laura74a
 begin
-    gsay_message(48, 322, 50);
+    gsay_message(NAME, 322, NEUTRAL_REACTION);
 end
 
 procedure laura75a
 begin
-    gsay_message(48, 324, 50);
+    gsay_message(NAME, 324, NEUTRAL_REACTION);
 end
 
 procedure laura76a
 begin
-    gsay_message(48, 326, 50);
+    gsay_message(NAME, 326, NEUTRAL_REACTION);
 end
 
 procedure laura77a
 begin
-    gsay_message(48, 328, 50);
+    gsay_message(NAME, 328, NEUTRAL_REACTION);
 end
 
 procedure laura78a
 begin
-    gsay_message(48, 330, 50);
+    gsay_message(NAME, 330, NEUTRAL_REACTION);
 end
 
 procedure laura79a
 begin
-    gsay_message(48, 332, 50);
+    gsay_message(NAME, 332, NEUTRAL_REACTION);
 end
 
 procedure laura80
 begin
-    gsay_message(48, 333, 50);
+    gsay_message(NAME, 333, NEUTRAL_REACTION);
 end
 
 procedure laura80a
 begin
-    gsay_message(48, 334, 50);
+    gsay_message(NAME, 334, NEUTRAL_REACTION);
 end
 
 procedure laura81
 begin
-    gsay_message(48, 335, 50);
+    gsay_message(NAME, 335, NEUTRAL_REACTION);
 end
 
 procedure laura81a
 begin
-    gsay_message(48, 336, 50);
+    gsay_message(NAME, 336, NEUTRAL_REACTION);
 end
 
 procedure laura82
 begin
-    gsay_message(48, 337, 50);
+    gsay_message(NAME, 337, NEUTRAL_REACTION);
 end
 
 procedure laura82a
 begin
-    gsay_message(48, 338, 50);
+    gsay_message(NAME, 338, NEUTRAL_REACTION);
 end
 
 procedure laura83
 begin
-    gsay_message(48, 339, 50);
+    gsay_message(NAME, 339, NEUTRAL_REACTION);
 end
 
 procedure laura83a
 begin
-    gsay_message(48, 340, 50);
+    gsay_message(NAME, 340, NEUTRAL_REACTION);
 end
 
 procedure laura84
 begin
-    gsay_message(48, 341, 50);
+    gsay_message(NAME, 341, NEUTRAL_REACTION);
 end
 
 procedure laura84a
 begin
-    gsay_message(48, 342, 50);
+    gsay_message(NAME, 342, NEUTRAL_REACTION);
 end
 
 procedure laura85
 begin
-    gsay_message(48, 343, 50);
+    gsay_message(NAME, 343, NEUTRAL_REACTION);
 end
 
 procedure laura85a
 begin
-    gsay_message(48, 344, 50);
+    gsay_message(NAME, 344, NEUTRAL_REACTION);
 end
 
 procedure laura86
 begin
-    gsay_message(48, 345, 50);
+    gsay_message(NAME, 345, NEUTRAL_REACTION);
 end
 
 procedure laura86a
 begin
-    gsay_message(48, 346, 50);
+    gsay_message(NAME, 346, NEUTRAL_REACTION);
 end
 
 procedure laura87
 begin
-    gsay_message(48, 347, 50);
+    gsay_message(NAME, 347, NEUTRAL_REACTION);
 end
 
 procedure laura87a
 begin
-    gsay_message(48, 348, 50);
+    gsay_message(NAME, 348, NEUTRAL_REACTION);
 end
 
 procedure laura88
 begin
-    gsay_message(48, 349, 50);
+    gsay_message(NAME, 349, NEUTRAL_REACTION);
 end
 
 procedure laura88a
 begin
-    gsay_message(48, 350, 50);
+    gsay_message(NAME, 350, NEUTRAL_REACTION);
 end
 
 procedure lauraend
 begin
 end
-
