diff --git a/Games/Fallout1.Steam/Original/LOWCRAPS.ssl b/Games/Fallout1.Steam/ReDefine/LOWCRAPS.ssl
index b61c454..86097db 100644
--- a/Games/Fallout1.Steam/Original/LOWCRAPS.ssl
+++ b/Games/Fallout1.Steam/ReDefine/LOWCRAPS.ssl
@@ -6,16 +6,16 @@ variable point;
 
 procedure start
 begin
-    if ((script_action == 21) or (script_action == 3)) then begin
+    if ((script_action == look_at_proc) or (script_action == description_proc)) then begin
         script_overrides;
         display_msg(message_str(0, 100));
     end
     else begin
-        if (script_action == 6) then begin
+        if (script_action == use_proc) then begin
             call point_run();
         end
         else begin
-            if (script_action == 11) then begin
+            if (script_action == talk_proc) then begin
                 script_overrides;
                 call point_run();
             end
@@ -77,4 +77,3 @@ begin
         end
     end
 end
-
