diff --git a/Games/Fallout1.Steam/Original/MANGLED.ssl b/Games/Fallout1.Steam/ReDefine/MANGLED.ssl
index eb8ed05..92e8f58 100644
--- a/Games/Fallout1.Steam/Original/MANGLED.ssl
+++ b/Games/Fallout1.Steam/ReDefine/MANGLED.ssl
@@ -5,14 +5,14 @@ procedure dead00;
 procedure start
 begin
     variable LVar0 := 0;
-    if (script_action == 3) then begin
+    if (script_action == description_proc) then begin
         script_overrides;
         call dead00();
     end
     else begin
-        if (script_action == 21) then begin
+        if (script_action == look_at_proc) then begin
             script_overrides;
-            display_msg(message_str(83, 100));
+            display_mstr(100);
         end
     end
 end
@@ -20,7 +20,6 @@ end
 procedure dead00
 begin
     if (is_success(do_check(dude_obj, 4, 0))) then begin
-        display_msg(message_str(83, 101));
+        display_mstr(101);
     end
 end
-
