diff --git a/Games/Fallout1.Steam/Original/MARNEY.ssl b/Games/Fallout1.Steam/ReDefine/MARNEY.ssl
index aa79dbd..ab9eeb5 100644
--- a/Games/Fallout1.Steam/Original/MARNEY.ssl
+++ b/Games/Fallout1.Steam/ReDefine/MARNEY.ssl
@@ -19,24 +19,24 @@ variable initialized;
 procedure start
 begin
     if (not(initialized)) then begin
-        critter_add_trait(self_obj, 1, 6, 46);
-        critter_add_trait(self_obj, 1, 5, 34);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 46);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_FOLLOWER_LEADER);
         initialized := 1;
     end
     else begin
-        if (script_action == 12) then begin
+        if (script_action == critter_proc) then begin
             call critter_p_proc();
         end
         else begin
-            if (script_action == 18) then begin
+            if (script_action == destroy_proc) then begin
                 call destroy_p_proc();
             end
             else begin
-                if (script_action == 21) then begin
+                if (script_action == look_at_proc) then begin
                     call look_at_p_proc();
                 end
                 else begin
-                    if (script_action == 11) then begin
+                    if (script_action == talk_proc) then begin
                         call talk_p_proc();
                     end
                 end
@@ -47,7 +47,7 @@ end
 
 procedure critter_p_proc
 begin
-    if (global_var(129) == 2) then begin
+    if (global_var(GVAR_DESTROY_FOLLOWERS) == 2) then begin
         kill_critter(self_obj, 59);
     end
 end
@@ -55,17 +55,17 @@ end
 procedure destroy_p_proc
 begin
     if (source_obj == dude_obj) then begin
-        if (((global_var(160) + global_var(159)) >= 25) and ((global_var(159) > (2 * global_var(160))) or (global_var(156) == 1))) then begin
-            set_global_var(156, 1);
-            set_global_var(157, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
+            set_global_var(GVAR_BERSERKER_REPUTATION, 1);
+            set_global_var(GVAR_CHAMPION_REPUTATION, 0);
         end
-        if (((global_var(160) + global_var(159)) >= 25) and ((global_var(160) > (3 * global_var(159))) or (global_var(157) == 1))) then begin
-            set_global_var(157, 1);
-            set_global_var(156, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(160) > (3 * global_var(GVAR_GOOD_MONSTER))) or (global_var(GVAR_CHAMPION_REPUTATION) == 1))) then begin
+            set_global_var(GVAR_CHAMPION_REPUTATION, 1);
+            set_global_var(GVAR_BERSERKER_REPUTATION, 0);
         end
-        set_global_var(159, global_var(159) + 1);
-        if ((global_var(159) % 2) == 0) then begin
-            set_global_var(155, global_var(155) - 1);
+        set_global_var(GVAR_GOOD_MONSTER, global_var(GVAR_GOOD_MONSTER) + 1);
+        if ((global_var(GVAR_GOOD_MONSTER) % 2) == 0) then begin
+            set_global_var(GVAR_PLAYER_REPUATION, global_var(GVAR_PLAYER_REPUATION) - 1);
         end
     end
 end
@@ -75,18 +75,18 @@ begin
     script_overrides;
     if (local_var(0)) then begin
         if ((game_time_hour >= 1900) or (game_time_hour < 600)) then begin
-            display_msg(message_str(270, 100));
+            display_mstr(100);
         end
         else begin
-            display_msg(message_str(270, 101));
+            display_mstr(101);
         end
     end
     else begin
         if ((game_time_hour >= 1900) or (game_time_hour < 600)) then begin
-            display_msg(message_str(270, 102));
+            display_mstr(102);
         end
         else begin
-            display_msg(message_str(270, 103));
+            display_mstr(103);
         end
     end
 end
@@ -104,25 +104,25 @@ end
 procedure Marney00
 begin
     variable LVar0 := 0;
-    LVar0 := get_critter_stat(dude_obj, 7) - get_critter_stat(dude_obj, 35);
-    gsay_reply(270, 104);
-    giq_option(-3, 270, 105, Marney01, 50);
-    giq_option(4, 270, 125, MarneyEnd, 50);
+    LVar0 := get_critter_stat(dude_obj, STAT_max_hp) - get_critter_stat(dude_obj, STAT_current_hp);
+    Reply(104);
+    giq_option(-3, NAME, 105, Marney01, NEUTRAL_REACTION);
+    giq_option(4, NAME, 125, MarneyEnd, NEUTRAL_REACTION);
     if (LVar0 > 0) then begin
-        giq_option(4, 270, 106, Marney01, 50);
+        giq_option(4, NAME, 106, Marney01, NEUTRAL_REACTION);
     end
     if (not(local_var(2))) then begin
-        giq_option(5, 270, 107, Marney06, 50);
+        giq_option(5, NAME, 107, Marney06, NEUTRAL_REACTION);
     end
 end
 
 procedure Marney01
 begin
-    gsay_reply(270, 108);
-    giq_option(-3, 270, 109, Marney02, 50);
-    giq_option(-3, 270, 110, Marney05, 50);
-    giq_option(4, 270, 111, Marney02, 50);
-    giq_option(4, 270, 112, Marney05, 50);
+    Reply(108);
+    giq_option(-3, NAME, 109, Marney02, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 110, Marney05, NEUTRAL_REACTION);
+    giq_option(4, NAME, 111, Marney02, NEUTRAL_REACTION);
+    giq_option(4, NAME, 112, Marney05, NEUTRAL_REACTION);
 end
 
 procedure Marney02
@@ -130,21 +130,21 @@ begin
     variable LVar0 := 0;
     gfade_out(600);
     gfade_in(600);
-    LVar0 := get_critter_stat(dude_obj, 7) - get_critter_stat(dude_obj, 35);
+    LVar0 := get_critter_stat(dude_obj, STAT_max_hp) - get_critter_stat(dude_obj, STAT_current_hp);
     game_time_advance(game_ticks(60 * 20 * LVar0));
     critter_heal(dude_obj, LVar0);
-    gsay_message(270, 113, 50);
+    gsay_message(NAME, 113, NEUTRAL_REACTION);
     if (not(local_var(1))) then begin
-        giq_option(4, 270, 114, Marney03, 50);
+        giq_option(4, NAME, 114, Marney03, NEUTRAL_REACTION);
     end
-    giq_option(4, 270, 115, MarneyEnd, 50);
+    giq_option(4, NAME, 115, MarneyEnd, NEUTRAL_REACTION);
 end
 
 procedure Marney03
 begin
-    gsay_reply(270, 116);
-    giq_option(4, 270, 117, Marney04, 50);
-    giq_option(4, 270, 118, Marney05, 50);
+    Reply(116);
+    giq_option(4, NAME, 117, Marney04, NEUTRAL_REACTION);
+    giq_option(4, NAME, 118, Marney05, NEUTRAL_REACTION);
 end
 
 procedure Marney04
@@ -152,21 +152,21 @@ begin
     gfade_out(600);
     gfade_in(600);
     game_time_advance(game_ticks(18000));
-    critter_mod_skill(dude_obj, 6, 15);
+    critter_mod_skill(dude_obj, SKILL_FIRST_AID, 15);
     set_local_var(1, 1);
-    gsay_message(270, 119, 50);
+    gsay_message(NAME, 119, NEUTRAL_REACTION);
 end
 
 procedure Marney05
 begin
-    gsay_message(270, 120, 50);
+    gsay_message(NAME, 120, NEUTRAL_REACTION);
 end
 
 procedure Marney06
 begin
-    gsay_reply(270, 121);
-    giq_option(5, 270, 122, Marney07, 50);
-    giq_option(5, 270, 123, Marney05, 50);
+    Reply(121);
+    giq_option(5, NAME, 122, Marney07, NEUTRAL_REACTION);
+    giq_option(5, NAME, 123, Marney05, NEUTRAL_REACTION);
 end
 
 procedure Marney07
@@ -174,12 +174,11 @@ begin
     gfade_out(600);
     gfade_in(600);
     game_time_advance(game_ticks(25200));
-    critter_mod_skill(dude_obj, 7, 15);
+    critter_mod_skill(dude_obj, SKILL_DOCTOR, 15);
     set_local_var(2, 1);
-    gsay_message(270, 124, 50);
+    gsay_message(NAME, 124, NEUTRAL_REACTION);
 end
 
 procedure MarneyEnd
 begin
 end
-
