diff --git a/Games/Fallout1.Steam/Original/MAT.ssl b/Games/Fallout1.Steam/ReDefine/MAT.ssl
index 18fa497..0d7cb44 100644
--- a/Games/Fallout1.Steam/Original/MAT.ssl
+++ b/Games/Fallout1.Steam/ReDefine/MAT.ssl
@@ -50,34 +50,34 @@ procedure start
 begin
     if (Only_Once) then begin
         Only_Once := 0;
-        critter_add_trait(self_obj, 1, 6, 64);
-        critter_add_trait(self_obj, 1, 5, 50);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 64);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_MERCHANT);
     end
-    if (script_action == 21) then begin
+    if (script_action == look_at_proc) then begin
         call look_at_p_proc();
     end
     else begin
-        if (script_action == 4) then begin
+        if (script_action == pickup_proc) then begin
             call pickup_p_proc();
         end
         else begin
-            if (script_action == 11) then begin
+            if (script_action == talk_proc) then begin
                 call talk_p_proc();
             end
             else begin
-                if (script_action == 12) then begin
+                if (script_action == critter_proc) then begin
                     call critter_p_proc();
                 end
                 else begin
-                    if (script_action == 18) then begin
+                    if (script_action == destroy_proc) then begin
                         call destroy_p_proc();
                     end
                     else begin
-                        if (script_action == 22) then begin
+                        if (script_action == timed_event_proc) then begin
                             call timed_event_p_proc();
                         end
                         else begin
-                            if (script_action == 15) then begin
+                            if (script_action == map_enter_proc) then begin
                                 call map_enter_p_proc();
                             end
                         end
@@ -97,7 +97,7 @@ procedure critter_p_proc
 begin
     if (hostile) then begin
         hostile := 0;
-        attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+        attack(dude_obj);
     end
     if ((game_time_hour >= 600) and (game_time_hour < 700) or ((game_time_hour >= 700) and (game_time_hour < 1800))) then begin
         call WakeUpCall();
@@ -135,17 +135,17 @@ end
 procedure destroy_p_proc
 begin
     if (source_obj == dude_obj) then begin
-        if (((global_var(160) + global_var(159)) >= 25) and ((global_var(159) > (2 * global_var(160))) or (global_var(156) == 1))) then begin
-            set_global_var(156, 1);
-            set_global_var(157, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
+            set_global_var(GVAR_BERSERKER_REPUTATION, 1);
+            set_global_var(GVAR_CHAMPION_REPUTATION, 0);
         end
-        if (((global_var(160) + global_var(159)) >= 25) and ((global_var(160) > (3 * global_var(159))) or (global_var(157) == 1))) then begin
-            set_global_var(157, 1);
-            set_global_var(156, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(160) > (3 * global_var(GVAR_GOOD_MONSTER))) or (global_var(GVAR_CHAMPION_REPUTATION) == 1))) then begin
+            set_global_var(GVAR_CHAMPION_REPUTATION, 1);
+            set_global_var(GVAR_BERSERKER_REPUTATION, 0);
         end
-        set_global_var(159, global_var(159) + 1);
-        if ((global_var(159) % 2) == 0) then begin
-            set_global_var(155, global_var(155) - 1);
+        set_global_var(GVAR_GOOD_MONSTER, global_var(GVAR_GOOD_MONSTER) + 1);
+        if ((global_var(GVAR_GOOD_MONSTER) % 2) == 0) then begin
+            set_global_var(GVAR_PLAYER_REPUATION, global_var(GVAR_PLAYER_REPUATION) - 1);
         end
     end
 end
@@ -153,7 +153,7 @@ end
 procedure look_at_p_proc
 begin
     script_overrides;
-    display_msg(message_str(609, 200));
+    display_msg(message_str(SCRIPT_HGENCVAN, 200));
 end
 
 procedure timed_event_p_proc
@@ -178,7 +178,7 @@ begin
     variable LVar0 := 0;
     LVar0 := obj_pid(source_obj);
     if (party_member_obj(LVar0) != 0) then begin
-        set_global_var(248, 1);
+        set_global_var(GVAR_ENEMY_HUB, 1);
     end
 end
 
@@ -266,7 +266,7 @@ begin
         Sleeping := 1;
         SetDayNight := 0;
         if (random(0, 100) >= 90) then begin
-            float_msg(self_obj, message_str(609, 201), 2);
+            float_msg(self_obj, message_str(SCRIPT_HGENCVAN, 201), FLOAT_MSG_RED);
         end
     end
 end
@@ -279,7 +279,7 @@ begin
         set_obj_visibility(self_obj, 0);
         add_timer_event(self_obj, game_ticks(1), 1);
         if (random(0, 100) >= 90) then begin
-            float_msg(self_obj, message_str(609, 202), 2);
+            float_msg(self_obj, message_str(SCRIPT_HGENCVAN, 202), FLOAT_MSG_RED);
         end
     end
 end
@@ -304,92 +304,92 @@ end
 
 procedure Mat1
 begin
-    gsay_reply(609, 203);
-    giq_option(-3, 609, 204, Mat2, 50);
-    giq_option(4, 609, 205, Mat3, 50);
-    giq_option(4, 609, 206, Mat4, 50);
-    giq_option(4, 609, 207, Mat5, 50);
-    giq_option(4, 609, 208, MatEnd, 50);
-    if (((global_var(160) + global_var(159)) >= 25) and ((global_var(159) > (2 * global_var(160))) or (global_var(156) == 1))) then begin
-        giq_option(4, 609, 209, MatCombat, 51);
+    gsay_reply(SCRIPT_HGENCVAN, 203);
+    giq_option(-3, SCRIPT_HGENCVAN, 204, Mat2, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_HGENCVAN, 205, Mat3, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_HGENCVAN, 206, Mat4, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_HGENCVAN, 207, Mat5, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_HGENCVAN, 208, MatEnd, NEUTRAL_REACTION);
+    if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
+        giq_option(4, SCRIPT_HGENCVAN, 209, MatCombat, BAD_REACTION);
     end
 end
 
 procedure Mat2
 begin
-    gsay_message(609, 210, 50);
+    gsay_message(SCRIPT_HGENCVAN, 210, NEUTRAL_REACTION);
 end
 
 procedure Mat3
 begin
-    gsay_reply(609, 211);
-    giq_option(4, 609, 212, Mat4, 50);
-    giq_option(4, 609, 213, Mat5, 50);
-    giq_option(4, 609, 214, Mat6, 50);
-    giq_option(4, 609, 215, Mat7, 50);
-    giq_option(4, 609, 216, MatEnd, 50);
+    gsay_reply(SCRIPT_HGENCVAN, 211);
+    giq_option(4, SCRIPT_HGENCVAN, 212, Mat4, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_HGENCVAN, 213, Mat5, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_HGENCVAN, 214, Mat6, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_HGENCVAN, 215, Mat7, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_HGENCVAN, 216, MatEnd, NEUTRAL_REACTION);
 end
 
 procedure Mat4
 begin
-    gsay_reply(609, 217);
-    giq_option(4, 609, 218, Mat5, 50);
-    giq_option(4, 609, 219, MatEnd, 50);
+    gsay_reply(SCRIPT_HGENCVAN, 217);
+    giq_option(4, SCRIPT_HGENCVAN, 218, Mat5, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_HGENCVAN, 219, MatEnd, NEUTRAL_REACTION);
 end
 
 procedure Mat5
 begin
-    if (global_var(71) == 0) then begin
-        set_global_var(71, 1);
+    if (global_var(GVAR_JUNKTOWN) == 0) then begin
+        set_global_var(GVAR_JUNKTOWN, 1);
     end
-    if (global_var(74) == 0) then begin
-        set_global_var(74, 1);
+    if (global_var(GVAR_BROTHERHOOD_OF_STEEL) == 0) then begin
+        set_global_var(GVAR_BROTHERHOOD_OF_STEEL, 1);
     end
-    if (global_var(68) == 0) then begin
-        set_global_var(68, 1);
+    if (global_var(GVAR_SHADY_SANDS) == 0) then begin
+        set_global_var(GVAR_SHADY_SANDS, 1);
     end
-    if (global_var(72) == 0) then begin
-        set_global_var(72, 1);
+    if (global_var(GVAR_NECROPOLIS) == 0) then begin
+        set_global_var(GVAR_NECROPOLIS, 1);
     end
-    if (global_var(76) == 0) then begin
-        set_global_var(76, 1);
+    if (global_var(GVAR_THE_GLOW) == 0) then begin
+        set_global_var(GVAR_THE_GLOW, 1);
     end
-    gsay_reply(609, 220);
+    gsay_reply(SCRIPT_HGENCVAN, 220);
     if (CaravanAgain == 1) then begin
-        giq_option(4, 609, 221, Mat8, 50);
+        giq_option(4, SCRIPT_HGENCVAN, 221, Mat8, NEUTRAL_REACTION);
         CaravanAgain := 0;
     end
-    giq_option(4, 609, 222, Mat6, 50);
-    giq_option(4, 609, 223, Mat3, 50);
-    giq_option(4, 609, 224, Mat7, 50);
-    giq_option(4, 609, 225, MatEnd, 50);
+    giq_option(4, SCRIPT_HGENCVAN, 222, Mat6, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_HGENCVAN, 223, Mat3, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_HGENCVAN, 224, Mat7, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_HGENCVAN, 225, MatEnd, NEUTRAL_REACTION);
 end
 
 procedure Mat6
 begin
-    gsay_reply(609, 226);
-    giq_option(4, 609, 227, Mat4, 50);
-    giq_option(4, 609, 228, Mat7, 50);
-    giq_option(4, 609, 229, MatEnd, 50);
+    gsay_reply(SCRIPT_HGENCVAN, 226);
+    giq_option(4, SCRIPT_HGENCVAN, 227, Mat4, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_HGENCVAN, 228, Mat7, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_HGENCVAN, 229, MatEnd, NEUTRAL_REACTION);
 end
 
 procedure Mat7
 begin
-    gsay_reply(609, 230);
-    giq_option(4, 609, 231, Mat3, 50);
-    giq_option(4, 609, 232, Mat4, 50);
-    giq_option(4, 609, 233, Mat5, 50);
-    giq_option(4, 609, 234, MatEnd, 50);
+    gsay_reply(SCRIPT_HGENCVAN, 230);
+    giq_option(4, SCRIPT_HGENCVAN, 231, Mat3, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_HGENCVAN, 232, Mat4, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_HGENCVAN, 233, Mat5, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_HGENCVAN, 234, MatEnd, NEUTRAL_REACTION);
 end
 
 procedure Mat8
 begin
-    gsay_reply(609, 235);
-    if (((global_var(160) + global_var(159)) >= 25) and ((global_var(159) > (2 * global_var(160))) or (global_var(156) == 1))) then begin
-        giq_option(4, 609, 236, MatCombat, 51);
+    gsay_reply(SCRIPT_HGENCVAN, 235);
+    if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
+        giq_option(4, SCRIPT_HGENCVAN, 236, MatCombat, BAD_REACTION);
     end
-    giq_option(4, 609, 237, Mat7, 50);
-    giq_option(4, 609, 238, MatEnd, 50);
+    giq_option(4, SCRIPT_HGENCVAN, 237, Mat7, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_HGENCVAN, 238, MatEnd, NEUTRAL_REACTION);
 end
 
 procedure get_reaction
@@ -398,31 +398,31 @@ begin
         set_local_var(0, 50);
         set_local_var(1, 2);
         set_local_var(2, 1);
-        set_local_var(0, local_var(0) + (5 * get_critter_stat(dude_obj, 3)) - 25);
-        set_local_var(0, local_var(0) + (10 * has_trait(0, dude_obj, 10)));
-        if (has_trait(0, dude_obj, 39)) then begin
-            if (global_var(155) > 0) then begin
-                set_local_var(0, local_var(0) + global_var(155));
+        set_local_var(0, local_var(0) + (5 * dude_charisma) - 25);
+        set_local_var(0, local_var(0) + (10 * has_trait(TRAIT_PERK, dude_obj, PERK_presence)));
+        if (has_trait(TRAIT_PERK, dude_obj, PERK_cult_of_personality)) then begin
+            if (global_var(GVAR_PLAYER_REPUATION) > 0) then begin
+                set_local_var(0, local_var(0) + global_var(GVAR_PLAYER_REPUATION));
             end
             else begin
-                set_local_var(0, local_var(0) - global_var(155));
+                set_local_var(0, local_var(0) - global_var(GVAR_PLAYER_REPUATION));
             end
         end
         else begin
             if (local_var(3) == 1) then begin
-                set_local_var(0, local_var(0) - global_var(155));
+                set_local_var(0, local_var(0) - global_var(GVAR_PLAYER_REPUATION));
             end
             else begin
-                set_local_var(0, local_var(0) + global_var(155));
+                set_local_var(0, local_var(0) + global_var(GVAR_PLAYER_REPUATION));
             end
         end
-        if (global_var(158) > 2) then begin
+        if (global_var(GVAR_CHILDKILLER_REPUATION) > 2) then begin
             set_local_var(0, local_var(0) - 30);
         end
-        if (((global_var(160) + global_var(159)) >= 25) and ((global_var(160) > (3 * global_var(159))) or (global_var(157) == 1))) then begin
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(160) > (3 * global_var(GVAR_GOOD_MONSTER))) or (global_var(GVAR_CHAMPION_REPUTATION) == 1))) then begin
             set_local_var(0, local_var(0) + 20);
         end
-        if (((global_var(160) + global_var(159)) >= 25) and ((global_var(159) > (2 * global_var(160))) or (global_var(156) == 1))) then begin
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
             set_local_var(0, local_var(0) - 20);
         end
         call ReactToLevel();
@@ -515,6 +515,5 @@ end
 
 procedure Goodbyes
 begin
-    exit_line := message_str(634, random(100, 105));
+    exit_line := message_str(SCRIPT_MODREACT, random(100, 105));
 end
-
