diff --git a/Games/Fallout1.Steam/Original/MAXGRD.ssl b/Games/Fallout1.Steam/ReDefine/MAXGRD.ssl
index d0ab240..e8da0c1 100644
--- a/Games/Fallout1.Steam/Original/MAXGRD.ssl
+++ b/Games/Fallout1.Steam/ReDefine/MAXGRD.ssl
@@ -31,26 +31,26 @@ procedure start
 begin
     if (only_once) then begin
         only_once := 0;
-        critter_add_trait(self_obj, 1, 6, 44);
-        critter_add_trait(self_obj, 1, 5, 65);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 44);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_BROTHERHOOD_PALADIN);
     end
-    if (script_action == 21) then begin
+    if (script_action == look_at_proc) then begin
         call look_at_p_proc();
     end
     else begin
-        if (script_action == 4) then begin
+        if (script_action == pickup_proc) then begin
             call pickup_p_proc();
         end
         else begin
-            if (script_action == 11) then begin
+            if (script_action == talk_proc) then begin
                 call talk_p_proc();
             end
             else begin
-                if (script_action == 12) then begin
+                if (script_action == critter_proc) then begin
                     call critter_p_proc();
                 end
                 else begin
-                    if (script_action == 18) then begin
+                    if (script_action == destroy_proc) then begin
                         call destroy_p_proc();
                     end
                 end
@@ -66,16 +66,16 @@ end
 
 procedure critter_p_proc
 begin
-    if (global_var(250)) then begin
+    if (global_var(GVAR_ENEMY_BROTHERHOOD)) then begin
         hostile := 1;
     end
     if (tile_distance_objs(self_obj, dude_obj) > 12) then begin
         hostile := 0;
     end
     if (hostile) then begin
-        set_global_var(250, 1);
+        set_global_var(GVAR_ENEMY_BROTHERHOOD, 1);
         hostile := 0;
-        attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+        attack(dude_obj);
     end
 end
 
@@ -90,39 +90,39 @@ procedure talk_p_proc
 begin
     Pick := random(1, 9);
     if (Pick == 1) then begin
-        message := message_str(618, 101);
+        message := message_str(SCRIPT_HALLGRD, 101);
     end
     else begin
         if (Pick == 2) then begin
-            message := message_str(618, 102);
+            message := message_str(SCRIPT_HALLGRD, 102);
         end
         else begin
             if (Pick == 3) then begin
-                message := message_str(618, 103);
+                message := message_str(SCRIPT_HALLGRD, 103);
             end
             else begin
                 if (Pick == 4) then begin
-                    message := message_str(618, 104);
+                    message := message_str(SCRIPT_HALLGRD, 104);
                 end
                 else begin
                     if (Pick == 5) then begin
-                        message := message_str(618, 105);
+                        message := message_str(SCRIPT_HALLGRD, 105);
                     end
                     else begin
                         if (Pick == 6) then begin
-                            message := message_str(618, 106);
+                            message := message_str(SCRIPT_HALLGRD, 106);
                         end
                         else begin
                             if (Pick == 7) then begin
-                                message := message_str(618, 107);
+                                message := message_str(SCRIPT_HALLGRD, 107);
                             end
                             else begin
                                 if (Pick == 8) then begin
-                                    message := message_str(618, 108);
+                                    message := message_str(SCRIPT_HALLGRD, 108);
                                 end
                                 else begin
                                     if (Pick == 9) then begin
-                                        message := message_str(618, 110) + proto_data(obj_pid(dude_obj), 1) + message_str(618, 111);
+                                        message := message_str(SCRIPT_HALLGRD, 110) + proto_data(obj_pid(dude_obj), 1) + message_str(SCRIPT_HALLGRD, 111);
                                     end
                                 end
                             end
@@ -132,25 +132,25 @@ begin
             end
         end
     end
-    float_msg(self_obj, message, 0);
+    float_msg(self_obj, message, FLOAT_MSG_NORMAL);
 end
 
 procedure destroy_p_proc
 begin
-    set_global_var(250, 1);
+    set_global_var(GVAR_ENEMY_BROTHERHOOD, 1);
     rm_timer_event(self_obj);
     if (source_obj == dude_obj) then begin
-        if (((global_var(160) + global_var(159)) >= 25) and ((global_var(159) > (2 * global_var(160))) or (global_var(156) == 1))) then begin
-            set_global_var(156, 1);
-            set_global_var(157, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
+            set_global_var(GVAR_BERSERKER_REPUTATION, 1);
+            set_global_var(GVAR_CHAMPION_REPUTATION, 0);
         end
-        if (((global_var(160) + global_var(159)) >= 25) and ((global_var(160) > (3 * global_var(159))) or (global_var(157) == 1))) then begin
-            set_global_var(157, 1);
-            set_global_var(156, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(160) > (3 * global_var(GVAR_GOOD_MONSTER))) or (global_var(GVAR_CHAMPION_REPUTATION) == 1))) then begin
+            set_global_var(GVAR_CHAMPION_REPUTATION, 1);
+            set_global_var(GVAR_BERSERKER_REPUTATION, 0);
         end
-        set_global_var(159, global_var(159) + 1);
-        if ((global_var(159) % 2) == 0) then begin
-            set_global_var(155, global_var(155) - 1);
+        set_global_var(GVAR_GOOD_MONSTER, global_var(GVAR_GOOD_MONSTER) + 1);
+        if ((global_var(GVAR_GOOD_MONSTER) % 2) == 0) then begin
+            set_global_var(GVAR_PLAYER_REPUATION, global_var(GVAR_PLAYER_REPUATION) - 1);
         end
     end
 end
@@ -158,7 +158,7 @@ end
 procedure look_at_p_proc
 begin
     script_overrides;
-    display_msg(message_str(618, 100));
+    display_msg(message_str(SCRIPT_HALLGRD, 100));
 end
 
 procedure get_reaction
@@ -167,31 +167,31 @@ begin
         set_local_var(0, 50);
         set_local_var(1, 2);
         set_local_var(2, 1);
-        set_local_var(0, local_var(0) + (5 * get_critter_stat(dude_obj, 3)) - 25);
-        set_local_var(0, local_var(0) + (10 * has_trait(0, dude_obj, 10)));
-        if (has_trait(0, dude_obj, 39)) then begin
-            if (global_var(155) > 0) then begin
-                set_local_var(0, local_var(0) + global_var(155));
+        set_local_var(0, local_var(0) + (5 * dude_charisma) - 25);
+        set_local_var(0, local_var(0) + (10 * has_trait(TRAIT_PERK, dude_obj, PERK_presence)));
+        if (has_trait(TRAIT_PERK, dude_obj, PERK_cult_of_personality)) then begin
+            if (global_var(GVAR_PLAYER_REPUATION) > 0) then begin
+                set_local_var(0, local_var(0) + global_var(GVAR_PLAYER_REPUATION));
             end
             else begin
-                set_local_var(0, local_var(0) - global_var(155));
+                set_local_var(0, local_var(0) - global_var(GVAR_PLAYER_REPUATION));
             end
         end
         else begin
             if (local_var(3) == 1) then begin
-                set_local_var(0, local_var(0) - global_var(155));
+                set_local_var(0, local_var(0) - global_var(GVAR_PLAYER_REPUATION));
             end
             else begin
-                set_local_var(0, local_var(0) + global_var(155));
+                set_local_var(0, local_var(0) + global_var(GVAR_PLAYER_REPUATION));
             end
         end
-        if (global_var(158) > 2) then begin
+        if (global_var(GVAR_CHILDKILLER_REPUATION) > 2) then begin
             set_local_var(0, local_var(0) - 30);
         end
-        if (((global_var(160) + global_var(159)) >= 25) and ((global_var(160) > (3 * global_var(159))) or (global_var(157) == 1))) then begin
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(160) > (3 * global_var(GVAR_GOOD_MONSTER))) or (global_var(GVAR_CHAMPION_REPUTATION) == 1))) then begin
             set_local_var(0, local_var(0) + 20);
         end
-        if (((global_var(160) + global_var(159)) >= 25) and ((global_var(159) > (2 * global_var(160))) or (global_var(156) == 1))) then begin
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
             set_local_var(0, local_var(0) - 20);
         end
         call ReactToLevel();
@@ -284,6 +284,5 @@ end
 
 procedure Goodbyes
 begin
-    exit_line := message_str(634, random(100, 105));
+    exit_line := message_str(SCRIPT_MODREACT, random(100, 105));
 end
-
