diff --git a/Games/Fallout1.Steam/Original/MAXSON.ssl b/Games/Fallout1.Steam/ReDefine/MAXSON.ssl
index 082af87..fd83203 100644
--- a/Games/Fallout1.Steam/Original/MAXSON.ssl
+++ b/Games/Fallout1.Steam/ReDefine/MAXSON.ssl
@@ -73,26 +73,26 @@ procedure start
 begin
     if (Only_Once) then begin
         Only_Once := 0;
-        critter_add_trait(self_obj, 1, 6, 44);
-        critter_add_trait(self_obj, 1, 5, 80);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 44);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_BOS_ELDERS);
     end
-    if (script_action == 21) then begin
+    if (script_action == look_at_proc) then begin
         call look_at_p_proc();
     end
     else begin
-        if (script_action == 4) then begin
+        if (script_action == pickup_proc) then begin
             call pickup_p_proc();
         end
         else begin
-            if (script_action == 11) then begin
+            if (script_action == talk_proc) then begin
                 call talk_p_proc();
             end
             else begin
-                if (script_action == 12) then begin
+                if (script_action == critter_proc) then begin
                     call critter_p_proc();
                 end
                 else begin
-                    if (script_action == 18) then begin
+                    if (script_action == destroy_proc) then begin
                         call destroy_p_proc();
                     end
                 end
@@ -108,16 +108,16 @@ end
 
 procedure critter_p_proc
 begin
-    if (global_var(250)) then begin
+    if (global_var(GVAR_ENEMY_BROTHERHOOD)) then begin
         hostile := 1;
     end
     if (tile_distance_objs(self_obj, dude_obj) > 12) then begin
         hostile := 0;
     end
     if (hostile) then begin
-        set_global_var(250, 1);
+        set_global_var(GVAR_ENEMY_BROTHERHOOD, 1);
         hostile := 0;
-        attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+        attack(dude_obj);
     end
 end
 
@@ -133,11 +133,11 @@ begin
     call get_reaction();
     start_gdialog(52, self_obj, 4, 12, 5);
     gsay_start;
-    if ((global_var(223) == 1) or (global_var(223) == 2)) then begin
+    if ((global_var(GVAR_BROTHERHOOD_INVASION) == 1) or (global_var(GVAR_BROTHERHOOD_INVASION) == 2)) then begin
         call Maxson34();
     end
     else begin
-        if (global_var(17) == 1) then begin
+        if (global_var(GVAR_VATS_BLOWN) == 1) then begin
             call Maxson22();
         end
         else begin
@@ -145,7 +145,7 @@ begin
                 call Maxson22();
             end
             else begin
-                if (global_var(78) == 2) then begin
+                if (global_var(GVAR_THE_VATS) == 2) then begin
                     call Maxson20();
                 end
                 else begin
@@ -170,7 +170,7 @@ begin
     if (local_var(8) == 1) then begin
         set_local_var(8, 2);
         temp := 1500;
-        display_msg(message_str(52, 351) + temp + message_str(52, 352));
+        display_msg(mstr(351) + temp + mstr(352));
         give_exp_points(temp);
     end
     if (Denounce_Player == 1) then begin
@@ -180,20 +180,20 @@ end
 
 procedure destroy_p_proc
 begin
-    set_global_var(250, 1);
+    set_global_var(GVAR_ENEMY_BROTHERHOOD, 1);
     rm_timer_event(self_obj);
     if (source_obj == dude_obj) then begin
-        if (((global_var(160) + global_var(159)) >= 25) and ((global_var(159) > (2 * global_var(160))) or (global_var(156) == 1))) then begin
-            set_global_var(156, 1);
-            set_global_var(157, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
+            set_global_var(GVAR_BERSERKER_REPUTATION, 1);
+            set_global_var(GVAR_CHAMPION_REPUTATION, 0);
         end
-        if (((global_var(160) + global_var(159)) >= 25) and ((global_var(160) > (3 * global_var(159))) or (global_var(157) == 1))) then begin
-            set_global_var(157, 1);
-            set_global_var(156, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(160) > (3 * global_var(GVAR_GOOD_MONSTER))) or (global_var(GVAR_CHAMPION_REPUTATION) == 1))) then begin
+            set_global_var(GVAR_CHAMPION_REPUTATION, 1);
+            set_global_var(GVAR_BERSERKER_REPUTATION, 0);
         end
-        set_global_var(159, global_var(159) + 1);
-        if ((global_var(159) % 2) == 0) then begin
-            set_global_var(155, global_var(155) - 1);
+        set_global_var(GVAR_GOOD_MONSTER, global_var(GVAR_GOOD_MONSTER) + 1);
+        if ((global_var(GVAR_GOOD_MONSTER) % 2) == 0) then begin
+            set_global_var(GVAR_PLAYER_REPUATION, global_var(GVAR_PLAYER_REPUATION) - 1);
         end
     end
 end
@@ -201,30 +201,30 @@ end
 procedure look_at_p_proc
 begin
     script_overrides;
-    display_msg(message_str(52, 100));
+    display_mstr(100);
 end
 
 procedure Maxson01
 begin
-    gsay_reply(52, 106);
-    giq_option(-3, 52, 333, Maxson32, 50);
-    if (global_var(106) == 2) then begin
-        giq_option(4, 52, 300, Maxson14, 50);
+    Reply(106);
+    giq_option(-3, NAME, 333, Maxson32, NEUTRAL_REACTION);
+    if (global_var(GVAR_MISSING_CARAVAN) == 2) then begin
+        giq_option(4, NAME, 300, Maxson14, NEUTRAL_REACTION);
     end
-    if (global_var(106) == 1) then begin
-        giq_option(4, 52, 301, Maxson06, 50);
+    if (global_var(GVAR_MISSING_CARAVAN) == 1) then begin
+        giq_option(4, NAME, 301, Maxson06, NEUTRAL_REACTION);
     end
-    giq_option(4, 52, 349, Maxson27, 50);
-    giq_option(4, 52, 350, Maxson18, 50);
+    giq_option(4, NAME, 349, Maxson27, NEUTRAL_REACTION);
+    giq_option(4, NAME, 350, Maxson18, NEUTRAL_REACTION);
 end
 
 procedure Maxson02
 begin
-    gsay_reply(52, 162);
-    giq_option(4, 52, 302, Maxson04, 50);
-    giq_option(4, 52, 303, Maxson09, 50);
-    giq_option(4, 52, 304, Maxson02a, 51);
-    giq_option(4, 52, 305, Maxson08, 50);
+    Reply(162);
+    giq_option(4, NAME, 302, Maxson04, NEUTRAL_REACTION);
+    giq_option(4, NAME, 303, Maxson09, NEUTRAL_REACTION);
+    giq_option(4, NAME, 304, Maxson02a, BAD_REACTION);
+    giq_option(4, NAME, 305, Maxson08, NEUTRAL_REACTION);
 end
 
 procedure Maxson02a
@@ -244,10 +244,10 @@ end
 
 procedure Maxson04
 begin
-    gsay_reply(52, 133);
-    giq_option(4, 52, 306, Maxson04a, 50);
-    giq_option(4, 52, 307, Maxson27, 50);
-    giq_option(4, 52, 308, MaxsonEnd, 50);
+    Reply(133);
+    giq_option(4, NAME, 306, Maxson04a, NEUTRAL_REACTION);
+    giq_option(4, NAME, 307, Maxson27, NEUTRAL_REACTION);
+    giq_option(4, NAME, 308, MaxsonEnd, NEUTRAL_REACTION);
 end
 
 procedure Maxson04a
@@ -258,35 +258,35 @@ end
 
 procedure Maxson05
 begin
-    gsay_reply(52, 276);
-    giq_option(4, 52, 309, MaxsonEnd, 50);
+    Reply(276);
+    giq_option(4, NAME, 309, MaxsonEnd, NEUTRAL_REACTION);
 end
 
 procedure Maxson06
 begin
-    gsay_reply(52, 122);
-    giq_option(4, 52, 310, Maxson07, 50);
-    giq_option(4, 52, 311, Maxson02, 50);
+    Reply(122);
+    giq_option(4, NAME, 310, Maxson07, NEUTRAL_REACTION);
+    giq_option(4, NAME, 311, Maxson02, NEUTRAL_REACTION);
 end
 
 procedure Maxson07
 begin
-    if (get_critter_stat(dude_obj, 34) == 0) then begin
-        gsay_reply(52, 355);
+    if (dude_gender == 0) then begin
+        Reply(355);
     end
     else begin
-        gsay_reply(52, 356);
+        Reply(356);
     end
-    giq_option(4, 52, 312, Maxson14, 50);
-    giq_option(4, 52, 313, Maxson13, 50);
+    giq_option(4, NAME, 312, Maxson14, NEUTRAL_REACTION);
+    giq_option(4, NAME, 313, Maxson13, NEUTRAL_REACTION);
 end
 
 procedure Maxson08
 begin
-    gsay_reply(52, 184);
-    giq_option(4, 52, 314, Maxson18, 50);
-    giq_option(4, 52, 315, Maxson04a, 50);
-    giq_option(4, 52, 316, Maxson08a, 50);
+    Reply(184);
+    giq_option(4, NAME, 314, Maxson18, NEUTRAL_REACTION);
+    giq_option(4, NAME, 315, Maxson04a, NEUTRAL_REACTION);
+    giq_option(4, NAME, 316, Maxson08a, NEUTRAL_REACTION);
 end
 
 procedure Maxson08a
@@ -301,10 +301,10 @@ end
 
 procedure Maxson09
 begin
-    gsay_reply(52, 223);
-    giq_option(4, 52, 317, Maxson09a, 51);
-    giq_option(4, 52, 318, Maxson18, 50);
-    giq_option(4, 52, 319, MaxsonEnd, 50);
+    Reply(223);
+    giq_option(4, NAME, 317, Maxson09a, BAD_REACTION);
+    giq_option(4, NAME, 318, Maxson18, NEUTRAL_REACTION);
+    giq_option(4, NAME, 319, MaxsonEnd, NEUTRAL_REACTION);
 end
 
 procedure Maxson09a
@@ -326,224 +326,224 @@ begin
         item_caps_adjust(dude_obj, 500);
     end
     call DownReact();
-    gsay_reply(52, 226);
-    giq_option(4, 52, 320, MaxsonEnd, 51);
+    Reply(226);
+    giq_option(4, NAME, 320, MaxsonEnd, BAD_REACTION);
 end
 
 procedure Maxson11
 begin
-    gsay_message(52, 229, 51);
+    gsay_message(NAME, 229, BAD_REACTION);
     Denounce_Player := 1;
 end
 
 procedure Maxson12
 begin
-    gsay_message(52, 229, 51);
+    gsay_message(NAME, 229, BAD_REACTION);
     Denounce_Player := 1;
 end
 
 procedure Maxson13
 begin
-    gsay_message(52, 231, 51);
+    gsay_message(NAME, 231, BAD_REACTION);
     Denounce_Player := 1;
 end
 
 procedure Maxson14
 begin
-    gsay_reply(52, 146);
-    giq_option(4, 52, 321, Maxson02, 50);
+    Reply(146);
+    giq_option(4, NAME, 321, Maxson02, NEUTRAL_REACTION);
 end
 
 procedure Maxson15
 begin
-    gsay_reply(52, 141);
-    giq_option(4, 52, 322, Maxson02, 50);
-    giq_option(4, 52, 323, Maxson16, 50);
+    Reply(141);
+    giq_option(4, NAME, 322, Maxson02, NEUTRAL_REACTION);
+    giq_option(4, NAME, 323, Maxson16, NEUTRAL_REACTION);
 end
 
 procedure Maxson16
 begin
-    gsay_reply(52, 223);
-    giq_option(4, 52, 324, Maxson17, 50);
-    giq_option(4, 52, 325, Maxson18, 50);
-    giq_option(4, 52, 326, Maxson17, 50);
+    Reply(223);
+    giq_option(4, NAME, 324, Maxson17, NEUTRAL_REACTION);
+    giq_option(4, NAME, 325, Maxson18, NEUTRAL_REACTION);
+    giq_option(4, NAME, 326, Maxson17, NEUTRAL_REACTION);
 end
 
 procedure Maxson17
 begin
-    gsay_reply(52, 168);
-    giq_option(4, 52, 327, MaxsonEnd, 51);
+    Reply(168);
+    giq_option(4, NAME, 327, MaxsonEnd, BAD_REACTION);
 end
 
 procedure Maxson18
 begin
-    gsay_message(52, 222, 50);
+    gsay_message(NAME, 222, NEUTRAL_REACTION);
 end
 
 procedure Maxson19
 begin
-    gsay_reply(52, 201);
-    giq_option(-3, 52, 333, Maxson32, 50);
-    if (global_var(106) == 2) then begin
-        giq_option(4, 52, 329, Maxson14, 50);
+    Reply(201);
+    giq_option(-3, NAME, 333, Maxson32, NEUTRAL_REACTION);
+    if (global_var(GVAR_MISSING_CARAVAN) == 2) then begin
+        giq_option(4, NAME, 329, Maxson14, NEUTRAL_REACTION);
     end
-    if (global_var(106) == 1) then begin
-        giq_option(4, 52, 330, Maxson06, 50);
+    if (global_var(GVAR_MISSING_CARAVAN) == 1) then begin
+        giq_option(4, NAME, 330, Maxson06, NEUTRAL_REACTION);
     end
-    giq_option(4, 52, 332, Maxson27, 50);
-    giq_option(4, 52, 331, Maxson18, 50);
+    giq_option(4, NAME, 332, Maxson27, NEUTRAL_REACTION);
+    giq_option(4, NAME, 331, Maxson18, NEUTRAL_REACTION);
 end
 
 procedure Maxson20
 begin
-    gsay_reply(52, 193);
-    giq_option(-3, 52, 333, Maxson32, 50);
-    giq_option(4, 52, 334, Maxson22, 50);
-    giq_option(4, 52, 335, Maxson27, 50);
-    giq_option(4, 52, 336, Maxson18, 50);
+    Reply(193);
+    giq_option(-3, NAME, 333, Maxson32, NEUTRAL_REACTION);
+    giq_option(4, NAME, 334, Maxson22, NEUTRAL_REACTION);
+    giq_option(4, NAME, 335, Maxson27, NEUTRAL_REACTION);
+    giq_option(4, NAME, 336, Maxson18, NEUTRAL_REACTION);
 end
 
 procedure Maxson21
 begin
-    gsay_reply(52, 201);
-    giq_option(-3, 52, 333, Maxson32, 50);
-    giq_option(4, 52, 334, Maxson22, 50);
-    giq_option(4, 52, 335, Maxson27, 50);
-    giq_option(4, 52, 336, Maxson18, 50);
+    Reply(201);
+    giq_option(-3, NAME, 333, Maxson32, NEUTRAL_REACTION);
+    giq_option(4, NAME, 334, Maxson22, NEUTRAL_REACTION);
+    giq_option(4, NAME, 335, Maxson27, NEUTRAL_REACTION);
+    giq_option(4, NAME, 336, Maxson18, NEUTRAL_REACTION);
 end
 
 procedure Maxson22
 begin
     temp := local_var(7);
-    gsay_reply(52, 205);
-    if ((temp == 0) and (global_var(78) != 2)) then begin
-        giq_option(4, 52, 337, Maxson18, 50);
+    Reply(205);
+    if ((temp == 0) and (global_var(GVAR_THE_VATS) != 2)) then begin
+        giq_option(4, NAME, 337, Maxson18, NEUTRAL_REACTION);
     end
     else begin
-        if ((temp == 0) and (global_var(78) == 2) and (global_var(79) == 2)) then begin
-            giq_option(4, 52, 339, Maxson23, 50);
+        if ((temp == 0) and (global_var(GVAR_THE_VATS) == 2) and (global_var(GVAR_MASTERS_LAIR) == 2)) then begin
+            giq_option(4, NAME, 339, Maxson23, NEUTRAL_REACTION);
         end
         else begin
-            if ((temp == 0) and (global_var(78) == 2) and (global_var(79) != 2)) then begin
-                giq_option(4, 52, 338, Maxson23, 50);
+            if ((temp == 0) and (global_var(GVAR_THE_VATS) == 2) and (global_var(GVAR_MASTERS_LAIR) != 2)) then begin
+                giq_option(4, NAME, 338, Maxson23, NEUTRAL_REACTION);
             end
         end
     end
-    if (global_var(17) == 1) then begin
-        giq_option(4, 52, 400, Maxson35, 50);
+    if (global_var(GVAR_VATS_BLOWN) == 1) then begin
+        giq_option(4, NAME, 400, Maxson35, NEUTRAL_REACTION);
     end
-    giq_option(4, 52, 335, Maxson27, 50);
-    giq_option(4, 52, 336, Maxson18, 50);
+    giq_option(4, NAME, 335, Maxson27, NEUTRAL_REACTION);
+    giq_option(4, NAME, 336, Maxson18, NEUTRAL_REACTION);
 end
 
 procedure Maxson23
 begin
     set_local_var(7, 1);
     set_local_var(8, 1);
-    gsay_reply(52, 213);
-    giq_option(4, 52, 340, Maxson24, 50);
+    Reply(213);
+    giq_option(4, NAME, 340, Maxson24, NEUTRAL_REACTION);
 end
 
 procedure Maxson24
 begin
     set_local_var(5, 1);
-    gsay_reply(52, 219);
-    giq_option(4, 52, 341, Maxson25, 50);
-    giq_option(4, 52, 342, Maxson33, 50);
+    Reply(219);
+    giq_option(4, NAME, 341, Maxson25, NEUTRAL_REACTION);
+    giq_option(4, NAME, 342, Maxson33, NEUTRAL_REACTION);
 end
 
 procedure Maxson25
 begin
     set_map_var(19, 1);
-    gsay_message(52, 221, 50);
+    gsay_message(NAME, 221, NEUTRAL_REACTION);
 end
 
 procedure Maxson26
 begin
-    gsay_reply(52, 210);
-    giq_option(4, 52, 343, Maxson23, 50);
+    Reply(210);
+    giq_option(4, NAME, 343, Maxson23, NEUTRAL_REACTION);
 end
 
 procedure Maxson27
 begin
-    gsay_reply(52, random(138, 140));
+    Reply(random(138, 140));
     if (local_var(6) == 0) then begin
-        giq_option(4, 52, 344, Maxson28, 50);
+        giq_option(4, NAME, 344, Maxson28, NEUTRAL_REACTION);
     end
-    giq_option(4, 52, 345, Maxson29, 50);
-    giq_option(4, 52, 346, Maxson30, 50);
-    giq_option(4, 52, 347, Maxson31, 50);
-    giq_option(4, 52, 348, Maxson18, 50);
+    giq_option(4, NAME, 345, Maxson29, NEUTRAL_REACTION);
+    giq_option(4, NAME, 346, Maxson30, NEUTRAL_REACTION);
+    giq_option(4, NAME, 347, Maxson31, NEUTRAL_REACTION);
+    giq_option(4, NAME, 348, Maxson18, NEUTRAL_REACTION);
 end
 
 procedure Maxson28
 begin
     variable LVar0 := 0;
     set_local_var(6, 1);
-    LVar0 := create_object_sid(216, 0, 0, -1);
+    LVar0 := create_object(216, 0, 0);
     add_obj_to_inven(dude_obj, LVar0);
-    gsay_reply(52, 186);
-    giq_option(4, 52, 349, Maxson27, 50);
-    giq_option(4, 52, 350, Maxson18, 50);
+    Reply(186);
+    giq_option(4, NAME, 349, Maxson27, NEUTRAL_REACTION);
+    giq_option(4, NAME, 350, Maxson18, NEUTRAL_REACTION);
 end
 
 procedure Maxson29
 begin
-    gsay_reply(52, 172);
-    giq_option(4, 52, 349, Maxson27, 50);
-    giq_option(4, 52, 350, Maxson18, 50);
+    Reply(172);
+    giq_option(4, NAME, 349, Maxson27, NEUTRAL_REACTION);
+    giq_option(4, NAME, 350, Maxson18, NEUTRAL_REACTION);
 end
 
 procedure Maxson30
 begin
-    gsay_reply(52, 244);
-    giq_option(4, 52, 349, Maxson27, 50);
-    giq_option(4, 52, 350, Maxson18, 50);
+    Reply(244);
+    giq_option(4, NAME, 349, Maxson27, NEUTRAL_REACTION);
+    giq_option(4, NAME, 350, Maxson18, NEUTRAL_REACTION);
 end
 
 procedure Maxson31
 begin
-    gsay_reply(52, 262);
-    giq_option(4, 52, 349, Maxson27, 50);
-    giq_option(4, 52, 350, Maxson18, 50);
+    Reply(262);
+    giq_option(4, NAME, 349, Maxson27, NEUTRAL_REACTION);
+    giq_option(4, NAME, 350, Maxson18, NEUTRAL_REACTION);
 end
 
 procedure Maxson32
 begin
-    gsay_reply(52, 275);
-    giq_option(4, 52, 349, Maxson27, 50);
-    giq_option(4, 52, 350, Maxson18, 50);
+    Reply(275);
+    giq_option(4, NAME, 349, Maxson27, NEUTRAL_REACTION);
+    giq_option(4, NAME, 350, Maxson18, NEUTRAL_REACTION);
 end
 
 procedure Maxson33
 begin
-    if (get_critter_stat(dude_obj, 34) == 0) then begin
-        gsay_reply(52, 357);
+    if (dude_gender == 0) then begin
+        Reply(357);
     end
     else begin
-        gsay_reply(52, 358);
+        Reply(358);
     end
-    giq_option(4, 52, 349, Maxson27, 50);
-    giq_option(4, 52, 350, Maxson18, 50);
+    giq_option(4, NAME, 349, Maxson27, NEUTRAL_REACTION);
+    giq_option(4, NAME, 350, Maxson18, NEUTRAL_REACTION);
 end
 
 procedure Maxson34
 begin
-    if (get_critter_stat(dude_obj, 34) == 0) then begin
-        gsay_reply(52, 359);
+    if (dude_gender == 0) then begin
+        Reply(359);
     end
     else begin
-        gsay_reply(52, 360);
+        Reply(360);
     end
-    giq_option(4, 52, 349, Maxson27, 50);
-    giq_option(4, 52, 350, Maxson18, 50);
+    giq_option(4, NAME, 349, Maxson27, NEUTRAL_REACTION);
+    giq_option(4, NAME, 350, Maxson18, NEUTRAL_REACTION);
 end
 
 procedure Maxson35
 begin
-    gsay_reply(52, 133);
-    giq_option(4, 52, 335, Maxson27, 50);
-    giq_option(4, 52, 336, Maxson18, 50);
+    Reply(133);
+    giq_option(4, NAME, 335, Maxson27, NEUTRAL_REACTION);
+    giq_option(4, NAME, 336, Maxson18, NEUTRAL_REACTION);
 end
 
 procedure MaxsonEnd
@@ -552,11 +552,11 @@ end
 
 procedure Remove_Player
 begin
-    set_global_var(108, 5);
-    set_global_var(583, 0);
-    set_global_var(584, 0);
-    set_global_var(585, 0);
-    set_global_var(586, 0);
+    set_global_var(GVAR_BECOME_AN_INITIATE, 5);
+    set_global_var(GVAR_MARK_BROTHER_2, 0);
+    set_global_var(GVAR_MARK_BROTHER_3, 0);
+    set_global_var(GVAR_MARK_BROTHER_4, 0);
+    set_global_var(GVAR_MARK_BROTHER_5, 0);
     load_map(13, 1);
 end
 
@@ -566,31 +566,31 @@ begin
         set_local_var(0, 50);
         set_local_var(1, 2);
         set_local_var(2, 1);
-        set_local_var(0, local_var(0) + (5 * get_critter_stat(dude_obj, 3)) - 25);
-        set_local_var(0, local_var(0) + (10 * has_trait(0, dude_obj, 10)));
-        if (has_trait(0, dude_obj, 39)) then begin
-            if (global_var(155) > 0) then begin
-                set_local_var(0, local_var(0) + global_var(155));
+        set_local_var(0, local_var(0) + (5 * dude_charisma) - 25);
+        set_local_var(0, local_var(0) + (10 * has_trait(TRAIT_PERK, dude_obj, PERK_presence)));
+        if (has_trait(TRAIT_PERK, dude_obj, PERK_cult_of_personality)) then begin
+            if (global_var(GVAR_PLAYER_REPUATION) > 0) then begin
+                set_local_var(0, local_var(0) + global_var(GVAR_PLAYER_REPUATION));
             end
             else begin
-                set_local_var(0, local_var(0) - global_var(155));
+                set_local_var(0, local_var(0) - global_var(GVAR_PLAYER_REPUATION));
             end
         end
         else begin
             if (local_var(3) == 1) then begin
-                set_local_var(0, local_var(0) - global_var(155));
+                set_local_var(0, local_var(0) - global_var(GVAR_PLAYER_REPUATION));
             end
             else begin
-                set_local_var(0, local_var(0) + global_var(155));
+                set_local_var(0, local_var(0) + global_var(GVAR_PLAYER_REPUATION));
             end
         end
-        if (global_var(158) > 2) then begin
+        if (global_var(GVAR_CHILDKILLER_REPUATION) > 2) then begin
             set_local_var(0, local_var(0) - 30);
         end
-        if (((global_var(160) + global_var(159)) >= 25) and ((global_var(160) > (3 * global_var(159))) or (global_var(157) == 1))) then begin
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(160) > (3 * global_var(GVAR_GOOD_MONSTER))) or (global_var(GVAR_CHAMPION_REPUTATION) == 1))) then begin
             set_local_var(0, local_var(0) + 20);
         end
-        if (((global_var(160) + global_var(159)) >= 25) and ((global_var(159) > (2 * global_var(160))) or (global_var(156) == 1))) then begin
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
             set_local_var(0, local_var(0) - 20);
         end
         call ReactToLevel();
@@ -683,6 +683,5 @@ end
 
 procedure Goodbyes
 begin
-    exit_line := message_str(634, random(100, 105));
+    exit_line := message_str(SCRIPT_MODREACT, random(100, 105));
 end
-
