diff --git a/Games/Fallout1.Steam/Original/MAXSON2.ssl b/Games/Fallout1.Steam/ReDefine/MAXSON2.ssl
index 129b514..6cccde4 100644
--- a/Games/Fallout1.Steam/Original/MAXSON2.ssl
+++ b/Games/Fallout1.Steam/ReDefine/MAXSON2.ssl
@@ -70,26 +70,26 @@ procedure start
 begin
     if (Only_Once) then begin
         Only_Once := 0;
-        critter_add_trait(self_obj, 1, 6, 44);
-        critter_add_trait(self_obj, 1, 5, 80);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 44);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_BOS_ELDERS);
     end
-    if (script_action == 21) then begin
+    if (script_action == look_at_proc) then begin
         call look_at_p_proc();
     end
     else begin
-        if (script_action == 4) then begin
+        if (script_action == pickup_proc) then begin
             call pickup_p_proc();
         end
         else begin
-            if (script_action == 11) then begin
+            if (script_action == talk_proc) then begin
                 call talk_p_proc();
             end
             else begin
-                if (script_action == 12) then begin
+                if (script_action == critter_proc) then begin
                     call critter_p_proc();
                 end
                 else begin
-                    if (script_action == 18) then begin
+                    if (script_action == destroy_proc) then begin
                         call destroy_p_proc();
                     end
                 end
@@ -105,16 +105,16 @@ end
 
 procedure critter_p_proc
 begin
-    if (global_var(250)) then begin
+    if (global_var(GVAR_ENEMY_BROTHERHOOD)) then begin
         hostile := 1;
     end
     if (tile_distance_objs(self_obj, dude_obj) > 12) then begin
         hostile := 0;
     end
     if (hostile) then begin
-        set_global_var(250, 1);
+        set_global_var(GVAR_ENEMY_BROTHERHOOD, 1);
         hostile := 0;
-        attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+        attack(dude_obj);
     end
 end
 
@@ -130,7 +130,7 @@ begin
     call get_reaction();
     start_gdialog(52, self_obj, 4, 12, 5);
     gsay_start;
-    if ((global_var(223) == 1) or (global_var(223) == 2)) then begin
+    if ((global_var(GVAR_BROTHERHOOD_INVASION) == 1) or (global_var(GVAR_BROTHERHOOD_INVASION) == 2)) then begin
         call Maxson34();
     end
     else begin
@@ -138,7 +138,7 @@ begin
             call Maxson22();
         end
         else begin
-            if (global_var(78) == 2) then begin
+            if (global_var(GVAR_THE_VATS) == 2) then begin
                 call Maxson20();
             end
             else begin
@@ -166,20 +166,20 @@ end
 
 procedure destroy_p_proc
 begin
-    set_global_var(250, 1);
+    set_global_var(GVAR_ENEMY_BROTHERHOOD, 1);
     rm_timer_event(self_obj);
     if (source_obj == dude_obj) then begin
-        if (((global_var(160) + global_var(159)) >= 25) and ((global_var(159) > (2 * global_var(160))) or (global_var(156) == 1))) then begin
-            set_global_var(156, 1);
-            set_global_var(157, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
+            set_global_var(GVAR_BERSERKER_REPUTATION, 1);
+            set_global_var(GVAR_CHAMPION_REPUTATION, 0);
         end
-        if (((global_var(160) + global_var(159)) >= 25) and ((global_var(160) > (3 * global_var(159))) or (global_var(157) == 1))) then begin
-            set_global_var(157, 1);
-            set_global_var(156, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(160) > (3 * global_var(GVAR_GOOD_MONSTER))) or (global_var(GVAR_CHAMPION_REPUTATION) == 1))) then begin
+            set_global_var(GVAR_CHAMPION_REPUTATION, 1);
+            set_global_var(GVAR_BERSERKER_REPUTATION, 0);
         end
-        set_global_var(159, global_var(159) + 1);
-        if ((global_var(159) % 5) == 0) then begin
-            set_global_var(155, global_var(155) - 1);
+        set_global_var(GVAR_GOOD_MONSTER, global_var(GVAR_GOOD_MONSTER) + 1);
+        if ((global_var(GVAR_GOOD_MONSTER) % 5) == 0) then begin
+            set_global_var(GVAR_PLAYER_REPUATION, global_var(GVAR_PLAYER_REPUATION) - 1);
         end
     end
 end
@@ -187,30 +187,30 @@ end
 procedure look_at_p_proc
 begin
     script_overrides;
-    display_msg(message_str(52, 100));
+    display_msg(message_str(SCRIPT_MAXSON, 100));
 end
 
 procedure Maxson01
 begin
-    gsay_reply(52, 106);
-    giq_option(-3, 52, 333, Maxson32, 50);
-    if (global_var(106) == 2) then begin
-        giq_option(4, 52, 300, Maxson14, 50);
+    gsay_reply(SCRIPT_MAXSON, 106);
+    giq_option(-3, SCRIPT_MAXSON, 333, Maxson32, NEUTRAL_REACTION);
+    if (global_var(GVAR_MISSING_CARAVAN) == 2) then begin
+        giq_option(4, SCRIPT_MAXSON, 300, Maxson14, NEUTRAL_REACTION);
     end
-    if (global_var(106) == 1) then begin
-        giq_option(4, 52, 301, Maxson06, 50);
+    if (global_var(GVAR_MISSING_CARAVAN) == 1) then begin
+        giq_option(4, SCRIPT_MAXSON, 301, Maxson06, NEUTRAL_REACTION);
     end
-    giq_option(4, 52, 349, Maxson27, 50);
-    giq_option(4, 52, 350, Maxson18, 50);
+    giq_option(4, SCRIPT_MAXSON, 349, Maxson27, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_MAXSON, 350, Maxson18, NEUTRAL_REACTION);
 end
 
 procedure Maxson02
 begin
-    gsay_reply(52, 162);
-    giq_option(4, 52, 302, Maxson04, 50);
-    giq_option(4, 52, 303, Maxson09, 50);
-    giq_option(4, 52, 304, Maxson02a, 51);
-    giq_option(4, 52, 305, Maxson08, 50);
+    gsay_reply(SCRIPT_MAXSON, 162);
+    giq_option(4, SCRIPT_MAXSON, 302, Maxson04, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_MAXSON, 303, Maxson09, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_MAXSON, 304, Maxson02a, BAD_REACTION);
+    giq_option(4, SCRIPT_MAXSON, 305, Maxson08, NEUTRAL_REACTION);
 end
 
 procedure Maxson02a
@@ -229,10 +229,10 @@ end
 
 procedure Maxson04
 begin
-    gsay_reply(52, 133);
-    giq_option(4, 52, 306, Maxson04a, 50);
-    giq_option(4, 52, 307, Maxson27, 50);
-    giq_option(4, 52, 308, MaxsonEnd, 50);
+    gsay_reply(SCRIPT_MAXSON, 133);
+    giq_option(4, SCRIPT_MAXSON, 306, Maxson04a, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_MAXSON, 307, Maxson27, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_MAXSON, 308, MaxsonEnd, NEUTRAL_REACTION);
 end
 
 procedure Maxson04a
@@ -243,30 +243,30 @@ end
 
 procedure Maxson05
 begin
-    gsay_reply(52, 276);
-    giq_option(4, 52, 309, MaxsonEnd, 50);
+    gsay_reply(SCRIPT_MAXSON, 276);
+    giq_option(4, SCRIPT_MAXSON, 309, MaxsonEnd, NEUTRAL_REACTION);
 end
 
 procedure Maxson06
 begin
-    gsay_reply(52, 122);
-    giq_option(4, 52, 310, Maxson07, 50);
-    giq_option(4, 52, 311, Maxson02, 50);
+    gsay_reply(SCRIPT_MAXSON, 122);
+    giq_option(4, SCRIPT_MAXSON, 310, Maxson07, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_MAXSON, 311, Maxson02, NEUTRAL_REACTION);
 end
 
 procedure Maxson07
 begin
-    gsay_reply(52, 128);
-    giq_option(4, 52, 312, Maxson14, 50);
-    giq_option(4, 52, 313, Maxson13, 50);
+    gsay_reply(SCRIPT_MAXSON, 128);
+    giq_option(4, SCRIPT_MAXSON, 312, Maxson14, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_MAXSON, 313, Maxson13, NEUTRAL_REACTION);
 end
 
 procedure Maxson08
 begin
-    gsay_reply(52, 184);
-    giq_option(4, 52, 314, Maxson18, 50);
-    giq_option(4, 52, 315, Maxson04a, 50);
-    giq_option(4, 52, 316, Maxson08a, 50);
+    gsay_reply(SCRIPT_MAXSON, 184);
+    giq_option(4, SCRIPT_MAXSON, 314, Maxson18, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_MAXSON, 315, Maxson04a, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_MAXSON, 316, Maxson08a, NEUTRAL_REACTION);
 end
 
 procedure Maxson08a
@@ -281,10 +281,10 @@ end
 
 procedure Maxson09
 begin
-    gsay_reply(52, 223);
-    giq_option(4, 52, 317, Maxson09a, 51);
-    giq_option(4, 52, 318, Maxson18, 50);
-    giq_option(4, 52, 319, MaxsonEnd, 50);
+    gsay_reply(SCRIPT_MAXSON, 223);
+    giq_option(4, SCRIPT_MAXSON, 317, Maxson09a, BAD_REACTION);
+    giq_option(4, SCRIPT_MAXSON, 318, Maxson18, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_MAXSON, 319, MaxsonEnd, NEUTRAL_REACTION);
 end
 
 procedure Maxson09a
@@ -300,195 +300,195 @@ end
 procedure Maxson10
 begin
     call DownReact();
-    gsay_reply(52, 226);
-    giq_option(4, 52, 320, MaxsonEnd, 51);
+    gsay_reply(SCRIPT_MAXSON, 226);
+    giq_option(4, SCRIPT_MAXSON, 320, MaxsonEnd, BAD_REACTION);
 end
 
 procedure Maxson11
 begin
-    gsay_message(52, 229, 51);
+    gsay_message(SCRIPT_MAXSON, 229, BAD_REACTION);
     Denounce_Player := 1;
 end
 
 procedure Maxson12
 begin
-    gsay_message(52, 229, 51);
+    gsay_message(SCRIPT_MAXSON, 229, BAD_REACTION);
     Denounce_Player := 1;
 end
 
 procedure Maxson13
 begin
-    gsay_message(52, 231, 51);
+    gsay_message(SCRIPT_MAXSON, 231, BAD_REACTION);
     Denounce_Player := 1;
 end
 
 procedure Maxson14
 begin
-    gsay_reply(52, 146);
-    giq_option(4, 52, 321, Maxson02, 50);
+    gsay_reply(SCRIPT_MAXSON, 146);
+    giq_option(4, SCRIPT_MAXSON, 321, Maxson02, NEUTRAL_REACTION);
 end
 
 procedure Maxson15
 begin
-    gsay_reply(52, 141);
-    giq_option(4, 52, 322, Maxson02, 50);
-    giq_option(4, 52, 323, Maxson16, 50);
+    gsay_reply(SCRIPT_MAXSON, 141);
+    giq_option(4, SCRIPT_MAXSON, 322, Maxson02, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_MAXSON, 323, Maxson16, NEUTRAL_REACTION);
 end
 
 procedure Maxson16
 begin
-    gsay_reply(52, 223);
-    giq_option(4, 52, 324, Maxson17, 50);
-    giq_option(4, 52, 325, Maxson18, 50);
-    giq_option(4, 52, 326, Maxson17, 50);
+    gsay_reply(SCRIPT_MAXSON, 223);
+    giq_option(4, SCRIPT_MAXSON, 324, Maxson17, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_MAXSON, 325, Maxson18, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_MAXSON, 326, Maxson17, NEUTRAL_REACTION);
 end
 
 procedure Maxson17
 begin
-    gsay_reply(52, 168);
-    giq_option(4, 52, 327, MaxsonEnd, 51);
+    gsay_reply(SCRIPT_MAXSON, 168);
+    giq_option(4, SCRIPT_MAXSON, 327, MaxsonEnd, BAD_REACTION);
 end
 
 procedure Maxson18
 begin
-    gsay_message(52, 222, 50);
+    gsay_message(SCRIPT_MAXSON, 222, NEUTRAL_REACTION);
 end
 
 procedure Maxson19
 begin
-    gsay_reply(52, 201);
-    giq_option(-3, 52, 333, Maxson32, 50);
-    giq_option(4, 52, 329, Maxson14, 50);
-    if (global_var(106) == 1) then begin
-        giq_option(4, 52, 330, Maxson06, 50);
+    gsay_reply(SCRIPT_MAXSON, 201);
+    giq_option(-3, SCRIPT_MAXSON, 333, Maxson32, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_MAXSON, 329, Maxson14, NEUTRAL_REACTION);
+    if (global_var(GVAR_MISSING_CARAVAN) == 1) then begin
+        giq_option(4, SCRIPT_MAXSON, 330, Maxson06, NEUTRAL_REACTION);
     end
-    giq_option(4, 52, 332, Maxson27, 50);
-    giq_option(4, 52, 331, Maxson18, 50);
+    giq_option(4, SCRIPT_MAXSON, 332, Maxson27, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_MAXSON, 331, Maxson18, NEUTRAL_REACTION);
 end
 
 procedure Maxson20
 begin
-    gsay_reply(52, 193);
-    giq_option(-3, 52, 333, Maxson32, 50);
-    giq_option(4, 52, 334, Maxson22, 50);
-    giq_option(4, 52, 335, Maxson27, 50);
-    giq_option(4, 52, 336, Maxson18, 50);
+    gsay_reply(SCRIPT_MAXSON, 193);
+    giq_option(-3, SCRIPT_MAXSON, 333, Maxson32, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_MAXSON, 334, Maxson22, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_MAXSON, 335, Maxson27, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_MAXSON, 336, Maxson18, NEUTRAL_REACTION);
 end
 
 procedure Maxson21
 begin
-    gsay_reply(52, 201);
-    giq_option(-3, 52, 333, Maxson32, 50);
-    giq_option(4, 52, 334, Maxson22, 50);
-    giq_option(4, 52, 335, Maxson27, 50);
-    giq_option(4, 52, 336, Maxson18, 50);
+    gsay_reply(SCRIPT_MAXSON, 201);
+    giq_option(-3, SCRIPT_MAXSON, 333, Maxson32, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_MAXSON, 334, Maxson22, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_MAXSON, 335, Maxson27, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_MAXSON, 336, Maxson18, NEUTRAL_REACTION);
 end
 
 procedure Maxson22
 begin
-    gsay_reply(52, 205);
-    if (global_var(78) != 2) then begin
-        giq_option(4, 52, 337, Maxson18, 50);
+    gsay_reply(SCRIPT_MAXSON, 205);
+    if (global_var(GVAR_THE_VATS) != 2) then begin
+        giq_option(4, SCRIPT_MAXSON, 337, Maxson18, NEUTRAL_REACTION);
     end
     else begin
-        if (global_var(18) == 1) then begin
-            giq_option(4, 52, 338, Maxson23, 50);
+        if (global_var(GVAR_MASTER_BLOWN) == 1) then begin
+            giq_option(4, SCRIPT_MAXSON, 338, Maxson23, NEUTRAL_REACTION);
         end
         else begin
-            giq_option(4, 52, 339, Maxson06, 50);
+            giq_option(4, SCRIPT_MAXSON, 339, Maxson06, NEUTRAL_REACTION);
         end
     end
 end
 
 procedure Maxson23
 begin
-    gsay_reply(52, 213);
-    giq_option(4, 52, 340, Maxson24, 50);
+    gsay_reply(SCRIPT_MAXSON, 213);
+    giq_option(4, SCRIPT_MAXSON, 340, Maxson24, NEUTRAL_REACTION);
 end
 
 procedure Maxson24
 begin
     set_local_var(5, 1);
-    gsay_reply(52, 219);
-    giq_option(4, 52, 341, Maxson25, 50);
-    giq_option(4, 52, 342, Maxson33, 50);
+    gsay_reply(SCRIPT_MAXSON, 219);
+    giq_option(4, SCRIPT_MAXSON, 341, Maxson25, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_MAXSON, 342, Maxson33, NEUTRAL_REACTION);
 end
 
 procedure Maxson25
 begin
     set_map_var(19, 1);
-    gsay_message(52, 221, 50);
+    gsay_message(SCRIPT_MAXSON, 221, NEUTRAL_REACTION);
 end
 
 procedure Maxson26
 begin
-    gsay_reply(52, 210);
-    giq_option(4, 52, 343, Maxson23, 50);
+    gsay_reply(SCRIPT_MAXSON, 210);
+    giq_option(4, SCRIPT_MAXSON, 343, Maxson23, NEUTRAL_REACTION);
 end
 
 procedure Maxson27
 begin
-    gsay_reply(52, random(138, 140));
+    gsay_reply(SCRIPT_MAXSON, random(138, 140));
     if (local_var(6) == 0) then begin
-        giq_option(4, 52, 344, Maxson28, 50);
+        giq_option(4, SCRIPT_MAXSON, 344, Maxson28, NEUTRAL_REACTION);
     end
-    giq_option(4, 52, 345, Maxson29, 50);
-    giq_option(4, 52, 346, Maxson30, 50);
-    giq_option(4, 52, 347, Maxson31, 50);
-    giq_option(4, 52, 348, Maxson18, 50);
+    giq_option(4, SCRIPT_MAXSON, 345, Maxson29, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_MAXSON, 346, Maxson30, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_MAXSON, 347, Maxson31, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_MAXSON, 348, Maxson18, NEUTRAL_REACTION);
 end
 
 procedure Maxson28
 begin
     variable LVar0 := 0;
     set_local_var(6, 1);
-    LVar0 := create_object_sid(216, 0, 0, -1);
+    LVar0 := create_object(216, 0, 0);
     add_obj_to_inven(dude_obj, LVar0);
-    gsay_reply(52, 186);
-    giq_option(4, 52, 349, Maxson27, 50);
-    giq_option(4, 52, 350, Maxson18, 50);
+    gsay_reply(SCRIPT_MAXSON, 186);
+    giq_option(4, SCRIPT_MAXSON, 349, Maxson27, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_MAXSON, 350, Maxson18, NEUTRAL_REACTION);
 end
 
 procedure Maxson29
 begin
-    gsay_reply(52, 172);
-    giq_option(4, 52, 349, Maxson27, 50);
-    giq_option(4, 52, 350, Maxson18, 50);
+    gsay_reply(SCRIPT_MAXSON, 172);
+    giq_option(4, SCRIPT_MAXSON, 349, Maxson27, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_MAXSON, 350, Maxson18, NEUTRAL_REACTION);
 end
 
 procedure Maxson30
 begin
-    gsay_reply(52, 244);
-    giq_option(4, 52, 349, Maxson27, 50);
-    giq_option(4, 52, 350, Maxson18, 50);
+    gsay_reply(SCRIPT_MAXSON, 244);
+    giq_option(4, SCRIPT_MAXSON, 349, Maxson27, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_MAXSON, 350, Maxson18, NEUTRAL_REACTION);
 end
 
 procedure Maxson31
 begin
-    gsay_reply(52, 262);
-    giq_option(4, 52, 349, Maxson27, 50);
-    giq_option(4, 52, 350, Maxson18, 50);
+    gsay_reply(SCRIPT_MAXSON, 262);
+    giq_option(4, SCRIPT_MAXSON, 349, Maxson27, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_MAXSON, 350, Maxson18, NEUTRAL_REACTION);
 end
 
 procedure Maxson32
 begin
-    gsay_reply(52, 275);
-    giq_option(4, 52, 349, Maxson27, 50);
-    giq_option(4, 52, 350, Maxson18, 50);
+    gsay_reply(SCRIPT_MAXSON, 275);
+    giq_option(4, SCRIPT_MAXSON, 349, Maxson27, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_MAXSON, 350, Maxson18, NEUTRAL_REACTION);
 end
 
 procedure Maxson33
 begin
-    gsay_reply(52, 156);
-    giq_option(4, 52, 349, Maxson27, 50);
-    giq_option(4, 52, 350, Maxson18, 50);
+    gsay_reply(SCRIPT_MAXSON, 156);
+    giq_option(4, SCRIPT_MAXSON, 349, Maxson27, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_MAXSON, 350, Maxson18, NEUTRAL_REACTION);
 end
 
 procedure Maxson34
 begin
-    gsay_reply(52, 193);
-    giq_option(4, 52, 349, Maxson27, 50);
-    giq_option(4, 52, 350, Maxson18, 50);
+    gsay_reply(SCRIPT_MAXSON, 193);
+    giq_option(4, SCRIPT_MAXSON, 349, Maxson27, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_MAXSON, 350, Maxson18, NEUTRAL_REACTION);
 end
 
 procedure MaxsonEnd
@@ -497,11 +497,11 @@ end
 
 procedure Remove_Player
 begin
-    set_global_var(108, 5);
-    set_global_var(583, 0);
-    set_global_var(584, 0);
-    set_global_var(585, 0);
-    set_global_var(586, 0);
+    set_global_var(GVAR_BECOME_AN_INITIATE, 5);
+    set_global_var(GVAR_MARK_BROTHER_2, 0);
+    set_global_var(GVAR_MARK_BROTHER_3, 0);
+    set_global_var(GVAR_MARK_BROTHER_4, 0);
+    set_global_var(GVAR_MARK_BROTHER_5, 0);
     world_map;
 end
 
@@ -511,31 +511,31 @@ begin
         set_local_var(0, 50);
         set_local_var(1, 2);
         set_local_var(2, 1);
-        set_local_var(0, local_var(0) + (5 * get_critter_stat(dude_obj, 3)) - 25);
-        set_local_var(0, local_var(0) + (10 * has_trait(0, dude_obj, 10)));
-        if (has_trait(0, dude_obj, 39)) then begin
-            if (global_var(155) > 0) then begin
-                set_local_var(0, local_var(0) + global_var(155));
+        set_local_var(0, local_var(0) + (5 * dude_charisma) - 25);
+        set_local_var(0, local_var(0) + (10 * has_trait(TRAIT_PERK, dude_obj, PERK_presence)));
+        if (has_trait(TRAIT_PERK, dude_obj, PERK_cult_of_personality)) then begin
+            if (global_var(GVAR_PLAYER_REPUATION) > 0) then begin
+                set_local_var(0, local_var(0) + global_var(GVAR_PLAYER_REPUATION));
             end
             else begin
-                set_local_var(0, local_var(0) - global_var(155));
+                set_local_var(0, local_var(0) - global_var(GVAR_PLAYER_REPUATION));
             end
         end
         else begin
             if (local_var(3) == 1) then begin
-                set_local_var(0, local_var(0) - global_var(155));
+                set_local_var(0, local_var(0) - global_var(GVAR_PLAYER_REPUATION));
             end
             else begin
-                set_local_var(0, local_var(0) + global_var(155));
+                set_local_var(0, local_var(0) + global_var(GVAR_PLAYER_REPUATION));
             end
         end
-        if (global_var(158) > 2) then begin
+        if (global_var(GVAR_CHILDKILLER_REPUATION) > 2) then begin
             set_local_var(0, local_var(0) - 30);
         end
-        if (((global_var(160) + global_var(159)) >= 25) and ((global_var(160) > (3 * global_var(159))) or (global_var(157) == 1))) then begin
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(160) > (3 * global_var(GVAR_GOOD_MONSTER))) or (global_var(GVAR_CHAMPION_REPUTATION) == 1))) then begin
             set_local_var(0, local_var(0) + 20);
         end
-        if (((global_var(160) + global_var(159)) >= 25) and ((global_var(159) > (2 * global_var(160))) or (global_var(156) == 1))) then begin
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
             set_local_var(0, local_var(0) - 20);
         end
         call ReactToLevel();
@@ -628,6 +628,5 @@ end
 
 procedure Goodbyes
 begin
-    exit_line := message_str(634, random(100, 105));
+    exit_line := message_str(SCRIPT_MODREACT, random(100, 105));
 end
-
