diff --git a/Games/Fallout1.Steam/Original/MBOUT2IN.ssl b/Games/Fallout1.Steam/ReDefine/MBOUT2IN.ssl
index 2da0434..855ad51 100644
--- a/Games/Fallout1.Steam/Original/MBOUT2IN.ssl
+++ b/Games/Fallout1.Steam/ReDefine/MBOUT2IN.ssl
@@ -12,19 +12,19 @@ procedure damage_p_proc;
 
 procedure start
 begin
-    if (script_action == 21) then begin
+    if (script_action == look_at_proc) then begin
         call look_at_p_proc();
     end
     else begin
-        if (script_action == 6) then begin
+        if (script_action == use_proc) then begin
             call use_p_proc();
         end
         else begin
-            if (script_action == 7) then begin
+            if (script_action == use_obj_on_proc) then begin
                 call use_obj_on_p_proc();
             end
             else begin
-                if (script_action == 8) then begin
+                if (script_action == use_skill_on_proc) then begin
                     call use_skill_on_p_proc();
                 end
             end
@@ -42,19 +42,19 @@ begin
     script_overrides;
     if (source_obj == Katja_ptr) then begin
         obj_unlock(self_obj);
-        float_msg(Katja_ptr, message_str(623, 172), 0);
+        float_msg(Katja_ptr, message_str(SCRIPT_KATJA, 172), FLOAT_MSG_NORMAL);
     end
     else begin
         if (not(obj_is_locked(self_obj))) then begin
             load_map(31, 12);
         end
         else begin
-            if ((map_var(1) == 1) or (global_var(271) == 1)) then begin
-                display_msg(message_str(526, 101));
+            if ((map_var(1) == 1) or (global_var(GVAR_WATER_CHIP_14) == 1)) then begin
+                display_mstr(101);
                 obj_unlock(self_obj);
             end
             else begin
-                display_msg(message_str(526, 100));
+                display_mstr(100);
             end
         end
     end
@@ -65,19 +65,19 @@ begin
     if (obj_pid(obj_being_used_with) == 77) then begin
         if (not(obj_is_locked(self_obj))) then begin
             script_overrides;
-            display_msg(message_str(526, 102));
+            display_mstr(102);
         end
         else begin
             if (is_success(roll_vs_skill(dude_obj, 9, -40))) then begin
                 script_overrides;
-                display_msg(message_str(526, 103));
+                display_mstr(103);
                 obj_unlock(self_obj);
-                display_msg(message_str(766, 103) + "65" + message_str(766, 104));
+                display_msg(message_str(SCRIPT_GENCHAT, 103) + "65" + message_str(SCRIPT_GENCHAT, 104));
                 give_exp_points(65);
             end
             else begin
                 script_overrides;
-                display_msg(message_str(526, 104));
+                display_mstr(104);
                 set_local_var(0, local_var(0) + 1);
             end
         end
@@ -89,19 +89,19 @@ procedure use_skill_on_p_proc
 begin
     if (action_being_used == 9) then begin
         if (not(obj_is_locked(self_obj))) then begin
-            display_msg(message_str(526, 102));
+            display_mstr(102);
         end
         else begin
             if (is_success(roll_vs_skill(dude_obj, 9, -60))) then begin
                 script_overrides;
                 obj_unlock(self_obj);
-                display_msg(message_str(526, 103));
+                display_mstr(103);
                 give_exp_points(85);
-                display_msg(message_str(766, 103) + "85" + message_str(766, 104));
+                display_msg(message_str(SCRIPT_GENCHAT, 103) + "85" + message_str(SCRIPT_GENCHAT, 104));
             end
             else begin
                 script_overrides;
-                display_msg(message_str(526, 105));
+                display_mstr(105);
                 set_local_var(0, local_var(0) + 1);
             end
         end
@@ -112,14 +112,13 @@ end
 procedure alert
 begin
     if (local_var(0) > 2) then begin
-        set_global_var(146, 1);
+        set_global_var(GVAR_VATS_ALERT, 1);
     end
 end
 
 procedure damage_p_proc
 begin
     set_local_var(0, local_var(0) + 1);
-    display_msg(message_str(526, 106));
+    display_mstr(106);
     call alert();
 end
-
