diff --git a/Games/Fallout1.Steam/Original/MEANDER.ssl b/Games/Fallout1.Steam/ReDefine/MEANDER.ssl
index f35bc2d..b799f25 100644
--- a/Games/Fallout1.Steam/Original/MEANDER.ssl
+++ b/Games/Fallout1.Steam/ReDefine/MEANDER.ssl
@@ -32,7 +32,7 @@ begin
         if (local_var(0) == 0) then begin
             set_local_var(0, 1);
         end
-        critter_add_trait(self_obj, 1, 6, 32);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 32);
         initialized := 1;
     end
     curtime := DAY;
@@ -41,22 +41,22 @@ begin
         prevtime := curtime;
         animate_move_obj_to_tile(self_obj, local_var(0), 0);
     end
-    if (script_action == 22) then begin
+    if (script_action == timed_event_proc) then begin
         call timeforwhat();
     end
     else begin
-        if (script_action == 11) then begin
+        if (script_action == talk_proc) then begin
             call do_dialogue();
         end
         else begin
-            if (script_action == 4) then begin
+            if (script_action == pickup_proc) then begin
                 hostile := 1;
             end
             else begin
-                if (script_action == 12) then begin
+                if (script_action == critter_proc) then begin
                     if (hostile) then begin
                         hostile := 0;
-                        attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+                        attack(dude_obj);
                     end
                     else begin
                         loopcount := loopcount + 1;
@@ -71,19 +71,19 @@ begin
                     end
                 end
                 else begin
-                    if (script_action == 18) then begin
+                    if (script_action == destroy_proc) then begin
                         if (source_obj == dude_obj) then begin
-                            if (((global_var(160) + global_var(159)) >= 25) and ((global_var(159) > (2 * global_var(160))) or (global_var(156) == 1))) then begin
-                                set_global_var(156, 1);
-                                set_global_var(157, 0);
+                            if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
+                                set_global_var(GVAR_BERSERKER_REPUTATION, 1);
+                                set_global_var(GVAR_CHAMPION_REPUTATION, 0);
                             end
-                            if (((global_var(160) + global_var(159)) >= 25) and ((global_var(160) > (3 * global_var(159))) or (global_var(157) == 1))) then begin
-                                set_global_var(157, 1);
-                                set_global_var(156, 0);
+                            if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(160) > (3 * global_var(GVAR_GOOD_MONSTER))) or (global_var(GVAR_CHAMPION_REPUTATION) == 1))) then begin
+                                set_global_var(GVAR_CHAMPION_REPUTATION, 1);
+                                set_global_var(GVAR_BERSERKER_REPUTATION, 0);
                             end
-                            set_global_var(160, global_var(160) + 1);
-                            if ((global_var(160) % 6) == 0) then begin
-                                set_global_var(155, global_var(155) + 1);
+                            set_global_var(GVAR_BAD_MONSTER, global_var(GVAR_BAD_MONSTER) + 1);
+                            if ((global_var(GVAR_BAD_MONSTER) % 6) == 0) then begin
+                                set_global_var(GVAR_PLAYER_REPUATION, global_var(GVAR_PLAYER_REPUATION) + 1);
                             end
                         end
                     end
@@ -137,19 +137,19 @@ procedure generic00
 begin
     rndx := random(1, 7);
     if (rndx == 1) then begin
-        gsay_message(84, 100, 50);
+        gsay_message(NAME, 100, NEUTRAL_REACTION);
     end
     if (rndx == 2) then begin
-        gsay_message(84, 101, 50);
+        gsay_message(NAME, 101, NEUTRAL_REACTION);
     end
     if (rndx == 3) then begin
-        gsay_message(84, 102, 50);
+        gsay_message(NAME, 102, NEUTRAL_REACTION);
     end
     if (rndx == 4) then begin
-        gsay_message(84, 103, 50);
+        gsay_message(NAME, 103, NEUTRAL_REACTION);
     end
     if (rndx == 5) then begin
-        gsay_message(84, 104, 50);
+        gsay_message(NAME, 104, NEUTRAL_REACTION);
     end
     if (rndx == 6) then begin
         call generic02();
@@ -163,16 +163,16 @@ procedure generic01
 begin
     rndx := random(1, 4);
     if (rndx == 1) then begin
-        gsay_message(84, 105, 50);
+        gsay_message(NAME, 105, NEUTRAL_REACTION);
     end
     if (rndx == 2) then begin
-        gsay_message(84, 106, 50);
+        gsay_message(NAME, 106, NEUTRAL_REACTION);
     end
     if (rndx == 3) then begin
-        gsay_message(84, 107, 50);
+        gsay_message(NAME, 107, NEUTRAL_REACTION);
     end
     if (rndx == 4) then begin
-        gsay_message(84, 108, 50);
+        gsay_message(NAME, 108, NEUTRAL_REACTION);
     end
 end
 
@@ -181,14 +181,14 @@ begin
     if (SET == 0) then begin
         rndx := random(1, 2);
         if (rndx == 1) then begin
-            gsay_message(84, 109, 50);
+            gsay_message(NAME, 109, NEUTRAL_REACTION);
         end
         if (rndx == 2) then begin
-            gsay_message(84, 110, 50);
+            gsay_message(NAME, 110, NEUTRAL_REACTION);
         end
     end
     else begin
-        gsay_message(84, 111, 50);
+        gsay_message(NAME, 111, NEUTRAL_REACTION);
     end
 end
 
@@ -197,14 +197,13 @@ begin
     if (MUTANTS == 0) then begin
         rndx := random(1, 2);
         if (rndx == 1) then begin
-            gsay_message(84, 112, 50);
+            gsay_message(NAME, 112, NEUTRAL_REACTION);
         end
         if (rndx == 2) then begin
-            gsay_message(84, 113, 50);
+            gsay_message(NAME, 113, NEUTRAL_REACTION);
         end
     end
     else begin
-        gsay_message(84, 114, 50);
+        gsay_message(NAME, 114, NEUTRAL_REACTION);
     end
 end
-
