diff --git a/Games/Fallout1.Steam/Original/MICHAE~1.ssl b/Games/Fallout1.Steam/ReDefine/MICHAE~1.ssl
index 8fa0104..1ff2985 100644
--- a/Games/Fallout1.Steam/Original/MICHAE~1.ssl
+++ b/Games/Fallout1.Steam/ReDefine/MICHAE~1.ssl
@@ -110,25 +110,25 @@ variable exit_line;
 procedure start
 begin
     if (Only_Once) then begin
-        critter_add_trait(self_obj, 1, 6, 44);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 44);
     end
-    if (script_action == 21) then begin
+    if (script_action == look_at_proc) then begin
         call look_at_p_proc();
     end
     else begin
-        if (script_action == 4) then begin
+        if (script_action == pickup_proc) then begin
             call pickup_p_proc();
         end
         else begin
-            if (script_action == 11) then begin
+            if (script_action == talk_proc) then begin
                 call talk_p_proc();
             end
             else begin
-                if (script_action == 12) then begin
+                if (script_action == critter_proc) then begin
                     call critter_p_proc();
                 end
                 else begin
-                    if (script_action == 18) then begin
+                    if (script_action == destroy_proc) then begin
                         call destroy_p_proc();
                     end
                 end
@@ -139,26 +139,26 @@ end
 
 procedure goto0
 begin
-    float_msg(self_obj, message_str(320, 103), 0);
+    float_msg(self_obj, message_str(SCRIPT_MICHAEL, 103), FLOAT_MSG_NORMAL);
 end
 
 procedure goto1
 begin
-    float_msg(self_obj, message_str(320, 104), 0);
+    float_msg(self_obj, message_str(SCRIPT_MICHAEL, 104), FLOAT_MSG_NORMAL);
 end
 
 procedure goto2
 begin
-    float_msg(self_obj, message_str(320, 105), 0);
+    float_msg(self_obj, message_str(SCRIPT_MICHAEL, 105), FLOAT_MSG_NORMAL);
 end
 
 procedure goto3
 begin
-    gsay_reply(0, message_str(320, 106));
-    giq_option(5, 320, message_str(320, 107), goto3a, 49);
-    giq_option(4, 320, message_str(320, 108), goto11, 50);
-    giq_option(4, 320, message_str(320, 109), goto10, 50);
-    giq_option(-3, 320, message_str(320, 110), goto4, 50);
+    gsay_reply(0, message_str(SCRIPT_MICHAEL, 106));
+    giq_option(5, SCRIPT_MICHAEL, message_str(SCRIPT_MICHAEL, 107), goto3a, GOOD_REACTION);
+    giq_option(4, SCRIPT_MICHAEL, message_str(SCRIPT_MICHAEL, 108), goto11, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_MICHAEL, message_str(SCRIPT_MICHAEL, 109), goto10, NEUTRAL_REACTION);
+    giq_option(-3, SCRIPT_MICHAEL, message_str(SCRIPT_MICHAEL, 110), goto4, NEUTRAL_REACTION);
 end
 
 procedure goto3a
@@ -169,27 +169,27 @@ end
 
 procedure goto4
 begin
-    gsay_reply(0, message_str(320, 111));
-    giq_option(-3, 320, message_str(320, 112), goto5, 50);
-    giq_option(-3, 320, message_str(320, 113), goto5, 50);
+    gsay_reply(0, message_str(SCRIPT_MICHAEL, 111));
+    giq_option(-3, SCRIPT_MICHAEL, message_str(SCRIPT_MICHAEL, 112), goto5, NEUTRAL_REACTION);
+    giq_option(-3, SCRIPT_MICHAEL, message_str(SCRIPT_MICHAEL, 113), goto5, NEUTRAL_REACTION);
 end
 
 procedure goto5
 begin
-    gsay_message(0, message_str(320, 114), 50);
+    gsay_message(0, message_str(SCRIPT_MICHAEL, 114), NEUTRAL_REACTION);
 end
 
 procedure goto10
 begin
-    gsay_message(0, message_str(320, 115), 50);
+    gsay_message(0, message_str(SCRIPT_MICHAEL, 115), NEUTRAL_REACTION);
 end
 
 procedure goto11
 begin
-    gsay_reply(0, message_str(320, 116));
-    giq_option(5, 320, message_str(320, 117), goto11a, 50);
-    giq_option(4, 320, message_str(320, 118), goto26, 50);
-    giq_option(4, 320, message_str(320, 119), gotoend, 50);
+    gsay_reply(0, message_str(SCRIPT_MICHAEL, 116));
+    giq_option(5, SCRIPT_MICHAEL, message_str(SCRIPT_MICHAEL, 117), goto11a, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_MICHAEL, message_str(SCRIPT_MICHAEL, 118), goto26, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_MICHAEL, message_str(SCRIPT_MICHAEL, 119), gotoend, NEUTRAL_REACTION);
 end
 
 procedure goto11a
@@ -204,21 +204,21 @@ end
 
 procedure goto12
 begin
-    gsay_message(0, message_str(320, 120), 50);
+    gsay_message(0, message_str(SCRIPT_MICHAEL, 120), NEUTRAL_REACTION);
 end
 
 procedure goto13
 begin
-    gsay_message(0, message_str(320, 121), 50);
-    giq_option(5, 320, message_str(320, 122), goto13a, 50);
-    giq_option(4, 320, message_str(320, 123), goto13b, 50);
-    giq_option(4, 320, message_str(320, 124), goto22, 50);
+    gsay_message(0, message_str(SCRIPT_MICHAEL, 121), NEUTRAL_REACTION);
+    giq_option(5, SCRIPT_MICHAEL, message_str(SCRIPT_MICHAEL, 122), goto13a, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_MICHAEL, message_str(SCRIPT_MICHAEL, 123), goto13b, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_MICHAEL, message_str(SCRIPT_MICHAEL, 124), goto22, NEUTRAL_REACTION);
 end
 
 procedure goto13a
 begin
     if (is_success(do_check(dude_obj, 3, 0))) then begin
-        if (get_critter_stat(dude_obj, 34) == 0) then begin
+        if (dude_gender == 0) then begin
             call goto16();
         end
         else begin
@@ -226,7 +226,7 @@ begin
         end
     end
     else begin
-        if (get_critter_stat(dude_obj, 34) == 0) then begin
+        if (dude_gender == 0) then begin
             call goto14();
         end
         else begin
@@ -247,21 +247,21 @@ end
 
 procedure goto14
 begin
-    gsay_message(0, message_str(320, 125), 50);
+    gsay_message(0, message_str(SCRIPT_MICHAEL, 125), NEUTRAL_REACTION);
     call DownReact();
 end
 
 procedure goto15
 begin
-    gsay_message(0, message_str(320, 126), 50);
+    gsay_message(0, message_str(SCRIPT_MICHAEL, 126), NEUTRAL_REACTION);
     call DownReact();
 end
 
 procedure goto16
 begin
-    gsay_reply(0, message_str(320, 127));
-    giq_option(4, 320, message_str(320, 128), goto16a, 50);
-    giq_option(4, 320, message_str(320, 129), goto16b, 50);
+    gsay_reply(0, message_str(SCRIPT_MICHAEL, 127));
+    giq_option(4, SCRIPT_MICHAEL, message_str(SCRIPT_MICHAEL, 128), goto16a, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_MICHAEL, message_str(SCRIPT_MICHAEL, 129), goto16b, NEUTRAL_REACTION);
 end
 
 procedure goto16a
@@ -278,164 +278,164 @@ end
 
 procedure goto17
 begin
-    gsay_message(0, message_str(320, 130), 50);
+    gsay_message(0, message_str(SCRIPT_MICHAEL, 130), NEUTRAL_REACTION);
     call DownReact();
 end
 
 procedure goto18
 begin
-    gsay_reply(0, message_str(320, 131));
-    giq_option(4, 320, message_str(320, 132), goto16a, 50);
-    giq_option(4, 320, message_str(320, 133), goto16b, 50);
+    gsay_reply(0, message_str(SCRIPT_MICHAEL, 131));
+    giq_option(4, SCRIPT_MICHAEL, message_str(SCRIPT_MICHAEL, 132), goto16a, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_MICHAEL, message_str(SCRIPT_MICHAEL, 133), goto16b, NEUTRAL_REACTION);
 end
 
 procedure goto19
 begin
-    gsay_message(0, message_str(320, 134), 50);
+    gsay_message(0, message_str(SCRIPT_MICHAEL, 134), NEUTRAL_REACTION);
     call TopReact();
-    item := create_object_sid(27, 0, 0, -1);
+    item := create_object(27, 0, 0);
     add_obj_to_inven(dude_obj, item);
-    item := create_object_sid(27, 0, 0, -1);
+    item := create_object(27, 0, 0);
     add_obj_to_inven(dude_obj, item);
-    item := create_object_sid(27, 0, 0, -1);
+    item := create_object(27, 0, 0);
     add_obj_to_inven(dude_obj, item);
 end
 
 procedure goto20
 begin
-    gsay_message(0, message_str(320, 135), 50);
+    gsay_message(0, message_str(SCRIPT_MICHAEL, 135), NEUTRAL_REACTION);
 end
 
 procedure goto21
 begin
-    gsay_message(0, message_str(320, 136), 50);
+    gsay_message(0, message_str(SCRIPT_MICHAEL, 136), NEUTRAL_REACTION);
 end
 
 procedure goto22
 begin
-    gsay_reply(0, message_str(320, 137));
-    giq_option(6, 320, message_str(320, 138), goto23, 50);
-    giq_option(4, 320, message_str(320, 139), goto24, 50);
-    giq_option(4, 320, message_str(320, 140), gotoend, 50);
+    gsay_reply(0, message_str(SCRIPT_MICHAEL, 137));
+    giq_option(6, SCRIPT_MICHAEL, message_str(SCRIPT_MICHAEL, 138), goto23, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_MICHAEL, message_str(SCRIPT_MICHAEL, 139), goto24, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_MICHAEL, message_str(SCRIPT_MICHAEL, 140), gotoend, NEUTRAL_REACTION);
 end
 
 procedure goto23
 begin
-    gsay_message(0, message_str(320, 141), 50);
+    gsay_message(0, message_str(SCRIPT_MICHAEL, 141), NEUTRAL_REACTION);
     call TopReact();
-    item := create_object_sid(50, 0, 0, -1);
+    item := create_object(50, 0, 0);
     add_obj_to_inven(dude_obj, item);
-    item := create_object_sid(50, 0, 0, -1);
+    item := create_object(50, 0, 0);
     add_obj_to_inven(dude_obj, item);
 end
 
 procedure goto24
 begin
-    if (get_critter_stat(dude_obj, 34) == 0) then begin
-        gsay_message(0, message_str(320, 142), 50);
+    if (dude_gender == 0) then begin
+        gsay_message(0, message_str(SCRIPT_MICHAEL, 142), NEUTRAL_REACTION);
     end
     else begin
-        gsay_message(0, message_str(320, 143), 50);
+        gsay_message(0, message_str(SCRIPT_MICHAEL, 143), NEUTRAL_REACTION);
     end
 end
 
 procedure goto26
 begin
-    gsay_reply(0, message_str(320, 144));
-    giq_option(4, 320, message_str(320, 145), goto27, 50);
+    gsay_reply(0, message_str(SCRIPT_MICHAEL, 144));
+    giq_option(4, SCRIPT_MICHAEL, message_str(SCRIPT_MICHAEL, 145), goto27, NEUTRAL_REACTION);
     if (map_var(9) > 0) then begin
-        giq_option(4, 320, message_str(320, 146), goto48, 50);
+        giq_option(4, SCRIPT_MICHAEL, message_str(SCRIPT_MICHAEL, 146), goto48, NEUTRAL_REACTION);
     end
-    giq_option(4, 320, message_str(320, 147), goto27, 50);
+    giq_option(4, SCRIPT_MICHAEL, message_str(SCRIPT_MICHAEL, 147), goto27, NEUTRAL_REACTION);
 end
 
 procedure goto27
 begin
-    gsay_reply(0, message_str(320, 148));
-    giq_option(4, 320, message_str(320, 149), goto28, 50);
-    giq_option(4, 320, message_str(320, 150), goto29, 50);
-    giq_option(4, 320, message_str(320, 151), gotoend, 50);
+    gsay_reply(0, message_str(SCRIPT_MICHAEL, 148));
+    giq_option(4, SCRIPT_MICHAEL, message_str(SCRIPT_MICHAEL, 149), goto28, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_MICHAEL, message_str(SCRIPT_MICHAEL, 150), goto29, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_MICHAEL, message_str(SCRIPT_MICHAEL, 151), gotoend, NEUTRAL_REACTION);
 end
 
 procedure goto28
 begin
     Hear_28 := 1;
-    gsay_reply(0, message_str(320, 152));
+    gsay_reply(0, message_str(SCRIPT_MICHAEL, 152));
     if (Hear_29 == 0) then begin
-        giq_option(4, 320, message_str(320, 153), goto29, 50);
+        giq_option(4, SCRIPT_MICHAEL, message_str(SCRIPT_MICHAEL, 153), goto29, NEUTRAL_REACTION);
     end
-    giq_option(4, 320, message_str(320, 154), gotoend, 50);
+    giq_option(4, SCRIPT_MICHAEL, message_str(SCRIPT_MICHAEL, 154), gotoend, NEUTRAL_REACTION);
 end
 
 procedure goto29
 begin
     Hear_29 := 1;
-    gsay_reply(0, message_str(320, 155));
+    gsay_reply(0, message_str(SCRIPT_MICHAEL, 155));
     if (Hear_28 == 0) then begin
-        giq_option(8, 320, message_str(320, 156), goto30, 50);
-        giq_option(4, 320, message_str(320, 157), goto28, 50);
+        giq_option(8, SCRIPT_MICHAEL, message_str(SCRIPT_MICHAEL, 156), goto30, NEUTRAL_REACTION);
+        giq_option(4, SCRIPT_MICHAEL, message_str(SCRIPT_MICHAEL, 157), goto28, NEUTRAL_REACTION);
     end
-    giq_option(4, 320, message_str(320, 158), gotoend, 50);
+    giq_option(4, SCRIPT_MICHAEL, message_str(SCRIPT_MICHAEL, 158), gotoend, NEUTRAL_REACTION);
 end
 
 procedure goto30
 begin
-    gsay_reply(0, message_str(320, 159));
-    giq_option(8, 320, message_str(320, 160), UpReact, 50);
+    gsay_reply(0, message_str(SCRIPT_MICHAEL, 159));
+    giq_option(8, SCRIPT_MICHAEL, message_str(SCRIPT_MICHAEL, 160), UpReact, NEUTRAL_REACTION);
 end
 
 procedure goto38
 begin
-    gsay_reply(0, message_str(320, 167));
+    gsay_reply(0, message_str(SCRIPT_MICHAEL, 167));
     temp2 := map_var(11) + 1;
     set_map_var(11, temp2);
-    item := create_object_sid(temp, 0, 0, -1);
+    item := create_object(temp, 0, 0);
     add_obj_to_inven(dude_obj, item);
-    giq_option(4, 320, message_str(320, 168), gotoend, 50);
+    giq_option(4, SCRIPT_MICHAEL, message_str(SCRIPT_MICHAEL, 168), gotoend, NEUTRAL_REACTION);
     if (map_var(9) > 0) then begin
-        giq_option(4, 320, message_str(320, 169), goto43, 50);
+        giq_option(4, SCRIPT_MICHAEL, message_str(SCRIPT_MICHAEL, 169), goto43, NEUTRAL_REACTION);
     end
 end
 
 procedure goto39
 begin
-    gsay_reply(0, message_str(320, 170));
-    giq_option(4, 320, message_str(320, 171), gotoend, 50);
+    gsay_reply(0, message_str(SCRIPT_MICHAEL, 170));
+    giq_option(4, SCRIPT_MICHAEL, message_str(SCRIPT_MICHAEL, 171), gotoend, NEUTRAL_REACTION);
     if (map_var(9) > 0) then begin
-        giq_option(4, 320, message_str(320, 172), goto43, 50);
+        giq_option(4, SCRIPT_MICHAEL, message_str(SCRIPT_MICHAEL, 172), goto43, NEUTRAL_REACTION);
     end
 end
 
 procedure goto40
 begin
-    gsay_reply(0, message_str(320, 173));
-    giq_option(4, 320, message_str(320, 174), gotoend, 50);
+    gsay_reply(0, message_str(SCRIPT_MICHAEL, 173));
+    giq_option(4, SCRIPT_MICHAEL, message_str(SCRIPT_MICHAEL, 174), gotoend, NEUTRAL_REACTION);
     if (map_var(9) > 0) then begin
-        giq_option(4, 320, message_str(320, 175), goto43, 50);
+        giq_option(4, SCRIPT_MICHAEL, message_str(SCRIPT_MICHAEL, 175), goto43, NEUTRAL_REACTION);
     end
 end
 
 procedure goto41
 begin
-    gsay_reply(0, message_str(320, 176));
-    giq_option(4, 320, message_str(320, 177), gotoend, 50);
+    gsay_reply(0, message_str(SCRIPT_MICHAEL, 176));
+    giq_option(4, SCRIPT_MICHAEL, message_str(SCRIPT_MICHAEL, 177), gotoend, NEUTRAL_REACTION);
     if (map_var(9) > 0) then begin
-        giq_option(4, 320, message_str(320, 178), goto43, 50);
+        giq_option(4, SCRIPT_MICHAEL, message_str(SCRIPT_MICHAEL, 178), goto43, NEUTRAL_REACTION);
     end
 end
 
 procedure goto42
 begin
-    gsay_reply(0, message_str(320, 179));
-    giq_option(4, 320, message_str(320, 180), gotoend, 50);
+    gsay_reply(0, message_str(SCRIPT_MICHAEL, 179));
+    giq_option(4, SCRIPT_MICHAEL, message_str(SCRIPT_MICHAEL, 180), gotoend, NEUTRAL_REACTION);
     if (map_var(9) > 0) then begin
-        giq_option(4, 320, message_str(320, 181), goto43, 50);
+        giq_option(4, SCRIPT_MICHAEL, message_str(SCRIPT_MICHAEL, 181), goto43, NEUTRAL_REACTION);
     end
 end
 
 procedure goto43
 begin
-    gsay_message(0, message_str(320, 182), 50);
+    gsay_message(0, message_str(SCRIPT_MICHAEL, 182), NEUTRAL_REACTION);
     call goto49();
 end
 
@@ -495,15 +495,15 @@ end
 
 procedure goto47
 begin
-    gsay_reply(0, message_str(320, 183));
+    gsay_reply(0, message_str(SCRIPT_MICHAEL, 183));
     if (Hear_50 == 0) then begin
-        giq_option(4, 320, message_str(320, 184), goto50, 50);
+        giq_option(4, SCRIPT_MICHAEL, message_str(SCRIPT_MICHAEL, 184), goto50, NEUTRAL_REACTION);
     end
     if (map_var(9) > 0) then begin
-        giq_option(4, 320, message_str(320, 185), goto48, 50);
+        giq_option(4, SCRIPT_MICHAEL, message_str(SCRIPT_MICHAEL, 185), goto48, NEUTRAL_REACTION);
     end
-    giq_option(4, 320, message_str(320, 186), gotoend, 50);
-    giq_option(4, 320, message_str(320, 187), goto47a, 50);
+    giq_option(4, SCRIPT_MICHAEL, message_str(SCRIPT_MICHAEL, 186), gotoend, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_MICHAEL, message_str(SCRIPT_MICHAEL, 187), goto47a, NEUTRAL_REACTION);
 end
 
 procedure goto47a
@@ -514,13 +514,13 @@ end
 
 procedure goto48
 begin
-    gsay_message(0, message_str(320, 188), 50);
+    gsay_message(0, message_str(SCRIPT_MICHAEL, 188), NEUTRAL_REACTION);
     call goto49();
 end
 
 procedure goto49
 begin
-    gsay_message(0, message_str(320, 189), 50);
+    gsay_message(0, message_str(SCRIPT_MICHAEL, 189), NEUTRAL_REACTION);
     if (map_var(10) > map_var(11)) then begin
         call goto144();
     end
@@ -564,37 +564,37 @@ end
 procedure goto50
 begin
     Hear_50 := 1;
-    gsay_reply(0, message_str(320, 190));
-    giq_option(4, 320, message_str(320, 191), goto51, 50);
-    giq_option(4, 320, message_str(320, 192), gotoend, 50);
-    giq_option(4, 320, message_str(320, 193), goto47a, 51);
+    gsay_reply(0, message_str(SCRIPT_MICHAEL, 190));
+    giq_option(4, SCRIPT_MICHAEL, message_str(SCRIPT_MICHAEL, 191), goto51, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_MICHAEL, message_str(SCRIPT_MICHAEL, 192), gotoend, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_MICHAEL, message_str(SCRIPT_MICHAEL, 193), goto47a, BAD_REACTION);
 end
 
 procedure goto51
 begin
-    gsay_message(0, message_str(320, 194), 50);
+    gsay_message(0, message_str(SCRIPT_MICHAEL, 194), NEUTRAL_REACTION);
     set_local_var(5, game_time / (10 * 60 * 60));
     set_map_var(6, 1);
-    item := create_object_sid(27, 0, 0, -1);
+    item := create_object(27, 0, 0);
     add_obj_to_inven(dude_obj, item);
-    item := create_object_sid(27, 0, 0, -1);
+    item := create_object(27, 0, 0);
     add_obj_to_inven(dude_obj, item);
-    item := create_object_sid(27, 0, 0, -1);
+    item := create_object(27, 0, 0);
     add_obj_to_inven(dude_obj, item);
 end
 
 procedure goto52
 begin
-    gsay_reply(0, message_str(320, 195));
-    giq_option(4, 320, message_str(320, 196), goto53, 50);
-    giq_option(4, 320, message_str(320, 197), goto57, 50);
-    giq_option(4, 320, message_str(320, 198), gotoend, 50);
-    giq_option(4, 320, message_str(320, 199), goto52a, 50);
+    gsay_reply(0, message_str(SCRIPT_MICHAEL, 195));
+    giq_option(4, SCRIPT_MICHAEL, message_str(SCRIPT_MICHAEL, 196), goto53, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_MICHAEL, message_str(SCRIPT_MICHAEL, 197), goto57, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_MICHAEL, message_str(SCRIPT_MICHAEL, 198), gotoend, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_MICHAEL, message_str(SCRIPT_MICHAEL, 199), goto52a, NEUTRAL_REACTION);
 end
 
 procedure goto52a
 begin
-    if (get_critter_stat(dude_obj, 34) == 0) then begin
+    if (dude_gender == 0) then begin
         call goto54();
     end
     else begin
@@ -604,139 +604,139 @@ end
 
 procedure goto53
 begin
-    gsay_message(0, message_str(320, 200), 50);
+    gsay_message(0, message_str(SCRIPT_MICHAEL, 200), NEUTRAL_REACTION);
 end
 
 procedure goto54
 begin
-    gsay_message(0, message_str(320, 201), 50);
+    gsay_message(0, message_str(SCRIPT_MICHAEL, 201), NEUTRAL_REACTION);
 end
 
 procedure goto55
 begin
-    gsay_message(0, message_str(320, 202), 50);
+    gsay_message(0, message_str(SCRIPT_MICHAEL, 202), NEUTRAL_REACTION);
 end
 
 procedure goto56
 begin
-    gsay_message(0, message_str(320, 203), 50);
+    gsay_message(0, message_str(SCRIPT_MICHAEL, 203), NEUTRAL_REACTION);
 end
 
 procedure goto57
 begin
-    gsay_message(0, message_str(320, 204), 50);
+    gsay_message(0, message_str(SCRIPT_MICHAEL, 204), NEUTRAL_REACTION);
 end
 
 procedure goto58
 begin
-    gsay_message(0, message_str(320, 205), 50);
+    gsay_message(0, message_str(SCRIPT_MICHAEL, 205), NEUTRAL_REACTION);
 end
 
 procedure goto59
 begin
-    gsay_message(0, message_str(320, 206), 50);
+    gsay_message(0, message_str(SCRIPT_MICHAEL, 206), NEUTRAL_REACTION);
 end
 
 procedure goto60
 begin
-    gsay_reply(0, message_str(320, 207));
+    gsay_reply(0, message_str(SCRIPT_MICHAEL, 207));
     if (map_var(9) > 0) then begin
-        giq_option(4, 320, message_str(320, 208), goto61, 50);
+        giq_option(4, SCRIPT_MICHAEL, message_str(SCRIPT_MICHAEL, 208), goto61, NEUTRAL_REACTION);
     end
-    giq_option(4, 320, message_str(320, 209), gotoend, 50);
+    giq_option(4, SCRIPT_MICHAEL, message_str(SCRIPT_MICHAEL, 209), gotoend, NEUTRAL_REACTION);
 end
 
 procedure goto61
 begin
-    gsay_message(0, message_str(320, 210), 50);
+    gsay_message(0, message_str(SCRIPT_MICHAEL, 210), NEUTRAL_REACTION);
     call goto49();
 end
 
 procedure goto144
 begin
-    gsay_reply(0, message_str(320, 211));
-    giq_option(4, 320, message_str(320, 212), goto145, 50);
+    gsay_reply(0, message_str(SCRIPT_MICHAEL, 211));
+    giq_option(4, SCRIPT_MICHAEL, message_str(SCRIPT_MICHAEL, 212), goto145, NEUTRAL_REACTION);
     if (map_var(10) > map_var(11)) then begin
-        giq_option(4, 320, message_str(320, 213), goto44a, 50);
-        giq_option(4, 320, message_str(320, 214), goto44b, 50);
-        giq_option(4, 320, message_str(320, 215), goto44c, 50);
-        giq_option(4, 320, message_str(320, 216), goto44d, 50);
-        giq_option(4, 320, message_str(320, 217), goto44e, 50);
-        giq_option(4, 320, message_str(320, 218), goto44f, 50);
-        giq_option(4, 320, message_str(320, 219), goto44g, 50);
-        giq_option(4, 320, message_str(320, 220), goto44h, 50);
-        giq_option(4, 320, message_str(320, 221), goto44i, 50);
+        giq_option(4, SCRIPT_MICHAEL, message_str(SCRIPT_MICHAEL, 213), goto44a, NEUTRAL_REACTION);
+        giq_option(4, SCRIPT_MICHAEL, message_str(SCRIPT_MICHAEL, 214), goto44b, NEUTRAL_REACTION);
+        giq_option(4, SCRIPT_MICHAEL, message_str(SCRIPT_MICHAEL, 215), goto44c, NEUTRAL_REACTION);
+        giq_option(4, SCRIPT_MICHAEL, message_str(SCRIPT_MICHAEL, 216), goto44d, NEUTRAL_REACTION);
+        giq_option(4, SCRIPT_MICHAEL, message_str(SCRIPT_MICHAEL, 217), goto44e, NEUTRAL_REACTION);
+        giq_option(4, SCRIPT_MICHAEL, message_str(SCRIPT_MICHAEL, 218), goto44f, NEUTRAL_REACTION);
+        giq_option(4, SCRIPT_MICHAEL, message_str(SCRIPT_MICHAEL, 219), goto44g, NEUTRAL_REACTION);
+        giq_option(4, SCRIPT_MICHAEL, message_str(SCRIPT_MICHAEL, 220), goto44h, NEUTRAL_REACTION);
+        giq_option(4, SCRIPT_MICHAEL, message_str(SCRIPT_MICHAEL, 221), goto44i, NEUTRAL_REACTION);
     end
     display_msg("sadg");
 end
 
 procedure goto145
 begin
-    gsay_message(0, message_str(320, 222), 50);
+    gsay_message(0, message_str(SCRIPT_MICHAEL, 222), NEUTRAL_REACTION);
 end
 
 procedure goto146
 begin
-    gsay_reply(0, message_str(320, 223));
+    gsay_reply(0, message_str(SCRIPT_MICHAEL, 223));
     temp2 := map_var(12) - 1;
     set_map_var(12, temp2);
-    item := create_object_sid(9, 0, 0, -1);
+    item := create_object(9, 0, 0);
     add_obj_to_inven(dude_obj, item);
-    giq_option(4, 320, message_str(320, 224), gotoend, 50);
+    giq_option(4, SCRIPT_MICHAEL, message_str(SCRIPT_MICHAEL, 224), gotoend, NEUTRAL_REACTION);
     if (map_var(9) > 0) then begin
-        giq_option(4, 320, message_str(320, 225), goto43, 50);
+        giq_option(4, SCRIPT_MICHAEL, message_str(SCRIPT_MICHAEL, 225), goto43, NEUTRAL_REACTION);
     end
 end
 
 procedure goto147
 begin
-    gsay_reply(0, message_str(320, 226));
+    gsay_reply(0, message_str(SCRIPT_MICHAEL, 226));
     temp2 := map_var(13) - 1;
     set_map_var(13, temp2);
-    item := create_object_sid(22, 0, 0, -1);
+    item := create_object(22, 0, 0);
     add_obj_to_inven(dude_obj, item);
-    giq_option(4, 320, message_str(320, 227), gotoend, 50);
+    giq_option(4, SCRIPT_MICHAEL, message_str(SCRIPT_MICHAEL, 227), gotoend, NEUTRAL_REACTION);
     if (map_var(9) > 0) then begin
-        giq_option(4, 320, message_str(320, 228), goto43, 50);
+        giq_option(4, SCRIPT_MICHAEL, message_str(SCRIPT_MICHAEL, 228), goto43, NEUTRAL_REACTION);
     end
 end
 
 procedure goto148
 begin
-    gsay_reply(0, message_str(320, 229));
+    gsay_reply(0, message_str(SCRIPT_MICHAEL, 229));
     temp2 := map_var(14) - 1;
     set_map_var(14, temp2);
-    item := create_object_sid(17, 0, 0, -1);
+    item := create_object(17, 0, 0);
     add_obj_to_inven(dude_obj, item);
-    giq_option(4, 320, message_str(320, 230), gotoend, 50);
+    giq_option(4, SCRIPT_MICHAEL, message_str(SCRIPT_MICHAEL, 230), gotoend, NEUTRAL_REACTION);
     if (map_var(9) > 0) then begin
-        giq_option(4, 320, message_str(320, 231), goto43, 50);
+        giq_option(4, SCRIPT_MICHAEL, message_str(SCRIPT_MICHAEL, 231), goto43, NEUTRAL_REACTION);
     end
 end
 
 procedure goto149
 begin
-    gsay_reply(0, message_str(320, 232));
+    gsay_reply(0, message_str(SCRIPT_MICHAEL, 232));
     temp2 := map_var(15) - 1;
     set_map_var(15, temp2);
-    item := create_object_sid(3, 0, 0, -1);
+    item := create_object(3, 0, 0);
     add_obj_to_inven(dude_obj, item);
-    giq_option(4, 320, message_str(320, 233), gotoend, 50);
+    giq_option(4, SCRIPT_MICHAEL, message_str(SCRIPT_MICHAEL, 233), gotoend, NEUTRAL_REACTION);
     if (map_var(9) > 0) then begin
-        giq_option(4, 320, message_str(320, 234), goto43, 50);
+        giq_option(4, SCRIPT_MICHAEL, message_str(SCRIPT_MICHAEL, 234), goto43, NEUTRAL_REACTION);
     end
 end
 
 procedure goto150
 begin
-    gsay_reply(0, message_str(320, 235));
+    gsay_reply(0, message_str(SCRIPT_MICHAEL, 235));
     temp2 := map_var(16) - 1;
     set_map_var(16, temp2);
-    item := create_object_sid(52, 0, 0, -1);
+    item := create_object(52, 0, 0);
     add_obj_to_inven(dude_obj, item);
-    giq_option(4, 320, message_str(320, 236), gotoend, 50);
+    giq_option(4, SCRIPT_MICHAEL, message_str(SCRIPT_MICHAEL, 236), gotoend, NEUTRAL_REACTION);
     if (map_var(9) > 0) then begin
-        giq_option(4, 320, message_str(320, 237), goto43, 50);
+        giq_option(4, SCRIPT_MICHAEL, message_str(SCRIPT_MICHAEL, 237), goto43, NEUTRAL_REACTION);
     end
 end
 
@@ -793,7 +793,7 @@ procedure critter_p_proc
 begin
     if (hostile) then begin
         hostile := 0;
-        attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+        attack(dude_obj);
     end
 end
 
@@ -817,14 +817,14 @@ end
 procedure destroy_p_proc
 begin
     if (source_obj == dude_obj) then begin
-        set_global_var(159, global_var(159) + 1);
+        set_global_var(GVAR_GOOD_MONSTER, global_var(GVAR_GOOD_MONSTER) + 1);
     end
 end
 
 procedure look_at_p_proc
 begin
     script_overrides;
-    display_msg(message_str(320, 100));
+    display_msg(message_str(SCRIPT_MICHAEL, 100));
 end
 
 procedure get_reaction
@@ -833,31 +833,31 @@ begin
         set_local_var(0, 50);
         set_local_var(1, 2);
         set_local_var(2, 1);
-        set_local_var(0, local_var(0) + (5 * get_critter_stat(dude_obj, 3)) - 25);
-        set_local_var(0, local_var(0) + (10 * has_trait(0, dude_obj, 10)));
-        if (has_trait(0, dude_obj, 39)) then begin
-            if (global_var(155) > 0) then begin
-                set_local_var(0, local_var(0) + global_var(155));
+        set_local_var(0, local_var(0) + (5 * dude_charisma) - 25);
+        set_local_var(0, local_var(0) + (10 * has_trait(TRAIT_PERK, dude_obj, PERK_presence)));
+        if (has_trait(TRAIT_PERK, dude_obj, PERK_cult_of_personality)) then begin
+            if (global_var(GVAR_PLAYER_REPUATION) > 0) then begin
+                set_local_var(0, local_var(0) + global_var(GVAR_PLAYER_REPUATION));
             end
             else begin
-                set_local_var(0, local_var(0) - global_var(155));
+                set_local_var(0, local_var(0) - global_var(GVAR_PLAYER_REPUATION));
             end
         end
         else begin
             if (local_var(3) == 1) then begin
-                set_local_var(0, local_var(0) - global_var(155));
+                set_local_var(0, local_var(0) - global_var(GVAR_PLAYER_REPUATION));
             end
             else begin
-                set_local_var(0, local_var(0) + global_var(155));
+                set_local_var(0, local_var(0) + global_var(GVAR_PLAYER_REPUATION));
             end
         end
-        if (global_var(158) > 2) then begin
+        if (global_var(GVAR_CHILDKILLER_REPUATION) > 2) then begin
             set_local_var(0, local_var(0) - 30);
         end
-        if ((global_var(160) > (3 * global_var(159))) or (global_var(157) == 1)) then begin
+        if ((global_var(GVAR_BAD_MONSTER) > (3 * global_var(GVAR_GOOD_MONSTER))) or (global_var(GVAR_CHAMPION_REPUTATION) == 1)) then begin
             set_local_var(0, local_var(0) + 20);
         end
-        if ((global_var(159) > (2 * global_var(160))) or (global_var(156) == 1)) then begin
+        if ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1)) then begin
             set_local_var(0, local_var(0) - 20);
         end
         call ReactToLevel();
@@ -950,6 +950,5 @@ end
 
 procedure Goodbyes
 begin
-    exit_line := message_str(634, random(100, 105));
+    exit_line := message_str(SCRIPT_MODREACT, random(100, 105));
 end
-
