diff --git a/Games/Fallout1.Steam/Original/MLOPS.ssl b/Games/Fallout1.Steam/ReDefine/MLOPS.ssl
index 4fe61f7..6450fe5 100644
--- a/Games/Fallout1.Steam/Original/MLOPS.ssl
+++ b/Games/Fallout1.Steam/ReDefine/MLOPS.ssl
@@ -17,25 +17,25 @@ procedure start
 begin
     if (Only_Once) then begin
         Only_Once := 0;
-        critter_add_trait(self_obj, 1, 6, 34);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 34);
     end
-    if (script_action == 21) then begin
+    if (script_action == look_at_proc) then begin
         call look_at_p_proc();
     end
     else begin
-        if (script_action == 4) then begin
+        if (script_action == pickup_proc) then begin
             call pickup_p_proc();
         end
         else begin
-            if (script_action == 11) then begin
+            if (script_action == talk_proc) then begin
                 call talk_p_proc();
             end
             else begin
-                if (script_action == 12) then begin
+                if (script_action == critter_proc) then begin
                     call critter_p_proc();
                 end
                 else begin
-                    if (script_action == 18) then begin
+                    if (script_action == destroy_proc) then begin
                         call destroy_p_proc();
                     end
                 end
@@ -53,7 +53,7 @@ procedure critter_p_proc
 begin
     if (hostile) then begin
         hostile := 0;
-        attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+        attack(dude_obj);
     end
 end
 
@@ -66,17 +66,17 @@ end
 
 procedure talk_p_proc
 begin
-    if (obj_pid(critter_inven_obj(dude_obj, 0)) == 113) then begin
+    if (obj_pid(critter_inven_obj(dude_obj, INVEN_TYPE_WORN)) == 113) then begin
         if (local_var(0) == 1) then begin
-            float_msg(self_obj, message_str(811, 102), 2);
+            float_msg(self_obj, mstr(102), FLOAT_MSG_RED);
         end
         else begin
-            float_msg(self_obj, message_str(811, 101), 2);
+            float_msg(self_obj, mstr(101), FLOAT_MSG_RED);
         end
         set_local_var(0, 1);
     end
     else begin
-        float_msg(self_obj, message_str(811, 103), 2);
+        float_msg(self_obj, mstr(103), FLOAT_MSG_RED);
         call combat();
     end
 end
@@ -84,17 +84,17 @@ end
 procedure destroy_p_proc
 begin
     if (source_obj == dude_obj) then begin
-        if (((global_var(160) + global_var(159)) >= 25) and ((global_var(159) > (2 * global_var(160))) or (global_var(156) == 1))) then begin
-            set_global_var(156, 1);
-            set_global_var(157, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
+            set_global_var(GVAR_BERSERKER_REPUTATION, 1);
+            set_global_var(GVAR_CHAMPION_REPUTATION, 0);
         end
-        if (((global_var(160) + global_var(159)) >= 25) and ((global_var(160) > (3 * global_var(159))) or (global_var(157) == 1))) then begin
-            set_global_var(157, 1);
-            set_global_var(156, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(160) > (3 * global_var(GVAR_GOOD_MONSTER))) or (global_var(GVAR_CHAMPION_REPUTATION) == 1))) then begin
+            set_global_var(GVAR_CHAMPION_REPUTATION, 1);
+            set_global_var(GVAR_BERSERKER_REPUTATION, 0);
         end
-        set_global_var(160, global_var(160) + 1);
-        if ((global_var(160) % 6) == 0) then begin
-            set_global_var(155, global_var(155) + 1);
+        set_global_var(GVAR_BAD_MONSTER, global_var(GVAR_BAD_MONSTER) + 1);
+        if ((global_var(GVAR_BAD_MONSTER) % 6) == 0) then begin
+            set_global_var(GVAR_PLAYER_REPUATION, global_var(GVAR_PLAYER_REPUATION) + 1);
         end
     end
 end
@@ -102,7 +102,7 @@ end
 procedure look_at_p_proc
 begin
     script_overrides;
-    display_msg(message_str(811, 100));
+    display_mstr(100);
 end
 
 procedure dialog_end
@@ -112,4 +112,3 @@ end
 procedure gotonull
 begin
 end
-
