diff --git a/Games/Fallout1.Steam/Original/MRHANDYB.ssl b/Games/Fallout1.Steam/ReDefine/MRHANDYB.ssl
index 3d05d65..3a03d57 100644
--- a/Games/Fallout1.Steam/Original/MRHANDYB.ssl
+++ b/Games/Fallout1.Steam/ReDefine/MRHANDYB.ssl
@@ -11,18 +11,18 @@ import variable valid_target;
 procedure start
 begin
     if (not(initialized)) then begin
-        critter_add_trait(self_obj, 1, 6, 34);
-        critter_add_trait(self_obj, 1, 5, 71);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 34);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_ROBOTS);
         initialized := 1;
     end
-    if (global_var(273) == 2) then begin
-        critter_add_trait(self_obj, 1, 6, 85);
+    if (global_var(GVAR_DESTROY_VATS_1) == 2) then begin
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 85);
     end
-    if (script_action == 12) then begin
+    if (script_action == critter_proc) then begin
         call critter_p_proc();
     end
     else begin
-        if (script_action == 8) then begin
+        if (script_action == use_skill_on_proc) then begin
             call use_skill_on_p_proc();
         end
     end
@@ -32,49 +32,49 @@ procedure critter_p_proc
 begin
     if (hostile) then begin
         hostile := 0;
-        attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+        attack(dude_obj);
     end
-    if (global_var(275) == 0) then begin
+    if (global_var(GVAR_DESTROY_VATS_3) == 0) then begin
         if ((critter_state(self_obj) bwand 64) == 0) then begin
             critter_injure(self_obj, 64);
         end
     end
-    if (global_var(274) != 0) then begin
+    if (global_var(GVAR_DESTROY_VATS_2) != 0) then begin
         if ((critter_state(self_obj) bwand 4) == 0) then begin
             critter_injure(self_obj, 4);
         end
     end
-    if (global_var(274) == 2) then begin
+    if (global_var(GVAR_DESTROY_VATS_2) == 2) then begin
         if ((critter_state(self_obj) bwand 8) == 0) then begin
             critter_injure(self_obj, 8);
         end
     end
-    if (global_var(146) == 1) then begin
-        if (global_var(273) == 0) then begin
+    if (global_var(GVAR_VATS_ALERT) == 1) then begin
+        if (global_var(GVAR_DESTROY_VATS_1) == 0) then begin
             if (obj_can_see_obj(self_obj, dude_obj)) then begin
                 hostile := 1;
             end
         end
         else begin
-            if (global_var(273) == 1) then begin
+            if (global_var(GVAR_DESTROY_VATS_1) == 1) then begin
                 if (obj_can_see_obj(self_obj, valid_target)) then begin
-                    attack_complex(valid_target, 0, 1, 0, 0, 30000, 0, 0);
+                    attack(valid_target);
                 end
             end
             else begin
-                if (global_var(273) == 2) then begin
+                if (global_var(GVAR_DESTROY_VATS_1) == 2) then begin
                     if (obj_can_see_obj(self_obj, valid_target)) then begin
-                        attack_complex(valid_target, 0, 1, 0, 0, 30000, 0, 0);
+                        attack(valid_target);
                     end
                 end
                 else begin
-                    if (global_var(273) == 3) then begin
+                    if (global_var(GVAR_DESTROY_VATS_1) == 3) then begin
                         if (obj_can_see_obj(self_obj, dude_obj)) then begin
                             hostile := 1;
                         end
                         else begin
                             if (obj_can_see_obj(self_obj, valid_target)) then begin
-                                attack_complex(valid_target, 0, 1, 0, 0, 30000, 0, 0);
+                                attack(valid_target);
                             end
                         end
                     end
@@ -91,4 +91,3 @@ begin
         script_overrides;
     end
 end
-
