diff --git a/Games/Fallout1.Steam/Original/MSTMERCH.ssl b/Games/Fallout1.Steam/ReDefine/MSTMERCH.ssl
index f3ab103..bc15753 100644
--- a/Games/Fallout1.Steam/Original/MSTMERCH.ssl
+++ b/Games/Fallout1.Steam/ReDefine/MSTMERCH.ssl
@@ -52,30 +52,30 @@ procedure start
 begin
     if (Only_Once) then begin
         Only_Once := 0;
-        critter_add_trait(self_obj, 1, 6, 35);
-        critter_add_trait(self_obj, 1, 5, 50);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 35);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_MERCHANT);
     end
-    if (script_action == 21) then begin
+    if (script_action == look_at_proc) then begin
         call look_at_p_proc();
     end
     else begin
-        if (script_action == 4) then begin
+        if (script_action == pickup_proc) then begin
             call pickup_p_proc();
         end
         else begin
-            if (script_action == 11) then begin
+            if (script_action == talk_proc) then begin
                 call talk_p_proc();
             end
             else begin
-                if (script_action == 12) then begin
+                if (script_action == critter_proc) then begin
                     call critter_p_proc();
                 end
                 else begin
-                    if (script_action == 18) then begin
+                    if (script_action == destroy_proc) then begin
                         call destroy_p_proc();
                     end
                     else begin
-                        if (script_action == 14) then begin
+                        if (script_action == damage_proc) then begin
                             call damage_p_proc();
                         end
                     end
@@ -94,7 +94,7 @@ procedure critter_p_proc
 begin
     if (hostile) then begin
         hostile := 0;
-        attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+        attack(dude_obj);
     end
 end
 
@@ -110,7 +110,7 @@ procedure talk_p_proc
 begin
     call get_reaction();
     if (map_var(2) == 0) then begin
-        float_msg(self_obj, message_str(598, 156), 2);
+        float_msg(self_obj, mstr(156), FLOAT_MSG_RED);
     end
     else begin
         if (local_var(4) == 0) then begin
@@ -122,7 +122,7 @@ begin
             end_dialogue;
         end
         else begin
-            if ((local_var(5) > 0) and (global_var(101) != 1)) then begin
+            if ((local_var(5) > 0) and (global_var(GVAR_FIND_WATER_CHIP) != 1)) then begin
                 start_gdialog(598, self_obj, 4, -1, -1);
                 gsay_start;
                 call MasterMerch19();
@@ -143,17 +143,17 @@ end
 procedure destroy_p_proc
 begin
     if (source_obj == dude_obj) then begin
-        if (((global_var(160) + global_var(159)) >= 25) and ((global_var(159) > (2 * global_var(160))) or (global_var(156) == 1))) then begin
-            set_global_var(156, 1);
-            set_global_var(157, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
+            set_global_var(GVAR_BERSERKER_REPUTATION, 1);
+            set_global_var(GVAR_CHAMPION_REPUTATION, 0);
         end
-        if (((global_var(160) + global_var(159)) >= 25) and ((global_var(160) > (3 * global_var(159))) or (global_var(157) == 1))) then begin
-            set_global_var(157, 1);
-            set_global_var(156, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(160) > (3 * global_var(GVAR_GOOD_MONSTER))) or (global_var(GVAR_CHAMPION_REPUTATION) == 1))) then begin
+            set_global_var(GVAR_CHAMPION_REPUTATION, 1);
+            set_global_var(GVAR_BERSERKER_REPUTATION, 0);
         end
-        set_global_var(159, global_var(159) + 1);
-        if ((global_var(159) % 2) == 0) then begin
-            set_global_var(155, global_var(155) - 1);
+        set_global_var(GVAR_GOOD_MONSTER, global_var(GVAR_GOOD_MONSTER) + 1);
+        if ((global_var(GVAR_GOOD_MONSTER) % 2) == 0) then begin
+            set_global_var(GVAR_PLAYER_REPUATION, global_var(GVAR_PLAYER_REPUATION) - 1);
         end
     end
 end
@@ -161,7 +161,7 @@ end
 procedure look_at_p_proc
 begin
     script_overrides;
-    display_msg(message_str(598, 100));
+    display_mstr(100);
 end
 
 procedure damage_p_proc
@@ -169,126 +169,126 @@ begin
     variable LVar0 := 0;
     LVar0 := obj_pid(source_obj);
     if (party_member_obj(LVar0) != 0) then begin
-        set_global_var(248, 1);
+        set_global_var(GVAR_ENEMY_HUB, 1);
     end
 end
 
 procedure MasterMerch00
 begin
-    float_msg(self_obj, message_str(598, 101), 2);
+    float_msg(self_obj, mstr(101), FLOAT_MSG_RED);
     call combat();
 end
 
 procedure MasterMerch01
 begin
-    gsay_reply(598, 102);
-    giq_option(4, 598, 103, MasterMerch02, 50);
-    if ((global_var(101) != 2) or ((global_var(30) == 1) and (global_var(31) == 0))) then begin
-        giq_option(4, 598, 104, MasterMerch04, 50);
+    Reply(102);
+    giq_option(4, NAME, 103, MasterMerch02, NEUTRAL_REACTION);
+    if ((global_var(GVAR_FIND_WATER_CHIP) != 2) or ((global_var(GVAR_NECROP_WATER_CHIP_TAKEN) == 1) and (global_var(GVAR_NECROP_WATER_PUMP_FIXED) == 0))) then begin
+        giq_option(4, NAME, 104, MasterMerch04, NEUTRAL_REACTION);
     end
-    giq_option(4, 598, 105, MasterMerchEnd, 50);
-    giq_option(-3, 598, 106, MasterMerch03, 50);
+    giq_option(4, NAME, 105, MasterMerchEnd, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 106, MasterMerch03, NEUTRAL_REACTION);
 end
 
 procedure MasterMerch02
 begin
     set_map_var(1, 1);
-    gsay_reply(598, 107);
-    giq_option(4, 598, 108, MasterMerchEnd, 50);
-    if ((global_var(101) != 2) or ((global_var(30) == 1) and (global_var(31) == 0))) then begin
-        giq_option(4, 598, 109, MasterMerch04, 50);
+    Reply(107);
+    giq_option(4, NAME, 108, MasterMerchEnd, NEUTRAL_REACTION);
+    if ((global_var(GVAR_FIND_WATER_CHIP) != 2) or ((global_var(GVAR_NECROP_WATER_CHIP_TAKEN) == 1) and (global_var(GVAR_NECROP_WATER_PUMP_FIXED) == 0))) then begin
+        giq_option(4, NAME, 109, MasterMerch04, NEUTRAL_REACTION);
     end
-    giq_option(-3, 598, 110, MasterMerch03, 50);
+    giq_option(-3, NAME, 110, MasterMerch03, NEUTRAL_REACTION);
 end
 
 procedure MasterMerch03
 begin
-    gsay_message(598, 111, 50);
+    gsay_message(NAME, 111, NEUTRAL_REACTION);
 end
 
 procedure MasterMerch04
 begin
-    if (global_var(72) == 0) then begin
-        set_global_var(72, 1);
+    if (global_var(GVAR_NECROPOLIS) == 0) then begin
+        set_global_var(GVAR_NECROPOLIS, 1);
     end
-    gsay_reply(598, 112);
-    giq_option(4, 598, 113, MasterMerch13, 50);
-    giq_option(4, 598, 114, MasterMerch05, 50);
-    giq_option(4, 598, 115, MasterMerch06, 50);
-    giq_option(4, 598, 116, MasterMerch07, 50);
+    Reply(112);
+    giq_option(4, NAME, 113, MasterMerch13, NEUTRAL_REACTION);
+    giq_option(4, NAME, 114, MasterMerch05, NEUTRAL_REACTION);
+    giq_option(4, NAME, 115, MasterMerch06, NEUTRAL_REACTION);
+    giq_option(4, NAME, 116, MasterMerch07, NEUTRAL_REACTION);
 end
 
 procedure MasterMerch05
 begin
     set_local_var(6, 1);
-    gsay_reply(598, 117);
-    giq_option(4, 598, 118, MasterMerch14, 50);
-    giq_option(4, 598, 119, MasterMerch08, 50);
-    giq_option(4, 598, 120, MasterMerch07, 50);
+    Reply(117);
+    giq_option(4, NAME, 118, MasterMerch14, NEUTRAL_REACTION);
+    giq_option(4, NAME, 119, MasterMerch08, NEUTRAL_REACTION);
+    giq_option(4, NAME, 120, MasterMerch07, NEUTRAL_REACTION);
 end
 
 procedure MasterMerch06
 begin
-    gsay_message(598, 121, 50);
+    gsay_message(NAME, 121, NEUTRAL_REACTION);
 end
 
 procedure MasterMerch07
 begin
-    gsay_message(598, 122, 50);
+    gsay_message(NAME, 122, NEUTRAL_REACTION);
 end
 
 procedure MasterMerch08
 begin
-    gsay_reply(598, 123);
-    giq_option(4, 598, 124, MasterMerch14, 50);
-    giq_option(4, 598, 125, MasterMerch09, 50);
+    Reply(123);
+    giq_option(4, NAME, 124, MasterMerch14, NEUTRAL_REACTION);
+    giq_option(4, NAME, 125, MasterMerch09, NEUTRAL_REACTION);
 end
 
 procedure MasterMerch09
 begin
-    gsay_message(598, 126, 50);
+    gsay_message(NAME, 126, NEUTRAL_REACTION);
 end
 
 procedure MasterMerch10
 begin
-    gsay_reply(598, 127);
-    if ((global_var(101) != 2) and (global_var(101) != 1) and (local_var(6) == 0)) then begin
-        giq_option(4, 598, 128, MasterMerch04, 50);
+    Reply(127);
+    if ((global_var(GVAR_FIND_WATER_CHIP) != 2) and (global_var(GVAR_FIND_WATER_CHIP) != 1) and (local_var(6) == 0)) then begin
+        giq_option(4, NAME, 128, MasterMerch04, NEUTRAL_REACTION);
     end
     else begin
-        if ((global_var(101) != 2) and (global_var(101) != 1) and (local_var(6) == 1)) then begin
-            giq_option(4, 598, 129, MasterMerch12, 50);
+        if ((global_var(GVAR_FIND_WATER_CHIP) != 2) and (global_var(GVAR_FIND_WATER_CHIP) != 1) and (local_var(6) == 1)) then begin
+            giq_option(4, NAME, 129, MasterMerch12, NEUTRAL_REACTION);
         end
     end
-    giq_option(4, 598, 130, MasterMerch11, 50);
-    giq_option(4, 598, 131, MasterMerchEnd, 50);
-    giq_option(-3, 598, 106, MasterMerch03, 50);
+    giq_option(4, NAME, 130, MasterMerch11, NEUTRAL_REACTION);
+    giq_option(4, NAME, 131, MasterMerchEnd, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 106, MasterMerch03, NEUTRAL_REACTION);
 end
 
 procedure MasterMerch11
 begin
     set_map_var(1, 1);
-    gsay_message(598, 132, 50);
+    gsay_message(NAME, 132, NEUTRAL_REACTION);
 end
 
 procedure MasterMerch12
 begin
-    gsay_reply(598, 133);
-    giq_option(4, 598, 134, MasterMerch13, 50);
-    giq_option(4, 598, 135, MasterMerch14, 50);
-    giq_option(4, 598, 136, MasterMerchEnd, 50);
+    Reply(133);
+    giq_option(4, NAME, 134, MasterMerch13, NEUTRAL_REACTION);
+    giq_option(4, NAME, 135, MasterMerch14, NEUTRAL_REACTION);
+    giq_option(4, NAME, 136, MasterMerchEnd, NEUTRAL_REACTION);
 end
 
 procedure MasterMerch13
 begin
-    gsay_reply(598, 137);
-    giq_option(4, 598, 138, MasterMerch14, 50);
-    giq_option(4, 598, 139, MasterMerch07, 50);
+    Reply(137);
+    giq_option(4, NAME, 138, MasterMerch14, NEUTRAL_REACTION);
+    giq_option(4, NAME, 139, MasterMerch07, NEUTRAL_REACTION);
 end
 
 procedure MasterMerch14
 begin
-    gsay_reply(598, 140);
+    Reply(140);
     call MasterMerch15();
 end
 
@@ -297,26 +297,26 @@ begin
     if (local_var(5) == 0) then begin
         set_local_var(5, 2000);
     end
-    giq_option(4, 598, 141, MasterMerch16, 50);
+    giq_option(4, NAME, 141, MasterMerch16, NEUTRAL_REACTION);
     if (local_var(7) == 0) then begin
-        giq_option(4, 598, 142, MasterMerch17, 50);
+        giq_option(4, NAME, 142, MasterMerch17, NEUTRAL_REACTION);
     end
     if (local_var(7) == 0) then begin
-        giq_option(4, 598, 143, MasterMerch18, 50);
+        giq_option(4, NAME, 143, MasterMerch18, NEUTRAL_REACTION);
     end
-    giq_option(4, 598, 144, MasterMerchEnd, 50);
+    giq_option(4, NAME, 144, MasterMerchEnd, NEUTRAL_REACTION);
 end
 
 procedure MasterMerch16
 begin
     if (item_caps_total(dude_obj) < local_var(5)) then begin
-        gsay_message(598, message_str(598, 145) + local_var(5) + message_str(598, 146), 50);
+        gsay_message(NAME, mstr(145) + local_var(5) + mstr(146), NEUTRAL_REACTION);
         call MasterMerchEnd();
     end
     else begin
-        gsay_reply(598, 147);
-        giq_option(4, 598, 148, MasterMerchEnd, 50);
-        giq_option(4, 598, 149, MasterMerch20, 50);
+        Reply(147);
+        giq_option(4, NAME, 148, MasterMerchEnd, NEUTRAL_REACTION);
+        giq_option(4, NAME, 149, MasterMerch20, NEUTRAL_REACTION);
     end
 end
 
@@ -325,11 +325,11 @@ begin
     set_local_var(7, 1);
     if (is_success(roll_vs_skill(dude_obj, 15, -15))) then begin
         set_local_var(5, 1000);
-        gsay_message(598, 150, 50);
+        gsay_message(NAME, 150, NEUTRAL_REACTION);
         call MasterMerch16();
     end
     else begin
-        gsay_reply(598, 151);
+        Reply(151);
         call MasterMerch15();
     end
 end
@@ -339,18 +339,18 @@ begin
     set_local_var(7, 1);
     if (is_success(roll_vs_skill(dude_obj, 15, -30))) then begin
         set_local_var(5, 500);
-        gsay_message(598, 152, 50);
+        gsay_message(NAME, 152, NEUTRAL_REACTION);
         call MasterMerch16();
     end
     else begin
-        gsay_reply(598, 153);
+        Reply(153);
         call MasterMerch15();
     end
 end
 
 procedure MasterMerch19
 begin
-    gsay_reply(598, message_str(598, 157) + local_var(5) + message_str(598, 158));
+    Reply(mstr(157) + local_var(5) + mstr(158));
     call MasterMerch15();
 end
 
@@ -358,14 +358,14 @@ procedure MasterMerch20
 begin
     variable LVar0 := 0;
     LVar0 := item_caps_adjust(dude_obj, -1 * local_var(5));
-    set_global_var(101, 1);
-    set_global_var(10, global_var(10) + 100);
-    if (global_var(154) >= 100) then begin
-        set_global_var(154, global_var(154) - 90);
+    set_global_var(GVAR_FIND_WATER_CHIP, 1);
+    set_global_var(GVAR_VAULT_WATER, global_var(GVAR_VAULT_WATER) + 100);
+    if (global_var(GVAR_VAULT_13_INVADED_DATE) >= 100) then begin
+        set_global_var(GVAR_VAULT_13_INVADED_DATE, global_var(GVAR_VAULT_13_INVADED_DATE) - 90);
     end
     give_exp_points(1000);
-    display_msg(message_str(766, 103) + 1000 + message_str(766, 104));
-    gsay_message(598, 155, 50);
+    display_msg(message_str(SCRIPT_GENCHAT, 103) + 1000 + message_str(SCRIPT_GENCHAT, 104));
+    gsay_message(NAME, 155, NEUTRAL_REACTION);
 end
 
 procedure MasterMerchEnd
@@ -378,31 +378,31 @@ begin
         set_local_var(0, 50);
         set_local_var(1, 2);
         set_local_var(2, 1);
-        set_local_var(0, local_var(0) + (5 * get_critter_stat(dude_obj, 3)) - 25);
-        set_local_var(0, local_var(0) + (10 * has_trait(0, dude_obj, 10)));
-        if (has_trait(0, dude_obj, 39)) then begin
-            if (global_var(155) > 0) then begin
-                set_local_var(0, local_var(0) + global_var(155));
+        set_local_var(0, local_var(0) + (5 * dude_charisma) - 25);
+        set_local_var(0, local_var(0) + (10 * has_trait(TRAIT_PERK, dude_obj, PERK_presence)));
+        if (has_trait(TRAIT_PERK, dude_obj, PERK_cult_of_personality)) then begin
+            if (global_var(GVAR_PLAYER_REPUATION) > 0) then begin
+                set_local_var(0, local_var(0) + global_var(GVAR_PLAYER_REPUATION));
             end
             else begin
-                set_local_var(0, local_var(0) - global_var(155));
+                set_local_var(0, local_var(0) - global_var(GVAR_PLAYER_REPUATION));
             end
         end
         else begin
             if (local_var(3) == 1) then begin
-                set_local_var(0, local_var(0) - global_var(155));
+                set_local_var(0, local_var(0) - global_var(GVAR_PLAYER_REPUATION));
             end
             else begin
-                set_local_var(0, local_var(0) + global_var(155));
+                set_local_var(0, local_var(0) + global_var(GVAR_PLAYER_REPUATION));
             end
         end
-        if (global_var(158) > 2) then begin
+        if (global_var(GVAR_CHILDKILLER_REPUATION) > 2) then begin
             set_local_var(0, local_var(0) - 30);
         end
-        if (((global_var(160) + global_var(159)) >= 25) and ((global_var(160) > (3 * global_var(159))) or (global_var(157) == 1))) then begin
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(160) > (3 * global_var(GVAR_GOOD_MONSTER))) or (global_var(GVAR_CHAMPION_REPUTATION) == 1))) then begin
             set_local_var(0, local_var(0) + 20);
         end
-        if (((global_var(160) + global_var(159)) >= 25) and ((global_var(159) > (2 * global_var(160))) or (global_var(156) == 1))) then begin
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
             set_local_var(0, local_var(0) - 20);
         end
         call ReactToLevel();
@@ -495,6 +495,5 @@ end
 
 procedure Goodbyes
 begin
-    exit_line := message_str(634, random(100, 105));
+    exit_line := message_str(SCRIPT_MODREACT, random(100, 105));
 end
-
