diff --git a/Games/Fallout1.Steam/Original/MUTANT4.ssl b/Games/Fallout1.Steam/ReDefine/MUTANT4.ssl
index 4d9d6a1..6540abc 100644
--- a/Games/Fallout1.Steam/Original/MUTANT4.ssl
+++ b/Games/Fallout1.Steam/ReDefine/MUTANT4.ssl
@@ -5,8 +5,8 @@ procedure start;
 
 procedure start
 begin
-    if (script_action == 12) then begin
-        if (global_var(13) == 0) then begin
+    if (script_action == critter_proc) then begin
+        if (global_var(GVAR_NECROPOLIS_INVADED) == 0) then begin
             set_obj_visibility(self_obj, 1);
         end
         else begin
@@ -18,7 +18,7 @@ begin
         end
     end
     else begin
-        if (script_action == 22) then begin
+        if (script_action == timed_event_proc) then begin
             if (fixed_param == 1) then begin
                 animate_move_obj_to_tile(self_obj, 10917, 0);
                 add_timer_event(self_obj, game_ticks(60), 2);
@@ -31,16 +31,15 @@ begin
             end
         end
         else begin
-            if (script_action == 21) then begin
+            if (script_action == look_at_proc) then begin
                 script_overrides;
                 display_msg(message_str(0, 100));
             end
             else begin
-                if (script_action == 18) then begin
-                    set_global_var(35, global_var(35) + 1);
+                if (script_action == destroy_proc) then begin
+                    set_global_var(GVAR_SUPER_MUTANTS_KILLED, global_var(GVAR_SUPER_MUTANTS_KILLED) + 1);
                 end
             end
         end
     end
 end
-
