diff --git a/Games/Fallout1.Steam/Original/MUTANT2.ssl b/Games/Fallout1.Steam/ReDefine/MUTANT2.ssl
index 4b15a48..7dd4de7 100644
--- a/Games/Fallout1.Steam/Original/MUTANT2.ssl
+++ b/Games/Fallout1.Steam/ReDefine/MUTANT2.ssl
@@ -21,30 +21,30 @@ variable new_tile;
 procedure start
 begin
     if (not(init_teams)) then begin
-        critter_add_trait(self_obj, 1, 6, 34);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 34);
         init_teams := 1;
     end
-    if (script_action == 22) then begin
+    if (script_action == timed_event_proc) then begin
         call timeforwhat();
     end
     else begin
         if (local_var(2) == 0) then begin
             set_local_var(2, tile_num(self_obj));
         end
-        if (script_action == 11) then begin
+        if (script_action == talk_proc) then begin
             call do_dialogue();
         end
         else begin
-            if (script_action == 4) then begin
+            if (script_action == pickup_proc) then begin
                 Hostile := 1;
             end
             else begin
-                if (script_action == 12) then begin
-                    if (global_var(13) == 0) then begin
+                if (script_action == critter_proc) then begin
+                    if (global_var(GVAR_NECROPOLIS_INVADED) == 0) then begin
                         set_obj_visibility(self_obj, 1);
                     end
                     else begin
-                        maxsight := get_critter_stat(self_obj, 1);
+                        maxsight := self_perception;
                         time := game_time_hour;
                         if ((time > 2300) or (time < 600)) then begin
                             maxsight := 4;
@@ -52,7 +52,7 @@ begin
                         if ((tile_distance_objs(self_obj, dude_obj) <= maxsight) and obj_can_see_obj(self_obj, dude_obj)) then begin
                             if ((Hostile > 0) or (local_var(0) > 0)) then begin
                                 Hostile := 0;
-                                attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+                                attack(dude_obj);
                             end
                             else begin
                                 call do_dialogue();
@@ -70,26 +70,26 @@ begin
                     end
                 end
                 else begin
-                    if (script_action == 18) then begin
-                        set_global_var(35, global_var(35) + 1);
+                    if (script_action == destroy_proc) then begin
+                        set_global_var(GVAR_SUPER_MUTANTS_KILLED, global_var(GVAR_SUPER_MUTANTS_KILLED) + 1);
                         if (source_obj == dude_obj) then begin
-                            if (((global_var(160) + global_var(159)) >= 25) and ((global_var(159) > (2 * global_var(160))) or (global_var(156) == 1))) then begin
-                                set_global_var(156, 1);
-                                set_global_var(157, 0);
+                            if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
+                                set_global_var(GVAR_BERSERKER_REPUTATION, 1);
+                                set_global_var(GVAR_CHAMPION_REPUTATION, 0);
                             end
-                            if (((global_var(160) + global_var(159)) >= 25) and ((global_var(160) > (3 * global_var(159))) or (global_var(157) == 1))) then begin
-                                set_global_var(157, 1);
-                                set_global_var(156, 0);
+                            if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(160) > (3 * global_var(GVAR_GOOD_MONSTER))) or (global_var(GVAR_CHAMPION_REPUTATION) == 1))) then begin
+                                set_global_var(GVAR_CHAMPION_REPUTATION, 1);
+                                set_global_var(GVAR_BERSERKER_REPUTATION, 0);
                             end
-                            set_global_var(160, global_var(160) + 1);
-                            if ((global_var(160) % 6) == 0) then begin
-                                set_global_var(155, global_var(155) + 1);
+                            set_global_var(GVAR_BAD_MONSTER, global_var(GVAR_BAD_MONSTER) + 1);
+                            if ((global_var(GVAR_BAD_MONSTER) % 6) == 0) then begin
+                                set_global_var(GVAR_PLAYER_REPUATION, global_var(GVAR_PLAYER_REPUATION) + 1);
                             end
                         end
                     end
                     else begin
-                        if (script_action == 21) then begin
-                            display_msg(message_str(13, 100));
+                        if (script_action == look_at_proc) then begin
+                            display_mstr(100);
                         end
                     end
                 end
@@ -103,11 +103,11 @@ begin
     start_gdialog(13, self_obj, 4, -1, -1);
     gsay_start;
     set_local_var(0, 1);
-    if (get_critter_stat(dude_obj, 34) == 0) then begin
-        gsay_message(13, 101, 50);
+    if (dude_gender == 0) then begin
+        gsay_message(NAME, 101, NEUTRAL_REACTION);
     end
     else begin
-        gsay_message(13, 102, 50);
+        gsay_message(NAME, 102, NEUTRAL_REACTION);
     end
     gsay_end;
     end_dialogue;
@@ -138,4 +138,3 @@ end
 procedure genericcbt
 begin
 end
-
