diff --git a/Games/Fallout1.Steam/Original/MVDWN2UP.ssl b/Games/Fallout1.Steam/ReDefine/MVDWN2UP.ssl
index 5762426..7e96cd5 100644
--- a/Games/Fallout1.Steam/Original/MVDWN2UP.ssl
+++ b/Games/Fallout1.Steam/ReDefine/MVDWN2UP.ssl
@@ -3,8 +3,7 @@ procedure start;
 
 procedure start
 begin
-    if (script_action == 6) then begin
+    if (script_action == use_proc) then begin
         load_map("master12.map", 1);
     end
 end
-
