diff --git a/Games/Fallout1.Steam/Original/NICOLE.ssl b/Games/Fallout1.Steam/ReDefine/NICOLE.ssl
index 57905b7..0b205de 100644
--- a/Games/Fallout1.Steam/Original/NICOLE.ssl
+++ b/Games/Fallout1.Steam/ReDefine/NICOLE.ssl
@@ -80,7 +80,7 @@ begin
         if (obj_is_carrying_obj_pid(self_obj, 41) == 0) then begin
             item_caps_adjust(self_obj, random(50, 150));
         end
-        critter_add_trait(self_obj, 1, 6, 46);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 46);
         Initialize := 0;
     end
 end
@@ -88,23 +88,23 @@ end
 procedure look_at_p_proc
 begin
     script_overrides;
-    display_msg(message_str(54, 100));
+    display_mstr(100);
 end
 
 procedure description_p_proc
 begin
     script_overrides;
-    display_msg(message_str(54, 100));
+    display_mstr(100);
 end
 
 procedure talk_p_proc
 begin
-    if (global_var(256) == 1) then begin
-        float_msg(self_obj, message_str(669, random(100, 105)), 2);
+    if (global_var(GVAR_ENEMY_FOLLOWERS) == 1) then begin
+        float_msg(self_obj, message_str(SCRIPT_ENEMY, random(100, 105)), FLOAT_MSG_RED);
     end
     else begin
-        if ((local_var(4) == 1) and (get_critter_stat(dude_obj, 4) < 4)) then begin
-            float_msg(self_obj, message_str(54, 286), 0);
+        if ((local_var(4) == 1) and (dude_iq < 4)) then begin
+            float_msg(self_obj, mstr(286), FLOAT_MSG_NORMAL);
         end
         else begin
             if (local_var(2) == 0) then begin
@@ -134,26 +134,26 @@ end
 procedure damage_p_proc
 begin
     if (source_obj == dude_obj) then begin
-        set_global_var(256, 1);
+        set_global_var(GVAR_ENEMY_FOLLOWERS, 1);
     end
 end
 
 procedure destroy_p_proc
 begin
     if (source_obj == dude_obj) then begin
-        set_global_var(256, 1);
+        set_global_var(GVAR_ENEMY_FOLLOWERS, 1);
         if (source_obj == dude_obj) then begin
-            if (((global_var(160) + global_var(159)) >= 25) and ((global_var(159) > (2 * global_var(160))) or (global_var(156) == 1))) then begin
-                set_global_var(156, 1);
-                set_global_var(157, 0);
+            if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
+                set_global_var(GVAR_BERSERKER_REPUTATION, 1);
+                set_global_var(GVAR_CHAMPION_REPUTATION, 0);
             end
-            if (((global_var(160) + global_var(159)) >= 25) and ((global_var(160) > (3 * global_var(159))) or (global_var(157) == 1))) then begin
-                set_global_var(157, 1);
-                set_global_var(156, 0);
+            if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(160) > (3 * global_var(GVAR_GOOD_MONSTER))) or (global_var(GVAR_CHAMPION_REPUTATION) == 1))) then begin
+                set_global_var(GVAR_CHAMPION_REPUTATION, 1);
+                set_global_var(GVAR_BERSERKER_REPUTATION, 0);
             end
-            set_global_var(159, global_var(159) + 1);
-            if ((global_var(159) % 2) == 0) then begin
-                set_global_var(155, global_var(155) - 1);
+            set_global_var(GVAR_GOOD_MONSTER, global_var(GVAR_GOOD_MONSTER) + 1);
+            if ((global_var(GVAR_GOOD_MONSTER) % 2) == 0) then begin
+                set_global_var(GVAR_PLAYER_REPUATION, global_var(GVAR_PLAYER_REPUATION) - 1);
             end
         end
     end
@@ -162,95 +162,95 @@ end
 procedure critter_p_proc
 begin
     if (obj_can_see_obj(self_obj, dude_obj)) then begin
-        if (global_var(256) == 1) then begin
-            attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+        if (global_var(GVAR_ENEMY_FOLLOWERS) == 1) then begin
+            attack(dude_obj);
         end
     end
 end
 
 procedure pickup_p_proc
 begin
-    set_global_var(256, 1);
+    set_global_var(GVAR_ENEMY_FOLLOWERS, 1);
 end
 
 procedure Nicole00
 begin
-    gsay_reply(54, DisplayMessage);
-    gsay_option(54, 101, Nicole29, 50);
-    gsay_option(54, 102, Nicole02, 50);
-    if (global_var(18) == 0) then begin
-        giq_option(4, 54, 319, Nicole50, 50);
+    Reply(DisplayMessage);
+    giq_option(1, NAME, 101, Nicole29, NEUTRAL_REACTION);
+    giq_option(1, NAME, 102, Nicole02, NEUTRAL_REACTION);
+    if (global_var(GVAR_MASTER_BLOWN) == 0) then begin
+        giq_option(4, NAME, 319, Nicole50, NEUTRAL_REACTION);
     end
-    gsay_option(54, 131, Nicole14, 50);
+    giq_option(1, NAME, 131, Nicole14, NEUTRAL_REACTION);
 end
 
 procedure Nicole01
 begin
-    gsay_reply(54, 105);
-    giq_option(4, 54, 106, Nicole02, 50);
-    giq_option(4, 54, 107, Nicole11, 51);
-    giq_option(4, 54, 108, Nicole09, 50);
-    giq_option(4, 54, 109, Nicole15, 50);
-    giq_option(-3, 54, 110, Nicole78, 50);
+    Reply(105);
+    giq_option(4, NAME, 106, Nicole02, NEUTRAL_REACTION);
+    giq_option(4, NAME, 107, Nicole11, BAD_REACTION);
+    giq_option(4, NAME, 108, Nicole09, NEUTRAL_REACTION);
+    giq_option(4, NAME, 109, Nicole15, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 110, Nicole78, NEUTRAL_REACTION);
     set_local_var(4, 1);
 end
 
 procedure Nicole02
 begin
-    gsay_reply(54, 111);
-    gsay_option(54, 112, Nicole04, 50);
+    Reply(111);
+    giq_option(1, NAME, 112, Nicole04, NEUTRAL_REACTION);
 end
 
 procedure Nicole04
 begin
-    gsay_reply(54, 113);
-    gsay_option(54, 114, Nicole05, 50);
-    gsay_option(54, 115, Nicole11, 51);
+    Reply(113);
+    giq_option(1, NAME, 114, Nicole05, NEUTRAL_REACTION);
+    giq_option(1, NAME, 115, Nicole11, BAD_REACTION);
 end
 
 procedure Nicole05
 begin
-    gsay_reply(54, 116);
-    gsay_option(54, 117, Nicole06, 50);
+    Reply(116);
+    giq_option(1, NAME, 117, Nicole06, NEUTRAL_REACTION);
 end
 
 procedure Nicole06
 begin
-    gsay_reply(54, 118);
-    gsay_option(54, 119, Nicole07, 50);
-    gsay_option(54, 123, Nicole08, 50);
+    Reply(118);
+    giq_option(1, NAME, 119, Nicole07, NEUTRAL_REACTION);
+    giq_option(1, NAME, 123, Nicole08, NEUTRAL_REACTION);
     if (local_var(1) >= 2) then begin
-        gsay_option(54, 121, Nicole08a, 50);
+        giq_option(1, NAME, 121, Nicole08a, NEUTRAL_REACTION);
     end
     else begin
-        gsay_option(54, 121, Nicole11, 51);
+        giq_option(1, NAME, 121, Nicole11, BAD_REACTION);
     end
 end
 
 procedure Nicole07
 begin
-    gsay_reply(54, 122);
-    gsay_option(54, 123, Nicole08, 50);
-    gsay_option(54, 124, Nicole28, 50);
-    gsay_option(54, 125, Nicole19, 49);
+    Reply(122);
+    giq_option(1, NAME, 123, Nicole08, NEUTRAL_REACTION);
+    giq_option(1, NAME, 124, Nicole28, NEUTRAL_REACTION);
+    giq_option(1, NAME, 125, Nicole19, GOOD_REACTION);
     if (local_var(1) >= 2) then begin
-        gsay_option(54, 126, Nicole08a, 50);
+        giq_option(1, NAME, 126, Nicole08a, NEUTRAL_REACTION);
     end
     else begin
-        gsay_option(54, 126, Nicole11, 51);
+        giq_option(1, NAME, 126, Nicole11, BAD_REACTION);
     end
 end
 
 procedure Nicole08
 begin
-    gsay_message(54, 127, 50);
+    gsay_message(NAME, 127, NEUTRAL_REACTION);
     DisplayMessage := 129;
     call NicoleMore();
 end
 
 procedure Nicole08a
 begin
-    gsay_message(54, 128, 50);
+    gsay_message(NAME, 128, NEUTRAL_REACTION);
     DisplayMessage := 129;
     call NicoleMore();
 end
@@ -263,22 +263,22 @@ end
 
 procedure Nicole10
 begin
-    gsay_message(54, 133, 50);
+    gsay_message(NAME, 133, NEUTRAL_REACTION);
     call NicoleShowOut();
 end
 
 procedure Nicole11
 begin
     call DownReactLevel();
-    gsay_reply(54, 134);
-    gsay_option(54, 135, Nicole12, 50);
-    gsay_option(54, 136, Nicole13, 50);
+    Reply(134);
+    giq_option(1, NAME, 135, Nicole12, NEUTRAL_REACTION);
+    giq_option(1, NAME, 136, Nicole13, NEUTRAL_REACTION);
 end
 
 procedure Nicole12
 begin
     call BottomReact();
-    gsay_message(54, 137, 50);
+    gsay_message(NAME, 137, NEUTRAL_REACTION);
     call NicoleShowOut();
 end
 
@@ -290,34 +290,34 @@ end
 
 procedure Nicole14
 begin
-    gsay_message(54, 141, 50);
+    gsay_message(NAME, 141, NEUTRAL_REACTION);
 end
 
 procedure Nicole15
 begin
-    gsay_reply(54, 142);
-    gsay_option(54, 143, Nicole16, 50);
+    Reply(142);
+    giq_option(1, NAME, 143, Nicole16, NEUTRAL_REACTION);
 end
 
 procedure Nicole16
 begin
-    gsay_reply(54, 144);
-    gsay_option(54, 145, Nicole17, 50);
-    gsay_option(54, 146, Nicole25, 50);
+    Reply(144);
+    giq_option(1, NAME, 145, Nicole17, NEUTRAL_REACTION);
+    giq_option(1, NAME, 146, Nicole25, NEUTRAL_REACTION);
 end
 
 procedure Nicole17
 begin
-    gsay_reply(54, 147);
-    gsay_option(54, 148, Nicole07, 50);
-    gsay_option(54, 149, Nicole18, 50);
+    Reply(147);
+    giq_option(1, NAME, 148, Nicole07, NEUTRAL_REACTION);
+    giq_option(1, NAME, 149, Nicole18, NEUTRAL_REACTION);
 end
 
 procedure Nicole18
 begin
-    gsay_reply(54, 151);
-    gsay_option(54, 152, Nicole19, 49);
-    gsay_option(54, 153, Nicole20, 50);
+    Reply(151);
+    giq_option(1, NAME, 152, Nicole19, GOOD_REACTION);
+    giq_option(1, NAME, 153, Nicole20, NEUTRAL_REACTION);
 end
 
 procedure Nicole19
@@ -328,48 +328,48 @@ end
 
 procedure Nicole20
 begin
-    gsay_reply(54, 157);
-    gsay_option(54, 158, Nicole21, 50);
-    gsay_option(54, 159, Nicole21a, 51);
+    Reply(157);
+    giq_option(1, NAME, 158, Nicole21, NEUTRAL_REACTION);
+    giq_option(1, NAME, 159, Nicole21a, BAD_REACTION);
     if (local_var(1) >= 2) then begin
-        gsay_option(54, 160, Nicole09, 50);
+        giq_option(1, NAME, 160, Nicole09, NEUTRAL_REACTION);
     end
     else begin
-        gsay_option(54, 160, Nicole10, 51);
+        giq_option(1, NAME, 160, Nicole10, BAD_REACTION);
     end
-    gsay_option(54, 164, Nicole14, 50);
+    giq_option(1, NAME, 164, Nicole14, NEUTRAL_REACTION);
 end
 
 procedure Nicole21
 begin
-    gsay_reply(54, 161);
+    Reply(161);
     if (local_var(1) >= 2) then begin
-        gsay_option(54, 163, Nicole09, 50);
+        giq_option(1, NAME, 163, Nicole09, NEUTRAL_REACTION);
     end
     else begin
-        gsay_option(54, 163, Nicole10, 51);
+        giq_option(1, NAME, 163, Nicole10, BAD_REACTION);
     end
-    gsay_option(54, 164, Nicole14, 50);
+    giq_option(1, NAME, 164, Nicole14, NEUTRAL_REACTION);
 end
 
 procedure Nicole21a
 begin
-    gsay_reply(54, 162);
+    Reply(162);
     if (local_var(1) >= 2) then begin
-        gsay_option(54, 163, Nicole09, 50);
+        giq_option(1, NAME, 163, Nicole09, NEUTRAL_REACTION);
     end
     else begin
-        gsay_option(54, 163, Nicole10, 51);
+        giq_option(1, NAME, 163, Nicole10, BAD_REACTION);
     end
-    gsay_option(54, 164, Nicole14, 50);
+    giq_option(1, NAME, 164, Nicole14, NEUTRAL_REACTION);
 end
 
 procedure Nicole25
 begin
-    gsay_reply(54, 175);
-    gsay_option(54, 176, Nicole26, 50);
-    gsay_option(54, 177, Nicole11, 50);
-    gsay_option(54, 178, Nicole27, 50);
+    Reply(175);
+    giq_option(1, NAME, 176, Nicole26, NEUTRAL_REACTION);
+    giq_option(1, NAME, 177, Nicole11, NEUTRAL_REACTION);
+    giq_option(1, NAME, 178, Nicole27, NEUTRAL_REACTION);
 end
 
 procedure Nicole26
@@ -386,88 +386,88 @@ end
 
 procedure Nicole28
 begin
-    gsay_reply(54, 185);
-    gsay_option(54, 186, Nicole19, 49);
+    Reply(185);
+    giq_option(1, NAME, 186, Nicole19, GOOD_REACTION);
     if (local_var(1) >= 2) then begin
-        gsay_option(54, 187, Nicole09, 50);
+        giq_option(1, NAME, 187, Nicole09, NEUTRAL_REACTION);
     end
     else begin
-        gsay_option(54, 187, Nicole10, 51);
+        giq_option(1, NAME, 187, Nicole10, BAD_REACTION);
     end
-    gsay_option(54, 188, Nicole14, 50);
+    giq_option(1, NAME, 188, Nicole14, NEUTRAL_REACTION);
 end
 
 procedure Nicole29
 begin
-    gsay_reply(54, 189);
-    gsay_option(54, 190, Nicole30, 50);
-    gsay_option(54, 191, Nicole31, 50);
-    gsay_option(54, 192, Nicole04, 50);
+    Reply(189);
+    giq_option(1, NAME, 190, Nicole30, NEUTRAL_REACTION);
+    giq_option(1, NAME, 191, Nicole31, NEUTRAL_REACTION);
+    giq_option(1, NAME, 192, Nicole04, NEUTRAL_REACTION);
 end
 
 procedure Nicole30
 begin
-    gsay_reply(54, 193);
-    gsay_option(54, 194, Nicole31, 50);
-    gsay_option(54, 195, Nicole36, 50);
+    Reply(193);
+    giq_option(1, NAME, 194, Nicole31, NEUTRAL_REACTION);
+    giq_option(1, NAME, 195, Nicole36, NEUTRAL_REACTION);
 end
 
 procedure Nicole31
 begin
-    gsay_reply(54, 196);
-    gsay_option(54, 197, Nicole39, 50);
-    gsay_option(54, 198, Nicole11, 50);
-    gsay_option(54, 199, Nicole32, 50);
-    gsay_option(54, 195, Nicole36, 50);
+    Reply(196);
+    giq_option(1, NAME, 197, Nicole39, NEUTRAL_REACTION);
+    giq_option(1, NAME, 198, Nicole11, NEUTRAL_REACTION);
+    giq_option(1, NAME, 199, Nicole32, NEUTRAL_REACTION);
+    giq_option(1, NAME, 195, Nicole36, NEUTRAL_REACTION);
 end
 
 procedure Nicole32
 begin
-    gsay_reply(54, 201);
-    gsay_option(54, 202, Nicole33, 50);
-    gsay_option(54, 195, Nicole36, 50);
+    Reply(201);
+    giq_option(1, NAME, 202, Nicole33, NEUTRAL_REACTION);
+    giq_option(1, NAME, 195, Nicole36, NEUTRAL_REACTION);
 end
 
 procedure Nicole33
 begin
-    gsay_reply(54, 204);
-    gsay_option(54, 205, Nicole11, 50);
-    gsay_option(54, 206, Nicole34, 50);
-    gsay_option(54, 207, Nicole36, 50);
+    Reply(204);
+    giq_option(1, NAME, 205, Nicole11, NEUTRAL_REACTION);
+    giq_option(1, NAME, 206, Nicole34, NEUTRAL_REACTION);
+    giq_option(1, NAME, 207, Nicole36, NEUTRAL_REACTION);
 end
 
 procedure Nicole34
 begin
-    gsay_reply(54, 208);
-    gsay_option(54, 209, Nicole36, 50);
-    gsay_option(54, 210, Nicole35, 50);
+    Reply(208);
+    giq_option(1, NAME, 209, Nicole36, NEUTRAL_REACTION);
+    giq_option(1, NAME, 210, Nicole35, NEUTRAL_REACTION);
     if (local_var(1) >= 2) then begin
-        gsay_option(54, 211, Nicole09, 50);
+        giq_option(1, NAME, 211, Nicole09, NEUTRAL_REACTION);
     end
     else begin
-        gsay_option(54, 211, Nicole10, 51);
+        giq_option(1, NAME, 211, Nicole10, BAD_REACTION);
     end
-    gsay_option(54, 212, Nicole14, 50);
+    giq_option(1, NAME, 212, Nicole14, NEUTRAL_REACTION);
 end
 
 procedure Nicole35
 begin
-    gsay_message(54, 213, 50);
+    gsay_message(NAME, 213, NEUTRAL_REACTION);
     call DownReactLevel();
 end
 
 procedure Nicole36
 begin
-    gsay_reply(54, 214);
-    gsay_option(54, 215, Nicole37, 50);
-    gsay_option(54, 216, Nicole41, 50);
+    Reply(214);
+    giq_option(1, NAME, 215, Nicole37, NEUTRAL_REACTION);
+    giq_option(1, NAME, 216, Nicole41, NEUTRAL_REACTION);
 end
 
 procedure Nicole37
 begin
-    gsay_reply(54, 217);
-    gsay_option(54, 218, Nicole39, 50);
-    gsay_option(54, 219, Nicole41, 50);
+    Reply(217);
+    giq_option(1, NAME, 218, Nicole39, NEUTRAL_REACTION);
+    giq_option(1, NAME, 219, Nicole41, NEUTRAL_REACTION);
 end
 
 procedure Nicole39
@@ -479,27 +479,27 @@ end
 
 procedure Nicole41
 begin
-    gsay_reply(54, 223);
-    gsay_option(54, 224, Nicole00, 50);
-    gsay_option(54, 225, NicoleEnd, 50);
+    Reply(223);
+    giq_option(1, NAME, 224, Nicole00, NEUTRAL_REACTION);
+    giq_option(1, NAME, 225, NicoleEnd, NEUTRAL_REACTION);
 end
 
 procedure Nicole50
 begin
     DisplayMessage := 132;
-    gsay_reply(54, 307);
-    gsay_option(54, 320, Nicole51, 50);
-    gsay_option(54, 321, Nicole00, 50);
+    Reply(307);
+    giq_option(1, NAME, 320, Nicole51, NEUTRAL_REACTION);
+    giq_option(1, NAME, 321, Nicole00, NEUTRAL_REACTION);
 end
 
 procedure Nicole51
 begin
-    gsay_reply(54, DisplayMessage);
-    gsay_option(54, 322, Nicole52, 50);
-    gsay_option(54, 323, Nicole53, 50);
-    gsay_option(54, 324, Nicole54, 50);
-    gsay_option(54, 321, Nicole00, 50);
-    gsay_option(54, 326, Nicole14, 50);
+    Reply(DisplayMessage);
+    giq_option(1, NAME, 322, Nicole52, NEUTRAL_REACTION);
+    giq_option(1, NAME, 323, Nicole53, NEUTRAL_REACTION);
+    giq_option(1, NAME, 324, Nicole54, NEUTRAL_REACTION);
+    giq_option(1, NAME, 321, Nicole00, NEUTRAL_REACTION);
+    giq_option(1, NAME, 326, Nicole14, NEUTRAL_REACTION);
 end
 
 procedure Nicole52
@@ -516,32 +516,32 @@ end
 
 procedure Nicole54
 begin
-    gsay_reply(54, 251);
-    gsay_option(54, 325, Nicole55, 50);
-    set_global_var(63, 1);
+    Reply(251);
+    giq_option(1, NAME, 325, Nicole55, NEUTRAL_REACTION);
+    set_global_var(GVAR_FOLLOWER_MACHO_HELP, 1);
 end
 
 procedure Nicole55
 begin
-    set_global_var(61, 1);
-    gsay_message(54, 254, 50);
+    set_global_var(GVAR_SHADOW_PASSWORD, 1);
+    gsay_message(NAME, 254, NEUTRAL_REACTION);
 end
 
 procedure Nicole78
 begin
-    gsay_message(54, 291, 50);
+    gsay_message(NAME, 291, NEUTRAL_REACTION);
 end
 
 procedure NicoleMore
 begin
-    gsay_reply(54, DisplayMessage);
+    Reply(DisplayMessage);
     if (local_var(1) >= 2) then begin
-        gsay_option(54, 155, Nicole09, 50);
+        giq_option(1, NAME, 155, Nicole09, NEUTRAL_REACTION);
     end
     else begin
-        gsay_option(54, 155, Nicole10, 51);
+        giq_option(1, NAME, 155, Nicole10, BAD_REACTION);
     end
-    gsay_option(54, 131, Nicole14, 50);
+    giq_option(1, NAME, 131, Nicole14, NEUTRAL_REACTION);
 end
 
 procedure NicoleShowOut
@@ -561,31 +561,31 @@ begin
         set_local_var(0, 50);
         set_local_var(1, 2);
         set_local_var(2, 1);
-        set_local_var(0, local_var(0) + (5 * get_critter_stat(dude_obj, 3)) - 25);
-        set_local_var(0, local_var(0) + (10 * has_trait(0, dude_obj, 10)));
-        if (has_trait(0, dude_obj, 39)) then begin
-            if (global_var(155) > 0) then begin
-                set_local_var(0, local_var(0) + global_var(155));
+        set_local_var(0, local_var(0) + (5 * dude_charisma) - 25);
+        set_local_var(0, local_var(0) + (10 * has_trait(TRAIT_PERK, dude_obj, PERK_presence)));
+        if (has_trait(TRAIT_PERK, dude_obj, PERK_cult_of_personality)) then begin
+            if (global_var(GVAR_PLAYER_REPUATION) > 0) then begin
+                set_local_var(0, local_var(0) + global_var(GVAR_PLAYER_REPUATION));
             end
             else begin
-                set_local_var(0, local_var(0) - global_var(155));
+                set_local_var(0, local_var(0) - global_var(GVAR_PLAYER_REPUATION));
             end
         end
         else begin
             if (local_var(3) == 1) then begin
-                set_local_var(0, local_var(0) - global_var(155));
+                set_local_var(0, local_var(0) - global_var(GVAR_PLAYER_REPUATION));
             end
             else begin
-                set_local_var(0, local_var(0) + global_var(155));
+                set_local_var(0, local_var(0) + global_var(GVAR_PLAYER_REPUATION));
             end
         end
-        if (global_var(158) > 2) then begin
+        if (global_var(GVAR_CHILDKILLER_REPUATION) > 2) then begin
             set_local_var(0, local_var(0) - 30);
         end
-        if (((global_var(160) + global_var(159)) >= 25) and ((global_var(160) > (3 * global_var(159))) or (global_var(157) == 1))) then begin
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(160) > (3 * global_var(GVAR_GOOD_MONSTER))) or (global_var(GVAR_CHAMPION_REPUTATION) == 1))) then begin
             set_local_var(0, local_var(0) + 20);
         end
-        if (((global_var(160) + global_var(159)) >= 25) and ((global_var(159) > (2 * global_var(160))) or (global_var(156) == 1))) then begin
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
             set_local_var(0, local_var(0) - 20);
         end
         call ReactToLevel();
@@ -678,6 +678,5 @@ end
 
 procedure Goodbyes
 begin
-    exit_line := message_str(634, random(100, 105));
+    exit_line := message_str(SCRIPT_MODREACT, random(100, 105));
 end
-
