diff --git a/Games/Fallout1.Steam/Original/NWUP2DN2.ssl b/Games/Fallout1.Steam/ReDefine/NWUP2DN2.ssl
index 9c3b820..a996785 100644
--- a/Games/Fallout1.Steam/Original/NWUP2DN2.ssl
+++ b/Games/Fallout1.Steam/ReDefine/NWUP2DN2.ssl
@@ -13,7 +13,7 @@ procedure use_bottom;
 
 procedure start
 begin
-    if (script_action == 6) then begin
+    if (script_action == use_proc) then begin
         move_to(dude_obj, 16289, 0);
     end
 end
@@ -95,4 +95,3 @@ begin
         end
     end
 end
-
