diff --git a/Games/Fallout1.Steam/Original/ORFEO.ssl b/Games/Fallout1.Steam/ReDefine/ORFEO.ssl
index 4c8c580..0add5a1 100644
--- a/Games/Fallout1.Steam/Original/ORFEO.ssl
+++ b/Games/Fallout1.Steam/ReDefine/ORFEO.ssl
@@ -18,27 +18,27 @@ variable hostile;
 procedure start
 begin
     if (not(initialized)) then begin
-        critter_add_trait(self_obj, 1, 6, 20);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 20);
         initialized := 1;
     end
     else begin
-        if (script_action == 12) then begin
+        if (script_action == critter_proc) then begin
             call critter_p_proc();
         end
         else begin
-            if (script_action == 18) then begin
+            if (script_action == destroy_proc) then begin
                 call destroy_p_proc();
             end
             else begin
-                if (script_action == 21) then begin
+                if (script_action == look_at_proc) then begin
                     call look_at_p_proc();
                 end
                 else begin
-                    if (script_action == 4) then begin
+                    if (script_action == pickup_proc) then begin
                         call pickup_p_proc();
                     end
                     else begin
-                        if (script_action == 11) then begin
+                        if (script_action == talk_proc) then begin
                             call talk_p_proc();
                         end
                     end
@@ -52,24 +52,24 @@ procedure critter_p_proc
 begin
     if (hostile) then begin
         hostile := 0;
-        attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+        attack(dude_obj);
     end
 end
 
 procedure destroy_p_proc
 begin
     if (source_obj == dude_obj) then begin
-        if (((global_var(160) + global_var(159)) >= 25) and ((global_var(159) > (2 * global_var(160))) or (global_var(156) == 1))) then begin
-            set_global_var(156, 1);
-            set_global_var(157, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
+            set_global_var(GVAR_BERSERKER_REPUTATION, 1);
+            set_global_var(GVAR_CHAMPION_REPUTATION, 0);
         end
-        if (((global_var(160) + global_var(159)) >= 25) and ((global_var(160) > (3 * global_var(159))) or (global_var(157) == 1))) then begin
-            set_global_var(157, 1);
-            set_global_var(156, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(160) > (3 * global_var(GVAR_GOOD_MONSTER))) or (global_var(GVAR_CHAMPION_REPUTATION) == 1))) then begin
+            set_global_var(GVAR_CHAMPION_REPUTATION, 1);
+            set_global_var(GVAR_BERSERKER_REPUTATION, 0);
         end
-        set_global_var(160, global_var(160) + 1);
-        if ((global_var(160) % 6) == 0) then begin
-            set_global_var(155, global_var(155) + 1);
+        set_global_var(GVAR_BAD_MONSTER, global_var(GVAR_BAD_MONSTER) + 1);
+        if ((global_var(GVAR_BAD_MONSTER) % 6) == 0) then begin
+            set_global_var(GVAR_PLAYER_REPUATION, global_var(GVAR_PLAYER_REPUATION) + 1);
         end
     end
 end
@@ -78,7 +78,7 @@ procedure look_at_p_proc
 begin
     if (local_var(0)) then begin
         script_overrides;
-        display_msg(message_str(269, 100));
+        display_mstr(100);
     end
 end
 
@@ -105,27 +105,27 @@ end
 
 procedure Orfeo0
 begin
-    gsay_reply(269, 101);
-    giq_option(-3, 269, 102, combat, 50);
-    giq_option(4, 269, 103, Orfeo1, 50);
-    giq_option(4, 269, 104, Orfeo2, 50);
+    Reply(101);
+    giq_option(-3, NAME, 102, combat, NEUTRAL_REACTION);
+    giq_option(4, NAME, 103, Orfeo1, NEUTRAL_REACTION);
+    giq_option(4, NAME, 104, Orfeo2, NEUTRAL_REACTION);
 end
 
 procedure Orfeo1
 begin
-    gsay_message(269, 105, 50);
+    gsay_message(NAME, 105, NEUTRAL_REACTION);
     call combat();
 end
 
 procedure Orfeo2
 begin
-    gsay_message(269, 106, 50);
+    gsay_message(NAME, 106, NEUTRAL_REACTION);
     call combat();
 end
 
 procedure Orfeo3
 begin
-    float_msg(self_obj, message_str(269, 107), 0);
+    float_msg(self_obj, mstr(107), FLOAT_MSG_NORMAL);
     call combat();
 end
 
@@ -137,4 +137,3 @@ procedure combat
 begin
     hostile := hostile + 1;
 end
-
