diff --git a/Games/Fallout1.Steam/Original/PRISFELD.ssl b/Games/Fallout1.Steam/ReDefine/PRISFELD.ssl
index 65036e6..ba5595f 100644
--- a/Games/Fallout1.Steam/Original/PRISFELD.ssl
+++ b/Games/Fallout1.Steam/ReDefine/PRISFELD.ssl
@@ -5,11 +5,11 @@ procedure map_update_p_proc;
 
 procedure start
 begin
-    if (script_action == 15) then begin
+    if (script_action == map_enter_proc) then begin
         call map_enter_p_proc();
     end
     else begin
-        if (script_action == 23) then begin
+        if (script_action == map_update_proc) then begin
             call map_update_p_proc();
         end
     end
@@ -35,4 +35,3 @@ begin
         set_obj_visibility(self_obj, 1);
     end
 end
-
