diff --git a/Games/Fallout1.Steam/Original/RAZLO.ssl b/Games/Fallout1.Steam/ReDefine/RAZLO.ssl
index 7b97dda..e057829 100644
--- a/Games/Fallout1.Steam/Original/RAZLO.ssl
+++ b/Games/Fallout1.Steam/ReDefine/RAZLO.ssl
@@ -106,31 +106,31 @@ variable exit_line;
 
 procedure start
 begin
-    if (script_action == 13) then begin
+    if (script_action == combat_proc) then begin
         call combat_p_proc();
     end
     else begin
-        if (script_action == 23) then begin
+        if (script_action == map_update_proc) then begin
             call map_update_p_proc();
         end
         else begin
-            if (script_action == 11) then begin
+            if (script_action == talk_proc) then begin
                 call talk_p_proc();
             end
             else begin
-                if (script_action == 4) then begin
+                if (script_action == pickup_proc) then begin
                     call pickup_p_proc();
                 end
                 else begin
-                    if (script_action == 12) then begin
+                    if (script_action == critter_proc) then begin
                         call critter_p_proc();
                     end
                     else begin
-                        if (script_action == 21) then begin
+                        if (script_action == look_at_proc) then begin
                             call look_at_p_proc();
                         end
                         else begin
-                            if (script_action == 18) then begin
+                            if (script_action == destroy_proc) then begin
                                 call destroy_p_proc();
                             end
                         end
@@ -147,9 +147,9 @@ begin
         round_counter := round_counter + 1;
     end
     if (round_counter > 3) then begin
-        if (global_var(246) == 0) then begin
-            set_global_var(246, 1);
-            set_global_var(155, global_var(155) - 5);
+        if (global_var(GVAR_ENEMY_SHADY_SANDS) == 0) then begin
+            set_global_var(GVAR_ENEMY_SHADY_SANDS, 1);
+            set_global_var(GVAR_PLAYER_REPUATION, global_var(GVAR_PLAYER_REPUATION) - 5);
         end
     end
 end
@@ -157,8 +157,8 @@ end
 procedure map_update_p_proc
 begin
     if (not(initialized)) then begin
-        critter_add_trait(self_obj, 1, 6, 2);
-        critter_add_trait(self_obj, 1, 5, 6);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 2);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_PEASANT);
         home_tile := 24529;
         sleep_tile := 23133;
         wake_time := 630;
@@ -169,19 +169,19 @@ end
 
 procedure talk_p_proc
 begin
-    MAX_HITS := get_critter_stat(dude_obj, 7);
-    HITS := get_critter_stat(dude_obj, 35);
+    MAX_HITS := get_critter_stat(dude_obj, STAT_max_hp);
+    HITS := get_critter_stat(dude_obj, STAT_current_hp);
     call get_reaction();
-    if (((global_var(160) + global_var(159)) >= 25) and ((global_var(159) > (2 * global_var(160))) or (global_var(156) == 1))) then begin
-        set_global_var(156, 1);
-        set_global_var(157, 0);
+    if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
+        set_global_var(GVAR_BERSERKER_REPUTATION, 1);
+        set_global_var(GVAR_CHAMPION_REPUTATION, 0);
     end
-    if (((global_var(160) + global_var(159)) >= 25) and ((global_var(160) > (3 * global_var(159))) or (global_var(157) == 1))) then begin
-        set_global_var(157, 1);
-        set_global_var(156, 0);
+    if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(160) > (3 * global_var(GVAR_GOOD_MONSTER))) or (global_var(GVAR_CHAMPION_REPUTATION) == 1))) then begin
+        set_global_var(GVAR_CHAMPION_REPUTATION, 1);
+        set_global_var(GVAR_BERSERKER_REPUTATION, 0);
     end
     if (local_var(9) == 1) then begin
-        float_msg(self_obj, message_str(185, 166), 0);
+        float_msg(self_obj, message_str(SCRIPT_GENVAULT, 166), FLOAT_MSG_NORMAL);
     end
     else begin
         if ((game_time_hour >= 1900) or (game_time_hour < 600)) then begin
@@ -193,8 +193,8 @@ begin
             end_dialogue;
         end
         else begin
-            if (global_var(43) == 1) then begin
-                if ((global_var(26) == 1) and (local_var(4) == 0) and (global_var(218) == 1)) then begin
+            if (global_var(GVAR_RADSCORPION_SEED) == 1) then begin
+                if ((global_var(GVAR_TANDI_STATUS) == 1) and (local_var(4) == 0) and (global_var(GVAR_TALKED_ABOUT_TANDI) == 1)) then begin
                     set_local_var(4, 1);
                     start_gdialog(129, self_obj, 4, -1, -1);
                     gsay_start;
@@ -237,16 +237,16 @@ begin
                             end_dialogue;
                         end
                         else begin
-                            if (global_var(26) == 3) then begin
+                            if (global_var(GVAR_TANDI_STATUS) == 3) then begin
                                 call razlo09();
                             end
                             else begin
-                                if ((global_var(26) == 2) and (local_var(5) == 0)) then begin
+                                if ((global_var(GVAR_TANDI_STATUS) == 2) and (local_var(5) == 0)) then begin
                                     set_local_var(5, 1);
                                     call razlo08();
                                 end
                                 else begin
-                                    if ((global_var(26) == 1) and (local_var(4) == 0) and (global_var(218) == 1)) then begin
+                                    if ((global_var(GVAR_TANDI_STATUS) == 1) and (local_var(4) == 0) and (global_var(GVAR_TALKED_ABOUT_TANDI) == 1)) then begin
                                         set_local_var(4, 1);
                                         start_gdialog(129, self_obj, 4, -1, -1);
                                         gsay_start;
@@ -255,18 +255,18 @@ begin
                                         end_dialogue;
                                     end
                                     else begin
-                                        if ((global_var(26) == 1) and (global_var(218) == 1) and (local_var(4) == 1)) then begin
+                                        if ((global_var(GVAR_TANDI_STATUS) == 1) and (global_var(GVAR_TALKED_ABOUT_TANDI) == 1) and (local_var(4) == 1)) then begin
                                             set_local_var(4, 2);
                                             call razlo07();
                                         end
                                         else begin
-                                            if (((global_var(160) + global_var(159)) >= 25) and ((global_var(159) > (2 * global_var(160))) or (global_var(156) == 1)) or (global_var(158) > 2)) then begin
+                                            if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1)) or (global_var(GVAR_CHILDKILLER_REPUATION) > 2)) then begin
                                                 start_gdialog(129, self_obj, 4, -1, -1);
                                                 gsay_start;
                                                 call razlo00();
                                                 gsay_end;
                                                 end_dialogue;
-                                                display_msg(message_str(129, 103));
+                                                display_mstr(103);
                                             end
                                             else begin
                                                 if (local_var(1) >= 2) then begin
@@ -299,61 +299,61 @@ end
 
 procedure razlo00
 begin
-    gsay_message(129, 101, 50);
+    gsay_message(NAME, 101, NEUTRAL_REACTION);
 end
 
 procedure razlo00a
 begin
-    display_msg(message_str(129, 103));
+    display_mstr(103);
 end
 
 procedure razlo01
 begin
-    gsay_reply(129, 104);
-    if ((global_var(26) == 1) and (global_var(218) == 1)) then begin
-        giq_option(4, 129, 105, razlo03, 50);
+    Reply(104);
+    if ((global_var(GVAR_TANDI_STATUS) == 1) and (global_var(GVAR_TALKED_ABOUT_TANDI) == 1)) then begin
+        giq_option(4, NAME, 105, razlo03, NEUTRAL_REACTION);
     end
-    giq_option(4, 129, 143, razlo22, 50);
-    giq_option(4, 129, 144, razlo23a, 50);
-    giq_option(-3, 129, 106, razlo02, 50);
+    giq_option(4, NAME, 143, razlo22, NEUTRAL_REACTION);
+    giq_option(4, NAME, 144, razlo23a, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 106, razlo02, NEUTRAL_REACTION);
 end
 
 procedure razlo02
 begin
-    gsay_message(129, 107, 50);
+    gsay_message(NAME, 107, NEUTRAL_REACTION);
     set_local_var(4, 1);
 end
 
 procedure razlo03
 begin
     set_local_var(4, 1);
-    gsay_reply(129, 108);
-    giq_option(5, 129, 110, razlo04, 50);
-    giq_option(4, 129, 109, razloend, 50);
-    giq_option(5, 129, 111, razloend, 50);
+    Reply(108);
+    giq_option(5, NAME, 110, razlo04, NEUTRAL_REACTION);
+    giq_option(4, NAME, 109, razloend, NEUTRAL_REACTION);
+    giq_option(5, NAME, 111, razloend, NEUTRAL_REACTION);
 end
 
 procedure razlo04
 begin
-    gsay_reply(129, 112);
-    giq_option(4, 129, 113, razlo05, 50);
-    giq_option(4, 129, 114, razlo06, 50);
+    Reply(112);
+    giq_option(4, NAME, 113, razlo05, NEUTRAL_REACTION);
+    giq_option(4, NAME, 114, razlo06, NEUTRAL_REACTION);
 end
 
 procedure razlo05
 begin
-    gsay_message(129, 115, 50);
+    gsay_message(NAME, 115, NEUTRAL_REACTION);
 end
 
 procedure razlo06
 begin
-    gsay_message(129, 116, 50);
+    gsay_message(NAME, 116, NEUTRAL_REACTION);
     call givestuff();
 end
 
 procedure razlo06na
 begin
-    gsay_message(129, 306, 50);
+    gsay_message(NAME, 306, NEUTRAL_REACTION);
     call razlo06nb();
 end
 
@@ -361,8 +361,8 @@ procedure razlo06nb
 begin
     variable LVar0 := 0;
     variable LVar1 := 0;
-    gsay_message(129, 307, 50);
-    LVar0 := create_object_sid(49, 0, 0, -1);
+    gsay_message(NAME, 307, NEUTRAL_REACTION);
+    LVar0 := create_object(49, 0, 0);
     add_obj_to_inven(dude_obj, LVar0);
     LVar1 := obj_carrying_pid_obj(dude_obj, 92);
     rm_obj_from_inven(dude_obj, LVar1);
@@ -370,7 +370,7 @@ begin
     if (local_var(7) == 0) then begin
         set_local_var(7, 1);
         give_exp_points(250);
-        display_msg(message_str(129, 300));
+        display_mstr(300);
     end
     gfade_out(400);
     gfade_in(400);
@@ -380,31 +380,31 @@ end
 
 procedure razlo07
 begin
-    float_msg(self_obj, message_str(129, 117), 2);
+    float_msg(self_obj, mstr(117), FLOAT_MSG_RED);
 end
 
 procedure razlo08
 begin
-    float_msg(self_obj, message_str(129, 118), 7);
+    float_msg(self_obj, mstr(118), FLOAT_MSG_LIGHT_RED);
 end
 
 procedure razlo09
 begin
-    float_msg(self_obj, message_str(129, 119), 12);
+    float_msg(self_obj, mstr(119), FLOAT_MSG_LIGHT_GREY);
 end
 
 procedure razlo10
 begin
-    gsay_reply(129, 120);
-    giq_option(4, 129, 122, razlo14, 50);
-    giq_option(4, 129, 144, razlo23a, 50);
-    giq_option(4, 129, 121, razloend, 50);
-    giq_option(-3, 129, 123, razlo11, 50);
+    Reply(120);
+    giq_option(4, NAME, 122, razlo14, NEUTRAL_REACTION);
+    giq_option(4, NAME, 144, razlo23a, NEUTRAL_REACTION);
+    giq_option(4, NAME, 121, razloend, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 123, razlo11, NEUTRAL_REACTION);
 end
 
 procedure razlo11
 begin
-    gsay_message(129, 124, 50);
+    gsay_message(NAME, 124, NEUTRAL_REACTION);
     damage := MAX_HITS - HITS;
     if (damage > (MAX_HITS / 4)) then begin
         call razlo14();
@@ -416,12 +416,12 @@ end
 
 procedure razlo12
 begin
-    gsay_message(129, 125, 50);
+    gsay_message(NAME, 125, NEUTRAL_REACTION);
 end
 
 procedure razlo14
 begin
-    gsay_message(129, 126, 50);
+    gsay_message(NAME, 126, NEUTRAL_REACTION);
     damage := MAX_HITS - HITS;
     if (damage < (MAX_HITS / 4)) then begin
         call razlowmpa();
@@ -443,80 +443,80 @@ end
 
 procedure razlo15
 begin
-    gsay_message(129, 127, 50);
+    gsay_message(NAME, 127, NEUTRAL_REACTION);
 end
 
 procedure Razlo17
 begin
     variable LVar0 := 0;
     if (damage < (MAX_HITS / 2)) then begin
-        LVar0 := create_object_sid(218, 0, 0, -1);
+        LVar0 := create_object(218, 0, 0);
         add_obj_to_inven(self_obj, LVar0);
     end
     else begin
         if (damage < (3 * MAX_HITS / 4)) then begin
-            LVar0 := create_object_sid(219, 0, 0, -1);
+            LVar0 := create_object(219, 0, 0);
             add_obj_to_inven(self_obj, LVar0);
         end
         else begin
-            LVar0 := create_object_sid(220, 0, 0, -1);
+            LVar0 := create_object(220, 0, 0);
             add_obj_to_inven(self_obj, LVar0);
         end
     end
-    gdialog_mod_barter(0);
-    gsay_message(129, 126, 50);
+    gdialog_barter;
+    gsay_message(NAME, 126, NEUTRAL_REACTION);
     call razlo08n();
 end
 
 procedure razlo18
 begin
-    gsay_reply(129, 128);
-    giq_option(4, 129, 130, razlo14, 50);
-    giq_option(4, 129, 144, razlo23a, 50);
-    giq_option(5, 129, 129, razloend, 50);
+    Reply(128);
+    giq_option(4, NAME, 130, razlo14, NEUTRAL_REACTION);
+    giq_option(4, NAME, 144, razlo23a, NEUTRAL_REACTION);
+    giq_option(5, NAME, 129, razloend, NEUTRAL_REACTION);
 end
 
 procedure razlo19
 begin
-    gsay_reply(129, 131);
-    giq_option(4, 129, 132, razlo20, 50);
-    giq_option(4, 129, 143, razlo22, 50);
-    giq_option(4, 129, 144, razlo23a, 50);
-    giq_option(-3, 129, 134, razlo11, 50);
+    Reply(131);
+    giq_option(4, NAME, 132, razlo20, NEUTRAL_REACTION);
+    giq_option(4, NAME, 143, razlo22, NEUTRAL_REACTION);
+    giq_option(4, NAME, 144, razlo23a, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 134, razlo11, NEUTRAL_REACTION);
 end
 
 procedure razlo20
 begin
-    gsay_message(129, 135, 50);
+    gsay_message(NAME, 135, NEUTRAL_REACTION);
 end
 
 procedure razlo21
 begin
-    gsay_reply(129, 136);
-    giq_option(4, 129, 138, razlo22, 50);
-    giq_option(4, 129, 144, razlo23a, 50);
-    giq_option(-3, 129, 139, razlo11, 50);
-    giq_option(4, 129, 137, razloend, 50);
+    Reply(136);
+    giq_option(4, NAME, 138, razlo22, NEUTRAL_REACTION);
+    giq_option(4, NAME, 144, razlo23a, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 139, razlo11, NEUTRAL_REACTION);
+    giq_option(4, NAME, 137, razloend, NEUTRAL_REACTION);
 end
 
 procedure razlo22
 begin
-    gsay_message(129, 140, 50);
+    gsay_message(NAME, 140, NEUTRAL_REACTION);
     call razlo14();
 end
 
 procedure razlo23
 begin
-    gsay_reply(129, 141);
-    giq_option(4, 129, 143, razlo22, 50);
-    giq_option(4, 129, 144, razlo23a, 50);
-    giq_option(5, 129, 145, razlo27, 50);
+    Reply(141);
+    giq_option(4, NAME, 143, razlo22, NEUTRAL_REACTION);
+    giq_option(4, NAME, 144, razlo23a, NEUTRAL_REACTION);
+    giq_option(5, NAME, 145, razlo27, NEUTRAL_REACTION);
     if (obj_is_carrying_obj_pid(dude_obj, 92)) then begin
-        giq_option(4, 129, 146, razlo30, 50);
-        giq_option(-3, 129, 303, razlo30, 50);
+        giq_option(4, NAME, 146, razlo30, NEUTRAL_REACTION);
+        giq_option(-3, NAME, 303, razlo30, NEUTRAL_REACTION);
     end
-    giq_option(4, 129, 142, razloend, 50);
-    giq_option(-3, 129, 147, razlo11, 50);
+    giq_option(4, NAME, 142, razloend, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 147, razlo11, NEUTRAL_REACTION);
 end
 
 procedure razlo23a
@@ -531,35 +531,35 @@ end
 
 procedure razlo24
 begin
-    gsay_message(129, 148, 50);
+    gsay_message(NAME, 148, NEUTRAL_REACTION);
 end
 
 procedure razlo25
 begin
-    gsay_message(129, 149, 50);
+    gsay_message(NAME, 149, NEUTRAL_REACTION);
     if (get_poison(dude_obj) > 0) then begin
-        gsay_message(129, 150, 50);
+        gsay_message(NAME, 150, NEUTRAL_REACTION);
         poison(dude_obj, -get_poison(dude_obj));
         call razlo26();
     end
     else begin
-        gsay_message(129, 151, 50);
+        gsay_message(NAME, 151, NEUTRAL_REACTION);
     end
     poison(dude_obj, -get_poison(dude_obj));
 end
 
 procedure razlo26
 begin
-    gsay_message(129, 152, 50);
+    gsay_message(NAME, 152, NEUTRAL_REACTION);
 end
 
 procedure razlo27
 begin
-    gsay_message(129, 153, 50);
-    gsay_reply(129, 154);
-    giq_option(5, 129, 156, razlo29, 50);
-    giq_option(5, 129, 157, razlo28, 50);
-    giq_option(4, 129, 155, razlo27a, 49);
+    gsay_message(NAME, 153, NEUTRAL_REACTION);
+    Reply(154);
+    giq_option(5, NAME, 156, razlo29, NEUTRAL_REACTION);
+    giq_option(5, NAME, 157, razlo28, NEUTRAL_REACTION);
+    giq_option(4, NAME, 155, razlo27a, GOOD_REACTION);
 end
 
 procedure razlo27a
@@ -569,18 +569,18 @@ end
 
 procedure razlo28
 begin
-    gsay_reply(129, 158);
-    giq_option(5, 129, 159, razloend, 50);
+    Reply(158);
+    giq_option(5, NAME, 159, razloend, NEUTRAL_REACTION);
 end
 
 procedure razlo29
 begin
-    gsay_message(129, 160, 50);
+    gsay_message(NAME, 160, NEUTRAL_REACTION);
 end
 
 procedure razlo30
 begin
-    gsay_message(129, 161, 50);
+    gsay_message(NAME, 161, NEUTRAL_REACTION);
     gfade_out(400);
     gfade_in(400);
     game_time_advance(game_ticks(14400));
@@ -591,8 +591,8 @@ end
 procedure razlo31
 begin
     variable LVar0 := 0;
-    gsay_message(129, 162, 50);
-    LVar0 := create_object_sid(49, 0, 0, -1);
+    gsay_message(NAME, 162, NEUTRAL_REACTION);
+    LVar0 := create_object(49, 0, 0);
     add_obj_to_inven(dude_obj, LVar0);
     tail := obj_carrying_pid_obj(dude_obj, 92);
     rm_obj_from_inven(dude_obj, tail);
@@ -600,46 +600,46 @@ begin
     if (local_var(7) == 0) then begin
         set_local_var(7, 1);
         give_exp_points(250);
-        display_msg(message_str(129, 300));
+        display_mstr(300);
     end
 end
 
 procedure razlo00n
 begin
-    gsay_reply(129, 163);
-    giq_option(4, 129, 164, razlo02n, 50);
-    if (global_var(43) == 1) then begin
-        giq_option(5, 129, 165, razlo09n, 50);
+    Reply(163);
+    giq_option(4, NAME, 164, razlo02n, NEUTRAL_REACTION);
+    if (global_var(GVAR_RADSCORPION_SEED) == 1) then begin
+        giq_option(5, NAME, 165, razlo09n, NEUTRAL_REACTION);
     end
-    if ((global_var(26) == 1) and (global_var(218) == 1)) then begin
-        giq_option(5, 129, 166, razlo15n, 50);
+    if ((global_var(GVAR_TANDI_STATUS) == 1) and (global_var(GVAR_TALKED_ABOUT_TANDI) == 1)) then begin
+        giq_option(5, NAME, 166, razlo15n, NEUTRAL_REACTION);
     end
-    giq_option(-3, 129, 167, razlo01n, 50);
+    giq_option(-3, NAME, 167, razlo01n, NEUTRAL_REACTION);
 end
 
 procedure razlo01n
 begin
-    gsay_message(129, 168, 50);
+    gsay_message(NAME, 168, NEUTRAL_REACTION);
     call DownReact();
 end
 
 procedure razlo02n
 begin
-    gsay_reply(129, 169);
-    giq_option(4, 129, 170, razlo04n, 50);
-    giq_option(4, 129, 171, razlo03n, 50);
+    Reply(169);
+    giq_option(4, NAME, 170, razlo04n, NEUTRAL_REACTION);
+    giq_option(4, NAME, 171, razlo03n, NEUTRAL_REACTION);
 end
 
 procedure razlo03n
 begin
-    gsay_message(129, 172, 50);
+    gsay_message(NAME, 172, NEUTRAL_REACTION);
     call DownReact();
 end
 
 procedure razlo04n
 begin
-    gsay_reply(129, 173);
-    giq_option(4, 129, 174, razlo05n, 50);
+    Reply(173);
+    giq_option(4, NAME, 174, razlo05n, NEUTRAL_REACTION);
 end
 
 procedure razlo05n
@@ -665,20 +665,20 @@ end
 
 procedure razlo06n
 begin
-    gsay_reply(129, 186);
-    giq_option(4, 129, 187, Razlo17, 50);
+    Reply(186);
+    giq_option(4, NAME, 187, Razlo17, NEUTRAL_REACTION);
     if (obj_is_carrying_obj_pid(dude_obj, 92)) then begin
-        giq_option(4, 129, 305, razlo06na, 50);
-        giq_option(-3, 129, 303, razlo06na, 50);
+        giq_option(4, NAME, 305, razlo06na, NEUTRAL_REACTION);
+        giq_option(-3, NAME, 303, razlo06na, NEUTRAL_REACTION);
     end
-    giq_option(4, 129, 188, razloend, 50);
-    giq_option(-3, 129, 304, Razlo17, 50);
-    giq_option(-3, 129, 302, razloend, 50);
+    giq_option(4, NAME, 188, razloend, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 304, Razlo17, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 302, razloend, NEUTRAL_REACTION);
 end
 
 procedure razlo07n
 begin
-    gsay_message(129, 189, 49);
+    gsay_message(NAME, 189, GOOD_REACTION);
 end
 
 procedure razlo08n
@@ -699,25 +699,25 @@ end
 
 procedure razlo09n
 begin
-    gsay_reply(129, 190);
-    giq_option(6, 129, 191, razlo10n, 50);
+    Reply(190);
+    giq_option(6, NAME, 191, razlo10n, NEUTRAL_REACTION);
 end
 
 procedure razlo10n
 begin
-    gsay_message(129, 192, 50);
+    gsay_message(NAME, 192, NEUTRAL_REACTION);
 end
 
 procedure razlo11n
 begin
-    gsay_reply(129, 108);
-    giq_option(5, 129, 194, razlo04, 50);
-    giq_option(5, 129, 195, razloend, 50);
+    Reply(108);
+    giq_option(5, NAME, 194, razlo04, NEUTRAL_REACTION);
+    giq_option(5, NAME, 195, razloend, NEUTRAL_REACTION);
 end
 
 procedure razlo14n
 begin
-    gsay_message(129, 196, 50);
+    gsay_message(NAME, 196, NEUTRAL_REACTION);
     gfade_out(400);
     gfade_in(400);
     game_time_advance(game_ticks(1200 * damage));
@@ -737,47 +737,47 @@ end
 
 procedure razlo16n
 begin
-    gsay_message(129, 197, 50);
+    gsay_message(NAME, 197, NEUTRAL_REACTION);
 end
 
 procedure razlowmpa
 begin
-    gsay_message(129, 175, 50);
+    gsay_message(NAME, 175, NEUTRAL_REACTION);
 end
 
 procedure razlowmpb
 begin
-    gsay_message(129, 176, 50);
+    gsay_message(NAME, 176, NEUTRAL_REACTION);
 end
 
 procedure razlofixa
 begin
     COST := 25;
-    gsay_reply(129, 177);
-    giq_option(4, 129, 178, razlo08n, 50);
-    giq_option(4, 129, 179, razlo06n, 50);
-    giq_option(-3, 129, 301, razlo08n, 50);
-    giq_option(-3, 129, 302, razloend, 50);
+    Reply(177);
+    giq_option(4, NAME, 178, razlo08n, NEUTRAL_REACTION);
+    giq_option(4, NAME, 179, razlo06n, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 301, razlo08n, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 302, razloend, NEUTRAL_REACTION);
 end
 
 procedure razlofixb
 begin
     COST := 50;
-    gsay_reply(129, 180);
-    giq_option(4, 129, 181, razlo08n, 50);
-    giq_option(4, 129, 182, razlo06n, 50);
-    giq_option(-3, 129, 301, razlo08n, 50);
-    giq_option(-3, 129, 302, razloend, 50);
+    Reply(180);
+    giq_option(4, NAME, 181, razlo08n, NEUTRAL_REACTION);
+    giq_option(4, NAME, 182, razlo06n, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 301, razlo08n, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 302, razloend, NEUTRAL_REACTION);
 end
 
 procedure razlofixc
 begin
     COST := 100;
-    gsay_reply(129, 183);
-    giq_option(4, 129, 184, razlo08n, 50);
-    giq_option(4, 129, 185, razlo06n, 50);
-    giq_option(-3, 129, 301, razlo08n, 50);
-    giq_option(-3, 129, 302, razloend, 50);
+    Reply(183);
+    giq_option(4, NAME, 184, razlo08n, NEUTRAL_REACTION);
+    giq_option(4, NAME, 185, razlo06n, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 301, razlo08n, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 302, razloend, NEUTRAL_REACTION);
 end
 
 procedure razlox
@@ -791,14 +791,14 @@ end
 procedure givestuff
 begin
     variable LVar0 := 0;
-    LVar0 := create_object_sid(40, 0, 0, -1);
+    LVar0 := create_object(40, 0, 0);
     add_obj_to_inven(dude_obj, LVar0);
-    LVar0 := create_object_sid(40, 0, 0, -1);
+    LVar0 := create_object(40, 0, 0);
     add_obj_to_inven(dude_obj, LVar0);
-    LVar0 := create_object_sid(71, 0, 0, -1);
+    LVar0 := create_object(71, 0, 0);
     add_obj_to_inven(dude_obj, LVar0);
     if (local_var(8) == 1) then begin
-        LVar0 := create_object_sid(49, 0, 0, -1);
+        LVar0 := create_object(49, 0, 0);
         add_obj_to_inven(dude_obj, LVar0);
     end
 end
@@ -844,46 +844,46 @@ begin
         set_local_var(10, local_var(10) + 1);
         set_map_var(2, 0);
         if (local_var(10) < 3) then begin
-            float_msg(self_obj, message_str(129, 308), 2);
+            float_msg(self_obj, mstr(308), FLOAT_MSG_RED);
         end
         else begin
             HOSTILE := 1;
         end
     end
     if (obj_can_see_obj(self_obj, dude_obj)) then begin
-        if (global_var(246) == 1) then begin
+        if (global_var(GVAR_ENEMY_SHADY_SANDS) == 1) then begin
             HOSTILE := 1;
         end
     end
     if (HOSTILE) then begin
         HOSTILE := 0;
-        attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+        attack(dude_obj);
     end
 end
 
 procedure look_at_p_proc
 begin
     script_overrides;
-    display_msg(message_str(129, 100));
+    display_mstr(100);
 end
 
 procedure destroy_p_proc
 begin
     if (source_obj == dude_obj) then begin
-        set_global_var(246, 1);
+        set_global_var(GVAR_ENEMY_SHADY_SANDS, 1);
     end
     if (source_obj == dude_obj) then begin
-        if (((global_var(160) + global_var(159)) >= 25) and ((global_var(159) > (2 * global_var(160))) or (global_var(156) == 1))) then begin
-            set_global_var(156, 1);
-            set_global_var(157, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
+            set_global_var(GVAR_BERSERKER_REPUTATION, 1);
+            set_global_var(GVAR_CHAMPION_REPUTATION, 0);
         end
-        if (((global_var(160) + global_var(159)) >= 25) and ((global_var(160) > (3 * global_var(159))) or (global_var(157) == 1))) then begin
-            set_global_var(157, 1);
-            set_global_var(156, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(160) > (3 * global_var(GVAR_GOOD_MONSTER))) or (global_var(GVAR_CHAMPION_REPUTATION) == 1))) then begin
+            set_global_var(GVAR_CHAMPION_REPUTATION, 1);
+            set_global_var(GVAR_BERSERKER_REPUTATION, 0);
         end
-        set_global_var(159, global_var(159) + 1);
-        if ((global_var(159) % 2) == 0) then begin
-            set_global_var(155, global_var(155) - 1);
+        set_global_var(GVAR_GOOD_MONSTER, global_var(GVAR_GOOD_MONSTER) + 1);
+        if ((global_var(GVAR_GOOD_MONSTER) % 2) == 0) then begin
+            set_global_var(GVAR_PLAYER_REPUATION, global_var(GVAR_PLAYER_REPUATION) - 1);
         end
     end
 end
@@ -941,31 +941,31 @@ begin
         set_local_var(0, 50);
         set_local_var(1, 2);
         set_local_var(2, 1);
-        set_local_var(0, local_var(0) + (5 * get_critter_stat(dude_obj, 3)) - 25);
-        set_local_var(0, local_var(0) + (10 * has_trait(0, dude_obj, 10)));
-        if (has_trait(0, dude_obj, 39)) then begin
-            if (global_var(155) > 0) then begin
-                set_local_var(0, local_var(0) + global_var(155));
+        set_local_var(0, local_var(0) + (5 * dude_charisma) - 25);
+        set_local_var(0, local_var(0) + (10 * has_trait(TRAIT_PERK, dude_obj, PERK_presence)));
+        if (has_trait(TRAIT_PERK, dude_obj, PERK_cult_of_personality)) then begin
+            if (global_var(GVAR_PLAYER_REPUATION) > 0) then begin
+                set_local_var(0, local_var(0) + global_var(GVAR_PLAYER_REPUATION));
             end
             else begin
-                set_local_var(0, local_var(0) - global_var(155));
+                set_local_var(0, local_var(0) - global_var(GVAR_PLAYER_REPUATION));
             end
         end
         else begin
             if (local_var(3) == 1) then begin
-                set_local_var(0, local_var(0) - global_var(155));
+                set_local_var(0, local_var(0) - global_var(GVAR_PLAYER_REPUATION));
             end
             else begin
-                set_local_var(0, local_var(0) + global_var(155));
+                set_local_var(0, local_var(0) + global_var(GVAR_PLAYER_REPUATION));
             end
         end
-        if (global_var(158) > 2) then begin
+        if (global_var(GVAR_CHILDKILLER_REPUATION) > 2) then begin
             set_local_var(0, local_var(0) - 30);
         end
-        if (((global_var(160) + global_var(159)) >= 25) and ((global_var(160) > (3 * global_var(159))) or (global_var(157) == 1))) then begin
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(160) > (3 * global_var(GVAR_GOOD_MONSTER))) or (global_var(GVAR_CHAMPION_REPUTATION) == 1))) then begin
             set_local_var(0, local_var(0) + 20);
         end
-        if (((global_var(160) + global_var(159)) >= 25) and ((global_var(159) > (2 * global_var(160))) or (global_var(156) == 1))) then begin
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
             set_local_var(0, local_var(0) - 20);
         end
         call ReactToLevel();
@@ -1058,6 +1058,5 @@ end
 
 procedure Goodbyes
 begin
-    exit_line := message_str(634, random(100, 105));
+    exit_line := message_str(SCRIPT_MODREACT, random(100, 105));
 end
-
