diff --git a/Games/Fallout1.Steam/Original/RD1INIT.ssl b/Games/Fallout1.Steam/ReDefine/RD1INIT.ssl
index 2621250..8c111c8 100644
--- a/Games/Fallout1.Steam/Original/RD1INIT.ssl
+++ b/Games/Fallout1.Steam/ReDefine/RD1INIT.ssl
@@ -41,26 +41,26 @@ procedure start
 begin
     if (Only_Once) then begin
         Only_Once := 0;
-        critter_add_trait(self_obj, 1, 6, 44);
-        critter_add_trait(self_obj, 1, 5, 62);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 44);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_BROTHERHOOD_INITIATE);
     end
-    if (script_action == 21) then begin
+    if (script_action == look_at_proc) then begin
         call look_at_p_proc();
     end
     else begin
-        if (script_action == 4) then begin
+        if (script_action == pickup_proc) then begin
             call pickup_p_proc();
         end
         else begin
-            if (script_action == 11) then begin
+            if (script_action == talk_proc) then begin
                 call talk_p_proc();
             end
             else begin
-                if (script_action == 12) then begin
+                if (script_action == critter_proc) then begin
                     call critter_p_proc();
                 end
                 else begin
-                    if (script_action == 18) then begin
+                    if (script_action == destroy_proc) then begin
                         call destroy_p_proc();
                     end
                 end
@@ -76,16 +76,16 @@ end
 
 procedure critter_p_proc
 begin
-    if (global_var(250)) then begin
+    if (global_var(GVAR_ENEMY_BROTHERHOOD)) then begin
         hostile := 1;
     end
     if (tile_distance_objs(self_obj, dude_obj) > 12) then begin
         hostile := 0;
     end
     if (hostile) then begin
-        set_global_var(250, 1);
+        set_global_var(GVAR_ENEMY_BROTHERHOOD, 1);
         hostile := 0;
-        attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+        attack(dude_obj);
     end
 end
 
@@ -112,20 +112,20 @@ end
 
 procedure destroy_p_proc
 begin
-    set_global_var(250, 1);
+    set_global_var(GVAR_ENEMY_BROTHERHOOD, 1);
     rm_timer_event(self_obj);
     if (source_obj == dude_obj) then begin
-        if (((global_var(160) + global_var(159)) >= 25) and ((global_var(159) > (2 * global_var(160))) or (global_var(156) == 1))) then begin
-            set_global_var(156, 1);
-            set_global_var(157, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
+            set_global_var(GVAR_BERSERKER_REPUTATION, 1);
+            set_global_var(GVAR_CHAMPION_REPUTATION, 0);
         end
-        if (((global_var(160) + global_var(159)) >= 25) and ((global_var(160) > (3 * global_var(159))) or (global_var(157) == 1))) then begin
-            set_global_var(157, 1);
-            set_global_var(156, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(160) > (3 * global_var(GVAR_GOOD_MONSTER))) or (global_var(GVAR_CHAMPION_REPUTATION) == 1))) then begin
+            set_global_var(GVAR_CHAMPION_REPUTATION, 1);
+            set_global_var(GVAR_BERSERKER_REPUTATION, 0);
         end
-        set_global_var(159, global_var(159) + 1);
-        if ((global_var(159) % 2) == 0) then begin
-            set_global_var(155, global_var(155) - 1);
+        set_global_var(GVAR_GOOD_MONSTER, global_var(GVAR_GOOD_MONSTER) + 1);
+        if ((global_var(GVAR_GOOD_MONSTER) % 2) == 0) then begin
+            set_global_var(GVAR_PLAYER_REPUATION, global_var(GVAR_PLAYER_REPUATION) - 1);
         end
     end
 end
@@ -133,59 +133,59 @@ end
 procedure look_at_p_proc
 begin
     script_overrides;
-    display_msg(message_str(691, 100));
+    display_mstr(100);
 end
 
 procedure goto00
 begin
-    gsay_reply(691, message_str(691, 103) + proto_data(obj_pid(dude_obj), 1) + message_str(691, 104));
-    giq_option(4, 691, 105, goto03, 50);
-    giq_option(4, 691, 106, goto01, 50);
-    giq_option(4, 691, 107, goto02, 50);
-    giq_option(-3, 691, 108, goto04, 50);
+    Reply(mstr(103) + proto_data(obj_pid(dude_obj), 1) + mstr(104));
+    giq_option(4, NAME, 105, goto03, NEUTRAL_REACTION);
+    giq_option(4, NAME, 106, goto01, NEUTRAL_REACTION);
+    giq_option(4, NAME, 107, goto02, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 108, goto04, NEUTRAL_REACTION);
 end
 
 procedure goto01
 begin
-    gsay_reply(691, 109);
-    giq_option(4, 691, 110, goto06, 50);
-    giq_option(4, 691, 111, goto05, 50);
+    Reply(109);
+    giq_option(4, NAME, 110, goto06, NEUTRAL_REACTION);
+    giq_option(4, NAME, 111, goto05, NEUTRAL_REACTION);
 end
 
 procedure goto02
 begin
-    gsay_reply(691, 112);
-    giq_option(4, 691, 110, goto06, 50);
-    giq_option(4, 691, 111, goto05, 50);
+    Reply(112);
+    giq_option(4, NAME, 110, goto06, NEUTRAL_REACTION);
+    giq_option(4, NAME, 111, goto05, NEUTRAL_REACTION);
 end
 
 procedure goto03
 begin
-    gsay_reply(691, 113);
-    giq_option(4, 691, 114, goto04, 50);
+    Reply(113);
+    giq_option(4, NAME, 114, goto04, NEUTRAL_REACTION);
 end
 
 procedure goto04
 begin
-    gsay_message(691, 115, 50);
+    gsay_message(NAME, 115, NEUTRAL_REACTION);
     call dialog_end();
 end
 
 procedure goto05
 begin
-    gsay_message(691, 116, 50);
+    gsay_message(NAME, 116, NEUTRAL_REACTION);
     call dialog_end();
 end
 
 procedure goto06
 begin
-    gsay_message(691, 117, 50);
+    gsay_message(NAME, 117, NEUTRAL_REACTION);
     call dialog_end();
 end
 
 procedure goto07
 begin
-    float_msg(self_obj, message_str(691, random(118, 125)), 2);
+    float_msg(self_obj, mstr(random(118, 125)), FLOAT_MSG_RED);
 end
 
 procedure gotoend
@@ -208,31 +208,31 @@ begin
         set_local_var(0, 50);
         set_local_var(1, 2);
         set_local_var(2, 1);
-        set_local_var(0, local_var(0) + (5 * get_critter_stat(dude_obj, 3)) - 25);
-        set_local_var(0, local_var(0) + (10 * has_trait(0, dude_obj, 10)));
-        if (has_trait(0, dude_obj, 39)) then begin
-            if (global_var(155) > 0) then begin
-                set_local_var(0, local_var(0) + global_var(155));
+        set_local_var(0, local_var(0) + (5 * dude_charisma) - 25);
+        set_local_var(0, local_var(0) + (10 * has_trait(TRAIT_PERK, dude_obj, PERK_presence)));
+        if (has_trait(TRAIT_PERK, dude_obj, PERK_cult_of_personality)) then begin
+            if (global_var(GVAR_PLAYER_REPUATION) > 0) then begin
+                set_local_var(0, local_var(0) + global_var(GVAR_PLAYER_REPUATION));
             end
             else begin
-                set_local_var(0, local_var(0) - global_var(155));
+                set_local_var(0, local_var(0) - global_var(GVAR_PLAYER_REPUATION));
             end
         end
         else begin
             if (local_var(3) == 1) then begin
-                set_local_var(0, local_var(0) - global_var(155));
+                set_local_var(0, local_var(0) - global_var(GVAR_PLAYER_REPUATION));
             end
             else begin
-                set_local_var(0, local_var(0) + global_var(155));
+                set_local_var(0, local_var(0) + global_var(GVAR_PLAYER_REPUATION));
             end
         end
-        if (global_var(158) > 2) then begin
+        if (global_var(GVAR_CHILDKILLER_REPUATION) > 2) then begin
             set_local_var(0, local_var(0) - 30);
         end
-        if (((global_var(160) + global_var(159)) >= 25) and ((global_var(160) > (3 * global_var(159))) or (global_var(157) == 1))) then begin
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(160) > (3 * global_var(GVAR_GOOD_MONSTER))) or (global_var(GVAR_CHAMPION_REPUTATION) == 1))) then begin
             set_local_var(0, local_var(0) + 20);
         end
-        if (((global_var(160) + global_var(159)) >= 25) and ((global_var(159) > (2 * global_var(160))) or (global_var(156) == 1))) then begin
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
             set_local_var(0, local_var(0) - 20);
         end
         call ReactToLevel();
@@ -325,6 +325,5 @@ end
 
 procedure Goodbyes
 begin
-    exit_line := message_str(634, random(100, 105));
+    exit_line := message_str(SCRIPT_MODREACT, random(100, 105));
 end
-
