diff --git a/Games/Fallout1.Steam/Original/SAMMY2.ssl b/Games/Fallout1.Steam/ReDefine/SAMMY2.ssl
index 9e67ccc..367dcf0 100644
--- a/Games/Fallout1.Steam/Original/SAMMY2.ssl
+++ b/Games/Fallout1.Steam/ReDefine/SAMMY2.ssl
@@ -4,17 +4,17 @@ procedure do_dialogue;
 
 procedure start
 begin
-    if (script_action == 11) then begin
+    if (script_action == talk_proc) then begin
         call do_dialogue();
     end
     else begin
-        if (script_action == 3) then begin
+        if (script_action == description_proc) then begin
             script_overrides;
-            display_msg(message_str(14, 100));
+            display_msg(message_str(SCRIPT_SSWELL, 100));
         end
         else begin
-            if (script_action == 18) then begin
-                set_global_var(6, 1);
+            if (script_action == destroy_proc) then begin
+                set_global_var(GVAR_QUEST_PEOPLE, 1);
             end
         end
     end
@@ -22,8 +22,7 @@ end
 
 procedure do_dialogue
 begin
-    start_gdialog(-1, 14, self_obj, -1, -1);
-    sayMessage(0, message_str(14, 101));
+    start_dialogue(14, self_obj);
+    sayMessage(0, message_str(SCRIPT_SSWELL, 101));
     end_dialogue;
 end
-
