diff --git a/Games/Fallout1.Steam/Original/SARAH.ssl b/Games/Fallout1.Steam/ReDefine/SARAH.ssl
index 749b299..f9c8547 100644
--- a/Games/Fallout1.Steam/Original/SARAH.ssl
+++ b/Games/Fallout1.Steam/ReDefine/SARAH.ssl
@@ -36,20 +36,20 @@ variable exit_line;
 procedure start
 begin
     if (not(initialized)) then begin
-        critter_add_trait(self_obj, 1, 6, 34);
-        critter_add_trait(self_obj, 1, 5, 67);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 34);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_BRAINWASHED_CHILDREN);
         initialized := 1;
     end
     else begin
-        if (script_action == 12) then begin
+        if (script_action == critter_proc) then begin
             call critter_p_proc();
         end
         else begin
-            if (script_action == 18) then begin
+            if (script_action == destroy_proc) then begin
                 call destroy_p_proc();
             end
             else begin
-                if (script_action == 11) then begin
+                if (script_action == talk_proc) then begin
                     call talk_p_proc();
                 end
             end
@@ -73,17 +73,17 @@ end
 procedure destroy_p_proc
 begin
     if (source_obj == dude_obj) then begin
-        if (((global_var(160) + global_var(159)) >= 25) and ((global_var(159) > (2 * global_var(160))) or (global_var(156) == 1))) then begin
-            set_global_var(156, 1);
-            set_global_var(157, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
+            set_global_var(GVAR_BERSERKER_REPUTATION, 1);
+            set_global_var(GVAR_CHAMPION_REPUTATION, 0);
         end
-        if (((global_var(160) + global_var(159)) >= 25) and ((global_var(160) > (3 * global_var(159))) or (global_var(157) == 1))) then begin
-            set_global_var(157, 1);
-            set_global_var(156, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(160) > (3 * global_var(GVAR_GOOD_MONSTER))) or (global_var(GVAR_CHAMPION_REPUTATION) == 1))) then begin
+            set_global_var(GVAR_CHAMPION_REPUTATION, 1);
+            set_global_var(GVAR_BERSERKER_REPUTATION, 0);
         end
-        set_global_var(159, global_var(159) + 1);
-        if ((global_var(159) % 2) == 0) then begin
-            set_global_var(155, global_var(155) - 1);
+        set_global_var(GVAR_GOOD_MONSTER, global_var(GVAR_GOOD_MONSTER) + 1);
+        if ((global_var(GVAR_GOOD_MONSTER) % 2) == 0) then begin
+            set_global_var(GVAR_PLAYER_REPUATION, global_var(GVAR_PLAYER_REPUATION) - 1);
         end
     end
 end
@@ -92,11 +92,11 @@ procedure talk_p_proc
 begin
     if (local_var(4)) then begin
         if (map_var(8) == 3) then begin
-            float_msg(self_obj, message_str(662, 105), 0);
+            float_msg(self_obj, mstr(105), FLOAT_MSG_NORMAL);
         end
         else begin
-            if (get_critter_stat(dude_obj, 4) < 4) then begin
-                float_msg(self_obj, message_str(662, 117), 0);
+            if (dude_iq < 4) then begin
+                float_msg(self_obj, mstr(117), FLOAT_MSG_NORMAL);
             end
             else begin
                 call Sarah10();
@@ -120,71 +120,71 @@ end
 
 procedure Sarah00
 begin
-    gsay_reply(662, 100);
-    giq_option(-3, 662, 101, Sarah01, 51);
-    giq_option(4, 662, 102, SarahEnd, 51);
-    giq_option(5, 662, 103, Sarah01, 51);
-    giq_option(7, 662, 104, sarah02, 51);
+    Reply(100);
+    giq_option(-3, NAME, 101, Sarah01, BAD_REACTION);
+    giq_option(4, NAME, 102, SarahEnd, BAD_REACTION);
+    giq_option(5, NAME, 103, Sarah01, BAD_REACTION);
+    giq_option(7, NAME, 104, sarah02, BAD_REACTION);
 end
 
 procedure Sarah01
 begin
-    gsay_message(662, 105, 51);
+    gsay_message(NAME, 105, BAD_REACTION);
 end
 
 procedure sarah02
 begin
-    gsay_reply(662, 106);
-    giq_option(7, 662, 107, Sarah03, 51);
-    giq_option(7, 662, 108, Sarah04, 51);
+    Reply(106);
+    giq_option(7, NAME, 107, Sarah03, BAD_REACTION);
+    giq_option(7, NAME, 108, Sarah04, BAD_REACTION);
 end
 
 procedure Sarah03
 begin
-    gsay_reply(662, 109);
-    giq_option(7, 662, 110, Sarah01, 51);
-    giq_option(7, 662, 111, Sarah04, 51);
+    Reply(109);
+    giq_option(7, NAME, 110, Sarah01, BAD_REACTION);
+    giq_option(7, NAME, 111, Sarah04, BAD_REACTION);
 end
 
 procedure Sarah04
 begin
-    gsay_message(662, 112, 51);
+    gsay_message(NAME, 112, BAD_REACTION);
 end
 
 procedure Sarah05
 begin
-    gsay_reply(662, 113);
-    giq_option(-3, 662, 114, Sarah06, 50);
-    giq_option(4, 662, 115, Sarah07, 50);
-    giq_option(6, 662, 116, Sarah09, 50);
+    Reply(113);
+    giq_option(-3, NAME, 114, Sarah06, NEUTRAL_REACTION);
+    giq_option(4, NAME, 115, Sarah07, NEUTRAL_REACTION);
+    giq_option(6, NAME, 116, Sarah09, NEUTRAL_REACTION);
 end
 
 procedure Sarah06
 begin
-    gsay_message(662, 117, 50);
+    gsay_message(NAME, 117, NEUTRAL_REACTION);
 end
 
 procedure Sarah07
 begin
-    gsay_reply(662, 118);
-    giq_option(4, 662, 119, SarahEnd, 50);
-    giq_option(6, 662, 120, Sarah08, 50);
+    Reply(118);
+    giq_option(4, NAME, 119, SarahEnd, NEUTRAL_REACTION);
+    giq_option(6, NAME, 120, Sarah08, NEUTRAL_REACTION);
 end
 
 procedure Sarah08
 begin
-    gsay_reply(662, 121);
-    giq_option(4, 662, 119, SarahEnd, 51);
+    Reply(121);
+    giq_option(4, NAME, 119, SarahEnd, BAD_REACTION);
 end
 
 procedure Sarah09
 begin
-    gsay_message(662, 122, 50);
+    gsay_message(NAME, 122, NEUTRAL_REACTION);
 end
 
 procedure Sarah10
 begin
-    float_msg(self_obj, message_str(662, 123), 0);
+    float_msg(self_obj, mstr(123), FLOAT_MSG_NORMAL);
 end
 
 procedure SarahEnd
@@ -197,31 +197,31 @@ begin
         set_local_var(0, 50);
         set_local_var(1, 2);
         set_local_var(2, 1);
-        set_local_var(0, local_var(0) + (5 * get_critter_stat(dude_obj, 3)) - 25);
-        set_local_var(0, local_var(0) + (10 * has_trait(0, dude_obj, 10)));
-        if (has_trait(0, dude_obj, 39)) then begin
-            if (global_var(155) > 0) then begin
-                set_local_var(0, local_var(0) + global_var(155));
+        set_local_var(0, local_var(0) + (5 * dude_charisma) - 25);
+        set_local_var(0, local_var(0) + (10 * has_trait(TRAIT_PERK, dude_obj, PERK_presence)));
+        if (has_trait(TRAIT_PERK, dude_obj, PERK_cult_of_personality)) then begin
+            if (global_var(GVAR_PLAYER_REPUATION) > 0) then begin
+                set_local_var(0, local_var(0) + global_var(GVAR_PLAYER_REPUATION));
             end
             else begin
-                set_local_var(0, local_var(0) - global_var(155));
+                set_local_var(0, local_var(0) - global_var(GVAR_PLAYER_REPUATION));
             end
         end
         else begin
             if (local_var(3) == 1) then begin
-                set_local_var(0, local_var(0) - global_var(155));
+                set_local_var(0, local_var(0) - global_var(GVAR_PLAYER_REPUATION));
             end
             else begin
-                set_local_var(0, local_var(0) + global_var(155));
+                set_local_var(0, local_var(0) + global_var(GVAR_PLAYER_REPUATION));
             end
         end
-        if (global_var(158) > 2) then begin
+        if (global_var(GVAR_CHILDKILLER_REPUATION) > 2) then begin
             set_local_var(0, local_var(0) - 30);
         end
-        if (((global_var(160) + global_var(159)) >= 25) and ((global_var(160) > (3 * global_var(159))) or (global_var(157) == 1))) then begin
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(160) > (3 * global_var(GVAR_GOOD_MONSTER))) or (global_var(GVAR_CHAMPION_REPUTATION) == 1))) then begin
             set_local_var(0, local_var(0) + 20);
         end
-        if (((global_var(160) + global_var(159)) >= 25) and ((global_var(159) > (2 * global_var(160))) or (global_var(156) == 1))) then begin
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
             set_local_var(0, local_var(0) - 20);
         end
         call ReactToLevel();
@@ -314,6 +314,5 @@ end
 
 procedure Goodbyes
 begin
-    exit_line := message_str(634, random(100, 105));
+    exit_line := message_str(SCRIPT_MODREACT, random(100, 105));
 end
-
