diff --git a/Games/Fallout1.Steam/Original/SAUL.ssl b/Games/Fallout1.Steam/ReDefine/SAUL.ssl
index cb61af2..78768a6 100644
--- a/Games/Fallout1.Steam/Original/SAUL.ssl
+++ b/Games/Fallout1.Steam/ReDefine/SAUL.ssl
@@ -94,39 +94,39 @@ procedure Goodbyes;
 
 procedure start
 begin
-    if (script_action == 13) then begin
+    if (script_action == combat_proc) then begin
         call combat_p_proc();
     end
     else begin
-        if (script_action == 12) then begin
+        if (script_action == critter_proc) then begin
             call critter_p_proc();
         end
         else begin
-            if (script_action == 3) then begin
+            if (script_action == description_proc) then begin
                 call description_p_proc();
             end
             else begin
-                if (script_action == 18) then begin
+                if (script_action == destroy_proc) then begin
                     call destroy_p_proc();
                 end
                 else begin
-                    if (script_action == 21) then begin
+                    if (script_action == look_at_proc) then begin
                         call look_at_p_proc();
                     end
                     else begin
-                        if (script_action == 15) then begin
+                        if (script_action == map_enter_proc) then begin
                             call map_enter_p_proc();
                         end
                         else begin
-                            if (script_action == 23) then begin
+                            if (script_action == map_update_proc) then begin
                                 call map_update_p_proc();
                             end
                             else begin
-                                if (script_action == 11) then begin
+                                if (script_action == talk_proc) then begin
                                     call talk_p_proc();
                                 end
                                 else begin
-                                    if (script_action == 22) then begin
+                                    if (script_action == timed_event_proc) then begin
                                         call timed_event_p_proc();
                                     end
                                 end
@@ -145,9 +145,9 @@ begin
         GVar7 := GVar7 + 1;
     end
     if (GVar7 > 3) then begin
-        if (global_var(247) == 0) then begin
-            set_global_var(247, 1);
-            set_global_var(155, global_var(155) - 5);
+        if (global_var(GVAR_ENEMY_JUNKTOWN) == 0) then begin
+            set_global_var(GVAR_ENEMY_JUNKTOWN, 1);
+            set_global_var(GVAR_PLAYER_REPUATION, global_var(GVAR_PLAYER_REPUATION) - 5);
         end
     end
 end
@@ -156,7 +156,7 @@ procedure critter_p_proc
 begin
     if (GVar5) then begin
         GVar5 := 0;
-        attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+        attack(dude_obj);
     end
     else begin
         if (cur_map_index == 11) then begin
@@ -180,58 +180,58 @@ begin
             call sleeping();
         end
     end
-    if (global_var(247) == 1) then begin
+    if (global_var(GVAR_ENEMY_JUNKTOWN) == 1) then begin
         if (obj_can_see_obj(self_obj, dude_obj)) then begin
-            attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+            attack(dude_obj);
         end
     end
 end
 
 procedure description_p_proc
 begin
-    if (is_success(do_check(dude_obj, 1, has_trait(0, dude_obj, 0)))) then begin
-        display_msg(message_str(528, 102));
+    if (is_success(do_check(dude_obj, 1, has_trait(TRAIT_PERK, dude_obj, PERK_bonus_awareness)))) then begin
+        display_mstr(102);
     end
 end
 
 procedure destroy_p_proc
 begin
     if (source_obj == dude_obj) then begin
-        set_global_var(247, 1);
+        set_global_var(GVAR_ENEMY_JUNKTOWN, 1);
         if (source_obj == dude_obj) then begin
-            if (((global_var(160) + global_var(159)) >= 25) and ((global_var(159) > (2 * global_var(160))) or (global_var(156) == 1))) then begin
-                set_global_var(156, 1);
-                set_global_var(157, 0);
+            if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
+                set_global_var(GVAR_BERSERKER_REPUTATION, 1);
+                set_global_var(GVAR_CHAMPION_REPUTATION, 0);
             end
-            if (((global_var(160) + global_var(159)) >= 25) and ((global_var(160) > (3 * global_var(159))) or (global_var(157) == 1))) then begin
-                set_global_var(157, 1);
-                set_global_var(156, 0);
+            if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(160) > (3 * global_var(GVAR_GOOD_MONSTER))) or (global_var(GVAR_CHAMPION_REPUTATION) == 1))) then begin
+                set_global_var(GVAR_CHAMPION_REPUTATION, 1);
+                set_global_var(GVAR_BERSERKER_REPUTATION, 0);
             end
-            set_global_var(159, global_var(159) + 1);
-            if ((global_var(159) % 2) == 0) then begin
-                set_global_var(155, global_var(155) - 1);
+            set_global_var(GVAR_GOOD_MONSTER, global_var(GVAR_GOOD_MONSTER) + 1);
+            if ((global_var(GVAR_GOOD_MONSTER) % 2) == 0) then begin
+                set_global_var(GVAR_PLAYER_REPUATION, global_var(GVAR_PLAYER_REPUATION) - 1);
             end
         end
     end
-    set_global_var(169, 3);
+    set_global_var(GVAR_SAUL_STATUS, 3);
 end
 
 procedure look_at_p_proc
 begin
     script_overrides;
-    if (not(global_var(169))) then begin
-        display_msg(message_str(528, 101));
+    if (not(global_var(GVAR_SAUL_STATUS))) then begin
+        display_mstr(101);
     end
     else begin
-        display_msg(message_str(528, 100));
+        display_mstr(100);
     end
 end
 
 procedure map_enter_p_proc
 begin
     variable LVar0 := 0;
-    if (global_var(15) == 1) then begin
-        LVar0 := create_object_sid(234, 0, 0, -1);
+    if (global_var(GVAR_JUNKTOWN_INVADED) == 1) then begin
+        LVar0 := create_object(234, 0, 0);
         add_obj_to_inven(self_obj, LVar0);
         kill_critter(self_obj, 48);
     end
@@ -247,7 +247,7 @@ begin
             GVar3 := 16892;
         end
     end
-    if (global_var(169) == 3) then begin
+    if (global_var(GVAR_SAUL_STATUS) == 3) then begin
         destroy_object(self_obj);
     end
     move_to(self_obj, GVar3, 0);
@@ -275,15 +275,15 @@ procedure talk_p_proc
 begin
     call get_reaction();
     if (local_var(4)) then begin
-        float_msg(self_obj, message_str(185, 166), 0);
+        float_msg(self_obj, message_str(SCRIPT_GENVAULT, 166), FLOAT_MSG_NORMAL);
     end
     else begin
         if (critter_state(self_obj) bwand 1) then begin
-            display_msg(message_str(528, 115));
+            display_mstr(115);
         end
         else begin
-            if (global_var(247)) then begin
-                float_msg(self_obj, message_str(669, 104), 2);
+            if (global_var(GVAR_ENEMY_JUNKTOWN)) then begin
+                float_msg(self_obj, message_str(SCRIPT_ENEMY, 104), FLOAT_MSG_RED);
             end
             else begin
                 if ((obj_item_subtype(critter_inven_obj(dude_obj, 1)) == 3) or (obj_item_subtype(critter_inven_obj(dude_obj, 2)) == 3)) then begin
@@ -292,7 +292,7 @@ begin
                 else begin
                     start_gdialog(528, self_obj, 4, -1, -1);
                     gsay_start;
-                    if (global_var(169)) then begin
+                    if (global_var(GVAR_SAUL_STATUS)) then begin
                         call Saul23();
                     end
                     else begin
@@ -304,9 +304,9 @@ begin
             end
         end
     end
-    if (not(global_var(557) bwand 16) and (global_var(557) bwand 8)) then begin
-        set_global_var(557, global_var(557) + 16);
-        display_msg(message_str(342, 173));
+    if (not(global_var(GVAR_TRISH_STATUS) bwand 16) and (global_var(GVAR_TRISH_STATUS) bwand 8)) then begin
+        set_global_var(GVAR_TRISH_STATUS, global_var(GVAR_TRISH_STATUS) + 16);
+        display_msg(message_str(SCRIPT_TRISH, 173));
         give_exp_points(250);
     end
 end
@@ -348,171 +348,171 @@ end
 procedure damage_p_proc
 begin
     if (source_obj == dude_obj) then begin
-        set_global_var(247, 1);
+        set_global_var(GVAR_ENEMY_JUNKTOWN, 1);
     end
 end
 
 procedure Saul06
 begin
-    float_msg(self_obj, message_str(528, 104), 2);
+    float_msg(self_obj, mstr(104), FLOAT_MSG_RED);
 end
 
 procedure Saul07
 begin
     if (local_var(1) == 1) then begin
-        gsay_reply(528, 105);
+        Reply(105);
     end
     else begin
-        gsay_reply(528, 106);
+        Reply(106);
     end
-    giq_option(4, 528, 107, Saul08, 50);
-    giq_option(4, 528, 108, Saul09, 51);
-    giq_option(4, 528, message_str(528, 109) + proto_data(obj_pid(dude_obj), 1) + message_str(528, 110), Saul09, 50);
+    giq_option(4, NAME, 107, Saul08, NEUTRAL_REACTION);
+    giq_option(4, NAME, 108, Saul09, BAD_REACTION);
+    giq_option(4, NAME, mstr(109) + proto_data(obj_pid(dude_obj), 1) + mstr(110), Saul09, NEUTRAL_REACTION);
 end
 
 procedure Saul08
 begin
-    gsay_reply(528, 111);
-    giq_option(4, 528, 112, Saul09, 51);
-    giq_option(4, 528, 113, Saul11, 50);
-    giq_option(4, 528, 103, SaulEnd, 50);
+    Reply(111);
+    giq_option(4, NAME, 112, Saul09, BAD_REACTION);
+    giq_option(4, NAME, 113, Saul11, NEUTRAL_REACTION);
+    giq_option(4, NAME, 103, SaulEnd, NEUTRAL_REACTION);
 end
 
 procedure Saul09
 begin
     if (local_var(1) == 1) then begin
-        gsay_reply(528, 128);
+        Reply(128);
     end
     else begin
-        gsay_reply(528, 129);
+        Reply(129);
     end
     if (local_var(1) == 1) then begin
-        giq_option(4, 528, 130, Saul10, 50);
-        giq_option(4, 528, 188, SaulCombat, 51);
+        giq_option(4, NAME, 130, Saul10, NEUTRAL_REACTION);
+        giq_option(4, NAME, 188, SaulCombat, BAD_REACTION);
     end
-    giq_option(4, 528, 131, Saul10, 50);
+    giq_option(4, NAME, 131, Saul10, NEUTRAL_REACTION);
     if (local_var(1) == 3) then begin
-        giq_option(4, 528, 135, Saul11, 50);
+        giq_option(4, NAME, 135, Saul11, NEUTRAL_REACTION);
     end
-    giq_option(4, 528, 132, SaulEnd, 50);
+    giq_option(4, NAME, 132, SaulEnd, NEUTRAL_REACTION);
 end
 
 procedure Saul10
 begin
     if (local_var(1) == 1) then begin
-        gsay_reply(528, 133);
+        Reply(133);
     end
     else begin
-        gsay_reply(528, 134);
+        Reply(134);
     end
-    giq_option(4, 528, 135, Saul11, 50);
-    giq_option(4, 528, 136, Saul12, 49);
-    giq_option(4, 528, 132, SaulEnd, 50);
+    giq_option(4, NAME, 135, Saul11, NEUTRAL_REACTION);
+    giq_option(4, NAME, 136, Saul12, GOOD_REACTION);
+    giq_option(4, NAME, 132, SaulEnd, NEUTRAL_REACTION);
 end
 
 procedure Saul11
 begin
-    gsay_reply(528, 137);
-    giq_option(4, 528, 136, Saul12, 49);
+    Reply(137);
+    giq_option(4, NAME, 136, Saul12, GOOD_REACTION);
     if (not(local_var(5))) then begin
-        giq_option(4, 528, 138, Saul13, 50);
+        giq_option(4, NAME, 138, Saul13, NEUTRAL_REACTION);
     end
     if (not(local_var(6))) then begin
-        giq_option(4, 528, 139, Saul14, 50);
+        giq_option(4, NAME, 139, Saul14, NEUTRAL_REACTION);
     end
     call Goodbyes();
-    giq_option(4, 528, GVar12, SaulEnd, 50);
+    giq_option(4, NAME, GVar12, SaulEnd, NEUTRAL_REACTION);
 end
 
 procedure Saul12
 begin
     call UpReact();
     if (local_var(1) == 1) then begin
-        gsay_reply(528, 140);
+        Reply(140);
     end
     else begin
-        gsay_reply(528, 141);
-        giq_option(5, 528, 142, Saul15, 49);
+        Reply(141);
+        giq_option(5, NAME, 142, Saul15, GOOD_REACTION);
     end
-    giq_option(4, 528, 143, Saul13, 50);
-    giq_option(4, 528, 144, Saul14, 50);
+    giq_option(4, NAME, 143, Saul13, NEUTRAL_REACTION);
+    giq_option(4, NAME, 144, Saul14, NEUTRAL_REACTION);
     call Goodbyes();
-    giq_option(4, 528, GVar12, SaulEnd, 50);
+    giq_option(4, NAME, GVar12, SaulEnd, NEUTRAL_REACTION);
 end
 
 procedure Saul13
 begin
     set_local_var(5, 1);
-    gsay_reply(528, 145);
-    giq_option(4, 528, 146, Saul19, 50);
-    giq_option(4, 528, 147, Saul14, 50);
+    Reply(145);
+    giq_option(4, NAME, 146, Saul19, NEUTRAL_REACTION);
+    giq_option(4, NAME, 147, Saul14, NEUTRAL_REACTION);
     call Goodbyes();
-    giq_option(4, 528, GVar12, SaulEnd, 50);
+    giq_option(4, NAME, GVar12, SaulEnd, NEUTRAL_REACTION);
 end
 
 procedure Saul14
 begin
     set_local_var(6, 1);
-    gsay_reply(528, 148);
+    Reply(148);
     if (not(local_var(5))) then begin
-        giq_option(4, 528, 149, Saul13, 50);
+        giq_option(4, NAME, 149, Saul13, NEUTRAL_REACTION);
     end
     call Goodbyes();
-    giq_option(4, 528, GVar12, SaulEnd, 50);
+    giq_option(4, NAME, GVar12, SaulEnd, NEUTRAL_REACTION);
 end
 
 procedure Saul15
 begin
     set_local_var(7, 1);
-    gsay_reply(528, 150);
-    giq_option(4, 528, 151, Saul17, 50);
-    giq_option(4, 528, 152, Saul18, 50);
-    if (get_critter_stat(dude_obj, 4) >= 6) then begin
-        giq_option(6, 528, 153, Saul16, 49);
+    Reply(150);
+    giq_option(4, NAME, 151, Saul17, NEUTRAL_REACTION);
+    giq_option(4, NAME, 152, Saul18, NEUTRAL_REACTION);
+    if (dude_iq >= 6) then begin
+        giq_option(6, NAME, 153, Saul16, GOOD_REACTION);
     end
     else begin
         call Goodbyes();
-        giq_option(4, 528, GVar12, SaulEnd, 50);
+        giq_option(4, NAME, GVar12, SaulEnd, NEUTRAL_REACTION);
     end
-    giq_option(6, 528, 154, Saul24, 50);
+    giq_option(6, NAME, 154, Saul24, NEUTRAL_REACTION);
 end
 
 procedure Saul16
 begin
     call UpReact();
-    gsay_reply(528, 155);
-    giq_option(4, 528, 156, Saul18, 50);
+    Reply(155);
+    giq_option(4, NAME, 156, Saul18, NEUTRAL_REACTION);
     call Goodbyes();
-    giq_option(4, 528, GVar12, SaulEnd, 50);
+    giq_option(4, NAME, GVar12, SaulEnd, NEUTRAL_REACTION);
 end
 
 procedure Saul17
 begin
-    gsay_reply(528, 157);
-    giq_option(4, 528, 158, Saul18, 50);
-    if ((global_var(557) bwand 1) or (global_var(557) bwand 4)) then begin
-        giq_option(5, 528, 159, Saul20, 50);
+    Reply(157);
+    giq_option(4, NAME, 158, Saul18, NEUTRAL_REACTION);
+    if ((global_var(GVAR_TRISH_STATUS) bwand 1) or (global_var(GVAR_TRISH_STATUS) bwand 4)) then begin
+        giq_option(5, NAME, 159, Saul20, NEUTRAL_REACTION);
     end
     call Goodbyes();
-    giq_option(4, 528, GVar12, SaulEnd, 50);
+    giq_option(4, NAME, GVar12, SaulEnd, NEUTRAL_REACTION);
 end
 
 procedure Saul18
 begin
     if (local_var(1) < 2) then begin
-        gsay_reply(528, 160);
+        Reply(160);
     end
     else begin
-        gsay_reply(528, 161);
+        Reply(161);
     end
     if (not(local_var(5))) then begin
-        giq_option(4, 528, 162, Saul13, 50);
+        giq_option(4, NAME, 162, Saul13, NEUTRAL_REACTION);
     end
     if (not(local_var(6))) then begin
-        giq_option(4, 528, 163, Saul14, 50);
+        giq_option(4, NAME, 163, Saul14, NEUTRAL_REACTION);
     end
-    giq_option(4, 528, 164, Saul18a, 51);
-    giq_option(4, 528, 165, SaulEnd, 50);
+    giq_option(4, NAME, 164, Saul18a, BAD_REACTION);
+    giq_option(4, NAME, 165, SaulEnd, NEUTRAL_REACTION);
 end
 
 procedure Saul18a
@@ -524,45 +524,45 @@ end
 
 procedure Saul19
 begin
-    gsay_reply(528, 166);
-    giq_option(4, 528, 168, Saul18, 50);
-    if (get_critter_stat(dude_obj, 4) >= 6) then begin
-        giq_option(6, 528, 167, UpReact, 49);
+    Reply(166);
+    giq_option(4, NAME, 168, Saul18, NEUTRAL_REACTION);
+    if (dude_iq >= 6) then begin
+        giq_option(6, NAME, 167, UpReact, GOOD_REACTION);
     end
     else begin
         call Goodbyes();
-        giq_option(4, 528, 169, SaulEnd, 50);
+        giq_option(4, NAME, 169, SaulEnd, NEUTRAL_REACTION);
     end
 end
 
 procedure Saul20
 begin
     if (local_var(1) == 1) then begin
-        gsay_reply(528, 169);
+        Reply(169);
     end
     else begin
         if (is_success(roll_vs_skill(dude_obj, 14, 0))) then begin
-            gsay_reply(528, message_str(528, 170) + message_str(528, 171));
-            if (get_critter_stat(dude_obj, 4) >= 6) then begin
-                giq_option(6, 528, 172, Saul21, 49);
+            Reply(mstr(170) + mstr(171));
+            if (dude_iq >= 6) then begin
+                giq_option(6, NAME, 172, Saul21, GOOD_REACTION);
             end
             else begin
-                giq_option(4, 528, 173, Saul22, 50);
+                giq_option(4, NAME, 173, Saul22, NEUTRAL_REACTION);
             end
         end
         else begin
-            gsay_reply(528, message_str(528, 170));
+            Reply(mstr(170));
         end
     end
-    giq_option(4, 528, 174, Saul18, 50);
-    giq_option(4, 528, 175, SaulEnd, 50);
+    giq_option(4, NAME, 174, Saul18, NEUTRAL_REACTION);
+    giq_option(4, NAME, 175, SaulEnd, NEUTRAL_REACTION);
 end
 
 procedure Saul21
 begin
     call UpReact();
-    set_global_var(557, global_var(557) + 8);
-    gsay_message(528, 176, 49);
+    set_global_var(GVAR_TRISH_STATUS, global_var(GVAR_TRISH_STATUS) + 8);
+    gsay_message(NAME, 176, GOOD_REACTION);
 end
 
 procedure Saul22
@@ -571,36 +571,36 @@ begin
         call Saul21();
     end
     else begin
-        gsay_reply(528, 177);
-        giq_option(4, 528, 178, Saul18, 50);
-        giq_option(4, 528, 179, SaulEnd, 50);
+        Reply(177);
+        giq_option(4, NAME, 178, Saul18, NEUTRAL_REACTION);
+        giq_option(4, NAME, 179, SaulEnd, NEUTRAL_REACTION);
     end
 end
 
 procedure Saul23
 begin
     if (local_var(1) < 2) then begin
-        gsay_reply(528, 180);
+        Reply(180);
     end
     else begin
-        gsay_reply(528, 181);
+        Reply(181);
     end
-    giq_option(4, 528, 182, Saul09, 51);
-    if ((global_var(557) bwand 1) or (global_var(557) bwand 4) and not(global_var(557) bwand 8)) then begin
-        giq_option(4, 528, 183, Saul20, 50);
+    giq_option(4, NAME, 182, Saul09, BAD_REACTION);
+    if ((global_var(GVAR_TRISH_STATUS) bwand 1) or (global_var(GVAR_TRISH_STATUS) bwand 4) and not(global_var(GVAR_TRISH_STATUS) bwand 8)) then begin
+        giq_option(4, NAME, 183, Saul20, NEUTRAL_REACTION);
     end
     if (not(local_var(5))) then begin
-        giq_option(4, 528, 184, Saul13, 50);
+        giq_option(4, NAME, 184, Saul13, NEUTRAL_REACTION);
     end
     call Goodbyes();
-    giq_option(4, 528, GVar12, SaulEnd, 50);
+    giq_option(4, NAME, GVar12, SaulEnd, NEUTRAL_REACTION);
 end
 
 procedure Saul24
 begin
-    gsay_reply(528, 185);
-    giq_option(4, 528, 186, SaulEnd, 0);
-    giq_option(4, 528, 187, Saul18, 0);
+    Reply(185);
+    giq_option(4, NAME, 186, SaulEnd, 0);
+    giq_option(4, NAME, 187, Saul18, 0);
 end
 
 procedure SaulBoxing
@@ -611,7 +611,7 @@ begin
         reg_anim_func(1, 1);
         reg_anim_animate(self_obj, 16, -1);
         if (is_success(roll_vs_skill(self_obj, 3, 0))) then begin
-            if (GVar9 >= (get_critter_stat(challenger_ptr, 2) * 2)) then begin
+            if (GVar9 >= (get_critter_stat(challenger_ptr, STAT_en) * 2)) then begin
                 reg_anim_animate(challenger_ptr, 20, 4);
                 reg_anim_animate(challenger_ptr, 48, -1);
                 add_timer_event(self_obj, game_ticks(2), 2);
@@ -632,7 +632,7 @@ begin
         reg_anim_func(1, 1);
         reg_anim_animate(challenger_ptr, 16, -1);
         if (is_success(roll_vs_skill(challenger_ptr, 3, 0))) then begin
-            if (GVar10 >= (get_critter_stat(self_obj, 2) * 2)) then begin
+            if (GVar10 >= (self_endurance * 2)) then begin
                 reg_anim_animate(self_obj, 20, 4);
                 reg_anim_animate(self_obj, 48, -1);
                 add_timer_event(self_obj, game_ticks(2), 3);
@@ -664,23 +664,23 @@ procedure create_challenger
 begin
     challenger_ptr := random(0, 4);
     if (challenger_ptr == 4) then begin
-        challenger_ptr := create_object_sid(16777226, 0, 0, 25);
+        challenger_ptr := create_object_sid(16777226, 0, 0, SCRIPT_DESK);
     end
     else begin
         if (challenger_ptr == 3) then begin
-            challenger_ptr := create_object_sid(16777227, 0, 0, 25);
+            challenger_ptr := create_object_sid(16777227, 0, 0, SCRIPT_DESK);
         end
         else begin
             if (challenger_ptr == 2) then begin
-                challenger_ptr := create_object_sid(16777238, 0, 0, 25);
+                challenger_ptr := create_object_sid(16777238, 0, 0, SCRIPT_DESK);
             end
             else begin
                 if (challenger_ptr == 1) then begin
-                    challenger_ptr := create_object_sid(16777218, 0, 0, 25);
+                    challenger_ptr := create_object_sid(16777218, 0, 0, SCRIPT_DESK);
                 end
                 else begin
                     if (challenger_ptr == 0) then begin
-                        challenger_ptr := create_object_sid(16777243, 0, 0, 25);
+                        challenger_ptr := create_object_sid(16777243, 0, 0, SCRIPT_DESK);
                     end
                 end
             end
@@ -738,31 +738,31 @@ begin
         set_local_var(0, 50);
         set_local_var(1, 2);
         set_local_var(2, 1);
-        set_local_var(0, local_var(0) + (5 * get_critter_stat(dude_obj, 3)) - 25);
-        set_local_var(0, local_var(0) + (10 * has_trait(0, dude_obj, 10)));
-        if (has_trait(0, dude_obj, 39)) then begin
-            if (global_var(155) > 0) then begin
-                set_local_var(0, local_var(0) + global_var(155));
+        set_local_var(0, local_var(0) + (5 * dude_charisma) - 25);
+        set_local_var(0, local_var(0) + (10 * has_trait(TRAIT_PERK, dude_obj, PERK_presence)));
+        if (has_trait(TRAIT_PERK, dude_obj, PERK_cult_of_personality)) then begin
+            if (global_var(GVAR_PLAYER_REPUATION) > 0) then begin
+                set_local_var(0, local_var(0) + global_var(GVAR_PLAYER_REPUATION));
             end
             else begin
-                set_local_var(0, local_var(0) - global_var(155));
+                set_local_var(0, local_var(0) - global_var(GVAR_PLAYER_REPUATION));
             end
         end
         else begin
             if (local_var(3) == 1) then begin
-                set_local_var(0, local_var(0) - global_var(155));
+                set_local_var(0, local_var(0) - global_var(GVAR_PLAYER_REPUATION));
             end
             else begin
-                set_local_var(0, local_var(0) + global_var(155));
+                set_local_var(0, local_var(0) + global_var(GVAR_PLAYER_REPUATION));
             end
         end
-        if (global_var(158) > 2) then begin
+        if (global_var(GVAR_CHILDKILLER_REPUATION) > 2) then begin
             set_local_var(0, local_var(0) - 30);
         end
-        if (((global_var(160) + global_var(159)) >= 25) and ((global_var(160) > (3 * global_var(159))) or (global_var(157) == 1))) then begin
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(160) > (3 * global_var(GVAR_GOOD_MONSTER))) or (global_var(GVAR_CHAMPION_REPUTATION) == 1))) then begin
             set_local_var(0, local_var(0) + 20);
         end
-        if (((global_var(160) + global_var(159)) >= 25) and ((global_var(159) > (2 * global_var(160))) or (global_var(156) == 1))) then begin
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
             set_local_var(0, local_var(0) - 20);
         end
         call ReactToLevel();
@@ -855,6 +855,5 @@ end
 
 procedure Goodbyes
 begin
-    GVar12 := message_str(634, random(100, 105));
+    GVar12 := message_str(SCRIPT_MODREACT, random(100, 105));
 end
-
