diff --git a/Games/Fallout1.Steam/Original/SBOKCASE.ssl b/Games/Fallout1.Steam/ReDefine/SBOKCASE.ssl
index 93c0383..b8cb4f1 100644
--- a/Games/Fallout1.Steam/Original/SBOKCASE.ssl
+++ b/Games/Fallout1.Steam/ReDefine/SBOKCASE.ssl
@@ -7,15 +7,15 @@ procedure chairend;
 
 procedure start
 begin
-    if (script_action == 4) then begin
+    if (script_action == pickup_proc) then begin
         call pickup_p_proc();
     end
     else begin
-        if (script_action == 3) then begin
+        if (script_action == description_proc) then begin
             call description_p_proc();
         end
         else begin
-            if (script_action == 21) then begin
+            if (script_action == look_at_proc) then begin
                 call look_at_p_proc();
             end
         end
@@ -24,7 +24,7 @@ end
 
 procedure pickup_p_proc
 begin
-    if (not(is_success(roll_vs_skill(dude_obj, 10, 0))) and (global_var(246) != 1)) then begin
+    if (not(is_success(roll_vs_skill(dude_obj, 10, 0))) and (global_var(GVAR_ENEMY_SHADY_SANDS) != 1)) then begin
         script_overrides;
         set_map_var(2, 1);
     end
@@ -33,16 +33,15 @@ end
 procedure description_p_proc
 begin
     script_overrides;
-    display_msg(message_str(122, 100));
+    display_mstr(100);
 end
 
 procedure look_at_p_proc
 begin
     script_overrides;
-    display_msg(message_str(122, 101));
+    display_mstr(101);
 end
 
 procedure chairend
 begin
 end
-
