diff --git a/Games/Fallout1.Steam/Original/SDOOR.ssl b/Games/Fallout1.Steam/ReDefine/SDOOR.ssl
index 6305358..a9ae88c 100644
--- a/Games/Fallout1.Steam/Original/SDOOR.ssl
+++ b/Games/Fallout1.Steam/ReDefine/SDOOR.ssl
@@ -8,23 +8,23 @@ procedure damage_p_proc;
 
 procedure start
 begin
-    if (script_action == 21) then begin
+    if (script_action == look_at_proc) then begin
         call look_at_p_proc();
     end
     else begin
-        if (script_action == 6) then begin
+        if (script_action == use_proc) then begin
             call use_p_proc();
         end
         else begin
-            if (script_action == 8) then begin
+            if (script_action == use_skill_on_proc) then begin
                 call use_skill_on_p_proc();
             end
             else begin
-                if (script_action == 7) then begin
+                if (script_action == use_obj_on_proc) then begin
                     call use_obj_on_p_proc();
                 end
                 else begin
-                    if (script_action == 14) then begin
+                    if (script_action == damage_proc) then begin
                         call damage_p_proc();
                     end
                 end
@@ -37,7 +37,7 @@ procedure use_p_proc
 begin
     script_overrides;
     if (source_obj == dude_obj) then begin
-        display_msg(message_str(526, 104));
+        display_msg(message_str(SCRIPT_MBOUT2IN, 104));
     end
 end
 
@@ -45,14 +45,14 @@ procedure use_skill_on_p_proc
 begin
     if (action_being_used == 9) then begin
         script_overrides;
-        display_msg(message_str(526, 104));
+        display_msg(message_str(SCRIPT_MBOUT2IN, 104));
     end
 end
 
 procedure look_at_p_proc
 begin
     script_overrides;
-    display_msg(message_str(526, 100));
+    display_msg(message_str(SCRIPT_MBOUT2IN, 100));
 end
 
 procedure use_obj_on_p_proc
@@ -61,13 +61,12 @@ begin
     LVar0 := obj_being_used_with;
     if (obj_pid(LVar0) == 84) then begin
         script_overrides;
-        display_msg(message_str(526, 104));
+        display_msg(message_str(SCRIPT_MBOUT2IN, 104));
     end
 end
 
 procedure damage_p_proc
 begin
     script_overrides;
-    display_msg(message_str(526, 106));
+    display_msg(message_str(SCRIPT_MBOUT2IN, 106));
 end
-
