diff --git a/Games/Fallout1.Steam/Original/SEMICOMP.ssl b/Games/Fallout1.Steam/ReDefine/SEMICOMP.ssl
index e15baa4..72c132e 100644
--- a/Games/Fallout1.Steam/Original/SEMICOMP.ssl
+++ b/Games/Fallout1.Steam/ReDefine/SEMICOMP.ssl
@@ -10,12 +10,12 @@ begin
     if (local_var(0) == 0) then begin
         set_local_var(0, random(1, 9));
     end
-    if (script_action == 21) then begin
+    if (script_action == look_at_proc) then begin
         script_overrides;
         call search();
     end
     else begin
-        if (script_action == 8) then begin
+        if (script_action == use_skill_on_proc) then begin
             use_skill := action_being_used;
             script_overrides;
             call skills();
@@ -26,43 +26,43 @@ end
 procedure search
 begin
     if (tile_distance_objs(self_obj, dude_obj) > 3) then begin
-        display_msg(message_str(418, 100));
+        display_mstr(100);
     end
     else begin
         if (local_var(0) == 1) then begin
-            display_msg(message_str(418, 102));
+            display_mstr(102);
         end
         else begin
             if (local_var(0) == 2) then begin
-                display_msg(message_str(418, 103));
+                display_mstr(103);
             end
             else begin
                 if (local_var(0) == 3) then begin
-                    display_msg(message_str(418, 104));
+                    display_mstr(104);
                 end
                 else begin
                     if (local_var(0) == 4) then begin
-                        display_msg(message_str(418, 105));
+                        display_mstr(105);
                     end
                     else begin
                         if (local_var(0) == 5) then begin
-                            display_msg(message_str(418, 106));
+                            display_mstr(106);
                         end
                         else begin
                             if (local_var(0) == 6) then begin
-                                display_msg(message_str(418, 107));
+                                display_mstr(107);
                             end
                             else begin
                                 if (local_var(0) == 7) then begin
-                                    display_msg(message_str(418, 108));
+                                    display_mstr(108);
                                 end
                                 else begin
                                     if (local_var(0) == 8) then begin
-                                        display_msg(message_str(418, 109));
+                                        display_mstr(109);
                                     end
                                     else begin
                                         if (local_var(0) == 9) then begin
-                                            display_msg(message_str(418, 111));
+                                            display_mstr(111);
                                         end
                                     end
                                 end
@@ -78,10 +78,9 @@ end
 procedure skills
 begin
     if ((use_skill == 13) or (use_skill == 12)) then begin
-        display_msg(message_str(418, 101));
+        display_mstr(101);
     end
     else begin
-        display_msg(message_str(418, 102));
+        display_mstr(102);
     end
 end
-
