diff --git a/Games/Fallout1.Steam/Original/SENTRYGN.ssl b/Games/Fallout1.Steam/ReDefine/SENTRYGN.ssl
index 9f16937..cf2ea3b 100644
--- a/Games/Fallout1.Steam/Original/SENTRYGN.ssl
+++ b/Games/Fallout1.Steam/ReDefine/SENTRYGN.ssl
@@ -5,53 +5,52 @@ procedure start
 begin
     variable LVar0 := 0;
     variable LVar1 := 0;
-    if (script_action == 2) then begin
+    if (script_action == spatial_proc) then begin
         if ((local_var(0) == 0) and (obj_type(source_obj) == dude_obj)) then begin
-            display_msg(message_str(304, 100));
+            display_mstr(100);
             set_local_var(0, 1);
         end
-        if ((local_var(1) == 0) and (obj_type(source_obj) == dude_obj) and global_var(139) and global_var(140)) then begin
-            display_msg(message_str(304, 101));
+        if ((local_var(1) == 0) and (obj_type(source_obj) == dude_obj) and global_var(GVAR_START_POWER) and global_var(GVAR_WEAPONS_ARMED)) then begin
+            display_mstr(101);
             set_local_var(1, 1);
             add_timer_event(self_obj, game_ticks(7), 1);
         end
     end
     else begin
-        if (script_action == 22) then begin
+        if (script_action == timed_event_proc) then begin
             LVar0 := random(1, 6) + 2;
             LVar1 := do_check(source_obj, 5, 0);
             if (is_success(LVar1)) then begin
                 if (is_critical(LVar1)) then begin
                     if (obj_type(source_obj) == dude_obj) then begin
-                        display_msg(message_str(304, 102));
+                        display_mstr(102);
                     end
                 end
                 else begin
                     if (obj_type(source_obj) == dude_obj) then begin
-                        display_msg(message_str(304, 103));
+                        display_mstr(103);
                     end
-                    critter_dmg(source_obj, LVar0, 0);
-                    display_msg(message_str(304, 104) + LVar0 + message_str(304, 105));
+                    critter_damage(source_obj, LVar0);
+                    display_msg(mstr(104) + LVar0 + mstr(105));
                 end
             end
             else begin
                 if (is_critical(LVar1)) then begin
                     if (obj_type(source_obj) == dude_obj) then begin
-                        display_msg(message_str(304, 106));
+                        display_mstr(106);
                     end
-                    critter_dmg(source_obj, LVar0, 0);
-                    display_msg(message_str(304, 107) + LVar0 + message_str(304, 108));
+                    critter_damage(source_obj, LVar0);
+                    display_msg(mstr(107) + LVar0 + mstr(108));
                 end
                 else begin
                     if (obj_type(source_obj) == dude_obj) then begin
-                        display_msg(message_str(304, 109));
+                        display_mstr(109);
                     end
-                    critter_dmg(source_obj, LVar0, 0);
-                    display_msg(message_str(304, 110) + LVar0 + message_str(304, 111));
+                    critter_damage(source_obj, LVar0);
+                    display_msg(mstr(110) + LVar0 + mstr(111));
                 end
             end
             set_local_var(1, 0);
         end
     end
 end
-
