diff --git a/Games/Fallout1.Steam/Original/SETDOOR1.ssl b/Games/Fallout1.Steam/ReDefine/SETDOOR1.ssl
index 128be24..9705a6f 100644
--- a/Games/Fallout1.Steam/Original/SETDOOR1.ssl
+++ b/Games/Fallout1.Steam/ReDefine/SETDOOR1.ssl
@@ -11,15 +11,15 @@ variable testa;
 procedure start
 begin
     variable LVar0 := 0;
-    if (script_action == 21) then begin
+    if (script_action == look_at_proc) then begin
         script_overrides;
         call see_stuff();
     end
-    if (script_action == 6) then begin
+    if (script_action == use_proc) then begin
         script_overrides;
         call do_stuff();
     end
-    if (script_action == 7) then begin
+    if (script_action == use_obj_on_proc) then begin
         lockpicks := obj_being_used_with;
         script_overrides;
         call do_skill();
@@ -28,41 +28,41 @@ end
 
 procedure do_stuff
 begin
-    display_msg(message_str(94, 107));
+    display_mstr(107);
 end
 
 procedure do_skill
 begin
-    if (has_skill(dude_obj, 9)) then begin
+    if (dude_skill(SKILL_LOCKPICK)) then begin
         testa := roll_vs_skill(dude_obj, 9, -10);
         if (is_success(testa)) then begin
-            display_msg(message_str(94, 100));
+            display_mstr(100);
             set_local_var(0, 1);
         end
         else begin
-            display_msg(message_str(94, 101));
+            display_mstr(101);
             if (is_critical(testa)) then begin
-                display_msg(message_str(94, 102));
+                display_mstr(102);
                 rm_obj_from_inven(dude_obj, lockpicks);
             end
         end
     end
     else begin
-        display_msg(message_str(94, 103));
+        display_mstr(103);
     end
 end
 
 procedure see_stuff
 begin
     if (local_var(0) == 0) then begin
-        display_msg(message_str(94, 104));
+        display_mstr(104);
     end
     else begin
         if (local_var(2) == 1) then begin
-            display_msg(message_str(94, 105));
+            display_mstr(105);
         end
         else begin
-            display_msg(message_str(94, 106));
+            display_mstr(106);
         end
     end
 end
@@ -70,4 +70,3 @@ end
 procedure doorend
 begin
 end
-
