diff --git a/Games/Fallout1.Steam/Original/SETSCTDR.ssl b/Games/Fallout1.Steam/ReDefine/SETSCTDR.ssl
index c259fcd..022f924 100644
--- a/Games/Fallout1.Steam/Original/SETSCTDR.ssl
+++ b/Games/Fallout1.Steam/ReDefine/SETSCTDR.ssl
@@ -16,7 +16,7 @@ variable use_skill;
 
 procedure start
 begin
-    if (script_action == 6) then begin
+    if (script_action == use_proc) then begin
         if (source_obj == dude_obj) then begin
             if (local_var(2)) then begin
                 call use_door();
@@ -25,7 +25,7 @@ begin
                 set_local_var(3, 1);
                 temp := 100;
                 give_exp_points(temp);
-                display_msg(message_str(96, 200) + temp + message_str(96, 201));
+                display_msg(mstr(200) + temp + mstr(201));
             end
             else begin
                 if (local_var(1)) then begin
@@ -46,7 +46,7 @@ begin
         end
     end
     else begin
-        if (script_action == 8) then begin
+        if (script_action == use_skill_on_proc) then begin
             call trap_stuff();
         end
     end
@@ -77,11 +77,11 @@ begin
     script_overrides;
     test := roll_vs_skill(dude_obj, 11, -20);
     if (is_success(test)) then begin
-        display_msg(message_str(96, 100));
+        display_mstr(100);
         set_local_var(0, 1);
     end
     else begin
-        display_msg(message_str(96, 104));
+        display_mstr(104);
         call explode();
     end
 end
@@ -91,11 +91,11 @@ begin
     script_overrides;
     test := do_check(dude_obj, 4, 2);
     if (is_success(test)) then begin
-        display_msg(message_str(96, 100));
+        display_mstr(100);
         set_local_var(0, 1);
     end
     else begin
-        display_msg(message_str(96, 104));
+        display_mstr(104);
         call explode();
     end
 end
@@ -105,16 +105,16 @@ begin
     script_overrides;
     test := roll_vs_skill(dude_obj, 11, 0);
     if (is_success(test)) then begin
-        display_msg(message_str(96, 102));
+        display_mstr(102);
         set_local_var(1, 1);
     end
     else begin
         if (is_critical(test)) then begin
-            display_msg(message_str(96, 104));
+            display_mstr(104);
             call explode();
         end
         else begin
-            display_msg(message_str(96, 103));
+            display_mstr(103);
         end
     end
 end
@@ -124,16 +124,16 @@ begin
     script_overrides;
     test := roll_vs_skill(dude_obj, 13, 20);
     if (is_success(test)) then begin
-        display_msg(message_str(96, 102));
+        display_mstr(102);
         set_local_var(1, 1);
     end
     else begin
         if (is_critical(test)) then begin
-            display_msg(message_str(96, 104));
+            display_mstr(104);
             call explode();
         end
         else begin
-            display_msg(message_str(96, 103));
+            display_mstr(103);
         end
     end
 end
@@ -141,7 +141,7 @@ end
 procedure failure
 begin
     script_overrides;
-    display_msg(message_str(96, 105));
+    display_mstr(105);
     call explode();
 end
 
@@ -150,7 +150,7 @@ begin
     set_local_var(1, 1);
     explosion(tile_num(self_obj), elevation(self_obj), 0);
     damage := random(1, 6) + random(1, 6) + random(1, 6) + random(1, 6) + random(1, 6) + random(1, 6);
-    critter_dmg(dude_obj, damage, 0);
+    critter_damage(dude_obj, damage);
 end
 
 procedure use_door
@@ -162,4 +162,3 @@ begin
         set_local_var(2, 1);
     end
 end
-
