diff --git a/Games/Fallout1.Steam/Original/SINK.ssl b/Games/Fallout1.Steam/ReDefine/SINK.ssl
index a08d3cc..77259d2 100644
--- a/Games/Fallout1.Steam/Original/SINK.ssl
+++ b/Games/Fallout1.Steam/ReDefine/SINK.ssl
@@ -7,19 +7,19 @@ procedure use_obj_on_p_proc;
 
 procedure start
 begin
-    if (script_action == 21) then begin
+    if (script_action == look_at_proc) then begin
         call look_at_p_proc();
     end
     else begin
-        if (script_action == 6) then begin
+        if (script_action == use_proc) then begin
             call use_p_proc();
         end
         else begin
-            if (script_action == 4) then begin
+            if (script_action == pickup_proc) then begin
                 call pickup_p_proc();
             end
             else begin
-                if (script_action == 7) then begin
+                if (script_action == use_obj_on_proc) then begin
                     call use_obj_on_p_proc();
                 end
             end
@@ -30,7 +30,7 @@ end
 procedure use_p_proc
 begin
     script_overrides;
-    display_msg(message_str(739, 101));
+    display_mstr(101);
 end
 
 procedure pickup_p_proc
@@ -41,12 +41,11 @@ end
 procedure look_at_p_proc
 begin
     script_overrides;
-    display_msg(message_str(739, 100));
+    display_mstr(100);
 end
 
 procedure use_obj_on_p_proc
 begin
     script_overrides;
-    display_msg(message_str(739, 102) + proto_data(obj_pid(obj_being_used_with), 1) + message_str(739, 103));
+    display_msg(mstr(102) + proto_data(obj_pid(obj_being_used_with), 1) + mstr(103));
 end
-
