diff --git a/Games/Fallout1.Steam/Original/SMITTY.ssl b/Games/Fallout1.Steam/ReDefine/SMITTY.ssl
index 117f2dc..7d9db19 100644
--- a/Games/Fallout1.Steam/Original/SMITTY.ssl
+++ b/Games/Fallout1.Steam/ReDefine/SMITTY.ssl
@@ -47,13 +47,13 @@ begin
         if (obj_is_carrying_obj_pid(self_obj, 41) == 0) then begin
             item_caps_adjust(self_obj, random(500, 750));
         end
-        if ((global_var(613) == 9103) or (global_var(613) == 9102)) then begin
-            critter_add_trait(self_obj, 1, 6, 0);
+        if ((global_var(GVAR_BLADES) == 9103) or (global_var(GVAR_BLADES) == 9102)) then begin
+            critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 0);
         end
         else begin
-            critter_add_trait(self_obj, 1, 6, 49);
+            critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 49);
         end
-        critter_add_trait(self_obj, 1, 5, 6);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_PEASANT);
         Initialize := 0;
     end
 end
@@ -62,10 +62,10 @@ procedure look_at_p_proc
 begin
     script_overrides;
     if (local_var(4) == 1) then begin
-        display_msg(message_str(250, 100));
+        display_mstr(100);
     end
     else begin
-        display_msg(message_str(250, 101));
+        display_mstr(101);
     end
 end
 
@@ -73,35 +73,35 @@ procedure description_p_proc
 begin
     script_overrides;
     if (local_var(4) == 1) then begin
-        display_msg(message_str(250, 100));
+        display_mstr(100);
     end
     else begin
-        display_msg(message_str(250, 101));
+        display_mstr(101);
     end
 end
 
 procedure talk_p_proc
 begin
-    if (global_var(251) == 1) then begin
-        float_msg(self_obj, message_str(669, random(100, 105)), 2);
+    if (global_var(GVAR_ENEMY_ADYTUM) == 1) then begin
+        float_msg(self_obj, message_str(SCRIPT_ENEMY, random(100, 105)), FLOAT_MSG_RED);
     end
     else begin
-        if ((local_var(4) == 1) and (get_critter_stat(dude_obj, 4) < 4)) then begin
-            float_msg(self_obj, message_str(250, 117), 0);
+        if ((local_var(4) == 1) and (dude_iq < 4)) then begin
+            float_msg(self_obj, mstr(117), FLOAT_MSG_NORMAL);
         end
         else begin
             if (local_var(5) == 2) then begin
-                float_msg(self_obj, message_str(250, 131), 0);
+                float_msg(self_obj, mstr(131), FLOAT_MSG_NORMAL);
             end
             else begin
                 call get_reaction();
                 start_gdialog(250, self_obj, -1, -1, -1);
                 gsay_start;
-                if (global_var(138) == 9303) then begin
+                if (global_var(GVAR_FIX_FARM) == 9303) then begin
                     call Smitty11();
                 end
                 else begin
-                    if (global_var(138) == 9304) then begin
+                    if (global_var(GVAR_FIX_FARM) == 9304) then begin
                         if (local_var(5) == 0) then begin
                             call Smitty12();
                         end
@@ -137,52 +137,52 @@ end
 procedure damage_p_proc
 begin
     if (source_obj == dude_obj) then begin
-        set_global_var(251, 1);
-        critter_add_trait(self_obj, 1, 6, 49);
+        set_global_var(GVAR_ENEMY_ADYTUM, 1);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 49);
     end
 end
 
 procedure destroy_p_proc
 begin
     if (source_obj == dude_obj) then begin
-        set_global_var(251, 1);
+        set_global_var(GVAR_ENEMY_ADYTUM, 1);
         if (source_obj == dude_obj) then begin
-            if (((global_var(160) + global_var(159)) >= 25) and ((global_var(159) > (2 * global_var(160))) or (global_var(156) == 1))) then begin
-                set_global_var(156, 1);
-                set_global_var(157, 0);
+            if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
+                set_global_var(GVAR_BERSERKER_REPUTATION, 1);
+                set_global_var(GVAR_CHAMPION_REPUTATION, 0);
             end
-            if (((global_var(160) + global_var(159)) >= 25) and ((global_var(160) > (3 * global_var(159))) or (global_var(157) == 1))) then begin
-                set_global_var(157, 1);
-                set_global_var(156, 0);
+            if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(160) > (3 * global_var(GVAR_GOOD_MONSTER))) or (global_var(GVAR_CHAMPION_REPUTATION) == 1))) then begin
+                set_global_var(GVAR_CHAMPION_REPUTATION, 1);
+                set_global_var(GVAR_BERSERKER_REPUTATION, 0);
             end
-            set_global_var(159, global_var(159) + 1);
-            if ((global_var(159) % 2) == 0) then begin
-                set_global_var(155, global_var(155) - 1);
+            set_global_var(GVAR_GOOD_MONSTER, global_var(GVAR_GOOD_MONSTER) + 1);
+            if ((global_var(GVAR_GOOD_MONSTER) % 2) == 0) then begin
+                set_global_var(GVAR_PLAYER_REPUATION, global_var(GVAR_PLAYER_REPUATION) - 1);
             end
         end
-        critter_add_trait(self_obj, 1, 6, 49);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 49);
     end
 end
 
 procedure pickup_p_proc
 begin
-    set_global_var(251, 1);
+    set_global_var(GVAR_ENEMY_ADYTUM, 1);
 end
 
 procedure Smitty01
 begin
-    gsay_reply(250, DisplayMessage);
+    Reply(DisplayMessage);
     DisplayMessage := 112;
     if (local_var(4) == 0) then begin
-        giq_option(4, 250, 104, Smitty03, 50);
+        giq_option(4, NAME, 104, Smitty03, NEUTRAL_REACTION);
     end
-    if ((global_var(138) < 9303) and (global_var(138) > 2)) then begin
-        giq_option(4, 250, 105, Smitty10, 50);
+    if ((global_var(GVAR_FIX_FARM) < 9303) and (global_var(GVAR_FIX_FARM) > 2)) then begin
+        giq_option(4, NAME, 105, Smitty10, NEUTRAL_REACTION);
     end
-    giq_option(4, 250, 106, Smitty02, 50);
-    giq_option(4, 250, 107, Smitty04, 50);
-    giq_option(4, 250, 108, Smitty06, 50);
-    giq_option(-3, 250, 109, Smitty05, 50);
+    giq_option(4, NAME, 106, Smitty02, NEUTRAL_REACTION);
+    giq_option(4, NAME, 107, Smitty04, NEUTRAL_REACTION);
+    giq_option(4, NAME, 108, Smitty06, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 109, Smitty05, NEUTRAL_REACTION);
 end
 
 procedure Smitty02
@@ -200,22 +200,22 @@ end
 
 procedure Smitty04
 begin
-    gsay_reply(250, DisplayMessage);
-    gsay_option(250, 113, Smitty07, 50);
-    gsay_option(250, 114, Smitty08, 50);
-    gsay_option(250, 115, Smitty09, 50);
-    gsay_option(250, 116, Smitty06, 50);
+    Reply(DisplayMessage);
+    giq_option(1, NAME, 113, Smitty07, NEUTRAL_REACTION);
+    giq_option(1, NAME, 114, Smitty08, NEUTRAL_REACTION);
+    giq_option(1, NAME, 115, Smitty09, NEUTRAL_REACTION);
+    giq_option(1, NAME, 116, Smitty06, NEUTRAL_REACTION);
 end
 
 procedure Smitty05
 begin
     set_local_var(4, 1);
-    gsay_message(250, 117, 50);
+    gsay_message(NAME, 117, NEUTRAL_REACTION);
 end
 
 procedure Smitty06
 begin
-    gsay_message(250, 118, 50);
+    gsay_message(NAME, 118, NEUTRAL_REACTION);
 end
 
 procedure Smitty07
@@ -238,22 +238,22 @@ end
 
 procedure Smitty10
 begin
-    gsay_message(250, 122, 50);
+    gsay_message(NAME, 122, NEUTRAL_REACTION);
     gfade_out(600);
     game_time_advance(10 * 60 * 60);
     gfade_in(600);
-    set_global_var(138, 9303);
-    gsay_message(250, 123, 50);
+    set_global_var(GVAR_FIX_FARM, 9303);
+    gsay_message(NAME, 123, NEUTRAL_REACTION);
 end
 
 procedure Smitty11
 begin
-    gsay_message(250, 124, 50);
+    gsay_message(NAME, 124, NEUTRAL_REACTION);
 end
 
 procedure Smitty12
 begin
-    gsay_message(250, 125, 50);
+    gsay_message(NAME, 125, NEUTRAL_REACTION);
     set_local_var(5, 9307);
     if (obj_is_carrying_obj_pid(dude_obj, 15)) then begin
         call Smitty14();
@@ -262,16 +262,16 @@ end
 
 procedure Smitty14
 begin
-    gsay_reply(250, 127);
-    gsay_option(250, 126, Smitty14a, 50);
+    Reply(127);
+    giq_option(1, NAME, 126, Smitty14a, NEUTRAL_REACTION);
 end
 
 procedure Smitty14a
 begin
     variable LVar0 := 0;
     variable LVar1 := 0;
-    gsay_message(250, 128, 50);
-    LVar1 := create_object_sid(42, 0, 0, -1);
+    gsay_message(NAME, 128, NEUTRAL_REACTION);
+    LVar1 := create_object(42, 0, 0);
     move_obj_inven_to_obj(dude_obj, LVar1);
     move_obj_inven_to_obj(LVar1, dude_obj);
     LVar0 := obj_carrying_pid_obj(dude_obj, 15);
@@ -281,15 +281,15 @@ begin
     gfade_out(600);
     game_time_advance(10 * 60 * 60 * 2);
     gfade_in(600);
-    gsay_message(250, 129, 50);
-    LVar0 := create_object_sid(233, 0, 0, -1);
+    gsay_message(NAME, 129, NEUTRAL_REACTION);
+    LVar0 := create_object(233, 0, 0);
     add_obj_to_inven(dude_obj, LVar0);
     set_local_var(5, 2);
 end
 
 procedure Smitty15
 begin
-    gsay_message(250, 130, 50);
+    gsay_message(NAME, 130, NEUTRAL_REACTION);
 end
 
 procedure SmittyEnd
@@ -302,31 +302,31 @@ begin
         set_local_var(0, 50);
         set_local_var(1, 2);
         set_local_var(2, 1);
-        set_local_var(0, local_var(0) + (5 * get_critter_stat(dude_obj, 3)) - 25);
-        set_local_var(0, local_var(0) + (10 * has_trait(0, dude_obj, 10)));
-        if (has_trait(0, dude_obj, 39)) then begin
-            if (global_var(155) > 0) then begin
-                set_local_var(0, local_var(0) + global_var(155));
+        set_local_var(0, local_var(0) + (5 * dude_charisma) - 25);
+        set_local_var(0, local_var(0) + (10 * has_trait(TRAIT_PERK, dude_obj, PERK_presence)));
+        if (has_trait(TRAIT_PERK, dude_obj, PERK_cult_of_personality)) then begin
+            if (global_var(GVAR_PLAYER_REPUATION) > 0) then begin
+                set_local_var(0, local_var(0) + global_var(GVAR_PLAYER_REPUATION));
             end
             else begin
-                set_local_var(0, local_var(0) - global_var(155));
+                set_local_var(0, local_var(0) - global_var(GVAR_PLAYER_REPUATION));
             end
         end
         else begin
             if (local_var(3) == 1) then begin
-                set_local_var(0, local_var(0) - global_var(155));
+                set_local_var(0, local_var(0) - global_var(GVAR_PLAYER_REPUATION));
             end
             else begin
-                set_local_var(0, local_var(0) + global_var(155));
+                set_local_var(0, local_var(0) + global_var(GVAR_PLAYER_REPUATION));
             end
         end
-        if (global_var(158) > 2) then begin
+        if (global_var(GVAR_CHILDKILLER_REPUATION) > 2) then begin
             set_local_var(0, local_var(0) - 30);
         end
-        if (((global_var(160) + global_var(159)) >= 25) and ((global_var(160) > (3 * global_var(159))) or (global_var(157) == 1))) then begin
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(160) > (3 * global_var(GVAR_GOOD_MONSTER))) or (global_var(GVAR_CHAMPION_REPUTATION) == 1))) then begin
             set_local_var(0, local_var(0) + 20);
         end
-        if (((global_var(160) + global_var(159)) >= 25) and ((global_var(159) > (2 * global_var(160))) or (global_var(156) == 1))) then begin
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
             set_local_var(0, local_var(0) - 20);
         end
         call ReactToLevel();
@@ -419,6 +419,5 @@ end
 
 procedure Goodbyes
 begin
-    exit_line := message_str(634, random(100, 105));
+    exit_line := message_str(SCRIPT_MODREACT, random(100, 105));
 end
-
