diff --git a/Games/Fallout1.Steam/Original/STAPLE.ssl b/Games/Fallout1.Steam/ReDefine/STAPLE.ssl
index e853502..2c47be4 100644
--- a/Games/Fallout1.Steam/Original/STAPLE.ssl
+++ b/Games/Fallout1.Steam/ReDefine/STAPLE.ssl
@@ -55,26 +55,26 @@ procedure start
 begin
     if (Only_Once) then begin
         Only_Once := 0;
-        critter_add_trait(self_obj, 1, 6, 70);
-        critter_add_trait(self_obj, 1, 5, 50);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 70);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_MERCHANT);
     end
-    if (script_action == 21) then begin
+    if (script_action == look_at_proc) then begin
         call look_at_p_proc();
     end
     else begin
-        if (script_action == 4) then begin
+        if (script_action == pickup_proc) then begin
             call pickup_p_proc();
         end
         else begin
-            if (script_action == 11) then begin
+            if (script_action == talk_proc) then begin
                 call talk_p_proc();
             end
             else begin
-                if (script_action == 12) then begin
+                if (script_action == critter_proc) then begin
                     call critter_p_proc();
                 end
                 else begin
-                    if (script_action == 18) then begin
+                    if (script_action == destroy_proc) then begin
                         call destroy_p_proc();
                     end
                 end
@@ -92,7 +92,7 @@ procedure critter_p_proc
 begin
     if (hostile) then begin
         hostile := 0;
-        attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+        attack(dude_obj);
     end
 end
 
@@ -134,17 +134,17 @@ end
 procedure destroy_p_proc
 begin
     if (source_obj == dude_obj) then begin
-        if (((global_var(160) + global_var(159)) >= 25) and ((global_var(159) > (2 * global_var(160))) or (global_var(156) == 1))) then begin
-            set_global_var(156, 1);
-            set_global_var(157, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
+            set_global_var(GVAR_BERSERKER_REPUTATION, 1);
+            set_global_var(GVAR_CHAMPION_REPUTATION, 0);
         end
-        if (((global_var(160) + global_var(159)) >= 25) and ((global_var(160) > (3 * global_var(159))) or (global_var(157) == 1))) then begin
-            set_global_var(157, 1);
-            set_global_var(156, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(160) > (3 * global_var(GVAR_GOOD_MONSTER))) or (global_var(GVAR_CHAMPION_REPUTATION) == 1))) then begin
+            set_global_var(GVAR_CHAMPION_REPUTATION, 1);
+            set_global_var(GVAR_BERSERKER_REPUTATION, 0);
         end
-        set_global_var(159, global_var(159) + 1);
-        if ((global_var(159) % 2) == 0) then begin
-            set_global_var(155, global_var(155) - 1);
+        set_global_var(GVAR_GOOD_MONSTER, global_var(GVAR_GOOD_MONSTER) + 1);
+        if ((global_var(GVAR_GOOD_MONSTER) % 2) == 0) then begin
+            set_global_var(GVAR_PLAYER_REPUATION, global_var(GVAR_PLAYER_REPUATION) - 1);
         end
     end
 end
@@ -154,70 +154,70 @@ begin
     variable LVar0 := 0;
     LVar0 := obj_pid(source_obj);
     if (party_member_obj(LVar0) != 0) then begin
-        set_global_var(248, 1);
+        set_global_var(GVAR_ENEMY_HUB, 1);
     end
 end
 
 procedure look_at_p_proc
 begin
     script_overrides;
-    display_msg(message_str(862, 100));
+    display_mstr(100);
 end
 
 procedure Staple00
 begin
-    gsay_reply(862, 102);
+    Reply(102);
     call Staple02();
 end
 
 procedure Staple01
 begin
-    gsay_reply(862, 103);
+    Reply(103);
     call Staple02();
 end
 
 procedure Staple02
 begin
-    giq_option(4, 862, 104, Staple03, 50);
-    if ((global_var(101) != 2) and (local_var(5) == 0)) then begin
-        giq_option(4, 862, 105, Staple04, 50);
+    giq_option(4, NAME, 104, Staple03, NEUTRAL_REACTION);
+    if ((global_var(GVAR_FIND_WATER_CHIP) != 2) and (local_var(5) == 0)) then begin
+        giq_option(4, NAME, 105, Staple04, NEUTRAL_REACTION);
     end
-    if (global_var(615) == 1) then begin
-        giq_option(4, 862, 130, Staple15, 50);
+    if (global_var(GVAR_CHEMISTRY_BOOK) == 1) then begin
+        giq_option(4, NAME, 130, Staple15, NEUTRAL_REACTION);
     end
-    giq_option(4, 862, 106, StapleEnd, 50);
-    giq_option(-3, 862, 125, Staple14, 50);
+    giq_option(4, NAME, 106, StapleEnd, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 125, Staple14, NEUTRAL_REACTION);
 end
 
 procedure Staple03
 begin
-    gsay_message(862, 107, 50);
+    gsay_message(NAME, 107, NEUTRAL_REACTION);
     gdialog_mod_barter(-50);
-    gsay_reply(862, 127);
-    giq_option(4, 862, 128, StapleEnd, 50);
-    giq_option(-3, 862, 129, StapleEnd, 50);
+    Reply(127);
+    giq_option(4, NAME, 128, StapleEnd, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 129, StapleEnd, NEUTRAL_REACTION);
 end
 
 procedure Staple04
 begin
-    gsay_reply(862, 108);
-    giq_option(4, 862, 109, Staple05, 50);
-    giq_option(4, 862, 110, Staple06, 50);
+    Reply(108);
+    giq_option(4, NAME, 109, Staple05, NEUTRAL_REACTION);
+    giq_option(4, NAME, 110, Staple06, NEUTRAL_REACTION);
 end
 
 procedure Staple05
 begin
-    gsay_reply(862, 111);
-    giq_option(4, 862, 112, Staple06, 50);
+    Reply(111);
+    giq_option(4, NAME, 112, Staple06, NEUTRAL_REACTION);
 end
 
 procedure Staple06
 begin
-    gsay_reply(862, 113);
-    giq_option(4, 862, 114, Staple06a, 50);
-    giq_option(4, 862, 115, Staple06b, 50);
-    giq_option(4, 862, 116, Staple06c, 50);
-    giq_option(4, 862, 117, StapleEnd, 50);
+    Reply(113);
+    giq_option(4, NAME, 114, Staple06a, NEUTRAL_REACTION);
+    giq_option(4, NAME, 115, Staple06b, NEUTRAL_REACTION);
+    giq_option(4, NAME, 116, Staple06c, NEUTRAL_REACTION);
+    giq_option(4, NAME, 117, StapleEnd, NEUTRAL_REACTION);
 end
 
 procedure Staple06a
@@ -257,20 +257,20 @@ end
 
 procedure Staple07
 begin
-    gsay_message(862, 118, 50);
+    gsay_message(NAME, 118, NEUTRAL_REACTION);
 end
 
 procedure Staple08
 begin
     set_local_var(6, 1);
-    gsay_message(862, 119, 51);
+    gsay_message(NAME, 119, BAD_REACTION);
 end
 
 procedure Staple09
 begin
     variable LVar0 := 0;
     LVar0 := item_caps_adjust(dude_obj, -750);
-    gsay_message(862, 120, 50);
+    gsay_message(NAME, 120, NEUTRAL_REACTION);
     call Staple12();
 end
 
@@ -278,14 +278,14 @@ procedure Staple10
 begin
     variable LVar0 := 0;
     LVar0 := item_caps_adjust(dude_obj, -500);
-    gsay_message(862, 121, 50);
+    gsay_message(NAME, 121, NEUTRAL_REACTION);
     call Staple12();
 end
 
 procedure Staple11
 begin
     set_local_var(6, 1);
-    gsay_message(862, 122, 51);
+    gsay_message(NAME, 122, BAD_REACTION);
     call Staple12();
 end
 
@@ -293,26 +293,26 @@ procedure Staple12
 begin
     variable LVar0 := 0;
     set_local_var(5, 1);
-    LVar0 := create_object_sid(230, 0, 0, -1);
+    LVar0 := create_object(230, 0, 0);
     add_obj_to_inven(dude_obj, LVar0);
-    gsay_message(862, 123, 50);
+    gsay_message(NAME, 123, NEUTRAL_REACTION);
 end
 
 procedure Staple13
 begin
-    float_msg(self_obj, message_str(862, 124), 2);
+    float_msg(self_obj, mstr(124), FLOAT_MSG_RED);
 end
 
 procedure Staple14
 begin
-    gsay_message(862, 126, 50);
+    gsay_message(NAME, 126, NEUTRAL_REACTION);
 end
 
 procedure Staple15
 begin
-    gsay_reply(862, 131);
-    giq_option(4, 862, 114, Staple15a, 50);
-    giq_option(4, 862, 117, StapleEnd, 50);
+    Reply(131);
+    giq_option(4, NAME, 114, Staple15a, NEUTRAL_REACTION);
+    giq_option(4, NAME, 117, StapleEnd, NEUTRAL_REACTION);
 end
 
 procedure Staple15a
@@ -330,15 +330,15 @@ begin
     variable LVar0 := 0;
     variable LVar1 := 0;
     LVar0 := item_caps_adjust(dude_obj, -750);
-    set_global_var(615, 2);
-    LVar1 := create_object_sid(237, 0, 0, -1);
+    set_global_var(GVAR_CHEMISTRY_BOOK, 2);
+    LVar1 := create_object(237, 0, 0);
     add_obj_to_inven(dude_obj, LVar1);
-    gsay_message(862, 132, 50);
+    gsay_message(NAME, 132, NEUTRAL_REACTION);
 end
 
 procedure Staple17
 begin
-    gsay_message(862, 133, 50);
+    gsay_message(NAME, 133, NEUTRAL_REACTION);
 end
 
 procedure StapleEnd
@@ -349,15 +349,15 @@ procedure Get_Stuff
 begin
     variable LVar0 := 0;
     variable LVar1 := 0;
-    LVar0 := create_object_sid(73, 0, 0, -1);
+    LVar0 := create_object(73, 0, 0);
     add_mult_objs_to_inven(self_obj, LVar0, 5);
-    LVar0 := create_object_sid(76, 0, 0, -1);
+    LVar0 := create_object(76, 0, 0);
     add_mult_objs_to_inven(self_obj, LVar0, 5);
-    LVar0 := create_object_sid(80, 0, 0, -1);
+    LVar0 := create_object(80, 0, 0);
     add_mult_objs_to_inven(self_obj, LVar0, 5);
-    LVar0 := create_object_sid(86, 0, 0, -1);
+    LVar0 := create_object(86, 0, 0);
     add_mult_objs_to_inven(self_obj, LVar0, 5);
-    LVar0 := create_object_sid(102, 0, 0, -1);
+    LVar0 := create_object(102, 0, 0);
     add_mult_objs_to_inven(self_obj, LVar0, 5);
     LVar1 := item_caps_adjust(self_obj, 500);
 end
@@ -401,31 +401,31 @@ begin
         set_local_var(0, 50);
         set_local_var(1, 2);
         set_local_var(2, 1);
-        set_local_var(0, local_var(0) + (5 * get_critter_stat(dude_obj, 3)) - 25);
-        set_local_var(0, local_var(0) + (10 * has_trait(0, dude_obj, 10)));
-        if (has_trait(0, dude_obj, 39)) then begin
-            if (global_var(155) > 0) then begin
-                set_local_var(0, local_var(0) + global_var(155));
+        set_local_var(0, local_var(0) + (5 * dude_charisma) - 25);
+        set_local_var(0, local_var(0) + (10 * has_trait(TRAIT_PERK, dude_obj, PERK_presence)));
+        if (has_trait(TRAIT_PERK, dude_obj, PERK_cult_of_personality)) then begin
+            if (global_var(GVAR_PLAYER_REPUATION) > 0) then begin
+                set_local_var(0, local_var(0) + global_var(GVAR_PLAYER_REPUATION));
             end
             else begin
-                set_local_var(0, local_var(0) - global_var(155));
+                set_local_var(0, local_var(0) - global_var(GVAR_PLAYER_REPUATION));
             end
         end
         else begin
             if (local_var(3) == 1) then begin
-                set_local_var(0, local_var(0) - global_var(155));
+                set_local_var(0, local_var(0) - global_var(GVAR_PLAYER_REPUATION));
             end
             else begin
-                set_local_var(0, local_var(0) + global_var(155));
+                set_local_var(0, local_var(0) + global_var(GVAR_PLAYER_REPUATION));
             end
         end
-        if (global_var(158) > 2) then begin
+        if (global_var(GVAR_CHILDKILLER_REPUATION) > 2) then begin
             set_local_var(0, local_var(0) - 30);
         end
-        if (((global_var(160) + global_var(159)) >= 25) and ((global_var(160) > (3 * global_var(159))) or (global_var(157) == 1))) then begin
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(160) > (3 * global_var(GVAR_GOOD_MONSTER))) or (global_var(GVAR_CHAMPION_REPUTATION) == 1))) then begin
             set_local_var(0, local_var(0) + 20);
         end
-        if (((global_var(160) + global_var(159)) >= 25) and ((global_var(159) > (2 * global_var(160))) or (global_var(156) == 1))) then begin
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
             set_local_var(0, local_var(0) - 20);
         end
         call ReactToLevel();
@@ -518,6 +518,5 @@ end
 
 procedure Goodbyes
 begin
-    exit_line := message_str(634, random(100, 105));
+    exit_line := message_str(SCRIPT_MODREACT, random(100, 105));
 end
-
