diff --git a/Games/Fallout1.Steam/Original/STUDENT.ssl b/Games/Fallout1.Steam/ReDefine/STUDENT.ssl
index a7df416..c360bc1 100644
--- a/Games/Fallout1.Steam/Original/STUDENT.ssl
+++ b/Games/Fallout1.Steam/ReDefine/STUDENT.ssl
@@ -18,8 +18,8 @@ begin
         if (local_var(3) == 0) then begin
             set_local_var(3, tile_num(self_obj));
         end
-        critter_add_trait(self_obj, 1, 6, 44);
-        critter_add_trait(self_obj, 1, 5, 62);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 44);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_BROTHERHOOD_INITIATE);
         my_knife := obj_carrying_pid_obj(self_obj, 4);
         if (local_var(0) != 0) then begin
             if (map_var(4) == local_var(0)) then begin
@@ -37,24 +37,24 @@ begin
         initialized := 1;
     end
     else begin
-        if (script_action == 18) then begin
+        if (script_action == destroy_proc) then begin
             call destroy_p_proc();
         end
         else begin
-            if (script_action == 12) then begin
-                if (global_var(250)) then begin
+            if (script_action == critter_proc) then begin
+                if (global_var(GVAR_ENEMY_BROTHERHOOD)) then begin
                     hostile := 1;
                 end
                 if (tile_distance_objs(self_obj, dude_obj) > 12) then begin
                     hostile := 0;
                 end
                 if (hostile) then begin
-                    set_global_var(250, 1);
+                    set_global_var(GVAR_ENEMY_BROTHERHOOD, 1);
                     hostile := 0;
-                    attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+                    attack(dude_obj);
                 end
                 else begin
-                    if (global_var(250) == 0) then begin
+                    if (global_var(GVAR_ENEMY_BROTHERHOOD) == 0) then begin
                         call critter_p_proc();
                     end
                 end
@@ -65,20 +65,20 @@ end
 
 procedure destroy_p_proc
 begin
-    set_global_var(250, 1);
+    set_global_var(GVAR_ENEMY_BROTHERHOOD, 1);
     rm_timer_event(self_obj);
     if (source_obj == dude_obj) then begin
-        if (((global_var(160) + global_var(159)) >= 25) and ((global_var(159) > (2 * global_var(160))) or (global_var(156) == 1))) then begin
-            set_global_var(156, 1);
-            set_global_var(157, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
+            set_global_var(GVAR_BERSERKER_REPUTATION, 1);
+            set_global_var(GVAR_CHAMPION_REPUTATION, 0);
         end
-        if (((global_var(160) + global_var(159)) >= 25) and ((global_var(160) > (3 * global_var(159))) or (global_var(157) == 1))) then begin
-            set_global_var(157, 1);
-            set_global_var(156, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(160) > (3 * global_var(GVAR_GOOD_MONSTER))) or (global_var(GVAR_CHAMPION_REPUTATION) == 1))) then begin
+            set_global_var(GVAR_CHAMPION_REPUTATION, 1);
+            set_global_var(GVAR_BERSERKER_REPUTATION, 0);
         end
-        set_global_var(159, global_var(159) + 1);
-        if ((global_var(159) % 2) == 0) then begin
-            set_global_var(155, global_var(155) - 1);
+        set_global_var(GVAR_GOOD_MONSTER, global_var(GVAR_GOOD_MONSTER) + 1);
+        if ((global_var(GVAR_GOOD_MONSTER) % 2) == 0) then begin
+            set_global_var(GVAR_PLAYER_REPUATION, global_var(GVAR_PLAYER_REPUATION) - 1);
         end
     end
 end
@@ -122,4 +122,3 @@ begin
         end
     end
 end
-
