diff --git a/Games/Fallout1.Steam/Original/SURF.ssl b/Games/Fallout1.Steam/ReDefine/SURF.ssl
index 6824582..2c28415 100644
--- a/Games/Fallout1.Steam/Original/SURF.ssl
+++ b/Games/Fallout1.Steam/ReDefine/SURF.ssl
@@ -99,19 +99,19 @@ variable exit_line;
 
 procedure start
 begin
-    if (script_action == 11) then begin
+    if (script_action == talk_proc) then begin
         call do_dialogue();
     end
     else begin
-        if ((script_action == 21) or (script_action == 3)) then begin
+        if ((script_action == look_at_proc) or (script_action == description_proc)) then begin
             script_overrides;
             display_msg(message_str(0, 100));
         end
         else begin
-            if (script_action == 12) then begin
+            if (script_action == critter_proc) then begin
                 if (hostile) then begin
                     hostile := 0;
-                    attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+                    attack(dude_obj);
                 end
             end
         end
@@ -121,17 +121,17 @@ end
 procedure do_dialogue
 begin
     call weapon_check();
-    start_gdialog(-1, self_obj, 4, -1, -1);
+    start_dialogue(self_obj, 4);
     sayStart;
-    if (global_var(195) == 1) then begin
+    if (global_var(GVAR_CATHEDRAL_ENEMY) == 1) then begin
         call surf00();
     end
     else begin
-        if (global_var(158) > 2) then begin
+        if (global_var(GVAR_CHILDKILLER_REPUATION) > 2) then begin
             call surf00();
         end
         else begin
-            if ((global_var(159) > (2 * global_var(160))) or (global_var(156) == 1)) then begin
+            if ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1)) then begin
                 call surf00();
             end
             else begin
@@ -139,7 +139,7 @@ begin
                     call surf02();
                 end
                 else begin
-                    if ((get_critter_stat(dude_obj, 3) > 6) and (GENDER == 1)) then begin
+                    if ((dude_charisma > 6) and (GENDER == 1)) then begin
                         call surf01();
                     end
                     else begin
@@ -179,7 +179,7 @@ end
 
 procedure surfcombat
 begin
-    set_global_var(195, 1);
+    set_global_var(GVAR_CATHEDRAL_ENEMY, 1);
     hostile := 1;
 end
 
@@ -202,24 +202,24 @@ end
 procedure surf01
 begin
     sayReply(0, message_str(0, 102));
-    if (get_critter_stat(dude_obj, 4) >= 7) then begin
+    if (dude_iq >= 7) then begin
         sayOption(message_str(0, 103), surf08);
     end
-    if (get_critter_stat(dude_obj, 4) >= 4) then begin
+    if (dude_iq >= 4) then begin
         sayOption(message_str(0, 104), surf09);
     end
-    if (get_critter_stat(dude_obj, 4) >= 4) then begin
+    if (dude_iq >= 4) then begin
         sayOption(message_str(0, 105), surf10);
     end
-    if (get_critter_stat(dude_obj, 4) >= 4) then begin
+    if (dude_iq >= 4) then begin
         sayOption(message_str(0, 106), surf11);
     end
-    if ((global_var(159) > (2 * global_var(160))) or (global_var(156) == 1)) then begin
-        if (get_critter_stat(dude_obj, 4) >= 4) then begin
+    if ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1)) then begin
+        if (dude_iq >= 4) then begin
             sayOption(message_str(0, 107), surf10);
         end
     end
-    if (-get_critter_stat(dude_obj, 4) >= -3) then begin
+    if (-dude_iq >= -3) then begin
         sayOption(message_str(0, 108), surf12);
     end
 end
@@ -228,20 +228,20 @@ procedure surf02
 begin
     set_local_var(5, 1);
     sayReply(0, message_str(0, 109));
-    if (get_critter_stat(dude_obj, 4) >= 7) then begin
+    if (dude_iq >= 7) then begin
         sayOption(message_str(0, 110), surf25);
     end
-    if (get_critter_stat(dude_obj, 4) >= 4) then begin
+    if (dude_iq >= 4) then begin
         sayOption(message_str(0, 111), surf26);
     end
-    if (get_critter_stat(dude_obj, 4) >= 4) then begin
+    if (dude_iq >= 4) then begin
         sayOption(message_str(0, 112), surf27);
     end
-    if (get_critter_stat(dude_obj, 4) >= 4) then begin
+    if (dude_iq >= 4) then begin
         sayOption(message_str(0, 113), surf28);
     end
-    if ((global_var(159) > (2 * global_var(160))) or (global_var(156) == 1)) then begin
-        if (get_critter_stat(dude_obj, 4) >= 4) then begin
+    if ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1)) then begin
+        if (dude_iq >= 4) then begin
             sayOption(message_str(0, 114), surf15);
         end
     end
@@ -250,20 +250,20 @@ end
 procedure surf03
 begin
     sayReply(0, message_str(0, 115));
-    if (get_critter_stat(dude_obj, 4) >= 7) then begin
+    if (dude_iq >= 7) then begin
         sayOption(message_str(0, 116), surf29);
     end
-    if (get_critter_stat(dude_obj, 4) >= 4) then begin
+    if (dude_iq >= 4) then begin
         sayOption(message_str(0, 117), surf03a);
     end
-    if (get_critter_stat(dude_obj, 4) >= 4) then begin
+    if (dude_iq >= 4) then begin
         sayOption(message_str(0, 118), surf03b);
     end
-    if (get_critter_stat(dude_obj, 4) >= 4) then begin
+    if (dude_iq >= 4) then begin
         sayOption(message_str(0, 119), surf33);
     end
-    if ((global_var(159) > (2 * global_var(160))) or (global_var(156) == 1)) then begin
-        if (get_critter_stat(dude_obj, 4) >= 4) then begin
+    if ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1)) then begin
+        if (dude_iq >= 4) then begin
             sayOption(message_str(0, 120), surf34);
         end
     end
@@ -290,23 +290,23 @@ begin
     else begin
         sayReply(0, message_str(0, 123));
     end
-    if (get_critter_stat(dude_obj, 4) >= 7) then begin
+    if (dude_iq >= 7) then begin
         sayOption(message_str(0, 124), surf05a);
     end
-    if (get_critter_stat(dude_obj, 4) >= 4) then begin
+    if (dude_iq >= 4) then begin
         sayOption(message_str(0, 125), surf36);
     end
-    if (get_critter_stat(dude_obj, 4) >= 4) then begin
+    if (dude_iq >= 4) then begin
         sayOption(message_str(0, 126), surf05b);
     end
-    if (get_critter_stat(dude_obj, 4) >= 4) then begin
+    if (dude_iq >= 4) then begin
         sayOption(message_str(0, 127), surf05c);
     end
-    if (get_critter_stat(dude_obj, 4) >= 4) then begin
+    if (dude_iq >= 4) then begin
         sayOption(message_str(0, 128), surf33);
     end
-    if ((global_var(159) > (2 * global_var(160))) or (global_var(156) == 1)) then begin
-        if (get_critter_stat(dude_obj, 4) >= 4) then begin
+    if ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1)) then begin
+        if (dude_iq >= 4) then begin
             sayOption(message_str(0, 129), surf34);
         end
     end
@@ -338,37 +338,37 @@ begin
     else begin
         sayReply(0, message_str(0, 132));
     end
-    if (get_critter_stat(dude_obj, 4) >= 7) then begin
+    if (dude_iq >= 7) then begin
         sayOption(message_str(0, 133), surf07a);
     end
-    if (get_critter_stat(dude_obj, 4) >= 4) then begin
+    if (dude_iq >= 4) then begin
         sayOption(message_str(0, 134), surf37);
     end
-    if (get_critter_stat(dude_obj, 4) >= 4) then begin
+    if (dude_iq >= 4) then begin
         sayOption(message_str(0, 135), surf07b);
     end
-    if (get_critter_stat(dude_obj, 4) >= 4) then begin
+    if (dude_iq >= 4) then begin
         sayOption(message_str(0, 136), surf07c);
     end
-    if (get_critter_stat(dude_obj, 4) >= 4) then begin
+    if (dude_iq >= 4) then begin
         sayOption(message_str(0, 137), surf33);
     end
-    if ((global_var(159) > (2 * global_var(160))) or (global_var(156) == 1)) then begin
-        if (get_critter_stat(dude_obj, 4) >= 4) then begin
+    if ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1)) then begin
+        if (dude_iq >= 4) then begin
             sayOption(message_str(0, 138), surf38);
         end
     end
-    if (get_critter_stat(dude_obj, 4) >= 4) then begin
+    if (dude_iq >= 4) then begin
         sayOption(message_str(0, 139), surf13);
     end
-    if (get_critter_stat(dude_obj, 4) >= 4) then begin
+    if (dude_iq >= 4) then begin
         sayOption(message_str(0, 140), surf14);
     end
-    if (get_critter_stat(dude_obj, 4) >= 4) then begin
+    if (dude_iq >= 4) then begin
         sayOption(message_str(0, 141), surf10);
     end
-    if ((global_var(159) > (2 * global_var(160))) or (global_var(156) == 1)) then begin
-        if (get_critter_stat(dude_obj, 4) >= 4) then begin
+    if ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1)) then begin
+        if (dude_iq >= 4) then begin
             sayOption(message_str(0, 142), surf15);
         end
     end
@@ -994,4 +994,3 @@ end
 *           Not implemented                            *
 *                                                      *
 *******************************************************/
-
