diff --git a/Games/Fallout1.Steam/Original/TERM.ssl b/Games/Fallout1.Steam/ReDefine/TERM.ssl
index f1da474..2b465ce 100644
--- a/Games/Fallout1.Steam/Original/TERM.ssl
+++ b/Games/Fallout1.Steam/ReDefine/TERM.ssl
@@ -4,12 +4,12 @@ procedure look_at_p_proc;
 
 procedure start
 begin
-    if (script_action == 21) then begin
+    if (script_action == look_at_proc) then begin
         call look_at_p_proc();
     end
     else begin
-        if (script_action == 6) then begin
-            display_msg(message_str(725, 101));
+        if (script_action == use_proc) then begin
+            display_mstr(101);
         end
     end
 end
@@ -17,6 +17,5 @@ end
 procedure look_at_p_proc
 begin
     script_overrides;
-    display_msg(message_str(725, 100));
+    display_mstr(100);
 end
-
