diff --git a/Games/Fallout1.Steam/Original/TOGGLE.ssl b/Games/Fallout1.Steam/ReDefine/TOGGLE.ssl
index f22f2e6..c2767f2 100644
--- a/Games/Fallout1.Steam/Original/TOGGLE.ssl
+++ b/Games/Fallout1.Steam/ReDefine/TOGGLE.ssl
@@ -7,11 +7,11 @@ import variable field_change;
 
 procedure start
 begin
-    if (script_action == 6) then begin
+    if (script_action == use_proc) then begin
         call use_p_proc();
     end
     else begin
-        if (script_action == 8) then begin
+        if (script_action == use_skill_on_proc) then begin
             call use_skill_on_p_proc();
         end
     end
@@ -42,13 +42,12 @@ procedure use_skill_on_p_proc
 begin
     if (action_being_used == 13) then begin
         script_overrides;
-        display_msg(message_str(864, 100));
+        display_mstr(100);
     end
     else begin
         if (action_being_used == 12) then begin
             script_overrides;
-            display_msg(message_str(864, 101));
+            display_mstr(101);
         end
     end
 end
-
