diff --git a/Games/Fallout1.Steam/Original/VALEL1.ssl b/Games/Fallout1.Steam/ReDefine/VALEL1.ssl
index a682d1e..9473b17 100644
--- a/Games/Fallout1.Steam/Original/VALEL1.ssl
+++ b/Games/Fallout1.Steam/ReDefine/VALEL1.ssl
@@ -3,14 +3,13 @@ procedure start;
 
 procedure start
 begin
-    if (script_action == 6) then begin
+    if (script_action == use_proc) then begin
         script_overrides;
         add_timer_event(self_obj, 8, 1);
     end
     else begin
-        if (script_action == 22) then begin
+        if (script_action == timed_event_proc) then begin
             move_to(dude_obj, 23118, 1);
         end
     end
 end
-
