diff --git a/Games/Fallout1.Steam/Original/VALTMEET.ssl b/Games/Fallout1.Steam/ReDefine/VALTMEET.ssl
index f5437d3..9f6faf0 100644
--- a/Games/Fallout1.Steam/Original/VALTMEET.ssl
+++ b/Games/Fallout1.Steam/ReDefine/VALTMEET.ssl
@@ -4,7 +4,7 @@ procedure spatial_p_proc;
 
 procedure start
 begin
-    if (script_action == 2) then begin
+    if (script_action == spatial_proc) then begin
         call spatial_p_proc();
     end
 end
@@ -13,9 +13,8 @@ procedure spatial_p_proc
 begin
     if (source_obj == dude_obj) then begin
         if (not(local_var(0))) then begin
-            display_msg(message_str(218, 100));
+            display_mstr(100);
             set_local_var(0, 1);
         end
     end
 end
-
