diff --git a/Games/Fallout1.Steam/Original/VAULTBUR.ssl b/Games/Fallout1.Steam/ReDefine/VAULTBUR.ssl
index 5b9c6e7..c71a97b 100644
--- a/Games/Fallout1.Steam/Original/VAULTBUR.ssl
+++ b/Games/Fallout1.Steam/ReDefine/VAULTBUR.ssl
@@ -18,15 +18,15 @@ export variable Tandi_ptr;
 
 procedure start
 begin
-    if (script_action == 15) then begin
+    if (script_action == map_enter_proc) then begin
         call map_enter_p_proc();
     end
     else begin
-        if (script_action == 23) then begin
+        if (script_action == map_update_proc) then begin
             call map_update_p_proc();
         end
         else begin
-            if (script_action == 16) then begin
+            if (script_action == map_exit_proc) then begin
                 call map_exit_p_proc();
             end
         end
@@ -36,15 +36,15 @@ end
 procedure map_enter_p_proc
 begin
     set_light_level(40);
-    if (global_var(32) == 1) then begin
+    if (global_var(GVAR_LOAD_MAP_INDEX) == 1) then begin
         override_map_start(94, 95, 0, 5);
     end
     else begin
-        if (global_var(32) == 2) then begin
+        if (global_var(GVAR_LOAD_MAP_INDEX) == 2) then begin
             override_map_start(109, 113, 1, 5);
         end
         else begin
-            if (global_var(32) == 3) then begin
+            if (global_var(GVAR_LOAD_MAP_INDEX) == 3) then begin
                 override_map_start(108, 93, 2, 5);
             end
             else begin
@@ -57,15 +57,15 @@ end
 
 procedure map_update_p_proc
 begin
-    if (elevation(dude_obj) == 0) then begin
-        set_global_var(563, 1);
+    if (dude_elevation == 0) then begin
+        set_global_var(GVAR_MARK_V15_2, 1);
     end
     else begin
-        if (elevation(dude_obj) == 1) then begin
-            set_global_var(564, 1);
+        if (dude_elevation == 1) then begin
+            set_global_var(GVAR_MARK_V15_3, 1);
         end
         else begin
-            set_global_var(565, 1);
+            set_global_var(GVAR_MARK_V15_4, 1);
         end
     end
     call update_party();
@@ -80,11 +80,11 @@ procedure add_party
 begin
     variable LVar0 := 0;
     variable LVar1 := 0;
-    party_elevation := elevation(dude_obj);
-    if (global_var(26) == 5) then begin
+    party_elevation := dude_elevation;
+    if (global_var(GVAR_TANDI_STATUS) == 5) then begin
         if (Tandi_ptr == 0) then begin
         end
-        critter_add_trait(Tandi_ptr, 1, 6, 0);
+        critter_add_trait(Tandi_ptr, TRAIT_OBJECT, OBJECT_TEAM_NUM, 0);
     end
 end
 
@@ -92,41 +92,41 @@ procedure update_party
 begin
     variable LVar0 := 0;
     variable LVar1 := 0;
-    if (elevation(dude_obj) != party_elevation) then begin
-        party_elevation := elevation(dude_obj);
-        if (global_var(118) == 2) then begin
+    if (dude_elevation != party_elevation) then begin
+        party_elevation := dude_elevation;
+        if (global_var(GVAR_IAN_STATUS) == 2) then begin
             if (Ian_ptr == 0) then begin
             end
             else begin
-                move_to(Ian_ptr, tile_num_in_direction(tile_num(dude_obj), 1, 2), elevation(dude_obj));
+                move_to(Ian_ptr, tile_num_in_direction(dude_tile, 1, 2), dude_elevation);
             end
         end
-        if (global_var(5)) then begin
+        if (global_var(GVAR_DOG_EMPATHY)) then begin
             if (Dog_ptr == 0) then begin
             end
             else begin
-                move_to(Dog_ptr, tile_num_in_direction(tile_num(dude_obj), 2, 1), elevation(dude_obj));
+                move_to(Dog_ptr, tile_num_in_direction(dude_tile, 2, 1), dude_elevation);
             end
         end
-        if (global_var(121) == 2) then begin
+        if (global_var(GVAR_HERNANDEZ_STATUS) == 2) then begin
             if (Tycho_ptr == 0) then begin
             end
             else begin
-                move_to(Tycho_ptr, tile_num_in_direction(tile_num(dude_obj), 3, 2), elevation(dude_obj));
+                move_to(Tycho_ptr, tile_num_in_direction(dude_tile, 3, 2), dude_elevation);
             end
         end
-        if (global_var(244) == 2) then begin
+        if (global_var(GVAR_KATJA_STATUS) == 2) then begin
             if (Katja_ptr == 0) then begin
             end
             else begin
-                move_to(Katja_ptr, tile_num_in_direction(tile_num(dude_obj), 4, 2), elevation(dude_obj));
+                move_to(Katja_ptr, tile_num_in_direction(dude_tile, 4, 2), dude_elevation);
             end
         end
-        if (global_var(26) == 5) then begin
+        if (global_var(GVAR_TANDI_STATUS) == 5) then begin
             if (Tandi_ptr == 0) then begin
             end
             else begin
-                move_to(Tandi_ptr, tile_num_in_direction(tile_num(dude_obj), 4, 4), elevation(dude_obj));
+                move_to(Tandi_ptr, tile_num_in_direction(dude_tile, 4, 4), dude_elevation);
             end
         end
     end
@@ -134,19 +134,18 @@ end
 
 procedure remove_party
 begin
-    if (global_var(118) == 2) then begin
-        set_global_var(118, 2);
+    if (global_var(GVAR_IAN_STATUS) == 2) then begin
+        set_global_var(GVAR_IAN_STATUS, 2);
     end
-    if (global_var(5)) then begin
-        set_global_var(5, 1);
+    if (global_var(GVAR_DOG_EMPATHY)) then begin
+        set_global_var(GVAR_DOG_EMPATHY, 1);
     end
-    if (global_var(121) == 2) then begin
-        set_global_var(121, 2);
+    if (global_var(GVAR_HERNANDEZ_STATUS) == 2) then begin
+        set_global_var(GVAR_HERNANDEZ_STATUS, 2);
     end
-    if (global_var(244) == 2) then begin
-        set_global_var(244, 2);
+    if (global_var(GVAR_KATJA_STATUS) == 2) then begin
+        set_global_var(GVAR_KATJA_STATUS, 2);
     end
-    if (global_var(26) == 5) then begin
+    if (global_var(GVAR_TANDI_STATUS) == 5) then begin
     end
 end
-
