diff --git a/Games/Fallout1.Steam/Original/VCOMPOP.ssl b/Games/Fallout1.Steam/ReDefine/VCOMPOP.ssl
index fefa752..6cf41cf 100644
--- a/Games/Fallout1.Steam/Original/VCOMPOP.ssl
+++ b/Games/Fallout1.Steam/ReDefine/VCOMPOP.ssl
@@ -5,11 +5,11 @@ procedure use_skill_on_p_proc;
 
 procedure start
 begin
-    if (script_action == 21) then begin
+    if (script_action == look_at_proc) then begin
         call look_at_p_proc();
     end
     else begin
-        if (script_action == 8) then begin
+        if (script_action == use_skill_on_proc) then begin
             call use_skill_on_p_proc();
         end
     end
@@ -18,30 +18,29 @@ end
 procedure look_at_p_proc
 begin
     script_overrides;
-    display_msg(message_str(214, 100));
-    if (global_var(101) == 2) then begin
-        display_msg(message_str(214, 104));
+    display_mstr(100);
+    if (global_var(GVAR_FIND_WATER_CHIP) == 2) then begin
+        display_mstr(104);
     end
 end
 
 procedure use_skill_on_p_proc
 begin
     script_overrides;
-    if (global_var(101) == 2) then begin
-        display_msg(message_str(214, 104));
+    if (global_var(GVAR_FIND_WATER_CHIP) == 2) then begin
+        display_mstr(104);
     end
     else begin
         if (action_being_used == 13) then begin
             if (is_success(roll_vs_skill(dude_obj, 13, 0))) then begin
-                display_msg(message_str(214, 101));
+                display_mstr(101);
             end
             else begin
-                display_msg(message_str(214, 102));
+                display_mstr(102);
             end
         end
         else begin
-            display_msg(message_str(214, 103));
+            display_mstr(103);
         end
     end
 end
-
