diff --git a/Games/Fallout1.Steam/Original/VWEPLOKR.ssl b/Games/Fallout1.Steam/ReDefine/VWEPLOKR.ssl
index 8eb4926..927a3de 100644
--- a/Games/Fallout1.Steam/Original/VWEPLOKR.ssl
+++ b/Games/Fallout1.Steam/ReDefine/VWEPLOKR.ssl
@@ -6,7 +6,7 @@ import variable VWeapLocker_ptr;
 
 procedure start
 begin
-    if (script_action == 15) then begin
+    if (script_action == map_enter_proc) then begin
         call map_enter_p_proc();
     end
 end
@@ -15,4 +15,3 @@ procedure map_enter_p_proc
 begin
     VWeapLocker_ptr := self_obj;
 end
-
