diff --git a/Games/Fallout1.Steam/Original/WIFE.ssl b/Games/Fallout1.Steam/ReDefine/WIFE.ssl
index 1456ccf..e00e73c 100644
--- a/Games/Fallout1.Steam/Original/WIFE.ssl
+++ b/Games/Fallout1.Steam/ReDefine/WIFE.ssl
@@ -19,26 +19,26 @@ procedure start
 begin
     if (Only_Once) then begin
         Only_Once := 0;
-        critter_add_trait(self_obj, 1, 6, 2);
-        critter_add_trait(self_obj, 1, 5, 6);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 2);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_PEASANT);
     end
-    if (script_action == 12) then begin
+    if (script_action == critter_proc) then begin
         call critter_p_proc();
     end
     else begin
-        if (script_action == 11) then begin
+        if (script_action == talk_proc) then begin
             call talk_p_proc();
         end
         else begin
-            if (script_action == 4) then begin
+            if (script_action == pickup_proc) then begin
                 call pickup_p_proc();
             end
             else begin
-                if (script_action == 18) then begin
+                if (script_action == destroy_proc) then begin
                     call destroy_p_proc();
                 end
                 else begin
-                    if (script_action == 14) then begin
+                    if (script_action == damage_proc) then begin
                         call damage_p_proc();
                     end
                 end
@@ -63,25 +63,25 @@ end
 
 procedure Wife01
 begin
-    gsay_reply(119, 100);
-    giq_option(4, 119, 101, Wife02, 50);
-    giq_option(4, 119, 107, WifeEnd, 50);
-    giq_option(-3, 119, 102, Wife03, 50);
+    Reply(100);
+    giq_option(4, NAME, 101, Wife02, NEUTRAL_REACTION);
+    giq_option(4, NAME, 107, WifeEnd, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 102, Wife03, NEUTRAL_REACTION);
 end
 
 procedure Wife02
 begin
-    gsay_message(119, 103, 50);
+    gsay_message(NAME, 103, NEUTRAL_REACTION);
 end
 
 procedure Wife03
 begin
-    gsay_message(119, 104, 50);
+    gsay_message(NAME, 104, NEUTRAL_REACTION);
 end
 
 procedure Wife00n
 begin
-    float_msg(self_obj, message_str(119, 105), 8);
+    float_msg(self_obj, mstr(105), FLOAT_MSG_YELLOW);
 end
 
 procedure WifeEnd
@@ -94,20 +94,20 @@ begin
         set_local_var(0, local_var(0) + 1);
         set_map_var(2, 0);
         if (local_var(0) < 3) then begin
-            float_msg(self_obj, message_str(129, 308), 2);
+            float_msg(self_obj, message_str(SCRIPT_RAZLO, 308), FLOAT_MSG_RED);
         end
         else begin
             HOSTILE := 1;
         end
     end
     if (obj_can_see_obj(self_obj, dude_obj)) then begin
-        if (global_var(246) == 1) then begin
+        if (global_var(GVAR_ENEMY_SHADY_SANDS) == 1) then begin
             HOSTILE := 1;
         end
     end
     if (HOSTILE) then begin
         HOSTILE := 0;
-        attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+        attack(dude_obj);
     end
 end
 
@@ -124,20 +124,20 @@ end
 procedure destroy_p_proc
 begin
     if (source_obj == dude_obj) then begin
-        set_global_var(246, 1);
+        set_global_var(GVAR_ENEMY_SHADY_SANDS, 1);
     end
     if (source_obj == dude_obj) then begin
-        if (((global_var(160) + global_var(159)) >= 25) and ((global_var(159) > (2 * global_var(160))) or (global_var(156) == 1))) then begin
-            set_global_var(156, 1);
-            set_global_var(157, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
+            set_global_var(GVAR_BERSERKER_REPUTATION, 1);
+            set_global_var(GVAR_CHAMPION_REPUTATION, 0);
         end
-        if (((global_var(160) + global_var(159)) >= 25) and ((global_var(160) > (3 * global_var(159))) or (global_var(157) == 1))) then begin
-            set_global_var(157, 1);
-            set_global_var(156, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(160) > (3 * global_var(GVAR_GOOD_MONSTER))) or (global_var(GVAR_CHAMPION_REPUTATION) == 1))) then begin
+            set_global_var(GVAR_CHAMPION_REPUTATION, 1);
+            set_global_var(GVAR_BERSERKER_REPUTATION, 0);
         end
-        set_global_var(159, global_var(159) + 1);
-        if ((global_var(159) % 2) == 0) then begin
-            set_global_var(155, global_var(155) - 1);
+        set_global_var(GVAR_GOOD_MONSTER, global_var(GVAR_GOOD_MONSTER) + 1);
+        if ((global_var(GVAR_GOOD_MONSTER) % 2) == 0) then begin
+            set_global_var(GVAR_PLAYER_REPUATION, global_var(GVAR_PLAYER_REPUATION) - 1);
         end
     end
 end
@@ -145,7 +145,6 @@ end
 procedure damage_p_proc
 begin
     if (source_obj == dude_obj) then begin
-        set_global_var(246, 1);
+        set_global_var(GVAR_ENEMY_SHADY_SANDS, 1);
     end
 end
-
