diff --git a/Games/Fallout1.Steam/Original/WMCARVN.ssl b/Games/Fallout1.Steam/ReDefine/WMCARVN.ssl
index b137c8c..4b04bd0 100644
--- a/Games/Fallout1.Steam/Original/WMCARVN.ssl
+++ b/Games/Fallout1.Steam/ReDefine/WMCARVN.ssl
@@ -65,26 +65,26 @@ procedure start
 begin
     if (Only_Once) then begin
         Only_Once := 0;
-        critter_add_trait(self_obj, 1, 6, 35);
-        critter_add_trait(self_obj, 1, 5, 50);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 35);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_MERCHANT);
     end
-    if (script_action == 21) then begin
+    if (script_action == look_at_proc) then begin
         call look_at_p_proc();
     end
     else begin
-        if (script_action == 4) then begin
+        if (script_action == pickup_proc) then begin
             call pickup_p_proc();
         end
         else begin
-            if (script_action == 11) then begin
+            if (script_action == talk_proc) then begin
                 call talk_p_proc();
             end
             else begin
-                if (script_action == 12) then begin
+                if (script_action == critter_proc) then begin
                     call critter_p_proc();
                 end
                 else begin
-                    if (script_action == 18) then begin
+                    if (script_action == destroy_proc) then begin
                         call destroy_p_proc();
                     end
                 end
@@ -102,7 +102,7 @@ procedure critter_p_proc
 begin
     if (hostile) then begin
         hostile := 0;
-        attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+        attack(dude_obj);
     end
 end
 
@@ -140,7 +140,7 @@ begin
             call WMDriver15();
         end
         else begin
-            if (global_var(205) == 3) then begin
+            if (global_var(GVAR_WATER_JOB) == 3) then begin
                 start_gdialog(605, self_obj, 4, -1, -1);
                 gsay_start;
                 call WMDriver19();
@@ -148,7 +148,7 @@ begin
                 end_dialogue;
             end
             else begin
-                if (global_var(205) == 4) then begin
+                if (global_var(GVAR_WATER_JOB) == 4) then begin
                     start_gdialog(605, self_obj, 4, -1, -1);
                     gsay_start;
                     call WMDriver27();
@@ -201,17 +201,17 @@ end
 procedure destroy_p_proc
 begin
     if (source_obj == dude_obj) then begin
-        if (((global_var(160) + global_var(159)) >= 25) and ((global_var(159) > (2 * global_var(160))) or (global_var(156) == 1))) then begin
-            set_global_var(156, 1);
-            set_global_var(157, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
+            set_global_var(GVAR_BERSERKER_REPUTATION, 1);
+            set_global_var(GVAR_CHAMPION_REPUTATION, 0);
         end
-        if (((global_var(160) + global_var(159)) >= 25) and ((global_var(160) > (3 * global_var(159))) or (global_var(157) == 1))) then begin
-            set_global_var(157, 1);
-            set_global_var(156, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(160) > (3 * global_var(GVAR_GOOD_MONSTER))) or (global_var(GVAR_CHAMPION_REPUTATION) == 1))) then begin
+            set_global_var(GVAR_CHAMPION_REPUTATION, 1);
+            set_global_var(GVAR_BERSERKER_REPUTATION, 0);
         end
-        set_global_var(159, global_var(159) + 1);
-        if ((global_var(159) % 2) == 0) then begin
-            set_global_var(155, global_var(155) - 1);
+        set_global_var(GVAR_GOOD_MONSTER, global_var(GVAR_GOOD_MONSTER) + 1);
+        if ((global_var(GVAR_GOOD_MONSTER) % 2) == 0) then begin
+            set_global_var(GVAR_PLAYER_REPUATION, global_var(GVAR_PLAYER_REPUATION) - 1);
         end
     end
 end
@@ -219,7 +219,7 @@ end
 procedure look_at_p_proc
 begin
     script_overrides;
-    display_msg(message_str(605, 100));
+    display_mstr(100);
 end
 
 procedure RecalcDate
@@ -294,51 +294,51 @@ begin
     LVar0 := local_var(6);
     LVar1 := local_var(5);
     if (LVar0 == 1) then begin
-        DayString := message_str(605, 204);
+        DayString := mstr(204);
     end
     else begin
         if (LVar0 == 2) then begin
-            DayString := message_str(605, 205);
+            DayString := mstr(205);
         end
         else begin
             if (LVar0 == 3) then begin
-                DayString := message_str(605, 206);
+                DayString := mstr(206);
             end
             else begin
                 if (LVar0 == 4) then begin
-                    DayString := message_str(605, 207);
+                    DayString := mstr(207);
                 end
                 else begin
                     if (LVar0 == 5) then begin
-                        DayString := message_str(605, 208);
+                        DayString := mstr(208);
                     end
                     else begin
                         if (LVar0 == 6) then begin
-                            DayString := message_str(605, 209);
+                            DayString := mstr(209);
                         end
                         else begin
                             if (LVar0 == 7) then begin
-                                DayString := message_str(605, 210);
+                                DayString := mstr(210);
                             end
                             else begin
                                 if (LVar0 == 8) then begin
-                                    DayString := message_str(605, 211);
+                                    DayString := mstr(211);
                                 end
                                 else begin
                                     if (LVar0 == 9) then begin
-                                        DayString := message_str(605, 212);
+                                        DayString := mstr(212);
                                     end
                                     else begin
                                         if (LVar0 == 10) then begin
-                                            DayString := message_str(605, 213);
+                                            DayString := mstr(213);
                                         end
                                         else begin
                                             if (LVar0 == 11) then begin
-                                                DayString := message_str(605, 214);
+                                                DayString := mstr(214);
                                             end
                                             else begin
                                                 if (LVar0 == 12) then begin
-                                                    DayString := message_str(605, 215);
+                                                    DayString := mstr(215);
                                                 end
                                             end
                                         end
@@ -352,27 +352,27 @@ begin
         end
     end
     if (LVar1 == 1) then begin
-        DayString := DayString + message_str(605, 197);
+        DayString := DayString + mstr(197);
     end
     else begin
         if (LVar1 == 5) then begin
-            DayString := DayString + message_str(605, 198);
+            DayString := DayString + mstr(198);
         end
         else begin
             if (LVar1 == 10) then begin
-                DayString := DayString + message_str(605, 199);
+                DayString := DayString + mstr(199);
             end
             else begin
                 if (LVar1 == 15) then begin
-                    DayString := DayString + message_str(605, 200);
+                    DayString := DayString + mstr(200);
                 end
                 else begin
                     if (LVar1 == 20) then begin
-                        DayString := DayString + message_str(605, 201);
+                        DayString := DayString + mstr(201);
                     end
                     else begin
                         if (LVar1 == 25) then begin
-                            DayString := DayString + message_str(605, 202);
+                            DayString := DayString + mstr(202);
                         end
                     end
                 end
@@ -386,32 +386,32 @@ begin
     variable LVar0 := 0;
     LVar0 := obj_pid(source_obj);
     if (party_member_obj(LVar0) != 0) then begin
-        set_global_var(248, 1);
+        set_global_var(GVAR_ENEMY_HUB, 1);
     end
 end
 
 procedure WMDriver00
 begin
     if (local_var(4) == 0) then begin
-        float_msg(self_obj, message_str(605, 101), 2);
+        float_msg(self_obj, mstr(101), FLOAT_MSG_RED);
         set_local_var(4, 1);
     end
     else begin
-        float_msg(self_obj, message_str(605, 102), 2);
+        float_msg(self_obj, mstr(102), FLOAT_MSG_RED);
     end
 end
 
 procedure WMDriver01
 begin
-    if (global_var(71) == 0) then begin
-        set_global_var(71, 1);
+    if (global_var(GVAR_JUNKTOWN) == 0) then begin
+        set_global_var(GVAR_JUNKTOWN, 1);
     end
-    if (global_var(75) == 0) then begin
-        set_global_var(75, 1);
+    if (global_var(GVAR_ANGELS_BONEYARD) == 0) then begin
+        set_global_var(GVAR_ANGELS_BONEYARD, 1);
     end
-    gsay_reply(605, 103);
-    giq_option(4, 605, 104, WMDriver02, 50);
-    giq_option(-3, 605, 105, WMDriver02, 50);
+    Reply(103);
+    giq_option(4, NAME, 104, WMDriver02, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 105, WMDriver02, NEUTRAL_REACTION);
 end
 
 procedure WMDriver02
@@ -420,10 +420,10 @@ begin
     call RecalcDateString();
     LVar0 := local_var(5);
     if (LVar0 == get_day) then begin
-        gsay_reply(605, message_str(605, 106) + message_str(605, 107));
+        Reply(mstr(106) + mstr(107));
     end
     else begin
-        gsay_reply(605, message_str(605, 106) + message_str(605, 108) + DayString + message_str(605, 111));
+        Reply(mstr(106) + mstr(108) + DayString + mstr(111));
     end
     call WMDriver04();
 end
@@ -435,26 +435,26 @@ begin
         call RecalcDateString();
         LVar0 := local_var(5);
         if (LVar0 == get_day) then begin
-            gsay_reply(605, message_str(605, 107));
+            Reply(mstr(107));
         end
         else begin
-            gsay_reply(605, message_str(605, 108) + DayString + message_str(605, 111));
+            Reply(mstr(108) + DayString + mstr(111));
         end
         call WMDriver04();
     end
     else begin
-        gsay_reply(605, 112);
+        Reply(112);
         call WMDriver04();
     end
 end
 
 procedure WMDriver04
 begin
-    giq_option(4, 605, 113, WMDriver04a, 50);
-    giq_option(4, 605, 114, WMDriver05, 50);
-    giq_option(4, 605, 115, WMDriver06, 50);
-    giq_option(-3, 605, 116, WMDriver07, 50);
-    giq_option(-3, 605, 117, WMDriver07, 50);
+    giq_option(4, NAME, 113, WMDriver04a, NEUTRAL_REACTION);
+    giq_option(4, NAME, 114, WMDriver05, NEUTRAL_REACTION);
+    giq_option(4, NAME, 115, WMDriver06, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 116, WMDriver07, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 117, WMDriver07, NEUTRAL_REACTION);
 end
 
 procedure WMDriver04a
@@ -474,23 +474,23 @@ end
 procedure WMDriver05
 begin
     set_local_var(10, 0);
-    gsay_message(605, 118, 50);
+    gsay_message(NAME, 118, NEUTRAL_REACTION);
 end
 
 procedure WMDriver06
 begin
     set_local_var(10, 0);
-    gsay_message(605, 120, 50);
+    gsay_message(NAME, 120, NEUTRAL_REACTION);
 end
 
 procedure WMDriver07
 begin
-    gsay_message(605, 121, 50);
+    gsay_message(NAME, 121, NEUTRAL_REACTION);
 end
 
 procedure WMDriver08
 begin
-    gsay_message(605, 123, 50);
+    gsay_message(NAME, 123, NEUTRAL_REACTION);
 end
 
 procedure WMDriver10
@@ -498,14 +498,14 @@ begin
     if (local_var(8) == 0) then begin
         set_local_var(8, 1);
         set_map_var(2, 0);
-        gsay_reply(605, 125);
-        giq_option(4, 605, 126, WMDriver08, 50);
-        giq_option(4, 605, 127, WMDriver10a, 50);
-        giq_option(4, 605, 128, WMDriver31, 51);
-        giq_option(-3, 605, 130, WMDriver31, 51);
+        Reply(125);
+        giq_option(4, NAME, 126, WMDriver08, NEUTRAL_REACTION);
+        giq_option(4, NAME, 127, WMDriver10a, NEUTRAL_REACTION);
+        giq_option(4, NAME, 128, WMDriver31, BAD_REACTION);
+        giq_option(-3, NAME, 130, WMDriver31, BAD_REACTION);
     end
     else begin
-        gsay_message(605, 131, 51);
+        gsay_message(NAME, 131, BAD_REACTION);
         call combat();
     end
 end
@@ -524,27 +524,27 @@ procedure WMDriver11
 begin
     set_map_var(1, 0);
     set_map_var(2, 0);
-    gsay_message(605, 132, 50);
+    gsay_message(NAME, 132, NEUTRAL_REACTION);
 end
 
 procedure WMDriver12
 begin
-    gsay_message(605, 133, 51);
+    gsay_message(NAME, 133, BAD_REACTION);
     call combat();
 end
 
 procedure WMDriver13
 begin
-    float_msg(self_obj, message_str(605, 137), 2);
+    float_msg(self_obj, mstr(137), FLOAT_MSG_RED);
 end
 
 procedure WMDriver14
 begin
     if (local_var(8) > 0) then begin
-        float_msg(self_obj, message_str(605, 138), 2);
+        float_msg(self_obj, mstr(138), FLOAT_MSG_RED);
     end
     else begin
-        float_msg(self_obj, message_str(605, 139), 2);
+        float_msg(self_obj, mstr(139), FLOAT_MSG_RED);
         call WMDriver31();
     end
 end
@@ -554,19 +554,19 @@ begin
     variable LVar0 := 0;
     LVar0 := random(1, 4);
     if (LVar0 == 1) then begin
-        float_msg(self_obj, message_str(605, 140), 2);
+        float_msg(self_obj, mstr(140), FLOAT_MSG_RED);
     end
     else begin
         if (LVar0 == 2) then begin
-            float_msg(self_obj, message_str(605, 141), 2);
+            float_msg(self_obj, mstr(141), FLOAT_MSG_RED);
         end
         else begin
             if (LVar0 == 3) then begin
-                float_msg(self_obj, message_str(605, 142), 2);
+                float_msg(self_obj, mstr(142), FLOAT_MSG_RED);
             end
             else begin
                 if (LVar0 == 4) then begin
-                    float_msg(self_obj, message_str(605, 143), 2);
+                    float_msg(self_obj, mstr(143), FLOAT_MSG_RED);
                 end
             end
         end
@@ -575,7 +575,7 @@ end
 
 procedure WMDriver17
 begin
-    gsay_reply(605, 145);
+    Reply(145);
     call WMDriver33();
     call WMDriver20();
 end
@@ -583,56 +583,56 @@ end
 procedure WMDriver18
 begin
     call RecalcDateString();
-    gsay_reply(605, message_str(605, 146) + DayString + message_str(605, 153));
-    giq_option(4, 605, 154, WMDriver30, 50);
-    giq_option(4, 605, 155, WMDriver34, 50);
+    Reply(mstr(146) + DayString + mstr(153));
+    giq_option(4, NAME, 154, WMDriver30, NEUTRAL_REACTION);
+    giq_option(4, NAME, 155, WMDriver34, NEUTRAL_REACTION);
 end
 
 procedure WMDriver19
 begin
     set_map_var(2, 0);
-    set_global_var(205, 0);
-    gsay_message(605, 156, 51);
+    set_global_var(GVAR_WATER_JOB, 0);
+    gsay_message(NAME, 156, BAD_REACTION);
 end
 
 procedure WMDriver20
 begin
-    giq_option(4, 605, 157, WMDriver21a, 50);
-    giq_option(4, 605, 158, WMDriver21b, 50);
-    giq_option(4, 605, 159, WMDriver34, 50);
+    giq_option(4, NAME, 157, WMDriver21a, NEUTRAL_REACTION);
+    giq_option(4, NAME, 158, WMDriver21b, NEUTRAL_REACTION);
+    giq_option(4, NAME, 159, WMDriver34, NEUTRAL_REACTION);
 end
 
 procedure WMDriver21a
 begin
     variable LVar0 := 0;
-    set_global_var(200, 1);
+    set_global_var(GVAR_WATER_MERCHANTS_STATUS, 1);
     LVar0 := random(1, 7);
     if (LVar0 == 1) then begin
-        gsay_message(605, 160, 50);
+        gsay_message(NAME, 160, NEUTRAL_REACTION);
     end
     else begin
         if (LVar0 == 2) then begin
-            gsay_message(605, 161, 50);
+            gsay_message(NAME, 161, NEUTRAL_REACTION);
         end
         else begin
             if (LVar0 == 3) then begin
-                gsay_message(605, 162, 50);
+                gsay_message(NAME, 162, NEUTRAL_REACTION);
             end
             else begin
                 if (LVar0 == 4) then begin
-                    gsay_message(605, 163, 50);
+                    gsay_message(NAME, 163, NEUTRAL_REACTION);
                 end
                 else begin
                     if (LVar0 == 5) then begin
-                        gsay_message(605, 167, 50);
+                        gsay_message(NAME, 167, NEUTRAL_REACTION);
                     end
                     else begin
                         if (LVar0 == 6) then begin
-                            gsay_message(605, 168, 50);
+                            gsay_message(NAME, 168, NEUTRAL_REACTION);
                         end
                         else begin
                             if (LVar0 == 7) then begin
-                                gsay_message(605, 169, 50);
+                                gsay_message(NAME, 169, NEUTRAL_REACTION);
                             end
                         end
                     end
@@ -652,34 +652,34 @@ end
 procedure WMDriver21b
 begin
     variable LVar0 := 0;
-    set_global_var(200, 1);
+    set_global_var(GVAR_WATER_MERCHANTS_STATUS, 1);
     LVar0 := random(1, 7);
     if (LVar0 == 1) then begin
-        gsay_message(605, 160, 50);
+        gsay_message(NAME, 160, NEUTRAL_REACTION);
     end
     else begin
         if (LVar0 == 2) then begin
-            gsay_message(605, 161, 50);
+            gsay_message(NAME, 161, NEUTRAL_REACTION);
         end
         else begin
             if (LVar0 == 3) then begin
-                gsay_message(605, 162, 50);
+                gsay_message(NAME, 162, NEUTRAL_REACTION);
             end
             else begin
                 if (LVar0 == 4) then begin
-                    gsay_message(605, 163, 50);
+                    gsay_message(NAME, 163, NEUTRAL_REACTION);
                 end
                 else begin
                     if (LVar0 == 5) then begin
-                        gsay_message(605, 164, 50);
+                        gsay_message(NAME, 164, NEUTRAL_REACTION);
                     end
                     else begin
                         if (LVar0 == 6) then begin
-                            gsay_message(605, 165, 50);
+                            gsay_message(NAME, 165, NEUTRAL_REACTION);
                         end
                         else begin
                             if (LVar0 == 7) then begin
-                                gsay_message(605, 166, 50);
+                                gsay_message(NAME, 166, NEUTRAL_REACTION);
                             end
                         end
                     end
@@ -698,20 +698,20 @@ end
 
 procedure WMDriver27
 begin
-    gsay_reply(605, 185);
-    giq_option(4, 605, 186, WMDriver04a, 50);
-    giq_option(4, 605, 187, WMDriver05, 50);
-    giq_option(4, 605, 188, WMDriver06, 50);
-    giq_option(-3, 605, 189, WMDriver07, 50);
-    giq_option(-3, 605, 190, WMDriver07, 50);
+    Reply(185);
+    giq_option(4, NAME, 186, WMDriver04a, NEUTRAL_REACTION);
+    giq_option(4, NAME, 187, WMDriver05, NEUTRAL_REACTION);
+    giq_option(4, NAME, 188, WMDriver06, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 189, WMDriver07, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 190, WMDriver07, NEUTRAL_REACTION);
 end
 
 procedure WMDriver28
 begin
-    gsay_reply(605, 191);
-    giq_option(4, 605, 192, WMDriver17, 50);
-    giq_option(4, 605, 193, WMDriver35, 50);
-    giq_option(-3, 605, 194, WMDriver07, 50);
+    Reply(191);
+    giq_option(4, NAME, 192, WMDriver17, NEUTRAL_REACTION);
+    giq_option(4, NAME, 193, WMDriver35, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 194, WMDriver07, NEUTRAL_REACTION);
 end
 
 procedure WMDriver29
@@ -719,7 +719,7 @@ begin
     variable LVar0 := 0;
     call RecalcDateString();
     LVar0 := local_var(5);
-    gsay_message(605, message_str(605, 196) + DayString + message_str(605, 203) + message_str(605, 195), 50);
+    gsay_message(NAME, mstr(196) + DayString + mstr(203) + mstr(195), NEUTRAL_REACTION);
 end
 
 procedure WMDriver30
@@ -740,7 +740,7 @@ procedure WMDriver33
 begin
     variable LVar0 := 0;
     set_local_var(10, 0);
-    set_global_var(205, 1);
+    set_global_var(GVAR_WATER_JOB, 1);
     LVar0 := local_var(9);
     set_local_var(9, LVar0 + 1);
 end
@@ -760,31 +760,31 @@ begin
         set_local_var(0, 50);
         set_local_var(1, 2);
         set_local_var(2, 1);
-        set_local_var(0, local_var(0) + (5 * get_critter_stat(dude_obj, 3)) - 25);
-        set_local_var(0, local_var(0) + (10 * has_trait(0, dude_obj, 10)));
-        if (has_trait(0, dude_obj, 39)) then begin
-            if (global_var(155) > 0) then begin
-                set_local_var(0, local_var(0) + global_var(155));
+        set_local_var(0, local_var(0) + (5 * dude_charisma) - 25);
+        set_local_var(0, local_var(0) + (10 * has_trait(TRAIT_PERK, dude_obj, PERK_presence)));
+        if (has_trait(TRAIT_PERK, dude_obj, PERK_cult_of_personality)) then begin
+            if (global_var(GVAR_PLAYER_REPUATION) > 0) then begin
+                set_local_var(0, local_var(0) + global_var(GVAR_PLAYER_REPUATION));
             end
             else begin
-                set_local_var(0, local_var(0) - global_var(155));
+                set_local_var(0, local_var(0) - global_var(GVAR_PLAYER_REPUATION));
             end
         end
         else begin
             if (local_var(3) == 1) then begin
-                set_local_var(0, local_var(0) - global_var(155));
+                set_local_var(0, local_var(0) - global_var(GVAR_PLAYER_REPUATION));
             end
             else begin
-                set_local_var(0, local_var(0) + global_var(155));
+                set_local_var(0, local_var(0) + global_var(GVAR_PLAYER_REPUATION));
             end
         end
-        if (global_var(158) > 2) then begin
+        if (global_var(GVAR_CHILDKILLER_REPUATION) > 2) then begin
             set_local_var(0, local_var(0) - 30);
         end
-        if (((global_var(160) + global_var(159)) >= 25) and ((global_var(160) > (3 * global_var(159))) or (global_var(157) == 1))) then begin
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(160) > (3 * global_var(GVAR_GOOD_MONSTER))) or (global_var(GVAR_CHAMPION_REPUTATION) == 1))) then begin
             set_local_var(0, local_var(0) + 20);
         end
-        if (((global_var(160) + global_var(159)) >= 25) and ((global_var(159) > (2 * global_var(160))) or (global_var(156) == 1))) then begin
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
             set_local_var(0, local_var(0) - 20);
         end
         call ReactToLevel();
@@ -877,6 +877,5 @@ end
 
 procedure Goodbyes
 begin
-    exit_line := message_str(634, random(100, 105));
+    exit_line := message_str(SCRIPT_MODREACT, random(100, 105));
 end
-
