diff --git a/Games/Fallout1.Steam/Original/WOODDOOR.ssl b/Games/Fallout1.Steam/ReDefine/WOODDOOR.ssl
index cef802a..9a1efac 100644
--- a/Games/Fallout1.Steam/Original/WOODDOOR.ssl
+++ b/Games/Fallout1.Steam/ReDefine/WOODDOOR.ssl
@@ -4,7 +4,7 @@ procedure damage_p_proc;
 
 procedure start
 begin
-    if (script_action == 14) then begin
+    if (script_action == damage_proc) then begin
         call damage_p_proc();
     end
 end
@@ -13,4 +13,3 @@ procedure damage_p_proc
 begin
     set_obj_visibility(self_obj, 1);
 end
-
