diff --git a/Games/Fallout1.Steam/Original/WORKROOM.ssl b/Games/Fallout1.Steam/ReDefine/WORKROOM.ssl
index 1368fc0..cc05572 100644
--- a/Games/Fallout1.Steam/Original/WORKROOM.ssl
+++ b/Games/Fallout1.Steam/ReDefine/WORKROOM.ssl
@@ -10,15 +10,15 @@ variable new_tile;
 
 procedure start
 begin
-    if (script_action == 15) then begin
+    if (script_action == map_enter_proc) then begin
         call initialize();
     end
     else begin
-        if (script_action == 22) then begin
+        if (script_action == timed_event_proc) then begin
             call explode();
         end
         else begin
-            if (script_action == 6) then begin
+            if (script_action == use_proc) then begin
                 call explode();
             end
         end
@@ -38,6 +38,5 @@ procedure explode
 begin
     new_tile := tile_num_in_direction(tile_num(self_obj), 1, 1);
     explosion(new_tile, elevation(self_obj), 0);
-    float_msg(self_obj, message_str(736, random(108, 115)), 8);
+    float_msg(self_obj, mstr(random(108, 115)), FLOAT_MSG_YELLOW);
 end
-
