diff --git a/Games/Fallout1.Steam/Original/WORLD.ssl b/Games/Fallout1.Steam/ReDefine/WORLD.ssl
index 95c9da8..1655c87 100644
--- a/Games/Fallout1.Steam/Original/WORLD.ssl
+++ b/Games/Fallout1.Steam/ReDefine/WORLD.ssl
@@ -3,7 +3,7 @@ procedure start;
 
 procedure start
 begin
-    if (script_action == 2) then begin
+    if (script_action == spatial_proc) then begin
         if (source_obj == dude_obj) then begin
             script_overrides;
             display_msg("You are now entering the world map.");
@@ -11,4 +11,3 @@ begin
         end
     end
 end
-
