diff --git a/Games/Fallout1.Steam/Original/WTRGRD.ssl b/Games/Fallout1.Steam/ReDefine/WTRGRD.ssl
index 202082a..da2c686 100644
--- a/Games/Fallout1.Steam/Original/WTRGRD.ssl
+++ b/Games/Fallout1.Steam/ReDefine/WTRGRD.ssl
@@ -47,31 +47,31 @@ variable exit_line;
 
 procedure start
 begin
-    if (script_action == 12) then begin
+    if (script_action == critter_proc) then begin
         call critter_p_proc();
     end
     else begin
-        if (script_action == 18) then begin
+        if (script_action == destroy_proc) then begin
             call destroy_p_proc();
         end
         else begin
-            if (script_action == 21) then begin
+            if (script_action == look_at_proc) then begin
                 call look_at_p_proc();
             end
             else begin
-                if (script_action == 15) then begin
+                if (script_action == map_enter_proc) then begin
                     call map_enter_p_proc();
                 end
                 else begin
-                    if (script_action == 4) then begin
+                    if (script_action == pickup_proc) then begin
                         call pickup_p_proc();
                     end
                     else begin
-                        if (script_action == 11) then begin
+                        if (script_action == talk_proc) then begin
                             call talk_p_proc();
                         end
                         else begin
-                            if (script_action == 22) then begin
+                            if (script_action == timed_event_proc) then begin
                                 call timed_event_p_proc();
                             end
                         end
@@ -86,14 +86,14 @@ procedure critter_p_proc
 begin
     if (hostile) then begin
         hostile := 0;
-        attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+        attack(dude_obj);
     end
     else begin
         if (obj_can_see_obj(self_obj, dude_obj)) then begin
             if (obj_is_carrying_obj_pid(dude_obj, 126) > 3) then begin
                 if (local_var(4) == 0) then begin
                     set_local_var(4, 1);
-                    float_msg(self_obj, message_str(163, 116), 0);
+                    float_msg(self_obj, mstr(116), FLOAT_MSG_NORMAL);
                 end
             end
         end
@@ -144,7 +144,7 @@ begin
         end
         if (getting_ration) then begin
             use_obj(recipient);
-            float_msg(self_obj, message_str(163, random(113, 114)), 3);
+            float_msg(self_obj, mstr(random(113, 114)), FLOAT_MSG_GREEN);
             getting_ration := 0;
         end
         if ((tile_num(self_obj) == 16912) and going_down) then begin
@@ -177,20 +177,20 @@ end
 procedure destroy_p_proc
 begin
     if (source_obj == dude_obj) then begin
-        set_global_var(261, 1);
+        set_global_var(GVAR_WATER_CHIP_4, 1);
     end
     if (source_obj == dude_obj) then begin
-        if (((global_var(160) + global_var(159)) >= 25) and ((global_var(159) > (2 * global_var(160))) or (global_var(156) == 1))) then begin
-            set_global_var(156, 1);
-            set_global_var(157, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
+            set_global_var(GVAR_BERSERKER_REPUTATION, 1);
+            set_global_var(GVAR_CHAMPION_REPUTATION, 0);
         end
-        if (((global_var(160) + global_var(159)) >= 25) and ((global_var(160) > (3 * global_var(159))) or (global_var(157) == 1))) then begin
-            set_global_var(157, 1);
-            set_global_var(156, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(160) > (3 * global_var(GVAR_GOOD_MONSTER))) or (global_var(GVAR_CHAMPION_REPUTATION) == 1))) then begin
+            set_global_var(GVAR_CHAMPION_REPUTATION, 1);
+            set_global_var(GVAR_BERSERKER_REPUTATION, 0);
         end
-        set_global_var(159, global_var(159) + 1);
-        if ((global_var(159) % 2) == 0) then begin
-            set_global_var(155, global_var(155) - 1);
+        set_global_var(GVAR_GOOD_MONSTER, global_var(GVAR_GOOD_MONSTER) + 1);
+        if ((global_var(GVAR_GOOD_MONSTER) % 2) == 0) then begin
+            set_global_var(GVAR_PLAYER_REPUATION, global_var(GVAR_PLAYER_REPUATION) - 1);
         end
     end
 end
@@ -198,7 +198,7 @@ end
 procedure look_at_p_proc
 begin
     script_overrides;
-    display_msg(message_str(163, 100));
+    display_mstr(100);
 end
 
 procedure map_enter_p_proc
@@ -217,22 +217,22 @@ end
 
 procedure pickup_p_proc
 begin
-    float_msg(self_obj, message_str(163, 115), 0);
+    float_msg(self_obj, mstr(115), FLOAT_MSG_NORMAL);
     hostile := 1;
 end
 
 procedure talk_p_proc
 begin
-    if (global_var(261) == 1) then begin
-        float_msg(self_obj, message_str(185, 102), 2);
+    if (global_var(GVAR_WATER_CHIP_4) == 1) then begin
+        float_msg(self_obj, message_str(SCRIPT_GENVAULT, 102), FLOAT_MSG_RED);
     end
     else begin
-        if (global_var(101) == 2) then begin
-            float_msg(self_obj, message_str(185, 109), 0);
+        if (global_var(GVAR_FIND_WATER_CHIP) == 2) then begin
+            float_msg(self_obj, message_str(SCRIPT_GENVAULT, 109), FLOAT_MSG_NORMAL);
         end
         else begin
-            if (global_var(188) == 2) then begin
-                float_msg(self_obj, message_str(507, 113), 0);
+            if (global_var(GVAR_WATER_THIEF) == 2) then begin
+                float_msg(self_obj, message_str(SCRIPT_LYLE, 113), FLOAT_MSG_NORMAL);
             end
             else begin
                 start_gdialog(163, self_obj, 4, -1, -1);
@@ -303,63 +303,63 @@ end
 
 procedure WtrGrd01
 begin
-    gsay_reply(163, message_str(163, 101) + proto_data(obj_pid(dude_obj), 1) + message_str(163, 102));
-    giq_option(4, 163, 103, WtrGrd03, 50);
-    giq_option(4, 163, 127, WtrGrd09, 50);
-    giq_option(-3, 163, 104, WtrGrd02, 50);
+    Reply(mstr(101) + proto_data(obj_pid(dude_obj), 1) + mstr(102));
+    giq_option(4, NAME, 103, WtrGrd03, NEUTRAL_REACTION);
+    giq_option(4, NAME, 127, WtrGrd09, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 104, WtrGrd02, NEUTRAL_REACTION);
 end
 
 procedure WtrGrd02
 begin
-    gsay_reply(163, 105);
-    giq_option(-3, 163, 106, WtrGrd03, 50);
+    Reply(105);
+    giq_option(-3, NAME, 106, WtrGrd03, NEUTRAL_REACTION);
 end
 
 procedure WtrGrd03
 begin
-    gsay_reply(163, 107);
-    giq_option(4, 163, 108, WtrGrd04, 50);
-    giq_option(4, 163, 127, WtrGrd09, 50);
-    giq_option(-3, 163, 109, WtrGrdEnd, 50);
+    Reply(107);
+    giq_option(4, NAME, 108, WtrGrd04, NEUTRAL_REACTION);
+    giq_option(4, NAME, 127, WtrGrd09, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 109, WtrGrdEnd, NEUTRAL_REACTION);
 end
 
 procedure WtrGrd04
 begin
-    gsay_reply(163, 110);
-    giq_option(4, 163, 111, WtrGrdQuest, 50);
-    giq_option(4, 163, 112, WtrGrdEnd, 50);
+    Reply(110);
+    giq_option(4, NAME, 111, WtrGrdQuest, NEUTRAL_REACTION);
+    giq_option(4, NAME, 112, WtrGrdEnd, NEUTRAL_REACTION);
 end
 
 procedure WtrGrd05
 begin
-    gsay_reply(163, message_str(163, 118) + proto_data(obj_pid(dude_obj), 1) + message_str(163, 119));
-    giq_option(4, 163, 120, WtrGrd06, 50);
-    giq_option(6, 163, 121, WtrGRd07, 50);
-    giq_option(4, 163, 127, WtrGrd09, 50);
-    giq_option(-3, 163, 104, WtrGrd02, 50);
+    Reply(mstr(118) + proto_data(obj_pid(dude_obj), 1) + mstr(119));
+    giq_option(4, NAME, 120, WtrGrd06, NEUTRAL_REACTION);
+    giq_option(6, NAME, 121, WtrGRd07, NEUTRAL_REACTION);
+    giq_option(4, NAME, 127, WtrGrd09, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 104, WtrGrd02, NEUTRAL_REACTION);
 end
 
 procedure WtrGrd06
 begin
-    gsay_reply(163, 122);
-    giq_option(4, 163, 121, WtrGRd07, 50);
-    giq_option(4, 634, 105, WtrGrdEnd, 50);
+    Reply(122);
+    giq_option(4, NAME, 121, WtrGRd07, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_MODREACT, 105, WtrGrdEnd, NEUTRAL_REACTION);
 end
 
 procedure WtrGRd07
 begin
-    gsay_reply(163, 123);
-    giq_option(4, 163, 124, WtrGRd08, 50);
+    Reply(123);
+    giq_option(4, NAME, 124, WtrGRd08, NEUTRAL_REACTION);
 end
 
 procedure WtrGRd08
 begin
-    gsay_message(163, 125, 50);
+    gsay_message(NAME, 125, NEUTRAL_REACTION);
 end
 
 procedure WtrGrd09
 begin
-    gsay_message(163, 126, 50);
+    gsay_message(NAME, 126, NEUTRAL_REACTION);
 end
 
 procedure WtrGrdEnd
@@ -368,8 +368,8 @@ end
 
 procedure WtrGrdQuest
 begin
-    if (not(global_var(188))) then begin
-        set_global_var(188, 1);
+    if (not(global_var(GVAR_WATER_THIEF))) then begin
+        set_global_var(GVAR_WATER_THIEF, 1);
     end
 end
 
@@ -379,31 +379,31 @@ begin
         set_local_var(0, 50);
         set_local_var(1, 2);
         set_local_var(2, 1);
-        set_local_var(0, local_var(0) + (5 * get_critter_stat(dude_obj, 3)) - 25);
-        set_local_var(0, local_var(0) + (10 * has_trait(0, dude_obj, 10)));
-        if (has_trait(0, dude_obj, 39)) then begin
-            if (global_var(155) > 0) then begin
-                set_local_var(0, local_var(0) + global_var(155));
+        set_local_var(0, local_var(0) + (5 * dude_charisma) - 25);
+        set_local_var(0, local_var(0) + (10 * has_trait(TRAIT_PERK, dude_obj, PERK_presence)));
+        if (has_trait(TRAIT_PERK, dude_obj, PERK_cult_of_personality)) then begin
+            if (global_var(GVAR_PLAYER_REPUATION) > 0) then begin
+                set_local_var(0, local_var(0) + global_var(GVAR_PLAYER_REPUATION));
             end
             else begin
-                set_local_var(0, local_var(0) - global_var(155));
+                set_local_var(0, local_var(0) - global_var(GVAR_PLAYER_REPUATION));
             end
         end
         else begin
             if (local_var(3) == 1) then begin
-                set_local_var(0, local_var(0) - global_var(155));
+                set_local_var(0, local_var(0) - global_var(GVAR_PLAYER_REPUATION));
             end
             else begin
-                set_local_var(0, local_var(0) + global_var(155));
+                set_local_var(0, local_var(0) + global_var(GVAR_PLAYER_REPUATION));
             end
         end
-        if (global_var(158) > 2) then begin
+        if (global_var(GVAR_CHILDKILLER_REPUATION) > 2) then begin
             set_local_var(0, local_var(0) - 30);
         end
-        if (((global_var(160) + global_var(159)) >= 25) and ((global_var(160) > (3 * global_var(159))) or (global_var(157) == 1))) then begin
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(160) > (3 * global_var(GVAR_GOOD_MONSTER))) or (global_var(GVAR_CHAMPION_REPUTATION) == 1))) then begin
             set_local_var(0, local_var(0) + 20);
         end
-        if (((global_var(160) + global_var(159)) >= 25) and ((global_var(159) > (2 * global_var(160))) or (global_var(156) == 1))) then begin
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
             set_local_var(0, local_var(0) - 20);
         end
         call ReactToLevel();
@@ -496,6 +496,5 @@ end
 
 procedure Goodbyes
 begin
-    exit_line := message_str(634, random(100, 105));
+    exit_line := message_str(SCRIPT_MODREACT, random(100, 105));
 end
-
