diff --git a/Mods/Resurrection/Original/AALEEA.ssl b/Mods/Resurrection/ReDefine/AALEEA.ssl
index 1641e3c..f9e13cc 100644
--- a/Mods/Resurrection/Original/AALEEA.ssl
+++ b/Mods/Resurrection/ReDefine/AALEEA.ssl
@@ -77,12 +77,12 @@ begin
             return 1;
         end
     end
-    if ((party_member_obj(16777779) != 0) and ((global_var(910) bwand 32) == 0)) then begin
+    if ((party_member_obj(16777779) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 32) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777779)) <= 5) then begin
             return 1;
         end
     end
-    if ((party_member_obj(16777811) != 0) and ((global_var(910) bwand 64) == 0)) then begin
+    if ((party_member_obj(16777811) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 64) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777811)) <= 5) then begin
             return 1;
         end
@@ -97,7 +97,7 @@ begin
             return 1;
         end
     end
-    if ((party_member_obj(16777814) != 0) and ((global_var(910) bwand 128) == 0)) then begin
+    if ((party_member_obj(16777814) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 128) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777814)) <= 5) then begin
             return 1;
         end
@@ -154,15 +154,15 @@ end
 
 procedure critter_p_proc
 begin
-    if ((local_var(4) == 1) or (global_var(862) == 1) and obj_can_see_obj(self_obj, dude_obj) and (local_var(7) == 0)) then begin
+    if ((local_var(4) == 1) or (global_var(GVAR_AQ_ENEMY_PRIMITIVES) == 1) and obj_can_see_obj(self_obj, dude_obj) and (local_var(7) == 0)) then begin
         if (critter_is_fleeing(self_obj)) then begin
             if (anim_busy(self_obj) == 0) then begin
                 if ((tile_distance_objs(self_obj, dude_obj) < 8) or obj_can_see_obj(self_obj, dude_obj)) then begin
                     if (anim_busy(self_obj) == 0) then begin
-                        global_temp := rotation_to_tile(tile_num(dude_obj), tile_num(self_obj));
+                        global_temp := rotation_to_tile(dude_tile, tile_num(self_obj));
                         animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(self_obj), global_temp, random(3, 10)), 1);
                         global_temp := (global_temp + 1) % 6;
-                        while ((anim_busy(self_obj) == 0) and (global_temp != rotation_to_tile(tile_num(dude_obj), tile_num(self_obj)))) do begin
+                        while ((anim_busy(self_obj) == 0) and (global_temp != rotation_to_tile(dude_tile, tile_num(self_obj)))) do begin
                             debug_msg("flee loop: rot == " + global_temp);
                             animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(self_obj), global_temp, random(3, 10)), 1);
                             global_temp := (global_temp + 1) % 6;
@@ -172,7 +172,7 @@ begin
             end
         end
         else begin
-            attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+            attack(dude_obj);
         end
     end
     if ((map_var(0) bwand 512) != 0) then begin
@@ -225,10 +225,10 @@ begin
                 set_local_var(5, local_var(5) bwor 4096);
             end
             if ((local_var(5) bwand 1024) != 0) then begin
-                set_global_var(865, global_var(865) bwor 2048);
+                set_global_var(GVAR_AQ_ALBUQ_EVENTS, global_var(GVAR_AQ_ALBUQ_EVENTS) bwor 2048);
             end
             if ((local_var(5) bwand 128) != 0) then begin
-                set_global_var(865, global_var(865) bwor 128);
+                set_global_var(GVAR_AQ_ALBUQ_EVENTS, global_var(GVAR_AQ_ALBUQ_EVENTS) bwor 128);
             end
             destroy_object(self_obj);
         end
@@ -237,17 +237,17 @@ begin
                 set_local_var(5, local_var(5) bwor 4096);
             end
             if ((local_var(5) bwand 1024) != 0) then begin
-                set_global_var(865, global_var(865) bwor 2048);
+                set_global_var(GVAR_AQ_ALBUQ_EVENTS, global_var(GVAR_AQ_ALBUQ_EVENTS) bwor 2048);
             end
             if ((local_var(5) bwand 128) != 0) then begin
-                set_global_var(865, global_var(865) bwor 128);
+                set_global_var(GVAR_AQ_ALBUQ_EVENTS, global_var(GVAR_AQ_ALBUQ_EVENTS) bwor 128);
             end
             destroy_object(self_obj);
         end
     end
     if (local_var(7) == 1) then begin
-        critter_add_trait(self_obj, 1, 6, 251);
-        critter_add_trait(self_obj, 1, 5, 180);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 251);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_PEASANT_NO_MOVE);
         set_local_var(7, 2);
         if ((art_anim(obj_art_fid(self_obj)) == 20) or (art_anim(obj_art_fid(self_obj)) == 21) or (art_anim(obj_art_fid(self_obj)) == 48) or (art_anim(obj_art_fid(self_obj)) == 49)) then begin
         end
@@ -281,76 +281,76 @@ begin
         end
         else begin
             if (((local_var(5) bwand 2) != 0) and (projed_smycku == 0)) then begin
-                if (global_var(886) == 0) then begin
-                    if (get_critter_stat(dude_obj, 34) == 0) then begin
-                        float_msg(self_obj, message_str(1609, 300), 3);
+                if (global_var(GVAR_AQ_FRIEND_OF_THE_TRIBE) == 0) then begin
+                    if (dude_gender == 0) then begin
+                        float_msg(self_obj, mstr(300), FLOAT_MSG_GREEN);
                     end
-                    if (get_critter_stat(dude_obj, 34) == 1) then begin
-                        float_msg(self_obj, message_str(1609, 301), 3);
+                    if (dude_gender == 1) then begin
+                        float_msg(self_obj, mstr(301), FLOAT_MSG_GREEN);
                     end
                 end
                 else begin
-                    if (get_critter_stat(dude_obj, 34) == 0) then begin
-                        float_msg(self_obj, message_str(1609, 310), 3);
+                    if (dude_gender == 0) then begin
+                        float_msg(self_obj, mstr(310), FLOAT_MSG_GREEN);
                     end
-                    if (get_critter_stat(dude_obj, 34) == 1) then begin
-                        float_msg(self_obj, message_str(1609, 311), 3);
+                    if (dude_gender == 1) then begin
+                        float_msg(self_obj, mstr(311), FLOAT_MSG_GREEN);
                     end
                 end
                 set_local_var(5, local_var(5) bwor 8192);
                 projed_smycku := 1;
             end
-            if (((global_var(865) bwand 64) != 0) and ((local_var(5) bwand 1024) != 0) and (which_float == 306) and (projed_smycku == 0)) then begin
-                float_msg(self_obj, obj_name(dude_obj) + message_str(1609, 306), 3);
+            if (((global_var(GVAR_AQ_ALBUQ_EVENTS) bwand 64) != 0) and ((local_var(5) bwand 1024) != 0) and (which_float == 306) and (projed_smycku == 0)) then begin
+                float_msg(self_obj, dude_name + mstr(306), FLOAT_MSG_GREEN);
                 set_local_var(5, local_var(5) bwor 8192);
                 projed_smycku := 1;
             end
-            if (((global_var(865) bwand 64) != 0) and ((local_var(5) bwand 2048) != 0) and (which_float == 307) and (projed_smycku == 0)) then begin
-                float_msg(self_obj, obj_name(dude_obj) + message_str(1609, 307), 3);
+            if (((global_var(GVAR_AQ_ALBUQ_EVENTS) bwand 64) != 0) and ((local_var(5) bwand 2048) != 0) and (which_float == 307) and (projed_smycku == 0)) then begin
+                float_msg(self_obj, dude_name + mstr(307), FLOAT_MSG_GREEN);
                 set_local_var(5, local_var(5) bwor 8192);
                 projed_smycku := 1;
             end
-            if (((global_var(865) bwand 4096) != 0) and not((global_var(865) bwand 64) != 0) and (projed_smycku == 0)) then begin
-                if (get_critter_stat(dude_obj, 34) == 0) then begin
-                    float_msg(self_obj, message_str(1609, 312), 2);
+            if (((global_var(GVAR_AQ_ALBUQ_EVENTS) bwand 4096) != 0) and not((global_var(GVAR_AQ_ALBUQ_EVENTS) bwand 64) != 0) and (projed_smycku == 0)) then begin
+                if (dude_gender == 0) then begin
+                    float_msg(self_obj, mstr(312), FLOAT_MSG_RED);
                 end
-                if (get_critter_stat(dude_obj, 34) == 1) then begin
-                    float_msg(self_obj, message_str(1609, 313), 2);
+                if (dude_gender == 1) then begin
+                    float_msg(self_obj, mstr(313), FLOAT_MSG_RED);
                 end
                 projed_smycku := 1;
             end
             if (((local_var(5) bwand 512) != 0) and (projed_smycku == 0)) then begin
-                float_msg(self_obj, message_str(1609, 302), 8);
+                float_msg(self_obj, mstr(302), FLOAT_MSG_YELLOW);
                 projed_smycku := 1;
             end
             if (((local_var(5) bwand 4) != 0) and ((local_var(5) bwand 8) == 0) and (projed_smycku == 0)) then begin
-                float_msg(self_obj, obj_name(dude_obj) + message_str(1609, 303), 2);
+                float_msg(self_obj, dude_name + mstr(303), FLOAT_MSG_RED);
                 set_local_var(5, local_var(5) bwor 8);
                 projed_smycku := 1;
             end
             if (((local_var(5) bwand 4) != 0) and not((local_var(5) bwand 8) == 0) and (projed_smycku == 0)) then begin
-                float_msg(self_obj, message_str(1609, 304) + obj_name(dude_obj) + message_str(1609, 305), 2);
+                float_msg(self_obj, mstr(304) + dude_name + mstr(305), FLOAT_MSG_RED);
                 projed_smycku := 1;
             end
-            if (((global_var(865) bwand 64) != 0) and not((local_var(5) bwand 16) == 0) and (which_float == 308) and (projed_smycku == 0)) then begin
-                float_msg(self_obj, obj_name(dude_obj) + message_str(1609, 308), 3);
+            if (((global_var(GVAR_AQ_ALBUQ_EVENTS) bwand 64) != 0) and not((local_var(5) bwand 16) == 0) and (which_float == 308) and (projed_smycku == 0)) then begin
+                float_msg(self_obj, dude_name + mstr(308), FLOAT_MSG_GREEN);
                 projed_smycku := 1;
             end
             which_float := 0;
             if (projed_smycku == 0) then begin
-                anim(dude_obj, 1000, rotation_to_tile(tile_num(dude_obj), tile_num(self_obj)));
-                anim(self_obj, 1000, rotation_to_tile(tile_num(self_obj), tile_num(dude_obj)));
+                anim(dude_obj, 1000, rotation_to_tile(dude_tile, tile_num(self_obj)));
+                anim(self_obj, 1000, rotation_to_tile(tile_num(self_obj), dude_tile));
                 start_gdialog(1609, self_obj, 4, -1, -1);
                 gsay_start;
-                if (((global_var(865) bwand 64) != 0) and ((game_time - local_var(6)) > (24 * (60 * (60 * 10)) * 2)) and ((game_time_hour > 2100) or (game_time_hour < 400)) and not((local_var(5) bwand 16) == 0) and (projed_smycku == 0)) then begin
+                if (((global_var(GVAR_AQ_ALBUQ_EVENTS) bwand 64) != 0) and ((game_time - local_var(6)) > (24 * (60 * (60 * 10)) * 2)) and ((game_time_hour > 2100) or (game_time_hour < 400)) and not((local_var(5) bwand 16) == 0) and (projed_smycku == 0)) then begin
                     call Node1300();
                     projed_smycku := 1;
                 end
-                if (((local_var(5) bwand 16) == 0) and not((local_var(5) bwand 1) == 0) and not((global_var(865) bwand 64) != 0) and (projed_smycku == 0)) then begin
+                if (((local_var(5) bwand 16) == 0) and not((local_var(5) bwand 1) == 0) and not((global_var(GVAR_AQ_ALBUQ_EVENTS) bwand 64) != 0) and (projed_smycku == 0)) then begin
                     call Node1000();
                     projed_smycku := 1;
                 end
-                if (((global_var(865) bwand 64) != 0) and not((local_var(5) bwand 1) == 0) and (projed_smycku == 0)) then begin
+                if (((global_var(GVAR_AQ_ALBUQ_EVENTS) bwand 64) != 0) and not((local_var(5) bwand 1) == 0) and (projed_smycku == 0)) then begin
                     call Node1150();
                     projed_smycku := 1;
                 end
@@ -362,7 +362,7 @@ begin
                 end_dialogue;
             end
             projed_smycku := 0;
-            if (((global_var(865) bwand 64) != 0) and (cur_map_index == 21)) then begin
+            if (((global_var(GVAR_AQ_ALBUQ_EVENTS) bwand 64) != 0) and (CUR_MAP_ALB_PRED)) then begin
                 set_local_var(6, game_time);
             end
             if ((local_var(5) bwand 32) != 0) then begin
@@ -376,59 +376,59 @@ end
 procedure destroy_p_proc
 begin
     if (source_obj == dude_obj) then begin
-        set_global_var(5, global_var(5) + 1);
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(4) > (3 * global_var(5))) or (global_var(2) == 1)) and (global_var(1) == 0)) then begin
-            set_global_var(2, 1);
-            set_global_var(3, 0);
+        set_global_var(GVAR_GOOD_MONSTER, global_var(GVAR_GOOD_MONSTER) + 1);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(4) > (3 * global_var(GVAR_GOOD_MONSTER))) or (global_var(GVAR_CHAMPION_REPUTATION) == 1)) and (global_var(GVAR_CHILDKILLER_REPUTATION) == 0)) then begin
+            set_global_var(GVAR_CHAMPION_REPUTATION, 1);
+            set_global_var(GVAR_BERSERKER_REPUTATION, 0);
         end
-        if (metarule(51, self_obj) == 2) then begin
+        if (critter_kill_type(self_obj) == KILL_TYPE_children_kills) then begin
             if (source_obj == dude_obj) then begin
-                set_global_var(2, 0);
-                set_global_var(1, global_var(1) + 1);
-                set_global_var(0, global_var(0) + -15);
+                set_global_var(GVAR_CHAMPION_REPUTATION, 0);
+                set_global_var(GVAR_CHILDKILLER_REPUTATION, global_var(GVAR_CHILDKILLER_REPUTATION) + 1);
+                set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + -15);
                 debug_msg("Player gains " + -15 + " Karma Points.");
-                set_global_var(37, 0);
-                set_global_var(38, 0);
-                set_global_var(39, 0);
-                set_global_var(40, 0);
-                set_global_var(41, 0);
-                set_global_var(42, 0);
-                set_global_var(43, 0);
-                set_global_var(44, 0);
-                set_global_var(45, 0);
-                if (global_var(0) >= 1000) then begin
-                    set_global_var(37, 1);
+                set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+                set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+                set_global_var(GVAR_KARMA_DEFENDER, 0);
+                set_global_var(GVAR_KARMA_WANDERER, 0);
+                set_global_var(GVAR_KARMA_BETRAYER, 0);
+                set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+                set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+                set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+                    set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
                 end
                 else begin
-                    if (global_var(0) >= 750) then begin
-                        set_global_var(38, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
                     end
                     else begin
-                        if (global_var(0) >= 500) then begin
-                            set_global_var(39, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                            set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                         end
                         else begin
-                            if (global_var(0) >= 250) then begin
-                                set_global_var(40, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                                set_global_var(GVAR_KARMA_DEFENDER, 1);
                             end
                             else begin
-                                if (global_var(0) > -250) then begin
-                                    set_global_var(41, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                                    set_global_var(GVAR_KARMA_WANDERER, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -500) then begin
-                                        set_global_var(42, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                        set_global_var(GVAR_KARMA_BETRAYER, 1);
                                     end
                                     else begin
-                                        if (global_var(0) > -750) then begin
-                                            set_global_var(43, 1);
+                                        if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                            set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                         end
                                         else begin
-                                            if (global_var(0) > -1000) then begin
-                                                set_global_var(44, 1);
+                                            if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                                set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                             end
                                             else begin
-                                                set_global_var(45, 1);
+                                                set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                             end
                                         end
                                     end
@@ -437,58 +437,58 @@ begin
                         end
                     end
                 end
-                set_global_var(51, global_var(51) + -8);
+                set_global_var(GVAR_TOWN_REP_ALBUQUERQUE, global_var(GVAR_TOWN_REP_ALBUQUERQUE) + -8);
                 debug_msg("Added " + -8 + " to Town Rep");
             end
         end
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(5) > (2 * global_var(4))) or (global_var(3) == 1))) then begin
-            set_global_var(3, 1);
-            set_global_var(2, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
+            set_global_var(GVAR_BERSERKER_REPUTATION, 1);
+            set_global_var(GVAR_CHAMPION_REPUTATION, 0);
         end
-        set_global_var(0, global_var(0) + -10);
+        set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + -10);
         debug_msg("Player gains " + -10 + " Karma Points.");
-        set_global_var(37, 0);
-        set_global_var(38, 0);
-        set_global_var(39, 0);
-        set_global_var(40, 0);
-        set_global_var(41, 0);
-        set_global_var(42, 0);
-        set_global_var(43, 0);
-        set_global_var(44, 0);
-        set_global_var(45, 0);
-        if (global_var(0) >= 1000) then begin
-            set_global_var(37, 1);
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+        set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+        set_global_var(GVAR_KARMA_DEFENDER, 0);
+        set_global_var(GVAR_KARMA_WANDERER, 0);
+        set_global_var(GVAR_KARMA_BETRAYER, 0);
+        set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+        set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+            set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
         end
         else begin
-            if (global_var(0) >= 750) then begin
-                set_global_var(38, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
             end
             else begin
-                if (global_var(0) >= 500) then begin
-                    set_global_var(39, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                 end
                 else begin
-                    if (global_var(0) >= 250) then begin
-                        set_global_var(40, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                        set_global_var(GVAR_KARMA_DEFENDER, 1);
                     end
                     else begin
-                        if (global_var(0) > -250) then begin
-                            set_global_var(41, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                            set_global_var(GVAR_KARMA_WANDERER, 1);
                         end
                         else begin
-                            if (global_var(0) > -500) then begin
-                                set_global_var(42, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                set_global_var(GVAR_KARMA_BETRAYER, 1);
                             end
                             else begin
-                                if (global_var(0) > -750) then begin
-                                    set_global_var(43, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -1000) then begin
-                                        set_global_var(44, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                     end
                                     else begin
-                                        set_global_var(45, 1);
+                                        set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                     end
                                 end
                             end
@@ -497,28 +497,28 @@ begin
                 end
             end
         end
-        set_global_var(51, global_var(51) + -5);
+        set_global_var(GVAR_TOWN_REP_ALBUQUERQUE, global_var(GVAR_TOWN_REP_ALBUQUERQUE) + -5);
         debug_msg("Added " + -5 + " to Town Rep");
     end
-    if (((global_var(865) bwand 128) != 0) or ((global_var(865) bwand 512) != 0) or ((global_var(865) bwand 2048) != 0) or (global_var(873) >= 2)) then begin
+    if (((global_var(GVAR_AQ_ALBUQ_EVENTS) bwand 128) != 0) or ((global_var(GVAR_AQ_ALBUQ_EVENTS) bwand 512) != 0) or ((global_var(GVAR_AQ_ALBUQ_EVENTS) bwand 2048) != 0) or (global_var(GVAR_AQ_QUEST_BEN_ALEEYA) >= 2)) then begin
         if (((local_var(5) bwand 64) != 0) and not((local_var(5) bwand 4096) != 0)) then begin
-            set_global_var(867, global_var(867) bwor 64);
+            set_global_var(GVAR_AQ_ALBUQ_DEATHS, global_var(GVAR_AQ_ALBUQ_DEATHS) bwor 64);
         end
     end
     else begin
-        set_global_var(867, global_var(867) bwor 64);
+        set_global_var(GVAR_AQ_ALBUQ_DEATHS, global_var(GVAR_AQ_ALBUQ_DEATHS) bwor 64);
     end
-    if (global_var(878) == 2) then begin
-        set_global_var(890, global_var(890) + 1);
+    if (global_var(GVAR_AQ_QUEST_HOSPITAL_TO_KILL_PRIMITIVES) == 2) then begin
+        set_global_var(GVAR_AQ_PRIMITIVES_COUNTER, global_var(GVAR_AQ_PRIMITIVES_COUNTER) + 1);
     end
-    if ((global_var(878) == 2) and (local_var(7) == 0)) then begin
-        set_global_var(899, global_var(899) + 1);
+    if ((global_var(GVAR_AQ_QUEST_HOSPITAL_TO_KILL_PRIMITIVES) == 2) and (local_var(7) == 0)) then begin
+        set_global_var(GVAR_AQ_HOSPITAL_GIVERS_STATUS, global_var(GVAR_AQ_HOSPITAL_GIVERS_STATUS) + 1);
     end
-    if ((global_var(878) == 2) and (global_var(889) <= global_var(899))) then begin
-        set_global_var(878, 3);
+    if ((global_var(GVAR_AQ_QUEST_HOSPITAL_TO_KILL_PRIMITIVES) == 2) and (global_var(GVAR_AQ_HUNTINGS_RESULTS) <= global_var(GVAR_AQ_HOSPITAL_GIVERS_STATUS))) then begin
+        set_global_var(GVAR_AQ_QUEST_HOSPITAL_TO_KILL_PRIMITIVES, 3);
     end
-    if ((global_var(878) == 2) and (global_var(862) == 0)) then begin
-        set_global_var(862, 1);
+    if ((global_var(GVAR_AQ_QUEST_HOSPITAL_TO_KILL_PRIMITIVES) == 2) and (global_var(GVAR_AQ_ENEMY_PRIMITIVES) == 0)) then begin
+        set_global_var(GVAR_AQ_ENEMY_PRIMITIVES, 1);
     end
 end
 
@@ -526,10 +526,10 @@ procedure look_at_p_proc
 begin
     script_overrides;
     if (not((local_var(5) bwand 1) == 0)) then begin
-        display_msg(message_str(1609, 150));
+        display_mstr(150);
     end
     else begin
-        display_msg(message_str(1609, 100));
+        display_mstr(100);
     end
 end
 
@@ -538,18 +538,18 @@ begin
     script_overrides;
     if (not((local_var(5) bwand 1) == 0)) then begin
         if (local_var(7) == 0) then begin
-            display_msg(message_str(1609, 250));
+            display_mstr(250);
         end
         else begin
-            display_msg(message_str(1609, 250) + message_str(1609, 201));
+            display_msg(mstr(250) + mstr(201));
         end
     end
     else begin
         if (local_var(7) == 0) then begin
-            display_msg(message_str(1609, 200));
+            display_mstr(200);
         end
         else begin
-            display_msg(message_str(1609, 200) + message_str(1609, 201));
+            display_msg(mstr(200) + mstr(201));
         end
     end
 end
@@ -561,38 +561,38 @@ end
 procedure damage_p_proc
 begin
     if (source_obj == dude_obj) then begin
-        set_global_var(862, 1);
+        set_global_var(GVAR_AQ_ENEMY_PRIMITIVES, 1);
     end
-    if (global_var(878) == 2) then begin
-        if ((obj_pid(critter_inven_obj(source_obj, 1)) == 601) or (obj_pid(critter_inven_obj(source_obj, 2)) == 601) and (local_var(7) != 2)) then begin
-            if (get_critter_stat(self_obj, 35) <= (get_critter_stat(self_obj, 7) / 3)) then begin
+    if (global_var(GVAR_AQ_QUEST_HOSPITAL_TO_KILL_PRIMITIVES) == 2) then begin
+        if ((obj_pid(critter_inven_obj(source_obj, INVEN_TYPE_RIGHT_HAND)) == 601) or (obj_pid(critter_inven_obj(source_obj, INVEN_TYPE_LEFT_HAND)) == 601) and (local_var(7) != 2)) then begin
+            if (get_critter_stat(self_obj, STAT_current_hp) <= (get_critter_stat(self_obj, STAT_max_hp) / 3)) then begin
                 if (local_var(7) == 0) then begin
-                    set_global_var(899, global_var(899) + 1);
+                    set_global_var(GVAR_AQ_HOSPITAL_GIVERS_STATUS, global_var(GVAR_AQ_HOSPITAL_GIVERS_STATUS) + 1);
                 end
-                if ((global_var(878) == 2) and (global_var(889) <= global_var(899)) and (local_var(7) == 0)) then begin
-                    set_global_var(878, 3);
+                if ((global_var(GVAR_AQ_QUEST_HOSPITAL_TO_KILL_PRIMITIVES) == 2) and (global_var(GVAR_AQ_HUNTINGS_RESULTS) <= global_var(GVAR_AQ_HOSPITAL_GIVERS_STATUS)) and (local_var(7) == 0)) then begin
+                    set_global_var(GVAR_AQ_QUEST_HOSPITAL_TO_KILL_PRIMITIVES, 3);
                 end
                 set_local_var(7, 1);
                 script_overrides;
                 critter_injure(self_obj, 2);
                 critter_injure(self_obj, 8);
                 critter_injure(self_obj, 4);
-                critter_add_trait(self_obj, 1, 6, 251);
-                critter_add_trait(self_obj, 1, 5, 180);
+                critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 251);
+                critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_PEASANT_NO_MOVE);
                 set_map_var(4, map_var(4) bwor 2048);
-                display_msg(message_str(1609, 410));
-                set_global_var(51, global_var(51) - 2);
-                if (get_critter_stat(self_obj, 35) > 0) then begin
-                    set_global_var(862, 0);
+                display_mstr(410);
+                set_global_var(GVAR_TOWN_REP_ALBUQUERQUE, global_var(GVAR_TOWN_REP_ALBUQUERQUE) - 2);
+                if (get_critter_stat(self_obj, STAT_current_hp) > 0) then begin
+                    set_global_var(GVAR_AQ_ENEMY_PRIMITIVES, 0);
                 end
-                if (get_critter_stat(self_obj, 35) > 0) then begin
+                if (get_critter_stat(self_obj, STAT_current_hp) > 0) then begin
                     terminate_combat;
                 end
             end
         end
         else begin
             if (source_obj == dude_obj) then begin
-                set_global_var(900, global_var(900) bwor 4);
+                set_global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS, global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS) bwor 4);
             end
         end
     end
@@ -601,23 +601,23 @@ end
 procedure map_enter_p_proc
 begin
     if (local_var(7) > 0) then begin
-        critter_add_trait(self_obj, 1, 6, 251);
-        critter_add_trait(self_obj, 1, 5, 180);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 251);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_PEASANT_NO_MOVE);
     end
     else begin
-        critter_add_trait(self_obj, 1, 6, 256);
-        critter_add_trait(self_obj, 1, 5, 1);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 256);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_BERSERKER_NO_TALKING);
     end
-    if ((global_var(878) == 1) and not(metarule(22, 0))) then begin
-        set_global_var(889, global_var(889) + 1);
+    if ((global_var(GVAR_AQ_QUEST_HOSPITAL_TO_KILL_PRIMITIVES) == 1) and not(is_loading_game)) then begin
+        set_global_var(GVAR_AQ_HUNTINGS_RESULTS, global_var(GVAR_AQ_HUNTINGS_RESULTS) + 1);
     end
-    if ((global_var(878) == 1) and not(metarule(22, 0))) then begin
+    if ((global_var(GVAR_AQ_QUEST_HOSPITAL_TO_KILL_PRIMITIVES) == 1) and not(is_loading_game)) then begin
         set_local_var(4, 0);
     end
-    if ((global_var(865) bwand 8388608) != 0) then begin
+    if ((global_var(GVAR_AQ_ALBUQ_EVENTS) bwand 8388608) != 0) then begin
         destroy_object(self_obj);
     end
-    if (((global_var(867) bwand 64) != 0) or ((global_var(865) bwand 128) != 0) or ((global_var(865) bwand 512) != 0) or ((global_var(865) bwand 2048) != 0) or (global_var(873) >= 2)) then begin
+    if (((global_var(GVAR_AQ_ALBUQ_DEATHS) bwand 64) != 0) or ((global_var(GVAR_AQ_ALBUQ_EVENTS) bwand 128) != 0) or ((global_var(GVAR_AQ_ALBUQ_EVENTS) bwand 512) != 0) or ((global_var(GVAR_AQ_ALBUQ_EVENTS) bwand 2048) != 0) or (global_var(GVAR_AQ_QUEST_BEN_ALEEYA) >= 2)) then begin
         destroy_object(self_obj);
     end
 end
@@ -628,13 +628,13 @@ begin
         probability := random(0, 20);
         if ((probability >= 20) and critter_is_fleeing(self_obj)) then begin
             if ((local_var(4) == 1) and obj_can_see_obj(self_obj, dude_obj)) then begin
-                float_msg(self_obj, message_str(1609, 400), 2);
+                float_msg(self_obj, mstr(400), FLOAT_MSG_RED);
             end
         end
     end
     if ((fixed_param == 4) and (local_var(7) == 2)) then begin
-        critter_add_trait(self_obj, 1, 6, 251);
-        critter_add_trait(self_obj, 1, 5, 180);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 251);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_PEASANT_NO_MOVE);
         anim(self_obj, 48, 0);
     end
 end
@@ -650,7 +650,7 @@ begin
             add_timer_event(self_obj, game_ticks(1), 1);
         end
         if (sequence == 2) then begin
-            if (get_critter_stat(dude_obj, 34) == 0) then begin
+            if (dude_gender == 0) then begin
                 if ((obj_is_carrying_obj_pid(dude_obj, 314) > 0) and (Condom_removed == 0)) then begin
                     removed_qty := obj_is_carrying_obj_pid(dude_obj, 314);
                     if (1 < removed_qty) then begin
@@ -660,17 +660,17 @@ begin
                         restock_obj := obj_carrying_pid_obj(dude_obj, 314);
                         if (obj_type(self_obj) == 1) then begin
                             restock_obj := obj_carrying_pid_obj(dude_obj, 314);
-                            if (critter_inven_obj(self_obj, 0) == restock_obj) then begin
-                                if (obj_item_subtype(critter_inven_obj(dude_obj, 0)) == 0) then begin
-                                    restock_obj := critter_inven_obj(dude_obj, 0);
+                            if (critter_inven_obj(self_obj, INVEN_TYPE_WORN) == restock_obj) then begin
+                                if (obj_item_subtype(critter_inven_obj(dude_obj, INVEN_TYPE_WORN)) == 0) then begin
+                                    restock_obj := critter_inven_obj(dude_obj, INVEN_TYPE_WORN);
                                     debug_msg("armour pid == " + obj_pid(restock_obj));
                                     rm_obj_from_inven(dude_obj, restock_obj);
                                     add_obj_to_inven(dude_obj, restock_obj);
                                 end
                             end
                             else begin
-                                if ((critter_inven_obj(dude_obj, 2) == restock_obj) or (critter_inven_obj(dude_obj, 1) == restock_obj)) then begin
-                                    metarule(43, dude_obj);
+                                if ((critter_inven_obj(dude_obj, INVEN_TYPE_LEFT_HAND) == restock_obj) or (critter_inven_obj(dude_obj, INVEN_TYPE_RIGHT_HAND) == restock_obj)) then begin
+                                    inven_unwield(dude_obj);
                                 end
                             end
                         end
@@ -701,17 +701,17 @@ begin
                         restock_obj := obj_carrying_pid_obj(dude_obj, 315);
                         if (obj_type(self_obj) == 1) then begin
                             restock_obj := obj_carrying_pid_obj(dude_obj, 315);
-                            if (critter_inven_obj(self_obj, 0) == restock_obj) then begin
-                                if (obj_item_subtype(critter_inven_obj(dude_obj, 0)) == 0) then begin
-                                    restock_obj := critter_inven_obj(dude_obj, 0);
+                            if (critter_inven_obj(self_obj, INVEN_TYPE_WORN) == restock_obj) then begin
+                                if (obj_item_subtype(critter_inven_obj(dude_obj, INVEN_TYPE_WORN)) == 0) then begin
+                                    restock_obj := critter_inven_obj(dude_obj, INVEN_TYPE_WORN);
                                     debug_msg("armour pid == " + obj_pid(restock_obj));
                                     rm_obj_from_inven(dude_obj, restock_obj);
                                     add_obj_to_inven(dude_obj, restock_obj);
                                 end
                             end
                             else begin
-                                if ((critter_inven_obj(dude_obj, 2) == restock_obj) or (critter_inven_obj(dude_obj, 1) == restock_obj)) then begin
-                                    metarule(43, dude_obj);
+                                if ((critter_inven_obj(dude_obj, INVEN_TYPE_LEFT_HAND) == restock_obj) or (critter_inven_obj(dude_obj, INVEN_TYPE_RIGHT_HAND) == restock_obj)) then begin
+                                    inven_unwield(dude_obj);
                                 end
                             end
                         end
@@ -742,17 +742,17 @@ begin
                         restock_obj := obj_carrying_pid_obj(dude_obj, 316);
                         if (obj_type(self_obj) == 1) then begin
                             restock_obj := obj_carrying_pid_obj(dude_obj, 316);
-                            if (critter_inven_obj(self_obj, 0) == restock_obj) then begin
-                                if (obj_item_subtype(critter_inven_obj(dude_obj, 0)) == 0) then begin
-                                    restock_obj := critter_inven_obj(dude_obj, 0);
+                            if (critter_inven_obj(self_obj, INVEN_TYPE_WORN) == restock_obj) then begin
+                                if (obj_item_subtype(critter_inven_obj(dude_obj, INVEN_TYPE_WORN)) == 0) then begin
+                                    restock_obj := critter_inven_obj(dude_obj, INVEN_TYPE_WORN);
                                     debug_msg("armour pid == " + obj_pid(restock_obj));
                                     rm_obj_from_inven(dude_obj, restock_obj);
                                     add_obj_to_inven(dude_obj, restock_obj);
                                 end
                             end
                             else begin
-                                if ((critter_inven_obj(dude_obj, 2) == restock_obj) or (critter_inven_obj(dude_obj, 1) == restock_obj)) then begin
-                                    metarule(43, dude_obj);
+                                if ((critter_inven_obj(dude_obj, INVEN_TYPE_LEFT_HAND) == restock_obj) or (critter_inven_obj(dude_obj, INVEN_TYPE_RIGHT_HAND) == restock_obj)) then begin
+                                    inven_unwield(dude_obj);
                                 end
                             end
                         end
@@ -776,20 +776,20 @@ begin
                 end
                 Condom_removed := 0;
             end
-            game_time_advance(get_critter_stat(dude_obj, 2) * random(4, 8) * (60 * 10));
-            set_global_var(194, global_var(194) + (((((get_critter_stat(dude_obj, 3) * 50) + (get_critter_stat(dude_obj, 2) * 25) + (get_critter_stat(dude_obj, 5) * 13) + (get_critter_stat(dude_obj, 0) * 12)) / 100) + (has_trait(0, dude_obj, 94) * 2) + has_trait(2, dude_obj, 13) + (((global_var(194) >= 100) or (global_var(232) > 0)) * 2 * (dude_obj == dude_obj))) * 1));
-            set_global_var(358, global_var(358) + 1);
-            if ((global_var(713) >= 1) and (global_var(714) >= 1) and (global_var(715) >= 1)) then begin
-                if (global_var(358) >= 16) then begin
-                    if (get_critter_stat(dude_obj, 34) == 0) then begin
-                        set_global_var(589, 1);
+            game_time_advance(dude_endurance * random(4, 8) * (60 * 10));
+            set_global_var(GVAR_PLAYER_SEX_LEVEL, global_var(194) + (((((dude_charisma * 50) + (dude_endurance * 25) + (get_critter_stat(dude_obj, STAT_ag) * 13) + (dude_strength * 12)) / 100) + (has_trait(TRAIT_PERK, dude_obj, PERK_kama_sutra_perk) * 2) + has_trait(2, dude_obj, 13) + (((global_var(GVAR_PLAYER_SEX_LEVEL) >= 100) or (global_var(GVAR_NEW_RENO_PORN_STAR) > 0)) * 2 * (dude_obj == dude_obj))) * 1));
+            set_global_var(GVAR_SEX_COUNTER, global_var(GVAR_SEX_COUNTER) + 1);
+            if ((global_var(GVAR_SEX_WITH_WOMAN) >= 1) and (global_var(GVAR_SEX_WITH_MAN) >= 1) and (global_var(GVAR_SEX_WITH_BRAHMIN) >= 1)) then begin
+                if (global_var(GVAR_SEX_COUNTER) >= 16) then begin
+                    if (dude_gender == 0) then begin
+                        set_global_var(GVAR_GIGALO, 1);
                     end
-                    if (get_critter_stat(dude_obj, 34) == 1) then begin
-                        set_global_var(588, 1);
+                    if (dude_gender == 1) then begin
+                        set_global_var(GVAR_SEXPERT, 1);
                     end
                 end
             end
-            set_global_var(713, global_var(713) + 1);
+            set_global_var(GVAR_SEX_WITH_WOMAN, global_var(GVAR_SEX_WITH_WOMAN) + 1);
             add_timer_event(self_obj, game_ticks(1), 1);
         end
         if (sequence == 3) then begin
@@ -817,7 +817,7 @@ begin
     end
     if (fixed_param == 6) then begin
         game_ui_enable;
-        set_global_var(862, 1);
+        set_global_var(GVAR_AQ_ENEMY_PRIMITIVES, 1);
     end
 end
 
@@ -829,55 +829,55 @@ end
 procedure Node999
 begin
     anim(self_obj, 1000, 2);
-    if (((local_var(5) bwand 1024) != 0) and ((party_member_obj(16777814) != 0) and ((global_var(910) bwand 128) == 0))) then begin
+    if (((local_var(5) bwand 1024) != 0) and ((party_member_obj(16777814) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 128) == 0))) then begin
         add_timer_event(party_member_obj(16777814), 10, 23);
     end
 end
 
 procedure Node500
 begin
-    gsay_reply(1609, 500);
-    if (global_var(873) == 1) then begin
-        giq_option(4, 1609, 501, Node550, 50);
+    Reply(500);
+    if (global_var(GVAR_AQ_QUEST_BEN_ALEEYA) == 1) then begin
+        giq_option(4, NAME, 501, Node550, NEUTRAL_REACTION);
     end
-    giq_option(4, 1609, 502, Node999, 50);
-    giq_option(-3, 1609, 503, Node999, 50);
+    giq_option(4, NAME, 502, Node999, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 503, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node550
 begin
-    if (global_var(886) == 0) then begin
-        if (get_critter_stat(dude_obj, 34) == 0) then begin
-            gsay_reply(1609, 550);
+    if (global_var(GVAR_AQ_FRIEND_OF_THE_TRIBE) == 0) then begin
+        if (dude_gender == 0) then begin
+            Reply(550);
         end
-        if (get_critter_stat(dude_obj, 34) == 1) then begin
-            gsay_reply(1609, 551);
+        if (dude_gender == 1) then begin
+            Reply(551);
         end
     end
     else begin
-        if (get_critter_stat(dude_obj, 34) == 0) then begin
-            gsay_reply(1609, 552);
+        if (dude_gender == 0) then begin
+            Reply(552);
         end
-        if (get_critter_stat(dude_obj, 34) == 1) then begin
-            gsay_reply(1609, 553);
+        if (dude_gender == 1) then begin
+            Reply(553);
         end
     end
     set_local_var(5, local_var(5) bwor 1);
     if (obj_is_carrying_obj_pid(dude_obj, 564)) then begin
-        giq_option(4, 1609, 554, Node600, 50);
+        giq_option(4, NAME, 554, Node600, NEUTRAL_REACTION);
     end
-    if ((get_critter_stat(dude_obj, 3) > 6) and not(obj_is_carrying_obj_pid(dude_obj, 564))) then begin
-        giq_option(4, 1609, 555, Node600, 50);
+    if ((dude_charisma > 6) and not(obj_is_carrying_obj_pid(dude_obj, 564))) then begin
+        giq_option(4, NAME, 555, Node600, NEUTRAL_REACTION);
     end
-    if ((get_critter_stat(dude_obj, 3) < 7) and not(obj_is_carrying_obj_pid(dude_obj, 564))) then begin
-        giq_option(4, 1609, 555, Node950, 50);
+    if ((dude_charisma < 7) and not(obj_is_carrying_obj_pid(dude_obj, 564))) then begin
+        giq_option(4, NAME, 555, Node950, NEUTRAL_REACTION);
     end
-    giq_option(4, 1609, 556, Node700, 50);
+    giq_option(4, NAME, 556, Node700, NEUTRAL_REACTION);
 end
 
 procedure Node600
 begin
-    gsay_reply(1609, 600);
+    Reply(600);
     removed_qty := obj_is_carrying_obj_pid(dude_obj, 564);
     if (1 < removed_qty) then begin
         removed_qty := 1;
@@ -886,17 +886,17 @@ begin
         restock_obj := obj_carrying_pid_obj(dude_obj, 564);
         if (obj_type(self_obj) == 1) then begin
             restock_obj := obj_carrying_pid_obj(dude_obj, 564);
-            if (critter_inven_obj(self_obj, 0) == restock_obj) then begin
-                if (obj_item_subtype(critter_inven_obj(dude_obj, 0)) == 0) then begin
-                    restock_obj := critter_inven_obj(dude_obj, 0);
+            if (critter_inven_obj(self_obj, INVEN_TYPE_WORN) == restock_obj) then begin
+                if (obj_item_subtype(critter_inven_obj(dude_obj, INVEN_TYPE_WORN)) == 0) then begin
+                    restock_obj := critter_inven_obj(dude_obj, INVEN_TYPE_WORN);
                     debug_msg("armour pid == " + obj_pid(restock_obj));
                     rm_obj_from_inven(dude_obj, restock_obj);
                     add_obj_to_inven(dude_obj, restock_obj);
                 end
             end
             else begin
-                if ((critter_inven_obj(dude_obj, 2) == restock_obj) or (critter_inven_obj(dude_obj, 1) == restock_obj)) then begin
-                    metarule(43, dude_obj);
+                if ((critter_inven_obj(dude_obj, INVEN_TYPE_LEFT_HAND) == restock_obj) or (critter_inven_obj(dude_obj, INVEN_TYPE_RIGHT_HAND) == restock_obj)) then begin
+                    inven_unwield(dude_obj);
                 end
             end
         end
@@ -916,64 +916,64 @@ begin
     end
     restock_obj := 0;
     restock_amt := 0;
-    giq_option(4, 1609, 601, Node650, 50);
+    giq_option(4, NAME, 601, Node650, NEUTRAL_REACTION);
 end
 
 procedure Node650
 begin
-    gsay_reply(1609, 650);
-    set_global_var(873, 2);
+    Reply(650);
+    set_global_var(GVAR_AQ_QUEST_BEN_ALEEYA, 2);
     set_local_var(5, local_var(5) bwor 2);
-    giq_option(4, 1609, 651, Node999_to_Ben, 50);
+    giq_option(4, NAME, 651, Node999_to_Ben, NEUTRAL_REACTION);
 end
 
 procedure Node700
 begin
-    if (global_var(886) == 0) then begin
-        if (get_critter_stat(dude_obj, 34) == 0) then begin
-            gsay_reply(1609, 700);
+    if (global_var(GVAR_AQ_FRIEND_OF_THE_TRIBE) == 0) then begin
+        if (dude_gender == 0) then begin
+            Reply(700);
         end
-        if (get_critter_stat(dude_obj, 34) == 1) then begin
-            gsay_reply(1609, 701);
+        if (dude_gender == 1) then begin
+            Reply(701);
         end
     end
     else begin
-        if (get_critter_stat(dude_obj, 34) == 0) then begin
-            gsay_reply(1609, 702);
+        if (dude_gender == 0) then begin
+            Reply(702);
         end
-        if (get_critter_stat(dude_obj, 34) == 1) then begin
-            gsay_reply(1609, 703);
+        if (dude_gender == 1) then begin
+            Reply(703);
         end
     end
-    if (get_critter_stat(dude_obj, 34) == 0) then begin
-        giq_option(4, 1609, 704, Node750, 50);
+    if (dude_gender == 0) then begin
+        giq_option(4, NAME, 704, Node750, NEUTRAL_REACTION);
     end
-    giq_option(4, 1609, 705, Node999_float302, 50);
+    giq_option(4, NAME, 705, Node999_float302, NEUTRAL_REACTION);
 end
 
 procedure Node750
 begin
-    gsay_reply(1609, 750);
+    Reply(750);
     if (obj_is_carrying_obj_pid(dude_obj, 564)) then begin
-        if (get_critter_stat(dude_obj, 3) > 5) then begin
-            giq_option(4, 1609, message_str(1609, 751) + obj_name(dude_obj) + message_str(1609, 752), Node800, 50);
+        if (dude_charisma > 5) then begin
+            giq_option(4, NAME, mstr(751) + dude_name + mstr(752), Node800, NEUTRAL_REACTION);
         end
         else begin
-            giq_option(4, 1609, message_str(1609, 751) + obj_name(dude_obj) + message_str(1609, 752), Node900, 50);
+            giq_option(4, NAME, mstr(751) + dude_name + mstr(752), Node900, NEUTRAL_REACTION);
         end
     end
-    if (get_critter_stat(dude_obj, 3) > 8) then begin
-        giq_option(4, 1609, message_str(1609, 753) + obj_name(dude_obj) + message_str(1609, 754), Node850, 50);
+    if (dude_charisma > 8) then begin
+        giq_option(4, NAME, mstr(753) + dude_name + mstr(754), Node850, NEUTRAL_REACTION);
     end
     else begin
-        giq_option(4, 1609, message_str(1609, 753) + obj_name(dude_obj) + message_str(1609, 754), Node900, 50);
+        giq_option(4, NAME, mstr(753) + dude_name + mstr(754), Node900, NEUTRAL_REACTION);
     end
-    giq_option(4, 1609, 755, Node999_float302, 50);
+    giq_option(4, NAME, 755, Node999_float302, NEUTRAL_REACTION);
 end
 
 procedure Node800
 begin
-    gsay_reply(1609, message_str(1609, 800) + obj_name(dude_obj) + message_str(1609, 801));
+    Reply(mstr(800) + dude_name + mstr(801));
     removed_qty := obj_is_carrying_obj_pid(dude_obj, 564);
     if (1 < removed_qty) then begin
         removed_qty := 1;
@@ -982,17 +982,17 @@ begin
         restock_obj := obj_carrying_pid_obj(dude_obj, 564);
         if (obj_type(self_obj) == 1) then begin
             restock_obj := obj_carrying_pid_obj(dude_obj, 564);
-            if (critter_inven_obj(self_obj, 0) == restock_obj) then begin
-                if (obj_item_subtype(critter_inven_obj(dude_obj, 0)) == 0) then begin
-                    restock_obj := critter_inven_obj(dude_obj, 0);
+            if (critter_inven_obj(self_obj, INVEN_TYPE_WORN) == restock_obj) then begin
+                if (obj_item_subtype(critter_inven_obj(dude_obj, INVEN_TYPE_WORN)) == 0) then begin
+                    restock_obj := critter_inven_obj(dude_obj, INVEN_TYPE_WORN);
                     debug_msg("armour pid == " + obj_pid(restock_obj));
                     rm_obj_from_inven(dude_obj, restock_obj);
                     add_obj_to_inven(dude_obj, restock_obj);
                 end
             end
             else begin
-                if ((critter_inven_obj(dude_obj, 2) == restock_obj) or (critter_inven_obj(dude_obj, 1) == restock_obj)) then begin
-                    metarule(43, dude_obj);
+                if ((critter_inven_obj(dude_obj, INVEN_TYPE_LEFT_HAND) == restock_obj) or (critter_inven_obj(dude_obj, INVEN_TYPE_RIGHT_HAND) == restock_obj)) then begin
+                    inven_unwield(dude_obj);
                 end
             end
         end
@@ -1012,49 +1012,49 @@ begin
     end
     restock_obj := 0;
     restock_amt := 0;
-    set_global_var(0, global_var(0) - 5);
-    set_global_var(37, 0);
-    set_global_var(38, 0);
-    set_global_var(39, 0);
-    set_global_var(40, 0);
-    set_global_var(41, 0);
-    set_global_var(42, 0);
-    set_global_var(43, 0);
-    set_global_var(44, 0);
-    set_global_var(45, 0);
-    if (global_var(0) >= 1000) then begin
-        set_global_var(37, 1);
+    set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) - 5);
+    set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+    set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+    set_global_var(GVAR_KARMA_DEFENDER, 0);
+    set_global_var(GVAR_KARMA_WANDERER, 0);
+    set_global_var(GVAR_KARMA_BETRAYER, 0);
+    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+    set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+    set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+    if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
     end
     else begin
-        if (global_var(0) >= 750) then begin
-            set_global_var(38, 1);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+            set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
         end
         else begin
-            if (global_var(0) >= 500) then begin
-                set_global_var(39, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
             end
             else begin
-                if (global_var(0) >= 250) then begin
-                    set_global_var(40, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                    set_global_var(GVAR_KARMA_DEFENDER, 1);
                 end
                 else begin
-                    if (global_var(0) > -250) then begin
-                        set_global_var(41, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                        set_global_var(GVAR_KARMA_WANDERER, 1);
                     end
                     else begin
-                        if (global_var(0) > -500) then begin
-                            set_global_var(42, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                            set_global_var(GVAR_KARMA_BETRAYER, 1);
                         end
                         else begin
-                            if (global_var(0) > -750) then begin
-                                set_global_var(43, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                             end
                             else begin
-                                if (global_var(0) > -1000) then begin
-                                    set_global_var(44, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                    set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                 end
                                 else begin
-                                    set_global_var(45, 1);
+                                    set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                 end
                             end
                         end
@@ -1064,87 +1064,87 @@ begin
         end
     end
     set_map_var(0, map_var(0) bwor 16777216);
-    giq_option(4, 1609, 802, Do_Sex, 50);
+    giq_option(4, NAME, 802, Do_Sex, NEUTRAL_REACTION);
 end
 
 procedure Node850
 begin
-    gsay_reply(1609, message_str(1609, 850) + obj_name(dude_obj) + message_str(1609, 851));
-    giq_option(4, 1609, 852, Do_Sex, 50);
+    Reply(mstr(850) + dude_name + mstr(851));
+    giq_option(4, NAME, 852, Do_Sex, NEUTRAL_REACTION);
 end
 
 procedure Node900
 begin
-    gsay_reply(1609, 900);
-    giq_option(4, 1609, 901, Node999_float302, 50);
+    Reply(900);
+    giq_option(4, NAME, 901, Node999_float302, NEUTRAL_REACTION);
 end
 
 procedure Node950
 begin
-    gsay_reply(1609, 950);
-    giq_option(4, 1609, 951, Node999_float302, 50);
+    Reply(950);
+    giq_option(4, NAME, 951, Node999_float302, NEUTRAL_REACTION);
 end
 
 procedure Node1000
 begin
-    gsay_reply(1609, message_str(1609, 1000) + obj_name(dude_obj) + message_str(1609, 1001));
+    Reply(mstr(1000) + dude_name + mstr(1001));
     set_local_var(5, local_var(5) bwor 16);
-    giq_option(4, 1609, 1002, Node1050, 50);
+    giq_option(4, NAME, 1002, Node1050, NEUTRAL_REACTION);
 end
 
 procedure Node1050
 begin
-    gsay_reply(1609, message_str(1609, 1050) + obj_name(dude_obj) + message_str(1609, 1051));
-    giq_option(4, 1609, 1052, Node1100, 50);
-    giq_option(4, 1609, 1053, Node999_float303, 50);
+    Reply(mstr(1050) + dude_name + mstr(1051));
+    giq_option(4, NAME, 1052, Node1100, NEUTRAL_REACTION);
+    giq_option(4, NAME, 1053, Node999_float303, NEUTRAL_REACTION);
 end
 
 procedure Node1100
 begin
-    gsay_reply(1609, obj_name(dude_obj) + message_str(1609, 1100));
-    set_global_var(865, global_var(865) bwor 64);
-    giq_option(4, 1609, 1101, Node999, 50);
+    Reply(dude_name + mstr(1100));
+    set_global_var(GVAR_AQ_ALBUQ_EVENTS, global_var(GVAR_AQ_ALBUQ_EVENTS) bwor 64);
+    giq_option(4, NAME, 1101, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node1150
 begin
-    gsay_reply(1609, obj_name(dude_obj) + message_str(1609, 1150));
-    giq_option(4, 1609, 1151, Node999, 50);
-    if ((global_var(866) bwand 64) != 0) then begin
-        giq_option(4, 1609, 1152, Node1250, 50);
+    Reply(dude_name + mstr(1150));
+    giq_option(4, NAME, 1151, Node999, NEUTRAL_REACTION);
+    if ((global_var(GVAR_AQ_ALBUQ_SPEECHES) bwand 64) != 0) then begin
+        giq_option(4, NAME, 1152, Node1250, NEUTRAL_REACTION);
     end
-    if ((global_var(866) bwand 128) != 0) then begin
-        giq_option(4, 1609, 1153, Node1200, 50);
+    if ((global_var(GVAR_AQ_ALBUQ_SPEECHES) bwand 128) != 0) then begin
+        giq_option(4, NAME, 1153, Node1200, NEUTRAL_REACTION);
     end
-    giq_option(-3, 1609, 1154, Node999, 50);
+    giq_option(-3, NAME, 1154, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node1200
 begin
-    gsay_reply(1609, message_str(1609, 1200) + obj_name(dude_obj) + message_str(1609, 1201));
-    giq_option(4, 1609, 1202, Node999_floa307_hosital, 50);
-    giq_option(4, 1609, 1203, Node999, 50);
+    Reply(mstr(1200) + dude_name + mstr(1201));
+    giq_option(4, NAME, 1202, Node999_floa307_hosital, NEUTRAL_REACTION);
+    giq_option(4, NAME, 1203, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node1250
 begin
-    gsay_reply(1609, message_str(1609, 1250) + obj_name(dude_obj) + message_str(1609, 1251));
-    giq_option(4, 1609, 1252, Node999_floa307_wildparadise, 50);
-    giq_option(4, 1609, 1253, Node999, 50);
+    Reply(mstr(1250) + dude_name + mstr(1251));
+    giq_option(4, NAME, 1252, Node999_floa307_wildparadise, NEUTRAL_REACTION);
+    giq_option(4, NAME, 1253, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node1300
 begin
-    gsay_reply(1609, obj_name(dude_obj) + message_str(1609, 1300));
-    giq_option(4, 1609, 1301, Do_Sex_308, 50);
-    giq_option(4, 1609, 1302, Node999, 50);
-    if ((global_var(866) bwand 64) != 0) then begin
-        giq_option(4, 1609, 1303, Node1250, 50);
+    Reply(dude_name + mstr(1300));
+    giq_option(4, NAME, 1301, Do_Sex_308, NEUTRAL_REACTION);
+    giq_option(4, NAME, 1302, Node999, NEUTRAL_REACTION);
+    if ((global_var(GVAR_AQ_ALBUQ_SPEECHES) bwand 64) != 0) then begin
+        giq_option(4, NAME, 1303, Node1250, NEUTRAL_REACTION);
     end
-    if ((global_var(866) bwand 128) != 0) then begin
-        giq_option(4, 1609, 1304, Node1200, 50);
+    if ((global_var(GVAR_AQ_ALBUQ_SPEECHES) bwand 128) != 0) then begin
+        giq_option(4, NAME, 1304, Node1200, NEUTRAL_REACTION);
     end
-    giq_option(-3, 1609, 1305, Node999, 50);
+    giq_option(-3, NAME, 1305, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node999_to_Ben
@@ -1164,49 +1164,49 @@ end
 procedure Node999_float303
 begin
     set_local_var(5, local_var(5) bwor 4);
-    set_global_var(0, global_var(0) - 3);
-    set_global_var(37, 0);
-    set_global_var(38, 0);
-    set_global_var(39, 0);
-    set_global_var(40, 0);
-    set_global_var(41, 0);
-    set_global_var(42, 0);
-    set_global_var(43, 0);
-    set_global_var(44, 0);
-    set_global_var(45, 0);
-    if (global_var(0) >= 1000) then begin
-        set_global_var(37, 1);
+    set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) - 3);
+    set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+    set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+    set_global_var(GVAR_KARMA_DEFENDER, 0);
+    set_global_var(GVAR_KARMA_WANDERER, 0);
+    set_global_var(GVAR_KARMA_BETRAYER, 0);
+    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+    set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+    set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+    if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
     end
     else begin
-        if (global_var(0) >= 750) then begin
-            set_global_var(38, 1);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+            set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
         end
         else begin
-            if (global_var(0) >= 500) then begin
-                set_global_var(39, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
             end
             else begin
-                if (global_var(0) >= 250) then begin
-                    set_global_var(40, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                    set_global_var(GVAR_KARMA_DEFENDER, 1);
                 end
                 else begin
-                    if (global_var(0) > -250) then begin
-                        set_global_var(41, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                        set_global_var(GVAR_KARMA_WANDERER, 1);
                     end
                     else begin
-                        if (global_var(0) > -500) then begin
-                            set_global_var(42, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                            set_global_var(GVAR_KARMA_BETRAYER, 1);
                         end
                         else begin
-                            if (global_var(0) > -750) then begin
-                                set_global_var(43, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                             end
                             else begin
-                                if (global_var(0) > -1000) then begin
-                                    set_global_var(44, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                    set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                 end
                                 else begin
-                                    set_global_var(45, 1);
+                                    set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                 end
                             end
                         end
@@ -1231,49 +1231,49 @@ end
 procedure Node999_floa307_wildparadise
 begin
     set_local_var(5, local_var(5) bwor 2048);
-    set_global_var(0, global_var(0) - 10);
-    set_global_var(37, 0);
-    set_global_var(38, 0);
-    set_global_var(39, 0);
-    set_global_var(40, 0);
-    set_global_var(41, 0);
-    set_global_var(42, 0);
-    set_global_var(43, 0);
-    set_global_var(44, 0);
-    set_global_var(45, 0);
-    if (global_var(0) >= 1000) then begin
-        set_global_var(37, 1);
+    set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) - 10);
+    set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+    set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+    set_global_var(GVAR_KARMA_DEFENDER, 0);
+    set_global_var(GVAR_KARMA_WANDERER, 0);
+    set_global_var(GVAR_KARMA_BETRAYER, 0);
+    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+    set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+    set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+    if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
     end
     else begin
-        if (global_var(0) >= 750) then begin
-            set_global_var(38, 1);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+            set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
         end
         else begin
-            if (global_var(0) >= 500) then begin
-                set_global_var(39, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
             end
             else begin
-                if (global_var(0) >= 250) then begin
-                    set_global_var(40, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                    set_global_var(GVAR_KARMA_DEFENDER, 1);
                 end
                 else begin
-                    if (global_var(0) > -250) then begin
-                        set_global_var(41, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                        set_global_var(GVAR_KARMA_WANDERER, 1);
                     end
                     else begin
-                        if (global_var(0) > -500) then begin
-                            set_global_var(42, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                            set_global_var(GVAR_KARMA_BETRAYER, 1);
                         end
                         else begin
-                            if (global_var(0) > -750) then begin
-                                set_global_var(43, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                             end
                             else begin
-                                if (global_var(0) > -1000) then begin
-                                    set_global_var(44, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                    set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                 end
                                 else begin
-                                    set_global_var(45, 1);
+                                    set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                 end
                             end
                         end
@@ -1290,49 +1290,49 @@ end
 
 procedure Do_Sex
 begin
-    set_global_var(0, global_var(0) - 5);
-    set_global_var(37, 0);
-    set_global_var(38, 0);
-    set_global_var(39, 0);
-    set_global_var(40, 0);
-    set_global_var(41, 0);
-    set_global_var(42, 0);
-    set_global_var(43, 0);
-    set_global_var(44, 0);
-    set_global_var(45, 0);
-    if (global_var(0) >= 1000) then begin
-        set_global_var(37, 1);
+    set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) - 5);
+    set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+    set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+    set_global_var(GVAR_KARMA_DEFENDER, 0);
+    set_global_var(GVAR_KARMA_WANDERER, 0);
+    set_global_var(GVAR_KARMA_BETRAYER, 0);
+    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+    set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+    set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+    if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
     end
     else begin
-        if (global_var(0) >= 750) then begin
-            set_global_var(38, 1);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+            set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
         end
         else begin
-            if (global_var(0) >= 500) then begin
-                set_global_var(39, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
             end
             else begin
-                if (global_var(0) >= 250) then begin
-                    set_global_var(40, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                    set_global_var(GVAR_KARMA_DEFENDER, 1);
                 end
                 else begin
-                    if (global_var(0) > -250) then begin
-                        set_global_var(41, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                        set_global_var(GVAR_KARMA_WANDERER, 1);
                     end
                     else begin
-                        if (global_var(0) > -500) then begin
-                            set_global_var(42, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                            set_global_var(GVAR_KARMA_BETRAYER, 1);
                         end
                         else begin
-                            if (global_var(0) > -750) then begin
-                                set_global_var(43, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                             end
                             else begin
-                                if (global_var(0) > -1000) then begin
-                                    set_global_var(44, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                    set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                 end
                                 else begin
-                                    set_global_var(45, 1);
+                                    set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                 end
                             end
                         end
@@ -1351,4 +1351,3 @@ begin
     set_local_var(5, local_var(5) bwor 32);
     call Node999();
 end
-
