diff --git a/Mods/Resurrection/Original/AALIEN1.ssl b/Mods/Resurrection/ReDefine/AALIEN1.ssl
index 446430f..45cebc7 100644
--- a/Mods/Resurrection/Original/AALIEN1.ssl
+++ b/Mods/Resurrection/ReDefine/AALIEN1.ssl
@@ -44,12 +44,12 @@ begin
             return 1;
         end
     end
-    if ((party_member_obj(16777779) != 0) and ((global_var(910) bwand 32) == 0)) then begin
+    if ((party_member_obj(16777779) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 32) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777779)) <= 5) then begin
             return 1;
         end
     end
-    if ((party_member_obj(16777811) != 0) and ((global_var(910) bwand 64) == 0)) then begin
+    if ((party_member_obj(16777811) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 64) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777811)) <= 5) then begin
             return 1;
         end
@@ -64,7 +64,7 @@ begin
             return 1;
         end
     end
-    if ((party_member_obj(16777814) != 0) and ((global_var(910) bwand 128) == 0)) then begin
+    if ((party_member_obj(16777814) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 128) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777814)) <= 5) then begin
             return 1;
         end
@@ -123,10 +123,10 @@ procedure look_at_p_proc
 begin
     script_overrides;
     if (local_var(4) == 17688) then begin
-        display_msg(message_str(1740, 101));
+        display_mstr(101);
     end
     else begin
-        display_msg(message_str(1740, 100));
+        display_mstr(100);
     end
 end
 
@@ -134,10 +134,10 @@ procedure description_p_proc
 begin
     script_overrides;
     if (local_var(4) == 17688) then begin
-        display_msg(message_str(1740, 201));
+        display_mstr(201);
     end
     else begin
-        display_msg(message_str(1740, 200));
+        display_mstr(200);
     end
 end
 
@@ -147,8 +147,8 @@ end
 
 procedure destroy_p_proc
 begin
-    set_global_var(1012, global_var(1012) + 1);
-    if ((global_var(1012) >= 11) and (elevation(dude_obj) == 1) and not((map_var(0) bwand 256) != 0)) then begin
+    set_global_var(GVAR_AQE_2ST_FLOOR_ALIENS_KILLED, global_var(GVAR_AQE_2ST_FLOOR_ALIENS_KILLED) + 1);
+    if ((global_var(GVAR_AQE_2ST_FLOOR_ALIENS_KILLED) >= 11) and (dude_elevation == 1) and not((map_var(0) bwand 256) != 0)) then begin
         add_timer_event(Map_event, game_ticks(0), 4);
     end
 end
@@ -158,42 +158,42 @@ begin
     if (local_var(4) == 0) then begin
         set_local_var(4, tile_num(self_obj));
     end
-    critter_add_trait(self_obj, 1, 6, 186);
-    critter_add_trait(self_obj, 1, 5, 130);
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 186);
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_ALIEN);
 end
 
 procedure critter_p_proc
 begin
     if (((map_var(0) bwand 256) != 0) and not(combat_is_initialized)) then begin
-        set_global_var(1012, global_var(1012) + 1);
-        critter_dmg(self_obj, 999, 0);
+        set_global_var(GVAR_AQE_2ST_FLOOR_ALIENS_KILLED, global_var(GVAR_AQE_2ST_FLOOR_ALIENS_KILLED) + 1);
+        critter_damage(self_obj, 999);
     end
     if (((map_var(0) bwand 4096) != 0) and ((local_var(4) == 20306) or (local_var(4) == 25288) or (local_var(4) == 25895) or (local_var(4) == 22926))) then begin
         if (local_var(4) == 20306) then begin
             move_to(self_obj, 22698, 1);
-            set_global_var(1012, global_var(1012) + 1);
-            critter_heal(self_obj, -get_critter_stat(self_obj, 7));
+            set_global_var(GVAR_AQE_2ST_FLOOR_ALIENS_KILLED, global_var(GVAR_AQE_2ST_FLOOR_ALIENS_KILLED) + 1);
+            critter_heal(self_obj, -get_critter_stat(self_obj, STAT_max_hp));
         end
         if (local_var(4) == 25288) then begin
             move_to(self_obj, 23095, 1);
-            set_global_var(1012, global_var(1012) + 1);
-            critter_heal(self_obj, -get_critter_stat(self_obj, 7));
+            set_global_var(GVAR_AQE_2ST_FLOOR_ALIENS_KILLED, global_var(GVAR_AQE_2ST_FLOOR_ALIENS_KILLED) + 1);
+            critter_heal(self_obj, -get_critter_stat(self_obj, STAT_max_hp));
         end
         if (local_var(4) == 25895) then begin
-            set_global_var(1012, global_var(1012) + 1);
+            set_global_var(GVAR_AQE_2ST_FLOOR_ALIENS_KILLED, global_var(GVAR_AQE_2ST_FLOOR_ALIENS_KILLED) + 1);
             move_to(self_obj, 23699, 1);
-            critter_heal(self_obj, -get_critter_stat(self_obj, 7));
+            critter_heal(self_obj, -get_critter_stat(self_obj, STAT_max_hp));
         end
         if (local_var(4) == 22926) then begin
             move_to(self_obj, 22692, 1);
-            set_global_var(1012, global_var(1012) + 1);
-            critter_heal(self_obj, -get_critter_stat(self_obj, 7));
+            set_global_var(GVAR_AQE_2ST_FLOOR_ALIENS_KILLED, global_var(GVAR_AQE_2ST_FLOOR_ALIENS_KILLED) + 1);
+            critter_heal(self_obj, -get_critter_stat(self_obj, STAT_max_hp));
         end
     end
-    if (elevation(dude_obj) == elevation(self_obj)) then begin
+    if (dude_elevation == elevation(self_obj)) then begin
         if (see_time < game_time) then begin
             if (obj_can_see_obj(self_obj, dude_obj)) then begin
-                attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+                attack(dude_obj);
             end
             see_time := game_time + random(10 * 2, 10 * 5);
         end
@@ -205,4 +205,3 @@ begin
         end
     end
 end
-
