diff --git a/Mods/Resurrection/Original/AANOZOL1.ssl b/Mods/Resurrection/ReDefine/AANOZOL1.ssl
index fab80be..3674c91 100644
--- a/Mods/Resurrection/Original/AANOZOL1.ssl
+++ b/Mods/Resurrection/ReDefine/AANOZOL1.ssl
@@ -56,12 +56,12 @@ begin
             return 1;
         end
     end
-    if ((party_member_obj(16777779) != 0) and ((global_var(910) bwand 32) == 0)) then begin
+    if ((party_member_obj(16777779) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 32) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777779)) <= 5) then begin
             return 1;
         end
     end
-    if ((party_member_obj(16777811) != 0) and ((global_var(910) bwand 64) == 0)) then begin
+    if ((party_member_obj(16777811) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 64) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777811)) <= 5) then begin
             return 1;
         end
@@ -76,7 +76,7 @@ begin
             return 1;
         end
     end
-    if ((party_member_obj(16777814) != 0) and ((global_var(910) bwand 128) == 0)) then begin
+    if ((party_member_obj(16777814) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 128) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777814)) <= 5) then begin
             return 1;
         end
@@ -134,15 +134,15 @@ end
 
 procedure critter_p_proc
 begin
-    if ((local_var(4) == 1) and obj_can_see_obj(self_obj, dude_obj) and (has_trait(1, self_obj, 666) == 1)) then begin
+    if ((local_var(4) == 1) and obj_can_see_obj(self_obj, dude_obj) and (obj_is_visible_flag(self_obj) == 1)) then begin
         if (critter_is_fleeing(self_obj)) then begin
             if (anim_busy(self_obj) == 0) then begin
                 if ((tile_distance_objs(self_obj, dude_obj) < 8) or obj_can_see_obj(self_obj, dude_obj)) then begin
                     if (anim_busy(self_obj) == 0) then begin
-                        global_temp := rotation_to_tile(tile_num(dude_obj), tile_num(self_obj));
+                        global_temp := rotation_to_tile(dude_tile, tile_num(self_obj));
                         animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(self_obj), global_temp, random(3, 10)), 1);
                         global_temp := (global_temp + 1) % 6;
-                        while ((anim_busy(self_obj) == 0) and (global_temp != rotation_to_tile(tile_num(dude_obj), tile_num(self_obj)))) do begin
+                        while ((anim_busy(self_obj) == 0) and (global_temp != rotation_to_tile(dude_tile, tile_num(self_obj)))) do begin
                             debug_msg("flee loop: rot == " + global_temp);
                             animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(self_obj), global_temp, random(3, 10)), 1);
                             global_temp := (global_temp + 1) % 6;
@@ -152,18 +152,18 @@ begin
             end
         end
         else begin
-            attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+            attack(dude_obj);
         end
     end
-    if (obj_can_see_obj(self_obj, dude_obj) and (has_trait(1, self_obj, 666) == 1) and ((local_var(6) bwand 8) == 0) and (tile_distance(tile_num(self_obj), tile_num(dude_obj)) < 7) and (local_var(5) != 1) and (global_var(996) == 2)) then begin
+    if (obj_can_see_obj(self_obj, dude_obj) and (obj_is_visible_flag(self_obj) == 1) and ((local_var(6) bwand 8) == 0) and (tile_distance(tile_num(self_obj), dude_tile) < 7) and (local_var(5) != 1) and (global_var(GVAR_AQE_QUEST_ANONYM_3) == 2)) then begin
         set_local_var(6, local_var(6) bwor 8);
         dialogue_system_enter;
     end
-    if (obj_can_see_obj(self_obj, dude_obj) and (has_trait(1, self_obj, 666) == 1) and (tile_distance(tile_num(self_obj), tile_num(dude_obj)) < 7) and (local_var(5) == 1) and ((map_var(0) bwand 16777216) == 0) and ((local_var(6) bwand 16) == 0) and (global_var(998) == 1)) then begin
+    if (obj_can_see_obj(self_obj, dude_obj) and (obj_is_visible_flag(self_obj) == 1) and (tile_distance(tile_num(self_obj), dude_tile) < 7) and (local_var(5) == 1) and ((map_var(0) bwand 16777216) == 0) and ((local_var(6) bwand 16) == 0) and (global_var(GVAR_AQE_ENEMY_EMPEROR) == 1)) then begin
         set_map_var(0, map_var(0) bwor 16777216);
         dialogue_system_enter;
     end
-    if ((local_var(5) == 1) and (elevation(dude_obj) == 2) and (has_trait(1, self_obj, 666) == 1) and (local_var(4) != 1)) then begin
+    if ((local_var(5) == 1) and (dude_elevation == 2) and (obj_is_visible_flag(self_obj) == 1) and (local_var(4) != 1)) then begin
         destroy_object(self_obj);
     end
 end
@@ -171,7 +171,7 @@ end
 procedure pickup_p_proc
 begin
     if (source_obj == dude_obj) then begin
-        set_global_var(904, global_var(904) bwor 1024);
+        set_global_var(GVAR_AQE_ALBUQ_EMPEROR_EVENTS, global_var(GVAR_AQE_ALBUQ_EMPEROR_EVENTS) bwor 1024);
         set_local_var(4, 1);
     end
 end
@@ -188,11 +188,11 @@ procedure talk_p_proc
 begin
     set_local_var(6, local_var(6) bwor 8);
     if ((local_var(5) == 1) and ((local_var(6) bwand 16) == 0)) then begin
-        float_msg(self_obj, message_str(115, random(300, 302)), 8);
+        float_msg(self_obj, mstr(random(300, 302)), FLOAT_MSG_YELLOW);
     end
     else begin
-        anim(dude_obj, 1000, rotation_to_tile(tile_num(dude_obj), tile_num(self_obj)));
-        anim(self_obj, 1000, rotation_to_tile(tile_num(self_obj), tile_num(dude_obj)));
+        anim(dude_obj, 1000, rotation_to_tile(dude_tile, tile_num(self_obj)));
+        anim(self_obj, 1000, rotation_to_tile(tile_num(self_obj), dude_tile));
         start_gdialog(115, self_obj, 4, -1, -1);
         gsay_start;
         if (((map_var(0) bwand 16777216) != 0) and ((local_var(6) bwand 16) == 0)) then begin
@@ -209,55 +209,55 @@ end
 procedure destroy_p_proc
 begin
     if (source_obj == dude_obj) then begin
-        set_global_var(4, global_var(4) + 1);
-        if (metarule(51, self_obj) == 2) then begin
+        set_global_var(GVAR_BAD_MONSTER, global_var(GVAR_BAD_MONSTER) + 1);
+        if (critter_kill_type(self_obj) == KILL_TYPE_children_kills) then begin
             if (source_obj == dude_obj) then begin
-                set_global_var(2, 0);
-                set_global_var(1, global_var(1) + 1);
-                set_global_var(0, global_var(0) + -15);
+                set_global_var(GVAR_CHAMPION_REPUTATION, 0);
+                set_global_var(GVAR_CHILDKILLER_REPUTATION, global_var(GVAR_CHILDKILLER_REPUTATION) + 1);
+                set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + -15);
                 debug_msg("Player gains " + -15 + " Karma Points.");
-                set_global_var(37, 0);
-                set_global_var(38, 0);
-                set_global_var(39, 0);
-                set_global_var(40, 0);
-                set_global_var(41, 0);
-                set_global_var(42, 0);
-                set_global_var(43, 0);
-                set_global_var(44, 0);
-                set_global_var(45, 0);
-                if (global_var(0) >= 1000) then begin
-                    set_global_var(37, 1);
+                set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+                set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+                set_global_var(GVAR_KARMA_DEFENDER, 0);
+                set_global_var(GVAR_KARMA_WANDERER, 0);
+                set_global_var(GVAR_KARMA_BETRAYER, 0);
+                set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+                set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+                set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+                    set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
                 end
                 else begin
-                    if (global_var(0) >= 750) then begin
-                        set_global_var(38, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
                     end
                     else begin
-                        if (global_var(0) >= 500) then begin
-                            set_global_var(39, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                            set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                         end
                         else begin
-                            if (global_var(0) >= 250) then begin
-                                set_global_var(40, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                                set_global_var(GVAR_KARMA_DEFENDER, 1);
                             end
                             else begin
-                                if (global_var(0) > -250) then begin
-                                    set_global_var(41, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                                    set_global_var(GVAR_KARMA_WANDERER, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -500) then begin
-                                        set_global_var(42, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                        set_global_var(GVAR_KARMA_BETRAYER, 1);
                                     end
                                     else begin
-                                        if (global_var(0) > -750) then begin
-                                            set_global_var(43, 1);
+                                        if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                            set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                         end
                                         else begin
-                                            if (global_var(0) > -1000) then begin
-                                                set_global_var(44, 1);
+                                            if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                                set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                             end
                                             else begin
-                                                set_global_var(45, 1);
+                                                set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                             end
                                         end
                                     end
@@ -266,62 +266,62 @@ begin
                         end
                     end
                 end
-                set_global_var(51, global_var(51) + -8);
+                set_global_var(GVAR_TOWN_REP_ALBUQUERQUE, global_var(GVAR_TOWN_REP_ALBUQUERQUE) + -8);
                 debug_msg("Added " + -8 + " to Town Rep");
             end
         end
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(5) > (2 * global_var(4))) or (global_var(3) == 1))) then begin
-            set_global_var(3, 1);
-            set_global_var(2, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
+            set_global_var(GVAR_BERSERKER_REPUTATION, 1);
+            set_global_var(GVAR_CHAMPION_REPUTATION, 0);
         end
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(4) > (3 * global_var(5))) or (global_var(2) == 1)) and (global_var(1) == 0)) then begin
-            set_global_var(2, 1);
-            set_global_var(3, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(4) > (3 * global_var(GVAR_GOOD_MONSTER))) or (global_var(GVAR_CHAMPION_REPUTATION) == 1)) and (global_var(GVAR_CHILDKILLER_REPUTATION) == 0)) then begin
+            set_global_var(GVAR_CHAMPION_REPUTATION, 1);
+            set_global_var(GVAR_BERSERKER_REPUTATION, 0);
         end
-        set_global_var(0, global_var(0) + 5);
+        set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + 5);
         debug_msg("Player gains " + 5 + " Karma Points.");
-        set_global_var(37, 0);
-        set_global_var(38, 0);
-        set_global_var(39, 0);
-        set_global_var(40, 0);
-        set_global_var(41, 0);
-        set_global_var(42, 0);
-        set_global_var(43, 0);
-        set_global_var(44, 0);
-        set_global_var(45, 0);
-        if (global_var(0) >= 1000) then begin
-            set_global_var(37, 1);
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+        set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+        set_global_var(GVAR_KARMA_DEFENDER, 0);
+        set_global_var(GVAR_KARMA_WANDERER, 0);
+        set_global_var(GVAR_KARMA_BETRAYER, 0);
+        set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+        set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+            set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
         end
         else begin
-            if (global_var(0) >= 750) then begin
-                set_global_var(38, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
             end
             else begin
-                if (global_var(0) >= 500) then begin
-                    set_global_var(39, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                 end
                 else begin
-                    if (global_var(0) >= 250) then begin
-                        set_global_var(40, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                        set_global_var(GVAR_KARMA_DEFENDER, 1);
                     end
                     else begin
-                        if (global_var(0) > -250) then begin
-                            set_global_var(41, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                            set_global_var(GVAR_KARMA_WANDERER, 1);
                         end
                         else begin
-                            if (global_var(0) > -500) then begin
-                                set_global_var(42, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                set_global_var(GVAR_KARMA_BETRAYER, 1);
                             end
                             else begin
-                                if (global_var(0) > -750) then begin
-                                    set_global_var(43, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -1000) then begin
-                                        set_global_var(44, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                     end
                                     else begin
-                                        set_global_var(45, 1);
+                                        set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                     end
                                 end
                             end
@@ -330,7 +330,7 @@ begin
                 end
             end
         end
-        set_global_var(51, global_var(51) + 2);
+        set_global_var(GVAR_TOWN_REP_ALBUQUERQUE, global_var(GVAR_TOWN_REP_ALBUQUERQUE) + 2);
         debug_msg("Added " + 2 + " to Town Rep");
     end
     AnonymGuard_Obj := 0;
@@ -339,13 +339,13 @@ end
 procedure look_at_p_proc
 begin
     script_overrides;
-    display_msg(message_str(115, 100));
+    display_mstr(100);
 end
 
 procedure description_p_proc
 begin
     script_overrides;
-    display_msg(message_str(115, 200));
+    display_mstr(200);
 end
 
 procedure use_skill_on_p_proc
@@ -355,23 +355,23 @@ end
 procedure damage_p_proc
 begin
     if (source_obj == dude_obj) then begin
-        set_global_var(904, global_var(904) bwor 1024);
+        set_global_var(GVAR_AQE_ALBUQ_EMPEROR_EVENTS, global_var(GVAR_AQE_ALBUQ_EMPEROR_EVENTS) bwor 1024);
         set_local_var(4, 1);
     end
 end
 
 procedure map_enter_p_proc
 begin
-    critter_add_trait(self_obj, 1, 6, 283);
-    critter_add_trait(self_obj, 1, 5, 3);
-    if (not(metarule(22, 0))) then begin
-        if (global_var(996) == 2) then begin
-            if (has_trait(1, self_obj, 666) == 0) then begin
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 283);
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_TOUGH_GUARD_NO_TALKING);
+    if (not(is_loading_game)) then begin
+        if (global_var(GVAR_AQE_QUEST_ANONYM_3) == 2) then begin
+            if (obj_is_visible_flag(self_obj) == 0) then begin
                 set_obj_visibility(self_obj, 0);
             end
         end
         else begin
-            if (has_trait(1, self_obj, 666) == 1) then begin
+            if (obj_is_visible_flag(self_obj) == 1) then begin
                 set_obj_visibility(self_obj, 1);
             end
         end
@@ -386,109 +386,108 @@ end
 procedure Node999
 begin
     anim(self_obj, 1000, 2);
-    if ((party_member_obj(16777814) != 0) and ((global_var(910) bwand 128) == 0)) then begin
+    if ((party_member_obj(16777814) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 128) == 0)) then begin
         add_timer_event(party_member_obj(16777814), 5, 27);
     end
 end
 
 procedure Node500
 begin
-    gsay_reply(115, 500);
-    if (global_var(998) == 1) then begin
-        giq_option(4, 115, 501, Node800, 50);
+    Reply(500);
+    if (global_var(GVAR_AQE_ENEMY_EMPEROR) == 1) then begin
+        giq_option(4, NAME, 501, Node800, NEUTRAL_REACTION);
     end
     else begin
-        giq_option(4, 115, 501, Node600, 50);
+        giq_option(4, NAME, 501, Node600, NEUTRAL_REACTION);
     end
-    giq_option(-3, 115, 502, Node550, 50);
+    giq_option(-3, NAME, 502, Node550, NEUTRAL_REACTION);
 end
 
 procedure Node550
 begin
-    gsay_reply(115, 550);
-    giq_option(-3, 115, 551, Node998, 50);
+    Reply(550);
+    giq_option(-3, NAME, 551, Node998, NEUTRAL_REACTION);
 end
 
 procedure Node600
 begin
-    gsay_reply(115, 600);
+    Reply(600);
     set_local_var(5, 1);
     set_map_var(0, map_var(0) bwor 4096);
     if ((local_var(6) bwand 1) == 0) then begin
-        giq_option(4, 115, 601, Node650, 50);
+        giq_option(4, NAME, 601, Node650, NEUTRAL_REACTION);
     end
     if ((local_var(6) bwand 2) == 0) then begin
-        giq_option(4, 115, 602, Node700, 50);
+        giq_option(4, NAME, 602, Node700, NEUTRAL_REACTION);
     end
     if ((local_var(6) bwand 4) == 0) then begin
-        giq_option(4, 115, 603, Node750, 50);
+        giq_option(4, NAME, 603, Node750, NEUTRAL_REACTION);
     end
-    giq_option(4, 115, 604, Node999, 50);
+    giq_option(4, NAME, 604, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node650
 begin
-    gsay_reply(115, 650);
+    Reply(650);
     set_local_var(6, local_var(6) bwor 1);
     if ((local_var(6) bwand 2) == 0) then begin
-        giq_option(4, 115, 651, Node700, 50);
+        giq_option(4, NAME, 651, Node700, NEUTRAL_REACTION);
     end
     if ((local_var(6) bwand 4) == 0) then begin
-        giq_option(4, 115, 652, Node750, 50);
+        giq_option(4, NAME, 652, Node750, NEUTRAL_REACTION);
     end
-    giq_option(4, 115, 653, Node999, 50);
+    giq_option(4, NAME, 653, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node700
 begin
-    if (get_critter_stat(dude_obj, 34) == 0) then begin
-        gsay_reply(115, 700);
+    if (dude_gender == 0) then begin
+        Reply(700);
     end
     else begin
-        gsay_reply(115, 701);
+        Reply(701);
     end
     set_local_var(6, local_var(6) bwor 2);
     if ((local_var(6) bwand 1) == 0) then begin
-        giq_option(4, 115, 702, Node650, 50);
+        giq_option(4, NAME, 702, Node650, NEUTRAL_REACTION);
     end
     if ((local_var(6) bwand 4) == 0) then begin
-        giq_option(4, 115, 703, Node750, 50);
+        giq_option(4, NAME, 703, Node750, NEUTRAL_REACTION);
     end
-    giq_option(4, 115, 704, Node999, 50);
+    giq_option(4, NAME, 704, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node750
 begin
-    if (get_critter_stat(dude_obj, 34) == 0) then begin
-        gsay_reply(115, 750);
+    if (dude_gender == 0) then begin
+        Reply(750);
     end
     else begin
-        gsay_reply(115, 751);
+        Reply(751);
     end
     set_local_var(6, local_var(6) bwor 4);
     if ((local_var(6) bwand 1) == 0) then begin
-        giq_option(4, 115, 752, Node650, 50);
+        giq_option(4, NAME, 752, Node650, NEUTRAL_REACTION);
     end
     if ((local_var(6) bwand 2) == 0) then begin
-        giq_option(4, 115, 753, Node700, 50);
+        giq_option(4, NAME, 753, Node700, NEUTRAL_REACTION);
     end
-    giq_option(4, 115, 754, Node999, 50);
+    giq_option(4, NAME, 754, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node800
 begin
-    gsay_reply(115, 800);
-    set_global_var(904, global_var(904) bwor 1024);
+    Reply(800);
+    set_global_var(GVAR_AQE_ALBUQ_EMPEROR_EVENTS, global_var(GVAR_AQE_ALBUQ_EMPEROR_EVENTS) bwor 1024);
     set_map_var(0, map_var(0) bwor 16777216);
     set_local_var(6, local_var(6) bwor 16);
-    giq_option(4, 115, 801, Node998, 50);
+    giq_option(4, NAME, 801, Node998, NEUTRAL_REACTION);
 end
 
 procedure Node850
 begin
-    gsay_reply(115, 850);
-    set_global_var(904, global_var(904) bwor 1024);
+    Reply(850);
+    set_global_var(GVAR_AQE_ALBUQ_EMPEROR_EVENTS, global_var(GVAR_AQE_ALBUQ_EMPEROR_EVENTS) bwor 1024);
     set_local_var(6, local_var(6) bwor 16);
-    giq_option(4, 115, 851, Node998, 50);
+    giq_option(4, NAME, 851, Node998, NEUTRAL_REACTION);
 end
-
