diff --git a/Mods/Resurrection/Original/ABRADSNE.ssl b/Mods/Resurrection/ReDefine/ABRADSNE.ssl
index ca15fe9..241772f 100644
--- a/Mods/Resurrection/Original/ABRADSNE.ssl
+++ b/Mods/Resurrection/ReDefine/ABRADSNE.ssl
@@ -99,12 +99,12 @@ begin
             return 1;
         end
     end
-    if ((party_member_obj(16777779) != 0) and ((global_var(910) bwand 32) == 0)) then begin
+    if ((party_member_obj(16777779) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 32) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777779)) <= 5) then begin
             return 1;
         end
     end
-    if ((party_member_obj(16777811) != 0) and ((global_var(910) bwand 64) == 0)) then begin
+    if ((party_member_obj(16777811) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 64) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777811)) <= 5) then begin
             return 1;
         end
@@ -119,7 +119,7 @@ begin
             return 1;
         end
     end
-    if ((party_member_obj(16777814) != 0) and ((global_var(910) bwand 128) == 0)) then begin
+    if ((party_member_obj(16777814) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 128) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777814)) <= 5) then begin
             return 1;
         end
@@ -177,15 +177,15 @@ end
 
 procedure critter_p_proc
 begin
-    if ((local_var(4) == 1) or (global_var(998) == 1) and ((local_var(6) bwand 1024) == 0) and obj_can_see_obj(self_obj, dude_obj) and (has_trait(1, self_obj, 666) == 1)) then begin
+    if ((local_var(4) == 1) or (global_var(GVAR_AQE_ENEMY_EMPEROR) == 1) and ((local_var(6) bwand 1024) == 0) and obj_can_see_obj(self_obj, dude_obj) and (obj_is_visible_flag(self_obj) == 1)) then begin
         if (critter_is_fleeing(self_obj)) then begin
             if (anim_busy(self_obj) == 0) then begin
                 if ((tile_distance_objs(self_obj, dude_obj) < 8) or obj_can_see_obj(self_obj, dude_obj)) then begin
                     if (anim_busy(self_obj) == 0) then begin
-                        global_temp := rotation_to_tile(tile_num(dude_obj), tile_num(self_obj));
+                        global_temp := rotation_to_tile(dude_tile, tile_num(self_obj));
                         animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(self_obj), global_temp, random(3, 10)), 1);
                         global_temp := (global_temp + 1) % 6;
-                        while ((anim_busy(self_obj) == 0) and (global_temp != rotation_to_tile(tile_num(dude_obj), tile_num(self_obj)))) do begin
+                        while ((anim_busy(self_obj) == 0) and (global_temp != rotation_to_tile(dude_tile, tile_num(self_obj)))) do begin
                             debug_msg("flee loop: rot == " + global_temp);
                             animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(self_obj), global_temp, random(3, 10)), 1);
                             global_temp := (global_temp + 1) % 6;
@@ -195,13 +195,13 @@ begin
             end
         end
         else begin
-            attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+            attack(dude_obj);
         end
     end
-    if ((global_var(997) == 3) and (has_trait(1, self_obj, 666) == 0) and (global_var(998) != 1) and not((global_var(904) bwand 4194304) != 0)) then begin
+    if ((global_var(GVAR_AQE_QUEST_B_SNEED) == 3) and (obj_is_visible_flag(self_obj) == 0) and (global_var(GVAR_AQE_ENEMY_EMPEROR) != 1) and not((global_var(GVAR_AQE_ALBUQ_EMPEROR_EVENTS) bwand 4194304) != 0)) then begin
         set_obj_visibility(self_obj, 0);
     end
-    if ((local_var(5) != 1) or ((local_var(6) bwand 32) == 0) and ((local_var(6) bwand 4096) == 0) and ((local_var(6) bwand 1024) == 0) and obj_can_see_obj(self_obj, dude_obj) and not(combat_is_initialized) and tile_in_tile_rect(15327, 15318, 18127, 18718, tile_num(dude_obj)) and (has_trait(1, self_obj, 666) == 1) and not((map_var(0) bwand 256) != 0)) then begin
+    if ((local_var(5) != 1) or ((local_var(6) bwand 32) == 0) and ((local_var(6) bwand 4096) == 0) and ((local_var(6) bwand 1024) == 0) and obj_can_see_obj(self_obj, dude_obj) and not(combat_is_initialized) and tile_in_tile_rect(15327, 15318, 18127, 18718, dude_tile) and (obj_is_visible_flag(self_obj) == 1) and not((map_var(0) bwand 256) != 0)) then begin
         set_local_var(6, local_var(6) bwor 4096);
         dialogue_system_enter;
     end
@@ -244,16 +244,16 @@ begin
             if ((map_var(0) bwand 256) != 0) then begin
                 if ((map_var(0) bwand 16) != 0) then begin
                     if ((local_var(6) bwand 2048) == 0) then begin
-                        float_msg(self_obj, message_str(111, 313), 8);
+                        float_msg(self_obj, mstr(313), FLOAT_MSG_YELLOW);
                         set_local_var(6, local_var(6) bwor 2048);
                     end
                     else begin
-                        float_msg(self_obj, message_str(111, 301), 8);
+                        float_msg(self_obj, mstr(301), FLOAT_MSG_YELLOW);
                     end
                 end
                 else begin
-                    anim(dude_obj, 1000, rotation_to_tile(tile_num(dude_obj), tile_num(self_obj)));
-                    anim(self_obj, 1000, rotation_to_tile(tile_num(self_obj), tile_num(dude_obj)));
+                    anim(dude_obj, 1000, rotation_to_tile(dude_tile, tile_num(self_obj)));
+                    anim(self_obj, 1000, rotation_to_tile(tile_num(self_obj), dude_tile));
                     start_gdialog(111, self_obj, 4, -1, -1);
                     gsay_start;
                     call Node1450();
@@ -268,11 +268,11 @@ begin
             else begin
                 if (Timer_Event == 0) then begin
                     if ((map_var(0) bwand 16) != 0) then begin
-                        float_msg(self_obj, message_str(111, 301), 8);
+                        float_msg(self_obj, mstr(301), FLOAT_MSG_YELLOW);
                     end
                     else begin
-                        anim(dude_obj, 1000, rotation_to_tile(tile_num(dude_obj), tile_num(self_obj)));
-                        anim(self_obj, 1000, rotation_to_tile(tile_num(self_obj), tile_num(dude_obj)));
+                        anim(dude_obj, 1000, rotation_to_tile(dude_tile, tile_num(self_obj)));
+                        anim(self_obj, 1000, rotation_to_tile(tile_num(self_obj), dude_tile));
                         if (obj_carrying_pid_obj(dude_obj, 8) or obj_carrying_pid_obj(dude_obj, 9) or obj_carrying_pid_obj(dude_obj, 16) or obj_carrying_pid_obj(dude_obj, 18) or obj_carrying_pid_obj(dude_obj, 22) or obj_carrying_pid_obj(dude_obj, 24) or obj_carrying_pid_obj(dude_obj, 120) or obj_carrying_pid_obj(dude_obj, 241) or obj_carrying_pid_obj(dude_obj, 300) or obj_carrying_pid_obj(dude_obj, 313) or obj_carrying_pid_obj(dude_obj, 388) or obj_carrying_pid_obj(dude_obj, 396) or obj_carrying_pid_obj(dude_obj, 402) or obj_carrying_pid_obj(dude_obj, 404) or obj_carrying_pid_obj(dude_obj, 406) or obj_carrying_pid_obj(dude_obj, 605) or obj_carrying_pid_obj(dude_obj, 594) or obj_carrying_pid_obj(dude_obj, 593) or obj_carrying_pid_obj(dude_obj, 398)) then begin
                             Wearing_Pistol := 1;
                         end
@@ -290,7 +290,7 @@ begin
                         gsay_end;
                         end_dialogue;
                     end
-                    if (global_var(997) == 4) then begin
+                    if (global_var(GVAR_AQE_QUEST_B_SNEED) == 4) then begin
                         game_ui_disable;
                         add_timer_event(self_obj, game_ticks(1), 10);
                     end
@@ -307,7 +307,7 @@ begin
                         game_ui_disable;
                         add_timer_event(self_obj, game_ticks(1), 40);
                     end
-                    if ((global_var(997) == 2) and ((global_var(904) bwand 2097152) != 0)) then begin
+                    if ((global_var(GVAR_AQE_QUEST_B_SNEED) == 2) and ((global_var(GVAR_AQE_ALBUQ_EMPEROR_EVENTS) bwand 2097152) != 0)) then begin
                         set_obj_visibility(self_obj, 1);
                     end
                 end
@@ -332,11 +332,11 @@ begin
                     end
                     gsay_end;
                     end_dialogue;
-                    if ((local_var(4) == 1) and (global_var(998) == 1) and not((local_var(6) bwand 64) == 0)) then begin
-                        critter_dmg(dude_obj, 30, 0);
+                    if ((local_var(4) == 1) and (global_var(GVAR_AQE_ENEMY_EMPEROR) == 1) and not((local_var(6) bwand 64) == 0)) then begin
+                        critter_damage(dude_obj, 30);
                     end
-                    if ((local_var(4) == 1) and (global_var(998) == 1) and ((local_var(6) bwand 32768) != 0)) then begin
-                        critter_dmg(dude_obj, 30, 0);
+                    if ((local_var(4) == 1) and (global_var(GVAR_AQE_ENEMY_EMPEROR) == 1) and ((local_var(6) bwand 32768) != 0)) then begin
+                        critter_damage(dude_obj, 30);
                     end
                     if (not((local_var(6) bwand 256) == 0)) then begin
                         game_ui_disable;
@@ -353,59 +353,59 @@ end
 procedure destroy_p_proc
 begin
     if (source_obj == dude_obj) then begin
-        set_global_var(5, global_var(5) + 1);
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(4) > (3 * global_var(5))) or (global_var(2) == 1)) and (global_var(1) == 0)) then begin
-            set_global_var(2, 1);
-            set_global_var(3, 0);
+        set_global_var(GVAR_GOOD_MONSTER, global_var(GVAR_GOOD_MONSTER) + 1);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(4) > (3 * global_var(GVAR_GOOD_MONSTER))) or (global_var(GVAR_CHAMPION_REPUTATION) == 1)) and (global_var(GVAR_CHILDKILLER_REPUTATION) == 0)) then begin
+            set_global_var(GVAR_CHAMPION_REPUTATION, 1);
+            set_global_var(GVAR_BERSERKER_REPUTATION, 0);
         end
-        if (metarule(51, self_obj) == 2) then begin
+        if (critter_kill_type(self_obj) == KILL_TYPE_children_kills) then begin
             if (source_obj == dude_obj) then begin
-                set_global_var(2, 0);
-                set_global_var(1, global_var(1) + 1);
-                set_global_var(0, global_var(0) + -15);
+                set_global_var(GVAR_CHAMPION_REPUTATION, 0);
+                set_global_var(GVAR_CHILDKILLER_REPUTATION, global_var(GVAR_CHILDKILLER_REPUTATION) + 1);
+                set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + -15);
                 debug_msg("Player gains " + -15 + " Karma Points.");
-                set_global_var(37, 0);
-                set_global_var(38, 0);
-                set_global_var(39, 0);
-                set_global_var(40, 0);
-                set_global_var(41, 0);
-                set_global_var(42, 0);
-                set_global_var(43, 0);
-                set_global_var(44, 0);
-                set_global_var(45, 0);
-                if (global_var(0) >= 1000) then begin
-                    set_global_var(37, 1);
+                set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+                set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+                set_global_var(GVAR_KARMA_DEFENDER, 0);
+                set_global_var(GVAR_KARMA_WANDERER, 0);
+                set_global_var(GVAR_KARMA_BETRAYER, 0);
+                set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+                set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+                set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+                    set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
                 end
                 else begin
-                    if (global_var(0) >= 750) then begin
-                        set_global_var(38, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
                     end
                     else begin
-                        if (global_var(0) >= 500) then begin
-                            set_global_var(39, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                            set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                         end
                         else begin
-                            if (global_var(0) >= 250) then begin
-                                set_global_var(40, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                                set_global_var(GVAR_KARMA_DEFENDER, 1);
                             end
                             else begin
-                                if (global_var(0) > -250) then begin
-                                    set_global_var(41, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                                    set_global_var(GVAR_KARMA_WANDERER, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -500) then begin
-                                        set_global_var(42, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                        set_global_var(GVAR_KARMA_BETRAYER, 1);
                                     end
                                     else begin
-                                        if (global_var(0) > -750) then begin
-                                            set_global_var(43, 1);
+                                        if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                            set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                         end
                                         else begin
-                                            if (global_var(0) > -1000) then begin
-                                                set_global_var(44, 1);
+                                            if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                                set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                             end
                                             else begin
-                                                set_global_var(45, 1);
+                                                set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                             end
                                         end
                                     end
@@ -414,58 +414,58 @@ begin
                         end
                     end
                 end
-                set_global_var(53, global_var(53) + -8);
+                set_global_var(GVAR_TOWN_REP_EMPORER_CITY, global_var(GVAR_TOWN_REP_EMPORER_CITY) + -8);
                 debug_msg("Added " + -8 + " to Town Rep");
             end
         end
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(5) > (2 * global_var(4))) or (global_var(3) == 1))) then begin
-            set_global_var(3, 1);
-            set_global_var(2, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
+            set_global_var(GVAR_BERSERKER_REPUTATION, 1);
+            set_global_var(GVAR_CHAMPION_REPUTATION, 0);
         end
-        set_global_var(0, global_var(0) + -10);
+        set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + -10);
         debug_msg("Player gains " + -10 + " Karma Points.");
-        set_global_var(37, 0);
-        set_global_var(38, 0);
-        set_global_var(39, 0);
-        set_global_var(40, 0);
-        set_global_var(41, 0);
-        set_global_var(42, 0);
-        set_global_var(43, 0);
-        set_global_var(44, 0);
-        set_global_var(45, 0);
-        if (global_var(0) >= 1000) then begin
-            set_global_var(37, 1);
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+        set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+        set_global_var(GVAR_KARMA_DEFENDER, 0);
+        set_global_var(GVAR_KARMA_WANDERER, 0);
+        set_global_var(GVAR_KARMA_BETRAYER, 0);
+        set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+        set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+            set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
         end
         else begin
-            if (global_var(0) >= 750) then begin
-                set_global_var(38, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
             end
             else begin
-                if (global_var(0) >= 500) then begin
-                    set_global_var(39, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                 end
                 else begin
-                    if (global_var(0) >= 250) then begin
-                        set_global_var(40, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                        set_global_var(GVAR_KARMA_DEFENDER, 1);
                     end
                     else begin
-                        if (global_var(0) > -250) then begin
-                            set_global_var(41, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                            set_global_var(GVAR_KARMA_WANDERER, 1);
                         end
                         else begin
-                            if (global_var(0) > -500) then begin
-                                set_global_var(42, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                set_global_var(GVAR_KARMA_BETRAYER, 1);
                             end
                             else begin
-                                if (global_var(0) > -750) then begin
-                                    set_global_var(43, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -1000) then begin
-                                        set_global_var(44, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                     end
                                     else begin
-                                        set_global_var(45, 1);
+                                        set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                     end
                                 end
                             end
@@ -474,22 +474,22 @@ begin
                 end
             end
         end
-        set_global_var(53, global_var(53) + -5);
+        set_global_var(GVAR_TOWN_REP_EMPORER_CITY, global_var(GVAR_TOWN_REP_EMPORER_CITY) + -5);
         debug_msg("Added " + -5 + " to Town Rep");
     end
     if ((local_var(6) bwand 256) == 0) then begin
-        if (global_var(997) != 4) then begin
+        if (global_var(GVAR_AQE_QUEST_B_SNEED) != 4) then begin
             set_map_var(0, map_var(0) bwor 8);
         end
     end
-    if (global_var(997) != 4) then begin
-        set_global_var(997, 1);
+    if (global_var(GVAR_AQE_QUEST_B_SNEED) != 4) then begin
+        set_global_var(GVAR_AQE_QUEST_B_SNEED, 1);
     end
-    if (((map_var(0) bwand 256) != 0) and ((map_var(0) bwand 32) != 0) and not((map_var(0) bwand 8) != 0) and not((map_var(0) bwand 64) != 0) and ((game_time - global_var(1001)) < (60 * (60 * 10) * 8))) then begin
+    if (((map_var(0) bwand 256) != 0) and ((map_var(0) bwand 32) != 0) and not((map_var(0) bwand 8) != 0) and not((map_var(0) bwand 64) != 0) and ((game_time - global_var(GVAR_AQE_BRAD_WAITING_TIME)) < (60 * (60 * 10) * 8))) then begin
         set_map_var(0, map_var(0) bwor 8);
     end
-    if (((global_var(904) bwand 8388608) != 0) and ((game_time - global_var(1001)) < (60 * (60 * 10) * 8))) then begin
-        set_global_var(904, global_var(904) bwand (-1 - 8388608));
+    if (((global_var(GVAR_AQE_ALBUQ_EMPEROR_EVENTS) bwand 8388608) != 0) and ((game_time - global_var(GVAR_AQE_BRAD_WAITING_TIME)) < (60 * (60 * 10) * 8))) then begin
+        set_global_var(GVAR_AQE_ALBUQ_EMPEROR_EVENTS, global_var(GVAR_AQE_ALBUQ_EMPEROR_EVENTS) bwand (-1 - 8388608));
     end
 end
 
@@ -497,10 +497,10 @@ procedure look_at_p_proc
 begin
     script_overrides;
     if (local_var(5) == 1) then begin
-        display_msg(message_str(111, 150));
+        display_mstr(150);
     end
     else begin
-        display_msg(message_str(111, 100));
+        display_mstr(100);
     end
 end
 
@@ -509,14 +509,14 @@ begin
     script_overrides;
     if (local_var(5) == 1) then begin
         if ((map_var(0) bwand 32) != 0) then begin
-            display_msg(message_str(111, 250));
+            display_mstr(250);
         end
         else begin
-            display_msg(message_str(111, 252));
+            display_mstr(252);
         end
     end
     else begin
-        display_msg(message_str(111, 200));
+        display_mstr(200);
     end
 end
 
@@ -528,8 +528,8 @@ procedure damage_p_proc
 begin
     if ((source_obj == dude_obj) and ((local_var(6) bwand 1024) == 0)) then begin
         set_local_var(4, 1);
-        set_global_var(998, 1);
-        if ((party_member_obj(16777814) != 0) and ((global_var(910) bwand 128) == 0)) then begin
+        set_global_var(GVAR_AQE_ENEMY_EMPEROR, 1);
+        if ((party_member_obj(16777814) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 128) == 0)) then begin
             terminate_combat;
             add_timer_event(party_member_obj(16777814), 0, 25);
         end
@@ -538,43 +538,43 @@ end
 
 procedure map_enter_p_proc
 begin
-    if (not(metarule(22, 0))) then begin
-        if ((global_var(997) == 1) or (global_var(997) == 4)) then begin
+    if (not(is_loading_game)) then begin
+        if ((global_var(GVAR_AQE_QUEST_B_SNEED) == 1) or (global_var(GVAR_AQE_QUEST_B_SNEED) == 4)) then begin
             destroy_object(self_obj);
         end
-        if (global_var(999) == 3) then begin
+        if (global_var(GVAR_AQE_QUEST_COLE) == 3) then begin
             destroy_object(self_obj);
         end
     end
-    if (global_var(997) == 2) then begin
-        if ((global_var(904) bwand 2097152) != 0) then begin
+    if (global_var(GVAR_AQE_QUEST_B_SNEED) == 2) then begin
+        if ((global_var(GVAR_AQE_ALBUQ_EMPEROR_EVENTS) bwand 2097152) != 0) then begin
             set_obj_visibility(self_obj, 1);
         end
     end
     else begin
-        if (has_trait(1, self_obj, 666) == 0) then begin
+        if (obj_is_visible_flag(self_obj) == 0) then begin
             set_obj_visibility(self_obj, 0);
         end
     end
     if ((map_var(0) bwand 32) != 0) then begin
-        critter_add_trait(self_obj, 1, 6, 35);
-        critter_add_trait(self_obj, 1, 5, 180);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 35);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_PEASANT_NO_MOVE);
     end
     else begin
-        critter_add_trait(self_obj, 1, 6, 282);
-        critter_add_trait(self_obj, 1, 5, 29);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 282);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_IMPERIAL_SOLDIER_NO_TALKING);
     end
-    if (((map_var(0) bwand 16) != 0) and ((game_time - global_var(1001)) > (60 * (60 * 10) * 24))) then begin
+    if (((map_var(0) bwand 16) != 0) and ((game_time - global_var(GVAR_AQE_BRAD_WAITING_TIME)) > (60 * (60 * 10) * 24))) then begin
         destroy_object(self_obj);
     end
-    if (((map_var(0) bwand 32) != 0) and ((game_time - global_var(1001)) >= (60 * (60 * 10) * 8))) then begin
+    if (((map_var(0) bwand 32) != 0) and ((game_time - global_var(GVAR_AQE_BRAD_WAITING_TIME)) >= (60 * (60 * 10) * 8))) then begin
         destroy_object(self_obj);
     end
 end
 
 procedure timed_event_p_proc
 begin
-    if (global_var(997) == 4) then begin
+    if (global_var(GVAR_AQE_QUEST_B_SNEED) == 4) then begin
         if (fixed_param == 10) then begin
             if (fixed_param == 10) then begin
                 sequence := sequence + 1;
@@ -615,7 +615,7 @@ begin
             if (sequence2 == 2) then begin
                 if (not((local_var(6) bwand 64) == 0) or ((local_var(6) bwand 32768) != 0)) then begin
                     play_sfx("WAC1XXX1");
-                    create_object_sid(33555758, tile_num(dude_obj), 0, -1);
+                    create_object(33555758, dude_tile, 0);
                 end
                 if (not((local_var(6) bwand 128) == 0)) then begin
                     if (obj_is_carrying_obj_pid(self_obj, 16) >= 1) then begin
@@ -627,17 +627,17 @@ begin
                             restock_obj := obj_carrying_pid_obj(self_obj, 16);
                             if (obj_type(self_obj) == 1) then begin
                                 restock_obj := obj_carrying_pid_obj(self_obj, 16);
-                                if (critter_inven_obj(self_obj, 0) == restock_obj) then begin
-                                    if (obj_item_subtype(critter_inven_obj(self_obj, 0)) == 0) then begin
-                                        restock_obj := critter_inven_obj(self_obj, 0);
+                                if (critter_inven_obj(self_obj, INVEN_TYPE_WORN) == restock_obj) then begin
+                                    if (obj_item_subtype(critter_inven_obj(self_obj, INVEN_TYPE_WORN)) == 0) then begin
+                                        restock_obj := critter_inven_obj(self_obj, INVEN_TYPE_WORN);
                                         debug_msg("armour pid == " + obj_pid(restock_obj));
                                         rm_obj_from_inven(self_obj, restock_obj);
                                         add_obj_to_inven(self_obj, restock_obj);
                                     end
                                 end
                                 else begin
-                                    if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                        metarule(43, self_obj);
+                                    if ((critter_inven_obj(self_obj, INVEN_TYPE_LEFT_HAND) == restock_obj) or (critter_inven_obj(self_obj, INVEN_TYPE_RIGHT_HAND) == restock_obj)) then begin
+                                        inven_unwield(self_obj);
                                     end
                                 end
                             end
@@ -658,7 +658,7 @@ begin
                         restock_obj := 0;
                         restock_amt := 0;
                     end
-                    create_object_sid(16, 16523, 0, -1);
+                    create_object(16, 16523, 0);
                 end
                 if ((local_var(6) bwand 65536) != 0) then begin
                     if (obj_is_carrying_obj_pid(self_obj, 16) >= 1) then begin
@@ -670,17 +670,17 @@ begin
                             restock_obj := obj_carrying_pid_obj(self_obj, 16);
                             if (obj_type(self_obj) == 1) then begin
                                 restock_obj := obj_carrying_pid_obj(self_obj, 16);
-                                if (critter_inven_obj(self_obj, 0) == restock_obj) then begin
-                                    if (obj_item_subtype(critter_inven_obj(self_obj, 0)) == 0) then begin
-                                        restock_obj := critter_inven_obj(self_obj, 0);
+                                if (critter_inven_obj(self_obj, INVEN_TYPE_WORN) == restock_obj) then begin
+                                    if (obj_item_subtype(critter_inven_obj(self_obj, INVEN_TYPE_WORN)) == 0) then begin
+                                        restock_obj := critter_inven_obj(self_obj, INVEN_TYPE_WORN);
                                         debug_msg("armour pid == " + obj_pid(restock_obj));
                                         rm_obj_from_inven(self_obj, restock_obj);
                                         add_obj_to_inven(self_obj, restock_obj);
                                     end
                                 end
                                 else begin
-                                    if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                        metarule(43, self_obj);
+                                    if ((critter_inven_obj(self_obj, INVEN_TYPE_LEFT_HAND) == restock_obj) or (critter_inven_obj(self_obj, INVEN_TYPE_RIGHT_HAND) == restock_obj)) then begin
+                                        inven_unwield(self_obj);
                                     end
                                 end
                             end
@@ -701,7 +701,7 @@ begin
                         restock_obj := 0;
                         restock_amt := 0;
                     end
-                    restock_obj := create_object_sid(16, 0, 0, -1);
+                    restock_obj := create_object(16, 0, 0);
                     add_mult_objs_to_inven(dude_obj, restock_obj, 1);
                     wield_obj_critter(dude_obj, obj_carrying_pid_obj(dude_obj, 16));
                 end
@@ -736,18 +736,18 @@ begin
                 if ((map_var(0) bwand 64) != 0) then begin
                     set_obj_visibility(self_obj, 1);
                     if ((local_var(6) bwand 16384) != 0) then begin
-                        Fake_Brad := create_object_sid(obj_pid(self_obj), 16522, 0, 111);
+                        Fake_Brad := create_object_sid(obj_pid(self_obj), 16522, 0, NAME);
                         move_obj_inven_to_obj(self_obj, Fake_Brad);
                     end
                     else begin
-                        Fake_Brad := create_object_sid(16778041, 16522, 0, 111);
+                        Fake_Brad := create_object_sid(16778041, 16522, 0, NAME);
                     end
                     anim(Fake_Brad, 1000, 3);
                     reg_anim_func(2, Fake_Brad);
                     reg_anim_func(1, 1);
                     reg_anim_animate(Fake_Brad, 34, -1);
                     reg_anim_func(3, 0);
-                    critter_heal(Fake_Brad, -get_critter_stat(self_obj, 7));
+                    critter_heal(Fake_Brad, -get_critter_stat(self_obj, STAT_max_hp));
                     set_local_var(6, local_var(6) bwor 512);
                 end
                 if ((map_var(0) bwand 32) != 0) then begin
@@ -755,8 +755,8 @@ begin
                     reg_anim_func(1, 1);
                     reg_anim_animate(self_obj, 20, -1);
                     reg_anim_func(3, 0);
-                    critter_add_trait(self_obj, 1, 6, 35);
-                    critter_add_trait(self_obj, 1, 5, 180);
+                    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 35);
+                    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_PEASANT_NO_MOVE);
                     set_local_var(6, local_var(6) bwor 1024);
                 end
                 add_timer_event(self_obj, game_ticks(2), 50);
@@ -770,7 +770,7 @@ begin
                 if ((map_var(0) bwand 64) != 0) then begin
                     dialogue_system_enter;
                 end
-                if (((map_var(0) bwand 32) != 0) and ((party_member_obj(16777814) != 0) and ((global_var(910) bwand 128) == 0))) then begin
+                if (((map_var(0) bwand 32) != 0) and ((party_member_obj(16777814) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 128) == 0))) then begin
                     add_timer_event(party_member_obj(16777814), 1, 25);
                 end
             end
@@ -780,10 +780,10 @@ begin
         destroy_object(self_obj);
     end
     if (fixed_param == 70) then begin
-        critter_attempt_placement(self_obj, tile_num_in_direction(tile_num(dude_obj), (has_trait(1, dude_obj, 10) + 3) % 6, 2), elevation(dude_obj));
+        critter_attempt_placement(self_obj, tile_num_in_direction(dude_tile, (obj_get_rot(dude_obj) + 3) % 6, 2), dude_elevation);
     end
-    if ((fixed_param == 99) and (has_trait(1, self_obj, 666) == 1) and not(combat_is_initialized)) then begin
-        float_msg(self_obj, message_str(111, 314), 2);
+    if ((fixed_param == 99) and (obj_is_visible_flag(self_obj) == 1) and not(combat_is_initialized)) then begin
+        float_msg(self_obj, mstr(314), FLOAT_MSG_RED);
     end
 end
 
@@ -795,7 +795,7 @@ end
 procedure Node998_Brad_wins
 begin
     set_local_var(4, 1);
-    set_global_var(998, 1);
+    set_global_var(GVAR_AQE_ENEMY_EMPEROR, 1);
 end
 
 procedure Node999
@@ -806,24 +806,24 @@ end
 
 procedure Node999_Sneed_to_party
 begin
-    set_global_var(997, 2);
-    set_global_var(904, global_var(904) bwor 2097152);
-    brad_partym := create_object_sid(16778026, tile_num(self_obj), elevation(self_obj), 118);
-    anim(brad_partym, 1000, has_trait(1, self_obj, 10));
-    restock_obj := create_object_sid(16, 0, 0, -1);
+    set_global_var(GVAR_AQE_QUEST_B_SNEED, 2);
+    set_global_var(GVAR_AQE_ALBUQ_EMPEROR_EVENTS, global_var(GVAR_AQE_ALBUQ_EMPEROR_EVENTS) bwor 2097152);
+    brad_partym := create_object_sid(16778026, tile_num(self_obj), elevation(self_obj), SCRIPT_ABRADSPT);
+    anim(brad_partym, 1000, obj_get_rot(self_obj));
+    restock_obj := create_object(16, 0, 0);
     add_mult_objs_to_inven(brad_partym, restock_obj, 1);
     wield_obj_critter(brad_partym, obj_carrying_pid_obj(brad_partym, 16));
-    critter_add_trait(brad_partym, 1, 6, 0);
+    critter_add_trait(brad_partym, TRAIT_OBJECT, OBJECT_TEAM_NUM, 0);
     add_timer_event(brad_partym, game_ticks(random(10, 20)), 1);
 end
 
 procedure Node999_strip_brad
 begin
     set_map_var(0, map_var(0) bwor 16);
-    set_global_var(1001, game_time);
-    restock_obj := create_object_sid(16, 0, 0, -1);
+    set_global_var(GVAR_AQE_BRAD_WAITING_TIME, game_time);
+    restock_obj := create_object(16, 0, 0);
     add_mult_objs_to_inven(dude_obj, restock_obj, 1);
-    restock_obj := create_object_sid(579, 0, 0, -1);
+    restock_obj := create_object(579, 0, 0);
     add_mult_objs_to_inven(dude_obj, restock_obj, 1);
     if (obj_is_carrying_obj_pid(self_obj, 579) >= 1) then begin
         removed_qty := obj_is_carrying_obj_pid(self_obj, 579);
@@ -834,17 +834,17 @@ begin
             restock_obj := obj_carrying_pid_obj(self_obj, 579);
             if (obj_type(self_obj) == 1) then begin
                 restock_obj := obj_carrying_pid_obj(self_obj, 579);
-                if (critter_inven_obj(self_obj, 0) == restock_obj) then begin
-                    if (obj_item_subtype(critter_inven_obj(self_obj, 0)) == 0) then begin
-                        restock_obj := critter_inven_obj(self_obj, 0);
+                if (critter_inven_obj(self_obj, INVEN_TYPE_WORN) == restock_obj) then begin
+                    if (obj_item_subtype(critter_inven_obj(self_obj, INVEN_TYPE_WORN)) == 0) then begin
+                        restock_obj := critter_inven_obj(self_obj, INVEN_TYPE_WORN);
                         debug_msg("armour pid == " + obj_pid(restock_obj));
                         rm_obj_from_inven(self_obj, restock_obj);
                         add_obj_to_inven(self_obj, restock_obj);
                     end
                 end
                 else begin
-                    if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                        metarule(43, self_obj);
+                    if ((critter_inven_obj(self_obj, INVEN_TYPE_LEFT_HAND) == restock_obj) or (critter_inven_obj(self_obj, INVEN_TYPE_RIGHT_HAND) == restock_obj)) then begin
+                        inven_unwield(self_obj);
                     end
                 end
             end
@@ -874,17 +874,17 @@ begin
             restock_obj := obj_carrying_pid_obj(self_obj, 16);
             if (obj_type(self_obj) == 1) then begin
                 restock_obj := obj_carrying_pid_obj(self_obj, 16);
-                if (critter_inven_obj(self_obj, 0) == restock_obj) then begin
-                    if (obj_item_subtype(critter_inven_obj(self_obj, 0)) == 0) then begin
-                        restock_obj := critter_inven_obj(self_obj, 0);
+                if (critter_inven_obj(self_obj, INVEN_TYPE_WORN) == restock_obj) then begin
+                    if (obj_item_subtype(critter_inven_obj(self_obj, INVEN_TYPE_WORN)) == 0) then begin
+                        restock_obj := critter_inven_obj(self_obj, INVEN_TYPE_WORN);
                         debug_msg("armour pid == " + obj_pid(restock_obj));
                         rm_obj_from_inven(self_obj, restock_obj);
                         add_obj_to_inven(self_obj, restock_obj);
                     end
                 end
                 else begin
-                    if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                        metarule(43, self_obj);
+                    if ((critter_inven_obj(self_obj, INVEN_TYPE_LEFT_HAND) == restock_obj) or (critter_inven_obj(self_obj, INVEN_TYPE_RIGHT_HAND) == restock_obj)) then begin
+                        inven_unwield(self_obj);
                     end
                 end
             end
@@ -928,8 +928,8 @@ end
 procedure Node999_Brad_uncosious
 begin
     set_map_var(0, map_var(0) bwor 32);
-    set_global_var(904, global_var(904) bwor 8388608);
-    set_global_var(1001, game_time);
+    set_global_var(GVAR_AQE_ALBUQ_EMPEROR_EVENTS, global_var(GVAR_AQE_ALBUQ_EMPEROR_EVENTS) bwor 8388608);
+    set_global_var(GVAR_AQE_BRAD_WAITING_TIME, game_time);
     Timer_Event := 1;
 end
 
@@ -937,49 +937,49 @@ procedure Node999_Brad_execute
 begin
     set_map_var(0, map_var(0) bwor 64);
     Timer_Event := 1;
-    set_global_var(0, global_var(0) - 10);
-    set_global_var(37, 0);
-    set_global_var(38, 0);
-    set_global_var(39, 0);
-    set_global_var(40, 0);
-    set_global_var(41, 0);
-    set_global_var(42, 0);
-    set_global_var(43, 0);
-    set_global_var(44, 0);
-    set_global_var(45, 0);
-    if (global_var(0) >= 1000) then begin
-        set_global_var(37, 1);
+    set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) - 10);
+    set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+    set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+    set_global_var(GVAR_KARMA_DEFENDER, 0);
+    set_global_var(GVAR_KARMA_WANDERER, 0);
+    set_global_var(GVAR_KARMA_BETRAYER, 0);
+    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+    set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+    set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+    if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
     end
     else begin
-        if (global_var(0) >= 750) then begin
-            set_global_var(38, 1);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+            set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
         end
         else begin
-            if (global_var(0) >= 500) then begin
-                set_global_var(39, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
             end
             else begin
-                if (global_var(0) >= 250) then begin
-                    set_global_var(40, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                    set_global_var(GVAR_KARMA_DEFENDER, 1);
                 end
                 else begin
-                    if (global_var(0) > -250) then begin
-                        set_global_var(41, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                        set_global_var(GVAR_KARMA_WANDERER, 1);
                     end
                     else begin
-                        if (global_var(0) > -500) then begin
-                            set_global_var(42, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                            set_global_var(GVAR_KARMA_BETRAYER, 1);
                         end
                         else begin
-                            if (global_var(0) > -750) then begin
-                                set_global_var(43, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                             end
                             else begin
-                                if (global_var(0) > -1000) then begin
-                                    set_global_var(44, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                    set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                 end
                                 else begin
-                                    set_global_var(45, 1);
+                                    set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                 end
                             end
                         end
@@ -999,51 +999,51 @@ end
 
 procedure Node999_Quest_ended
 begin
-    set_global_var(997, 4);
-    set_global_var(0, global_var(0) + 15);
+    set_global_var(GVAR_AQE_QUEST_B_SNEED, 4);
+    set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + 15);
     debug_msg("Player gains " + 15 + " Karma Points.");
-    set_global_var(37, 0);
-    set_global_var(38, 0);
-    set_global_var(39, 0);
-    set_global_var(40, 0);
-    set_global_var(41, 0);
-    set_global_var(42, 0);
-    set_global_var(43, 0);
-    set_global_var(44, 0);
-    set_global_var(45, 0);
-    if (global_var(0) >= 1000) then begin
-        set_global_var(37, 1);
+    set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+    set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+    set_global_var(GVAR_KARMA_DEFENDER, 0);
+    set_global_var(GVAR_KARMA_WANDERER, 0);
+    set_global_var(GVAR_KARMA_BETRAYER, 0);
+    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+    set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+    set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+    if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
     end
     else begin
-        if (global_var(0) >= 750) then begin
-            set_global_var(38, 1);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+            set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
         end
         else begin
-            if (global_var(0) >= 500) then begin
-                set_global_var(39, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
             end
             else begin
-                if (global_var(0) >= 250) then begin
-                    set_global_var(40, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                    set_global_var(GVAR_KARMA_DEFENDER, 1);
                 end
                 else begin
-                    if (global_var(0) > -250) then begin
-                        set_global_var(41, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                        set_global_var(GVAR_KARMA_WANDERER, 1);
                     end
                     else begin
-                        if (global_var(0) > -500) then begin
-                            set_global_var(42, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                            set_global_var(GVAR_KARMA_BETRAYER, 1);
                         end
                         else begin
-                            if (global_var(0) > -750) then begin
-                                set_global_var(43, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                             end
                             else begin
-                                if (global_var(0) > -1000) then begin
-                                    set_global_var(44, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                    set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                 end
                                 else begin
-                                    set_global_var(45, 1);
+                                    set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                 end
                             end
                         end
@@ -1052,18 +1052,18 @@ begin
             end
         end
     end
-    set_global_var(53, global_var(53) + 3);
+    set_global_var(GVAR_TOWN_REP_EMPORER_CITY, global_var(GVAR_TOWN_REP_EMPORER_CITY) + 3);
     give_exp_points(1000);
-    display_msg(message_str(14, 100) + 1000 + message_str(14, 101));
-    if ((party_member_obj(16777814) != 0) and ((global_var(910) bwand 128) == 0)) then begin
-        set_global_var(1019, global_var(1019) bwor --2147483648);
+    display_msg(g_mstr(100) + 1000 + g_mstr(101));
+    if ((party_member_obj(16777814) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 128) == 0)) then begin
+        set_global_var(GVAR_PTR_LYSTRA_REACTIONS_1, global_var(GVAR_PTR_LYSTRA_REACTIONS_1) bwor --2147483648);
     end
 end
 
 procedure Node999_Killed
 begin
     Wearing_Pistol := 0;
-    if ((party_member_obj(16777814) != 0) and ((global_var(910) bwand 128) == 0)) then begin
+    if ((party_member_obj(16777814) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 128) == 0)) then begin
         add_timer_event(party_member_obj(16777814), 1, 25);
     end
 end
@@ -1083,423 +1083,422 @@ end
 procedure Node500
 begin
     if ((local_var(6) bwand 32) == 0) then begin
-        if ((obj_item_subtype(critter_inven_obj(dude_obj, 1)) == 3) or (obj_item_subtype(critter_inven_obj(dude_obj, 2)) == 3)) then begin
-            gsay_reply(111, 501);
+        if ((obj_item_subtype(critter_inven_obj(dude_obj, INVEN_TYPE_RIGHT_HAND)) == 3) or (obj_item_subtype(critter_inven_obj(dude_obj, INVEN_TYPE_LEFT_HAND)) == 3)) then begin
+            Reply(501);
         end
         else begin
-            gsay_reply(111, 500);
+            Reply(500);
         end
     end
     else begin
-        if ((obj_item_subtype(critter_inven_obj(dude_obj, 1)) == 3) or (obj_item_subtype(critter_inven_obj(dude_obj, 2)) == 3)) then begin
-            gsay_reply(111, 506);
+        if ((obj_item_subtype(critter_inven_obj(dude_obj, INVEN_TYPE_RIGHT_HAND)) == 3) or (obj_item_subtype(critter_inven_obj(dude_obj, INVEN_TYPE_LEFT_HAND)) == 3)) then begin
+            Reply(506);
         end
         else begin
-            gsay_reply(111, 505);
+            Reply(505);
         end
     end
-    if (global_var(999) == 2) then begin
-        giq_option(4, 111, 503, Node600, 50);
+    if (global_var(GVAR_AQE_QUEST_COLE) == 2) then begin
+        giq_option(4, NAME, 503, Node600, NEUTRAL_REACTION);
     end
     else begin
-        giq_option(4, 111, 502, Node550, 50);
+        giq_option(4, NAME, 502, Node550, NEUTRAL_REACTION);
     end
-    giq_option(-3, 111, 504, Node1750, 50);
+    giq_option(-3, NAME, 504, Node1750, NEUTRAL_REACTION);
 end
 
 procedure Node550
 begin
-    gsay_reply(111, 550);
+    Reply(550);
     set_local_var(6, local_var(6) bwor 32);
-    giq_option(1, 111, 551, Node999, 50);
+    giq_option(1, NAME, 551, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node600
 begin
-    gsay_reply(111, 600);
+    Reply(600);
     set_map_var(0, map_var(0) bwor 4);
     set_local_var(5, 1);
-    set_global_var(1019, global_var(1019) bwand (-1 - 536870912));
-    if ((party_member_obj(16777814) != 0) and ((global_var(910) bwand 128) == 0)) then begin
-        set_global_var(1019, global_var(1019) bwor 1073741824);
+    set_global_var(GVAR_PTR_LYSTRA_REACTIONS_1, global_var(GVAR_PTR_LYSTRA_REACTIONS_1) bwand (-1 - 536870912));
+    if ((party_member_obj(16777814) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 128) == 0)) then begin
+        set_global_var(GVAR_PTR_LYSTRA_REACTIONS_1, global_var(GVAR_PTR_LYSTRA_REACTIONS_1) bwor 1073741824);
     end
     if ((local_var(6) bwand 1) == 0) then begin
-        giq_option(4, 111, 601, Node650, 50);
+        giq_option(4, NAME, 601, Node650, NEUTRAL_REACTION);
     end
     if ((local_var(6) bwand 2) == 0) then begin
-        giq_option(4, 111, 602, Node700, 50);
+        giq_option(4, NAME, 602, Node700, NEUTRAL_REACTION);
     end
     if ((local_var(6) bwand 4) == 0) then begin
-        giq_option(4, 111, 603, Node750, 50);
+        giq_option(4, NAME, 603, Node750, NEUTRAL_REACTION);
     end
-    if (get_critter_stat(dude_obj, 34) == 0) then begin
-        if ((global_var(904) bwand 131072) != 0) then begin
-            giq_option(4, 111, 604, Node800, 50);
+    if (dude_gender == 0) then begin
+        if ((global_var(GVAR_AQE_ALBUQ_EMPEROR_EVENTS) bwand 131072) != 0) then begin
+            giq_option(4, NAME, 604, Node800, NEUTRAL_REACTION);
         end
-        if ((global_var(904) bwand 262144) != 0) then begin
-            giq_option(4, 111, 605, Node800, 50);
+        if ((global_var(GVAR_AQE_ALBUQ_EMPEROR_EVENTS) bwand 262144) != 0) then begin
+            giq_option(4, NAME, 605, Node800, NEUTRAL_REACTION);
         end
     end
     else begin
-        if ((global_var(904) bwand 131072) != 0) then begin
-            giq_option(4, 111, 606, Node800, 50);
+        if ((global_var(GVAR_AQE_ALBUQ_EMPEROR_EVENTS) bwand 131072) != 0) then begin
+            giq_option(4, NAME, 606, Node800, NEUTRAL_REACTION);
         end
-        if ((global_var(904) bwand 262144) != 0) then begin
-            giq_option(4, 111, 607, Node800, 50);
+        if ((global_var(GVAR_AQE_ALBUQ_EMPEROR_EVENTS) bwand 262144) != 0) then begin
+            giq_option(4, NAME, 607, Node800, NEUTRAL_REACTION);
         end
     end
-    giq_option(4, 111, 608, Node999, 50);
+    giq_option(4, NAME, 608, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node650
 begin
-    gsay_reply(111, 650);
+    Reply(650);
     set_local_var(6, local_var(6) bwor 1);
     if ((local_var(6) bwand 2) == 0) then begin
-        giq_option(4, 111, 651, Node700, 50);
+        giq_option(4, NAME, 651, Node700, NEUTRAL_REACTION);
     end
     if ((local_var(6) bwand 4) == 0) then begin
-        giq_option(4, 111, 652, Node750, 50);
+        giq_option(4, NAME, 652, Node750, NEUTRAL_REACTION);
     end
-    if (get_critter_stat(dude_obj, 34) == 0) then begin
-        if ((global_var(904) bwand 131072) != 0) then begin
-            giq_option(4, 111, 653, Node800, 50);
+    if (dude_gender == 0) then begin
+        if ((global_var(GVAR_AQE_ALBUQ_EMPEROR_EVENTS) bwand 131072) != 0) then begin
+            giq_option(4, NAME, 653, Node800, NEUTRAL_REACTION);
         end
-        if ((global_var(904) bwand 262144) != 0) then begin
-            giq_option(4, 111, 654, Node800, 50);
+        if ((global_var(GVAR_AQE_ALBUQ_EMPEROR_EVENTS) bwand 262144) != 0) then begin
+            giq_option(4, NAME, 654, Node800, NEUTRAL_REACTION);
         end
     end
     else begin
-        if ((global_var(904) bwand 131072) != 0) then begin
-            giq_option(4, 111, 655, Node800, 50);
+        if ((global_var(GVAR_AQE_ALBUQ_EMPEROR_EVENTS) bwand 131072) != 0) then begin
+            giq_option(4, NAME, 655, Node800, NEUTRAL_REACTION);
         end
-        if ((global_var(904) bwand 262144) != 0) then begin
-            giq_option(4, 111, 656, Node800, 50);
+        if ((global_var(GVAR_AQE_ALBUQ_EMPEROR_EVENTS) bwand 262144) != 0) then begin
+            giq_option(4, NAME, 656, Node800, NEUTRAL_REACTION);
         end
     end
-    giq_option(4, 111, 657, Node999, 50);
+    giq_option(4, NAME, 657, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node700
 begin
-    gsay_reply(111, 700);
+    Reply(700);
     set_local_var(6, local_var(6) bwor 2);
     if ((local_var(6) bwand 1) == 0) then begin
-        giq_option(4, 111, 701, Node650, 50);
+        giq_option(4, NAME, 701, Node650, NEUTRAL_REACTION);
     end
     if ((local_var(6) bwand 4) == 0) then begin
-        giq_option(4, 111, 702, Node750, 50);
+        giq_option(4, NAME, 702, Node750, NEUTRAL_REACTION);
     end
-    if (get_critter_stat(dude_obj, 34) == 0) then begin
-        if ((global_var(904) bwand 131072) != 0) then begin
-            giq_option(4, 111, 703, Node800, 50);
+    if (dude_gender == 0) then begin
+        if ((global_var(GVAR_AQE_ALBUQ_EMPEROR_EVENTS) bwand 131072) != 0) then begin
+            giq_option(4, NAME, 703, Node800, NEUTRAL_REACTION);
         end
-        if ((global_var(904) bwand 262144) != 0) then begin
-            giq_option(4, 111, 704, Node800, 50);
+        if ((global_var(GVAR_AQE_ALBUQ_EMPEROR_EVENTS) bwand 262144) != 0) then begin
+            giq_option(4, NAME, 704, Node800, NEUTRAL_REACTION);
         end
     end
     else begin
-        if ((global_var(904) bwand 131072) != 0) then begin
-            giq_option(4, 111, 705, Node800, 50);
+        if ((global_var(GVAR_AQE_ALBUQ_EMPEROR_EVENTS) bwand 131072) != 0) then begin
+            giq_option(4, NAME, 705, Node800, NEUTRAL_REACTION);
         end
-        if ((global_var(904) bwand 262144) != 0) then begin
-            giq_option(4, 111, 706, Node800, 50);
+        if ((global_var(GVAR_AQE_ALBUQ_EMPEROR_EVENTS) bwand 262144) != 0) then begin
+            giq_option(4, NAME, 706, Node800, NEUTRAL_REACTION);
         end
     end
-    giq_option(4, 111, 707, Node999, 50);
+    giq_option(4, NAME, 707, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node750
 begin
-    gsay_reply(111, 750);
+    Reply(750);
     set_local_var(6, local_var(6) bwor 4);
     if ((local_var(6) bwand 1) == 0) then begin
-        giq_option(4, 111, 751, Node650, 50);
+        giq_option(4, NAME, 751, Node650, NEUTRAL_REACTION);
     end
     if ((local_var(6) bwand 2) == 0) then begin
-        giq_option(4, 111, 752, Node700, 50);
+        giq_option(4, NAME, 752, Node700, NEUTRAL_REACTION);
     end
-    if (get_critter_stat(dude_obj, 34) == 0) then begin
-        if ((global_var(904) bwand 131072) != 0) then begin
-            giq_option(4, 111, 753, Node800, 50);
+    if (dude_gender == 0) then begin
+        if ((global_var(GVAR_AQE_ALBUQ_EMPEROR_EVENTS) bwand 131072) != 0) then begin
+            giq_option(4, NAME, 753, Node800, NEUTRAL_REACTION);
         end
-        if ((global_var(904) bwand 262144) != 0) then begin
-            giq_option(4, 111, 754, Node800, 50);
+        if ((global_var(GVAR_AQE_ALBUQ_EMPEROR_EVENTS) bwand 262144) != 0) then begin
+            giq_option(4, NAME, 754, Node800, NEUTRAL_REACTION);
         end
     end
     else begin
-        if ((global_var(904) bwand 131072) != 0) then begin
-            giq_option(4, 111, 755, Node800, 50);
+        if ((global_var(GVAR_AQE_ALBUQ_EMPEROR_EVENTS) bwand 131072) != 0) then begin
+            giq_option(4, NAME, 755, Node800, NEUTRAL_REACTION);
         end
-        if ((global_var(904) bwand 262144) != 0) then begin
-            giq_option(4, 111, 756, Node800, 50);
+        if ((global_var(GVAR_AQE_ALBUQ_EMPEROR_EVENTS) bwand 262144) != 0) then begin
+            giq_option(4, NAME, 756, Node800, NEUTRAL_REACTION);
         end
     end
-    giq_option(4, 111, 757, Node999, 50);
+    giq_option(4, NAME, 757, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node800
 begin
-    gsay_reply(111, 800);
+    Reply(800);
     set_local_var(6, local_var(6) bwor 8);
-    if (get_critter_stat(dude_obj, 34) == 0) then begin
-        giq_option(4, 111, 801, Node850, 50);
+    if (dude_gender == 0) then begin
+        giq_option(4, NAME, 801, Node850, NEUTRAL_REACTION);
     end
-    if (get_critter_stat(dude_obj, 34) == 1) then begin
-        giq_option(4, 111, 802, Node850, 50);
+    if (dude_gender == 1) then begin
+        giq_option(4, NAME, 802, Node850, NEUTRAL_REACTION);
     end
-    if (get_critter_stat(dude_obj, 34) == 0) then begin
-        giq_option(4, 111, 803, Node900, 50);
+    if (dude_gender == 0) then begin
+        giq_option(4, NAME, 803, Node900, NEUTRAL_REACTION);
     end
-    if (get_critter_stat(dude_obj, 34) == 1) then begin
-        giq_option(4, 111, 804, Node900, 50);
+    if (dude_gender == 1) then begin
+        giq_option(4, NAME, 804, Node900, NEUTRAL_REACTION);
     end
 end
 
 procedure Node850
 begin
-    gsay_reply(111, 850);
-    giq_option(4, 111, 851, Node999, 50);
-    if (get_critter_stat(dude_obj, 34) == 0) then begin
-        giq_option(4, 111, 803, Node900, 50);
+    Reply(850);
+    giq_option(4, NAME, 851, Node999, NEUTRAL_REACTION);
+    if (dude_gender == 0) then begin
+        giq_option(4, NAME, 803, Node900, NEUTRAL_REACTION);
     end
-    if (get_critter_stat(dude_obj, 34) == 1) then begin
-        giq_option(4, 111, 804, Node900, 50);
+    if (dude_gender == 1) then begin
+        giq_option(4, NAME, 804, Node900, NEUTRAL_REACTION);
     end
 end
 
 procedure Node900
 begin
-    gsay_reply(111, 900);
-    giq_option(4, 111, 901, Node950, 50);
-    giq_option(4, 111, 902, Node1000, 50);
-    if (get_critter_stat(dude_obj, 34) == 0) then begin
-        giq_option(4, 111, 903, Node1050, 50);
+    Reply(900);
+    giq_option(4, NAME, 901, Node950, NEUTRAL_REACTION);
+    giq_option(4, NAME, 902, Node1000, NEUTRAL_REACTION);
+    if (dude_gender == 0) then begin
+        giq_option(4, NAME, 903, Node1050, NEUTRAL_REACTION);
     end
-    if (get_critter_stat(dude_obj, 34) == 1) then begin
-        giq_option(4, 111, 903, Node1100, 50);
+    if (dude_gender == 1) then begin
+        giq_option(4, NAME, 903, Node1100, NEUTRAL_REACTION);
     end
 end
 
 procedure Node950
 begin
-    gsay_reply(111, 950);
-    giq_option(4, 111, 951, Node1000, 50);
-    if (get_critter_stat(dude_obj, 34) == 0) then begin
-        giq_option(4, 111, 952, Node1050, 50);
+    Reply(950);
+    giq_option(4, NAME, 951, Node1000, NEUTRAL_REACTION);
+    if (dude_gender == 0) then begin
+        giq_option(4, NAME, 952, Node1050, NEUTRAL_REACTION);
     end
-    if (get_critter_stat(dude_obj, 34) == 1) then begin
-        giq_option(4, 111, 952, Node1100, 50);
+    if (dude_gender == 1) then begin
+        giq_option(4, NAME, 952, Node1100, NEUTRAL_REACTION);
     end
 end
 
 procedure Node1000
 begin
-    gsay_reply(111, 1000);
-    giq_option(4, 111, 1001, Node999_Sneed_to_party, 50);
+    Reply(1000);
+    giq_option(4, NAME, 1001, Node999_Sneed_to_party, NEUTRAL_REACTION);
 end
 
 procedure Node1050
 begin
-    gsay_reply(111, 1050);
+    Reply(1050);
     set_local_var(6, local_var(6) bwor 16);
-    giq_option(4, 111, 1051, Node999, 50);
+    giq_option(4, NAME, 1051, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node1100
 begin
-    gsay_reply(111, 1100);
+    Reply(1100);
     set_local_var(6, local_var(6) bwor 16);
-    giq_option(4, 111, 1101, Node999, 50);
+    giq_option(4, NAME, 1101, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node1150
 begin
-    if (global_var(997) != 3) then begin
-        gsay_reply(111, 1150);
+    if (global_var(GVAR_AQE_QUEST_B_SNEED) != 3) then begin
+        Reply(1150);
     end
     else begin
-        if (get_critter_stat(dude_obj, 34) == 0) then begin
-            gsay_reply(111, 1162);
+        if (dude_gender == 0) then begin
+            Reply(1162);
         end
         else begin
-            gsay_reply(111, 1163);
+            Reply(1163);
         end
     end
-    if (global_var(999) != 3) then begin
-        if (global_var(997) != 3) then begin
+    if (global_var(GVAR_AQE_QUEST_COLE) != 3) then begin
+        if (global_var(GVAR_AQE_QUEST_B_SNEED) != 3) then begin
             if ((local_var(6) bwand 8) == 0) then begin
-                if (get_critter_stat(dude_obj, 34) == 0) then begin
-                    if ((global_var(904) bwand 131072) != 0) then begin
-                        giq_option(4, 111, 1151, Node800, 50);
+                if (dude_gender == 0) then begin
+                    if ((global_var(GVAR_AQE_ALBUQ_EMPEROR_EVENTS) bwand 131072) != 0) then begin
+                        giq_option(4, NAME, 1151, Node800, NEUTRAL_REACTION);
                     end
-                    if ((global_var(904) bwand 262144) != 0) then begin
-                        giq_option(4, 111, 1152, Node800, 50);
+                    if ((global_var(GVAR_AQE_ALBUQ_EMPEROR_EVENTS) bwand 262144) != 0) then begin
+                        giq_option(4, NAME, 1152, Node800, NEUTRAL_REACTION);
                     end
                 end
                 else begin
-                    if ((global_var(904) bwand 131072) != 0) then begin
-                        giq_option(4, 111, 1153, Node800, 50);
+                    if ((global_var(GVAR_AQE_ALBUQ_EMPEROR_EVENTS) bwand 131072) != 0) then begin
+                        giq_option(4, NAME, 1153, Node800, NEUTRAL_REACTION);
                     end
-                    if ((global_var(904) bwand 262144) != 0) then begin
-                        giq_option(4, 111, 1154, Node800, 50);
+                    if ((global_var(GVAR_AQE_ALBUQ_EMPEROR_EVENTS) bwand 262144) != 0) then begin
+                        giq_option(4, NAME, 1154, Node800, NEUTRAL_REACTION);
                     end
                 end
             end
             if ((map_var(0) bwand 1) != 0) then begin
-                giq_option(4, 111, 1156, Node1250, 50);
+                giq_option(4, NAME, 1156, Node1250, NEUTRAL_REACTION);
                 if (Wearing_Pistol == 1) then begin
-                    if (get_critter_stat(dude_obj, 5) < 8) then begin
-                        giq_option(4, 111, 1157, Node999_Brad_wins, 50);
+                    if (get_critter_stat(dude_obj, STAT_ag) < 8) then begin
+                        giq_option(4, NAME, 1157, Node999_Brad_wins, NEUTRAL_REACTION);
                     end
                     else begin
-                        giq_option(4, 111, 1157, Node999_Brad_loose, 50);
+                        giq_option(4, NAME, 1157, Node999_Brad_loose, NEUTRAL_REACTION);
                     end
                 end
-                if (get_critter_stat(dude_obj, 0) >= 10) then begin
-                    giq_option(4, 111, 1164, Node999_Brad_execute_Punch, 50);
+                if (dude_strength >= 10) then begin
+                    giq_option(4, NAME, 1164, Node999_Brad_execute_Punch, NEUTRAL_REACTION);
                 end
                 else begin
-                    if (get_critter_stat(dude_obj, 0) > 6) then begin
-                        giq_option(4, 111, 1164, Node999_Brad_loose_punch, 50);
+                    if (dude_strength > 6) then begin
+                        giq_option(4, NAME, 1164, Node999_Brad_loose_punch, NEUTRAL_REACTION);
                     end
                     else begin
-                        giq_option(4, 111, 1164, Node999_Brad_wins_punch, 50);
+                        giq_option(4, NAME, 1164, Node999_Brad_wins_punch, NEUTRAL_REACTION);
                     end
                 end
             end
             if ((local_var(6) bwand 8) != 0) then begin
-                giq_option(4, 111, 1155, Node1200, 50);
+                giq_option(4, NAME, 1155, Node1200, NEUTRAL_REACTION);
             end
         end
-        if ((global_var(999) == 1) and ((global_var(904) bwand 4194304) != 0)) then begin
-            if (get_critter_stat(dude_obj, 34) == 0) then begin
-                giq_option(4, 111, 1158, Node1700, 50);
+        if ((global_var(GVAR_AQE_QUEST_COLE) == 1) and ((global_var(GVAR_AQE_ALBUQ_EMPEROR_EVENTS) bwand 4194304) != 0)) then begin
+            if (dude_gender == 0) then begin
+                giq_option(4, NAME, 1158, Node1700, NEUTRAL_REACTION);
             end
-            if (get_critter_stat(dude_obj, 34) == 1) then begin
-                giq_option(4, 111, 1159, Node1700, 50);
+            if (dude_gender == 1) then begin
+                giq_option(4, NAME, 1159, Node1700, NEUTRAL_REACTION);
             end
         end
     end
-    giq_option(4, 111, 1160, Node999, 50);
-    giq_option(-3, 111, 1161, Node999, 50);
+    giq_option(4, NAME, 1160, Node999, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 1161, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node1200
 begin
-    gsay_reply(111, 1200);
-    giq_option(4, 111, 1201, Node1000, 50);
+    Reply(1200);
+    giq_option(4, NAME, 1201, Node1000, NEUTRAL_REACTION);
 end
 
 procedure Node1250
 begin
-    gsay_reply(111, 1250);
-    if (has_skill(dude_obj, 14) > 100) then begin
-        if (get_critter_stat(dude_obj, 34) == 0) then begin
-            giq_option(4, 111, 1251, Node1300, 50);
+    Reply(1250);
+    if (dude_skill(SKILL_SPEECH) > 100) then begin
+        if (dude_gender == 0) then begin
+            giq_option(4, NAME, 1251, Node1300, NEUTRAL_REACTION);
         end
-        if (get_critter_stat(dude_obj, 34) == 1) then begin
-            giq_option(4, 111, 1252, Node1300, 50);
+        if (dude_gender == 1) then begin
+            giq_option(4, NAME, 1252, Node1300, NEUTRAL_REACTION);
         end
     end
-    giq_option(4, 111, 1253, Node999, 50);
+    giq_option(4, NAME, 1253, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node1300
 begin
-    gsay_reply(111, 1300);
-    giq_option(4, 111, 1301, Node999_strip_brad, 50);
+    Reply(1300);
+    giq_option(4, NAME, 1301, Node999_strip_brad, NEUTRAL_REACTION);
 end
 
 procedure Node1350
 begin
-    if (get_critter_stat(dude_obj, 34) == 0) then begin
-        gsay_reply(111, 1350);
+    if (dude_gender == 0) then begin
+        Reply(1350);
     end
-    if (get_critter_stat(dude_obj, 34) == 1) then begin
-        gsay_reply(111, 1351);
+    if (dude_gender == 1) then begin
+        Reply(1351);
     end
-    giq_option(4, 111, 1352, Node998_Brad_wins, 50);
+    giq_option(4, NAME, 1352, Node998_Brad_wins, NEUTRAL_REACTION);
 end
 
 procedure Node1400
 begin
-    if (get_critter_stat(dude_obj, 34) == 0) then begin
-        gsay_reply(111, 1400);
+    if (dude_gender == 0) then begin
+        Reply(1400);
     end
-    if (get_critter_stat(dude_obj, 34) == 1) then begin
-        gsay_reply(111, 1401);
+    if (dude_gender == 1) then begin
+        Reply(1401);
     end
-    giq_option(4, 111, 1402, Node999_drop_pistol, 50);
+    giq_option(4, NAME, 1402, Node999_drop_pistol, NEUTRAL_REACTION);
 end
 
 procedure Node1450
 begin
-    gsay_reply(111, 1450);
-    if (get_critter_stat(dude_obj, 34) == 0) then begin
-        giq_option(4, 111, 1451, Node1500, 50);
+    Reply(1450);
+    if (dude_gender == 0) then begin
+        giq_option(4, NAME, 1451, Node1500, NEUTRAL_REACTION);
     end
-    if (get_critter_stat(dude_obj, 34) == 1) then begin
-        giq_option(4, 111, 1452, Node1500, 50);
+    if (dude_gender == 1) then begin
+        giq_option(4, NAME, 1452, Node1500, NEUTRAL_REACTION);
     end
-    giq_option(4, 111, 1453, Node999_Brad_execute, 50);
+    giq_option(4, NAME, 1453, Node999_Brad_execute, NEUTRAL_REACTION);
 end
 
 procedure Node1500
 begin
-    if (get_critter_stat(dude_obj, 34) == 0) then begin
-        gsay_reply(111, 1500);
+    if (dude_gender == 0) then begin
+        Reply(1500);
     end
-    if (get_critter_stat(dude_obj, 34) == 1) then begin
-        gsay_reply(111, 1501);
+    if (dude_gender == 1) then begin
+        Reply(1501);
     end
-    giq_option(4, 111, 1502, Node999_Brad_uncosious, 50);
+    giq_option(4, NAME, 1502, Node999_Brad_uncosious, NEUTRAL_REACTION);
 end
 
 procedure Node1550
 begin
-    if (get_critter_stat(dude_obj, 34) == 0) then begin
-        gsay_reply(111, 1550);
+    if (dude_gender == 0) then begin
+        Reply(1550);
     end
-    if (get_critter_stat(dude_obj, 34) == 1) then begin
-        gsay_reply(111, 1551);
+    if (dude_gender == 1) then begin
+        Reply(1551);
     end
-    giq_option(4, 111, 1552, Node999_Killed, 50);
+    giq_option(4, NAME, 1552, Node999_Killed, NEUTRAL_REACTION);
 end
 
 procedure Node1700
 begin
-    if (get_critter_stat(dude_obj, 34) == 0) then begin
-        gsay_reply(111, 1700);
+    if (dude_gender == 0) then begin
+        Reply(1700);
     end
-    if (get_critter_stat(dude_obj, 34) == 1) then begin
-        gsay_reply(111, 1701);
+    if (dude_gender == 1) then begin
+        Reply(1701);
     end
-    giq_option(4, 111, 1702, Node999_Quest_ended, 50);
+    giq_option(4, NAME, 1702, Node999_Quest_ended, NEUTRAL_REACTION);
 end
 
 procedure Node1750
 begin
-    gsay_reply(111, 1750);
+    Reply(1750);
     set_local_var(5, 1);
-    giq_option(-3, 111, 1751, Node999, 50);
+    giq_option(-3, NAME, 1751, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node1800
 begin
-    gsay_reply(111, 1800 + (get_critter_stat(dude_obj, 34) == 1));
-    giq_option(4, 111, 1802, Node999_drop_pistol, 50);
+    Reply(1800 + (dude_gender == 1));
+    giq_option(4, NAME, 1802, Node999_drop_pistol, NEUTRAL_REACTION);
 end
 
 procedure Node1850
 begin
-    gsay_reply(111, 1850);
-    giq_option(4, 111, 1851, Node998_Brad_wins, 50);
+    Reply(1850);
+    giq_option(4, NAME, 1851, Node998_Brad_wins, NEUTRAL_REACTION);
 end
 
 procedure Node1900
 begin
-    gsay_reply(111, 1900 + (get_critter_stat(dude_obj, 34) == 1));
-    giq_option(4, 111, 1902, Node999_Killed, 50);
+    Reply(1900 + (dude_gender == 1));
+    giq_option(4, NAME, 1902, Node999_Killed, NEUTRAL_REACTION);
 end
-
