diff --git a/Mods/Resurrection/Original/ACHSHELF.ssl b/Mods/Resurrection/ReDefine/ACHSHELF.ssl
index 5080cfd..67b5fa5 100644
--- a/Mods/Resurrection/Original/ACHSHELF.ssl
+++ b/Mods/Resurrection/ReDefine/ACHSHELF.ssl
@@ -68,12 +68,12 @@ begin
             return 1;
         end
     end
-    if ((party_member_obj(16777779) != 0) and ((global_var(910) bwand 32) == 0)) then begin
+    if ((party_member_obj(16777779) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 32) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777779)) <= 5) then begin
             return 1;
         end
     end
-    if ((party_member_obj(16777811) != 0) and ((global_var(910) bwand 64) == 0)) then begin
+    if ((party_member_obj(16777811) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 64) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777811)) <= 5) then begin
             return 1;
         end
@@ -88,7 +88,7 @@ begin
             return 1;
         end
     end
-    if ((party_member_obj(16777814) != 0) and ((global_var(910) bwand 128) == 0)) then begin
+    if ((party_member_obj(16777814) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 128) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777814)) <= 5) then begin
             return 1;
         end
@@ -170,17 +170,17 @@ begin
         GVar16 := obj_carrying_pid_obj(self_obj, 558);
         if (obj_type(self_obj) == 1) then begin
             GVar16 := obj_carrying_pid_obj(self_obj, 558);
-            if (critter_inven_obj(self_obj, 0) == GVar16) then begin
-                if (obj_item_subtype(critter_inven_obj(self_obj, 0)) == 0) then begin
-                    GVar16 := critter_inven_obj(self_obj, 0);
+            if (critter_inven_obj(self_obj, INVEN_TYPE_WORN) == GVar16) then begin
+                if (obj_item_subtype(critter_inven_obj(self_obj, INVEN_TYPE_WORN)) == 0) then begin
+                    GVar16 := critter_inven_obj(self_obj, INVEN_TYPE_WORN);
                     debug_msg("armour pid == " + obj_pid(GVar16));
                     rm_obj_from_inven(self_obj, GVar16);
                     add_obj_to_inven(self_obj, GVar16);
                 end
             end
             else begin
-                if ((critter_inven_obj(self_obj, 2) == GVar16) or (critter_inven_obj(self_obj, 1) == GVar16)) then begin
-                    metarule(43, self_obj);
+                if ((critter_inven_obj(self_obj, INVEN_TYPE_LEFT_HAND) == GVar16) or (critter_inven_obj(self_obj, INVEN_TYPE_RIGHT_HAND) == GVar16)) then begin
+                    inven_unwield(self_obj);
                 end
             end
         end
@@ -201,4 +201,3 @@ begin
     GVar16 := 0;
     GVar15 := 0;
 end
-
