diff --git a/Mods/Resurrection/Original/ACHUCKSH.ssl b/Mods/Resurrection/ReDefine/ACHUCKSH.ssl
index 1e813e7..4b1059d 100644
--- a/Mods/Resurrection/Original/ACHUCKSH.ssl
+++ b/Mods/Resurrection/ReDefine/ACHUCKSH.ssl
@@ -63,12 +63,12 @@ begin
             return 1;
         end
     end
-    if ((party_member_obj(16777779) != 0) and ((global_var(910) bwand 32) == 0)) then begin
+    if ((party_member_obj(16777779) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 32) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777779)) <= 5) then begin
             return 1;
         end
     end
-    if ((party_member_obj(16777811) != 0) and ((global_var(910) bwand 64) == 0)) then begin
+    if ((party_member_obj(16777811) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 64) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777811)) <= 5) then begin
             return 1;
         end
@@ -83,7 +83,7 @@ begin
             return 1;
         end
     end
-    if ((party_member_obj(16777814) != 0) and ((global_var(910) bwand 128) == 0)) then begin
+    if ((party_member_obj(16777814) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 128) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777814)) <= 5) then begin
             return 1;
         end
@@ -145,10 +145,10 @@ begin
             if (anim_busy(self_obj) == 0) then begin
                 if ((tile_distance_objs(self_obj, dude_obj) < 8) or obj_can_see_obj(self_obj, dude_obj)) then begin
                     if (anim_busy(self_obj) == 0) then begin
-                        global_temp := rotation_to_tile(tile_num(dude_obj), tile_num(self_obj));
+                        global_temp := rotation_to_tile(dude_tile, tile_num(self_obj));
                         animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(self_obj), global_temp, random(3, 10)), 1);
                         global_temp := (global_temp + 1) % 6;
-                        while ((anim_busy(self_obj) == 0) and (global_temp != rotation_to_tile(tile_num(dude_obj), tile_num(self_obj)))) do begin
+                        while ((anim_busy(self_obj) == 0) and (global_temp != rotation_to_tile(dude_tile, tile_num(self_obj)))) do begin
                             debug_msg("flee loop: rot == " + global_temp);
                             animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(self_obj), global_temp, random(3, 10)), 1);
                             global_temp := (global_temp + 1) % 6;
@@ -158,7 +158,7 @@ begin
             end
         end
         else begin
-            attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+            attack(dude_obj);
         end
     end
     if ((local_var(7) == 27272) and (local_var(5) == 0) and obj_can_see_obj(self_obj, dude_obj)) then begin
@@ -167,19 +167,19 @@ begin
     if (((map_var(0) bwand 512) != 0) and (local_var(7) != 27272)) then begin
         destroy_object(self_obj);
     end
-    if ((global_var(865) bwand 16777216) != 0) then begin
+    if ((global_var(GVAR_AQ_ALBUQ_EVENTS) bwand 16777216) != 0) then begin
         blood_type := random(1, 4);
         if (blood_type == 1) then begin
-            create_object_sid(33555755, tile_num(self_obj), elevation(self_obj), -1);
+            create_object(33555755, tile_num(self_obj), elevation(self_obj));
         end
         if (blood_type == 2) then begin
-            create_object_sid(33555756, tile_num(self_obj), elevation(self_obj), -1);
+            create_object(33555756, tile_num(self_obj), elevation(self_obj));
         end
         if (blood_type == 3) then begin
-            create_object_sid(33555757, tile_num(self_obj), elevation(self_obj), -1);
+            create_object(33555757, tile_num(self_obj), elevation(self_obj));
         end
         if (blood_type == 4) then begin
-            create_object_sid(33555758, tile_num(self_obj), elevation(self_obj), -1);
+            create_object(33555758, tile_num(self_obj), elevation(self_obj));
         end
         destroy_object(self_obj);
     end
@@ -200,66 +200,66 @@ begin
         set_local_var(5, 1);
     end
     else begin
-        animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(self_obj), rotation_to_tile(tile_num(dude_obj), tile_num(self_obj)), 1), 0);
+        animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(self_obj), rotation_to_tile(dude_tile, tile_num(self_obj)), 1), 0);
     end
 end
 
 procedure destroy_p_proc
 begin
     if (source_obj == dude_obj) then begin
-        set_global_var(5, global_var(5) + 1);
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(4) > (3 * global_var(5))) or (global_var(2) == 1)) and (global_var(1) == 0)) then begin
-            set_global_var(2, 1);
-            set_global_var(3, 0);
+        set_global_var(GVAR_GOOD_MONSTER, global_var(GVAR_GOOD_MONSTER) + 1);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(4) > (3 * global_var(GVAR_GOOD_MONSTER))) or (global_var(GVAR_CHAMPION_REPUTATION) == 1)) and (global_var(GVAR_CHILDKILLER_REPUTATION) == 0)) then begin
+            set_global_var(GVAR_CHAMPION_REPUTATION, 1);
+            set_global_var(GVAR_BERSERKER_REPUTATION, 0);
         end
-        if (metarule(51, self_obj) == 2) then begin
+        if (critter_kill_type(self_obj) == KILL_TYPE_children_kills) then begin
             if (source_obj == dude_obj) then begin
-                set_global_var(2, 0);
-                set_global_var(1, global_var(1) + 1);
-                set_global_var(0, global_var(0) + -15);
+                set_global_var(GVAR_CHAMPION_REPUTATION, 0);
+                set_global_var(GVAR_CHILDKILLER_REPUTATION, global_var(GVAR_CHILDKILLER_REPUTATION) + 1);
+                set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + -15);
                 debug_msg("Player gains " + -15 + " Karma Points.");
-                set_global_var(37, 0);
-                set_global_var(38, 0);
-                set_global_var(39, 0);
-                set_global_var(40, 0);
-                set_global_var(41, 0);
-                set_global_var(42, 0);
-                set_global_var(43, 0);
-                set_global_var(44, 0);
-                set_global_var(45, 0);
-                if (global_var(0) >= 1000) then begin
-                    set_global_var(37, 1);
+                set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+                set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+                set_global_var(GVAR_KARMA_DEFENDER, 0);
+                set_global_var(GVAR_KARMA_WANDERER, 0);
+                set_global_var(GVAR_KARMA_BETRAYER, 0);
+                set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+                set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+                set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+                    set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
                 end
                 else begin
-                    if (global_var(0) >= 750) then begin
-                        set_global_var(38, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
                     end
                     else begin
-                        if (global_var(0) >= 500) then begin
-                            set_global_var(39, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                            set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                         end
                         else begin
-                            if (global_var(0) >= 250) then begin
-                                set_global_var(40, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                                set_global_var(GVAR_KARMA_DEFENDER, 1);
                             end
                             else begin
-                                if (global_var(0) > -250) then begin
-                                    set_global_var(41, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                                    set_global_var(GVAR_KARMA_WANDERER, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -500) then begin
-                                        set_global_var(42, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                        set_global_var(GVAR_KARMA_BETRAYER, 1);
                                     end
                                     else begin
-                                        if (global_var(0) > -750) then begin
-                                            set_global_var(43, 1);
+                                        if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                            set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                         end
                                         else begin
-                                            if (global_var(0) > -1000) then begin
-                                                set_global_var(44, 1);
+                                            if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                                set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                             end
                                             else begin
-                                                set_global_var(45, 1);
+                                                set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                             end
                                         end
                                     end
@@ -268,58 +268,58 @@ begin
                         end
                     end
                 end
-                set_global_var(51, global_var(51) + -8);
+                set_global_var(GVAR_TOWN_REP_ALBUQUERQUE, global_var(GVAR_TOWN_REP_ALBUQUERQUE) + -8);
                 debug_msg("Added " + -8 + " to Town Rep");
             end
         end
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(5) > (2 * global_var(4))) or (global_var(3) == 1))) then begin
-            set_global_var(3, 1);
-            set_global_var(2, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
+            set_global_var(GVAR_BERSERKER_REPUTATION, 1);
+            set_global_var(GVAR_CHAMPION_REPUTATION, 0);
         end
-        set_global_var(0, global_var(0) + -10);
+        set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + -10);
         debug_msg("Player gains " + -10 + " Karma Points.");
-        set_global_var(37, 0);
-        set_global_var(38, 0);
-        set_global_var(39, 0);
-        set_global_var(40, 0);
-        set_global_var(41, 0);
-        set_global_var(42, 0);
-        set_global_var(43, 0);
-        set_global_var(44, 0);
-        set_global_var(45, 0);
-        if (global_var(0) >= 1000) then begin
-            set_global_var(37, 1);
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+        set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+        set_global_var(GVAR_KARMA_DEFENDER, 0);
+        set_global_var(GVAR_KARMA_WANDERER, 0);
+        set_global_var(GVAR_KARMA_BETRAYER, 0);
+        set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+        set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+            set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
         end
         else begin
-            if (global_var(0) >= 750) then begin
-                set_global_var(38, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
             end
             else begin
-                if (global_var(0) >= 500) then begin
-                    set_global_var(39, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                 end
                 else begin
-                    if (global_var(0) >= 250) then begin
-                        set_global_var(40, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                        set_global_var(GVAR_KARMA_DEFENDER, 1);
                     end
                     else begin
-                        if (global_var(0) > -250) then begin
-                            set_global_var(41, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                            set_global_var(GVAR_KARMA_WANDERER, 1);
                         end
                         else begin
-                            if (global_var(0) > -500) then begin
-                                set_global_var(42, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                set_global_var(GVAR_KARMA_BETRAYER, 1);
                             end
                             else begin
-                                if (global_var(0) > -750) then begin
-                                    set_global_var(43, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -1000) then begin
-                                        set_global_var(44, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                     end
                                     else begin
-                                        set_global_var(45, 1);
+                                        set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                     end
                                 end
                             end
@@ -328,10 +328,10 @@ begin
                 end
             end
         end
-        set_global_var(51, global_var(51) + -5);
+        set_global_var(GVAR_TOWN_REP_ALBUQUERQUE, global_var(GVAR_TOWN_REP_ALBUQUERQUE) + -5);
         debug_msg("Added " + -5 + " to Town Rep");
     end
-    if (global_var(872) == 1) then begin
+    if (global_var(GVAR_AQ_QUEST_CHUCK_SWEETHEARTS) == 1) then begin
         set_map_var(6, map_var(6) + 1);
     end
     if (map_var(6) == 1) then begin
@@ -348,13 +348,13 @@ end
 procedure look_at_p_proc
 begin
     script_overrides;
-    display_msg(message_str(1620, 100));
+    display_mstr(100);
 end
 
 procedure description_p_proc
 begin
     script_overrides;
-    display_msg(message_str(1620, 200));
+    display_mstr(200);
 end
 
 procedure use_skill_on_p_proc
@@ -365,8 +365,8 @@ procedure damage_p_proc
 begin
     if (source_obj == dude_obj) then begin
         set_local_var(4, 1);
-        if ((party_member_obj(16777814) != 0) and ((global_var(910) bwand 128) == 0)) then begin
-            set_global_var(1020, global_var(1020) bwor 65536);
+        if ((party_member_obj(16777814) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 128) == 0)) then begin
+            set_global_var(GVAR_PTR_LYSTRA_REACTIONS_2, global_var(GVAR_PTR_LYSTRA_REACTIONS_2) bwor 65536);
         end
     end
 end
@@ -376,8 +376,8 @@ begin
     if (local_var(7) == 0) then begin
         set_local_var(7, tile_num(self_obj));
     end
-    critter_add_trait(self_obj, 1, 6, 262);
-    critter_add_trait(self_obj, 1, 5, 15);
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 262);
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_CHILD);
 end
 
 procedure timed_event_p_proc
@@ -413,88 +413,87 @@ end
 
 procedure Node500
 begin
-    gsay_reply(1620, 500);
-    if ((global_var(867) bwand 524288) != 0) then begin
-        giq_option(4, 1620, 501, Node600, 50);
+    Reply(500);
+    if ((global_var(GVAR_AQ_ALBUQ_DEATHS) bwand 524288) != 0) then begin
+        giq_option(4, NAME, 501, Node600, NEUTRAL_REACTION);
     end
-    if (((local_var(6) bwand 1) == 0) and not((global_var(867) bwand 524288) != 0)) then begin
-        giq_option(4, 1620, 502, Node550, 50);
+    if (((local_var(6) bwand 1) == 0) and not((global_var(GVAR_AQ_ALBUQ_DEATHS) bwand 524288) != 0)) then begin
+        giq_option(4, NAME, 502, Node550, NEUTRAL_REACTION);
     end
-    if (global_var(872) == 1) then begin
-        giq_option(4, 1620, 503, Node998, 50);
+    if (global_var(GVAR_AQ_QUEST_CHUCK_SWEETHEARTS) == 1) then begin
+        giq_option(4, NAME, 503, Node998, NEUTRAL_REACTION);
     end
-    giq_option(4, 1620, 504, Node999, 50);
-    giq_option(-3, 1620, 505, Node999, 50);
+    giq_option(4, NAME, 504, Node999, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 505, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node550
 begin
-    gsay_reply(1620, 550);
+    Reply(550);
     set_local_var(6, local_var(6) bwor 1);
-    if (global_var(872) == 1) then begin
-        giq_option(4, 1620, 551, Node998, 50);
+    if (global_var(GVAR_AQ_QUEST_CHUCK_SWEETHEARTS) == 1) then begin
+        giq_option(4, NAME, 551, Node998, NEUTRAL_REACTION);
     end
-    giq_option(4, 1620, 552, Node999, 50);
+    giq_option(4, NAME, 552, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node600
 begin
-    if (get_critter_stat(dude_obj, 34) == 0) then begin
-        gsay_reply(1620, 600);
+    if (dude_gender == 0) then begin
+        Reply(600);
     end
     else begin
-        gsay_reply(1620, 601);
+        Reply(601);
     end
-    if ((global_var(866) bwand 65536) != 0) then begin
-        giq_option(4, 1620, 603, Node800, 50);
+    if ((global_var(GVAR_AQ_ALBUQ_SPEECHES) bwand 65536) != 0) then begin
+        giq_option(4, NAME, 603, Node800, NEUTRAL_REACTION);
     end
     else begin
-        giq_option(4, 1620, 602, Node650, 50);
+        giq_option(4, NAME, 602, Node650, NEUTRAL_REACTION);
     end
 end
 
 procedure Node650
 begin
-    if (get_critter_stat(dude_obj, 34) == 0) then begin
-        gsay_reply(1620, 650);
+    if (dude_gender == 0) then begin
+        Reply(650);
     end
     else begin
-        gsay_reply(1620, 651);
+        Reply(651);
     end
-    giq_option(4, 1620, 652, Node700, 50);
+    giq_option(4, NAME, 652, Node700, NEUTRAL_REACTION);
 end
 
 procedure Node700
 begin
-    gsay_reply(1620, 700);
-    giq_option(4, 1620, 701, Node750, 50);
+    Reply(700);
+    giq_option(4, NAME, 701, Node750, NEUTRAL_REACTION);
 end
 
 procedure Node750
 begin
-    gsay_reply(1620, 750);
-    if (not((global_var(867) bwand 67108864) != 0)) then begin
-        set_global_var(867, global_var(867) bwor 67108864);
+    Reply(750);
+    if (not((global_var(GVAR_AQ_ALBUQ_DEATHS) bwand 67108864) != 0)) then begin
+        set_global_var(GVAR_AQ_ALBUQ_DEATHS, global_var(GVAR_AQ_ALBUQ_DEATHS) bwor 67108864);
     end
-    set_global_var(1019, global_var(1019) bwand (-1 - 1048576));
-    if ((party_member_obj(16777814) != 0) and ((global_var(910) bwand 128) == 0)) then begin
-        set_global_var(1019, global_var(1019) bwor 8388608);
+    set_global_var(GVAR_PTR_LYSTRA_REACTIONS_1, global_var(GVAR_PTR_LYSTRA_REACTIONS_1) bwand (-1 - 1048576));
+    if ((party_member_obj(16777814) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 128) == 0)) then begin
+        set_global_var(GVAR_PTR_LYSTRA_REACTIONS_1, global_var(GVAR_PTR_LYSTRA_REACTIONS_1) bwor 8388608);
     end
-    giq_option(4, 1620, 751, Node800, 50);
+    giq_option(4, NAME, 751, Node800, NEUTRAL_REACTION);
 end
 
 procedure Node800
 begin
-    gsay_reply(1620, 800);
-    giq_option(4, 1620, 801, GO_HOME_KIDS, 50);
+    Reply(800);
+    giq_option(4, NAME, 801, GO_HOME_KIDS, NEUTRAL_REACTION);
 end
 
 procedure GO_HOME_KIDS
 begin
     add_timer_event(self_obj, 2, 1);
-    if (not((global_var(867) bwand 67108864) != 0)) then begin
-        set_global_var(867, global_var(867) bwor 67108864);
+    if (not((global_var(GVAR_AQ_ALBUQ_DEATHS) bwand 67108864) != 0)) then begin
+        set_global_var(GVAR_AQ_ALBUQ_DEATHS, global_var(GVAR_AQ_ALBUQ_DEATHS) bwor 67108864);
     end
-    set_global_var(866, global_var(866) bwor 134217728);
+    set_global_var(GVAR_AQ_ALBUQ_SPEECHES, global_var(GVAR_AQ_ALBUQ_SPEECHES) bwor 134217728);
 end
-
