diff --git a/Mods/Resurrection/Original/ACISVOJ4.ssl b/Mods/Resurrection/ReDefine/ACISVOJ4.ssl
index ed69445..57de8ae 100644
--- a/Mods/Resurrection/Original/ACISVOJ4.ssl
+++ b/Mods/Resurrection/ReDefine/ACISVOJ4.ssl
@@ -51,12 +51,12 @@ begin
             return 1;
         end
     end
-    if ((party_member_obj(16777779) != 0) and ((global_var(910) bwand 32) == 0)) then begin
+    if ((party_member_obj(16777779) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 32) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777779)) <= 5) then begin
             return 1;
         end
     end
-    if ((party_member_obj(16777811) != 0) and ((global_var(910) bwand 64) == 0)) then begin
+    if ((party_member_obj(16777811) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 64) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777811)) <= 5) then begin
             return 1;
         end
@@ -71,7 +71,7 @@ begin
             return 1;
         end
     end
-    if ((party_member_obj(16777814) != 0) and ((global_var(910) bwand 128) == 0)) then begin
+    if ((party_member_obj(16777814) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 128) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777814)) <= 5) then begin
             return 1;
         end
@@ -128,19 +128,19 @@ end
 
 procedure critter_p_proc
 begin
-    if ((global_var(998) == 1) and (Team_swap2 == 0)) then begin
-        critter_add_trait(self_obj, 1, 6, 282);
+    if ((global_var(GVAR_AQE_ENEMY_EMPEROR) == 1) and (Team_swap2 == 0)) then begin
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 282);
         Team_swap2 := 1;
     end
-    if ((global_var(998) == 1) and (has_trait(1, self_obj, 666) == 1) and obj_can_see_obj(self_obj, dude_obj)) then begin
+    if ((global_var(GVAR_AQE_ENEMY_EMPEROR) == 1) and (obj_is_visible_flag(self_obj) == 1) and obj_can_see_obj(self_obj, dude_obj)) then begin
         if (critter_is_fleeing(self_obj)) then begin
             if (anim_busy(self_obj) == 0) then begin
                 if ((tile_distance_objs(self_obj, dude_obj) < 8) or obj_can_see_obj(self_obj, dude_obj)) then begin
                     if (anim_busy(self_obj) == 0) then begin
-                        global_temp := rotation_to_tile(tile_num(dude_obj), tile_num(self_obj));
+                        global_temp := rotation_to_tile(dude_tile, tile_num(self_obj));
                         animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(self_obj), global_temp, random(3, 10)), 1);
                         global_temp := (global_temp + 1) % 6;
-                        while ((anim_busy(self_obj) == 0) and (global_temp != rotation_to_tile(tile_num(dude_obj), tile_num(self_obj)))) do begin
+                        while ((anim_busy(self_obj) == 0) and (global_temp != rotation_to_tile(dude_tile, tile_num(self_obj)))) do begin
                             debug_msg("flee loop: rot == " + global_temp);
                             animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(self_obj), global_temp, random(3, 10)), 1);
                             global_temp := (global_temp + 1) % 6;
@@ -150,35 +150,35 @@ begin
             end
         end
         else begin
-            attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+            attack(dude_obj);
         end
     end
-    if ((global_var(903) == 2) and (elevation(self_obj) == 1) and ((local_var(5) == 22494) or (local_var(5) == 22698) or (local_var(5) == 22896))) then begin
-        if (((map_var(0) bwand 16384) != 0) and (elevation(dude_obj) == 1) and (global_var(998) != 1) and not(combat_is_initialized)) then begin
+    if ((global_var(GVAR_AQE_QUEST_GET_OUR_VAULT_BACK) == 2) and (elevation(self_obj) == 1) and ((local_var(5) == 22494) or (local_var(5) == 22698) or (local_var(5) == 22896))) then begin
+        if (((map_var(0) bwand 16384) != 0) and (dude_elevation == 1) and (global_var(GVAR_AQE_ENEMY_EMPEROR) != 1) and not(combat_is_initialized)) then begin
             if (tile_distance_objs(self_obj, dude_obj) > 6) then begin
                 if (tile_distance_objs(self_obj, dude_obj) > (6 * 2)) then begin
-                    if ((tile_num(self_obj) != tile_num_in_direction(tile_num(dude_obj), rotation_to_tile(tile_num(dude_obj), tile_num(self_obj)), 2)) and (tile_num_in_direction(tile_num(dude_obj), rotation_to_tile(tile_num(dude_obj), tile_num(self_obj)), 2) != 0)) then begin
-                        if ((dest_tile == tile_num_in_direction(tile_num(dude_obj), rotation_to_tile(tile_num(dude_obj), tile_num(self_obj)), 2)) and (step_tile < 0)) then begin
+                    if ((tile_num(self_obj) != tile_num_in_direction(dude_tile, rotation_to_tile(dude_tile, tile_num(self_obj)), 2)) and (tile_num_in_direction(dude_tile, rotation_to_tile(dude_tile, tile_num(self_obj)), 2) != 0)) then begin
+                        if ((dest_tile == tile_num_in_direction(dude_tile, rotation_to_tile(dude_tile, tile_num(self_obj)), 2)) and (step_tile < 0)) then begin
                             step_tile := step_tile + 1;
                         end
                         else begin
                             if (anim_busy(self_obj) == 0) then begin
-                                dest_tile := tile_num_in_direction(tile_num(dude_obj), rotation_to_tile(tile_num(dude_obj), tile_num(self_obj)), 2);
-                                step_tile := tile_num_in_direction(tile_num(dude_obj), rotation_to_tile(tile_num(dude_obj), tile_num(self_obj)), 2);
+                                dest_tile := tile_num_in_direction(dude_tile, rotation_to_tile(dude_tile, tile_num(self_obj)), 2);
+                                step_tile := tile_num_in_direction(dude_tile, rotation_to_tile(dude_tile, tile_num(self_obj)), 2);
                                 animate_move_obj_to_tile(self_obj, step_tile, 1);
                                 debug_msg("move_to_point: self_tile == " + tile_num(self_obj));
-                                while ((anim_busy(self_obj) == 0) and (tile_distance(tile_num(self_obj), tile_num_in_direction(tile_num(dude_obj), rotation_to_tile(tile_num(dude_obj), tile_num(self_obj)), 2)) >= tile_distance(step_tile, tile_num_in_direction(tile_num(dude_obj), rotation_to_tile(tile_num(dude_obj), tile_num(self_obj)), 2)))) do begin
+                                while ((anim_busy(self_obj) == 0) and (tile_distance(tile_num(self_obj), tile_num_in_direction(dude_tile, rotation_to_tile(dude_tile, tile_num(self_obj)), 2)) >= tile_distance(step_tile, tile_num_in_direction(dude_tile, rotation_to_tile(dude_tile, tile_num(self_obj)), 2)))) do begin
                                     step_tile := tile_num_in_direction(step_tile, rotation_to_tile(step_tile, tile_num(self_obj)), ((tile_distance_objs(dude_obj, self_obj) / 5) + 1) % tile_distance(step_tile, tile_num(self_obj)));
                                     debug_msg("move_to_point: step_tile == " + step_tile);
                                     animate_move_obj_to_tile(self_obj, step_tile, 1);
                                 end
                                 if (anim_busy(self_obj) == 0) then begin
-                                    if (metarule3(106, tile_num_in_direction(tile_num(self_obj), has_trait(1, self_obj, 10), 1), elevation(self_obj), 0) != 0) then begin
+                                    if (tile_get_next_critter(tile_num_in_direction(tile_num(self_obj), obj_get_rot(self_obj), 1), elevation(self_obj), 0) != 0) then begin
                                         if (anim_busy(self_obj) == 0) then begin
-                                            global_temp := rotation_to_tile(tile_num_in_direction(tile_num(self_obj), rotation_to_tile(tile_num_in_direction(tile_num(self_obj), has_trait(1, self_obj, 10), 1), tile_num(self_obj)), 1), tile_num(self_obj));
+                                            global_temp := rotation_to_tile(tile_num_in_direction(tile_num(self_obj), rotation_to_tile(tile_num_in_direction(tile_num(self_obj), obj_get_rot(self_obj), 1), tile_num(self_obj)), 1), tile_num(self_obj));
                                             animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(self_obj), global_temp, random(2, 5)), 0);
                                             global_temp := (global_temp + 1) % 6;
-                                            while ((anim_busy(self_obj) == 0) and (global_temp != rotation_to_tile(tile_num_in_direction(tile_num(self_obj), rotation_to_tile(tile_num_in_direction(tile_num(self_obj), has_trait(1, self_obj, 10), 1), tile_num(self_obj)), 1), tile_num(self_obj)))) do begin
+                                            while ((anim_busy(self_obj) == 0) and (global_temp != rotation_to_tile(tile_num_in_direction(tile_num(self_obj), rotation_to_tile(tile_num_in_direction(tile_num(self_obj), obj_get_rot(self_obj), 1), tile_num(self_obj)), 1), tile_num(self_obj)))) do begin
                                                 debug_msg("flee loop: rot == " + global_temp);
                                                 animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(self_obj), global_temp, random(2, 5)), 0);
                                                 global_temp := (global_temp + 1) % 6;
@@ -191,7 +191,7 @@ begin
                                 end
                             end
                             else begin
-                                if (tile_distance(tile_num(self_obj), tile_num_in_direction(tile_num(dude_obj), rotation_to_tile(tile_num(dude_obj), tile_num(self_obj)), 2)) < tile_distance(dest_tile, tile_num_in_direction(tile_num(dude_obj), rotation_to_tile(tile_num(dude_obj), tile_num(self_obj)), 2))) then begin
+                                if (tile_distance(tile_num(self_obj), tile_num_in_direction(dude_tile, rotation_to_tile(dude_tile, tile_num(self_obj)), 2)) < tile_distance(dest_tile, tile_num_in_direction(dude_tile, rotation_to_tile(dude_tile, tile_num(self_obj)), 2))) then begin
                                     reg_anim_func(2, self_obj);
                                 end
                             end
@@ -199,28 +199,28 @@ begin
                     end
                 end
                 else begin
-                    if ((tile_num(self_obj) != tile_num_in_direction(tile_num(dude_obj), rotation_to_tile(tile_num(dude_obj), tile_num(self_obj)), 2)) and (tile_num_in_direction(tile_num(dude_obj), rotation_to_tile(tile_num(dude_obj), tile_num(self_obj)), 2) != 0)) then begin
-                        if ((dest_tile == tile_num_in_direction(tile_num(dude_obj), rotation_to_tile(tile_num(dude_obj), tile_num(self_obj)), 2)) and (step_tile < 0)) then begin
+                    if ((tile_num(self_obj) != tile_num_in_direction(dude_tile, rotation_to_tile(dude_tile, tile_num(self_obj)), 2)) and (tile_num_in_direction(dude_tile, rotation_to_tile(dude_tile, tile_num(self_obj)), 2) != 0)) then begin
+                        if ((dest_tile == tile_num_in_direction(dude_tile, rotation_to_tile(dude_tile, tile_num(self_obj)), 2)) and (step_tile < 0)) then begin
                             step_tile := step_tile + 1;
                         end
                         else begin
                             if (anim_busy(self_obj) == 0) then begin
-                                dest_tile := tile_num_in_direction(tile_num(dude_obj), rotation_to_tile(tile_num(dude_obj), tile_num(self_obj)), 2);
-                                step_tile := tile_num_in_direction(tile_num(dude_obj), rotation_to_tile(tile_num(dude_obj), tile_num(self_obj)), 2);
+                                dest_tile := tile_num_in_direction(dude_tile, rotation_to_tile(dude_tile, tile_num(self_obj)), 2);
+                                step_tile := tile_num_in_direction(dude_tile, rotation_to_tile(dude_tile, tile_num(self_obj)), 2);
                                 animate_move_obj_to_tile(self_obj, step_tile, 0);
                                 debug_msg("move_to_point: self_tile == " + tile_num(self_obj));
-                                while ((anim_busy(self_obj) == 0) and (tile_distance(tile_num(self_obj), tile_num_in_direction(tile_num(dude_obj), rotation_to_tile(tile_num(dude_obj), tile_num(self_obj)), 2)) >= tile_distance(step_tile, tile_num_in_direction(tile_num(dude_obj), rotation_to_tile(tile_num(dude_obj), tile_num(self_obj)), 2)))) do begin
+                                while ((anim_busy(self_obj) == 0) and (tile_distance(tile_num(self_obj), tile_num_in_direction(dude_tile, rotation_to_tile(dude_tile, tile_num(self_obj)), 2)) >= tile_distance(step_tile, tile_num_in_direction(dude_tile, rotation_to_tile(dude_tile, tile_num(self_obj)), 2)))) do begin
                                     step_tile := tile_num_in_direction(step_tile, rotation_to_tile(step_tile, tile_num(self_obj)), ((tile_distance_objs(dude_obj, self_obj) / 5) + 1) % tile_distance(step_tile, tile_num(self_obj)));
                                     debug_msg("move_to_point: step_tile == " + step_tile);
                                     animate_move_obj_to_tile(self_obj, step_tile, 0);
                                 end
                                 if (anim_busy(self_obj) == 0) then begin
-                                    if (metarule3(106, tile_num_in_direction(tile_num(self_obj), has_trait(1, self_obj, 10), 1), elevation(self_obj), 0) != 0) then begin
+                                    if (tile_get_next_critter(tile_num_in_direction(tile_num(self_obj), obj_get_rot(self_obj), 1), elevation(self_obj), 0) != 0) then begin
                                         if (anim_busy(self_obj) == 0) then begin
-                                            global_temp := rotation_to_tile(tile_num_in_direction(tile_num(self_obj), rotation_to_tile(tile_num_in_direction(tile_num(self_obj), has_trait(1, self_obj, 10), 1), tile_num(self_obj)), 1), tile_num(self_obj));
+                                            global_temp := rotation_to_tile(tile_num_in_direction(tile_num(self_obj), rotation_to_tile(tile_num_in_direction(tile_num(self_obj), obj_get_rot(self_obj), 1), tile_num(self_obj)), 1), tile_num(self_obj));
                                             animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(self_obj), global_temp, random(2, 5)), 0);
                                             global_temp := (global_temp + 1) % 6;
-                                            while ((anim_busy(self_obj) == 0) and (global_temp != rotation_to_tile(tile_num_in_direction(tile_num(self_obj), rotation_to_tile(tile_num_in_direction(tile_num(self_obj), has_trait(1, self_obj, 10), 1), tile_num(self_obj)), 1), tile_num(self_obj)))) do begin
+                                            while ((anim_busy(self_obj) == 0) and (global_temp != rotation_to_tile(tile_num_in_direction(tile_num(self_obj), rotation_to_tile(tile_num_in_direction(tile_num(self_obj), obj_get_rot(self_obj), 1), tile_num(self_obj)), 1), tile_num(self_obj)))) do begin
                                                 debug_msg("flee loop: rot == " + global_temp);
                                                 animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(self_obj), global_temp, random(2, 5)), 0);
                                                 global_temp := (global_temp + 1) % 6;
@@ -233,7 +233,7 @@ begin
                                 end
                             end
                             else begin
-                                if (tile_distance(tile_num(self_obj), tile_num_in_direction(tile_num(dude_obj), rotation_to_tile(tile_num(dude_obj), tile_num(self_obj)), 2)) < tile_distance(dest_tile, tile_num_in_direction(tile_num(dude_obj), rotation_to_tile(tile_num(dude_obj), tile_num(self_obj)), 2))) then begin
+                                if (tile_distance(tile_num(self_obj), tile_num_in_direction(dude_tile, rotation_to_tile(dude_tile, tile_num(self_obj)), 2)) < tile_distance(dest_tile, tile_num_in_direction(dude_tile, rotation_to_tile(dude_tile, tile_num(self_obj)), 2))) then begin
                                     reg_anim_func(2, self_obj);
                                 end
                             end
@@ -242,7 +242,7 @@ begin
                 end
             end
         end
-        if (((global_var(1000) bwand 33554432) != 0) and not(combat_is_initialized)) then begin
+        if (((global_var(GVAR_AQE_ALBUQ_EMPEROR_EVENTS_NEXT) bwand 33554432) != 0) and not(combat_is_initialized)) then begin
             if ((map_var(0) bwand 16384) != 0) then begin
                 set_map_var(0, map_var(0) bwand (-1 - 16384));
             end
@@ -250,11 +250,11 @@ begin
             if ((tile_num(self_obj) == 22296) or (tile_num(self_obj) == 22295) or (tile_num(self_obj) == 22495)) then begin
                 destroy_object(self_obj);
             end
-            if (elevation(dude_obj) == 0) then begin
+            if (dude_elevation == 0) then begin
                 destroy_object(self_obj);
             end
         end
-        if ((global_var(1012) >= 11) and not(combat_is_initialized)) then begin
+        if ((global_var(GVAR_AQE_2ST_FLOOR_ALIENS_KILLED) >= 11) and not(combat_is_initialized)) then begin
             if ((map_var(0) bwand 16384) != 0) then begin
                 set_map_var(0, map_var(0) bwand (-1 - 16384));
             end
@@ -262,41 +262,41 @@ begin
             if ((tile_num(self_obj) == 22296) or (tile_num(self_obj) == 22295) or (tile_num(self_obj) == 22495)) then begin
                 destroy_object(self_obj);
             end
-            if (elevation(dude_obj) == 0) then begin
+            if (dude_elevation == 0) then begin
                 destroy_object(self_obj);
             end
         end
     end
-    if ((global_var(903) == 3) and (elevation(self_obj) == 2)) then begin
-        if ((global_var(864) == 1) and not((map_var(0) bwand 16384) != 0)) then begin
-            critter_add_trait(self_obj, 1, 6, 0);
+    if ((global_var(GVAR_AQE_QUEST_GET_OUR_VAULT_BACK) == 3) and (elevation(self_obj) == 2)) then begin
+        if ((global_var(GVAR_AQ_ENEMY_HUNTERS) == 1) and not((map_var(0) bwand 16384) != 0)) then begin
+            critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 0);
             set_map_var(0, map_var(0) bwor 16384);
         end
-        if (((map_var(0) bwand 16384) != 0) and (elevation(dude_obj) == 2) and (global_var(998) != 1) and not(combat_is_initialized)) then begin
+        if (((map_var(0) bwand 16384) != 0) and (dude_elevation == 2) and (global_var(GVAR_AQE_ENEMY_EMPEROR) != 1) and not(combat_is_initialized)) then begin
             if (tile_distance_objs(self_obj, dude_obj) > 6) then begin
                 if (tile_distance_objs(self_obj, dude_obj) > (6 * 2)) then begin
-                    if ((tile_num(self_obj) != tile_num_in_direction(tile_num(dude_obj), rotation_to_tile(tile_num(dude_obj), tile_num(self_obj)), 2)) and (tile_num_in_direction(tile_num(dude_obj), rotation_to_tile(tile_num(dude_obj), tile_num(self_obj)), 2) != 0)) then begin
-                        if ((dest_tile == tile_num_in_direction(tile_num(dude_obj), rotation_to_tile(tile_num(dude_obj), tile_num(self_obj)), 2)) and (step_tile < 0)) then begin
+                    if ((tile_num(self_obj) != tile_num_in_direction(dude_tile, rotation_to_tile(dude_tile, tile_num(self_obj)), 2)) and (tile_num_in_direction(dude_tile, rotation_to_tile(dude_tile, tile_num(self_obj)), 2) != 0)) then begin
+                        if ((dest_tile == tile_num_in_direction(dude_tile, rotation_to_tile(dude_tile, tile_num(self_obj)), 2)) and (step_tile < 0)) then begin
                             step_tile := step_tile + 1;
                         end
                         else begin
                             if (anim_busy(self_obj) == 0) then begin
-                                dest_tile := tile_num_in_direction(tile_num(dude_obj), rotation_to_tile(tile_num(dude_obj), tile_num(self_obj)), 2);
-                                step_tile := tile_num_in_direction(tile_num(dude_obj), rotation_to_tile(tile_num(dude_obj), tile_num(self_obj)), 2);
+                                dest_tile := tile_num_in_direction(dude_tile, rotation_to_tile(dude_tile, tile_num(self_obj)), 2);
+                                step_tile := tile_num_in_direction(dude_tile, rotation_to_tile(dude_tile, tile_num(self_obj)), 2);
                                 animate_move_obj_to_tile(self_obj, step_tile, 1);
                                 debug_msg("move_to_point: self_tile == " + tile_num(self_obj));
-                                while ((anim_busy(self_obj) == 0) and (tile_distance(tile_num(self_obj), tile_num_in_direction(tile_num(dude_obj), rotation_to_tile(tile_num(dude_obj), tile_num(self_obj)), 2)) >= tile_distance(step_tile, tile_num_in_direction(tile_num(dude_obj), rotation_to_tile(tile_num(dude_obj), tile_num(self_obj)), 2)))) do begin
+                                while ((anim_busy(self_obj) == 0) and (tile_distance(tile_num(self_obj), tile_num_in_direction(dude_tile, rotation_to_tile(dude_tile, tile_num(self_obj)), 2)) >= tile_distance(step_tile, tile_num_in_direction(dude_tile, rotation_to_tile(dude_tile, tile_num(self_obj)), 2)))) do begin
                                     step_tile := tile_num_in_direction(step_tile, rotation_to_tile(step_tile, tile_num(self_obj)), ((tile_distance_objs(dude_obj, self_obj) / 5) + 1) % tile_distance(step_tile, tile_num(self_obj)));
                                     debug_msg("move_to_point: step_tile == " + step_tile);
                                     animate_move_obj_to_tile(self_obj, step_tile, 1);
                                 end
                                 if (anim_busy(self_obj) == 0) then begin
-                                    if (metarule3(106, tile_num_in_direction(tile_num(self_obj), has_trait(1, self_obj, 10), 1), elevation(self_obj), 0) != 0) then begin
+                                    if (tile_get_next_critter(tile_num_in_direction(tile_num(self_obj), obj_get_rot(self_obj), 1), elevation(self_obj), 0) != 0) then begin
                                         if (anim_busy(self_obj) == 0) then begin
-                                            global_temp := rotation_to_tile(tile_num_in_direction(tile_num(self_obj), rotation_to_tile(tile_num_in_direction(tile_num(self_obj), has_trait(1, self_obj, 10), 1), tile_num(self_obj)), 1), tile_num(self_obj));
+                                            global_temp := rotation_to_tile(tile_num_in_direction(tile_num(self_obj), rotation_to_tile(tile_num_in_direction(tile_num(self_obj), obj_get_rot(self_obj), 1), tile_num(self_obj)), 1), tile_num(self_obj));
                                             animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(self_obj), global_temp, random(2, 5)), 0);
                                             global_temp := (global_temp + 1) % 6;
-                                            while ((anim_busy(self_obj) == 0) and (global_temp != rotation_to_tile(tile_num_in_direction(tile_num(self_obj), rotation_to_tile(tile_num_in_direction(tile_num(self_obj), has_trait(1, self_obj, 10), 1), tile_num(self_obj)), 1), tile_num(self_obj)))) do begin
+                                            while ((anim_busy(self_obj) == 0) and (global_temp != rotation_to_tile(tile_num_in_direction(tile_num(self_obj), rotation_to_tile(tile_num_in_direction(tile_num(self_obj), obj_get_rot(self_obj), 1), tile_num(self_obj)), 1), tile_num(self_obj)))) do begin
                                                 debug_msg("flee loop: rot == " + global_temp);
                                                 animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(self_obj), global_temp, random(2, 5)), 0);
                                                 global_temp := (global_temp + 1) % 6;
@@ -309,7 +309,7 @@ begin
                                 end
                             end
                             else begin
-                                if (tile_distance(tile_num(self_obj), tile_num_in_direction(tile_num(dude_obj), rotation_to_tile(tile_num(dude_obj), tile_num(self_obj)), 2)) < tile_distance(dest_tile, tile_num_in_direction(tile_num(dude_obj), rotation_to_tile(tile_num(dude_obj), tile_num(self_obj)), 2))) then begin
+                                if (tile_distance(tile_num(self_obj), tile_num_in_direction(dude_tile, rotation_to_tile(dude_tile, tile_num(self_obj)), 2)) < tile_distance(dest_tile, tile_num_in_direction(dude_tile, rotation_to_tile(dude_tile, tile_num(self_obj)), 2))) then begin
                                     reg_anim_func(2, self_obj);
                                 end
                             end
@@ -317,28 +317,28 @@ begin
                     end
                 end
                 else begin
-                    if ((tile_num(self_obj) != tile_num_in_direction(tile_num(dude_obj), rotation_to_tile(tile_num(dude_obj), tile_num(self_obj)), 2)) and (tile_num_in_direction(tile_num(dude_obj), rotation_to_tile(tile_num(dude_obj), tile_num(self_obj)), 2) != 0)) then begin
-                        if ((dest_tile == tile_num_in_direction(tile_num(dude_obj), rotation_to_tile(tile_num(dude_obj), tile_num(self_obj)), 2)) and (step_tile < 0)) then begin
+                    if ((tile_num(self_obj) != tile_num_in_direction(dude_tile, rotation_to_tile(dude_tile, tile_num(self_obj)), 2)) and (tile_num_in_direction(dude_tile, rotation_to_tile(dude_tile, tile_num(self_obj)), 2) != 0)) then begin
+                        if ((dest_tile == tile_num_in_direction(dude_tile, rotation_to_tile(dude_tile, tile_num(self_obj)), 2)) and (step_tile < 0)) then begin
                             step_tile := step_tile + 1;
                         end
                         else begin
                             if (anim_busy(self_obj) == 0) then begin
-                                dest_tile := tile_num_in_direction(tile_num(dude_obj), rotation_to_tile(tile_num(dude_obj), tile_num(self_obj)), 2);
-                                step_tile := tile_num_in_direction(tile_num(dude_obj), rotation_to_tile(tile_num(dude_obj), tile_num(self_obj)), 2);
+                                dest_tile := tile_num_in_direction(dude_tile, rotation_to_tile(dude_tile, tile_num(self_obj)), 2);
+                                step_tile := tile_num_in_direction(dude_tile, rotation_to_tile(dude_tile, tile_num(self_obj)), 2);
                                 animate_move_obj_to_tile(self_obj, step_tile, 0);
                                 debug_msg("move_to_point: self_tile == " + tile_num(self_obj));
-                                while ((anim_busy(self_obj) == 0) and (tile_distance(tile_num(self_obj), tile_num_in_direction(tile_num(dude_obj), rotation_to_tile(tile_num(dude_obj), tile_num(self_obj)), 2)) >= tile_distance(step_tile, tile_num_in_direction(tile_num(dude_obj), rotation_to_tile(tile_num(dude_obj), tile_num(self_obj)), 2)))) do begin
+                                while ((anim_busy(self_obj) == 0) and (tile_distance(tile_num(self_obj), tile_num_in_direction(dude_tile, rotation_to_tile(dude_tile, tile_num(self_obj)), 2)) >= tile_distance(step_tile, tile_num_in_direction(dude_tile, rotation_to_tile(dude_tile, tile_num(self_obj)), 2)))) do begin
                                     step_tile := tile_num_in_direction(step_tile, rotation_to_tile(step_tile, tile_num(self_obj)), ((tile_distance_objs(dude_obj, self_obj) / 5) + 1) % tile_distance(step_tile, tile_num(self_obj)));
                                     debug_msg("move_to_point: step_tile == " + step_tile);
                                     animate_move_obj_to_tile(self_obj, step_tile, 0);
                                 end
                                 if (anim_busy(self_obj) == 0) then begin
-                                    if (metarule3(106, tile_num_in_direction(tile_num(self_obj), has_trait(1, self_obj, 10), 1), elevation(self_obj), 0) != 0) then begin
+                                    if (tile_get_next_critter(tile_num_in_direction(tile_num(self_obj), obj_get_rot(self_obj), 1), elevation(self_obj), 0) != 0) then begin
                                         if (anim_busy(self_obj) == 0) then begin
-                                            global_temp := rotation_to_tile(tile_num_in_direction(tile_num(self_obj), rotation_to_tile(tile_num_in_direction(tile_num(self_obj), has_trait(1, self_obj, 10), 1), tile_num(self_obj)), 1), tile_num(self_obj));
+                                            global_temp := rotation_to_tile(tile_num_in_direction(tile_num(self_obj), rotation_to_tile(tile_num_in_direction(tile_num(self_obj), obj_get_rot(self_obj), 1), tile_num(self_obj)), 1), tile_num(self_obj));
                                             animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(self_obj), global_temp, random(2, 5)), 0);
                                             global_temp := (global_temp + 1) % 6;
-                                            while ((anim_busy(self_obj) == 0) and (global_temp != rotation_to_tile(tile_num_in_direction(tile_num(self_obj), rotation_to_tile(tile_num_in_direction(tile_num(self_obj), has_trait(1, self_obj, 10), 1), tile_num(self_obj)), 1), tile_num(self_obj)))) do begin
+                                            while ((anim_busy(self_obj) == 0) and (global_temp != rotation_to_tile(tile_num_in_direction(tile_num(self_obj), rotation_to_tile(tile_num_in_direction(tile_num(self_obj), obj_get_rot(self_obj), 1), tile_num(self_obj)), 1), tile_num(self_obj)))) do begin
                                                 debug_msg("flee loop: rot == " + global_temp);
                                                 animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(self_obj), global_temp, random(2, 5)), 0);
                                                 global_temp := (global_temp + 1) % 6;
@@ -351,7 +351,7 @@ begin
                                 end
                             end
                             else begin
-                                if (tile_distance(tile_num(self_obj), tile_num_in_direction(tile_num(dude_obj), rotation_to_tile(tile_num(dude_obj), tile_num(self_obj)), 2)) < tile_distance(dest_tile, tile_num_in_direction(tile_num(dude_obj), rotation_to_tile(tile_num(dude_obj), tile_num(self_obj)), 2))) then begin
+                                if (tile_distance(tile_num(self_obj), tile_num_in_direction(dude_tile, rotation_to_tile(dude_tile, tile_num(self_obj)), 2)) < tile_distance(dest_tile, tile_num_in_direction(dude_tile, rotation_to_tile(dude_tile, tile_num(self_obj)), 2))) then begin
                                     reg_anim_func(2, self_obj);
                                 end
                             end
@@ -360,34 +360,34 @@ begin
                 end
             end
         end
-        if (((map_var(0) bwand 16384) != 0) and (global_var(864) == 1) and not((global_var(865) bwand 67108864) != 0) and not(combat_is_initialized)) then begin
+        if (((map_var(0) bwand 16384) != 0) and (global_var(GVAR_AQ_ENEMY_HUNTERS) == 1) and not((global_var(GVAR_AQ_ALBUQ_EVENTS) bwand 67108864) != 0) and not(combat_is_initialized)) then begin
             if ((coridor_guard1 != -1) and obj_can_see_obj(self_obj, coridor_guard1)) then begin
-                attack_complex(coridor_guard1, 0, 1, 0, 0, 30000, 0, 0);
+                attack(coridor_guard1);
             end
             if ((coridor_guard2 != -1) and obj_can_see_obj(self_obj, coridor_guard2)) then begin
-                attack_complex(coridor_guard2, 0, 1, 0, 0, 30000, 0, 0);
+                attack(coridor_guard2);
             end
         end
         if ((map_var(0) bwand 131072) != 0) then begin
             set_map_var(0, map_var(0) bwand (-1 - 131072));
-            float_msg(self_obj, message_str(128, random(382, 383)), 8);
+            float_msg(self_obj, mstr(random(382, 383)), FLOAT_MSG_YELLOW);
         end
         if (((map_var(0) bwand 16384) != 0) and (Team_swap == 0)) then begin
-            critter_add_trait(self_obj, 1, 6, 0);
+            critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 0);
             Team_swap := 1;
         end
     end
-    if ((global_var(903) == 4) and (elevation(self_obj) == 2) and (elevation(dude_obj) == 1)) then begin
+    if ((global_var(GVAR_AQE_QUEST_GET_OUR_VAULT_BACK) == 4) and (elevation(self_obj) == 2) and (dude_elevation == 1)) then begin
         destroy_object(self_obj);
     end
     if (local_var(5) == 0) then begin
         if ((map_var(0) bwand 16384) != 0) then begin
-            critter_add_trait(self_obj, 1, 6, 0);
+            critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 0);
         end
         else begin
-            critter_add_trait(self_obj, 1, 6, 282);
+            critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 282);
         end
-        critter_add_trait(self_obj, 1, 5, 29);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_IMPERIAL_SOLDIER_NO_TALKING);
         set_local_var(5, tile_num(self_obj));
     end
 end
@@ -395,35 +395,35 @@ end
 procedure pickup_p_proc
 begin
     if ((source_obj == dude_obj) and not((map_var(0) bwand 16384) != 0)) then begin
-        set_global_var(998, 1);
+        set_global_var(GVAR_AQE_ENEMY_EMPEROR, 1);
     end
 end
 
 procedure talk_p_proc
 begin
-    if ((Projed_smycku == 0) and ((map_var(0) bwand 16384) != 0) and (global_var(903) == 2) and (global_var(1012) < 11) and (elevation(self_obj) == 1)) then begin
-        float_msg(self_obj, message_str(128, 387), 8);
+    if ((Projed_smycku == 0) and ((map_var(0) bwand 16384) != 0) and (global_var(GVAR_AQE_QUEST_GET_OUR_VAULT_BACK) == 2) and (global_var(GVAR_AQE_2ST_FLOOR_ALIENS_KILLED) < 11) and (elevation(self_obj) == 1)) then begin
+        float_msg(self_obj, mstr(387), FLOAT_MSG_YELLOW);
         Projed_smycku := 1;
     end
-    if ((Projed_smycku == 0) and (global_var(903) == 3) and (elevation(self_obj) == 2)) then begin
-        float_msg(self_obj, message_str(128, random(384, 390)), 8);
+    if ((Projed_smycku == 0) and (global_var(GVAR_AQE_QUEST_GET_OUR_VAULT_BACK) == 3) and (elevation(self_obj) == 2)) then begin
+        float_msg(self_obj, mstr(random(384, 390)), FLOAT_MSG_YELLOW);
         Projed_smycku := 1;
     end
-    if ((Projed_smycku == 0) and (global_var(903) >= 4) and (elevation(self_obj) == 2)) then begin
-        if (get_critter_stat(dude_obj, 34) == 0) then begin
-            float_msg(self_obj, message_str(128, random(350, 365)), 8);
+    if ((Projed_smycku == 0) and (global_var(GVAR_AQE_QUEST_GET_OUR_VAULT_BACK) >= 4) and (elevation(self_obj) == 2)) then begin
+        if (dude_gender == 0) then begin
+            float_msg(self_obj, mstr(random(350, 365)), FLOAT_MSG_YELLOW);
         end
         else begin
-            float_msg(self_obj, message_str(128, random(352, 367)), 8);
+            float_msg(self_obj, mstr(random(352, 367)), FLOAT_MSG_YELLOW);
         end
         Projed_smycku := 1;
     end
-    if ((Projed_smycku == 0) and (global_var(903) >= 2) and (elevation(self_obj) == 1)) then begin
-        if (get_critter_stat(dude_obj, 34) == 0) then begin
-            float_msg(self_obj, message_str(128, random(300, 309)), 8);
+    if ((Projed_smycku == 0) and (global_var(GVAR_AQE_QUEST_GET_OUR_VAULT_BACK) >= 2) and (elevation(self_obj) == 1)) then begin
+        if (dude_gender == 0) then begin
+            float_msg(self_obj, mstr(random(300, 309)), FLOAT_MSG_YELLOW);
         end
         else begin
-            float_msg(self_obj, message_str(128, random(301, 310)), 8);
+            float_msg(self_obj, mstr(random(301, 310)), FLOAT_MSG_YELLOW);
         end
         Projed_smycku := 1;
     end
@@ -433,63 +433,63 @@ end
 procedure destroy_p_proc
 begin
     if ((map_var(0) bwand 16384) != 0) then begin
-        set_global_var(1013, global_var(1013) + 1);
+        set_global_var(GVAR_AQE_VAULT_SOLDIERS_KILLED, global_var(GVAR_AQE_VAULT_SOLDIERS_KILLED) + 1);
     end
     else begin
         if (source_obj == dude_obj) then begin
-            set_global_var(5, global_var(5) + 1);
-            if (((global_var(4) + global_var(5)) >= 25) and ((global_var(4) > (3 * global_var(5))) or (global_var(2) == 1)) and (global_var(1) == 0)) then begin
-                set_global_var(2, 1);
-                set_global_var(3, 0);
+            set_global_var(GVAR_GOOD_MONSTER, global_var(GVAR_GOOD_MONSTER) + 1);
+            if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(4) > (3 * global_var(GVAR_GOOD_MONSTER))) or (global_var(GVAR_CHAMPION_REPUTATION) == 1)) and (global_var(GVAR_CHILDKILLER_REPUTATION) == 0)) then begin
+                set_global_var(GVAR_CHAMPION_REPUTATION, 1);
+                set_global_var(GVAR_BERSERKER_REPUTATION, 0);
             end
-            if (metarule(51, self_obj) == 2) then begin
+            if (critter_kill_type(self_obj) == KILL_TYPE_children_kills) then begin
                 if (source_obj == dude_obj) then begin
-                    set_global_var(2, 0);
-                    set_global_var(1, global_var(1) + 1);
-                    set_global_var(0, global_var(0) + -15);
+                    set_global_var(GVAR_CHAMPION_REPUTATION, 0);
+                    set_global_var(GVAR_CHILDKILLER_REPUTATION, global_var(GVAR_CHILDKILLER_REPUTATION) + 1);
+                    set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + -15);
                     debug_msg("Player gains " + -15 + " Karma Points.");
-                    set_global_var(37, 0);
-                    set_global_var(38, 0);
-                    set_global_var(39, 0);
-                    set_global_var(40, 0);
-                    set_global_var(41, 0);
-                    set_global_var(42, 0);
-                    set_global_var(43, 0);
-                    set_global_var(44, 0);
-                    set_global_var(45, 0);
-                    if (global_var(0) >= 1000) then begin
-                        set_global_var(37, 1);
+                    set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+                    set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+                    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+                    set_global_var(GVAR_KARMA_DEFENDER, 0);
+                    set_global_var(GVAR_KARMA_WANDERER, 0);
+                    set_global_var(GVAR_KARMA_BETRAYER, 0);
+                    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+                    set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+                    set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+                        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
                     end
                     else begin
-                        if (global_var(0) >= 750) then begin
-                            set_global_var(38, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                            set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
                         end
                         else begin
-                            if (global_var(0) >= 500) then begin
-                                set_global_var(39, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                                set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                             end
                             else begin
-                                if (global_var(0) >= 250) then begin
-                                    set_global_var(40, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                                    set_global_var(GVAR_KARMA_DEFENDER, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -250) then begin
-                                        set_global_var(41, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                                        set_global_var(GVAR_KARMA_WANDERER, 1);
                                     end
                                     else begin
-                                        if (global_var(0) > -500) then begin
-                                            set_global_var(42, 1);
+                                        if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                            set_global_var(GVAR_KARMA_BETRAYER, 1);
                                         end
                                         else begin
-                                            if (global_var(0) > -750) then begin
-                                                set_global_var(43, 1);
+                                            if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                                set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                             end
                                             else begin
-                                                if (global_var(0) > -1000) then begin
-                                                    set_global_var(44, 1);
+                                                if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                                    set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                                 end
                                                 else begin
-                                                    set_global_var(45, 1);
+                                                    set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                                 end
                                             end
                                         end
@@ -498,58 +498,58 @@ begin
                             end
                         end
                     end
-                    set_global_var(53, global_var(53) + -8);
+                    set_global_var(GVAR_TOWN_REP_EMPORER_CITY, global_var(GVAR_TOWN_REP_EMPORER_CITY) + -8);
                     debug_msg("Added " + -8 + " to Town Rep");
                 end
             end
-            if (((global_var(4) + global_var(5)) >= 25) and ((global_var(5) > (2 * global_var(4))) or (global_var(3) == 1))) then begin
-                set_global_var(3, 1);
-                set_global_var(2, 0);
+            if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
+                set_global_var(GVAR_BERSERKER_REPUTATION, 1);
+                set_global_var(GVAR_CHAMPION_REPUTATION, 0);
             end
-            set_global_var(0, global_var(0) + -10);
+            set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + -10);
             debug_msg("Player gains " + -10 + " Karma Points.");
-            set_global_var(37, 0);
-            set_global_var(38, 0);
-            set_global_var(39, 0);
-            set_global_var(40, 0);
-            set_global_var(41, 0);
-            set_global_var(42, 0);
-            set_global_var(43, 0);
-            set_global_var(44, 0);
-            set_global_var(45, 0);
-            if (global_var(0) >= 1000) then begin
-                set_global_var(37, 1);
+            set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+            set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+            set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+            set_global_var(GVAR_KARMA_DEFENDER, 0);
+            set_global_var(GVAR_KARMA_WANDERER, 0);
+            set_global_var(GVAR_KARMA_BETRAYER, 0);
+            set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+            set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+            set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+                set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
             end
             else begin
-                if (global_var(0) >= 750) then begin
-                    set_global_var(38, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                    set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
                 end
                 else begin
-                    if (global_var(0) >= 500) then begin
-                        set_global_var(39, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                        set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                     end
                     else begin
-                        if (global_var(0) >= 250) then begin
-                            set_global_var(40, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                            set_global_var(GVAR_KARMA_DEFENDER, 1);
                         end
                         else begin
-                            if (global_var(0) > -250) then begin
-                                set_global_var(41, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                                set_global_var(GVAR_KARMA_WANDERER, 1);
                             end
                             else begin
-                                if (global_var(0) > -500) then begin
-                                    set_global_var(42, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                    set_global_var(GVAR_KARMA_BETRAYER, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -750) then begin
-                                        set_global_var(43, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                        set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                     end
                                     else begin
-                                        if (global_var(0) > -1000) then begin
-                                            set_global_var(44, 1);
+                                        if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                            set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                         end
                                         else begin
-                                            set_global_var(45, 1);
+                                            set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                         end
                                     end
                                 end
@@ -558,7 +558,7 @@ begin
                     end
                 end
             end
-            set_global_var(53, global_var(53) + -5);
+            set_global_var(GVAR_TOWN_REP_EMPORER_CITY, global_var(GVAR_TOWN_REP_EMPORER_CITY) + -5);
             debug_msg("Added " + -5 + " to Town Rep");
         end
     end
@@ -567,13 +567,13 @@ end
 procedure look_at_p_proc
 begin
     script_overrides;
-    display_msg(message_str(128, 100));
+    display_mstr(100);
 end
 
 procedure description_p_proc
 begin
     script_overrides;
-    display_msg(message_str(128, 200));
+    display_mstr(200);
 end
 
 procedure use_skill_on_p_proc
@@ -583,28 +583,28 @@ end
 procedure damage_p_proc
 begin
     if ((source_obj == dude_obj) and not((map_var(0) bwand 16384) != 0)) then begin
-        set_global_var(998, 1);
+        set_global_var(GVAR_AQE_ENEMY_EMPEROR, 1);
     end
 end
 
 procedure map_enter_p_proc
 begin
     if ((map_var(0) bwand 16384) != 0) then begin
-        critter_add_trait(self_obj, 1, 6, 0);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 0);
         Team_swap := 1;
     end
     else begin
-        if ((global_var(864) == 1) and (global_var(903) == 3) and ((local_var(5) == 17311) or (local_var(5) == 17313)) and (elevation(self_obj) == 2)) then begin
-            critter_add_trait(self_obj, 1, 6, 0);
+        if ((global_var(GVAR_AQ_ENEMY_HUNTERS) == 1) and (global_var(GVAR_AQE_QUEST_GET_OUR_VAULT_BACK) == 3) and ((local_var(5) == 17311) or (local_var(5) == 17313)) and (elevation(self_obj) == 2)) then begin
+            critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 0);
         end
         else begin
-            critter_add_trait(self_obj, 1, 6, 282);
+            critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 282);
         end
     end
-    if (global_var(998) == 1) then begin
-        critter_add_trait(self_obj, 1, 6, 282);
+    if (global_var(GVAR_AQE_ENEMY_EMPEROR) == 1) then begin
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 282);
     end
-    critter_add_trait(self_obj, 1, 5, 29);
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_IMPERIAL_SOLDIER_NO_TALKING);
     if (local_var(5) == 0) then begin
         set_local_var(5, tile_num(self_obj));
     end
@@ -616,13 +616,12 @@ end
 
 procedure combat_p_proc
 begin
-    if (((map_var(0) bwand 16384) != 0) or (global_var(864) == 1) and (global_var(903) == 3) and (Team_swap == 0) and (elevation(self_obj) == 2)) then begin
-        critter_add_trait(self_obj, 1, 6, 0);
+    if (((map_var(0) bwand 16384) != 0) or (global_var(GVAR_AQ_ENEMY_HUNTERS) == 1) and (global_var(GVAR_AQE_QUEST_GET_OUR_VAULT_BACK) == 3) and (Team_swap == 0) and (elevation(self_obj) == 2)) then begin
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 0);
         Team_swap := 1;
     end
-    if ((global_var(998) == 1) and (Team_swap2 == 0)) then begin
-        critter_add_trait(self_obj, 1, 6, 282);
+    if ((global_var(GVAR_AQE_ENEMY_EMPEROR) == 1) and (Team_swap2 == 0)) then begin
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 282);
         Team_swap2 := 1;
     end
 end
-
