diff --git a/Mods/Resurrection/Original/ACISVOJ9.ssl b/Mods/Resurrection/ReDefine/ACISVOJ9.ssl
index 0b77832..9884a69 100644
--- a/Mods/Resurrection/Original/ACISVOJ9.ssl
+++ b/Mods/Resurrection/ReDefine/ACISVOJ9.ssl
@@ -52,12 +52,12 @@ begin
             return 1;
         end
     end
-    if ((party_member_obj(16777779) != 0) and ((global_var(910) bwand 32) == 0)) then begin
+    if ((party_member_obj(16777779) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 32) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777779)) <= 5) then begin
             return 1;
         end
     end
-    if ((party_member_obj(16777811) != 0) and ((global_var(910) bwand 64) == 0)) then begin
+    if ((party_member_obj(16777811) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 64) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777811)) <= 5) then begin
             return 1;
         end
@@ -72,7 +72,7 @@ begin
             return 1;
         end
     end
-    if ((party_member_obj(16777814) != 0) and ((global_var(910) bwand 128) == 0)) then begin
+    if ((party_member_obj(16777814) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 128) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777814)) <= 5) then begin
             return 1;
         end
@@ -129,15 +129,15 @@ end
 
 procedure critter_p_proc
 begin
-    if ((global_var(998) == 1) and (has_trait(1, self_obj, 666) == 1) and obj_can_see_obj(self_obj, dude_obj)) then begin
+    if ((global_var(GVAR_AQE_ENEMY_EMPEROR) == 1) and (obj_is_visible_flag(self_obj) == 1) and obj_can_see_obj(self_obj, dude_obj)) then begin
         if (critter_is_fleeing(self_obj)) then begin
             if (anim_busy(self_obj) == 0) then begin
                 if ((tile_distance_objs(self_obj, dude_obj) < 8) or obj_can_see_obj(self_obj, dude_obj)) then begin
                     if (anim_busy(self_obj) == 0) then begin
-                        global_temp := rotation_to_tile(tile_num(dude_obj), tile_num(self_obj));
+                        global_temp := rotation_to_tile(dude_tile, tile_num(self_obj));
                         animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(self_obj), global_temp, random(3, 10)), 1);
                         global_temp := (global_temp + 1) % 6;
-                        while ((anim_busy(self_obj) == 0) and (global_temp != rotation_to_tile(tile_num(dude_obj), tile_num(self_obj)))) do begin
+                        while ((anim_busy(self_obj) == 0) and (global_temp != rotation_to_tile(dude_tile, tile_num(self_obj)))) do begin
                             debug_msg("flee loop: rot == " + global_temp);
                             animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(self_obj), global_temp, random(3, 10)), 1);
                             global_temp := (global_temp + 1) % 6;
@@ -147,39 +147,39 @@ begin
             end
         end
         else begin
-            attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+            attack(dude_obj);
         end
     end
-    if ((global_var(1008) >= 9) and ((global_var(904) bwand 1) != 0)) then begin
+    if ((global_var(GVAR_AQE_HELP_TO_ARAN) >= 9) and ((global_var(GVAR_AQE_ALBUQ_EMPEROR_EVENTS) bwand 1) != 0)) then begin
         destroy_object(self_obj);
     end
-    if ((global_var(1000) bwand 64) != 0) then begin
-        float_msg(self_obj, message_str(140, 600), 8);
+    if ((global_var(GVAR_AQE_ALBUQ_EMPEROR_EVENTS_NEXT) bwand 64) != 0) then begin
+        float_msg(self_obj, mstr(600), FLOAT_MSG_YELLOW);
     end
     if (local_var(5) == 0) then begin
-        critter_add_trait(self_obj, 1, 6, 0);
-        critter_add_trait(self_obj, 1, 5, 29);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 0);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_IMPERIAL_SOLDIER_NO_TALKING);
         set_local_var(5, tile_num(self_obj));
     end
-    if ((cur_map_index == 25) and (has_trait(1, self_obj, 666) == 1) and not(combat_is_initialized)) then begin
+    if ((CUR_MAP_ALB_CISA) and (obj_is_visible_flag(self_obj) == 1) and not(combat_is_initialized)) then begin
         if ((self_talker == 0) and ((map_var(0) bwand 524288) != 0) and not((map_var(0) bwand 2097152) != 0)) then begin
-            float_msg(self_obj, message_str(140, 4000), 8);
+            float_msg(self_obj, mstr(4000), FLOAT_MSG_YELLOW);
             set_map_var(0, map_var(0) bwor 2097152);
             self_talker := 1;
         end
         if ((self_talker == 0) and ((map_var(0) bwand 1048576) != 0) and not((map_var(0) bwand 4194304) != 0)) then begin
-            float_msg(self_obj, message_str(140, 4050), 8);
+            float_msg(self_obj, mstr(4050), FLOAT_MSG_YELLOW);
             set_map_var(0, map_var(0) bwor 4194304);
             self_talker := 1;
         end
         if ((self_talker == 0) and ((map_var(0) bwand 8388608) != 0)) then begin
             probability := random(1, 2);
             if (probability == 1) then begin
-                float_msg(self_obj, message_str(140, 4100), 8);
+                float_msg(self_obj, mstr(4100), FLOAT_MSG_YELLOW);
             end
             self_talker := 1;
         end
-        if (global_var(1008) == 5) then begin
+        if (global_var(GVAR_AQE_HELP_TO_ARAN) == 5) then begin
             if (not(combat_is_initialized)) then begin
                 if (tile_in_tile_rect(29525, 29493, 32125, 32093, tile_num(self_obj))) then begin
                     animate_move_obj_to_tile(self_obj, 28305, 1);
@@ -290,7 +290,7 @@ begin
                 end
             end
         end
-        if ((global_var(1008) == 7) and ((global_var(1000) bwand 4096) != 0)) then begin
+        if ((global_var(GVAR_AQE_HELP_TO_ARAN) == 7) and ((global_var(GVAR_AQE_ALBUQ_EMPEROR_EVENTS_NEXT) bwand 4096) != 0)) then begin
             if (not(combat_is_initialized)) then begin
                 if (local_var(5) == 30309) then begin
                     if (tile_num(self_obj) != 19286) then begin
@@ -397,9 +397,9 @@ begin
             end
         end
     end
-    if (cur_map_index == 23) then begin
-        if (global_var(1008) == 3) then begin
-            if ((global_var(1011) >= 5) and not(combat_is_initialized)) then begin
+    if (CUR_MAP_ALB_LOVA) then begin
+        if (global_var(GVAR_AQE_HELP_TO_ARAN) == 3) then begin
+            if ((global_var(GVAR_AQE_EMPEROR_SOLDIERS_COUNT) >= 5) and not(combat_is_initialized)) then begin
                 if (tile_num(self_obj) != 9302) then begin
                     animate_move_obj_to_tile(self_obj, 9302, 1);
                 end
@@ -420,21 +420,21 @@ end
 
 procedure talk_p_proc
 begin
-    if (cur_map_index == 22) then begin
-        if (((global_var(904) bwand 64) != 0) or ((global_var(1017) bwand 512) != 0)) then begin
-            float_msg(self_obj, message_str(140, random(500, 502)), 8);
+    if (CUR_MAP_ALB_MRTV) then begin
+        if (((global_var(GVAR_AQE_ALBUQ_EMPEROR_EVENTS) bwand 64) != 0) or ((global_var(GVAR_AQE_DIRECTIONS_SETUP) bwand 512) != 0)) then begin
+            float_msg(self_obj, mstr(random(500, 502)), FLOAT_MSG_YELLOW);
         end
     end
-    if (cur_map_index == 25) then begin
-        if ((global_var(1008) >= 3) and (global_var(1008) <= 9)) then begin
-            float_msg(self_obj, message_str(140, random(550, 552)), 8);
+    if (CUR_MAP_ALB_CISA) then begin
+        if ((global_var(GVAR_AQE_HELP_TO_ARAN) >= 3) and (global_var(GVAR_AQE_HELP_TO_ARAN) <= 9)) then begin
+            float_msg(self_obj, mstr(random(550, 552)), FLOAT_MSG_YELLOW);
         end
     end
 end
 
 procedure destroy_p_proc
 begin
-    if ((cur_map_index == 25) and ((global_var(1008) == 8) or (global_var(1008) == 4))) then begin
+    if ((CUR_MAP_ALB_CISA) and ((global_var(GVAR_AQE_HELP_TO_ARAN) == 8) or (global_var(GVAR_AQE_HELP_TO_ARAN) == 4))) then begin
         if ((smycka == 0) and (map_var(5) == 0)) then begin
             set_map_var(5, tile_num(self_obj));
             smycka := 1;
@@ -532,24 +532,24 @@ begin
             smycka := 1;
         end
     end
-    if (cur_map_index == 23) then begin
-        if ((global_var(1008) == 3) and (global_var(1011) >= 5)) then begin
+    if (CUR_MAP_ALB_LOVA) then begin
+        if ((global_var(GVAR_AQE_HELP_TO_ARAN) == 3) and (global_var(GVAR_AQE_EMPEROR_SOLDIERS_COUNT) >= 5)) then begin
         end
         else begin
             if (local_var(5) == 10904) then begin
-                set_global_var(1000, global_var(1000) bwor 128);
+                set_global_var(GVAR_AQE_ALBUQ_EMPEROR_EVENTS_NEXT, global_var(GVAR_AQE_ALBUQ_EMPEROR_EVENTS_NEXT) bwor 128);
             end
             if (local_var(5) == 10899) then begin
-                set_global_var(1000, global_var(1000) bwor 256);
+                set_global_var(GVAR_AQE_ALBUQ_EMPEROR_EVENTS_NEXT, global_var(GVAR_AQE_ALBUQ_EMPEROR_EVENTS_NEXT) bwor 256);
             end
             if (local_var(5) == 11302) then begin
-                set_global_var(1000, global_var(1000) bwor 512);
+                set_global_var(GVAR_AQE_ALBUQ_EMPEROR_EVENTS_NEXT, global_var(GVAR_AQE_ALBUQ_EMPEROR_EVENTS_NEXT) bwor 512);
             end
             if (local_var(5) == 11507) then begin
-                set_global_var(1000, global_var(1000) bwor 1024);
+                set_global_var(GVAR_AQE_ALBUQ_EMPEROR_EVENTS_NEXT, global_var(GVAR_AQE_ALBUQ_EMPEROR_EVENTS_NEXT) bwor 1024);
             end
             if (local_var(5) == 11903) then begin
-                set_global_var(1000, global_var(1000) bwor 2048);
+                set_global_var(GVAR_AQE_ALBUQ_EMPEROR_EVENTS_NEXT, global_var(GVAR_AQE_ALBUQ_EMPEROR_EVENTS_NEXT) bwor 2048);
             end
             if ((smycka == 0) and (map_var(5) == 0)) then begin
                 set_map_var(5, tile_num(self_obj));
@@ -590,13 +590,13 @@ end
 procedure look_at_p_proc
 begin
     script_overrides;
-    display_msg(message_str(140, 100));
+    display_mstr(100);
 end
 
 procedure description_p_proc
 begin
     script_overrides;
-    display_msg(message_str(140, 200));
+    display_mstr(200);
 end
 
 procedure use_skill_on_p_proc
@@ -612,27 +612,27 @@ end
 
 procedure map_enter_p_proc
 begin
-    critter_add_trait(self_obj, 1, 6, 0);
-    critter_add_trait(self_obj, 1, 5, 29);
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 0);
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_IMPERIAL_SOLDIER_NO_TALKING);
     if (local_var(5) == 0) then begin
         set_local_var(5, tile_num(self_obj));
     end
-    if ((global_var(1008) == 1) and ((global_var(904) bwand 64) != 0)) then begin
+    if ((global_var(GVAR_AQE_HELP_TO_ARAN) == 1) and ((global_var(GVAR_AQE_ALBUQ_EMPEROR_EVENTS) bwand 64) != 0)) then begin
         destroy_object(self_obj);
     end
-    if ((global_var(1008) >= 9) and ((global_var(904) bwand 1) != 0)) then begin
+    if ((global_var(GVAR_AQE_HELP_TO_ARAN) >= 9) and ((global_var(GVAR_AQE_ALBUQ_EMPEROR_EVENTS) bwand 1) != 0)) then begin
         destroy_object(self_obj);
     end
-    if ((cur_map_index == 22) and not(metarule(22, 0))) then begin
-        if ((global_var(1008) != 0) or (global_var(993) != 0)) then begin
+    if ((CUR_MAP_ALB_MRTV) and not(is_loading_game)) then begin
+        if ((global_var(GVAR_AQE_HELP_TO_ARAN) != 0) or (global_var(GVAR_AQE_QUEST_ANONYM_1) != 0)) then begin
             destroy_object(self_obj);
         end
-        if (global_var(992) == 4) then begin
+        if (global_var(GVAR_AQE_QUEST_ELISA_3) == 4) then begin
             destroy_object(self_obj);
         end
     end
-    if ((cur_map_index == 23) and not(metarule(22, 0))) then begin
-        if ((global_var(1008) >= 3) or (global_var(998) == 1) or (global_var(1008) == 1)) then begin
+    if ((CUR_MAP_ALB_LOVA) and not(is_loading_game)) then begin
+        if ((global_var(GVAR_AQE_HELP_TO_ARAN) >= 3) or (global_var(GVAR_AQE_ENEMY_EMPEROR) == 1) or (global_var(GVAR_AQE_HELP_TO_ARAN) == 1)) then begin
             destroy_object(self_obj);
         end
     end
@@ -641,17 +641,17 @@ end
 procedure combat_p_proc
 begin
     if (fixed_param == 4) then begin
-        if (cur_map_index == 25) then begin
-            if (global_var(1008) == 6) then begin
+        if (CUR_MAP_ALB_CISA) then begin
+            if (global_var(GVAR_AQE_HELP_TO_ARAN) == 6) then begin
                 if (team_switch1 == 0) then begin
-                    critter_add_trait(self_obj, 1, 6, 145);
+                    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 145);
                     team_switch1 := 1;
                 end
             end
             else begin
-                if ((global_var(1008) == 7) or (global_var(1008) == 8)) then begin
+                if ((global_var(GVAR_AQE_HELP_TO_ARAN) == 7) or (global_var(GVAR_AQE_HELP_TO_ARAN) == 8)) then begin
                     if (team_switch2 == 0) then begin
-                        critter_add_trait(self_obj, 1, 6, 0);
+                        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 0);
                         team_switch2 := 1;
                     end
                 end
@@ -659,4 +659,3 @@ begin
         end
     end
 end
-
