diff --git a/Mods/Resurrection/Original/ACISVOJ1.ssl b/Mods/Resurrection/ReDefine/ACISVOJ1.ssl
index c086638..85d42d3 100644
--- a/Mods/Resurrection/Original/ACISVOJ1.ssl
+++ b/Mods/Resurrection/ReDefine/ACISVOJ1.ssl
@@ -73,12 +73,12 @@ begin
             return 1;
         end
     end
-    if ((party_member_obj(16777779) != 0) and ((global_var(910) bwand 32) == 0)) then begin
+    if ((party_member_obj(16777779) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 32) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777779)) <= 5) then begin
             return 1;
         end
     end
-    if ((party_member_obj(16777811) != 0) and ((global_var(910) bwand 64) == 0)) then begin
+    if ((party_member_obj(16777811) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 64) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777811)) <= 5) then begin
             return 1;
         end
@@ -93,7 +93,7 @@ begin
             return 1;
         end
     end
-    if ((party_member_obj(16777814) != 0) and ((global_var(910) bwand 128) == 0)) then begin
+    if ((party_member_obj(16777814) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 128) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777814)) <= 5) then begin
             return 1;
         end
@@ -151,15 +151,15 @@ end
 
 procedure critter_p_proc
 begin
-    if ((local_var(4) == 1) and (has_trait(1, self_obj, 666) == 1) and obj_can_see_obj(self_obj, dude_obj)) then begin
+    if ((local_var(4) == 1) and (obj_is_visible_flag(self_obj) == 1) and obj_can_see_obj(self_obj, dude_obj)) then begin
         if (critter_is_fleeing(self_obj)) then begin
             if (anim_busy(self_obj) == 0) then begin
                 if ((tile_distance_objs(self_obj, dude_obj) < 8) or obj_can_see_obj(self_obj, dude_obj)) then begin
                     if (anim_busy(self_obj) == 0) then begin
-                        global_temp := rotation_to_tile(tile_num(dude_obj), tile_num(self_obj));
+                        global_temp := rotation_to_tile(dude_tile, tile_num(self_obj));
                         animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(self_obj), global_temp, random(3, 10)), 1);
                         global_temp := (global_temp + 1) % 6;
-                        while ((anim_busy(self_obj) == 0) and (global_temp != rotation_to_tile(tile_num(dude_obj), tile_num(self_obj)))) do begin
+                        while ((anim_busy(self_obj) == 0) and (global_temp != rotation_to_tile(dude_tile, tile_num(self_obj)))) do begin
                             debug_msg("flee loop: rot == " + global_temp);
                             animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(self_obj), global_temp, random(3, 10)), 1);
                             global_temp := (global_temp + 1) % 6;
@@ -169,13 +169,13 @@ begin
             end
         end
         else begin
-            attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+            attack(dude_obj);
         end
     end
-    if ((art_anim(obj_art_fid(self_obj)) != 1) and (tile_num(self_obj) != local_var(7)) and (has_trait(1, self_obj, 666) == 1)) then begin
+    if ((art_anim(obj_art_fid(self_obj)) != 1) and (tile_num(self_obj) != local_var(7)) and (obj_is_visible_flag(self_obj) == 1)) then begin
         animate_move_obj_to_tile(self_obj, local_var(7), 0);
     end
-    if ((art_anim(obj_art_fid(self_obj)) != 1) and (tile_num(self_obj) == local_var(7)) and (local_var(8) != has_trait(1, self_obj, 10)) and (has_trait(1, self_obj, 666) == 1)) then begin
+    if ((art_anim(obj_art_fid(self_obj)) != 1) and (tile_num(self_obj) == local_var(7)) and (local_var(8) != obj_get_rot(self_obj)) and (obj_is_visible_flag(self_obj) == 1)) then begin
         anim(self_obj, 1000, local_var(8));
     end
 end
@@ -189,56 +189,56 @@ end
 
 procedure talk_p_proc
 begin
-    if (global_var(998) == 1) then begin
-        if (get_critter_stat(dude_obj, 34) == 0) then begin
-            float_msg(self_obj, message_str(88, 330), 2);
+    if (global_var(GVAR_AQE_ENEMY_EMPEROR) == 1) then begin
+        if (dude_gender == 0) then begin
+            float_msg(self_obj, mstr(330), FLOAT_MSG_RED);
         end
         else begin
-            float_msg(self_obj, message_str(88, 331), 2);
+            float_msg(self_obj, mstr(331), FLOAT_MSG_RED);
         end
     end
     else begin
         if ((local_var(6) bwand 4096) == 0) then begin
-            if (((global_var(904) bwand 1) != 0) or (((global_var(904) bwand 2) != 0) and not((global_var(904) bwand 8192) != 0))) then begin
+            if (((global_var(GVAR_AQE_ALBUQ_EMPEROR_EVENTS) bwand 1) != 0) or (((global_var(GVAR_AQE_ALBUQ_EMPEROR_EVENTS) bwand 2) != 0) and not((global_var(GVAR_AQE_ALBUQ_EMPEROR_EVENTS) bwand 8192) != 0))) then begin
                 if ((local_var(6) bwand 64) == 0) then begin
-                    if ((global_var(904) bwand 2) != 0) then begin
-                        float_msg(self_obj, message_str(88, 311), 8);
+                    if ((global_var(GVAR_AQE_ALBUQ_EMPEROR_EVENTS) bwand 2) != 0) then begin
+                        float_msg(self_obj, mstr(311), FLOAT_MSG_YELLOW);
                     end
-                    if ((global_var(904) bwand 1) != 0) then begin
-                        float_msg(self_obj, message_str(88, 310), 8);
+                    if ((global_var(GVAR_AQE_ALBUQ_EMPEROR_EVENTS) bwand 1) != 0) then begin
+                        float_msg(self_obj, mstr(310), FLOAT_MSG_YELLOW);
                     end
                     set_local_var(6, local_var(6) bwor 64);
                 end
                 else begin
-                    if ((global_var(904) bwand 1) != 0) then begin
-                        if (global_var(903) != 5) then begin
-                            float_msg(self_obj, message_str(88, random(314, 318)), 8);
+                    if ((global_var(GVAR_AQE_ALBUQ_EMPEROR_EVENTS) bwand 1) != 0) then begin
+                        if (global_var(GVAR_AQE_QUEST_GET_OUR_VAULT_BACK) != 5) then begin
+                            float_msg(self_obj, mstr(random(314, 318)), FLOAT_MSG_YELLOW);
                         end
                         else begin
-                            float_msg(self_obj, message_str(88, random(314, 317)), 8);
+                            float_msg(self_obj, mstr(random(314, 317)), FLOAT_MSG_YELLOW);
                         end
                     end
                     else begin
-                        if (global_var(903) != 5) then begin
-                            float_msg(self_obj, message_str(88, random(312, 316)), 8);
+                        if (global_var(GVAR_AQE_QUEST_GET_OUR_VAULT_BACK) != 5) then begin
+                            float_msg(self_obj, mstr(random(312, 316)), FLOAT_MSG_YELLOW);
                         end
                         else begin
-                            float_msg(self_obj, message_str(88, random(313, 316)), 8);
+                            float_msg(self_obj, mstr(random(313, 316)), FLOAT_MSG_YELLOW);
                         end
                     end
                 end
             end
             else begin
-                if ((party_member_obj(16777814) != 0) and ((global_var(910) bwand 128) == 0)) then begin
-                    float_msg(self_obj, message_str(88, 305), 8);
+                if ((party_member_obj(16777814) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 128) == 0)) then begin
+                    float_msg(self_obj, mstr(305), FLOAT_MSG_YELLOW);
                 end
                 else begin
-                    if (((get_critter_stat(dude_obj, 4) + has_trait(0, dude_obj, 49)) <= 3) and not((local_var(6) bwand 128) == 0)) then begin
-                        float_msg(self_obj, message_str(88, random(302, 303)), 8);
+                    if (((dude_iq + has_trait(TRAIT_PERK, dude_obj, PERK_smooth_talker)) <= 3) and not((local_var(6) bwand 128) == 0)) then begin
+                        float_msg(self_obj, mstr(random(302, 303)), FLOAT_MSG_YELLOW);
                     end
                     else begin
-                        anim(dude_obj, 1000, rotation_to_tile(tile_num(dude_obj), tile_num(self_obj)));
-                        anim(self_obj, 1000, rotation_to_tile(tile_num(self_obj), tile_num(dude_obj)));
+                        anim(dude_obj, 1000, rotation_to_tile(dude_tile, tile_num(self_obj)));
+                        anim(self_obj, 1000, rotation_to_tile(tile_num(self_obj), dude_tile));
                         start_gdialog(88, self_obj, 4, -1, -1);
                         gsay_start;
                         if (local_var(5) == 1) then begin
@@ -254,11 +254,11 @@ begin
             end
         end
         else begin
-            if ((party_member_obj(16777811) != 0) and ((global_var(910) bwand 64) == 0) or ((party_member_obj(16777814) != 0) and ((global_var(910) bwand 128) == 0)) or ((party_member_obj(16777779) != 0) and ((global_var(910) bwand 32) == 0))) then begin
-                float_msg(self_obj, message_str(88, 301), 8);
+            if ((party_member_obj(16777811) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 64) == 0) or ((party_member_obj(16777814) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 128) == 0)) or ((party_member_obj(16777779) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 32) == 0))) then begin
+                float_msg(self_obj, mstr(301), FLOAT_MSG_YELLOW);
             end
             else begin
-                float_msg(self_obj, message_str(88, 300), 8);
+                float_msg(self_obj, mstr(300), FLOAT_MSG_YELLOW);
             end
         end
     end
@@ -269,59 +269,59 @@ procedure destroy_p_proc
 begin
     add_timer_event(cisdoor_obj, 0, 1);
     if (source_obj == dude_obj) then begin
-        set_global_var(5, global_var(5) + 1);
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(4) > (3 * global_var(5))) or (global_var(2) == 1)) and (global_var(1) == 0)) then begin
-            set_global_var(2, 1);
-            set_global_var(3, 0);
+        set_global_var(GVAR_GOOD_MONSTER, global_var(GVAR_GOOD_MONSTER) + 1);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(4) > (3 * global_var(GVAR_GOOD_MONSTER))) or (global_var(GVAR_CHAMPION_REPUTATION) == 1)) and (global_var(GVAR_CHILDKILLER_REPUTATION) == 0)) then begin
+            set_global_var(GVAR_CHAMPION_REPUTATION, 1);
+            set_global_var(GVAR_BERSERKER_REPUTATION, 0);
         end
-        if (metarule(51, self_obj) == 2) then begin
+        if (critter_kill_type(self_obj) == KILL_TYPE_children_kills) then begin
             if (source_obj == dude_obj) then begin
-                set_global_var(2, 0);
-                set_global_var(1, global_var(1) + 1);
-                set_global_var(0, global_var(0) + -15);
+                set_global_var(GVAR_CHAMPION_REPUTATION, 0);
+                set_global_var(GVAR_CHILDKILLER_REPUTATION, global_var(GVAR_CHILDKILLER_REPUTATION) + 1);
+                set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + -15);
                 debug_msg("Player gains " + -15 + " Karma Points.");
-                set_global_var(37, 0);
-                set_global_var(38, 0);
-                set_global_var(39, 0);
-                set_global_var(40, 0);
-                set_global_var(41, 0);
-                set_global_var(42, 0);
-                set_global_var(43, 0);
-                set_global_var(44, 0);
-                set_global_var(45, 0);
-                if (global_var(0) >= 1000) then begin
-                    set_global_var(37, 1);
+                set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+                set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+                set_global_var(GVAR_KARMA_DEFENDER, 0);
+                set_global_var(GVAR_KARMA_WANDERER, 0);
+                set_global_var(GVAR_KARMA_BETRAYER, 0);
+                set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+                set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+                set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+                    set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
                 end
                 else begin
-                    if (global_var(0) >= 750) then begin
-                        set_global_var(38, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
                     end
                     else begin
-                        if (global_var(0) >= 500) then begin
-                            set_global_var(39, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                            set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                         end
                         else begin
-                            if (global_var(0) >= 250) then begin
-                                set_global_var(40, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                                set_global_var(GVAR_KARMA_DEFENDER, 1);
                             end
                             else begin
-                                if (global_var(0) > -250) then begin
-                                    set_global_var(41, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                                    set_global_var(GVAR_KARMA_WANDERER, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -500) then begin
-                                        set_global_var(42, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                        set_global_var(GVAR_KARMA_BETRAYER, 1);
                                     end
                                     else begin
-                                        if (global_var(0) > -750) then begin
-                                            set_global_var(43, 1);
+                                        if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                            set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                         end
                                         else begin
-                                            if (global_var(0) > -1000) then begin
-                                                set_global_var(44, 1);
+                                            if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                                set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                             end
                                             else begin
-                                                set_global_var(45, 1);
+                                                set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                             end
                                         end
                                     end
@@ -330,58 +330,58 @@ begin
                         end
                     end
                 end
-                set_global_var(53, global_var(53) + -8);
+                set_global_var(GVAR_TOWN_REP_EMPORER_CITY, global_var(GVAR_TOWN_REP_EMPORER_CITY) + -8);
                 debug_msg("Added " + -8 + " to Town Rep");
             end
         end
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(5) > (2 * global_var(4))) or (global_var(3) == 1))) then begin
-            set_global_var(3, 1);
-            set_global_var(2, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
+            set_global_var(GVAR_BERSERKER_REPUTATION, 1);
+            set_global_var(GVAR_CHAMPION_REPUTATION, 0);
         end
-        set_global_var(0, global_var(0) + -10);
+        set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + -10);
         debug_msg("Player gains " + -10 + " Karma Points.");
-        set_global_var(37, 0);
-        set_global_var(38, 0);
-        set_global_var(39, 0);
-        set_global_var(40, 0);
-        set_global_var(41, 0);
-        set_global_var(42, 0);
-        set_global_var(43, 0);
-        set_global_var(44, 0);
-        set_global_var(45, 0);
-        if (global_var(0) >= 1000) then begin
-            set_global_var(37, 1);
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+        set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+        set_global_var(GVAR_KARMA_DEFENDER, 0);
+        set_global_var(GVAR_KARMA_WANDERER, 0);
+        set_global_var(GVAR_KARMA_BETRAYER, 0);
+        set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+        set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+            set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
         end
         else begin
-            if (global_var(0) >= 750) then begin
-                set_global_var(38, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
             end
             else begin
-                if (global_var(0) >= 500) then begin
-                    set_global_var(39, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                 end
                 else begin
-                    if (global_var(0) >= 250) then begin
-                        set_global_var(40, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                        set_global_var(GVAR_KARMA_DEFENDER, 1);
                     end
                     else begin
-                        if (global_var(0) > -250) then begin
-                            set_global_var(41, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                            set_global_var(GVAR_KARMA_WANDERER, 1);
                         end
                         else begin
-                            if (global_var(0) > -500) then begin
-                                set_global_var(42, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                set_global_var(GVAR_KARMA_BETRAYER, 1);
                             end
                             else begin
-                                if (global_var(0) > -750) then begin
-                                    set_global_var(43, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -1000) then begin
-                                        set_global_var(44, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                     end
                                     else begin
-                                        set_global_var(45, 1);
+                                        set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                     end
                                 end
                             end
@@ -390,7 +390,7 @@ begin
                 end
             end
         end
-        set_global_var(53, global_var(53) + -5);
+        set_global_var(GVAR_TOWN_REP_EMPORER_CITY, global_var(GVAR_TOWN_REP_EMPORER_CITY) + -5);
         debug_msg("Added " + -5 + " to Town Rep");
     end
 end
@@ -398,13 +398,13 @@ end
 procedure look_at_p_proc
 begin
     script_overrides;
-    display_msg(message_str(88, 100));
+    display_mstr(100);
 end
 
 procedure description_p_proc
 begin
     script_overrides;
-    display_msg(message_str(88, 200));
+    display_mstr(200);
 end
 
 procedure use_skill_on_p_proc
@@ -414,30 +414,30 @@ end
 procedure damage_p_proc
 begin
     if (source_obj == dude_obj) then begin
-        set_global_var(998, 1);
+        set_global_var(GVAR_AQE_ENEMY_EMPEROR, 1);
         set_local_var(4, 1);
     end
 end
 
 procedure map_enter_p_proc
 begin
-    critter_add_trait(self_obj, 1, 6, 282);
-    critter_add_trait(self_obj, 1, 5, 29);
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 282);
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_IMPERIAL_SOLDIER_NO_TALKING);
     if (local_var(7) == 0) then begin
         set_local_var(7, tile_num(self_obj));
-        set_local_var(8, has_trait(1, self_obj, 10));
+        set_local_var(8, obj_get_rot(self_obj));
     end
-    if (not(metarule(22, 0))) then begin
+    if (not(is_loading_game)) then begin
         if ((local_var(6) bwand 4096) != 0) then begin
             set_local_var(6, local_var(6) bwand (-1 - 4096));
         end
-        if ((global_var(1008) >= 2) and (global_var(1008) != 10)) then begin
-            if (has_trait(1, self_obj, 666) == 1) then begin
+        if ((global_var(GVAR_AQE_HELP_TO_ARAN) >= 2) and (global_var(GVAR_AQE_HELP_TO_ARAN) != 10)) then begin
+            if (obj_is_visible_flag(self_obj) == 1) then begin
                 set_obj_visibility(self_obj, 1);
             end
         end
         else begin
-            if (has_trait(1, self_obj, 666) == 0) then begin
+            if (obj_is_visible_flag(self_obj) == 0) then begin
                 set_obj_visibility(self_obj, 0);
             end
         end
@@ -446,41 +446,41 @@ end
 
 procedure timed_event_p_proc
 begin
-    if (has_trait(1, self_obj, 666) == 1) then begin
+    if (obj_is_visible_flag(self_obj) == 1) then begin
         if (fixed_param == 1) then begin
             dialogue_system_enter;
         end
         if (fixed_param == 2) then begin
-            if (((global_var(904) bwand 1) != 0) or (((global_var(904) bwand 2) != 0) and not((global_var(904) bwand 8192) != 0))) then begin
+            if (((global_var(GVAR_AQE_ALBUQ_EMPEROR_EVENTS) bwand 1) != 0) or (((global_var(GVAR_AQE_ALBUQ_EMPEROR_EVENTS) bwand 2) != 0) and not((global_var(GVAR_AQE_ALBUQ_EMPEROR_EVENTS) bwand 8192) != 0))) then begin
                 if ((local_var(6) bwand 64) == 0) then begin
-                    if ((global_var(904) bwand 2) != 0) then begin
-                        float_msg(self_obj, message_str(88, 311), 8);
+                    if ((global_var(GVAR_AQE_ALBUQ_EMPEROR_EVENTS) bwand 2) != 0) then begin
+                        float_msg(self_obj, mstr(311), FLOAT_MSG_YELLOW);
                     end
-                    if ((global_var(904) bwand 1) != 0) then begin
-                        float_msg(self_obj, message_str(88, 310), 8);
+                    if ((global_var(GVAR_AQE_ALBUQ_EMPEROR_EVENTS) bwand 1) != 0) then begin
+                        float_msg(self_obj, mstr(310), FLOAT_MSG_YELLOW);
                     end
                     set_local_var(6, local_var(6) bwor 64);
                 end
                 else begin
-                    if ((party_member_obj(16777811) != 0) and ((global_var(910) bwand 64) == 0) or ((party_member_obj(16777814) != 0) and ((global_var(910) bwand 128) == 0)) or ((party_member_obj(16777779) != 0) and ((global_var(910) bwand 32) == 0))) then begin
-                        float_msg(self_obj, message_str(88, 301), 8);
+                    if ((party_member_obj(16777811) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 64) == 0) or ((party_member_obj(16777814) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 128) == 0)) or ((party_member_obj(16777779) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 32) == 0))) then begin
+                        float_msg(self_obj, mstr(301), FLOAT_MSG_YELLOW);
                     end
                     else begin
-                        float_msg(self_obj, message_str(88, 300), 8);
+                        float_msg(self_obj, mstr(300), FLOAT_MSG_YELLOW);
                     end
                 end
             end
             else begin
-                if ((party_member_obj(16777811) != 0) and ((global_var(910) bwand 64) == 0) or ((party_member_obj(16777814) != 0) and ((global_var(910) bwand 128) == 0)) or ((party_member_obj(16777779) != 0) and ((global_var(910) bwand 32) == 0))) then begin
-                    if ((party_member_obj(16777814) != 0) and ((global_var(910) bwand 128) == 0)) then begin
-                        float_msg(self_obj, message_str(88, 304), 8);
+                if ((party_member_obj(16777811) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 64) == 0) or ((party_member_obj(16777814) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 128) == 0)) or ((party_member_obj(16777779) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 32) == 0))) then begin
+                    if ((party_member_obj(16777814) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 128) == 0)) then begin
+                        float_msg(self_obj, mstr(304), FLOAT_MSG_YELLOW);
                     end
                     else begin
-                        float_msg(self_obj, message_str(88, 301), 8);
+                        float_msg(self_obj, mstr(301), FLOAT_MSG_YELLOW);
                     end
                 end
                 else begin
-                    float_msg(self_obj, message_str(88, 300), 8);
+                    float_msg(self_obj, mstr(300), FLOAT_MSG_YELLOW);
                 end
             end
         end
@@ -489,13 +489,13 @@ end
 
 procedure Node998
 begin
-    if (get_critter_stat(dude_obj, 34) == 0) then begin
-        float_msg(self_obj, message_str(88, 400), 2);
+    if (dude_gender == 0) then begin
+        float_msg(self_obj, mstr(400), FLOAT_MSG_RED);
     end
-    if (get_critter_stat(dude_obj, 34) == 1) then begin
-        float_msg(self_obj, message_str(88, 401), 2);
+    if (dude_gender == 1) then begin
+        float_msg(self_obj, mstr(401), FLOAT_MSG_RED);
     end
-    set_global_var(998, 1);
+    set_global_var(GVAR_AQE_ENEMY_EMPEROR, 1);
     set_local_var(4, 1);
 end
 
@@ -506,312 +506,312 @@ end
 
 procedure Node500
 begin
-    gsay_reply(88, 500);
-    if (((local_var(6) bwand 1) == 0) and not((global_var(865) bwand 1) != 0)) then begin
-        giq_option(4, 88, 501, Node550, 50);
+    Reply(500);
+    if (((local_var(6) bwand 1) == 0) and not((global_var(GVAR_AQ_ALBUQ_EVENTS) bwand 1) != 0)) then begin
+        giq_option(4, NAME, 501, Node550, NEUTRAL_REACTION);
     end
-    if (get_critter_stat(dude_obj, 34) == 0) then begin
-        giq_option(4, 88, 502, Node600, 50);
+    if (dude_gender == 0) then begin
+        giq_option(4, NAME, 502, Node600, NEUTRAL_REACTION);
     end
-    if (get_critter_stat(dude_obj, 34) == 1) then begin
-        giq_option(4, 88, 503, Node600, 50);
+    if (dude_gender == 1) then begin
+        giq_option(4, NAME, 503, Node600, NEUTRAL_REACTION);
     end
-    if ((global_var(993) == 2) and ((local_var(6) bwand 4) == 0)) then begin
-        giq_option(4, 88, 504, Node650, 50);
+    if ((global_var(GVAR_AQE_QUEST_ANONYM_1) == 2) and ((local_var(6) bwand 4) == 0)) then begin
+        giq_option(4, NAME, 504, Node650, NEUTRAL_REACTION);
     end
-    if ((global_var(874) == 1) and ((local_var(6) bwand 256) == 0)) then begin
-        giq_option(4, 88, 505, Node800, 50);
+    if ((global_var(GVAR_AQ_QUEST_FIND_FELIX_VELAZQUEZ) == 1) and ((local_var(6) bwand 256) == 0)) then begin
+        giq_option(4, NAME, 505, Node800, NEUTRAL_REACTION);
     end
-    giq_option(4, 88, 506, Node999, 50);
-    giq_option(-3, 88, 507, Node750, 50);
+    giq_option(4, NAME, 506, Node999, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 507, Node750, NEUTRAL_REACTION);
 end
 
 procedure Node550
 begin
-    if (get_critter_stat(dude_obj, 34) == 0) then begin
-        gsay_reply(88, 550);
+    if (dude_gender == 0) then begin
+        Reply(550);
     end
-    if (get_critter_stat(dude_obj, 34) == 1) then begin
-        gsay_reply(88, 551);
+    if (dude_gender == 1) then begin
+        Reply(551);
     end
     set_local_var(6, local_var(6) bwor 1);
-    if (get_critter_stat(dude_obj, 34) == 0) then begin
-        giq_option(4, 88, 552, Node600, 50);
+    if (dude_gender == 0) then begin
+        giq_option(4, NAME, 552, Node600, NEUTRAL_REACTION);
     end
-    if (get_critter_stat(dude_obj, 34) == 1) then begin
-        giq_option(4, 88, 553, Node600, 50);
+    if (dude_gender == 1) then begin
+        giq_option(4, NAME, 553, Node600, NEUTRAL_REACTION);
     end
-    giq_option(4, 88, 554, Node999, 50);
+    giq_option(4, NAME, 554, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node600
 begin
     if ((local_var(6) bwand 2) == 0) then begin
-        if (get_critter_stat(dude_obj, 34) == 0) then begin
-            gsay_reply(88, 600);
+        if (dude_gender == 0) then begin
+            Reply(600);
         end
-        if (get_critter_stat(dude_obj, 34) == 1) then begin
-            gsay_reply(88, 601);
+        if (dude_gender == 1) then begin
+            Reply(601);
         end
         set_local_var(6, local_var(6) bwor 2);
     end
     else begin
-        if (get_critter_stat(dude_obj, 34) == 0) then begin
-            gsay_reply(88, 602);
+        if (dude_gender == 0) then begin
+            Reply(602);
         end
-        if (get_critter_stat(dude_obj, 34) == 1) then begin
-            gsay_reply(88, 603);
+        if (dude_gender == 1) then begin
+            Reply(603);
         end
     end
-    if ((global_var(1029) == 3) or (global_var(1029) == 4)) then begin
-        if ((global_var(1000) bwand --2147483648) != 0) then begin
-            giq_option(4, 88, 614, Node950, 50);
+    if ((global_var(GVAR_AQE_KICK_BENICIO_FROM_ALBUQUERQUE) == 3) or (global_var(GVAR_AQE_KICK_BENICIO_FROM_ALBUQUERQUE) == 4)) then begin
+        if ((global_var(GVAR_AQE_ALBUQ_EMPEROR_EVENTS_NEXT) bwand --2147483648) != 0) then begin
+            giq_option(4, NAME, 614, Node950, NEUTRAL_REACTION);
         end
         else begin
-            giq_option(4, 88, 605, Node1400, 50);
+            giq_option(4, NAME, 605, Node1400, NEUTRAL_REACTION);
         end
     end
     else begin
-        if ((global_var(996) == 3) and ((global_var(904) bwand 64) != 0)) then begin
-            giq_option(4, 88, 605, Node950, 50);
+        if ((global_var(GVAR_AQE_QUEST_ANONYM_3) == 3) and ((global_var(GVAR_AQE_ALBUQ_EMPEROR_EVENTS) bwand 64) != 0)) then begin
+            giq_option(4, NAME, 605, Node950, NEUTRAL_REACTION);
         end
     end
-    if (((global_var(904) bwand 256) != 0) and (global_var(990) == 0) and ((local_var(6) bwand 8) == 0) and ((global_var(1000) bwand 536870912) == 0)) then begin
-        giq_option(4, 88, 604, Node900, 50);
+    if (((global_var(GVAR_AQE_ALBUQ_EMPEROR_EVENTS) bwand 256) != 0) and (global_var(GVAR_AQE_QUEST_ELISA_1) == 0) and ((local_var(6) bwand 8) == 0) and ((global_var(GVAR_AQE_ALBUQ_EMPEROR_EVENTS_NEXT) bwand 536870912) == 0)) then begin
+        giq_option(4, NAME, 604, Node900, NEUTRAL_REACTION);
     end
-    if ((global_var(996) == 0) and not((global_var(904) bwand 1) != 0)) then begin
-        if ((global_var(904) bwand 2) != 0) then begin
-            if ((global_var(990) > 0) and ((local_var(6) bwand 16) == 0) and ((global_var(1000) bwand --2147483648) == 0)) then begin
-                giq_option(4, 88, 605, Node1000, 50);
+    if ((global_var(GVAR_AQE_QUEST_ANONYM_3) == 0) and not((global_var(GVAR_AQE_ALBUQ_EMPEROR_EVENTS) bwand 1) != 0)) then begin
+        if ((global_var(GVAR_AQE_ALBUQ_EMPEROR_EVENTS) bwand 2) != 0) then begin
+            if ((global_var(GVAR_AQE_QUEST_ELISA_1) > 0) and ((local_var(6) bwand 16) == 0) and ((global_var(GVAR_AQE_ALBUQ_EMPEROR_EVENTS_NEXT) bwand --2147483648) == 0)) then begin
+                giq_option(4, NAME, 605, Node1000, NEUTRAL_REACTION);
             end
         end
         else begin
-            if ((global_var(990) > 0) and ((local_var(6) bwand 16) == 0)) then begin
-                giq_option(4, 88, 605, Node950, 50);
+            if ((global_var(GVAR_AQE_QUEST_ELISA_1) > 0) and ((local_var(6) bwand 16) == 0)) then begin
+                giq_option(4, NAME, 605, Node950, NEUTRAL_REACTION);
             end
         end
     end
-    if (global_var(997) == 2) then begin
-        if (get_critter_stat(dude_obj, 34) == 0) then begin
-            giq_option(4, 88, 606, Node1050, 50);
+    if (global_var(GVAR_AQE_QUEST_B_SNEED) == 2) then begin
+        if (dude_gender == 0) then begin
+            giq_option(4, NAME, 606, Node1050, NEUTRAL_REACTION);
         end
-        if (get_critter_stat(dude_obj, 34) == 1) then begin
-            giq_option(4, 88, 607, Node1050, 50);
+        if (dude_gender == 1) then begin
+            giq_option(4, NAME, 607, Node1050, NEUTRAL_REACTION);
         end
     end
-    if (global_var(874) == 1) then begin
-        giq_option(4, 88, 608, Node1050, 50);
+    if (global_var(GVAR_AQ_QUEST_FIND_FELIX_VELAZQUEZ) == 1) then begin
+        giq_option(4, NAME, 608, Node1050, NEUTRAL_REACTION);
     end
-    if ((global_var(993) == 2) and ((local_var(6) bwand 4) == 0)) then begin
-        giq_option(4, 88, 609, Node650, 50);
+    if ((global_var(GVAR_AQE_QUEST_ANONYM_1) == 2) and ((local_var(6) bwand 4) == 0)) then begin
+        giq_option(4, NAME, 609, Node650, NEUTRAL_REACTION);
     end
     if ((item_caps_total(dude_obj) > 699) and not((local_var(6) bwand 2048) == 0)) then begin
-        giq_option(4, 88, 610, pay_700_caps, 50);
+        giq_option(4, NAME, 610, pay_700_caps, NEUTRAL_REACTION);
     end
     if ((item_caps_total(dude_obj) > 499) and not((local_var(6) bwand 1024) == 0)) then begin
-        giq_option(4, 88, 611, pay_500_caps, 50);
+        giq_option(4, NAME, 611, pay_500_caps, NEUTRAL_REACTION);
     end
     if ((item_caps_total(dude_obj) > 299) and not((local_var(6) bwand 512) == 0)) then begin
-        giq_option(4, 88, 612, pay_300_caps, 50);
+        giq_option(4, NAME, 612, pay_300_caps, NEUTRAL_REACTION);
     end
-    giq_option(4, 88, 613, Node1150, 50);
+    giq_option(4, NAME, 613, Node1150, NEUTRAL_REACTION);
 end
 
 procedure Node650
 begin
-    gsay_reply(88, 650);
+    Reply(650);
     set_local_var(6, local_var(6) bwor 4);
-    if (get_critter_stat(dude_obj, 34) == 0) then begin
-        giq_option(4, 88, 651, Node998, 50);
+    if (dude_gender == 0) then begin
+        giq_option(4, NAME, 651, Node998, NEUTRAL_REACTION);
     end
-    if (get_critter_stat(dude_obj, 34) == 1) then begin
-        giq_option(4, 88, 652, Node998, 50);
+    if (dude_gender == 1) then begin
+        giq_option(4, NAME, 652, Node998, NEUTRAL_REACTION);
     end
-    giq_option(4, 88, 653, Node700, 50);
-    giq_option(4, 88, 654, Node999, 50);
+    giq_option(4, NAME, 653, Node700, NEUTRAL_REACTION);
+    giq_option(4, NAME, 654, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node700
 begin
-    gsay_reply(88, 700);
-    giq_option(4, 88, 701, Node998, 50);
-    giq_option(4, 88, 702, Node999, 50);
+    Reply(700);
+    giq_option(4, NAME, 701, Node998, NEUTRAL_REACTION);
+    giq_option(4, NAME, 702, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node750
 begin
-    gsay_reply(88, 750);
+    Reply(750);
     set_local_var(6, local_var(6) bwor 128);
-    giq_option(-3, 88, 751, Node998, 50);
-    giq_option(-3, 88, 752, Node999, 50);
+    giq_option(-3, NAME, 751, Node998, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 752, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node800
 begin
-    gsay_reply(88, 800);
-    giq_option(4, 88, 801, Node850, 50);
-    giq_option(4, 88, 802, Node999, 50);
+    Reply(800);
+    giq_option(4, NAME, 801, Node850, NEUTRAL_REACTION);
+    giq_option(4, NAME, 802, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node850
 begin
-    gsay_reply(88, 850);
+    Reply(850);
     set_local_var(6, local_var(6) bwor 256);
-    set_global_var(904, global_var(904) bwor 16384);
-    giq_option(4, 88, 851, Node999, 50);
+    set_global_var(GVAR_AQE_ALBUQ_EMPEROR_EVENTS, global_var(GVAR_AQE_ALBUQ_EMPEROR_EVENTS) bwor 16384);
+    giq_option(4, NAME, 851, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node900
 begin
-    gsay_reply(88, 900);
+    Reply(900);
     set_local_var(6, local_var(6) bwor 8);
-    set_global_var(1000, global_var(1000) bwor 536870912);
-    if ((global_var(1017) bwand 262144) == 0) then begin
-        giq_option(4, 88, 901, Node1250, 50);
+    set_global_var(GVAR_AQE_ALBUQ_EMPEROR_EVENTS_NEXT, global_var(GVAR_AQE_ALBUQ_EMPEROR_EVENTS_NEXT) bwor 536870912);
+    if ((global_var(GVAR_AQE_DIRECTIONS_SETUP) bwand 262144) == 0) then begin
+        giq_option(4, NAME, 901, Node1250, NEUTRAL_REACTION);
     end
     else begin
-        giq_option(4, 88, 901, Node999_Open_Gate, 50);
+        giq_option(4, NAME, 901, Node999_Open_Gate, NEUTRAL_REACTION);
     end
 end
 
 procedure Node950
 begin
-    gsay_reply(88, 950);
-    giq_option(4, 88, 951, Node999_Open_Gate, 50);
+    Reply(950);
+    giq_option(4, NAME, 951, Node999_Open_Gate, NEUTRAL_REACTION);
 end
 
 procedure Node1000
 begin
-    if (get_critter_stat(dude_obj, 34) == 0) then begin
-        gsay_reply(88, 1000);
+    if (dude_gender == 0) then begin
+        Reply(1000);
     end
-    if (get_critter_stat(dude_obj, 34) == 1) then begin
-        gsay_reply(88, 1001);
+    if (dude_gender == 1) then begin
+        Reply(1001);
     end
     set_local_var(6, local_var(6) bwor 16);
-    set_global_var(1000, global_var(1000) bwor --2147483648);
-    giq_option(4, 88, 1002, Node1300, 50);
+    set_global_var(GVAR_AQE_ALBUQ_EMPEROR_EVENTS_NEXT, global_var(GVAR_AQE_ALBUQ_EMPEROR_EVENTS_NEXT) bwor --2147483648);
+    giq_option(4, NAME, 1002, Node1300, NEUTRAL_REACTION);
     if ((item_caps_total(dude_obj) > 699) and not((local_var(6) bwand 512) == 0)) then begin
-        giq_option(4, 88, 1003, pay_700_caps, 50);
+        giq_option(4, NAME, 1003, pay_700_caps, NEUTRAL_REACTION);
     end
     if ((item_caps_total(dude_obj) > 499) and not((local_var(6) bwand 1024) == 0)) then begin
-        giq_option(4, 88, 1004, pay_500_caps, 50);
+        giq_option(4, NAME, 1004, pay_500_caps, NEUTRAL_REACTION);
     end
     if ((item_caps_total(dude_obj) > 299) and not((local_var(6) bwand 2048) == 0)) then begin
-        giq_option(4, 88, 1005, pay_300_caps, 50);
+        giq_option(4, NAME, 1005, pay_300_caps, NEUTRAL_REACTION);
     end
-    giq_option(4, 88, 1006, Node999, 50);
+    giq_option(4, NAME, 1006, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node1050
 begin
-    gsay_reply(88, 1050);
-    giq_option(4, 88, 1051, Node999_Open_Gate, 50);
+    Reply(1050);
+    giq_option(4, NAME, 1051, Node999_Open_Gate, NEUTRAL_REACTION);
 end
 
 procedure Node1100
 begin
-    gsay_reply(88, 1100);
-    if ((item_caps_total(dude_obj) > 699) and (has_skill(dude_obj, 15) < 50)) then begin
-        giq_option(4, 88, 1101, pay_700_caps, 50);
+    Reply(1100);
+    if ((item_caps_total(dude_obj) > 699) and (dude_skill(SKILL_BARTER) < 50)) then begin
+        giq_option(4, NAME, 1101, pay_700_caps, NEUTRAL_REACTION);
     end
     else begin
-        if ((item_caps_total(dude_obj) > 499) and (has_skill(dude_obj, 15) < 90)) then begin
-            giq_option(4, 88, 1102, pay_500_caps, 50);
+        if ((item_caps_total(dude_obj) > 499) and (dude_skill(SKILL_BARTER) < 90)) then begin
+            giq_option(4, NAME, 1102, pay_500_caps, NEUTRAL_REACTION);
         end
-        if ((item_caps_total(dude_obj) > 299) and (has_skill(dude_obj, 15) > 89)) then begin
-            giq_option(4, 88, 1103, pay_300_caps, 50);
+        if ((item_caps_total(dude_obj) > 299) and (dude_skill(SKILL_BARTER) > 89)) then begin
+            giq_option(4, NAME, 1103, pay_300_caps, NEUTRAL_REACTION);
         end
     end
-    giq_option(4, 88, 1104, Node999, 50);
+    giq_option(4, NAME, 1104, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node1150
 begin
-    gsay_reply(88, 1150);
+    Reply(1150);
     if ((local_var(6) bwand 32) == 0) then begin
-        giq_option(4, 88, 1151, Node1100, 50);
+        giq_option(4, NAME, 1151, Node1100, NEUTRAL_REACTION);
     end
-    giq_option(4, 88, 1152, Node999, 50);
+    giq_option(4, NAME, 1152, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node1200
 begin
-    if ((global_var(1017) bwand 262144) == 0) then begin
-        gsay_reply(88, 1200);
+    if ((global_var(GVAR_AQE_DIRECTIONS_SETUP) bwand 262144) == 0) then begin
+        Reply(1200);
         set_local_var(6, local_var(6) bwor 32);
     end
     else begin
-        gsay_reply(88, 1201);
+        Reply(1201);
     end
-    if ((global_var(1017) bwand 262144) == 0) then begin
-        giq_option(4, 88, 1202, Node1250, 50);
+    if ((global_var(GVAR_AQE_DIRECTIONS_SETUP) bwand 262144) == 0) then begin
+        giq_option(4, NAME, 1202, Node1250, NEUTRAL_REACTION);
     end
     else begin
-        giq_option(4, 88, 1202, Node999_Open_Gate, 50);
+        giq_option(4, NAME, 1202, Node999_Open_Gate, NEUTRAL_REACTION);
     end
 end
 
 procedure Node1250
 begin
-    if (get_critter_stat(dude_obj, 34) == 0) then begin
-        gsay_reply(88, 1250);
+    if (dude_gender == 0) then begin
+        Reply(1250);
     end
-    if (get_critter_stat(dude_obj, 34) == 1) then begin
-        gsay_reply(88, 1251);
+    if (dude_gender == 1) then begin
+        Reply(1251);
     end
-    set_global_var(1017, global_var(1017) bwor 262144);
-    if (get_critter_stat(dude_obj, 34) == 0) then begin
-        giq_option(4, 88, 1252, Node999_Open_Gate, 50);
+    set_global_var(GVAR_AQE_DIRECTIONS_SETUP, global_var(GVAR_AQE_DIRECTIONS_SETUP) bwor 262144);
+    if (dude_gender == 0) then begin
+        giq_option(4, NAME, 1252, Node999_Open_Gate, NEUTRAL_REACTION);
     end
-    if (get_critter_stat(dude_obj, 34) == 1) then begin
-        giq_option(4, 88, 1253, Node999_Open_Gate, 50);
+    if (dude_gender == 1) then begin
+        giq_option(4, NAME, 1253, Node999_Open_Gate, NEUTRAL_REACTION);
     end
 end
 
 procedure Node1300
 begin
-    if (get_critter_stat(dude_obj, 34) == 0) then begin
-        gsay_reply(88, 1300);
+    if (dude_gender == 0) then begin
+        Reply(1300);
     end
-    if (get_critter_stat(dude_obj, 34) == 1) then begin
-        gsay_reply(88, 1301);
+    if (dude_gender == 1) then begin
+        Reply(1301);
     end
-    giq_option(4, 88, 1302, Node998, 50);
-    giq_option(4, 88, 1303, Node999, 50);
+    giq_option(4, NAME, 1302, Node998, NEUTRAL_REACTION);
+    giq_option(4, NAME, 1303, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node1350
 begin
-    gsay_reply(88, 1350);
-    if (get_critter_stat(dude_obj, 34) == 0) then begin
-        giq_option(4, 88, 1351, Node600, 50);
+    Reply(1350);
+    if (dude_gender == 0) then begin
+        giq_option(4, NAME, 1351, Node600, NEUTRAL_REACTION);
     end
-    if (get_critter_stat(dude_obj, 34) == 1) then begin
-        giq_option(4, 88, 1352, Node600, 50);
+    if (dude_gender == 1) then begin
+        giq_option(4, NAME, 1352, Node600, NEUTRAL_REACTION);
     end
-    if ((global_var(993) == 2) and ((local_var(6) bwand 4) == 0)) then begin
-        giq_option(4, 88, 1353, Node650, 50);
+    if ((global_var(GVAR_AQE_QUEST_ANONYM_1) == 2) and ((local_var(6) bwand 4) == 0)) then begin
+        giq_option(4, NAME, 1353, Node650, NEUTRAL_REACTION);
     end
-    if ((global_var(874) == 1) and ((local_var(6) bwand 256) == 0)) then begin
-        giq_option(4, 88, 1354, Node800, 50);
+    if ((global_var(GVAR_AQ_QUEST_FIND_FELIX_VELAZQUEZ) == 1) and ((local_var(6) bwand 256) == 0)) then begin
+        giq_option(4, NAME, 1354, Node800, NEUTRAL_REACTION);
     end
-    giq_option(4, 88, 1355, Node999, 50);
-    giq_option(-3, 88, 1356, Node999, 50);
+    giq_option(4, NAME, 1355, Node999, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 1356, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node1400
 begin
-    gsay_reply(88, 1400);
-    set_global_var(1000, global_var(1000) bwor --2147483648);
-    giq_option(4, 88, 1401, Node1450, 50);
-    giq_option(4, 88, 1402, Node999_Open_Gate, 50);
+    Reply(1400);
+    set_global_var(GVAR_AQE_ALBUQ_EMPEROR_EVENTS_NEXT, global_var(GVAR_AQE_ALBUQ_EMPEROR_EVENTS_NEXT) bwor --2147483648);
+    giq_option(4, NAME, 1401, Node1450, NEUTRAL_REACTION);
+    giq_option(4, NAME, 1402, Node999_Open_Gate, NEUTRAL_REACTION);
 end
 
 procedure Node1450
 begin
-    gsay_reply(88, 1450);
-    giq_option(4, 88, 1451, Node999_Open_Gate, 50);
+    Reply(1450);
+    giq_option(4, NAME, 1451, Node999_Open_Gate, NEUTRAL_REACTION);
 end
 
 procedure pay_300_caps
@@ -847,4 +847,3 @@ begin
     set_local_var(6, local_var(6) bwor 4096);
     call Node999();
 end
-
