diff --git a/Mods/Resurrection/Original/ADIVLV.ssl b/Mods/Resurrection/ReDefine/ADIVLV.ssl
index 2f8ed83..b73bc6b 100644
--- a/Mods/Resurrection/Original/ADIVLV.ssl
+++ b/Mods/Resurrection/ReDefine/ADIVLV.ssl
@@ -53,12 +53,12 @@ begin
             return 1;
         end
     end
-    if ((party_member_obj(16777779) != 0) and ((global_var(910) bwand 32) == 0)) then begin
+    if ((party_member_obj(16777779) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 32) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777779)) <= 5) then begin
             return 1;
         end
     end
-    if ((party_member_obj(16777811) != 0) and ((global_var(910) bwand 64) == 0)) then begin
+    if ((party_member_obj(16777811) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 64) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777811)) <= 5) then begin
             return 1;
         end
@@ -73,7 +73,7 @@ begin
             return 1;
         end
     end
-    if ((party_member_obj(16777814) != 0) and ((global_var(910) bwand 128) == 0)) then begin
+    if ((party_member_obj(16777814) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 128) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777814)) <= 5) then begin
             return 1;
         end
@@ -130,15 +130,15 @@ end
 
 procedure critter_p_proc
 begin
-    if ((global_var(862) == 1) or (global_var(857) > 2) and obj_can_see_obj(self_obj, dude_obj)) then begin
+    if ((global_var(GVAR_AQ_ENEMY_PRIMITIVES) == 1) or (global_var(GVAR_FRIENDLY_FIRE) > 2) and obj_can_see_obj(self_obj, dude_obj)) then begin
         if (critter_is_fleeing(self_obj)) then begin
             if (anim_busy(self_obj) == 0) then begin
                 if ((tile_distance_objs(self_obj, dude_obj) < 8) or obj_can_see_obj(self_obj, dude_obj)) then begin
                     if (anim_busy(self_obj) == 0) then begin
-                        global_temp := rotation_to_tile(tile_num(dude_obj), tile_num(self_obj));
+                        global_temp := rotation_to_tile(dude_tile, tile_num(self_obj));
                         animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(self_obj), global_temp, random(3, 10)), 1);
                         global_temp := (global_temp + 1) % 6;
-                        while ((anim_busy(self_obj) == 0) and (global_temp != rotation_to_tile(tile_num(dude_obj), tile_num(self_obj)))) do begin
+                        while ((anim_busy(self_obj) == 0) and (global_temp != rotation_to_tile(dude_tile, tile_num(self_obj)))) do begin
                             debug_msg("flee loop: rot == " + global_temp);
                             animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(self_obj), global_temp, random(3, 10)), 1);
                             global_temp := (global_temp + 1) % 6;
@@ -148,26 +148,26 @@ begin
             end
         end
         else begin
-            attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+            attack(dude_obj);
         end
     end
-    if (not((global_var(866) bwand 256) != 0) and (cur_map_index == 92) and (global_var(889) == 9)) then begin
-        float_msg(self_obj, message_str(1662, 403), 8);
-        set_global_var(866, global_var(866) bwor 256);
+    if (not((global_var(GVAR_AQ_ALBUQ_SPEECHES) bwand 256) != 0) and (CUR_MAP_ALB_DIV1) and (global_var(GVAR_AQ_HUNTINGS_RESULTS) == 9)) then begin
+        float_msg(self_obj, mstr(403), FLOAT_MSG_YELLOW);
+        set_global_var(GVAR_AQ_ALBUQ_SPEECHES, global_var(GVAR_AQ_ALBUQ_SPEECHES) bwor 256);
         script_overrides;
-        display_msg(message_str(1662, 404));
+        display_mstr(404);
     end
-    if (not((global_var(866) bwand 256) != 0) and (cur_map_index == 93) and (global_var(889) == 9)) then begin
-        float_msg(self_obj, message_str(1662, 403), 8);
-        set_global_var(866, global_var(866) bwor 256);
+    if (not((global_var(GVAR_AQ_ALBUQ_SPEECHES) bwand 256) != 0) and (CUR_MAP_ALB_DIV2) and (global_var(GVAR_AQ_HUNTINGS_RESULTS) == 9)) then begin
+        float_msg(self_obj, mstr(403), FLOAT_MSG_YELLOW);
+        set_global_var(GVAR_AQ_ALBUQ_SPEECHES, global_var(GVAR_AQ_ALBUQ_SPEECHES) bwor 256);
         script_overrides;
-        display_msg(message_str(1662, 404));
+        display_mstr(404);
     end
-    if (not((global_var(866) bwand 256) != 0) and (cur_map_index == 94) and (global_var(889) == 12)) then begin
-        float_msg(self_obj, message_str(1662, 403), 8);
-        set_global_var(866, global_var(866) bwor 256);
+    if (not((global_var(GVAR_AQ_ALBUQ_SPEECHES) bwand 256) != 0) and (CUR_MAP_ALB_DIV3) and (global_var(GVAR_AQ_HUNTINGS_RESULTS) == 12)) then begin
+        float_msg(self_obj, mstr(403), FLOAT_MSG_YELLOW);
+        set_global_var(GVAR_AQ_ALBUQ_SPEECHES, global_var(GVAR_AQ_ALBUQ_SPEECHES) bwor 256);
         script_overrides;
-        display_msg(message_str(1662, 404));
+        display_mstr(404);
     end
 end
 
@@ -175,7 +175,7 @@ procedure pickup_p_proc
 begin
     if (source_obj == dude_obj) then begin
         if (counter > 2) then begin
-            set_global_var(857, 3);
+            set_global_var(GVAR_FRIENDLY_FIRE, 3);
         end
         counter := counter + 1;
     end
@@ -183,20 +183,20 @@ end
 
 procedure talk_p_proc
 begin
-    if ((cur_map_index == 92) and (global_var(889) == 9) or ((cur_map_index == 93) and (global_var(889) == 9)) or ((cur_map_index == 94) and (global_var(889) == 12))) then begin
-        if (get_critter_stat(dude_obj, 34) == 0) then begin
-            float_msg(self_obj, message_str(1662, random(330, 332)), 8);
+    if ((CUR_MAP_ALB_DIV1) and (global_var(GVAR_AQ_HUNTINGS_RESULTS) == 9) or ((CUR_MAP_ALB_DIV2) and (global_var(GVAR_AQ_HUNTINGS_RESULTS) == 9)) or ((CUR_MAP_ALB_DIV3) and (global_var(GVAR_AQ_HUNTINGS_RESULTS) == 12))) then begin
+        if (dude_gender == 0) then begin
+            float_msg(self_obj, mstr(random(330, 332)), FLOAT_MSG_YELLOW);
         end
-        if (get_critter_stat(dude_obj, 34) == 1) then begin
-            float_msg(self_obj, message_str(1662, random(332, 334)), 8);
+        if (dude_gender == 1) then begin
+            float_msg(self_obj, mstr(random(332, 334)), FLOAT_MSG_YELLOW);
         end
     end
-    if ((cur_map_index == 95) and (global_var(899) == 6)) then begin
-        if (get_critter_stat(dude_obj, 34) == 0) then begin
-            float_msg(self_obj, message_str(1662, random(340, 342)), 8);
+    if ((CUR_MAP_ALB_NEM4) and (global_var(GVAR_AQ_HOSPITAL_GIVERS_STATUS) == 6)) then begin
+        if (dude_gender == 0) then begin
+            float_msg(self_obj, mstr(random(340, 342)), FLOAT_MSG_YELLOW);
         end
-        if (get_critter_stat(dude_obj, 34) == 1) then begin
-            float_msg(self_obj, message_str(1662, random(341, 343)), 8);
+        if (dude_gender == 1) then begin
+            float_msg(self_obj, mstr(random(341, 343)), FLOAT_MSG_YELLOW);
         end
     end
 end
@@ -204,59 +204,59 @@ end
 procedure destroy_p_proc
 begin
     if (source_obj == dude_obj) then begin
-        set_global_var(5, global_var(5) + 1);
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(4) > (3 * global_var(5))) or (global_var(2) == 1)) and (global_var(1) == 0)) then begin
-            set_global_var(2, 1);
-            set_global_var(3, 0);
+        set_global_var(GVAR_GOOD_MONSTER, global_var(GVAR_GOOD_MONSTER) + 1);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(4) > (3 * global_var(GVAR_GOOD_MONSTER))) or (global_var(GVAR_CHAMPION_REPUTATION) == 1)) and (global_var(GVAR_CHILDKILLER_REPUTATION) == 0)) then begin
+            set_global_var(GVAR_CHAMPION_REPUTATION, 1);
+            set_global_var(GVAR_BERSERKER_REPUTATION, 0);
         end
-        if (metarule(51, self_obj) == 2) then begin
+        if (critter_kill_type(self_obj) == KILL_TYPE_children_kills) then begin
             if (source_obj == dude_obj) then begin
-                set_global_var(2, 0);
-                set_global_var(1, global_var(1) + 1);
-                set_global_var(0, global_var(0) + -15);
+                set_global_var(GVAR_CHAMPION_REPUTATION, 0);
+                set_global_var(GVAR_CHILDKILLER_REPUTATION, global_var(GVAR_CHILDKILLER_REPUTATION) + 1);
+                set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + -15);
                 debug_msg("Player gains " + -15 + " Karma Points.");
-                set_global_var(37, 0);
-                set_global_var(38, 0);
-                set_global_var(39, 0);
-                set_global_var(40, 0);
-                set_global_var(41, 0);
-                set_global_var(42, 0);
-                set_global_var(43, 0);
-                set_global_var(44, 0);
-                set_global_var(45, 0);
-                if (global_var(0) >= 1000) then begin
-                    set_global_var(37, 1);
+                set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+                set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+                set_global_var(GVAR_KARMA_DEFENDER, 0);
+                set_global_var(GVAR_KARMA_WANDERER, 0);
+                set_global_var(GVAR_KARMA_BETRAYER, 0);
+                set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+                set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+                set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+                    set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
                 end
                 else begin
-                    if (global_var(0) >= 750) then begin
-                        set_global_var(38, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
                     end
                     else begin
-                        if (global_var(0) >= 500) then begin
-                            set_global_var(39, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                            set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                         end
                         else begin
-                            if (global_var(0) >= 250) then begin
-                                set_global_var(40, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                                set_global_var(GVAR_KARMA_DEFENDER, 1);
                             end
                             else begin
-                                if (global_var(0) > -250) then begin
-                                    set_global_var(41, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                                    set_global_var(GVAR_KARMA_WANDERER, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -500) then begin
-                                        set_global_var(42, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                        set_global_var(GVAR_KARMA_BETRAYER, 1);
                                     end
                                     else begin
-                                        if (global_var(0) > -750) then begin
-                                            set_global_var(43, 1);
+                                        if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                            set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                         end
                                         else begin
-                                            if (global_var(0) > -1000) then begin
-                                                set_global_var(44, 1);
+                                            if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                                set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                             end
                                             else begin
-                                                set_global_var(45, 1);
+                                                set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                             end
                                         end
                                     end
@@ -265,58 +265,58 @@ begin
                         end
                     end
                 end
-                set_global_var(51, global_var(51) + -8);
+                set_global_var(GVAR_TOWN_REP_ALBUQUERQUE, global_var(GVAR_TOWN_REP_ALBUQUERQUE) + -8);
                 debug_msg("Added " + -8 + " to Town Rep");
             end
         end
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(5) > (2 * global_var(4))) or (global_var(3) == 1))) then begin
-            set_global_var(3, 1);
-            set_global_var(2, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
+            set_global_var(GVAR_BERSERKER_REPUTATION, 1);
+            set_global_var(GVAR_CHAMPION_REPUTATION, 0);
         end
-        set_global_var(0, global_var(0) + -10);
+        set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + -10);
         debug_msg("Player gains " + -10 + " Karma Points.");
-        set_global_var(37, 0);
-        set_global_var(38, 0);
-        set_global_var(39, 0);
-        set_global_var(40, 0);
-        set_global_var(41, 0);
-        set_global_var(42, 0);
-        set_global_var(43, 0);
-        set_global_var(44, 0);
-        set_global_var(45, 0);
-        if (global_var(0) >= 1000) then begin
-            set_global_var(37, 1);
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+        set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+        set_global_var(GVAR_KARMA_DEFENDER, 0);
+        set_global_var(GVAR_KARMA_WANDERER, 0);
+        set_global_var(GVAR_KARMA_BETRAYER, 0);
+        set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+        set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+            set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
         end
         else begin
-            if (global_var(0) >= 750) then begin
-                set_global_var(38, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
             end
             else begin
-                if (global_var(0) >= 500) then begin
-                    set_global_var(39, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                 end
                 else begin
-                    if (global_var(0) >= 250) then begin
-                        set_global_var(40, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                        set_global_var(GVAR_KARMA_DEFENDER, 1);
                     end
                     else begin
-                        if (global_var(0) > -250) then begin
-                            set_global_var(41, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                            set_global_var(GVAR_KARMA_WANDERER, 1);
                         end
                         else begin
-                            if (global_var(0) > -500) then begin
-                                set_global_var(42, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                set_global_var(GVAR_KARMA_BETRAYER, 1);
                             end
                             else begin
-                                if (global_var(0) > -750) then begin
-                                    set_global_var(43, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -1000) then begin
-                                        set_global_var(44, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                     end
                                     else begin
-                                        set_global_var(45, 1);
+                                        set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                     end
                                 end
                             end
@@ -325,55 +325,55 @@ begin
                 end
             end
         end
-        set_global_var(51, global_var(51) + -5);
+        set_global_var(GVAR_TOWN_REP_ALBUQUERQUE, global_var(GVAR_TOWN_REP_ALBUQUERQUE) + -5);
         debug_msg("Added " + -5 + " to Town Rep");
     end
-    if (cur_map_index == 92) then begin
+    if (CUR_MAP_ALB_DIV1) then begin
         if (local_var(4) == 17074) then begin
-            set_global_var(900, global_var(900) bwor 4194304);
+            set_global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS, global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS) bwor 4194304);
         end
         if (local_var(4) == 18476) then begin
-            set_global_var(900, global_var(900) bwor 8388608);
+            set_global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS, global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS) bwor 8388608);
         end
         if (local_var(4) == 17471) then begin
-            set_global_var(900, global_var(900) bwor 16777216);
+            set_global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS, global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS) bwor 16777216);
         end
     end
-    if (cur_map_index == 93) then begin
+    if (CUR_MAP_ALB_DIV2) then begin
         if (local_var(4) == 20727) then begin
-            set_global_var(900, global_var(900) bwor 4194304);
+            set_global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS, global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS) bwor 4194304);
         end
         if (local_var(4) == 20531) then begin
-            set_global_var(900, global_var(900) bwor 8388608);
+            set_global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS, global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS) bwor 8388608);
         end
         if (local_var(4) == 21532) then begin
-            set_global_var(900, global_var(900) bwor 16777216);
+            set_global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS, global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS) bwor 16777216);
         end
     end
-    if (cur_map_index == 94) then begin
+    if (CUR_MAP_ALB_DIV3) then begin
         if (local_var(4) == 26504) then begin
-            set_global_var(900, global_var(900) bwor 4194304);
+            set_global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS, global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS) bwor 4194304);
         end
         if (local_var(4) == 27707) then begin
-            set_global_var(900, global_var(900) bwor 8388608);
+            set_global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS, global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS) bwor 8388608);
         end
         if (local_var(4) == 27312) then begin
-            set_global_var(900, global_var(900) bwor 16777216);
+            set_global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS, global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS) bwor 16777216);
         end
     end
-    set_global_var(892, global_var(892) + 1);
+    set_global_var(GVAR_AQ_PRIMITIVES_HUNTING_COUNTER, global_var(GVAR_AQ_PRIMITIVES_HUNTING_COUNTER) + 1);
 end
 
 procedure look_at_p_proc
 begin
     script_overrides;
-    display_msg(message_str(1662, 100));
+    display_mstr(100);
 end
 
 procedure description_p_proc
 begin
     script_overrides;
-    display_msg(message_str(1662, 200));
+    display_mstr(200);
 end
 
 procedure use_skill_on_p_proc
@@ -383,26 +383,26 @@ end
 procedure damage_p_proc
 begin
     if (source_obj == dude_obj) then begin
-        set_global_var(857, global_var(857) + 1);
-        if (global_var(857) > 2) then begin
-            critter_add_trait(self_obj, 1, 6, 256);
+        set_global_var(GVAR_FRIENDLY_FIRE, global_var(GVAR_FRIENDLY_FIRE) + 1);
+        if (global_var(GVAR_FRIENDLY_FIRE) > 2) then begin
+            critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 256);
         end
     end
 end
 
 procedure map_enter_p_proc
 begin
-    if (global_var(857) < 3) then begin
-        critter_add_trait(self_obj, 1, 6, 0);
+    if (global_var(GVAR_FRIENDLY_FIRE) < 3) then begin
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 0);
     end
     else begin
-        critter_add_trait(self_obj, 1, 6, 256);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 256);
     end
-    critter_add_trait(self_obj, 1, 5, 1);
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_BERSERKER_NO_TALKING);
     if (local_var(4) == 0) then begin
         set_local_var(4, tile_num(self_obj));
     end
-    if (cur_map_index == 92) then begin
+    if (CUR_MAP_ALB_DIV1) then begin
         if (local_var(4) == 17074) then begin
             lovec1_obj := self_obj;
         end
@@ -413,7 +413,7 @@ begin
             lovec3_obj := self_obj;
         end
     end
-    if (cur_map_index == 93) then begin
+    if (CUR_MAP_ALB_DIV2) then begin
         if (local_var(4) == 20727) then begin
             lovec1_obj := self_obj;
         end
@@ -424,7 +424,7 @@ begin
             lovec3_obj := self_obj;
         end
     end
-    if (cur_map_index == 94) then begin
+    if (CUR_MAP_ALB_DIV3) then begin
         if (local_var(4) == 26504) then begin
             lovec1_obj := self_obj;
         end
@@ -439,24 +439,24 @@ end
 
 procedure combat_p_proc
 begin
-    if ((global_var(862) == 1) or (global_var(857) > 2) and ((local_var(5) bwand 1) == 0)) then begin
-        critter_add_trait(self_obj, 1, 6, 256);
+    if ((global_var(GVAR_AQ_ENEMY_PRIMITIVES) == 1) or (global_var(GVAR_FRIENDLY_FIRE) > 2) and ((local_var(5) bwand 1) == 0)) then begin
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 256);
         set_local_var(5, local_var(5) bwor 1);
     end
     if (fixed_param == 4) then begin
-        if (cur_map_index == 95) then begin
+        if (CUR_MAP_ALB_NEM4) then begin
             probability := random(0, 20);
             if ((probability == 16) and not(critter_is_fleeing(self_obj))) then begin
-                float_msg(self_obj, message_str(1662, 400), 2);
+                float_msg(self_obj, mstr(400), FLOAT_MSG_RED);
             end
             if ((probability == 17) and not(critter_is_fleeing(self_obj))) then begin
-                float_msg(self_obj, message_str(1662, 402), 2);
+                float_msg(self_obj, mstr(402), FLOAT_MSG_RED);
             end
         end
         else begin
             probability := random(0, 15);
             if ((probability >= 15) and not(critter_is_fleeing(self_obj))) then begin
-                float_msg(self_obj, message_str(1662, random(401, 402)), 2);
+                float_msg(self_obj, mstr(random(401, 402)), FLOAT_MSG_RED);
             end
         end
     end
@@ -464,13 +464,12 @@ end
 
 procedure map_exit_p_proc
 begin
-    set_global_var(866, global_var(866) bwand (-1 - 256));
+    set_global_var(GVAR_AQ_ALBUQ_SPEECHES, global_var(GVAR_AQ_ALBUQ_SPEECHES) bwand (-1 - 256));
 end
 
 procedure push_p_proc
 begin
-    if (has_trait(1, self_obj, 6) != 0) then begin
+    if (has_trait(TRAIT_OBJECT, self_obj, OBJECT_TEAM_NUM) != 0) then begin
         script_overrides;
     end
 end
-
