diff --git a/Mods/Resurrection/Original/ADIVLOV.ssl b/Mods/Resurrection/ReDefine/ADIVLOV.ssl
index a8ebabc..26390cc 100644
--- a/Mods/Resurrection/Original/ADIVLOV.ssl
+++ b/Mods/Resurrection/ReDefine/ADIVLOV.ssl
@@ -59,12 +59,12 @@ begin
             return 1;
         end
     end
-    if ((party_member_obj(16777779) != 0) and ((global_var(910) bwand 32) == 0)) then begin
+    if ((party_member_obj(16777779) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 32) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777779)) <= 5) then begin
             return 1;
         end
     end
-    if ((party_member_obj(16777811) != 0) and ((global_var(910) bwand 64) == 0)) then begin
+    if ((party_member_obj(16777811) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 64) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777811)) <= 5) then begin
             return 1;
         end
@@ -79,7 +79,7 @@ begin
             return 1;
         end
     end
-    if ((party_member_obj(16777814) != 0) and ((global_var(910) bwand 128) == 0)) then begin
+    if ((party_member_obj(16777814) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 128) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777814)) <= 5) then begin
             return 1;
         end
@@ -136,15 +136,15 @@ end
 
 procedure critter_p_proc
 begin
-    if ((local_var(4) == 1) or (global_var(862) == 1) and (obj_can_see_obj(self_obj, dude_obj) or (tile_distance(tile_num(self_obj), tile_num(dude_obj)) < 15)) and (local_var(7) == 0)) then begin
+    if ((local_var(4) == 1) or (global_var(GVAR_AQ_ENEMY_PRIMITIVES) == 1) and (obj_can_see_obj(self_obj, dude_obj) or (tile_distance(tile_num(self_obj), dude_tile) < 15)) and (local_var(7) == 0)) then begin
         if (critter_is_fleeing(self_obj)) then begin
             if (anim_busy(self_obj) == 0) then begin
                 if ((tile_distance_objs(self_obj, dude_obj) < 8) or obj_can_see_obj(self_obj, dude_obj)) then begin
                     if (anim_busy(self_obj) == 0) then begin
-                        global_temp := rotation_to_tile(tile_num(dude_obj), tile_num(self_obj));
+                        global_temp := rotation_to_tile(dude_tile, tile_num(self_obj));
                         animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(self_obj), global_temp, random(3, 10)), 1);
                         global_temp := (global_temp + 1) % 6;
-                        while ((anim_busy(self_obj) == 0) and (global_temp != rotation_to_tile(tile_num(dude_obj), tile_num(self_obj)))) do begin
+                        while ((anim_busy(self_obj) == 0) and (global_temp != rotation_to_tile(dude_tile, tile_num(self_obj)))) do begin
                             debug_msg("flee loop: rot == " + global_temp);
                             animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(self_obj), global_temp, random(3, 10)), 1);
                             global_temp := (global_temp + 1) % 6;
@@ -154,13 +154,13 @@ begin
             end
         end
         else begin
-            attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+            attack(dude_obj);
         end
     end
     if ((map_var(0) bwand 512) != 0) then begin
         destroy_object(self_obj);
     end
-    if ((global_var(865) bwand 32768) != 0) then begin
+    if ((global_var(GVAR_AQ_ALBUQ_EVENTS) bwand 32768) != 0) then begin
         destroy_object(self_obj);
     end
     if (tile_in_tile_rect(25324, 25519, 25524, 26719, tile_num(self_obj)) or tile_in_tile_rect(21342, 21337, 22742, 22737, tile_num(self_obj)) or tile_in_tile_rect(18340, 18334, 19740, 19734, tile_num(self_obj)) or tile_in_tile_rect(24132, 24127, 25532, 25527, tile_num(self_obj)) and (local_var(7) == 0)) then begin
@@ -176,8 +176,8 @@ begin
         end
     end
     if (local_var(7) == 1) then begin
-        critter_add_trait(self_obj, 1, 6, 251);
-        critter_add_trait(self_obj, 1, 5, 180);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 251);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_PEASANT_NO_MOVE);
         set_local_var(7, 2);
         if ((art_anim(obj_art_fid(self_obj)) == 20) or (art_anim(obj_art_fid(self_obj)) == 21) or (art_anim(obj_art_fid(self_obj)) == 48) or (art_anim(obj_art_fid(self_obj)) == 49)) then begin
         end
@@ -207,89 +207,89 @@ end
 procedure talk_p_proc
 begin
     if (local_var(7) == 0) then begin
-        if (((global_var(865) bwand 4096) != 0) and (Projed_smycku == 0)) then begin
-            if (get_critter_stat(dude_obj, 34) == 0) then begin
-                float_msg(self_obj, message_str(1665, random(320, 323)), 2);
+        if (((global_var(GVAR_AQ_ALBUQ_EVENTS) bwand 4096) != 0) and (Projed_smycku == 0)) then begin
+            if (dude_gender == 0) then begin
+                float_msg(self_obj, mstr(random(320, 323)), FLOAT_MSG_RED);
             end
-            if (get_critter_stat(dude_obj, 34) == 1) then begin
-                float_msg(self_obj, message_str(1665, random(323, 326)), 2);
+            if (dude_gender == 1) then begin
+                float_msg(self_obj, mstr(random(323, 326)), FLOAT_MSG_RED);
             end
             Projed_smycku := 1;
         end
-        if ((party_member_obj(16777812) != 0) and (global_var(886) == 0) and (Projed_smycku == 0)) then begin
-            if (get_critter_stat(dude_obj, 34) == 0) then begin
-                float_msg(self_obj, message_str(1665, random(314, 315)), 8);
+        if ((party_member_obj(16777812) != 0) and (global_var(GVAR_AQ_FRIEND_OF_THE_TRIBE) == 0) and (Projed_smycku == 0)) then begin
+            if (dude_gender == 0) then begin
+                float_msg(self_obj, mstr(random(314, 315)), FLOAT_MSG_YELLOW);
             end
-            if (get_critter_stat(dude_obj, 34) == 1) then begin
-                float_msg(self_obj, message_str(1665, random(316, 317)), 8);
+            if (dude_gender == 1) then begin
+                float_msg(self_obj, mstr(random(316, 317)), FLOAT_MSG_YELLOW);
             end
             Projed_smycku := 1;
         end
-        if ((global_var(886) == 1) and (Projed_smycku == 0)) then begin
+        if ((global_var(GVAR_AQ_FRIEND_OF_THE_TRIBE) == 1) and (Projed_smycku == 0)) then begin
             probability := random(1, 9);
-            if ((global_var(867) bwand 512) != 0) then begin
+            if ((global_var(GVAR_AQ_ALBUQ_DEATHS) bwand 512) != 0) then begin
                 if (probability == 1) then begin
-                    float_msg(self_obj, message_str(1665, 313), 8);
+                    float_msg(self_obj, mstr(313), FLOAT_MSG_YELLOW);
                 end
                 if (probability == 2) then begin
-                    float_msg(self_obj, message_str(1665, 303), 8);
+                    float_msg(self_obj, mstr(303), FLOAT_MSG_YELLOW);
                 end
                 if (probability == 3) then begin
-                    float_msg(self_obj, message_str(1665, 304), 8);
+                    float_msg(self_obj, mstr(304), FLOAT_MSG_YELLOW);
                 end
                 if (probability == 4) then begin
-                    float_msg(self_obj, message_str(1665, 305), 8);
+                    float_msg(self_obj, mstr(305), FLOAT_MSG_YELLOW);
                 end
                 if (probability == 5) then begin
-                    float_msg(self_obj, message_str(1665, 306), 8);
+                    float_msg(self_obj, mstr(306), FLOAT_MSG_YELLOW);
                 end
-                if ((get_critter_stat(dude_obj, 34) == 0) and (probability >= 6)) then begin
-                    float_msg(self_obj, message_str(1665, random(309, 310)), 8);
+                if ((dude_gender == 0) and (probability >= 6)) then begin
+                    float_msg(self_obj, mstr(random(309, 310)), FLOAT_MSG_YELLOW);
                 end
-                if ((get_critter_stat(dude_obj, 34) == 1) and (probability >= 6)) then begin
-                    float_msg(self_obj, message_str(1665, random(311, 312)), 8);
+                if ((dude_gender == 1) and (probability >= 6)) then begin
+                    float_msg(self_obj, mstr(random(311, 312)), FLOAT_MSG_YELLOW);
                 end
             end
             else begin
                 if (probability == 1) then begin
-                    float_msg(self_obj, message_str(1665, 301), 8);
+                    float_msg(self_obj, mstr(301), FLOAT_MSG_YELLOW);
                 end
                 if (probability == 2) then begin
-                    float_msg(self_obj, message_str(1665, 303), 8);
+                    float_msg(self_obj, mstr(303), FLOAT_MSG_YELLOW);
                 end
                 if (probability == 3) then begin
-                    float_msg(self_obj, message_str(1665, 304), 8);
+                    float_msg(self_obj, mstr(304), FLOAT_MSG_YELLOW);
                 end
                 if (probability == 4) then begin
-                    float_msg(self_obj, message_str(1665, 305), 8);
+                    float_msg(self_obj, mstr(305), FLOAT_MSG_YELLOW);
                 end
                 if (probability == 5) then begin
-                    float_msg(self_obj, message_str(1665, 306), 8);
+                    float_msg(self_obj, mstr(306), FLOAT_MSG_YELLOW);
                 end
-                if ((get_critter_stat(dude_obj, 34) == 0) and (probability >= 6)) then begin
-                    float_msg(self_obj, message_str(1665, random(309, 310)), 8);
+                if ((dude_gender == 0) and (probability >= 6)) then begin
+                    float_msg(self_obj, mstr(random(309, 310)), FLOAT_MSG_YELLOW);
                 end
-                if ((get_critter_stat(dude_obj, 34) == 1) and (probability >= 6)) then begin
-                    float_msg(self_obj, message_str(1665, random(311, 312)), 8);
+                if ((dude_gender == 1) and (probability >= 6)) then begin
+                    float_msg(self_obj, mstr(random(311, 312)), FLOAT_MSG_YELLOW);
                 end
             end
             Projed_smycku := 1;
         end
-        if ((global_var(886) == 0) and (Projed_smycku == 0)) then begin
-            if (global_var(891) < 7) then begin
+        if ((global_var(GVAR_AQ_FRIEND_OF_THE_TRIBE) == 0) and (Projed_smycku == 0)) then begin
+            if (global_var(GVAR_AQ_DOCTORS_COUNTER) < 7) then begin
                 probability := random(1, 8);
             end
             else begin
                 probability := random(1, 7);
             end
-            if ((global_var(867) bwand 512) != 0) then begin
+            if ((global_var(GVAR_AQ_ALBUQ_DEATHS) bwand 512) != 0) then begin
                 if (probability == 7) then begin
-                    if (global_var(888) >= 4) then begin
-                        if (get_critter_stat(dude_obj, 34) == 0) then begin
-                            float_msg(self_obj, message_str(1665, 307), 8);
+                    if (global_var(GVAR_AQ_QUEST_TRIBE_HUNTINGS) >= 4) then begin
+                        if (dude_gender == 0) then begin
+                            float_msg(self_obj, mstr(307), FLOAT_MSG_YELLOW);
                         end
-                        if (get_critter_stat(dude_obj, 34) == 1) then begin
-                            float_msg(self_obj, message_str(1665, 308), 8);
+                        if (dude_gender == 1) then begin
+                            float_msg(self_obj, mstr(308), FLOAT_MSG_YELLOW);
                         end
                     end
                     else begin
@@ -297,35 +297,35 @@ begin
                     end
                 end
                 if (probability == 1) then begin
-                    float_msg(self_obj, message_str(1665, 313), 8);
+                    float_msg(self_obj, mstr(313), FLOAT_MSG_YELLOW);
                 end
                 if (probability == 2) then begin
-                    float_msg(self_obj, message_str(1665, 303), 8);
+                    float_msg(self_obj, mstr(303), FLOAT_MSG_YELLOW);
                 end
                 if (probability == 3) then begin
-                    float_msg(self_obj, message_str(1665, 304), 8);
+                    float_msg(self_obj, mstr(304), FLOAT_MSG_YELLOW);
                 end
                 if (probability == 4) then begin
-                    float_msg(self_obj, message_str(1665, 305), 8);
+                    float_msg(self_obj, mstr(305), FLOAT_MSG_YELLOW);
                 end
                 if (probability == 5) then begin
-                    float_msg(self_obj, message_str(1665, 306), 8);
+                    float_msg(self_obj, mstr(306), FLOAT_MSG_YELLOW);
                 end
                 if (probability == 6) then begin
-                    float_msg(self_obj, message_str(1665, 302), 8);
+                    float_msg(self_obj, mstr(302), FLOAT_MSG_YELLOW);
                 end
                 if (probability == 8) then begin
-                    float_msg(self_obj, message_str(1665, 300), 8);
+                    float_msg(self_obj, mstr(300), FLOAT_MSG_YELLOW);
                 end
             end
             else begin
                 if (probability == 7) then begin
-                    if (global_var(888) >= 4) then begin
-                        if (get_critter_stat(dude_obj, 34) == 0) then begin
-                            float_msg(self_obj, message_str(1665, 307), 8);
+                    if (global_var(GVAR_AQ_QUEST_TRIBE_HUNTINGS) >= 4) then begin
+                        if (dude_gender == 0) then begin
+                            float_msg(self_obj, mstr(307), FLOAT_MSG_YELLOW);
                         end
-                        if (get_critter_stat(dude_obj, 34) == 1) then begin
-                            float_msg(self_obj, message_str(1665, 308), 8);
+                        if (dude_gender == 1) then begin
+                            float_msg(self_obj, mstr(308), FLOAT_MSG_YELLOW);
                         end
                     end
                     else begin
@@ -333,25 +333,25 @@ begin
                     end
                 end
                 if (probability == 1) then begin
-                    float_msg(self_obj, message_str(1665, 301), 8);
+                    float_msg(self_obj, mstr(301), FLOAT_MSG_YELLOW);
                 end
                 if (probability == 2) then begin
-                    float_msg(self_obj, message_str(1665, 303), 8);
+                    float_msg(self_obj, mstr(303), FLOAT_MSG_YELLOW);
                 end
                 if (probability == 3) then begin
-                    float_msg(self_obj, message_str(1665, 304), 8);
+                    float_msg(self_obj, mstr(304), FLOAT_MSG_YELLOW);
                 end
                 if (probability == 4) then begin
-                    float_msg(self_obj, message_str(1665, 305), 8);
+                    float_msg(self_obj, mstr(305), FLOAT_MSG_YELLOW);
                 end
                 if (probability == 5) then begin
-                    float_msg(self_obj, message_str(1665, 306), 8);
+                    float_msg(self_obj, mstr(306), FLOAT_MSG_YELLOW);
                 end
                 if (probability == 6) then begin
-                    float_msg(self_obj, message_str(1665, 302), 8);
+                    float_msg(self_obj, mstr(302), FLOAT_MSG_YELLOW);
                 end
                 if (probability == 8) then begin
-                    float_msg(self_obj, message_str(1665, 300), 8);
+                    float_msg(self_obj, mstr(300), FLOAT_MSG_YELLOW);
                 end
             end
             Projed_smycku := 1;
@@ -363,59 +363,59 @@ end
 procedure destroy_p_proc
 begin
     if (source_obj == dude_obj) then begin
-        set_global_var(5, global_var(5) + 1);
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(4) > (3 * global_var(5))) or (global_var(2) == 1)) and (global_var(1) == 0)) then begin
-            set_global_var(2, 1);
-            set_global_var(3, 0);
+        set_global_var(GVAR_GOOD_MONSTER, global_var(GVAR_GOOD_MONSTER) + 1);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(4) > (3 * global_var(GVAR_GOOD_MONSTER))) or (global_var(GVAR_CHAMPION_REPUTATION) == 1)) and (global_var(GVAR_CHILDKILLER_REPUTATION) == 0)) then begin
+            set_global_var(GVAR_CHAMPION_REPUTATION, 1);
+            set_global_var(GVAR_BERSERKER_REPUTATION, 0);
         end
-        if (metarule(51, self_obj) == 2) then begin
+        if (critter_kill_type(self_obj) == KILL_TYPE_children_kills) then begin
             if (source_obj == dude_obj) then begin
-                set_global_var(2, 0);
-                set_global_var(1, global_var(1) + 1);
-                set_global_var(0, global_var(0) + -15);
+                set_global_var(GVAR_CHAMPION_REPUTATION, 0);
+                set_global_var(GVAR_CHILDKILLER_REPUTATION, global_var(GVAR_CHILDKILLER_REPUTATION) + 1);
+                set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + -15);
                 debug_msg("Player gains " + -15 + " Karma Points.");
-                set_global_var(37, 0);
-                set_global_var(38, 0);
-                set_global_var(39, 0);
-                set_global_var(40, 0);
-                set_global_var(41, 0);
-                set_global_var(42, 0);
-                set_global_var(43, 0);
-                set_global_var(44, 0);
-                set_global_var(45, 0);
-                if (global_var(0) >= 1000) then begin
-                    set_global_var(37, 1);
+                set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+                set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+                set_global_var(GVAR_KARMA_DEFENDER, 0);
+                set_global_var(GVAR_KARMA_WANDERER, 0);
+                set_global_var(GVAR_KARMA_BETRAYER, 0);
+                set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+                set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+                set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+                    set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
                 end
                 else begin
-                    if (global_var(0) >= 750) then begin
-                        set_global_var(38, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
                     end
                     else begin
-                        if (global_var(0) >= 500) then begin
-                            set_global_var(39, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                            set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                         end
                         else begin
-                            if (global_var(0) >= 250) then begin
-                                set_global_var(40, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                                set_global_var(GVAR_KARMA_DEFENDER, 1);
                             end
                             else begin
-                                if (global_var(0) > -250) then begin
-                                    set_global_var(41, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                                    set_global_var(GVAR_KARMA_WANDERER, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -500) then begin
-                                        set_global_var(42, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                        set_global_var(GVAR_KARMA_BETRAYER, 1);
                                     end
                                     else begin
-                                        if (global_var(0) > -750) then begin
-                                            set_global_var(43, 1);
+                                        if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                            set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                         end
                                         else begin
-                                            if (global_var(0) > -1000) then begin
-                                                set_global_var(44, 1);
+                                            if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                                set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                             end
                                             else begin
-                                                set_global_var(45, 1);
+                                                set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                             end
                                         end
                                     end
@@ -424,58 +424,58 @@ begin
                         end
                     end
                 end
-                set_global_var(51, global_var(51) + -8);
+                set_global_var(GVAR_TOWN_REP_ALBUQUERQUE, global_var(GVAR_TOWN_REP_ALBUQUERQUE) + -8);
                 debug_msg("Added " + -8 + " to Town Rep");
             end
         end
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(5) > (2 * global_var(4))) or (global_var(3) == 1))) then begin
-            set_global_var(3, 1);
-            set_global_var(2, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
+            set_global_var(GVAR_BERSERKER_REPUTATION, 1);
+            set_global_var(GVAR_CHAMPION_REPUTATION, 0);
         end
-        set_global_var(0, global_var(0) + -10);
+        set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + -10);
         debug_msg("Player gains " + -10 + " Karma Points.");
-        set_global_var(37, 0);
-        set_global_var(38, 0);
-        set_global_var(39, 0);
-        set_global_var(40, 0);
-        set_global_var(41, 0);
-        set_global_var(42, 0);
-        set_global_var(43, 0);
-        set_global_var(44, 0);
-        set_global_var(45, 0);
-        if (global_var(0) >= 1000) then begin
-            set_global_var(37, 1);
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+        set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+        set_global_var(GVAR_KARMA_DEFENDER, 0);
+        set_global_var(GVAR_KARMA_WANDERER, 0);
+        set_global_var(GVAR_KARMA_BETRAYER, 0);
+        set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+        set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+            set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
         end
         else begin
-            if (global_var(0) >= 750) then begin
-                set_global_var(38, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
             end
             else begin
-                if (global_var(0) >= 500) then begin
-                    set_global_var(39, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                 end
                 else begin
-                    if (global_var(0) >= 250) then begin
-                        set_global_var(40, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                        set_global_var(GVAR_KARMA_DEFENDER, 1);
                     end
                     else begin
-                        if (global_var(0) > -250) then begin
-                            set_global_var(41, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                            set_global_var(GVAR_KARMA_WANDERER, 1);
                         end
                         else begin
-                            if (global_var(0) > -500) then begin
-                                set_global_var(42, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                set_global_var(GVAR_KARMA_BETRAYER, 1);
                             end
                             else begin
-                                if (global_var(0) > -750) then begin
-                                    set_global_var(43, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -1000) then begin
-                                        set_global_var(44, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                     end
                                     else begin
-                                        set_global_var(45, 1);
+                                        set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                     end
                                 end
                             end
@@ -484,37 +484,37 @@ begin
                 end
             end
         end
-        set_global_var(51, global_var(51) + -5);
+        set_global_var(GVAR_TOWN_REP_ALBUQUERQUE, global_var(GVAR_TOWN_REP_ALBUQUERQUE) + -5);
         debug_msg("Added " + -5 + " to Town Rep");
     end
-    if (global_var(878) == 2) then begin
-        set_global_var(890, global_var(890) + 1);
+    if (global_var(GVAR_AQ_QUEST_HOSPITAL_TO_KILL_PRIMITIVES) == 2) then begin
+        set_global_var(GVAR_AQ_PRIMITIVES_COUNTER, global_var(GVAR_AQ_PRIMITIVES_COUNTER) + 1);
     end
-    if ((global_var(878) == 2) and (local_var(7) == 0)) then begin
-        set_global_var(899, global_var(899) + 1);
+    if ((global_var(GVAR_AQ_QUEST_HOSPITAL_TO_KILL_PRIMITIVES) == 2) and (local_var(7) == 0)) then begin
+        set_global_var(GVAR_AQ_HOSPITAL_GIVERS_STATUS, global_var(GVAR_AQ_HOSPITAL_GIVERS_STATUS) + 1);
     end
-    if ((global_var(878) == 2) and (global_var(889) <= global_var(899))) then begin
-        set_global_var(878, 3);
+    if ((global_var(GVAR_AQ_QUEST_HOSPITAL_TO_KILL_PRIMITIVES) == 2) and (global_var(GVAR_AQ_HUNTINGS_RESULTS) <= global_var(GVAR_AQ_HOSPITAL_GIVERS_STATUS))) then begin
+        set_global_var(GVAR_AQ_QUEST_HOSPITAL_TO_KILL_PRIMITIVES, 3);
     end
-    if ((global_var(878) == 2) and (global_var(862) == 0)) then begin
-        set_global_var(862, 1);
+    if ((global_var(GVAR_AQ_QUEST_HOSPITAL_TO_KILL_PRIMITIVES) == 2) and (global_var(GVAR_AQ_ENEMY_PRIMITIVES) == 0)) then begin
+        set_global_var(GVAR_AQ_ENEMY_PRIMITIVES, 1);
     end
 end
 
 procedure look_at_p_proc
 begin
     script_overrides;
-    display_msg(message_str(1665, 100));
+    display_mstr(100);
 end
 
 procedure description_p_proc
 begin
     script_overrides;
     if (local_var(7) == 0) then begin
-        display_msg(message_str(1665, 200));
+        display_mstr(200);
     end
     else begin
-        display_msg(message_str(1665, 200) + message_str(1665, 201));
+        display_msg(mstr(200) + mstr(201));
     end
 end
 
@@ -525,24 +525,24 @@ end
 procedure damage_p_proc
 begin
     if (source_obj == dude_obj) then begin
-        set_global_var(862, 1);
+        set_global_var(GVAR_AQ_ENEMY_PRIMITIVES, 1);
     end
-    if (global_var(878) == 2) then begin
-        if ((obj_pid(critter_inven_obj(source_obj, 1)) == 601) or (obj_pid(critter_inven_obj(source_obj, 2)) == 601) and (local_var(7) != 2)) then begin
-            if (get_critter_stat(self_obj, 35) <= (get_critter_stat(self_obj, 7) / 3)) then begin
+    if (global_var(GVAR_AQ_QUEST_HOSPITAL_TO_KILL_PRIMITIVES) == 2) then begin
+        if ((obj_pid(critter_inven_obj(source_obj, INVEN_TYPE_RIGHT_HAND)) == 601) or (obj_pid(critter_inven_obj(source_obj, INVEN_TYPE_LEFT_HAND)) == 601) and (local_var(7) != 2)) then begin
+            if (get_critter_stat(self_obj, STAT_current_hp) <= (get_critter_stat(self_obj, STAT_max_hp) / 3)) then begin
                 if (local_var(7) == 0) then begin
-                    set_global_var(899, global_var(899) + 1);
+                    set_global_var(GVAR_AQ_HOSPITAL_GIVERS_STATUS, global_var(GVAR_AQ_HOSPITAL_GIVERS_STATUS) + 1);
                 end
-                if ((global_var(878) == 2) and (global_var(889) <= global_var(899)) and (local_var(7) == 0)) then begin
-                    set_global_var(878, 3);
+                if ((global_var(GVAR_AQ_QUEST_HOSPITAL_TO_KILL_PRIMITIVES) == 2) and (global_var(GVAR_AQ_HUNTINGS_RESULTS) <= global_var(GVAR_AQ_HOSPITAL_GIVERS_STATUS)) and (local_var(7) == 0)) then begin
+                    set_global_var(GVAR_AQ_QUEST_HOSPITAL_TO_KILL_PRIMITIVES, 3);
                 end
                 set_local_var(7, 1);
                 script_overrides;
                 critter_injure(self_obj, 2);
                 critter_injure(self_obj, 8);
                 critter_injure(self_obj, 4);
-                critter_add_trait(self_obj, 1, 6, 251);
-                critter_add_trait(self_obj, 1, 5, 180);
+                critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 251);
+                critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_PEASANT_NO_MOVE);
                 if (local_var(5) == 24122) then begin
                     set_map_var(4, map_var(4) bwor 1);
                 end
@@ -564,19 +564,19 @@ begin
                 if (local_var(5) == 22332) then begin
                     set_map_var(4, map_var(4) bwor 64);
                 end
-                display_msg(obj_name(self_obj) + message_str(1665, 410));
-                set_global_var(51, global_var(51) - 2);
-                if (get_critter_stat(self_obj, 35) > 0) then begin
-                    set_global_var(862, 0);
+                display_msg(self_name + mstr(410));
+                set_global_var(GVAR_TOWN_REP_ALBUQUERQUE, global_var(GVAR_TOWN_REP_ALBUQUERQUE) - 2);
+                if (get_critter_stat(self_obj, STAT_current_hp) > 0) then begin
+                    set_global_var(GVAR_AQ_ENEMY_PRIMITIVES, 0);
                 end
-                if (get_critter_stat(self_obj, 35) > 0) then begin
+                if (get_critter_stat(self_obj, STAT_current_hp) > 0) then begin
                     terminate_combat;
                 end
             end
         end
         else begin
             if (source_obj == dude_obj) then begin
-                set_global_var(900, global_var(900) bwor 4);
+                set_global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS, global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS) bwor 4);
             end
         end
     end
@@ -585,26 +585,26 @@ end
 procedure map_enter_p_proc
 begin
     if (local_var(7) > 0) then begin
-        critter_add_trait(self_obj, 1, 6, 251);
-        critter_add_trait(self_obj, 1, 5, 180);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 251);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_PEASANT_NO_MOVE);
     end
     else begin
-        critter_add_trait(self_obj, 1, 6, 256);
-        critter_add_trait(self_obj, 1, 5, 1);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 256);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_BERSERKER_NO_TALKING);
     end
-    if ((global_var(878) == 1) and not(metarule(22, 0))) then begin
-        set_global_var(889, global_var(889) + 1);
+    if ((global_var(GVAR_AQ_QUEST_HOSPITAL_TO_KILL_PRIMITIVES) == 1) and not(is_loading_game)) then begin
+        set_global_var(GVAR_AQ_HUNTINGS_RESULTS, global_var(GVAR_AQ_HUNTINGS_RESULTS) + 1);
     end
-    if ((global_var(878) == 1) and not(metarule(22, 0))) then begin
+    if ((global_var(GVAR_AQ_QUEST_HOSPITAL_TO_KILL_PRIMITIVES) == 1) and not(is_loading_game)) then begin
         set_local_var(4, 0);
     end
-    if ((global_var(865) bwand 32768) != 0) then begin
+    if ((global_var(GVAR_AQ_ALBUQ_EVENTS) bwand 32768) != 0) then begin
         destroy_object(self_obj);
     end
-    if ((global_var(865) bwand 8388608) != 0) then begin
+    if ((global_var(GVAR_AQ_ALBUQ_EVENTS) bwand 8388608) != 0) then begin
         destroy_object(self_obj);
     end
-    if (not(metarule(22, 0))) then begin
+    if (not(is_loading_game)) then begin
         if (local_var(5) == 0) then begin
             set_local_var(5, tile_num(self_obj));
         end
@@ -620,31 +620,31 @@ begin
         if (local_var(5) == 18945) then begin
             set_local_var(6, 1);
         end
-        if ((tile_num(self_obj) != local_var(5)) and not(metarule(14, 0))) then begin
-            if (global_var(862) != 1) then begin
+        if ((tile_num(self_obj) != local_var(5)) and not(map_first_run)) then begin
+            if (global_var(GVAR_AQ_ENEMY_PRIMITIVES) != 1) then begin
                 move_to(self_obj, local_var(5), elevation(self_obj));
             end
         end
     end
-    if ((global_var(893) > 0) and (local_var(5) == 24122)) then begin
+    if ((global_var(GVAR_AQ_PRIMATACKERS_COUNTER) > 0) and (local_var(5) == 24122)) then begin
         destroy_object(self_obj);
     end
-    if ((global_var(893) > 1) and (local_var(5) == 22528)) then begin
+    if ((global_var(GVAR_AQ_PRIMATACKERS_COUNTER) > 1) and (local_var(5) == 22528)) then begin
         destroy_object(self_obj);
     end
-    if ((global_var(893) > 2) and (local_var(5) == 26144)) then begin
+    if ((global_var(GVAR_AQ_PRIMATACKERS_COUNTER) > 2) and (local_var(5) == 26144)) then begin
         destroy_object(self_obj);
     end
-    if ((global_var(893) > 3) and (local_var(5) == 26933)) then begin
+    if ((global_var(GVAR_AQ_PRIMATACKERS_COUNTER) > 3) and (local_var(5) == 26933)) then begin
         destroy_object(self_obj);
     end
-    if ((global_var(893) > 4) and (local_var(5) == 20536)) then begin
+    if ((global_var(GVAR_AQ_PRIMATACKERS_COUNTER) > 4) and (local_var(5) == 20536)) then begin
         destroy_object(self_obj);
     end
-    if ((global_var(893) > 5) and (local_var(5) == 18945)) then begin
+    if ((global_var(GVAR_AQ_PRIMATACKERS_COUNTER) > 5) and (local_var(5) == 18945)) then begin
         destroy_object(self_obj);
     end
-    if ((global_var(893) > 6) and (local_var(5) == 22332)) then begin
+    if ((global_var(GVAR_AQ_PRIMATACKERS_COUNTER) > 6) and (local_var(5) == 22332)) then begin
         destroy_object(self_obj);
     end
     if (local_var(5) == 24122) then begin
@@ -668,8 +668,8 @@ begin
     if (local_var(5) == 22332) then begin
         primi7_obj := self_obj;
     end
-    if ((local_var(6) == 1) and (has_trait(1, self_obj, 666) == 1) and (local_var(7) == 0)) then begin
-        metarule3(100, self_obj, 1, 0);
+    if ((local_var(6) == 1) and (obj_is_visible_flag(self_obj) == 1) and (local_var(7) == 0)) then begin
+        rm_fixed_timer_event(self_obj, 1);
     end
     add_timer_event(self_obj, game_ticks(random(15, 30)), 1);
 end
@@ -688,13 +688,13 @@ begin
             reg_anim_func(3, 0);
         end
         if (local_var(6) == 1) then begin
-            metarule3(100, self_obj, 1, 0);
+            rm_fixed_timer_event(self_obj, 1);
         end
         add_timer_event(self_obj, game_ticks(random(15, 30)), 1);
     end
     if (fixed_param == 2) then begin
         game_ui_enable;
-        set_global_var(862, 1);
+        set_global_var(GVAR_AQ_ENEMY_PRIMITIVES, 1);
     end
 end
 
@@ -703,17 +703,17 @@ begin
     if ((fixed_param == 4) and (local_var(7) == 0)) then begin
         probability := random(0, 12);
         if ((probability >= 12) and not(critter_is_fleeing(self_obj))) then begin
-            if ((local_var(4) == 1) and obj_can_see_obj(self_obj, dude_obj) and (get_critter_stat(dude_obj, 34) == 0)) then begin
-                float_msg(self_obj, message_str(1665, random(400, 403)), 2);
+            if ((local_var(4) == 1) and obj_can_see_obj(self_obj, dude_obj) and (dude_gender == 0)) then begin
+                float_msg(self_obj, mstr(random(400, 403)), FLOAT_MSG_RED);
             end
-            if ((local_var(4) == 1) and obj_can_see_obj(self_obj, dude_obj) and (get_critter_stat(dude_obj, 34) == 1)) then begin
-                float_msg(self_obj, message_str(1665, random(401, 404)), 2);
+            if ((local_var(4) == 1) and obj_can_see_obj(self_obj, dude_obj) and (dude_gender == 1)) then begin
+                float_msg(self_obj, mstr(random(401, 404)), FLOAT_MSG_RED);
             end
         end
     end
     if ((fixed_param == 4) and (local_var(7) == 2)) then begin
-        critter_add_trait(self_obj, 1, 6, 251);
-        critter_add_trait(self_obj, 1, 5, 180);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 251);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_PEASANT_NO_MOVE);
         anim(self_obj, 48, 0);
     end
 end
@@ -724,4 +724,3 @@ begin
         script_overrides;
     end
 end
-
