diff --git a/Mods/Resurrection/Original/ADOREEN.ssl b/Mods/Resurrection/ReDefine/ADOREEN.ssl
index c67dc12..2f1d897 100644
--- a/Mods/Resurrection/Original/ADOREEN.ssl
+++ b/Mods/Resurrection/ReDefine/ADOREEN.ssl
@@ -70,12 +70,12 @@ begin
             return 1;
         end
     end
-    if ((party_member_obj(16777779) != 0) and ((global_var(910) bwand 32) == 0)) then begin
+    if ((party_member_obj(16777779) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 32) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777779)) <= 5) then begin
             return 1;
         end
     end
-    if ((party_member_obj(16777811) != 0) and ((global_var(910) bwand 64) == 0)) then begin
+    if ((party_member_obj(16777811) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 64) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777811)) <= 5) then begin
             return 1;
         end
@@ -90,7 +90,7 @@ begin
             return 1;
         end
     end
-    if ((party_member_obj(16777814) != 0) and ((global_var(910) bwand 128) == 0)) then begin
+    if ((party_member_obj(16777814) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 128) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777814)) <= 5) then begin
             return 1;
         end
@@ -147,15 +147,15 @@ end
 
 procedure critter_p_proc
 begin
-    if ((local_var(4) == 1) or (global_var(860) == 1) and obj_can_see_obj(self_obj, dude_obj)) then begin
+    if ((local_var(4) == 1) or (global_var(GVAR_AQ_ENEMY_PRISTEH) == 1) and obj_can_see_obj(self_obj, dude_obj)) then begin
         if (critter_is_fleeing(self_obj)) then begin
             if (anim_busy(self_obj) == 0) then begin
                 if ((tile_distance_objs(self_obj, dude_obj) < 8) or obj_can_see_obj(self_obj, dude_obj)) then begin
                     if (anim_busy(self_obj) == 0) then begin
-                        global_temp := rotation_to_tile(tile_num(dude_obj), tile_num(self_obj));
+                        global_temp := rotation_to_tile(dude_tile, tile_num(self_obj));
                         animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(self_obj), global_temp, random(3, 10)), 1);
                         global_temp := (global_temp + 1) % 6;
-                        while ((anim_busy(self_obj) == 0) and (global_temp != rotation_to_tile(tile_num(dude_obj), tile_num(self_obj)))) do begin
+                        while ((anim_busy(self_obj) == 0) and (global_temp != rotation_to_tile(dude_tile, tile_num(self_obj)))) do begin
                             debug_msg("flee loop: rot == " + global_temp);
                             animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(self_obj), global_temp, random(3, 10)), 1);
                             global_temp := (global_temp + 1) % 6;
@@ -165,18 +165,18 @@ begin
             end
         end
         else begin
-            attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+            attack(dude_obj);
         end
     end
-    if (((global_var(866) bwand 1048576) != 0) and obj_can_see_obj(self_obj, dude_obj)) then begin
+    if (((global_var(GVAR_AQ_ALBUQ_SPEECHES) bwand 1048576) != 0) and obj_can_see_obj(self_obj, dude_obj)) then begin
         if (critter_is_fleeing(self_obj)) then begin
             if (anim_busy(self_obj) == 0) then begin
                 if ((tile_distance_objs(self_obj, dude_obj) < 8) or obj_can_see_obj(self_obj, dude_obj)) then begin
                     if (anim_busy(self_obj) == 0) then begin
-                        global_temp := rotation_to_tile(tile_num(dude_obj), tile_num(self_obj));
+                        global_temp := rotation_to_tile(dude_tile, tile_num(self_obj));
                         animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(self_obj), global_temp, random(3, 10)), 1);
                         global_temp := (global_temp + 1) % 6;
-                        while ((anim_busy(self_obj) == 0) and (global_temp != rotation_to_tile(tile_num(dude_obj), tile_num(self_obj)))) do begin
+                        while ((anim_busy(self_obj) == 0) and (global_temp != rotation_to_tile(dude_tile, tile_num(self_obj)))) do begin
                             debug_msg("flee loop: rot == " + global_temp);
                             animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(self_obj), global_temp, random(3, 10)), 1);
                             global_temp := (global_temp + 1) % 6;
@@ -186,7 +186,7 @@ begin
             end
         end
         else begin
-            attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+            attack(dude_obj);
         end
     end
 end
@@ -195,14 +195,14 @@ procedure pickup_p_proc
 begin
     if (source_obj == dude_obj) then begin
         set_local_var(4, 1);
-        set_global_var(866, global_var(866) bwor 1048576);
+        set_global_var(GVAR_AQ_ALBUQ_SPEECHES, global_var(GVAR_AQ_ALBUQ_SPEECHES) bwor 1048576);
     end
 end
 
 procedure talk_p_proc
 begin
-    anim(dude_obj, 1000, rotation_to_tile(tile_num(dude_obj), tile_num(self_obj)));
-    anim(self_obj, 1000, rotation_to_tile(tile_num(self_obj), tile_num(dude_obj)));
+    anim(dude_obj, 1000, rotation_to_tile(dude_tile, tile_num(self_obj)));
+    anim(self_obj, 1000, rotation_to_tile(tile_num(self_obj), dude_tile));
     start_gdialog(1614, self_obj, 4, -1, -1);
     gsay_start;
     if (local_var(5) == 1) then begin
@@ -222,59 +222,59 @@ end
 procedure destroy_p_proc
 begin
     if (source_obj == dude_obj) then begin
-        set_global_var(5, global_var(5) + 1);
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(4) > (3 * global_var(5))) or (global_var(2) == 1)) and (global_var(1) == 0)) then begin
-            set_global_var(2, 1);
-            set_global_var(3, 0);
+        set_global_var(GVAR_GOOD_MONSTER, global_var(GVAR_GOOD_MONSTER) + 1);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(4) > (3 * global_var(GVAR_GOOD_MONSTER))) or (global_var(GVAR_CHAMPION_REPUTATION) == 1)) and (global_var(GVAR_CHILDKILLER_REPUTATION) == 0)) then begin
+            set_global_var(GVAR_CHAMPION_REPUTATION, 1);
+            set_global_var(GVAR_BERSERKER_REPUTATION, 0);
         end
-        if (metarule(51, self_obj) == 2) then begin
+        if (critter_kill_type(self_obj) == KILL_TYPE_children_kills) then begin
             if (source_obj == dude_obj) then begin
-                set_global_var(2, 0);
-                set_global_var(1, global_var(1) + 1);
-                set_global_var(0, global_var(0) + -15);
+                set_global_var(GVAR_CHAMPION_REPUTATION, 0);
+                set_global_var(GVAR_CHILDKILLER_REPUTATION, global_var(GVAR_CHILDKILLER_REPUTATION) + 1);
+                set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + -15);
                 debug_msg("Player gains " + -15 + " Karma Points.");
-                set_global_var(37, 0);
-                set_global_var(38, 0);
-                set_global_var(39, 0);
-                set_global_var(40, 0);
-                set_global_var(41, 0);
-                set_global_var(42, 0);
-                set_global_var(43, 0);
-                set_global_var(44, 0);
-                set_global_var(45, 0);
-                if (global_var(0) >= 1000) then begin
-                    set_global_var(37, 1);
+                set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+                set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+                set_global_var(GVAR_KARMA_DEFENDER, 0);
+                set_global_var(GVAR_KARMA_WANDERER, 0);
+                set_global_var(GVAR_KARMA_BETRAYER, 0);
+                set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+                set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+                set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+                    set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
                 end
                 else begin
-                    if (global_var(0) >= 750) then begin
-                        set_global_var(38, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
                     end
                     else begin
-                        if (global_var(0) >= 500) then begin
-                            set_global_var(39, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                            set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                         end
                         else begin
-                            if (global_var(0) >= 250) then begin
-                                set_global_var(40, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                                set_global_var(GVAR_KARMA_DEFENDER, 1);
                             end
                             else begin
-                                if (global_var(0) > -250) then begin
-                                    set_global_var(41, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                                    set_global_var(GVAR_KARMA_WANDERER, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -500) then begin
-                                        set_global_var(42, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                        set_global_var(GVAR_KARMA_BETRAYER, 1);
                                     end
                                     else begin
-                                        if (global_var(0) > -750) then begin
-                                            set_global_var(43, 1);
+                                        if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                            set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                         end
                                         else begin
-                                            if (global_var(0) > -1000) then begin
-                                                set_global_var(44, 1);
+                                            if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                                set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                             end
                                             else begin
-                                                set_global_var(45, 1);
+                                                set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                             end
                                         end
                                     end
@@ -283,58 +283,58 @@ begin
                         end
                     end
                 end
-                set_global_var(51, global_var(51) + -8);
+                set_global_var(GVAR_TOWN_REP_ALBUQUERQUE, global_var(GVAR_TOWN_REP_ALBUQUERQUE) + -8);
                 debug_msg("Added " + -8 + " to Town Rep");
             end
         end
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(5) > (2 * global_var(4))) or (global_var(3) == 1))) then begin
-            set_global_var(3, 1);
-            set_global_var(2, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
+            set_global_var(GVAR_BERSERKER_REPUTATION, 1);
+            set_global_var(GVAR_CHAMPION_REPUTATION, 0);
         end
-        set_global_var(0, global_var(0) + -10);
+        set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + -10);
         debug_msg("Player gains " + -10 + " Karma Points.");
-        set_global_var(37, 0);
-        set_global_var(38, 0);
-        set_global_var(39, 0);
-        set_global_var(40, 0);
-        set_global_var(41, 0);
-        set_global_var(42, 0);
-        set_global_var(43, 0);
-        set_global_var(44, 0);
-        set_global_var(45, 0);
-        if (global_var(0) >= 1000) then begin
-            set_global_var(37, 1);
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+        set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+        set_global_var(GVAR_KARMA_DEFENDER, 0);
+        set_global_var(GVAR_KARMA_WANDERER, 0);
+        set_global_var(GVAR_KARMA_BETRAYER, 0);
+        set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+        set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+            set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
         end
         else begin
-            if (global_var(0) >= 750) then begin
-                set_global_var(38, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
             end
             else begin
-                if (global_var(0) >= 500) then begin
-                    set_global_var(39, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                 end
                 else begin
-                    if (global_var(0) >= 250) then begin
-                        set_global_var(40, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                        set_global_var(GVAR_KARMA_DEFENDER, 1);
                     end
                     else begin
-                        if (global_var(0) > -250) then begin
-                            set_global_var(41, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                            set_global_var(GVAR_KARMA_WANDERER, 1);
                         end
                         else begin
-                            if (global_var(0) > -500) then begin
-                                set_global_var(42, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                set_global_var(GVAR_KARMA_BETRAYER, 1);
                             end
                             else begin
-                                if (global_var(0) > -750) then begin
-                                    set_global_var(43, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -1000) then begin
-                                        set_global_var(44, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                     end
                                     else begin
-                                        set_global_var(45, 1);
+                                        set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                     end
                                 end
                             end
@@ -343,19 +343,19 @@ begin
                 end
             end
         end
-        set_global_var(51, global_var(51) + -5);
+        set_global_var(GVAR_TOWN_REP_ALBUQUERQUE, global_var(GVAR_TOWN_REP_ALBUQUERQUE) + -5);
         debug_msg("Added " + -5 + " to Town Rep");
     end
-    set_global_var(867, global_var(867) bwor 8);
-    set_global_var(866, global_var(866) bwor 1048576);
-    if ((party_member_obj(16777814) != 0) and ((global_var(910) bwand 128) == 0)) then begin
+    set_global_var(GVAR_AQ_ALBUQ_DEATHS, global_var(GVAR_AQ_ALBUQ_DEATHS) bwor 8);
+    set_global_var(GVAR_AQ_ALBUQ_SPEECHES, global_var(GVAR_AQ_ALBUQ_SPEECHES) bwor 1048576);
+    if ((party_member_obj(16777814) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 128) == 0)) then begin
         add_timer_event(party_member_obj(16777814), 5, 21);
     end
-    if ((global_var(867) bwand 4) != 0) then begin
-        set_global_var(860, 1);
+    if ((global_var(GVAR_AQ_ALBUQ_DEATHS) bwand 4) != 0) then begin
+        set_global_var(GVAR_AQ_ENEMY_PRISTEH, 1);
     end
-    if (global_var(871) == 1) then begin
-        set_global_var(871, 2);
+    if (global_var(GVAR_AQ_QUEST_KILL_WILLIAM_WIFE) == 1) then begin
+        set_global_var(GVAR_AQ_QUEST_KILL_WILLIAM_WIFE, 2);
     end
 end
 
@@ -363,10 +363,10 @@ procedure look_at_p_proc
 begin
     script_overrides;
     if (local_var(5) == 1) then begin
-        display_msg(message_str(1614, 150));
+        display_mstr(150);
     end
     else begin
-        display_msg(message_str(1614, 100));
+        display_mstr(100);
     end
 end
 
@@ -374,10 +374,10 @@ procedure description_p_proc
 begin
     script_overrides;
     if (local_var(5) == 1) then begin
-        display_msg(message_str(1614, 250));
+        display_mstr(250);
     end
     else begin
-        display_msg(message_str(1614, 200));
+        display_mstr(200);
     end
 end
 
@@ -388,17 +388,17 @@ end
 procedure damage_p_proc
 begin
     if (source_obj == dude_obj) then begin
-        if ((obj_pid(critter_inven_obj(dude_obj, 1)) == 4) or (obj_pid(critter_inven_obj(dude_obj, 1)) == 45) or (obj_pid(critter_inven_obj(dude_obj, 1)) == 116) or (obj_pid(critter_inven_obj(dude_obj, 1)) == 236) or (obj_pid(critter_inven_obj(dude_obj, 2)) == 4) or (obj_pid(critter_inven_obj(dude_obj, 2)) == 45) or (obj_pid(critter_inven_obj(dude_obj, 2)) == 116) or (obj_pid(critter_inven_obj(dude_obj, 2)) == 236) and ((local_var(6) bwand 2) == 0) and has_trait(0, dude_obj, 25) and (has_trait(1, self_obj, 10) == 5) and (global_var(860) == 0) and (tile_num(self_obj) == 19093) and ((tile_num(dude_obj) == 19293) or (tile_num(dude_obj) == 19092) or (tile_num(dude_obj) == 19094)) and using_skill(dude_obj, 8) and is_success(roll_vs_skill(dude_obj, 8, -50))) then begin
-            set_global_var(867, global_var(867) bwor 8);
-            if (global_var(871) == 1) then begin
-                set_global_var(871, 2);
+        if ((obj_pid(critter_inven_obj(dude_obj, INVEN_TYPE_RIGHT_HAND)) == 4) or (obj_pid(critter_inven_obj(dude_obj, INVEN_TYPE_RIGHT_HAND)) == 45) or (obj_pid(critter_inven_obj(dude_obj, INVEN_TYPE_RIGHT_HAND)) == 116) or (obj_pid(critter_inven_obj(dude_obj, INVEN_TYPE_RIGHT_HAND)) == 236) or (obj_pid(critter_inven_obj(dude_obj, INVEN_TYPE_LEFT_HAND)) == 4) or (obj_pid(critter_inven_obj(dude_obj, INVEN_TYPE_LEFT_HAND)) == 45) or (obj_pid(critter_inven_obj(dude_obj, INVEN_TYPE_LEFT_HAND)) == 116) or (obj_pid(critter_inven_obj(dude_obj, INVEN_TYPE_LEFT_HAND)) == 236) and ((local_var(6) bwand 2) == 0) and has_trait(TRAIT_PERK, dude_obj, PERK_silent_death) and (obj_get_rot(self_obj) == 5) and (global_var(GVAR_AQ_ENEMY_PRISTEH) == 0) and (tile_num(self_obj) == 19093) and ((dude_tile == 19293) or (dude_tile == 19092) or (dude_tile == 19094)) and using_skill(dude_obj, 8) and is_success(roll_vs_skill(dude_obj, 8, -50))) then begin
+            set_global_var(GVAR_AQ_ALBUQ_DEATHS, global_var(GVAR_AQ_ALBUQ_DEATHS) bwor 8);
+            if (global_var(GVAR_AQ_QUEST_KILL_WILLIAM_WIFE) == 1) then begin
+                set_global_var(GVAR_AQ_QUEST_KILL_WILLIAM_WIFE, 2);
             end
-            critter_heal(self_obj, -get_critter_stat(self_obj, 7));
+            critter_heal(self_obj, -get_critter_stat(self_obj, STAT_max_hp));
             terminate_combat;
         end
         else begin
             set_local_var(4, 1);
-            set_global_var(866, global_var(866) bwor 1048576);
+            set_global_var(GVAR_AQ_ALBUQ_SPEECHES, global_var(GVAR_AQ_ALBUQ_SPEECHES) bwor 1048576);
         end
         if ((local_var(6) bwand 2) == 0) then begin
             set_local_var(6, local_var(6) bwor 2);
@@ -408,9 +408,9 @@ end
 
 procedure map_enter_p_proc
 begin
-    critter_add_trait(self_obj, 1, 6, 267);
-    critter_add_trait(self_obj, 1, 5, 14);
-    if ((global_var(867) bwand 2) != 0) then begin
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 267);
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_PEASANTS);
+    if ((global_var(GVAR_AQ_ALBUQ_DEATHS) bwand 2) != 0) then begin
         destroy_object(self_obj);
     end
 end
@@ -433,9 +433,9 @@ begin
             add_timer_event(self_obj, game_ticks(1), 1);
         end
         if (sequence == 3) then begin
-            set_global_var(867, global_var(867) bwor 8);
-            if (global_var(871) == 1) then begin
-                set_global_var(871, 2);
+            set_global_var(GVAR_AQ_ALBUQ_DEATHS, global_var(GVAR_AQ_ALBUQ_DEATHS) bwor 8);
+            if (global_var(GVAR_AQ_QUEST_KILL_WILLIAM_WIFE) == 1) then begin
+                set_global_var(GVAR_AQ_QUEST_KILL_WILLIAM_WIFE, 2);
             end
             game_ui_enable;
             add_timer_event(self_obj, game_ticks(0), 1);
@@ -453,8 +453,8 @@ end
 procedure Node998
 begin
     set_local_var(4, 1);
-    set_global_var(866, global_var(866) bwor 1048576);
-    float_msg(self_obj, message_str(1614, 400), 2);
+    set_global_var(GVAR_AQ_ALBUQ_SPEECHES, global_var(GVAR_AQ_ALBUQ_SPEECHES) bwor 1048576);
+    float_msg(self_obj, mstr(400), FLOAT_MSG_RED);
 end
 
 procedure Node999
@@ -466,220 +466,220 @@ end
 
 procedure Node500
 begin
-    gsay_reply(1614, 500);
-    giq_option(4, 1614, 501, Node550, 50);
-    giq_option(-3, 1614, 502, Node1100, 50);
+    Reply(500);
+    giq_option(4, NAME, 501, Node550, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 502, Node1100, NEUTRAL_REACTION);
 end
 
 procedure Node550
 begin
-    gsay_reply(1614, 550);
-    giq_option(4, 1614, 551, Node600, 50);
-    giq_option(4, 1614, 552, Node850, 50);
-    if ((global_var(871) == 1) and (get_critter_stat(dude_obj, 5) > 5) and not((global_var(867) bwand 8) != 0) and not((global_var(867) bwand 4) != 0) and (obj_is_carrying_obj_pid(dude_obj, 4) or obj_is_carrying_obj_pid(dude_obj, 45) or obj_is_carrying_obj_pid(dude_obj, 116) or obj_is_carrying_obj_pid(dude_obj, 517) or obj_is_carrying_obj_pid(dude_obj, 592) or obj_is_carrying_obj_pid(dude_obj, 236))) then begin
-        giq_option(4, 1614, 553, Node1050, 50);
+    Reply(550);
+    giq_option(4, NAME, 551, Node600, NEUTRAL_REACTION);
+    giq_option(4, NAME, 552, Node850, NEUTRAL_REACTION);
+    if ((global_var(GVAR_AQ_QUEST_KILL_WILLIAM_WIFE) == 1) and (get_critter_stat(dude_obj, STAT_ag) > 5) and not((global_var(GVAR_AQ_ALBUQ_DEATHS) bwand 8) != 0) and not((global_var(GVAR_AQ_ALBUQ_DEATHS) bwand 4) != 0) and (obj_is_carrying_obj_pid(dude_obj, 4) or obj_is_carrying_obj_pid(dude_obj, 45) or obj_is_carrying_obj_pid(dude_obj, 116) or obj_is_carrying_obj_pid(dude_obj, 517) or obj_is_carrying_obj_pid(dude_obj, 592) or obj_is_carrying_obj_pid(dude_obj, 236))) then begin
+        giq_option(4, NAME, 553, Node1050, NEUTRAL_REACTION);
     end
-    if ((global_var(871) == 1) and (get_critter_stat(dude_obj, 5) < 6) and not((global_var(867) bwand 8) != 0) and not((global_var(867) bwand 4) != 0) and (obj_is_carrying_obj_pid(dude_obj, 4) or obj_is_carrying_obj_pid(dude_obj, 45) or obj_is_carrying_obj_pid(dude_obj, 116) or obj_is_carrying_obj_pid(dude_obj, 517) or obj_is_carrying_obj_pid(dude_obj, 592) or obj_is_carrying_obj_pid(dude_obj, 236))) then begin
-        giq_option(4, 1614, 553, Node1000, 50);
+    if ((global_var(GVAR_AQ_QUEST_KILL_WILLIAM_WIFE) == 1) and (get_critter_stat(dude_obj, STAT_ag) < 6) and not((global_var(GVAR_AQ_ALBUQ_DEATHS) bwand 8) != 0) and not((global_var(GVAR_AQ_ALBUQ_DEATHS) bwand 4) != 0) and (obj_is_carrying_obj_pid(dude_obj, 4) or obj_is_carrying_obj_pid(dude_obj, 45) or obj_is_carrying_obj_pid(dude_obj, 116) or obj_is_carrying_obj_pid(dude_obj, 517) or obj_is_carrying_obj_pid(dude_obj, 592) or obj_is_carrying_obj_pid(dude_obj, 236))) then begin
+        giq_option(4, NAME, 553, Node1000, NEUTRAL_REACTION);
     end
-    if ((global_var(871) == 1) and not((global_var(867) bwand 8) != 0) and not((global_var(867) bwand 4) != 0)) then begin
-        giq_option(4, 1614, 554, Node998, 50);
+    if ((global_var(GVAR_AQ_QUEST_KILL_WILLIAM_WIFE) == 1) and not((global_var(GVAR_AQ_ALBUQ_DEATHS) bwand 8) != 0) and not((global_var(GVAR_AQ_ALBUQ_DEATHS) bwand 4) != 0)) then begin
+        giq_option(4, NAME, 554, Node998, NEUTRAL_REACTION);
     end
-    giq_option(4, 1614, 555, Node999, 50);
+    giq_option(4, NAME, 555, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node600
 begin
-    gsay_reply(1614, 600);
-    giq_option(4, 1614, 601, Node650, 50);
-    giq_option(4, 1614, 602, Node750, 50);
-    giq_option(4, 1614, 603, Node900, 50);
-    giq_option(4, 1614, 604, Node800, 50);
-    giq_option(4, 1614, 605, Node850, 50);
-    if ((global_var(871) == 1) and (get_critter_stat(dude_obj, 5) > 5) and not((global_var(867) bwand 8) != 0) and not((global_var(867) bwand 4) != 0) and (obj_is_carrying_obj_pid(dude_obj, 4) or obj_is_carrying_obj_pid(dude_obj, 45) or obj_is_carrying_obj_pid(dude_obj, 116) or obj_is_carrying_obj_pid(dude_obj, 517) or obj_is_carrying_obj_pid(dude_obj, 592) or obj_is_carrying_obj_pid(dude_obj, 236))) then begin
-        giq_option(4, 1614, 606, Node1050, 50);
-    end
-    if ((global_var(871) == 1) and (get_critter_stat(dude_obj, 5) < 6) and not((global_var(867) bwand 8) != 0) and not((global_var(867) bwand 4) != 0) and (obj_is_carrying_obj_pid(dude_obj, 4) or obj_is_carrying_obj_pid(dude_obj, 45) or obj_is_carrying_obj_pid(dude_obj, 116) or obj_is_carrying_obj_pid(dude_obj, 517) or obj_is_carrying_obj_pid(dude_obj, 592) or obj_is_carrying_obj_pid(dude_obj, 236))) then begin
-        giq_option(4, 1614, 606, Node1000, 50);
-    end
-    if ((global_var(871) == 1) and not((global_var(867) bwand 8) != 0) and not((global_var(867) bwand 4) != 0)) then begin
-        giq_option(4, 1614, 607, Node998, 50);
-    end
-    giq_option(4, 1614, 608, Node999, 50);
+    Reply(600);
+    giq_option(4, NAME, 601, Node650, NEUTRAL_REACTION);
+    giq_option(4, NAME, 602, Node750, NEUTRAL_REACTION);
+    giq_option(4, NAME, 603, Node900, NEUTRAL_REACTION);
+    giq_option(4, NAME, 604, Node800, NEUTRAL_REACTION);
+    giq_option(4, NAME, 605, Node850, NEUTRAL_REACTION);
+    if ((global_var(GVAR_AQ_QUEST_KILL_WILLIAM_WIFE) == 1) and (get_critter_stat(dude_obj, STAT_ag) > 5) and not((global_var(GVAR_AQ_ALBUQ_DEATHS) bwand 8) != 0) and not((global_var(GVAR_AQ_ALBUQ_DEATHS) bwand 4) != 0) and (obj_is_carrying_obj_pid(dude_obj, 4) or obj_is_carrying_obj_pid(dude_obj, 45) or obj_is_carrying_obj_pid(dude_obj, 116) or obj_is_carrying_obj_pid(dude_obj, 517) or obj_is_carrying_obj_pid(dude_obj, 592) or obj_is_carrying_obj_pid(dude_obj, 236))) then begin
+        giq_option(4, NAME, 606, Node1050, NEUTRAL_REACTION);
+    end
+    if ((global_var(GVAR_AQ_QUEST_KILL_WILLIAM_WIFE) == 1) and (get_critter_stat(dude_obj, STAT_ag) < 6) and not((global_var(GVAR_AQ_ALBUQ_DEATHS) bwand 8) != 0) and not((global_var(GVAR_AQ_ALBUQ_DEATHS) bwand 4) != 0) and (obj_is_carrying_obj_pid(dude_obj, 4) or obj_is_carrying_obj_pid(dude_obj, 45) or obj_is_carrying_obj_pid(dude_obj, 116) or obj_is_carrying_obj_pid(dude_obj, 517) or obj_is_carrying_obj_pid(dude_obj, 592) or obj_is_carrying_obj_pid(dude_obj, 236))) then begin
+        giq_option(4, NAME, 606, Node1000, NEUTRAL_REACTION);
+    end
+    if ((global_var(GVAR_AQ_QUEST_KILL_WILLIAM_WIFE) == 1) and not((global_var(GVAR_AQ_ALBUQ_DEATHS) bwand 8) != 0) and not((global_var(GVAR_AQ_ALBUQ_DEATHS) bwand 4) != 0)) then begin
+        giq_option(4, NAME, 607, Node998, NEUTRAL_REACTION);
+    end
+    giq_option(4, NAME, 608, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node650
 begin
-    gsay_reply(1614, 650);
-    giq_option(6, 1614, 651, Node700, 50);
-    giq_option(4, 1614, 652, Node750, 50);
-    giq_option(4, 1614, 653, Node900, 50);
-    giq_option(4, 1614, 654, Node800, 50);
-    giq_option(4, 1614, 655, Node850, 50);
-    if ((global_var(871) == 1) and (get_critter_stat(dude_obj, 5) > 5) and not((global_var(867) bwand 8) != 0) and not((global_var(867) bwand 4) != 0) and (obj_is_carrying_obj_pid(dude_obj, 4) or obj_is_carrying_obj_pid(dude_obj, 45) or obj_is_carrying_obj_pid(dude_obj, 116) or obj_is_carrying_obj_pid(dude_obj, 517) or obj_is_carrying_obj_pid(dude_obj, 592) or obj_is_carrying_obj_pid(dude_obj, 236))) then begin
-        giq_option(4, 1614, 656, Node1050, 50);
-    end
-    if ((global_var(871) == 1) and (get_critter_stat(dude_obj, 5) < 6) and not((global_var(867) bwand 8) != 0) and not((global_var(867) bwand 4) != 0) and (obj_is_carrying_obj_pid(dude_obj, 4) or obj_is_carrying_obj_pid(dude_obj, 45) or obj_is_carrying_obj_pid(dude_obj, 116) or obj_is_carrying_obj_pid(dude_obj, 517) or obj_is_carrying_obj_pid(dude_obj, 592) or obj_is_carrying_obj_pid(dude_obj, 236))) then begin
-        giq_option(4, 1614, 656, Node1000, 50);
-    end
-    if ((global_var(871) == 1) and not((global_var(867) bwand 8) != 0) and not((global_var(867) bwand 4) != 0)) then begin
-        giq_option(4, 1614, 657, Node998, 50);
-    end
-    giq_option(4, 1614, 658, Node999, 50);
+    Reply(650);
+    giq_option(6, NAME, 651, Node700, NEUTRAL_REACTION);
+    giq_option(4, NAME, 652, Node750, NEUTRAL_REACTION);
+    giq_option(4, NAME, 653, Node900, NEUTRAL_REACTION);
+    giq_option(4, NAME, 654, Node800, NEUTRAL_REACTION);
+    giq_option(4, NAME, 655, Node850, NEUTRAL_REACTION);
+    if ((global_var(GVAR_AQ_QUEST_KILL_WILLIAM_WIFE) == 1) and (get_critter_stat(dude_obj, STAT_ag) > 5) and not((global_var(GVAR_AQ_ALBUQ_DEATHS) bwand 8) != 0) and not((global_var(GVAR_AQ_ALBUQ_DEATHS) bwand 4) != 0) and (obj_is_carrying_obj_pid(dude_obj, 4) or obj_is_carrying_obj_pid(dude_obj, 45) or obj_is_carrying_obj_pid(dude_obj, 116) or obj_is_carrying_obj_pid(dude_obj, 517) or obj_is_carrying_obj_pid(dude_obj, 592) or obj_is_carrying_obj_pid(dude_obj, 236))) then begin
+        giq_option(4, NAME, 656, Node1050, NEUTRAL_REACTION);
+    end
+    if ((global_var(GVAR_AQ_QUEST_KILL_WILLIAM_WIFE) == 1) and (get_critter_stat(dude_obj, STAT_ag) < 6) and not((global_var(GVAR_AQ_ALBUQ_DEATHS) bwand 8) != 0) and not((global_var(GVAR_AQ_ALBUQ_DEATHS) bwand 4) != 0) and (obj_is_carrying_obj_pid(dude_obj, 4) or obj_is_carrying_obj_pid(dude_obj, 45) or obj_is_carrying_obj_pid(dude_obj, 116) or obj_is_carrying_obj_pid(dude_obj, 517) or obj_is_carrying_obj_pid(dude_obj, 592) or obj_is_carrying_obj_pid(dude_obj, 236))) then begin
+        giq_option(4, NAME, 656, Node1000, NEUTRAL_REACTION);
+    end
+    if ((global_var(GVAR_AQ_QUEST_KILL_WILLIAM_WIFE) == 1) and not((global_var(GVAR_AQ_ALBUQ_DEATHS) bwand 8) != 0) and not((global_var(GVAR_AQ_ALBUQ_DEATHS) bwand 4) != 0)) then begin
+        giq_option(4, NAME, 657, Node998, NEUTRAL_REACTION);
+    end
+    giq_option(4, NAME, 658, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node700
 begin
-    gsay_reply(1614, 700);
-    giq_option(4, 1614, 701, Node750, 50);
-    giq_option(4, 1614, 702, Node900, 50);
-    giq_option(4, 1614, 703, Node800, 50);
-    giq_option(4, 1614, 704, Node850, 50);
-    if ((global_var(871) == 1) and (get_critter_stat(dude_obj, 5) > 5) and not((global_var(867) bwand 8) != 0) and not((global_var(867) bwand 4) != 0) and (obj_is_carrying_obj_pid(dude_obj, 4) or obj_is_carrying_obj_pid(dude_obj, 45) or obj_is_carrying_obj_pid(dude_obj, 116) or obj_is_carrying_obj_pid(dude_obj, 517) or obj_is_carrying_obj_pid(dude_obj, 592) or obj_is_carrying_obj_pid(dude_obj, 236))) then begin
-        giq_option(4, 1614, 705, Node1050, 50);
+    Reply(700);
+    giq_option(4, NAME, 701, Node750, NEUTRAL_REACTION);
+    giq_option(4, NAME, 702, Node900, NEUTRAL_REACTION);
+    giq_option(4, NAME, 703, Node800, NEUTRAL_REACTION);
+    giq_option(4, NAME, 704, Node850, NEUTRAL_REACTION);
+    if ((global_var(GVAR_AQ_QUEST_KILL_WILLIAM_WIFE) == 1) and (get_critter_stat(dude_obj, STAT_ag) > 5) and not((global_var(GVAR_AQ_ALBUQ_DEATHS) bwand 8) != 0) and not((global_var(GVAR_AQ_ALBUQ_DEATHS) bwand 4) != 0) and (obj_is_carrying_obj_pid(dude_obj, 4) or obj_is_carrying_obj_pid(dude_obj, 45) or obj_is_carrying_obj_pid(dude_obj, 116) or obj_is_carrying_obj_pid(dude_obj, 517) or obj_is_carrying_obj_pid(dude_obj, 592) or obj_is_carrying_obj_pid(dude_obj, 236))) then begin
+        giq_option(4, NAME, 705, Node1050, NEUTRAL_REACTION);
     end
-    if ((global_var(871) == 1) and (get_critter_stat(dude_obj, 5) < 6) and not((global_var(867) bwand 8) != 0) and not((global_var(867) bwand 4) != 0) and (obj_is_carrying_obj_pid(dude_obj, 4) or obj_is_carrying_obj_pid(dude_obj, 45) or obj_is_carrying_obj_pid(dude_obj, 116) or obj_is_carrying_obj_pid(dude_obj, 517) or obj_is_carrying_obj_pid(dude_obj, 592) or obj_is_carrying_obj_pid(dude_obj, 236))) then begin
-        giq_option(4, 1614, 705, Node1000, 50);
+    if ((global_var(GVAR_AQ_QUEST_KILL_WILLIAM_WIFE) == 1) and (get_critter_stat(dude_obj, STAT_ag) < 6) and not((global_var(GVAR_AQ_ALBUQ_DEATHS) bwand 8) != 0) and not((global_var(GVAR_AQ_ALBUQ_DEATHS) bwand 4) != 0) and (obj_is_carrying_obj_pid(dude_obj, 4) or obj_is_carrying_obj_pid(dude_obj, 45) or obj_is_carrying_obj_pid(dude_obj, 116) or obj_is_carrying_obj_pid(dude_obj, 517) or obj_is_carrying_obj_pid(dude_obj, 592) or obj_is_carrying_obj_pid(dude_obj, 236))) then begin
+        giq_option(4, NAME, 705, Node1000, NEUTRAL_REACTION);
     end
-    if ((global_var(871) == 1) and not((global_var(867) bwand 8) != 0) and not((global_var(867) bwand 4) != 0)) then begin
-        giq_option(4, 1614, 706, Node998, 50);
+    if ((global_var(GVAR_AQ_QUEST_KILL_WILLIAM_WIFE) == 1) and not((global_var(GVAR_AQ_ALBUQ_DEATHS) bwand 8) != 0) and not((global_var(GVAR_AQ_ALBUQ_DEATHS) bwand 4) != 0)) then begin
+        giq_option(4, NAME, 706, Node998, NEUTRAL_REACTION);
     end
-    giq_option(4, 1614, 707, Node999, 50);
+    giq_option(4, NAME, 707, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node750
 begin
-    gsay_reply(1614, 750);
-    giq_option(4, 1614, 751, Node650, 50);
-    giq_option(4, 1614, 752, Node900, 50);
-    giq_option(4, 1614, 753, Node800, 50);
-    giq_option(4, 1614, 754, Node850, 50);
-    if ((global_var(871) == 1) and (get_critter_stat(dude_obj, 5) > 5) and not((global_var(867) bwand 8) != 0) and not((global_var(867) bwand 4) != 0) and (obj_is_carrying_obj_pid(dude_obj, 4) or obj_is_carrying_obj_pid(dude_obj, 45) or obj_is_carrying_obj_pid(dude_obj, 116) or obj_is_carrying_obj_pid(dude_obj, 517) or obj_is_carrying_obj_pid(dude_obj, 592) or obj_is_carrying_obj_pid(dude_obj, 236))) then begin
-        giq_option(4, 1614, 755, Node1050, 50);
+    Reply(750);
+    giq_option(4, NAME, 751, Node650, NEUTRAL_REACTION);
+    giq_option(4, NAME, 752, Node900, NEUTRAL_REACTION);
+    giq_option(4, NAME, 753, Node800, NEUTRAL_REACTION);
+    giq_option(4, NAME, 754, Node850, NEUTRAL_REACTION);
+    if ((global_var(GVAR_AQ_QUEST_KILL_WILLIAM_WIFE) == 1) and (get_critter_stat(dude_obj, STAT_ag) > 5) and not((global_var(GVAR_AQ_ALBUQ_DEATHS) bwand 8) != 0) and not((global_var(GVAR_AQ_ALBUQ_DEATHS) bwand 4) != 0) and (obj_is_carrying_obj_pid(dude_obj, 4) or obj_is_carrying_obj_pid(dude_obj, 45) or obj_is_carrying_obj_pid(dude_obj, 116) or obj_is_carrying_obj_pid(dude_obj, 517) or obj_is_carrying_obj_pid(dude_obj, 592) or obj_is_carrying_obj_pid(dude_obj, 236))) then begin
+        giq_option(4, NAME, 755, Node1050, NEUTRAL_REACTION);
     end
-    if ((global_var(871) == 1) and (get_critter_stat(dude_obj, 5) < 6) and not((global_var(867) bwand 8) != 0) and not((global_var(867) bwand 4) != 0) and (obj_is_carrying_obj_pid(dude_obj, 4) or obj_is_carrying_obj_pid(dude_obj, 45) or obj_is_carrying_obj_pid(dude_obj, 116) or obj_is_carrying_obj_pid(dude_obj, 517) or obj_is_carrying_obj_pid(dude_obj, 592) or obj_is_carrying_obj_pid(dude_obj, 236))) then begin
-        giq_option(4, 1614, 755, Node1000, 50);
+    if ((global_var(GVAR_AQ_QUEST_KILL_WILLIAM_WIFE) == 1) and (get_critter_stat(dude_obj, STAT_ag) < 6) and not((global_var(GVAR_AQ_ALBUQ_DEATHS) bwand 8) != 0) and not((global_var(GVAR_AQ_ALBUQ_DEATHS) bwand 4) != 0) and (obj_is_carrying_obj_pid(dude_obj, 4) or obj_is_carrying_obj_pid(dude_obj, 45) or obj_is_carrying_obj_pid(dude_obj, 116) or obj_is_carrying_obj_pid(dude_obj, 517) or obj_is_carrying_obj_pid(dude_obj, 592) or obj_is_carrying_obj_pid(dude_obj, 236))) then begin
+        giq_option(4, NAME, 755, Node1000, NEUTRAL_REACTION);
     end
-    if ((global_var(871) == 1) and not((global_var(867) bwand 8) != 0) and not((global_var(867) bwand 4) != 0)) then begin
-        giq_option(4, 1614, 756, Node998, 50);
+    if ((global_var(GVAR_AQ_QUEST_KILL_WILLIAM_WIFE) == 1) and not((global_var(GVAR_AQ_ALBUQ_DEATHS) bwand 8) != 0) and not((global_var(GVAR_AQ_ALBUQ_DEATHS) bwand 4) != 0)) then begin
+        giq_option(4, NAME, 756, Node998, NEUTRAL_REACTION);
     end
-    giq_option(4, 1614, 757, Node999, 50);
+    giq_option(4, NAME, 757, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node800
 begin
-    if (get_critter_stat(dude_obj, 34) == 0) then begin
-        gsay_reply(1614, 800);
+    if (dude_gender == 0) then begin
+        Reply(800);
     end
-    if (get_critter_stat(dude_obj, 34) == 1) then begin
-        gsay_reply(1614, 801);
+    if (dude_gender == 1) then begin
+        Reply(801);
     end
-    giq_option(4, 1614, 802, Node650, 50);
-    giq_option(4, 1614, 803, Node750, 50);
-    giq_option(4, 1614, 804, Node900, 50);
-    giq_option(4, 1614, 805, Node850, 50);
-    if ((global_var(871) == 1) and (get_critter_stat(dude_obj, 5) > 5) and not((global_var(867) bwand 8) != 0) and not((global_var(867) bwand 4) != 0) and (obj_is_carrying_obj_pid(dude_obj, 4) or obj_is_carrying_obj_pid(dude_obj, 45) or obj_is_carrying_obj_pid(dude_obj, 116) or obj_is_carrying_obj_pid(dude_obj, 517) or obj_is_carrying_obj_pid(dude_obj, 592) or obj_is_carrying_obj_pid(dude_obj, 236))) then begin
-        giq_option(4, 1614, 806, Node1050, 50);
+    giq_option(4, NAME, 802, Node650, NEUTRAL_REACTION);
+    giq_option(4, NAME, 803, Node750, NEUTRAL_REACTION);
+    giq_option(4, NAME, 804, Node900, NEUTRAL_REACTION);
+    giq_option(4, NAME, 805, Node850, NEUTRAL_REACTION);
+    if ((global_var(GVAR_AQ_QUEST_KILL_WILLIAM_WIFE) == 1) and (get_critter_stat(dude_obj, STAT_ag) > 5) and not((global_var(GVAR_AQ_ALBUQ_DEATHS) bwand 8) != 0) and not((global_var(GVAR_AQ_ALBUQ_DEATHS) bwand 4) != 0) and (obj_is_carrying_obj_pid(dude_obj, 4) or obj_is_carrying_obj_pid(dude_obj, 45) or obj_is_carrying_obj_pid(dude_obj, 116) or obj_is_carrying_obj_pid(dude_obj, 517) or obj_is_carrying_obj_pid(dude_obj, 592) or obj_is_carrying_obj_pid(dude_obj, 236))) then begin
+        giq_option(4, NAME, 806, Node1050, NEUTRAL_REACTION);
     end
-    if ((global_var(871) == 1) and (get_critter_stat(dude_obj, 5) < 6) and not((global_var(867) bwand 8) != 0) and not((global_var(867) bwand 4) != 0) and (obj_is_carrying_obj_pid(dude_obj, 4) or obj_is_carrying_obj_pid(dude_obj, 45) or obj_is_carrying_obj_pid(dude_obj, 116) or obj_is_carrying_obj_pid(dude_obj, 517) or obj_is_carrying_obj_pid(dude_obj, 592) or obj_is_carrying_obj_pid(dude_obj, 236))) then begin
-        giq_option(4, 1614, 806, Node1000, 50);
+    if ((global_var(GVAR_AQ_QUEST_KILL_WILLIAM_WIFE) == 1) and (get_critter_stat(dude_obj, STAT_ag) < 6) and not((global_var(GVAR_AQ_ALBUQ_DEATHS) bwand 8) != 0) and not((global_var(GVAR_AQ_ALBUQ_DEATHS) bwand 4) != 0) and (obj_is_carrying_obj_pid(dude_obj, 4) or obj_is_carrying_obj_pid(dude_obj, 45) or obj_is_carrying_obj_pid(dude_obj, 116) or obj_is_carrying_obj_pid(dude_obj, 517) or obj_is_carrying_obj_pid(dude_obj, 592) or obj_is_carrying_obj_pid(dude_obj, 236))) then begin
+        giq_option(4, NAME, 806, Node1000, NEUTRAL_REACTION);
     end
-    if ((global_var(871) == 1) and not((global_var(867) bwand 8) != 0) and not((global_var(867) bwand 4) != 0)) then begin
-        giq_option(4, 1614, 807, Node998, 50);
+    if ((global_var(GVAR_AQ_QUEST_KILL_WILLIAM_WIFE) == 1) and not((global_var(GVAR_AQ_ALBUQ_DEATHS) bwand 8) != 0) and not((global_var(GVAR_AQ_ALBUQ_DEATHS) bwand 4) != 0)) then begin
+        giq_option(4, NAME, 807, Node998, NEUTRAL_REACTION);
     end
-    giq_option(4, 1614, 808, Node999, 50);
+    giq_option(4, NAME, 808, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node850
 begin
-    gsay_reply(1614, 850);
-    giq_option(4, 1614, 851, Node600, 50);
-    if ((global_var(871) == 1) and (get_critter_stat(dude_obj, 5) > 5) and not((global_var(867) bwand 8) != 0) and not((global_var(867) bwand 4) != 0) and (obj_is_carrying_obj_pid(dude_obj, 4) or obj_is_carrying_obj_pid(dude_obj, 45) or obj_is_carrying_obj_pid(dude_obj, 116) or obj_is_carrying_obj_pid(dude_obj, 517) or obj_is_carrying_obj_pid(dude_obj, 592) or obj_is_carrying_obj_pid(dude_obj, 236))) then begin
-        giq_option(4, 1614, 852, Node1050, 50);
+    Reply(850);
+    giq_option(4, NAME, 851, Node600, NEUTRAL_REACTION);
+    if ((global_var(GVAR_AQ_QUEST_KILL_WILLIAM_WIFE) == 1) and (get_critter_stat(dude_obj, STAT_ag) > 5) and not((global_var(GVAR_AQ_ALBUQ_DEATHS) bwand 8) != 0) and not((global_var(GVAR_AQ_ALBUQ_DEATHS) bwand 4) != 0) and (obj_is_carrying_obj_pid(dude_obj, 4) or obj_is_carrying_obj_pid(dude_obj, 45) or obj_is_carrying_obj_pid(dude_obj, 116) or obj_is_carrying_obj_pid(dude_obj, 517) or obj_is_carrying_obj_pid(dude_obj, 592) or obj_is_carrying_obj_pid(dude_obj, 236))) then begin
+        giq_option(4, NAME, 852, Node1050, NEUTRAL_REACTION);
     end
-    if ((global_var(871) == 1) and (get_critter_stat(dude_obj, 5) < 6) and not((global_var(867) bwand 8) != 0) and not((global_var(867) bwand 4) != 0) and (obj_is_carrying_obj_pid(dude_obj, 4) or obj_is_carrying_obj_pid(dude_obj, 45) or obj_is_carrying_obj_pid(dude_obj, 116) or obj_is_carrying_obj_pid(dude_obj, 517) or obj_is_carrying_obj_pid(dude_obj, 592) or obj_is_carrying_obj_pid(dude_obj, 236))) then begin
-        giq_option(4, 1614, 852, Node1000, 50);
+    if ((global_var(GVAR_AQ_QUEST_KILL_WILLIAM_WIFE) == 1) and (get_critter_stat(dude_obj, STAT_ag) < 6) and not((global_var(GVAR_AQ_ALBUQ_DEATHS) bwand 8) != 0) and not((global_var(GVAR_AQ_ALBUQ_DEATHS) bwand 4) != 0) and (obj_is_carrying_obj_pid(dude_obj, 4) or obj_is_carrying_obj_pid(dude_obj, 45) or obj_is_carrying_obj_pid(dude_obj, 116) or obj_is_carrying_obj_pid(dude_obj, 517) or obj_is_carrying_obj_pid(dude_obj, 592) or obj_is_carrying_obj_pid(dude_obj, 236))) then begin
+        giq_option(4, NAME, 852, Node1000, NEUTRAL_REACTION);
     end
-    if ((global_var(871) == 1) and not((global_var(867) bwand 8) != 0) and not((global_var(867) bwand 4) != 0)) then begin
-        giq_option(4, 1614, 853, Node998, 50);
+    if ((global_var(GVAR_AQ_QUEST_KILL_WILLIAM_WIFE) == 1) and not((global_var(GVAR_AQ_ALBUQ_DEATHS) bwand 8) != 0) and not((global_var(GVAR_AQ_ALBUQ_DEATHS) bwand 4) != 0)) then begin
+        giq_option(4, NAME, 853, Node998, NEUTRAL_REACTION);
     end
-    giq_option(4, 1614, 854, Node999, 50);
+    giq_option(4, NAME, 854, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node900
 begin
-    gsay_reply(1614, 900);
-    giq_option(4, 1614, 901, Node650, 50);
-    giq_option(4, 1614, 902, Node750, 50);
-    giq_option(4, 1614, 903, Node800, 50);
-    giq_option(4, 1614, 904, Node850, 50);
-    if ((global_var(871) == 1) and (get_critter_stat(dude_obj, 5) > 5) and not((global_var(867) bwand 8) != 0) and not((global_var(867) bwand 4) != 0) and (obj_is_carrying_obj_pid(dude_obj, 4) or obj_is_carrying_obj_pid(dude_obj, 45) or obj_is_carrying_obj_pid(dude_obj, 116) or obj_is_carrying_obj_pid(dude_obj, 517) or obj_is_carrying_obj_pid(dude_obj, 592) or obj_is_carrying_obj_pid(dude_obj, 236))) then begin
-        giq_option(4, 1614, 905, Node1050, 50);
+    Reply(900);
+    giq_option(4, NAME, 901, Node650, NEUTRAL_REACTION);
+    giq_option(4, NAME, 902, Node750, NEUTRAL_REACTION);
+    giq_option(4, NAME, 903, Node800, NEUTRAL_REACTION);
+    giq_option(4, NAME, 904, Node850, NEUTRAL_REACTION);
+    if ((global_var(GVAR_AQ_QUEST_KILL_WILLIAM_WIFE) == 1) and (get_critter_stat(dude_obj, STAT_ag) > 5) and not((global_var(GVAR_AQ_ALBUQ_DEATHS) bwand 8) != 0) and not((global_var(GVAR_AQ_ALBUQ_DEATHS) bwand 4) != 0) and (obj_is_carrying_obj_pid(dude_obj, 4) or obj_is_carrying_obj_pid(dude_obj, 45) or obj_is_carrying_obj_pid(dude_obj, 116) or obj_is_carrying_obj_pid(dude_obj, 517) or obj_is_carrying_obj_pid(dude_obj, 592) or obj_is_carrying_obj_pid(dude_obj, 236))) then begin
+        giq_option(4, NAME, 905, Node1050, NEUTRAL_REACTION);
     end
-    if ((global_var(871) == 1) and (get_critter_stat(dude_obj, 5) < 6) and not((global_var(867) bwand 8) != 0) and not((global_var(867) bwand 4) != 0) and (obj_is_carrying_obj_pid(dude_obj, 4) or obj_is_carrying_obj_pid(dude_obj, 45) or obj_is_carrying_obj_pid(dude_obj, 116) or obj_is_carrying_obj_pid(dude_obj, 517) or obj_is_carrying_obj_pid(dude_obj, 592) or obj_is_carrying_obj_pid(dude_obj, 236))) then begin
-        giq_option(4, 1614, 905, Node1000, 50);
+    if ((global_var(GVAR_AQ_QUEST_KILL_WILLIAM_WIFE) == 1) and (get_critter_stat(dude_obj, STAT_ag) < 6) and not((global_var(GVAR_AQ_ALBUQ_DEATHS) bwand 8) != 0) and not((global_var(GVAR_AQ_ALBUQ_DEATHS) bwand 4) != 0) and (obj_is_carrying_obj_pid(dude_obj, 4) or obj_is_carrying_obj_pid(dude_obj, 45) or obj_is_carrying_obj_pid(dude_obj, 116) or obj_is_carrying_obj_pid(dude_obj, 517) or obj_is_carrying_obj_pid(dude_obj, 592) or obj_is_carrying_obj_pid(dude_obj, 236))) then begin
+        giq_option(4, NAME, 905, Node1000, NEUTRAL_REACTION);
     end
-    if ((global_var(871) == 1) and not((global_var(867) bwand 8) != 0) and not((global_var(867) bwand 4) != 0)) then begin
-        giq_option(4, 1614, 906, Node998, 50);
+    if ((global_var(GVAR_AQ_QUEST_KILL_WILLIAM_WIFE) == 1) and not((global_var(GVAR_AQ_ALBUQ_DEATHS) bwand 8) != 0) and not((global_var(GVAR_AQ_ALBUQ_DEATHS) bwand 4) != 0)) then begin
+        giq_option(4, NAME, 906, Node998, NEUTRAL_REACTION);
     end
-    giq_option(4, 1614, 907, Node999, 50);
+    giq_option(4, NAME, 907, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node950
 begin
-    if (get_critter_stat(dude_obj, 34) == 0) then begin
-        gsay_reply(1614, 950);
+    if (dude_gender == 0) then begin
+        Reply(950);
     end
-    if (get_critter_stat(dude_obj, 34) == 1) then begin
-        gsay_reply(1614, 951);
+    if (dude_gender == 1) then begin
+        Reply(951);
     end
-    giq_option(4, 1614, 952, Node600, 50);
-    giq_option(4, 1614, 953, Node850, 50);
-    if ((global_var(871) == 1) and (get_critter_stat(dude_obj, 5) > 5) and not((global_var(867) bwand 8) != 0) and not((global_var(867) bwand 4) != 0) and (obj_is_carrying_obj_pid(dude_obj, 4) or obj_is_carrying_obj_pid(dude_obj, 45) or obj_is_carrying_obj_pid(dude_obj, 116) or obj_is_carrying_obj_pid(dude_obj, 517) or obj_is_carrying_obj_pid(dude_obj, 592) or obj_is_carrying_obj_pid(dude_obj, 236))) then begin
-        giq_option(4, 1614, 954, Node1050, 50);
+    giq_option(4, NAME, 952, Node600, NEUTRAL_REACTION);
+    giq_option(4, NAME, 953, Node850, NEUTRAL_REACTION);
+    if ((global_var(GVAR_AQ_QUEST_KILL_WILLIAM_WIFE) == 1) and (get_critter_stat(dude_obj, STAT_ag) > 5) and not((global_var(GVAR_AQ_ALBUQ_DEATHS) bwand 8) != 0) and not((global_var(GVAR_AQ_ALBUQ_DEATHS) bwand 4) != 0) and (obj_is_carrying_obj_pid(dude_obj, 4) or obj_is_carrying_obj_pid(dude_obj, 45) or obj_is_carrying_obj_pid(dude_obj, 116) or obj_is_carrying_obj_pid(dude_obj, 517) or obj_is_carrying_obj_pid(dude_obj, 592) or obj_is_carrying_obj_pid(dude_obj, 236))) then begin
+        giq_option(4, NAME, 954, Node1050, NEUTRAL_REACTION);
     end
-    if ((global_var(871) == 1) and (get_critter_stat(dude_obj, 5) < 6) and not((global_var(867) bwand 8) != 0) and not((global_var(867) bwand 4) != 0) and (obj_is_carrying_obj_pid(dude_obj, 4) or obj_is_carrying_obj_pid(dude_obj, 45) or obj_is_carrying_obj_pid(dude_obj, 116) or obj_is_carrying_obj_pid(dude_obj, 517) or obj_is_carrying_obj_pid(dude_obj, 592) or obj_is_carrying_obj_pid(dude_obj, 236))) then begin
-        giq_option(4, 1614, 954, Node1000, 50);
+    if ((global_var(GVAR_AQ_QUEST_KILL_WILLIAM_WIFE) == 1) and (get_critter_stat(dude_obj, STAT_ag) < 6) and not((global_var(GVAR_AQ_ALBUQ_DEATHS) bwand 8) != 0) and not((global_var(GVAR_AQ_ALBUQ_DEATHS) bwand 4) != 0) and (obj_is_carrying_obj_pid(dude_obj, 4) or obj_is_carrying_obj_pid(dude_obj, 45) or obj_is_carrying_obj_pid(dude_obj, 116) or obj_is_carrying_obj_pid(dude_obj, 517) or obj_is_carrying_obj_pid(dude_obj, 592) or obj_is_carrying_obj_pid(dude_obj, 236))) then begin
+        giq_option(4, NAME, 954, Node1000, NEUTRAL_REACTION);
     end
-    if ((global_var(871) == 1) and not((global_var(867) bwand 8) != 0) and not((global_var(867) bwand 4) != 0)) then begin
-        giq_option(4, 1614, 955, Node998, 50);
+    if ((global_var(GVAR_AQ_QUEST_KILL_WILLIAM_WIFE) == 1) and not((global_var(GVAR_AQ_ALBUQ_DEATHS) bwand 8) != 0) and not((global_var(GVAR_AQ_ALBUQ_DEATHS) bwand 4) != 0)) then begin
+        giq_option(4, NAME, 955, Node998, NEUTRAL_REACTION);
     end
-    giq_option(4, 1614, 956, Node999, 50);
-    if ((global_var(871) == 1) and not((global_var(867) bwand 8) != 0) and not((global_var(867) bwand 4) != 0)) then begin
-        giq_option(-3, 1614, 957, Node998, 50);
+    giq_option(4, NAME, 956, Node999, NEUTRAL_REACTION);
+    if ((global_var(GVAR_AQ_QUEST_KILL_WILLIAM_WIFE) == 1) and not((global_var(GVAR_AQ_ALBUQ_DEATHS) bwand 8) != 0) and not((global_var(GVAR_AQ_ALBUQ_DEATHS) bwand 4) != 0)) then begin
+        giq_option(-3, NAME, 957, Node998, NEUTRAL_REACTION);
     end
-    giq_option(-3, 1614, 958, Node999, 50);
+    giq_option(-3, NAME, 958, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node1000
 begin
-    if (get_critter_stat(dude_obj, 34) == 0) then begin
-        gsay_reply(1614, 1000);
+    if (dude_gender == 0) then begin
+        Reply(1000);
     end
-    if (get_critter_stat(dude_obj, 34) == 1) then begin
-        gsay_reply(1614, 1001);
+    if (dude_gender == 1) then begin
+        Reply(1001);
     end
-    giq_option(4, 1614, 1002, Node998, 50);
+    giq_option(4, NAME, 1002, Node998, NEUTRAL_REACTION);
 end
 
 procedure Node1050
 begin
-    if (get_critter_stat(dude_obj, 34) == 0) then begin
-        gsay_reply(1614, 1050);
+    if (dude_gender == 0) then begin
+        Reply(1050);
     end
-    if (get_critter_stat(dude_obj, 34) == 1) then begin
-        gsay_reply(1614, 1051);
+    if (dude_gender == 1) then begin
+        Reply(1051);
     end
-    giq_option(4, 1614, 1052, Kill_is_sucesfull, 50);
+    giq_option(4, NAME, 1052, Kill_is_sucesfull, NEUTRAL_REACTION);
 end
 
 procedure Node1100
 begin
-    gsay_reply(1614, 1100);
-    if ((global_var(871) == 1) and not((global_var(867) bwand 8) != 0) and not((global_var(867) bwand 4) != 0)) then begin
-        giq_option(-3, 1614, 1101, Node998, 50);
+    Reply(1100);
+    if ((global_var(GVAR_AQ_QUEST_KILL_WILLIAM_WIFE) == 1) and not((global_var(GVAR_AQ_ALBUQ_DEATHS) bwand 8) != 0) and not((global_var(GVAR_AQ_ALBUQ_DEATHS) bwand 4) != 0)) then begin
+        giq_option(-3, NAME, 1101, Node998, NEUTRAL_REACTION);
     end
-    giq_option(-3, 1614, 1102, Node999, 50);
+    giq_option(-3, NAME, 1102, Node999, NEUTRAL_REACTION);
 end
 
 procedure Kill_is_sucesfull
@@ -687,4 +687,3 @@ begin
     set_local_var(6, local_var(6) bwor 1);
     call Node999();
 end
-
