diff --git a/Mods/Resurrection/Original/ADRAJDIV.ssl b/Mods/Resurrection/ReDefine/ADRAJDIV.ssl
index 5719a8b..ec09343 100644
--- a/Mods/Resurrection/Original/ADRAJDIV.ssl
+++ b/Mods/Resurrection/ReDefine/ADRAJDIV.ssl
@@ -83,12 +83,12 @@ begin
             return 1;
         end
     end
-    if ((party_member_obj(16777779) != 0) and ((global_var(910) bwand 32) == 0)) then begin
+    if ((party_member_obj(16777779) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 32) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777779)) <= 5) then begin
             return 1;
         end
     end
-    if ((party_member_obj(16777811) != 0) and ((global_var(910) bwand 64) == 0)) then begin
+    if ((party_member_obj(16777811) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 64) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777811)) <= 5) then begin
             return 1;
         end
@@ -103,7 +103,7 @@ begin
             return 1;
         end
     end
-    if ((party_member_obj(16777814) != 0) and ((global_var(910) bwand 128) == 0)) then begin
+    if ((party_member_obj(16777814) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 128) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777814)) <= 5) then begin
             return 1;
         end
@@ -160,15 +160,15 @@ end
 
 procedure critter_p_proc
 begin
-    if ((map_var(1) == 1) and (has_trait(1, self_obj, 666) == 1) and obj_can_see_obj(self_obj, dude_obj)) then begin
+    if ((map_var(1) == 1) and (obj_is_visible_flag(self_obj) == 1) and obj_can_see_obj(self_obj, dude_obj)) then begin
         if (critter_is_fleeing(self_obj)) then begin
             if (anim_busy(self_obj) == 0) then begin
                 if ((tile_distance_objs(self_obj, dude_obj) < 8) or obj_can_see_obj(self_obj, dude_obj)) then begin
                     if (anim_busy(self_obj) == 0) then begin
-                        GVar11 := rotation_to_tile(tile_num(dude_obj), tile_num(self_obj));
+                        GVar11 := rotation_to_tile(dude_tile, tile_num(self_obj));
                         animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(self_obj), GVar11, random(3, 10)), 1);
                         GVar11 := (GVar11 + 1) % 6;
-                        while ((anim_busy(self_obj) == 0) and (GVar11 != rotation_to_tile(tile_num(dude_obj), tile_num(self_obj)))) do begin
+                        while ((anim_busy(self_obj) == 0) and (GVar11 != rotation_to_tile(dude_tile, tile_num(self_obj)))) do begin
                             debug_msg("flee loop: rot == " + GVar11);
                             animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(self_obj), GVar11, random(3, 10)), 1);
                             GVar11 := (GVar11 + 1) % 6;
@@ -178,7 +178,7 @@ begin
             end
         end
         else begin
-            attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+            attack(dude_obj);
         end
     end
     if (local_var(4) == 1) then begin
@@ -187,10 +187,10 @@ begin
             if (anim_busy(self_obj) == 0) then begin
                 if ((tile_distance_objs(self_obj, dude_obj) < 8) or obj_can_see_obj(self_obj, dude_obj)) then begin
                     if (anim_busy(self_obj) == 0) then begin
-                        GVar11 := rotation_to_tile(tile_num(dude_obj), tile_num(self_obj));
+                        GVar11 := rotation_to_tile(dude_tile, tile_num(self_obj));
                         animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(self_obj), GVar11, random(3, 10)), 1);
                         GVar11 := (GVar11 + 1) % 6;
-                        while ((anim_busy(self_obj) == 0) and (GVar11 != rotation_to_tile(tile_num(dude_obj), tile_num(self_obj)))) do begin
+                        while ((anim_busy(self_obj) == 0) and (GVar11 != rotation_to_tile(dude_tile, tile_num(self_obj)))) do begin
                             debug_msg("flee loop: rot == " + GVar11);
                             animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(self_obj), GVar11, random(3, 10)), 1);
                             GVar11 := (GVar11 + 1) % 6;
@@ -200,36 +200,36 @@ begin
             end
         end
         else begin
-            attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+            attack(dude_obj);
         end
     end
-    if ((has_trait(1, self_obj, 666) == 1) and ((local_var(5) bwand 16) == 0) and (((game_time - GVar21) > (10 * 6)) or (GVar21 == 0))) then begin
+    if ((obj_is_visible_flag(self_obj) == 1) and ((local_var(5) bwand 16) == 0) and (((game_time - GVar21) > (10 * 6)) or (GVar21 == 0))) then begin
         if (tile_num(self_obj) == 23277) then begin
-            if (tile_in_tile_rect(22480, 22473, 24280, 24473, tile_num(dude_obj)) and (elevation(dude_obj) == 1)) then begin
+            if (tile_in_tile_rect(22480, 22473, 24280, 24473, dude_tile) and (dude_elevation == 1)) then begin
                 dialogue_system_enter;
                 GVar21 := game_time;
             end
         end
         if (tile_num(self_obj) == 21275) then begin
-            if (tile_in_tile_rect(20480, 20473, 22080, 22073, tile_num(dude_obj)) and (elevation(dude_obj) == 1)) then begin
+            if (tile_in_tile_rect(20480, 20473, 22080, 22073, dude_tile) and (dude_elevation == 1)) then begin
                 dialogue_system_enter;
                 GVar21 := game_time;
             end
         end
         if (tile_num(self_obj) == 20058) then begin
-            if (tile_in_tile_rect(19663, 19655, 21463, 21455, tile_num(dude_obj)) and (elevation(dude_obj) == 1)) then begin
+            if (tile_in_tile_rect(19663, 19655, 21463, 21455, dude_tile) and (dude_elevation == 1)) then begin
                 dialogue_system_enter;
                 GVar21 := game_time;
             end
         end
         if (tile_num(self_obj) == 19076) then begin
-            if (tile_in_tile_rect(18480, 18473, 20280, 20273, tile_num(dude_obj)) and (elevation(dude_obj) == 1)) then begin
+            if (tile_in_tile_rect(18480, 18473, 20280, 20273, dude_tile) and (dude_elevation == 1)) then begin
                 dialogue_system_enter;
                 GVar21 := game_time;
             end
         end
     end
-    if (tile_in_tile_rect(18471, 18465, 24071, 24065, tile_num(dude_obj)) and (elevation(dude_obj) == 1) and not((local_var(5) bwand 4) == 0)) then begin
+    if (tile_in_tile_rect(18471, 18465, 24071, 24065, dude_tile) and (dude_elevation == 1) and not((local_var(5) bwand 4) == 0)) then begin
         set_local_var(5, local_var(5) bwand (-1 - 4));
     end
 end
@@ -245,16 +245,16 @@ procedure talk_p_proc
 begin
     if ((local_var(5) bwand 16) == 0) then begin
         if ((local_var(5) bwand 4) == 0) then begin
-            if (has_trait(1, self_obj, 666) == 1) then begin
-                float_msg(self_obj, message_str(1646, random(300, 306)), 8);
+            if (obj_is_visible_flag(self_obj) == 1) then begin
+                float_msg(self_obj, mstr(random(300, 306)), FLOAT_MSG_YELLOW);
             end
         end
         else begin
-            if (global_var(859) == 1) then begin
-                float_msg(self_obj, message_str(1646, random(307, 312)), 8);
+            if (global_var(GVAR_AQ_MUTANT_HUNTER) == 1) then begin
+                float_msg(self_obj, mstr(random(307, 312)), FLOAT_MSG_YELLOW);
             end
             else begin
-                float_msg(self_obj, message_str(1646, random(307, 313)), 8);
+                float_msg(self_obj, mstr(random(307, 313)), FLOAT_MSG_YELLOW);
             end
         end
     end
@@ -263,58 +263,58 @@ end
 procedure destroy_p_proc
 begin
     if (source_obj == dude_obj) then begin
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(4) > (3 * global_var(5))) or (global_var(2) == 1)) and (global_var(1) == 0)) then begin
-            set_global_var(2, 1);
-            set_global_var(3, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(4) > (3 * global_var(GVAR_GOOD_MONSTER))) or (global_var(GVAR_CHAMPION_REPUTATION) == 1)) and (global_var(GVAR_CHILDKILLER_REPUTATION) == 0)) then begin
+            set_global_var(GVAR_CHAMPION_REPUTATION, 1);
+            set_global_var(GVAR_BERSERKER_REPUTATION, 0);
         end
-        if (metarule(51, self_obj) == 2) then begin
+        if (critter_kill_type(self_obj) == KILL_TYPE_children_kills) then begin
             if (source_obj == dude_obj) then begin
-                set_global_var(2, 0);
-                set_global_var(1, global_var(1) + 1);
-                set_global_var(0, global_var(0) + -15);
+                set_global_var(GVAR_CHAMPION_REPUTATION, 0);
+                set_global_var(GVAR_CHILDKILLER_REPUTATION, global_var(GVAR_CHILDKILLER_REPUTATION) + 1);
+                set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + -15);
                 debug_msg("Player gains " + -15 + " Karma Points.");
-                set_global_var(37, 0);
-                set_global_var(38, 0);
-                set_global_var(39, 0);
-                set_global_var(40, 0);
-                set_global_var(41, 0);
-                set_global_var(42, 0);
-                set_global_var(43, 0);
-                set_global_var(44, 0);
-                set_global_var(45, 0);
-                if (global_var(0) >= 1000) then begin
-                    set_global_var(37, 1);
+                set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+                set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+                set_global_var(GVAR_KARMA_DEFENDER, 0);
+                set_global_var(GVAR_KARMA_WANDERER, 0);
+                set_global_var(GVAR_KARMA_BETRAYER, 0);
+                set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+                set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+                set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+                    set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
                 end
                 else begin
-                    if (global_var(0) >= 750) then begin
-                        set_global_var(38, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
                     end
                     else begin
-                        if (global_var(0) >= 500) then begin
-                            set_global_var(39, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                            set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                         end
                         else begin
-                            if (global_var(0) >= 250) then begin
-                                set_global_var(40, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                                set_global_var(GVAR_KARMA_DEFENDER, 1);
                             end
                             else begin
-                                if (global_var(0) > -250) then begin
-                                    set_global_var(41, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                                    set_global_var(GVAR_KARMA_WANDERER, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -500) then begin
-                                        set_global_var(42, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                        set_global_var(GVAR_KARMA_BETRAYER, 1);
                                     end
                                     else begin
-                                        if (global_var(0) > -750) then begin
-                                            set_global_var(43, 1);
+                                        if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                            set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                         end
                                         else begin
-                                            if (global_var(0) > -1000) then begin
-                                                set_global_var(44, 1);
+                                            if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                                set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                             end
                                             else begin
-                                                set_global_var(45, 1);
+                                                set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                             end
                                         end
                                     end
@@ -323,58 +323,58 @@ begin
                         end
                     end
                 end
-                set_global_var(51, global_var(51) + -8);
+                set_global_var(GVAR_TOWN_REP_ALBUQUERQUE, global_var(GVAR_TOWN_REP_ALBUQUERQUE) + -8);
                 debug_msg("Added " + -8 + " to Town Rep");
             end
         end
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(5) > (2 * global_var(4))) or (global_var(3) == 1))) then begin
-            set_global_var(3, 1);
-            set_global_var(2, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
+            set_global_var(GVAR_BERSERKER_REPUTATION, 1);
+            set_global_var(GVAR_CHAMPION_REPUTATION, 0);
         end
-        set_global_var(0, global_var(0) + 0);
+        set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + 0);
         debug_msg("Player gains " + 0 + " Karma Points.");
-        set_global_var(37, 0);
-        set_global_var(38, 0);
-        set_global_var(39, 0);
-        set_global_var(40, 0);
-        set_global_var(41, 0);
-        set_global_var(42, 0);
-        set_global_var(43, 0);
-        set_global_var(44, 0);
-        set_global_var(45, 0);
-        if (global_var(0) >= 1000) then begin
-            set_global_var(37, 1);
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+        set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+        set_global_var(GVAR_KARMA_DEFENDER, 0);
+        set_global_var(GVAR_KARMA_WANDERER, 0);
+        set_global_var(GVAR_KARMA_BETRAYER, 0);
+        set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+        set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+            set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
         end
         else begin
-            if (global_var(0) >= 750) then begin
-                set_global_var(38, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
             end
             else begin
-                if (global_var(0) >= 500) then begin
-                    set_global_var(39, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                 end
                 else begin
-                    if (global_var(0) >= 250) then begin
-                        set_global_var(40, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                        set_global_var(GVAR_KARMA_DEFENDER, 1);
                     end
                     else begin
-                        if (global_var(0) > -250) then begin
-                            set_global_var(41, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                            set_global_var(GVAR_KARMA_WANDERER, 1);
                         end
                         else begin
-                            if (global_var(0) > -500) then begin
-                                set_global_var(42, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                set_global_var(GVAR_KARMA_BETRAYER, 1);
                             end
                             else begin
-                                if (global_var(0) > -750) then begin
-                                    set_global_var(43, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -1000) then begin
-                                        set_global_var(44, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                     end
                                     else begin
-                                        set_global_var(45, 1);
+                                        set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                     end
                                 end
                             end
@@ -389,13 +389,13 @@ end
 procedure look_at_p_proc
 begin
     script_overrides;
-    display_msg(message_str(1646, 100));
+    display_mstr(100);
 end
 
 procedure description_p_proc
 begin
     script_overrides;
-    display_msg(message_str(1646, 200));
+    display_mstr(200);
 end
 
 procedure use_skill_on_p_proc
@@ -412,12 +412,12 @@ end
 procedure map_enter_p_proc
 begin
     if ((local_var(5) bwand 1) == 0) then begin
-        critter_add_trait(self_obj, 1, 6, 260);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 260);
     end
     else begin
-        critter_add_trait(self_obj, 1, 6, 258);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 258);
     end
-    critter_add_trait(self_obj, 1, 5, 14);
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_PEASANTS);
     if (tile_num(self_obj) == 23277) then begin
         Devka1_obj := self_obj;
     end
@@ -435,7 +435,7 @@ begin
             set_obj_visibility(self_obj, 1);
         end
     end
-    if (((local_var(5) bwand 8) != 0) and (has_trait(1, self_obj, 666) == 0) and not(metarule(22, 0))) then begin
+    if (((local_var(5) bwand 8) != 0) and (obj_is_visible_flag(self_obj) == 0) and not(is_loading_game)) then begin
         set_obj_visibility(self_obj, 0);
         set_local_var(5, local_var(5) bwand (-1 - 8));
     end
@@ -443,14 +443,14 @@ end
 
 procedure map_exit_p_proc
 begin
-    float_msg(self_obj, 0, 0);
+    float_msg_clear(self_obj);
 end
 
 procedure combat_p_proc
 begin
     if ((map_var(1) == 1) and ((local_var(5) bwand 1) == 0)) then begin
-        critter_add_trait(self_obj, 1, 6, 258);
-        set_global_var(864, 1);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 258);
+        set_global_var(GVAR_AQ_ENEMY_HUNTERS, 1);
         set_local_var(5, local_var(5) bwor 1);
     end
 end
@@ -465,7 +465,7 @@ begin
         set_local_var(5, local_var(5) bwand (-1 - 2));
         set_obj_visibility(self_obj, 0);
     end
-    if ((fixed_param == 3) and (has_trait(1, self_obj, 666) == 1)) then begin
+    if ((fixed_param == 3) and (obj_is_visible_flag(self_obj) == 1)) then begin
         if ((local_var(5) bwand 4) == 0) then begin
             set_local_var(5, local_var(5) bwor 4);
         end
@@ -474,21 +474,21 @@ begin
         end
         add_timer_event(self_obj, game_ticks(3), 4);
     end
-    if ((fixed_param == 4) and (has_trait(1, self_obj, 666) == 1)) then begin
-        if (get_critter_stat(dude_obj, 34) == 0) then begin
+    if ((fixed_param == 4) and (obj_is_visible_flag(self_obj) == 1)) then begin
+        if (dude_gender == 0) then begin
             if (get_pc_base_stat(0) > 6) then begin
-                float_msg(self_obj, message_str(1646, random(314, 325)), 8);
+                float_msg(self_obj, mstr(random(314, 325)), FLOAT_MSG_YELLOW);
             end
             else begin
-                float_msg(self_obj, message_str(1646, random(315, 325)), 8);
+                float_msg(self_obj, mstr(random(315, 325)), FLOAT_MSG_YELLOW);
             end
         end
         else begin
-            if (get_critter_stat(dude_obj, 3) > 6) then begin
-                float_msg(self_obj, message_str(1646, random(318, 329)), 8);
+            if (dude_charisma > 6) then begin
+                float_msg(self_obj, mstr(random(318, 329)), FLOAT_MSG_YELLOW);
             end
             else begin
-                float_msg(self_obj, message_str(1646, random(318, 328)), 8);
+                float_msg(self_obj, mstr(random(318, 328)), FLOAT_MSG_YELLOW);
             end
         end
         add_timer_event(self_obj, game_ticks(7), 6);
@@ -500,7 +500,7 @@ begin
     if (fixed_param == 6) then begin
         set_local_var(5, local_var(5) bwand (-1 - 16));
     end
-    if ((fixed_param == 7) and (has_trait(1, self_obj, 666) == 1)) then begin
+    if ((fixed_param == 7) and (obj_is_visible_flag(self_obj) == 1)) then begin
         if ((local_var(5) bwand 4) == 0) then begin
             set_local_var(5, local_var(5) bwor 4);
         end
@@ -509,11 +509,11 @@ begin
         end
         add_timer_event(self_obj, game_ticks(3), 8);
     end
-    if ((fixed_param == 8) and (has_trait(1, self_obj, 666) == 1)) then begin
-        float_msg(self_obj, message_str(1646, random(318, 328)), 8);
+    if ((fixed_param == 8) and (obj_is_visible_flag(self_obj) == 1)) then begin
+        float_msg(self_obj, mstr(random(318, 328)), FLOAT_MSG_YELLOW);
         add_timer_event(self_obj, game_ticks(7), 6);
     end
-    if ((fixed_param == 9) and (has_trait(1, self_obj, 666) == 1)) then begin
+    if ((fixed_param == 9) and (obj_is_visible_flag(self_obj) == 1)) then begin
         if ((local_var(5) bwand 4) == 0) then begin
             set_local_var(5, local_var(5) bwor 4);
         end
@@ -522,9 +522,8 @@ begin
         end
         add_timer_event(self_obj, game_ticks(3), 10);
     end
-    if ((fixed_param == 10) and (has_trait(1, self_obj, 666) == 1)) then begin
-        float_msg(self_obj, message_str(1646, random(314, 325)), 8);
+    if ((fixed_param == 10) and (obj_is_visible_flag(self_obj) == 1)) then begin
+        float_msg(self_obj, mstr(random(314, 325)), FLOAT_MSG_YELLOW);
         add_timer_event(self_obj, game_ticks(7), 6);
     end
 end
-
