diff --git a/Mods/Resurrection/Original/ADRAJSER.ssl b/Mods/Resurrection/ReDefine/ADRAJSER.ssl
index e40a095..df5ae89 100644
--- a/Mods/Resurrection/Original/ADRAJSER.ssl
+++ b/Mods/Resurrection/ReDefine/ADRAJSER.ssl
@@ -54,12 +54,12 @@ begin
             return 1;
         end
     end
-    if ((party_member_obj(16777779) != 0) and ((global_var(910) bwand 32) == 0)) then begin
+    if ((party_member_obj(16777779) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 32) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777779)) <= 5) then begin
             return 1;
         end
     end
-    if ((party_member_obj(16777811) != 0) and ((global_var(910) bwand 64) == 0)) then begin
+    if ((party_member_obj(16777811) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 64) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777811)) <= 5) then begin
             return 1;
         end
@@ -74,7 +74,7 @@ begin
             return 1;
         end
     end
-    if ((party_member_obj(16777814) != 0) and ((global_var(910) bwand 128) == 0)) then begin
+    if ((party_member_obj(16777814) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 128) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777814)) <= 5) then begin
             return 1;
         end
@@ -136,10 +136,10 @@ begin
             if (anim_busy(self_obj) == 0) then begin
                 if ((tile_distance_objs(self_obj, dude_obj) < 8) or obj_can_see_obj(self_obj, dude_obj)) then begin
                     if (anim_busy(self_obj) == 0) then begin
-                        global_temp := rotation_to_tile(tile_num(dude_obj), tile_num(self_obj));
+                        global_temp := rotation_to_tile(dude_tile, tile_num(self_obj));
                         animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(self_obj), global_temp, random(3, 10)), 1);
                         global_temp := (global_temp + 1) % 6;
-                        while ((anim_busy(self_obj) == 0) and (global_temp != rotation_to_tile(tile_num(dude_obj), tile_num(self_obj)))) do begin
+                        while ((anim_busy(self_obj) == 0) and (global_temp != rotation_to_tile(dude_tile, tile_num(self_obj)))) do begin
                             debug_msg("flee loop: rot == " + global_temp);
                             animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(self_obj), global_temp, random(3, 10)), 1);
                             global_temp := (global_temp + 1) % 6;
@@ -149,7 +149,7 @@ begin
             end
         end
         else begin
-            attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+            attack(dude_obj);
         end
     end
     if (local_var(4) == 1) then begin
@@ -158,10 +158,10 @@ begin
             if (anim_busy(self_obj) == 0) then begin
                 if ((tile_distance_objs(self_obj, dude_obj) < 8) or obj_can_see_obj(self_obj, dude_obj)) then begin
                     if (anim_busy(self_obj) == 0) then begin
-                        global_temp := rotation_to_tile(tile_num(dude_obj), tile_num(self_obj));
+                        global_temp := rotation_to_tile(dude_tile, tile_num(self_obj));
                         animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(self_obj), global_temp, random(3, 10)), 1);
                         global_temp := (global_temp + 1) % 6;
-                        while ((anim_busy(self_obj) == 0) and (global_temp != rotation_to_tile(tile_num(dude_obj), tile_num(self_obj)))) do begin
+                        while ((anim_busy(self_obj) == 0) and (global_temp != rotation_to_tile(dude_tile, tile_num(self_obj)))) do begin
                             debug_msg("flee loop: rot == " + global_temp);
                             animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(self_obj), global_temp, random(3, 10)), 1);
                             global_temp := (global_temp + 1) % 6;
@@ -171,7 +171,7 @@ begin
             end
         end
         else begin
-            attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+            attack(dude_obj);
         end
     end
     if (tile_in_tile_rect(16664, 16654, 21464, 21454, tile_num(self_obj)) or tile_in_tile_rect(18286, 18280, 22286, 20280, tile_num(self_obj)) or tile_in_tile_rect(16680, 16664, 18080, 18064, tile_num(self_obj))) then begin
@@ -198,11 +198,11 @@ end
 procedure talk_p_proc
 begin
     if (party_member_obj(16777812) != 0) then begin
-        float_msg(self_obj, message_str(1616, random(300, 302)), 2);
+        float_msg(self_obj, mstr(random(300, 302)), FLOAT_MSG_RED);
     end
     else begin
-        anim(dude_obj, 1000, rotation_to_tile(tile_num(dude_obj), tile_num(self_obj)));
-        anim(self_obj, 1000, rotation_to_tile(tile_num(self_obj), tile_num(dude_obj)));
+        anim(dude_obj, 1000, rotation_to_tile(dude_tile, tile_num(self_obj)));
+        anim(self_obj, 1000, rotation_to_tile(tile_num(self_obj), dude_tile));
         start_gdialog(1616, self_obj, 4, -1, -1);
         gsay_start;
         call Node500();
@@ -214,58 +214,58 @@ end
 procedure destroy_p_proc
 begin
     if (source_obj == dude_obj) then begin
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(4) > (3 * global_var(5))) or (global_var(2) == 1)) and (global_var(1) == 0)) then begin
-            set_global_var(2, 1);
-            set_global_var(3, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(4) > (3 * global_var(GVAR_GOOD_MONSTER))) or (global_var(GVAR_CHAMPION_REPUTATION) == 1)) and (global_var(GVAR_CHILDKILLER_REPUTATION) == 0)) then begin
+            set_global_var(GVAR_CHAMPION_REPUTATION, 1);
+            set_global_var(GVAR_BERSERKER_REPUTATION, 0);
         end
-        if (metarule(51, self_obj) == 2) then begin
+        if (critter_kill_type(self_obj) == KILL_TYPE_children_kills) then begin
             if (source_obj == dude_obj) then begin
-                set_global_var(2, 0);
-                set_global_var(1, global_var(1) + 1);
-                set_global_var(0, global_var(0) + -15);
+                set_global_var(GVAR_CHAMPION_REPUTATION, 0);
+                set_global_var(GVAR_CHILDKILLER_REPUTATION, global_var(GVAR_CHILDKILLER_REPUTATION) + 1);
+                set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + -15);
                 debug_msg("Player gains " + -15 + " Karma Points.");
-                set_global_var(37, 0);
-                set_global_var(38, 0);
-                set_global_var(39, 0);
-                set_global_var(40, 0);
-                set_global_var(41, 0);
-                set_global_var(42, 0);
-                set_global_var(43, 0);
-                set_global_var(44, 0);
-                set_global_var(45, 0);
-                if (global_var(0) >= 1000) then begin
-                    set_global_var(37, 1);
+                set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+                set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+                set_global_var(GVAR_KARMA_DEFENDER, 0);
+                set_global_var(GVAR_KARMA_WANDERER, 0);
+                set_global_var(GVAR_KARMA_BETRAYER, 0);
+                set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+                set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+                set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+                    set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
                 end
                 else begin
-                    if (global_var(0) >= 750) then begin
-                        set_global_var(38, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
                     end
                     else begin
-                        if (global_var(0) >= 500) then begin
-                            set_global_var(39, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                            set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                         end
                         else begin
-                            if (global_var(0) >= 250) then begin
-                                set_global_var(40, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                                set_global_var(GVAR_KARMA_DEFENDER, 1);
                             end
                             else begin
-                                if (global_var(0) > -250) then begin
-                                    set_global_var(41, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                                    set_global_var(GVAR_KARMA_WANDERER, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -500) then begin
-                                        set_global_var(42, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                        set_global_var(GVAR_KARMA_BETRAYER, 1);
                                     end
                                     else begin
-                                        if (global_var(0) > -750) then begin
-                                            set_global_var(43, 1);
+                                        if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                            set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                         end
                                         else begin
-                                            if (global_var(0) > -1000) then begin
-                                                set_global_var(44, 1);
+                                            if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                                set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                             end
                                             else begin
-                                                set_global_var(45, 1);
+                                                set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                             end
                                         end
                                     end
@@ -274,58 +274,58 @@ begin
                         end
                     end
                 end
-                set_global_var(51, global_var(51) + -8);
+                set_global_var(GVAR_TOWN_REP_ALBUQUERQUE, global_var(GVAR_TOWN_REP_ALBUQUERQUE) + -8);
                 debug_msg("Added " + -8 + " to Town Rep");
             end
         end
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(5) > (2 * global_var(4))) or (global_var(3) == 1))) then begin
-            set_global_var(3, 1);
-            set_global_var(2, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
+            set_global_var(GVAR_BERSERKER_REPUTATION, 1);
+            set_global_var(GVAR_CHAMPION_REPUTATION, 0);
         end
-        set_global_var(0, global_var(0) + 0);
+        set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + 0);
         debug_msg("Player gains " + 0 + " Karma Points.");
-        set_global_var(37, 0);
-        set_global_var(38, 0);
-        set_global_var(39, 0);
-        set_global_var(40, 0);
-        set_global_var(41, 0);
-        set_global_var(42, 0);
-        set_global_var(43, 0);
-        set_global_var(44, 0);
-        set_global_var(45, 0);
-        if (global_var(0) >= 1000) then begin
-            set_global_var(37, 1);
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+        set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+        set_global_var(GVAR_KARMA_DEFENDER, 0);
+        set_global_var(GVAR_KARMA_WANDERER, 0);
+        set_global_var(GVAR_KARMA_BETRAYER, 0);
+        set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+        set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+            set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
         end
         else begin
-            if (global_var(0) >= 750) then begin
-                set_global_var(38, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
             end
             else begin
-                if (global_var(0) >= 500) then begin
-                    set_global_var(39, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                 end
                 else begin
-                    if (global_var(0) >= 250) then begin
-                        set_global_var(40, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                        set_global_var(GVAR_KARMA_DEFENDER, 1);
                     end
                     else begin
-                        if (global_var(0) > -250) then begin
-                            set_global_var(41, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                            set_global_var(GVAR_KARMA_WANDERER, 1);
                         end
                         else begin
-                            if (global_var(0) > -500) then begin
-                                set_global_var(42, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                set_global_var(GVAR_KARMA_BETRAYER, 1);
                             end
                             else begin
-                                if (global_var(0) > -750) then begin
-                                    set_global_var(43, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -1000) then begin
-                                        set_global_var(44, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                     end
                                     else begin
-                                        set_global_var(45, 1);
+                                        set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                     end
                                 end
                             end
@@ -340,13 +340,13 @@ end
 procedure look_at_p_proc
 begin
     script_overrides;
-    display_msg(message_str(1616, 100));
+    display_mstr(100);
 end
 
 procedure description_p_proc
 begin
     script_overrides;
-    display_msg(message_str(1616, 200));
+    display_mstr(200);
 end
 
 procedure use_skill_on_p_proc
@@ -363,22 +363,22 @@ end
 procedure map_enter_p_proc
 begin
     if ((local_var(6) bwand 1) == 0) then begin
-        critter_add_trait(self_obj, 1, 6, 260);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 260);
     end
     else begin
-        critter_add_trait(self_obj, 1, 6, 258);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 258);
     end
-    critter_add_trait(self_obj, 1, 5, 14);
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_PEASANTS);
     if (local_var(5) == 0) then begin
         set_local_var(5, tile_num(self_obj));
     end
-    if ((local_var(7) < game_time) and not(metarule(22, 0))) then begin
+    if ((local_var(7) < game_time) and not(is_loading_game)) then begin
         item_caps_adjust(self_obj, random(10, 20) - item_caps_total(self_obj));
         restock_amt := random(1, 2);
         if (obj_is_carrying_obj_pid(self_obj, 124) < restock_amt) then begin
             if (100 >= random(1, 100)) then begin
                 if (obj_is_carrying_obj_pid(self_obj, 124) < restock_amt) then begin
-                    restock_obj := create_object_sid(124, 0, 0, -1);
+                    restock_obj := create_object(124, 0, 0);
                     add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 124));
                 end
                 else begin
@@ -391,17 +391,17 @@ begin
                             restock_obj := obj_carrying_pid_obj(self_obj, 124);
                             if (obj_type(self_obj) == 1) then begin
                                 restock_obj := obj_carrying_pid_obj(self_obj, 124);
-                                if (critter_inven_obj(self_obj, 0) == restock_obj) then begin
-                                    if (obj_item_subtype(critter_inven_obj(self_obj, 0)) == 0) then begin
-                                        restock_obj := critter_inven_obj(self_obj, 0);
+                                if (critter_inven_obj(self_obj, INVEN_TYPE_WORN) == restock_obj) then begin
+                                    if (obj_item_subtype(critter_inven_obj(self_obj, INVEN_TYPE_WORN)) == 0) then begin
+                                        restock_obj := critter_inven_obj(self_obj, INVEN_TYPE_WORN);
                                         debug_msg("armour pid == " + obj_pid(restock_obj));
                                         rm_obj_from_inven(self_obj, restock_obj);
                                         add_obj_to_inven(self_obj, restock_obj);
                                     end
                                 end
                                 else begin
-                                    if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                        metarule(43, self_obj);
+                                    if ((critter_inven_obj(self_obj, INVEN_TYPE_LEFT_HAND) == restock_obj) or (critter_inven_obj(self_obj, INVEN_TYPE_RIGHT_HAND) == restock_obj)) then begin
+                                        inven_unwield(self_obj);
                                     end
                                 end
                             end
@@ -430,7 +430,7 @@ begin
         end
         else begin
             if (obj_is_carrying_obj_pid(self_obj, 124) < restock_amt) then begin
-                restock_obj := create_object_sid(124, 0, 0, -1);
+                restock_obj := create_object(124, 0, 0);
                 add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 124));
             end
             else begin
@@ -443,17 +443,17 @@ begin
                         restock_obj := obj_carrying_pid_obj(self_obj, 124);
                         if (obj_type(self_obj) == 1) then begin
                             restock_obj := obj_carrying_pid_obj(self_obj, 124);
-                            if (critter_inven_obj(self_obj, 0) == restock_obj) then begin
-                                if (obj_item_subtype(critter_inven_obj(self_obj, 0)) == 0) then begin
-                                    restock_obj := critter_inven_obj(self_obj, 0);
+                            if (critter_inven_obj(self_obj, INVEN_TYPE_WORN) == restock_obj) then begin
+                                if (obj_item_subtype(critter_inven_obj(self_obj, INVEN_TYPE_WORN)) == 0) then begin
+                                    restock_obj := critter_inven_obj(self_obj, INVEN_TYPE_WORN);
                                     debug_msg("armour pid == " + obj_pid(restock_obj));
                                     rm_obj_from_inven(self_obj, restock_obj);
                                     add_obj_to_inven(self_obj, restock_obj);
                                 end
                             end
                             else begin
-                                if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                    metarule(43, self_obj);
+                                if ((critter_inven_obj(self_obj, INVEN_TYPE_LEFT_HAND) == restock_obj) or (critter_inven_obj(self_obj, INVEN_TYPE_RIGHT_HAND) == restock_obj)) then begin
+                                    inven_unwield(self_obj);
                                 end
                             end
                         end
@@ -483,7 +483,7 @@ begin
         if (obj_is_carrying_obj_pid(self_obj, 125) < restock_amt) then begin
             if (95 >= random(1, 100)) then begin
                 if (obj_is_carrying_obj_pid(self_obj, 125) < restock_amt) then begin
-                    restock_obj := create_object_sid(125, 0, 0, -1);
+                    restock_obj := create_object(125, 0, 0);
                     add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 125));
                 end
                 else begin
@@ -496,17 +496,17 @@ begin
                             restock_obj := obj_carrying_pid_obj(self_obj, 125);
                             if (obj_type(self_obj) == 1) then begin
                                 restock_obj := obj_carrying_pid_obj(self_obj, 125);
-                                if (critter_inven_obj(self_obj, 0) == restock_obj) then begin
-                                    if (obj_item_subtype(critter_inven_obj(self_obj, 0)) == 0) then begin
-                                        restock_obj := critter_inven_obj(self_obj, 0);
+                                if (critter_inven_obj(self_obj, INVEN_TYPE_WORN) == restock_obj) then begin
+                                    if (obj_item_subtype(critter_inven_obj(self_obj, INVEN_TYPE_WORN)) == 0) then begin
+                                        restock_obj := critter_inven_obj(self_obj, INVEN_TYPE_WORN);
                                         debug_msg("armour pid == " + obj_pid(restock_obj));
                                         rm_obj_from_inven(self_obj, restock_obj);
                                         add_obj_to_inven(self_obj, restock_obj);
                                     end
                                 end
                                 else begin
-                                    if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                        metarule(43, self_obj);
+                                    if ((critter_inven_obj(self_obj, INVEN_TYPE_LEFT_HAND) == restock_obj) or (critter_inven_obj(self_obj, INVEN_TYPE_RIGHT_HAND) == restock_obj)) then begin
+                                        inven_unwield(self_obj);
                                     end
                                 end
                             end
@@ -535,7 +535,7 @@ begin
         end
         else begin
             if (obj_is_carrying_obj_pid(self_obj, 125) < restock_amt) then begin
-                restock_obj := create_object_sid(125, 0, 0, -1);
+                restock_obj := create_object(125, 0, 0);
                 add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 125));
             end
             else begin
@@ -548,17 +548,17 @@ begin
                         restock_obj := obj_carrying_pid_obj(self_obj, 125);
                         if (obj_type(self_obj) == 1) then begin
                             restock_obj := obj_carrying_pid_obj(self_obj, 125);
-                            if (critter_inven_obj(self_obj, 0) == restock_obj) then begin
-                                if (obj_item_subtype(critter_inven_obj(self_obj, 0)) == 0) then begin
-                                    restock_obj := critter_inven_obj(self_obj, 0);
+                            if (critter_inven_obj(self_obj, INVEN_TYPE_WORN) == restock_obj) then begin
+                                if (obj_item_subtype(critter_inven_obj(self_obj, INVEN_TYPE_WORN)) == 0) then begin
+                                    restock_obj := critter_inven_obj(self_obj, INVEN_TYPE_WORN);
                                     debug_msg("armour pid == " + obj_pid(restock_obj));
                                     rm_obj_from_inven(self_obj, restock_obj);
                                     add_obj_to_inven(self_obj, restock_obj);
                                 end
                             end
                             else begin
-                                if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                    metarule(43, self_obj);
+                                if ((critter_inven_obj(self_obj, INVEN_TYPE_LEFT_HAND) == restock_obj) or (critter_inven_obj(self_obj, INVEN_TYPE_RIGHT_HAND) == restock_obj)) then begin
+                                    inven_unwield(self_obj);
                                 end
                             end
                         end
@@ -588,7 +588,7 @@ begin
         if (obj_is_carrying_obj_pid(self_obj, 106) < restock_amt) then begin
             if (100 >= random(1, 100)) then begin
                 if (obj_is_carrying_obj_pid(self_obj, 106) < restock_amt) then begin
-                    restock_obj := create_object_sid(106, 0, 0, -1);
+                    restock_obj := create_object(106, 0, 0);
                     add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 106));
                 end
                 else begin
@@ -601,17 +601,17 @@ begin
                             restock_obj := obj_carrying_pid_obj(self_obj, 106);
                             if (obj_type(self_obj) == 1) then begin
                                 restock_obj := obj_carrying_pid_obj(self_obj, 106);
-                                if (critter_inven_obj(self_obj, 0) == restock_obj) then begin
-                                    if (obj_item_subtype(critter_inven_obj(self_obj, 0)) == 0) then begin
-                                        restock_obj := critter_inven_obj(self_obj, 0);
+                                if (critter_inven_obj(self_obj, INVEN_TYPE_WORN) == restock_obj) then begin
+                                    if (obj_item_subtype(critter_inven_obj(self_obj, INVEN_TYPE_WORN)) == 0) then begin
+                                        restock_obj := critter_inven_obj(self_obj, INVEN_TYPE_WORN);
                                         debug_msg("armour pid == " + obj_pid(restock_obj));
                                         rm_obj_from_inven(self_obj, restock_obj);
                                         add_obj_to_inven(self_obj, restock_obj);
                                     end
                                 end
                                 else begin
-                                    if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                        metarule(43, self_obj);
+                                    if ((critter_inven_obj(self_obj, INVEN_TYPE_LEFT_HAND) == restock_obj) or (critter_inven_obj(self_obj, INVEN_TYPE_RIGHT_HAND) == restock_obj)) then begin
+                                        inven_unwield(self_obj);
                                     end
                                 end
                             end
@@ -640,7 +640,7 @@ begin
         end
         else begin
             if (obj_is_carrying_obj_pid(self_obj, 106) < restock_amt) then begin
-                restock_obj := create_object_sid(106, 0, 0, -1);
+                restock_obj := create_object(106, 0, 0);
                 add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 106));
             end
             else begin
@@ -653,17 +653,17 @@ begin
                         restock_obj := obj_carrying_pid_obj(self_obj, 106);
                         if (obj_type(self_obj) == 1) then begin
                             restock_obj := obj_carrying_pid_obj(self_obj, 106);
-                            if (critter_inven_obj(self_obj, 0) == restock_obj) then begin
-                                if (obj_item_subtype(critter_inven_obj(self_obj, 0)) == 0) then begin
-                                    restock_obj := critter_inven_obj(self_obj, 0);
+                            if (critter_inven_obj(self_obj, INVEN_TYPE_WORN) == restock_obj) then begin
+                                if (obj_item_subtype(critter_inven_obj(self_obj, INVEN_TYPE_WORN)) == 0) then begin
+                                    restock_obj := critter_inven_obj(self_obj, INVEN_TYPE_WORN);
                                     debug_msg("armour pid == " + obj_pid(restock_obj));
                                     rm_obj_from_inven(self_obj, restock_obj);
                                     add_obj_to_inven(self_obj, restock_obj);
                                 end
                             end
                             else begin
-                                if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                    metarule(43, self_obj);
+                                if ((critter_inven_obj(self_obj, INVEN_TYPE_LEFT_HAND) == restock_obj) or (critter_inven_obj(self_obj, INVEN_TYPE_RIGHT_HAND) == restock_obj)) then begin
+                                    inven_unwield(self_obj);
                                 end
                             end
                         end
@@ -693,7 +693,7 @@ begin
         if (obj_is_carrying_obj_pid(self_obj, 311) < restock_amt) then begin
             if (50 >= random(1, 100)) then begin
                 if (obj_is_carrying_obj_pid(self_obj, 311) < restock_amt) then begin
-                    restock_obj := create_object_sid(311, 0, 0, -1);
+                    restock_obj := create_object(311, 0, 0);
                     add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 311));
                 end
                 else begin
@@ -706,17 +706,17 @@ begin
                             restock_obj := obj_carrying_pid_obj(self_obj, 311);
                             if (obj_type(self_obj) == 1) then begin
                                 restock_obj := obj_carrying_pid_obj(self_obj, 311);
-                                if (critter_inven_obj(self_obj, 0) == restock_obj) then begin
-                                    if (obj_item_subtype(critter_inven_obj(self_obj, 0)) == 0) then begin
-                                        restock_obj := critter_inven_obj(self_obj, 0);
+                                if (critter_inven_obj(self_obj, INVEN_TYPE_WORN) == restock_obj) then begin
+                                    if (obj_item_subtype(critter_inven_obj(self_obj, INVEN_TYPE_WORN)) == 0) then begin
+                                        restock_obj := critter_inven_obj(self_obj, INVEN_TYPE_WORN);
                                         debug_msg("armour pid == " + obj_pid(restock_obj));
                                         rm_obj_from_inven(self_obj, restock_obj);
                                         add_obj_to_inven(self_obj, restock_obj);
                                     end
                                 end
                                 else begin
-                                    if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                        metarule(43, self_obj);
+                                    if ((critter_inven_obj(self_obj, INVEN_TYPE_LEFT_HAND) == restock_obj) or (critter_inven_obj(self_obj, INVEN_TYPE_RIGHT_HAND) == restock_obj)) then begin
+                                        inven_unwield(self_obj);
                                     end
                                 end
                             end
@@ -745,7 +745,7 @@ begin
         end
         else begin
             if (obj_is_carrying_obj_pid(self_obj, 311) < restock_amt) then begin
-                restock_obj := create_object_sid(311, 0, 0, -1);
+                restock_obj := create_object(311, 0, 0);
                 add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 311));
             end
             else begin
@@ -758,17 +758,17 @@ begin
                         restock_obj := obj_carrying_pid_obj(self_obj, 311);
                         if (obj_type(self_obj) == 1) then begin
                             restock_obj := obj_carrying_pid_obj(self_obj, 311);
-                            if (critter_inven_obj(self_obj, 0) == restock_obj) then begin
-                                if (obj_item_subtype(critter_inven_obj(self_obj, 0)) == 0) then begin
-                                    restock_obj := critter_inven_obj(self_obj, 0);
+                            if (critter_inven_obj(self_obj, INVEN_TYPE_WORN) == restock_obj) then begin
+                                if (obj_item_subtype(critter_inven_obj(self_obj, INVEN_TYPE_WORN)) == 0) then begin
+                                    restock_obj := critter_inven_obj(self_obj, INVEN_TYPE_WORN);
                                     debug_msg("armour pid == " + obj_pid(restock_obj));
                                     rm_obj_from_inven(self_obj, restock_obj);
                                     add_obj_to_inven(self_obj, restock_obj);
                                 end
                             end
                             else begin
-                                if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                    metarule(43, self_obj);
+                                if ((critter_inven_obj(self_obj, INVEN_TYPE_LEFT_HAND) == restock_obj) or (critter_inven_obj(self_obj, INVEN_TYPE_RIGHT_HAND) == restock_obj)) then begin
+                                    inven_unwield(self_obj);
                                 end
                             end
                         end
@@ -802,14 +802,14 @@ end
 
 procedure map_exit_p_proc
 begin
-    float_msg(self_obj, 0, 0);
+    float_msg_clear(self_obj);
 end
 
 procedure combat_p_proc
 begin
     if ((map_var(1) == 1) and ((local_var(6) bwand 1) == 0)) then begin
-        critter_add_trait(self_obj, 1, 6, 258);
-        set_global_var(864, 1);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 258);
+        set_global_var(GVAR_AQ_ENEMY_HUNTERS, 1);
         set_local_var(6, local_var(6) bwor 1);
     end
 end
@@ -818,7 +818,7 @@ procedure timed_event_p_proc
 begin
     if (fixed_param == 1) then begin
         if (combat_is_initialized == 0) then begin
-            float_msg(self_obj, message_str(1616, random(303, 311)), 8);
+            float_msg(self_obj, mstr(random(303, 311)), FLOAT_MSG_YELLOW);
         end
         add_timer_event(self_obj, game_ticks(random(20, 40)), 1);
     end
@@ -846,25 +846,25 @@ end
 
 procedure Node500
 begin
-    gsay_reply(1616, 500);
-    giq_option(4, 1616, 501, Barter, 50);
-    giq_option(4, 1616, 502, Node999, 50);
-    giq_option(-3, 1616, 503, Barter, 50);
-    giq_option(-3, 1616, 504, Node999, 50);
+    Reply(500);
+    giq_option(4, NAME, 501, Barter, NEUTRAL_REACTION);
+    giq_option(4, NAME, 502, Node999, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 503, Barter, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 504, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node550
 begin
-    gsay_reply(1616, 550);
-    giq_option(4, 1616, 551, Barter, 50);
-    giq_option(4, 1616, 552, Node999, 50);
-    giq_option(-3, 1616, 553, Barter, 50);
-    giq_option(-3, 1616, 554, Node999, 50);
+    Reply(550);
+    giq_option(4, NAME, 551, Barter, NEUTRAL_REACTION);
+    giq_option(4, NAME, 552, Node999, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 553, Barter, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 554, Node999, NEUTRAL_REACTION);
 end
 
 procedure Barter
 begin
-    if ((party_member_obj(16777779) != 0) and ((global_var(910) bwand 32) == 0)) then begin
+    if ((party_member_obj(16777779) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 32) == 0)) then begin
         gdialog_mod_barter(30 - 15);
     end
     else begin
@@ -872,4 +872,3 @@ begin
     end
     call Node550();
 end
-
