diff --git a/Mods/Resurrection/Original/AFORFIEL.ssl b/Mods/Resurrection/ReDefine/AFORFIEL.ssl
index cd2b37f..3a0fdbc 100644
--- a/Mods/Resurrection/Original/AFORFIEL.ssl
+++ b/Mods/Resurrection/ReDefine/AFORFIEL.ssl
@@ -44,12 +44,12 @@ begin
             return 1;
         end
     end
-    if ((party_member_obj(16777779) != 0) and ((global_var(910) bwand 32) == 0)) then begin
+    if ((party_member_obj(16777779) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 32) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777779)) <= 5) then begin
             return 1;
         end
     end
-    if ((party_member_obj(16777811) != 0) and ((global_var(910) bwand 64) == 0)) then begin
+    if ((party_member_obj(16777811) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 64) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777811)) <= 5) then begin
             return 1;
         end
@@ -64,7 +64,7 @@ begin
             return 1;
         end
     end
-    if ((party_member_obj(16777814) != 0) and ((global_var(910) bwand 128) == 0)) then begin
+    if ((party_member_obj(16777814) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 128) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777814)) <= 5) then begin
             return 1;
         end
@@ -121,17 +121,17 @@ end
 
 procedure timed_event_p_proc
 begin
-    if ((fixed_param == 1) and ((global_var(998) == 1) or ((map_var(0) bwand 2048) != 0)) and (local_var(4) == 1)) then begin
+    if ((fixed_param == 1) and ((global_var(GVAR_AQE_ENEMY_EMPEROR) == 1) or ((map_var(0) bwand 2048) != 0)) and (local_var(4) == 1)) then begin
         call Unplug_Forcefield();
     end
-    if ((fixed_param == 2) and (global_var(998) == 0)) then begin
+    if ((fixed_param == 2) and (global_var(GVAR_AQE_ENEMY_EMPEROR) == 0)) then begin
         call Plug_Forcefield();
     end
 end
 
 procedure map_enter_p_proc
 begin
-    if (metarule(14, 0)) then begin
+    if (map_first_run) then begin
         set_local_var(4, 1);
     end
     if (local_var(5) == 0) then begin
@@ -162,7 +162,7 @@ end
 
 procedure Unplug_Forcefield
 begin
-    if ((global_var(998) == 1) or ((map_var(0) bwand 2048) != 0) and (local_var(4) == 1)) then begin
+    if ((global_var(GVAR_AQE_ENEMY_EMPEROR) == 1) or ((map_var(0) bwand 2048) != 0) and (local_var(4) == 1)) then begin
         if (local_var(5) == 15092) then begin
             set_obj_visibility(self_obj, 1);
         end
@@ -197,4 +197,3 @@ begin
         set_local_var(4, 1);
     end
 end
-
