diff --git a/Mods/Resurrection/Original/AGASCHAM.ssl b/Mods/Resurrection/ReDefine/AGASCHAM.ssl
index 0a93671..bc20ac4 100644
--- a/Mods/Resurrection/Original/AGASCHAM.ssl
+++ b/Mods/Resurrection/ReDefine/AGASCHAM.ssl
@@ -140,12 +140,12 @@ begin
             return 1;
         end
     end
-    if ((party_member_obj(16777779) != 0) and ((global_var(910) bwand 32) == 0)) then begin
+    if ((party_member_obj(16777779) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 32) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777779)) <= 5) then begin
             return 1;
         end
     end
-    if ((party_member_obj(16777811) != 0) and ((global_var(910) bwand 64) == 0)) then begin
+    if ((party_member_obj(16777811) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 64) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777811)) <= 5) then begin
             return 1;
         end
@@ -160,7 +160,7 @@ begin
             return 1;
         end
     end
-    if ((party_member_obj(16777814) != 0) and ((global_var(910) bwand 128) == 0)) then begin
+    if ((party_member_obj(16777814) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 128) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777814)) <= 5) then begin
             return 1;
         end
@@ -231,109 +231,109 @@ begin
     if (fixed_param == 1) then begin
         call read_tile_content();
         if (GVar20 != 0) then begin
-            critter_dmg(GVar20, 999, 0);
+            critter_damage(GVar20, 999);
         end
         if (GVar21 != 0) then begin
-            critter_dmg(GVar21, 999, 0);
+            critter_damage(GVar21, 999);
         end
         if (GVar22 != 0) then begin
-            critter_dmg(GVar22, 999, 0);
+            critter_damage(GVar22, 999);
         end
         if (GVar23 != 0) then begin
-            critter_dmg(GVar23, 999, 0);
+            critter_damage(GVar23, 999);
         end
         if (GVar24 != 0) then begin
-            critter_dmg(GVar24, 999, 0);
+            critter_damage(GVar24, 999);
         end
         if (GVar25 != 0) then begin
-            critter_dmg(GVar25, 999, 0);
+            critter_damage(GVar25, 999);
         end
         if (GVar26 != 0) then begin
-            critter_dmg(GVar26, 999, 0);
+            critter_damage(GVar26, 999);
         end
         if (GVar27 != 0) then begin
-            critter_dmg(GVar27, 999, 0);
+            critter_damage(GVar27, 999);
         end
         if (GVar28 != 0) then begin
-            critter_dmg(GVar28, 999, 0);
+            critter_damage(GVar28, 999);
         end
         if (GVar29 != 0) then begin
-            critter_dmg(GVar29, 999, 0);
+            critter_damage(GVar29, 999);
         end
         if (GVar30 != 0) then begin
-            critter_dmg(GVar30, 999, 0);
+            critter_damage(GVar30, 999);
         end
         if (GVar31 != 0) then begin
-            critter_dmg(GVar31, 999, 0);
+            critter_damage(GVar31, 999);
         end
         if (GVar32 != 0) then begin
-            critter_dmg(GVar32, 999, 0);
+            critter_damage(GVar32, 999);
         end
         if (GVar33 != 0) then begin
-            critter_dmg(GVar33, 999, 0);
+            critter_damage(GVar33, 999);
         end
         if (GVar34 != 0) then begin
-            critter_dmg(GVar34, 999, 0);
+            critter_damage(GVar34, 999);
         end
         if (GVar35 != 0) then begin
-            critter_dmg(GVar35, 999, 0);
+            critter_damage(GVar35, 999);
         end
         if (GVar36 != 0) then begin
-            critter_dmg(GVar36, 999, 0);
+            critter_damage(GVar36, 999);
         end
         if (GVar37 != 0) then begin
-            critter_dmg(GVar37, 999, 0);
+            critter_damage(GVar37, 999);
         end
         if (GVar38 != 0) then begin
-            critter_dmg(GVar38, 999, 0);
+            critter_damage(GVar38, 999);
         end
         if (GVar39 != 0) then begin
-            critter_dmg(GVar39, 999, 0);
+            critter_damage(GVar39, 999);
         end
         if (GVar40 != 0) then begin
-            critter_dmg(GVar40, 999, 0);
+            critter_damage(GVar40, 999);
         end
         if (GVar41 != 0) then begin
-            critter_dmg(GVar41, 999, 0);
+            critter_damage(GVar41, 999);
         end
         if (GVar42 != 0) then begin
-            critter_dmg(GVar42, 999, 0);
+            critter_damage(GVar42, 999);
         end
         if (GVar43 != 0) then begin
-            critter_dmg(GVar43, 999, 0);
+            critter_damage(GVar43, 999);
         end
         if (GVar44 != 0) then begin
-            critter_dmg(GVar44, 999, 0);
+            critter_damage(GVar44, 999);
         end
         if (GVar45 != 0) then begin
-            critter_dmg(GVar45, 999, 0);
+            critter_damage(GVar45, 999);
         end
         if (GVar46 != 0) then begin
-            critter_dmg(GVar46, 999, 0);
+            critter_damage(GVar46, 999);
         end
         if (GVar47 != 0) then begin
-            critter_dmg(GVar47, 999, 0);
+            critter_damage(GVar47, 999);
         end
         if (GVar48 != 0) then begin
-            critter_dmg(GVar48, 999, 0);
+            critter_damage(GVar48, 999);
         end
         if (GVar49 != 0) then begin
-            critter_dmg(GVar39, 999, 0);
+            critter_damage(GVar39, 999);
         end
         if (GVar50 != 0) then begin
-            critter_dmg(GVar50, 999, 0);
+            critter_damage(GVar50, 999);
         end
         if (GVar51 != 0) then begin
-            critter_dmg(GVar51, 999, 0);
+            critter_damage(GVar51, 999);
         end
         if (GVar52 != 0) then begin
-            critter_dmg(GVar52, 999, 0);
+            critter_damage(GVar52, 999);
         end
         if (GVar53 != 0) then begin
-            critter_dmg(GVar53, 999, 0);
+            critter_damage(GVar53, 999);
         end
         if (GVar54 != 0) then begin
-            critter_dmg(GVar54, 999, 0);
+            critter_damage(GVar54, 999);
         end
     end
     if (fixed_param == 2) then begin
@@ -346,109 +346,109 @@ begin
     if ((fixed_param == 4) and (local_var(4) == 1)) then begin
         call read_tile_content();
         if (GVar20 != 0) then begin
-            critter_dmg(GVar20, 5, 0);
+            critter_damage(GVar20, 5);
         end
         if (GVar21 != 0) then begin
-            critter_dmg(GVar21, 5, 0);
+            critter_damage(GVar21, 5);
         end
         if (GVar22 != 0) then begin
-            critter_dmg(GVar22, 5, 0);
+            critter_damage(GVar22, 5);
         end
         if (GVar23 != 0) then begin
-            critter_dmg(GVar23, 5, 0);
+            critter_damage(GVar23, 5);
         end
         if (GVar24 != 0) then begin
-            critter_dmg(GVar24, 5, 0);
+            critter_damage(GVar24, 5);
         end
         if (GVar25 != 0) then begin
-            critter_dmg(GVar25, 5, 0);
+            critter_damage(GVar25, 5);
         end
         if (GVar26 != 0) then begin
-            critter_dmg(GVar26, 5, 0);
+            critter_damage(GVar26, 5);
         end
         if (GVar27 != 0) then begin
-            critter_dmg(GVar27, 5, 0);
+            critter_damage(GVar27, 5);
         end
         if (GVar28 != 0) then begin
-            critter_dmg(GVar28, 5, 0);
+            critter_damage(GVar28, 5);
         end
         if (GVar29 != 0) then begin
-            critter_dmg(GVar29, 5, 0);
+            critter_damage(GVar29, 5);
         end
         if (GVar30 != 0) then begin
-            critter_dmg(GVar30, 5, 0);
+            critter_damage(GVar30, 5);
         end
         if (GVar31 != 0) then begin
-            critter_dmg(GVar31, 5, 0);
+            critter_damage(GVar31, 5);
         end
         if (GVar32 != 0) then begin
-            critter_dmg(GVar32, 5, 0);
+            critter_damage(GVar32, 5);
         end
         if (GVar33 != 0) then begin
-            critter_dmg(GVar33, 5, 0);
+            critter_damage(GVar33, 5);
         end
         if (GVar34 != 0) then begin
-            critter_dmg(GVar34, 5, 0);
+            critter_damage(GVar34, 5);
         end
         if (GVar35 != 0) then begin
-            critter_dmg(GVar35, 5, 0);
+            critter_damage(GVar35, 5);
         end
         if (GVar36 != 0) then begin
-            critter_dmg(GVar36, 5, 0);
+            critter_damage(GVar36, 5);
         end
         if (GVar37 != 0) then begin
-            critter_dmg(GVar37, 5, 0);
+            critter_damage(GVar37, 5);
         end
         if (GVar38 != 0) then begin
-            critter_dmg(GVar38, 5, 0);
+            critter_damage(GVar38, 5);
         end
         if (GVar39 != 0) then begin
-            critter_dmg(GVar39, 5, 0);
+            critter_damage(GVar39, 5);
         end
         if (GVar40 != 0) then begin
-            critter_dmg(GVar40, 5, 0);
+            critter_damage(GVar40, 5);
         end
         if (GVar41 != 0) then begin
-            critter_dmg(GVar41, 5, 0);
+            critter_damage(GVar41, 5);
         end
         if (GVar42 != 0) then begin
-            critter_dmg(GVar42, 5, 0);
+            critter_damage(GVar42, 5);
         end
         if (GVar43 != 0) then begin
-            critter_dmg(GVar43, 5, 0);
+            critter_damage(GVar43, 5);
         end
         if (GVar44 != 0) then begin
-            critter_dmg(GVar44, 5, 0);
+            critter_damage(GVar44, 5);
         end
         if (GVar45 != 0) then begin
-            critter_dmg(GVar45, 5, 0);
+            critter_damage(GVar45, 5);
         end
         if (GVar46 != 0) then begin
-            critter_dmg(GVar46, 5, 0);
+            critter_damage(GVar46, 5);
         end
         if (GVar47 != 0) then begin
-            critter_dmg(GVar47, 5, 0);
+            critter_damage(GVar47, 5);
         end
         if (GVar48 != 0) then begin
-            critter_dmg(GVar48, 5, 0);
+            critter_damage(GVar48, 5);
         end
         if (GVar49 != 0) then begin
-            critter_dmg(GVar39, 5, 0);
+            critter_damage(GVar39, 5);
         end
         if (GVar50 != 0) then begin
-            critter_dmg(GVar50, 5, 0);
+            critter_damage(GVar50, 5);
         end
         if (GVar51 != 0) then begin
-            critter_dmg(GVar51, 5, 0);
+            critter_damage(GVar51, 5);
         end
         if (GVar52 != 0) then begin
-            critter_dmg(GVar52, 5, 0);
+            critter_damage(GVar52, 5);
         end
         if (GVar53 != 0) then begin
-            critter_dmg(GVar53, 5, 0);
+            critter_damage(GVar53, 5);
         end
         if (GVar54 != 0) then begin
-            critter_dmg(GVar54, 5, 0);
+            critter_damage(GVar54, 5);
         end
         add_timer_event(self_obj, game_ticks(3), 4);
     end
@@ -456,40 +456,39 @@ end
 
 procedure read_tile_content
 begin
-    GVar20 := metarule3(106, 20509, 0, 0);
-    GVar21 := metarule3(106, 20709, 0, 0);
-    GVar22 := metarule3(106, 20909, 0, 0);
-    GVar23 := metarule3(106, 21109, 0, 0);
-    GVar24 := metarule3(106, 21309, 0, 0);
-    GVar25 := metarule3(106, 20508, 0, 0);
-    GVar26 := metarule3(106, 20708, 0, 0);
-    GVar27 := metarule3(106, 20908, 0, 0);
-    GVar28 := metarule3(106, 21108, 0, 0);
-    GVar29 := metarule3(106, 21307, 0, 0);
-    GVar30 := metarule3(106, 20507, 0, 0);
-    GVar31 := metarule3(106, 20707, 0, 0);
-    GVar32 := metarule3(106, 20907, 0, 0);
-    GVar33 := metarule3(106, 21107, 0, 0);
-    GVar34 := metarule3(106, 21307, 0, 0);
-    GVar35 := metarule3(106, 20506, 0, 0);
-    GVar36 := metarule3(106, 20706, 0, 0);
-    GVar37 := metarule3(106, 20906, 0, 0);
-    GVar38 := metarule3(106, 21106, 0, 0);
-    GVar39 := metarule3(106, 21306, 0, 0);
-    GVar40 := metarule3(106, 20505, 0, 0);
-    GVar41 := metarule3(106, 20705, 0, 0);
-    GVar42 := metarule3(106, 20905, 0, 0);
-    GVar43 := metarule3(106, 21105, 0, 0);
-    GVar44 := metarule3(106, 21305, 0, 0);
-    GVar45 := metarule3(106, 20504, 0, 0);
-    GVar46 := metarule3(106, 20704, 0, 0);
-    GVar47 := metarule3(106, 20904, 0, 0);
-    GVar48 := metarule3(106, 21104, 0, 0);
-    GVar49 := metarule3(106, 21304, 0, 0);
-    GVar50 := metarule3(106, 20503, 0, 0);
-    GVar51 := metarule3(106, 20703, 0, 0);
-    GVar52 := metarule3(106, 20903, 0, 0);
-    GVar53 := metarule3(106, 21103, 0, 0);
-    GVar54 := metarule3(106, 21303, 0, 0);
+    GVar20 := tile_get_next_critter(20509, 0, 0);
+    GVar21 := tile_get_next_critter(20709, 0, 0);
+    GVar22 := tile_get_next_critter(20909, 0, 0);
+    GVar23 := tile_get_next_critter(21109, 0, 0);
+    GVar24 := tile_get_next_critter(21309, 0, 0);
+    GVar25 := tile_get_next_critter(20508, 0, 0);
+    GVar26 := tile_get_next_critter(20708, 0, 0);
+    GVar27 := tile_get_next_critter(20908, 0, 0);
+    GVar28 := tile_get_next_critter(21108, 0, 0);
+    GVar29 := tile_get_next_critter(21307, 0, 0);
+    GVar30 := tile_get_next_critter(20507, 0, 0);
+    GVar31 := tile_get_next_critter(20707, 0, 0);
+    GVar32 := tile_get_next_critter(20907, 0, 0);
+    GVar33 := tile_get_next_critter(21107, 0, 0);
+    GVar34 := tile_get_next_critter(21307, 0, 0);
+    GVar35 := tile_get_next_critter(20506, 0, 0);
+    GVar36 := tile_get_next_critter(20706, 0, 0);
+    GVar37 := tile_get_next_critter(20906, 0, 0);
+    GVar38 := tile_get_next_critter(21106, 0, 0);
+    GVar39 := tile_get_next_critter(21306, 0, 0);
+    GVar40 := tile_get_next_critter(20505, 0, 0);
+    GVar41 := tile_get_next_critter(20705, 0, 0);
+    GVar42 := tile_get_next_critter(20905, 0, 0);
+    GVar43 := tile_get_next_critter(21105, 0, 0);
+    GVar44 := tile_get_next_critter(21305, 0, 0);
+    GVar45 := tile_get_next_critter(20504, 0, 0);
+    GVar46 := tile_get_next_critter(20704, 0, 0);
+    GVar47 := tile_get_next_critter(20904, 0, 0);
+    GVar48 := tile_get_next_critter(21104, 0, 0);
+    GVar49 := tile_get_next_critter(21304, 0, 0);
+    GVar50 := tile_get_next_critter(20503, 0, 0);
+    GVar51 := tile_get_next_critter(20703, 0, 0);
+    GVar52 := tile_get_next_critter(20903, 0, 0);
+    GVar53 := tile_get_next_critter(21103, 0, 0);
+    GVar54 := tile_get_next_critter(21303, 0, 0);
 end
-
