diff --git a/Mods/Resurrection/Original/AGGECKON.ssl b/Mods/Resurrection/ReDefine/AGGECKON.ssl
index 17206e1..19bf308 100644
--- a/Mods/Resurrection/Original/AGGECKON.ssl
+++ b/Mods/Resurrection/ReDefine/AGGECKON.ssl
@@ -51,12 +51,12 @@ begin
             return 1;
         end
     end
-    if ((party_member_obj(16777779) != 0) and ((global_var(910) bwand 32) == 0)) then begin
+    if ((party_member_obj(16777779) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 32) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777779)) <= 5) then begin
             return 1;
         end
     end
-    if ((party_member_obj(16777811) != 0) and ((global_var(910) bwand 64) == 0)) then begin
+    if ((party_member_obj(16777811) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 64) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777811)) <= 5) then begin
             return 1;
         end
@@ -71,7 +71,7 @@ begin
             return 1;
         end
     end
-    if ((party_member_obj(16777814) != 0) and ((global_var(910) bwand 128) == 0)) then begin
+    if ((party_member_obj(16777814) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 128) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777814)) <= 5) then begin
             return 1;
         end
@@ -130,149 +130,149 @@ procedure critter_p_proc
 begin
     if (not(combat_is_initialized)) then begin
         if (obj_can_see_obj(self_obj, dude_obj)) then begin
-            attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+            attack(dude_obj);
         end
-        if (((global_var(900) bwand 4194304) == 0) and obj_can_see_obj(self_obj, lovec1_obj)) then begin
-            attack_complex(lovec1_obj, 0, 1, 0, 0, 30000, 0, 0);
+        if (((global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS) bwand 4194304) == 0) and obj_can_see_obj(self_obj, lovec1_obj)) then begin
+            attack(lovec1_obj);
         end
-        if (((global_var(900) bwand 8388608) == 0) and obj_can_see_obj(self_obj, lovec2_obj)) then begin
-            attack_complex(lovec2_obj, 0, 1, 0, 0, 30000, 0, 0);
+        if (((global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS) bwand 8388608) == 0) and obj_can_see_obj(self_obj, lovec2_obj)) then begin
+            attack(lovec2_obj);
         end
-        if (((global_var(900) bwand 16777216) == 0) and obj_can_see_obj(self_obj, lovec3_obj)) then begin
-            attack_complex(lovec3_obj, 0, 1, 0, 0, 30000, 0, 0);
+        if (((global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS) bwand 16777216) == 0) and obj_can_see_obj(self_obj, lovec3_obj)) then begin
+            attack(lovec3_obj);
         end
-        if (cur_map_index == 92) then begin
-            if ((global_var(900) bwand 16) == 0) then begin
-                if ((local_var(4) == 19093) and ((global_var(900) bwand 4194304) == 0)) then begin
+        if (CUR_MAP_ALB_DIV1) then begin
+            if ((global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS) bwand 16) == 0) then begin
+                if ((local_var(4) == 19093) and ((global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS) bwand 4194304) == 0)) then begin
                     animate_move_obj_to_tile(lovec1_obj, tile_num(self_obj), 1);
                 end
-                if ((local_var(4) == 19093) and ((global_var(900) bwand 8388608) == 0)) then begin
+                if ((local_var(4) == 19093) and ((global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS) bwand 8388608) == 0)) then begin
                     animate_move_obj_to_tile(lovec2_obj, tile_num(self_obj), 1);
                 end
-                if ((local_var(4) == 19093) and ((global_var(900) bwand 16777216) == 0)) then begin
+                if ((local_var(4) == 19093) and ((global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS) bwand 16777216) == 0)) then begin
                     animate_move_obj_to_tile(lovec3_obj, tile_num(self_obj), 1);
                 end
-                if ((local_var(4) == 19093) and ((global_var(900) bwand 33554432) == 0)) then begin
+                if ((local_var(4) == 19093) and ((global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS) bwand 33554432) == 0)) then begin
                     animate_move_obj_to_tile(dog_obj, tile_num(self_obj), 1);
                 end
             end
             else begin
-                if ((global_var(900) bwand 32) == 0) then begin
-                    if ((local_var(4) == 20493) and ((global_var(900) bwand 4194304) == 0)) then begin
+                if ((global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS) bwand 32) == 0) then begin
+                    if ((local_var(4) == 20493) and ((global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS) bwand 4194304) == 0)) then begin
                         animate_move_obj_to_tile(lovec1_obj, tile_num(self_obj), 1);
                     end
-                    if ((local_var(4) == 20493) and ((global_var(900) bwand 8388608) == 0)) then begin
+                    if ((local_var(4) == 20493) and ((global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS) bwand 8388608) == 0)) then begin
                         animate_move_obj_to_tile(lovec2_obj, tile_num(self_obj), 1);
                     end
-                    if ((local_var(4) == 20493) and ((global_var(900) bwand 16777216) == 0)) then begin
+                    if ((local_var(4) == 20493) and ((global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS) bwand 16777216) == 0)) then begin
                         animate_move_obj_to_tile(lovec3_obj, tile_num(self_obj), 1);
                     end
-                    if ((local_var(4) == 20493) and ((global_var(900) bwand 33554432) == 0)) then begin
+                    if ((local_var(4) == 20493) and ((global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS) bwand 33554432) == 0)) then begin
                         animate_move_obj_to_tile(dog_obj, tile_num(self_obj), 1);
                     end
                 end
                 else begin
-                    if ((global_var(900) bwand 64) == 0) then begin
-                        if ((local_var(4) == 19103) and ((global_var(900) bwand 4194304) == 0)) then begin
+                    if ((global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS) bwand 64) == 0) then begin
+                        if ((local_var(4) == 19103) and ((global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS) bwand 4194304) == 0)) then begin
                             animate_move_obj_to_tile(lovec1_obj, tile_num(self_obj), 1);
                         end
-                        if ((local_var(4) == 19103) and ((global_var(900) bwand 8388608) == 0)) then begin
+                        if ((local_var(4) == 19103) and ((global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS) bwand 8388608) == 0)) then begin
                             animate_move_obj_to_tile(lovec2_obj, tile_num(self_obj), 1);
                         end
-                        if ((local_var(4) == 19103) and ((global_var(900) bwand 16777216) == 0)) then begin
+                        if ((local_var(4) == 19103) and ((global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS) bwand 16777216) == 0)) then begin
                             animate_move_obj_to_tile(lovec3_obj, tile_num(self_obj), 1);
                         end
-                        if ((local_var(4) == 19103) and ((global_var(900) bwand 33554432) == 0)) then begin
+                        if ((local_var(4) == 19103) and ((global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS) bwand 33554432) == 0)) then begin
                             animate_move_obj_to_tile(dog_obj, tile_num(self_obj), 1);
                         end
                     end
                     else begin
-                        if ((global_var(900) bwand 128) == 0) then begin
-                            if ((local_var(4) == 20302) and ((global_var(900) bwand 4194304) == 0)) then begin
+                        if ((global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS) bwand 128) == 0) then begin
+                            if ((local_var(4) == 20302) and ((global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS) bwand 4194304) == 0)) then begin
                                 animate_move_obj_to_tile(lovec1_obj, tile_num(self_obj), 1);
                             end
-                            if ((local_var(4) == 20302) and ((global_var(900) bwand 8388608) == 0)) then begin
+                            if ((local_var(4) == 20302) and ((global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS) bwand 8388608) == 0)) then begin
                                 animate_move_obj_to_tile(lovec2_obj, tile_num(self_obj), 1);
                             end
-                            if ((local_var(4) == 20302) and ((global_var(900) bwand 16777216) == 0)) then begin
+                            if ((local_var(4) == 20302) and ((global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS) bwand 16777216) == 0)) then begin
                                 animate_move_obj_to_tile(lovec3_obj, tile_num(self_obj), 1);
                             end
-                            if ((local_var(4) == 20302) and ((global_var(900) bwand 33554432) == 0)) then begin
+                            if ((local_var(4) == 20302) and ((global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS) bwand 33554432) == 0)) then begin
                                 animate_move_obj_to_tile(dog_obj, tile_num(self_obj), 1);
                             end
                         end
                         else begin
-                            if ((global_var(900) bwand 256) == 0) then begin
-                                if ((local_var(4) == 18909) and ((global_var(900) bwand 4194304) == 0)) then begin
+                            if ((global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS) bwand 256) == 0) then begin
+                                if ((local_var(4) == 18909) and ((global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS) bwand 4194304) == 0)) then begin
                                     animate_move_obj_to_tile(lovec1_obj, tile_num(self_obj), 1);
                                 end
-                                if ((local_var(4) == 18909) and ((global_var(900) bwand 8388608) == 0)) then begin
+                                if ((local_var(4) == 18909) and ((global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS) bwand 8388608) == 0)) then begin
                                     animate_move_obj_to_tile(lovec2_obj, tile_num(self_obj), 1);
                                 end
-                                if ((local_var(4) == 18909) and ((global_var(900) bwand 16777216) == 0)) then begin
+                                if ((local_var(4) == 18909) and ((global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS) bwand 16777216) == 0)) then begin
                                     animate_move_obj_to_tile(lovec3_obj, tile_num(self_obj), 1);
                                 end
-                                if ((local_var(4) == 18909) and ((global_var(900) bwand 33554432) == 0)) then begin
+                                if ((local_var(4) == 18909) and ((global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS) bwand 33554432) == 0)) then begin
                                     animate_move_obj_to_tile(dog_obj, tile_num(self_obj), 1);
                                 end
                             end
                             else begin
-                                if ((global_var(900) bwand 512) == 0) then begin
-                                    if ((local_var(4) == 20514) and ((global_var(900) bwand 4194304) == 0)) then begin
+                                if ((global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS) bwand 512) == 0) then begin
+                                    if ((local_var(4) == 20514) and ((global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS) bwand 4194304) == 0)) then begin
                                         animate_move_obj_to_tile(lovec1_obj, tile_num(self_obj), 1);
                                     end
-                                    if ((local_var(4) == 20514) and ((global_var(900) bwand 8388608) == 0)) then begin
+                                    if ((local_var(4) == 20514) and ((global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS) bwand 8388608) == 0)) then begin
                                         animate_move_obj_to_tile(lovec2_obj, tile_num(self_obj), 1);
                                     end
-                                    if ((local_var(4) == 20514) and ((global_var(900) bwand 16777216) == 0)) then begin
+                                    if ((local_var(4) == 20514) and ((global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS) bwand 16777216) == 0)) then begin
                                         animate_move_obj_to_tile(lovec3_obj, tile_num(self_obj), 1);
                                     end
-                                    if ((local_var(4) == 20514) and ((global_var(900) bwand 33554432) == 0)) then begin
+                                    if ((local_var(4) == 20514) and ((global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS) bwand 33554432) == 0)) then begin
                                         animate_move_obj_to_tile(dog_obj, tile_num(self_obj), 1);
                                     end
                                 end
                                 else begin
-                                    if ((global_var(900) bwand 1024) == 0) then begin
-                                        if ((local_var(4) == 22112) and ((global_var(900) bwand 4194304) == 0)) then begin
+                                    if ((global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS) bwand 1024) == 0) then begin
+                                        if ((local_var(4) == 22112) and ((global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS) bwand 4194304) == 0)) then begin
                                             animate_move_obj_to_tile(lovec1_obj, tile_num(self_obj), 1);
                                         end
-                                        if ((local_var(4) == 22112) and ((global_var(900) bwand 8388608) == 0)) then begin
+                                        if ((local_var(4) == 22112) and ((global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS) bwand 8388608) == 0)) then begin
                                             animate_move_obj_to_tile(lovec2_obj, tile_num(self_obj), 1);
                                         end
-                                        if ((local_var(4) == 22112) and ((global_var(900) bwand 16777216) == 0)) then begin
+                                        if ((local_var(4) == 22112) and ((global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS) bwand 16777216) == 0)) then begin
                                             animate_move_obj_to_tile(lovec3_obj, tile_num(self_obj), 1);
                                         end
-                                        if ((local_var(4) == 22112) and ((global_var(900) bwand 33554432) == 0)) then begin
+                                        if ((local_var(4) == 22112) and ((global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS) bwand 33554432) == 0)) then begin
                                             animate_move_obj_to_tile(dog_obj, tile_num(self_obj), 1);
                                         end
                                     end
                                     else begin
-                                        if ((global_var(900) bwand 2048) == 0) then begin
-                                            if ((local_var(4) == 17697) and ((global_var(900) bwand 4194304) == 0)) then begin
+                                        if ((global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS) bwand 2048) == 0) then begin
+                                            if ((local_var(4) == 17697) and ((global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS) bwand 4194304) == 0)) then begin
                                                 animate_move_obj_to_tile(lovec1_obj, tile_num(self_obj), 1);
                                             end
-                                            if ((local_var(4) == 17697) and ((global_var(900) bwand 8388608) == 0)) then begin
+                                            if ((local_var(4) == 17697) and ((global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS) bwand 8388608) == 0)) then begin
                                                 animate_move_obj_to_tile(lovec2_obj, tile_num(self_obj), 1);
                                             end
-                                            if ((local_var(4) == 17697) and ((global_var(900) bwand 16777216) == 0)) then begin
+                                            if ((local_var(4) == 17697) and ((global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS) bwand 16777216) == 0)) then begin
                                                 animate_move_obj_to_tile(lovec3_obj, tile_num(self_obj), 1);
                                             end
-                                            if ((local_var(4) == 17697) and ((global_var(900) bwand 33554432) == 0)) then begin
+                                            if ((local_var(4) == 17697) and ((global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS) bwand 33554432) == 0)) then begin
                                                 animate_move_obj_to_tile(dog_obj, tile_num(self_obj), 1);
                                             end
                                         end
                                         else begin
-                                            if ((global_var(900) bwand 4096) == 0) then begin
-                                                if ((local_var(4) == 21701) and ((global_var(900) bwand 4194304) == 0)) then begin
+                                            if ((global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS) bwand 4096) == 0) then begin
+                                                if ((local_var(4) == 21701) and ((global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS) bwand 4194304) == 0)) then begin
                                                     animate_move_obj_to_tile(lovec1_obj, tile_num(self_obj), 1);
                                                 end
-                                                if ((local_var(4) == 21701) and ((global_var(900) bwand 8388608) == 0)) then begin
+                                                if ((local_var(4) == 21701) and ((global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS) bwand 8388608) == 0)) then begin
                                                     animate_move_obj_to_tile(lovec2_obj, tile_num(self_obj), 1);
                                                 end
-                                                if ((local_var(4) == 21701) and ((global_var(900) bwand 16777216) == 0)) then begin
+                                                if ((local_var(4) == 21701) and ((global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS) bwand 16777216) == 0)) then begin
                                                     animate_move_obj_to_tile(lovec3_obj, tile_num(self_obj), 1);
                                                 end
-                                                if ((local_var(4) == 21701) and ((global_var(900) bwand 33554432) == 0)) then begin
+                                                if ((local_var(4) == 21701) and ((global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS) bwand 33554432) == 0)) then begin
                                                     animate_move_obj_to_tile(dog_obj, tile_num(self_obj), 1);
                                                 end
                                             end
@@ -285,138 +285,138 @@ begin
                 end
             end
         end
-        if (cur_map_index == 93) then begin
-            if ((global_var(900) bwand 16) == 0) then begin
-                if ((local_var(4) == 20905) and ((global_var(900) bwand 4194304) == 0)) then begin
+        if (CUR_MAP_ALB_DIV2) then begin
+            if ((global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS) bwand 16) == 0) then begin
+                if ((local_var(4) == 20905) and ((global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS) bwand 4194304) == 0)) then begin
                     animate_move_obj_to_tile(lovec1_obj, tile_num(self_obj), 1);
                 end
-                if ((local_var(4) == 20905) and ((global_var(900) bwand 8388608) == 0)) then begin
+                if ((local_var(4) == 20905) and ((global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS) bwand 8388608) == 0)) then begin
                     animate_move_obj_to_tile(lovec2_obj, tile_num(self_obj), 1);
                 end
-                if ((local_var(4) == 20905) and ((global_var(900) bwand 16777216) == 0)) then begin
+                if ((local_var(4) == 20905) and ((global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS) bwand 16777216) == 0)) then begin
                     animate_move_obj_to_tile(lovec3_obj, tile_num(self_obj), 1);
                 end
-                if ((local_var(4) == 20905) and ((global_var(900) bwand 33554432) == 0)) then begin
+                if ((local_var(4) == 20905) and ((global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS) bwand 33554432) == 0)) then begin
                     animate_move_obj_to_tile(dog_obj, tile_num(self_obj), 1);
                 end
             end
             else begin
-                if ((global_var(900) bwand 32) == 0) then begin
-                    if ((local_var(4) == 23511) and ((global_var(900) bwand 4194304) == 0)) then begin
+                if ((global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS) bwand 32) == 0) then begin
+                    if ((local_var(4) == 23511) and ((global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS) bwand 4194304) == 0)) then begin
                         animate_move_obj_to_tile(lovec1_obj, tile_num(self_obj), 1);
                     end
-                    if ((local_var(4) == 23511) and ((global_var(900) bwand 8388608) == 0)) then begin
+                    if ((local_var(4) == 23511) and ((global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS) bwand 8388608) == 0)) then begin
                         animate_move_obj_to_tile(lovec2_obj, tile_num(self_obj), 1);
                     end
-                    if ((local_var(4) == 23511) and ((global_var(900) bwand 16777216) == 0)) then begin
+                    if ((local_var(4) == 23511) and ((global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS) bwand 16777216) == 0)) then begin
                         animate_move_obj_to_tile(lovec3_obj, tile_num(self_obj), 1);
                     end
-                    if ((local_var(4) == 23511) and ((global_var(900) bwand 33554432) == 0)) then begin
+                    if ((local_var(4) == 23511) and ((global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS) bwand 33554432) == 0)) then begin
                         animate_move_obj_to_tile(dog_obj, tile_num(self_obj), 1);
                     end
                 end
                 else begin
-                    if ((global_var(900) bwand 64) == 0) then begin
-                        if ((local_var(4) == 23910) and ((global_var(900) bwand 4194304) == 0)) then begin
+                    if ((global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS) bwand 64) == 0) then begin
+                        if ((local_var(4) == 23910) and ((global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS) bwand 4194304) == 0)) then begin
                             animate_move_obj_to_tile(lovec1_obj, tile_num(self_obj), 1);
                         end
-                        if ((local_var(4) == 23910) and ((global_var(900) bwand 8388608) == 0)) then begin
+                        if ((local_var(4) == 23910) and ((global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS) bwand 8388608) == 0)) then begin
                             animate_move_obj_to_tile(lovec2_obj, tile_num(self_obj), 1);
                         end
-                        if ((local_var(4) == 23910) and ((global_var(900) bwand 16777216) == 0)) then begin
+                        if ((local_var(4) == 23910) and ((global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS) bwand 16777216) == 0)) then begin
                             animate_move_obj_to_tile(lovec3_obj, tile_num(self_obj), 1);
                         end
-                        if ((local_var(4) == 23910) and ((global_var(900) bwand 33554432) == 0)) then begin
+                        if ((local_var(4) == 23910) and ((global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS) bwand 33554432) == 0)) then begin
                             animate_move_obj_to_tile(dog_obj, tile_num(self_obj), 1);
                         end
                     end
                     else begin
-                        if ((global_var(900) bwand 128) == 0) then begin
-                            if ((local_var(4) == 21298) and ((global_var(900) bwand 4194304) == 0)) then begin
+                        if ((global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS) bwand 128) == 0) then begin
+                            if ((local_var(4) == 21298) and ((global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS) bwand 4194304) == 0)) then begin
                                 animate_move_obj_to_tile(lovec1_obj, tile_num(self_obj), 1);
                             end
-                            if ((local_var(4) == 21298) and ((global_var(900) bwand 8388608) == 0)) then begin
+                            if ((local_var(4) == 21298) and ((global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS) bwand 8388608) == 0)) then begin
                                 animate_move_obj_to_tile(lovec2_obj, tile_num(self_obj), 1);
                             end
-                            if ((local_var(4) == 21298) and ((global_var(900) bwand 16777216) == 0)) then begin
+                            if ((local_var(4) == 21298) and ((global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS) bwand 16777216) == 0)) then begin
                                 animate_move_obj_to_tile(lovec3_obj, tile_num(self_obj), 1);
                             end
-                            if ((local_var(4) == 21298) and ((global_var(900) bwand 33554432) == 0)) then begin
+                            if ((local_var(4) == 21298) and ((global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS) bwand 33554432) == 0)) then begin
                                 animate_move_obj_to_tile(dog_obj, tile_num(self_obj), 1);
                             end
                         end
                         else begin
-                            if ((global_var(900) bwand 256) == 0) then begin
-                                if ((local_var(4) == 20494) and ((global_var(900) bwand 4194304) == 0)) then begin
+                            if ((global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS) bwand 256) == 0) then begin
+                                if ((local_var(4) == 20494) and ((global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS) bwand 4194304) == 0)) then begin
                                     animate_move_obj_to_tile(lovec1_obj, tile_num(self_obj), 1);
                                 end
-                                if ((local_var(4) == 20494) and ((global_var(900) bwand 8388608) == 0)) then begin
+                                if ((local_var(4) == 20494) and ((global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS) bwand 8388608) == 0)) then begin
                                     animate_move_obj_to_tile(lovec2_obj, tile_num(self_obj), 1);
                                 end
-                                if ((local_var(4) == 20494) and ((global_var(900) bwand 16777216) == 0)) then begin
+                                if ((local_var(4) == 20494) and ((global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS) bwand 16777216) == 0)) then begin
                                     animate_move_obj_to_tile(lovec3_obj, tile_num(self_obj), 1);
                                 end
-                                if ((local_var(4) == 20494) and ((global_var(900) bwand 33554432) == 0)) then begin
+                                if ((local_var(4) == 20494) and ((global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS) bwand 33554432) == 0)) then begin
                                     animate_move_obj_to_tile(dog_obj, tile_num(self_obj), 1);
                                 end
                             end
                             else begin
-                                if ((global_var(900) bwand 512) == 0) then begin
-                                    if ((local_var(4) == 23497) and ((global_var(900) bwand 4194304) == 0)) then begin
+                                if ((global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS) bwand 512) == 0) then begin
+                                    if ((local_var(4) == 23497) and ((global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS) bwand 4194304) == 0)) then begin
                                         animate_move_obj_to_tile(lovec1_obj, tile_num(self_obj), 1);
                                     end
-                                    if ((local_var(4) == 23497) and ((global_var(900) bwand 8388608) == 0)) then begin
+                                    if ((local_var(4) == 23497) and ((global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS) bwand 8388608) == 0)) then begin
                                         animate_move_obj_to_tile(lovec2_obj, tile_num(self_obj), 1);
                                     end
-                                    if ((local_var(4) == 23497) and ((global_var(900) bwand 16777216) == 0)) then begin
+                                    if ((local_var(4) == 23497) and ((global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS) bwand 16777216) == 0)) then begin
                                         animate_move_obj_to_tile(lovec3_obj, tile_num(self_obj), 1);
                                     end
-                                    if ((local_var(4) == 23497) and ((global_var(900) bwand 33554432) == 0)) then begin
+                                    if ((local_var(4) == 23497) and ((global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS) bwand 33554432) == 0)) then begin
                                         animate_move_obj_to_tile(dog_obj, tile_num(self_obj), 1);
                                     end
                                 end
                                 else begin
-                                    if ((global_var(900) bwand 1024) == 0) then begin
-                                        if ((local_var(4) == 22691) and ((global_var(900) bwand 4194304) == 0)) then begin
+                                    if ((global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS) bwand 1024) == 0) then begin
+                                        if ((local_var(4) == 22691) and ((global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS) bwand 4194304) == 0)) then begin
                                             animate_move_obj_to_tile(lovec1_obj, tile_num(self_obj), 1);
                                         end
-                                        if ((local_var(4) == 22691) and ((global_var(900) bwand 8388608) == 0)) then begin
+                                        if ((local_var(4) == 22691) and ((global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS) bwand 8388608) == 0)) then begin
                                             animate_move_obj_to_tile(lovec2_obj, tile_num(self_obj), 1);
                                         end
-                                        if ((local_var(4) == 22691) and ((global_var(900) bwand 16777216) == 0)) then begin
+                                        if ((local_var(4) == 22691) and ((global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS) bwand 16777216) == 0)) then begin
                                             animate_move_obj_to_tile(lovec3_obj, tile_num(self_obj), 1);
                                         end
-                                        if ((local_var(4) == 22691) and ((global_var(900) bwand 33554432) == 0)) then begin
+                                        if ((local_var(4) == 22691) and ((global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS) bwand 33554432) == 0)) then begin
                                             animate_move_obj_to_tile(dog_obj, tile_num(self_obj), 1);
                                         end
                                     end
                                     else begin
-                                        if ((global_var(900) bwand 2048) == 0) then begin
-                                            if ((local_var(4) == 25496) and ((global_var(900) bwand 4194304) == 0)) then begin
+                                        if ((global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS) bwand 2048) == 0) then begin
+                                            if ((local_var(4) == 25496) and ((global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS) bwand 4194304) == 0)) then begin
                                                 animate_move_obj_to_tile(lovec1_obj, tile_num(self_obj), 1);
                                             end
-                                            if ((local_var(4) == 25496) and ((global_var(900) bwand 8388608) == 0)) then begin
+                                            if ((local_var(4) == 25496) and ((global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS) bwand 8388608) == 0)) then begin
                                                 animate_move_obj_to_tile(lovec2_obj, tile_num(self_obj), 1);
                                             end
-                                            if ((local_var(4) == 25496) and ((global_var(900) bwand 16777216) == 0)) then begin
+                                            if ((local_var(4) == 25496) and ((global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS) bwand 16777216) == 0)) then begin
                                                 animate_move_obj_to_tile(lovec3_obj, tile_num(self_obj), 1);
                                             end
-                                            if ((local_var(4) == 25496) and ((global_var(900) bwand 33554432) == 0)) then begin
+                                            if ((local_var(4) == 25496) and ((global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS) bwand 33554432) == 0)) then begin
                                                 animate_move_obj_to_tile(dog_obj, tile_num(self_obj), 1);
                                             end
                                         end
                                         else begin
-                                            if ((global_var(900) bwand 4096) == 0) then begin
-                                                if ((local_var(4) == 24303) and ((global_var(900) bwand 4194304) == 0)) then begin
+                                            if ((global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS) bwand 4096) == 0) then begin
+                                                if ((local_var(4) == 24303) and ((global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS) bwand 4194304) == 0)) then begin
                                                     animate_move_obj_to_tile(lovec1_obj, tile_num(self_obj), 1);
                                                 end
-                                                if ((local_var(4) == 24303) and ((global_var(900) bwand 8388608) == 0)) then begin
+                                                if ((local_var(4) == 24303) and ((global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS) bwand 8388608) == 0)) then begin
                                                     animate_move_obj_to_tile(lovec2_obj, tile_num(self_obj), 1);
                                                 end
-                                                if ((local_var(4) == 24303) and ((global_var(900) bwand 16777216) == 0)) then begin
+                                                if ((local_var(4) == 24303) and ((global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS) bwand 16777216) == 0)) then begin
                                                     animate_move_obj_to_tile(lovec3_obj, tile_num(self_obj), 1);
                                                 end
-                                                if ((local_var(4) == 24303) and ((global_var(900) bwand 33554432) == 0)) then begin
+                                                if ((local_var(4) == 24303) and ((global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS) bwand 33554432) == 0)) then begin
                                                     animate_move_obj_to_tile(dog_obj, tile_num(self_obj), 1);
                                                 end
                                             end
@@ -429,183 +429,183 @@ begin
                 end
             end
         end
-        if (cur_map_index == 94) then begin
-            if ((global_var(900) bwand 16) == 0) then begin
-                if ((local_var(4) == 21103) and ((global_var(900) bwand 4194304) == 0)) then begin
+        if (CUR_MAP_ALB_DIV3) then begin
+            if ((global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS) bwand 16) == 0) then begin
+                if ((local_var(4) == 21103) and ((global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS) bwand 4194304) == 0)) then begin
                     animate_move_obj_to_tile(lovec1_obj, tile_num(self_obj), 1);
                 end
-                if ((local_var(4) == 21103) and ((global_var(900) bwand 8388608) == 0)) then begin
+                if ((local_var(4) == 21103) and ((global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS) bwand 8388608) == 0)) then begin
                     animate_move_obj_to_tile(lovec2_obj, tile_num(self_obj), 1);
                 end
-                if ((local_var(4) == 21103) and ((global_var(900) bwand 16777216) == 0)) then begin
+                if ((local_var(4) == 21103) and ((global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS) bwand 16777216) == 0)) then begin
                     animate_move_obj_to_tile(lovec3_obj, tile_num(self_obj), 1);
                 end
-                if ((local_var(4) == 21103) and ((global_var(900) bwand 33554432) == 0)) then begin
+                if ((local_var(4) == 21103) and ((global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS) bwand 33554432) == 0)) then begin
                     animate_move_obj_to_tile(dog_obj, tile_num(self_obj), 1);
                 end
             end
             else begin
-                if ((global_var(900) bwand 32) == 0) then begin
-                    if ((local_var(4) == 19497) and ((global_var(900) bwand 4194304) == 0)) then begin
+                if ((global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS) bwand 32) == 0) then begin
+                    if ((local_var(4) == 19497) and ((global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS) bwand 4194304) == 0)) then begin
                         animate_move_obj_to_tile(lovec1_obj, tile_num(self_obj), 1);
                     end
-                    if ((local_var(4) == 19497) and ((global_var(900) bwand 8388608) == 0)) then begin
+                    if ((local_var(4) == 19497) and ((global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS) bwand 8388608) == 0)) then begin
                         animate_move_obj_to_tile(lovec2_obj, tile_num(self_obj), 1);
                     end
-                    if ((local_var(4) == 19497) and ((global_var(900) bwand 16777216) == 0)) then begin
+                    if ((local_var(4) == 19497) and ((global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS) bwand 16777216) == 0)) then begin
                         animate_move_obj_to_tile(lovec3_obj, tile_num(self_obj), 1);
                     end
-                    if ((local_var(4) == 19497) and ((global_var(900) bwand 33554432) == 0)) then begin
+                    if ((local_var(4) == 19497) and ((global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS) bwand 33554432) == 0)) then begin
                         animate_move_obj_to_tile(dog_obj, tile_num(self_obj), 1);
                     end
                 end
                 else begin
-                    if ((global_var(900) bwand 64) == 0) then begin
-                        if ((local_var(4) == 19706) and ((global_var(900) bwand 4194304) == 0)) then begin
+                    if ((global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS) bwand 64) == 0) then begin
+                        if ((local_var(4) == 19706) and ((global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS) bwand 4194304) == 0)) then begin
                             animate_move_obj_to_tile(lovec1_obj, tile_num(self_obj), 1);
                         end
-                        if ((local_var(4) == 19706) and ((global_var(900) bwand 8388608) == 0)) then begin
+                        if ((local_var(4) == 19706) and ((global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS) bwand 8388608) == 0)) then begin
                             animate_move_obj_to_tile(lovec2_obj, tile_num(self_obj), 1);
                         end
-                        if ((local_var(4) == 19706) and ((global_var(900) bwand 16777216) == 0)) then begin
+                        if ((local_var(4) == 19706) and ((global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS) bwand 16777216) == 0)) then begin
                             animate_move_obj_to_tile(lovec3_obj, tile_num(self_obj), 1);
                         end
-                        if ((local_var(4) == 19706) and ((global_var(900) bwand 33554432) == 0)) then begin
+                        if ((local_var(4) == 19706) and ((global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS) bwand 33554432) == 0)) then begin
                             animate_move_obj_to_tile(dog_obj, tile_num(self_obj), 1);
                         end
                     end
                     else begin
-                        if ((global_var(900) bwand 128) == 0) then begin
-                            if ((local_var(4) == 20289) and ((global_var(900) bwand 4194304) == 0)) then begin
+                        if ((global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS) bwand 128) == 0) then begin
+                            if ((local_var(4) == 20289) and ((global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS) bwand 4194304) == 0)) then begin
                                 animate_move_obj_to_tile(lovec1_obj, tile_num(self_obj), 1);
                             end
-                            if ((local_var(4) == 20289) and ((global_var(900) bwand 8388608) == 0)) then begin
+                            if ((local_var(4) == 20289) and ((global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS) bwand 8388608) == 0)) then begin
                                 animate_move_obj_to_tile(lovec2_obj, tile_num(self_obj), 1);
                             end
-                            if ((local_var(4) == 20289) and ((global_var(900) bwand 16777216) == 0)) then begin
+                            if ((local_var(4) == 20289) and ((global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS) bwand 16777216) == 0)) then begin
                                 animate_move_obj_to_tile(lovec3_obj, tile_num(self_obj), 1);
                             end
-                            if ((local_var(4) == 20289) and ((global_var(900) bwand 33554432) == 0)) then begin
+                            if ((local_var(4) == 20289) and ((global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS) bwand 33554432) == 0)) then begin
                                 animate_move_obj_to_tile(dog_obj, tile_num(self_obj), 1);
                             end
                         end
                         else begin
-                            if ((global_var(900) bwand 256) == 0) then begin
-                                if ((local_var(4) == 18290) and ((global_var(900) bwand 4194304) == 0)) then begin
+                            if ((global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS) bwand 256) == 0) then begin
+                                if ((local_var(4) == 18290) and ((global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS) bwand 4194304) == 0)) then begin
                                     animate_move_obj_to_tile(lovec1_obj, tile_num(self_obj), 1);
                                 end
-                                if ((local_var(4) == 18290) and ((global_var(900) bwand 8388608) == 0)) then begin
+                                if ((local_var(4) == 18290) and ((global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS) bwand 8388608) == 0)) then begin
                                     animate_move_obj_to_tile(lovec2_obj, tile_num(self_obj), 1);
                                 end
-                                if ((local_var(4) == 18290) and ((global_var(900) bwand 16777216) == 0)) then begin
+                                if ((local_var(4) == 18290) and ((global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS) bwand 16777216) == 0)) then begin
                                     animate_move_obj_to_tile(lovec3_obj, tile_num(self_obj), 1);
                                 end
-                                if ((local_var(4) == 18290) and ((global_var(900) bwand 33554432) == 0)) then begin
+                                if ((local_var(4) == 18290) and ((global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS) bwand 33554432) == 0)) then begin
                                     animate_move_obj_to_tile(dog_obj, tile_num(self_obj), 1);
                                 end
                             end
                             else begin
-                                if ((global_var(900) bwand 512) == 0) then begin
-                                    if ((local_var(4) == 18502) and ((global_var(900) bwand 4194304) == 0)) then begin
+                                if ((global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS) bwand 512) == 0) then begin
+                                    if ((local_var(4) == 18502) and ((global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS) bwand 4194304) == 0)) then begin
                                         animate_move_obj_to_tile(lovec1_obj, tile_num(self_obj), 1);
                                     end
-                                    if ((local_var(4) == 18502) and ((global_var(900) bwand 8388608) == 0)) then begin
+                                    if ((local_var(4) == 18502) and ((global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS) bwand 8388608) == 0)) then begin
                                         animate_move_obj_to_tile(lovec2_obj, tile_num(self_obj), 1);
                                     end
-                                    if ((local_var(4) == 18502) and ((global_var(900) bwand 16777216) == 0)) then begin
+                                    if ((local_var(4) == 18502) and ((global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS) bwand 16777216) == 0)) then begin
                                         animate_move_obj_to_tile(lovec3_obj, tile_num(self_obj), 1);
                                     end
-                                    if ((local_var(4) == 18502) and ((global_var(900) bwand 33554432) == 0)) then begin
+                                    if ((local_var(4) == 18502) and ((global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS) bwand 33554432) == 0)) then begin
                                         animate_move_obj_to_tile(dog_obj, tile_num(self_obj), 1);
                                     end
                                 end
                                 else begin
-                                    if ((global_var(900) bwand 1024) == 0) then begin
-                                        if ((local_var(4) == 18509) and ((global_var(900) bwand 4194304) == 0)) then begin
+                                    if ((global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS) bwand 1024) == 0) then begin
+                                        if ((local_var(4) == 18509) and ((global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS) bwand 4194304) == 0)) then begin
                                             animate_move_obj_to_tile(lovec1_obj, tile_num(self_obj), 1);
                                         end
-                                        if ((local_var(4) == 18509) and ((global_var(900) bwand 8388608) == 0)) then begin
+                                        if ((local_var(4) == 18509) and ((global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS) bwand 8388608) == 0)) then begin
                                             animate_move_obj_to_tile(lovec2_obj, tile_num(self_obj), 1);
                                         end
-                                        if ((local_var(4) == 18509) and ((global_var(900) bwand 16777216) == 0)) then begin
+                                        if ((local_var(4) == 18509) and ((global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS) bwand 16777216) == 0)) then begin
                                             animate_move_obj_to_tile(lovec3_obj, tile_num(self_obj), 1);
                                         end
-                                        if ((local_var(4) == 18509) and ((global_var(900) bwand 33554432) == 0)) then begin
+                                        if ((local_var(4) == 18509) and ((global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS) bwand 33554432) == 0)) then begin
                                             animate_move_obj_to_tile(dog_obj, tile_num(self_obj), 1);
                                         end
                                     end
                                     else begin
-                                        if ((global_var(900) bwand 2048) == 0) then begin
-                                            if ((local_var(4) == 19116) and ((global_var(900) bwand 4194304) == 0)) then begin
+                                        if ((global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS) bwand 2048) == 0) then begin
+                                            if ((local_var(4) == 19116) and ((global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS) bwand 4194304) == 0)) then begin
                                                 animate_move_obj_to_tile(lovec1_obj, tile_num(self_obj), 1);
                                             end
-                                            if ((local_var(4) == 19116) and ((global_var(900) bwand 8388608) == 0)) then begin
+                                            if ((local_var(4) == 19116) and ((global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS) bwand 8388608) == 0)) then begin
                                                 animate_move_obj_to_tile(lovec2_obj, tile_num(self_obj), 1);
                                             end
-                                            if ((local_var(4) == 19116) and ((global_var(900) bwand 16777216) == 0)) then begin
+                                            if ((local_var(4) == 19116) and ((global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS) bwand 16777216) == 0)) then begin
                                                 animate_move_obj_to_tile(lovec3_obj, tile_num(self_obj), 1);
                                             end
-                                            if ((local_var(4) == 19116) and ((global_var(900) bwand 33554432) == 0)) then begin
+                                            if ((local_var(4) == 19116) and ((global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS) bwand 33554432) == 0)) then begin
                                                 animate_move_obj_to_tile(dog_obj, tile_num(self_obj), 1);
                                             end
                                         end
                                         else begin
-                                            if ((global_var(900) bwand 4096) == 0) then begin
-                                                if ((local_var(4) == 16912) and ((global_var(900) bwand 4194304) == 0)) then begin
+                                            if ((global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS) bwand 4096) == 0) then begin
+                                                if ((local_var(4) == 16912) and ((global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS) bwand 4194304) == 0)) then begin
                                                     animate_move_obj_to_tile(lovec1_obj, tile_num(self_obj), 1);
                                                 end
-                                                if ((local_var(4) == 16912) and ((global_var(900) bwand 8388608) == 0)) then begin
+                                                if ((local_var(4) == 16912) and ((global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS) bwand 8388608) == 0)) then begin
                                                     animate_move_obj_to_tile(lovec2_obj, tile_num(self_obj), 1);
                                                 end
-                                                if ((local_var(4) == 16912) and ((global_var(900) bwand 16777216) == 0)) then begin
+                                                if ((local_var(4) == 16912) and ((global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS) bwand 16777216) == 0)) then begin
                                                     animate_move_obj_to_tile(lovec3_obj, tile_num(self_obj), 1);
                                                 end
-                                                if ((local_var(4) == 16912) and ((global_var(900) bwand 33554432) == 0)) then begin
+                                                if ((local_var(4) == 16912) and ((global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS) bwand 33554432) == 0)) then begin
                                                     animate_move_obj_to_tile(dog_obj, tile_num(self_obj), 1);
                                                 end
                                             end
                                             else begin
-                                                if ((global_var(900) bwand 8192) == 0) then begin
-                                                    if ((local_var(4) == 17719) and ((global_var(900) bwand 4194304) == 0)) then begin
+                                                if ((global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS) bwand 8192) == 0) then begin
+                                                    if ((local_var(4) == 17719) and ((global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS) bwand 4194304) == 0)) then begin
                                                         animate_move_obj_to_tile(lovec1_obj, tile_num(self_obj), 1);
                                                     end
-                                                    if ((local_var(4) == 17719) and ((global_var(900) bwand 8388608) == 0)) then begin
+                                                    if ((local_var(4) == 17719) and ((global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS) bwand 8388608) == 0)) then begin
                                                         animate_move_obj_to_tile(lovec2_obj, tile_num(self_obj), 1);
                                                     end
-                                                    if ((local_var(4) == 17719) and ((global_var(900) bwand 16777216) == 0)) then begin
+                                                    if ((local_var(4) == 17719) and ((global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS) bwand 16777216) == 0)) then begin
                                                         animate_move_obj_to_tile(lovec3_obj, tile_num(self_obj), 1);
                                                     end
-                                                    if ((local_var(4) == 17719) and ((global_var(900) bwand 33554432) == 0)) then begin
+                                                    if ((local_var(4) == 17719) and ((global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS) bwand 33554432) == 0)) then begin
                                                         animate_move_obj_to_tile(dog_obj, tile_num(self_obj), 1);
                                                     end
                                                 end
                                                 else begin
-                                                    if ((global_var(900) bwand 16384) == 0) then begin
-                                                        if ((local_var(4) == 17499) and ((global_var(900) bwand 4194304) == 0)) then begin
+                                                    if ((global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS) bwand 16384) == 0) then begin
+                                                        if ((local_var(4) == 17499) and ((global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS) bwand 4194304) == 0)) then begin
                                                             animate_move_obj_to_tile(lovec1_obj, tile_num(self_obj), 1);
                                                         end
-                                                        if ((local_var(4) == 17499) and ((global_var(900) bwand 8388608) == 0)) then begin
+                                                        if ((local_var(4) == 17499) and ((global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS) bwand 8388608) == 0)) then begin
                                                             animate_move_obj_to_tile(lovec2_obj, tile_num(self_obj), 1);
                                                         end
-                                                        if ((local_var(4) == 17499) and ((global_var(900) bwand 16777216) == 0)) then begin
+                                                        if ((local_var(4) == 17499) and ((global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS) bwand 16777216) == 0)) then begin
                                                             animate_move_obj_to_tile(lovec3_obj, tile_num(self_obj), 1);
                                                         end
-                                                        if ((local_var(4) == 17499) and ((global_var(900) bwand 33554432) == 0)) then begin
+                                                        if ((local_var(4) == 17499) and ((global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS) bwand 33554432) == 0)) then begin
                                                             animate_move_obj_to_tile(dog_obj, tile_num(self_obj), 1);
                                                         end
                                                     end
                                                     else begin
-                                                        if ((global_var(900) bwand 32768) == 0) then begin
-                                                            if ((local_var(4) == 16692) and ((global_var(900) bwand 4194304) == 0)) then begin
+                                                        if ((global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS) bwand 32768) == 0) then begin
+                                                            if ((local_var(4) == 16692) and ((global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS) bwand 4194304) == 0)) then begin
                                                                 animate_move_obj_to_tile(lovec1_obj, tile_num(self_obj), 1);
                                                             end
-                                                            if ((local_var(4) == 16692) and ((global_var(900) bwand 8388608) == 0)) then begin
+                                                            if ((local_var(4) == 16692) and ((global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS) bwand 8388608) == 0)) then begin
                                                                 animate_move_obj_to_tile(lovec2_obj, tile_num(self_obj), 1);
                                                             end
-                                                            if ((local_var(4) == 16692) and ((global_var(900) bwand 16777216) == 0)) then begin
+                                                            if ((local_var(4) == 16692) and ((global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS) bwand 16777216) == 0)) then begin
                                                                 animate_move_obj_to_tile(lovec3_obj, tile_num(self_obj), 1);
                                                             end
-                                                            if ((local_var(4) == 16692) and ((global_var(900) bwand 33554432) == 0)) then begin
+                                                            if ((local_var(4) == 16692) and ((global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS) bwand 33554432) == 0)) then begin
                                                                 animate_move_obj_to_tile(dog_obj, tile_num(self_obj), 1);
                                                             end
                                                         end
@@ -633,11 +633,11 @@ begin
     if (fixed_param == 2) then begin
         if (target_obj == dude_obj) then begin
             if (not(is_success(do_check(dude_obj, 6, -1)))) then begin
-                LVar0 := random(1, get_pc_stat(1) * 3);
+                LVar0 := random(1, get_pc_stat(PCSTAT_level) * 3);
                 if (LVar0 > 5) then begin
                     LVar0 := 5;
                 end
-                if (cur_map_index == 93) then begin
+                if (CUR_MAP_ALB_DIV2) then begin
                     LVar0 := random(0, 1);
                 end
                 radiation_inc(target_obj, LVar0);
@@ -648,98 +648,98 @@ end
 
 procedure pickup_p_proc
 begin
-    attack_complex(source_obj, 0, 1, 0, 0, 30000, 0, 0);
+    attack(source_obj);
 end
 
 procedure destroy_p_proc
 begin
     variable LVar0 := 0;
-    set_global_var(889, global_var(889) + 1);
-    if (cur_map_index == 92) then begin
+    set_global_var(GVAR_AQ_HUNTINGS_RESULTS, global_var(GVAR_AQ_HUNTINGS_RESULTS) + 1);
+    if (CUR_MAP_ALB_DIV1) then begin
         if (local_var(4) == 19093) then begin
-            set_global_var(900, global_var(900) bwor 16);
+            set_global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS, global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS) bwor 16);
         end
         if (local_var(4) == 20493) then begin
-            set_global_var(900, global_var(900) bwor 32);
+            set_global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS, global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS) bwor 32);
         end
         if (local_var(4) == 19103) then begin
-            set_global_var(900, global_var(900) bwor 64);
+            set_global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS, global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS) bwor 64);
         end
         if (local_var(4) == 20302) then begin
-            set_global_var(900, global_var(900) bwor 128);
+            set_global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS, global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS) bwor 128);
         end
         if (local_var(4) == 18909) then begin
-            set_global_var(900, global_var(900) bwor 256);
+            set_global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS, global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS) bwor 256);
         end
         if (local_var(4) == 20514) then begin
-            set_global_var(900, global_var(900) bwor 512);
+            set_global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS, global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS) bwor 512);
         end
         if (local_var(4) == 22112) then begin
-            set_global_var(900, global_var(900) bwor 1024);
+            set_global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS, global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS) bwor 1024);
         end
         if (local_var(4) == 17697) then begin
-            set_global_var(900, global_var(900) bwor 2048);
+            set_global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS, global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS) bwor 2048);
         end
         if (local_var(4) == 21701) then begin
-            set_global_var(900, global_var(900) bwor 4096);
+            set_global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS, global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS) bwor 4096);
         end
     end
-    if (cur_map_index == 93) then begin
+    if (CUR_MAP_ALB_DIV2) then begin
         if (local_var(4) == 20905) then begin
-            set_global_var(900, global_var(900) bwor 16);
+            set_global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS, global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS) bwor 16);
         end
         if (local_var(4) == 23511) then begin
-            set_global_var(900, global_var(900) bwor 32);
+            set_global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS, global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS) bwor 32);
         end
         if (local_var(4) == 23910) then begin
-            set_global_var(900, global_var(900) bwor 64);
+            set_global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS, global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS) bwor 64);
         end
         if (local_var(4) == 21298) then begin
-            set_global_var(900, global_var(900) bwor 128);
+            set_global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS, global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS) bwor 128);
         end
         if (local_var(4) == 20494) then begin
-            set_global_var(900, global_var(900) bwor 256);
+            set_global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS, global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS) bwor 256);
         end
         if (local_var(4) == 23497) then begin
-            set_global_var(900, global_var(900) bwor 512);
+            set_global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS, global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS) bwor 512);
         end
         if (local_var(4) == 22691) then begin
-            set_global_var(900, global_var(900) bwor 1024);
+            set_global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS, global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS) bwor 1024);
         end
         if (local_var(4) == 25496) then begin
-            set_global_var(900, global_var(900) bwor 2048);
+            set_global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS, global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS) bwor 2048);
         end
         if (local_var(4) == 24303) then begin
-            set_global_var(900, global_var(900) bwor 4096);
+            set_global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS, global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS) bwor 4096);
         end
     end
-    if (cur_map_index == 94) then begin
+    if (CUR_MAP_ALB_DIV3) then begin
         if (local_var(4) == 21103) then begin
-            set_global_var(900, global_var(900) bwor 16);
+            set_global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS, global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS) bwor 16);
         end
         if (local_var(4) == 19497) then begin
-            set_global_var(900, global_var(900) bwor 32);
+            set_global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS, global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS) bwor 32);
         end
         if (local_var(4) == 19706) then begin
-            set_global_var(900, global_var(900) bwor 64);
+            set_global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS, global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS) bwor 64);
         end
         if (local_var(4) == 20289) then begin
-            set_global_var(900, global_var(900) bwor 128);
+            set_global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS, global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS) bwor 128);
         end
         if (local_var(4) == 18290) then begin
-            set_global_var(900, global_var(900) bwor 256);
+            set_global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS, global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS) bwor 256);
         end
         if (local_var(4) == 18502) then begin
-            set_global_var(900, global_var(900) bwor 512);
+            set_global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS, global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS) bwor 512);
         end
         if (local_var(4) == 18509) then begin
-            set_global_var(900, global_var(900) bwor 1024);
+            set_global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS, global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS) bwor 1024);
         end
         if (local_var(4) == 19116) then begin
-            set_global_var(900, global_var(900) bwor 2048);
+            set_global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS, global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS) bwor 2048);
         end
         if (local_var(4) == 16912) then begin
-            set_global_var(900, global_var(900) bwor 4096);
+            set_global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS, global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS) bwor 4096);
         end
     end
 end
@@ -751,8 +751,8 @@ end
 procedure map_enter_p_proc
 begin
     Only_Once := 0;
-    critter_add_trait(self_obj, 1, 6, 49);
-    critter_add_trait(self_obj, 1, 5, 26);
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 49);
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_GECKO);
     if (local_var(4) == 0) then begin
         set_local_var(4, tile_num(self_obj));
     end
@@ -772,6 +772,5 @@ end
 
 procedure damage_p_proc
 begin
-    attack_complex(source_obj, 0, 1, 0, 0, 30000, 0, 0);
+    attack(source_obj);
 end
-
