diff --git a/Mods/Resurrection/Original/AHERMAN.ssl b/Mods/Resurrection/ReDefine/AHERMAN.ssl
index 01b8c80..00d62f6 100644
--- a/Mods/Resurrection/Original/AHERMAN.ssl
+++ b/Mods/Resurrection/ReDefine/AHERMAN.ssl
@@ -100,12 +100,12 @@ begin
             return 1;
         end
     end
-    if ((party_member_obj(16777779) != 0) and ((global_var(910) bwand 32) == 0)) then begin
+    if ((party_member_obj(16777779) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 32) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777779)) <= 5) then begin
             return 1;
         end
     end
-    if ((party_member_obj(16777811) != 0) and ((global_var(910) bwand 64) == 0)) then begin
+    if ((party_member_obj(16777811) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 64) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777811)) <= 5) then begin
             return 1;
         end
@@ -120,7 +120,7 @@ begin
             return 1;
         end
     end
-    if ((party_member_obj(16777814) != 0) and ((global_var(910) bwand 128) == 0)) then begin
+    if ((party_member_obj(16777814) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 128) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777814)) <= 5) then begin
             return 1;
         end
@@ -177,15 +177,15 @@ end
 
 procedure critter_p_proc
 begin
-    if ((local_var(4) == 1) or (global_var(898) == 0) or (global_var(864) == 1) and obj_can_see_obj(self_obj, dude_obj)) then begin
+    if ((local_var(4) == 1) or (global_var(GVAR_AQ_MUTANT_HUNTERS_CANDIDATE) == 0) or (global_var(GVAR_AQ_ENEMY_HUNTERS) == 1) and obj_can_see_obj(self_obj, dude_obj)) then begin
         if (critter_is_fleeing(self_obj)) then begin
             if (anim_busy(self_obj) == 0) then begin
                 if ((tile_distance_objs(self_obj, dude_obj) < 8) or obj_can_see_obj(self_obj, dude_obj)) then begin
                     if (anim_busy(self_obj) == 0) then begin
-                        global_temp := rotation_to_tile(tile_num(dude_obj), tile_num(self_obj));
+                        global_temp := rotation_to_tile(dude_tile, tile_num(self_obj));
                         animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(self_obj), global_temp, random(3, 10)), 1);
                         global_temp := (global_temp + 1) % 6;
-                        while ((anim_busy(self_obj) == 0) and (global_temp != rotation_to_tile(tile_num(dude_obj), tile_num(self_obj)))) do begin
+                        while ((anim_busy(self_obj) == 0) and (global_temp != rotation_to_tile(dude_tile, tile_num(self_obj)))) do begin
                             debug_msg("flee loop: rot == " + global_temp);
                             animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(self_obj), global_temp, random(3, 10)), 1);
                             global_temp := (global_temp + 1) % 6;
@@ -195,7 +195,7 @@ begin
             end
         end
         else begin
-            attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+            attack(dude_obj);
         end
     end
 end
@@ -204,25 +204,25 @@ procedure pickup_p_proc
 begin
     if (source_obj == dude_obj) then begin
         set_local_var(4, 1);
-        set_global_var(864, 1);
+        set_global_var(GVAR_AQ_ENEMY_HUNTERS, 1);
     end
 end
 
 procedure talk_p_proc
 begin
-    anim(dude_obj, 1000, rotation_to_tile(tile_num(dude_obj), tile_num(self_obj)));
-    anim(self_obj, 1000, rotation_to_tile(tile_num(self_obj), tile_num(dude_obj)));
+    anim(dude_obj, 1000, rotation_to_tile(dude_tile, tile_num(self_obj)));
+    anim(self_obj, 1000, rotation_to_tile(tile_num(self_obj), dude_tile));
     start_gdialog(1706, self_obj, 4, -1, -1);
     gsay_start;
-    if (global_var(859) == 1) then begin
+    if (global_var(GVAR_AQ_MUTANT_HUNTER) == 1) then begin
         call Node2150();
     end
     else begin
-        if ((global_var(882) == 2) or (global_var(882) == 3) and ((local_var(6) bwand 1024) == 0)) then begin
-            if (global_var(882) == 2) then begin
+        if ((global_var(GVAR_AQ_QUEST_MTHUNT_HERMAN_QUEST) == 2) or (global_var(GVAR_AQ_QUEST_MTHUNT_HERMAN_QUEST) == 3) and ((local_var(6) bwand 1024) == 0)) then begin
+            if (global_var(GVAR_AQ_QUEST_MTHUNT_HERMAN_QUEST) == 2) then begin
                 call Node1900();
             end
-            if (global_var(882) == 3) then begin
+            if (global_var(GVAR_AQ_QUEST_MTHUNT_HERMAN_QUEST) == 3) then begin
                 call Node1950();
             end
         end
@@ -245,58 +245,58 @@ end
 procedure destroy_p_proc
 begin
     if (source_obj == dude_obj) then begin
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(4) > (3 * global_var(5))) or (global_var(2) == 1)) and (global_var(1) == 0)) then begin
-            set_global_var(2, 1);
-            set_global_var(3, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(4) > (3 * global_var(GVAR_GOOD_MONSTER))) or (global_var(GVAR_CHAMPION_REPUTATION) == 1)) and (global_var(GVAR_CHILDKILLER_REPUTATION) == 0)) then begin
+            set_global_var(GVAR_CHAMPION_REPUTATION, 1);
+            set_global_var(GVAR_BERSERKER_REPUTATION, 0);
         end
-        if (metarule(51, self_obj) == 2) then begin
+        if (critter_kill_type(self_obj) == KILL_TYPE_children_kills) then begin
             if (source_obj == dude_obj) then begin
-                set_global_var(2, 0);
-                set_global_var(1, global_var(1) + 1);
-                set_global_var(0, global_var(0) + -15);
+                set_global_var(GVAR_CHAMPION_REPUTATION, 0);
+                set_global_var(GVAR_CHILDKILLER_REPUTATION, global_var(GVAR_CHILDKILLER_REPUTATION) + 1);
+                set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + -15);
                 debug_msg("Player gains " + -15 + " Karma Points.");
-                set_global_var(37, 0);
-                set_global_var(38, 0);
-                set_global_var(39, 0);
-                set_global_var(40, 0);
-                set_global_var(41, 0);
-                set_global_var(42, 0);
-                set_global_var(43, 0);
-                set_global_var(44, 0);
-                set_global_var(45, 0);
-                if (global_var(0) >= 1000) then begin
-                    set_global_var(37, 1);
+                set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+                set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+                set_global_var(GVAR_KARMA_DEFENDER, 0);
+                set_global_var(GVAR_KARMA_WANDERER, 0);
+                set_global_var(GVAR_KARMA_BETRAYER, 0);
+                set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+                set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+                set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+                    set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
                 end
                 else begin
-                    if (global_var(0) >= 750) then begin
-                        set_global_var(38, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
                     end
                     else begin
-                        if (global_var(0) >= 500) then begin
-                            set_global_var(39, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                            set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                         end
                         else begin
-                            if (global_var(0) >= 250) then begin
-                                set_global_var(40, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                                set_global_var(GVAR_KARMA_DEFENDER, 1);
                             end
                             else begin
-                                if (global_var(0) > -250) then begin
-                                    set_global_var(41, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                                    set_global_var(GVAR_KARMA_WANDERER, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -500) then begin
-                                        set_global_var(42, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                        set_global_var(GVAR_KARMA_BETRAYER, 1);
                                     end
                                     else begin
-                                        if (global_var(0) > -750) then begin
-                                            set_global_var(43, 1);
+                                        if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                            set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                         end
                                         else begin
-                                            if (global_var(0) > -1000) then begin
-                                                set_global_var(44, 1);
+                                            if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                                set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                             end
                                             else begin
-                                                set_global_var(45, 1);
+                                                set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                             end
                                         end
                                     end
@@ -305,58 +305,58 @@ begin
                         end
                     end
                 end
-                set_global_var(51, global_var(51) + -8);
+                set_global_var(GVAR_TOWN_REP_ALBUQUERQUE, global_var(GVAR_TOWN_REP_ALBUQUERQUE) + -8);
                 debug_msg("Added " + -8 + " to Town Rep");
             end
         end
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(5) > (2 * global_var(4))) or (global_var(3) == 1))) then begin
-            set_global_var(3, 1);
-            set_global_var(2, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
+            set_global_var(GVAR_BERSERKER_REPUTATION, 1);
+            set_global_var(GVAR_CHAMPION_REPUTATION, 0);
         end
-        set_global_var(0, global_var(0) + 0);
+        set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + 0);
         debug_msg("Player gains " + 0 + " Karma Points.");
-        set_global_var(37, 0);
-        set_global_var(38, 0);
-        set_global_var(39, 0);
-        set_global_var(40, 0);
-        set_global_var(41, 0);
-        set_global_var(42, 0);
-        set_global_var(43, 0);
-        set_global_var(44, 0);
-        set_global_var(45, 0);
-        if (global_var(0) >= 1000) then begin
-            set_global_var(37, 1);
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+        set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+        set_global_var(GVAR_KARMA_DEFENDER, 0);
+        set_global_var(GVAR_KARMA_WANDERER, 0);
+        set_global_var(GVAR_KARMA_BETRAYER, 0);
+        set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+        set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+            set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
         end
         else begin
-            if (global_var(0) >= 750) then begin
-                set_global_var(38, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
             end
             else begin
-                if (global_var(0) >= 500) then begin
-                    set_global_var(39, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                 end
                 else begin
-                    if (global_var(0) >= 250) then begin
-                        set_global_var(40, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                        set_global_var(GVAR_KARMA_DEFENDER, 1);
                     end
                     else begin
-                        if (global_var(0) > -250) then begin
-                            set_global_var(41, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                            set_global_var(GVAR_KARMA_WANDERER, 1);
                         end
                         else begin
-                            if (global_var(0) > -500) then begin
-                                set_global_var(42, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                set_global_var(GVAR_KARMA_BETRAYER, 1);
                             end
                             else begin
-                                if (global_var(0) > -750) then begin
-                                    set_global_var(43, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -1000) then begin
-                                        set_global_var(44, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                     end
                                     else begin
-                                        set_global_var(45, 1);
+                                        set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                     end
                                 end
                             end
@@ -366,17 +366,17 @@ begin
             end
         end
     end
-    set_global_var(867, global_var(867) bwor 262144);
+    set_global_var(GVAR_AQ_ALBUQ_DEATHS, global_var(GVAR_AQ_ALBUQ_DEATHS) bwor 262144);
 end
 
 procedure look_at_p_proc
 begin
     script_overrides;
     if (local_var(5) == 1) then begin
-        display_msg(message_str(1706, 150));
+        display_mstr(150);
     end
     else begin
-        display_msg(message_str(1706, 100));
+        display_mstr(100);
     end
 end
 
@@ -384,10 +384,10 @@ procedure description_p_proc
 begin
     script_overrides;
     if (local_var(5) == 1) then begin
-        display_msg(message_str(1706, 250));
+        display_mstr(250);
     end
     else begin
-        display_msg(message_str(1706, 200));
+        display_mstr(200);
     end
 end
 
@@ -399,14 +399,14 @@ procedure damage_p_proc
 begin
     if (source_obj == dude_obj) then begin
         set_local_var(4, 1);
-        set_global_var(864, 1);
+        set_global_var(GVAR_AQ_ENEMY_HUNTERS, 1);
     end
 end
 
 procedure map_enter_p_proc
 begin
-    critter_add_trait(self_obj, 1, 6, 258);
-    critter_add_trait(self_obj, 1, 5, 3);
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 258);
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_TOUGH_GUARD_NO_TALKING);
 end
 
 procedure combat_p_proc
@@ -415,7 +415,7 @@ begin
         probability := random(0, 12);
         if ((probability >= 12) and not(critter_is_fleeing(self_obj))) then begin
             if ((local_var(4) == 1) and obj_can_see_obj(self_obj, dude_obj)) then begin
-                float_msg(self_obj, message_str(1706, random(400, 402)), 2);
+                float_msg(self_obj, mstr(random(400, 402)), FLOAT_MSG_RED);
             end
         end
     end
@@ -427,7 +427,7 @@ begin
         set_local_var(6, local_var(6) bwand (-1 - 8192));
     end
     set_local_var(4, 1);
-    set_global_var(864, 1);
+    set_global_var(GVAR_AQ_ENEMY_HUNTERS, 1);
 end
 
 procedure Node999
@@ -437,120 +437,120 @@ end
 
 procedure Node500
 begin
-    gsay_reply(1706, 500);
-    giq_option(4, 1706, 501, Node600, 50);
-    if (((local_var(6) bwand 1) == 0) and (global_var(898) == 1)) then begin
-        giq_option(4, 1706, 502, Node550, 50);
+    Reply(500);
+    giq_option(4, NAME, 501, Node600, NEUTRAL_REACTION);
+    if (((local_var(6) bwand 1) == 0) and (global_var(GVAR_AQ_MUTANT_HUNTERS_CANDIDATE) == 1)) then begin
+        giq_option(4, NAME, 502, Node550, NEUTRAL_REACTION);
     end
-    giq_option(4, 1706, 503, Node999, 50);
-    giq_option(-3, 1706, 504, Node2700, 50);
+    giq_option(4, NAME, 503, Node999, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 504, Node2700, NEUTRAL_REACTION);
 end
 
 procedure Node550
 begin
-    gsay_reply(1706, 550);
+    Reply(550);
     set_local_var(6, local_var(6) bwor 1);
-    giq_option(4, 1706, 551, Node999, 50);
+    giq_option(4, NAME, 551, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node600
 begin
-    gsay_reply(1706, 600);
+    Reply(600);
     set_local_var(5, 1);
-    if (global_var(897) == 1) then begin
-        giq_option(4, 1706, 1555, Node2800, 50);
+    if (global_var(GVAR_AQ_QUEST_HOSPITAL_HERMAN) == 1) then begin
+        giq_option(4, NAME, 1555, Node2800, NEUTRAL_REACTION);
     end
-    if (((local_var(6) bwand 1) == 0) and (global_var(898) == 1)) then begin
-        giq_option(4, 1706, 601, Node550, 50);
+    if (((local_var(6) bwand 1) == 0) and (global_var(GVAR_AQ_MUTANT_HUNTERS_CANDIDATE) == 1)) then begin
+        giq_option(4, NAME, 601, Node550, NEUTRAL_REACTION);
     end
-    giq_option(4, 1706, 602, Node650, 50);
-    if (((local_var(6) bwand 512) == 0) and (global_var(898) == 3) and (global_var(882) == 0)) then begin
-        if (global_var(886) == 0) then begin
-            if ((global_var(1006) == 0) and (global_var(1007) == 0)) then begin
-                giq_option(4, 1706, 603, Node1600, 50);
+    giq_option(4, NAME, 602, Node650, NEUTRAL_REACTION);
+    if (((local_var(6) bwand 512) == 0) and (global_var(GVAR_AQ_MUTANT_HUNTERS_CANDIDATE) == 3) and (global_var(GVAR_AQ_QUEST_MTHUNT_HERMAN_QUEST) == 0)) then begin
+        if (global_var(GVAR_AQ_FRIEND_OF_THE_TRIBE) == 0) then begin
+            if ((global_var(GVAR_AQE_MESAGE_FROM_ELISA) == 0) and (global_var(GVAR_AQE_MESAGE_FROM_ARAN) == 0)) then begin
+                giq_option(4, NAME, 603, Node1600, NEUTRAL_REACTION);
             end
             else begin
-                giq_option(4, 1706, 603, Node2950, 50);
+                giq_option(4, NAME, 603, Node2950, NEUTRAL_REACTION);
             end
         end
         else begin
-            giq_option(4, 1706, 603, Node2900, 50);
+            giq_option(4, NAME, 603, Node2900, NEUTRAL_REACTION);
         end
     end
-    giq_option(4, 1706, 604, Node999, 50);
+    giq_option(4, NAME, 604, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node650
 begin
-    gsay_reply(1706, 650);
-    if (((local_var(6) bwand 2) == 0) and (((global_var(866) bwand 8) != 0) or ((global_var(866) bwand 1) != 0)) and not((global_var(865) bwand 4) != 0) and not((global_var(867) bwand 256) != 0) and not((global_var(867) bwand 128) != 0)) then begin
-        giq_option(4, 1706, 651, Node700, 50);
+    Reply(650);
+    if (((local_var(6) bwand 2) == 0) and (((global_var(GVAR_AQ_ALBUQ_SPEECHES) bwand 8) != 0) or ((global_var(GVAR_AQ_ALBUQ_SPEECHES) bwand 1) != 0)) and not((global_var(GVAR_AQ_ALBUQ_EVENTS) bwand 4) != 0) and not((global_var(GVAR_AQ_ALBUQ_DEATHS) bwand 256) != 0) and not((global_var(GVAR_AQ_ALBUQ_DEATHS) bwand 128) != 0)) then begin
+        giq_option(4, NAME, 651, Node700, NEUTRAL_REACTION);
     end
-    if (((local_var(6) bwand 16) == 0) and ((global_var(866) bwand 65536) != 0) and not((global_var(867) bwand 524288) != 0) and not((local_var(6) bwand 8192) != 0) and (global_var(872) != 2) and (map_var(6) == 0)) then begin
-        giq_option(4, 1706, 652, Node950, 50);
+    if (((local_var(6) bwand 16) == 0) and ((global_var(GVAR_AQ_ALBUQ_SPEECHES) bwand 65536) != 0) and not((global_var(GVAR_AQ_ALBUQ_DEATHS) bwand 524288) != 0) and not((local_var(6) bwand 8192) != 0) and (global_var(GVAR_AQ_QUEST_CHUCK_SWEETHEARTS) != 2) and (map_var(6) == 0)) then begin
+        giq_option(4, NAME, 652, Node950, NEUTRAL_REACTION);
     end
-    if (((local_var(6) bwand 32) == 0) and ((global_var(866) bwand 65536) != 0) and not((global_var(867) bwand 524288) != 0) and not((local_var(6) bwand 8192) != 0) and (global_var(872) != 2) and (map_var(6) == 0)) then begin
-        giq_option(4, 1706, 653, Node1050, 50);
+    if (((local_var(6) bwand 32) == 0) and ((global_var(GVAR_AQ_ALBUQ_SPEECHES) bwand 65536) != 0) and not((global_var(GVAR_AQ_ALBUQ_DEATHS) bwand 524288) != 0) and not((local_var(6) bwand 8192) != 0) and (global_var(GVAR_AQ_QUEST_CHUCK_SWEETHEARTS) != 2) and (map_var(6) == 0)) then begin
+        giq_option(4, NAME, 653, Node1050, NEUTRAL_REACTION);
     end
-    if (((local_var(6) bwand 128) == 0) and ((global_var(866) bwand 16384) != 0) and not((global_var(867) bwand 131072) != 0)) then begin
-        giq_option(4, 1706, 654, Node1400, 50);
+    if (((local_var(6) bwand 128) == 0) and ((global_var(GVAR_AQ_ALBUQ_SPEECHES) bwand 16384) != 0) and not((global_var(GVAR_AQ_ALBUQ_DEATHS) bwand 131072) != 0)) then begin
+        giq_option(4, NAME, 654, Node1400, NEUTRAL_REACTION);
     end
-    if (((local_var(6) bwand 64) == 0) and ((global_var(866) bwand 8192) != 0) and not((global_var(867) bwand 1048576) != 0)) then begin
-        giq_option(4, 1706, 655, Node1350, 50);
+    if (((local_var(6) bwand 64) == 0) and ((global_var(GVAR_AQ_ALBUQ_SPEECHES) bwand 8192) != 0) and not((global_var(GVAR_AQ_ALBUQ_DEATHS) bwand 1048576) != 0)) then begin
+        giq_option(4, NAME, 655, Node1350, NEUTRAL_REACTION);
     end
-    if (((local_var(6) bwand 256) == 0) and ((global_var(866) bwand 32768) != 0) and not((global_var(867) bwand 16) != 0)) then begin
-        giq_option(4, 1706, 656, Node1500, 50);
+    if (((local_var(6) bwand 256) == 0) and ((global_var(GVAR_AQ_ALBUQ_SPEECHES) bwand 32768) != 0) and not((global_var(GVAR_AQ_ALBUQ_DEATHS) bwand 16) != 0)) then begin
+        giq_option(4, NAME, 656, Node1500, NEUTRAL_REACTION);
     end
-    giq_option(4, 1706, 657, Node999, 50);
+    giq_option(4, NAME, 657, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node700
 begin
-    gsay_reply(1706, 700);
+    Reply(700);
     set_local_var(6, local_var(6) bwor 2);
-    set_global_var(0, global_var(0) - 20);
-    set_global_var(37, 0);
-    set_global_var(38, 0);
-    set_global_var(39, 0);
-    set_global_var(40, 0);
-    set_global_var(41, 0);
-    set_global_var(42, 0);
-    set_global_var(43, 0);
-    set_global_var(44, 0);
-    set_global_var(45, 0);
-    if (global_var(0) >= 1000) then begin
-        set_global_var(37, 1);
+    set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) - 20);
+    set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+    set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+    set_global_var(GVAR_KARMA_DEFENDER, 0);
+    set_global_var(GVAR_KARMA_WANDERER, 0);
+    set_global_var(GVAR_KARMA_BETRAYER, 0);
+    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+    set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+    set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+    if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
     end
     else begin
-        if (global_var(0) >= 750) then begin
-            set_global_var(38, 1);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+            set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
         end
         else begin
-            if (global_var(0) >= 500) then begin
-                set_global_var(39, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
             end
             else begin
-                if (global_var(0) >= 250) then begin
-                    set_global_var(40, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                    set_global_var(GVAR_KARMA_DEFENDER, 1);
                 end
                 else begin
-                    if (global_var(0) > -250) then begin
-                        set_global_var(41, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                        set_global_var(GVAR_KARMA_WANDERER, 1);
                     end
                     else begin
-                        if (global_var(0) > -500) then begin
-                            set_global_var(42, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                            set_global_var(GVAR_KARMA_BETRAYER, 1);
                         end
                         else begin
-                            if (global_var(0) > -750) then begin
-                                set_global_var(43, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                             end
                             else begin
-                                if (global_var(0) > -1000) then begin
-                                    set_global_var(44, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                    set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                 end
                                 else begin
-                                    set_global_var(45, 1);
+                                    set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                 end
                             end
                         end
@@ -559,535 +559,535 @@ begin
             end
         end
     end
-    set_global_var(865, global_var(865) bwor 2);
-    set_global_var(51, global_var(51) - 5);
+    set_global_var(GVAR_AQ_ALBUQ_EVENTS, global_var(GVAR_AQ_ALBUQ_EVENTS) bwor 2);
+    set_global_var(GVAR_TOWN_REP_ALBUQUERQUE, global_var(GVAR_TOWN_REP_ALBUQUERQUE) - 5);
     if ((local_var(6) bwand 4) == 0) then begin
-        giq_option(4, 1706, 701, Node750, 50);
+        giq_option(4, NAME, 701, Node750, NEUTRAL_REACTION);
     end
     if ((local_var(6) bwand 8) == 0) then begin
-        giq_option(4, 1706, 702, Node800, 50);
+        giq_option(4, NAME, 702, Node800, NEUTRAL_REACTION);
     end
-    giq_option(4, 1706, 703, Node650, 50);
-    giq_option(4, 1706, 704, Node999, 50);
+    giq_option(4, NAME, 703, Node650, NEUTRAL_REACTION);
+    giq_option(4, NAME, 704, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node750
 begin
-    gsay_reply(1706, 750);
+    Reply(750);
     set_local_var(6, local_var(6) bwor 4);
-    giq_option(4, 1706, 751, Node850, 50);
+    giq_option(4, NAME, 751, Node850, NEUTRAL_REACTION);
     if ((local_var(6) bwand 8) == 0) then begin
-        giq_option(4, 1706, 752, Node800, 50);
+        giq_option(4, NAME, 752, Node800, NEUTRAL_REACTION);
     end
-    giq_option(4, 1706, 753, Node650, 50);
-    giq_option(4, 1706, 754, Node999, 50);
+    giq_option(4, NAME, 753, Node650, NEUTRAL_REACTION);
+    giq_option(4, NAME, 754, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node800
 begin
-    gsay_reply(1706, 800);
+    Reply(800);
     set_local_var(6, local_var(6) bwor 8);
     if ((local_var(6) bwand 4) == 0) then begin
-        giq_option(4, 1706, 801, Node750, 50);
+        giq_option(4, NAME, 801, Node750, NEUTRAL_REACTION);
     end
-    giq_option(4, 1706, 802, Node900, 50);
-    giq_option(4, 1706, 803, Node650, 50);
-    giq_option(4, 1706, 804, Node999, 50);
+    giq_option(4, NAME, 802, Node900, NEUTRAL_REACTION);
+    giq_option(4, NAME, 803, Node650, NEUTRAL_REACTION);
+    giq_option(4, NAME, 804, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node850
 begin
-    gsay_reply(1706, 850);
-    set_global_var(865, global_var(865) bwand (-1 - 2));
-    giq_option(4, 1706, 851, Node998, 50);
+    Reply(850);
+    set_global_var(GVAR_AQ_ALBUQ_EVENTS, global_var(GVAR_AQ_ALBUQ_EVENTS) bwand (-1 - 2));
+    giq_option(4, NAME, 851, Node998, NEUTRAL_REACTION);
 end
 
 procedure Node900
 begin
-    gsay_reply(1706, 900);
-    giq_option(4, 1706, 901, Node998, 50);
+    Reply(900);
+    giq_option(4, NAME, 901, Node998, NEUTRAL_REACTION);
 end
 
 procedure Node950
 begin
-    gsay_reply(1706, 950);
+    Reply(950);
     set_local_var(6, local_var(6) bwor 16);
-    giq_option(4, 1706, 951, Node650, 50);
-    giq_option(4, 1706, 952, Node1000, 50);
-    giq_option(4, 1706, 953, Node999, 50);
+    giq_option(4, NAME, 951, Node650, NEUTRAL_REACTION);
+    giq_option(4, NAME, 952, Node1000, NEUTRAL_REACTION);
+    giq_option(4, NAME, 953, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node1000
 begin
-    gsay_reply(1706, 1000);
-    giq_option(4, 1706, 1001, Node999, 50);
-    giq_option(4, 1706, 1002, Node998, 50);
+    Reply(1000);
+    giq_option(4, NAME, 1001, Node999, NEUTRAL_REACTION);
+    giq_option(4, NAME, 1002, Node998, NEUTRAL_REACTION);
 end
 
 procedure Node1050
 begin
-    gsay_reply(1706, 1050);
+    Reply(1050);
     set_local_var(6, local_var(6) bwor 32);
     if (obj_is_carrying_obj_pid(dude_obj, 473)) then begin
-        if (get_critter_stat(dude_obj, 6) == 10) then begin
-            giq_option(4, 1706, 1051, Node1150, 50);
+        if (dude_luck == 10) then begin
+            giq_option(4, NAME, 1051, Node1150, NEUTRAL_REACTION);
         end
         else begin
-            giq_option(4, 1706, 1051, Node1100, 50);
+            giq_option(4, NAME, 1051, Node1100, NEUTRAL_REACTION);
         end
     end
-    if (has_skill(dude_obj, 14) > 85) then begin
-        if (get_critter_stat(dude_obj, 6) == 10) then begin
-            giq_option(4, 1706, 1052, Node1300, 50);
+    if (dude_skill(SKILL_SPEECH) > 85) then begin
+        if (dude_luck == 10) then begin
+            giq_option(4, NAME, 1052, Node1300, NEUTRAL_REACTION);
         end
         else begin
-            giq_option(4, 1706, 1052, Node1250, 50);
+            giq_option(4, NAME, 1052, Node1250, NEUTRAL_REACTION);
         end
     end
-    giq_option(4, 1706, 1053, Node999, 50);
+    giq_option(4, NAME, 1053, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node1100
 begin
-    gsay_reply(1706, 1100);
+    Reply(1100);
     set_local_var(6, local_var(6) bwor 8192);
-    set_global_var(865, global_var(865) bwor 16777216);
-    giq_option(4, 1706, 1101, Node1200, 50);
-    giq_option(4, 1706, 1102, Node650, 50);
-    giq_option(4, 1706, 1103, Node999, 50);
+    set_global_var(GVAR_AQ_ALBUQ_EVENTS, global_var(GVAR_AQ_ALBUQ_EVENTS) bwor 16777216);
+    giq_option(4, NAME, 1101, Node1200, NEUTRAL_REACTION);
+    giq_option(4, NAME, 1102, Node650, NEUTRAL_REACTION);
+    giq_option(4, NAME, 1103, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node1150
 begin
-    gsay_reply(1706, 1150);
+    Reply(1150);
     set_local_var(6, local_var(6) bwor 8192);
-    set_global_var(865, global_var(865) bwor 16777216);
-    giq_option(4, 1706, 1151, Node1200, 50);
-    giq_option(4, 1706, 1152, Node650, 50);
-    giq_option(4, 1706, 1153, Node999, 50);
+    set_global_var(GVAR_AQ_ALBUQ_EVENTS, global_var(GVAR_AQ_ALBUQ_EVENTS) bwor 16777216);
+    giq_option(4, NAME, 1151, Node1200, NEUTRAL_REACTION);
+    giq_option(4, NAME, 1152, Node650, NEUTRAL_REACTION);
+    giq_option(4, NAME, 1153, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node1200
 begin
-    gsay_reply(1706, 1200);
-    giq_option(4, 1706, 1201, Node900, 50);
-    giq_option(4, 1706, 1202, Node650, 50);
-    giq_option(4, 1706, 1203, Node999, 50);
+    Reply(1200);
+    giq_option(4, NAME, 1201, Node900, NEUTRAL_REACTION);
+    giq_option(4, NAME, 1202, Node650, NEUTRAL_REACTION);
+    giq_option(4, NAME, 1203, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node1250
 begin
-    gsay_reply(1706, 1250);
+    Reply(1250);
     set_local_var(6, local_var(6) bwor 8192);
-    set_global_var(865, global_var(865) bwor 16777216);
-    giq_option(4, 1706, 1251, Node1200, 50);
-    giq_option(4, 1706, 1252, Node650, 50);
-    giq_option(4, 1706, 1253, Node999, 50);
+    set_global_var(GVAR_AQ_ALBUQ_EVENTS, global_var(GVAR_AQ_ALBUQ_EVENTS) bwor 16777216);
+    giq_option(4, NAME, 1251, Node1200, NEUTRAL_REACTION);
+    giq_option(4, NAME, 1252, Node650, NEUTRAL_REACTION);
+    giq_option(4, NAME, 1253, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node1300
 begin
-    gsay_reply(1706, 1300);
+    Reply(1300);
     set_local_var(6, local_var(6) bwor 8192);
-    set_global_var(865, global_var(865) bwor 16777216);
-    giq_option(4, 1706, 1301, Node1200, 50);
-    giq_option(4, 1706, 1302, Node650, 50);
-    giq_option(4, 1706, 1303, Node999, 50);
+    set_global_var(GVAR_AQ_ALBUQ_EVENTS, global_var(GVAR_AQ_ALBUQ_EVENTS) bwor 16777216);
+    giq_option(4, NAME, 1301, Node1200, NEUTRAL_REACTION);
+    giq_option(4, NAME, 1302, Node650, NEUTRAL_REACTION);
+    giq_option(4, NAME, 1303, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node1350
 begin
-    gsay_reply(1706, 1350);
+    Reply(1350);
     set_local_var(6, local_var(6) bwor 64);
-    giq_option(4, 1706, 1351, Node650, 50);
-    giq_option(4, 1706, 1352, Node1450, 50);
-    giq_option(4, 1706, 1353, Node999, 50);
+    giq_option(4, NAME, 1351, Node650, NEUTRAL_REACTION);
+    giq_option(4, NAME, 1352, Node1450, NEUTRAL_REACTION);
+    giq_option(4, NAME, 1353, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node1400
 begin
-    gsay_reply(1706, 1400);
+    Reply(1400);
     set_local_var(6, local_var(6) bwor 128);
     set_map_var(0, map_var(0) bwor 256);
-    giq_option(4, 1706, 1401, Node650, 50);
-    giq_option(4, 1706, 1402, Node1450, 50);
-    giq_option(4, 1706, 1403, Node999, 50);
+    giq_option(4, NAME, 1401, Node650, NEUTRAL_REACTION);
+    giq_option(4, NAME, 1402, Node1450, NEUTRAL_REACTION);
+    giq_option(4, NAME, 1403, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node1450
 begin
-    gsay_reply(1706, 1450);
-    giq_option(4, 1706, 1451, Node999, 50);
-    giq_option(4, 1706, 1452, Node998, 50);
+    Reply(1450);
+    giq_option(4, NAME, 1451, Node999, NEUTRAL_REACTION);
+    giq_option(4, NAME, 1452, Node998, NEUTRAL_REACTION);
 end
 
 procedure Node1500
 begin
-    gsay_reply(1706, 1500);
+    Reply(1500);
     set_local_var(6, local_var(6) bwor 256);
-    giq_option(4, 1706, 1501, Node650, 50);
-    giq_option(4, 1706, 1502, Node1450, 50);
-    giq_option(4, 1706, 1503, Node999, 50);
+    giq_option(4, NAME, 1501, Node650, NEUTRAL_REACTION);
+    giq_option(4, NAME, 1502, Node1450, NEUTRAL_REACTION);
+    giq_option(4, NAME, 1503, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node1550
 begin
-    gsay_reply(1706, 1550);
-    if (global_var(897) == 1) then begin
-        giq_option(4, 1706, 1555, Node2800, 50);
+    Reply(1550);
+    if (global_var(GVAR_AQ_QUEST_HOSPITAL_HERMAN) == 1) then begin
+        giq_option(4, NAME, 1555, Node2800, NEUTRAL_REACTION);
     end
-    if (((local_var(6) bwand 1) == 0) and (global_var(898) == 1)) then begin
-        giq_option(4, 1706, 1551, Node550, 50);
+    if (((local_var(6) bwand 1) == 0) and (global_var(GVAR_AQ_MUTANT_HUNTERS_CANDIDATE) == 1)) then begin
+        giq_option(4, NAME, 1551, Node550, NEUTRAL_REACTION);
     end
-    giq_option(4, 1706, 1552, Node650, 50);
-    if (((local_var(6) bwand 512) == 0) and (global_var(898) == 3) and (global_var(882) == 0)) then begin
-        if (global_var(886) == 0) then begin
-            if ((global_var(1006) == 0) and (global_var(1007) == 0)) then begin
-                giq_option(4, 1706, 1553, Node1600, 50);
+    giq_option(4, NAME, 1552, Node650, NEUTRAL_REACTION);
+    if (((local_var(6) bwand 512) == 0) and (global_var(GVAR_AQ_MUTANT_HUNTERS_CANDIDATE) == 3) and (global_var(GVAR_AQ_QUEST_MTHUNT_HERMAN_QUEST) == 0)) then begin
+        if (global_var(GVAR_AQ_FRIEND_OF_THE_TRIBE) == 0) then begin
+            if ((global_var(GVAR_AQE_MESAGE_FROM_ELISA) == 0) and (global_var(GVAR_AQE_MESAGE_FROM_ARAN) == 0)) then begin
+                giq_option(4, NAME, 1553, Node1600, NEUTRAL_REACTION);
             end
             else begin
-                giq_option(4, 1706, 1553, Node2950, 50);
+                giq_option(4, NAME, 1553, Node2950, NEUTRAL_REACTION);
             end
         end
         else begin
-            giq_option(4, 1706, 1553, Node2900, 50);
+            giq_option(4, NAME, 1553, Node2900, NEUTRAL_REACTION);
         end
     end
     if (((local_var(6) bwand 4096) != 0) and ((local_var(6) bwand 2048) == 0)) then begin
         if ((game_time - local_var(7)) > (60 * 10 * 60)) then begin
-            giq_option(4, 1706, 1554, Node2100, 50);
+            giq_option(4, NAME, 1554, Node2100, NEUTRAL_REACTION);
         end
         else begin
-            giq_option(4, 1706, 1554, Node2050, 50);
+            giq_option(4, NAME, 1554, Node2050, NEUTRAL_REACTION);
         end
     end
-    giq_option(4, 1706, 1556, Node999, 50);
-    giq_option(-3, 1706, 1557, Node2700, 50);
+    giq_option(4, NAME, 1556, Node999, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 1557, Node2700, NEUTRAL_REACTION);
 end
 
 procedure Node1600
 begin
-    gsay_reply(1706, 1600);
+    Reply(1600);
     set_local_var(6, local_var(6) bwor 512);
-    giq_option(4, 1706, 1601, Node1650, 50);
+    giq_option(4, NAME, 1601, Node1650, NEUTRAL_REACTION);
 end
 
 procedure Node1650
 begin
-    gsay_reply(1706, 1650);
-    giq_option(4, 1706, 1651, Node1700, 50);
+    Reply(1650);
+    giq_option(4, NAME, 1651, Node1700, NEUTRAL_REACTION);
 end
 
 procedure Node1700
 begin
-    gsay_reply(1706, 1700);
-    giq_option(4, 1706, 1701, Node1750, 50);
-    giq_option(4, 1706, 1702, Herman_Quest_Start, 50);
-    giq_option(4, 1706, 1703, Node1800, 50);
+    Reply(1700);
+    giq_option(4, NAME, 1701, Node1750, NEUTRAL_REACTION);
+    giq_option(4, NAME, 1702, Herman_Quest_Start, NEUTRAL_REACTION);
+    giq_option(4, NAME, 1703, Node1800, NEUTRAL_REACTION);
 end
 
 procedure Node1750
 begin
-    gsay_reply(1706, 1750);
-    giq_option(4, 1706, 1751, Herman_Quest_Start, 50);
-    giq_option(4, 1706, 1752, Node1800, 50);
+    Reply(1750);
+    giq_option(4, NAME, 1751, Herman_Quest_Start, NEUTRAL_REACTION);
+    giq_option(4, NAME, 1752, Node1800, NEUTRAL_REACTION);
 end
 
 procedure Node1800
 begin
-    gsay_reply(1706, 1800);
-    giq_option(4, 1706, 1801, Herman_Quest_Start, 50);
-    giq_option(4, 1706, 1802, Node1850, 50);
+    Reply(1800);
+    giq_option(4, NAME, 1801, Herman_Quest_Start, NEUTRAL_REACTION);
+    giq_option(4, NAME, 1802, Node1850, NEUTRAL_REACTION);
 end
 
 procedure Node1850
 begin
-    gsay_reply(1706, 1850);
-    set_global_var(898, 2);
-    if (global_var(881) != 2) then begin
-        set_global_var(881, 0);
+    Reply(1850);
+    set_global_var(GVAR_AQ_MUTANT_HUNTERS_CANDIDATE, 2);
+    if (global_var(GVAR_AQ_QUEST_MTHUNT_TULSA_QUEST) != 2) then begin
+        set_global_var(GVAR_AQ_QUEST_MTHUNT_TULSA_QUEST, 0);
     end
-    if (global_var(882) != 4) then begin
-        set_global_var(882, 0);
+    if (global_var(GVAR_AQ_QUEST_MTHUNT_HERMAN_QUEST) != 4) then begin
+        set_global_var(GVAR_AQ_QUEST_MTHUNT_HERMAN_QUEST, 0);
     end
-    if (global_var(883) != 2) then begin
-        set_global_var(883, 0);
+    if (global_var(GVAR_AQ_QUEST_MTHUNT_DUTCH_QUEST) != 2) then begin
+        set_global_var(GVAR_AQ_QUEST_MTHUNT_DUTCH_QUEST, 0);
     end
     set_map_var(0, map_var(0) bwor 64);
-    giq_option(4, 1706, 1851, Node999, 50);
-    giq_option(4, 1706, 1852, Node998, 50);
+    giq_option(4, NAME, 1851, Node999, NEUTRAL_REACTION);
+    giq_option(4, NAME, 1852, Node998, NEUTRAL_REACTION);
 end
 
 procedure Node1900
 begin
-    if (get_critter_stat(dude_obj, 34) == 0) then begin
-        gsay_reply(1706, 1900);
+    if (dude_gender == 0) then begin
+        Reply(1900);
     end
     else begin
-        gsay_reply(1706, 1901);
+        Reply(1901);
     end
-    set_global_var(882, 4);
-    if ((party_member_obj(16777814) != 0) and ((global_var(910) bwand 128) == 0)) then begin
-        set_global_var(1019, global_var(1019) bwor 524288);
+    set_global_var(GVAR_AQ_QUEST_MTHUNT_HERMAN_QUEST, 4);
+    if ((party_member_obj(16777814) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 128) == 0)) then begin
+        set_global_var(GVAR_PTR_LYSTRA_REACTIONS_1, global_var(GVAR_PTR_LYSTRA_REACTIONS_1) bwor 524288);
     end
-    if ((global_var(1017) bwand 65536) != 0) then begin
-        set_global_var(1010, game_time);
+    if ((global_var(GVAR_AQE_DIRECTIONS_SETUP) bwand 65536) != 0) then begin
+        set_global_var(GVAR_AQE_EMPEROR_TIME_LIMIT, game_time);
     end
     give_exp_points(1000);
-    display_msg(message_str(14, 100) + 1000 + message_str(14, 101));
+    display_msg(g_mstr(100) + 1000 + g_mstr(101));
     set_local_var(6, local_var(6) bwor 1024);
-    giq_option(4, 1706, 1902, Node999, 50);
-    giq_option(-3, 1706, 1903, Node999, 50);
+    giq_option(4, NAME, 1902, Node999, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 1903, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node1950
 begin
-    if (get_critter_stat(dude_obj, 34) == 0) then begin
-        gsay_reply(1706, 1950);
+    if (dude_gender == 0) then begin
+        Reply(1950);
     end
     else begin
-        gsay_reply(1706, 1951);
+        Reply(1951);
     end
-    set_global_var(882, 4);
-    if ((global_var(1017) bwand 65536) != 0) then begin
-        set_global_var(1010, game_time);
+    set_global_var(GVAR_AQ_QUEST_MTHUNT_HERMAN_QUEST, 4);
+    if ((global_var(GVAR_AQE_DIRECTIONS_SETUP) bwand 65536) != 0) then begin
+        set_global_var(GVAR_AQE_EMPEROR_TIME_LIMIT, game_time);
     end
-    if ((party_member_obj(16777814) != 0) and ((global_var(910) bwand 128) == 0)) then begin
-        set_global_var(1019, global_var(1019) bwor 524288);
+    if ((party_member_obj(16777814) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 128) == 0)) then begin
+        set_global_var(GVAR_PTR_LYSTRA_REACTIONS_1, global_var(GVAR_PTR_LYSTRA_REACTIONS_1) bwor 524288);
     end
     give_exp_points(1000);
-    display_msg(message_str(14, 100) + 1000 + message_str(14, 101));
+    display_msg(g_mstr(100) + 1000 + g_mstr(101));
     set_local_var(6, local_var(6) bwor 1024);
     set_map_var(0, map_var(0) bwor 128);
-    giq_option(4, 1706, 1952, Node2000, 50);
-    giq_option(4, 1706, 1953, Node999, 50);
-    giq_option(-3, 1706, 1954, Node999, 50);
+    giq_option(4, NAME, 1952, Node2000, NEUTRAL_REACTION);
+    giq_option(4, NAME, 1953, Node999, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 1954, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node2000
 begin
-    gsay_reply(1706, 2000);
-    giq_option(1, 1706, 2001, Node999, 50);
+    Reply(2000);
+    giq_option(1, NAME, 2001, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node2050
 begin
-    gsay_reply(1706, 2050);
-    giq_option(4, 1706, 2051, Node650, 50);
-    giq_option(4, 1706, 2052, Node999, 50);
+    Reply(2050);
+    giq_option(4, NAME, 2051, Node650, NEUTRAL_REACTION);
+    giq_option(4, NAME, 2052, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node2100
 begin
-    gsay_reply(1706, 2100);
+    Reply(2100);
     set_local_var(6, local_var(6) bwor 2048);
-    giq_option(6, 1706, 2102, Node2750, 50);
-    giq_option(4, 1706, 2101, Node650, 50);
-    giq_option(4, 1706, 2103, Node999, 50);
-    giq_option(4, 1706, 2104, Node998, 50);
+    giq_option(6, NAME, 2102, Node2750, NEUTRAL_REACTION);
+    giq_option(4, NAME, 2101, Node650, NEUTRAL_REACTION);
+    giq_option(4, NAME, 2103, Node999, NEUTRAL_REACTION);
+    giq_option(4, NAME, 2104, Node998, NEUTRAL_REACTION);
 end
 
 procedure Node2150
 begin
-    gsay_reply(1706, 2150);
-    if (global_var(897) == 1) then begin
-        giq_option(4, 1706, 2154, Node2800, 50);
+    Reply(2150);
+    if (global_var(GVAR_AQ_QUEST_HOSPITAL_HERMAN) == 1) then begin
+        giq_option(4, NAME, 2154, Node2800, NEUTRAL_REACTION);
     end
-    giq_option(4, 1706, 2151, Node650, 50);
-    giq_option(4, 1706, 2152, Node2200, 50);
+    giq_option(4, NAME, 2151, Node650, NEUTRAL_REACTION);
+    giq_option(4, NAME, 2152, Node2200, NEUTRAL_REACTION);
     if (((local_var(6) bwand 4096) != 0) and ((local_var(6) bwand 2048) == 0)) then begin
         if ((game_time - local_var(7)) > (60 * 10 * 60)) then begin
-            giq_option(4, 1706, 2153, Node2100, 50);
+            giq_option(4, NAME, 2153, Node2100, NEUTRAL_REACTION);
         end
         else begin
-            giq_option(4, 1706, 2153, Node2050, 50);
+            giq_option(4, NAME, 2153, Node2050, NEUTRAL_REACTION);
         end
     end
-    giq_option(4, 1706, 2155, Node999, 50);
-    giq_option(-3, 1706, 2156, Node999, 50);
+    giq_option(4, NAME, 2155, Node999, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 2156, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node2200
 begin
-    gsay_reply(1706, 2200);
-    giq_option(4, 1706, 2201, Node2250, 50);
-    if ((global_var(867) bwand 1) != 0) then begin
-        giq_option(4, 1706, 2202, Node2350, 50);
+    Reply(2200);
+    giq_option(4, NAME, 2201, Node2250, NEUTRAL_REACTION);
+    if ((global_var(GVAR_AQ_ALBUQ_DEATHS) bwand 1) != 0) then begin
+        giq_option(4, NAME, 2202, Node2350, NEUTRAL_REACTION);
     end
     else begin
-        giq_option(4, 1706, 2202, Node2300, 50);
+        giq_option(4, NAME, 2202, Node2300, NEUTRAL_REACTION);
     end
-    giq_option(4, 1706, 2203, Node2450, 50);
-    giq_option(6, 1706, 2204, Node2400, 50);
-    giq_option(4, 1706, 2205, Node999, 50);
+    giq_option(4, NAME, 2203, Node2450, NEUTRAL_REACTION);
+    giq_option(6, NAME, 2204, Node2400, NEUTRAL_REACTION);
+    giq_option(4, NAME, 2205, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node2250
 begin
-    gsay_reply(1706, 2250);
-    if ((global_var(867) bwand 1) != 0) then begin
-        giq_option(4, 1706, 2251, Node2350, 50);
+    Reply(2250);
+    if ((global_var(GVAR_AQ_ALBUQ_DEATHS) bwand 1) != 0) then begin
+        giq_option(4, NAME, 2251, Node2350, NEUTRAL_REACTION);
     end
     else begin
-        giq_option(4, 1706, 2251, Node2300, 50);
+        giq_option(4, NAME, 2251, Node2300, NEUTRAL_REACTION);
     end
-    giq_option(4, 1706, 2252, Node2450, 50);
-    giq_option(6, 1706, 2253, Node2400, 50);
-    giq_option(4, 1706, 2254, Node999, 50);
+    giq_option(4, NAME, 2252, Node2450, NEUTRAL_REACTION);
+    giq_option(6, NAME, 2253, Node2400, NEUTRAL_REACTION);
+    giq_option(4, NAME, 2254, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node2300
 begin
-    gsay_reply(1706, 2300);
-    giq_option(4, 1706, 2301, Node2250, 50);
-    giq_option(4, 1706, 2302, Node2450, 50);
-    giq_option(6, 1706, 2303, Node2400, 50);
-    giq_option(4, 1706, 2304, Node999, 50);
+    Reply(2300);
+    giq_option(4, NAME, 2301, Node2250, NEUTRAL_REACTION);
+    giq_option(4, NAME, 2302, Node2450, NEUTRAL_REACTION);
+    giq_option(6, NAME, 2303, Node2400, NEUTRAL_REACTION);
+    giq_option(4, NAME, 2304, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node2350
 begin
-    gsay_reply(1706, 2350);
-    giq_option(4, 1706, 2351, Node2250, 50);
-    giq_option(4, 1706, 2352, Node2450, 50);
-    giq_option(6, 1706, 2353, Node2400, 50);
-    giq_option(4, 1706, 2354, Node999, 50);
+    Reply(2350);
+    giq_option(4, NAME, 2351, Node2250, NEUTRAL_REACTION);
+    giq_option(4, NAME, 2352, Node2450, NEUTRAL_REACTION);
+    giq_option(6, NAME, 2353, Node2400, NEUTRAL_REACTION);
+    giq_option(4, NAME, 2354, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node2400
 begin
-    gsay_reply(1706, 2400);
-    giq_option(4, 1706, 2401, Node2250, 50);
-    if ((global_var(867) bwand 1) != 0) then begin
-        giq_option(4, 1706, 2402, Node2350, 50);
+    Reply(2400);
+    giq_option(4, NAME, 2401, Node2250, NEUTRAL_REACTION);
+    if ((global_var(GVAR_AQ_ALBUQ_DEATHS) bwand 1) != 0) then begin
+        giq_option(4, NAME, 2402, Node2350, NEUTRAL_REACTION);
     end
     else begin
-        giq_option(4, 1706, 2402, Node2300, 50);
+        giq_option(4, NAME, 2402, Node2300, NEUTRAL_REACTION);
     end
-    giq_option(4, 1706, 2403, Node2450, 50);
-    giq_option(4, 1706, 2404, Node999, 50);
+    giq_option(4, NAME, 2403, Node2450, NEUTRAL_REACTION);
+    giq_option(4, NAME, 2404, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node2450
 begin
-    gsay_reply(1706, 2450);
-    giq_option(4, 1706, 2451, Node2500, 50);
-    giq_option(4, 1706, 2452, Node2550, 50);
-    giq_option(4, 1706, 2453, Node2600, 50);
-    giq_option(4, 1706, 2454, Node2650, 50);
-    giq_option(4, 1706, 2455, Node2200, 50);
-    giq_option(4, 1706, 2456, Node999, 50);
+    Reply(2450);
+    giq_option(4, NAME, 2451, Node2500, NEUTRAL_REACTION);
+    giq_option(4, NAME, 2452, Node2550, NEUTRAL_REACTION);
+    giq_option(4, NAME, 2453, Node2600, NEUTRAL_REACTION);
+    giq_option(4, NAME, 2454, Node2650, NEUTRAL_REACTION);
+    giq_option(4, NAME, 2455, Node2200, NEUTRAL_REACTION);
+    giq_option(4, NAME, 2456, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node2500
 begin
-    gsay_reply(1706, 2500);
-    giq_option(4, 1706, 2501, Node2550, 50);
-    giq_option(4, 1706, 2502, Node2600, 50);
-    giq_option(4, 1706, 2503, Node2650, 50);
-    giq_option(4, 1706, 2504, Node2200, 50);
-    giq_option(4, 1706, 2505, Node999, 50);
+    Reply(2500);
+    giq_option(4, NAME, 2501, Node2550, NEUTRAL_REACTION);
+    giq_option(4, NAME, 2502, Node2600, NEUTRAL_REACTION);
+    giq_option(4, NAME, 2503, Node2650, NEUTRAL_REACTION);
+    giq_option(4, NAME, 2504, Node2200, NEUTRAL_REACTION);
+    giq_option(4, NAME, 2505, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node2550
 begin
-    gsay_reply(1706, 2550);
-    giq_option(4, 1706, 2551, Node2500, 50);
-    giq_option(4, 1706, 2552, Node2600, 50);
-    giq_option(4, 1706, 2553, Node2650, 50);
-    giq_option(4, 1706, 2554, Node2200, 50);
-    giq_option(4, 1706, 2555, Node999, 50);
+    Reply(2550);
+    giq_option(4, NAME, 2551, Node2500, NEUTRAL_REACTION);
+    giq_option(4, NAME, 2552, Node2600, NEUTRAL_REACTION);
+    giq_option(4, NAME, 2553, Node2650, NEUTRAL_REACTION);
+    giq_option(4, NAME, 2554, Node2200, NEUTRAL_REACTION);
+    giq_option(4, NAME, 2555, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node2600
 begin
-    gsay_reply(1706, 2600);
-    giq_option(4, 1706, 2601, Node2500, 50);
-    giq_option(4, 1706, 2602, Node2550, 50);
-    giq_option(4, 1706, 2603, Node2650, 50);
-    giq_option(4, 1706, 2604, Node2200, 50);
-    giq_option(4, 1706, 2605, Node999, 50);
+    Reply(2600);
+    giq_option(4, NAME, 2601, Node2500, NEUTRAL_REACTION);
+    giq_option(4, NAME, 2602, Node2550, NEUTRAL_REACTION);
+    giq_option(4, NAME, 2603, Node2650, NEUTRAL_REACTION);
+    giq_option(4, NAME, 2604, Node2200, NEUTRAL_REACTION);
+    giq_option(4, NAME, 2605, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node2650
 begin
-    gsay_reply(1706, 2650);
-    giq_option(4, 1706, 2651, Node2500, 50);
-    giq_option(4, 1706, 2652, Node2550, 50);
-    giq_option(4, 1706, 2653, Node2600, 50);
-    giq_option(4, 1706, 2654, Node2200, 50);
-    giq_option(4, 1706, 2655, Node999, 50);
+    Reply(2650);
+    giq_option(4, NAME, 2651, Node2500, NEUTRAL_REACTION);
+    giq_option(4, NAME, 2652, Node2550, NEUTRAL_REACTION);
+    giq_option(4, NAME, 2653, Node2600, NEUTRAL_REACTION);
+    giq_option(4, NAME, 2654, Node2200, NEUTRAL_REACTION);
+    giq_option(4, NAME, 2655, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node2700
 begin
-    gsay_reply(1706, 2700);
-    giq_option(-3, 1706, 2701, Node999, 50);
-    giq_option(-3, 1706, 2702, Node998, 50);
+    Reply(2700);
+    giq_option(-3, NAME, 2701, Node999, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 2702, Node998, NEUTRAL_REACTION);
 end
 
 procedure Node2750
 begin
-    gsay_reply(1706, 2750);
-    giq_option(4, 1706, 2751, Node650, 50);
-    giq_option(4, 1706, 2752, Node999, 50);
-    giq_option(4, 1706, 2753, Node998, 50);
+    Reply(2750);
+    giq_option(4, NAME, 2751, Node650, NEUTRAL_REACTION);
+    giq_option(4, NAME, 2752, Node999, NEUTRAL_REACTION);
+    giq_option(4, NAME, 2753, Node998, NEUTRAL_REACTION);
 end
 
 procedure Node2800
 begin
-    gsay_reply(1706, 2800);
-    set_global_var(897, 2);
-    if (has_skill(dude_obj, 14) > 120) then begin
-        giq_option(4, 1706, 2802, Node2850, 50);
+    Reply(2800);
+    set_global_var(GVAR_AQ_QUEST_HOSPITAL_HERMAN, 2);
+    if (dude_skill(SKILL_SPEECH) > 120) then begin
+        giq_option(4, NAME, 2802, Node2850, NEUTRAL_REACTION);
     end
-    giq_option(4, 1706, 2801, Node999, 50);
+    giq_option(4, NAME, 2801, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node2850
 begin
-    gsay_reply(1706, 2850);
-    set_global_var(897, 3);
-    giq_option(4, 1706, 2851, Node999, 50);
+    Reply(2850);
+    set_global_var(GVAR_AQ_QUEST_HOSPITAL_HERMAN, 3);
+    giq_option(4, NAME, 2851, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node2900
 begin
-    gsay_reply(1706, 2900);
+    Reply(2900);
     set_local_var(6, local_var(6) bwor 512);
-    set_global_var(898, 2);
-    if (global_var(881) != 2) then begin
-        set_global_var(881, 0);
+    set_global_var(GVAR_AQ_MUTANT_HUNTERS_CANDIDATE, 2);
+    if (global_var(GVAR_AQ_QUEST_MTHUNT_TULSA_QUEST) != 2) then begin
+        set_global_var(GVAR_AQ_QUEST_MTHUNT_TULSA_QUEST, 0);
     end
-    if (global_var(882) != 4) then begin
-        set_global_var(882, 0);
+    if (global_var(GVAR_AQ_QUEST_MTHUNT_HERMAN_QUEST) != 4) then begin
+        set_global_var(GVAR_AQ_QUEST_MTHUNT_HERMAN_QUEST, 0);
     end
-    if (global_var(883) != 2) then begin
-        set_global_var(883, 0);
+    if (global_var(GVAR_AQ_QUEST_MTHUNT_DUTCH_QUEST) != 2) then begin
+        set_global_var(GVAR_AQ_QUEST_MTHUNT_DUTCH_QUEST, 0);
     end
-    giq_option(4, 1706, 2901, Node999, 50);
-    giq_option(4, 1706, 2902, Node998, 50);
+    giq_option(4, NAME, 2901, Node999, NEUTRAL_REACTION);
+    giq_option(4, NAME, 2902, Node998, NEUTRAL_REACTION);
 end
 
 procedure Node2950
 begin
-    if (get_critter_stat(dude_obj, 34) == 0) then begin
-        gsay_reply(1706, 2950);
+    if (dude_gender == 0) then begin
+        Reply(2950);
     end
     else begin
-        gsay_reply(1706, 2951);
+        Reply(2951);
     end
-    set_global_var(898, 2);
-    if (global_var(882) != 4) then begin
-        set_global_var(882, 0);
+    set_global_var(GVAR_AQ_MUTANT_HUNTERS_CANDIDATE, 2);
+    if (global_var(GVAR_AQ_QUEST_MTHUNT_HERMAN_QUEST) != 4) then begin
+        set_global_var(GVAR_AQ_QUEST_MTHUNT_HERMAN_QUEST, 0);
     end
     set_local_var(6, local_var(6) bwor 512);
-    giq_option(4, 1706, 2952, Node999, 50);
+    giq_option(4, NAME, 2952, Node999, NEUTRAL_REACTION);
 end
 
 procedure Herman_Quest_Start
 begin
-    set_global_var(882, 1);
+    set_global_var(GVAR_AQ_QUEST_MTHUNT_HERMAN_QUEST, 1);
     load_map(21, 0);
 end
 
@@ -1096,10 +1096,9 @@ begin
     set_local_var(6, local_var(6) bwor 4096);
     set_local_var(6, local_var(6) bwand (-1 - 8192));
     set_map_var(5, 1);
-    set_global_var(867, global_var(867) bwor 524288);
-    if (global_var(872) == 1) then begin
-        set_global_var(872, 0);
+    set_global_var(GVAR_AQ_ALBUQ_DEATHS, global_var(GVAR_AQ_ALBUQ_DEATHS) bwor 524288);
+    if (global_var(GVAR_AQ_QUEST_CHUCK_SWEETHEARTS) == 1) then begin
+        set_global_var(GVAR_AQ_QUEST_CHUCK_SWEETHEARTS, 0);
     end
     set_local_var(7, game_time);
 end
-
