diff --git a/Mods/Resurrection/Original/ALB_VLTA.ssl b/Mods/Resurrection/ReDefine/ALB_VLTA.ssl
index 335b03f..c61d4f6 100644
--- a/Mods/Resurrection/Original/ALB_VLTA.ssl
+++ b/Mods/Resurrection/ReDefine/ALB_VLTA.ssl
@@ -87,12 +87,12 @@ begin
             return 1;
         end
     end
-    if ((party_member_obj(16777779) != 0) and ((global_var(910) bwand 32) == 0)) then begin
+    if ((party_member_obj(16777779) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 32) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777779)) <= 5) then begin
             return 1;
         end
     end
-    if ((party_member_obj(16777811) != 0) and ((global_var(910) bwand 64) == 0)) then begin
+    if ((party_member_obj(16777811) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 64) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777811)) <= 5) then begin
             return 1;
         end
@@ -107,7 +107,7 @@ begin
             return 1;
         end
     end
-    if ((party_member_obj(16777814) != 0) and ((global_var(910) bwand 128) == 0)) then begin
+    if ((party_member_obj(16777814) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 128) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777814)) <= 5) then begin
             return 1;
         end
@@ -165,8 +165,8 @@ end
 
 procedure map_enter_p_proc
 begin
-    if (elevation(dude_obj) == 1) then begin
-        if ((global_var(1000) bwand 67108864) != 0) then begin
+    if (dude_elevation == 1) then begin
+        if ((global_var(GVAR_AQE_ALBUQ_EMPEROR_EVENTS_NEXT) bwand 67108864) != 0) then begin
             set_light_level(100);
         end
         else begin
@@ -176,103 +176,103 @@ begin
     else begin
         set_light_level(100);
     end
-    if (not(metarule(22, 0))) then begin
-        if ((global_var(1000) bwand 268435456) != 0) then begin
+    if (not(is_loading_game)) then begin
+        if ((global_var(GVAR_AQE_ALBUQ_EMPEROR_EVENTS_NEXT) bwand 268435456) != 0) then begin
             move_to(dude_obj, 22076, 0);
             anim(dude_obj, 1000, 2);
             if (party_member_obj(16777729) != 0) then begin
                 if (party_member_obj(16777729) != 0) then begin
-                    if (has_trait(1, party_member_obj(16777729), 666)) then begin
-                        critter_attempt_placement(party_member_obj(16777729), tile_num_in_direction(tile_num(dude_obj), (has_trait(1, dude_obj, 10) + 3) % 6, 1), elevation(dude_obj));
+                    if (obj_is_visible_flag(party_member_obj(16777729))) then begin
+                        critter_attempt_placement(party_member_obj(16777729), tile_num_in_direction(dude_tile, (obj_get_rot(dude_obj) + 3) % 6, 1), dude_elevation);
                     end
                 end
             end
-            if ((party_member_obj(16777779) != 0) and ((global_var(910) bwand 32) == 0)) then begin
+            if ((party_member_obj(16777779) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 32) == 0)) then begin
                 if (party_member_obj(16777779) != 0) then begin
-                    if (has_trait(1, party_member_obj(16777779), 666)) then begin
-                        critter_attempt_placement(party_member_obj(16777779), tile_num_in_direction(tile_num(dude_obj), (has_trait(1, dude_obj, 10) + 3) % 6, 1), elevation(dude_obj));
+                    if (obj_is_visible_flag(party_member_obj(16777779))) then begin
+                        critter_attempt_placement(party_member_obj(16777779), tile_num_in_direction(dude_tile, (obj_get_rot(dude_obj) + 3) % 6, 1), dude_elevation);
                     end
                 end
             end
-            if ((party_member_obj(16777811) != 0) and ((global_var(910) bwand 64) == 0)) then begin
+            if ((party_member_obj(16777811) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 64) == 0)) then begin
                 if (party_member_obj(16777811) != 0) then begin
-                    if (has_trait(1, party_member_obj(16777811), 666)) then begin
-                        critter_attempt_placement(party_member_obj(16777811), tile_num_in_direction(tile_num(dude_obj), (has_trait(1, dude_obj, 10) + 3) % 6, 1), elevation(dude_obj));
+                    if (obj_is_visible_flag(party_member_obj(16777811))) then begin
+                        critter_attempt_placement(party_member_obj(16777811), tile_num_in_direction(dude_tile, (obj_get_rot(dude_obj) + 3) % 6, 1), dude_elevation);
                     end
                 end
             end
             if (party_member_obj(16777812) != 0) then begin
                 if (party_member_obj(16777812) != 0) then begin
-                    if (has_trait(1, party_member_obj(16777812), 666)) then begin
-                        critter_attempt_placement(party_member_obj(16777812), tile_num_in_direction(tile_num(dude_obj), (has_trait(1, dude_obj, 10) + 3) % 6, 1), elevation(dude_obj));
+                    if (obj_is_visible_flag(party_member_obj(16777812))) then begin
+                        critter_attempt_placement(party_member_obj(16777812), tile_num_in_direction(dude_tile, (obj_get_rot(dude_obj) + 3) % 6, 1), dude_elevation);
                     end
                 end
             end
-            if ((party_member_obj(16777814) != 0) and ((global_var(910) bwand 128) == 0)) then begin
+            if ((party_member_obj(16777814) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 128) == 0)) then begin
                 if (party_member_obj(16777814) != 0) then begin
-                    if (has_trait(1, party_member_obj(16777814), 666)) then begin
-                        critter_attempt_placement(party_member_obj(16777814), tile_num_in_direction(tile_num(dude_obj), (has_trait(1, dude_obj, 10) + 3) % 6, 1), elevation(dude_obj));
+                    if (obj_is_visible_flag(party_member_obj(16777814))) then begin
+                        critter_attempt_placement(party_member_obj(16777814), tile_num_in_direction(dude_tile, (obj_get_rot(dude_obj) + 3) % 6, 1), dude_elevation);
                     end
                 end
             end
             if (party_member_obj(16777748) != 0) then begin
                 if (party_member_obj(16777748) != 0) then begin
-                    if (has_trait(1, party_member_obj(16777748), 666)) then begin
-                        critter_attempt_placement(party_member_obj(16777748), tile_num_in_direction(tile_num(dude_obj), (has_trait(1, dude_obj, 10) + 3) % 6, 1), elevation(dude_obj));
+                    if (obj_is_visible_flag(party_member_obj(16777748))) then begin
+                        critter_attempt_placement(party_member_obj(16777748), tile_num_in_direction(dude_tile, (obj_get_rot(dude_obj) + 3) % 6, 1), dude_elevation);
                     end
                 end
             end
             if (party_member_obj(16778017) != 0) then begin
                 if (party_member_obj(16778017) != 0) then begin
-                    if (has_trait(1, party_member_obj(16778017), 666)) then begin
-                        critter_attempt_placement(party_member_obj(16778017), tile_num_in_direction(tile_num(dude_obj), (has_trait(1, dude_obj, 10) + 3) % 6, 1), elevation(dude_obj));
+                    if (obj_is_visible_flag(party_member_obj(16778017))) then begin
+                        critter_attempt_placement(party_member_obj(16778017), tile_num_in_direction(dude_tile, (obj_get_rot(dude_obj) + 3) % 6, 1), dude_elevation);
                     end
                 end
             end
             if (party_member_obj(16778016) != 0) then begin
                 if (party_member_obj(16778016) != 0) then begin
-                    if (has_trait(1, party_member_obj(16778016), 666)) then begin
-                        critter_attempt_placement(party_member_obj(16778016), tile_num_in_direction(tile_num(dude_obj), (has_trait(1, dude_obj, 10) + 3) % 6, 1), elevation(dude_obj));
+                    if (obj_is_visible_flag(party_member_obj(16778016))) then begin
+                        critter_attempt_placement(party_member_obj(16778016), tile_num_in_direction(dude_tile, (obj_get_rot(dude_obj) + 3) % 6, 1), dude_elevation);
                     end
                 end
             end
             if (party_member_obj(16778049) != 0) then begin
                 if (party_member_obj(16778049) != 0) then begin
-                    if (has_trait(1, party_member_obj(16778049), 666)) then begin
-                        critter_attempt_placement(party_member_obj(16778049), tile_num_in_direction(tile_num(dude_obj), (has_trait(1, dude_obj, 10) + 3) % 6, 1), elevation(dude_obj));
+                    if (obj_is_visible_flag(party_member_obj(16778049))) then begin
+                        critter_attempt_placement(party_member_obj(16778049), tile_num_in_direction(dude_tile, (obj_get_rot(dude_obj) + 3) % 6, 1), dude_elevation);
                     end
                 end
             end
             if (party_member_obj(16778050) != 0) then begin
                 if (party_member_obj(16778050) != 0) then begin
-                    if (has_trait(1, party_member_obj(16778050), 666)) then begin
-                        critter_attempt_placement(party_member_obj(16778050), tile_num_in_direction(tile_num(dude_obj), (has_trait(1, dude_obj, 10) + 3) % 6, 1), elevation(dude_obj));
+                    if (obj_is_visible_flag(party_member_obj(16778050))) then begin
+                        critter_attempt_placement(party_member_obj(16778050), tile_num_in_direction(dude_tile, (obj_get_rot(dude_obj) + 3) % 6, 1), dude_elevation);
                     end
                 end
             end
             if (party_member_obj(16778051) != 0) then begin
                 if (party_member_obj(16778051) != 0) then begin
-                    if (has_trait(1, party_member_obj(16778051), 666)) then begin
-                        critter_attempt_placement(party_member_obj(16778051), tile_num_in_direction(tile_num(dude_obj), (has_trait(1, dude_obj, 10) + 3) % 6, 1), elevation(dude_obj));
+                    if (obj_is_visible_flag(party_member_obj(16778051))) then begin
+                        critter_attempt_placement(party_member_obj(16778051), tile_num_in_direction(dude_tile, (obj_get_rot(dude_obj) + 3) % 6, 1), dude_elevation);
                     end
                 end
             end
             if (party_member_obj(16777294) != 0) then begin
                 if (party_member_obj(16777294) != 0) then begin
-                    if (has_trait(1, party_member_obj(16777294), 666)) then begin
-                        critter_attempt_placement(party_member_obj(16777294), tile_num_in_direction(tile_num(dude_obj), (has_trait(1, dude_obj, 10) + 3) % 6, 1), elevation(dude_obj));
+                    if (obj_is_visible_flag(party_member_obj(16777294))) then begin
+                        critter_attempt_placement(party_member_obj(16777294), tile_num_in_direction(dude_tile, (obj_get_rot(dude_obj) + 3) % 6, 1), dude_elevation);
                     end
                 end
             end
             add_timer_event(Vault_elevator_door, game_ticks(0), 3);
         end
-        if (((global_var(865) bwand 268435456) != 0) and not((map_var(0) bwand 4) != 0)) then begin
+        if (((global_var(GVAR_AQ_ALBUQ_EVENTS) bwand 268435456) != 0) and not((map_var(0) bwand 4) != 0)) then begin
             call destroy_cave_wall();
             if (not((map_var(0) bwand 4) != 0)) then begin
                 set_map_var(0, map_var(0) bwor 4);
             end
         end
-        if ((global_var(903) == 4) or (global_var(903) == 5) and not((map_var(0) bwand 1048576) != 0)) then begin
+        if ((global_var(GVAR_AQE_QUEST_GET_OUR_VAULT_BACK) == 4) or (global_var(GVAR_AQE_QUEST_GET_OUR_VAULT_BACK) == 5) and not((map_var(0) bwand 1048576) != 0)) then begin
             set_map_var(0, map_var(0) bwor 1048576);
             add_timer_event(self_obj, game_ticks(0), 60);
         end
@@ -281,8 +281,8 @@ end
 
 procedure map_update_p_proc
 begin
-    if (elevation(dude_obj) == 1) then begin
-        if ((global_var(1000) bwand 67108864) != 0) then begin
+    if (dude_elevation == 1) then begin
+        if ((global_var(GVAR_AQE_ALBUQ_EMPEROR_EVENTS_NEXT) bwand 67108864) != 0) then begin
             set_light_level(100);
         end
         else begin
@@ -296,15 +296,15 @@ end
 
 procedure map_exit_p_proc
 begin
-    if ((global_var(903) < 4) and (((global_var(865) bwand 268435456) != 0) and not((map_var(0) bwand 8) != 0))) then begin
+    if ((global_var(GVAR_AQE_QUEST_GET_OUR_VAULT_BACK) < 4) and (((global_var(GVAR_AQ_ALBUQ_EVENTS) bwand 268435456) != 0) and not((map_var(0) bwand 8) != 0))) then begin
         if ((map_var(0) bwand 16) != 0) then begin
-            set_global_var(864, 1);
+            set_global_var(GVAR_AQ_ENEMY_HUNTERS, 1);
         end
-        if (not((global_var(865) bwand 67108864) != 0)) then begin
-            set_global_var(864, 1);
+        if (not((global_var(GVAR_AQ_ALBUQ_EVENTS) bwand 67108864) != 0)) then begin
+            set_global_var(GVAR_AQ_ENEMY_HUNTERS, 1);
         end
-        if ((elevation(dude_obj) == 2) and not((global_var(865) bwand 536870912) != 0)) then begin
-            set_global_var(865, global_var(865) bwor 536870912);
+        if ((dude_elevation == 2) and not((global_var(GVAR_AQ_ALBUQ_EVENTS) bwand 536870912) != 0)) then begin
+            set_global_var(GVAR_AQ_ALBUQ_EVENTS, global_var(GVAR_AQ_ALBUQ_EVENTS) bwor 536870912);
         end
     end
 end
@@ -316,11 +316,11 @@ begin
             sequence := sequence + 1;
         end
         if (sequence == 1) then begin
-            set_global_var(865, global_var(865) bwor 268435456);
-            if (global_var(864) == 1) then begin
+            set_global_var(GVAR_AQ_ALBUQ_EVENTS, global_var(GVAR_AQ_ALBUQ_EVENTS) bwor 268435456);
+            if (global_var(GVAR_AQ_ENEMY_HUNTERS) == 1) then begin
                 set_map_var(0, map_var(0) bwor 16);
             end
-            set_global_var(864, 0);
+            set_global_var(GVAR_AQ_ENEMY_HUNTERS, 0);
             if (combat_is_initialized) then begin
                 terminate_combat;
             end
@@ -389,35 +389,35 @@ begin
         end
         if (sequence == 17) then begin
             game_ui_enable;
-            if (tile_in_tile_rect(27481, 27473, 32281, 32273, tile_num(dude_obj)) and (elevation(dude_obj) == 2)) then begin
-                critter_heal(dude_obj, get_critter_stat(dude_obj, 7) * 2);
+            if (tile_in_tile_rect(27481, 27473, 32281, 32273, dude_tile) and (dude_elevation == 2)) then begin
+                critter_heal(dude_obj, get_critter_stat(dude_obj, STAT_max_hp) * 2);
             end
-            if (((map_var(0) bwand 16) != 0) or (global_var(859) == 1)) then begin
-                set_global_var(864, 1);
+            if (((map_var(0) bwand 16) != 0) or (global_var(GVAR_AQ_MUTANT_HUNTER) == 1)) then begin
+                set_global_var(GVAR_AQ_ENEMY_HUNTERS, 1);
             end
-            if (not((global_var(865) bwand 67108864) != 0)) then begin
-                set_global_var(864, 1);
+            if (not((global_var(GVAR_AQ_ALBUQ_EVENTS) bwand 67108864) != 0)) then begin
+                set_global_var(GVAR_AQ_ENEMY_HUNTERS, 1);
             end
             set_map_var(0, map_var(0) bwor 8);
             sequence := 0;
-            if (global_var(903) >= 1) then begin
+            if (global_var(GVAR_AQE_QUEST_GET_OUR_VAULT_BACK) >= 1) then begin
                 add_timer_event(self_obj, game_ticks(0), 22);
             end
         end
     end
     if (fixed_param == 2) then begin
-        if ((obj_pid(critter_inven_obj(dude_obj, 0)) == 3) or (obj_pid(critter_inven_obj(dude_obj, 0)) == 348) or (obj_pid(critter_inven_obj(dude_obj, 0)) == 349) or (obj_pid(critter_inven_obj(dude_obj, 0)) == 232)) then begin
-            display_msg(message_str(1638, 100) + message_str(1638, 102));
+        if ((obj_pid(critter_inven_obj(dude_obj, INVEN_TYPE_WORN)) == 3) or (obj_pid(critter_inven_obj(dude_obj, INVEN_TYPE_WORN)) == 348) or (obj_pid(critter_inven_obj(dude_obj, INVEN_TYPE_WORN)) == 349) or (obj_pid(critter_inven_obj(dude_obj, INVEN_TYPE_WORN)) == 232)) then begin
+            display_msg(mstr(100) + mstr(102));
         end
         else begin
-            display_msg(message_str(1638, 100) + message_str(1638, 101));
+            display_msg(mstr(100) + mstr(101));
         end
     end
     if (fixed_param == 3) then begin
-        display_msg(message_str(1638, 103));
+        display_mstr(103);
     end
     if (fixed_param == 4) then begin
-        display_msg(message_str(1638, 104));
+        display_mstr(104);
     end
     if (fixed_param == 5) then begin
         if (fixed_param == 5) then begin
@@ -437,46 +437,46 @@ begin
         end
         if (sequence_floor_three == 4) then begin
             explosion(17111, 2, 999);
-            if (get_critter_stat(dude_obj, 34) == 0) then begin
-                display_msg(message_str(1638, 105));
+            if (dude_gender == 0) then begin
+                display_mstr(105);
             end
             else begin
-                display_msg(message_str(1638, 106));
+                display_mstr(106);
             end
             add_timer_event(self_obj, game_ticks(0), 5);
         end
         if (sequence_floor_three == 5) then begin
             game_ui_enable;
-            if (get_critter_stat(dude_obj, 35) > 0) then begin
-                critter_heal(dude_obj, -get_critter_stat(dude_obj, 35));
+            if (get_critter_stat(dude_obj, STAT_current_hp) > 0) then begin
+                critter_heal(dude_obj, -get_critter_stat(dude_obj, STAT_current_hp));
             end
         end
     end
     if (fixed_param == 20) then begin
-        display_msg(message_str(1638, 108));
+        display_mstr(108);
     end
     if (fixed_param == 21) then begin
-        display_msg(message_str(1638, 109));
+        display_mstr(109);
     end
     if (fixed_param == 22) then begin
-        if (((global_var(865) bwand 134217728) != 0) and not((global_var(865) bwand 4194304) != 0)) then begin
-            display_msg(message_str(1638, 110));
+        if (((global_var(GVAR_AQ_ALBUQ_EVENTS) bwand 134217728) != 0) and not((global_var(GVAR_AQ_ALBUQ_EVENTS) bwand 4194304) != 0)) then begin
+            display_mstr(110);
             add_timer_event(self_obj, game_ticks(1), 25);
         end
         else begin
-            if ((test_podminky == 0) and (((global_var(865) bwand 4194304) != 0) and (global_var(1012) < 18) or ((global_var(1014) < (20 + (2 * map_var(1)))) and not((global_var(865) bwand 134217728) != 0)))) then begin
-                display_msg(message_str(1638, 110) + message_str(1638, 111));
+            if ((test_podminky == 0) and (((global_var(GVAR_AQ_ALBUQ_EVENTS) bwand 4194304) != 0) and (global_var(GVAR_AQE_2ST_FLOOR_ALIENS_KILLED) < 18) or ((global_var(GVAR_AQE_VAULT_HUNTERS_KILLED) < (20 + (2 * map_var(1)))) and not((global_var(GVAR_AQ_ALBUQ_EVENTS) bwand 134217728) != 0)))) then begin
+                display_msg(mstr(110) + mstr(111));
                 set_map_var(0, map_var(0) bwor 16384);
                 add_timer_event(self_obj, game_ticks(0), 40);
                 test_podminky := 1;
             end
-            if ((test_podminky == 0) and ((global_var(865) bwand 4194304) != 0) and (global_var(1012) >= 18)) then begin
-                display_msg(message_str(1638, 110));
+            if ((test_podminky == 0) and ((global_var(GVAR_AQ_ALBUQ_EVENTS) bwand 4194304) != 0) and (global_var(GVAR_AQE_2ST_FLOOR_ALIENS_KILLED) >= 18)) then begin
+                display_mstr(110);
                 test_podminky := 1;
                 add_timer_event(self_obj, game_ticks(1), 25);
             end
-            if ((test_podminky == 0) and (global_var(1014) >= (20 + (2 * map_var(1)))) and not((global_var(865) bwand 134217728) != 0)) then begin
-                display_msg(message_str(1638, 110));
+            if ((test_podminky == 0) and (global_var(GVAR_AQE_VAULT_HUNTERS_KILLED) >= (20 + (2 * map_var(1)))) and not((global_var(GVAR_AQ_ALBUQ_EVENTS) bwand 134217728) != 0)) then begin
+                display_mstr(110);
                 test_podminky := 1;
                 add_timer_event(self_obj, game_ticks(1), 25);
             end
@@ -484,19 +484,19 @@ begin
                 test_podminky := 1;
                 set_map_var(0, map_var(0) bwor 16384);
                 add_timer_event(self_obj, game_ticks(0), 40);
-                display_msg(message_str(1638, 110) + message_str(1638, 111));
+                display_msg(mstr(110) + mstr(111));
             end
             test_podminky := 0;
         end
     end
     if (fixed_param == 25) then begin
-        if (get_critter_stat(dude_obj, 34) == 0) then begin
-            display_msg(message_str(1638, 112));
+        if (dude_gender == 0) then begin
+            display_mstr(112);
         end
         else begin
-            display_msg(message_str(1638, 113));
+            display_mstr(113);
         end
-        set_global_var(903, 4);
+        set_global_var(GVAR_AQE_QUEST_GET_OUR_VAULT_BACK, 4);
         if ((map_var(0) bwand 16384) != 0) then begin
             set_map_var(0, map_var(0) bwand (-1 - 16384));
         end
@@ -507,12 +507,12 @@ begin
         end
         if (sequence2 == 1) then begin
             set_map_var(0, map_var(0) bwor 65536);
-            set_global_var(632, global_var(632) bwor 64);
+            set_global_var(GVAR_PARTY_MEMBERS_HIDDEN, global_var(GVAR_PARTY_MEMBERS_HIDDEN) bwor 64);
             if (party_member_obj(16777729) != 0) then begin
                 if (party_member_obj(16777729)) then begin
-                    if (has_trait(1, party_member_obj(16777729), 666)) then begin
+                    if (obj_is_visible_flag(party_member_obj(16777729))) then begin
                         debug_msg("set party member hidden: " + obj_name(party_member_obj(16777729)));
-                        set_global_var(632, global_var(632) bwor 1);
+                        set_global_var(GVAR_PARTY_MEMBERS_HIDDEN, global_var(GVAR_PARTY_MEMBERS_HIDDEN) bwor 1);
                         set_obj_visibility(party_member_obj(16777729), 1);
                     end
                     else begin
@@ -520,11 +520,11 @@ begin
                     end
                 end
             end
-            if ((party_member_obj(16777779) != 0) and ((global_var(910) bwand 32) == 0)) then begin
+            if ((party_member_obj(16777779) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 32) == 0)) then begin
                 if (party_member_obj(16777779)) then begin
-                    if (has_trait(1, party_member_obj(16777779), 666)) then begin
+                    if (obj_is_visible_flag(party_member_obj(16777779))) then begin
                         debug_msg("set party member hidden: " + obj_name(party_member_obj(16777779)));
-                        set_global_var(632, global_var(632) bwor 2);
+                        set_global_var(GVAR_PARTY_MEMBERS_HIDDEN, global_var(GVAR_PARTY_MEMBERS_HIDDEN) bwor 2);
                         set_obj_visibility(party_member_obj(16777779), 1);
                     end
                     else begin
@@ -532,11 +532,11 @@ begin
                     end
                 end
             end
-            if ((party_member_obj(16777811) != 0) and ((global_var(910) bwand 64) == 0)) then begin
+            if ((party_member_obj(16777811) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 64) == 0)) then begin
                 if (party_member_obj(16777811)) then begin
-                    if (has_trait(1, party_member_obj(16777811), 666)) then begin
+                    if (obj_is_visible_flag(party_member_obj(16777811))) then begin
                         debug_msg("set party member hidden: " + obj_name(party_member_obj(16777811)));
-                        set_global_var(632, global_var(632) bwor 4);
+                        set_global_var(GVAR_PARTY_MEMBERS_HIDDEN, global_var(GVAR_PARTY_MEMBERS_HIDDEN) bwor 4);
                         set_obj_visibility(party_member_obj(16777811), 1);
                     end
                     else begin
@@ -546,9 +546,9 @@ begin
             end
             if (party_member_obj(16777812) != 0) then begin
                 if (party_member_obj(16777812)) then begin
-                    if (has_trait(1, party_member_obj(16777812), 666)) then begin
+                    if (obj_is_visible_flag(party_member_obj(16777812))) then begin
                         debug_msg("set party member hidden: " + obj_name(party_member_obj(16777812)));
-                        set_global_var(632, global_var(632) bwor 8);
+                        set_global_var(GVAR_PARTY_MEMBERS_HIDDEN, global_var(GVAR_PARTY_MEMBERS_HIDDEN) bwor 8);
                         set_obj_visibility(party_member_obj(16777812), 1);
                     end
                     else begin
@@ -558,9 +558,9 @@ begin
             end
             if (party_member_obj(16777813) != 0) then begin
                 if (party_member_obj(16777813)) then begin
-                    if (has_trait(1, party_member_obj(16777813), 666)) then begin
+                    if (obj_is_visible_flag(party_member_obj(16777813))) then begin
                         debug_msg("set party member hidden: " + obj_name(party_member_obj(16777813)));
-                        set_global_var(632, global_var(632) bwor 16);
+                        set_global_var(GVAR_PARTY_MEMBERS_HIDDEN, global_var(GVAR_PARTY_MEMBERS_HIDDEN) bwor 16);
                         set_obj_visibility(party_member_obj(16777813), 1);
                     end
                     else begin
@@ -568,11 +568,11 @@ begin
                     end
                 end
             end
-            if ((party_member_obj(16777814) != 0) and ((global_var(910) bwand 128) == 0)) then begin
+            if ((party_member_obj(16777814) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 128) == 0)) then begin
                 if (party_member_obj(16777814)) then begin
-                    if (has_trait(1, party_member_obj(16777814), 666)) then begin
+                    if (obj_is_visible_flag(party_member_obj(16777814))) then begin
                         debug_msg("set party member hidden: " + obj_name(party_member_obj(16777814)));
-                        set_global_var(632, global_var(632) bwor 32);
+                        set_global_var(GVAR_PARTY_MEMBERS_HIDDEN, global_var(GVAR_PARTY_MEMBERS_HIDDEN) bwor 32);
                         set_obj_visibility(party_member_obj(16777814), 1);
                     end
                     else begin
@@ -582,21 +582,21 @@ begin
             end
             move_to(dude_obj, 17112, 2);
             anim(dude_obj, 1000, 2);
-            is1 := create_object_sid(16778031, 17311, 2, 128);
-            is2 := create_object_sid(16778031, 17313, 2, 128);
+            is1 := create_object_sid(16778031, 17311, 2, SCRIPT_ACISVOJ4);
+            is2 := create_object_sid(16778031, 17313, 2, SCRIPT_ACISVOJ4);
             anim(is1, 1000, 2);
             anim(is2, 1000, 2);
-            restock_obj := create_object_sid(40, 0, 0, -1);
+            restock_obj := create_object(40, 0, 0);
             add_mult_objs_to_inven(is1, restock_obj, 2);
-            restock_obj := create_object_sid(40, 0, 0, -1);
+            restock_obj := create_object(40, 0, 0);
             add_mult_objs_to_inven(is2, restock_obj, 2);
-            restock_obj := create_object_sid(396, 0, 0, -1);
+            restock_obj := create_object(396, 0, 0);
             add_mult_objs_to_inven(is1, restock_obj, 1);
-            restock_obj := create_object_sid(396, 0, 0, -1);
+            restock_obj := create_object(396, 0, 0);
             add_mult_objs_to_inven(is2, restock_obj, 1);
-            restock_obj := create_object_sid(38, 0, 0, -1);
+            restock_obj := create_object(38, 0, 0);
             add_mult_objs_to_inven(is1, restock_obj, 1);
-            restock_obj := create_object_sid(38, 0, 0, -1);
+            restock_obj := create_object(38, 0, 0);
             add_mult_objs_to_inven(is2, restock_obj, 1);
             game_ui_disable;
             add_timer_event(self_obj, 1, 30);
@@ -664,7 +664,7 @@ begin
             reg_anim_func(1, 1);
             reg_anim_animate(is2, 10, -1);
             reg_anim_func(3, 0);
-            display_msg(message_str(1638, 107));
+            display_mstr(107);
             add_timer_event(self_obj, game_ticks(1), 30);
         end
         if (sequence2 == 9) then begin
@@ -689,22 +689,22 @@ begin
             add_timer_event(self_obj, game_ticks(1), 30);
         end
         if (sequence2 == 12) then begin
-            float_msg(is1, message_str(1638, 380), 8);
+            float_msg(is1, mstr(380), FLOAT_MSG_YELLOW);
             add_timer_event(self_obj, game_ticks(1), 30);
         end
         if (sequence2 == 13) then begin
-            float_msg(is2, message_str(1638, 381), 8);
+            float_msg(is2, mstr(381), FLOAT_MSG_YELLOW);
             animate_move_obj_to_tile(dude_obj, 17911, 0);
             add_timer_event(self_obj, game_ticks(1), 30);
         end
         if (sequence2 == 14) then begin
-            set_global_var(632, global_var(632) bwand (-1 - 64));
+            set_global_var(GVAR_PARTY_MEMBERS_HIDDEN, global_var(GVAR_PARTY_MEMBERS_HIDDEN) bwand (-1 - 64));
             if (party_member_obj(16777729) != 0) then begin
                 if (party_member_obj(16777729)) then begin
-                    if (has_trait(1, party_member_obj(16777729), 666) == 0) then begin
-                        if ((global_var(632) bwand 1) != 0) then begin
+                    if (obj_is_visible_flag(party_member_obj(16777729)) == 0) then begin
+                        if ((global_var(GVAR_PARTY_MEMBERS_HIDDEN) bwand 1) != 0) then begin
                             debug_msg("unhiding party member " + obj_name(party_member_obj(16777729)));
-                            set_global_var(632, global_var(632) bwand (-1 - 1));
+                            set_global_var(GVAR_PARTY_MEMBERS_HIDDEN, global_var(GVAR_PARTY_MEMBERS_HIDDEN) bwand (-1 - 1));
                             set_obj_visibility(party_member_obj(16777729), 0);
                         end
                         else begin
@@ -713,18 +713,18 @@ begin
                     end
                     else begin
                         debug_msg("unhiding party member " + obj_name(party_member_obj(16777729)));
-                        set_global_var(632, global_var(632) bwand (-1 - 1));
+                        set_global_var(GVAR_PARTY_MEMBERS_HIDDEN, global_var(GVAR_PARTY_MEMBERS_HIDDEN) bwand (-1 - 1));
                         set_obj_visibility(party_member_obj(16777729), 0);
                         debug_msg("PARTY.H:::: party member wasn't hidden.. clearing my state");
                     end
                 end
             end
-            if ((party_member_obj(16777779) != 0) and ((global_var(910) bwand 32) == 0)) then begin
+            if ((party_member_obj(16777779) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 32) == 0)) then begin
                 if (party_member_obj(16777779)) then begin
-                    if (has_trait(1, party_member_obj(16777779), 666) == 0) then begin
-                        if ((global_var(632) bwand 2) != 0) then begin
+                    if (obj_is_visible_flag(party_member_obj(16777779)) == 0) then begin
+                        if ((global_var(GVAR_PARTY_MEMBERS_HIDDEN) bwand 2) != 0) then begin
                             debug_msg("unhiding party member " + obj_name(party_member_obj(16777779)));
-                            set_global_var(632, global_var(632) bwand (-1 - 2));
+                            set_global_var(GVAR_PARTY_MEMBERS_HIDDEN, global_var(GVAR_PARTY_MEMBERS_HIDDEN) bwand (-1 - 2));
                             set_obj_visibility(party_member_obj(16777779), 0);
                         end
                         else begin
@@ -733,18 +733,18 @@ begin
                     end
                     else begin
                         debug_msg("unhiding party member " + obj_name(party_member_obj(16777779)));
-                        set_global_var(632, global_var(632) bwand (-1 - 2));
+                        set_global_var(GVAR_PARTY_MEMBERS_HIDDEN, global_var(GVAR_PARTY_MEMBERS_HIDDEN) bwand (-1 - 2));
                         set_obj_visibility(party_member_obj(16777779), 0);
                         debug_msg("PARTY.H:::: party member wasn't hidden.. clearing my state");
                     end
                 end
             end
-            if ((party_member_obj(16777811) != 0) and ((global_var(910) bwand 64) == 0)) then begin
+            if ((party_member_obj(16777811) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 64) == 0)) then begin
                 if (party_member_obj(16777811)) then begin
-                    if (has_trait(1, party_member_obj(16777811), 666) == 0) then begin
-                        if ((global_var(632) bwand 4) != 0) then begin
+                    if (obj_is_visible_flag(party_member_obj(16777811)) == 0) then begin
+                        if ((global_var(GVAR_PARTY_MEMBERS_HIDDEN) bwand 4) != 0) then begin
                             debug_msg("unhiding party member " + obj_name(party_member_obj(16777811)));
-                            set_global_var(632, global_var(632) bwand (-1 - 4));
+                            set_global_var(GVAR_PARTY_MEMBERS_HIDDEN, global_var(GVAR_PARTY_MEMBERS_HIDDEN) bwand (-1 - 4));
                             set_obj_visibility(party_member_obj(16777811), 0);
                         end
                         else begin
@@ -753,7 +753,7 @@ begin
                     end
                     else begin
                         debug_msg("unhiding party member " + obj_name(party_member_obj(16777811)));
-                        set_global_var(632, global_var(632) bwand (-1 - 4));
+                        set_global_var(GVAR_PARTY_MEMBERS_HIDDEN, global_var(GVAR_PARTY_MEMBERS_HIDDEN) bwand (-1 - 4));
                         set_obj_visibility(party_member_obj(16777811), 0);
                         debug_msg("PARTY.H:::: party member wasn't hidden.. clearing my state");
                     end
@@ -761,10 +761,10 @@ begin
             end
             if (party_member_obj(16777812) != 0) then begin
                 if (party_member_obj(16777812)) then begin
-                    if (has_trait(1, party_member_obj(16777812), 666) == 0) then begin
-                        if ((global_var(632) bwand 8) != 0) then begin
+                    if (obj_is_visible_flag(party_member_obj(16777812)) == 0) then begin
+                        if ((global_var(GVAR_PARTY_MEMBERS_HIDDEN) bwand 8) != 0) then begin
                             debug_msg("unhiding party member " + obj_name(party_member_obj(16777812)));
-                            set_global_var(632, global_var(632) bwand (-1 - 8));
+                            set_global_var(GVAR_PARTY_MEMBERS_HIDDEN, global_var(GVAR_PARTY_MEMBERS_HIDDEN) bwand (-1 - 8));
                             set_obj_visibility(party_member_obj(16777812), 0);
                         end
                         else begin
@@ -773,7 +773,7 @@ begin
                     end
                     else begin
                         debug_msg("unhiding party member " + obj_name(party_member_obj(16777812)));
-                        set_global_var(632, global_var(632) bwand (-1 - 8));
+                        set_global_var(GVAR_PARTY_MEMBERS_HIDDEN, global_var(GVAR_PARTY_MEMBERS_HIDDEN) bwand (-1 - 8));
                         set_obj_visibility(party_member_obj(16777812), 0);
                         debug_msg("PARTY.H:::: party member wasn't hidden.. clearing my state");
                     end
@@ -781,10 +781,10 @@ begin
             end
             if (party_member_obj(16777813) != 0) then begin
                 if (party_member_obj(16777813)) then begin
-                    if (has_trait(1, party_member_obj(16777813), 666) == 0) then begin
-                        if ((global_var(632) bwand 16) != 0) then begin
+                    if (obj_is_visible_flag(party_member_obj(16777813)) == 0) then begin
+                        if ((global_var(GVAR_PARTY_MEMBERS_HIDDEN) bwand 16) != 0) then begin
                             debug_msg("unhiding party member " + obj_name(party_member_obj(16777813)));
-                            set_global_var(632, global_var(632) bwand (-1 - 16));
+                            set_global_var(GVAR_PARTY_MEMBERS_HIDDEN, global_var(GVAR_PARTY_MEMBERS_HIDDEN) bwand (-1 - 16));
                             set_obj_visibility(party_member_obj(16777813), 0);
                         end
                         else begin
@@ -793,18 +793,18 @@ begin
                     end
                     else begin
                         debug_msg("unhiding party member " + obj_name(party_member_obj(16777813)));
-                        set_global_var(632, global_var(632) bwand (-1 - 16));
+                        set_global_var(GVAR_PARTY_MEMBERS_HIDDEN, global_var(GVAR_PARTY_MEMBERS_HIDDEN) bwand (-1 - 16));
                         set_obj_visibility(party_member_obj(16777813), 0);
                         debug_msg("PARTY.H:::: party member wasn't hidden.. clearing my state");
                     end
                 end
             end
-            if ((party_member_obj(16777814) != 0) and ((global_var(910) bwand 128) == 0)) then begin
+            if ((party_member_obj(16777814) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 128) == 0)) then begin
                 if (party_member_obj(16777814)) then begin
-                    if (has_trait(1, party_member_obj(16777814), 666) == 0) then begin
-                        if ((global_var(632) bwand 32) != 0) then begin
+                    if (obj_is_visible_flag(party_member_obj(16777814)) == 0) then begin
+                        if ((global_var(GVAR_PARTY_MEMBERS_HIDDEN) bwand 32) != 0) then begin
                             debug_msg("unhiding party member " + obj_name(party_member_obj(16777814)));
-                            set_global_var(632, global_var(632) bwand (-1 - 32));
+                            set_global_var(GVAR_PARTY_MEMBERS_HIDDEN, global_var(GVAR_PARTY_MEMBERS_HIDDEN) bwand (-1 - 32));
                             set_obj_visibility(party_member_obj(16777814), 0);
                         end
                         else begin
@@ -813,7 +813,7 @@ begin
                     end
                     else begin
                         debug_msg("unhiding party member " + obj_name(party_member_obj(16777814)));
-                        set_global_var(632, global_var(632) bwand (-1 - 32));
+                        set_global_var(GVAR_PARTY_MEMBERS_HIDDEN, global_var(GVAR_PARTY_MEMBERS_HIDDEN) bwand (-1 - 32));
                         set_obj_visibility(party_member_obj(16777814), 0);
                         debug_msg("PARTY.H:::: party member wasn't hidden.. clearing my state");
                     end
@@ -821,98 +821,98 @@ begin
             end
             if (party_member_obj(16777729) != 0) then begin
                 if (party_member_obj(16777729) != 0) then begin
-                    if (has_trait(1, party_member_obj(16777729), 666)) then begin
-                        critter_attempt_placement(party_member_obj(16777729), 17112, elevation(dude_obj));
+                    if (obj_is_visible_flag(party_member_obj(16777729))) then begin
+                        critter_attempt_placement(party_member_obj(16777729), 17112, dude_elevation);
                     end
                 end
             end
-            if ((party_member_obj(16777779) != 0) and ((global_var(910) bwand 32) == 0)) then begin
+            if ((party_member_obj(16777779) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 32) == 0)) then begin
                 if (party_member_obj(16777779) != 0) then begin
-                    if (has_trait(1, party_member_obj(16777779), 666)) then begin
-                        critter_attempt_placement(party_member_obj(16777779), 17112, elevation(dude_obj));
+                    if (obj_is_visible_flag(party_member_obj(16777779))) then begin
+                        critter_attempt_placement(party_member_obj(16777779), 17112, dude_elevation);
                     end
                 end
             end
-            if ((party_member_obj(16777811) != 0) and ((global_var(910) bwand 64) == 0)) then begin
+            if ((party_member_obj(16777811) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 64) == 0)) then begin
                 if (party_member_obj(16777811) != 0) then begin
-                    if (has_trait(1, party_member_obj(16777811), 666)) then begin
-                        critter_attempt_placement(party_member_obj(16777811), 17112, elevation(dude_obj));
+                    if (obj_is_visible_flag(party_member_obj(16777811))) then begin
+                        critter_attempt_placement(party_member_obj(16777811), 17112, dude_elevation);
                     end
                 end
             end
             if (party_member_obj(16777812) != 0) then begin
                 if (party_member_obj(16777812) != 0) then begin
-                    if (has_trait(1, party_member_obj(16777812), 666)) then begin
-                        critter_attempt_placement(party_member_obj(16777812), 17112, elevation(dude_obj));
+                    if (obj_is_visible_flag(party_member_obj(16777812))) then begin
+                        critter_attempt_placement(party_member_obj(16777812), 17112, dude_elevation);
                     end
                 end
             end
-            if ((party_member_obj(16777814) != 0) and ((global_var(910) bwand 128) == 0)) then begin
+            if ((party_member_obj(16777814) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 128) == 0)) then begin
                 if (party_member_obj(16777814) != 0) then begin
-                    if (has_trait(1, party_member_obj(16777814), 666)) then begin
-                        critter_attempt_placement(party_member_obj(16777814), 17112, elevation(dude_obj));
+                    if (obj_is_visible_flag(party_member_obj(16777814))) then begin
+                        critter_attempt_placement(party_member_obj(16777814), 17112, dude_elevation);
                     end
                 end
             end
             if (party_member_obj(16777748) != 0) then begin
                 if (party_member_obj(16777748) != 0) then begin
-                    if (has_trait(1, party_member_obj(16777748), 666)) then begin
-                        critter_attempt_placement(party_member_obj(16777748), 17112, elevation(dude_obj));
+                    if (obj_is_visible_flag(party_member_obj(16777748))) then begin
+                        critter_attempt_placement(party_member_obj(16777748), 17112, dude_elevation);
                     end
                 end
             end
             if (party_member_obj(16778017) != 0) then begin
                 if (party_member_obj(16778017) != 0) then begin
-                    if (has_trait(1, party_member_obj(16778017), 666)) then begin
-                        critter_attempt_placement(party_member_obj(16778017), 17112, elevation(dude_obj));
+                    if (obj_is_visible_flag(party_member_obj(16778017))) then begin
+                        critter_attempt_placement(party_member_obj(16778017), 17112, dude_elevation);
                     end
                 end
             end
             if (party_member_obj(16778016) != 0) then begin
                 if (party_member_obj(16778016) != 0) then begin
-                    if (has_trait(1, party_member_obj(16778016), 666)) then begin
-                        critter_attempt_placement(party_member_obj(16778016), 17112, elevation(dude_obj));
+                    if (obj_is_visible_flag(party_member_obj(16778016))) then begin
+                        critter_attempt_placement(party_member_obj(16778016), 17112, dude_elevation);
                     end
                 end
             end
             if (party_member_obj(16778049) != 0) then begin
                 if (party_member_obj(16778049) != 0) then begin
-                    if (has_trait(1, party_member_obj(16778049), 666)) then begin
-                        critter_attempt_placement(party_member_obj(16778049), 17112, elevation(dude_obj));
+                    if (obj_is_visible_flag(party_member_obj(16778049))) then begin
+                        critter_attempt_placement(party_member_obj(16778049), 17112, dude_elevation);
                     end
                 end
             end
             if (party_member_obj(16778050) != 0) then begin
                 if (party_member_obj(16778050) != 0) then begin
-                    if (has_trait(1, party_member_obj(16778050), 666)) then begin
-                        critter_attempt_placement(party_member_obj(16778050), 17112, elevation(dude_obj));
+                    if (obj_is_visible_flag(party_member_obj(16778050))) then begin
+                        critter_attempt_placement(party_member_obj(16778050), 17112, dude_elevation);
                     end
                 end
             end
             if (party_member_obj(16778051) != 0) then begin
                 if (party_member_obj(16778051) != 0) then begin
-                    if (has_trait(1, party_member_obj(16778051), 666)) then begin
-                        critter_attempt_placement(party_member_obj(16778051), 17112, elevation(dude_obj));
+                    if (obj_is_visible_flag(party_member_obj(16778051))) then begin
+                        critter_attempt_placement(party_member_obj(16778051), 17112, dude_elevation);
                     end
                 end
             end
             if (party_member_obj(16777294) != 0) then begin
                 if (party_member_obj(16777294) != 0) then begin
-                    if (has_trait(1, party_member_obj(16777294), 666)) then begin
-                        critter_attempt_placement(party_member_obj(16777294), 17112, elevation(dude_obj));
+                    if (obj_is_visible_flag(party_member_obj(16777294))) then begin
+                        critter_attempt_placement(party_member_obj(16777294), 17112, dude_elevation);
                     end
                 end
             end
             if (party_member_obj(16777729) != 0) then begin
                 anim(party_member_obj(16777729), 1000, 2);
             end
-            if ((party_member_obj(16777779) != 0) and ((global_var(910) bwand 32) == 0)) then begin
+            if ((party_member_obj(16777779) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 32) == 0)) then begin
                 anim(party_member_obj(16777779), 1000, 2);
             end
-            if ((party_member_obj(16777811) != 0) and ((global_var(910) bwand 64) == 0)) then begin
+            if ((party_member_obj(16777811) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 64) == 0)) then begin
                 anim(party_member_obj(16777811), 1000, 2);
             end
-            if ((party_member_obj(16777814) != 0) and ((global_var(910) bwand 128) == 0)) then begin
+            if ((party_member_obj(16777814) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 128) == 0)) then begin
                 anim(party_member_obj(16777814), 1000, 2);
             end
             if (party_member_obj(16778017) != 0) then begin
@@ -937,29 +937,29 @@ begin
         end
     end
     if (fixed_param == 40) then begin
-        is3 := create_object_sid(16778031, 17311, 2, 128);
-        is4 := create_object_sid(16778031, 17312, 2, 128);
-        is5 := create_object_sid(16778032, 17313, 2, 128);
+        is3 := create_object_sid(16778031, 17311, 2, SCRIPT_ACISVOJ4);
+        is4 := create_object_sid(16778031, 17312, 2, SCRIPT_ACISVOJ4);
+        is5 := create_object_sid(16778032, 17313, 2, SCRIPT_ACISVOJ4);
         anim(is3, 1000, 2);
         anim(is4, 1000, 2);
         anim(is5, 1000, 2);
-        restock_obj := create_object_sid(40, 0, 0, -1);
+        restock_obj := create_object(40, 0, 0);
         add_mult_objs_to_inven(is3, restock_obj, 2);
-        restock_obj := create_object_sid(40, 0, 0, -1);
+        restock_obj := create_object(40, 0, 0);
         add_mult_objs_to_inven(is4, restock_obj, 2);
-        restock_obj := create_object_sid(40, 0, 0, -1);
+        restock_obj := create_object(40, 0, 0);
         add_mult_objs_to_inven(is5, restock_obj, 2);
-        restock_obj := create_object_sid(599, 0, 0, -1);
+        restock_obj := create_object(599, 0, 0);
         add_mult_objs_to_inven(is3, restock_obj, 1);
-        restock_obj := create_object_sid(599, 0, 0, -1);
+        restock_obj := create_object(599, 0, 0);
         add_mult_objs_to_inven(is4, restock_obj, 1);
-        restock_obj := create_object_sid(118, 0, 0, -1);
+        restock_obj := create_object(118, 0, 0);
         add_mult_objs_to_inven(is5, restock_obj, 1);
-        restock_obj := create_object_sid(38, 0, 0, -1);
+        restock_obj := create_object(38, 0, 0);
         add_mult_objs_to_inven(is3, restock_obj, 1);
-        restock_obj := create_object_sid(38, 0, 0, -1);
+        restock_obj := create_object(38, 0, 0);
         add_mult_objs_to_inven(is4, restock_obj, 1);
-        restock_obj := create_object_sid(39, 0, 0, -1);
+        restock_obj := create_object(39, 0, 0);
         add_mult_objs_to_inven(is5, restock_obj, 1);
         wield_obj_critter(is3, obj_carrying_pid_obj(is3, 599));
         wield_obj_critter(is4, obj_carrying_pid_obj(is4, 599));
@@ -972,7 +972,7 @@ begin
         call alarm_off();
     end
     if (fixed_param == 60) then begin
-        if ((global_var(903) == 4) or (global_var(903) == 5)) then begin
+        if ((global_var(GVAR_AQE_QUEST_GET_OUR_VAULT_BACK) == 4) or (global_var(GVAR_AQE_QUEST_GET_OUR_VAULT_BACK) == 5)) then begin
             call Generate_New_Soldiers2();
             if ((map_var(0) bwand 262144) != 0) then begin
                 add_timer_event(self_obj, game_ticks(0), 51);
@@ -1067,55 +1067,55 @@ begin
     destroy_object(removable_wall);
     removable_wall := tile_contains_pid_obj(32675, 2, 83886102);
     destroy_object(removable_wall);
-    create_object_sid(67108980, 28875, 2, -1);
-    create_object_sid(67108979, 28877, 2, -1);
-    create_object_sid(67108980, 28879, 2, -1);
-    create_object_sid(67108979, 28881, 2, -1);
-    create_object_sid(67108980, 29275, 2, -1);
-    create_object_sid(67108979, 29277, 2, -1);
-    create_object_sid(67108980, 29279, 2, -1);
-    create_object_sid(67108979, 29281, 2, -1);
-    create_object_sid(67108980, 29675, 2, -1);
-    create_object_sid(67108979, 29677, 2, -1);
-    create_object_sid(67108980, 29679, 2, -1);
-    create_object_sid(67108979, 29681, 2, -1);
-    create_object_sid(67108980, 30075, 2, -1);
-    create_object_sid(67108979, 30077, 2, -1);
-    create_object_sid(67108980, 30079, 2, -1);
-    create_object_sid(67108979, 30081, 2, -1);
-    create_object_sid(67108979, 30481, 2, -1);
-    create_object_sid(50331867, 30275, 2, -1);
-    create_object_sid(50331866, 30076, 2, -1);
-    create_object_sid(50331865, 30277, 2, -1);
-    create_object_sid(50331864, 30077, 2, -1);
-    create_object_sid(50331883, 30279, 2, -1);
-    create_object_sid(50331875, 28474, 2, -1);
-    create_object_sid(50331874, 28675, 2, -1);
-    create_object_sid(50331873, 28476, 2, -1);
-    create_object_sid(50331872, 28677, 2, -1);
-    create_object_sid(50331873, 28478, 2, -1);
-    create_object_sid(50331872, 28679, 2, -1);
-    create_object_sid(50331875, 28480, 2, -1);
-    create_object_sid(50331872, 28681, 2, -1);
-    create_object_sid(50331841, 30875, 2, -1);
-    create_object_sid(50331867, 30475, 2, -1);
-    create_object_sid(50331866, 30276, 2, -1);
-    create_object_sid(33556237, 28078, 2, -1);
-    create_object_sid(33556235, 27478, 2, -1);
-    create_object_sid(33554512, 31276, 2, -1);
-    create_object_sid(33554511, 31077, 2, -1);
-    create_object_sid(33554513, 30277, 2, -1);
-    create_object_sid(33554510, 30278, 2, -1);
-    create_object_sid(33554511, 30476, 2, -1);
-    create_object_sid(33554499, 27875, 2, -1);
-    create_object_sid(33554499, 27876, 2, -1);
-    create_object_sid(33554499, 28077, 2, -1);
-    create_object_sid(33554499, 28078, 2, -1);
-    create_object_sid(33554499, 28279, 2, -1);
-    create_object_sid(33554499, 28079, 2, -1);
-    create_object_sid(33554499, 27478, 2, -1);
-    create_object_sid(33554499, 27679, 2, -1);
-    create_object_sid(33554499, 27877, 2, -1);
+    create_object(67108980, 28875, 2);
+    create_object(67108979, 28877, 2);
+    create_object(67108980, 28879, 2);
+    create_object(67108979, 28881, 2);
+    create_object(67108980, 29275, 2);
+    create_object(67108979, 29277, 2);
+    create_object(67108980, 29279, 2);
+    create_object(67108979, 29281, 2);
+    create_object(67108980, 29675, 2);
+    create_object(67108979, 29677, 2);
+    create_object(67108980, 29679, 2);
+    create_object(67108979, 29681, 2);
+    create_object(67108980, 30075, 2);
+    create_object(67108979, 30077, 2);
+    create_object(67108980, 30079, 2);
+    create_object(67108979, 30081, 2);
+    create_object(67108979, 30481, 2);
+    create_object(50331867, 30275, 2);
+    create_object(50331866, 30076, 2);
+    create_object(50331865, 30277, 2);
+    create_object(50331864, 30077, 2);
+    create_object(50331883, 30279, 2);
+    create_object(50331875, 28474, 2);
+    create_object(50331874, 28675, 2);
+    create_object(50331873, 28476, 2);
+    create_object(50331872, 28677, 2);
+    create_object(50331873, 28478, 2);
+    create_object(50331872, 28679, 2);
+    create_object(50331875, 28480, 2);
+    create_object(50331872, 28681, 2);
+    create_object(50331841, 30875, 2);
+    create_object(50331867, 30475, 2);
+    create_object(50331866, 30276, 2);
+    create_object(33556237, 28078, 2);
+    create_object(33556235, 27478, 2);
+    create_object(33554512, 31276, 2);
+    create_object(33554511, 31077, 2);
+    create_object(33554513, 30277, 2);
+    create_object(33554510, 30278, 2);
+    create_object(33554511, 30476, 2);
+    create_object(33554499, 27875, 2);
+    create_object(33554499, 27876, 2);
+    create_object(33554499, 28077, 2);
+    create_object(33554499, 28078, 2);
+    create_object(33554499, 28279, 2);
+    create_object(33554499, 28079, 2);
+    create_object(33554499, 27478, 2);
+    create_object(33554499, 27679, 2);
+    create_object(33554499, 27877, 2);
     set_map_var(0, map_var(0) bwor 4);
 end
 
@@ -1131,11 +1131,11 @@ begin
     destroy_object(removable_light);
     removable_light := tile_contains_pid_obj(27482, 2, 33555260);
     destroy_object(removable_light);
-    create_object_sid(33555340, 20710, 2, -1);
-    create_object_sid(33555339, 22121, 2, -1);
-    create_object_sid(33555339, 17321, 2, -1);
-    create_object_sid(33555340, 23898, 2, -1);
-    create_object_sid(33555340, 27482, 2, -1);
+    create_object(33555340, 20710, 2);
+    create_object(33555339, 22121, 2);
+    create_object(33555339, 17321, 2);
+    create_object(33555340, 23898, 2);
+    create_object(33555340, 27482, 2);
     set_map_var(0, map_var(0) bwor 262144);
 end
 
@@ -1151,42 +1151,41 @@ begin
     destroy_object(removable_light);
     removable_light := tile_contains_pid_obj(27482, 2, 33555340);
     destroy_object(removable_light);
-    create_object_sid(33555260, 20710, 2, -1);
-    create_object_sid(33555259, 22121, 2, -1);
-    create_object_sid(33555259, 17321, 2, -1);
-    create_object_sid(33555260, 23898, 2, -1);
-    create_object_sid(33555260, 27482, 2, -1);
+    create_object(33555260, 20710, 2);
+    create_object(33555259, 22121, 2);
+    create_object(33555259, 17321, 2);
+    create_object(33555260, 23898, 2);
+    create_object(33555260, 27482, 2);
 end
 
 procedure Generate_New_Soldiers2
 begin
-    is1 := create_object_sid(16778031, 18109, 2, 128);
-    is2 := create_object_sid(16778045, 22122, 2, 128);
-    is3 := create_object_sid(16778031, 22699, 2, 128);
-    is4 := create_object_sid(16778032, 21711, 2, 128);
-    is5 := create_object_sid(16778045, 22463, 2, 128);
-    is6 := create_object_sid(16778031, 20327, 2, 128);
+    is1 := create_object_sid(16778031, 18109, 2, SCRIPT_ACISVOJ4);
+    is2 := create_object_sid(16778045, 22122, 2, SCRIPT_ACISVOJ4);
+    is3 := create_object_sid(16778031, 22699, 2, SCRIPT_ACISVOJ4);
+    is4 := create_object_sid(16778032, 21711, 2, SCRIPT_ACISVOJ4);
+    is5 := create_object_sid(16778045, 22463, 2, SCRIPT_ACISVOJ4);
+    is6 := create_object_sid(16778031, 20327, 2, SCRIPT_ACISVOJ4);
     anim(is1, 1000, 3);
     anim(is2, 1000, 2);
     anim(is3, 1000, 3);
     anim(is4, 1000, 3);
     anim(is5, 1000, 5);
     anim(is6, 1000, 0);
-    restock_obj := create_object_sid(118, 0, 0, -1);
+    restock_obj := create_object(118, 0, 0);
     add_mult_objs_to_inven(is1, restock_obj, 1);
-    restock_obj := create_object_sid(406, 0, 0, -1);
+    restock_obj := create_object(406, 0, 0);
     add_mult_objs_to_inven(is2, restock_obj, 1);
-    restock_obj := create_object_sid(406, 0, 0, -1);
+    restock_obj := create_object(406, 0, 0);
     add_mult_objs_to_inven(is3, restock_obj, 1);
-    restock_obj := create_object_sid(599, 0, 0, -1);
+    restock_obj := create_object(599, 0, 0);
     add_mult_objs_to_inven(is4, restock_obj, 1);
-    restock_obj := create_object_sid(406, 0, 0, -1);
+    restock_obj := create_object(406, 0, 0);
     add_mult_objs_to_inven(is5, restock_obj, 1);
-    restock_obj := create_object_sid(406, 0, 0, -1);
+    restock_obj := create_object(406, 0, 0);
     add_mult_objs_to_inven(is6, restock_obj, 1);
     wield_obj_critter(is1, obj_carrying_pid_obj(is1, 118));
     wield_obj_critter(is2, obj_carrying_pid_obj(is2, 406));
     wield_obj_critter(is3, obj_carrying_pid_obj(is3, 406));
     wield_obj_critter(is3, obj_carrying_pid_obj(is4, 599));
 end
-
