diff --git a/Mods/Resurrection/Original/ALB_NEM4.ssl b/Mods/Resurrection/ReDefine/ALB_NEM4.ssl
index 367f969..032005d 100644
--- a/Mods/Resurrection/Original/ALB_NEM4.ssl
+++ b/Mods/Resurrection/ReDefine/ALB_NEM4.ssl
@@ -62,12 +62,12 @@ begin
             return 1;
         end
     end
-    if ((party_member_obj(16777779) != 0) and ((global_var(910) bwand 32) == 0)) then begin
+    if ((party_member_obj(16777779) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 32) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777779)) <= 5) then begin
             return 1;
         end
     end
-    if ((party_member_obj(16777811) != 0) and ((global_var(910) bwand 64) == 0)) then begin
+    if ((party_member_obj(16777811) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 64) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777811)) <= 5) then begin
             return 1;
         end
@@ -82,7 +82,7 @@ begin
             return 1;
         end
     end
-    if ((party_member_obj(16777814) != 0) and ((global_var(910) bwand 128) == 0)) then begin
+    if ((party_member_obj(16777814) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 128) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777814)) <= 5) then begin
             return 1;
         end
@@ -216,108 +216,108 @@ begin
             end
         end
     end
-    if (not(metarule(22, 0))) then begin
-        if (metarule(14, 0)) then begin
+    if (not(is_loading_game)) then begin
+        if (map_first_run) then begin
             wm_area_set_pos(28, random(520, 580), random(680, 750));
             game_time_advance(8 * (60 * (60 * 10)));
-            if (global_var(888) == 7) then begin
+            if (global_var(GVAR_AQ_QUEST_TRIBE_HUNTINGS) == 7) then begin
                 call Create_Fight_From_Primitives();
                 move_to(dude_obj, 20679, 0);
                 anim(dude_obj, 1000, 3);
                 if (party_member_obj(16777729) != 0) then begin
                     if (party_member_obj(16777729) != 0) then begin
-                        if (has_trait(1, party_member_obj(16777729), 666)) then begin
-                            critter_attempt_placement(party_member_obj(16777729), tile_num_in_direction(tile_num(dude_obj), (has_trait(1, dude_obj, 10) + 3) % 6, 4), elevation(dude_obj));
+                        if (obj_is_visible_flag(party_member_obj(16777729))) then begin
+                            critter_attempt_placement(party_member_obj(16777729), tile_num_in_direction(dude_tile, (obj_get_rot(dude_obj) + 3) % 6, 4), dude_elevation);
                         end
                     end
                 end
-                if ((party_member_obj(16777779) != 0) and ((global_var(910) bwand 32) == 0)) then begin
+                if ((party_member_obj(16777779) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 32) == 0)) then begin
                     if (party_member_obj(16777779) != 0) then begin
-                        if (has_trait(1, party_member_obj(16777779), 666)) then begin
-                            critter_attempt_placement(party_member_obj(16777779), tile_num_in_direction(tile_num(dude_obj), (has_trait(1, dude_obj, 10) + 3) % 6, 4), elevation(dude_obj));
+                        if (obj_is_visible_flag(party_member_obj(16777779))) then begin
+                            critter_attempt_placement(party_member_obj(16777779), tile_num_in_direction(dude_tile, (obj_get_rot(dude_obj) + 3) % 6, 4), dude_elevation);
                         end
                     end
                 end
-                if ((party_member_obj(16777811) != 0) and ((global_var(910) bwand 64) == 0)) then begin
+                if ((party_member_obj(16777811) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 64) == 0)) then begin
                     if (party_member_obj(16777811) != 0) then begin
-                        if (has_trait(1, party_member_obj(16777811), 666)) then begin
-                            critter_attempt_placement(party_member_obj(16777811), tile_num_in_direction(tile_num(dude_obj), (has_trait(1, dude_obj, 10) + 3) % 6, 4), elevation(dude_obj));
+                        if (obj_is_visible_flag(party_member_obj(16777811))) then begin
+                            critter_attempt_placement(party_member_obj(16777811), tile_num_in_direction(dude_tile, (obj_get_rot(dude_obj) + 3) % 6, 4), dude_elevation);
                         end
                     end
                 end
                 if (party_member_obj(16777812) != 0) then begin
                     if (party_member_obj(16777812) != 0) then begin
-                        if (has_trait(1, party_member_obj(16777812), 666)) then begin
-                            critter_attempt_placement(party_member_obj(16777812), tile_num_in_direction(tile_num(dude_obj), (has_trait(1, dude_obj, 10) + 3) % 6, 4), elevation(dude_obj));
+                        if (obj_is_visible_flag(party_member_obj(16777812))) then begin
+                            critter_attempt_placement(party_member_obj(16777812), tile_num_in_direction(dude_tile, (obj_get_rot(dude_obj) + 3) % 6, 4), dude_elevation);
                         end
                     end
                 end
-                if ((party_member_obj(16777814) != 0) and ((global_var(910) bwand 128) == 0)) then begin
+                if ((party_member_obj(16777814) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 128) == 0)) then begin
                     if (party_member_obj(16777814) != 0) then begin
-                        if (has_trait(1, party_member_obj(16777814), 666)) then begin
-                            critter_attempt_placement(party_member_obj(16777814), tile_num_in_direction(tile_num(dude_obj), (has_trait(1, dude_obj, 10) + 3) % 6, 4), elevation(dude_obj));
+                        if (obj_is_visible_flag(party_member_obj(16777814))) then begin
+                            critter_attempt_placement(party_member_obj(16777814), tile_num_in_direction(dude_tile, (obj_get_rot(dude_obj) + 3) % 6, 4), dude_elevation);
                         end
                     end
                 end
                 if (party_member_obj(16777748) != 0) then begin
                     if (party_member_obj(16777748) != 0) then begin
-                        if (has_trait(1, party_member_obj(16777748), 666)) then begin
-                            critter_attempt_placement(party_member_obj(16777748), tile_num_in_direction(tile_num(dude_obj), (has_trait(1, dude_obj, 10) + 3) % 6, 4), elevation(dude_obj));
+                        if (obj_is_visible_flag(party_member_obj(16777748))) then begin
+                            critter_attempt_placement(party_member_obj(16777748), tile_num_in_direction(dude_tile, (obj_get_rot(dude_obj) + 3) % 6, 4), dude_elevation);
                         end
                     end
                 end
                 if (party_member_obj(16778017) != 0) then begin
                     if (party_member_obj(16778017) != 0) then begin
-                        if (has_trait(1, party_member_obj(16778017), 666)) then begin
-                            critter_attempt_placement(party_member_obj(16778017), tile_num_in_direction(tile_num(dude_obj), (has_trait(1, dude_obj, 10) + 3) % 6, 4), elevation(dude_obj));
+                        if (obj_is_visible_flag(party_member_obj(16778017))) then begin
+                            critter_attempt_placement(party_member_obj(16778017), tile_num_in_direction(dude_tile, (obj_get_rot(dude_obj) + 3) % 6, 4), dude_elevation);
                         end
                     end
                 end
                 if (party_member_obj(16778016) != 0) then begin
                     if (party_member_obj(16778016) != 0) then begin
-                        if (has_trait(1, party_member_obj(16778016), 666)) then begin
-                            critter_attempt_placement(party_member_obj(16778016), tile_num_in_direction(tile_num(dude_obj), (has_trait(1, dude_obj, 10) + 3) % 6, 4), elevation(dude_obj));
+                        if (obj_is_visible_flag(party_member_obj(16778016))) then begin
+                            critter_attempt_placement(party_member_obj(16778016), tile_num_in_direction(dude_tile, (obj_get_rot(dude_obj) + 3) % 6, 4), dude_elevation);
                         end
                     end
                 end
                 if (party_member_obj(16778049) != 0) then begin
                     if (party_member_obj(16778049) != 0) then begin
-                        if (has_trait(1, party_member_obj(16778049), 666)) then begin
-                            critter_attempt_placement(party_member_obj(16778049), tile_num_in_direction(tile_num(dude_obj), (has_trait(1, dude_obj, 10) + 3) % 6, 4), elevation(dude_obj));
+                        if (obj_is_visible_flag(party_member_obj(16778049))) then begin
+                            critter_attempt_placement(party_member_obj(16778049), tile_num_in_direction(dude_tile, (obj_get_rot(dude_obj) + 3) % 6, 4), dude_elevation);
                         end
                     end
                 end
                 if (party_member_obj(16778050) != 0) then begin
                     if (party_member_obj(16778050) != 0) then begin
-                        if (has_trait(1, party_member_obj(16778050), 666)) then begin
-                            critter_attempt_placement(party_member_obj(16778050), tile_num_in_direction(tile_num(dude_obj), (has_trait(1, dude_obj, 10) + 3) % 6, 4), elevation(dude_obj));
+                        if (obj_is_visible_flag(party_member_obj(16778050))) then begin
+                            critter_attempt_placement(party_member_obj(16778050), tile_num_in_direction(dude_tile, (obj_get_rot(dude_obj) + 3) % 6, 4), dude_elevation);
                         end
                     end
                 end
                 if (party_member_obj(16778051) != 0) then begin
                     if (party_member_obj(16778051) != 0) then begin
-                        if (has_trait(1, party_member_obj(16778051), 666)) then begin
-                            critter_attempt_placement(party_member_obj(16778051), tile_num_in_direction(tile_num(dude_obj), (has_trait(1, dude_obj, 10) + 3) % 6, 4), elevation(dude_obj));
+                        if (obj_is_visible_flag(party_member_obj(16778051))) then begin
+                            critter_attempt_placement(party_member_obj(16778051), tile_num_in_direction(dude_tile, (obj_get_rot(dude_obj) + 3) % 6, 4), dude_elevation);
                         end
                     end
                 end
                 if (party_member_obj(16777293) != 0) then begin
                     if (party_member_obj(16777293) != 0) then begin
-                        if (has_trait(1, party_member_obj(16777293), 666)) then begin
-                            critter_attempt_placement(party_member_obj(16777293), tile_num_in_direction(tile_num(dude_obj), (has_trait(1, dude_obj, 10) + 3) % 6, 4), elevation(dude_obj));
+                        if (obj_is_visible_flag(party_member_obj(16777293))) then begin
+                            critter_attempt_placement(party_member_obj(16777293), tile_num_in_direction(dude_tile, (obj_get_rot(dude_obj) + 3) % 6, 4), dude_elevation);
                         end
                     end
                 end
                 if (party_member_obj(16777729) != 0) then begin
                     anim(party_member_obj(16777729), 1000, 3);
                 end
-                if ((party_member_obj(16777779) != 0) and ((global_var(910) bwand 32) == 0)) then begin
+                if ((party_member_obj(16777779) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 32) == 0)) then begin
                     anim(party_member_obj(16777779), 1000, 3);
                 end
-                if ((party_member_obj(16777811) != 0) and ((global_var(910) bwand 64) == 0)) then begin
+                if ((party_member_obj(16777811) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 64) == 0)) then begin
                     anim(party_member_obj(16777811), 1000, 3);
                 end
-                if ((party_member_obj(16777814) != 0) and ((global_var(910) bwand 128) == 0)) then begin
+                if ((party_member_obj(16777814) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 128) == 0)) then begin
                     anim(party_member_obj(16777814), 1000, 3);
                 end
                 if (party_member_obj(16778017) != 0) then begin
@@ -340,108 +340,108 @@ begin
                 end
                 add_timer_event(self_obj, game_ticks(0), 2);
                 script_overrides;
-                display_msg(message_str(1657, 101));
+                display_mstr(101);
             end
             else begin
-                set_global_var(889, 6);
-                set_global_var(899, 0);
+                set_global_var(GVAR_AQ_HUNTINGS_RESULTS, 6);
+                set_global_var(GVAR_AQ_HOSPITAL_GIVERS_STATUS, 0);
                 call Create_Fight_From_Hospital();
                 move_to(dude_obj, 17305, 0);
                 anim(dude_obj, 1000, 1);
                 if (party_member_obj(16777729) != 0) then begin
                     if (party_member_obj(16777729) != 0) then begin
-                        if (has_trait(1, party_member_obj(16777729), 666)) then begin
-                            critter_attempt_placement(party_member_obj(16777729), tile_num_in_direction(tile_num(dude_obj), (has_trait(1, dude_obj, 10) + 3) % 6, 4), elevation(dude_obj));
+                        if (obj_is_visible_flag(party_member_obj(16777729))) then begin
+                            critter_attempt_placement(party_member_obj(16777729), tile_num_in_direction(dude_tile, (obj_get_rot(dude_obj) + 3) % 6, 4), dude_elevation);
                         end
                     end
                 end
-                if ((party_member_obj(16777779) != 0) and ((global_var(910) bwand 32) == 0)) then begin
+                if ((party_member_obj(16777779) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 32) == 0)) then begin
                     if (party_member_obj(16777779) != 0) then begin
-                        if (has_trait(1, party_member_obj(16777779), 666)) then begin
-                            critter_attempt_placement(party_member_obj(16777779), tile_num_in_direction(tile_num(dude_obj), (has_trait(1, dude_obj, 10) + 3) % 6, 4), elevation(dude_obj));
+                        if (obj_is_visible_flag(party_member_obj(16777779))) then begin
+                            critter_attempt_placement(party_member_obj(16777779), tile_num_in_direction(dude_tile, (obj_get_rot(dude_obj) + 3) % 6, 4), dude_elevation);
                         end
                     end
                 end
-                if ((party_member_obj(16777811) != 0) and ((global_var(910) bwand 64) == 0)) then begin
+                if ((party_member_obj(16777811) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 64) == 0)) then begin
                     if (party_member_obj(16777811) != 0) then begin
-                        if (has_trait(1, party_member_obj(16777811), 666)) then begin
-                            critter_attempt_placement(party_member_obj(16777811), tile_num_in_direction(tile_num(dude_obj), (has_trait(1, dude_obj, 10) + 3) % 6, 4), elevation(dude_obj));
+                        if (obj_is_visible_flag(party_member_obj(16777811))) then begin
+                            critter_attempt_placement(party_member_obj(16777811), tile_num_in_direction(dude_tile, (obj_get_rot(dude_obj) + 3) % 6, 4), dude_elevation);
                         end
                     end
                 end
                 if (party_member_obj(16777812) != 0) then begin
                     if (party_member_obj(16777812) != 0) then begin
-                        if (has_trait(1, party_member_obj(16777812), 666)) then begin
-                            critter_attempt_placement(party_member_obj(16777812), tile_num_in_direction(tile_num(dude_obj), (has_trait(1, dude_obj, 10) + 3) % 6, 4), elevation(dude_obj));
+                        if (obj_is_visible_flag(party_member_obj(16777812))) then begin
+                            critter_attempt_placement(party_member_obj(16777812), tile_num_in_direction(dude_tile, (obj_get_rot(dude_obj) + 3) % 6, 4), dude_elevation);
                         end
                     end
                 end
-                if ((party_member_obj(16777814) != 0) and ((global_var(910) bwand 128) == 0)) then begin
+                if ((party_member_obj(16777814) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 128) == 0)) then begin
                     if (party_member_obj(16777814) != 0) then begin
-                        if (has_trait(1, party_member_obj(16777814), 666)) then begin
-                            critter_attempt_placement(party_member_obj(16777814), tile_num_in_direction(tile_num(dude_obj), (has_trait(1, dude_obj, 10) + 3) % 6, 4), elevation(dude_obj));
+                        if (obj_is_visible_flag(party_member_obj(16777814))) then begin
+                            critter_attempt_placement(party_member_obj(16777814), tile_num_in_direction(dude_tile, (obj_get_rot(dude_obj) + 3) % 6, 4), dude_elevation);
                         end
                     end
                 end
                 if (party_member_obj(16777748) != 0) then begin
                     if (party_member_obj(16777748) != 0) then begin
-                        if (has_trait(1, party_member_obj(16777748), 666)) then begin
-                            critter_attempt_placement(party_member_obj(16777748), tile_num_in_direction(tile_num(dude_obj), (has_trait(1, dude_obj, 10) + 3) % 6, 4), elevation(dude_obj));
+                        if (obj_is_visible_flag(party_member_obj(16777748))) then begin
+                            critter_attempt_placement(party_member_obj(16777748), tile_num_in_direction(dude_tile, (obj_get_rot(dude_obj) + 3) % 6, 4), dude_elevation);
                         end
                     end
                 end
                 if (party_member_obj(16778017) != 0) then begin
                     if (party_member_obj(16778017) != 0) then begin
-                        if (has_trait(1, party_member_obj(16778017), 666)) then begin
-                            critter_attempt_placement(party_member_obj(16778017), tile_num_in_direction(tile_num(dude_obj), (has_trait(1, dude_obj, 10) + 3) % 6, 4), elevation(dude_obj));
+                        if (obj_is_visible_flag(party_member_obj(16778017))) then begin
+                            critter_attempt_placement(party_member_obj(16778017), tile_num_in_direction(dude_tile, (obj_get_rot(dude_obj) + 3) % 6, 4), dude_elevation);
                         end
                     end
                 end
                 if (party_member_obj(16778016) != 0) then begin
                     if (party_member_obj(16778016) != 0) then begin
-                        if (has_trait(1, party_member_obj(16778016), 666)) then begin
-                            critter_attempt_placement(party_member_obj(16778016), tile_num_in_direction(tile_num(dude_obj), (has_trait(1, dude_obj, 10) + 3) % 6, 4), elevation(dude_obj));
+                        if (obj_is_visible_flag(party_member_obj(16778016))) then begin
+                            critter_attempt_placement(party_member_obj(16778016), tile_num_in_direction(dude_tile, (obj_get_rot(dude_obj) + 3) % 6, 4), dude_elevation);
                         end
                     end
                 end
                 if (party_member_obj(16778049) != 0) then begin
                     if (party_member_obj(16778049) != 0) then begin
-                        if (has_trait(1, party_member_obj(16778049), 666)) then begin
-                            critter_attempt_placement(party_member_obj(16778049), tile_num_in_direction(tile_num(dude_obj), (has_trait(1, dude_obj, 10) + 3) % 6, 4), elevation(dude_obj));
+                        if (obj_is_visible_flag(party_member_obj(16778049))) then begin
+                            critter_attempt_placement(party_member_obj(16778049), tile_num_in_direction(dude_tile, (obj_get_rot(dude_obj) + 3) % 6, 4), dude_elevation);
                         end
                     end
                 end
                 if (party_member_obj(16778050) != 0) then begin
                     if (party_member_obj(16778050) != 0) then begin
-                        if (has_trait(1, party_member_obj(16778050), 666)) then begin
-                            critter_attempt_placement(party_member_obj(16778050), tile_num_in_direction(tile_num(dude_obj), (has_trait(1, dude_obj, 10) + 3) % 6, 4), elevation(dude_obj));
+                        if (obj_is_visible_flag(party_member_obj(16778050))) then begin
+                            critter_attempt_placement(party_member_obj(16778050), tile_num_in_direction(dude_tile, (obj_get_rot(dude_obj) + 3) % 6, 4), dude_elevation);
                         end
                     end
                 end
                 if (party_member_obj(16778051) != 0) then begin
                     if (party_member_obj(16778051) != 0) then begin
-                        if (has_trait(1, party_member_obj(16778051), 666)) then begin
-                            critter_attempt_placement(party_member_obj(16778051), tile_num_in_direction(tile_num(dude_obj), (has_trait(1, dude_obj, 10) + 3) % 6, 4), elevation(dude_obj));
+                        if (obj_is_visible_flag(party_member_obj(16778051))) then begin
+                            critter_attempt_placement(party_member_obj(16778051), tile_num_in_direction(dude_tile, (obj_get_rot(dude_obj) + 3) % 6, 4), dude_elevation);
                         end
                     end
                 end
                 if (party_member_obj(16777293) != 0) then begin
                     if (party_member_obj(16777293) != 0) then begin
-                        if (has_trait(1, party_member_obj(16777293), 666)) then begin
-                            critter_attempt_placement(party_member_obj(16777293), tile_num_in_direction(tile_num(dude_obj), (has_trait(1, dude_obj, 10) + 3) % 6, 4), elevation(dude_obj));
+                        if (obj_is_visible_flag(party_member_obj(16777293))) then begin
+                            critter_attempt_placement(party_member_obj(16777293), tile_num_in_direction(dude_tile, (obj_get_rot(dude_obj) + 3) % 6, 4), dude_elevation);
                         end
                     end
                 end
                 if (party_member_obj(16777729) != 0) then begin
                     anim(party_member_obj(16777729), 1000, 1);
                 end
-                if ((party_member_obj(16777779) != 0) and ((global_var(910) bwand 32) == 0)) then begin
+                if ((party_member_obj(16777779) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 32) == 0)) then begin
                     anim(party_member_obj(16777779), 1000, 1);
                 end
-                if ((party_member_obj(16777811) != 0) and ((global_var(910) bwand 64) == 0)) then begin
+                if ((party_member_obj(16777811) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 64) == 0)) then begin
                     anim(party_member_obj(16777811), 1000, 1);
                 end
-                if ((party_member_obj(16777814) != 0) and ((global_var(910) bwand 128) == 0)) then begin
+                if ((party_member_obj(16777814) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 128) == 0)) then begin
                     anim(party_member_obj(16777814), 1000, 1);
                 end
                 if (party_member_obj(16778017) != 0) then begin
@@ -462,7 +462,7 @@ begin
                 if (party_member_obj(16777812) != 0) then begin
                     anim(party_member_obj(16777812), 1000, 1);
                 end
-                if ((obj_pid(critter_inven_obj(dude_obj, 1)) == 601) or (obj_pid(critter_inven_obj(dude_obj, 2)) == 601)) then begin
+                if ((obj_pid(critter_inven_obj(dude_obj, INVEN_TYPE_RIGHT_HAND)) == 601) or (obj_pid(critter_inven_obj(dude_obj, INVEN_TYPE_LEFT_HAND)) == 601)) then begin
                 end
                 else begin
                     if (obj_is_carrying_obj_pid(dude_obj, 601)) then begin
@@ -471,7 +471,7 @@ begin
                 end
                 add_timer_event(self_obj, game_ticks(0), 1);
                 script_overrides;
-                display_msg(message_str(1657, 100));
+                display_mstr(100);
             end
         end
     end
@@ -559,40 +559,40 @@ end
 
 procedure map_exit_p_proc
 begin
-    if (global_var(877) == 7) then begin
-        if ((global_var(899) < 6) and (cur_map_index == 98)) then begin
-            set_global_var(1017, global_var(1017) bwor 32768);
+    if (global_var(GVAR_AQ_QUEST_HOSPITAL_HUNTINGS) == 7) then begin
+        if ((global_var(GVAR_AQ_HOSPITAL_GIVERS_STATUS) < 6) and (CUR_MAP_ALB_NEM1)) then begin
+            set_global_var(GVAR_AQE_DIRECTIONS_SETUP, global_var(GVAR_AQE_DIRECTIONS_SETUP) bwor 32768);
         end
-        if ((global_var(899) < 14) and (cur_map_index == 97)) then begin
-            set_global_var(1017, global_var(1017) bwor 32768);
+        if ((global_var(GVAR_AQ_HOSPITAL_GIVERS_STATUS) < 14) and (CUR_MAP_ALB_NEM2)) then begin
+            set_global_var(GVAR_AQE_DIRECTIONS_SETUP, global_var(GVAR_AQE_DIRECTIONS_SETUP) bwor 32768);
         end
-        if ((global_var(899) < 16) and (cur_map_index == 96)) then begin
-            set_global_var(1017, global_var(1017) bwor 32768);
+        if ((global_var(GVAR_AQ_HOSPITAL_GIVERS_STATUS) < 16) and (CUR_MAP_ALB_NEM3)) then begin
+            set_global_var(GVAR_AQE_DIRECTIONS_SETUP, global_var(GVAR_AQE_DIRECTIONS_SETUP) bwor 32768);
         end
-        if ((global_var(899) < 5) and (cur_map_index == 95)) then begin
-            set_global_var(1017, global_var(1017) bwor 32768);
+        if ((global_var(GVAR_AQ_HOSPITAL_GIVERS_STATUS) < 5) and (CUR_MAP_ALB_NEM4)) then begin
+            set_global_var(GVAR_AQE_DIRECTIONS_SETUP, global_var(GVAR_AQE_DIRECTIONS_SETUP) bwor 32768);
         end
     end
-    if (global_var(888) == 7) then begin
-        set_global_var(888, 8);
+    if (global_var(GVAR_AQ_QUEST_TRIBE_HUNTINGS) == 7) then begin
+        set_global_var(GVAR_AQ_QUEST_TRIBE_HUNTINGS, 8);
     end
-    if ((global_var(877) == 7) and (global_var(889) > 0)) then begin
-        set_global_var(877, 8);
+    if ((global_var(GVAR_AQ_QUEST_HOSPITAL_HUNTINGS) == 7) and (global_var(GVAR_AQ_HUNTINGS_RESULTS) > 0)) then begin
+        set_global_var(GVAR_AQ_QUEST_HOSPITAL_HUNTINGS, 8);
     end
-    set_global_var(900, global_var(900) bwand (-1 - 16));
-    set_global_var(900, global_var(900) bwand (-1 - 32));
-    set_global_var(900, global_var(900) bwand (-1 - 64));
-    set_global_var(900, global_var(900) bwand (-1 - 128));
-    set_global_var(900, global_var(900) bwand (-1 - 256));
-    set_global_var(900, global_var(900) bwand (-1 - 512));
-    set_global_var(900, global_var(900) bwand (-1 - 4194304));
-    set_global_var(900, global_var(900) bwand (-1 - 8388608));
-    set_global_var(900, global_var(900) bwand (-1 - 16777216));
-    set_global_var(900, global_var(900) bwand (-1 - 1));
-    set_global_var(900, global_var(900) bwand (-1 - 2));
-    set_global_var(900, global_var(900) bwand (-1 - 4));
-    set_global_var(900, global_var(900) bwand (-1 - 8));
-    set_global_var(857, 0);
+    set_global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS, global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS) bwand (-1 - 16));
+    set_global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS, global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS) bwand (-1 - 32));
+    set_global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS, global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS) bwand (-1 - 64));
+    set_global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS, global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS) bwand (-1 - 128));
+    set_global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS, global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS) bwand (-1 - 256));
+    set_global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS, global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS) bwand (-1 - 512));
+    set_global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS, global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS) bwand (-1 - 4194304));
+    set_global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS, global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS) bwand (-1 - 8388608));
+    set_global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS, global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS) bwand (-1 - 16777216));
+    set_global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS, global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS) bwand (-1 - 1));
+    set_global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS, global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS) bwand (-1 - 2));
+    set_global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS, global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS) bwand (-1 - 4));
+    set_global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS, global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS) bwand (-1 - 8));
+    set_global_var(GVAR_FRIENDLY_FIRE, 0);
 end
 
 procedure timed_event_p_proc
@@ -602,7 +602,7 @@ begin
             sequence := sequence + 1;
         end
         if (sequence == 1) then begin
-            metarule3(108, 16900, 0, 0);
+            tile_set_center(16900);
             game_ui_disable;
             add_timer_event(self_obj, game_ticks(1), 1);
         end
@@ -623,7 +623,7 @@ begin
         end
         if (sequence == 3) then begin
             game_ui_enable;
-            set_global_var(900, global_var(900) bwor 1);
+            set_global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS, global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS) bwor 1);
             add_timer_event(self_obj, 5, 1);
         end
         if ((sequence == 4) and not(combat_is_initialized)) then begin
@@ -637,7 +637,7 @@ begin
             sequence := sequence + 1;
         end
         if (sequence == 1) then begin
-            metarule3(108, 21085, 0, 0);
+            tile_set_center(21085);
             game_ui_disable;
             add_timer_event(self_obj, game_ticks(1), 2);
         end
@@ -653,22 +653,22 @@ begin
         end
         if (sequence == 3) then begin
             game_ui_enable;
-            set_global_var(900, global_var(900) bwor 1);
+            set_global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS, global_var(GVAR_AQ_HOSPITAL_ATTACKERS_STATUS) bwor 1);
         end
     end
 end
 
 procedure Create_Fight_From_Primitives
 begin
-    primitiv1 := create_object_sid(16777917, 20477, 0, 1662);
-    primitiv2 := create_object_sid(16777917, 21077, 0, 1662);
-    primitiv3 := create_object_sid(16777917, 20675, 0, 1662);
-    zoldak1 := create_object_sid(16777928, 19683, 0, 1693);
-    zoldak2 := create_object_sid(16777928, 19485, 0, 1693);
-    zoldak3 := create_object_sid(16777929, 18883, 0, 1693);
-    zoldak4 := create_object_sid(16777928, 18283, 0, 1693);
-    zoldak5 := create_object_sid(16777928, 18685, 0, 1693);
-    zoldak6 := create_object_sid(16777929, 18285, 0, 1693);
+    primitiv1 := create_object_sid(16777917, 20477, 0, SCRIPT_ADIVLV);
+    primitiv2 := create_object_sid(16777917, 21077, 0, SCRIPT_ADIVLV);
+    primitiv3 := create_object_sid(16777917, 20675, 0, SCRIPT_ADIVLV);
+    zoldak1 := create_object_sid(16777928, 19683, 0, SCRIPT_ANEMNUNT);
+    zoldak2 := create_object_sid(16777928, 19485, 0, SCRIPT_ANEMNUNT);
+    zoldak3 := create_object_sid(16777929, 18883, 0, SCRIPT_ANEMNUNT);
+    zoldak4 := create_object_sid(16777928, 18283, 0, SCRIPT_ANEMNUNT);
+    zoldak5 := create_object_sid(16777928, 18685, 0, SCRIPT_ANEMNUNT);
+    zoldak6 := create_object_sid(16777929, 18285, 0, SCRIPT_ANEMNUNT);
     anim(zoldak1, 1000, 2);
     anim(zoldak2, 1000, 2);
     anim(zoldak3, 1000, 2);
@@ -678,29 +678,29 @@ begin
     anim(primitiv1, 1000, 3);
     anim(primitiv2, 1000, 3);
     anim(primitiv3, 1000, 3);
-    restock_obj := create_object_sid(601, 0, 0, -1);
+    restock_obj := create_object(601, 0, 0);
     add_mult_objs_to_inven(zoldak1, restock_obj, 1);
-    restock_obj := create_object_sid(601, 0, 0, -1);
+    restock_obj := create_object(601, 0, 0);
     add_mult_objs_to_inven(zoldak2, restock_obj, 1);
-    restock_obj := create_object_sid(601, 0, 0, -1);
+    restock_obj := create_object(601, 0, 0);
     add_mult_objs_to_inven(zoldak3, restock_obj, 1);
-    restock_obj := create_object_sid(601, 0, 0, -1);
+    restock_obj := create_object(601, 0, 0);
     add_mult_objs_to_inven(zoldak4, restock_obj, 1);
-    restock_obj := create_object_sid(601, 0, 0, -1);
+    restock_obj := create_object(601, 0, 0);
     add_mult_objs_to_inven(zoldak5, restock_obj, 1);
-    restock_obj := create_object_sid(601, 0, 0, -1);
+    restock_obj := create_object(601, 0, 0);
     add_mult_objs_to_inven(zoldak6, restock_obj, 1);
-    restock_obj := create_object_sid(40, 0, 0, -1);
+    restock_obj := create_object(40, 0, 0);
     add_mult_objs_to_inven(zoldak1, restock_obj, 1);
-    restock_obj := create_object_sid(40, 0, 0, -1);
+    restock_obj := create_object(40, 0, 0);
     add_mult_objs_to_inven(zoldak2, restock_obj, 1);
-    restock_obj := create_object_sid(40, 0, 0, -1);
+    restock_obj := create_object(40, 0, 0);
     add_mult_objs_to_inven(zoldak3, restock_obj, 1);
-    restock_obj := create_object_sid(40, 0, 0, -1);
+    restock_obj := create_object(40, 0, 0);
     add_mult_objs_to_inven(zoldak4, restock_obj, 1);
-    restock_obj := create_object_sid(40, 0, 0, -1);
+    restock_obj := create_object(40, 0, 0);
     add_mult_objs_to_inven(zoldak5, restock_obj, 1);
-    restock_obj := create_object_sid(40, 0, 0, -1);
+    restock_obj := create_object(40, 0, 0);
     add_mult_objs_to_inven(zoldak6, restock_obj, 1);
     wield_obj_critter(zoldak1, obj_carrying_pid_obj(zoldak1, 601));
     wield_obj_critter(zoldak2, obj_carrying_pid_obj(zoldak2, 601));
@@ -708,13 +708,13 @@ begin
     wield_obj_critter(zoldak4, obj_carrying_pid_obj(zoldak4, 601));
     wield_obj_critter(zoldak5, obj_carrying_pid_obj(zoldak5, 601));
     wield_obj_critter(zoldak6, obj_carrying_pid_obj(zoldak6, 601));
-    restock_obj := create_object_sid(280, 0, 0, -1);
+    restock_obj := create_object(280, 0, 0);
     add_mult_objs_to_inven(primitiv1, restock_obj, 1);
-    restock_obj := create_object_sid(280, 0, 0, -1);
+    restock_obj := create_object(280, 0, 0);
     add_mult_objs_to_inven(primitiv2, restock_obj, 1);
-    restock_obj := create_object_sid(565, 0, 0, -1);
+    restock_obj := create_object(565, 0, 0);
     add_mult_objs_to_inven(primitiv2, restock_obj, 1);
-    restock_obj := create_object_sid(280, 0, 0, -1);
+    restock_obj := create_object(280, 0, 0);
     add_mult_objs_to_inven(primitiv3, restock_obj, 1);
     wield_obj_critter(primitiv1, obj_carrying_pid_obj(primitiv1, 280));
     wield_obj_critter(primitiv2, obj_carrying_pid_obj(primitiv2, 280));
@@ -723,15 +723,15 @@ end
 
 procedure Create_Fight_From_Hospital
 begin
-    primitiv1 := create_object_sid(16777917, 16300, 0, 1693);
-    primitiv2 := create_object_sid(16777917, 15498, 0, 1693);
-    primitiv3 := create_object_sid(16777917, 14698, 0, 1693);
-    primitiv4 := create_object_sid(16777917, 14895, 0, 1693);
-    primitiv5 := create_object_sid(16777917, 15501, 0, 1693);
-    primitiv6 := create_object_sid(16777917, 16297, 0, 1693);
-    zoldak1 := create_object_sid(16777928, 16908, 0, 1694);
-    zoldak2 := create_object_sid(16777928, 16508, 0, 1694);
-    zoldak3 := create_object_sid(16777929, 16906, 0, 1694);
+    primitiv1 := create_object_sid(16777917, 16300, 0, SCRIPT_ANEMNUNT);
+    primitiv2 := create_object_sid(16777917, 15498, 0, SCRIPT_ANEMNUNT);
+    primitiv3 := create_object_sid(16777917, 14698, 0, SCRIPT_ANEMNUNT);
+    primitiv4 := create_object_sid(16777917, 14895, 0, SCRIPT_ANEMNUNT);
+    primitiv5 := create_object_sid(16777917, 15501, 0, SCRIPT_ANEMNUNT);
+    primitiv6 := create_object_sid(16777917, 16297, 0, SCRIPT_ANEMNUNT);
+    zoldak1 := create_object_sid(16777928, 16908, 0, SCRIPT_ASBERLOV);
+    zoldak2 := create_object_sid(16777928, 16508, 0, SCRIPT_ASBERLOV);
+    zoldak3 := create_object_sid(16777929, 16906, 0, SCRIPT_ASBERLOV);
     anim(zoldak1, 1000, 1);
     anim(zoldak2, 1000, 1);
     anim(zoldak3, 1000, 1);
@@ -741,42 +741,42 @@ begin
     anim(primitiv4, 1000, 2);
     anim(primitiv5, 1000, 2);
     anim(primitiv6, 1000, 2);
-    restock_obj := create_object_sid(601, 0, 0, -1);
+    restock_obj := create_object(601, 0, 0);
     add_mult_objs_to_inven(zoldak1, restock_obj, 1);
-    restock_obj := create_object_sid(601, 0, 0, -1);
+    restock_obj := create_object(601, 0, 0);
     add_mult_objs_to_inven(zoldak2, restock_obj, 1);
-    restock_obj := create_object_sid(601, 0, 0, -1);
+    restock_obj := create_object(601, 0, 0);
     add_mult_objs_to_inven(zoldak3, restock_obj, 1);
-    restock_obj := create_object_sid(40, 0, 0, -1);
+    restock_obj := create_object(40, 0, 0);
     add_mult_objs_to_inven(zoldak1, restock_obj, 1);
-    restock_obj := create_object_sid(40, 0, 0, -1);
+    restock_obj := create_object(40, 0, 0);
     add_mult_objs_to_inven(zoldak2, restock_obj, 1);
-    restock_obj := create_object_sid(40, 0, 0, -1);
+    restock_obj := create_object(40, 0, 0);
     add_mult_objs_to_inven(zoldak3, restock_obj, 1);
-    restock_obj := create_object_sid(38, 0, 0, -1);
+    restock_obj := create_object(38, 0, 0);
     add_mult_objs_to_inven(zoldak1, restock_obj, 1);
-    restock_obj := create_object_sid(38, 0, 0, -1);
+    restock_obj := create_object(38, 0, 0);
     add_mult_objs_to_inven(zoldak2, restock_obj, 1);
-    restock_obj := create_object_sid(38, 0, 0, -1);
+    restock_obj := create_object(38, 0, 0);
     add_mult_objs_to_inven(zoldak3, restock_obj, 1);
     wield_obj_critter(zoldak1, obj_carrying_pid_obj(zoldak1, 601));
     wield_obj_critter(zoldak2, obj_carrying_pid_obj(zoldak2, 601));
     wield_obj_critter(zoldak3, obj_carrying_pid_obj(zoldak3, 601));
-    restock_obj := create_object_sid(280, 0, 0, -1);
+    restock_obj := create_object(280, 0, 0);
     add_mult_objs_to_inven(primitiv1, restock_obj, 1);
-    restock_obj := create_object_sid(280, 0, 0, -1);
+    restock_obj := create_object(280, 0, 0);
     add_mult_objs_to_inven(primitiv2, restock_obj, 1);
-    restock_obj := create_object_sid(565, 0, 0, -1);
+    restock_obj := create_object(565, 0, 0);
     add_mult_objs_to_inven(primitiv2, restock_obj, 1);
-    restock_obj := create_object_sid(280, 0, 0, -1);
+    restock_obj := create_object(280, 0, 0);
     add_mult_objs_to_inven(primitiv3, restock_obj, 1);
-    restock_obj := create_object_sid(280, 0, 0, -1);
+    restock_obj := create_object(280, 0, 0);
     add_mult_objs_to_inven(primitiv4, restock_obj, 1);
-    restock_obj := create_object_sid(565, 0, 0, -1);
+    restock_obj := create_object(565, 0, 0);
     add_mult_objs_to_inven(primitiv4, restock_obj, 1);
-    restock_obj := create_object_sid(280, 0, 0, -1);
+    restock_obj := create_object(280, 0, 0);
     add_mult_objs_to_inven(primitiv5, restock_obj, 1);
-    restock_obj := create_object_sid(280, 0, 0, -1);
+    restock_obj := create_object(280, 0, 0);
     add_mult_objs_to_inven(primitiv6, restock_obj, 1);
     wield_obj_critter(primitiv1, obj_carrying_pid_obj(primitiv1, 280));
     wield_obj_critter(primitiv2, obj_carrying_pid_obj(primitiv2, 280));
@@ -785,4 +785,3 @@ begin
     wield_obj_critter(primitiv5, obj_carrying_pid_obj(primitiv5, 280));
     wield_obj_critter(primitiv6, obj_carrying_pid_obj(primitiv6, 280));
 end
-
