diff --git a/Mods/Resurrection/Original/ALOVCOM1.ssl b/Mods/Resurrection/ReDefine/ALOVCOM1.ssl
index cbb389f..0d8ed44 100644
--- a/Mods/Resurrection/Original/ALOVCOM1.ssl
+++ b/Mods/Resurrection/ReDefine/ALOVCOM1.ssl
@@ -87,12 +87,12 @@ begin
             return 1;
         end
     end
-    if ((party_member_obj(16777779) != 0) and ((global_var(910) bwand 32) == 0)) then begin
+    if ((party_member_obj(16777779) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 32) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777779)) <= 5) then begin
             return 1;
         end
     end
-    if ((party_member_obj(16777811) != 0) and ((global_var(910) bwand 64) == 0)) then begin
+    if ((party_member_obj(16777811) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 64) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777811)) <= 5) then begin
             return 1;
         end
@@ -107,7 +107,7 @@ begin
             return 1;
         end
     end
-    if ((party_member_obj(16777814) != 0) and ((global_var(910) bwand 128) == 0)) then begin
+    if ((party_member_obj(16777814) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 128) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777814)) <= 5) then begin
             return 1;
         end
@@ -175,10 +175,10 @@ procedure use_p_proc
 begin
     script_overrides;
     if (combat_is_initialized) then begin
-        display_msg(message_str(1732, 300));
+        display_mstr(300);
     end
     else begin
-        anim(self_obj, 1000, rotation_to_tile(tile_num(self_obj), tile_num(dude_obj)));
+        anim(self_obj, 1000, rotation_to_tile(tile_num(self_obj), dude_tile));
         start_gdialog(1732, self_obj, 4, -1, -1);
         gsay_start;
         call Node500();
@@ -194,13 +194,13 @@ end
 procedure look_at_p_proc
 begin
     script_overrides;
-    display_msg(message_str(1732, 100));
+    display_mstr(100);
 end
 
 procedure description_p_proc
 begin
     script_overrides;
-    display_msg(message_str(1732, 200));
+    display_mstr(200);
 end
 
 procedure use_skill_on_p_proc
@@ -214,10 +214,10 @@ procedure timed_event_p_proc
 begin
     if (fixed_param == 1) then begin
         add_timer_event(Map_Event, game_ticks(0), 50);
-        metarule3(108, 26475, 0, 0);
+        tile_set_center(26475);
         game_ui_disable;
-        metarule3(108, 26475, 0, 0);
-        if ((global_var(865) bwand 4194304) != 0) then begin
+        tile_set_center(26475);
+        if ((global_var(GVAR_AQ_ALBUQ_EVENTS) bwand 4194304) != 0) then begin
             add_timer_event(Map_Event, game_ticks(0), 21);
         end
         else begin
@@ -226,11 +226,11 @@ begin
         add_timer_event(self_obj, game_ticks(5), 2);
     end
     if (fixed_param == 2) then begin
-        metarule3(108, tile_num(dude_obj), 0, 0);
+        tile_set_center(dude_tile);
         add_timer_event(self_obj, 2, 3);
     end
     if (fixed_param == 3) then begin
-        set_global_var(865, global_var(865) bwor 134217728);
+        set_global_var(GVAR_AQ_ALBUQ_EVENTS, global_var(GVAR_AQ_ALBUQ_EVENTS) bwor 134217728);
         add_timer_event(self_obj, 3, 4);
     end
     if (fixed_param == 4) then begin
@@ -244,108 +244,107 @@ end
 
 procedure Node500
 begin
-    if ((global_var(865) bwand 67108864) != 0) then begin
-        gsay_reply(1732, 550);
+    if ((global_var(GVAR_AQ_ALBUQ_EVENTS) bwand 67108864) != 0) then begin
+        Reply(550);
     end
     else begin
-        gsay_reply(1732, 500);
+        Reply(500);
     end
-    if (not((global_var(865) bwand 67108864) != 0)) then begin
-        if ((global_var(865) bwand 33554432) != 0) then begin
-            if (global_var(903) == 3) then begin
-                giq_option(4, 1732, 501, Node600, 50);
+    if (not((global_var(GVAR_AQ_ALBUQ_EVENTS) bwand 67108864) != 0)) then begin
+        if ((global_var(GVAR_AQ_ALBUQ_EVENTS) bwand 33554432) != 0) then begin
+            if (global_var(GVAR_AQE_QUEST_GET_OUR_VAULT_BACK) == 3) then begin
+                giq_option(4, NAME, 501, Node600, NEUTRAL_REACTION);
             end
             else begin
-                giq_option(4, 1732, 501, Node550, 50);
+                giq_option(4, NAME, 501, Node550, NEUTRAL_REACTION);
             end
         end
         else begin
-            if (has_skill(dude_obj, 12) > 94) then begin
-                if (global_var(903) == 3) then begin
-                    giq_option(4, 1732, 502, Node700, 50);
+            if (dude_skill(SKILL_SCIENCE) > 94) then begin
+                if (global_var(GVAR_AQE_QUEST_GET_OUR_VAULT_BACK) == 3) then begin
+                    giq_option(4, NAME, 502, Node700, NEUTRAL_REACTION);
                 end
                 else begin
-                    giq_option(4, 1732, 502, Node650, 50);
+                    giq_option(4, NAME, 502, Node650, NEUTRAL_REACTION);
                 end
             end
             else begin
-                giq_option(4, 1732, 502, Node750, 50);
+                giq_option(4, NAME, 502, Node750, NEUTRAL_REACTION);
             end
-            if (has_skill(dude_obj, 16) > 94) then begin
-                if (global_var(903) == 3) then begin
-                    giq_option(4, 1732, 503, Node850, 50);
+            if (dude_skill(SKILL_GAMBLING) > 94) then begin
+                if (global_var(GVAR_AQE_QUEST_GET_OUR_VAULT_BACK) == 3) then begin
+                    giq_option(4, NAME, 503, Node850, NEUTRAL_REACTION);
                 end
                 else begin
-                    giq_option(4, 1732, 503, Node800, 50);
+                    giq_option(4, NAME, 503, Node800, NEUTRAL_REACTION);
                 end
             end
             else begin
-                giq_option(4, 1732, 503, Node900, 50);
+                giq_option(4, NAME, 503, Node900, NEUTRAL_REACTION);
             end
         end
     end
-    giq_option(4, 1732, 504, Node999, 50);
-    giq_option(-3, 1732, 505, Node999, 50);
+    giq_option(4, NAME, 504, Node999, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 505, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node550
 begin
-    gsay_reply(1732, 550);
-    giq_option(4, 1732, 551, Node999, 50);
+    Reply(550);
+    giq_option(4, NAME, 551, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node600
 begin
-    gsay_reply(1732, 600);
-    giq_option(4, 1732, 601, SET_ALARM, 50);
-    giq_option(4, 1732, 602, Node999, 50);
+    Reply(600);
+    giq_option(4, NAME, 601, SET_ALARM, NEUTRAL_REACTION);
+    giq_option(4, NAME, 602, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node650
 begin
-    gsay_reply(1732, 650);
-    set_global_var(865, global_var(865) bwor 33554432);
-    giq_option(4, 1732, 651, Node999, 50);
+    Reply(650);
+    set_global_var(GVAR_AQ_ALBUQ_EVENTS, global_var(GVAR_AQ_ALBUQ_EVENTS) bwor 33554432);
+    giq_option(4, NAME, 651, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node700
 begin
-    gsay_reply(1732, 700);
-    set_global_var(865, global_var(865) bwor 33554432);
-    giq_option(4, 1732, 701, SET_ALARM, 50);
-    giq_option(4, 1732, 702, Node999, 50);
+    Reply(700);
+    set_global_var(GVAR_AQ_ALBUQ_EVENTS, global_var(GVAR_AQ_ALBUQ_EVENTS) bwor 33554432);
+    giq_option(4, NAME, 701, SET_ALARM, NEUTRAL_REACTION);
+    giq_option(4, NAME, 702, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node750
 begin
-    gsay_reply(1732, 750);
-    giq_option(4, 1732, 751, Node999, 50);
+    Reply(750);
+    giq_option(4, NAME, 751, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node800
 begin
-    gsay_reply(1732, 800);
-    set_global_var(865, global_var(865) bwor 33554432);
-    giq_option(4, 1732, 801, Node999, 50);
+    Reply(800);
+    set_global_var(GVAR_AQ_ALBUQ_EVENTS, global_var(GVAR_AQ_ALBUQ_EVENTS) bwor 33554432);
+    giq_option(4, NAME, 801, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node850
 begin
-    gsay_reply(1732, 850);
-    set_global_var(865, global_var(865) bwor 33554432);
-    giq_option(4, 1732, 851, SET_ALARM, 50);
-    giq_option(4, 1732, 852, Node999, 50);
+    Reply(850);
+    set_global_var(GVAR_AQ_ALBUQ_EVENTS, global_var(GVAR_AQ_ALBUQ_EVENTS) bwor 33554432);
+    giq_option(4, NAME, 851, SET_ALARM, NEUTRAL_REACTION);
+    giq_option(4, NAME, 852, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node900
 begin
-    gsay_reply(1732, 900);
-    giq_option(4, 1732, 901, Node999, 50);
+    Reply(900);
+    giq_option(4, NAME, 901, Node999, NEUTRAL_REACTION);
 end
 
 procedure SET_ALARM
 begin
-    set_global_var(865, global_var(865) bwor 67108864);
+    set_global_var(GVAR_AQ_ALBUQ_EVENTS, global_var(GVAR_AQ_ALBUQ_EVENTS) bwor 67108864);
     add_timer_event(self_obj, 2, 1);
 end
-
