diff --git a/Mods/Resurrection/Original/ALOVGEN.ssl b/Mods/Resurrection/ReDefine/ALOVGEN.ssl
index bd016cb..7e5dee2 100644
--- a/Mods/Resurrection/Original/ALOVGEN.ssl
+++ b/Mods/Resurrection/ReDefine/ALOVGEN.ssl
@@ -39,12 +39,12 @@ begin
             return 1;
         end
     end
-    if ((party_member_obj(16777779) != 0) and ((global_var(910) bwand 32) == 0)) then begin
+    if ((party_member_obj(16777779) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 32) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777779)) <= 5) then begin
             return 1;
         end
     end
-    if ((party_member_obj(16777811) != 0) and ((global_var(910) bwand 64) == 0)) then begin
+    if ((party_member_obj(16777811) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 64) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777811)) <= 5) then begin
             return 1;
         end
@@ -59,7 +59,7 @@ begin
             return 1;
         end
     end
-    if ((party_member_obj(16777814) != 0) and ((global_var(910) bwand 128) == 0)) then begin
+    if ((party_member_obj(16777814) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 128) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777814)) <= 5) then begin
             return 1;
         end
@@ -118,30 +118,30 @@ procedure use_skill_on_p_proc
 begin
     if (map_var(1) == 0) then begin
         if (action_being_used == 12) then begin
-            if (has_skill(dude_obj, 13) > 70) then begin
+            if (dude_skill(SKILL_REPAIR) > 70) then begin
                 script_overrides;
-                display_msg(message_str(1730, 101));
+                display_mstr(101);
             end
             else begin
                 script_overrides;
-                display_msg(message_str(1730, 105));
+                display_mstr(105);
             end
         end
         if (action_being_used == 13) then begin
-            if (has_skill(dude_obj, 13) > 70) then begin
+            if (dude_skill(SKILL_REPAIR) > 70) then begin
                 script_overrides;
-                display_msg(message_str(1730, 101));
+                display_mstr(101);
             end
             else begin
                 script_overrides;
-                display_msg(message_str(1730, 105));
+                display_mstr(105);
             end
         end
     end
     else begin
         if (action_being_used == 13) then begin
             script_overrides;
-            display_msg(message_str(1730, 104));
+            display_mstr(104);
         end
     end
 end
@@ -153,11 +153,11 @@ begin
         LVar0 := obj_pid(obj_being_used_with);
         if (LVar0 == 560) then begin
             script_overrides;
-            if (get_critter_stat(dude_obj, 34) == 0) then begin
-                display_msg(message_str(1730, 102));
+            if (dude_gender == 0) then begin
+                display_mstr(102);
             end
             else begin
-                display_msg(message_str(1730, 103));
+                display_mstr(103);
             end
             removed_qty := obj_is_carrying_obj_pid(dude_obj, 560);
             if (1 < removed_qty) then begin
@@ -167,17 +167,17 @@ begin
                 restock_obj := obj_carrying_pid_obj(dude_obj, 560);
                 if (obj_type(self_obj) == 1) then begin
                     restock_obj := obj_carrying_pid_obj(dude_obj, 560);
-                    if (critter_inven_obj(self_obj, 0) == restock_obj) then begin
-                        if (obj_item_subtype(critter_inven_obj(dude_obj, 0)) == 0) then begin
-                            restock_obj := critter_inven_obj(dude_obj, 0);
+                    if (critter_inven_obj(self_obj, INVEN_TYPE_WORN) == restock_obj) then begin
+                        if (obj_item_subtype(critter_inven_obj(dude_obj, INVEN_TYPE_WORN)) == 0) then begin
+                            restock_obj := critter_inven_obj(dude_obj, INVEN_TYPE_WORN);
                             debug_msg("armour pid == " + obj_pid(restock_obj));
                             rm_obj_from_inven(dude_obj, restock_obj);
                             add_obj_to_inven(dude_obj, restock_obj);
                         end
                     end
                     else begin
-                        if ((critter_inven_obj(dude_obj, 2) == restock_obj) or (critter_inven_obj(dude_obj, 1) == restock_obj)) then begin
-                            metarule(43, dude_obj);
+                        if ((critter_inven_obj(dude_obj, INVEN_TYPE_LEFT_HAND) == restock_obj) or (critter_inven_obj(dude_obj, INVEN_TYPE_RIGHT_HAND) == restock_obj)) then begin
+                            inven_unwield(dude_obj);
                         end
                     end
                 end
@@ -199,16 +199,16 @@ begin
             restock_amt := 0;
             set_map_var(1, 1);
             if (map_var(4) > 0) then begin
-                set_global_var(1014, global_var(1014) + map_var(4));
+                set_global_var(GVAR_AQE_VAULT_HUNTERS_KILLED, global_var(GVAR_AQE_VAULT_HUNTERS_KILLED) + map_var(4));
             end
         end
         if ((LVar0 == 75) or (LVar0 == 308)) then begin
             script_overrides;
-            if (has_skill(dude_obj, 13) > 70) then begin
-                display_msg(message_str(1730, 101));
+            if (dude_skill(SKILL_REPAIR) > 70) then begin
+                display_mstr(101);
             end
             else begin
-                display_msg(message_str(1730, 105));
+                display_mstr(105);
             end
         end
     end
@@ -217,16 +217,15 @@ end
 procedure description_p_proc
 begin
     if (map_var(1) == 0) then begin
-        display_msg(message_str(1730, 100));
+        display_mstr(100);
     end
 end
 
 procedure damage_p_proc
 begin
-    set_global_var(864, 1);
+    set_global_var(GVAR_AQ_ENEMY_HUNTERS, 1);
 end
 
 procedure map_enter_p_proc
 begin
 end
-
