diff --git a/Mods/Resurrection/Original/ALTURRET.ssl b/Mods/Resurrection/ReDefine/ALTURRET.ssl
index 1571363..9ec3a3f 100644
--- a/Mods/Resurrection/Original/ALTURRET.ssl
+++ b/Mods/Resurrection/ReDefine/ALTURRET.ssl
@@ -40,12 +40,12 @@ begin
             return 1;
         end
     end
-    if ((party_member_obj(16777779) != 0) and ((global_var(910) bwand 32) == 0)) then begin
+    if ((party_member_obj(16777779) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 32) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777779)) <= 5) then begin
             return 1;
         end
     end
-    if ((party_member_obj(16777811) != 0) and ((global_var(910) bwand 64) == 0)) then begin
+    if ((party_member_obj(16777811) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 64) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777811)) <= 5) then begin
             return 1;
         end
@@ -60,7 +60,7 @@ begin
             return 1;
         end
     end
-    if ((party_member_obj(16777814) != 0) and ((global_var(910) bwand 128) == 0)) then begin
+    if ((party_member_obj(16777814) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 128) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777814)) <= 5) then begin
             return 1;
         end
@@ -118,17 +118,17 @@ end
 procedure map_enter_p_proc
 begin
     if ((map_var(1) == 1) and (local_var(1) == 1)) then begin
-        if (global_var(903) >= 4) then begin
-            critter_add_trait(self_obj, 1, 6, 282);
+        if (global_var(GVAR_AQE_QUEST_GET_OUR_VAULT_BACK) >= 4) then begin
+            critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 282);
         end
         else begin
-            critter_add_trait(self_obj, 1, 6, 258);
+            critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 258);
         end
-        critter_add_trait(self_obj, 1, 5, 176);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_GUN_TURRET);
     end
     else begin
-        critter_add_trait(self_obj, 1, 6, 291);
-        critter_add_trait(self_obj, 1, 5, 2);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 291);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_PEASANTS_NO_TALKING);
     end
 end
 
@@ -136,15 +136,15 @@ procedure critter_p_proc
 begin
     variable LVar0 := 0;
     if ((map_var(1) == 1) and (local_var(1) == 1)) then begin
-        if ((global_var(864) == 1) and obj_can_see_obj(self_obj, dude_obj) and (global_var(903) != 5) and not((global_var(865) bwand 67108864) != 0)) then begin
+        if ((global_var(GVAR_AQ_ENEMY_HUNTERS) == 1) and obj_can_see_obj(self_obj, dude_obj) and (global_var(GVAR_AQE_QUEST_GET_OUR_VAULT_BACK) != 5) and not((global_var(GVAR_AQ_ALBUQ_EVENTS) bwand 67108864) != 0)) then begin
             if (critter_is_fleeing(self_obj)) then begin
                 if (anim_busy(self_obj) == 0) then begin
                     if ((tile_distance_objs(self_obj, dude_obj) < 8) or obj_can_see_obj(self_obj, dude_obj)) then begin
                         if (anim_busy(self_obj) == 0) then begin
-                            global_temp := rotation_to_tile(tile_num(dude_obj), tile_num(self_obj));
+                            global_temp := rotation_to_tile(dude_tile, tile_num(self_obj));
                             animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(self_obj), global_temp, random(3, 10)), 1);
                             global_temp := (global_temp + 1) % 6;
-                            while ((anim_busy(self_obj) == 0) and (global_temp != rotation_to_tile(tile_num(dude_obj), tile_num(self_obj)))) do begin
+                            while ((anim_busy(self_obj) == 0) and (global_temp != rotation_to_tile(dude_tile, tile_num(self_obj)))) do begin
                                 debug_msg("flee loop: rot == " + global_temp);
                                 animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(self_obj), global_temp, random(3, 10)), 1);
                                 global_temp := (global_temp + 1) % 6;
@@ -154,18 +154,18 @@ begin
                 end
             end
             else begin
-                attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+                attack(dude_obj);
             end
         end
-        if ((global_var(998) == 1) and obj_can_see_obj(self_obj, dude_obj) and (global_var(903) == 5)) then begin
+        if ((global_var(GVAR_AQE_ENEMY_EMPEROR) == 1) and obj_can_see_obj(self_obj, dude_obj) and (global_var(GVAR_AQE_QUEST_GET_OUR_VAULT_BACK) == 5)) then begin
             if (critter_is_fleeing(self_obj)) then begin
                 if (anim_busy(self_obj) == 0) then begin
                     if ((tile_distance_objs(self_obj, dude_obj) < 8) or obj_can_see_obj(self_obj, dude_obj)) then begin
                         if (anim_busy(self_obj) == 0) then begin
-                            global_temp := rotation_to_tile(tile_num(dude_obj), tile_num(self_obj));
+                            global_temp := rotation_to_tile(dude_tile, tile_num(self_obj));
                             animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(self_obj), global_temp, random(3, 10)), 1);
                             global_temp := (global_temp + 1) % 6;
-                            while ((anim_busy(self_obj) == 0) and (global_temp != rotation_to_tile(tile_num(dude_obj), tile_num(self_obj)))) do begin
+                            while ((anim_busy(self_obj) == 0) and (global_temp != rotation_to_tile(dude_tile, tile_num(self_obj)))) do begin
                                 debug_msg("flee loop: rot == " + global_temp);
                                 animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(self_obj), global_temp, random(3, 10)), 1);
                                 global_temp := (global_temp + 1) % 6;
@@ -175,24 +175,24 @@ begin
                 end
             end
             else begin
-                attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+                attack(dude_obj);
             end
         end
-        if ((global_var(865) bwand 4194304) != 0) then begin
+        if ((global_var(GVAR_AQ_ALBUQ_EVENTS) bwand 4194304) != 0) then begin
         end
     end
     if ((map_var(1) == 1) and (local_var(1) == 0)) then begin
-        LVar0 := create_object_sid(520, 0, 0, -1);
+        LVar0 := create_object(520, 0, 0);
         add_obj_to_inven(self_obj, LVar0);
         LVar0 := obj_carrying_pid_obj(self_obj, 520);
-        wield_obj_critter(self_obj, LVar0);
-        if (global_var(903) >= 4) then begin
-            critter_add_trait(self_obj, 1, 6, 282);
+        wield_obj(LVar0);
+        if (global_var(GVAR_AQE_QUEST_GET_OUR_VAULT_BACK) >= 4) then begin
+            critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 282);
         end
         else begin
-            critter_add_trait(self_obj, 1, 6, 258);
+            critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 258);
         end
-        critter_add_trait(self_obj, 1, 5, 176);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_GUN_TURRET);
         set_local_var(1, 1);
     end
 end
@@ -200,11 +200,11 @@ end
 procedure damage_p_proc
 begin
     if (source_obj == dude_obj) then begin
-        if (global_var(903) >= 4) then begin
-            set_global_var(998, 1);
+        if (global_var(GVAR_AQE_QUEST_GET_OUR_VAULT_BACK) >= 4) then begin
+            set_global_var(GVAR_AQE_ENEMY_EMPEROR, 1);
         end
         else begin
-            set_global_var(864, 1);
+            set_global_var(GVAR_AQ_ENEMY_HUNTERS, 1);
         end
     end
 end
@@ -213,10 +213,9 @@ procedure destroy_p_proc
 begin
     set_map_var(4, map_var(4) + 1);
     if (map_var(1) == 1) then begin
-        set_global_var(1014, global_var(1014) + 1);
-        if ((global_var(1014) >= (20 + (2 * map_var(1)))) and ((global_var(865) bwand 268435456) != 0) and (elevation(dude_obj) == 2)) then begin
+        set_global_var(GVAR_AQE_VAULT_HUNTERS_KILLED, global_var(GVAR_AQE_VAULT_HUNTERS_KILLED) + 1);
+        if ((global_var(GVAR_AQE_VAULT_HUNTERS_KILLED) >= (20 + (2 * map_var(1)))) and ((global_var(GVAR_AQ_ALBUQ_EVENTS) bwand 268435456) != 0) and (dude_elevation == 2)) then begin
             add_timer_event(Map_event, game_ticks(0), 25);
         end
     end
 end
-
