diff --git a/Mods/Resurrection/Original/AROBSEC1.ssl b/Mods/Resurrection/ReDefine/AROBSEC1.ssl
index dd9aa04..cd505fd 100644
--- a/Mods/Resurrection/Original/AROBSEC1.ssl
+++ b/Mods/Resurrection/ReDefine/AROBSEC1.ssl
@@ -53,12 +53,12 @@ begin
             return 1;
         end
     end
-    if ((party_member_obj(16777779) != 0) and ((global_var(910) bwand 32) == 0)) then begin
+    if ((party_member_obj(16777779) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 32) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777779)) <= 5) then begin
             return 1;
         end
     end
-    if ((party_member_obj(16777811) != 0) and ((global_var(910) bwand 64) == 0)) then begin
+    if ((party_member_obj(16777811) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 64) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777811)) <= 5) then begin
             return 1;
         end
@@ -73,7 +73,7 @@ begin
             return 1;
         end
     end
-    if ((party_member_obj(16777814) != 0) and ((global_var(910) bwand 128) == 0)) then begin
+    if ((party_member_obj(16777814) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 128) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777814)) <= 5) then begin
             return 1;
         end
@@ -130,36 +130,36 @@ end
 
 procedure critter_p_proc
 begin
-    if (((map_var(0) bwand 262144) != 0) and (has_trait(1, self_obj, 666) == 1)) then begin
+    if (((map_var(0) bwand 262144) != 0) and (obj_is_visible_flag(self_obj) == 1)) then begin
         set_obj_visibility(self_obj, 1);
     end
-    if ((has_trait(1, self_obj, 666) == 0) and not((map_var(0) bwand 262144) != 0)) then begin
+    if ((obj_is_visible_flag(self_obj) == 0) and not((map_var(0) bwand 262144) != 0)) then begin
         set_obj_visibility(self_obj, 0);
     end
-    if (has_trait(1, self_obj, 666) == 1) then begin
-        if ((global_var(1008) >= 9) and ((global_var(904) bwand 1) != 0)) then begin
+    if (obj_is_visible_flag(self_obj) == 1) then begin
+        if ((global_var(GVAR_AQE_HELP_TO_ARAN) >= 9) and ((global_var(GVAR_AQE_ALBUQ_EMPEROR_EVENTS) bwand 1) != 0)) then begin
             set_map_var(2, 0);
             set_map_var(0, map_var(0) bwand (-1 - 1024));
-            critter_add_trait(self_obj, 1, 6, 282);
+            critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 282);
             move_to(self_obj, local_var(5), 0);
         end
         if (map_var(2) == 2) then begin
             if (local_var(4) == 0) then begin
-                critter_add_trait(self_obj, 1, 6, 0);
+                critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 0);
                 set_local_var(4, 1);
             end
         end
         else begin
             call Unplug_Forcfield();
-            if ((global_var(998) == 1) and not((map_var(0) bwand 1024) != 0) and obj_can_see_obj(self_obj, dude_obj) and ((global_var(1017) bwand 2097152) == 0)) then begin
+            if ((global_var(GVAR_AQE_ENEMY_EMPEROR) == 1) and not((map_var(0) bwand 1024) != 0) and obj_can_see_obj(self_obj, dude_obj) and ((global_var(GVAR_AQE_DIRECTIONS_SETUP) bwand 2097152) == 0)) then begin
                 if (critter_is_fleeing(self_obj)) then begin
                     if (anim_busy(self_obj) == 0) then begin
                         if ((tile_distance_objs(self_obj, dude_obj) < 8) or obj_can_see_obj(self_obj, dude_obj)) then begin
                             if (anim_busy(self_obj) == 0) then begin
-                                global_temp := rotation_to_tile(tile_num(dude_obj), tile_num(self_obj));
+                                global_temp := rotation_to_tile(dude_tile, tile_num(self_obj));
                                 animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(self_obj), global_temp, random(3, 10)), 1);
                                 global_temp := (global_temp + 1) % 6;
-                                while ((anim_busy(self_obj) == 0) and (global_temp != rotation_to_tile(tile_num(dude_obj), tile_num(self_obj)))) do begin
+                                while ((anim_busy(self_obj) == 0) and (global_temp != rotation_to_tile(dude_tile, tile_num(self_obj)))) do begin
                                     debug_msg("flee loop: rot == " + global_temp);
                                     animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(self_obj), global_temp, random(3, 10)), 1);
                                     global_temp := (global_temp + 1) % 6;
@@ -169,18 +169,18 @@ begin
                     end
                 end
                 else begin
-                    attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+                    attack(dude_obj);
                 end
             end
-            if ((map_var(2) == 4) and obj_can_see_obj(self_obj, dude_obj) and ((global_var(1017) bwand 2097152) == 0)) then begin
+            if ((map_var(2) == 4) and obj_can_see_obj(self_obj, dude_obj) and ((global_var(GVAR_AQE_DIRECTIONS_SETUP) bwand 2097152) == 0)) then begin
                 if (critter_is_fleeing(self_obj)) then begin
                     if (anim_busy(self_obj) == 0) then begin
                         if ((tile_distance_objs(self_obj, dude_obj) < 8) or obj_can_see_obj(self_obj, dude_obj)) then begin
                             if (anim_busy(self_obj) == 0) then begin
-                                global_temp := rotation_to_tile(tile_num(dude_obj), tile_num(self_obj));
+                                global_temp := rotation_to_tile(dude_tile, tile_num(self_obj));
                                 animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(self_obj), global_temp, random(3, 10)), 1);
                                 global_temp := (global_temp + 1) % 6;
-                                while ((anim_busy(self_obj) == 0) and (global_temp != rotation_to_tile(tile_num(dude_obj), tile_num(self_obj)))) do begin
+                                while ((anim_busy(self_obj) == 0) and (global_temp != rotation_to_tile(dude_tile, tile_num(self_obj)))) do begin
                                     debug_msg("flee loop: rot == " + global_temp);
                                     animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(self_obj), global_temp, random(3, 10)), 1);
                                     global_temp := (global_temp + 1) % 6;
@@ -190,11 +190,11 @@ begin
                     end
                 end
                 else begin
-                    attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+                    attack(dude_obj);
                 end
             end
             if ((local_var(4) == 0) and ((map_var(0) bwand 1024) != 0)) then begin
-                critter_add_trait(self_obj, 1, 6, 242);
+                critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 242);
                 set_local_var(4, 1);
             end
         end
@@ -211,7 +211,7 @@ end
 
 procedure destroy_p_proc
 begin
-    if ((cur_map_index == 25) and ((global_var(1008) == 8) or (global_var(1008) == 4))) then begin
+    if ((CUR_MAP_ALB_CISA) and ((global_var(GVAR_AQE_HELP_TO_ARAN) == 8) or (global_var(GVAR_AQE_HELP_TO_ARAN) == 4))) then begin
         if ((smycka == 0) and (map_var(5) == 0)) then begin
             set_map_var(5, tile_num(self_obj));
             smycka := 1;
@@ -310,58 +310,58 @@ begin
         end
     end
     if (source_obj == dude_obj) then begin
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(4) > (3 * global_var(5))) or (global_var(2) == 1)) and (global_var(1) == 0)) then begin
-            set_global_var(2, 1);
-            set_global_var(3, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(4) > (3 * global_var(GVAR_GOOD_MONSTER))) or (global_var(GVAR_CHAMPION_REPUTATION) == 1)) and (global_var(GVAR_CHILDKILLER_REPUTATION) == 0)) then begin
+            set_global_var(GVAR_CHAMPION_REPUTATION, 1);
+            set_global_var(GVAR_BERSERKER_REPUTATION, 0);
         end
-        if (metarule(51, self_obj) == 2) then begin
+        if (critter_kill_type(self_obj) == KILL_TYPE_children_kills) then begin
             if (source_obj == dude_obj) then begin
-                set_global_var(2, 0);
-                set_global_var(1, global_var(1) + 1);
-                set_global_var(0, global_var(0) + -15);
+                set_global_var(GVAR_CHAMPION_REPUTATION, 0);
+                set_global_var(GVAR_CHILDKILLER_REPUTATION, global_var(GVAR_CHILDKILLER_REPUTATION) + 1);
+                set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + -15);
                 debug_msg("Player gains " + -15 + " Karma Points.");
-                set_global_var(37, 0);
-                set_global_var(38, 0);
-                set_global_var(39, 0);
-                set_global_var(40, 0);
-                set_global_var(41, 0);
-                set_global_var(42, 0);
-                set_global_var(43, 0);
-                set_global_var(44, 0);
-                set_global_var(45, 0);
-                if (global_var(0) >= 1000) then begin
-                    set_global_var(37, 1);
+                set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+                set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+                set_global_var(GVAR_KARMA_DEFENDER, 0);
+                set_global_var(GVAR_KARMA_WANDERER, 0);
+                set_global_var(GVAR_KARMA_BETRAYER, 0);
+                set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+                set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+                set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+                    set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
                 end
                 else begin
-                    if (global_var(0) >= 750) then begin
-                        set_global_var(38, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
                     end
                     else begin
-                        if (global_var(0) >= 500) then begin
-                            set_global_var(39, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                            set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                         end
                         else begin
-                            if (global_var(0) >= 250) then begin
-                                set_global_var(40, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                                set_global_var(GVAR_KARMA_DEFENDER, 1);
                             end
                             else begin
-                                if (global_var(0) > -250) then begin
-                                    set_global_var(41, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                                    set_global_var(GVAR_KARMA_WANDERER, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -500) then begin
-                                        set_global_var(42, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                        set_global_var(GVAR_KARMA_BETRAYER, 1);
                                     end
                                     else begin
-                                        if (global_var(0) > -750) then begin
-                                            set_global_var(43, 1);
+                                        if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                            set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                         end
                                         else begin
-                                            if (global_var(0) > -1000) then begin
-                                                set_global_var(44, 1);
+                                            if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                                set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                             end
                                             else begin
-                                                set_global_var(45, 1);
+                                                set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                             end
                                         end
                                     end
@@ -370,58 +370,58 @@ begin
                         end
                     end
                 end
-                set_global_var(53, global_var(53) + -8);
+                set_global_var(GVAR_TOWN_REP_EMPORER_CITY, global_var(GVAR_TOWN_REP_EMPORER_CITY) + -8);
                 debug_msg("Added " + -8 + " to Town Rep");
             end
         end
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(5) > (2 * global_var(4))) or (global_var(3) == 1))) then begin
-            set_global_var(3, 1);
-            set_global_var(2, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
+            set_global_var(GVAR_BERSERKER_REPUTATION, 1);
+            set_global_var(GVAR_CHAMPION_REPUTATION, 0);
         end
-        set_global_var(0, global_var(0) + 0);
+        set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + 0);
         debug_msg("Player gains " + 0 + " Karma Points.");
-        set_global_var(37, 0);
-        set_global_var(38, 0);
-        set_global_var(39, 0);
-        set_global_var(40, 0);
-        set_global_var(41, 0);
-        set_global_var(42, 0);
-        set_global_var(43, 0);
-        set_global_var(44, 0);
-        set_global_var(45, 0);
-        if (global_var(0) >= 1000) then begin
-            set_global_var(37, 1);
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+        set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+        set_global_var(GVAR_KARMA_DEFENDER, 0);
+        set_global_var(GVAR_KARMA_WANDERER, 0);
+        set_global_var(GVAR_KARMA_BETRAYER, 0);
+        set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+        set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+            set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
         end
         else begin
-            if (global_var(0) >= 750) then begin
-                set_global_var(38, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
             end
             else begin
-                if (global_var(0) >= 500) then begin
-                    set_global_var(39, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                 end
                 else begin
-                    if (global_var(0) >= 250) then begin
-                        set_global_var(40, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                        set_global_var(GVAR_KARMA_DEFENDER, 1);
                     end
                     else begin
-                        if (global_var(0) > -250) then begin
-                            set_global_var(41, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                            set_global_var(GVAR_KARMA_WANDERER, 1);
                         end
                         else begin
-                            if (global_var(0) > -500) then begin
-                                set_global_var(42, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                set_global_var(GVAR_KARMA_BETRAYER, 1);
                             end
                             else begin
-                                if (global_var(0) > -750) then begin
-                                    set_global_var(43, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -1000) then begin
-                                        set_global_var(44, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                     end
                                     else begin
-                                        set_global_var(45, 1);
+                                        set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                     end
                                 end
                             end
@@ -436,26 +436,26 @@ end
 procedure look_at_p_proc
 begin
     script_overrides;
-    display_msg(message_str(119, 100));
+    display_mstr(100);
 end
 
 procedure description_p_proc
 begin
     script_overrides;
     if (map_var(2) == 0) then begin
-        if (get_critter_stat(dude_obj, 4) < 8) then begin
-            display_msg(message_str(119, 200));
+        if (dude_iq < 8) then begin
+            display_mstr(200);
         end
         else begin
-            display_msg(message_str(119, 201));
+            display_mstr(201);
         end
     end
     else begin
-        if (get_critter_stat(dude_obj, 4) < 8) then begin
-            display_msg(message_str(119, 202));
+        if (dude_iq < 8) then begin
+            display_mstr(202);
         end
         else begin
-            display_msg(message_str(119, 203));
+            display_mstr(203);
         end
     end
 end
@@ -468,8 +468,8 @@ procedure damage_p_proc
 begin
     if (source_obj == dude_obj) then begin
         if (map_var(2) != 2) then begin
-            if (global_var(1008) != 8) then begin
-                set_global_var(998, 1);
+            if (global_var(GVAR_AQE_HELP_TO_ARAN) != 8) then begin
+                set_global_var(GVAR_AQE_ENEMY_EMPEROR, 1);
             end
             if ((map_var(0) bwand 1024) != 0) then begin
                 set_map_var(2, 4);
@@ -483,20 +483,20 @@ end
 procedure map_enter_p_proc
 begin
     if (map_var(2) == 2) then begin
-        critter_add_trait(self_obj, 1, 6, 0);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 0);
     end
     else begin
         if ((map_var(0) bwand 1024) != 0) then begin
-            critter_add_trait(self_obj, 1, 6, 242);
+            critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 242);
         end
         else begin
-            critter_add_trait(self_obj, 1, 6, 282);
+            critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 282);
         end
         if (map_var(2) == 4) then begin
-            critter_add_trait(self_obj, 1, 6, 242);
+            critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 242);
         end
     end
-    critter_add_trait(self_obj, 1, 5, 31);
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_SECURITY_BOT);
     if (local_var(5) == 0) then begin
         set_local_var(5, tile_num(self_obj));
     end
@@ -506,12 +506,12 @@ procedure combat_p_proc
 begin
     call Unplug_Forcfield();
     if ((map_var(2) == 4) and (Change_status == 0)) then begin
-        critter_add_trait(self_obj, 1, 6, 242);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 242);
         Change_status := 1;
         set_local_var(4, 0);
     end
     if (fixed_param == 4) then begin
-        if ((has_trait(1, self_obj, 6) == 0) and (global_var(1008) > 2) and (global_var(1008) < 10) and not((global_var(904) bwand 32) != 0)) then begin
+        if ((has_trait(TRAIT_OBJECT, self_obj, OBJECT_TEAM_NUM) == 0) and (global_var(GVAR_AQE_HELP_TO_ARAN) > 2) and (global_var(GVAR_AQE_HELP_TO_ARAN) < 10) and not((global_var(GVAR_AQE_ALBUQ_EMPEROR_EVENTS) bwand 32) != 0)) then begin
             script_overrides;
         end
     end
@@ -519,17 +519,16 @@ end
 
 procedure Unplug_Forcfield
 begin
-    if ((global_var(998) == 1) or (global_var(1008) == 8) and not((map_var(0) bwand 1024) != 0) or (map_var(2) == 4) or (map_var(2) == 2)) then begin
+    if ((global_var(GVAR_AQE_ENEMY_EMPEROR) == 1) or (global_var(GVAR_AQE_HELP_TO_ARAN) == 8) and not((map_var(0) bwand 1024) != 0) or (map_var(2) == 4) or (map_var(2) == 2)) then begin
         if (not((map_var(0) bwand 2048) != 0)) then begin
             add_timer_event(forcefield_1, game_ticks(0), 1);
             add_timer_event(forcefield_2, game_ticks(0), 1);
             add_timer_event(forcefield_3, game_ticks(0), 1);
             add_timer_event(forcefield_4, game_ticks(0), 1);
-            if ((global_var(998) == 1) and (map_var(2) != 2) and not((map_var(0) bwand 1024) != 0)) then begin
+            if ((global_var(GVAR_AQE_ENEMY_EMPEROR) == 1) and (map_var(2) != 2) and not((map_var(0) bwand 1024) != 0)) then begin
                 set_map_var(2, 1);
             end
             set_map_var(0, map_var(0) bwor 2048);
         end
     end
 end
-
