diff --git a/Mods/Resurrection/Original/AROBCOMP.ssl b/Mods/Resurrection/ReDefine/AROBCOMP.ssl
index 358600c..511f547 100644
--- a/Mods/Resurrection/Original/AROBCOMP.ssl
+++ b/Mods/Resurrection/ReDefine/AROBCOMP.ssl
@@ -52,12 +52,12 @@ begin
             return 1;
         end
     end
-    if ((party_member_obj(16777779) != 0) and ((global_var(910) bwand 32) == 0)) then begin
+    if ((party_member_obj(16777779) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 32) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777779)) <= 5) then begin
             return 1;
         end
     end
-    if ((party_member_obj(16777811) != 0) and ((global_var(910) bwand 64) == 0)) then begin
+    if ((party_member_obj(16777811) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 64) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777811)) <= 5) then begin
             return 1;
         end
@@ -72,7 +72,7 @@ begin
             return 1;
         end
     end
-    if ((party_member_obj(16777814) != 0) and ((global_var(910) bwand 128) == 0)) then begin
+    if ((party_member_obj(16777814) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 128) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777814)) <= 5) then begin
             return 1;
         end
@@ -150,11 +150,11 @@ end
 procedure timed_event_p_proc
 begin
     if (fixed_param == 1) then begin
-        anim(self_obj, 1000, rotation_to_tile(tile_num(self_obj), tile_num(dude_obj)));
+        anim(self_obj, 1000, rotation_to_tile(tile_num(self_obj), dude_tile));
         start_gdialog(106, self_obj, 4, -1, -1);
         gsay_start;
-        if ((map_var(2) > 0) or ((global_var(1008) >= 6) and (global_var(1008) <= 8) and (local_var(4) == 0))) then begin
-            if (get_critter_stat(dude_obj, 4) > 3) then begin
+        if ((map_var(2) > 0) or ((global_var(GVAR_AQE_HELP_TO_ARAN) >= 6) and (global_var(GVAR_AQE_HELP_TO_ARAN) <= 8) and (local_var(4) == 0))) then begin
+            if (dude_iq > 3) then begin
                 call Node500();
             end
             else begin
@@ -162,7 +162,7 @@ begin
             end
         end
         else begin
-            if (get_critter_stat(dude_obj, 4) > 3) then begin
+            if (dude_iq > 3) then begin
                 call Node850();
             end
             else begin
@@ -181,13 +181,13 @@ end
 procedure look_at_p_proc
 begin
     script_overrides;
-    display_msg(message_str(106, 100));
+    display_mstr(100);
 end
 
 procedure description_p_proc
 begin
     script_overrides;
-    display_msg(message_str(106, 200));
+    display_mstr(200);
 end
 
 procedure use_skill_on_p_proc
@@ -203,100 +203,99 @@ end
 
 procedure Node500
 begin
-    gsay_reply(106, 500);
+    Reply(500);
     if (is_success(roll_vs_skill(dude_obj, 12, -30))) then begin
-        giq_option(4, 106, 501, Node600, 50);
+        giq_option(4, NAME, 501, Node600, NEUTRAL_REACTION);
     end
     else begin
-        giq_option(4, 106, 501, Node550, 50);
+        giq_option(4, NAME, 501, Node550, NEUTRAL_REACTION);
     end
-    giq_option(4, 106, 502, Node999, 50);
+    giq_option(4, NAME, 502, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node550
 begin
-    gsay_reply(106, 550);
-    giq_option(4, 106, 551, Node999, 50);
+    Reply(550);
+    giq_option(4, NAME, 551, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node600
 begin
-    if (get_critter_stat(dude_obj, 34) == 0) then begin
-        gsay_reply(106, 600);
+    if (dude_gender == 0) then begin
+        Reply(600);
     end
     else begin
-        gsay_reply(106, 601);
+        Reply(601);
     end
     if ((map_var(2) != 2) and (local_var(4) == 0)) then begin
-        giq_option(7, 106, 603, Node700, 50);
+        giq_option(7, NAME, 603, Node700, NEUTRAL_REACTION);
     end
     if (local_var(4) == 0) then begin
-        giq_option(4, 106, 602, Node650, 50);
+        giq_option(4, NAME, 602, Node650, NEUTRAL_REACTION);
     end
-    giq_option(4, 106, 604, Node999, 50);
+    giq_option(4, NAME, 604, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node650
 begin
-    gsay_reply(106, 650);
+    Reply(650);
     set_map_var(2, 0);
     set_map_var(0, map_var(0) bwor 1024);
     set_local_var(4, 1);
-    giq_option(4, 106, 651, Node999, 50);
+    giq_option(4, NAME, 651, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node700
 begin
-    gsay_reply(106, 700);
-    if (has_skill(dude_obj, 12) > 109) then begin
-        giq_option(4, 106, 701, Node800, 50);
+    Reply(700);
+    if (dude_skill(SKILL_SCIENCE) > 109) then begin
+        giq_option(4, NAME, 701, Node800, NEUTRAL_REACTION);
     end
     else begin
-        giq_option(4, 106, 701, Node750, 50);
+        giq_option(4, NAME, 701, Node750, NEUTRAL_REACTION);
     end
-    giq_option(4, 106, 702, Node650, 50);
-    giq_option(4, 106, 703, Node999, 50);
+    giq_option(4, NAME, 702, Node650, NEUTRAL_REACTION);
+    giq_option(4, NAME, 703, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node750
 begin
-    if (get_critter_stat(dude_obj, 34) == 0) then begin
-        gsay_reply(106, 750);
+    if (dude_gender == 0) then begin
+        Reply(750);
     end
     else begin
-        gsay_reply(106, 751);
+        Reply(751);
     end
-    giq_option(4, 106, 752, Node650, 50);
-    giq_option(4, 106, 753, Node999, 50);
+    giq_option(4, NAME, 752, Node650, NEUTRAL_REACTION);
+    giq_option(4, NAME, 753, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node800
 begin
-    if (get_critter_stat(dude_obj, 34) == 0) then begin
-        gsay_reply(106, 800);
+    if (dude_gender == 0) then begin
+        Reply(800);
     end
     else begin
-        gsay_reply(106, 801);
+        Reply(801);
     end
     set_map_var(2, 2);
     set_local_var(4, 1);
-    giq_option(4, 106, 802, Node999, 50);
+    giq_option(4, NAME, 802, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node850
 begin
-    if (get_critter_stat(dude_obj, 4) < 8) then begin
-        gsay_reply(106, 850);
+    if (dude_iq < 8) then begin
+        Reply(850);
     end
     else begin
-        gsay_reply(106, 851);
+        Reply(851);
     end
-    giq_option(4, 106, 852, Node999, 50);
+    giq_option(4, NAME, 852, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node900
 begin
-    gsay_reply(106, 900);
-    giq_option(-3, 106, 901, Node999, 50);
+    Reply(900);
+    giq_option(-3, NAME, 901, Node999, NEUTRAL_REACTION);
 end
-
