diff --git a/Mods/Resurrection/Original/ASMOBEAS.ssl b/Mods/Resurrection/ReDefine/ASMOBEAS.ssl
index 33283ea..ec92f6f 100644
--- a/Mods/Resurrection/Original/ASMOBEAS.ssl
+++ b/Mods/Resurrection/ReDefine/ASMOBEAS.ssl
@@ -46,12 +46,12 @@ begin
             return 1;
         end
     end
-    if ((party_member_obj(16777779) != 0) and ((global_var(910) bwand 32) == 0)) then begin
+    if ((party_member_obj(16777779) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 32) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777779)) <= 5) then begin
             return 1;
         end
     end
-    if ((party_member_obj(16777811) != 0) and ((global_var(910) bwand 64) == 0)) then begin
+    if ((party_member_obj(16777811) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 64) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777811)) <= 5) then begin
             return 1;
         end
@@ -66,7 +66,7 @@ begin
             return 1;
         end
     end
-    if ((party_member_obj(16777814) != 0) and ((global_var(910) bwand 128) == 0)) then begin
+    if ((party_member_obj(16777814) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 128) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777814)) <= 5) then begin
             return 1;
         end
@@ -124,16 +124,16 @@ end
 procedure critter_p_proc
 begin
     if (obj_can_see_obj(self_obj, dude_obj)) then begin
-        attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+        attack(dude_obj);
     end
 end
 
 procedure destroy_p_proc
 begin
     set_map_var(0, map_var(0) + 1);
-    if ((map_var(0) == 10) and (global_var(741) == 1)) then begin
-        set_global_var(741, 2);
-        display_msg(message_str(1439, 100 + (get_critter_stat(dude_obj, 34) == 1)));
+    if ((map_var(0) == 10) and (global_var(GVAR_QUEST_JOHNATAN_CAVE) == 1)) then begin
+        set_global_var(GVAR_QUEST_JOHNATAN_CAVE, 2);
+        display_mstr(100 + (dude_gender == 1));
     end
 end
 
@@ -155,25 +155,25 @@ end
 procedure damage_p_proc
 begin
     if (source_obj == dude_obj) then begin
-        attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+        attack(dude_obj);
     end
 end
 
 procedure map_enter_p_proc
 begin
-    critter_add_trait(self_obj, 1, 6, 232);
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 232);
     if (obj_pid(self_obj) == 16777462) then begin
-        critter_add_trait(self_obj, 1, 5, 132);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_FLOATER);
     end
     if (obj_pid(self_obj) == 16777464) then begin
-        critter_add_trait(self_obj, 1, 5, 133);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_CENTAUR);
     end
     starttile := tile_num(self_obj);
     reg_anim_func(2, self_obj);
     reg_anim_func(1, 1);
     anim(self_obj, 1000, random(0, 5));
     reg_anim_func(3, 0);
-    metarule3(100, self_obj, 1, 0);
+    rm_fixed_timer_event(self_obj, 1);
     add_timer_event(self_obj, game_ticks(random(5, 15)), 1);
 end
 
@@ -186,7 +186,7 @@ begin
             animate_move_obj_to_tile(self_obj, tile_num_in_direction(starttile, random(0, 5), random(1, 5)), 0);
             reg_anim_func(3, 0);
         end
-        metarule3(100, self_obj, 1, 0);
+        rm_fixed_timer_event(self_obj, 1);
         add_timer_event(self_obj, game_ticks(random(5, 15)), 1);
     end
 end
@@ -211,4 +211,3 @@ end
 procedure use_obj_on_p_proc
 begin
 end
-
