diff --git a/Mods/Resurrection/Original/ASOPHIE.ssl b/Mods/Resurrection/ReDefine/ASOPHIE.ssl
index 42539da..3408b28 100644
--- a/Mods/Resurrection/Original/ASOPHIE.ssl
+++ b/Mods/Resurrection/ReDefine/ASOPHIE.ssl
@@ -72,12 +72,12 @@ begin
             return 1;
         end
     end
-    if ((party_member_obj(16777779) != 0) and ((global_var(910) bwand 32) == 0)) then begin
+    if ((party_member_obj(16777779) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 32) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777779)) <= 5) then begin
             return 1;
         end
     end
-    if ((party_member_obj(16777811) != 0) and ((global_var(910) bwand 64) == 0)) then begin
+    if ((party_member_obj(16777811) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 64) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777811)) <= 5) then begin
             return 1;
         end
@@ -92,7 +92,7 @@ begin
             return 1;
         end
     end
-    if ((party_member_obj(16777814) != 0) and ((global_var(910) bwand 128) == 0)) then begin
+    if ((party_member_obj(16777814) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 128) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777814)) <= 5) then begin
             return 1;
         end
@@ -150,15 +150,15 @@ end
 
 procedure critter_p_proc
 begin
-    if ((has_trait(1, self_obj, 666) == 1) and ((global_var(860) == 1) and (cur_map_index == 21)) and obj_can_see_obj(self_obj, dude_obj)) then begin
+    if ((obj_is_visible_flag(self_obj) == 1) and ((global_var(GVAR_AQ_ENEMY_PRISTEH) == 1) and (CUR_MAP_ALB_PRED)) and obj_can_see_obj(self_obj, dude_obj)) then begin
         if (critter_is_fleeing(self_obj)) then begin
             if (anim_busy(self_obj) == 0) then begin
                 if ((tile_distance_objs(self_obj, dude_obj) < 8) or obj_can_see_obj(self_obj, dude_obj)) then begin
                     if (anim_busy(self_obj) == 0) then begin
-                        global_temp := rotation_to_tile(tile_num(dude_obj), tile_num(self_obj));
+                        global_temp := rotation_to_tile(dude_tile, tile_num(self_obj));
                         animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(self_obj), global_temp, random(3, 10)), 1);
                         global_temp := (global_temp + 1) % 6;
-                        while ((anim_busy(self_obj) == 0) and (global_temp != rotation_to_tile(tile_num(dude_obj), tile_num(self_obj)))) do begin
+                        while ((anim_busy(self_obj) == 0) and (global_temp != rotation_to_tile(dude_tile, tile_num(self_obj)))) do begin
                             debug_msg("flee loop: rot == " + global_temp);
                             animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(self_obj), global_temp, random(3, 10)), 1);
                             global_temp := (global_temp + 1) % 6;
@@ -168,18 +168,18 @@ begin
             end
         end
         else begin
-            attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+            attack(dude_obj);
         end
     end
-    if (((global_var(865) bwand 4) != 0) and ((has_trait(1, self_obj, 666) == 1) and ((global_var(998) == 1) and not(cur_map_index == 21))) and obj_can_see_obj(self_obj, dude_obj)) then begin
+    if (((global_var(GVAR_AQ_ALBUQ_EVENTS) bwand 4) != 0) and ((obj_is_visible_flag(self_obj) == 1) and ((global_var(GVAR_AQE_ENEMY_EMPEROR) == 1) and not(CUR_MAP_ALB_PRED))) and obj_can_see_obj(self_obj, dude_obj)) then begin
         if (critter_is_fleeing(self_obj)) then begin
             if (anim_busy(self_obj) == 0) then begin
                 if ((tile_distance_objs(self_obj, dude_obj) < 8) or obj_can_see_obj(self_obj, dude_obj)) then begin
                     if (anim_busy(self_obj) == 0) then begin
-                        global_temp := rotation_to_tile(tile_num(dude_obj), tile_num(self_obj));
+                        global_temp := rotation_to_tile(dude_tile, tile_num(self_obj));
                         animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(self_obj), global_temp, random(3, 10)), 1);
                         global_temp := (global_temp + 1) % 6;
-                        while ((anim_busy(self_obj) == 0) and (global_temp != rotation_to_tile(tile_num(dude_obj), tile_num(self_obj)))) do begin
+                        while ((anim_busy(self_obj) == 0) and (global_temp != rotation_to_tile(dude_tile, tile_num(self_obj)))) do begin
                             debug_msg("flee loop: rot == " + global_temp);
                             animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(self_obj), global_temp, random(3, 10)), 1);
                             global_temp := (global_temp + 1) % 6;
@@ -189,10 +189,10 @@ begin
             end
         end
         else begin
-            attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+            attack(dude_obj);
         end
     end
-    if (((global_var(865) bwand 4) != 0) and (has_trait(1, self_obj, 666) == 0) and not(cur_map_index == 21)) then begin
+    if (((global_var(GVAR_AQ_ALBUQ_EVENTS) bwand 4) != 0) and (obj_is_visible_flag(self_obj) == 0) and not(CUR_MAP_ALB_PRED)) then begin
         set_obj_visibility(self_obj, 0);
     end
 end
@@ -206,15 +206,15 @@ end
 
 procedure talk_p_proc
 begin
-    if ((global_var(859) == 1) and (cur_map_index == 21)) then begin
+    if ((global_var(GVAR_AQ_MUTANT_HUNTER) == 1) and (CUR_MAP_ALB_PRED)) then begin
         add_timer_event(Catherin_obj, game_ticks(0), 1);
     end
     else begin
-        anim(dude_obj, 1000, rotation_to_tile(tile_num(dude_obj), tile_num(self_obj)));
-        anim(self_obj, 1000, rotation_to_tile(tile_num(self_obj), tile_num(dude_obj)));
+        anim(dude_obj, 1000, rotation_to_tile(dude_tile, tile_num(self_obj)));
+        anim(self_obj, 1000, rotation_to_tile(tile_num(self_obj), dude_tile));
         start_gdialog(1625, self_obj, 4, -1, -1);
         gsay_start;
-        if (cur_map_index == 21) then begin
+        if (CUR_MAP_ALB_PRED) then begin
             if ((local_var(6) bwand 1) == 0) then begin
                 call Node500();
             end
@@ -234,59 +234,59 @@ end
 procedure destroy_p_proc
 begin
     if (source_obj == dude_obj) then begin
-        set_global_var(5, global_var(5) + 1);
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(4) > (3 * global_var(5))) or (global_var(2) == 1)) and (global_var(1) == 0)) then begin
-            set_global_var(2, 1);
-            set_global_var(3, 0);
+        set_global_var(GVAR_GOOD_MONSTER, global_var(GVAR_GOOD_MONSTER) + 1);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(4) > (3 * global_var(GVAR_GOOD_MONSTER))) or (global_var(GVAR_CHAMPION_REPUTATION) == 1)) and (global_var(GVAR_CHILDKILLER_REPUTATION) == 0)) then begin
+            set_global_var(GVAR_CHAMPION_REPUTATION, 1);
+            set_global_var(GVAR_BERSERKER_REPUTATION, 0);
         end
-        if (metarule(51, self_obj) == 2) then begin
+        if (critter_kill_type(self_obj) == KILL_TYPE_children_kills) then begin
             if (source_obj == dude_obj) then begin
-                set_global_var(2, 0);
-                set_global_var(1, global_var(1) + 1);
-                set_global_var(0, global_var(0) + -15);
+                set_global_var(GVAR_CHAMPION_REPUTATION, 0);
+                set_global_var(GVAR_CHILDKILLER_REPUTATION, global_var(GVAR_CHILDKILLER_REPUTATION) + 1);
+                set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + -15);
                 debug_msg("Player gains " + -15 + " Karma Points.");
-                set_global_var(37, 0);
-                set_global_var(38, 0);
-                set_global_var(39, 0);
-                set_global_var(40, 0);
-                set_global_var(41, 0);
-                set_global_var(42, 0);
-                set_global_var(43, 0);
-                set_global_var(44, 0);
-                set_global_var(45, 0);
-                if (global_var(0) >= 1000) then begin
-                    set_global_var(37, 1);
+                set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+                set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+                set_global_var(GVAR_KARMA_DEFENDER, 0);
+                set_global_var(GVAR_KARMA_WANDERER, 0);
+                set_global_var(GVAR_KARMA_BETRAYER, 0);
+                set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+                set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+                set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+                    set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
                 end
                 else begin
-                    if (global_var(0) >= 750) then begin
-                        set_global_var(38, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
                     end
                     else begin
-                        if (global_var(0) >= 500) then begin
-                            set_global_var(39, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                            set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                         end
                         else begin
-                            if (global_var(0) >= 250) then begin
-                                set_global_var(40, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                                set_global_var(GVAR_KARMA_DEFENDER, 1);
                             end
                             else begin
-                                if (global_var(0) > -250) then begin
-                                    set_global_var(41, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                                    set_global_var(GVAR_KARMA_WANDERER, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -500) then begin
-                                        set_global_var(42, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                        set_global_var(GVAR_KARMA_BETRAYER, 1);
                                     end
                                     else begin
-                                        if (global_var(0) > -750) then begin
-                                            set_global_var(43, 1);
+                                        if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                            set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                         end
                                         else begin
-                                            if (global_var(0) > -1000) then begin
-                                                set_global_var(44, 1);
+                                            if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                                set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                             end
                                             else begin
-                                                set_global_var(45, 1);
+                                                set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                             end
                                         end
                                     end
@@ -295,58 +295,58 @@ begin
                         end
                     end
                 end
-                set_global_var(51, global_var(51) + -8);
+                set_global_var(GVAR_TOWN_REP_ALBUQUERQUE, global_var(GVAR_TOWN_REP_ALBUQUERQUE) + -8);
                 debug_msg("Added " + -8 + " to Town Rep");
             end
         end
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(5) > (2 * global_var(4))) or (global_var(3) == 1))) then begin
-            set_global_var(3, 1);
-            set_global_var(2, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
+            set_global_var(GVAR_BERSERKER_REPUTATION, 1);
+            set_global_var(GVAR_CHAMPION_REPUTATION, 0);
         end
-        set_global_var(0, global_var(0) + -10);
+        set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + -10);
         debug_msg("Player gains " + -10 + " Karma Points.");
-        set_global_var(37, 0);
-        set_global_var(38, 0);
-        set_global_var(39, 0);
-        set_global_var(40, 0);
-        set_global_var(41, 0);
-        set_global_var(42, 0);
-        set_global_var(43, 0);
-        set_global_var(44, 0);
-        set_global_var(45, 0);
-        if (global_var(0) >= 1000) then begin
-            set_global_var(37, 1);
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+        set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+        set_global_var(GVAR_KARMA_DEFENDER, 0);
+        set_global_var(GVAR_KARMA_WANDERER, 0);
+        set_global_var(GVAR_KARMA_BETRAYER, 0);
+        set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+        set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+            set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
         end
         else begin
-            if (global_var(0) >= 750) then begin
-                set_global_var(38, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
             end
             else begin
-                if (global_var(0) >= 500) then begin
-                    set_global_var(39, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                 end
                 else begin
-                    if (global_var(0) >= 250) then begin
-                        set_global_var(40, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                        set_global_var(GVAR_KARMA_DEFENDER, 1);
                     end
                     else begin
-                        if (global_var(0) > -250) then begin
-                            set_global_var(41, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                            set_global_var(GVAR_KARMA_WANDERER, 1);
                         end
                         else begin
-                            if (global_var(0) > -500) then begin
-                                set_global_var(42, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                set_global_var(GVAR_KARMA_BETRAYER, 1);
                             end
                             else begin
-                                if (global_var(0) > -750) then begin
-                                    set_global_var(43, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -1000) then begin
-                                        set_global_var(44, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                     end
                                     else begin
-                                        set_global_var(45, 1);
+                                        set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                     end
                                 end
                             end
@@ -355,38 +355,38 @@ begin
                 end
             end
         end
-        set_global_var(51, global_var(51) + -5);
+        set_global_var(GVAR_TOWN_REP_ALBUQUERQUE, global_var(GVAR_TOWN_REP_ALBUQUERQUE) + -5);
         debug_msg("Added " + -5 + " to Town Rep");
     end
-    if (not((global_var(865) bwand 4) != 0)) then begin
-        set_global_var(867, global_var(867) bwor 128);
+    if (not((global_var(GVAR_AQ_ALBUQ_EVENTS) bwand 4) != 0)) then begin
+        set_global_var(GVAR_AQ_ALBUQ_DEATHS, global_var(GVAR_AQ_ALBUQ_DEATHS) bwor 128);
     end
 end
 
 procedure look_at_p_proc
 begin
     script_overrides;
-    if (((local_var(6) bwand 1) == 0) and (cur_map_index == 21)) then begin
-        display_msg(message_str(1625, 100));
+    if (((local_var(6) bwand 1) == 0) and (CUR_MAP_ALB_PRED)) then begin
+        display_mstr(100);
     end
     else begin
-        display_msg(message_str(1625, 150));
+        display_mstr(150);
     end
 end
 
 procedure description_p_proc
 begin
     script_overrides;
-    if (cur_map_index == 21) then begin
+    if (CUR_MAP_ALB_PRED) then begin
         if ((local_var(6) bwand 1) == 0) then begin
-            display_msg(message_str(1625, 200));
+            display_mstr(200);
         end
         else begin
-            display_msg(message_str(1625, 250));
+            display_mstr(250);
         end
     end
     else begin
-        display_msg(message_str(1625, 251));
+        display_mstr(251);
     end
 end
 
@@ -397,38 +397,38 @@ end
 procedure damage_p_proc
 begin
     if (source_obj == dude_obj) then begin
-        if (cur_map_index == 21) then begin
-            set_global_var(860, 1);
+        if (CUR_MAP_ALB_PRED) then begin
+            set_global_var(GVAR_AQ_ENEMY_PRISTEH, 1);
         end
         else begin
-            set_global_var(998, 1);
+            set_global_var(GVAR_AQE_ENEMY_EMPEROR, 1);
         end
     end
 end
 
 procedure map_enter_p_proc
 begin
-    if (cur_map_index == 21) then begin
-        critter_add_trait(self_obj, 1, 6, 255);
+    if (CUR_MAP_ALB_PRED) then begin
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 255);
     end
     else begin
-        critter_add_trait(self_obj, 1, 6, 282);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 282);
     end
-    critter_add_trait(self_obj, 1, 5, 27);
-    if (not(metarule(22, 0))) then begin
-        if (((global_var(867) bwand 128) != 0) or ((global_var(865) bwand 2) != 0)) then begin
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_COWARD_NO_TALKING);
+    if (not(is_loading_game)) then begin
+        if (((global_var(GVAR_AQ_ALBUQ_DEATHS) bwand 128) != 0) or ((global_var(GVAR_AQ_ALBUQ_EVENTS) bwand 2) != 0)) then begin
             destroy_object(self_obj);
         end
-        if (cur_map_index == 21) then begin
-            if ((global_var(865) bwand 4) != 0) then begin
+        if (CUR_MAP_ALB_PRED) then begin
+            if ((global_var(GVAR_AQ_ALBUQ_EVENTS) bwand 4) != 0) then begin
                 destroy_object(self_obj);
             end
         end
         else begin
-            if (((global_var(865) bwand 4) != 0) and (has_trait(1, self_obj, 666) == 0)) then begin
+            if (((global_var(GVAR_AQ_ALBUQ_EVENTS) bwand 4) != 0) and (obj_is_visible_flag(self_obj) == 0)) then begin
                 set_obj_visibility(self_obj, 0);
             end
-            if ((has_trait(1, self_obj, 666) == 1) and not((global_var(865) bwand 4) != 0)) then begin
+            if ((obj_is_visible_flag(self_obj) == 1) and not((global_var(GVAR_AQ_ALBUQ_EVENTS) bwand 4) != 0)) then begin
                 set_obj_visibility(self_obj, 1);
             end
         end
@@ -442,7 +442,7 @@ end
 
 procedure Node999
 begin
-    if (cur_map_index == 21) then begin
+    if (CUR_MAP_ALB_PRED) then begin
         anim(self_obj, 1000, 3);
     end
     else begin
@@ -452,142 +452,141 @@ end
 
 procedure Node500
 begin
-    gsay_reply(1625, 500);
-    giq_option(4, 1625, message_str(1625, 501) + obj_name(dude_obj) + message_str(1625, 502), Node600, 50);
-    giq_option(4, 1625, 503, Node999, 50);
-    giq_option(-3, 1625, 504, Node550, 50);
+    Reply(500);
+    giq_option(4, NAME, mstr(501) + dude_name + mstr(502), Node600, NEUTRAL_REACTION);
+    giq_option(4, NAME, 503, Node999, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 504, Node550, NEUTRAL_REACTION);
 end
 
 procedure Node550
 begin
-    gsay_reply(1625, 550);
-    giq_option(-3, 1625, 551, Node999, 50);
+    Reply(550);
+    giq_option(-3, NAME, 551, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node600
 begin
-    gsay_reply(1625, 600);
+    Reply(600);
     set_local_var(6, local_var(6) bwor 1);
-    giq_option(4, 1625, 601, Node650, 50);
-    giq_option(4, 1625, 602, Node700, 50);
+    giq_option(4, NAME, 601, Node650, NEUTRAL_REACTION);
+    giq_option(4, NAME, 602, Node700, NEUTRAL_REACTION);
 end
 
 procedure Node650
 begin
-    gsay_reply(1625, 650);
-    giq_option(4, 1625, 651, Node750, 50);
-    giq_option(4, 1625, 652, Node999, 50);
+    Reply(650);
+    giq_option(4, NAME, 651, Node750, NEUTRAL_REACTION);
+    giq_option(4, NAME, 652, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node700
 begin
-    gsay_reply(1625, 700);
-    giq_option(4, 1625, 701, Node850, 50);
-    giq_option(4, 1625, 702, Node999, 50);
+    Reply(700);
+    giq_option(4, NAME, 701, Node850, NEUTRAL_REACTION);
+    giq_option(4, NAME, 702, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node750
 begin
-    gsay_reply(1625, 750);
-    giq_option(4, 1625, 751, Node800, 50);
-    giq_option(4, 1625, 752, Node999, 50);
+    Reply(750);
+    giq_option(4, NAME, 751, Node800, NEUTRAL_REACTION);
+    giq_option(4, NAME, 752, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node800
 begin
-    gsay_reply(1625, 800);
-    giq_option(4, 1625, 801, Node999, 50);
+    Reply(800);
+    giq_option(4, NAME, 801, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node850
 begin
-    gsay_reply(1625, 850);
-    giq_option(4, 1625, 851, Node750, 50);
-    giq_option(4, 1625, 852, Node999, 50);
+    Reply(850);
+    giq_option(4, NAME, 851, Node750, NEUTRAL_REACTION);
+    giq_option(4, NAME, 852, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node900
 begin
-    gsay_reply(1625, 900);
+    Reply(900);
     if ((local_var(6) bwand 2) == 0) then begin
-        giq_option(4, 1625, 901, Node950, 50);
+        giq_option(4, NAME, 901, Node950, NEUTRAL_REACTION);
     end
     else begin
-        giq_option(4, 1625, 901, Node1050, 50);
+        giq_option(4, NAME, 901, Node1050, NEUTRAL_REACTION);
     end
-    giq_option(4, 1625, 902, Node999, 50);
-    giq_option(-3, 1625, 903, Node550, 50);
+    giq_option(4, NAME, 902, Node999, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 903, Node550, NEUTRAL_REACTION);
 end
 
 procedure Node950
 begin
-    gsay_reply(1625, 950);
+    Reply(950);
     set_local_var(6, local_var(6) bwor 2);
-    giq_option(4, 1625, 951, Node1000, 50);
-    giq_option(4, 1625, 952, Node999, 50);
+    giq_option(4, NAME, 951, Node1000, NEUTRAL_REACTION);
+    giq_option(4, NAME, 952, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node1000
 begin
-    gsay_reply(1625, 1000);
-    giq_option(4, 1625, 1001, Node750, 50);
-    giq_option(4, 1625, 1002, Node999, 50);
+    Reply(1000);
+    giq_option(4, NAME, 1001, Node750, NEUTRAL_REACTION);
+    giq_option(4, NAME, 1002, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node1050
 begin
-    gsay_reply(1625, 1050);
-    giq_option(4, 1625, 1051, Node999, 50);
+    Reply(1050);
+    giq_option(4, NAME, 1051, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node1100
 begin
-    gsay_reply(1625, 1100);
-    if (((local_var(6) bwand 4) == 0) and (global_var(868) == 0)) then begin
-        giq_option(4, 1625, 1101, Node1150, 50);
+    Reply(1100);
+    if (((local_var(6) bwand 4) == 0) and (global_var(GVAR_AQ_SOPHI_CATR_DIALOG) == 0)) then begin
+        giq_option(4, NAME, 1101, Node1150, NEUTRAL_REACTION);
     end
-    if (((local_var(6) bwand 8) == 0) and (global_var(868) == 2)) then begin
-        giq_option(4, 1625, 1101, Node1200, 50);
+    if (((local_var(6) bwand 8) == 0) and (global_var(GVAR_AQ_SOPHI_CATR_DIALOG) == 2)) then begin
+        giq_option(4, NAME, 1101, Node1200, NEUTRAL_REACTION);
     end
-    if (((local_var(6) bwand 16) == 0) and (global_var(868) == 4)) then begin
-        giq_option(4, 1625, 1101, Node1250, 50);
+    if (((local_var(6) bwand 16) == 0) and (global_var(GVAR_AQ_SOPHI_CATR_DIALOG) == 4)) then begin
+        giq_option(4, NAME, 1101, Node1250, NEUTRAL_REACTION);
     end
-    if (((local_var(6) bwand 32) == 0) and (global_var(868) == 6)) then begin
-        giq_option(4, 1625, 1101, Node1300, 50);
+    if (((local_var(6) bwand 32) == 0) and (global_var(GVAR_AQ_SOPHI_CATR_DIALOG) == 6)) then begin
+        giq_option(4, NAME, 1101, Node1300, NEUTRAL_REACTION);
     end
-    giq_option(4, 1625, 1102, Node999, 50);
-    giq_option(-3, 1625, 1103, Node550, 50);
+    giq_option(4, NAME, 1102, Node999, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 1103, Node550, NEUTRAL_REACTION);
 end
 
 procedure Node1150
 begin
-    gsay_reply(1625, 1150);
-    set_global_var(868, 1);
+    Reply(1150);
+    set_global_var(GVAR_AQ_SOPHI_CATR_DIALOG, 1);
     set_local_var(6, local_var(6) bwor 4);
-    giq_option(4, 1625, 1151, Node999, 50);
+    giq_option(4, NAME, 1151, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node1200
 begin
-    gsay_reply(1625, 1200);
-    set_global_var(868, 3);
+    Reply(1200);
+    set_global_var(GVAR_AQ_SOPHI_CATR_DIALOG, 3);
     set_local_var(6, local_var(6) bwor 8);
-    giq_option(4, 1625, 1201, Node999, 50);
+    giq_option(4, NAME, 1201, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node1250
 begin
-    gsay_reply(1625, 1250);
-    set_global_var(868, 5);
+    Reply(1250);
+    set_global_var(GVAR_AQ_SOPHI_CATR_DIALOG, 5);
     set_local_var(6, local_var(6) bwor 16);
-    giq_option(4, 1625, 1251, Node999, 50);
+    giq_option(4, NAME, 1251, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node1300
 begin
-    gsay_reply(1625, 1300);
-    set_global_var(868, 7);
+    Reply(1300);
+    set_global_var(GVAR_AQ_SOPHI_CATR_DIALOG, 7);
     set_local_var(6, local_var(6) bwor 32);
-    giq_option(4, 1625, 1301, Node999, 50);
+    giq_option(4, NAME, 1301, Node999, NEUTRAL_REACTION);
 end
-
