diff --git a/Mods/Resurrection/Original/ATBRCOMP.ssl b/Mods/Resurrection/ReDefine/ATBRCOMP.ssl
index c15f128..e2badae 100644
--- a/Mods/Resurrection/Original/ATBRCOMP.ssl
+++ b/Mods/Resurrection/ReDefine/ATBRCOMP.ssl
@@ -56,12 +56,12 @@ begin
             return 1;
         end
     end
-    if ((party_member_obj(16777779) != 0) and ((global_var(910) bwand 32) == 0)) then begin
+    if ((party_member_obj(16777779) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 32) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777779)) <= 5) then begin
             return 1;
         end
     end
-    if ((party_member_obj(16777811) != 0) and ((global_var(910) bwand 64) == 0)) then begin
+    if ((party_member_obj(16777811) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 64) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777811)) <= 5) then begin
             return 1;
         end
@@ -76,7 +76,7 @@ begin
             return 1;
         end
     end
-    if ((party_member_obj(16777814) != 0) and ((global_var(910) bwand 128) == 0)) then begin
+    if ((party_member_obj(16777814) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 128) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777814)) <= 5) then begin
             return 1;
         end
@@ -143,17 +143,17 @@ procedure use_p_proc
 begin
     script_overrides;
     if (combat_is_initialized) then begin
-        display_msg(message_str(105, 300));
+        display_mstr(300);
     end
     else begin
-        if ((global_var(998) != 1) and not((global_var(1000) bwand 65536) != 0)) then begin
+        if ((global_var(GVAR_AQE_ENEMY_EMPEROR) != 1) and not((global_var(GVAR_AQE_ALBUQ_EMPEROR_EVENTS_NEXT) bwand 65536) != 0)) then begin
             set_map_var(0, map_var(0) bwor 128);
         end
         else begin
-            anim(self_obj, 1000, rotation_to_tile(tile_num(self_obj), tile_num(dude_obj)));
+            anim(self_obj, 1000, rotation_to_tile(tile_num(self_obj), dude_tile));
             start_gdialog(105, self_obj, 4, -1, -1);
             gsay_start;
-            if (get_critter_stat(dude_obj, 4) > 3) then begin
+            if (dude_iq > 3) then begin
                 call Node500();
             end
             else begin
@@ -172,13 +172,13 @@ end
 procedure look_at_p_proc
 begin
     script_overrides;
-    display_msg(message_str(105, 100));
+    display_mstr(100);
 end
 
 procedure description_p_proc
 begin
     script_overrides;
-    display_msg(message_str(105, 200));
+    display_mstr(200);
 end
 
 procedure use_skill_on_p_proc
@@ -194,166 +194,166 @@ end
 
 procedure Node500
 begin
-    gsay_reply(105, 500);
-    if (is_success(roll_vs_skill(dude_obj, 12, 0)) and (has_skill(dude_obj, 12) > 39)) then begin
-        giq_option(4, 105, 501, Node600, 50);
+    Reply(500);
+    if (is_success(roll_vs_skill(dude_obj, 12, 0)) and (dude_skill(SKILL_SCIENCE) > 39)) then begin
+        giq_option(4, NAME, 501, Node600, NEUTRAL_REACTION);
     end
     else begin
-        giq_option(4, 105, 501, Node550, 50);
+        giq_option(4, NAME, 501, Node550, NEUTRAL_REACTION);
     end
-    giq_option(4, 105, 502, Node999, 50);
+    giq_option(4, NAME, 502, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node550
 begin
-    gsay_reply(105, 550);
-    giq_option(4, 105, 551, Node999, 50);
+    Reply(550);
+    giq_option(4, NAME, 551, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node600
 begin
-    if (get_critter_stat(dude_obj, 34) == 0) then begin
-        gsay_reply(105, 600);
+    if (dude_gender == 0) then begin
+        Reply(600);
     end
     else begin
-        gsay_reply(105, 601);
+        Reply(601);
     end
     if (local_var(5) < 3) then begin
-        giq_option(4, 105, 602, Give_Stimpack, 50);
+        giq_option(4, NAME, 602, Give_Stimpack, NEUTRAL_REACTION);
     end
     else begin
-        giq_option(4, 105, 602, Node700, 50);
+        giq_option(4, NAME, 602, Node700, NEUTRAL_REACTION);
     end
     if (local_var(5) < 3) then begin
-        giq_option(4, 105, 603, Give_RadAway, 50);
+        giq_option(4, NAME, 603, Give_RadAway, NEUTRAL_REACTION);
     end
     else begin
-        giq_option(4, 105, 603, Node700, 50);
+        giq_option(4, NAME, 603, Node700, NEUTRAL_REACTION);
     end
     if (local_var(5) < 3) then begin
-        giq_option(4, 105, 604, Give_RadX, 50);
+        giq_option(4, NAME, 604, Give_RadX, NEUTRAL_REACTION);
     end
     else begin
-        giq_option(4, 105, 604, Node700, 50);
+        giq_option(4, NAME, 604, Node700, NEUTRAL_REACTION);
     end
-    if ((global_var(449) == 1) or (global_var(778) == 1) or (global_var(779) == 1) or (global_var(780) == 1)) then begin
-        giq_option(4, 105, 605, Node750, 50);
+    if ((global_var(GVAR_PLAYER_WAS_MARRIED) == 1) or (global_var(GVAR_AIDS_STADIUM_1) == 1) or (global_var(GVAR_AIDS_STADIUM_2) == 1) or (global_var(GVAR_AIDS_STADIUM_3) == 1)) then begin
+        giq_option(4, NAME, 605, Node750, NEUTRAL_REACTION);
     end
-    giq_option(4, 105, 606, Node999, 50);
+    giq_option(4, NAME, 606, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node650
 begin
     if (Stimpack == 1) then begin
-        if (get_critter_stat(dude_obj, 34) == 0) then begin
-            gsay_reply(105, message_str(105, 650) + message_str(105, 652) + message_str(105, 655));
+        if (dude_gender == 0) then begin
+            Reply(mstr(650) + mstr(652) + mstr(655));
         end
         else begin
-            gsay_reply(105, message_str(105, 651) + message_str(105, 652) + message_str(105, 655));
+            Reply(mstr(651) + mstr(652) + mstr(655));
         end
         Stimpack := 0;
     end
     if (RadAway == 1) then begin
-        if (get_critter_stat(dude_obj, 34) == 0) then begin
-            gsay_reply(105, message_str(105, 650) + message_str(105, 653) + message_str(105, 655));
+        if (dude_gender == 0) then begin
+            Reply(mstr(650) + mstr(653) + mstr(655));
         end
         else begin
-            gsay_reply(105, message_str(105, 651) + message_str(105, 653) + message_str(105, 655));
+            Reply(mstr(651) + mstr(653) + mstr(655));
         end
         RadAway := 0;
     end
     if (RadX == 1) then begin
-        if (get_critter_stat(dude_obj, 34) == 0) then begin
-            gsay_reply(105, message_str(105, 650) + message_str(105, 654) + message_str(105, 655));
+        if (dude_gender == 0) then begin
+            Reply(mstr(650) + mstr(654) + mstr(655));
         end
         else begin
-            gsay_reply(105, message_str(105, 651) + message_str(105, 654) + message_str(105, 655));
+            Reply(mstr(651) + mstr(654) + mstr(655));
         end
         RadX := 0;
     end
     set_local_var(5, local_var(5) + 1);
     if (local_var(5) < 3) then begin
-        giq_option(4, 105, 656, Give_Stimpack, 50);
+        giq_option(4, NAME, 656, Give_Stimpack, NEUTRAL_REACTION);
     end
     else begin
-        giq_option(4, 105, 656, Node700, 50);
+        giq_option(4, NAME, 656, Node700, NEUTRAL_REACTION);
     end
     if (local_var(5) < 3) then begin
-        giq_option(4, 105, 657, Give_RadAway, 50);
+        giq_option(4, NAME, 657, Give_RadAway, NEUTRAL_REACTION);
     end
     else begin
-        giq_option(4, 105, 657, Node700, 50);
+        giq_option(4, NAME, 657, Node700, NEUTRAL_REACTION);
     end
     if (local_var(5) < 3) then begin
-        giq_option(4, 105, 658, Give_RadX, 50);
+        giq_option(4, NAME, 658, Give_RadX, NEUTRAL_REACTION);
     end
     else begin
-        giq_option(4, 105, 658, Node700, 50);
+        giq_option(4, NAME, 658, Node700, NEUTRAL_REACTION);
     end
-    if ((global_var(449) == 1) or (global_var(778) == 1) or (global_var(779) == 1) or (global_var(780) == 1)) then begin
-        giq_option(4, 105, 659, Node750, 50);
+    if ((global_var(GVAR_PLAYER_WAS_MARRIED) == 1) or (global_var(GVAR_AIDS_STADIUM_1) == 1) or (global_var(GVAR_AIDS_STADIUM_2) == 1) or (global_var(GVAR_AIDS_STADIUM_3) == 1)) then begin
+        giq_option(4, NAME, 659, Node750, NEUTRAL_REACTION);
     end
-    giq_option(4, 105, 660, Node999, 50);
+    giq_option(4, NAME, 660, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node700
 begin
-    gsay_reply(105, 700);
+    Reply(700);
     set_local_var(5, 5);
-    if ((global_var(449) == 1) or (global_var(778) == 1) or (global_var(779) == 1) or (global_var(780) == 1)) then begin
-        giq_option(4, 105, 701, Node750, 50);
+    if ((global_var(GVAR_PLAYER_WAS_MARRIED) == 1) or (global_var(GVAR_AIDS_STADIUM_1) == 1) or (global_var(GVAR_AIDS_STADIUM_2) == 1) or (global_var(GVAR_AIDS_STADIUM_3) == 1)) then begin
+        giq_option(4, NAME, 701, Node750, NEUTRAL_REACTION);
     end
-    giq_option(4, 105, 702, Node999, 50);
+    giq_option(4, NAME, 702, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node750
 begin
-    if (get_critter_stat(dude_obj, 34) == 0) then begin
-        gsay_reply(105, 750);
+    if (dude_gender == 0) then begin
+        Reply(750);
     end
     else begin
-        gsay_reply(105, 755);
+        Reply(755);
     end
-    set_global_var(449, 0);
-    set_global_var(778, 0);
-    set_global_var(779, 0);
-    set_global_var(780, 0);
-    if (global_var(1034) > 0) then begin
-        set_pc_base_stat(2, get_pc_base_stat(2) + global_var(1034));
+    set_global_var(GVAR_PLAYER_WAS_MARRIED, 0);
+    set_global_var(GVAR_AIDS_STADIUM_1, 0);
+    set_global_var(GVAR_AIDS_STADIUM_2, 0);
+    set_global_var(GVAR_AIDS_STADIUM_3, 0);
+    if (global_var(GVAR_AIDS_ENDURANCE_LOSS) > 0) then begin
+        set_pc_base_stat(2, get_pc_base_stat(2) + global_var(GVAR_AIDS_ENDURANCE_LOSS));
     end
-    if (global_var(1035) > 0) then begin
-        set_pc_base_stat(3, get_pc_base_stat(3) + global_var(1035));
+    if (global_var(GVAR_AIDS_CHARISMA_LOSS) > 0) then begin
+        set_pc_base_stat(3, get_pc_base_stat(3) + global_var(GVAR_AIDS_CHARISMA_LOSS));
     end
     if (local_var(5) < 3) then begin
-        giq_option(4, 105, 751, Give_Stimpack, 50);
+        giq_option(4, NAME, 751, Give_Stimpack, NEUTRAL_REACTION);
     end
     else begin
-        giq_option(4, 105, 751, Node700, 50);
+        giq_option(4, NAME, 751, Node700, NEUTRAL_REACTION);
     end
     if (local_var(5) < 3) then begin
-        giq_option(4, 105, 752, Give_RadAway, 50);
+        giq_option(4, NAME, 752, Give_RadAway, NEUTRAL_REACTION);
     end
     else begin
-        giq_option(4, 105, 752, Node700, 50);
+        giq_option(4, NAME, 752, Node700, NEUTRAL_REACTION);
     end
     if (local_var(5) < 3) then begin
-        giq_option(4, 105, 753, Give_RadX, 50);
+        giq_option(4, NAME, 753, Give_RadX, NEUTRAL_REACTION);
     end
     else begin
-        giq_option(4, 105, 753, Node700, 50);
+        giq_option(4, NAME, 753, Node700, NEUTRAL_REACTION);
     end
-    giq_option(4, 105, 754, Node999, 50);
+    giq_option(4, NAME, 754, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node800
 begin
-    gsay_reply(105, 800);
-    giq_option(-3, 105, 801, Node999, 50);
+    Reply(800);
+    giq_option(-3, NAME, 801, Node999, NEUTRAL_REACTION);
 end
 
 procedure Give_Stimpack
 begin
-    restock_obj := create_object_sid(40, 0, 0, -1);
+    restock_obj := create_object(40, 0, 0);
     add_mult_objs_to_inven(dude_obj, restock_obj, 1);
     Stimpack := 1;
     call Node650();
@@ -361,7 +361,7 @@ end
 
 procedure Give_RadAway
 begin
-    restock_obj := create_object_sid(48, 0, 0, -1);
+    restock_obj := create_object(48, 0, 0);
     add_mult_objs_to_inven(dude_obj, restock_obj, 1);
     RadAway := 1;
     call Node650();
@@ -369,9 +369,8 @@ end
 
 procedure Give_RadX
 begin
-    restock_obj := create_object_sid(109, 0, 0, -1);
+    restock_obj := create_object(109, 0, 0);
     add_mult_objs_to_inven(dude_obj, restock_obj, 1);
     RadX := 1;
     call Node650();
 end
-
