diff --git a/Mods/Resurrection/Original/AUNOSCE1.ssl b/Mods/Resurrection/ReDefine/AUNOSCE1.ssl
index d55027f..bea606d 100644
--- a/Mods/Resurrection/Original/AUNOSCE1.ssl
+++ b/Mods/Resurrection/ReDefine/AUNOSCE1.ssl
@@ -67,12 +67,12 @@ begin
             return 1;
         end
     end
-    if ((party_member_obj(16777779) != 0) and ((global_var(910) bwand 32) == 0)) then begin
+    if ((party_member_obj(16777779) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 32) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777779)) <= 5) then begin
             return 1;
         end
     end
-    if ((party_member_obj(16777811) != 0) and ((global_var(910) bwand 64) == 0)) then begin
+    if ((party_member_obj(16777811) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 64) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777811)) <= 5) then begin
             return 1;
         end
@@ -87,7 +87,7 @@ begin
             return 1;
         end
     end
-    if ((party_member_obj(16777814) != 0) and ((global_var(910) bwand 128) == 0)) then begin
+    if ((party_member_obj(16777814) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 128) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777814)) <= 5) then begin
             return 1;
         end
@@ -144,15 +144,15 @@ end
 
 procedure critter_p_proc
 begin
-    if ((local_var(4) == 1) and (tile_distance(tile_num(self_obj), tile_num(dude_obj)) < 15) and (has_trait(1, self_obj, 666) == 1) and obj_can_see_obj(self_obj, dude_obj)) then begin
+    if ((local_var(4) == 1) and (tile_distance(tile_num(self_obj), dude_tile) < 15) and (obj_is_visible_flag(self_obj) == 1) and obj_can_see_obj(self_obj, dude_obj)) then begin
         if (critter_is_fleeing(self_obj)) then begin
             if (anim_busy(self_obj) == 0) then begin
                 if ((tile_distance_objs(self_obj, dude_obj) < 8) or obj_can_see_obj(self_obj, dude_obj)) then begin
                     if (anim_busy(self_obj) == 0) then begin
-                        global_temp := rotation_to_tile(tile_num(dude_obj), tile_num(self_obj));
+                        global_temp := rotation_to_tile(dude_tile, tile_num(self_obj));
                         animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(self_obj), global_temp, random(3, 10)), 1);
                         global_temp := (global_temp + 1) % 6;
-                        while ((anim_busy(self_obj) == 0) and (global_temp != rotation_to_tile(tile_num(dude_obj), tile_num(self_obj)))) do begin
+                        while ((anim_busy(self_obj) == 0) and (global_temp != rotation_to_tile(dude_tile, tile_num(self_obj)))) do begin
                             debug_msg("flee loop: rot == " + global_temp);
                             animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(self_obj), global_temp, random(3, 10)), 1);
                             global_temp := (global_temp + 1) % 6;
@@ -162,7 +162,7 @@ begin
             end
         end
         else begin
-            attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+            attack(dude_obj);
         end
     end
 end
@@ -176,12 +176,12 @@ end
 
 procedure talk_p_proc
 begin
-    if (global_var(723) >= 6) then begin
-        float_msg(self_obj, message_str(1676, 300), 8);
+    if (global_var(GVAR_QUEST_S_DAVID_KIDNAP_INVESTIGATION) >= 6) then begin
+        float_msg(self_obj, mstr(300), FLOAT_MSG_YELLOW);
     end
     else begin
-        anim(dude_obj, 1000, rotation_to_tile(tile_num(dude_obj), tile_num(self_obj)));
-        anim(self_obj, 1000, rotation_to_tile(tile_num(self_obj), tile_num(dude_obj)));
+        anim(dude_obj, 1000, rotation_to_tile(dude_tile, tile_num(self_obj)));
+        anim(self_obj, 1000, rotation_to_tile(tile_num(self_obj), dude_tile));
         start_gdialog(1676, self_obj, 4, -1, -1);
         gsay_start;
         call Node1150();
@@ -193,55 +193,55 @@ end
 procedure destroy_p_proc
 begin
     if (source_obj == dude_obj) then begin
-        set_global_var(4, global_var(4) + 1);
-        if (metarule(51, self_obj) == 2) then begin
+        set_global_var(GVAR_BAD_MONSTER, global_var(GVAR_BAD_MONSTER) + 1);
+        if (critter_kill_type(self_obj) == KILL_TYPE_children_kills) then begin
             if (source_obj == dude_obj) then begin
-                set_global_var(2, 0);
-                set_global_var(1, global_var(1) + 1);
-                set_global_var(0, global_var(0) + -15);
+                set_global_var(GVAR_CHAMPION_REPUTATION, 0);
+                set_global_var(GVAR_CHILDKILLER_REPUTATION, global_var(GVAR_CHILDKILLER_REPUTATION) + 1);
+                set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + -15);
                 debug_msg("Player gains " + -15 + " Karma Points.");
-                set_global_var(37, 0);
-                set_global_var(38, 0);
-                set_global_var(39, 0);
-                set_global_var(40, 0);
-                set_global_var(41, 0);
-                set_global_var(42, 0);
-                set_global_var(43, 0);
-                set_global_var(44, 0);
-                set_global_var(45, 0);
-                if (global_var(0) >= 1000) then begin
-                    set_global_var(37, 1);
+                set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+                set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+                set_global_var(GVAR_KARMA_DEFENDER, 0);
+                set_global_var(GVAR_KARMA_WANDERER, 0);
+                set_global_var(GVAR_KARMA_BETRAYER, 0);
+                set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+                set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+                set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+                    set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
                 end
                 else begin
-                    if (global_var(0) >= 750) then begin
-                        set_global_var(38, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
                     end
                     else begin
-                        if (global_var(0) >= 500) then begin
-                            set_global_var(39, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                            set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                         end
                         else begin
-                            if (global_var(0) >= 250) then begin
-                                set_global_var(40, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                                set_global_var(GVAR_KARMA_DEFENDER, 1);
                             end
                             else begin
-                                if (global_var(0) > -250) then begin
-                                    set_global_var(41, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                                    set_global_var(GVAR_KARMA_WANDERER, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -500) then begin
-                                        set_global_var(42, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                        set_global_var(GVAR_KARMA_BETRAYER, 1);
                                     end
                                     else begin
-                                        if (global_var(0) > -750) then begin
-                                            set_global_var(43, 1);
+                                        if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                            set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                         end
                                         else begin
-                                            if (global_var(0) > -1000) then begin
-                                                set_global_var(44, 1);
+                                            if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                                set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                             end
                                             else begin
-                                                set_global_var(45, 1);
+                                                set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                             end
                                         end
                                     end
@@ -250,62 +250,62 @@ begin
                         end
                     end
                 end
-                set_global_var(51, global_var(51) + -8);
+                set_global_var(GVAR_TOWN_REP_ALBUQUERQUE, global_var(GVAR_TOWN_REP_ALBUQUERQUE) + -8);
                 debug_msg("Added " + -8 + " to Town Rep");
             end
         end
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(5) > (2 * global_var(4))) or (global_var(3) == 1))) then begin
-            set_global_var(3, 1);
-            set_global_var(2, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
+            set_global_var(GVAR_BERSERKER_REPUTATION, 1);
+            set_global_var(GVAR_CHAMPION_REPUTATION, 0);
         end
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(4) > (3 * global_var(5))) or (global_var(2) == 1)) and (global_var(1) == 0)) then begin
-            set_global_var(2, 1);
-            set_global_var(3, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(4) > (3 * global_var(GVAR_GOOD_MONSTER))) or (global_var(GVAR_CHAMPION_REPUTATION) == 1)) and (global_var(GVAR_CHILDKILLER_REPUTATION) == 0)) then begin
+            set_global_var(GVAR_CHAMPION_REPUTATION, 1);
+            set_global_var(GVAR_BERSERKER_REPUTATION, 0);
         end
-        set_global_var(0, global_var(0) + 5);
+        set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + 5);
         debug_msg("Player gains " + 5 + " Karma Points.");
-        set_global_var(37, 0);
-        set_global_var(38, 0);
-        set_global_var(39, 0);
-        set_global_var(40, 0);
-        set_global_var(41, 0);
-        set_global_var(42, 0);
-        set_global_var(43, 0);
-        set_global_var(44, 0);
-        set_global_var(45, 0);
-        if (global_var(0) >= 1000) then begin
-            set_global_var(37, 1);
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+        set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+        set_global_var(GVAR_KARMA_DEFENDER, 0);
+        set_global_var(GVAR_KARMA_WANDERER, 0);
+        set_global_var(GVAR_KARMA_BETRAYER, 0);
+        set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+        set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+            set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
         end
         else begin
-            if (global_var(0) >= 750) then begin
-                set_global_var(38, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
             end
             else begin
-                if (global_var(0) >= 500) then begin
-                    set_global_var(39, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                 end
                 else begin
-                    if (global_var(0) >= 250) then begin
-                        set_global_var(40, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                        set_global_var(GVAR_KARMA_DEFENDER, 1);
                     end
                     else begin
-                        if (global_var(0) > -250) then begin
-                            set_global_var(41, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                            set_global_var(GVAR_KARMA_WANDERER, 1);
                         end
                         else begin
-                            if (global_var(0) > -500) then begin
-                                set_global_var(42, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                set_global_var(GVAR_KARMA_BETRAYER, 1);
                             end
                             else begin
-                                if (global_var(0) > -750) then begin
-                                    set_global_var(43, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -1000) then begin
-                                        set_global_var(44, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                     end
                                     else begin
-                                        set_global_var(45, 1);
+                                        set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                     end
                                 end
                             end
@@ -314,7 +314,7 @@ begin
                 end
             end
         end
-        set_global_var(51, global_var(51) + 2);
+        set_global_var(GVAR_TOWN_REP_ALBUQUERQUE, global_var(GVAR_TOWN_REP_ALBUQUERQUE) + 2);
         debug_msg("Added " + 2 + " to Town Rep");
     end
 end
@@ -322,13 +322,13 @@ end
 procedure look_at_p_proc
 begin
     script_overrides;
-    display_msg(message_str(1676, 100));
+    display_mstr(100);
 end
 
 procedure description_p_proc
 begin
     script_overrides;
-    display_msg(message_str(1676, 200));
+    display_mstr(200);
 end
 
 procedure use_skill_on_p_proc
@@ -344,22 +344,22 @@ end
 
 procedure map_enter_p_proc
 begin
-    critter_add_trait(self_obj, 1, 6, 266);
-    critter_add_trait(self_obj, 1, 5, 13);
-    if (global_var(723) == 5) then begin
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 266);
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_THUGS);
+    if (global_var(GVAR_QUEST_S_DAVID_KIDNAP_INVESTIGATION) == 5) then begin
         set_obj_visibility(self_obj, 0);
     end
     else begin
         set_obj_visibility(self_obj, 1);
     end
-    if (global_var(723) >= 6) then begin
+    if (global_var(GVAR_QUEST_S_DAVID_KIDNAP_INVESTIGATION) >= 6) then begin
         destroy_object(self_obj);
     end
 end
 
 procedure timed_event_p_proc
 begin
-    if (global_var(723) >= 6) then begin
+    if (global_var(GVAR_QUEST_S_DAVID_KIDNAP_INVESTIGATION) >= 6) then begin
         if (fixed_param == 10) then begin
             sequence := sequence + 1;
         end
@@ -381,7 +381,7 @@ begin
             add_timer_event(self_obj, game_ticks(1), 10);
         end
         if (sequence == 5) then begin
-            if ((party_member_obj(16777811) != 0) and ((global_var(910) bwand 64) == 0)) then begin
+            if ((party_member_obj(16777811) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 64) == 0)) then begin
                 add_timer_event(party_member_obj(16777811), game_ticks(1), 21);
             end
             destroy_object(self_obj);
@@ -391,11 +391,11 @@ end
 
 procedure Node998
 begin
-    if ((global_var(867) bwand 16) != 0) then begin
-        float_msg(self_obj, message_str(1676, 302), 2);
+    if ((global_var(GVAR_AQ_ALBUQ_DEATHS) bwand 16) != 0) then begin
+        float_msg(self_obj, mstr(302), FLOAT_MSG_RED);
     end
     else begin
-        float_msg(self_obj, message_str(1676, 301), 2);
+        float_msg(self_obj, mstr(301), FLOAT_MSG_RED);
     end
     set_local_var(4, 1);
 end
@@ -413,98 +413,98 @@ end
 
 procedure Node1150
 begin
-    gsay_reply(1676, 1150);
-    if (global_var(723) == 5) then begin
-        giq_option(4, 1676, 1151, Node1200, 50);
+    Reply(1150);
+    if (global_var(GVAR_QUEST_S_DAVID_KIDNAP_INVESTIGATION) == 5) then begin
+        giq_option(4, NAME, 1151, Node1200, NEUTRAL_REACTION);
     end
-    giq_option(4, 1676, 1152, Node999, 50);
-    giq_option(-3, 1676, 1153, Node999, 50);
+    giq_option(4, NAME, 1152, Node999, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 1153, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node1200
 begin
-    gsay_reply(1676, 1200);
-    giq_option(4, 1676, 1201, Node1250, 50);
+    Reply(1200);
+    giq_option(4, NAME, 1201, Node1250, NEUTRAL_REACTION);
 end
 
 procedure Node1250
 begin
-    gsay_reply(1676, 1250);
-    if ((get_critter_stat(dude_obj, 0) > 7) and (get_critter_stat(dude_obj, 5) > 7)) then begin
-        giq_option(4, 1676, 1251, Node1350, 50);
+    Reply(1250);
+    if ((dude_strength > 7) and (get_critter_stat(dude_obj, STAT_ag) > 7)) then begin
+        giq_option(4, NAME, 1251, Node1350, NEUTRAL_REACTION);
     end
     else begin
-        giq_option(4, 1676, 1251, Node1300, 50);
+        giq_option(4, NAME, 1251, Node1300, NEUTRAL_REACTION);
     end
-    giq_option(4, 1676, 1252, Node1400, 50);
-    giq_option(4, 1676, 1253, Node998, 50);
+    giq_option(4, NAME, 1252, Node1400, NEUTRAL_REACTION);
+    giq_option(4, NAME, 1253, Node998, NEUTRAL_REACTION);
 end
 
 procedure Node1300
 begin
-    gsay_reply(1676, 1300);
-    giq_option(4, 1676, 1301, Node998, 50);
+    Reply(1300);
+    giq_option(4, NAME, 1301, Node998, NEUTRAL_REACTION);
 end
 
 procedure Node1350
 begin
-    if ((global_var(867) bwand 16) != 0) then begin
-        gsay_reply(1676, 1352);
+    if ((global_var(GVAR_AQ_ALBUQ_DEATHS) bwand 16) != 0) then begin
+        Reply(1352);
     end
     else begin
-        gsay_reply(1676, 1350);
+        Reply(1350);
     end
-    set_global_var(723, 6);
-    if ((party_member_obj(16777814) != 0) and ((global_var(910) bwand 128) == 0)) then begin
-        set_global_var(1019, global_var(1019) bwor 8192);
+    set_global_var(GVAR_QUEST_S_DAVID_KIDNAP_INVESTIGATION, 6);
+    if ((party_member_obj(16777814) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 128) == 0)) then begin
+        set_global_var(GVAR_PTR_LYSTRA_REACTIONS_1, global_var(GVAR_PTR_LYSTRA_REACTIONS_1) bwor 8192);
     end
-    giq_option(4, 1676, 1351, Node999_quest, 50);
+    giq_option(4, NAME, 1351, Node999_quest, NEUTRAL_REACTION);
 end
 
 procedure Node1400
 begin
-    gsay_reply(1676, 1400);
+    Reply(1400);
     if (item_caps_total(dude_obj) > 999) then begin
-        if (has_skill(dude_obj, 15) > 70) then begin
-            giq_option(4, 1676, 1401, Node1450_1000, 50);
+        if (dude_skill(SKILL_BARTER) > 70) then begin
+            giq_option(4, NAME, 1401, Node1450_1000, NEUTRAL_REACTION);
         end
         else begin
-            giq_option(4, 1676, 1401, Node1500, 50);
+            giq_option(4, NAME, 1401, Node1500, NEUTRAL_REACTION);
         end
     end
     if (item_caps_total(dude_obj) > 1999) then begin
-        if (has_skill(dude_obj, 15) > 40) then begin
-            giq_option(4, 1676, 1402, Node1450_2000, 50);
+        if (dude_skill(SKILL_BARTER) > 40) then begin
+            giq_option(4, NAME, 1402, Node1450_2000, NEUTRAL_REACTION);
         end
         else begin
-            giq_option(4, 1676, 1402, Node1500, 50);
+            giq_option(4, NAME, 1402, Node1500, NEUTRAL_REACTION);
         end
     end
     if (item_caps_total(dude_obj) > 2999) then begin
-        giq_option(4, 1676, 1403, Node1450_3000, 50);
+        giq_option(4, NAME, 1403, Node1450_3000, NEUTRAL_REACTION);
     end
-    giq_option(4, 1676, 1404, Node998, 50);
+    giq_option(4, NAME, 1404, Node998, NEUTRAL_REACTION);
 end
 
 procedure Node1450
 begin
-    if ((global_var(867) bwand 16) != 0) then begin
-        gsay_reply(1676, 1452);
+    if ((global_var(GVAR_AQ_ALBUQ_DEATHS) bwand 16) != 0) then begin
+        Reply(1452);
     end
     else begin
-        gsay_reply(1676, 1450);
+        Reply(1450);
     end
-    set_global_var(723, 6);
-    if ((party_member_obj(16777814) != 0) and ((global_var(910) bwand 128) == 0)) then begin
-        set_global_var(1019, global_var(1019) bwor 8192);
+    set_global_var(GVAR_QUEST_S_DAVID_KIDNAP_INVESTIGATION, 6);
+    if ((party_member_obj(16777814) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 128) == 0)) then begin
+        set_global_var(GVAR_PTR_LYSTRA_REACTIONS_1, global_var(GVAR_PTR_LYSTRA_REACTIONS_1) bwor 8192);
     end
-    giq_option(4, 1676, 1451, Node999_quest, 50);
+    giq_option(4, NAME, 1451, Node999_quest, NEUTRAL_REACTION);
 end
 
 procedure Node1500
 begin
-    gsay_reply(1676, 1500);
-    giq_option(4, 1676, 1501, Node998, 50);
+    Reply(1500);
+    giq_option(4, NAME, 1501, Node998, NEUTRAL_REACTION);
 end
 
 procedure Node1450_1000
@@ -524,4 +524,3 @@ begin
     item_caps_adjust(dude_obj, -3000);
     call Node1450();
 end
-
