diff --git a/Mods/Resurrection/Original/AVSACHTA.ssl b/Mods/Resurrection/ReDefine/AVSACHTA.ssl
index 0557f3e..8e8c743 100644
--- a/Mods/Resurrection/Original/AVSACHTA.ssl
+++ b/Mods/Resurrection/ReDefine/AVSACHTA.ssl
@@ -38,12 +38,12 @@ begin
             return 1;
         end
     end
-    if ((party_member_obj(16777779) != 0) and ((global_var(910) bwand 32) == 0)) then begin
+    if ((party_member_obj(16777779) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 32) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777779)) <= 5) then begin
             return 1;
         end
     end
-    if ((party_member_obj(16777811) != 0) and ((global_var(910) bwand 64) == 0)) then begin
+    if ((party_member_obj(16777811) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 64) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777811)) <= 5) then begin
             return 1;
         end
@@ -58,7 +58,7 @@ begin
             return 1;
         end
     end
-    if ((party_member_obj(16777814) != 0) and ((global_var(910) bwand 128) == 0)) then begin
+    if ((party_member_obj(16777814) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 128) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777814)) <= 5) then begin
             return 1;
         end
@@ -119,8 +119,8 @@ begin
     variable LVar0 := 0;
     LVar0 := obj_pid(obj_being_used_with);
     if (LVar0 == 127) then begin
-        create_object_sid(33555334, 21912, 1, 1746);
-        create_object_sid(33555333, 16712, 2, 1746);
+        create_object_sid(33555334, 21912, 1, SCRIPT_AVTROPE2);
+        create_object_sid(33555333, 16712, 2, SCRIPT_AVTROPE2);
         if (obj_is_carrying_obj_pid(dude_obj, 127) >= 1) then begin
             removed_qty := obj_is_carrying_obj_pid(dude_obj, 127);
         end
@@ -131,17 +131,17 @@ begin
             restock_obj := obj_carrying_pid_obj(dude_obj, 127);
             if (obj_type(self_obj) == 1) then begin
                 restock_obj := obj_carrying_pid_obj(dude_obj, 127);
-                if (critter_inven_obj(self_obj, 0) == restock_obj) then begin
-                    if (obj_item_subtype(critter_inven_obj(dude_obj, 0)) == 0) then begin
-                        restock_obj := critter_inven_obj(dude_obj, 0);
+                if (critter_inven_obj(self_obj, INVEN_TYPE_WORN) == restock_obj) then begin
+                    if (obj_item_subtype(critter_inven_obj(dude_obj, INVEN_TYPE_WORN)) == 0) then begin
+                        restock_obj := critter_inven_obj(dude_obj, INVEN_TYPE_WORN);
                         debug_msg("armour pid == " + obj_pid(restock_obj));
                         rm_obj_from_inven(dude_obj, restock_obj);
                         add_obj_to_inven(dude_obj, restock_obj);
                     end
                 end
                 else begin
-                    if ((critter_inven_obj(dude_obj, 2) == restock_obj) or (critter_inven_obj(dude_obj, 1) == restock_obj)) then begin
-                        metarule(43, dude_obj);
+                    if ((critter_inven_obj(dude_obj, INVEN_TYPE_LEFT_HAND) == restock_obj) or (critter_inven_obj(dude_obj, INVEN_TYPE_RIGHT_HAND) == restock_obj)) then begin
+                        inven_unwield(dude_obj);
                     end
                 end
             end
@@ -162,8 +162,8 @@ begin
         restock_obj := 0;
         restock_amt := 0;
         add_timer_event(vytah_sachta, game_ticks(0), 1);
-        if (not((global_var(904) bwand 4) != 0)) then begin
-            set_global_var(904, global_var(904) bwor 4);
+        if (not((global_var(GVAR_AQE_ALBUQ_EMPEROR_EVENTS) bwand 4) != 0)) then begin
+            set_global_var(GVAR_AQE_ALBUQ_EMPEROR_EVENTS, global_var(GVAR_AQE_ALBUQ_EMPEROR_EVENTS) bwor 4);
         end
         destroy_object(self_obj);
     end
@@ -172,13 +172,12 @@ end
 procedure timed_event_p_proc
 begin
     if (fixed_param == 1) then begin
-        create_object_sid(33555334, 21912, 1, 1746);
-        create_object_sid(33555333, 16712, 2, 1746);
+        create_object_sid(33555334, 21912, 1, SCRIPT_AVTROPE2);
+        create_object_sid(33555333, 16712, 2, SCRIPT_AVTROPE2);
         add_timer_event(vytah_sachta, game_ticks(0), 1);
-        if (not((global_var(904) bwand 4) != 0)) then begin
-            set_global_var(904, global_var(904) bwor 4);
+        if (not((global_var(GVAR_AQE_ALBUQ_EMPEROR_EVENTS) bwand 4) != 0)) then begin
+            set_global_var(GVAR_AQE_ALBUQ_EMPEROR_EVENTS, global_var(GVAR_AQE_ALBUQ_EMPEROR_EVENTS) bwor 4);
         end
         destroy_object(self_obj);
     end
 end
-
