diff --git a/Mods/Resurrection/Original/AWILLIAM.ssl b/Mods/Resurrection/ReDefine/AWILLIAM.ssl
index 9c6247f..f72b034 100644
--- a/Mods/Resurrection/Original/AWILLIAM.ssl
+++ b/Mods/Resurrection/ReDefine/AWILLIAM.ssl
@@ -86,12 +86,12 @@ begin
             return 1;
         end
     end
-    if ((party_member_obj(16777779) != 0) and ((global_var(910) bwand 32) == 0)) then begin
+    if ((party_member_obj(16777779) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 32) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777779)) <= 5) then begin
             return 1;
         end
     end
-    if ((party_member_obj(16777811) != 0) and ((global_var(910) bwand 64) == 0)) then begin
+    if ((party_member_obj(16777811) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 64) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777811)) <= 5) then begin
             return 1;
         end
@@ -106,7 +106,7 @@ begin
             return 1;
         end
     end
-    if ((party_member_obj(16777814) != 0) and ((global_var(910) bwand 128) == 0)) then begin
+    if ((party_member_obj(16777814) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 128) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777814)) <= 5) then begin
             return 1;
         end
@@ -164,15 +164,15 @@ end
 
 procedure critter_p_proc
 begin
-    if ((local_var(4) == 1) or (global_var(860) == 1) and obj_can_see_obj(self_obj, dude_obj)) then begin
+    if ((local_var(4) == 1) or (global_var(GVAR_AQ_ENEMY_PRISTEH) == 1) and obj_can_see_obj(self_obj, dude_obj)) then begin
         if (critter_is_fleeing(self_obj)) then begin
             if (anim_busy(self_obj) == 0) then begin
                 if ((tile_distance_objs(self_obj, dude_obj) < 8) or obj_can_see_obj(self_obj, dude_obj)) then begin
                     if (anim_busy(self_obj) == 0) then begin
-                        global_temp := rotation_to_tile(tile_num(dude_obj), tile_num(self_obj));
+                        global_temp := rotation_to_tile(dude_tile, tile_num(self_obj));
                         animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(self_obj), global_temp, random(3, 10)), 1);
                         global_temp := (global_temp + 1) % 6;
-                        while ((anim_busy(self_obj) == 0) and (global_temp != rotation_to_tile(tile_num(dude_obj), tile_num(self_obj)))) do begin
+                        while ((anim_busy(self_obj) == 0) and (global_temp != rotation_to_tile(dude_tile, tile_num(self_obj)))) do begin
                             debug_msg("flee loop: rot == " + global_temp);
                             animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(self_obj), global_temp, random(3, 10)), 1);
                             global_temp := (global_temp + 1) % 6;
@@ -182,18 +182,18 @@ begin
             end
         end
         else begin
-            attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+            attack(dude_obj);
         end
     end
-    if (((global_var(866) bwand 1048576) != 0) and obj_can_see_obj(self_obj, dude_obj) and not((map_var(0) bwand 131072) != 0)) then begin
+    if (((global_var(GVAR_AQ_ALBUQ_SPEECHES) bwand 1048576) != 0) and obj_can_see_obj(self_obj, dude_obj) and not((map_var(0) bwand 131072) != 0)) then begin
         if (critter_is_fleeing(self_obj)) then begin
             if (anim_busy(self_obj) == 0) then begin
                 if ((tile_distance_objs(self_obj, dude_obj) < 8) or obj_can_see_obj(self_obj, dude_obj)) then begin
                     if (anim_busy(self_obj) == 0) then begin
-                        global_temp := rotation_to_tile(tile_num(dude_obj), tile_num(self_obj));
+                        global_temp := rotation_to_tile(dude_tile, tile_num(self_obj));
                         animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(self_obj), global_temp, random(3, 10)), 1);
                         global_temp := (global_temp + 1) % 6;
-                        while ((anim_busy(self_obj) == 0) and (global_temp != rotation_to_tile(tile_num(dude_obj), tile_num(self_obj)))) do begin
+                        while ((anim_busy(self_obj) == 0) and (global_temp != rotation_to_tile(dude_tile, tile_num(self_obj)))) do begin
                             debug_msg("flee loop: rot == " + global_temp);
                             animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(self_obj), global_temp, random(3, 10)), 1);
                             global_temp := (global_temp + 1) % 6;
@@ -203,16 +203,16 @@ begin
             end
         end
         else begin
-            attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+            attack(dude_obj);
         end
     end
-    if (((global_var(866) bwand 1048576) != 0) and ((map_var(0) bwand 131072) != 0) and (global_var(860) == 0) and not(combat_is_initialized)) then begin
+    if (((global_var(GVAR_AQ_ALBUQ_SPEECHES) bwand 1048576) != 0) and ((map_var(0) bwand 131072) != 0) and (global_var(GVAR_AQ_ENEMY_PRISTEH) == 0) and not(combat_is_initialized)) then begin
         critter_stop_attacking(self_obj);
     end
-    if ((elevation(dude_obj) == 0) and tile_in_tile_rect(20893, 20879, 22093, 22079, tile_num(dude_obj)) and (local_var(5) != 1)) then begin
+    if ((dude_elevation == 0) and tile_in_tile_rect(20893, 20879, 22093, 22079, dude_tile) and (local_var(5) != 1)) then begin
         dialogue_system_enter;
     end
-    if (((global_var(866) bwand 1048576) != 0) and (tile_num(self_obj) != 21288) and (global_var(860) == 0) and not(combat_is_initialized)) then begin
+    if (((global_var(GVAR_AQ_ALBUQ_SPEECHES) bwand 1048576) != 0) and (tile_num(self_obj) != 21288) and (global_var(GVAR_AQ_ENEMY_PRISTEH) == 0) and not(combat_is_initialized)) then begin
         if (obj_on_screen(self_obj) == 0) then begin
             move_to(self_obj, 21288, elevation(self_obj));
         end
@@ -226,34 +226,34 @@ procedure pickup_p_proc
 begin
     if (source_obj == dude_obj) then begin
         set_local_var(4, 1);
-        set_global_var(866, global_var(866) bwor 1048576);
+        set_global_var(GVAR_AQ_ALBUQ_SPEECHES, global_var(GVAR_AQ_ALBUQ_SPEECHES) bwor 1048576);
     end
 end
 
 procedure talk_p_proc
 begin
-    if ((local_var(4) == 1) or ((global_var(866) bwand 1048576) != 0) or (global_var(860) == 1)) then begin
+    if ((local_var(4) == 1) or ((global_var(GVAR_AQ_ALBUQ_SPEECHES) bwand 1048576) != 0) or (global_var(GVAR_AQ_ENEMY_PRISTEH) == 1)) then begin
         Projed_smycku := 1;
-        anim(self_obj, 1000, rotation_to_tile(tile_num(self_obj), tile_num(dude_obj)));
+        anim(self_obj, 1000, rotation_to_tile(tile_num(self_obj), dude_tile));
     end
     if (((local_var(6) bwand 4) != 0) and (Projed_smycku == 0)) then begin
-        float_msg(self_obj, message_str(1629, 300), 2);
+        float_msg(self_obj, mstr(300), FLOAT_MSG_RED);
         Projed_smycku := 1;
     end
-    if ((global_var(870) == 2) and ((get_critter_stat(dude_obj, 4) + has_trait(0, dude_obj, 49)) <= 3) and not((local_var(6) bwand 8) != 0) and (Projed_smycku == 0)) then begin
-        float_msg(self_obj, message_str(1629, 301), 8);
+    if ((global_var(GVAR_AQ_QUEST_BRING_VEX_HEAD) == 2) and ((dude_iq + has_trait(TRAIT_PERK, dude_obj, PERK_smooth_talker)) <= 3) and not((local_var(6) bwand 8) != 0) and (Projed_smycku == 0)) then begin
+        float_msg(self_obj, mstr(301), FLOAT_MSG_YELLOW);
         Projed_smycku := 1;
     end
-    if (((local_var(6) bwand 8) != 0) and ((get_critter_stat(dude_obj, 4) + has_trait(0, dude_obj, 49)) <= 3) and (local_var(5) == 1) and (global_var(870) != 1) and (Projed_smycku == 0)) then begin
-        float_msg(self_obj, message_str(1629, 302), 8);
+    if (((local_var(6) bwand 8) != 0) and ((dude_iq + has_trait(TRAIT_PERK, dude_obj, PERK_smooth_talker)) <= 3) and (local_var(5) == 1) and (global_var(GVAR_AQ_QUEST_BRING_VEX_HEAD) != 1) and (Projed_smycku == 0)) then begin
+        float_msg(self_obj, mstr(302), FLOAT_MSG_YELLOW);
         Projed_smycku := 1;
     end
     if (Projed_smycku == 0) then begin
-        anim(dude_obj, 1000, rotation_to_tile(tile_num(dude_obj), tile_num(self_obj)));
-        anim(self_obj, 1000, rotation_to_tile(tile_num(self_obj), tile_num(dude_obj)));
+        anim(dude_obj, 1000, rotation_to_tile(dude_tile, tile_num(self_obj)));
+        anim(self_obj, 1000, rotation_to_tile(tile_num(self_obj), dude_tile));
         start_gdialog(1629, self_obj, 4, -1, -1);
         gsay_start;
-        if ((local_var(5) == 1) and (global_var(870) == 2) and (Projed_smycku == 0)) then begin
+        if ((local_var(5) == 1) and (global_var(GVAR_AQ_QUEST_BRING_VEX_HEAD) == 2) and (Projed_smycku == 0)) then begin
             call Node1500();
             Projed_smycku := 1;
         end
@@ -275,59 +275,59 @@ end
 procedure destroy_p_proc
 begin
     if (source_obj == dude_obj) then begin
-        set_global_var(5, global_var(5) + 1);
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(4) > (3 * global_var(5))) or (global_var(2) == 1)) and (global_var(1) == 0)) then begin
-            set_global_var(2, 1);
-            set_global_var(3, 0);
+        set_global_var(GVAR_GOOD_MONSTER, global_var(GVAR_GOOD_MONSTER) + 1);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(4) > (3 * global_var(GVAR_GOOD_MONSTER))) or (global_var(GVAR_CHAMPION_REPUTATION) == 1)) and (global_var(GVAR_CHILDKILLER_REPUTATION) == 0)) then begin
+            set_global_var(GVAR_CHAMPION_REPUTATION, 1);
+            set_global_var(GVAR_BERSERKER_REPUTATION, 0);
         end
-        if (metarule(51, self_obj) == 2) then begin
+        if (critter_kill_type(self_obj) == KILL_TYPE_children_kills) then begin
             if (source_obj == dude_obj) then begin
-                set_global_var(2, 0);
-                set_global_var(1, global_var(1) + 1);
-                set_global_var(0, global_var(0) + -15);
+                set_global_var(GVAR_CHAMPION_REPUTATION, 0);
+                set_global_var(GVAR_CHILDKILLER_REPUTATION, global_var(GVAR_CHILDKILLER_REPUTATION) + 1);
+                set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + -15);
                 debug_msg("Player gains " + -15 + " Karma Points.");
-                set_global_var(37, 0);
-                set_global_var(38, 0);
-                set_global_var(39, 0);
-                set_global_var(40, 0);
-                set_global_var(41, 0);
-                set_global_var(42, 0);
-                set_global_var(43, 0);
-                set_global_var(44, 0);
-                set_global_var(45, 0);
-                if (global_var(0) >= 1000) then begin
-                    set_global_var(37, 1);
+                set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+                set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+                set_global_var(GVAR_KARMA_DEFENDER, 0);
+                set_global_var(GVAR_KARMA_WANDERER, 0);
+                set_global_var(GVAR_KARMA_BETRAYER, 0);
+                set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+                set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+                set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+                    set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
                 end
                 else begin
-                    if (global_var(0) >= 750) then begin
-                        set_global_var(38, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
                     end
                     else begin
-                        if (global_var(0) >= 500) then begin
-                            set_global_var(39, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                            set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                         end
                         else begin
-                            if (global_var(0) >= 250) then begin
-                                set_global_var(40, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                                set_global_var(GVAR_KARMA_DEFENDER, 1);
                             end
                             else begin
-                                if (global_var(0) > -250) then begin
-                                    set_global_var(41, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                                    set_global_var(GVAR_KARMA_WANDERER, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -500) then begin
-                                        set_global_var(42, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                        set_global_var(GVAR_KARMA_BETRAYER, 1);
                                     end
                                     else begin
-                                        if (global_var(0) > -750) then begin
-                                            set_global_var(43, 1);
+                                        if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                            set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                         end
                                         else begin
-                                            if (global_var(0) > -1000) then begin
-                                                set_global_var(44, 1);
+                                            if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                                set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                             end
                                             else begin
-                                                set_global_var(45, 1);
+                                                set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                             end
                                         end
                                     end
@@ -336,58 +336,58 @@ begin
                         end
                     end
                 end
-                set_global_var(51, global_var(51) + -8);
+                set_global_var(GVAR_TOWN_REP_ALBUQUERQUE, global_var(GVAR_TOWN_REP_ALBUQUERQUE) + -8);
                 debug_msg("Added " + -8 + " to Town Rep");
             end
         end
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(5) > (2 * global_var(4))) or (global_var(3) == 1))) then begin
-            set_global_var(3, 1);
-            set_global_var(2, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
+            set_global_var(GVAR_BERSERKER_REPUTATION, 1);
+            set_global_var(GVAR_CHAMPION_REPUTATION, 0);
         end
-        set_global_var(0, global_var(0) + -10);
+        set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + -10);
         debug_msg("Player gains " + -10 + " Karma Points.");
-        set_global_var(37, 0);
-        set_global_var(38, 0);
-        set_global_var(39, 0);
-        set_global_var(40, 0);
-        set_global_var(41, 0);
-        set_global_var(42, 0);
-        set_global_var(43, 0);
-        set_global_var(44, 0);
-        set_global_var(45, 0);
-        if (global_var(0) >= 1000) then begin
-            set_global_var(37, 1);
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+        set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+        set_global_var(GVAR_KARMA_DEFENDER, 0);
+        set_global_var(GVAR_KARMA_WANDERER, 0);
+        set_global_var(GVAR_KARMA_BETRAYER, 0);
+        set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+        set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+            set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
         end
         else begin
-            if (global_var(0) >= 750) then begin
-                set_global_var(38, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
             end
             else begin
-                if (global_var(0) >= 500) then begin
-                    set_global_var(39, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                 end
                 else begin
-                    if (global_var(0) >= 250) then begin
-                        set_global_var(40, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                        set_global_var(GVAR_KARMA_DEFENDER, 1);
                     end
                     else begin
-                        if (global_var(0) > -250) then begin
-                            set_global_var(41, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                            set_global_var(GVAR_KARMA_WANDERER, 1);
                         end
                         else begin
-                            if (global_var(0) > -500) then begin
-                                set_global_var(42, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                set_global_var(GVAR_KARMA_BETRAYER, 1);
                             end
                             else begin
-                                if (global_var(0) > -750) then begin
-                                    set_global_var(43, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -1000) then begin
-                                        set_global_var(44, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                     end
                                     else begin
-                                        set_global_var(45, 1);
+                                        set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                     end
                                 end
                             end
@@ -396,21 +396,21 @@ begin
                 end
             end
         end
-        set_global_var(51, global_var(51) + -5);
+        set_global_var(GVAR_TOWN_REP_ALBUQUERQUE, global_var(GVAR_TOWN_REP_ALBUQUERQUE) + -5);
         debug_msg("Added " + -5 + " to Town Rep");
     end
-    set_global_var(867, global_var(867) bwor 2);
-    set_global_var(860, 1);
+    set_global_var(GVAR_AQ_ALBUQ_DEATHS, global_var(GVAR_AQ_ALBUQ_DEATHS) bwor 2);
+    set_global_var(GVAR_AQ_ENEMY_PRISTEH, 1);
 end
 
 procedure look_at_p_proc
 begin
     script_overrides;
     if (local_var(5) == 1) then begin
-        display_msg(message_str(1629, 150));
+        display_mstr(150);
     end
     else begin
-        display_msg(message_str(1629, 100));
+        display_mstr(100);
     end
 end
 
@@ -418,10 +418,10 @@ procedure description_p_proc
 begin
     script_overrides;
     if (local_var(5) == 1) then begin
-        display_msg(message_str(1629, 250));
+        display_mstr(250);
     end
     else begin
-        display_msg(message_str(1629, 200));
+        display_mstr(200);
     end
 end
 
@@ -432,15 +432,15 @@ end
 procedure damage_p_proc
 begin
     if (source_obj == dude_obj) then begin
-        set_global_var(866, global_var(866) bwor 1048576);
+        set_global_var(GVAR_AQ_ALBUQ_SPEECHES, global_var(GVAR_AQ_ALBUQ_SPEECHES) bwor 1048576);
     end
 end
 
 procedure map_enter_p_proc
 begin
-    critter_add_trait(self_obj, 1, 6, 267);
-    critter_add_trait(self_obj, 1, 5, 14);
-    if ((global_var(867) bwand 2) != 0) then begin
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 267);
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_PEASANTS);
+    if ((global_var(GVAR_AQ_ALBUQ_DEATHS) bwand 2) != 0) then begin
         destroy_object(self_obj);
     end
 end
@@ -448,7 +448,7 @@ end
 procedure timed_event_p_proc
 begin
     if (fixed_param == 1) then begin
-        float_msg(self_obj, message_str(1629, 350), 2);
+        float_msg(self_obj, mstr(350), FLOAT_MSG_RED);
     end
     if (fixed_param == 3) then begin
         if (fixed_param == 3) then begin
@@ -481,12 +481,12 @@ end
 procedure combat_p_proc
 begin
     if (fixed_param == 4) then begin
-        if (((global_var(866) bwand 1048576) != 0) and (elevation(dude_obj) == 0) and ((local_var(6) bwand 32) == 0)) then begin
-            float_msg(self_obj, message_str(1629, 400), 2);
+        if (((global_var(GVAR_AQ_ALBUQ_SPEECHES) bwand 1048576) != 0) and (dude_elevation == 0) and ((local_var(6) bwand 32) == 0)) then begin
+            float_msg(self_obj, mstr(400), FLOAT_MSG_RED);
             set_local_var(6, local_var(6) bwor 32);
         end
     end
-    if ((global_var(871) == 1) or (global_var(871) == 2) and ((global_var(866) bwand 1048576) != 0) and tile_in_tile_rect(11144, 11111, 15944, 15911, tile_num(dude_obj)) and (global_var(860) == 0) and (global_var(861) == 0) and (global_var(864) == 0) and not((map_var(0) bwand 131072) != 0)) then begin
+    if ((global_var(GVAR_AQ_QUEST_KILL_WILLIAM_WIFE) == 1) or (global_var(GVAR_AQ_QUEST_KILL_WILLIAM_WIFE) == 2) and ((global_var(GVAR_AQ_ALBUQ_SPEECHES) bwand 1048576) != 0) and tile_in_tile_rect(11144, 11111, 15944, 15911, dude_tile) and (global_var(GVAR_AQ_ENEMY_PRISTEH) == 0) and (global_var(GVAR_AQ_ENEMY_VEX) == 0) and (global_var(GVAR_AQ_ENEMY_HUNTERS) == 0) and not((map_var(0) bwand 131072) != 0)) then begin
         if (tile_in_tile_rect(11144, 11111, 15944, 15911, tile_num(self_obj))) then begin
             if (tile_in_tile_rect(13722, 13717, 16722, 16717, tile_num(self_obj))) then begin
                 set_map_var(0, map_var(0) bwor 1048576);
@@ -507,8 +507,8 @@ end
 
 procedure Node998
 begin
-    float_msg(self_obj, message_str(1629, 400), 2);
-    set_global_var(866, global_var(866) bwor 1048576);
+    float_msg(self_obj, mstr(400), FLOAT_MSG_RED);
+    set_global_var(GVAR_AQ_ALBUQ_SPEECHES, global_var(GVAR_AQ_ALBUQ_SPEECHES) bwor 1048576);
     set_local_var(6, local_var(6) bwor 32);
 end
 
@@ -519,354 +519,354 @@ end
 
 procedure Node500
 begin
-    gsay_reply(1629, 500);
-    if ((get_critter_stat(dude_obj, 4) + has_trait(0, dude_obj, 49)) > 3) then begin
+    Reply(500);
+    if ((dude_iq + has_trait(TRAIT_PERK, dude_obj, PERK_smooth_talker)) > 3) then begin
         set_local_var(6, local_var(6) bwor 8);
     end
-    giq_option(4, 1629, message_str(1629, 501) + obj_name(dude_obj) + message_str(1629, 502), Node550, 50);
-    giq_option(4, 1629, 503, Node600, 50);
-    giq_option(4, 1629, 504, Node1200, 50);
-    giq_option(-3, 1629, 505, Node1000, 50);
+    giq_option(4, NAME, mstr(501) + dude_name + mstr(502), Node550, NEUTRAL_REACTION);
+    giq_option(4, NAME, 503, Node600, NEUTRAL_REACTION);
+    giq_option(4, NAME, 504, Node1200, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 505, Node1000, NEUTRAL_REACTION);
 end
 
 procedure Node550
 begin
-    if (get_critter_stat(dude_obj, 34) == 0) then begin
-        gsay_reply(1629, 550);
+    if (dude_gender == 0) then begin
+        Reply(550);
     end
-    if (get_critter_stat(dude_obj, 34) == 1) then begin
-        gsay_reply(1629, 551);
+    if (dude_gender == 1) then begin
+        Reply(551);
     end
-    if (get_critter_stat(dude_obj, 34) == 0) then begin
-        giq_option(4, 1629, 552, Node650, 50);
+    if (dude_gender == 0) then begin
+        giq_option(4, NAME, 552, Node650, NEUTRAL_REACTION);
     end
-    if (get_critter_stat(dude_obj, 34) == 1) then begin
-        giq_option(4, 1629, 553, Node650, 50);
+    if (dude_gender == 1) then begin
+        giq_option(4, NAME, 553, Node650, NEUTRAL_REACTION);
     end
-    if ((global_var(870) == 0) and ((map_var(0) bwand 1) != 0) and not((global_var(867) bwand 16) != 0)) then begin
-        giq_option(4, 1629, 554, Node850, 50);
+    if ((global_var(GVAR_AQ_QUEST_BRING_VEX_HEAD) == 0) and ((map_var(0) bwand 1) != 0) and not((global_var(GVAR_AQ_ALBUQ_DEATHS) bwand 16) != 0)) then begin
+        giq_option(4, NAME, 554, Node850, NEUTRAL_REACTION);
     end
-    if ((global_var(870) == 0) and ((map_var(0) bwand 1) != 0) and ((global_var(867) bwand 16) != 0)) then begin
-        giq_option(4, 1629, 554, Node1600, 50);
+    if ((global_var(GVAR_AQ_QUEST_BRING_VEX_HEAD) == 0) and ((map_var(0) bwand 1) != 0) and ((global_var(GVAR_AQ_ALBUQ_DEATHS) bwand 16) != 0)) then begin
+        giq_option(4, NAME, 554, Node1600, NEUTRAL_REACTION);
     end
-    if ((global_var(870) == 0) and ((map_var(0) bwand 1) == 0) and not((global_var(867) bwand 16) != 0)) then begin
-        giq_option(6, 1629, 556, Node1750, 50);
+    if ((global_var(GVAR_AQ_QUEST_BRING_VEX_HEAD) == 0) and ((map_var(0) bwand 1) == 0) and not((global_var(GVAR_AQ_ALBUQ_DEATHS) bwand 16) != 0)) then begin
+        giq_option(6, NAME, 556, Node1750, NEUTRAL_REACTION);
     end
-    if ((global_var(870) == 0) and ((map_var(0) bwand 1) == 0) and ((global_var(867) bwand 16) != 0)) then begin
-        giq_option(6, 1629, 556, Node1800, 50);
+    if ((global_var(GVAR_AQ_QUEST_BRING_VEX_HEAD) == 0) and ((map_var(0) bwand 1) == 0) and ((global_var(GVAR_AQ_ALBUQ_DEATHS) bwand 16) != 0)) then begin
+        giq_option(6, NAME, 556, Node1800, NEUTRAL_REACTION);
     end
-    if ((global_var(875) == 1) and not((map_var(0) bwand 2) != 0)) then begin
-        giq_option(4, 1629, 1258, Node1300, 50);
+    if ((global_var(GVAR_AQ_QUEST_TRIBE_SOLVE_LOSTS) == 1) and not((map_var(0) bwand 2) != 0)) then begin
+        giq_option(4, NAME, 1258, Node1300, NEUTRAL_REACTION);
     end
-    if ((global_var(876) == 1) and not((map_var(0) bwand 4) != 0)) then begin
-        giq_option(4, 1629, 1259, Node1550, 50);
+    if ((global_var(GVAR_AQ_QUEST_TRIBE_HELP_REVEGE) == 1) and not((map_var(0) bwand 4) != 0)) then begin
+        giq_option(4, NAME, 1259, Node1550, NEUTRAL_REACTION);
     end
     if ((local_var(6) bwand 1) == 0) then begin
-        giq_option(4, 1629, 555, Node1050, 50);
+        giq_option(4, NAME, 555, Node1050, NEUTRAL_REACTION);
     end
     else begin
-        giq_option(4, 1629, 555, Node999, 50);
+        giq_option(4, NAME, 555, Node999, NEUTRAL_REACTION);
     end
 end
 
 procedure Node600
 begin
-    if (get_critter_stat(dude_obj, 34) == 0) then begin
-        gsay_reply(1629, 600);
+    if (dude_gender == 0) then begin
+        Reply(600);
     end
-    if (get_critter_stat(dude_obj, 34) == 1) then begin
-        gsay_reply(1629, 601);
+    if (dude_gender == 1) then begin
+        Reply(601);
     end
-    if (get_critter_stat(dude_obj, 34) == 0) then begin
-        giq_option(4, 1629, 602, Node650, 50);
+    if (dude_gender == 0) then begin
+        giq_option(4, NAME, 602, Node650, NEUTRAL_REACTION);
     end
-    if (get_critter_stat(dude_obj, 34) == 1) then begin
-        giq_option(4, 1629, 553, Node650, 50);
+    if (dude_gender == 1) then begin
+        giq_option(4, NAME, 553, Node650, NEUTRAL_REACTION);
     end
-    if ((global_var(870) == 0) and ((map_var(0) bwand 1) != 0) and not((global_var(867) bwand 16) != 0)) then begin
-        giq_option(4, 1629, 603, Node850, 50);
+    if ((global_var(GVAR_AQ_QUEST_BRING_VEX_HEAD) == 0) and ((map_var(0) bwand 1) != 0) and not((global_var(GVAR_AQ_ALBUQ_DEATHS) bwand 16) != 0)) then begin
+        giq_option(4, NAME, 603, Node850, NEUTRAL_REACTION);
     end
-    if ((global_var(870) == 0) and ((map_var(0) bwand 1) != 0) and ((global_var(867) bwand 16) != 0)) then begin
-        giq_option(4, 1629, 603, Node1600, 50);
+    if ((global_var(GVAR_AQ_QUEST_BRING_VEX_HEAD) == 0) and ((map_var(0) bwand 1) != 0) and ((global_var(GVAR_AQ_ALBUQ_DEATHS) bwand 16) != 0)) then begin
+        giq_option(4, NAME, 603, Node1600, NEUTRAL_REACTION);
     end
-    if ((global_var(870) == 0) and ((map_var(0) bwand 1) == 0) and not((global_var(867) bwand 16) != 0)) then begin
-        giq_option(6, 1629, 605, Node1750, 50);
+    if ((global_var(GVAR_AQ_QUEST_BRING_VEX_HEAD) == 0) and ((map_var(0) bwand 1) == 0) and not((global_var(GVAR_AQ_ALBUQ_DEATHS) bwand 16) != 0)) then begin
+        giq_option(6, NAME, 605, Node1750, NEUTRAL_REACTION);
     end
-    if ((global_var(870) == 0) and ((map_var(0) bwand 1) == 0) and ((global_var(867) bwand 16) != 0)) then begin
-        giq_option(6, 1629, 605, Node1800, 50);
+    if ((global_var(GVAR_AQ_QUEST_BRING_VEX_HEAD) == 0) and ((map_var(0) bwand 1) == 0) and ((global_var(GVAR_AQ_ALBUQ_DEATHS) bwand 16) != 0)) then begin
+        giq_option(6, NAME, 605, Node1800, NEUTRAL_REACTION);
     end
-    if ((global_var(875) == 1) and not((map_var(0) bwand 2) != 0)) then begin
-        giq_option(4, 1629, 1258, Node1300, 50);
+    if ((global_var(GVAR_AQ_QUEST_TRIBE_SOLVE_LOSTS) == 1) and not((map_var(0) bwand 2) != 0)) then begin
+        giq_option(4, NAME, 1258, Node1300, NEUTRAL_REACTION);
     end
-    if ((global_var(876) == 1) and not((map_var(0) bwand 4) != 0)) then begin
-        giq_option(4, 1629, 1259, Node1550, 50);
+    if ((global_var(GVAR_AQ_QUEST_TRIBE_HELP_REVEGE) == 1) and not((map_var(0) bwand 4) != 0)) then begin
+        giq_option(4, NAME, 1259, Node1550, NEUTRAL_REACTION);
     end
     if ((local_var(6) bwand 1) == 0) then begin
-        giq_option(4, 1629, 604, Node1050, 50);
+        giq_option(4, NAME, 604, Node1050, NEUTRAL_REACTION);
     end
     else begin
-        giq_option(4, 1629, 604, Node999, 50);
+        giq_option(4, NAME, 604, Node999, NEUTRAL_REACTION);
     end
 end
 
 procedure Node650
 begin
-    gsay_reply(1629, 650);
-    giq_option(4, 1629, 651, Node700, 50);
-    giq_option(4, 1629, 652, Node750, 50);
-    if (global_var(878) >= 3) then begin
-        giq_option(4, 1629, 653, Node1350, 50);
+    Reply(650);
+    giq_option(4, NAME, 651, Node700, NEUTRAL_REACTION);
+    giq_option(4, NAME, 652, Node750, NEUTRAL_REACTION);
+    if (global_var(GVAR_AQ_QUEST_HOSPITAL_TO_KILL_PRIMITIVES) >= 3) then begin
+        giq_option(4, NAME, 653, Node1350, NEUTRAL_REACTION);
     end
     else begin
-        giq_option(4, 1629, 653, Node800, 50);
+        giq_option(4, NAME, 653, Node800, NEUTRAL_REACTION);
     end
     if ((local_var(6) bwand 1) == 0) then begin
-        giq_option(4, 1629, 654, Node1050, 50);
+        giq_option(4, NAME, 654, Node1050, NEUTRAL_REACTION);
     end
     else begin
-        giq_option(4, 1629, 654, Node999, 50);
+        giq_option(4, NAME, 654, Node999, NEUTRAL_REACTION);
     end
 end
 
 procedure Node700
 begin
-    gsay_reply(1629, 700);
-    giq_option(4, 1629, 701, Node1150, 50);
+    Reply(700);
+    giq_option(4, NAME, 701, Node1150, NEUTRAL_REACTION);
     if ((local_var(6) bwand 1) == 0) then begin
-        giq_option(4, 1629, 702, Node1050, 50);
+        giq_option(4, NAME, 702, Node1050, NEUTRAL_REACTION);
     end
     else begin
-        giq_option(4, 1629, 702, Node999, 50);
+        giq_option(4, NAME, 702, Node999, NEUTRAL_REACTION);
     end
 end
 
 procedure Node750
 begin
-    gsay_reply(1629, 750);
-    giq_option(4, 1629, 751, Node1150, 50);
+    Reply(750);
+    giq_option(4, NAME, 751, Node1150, NEUTRAL_REACTION);
     if ((local_var(6) bwand 1) == 0) then begin
-        giq_option(4, 1629, 752, Node1050, 50);
+        giq_option(4, NAME, 752, Node1050, NEUTRAL_REACTION);
     end
     else begin
-        giq_option(4, 1629, 752, Node999, 50);
+        giq_option(4, NAME, 752, Node999, NEUTRAL_REACTION);
     end
 end
 
 procedure Node800
 begin
-    if (global_var(859) == 0) then begin
-        gsay_reply(1629, 800);
+    if (global_var(GVAR_AQ_MUTANT_HUNTER) == 0) then begin
+        Reply(800);
     end
     else begin
-        if (get_critter_stat(dude_obj, 34) == 0) then begin
-            gsay_reply(1629, 801);
+        if (dude_gender == 0) then begin
+            Reply(801);
         end
         else begin
-            gsay_reply(1629, 802);
+            Reply(802);
         end
     end
-    giq_option(4, 1629, 803, Node1150, 50);
+    giq_option(4, NAME, 803, Node1150, NEUTRAL_REACTION);
     if ((local_var(6) bwand 1) == 0) then begin
-        giq_option(4, 1629, 804, Node1050, 50);
+        giq_option(4, NAME, 804, Node1050, NEUTRAL_REACTION);
     end
     else begin
-        giq_option(4, 1629, 804, Node999, 50);
+        giq_option(4, NAME, 804, Node999, NEUTRAL_REACTION);
     end
 end
 
 procedure Node850
 begin
-    gsay_reply(1629, 850);
-    if (not((global_var(866) bwand 32768) != 0)) then begin
-        set_global_var(866, global_var(866) bwor 32768);
+    Reply(850);
+    if (not((global_var(GVAR_AQ_ALBUQ_SPEECHES) bwand 32768) != 0)) then begin
+        set_global_var(GVAR_AQ_ALBUQ_SPEECHES, global_var(GVAR_AQ_ALBUQ_SPEECHES) bwor 32768);
     end
-    giq_option(4, 1629, 851, Node900, 50);
+    giq_option(4, NAME, 851, Node900, NEUTRAL_REACTION);
     if ((local_var(6) bwand 1) == 0) then begin
-        giq_option(4, 1629, 852, Node1050, 50);
+        giq_option(4, NAME, 852, Node1050, NEUTRAL_REACTION);
     end
     else begin
-        giq_option(4, 1629, 852, Node999, 50);
+        giq_option(4, NAME, 852, Node999, NEUTRAL_REACTION);
     end
 end
 
 procedure Node900
 begin
-    gsay_reply(1629, 900);
-    giq_option(4, 1629, 901, Node1450, 50);
+    Reply(900);
+    giq_option(4, NAME, 901, Node1450, NEUTRAL_REACTION);
 end
 
 procedure Node950
 begin
-    if (get_critter_stat(dude_obj, 34) == 0) then begin
-        gsay_reply(1629, 950);
+    if (dude_gender == 0) then begin
+        Reply(950);
     end
-    if (get_critter_stat(dude_obj, 34) == 1) then begin
-        gsay_reply(1629, 951);
+    if (dude_gender == 1) then begin
+        Reply(951);
     end
     if ((local_var(6) bwand 1) == 0) then begin
-        giq_option(4, 1629, 952, Node1050, 50);
+        giq_option(4, NAME, 952, Node1050, NEUTRAL_REACTION);
     end
     else begin
-        giq_option(4, 1629, 952, Node999, 50);
+        giq_option(4, NAME, 952, Node999, NEUTRAL_REACTION);
     end
 end
 
 procedure Node1000
 begin
-    gsay_reply(1629, 1000);
-    giq_option(-3, 1629, message_str(1629, 1001) + obj_name(dude_obj) + message_str(1629, 1002), Node1650, 50);
-    giq_option(-3, 1629, 1003, Node998, 50);
+    Reply(1000);
+    giq_option(-3, NAME, mstr(1001) + dude_name + mstr(1002), Node1650, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 1003, Node998, NEUTRAL_REACTION);
 end
 
 procedure Node1050
 begin
-    if (get_critter_stat(dude_obj, 34) == 0) then begin
-        gsay_reply(1629, 1050);
+    if (dude_gender == 0) then begin
+        Reply(1050);
     end
-    if (get_critter_stat(dude_obj, 34) == 1) then begin
-        gsay_reply(1629, 1051);
+    if (dude_gender == 1) then begin
+        Reply(1051);
     end
     set_local_var(6, local_var(6) bwor 1);
-    giq_option(4, 1629, 1052, Node999, 50);
-    giq_option(4, 1629, 1053, Node1100, 50);
+    giq_option(4, NAME, 1052, Node999, NEUTRAL_REACTION);
+    giq_option(4, NAME, 1053, Node1100, NEUTRAL_REACTION);
 end
 
 procedure Node1100
 begin
-    gsay_reply(1629, 1100);
-    giq_option(4, 1629, 1101, Node999, 50);
-    if (get_critter_stat(dude_obj, 34) == 0) then begin
-        giq_option(4, 1629, 1102, Node998, 50);
+    Reply(1100);
+    giq_option(4, NAME, 1101, Node999, NEUTRAL_REACTION);
+    if (dude_gender == 0) then begin
+        giq_option(4, NAME, 1102, Node998, NEUTRAL_REACTION);
     end
-    if (get_critter_stat(dude_obj, 34) == 1) then begin
-        giq_option(4, 1629, 1103, Node998, 50);
+    if (dude_gender == 1) then begin
+        giq_option(4, NAME, 1103, Node998, NEUTRAL_REACTION);
     end
 end
 
 procedure Node1150
 begin
-    gsay_reply(1629, 1150);
-    giq_option(4, 1629, 1151, Node700, 50);
-    giq_option(4, 1629, 1152, Node750, 50);
-    if (global_var(878) >= 3) then begin
-        giq_option(4, 1629, 1153, Node1350, 50);
+    Reply(1150);
+    giq_option(4, NAME, 1151, Node700, NEUTRAL_REACTION);
+    giq_option(4, NAME, 1152, Node750, NEUTRAL_REACTION);
+    if (global_var(GVAR_AQ_QUEST_HOSPITAL_TO_KILL_PRIMITIVES) >= 3) then begin
+        giq_option(4, NAME, 1153, Node1350, NEUTRAL_REACTION);
     end
     else begin
-        giq_option(4, 1629, 1153, Node800, 50);
+        giq_option(4, NAME, 1153, Node800, NEUTRAL_REACTION);
     end
     if ((local_var(6) bwand 1) == 0) then begin
-        giq_option(4, 1629, 1154, Node1050, 50);
+        giq_option(4, NAME, 1154, Node1050, NEUTRAL_REACTION);
     end
     else begin
-        giq_option(4, 1629, 1154, Node999, 50);
+        giq_option(4, NAME, 1154, Node999, NEUTRAL_REACTION);
     end
 end
 
 procedure Node1200
 begin
-    gsay_reply(1629, 1200);
-    giq_option(4, 1629, message_str(1629, 1201) + obj_name(dude_obj) + message_str(1629, 1202), Node550, 50);
-    giq_option(4, 1629, 1203, Node998, 50);
+    Reply(1200);
+    giq_option(4, NAME, mstr(1201) + dude_name + mstr(1202), Node550, NEUTRAL_REACTION);
+    giq_option(4, NAME, 1203, Node998, NEUTRAL_REACTION);
 end
 
 procedure Node1250
 begin
-    gsay_reply(1629, 1250);
-    if (get_critter_stat(dude_obj, 34) == 0) then begin
-        giq_option(4, 1629, 1251, Node650, 50);
+    Reply(1250);
+    if (dude_gender == 0) then begin
+        giq_option(4, NAME, 1251, Node650, NEUTRAL_REACTION);
     end
-    if (get_critter_stat(dude_obj, 34) == 1) then begin
-        giq_option(4, 1629, 1252, Node650, 50);
+    if (dude_gender == 1) then begin
+        giq_option(4, NAME, 1252, Node650, NEUTRAL_REACTION);
     end
-    if ((global_var(870) == 0) and ((map_var(0) bwand 1) != 0) and not((global_var(867) bwand 16) != 0) and not((local_var(6) bwand 16) != 0)) then begin
-        giq_option(4, 1629, 1253, Node850, 50);
+    if ((global_var(GVAR_AQ_QUEST_BRING_VEX_HEAD) == 0) and ((map_var(0) bwand 1) != 0) and not((global_var(GVAR_AQ_ALBUQ_DEATHS) bwand 16) != 0) and not((local_var(6) bwand 16) != 0)) then begin
+        giq_option(4, NAME, 1253, Node850, NEUTRAL_REACTION);
     end
-    if ((global_var(870) == 0) and ((map_var(0) bwand 1) != 0) and ((global_var(867) bwand 16) != 0) and not((local_var(6) bwand 16) != 0)) then begin
-        giq_option(4, 1629, 1253, Node1600, 50);
+    if ((global_var(GVAR_AQ_QUEST_BRING_VEX_HEAD) == 0) and ((map_var(0) bwand 1) != 0) and ((global_var(GVAR_AQ_ALBUQ_DEATHS) bwand 16) != 0) and not((local_var(6) bwand 16) != 0)) then begin
+        giq_option(4, NAME, 1253, Node1600, NEUTRAL_REACTION);
     end
-    if ((global_var(870) == 0) and ((map_var(0) bwand 1) == 0) and not((global_var(867) bwand 16) != 0) and not((local_var(6) bwand 16) != 0)) then begin
-        giq_option(6, 1629, 1262, Node1750, 50);
+    if ((global_var(GVAR_AQ_QUEST_BRING_VEX_HEAD) == 0) and ((map_var(0) bwand 1) == 0) and not((global_var(GVAR_AQ_ALBUQ_DEATHS) bwand 16) != 0) and not((local_var(6) bwand 16) != 0)) then begin
+        giq_option(6, NAME, 1262, Node1750, NEUTRAL_REACTION);
     end
-    if ((global_var(870) == 0) and ((map_var(0) bwand 1) == 0) and ((global_var(867) bwand 16) != 0) and not((local_var(6) bwand 16) != 0)) then begin
-        giq_option(6, 1629, 1262, Node1800, 50);
+    if ((global_var(GVAR_AQ_QUEST_BRING_VEX_HEAD) == 0) and ((map_var(0) bwand 1) == 0) and ((global_var(GVAR_AQ_ALBUQ_DEATHS) bwand 16) != 0) and not((local_var(6) bwand 16) != 0)) then begin
+        giq_option(6, NAME, 1262, Node1800, NEUTRAL_REACTION);
     end
-    if ((global_var(870) == 1) and ((global_var(867) bwand 16) != 0) and obj_is_carrying_obj_pid(dude_obj, 541)) then begin
-        giq_option(4, 1629, 1254, Node1400, 50);
+    if ((global_var(GVAR_AQ_QUEST_BRING_VEX_HEAD) == 1) and ((global_var(GVAR_AQ_ALBUQ_DEATHS) bwand 16) != 0) and obj_is_carrying_obj_pid(dude_obj, 541)) then begin
+        giq_option(4, NAME, 1254, Node1400, NEUTRAL_REACTION);
     end
-    if ((global_var(870) == 1) and ((global_var(867) bwand 16) != 0) and obj_is_carrying_obj_pid(dude_obj, 541)) then begin
-        giq_option(-3, 1629, 1255, Node1400, 50);
+    if ((global_var(GVAR_AQ_QUEST_BRING_VEX_HEAD) == 1) and ((global_var(GVAR_AQ_ALBUQ_DEATHS) bwand 16) != 0) and obj_is_carrying_obj_pid(dude_obj, 541)) then begin
+        giq_option(-3, NAME, 1255, Node1400, NEUTRAL_REACTION);
     end
-    if ((global_var(870) == 0) and (get_critter_stat(dude_obj, 34) == 0) and ((local_var(6) bwand 2) != 0)) then begin
-        giq_option(4, 1629, 1256, Quest_accepted, 50);
+    if ((global_var(GVAR_AQ_QUEST_BRING_VEX_HEAD) == 0) and (dude_gender == 0) and ((local_var(6) bwand 2) != 0)) then begin
+        giq_option(4, NAME, 1256, Quest_accepted, NEUTRAL_REACTION);
     end
-    if ((global_var(870) == 0) and (get_critter_stat(dude_obj, 34) == 1) and ((local_var(6) bwand 2) != 0)) then begin
-        giq_option(4, 1629, 1257, Quest_accepted, 50);
+    if ((global_var(GVAR_AQ_QUEST_BRING_VEX_HEAD) == 0) and (dude_gender == 1) and ((local_var(6) bwand 2) != 0)) then begin
+        giq_option(4, NAME, 1257, Quest_accepted, NEUTRAL_REACTION);
     end
-    if ((global_var(875) == 1) and not((map_var(0) bwand 2) != 0)) then begin
-        giq_option(4, 1629, 1258, Node1300, 50);
+    if ((global_var(GVAR_AQ_QUEST_TRIBE_SOLVE_LOSTS) == 1) and not((map_var(0) bwand 2) != 0)) then begin
+        giq_option(4, NAME, 1258, Node1300, NEUTRAL_REACTION);
     end
-    if ((global_var(876) == 1) and not((map_var(0) bwand 4) != 0)) then begin
-        giq_option(4, 1629, 1259, Node1550, 50);
+    if ((global_var(GVAR_AQ_QUEST_TRIBE_HELP_REVEGE) == 1) and not((map_var(0) bwand 4) != 0)) then begin
+        giq_option(4, NAME, 1259, Node1550, NEUTRAL_REACTION);
     end
-    giq_option(4, 1629, 1260, Node999, 50);
-    giq_option(-3, 1629, 1261, Node999, 50);
+    giq_option(4, NAME, 1260, Node999, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 1261, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node1300
 begin
-    if (get_critter_stat(dude_obj, 34) == 0) then begin
-        gsay_reply(1629, 1300);
+    if (dude_gender == 0) then begin
+        Reply(1300);
     end
-    if (get_critter_stat(dude_obj, 34) == 1) then begin
-        gsay_reply(1629, 1301);
+    if (dude_gender == 1) then begin
+        Reply(1301);
     end
     set_map_var(0, map_var(0) bwor 2);
     if ((local_var(6) bwand 1) == 0) then begin
-        giq_option(4, 1629, 1302, Node1050, 50);
+        giq_option(4, NAME, 1302, Node1050, NEUTRAL_REACTION);
     end
     else begin
-        giq_option(4, 1629, 1302, Node999, 50);
+        giq_option(4, NAME, 1302, Node999, NEUTRAL_REACTION);
     end
 end
 
 procedure Node1350
 begin
-    if (global_var(859) == 0) then begin
-        gsay_reply(1629, 1350);
+    if (global_var(GVAR_AQ_MUTANT_HUNTER) == 0) then begin
+        Reply(1350);
     end
     else begin
-        if (get_critter_stat(dude_obj, 34) == 0) then begin
-            gsay_reply(1629, 1351);
+        if (dude_gender == 0) then begin
+            Reply(1351);
         end
         else begin
-            gsay_reply(1629, 1352);
+            Reply(1352);
         end
     end
-    giq_option(4, 1629, 1353, Node1150, 50);
+    giq_option(4, NAME, 1353, Node1150, NEUTRAL_REACTION);
     if ((local_var(6) bwand 1) == 0) then begin
-        giq_option(4, 1629, 1354, Node1050, 50);
+        giq_option(4, NAME, 1354, Node1050, NEUTRAL_REACTION);
     end
     else begin
-        giq_option(4, 1629, 1354, Node999, 50);
+        giq_option(4, NAME, 1354, Node999, NEUTRAL_REACTION);
     end
 end
 
 procedure Node1400
 begin
-    if ((get_critter_stat(dude_obj, 4) + has_trait(0, dude_obj, 49)) > 3) then begin
-        if (get_critter_stat(dude_obj, 34) == 0) then begin
-            gsay_reply(1629, 1400);
+    if ((dude_iq + has_trait(TRAIT_PERK, dude_obj, PERK_smooth_talker)) > 3) then begin
+        if (dude_gender == 0) then begin
+            Reply(1400);
         end
-        if (get_critter_stat(dude_obj, 34) == 1) then begin
-            gsay_reply(1629, 1401);
+        if (dude_gender == 1) then begin
+            Reply(1401);
         end
     end
     else begin
-        if (get_critter_stat(dude_obj, 34) == 0) then begin
-            gsay_reply(1629, 1402);
+        if (dude_gender == 0) then begin
+            Reply(1402);
         end
-        if (get_critter_stat(dude_obj, 34) == 1) then begin
-            gsay_reply(1629, 1403);
+        if (dude_gender == 1) then begin
+            Reply(1403);
         end
     end
     removed_qty := obj_is_carrying_obj_pid(dude_obj, 541);
@@ -877,17 +877,17 @@ begin
         restock_obj := obj_carrying_pid_obj(dude_obj, 541);
         if (obj_type(self_obj) == 1) then begin
             restock_obj := obj_carrying_pid_obj(dude_obj, 541);
-            if (critter_inven_obj(self_obj, 0) == restock_obj) then begin
-                if (obj_item_subtype(critter_inven_obj(dude_obj, 0)) == 0) then begin
-                    restock_obj := critter_inven_obj(dude_obj, 0);
+            if (critter_inven_obj(self_obj, INVEN_TYPE_WORN) == restock_obj) then begin
+                if (obj_item_subtype(critter_inven_obj(dude_obj, INVEN_TYPE_WORN)) == 0) then begin
+                    restock_obj := critter_inven_obj(dude_obj, INVEN_TYPE_WORN);
                     debug_msg("armour pid == " + obj_pid(restock_obj));
                     rm_obj_from_inven(dude_obj, restock_obj);
                     add_obj_to_inven(dude_obj, restock_obj);
                 end
             end
             else begin
-                if ((critter_inven_obj(dude_obj, 2) == restock_obj) or (critter_inven_obj(dude_obj, 1) == restock_obj)) then begin
-                    metarule(43, dude_obj);
+                if ((critter_inven_obj(dude_obj, INVEN_TYPE_LEFT_HAND) == restock_obj) or (critter_inven_obj(dude_obj, INVEN_TYPE_RIGHT_HAND) == restock_obj)) then begin
+                    inven_unwield(dude_obj);
                 end
             end
         end
@@ -907,54 +907,54 @@ begin
     end
     restock_obj := 0;
     restock_amt := 0;
-    set_global_var(870, 2);
-    set_global_var(1015, global_var(1015) bwand (-1 - 65536));
+    set_global_var(GVAR_AQ_QUEST_BRING_VEX_HEAD, 2);
+    set_global_var(GVAR_PTR_KERI_REACTIONS, global_var(GVAR_PTR_KERI_REACTIONS) bwand (-1 - 65536));
     give_exp_points(3000);
-    display_msg(message_str(14, 100) + 3000 + message_str(14, 101));
-    set_global_var(0, global_var(0) + 30);
+    display_msg(g_mstr(100) + 3000 + g_mstr(101));
+    set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + 30);
     debug_msg("Player gains " + 30 + " Karma Points.");
-    set_global_var(37, 0);
-    set_global_var(38, 0);
-    set_global_var(39, 0);
-    set_global_var(40, 0);
-    set_global_var(41, 0);
-    set_global_var(42, 0);
-    set_global_var(43, 0);
-    set_global_var(44, 0);
-    set_global_var(45, 0);
-    if (global_var(0) >= 1000) then begin
-        set_global_var(37, 1);
+    set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+    set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+    set_global_var(GVAR_KARMA_DEFENDER, 0);
+    set_global_var(GVAR_KARMA_WANDERER, 0);
+    set_global_var(GVAR_KARMA_BETRAYER, 0);
+    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+    set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+    set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+    if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
     end
     else begin
-        if (global_var(0) >= 750) then begin
-            set_global_var(38, 1);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+            set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
         end
         else begin
-            if (global_var(0) >= 500) then begin
-                set_global_var(39, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
             end
             else begin
-                if (global_var(0) >= 250) then begin
-                    set_global_var(40, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                    set_global_var(GVAR_KARMA_DEFENDER, 1);
                 end
                 else begin
-                    if (global_var(0) > -250) then begin
-                        set_global_var(41, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                        set_global_var(GVAR_KARMA_WANDERER, 1);
                     end
                     else begin
-                        if (global_var(0) > -500) then begin
-                            set_global_var(42, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                            set_global_var(GVAR_KARMA_BETRAYER, 1);
                         end
                         else begin
-                            if (global_var(0) > -750) then begin
-                                set_global_var(43, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                             end
                             else begin
-                                if (global_var(0) > -1000) then begin
-                                    set_global_var(44, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                    set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                 end
                                 else begin
-                                    set_global_var(45, 1);
+                                    set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                 end
                             end
                         end
@@ -963,118 +963,118 @@ begin
             end
         end
     end
-    set_global_var(51, global_var(51) + 5);
-    if (global_var(871) != 3) then begin
-        set_global_var(871, 0);
+    set_global_var(GVAR_TOWN_REP_ALBUQUERQUE, global_var(GVAR_TOWN_REP_ALBUQUERQUE) + 5);
+    if (global_var(GVAR_AQ_QUEST_KILL_WILLIAM_WIFE) != 3) then begin
+        set_global_var(GVAR_AQ_QUEST_KILL_WILLIAM_WIFE, 0);
     end
-    giq_option(4, 1629, 1404, Node999_2500caps, 50);
-    giq_option(-3, 1629, 1405, Node999_500caps, 50);
+    giq_option(4, NAME, 1404, Node999_2500caps, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 1405, Node999_500caps, NEUTRAL_REACTION);
 end
 
 procedure Node1450
 begin
-    gsay_reply(1629, 1450);
+    Reply(1450);
     if (not((local_var(6) bwand 16) != 0)) then begin
         set_local_var(6, local_var(6) bwor 16);
     end
-    giq_option(4, 1629, 1451, Quest_accepted, 50);
-    giq_option(4, 1629, 1452, Quest_declined, 50);
+    giq_option(4, NAME, 1451, Quest_accepted, NEUTRAL_REACTION);
+    giq_option(4, NAME, 1452, Quest_declined, NEUTRAL_REACTION);
 end
 
 procedure Node1500
 begin
-    gsay_reply(1629, 1500);
-    if ((global_var(875) == 1) and not((map_var(0) bwand 2) != 0)) then begin
-        giq_option(4, 1629, 1501, Node1300, 50);
+    Reply(1500);
+    if ((global_var(GVAR_AQ_QUEST_TRIBE_SOLVE_LOSTS) == 1) and not((map_var(0) bwand 2) != 0)) then begin
+        giq_option(4, NAME, 1501, Node1300, NEUTRAL_REACTION);
     end
-    if ((global_var(876) == 1) and not((map_var(0) bwand 4) != 0)) then begin
-        giq_option(4, 1629, 1502, Node1550, 50);
+    if ((global_var(GVAR_AQ_QUEST_TRIBE_HELP_REVEGE) == 1) and not((map_var(0) bwand 4) != 0)) then begin
+        giq_option(4, NAME, 1502, Node1550, NEUTRAL_REACTION);
     end
-    giq_option(4, 1629, 1503, Node999, 50);
-    giq_option(-3, 1629, 1504, Node999, 50);
+    giq_option(4, NAME, 1503, Node999, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 1504, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node1550
 begin
-    gsay_reply(1629, 1550);
+    Reply(1550);
     set_map_var(0, map_var(0) bwor 4);
     if ((local_var(6) bwand 1) == 0) then begin
-        giq_option(4, 1629, 1551, Node1050, 50);
+        giq_option(4, NAME, 1551, Node1050, NEUTRAL_REACTION);
     end
     else begin
-        giq_option(4, 1629, 1551, Node999, 50);
+        giq_option(4, NAME, 1551, Node999, NEUTRAL_REACTION);
     end
 end
 
 procedure Node1600
 begin
-    gsay_reply(1629, 1600);
+    Reply(1600);
     if (not((local_var(6) bwand 16) != 0)) then begin
         set_local_var(6, local_var(6) bwor 16);
     end
-    giq_option(4, 1629, 1601, Node999, 50);
+    giq_option(4, NAME, 1601, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node1650
 begin
-    if (get_critter_stat(dude_obj, 34) == 0) then begin
-        gsay_reply(1629, 1650);
+    if (dude_gender == 0) then begin
+        Reply(1650);
     end
-    if (get_critter_stat(dude_obj, 34) == 1) then begin
-        gsay_reply(1629, 1651);
+    if (dude_gender == 1) then begin
+        Reply(1651);
     end
-    if (get_critter_stat(dude_obj, 34) == 0) then begin
-        giq_option(-3, 1629, obj_name(dude_obj) + message_str(1629, 1652), Node1700, 50);
+    if (dude_gender == 0) then begin
+        giq_option(-3, NAME, dude_name + mstr(1652), Node1700, NEUTRAL_REACTION);
     end
-    if (get_critter_stat(dude_obj, 34) == 1) then begin
-        giq_option(-3, 1629, obj_name(dude_obj) + message_str(1629, 1653), Node1700, 50);
+    if (dude_gender == 1) then begin
+        giq_option(-3, NAME, dude_name + mstr(1653), Node1700, NEUTRAL_REACTION);
     end
-    giq_option(-3, 1629, 1654, Quest_declined_idiot, 50);
+    giq_option(-3, NAME, 1654, Quest_declined_idiot, NEUTRAL_REACTION);
 end
 
 procedure Node1700
 begin
-    if (get_critter_stat(dude_obj, 34) == 0) then begin
-        gsay_reply(1629, 1700);
+    if (dude_gender == 0) then begin
+        Reply(1700);
     end
-    if (get_critter_stat(dude_obj, 34) == 1) then begin
-        gsay_reply(1629, 1701);
+    if (dude_gender == 1) then begin
+        Reply(1701);
     end
-    giq_option(-3, 1629, 1702, Quest_accepted_idiot, 50);
-    giq_option(-3, 1629, 1703, Quest_declined_idiot, 50);
+    giq_option(-3, NAME, 1702, Quest_accepted_idiot, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 1703, Quest_declined_idiot, NEUTRAL_REACTION);
 end
 
 procedure Node1750
 begin
-    gsay_reply(1629, 1750);
-    if (not((global_var(866) bwand 32768) != 0)) then begin
-        set_global_var(866, global_var(866) bwor 32768);
+    Reply(1750);
+    if (not((global_var(GVAR_AQ_ALBUQ_SPEECHES) bwand 32768) != 0)) then begin
+        set_global_var(GVAR_AQ_ALBUQ_SPEECHES, global_var(GVAR_AQ_ALBUQ_SPEECHES) bwor 32768);
     end
-    giq_option(4, 1629, 1751, Node900, 50);
+    giq_option(4, NAME, 1751, Node900, NEUTRAL_REACTION);
     if ((local_var(6) bwand 1) == 0) then begin
-        giq_option(4, 1629, 1752, Node1050, 50);
+        giq_option(4, NAME, 1752, Node1050, NEUTRAL_REACTION);
     end
     else begin
-        giq_option(4, 1629, 1752, Node999, 50);
+        giq_option(4, NAME, 1752, Node999, NEUTRAL_REACTION);
     end
 end
 
 procedure Node1800
 begin
-    gsay_reply(1629, 1800);
+    Reply(1800);
     if (not((local_var(6) bwand 16) != 0)) then begin
         set_local_var(6, local_var(6) bwor 16);
     end
-    giq_option(4, 1629, 1801, Node999, 50);
+    giq_option(4, NAME, 1801, Node999, NEUTRAL_REACTION);
 end
 
 procedure Quest_accepted
 begin
-    if (global_var(870) == 0) then begin
-        set_global_var(870, 1);
+    if (global_var(GVAR_AQ_QUEST_BRING_VEX_HEAD) == 0) then begin
+        set_global_var(GVAR_AQ_QUEST_BRING_VEX_HEAD, 1);
     end
-    if ((party_member_obj(16777779) != 0) and ((global_var(910) bwand 32) == 0)) then begin
-        set_global_var(1015, global_var(1015) bwor 65536);
+    if ((party_member_obj(16777779) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 32) == 0)) then begin
+        set_global_var(GVAR_PTR_KERI_REACTIONS, global_var(GVAR_PTR_KERI_REACTIONS) bwor 65536);
     end
     if ((local_var(6) bwand 1) == 0) then begin
         call Node1050();
@@ -1094,11 +1094,11 @@ end
 
 procedure Quest_accepted_idiot
 begin
-    if (global_var(870) == 0) then begin
-        set_global_var(870, 1);
+    if (global_var(GVAR_AQ_QUEST_BRING_VEX_HEAD) == 0) then begin
+        set_global_var(GVAR_AQ_QUEST_BRING_VEX_HEAD, 1);
     end
-    if ((party_member_obj(16777779) != 0) and ((global_var(910) bwand 32) == 0)) then begin
-        set_global_var(1015, global_var(1015) bwor 65536);
+    if ((party_member_obj(16777779) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 32) == 0)) then begin
+        set_global_var(GVAR_PTR_KERI_REACTIONS, global_var(GVAR_PTR_KERI_REACTIONS) bwor 65536);
     end
     call Node999();
 end
@@ -1122,4 +1122,3 @@ begin
     add_timer_event(self_obj, 3, 3);
     call Node999();
 end
-
