diff --git a/Mods/Resurrection/Original/CCHADWIC.ssl b/Mods/Resurrection/ReDefine/CCHADWIC.ssl
index a191715..68894d8 100644
--- a/Mods/Resurrection/Original/CCHADWIC.ssl
+++ b/Mods/Resurrection/ReDefine/CCHADWIC.ssl
@@ -127,12 +127,12 @@ begin
             return 1;
         end
     end
-    if ((party_member_obj(16777779) != 0) and ((global_var(910) bwand 32) == 0)) then begin
+    if ((party_member_obj(16777779) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 32) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777779)) <= 5) then begin
             return 1;
         end
     end
-    if ((party_member_obj(16777811) != 0) and ((global_var(910) bwand 64) == 0)) then begin
+    if ((party_member_obj(16777811) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 64) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777811)) <= 5) then begin
             return 1;
         end
@@ -147,7 +147,7 @@ begin
             return 1;
         end
     end
-    if ((party_member_obj(16777814) != 0) and ((global_var(910) bwand 128) == 0)) then begin
+    if ((party_member_obj(16777814) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 128) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777814)) <= 5) then begin
             return 1;
         end
@@ -205,23 +205,23 @@ end
 
 procedure critter_p_proc
 begin
-    if (((global_var(843) bwand 1024) != 0) or ((global_var(843) bwand 4096) != 0) or ((global_var(843) bwand 2048) != 0) and (cur_map_index == 15) and (global_var(849) == 0)) then begin
+    if (((global_var(GVAR_C_RING_VICTORIES) bwand 1024) != 0) or ((global_var(GVAR_C_RING_VICTORIES) bwand 4096) != 0) or ((global_var(GVAR_C_RING_VICTORIES) bwand 2048) != 0) and (CUR_MAP_CORATH1) and (global_var(GVAR_C_RING_POINTER) == 0)) then begin
         destroy_object(self_obj);
     end
-    if ((global_var(849) == 2) and ((global_var(843) bwand 262144) != 0) and (local_var(4) != 1)) then begin
-        critter_add_trait(self_obj, 1, 6, 246);
-        critter_add_trait(self_obj, 1, 5, 2);
+    if ((global_var(GVAR_C_RING_POINTER) == 2) and ((global_var(GVAR_C_RING_VICTORIES) bwand 262144) != 0) and (local_var(4) != 1)) then begin
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 246);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_PEASANTS_NO_TALKING);
         set_local_var(4, 1);
     end
-    if ((local_var(4) == 1) or (global_var(853) == 1) or (global_var(824) == 1) and obj_can_see_obj(self_obj, dude_obj)) then begin
+    if ((local_var(4) == 1) or (global_var(GVAR_C_ENEMY_OBYVAT) == 1) or (global_var(GVAR_ENEMY_CORATH) == 1) and obj_can_see_obj(self_obj, dude_obj)) then begin
         if (critter_is_fleeing(self_obj)) then begin
             if (anim_busy(self_obj) == 0) then begin
                 if ((tile_distance_objs(self_obj, dude_obj) < 8) or obj_can_see_obj(self_obj, dude_obj)) then begin
                     if (anim_busy(self_obj) == 0) then begin
-                        GVar11 := rotation_to_tile(tile_num(dude_obj), tile_num(self_obj));
+                        GVar11 := rotation_to_tile(dude_tile, tile_num(self_obj));
                         animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(self_obj), GVar11, random(3, 10)), 1);
                         GVar11 := (GVar11 + 1) % 6;
-                        while ((anim_busy(self_obj) == 0) and (GVar11 != rotation_to_tile(tile_num(dude_obj), tile_num(self_obj)))) do begin
+                        while ((anim_busy(self_obj) == 0) and (GVar11 != rotation_to_tile(dude_tile, tile_num(self_obj)))) do begin
                             debug_msg("flee loop: rot == " + GVar11);
                             animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(self_obj), GVar11, random(3, 10)), 1);
                             GVar11 := (GVar11 + 1) % 6;
@@ -231,10 +231,10 @@ begin
             end
         end
         else begin
-            attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+            attack(dude_obj);
         end
     end
-    if (((local_var(6) bwand 16) != 0) and tile_in_tile_rect(27086, 27078, 28686, 28678, tile_num(dude_obj))) then begin
+    if (((local_var(6) bwand 16) != 0) and tile_in_tile_rect(27086, 27078, 28686, 28678, dude_tile)) then begin
         set_local_var(4, 1);
     end
 end
@@ -248,7 +248,7 @@ end
 
 procedure talk_p_proc
 begin
-    if (global_var(849) == 2) then begin
+    if (global_var(GVAR_C_RING_POINTER) == 2) then begin
         set_local_var(4, 1);
     end
     else begin
@@ -256,28 +256,28 @@ begin
         end
         else begin
             if (((local_var(6) bwand 8) != 0) and not(obj_carrying_pid_obj(self_obj, 540))) then begin
-                float_msg(self_obj, message_str(1534, random(330, 333)), 8);
+                float_msg(self_obj, mstr(random(330, 333)), FLOAT_MSG_YELLOW);
             end
             else begin
-                if ((global_var(840) != 0) and (global_var(825) != 0) and ((get_critter_stat(dude_obj, 4) + has_trait(0, dude_obj, 49)) > 3)) then begin
-                    anim(dude_obj, 1000, rotation_to_tile(tile_num(dude_obj), tile_num(self_obj)));
-                    anim(self_obj, 1000, rotation_to_tile(tile_num(self_obj), tile_num(dude_obj)));
+                if ((global_var(GVAR_C_CHADWIC_IS_MURDERER) != 0) and (global_var(GVAR_C_QUEST_MURDERS) != 0) and ((dude_iq + has_trait(TRAIT_PERK, dude_obj, PERK_smooth_talker)) > 3)) then begin
+                    anim(dude_obj, 1000, rotation_to_tile(dude_tile, tile_num(self_obj)));
+                    anim(self_obj, 1000, rotation_to_tile(tile_num(self_obj), dude_tile));
                     start_gdialog(1534, self_obj, 4, -1, -1);
                     gsay_start;
                     if ((local_var(6) bwand 1) == 0) then begin
                         call Node500();
                     end
                     else begin
-                        if ((global_var(827) == 2) and ((local_var(6) bwand 2) != 0)) then begin
+                        if ((global_var(GVAR_C_QUEST_CHADWIC_WANT_SUZAN) == 2) and ((local_var(6) bwand 2) != 0)) then begin
                             call Node1350();
                         end
-                        if ((global_var(827) == 2) and ((local_var(6) bwand 4) != 0)) then begin
+                        if ((global_var(GVAR_C_QUEST_CHADWIC_WANT_SUZAN) == 2) and ((local_var(6) bwand 4) != 0)) then begin
                             call Node1450();
                         end
-                        if (global_var(827) == 1) then begin
+                        if (global_var(GVAR_C_QUEST_CHADWIC_WANT_SUZAN) == 1) then begin
                             call Node1300();
                         end
-                        if (global_var(827) == 0) then begin
+                        if (global_var(GVAR_C_QUEST_CHADWIC_WANT_SUZAN) == 0) then begin
                             call Node1250();
                         end
                     end
@@ -286,45 +286,45 @@ begin
                     set_local_var(5, 1);
                 end
                 else begin
-                    if ((global_var(840) != 0) and (GVar27 == 0)) then begin
-                        float_msg(self_obj, message_str(1534, 311), 8);
+                    if ((global_var(GVAR_C_CHADWIC_IS_MURDERER) != 0) and (GVar27 == 0)) then begin
+                        float_msg(self_obj, mstr(311), FLOAT_MSG_YELLOW);
                     end
                     if ((party_member_obj(16777812) != 0) and (GVar27 == 0)) then begin
-                        if (get_critter_stat(dude_obj, 34) == 0) then begin
-                            float_msg(self_obj, message_str(1534, random(319, 322)), 8);
+                        if (dude_gender == 0) then begin
+                            float_msg(self_obj, mstr(random(319, 322)), FLOAT_MSG_YELLOW);
                         end
-                        if (get_critter_stat(dude_obj, 34) == 1) then begin
-                            float_msg(self_obj, message_str(1534, random(320, 323)), 8);
+                        if (dude_gender == 1) then begin
+                            float_msg(self_obj, mstr(random(320, 323)), FLOAT_MSG_YELLOW);
                         end
                         GVar27 := 1;
                     end
                     if (GVar27 == 0) then begin
-                        if (get_critter_stat(dude_obj, 34) == 0) then begin
-                            float_msg(self_obj, message_str(1534, random(312, 317)), 8);
+                        if (dude_gender == 0) then begin
+                            float_msg(self_obj, mstr(random(312, 317)), FLOAT_MSG_YELLOW);
                         end
-                        if (get_critter_stat(dude_obj, 34) == 1) then begin
-                            float_msg(self_obj, message_str(1534, random(314, 318)), 8);
+                        if (dude_gender == 1) then begin
+                            float_msg(self_obj, mstr(random(314, 318)), FLOAT_MSG_YELLOW);
                         end
                         GVar27 := 1;
                     end
                     GVar27 := 0;
                 end
             end
-            if (global_var(827) == 3) then begin
+            if (global_var(GVAR_C_QUEST_CHADWIC_WANT_SUZAN) == 3) then begin
                 game_ui_disable;
                 add_timer_event(self_obj, game_ticks(1), 10);
             end
-            if (global_var(840) == 3) then begin
+            if (global_var(GVAR_C_CHADWIC_IS_MURDERER) == 3) then begin
                 game_ui_disable;
                 add_timer_event(self_obj, game_ticks(1), 2);
             end
             if (((local_var(6) bwand 8) == 0) and not((local_var(6) bwand 1) == 0)) then begin
-                GVar17 := create_object_sid(540, 0, 0, -1);
+                GVar17 := create_object(540, 0, 0);
                 add_mult_objs_to_inven(self_obj, GVar17, 1);
                 set_local_var(6, local_var(6) bwor 8);
             end
         end
-        if ((global_var(849) == 2) and (GVar28 == 0)) then begin
+        if ((global_var(GVAR_C_RING_POINTER) == 2) and (GVar28 == 0)) then begin
             add_timer_event(map_event, game_ticks(1), 40);
             GVar28 := 1;
         end
@@ -333,58 +333,58 @@ end
 
 procedure destroy_p_proc
 begin
-    if (global_var(849) == 2) then begin
-        set_global_var(843, global_var(843) bwor 1024);
-        if (cur_map_index == 15) then begin
+    if (global_var(GVAR_C_RING_POINTER) == 2) then begin
+        set_global_var(GVAR_C_RING_VICTORIES, global_var(GVAR_C_RING_VICTORIES) bwor 1024);
+        if (CUR_MAP_CORATH1) then begin
             add_timer_event(map_event, game_ticks(0), 999);
         end
     end
     else begin
-        if (global_var(853) == 1) then begin
-            set_global_var(0, global_var(0) - 5);
+        if (global_var(GVAR_C_ENEMY_OBYVAT) == 1) then begin
+            set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) - 5);
         end
-        set_global_var(37, 0);
-        set_global_var(38, 0);
-        set_global_var(39, 0);
-        set_global_var(40, 0);
-        set_global_var(41, 0);
-        set_global_var(42, 0);
-        set_global_var(43, 0);
-        set_global_var(44, 0);
-        set_global_var(45, 0);
-        if (global_var(0) >= 1000) then begin
-            set_global_var(37, 1);
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+        set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+        set_global_var(GVAR_KARMA_DEFENDER, 0);
+        set_global_var(GVAR_KARMA_WANDERER, 0);
+        set_global_var(GVAR_KARMA_BETRAYER, 0);
+        set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+        set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+            set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
         end
         else begin
-            if (global_var(0) >= 750) then begin
-                set_global_var(38, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
             end
             else begin
-                if (global_var(0) >= 500) then begin
-                    set_global_var(39, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                 end
                 else begin
-                    if (global_var(0) >= 250) then begin
-                        set_global_var(40, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                        set_global_var(GVAR_KARMA_DEFENDER, 1);
                     end
                     else begin
-                        if (global_var(0) > -250) then begin
-                            set_global_var(41, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                            set_global_var(GVAR_KARMA_WANDERER, 1);
                         end
                         else begin
-                            if (global_var(0) > -500) then begin
-                                set_global_var(42, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                set_global_var(GVAR_KARMA_BETRAYER, 1);
                             end
                             else begin
-                                if (global_var(0) > -750) then begin
-                                    set_global_var(43, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -1000) then begin
-                                        set_global_var(44, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                     end
                                     else begin
-                                        set_global_var(45, 1);
+                                        set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                     end
                                 end
                             end
@@ -394,33 +394,33 @@ begin
             end
         end
     end
-    set_global_var(1018, global_var(1018) bwand (-1 - 8192));
-    set_global_var(1019, global_var(1019) bwand (-1 - 2048));
+    set_global_var(GVAR_PTR_GABRIEL_REACTIONS, global_var(GVAR_PTR_GABRIEL_REACTIONS) bwand (-1 - 8192));
+    set_global_var(GVAR_PTR_LYSTRA_REACTIONS_1, global_var(GVAR_PTR_LYSTRA_REACTIONS_1) bwand (-1 - 2048));
 end
 
 procedure look_at_p_proc
 begin
     script_overrides;
     if (local_var(5) == 1) then begin
-        display_msg(message_str(1534, 150));
+        display_mstr(150);
     end
     else begin
-        display_msg(message_str(1534, 100));
+        display_mstr(100);
     end
 end
 
 procedure description_p_proc
 begin
     script_overrides;
-    if (global_var(840) == 3) then begin
-        display_msg(message_str(1534, 251));
+    if (global_var(GVAR_C_CHADWIC_IS_MURDERER) == 3) then begin
+        display_mstr(251);
     end
     else begin
         if (local_var(5) == 1) then begin
-            display_msg(message_str(1534, 250));
+            display_mstr(250);
         end
         else begin
-            display_msg(message_str(1534, 200));
+            display_mstr(200);
         end
     end
 end
@@ -433,8 +433,8 @@ procedure damage_p_proc
 begin
     if (source_obj == dude_obj) then begin
         set_local_var(4, 1);
-        if (global_var(849) != 2) then begin
-            set_global_var(853, 1);
+        if (global_var(GVAR_C_RING_POINTER) != 2) then begin
+            set_global_var(GVAR_C_ENEMY_OBYVAT, 1);
         end
         set_local_var(6, local_var(6) bwor 64);
     end
@@ -442,24 +442,24 @@ end
 
 procedure map_enter_p_proc
 begin
-    if (global_var(849) == 2) then begin
-        critter_add_trait(self_obj, 1, 6, 246);
-        critter_add_trait(self_obj, 1, 5, 52);
+    if (global_var(GVAR_C_RING_POINTER) == 2) then begin
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 246);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_TOUGH_CITIZEN_NO_TALKING);
     end
     else begin
         if ((local_var(6) bwand 16) != 0) then begin
-            critter_add_trait(self_obj, 1, 6, 109);
+            critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 109);
         end
         else begin
-            critter_add_trait(self_obj, 1, 6, 243);
+            critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 243);
         end
-        critter_add_trait(self_obj, 1, 5, 52);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_TOUGH_CITIZEN_NO_TALKING);
     end
-    if ((global_var(827) == 2) and ((map_var(0) bwand 1) == 0)) then begin
+    if ((global_var(GVAR_C_QUEST_CHADWIC_WANT_SUZAN) == 2) and ((map_var(0) bwand 1) == 0)) then begin
         add_timer_event(self_obj, game_ticks(0), 1);
     end
     if (((game_time - local_var(7)) >= (24 * (60 * (60 * 10)) * 1)) and ((local_var(6) bwand 16) != 0)) then begin
-        set_global_var(842, global_var(842) bwor 65536);
+        set_global_var(GVAR_C_CORATH_STATUS, global_var(GVAR_C_CORATH_STATUS) bwor 65536);
         destroy_object(self_obj);
     end
 end
@@ -469,34 +469,34 @@ begin
     if (fixed_param == 4) then begin
         GVar24 := random(0, 10);
         if ((local_var(4) == 1) and obj_can_see_obj(self_obj, dude_obj) and (GVar24 >= 6) and not(critter_is_fleeing(self_obj))) then begin
-            if (global_var(840) != 0) then begin
-                float_msg(self_obj, message_str(1534, random(402, 408)), 2);
+            if (global_var(GVAR_C_CHADWIC_IS_MURDERER) != 0) then begin
+                float_msg(self_obj, mstr(random(402, 408)), FLOAT_MSG_RED);
             end
         end
     end
-    if ((global_var(849) == 2) and ((local_var(6) bwand 32) == 0) and ((local_var(6) bwand 64) != 0)) then begin
-        GVar17 := create_object_sid(319, 0, 0, -1);
+    if ((global_var(GVAR_C_RING_POINTER) == 2) and ((local_var(6) bwand 32) == 0) and ((local_var(6) bwand 64) != 0)) then begin
+        GVar17 := create_object(319, 0, 0);
         add_mult_objs_to_inven(self_obj, GVar17, 1);
         if (obj_is_carrying_obj_pid(self_obj, 319)) then begin
-            wield_obj_critter(self_obj, obj_carrying_pid_obj(self_obj, 319));
+            wield_obj(obj_carrying_pid_obj(self_obj, 319));
         end
-        float_msg(self_obj, message_str(1534, 401), 2);
+        float_msg(self_obj, mstr(401), FLOAT_MSG_RED);
         add_timer_event(map_event, game_ticks(0), 45);
         add_timer_event(map_event, game_ticks(1), 46);
         set_local_var(6, local_var(6) bwor 32);
-        set_global_var(1027, global_var(1027) bwor 32);
+        set_global_var(GVAR_CORATH_DIALOG_BITS_NEXT, global_var(GVAR_CORATH_DIALOG_BITS_NEXT) bwor 32);
     end
 end
 
 procedure timed_event_p_proc
 begin
-    if (((map_var(0) bwand 1) == 0) and (global_var(827) == 2)) then begin
+    if (((map_var(0) bwand 1) == 0) and (global_var(GVAR_C_QUEST_CHADWIC_WANT_SUZAN) == 2)) then begin
         if (fixed_param == 1) then begin
             GVar23 := GVar23 + 1;
         end
         if (GVar23 == 1) then begin
             game_ui_disable;
-            GVar26 := create_object_sid(16777875, 21710, 0, 1535);
+            GVar26 := create_object_sid(16777875, 21710, 0, SCRIPT_CSUZAN);
             add_timer_event(self_obj, game_ticks(0), 1);
         end
         if (GVar23 == 2) then begin
@@ -510,7 +510,7 @@ begin
             rm_timer_event(self_obj);
         end
     end
-    if ((global_var(840) == 3) or (global_var(825) == 3)) then begin
+    if ((global_var(GVAR_C_CHADWIC_IS_MURDERER) == 3) or (global_var(GVAR_C_QUEST_MURDERS) == 3)) then begin
         if (fixed_param == 2) then begin
             GVar22 := GVar22 + 1;
         end
@@ -523,13 +523,13 @@ begin
                 move_to(self_obj, 27479, elevation(self_obj));
                 anim(self_obj, 1000, 3);
                 set_local_var(6, local_var(6) bwor 16);
-                GVar11 := create_object_sid(128, 0, 0, -1);
+                GVar11 := create_object(128, 0, 0);
                 move_obj_inven_to_obj(self_obj, GVar11);
                 destroy_object(GVar11);
                 GVar31 := tile_contains_pid_obj(24668, 0, 183);
-                GVar17 := create_object_sid(319, 0, 0, -1);
+                GVar17 := create_object(319, 0, 0);
                 add_mult_objs_to_inven(GVar31, GVar17, 1);
-                critter_add_trait(self_obj, 1, 6, 109);
+                critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 109);
                 set_local_var(7, game_time);
                 add_timer_event(self_obj, game_ticks(1), 2);
             end
@@ -539,7 +539,7 @@ begin
             end
         end
     end
-    if (global_var(827) == 3) then begin
+    if (global_var(GVAR_C_QUEST_CHADWIC_WANT_SUZAN) == 3) then begin
         if (fixed_param == 10) then begin
             GVar21 := GVar21 + 1;
         end
@@ -554,7 +554,7 @@ begin
         if (GVar21 == 3) then begin
             gfade_in(600);
             game_ui_enable;
-            set_global_var(840, 0);
+            set_global_var(GVAR_C_CHADWIC_IS_MURDERER, 0);
             add_timer_event(self_obj, game_ticks(1), 10);
         end
         if (GVar21 == 4) then begin
@@ -570,52 +570,52 @@ end
 
 procedure Node999
 begin
-    if (global_var(827) == 3) then begin
+    if (global_var(GVAR_C_QUEST_CHADWIC_WANT_SUZAN) == 3) then begin
         if (GVar29 == 1) then begin
-            set_global_var(0, global_var(0) - 20);
+            set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) - 20);
         end
-        set_global_var(37, 0);
-        set_global_var(38, 0);
-        set_global_var(39, 0);
-        set_global_var(40, 0);
-        set_global_var(41, 0);
-        set_global_var(42, 0);
-        set_global_var(43, 0);
-        set_global_var(44, 0);
-        set_global_var(45, 0);
-        if (global_var(0) >= 1000) then begin
-            set_global_var(37, 1);
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+        set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+        set_global_var(GVAR_KARMA_DEFENDER, 0);
+        set_global_var(GVAR_KARMA_WANDERER, 0);
+        set_global_var(GVAR_KARMA_BETRAYER, 0);
+        set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+        set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+            set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
         end
         else begin
-            if (global_var(0) >= 750) then begin
-                set_global_var(38, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
             end
             else begin
-                if (global_var(0) >= 500) then begin
-                    set_global_var(39, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                 end
                 else begin
-                    if (global_var(0) >= 250) then begin
-                        set_global_var(40, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                        set_global_var(GVAR_KARMA_DEFENDER, 1);
                     end
                     else begin
-                        if (global_var(0) > -250) then begin
-                            set_global_var(41, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                            set_global_var(GVAR_KARMA_WANDERER, 1);
                         end
                         else begin
-                            if (global_var(0) > -500) then begin
-                                set_global_var(42, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                set_global_var(GVAR_KARMA_BETRAYER, 1);
                             end
                             else begin
-                                if (global_var(0) > -750) then begin
-                                    set_global_var(43, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -1000) then begin
-                                        set_global_var(44, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                     end
                                     else begin
-                                        set_global_var(45, 1);
+                                        set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                     end
                                 end
                             end
@@ -625,50 +625,50 @@ begin
             end
         end
         if (GVar30 == 1) then begin
-            set_global_var(0, global_var(0) - 10);
+            set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) - 10);
         end
-        set_global_var(37, 0);
-        set_global_var(38, 0);
-        set_global_var(39, 0);
-        set_global_var(40, 0);
-        set_global_var(41, 0);
-        set_global_var(42, 0);
-        set_global_var(43, 0);
-        set_global_var(44, 0);
-        set_global_var(45, 0);
-        if (global_var(0) >= 1000) then begin
-            set_global_var(37, 1);
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+        set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+        set_global_var(GVAR_KARMA_DEFENDER, 0);
+        set_global_var(GVAR_KARMA_WANDERER, 0);
+        set_global_var(GVAR_KARMA_BETRAYER, 0);
+        set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+        set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+            set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
         end
         else begin
-            if (global_var(0) >= 750) then begin
-                set_global_var(38, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
             end
             else begin
-                if (global_var(0) >= 500) then begin
-                    set_global_var(39, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                 end
                 else begin
-                    if (global_var(0) >= 250) then begin
-                        set_global_var(40, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                        set_global_var(GVAR_KARMA_DEFENDER, 1);
                     end
                     else begin
-                        if (global_var(0) > -250) then begin
-                            set_global_var(41, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                            set_global_var(GVAR_KARMA_WANDERER, 1);
                         end
                         else begin
-                            if (global_var(0) > -500) then begin
-                                set_global_var(42, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                set_global_var(GVAR_KARMA_BETRAYER, 1);
                             end
                             else begin
-                                if (global_var(0) > -750) then begin
-                                    set_global_var(43, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -1000) then begin
-                                        set_global_var(44, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                     end
                                     else begin
-                                        set_global_var(45, 1);
+                                        set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                     end
                                 end
                             end
@@ -685,217 +685,217 @@ end
 
 procedure Node500
 begin
-    gsay_reply(1534, 500);
-    giq_option(6, 1534, 501, Node550, 50);
-    if (get_critter_stat(dude_obj, 4) < 8) then begin
-        giq_option(4, 1534, 502, Node600, 50);
+    Reply(500);
+    giq_option(6, NAME, 501, Node550, NEUTRAL_REACTION);
+    if (dude_iq < 8) then begin
+        giq_option(4, NAME, 502, Node600, NEUTRAL_REACTION);
     end
-    giq_option(4, 1534, 503, Node999, 50);
+    giq_option(4, NAME, 503, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node550
 begin
-    gsay_reply(1534, 550);
-    giq_option(4, 1534, 551, Node600, 50);
-    giq_option(4, 1534, 552, Node999, 50);
+    Reply(550);
+    giq_option(4, NAME, 551, Node600, NEUTRAL_REACTION);
+    giq_option(4, NAME, 552, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node600
 begin
-    gsay_reply(1534, 600);
+    Reply(600);
     set_local_var(6, local_var(6) bwor 1);
-    giq_option(4, 1534, 602, Node650, 50);
-    giq_option(4, 1534, 601, Node700, 50);
+    giq_option(4, NAME, 602, Node650, NEUTRAL_REACTION);
+    giq_option(4, NAME, 601, Node700, NEUTRAL_REACTION);
 end
 
 procedure Node650
 begin
-    gsay_reply(1534, 650);
-    giq_option(4, 1534, 652, Node750, 50);
-    giq_option(4, 1534, 651, Node800, 50);
-    giq_option(4, 1534, 653, Node999, 50);
+    Reply(650);
+    giq_option(4, NAME, 652, Node750, NEUTRAL_REACTION);
+    giq_option(4, NAME, 651, Node800, NEUTRAL_REACTION);
+    giq_option(4, NAME, 653, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node700
 begin
-    gsay_reply(1534, 700);
-    giq_option(4, 1534, 702, Node750, 50);
-    giq_option(4, 1534, 701, Node800, 50);
-    giq_option(4, 1534, 703, Node999, 50);
+    Reply(700);
+    giq_option(4, NAME, 702, Node750, NEUTRAL_REACTION);
+    giq_option(4, NAME, 701, Node800, NEUTRAL_REACTION);
+    giq_option(4, NAME, 703, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node750
 begin
-    gsay_reply(1534, 750);
-    giq_option(4, 1534, 751, Go_To_Ring, 50);
+    Reply(750);
+    giq_option(4, NAME, 751, Go_To_Ring, NEUTRAL_REACTION);
 end
 
 procedure Node800
 begin
-    gsay_reply(1534, 800);
-    if (has_skill(dude_obj, 14) > 80) then begin
-        giq_option(7, 1534, 801, Node850, 50);
+    Reply(800);
+    if (dude_skill(SKILL_SPEECH) > 80) then begin
+        giq_option(7, NAME, 801, Node850, NEUTRAL_REACTION);
     end
-    if (get_critter_stat(dude_obj, 34) == 0) then begin
-        giq_option(4, 1534, 802, Node950, 50);
+    if (dude_gender == 0) then begin
+        giq_option(4, NAME, 802, Node950, NEUTRAL_REACTION);
     end
-    if (get_critter_stat(dude_obj, 34) == 1) then begin
-        giq_option(4, 1534, 803, Node950, 50);
+    if (dude_gender == 1) then begin
+        giq_option(4, NAME, 803, Node950, NEUTRAL_REACTION);
     end
-    giq_option(4, 1534, 804, Node750, 50);
-    giq_option(4, 1534, 805, Node999, 50);
+    giq_option(4, NAME, 804, Node750, NEUTRAL_REACTION);
+    giq_option(4, NAME, 805, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node850
 begin
-    gsay_reply(1534, 850);
-    giq_option(4, 1534, 851, Node900, 50);
+    Reply(850);
+    giq_option(4, NAME, 851, Node900, NEUTRAL_REACTION);
 end
 
 procedure Node900
 begin
-    gsay_reply(1534, 900);
-    set_global_var(840, 3);
-    set_global_var(1018, global_var(1018) bwand (-1 - 8192));
-    giq_option(4, 1534, 901, Node999, 50);
+    Reply(900);
+    set_global_var(GVAR_C_CHADWIC_IS_MURDERER, 3);
+    set_global_var(GVAR_PTR_GABRIEL_REACTIONS, global_var(GVAR_PTR_GABRIEL_REACTIONS) bwand (-1 - 8192));
+    giq_option(4, NAME, 901, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node950
 begin
-    gsay_reply(1534, 950);
-    set_global_var(840, 2);
-    giq_option(4, 1534, 951, Node1000, 50);
-    giq_option(4, 1534, 952, Node1050, 50);
-    giq_option(4, 1534, 953, Node1200, 50);
-    giq_option(4, 1534, 954, Node999, 50);
+    Reply(950);
+    set_global_var(GVAR_C_CHADWIC_IS_MURDERER, 2);
+    giq_option(4, NAME, 951, Node1000, NEUTRAL_REACTION);
+    giq_option(4, NAME, 952, Node1050, NEUTRAL_REACTION);
+    giq_option(4, NAME, 953, Node1200, NEUTRAL_REACTION);
+    giq_option(4, NAME, 954, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node1000
 begin
-    gsay_reply(1534, 1000);
-    if (has_skill(dude_obj, 14) > 80) then begin
-        giq_option(7, 1534, 1003, Node850, 50);
+    Reply(1000);
+    if (dude_skill(SKILL_SPEECH) > 80) then begin
+        giq_option(7, NAME, 1003, Node850, NEUTRAL_REACTION);
     end
-    giq_option(4, 1534, 1001, Node1050, 50);
-    giq_option(4, 1534, 1002, Node1200, 50);
-    giq_option(4, 1534, 1004, Node750, 50);
-    giq_option(4, 1534, 1005, Node999, 50);
+    giq_option(4, NAME, 1001, Node1050, NEUTRAL_REACTION);
+    giq_option(4, NAME, 1002, Node1200, NEUTRAL_REACTION);
+    giq_option(4, NAME, 1004, Node750, NEUTRAL_REACTION);
+    giq_option(4, NAME, 1005, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node1050
 begin
-    gsay_reply(1534, 1050);
+    Reply(1050);
     set_local_var(6, local_var(6) bwor 2);
-    set_global_var(840, 2);
-    giq_option(4, 1534, 1051, Node1100, 50);
-    giq_option(4, 1534, 1052, Node999, 50);
+    set_global_var(GVAR_C_CHADWIC_IS_MURDERER, 2);
+    giq_option(4, NAME, 1051, Node1100, NEUTRAL_REACTION);
+    giq_option(4, NAME, 1052, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node1100
 begin
-    gsay_reply(1534, 1100);
-    giq_option(4, 1534, 1101, Node1150, 50);
+    Reply(1100);
+    giq_option(4, NAME, 1101, Node1150, NEUTRAL_REACTION);
 end
 
 procedure Node1150
 begin
-    gsay_reply(1534, 1150);
-    giq_option(4, 1534, 1151, Chadwic_Quest_Accepted, 50);
-    giq_option(4, 1534, 1152, Node999, 50);
+    Reply(1150);
+    giq_option(4, NAME, 1151, Chadwic_Quest_Accepted, NEUTRAL_REACTION);
+    giq_option(4, NAME, 1152, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node1200
 begin
-    gsay_reply(1534, 1200);
+    Reply(1200);
     set_local_var(6, local_var(6) bwor 4);
-    set_global_var(840, 2);
-    giq_option(4, 1534, 1201, Node1100, 50);
-    giq_option(4, 1534, 1202, Node999, 50);
+    set_global_var(GVAR_C_CHADWIC_IS_MURDERER, 2);
+    giq_option(4, NAME, 1201, Node1100, NEUTRAL_REACTION);
+    giq_option(4, NAME, 1202, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node1250
 begin
-    gsay_reply(1534, 1250);
-    giq_option(4, 1534, 1251, Node800, 50);
-    giq_option(4, 1534, 1252, Node750, 50);
-    giq_option(4, 1534, 1253, Node999, 50);
+    Reply(1250);
+    giq_option(4, NAME, 1251, Node800, NEUTRAL_REACTION);
+    giq_option(4, NAME, 1252, Node750, NEUTRAL_REACTION);
+    giq_option(4, NAME, 1253, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node1300
 begin
-    gsay_reply(1534, 1300);
-    if ((get_critter_stat(dude_obj, 34) == 0) and (has_skill(dude_obj, 14) > 80)) then begin
-        giq_option(7, 1534, 1301, Node850, 50);
+    Reply(1300);
+    if ((dude_gender == 0) and (dude_skill(SKILL_SPEECH) > 80)) then begin
+        giq_option(7, NAME, 1301, Node850, NEUTRAL_REACTION);
     end
-    if ((get_critter_stat(dude_obj, 34) == 1) and (has_skill(dude_obj, 14) > 80)) then begin
-        giq_option(7, 1534, 1302, Node850, 50);
+    if ((dude_gender == 1) and (dude_skill(SKILL_SPEECH) > 80)) then begin
+        giq_option(7, NAME, 1302, Node850, NEUTRAL_REACTION);
     end
-    giq_option(4, 1534, 1303, Node750, 50);
-    giq_option(4, 1534, 1304, Node999, 50);
+    giq_option(4, NAME, 1303, Node750, NEUTRAL_REACTION);
+    giq_option(4, NAME, 1304, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node1350
 begin
-    if (get_critter_stat(dude_obj, 6) == 10) then begin
-        gsay_reply(1534, 1353);
+    if (dude_luck == 10) then begin
+        Reply(1353);
     end
     else begin
-        gsay_reply(1534, 1350);
+        Reply(1350);
     end
-    set_global_var(827, 3);
+    set_global_var(GVAR_C_QUEST_CHADWIC_WANT_SUZAN, 3);
     give_exp_points(800);
-    display_msg(message_str(14, 100) + 800 + message_str(14, 101));
-    set_global_var(1018, global_var(1018) bwand (-1 - 8192));
+    display_msg(g_mstr(100) + 800 + g_mstr(101));
+    set_global_var(GVAR_PTR_GABRIEL_REACTIONS, global_var(GVAR_PTR_GABRIEL_REACTIONS) bwand (-1 - 8192));
     GVar29 := 1;
-    giq_option(4, 1534, 1351, Node1400, 50);
-    if ((party_member_obj(16777814) != 0) and ((global_var(910) bwand 128) == 0)) then begin
-        giq_option(4, 1534, 1352, Quest_Done_LystraReact, 50);
+    giq_option(4, NAME, 1351, Node1400, NEUTRAL_REACTION);
+    if ((party_member_obj(16777814) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 128) == 0)) then begin
+        giq_option(4, NAME, 1352, Quest_Done_LystraReact, NEUTRAL_REACTION);
     end
     else begin
-        giq_option(4, 1534, 1352, Node999, 50);
+        giq_option(4, NAME, 1352, Node999, NEUTRAL_REACTION);
     end
 end
 
 procedure Node1400
 begin
-    gsay_reply(1534, 1400);
+    Reply(1400);
     item_caps_adjust(dude_obj, 600);
     GVar29 := 0;
     GVar30 := 1;
-    if ((party_member_obj(16777814) != 0) and ((global_var(910) bwand 128) == 0)) then begin
-        giq_option(4, 1534, 1401, Quest_Done_LystraReact, 50);
+    if ((party_member_obj(16777814) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 128) == 0)) then begin
+        giq_option(4, NAME, 1401, Quest_Done_LystraReact, NEUTRAL_REACTION);
     end
     else begin
-        giq_option(4, 1534, 1401, Node999, 50);
+        giq_option(4, NAME, 1401, Node999, NEUTRAL_REACTION);
     end
 end
 
 procedure Node1450
 begin
-    if (get_critter_stat(dude_obj, 6) == 10) then begin
-        gsay_reply(1534, 1452);
+    if (dude_luck == 10) then begin
+        Reply(1452);
     end
     else begin
-        gsay_reply(1534, 1450);
+        Reply(1450);
     end
     item_caps_adjust(dude_obj, 600);
     give_exp_points(800);
-    display_msg(message_str(14, 100) + 800 + message_str(14, 101));
-    set_global_var(827, 3);
-    set_global_var(1018, global_var(1018) bwand (-1 - 8192));
+    display_msg(g_mstr(100) + 800 + g_mstr(101));
+    set_global_var(GVAR_C_QUEST_CHADWIC_WANT_SUZAN, 3);
+    set_global_var(GVAR_PTR_GABRIEL_REACTIONS, global_var(GVAR_PTR_GABRIEL_REACTIONS) bwand (-1 - 8192));
     GVar29 := 0;
     GVar30 := 1;
-    if ((party_member_obj(16777814) != 0) and ((global_var(910) bwand 128) == 0)) then begin
-        giq_option(4, 1534, 1451, Quest_Done_LystraReact, 50);
+    if ((party_member_obj(16777814) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 128) == 0)) then begin
+        giq_option(4, NAME, 1451, Quest_Done_LystraReact, NEUTRAL_REACTION);
     end
     else begin
-        giq_option(4, 1534, 1451, Node999, 50);
+        giq_option(4, NAME, 1451, Node999, NEUTRAL_REACTION);
     end
 end
 
 procedure Go_To_Ring
 begin
-    set_global_var(849, 2);
+    set_global_var(GVAR_C_RING_POINTER, 2);
     if (obj_is_carrying_obj_pid(self_obj, 540)) then begin
         GVar19 := obj_is_carrying_obj_pid(self_obj, 540);
         if (1 < GVar19) then begin
@@ -905,17 +905,17 @@ begin
             GVar17 := obj_carrying_pid_obj(self_obj, 540);
             if (obj_type(self_obj) == 1) then begin
                 GVar17 := obj_carrying_pid_obj(self_obj, 540);
-                if (critter_inven_obj(self_obj, 0) == GVar17) then begin
-                    if (obj_item_subtype(critter_inven_obj(self_obj, 0)) == 0) then begin
-                        GVar17 := critter_inven_obj(self_obj, 0);
+                if (critter_inven_obj(self_obj, INVEN_TYPE_WORN) == GVar17) then begin
+                    if (obj_item_subtype(critter_inven_obj(self_obj, INVEN_TYPE_WORN)) == 0) then begin
+                        GVar17 := critter_inven_obj(self_obj, INVEN_TYPE_WORN);
                         debug_msg("armour pid == " + obj_pid(GVar17));
                         rm_obj_from_inven(self_obj, GVar17);
                         add_obj_to_inven(self_obj, GVar17);
                     end
                 end
                 else begin
-                    if ((critter_inven_obj(self_obj, 2) == GVar17) or (critter_inven_obj(self_obj, 1) == GVar17)) then begin
-                        metarule(43, self_obj);
+                    if ((critter_inven_obj(self_obj, INVEN_TYPE_LEFT_HAND) == GVar17) or (critter_inven_obj(self_obj, INVEN_TYPE_RIGHT_HAND) == GVar17)) then begin
+                        inven_unwield(self_obj);
                     end
                 end
             end
@@ -936,70 +936,70 @@ begin
         GVar17 := 0;
         GVar16 := 0;
     end
-    set_global_var(1018, global_var(1018) bwand (-1 - 8192));
+    set_global_var(GVAR_PTR_GABRIEL_REACTIONS, global_var(GVAR_PTR_GABRIEL_REACTIONS) bwand (-1 - 8192));
     call Node999();
 end
 
 procedure Chadwic_Quest_Accepted
 begin
-    set_global_var(827, 1);
-    if ((party_member_obj(16777811) != 0) and ((global_var(910) bwand 64) == 0)) then begin
-        set_global_var(1018, global_var(1018) bwor 8192);
+    set_global_var(GVAR_C_QUEST_CHADWIC_WANT_SUZAN, 1);
+    if ((party_member_obj(16777811) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 64) == 0)) then begin
+        set_global_var(GVAR_PTR_GABRIEL_REACTIONS, global_var(GVAR_PTR_GABRIEL_REACTIONS) bwor 8192);
     end
-    if ((party_member_obj(16777814) != 0) and ((global_var(910) bwand 128) == 0)) then begin
-        set_global_var(1019, global_var(1019) bwor 2048);
+    if ((party_member_obj(16777814) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 128) == 0)) then begin
+        set_global_var(GVAR_PTR_LYSTRA_REACTIONS_1, global_var(GVAR_PTR_LYSTRA_REACTIONS_1) bwor 2048);
     end
     call Node999();
 end
 
 procedure Quest_Done_LystraReact
 begin
-    if (global_var(827) == 3) then begin
+    if (global_var(GVAR_C_QUEST_CHADWIC_WANT_SUZAN) == 3) then begin
         if (GVar29 == 1) then begin
-            set_global_var(0, global_var(0) - 20);
+            set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) - 20);
         end
-        set_global_var(37, 0);
-        set_global_var(38, 0);
-        set_global_var(39, 0);
-        set_global_var(40, 0);
-        set_global_var(41, 0);
-        set_global_var(42, 0);
-        set_global_var(43, 0);
-        set_global_var(44, 0);
-        set_global_var(45, 0);
-        if (global_var(0) >= 1000) then begin
-            set_global_var(37, 1);
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+        set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+        set_global_var(GVAR_KARMA_DEFENDER, 0);
+        set_global_var(GVAR_KARMA_WANDERER, 0);
+        set_global_var(GVAR_KARMA_BETRAYER, 0);
+        set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+        set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+            set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
         end
         else begin
-            if (global_var(0) >= 750) then begin
-                set_global_var(38, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
             end
             else begin
-                if (global_var(0) >= 500) then begin
-                    set_global_var(39, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                 end
                 else begin
-                    if (global_var(0) >= 250) then begin
-                        set_global_var(40, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                        set_global_var(GVAR_KARMA_DEFENDER, 1);
                     end
                     else begin
-                        if (global_var(0) > -250) then begin
-                            set_global_var(41, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                            set_global_var(GVAR_KARMA_WANDERER, 1);
                         end
                         else begin
-                            if (global_var(0) > -500) then begin
-                                set_global_var(42, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                set_global_var(GVAR_KARMA_BETRAYER, 1);
                             end
                             else begin
-                                if (global_var(0) > -750) then begin
-                                    set_global_var(43, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -1000) then begin
-                                        set_global_var(44, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                     end
                                     else begin
-                                        set_global_var(45, 1);
+                                        set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                     end
                                 end
                             end
@@ -1009,50 +1009,50 @@ begin
             end
         end
         if (GVar30 == 1) then begin
-            set_global_var(0, global_var(0) - 10);
+            set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) - 10);
         end
-        set_global_var(37, 0);
-        set_global_var(38, 0);
-        set_global_var(39, 0);
-        set_global_var(40, 0);
-        set_global_var(41, 0);
-        set_global_var(42, 0);
-        set_global_var(43, 0);
-        set_global_var(44, 0);
-        set_global_var(45, 0);
-        if (global_var(0) >= 1000) then begin
-            set_global_var(37, 1);
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+        set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+        set_global_var(GVAR_KARMA_DEFENDER, 0);
+        set_global_var(GVAR_KARMA_WANDERER, 0);
+        set_global_var(GVAR_KARMA_BETRAYER, 0);
+        set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+        set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+            set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
         end
         else begin
-            if (global_var(0) >= 750) then begin
-                set_global_var(38, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
             end
             else begin
-                if (global_var(0) >= 500) then begin
-                    set_global_var(39, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                 end
                 else begin
-                    if (global_var(0) >= 250) then begin
-                        set_global_var(40, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                        set_global_var(GVAR_KARMA_DEFENDER, 1);
                     end
                     else begin
-                        if (global_var(0) > -250) then begin
-                            set_global_var(41, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                            set_global_var(GVAR_KARMA_WANDERER, 1);
                         end
                         else begin
-                            if (global_var(0) > -500) then begin
-                                set_global_var(42, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                set_global_var(GVAR_KARMA_BETRAYER, 1);
                             end
                             else begin
-                                if (global_var(0) > -750) then begin
-                                    set_global_var(43, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -1000) then begin
-                                        set_global_var(44, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                     end
                                     else begin
-                                        set_global_var(45, 1);
+                                        set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                     end
                                 end
                             end
@@ -1067,4 +1067,3 @@ begin
     anim(self_obj, 1000, 4);
     add_timer_event(party_member_obj(16777814), 30, 17);
 end
-
