diff --git a/Mods/Resurrection/Original/CHORNIK1.ssl b/Mods/Resurrection/ReDefine/CHORNIK1.ssl
index f3b1514..5f1f9ea 100644
--- a/Mods/Resurrection/Original/CHORNIK1.ssl
+++ b/Mods/Resurrection/ReDefine/CHORNIK1.ssl
@@ -108,12 +108,12 @@ begin
             return 1;
         end
     end
-    if ((party_member_obj(16777779) != 0) and ((global_var(910) bwand 32) == 0)) then begin
+    if ((party_member_obj(16777779) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 32) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777779)) <= 5) then begin
             return 1;
         end
     end
-    if ((party_member_obj(16777811) != 0) and ((global_var(910) bwand 64) == 0)) then begin
+    if ((party_member_obj(16777811) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 64) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777811)) <= 5) then begin
             return 1;
         end
@@ -128,7 +128,7 @@ begin
             return 1;
         end
     end
-    if ((party_member_obj(16777814) != 0) and ((global_var(910) bwand 128) == 0)) then begin
+    if ((party_member_obj(16777814) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 128) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777814)) <= 5) then begin
             return 1;
         end
@@ -188,15 +188,15 @@ begin
     if (map_var(3) != 0) then begin
         set_local_var(4, 0);
     end
-    if ((local_var(4) == 1) or (global_var(853) == 1) or (global_var(850) == 1) and obj_can_see_obj(self_obj, dude_obj) and (has_trait(1, self_obj, 666) == 1)) then begin
+    if ((local_var(4) == 1) or (global_var(GVAR_C_ENEMY_OBYVAT) == 1) or (global_var(GVAR_C_ENEMY_NESTOR) == 1) and obj_can_see_obj(self_obj, dude_obj) and (obj_is_visible_flag(self_obj) == 1)) then begin
         if (critter_is_fleeing(self_obj)) then begin
             if (anim_busy(self_obj) == 0) then begin
                 if ((tile_distance_objs(self_obj, dude_obj) < 8) or obj_can_see_obj(self_obj, dude_obj)) then begin
                     if (anim_busy(self_obj) == 0) then begin
-                        GVar11 := rotation_to_tile(tile_num(dude_obj), tile_num(self_obj));
+                        GVar11 := rotation_to_tile(dude_tile, tile_num(self_obj));
                         animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(self_obj), GVar11, random(3, 10)), 1);
                         GVar11 := (GVar11 + 1) % 6;
-                        while ((anim_busy(self_obj) == 0) and (GVar11 != rotation_to_tile(tile_num(dude_obj), tile_num(self_obj)))) do begin
+                        while ((anim_busy(self_obj) == 0) and (GVar11 != rotation_to_tile(dude_tile, tile_num(self_obj)))) do begin
                             debug_msg("flee loop: rot == " + GVar11);
                             animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(self_obj), GVar11, random(3, 10)), 1);
                             GVar11 := (GVar11 + 1) % 6;
@@ -206,11 +206,11 @@ begin
             end
         end
         else begin
-            attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+            attack(dude_obj);
         end
     end
     if ((map_var(1) == 1) and ((map_var(0) bwand 64) == 0) and (GVar21 == 0)) then begin
-        float_msg(self_obj, message_str(1548, random(409, 418)), 2);
+        float_msg(self_obj, mstr(random(409, 418)), FLOAT_MSG_RED);
         GVar21 := 1;
     end
 end
@@ -225,8 +225,8 @@ end
 procedure talk_p_proc
 begin
     if ((map_var(0) bwand 32) == 0) then begin
-        anim(dude_obj, 1000, rotation_to_tile(tile_num(dude_obj), tile_num(self_obj)));
-        anim(self_obj, 1000, rotation_to_tile(tile_num(self_obj), tile_num(dude_obj)));
+        anim(dude_obj, 1000, rotation_to_tile(dude_tile, tile_num(self_obj)));
+        anim(self_obj, 1000, rotation_to_tile(tile_num(self_obj), dude_tile));
         start_gdialog(1548, self_obj, 4, -1, -1);
         gsay_start;
         if ((map_var(0) bwand 1) == 0) then begin
@@ -240,14 +240,14 @@ begin
     end
     else begin
         if (not(combat_is_initialized)) then begin
-            float_msg(self_obj, message_str(1548, random(302, 305)), 8);
+            float_msg(self_obj, mstr(random(302, 305)), FLOAT_MSG_YELLOW);
         end
     end
-    if (global_var(841) == 2) then begin
+    if (global_var(GVAR_C_STRIKE_IN_MINE) == 2) then begin
         game_ui_disable;
         add_timer_event(Mine_map, game_ticks(1), 2);
     end
-    if (global_var(841) == 3) then begin
+    if (global_var(GVAR_C_STRIKE_IN_MINE) == 3) then begin
         game_ui_disable;
         add_timer_event(Mine_map, game_ticks(1), 3);
     end
@@ -255,51 +255,51 @@ end
 
 procedure destroy_p_proc
 begin
-    if (global_var(841) == 1) then begin
-        set_global_var(0, global_var(0) - 5);
-    end
-    set_global_var(37, 0);
-    set_global_var(38, 0);
-    set_global_var(39, 0);
-    set_global_var(40, 0);
-    set_global_var(41, 0);
-    set_global_var(42, 0);
-    set_global_var(43, 0);
-    set_global_var(44, 0);
-    set_global_var(45, 0);
-    if (global_var(0) >= 1000) then begin
-        set_global_var(37, 1);
+    if (global_var(GVAR_C_STRIKE_IN_MINE) == 1) then begin
+        set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) - 5);
+    end
+    set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+    set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+    set_global_var(GVAR_KARMA_DEFENDER, 0);
+    set_global_var(GVAR_KARMA_WANDERER, 0);
+    set_global_var(GVAR_KARMA_BETRAYER, 0);
+    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+    set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+    set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+    if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
     end
     else begin
-        if (global_var(0) >= 750) then begin
-            set_global_var(38, 1);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+            set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
         end
         else begin
-            if (global_var(0) >= 500) then begin
-                set_global_var(39, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
             end
             else begin
-                if (global_var(0) >= 250) then begin
-                    set_global_var(40, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                    set_global_var(GVAR_KARMA_DEFENDER, 1);
                 end
                 else begin
-                    if (global_var(0) > -250) then begin
-                        set_global_var(41, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                        set_global_var(GVAR_KARMA_WANDERER, 1);
                     end
                     else begin
-                        if (global_var(0) > -500) then begin
-                            set_global_var(42, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                            set_global_var(GVAR_KARMA_BETRAYER, 1);
                         end
                         else begin
-                            if (global_var(0) > -750) then begin
-                                set_global_var(43, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                             end
                             else begin
-                                if (global_var(0) > -1000) then begin
-                                    set_global_var(44, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                    set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                 end
                                 else begin
-                                    set_global_var(45, 1);
+                                    set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                 end
                             end
                         end
@@ -314,13 +314,13 @@ end
 procedure look_at_p_proc
 begin
     script_overrides;
-    display_msg(message_str(1548, 100));
+    display_mstr(100);
 end
 
 procedure description_p_proc
 begin
     script_overrides;
-    display_msg(message_str(1548, 200));
+    display_mstr(200);
 end
 
 procedure use_skill_on_p_proc
@@ -338,9 +338,9 @@ end
 
 procedure map_enter_p_proc
 begin
-    critter_add_trait(self_obj, 1, 6, 248);
-    critter_add_trait(self_obj, 1, 5, 2);
-    if (global_var(841) == 1) then begin
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 248);
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_PEASANTS_NO_TALKING);
+    if (global_var(GVAR_C_STRIKE_IN_MINE) == 1) then begin
         set_obj_visibility(self_obj, 0);
     end
     else begin
@@ -350,22 +350,22 @@ end
 
 procedure timed_event_p_proc
 begin
-    if ((fixed_param == 1) and (global_var(841) == 1) and not(combat_is_initialized)) then begin
-        float_msg(self_obj, message_str(1548, random(306, 318)), 8);
+    if ((fixed_param == 1) and (global_var(GVAR_C_STRIKE_IN_MINE) == 1) and not(combat_is_initialized)) then begin
+        float_msg(self_obj, mstr(random(306, 318)), FLOAT_MSG_YELLOW);
     end
 end
 
 procedure combat_p_proc
 begin
-    if ((global_var(841) == 1) and (map_var(3) != 0)) then begin
-        critter_add_trait(self_obj, 1, 6, 244);
-        critter_add_trait(self_obj, 1, 5, 2);
+    if ((global_var(GVAR_C_STRIKE_IN_MINE) == 1) and (map_var(3) != 0)) then begin
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 244);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_PEASANTS_NO_TALKING);
     end
     if (fixed_param == 4) then begin
         GVar22 := random(0, 12);
-        if ((GVar22 >= 12) and (has_trait(1, self_obj, 666) == 1) and (map_var(3) != 0) and not(critter_is_fleeing(self_obj))) then begin
+        if ((GVar22 >= 12) and (obj_is_visible_flag(self_obj) == 1) and (map_var(3) != 0) and not(critter_is_fleeing(self_obj))) then begin
             if (obj_can_see_obj(self_obj, dude_obj)) then begin
-                float_msg(self_obj, message_str(1548, random(401, 408)), 2);
+                float_msg(self_obj, mstr(random(401, 408)), FLOAT_MSG_RED);
             end
         end
     end
@@ -374,55 +374,55 @@ end
 procedure Node998
 begin
     set_local_var(4, 1);
-    if (get_critter_stat(dude_obj, 34) == 0) then begin
-        float_msg(self_obj, message_str(1548, 300), 2);
-    end
-    if (get_critter_stat(dude_obj, 34) == 1) then begin
-        float_msg(self_obj, message_str(1548, 301), 2);
-    end
-    set_global_var(0, global_var(0) - 10);
-    set_global_var(37, 0);
-    set_global_var(38, 0);
-    set_global_var(39, 0);
-    set_global_var(40, 0);
-    set_global_var(41, 0);
-    set_global_var(42, 0);
-    set_global_var(43, 0);
-    set_global_var(44, 0);
-    set_global_var(45, 0);
-    if (global_var(0) >= 1000) then begin
-        set_global_var(37, 1);
+    if (dude_gender == 0) then begin
+        float_msg(self_obj, mstr(300), FLOAT_MSG_RED);
+    end
+    if (dude_gender == 1) then begin
+        float_msg(self_obj, mstr(301), FLOAT_MSG_RED);
+    end
+    set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) - 10);
+    set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+    set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+    set_global_var(GVAR_KARMA_DEFENDER, 0);
+    set_global_var(GVAR_KARMA_WANDERER, 0);
+    set_global_var(GVAR_KARMA_BETRAYER, 0);
+    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+    set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+    set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+    if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
     end
     else begin
-        if (global_var(0) >= 750) then begin
-            set_global_var(38, 1);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+            set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
         end
         else begin
-            if (global_var(0) >= 500) then begin
-                set_global_var(39, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
             end
             else begin
-                if (global_var(0) >= 250) then begin
-                    set_global_var(40, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                    set_global_var(GVAR_KARMA_DEFENDER, 1);
                 end
                 else begin
-                    if (global_var(0) > -250) then begin
-                        set_global_var(41, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                        set_global_var(GVAR_KARMA_WANDERER, 1);
                     end
                     else begin
-                        if (global_var(0) > -500) then begin
-                            set_global_var(42, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                            set_global_var(GVAR_KARMA_BETRAYER, 1);
                         end
                         else begin
-                            if (global_var(0) > -750) then begin
-                                set_global_var(43, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                             end
                             else begin
-                                if (global_var(0) > -1000) then begin
-                                    set_global_var(44, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                    set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                 end
                                 else begin
-                                    set_global_var(45, 1);
+                                    set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                 end
                             end
                         end
@@ -454,237 +454,237 @@ end
 
 procedure Node500
 begin
-    gsay_reply(1548, 500);
-    if ((get_critter_stat(dude_obj, 34) == 0) and ((global_var(835) == 1) or (global_var(831) == 1))) then begin
-        giq_option(4, 1548, 501, Node550, 50);
+    Reply(500);
+    if ((dude_gender == 0) and ((global_var(GVAR_C_QUEST_NESTOR_Q2) == 1) or (global_var(GVAR_C_QUEST_UNCLE_Q2) == 1))) then begin
+        giq_option(4, NAME, 501, Node550, NEUTRAL_REACTION);
     end
-    if ((get_critter_stat(dude_obj, 34) == 1) and ((global_var(835) == 1) or (global_var(831) == 1))) then begin
-        giq_option(4, 1548, 502, Node550, 50);
+    if ((dude_gender == 1) and ((global_var(GVAR_C_QUEST_NESTOR_Q2) == 1) or (global_var(GVAR_C_QUEST_UNCLE_Q2) == 1))) then begin
+        giq_option(4, NAME, 502, Node550, NEUTRAL_REACTION);
     end
-    giq_option(4, 1548, 503, Node600, 50);
-    giq_option(4, 1548, 504, Node1000, 50);
+    giq_option(4, NAME, 503, Node600, NEUTRAL_REACTION);
+    giq_option(4, NAME, 504, Node1000, NEUTRAL_REACTION);
 end
 
 procedure Node550
 begin
-    gsay_reply(1548, 550);
-    giq_option(4, 1548, 551, Node650, 50);
-    giq_option(4, 1548, 552, Node999, 50);
+    Reply(550);
+    giq_option(4, NAME, 551, Node650, NEUTRAL_REACTION);
+    giq_option(4, NAME, 552, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node600
 begin
-    gsay_reply(1548, 600);
-    giq_option(4, 1548, 601, Node650, 50);
-    giq_option(4, 1548, 602, Node999, 50);
+    Reply(600);
+    giq_option(4, NAME, 601, Node650, NEUTRAL_REACTION);
+    giq_option(4, NAME, 602, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node650
 begin
-    gsay_reply(1548, 650);
+    Reply(650);
     set_map_var(0, map_var(0) bwor 1);
     if ((map_var(0) bwand 2) == 0) then begin
-        giq_option(4, 1548, 651, Node700, 50);
+        giq_option(4, NAME, 651, Node700, NEUTRAL_REACTION);
     end
     if ((map_var(0) bwand 4) == 0) then begin
-        giq_option(4, 1548, 652, Node750, 50);
+        giq_option(4, NAME, 652, Node750, NEUTRAL_REACTION);
     end
     if ((map_var(0) bwand 8) == 0) then begin
-        giq_option(4, 1548, 653, Node800, 50);
+        giq_option(4, NAME, 653, Node800, NEUTRAL_REACTION);
     end
     if ((map_var(0) bwand 16) == 0) then begin
-        giq_option(4, 1548, 654, Node850, 50);
+        giq_option(4, NAME, 654, Node850, NEUTRAL_REACTION);
     end
-    giq_option(4, 1548, 655, Node1050, 50);
-    giq_option(4, 1548, 656, Node1350, 50);
-    giq_option(4, 1548, 657, Node999, 50);
+    giq_option(4, NAME, 655, Node1050, NEUTRAL_REACTION);
+    giq_option(4, NAME, 656, Node1350, NEUTRAL_REACTION);
+    giq_option(4, NAME, 657, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node700
 begin
-    gsay_reply(1548, 700);
+    Reply(700);
     set_map_var(0, map_var(0) bwor 2);
     if ((map_var(0) bwand 4) == 0) then begin
-        giq_option(4, 1548, 701, Node750, 50);
+        giq_option(4, NAME, 701, Node750, NEUTRAL_REACTION);
     end
     if ((map_var(0) bwand 8) == 0) then begin
-        giq_option(4, 1548, 702, Node800, 50);
+        giq_option(4, NAME, 702, Node800, NEUTRAL_REACTION);
     end
     if ((map_var(0) bwand 16) == 0) then begin
-        giq_option(4, 1548, 703, Node850, 50);
+        giq_option(4, NAME, 703, Node850, NEUTRAL_REACTION);
     end
-    giq_option(4, 1548, 704, Node1050, 50);
-    giq_option(4, 1548, 705, Node1350, 50);
-    giq_option(4, 1548, 706, Node999, 50);
+    giq_option(4, NAME, 704, Node1050, NEUTRAL_REACTION);
+    giq_option(4, NAME, 705, Node1350, NEUTRAL_REACTION);
+    giq_option(4, NAME, 706, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node750
 begin
-    gsay_reply(1548, 750);
+    Reply(750);
     set_map_var(0, map_var(0) bwor 4);
     if ((map_var(0) bwand 2) == 0) then begin
-        giq_option(4, 1548, 751, Node700, 50);
+        giq_option(4, NAME, 751, Node700, NEUTRAL_REACTION);
     end
     if ((map_var(0) bwand 8) == 0) then begin
-        giq_option(4, 1548, 752, Node800, 50);
+        giq_option(4, NAME, 752, Node800, NEUTRAL_REACTION);
     end
     if ((map_var(0) bwand 16) == 0) then begin
-        giq_option(4, 1548, 753, Node850, 50);
+        giq_option(4, NAME, 753, Node850, NEUTRAL_REACTION);
     end
-    giq_option(4, 1548, 754, Node1050, 50);
-    giq_option(4, 1548, 755, Node1350, 50);
-    giq_option(4, 1548, 756, Node999, 50);
+    giq_option(4, NAME, 754, Node1050, NEUTRAL_REACTION);
+    giq_option(4, NAME, 755, Node1350, NEUTRAL_REACTION);
+    giq_option(4, NAME, 756, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node800
 begin
-    gsay_reply(1548, 800);
+    Reply(800);
     set_map_var(0, map_var(0) bwor 8);
-    giq_option(4, 1548, 801, Node998, 50);
+    giq_option(4, NAME, 801, Node998, NEUTRAL_REACTION);
     if ((map_var(0) bwand 2) == 0) then begin
-        giq_option(4, 1548, 802, Node700, 50);
+        giq_option(4, NAME, 802, Node700, NEUTRAL_REACTION);
     end
     if ((map_var(0) bwand 4) == 0) then begin
-        giq_option(4, 1548, 803, Node750, 50);
+        giq_option(4, NAME, 803, Node750, NEUTRAL_REACTION);
     end
     if ((map_var(0) bwand 16) == 0) then begin
-        giq_option(4, 1548, 804, Node850, 50);
+        giq_option(4, NAME, 804, Node850, NEUTRAL_REACTION);
     end
-    giq_option(4, 1548, 805, Node1050, 50);
-    giq_option(4, 1548, 806, Node1350, 50);
-    giq_option(4, 1548, 807, Node999, 50);
+    giq_option(4, NAME, 805, Node1050, NEUTRAL_REACTION);
+    giq_option(4, NAME, 806, Node1350, NEUTRAL_REACTION);
+    giq_option(4, NAME, 807, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node850
 begin
-    gsay_reply(1548, 850);
+    Reply(850);
     set_map_var(0, map_var(0) bwor 16);
-    giq_option(4, 1548, 851, Node900, 50);
+    giq_option(4, NAME, 851, Node900, NEUTRAL_REACTION);
     if ((map_var(0) bwand 2) == 0) then begin
-        giq_option(4, 1548, 852, Node700, 50);
+        giq_option(4, NAME, 852, Node700, NEUTRAL_REACTION);
     end
     if ((map_var(0) bwand 4) == 0) then begin
-        giq_option(4, 1548, 853, Node750, 50);
+        giq_option(4, NAME, 853, Node750, NEUTRAL_REACTION);
     end
     if ((map_var(0) bwand 8) == 0) then begin
-        giq_option(4, 1548, 854, Node800, 50);
+        giq_option(4, NAME, 854, Node800, NEUTRAL_REACTION);
     end
-    giq_option(4, 1548, 855, Node1050, 50);
-    giq_option(4, 1548, 856, Node1350, 50);
-    giq_option(4, 1548, 857, Node999, 50);
+    giq_option(4, NAME, 855, Node1050, NEUTRAL_REACTION);
+    giq_option(4, NAME, 856, Node1350, NEUTRAL_REACTION);
+    giq_option(4, NAME, 857, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node900
 begin
-    gsay_reply(1548, 900);
-    giq_option(4, 1548, 901, Node950, 50);
+    Reply(900);
+    giq_option(4, NAME, 901, Node950, NEUTRAL_REACTION);
     if ((map_var(0) bwand 2) == 0) then begin
-        giq_option(4, 1548, 902, Node700, 50);
+        giq_option(4, NAME, 902, Node700, NEUTRAL_REACTION);
     end
     if ((map_var(0) bwand 4) == 0) then begin
-        giq_option(4, 1548, 903, Node750, 50);
+        giq_option(4, NAME, 903, Node750, NEUTRAL_REACTION);
     end
     if ((map_var(0) bwand 8) == 0) then begin
-        giq_option(4, 1548, 904, Node800, 50);
+        giq_option(4, NAME, 904, Node800, NEUTRAL_REACTION);
     end
-    giq_option(4, 1548, 905, Node1050, 50);
-    giq_option(4, 1548, 906, Node1350, 50);
-    giq_option(4, 1548, 907, Node999, 50);
+    giq_option(4, NAME, 905, Node1050, NEUTRAL_REACTION);
+    giq_option(4, NAME, 906, Node1350, NEUTRAL_REACTION);
+    giq_option(4, NAME, 907, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node950
 begin
-    gsay_reply(1548, 950);
+    Reply(950);
     set_map_var(0, map_var(0) bwor 32);
-    giq_option(4, 1548, 951, Node999, 50);
+    giq_option(4, NAME, 951, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node1000
 begin
-    gsay_reply(1548, 1000);
-    giq_option(4, 1548, 1001, Node600, 50);
-    giq_option(4, 1548, 1002, Node999, 50);
+    Reply(1000);
+    giq_option(4, NAME, 1001, Node600, NEUTRAL_REACTION);
+    giq_option(4, NAME, 1002, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node1050
 begin
-    gsay_reply(1548, 1050);
-    if (has_skill(dude_obj, 14) > 60) then begin
-        giq_option(4, 1548, 1051, Node1100, 50);
+    Reply(1050);
+    if (dude_skill(SKILL_SPEECH) > 60) then begin
+        giq_option(4, NAME, 1051, Node1100, NEUTRAL_REACTION);
     end
     if (item_caps_total(dude_obj) > 999) then begin
-        giq_option(4, 1548, 1052, Node1250, 50);
+        giq_option(4, NAME, 1052, Node1250, NEUTRAL_REACTION);
     end
-    giq_option(4, 1548, 1053, Node998, 50);
-    giq_option(4, 1548, 1054, Node999, 50);
+    giq_option(4, NAME, 1053, Node998, NEUTRAL_REACTION);
+    giq_option(4, NAME, 1054, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node1100
 begin
-    gsay_reply(1548, 1100);
-    if (has_skill(dude_obj, 14) > 70) then begin
-        giq_option(4, 1548, 1101, Node1150, 50);
+    Reply(1100);
+    if (dude_skill(SKILL_SPEECH) > 70) then begin
+        giq_option(4, NAME, 1101, Node1150, NEUTRAL_REACTION);
     end
     if (item_caps_total(dude_obj) > 999) then begin
-        giq_option(4, 1548, 1102, Node1250, 50);
+        giq_option(4, NAME, 1102, Node1250, NEUTRAL_REACTION);
     end
-    giq_option(4, 1548, 1103, Node998, 50);
-    giq_option(4, 1548, 1104, Node999, 50);
+    giq_option(4, NAME, 1103, Node998, NEUTRAL_REACTION);
+    giq_option(4, NAME, 1104, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node1150
 begin
-    gsay_reply(1548, 1150);
-    giq_option(4, 1548, 1151, Node1200, 50);
+    Reply(1150);
+    giq_option(4, NAME, 1151, Node1200, NEUTRAL_REACTION);
 end
 
 procedure Node1200
 begin
-    gsay_reply(1548, 1200);
-    set_global_var(0, global_var(0) + 5);
+    Reply(1200);
+    set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + 5);
     debug_msg("Player gains " + 5 + " Karma Points.");
-    set_global_var(37, 0);
-    set_global_var(38, 0);
-    set_global_var(39, 0);
-    set_global_var(40, 0);
-    set_global_var(41, 0);
-    set_global_var(42, 0);
-    set_global_var(43, 0);
-    set_global_var(44, 0);
-    set_global_var(45, 0);
-    if (global_var(0) >= 1000) then begin
-        set_global_var(37, 1);
+    set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+    set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+    set_global_var(GVAR_KARMA_DEFENDER, 0);
+    set_global_var(GVAR_KARMA_WANDERER, 0);
+    set_global_var(GVAR_KARMA_BETRAYER, 0);
+    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+    set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+    set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+    if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
     end
     else begin
-        if (global_var(0) >= 750) then begin
-            set_global_var(38, 1);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+            set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
         end
         else begin
-            if (global_var(0) >= 500) then begin
-                set_global_var(39, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
             end
             else begin
-                if (global_var(0) >= 250) then begin
-                    set_global_var(40, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                    set_global_var(GVAR_KARMA_DEFENDER, 1);
                 end
                 else begin
-                    if (global_var(0) > -250) then begin
-                        set_global_var(41, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                        set_global_var(GVAR_KARMA_WANDERER, 1);
                     end
                     else begin
-                        if (global_var(0) > -500) then begin
-                            set_global_var(42, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                            set_global_var(GVAR_KARMA_BETRAYER, 1);
                         end
                         else begin
-                            if (global_var(0) > -750) then begin
-                                set_global_var(43, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                             end
                             else begin
-                                if (global_var(0) > -1000) then begin
-                                    set_global_var(44, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                    set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                 end
                                 else begin
-                                    set_global_var(45, 1);
+                                    set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                 end
                             end
                         end
@@ -693,79 +693,78 @@ begin
             end
         end
     end
-    giq_option(4, 1548, 1201, Strike_End, 50);
+    giq_option(4, NAME, 1201, Strike_End, NEUTRAL_REACTION);
 end
 
 procedure Node1250
 begin
-    gsay_reply(1548, 1250);
-    giq_option(4, 1548, 1251, Node1300, 50);
-    if (get_critter_stat(dude_obj, 34) == 0) then begin
-        giq_option(4, 1548, 1252, Node998, 50);
+    Reply(1250);
+    giq_option(4, NAME, 1251, Node1300, NEUTRAL_REACTION);
+    if (dude_gender == 0) then begin
+        giq_option(4, NAME, 1252, Node998, NEUTRAL_REACTION);
     end
-    if (get_critter_stat(dude_obj, 34) == 1) then begin
-        giq_option(4, 1548, 1253, Node998, 50);
+    if (dude_gender == 1) then begin
+        giq_option(4, NAME, 1253, Node998, NEUTRAL_REACTION);
     end
-    giq_option(4, 1548, 1254, Node999, 50);
+    giq_option(4, NAME, 1254, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node1300
 begin
-    gsay_reply(1548, 1300);
+    Reply(1300);
     item_caps_adjust(dude_obj, -1000);
-    giq_option(4, 1548, 1301, Strike_End, 50);
+    giq_option(4, NAME, 1301, Strike_End, NEUTRAL_REACTION);
 end
 
 procedure Node1350
 begin
-    gsay_reply(1548, 1350);
-    if (has_skill(dude_obj, 14) > 70) then begin
-        giq_option(4, 1548, 1351, Node1400, 50);
+    Reply(1350);
+    if (dude_skill(SKILL_SPEECH) > 70) then begin
+        giq_option(4, NAME, 1351, Node1400, NEUTRAL_REACTION);
     end
     if (item_caps_total(dude_obj) > 999) then begin
-        giq_option(4, 1548, 1352, Node1450, 50);
+        giq_option(4, NAME, 1352, Node1450, NEUTRAL_REACTION);
     end
-    giq_option(4, 1548, 1353, Node999, 50);
+    giq_option(4, NAME, 1353, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node1400
 begin
-    gsay_reply(1548, 1400);
-    giq_option(4, 1548, 1401, Rebelion_Start, 50);
+    Reply(1400);
+    giq_option(4, NAME, 1401, Rebelion_Start, NEUTRAL_REACTION);
 end
 
 procedure Node1450
 begin
-    gsay_reply(1548, 1450);
-    giq_option(4, 1548, 1451, Node1500, 50);
-    giq_option(4, 1548, 1452, Node999, 50);
+    Reply(1450);
+    giq_option(4, NAME, 1451, Node1500, NEUTRAL_REACTION);
+    giq_option(4, NAME, 1452, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node1500
 begin
-    gsay_reply(1548, 1500);
+    Reply(1500);
     item_caps_adjust(dude_obj, -1000);
-    giq_option(4, 1548, 1501, Rebelion_Start, 50);
+    giq_option(4, NAME, 1501, Rebelion_Start, NEUTRAL_REACTION);
 end
 
 procedure Node1550
 begin
-    gsay_reply(1548, 1550);
-    giq_option(4, 1548, 1551, Node650, 50);
-    giq_option(4, 1548, 1552, Node1000, 50);
+    Reply(1550);
+    giq_option(4, NAME, 1551, Node650, NEUTRAL_REACTION);
+    giq_option(4, NAME, 1552, Node1000, NEUTRAL_REACTION);
 end
 
 procedure Strike_End
 begin
     game_time_advance(30 * (60 * 10));
-    set_global_var(841, 2);
+    set_global_var(GVAR_C_STRIKE_IN_MINE, 2);
     call Node999();
 end
 
 procedure Rebelion_Start
 begin
-    set_global_var(841, 3);
+    set_global_var(GVAR_C_STRIKE_IN_MINE, 3);
     set_map_var(1, 1);
     call Node999();
 end
-
