diff --git a/Mods/Resurrection/Original/CORATHNX.ssl b/Mods/Resurrection/ReDefine/CORATHNX.ssl
index 349bff3..c9ff829 100644
--- a/Mods/Resurrection/Original/CORATHNX.ssl
+++ b/Mods/Resurrection/ReDefine/CORATHNX.ssl
@@ -54,12 +54,12 @@ begin
             return 1;
         end
     end
-    if ((party_member_obj(16777779) != 0) and ((global_var(910) bwand 32) == 0)) then begin
+    if ((party_member_obj(16777779) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 32) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777779)) <= 5) then begin
             return 1;
         end
     end
-    if ((party_member_obj(16777811) != 0) and ((global_var(910) bwand 64) == 0)) then begin
+    if ((party_member_obj(16777811) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 64) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777811)) <= 5) then begin
             return 1;
         end
@@ -74,7 +74,7 @@ begin
             return 1;
         end
     end
-    if ((party_member_obj(16777814) != 0) and ((global_var(910) bwand 128) == 0)) then begin
+    if ((party_member_obj(16777814) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 128) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777814)) <= 5) then begin
             return 1;
         end
@@ -131,10 +131,10 @@ end
 
 procedure map_enter_p_proc
 begin
-    if (not(metarule(22, 0))) then begin
-        if ((global_var(843) bwand 2097152) != 0) then begin
-            gfade_in(1);
-            set_global_var(843, global_var(843) bwand (-1 - 2097152));
+    if (not(is_loading_game)) then begin
+        if ((global_var(GVAR_C_RING_VICTORIES) bwand 2097152) != 0) then begin
+            fade_in;
+            set_global_var(GVAR_C_RING_VICTORIES, global_var(GVAR_C_RING_VICTORIES) bwand (-1 - 2097152));
         end
         game_time_advance(6 * (60 * (60 * 10)));
         add_timer_event(self_obj, game_ticks(0), 10);
@@ -151,7 +151,7 @@ begin
         police2 := tile_contains_pid_obj(21911, 0, 16777894);
         police3 := tile_contains_pid_obj(21508, 0, 16777895);
         police4 := tile_contains_pid_obj(21501, 0, 16777895);
-        if ((global_var(830) == 4) or ((global_var(842) bwand 1024) != 0) and ((global_var(831) == 3) or (global_var(841) == 3))) then begin
+        if ((global_var(GVAR_C_QUEST_UNCLE_Q1) == 4) or ((global_var(GVAR_C_CORATH_STATUS) bwand 1024) != 0) and ((global_var(GVAR_C_QUEST_UNCLE_Q2) == 3) or (global_var(GVAR_C_STRIKE_IN_MINE) == 3))) then begin
             destroy_object(neszol1);
             destroy_object(neszol6);
             destroy_object(neszol7);
@@ -327,103 +327,103 @@ begin
         end
         if (sequence == 1) then begin
             game_ui_disable;
-            display_msg(message_str(1584, 99));
+            display_mstr(99);
             move_to(dude_obj, 20914, elevation(self_obj));
             anim(dude_obj, 1000, 0);
             if (party_member_obj(16777729) != 0) then begin
                 if (party_member_obj(16777729) != 0) then begin
-                    if (has_trait(1, party_member_obj(16777729), 666)) then begin
-                        critter_attempt_placement(party_member_obj(16777729), tile_num_in_direction(tile_num(dude_obj), (has_trait(1, dude_obj, 10) + 3) % 6, 2), elevation(dude_obj));
+                    if (obj_is_visible_flag(party_member_obj(16777729))) then begin
+                        critter_attempt_placement(party_member_obj(16777729), tile_num_in_direction(dude_tile, (obj_get_rot(dude_obj) + 3) % 6, 2), dude_elevation);
                     end
                 end
             end
-            if ((party_member_obj(16777779) != 0) and ((global_var(910) bwand 32) == 0)) then begin
+            if ((party_member_obj(16777779) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 32) == 0)) then begin
                 if (party_member_obj(16777779) != 0) then begin
-                    if (has_trait(1, party_member_obj(16777779), 666)) then begin
-                        critter_attempt_placement(party_member_obj(16777779), tile_num_in_direction(tile_num(dude_obj), (has_trait(1, dude_obj, 10) + 3) % 6, 2), elevation(dude_obj));
+                    if (obj_is_visible_flag(party_member_obj(16777779))) then begin
+                        critter_attempt_placement(party_member_obj(16777779), tile_num_in_direction(dude_tile, (obj_get_rot(dude_obj) + 3) % 6, 2), dude_elevation);
                     end
                 end
             end
-            if ((party_member_obj(16777811) != 0) and ((global_var(910) bwand 64) == 0)) then begin
+            if ((party_member_obj(16777811) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 64) == 0)) then begin
                 if (party_member_obj(16777811) != 0) then begin
-                    if (has_trait(1, party_member_obj(16777811), 666)) then begin
-                        critter_attempt_placement(party_member_obj(16777811), tile_num_in_direction(tile_num(dude_obj), (has_trait(1, dude_obj, 10) + 3) % 6, 2), elevation(dude_obj));
+                    if (obj_is_visible_flag(party_member_obj(16777811))) then begin
+                        critter_attempt_placement(party_member_obj(16777811), tile_num_in_direction(dude_tile, (obj_get_rot(dude_obj) + 3) % 6, 2), dude_elevation);
                     end
                 end
             end
             if (party_member_obj(16777812) != 0) then begin
                 if (party_member_obj(16777812) != 0) then begin
-                    if (has_trait(1, party_member_obj(16777812), 666)) then begin
-                        critter_attempt_placement(party_member_obj(16777812), tile_num_in_direction(tile_num(dude_obj), (has_trait(1, dude_obj, 10) + 3) % 6, 2), elevation(dude_obj));
+                    if (obj_is_visible_flag(party_member_obj(16777812))) then begin
+                        critter_attempt_placement(party_member_obj(16777812), tile_num_in_direction(dude_tile, (obj_get_rot(dude_obj) + 3) % 6, 2), dude_elevation);
                     end
                 end
             end
-            if ((party_member_obj(16777814) != 0) and ((global_var(910) bwand 128) == 0)) then begin
+            if ((party_member_obj(16777814) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 128) == 0)) then begin
                 if (party_member_obj(16777814) != 0) then begin
-                    if (has_trait(1, party_member_obj(16777814), 666)) then begin
-                        critter_attempt_placement(party_member_obj(16777814), tile_num_in_direction(tile_num(dude_obj), (has_trait(1, dude_obj, 10) + 3) % 6, 2), elevation(dude_obj));
+                    if (obj_is_visible_flag(party_member_obj(16777814))) then begin
+                        critter_attempt_placement(party_member_obj(16777814), tile_num_in_direction(dude_tile, (obj_get_rot(dude_obj) + 3) % 6, 2), dude_elevation);
                     end
                 end
             end
             if (party_member_obj(16777748) != 0) then begin
                 if (party_member_obj(16777748) != 0) then begin
-                    if (has_trait(1, party_member_obj(16777748), 666)) then begin
-                        critter_attempt_placement(party_member_obj(16777748), tile_num_in_direction(tile_num(dude_obj), (has_trait(1, dude_obj, 10) + 3) % 6, 2), elevation(dude_obj));
+                    if (obj_is_visible_flag(party_member_obj(16777748))) then begin
+                        critter_attempt_placement(party_member_obj(16777748), tile_num_in_direction(dude_tile, (obj_get_rot(dude_obj) + 3) % 6, 2), dude_elevation);
                     end
                 end
             end
             if (party_member_obj(16778017) != 0) then begin
                 if (party_member_obj(16778017) != 0) then begin
-                    if (has_trait(1, party_member_obj(16778017), 666)) then begin
-                        critter_attempt_placement(party_member_obj(16778017), tile_num_in_direction(tile_num(dude_obj), (has_trait(1, dude_obj, 10) + 3) % 6, 2), elevation(dude_obj));
+                    if (obj_is_visible_flag(party_member_obj(16778017))) then begin
+                        critter_attempt_placement(party_member_obj(16778017), tile_num_in_direction(dude_tile, (obj_get_rot(dude_obj) + 3) % 6, 2), dude_elevation);
                     end
                 end
             end
             if (party_member_obj(16778016) != 0) then begin
                 if (party_member_obj(16778016) != 0) then begin
-                    if (has_trait(1, party_member_obj(16778016), 666)) then begin
-                        critter_attempt_placement(party_member_obj(16778016), tile_num_in_direction(tile_num(dude_obj), (has_trait(1, dude_obj, 10) + 3) % 6, 2), elevation(dude_obj));
+                    if (obj_is_visible_flag(party_member_obj(16778016))) then begin
+                        critter_attempt_placement(party_member_obj(16778016), tile_num_in_direction(dude_tile, (obj_get_rot(dude_obj) + 3) % 6, 2), dude_elevation);
                     end
                 end
             end
             if (party_member_obj(16778049) != 0) then begin
                 if (party_member_obj(16778049) != 0) then begin
-                    if (has_trait(1, party_member_obj(16778049), 666)) then begin
-                        critter_attempt_placement(party_member_obj(16778049), tile_num_in_direction(tile_num(dude_obj), (has_trait(1, dude_obj, 10) + 3) % 6, 2), elevation(dude_obj));
+                    if (obj_is_visible_flag(party_member_obj(16778049))) then begin
+                        critter_attempt_placement(party_member_obj(16778049), tile_num_in_direction(dude_tile, (obj_get_rot(dude_obj) + 3) % 6, 2), dude_elevation);
                     end
                 end
             end
             if (party_member_obj(16778050) != 0) then begin
                 if (party_member_obj(16778050) != 0) then begin
-                    if (has_trait(1, party_member_obj(16778050), 666)) then begin
-                        critter_attempt_placement(party_member_obj(16778050), tile_num_in_direction(tile_num(dude_obj), (has_trait(1, dude_obj, 10) + 3) % 6, 2), elevation(dude_obj));
+                    if (obj_is_visible_flag(party_member_obj(16778050))) then begin
+                        critter_attempt_placement(party_member_obj(16778050), tile_num_in_direction(dude_tile, (obj_get_rot(dude_obj) + 3) % 6, 2), dude_elevation);
                     end
                 end
             end
             if (party_member_obj(16778051) != 0) then begin
                 if (party_member_obj(16778051) != 0) then begin
-                    if (has_trait(1, party_member_obj(16778051), 666)) then begin
-                        critter_attempt_placement(party_member_obj(16778051), tile_num_in_direction(tile_num(dude_obj), (has_trait(1, dude_obj, 10) + 3) % 6, 2), elevation(dude_obj));
+                    if (obj_is_visible_flag(party_member_obj(16778051))) then begin
+                        critter_attempt_placement(party_member_obj(16778051), tile_num_in_direction(dude_tile, (obj_get_rot(dude_obj) + 3) % 6, 2), dude_elevation);
                     end
                 end
             end
             if (party_member_obj(16777294) != 0) then begin
                 if (party_member_obj(16777294) != 0) then begin
-                    if (has_trait(1, party_member_obj(16777294), 666)) then begin
-                        critter_attempt_placement(party_member_obj(16777294), tile_num_in_direction(tile_num(dude_obj), (has_trait(1, dude_obj, 10) + 3) % 6, 2), elevation(dude_obj));
+                    if (obj_is_visible_flag(party_member_obj(16777294))) then begin
+                        critter_attempt_placement(party_member_obj(16777294), tile_num_in_direction(dude_tile, (obj_get_rot(dude_obj) + 3) % 6, 2), dude_elevation);
                     end
                 end
             end
             if (party_member_obj(16777729) != 0) then begin
                 anim(party_member_obj(16777729), 1000, 0);
             end
-            if ((party_member_obj(16777779) != 0) and ((global_var(910) bwand 32) == 0)) then begin
+            if ((party_member_obj(16777779) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 32) == 0)) then begin
                 anim(party_member_obj(16777779), 1000, 0);
             end
-            if ((party_member_obj(16777811) != 0) and ((global_var(910) bwand 64) == 0)) then begin
+            if ((party_member_obj(16777811) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 64) == 0)) then begin
                 anim(party_member_obj(16777811), 1000, 0);
             end
-            if ((party_member_obj(16777814) != 0) and ((global_var(910) bwand 128) == 0)) then begin
+            if ((party_member_obj(16777814) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 128) == 0)) then begin
                 anim(party_member_obj(16777814), 1000, 0);
             end
             if (party_member_obj(16778017) != 0) then begin
@@ -444,31 +444,31 @@ begin
             if (party_member_obj(16777812) != 0) then begin
                 anim(party_member_obj(16777812), 1000, 0);
             end
-            metarule3(108, 19905, 0, 0);
+            tile_set_center(19905);
             add_timer_event(self_obj, game_ticks(3), 10);
         end
         if (sequence == 2) then begin
-            float_msg(francis, message_str(1584, 100), 4);
+            float_msg(francis, mstr(100), FLOAT_MSG_BLUE);
             add_timer_event(self_obj, game_ticks(4), 10);
         end
         if (sequence == 3) then begin
-            float_msg(francis, message_str(1584, 101), 4);
+            float_msg(francis, mstr(101), FLOAT_MSG_BLUE);
             add_timer_event(self_obj, game_ticks(4), 10);
         end
         if (sequence == 4) then begin
-            float_msg(nestor, message_str(1584, 200), 8);
+            float_msg(nestor, mstr(200), FLOAT_MSG_YELLOW);
             add_timer_event(self_obj, game_ticks(7), 10);
         end
         if (sequence == 5) then begin
-            float_msg(francis, message_str(1584, 300), 4);
+            float_msg(francis, mstr(300), FLOAT_MSG_BLUE);
             add_timer_event(self_obj, game_ticks(7), 10);
         end
         if (sequence == 6) then begin
-            float_msg(nestor, message_str(1584, 400), 8);
+            float_msg(nestor, mstr(400), FLOAT_MSG_YELLOW);
             add_timer_event(self_obj, game_ticks(7), 10);
         end
         if (sequence == 7) then begin
-            float_msg(francis, message_str(1584, 500), 4);
+            float_msg(francis, mstr(500), FLOAT_MSG_BLUE);
             add_timer_event(self_obj, game_ticks(8), 10);
         end
         if (sequence == 8) then begin
@@ -476,7 +476,7 @@ begin
             add_timer_event(self_obj, game_ticks(11), 10);
         end
         if (sequence == 9) then begin
-            if ((global_var(830) == 4) or ((global_var(842) bwand 1024) != 0) and ((global_var(831) == 3) or (global_var(841) == 3))) then begin
+            if ((global_var(GVAR_C_QUEST_UNCLE_Q1) == 4) or ((global_var(GVAR_C_CORATH_STATUS) bwand 1024) != 0) and ((global_var(GVAR_C_QUEST_UNCLE_Q2) == 3) or (global_var(GVAR_C_STRIKE_IN_MINE) == 3))) then begin
                 animate_move_obj_to_tile(neszol2, 15703, 0);
                 animate_move_obj_to_tile(neszol3, 15702, 0);
                 animate_move_obj_to_tile(neszol4, 16709, 0);
@@ -516,15 +516,15 @@ begin
             destroy_object(neszol3);
             destroy_object(neszol4);
             destroy_object(neszol5);
-            if ((global_var(830) == 4) or ((global_var(842) bwand 1024) != 0) and ((global_var(831) == 3) or (global_var(841) == 3))) then begin
+            if ((global_var(GVAR_C_QUEST_UNCLE_Q1) == 4) or ((global_var(GVAR_C_CORATH_STATUS) bwand 1024) != 0) and ((global_var(GVAR_C_QUEST_UNCLE_Q2) == 3) or (global_var(GVAR_C_STRIKE_IN_MINE) == 3))) then begin
             end
             else begin
                 destroy_object(neszol1);
                 destroy_object(neszol6);
                 destroy_object(neszol7);
             end
-            set_global_var(848, 1);
-            set_global_var(1021, game_time);
+            set_global_var(GVAR_NESTOR_LEAVED_CORATH, 1);
+            set_global_var(GVAR_NESTOR_LEAVED_CORATH_TIME, game_time);
             add_timer_event(self_obj, game_ticks(1), 10);
         end
         if (sequence == 13) then begin
@@ -536,4 +536,3 @@ begin
         end
     end
 end
-
