diff --git a/Mods/Resurrection/Original/ECCARTER.ssl b/Mods/Resurrection/ReDefine/ECCARTER.ssl
index 390a472..5026920 100644
--- a/Mods/Resurrection/Original/ECCARTER.ssl
+++ b/Mods/Resurrection/ReDefine/ECCARTER.ssl
@@ -117,12 +117,12 @@ begin
             return 1;
         end
     end
-    if ((party_member_obj(16777779) != 0) and ((global_var(910) bwand 32) == 0)) then begin
+    if ((party_member_obj(16777779) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 32) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777779)) <= 5) then begin
             return 1;
         end
     end
-    if ((party_member_obj(16777811) != 0) and ((global_var(910) bwand 64) == 0)) then begin
+    if ((party_member_obj(16777811) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 64) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777811)) <= 5) then begin
             return 1;
         end
@@ -137,7 +137,7 @@ begin
             return 1;
         end
     end
-    if ((party_member_obj(16777814) != 0) and ((global_var(910) bwand 128) == 0)) then begin
+    if ((party_member_obj(16777814) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 128) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777814)) <= 5) then begin
             return 1;
         end
@@ -196,7 +196,7 @@ end
 procedure critter_p_proc
 begin
     if ((map_var(1) == 1) and obj_can_see_obj(self_obj, dude_obj)) then begin
-        attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+        attack(dude_obj);
     end
 end
 
@@ -209,8 +209,8 @@ end
 
 procedure talk_p_proc
 begin
-    anim(dude_obj, 1000, rotation_to_tile(tile_num(dude_obj), tile_num(self_obj)));
-    anim(self_obj, 1000, rotation_to_tile(tile_num(self_obj), tile_num(dude_obj)));
+    anim(dude_obj, 1000, rotation_to_tile(dude_tile, tile_num(self_obj)));
+    anim(self_obj, 1000, rotation_to_tile(tile_num(self_obj), dude_tile));
     start_gdialog(1592, self_obj, 4, -1, -1);
     gsay_start;
     call Node500();
@@ -222,55 +222,55 @@ end
 procedure destroy_p_proc
 begin
     if (source_obj == dude_obj) then begin
-        set_global_var(5, global_var(5) + 1);
-        if (metarule(51, self_obj) == 2) then begin
+        set_global_var(GVAR_GOOD_MONSTER, global_var(GVAR_GOOD_MONSTER) + 1);
+        if (critter_kill_type(self_obj) == KILL_TYPE_children_kills) then begin
             if (source_obj == dude_obj) then begin
-                set_global_var(2, 0);
-                set_global_var(1, global_var(1) + 1);
-                set_global_var(0, global_var(0) + -15);
+                set_global_var(GVAR_CHAMPION_REPUTATION, 0);
+                set_global_var(GVAR_CHILDKILLER_REPUTATION, global_var(GVAR_CHILDKILLER_REPUTATION) + 1);
+                set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + -15);
                 debug_msg("Player gains " + -15 + " Karma Points.");
-                set_global_var(37, 0);
-                set_global_var(38, 0);
-                set_global_var(39, 0);
-                set_global_var(40, 0);
-                set_global_var(41, 0);
-                set_global_var(42, 0);
-                set_global_var(43, 0);
-                set_global_var(44, 0);
-                set_global_var(45, 0);
-                if (global_var(0) >= 1000) then begin
-                    set_global_var(37, 1);
+                set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+                set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+                set_global_var(GVAR_KARMA_DEFENDER, 0);
+                set_global_var(GVAR_KARMA_WANDERER, 0);
+                set_global_var(GVAR_KARMA_BETRAYER, 0);
+                set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+                set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+                set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+                    set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
                 end
                 else begin
-                    if (global_var(0) >= 750) then begin
-                        set_global_var(38, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
                     end
                     else begin
-                        if (global_var(0) >= 500) then begin
-                            set_global_var(39, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                            set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                         end
                         else begin
-                            if (global_var(0) >= 250) then begin
-                                set_global_var(40, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                                set_global_var(GVAR_KARMA_DEFENDER, 1);
                             end
                             else begin
-                                if (global_var(0) > -250) then begin
-                                    set_global_var(41, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                                    set_global_var(GVAR_KARMA_WANDERER, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -500) then begin
-                                        set_global_var(42, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                        set_global_var(GVAR_KARMA_BETRAYER, 1);
                                     end
                                     else begin
-                                        if (global_var(0) > -750) then begin
-                                            set_global_var(43, 1);
+                                        if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                            set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                         end
                                         else begin
-                                            if (global_var(0) > -1000) then begin
-                                                set_global_var(44, 1);
+                                            if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                                set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                             end
                                             else begin
-                                                set_global_var(45, 1);
+                                                set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                             end
                                         end
                                     end
@@ -281,58 +281,58 @@ begin
                 end
             end
         end
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(4) > (3 * global_var(5))) or (global_var(2) == 1)) and (global_var(1) == 0)) then begin
-            set_global_var(2, 1);
-            set_global_var(3, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(4) > (3 * global_var(GVAR_GOOD_MONSTER))) or (global_var(GVAR_CHAMPION_REPUTATION) == 1)) and (global_var(GVAR_CHILDKILLER_REPUTATION) == 0)) then begin
+            set_global_var(GVAR_CHAMPION_REPUTATION, 1);
+            set_global_var(GVAR_BERSERKER_REPUTATION, 0);
         end
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(5) > (2 * global_var(4))) or (global_var(3) == 1))) then begin
-            set_global_var(3, 1);
-            set_global_var(2, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
+            set_global_var(GVAR_BERSERKER_REPUTATION, 1);
+            set_global_var(GVAR_CHAMPION_REPUTATION, 0);
         end
-        set_global_var(0, global_var(0) + -10);
+        set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + -10);
         debug_msg("Player gains " + -10 + " Karma Points.");
-        set_global_var(37, 0);
-        set_global_var(38, 0);
-        set_global_var(39, 0);
-        set_global_var(40, 0);
-        set_global_var(41, 0);
-        set_global_var(42, 0);
-        set_global_var(43, 0);
-        set_global_var(44, 0);
-        set_global_var(45, 0);
-        if (global_var(0) >= 1000) then begin
-            set_global_var(37, 1);
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+        set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+        set_global_var(GVAR_KARMA_DEFENDER, 0);
+        set_global_var(GVAR_KARMA_WANDERER, 0);
+        set_global_var(GVAR_KARMA_BETRAYER, 0);
+        set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+        set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+            set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
         end
         else begin
-            if (global_var(0) >= 750) then begin
-                set_global_var(38, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
             end
             else begin
-                if (global_var(0) >= 500) then begin
-                    set_global_var(39, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                 end
                 else begin
-                    if (global_var(0) >= 250) then begin
-                        set_global_var(40, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                        set_global_var(GVAR_KARMA_DEFENDER, 1);
                     end
                     else begin
-                        if (global_var(0) > -250) then begin
-                            set_global_var(41, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                            set_global_var(GVAR_KARMA_WANDERER, 1);
                         end
                         else begin
-                            if (global_var(0) > -500) then begin
-                                set_global_var(42, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                set_global_var(GVAR_KARMA_BETRAYER, 1);
                             end
                             else begin
-                                if (global_var(0) > -750) then begin
-                                    set_global_var(43, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -1000) then begin
-                                        set_global_var(44, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                     end
                                     else begin
-                                        set_global_var(45, 1);
+                                        set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                     end
                                 end
                             end
@@ -348,23 +348,23 @@ procedure look_at_p_proc
 begin
     script_overrides;
     if (local_var(4) == 1) then begin
-        if (get_critter_stat(dude_obj, 4) < 4) then begin
-            display_msg(message_str(1592, 150));
+        if (dude_iq < 4) then begin
+            display_mstr(150);
         end
         else begin
-            display_msg(message_str(1592, 151));
+            display_mstr(151);
         end
     end
     else begin
-        if (get_critter_stat(dude_obj, 4) < 4) then begin
-            display_msg(message_str(1592, 100));
+        if (dude_iq < 4) then begin
+            display_mstr(100);
         end
         else begin
-            if (get_critter_stat(dude_obj, 4) < 8) then begin
-                display_msg(message_str(1592, 101 + (get_critter_stat(dude_obj, 34) == 1)));
+            if (dude_iq < 8) then begin
+                display_mstr(101 + (dude_gender == 1));
             end
             else begin
-                display_msg(message_str(1592, 103));
+                display_mstr(103);
             end
         end
     end
@@ -374,10 +374,10 @@ procedure description_p_proc
 begin
     script_overrides;
     if (local_var(4) == 1) then begin
-        display_msg(message_str(1592, 250));
+        display_mstr(250);
     end
     else begin
-        display_msg(message_str(1592, 200));
+        display_mstr(200);
     end
 end
 
@@ -394,9 +394,9 @@ end
 
 procedure map_enter_p_proc
 begin
-    critter_add_trait(self_obj, 1, 6, 285);
-    critter_add_trait(self_obj, 1, 5, 3);
-    metarule3(100, self_obj, 1, 0);
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 285);
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_TOUGH_GUARD_NO_TALKING);
+    rm_fixed_timer_event(self_obj, 1);
     add_timer_event(self_obj, game_ticks(random(18, 24)), 1);
 end
 
@@ -404,9 +404,9 @@ procedure timed_event_p_proc
 begin
     if (fixed_param == 1) then begin
         if ((map_var(2) == 0) and not(combat_is_initialized)) then begin
-            float_msg(self_obj, message_str(1592, random(300, 302)), 8);
+            float_msg(self_obj, mstr(random(300, 302)), FLOAT_MSG_YELLOW);
         end
-        metarule3(100, self_obj, 1, 0);
+        rm_fixed_timer_event(self_obj, 1);
         add_timer_event(self_obj, game_ticks(random(18, 24)), 1);
     end
 end
@@ -422,501 +422,500 @@ end
 
 procedure Node999_Hurt
 begin
-    display_msg(message_str(1592, 3900));
-    critter_dmg(dude_obj, 1, 0);
+    display_mstr(3900);
+    critter_damage(dude_obj, 1);
 end
 
 procedure Node500
 begin
-    gsay_reply(1592, 500);
-    giq_option(-3, 1592, 501, Node550, 50);
-    giq_option(4, 1592, 502, Node600, 50);
+    Reply(500);
+    giq_option(-3, NAME, 501, Node550, NEUTRAL_REACTION);
+    giq_option(4, NAME, 502, Node600, NEUTRAL_REACTION);
     if (local_var(5) == 1) then begin
-        giq_option(4, 1592, 503, Node650, 50);
+        giq_option(4, NAME, 503, Node650, NEUTRAL_REACTION);
     end
-    giq_option(4, 1592, 504, Node700, 50);
+    giq_option(4, NAME, 504, Node700, NEUTRAL_REACTION);
 end
 
 procedure Node550
 begin
-    gsay_reply(1592, 550);
-    giq_option(-3, 1592, 551, Node750, 50);
-    giq_option(-3, 1592, 552, Node800, 50);
+    Reply(550);
+    giq_option(-3, NAME, 551, Node750, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 552, Node800, NEUTRAL_REACTION);
 end
 
 procedure Node600
 begin
-    gsay_reply(1592, 600);
+    Reply(600);
     set_local_var(5, 1);
-    giq_option(4, 1592, 601 + (get_critter_stat(dude_obj, 34) == 1), Node700, 50);
-    giq_option(4, 1592, 603, Node999, 50);
-    giq_option(4, 1592, 604, Node850, 50);
-    if (get_critter_stat(dude_obj, 34) == 0) then begin
-        giq_option(4, 1592, 605, Node900, 50);
+    giq_option(4, NAME, 601 + (dude_gender == 1), Node700, NEUTRAL_REACTION);
+    giq_option(4, NAME, 603, Node999, NEUTRAL_REACTION);
+    giq_option(4, NAME, 604, Node850, NEUTRAL_REACTION);
+    if (dude_gender == 0) then begin
+        giq_option(4, NAME, 605, Node900, NEUTRAL_REACTION);
     end
 end
 
 procedure Node650
 begin
-    gsay_reply(1592, 650);
-    giq_option(4, 1592, 651, Node950, 50);
-    giq_option(6, 1592, 652, Node1000, 50);
-    giq_option(9, 1592, 653, Node1050, 50);
-    giq_option(4, 1592, 654, Node999, 50);
+    Reply(650);
+    giq_option(4, NAME, 651, Node950, NEUTRAL_REACTION);
+    giq_option(6, NAME, 652, Node1000, NEUTRAL_REACTION);
+    giq_option(9, NAME, 653, Node1050, NEUTRAL_REACTION);
+    giq_option(4, NAME, 654, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node700
 begin
-    gsay_reply(1592, 700);
-    giq_option(4, 1592, 701, Node1100, 50);
-    giq_option(4, 1592, 702, Node1150, 50);
-    giq_option(4, 1592, 703, Node1200, 50);
-    giq_option(4, 1592, 704, Node1300, 50);
-    giq_option(4, 1592, 705, Node1350, 50);
-    giq_option(4, 1592, 706, Node999, 50);
+    Reply(700);
+    giq_option(4, NAME, 701, Node1100, NEUTRAL_REACTION);
+    giq_option(4, NAME, 702, Node1150, NEUTRAL_REACTION);
+    giq_option(4, NAME, 703, Node1200, NEUTRAL_REACTION);
+    giq_option(4, NAME, 704, Node1300, NEUTRAL_REACTION);
+    giq_option(4, NAME, 705, Node1350, NEUTRAL_REACTION);
+    giq_option(4, NAME, 706, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node750
 begin
-    gsay_reply(1592, 750);
-    giq_option(-3, 1592, 751, Node1400, 50);
-    giq_option(-3, 1592, 752, Node1450, 50);
+    Reply(750);
+    giq_option(-3, NAME, 751, Node1400, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 752, Node1450, NEUTRAL_REACTION);
 end
 
 procedure Node800
 begin
-    gsay_reply(1592, 800);
-    giq_option(-3, 1592, 801, Node1500, 50);
-    giq_option(-3, 1592, 802, Node1550, 50);
+    Reply(800);
+    giq_option(-3, NAME, 801, Node1500, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 802, Node1550, NEUTRAL_REACTION);
 end
 
 procedure Node850
 begin
-    gsay_reply(1592, 850);
-    giq_option(4, 1592, 851, Node1000, 50);
-    giq_option(4, 1592, 852, Node700, 50);
-    giq_option(4, 1592, 853, Node999, 50);
+    Reply(850);
+    giq_option(4, NAME, 851, Node1000, NEUTRAL_REACTION);
+    giq_option(4, NAME, 852, Node700, NEUTRAL_REACTION);
+    giq_option(4, NAME, 853, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node900
 begin
-    gsay_reply(1592, 900);
-    giq_option(4, 1592, 901, Node1600, 50);
-    giq_option(4, 1592, 902, Node1650, 50);
+    Reply(900);
+    giq_option(4, NAME, 901, Node1600, NEUTRAL_REACTION);
+    giq_option(4, NAME, 902, Node1650, NEUTRAL_REACTION);
 end
 
 procedure Node950
 begin
-    gsay_reply(1592, 950);
-    giq_option(4, 1592, 951, Node700, 50);
-    giq_option(4, 1592, 952, Node999, 50);
+    Reply(950);
+    giq_option(4, NAME, 951, Node700, NEUTRAL_REACTION);
+    giq_option(4, NAME, 952, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node1000
 begin
-    gsay_reply(1592, 1000);
-    giq_option(4, 1592, 1001, Node700, 50);
-    giq_option(4, 1592, 1002, Node1700, 50);
+    Reply(1000);
+    giq_option(4, NAME, 1001, Node700, NEUTRAL_REACTION);
+    giq_option(4, NAME, 1002, Node1700, NEUTRAL_REACTION);
 end
 
 procedure Node1050
 begin
-    gsay_reply(1592, 1050);
-    giq_option(4, 1592, 1051, Node1750, 50);
-    giq_option(4, 1592, 1052, Node1800, 50);
-    giq_option(4, 1592, 1053, Node1850, 50);
+    Reply(1050);
+    giq_option(4, NAME, 1051, Node1750, NEUTRAL_REACTION);
+    giq_option(4, NAME, 1052, Node1800, NEUTRAL_REACTION);
+    giq_option(4, NAME, 1053, Node1850, NEUTRAL_REACTION);
 end
 
 procedure Node1100
 begin
-    gsay_reply(1592, 1100);
-    giq_option(4, 1592, 1101, Node1900, 50);
-    giq_option(4, 1592, 1102, Node700, 50);
+    Reply(1100);
+    giq_option(4, NAME, 1101, Node1900, NEUTRAL_REACTION);
+    giq_option(4, NAME, 1102, Node700, NEUTRAL_REACTION);
 end
 
 procedure Node1150
 begin
-    gsay_reply(1592, 1150);
-    giq_option(4, 1592, 1151, Node700, 50);
-    giq_option(4, 1592, 1152, Node1950, 50);
+    Reply(1150);
+    giq_option(4, NAME, 1151, Node700, NEUTRAL_REACTION);
+    giq_option(4, NAME, 1152, Node1950, NEUTRAL_REACTION);
 end
 
 procedure Node1200
 begin
-    gsay_reply(1592, 1200);
-    giq_option(4, 1592, 1201, Node1250, 50);
-    giq_option(4, 1592, 1202, Node700, 50);
-    giq_option(4, 1592, 1203, Node2000, 50);
+    Reply(1200);
+    giq_option(4, NAME, 1201, Node1250, NEUTRAL_REACTION);
+    giq_option(4, NAME, 1202, Node700, NEUTRAL_REACTION);
+    giq_option(4, NAME, 1203, Node2000, NEUTRAL_REACTION);
 end
 
 procedure Node1250
 begin
-    gsay_reply(1592, 1250);
-    giq_option(4, 1592, 1251, Node2050, 50);
-    giq_option(7, 1592, 1252, Node2100, 50);
-    giq_option(8, 1592, 1253, Node2150, 50);
-    giq_option(6, 1592, 1254, Node2200, 50);
-    giq_option(4, 1592, 1255, Node2250, 50);
-    giq_option(4, 1592, 1256, Node999, 50);
+    Reply(1250);
+    giq_option(4, NAME, 1251, Node2050, NEUTRAL_REACTION);
+    giq_option(7, NAME, 1252, Node2100, NEUTRAL_REACTION);
+    giq_option(8, NAME, 1253, Node2150, NEUTRAL_REACTION);
+    giq_option(6, NAME, 1254, Node2200, NEUTRAL_REACTION);
+    giq_option(4, NAME, 1255, Node2250, NEUTRAL_REACTION);
+    giq_option(4, NAME, 1256, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node1300
 begin
-    gsay_reply(1592, 1300);
-    giq_option(4, 1592, 1301, Node700, 50);
-    giq_option(4, 1592, 1302, Node2300, 50);
-    giq_option(8, 1592, 1303, Node2350, 50);
+    Reply(1300);
+    giq_option(4, NAME, 1301, Node700, NEUTRAL_REACTION);
+    giq_option(4, NAME, 1302, Node2300, NEUTRAL_REACTION);
+    giq_option(8, NAME, 1303, Node2350, NEUTRAL_REACTION);
 end
 
 procedure Node1350
 begin
-    gsay_reply(1592, 1350);
-    giq_option(4, 1592, 1351, Node700, 50);
-    giq_option(4, 1592, 1352, Node999, 50);
+    Reply(1350);
+    giq_option(4, NAME, 1351, Node700, NEUTRAL_REACTION);
+    giq_option(4, NAME, 1352, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node1400
 begin
-    gsay_reply(1592, 1400 + (get_critter_stat(dude_obj, 34) == 1));
-    giq_option(-3, 1592, 1402, Node999, 50);
+    Reply(1400 + (dude_gender == 1));
+    giq_option(-3, NAME, 1402, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node1450
 begin
-    gsay_reply(1592, 1450 + (get_critter_stat(dude_obj, 34) == 1));
-    giq_option(4, 1592, 1452, Node999, 50);
+    Reply(1450 + (dude_gender == 1));
+    giq_option(4, NAME, 1452, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node1500
 begin
-    gsay_reply(1592, 1500);
-    giq_option(-3, 1592, 1501, Node2400, 50);
-    giq_option(-3, 1592, 1502, Node2450, 50);
+    Reply(1500);
+    giq_option(-3, NAME, 1501, Node2400, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 1502, Node2450, NEUTRAL_REACTION);
 end
 
 procedure Node1550
 begin
-    gsay_reply(1592, 1550);
-    giq_option(-3, 1592, 1551, Node999, 50);
+    Reply(1550);
+    giq_option(-3, NAME, 1551, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node1600
 begin
-    gsay_reply(1592, 1600);
-    giq_option(4, 1592, 1601, Node999, 50);
+    Reply(1600);
+    giq_option(4, NAME, 1601, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node1650
 begin
-    gsay_reply(1592, 1650);
-    giq_option(4, 1592, 1651, Node700, 50);
-    giq_option(4, 1592, 1652, Node999, 50);
+    Reply(1650);
+    giq_option(4, NAME, 1651, Node700, NEUTRAL_REACTION);
+    giq_option(4, NAME, 1652, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node1700
 begin
-    gsay_reply(1592, 1700);
-    giq_option(4, 1592, 1701, Node2500, 50);
-    giq_option(4, 1592, 1702, Node2550, 50);
+    Reply(1700);
+    giq_option(4, NAME, 1701, Node2500, NEUTRAL_REACTION);
+    giq_option(4, NAME, 1702, Node2550, NEUTRAL_REACTION);
 end
 
 procedure Node1750
 begin
-    gsay_reply(1592, 1750);
-    giq_option(4, 1592, 1751, Node1050, 50);
-    giq_option(4, 1592, 1752, Node1700, 50);
+    Reply(1750);
+    giq_option(4, NAME, 1751, Node1050, NEUTRAL_REACTION);
+    giq_option(4, NAME, 1752, Node1700, NEUTRAL_REACTION);
 end
 
 procedure Node1800
 begin
-    gsay_reply(1592, 1800);
-    giq_option(4, 1592, 1801, Node1700, 50);
-    giq_option(4, 1592, 1802, Node2550, 50);
+    Reply(1800);
+    giq_option(4, NAME, 1801, Node1700, NEUTRAL_REACTION);
+    giq_option(4, NAME, 1802, Node2550, NEUTRAL_REACTION);
 end
 
 procedure Node1850
 begin
-    gsay_reply(1592, 1850);
-    giq_option(4, 1592, 1851, Node2600, 50);
-    giq_option(4, 1592, 1852, Node1700, 50);
+    Reply(1850);
+    giq_option(4, NAME, 1851, Node2600, NEUTRAL_REACTION);
+    giq_option(4, NAME, 1852, Node1700, NEUTRAL_REACTION);
 end
 
 procedure Node1900
 begin
-    gsay_reply(1592, 1900);
-    giq_option(4, 1592, 1901, Node700, 50);
-    giq_option(4, 1592, 1902, Node999, 50);
+    Reply(1900);
+    giq_option(4, NAME, 1901, Node700, NEUTRAL_REACTION);
+    giq_option(4, NAME, 1902, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node1950
 begin
-    gsay_reply(1592, 1950);
-    giq_option(4, 1592, 1951, Node2650, 50);
-    giq_option(4, 1592, 1952, Node2700, 50);
-    giq_option(4, 1592, 1953, Node700, 50);
+    Reply(1950);
+    giq_option(4, NAME, 1951, Node2650, NEUTRAL_REACTION);
+    giq_option(4, NAME, 1952, Node2700, NEUTRAL_REACTION);
+    giq_option(4, NAME, 1953, Node700, NEUTRAL_REACTION);
 end
 
 procedure Node2000
 begin
-    gsay_reply(1592, 2000);
-    giq_option(4, 1592, 2001, Node2300, 50);
-    giq_option(4, 1592, 2002, Node2750, 50);
+    Reply(2000);
+    giq_option(4, NAME, 2001, Node2300, NEUTRAL_REACTION);
+    giq_option(4, NAME, 2002, Node2750, NEUTRAL_REACTION);
 end
 
 procedure Node2050
 begin
-    gsay_reply(1592, 2050);
-    giq_option(4, 1592, 2051, Node1250, 50);
-    giq_option(4, 1592, 2052, Node700, 50);
+    Reply(2050);
+    giq_option(4, NAME, 2051, Node1250, NEUTRAL_REACTION);
+    giq_option(4, NAME, 2052, Node700, NEUTRAL_REACTION);
 end
 
 procedure Node2100
 begin
-    gsay_reply(1592, 2100);
-    giq_option(4, 1592, 2101, Node1250, 50);
-    giq_option(4, 1592, 2102, Node700, 50);
+    Reply(2100);
+    giq_option(4, NAME, 2101, Node1250, NEUTRAL_REACTION);
+    giq_option(4, NAME, 2102, Node700, NEUTRAL_REACTION);
 end
 
 procedure Node2150
 begin
-    gsay_reply(1592, 2150);
-    giq_option(4, 1592, 2151, Node1250, 50);
-    giq_option(4, 1592, 2152, Node700, 50);
+    Reply(2150);
+    giq_option(4, NAME, 2151, Node1250, NEUTRAL_REACTION);
+    giq_option(4, NAME, 2152, Node700, NEUTRAL_REACTION);
 end
 
 procedure Node2200
 begin
-    gsay_reply(1592, 2200);
-    giq_option(4, 1592, 2201, Node1250, 50);
-    giq_option(4, 1592, 2202, Node999, 50);
-    if (has_skill(dude_obj, 12) > 70) then begin
-        giq_option(8, 1592, 2203, Node2800, 50);
+    Reply(2200);
+    giq_option(4, NAME, 2201, Node1250, NEUTRAL_REACTION);
+    giq_option(4, NAME, 2202, Node999, NEUTRAL_REACTION);
+    if (dude_skill(SKILL_SCIENCE) > 70) then begin
+        giq_option(8, NAME, 2203, Node2800, NEUTRAL_REACTION);
     end
-    giq_option(4, 1592, 2204, Node2750, 50);
+    giq_option(4, NAME, 2204, Node2750, NEUTRAL_REACTION);
 end
 
 procedure Node2250
 begin
-    gsay_reply(1592, 2250);
-    giq_option(4, 1592, 2251, Node2850, 50);
-    giq_option(4, 1592, 2252, Node2900, 50);
+    Reply(2250);
+    giq_option(4, NAME, 2251, Node2850, NEUTRAL_REACTION);
+    giq_option(4, NAME, 2252, Node2900, NEUTRAL_REACTION);
 end
 
 procedure Node2300
 begin
-    gsay_reply(1592, 2300);
-    giq_option(4, 1592, 2301, Node700, 50);
-    giq_option(6, 1592, 2302, Node2950, 50);
-    giq_option(8, 1592, 2303, Node3000, 50);
+    Reply(2300);
+    giq_option(4, NAME, 2301, Node700, NEUTRAL_REACTION);
+    giq_option(6, NAME, 2302, Node2950, NEUTRAL_REACTION);
+    giq_option(8, NAME, 2303, Node3000, NEUTRAL_REACTION);
 end
 
 procedure Node2350
 begin
-    gsay_reply(1592, 2350);
-    giq_option(4, 1592, 2351, Node3050, 50);
-    giq_option(4, 1592, 2352, Node3100, 50);
+    Reply(2350);
+    giq_option(4, NAME, 2351, Node3050, NEUTRAL_REACTION);
+    giq_option(4, NAME, 2352, Node3100, NEUTRAL_REACTION);
 end
 
 procedure Node2400
 begin
-    gsay_reply(1592, 2400);
-    giq_option(-3, 1592, 2401, Node999, 50);
+    Reply(2400);
+    giq_option(-3, NAME, 2401, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node2450
 begin
-    gsay_reply(1592, 2450);
-    giq_option(-3, 1592, 2451, Node999, 50);
+    Reply(2450);
+    giq_option(-3, NAME, 2451, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node2500
 begin
-    gsay_reply(1592, 2500);
-    giq_option(4, 1592, 2501, Node700, 50);
-    giq_option(4, 1592, 2502, Node1250, 50);
-    giq_option(4, 1592, 2503, Node999, 50);
+    Reply(2500);
+    giq_option(4, NAME, 2501, Node700, NEUTRAL_REACTION);
+    giq_option(4, NAME, 2502, Node1250, NEUTRAL_REACTION);
+    giq_option(4, NAME, 2503, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node2550
 begin
-    gsay_reply(1592, 2550);
-    giq_option(4, 1592, 2551, Node3500, 50);
+    Reply(2550);
+    giq_option(4, NAME, 2551, Node3500, NEUTRAL_REACTION);
     if ((map_var(0) == 1) and (local_var(6) == 0)) then begin
-        giq_option(4, 1592, 2552, Node3200, 50);
+        giq_option(4, NAME, 2552, Node3200, NEUTRAL_REACTION);
     end
 end
 
 procedure Node2600
 begin
-    gsay_reply(1592, 2600);
-    giq_option(4, 1592, 2601, Node3250, 50);
-    giq_option(4, 1592, 2602, Node1800, 50);
+    Reply(2600);
+    giq_option(4, NAME, 2601, Node3250, NEUTRAL_REACTION);
+    giq_option(4, NAME, 2602, Node1800, NEUTRAL_REACTION);
 end
 
 procedure Node2650
 begin
-    gsay_reply(1592, 2650);
-    giq_option(4, 1592, 2651, Node2700, 50);
-    giq_option(4, 1592, 2652, Node700, 50);
+    Reply(2650);
+    giq_option(4, NAME, 2651, Node2700, NEUTRAL_REACTION);
+    giq_option(4, NAME, 2652, Node700, NEUTRAL_REACTION);
 end
 
 procedure Node2700
 begin
-    gsay_reply(1592, 2700);
-    giq_option(4, 1592, 2701, Node700, 50);
+    Reply(2700);
+    giq_option(4, NAME, 2701, Node700, NEUTRAL_REACTION);
 end
 
 procedure Node2750
 begin
-    gsay_reply(1592, 2750);
-    giq_option(4, 1592, 2751, Node700, 50);
-    giq_option(4, 1592, 2752, Node2300, 50);
-    giq_option(8, 1592, 2753, Node2350, 50);
+    Reply(2750);
+    giq_option(4, NAME, 2751, Node700, NEUTRAL_REACTION);
+    giq_option(4, NAME, 2752, Node2300, NEUTRAL_REACTION);
+    giq_option(8, NAME, 2753, Node2350, NEUTRAL_REACTION);
 end
 
 procedure Node2800
 begin
-    gsay_reply(1592, 2800);
-    giq_option(4, 1592, 2801, Node3300, 50);
-    giq_option(4, 1592, 2802, Node700, 50);
+    Reply(2800);
+    giq_option(4, NAME, 2801, Node3300, NEUTRAL_REACTION);
+    giq_option(4, NAME, 2802, Node700, NEUTRAL_REACTION);
 end
 
 procedure Node2850
 begin
-    gsay_reply(1592, 2850);
-    giq_option(4, 1592, 2851, Node700, 50);
-    giq_option(4, 1592, 2852, Node3350, 50);
+    Reply(2850);
+    giq_option(4, NAME, 2851, Node700, NEUTRAL_REACTION);
+    giq_option(4, NAME, 2852, Node3350, NEUTRAL_REACTION);
 end
 
 procedure Node2900
 begin
-    gsay_reply(1592, 2900);
-    giq_option(4, 1592, 2901, Node2850, 50);
+    Reply(2900);
+    giq_option(4, NAME, 2901, Node2850, NEUTRAL_REACTION);
 end
 
 procedure Node2950
 begin
-    gsay_reply(1592, 2950);
-    giq_option(4, 1592, 2951, Node3500, 50);
-    giq_option(8, 1592, 2952, Node3550, 50);
+    Reply(2950);
+    giq_option(4, NAME, 2951, Node3500, NEUTRAL_REACTION);
+    giq_option(8, NAME, 2952, Node3550, NEUTRAL_REACTION);
 end
 
 procedure Node3000
 begin
-    gsay_reply(1592, 3000);
-    giq_option(4, 1592, 3001, Node999, 50);
+    Reply(3000);
+    giq_option(4, NAME, 3001, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node3050
 begin
-    gsay_reply(1592, 3050);
-    giq_option(4, 1592, 3051, Node999, 50);
-    giq_option(4, 1592, 3052, Node2350, 50);
+    Reply(3050);
+    giq_option(4, NAME, 3051, Node999, NEUTRAL_REACTION);
+    giq_option(4, NAME, 3052, Node2350, NEUTRAL_REACTION);
 end
 
 procedure Node3100
 begin
-    gsay_reply(1592, 3100);
-    giq_option(4, 1592, 3101, Node3150, 50);
-    giq_option(4, 1592, 3102, Node3600, 50);
+    Reply(3100);
+    giq_option(4, NAME, 3101, Node3150, NEUTRAL_REACTION);
+    giq_option(4, NAME, 3102, Node3600, NEUTRAL_REACTION);
 end
 
 procedure Node3150
 begin
-    gsay_reply(1592, 3150);
-    giq_option(4, 1592, 3151, Node2950, 50);
-    giq_option(4, 1592, 3152, Node999, 50);
+    Reply(3150);
+    giq_option(4, NAME, 3151, Node2950, NEUTRAL_REACTION);
+    giq_option(4, NAME, 3152, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node3200
 begin
-    gsay_reply(1592, 3200);
-    giq_option(4, 1592, 3201, Node3650, 50);
+    Reply(3200);
+    giq_option(4, NAME, 3201, Node3650, NEUTRAL_REACTION);
 end
 
 procedure Node3250
 begin
-    gsay_reply(1592, 3250);
-    giq_option(4, 1592, 3251, Node700, 50);
-    giq_option(4, 1592, 3252, Node999, 50);
+    Reply(3250);
+    giq_option(4, NAME, 3251, Node700, NEUTRAL_REACTION);
+    giq_option(4, NAME, 3252, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node3300
 begin
-    gsay_reply(1592, 3300);
-    giq_option(4, 1592, 3301, Node3400, 50);
-    giq_option(4, 1592, 3302, Node3450, 50);
+    Reply(3300);
+    giq_option(4, NAME, 3301, Node3400, NEUTRAL_REACTION);
+    giq_option(4, NAME, 3302, Node3450, NEUTRAL_REACTION);
 end
 
 procedure Node3350
 begin
-    gsay_reply(1592, 3350);
-    giq_option(4, 1592, 3351, Node999, 50);
+    Reply(3350);
+    giq_option(4, NAME, 3351, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node3400
 begin
-    gsay_reply(1592, 3400);
-    giq_option(4, 1592, 3401, Node999, 50);
+    Reply(3400);
+    giq_option(4, NAME, 3401, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node3450
 begin
-    gsay_reply(1592, 3450);
-    giq_option(4, 1592, 3451, Node2850, 50);
+    Reply(3450);
+    giq_option(4, NAME, 3451, Node2850, NEUTRAL_REACTION);
 end
 
 procedure Node3500
 begin
-    gsay_reply(1592, 3500);
-    giq_option(4, 1592, 3501, Node999, 50);
+    Reply(3500);
+    giq_option(4, NAME, 3501, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node3550
 begin
-    gsay_reply(1592, 3550);
-    giq_option(4, 1592, 3551, Node3000, 50);
+    Reply(3550);
+    giq_option(4, NAME, 3551, Node3000, NEUTRAL_REACTION);
 end
 
 procedure Node3600
 begin
-    gsay_reply(1592, 3600 + (get_critter_stat(dude_obj, 34) == 1));
-    giq_option(4, 1592, 3602 + (get_critter_stat(dude_obj, 34) == 1), Node3700, 50);
-    giq_option(7, 1592, 3604, Node3750, 50);
-    giq_option(6, 1592, 3605, Node3400, 50);
-    giq_option(4, 1592, 3606, Node999_Hurt, 50);
+    Reply(3600 + (dude_gender == 1));
+    giq_option(4, NAME, 3602 + (dude_gender == 1), Node3700, NEUTRAL_REACTION);
+    giq_option(7, NAME, 3604, Node3750, NEUTRAL_REACTION);
+    giq_option(6, NAME, 3605, Node3400, NEUTRAL_REACTION);
+    giq_option(4, NAME, 3606, Node999_Hurt, NEUTRAL_REACTION);
 end
 
 procedure Node3650
 begin
-    gsay_reply(1592, 3650);
-    giq_option(4, 1592, 3651, Node3800, 50);
-    giq_option(4, 1592, 3652, Node3400, 50);
+    Reply(3650);
+    giq_option(4, NAME, 3651, Node3800, NEUTRAL_REACTION);
+    giq_option(4, NAME, 3652, Node3400, NEUTRAL_REACTION);
 end
 
 procedure Node3700
 begin
-    gsay_reply(1592, 3700 + (get_critter_stat(dude_obj, 34) == 1));
-    giq_option(4, 1592, 3702, Node999, 50);
+    Reply(3700 + (dude_gender == 1));
+    giq_option(4, NAME, 3702, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node3750
 begin
-    gsay_reply(1592, 3750);
-    giq_option(4, 1592, 3751, Node2300, 50);
-    giq_option(4, 1592, 3752, Node999, 50);
+    Reply(3750);
+    giq_option(4, NAME, 3751, Node2300, NEUTRAL_REACTION);
+    giq_option(4, NAME, 3752, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node3800
 begin
-    gsay_reply(1592, 3800);
-    giq_option(4, 1592, 3801, Node3850, 50);
+    Reply(3800);
+    giq_option(4, NAME, 3801, Node3850, NEUTRAL_REACTION);
 end
 
 procedure Node3850
 begin
-    gsay_reply(1592, 3850 + (get_critter_stat(dude_obj, 34) == 1));
+    Reply(3850 + (dude_gender == 1));
     set_local_var(6, 1);
-    giq_option(4, 1592, 3852, Node999, 50);
+    giq_option(4, NAME, 3852, Node999, NEUTRAL_REACTION);
 end
-
