diff --git a/Mods/Resurrection/Original/ECDTHCLW.ssl b/Mods/Resurrection/ReDefine/ECDTHCLW.ssl
index ec8a464..3692f96 100644
--- a/Mods/Resurrection/Original/ECDTHCLW.ssl
+++ b/Mods/Resurrection/ReDefine/ECDTHCLW.ssl
@@ -46,12 +46,12 @@ begin
             return 1;
         end
     end
-    if ((party_member_obj(16777779) != 0) and ((global_var(910) bwand 32) == 0)) then begin
+    if ((party_member_obj(16777779) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 32) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777779)) <= 5) then begin
             return 1;
         end
     end
-    if ((party_member_obj(16777811) != 0) and ((global_var(910) bwand 64) == 0)) then begin
+    if ((party_member_obj(16777811) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 64) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777811)) <= 5) then begin
             return 1;
         end
@@ -66,7 +66,7 @@ begin
             return 1;
         end
     end
-    if ((party_member_obj(16777814) != 0) and ((global_var(910) bwand 128) == 0)) then begin
+    if ((party_member_obj(16777814) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 128) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777814)) <= 5) then begin
             return 1;
         end
@@ -119,11 +119,11 @@ end
 
 procedure critter_p_proc
 begin
-    if (obj_can_see_obj(self_obj, dude_obj) and not((global_var(911) bwand 1) != 0)) then begin
-        attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+    if (obj_can_see_obj(self_obj, dude_obj) and not((global_var(GVAR_KARAVAN_ROUTES_COMMAND_BITS) bwand 1) != 0)) then begin
+        attack(dude_obj);
     end
-    if (obj_can_see_obj(self_obj, dude_obj) and ((global_var(911) bwand 1) != 0) and ((global_var(911) bwand 8388608) != 0)) then begin
-        attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+    if (obj_can_see_obj(self_obj, dude_obj) and ((global_var(GVAR_KARAVAN_ROUTES_COMMAND_BITS) bwand 1) != 0) and ((global_var(GVAR_KARAVAN_ROUTES_COMMAND_BITS) bwand 8388608) != 0)) then begin
+        attack(dude_obj);
     end
     if (not(anim_busy(self_obj))) then begin
         reg_anim_func(2, self_obj);
@@ -135,23 +135,23 @@ end
 
 procedure map_enter_p_proc
 begin
-    critter_add_trait(self_obj, 1, 6, 187);
-    critter_add_trait(self_obj, 1, 5, 10);
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 187);
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_DEATHCLAW);
     starttile := tile_num(self_obj);
     reg_anim_func(2, self_obj);
     reg_anim_func(1, 1);
     anim(self_obj, 1000, random(0, 5));
     reg_anim_func(3, 0);
-    if (not(metarule(22, 0))) then begin
+    if (not(is_loading_game)) then begin
         if (elevation(self_obj) == 0) then begin
-            if (global_var(1045) >= 10) then begin
+            if (global_var(GVAR_ENC_NUMBERS_ANIMALS) >= 10) then begin
                 destroy_object(self_obj);
             end
             else begin
-                set_global_var(1045, global_var(1045) + 1);
+                set_global_var(GVAR_ENC_NUMBERS_ANIMALS, global_var(GVAR_ENC_NUMBERS_ANIMALS) + 1);
             end
         end
-        if ((global_var(1023) bwand 1) != 0) then begin
+        if ((global_var(GVAR_CARAVANS_RANDOM_ECCS) bwand 1) != 0) then begin
             critter_attempt_placement(self_obj, tile_num(self_obj), 1);
         end
     end
@@ -163,11 +163,10 @@ end
 
 procedure destroy_p_proc
 begin
-    if ((global_var(911) bwand 1) != 0) then begin
-        set_global_var(916, global_var(916) - 1);
+    if ((global_var(GVAR_KARAVAN_ROUTES_COMMAND_BITS) bwand 1) != 0) then begin
+        set_global_var(GVAR_KARAVAN_ATACKERS_COUNTER, global_var(GVAR_KARAVAN_ATACKERS_COUNTER) - 1);
     end
-    if (cur_map_index == 82) then begin
+    if (CUR_MAP_BHRNDDST) then begin
         set_map_var(1, map_var(1) - 1);
     end
 end
-
