diff --git a/Mods/Resurrection/Original/ECMADDO.ssl b/Mods/Resurrection/ReDefine/ECMADDO.ssl
index 71b1a83..ea6f1d6 100644
--- a/Mods/Resurrection/Original/ECMADDO.ssl
+++ b/Mods/Resurrection/ReDefine/ECMADDO.ssl
@@ -45,12 +45,12 @@ begin
             return 1;
         end
     end
-    if ((party_member_obj(16777779) != 0) and ((global_var(910) bwand 32) == 0)) then begin
+    if ((party_member_obj(16777779) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 32) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777779)) <= 5) then begin
             return 1;
         end
     end
-    if ((party_member_obj(16777811) != 0) and ((global_var(910) bwand 64) == 0)) then begin
+    if ((party_member_obj(16777811) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 64) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777811)) <= 5) then begin
             return 1;
         end
@@ -65,7 +65,7 @@ begin
             return 1;
         end
     end
-    if ((party_member_obj(16777814) != 0) and ((global_var(910) bwand 128) == 0)) then begin
+    if ((party_member_obj(16777814) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 128) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777814)) <= 5) then begin
             return 1;
         end
@@ -119,8 +119,8 @@ end
 procedure put_critter(variable arg0, variable arg1, variable arg2)
 begin
     variable LVar3 := 0;
-    LVar3 := create_object_sid(arg0, arg1, elevation(dude_obj), arg2);
-    critter_attempt_placement(LVar3, arg1, elevation(dude_obj));
+    LVar3 := create_object_sid(arg0, arg1, dude_elevation, arg2);
+    critter_attempt_placement(LVar3, arg1, dude_elevation);
 end
 
 procedure start
@@ -147,23 +147,23 @@ begin
         end
     end
     if (tile_distance_objs(self_obj, dude_obj) > 5) then begin
-        animate_move_obj_to_tile(self_obj, tile_num(dude_obj) + random(-10, 10), 1);
+        animate_move_obj_to_tile(self_obj, dude_tile + random(-10, 10), 1);
     end
     else begin
-        animate_move_obj_to_tile(self_obj, tile_num(dude_obj) + random(-10, 10), 0);
+        animate_move_obj_to_tile(self_obj, dude_tile + random(-10, 10), 0);
     end
 end
 
 procedure map_enter_p_proc
 begin
-    critter_add_trait(self_obj, 1, 6, 286);
-    critter_add_trait(self_obj, 1, 5, 1);
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 286);
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_BERSERKER_NO_TALKING);
 end
 
 procedure add_rocks
 begin
     variable LVar0 := 0;
-    LVar0 := create_object_sid(19, 0, 0, -1);
+    LVar0 := create_object(19, 0, 0);
     add_mult_objs_to_inven(self_obj, LVar0, 15);
 end
 
@@ -175,7 +175,7 @@ end
 procedure look_at_p_proc
 begin
     script_overrides;
-    display_msg(message_str(1595, 100));
+    display_mstr(100);
 end
 
 procedure combat_p_proc
@@ -185,6 +185,5 @@ end
 
 procedure node001
 begin
-    float_msg(self_obj, message_str(1595, random(111, 117)), 2);
+    float_msg(self_obj, mstr(random(111, 117)), FLOAT_MSG_RED);
 end
-
