diff --git a/Mods/Resurrection/Original/ECSAVAG2.ssl b/Mods/Resurrection/ReDefine/ECSAVAG2.ssl
index 79e220f..a65626c 100644
--- a/Mods/Resurrection/Original/ECSAVAG2.ssl
+++ b/Mods/Resurrection/ReDefine/ECSAVAG2.ssl
@@ -46,12 +46,12 @@ begin
             return 1;
         end
     end
-    if ((party_member_obj(16777779) != 0) and ((global_var(910) bwand 32) == 0)) then begin
+    if ((party_member_obj(16777779) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 32) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777779)) <= 5) then begin
             return 1;
         end
     end
-    if ((party_member_obj(16777811) != 0) and ((global_var(910) bwand 64) == 0)) then begin
+    if ((party_member_obj(16777811) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 64) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777811)) <= 5) then begin
             return 1;
         end
@@ -66,7 +66,7 @@ begin
             return 1;
         end
     end
-    if ((party_member_obj(16777814) != 0) and ((global_var(910) bwand 128) == 0)) then begin
+    if ((party_member_obj(16777814) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 128) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777814)) <= 5) then begin
             return 1;
         end
@@ -126,13 +126,13 @@ begin
     variable LVar0 := 0;
     if (not(combat_is_initialized)) then begin
         if (obj_can_see_obj(self_obj, dude_obj)) then begin
-            if (get_critter_stat(dude_obj, 34) == 0) then begin
-                float_msg(self_obj, message_str(225, random(300, 304)), 2);
+            if (dude_gender == 0) then begin
+                float_msg(self_obj, mstr(random(300, 304)), FLOAT_MSG_RED);
             end
-            if (get_critter_stat(dude_obj, 34) == 1) then begin
-                float_msg(self_obj, message_str(225, random(303, 307)), 2);
+            if (dude_gender == 1) then begin
+                float_msg(self_obj, mstr(random(303, 307)), FLOAT_MSG_RED);
             end
-            attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+            attack(dude_obj);
         end
         if (not(anim_busy(self_obj))) then begin
             LVar0 := random(1, 20);
@@ -160,53 +160,53 @@ end
 
 procedure pickup_p_proc
 begin
-    attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+    attack(dude_obj);
 end
 
 procedure talk_p_proc
 begin
-    attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+    attack(dude_obj);
 end
 
 procedure look_at_p_proc
 begin
     script_overrides;
-    display_msg(message_str(225, 100 + (get_critter_stat(self_obj, 34) == 1)));
+    display_mstr(100 + (self_gender == 1));
 end
 
 procedure description_p_proc
 begin
     script_overrides;
-    display_msg(message_str(225, 200 + (get_critter_stat(self_obj, 34) == 1)));
+    display_mstr(200 + (self_gender == 1));
 end
 
 procedure timed_event_p_proc
 begin
     if (fixed_param == 1) then begin
         if (obj_carrying_pid_obj(self_obj, wep_pid)) then begin
-            wield_obj_critter(self_obj, obj_carrying_pid_obj(self_obj, wep_pid));
+            wield_obj(obj_carrying_pid_obj(self_obj, wep_pid));
         end
     end
 end
 
 procedure map_enter_p_proc
 begin
-    critter_add_trait(self_obj, 1, 6, 161);
-    critter_add_trait(self_obj, 1, 5, 1);
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 161);
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_BERSERKER_NO_TALKING);
     starttile := tile_num(self_obj);
-    if ((elevation(self_obj) != 0) and not(metarule(22, 0))) then begin
+    if ((elevation(self_obj) != 0) and not(is_loading_game)) then begin
         val := random(0, 1);
         if (val == 1) then begin
-            restock_obj := create_object_sid(273, 0, 0, -1);
+            restock_obj := create_object(273, 0, 0);
             add_mult_objs_to_inven(self_obj, restock_obj, val);
         end
-        if (get_critter_stat(self_obj, 34) == 0) then begin
+        if (self_gender == 0) then begin
             wep_pid := 7;
         end
         else begin
             wep_pid := 4;
         end
-        restock_obj := create_object_sid(wep_pid, 0, 0, -1);
+        restock_obj := create_object(wep_pid, 0, 0);
         add_mult_objs_to_inven(self_obj, restock_obj, 1);
         add_timer_event(self_obj, 1, 1);
     end
@@ -218,10 +218,9 @@ end
 
 procedure damage_p_proc
 begin
-    attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+    attack(dude_obj);
 end
 
 procedure destroy_p_proc
 begin
 end
-
