diff --git a/Mods/Resurrection/Original/FETAK2.ssl b/Mods/Resurrection/ReDefine/FETAK2.ssl
index 964c17e..b1d8afd 100644
--- a/Mods/Resurrection/Original/FETAK2.ssl
+++ b/Mods/Resurrection/ReDefine/FETAK2.ssl
@@ -46,12 +46,12 @@ begin
             return 1;
         end
     end
-    if ((party_member_obj(16777779) != 0) and ((global_var(910) bwand 32) == 0)) then begin
+    if ((party_member_obj(16777779) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 32) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777779)) <= 5) then begin
             return 1;
         end
     end
-    if ((party_member_obj(16777811) != 0) and ((global_var(910) bwand 64) == 0)) then begin
+    if ((party_member_obj(16777811) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 64) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777811)) <= 5) then begin
             return 1;
         end
@@ -66,7 +66,7 @@ begin
             return 1;
         end
     end
-    if ((party_member_obj(16777814) != 0) and ((global_var(910) bwand 128) == 0)) then begin
+    if ((party_member_obj(16777814) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 128) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777814)) <= 5) then begin
             return 1;
         end
@@ -125,11 +125,11 @@ procedure critter_p_proc
 begin
     if ((local_var(5) == 2) and obj_can_see_obj(self_obj, dude_obj)) then begin
         set_local_var(5, 1);
-        attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+        attack(dude_obj);
     end
     if (local_var(6) == 1) then begin
         if (obj_can_see_obj(self_obj, dude_obj)) then begin
-            attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+            attack(dude_obj);
         end
     end
 end
@@ -143,57 +143,57 @@ end
 
 procedure talk_p_proc
 begin
-    if (get_critter_stat(dude_obj, 34) == 0) then begin
-        float_msg(self_obj, message_str(1434, random(300, 309)), 8);
+    if (dude_gender == 0) then begin
+        float_msg(self_obj, mstr(random(300, 309)), FLOAT_MSG_YELLOW);
     end
-    if (get_critter_stat(dude_obj, 34) == 1) then begin
-        float_msg(self_obj, message_str(1434, random(301, 310)), 8);
+    if (dude_gender == 1) then begin
+        float_msg(self_obj, mstr(random(301, 310)), FLOAT_MSG_YELLOW);
     end
 end
 
 procedure look_at_p_proc
 begin
     script_overrides;
-    display_msg(message_str(1434, 100 + (get_critter_stat(self_obj, 34) == 1)));
+    display_mstr(100 + (self_gender == 1));
 end
 
 procedure description_p_proc
 begin
     script_overrides;
-    if (get_critter_stat(dude_obj, 4) > 5) then begin
-        display_msg(message_str(1434, 201));
+    if (dude_iq > 5) then begin
+        display_mstr(201);
     end
     else begin
-        display_msg(message_str(1434, 200));
+        display_mstr(200);
     end
 end
 
 procedure timed_event_p_proc
 begin
     if (fixed_param == 1) then begin
-        if (not(anim_busy(self_obj)) and (has_trait(1, self_obj, 666) == 1)) then begin
+        if (not(anim_busy(self_obj)) and (obj_is_visible_flag(self_obj) == 1)) then begin
             reg_anim_func(2, self_obj);
             reg_anim_func(1, 1);
             animate_move_obj_to_tile(self_obj, tile_num_in_direction(local_var(7), random(0, 5), random(1, 5)), 0);
             reg_anim_func(3, 0);
         end
-        metarule3(100, self_obj, 1, 0);
+        rm_fixed_timer_event(self_obj, 1);
         add_timer_event(self_obj, game_ticks(random(5, 25)), 1);
     end
     if (fixed_param == 2) then begin
-        if ((tile_distance(tile_num(self_obj), tile_num(dude_obj)) <= 4) and not(combat_is_initialized) and (has_trait(1, self_obj, 666) == 1)) then begin
-            float_msg(self_obj, message_str(1434, random(351, 359)), 8);
+        if ((tile_distance(tile_num(self_obj), dude_tile) <= 4) and not(combat_is_initialized) and (obj_is_visible_flag(self_obj) == 1)) then begin
+            float_msg(self_obj, mstr(random(351, 359)), FLOAT_MSG_YELLOW);
         end
-        metarule3(100, self_obj, 2, 0);
+        rm_fixed_timer_event(self_obj, 2);
         add_timer_event(self_obj, game_ticks(random(5, 10)), 2);
     end
 end
 
 procedure map_enter_p_proc
 begin
-    critter_add_trait(self_obj, 1, 6, 229);
-    critter_add_trait(self_obj, 1, 5, 15);
-    if (not(metarule(22, 0))) then begin
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 229);
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_CHILD);
+    if (not(is_loading_game)) then begin
         if (local_var(7) == 0) then begin
             set_local_var(7, tile_num(self_obj));
         end
@@ -201,18 +201,18 @@ begin
         reg_anim_func(1, 1);
         anim(self_obj, 1000, random(0, 5));
         reg_anim_func(3, 0);
-        if (cur_map_index == 13) then begin
-            if ((global_var(848) == 1) and not((global_var(842) bwand 33554432) != 0) and (tile_distance(20663, tile_num(self_obj)) <= 12)) then begin
+        if (CUR_MAP_SED_ZTR) then begin
+            if ((global_var(GVAR_NESTOR_LEAVED_CORATH) == 1) and not((global_var(GVAR_C_CORATH_STATUS) bwand 33554432) != 0) and (tile_distance(20663, tile_num(self_obj)) <= 12)) then begin
                 set_obj_visibility(self_obj, 1);
             end
             else begin
-                if ((has_trait(1, self_obj, 666) == 0) and (random(1, 5) == 5)) then begin
+                if ((obj_is_visible_flag(self_obj) == 0) and (random(1, 5) == 5)) then begin
                     set_obj_visibility(self_obj, 0);
                 end
             end
         end
-        if (cur_map_index == 22) then begin
-            if (((global_var(904) bwand 1) != 0) or (((global_var(1017) bwand 2048) != 0) and ((global_var(904) bwand 8192) != 0) or (global_var(1029) == 2)) and (has_trait(1, self_obj, 666) == 1)) then begin
+        if (CUR_MAP_ALB_MRTV) then begin
+            if (((global_var(GVAR_AQE_ALBUQ_EMPEROR_EVENTS) bwand 1) != 0) or (((global_var(GVAR_AQE_DIRECTIONS_SETUP) bwand 2048) != 0) and ((global_var(GVAR_AQE_ALBUQ_EMPEROR_EVENTS) bwand 8192) != 0) or (global_var(GVAR_AQE_KICK_BENICIO_FROM_ALBUQUERQUE) == 2)) and (obj_is_visible_flag(self_obj) == 1)) then begin
                 if (((map_var(0) bwand 32768) != 0) and (tile_distance(tile_num(self_obj), 30277) <= 15)) then begin
                     set_obj_visibility(self_obj, 1);
                 end
@@ -224,15 +224,15 @@ begin
                 end
             end
             else begin
-                if (has_trait(1, self_obj, 666) == 0) then begin
+                if (obj_is_visible_flag(self_obj) == 0) then begin
                     set_obj_visibility(self_obj, 0);
                 end
             end
         end
     end
-    metarule3(100, self_obj, 1, 0);
+    rm_fixed_timer_event(self_obj, 1);
     add_timer_event(self_obj, game_ticks(random(5, 25)), 1);
-    metarule3(100, self_obj, 2, 0);
+    rm_fixed_timer_event(self_obj, 2);
     add_timer_event(self_obj, game_ticks(random(5, 10)), 2);
 end
 
@@ -243,7 +243,7 @@ begin
         set_local_var(6, 1);
     end
     if (random(1, 4) == 1) then begin
-        float_msg(self_obj, message_str(1434, random(400, 404)), 8);
+        float_msg(self_obj, mstr(random(400, 404)), FLOAT_MSG_YELLOW);
     end
 end
 
@@ -251,52 +251,52 @@ procedure destroy_p_proc
 begin
     if (source_obj == dude_obj) then begin
         if (source_obj == dude_obj) then begin
-            set_global_var(2, 0);
-            set_global_var(1, global_var(1) + 1);
-            set_global_var(0, global_var(0) + -15);
+            set_global_var(GVAR_CHAMPION_REPUTATION, 0);
+            set_global_var(GVAR_CHILDKILLER_REPUTATION, global_var(GVAR_CHILDKILLER_REPUTATION) + 1);
+            set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + -15);
             debug_msg("Player gains " + -15 + " Karma Points.");
-            set_global_var(37, 0);
-            set_global_var(38, 0);
-            set_global_var(39, 0);
-            set_global_var(40, 0);
-            set_global_var(41, 0);
-            set_global_var(42, 0);
-            set_global_var(43, 0);
-            set_global_var(44, 0);
-            set_global_var(45, 0);
-            if (global_var(0) >= 1000) then begin
-                set_global_var(37, 1);
+            set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+            set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+            set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+            set_global_var(GVAR_KARMA_DEFENDER, 0);
+            set_global_var(GVAR_KARMA_WANDERER, 0);
+            set_global_var(GVAR_KARMA_BETRAYER, 0);
+            set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+            set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+            set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+                set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
             end
             else begin
-                if (global_var(0) >= 750) then begin
-                    set_global_var(38, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                    set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
                 end
                 else begin
-                    if (global_var(0) >= 500) then begin
-                        set_global_var(39, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                        set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                     end
                     else begin
-                        if (global_var(0) >= 250) then begin
-                            set_global_var(40, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                            set_global_var(GVAR_KARMA_DEFENDER, 1);
                         end
                         else begin
-                            if (global_var(0) > -250) then begin
-                                set_global_var(41, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                                set_global_var(GVAR_KARMA_WANDERER, 1);
                             end
                             else begin
-                                if (global_var(0) > -500) then begin
-                                    set_global_var(42, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                    set_global_var(GVAR_KARMA_BETRAYER, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -750) then begin
-                                        set_global_var(43, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                        set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                     end
                                     else begin
-                                        if (global_var(0) > -1000) then begin
-                                            set_global_var(44, 1);
+                                        if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                            set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                         end
                                         else begin
-                                            set_global_var(45, 1);
+                                            set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                         end
                                     end
                                 end
@@ -307,8 +307,8 @@ begin
             end
         end
     end
-    if (metarule(46, 0) != 1) then begin
-        temp_critter := create_object_sid(obj_pid(self_obj), 0, 0, 1434);
+    if (cur_town != 1) then begin
+        temp_critter := create_object_sid(obj_pid(self_obj), 0, 0, NAME);
         set_obj_visibility(temp_critter, 1);
         move_to(temp_critter, local_var(7), elevation(self_obj));
     end
@@ -317,4 +317,3 @@ end
 procedure push_p_proc
 begin
 end
-
