diff --git a/Mods/Resurrection/Original/G11TY.ssl b/Mods/Resurrection/ReDefine/G11TY.ssl
index 94f4d62..9f77d7a 100644
--- a/Mods/Resurrection/Original/G11TY.ssl
+++ b/Mods/Resurrection/ReDefine/G11TY.ssl
@@ -47,12 +47,12 @@ begin
             return 1;
         end
     end
-    if ((party_member_obj(16777779) != 0) and ((global_var(910) bwand 32) == 0)) then begin
+    if ((party_member_obj(16777779) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 32) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777779)) <= 5) then begin
             return 1;
         end
     end
-    if ((party_member_obj(16777811) != 0) and ((global_var(910) bwand 64) == 0)) then begin
+    if ((party_member_obj(16777811) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 64) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777811)) <= 5) then begin
             return 1;
         end
@@ -67,7 +67,7 @@ begin
             return 1;
         end
     end
-    if ((party_member_obj(16777814) != 0) and ((global_var(910) bwand 128) == 0)) then begin
+    if ((party_member_obj(16777814) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 128) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777814)) <= 5) then begin
             return 1;
         end
@@ -133,7 +133,7 @@ end
 
 procedure talk_p_proc
 begin
-    float_msg(self_obj, message_str(461, random(300, 305)), 8);
+    float_msg(self_obj, mstr(random(300, 305)), FLOAT_MSG_YELLOW);
 end
 
 procedure destroy_p_proc
@@ -144,10 +144,10 @@ procedure look_at_p_proc
 begin
     script_overrides;
     if (local_var(4) == 1) then begin
-        display_msg(message_str(461, 150));
+        display_mstr(150);
     end
     else begin
-        display_msg(message_str(461, 100));
+        display_mstr(100);
     end
 end
 
@@ -155,10 +155,10 @@ procedure description_p_proc
 begin
     script_overrides;
     if (local_var(4) == 1) then begin
-        display_msg(message_str(461, 250));
+        display_mstr(250);
     end
     else begin
-        display_msg(message_str(461, 200));
+        display_mstr(200);
     end
 end
 
@@ -172,8 +172,8 @@ end
 
 procedure map_enter_p_proc
 begin
-    critter_add_trait(self_obj, 1, 6, 279);
-    critter_add_trait(self_obj, 1, 5, 180);
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 279);
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_PEASANT_NO_MOVE);
     add_timer_event(self_obj, 1, 2);
 end
 
@@ -193,4 +193,3 @@ begin
         end
     end
 end
-
