diff --git a/Mods/Resurrection/Original/GCOM6REA.ssl b/Mods/Resurrection/ReDefine/GCOM6REA.ssl
index 1b3a22b..202d33d 100644
--- a/Mods/Resurrection/Original/GCOM6REA.ssl
+++ b/Mods/Resurrection/ReDefine/GCOM6REA.ssl
@@ -61,12 +61,12 @@ begin
             return 1;
         end
     end
-    if ((party_member_obj(16777779) != 0) and ((global_var(910) bwand 32) == 0)) then begin
+    if ((party_member_obj(16777779) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 32) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777779)) <= 5) then begin
             return 1;
         end
     end
-    if ((party_member_obj(16777811) != 0) and ((global_var(910) bwand 64) == 0)) then begin
+    if ((party_member_obj(16777811) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 64) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777811)) <= 5) then begin
             return 1;
         end
@@ -81,7 +81,7 @@ begin
             return 1;
         end
     end
-    if ((party_member_obj(16777814) != 0) and ((global_var(910) bwand 128) == 0)) then begin
+    if ((party_member_obj(16777814) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 128) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777814)) <= 5) then begin
             return 1;
         end
@@ -140,26 +140,26 @@ procedure use_p_proc
 begin
     script_overrides;
     if (combat_is_initialized) then begin
-        display_msg(message_str(450, 300));
+        display_mstr(300);
     end
     else begin
         start_gdialog(450, self_obj, 4, -1, -1);
         gsay_start;
-        if (get_critter_stat(dude_obj, 4) < 4) then begin
+        if (dude_iq < 4) then begin
             call Node500();
         end
         else begin
-            if (global_var(953) == 0) then begin
+            if (global_var(GVAR_G_BASE_DESTROYED) == 0) then begin
                 call Node550();
             end
             else begin
-                if (global_var(953) == 1) then begin
+                if (global_var(GVAR_G_BASE_DESTROYED) == 1) then begin
                     call Node1050();
                 end
-                if (global_var(953) == 2) then begin
+                if (global_var(GVAR_G_BASE_DESTROYED) == 2) then begin
                     call Node1000();
                 end
-                if ((global_var(953) == 3) or (global_var(953) == 4)) then begin
+                if ((global_var(GVAR_G_BASE_DESTROYED) == 3) or (global_var(GVAR_G_BASE_DESTROYED) == 4)) then begin
                     call Node1450();
                 end
             end
@@ -172,13 +172,13 @@ end
 procedure look_at_p_proc
 begin
     script_overrides;
-    display_msg(message_str(450, 100));
+    display_mstr(100);
 end
 
 procedure description_p_proc
 begin
     script_overrides;
-    display_msg(message_str(450, 200));
+    display_mstr(200);
 end
 
 procedure use_skill_on_p_proc
@@ -209,192 +209,191 @@ end
 
 procedure Node500
 begin
-    gsay_reply(450, 500);
-    giq_option(-3, 450, 501, Node999, 50);
+    Reply(500);
+    giq_option(-3, NAME, 501, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node550
 begin
-    gsay_reply(450, 550);
-    giq_option(4, 450, 551, Node1100, 50);
+    Reply(550);
+    giq_option(4, NAME, 551, Node1100, NEUTRAL_REACTION);
     if ((local_var(0) bwand 1) == 0) then begin
-        giq_option(4, 450, 552, Node600, 50);
+        giq_option(4, NAME, 552, Node600, NEUTRAL_REACTION);
     end
     if (map_var(7) == 0) then begin
-        giq_option(4, 450, 553, Node650, 50);
+        giq_option(4, NAME, 553, Node650, NEUTRAL_REACTION);
     end
     else begin
-        giq_option(4, 450, 553, Node850, 50);
+        giq_option(4, NAME, 553, Node850, NEUTRAL_REACTION);
     end
-    giq_option(4, 450, 554, Node999, 50);
+    giq_option(4, NAME, 554, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node600
 begin
-    gsay_reply(450, 600);
+    Reply(600);
     set_local_var(0, local_var(0) bwor 1);
     if (map_var(7) == 0) then begin
-        giq_option(4, 450, 601, Node650, 50);
+        giq_option(4, NAME, 601, Node650, NEUTRAL_REACTION);
     end
     else begin
-        giq_option(4, 450, 601, Node850, 50);
+        giq_option(4, NAME, 601, Node850, NEUTRAL_REACTION);
     end
-    giq_option(4, 450, 602, Node999, 50);
+    giq_option(4, NAME, 602, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node650
 begin
-    gsay_reply(450, 650);
-    giq_option(4, 450, 651, Node1250, 50);
-    giq_option(4, 450, 652, Node700, 50);
-    giq_option(4, 450, 653, Node1350, 50);
-    giq_option(4, 450, 654, Node800, 50);
-    giq_option(4, 450, 655, Node700, 50);
-    giq_option(4, 450, 656, Node550, 50);
-    giq_option(4, 450, 657, Node999, 50);
+    Reply(650);
+    giq_option(4, NAME, 651, Node1250, NEUTRAL_REACTION);
+    giq_option(4, NAME, 652, Node700, NEUTRAL_REACTION);
+    giq_option(4, NAME, 653, Node1350, NEUTRAL_REACTION);
+    giq_option(4, NAME, 654, Node800, NEUTRAL_REACTION);
+    giq_option(4, NAME, 655, Node700, NEUTRAL_REACTION);
+    giq_option(4, NAME, 656, Node550, NEUTRAL_REACTION);
+    giq_option(4, NAME, 657, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node700
 begin
-    gsay_reply(450, 700);
-    giq_option(4, 450, 701, Node750, 50);
-    giq_option(4, 450, 702, Node650, 50);
-    giq_option(4, 450, 703, Node999, 50);
+    Reply(700);
+    giq_option(4, NAME, 701, Node750, NEUTRAL_REACTION);
+    giq_option(4, NAME, 702, Node650, NEUTRAL_REACTION);
+    giq_option(4, NAME, 703, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node750
 begin
-    gsay_reply(450, 750);
-    gfade_out(1);
+    Reply(750);
+    fade_out;
     game_time_advance(2 * 60 * 10);
-    gfade_in(1);
-    giq_option(4, 450, 751, Node650, 50);
-    giq_option(4, 450, 752, Node550, 50);
-    giq_option(4, 450, 753, Node999, 50);
+    fade_in;
+    giq_option(4, NAME, 751, Node650, NEUTRAL_REACTION);
+    giq_option(4, NAME, 752, Node550, NEUTRAL_REACTION);
+    giq_option(4, NAME, 753, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node800
 begin
-    gsay_reply(450, 800);
-    giq_option(4, 450, 801, Node999_send_robot, 50);
-    if (((local_var(0) bwand 2) == 0) and (has_skill(dude_obj, 12) > 120)) then begin
-        giq_option(4, 450, 802, Node900, 50);
+    Reply(800);
+    giq_option(4, NAME, 801, Node999_send_robot, NEUTRAL_REACTION);
+    if (((local_var(0) bwand 2) == 0) and (dude_skill(SKILL_SCIENCE) > 120)) then begin
+        giq_option(4, NAME, 802, Node900, NEUTRAL_REACTION);
     end
-    giq_option(4, 450, 803, Node650, 50);
-    giq_option(4, 450, 804, Node999, 50);
+    giq_option(4, NAME, 803, Node650, NEUTRAL_REACTION);
+    giq_option(4, NAME, 804, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node850
 begin
-    gsay_reply(450, 850);
-    giq_option(4, 450, 851, Node550, 50);
-    giq_option(4, 450, 852, Node999, 50);
+    Reply(850);
+    giq_option(4, NAME, 851, Node550, NEUTRAL_REACTION);
+    giq_option(4, NAME, 852, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node900
 begin
-    gsay_reply(450, 900);
-    if (has_skill(dude_obj, 12) >= 130) then begin
-        giq_option(4, 450, 901, Node950, 50);
+    Reply(900);
+    if (dude_skill(SKILL_SCIENCE) >= 130) then begin
+        giq_option(4, NAME, 901, Node950, NEUTRAL_REACTION);
     end
-    giq_option(4, 450, 902, Node650, 50);
-    giq_option(4, 450, 903, Node999, 50);
+    giq_option(4, NAME, 902, Node650, NEUTRAL_REACTION);
+    giq_option(4, NAME, 903, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node950
 begin
-    gsay_reply(450, 950);
+    Reply(950);
     set_local_var(0, local_var(0) bwor 2);
-    giq_option(4, 450, 951, Node999_send_robot, 50);
-    giq_option(4, 450, 952, Node650, 50);
-    giq_option(4, 450, 953, Node999, 50);
+    giq_option(4, NAME, 951, Node999_send_robot, NEUTRAL_REACTION);
+    giq_option(4, NAME, 952, Node650, NEUTRAL_REACTION);
+    giq_option(4, NAME, 953, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node1000
 begin
-    gsay_reply(450, 1000);
-    giq_option(4, 450, 1001, Node1200, 50);
-    giq_option(4, 450, 1002, Node999, 50);
+    Reply(1000);
+    giq_option(4, NAME, 1001, Node1200, NEUTRAL_REACTION);
+    giq_option(4, NAME, 1002, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node1050
 begin
-    gsay_reply(450, 1050);
+    Reply(1050);
     if ((local_var(0) bwand 4) == 0) then begin
-        giq_option(4, 450, 1051, Node1150, 50);
+        giq_option(4, NAME, 1051, Node1150, NEUTRAL_REACTION);
     end
-    giq_option(4, 450, 1052, Node999, 50);
+    giq_option(4, NAME, 1052, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node1100
 begin
-    gsay_reply(450, 1100);
+    Reply(1100);
     if ((local_var(0) bwand 1) == 0) then begin
-        giq_option(4, 450, 1101, Node600, 50);
+        giq_option(4, NAME, 1101, Node600, NEUTRAL_REACTION);
     end
     if (map_var(7) == 0) then begin
-        giq_option(4, 450, 1102, Node650, 50);
+        giq_option(4, NAME, 1102, Node650, NEUTRAL_REACTION);
     end
     else begin
-        giq_option(4, 450, 1102, Node850, 50);
+        giq_option(4, NAME, 1102, Node850, NEUTRAL_REACTION);
     end
-    giq_option(4, 450, 1103, Node999, 50);
+    giq_option(4, NAME, 1103, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node1150
 begin
-    gsay_reply(450, 1150);
+    Reply(1150);
     set_local_var(0, local_var(0) bwor 4);
-    giq_option(4, 450, 1151, Node999, 50);
+    giq_option(4, NAME, 1151, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node1200
 begin
-    gsay_reply(450, 1200);
-    giq_option(4, 450, 1201, Node999, 50);
+    Reply(1200);
+    giq_option(4, NAME, 1201, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node1250
 begin
-    gsay_reply(450, 1250);
-    giq_option(4, 450, 1251, Node1300, 50);
-    giq_option(4, 450, 1252, Node650, 50);
-    giq_option(4, 450, 1253, Node999, 50);
+    Reply(1250);
+    giq_option(4, NAME, 1251, Node1300, NEUTRAL_REACTION);
+    giq_option(4, NAME, 1252, Node650, NEUTRAL_REACTION);
+    giq_option(4, NAME, 1253, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node1300
 begin
-    gsay_reply(450, 1300);
-    gfade_out(1);
+    Reply(1300);
+    fade_out;
     game_time_advance(2 * 60 * 10);
-    gfade_in(1);
-    giq_option(4, 450, 1301, Node650, 50);
-    giq_option(4, 450, 1302, Node550, 50);
-    giq_option(4, 450, 1303, Node999, 50);
+    fade_in;
+    giq_option(4, NAME, 1301, Node650, NEUTRAL_REACTION);
+    giq_option(4, NAME, 1302, Node550, NEUTRAL_REACTION);
+    giq_option(4, NAME, 1303, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node1350
 begin
-    gsay_reply(450, 1350);
-    giq_option(4, 450, 1351, Node1400, 50);
-    giq_option(4, 450, 1352, Node650, 50);
-    giq_option(4, 450, 1353, Node999, 50);
+    Reply(1350);
+    giq_option(4, NAME, 1351, Node1400, NEUTRAL_REACTION);
+    giq_option(4, NAME, 1352, Node650, NEUTRAL_REACTION);
+    giq_option(4, NAME, 1353, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node1400
 begin
-    gsay_reply(450, 1400);
-    gfade_out(1);
+    Reply(1400);
+    fade_out;
     game_time_advance(2 * 60 * 10);
-    gfade_in(1);
-    giq_option(4, 450, 1401, Node650, 50);
-    giq_option(4, 450, 1402, Node550, 50);
-    giq_option(4, 450, 1403, Node999, 50);
+    fade_in;
+    giq_option(4, NAME, 1401, Node650, NEUTRAL_REACTION);
+    giq_option(4, NAME, 1402, Node550, NEUTRAL_REACTION);
+    giq_option(4, NAME, 1403, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node1450
 begin
-    gsay_reply(450, 1450);
-    giq_option(4, 450, 1451, Node999, 50);
+    Reply(1450);
+    giq_option(4, NAME, 1451, Node999, NEUTRAL_REACTION);
 end
-
