diff --git a/Mods/Resurrection/Original/GCOM2SCR.ssl b/Mods/Resurrection/ReDefine/GCOM2SCR.ssl
index 4032216..764786e 100644
--- a/Mods/Resurrection/Original/GCOM2SCR.ssl
+++ b/Mods/Resurrection/ReDefine/GCOM2SCR.ssl
@@ -61,12 +61,12 @@ begin
             return 1;
         end
     end
-    if ((party_member_obj(16777779) != 0) and ((global_var(910) bwand 32) == 0)) then begin
+    if ((party_member_obj(16777779) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 32) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777779)) <= 5) then begin
             return 1;
         end
     end
-    if ((party_member_obj(16777811) != 0) and ((global_var(910) bwand 64) == 0)) then begin
+    if ((party_member_obj(16777811) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 64) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777811)) <= 5) then begin
             return 1;
         end
@@ -81,7 +81,7 @@ begin
             return 1;
         end
     end
-    if ((party_member_obj(16777814) != 0) and ((global_var(910) bwand 128) == 0)) then begin
+    if ((party_member_obj(16777814) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 128) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777814)) <= 5) then begin
             return 1;
         end
@@ -139,7 +139,7 @@ end
 procedure use_p_proc
 begin
     script_overrides;
-    set_global_var(963, global_var(963) bwor 64);
+    set_global_var(GVAR_G_SEC_ROBOTS_ACTIVATION, global_var(GVAR_G_SEC_ROBOTS_ACTIVATION) bwor 64);
     if (combat_is_initialized) then begin
         terminate_combat;
         add_timer_event(self_obj, 1, 1);
@@ -162,17 +162,17 @@ end
 procedure look_at_p_proc
 begin
     script_overrides;
-    display_msg(message_str(475, 100));
+    display_mstr(100);
 end
 
 procedure description_p_proc
 begin
     script_overrides;
-    if (global_var(927) == 1) then begin
-        display_msg(message_str(475, 201));
+    if (global_var(GVAR_ENEMY_OBRODA) == 1) then begin
+        display_mstr(201);
     end
     else begin
-        display_msg(message_str(475, 200));
+        display_mstr(200);
     end
 end
 
@@ -181,7 +181,7 @@ begin
     if (fixed_param == 1) then begin
         start_gdialog(475, self_obj, 4, -1, -1);
         gsay_start;
-        if (global_var(927) == 1) then begin
+        if (global_var(GVAR_ENEMY_OBRODA) == 1) then begin
             call Node550();
         end
         else begin
@@ -202,306 +202,306 @@ end
 
 procedure Node999
 begin
-    set_global_var(963, global_var(963) bwand (-1 - 64));
+    set_global_var(GVAR_G_SEC_ROBOTS_ACTIVATION, global_var(GVAR_G_SEC_ROBOTS_ACTIVATION) bwand (-1 - 64));
 end
 
 procedure Node500
 begin
-    gsay_reply(475, 500);
-    if (has_skill(dude_obj, 12) >= 130) then begin
-        giq_option(4, 475, 501, Node950, 50);
+    Reply(500);
+    if (dude_skill(SKILL_SCIENCE) >= 130) then begin
+        giq_option(4, NAME, 501, Node950, NEUTRAL_REACTION);
     end
-    if ((has_skill(dude_obj, 12) > 115) and (has_skill(dude_obj, 12) < 130)) then begin
-        giq_option(4, 475, 501, Node750, 50);
+    if ((dude_skill(SKILL_SCIENCE) > 115) and (dude_skill(SKILL_SCIENCE) < 130)) then begin
+        giq_option(4, NAME, 501, Node750, NEUTRAL_REACTION);
     end
-    if ((has_skill(dude_obj, 12) > 100) and (has_skill(dude_obj, 12) <= 115)) then begin
-        giq_option(4, 475, 501, Node600, 50);
+    if ((dude_skill(SKILL_SCIENCE) > 100) and (dude_skill(SKILL_SCIENCE) <= 115)) then begin
+        giq_option(4, NAME, 501, Node600, NEUTRAL_REACTION);
     end
-    if (has_skill(dude_obj, 12) <= 100) then begin
-        giq_option(4, 475, 501, Node1150, 50);
+    if (dude_skill(SKILL_SCIENCE) <= 100) then begin
+        giq_option(4, NAME, 501, Node1150, NEUTRAL_REACTION);
     end
-    giq_option(4, 475, 502, Node1300, 50);
-    giq_option(4, 475, 503, Node999, 50);
-    giq_option(-3, 475, 504, Node999, 50);
+    giq_option(4, NAME, 502, Node1300, NEUTRAL_REACTION);
+    giq_option(4, NAME, 503, Node999, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 504, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node550
 begin
-    gsay_reply(475, 550);
-    if (has_skill(dude_obj, 12) >= 130) then begin
-        giq_option(4, 475, 551, Node950, 50);
+    Reply(550);
+    if (dude_skill(SKILL_SCIENCE) >= 130) then begin
+        giq_option(4, NAME, 551, Node950, NEUTRAL_REACTION);
     end
-    if ((has_skill(dude_obj, 12) > 115) and (has_skill(dude_obj, 12) < 130)) then begin
-        giq_option(4, 475, 551, Node750, 50);
+    if ((dude_skill(SKILL_SCIENCE) > 115) and (dude_skill(SKILL_SCIENCE) < 130)) then begin
+        giq_option(4, NAME, 551, Node750, NEUTRAL_REACTION);
     end
-    if ((has_skill(dude_obj, 12) > 100) and (has_skill(dude_obj, 12) <= 115)) then begin
-        giq_option(4, 475, 551, Node600, 50);
+    if ((dude_skill(SKILL_SCIENCE) > 100) and (dude_skill(SKILL_SCIENCE) <= 115)) then begin
+        giq_option(4, NAME, 551, Node600, NEUTRAL_REACTION);
     end
-    if (has_skill(dude_obj, 12) <= 100) then begin
-        giq_option(4, 475, 551, Node1250, 50);
+    if (dude_skill(SKILL_SCIENCE) <= 100) then begin
+        giq_option(4, NAME, 551, Node1250, NEUTRAL_REACTION);
     end
-    giq_option(4, 475, 552, Node999, 50);
-    giq_option(-3, 475, 553, Node999, 50);
+    giq_option(4, NAME, 552, Node999, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 553, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node600
 begin
-    gsay_reply(475, 600);
+    Reply(600);
     set_local_var(0, 0);
-    if ((global_var(963) bwand 2) != 0) then begin
-        giq_option(4, 475, 601, Node650, 50);
+    if ((global_var(GVAR_G_SEC_ROBOTS_ACTIVATION) bwand 2) != 0) then begin
+        giq_option(4, NAME, 601, Node650, NEUTRAL_REACTION);
     end
     else begin
-        giq_option(4, 475, 602, Node700, 50);
+        giq_option(4, NAME, 602, Node700, NEUTRAL_REACTION);
     end
-    if (global_var(927) == 1) then begin
-        giq_option(4, 475, 603, Node550, 50);
+    if (global_var(GVAR_ENEMY_OBRODA) == 1) then begin
+        giq_option(4, NAME, 603, Node550, NEUTRAL_REACTION);
     end
     else begin
-        giq_option(4, 475, 603, Node500, 50);
+        giq_option(4, NAME, 603, Node500, NEUTRAL_REACTION);
     end
-    giq_option(4, 475, 604, Node999, 50);
+    giq_option(4, NAME, 604, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node650
 begin
-    gsay_reply(475, 650);
-    set_global_var(963, global_var(963) bwand (-1 - 2));
-    giq_option(4, 475, 651, Node700, 50);
-    if (global_var(927) == 1) then begin
-        giq_option(4, 475, 652, Node550, 50);
+    Reply(650);
+    set_global_var(GVAR_G_SEC_ROBOTS_ACTIVATION, global_var(GVAR_G_SEC_ROBOTS_ACTIVATION) bwand (-1 - 2));
+    giq_option(4, NAME, 651, Node700, NEUTRAL_REACTION);
+    if (global_var(GVAR_ENEMY_OBRODA) == 1) then begin
+        giq_option(4, NAME, 652, Node550, NEUTRAL_REACTION);
     end
     else begin
-        giq_option(4, 475, 652, Node500, 50);
+        giq_option(4, NAME, 652, Node500, NEUTRAL_REACTION);
     end
-    giq_option(4, 475, 653, Node999, 50);
+    giq_option(4, NAME, 653, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node700
 begin
-    gsay_reply(475, 700);
-    set_global_var(963, global_var(963) bwor 2);
-    giq_option(4, 475, 701, Node650, 50);
-    if (global_var(927) == 1) then begin
-        giq_option(4, 475, 702, Node550, 50);
+    Reply(700);
+    set_global_var(GVAR_G_SEC_ROBOTS_ACTIVATION, global_var(GVAR_G_SEC_ROBOTS_ACTIVATION) bwor 2);
+    giq_option(4, NAME, 701, Node650, NEUTRAL_REACTION);
+    if (global_var(GVAR_ENEMY_OBRODA) == 1) then begin
+        giq_option(4, NAME, 702, Node550, NEUTRAL_REACTION);
     end
     else begin
-        giq_option(4, 475, 702, Node500, 50);
+        giq_option(4, NAME, 702, Node500, NEUTRAL_REACTION);
     end
-    giq_option(4, 475, 703, Node999, 50);
+    giq_option(4, NAME, 703, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node750
 begin
-    gsay_reply(475, 750);
+    Reply(750);
     set_local_var(0, 1);
-    if ((global_var(963) bwand 2) != 0) then begin
-        giq_option(4, 475, 751, Node800, 50);
+    if ((global_var(GVAR_G_SEC_ROBOTS_ACTIVATION) bwand 2) != 0) then begin
+        giq_option(4, NAME, 751, Node800, NEUTRAL_REACTION);
     end
     else begin
-        giq_option(4, 475, 752, Node850, 50);
+        giq_option(4, NAME, 752, Node850, NEUTRAL_REACTION);
     end
     if (map_var(8) != 1) then begin
-        giq_option(4, 475, 753, Set_status_Attack_ALL, 50);
+        giq_option(4, NAME, 753, Set_status_Attack_ALL, NEUTRAL_REACTION);
     end
     if (map_var(8) != 0) then begin
-        giq_option(4, 475, 754, Set_status_Attack_DUDE, 50);
+        giq_option(4, NAME, 754, Set_status_Attack_DUDE, NEUTRAL_REACTION);
     end
-    if (global_var(927) == 1) then begin
-        giq_option(4, 475, 755, Node550, 50);
+    if (global_var(GVAR_ENEMY_OBRODA) == 1) then begin
+        giq_option(4, NAME, 755, Node550, NEUTRAL_REACTION);
     end
     else begin
-        giq_option(4, 475, 755, Node500, 50);
+        giq_option(4, NAME, 755, Node500, NEUTRAL_REACTION);
     end
-    giq_option(4, 475, 756, Node999, 50);
+    giq_option(4, NAME, 756, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node800
 begin
-    gsay_reply(475, 800);
-    set_global_var(963, global_var(963) bwand (-1 - 2));
-    giq_option(4, 475, 801, Node850, 50);
+    Reply(800);
+    set_global_var(GVAR_G_SEC_ROBOTS_ACTIVATION, global_var(GVAR_G_SEC_ROBOTS_ACTIVATION) bwand (-1 - 2));
+    giq_option(4, NAME, 801, Node850, NEUTRAL_REACTION);
     if (map_var(8) != 1) then begin
-        giq_option(4, 475, 802, Set_status_Attack_ALL, 50);
+        giq_option(4, NAME, 802, Set_status_Attack_ALL, NEUTRAL_REACTION);
     end
     if (map_var(8) != 0) then begin
-        giq_option(4, 475, 803, Set_status_Attack_DUDE, 50);
+        giq_option(4, NAME, 803, Set_status_Attack_DUDE, NEUTRAL_REACTION);
     end
-    if (global_var(927) == 1) then begin
-        giq_option(4, 475, 804, Node550, 50);
+    if (global_var(GVAR_ENEMY_OBRODA) == 1) then begin
+        giq_option(4, NAME, 804, Node550, NEUTRAL_REACTION);
     end
     else begin
-        giq_option(4, 475, 804, Node500, 50);
+        giq_option(4, NAME, 804, Node500, NEUTRAL_REACTION);
     end
-    giq_option(4, 475, 805, Node999, 50);
+    giq_option(4, NAME, 805, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node850
 begin
-    gsay_reply(475, 850);
-    set_global_var(963, global_var(963) bwor 2);
-    giq_option(4, 475, 851, Node800, 50);
+    Reply(850);
+    set_global_var(GVAR_G_SEC_ROBOTS_ACTIVATION, global_var(GVAR_G_SEC_ROBOTS_ACTIVATION) bwor 2);
+    giq_option(4, NAME, 851, Node800, NEUTRAL_REACTION);
     if (map_var(8) != 1) then begin
-        giq_option(4, 475, 852, Set_status_Attack_ALL, 50);
+        giq_option(4, NAME, 852, Set_status_Attack_ALL, NEUTRAL_REACTION);
     end
     if (map_var(8) != 0) then begin
-        giq_option(4, 475, 853, Set_status_Attack_DUDE, 50);
+        giq_option(4, NAME, 853, Set_status_Attack_DUDE, NEUTRAL_REACTION);
     end
-    if (global_var(927) == 1) then begin
-        giq_option(4, 475, 854, Node550, 50);
+    if (global_var(GVAR_ENEMY_OBRODA) == 1) then begin
+        giq_option(4, NAME, 854, Node550, NEUTRAL_REACTION);
     end
     else begin
-        giq_option(4, 475, 854, Node500, 50);
+        giq_option(4, NAME, 854, Node500, NEUTRAL_REACTION);
     end
-    giq_option(4, 475, 855, Node999, 50);
+    giq_option(4, NAME, 855, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node900
 begin
-    gsay_reply(475, 900);
-    if ((global_var(963) bwand 2) != 0) then begin
-        giq_option(4, 475, 901, Node800, 50);
+    Reply(900);
+    if ((global_var(GVAR_G_SEC_ROBOTS_ACTIVATION) bwand 2) != 0) then begin
+        giq_option(4, NAME, 901, Node800, NEUTRAL_REACTION);
     end
     else begin
-        giq_option(4, 475, 902, Node850, 50);
+        giq_option(4, NAME, 902, Node850, NEUTRAL_REACTION);
     end
     if (map_var(8) != 1) then begin
-        giq_option(4, 475, 903, Set_status_Attack_ALL, 50);
+        giq_option(4, NAME, 903, Set_status_Attack_ALL, NEUTRAL_REACTION);
     end
     if (map_var(8) != 0) then begin
-        giq_option(4, 475, 904, Set_status_Attack_DUDE, 50);
+        giq_option(4, NAME, 904, Set_status_Attack_DUDE, NEUTRAL_REACTION);
     end
-    if (global_var(927) == 1) then begin
-        giq_option(4, 475, 905, Node550, 50);
+    if (global_var(GVAR_ENEMY_OBRODA) == 1) then begin
+        giq_option(4, NAME, 905, Node550, NEUTRAL_REACTION);
     end
     else begin
-        giq_option(4, 475, 905, Node500, 50);
+        giq_option(4, NAME, 905, Node500, NEUTRAL_REACTION);
     end
-    giq_option(4, 475, 906, Node999, 50);
+    giq_option(4, NAME, 906, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node950
 begin
-    gsay_reply(475, 950);
+    Reply(950);
     set_local_var(0, 2);
-    if ((global_var(963) bwand 2) != 0) then begin
-        giq_option(4, 475, 951, Node1000, 50);
+    if ((global_var(GVAR_G_SEC_ROBOTS_ACTIVATION) bwand 2) != 0) then begin
+        giq_option(4, NAME, 951, Node1000, NEUTRAL_REACTION);
     end
     else begin
-        giq_option(4, 475, 952, Node1050, 50);
+        giq_option(4, NAME, 952, Node1050, NEUTRAL_REACTION);
     end
     if (map_var(8) != 1) then begin
-        giq_option(4, 475, 953, Set_status_Attack_ALL, 50);
+        giq_option(4, NAME, 953, Set_status_Attack_ALL, NEUTRAL_REACTION);
     end
     if (map_var(8) != 0) then begin
-        giq_option(4, 475, 954, Set_status_Attack_DUDE, 50);
+        giq_option(4, NAME, 954, Set_status_Attack_DUDE, NEUTRAL_REACTION);
     end
     if (map_var(8) != 2) then begin
-        giq_option(4, 475, 955, Set_status_Attack_OBRODA, 50);
+        giq_option(4, NAME, 955, Set_status_Attack_OBRODA, NEUTRAL_REACTION);
     end
-    if (global_var(927) == 1) then begin
-        giq_option(4, 475, 956, Node550, 50);
+    if (global_var(GVAR_ENEMY_OBRODA) == 1) then begin
+        giq_option(4, NAME, 956, Node550, NEUTRAL_REACTION);
     end
     else begin
-        giq_option(4, 475, 956, Node500, 50);
+        giq_option(4, NAME, 956, Node500, NEUTRAL_REACTION);
     end
-    giq_option(4, 475, 957, Node999, 50);
+    giq_option(4, NAME, 957, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node1000
 begin
-    gsay_reply(475, 1000);
-    set_global_var(963, global_var(963) bwand (-1 - 2));
-    giq_option(4, 475, 1001, Node1050, 50);
+    Reply(1000);
+    set_global_var(GVAR_G_SEC_ROBOTS_ACTIVATION, global_var(GVAR_G_SEC_ROBOTS_ACTIVATION) bwand (-1 - 2));
+    giq_option(4, NAME, 1001, Node1050, NEUTRAL_REACTION);
     if (map_var(8) != 1) then begin
-        giq_option(4, 475, 1002, Set_status_Attack_ALL, 50);
+        giq_option(4, NAME, 1002, Set_status_Attack_ALL, NEUTRAL_REACTION);
     end
     if (map_var(8) != 0) then begin
-        giq_option(4, 475, 1003, Set_status_Attack_DUDE, 50);
+        giq_option(4, NAME, 1003, Set_status_Attack_DUDE, NEUTRAL_REACTION);
     end
     if (map_var(8) != 2) then begin
-        giq_option(4, 475, 1004, Set_status_Attack_OBRODA, 50);
+        giq_option(4, NAME, 1004, Set_status_Attack_OBRODA, NEUTRAL_REACTION);
     end
-    if (global_var(927) == 1) then begin
-        giq_option(4, 475, 1005, Node550, 50);
+    if (global_var(GVAR_ENEMY_OBRODA) == 1) then begin
+        giq_option(4, NAME, 1005, Node550, NEUTRAL_REACTION);
     end
     else begin
-        giq_option(4, 475, 1005, Node500, 50);
+        giq_option(4, NAME, 1005, Node500, NEUTRAL_REACTION);
     end
-    giq_option(4, 475, 1006, Node999, 50);
+    giq_option(4, NAME, 1006, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node1050
 begin
-    gsay_reply(475, 1050);
-    set_global_var(963, global_var(963) bwor 2);
-    giq_option(4, 475, 1051, Node1000, 50);
+    Reply(1050);
+    set_global_var(GVAR_G_SEC_ROBOTS_ACTIVATION, global_var(GVAR_G_SEC_ROBOTS_ACTIVATION) bwor 2);
+    giq_option(4, NAME, 1051, Node1000, NEUTRAL_REACTION);
     if (map_var(8) != 1) then begin
-        giq_option(4, 475, 1052, Set_status_Attack_ALL, 50);
+        giq_option(4, NAME, 1052, Set_status_Attack_ALL, NEUTRAL_REACTION);
     end
     if (map_var(8) != 0) then begin
-        giq_option(4, 475, 1053, Set_status_Attack_DUDE, 50);
+        giq_option(4, NAME, 1053, Set_status_Attack_DUDE, NEUTRAL_REACTION);
     end
     if (map_var(8) != 2) then begin
-        giq_option(4, 475, 1054, Set_status_Attack_OBRODA, 50);
+        giq_option(4, NAME, 1054, Set_status_Attack_OBRODA, NEUTRAL_REACTION);
     end
-    if (global_var(927) == 1) then begin
-        giq_option(4, 475, 1055, Node550, 50);
+    if (global_var(GVAR_ENEMY_OBRODA) == 1) then begin
+        giq_option(4, NAME, 1055, Node550, NEUTRAL_REACTION);
     end
     else begin
-        giq_option(4, 475, 1055, Node500, 50);
+        giq_option(4, NAME, 1055, Node500, NEUTRAL_REACTION);
     end
-    giq_option(4, 475, 1056, Node999, 50);
+    giq_option(4, NAME, 1056, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node1100
 begin
-    gsay_reply(475, 1100);
-    if ((global_var(963) bwand 2) != 0) then begin
-        giq_option(4, 475, 1101, Node1000, 50);
+    Reply(1100);
+    if ((global_var(GVAR_G_SEC_ROBOTS_ACTIVATION) bwand 2) != 0) then begin
+        giq_option(4, NAME, 1101, Node1000, NEUTRAL_REACTION);
     end
     else begin
-        giq_option(4, 475, 1102, Node1050, 50);
+        giq_option(4, NAME, 1102, Node1050, NEUTRAL_REACTION);
     end
     if (map_var(8) != 1) then begin
-        giq_option(4, 475, 1103, Set_status_Attack_ALL, 50);
+        giq_option(4, NAME, 1103, Set_status_Attack_ALL, NEUTRAL_REACTION);
     end
     if (map_var(8) != 0) then begin
-        giq_option(4, 475, 1104, Set_status_Attack_DUDE, 50);
+        giq_option(4, NAME, 1104, Set_status_Attack_DUDE, NEUTRAL_REACTION);
     end
     if (map_var(8) != 2) then begin
-        giq_option(4, 475, 1105, Set_status_Attack_OBRODA, 50);
+        giq_option(4, NAME, 1105, Set_status_Attack_OBRODA, NEUTRAL_REACTION);
     end
-    if (global_var(927) == 1) then begin
-        giq_option(4, 475, 1106, Node550, 50);
+    if (global_var(GVAR_ENEMY_OBRODA) == 1) then begin
+        giq_option(4, NAME, 1106, Node550, NEUTRAL_REACTION);
     end
     else begin
-        giq_option(4, 475, 1106, Node500, 50);
+        giq_option(4, NAME, 1106, Node500, NEUTRAL_REACTION);
     end
-    giq_option(4, 475, 1107, Node999, 50);
+    giq_option(4, NAME, 1107, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node1150
 begin
-    gsay_reply(475, 1150);
-    set_global_var(927, 1);
-    giq_option(4, 475, 1151, Node550, 50);
-    giq_option(4, 475, 1152, Node999, 50);
+    Reply(1150);
+    set_global_var(GVAR_ENEMY_OBRODA, 1);
+    giq_option(4, NAME, 1151, Node550, NEUTRAL_REACTION);
+    giq_option(4, NAME, 1152, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node1250
 begin
-    gsay_reply(475, 1250);
-    giq_option(4, 475, 1251, Node550, 50);
-    giq_option(4, 475, 1252, Node999, 50);
+    Reply(1250);
+    giq_option(4, NAME, 1251, Node550, NEUTRAL_REACTION);
+    giq_option(4, NAME, 1252, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node1300
 begin
-    gsay_reply(475, 1300);
-    set_global_var(927, 1);
-    giq_option(4, 475, 1301, Node550, 50);
-    giq_option(4, 475, 1302, Node999, 50);
+    Reply(1300);
+    set_global_var(GVAR_ENEMY_OBRODA, 1);
+    giq_option(4, NAME, 1301, Node550, NEUTRAL_REACTION);
+    giq_option(4, NAME, 1302, Node999, NEUTRAL_REACTION);
 end
 
 procedure Set_status_Attack_ALL
@@ -536,4 +536,3 @@ begin
         call Node1100();
     end
 end
-
