diff --git a/Mods/Resurrection/Original/GCOM5MAS.ssl b/Mods/Resurrection/ReDefine/GCOM5MAS.ssl
index 2eb8c76..189b204 100644
--- a/Mods/Resurrection/Original/GCOM5MAS.ssl
+++ b/Mods/Resurrection/ReDefine/GCOM5MAS.ssl
@@ -67,12 +67,12 @@ begin
             return 1;
         end
     end
-    if ((party_member_obj(16777779) != 0) and ((global_var(910) bwand 32) == 0)) then begin
+    if ((party_member_obj(16777779) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 32) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777779)) <= 5) then begin
             return 1;
         end
     end
-    if ((party_member_obj(16777811) != 0) and ((global_var(910) bwand 64) == 0)) then begin
+    if ((party_member_obj(16777811) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 64) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777811)) <= 5) then begin
             return 1;
         end
@@ -87,7 +87,7 @@ begin
             return 1;
         end
     end
-    if ((party_member_obj(16777814) != 0) and ((global_var(910) bwand 128) == 0)) then begin
+    if ((party_member_obj(16777814) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 128) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777814)) <= 5) then begin
             return 1;
         end
@@ -146,16 +146,16 @@ procedure use_p_proc
 begin
     script_overrides;
     if (combat_is_initialized) then begin
-        display_msg(message_str(467, 300));
+        display_mstr(300);
     end
     else begin
         start_gdialog(467, self_obj, 4, -1, -1);
         gsay_start;
-        if (get_critter_stat(dude_obj, 4) < 4) then begin
+        if (dude_iq < 4) then begin
             call Node500();
         end
         else begin
-            if (global_var(953) == 0) then begin
+            if (global_var(GVAR_G_BASE_DESTROYED) == 0) then begin
                 if ((local_var(0) bwand 8) != 0) then begin
                     call Node750();
                 end
@@ -164,7 +164,7 @@ begin
                 end
             end
             else begin
-                if ((global_var(953) == 1) or (global_var(953) == 2)) then begin
+                if ((global_var(GVAR_G_BASE_DESTROYED) == 1) or (global_var(GVAR_G_BASE_DESTROYED) == 2)) then begin
                     call Node950();
                 end
                 else begin
@@ -180,13 +180,13 @@ end
 procedure look_at_p_proc
 begin
     script_overrides;
-    display_msg(message_str(467, 100));
+    display_mstr(100);
 end
 
 procedure description_p_proc
 begin
     script_overrides;
-    display_msg(message_str(467, 200));
+    display_mstr(200);
 end
 
 procedure use_skill_on_p_proc
@@ -204,14 +204,14 @@ end
 procedure timed_event_p_proc
 begin
     if (fixed_param == 1) then begin
-        if (global_var(953) == 0) then begin
-            set_global_var(953, 4);
+        if (global_var(GVAR_G_BASE_DESTROYED) == 0) then begin
+            set_global_var(GVAR_G_BASE_DESTROYED, 4);
             add_timer_event(ghoul46_map, 10, 12);
         end
     end
     if (fixed_param == 2) then begin
-        if (global_var(953) == 0) then begin
-            set_global_var(953, 3);
+        if (global_var(GVAR_G_BASE_DESTROYED) == 0) then begin
+            set_global_var(GVAR_G_BASE_DESTROYED, 3);
             add_timer_event(ghoul46_map, 10, 8);
         end
     end
@@ -233,146 +233,146 @@ end
 
 procedure Node500
 begin
-    gsay_reply(467, 500);
+    Reply(500);
     if ((local_var(0) bwand 8) != 0) then begin
         if ((local_var(0) bwand 32) == 0) then begin
-            giq_option(-3, 467, 502, Node1200, 50);
+            giq_option(-3, NAME, 502, Node1200, NEUTRAL_REACTION);
         end
         else begin
             if ((local_var(0) bwand 64) == 0) then begin
-                giq_option(-3, 467, 502, Node1250, 50);
+                giq_option(-3, NAME, 502, Node1250, NEUTRAL_REACTION);
             end
             else begin
-                if (global_var(953) == 0) then begin
-                    giq_option(-3, 467, 502, Node1300, 50);
+                if (global_var(GVAR_G_BASE_DESTROYED) == 0) then begin
+                    giq_option(-3, NAME, 502, Node1300, NEUTRAL_REACTION);
                 end
             end
         end
     end
     else begin
-        giq_option(-3, 467, 502, Node1050, 50);
+        giq_option(-3, NAME, 502, Node1050, NEUTRAL_REACTION);
     end
-    giq_option(-3, 467, 501, Node999, 50);
+    giq_option(-3, NAME, 501, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node550
 begin
-    gsay_reply(467, 550);
+    Reply(550);
     if (obj_is_carrying_obj_pid(dude_obj, 221)) then begin
-        giq_option(4, 467, 551, Node600, 50);
+        giq_option(4, NAME, 551, Node600, NEUTRAL_REACTION);
     end
-    giq_option(4, 467, 552, Node999, 50);
+    giq_option(4, NAME, 552, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node600
 begin
-    gsay_reply(467, 600);
-    giq_option(4, 467, 601, Node800, 50);
-    giq_option(4, 467, 602, Node650, 50);
-    giq_option(4, 467, 603, Node999, 50);
+    Reply(600);
+    giq_option(4, NAME, 601, Node800, NEUTRAL_REACTION);
+    giq_option(4, NAME, 602, Node650, NEUTRAL_REACTION);
+    giq_option(4, NAME, 603, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node650
 begin
-    gsay_reply(467, 650);
-    giq_option(4, 467, (5 * 60 * 10 / 60 / 10) + message_str(467, 651), silent_autodestruct, 50);
-    giq_option(4, 467, (5 * 60 * 10 / 60 / 10) + message_str(467, 652), noise_autodestruct, 50);
-    giq_option(4, 467, 653, Node750, 50);
-    giq_option(4, 467, 654, Node999, 50);
+    Reply(650);
+    giq_option(4, NAME, (5 * 60 * 10 / 60 / 10) + mstr(651), silent_autodestruct, NEUTRAL_REACTION);
+    giq_option(4, NAME, (5 * 60 * 10 / 60 / 10) + mstr(652), noise_autodestruct, NEUTRAL_REACTION);
+    giq_option(4, NAME, 653, Node750, NEUTRAL_REACTION);
+    giq_option(4, NAME, 654, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node700
 begin
-    gsay_reply(467, 700);
-    giq_option(4, 467, 701, Start_base_destraction, 50);
-    giq_option(4, 467, 702, Node750, 50);
+    Reply(700);
+    giq_option(4, NAME, 701, Start_base_destraction, NEUTRAL_REACTION);
+    giq_option(4, NAME, 702, Node750, NEUTRAL_REACTION);
 end
 
 procedure Node750
 begin
-    gsay_reply(467, 750);
+    Reply(750);
     if ((local_var(0) bwand 1) != 0) then begin
         set_local_var(0, local_var(0) bwand (-1 - 1));
     end
     if ((local_var(0) bwand 2) != 0) then begin
         set_local_var(0, local_var(0) bwand (-1 - 2));
     end
-    giq_option(4, 467, 751, Node800, 50);
-    giq_option(4, 467, 752, Node650, 50);
-    giq_option(4, 467, 753, Node999, 50);
+    giq_option(4, NAME, 751, Node800, NEUTRAL_REACTION);
+    giq_option(4, NAME, 752, Node650, NEUTRAL_REACTION);
+    giq_option(4, NAME, 753, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node800
 begin
-    gsay_reply(467, 800);
-    giq_option(4, 467, 804, Node800b, 50);
-    giq_option(4, 467, 801, Node750, 50);
-    giq_option(4, 467, 802, Node999, 50);
+    Reply(800);
+    giq_option(4, NAME, 804, Node800b, NEUTRAL_REACTION);
+    giq_option(4, NAME, 801, Node750, NEUTRAL_REACTION);
+    giq_option(4, NAME, 802, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node800b
 begin
-    gsay_reply(467, 805);
-    giq_option(4, 467, 804, Node800c, 50);
-    giq_option(4, 467, 803, Node800, 50);
-    giq_option(4, 467, 801, Node750, 50);
-    giq_option(4, 467, 802, Node999, 50);
+    Reply(805);
+    giq_option(4, NAME, 804, Node800c, NEUTRAL_REACTION);
+    giq_option(4, NAME, 803, Node800, NEUTRAL_REACTION);
+    giq_option(4, NAME, 801, Node750, NEUTRAL_REACTION);
+    giq_option(4, NAME, 802, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node800c
 begin
-    gsay_reply(467, 806);
-    giq_option(4, 467, 803, Node800b, 50);
-    giq_option(4, 467, 801, Node750, 50);
-    giq_option(4, 467, 802, Node999, 50);
+    Reply(806);
+    giq_option(4, NAME, 803, Node800b, NEUTRAL_REACTION);
+    giq_option(4, NAME, 801, Node750, NEUTRAL_REACTION);
+    giq_option(4, NAME, 802, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node850
 begin
-    gsay_reply(467, message_str(467, 850) + ((global_var(954) - game_time) / 10) + message_str(467, 851));
-    giq_option(4, 467, 852, Node900, 50);
-    giq_option(4, 467, 853, Node999, 50);
+    Reply(mstr(850) + ((global_var(GVAR_G_BASE_TIME_OF_DESTRUCTION) - game_time) / 10) + mstr(851));
+    giq_option(4, NAME, 852, Node900, NEUTRAL_REACTION);
+    giq_option(4, NAME, 853, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node900
 begin
-    gsay_reply(467, 900);
-    giq_option(4, 467, 901, Node999, 50);
+    Reply(900);
+    giq_option(4, NAME, 901, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node950
 begin
-    gsay_reply(467, message_str(467, 950) + ((global_var(954) - game_time) / 10) + message_str(467, 951));
+    Reply(mstr(950) + ((global_var(GVAR_G_BASE_TIME_OF_DESTRUCTION) - game_time) / 10) + mstr(951));
     if ((local_var(0) bwand 4) == 0) then begin
-        giq_option(4, 467, 952, Node1000, 50);
+        giq_option(4, NAME, 952, Node1000, NEUTRAL_REACTION);
     end
-    giq_option(4, 467, 953, Node999, 50);
+    giq_option(4, NAME, 953, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node1000
 begin
-    gsay_reply(467, 1000);
+    Reply(1000);
     set_local_var(0, local_var(0) bwor 4);
-    giq_option(4, 467, 1001, Node999, 50);
+    giq_option(4, NAME, 1001, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node1050
 begin
     if (obj_is_carrying_obj_pid(dude_obj, 221)) then begin
-        gsay_reply(467, 1051);
+        Reply(1051);
     end
     else begin
-        gsay_reply(467, 1050);
+        Reply(1050);
     end
     if (obj_is_carrying_obj_pid(dude_obj, 221)) then begin
-        giq_option(-3, 467, 1052, Node1100, 50);
+        giq_option(-3, NAME, 1052, Node1100, NEUTRAL_REACTION);
     end
-    giq_option(-3, 467, 1053, Node999, 50);
+    giq_option(-3, NAME, 1053, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node1100
 begin
-    gsay_reply(467, 1100 + (get_critter_stat(dude_obj, 34) == 1));
+    Reply(1100 + (dude_gender == 1));
     set_local_var(0, local_var(0) bwor 8);
     removed_qty := obj_is_carrying_obj_pid(dude_obj, 221);
     if (1 < removed_qty) then begin
@@ -382,17 +382,17 @@ begin
         restock_obj := obj_carrying_pid_obj(dude_obj, 221);
         if (obj_type(self_obj) == 1) then begin
             restock_obj := obj_carrying_pid_obj(dude_obj, 221);
-            if (critter_inven_obj(self_obj, 0) == restock_obj) then begin
-                if (obj_item_subtype(critter_inven_obj(dude_obj, 0)) == 0) then begin
-                    restock_obj := critter_inven_obj(dude_obj, 0);
+            if (critter_inven_obj(self_obj, INVEN_TYPE_WORN) == restock_obj) then begin
+                if (obj_item_subtype(critter_inven_obj(dude_obj, INVEN_TYPE_WORN)) == 0) then begin
+                    restock_obj := critter_inven_obj(dude_obj, INVEN_TYPE_WORN);
                     debug_msg("armour pid == " + obj_pid(restock_obj));
                     rm_obj_from_inven(dude_obj, restock_obj);
                     add_obj_to_inven(dude_obj, restock_obj);
                 end
             end
             else begin
-                if ((critter_inven_obj(dude_obj, 2) == restock_obj) or (critter_inven_obj(dude_obj, 1) == restock_obj)) then begin
-                    metarule(43, dude_obj);
+                if ((critter_inven_obj(dude_obj, INVEN_TYPE_LEFT_HAND) == restock_obj) or (critter_inven_obj(dude_obj, INVEN_TYPE_RIGHT_HAND) == restock_obj)) then begin
+                    inven_unwield(dude_obj);
                 end
             end
         end
@@ -412,71 +412,71 @@ begin
     end
     restock_obj := 0;
     restock_amt := 0;
-    giq_option(-3, 467, 1102, Node1200, 50);
+    giq_option(-3, NAME, 1102, Node1200, NEUTRAL_REACTION);
     if ((local_var(0) bwand 16) == 0) then begin
-        giq_option(-3, 467, 1103, Node1150, 50);
+        giq_option(-3, NAME, 1103, Node1150, NEUTRAL_REACTION);
     end
-    giq_option(-3, 467, 1104, Node999, 50);
+    giq_option(-3, NAME, 1104, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node1150
 begin
-    gsay_reply(467, 1150);
+    Reply(1150);
     set_local_var(0, local_var(0) bwor 16);
     if ((local_var(0) bwand 32) == 0) then begin
-        giq_option(-3, 467, 1151, Node1200, 50);
+        giq_option(-3, NAME, 1151, Node1200, NEUTRAL_REACTION);
     end
     else begin
         if ((local_var(0) bwand 64) == 0) then begin
-            giq_option(-3, 467, 1151, Node1250, 50);
+            giq_option(-3, NAME, 1151, Node1250, NEUTRAL_REACTION);
         end
         else begin
-            giq_option(-3, 467, 1151, Node1300, 50);
+            giq_option(-3, NAME, 1151, Node1300, NEUTRAL_REACTION);
         end
     end
-    giq_option(-3, 467, 1152, Node999, 50);
+    giq_option(-3, NAME, 1152, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node1200
 begin
-    gsay_reply(467, 1200);
+    Reply(1200);
     set_local_var(0, local_var(0) bwor 32);
-    giq_option(-3, 467, 1201, Node1250, 50);
+    giq_option(-3, NAME, 1201, Node1250, NEUTRAL_REACTION);
     if ((local_var(0) bwand 16) == 0) then begin
-        giq_option(-3, 467, 1202, Node1150, 50);
+        giq_option(-3, NAME, 1202, Node1150, NEUTRAL_REACTION);
     end
-    giq_option(-3, 467, 1203, Node999, 50);
+    giq_option(-3, NAME, 1203, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node1250
 begin
-    gsay_reply(467, 1250);
+    Reply(1250);
     set_local_var(0, local_var(0) bwor 64);
-    giq_option(-3, 467, 1251, Node1300, 50);
+    giq_option(-3, NAME, 1251, Node1300, NEUTRAL_REACTION);
     if ((local_var(0) bwand 16) == 0) then begin
-        giq_option(-3, 467, 1252, Node1150, 50);
+        giq_option(-3, NAME, 1252, Node1150, NEUTRAL_REACTION);
     end
-    giq_option(-3, 467, 1253, Node999, 50);
+    giq_option(-3, NAME, 1253, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node1300
 begin
-    gsay_reply(467, 1300);
-    giq_option(-3, 467, 1301, Node1400, 50);
-    giq_option(-3, 467, 1302, Node1350, 50);
+    Reply(1300);
+    giq_option(-3, NAME, 1301, Node1400, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 1302, Node1350, NEUTRAL_REACTION);
 end
 
 procedure Node1350
 begin
-    gsay_reply(467, 1350 + (get_critter_stat(dude_obj, 34) == 1));
-    giq_option(-3, 467, 1352, Node999, 50);
+    Reply(1350 + (dude_gender == 1));
+    giq_option(-3, NAME, 1352, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node1400
 begin
-    gsay_reply(467, message_str(467, 1400) + (5 * 60 * 10 / 10) + message_str(467, 1401));
+    Reply(mstr(1400) + (5 * 60 * 10 / 10) + mstr(1401));
     set_local_var(0, local_var(0) bwor 2);
-    giq_option(-3, 467, 1402, Start_base_destraction, 50);
+    giq_option(-3, NAME, 1402, Start_base_destraction, NEUTRAL_REACTION);
 end
 
 procedure silent_autodestruct
@@ -490,4 +490,3 @@ begin
     set_local_var(0, local_var(0) bwor 2);
     call Node700();
 end
-
