diff --git a/Mods/Resurrection/Original/GCOM3FEA.ssl b/Mods/Resurrection/ReDefine/GCOM3FEA.ssl
index 791d44c..5f2c312 100644
--- a/Mods/Resurrection/Original/GCOM3FEA.ssl
+++ b/Mods/Resurrection/ReDefine/GCOM3FEA.ssl
@@ -52,12 +52,12 @@ begin
             return 1;
         end
     end
-    if ((party_member_obj(16777779) != 0) and ((global_var(910) bwand 32) == 0)) then begin
+    if ((party_member_obj(16777779) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 32) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777779)) <= 5) then begin
             return 1;
         end
     end
-    if ((party_member_obj(16777811) != 0) and ((global_var(910) bwand 64) == 0)) then begin
+    if ((party_member_obj(16777811) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 64) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777811)) <= 5) then begin
             return 1;
         end
@@ -72,7 +72,7 @@ begin
             return 1;
         end
     end
-    if ((party_member_obj(16777814) != 0) and ((global_var(910) bwand 128) == 0)) then begin
+    if ((party_member_obj(16777814) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 128) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777814)) <= 5) then begin
             return 1;
         end
@@ -131,7 +131,7 @@ procedure use_p_proc
 begin
     script_overrides;
     if (combat_is_initialized) then begin
-        display_msg(message_str(404, 300));
+        display_mstr(300);
     end
     else begin
         start_gdialog(404, self_obj, 4, -1, -1);
@@ -145,13 +145,13 @@ end
 procedure look_at_p_proc
 begin
     script_overrides;
-    display_msg(message_str(404, 100));
+    display_mstr(100);
 end
 
 procedure description_p_proc
 begin
     script_overrides;
-    display_msg(message_str(404, 200));
+    display_mstr(200);
 end
 
 procedure timed_event_p_proc
@@ -178,96 +178,95 @@ end
 
 procedure Node999_Alarm
 begin
-    animate_move_obj_to_tile(Feargus, tile_num_in_direction(tile_num(dude_obj), (has_trait(1, dude_obj, 10) + 3) % 6, 1), 1);
+    animate_move_obj_to_tile(Feargus, tile_num_in_direction(dude_tile, (obj_get_rot(dude_obj) + 3) % 6, 1), 1);
 end
 
 procedure Node500
 begin
-    gsay_reply(404, message_str(404, 500) + message_str(404, 501) + message_str(404, 502));
-    if (has_skill(dude_obj, 12) > 105) then begin
-        giq_option(4, 404, 503, Node550, 50);
+    Reply(mstr(500) + mstr(501) + mstr(502));
+    if (dude_skill(SKILL_SCIENCE) > 105) then begin
+        giq_option(4, NAME, 503, Node550, NEUTRAL_REACTION);
     end
     else begin
-        giq_option(4, 404, 503, Node600, 50);
+        giq_option(4, NAME, 503, Node600, NEUTRAL_REACTION);
     end
-    giq_option(4, 404, 504, Node999, 50);
-    giq_option(-3, 404, 505, Node999, 50);
+    giq_option(4, NAME, 504, Node999, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 505, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node550
 begin
-    gsay_reply(404, 550);
-    if ((global_var(936) == 3) and ((local_var(4) bwand 1) == 0)) then begin
-        giq_option(4, 404, 551, Node650, 50);
+    Reply(550);
+    if ((global_var(GVAR_G_MAX_REGISTRATION) == 3) and ((local_var(4) bwand 1) == 0)) then begin
+        giq_option(4, NAME, 551, Node650, NEUTRAL_REACTION);
     end
     if ((local_var(4) bwand 2) == 0) then begin
-        if ((has_skill(dude_obj, 12) > 120) and (has_skill(dude_obj, 13) > 120)) then begin
-            giq_option(4, 404, 552, Node850, 50);
+        if ((dude_skill(SKILL_SCIENCE) > 120) and (dude_skill(SKILL_REPAIR) > 120)) then begin
+            giq_option(4, NAME, 552, Node850, NEUTRAL_REACTION);
         end
         else begin
-            giq_option(4, 404, 552, Node800, 50);
+            giq_option(4, NAME, 552, Node800, NEUTRAL_REACTION);
         end
     end
-    giq_option(4, 404, 553, Node999, 50);
+    giq_option(4, NAME, 553, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node600
 begin
-    gsay_reply(404, 600);
-    set_global_var(927, 1);
-    giq_option(4, 404, 601, Node999_Alarm, 50);
+    Reply(600);
+    set_global_var(GVAR_ENEMY_OBRODA, 1);
+    giq_option(4, NAME, 601, Node999_Alarm, NEUTRAL_REACTION);
 end
 
 procedure Node650
 begin
-    gsay_reply(404, 650);
+    Reply(650);
     set_local_var(4, local_var(4) bwor 1);
-    giq_option(4, 404, 651, Node700, 50);
+    giq_option(4, NAME, 651, Node700, NEUTRAL_REACTION);
 end
 
 procedure Node700
 begin
-    gsay_reply(404, 700);
-    giq_option(4, 404, 701, Node750, 50);
+    Reply(700);
+    giq_option(4, NAME, 701, Node750, NEUTRAL_REACTION);
 end
 
 procedure Node750
 begin
-    gsay_reply(404, 750);
+    Reply(750);
     set_map_var(2, 1);
-    restock_obj := create_object_sid(550, 0, 0, -1);
+    restock_obj := create_object(550, 0, 0);
     add_mult_objs_to_inven(dude_obj, restock_obj, 1);
     give_exp_points(500);
-    display_msg(message_str(14, 100) + 500 + message_str(14, 101));
+    display_msg(g_mstr(100) + 500 + g_mstr(101));
     if ((local_var(4) bwand 2) == 0) then begin
-        if ((has_skill(dude_obj, 12) > 120) and (has_skill(dude_obj, 13) > 120)) then begin
-            giq_option(4, 404, 751, Node850, 50);
+        if ((dude_skill(SKILL_SCIENCE) > 120) and (dude_skill(SKILL_REPAIR) > 120)) then begin
+            giq_option(4, NAME, 751, Node850, NEUTRAL_REACTION);
         end
         else begin
-            giq_option(4, 404, 751, Node800, 50);
+            giq_option(4, NAME, 751, Node800, NEUTRAL_REACTION);
         end
     end
-    giq_option(4, 404, 752, Node999, 50);
+    giq_option(4, NAME, 752, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node800
 begin
-    gsay_reply(404, 800 + (get_critter_stat(dude_obj, 34) == 1));
+    Reply(800 + (dude_gender == 1));
     set_local_var(4, local_var(4) bwor 2);
-    if ((global_var(936) == 3) and ((local_var(4) bwand 1) == 0)) then begin
-        giq_option(4, 404, 802, Node650, 50);
+    if ((global_var(GVAR_G_MAX_REGISTRATION) == 3) and ((local_var(4) bwand 1) == 0)) then begin
+        giq_option(4, NAME, 802, Node650, NEUTRAL_REACTION);
     end
-    giq_option(4, 404, 803, Node999, 50);
+    giq_option(4, NAME, 803, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node850
 begin
-    gsay_reply(404, 850 + (get_critter_stat(dude_obj, 34) == 1));
+    Reply(850 + (dude_gender == 1));
     set_local_var(4, local_var(4) bwor 2);
-    critter_mod_skill(dude_obj, 2, 5);
-    if ((global_var(936) == 3) and ((local_var(4) bwand 1) == 0)) then begin
-        giq_option(4, 404, 852, Node650, 50);
+    critter_mod_skill(dude_obj, SKILL_ENERGY_WEAPONS, 5);
+    if ((global_var(GVAR_G_MAX_REGISTRATION) == 3) and ((local_var(4) bwand 1) == 0)) then begin
+        giq_option(4, NAME, 852, Node650, NEUTRAL_REACTION);
     end
-    giq_option(4, 404, 853, Node999, 50);
+    giq_option(4, NAME, 853, Node999, NEUTRAL_REACTION);
 end
-
