diff --git a/Mods/Resurrection/Original/GCOM1LEO.ssl b/Mods/Resurrection/ReDefine/GCOM1LEO.ssl
index 5861873..b6c1933 100644
--- a/Mods/Resurrection/Original/GCOM1LEO.ssl
+++ b/Mods/Resurrection/ReDefine/GCOM1LEO.ssl
@@ -79,12 +79,12 @@ begin
             return 1;
         end
     end
-    if ((party_member_obj(16777779) != 0) and ((global_var(910) bwand 32) == 0)) then begin
+    if ((party_member_obj(16777779) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 32) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777779)) <= 5) then begin
             return 1;
         end
     end
-    if ((party_member_obj(16777811) != 0) and ((global_var(910) bwand 64) == 0)) then begin
+    if ((party_member_obj(16777811) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 64) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777811)) <= 5) then begin
             return 1;
         end
@@ -99,7 +99,7 @@ begin
             return 1;
         end
     end
-    if ((party_member_obj(16777814) != 0) and ((global_var(910) bwand 128) == 0)) then begin
+    if ((party_member_obj(16777814) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 128) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777814)) <= 5) then begin
             return 1;
         end
@@ -158,12 +158,12 @@ procedure use_p_proc
 begin
     script_overrides;
     if (combat_is_initialized) then begin
-        display_msg(message_str(394, 300));
+        display_mstr(300);
     end
     else begin
         start_gdialog(394, self_obj, 4, -1, -1);
         gsay_start;
-        if (get_critter_stat(dude_obj, 4) <= 3) then begin
+        if (dude_is_stupid) then begin
             call Node500();
         end
         else begin
@@ -177,13 +177,13 @@ end
 procedure look_at_p_proc
 begin
     script_overrides;
-    display_msg(message_str(394, 100));
+    display_mstr(100);
 end
 
 procedure description_p_proc
 begin
     script_overrides;
-    display_msg(message_str(394, 200));
+    display_mstr(200);
 end
 
 procedure timed_event_p_proc
@@ -210,382 +210,381 @@ end
 
 procedure Node500
 begin
-    gsay_reply(394, 500);
-    giq_option(-3, 394, 501, Node999, 50);
+    Reply(500);
+    giq_option(-3, NAME, 501, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node550
 begin
-    gsay_reply(394, 550);
-    if (global_var(595) != 1) then begin
-        giq_option(4, 394, 551, Download_holo_VT_base, 50);
+    Reply(550);
+    if (global_var(GVAR_NCR_SPY_HOLO_DOWNLOAD) != 1) then begin
+        giq_option(4, NAME, 551, Download_holo_VT_base, NEUTRAL_REACTION);
     end
-    if (global_var(596) != 1) then begin
-        giq_option(4, 394, 552, Download_holo_ultrastimpack, 50);
+    if (global_var(GVAR_NCR_HISTORY_HOLO_DOWNLOAD) != 1) then begin
+        giq_option(4, NAME, 552, Download_holo_ultrastimpack, NEUTRAL_REACTION);
     end
     if ((local_var(4) bwand 1) == 0) then begin
-        giq_option(4, 394, 553, Node650, 50);
+        giq_option(4, NAME, 553, Node650, NEUTRAL_REACTION);
     end
-    giq_option(4, 394, 554, Node1300, 50);
-    giq_option(4, 394, 555, Node999, 50);
+    giq_option(4, NAME, 554, Node1300, NEUTRAL_REACTION);
+    giq_option(4, NAME, 555, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node600
 begin
-    gsay_reply(394, 600);
-    if (global_var(595) != 1) then begin
-        giq_option(4, 394, 601, Download_holo_VT_base, 50);
+    Reply(600);
+    if (global_var(GVAR_NCR_SPY_HOLO_DOWNLOAD) != 1) then begin
+        giq_option(4, NAME, 601, Download_holo_VT_base, NEUTRAL_REACTION);
     end
-    if (global_var(596) != 1) then begin
-        giq_option(4, 394, 602, Download_holo_ultrastimpack, 50);
+    if (global_var(GVAR_NCR_HISTORY_HOLO_DOWNLOAD) != 1) then begin
+        giq_option(4, NAME, 602, Download_holo_ultrastimpack, NEUTRAL_REACTION);
     end
     if ((local_var(4) bwand 1) == 0) then begin
-        giq_option(4, 394, 603, Node650, 50);
+        giq_option(4, NAME, 603, Node650, NEUTRAL_REACTION);
     end
-    giq_option(4, 394, 604, Node1300, 50);
-    giq_option(4, 394, 605, Node999, 50);
+    giq_option(4, NAME, 604, Node1300, NEUTRAL_REACTION);
+    giq_option(4, NAME, 605, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node650
 begin
-    gsay_reply(394, 650);
+    Reply(650);
     set_local_var(4, local_var(4) bwor 1);
-    if (get_critter_stat(dude_obj, 34) == 0) then begin
-        giq_option(4, 394, 651, Node700, 50);
+    if (dude_gender == 0) then begin
+        giq_option(4, NAME, 651, Node700, NEUTRAL_REACTION);
     end
-    if (get_critter_stat(dude_obj, 34) == 1) then begin
-        giq_option(4, 394, 651, Node1000, 50);
+    if (dude_gender == 1) then begin
+        giq_option(4, NAME, 651, Node1000, NEUTRAL_REACTION);
     end
 end
 
 procedure Node700
 begin
-    gsay_reply(394, 700);
-    giq_option(4, 394, 701, Node750, 50);
+    Reply(700);
+    giq_option(4, NAME, 701, Node750, NEUTRAL_REACTION);
 end
 
 procedure Node750
 begin
-    gsay_reply(394, 750);
-    giq_option(4, 394, 751, Node800, 50);
+    Reply(750);
+    giq_option(4, NAME, 751, Node800, NEUTRAL_REACTION);
 end
 
 procedure Node800
 begin
-    gsay_reply(394, 800);
-    giq_option(4, 394, 801, Node850, 50);
+    Reply(800);
+    giq_option(4, NAME, 801, Node850, NEUTRAL_REACTION);
 end
 
 procedure Node850
 begin
-    gsay_reply(394, 850);
-    giq_option(4, 394, 851, Node900, 50);
+    Reply(850);
+    giq_option(4, NAME, 851, Node900, NEUTRAL_REACTION);
 end
 
 procedure Node900
 begin
-    gsay_reply(394, 900);
-    giq_option(4, 394, 901, Node950, 50);
+    Reply(900);
+    giq_option(4, NAME, 901, Node950, NEUTRAL_REACTION);
 end
 
 procedure Node950
 begin
-    gsay_reply(394, 950);
+    Reply(950);
     give_exp_points(500);
-    display_msg(message_str(14, 100) + 500 + message_str(14, 101));
-    giq_option(4, 394, 951, Node550, 50);
-    giq_option(4, 394, 952, Node999, 50);
+    display_msg(g_mstr(100) + 500 + g_mstr(101));
+    giq_option(4, NAME, 951, Node550, NEUTRAL_REACTION);
+    giq_option(4, NAME, 952, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node1000
 begin
-    gsay_reply(394, 1000);
-    giq_option(4, 394, 1001, Node1050, 50);
+    Reply(1000);
+    giq_option(4, NAME, 1001, Node1050, NEUTRAL_REACTION);
 end
 
 procedure Node1050
 begin
-    gsay_reply(394, 1050);
-    giq_option(4, 394, 1051, Node1100, 50);
+    Reply(1050);
+    giq_option(4, NAME, 1051, Node1100, NEUTRAL_REACTION);
 end
 
 procedure Node1100
 begin
-    gsay_reply(394, 1100);
-    giq_option(4, 394, 1101, Node1150, 50);
+    Reply(1100);
+    giq_option(4, NAME, 1101, Node1150, NEUTRAL_REACTION);
 end
 
 procedure Node1150
 begin
-    gsay_reply(394, 1150);
-    giq_option(4, 394, 1151, Node1200, 50);
+    Reply(1150);
+    giq_option(4, NAME, 1151, Node1200, NEUTRAL_REACTION);
 end
 
 procedure Node1200
 begin
-    gsay_reply(394, 1200);
-    giq_option(4, 394, 1201, Node1250, 50);
+    Reply(1200);
+    giq_option(4, NAME, 1201, Node1250, NEUTRAL_REACTION);
 end
 
 procedure Node1250
 begin
-    gsay_reply(394, 1250);
+    Reply(1250);
     give_exp_points(500);
-    display_msg(message_str(14, 100) + 500 + message_str(14, 101));
-    giq_option(4, 394, 1251, Node550, 50);
-    giq_option(4, 394, 1252, Node999, 50);
+    display_msg(g_mstr(100) + 500 + g_mstr(101));
+    giq_option(4, NAME, 1251, Node550, NEUTRAL_REACTION);
+    giq_option(4, NAME, 1252, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node1300
 begin
-    gsay_reply(394, 1300);
-    if (has_skill(dude_obj, 12) > 100) then begin
-        giq_option(4, 394, 1301, Node1400, 50);
+    Reply(1300);
+    if (dude_skill(SKILL_SCIENCE) > 100) then begin
+        giq_option(4, NAME, 1301, Node1400, NEUTRAL_REACTION);
     end
     else begin
-        giq_option(4, 394, 1301, Node1350, 50);
+        giq_option(4, NAME, 1301, Node1350, NEUTRAL_REACTION);
     end
-    giq_option(4, 394, 1302, Node550, 50);
-    giq_option(4, 394, 1303, Node999, 50);
+    giq_option(4, NAME, 1302, Node550, NEUTRAL_REACTION);
+    giq_option(4, NAME, 1303, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node1350
 begin
-    gsay_reply(394, 1350);
-    set_global_var(927, 1);
-    giq_option(4, 394, 1351, Node999, 50);
+    Reply(1350);
+    set_global_var(GVAR_ENEMY_OBRODA, 1);
+    giq_option(4, NAME, 1351, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node1400
 begin
-    gsay_reply(394, 1400);
-    if (global_var(936) == 0) then begin
-        giq_option(4, 394, 1401, Node1450, 50);
+    Reply(1400);
+    if (global_var(GVAR_G_MAX_REGISTRATION) == 0) then begin
+        giq_option(4, NAME, 1401, Node1450, NEUTRAL_REACTION);
     end
-    if (global_var(936) == 1) then begin
-        giq_option(4, 394, 1402, Node1950, 50);
+    if (global_var(GVAR_G_MAX_REGISTRATION) == 1) then begin
+        giq_option(4, NAME, 1402, Node1950, NEUTRAL_REACTION);
     end
-    if (global_var(936) == 2) then begin
-        giq_option(4, 394, 1402, Node2000, 50);
+    if (global_var(GVAR_G_MAX_REGISTRATION) == 2) then begin
+        giq_option(4, NAME, 1402, Node2000, NEUTRAL_REACTION);
     end
-    if (global_var(936) == 3) then begin
-        giq_option(4, 394, 1402, Node2050, 50);
+    if (global_var(GVAR_G_MAX_REGISTRATION) == 3) then begin
+        giq_option(4, NAME, 1402, Node2050, NEUTRAL_REACTION);
     end
-    giq_option(4, 394, 1403, Node999, 50);
+    giq_option(4, NAME, 1403, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node1450
 begin
-    gsay_reply(394, 1450);
-    set_global_var(936, 1);
-    giq_option(4, 394, 1451, Node1500, 50);
+    Reply(1450);
+    set_global_var(GVAR_G_MAX_REGISTRATION, 1);
+    giq_option(4, NAME, 1451, Node1500, NEUTRAL_REACTION);
 end
 
 procedure Node1500
 begin
-    gsay_reply(394, 1500);
-    giq_option(4, 394, 1501, Node1550, 50);
-    giq_option(4, 394, 1502, Node999, 50);
+    Reply(1500);
+    giq_option(4, NAME, 1501, Node1550, NEUTRAL_REACTION);
+    giq_option(4, NAME, 1502, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node1550
 begin
-    gsay_reply(394, 1550);
-    if (has_skill(dude_obj, 12) > 110) then begin
-        giq_option(4, 394, 1551, Node1600, 50);
+    Reply(1550);
+    if (dude_skill(SKILL_SCIENCE) > 110) then begin
+        giq_option(4, NAME, 1551, Node1600, NEUTRAL_REACTION);
     end
     else begin
-        giq_option(4, 394, 1551, Node1350, 50);
+        giq_option(4, NAME, 1551, Node1350, NEUTRAL_REACTION);
     end
-    giq_option(4, 394, 1552, Node1950, 50);
-    giq_option(4, 394, 1553, Node999, 50);
+    giq_option(4, NAME, 1552, Node1950, NEUTRAL_REACTION);
+    giq_option(4, NAME, 1553, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node1600
 begin
-    gsay_reply(394, 1600);
-    set_global_var(936, 2);
+    Reply(1600);
+    set_global_var(GVAR_G_MAX_REGISTRATION, 2);
     give_exp_points(500);
-    display_msg(message_str(14, 100) + 500 + message_str(14, 101));
-    giq_option(4, 394, 1601, Node1650, 50);
-    if (global_var(937) == 0) then begin
-        giq_option(4, 394, 1602, Node1750, 50);
+    display_msg(g_mstr(100) + 500 + g_mstr(101));
+    giq_option(4, NAME, 1601, Node1650, NEUTRAL_REACTION);
+    if (global_var(GVAR_G_MAX_EQUIP) == 0) then begin
+        giq_option(4, NAME, 1602, Node1750, NEUTRAL_REACTION);
     end
-    giq_option(4, 394, 1603, Node999, 50);
+    giq_option(4, NAME, 1603, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node1650
 begin
-    gsay_reply(394, 1650);
-    if (has_skill(dude_obj, 12) > 120) then begin
-        giq_option(4, 394, 1651, Node1700, 50);
+    Reply(1650);
+    if (dude_skill(SKILL_SCIENCE) > 120) then begin
+        giq_option(4, NAME, 1651, Node1700, NEUTRAL_REACTION);
     end
     else begin
-        giq_option(4, 394, 1651, Node1350, 50);
+        giq_option(4, NAME, 1651, Node1350, NEUTRAL_REACTION);
     end
-    giq_option(4, 394, 1652, Node2000, 50);
-    giq_option(4, 394, 1653, Node999, 50);
+    giq_option(4, NAME, 1652, Node2000, NEUTRAL_REACTION);
+    giq_option(4, NAME, 1653, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node1700
 begin
-    gsay_reply(394, 1700);
-    set_global_var(936, 3);
+    Reply(1700);
+    set_global_var(GVAR_G_MAX_REGISTRATION, 3);
     give_exp_points(500);
-    display_msg(message_str(14, 100) + 500 + message_str(14, 101));
-    if (global_var(937) == 0) then begin
-        giq_option(4, 394, 1701, Node1750, 50);
+    display_msg(g_mstr(100) + 500 + g_mstr(101));
+    if (global_var(GVAR_G_MAX_EQUIP) == 0) then begin
+        giq_option(4, NAME, 1701, Node1750, NEUTRAL_REACTION);
     end
-    giq_option(4, 394, 1702, Node999, 50);
+    giq_option(4, NAME, 1702, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node1750
 begin
-    gsay_reply(394, 1750);
-    if (has_skill(dude_obj, 12) >= 130) then begin
-        giq_option(4, 394, 1751, Node1850, 50);
+    Reply(1750);
+    if (dude_skill(SKILL_SCIENCE) >= 130) then begin
+        giq_option(4, NAME, 1751, Node1850, NEUTRAL_REACTION);
     end
     else begin
-        if (has_skill(dude_obj, 12) > 125) then begin
-            giq_option(4, 394, 1751, Node1800, 50);
+        if (dude_skill(SKILL_SCIENCE) > 125) then begin
+            giq_option(4, NAME, 1751, Node1800, NEUTRAL_REACTION);
         end
         else begin
-            giq_option(4, 394, 1751, Node1350, 50);
+            giq_option(4, NAME, 1751, Node1350, NEUTRAL_REACTION);
         end
     end
-    if (global_var(936) == 2) then begin
-        giq_option(4, 394, 1752, Node2000, 50);
+    if (global_var(GVAR_G_MAX_REGISTRATION) == 2) then begin
+        giq_option(4, NAME, 1752, Node2000, NEUTRAL_REACTION);
     end
-    if (global_var(936) == 3) then begin
-        giq_option(4, 394, 1752, Node2050, 50);
+    if (global_var(GVAR_G_MAX_REGISTRATION) == 3) then begin
+        giq_option(4, NAME, 1752, Node2050, NEUTRAL_REACTION);
     end
-    giq_option(4, 394, 1753, Node999, 50);
+    giq_option(4, NAME, 1753, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node1800
 begin
-    gsay_reply(394, 1800);
-    giq_option(4, 394, 1801, Bonus_gun_only, 50);
-    if (global_var(936) == 2) then begin
-        giq_option(4, 394, 1802, Node2000, 50);
+    Reply(1800);
+    giq_option(4, NAME, 1801, Bonus_gun_only, NEUTRAL_REACTION);
+    if (global_var(GVAR_G_MAX_REGISTRATION) == 2) then begin
+        giq_option(4, NAME, 1802, Node2000, NEUTRAL_REACTION);
     end
-    if (global_var(936) == 3) then begin
-        giq_option(4, 394, 1802, Node2050, 50);
+    if (global_var(GVAR_G_MAX_REGISTRATION) == 3) then begin
+        giq_option(4, NAME, 1802, Node2050, NEUTRAL_REACTION);
     end
-    giq_option(4, 394, 1803, Node999, 50);
+    giq_option(4, NAME, 1803, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node1850
 begin
-    gsay_reply(394, 1850);
-    if ((global_var(937) != 1) and (global_var(937) != 3)) then begin
-        giq_option(4, 394, 1851, Bonus_gun, 50);
+    Reply(1850);
+    if ((global_var(GVAR_G_MAX_EQUIP) != 1) and (global_var(GVAR_G_MAX_EQUIP) != 3)) then begin
+        giq_option(4, NAME, 1851, Bonus_gun, NEUTRAL_REACTION);
     end
-    if ((global_var(937) != 2) and (global_var(937) != 3)) then begin
-        giq_option(4, 394, 1852, Bonus_stimpack, 50);
+    if ((global_var(GVAR_G_MAX_EQUIP) != 2) and (global_var(GVAR_G_MAX_EQUIP) != 3)) then begin
+        giq_option(4, NAME, 1852, Bonus_stimpack, NEUTRAL_REACTION);
     end
-    if (global_var(936) == 2) then begin
-        giq_option(4, 394, 1853, Node2000, 50);
+    if (global_var(GVAR_G_MAX_REGISTRATION) == 2) then begin
+        giq_option(4, NAME, 1853, Node2000, NEUTRAL_REACTION);
     end
-    if (global_var(936) == 3) then begin
-        giq_option(4, 394, 1853, Node2050, 50);
+    if (global_var(GVAR_G_MAX_REGISTRATION) == 3) then begin
+        giq_option(4, NAME, 1853, Node2050, NEUTRAL_REACTION);
     end
-    giq_option(4, 394, 1854, Node999, 50);
+    giq_option(4, NAME, 1854, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node1900
 begin
-    gsay_reply(394, 1900);
-    if ((global_var(937) != 1) and (global_var(937) != 3)) then begin
-        giq_option(4, 394, 1901, Bonus_gun, 50);
+    Reply(1900);
+    if ((global_var(GVAR_G_MAX_EQUIP) != 1) and (global_var(GVAR_G_MAX_EQUIP) != 3)) then begin
+        giq_option(4, NAME, 1901, Bonus_gun, NEUTRAL_REACTION);
     end
-    if ((global_var(937) != 2) and (global_var(937) != 3)) then begin
-        giq_option(4, 394, 1902, Bonus_stimpack, 50);
+    if ((global_var(GVAR_G_MAX_EQUIP) != 2) and (global_var(GVAR_G_MAX_EQUIP) != 3)) then begin
+        giq_option(4, NAME, 1902, Bonus_stimpack, NEUTRAL_REACTION);
     end
-    if (global_var(936) == 2) then begin
-        giq_option(4, 394, 1903, Node2000, 50);
+    if (global_var(GVAR_G_MAX_REGISTRATION) == 2) then begin
+        giq_option(4, NAME, 1903, Node2000, NEUTRAL_REACTION);
     end
-    if (global_var(936) == 3) then begin
-        giq_option(4, 394, 1903, Node2050, 50);
+    if (global_var(GVAR_G_MAX_REGISTRATION) == 3) then begin
+        giq_option(4, NAME, 1903, Node2050, NEUTRAL_REACTION);
     end
-    giq_option(4, 394, 1904, Node999, 50);
+    giq_option(4, NAME, 1904, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node1950
 begin
-    gsay_reply(394, message_str(394, 1950) + message_str(394, 1951) + message_str(394, 1952));
-    giq_option(4, 394, 1953, Node1550, 50);
-    giq_option(4, 394, 1954, Node999, 50);
+    Reply(mstr(1950) + mstr(1951) + mstr(1952));
+    giq_option(4, NAME, 1953, Node1550, NEUTRAL_REACTION);
+    giq_option(4, NAME, 1954, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node2000
 begin
-    gsay_reply(394, message_str(394, 2000) + message_str(394, 2001) + message_str(394, 2002));
-    giq_option(4, 394, 2003, Node1650, 50);
-    if (global_var(937) == 0) then begin
-        giq_option(4, 394, 2004, Node1750, 50);
+    Reply(mstr(2000) + mstr(2001) + mstr(2002));
+    giq_option(4, NAME, 2003, Node1650, NEUTRAL_REACTION);
+    if (global_var(GVAR_G_MAX_EQUIP) == 0) then begin
+        giq_option(4, NAME, 2004, Node1750, NEUTRAL_REACTION);
     end
-    giq_option(4, 394, 2005, Node999, 50);
+    giq_option(4, NAME, 2005, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node2050
 begin
-    gsay_reply(394, message_str(394, 2050) + message_str(394, 2051) + message_str(394, 2052));
-    if (global_var(937) == 0) then begin
-        giq_option(4, 394, 2053, Node1750, 50);
+    Reply(mstr(2050) + mstr(2051) + mstr(2052));
+    if (global_var(GVAR_G_MAX_EQUIP) == 0) then begin
+        giq_option(4, NAME, 2053, Node1750, NEUTRAL_REACTION);
     end
-    giq_option(4, 394, 2054, Node999, 50);
+    giq_option(4, NAME, 2054, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node2100
 begin
-    gsay_reply(394, 2100);
-    if (global_var(936) == 2) then begin
-        giq_option(4, 394, 2101, Node2000, 50);
+    Reply(2100);
+    if (global_var(GVAR_G_MAX_REGISTRATION) == 2) then begin
+        giq_option(4, NAME, 2101, Node2000, NEUTRAL_REACTION);
     end
-    if (global_var(936) == 3) then begin
-        giq_option(4, 394, 2101, Node2050, 50);
+    if (global_var(GVAR_G_MAX_REGISTRATION) == 3) then begin
+        giq_option(4, NAME, 2101, Node2050, NEUTRAL_REACTION);
     end
-    giq_option(4, 394, 2102, Node999, 50);
+    giq_option(4, NAME, 2102, Node999, NEUTRAL_REACTION);
 end
 
 procedure Download_holo_VT_base
 begin
-    set_global_var(595, 1);
+    set_global_var(GVAR_NCR_SPY_HOLO_DOWNLOAD, 1);
     call Node600();
 end
 
 procedure Download_holo_ultrastimpack
 begin
-    set_global_var(596, 1);
+    set_global_var(GVAR_NCR_HISTORY_HOLO_DOWNLOAD, 1);
     call Node600();
 end
 
 procedure Bonus_gun_only
 begin
-    set_global_var(937, 1);
+    set_global_var(GVAR_G_MAX_EQUIP, 1);
     call Node2100();
 end
 
 procedure Bonus_gun
 begin
-    if (global_var(937) == 0) then begin
-        set_global_var(937, 1);
+    if (global_var(GVAR_G_MAX_EQUIP) == 0) then begin
+        set_global_var(GVAR_G_MAX_EQUIP, 1);
     end
     else begin
-        set_global_var(937, 3);
+        set_global_var(GVAR_G_MAX_EQUIP, 3);
     end
     call Node1900();
 end
 
 procedure Bonus_stimpack
 begin
-    if (global_var(937) == 0) then begin
-        set_global_var(937, 2);
+    if (global_var(GVAR_G_MAX_EQUIP) == 0) then begin
+        set_global_var(GVAR_G_MAX_EQUIP, 2);
     end
     else begin
-        set_global_var(937, 3);
+        set_global_var(GVAR_G_MAX_EQUIP, 3);
     end
     call Node1900();
 end
-
