diff --git a/Mods/Resurrection/Original/GDOR6WAT.ssl b/Mods/Resurrection/ReDefine/GDOR6WAT.ssl
index 050d8fe..75fba34 100644
--- a/Mods/Resurrection/Original/GDOR6WAT.ssl
+++ b/Mods/Resurrection/ReDefine/GDOR6WAT.ssl
@@ -62,12 +62,12 @@ begin
             return 1;
         end
     end
-    if ((party_member_obj(16777779) != 0) and ((global_var(910) bwand 32) == 0)) then begin
+    if ((party_member_obj(16777779) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 32) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777779)) <= 5) then begin
             return 1;
         end
     end
-    if ((party_member_obj(16777811) != 0) and ((global_var(910) bwand 64) == 0)) then begin
+    if ((party_member_obj(16777811) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 64) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777811)) <= 5) then begin
             return 1;
         end
@@ -82,7 +82,7 @@ begin
             return 1;
         end
     end
-    if ((party_member_obj(16777814) != 0) and ((global_var(910) bwand 128) == 0)) then begin
+    if ((party_member_obj(16777814) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 128) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777814)) <= 5) then begin
             return 1;
         end
@@ -146,16 +146,16 @@ begin
             set_local_var(2, 1);
             reg_anim_func(2, source_obj);
             if (source_obj == dude_obj) then begin
-                display_msg(message_str(413, 204));
+                display_msg(message_str(SCRIPT_GDOORS, 204));
             end
             else begin
-                display_msg(message_str(413, 205));
+                display_msg(message_str(SCRIPT_GDOORS, 205));
             end
         end
         else begin
             if (obj_is_locked(self_obj)) then begin
                 script_overrides;
-                display_msg(message_str(413, 203));
+                display_msg(message_str(SCRIPT_GDOORS, 203));
                 call Damage_Critter();
             end
             else begin
@@ -166,7 +166,7 @@ begin
     else begin
         if (obj_is_locked(self_obj)) then begin
             script_overrides;
-            display_msg(message_str(413, 203));
+            display_msg(message_str(SCRIPT_GDOORS, 203));
         end
     end
     if (0 == 1) then begin
@@ -179,10 +179,10 @@ procedure look_at_p_proc
 begin
     script_overrides;
     if (1 == 0) then begin
-        display_msg(message_str(413, 100));
+        display_msg(message_str(SCRIPT_GDOORS, 100));
     end
     else begin
-        display_msg(message_str(413, 100));
+        display_msg(message_str(SCRIPT_GDOORS, 100));
     end
 end
 
@@ -202,10 +202,10 @@ begin
             end
             else begin
                 if (1 == 0) then begin
-                    display_msg(message_str(413, 100));
+                    display_msg(message_str(SCRIPT_GDOORS, 100));
                 end
                 else begin
-                    display_msg(message_str(413, 100));
+                    display_msg(message_str(SCRIPT_GDOORS, 100));
                 end
             end
         end
@@ -231,10 +231,10 @@ begin
             end
             else begin
                 if (source_obj == dude_obj) then begin
-                    display_msg(message_str(413, 195));
+                    display_msg(message_str(SCRIPT_GDOORS, 195));
                 end
                 else begin
-                    display_msg(obj_name(source_obj) + message_str(413, 200));
+                    display_msg(obj_name(source_obj) + message_str(SCRIPT_GDOORS, 200));
                 end
             end
         end
@@ -297,7 +297,7 @@ end
 procedure damage_p_proc
 begin
     variable LVar0 := 0;
-    if (metarule(49, target_obj) == 6) then begin
+    if (weapon_dmg_type(target_obj) == 6) then begin
         if (1 == 0) then begin
             if (local_var(1) == 0) then begin
                 set_local_var(1, 1);
@@ -319,21 +319,21 @@ begin
                     set_local_var(0, 1);
                     LVar0 := random(10, 20);
                     explosion(tile_num(self_obj), elevation(self_obj), LVar0);
-                    set_global_var(927, 1);
+                    set_global_var(GVAR_ENEMY_OBRODA, 1);
                 end
                 else begin
                     if (local_var(4) > 2) then begin
                         set_local_var(1, 1);
                         set_local_var(0, 1);
                         destroy_object(self_obj);
-                        set_global_var(927, 1);
+                        set_global_var(GVAR_ENEMY_OBRODA, 1);
                     end
                 end
             end
         end
     end
     else begin
-        display_msg(message_str(413, 193));
+        display_msg(message_str(SCRIPT_GDOORS, 193));
     end
 end
 
@@ -394,36 +394,36 @@ begin
                 if (is_critical(LVar1)) then begin
                     if (is_success(LVar2)) then begin
                         if (is_critical(LVar2)) then begin
-                            display_msg(message_str(413, 102));
+                            display_msg(message_str(SCRIPT_GDOORS, 102));
                         end
                         else begin
-                            display_msg(message_str(413, 103));
+                            display_msg(message_str(SCRIPT_GDOORS, 103));
                         end
                     end
                     else begin
                         if (is_critical(LVar2)) then begin
-                            display_msg(message_str(413, 104));
+                            display_msg(message_str(SCRIPT_GDOORS, 104));
                         end
                         else begin
-                            display_msg(message_str(413, 105));
+                            display_msg(message_str(SCRIPT_GDOORS, 105));
                         end
                     end
                 end
                 else begin
                     if (is_success(LVar2)) then begin
                         if (is_critical(LVar2)) then begin
-                            display_msg(message_str(413, 106));
+                            display_msg(message_str(SCRIPT_GDOORS, 106));
                         end
                         else begin
-                            display_msg(message_str(413, 107));
+                            display_msg(message_str(SCRIPT_GDOORS, 107));
                         end
                     end
                     else begin
                         if (is_critical(LVar2)) then begin
-                            display_msg(message_str(413, 108));
+                            display_msg(message_str(SCRIPT_GDOORS, 108));
                         end
                         else begin
-                            display_msg(message_str(413, 109));
+                            display_msg(message_str(SCRIPT_GDOORS, 109));
                         end
                     end
                 end
@@ -432,36 +432,36 @@ begin
                 if (is_critical(LVar1)) then begin
                     if (is_success(LVar2)) then begin
                         if (is_critical(LVar2)) then begin
-                            display_msg(message_str(413, 110));
+                            display_msg(message_str(SCRIPT_GDOORS, 110));
                         end
                         else begin
-                            display_msg(message_str(413, 111));
+                            display_msg(message_str(SCRIPT_GDOORS, 111));
                         end
                     end
                     else begin
                         if (is_critical(LVar2)) then begin
-                            display_msg(message_str(413, 112));
+                            display_msg(message_str(SCRIPT_GDOORS, 112));
                         end
                         else begin
-                            display_msg(message_str(413, 113));
+                            display_msg(message_str(SCRIPT_GDOORS, 113));
                         end
                     end
                 end
                 else begin
                     if (is_success(LVar2)) then begin
                         if (is_critical(LVar2)) then begin
-                            display_msg(message_str(413, 114));
+                            display_msg(message_str(SCRIPT_GDOORS, 114));
                         end
                         else begin
-                            display_msg(message_str(413, 115));
+                            display_msg(message_str(SCRIPT_GDOORS, 115));
                         end
                     end
                     else begin
                         if (is_critical(LVar2)) then begin
-                            display_msg(message_str(413, 116));
+                            display_msg(message_str(SCRIPT_GDOORS, 116));
                         end
                         else begin
-                            display_msg(message_str(413, 117));
+                            display_msg(message_str(SCRIPT_GDOORS, 117));
                         end
                     end
                 end
@@ -473,36 +473,36 @@ begin
                 if (is_critical(LVar1)) then begin
                     if (is_success(LVar2)) then begin
                         if (is_critical(LVar2)) then begin
-                            display_msg(message_str(413, 118));
+                            display_msg(message_str(SCRIPT_GDOORS, 118));
                         end
                         else begin
-                            display_msg(message_str(413, 119));
+                            display_msg(message_str(SCRIPT_GDOORS, 119));
                         end
                     end
                     else begin
                         if (is_critical(LVar2)) then begin
-                            display_msg(message_str(413, 120));
+                            display_msg(message_str(SCRIPT_GDOORS, 120));
                         end
                         else begin
-                            display_msg(message_str(413, 121));
+                            display_msg(message_str(SCRIPT_GDOORS, 121));
                         end
                     end
                 end
                 else begin
                     if (is_success(LVar2)) then begin
                         if (is_critical(LVar2)) then begin
-                            display_msg(message_str(413, 122));
+                            display_msg(message_str(SCRIPT_GDOORS, 122));
                         end
                         else begin
-                            display_msg(message_str(413, 123));
+                            display_msg(message_str(SCRIPT_GDOORS, 123));
                         end
                     end
                     else begin
                         if (is_critical(LVar2)) then begin
-                            display_msg(message_str(413, 124));
+                            display_msg(message_str(SCRIPT_GDOORS, 124));
                         end
                         else begin
-                            display_msg(message_str(413, 125));
+                            display_msg(message_str(SCRIPT_GDOORS, 125));
                         end
                     end
                 end
@@ -511,36 +511,36 @@ begin
                 if (is_critical(LVar1)) then begin
                     if (is_success(LVar2)) then begin
                         if (is_critical(LVar2)) then begin
-                            display_msg(message_str(413, 126));
+                            display_msg(message_str(SCRIPT_GDOORS, 126));
                         end
                         else begin
-                            display_msg(message_str(413, 127));
+                            display_msg(message_str(SCRIPT_GDOORS, 127));
                         end
                     end
                     else begin
                         if (is_critical(LVar2)) then begin
-                            display_msg(message_str(413, 128));
+                            display_msg(message_str(SCRIPT_GDOORS, 128));
                         end
                         else begin
-                            display_msg(message_str(413, 129));
+                            display_msg(message_str(SCRIPT_GDOORS, 129));
                         end
                     end
                 end
                 else begin
                     if (is_success(LVar2)) then begin
                         if (is_critical(LVar2)) then begin
-                            display_msg(message_str(413, 130));
+                            display_msg(message_str(SCRIPT_GDOORS, 130));
                         end
                         else begin
-                            display_msg(message_str(413, 131));
+                            display_msg(message_str(SCRIPT_GDOORS, 131));
                         end
                     end
                     else begin
                         if (is_critical(LVar2)) then begin
-                            display_msg(message_str(413, 132));
+                            display_msg(message_str(SCRIPT_GDOORS, 132));
                         end
                         else begin
-                            display_msg(message_str(413, 133));
+                            display_msg(message_str(SCRIPT_GDOORS, 133));
                         end
                     end
                 end
@@ -554,36 +554,36 @@ begin
                 if (is_critical(LVar1)) then begin
                     if (is_success(LVar2)) then begin
                         if (is_critical(LVar2)) then begin
-                            display_msg(message_str(413, 134));
+                            display_msg(message_str(SCRIPT_GDOORS, 134));
                         end
                         else begin
-                            display_msg(message_str(413, 135));
+                            display_msg(message_str(SCRIPT_GDOORS, 135));
                         end
                     end
                     else begin
                         if (is_critical(LVar2)) then begin
-                            display_msg(message_str(413, 136));
+                            display_msg(message_str(SCRIPT_GDOORS, 136));
                         end
                         else begin
-                            display_msg(message_str(413, 137));
+                            display_msg(message_str(SCRIPT_GDOORS, 137));
                         end
                     end
                 end
                 else begin
                     if (is_success(LVar2)) then begin
                         if (is_critical(LVar2)) then begin
-                            display_msg(message_str(413, 138));
+                            display_msg(message_str(SCRIPT_GDOORS, 138));
                         end
                         else begin
-                            display_msg(message_str(413, 139));
+                            display_msg(message_str(SCRIPT_GDOORS, 139));
                         end
                     end
                     else begin
                         if (is_critical(LVar2)) then begin
-                            display_msg(message_str(413, 140));
+                            display_msg(message_str(SCRIPT_GDOORS, 140));
                         end
                         else begin
-                            display_msg(message_str(413, 141));
+                            display_msg(message_str(SCRIPT_GDOORS, 141));
                         end
                     end
                 end
@@ -592,36 +592,36 @@ begin
                 if (is_critical(LVar1)) then begin
                     if (is_success(LVar2)) then begin
                         if (is_critical(LVar2)) then begin
-                            display_msg(message_str(413, 142));
+                            display_msg(message_str(SCRIPT_GDOORS, 142));
                         end
                         else begin
-                            display_msg(message_str(413, 143));
+                            display_msg(message_str(SCRIPT_GDOORS, 143));
                         end
                     end
                     else begin
                         if (is_critical(LVar2)) then begin
-                            display_msg(message_str(413, 144));
+                            display_msg(message_str(SCRIPT_GDOORS, 144));
                         end
                         else begin
-                            display_msg(message_str(413, 145));
+                            display_msg(message_str(SCRIPT_GDOORS, 145));
                         end
                     end
                 end
                 else begin
                     if (is_success(LVar2)) then begin
                         if (is_critical(LVar2)) then begin
-                            display_msg(message_str(413, 146));
+                            display_msg(message_str(SCRIPT_GDOORS, 146));
                         end
                         else begin
-                            display_msg(message_str(413, 147));
+                            display_msg(message_str(SCRIPT_GDOORS, 147));
                         end
                     end
                     else begin
                         if (is_critical(LVar2)) then begin
-                            display_msg(message_str(413, 148));
+                            display_msg(message_str(SCRIPT_GDOORS, 148));
                         end
                         else begin
-                            display_msg(message_str(413, 149));
+                            display_msg(message_str(SCRIPT_GDOORS, 149));
                         end
                     end
                 end
@@ -633,36 +633,36 @@ begin
                 if (is_critical(LVar1)) then begin
                     if (is_success(LVar2)) then begin
                         if (is_critical(LVar2)) then begin
-                            display_msg(message_str(413, 150));
+                            display_msg(message_str(SCRIPT_GDOORS, 150));
                         end
                         else begin
-                            display_msg(message_str(413, 151));
+                            display_msg(message_str(SCRIPT_GDOORS, 151));
                         end
                     end
                     else begin
                         if (is_critical(LVar2)) then begin
-                            display_msg(message_str(413, 152));
+                            display_msg(message_str(SCRIPT_GDOORS, 152));
                         end
                         else begin
-                            display_msg(message_str(413, 153));
+                            display_msg(message_str(SCRIPT_GDOORS, 153));
                         end
                     end
                 end
                 else begin
                     if (is_success(LVar2)) then begin
                         if (is_critical(LVar2)) then begin
-                            display_msg(message_str(413, 154));
+                            display_msg(message_str(SCRIPT_GDOORS, 154));
                         end
                         else begin
-                            display_msg(message_str(413, 155));
+                            display_msg(message_str(SCRIPT_GDOORS, 155));
                         end
                     end
                     else begin
                         if (is_critical(LVar2)) then begin
-                            display_msg(message_str(413, 156));
+                            display_msg(message_str(SCRIPT_GDOORS, 156));
                         end
                         else begin
-                            display_msg(message_str(413, 157));
+                            display_msg(message_str(SCRIPT_GDOORS, 157));
                         end
                     end
                 end
@@ -671,36 +671,36 @@ begin
                 if (is_critical(LVar1)) then begin
                     if (is_success(LVar2)) then begin
                         if (is_critical(LVar2)) then begin
-                            display_msg(message_str(413, 158));
+                            display_msg(message_str(SCRIPT_GDOORS, 158));
                         end
                         else begin
-                            display_msg(message_str(413, 159));
+                            display_msg(message_str(SCRIPT_GDOORS, 159));
                         end
                     end
                     else begin
                         if (is_critical(LVar2)) then begin
-                            display_msg(message_str(413, 160));
+                            display_msg(message_str(SCRIPT_GDOORS, 160));
                         end
                         else begin
-                            display_msg(message_str(413, 161));
+                            display_msg(message_str(SCRIPT_GDOORS, 161));
                         end
                     end
                 end
                 else begin
                     if (is_success(LVar2)) then begin
                         if (is_critical(LVar2)) then begin
-                            display_msg(message_str(413, 162));
+                            display_msg(message_str(SCRIPT_GDOORS, 162));
                         end
                         else begin
-                            display_msg(message_str(413, 163));
+                            display_msg(message_str(SCRIPT_GDOORS, 163));
                         end
                     end
                     else begin
                         if (is_critical(LVar2)) then begin
-                            display_msg(message_str(413, 164));
+                            display_msg(message_str(SCRIPT_GDOORS, 164));
                         end
                         else begin
-                            display_msg(message_str(413, 165));
+                            display_msg(message_str(SCRIPT_GDOORS, 165));
                         end
                     end
                 end
@@ -719,36 +719,36 @@ begin
         if (is_critical(LVar0)) then begin
             if (is_success(LVar1)) then begin
                 if (is_critical(LVar1)) then begin
-                    display_msg(message_str(413, 104));
+                    display_msg(message_str(SCRIPT_GDOORS, 104));
                 end
                 else begin
-                    display_msg(message_str(413, 108));
+                    display_msg(message_str(SCRIPT_GDOORS, 108));
                 end
             end
             else begin
                 if (is_critical(LVar1)) then begin
-                    display_msg(message_str(413, 112));
+                    display_msg(message_str(SCRIPT_GDOORS, 112));
                 end
                 else begin
-                    display_msg(message_str(413, 116));
+                    display_msg(message_str(SCRIPT_GDOORS, 116));
                 end
             end
         end
         else begin
             if (is_success(LVar1)) then begin
                 if (is_critical(LVar1)) then begin
-                    display_msg(message_str(413, 120));
+                    display_msg(message_str(SCRIPT_GDOORS, 120));
                 end
                 else begin
-                    display_msg(message_str(413, 124));
+                    display_msg(message_str(SCRIPT_GDOORS, 124));
                 end
             end
             else begin
                 if (is_critical(LVar1)) then begin
-                    display_msg(message_str(413, 128));
+                    display_msg(message_str(SCRIPT_GDOORS, 128));
                 end
                 else begin
-                    display_msg(message_str(413, 132));
+                    display_msg(message_str(SCRIPT_GDOORS, 132));
                 end
             end
         end
@@ -757,36 +757,36 @@ begin
         if (is_critical(LVar0)) then begin
             if (is_success(LVar1)) then begin
                 if (is_critical(LVar1)) then begin
-                    display_msg(message_str(413, 136));
+                    display_msg(message_str(SCRIPT_GDOORS, 136));
                 end
                 else begin
-                    display_msg(message_str(413, 140));
+                    display_msg(message_str(SCRIPT_GDOORS, 140));
                 end
             end
             else begin
                 if (is_critical(LVar1)) then begin
-                    display_msg(message_str(413, 144));
+                    display_msg(message_str(SCRIPT_GDOORS, 144));
                 end
                 else begin
-                    display_msg(message_str(413, 148));
+                    display_msg(message_str(SCRIPT_GDOORS, 148));
                 end
             end
         end
         else begin
             if (is_success(LVar1)) then begin
                 if (is_critical(LVar1)) then begin
-                    display_msg(message_str(413, 152));
+                    display_msg(message_str(SCRIPT_GDOORS, 152));
                 end
                 else begin
-                    display_msg(message_str(413, 156));
+                    display_msg(message_str(SCRIPT_GDOORS, 156));
                 end
             end
             else begin
                 if (is_critical(LVar1)) then begin
-                    display_msg(message_str(413, 160));
+                    display_msg(message_str(SCRIPT_GDOORS, 160));
                 end
                 else begin
-                    display_msg(message_str(413, 164));
+                    display_msg(message_str(SCRIPT_GDOORS, 164));
                 end
             end
         end
@@ -803,36 +803,36 @@ begin
         if (is_critical(LVar0)) then begin
             if (is_success(LVar1)) then begin
                 if (is_critical(LVar1)) then begin
-                    display_msg(message_str(413, 114));
+                    display_msg(message_str(SCRIPT_GDOORS, 114));
                 end
                 else begin
-                    display_msg(message_str(413, 115));
+                    display_msg(message_str(SCRIPT_GDOORS, 115));
                 end
             end
             else begin
                 if (is_critical(LVar1)) then begin
-                    display_msg(message_str(413, 116));
+                    display_msg(message_str(SCRIPT_GDOORS, 116));
                 end
                 else begin
-                    display_msg(message_str(413, 117));
+                    display_msg(message_str(SCRIPT_GDOORS, 117));
                 end
             end
         end
         else begin
             if (is_success(LVar1)) then begin
                 if (is_critical(LVar1)) then begin
-                    display_msg(message_str(413, 130));
+                    display_msg(message_str(SCRIPT_GDOORS, 130));
                 end
                 else begin
-                    display_msg(message_str(413, 131));
+                    display_msg(message_str(SCRIPT_GDOORS, 131));
                 end
             end
             else begin
                 if (is_critical(LVar1)) then begin
-                    display_msg(message_str(413, 132));
+                    display_msg(message_str(SCRIPT_GDOORS, 132));
                 end
                 else begin
-                    display_msg(message_str(413, 133));
+                    display_msg(message_str(SCRIPT_GDOORS, 133));
                 end
             end
         end
@@ -841,36 +841,36 @@ begin
         if (is_critical(LVar0)) then begin
             if (is_success(LVar1)) then begin
                 if (is_critical(LVar1)) then begin
-                    display_msg(message_str(413, 146));
+                    display_msg(message_str(SCRIPT_GDOORS, 146));
                 end
                 else begin
-                    display_msg(message_str(413, 147));
+                    display_msg(message_str(SCRIPT_GDOORS, 147));
                 end
             end
             else begin
                 if (is_critical(LVar1)) then begin
-                    display_msg(message_str(413, 148));
+                    display_msg(message_str(SCRIPT_GDOORS, 148));
                 end
                 else begin
-                    display_msg(message_str(413, 149));
+                    display_msg(message_str(SCRIPT_GDOORS, 149));
                 end
             end
         end
         else begin
             if (is_success(LVar1)) then begin
                 if (is_critical(LVar1)) then begin
-                    display_msg(message_str(413, 162));
+                    display_msg(message_str(SCRIPT_GDOORS, 162));
                 end
                 else begin
-                    display_msg(message_str(413, 163));
+                    display_msg(message_str(SCRIPT_GDOORS, 163));
                 end
             end
             else begin
                 if (is_critical(LVar1)) then begin
-                    display_msg(message_str(413, 164));
+                    display_msg(message_str(SCRIPT_GDOORS, 164));
                 end
                 else begin
-                    display_msg(message_str(413, 165));
+                    display_msg(message_str(SCRIPT_GDOORS, 165));
                 end
             end
         end
@@ -889,24 +889,24 @@ begin
         if (local_var(2) == 0) then begin
             set_local_var(2, 1);
             if (source_obj == dude_obj) then begin
-                display_msg(message_str(413, 194));
+                display_msg(message_str(SCRIPT_GDOORS, 194));
             end
             else begin
-                display_msg(message_str(413, 199));
+                display_msg(message_str(SCRIPT_GDOORS, 199));
             end
             if (local_var(6) == 0) then begin
                 set_local_var(6, 1);
                 give_exp_points(25);
-                display_msg(message_str(14, 100) + 25 + message_str(14, 101));
+                display_msg(g_mstr(100) + 25 + g_mstr(101));
             end
         end
         else begin
             set_local_var(1, 1);
             if (source_obj == dude_obj) then begin
-                display_msg(message_str(413, 196));
+                display_msg(message_str(SCRIPT_GDOORS, 196));
             end
             else begin
-                display_msg(obj_name(source_obj) + message_str(413, 201));
+                display_msg(obj_name(source_obj) + message_str(SCRIPT_GDOORS, 201));
             end
         end
     end
@@ -917,14 +917,14 @@ begin
         else begin
             if (local_var(2) == 0) then begin
                 if (source_obj == dude_obj) then begin
-                    display_msg(message_str(413, 195));
+                    display_msg(message_str(SCRIPT_GDOORS, 195));
                 end
                 else begin
-                    display_msg(obj_name(source_obj) + message_str(413, 200));
+                    display_msg(obj_name(source_obj) + message_str(SCRIPT_GDOORS, 200));
                 end
             end
             else begin
-                display_msg(obj_name(source_obj) + message_str(413, 202));
+                display_msg(obj_name(source_obj) + message_str(SCRIPT_GDOORS, 202));
             end
         end
     end
@@ -948,33 +948,33 @@ begin
         set_local_var(0, 1);
         obj_unlock(self_obj);
         if (source_obj == dude_obj) then begin
-            display_msg(message_str(413, 170));
+            display_msg(message_str(SCRIPT_GDOORS, 170));
         end
         else begin
-            display_msg(obj_name(source_obj) + message_str(413, 173));
+            display_msg(obj_name(source_obj) + message_str(SCRIPT_GDOORS, 173));
         end
         if (local_var(5) == 0) then begin
             set_local_var(5, 1);
             give_exp_points(25);
-            display_msg(message_str(14, 100) + 25 + message_str(14, 101));
+            display_msg(g_mstr(100) + 25 + g_mstr(101));
         end
     end
     else begin
         if (is_critical(Locks_Roll)) then begin
             jam_lock(self_obj);
             if (source_obj == dude_obj) then begin
-                display_msg(message_str(413, 171));
+                display_msg(message_str(SCRIPT_GDOORS, 171));
             end
             else begin
-                display_msg(obj_name(source_obj) + message_str(413, 174));
+                display_msg(obj_name(source_obj) + message_str(SCRIPT_GDOORS, 174));
             end
         end
         else begin
             if (source_obj == dude_obj) then begin
-                display_msg(message_str(413, 172));
+                display_msg(message_str(SCRIPT_GDOORS, 172));
             end
             else begin
-                display_msg(obj_name(source_obj) + message_str(413, 175));
+                display_msg(obj_name(source_obj) + message_str(SCRIPT_GDOORS, 175));
             end
         end
     end
@@ -992,23 +992,23 @@ begin
         set_local_var(0, 0);
         obj_lock(self_obj);
         if (source_obj == dude_obj) then begin
-            display_msg(message_str(413, 189));
+            display_msg(message_str(SCRIPT_GDOORS, 189));
         end
         else begin
-            display_msg(obj_name(source_obj) + message_str(413, 191));
+            display_msg(obj_name(source_obj) + message_str(SCRIPT_GDOORS, 191));
         end
         if (local_var(5) == 0) then begin
             set_local_var(5, 1);
             give_exp_points(25);
-            display_msg(message_str(14, 100) + 25 + message_str(14, 101));
+            display_msg(g_mstr(100) + 25 + g_mstr(101));
         end
     end
     else begin
         if (source_obj == dude_obj) then begin
-            display_msg(message_str(413, 190));
+            display_msg(message_str(SCRIPT_GDOORS, 190));
         end
         else begin
-            display_msg(obj_name(source_obj) + message_str(413, 192));
+            display_msg(obj_name(source_obj) + message_str(SCRIPT_GDOORS, 192));
         end
     end
 end
@@ -1040,10 +1040,10 @@ begin
     if (obj_is_open(self_obj)) then begin
         script_overrides;
         if (source_obj == dude_obj) then begin
-            display_msg(message_str(413, 206));
+            display_msg(message_str(SCRIPT_GDOORS, 206));
         end
         else begin
-            display_msg(obj_name(source_obj) + message_str(413, 207));
+            display_msg(obj_name(source_obj) + message_str(SCRIPT_GDOORS, 207));
         end
     end
     else begin
@@ -1052,10 +1052,10 @@ begin
         if (is_success(Traps_Roll)) then begin
             set_local_var(1, 0);
             if (source_obj == dude_obj) then begin
-                display_msg(message_str(413, 208));
+                display_msg(message_str(SCRIPT_GDOORS, 208));
             end
             else begin
-                display_msg(obj_name(source_obj) + message_str(413, 210));
+                display_msg(obj_name(source_obj) + message_str(SCRIPT_GDOORS, 210));
             end
         end
         else begin
@@ -1064,10 +1064,10 @@ begin
             end
             else begin
                 if (source_obj == dude_obj) then begin
-                    display_msg(message_str(413, 209));
+                    display_msg(message_str(SCRIPT_GDOORS, 209));
                 end
                 else begin
-                    display_msg(obj_name(source_obj) + message_str(413, 211));
+                    display_msg(obj_name(source_obj) + message_str(SCRIPT_GDOORS, 211));
                 end
             end
         end
@@ -1082,10 +1082,10 @@ begin
         set_local_var(0, 1);
         obj_unlock(self_obj);
         if (source_obj == dude_obj) then begin
-            display_msg(message_str(413, 176));
+            display_msg(message_str(SCRIPT_GDOORS, 176));
         end
         else begin
-            display_msg(message_str(413, 181));
+            display_msg(message_str(SCRIPT_GDOORS, 181));
         end
     end
     else begin
@@ -1093,37 +1093,37 @@ begin
             critter_dmg(source_obj, 2, 0 bwor 256);
             if (source_obj == dude_obj) then begin
                 if (2 == 1) then begin
-                    display_msg(message_str(413, 177));
+                    display_msg(message_str(SCRIPT_GDOORS, 177));
                 end
                 else begin
-                    display_msg(message_str(413, 178) + 2 + message_str(413, 179));
+                    display_msg(message_str(SCRIPT_GDOORS, 178) + 2 + message_str(SCRIPT_GDOORS, 179));
                 end
             end
             else begin
-                if (get_critter_stat(source_obj, 34) == 0) then begin
+                if (get_critter_stat(source_obj, STAT_gender) == 0) then begin
                     if (2 == 1) then begin
-                        display_msg(message_str(413, 182));
+                        display_msg(message_str(SCRIPT_GDOORS, 182));
                     end
                     else begin
-                        display_msg(message_str(413, 183) + 2 + message_str(413, 184));
+                        display_msg(message_str(SCRIPT_GDOORS, 183) + 2 + message_str(SCRIPT_GDOORS, 184));
                     end
                 end
                 else begin
                     if (2 == 1) then begin
-                        display_msg(message_str(413, 186));
+                        display_msg(message_str(SCRIPT_GDOORS, 186));
                     end
                     else begin
-                        display_msg(message_str(413, 187) + 2 + message_str(413, 188));
+                        display_msg(message_str(SCRIPT_GDOORS, 187) + 2 + message_str(SCRIPT_GDOORS, 188));
                     end
                 end
             end
         end
         else begin
             if (source_obj == dude_obj) then begin
-                display_msg(message_str(413, 180));
+                display_msg(message_str(SCRIPT_GDOORS, 180));
             end
             else begin
-                display_msg(message_str(413, 185));
+                display_msg(message_str(SCRIPT_GDOORS, 185));
             end
         end
     end
@@ -1135,12 +1135,11 @@ begin
     LVar0 := random(10, 20);
     if (source_obj == dude_obj) then begin
         critter_dmg(dude_obj, LVar0, 6 bwor 256);
-        display_msg(message_str(413, 166) + LVar0 + message_str(413, 167));
+        display_msg(message_str(SCRIPT_GDOORS, 166) + LVar0 + message_str(SCRIPT_GDOORS, 167));
     end
     else begin
         critter_dmg(source_obj, LVar0, 6 bwor 256);
-        display_msg(obj_name(source_obj) + message_str(413, 168) + LVar0 + message_str(413, 169));
+        display_msg(obj_name(source_obj) + message_str(SCRIPT_GDOORS, 168) + LVar0 + message_str(SCRIPT_GDOORS, 169));
     end
     set_local_var(1, 1);
 end
-
