diff --git a/Mods/Resurrection/Original/GEMIT4_2.ssl b/Mods/Resurrection/ReDefine/GEMIT4_2.ssl
index fde585c..396d904 100644
--- a/Mods/Resurrection/Original/GEMIT4_2.ssl
+++ b/Mods/Resurrection/ReDefine/GEMIT4_2.ssl
@@ -41,12 +41,12 @@ begin
             return 1;
         end
     end
-    if ((party_member_obj(16777779) != 0) and ((global_var(910) bwand 32) == 0)) then begin
+    if ((party_member_obj(16777779) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 32) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777779)) <= 5) then begin
             return 1;
         end
     end
-    if ((party_member_obj(16777811) != 0) and ((global_var(910) bwand 64) == 0)) then begin
+    if ((party_member_obj(16777811) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 64) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777811)) <= 5) then begin
             return 1;
         end
@@ -61,7 +61,7 @@ begin
             return 1;
         end
     end
-    if ((party_member_obj(16777814) != 0) and ((global_var(910) bwand 128) == 0)) then begin
+    if ((party_member_obj(16777814) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 128) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777814)) <= 5) then begin
             return 1;
         end
@@ -122,14 +122,14 @@ end
 
 procedure map_enter_p_proc
 begin
-    if (metarule(14, 0)) then begin
+    if (map_first_run) then begin
         set_local_var(0, 1);
     end
 end
 
 procedure map_update_p_proc
 begin
-    if ((global_var(952) == 2) and (local_var(0) == 1) and (elevation(dude_obj) == 0)) then begin
+    if ((global_var(GVAR_G_QUEST_FREE_PRISONERS_SNEAK) == 2) and (local_var(0) == 1) and (dude_elevation == 0)) then begin
         fieldPtr := tile_contains_pid_obj(23494, 0, 33554922);
         set_obj_visibility(fieldPtr, 1);
         set_local_var(0, 0);
@@ -137,7 +137,7 @@ begin
     else begin
         if (local_var(0) == 0) then begin
             fieldPtr := tile_contains_pid_obj(23494, 0, 33554922);
-            if (has_trait(1, fieldPtr, 666)) then begin
+            if (obj_is_visible_flag(fieldPtr)) then begin
                 set_obj_visibility(fieldPtr, 1);
             end
         end
@@ -151,4 +151,3 @@ end
 procedure destroy_p_proc
 begin
 end
-
