diff --git a/Mods/Resurrection/Original/GHANDY2.ssl b/Mods/Resurrection/ReDefine/GHANDY2.ssl
index d5c05bc..175f4b3 100644
--- a/Mods/Resurrection/Original/GHANDY2.ssl
+++ b/Mods/Resurrection/ReDefine/GHANDY2.ssl
@@ -64,12 +64,12 @@ begin
             return 1;
         end
     end
-    if ((party_member_obj(16777779) != 0) and ((global_var(910) bwand 32) == 0)) then begin
+    if ((party_member_obj(16777779) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 32) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777779)) <= 5) then begin
             return 1;
         end
     end
-    if ((party_member_obj(16777811) != 0) and ((global_var(910) bwand 64) == 0)) then begin
+    if ((party_member_obj(16777811) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 64) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777811)) <= 5) then begin
             return 1;
         end
@@ -84,7 +84,7 @@ begin
             return 1;
         end
     end
-    if ((party_member_obj(16777814) != 0) and ((global_var(910) bwand 128) == 0)) then begin
+    if ((party_member_obj(16777814) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 128) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777814)) <= 5) then begin
             return 1;
         end
@@ -149,12 +149,12 @@ end
 
 procedure talk_p_proc
 begin
-    anim(dude_obj, 1000, rotation_to_tile(tile_num(dude_obj), tile_num(self_obj)));
-    anim(self_obj, 1000, rotation_to_tile(tile_num(self_obj), tile_num(dude_obj)));
+    anim(dude_obj, 1000, rotation_to_tile(dude_tile, tile_num(self_obj)));
+    anim(self_obj, 1000, rotation_to_tile(tile_num(self_obj), dude_tile));
     start_gdialog(424, self_obj, 4, -1, -1);
     gsay_start;
     if ((local_var(4) bwand 1) == 0) then begin
-        if (get_critter_stat(dude_obj, 4) > 7) then begin
+        if (dude_iq > 7) then begin
             call Node500();
         end
         else begin
@@ -171,13 +171,13 @@ end
 procedure look_at_p_proc
 begin
     script_overrides;
-    display_msg(message_str(424, 100));
+    display_mstr(100);
 end
 
 procedure description_p_proc
 begin
     script_overrides;
-    display_msg(message_str(424, 200));
+    display_mstr(200);
 end
 
 procedure timed_event_p_proc
@@ -186,14 +186,14 @@ end
 
 procedure map_enter_p_proc
 begin
-    critter_add_trait(self_obj, 1, 6, 271);
-    critter_add_trait(self_obj, 1, 5, 180);
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 271);
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_PEASANT_NO_MOVE);
 end
 
 procedure damage_p_proc
 begin
     if (source_obj == dude_obj) then begin
-        set_global_var(927, 1);
+        set_global_var(GVAR_ENEMY_OBRODA, 1);
     end
 end
 
@@ -208,143 +208,143 @@ end
 
 procedure Node500
 begin
-    gsay_reply(424, 500);
-    giq_option(4, 424, 501, Node700, 50);
-    giq_option(4, 424, 502, Node999, 50);
+    Reply(500);
+    giq_option(4, NAME, 501, Node700, NEUTRAL_REACTION);
+    giq_option(4, NAME, 502, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node550
 begin
-    gsay_reply(424, 550);
-    giq_option(4, 424, 551, Node600, 50);
-    giq_option(4, 424, 552, Node999, 50);
-    giq_option(-3, 424, 553, Node999, 50);
+    Reply(550);
+    giq_option(4, NAME, 551, Node600, NEUTRAL_REACTION);
+    giq_option(4, NAME, 552, Node999, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 553, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node600
 begin
-    gsay_reply(424, 600);
-    giq_option(4, 424, 601, Node650, 50);
-    giq_option(4, 424, 602, Node999, 50);
+    Reply(600);
+    giq_option(4, NAME, 601, Node650, NEUTRAL_REACTION);
+    giq_option(4, NAME, 602, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node650
 begin
-    gsay_reply(424, 650);
+    Reply(650);
     set_local_var(4, local_var(4) bwor 1);
-    if (has_skill(dude_obj, 12) >= 130) then begin
-        giq_option(4, 424, 651, Node750, 50);
+    if (dude_skill(SKILL_SCIENCE) >= 130) then begin
+        giq_option(4, NAME, 651, Node750, NEUTRAL_REACTION);
     end
-    giq_option(4, 424, 652, Node999, 50);
+    giq_option(4, NAME, 652, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node700
 begin
-    gsay_reply(424, 700);
+    Reply(700);
     set_local_var(4, local_var(4) bwor 1);
-    if (has_skill(dude_obj, 12) >= 130) then begin
-        giq_option(4, 424, 701, Node750, 50);
+    if (dude_skill(SKILL_SCIENCE) >= 130) then begin
+        giq_option(4, NAME, 701, Node750, NEUTRAL_REACTION);
     end
-    giq_option(4, 424, 702, Node999, 50);
+    giq_option(4, NAME, 702, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node750
 begin
-    gsay_reply(424, 750 + (get_critter_stat(dude_obj, 34) == 1));
+    Reply(750 + (dude_gender == 1));
     set_local_var(4, local_var(4) bwor 2);
-    giq_option(4, 424, 752, Node800, 50);
+    giq_option(4, NAME, 752, Node800, NEUTRAL_REACTION);
 end
 
 procedure Node800
 begin
-    gsay_reply(424, 800);
-    giq_option(4, 424, 801, Node850, 50);
-    giq_option(4, 424, 802, Node900, 50);
-    giq_option(4, 424, 803, Node999, 50);
+    Reply(800);
+    giq_option(4, NAME, 801, Node850, NEUTRAL_REACTION);
+    giq_option(4, NAME, 802, Node900, NEUTRAL_REACTION);
+    giq_option(4, NAME, 803, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node850
 begin
-    if ((has_skill(dude_obj, 13) >= 130) and (has_skill(dude_obj, 0) >= 130) and (obj_carrying_pid_obj(dude_obj, 18) or obj_carrying_pid_obj(dude_obj, 10) or obj_carrying_pid_obj(dude_obj, 94) or obj_carrying_pid_obj(dude_obj, 23))) then begin
-        gsay_reply(424, 850);
+    if ((dude_skill(SKILL_REPAIR) >= 130) and (dude_skill(SKILL_SMALL_GUNS) >= 130) and (obj_carrying_pid_obj(dude_obj, 18) or obj_carrying_pid_obj(dude_obj, 10) or obj_carrying_pid_obj(dude_obj, 94) or obj_carrying_pid_obj(dude_obj, 23))) then begin
+        Reply(850);
     end
     else begin
-        gsay_reply(424, 851);
+        Reply(851);
     end
-    if ((has_skill(dude_obj, 13) >= 130) and (has_skill(dude_obj, 0) >= 130)) then begin
+    if ((dude_skill(SKILL_REPAIR) >= 130) and (dude_skill(SKILL_SMALL_GUNS) >= 130)) then begin
         if (obj_carrying_pid_obj(dude_obj, 18)) then begin
-            giq_option(4, 424, 852, Upgrade_eagle, 50);
+            giq_option(4, NAME, 852, Upgrade_eagle, NEUTRAL_REACTION);
         end
         if (obj_carrying_pid_obj(dude_obj, 10)) then begin
-            giq_option(4, 424, 853, Upgrade_hunting, 50);
+            giq_option(4, NAME, 853, Upgrade_hunting, NEUTRAL_REACTION);
         end
         if (obj_carrying_pid_obj(dude_obj, 94)) then begin
-            giq_option(4, 424, 854, Upgrade_shotgun, 50);
+            giq_option(4, NAME, 854, Upgrade_shotgun, NEUTRAL_REACTION);
         end
         if (obj_carrying_pid_obj(dude_obj, 23)) then begin
-            giq_option(4, 424, 855, Upgrade_assault, 50);
+            giq_option(4, NAME, 855, Upgrade_assault, NEUTRAL_REACTION);
         end
         if (obj_carrying_pid_obj(dude_obj, 313)) then begin
-            giq_option(4, 424, 856, Upgrade_revolver, 50);
+            giq_option(4, NAME, 856, Upgrade_revolver, NEUTRAL_REACTION);
         end
         if (obj_carrying_pid_obj(dude_obj, 242)) then begin
-            giq_option(4, 424, 857, Upgrade_combatshotgun, 50);
+            giq_option(4, NAME, 857, Upgrade_combatshotgun, NEUTRAL_REACTION);
         end
     end
-    giq_option(4, 424, 858, Node900, 50);
-    giq_option(4, 424, 859, Node999, 50);
+    giq_option(4, NAME, 858, Node900, NEUTRAL_REACTION);
+    giq_option(4, NAME, 859, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node900
 begin
-    if ((has_skill(dude_obj, 13) >= 130) and (has_skill(dude_obj, 1) >= 130) and (obj_carrying_pid_obj(dude_obj, 12) or obj_carrying_pid_obj(dude_obj, 11) or obj_carrying_pid_obj(dude_obj, 32))) then begin
-        gsay_reply(424, 900);
+    if ((dude_skill(SKILL_REPAIR) >= 130) and (dude_skill(SKILL_BIG_GUNS) >= 130) and (obj_carrying_pid_obj(dude_obj, 12) or obj_carrying_pid_obj(dude_obj, 11) or obj_carrying_pid_obj(dude_obj, 32))) then begin
+        Reply(900);
     end
     else begin
-        gsay_reply(424, 901);
+        Reply(901);
     end
-    if ((has_skill(dude_obj, 13) >= 130) and (has_skill(dude_obj, 1) >= 130)) then begin
+    if ((dude_skill(SKILL_REPAIR) >= 130) and (dude_skill(SKILL_BIG_GUNS) >= 130)) then begin
         if (obj_carrying_pid_obj(dude_obj, 12)) then begin
-            giq_option(4, 424, 902, Upgrade_minigun, 50);
+            giq_option(4, NAME, 902, Upgrade_minigun, NEUTRAL_REACTION);
         end
         if (obj_carrying_pid_obj(dude_obj, 11)) then begin
-            giq_option(4, 424, 903, Upgrade_flamethrower, 50);
+            giq_option(4, NAME, 903, Upgrade_flamethrower, NEUTRAL_REACTION);
         end
         if (obj_carrying_pid_obj(dude_obj, 32)) then begin
-            giq_option(4, 424, 904, Upgrade_flamefuel, 50);
+            giq_option(4, NAME, 904, Upgrade_flamefuel, NEUTRAL_REACTION);
         end
     end
-    giq_option(4, 424, 905, Node850, 50);
-    giq_option(4, 424, 906, Node999, 50);
+    giq_option(4, NAME, 905, Node850, NEUTRAL_REACTION);
+    giq_option(4, NAME, 906, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node950
 begin
-    gsay_reply(424, 950 + (get_critter_stat(dude_obj, 34) == 1));
-    giq_option(4, 424, 952, Node850, 50);
-    giq_option(4, 424, 953, Node900, 50);
-    giq_option(4, 424, 954, Node999, 50);
+    Reply(950 + (dude_gender == 1));
+    giq_option(4, NAME, 952, Node850, NEUTRAL_REACTION);
+    giq_option(4, NAME, 953, Node900, NEUTRAL_REACTION);
+    giq_option(4, NAME, 954, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node1000
 begin
-    gsay_reply(424, 1000 + (get_critter_stat(dude_obj, 34) == 1));
-    giq_option(4, 424, 1002, Node850, 50);
-    giq_option(4, 424, 1003, Node900, 50);
-    giq_option(4, 424, 1004, Node999, 50);
+    Reply(1000 + (dude_gender == 1));
+    giq_option(4, NAME, 1002, Node850, NEUTRAL_REACTION);
+    giq_option(4, NAME, 1003, Node900, NEUTRAL_REACTION);
+    giq_option(4, NAME, 1004, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node1050
 begin
-    gsay_reply(424, 1050);
+    Reply(1050);
     if ((local_var(4) bwand 2) == 0) then begin
-        giq_option(4, 424, 1051, Node650, 50);
+        giq_option(4, NAME, 1051, Node650, NEUTRAL_REACTION);
     end
     if ((local_var(4) bwand 2) != 0) then begin
-        giq_option(4, 424, 1052, Node800, 50);
+        giq_option(4, NAME, 1052, Node800, NEUTRAL_REACTION);
     end
-    giq_option(4, 424, 1053, Node999, 50);
-    giq_option(-3, 424, 1054, Node999, 50);
+    giq_option(4, NAME, 1053, Node999, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 1054, Node999, NEUTRAL_REACTION);
 end
 
 procedure Upgrade_eagle
@@ -357,17 +357,17 @@ begin
         restock_obj := obj_carrying_pid_obj(dude_obj, 18);
         if (obj_type(self_obj) == 1) then begin
             restock_obj := obj_carrying_pid_obj(dude_obj, 18);
-            if (critter_inven_obj(self_obj, 0) == restock_obj) then begin
-                if (obj_item_subtype(critter_inven_obj(dude_obj, 0)) == 0) then begin
-                    restock_obj := critter_inven_obj(dude_obj, 0);
+            if (critter_inven_obj(self_obj, INVEN_TYPE_WORN) == restock_obj) then begin
+                if (obj_item_subtype(critter_inven_obj(dude_obj, INVEN_TYPE_WORN)) == 0) then begin
+                    restock_obj := critter_inven_obj(dude_obj, INVEN_TYPE_WORN);
                     debug_msg("armour pid == " + obj_pid(restock_obj));
                     rm_obj_from_inven(dude_obj, restock_obj);
                     add_obj_to_inven(dude_obj, restock_obj);
                 end
             end
             else begin
-                if ((critter_inven_obj(dude_obj, 2) == restock_obj) or (critter_inven_obj(dude_obj, 1) == restock_obj)) then begin
-                    metarule(43, dude_obj);
+                if ((critter_inven_obj(dude_obj, INVEN_TYPE_LEFT_HAND) == restock_obj) or (critter_inven_obj(dude_obj, INVEN_TYPE_RIGHT_HAND) == restock_obj)) then begin
+                    inven_unwield(dude_obj);
                 end
             end
         end
@@ -387,7 +387,7 @@ begin
     end
     restock_obj := 0;
     restock_amt := 0;
-    restock_obj := create_object_sid(404, 0, 0, -1);
+    restock_obj := create_object(404, 0, 0);
     add_mult_objs_to_inven(dude_obj, restock_obj, 1);
     game_time_advance(game_ticks(60 * 60));
     call Node950();
@@ -403,17 +403,17 @@ begin
         restock_obj := obj_carrying_pid_obj(dude_obj, 10);
         if (obj_type(self_obj) == 1) then begin
             restock_obj := obj_carrying_pid_obj(dude_obj, 10);
-            if (critter_inven_obj(self_obj, 0) == restock_obj) then begin
-                if (obj_item_subtype(critter_inven_obj(dude_obj, 0)) == 0) then begin
-                    restock_obj := critter_inven_obj(dude_obj, 0);
+            if (critter_inven_obj(self_obj, INVEN_TYPE_WORN) == restock_obj) then begin
+                if (obj_item_subtype(critter_inven_obj(dude_obj, INVEN_TYPE_WORN)) == 0) then begin
+                    restock_obj := critter_inven_obj(dude_obj, INVEN_TYPE_WORN);
                     debug_msg("armour pid == " + obj_pid(restock_obj));
                     rm_obj_from_inven(dude_obj, restock_obj);
                     add_obj_to_inven(dude_obj, restock_obj);
                 end
             end
             else begin
-                if ((critter_inven_obj(dude_obj, 2) == restock_obj) or (critter_inven_obj(dude_obj, 1) == restock_obj)) then begin
-                    metarule(43, dude_obj);
+                if ((critter_inven_obj(dude_obj, INVEN_TYPE_LEFT_HAND) == restock_obj) or (critter_inven_obj(dude_obj, INVEN_TYPE_RIGHT_HAND) == restock_obj)) then begin
+                    inven_unwield(dude_obj);
                 end
             end
         end
@@ -433,7 +433,7 @@ begin
     end
     restock_obj := 0;
     restock_amt := 0;
-    restock_obj := create_object_sid(287, 0, 0, -1);
+    restock_obj := create_object(287, 0, 0);
     add_mult_objs_to_inven(dude_obj, restock_obj, 1);
     game_time_advance(game_ticks(60 * 60));
     call Node950();
@@ -449,17 +449,17 @@ begin
         restock_obj := obj_carrying_pid_obj(dude_obj, 94);
         if (obj_type(self_obj) == 1) then begin
             restock_obj := obj_carrying_pid_obj(dude_obj, 94);
-            if (critter_inven_obj(self_obj, 0) == restock_obj) then begin
-                if (obj_item_subtype(critter_inven_obj(dude_obj, 0)) == 0) then begin
-                    restock_obj := critter_inven_obj(dude_obj, 0);
+            if (critter_inven_obj(self_obj, INVEN_TYPE_WORN) == restock_obj) then begin
+                if (obj_item_subtype(critter_inven_obj(dude_obj, INVEN_TYPE_WORN)) == 0) then begin
+                    restock_obj := critter_inven_obj(dude_obj, INVEN_TYPE_WORN);
                     debug_msg("armour pid == " + obj_pid(restock_obj));
                     rm_obj_from_inven(dude_obj, restock_obj);
                     add_obj_to_inven(dude_obj, restock_obj);
                 end
             end
             else begin
-                if ((critter_inven_obj(dude_obj, 2) == restock_obj) or (critter_inven_obj(dude_obj, 1) == restock_obj)) then begin
-                    metarule(43, dude_obj);
+                if ((critter_inven_obj(dude_obj, INVEN_TYPE_LEFT_HAND) == restock_obj) or (critter_inven_obj(dude_obj, INVEN_TYPE_RIGHT_HAND) == restock_obj)) then begin
+                    inven_unwield(dude_obj);
                 end
             end
         end
@@ -479,7 +479,7 @@ begin
     end
     restock_obj := 0;
     restock_amt := 0;
-    restock_obj := create_object_sid(385, 0, 0, -1);
+    restock_obj := create_object(385, 0, 0);
     add_mult_objs_to_inven(dude_obj, restock_obj, 1);
     game_time_advance(game_ticks(60 * 60));
     call Node950();
@@ -495,17 +495,17 @@ begin
         restock_obj := obj_carrying_pid_obj(dude_obj, 23);
         if (obj_type(self_obj) == 1) then begin
             restock_obj := obj_carrying_pid_obj(dude_obj, 23);
-            if (critter_inven_obj(self_obj, 0) == restock_obj) then begin
-                if (obj_item_subtype(critter_inven_obj(dude_obj, 0)) == 0) then begin
-                    restock_obj := critter_inven_obj(dude_obj, 0);
+            if (critter_inven_obj(self_obj, INVEN_TYPE_WORN) == restock_obj) then begin
+                if (obj_item_subtype(critter_inven_obj(dude_obj, INVEN_TYPE_WORN)) == 0) then begin
+                    restock_obj := critter_inven_obj(dude_obj, INVEN_TYPE_WORN);
                     debug_msg("armour pid == " + obj_pid(restock_obj));
                     rm_obj_from_inven(dude_obj, restock_obj);
                     add_obj_to_inven(dude_obj, restock_obj);
                 end
             end
             else begin
-                if ((critter_inven_obj(dude_obj, 2) == restock_obj) or (critter_inven_obj(dude_obj, 1) == restock_obj)) then begin
-                    metarule(43, dude_obj);
+                if ((critter_inven_obj(dude_obj, INVEN_TYPE_LEFT_HAND) == restock_obj) or (critter_inven_obj(dude_obj, INVEN_TYPE_RIGHT_HAND) == restock_obj)) then begin
+                    inven_unwield(dude_obj);
                 end
             end
         end
@@ -525,7 +525,7 @@ begin
     end
     restock_obj := 0;
     restock_amt := 0;
-    restock_obj := create_object_sid(405, 0, 0, -1);
+    restock_obj := create_object(405, 0, 0);
     add_mult_objs_to_inven(dude_obj, restock_obj, 1);
     game_time_advance(game_ticks(60 * 60));
     call Node950();
@@ -541,17 +541,17 @@ begin
         restock_obj := obj_carrying_pid_obj(dude_obj, 313);
         if (obj_type(self_obj) == 1) then begin
             restock_obj := obj_carrying_pid_obj(dude_obj, 313);
-            if (critter_inven_obj(self_obj, 0) == restock_obj) then begin
-                if (obj_item_subtype(critter_inven_obj(dude_obj, 0)) == 0) then begin
-                    restock_obj := critter_inven_obj(dude_obj, 0);
+            if (critter_inven_obj(self_obj, INVEN_TYPE_WORN) == restock_obj) then begin
+                if (obj_item_subtype(critter_inven_obj(dude_obj, INVEN_TYPE_WORN)) == 0) then begin
+                    restock_obj := critter_inven_obj(dude_obj, INVEN_TYPE_WORN);
                     debug_msg("armour pid == " + obj_pid(restock_obj));
                     rm_obj_from_inven(dude_obj, restock_obj);
                     add_obj_to_inven(dude_obj, restock_obj);
                 end
             end
             else begin
-                if ((critter_inven_obj(dude_obj, 2) == restock_obj) or (critter_inven_obj(dude_obj, 1) == restock_obj)) then begin
-                    metarule(43, dude_obj);
+                if ((critter_inven_obj(dude_obj, INVEN_TYPE_LEFT_HAND) == restock_obj) or (critter_inven_obj(dude_obj, INVEN_TYPE_RIGHT_HAND) == restock_obj)) then begin
+                    inven_unwield(dude_obj);
                 end
             end
         end
@@ -571,7 +571,7 @@ begin
     end
     restock_obj := 0;
     restock_amt := 0;
-    restock_obj := create_object_sid(398, 0, 0, -1);
+    restock_obj := create_object(398, 0, 0);
     add_mult_objs_to_inven(dude_obj, restock_obj, 1);
     game_time_advance(game_ticks(60 * 60));
     call Node950();
@@ -587,17 +587,17 @@ begin
         restock_obj := obj_carrying_pid_obj(dude_obj, 242);
         if (obj_type(self_obj) == 1) then begin
             restock_obj := obj_carrying_pid_obj(dude_obj, 242);
-            if (critter_inven_obj(self_obj, 0) == restock_obj) then begin
-                if (obj_item_subtype(critter_inven_obj(dude_obj, 0)) == 0) then begin
-                    restock_obj := critter_inven_obj(dude_obj, 0);
+            if (critter_inven_obj(self_obj, INVEN_TYPE_WORN) == restock_obj) then begin
+                if (obj_item_subtype(critter_inven_obj(dude_obj, INVEN_TYPE_WORN)) == 0) then begin
+                    restock_obj := critter_inven_obj(dude_obj, INVEN_TYPE_WORN);
                     debug_msg("armour pid == " + obj_pid(restock_obj));
                     rm_obj_from_inven(dude_obj, restock_obj);
                     add_obj_to_inven(dude_obj, restock_obj);
                 end
             end
             else begin
-                if ((critter_inven_obj(dude_obj, 2) == restock_obj) or (critter_inven_obj(dude_obj, 1) == restock_obj)) then begin
-                    metarule(43, dude_obj);
+                if ((critter_inven_obj(dude_obj, INVEN_TYPE_LEFT_HAND) == restock_obj) or (critter_inven_obj(dude_obj, INVEN_TYPE_RIGHT_HAND) == restock_obj)) then begin
+                    inven_unwield(dude_obj);
                 end
             end
         end
@@ -617,7 +617,7 @@ begin
     end
     restock_obj := 0;
     restock_amt := 0;
-    restock_obj := create_object_sid(598, 0, 0, -1);
+    restock_obj := create_object(598, 0, 0);
     add_mult_objs_to_inven(dude_obj, restock_obj, 1);
     game_time_advance(game_ticks(60 * 60));
     call Node950();
@@ -633,17 +633,17 @@ begin
         restock_obj := obj_carrying_pid_obj(dude_obj, 12);
         if (obj_type(self_obj) == 1) then begin
             restock_obj := obj_carrying_pid_obj(dude_obj, 12);
-            if (critter_inven_obj(self_obj, 0) == restock_obj) then begin
-                if (obj_item_subtype(critter_inven_obj(dude_obj, 0)) == 0) then begin
-                    restock_obj := critter_inven_obj(dude_obj, 0);
+            if (critter_inven_obj(self_obj, INVEN_TYPE_WORN) == restock_obj) then begin
+                if (obj_item_subtype(critter_inven_obj(dude_obj, INVEN_TYPE_WORN)) == 0) then begin
+                    restock_obj := critter_inven_obj(dude_obj, INVEN_TYPE_WORN);
                     debug_msg("armour pid == " + obj_pid(restock_obj));
                     rm_obj_from_inven(dude_obj, restock_obj);
                     add_obj_to_inven(dude_obj, restock_obj);
                 end
             end
             else begin
-                if ((critter_inven_obj(dude_obj, 2) == restock_obj) or (critter_inven_obj(dude_obj, 1) == restock_obj)) then begin
-                    metarule(43, dude_obj);
+                if ((critter_inven_obj(dude_obj, INVEN_TYPE_LEFT_HAND) == restock_obj) or (critter_inven_obj(dude_obj, INVEN_TYPE_RIGHT_HAND) == restock_obj)) then begin
+                    inven_unwield(dude_obj);
                 end
             end
         end
@@ -663,7 +663,7 @@ begin
     end
     restock_obj := 0;
     restock_amt := 0;
-    restock_obj := create_object_sid(389, 0, 0, -1);
+    restock_obj := create_object(389, 0, 0);
     add_mult_objs_to_inven(dude_obj, restock_obj, 1);
     game_time_advance(game_ticks(60 * 60));
     call Node950();
@@ -679,17 +679,17 @@ begin
         restock_obj := obj_carrying_pid_obj(dude_obj, 11);
         if (obj_type(self_obj) == 1) then begin
             restock_obj := obj_carrying_pid_obj(dude_obj, 11);
-            if (critter_inven_obj(self_obj, 0) == restock_obj) then begin
-                if (obj_item_subtype(critter_inven_obj(dude_obj, 0)) == 0) then begin
-                    restock_obj := critter_inven_obj(dude_obj, 0);
+            if (critter_inven_obj(self_obj, INVEN_TYPE_WORN) == restock_obj) then begin
+                if (obj_item_subtype(critter_inven_obj(dude_obj, INVEN_TYPE_WORN)) == 0) then begin
+                    restock_obj := critter_inven_obj(dude_obj, INVEN_TYPE_WORN);
                     debug_msg("armour pid == " + obj_pid(restock_obj));
                     rm_obj_from_inven(dude_obj, restock_obj);
                     add_obj_to_inven(dude_obj, restock_obj);
                 end
             end
             else begin
-                if ((critter_inven_obj(dude_obj, 2) == restock_obj) or (critter_inven_obj(dude_obj, 1) == restock_obj)) then begin
-                    metarule(43, dude_obj);
+                if ((critter_inven_obj(dude_obj, INVEN_TYPE_LEFT_HAND) == restock_obj) or (critter_inven_obj(dude_obj, INVEN_TYPE_RIGHT_HAND) == restock_obj)) then begin
+                    inven_unwield(dude_obj);
                 end
             end
         end
@@ -709,7 +709,7 @@ begin
     end
     restock_obj := 0;
     restock_amt := 0;
-    restock_obj := create_object_sid(400, 0, 0, -1);
+    restock_obj := create_object(400, 0, 0);
     add_mult_objs_to_inven(dude_obj, restock_obj, 1);
     game_time_advance(game_ticks(60 * 60));
     call Node1000();
@@ -725,17 +725,17 @@ begin
         restock_obj := obj_carrying_pid_obj(dude_obj, 32);
         if (obj_type(self_obj) == 1) then begin
             restock_obj := obj_carrying_pid_obj(dude_obj, 32);
-            if (critter_inven_obj(self_obj, 0) == restock_obj) then begin
-                if (obj_item_subtype(critter_inven_obj(dude_obj, 0)) == 0) then begin
-                    restock_obj := critter_inven_obj(dude_obj, 0);
+            if (critter_inven_obj(self_obj, INVEN_TYPE_WORN) == restock_obj) then begin
+                if (obj_item_subtype(critter_inven_obj(dude_obj, INVEN_TYPE_WORN)) == 0) then begin
+                    restock_obj := critter_inven_obj(dude_obj, INVEN_TYPE_WORN);
                     debug_msg("armour pid == " + obj_pid(restock_obj));
                     rm_obj_from_inven(dude_obj, restock_obj);
                     add_obj_to_inven(dude_obj, restock_obj);
                 end
             end
             else begin
-                if ((critter_inven_obj(dude_obj, 2) == restock_obj) or (critter_inven_obj(dude_obj, 1) == restock_obj)) then begin
-                    metarule(43, dude_obj);
+                if ((critter_inven_obj(dude_obj, INVEN_TYPE_LEFT_HAND) == restock_obj) or (critter_inven_obj(dude_obj, INVEN_TYPE_RIGHT_HAND) == restock_obj)) then begin
+                    inven_unwield(dude_obj);
                 end
             end
         end
@@ -755,9 +755,8 @@ begin
     end
     restock_obj := 0;
     restock_amt := 0;
-    restock_obj := create_object_sid(382, 0, 0, -1);
+    restock_obj := create_object(382, 0, 0);
     add_mult_objs_to_inven(dude_obj, restock_obj, 1);
     game_time_advance(game_ticks(60 * 60));
     call Node1000();
 end
-
