diff --git a/Mods/Resurrection/Original/GHOULMET.ssl b/Mods/Resurrection/ReDefine/GHOULMET.ssl
index 40026c5..acd0bf9 100644
--- a/Mods/Resurrection/Original/GHOULMET.ssl
+++ b/Mods/Resurrection/ReDefine/GHOULMET.ssl
@@ -35,12 +35,12 @@ begin
             return 1;
         end
     end
-    if ((party_member_obj(16777779) != 0) and ((global_var(910) bwand 32) == 0)) then begin
+    if ((party_member_obj(16777779) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 32) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777779)) <= 5) then begin
             return 1;
         end
     end
-    if ((party_member_obj(16777811) != 0) and ((global_var(910) bwand 64) == 0)) then begin
+    if ((party_member_obj(16777811) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 64) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777811)) <= 5) then begin
             return 1;
         end
@@ -55,7 +55,7 @@ begin
             return 1;
         end
     end
-    if ((party_member_obj(16777814) != 0) and ((global_var(910) bwand 128) == 0)) then begin
+    if ((party_member_obj(16777814) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 128) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777814)) <= 5) then begin
             return 1;
         end
@@ -192,8 +192,8 @@ begin
             end
         end
     end
-    if (metarule(14, 0)) then begin
-        display_msg(message_str(352, 100 + (get_critter_stat(dude_obj, 34) == 1)));
+    if (map_first_run) then begin
+        display_mstr(100 + (dude_gender == 1));
     end
 end
 
@@ -276,4 +276,3 @@ begin
         end
     end
 end
-
