diff --git a/Mods/Resurrection/Original/GLOVEC.ssl b/Mods/Resurrection/ReDefine/GLOVEC.ssl
index ecc06f5..8433c88 100644
--- a/Mods/Resurrection/Original/GLOVEC.ssl
+++ b/Mods/Resurrection/ReDefine/GLOVEC.ssl
@@ -48,12 +48,12 @@ begin
             return 1;
         end
     end
-    if ((party_member_obj(16777779) != 0) and ((global_var(910) bwand 32) == 0)) then begin
+    if ((party_member_obj(16777779) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 32) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777779)) <= 5) then begin
             return 1;
         end
     end
-    if ((party_member_obj(16777811) != 0) and ((global_var(910) bwand 64) == 0)) then begin
+    if ((party_member_obj(16777811) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 64) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777811)) <= 5) then begin
             return 1;
         end
@@ -68,7 +68,7 @@ begin
             return 1;
         end
     end
-    if ((party_member_obj(16777814) != 0) and ((global_var(910) bwand 128) == 0)) then begin
+    if ((party_member_obj(16777814) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 128) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777814)) <= 5) then begin
             return 1;
         end
@@ -125,23 +125,23 @@ end
 
 procedure map_enter_p_proc
 begin
-    critter_add_trait(self_obj, 1, 6, 0);
-    critter_add_trait(self_obj, 1, 5, 3);
-    if (cur_map_index == 30) then begin
-        if (not(has_trait(1, self_obj, 666))) then begin
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 0);
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_TOUGH_GUARD_NO_TALKING);
+    if (CUR_MAP_GHOUL0) then begin
+        if (not(obj_is_visible_flag(self_obj))) then begin
             set_obj_visibility(self_obj, 0);
-            critter_attempt_placement(self_obj, tile_num_in_direction(tile_num(dude_obj), has_trait(1, dude_obj, 10), 2), 0);
-            anim(self_obj, 1000, has_trait(1, dude_obj, 10));
+            critter_attempt_placement(self_obj, tile_num_in_direction(dude_tile, obj_get_rot(dude_obj), 2), 0);
+            anim(self_obj, 1000, obj_get_rot(dude_obj));
         end
-        if ((global_var(964) bwand 16384) != 0) then begin
-            set_global_var(964, global_var(964) bwand (-1 - 16384));
+        if ((global_var(GVAR_G_HUNTERS_FLOATS) bwand 16384) != 0) then begin
+            set_global_var(GVAR_G_HUNTERS_FLOATS, global_var(GVAR_G_HUNTERS_FLOATS) bwand (-1 - 16384));
         end
     end
-    if ((cur_map_index != 30) and (cur_map_index != 31) and (cur_map_index != 32) and has_trait(1, self_obj, 666)) then begin
+    if ((cur_map_index != MAP_GHOUL0) and (cur_map_index != MAP_GHOUL13) and (cur_map_index != MAP_GHOUL46) and obj_is_visible_flag(self_obj)) then begin
         set_obj_visibility(self_obj, 1);
-        if ((global_var(964) bwand 16384) == 0) then begin
-            set_global_var(964, global_var(964) bwor 16384);
-            display_msg(message_str(520, 250));
+        if ((global_var(GVAR_G_HUNTERS_FLOATS) bwand 16384) == 0) then begin
+            set_global_var(GVAR_G_HUNTERS_FLOATS, global_var(GVAR_G_HUNTERS_FLOATS) bwor 16384);
+            display_mstr(250);
         end
     end
     if (local_var(6) == 0) then begin
@@ -154,92 +154,92 @@ end
 procedure combat_p_proc
 begin
     if (fixed_param == 4) then begin
-        if ((cur_map_index == 31) and (elevation(self_obj) == 0)) then begin
-            if ((global_var(964) bwand 4) == 0) then begin
-                set_global_var(964, global_var(964) bwor 4);
+        if ((CUR_MAP_GHOUL13) and (elevation(self_obj) == 0)) then begin
+            if ((global_var(GVAR_G_HUNTERS_FLOATS) bwand 4) == 0) then begin
+                set_global_var(GVAR_G_HUNTERS_FLOATS, global_var(GVAR_G_HUNTERS_FLOATS) bwor 4);
                 floated := 1;
-                float_msg(self_obj, message_str(520, 302), 2);
+                float_msg(self_obj, mstr(302), FLOAT_MSG_RED);
             end
             else begin
-                if ((global_var(964) bwand 8) == 0) then begin
-                    set_global_var(964, global_var(964) bwor 8);
+                if ((global_var(GVAR_G_HUNTERS_FLOATS) bwand 8) == 0) then begin
+                    set_global_var(GVAR_G_HUNTERS_FLOATS, global_var(GVAR_G_HUNTERS_FLOATS) bwor 8);
                     floated := 1;
-                    float_msg(self_obj, message_str(520, 303), 2);
+                    float_msg(self_obj, mstr(303), FLOAT_MSG_RED);
                 end
             end
         end
-        if ((cur_map_index == 31) and (elevation(self_obj) == 1)) then begin
-            if ((global_var(964) bwand 16) == 0) then begin
-                set_global_var(964, global_var(964) bwor 16);
+        if ((CUR_MAP_GHOUL13) and (elevation(self_obj) == 1)) then begin
+            if ((global_var(GVAR_G_HUNTERS_FLOATS) bwand 16) == 0) then begin
+                set_global_var(GVAR_G_HUNTERS_FLOATS, global_var(GVAR_G_HUNTERS_FLOATS) bwor 16);
                 floated := 1;
-                float_msg(self_obj, message_str(520, 305), 2);
+                float_msg(self_obj, mstr(305), FLOAT_MSG_RED);
             end
             else begin
-                if ((global_var(964) bwand 32) == 0) then begin
-                    set_global_var(964, global_var(964) bwor 32);
+                if ((global_var(GVAR_G_HUNTERS_FLOATS) bwand 32) == 0) then begin
+                    set_global_var(GVAR_G_HUNTERS_FLOATS, global_var(GVAR_G_HUNTERS_FLOATS) bwor 32);
                     floated := 1;
-                    float_msg(self_obj, message_str(520, 306), 2);
+                    float_msg(self_obj, mstr(306), FLOAT_MSG_RED);
                 end
             end
         end
-        if ((cur_map_index == 31) and (elevation(self_obj) == 2)) then begin
-            if ((global_var(964) bwand 64) == 0) then begin
-                set_global_var(964, global_var(964) bwor 64);
+        if ((CUR_MAP_GHOUL13) and (elevation(self_obj) == 2)) then begin
+            if ((global_var(GVAR_G_HUNTERS_FLOATS) bwand 64) == 0) then begin
+                set_global_var(GVAR_G_HUNTERS_FLOATS, global_var(GVAR_G_HUNTERS_FLOATS) bwor 64);
                 floated := 1;
-                float_msg(self_obj, message_str(520, 307), 2);
+                float_msg(self_obj, mstr(307), FLOAT_MSG_RED);
             end
             else begin
-                if ((global_var(964) bwand 128) == 0) then begin
-                    set_global_var(964, global_var(964) bwor 128);
+                if ((global_var(GVAR_G_HUNTERS_FLOATS) bwand 128) == 0) then begin
+                    set_global_var(GVAR_G_HUNTERS_FLOATS, global_var(GVAR_G_HUNTERS_FLOATS) bwor 128);
                     floated := 1;
-                    float_msg(self_obj, message_str(520, 308), 2);
+                    float_msg(self_obj, mstr(308), FLOAT_MSG_RED);
                 end
             end
         end
-        if ((cur_map_index == 32) and (elevation(self_obj) == 0)) then begin
-            if ((global_var(964) bwand 256) == 0) then begin
-                set_global_var(964, global_var(964) bwor 256);
+        if ((CUR_MAP_GHOUL46) and (elevation(self_obj) == 0)) then begin
+            if ((global_var(GVAR_G_HUNTERS_FLOATS) bwand 256) == 0) then begin
+                set_global_var(GVAR_G_HUNTERS_FLOATS, global_var(GVAR_G_HUNTERS_FLOATS) bwor 256);
                 floated := 1;
-                float_msg(self_obj, message_str(520, 309), 2);
+                float_msg(self_obj, mstr(309), FLOAT_MSG_RED);
             end
             else begin
-                if ((global_var(964) bwand 512) == 0) then begin
-                    set_global_var(964, global_var(964) bwor 512);
+                if ((global_var(GVAR_G_HUNTERS_FLOATS) bwand 512) == 0) then begin
+                    set_global_var(GVAR_G_HUNTERS_FLOATS, global_var(GVAR_G_HUNTERS_FLOATS) bwor 512);
                     floated := 1;
-                    float_msg(self_obj, message_str(520, 310), 2);
+                    float_msg(self_obj, mstr(310), FLOAT_MSG_RED);
                 end
             end
         end
-        if ((cur_map_index == 32) and (elevation(self_obj) == 1)) then begin
-            if ((global_var(964) bwand 1024) == 0) then begin
-                set_global_var(964, global_var(964) bwor 1024);
+        if ((CUR_MAP_GHOUL46) and (elevation(self_obj) == 1)) then begin
+            if ((global_var(GVAR_G_HUNTERS_FLOATS) bwand 1024) == 0) then begin
+                set_global_var(GVAR_G_HUNTERS_FLOATS, global_var(GVAR_G_HUNTERS_FLOATS) bwor 1024);
                 floated := 1;
-                float_msg(self_obj, message_str(520, 311), 2);
+                float_msg(self_obj, mstr(311), FLOAT_MSG_RED);
             end
             else begin
-                if ((global_var(964) bwand 2048) == 0) then begin
-                    set_global_var(964, global_var(964) bwor 2048);
+                if ((global_var(GVAR_G_HUNTERS_FLOATS) bwand 2048) == 0) then begin
+                    set_global_var(GVAR_G_HUNTERS_FLOATS, global_var(GVAR_G_HUNTERS_FLOATS) bwor 2048);
                     floated := 1;
-                    float_msg(self_obj, message_str(520, 312), 2);
+                    float_msg(self_obj, mstr(312), FLOAT_MSG_RED);
                 end
             end
         end
-        if ((cur_map_index == 32) and (elevation(self_obj) == 2)) then begin
-            if ((global_var(964) bwand 4096) == 0) then begin
-                set_global_var(964, global_var(964) bwor 4096);
+        if ((CUR_MAP_GHOUL46) and (elevation(self_obj) == 2)) then begin
+            if ((global_var(GVAR_G_HUNTERS_FLOATS) bwand 4096) == 0) then begin
+                set_global_var(GVAR_G_HUNTERS_FLOATS, global_var(GVAR_G_HUNTERS_FLOATS) bwor 4096);
                 floated := 1;
-                float_msg(self_obj, message_str(520, 313), 2);
+                float_msg(self_obj, mstr(313), FLOAT_MSG_RED);
             end
             else begin
-                if ((global_var(964) bwand 8192) == 0) then begin
-                    set_global_var(964, global_var(964) bwor 8192);
+                if ((global_var(GVAR_G_HUNTERS_FLOATS) bwand 8192) == 0) then begin
+                    set_global_var(GVAR_G_HUNTERS_FLOATS, global_var(GVAR_G_HUNTERS_FLOATS) bwor 8192);
                     floated := 1;
-                    float_msg(self_obj, message_str(520, 314), 2);
+                    float_msg(self_obj, mstr(314), FLOAT_MSG_RED);
                 end
             end
         end
-        if ((random(1, 6) == 1) and (floated == 0) and has_trait(1, self_obj, 666)) then begin
-            float_msg(self_obj, message_str(520, random(400, 409)), 2);
+        if ((random(1, 6) == 1) and (floated == 0) and obj_is_visible_flag(self_obj)) then begin
+            float_msg(self_obj, mstr(random(400, 409)), FLOAT_MSG_RED);
         end
         floated := 0;
     end
@@ -251,31 +251,31 @@ end
 
 procedure critter_p_proc
 begin
-    if (not(combat_is_initialized) and has_trait(1, self_obj, 666)) then begin
+    if (not(combat_is_initialized) and obj_is_visible_flag(self_obj)) then begin
         if (tile_distance_objs(self_obj, dude_obj) > 5) then begin
             if (tile_distance_objs(self_obj, dude_obj) > (5 * 2)) then begin
-                if ((tile_num(self_obj) != tile_num_in_direction(tile_num(dude_obj), rotation_to_tile(tile_num(dude_obj), tile_num(self_obj)), 3)) and (tile_num_in_direction(tile_num(dude_obj), rotation_to_tile(tile_num(dude_obj), tile_num(self_obj)), 3) != 0)) then begin
-                    if ((dest_tile == tile_num_in_direction(tile_num(dude_obj), rotation_to_tile(tile_num(dude_obj), tile_num(self_obj)), 3)) and (step_tile < 0)) then begin
+                if ((tile_num(self_obj) != tile_num_in_direction(dude_tile, rotation_to_tile(dude_tile, tile_num(self_obj)), 3)) and (tile_num_in_direction(dude_tile, rotation_to_tile(dude_tile, tile_num(self_obj)), 3) != 0)) then begin
+                    if ((dest_tile == tile_num_in_direction(dude_tile, rotation_to_tile(dude_tile, tile_num(self_obj)), 3)) and (step_tile < 0)) then begin
                         step_tile := step_tile + 1;
                     end
                     else begin
                         if (anim_busy(self_obj) == 0) then begin
-                            dest_tile := tile_num_in_direction(tile_num(dude_obj), rotation_to_tile(tile_num(dude_obj), tile_num(self_obj)), 3);
-                            step_tile := tile_num_in_direction(tile_num(dude_obj), rotation_to_tile(tile_num(dude_obj), tile_num(self_obj)), 3);
+                            dest_tile := tile_num_in_direction(dude_tile, rotation_to_tile(dude_tile, tile_num(self_obj)), 3);
+                            step_tile := tile_num_in_direction(dude_tile, rotation_to_tile(dude_tile, tile_num(self_obj)), 3);
                             animate_move_obj_to_tile(self_obj, step_tile, 1);
                             debug_msg("move_to_point: self_tile == " + tile_num(self_obj));
-                            while ((anim_busy(self_obj) == 0) and (tile_distance(tile_num(self_obj), tile_num_in_direction(tile_num(dude_obj), rotation_to_tile(tile_num(dude_obj), tile_num(self_obj)), 3)) >= tile_distance(step_tile, tile_num_in_direction(tile_num(dude_obj), rotation_to_tile(tile_num(dude_obj), tile_num(self_obj)), 3)))) do begin
+                            while ((anim_busy(self_obj) == 0) and (tile_distance(tile_num(self_obj), tile_num_in_direction(dude_tile, rotation_to_tile(dude_tile, tile_num(self_obj)), 3)) >= tile_distance(step_tile, tile_num_in_direction(dude_tile, rotation_to_tile(dude_tile, tile_num(self_obj)), 3)))) do begin
                                 step_tile := tile_num_in_direction(step_tile, rotation_to_tile(step_tile, tile_num(self_obj)), ((tile_distance_objs(dude_obj, self_obj) / 5) + 1) % tile_distance(step_tile, tile_num(self_obj)));
                                 debug_msg("move_to_point: step_tile == " + step_tile);
                                 animate_move_obj_to_tile(self_obj, step_tile, 1);
                             end
                             if (anim_busy(self_obj) == 0) then begin
-                                if (metarule3(106, tile_num_in_direction(tile_num(self_obj), has_trait(1, self_obj, 10), 1), elevation(self_obj), 0) != 0) then begin
+                                if (tile_get_next_critter(tile_num_in_direction(tile_num(self_obj), obj_get_rot(self_obj), 1), elevation(self_obj), 0) != 0) then begin
                                     if (anim_busy(self_obj) == 0) then begin
-                                        global_temp := rotation_to_tile(tile_num_in_direction(tile_num(self_obj), rotation_to_tile(tile_num_in_direction(tile_num(self_obj), has_trait(1, self_obj, 10), 1), tile_num(self_obj)), 1), tile_num(self_obj));
+                                        global_temp := rotation_to_tile(tile_num_in_direction(tile_num(self_obj), rotation_to_tile(tile_num_in_direction(tile_num(self_obj), obj_get_rot(self_obj), 1), tile_num(self_obj)), 1), tile_num(self_obj));
                                         animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(self_obj), global_temp, random(2, 5)), 0);
                                         global_temp := (global_temp + 1) % 6;
-                                        while ((anim_busy(self_obj) == 0) and (global_temp != rotation_to_tile(tile_num_in_direction(tile_num(self_obj), rotation_to_tile(tile_num_in_direction(tile_num(self_obj), has_trait(1, self_obj, 10), 1), tile_num(self_obj)), 1), tile_num(self_obj)))) do begin
+                                        while ((anim_busy(self_obj) == 0) and (global_temp != rotation_to_tile(tile_num_in_direction(tile_num(self_obj), rotation_to_tile(tile_num_in_direction(tile_num(self_obj), obj_get_rot(self_obj), 1), tile_num(self_obj)), 1), tile_num(self_obj)))) do begin
                                             debug_msg("flee loop: rot == " + global_temp);
                                             animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(self_obj), global_temp, random(2, 5)), 0);
                                             global_temp := (global_temp + 1) % 6;
@@ -288,7 +288,7 @@ begin
                             end
                         end
                         else begin
-                            if (tile_distance(tile_num(self_obj), tile_num_in_direction(tile_num(dude_obj), rotation_to_tile(tile_num(dude_obj), tile_num(self_obj)), 3)) < tile_distance(dest_tile, tile_num_in_direction(tile_num(dude_obj), rotation_to_tile(tile_num(dude_obj), tile_num(self_obj)), 3))) then begin
+                            if (tile_distance(tile_num(self_obj), tile_num_in_direction(dude_tile, rotation_to_tile(dude_tile, tile_num(self_obj)), 3)) < tile_distance(dest_tile, tile_num_in_direction(dude_tile, rotation_to_tile(dude_tile, tile_num(self_obj)), 3))) then begin
                                 reg_anim_func(2, self_obj);
                             end
                         end
@@ -296,28 +296,28 @@ begin
                 end
             end
             else begin
-                if ((tile_num(self_obj) != tile_num_in_direction(tile_num(dude_obj), rotation_to_tile(tile_num(dude_obj), tile_num(self_obj)), 3)) and (tile_num_in_direction(tile_num(dude_obj), rotation_to_tile(tile_num(dude_obj), tile_num(self_obj)), 3) != 0)) then begin
-                    if ((dest_tile == tile_num_in_direction(tile_num(dude_obj), rotation_to_tile(tile_num(dude_obj), tile_num(self_obj)), 3)) and (step_tile < 0)) then begin
+                if ((tile_num(self_obj) != tile_num_in_direction(dude_tile, rotation_to_tile(dude_tile, tile_num(self_obj)), 3)) and (tile_num_in_direction(dude_tile, rotation_to_tile(dude_tile, tile_num(self_obj)), 3) != 0)) then begin
+                    if ((dest_tile == tile_num_in_direction(dude_tile, rotation_to_tile(dude_tile, tile_num(self_obj)), 3)) and (step_tile < 0)) then begin
                         step_tile := step_tile + 1;
                     end
                     else begin
                         if (anim_busy(self_obj) == 0) then begin
-                            dest_tile := tile_num_in_direction(tile_num(dude_obj), rotation_to_tile(tile_num(dude_obj), tile_num(self_obj)), 3);
-                            step_tile := tile_num_in_direction(tile_num(dude_obj), rotation_to_tile(tile_num(dude_obj), tile_num(self_obj)), 3);
+                            dest_tile := tile_num_in_direction(dude_tile, rotation_to_tile(dude_tile, tile_num(self_obj)), 3);
+                            step_tile := tile_num_in_direction(dude_tile, rotation_to_tile(dude_tile, tile_num(self_obj)), 3);
                             animate_move_obj_to_tile(self_obj, step_tile, 0);
                             debug_msg("move_to_point: self_tile == " + tile_num(self_obj));
-                            while ((anim_busy(self_obj) == 0) and (tile_distance(tile_num(self_obj), tile_num_in_direction(tile_num(dude_obj), rotation_to_tile(tile_num(dude_obj), tile_num(self_obj)), 3)) >= tile_distance(step_tile, tile_num_in_direction(tile_num(dude_obj), rotation_to_tile(tile_num(dude_obj), tile_num(self_obj)), 3)))) do begin
+                            while ((anim_busy(self_obj) == 0) and (tile_distance(tile_num(self_obj), tile_num_in_direction(dude_tile, rotation_to_tile(dude_tile, tile_num(self_obj)), 3)) >= tile_distance(step_tile, tile_num_in_direction(dude_tile, rotation_to_tile(dude_tile, tile_num(self_obj)), 3)))) do begin
                                 step_tile := tile_num_in_direction(step_tile, rotation_to_tile(step_tile, tile_num(self_obj)), ((tile_distance_objs(dude_obj, self_obj) / 5) + 1) % tile_distance(step_tile, tile_num(self_obj)));
                                 debug_msg("move_to_point: step_tile == " + step_tile);
                                 animate_move_obj_to_tile(self_obj, step_tile, 0);
                             end
                             if (anim_busy(self_obj) == 0) then begin
-                                if (metarule3(106, tile_num_in_direction(tile_num(self_obj), has_trait(1, self_obj, 10), 1), elevation(self_obj), 0) != 0) then begin
+                                if (tile_get_next_critter(tile_num_in_direction(tile_num(self_obj), obj_get_rot(self_obj), 1), elevation(self_obj), 0) != 0) then begin
                                     if (anim_busy(self_obj) == 0) then begin
-                                        global_temp := rotation_to_tile(tile_num_in_direction(tile_num(self_obj), rotation_to_tile(tile_num_in_direction(tile_num(self_obj), has_trait(1, self_obj, 10), 1), tile_num(self_obj)), 1), tile_num(self_obj));
+                                        global_temp := rotation_to_tile(tile_num_in_direction(tile_num(self_obj), rotation_to_tile(tile_num_in_direction(tile_num(self_obj), obj_get_rot(self_obj), 1), tile_num(self_obj)), 1), tile_num(self_obj));
                                         animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(self_obj), global_temp, random(2, 5)), 0);
                                         global_temp := (global_temp + 1) % 6;
-                                        while ((anim_busy(self_obj) == 0) and (global_temp != rotation_to_tile(tile_num_in_direction(tile_num(self_obj), rotation_to_tile(tile_num_in_direction(tile_num(self_obj), has_trait(1, self_obj, 10), 1), tile_num(self_obj)), 1), tile_num(self_obj)))) do begin
+                                        while ((anim_busy(self_obj) == 0) and (global_temp != rotation_to_tile(tile_num_in_direction(tile_num(self_obj), rotation_to_tile(tile_num_in_direction(tile_num(self_obj), obj_get_rot(self_obj), 1), tile_num(self_obj)), 1), tile_num(self_obj)))) do begin
                                             debug_msg("flee loop: rot == " + global_temp);
                                             animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(self_obj), global_temp, random(2, 5)), 0);
                                             global_temp := (global_temp + 1) % 6;
@@ -330,7 +330,7 @@ begin
                             end
                         end
                         else begin
-                            if (tile_distance(tile_num(self_obj), tile_num_in_direction(tile_num(dude_obj), rotation_to_tile(tile_num(dude_obj), tile_num(self_obj)), 3)) < tile_distance(dest_tile, tile_num_in_direction(tile_num(dude_obj), rotation_to_tile(tile_num(dude_obj), tile_num(self_obj)), 3))) then begin
+                            if (tile_distance(tile_num(self_obj), tile_num_in_direction(dude_tile, rotation_to_tile(dude_tile, tile_num(self_obj)), 3)) < tile_distance(dest_tile, tile_num_in_direction(dude_tile, rotation_to_tile(dude_tile, tile_num(self_obj)), 3))) then begin
                                 reg_anim_func(2, self_obj);
                             end
                         end
@@ -351,43 +351,43 @@ end
 
 procedure talk_p_proc
 begin
-    if (get_critter_stat(dude_obj, 34) == 0) then begin
-        float_msg(self_obj, message_str(520, random(350, 359)), 8);
+    if (dude_gender == 0) then begin
+        float_msg(self_obj, mstr(random(350, 359)), FLOAT_MSG_YELLOW);
     end
-    if (get_critter_stat(dude_obj, 34) == 1) then begin
-        float_msg(self_obj, message_str(520, random(351, 360)), 8);
+    if (dude_gender == 1) then begin
+        float_msg(self_obj, mstr(random(351, 360)), FLOAT_MSG_YELLOW);
     end
 end
 
 procedure destroy_p_proc
 begin
-    set_global_var(930, global_var(930) - 1);
+    set_global_var(GVAR_G_HUNTERS_COUNTER, global_var(GVAR_G_HUNTERS_COUNTER) - 1);
 end
 
 procedure look_at_p_proc
 begin
     script_overrides;
-    display_msg(message_str(520, 100));
+    display_mstr(100);
 end
 
 procedure description_p_proc
 begin
     script_overrides;
-    display_msg(message_str(520, 200));
+    display_mstr(200);
 end
 
 procedure timed_event_p_proc
 begin
     if (fixed_param == 1) then begin
-        if (cur_map_index == 30) then begin
-            if ((global_var(964) bwand 1) == 0) then begin
-                set_global_var(964, global_var(964) bwor 1);
-                float_msg(self_obj, message_str(520, 300), 2);
+        if (CUR_MAP_GHOUL0) then begin
+            if ((global_var(GVAR_G_HUNTERS_FLOATS) bwand 1) == 0) then begin
+                set_global_var(GVAR_G_HUNTERS_FLOATS, global_var(GVAR_G_HUNTERS_FLOATS) bwor 1);
+                float_msg(self_obj, mstr(300), FLOAT_MSG_RED);
             end
             else begin
-                if ((global_var(964) bwand 2) == 0) then begin
-                    set_global_var(964, global_var(964) bwor 2);
-                    float_msg(self_obj, message_str(520, 301), 2);
+                if ((global_var(GVAR_G_HUNTERS_FLOATS) bwand 2) == 0) then begin
+                    set_global_var(GVAR_G_HUNTERS_FLOATS, global_var(GVAR_G_HUNTERS_FLOATS) bwor 2);
+                    float_msg(self_obj, mstr(301), FLOAT_MSG_RED);
                 end
             end
         end
@@ -401,4 +401,3 @@ end
 procedure push_p_proc
 begin
 end
-
