diff --git a/Mods/Resurrection/Original/JOHNATAN.ssl b/Mods/Resurrection/ReDefine/JOHNATAN.ssl
index fb34b51..a56f77f 100644
--- a/Mods/Resurrection/Original/JOHNATAN.ssl
+++ b/Mods/Resurrection/ReDefine/JOHNATAN.ssl
@@ -90,12 +90,12 @@ begin
             return 1;
         end
     end
-    if ((party_member_obj(16777779) != 0) and ((global_var(910) bwand 32) == 0)) then begin
+    if ((party_member_obj(16777779) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 32) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777779)) <= 5) then begin
             return 1;
         end
     end
-    if ((party_member_obj(16777811) != 0) and ((global_var(910) bwand 64) == 0)) then begin
+    if ((party_member_obj(16777811) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 64) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777811)) <= 5) then begin
             return 1;
         end
@@ -110,7 +110,7 @@ begin
             return 1;
         end
     end
-    if ((party_member_obj(16777814) != 0) and ((global_var(910) bwand 128) == 0)) then begin
+    if ((party_member_obj(16777814) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 128) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777814)) <= 5) then begin
             return 1;
         end
@@ -168,16 +168,16 @@ end
 procedure critter_p_proc
 begin
     if ((local_var(4) == 2) and obj_can_see_obj(self_obj, dude_obj)) then begin
-        float_msg(self_obj, message_str(1376, 400), 2);
+        float_msg(self_obj, mstr(400), FLOAT_MSG_RED);
         set_local_var(4, 1);
         if (critter_is_fleeing(self_obj)) then begin
             if (anim_busy(self_obj) == 0) then begin
                 if ((tile_distance_objs(self_obj, dude_obj) < 8) or obj_can_see_obj(self_obj, dude_obj)) then begin
                     if (anim_busy(self_obj) == 0) then begin
-                        global_temp := rotation_to_tile(tile_num(dude_obj), tile_num(self_obj));
+                        global_temp := rotation_to_tile(dude_tile, tile_num(self_obj));
                         animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(self_obj), global_temp, random(3, 10)), 1);
                         global_temp := (global_temp + 1) % 6;
-                        while ((anim_busy(self_obj) == 0) and (global_temp != rotation_to_tile(tile_num(dude_obj), tile_num(self_obj)))) do begin
+                        while ((anim_busy(self_obj) == 0) and (global_temp != rotation_to_tile(dude_tile, tile_num(self_obj)))) do begin
                             debug_msg("flee loop: rot == " + global_temp);
                             animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(self_obj), global_temp, random(3, 10)), 1);
                             global_temp := (global_temp + 1) % 6;
@@ -187,13 +187,13 @@ begin
             end
         end
         else begin
-            attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+            attack(dude_obj);
         end
     end
     if (map_var(0) == 1) then begin
         if (obj_can_see_obj(self_obj, dude_obj) or obj_can_hear_obj(self_obj, dude_obj)) then begin
-            float_msg(self_obj, message_str(1376, 400), 2);
-            attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+            float_msg(self_obj, mstr(400), FLOAT_MSG_RED);
+            attack(dude_obj);
         end
     end
 end
@@ -207,9 +207,9 @@ end
 
 procedure talk_p_proc
 begin
-    anim(dude_obj, 1000, rotation_to_tile(tile_num(dude_obj), tile_num(self_obj)));
-    anim(self_obj, 1000, rotation_to_tile(tile_num(self_obj), tile_num(dude_obj)));
-    if ((global_var(741) == 0) and (local_var(5) != 1)) then begin
+    anim(dude_obj, 1000, rotation_to_tile(dude_tile, tile_num(self_obj)));
+    anim(self_obj, 1000, rotation_to_tile(tile_num(self_obj), dude_tile));
+    if ((global_var(GVAR_QUEST_JOHNATAN_CAVE) == 0) and (local_var(5) != 1)) then begin
         start_gdialog(1376, self_obj, 4, -1, -1);
         gsay_start;
         call Node500();
@@ -217,7 +217,7 @@ begin
         end_dialogue;
         projed_smycku := 1;
     end
-    if ((local_var(5) == 1) and (projed_smycku == 0) and ((global_var(741) == 0) or (global_var(741) == 3))) then begin
+    if ((local_var(5) == 1) and (projed_smycku == 0) and ((global_var(GVAR_QUEST_JOHNATAN_CAVE) == 0) or (global_var(GVAR_QUEST_JOHNATAN_CAVE) == 3))) then begin
         start_gdialog(1376, self_obj, 4, -1, -1);
         gsay_start;
         call Node1650();
@@ -225,7 +225,7 @@ begin
         end_dialogue;
         projed_smycku := 1;
     end
-    if ((local_var(5) == 1) and (projed_smycku == 0) and ((global_var(741) == 1) or (global_var(741) == 2))) then begin
+    if ((local_var(5) == 1) and (projed_smycku == 0) and ((global_var(GVAR_QUEST_JOHNATAN_CAVE) == 1) or (global_var(GVAR_QUEST_JOHNATAN_CAVE) == 2))) then begin
         start_gdialog(1376, self_obj, 4, -1, -1);
         gsay_start;
         if (local_var(8) != 1) then begin
@@ -243,50 +243,50 @@ end
 
 procedure destroy_p_proc
 begin
-    set_global_var(0, global_var(0) + -10);
+    set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + -10);
     debug_msg("Player gains " + -10 + " Karma Points.");
-    set_global_var(37, 0);
-    set_global_var(38, 0);
-    set_global_var(39, 0);
-    set_global_var(40, 0);
-    set_global_var(41, 0);
-    set_global_var(42, 0);
-    set_global_var(43, 0);
-    set_global_var(44, 0);
-    set_global_var(45, 0);
-    if (global_var(0) >= 1000) then begin
-        set_global_var(37, 1);
+    set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+    set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+    set_global_var(GVAR_KARMA_DEFENDER, 0);
+    set_global_var(GVAR_KARMA_WANDERER, 0);
+    set_global_var(GVAR_KARMA_BETRAYER, 0);
+    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+    set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+    set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+    if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
     end
     else begin
-        if (global_var(0) >= 750) then begin
-            set_global_var(38, 1);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+            set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
         end
         else begin
-            if (global_var(0) >= 500) then begin
-                set_global_var(39, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
             end
             else begin
-                if (global_var(0) >= 250) then begin
-                    set_global_var(40, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                    set_global_var(GVAR_KARMA_DEFENDER, 1);
                 end
                 else begin
-                    if (global_var(0) > -250) then begin
-                        set_global_var(41, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                        set_global_var(GVAR_KARMA_WANDERER, 1);
                     end
                     else begin
-                        if (global_var(0) > -500) then begin
-                            set_global_var(42, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                            set_global_var(GVAR_KARMA_BETRAYER, 1);
                         end
                         else begin
-                            if (global_var(0) > -750) then begin
-                                set_global_var(43, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                             end
                             else begin
-                                if (global_var(0) > -1000) then begin
-                                    set_global_var(44, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                    set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                 end
                                 else begin
-                                    set_global_var(45, 1);
+                                    set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                 end
                             end
                         end
@@ -301,10 +301,10 @@ procedure look_at_p_proc
 begin
     script_overrides;
     if (local_var(5) == 1) then begin
-        display_msg(message_str(1376, 150));
+        display_mstr(150);
     end
     else begin
-        display_msg(message_str(1376, 100));
+        display_mstr(100);
     end
 end
 
@@ -312,10 +312,10 @@ procedure description_p_proc
 begin
     script_overrides;
     if (local_var(5) == 1) then begin
-        display_msg(message_str(1376, 250));
+        display_mstr(250);
     end
     else begin
-        display_msg(message_str(1376, 200));
+        display_mstr(200);
     end
 end
 
@@ -333,8 +333,8 @@ end
 
 procedure map_enter_p_proc
 begin
-    critter_add_trait(self_obj, 1, 6, 231);
-    critter_add_trait(self_obj, 1, 5, 2);
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 231);
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_PEASANTS_NO_TALKING);
 end
 
 procedure Node998
@@ -349,158 +349,158 @@ end
 
 procedure Node500
 begin
-    gsay_reply(1376, 500);
-    giq_option(4, 1376, message_str(1376, 501) + obj_name(dude_obj), Node550, 50);
-    if (global_var(740) == 1) then begin
-        giq_option(4, 1376, 502, Node800, 50);
+    Reply(500);
+    giq_option(4, NAME, mstr(501) + dude_name, Node550, NEUTRAL_REACTION);
+    if (global_var(GVAR_QUEST_S_CHARLIE_JUNK) == 1) then begin
+        giq_option(4, NAME, 502, Node800, NEUTRAL_REACTION);
     end
-    giq_option(4, 1376, 503, Node999, 50);
-    if ((get_critter_stat(dude_obj, 34) == 0) and obj_is_carrying_obj_pid(dude_obj, 98) and (global_var(740) == 1)) then begin
-        giq_option(-3, 1376, 504, Node2250, 50);
+    giq_option(4, NAME, 503, Node999, NEUTRAL_REACTION);
+    if ((dude_gender == 0) and obj_is_carrying_obj_pid(dude_obj, 98) and (global_var(GVAR_QUEST_S_CHARLIE_JUNK) == 1)) then begin
+        giq_option(-3, NAME, 504, Node2250, NEUTRAL_REACTION);
     end
-    if ((get_critter_stat(dude_obj, 34) == 1) and obj_is_carrying_obj_pid(dude_obj, 98) and (global_var(740) == 1)) then begin
-        giq_option(-3, 1376, 505, Node2250, 50);
+    if ((dude_gender == 1) and obj_is_carrying_obj_pid(dude_obj, 98) and (global_var(GVAR_QUEST_S_CHARLIE_JUNK) == 1)) then begin
+        giq_option(-3, NAME, 505, Node2250, NEUTRAL_REACTION);
     end
-    giq_option(-3, 1376, 506, Node999, 50);
+    giq_option(-3, NAME, 506, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node550
 begin
-    gsay_reply(1376, 550);
+    Reply(550);
     set_local_var(5, 1);
-    giq_option(8, 1376, 551, Node650, 50);
-    giq_option(4, 1376, 552, Node600, 50);
-    giq_option(4, 1376, 553, Node750, 50);
-    if ((get_critter_stat(dude_obj, 34) == 0) and (global_var(740) == 1)) then begin
-        giq_option(4, 1376, 554, Node850, 50);
+    giq_option(8, NAME, 551, Node650, NEUTRAL_REACTION);
+    giq_option(4, NAME, 552, Node600, NEUTRAL_REACTION);
+    giq_option(4, NAME, 553, Node750, NEUTRAL_REACTION);
+    if ((dude_gender == 0) and (global_var(GVAR_QUEST_S_CHARLIE_JUNK) == 1)) then begin
+        giq_option(4, NAME, 554, Node850, NEUTRAL_REACTION);
     end
-    if ((get_critter_stat(dude_obj, 34) == 1) and (global_var(740) == 1)) then begin
-        giq_option(4, 1376, 555, Node900, 50);
+    if ((dude_gender == 1) and (global_var(GVAR_QUEST_S_CHARLIE_JUNK) == 1)) then begin
+        giq_option(4, NAME, 555, Node900, NEUTRAL_REACTION);
     end
-    giq_option(4, 1376, 556, Node999, 50);
+    giq_option(4, NAME, 556, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node600
 begin
-    gsay_reply(1376, 600);
-    giq_option(8, 1376, 601, Node650, 50);
-    giq_option(4, 1376, 602, Node750, 50);
-    if ((get_critter_stat(dude_obj, 34) == 0) and (global_var(740) == 1) and (local_var(6) != 1)) then begin
-        giq_option(4, 1376, 603, Node850, 50);
+    Reply(600);
+    giq_option(8, NAME, 601, Node650, NEUTRAL_REACTION);
+    giq_option(4, NAME, 602, Node750, NEUTRAL_REACTION);
+    if ((dude_gender == 0) and (global_var(GVAR_QUEST_S_CHARLIE_JUNK) == 1) and (local_var(6) != 1)) then begin
+        giq_option(4, NAME, 603, Node850, NEUTRAL_REACTION);
     end
-    if ((get_critter_stat(dude_obj, 34) == 1) and (global_var(740) == 1) and (local_var(6) != 1)) then begin
-        giq_option(4, 1376, 604, Node900, 50);
+    if ((dude_gender == 1) and (global_var(GVAR_QUEST_S_CHARLIE_JUNK) == 1) and (local_var(6) != 1)) then begin
+        giq_option(4, NAME, 604, Node900, NEUTRAL_REACTION);
     end
-    giq_option(4, 1376, 605, Node999, 50);
+    giq_option(4, NAME, 605, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node650
 begin
-    gsay_reply(1376, 650);
-    giq_option(4, 1376, 651, Node600, 50);
-    giq_option(4, 1376, 652, Node700, 50);
-    giq_option(4, 1376, 653, Node750, 50);
-    if ((get_critter_stat(dude_obj, 34) == 0) and (global_var(740) == 1) and (local_var(6) != 1)) then begin
-        giq_option(4, 1376, 654, Node850, 50);
+    Reply(650);
+    giq_option(4, NAME, 651, Node600, NEUTRAL_REACTION);
+    giq_option(4, NAME, 652, Node700, NEUTRAL_REACTION);
+    giq_option(4, NAME, 653, Node750, NEUTRAL_REACTION);
+    if ((dude_gender == 0) and (global_var(GVAR_QUEST_S_CHARLIE_JUNK) == 1) and (local_var(6) != 1)) then begin
+        giq_option(4, NAME, 654, Node850, NEUTRAL_REACTION);
     end
-    if ((get_critter_stat(dude_obj, 34) == 1) and (global_var(740) == 1) and (local_var(6) != 1)) then begin
-        giq_option(4, 1376, 655, Node900, 50);
+    if ((dude_gender == 1) and (global_var(GVAR_QUEST_S_CHARLIE_JUNK) == 1) and (local_var(6) != 1)) then begin
+        giq_option(4, NAME, 655, Node900, NEUTRAL_REACTION);
     end
-    giq_option(4, 1376, 656, Node999, 50);
+    giq_option(4, NAME, 656, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node700
 begin
-    gsay_reply(1376, 700);
-    giq_option(4, 1376, 701, Node750, 50);
-    giq_option(4, 1376, 702, Node600, 50);
-    if ((get_critter_stat(dude_obj, 34) == 0) and (global_var(740) == 1) and (local_var(6) != 1)) then begin
-        giq_option(4, 1376, 703, Node850, 50);
+    Reply(700);
+    giq_option(4, NAME, 701, Node750, NEUTRAL_REACTION);
+    giq_option(4, NAME, 702, Node600, NEUTRAL_REACTION);
+    if ((dude_gender == 0) and (global_var(GVAR_QUEST_S_CHARLIE_JUNK) == 1) and (local_var(6) != 1)) then begin
+        giq_option(4, NAME, 703, Node850, NEUTRAL_REACTION);
     end
-    if ((get_critter_stat(dude_obj, 34) == 1) and (global_var(740) == 1) and (local_var(6) != 1)) then begin
-        giq_option(4, 1376, 704, Node900, 50);
+    if ((dude_gender == 1) and (global_var(GVAR_QUEST_S_CHARLIE_JUNK) == 1) and (local_var(6) != 1)) then begin
+        giq_option(4, NAME, 704, Node900, NEUTRAL_REACTION);
     end
-    giq_option(4, 1376, 705, Node999, 50);
+    giq_option(4, NAME, 705, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node750
 begin
-    gsay_reply(1376, 750);
-    giq_option(8, 1376, 751, Node650, 50);
-    giq_option(4, 1376, 752, Node600, 50);
-    if ((get_critter_stat(dude_obj, 34) == 0) and (global_var(740) == 1) and (local_var(6) != 1)) then begin
-        giq_option(4, 1376, 753, Node850, 50);
+    Reply(750);
+    giq_option(8, NAME, 751, Node650, NEUTRAL_REACTION);
+    giq_option(4, NAME, 752, Node600, NEUTRAL_REACTION);
+    if ((dude_gender == 0) and (global_var(GVAR_QUEST_S_CHARLIE_JUNK) == 1) and (local_var(6) != 1)) then begin
+        giq_option(4, NAME, 753, Node850, NEUTRAL_REACTION);
     end
-    if ((get_critter_stat(dude_obj, 34) == 1) and (global_var(740) == 1) and (local_var(6) != 1)) then begin
-        giq_option(4, 1376, 754, Node900, 50);
+    if ((dude_gender == 1) and (global_var(GVAR_QUEST_S_CHARLIE_JUNK) == 1) and (local_var(6) != 1)) then begin
+        giq_option(4, NAME, 754, Node900, NEUTRAL_REACTION);
     end
-    giq_option(4, 1376, 755, Node999, 50);
+    giq_option(4, NAME, 755, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node800
 begin
-    gsay_reply(1376, 800);
+    Reply(800);
     set_local_var(5, 1);
-    if ((get_critter_stat(dude_obj, 34) == 0) and (global_var(740) == 1)) then begin
-        giq_option(4, 1376, 801, Node850, 50);
+    if ((dude_gender == 0) and (global_var(GVAR_QUEST_S_CHARLIE_JUNK) == 1)) then begin
+        giq_option(4, NAME, 801, Node850, NEUTRAL_REACTION);
     end
-    if ((get_critter_stat(dude_obj, 34) == 1) and (global_var(740) == 1)) then begin
-        giq_option(4, 1376, 802, Node900, 50);
+    if ((dude_gender == 1) and (global_var(GVAR_QUEST_S_CHARLIE_JUNK) == 1)) then begin
+        giq_option(4, NAME, 802, Node900, NEUTRAL_REACTION);
     end
-    giq_option(4, 1376, 803, Node999, 50);
+    giq_option(4, NAME, 803, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node850
 begin
-    gsay_reply(1376, 850);
+    Reply(850);
     set_local_var(6, 1);
     if (obj_is_carrying_obj_pid(dude_obj, 98)) then begin
-        giq_option(4, 1376, 851, Node1000, 50);
+        giq_option(4, NAME, 851, Node1000, NEUTRAL_REACTION);
     end
     if (not(obj_is_carrying_obj_pid(dude_obj, 98))) then begin
-        giq_option(4, 1376, 852, Node950, 50);
+        giq_option(4, NAME, 852, Node950, NEUTRAL_REACTION);
     end
 end
 
 procedure Node900
 begin
-    gsay_reply(1376, 900);
+    Reply(900);
     set_local_var(6, 1);
     if (obj_is_carrying_obj_pid(dude_obj, 98)) then begin
-        giq_option(4, 1376, 901, Node1000, 50);
+        giq_option(4, NAME, 901, Node1000, NEUTRAL_REACTION);
     end
     if (not(obj_is_carrying_obj_pid(dude_obj, 98))) then begin
-        giq_option(4, 1376, 902, Node950, 50);
+        giq_option(4, NAME, 902, Node950, NEUTRAL_REACTION);
     end
 end
 
 procedure Node950
 begin
-    gsay_reply(1376, 950);
-    if (get_critter_stat(dude_obj, 34) == 0) then begin
-        giq_option(4, 1376, 951, Node1200, 50);
+    Reply(950);
+    if (dude_gender == 0) then begin
+        giq_option(4, NAME, 951, Node1200, NEUTRAL_REACTION);
     end
-    if (get_critter_stat(dude_obj, 34) == 1) then begin
-        giq_option(4, 1376, 952, Node1200, 50);
+    if (dude_gender == 1) then begin
+        giq_option(4, NAME, 952, Node1200, NEUTRAL_REACTION);
     end
-    giq_option(4, 1376, 953, Node999, 50);
+    giq_option(4, NAME, 953, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node1000
 begin
-    gsay_reply(1376, 1000);
-    giq_option(4, 1376, 400 + message_str(1376, 1001), Node1050, 50);
-    if (has_skill(dude_obj, 15) <= 50) then begin
-        giq_option(4, 1376, 600 + message_str(1376, 1002), Node1100, 50);
+    Reply(1000);
+    giq_option(4, NAME, 400 + mstr(1001), Node1050, NEUTRAL_REACTION);
+    if (dude_skill(SKILL_BARTER) <= 50) then begin
+        giq_option(4, NAME, 600 + mstr(1002), Node1100, NEUTRAL_REACTION);
     end
-    if (has_skill(dude_obj, 15) > 50) then begin
-        giq_option(4, 1376, 600 + message_str(1376, 1002), Node1150, 50);
+    if (dude_skill(SKILL_BARTER) > 50) then begin
+        giq_option(4, NAME, 600 + mstr(1002), Node1150, NEUTRAL_REACTION);
     end
 end
 
 procedure Node1050
 begin
-    gsay_reply(1376, 1050);
+    Reply(1050);
     removed_qty := obj_is_carrying_obj_pid(dude_obj, 98);
     if (1 < removed_qty) then begin
         removed_qty := 1;
@@ -509,17 +509,17 @@ begin
         restock_obj := obj_carrying_pid_obj(dude_obj, 98);
         if (obj_type(self_obj) == 1) then begin
             restock_obj := obj_carrying_pid_obj(dude_obj, 98);
-            if (critter_inven_obj(self_obj, 0) == restock_obj) then begin
-                if (obj_item_subtype(critter_inven_obj(dude_obj, 0)) == 0) then begin
-                    restock_obj := critter_inven_obj(dude_obj, 0);
+            if (critter_inven_obj(self_obj, INVEN_TYPE_WORN) == restock_obj) then begin
+                if (obj_item_subtype(critter_inven_obj(dude_obj, INVEN_TYPE_WORN)) == 0) then begin
+                    restock_obj := critter_inven_obj(dude_obj, INVEN_TYPE_WORN);
                     debug_msg("armour pid == " + obj_pid(restock_obj));
                     rm_obj_from_inven(dude_obj, restock_obj);
                     add_obj_to_inven(dude_obj, restock_obj);
                 end
             end
             else begin
-                if ((critter_inven_obj(dude_obj, 2) == restock_obj) or (critter_inven_obj(dude_obj, 1) == restock_obj)) then begin
-                    metarule(43, dude_obj);
+                if ((critter_inven_obj(dude_obj, INVEN_TYPE_LEFT_HAND) == restock_obj) or (critter_inven_obj(dude_obj, INVEN_TYPE_RIGHT_HAND) == restock_obj)) then begin
+                    inven_unwield(dude_obj);
                 end
             end
         end
@@ -539,29 +539,29 @@ begin
     end
     restock_obj := 0;
     restock_amt := 0;
-    restock_obj := create_object_sid(98, 0, 0, -1);
+    restock_obj := create_object(98, 0, 0);
     add_mult_objs_to_inven(self_obj, restock_obj, 1);
-    set_global_var(740, 2);
+    set_global_var(GVAR_QUEST_S_CHARLIE_JUNK, 2);
     item_caps_adjust(dude_obj, 400);
     give_exp_points(500);
-    display_msg(message_str(14, 100) + 500 + message_str(14, 101));
-    if (get_critter_stat(dude_obj, 34) == 0) then begin
-        giq_option(4, 1376, 1051, Node1200, 50);
+    display_msg(g_mstr(100) + 500 + g_mstr(101));
+    if (dude_gender == 0) then begin
+        giq_option(4, NAME, 1051, Node1200, NEUTRAL_REACTION);
     end
-    if (get_critter_stat(dude_obj, 34) == 1) then begin
-        giq_option(4, 1376, 1052, Node1200, 50);
+    if (dude_gender == 1) then begin
+        giq_option(4, NAME, 1052, Node1200, NEUTRAL_REACTION);
     end
-    if (get_critter_stat(dude_obj, 34) == 0) then begin
-        giq_option(4, 1376, 1053, Node1350, 50);
+    if (dude_gender == 0) then begin
+        giq_option(4, NAME, 1053, Node1350, NEUTRAL_REACTION);
     end
-    if (get_critter_stat(dude_obj, 34) == 1) then begin
-        giq_option(4, 1376, 1054, Node1400, 50);
+    if (dude_gender == 1) then begin
+        giq_option(4, NAME, 1054, Node1400, NEUTRAL_REACTION);
     end
 end
 
 procedure Node1100
 begin
-    gsay_reply(1376, message_str(1376, 1100) + 600 + message_str(1376, 1101) + 400 + message_str(1376, 1102));
+    Reply(mstr(1100) + 600 + mstr(1101) + 400 + mstr(1102));
     removed_qty := obj_is_carrying_obj_pid(dude_obj, 98);
     if (1 < removed_qty) then begin
         removed_qty := 1;
@@ -570,17 +570,17 @@ begin
         restock_obj := obj_carrying_pid_obj(dude_obj, 98);
         if (obj_type(self_obj) == 1) then begin
             restock_obj := obj_carrying_pid_obj(dude_obj, 98);
-            if (critter_inven_obj(self_obj, 0) == restock_obj) then begin
-                if (obj_item_subtype(critter_inven_obj(dude_obj, 0)) == 0) then begin
-                    restock_obj := critter_inven_obj(dude_obj, 0);
+            if (critter_inven_obj(self_obj, INVEN_TYPE_WORN) == restock_obj) then begin
+                if (obj_item_subtype(critter_inven_obj(dude_obj, INVEN_TYPE_WORN)) == 0) then begin
+                    restock_obj := critter_inven_obj(dude_obj, INVEN_TYPE_WORN);
                     debug_msg("armour pid == " + obj_pid(restock_obj));
                     rm_obj_from_inven(dude_obj, restock_obj);
                     add_obj_to_inven(dude_obj, restock_obj);
                 end
             end
             else begin
-                if ((critter_inven_obj(dude_obj, 2) == restock_obj) or (critter_inven_obj(dude_obj, 1) == restock_obj)) then begin
-                    metarule(43, dude_obj);
+                if ((critter_inven_obj(dude_obj, INVEN_TYPE_LEFT_HAND) == restock_obj) or (critter_inven_obj(dude_obj, INVEN_TYPE_RIGHT_HAND) == restock_obj)) then begin
+                    inven_unwield(dude_obj);
                 end
             end
         end
@@ -600,29 +600,29 @@ begin
     end
     restock_obj := 0;
     restock_amt := 0;
-    restock_obj := create_object_sid(98, 0, 0, -1);
+    restock_obj := create_object(98, 0, 0);
     add_mult_objs_to_inven(self_obj, restock_obj, 1);
-    set_global_var(740, 2);
+    set_global_var(GVAR_QUEST_S_CHARLIE_JUNK, 2);
     give_exp_points(500);
-    display_msg(message_str(14, 100) + 500 + message_str(14, 101));
+    display_msg(g_mstr(100) + 500 + g_mstr(101));
     item_caps_adjust(dude_obj, 400);
-    if (get_critter_stat(dude_obj, 34) == 0) then begin
-        giq_option(4, 1376, 1103, Node1200, 50);
+    if (dude_gender == 0) then begin
+        giq_option(4, NAME, 1103, Node1200, NEUTRAL_REACTION);
     end
-    if (get_critter_stat(dude_obj, 34) == 1) then begin
-        giq_option(4, 1376, 1104, Node1200, 50);
+    if (dude_gender == 1) then begin
+        giq_option(4, NAME, 1104, Node1200, NEUTRAL_REACTION);
     end
-    if (get_critter_stat(dude_obj, 34) == 0) then begin
-        giq_option(4, 1376, 1105, Node1350, 50);
+    if (dude_gender == 0) then begin
+        giq_option(4, NAME, 1105, Node1350, NEUTRAL_REACTION);
     end
-    if (get_critter_stat(dude_obj, 34) == 1) then begin
-        giq_option(4, 1376, 1106, Node1400, 50);
+    if (dude_gender == 1) then begin
+        giq_option(4, NAME, 1106, Node1400, NEUTRAL_REACTION);
     end
 end
 
 procedure Node1150
 begin
-    gsay_reply(1376, 1150);
+    Reply(1150);
     removed_qty := obj_is_carrying_obj_pid(dude_obj, 98);
     if (1 < removed_qty) then begin
         removed_qty := 1;
@@ -631,17 +631,17 @@ begin
         restock_obj := obj_carrying_pid_obj(dude_obj, 98);
         if (obj_type(self_obj) == 1) then begin
             restock_obj := obj_carrying_pid_obj(dude_obj, 98);
-            if (critter_inven_obj(self_obj, 0) == restock_obj) then begin
-                if (obj_item_subtype(critter_inven_obj(dude_obj, 0)) == 0) then begin
-                    restock_obj := critter_inven_obj(dude_obj, 0);
+            if (critter_inven_obj(self_obj, INVEN_TYPE_WORN) == restock_obj) then begin
+                if (obj_item_subtype(critter_inven_obj(dude_obj, INVEN_TYPE_WORN)) == 0) then begin
+                    restock_obj := critter_inven_obj(dude_obj, INVEN_TYPE_WORN);
                     debug_msg("armour pid == " + obj_pid(restock_obj));
                     rm_obj_from_inven(dude_obj, restock_obj);
                     add_obj_to_inven(dude_obj, restock_obj);
                 end
             end
             else begin
-                if ((critter_inven_obj(dude_obj, 2) == restock_obj) or (critter_inven_obj(dude_obj, 1) == restock_obj)) then begin
-                    metarule(43, dude_obj);
+                if ((critter_inven_obj(dude_obj, INVEN_TYPE_LEFT_HAND) == restock_obj) or (critter_inven_obj(dude_obj, INVEN_TYPE_RIGHT_HAND) == restock_obj)) then begin
+                    inven_unwield(dude_obj);
                 end
             end
         end
@@ -661,251 +661,251 @@ begin
     end
     restock_obj := 0;
     restock_amt := 0;
-    restock_obj := create_object_sid(98, 0, 0, -1);
+    restock_obj := create_object(98, 0, 0);
     add_mult_objs_to_inven(self_obj, restock_obj, 1);
-    set_global_var(740, 2);
+    set_global_var(GVAR_QUEST_S_CHARLIE_JUNK, 2);
     give_exp_points(500);
-    display_msg(message_str(14, 100) + 500 + message_str(14, 101));
+    display_msg(g_mstr(100) + 500 + g_mstr(101));
     item_caps_adjust(dude_obj, 600);
-    if (get_critter_stat(dude_obj, 34) == 0) then begin
-        giq_option(4, 1376, 1151, Node1200, 50);
+    if (dude_gender == 0) then begin
+        giq_option(4, NAME, 1151, Node1200, NEUTRAL_REACTION);
     end
-    if (get_critter_stat(dude_obj, 34) == 1) then begin
-        giq_option(4, 1376, 1152, Node1200, 50);
+    if (dude_gender == 1) then begin
+        giq_option(4, NAME, 1152, Node1200, NEUTRAL_REACTION);
     end
-    if (get_critter_stat(dude_obj, 34) == 0) then begin
-        giq_option(4, 1376, 1153, Node1350, 50);
+    if (dude_gender == 0) then begin
+        giq_option(4, NAME, 1153, Node1350, NEUTRAL_REACTION);
     end
-    if (get_critter_stat(dude_obj, 34) == 1) then begin
-        giq_option(4, 1376, 1154, Node1400, 50);
+    if (dude_gender == 1) then begin
+        giq_option(4, NAME, 1154, Node1400, NEUTRAL_REACTION);
     end
 end
 
 procedure Node1200
 begin
-    gsay_reply(1376, 1200);
-    giq_option(4, 1376, 1201, Node1250, 50);
-    giq_option(4, 1376, 1202, Node1300, 50);
-    if (global_var(740) != 2) then begin
-        giq_option(4, 1376, 1203, Node999, 50);
+    Reply(1200);
+    giq_option(4, NAME, 1201, Node1250, NEUTRAL_REACTION);
+    giq_option(4, NAME, 1202, Node1300, NEUTRAL_REACTION);
+    if (global_var(GVAR_QUEST_S_CHARLIE_JUNK) != 2) then begin
+        giq_option(4, NAME, 1203, Node999, NEUTRAL_REACTION);
     end
-    if ((get_critter_stat(dude_obj, 34) == 0) and (global_var(740) == 2)) then begin
-        giq_option(4, 1376, 1203, Node1350, 50);
+    if ((dude_gender == 0) and (global_var(GVAR_QUEST_S_CHARLIE_JUNK) == 2)) then begin
+        giq_option(4, NAME, 1203, Node1350, NEUTRAL_REACTION);
     end
-    if ((get_critter_stat(dude_obj, 34) == 1) and (global_var(740) == 2)) then begin
-        giq_option(4, 1376, 1204, Node1400, 50);
+    if ((dude_gender == 1) and (global_var(GVAR_QUEST_S_CHARLIE_JUNK) == 2)) then begin
+        giq_option(4, NAME, 1204, Node1400, NEUTRAL_REACTION);
     end
 end
 
 procedure Node1250
 begin
-    gsay_reply(1376, 1250);
-    giq_option(4, 1376, 1251, Node1300, 50);
-    if (global_var(740) != 2) then begin
-        giq_option(4, 1376, 1252, Node999, 50);
+    Reply(1250);
+    giq_option(4, NAME, 1251, Node1300, NEUTRAL_REACTION);
+    if (global_var(GVAR_QUEST_S_CHARLIE_JUNK) != 2) then begin
+        giq_option(4, NAME, 1252, Node999, NEUTRAL_REACTION);
     end
-    if ((get_critter_stat(dude_obj, 34) == 0) and (global_var(740) == 2)) then begin
-        giq_option(4, 1376, 1252, Node1350, 50);
+    if ((dude_gender == 0) and (global_var(GVAR_QUEST_S_CHARLIE_JUNK) == 2)) then begin
+        giq_option(4, NAME, 1252, Node1350, NEUTRAL_REACTION);
     end
-    if ((get_critter_stat(dude_obj, 34) == 1) and (global_var(740) == 2)) then begin
-        giq_option(4, 1376, 1253, Node1400, 50);
+    if ((dude_gender == 1) and (global_var(GVAR_QUEST_S_CHARLIE_JUNK) == 2)) then begin
+        giq_option(4, NAME, 1253, Node1400, NEUTRAL_REACTION);
     end
 end
 
 procedure Node1300
 begin
-    gsay_reply(1376, 1300);
-    giq_option(4, 1376, 1301, Node1250, 50);
-    if (global_var(740) != 2) then begin
-        giq_option(4, 1376, 1302, Node999, 50);
+    Reply(1300);
+    giq_option(4, NAME, 1301, Node1250, NEUTRAL_REACTION);
+    if (global_var(GVAR_QUEST_S_CHARLIE_JUNK) != 2) then begin
+        giq_option(4, NAME, 1302, Node999, NEUTRAL_REACTION);
     end
-    if ((get_critter_stat(dude_obj, 34) == 0) and (global_var(740) == 2)) then begin
-        giq_option(4, 1376, 1302, Node1350, 50);
+    if ((dude_gender == 0) and (global_var(GVAR_QUEST_S_CHARLIE_JUNK) == 2)) then begin
+        giq_option(4, NAME, 1302, Node1350, NEUTRAL_REACTION);
     end
-    if ((get_critter_stat(dude_obj, 34) == 1) and (global_var(740) == 2)) then begin
-        giq_option(4, 1376, 1303, Node1400, 50);
+    if ((dude_gender == 1) and (global_var(GVAR_QUEST_S_CHARLIE_JUNK) == 2)) then begin
+        giq_option(4, NAME, 1303, Node1400, NEUTRAL_REACTION);
     end
 end
 
 procedure Node1350
 begin
-    gsay_reply(1376, 1350);
-    giq_option(4, 1376, 1351, Node1450, 50);
+    Reply(1350);
+    giq_option(4, NAME, 1351, Node1450, NEUTRAL_REACTION);
 end
 
 procedure Node1400
 begin
-    gsay_reply(1376, 1400);
-    giq_option(4, 1376, 1401, Node1450, 50);
+    Reply(1400);
+    giq_option(4, NAME, 1401, Node1450, NEUTRAL_REACTION);
 end
 
 procedure Node1450
 begin
-    gsay_reply(1376, 1450);
-    giq_option(4, 1376, 1451, Node1500, 50);
+    Reply(1450);
+    giq_option(4, NAME, 1451, Node1500, NEUTRAL_REACTION);
 end
 
 procedure Node1500
 begin
-    gsay_reply(1376, 1500);
-    giq_option(4, 1376, 1501, Node1550, 50);
-    giq_option(4, 1376, 1502, Node1600, 50);
+    Reply(1500);
+    giq_option(4, NAME, 1501, Node1550, NEUTRAL_REACTION);
+    giq_option(4, NAME, 1502, Node1600, NEUTRAL_REACTION);
 end
 
 procedure Node1550
 begin
-    gsay_reply(1376, 1550);
-    set_global_var(741, 1);
-    if (metarule(17, 5) == 0) then begin
+    Reply(1550);
+    set_global_var(GVAR_QUEST_JOHNATAN_CAVE, 1);
+    if (town_known(5) == MARK_STATE_UNKNOWN) then begin
         debug_msg("  mark_on_map(" + 5 + ")");
-        mark_area_known(0, 5, 1);
+        mark_area_known(MARK_TYPE_TOWN, 5, MARK_STATE_KNOWN);
     end
-    giq_option(4, 1376, 1551, Node999, 50);
+    giq_option(4, NAME, 1551, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node1600
 begin
-    gsay_reply(1376, 1600);
+    Reply(1600);
     set_local_var(7, 1);
-    giq_option(4, 1376, 1601, Node999, 50);
+    giq_option(4, NAME, 1601, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node1650
 begin
-    gsay_reply(1376, 1650);
-    if (get_critter_stat(dude_obj, 34) == 0) then begin
-        giq_option(4, 1376, 1651, Node1700, 50);
+    Reply(1650);
+    if (dude_gender == 0) then begin
+        giq_option(4, NAME, 1651, Node1700, NEUTRAL_REACTION);
     end
-    if (get_critter_stat(dude_obj, 34) == 1) then begin
-        giq_option(4, 1376, 1652, Node1700, 50);
+    if (dude_gender == 1) then begin
+        giq_option(4, NAME, 1652, Node1700, NEUTRAL_REACTION);
     end
-    if (global_var(740) == 2) then begin
-        giq_option(4, 1376, 1653, Node1750, 50);
+    if (global_var(GVAR_QUEST_S_CHARLIE_JUNK) == 2) then begin
+        giq_option(4, NAME, 1653, Node1750, NEUTRAL_REACTION);
     end
-    if ((get_critter_stat(dude_obj, 34) == 0) and (local_var(6) != 1) and (global_var(740) == 1)) then begin
-        giq_option(4, 1376, 1654, Node850, 50);
+    if ((dude_gender == 0) and (local_var(6) != 1) and (global_var(GVAR_QUEST_S_CHARLIE_JUNK) == 1)) then begin
+        giq_option(4, NAME, 1654, Node850, NEUTRAL_REACTION);
     end
-    if ((get_critter_stat(dude_obj, 34) == 1) and (local_var(6) != 1) and (global_var(740) == 1)) then begin
-        giq_option(4, 1376, 1655, Node900, 50);
+    if ((dude_gender == 1) and (local_var(6) != 1) and (global_var(GVAR_QUEST_S_CHARLIE_JUNK) == 1)) then begin
+        giq_option(4, NAME, 1655, Node900, NEUTRAL_REACTION);
     end
-    if ((local_var(6) == 1) and obj_is_carrying_obj_pid(dude_obj, 98) and (global_var(740) == 1)) then begin
-        giq_option(4, 1376, 1656, Node1000, 50);
+    if ((local_var(6) == 1) and obj_is_carrying_obj_pid(dude_obj, 98) and (global_var(GVAR_QUEST_S_CHARLIE_JUNK) == 1)) then begin
+        giq_option(4, NAME, 1656, Node1000, NEUTRAL_REACTION);
     end
-    if ((get_critter_stat(dude_obj, 34) == 0) and (global_var(741) == 0) and (local_var(7) == 1)) then begin
-        giq_option(4, 1376, 1657, Node1850, 50);
+    if ((dude_gender == 0) and (global_var(GVAR_QUEST_JOHNATAN_CAVE) == 0) and (local_var(7) == 1)) then begin
+        giq_option(4, NAME, 1657, Node1850, NEUTRAL_REACTION);
     end
-    if ((get_critter_stat(dude_obj, 34) == 1) and (global_var(741) == 0) and (local_var(7) == 1)) then begin
-        giq_option(4, 1376, 1658, Node1850, 50);
+    if ((dude_gender == 1) and (global_var(GVAR_QUEST_JOHNATAN_CAVE) == 0) and (local_var(7) == 1)) then begin
+        giq_option(4, NAME, 1658, Node1850, NEUTRAL_REACTION);
     end
-    giq_option(4, 1376, 1659, Node999, 50);
-    if ((global_var(741) == 0) and (local_var(7) == 1)) then begin
-        giq_option(-3, 1376, 1660, Node2450, 50);
+    giq_option(4, NAME, 1659, Node999, NEUTRAL_REACTION);
+    if ((global_var(GVAR_QUEST_JOHNATAN_CAVE) == 0) and (local_var(7) == 1)) then begin
+        giq_option(-3, NAME, 1660, Node2450, NEUTRAL_REACTION);
     end
-    giq_option(-3, 1376, 1661, Node999, 50);
+    giq_option(-3, NAME, 1661, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node1700
 begin
-    gsay_reply(1376, 1700);
-    giq_option(8, 1376, 1701, Node650, 50);
-    giq_option(4, 1376, 1702, Node600, 50);
-    giq_option(4, 1376, 1703, Node750, 50);
-    giq_option(4, 1376, 1704, Node999, 50);
+    Reply(1700);
+    giq_option(8, NAME, 1701, Node650, NEUTRAL_REACTION);
+    giq_option(4, NAME, 1702, Node600, NEUTRAL_REACTION);
+    giq_option(4, NAME, 1703, Node750, NEUTRAL_REACTION);
+    giq_option(4, NAME, 1704, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node1750
 begin
-    gsay_reply(1376, 1750);
-    giq_option(4, 1376, 1751, Node1800, 50);
-    giq_option(4, 1376, 1752, Node999, 50);
+    Reply(1750);
+    giq_option(4, NAME, 1751, Node1800, NEUTRAL_REACTION);
+    giq_option(4, NAME, 1752, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node1800
 begin
-    gsay_reply(1376, 1800);
-    giq_option(4, 1376, 1801, Node1750, 50);
-    giq_option(4, 1376, 1802, Node999, 50);
+    Reply(1800);
+    giq_option(4, NAME, 1801, Node1750, NEUTRAL_REACTION);
+    giq_option(4, NAME, 1802, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node1850
 begin
-    gsay_reply(1376, 1850);
-    set_global_var(741, 1);
-    if (metarule(17, 5) == 0) then begin
+    Reply(1850);
+    set_global_var(GVAR_QUEST_JOHNATAN_CAVE, 1);
+    if (town_known(5) == MARK_STATE_UNKNOWN) then begin
         debug_msg("  mark_on_map(" + 5 + ")");
-        mark_area_known(0, 5, 1);
+        mark_area_known(MARK_TYPE_TOWN, 5, MARK_STATE_KNOWN);
     end
-    giq_option(4, 1376, 1851, Node999, 50);
+    giq_option(4, NAME, 1851, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node1900
 begin
-    gsay_reply(1376, 1900);
-    if ((get_critter_stat(dude_obj, 34) == 0) and (global_var(222) == 1)) then begin
-        giq_option(4, 1376, 1901, Node1950, 50);
+    Reply(1900);
+    if ((dude_gender == 0) and (global_var(GVAR_HOLODISK_MB_LEVEL_2) == 1)) then begin
+        giq_option(4, NAME, 1901, Node1950, NEUTRAL_REACTION);
     end
-    if ((get_critter_stat(dude_obj, 34) == 1) and (global_var(222) == 1)) then begin
-        giq_option(4, 1376, 1902, Node2000, 50);
+    if ((dude_gender == 1) and (global_var(GVAR_HOLODISK_MB_LEVEL_2) == 1)) then begin
+        giq_option(4, NAME, 1902, Node2000, NEUTRAL_REACTION);
     end
-    if ((get_critter_stat(dude_obj, 34) == 0) and (global_var(741) == 2) and obj_carrying_pid_obj(Johnatan_box, 560) and (global_var(222) == 0)) then begin
-        giq_option(4, 1376, 1903, Node2050, 50);
+    if ((dude_gender == 0) and (global_var(GVAR_QUEST_JOHNATAN_CAVE) == 2) and obj_carrying_pid_obj(Johnatan_box, 560) and (global_var(GVAR_HOLODISK_MB_LEVEL_2) == 0)) then begin
+        giq_option(4, NAME, 1903, Node2050, NEUTRAL_REACTION);
     end
-    if ((get_critter_stat(dude_obj, 34) == 0) and (global_var(741) == 2) and not(obj_carrying_pid_obj(Johnatan_box, 560)) and (global_var(222) == 0)) then begin
-        giq_option(4, 1376, 1903, Node2100, 50);
+    if ((dude_gender == 0) and (global_var(GVAR_QUEST_JOHNATAN_CAVE) == 2) and not(obj_carrying_pid_obj(Johnatan_box, 560)) and (global_var(GVAR_HOLODISK_MB_LEVEL_2) == 0)) then begin
+        giq_option(4, NAME, 1903, Node2100, NEUTRAL_REACTION);
     end
-    if ((get_critter_stat(dude_obj, 34) == 1) and (global_var(741) == 2) and obj_carrying_pid_obj(Johnatan_box, 560) and (global_var(222) == 0)) then begin
-        giq_option(4, 1376, 1904, Node2050, 50);
+    if ((dude_gender == 1) and (global_var(GVAR_QUEST_JOHNATAN_CAVE) == 2) and obj_carrying_pid_obj(Johnatan_box, 560) and (global_var(GVAR_HOLODISK_MB_LEVEL_2) == 0)) then begin
+        giq_option(4, NAME, 1904, Node2050, NEUTRAL_REACTION);
     end
-    if ((get_critter_stat(dude_obj, 34) == 1) and (global_var(741) == 2) and not(obj_carrying_pid_obj(Johnatan_box, 560)) and (global_var(222) == 0)) then begin
-        giq_option(4, 1376, 1904, Node2100, 50);
+    if ((dude_gender == 1) and (global_var(GVAR_QUEST_JOHNATAN_CAVE) == 2) and not(obj_carrying_pid_obj(Johnatan_box, 560)) and (global_var(GVAR_HOLODISK_MB_LEVEL_2) == 0)) then begin
+        giq_option(4, NAME, 1904, Node2100, NEUTRAL_REACTION);
     end
-    if (get_critter_stat(dude_obj, 34) == 0) then begin
-        giq_option(4, 1376, 1905, Node1700, 50);
+    if (dude_gender == 0) then begin
+        giq_option(4, NAME, 1905, Node1700, NEUTRAL_REACTION);
     end
-    if (get_critter_stat(dude_obj, 34) == 1) then begin
-        giq_option(4, 1376, 1906, Node1700, 50);
+    if (dude_gender == 1) then begin
+        giq_option(4, NAME, 1906, Node1700, NEUTRAL_REACTION);
     end
-    giq_option(4, 1376, 1907, Node1750, 50);
-    giq_option(4, 1376, 1908, Node999, 50);
-    if (global_var(741) == 2) then begin
-        giq_option(-3, 1376, 1909, Node2500, 50);
+    giq_option(4, NAME, 1907, Node1750, NEUTRAL_REACTION);
+    giq_option(4, NAME, 1908, Node999, NEUTRAL_REACTION);
+    if (global_var(GVAR_QUEST_JOHNATAN_CAVE) == 2) then begin
+        giq_option(-3, NAME, 1909, Node2500, NEUTRAL_REACTION);
     end
-    if (global_var(741) == 1) then begin
-        giq_option(-3, 1376, 1910, Node999, 50);
+    if (global_var(GVAR_QUEST_JOHNATAN_CAVE) == 1) then begin
+        giq_option(-3, NAME, 1910, Node999, NEUTRAL_REACTION);
     end
 end
 
 procedure Node1950
 begin
-    gsay_reply(1376, 1950);
-    if (obj_carrying_pid_obj(Johnatan_box, 560) and (global_var(741) == 2)) then begin
-        giq_option(4, 1376, 1951, Node2050, 50);
+    Reply(1950);
+    if (obj_carrying_pid_obj(Johnatan_box, 560) and (global_var(GVAR_QUEST_JOHNATAN_CAVE) == 2)) then begin
+        giq_option(4, NAME, 1951, Node2050, NEUTRAL_REACTION);
     end
-    if (not(obj_carrying_pid_obj(Johnatan_box, 560)) and (global_var(741) == 2)) then begin
-        giq_option(4, 1376, 1952, Node2100, 50);
+    if (not(obj_carrying_pid_obj(Johnatan_box, 560)) and (global_var(GVAR_QUEST_JOHNATAN_CAVE) == 2)) then begin
+        giq_option(4, NAME, 1952, Node2100, NEUTRAL_REACTION);
     end
-    if (global_var(741) == 1) then begin
-        giq_option(4, 1376, 1953, Node2150, 50);
+    if (global_var(GVAR_QUEST_JOHNATAN_CAVE) == 1) then begin
+        giq_option(4, NAME, 1953, Node2150, NEUTRAL_REACTION);
     end
 end
 
 procedure Node2000
 begin
-    gsay_reply(1376, 2000);
-    if (obj_carrying_pid_obj(Johnatan_box, 560) and (global_var(741) == 2)) then begin
-        giq_option(4, 1376, 2001, Node2050, 50);
+    Reply(2000);
+    if (obj_carrying_pid_obj(Johnatan_box, 560) and (global_var(GVAR_QUEST_JOHNATAN_CAVE) == 2)) then begin
+        giq_option(4, NAME, 2001, Node2050, NEUTRAL_REACTION);
     end
-    if (not(obj_carrying_pid_obj(Johnatan_box, 560)) and (global_var(741) == 2)) then begin
-        giq_option(4, 1376, 2002, Node2100, 50);
+    if (not(obj_carrying_pid_obj(Johnatan_box, 560)) and (global_var(GVAR_QUEST_JOHNATAN_CAVE) == 2)) then begin
+        giq_option(4, NAME, 2002, Node2100, NEUTRAL_REACTION);
     end
-    if (global_var(741) == 1) then begin
-        giq_option(4, 1376, 2003, Node2150, 50);
+    if (global_var(GVAR_QUEST_JOHNATAN_CAVE) == 1) then begin
+        giq_option(4, NAME, 2003, Node2150, NEUTRAL_REACTION);
     end
 end
 
 procedure Node2050
 begin
-    gsay_reply(1376, message_str(1376, 2050) + 2000 + message_str(1376, 2051));
+    Reply(mstr(2050) + 2000 + mstr(2051));
     set_local_var(8, 8);
     item_caps_adjust(dude_obj, 2000);
     removed_qty := obj_is_carrying_obj_pid(Johnatan_box, 560);
@@ -916,17 +916,17 @@ begin
         restock_obj := obj_carrying_pid_obj(Johnatan_box, 560);
         if (obj_type(self_obj) == 1) then begin
             restock_obj := obj_carrying_pid_obj(Johnatan_box, 560);
-            if (critter_inven_obj(self_obj, 0) == restock_obj) then begin
-                if (obj_item_subtype(critter_inven_obj(Johnatan_box, 0)) == 0) then begin
-                    restock_obj := critter_inven_obj(Johnatan_box, 0);
+            if (critter_inven_obj(self_obj, INVEN_TYPE_WORN) == restock_obj) then begin
+                if (obj_item_subtype(critter_inven_obj(Johnatan_box, INVEN_TYPE_WORN)) == 0) then begin
+                    restock_obj := critter_inven_obj(Johnatan_box, INVEN_TYPE_WORN);
                     debug_msg("armour pid == " + obj_pid(restock_obj));
                     rm_obj_from_inven(Johnatan_box, restock_obj);
                     add_obj_to_inven(Johnatan_box, restock_obj);
                 end
             end
             else begin
-                if ((critter_inven_obj(Johnatan_box, 2) == restock_obj) or (critter_inven_obj(Johnatan_box, 1) == restock_obj)) then begin
-                    metarule(43, Johnatan_box);
+                if ((critter_inven_obj(Johnatan_box, INVEN_TYPE_LEFT_HAND) == restock_obj) or (critter_inven_obj(Johnatan_box, INVEN_TYPE_RIGHT_HAND) == restock_obj)) then begin
+                    inven_unwield(Johnatan_box);
                 end
             end
         end
@@ -946,53 +946,53 @@ begin
     end
     restock_obj := 0;
     restock_amt := 0;
-    restock_obj := create_object_sid(560, 0, 0, -1);
+    restock_obj := create_object(560, 0, 0);
     add_mult_objs_to_inven(dude_obj, restock_obj, 1);
-    set_global_var(741, 3);
-    set_global_var(0, global_var(0) + 20);
+    set_global_var(GVAR_QUEST_JOHNATAN_CAVE, 3);
+    set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + 20);
     debug_msg("Player gains " + 20 + " Karma Points.");
-    set_global_var(37, 0);
-    set_global_var(38, 0);
-    set_global_var(39, 0);
-    set_global_var(40, 0);
-    set_global_var(41, 0);
-    set_global_var(42, 0);
-    set_global_var(43, 0);
-    set_global_var(44, 0);
-    set_global_var(45, 0);
-    if (global_var(0) >= 1000) then begin
-        set_global_var(37, 1);
+    set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+    set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+    set_global_var(GVAR_KARMA_DEFENDER, 0);
+    set_global_var(GVAR_KARMA_WANDERER, 0);
+    set_global_var(GVAR_KARMA_BETRAYER, 0);
+    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+    set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+    set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+    if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
     end
     else begin
-        if (global_var(0) >= 750) then begin
-            set_global_var(38, 1);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+            set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
         end
         else begin
-            if (global_var(0) >= 500) then begin
-                set_global_var(39, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
             end
             else begin
-                if (global_var(0) >= 250) then begin
-                    set_global_var(40, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                    set_global_var(GVAR_KARMA_DEFENDER, 1);
                 end
                 else begin
-                    if (global_var(0) > -250) then begin
-                        set_global_var(41, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                        set_global_var(GVAR_KARMA_WANDERER, 1);
                     end
                     else begin
-                        if (global_var(0) > -500) then begin
-                            set_global_var(42, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                            set_global_var(GVAR_KARMA_BETRAYER, 1);
                         end
                         else begin
-                            if (global_var(0) > -750) then begin
-                                set_global_var(43, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                             end
                             else begin
-                                if (global_var(0) > -1000) then begin
-                                    set_global_var(44, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                    set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                 end
                                 else begin
-                                    set_global_var(45, 1);
+                                    set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                 end
                             end
                         end
@@ -1002,61 +1002,61 @@ begin
         end
     end
     give_exp_points(2000);
-    display_msg(message_str(14, 100) + 2000 + message_str(14, 101));
-    giq_option(4, 1376, 2052, Node2550, 50);
-    giq_option(4, 1376, 2053, Node999, 50);
+    display_msg(g_mstr(100) + 2000 + g_mstr(101));
+    giq_option(4, NAME, 2052, Node2550, NEUTRAL_REACTION);
+    giq_option(4, NAME, 2053, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node2100
 begin
-    gsay_reply(1376, message_str(1376, 2100) + 2000 + message_str(1376, 2101));
+    Reply(mstr(2100) + 2000 + mstr(2101));
     set_local_var(8, 8);
     item_caps_adjust(dude_obj, 2000);
-    set_global_var(741, 3);
-    set_global_var(0, global_var(0) + 20);
+    set_global_var(GVAR_QUEST_JOHNATAN_CAVE, 3);
+    set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + 20);
     debug_msg("Player gains " + 20 + " Karma Points.");
-    set_global_var(37, 0);
-    set_global_var(38, 0);
-    set_global_var(39, 0);
-    set_global_var(40, 0);
-    set_global_var(41, 0);
-    set_global_var(42, 0);
-    set_global_var(43, 0);
-    set_global_var(44, 0);
-    set_global_var(45, 0);
-    if (global_var(0) >= 1000) then begin
-        set_global_var(37, 1);
+    set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+    set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+    set_global_var(GVAR_KARMA_DEFENDER, 0);
+    set_global_var(GVAR_KARMA_WANDERER, 0);
+    set_global_var(GVAR_KARMA_BETRAYER, 0);
+    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+    set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+    set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+    if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
     end
     else begin
-        if (global_var(0) >= 750) then begin
-            set_global_var(38, 1);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+            set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
         end
         else begin
-            if (global_var(0) >= 500) then begin
-                set_global_var(39, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
             end
             else begin
-                if (global_var(0) >= 250) then begin
-                    set_global_var(40, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                    set_global_var(GVAR_KARMA_DEFENDER, 1);
                 end
                 else begin
-                    if (global_var(0) > -250) then begin
-                        set_global_var(41, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                        set_global_var(GVAR_KARMA_WANDERER, 1);
                     end
                     else begin
-                        if (global_var(0) > -500) then begin
-                            set_global_var(42, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                            set_global_var(GVAR_KARMA_BETRAYER, 1);
                         end
                         else begin
-                            if (global_var(0) > -750) then begin
-                                set_global_var(43, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                             end
                             else begin
-                                if (global_var(0) > -1000) then begin
-                                    set_global_var(44, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                    set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                 end
                                 else begin
-                                    set_global_var(45, 1);
+                                    set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                 end
                             end
                         end
@@ -1066,42 +1066,42 @@ begin
         end
     end
     give_exp_points(2000);
-    display_msg(message_str(14, 100) + 2000 + message_str(14, 101));
-    giq_option(4, 1376, 2102, Node2550, 50);
-    giq_option(4, 1376, 2103, Node999, 50);
+    display_msg(g_mstr(100) + 2000 + g_mstr(101));
+    giq_option(4, NAME, 2102, Node2550, NEUTRAL_REACTION);
+    giq_option(4, NAME, 2103, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node2150
 begin
-    gsay_reply(1376, 2150);
+    Reply(2150);
     set_local_var(8, 1);
-    giq_option(4, 1376, 2151, Node999, 50);
+    giq_option(4, NAME, 2151, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node2200
 begin
-    gsay_reply(1376, 2200);
-    if (obj_carrying_pid_obj(Johnatan_box, 560) and (global_var(741) == 2)) then begin
-        giq_option(4, 1376, 2201, Node2050, 50);
+    Reply(2200);
+    if (obj_carrying_pid_obj(Johnatan_box, 560) and (global_var(GVAR_QUEST_JOHNATAN_CAVE) == 2)) then begin
+        giq_option(4, NAME, 2201, Node2050, NEUTRAL_REACTION);
     end
-    if (not(obj_carrying_pid_obj(Johnatan_box, 560)) and (global_var(741) == 2)) then begin
-        giq_option(4, 1376, 2202, Node2100, 50);
+    if (not(obj_carrying_pid_obj(Johnatan_box, 560)) and (global_var(GVAR_QUEST_JOHNATAN_CAVE) == 2)) then begin
+        giq_option(4, NAME, 2202, Node2100, NEUTRAL_REACTION);
     end
-    if (global_var(741) == 1) then begin
-        giq_option(4, 1376, 2203, Node999, 50);
+    if (global_var(GVAR_QUEST_JOHNATAN_CAVE) == 1) then begin
+        giq_option(4, NAME, 2203, Node999, NEUTRAL_REACTION);
     end
-    if (get_critter_stat(dude_obj, 34) == 0) then begin
-        giq_option(4, 1376, 2204, Node1700, 50);
+    if (dude_gender == 0) then begin
+        giq_option(4, NAME, 2204, Node1700, NEUTRAL_REACTION);
     end
-    if (get_critter_stat(dude_obj, 34) == 1) then begin
-        giq_option(4, 1376, 2205, Node1700, 50);
+    if (dude_gender == 1) then begin
+        giq_option(4, NAME, 2205, Node1700, NEUTRAL_REACTION);
     end
-    giq_option(4, 1376, 2206, Node1750, 50);
+    giq_option(4, NAME, 2206, Node1750, NEUTRAL_REACTION);
 end
 
 procedure Node2250
 begin
-    gsay_reply(1376, message_str(1376, 2250) + 400 + message_str(1376, 2251));
+    Reply(mstr(2250) + 400 + mstr(2251));
     removed_qty := obj_is_carrying_obj_pid(dude_obj, 98);
     if (1 < removed_qty) then begin
         removed_qty := 1;
@@ -1110,17 +1110,17 @@ begin
         restock_obj := obj_carrying_pid_obj(dude_obj, 98);
         if (obj_type(self_obj) == 1) then begin
             restock_obj := obj_carrying_pid_obj(dude_obj, 98);
-            if (critter_inven_obj(self_obj, 0) == restock_obj) then begin
-                if (obj_item_subtype(critter_inven_obj(dude_obj, 0)) == 0) then begin
-                    restock_obj := critter_inven_obj(dude_obj, 0);
+            if (critter_inven_obj(self_obj, INVEN_TYPE_WORN) == restock_obj) then begin
+                if (obj_item_subtype(critter_inven_obj(dude_obj, INVEN_TYPE_WORN)) == 0) then begin
+                    restock_obj := critter_inven_obj(dude_obj, INVEN_TYPE_WORN);
                     debug_msg("armour pid == " + obj_pid(restock_obj));
                     rm_obj_from_inven(dude_obj, restock_obj);
                     add_obj_to_inven(dude_obj, restock_obj);
                 end
             end
             else begin
-                if ((critter_inven_obj(dude_obj, 2) == restock_obj) or (critter_inven_obj(dude_obj, 1) == restock_obj)) then begin
-                    metarule(43, dude_obj);
+                if ((critter_inven_obj(dude_obj, INVEN_TYPE_LEFT_HAND) == restock_obj) or (critter_inven_obj(dude_obj, INVEN_TYPE_RIGHT_HAND) == restock_obj)) then begin
+                    inven_unwield(dude_obj);
                 end
             end
         end
@@ -1140,98 +1140,98 @@ begin
     end
     restock_obj := 0;
     restock_amt := 0;
-    restock_obj := create_object_sid(98, 0, 0, -1);
+    restock_obj := create_object(98, 0, 0);
     add_mult_objs_to_inven(self_obj, restock_obj, 1);
-    set_global_var(740, 2);
+    set_global_var(GVAR_QUEST_S_CHARLIE_JUNK, 2);
     item_caps_adjust(dude_obj, 400);
     give_exp_points(500);
-    display_msg(message_str(14, 100) + 500 + message_str(14, 101));
+    display_msg(g_mstr(100) + 500 + g_mstr(101));
     set_local_var(5, 1);
-    giq_option(-3, 1376, 2252, Node2300, 50);
+    giq_option(-3, NAME, 2252, Node2300, NEUTRAL_REACTION);
 end
 
 procedure Node2300
 begin
-    gsay_reply(1376, 2301);
+    Reply(2301);
     set_local_var(7, 1);
-    giq_option(-3, 1376, 2302, Node2350, 50);
-    giq_option(-3, 1376, 2303, Node999, 50);
+    giq_option(-3, NAME, 2302, Node2350, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 2303, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node2350
 begin
-    gsay_reply(1376, 2350);
-    giq_option(-3, 1376, 2351, Node2400, 50);
+    Reply(2350);
+    giq_option(-3, NAME, 2351, Node2400, NEUTRAL_REACTION);
 end
 
 procedure Node2400
 begin
-    gsay_reply(1376, 2400);
-    set_global_var(741, 1);
-    if (metarule(17, 5) == 0) then begin
+    Reply(2400);
+    set_global_var(GVAR_QUEST_JOHNATAN_CAVE, 1);
+    if (town_known(5) == MARK_STATE_UNKNOWN) then begin
         debug_msg("  mark_on_map(" + 5 + ")");
-        mark_area_known(0, 5, 1);
+        mark_area_known(MARK_TYPE_TOWN, 5, MARK_STATE_KNOWN);
     end
-    giq_option(-3, 1376, 2401, Node999, 50);
+    giq_option(-3, NAME, 2401, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node2450
 begin
-    gsay_reply(1376, 2451);
+    Reply(2451);
     set_local_var(7, 1);
-    giq_option(-3, 1376, 2452, Node2350, 50);
-    giq_option(-3, 1376, 2453, Node999, 50);
+    giq_option(-3, NAME, 2452, Node2350, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 2453, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node2500
 begin
-    gsay_reply(1376, message_str(1376, 2500) + 2000 + message_str(1376, 2501));
-    set_global_var(741, 3);
+    Reply(mstr(2500) + 2000 + mstr(2501));
+    set_global_var(GVAR_QUEST_JOHNATAN_CAVE, 3);
     item_caps_adjust(dude_obj, 2000);
-    set_global_var(0, global_var(0) + 20);
+    set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + 20);
     debug_msg("Player gains " + 20 + " Karma Points.");
-    set_global_var(37, 0);
-    set_global_var(38, 0);
-    set_global_var(39, 0);
-    set_global_var(40, 0);
-    set_global_var(41, 0);
-    set_global_var(42, 0);
-    set_global_var(43, 0);
-    set_global_var(44, 0);
-    set_global_var(45, 0);
-    if (global_var(0) >= 1000) then begin
-        set_global_var(37, 1);
+    set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+    set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+    set_global_var(GVAR_KARMA_DEFENDER, 0);
+    set_global_var(GVAR_KARMA_WANDERER, 0);
+    set_global_var(GVAR_KARMA_BETRAYER, 0);
+    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+    set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+    set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+    if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
     end
     else begin
-        if (global_var(0) >= 750) then begin
-            set_global_var(38, 1);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+            set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
         end
         else begin
-            if (global_var(0) >= 500) then begin
-                set_global_var(39, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
             end
             else begin
-                if (global_var(0) >= 250) then begin
-                    set_global_var(40, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                    set_global_var(GVAR_KARMA_DEFENDER, 1);
                 end
                 else begin
-                    if (global_var(0) > -250) then begin
-                        set_global_var(41, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                        set_global_var(GVAR_KARMA_WANDERER, 1);
                     end
                     else begin
-                        if (global_var(0) > -500) then begin
-                            set_global_var(42, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                            set_global_var(GVAR_KARMA_BETRAYER, 1);
                         end
                         else begin
-                            if (global_var(0) > -750) then begin
-                                set_global_var(43, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                             end
                             else begin
-                                if (global_var(0) > -1000) then begin
-                                    set_global_var(44, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                    set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                 end
                                 else begin
-                                    set_global_var(45, 1);
+                                    set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                 end
                             end
                         end
@@ -1241,13 +1241,12 @@ begin
         end
     end
     give_exp_points(2000);
-    display_msg(message_str(14, 100) + 2000 + message_str(14, 101));
-    giq_option(-3, 1376, 2502, Node999, 50);
+    display_msg(g_mstr(100) + 2000 + g_mstr(101));
+    giq_option(-3, NAME, 2502, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node2550
 begin
-    gsay_reply(1376, 2550);
-    giq_option(4, 1376, 2551, Node999, 50);
+    Reply(2550);
+    giq_option(4, NAME, 2551, Node999, NEUTRAL_REACTION);
 end
-
