diff --git a/Mods/Resurrection/Original/KARMEMBR.ssl b/Mods/Resurrection/ReDefine/KARMEMBR.ssl
index 50c58f0..515fbb6 100644
--- a/Mods/Resurrection/Original/KARMEMBR.ssl
+++ b/Mods/Resurrection/ReDefine/KARMEMBR.ssl
@@ -47,12 +47,12 @@ begin
             return 1;
         end
     end
-    if ((party_member_obj(16777779) != 0) and ((global_var(910) bwand 32) == 0)) then begin
+    if ((party_member_obj(16777779) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 32) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777779)) <= 5) then begin
             return 1;
         end
     end
-    if ((party_member_obj(16777811) != 0) and ((global_var(910) bwand 64) == 0)) then begin
+    if ((party_member_obj(16777811) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 64) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777811)) <= 5) then begin
             return 1;
         end
@@ -67,7 +67,7 @@ begin
             return 1;
         end
     end
-    if ((party_member_obj(16777814) != 0) and ((global_var(910) bwand 128) == 0)) then begin
+    if ((party_member_obj(16777814) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 128) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777814)) <= 5) then begin
             return 1;
         end
@@ -127,171 +127,171 @@ begin
     if (local_var(6) == 0) then begin
         set_local_var(6, tile_num(self_obj));
     end
-    critter_add_trait(self_obj, 1, 6, 92);
-    critter_add_trait(self_obj, 1, 5, 181);
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 92);
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_CARAVAN_GUARD);
     if ((game_time - local_var(7)) > (60 * (60 * 10) * 24)) then begin
         set_local_var(5, 0);
         set_local_var(7, 0);
     end
-    if ((global_var(911) bwand 1) != 0) then begin
+    if ((global_var(GVAR_KARAVAN_ROUTES_COMMAND_BITS) bwand 1) != 0) then begin
         set_local_var(5, 0);
-        if ((global_var(911) bwand 16384) != 0) then begin
-            if ((cur_map_index == 21) and (local_var(6) == 25502)) then begin
+        if ((global_var(GVAR_KARAVAN_ROUTES_COMMAND_BITS) bwand 16384) != 0) then begin
+            if ((CUR_MAP_ALB_PRED) and (local_var(6) == 25502)) then begin
                 set_local_var(5, 1);
             end
-            if ((cur_map_index == 21) and (local_var(6) == 25699)) then begin
+            if ((CUR_MAP_ALB_PRED) and (local_var(6) == 25699)) then begin
                 set_local_var(5, 1);
             end
-            if ((cur_map_index == 15) and (local_var(6) == 25752)) then begin
+            if ((CUR_MAP_CORATH1) and (local_var(6) == 25752)) then begin
                 set_local_var(5, 1);
             end
-            if ((cur_map_index == 15) and (local_var(6) == 25752)) then begin
+            if ((CUR_MAP_CORATH1) and (local_var(6) == 25752)) then begin
                 set_local_var(5, 1);
             end
-            if ((cur_map_index == 6) and (local_var(6) == 25717)) then begin
+            if ((CUR_MAP_NH_VILL) and (local_var(6) == 25717)) then begin
                 set_local_var(5, 1);
             end
         end
-        if ((global_var(911) bwand 32768) != 0) then begin
-            if ((cur_map_index == 21) and (local_var(6) == 23704)) then begin
+        if ((global_var(GVAR_KARAVAN_ROUTES_COMMAND_BITS) bwand 32768) != 0) then begin
+            if ((CUR_MAP_ALB_PRED) and (local_var(6) == 23704)) then begin
                 set_local_var(5, 1);
             end
-            if ((cur_map_index == 21) and (local_var(6) == 22703)) then begin
+            if ((CUR_MAP_ALB_PRED) and (local_var(6) == 22703)) then begin
                 set_local_var(5, 1);
             end
-            if ((cur_map_index == 15) and (local_var(6) == 26745)) then begin
+            if ((CUR_MAP_CORATH1) and (local_var(6) == 26745)) then begin
                 set_local_var(5, 1);
             end
-            if ((cur_map_index == 15) and (local_var(6) == 26543)) then begin
+            if ((CUR_MAP_CORATH1) and (local_var(6) == 26543)) then begin
                 set_local_var(5, 1);
             end
-            if ((cur_map_index == 6) and (local_var(6) == 23520)) then begin
+            if ((CUR_MAP_NH_VILL) and (local_var(6) == 23520)) then begin
                 set_local_var(5, 1);
             end
         end
-        if ((global_var(911) bwand 65536) != 0) then begin
-            if ((cur_map_index == 21) and (local_var(6) == 22901)) then begin
+        if ((global_var(GVAR_KARAVAN_ROUTES_COMMAND_BITS) bwand 65536) != 0) then begin
+            if ((CUR_MAP_ALB_PRED) and (local_var(6) == 22901)) then begin
                 set_local_var(5, 1);
             end
-            if ((cur_map_index == 21) and (local_var(6) == 22899)) then begin
+            if ((CUR_MAP_ALB_PRED) and (local_var(6) == 22899)) then begin
                 set_local_var(5, 1);
             end
-            if ((cur_map_index == 15) and (local_var(6) == 24354)) then begin
+            if ((CUR_MAP_CORATH1) and (local_var(6) == 24354)) then begin
                 set_local_var(5, 1);
             end
-            if ((cur_map_index == 15) and (local_var(6) == 24148)) then begin
+            if ((CUR_MAP_CORATH1) and (local_var(6) == 24148)) then begin
                 set_local_var(5, 1);
             end
-            if ((cur_map_index == 6) and (local_var(6) == 25113)) then begin
+            if ((CUR_MAP_NH_VILL) and (local_var(6) == 25113)) then begin
                 set_local_var(5, 1);
             end
         end
-        if ((global_var(911) bwand 8192) != 0) then begin
-            if ((global_var(911) bwand 4) != 0) then begin
-                if (not((global_var(911) bwand 65536) != 0)) then begin
-                    if ((cur_map_index == 21) and (local_var(6) == 22901)) then begin
+        if ((global_var(GVAR_KARAVAN_ROUTES_COMMAND_BITS) bwand 8192) != 0) then begin
+            if ((global_var(GVAR_KARAVAN_ROUTES_COMMAND_BITS) bwand 4) != 0) then begin
+                if (not((global_var(GVAR_KARAVAN_ROUTES_COMMAND_BITS) bwand 65536) != 0)) then begin
+                    if ((CUR_MAP_ALB_PRED) and (local_var(6) == 22901)) then begin
                         set_local_var(5, 1);
                     end
-                    if ((cur_map_index == 21) and (local_var(6) == 22899)) then begin
+                    if ((CUR_MAP_ALB_PRED) and (local_var(6) == 22899)) then begin
                         set_local_var(5, 1);
                     end
-                    if ((cur_map_index == 15) and (local_var(6) == 24354)) then begin
+                    if ((CUR_MAP_CORATH1) and (local_var(6) == 24354)) then begin
                         set_local_var(5, 1);
                     end
-                    if ((cur_map_index == 15) and (local_var(6) == 24148)) then begin
+                    if ((CUR_MAP_CORATH1) and (local_var(6) == 24148)) then begin
                         set_local_var(5, 1);
                     end
-                    if ((cur_map_index == 6) and (local_var(6) == 25113)) then begin
+                    if ((CUR_MAP_NH_VILL) and (local_var(6) == 25113)) then begin
                         set_local_var(5, 1);
                     end
                 end
                 else begin
-                    if (not((global_var(911) bwand 32768) != 0)) then begin
-                        if ((cur_map_index == 21) and (local_var(6) == 23704)) then begin
+                    if (not((global_var(GVAR_KARAVAN_ROUTES_COMMAND_BITS) bwand 32768) != 0)) then begin
+                        if ((CUR_MAP_ALB_PRED) and (local_var(6) == 23704)) then begin
                             set_local_var(5, 1);
                         end
-                        if ((cur_map_index == 21) and (local_var(6) == 22703)) then begin
+                        if ((CUR_MAP_ALB_PRED) and (local_var(6) == 22703)) then begin
                             set_local_var(5, 1);
                         end
-                        if ((cur_map_index == 15) and (local_var(6) == 26745)) then begin
+                        if ((CUR_MAP_CORATH1) and (local_var(6) == 26745)) then begin
                             set_local_var(5, 1);
                         end
-                        if ((cur_map_index == 15) and (local_var(6) == 26543)) then begin
+                        if ((CUR_MAP_CORATH1) and (local_var(6) == 26543)) then begin
                             set_local_var(5, 1);
                         end
-                        if ((cur_map_index == 6) and (local_var(6) == 23520)) then begin
+                        if ((CUR_MAP_NH_VILL) and (local_var(6) == 23520)) then begin
                             set_local_var(5, 1);
                         end
                     end
                     else begin
-                        if ((cur_map_index == 21) and (local_var(6) == 25502)) then begin
+                        if ((CUR_MAP_ALB_PRED) and (local_var(6) == 25502)) then begin
                             set_local_var(5, 1);
                         end
-                        if ((cur_map_index == 21) and (local_var(6) == 25699)) then begin
+                        if ((CUR_MAP_ALB_PRED) and (local_var(6) == 25699)) then begin
                             set_local_var(5, 1);
                         end
-                        if ((cur_map_index == 15) and (local_var(6) == 25752)) then begin
+                        if ((CUR_MAP_CORATH1) and (local_var(6) == 25752)) then begin
                             set_local_var(5, 1);
                         end
-                        if ((cur_map_index == 15) and (local_var(6) == 25752)) then begin
+                        if ((CUR_MAP_CORATH1) and (local_var(6) == 25752)) then begin
                             set_local_var(5, 1);
                         end
-                        if ((cur_map_index == 6) and (local_var(6) == 25717)) then begin
+                        if ((CUR_MAP_NH_VILL) and (local_var(6) == 25717)) then begin
                             set_local_var(5, 1);
                         end
                     end
                 end
             end
             else begin
-                if (not((global_var(911) bwand 16384) != 0)) then begin
-                    if ((cur_map_index == 21) and (local_var(6) == 25502)) then begin
+                if (not((global_var(GVAR_KARAVAN_ROUTES_COMMAND_BITS) bwand 16384) != 0)) then begin
+                    if ((CUR_MAP_ALB_PRED) and (local_var(6) == 25502)) then begin
                         set_local_var(5, 1);
                     end
-                    if ((cur_map_index == 21) and (local_var(6) == 25699)) then begin
+                    if ((CUR_MAP_ALB_PRED) and (local_var(6) == 25699)) then begin
                         set_local_var(5, 1);
                     end
-                    if ((cur_map_index == 15) and (local_var(6) == 25752)) then begin
+                    if ((CUR_MAP_CORATH1) and (local_var(6) == 25752)) then begin
                         set_local_var(5, 1);
                     end
-                    if ((cur_map_index == 15) and (local_var(6) == 25752)) then begin
+                    if ((CUR_MAP_CORATH1) and (local_var(6) == 25752)) then begin
                         set_local_var(5, 1);
                     end
-                    if ((cur_map_index == 6) and (local_var(6) == 25717)) then begin
+                    if ((CUR_MAP_NH_VILL) and (local_var(6) == 25717)) then begin
                         set_local_var(5, 1);
                     end
                 end
                 else begin
-                    if (not((global_var(911) bwand 32768) != 0)) then begin
-                        if ((cur_map_index == 21) and (local_var(6) == 23704)) then begin
+                    if (not((global_var(GVAR_KARAVAN_ROUTES_COMMAND_BITS) bwand 32768) != 0)) then begin
+                        if ((CUR_MAP_ALB_PRED) and (local_var(6) == 23704)) then begin
                             set_local_var(5, 1);
                         end
-                        if ((cur_map_index == 21) and (local_var(6) == 22703)) then begin
+                        if ((CUR_MAP_ALB_PRED) and (local_var(6) == 22703)) then begin
                             set_local_var(5, 1);
                         end
-                        if ((cur_map_index == 15) and (local_var(6) == 26745)) then begin
+                        if ((CUR_MAP_CORATH1) and (local_var(6) == 26745)) then begin
                             set_local_var(5, 1);
                         end
-                        if ((cur_map_index == 15) and (local_var(6) == 26543)) then begin
+                        if ((CUR_MAP_CORATH1) and (local_var(6) == 26543)) then begin
                             set_local_var(5, 1);
                         end
-                        if ((cur_map_index == 6) and (local_var(6) == 23520)) then begin
+                        if ((CUR_MAP_NH_VILL) and (local_var(6) == 23520)) then begin
                             set_local_var(5, 1);
                         end
                     end
                     else begin
-                        if ((cur_map_index == 21) and (local_var(6) == 22901)) then begin
+                        if ((CUR_MAP_ALB_PRED) and (local_var(6) == 22901)) then begin
                             set_local_var(5, 1);
                         end
-                        if ((cur_map_index == 21) and (local_var(6) == 22899)) then begin
+                        if ((CUR_MAP_ALB_PRED) and (local_var(6) == 22899)) then begin
                             set_local_var(5, 1);
                         end
-                        if ((cur_map_index == 15) and (local_var(6) == 24354)) then begin
+                        if ((CUR_MAP_CORATH1) and (local_var(6) == 24354)) then begin
                             set_local_var(5, 1);
                         end
-                        if ((cur_map_index == 15) and (local_var(6) == 24148)) then begin
+                        if ((CUR_MAP_CORATH1) and (local_var(6) == 24148)) then begin
                             set_local_var(5, 1);
                         end
-                        if ((cur_map_index == 6) and (local_var(6) == 25113)) then begin
+                        if ((CUR_MAP_NH_VILL) and (local_var(6) == 25113)) then begin
                             set_local_var(5, 1);
                         end
                     end
@@ -304,15 +304,15 @@ end
 
 procedure critter_p_proc
 begin
-    if ((has_trait(1, self_obj, 666) == 1) and (local_var(4) == 1) and obj_can_see_obj(self_obj, dude_obj)) then begin
+    if ((obj_is_visible_flag(self_obj) == 1) and (local_var(4) == 1) and obj_can_see_obj(self_obj, dude_obj)) then begin
         if (critter_is_fleeing(self_obj)) then begin
             if (anim_busy(self_obj) == 0) then begin
                 if ((tile_distance_objs(self_obj, dude_obj) < 8) or obj_can_see_obj(self_obj, dude_obj)) then begin
                     if (anim_busy(self_obj) == 0) then begin
-                        global_temp := rotation_to_tile(tile_num(dude_obj), tile_num(self_obj));
+                        global_temp := rotation_to_tile(dude_tile, tile_num(self_obj));
                         animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(self_obj), global_temp, random(3, 10)), 1);
                         global_temp := (global_temp + 1) % 6;
-                        while ((anim_busy(self_obj) == 0) and (global_temp != rotation_to_tile(tile_num(dude_obj), tile_num(self_obj)))) do begin
+                        while ((anim_busy(self_obj) == 0) and (global_temp != rotation_to_tile(dude_tile, tile_num(self_obj)))) do begin
                             debug_msg("flee loop: rot == " + global_temp);
                             animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(self_obj), global_temp, random(3, 10)), 1);
                             global_temp := (global_temp + 1) % 6;
@@ -322,16 +322,16 @@ begin
             end
         end
         else begin
-            attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+            attack(dude_obj);
         end
     end
-    if ((has_trait(1, self_obj, 666) == 0) and (local_var(4) == 1)) then begin
+    if ((obj_is_visible_flag(self_obj) == 0) and (local_var(4) == 1)) then begin
         set_local_var(4, 0);
     end
-    if (((global_var(911) bwand 134217728) != 0) or ((global_var(911) bwand 268435456) != 0) or ((global_var(911) bwand 536870912) != 0)) then begin
-        if (cur_map_index == 21) then begin
+    if (((global_var(GVAR_KARAVAN_ROUTES_COMMAND_BITS) bwand 134217728) != 0) or ((global_var(GVAR_KARAVAN_ROUTES_COMMAND_BITS) bwand 268435456) != 0) or ((global_var(GVAR_KARAVAN_ROUTES_COMMAND_BITS) bwand 536870912) != 0)) then begin
+        if (CUR_MAP_ALB_PRED) then begin
             if ((local_var(6) == 25699) or (local_var(6) == 23704) or (local_var(6) == 22901)) then begin
-                if (((global_var(911) bwand 134217728) != 0) and (local_var(5) == 0) and (global_var(913) == 0)) then begin
+                if (((global_var(GVAR_KARAVAN_ROUTES_COMMAND_BITS) bwand 134217728) != 0) and (local_var(5) == 0) and (global_var(GVAR_KARAVAN_SEDIT_ELIMINATED) == 0)) then begin
                     set_obj_visibility(self_obj, 0);
                 end
                 else begin
@@ -339,7 +339,7 @@ begin
                 end
             end
             if ((local_var(6) == 25502) or (local_var(6) == 22703) or (local_var(6) == 22899)) then begin
-                if (((global_var(911) bwand 268435456) != 0) and (local_var(5) == 0) and (global_var(914) == 0) or (((global_var(911) bwand 536870912) != 0) and (local_var(5) == 0) and (global_var(912) == 0))) then begin
+                if (((global_var(GVAR_KARAVAN_ROUTES_COMMAND_BITS) bwand 268435456) != 0) and (local_var(5) == 0) and (global_var(GVAR_KARAVAN_CRIMSON_ELIMINATED) == 0) or (((global_var(GVAR_KARAVAN_ROUTES_COMMAND_BITS) bwand 536870912) != 0) and (local_var(5) == 0) and (global_var(GVAR_KARAVAN_FREDERYK_ELIMINATED) == 0))) then begin
                     set_obj_visibility(self_obj, 0);
                 end
                 else begin
@@ -347,9 +347,9 @@ begin
                 end
             end
         end
-        if (cur_map_index == 15) then begin
+        if (CUR_MAP_CORATH1) then begin
             if ((local_var(6) == 26352) or (local_var(6) == 26745) or (local_var(6) == 24354)) then begin
-                if (((global_var(911) bwand 134217728) != 0) and (local_var(5) == 0) and (global_var(913) == 0)) then begin
+                if (((global_var(GVAR_KARAVAN_ROUTES_COMMAND_BITS) bwand 134217728) != 0) and (local_var(5) == 0) and (global_var(GVAR_KARAVAN_SEDIT_ELIMINATED) == 0)) then begin
                     set_obj_visibility(self_obj, 0);
                 end
                 else begin
@@ -357,7 +357,7 @@ begin
                 end
             end
             if ((local_var(6) == 25752) or (local_var(6) == 26543) or (local_var(6) == 24148)) then begin
-                if (((global_var(911) bwand 268435456) != 0) and (local_var(5) == 0) and (global_var(914) == 0) or (((global_var(911) bwand 536870912) != 0) and (local_var(5) == 0) and (global_var(912) == 0))) then begin
+                if (((global_var(GVAR_KARAVAN_ROUTES_COMMAND_BITS) bwand 268435456) != 0) and (local_var(5) == 0) and (global_var(GVAR_KARAVAN_CRIMSON_ELIMINATED) == 0) or (((global_var(GVAR_KARAVAN_ROUTES_COMMAND_BITS) bwand 536870912) != 0) and (local_var(5) == 0) and (global_var(GVAR_KARAVAN_FREDERYK_ELIMINATED) == 0))) then begin
                     set_obj_visibility(self_obj, 0);
                 end
                 else begin
@@ -365,8 +365,8 @@ begin
                 end
             end
         end
-        if (cur_map_index == 6) then begin
-            if (((global_var(911) bwand 268435456) != 0) and (local_var(5) == 0) and (global_var(914) == 0)) then begin
+        if (CUR_MAP_NH_VILL) then begin
+            if (((global_var(GVAR_KARAVAN_ROUTES_COMMAND_BITS) bwand 268435456) != 0) and (local_var(5) == 0) and (global_var(GVAR_KARAVAN_CRIMSON_ELIMINATED) == 0)) then begin
                 set_obj_visibility(self_obj, 0);
             end
             else begin
@@ -375,7 +375,7 @@ begin
         end
     end
     else begin
-        if (has_trait(1, self_obj, 666) == 1) then begin
+        if (obj_is_visible_flag(self_obj) == 1) then begin
             set_obj_visibility(self_obj, 1);
         end
     end
@@ -397,7 +397,7 @@ end
 
 procedure talk_p_proc
 begin
-    float_msg(self_obj, message_str(27, random(340, 345)), 8);
+    float_msg(self_obj, mstr(random(340, 345)), FLOAT_MSG_YELLOW);
 end
 
 procedure destroy_p_proc
@@ -407,25 +407,25 @@ end
 procedure look_at_p_proc
 begin
     script_overrides;
-    display_msg(message_str(27, 100));
+    display_mstr(100);
 end
 
 procedure description_p_proc
 begin
     script_overrides;
-    if (has_trait(1, self_obj, 666) == 1) then begin
-        if (cur_map_index == 6) then begin
-            display_msg(message_str(27, 200));
+    if (obj_is_visible_flag(self_obj) == 1) then begin
+        if (CUR_MAP_NH_VILL) then begin
+            display_mstr(200);
         end
         else begin
-            if ((global_var(911) bwand 134217728) != 0) then begin
-                display_msg(message_str(27, 201));
+            if ((global_var(GVAR_KARAVAN_ROUTES_COMMAND_BITS) bwand 134217728) != 0) then begin
+                display_mstr(201);
             end
-            if ((global_var(911) bwand 268435456) != 0) then begin
-                display_msg(message_str(27, 200));
+            if ((global_var(GVAR_KARAVAN_ROUTES_COMMAND_BITS) bwand 268435456) != 0) then begin
+                display_mstr(200);
             end
-            if ((global_var(911) bwand 536870912) != 0) then begin
-                display_msg(message_str(27, 202));
+            if ((global_var(GVAR_KARAVAN_ROUTES_COMMAND_BITS) bwand 536870912) != 0) then begin
+                display_mstr(202);
             end
         end
     end
@@ -441,25 +441,24 @@ end
 
 procedure combat_p_proc
 begin
-    if ((fixed_param == 4) and (local_var(4) == 1) and (has_trait(1, self_obj, 666) == 1)) then begin
+    if ((fixed_param == 4) and (local_var(4) == 1) and (obj_is_visible_flag(self_obj) == 1)) then begin
         probability := random(0, 12);
         if (not(critter_is_fleeing(self_obj))) then begin
             if (probability == 1) then begin
-                float_msg(self_obj, message_str(27, 400), 2);
+                float_msg(self_obj, mstr(400), FLOAT_MSG_RED);
             end
-            if ((probability == 2) and (get_critter_stat(dude_obj, 34) == 0)) then begin
-                float_msg(self_obj, message_str(27, 401), 2);
+            if ((probability == 2) and (dude_gender == 0)) then begin
+                float_msg(self_obj, mstr(401), FLOAT_MSG_RED);
             end
-            if ((probability == 3) and (get_critter_stat(dude_obj, 34) == 1)) then begin
-                float_msg(self_obj, message_str(27, 402), 2);
+            if ((probability == 3) and (dude_gender == 1)) then begin
+                float_msg(self_obj, mstr(402), FLOAT_MSG_RED);
             end
-            if ((probability == 4) and (get_critter_stat(self_obj, 34) == 0)) then begin
-                float_msg(self_obj, message_str(27, 403), 2);
+            if ((probability == 4) and (self_gender == 0)) then begin
+                float_msg(self_obj, mstr(403), FLOAT_MSG_RED);
             end
-            if ((probability == 5) and (get_critter_stat(self_obj, 34) == 1)) then begin
-                float_msg(self_obj, message_str(27, 404), 2);
+            if ((probability == 5) and (self_gender == 1)) then begin
+                float_msg(self_obj, mstr(404), FLOAT_MSG_RED);
             end
         end
     end
 end
-
