diff --git a/Mods/Resurrection/Original/MCOMP.ssl b/Mods/Resurrection/ReDefine/MCOMP.ssl
index 6c47c72..61b7db4 100644
--- a/Mods/Resurrection/Original/MCOMP.ssl
+++ b/Mods/Resurrection/ReDefine/MCOMP.ssl
@@ -57,12 +57,12 @@ begin
             return 1;
         end
     end
-    if ((party_member_obj(16777779) != 0) and ((global_var(910) bwand 32) == 0)) then begin
+    if ((party_member_obj(16777779) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 32) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777779)) <= 5) then begin
             return 1;
         end
     end
-    if ((party_member_obj(16777811) != 0) and ((global_var(910) bwand 64) == 0)) then begin
+    if ((party_member_obj(16777811) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 64) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777811)) <= 5) then begin
             return 1;
         end
@@ -77,7 +77,7 @@ begin
             return 1;
         end
     end
-    if ((party_member_obj(16777814) != 0) and ((global_var(910) bwand 128) == 0)) then begin
+    if ((party_member_obj(16777814) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 128) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777814)) <= 5) then begin
             return 1;
         end
@@ -142,11 +142,11 @@ end
 
 procedure use_p_proc
 begin
-    anim(self_obj, 1000, rotation_to_tile(tile_num(self_obj), tile_num(dude_obj)));
+    anim(self_obj, 1000, rotation_to_tile(tile_num(self_obj), dude_tile));
     start_gdialog(1769, self_obj, 4, -1, -1);
     gsay_start;
     if (local_var(4) == 0) then begin
-        if ((get_critter_stat(dude_obj, 4) + has_trait(0, dude_obj, 49)) > 3) then begin
+        if ((dude_iq + has_trait(TRAIT_PERK, dude_obj, PERK_smooth_talker)) > 3) then begin
             call Node500();
         end
         else begin
@@ -154,7 +154,7 @@ begin
         end
     end
     else begin
-        if ((get_critter_stat(dude_obj, 4) + has_trait(0, dude_obj, 49)) > 3) then begin
+        if ((dude_iq + has_trait(TRAIT_PERK, dude_obj, PERK_smooth_talker)) > 3) then begin
             call Node550();
         end
         else begin
@@ -172,13 +172,13 @@ end
 procedure look_at_p_proc
 begin
     script_overrides;
-    display_msg(message_str(1769, 100));
+    display_mstr(100);
 end
 
 procedure description_p_proc
 begin
     script_overrides;
-    display_msg(message_str(1769, 200));
+    display_mstr(200);
 end
 
 procedure use_skill_on_p_proc
@@ -194,34 +194,33 @@ end
 
 procedure Node500
 begin
-    gsay_reply(1769, 500);
-    giq_option(4, 1769, 501, Get_Disc, 50);
-    giq_option(4, 1769, 502, Node999, 50);
+    Reply(500);
+    giq_option(4, NAME, 501, Get_Disc, NEUTRAL_REACTION);
+    giq_option(4, NAME, 502, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node550
 begin
-    gsay_reply(1769, 550);
-    giq_option(4, 1769, 551, Node999, 50);
+    Reply(550);
+    giq_option(4, NAME, 551, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node600
 begin
-    gsay_reply(1769, 600);
-    giq_option(-3, 1769, 601, Get_Disc, 50);
-    giq_option(-3, 1769, 602, Node999, 50);
+    Reply(600);
+    giq_option(-3, NAME, 601, Get_Disc, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 602, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node650
 begin
-    gsay_reply(1769, 650);
-    giq_option(-3, 1769, 651, Node999, 50);
+    Reply(650);
+    giq_option(-3, NAME, 651, Node999, NEUTRAL_REACTION);
 end
 
 procedure Get_Disc
 begin
     set_local_var(4, 1);
-    holodisk := create_object_sid(576, 0, 0, 1775);
+    holodisk := create_object_sid(576, 0, 0, SCRIPT_MHLTHURD);
     add_obj_to_inven(dude_obj, holodisk);
 end
-
