diff --git a/Mods/Resurrection/Original/MMARRY.ssl b/Mods/Resurrection/ReDefine/MMARRY.ssl
index 58fb5cd..609179d 100644
--- a/Mods/Resurrection/Original/MMARRY.ssl
+++ b/Mods/Resurrection/ReDefine/MMARRY.ssl
@@ -81,12 +81,12 @@ begin
             return 1;
         end
     end
-    if ((party_member_obj(16777779) != 0) and ((global_var(910) bwand 32) == 0)) then begin
+    if ((party_member_obj(16777779) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 32) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777779)) <= 5) then begin
             return 1;
         end
     end
-    if ((party_member_obj(16777811) != 0) and ((global_var(910) bwand 64) == 0)) then begin
+    if ((party_member_obj(16777811) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 64) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777811)) <= 5) then begin
             return 1;
         end
@@ -101,7 +101,7 @@ begin
             return 1;
         end
     end
-    if ((party_member_obj(16777814) != 0) and ((global_var(910) bwand 128) == 0)) then begin
+    if ((party_member_obj(16777814) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 128) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777814)) <= 5) then begin
             return 1;
         end
@@ -159,9 +159,9 @@ end
 
 procedure critter_p_proc
 begin
-    if (has_trait(1, self_obj, 6) == 0) then begin
+    if (has_trait(TRAIT_OBJECT, self_obj, OBJECT_TEAM_NUM) == 0) then begin
         if ((local_var(4) == 1) or (map_var(2) >= 3) or (map_var(4) >= 7)) then begin
-            critter_add_trait(self_obj, 1, 6, 268);
+            critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 268);
         end
     end
     if ((local_var(4) == 1) or (map_var(2) >= 3) and obj_can_see_obj(self_obj, dude_obj)) then begin
@@ -169,10 +169,10 @@ begin
             if (anim_busy(self_obj) == 0) then begin
                 if ((tile_distance_objs(self_obj, dude_obj) < 8) or obj_can_see_obj(self_obj, dude_obj)) then begin
                     if (anim_busy(self_obj) == 0) then begin
-                        global_temp := rotation_to_tile(tile_num(dude_obj), tile_num(self_obj));
+                        global_temp := rotation_to_tile(dude_tile, tile_num(self_obj));
                         animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(self_obj), global_temp, random(3, 10)), 1);
                         global_temp := (global_temp + 1) % 6;
-                        while ((anim_busy(self_obj) == 0) and (global_temp != rotation_to_tile(tile_num(dude_obj), tile_num(self_obj)))) do begin
+                        while ((anim_busy(self_obj) == 0) and (global_temp != rotation_to_tile(dude_tile, tile_num(self_obj)))) do begin
                             debug_msg("flee loop: rot == " + global_temp);
                             animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(self_obj), global_temp, random(3, 10)), 1);
                             global_temp := (global_temp + 1) % 6;
@@ -182,13 +182,13 @@ begin
             end
         end
         else begin
-            attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+            attack(dude_obj);
         end
     end
-    if (((local_var(6) bwand 1) != 0) and (map_var(2) < 3) and (tile_num(self_obj) == 15737) and (has_trait(1, self_obj, 10) != 3)) then begin
+    if (((local_var(6) bwand 1) != 0) and (map_var(2) < 3) and (tile_num(self_obj) == 15737) and (obj_get_rot(self_obj) != 3)) then begin
         anim(self_obj, 1000, 3);
     end
-    if (((local_var(6) bwand 1) != 0) and (map_var(2) < 3) and (tile_num(self_obj) != 15737) and (global_var(907) != 2)) then begin
+    if (((local_var(6) bwand 1) != 0) and (map_var(2) < 3) and (tile_num(self_obj) != 15737) and (global_var(GVAR_MC_QUEST_BRING_SEBASTIAN) != 2)) then begin
         animate_move_obj_to_tile(self_obj, 15737, 0);
     end
 end
@@ -202,12 +202,12 @@ end
 
 procedure talk_p_proc
 begin
-    anim(dude_obj, 1000, rotation_to_tile(tile_num(dude_obj), tile_num(self_obj)));
-    anim(self_obj, 1000, rotation_to_tile(tile_num(self_obj), tile_num(dude_obj)));
+    anim(dude_obj, 1000, rotation_to_tile(dude_tile, tile_num(self_obj)));
+    anim(self_obj, 1000, rotation_to_tile(tile_num(self_obj), dude_tile));
     start_gdialog(1767, self_obj, 4, -1, -1);
     gsay_start;
     if ((local_var(6) bwand 1) != 0) then begin
-        if ((get_critter_stat(dude_obj, 4) + has_trait(0, dude_obj, 49)) > 3) then begin
+        if ((dude_iq + has_trait(TRAIT_PERK, dude_obj, PERK_smooth_talker)) > 3) then begin
             call Node1700();
         end
         else begin
@@ -215,11 +215,11 @@ begin
         end
     end
     else begin
-        if ((obj_pid(critter_inven_obj(dude_obj, 0)) == 113) and ((get_critter_stat(dude_obj, 4) + has_trait(0, dude_obj, 49)) > 3)) then begin
+        if ((obj_pid(critter_inven_obj(dude_obj, INVEN_TYPE_WORN)) == 113) and ((dude_iq + has_trait(TRAIT_PERK, dude_obj, PERK_smooth_talker)) > 3)) then begin
             call Node1200();
         end
         else begin
-            if ((get_critter_stat(dude_obj, 4) + has_trait(0, dude_obj, 49)) > 3) then begin
+            if ((dude_iq + has_trait(TRAIT_PERK, dude_obj, PERK_smooth_talker)) > 3) then begin
                 call Node500();
             end
             else begin
@@ -234,58 +234,58 @@ end
 procedure destroy_p_proc
 begin
     if (source_obj == dude_obj) then begin
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(4) > (3 * global_var(5))) or (global_var(2) == 1)) and (global_var(1) == 0)) then begin
-            set_global_var(2, 1);
-            set_global_var(3, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(4) > (3 * global_var(GVAR_GOOD_MONSTER))) or (global_var(GVAR_CHAMPION_REPUTATION) == 1)) and (global_var(GVAR_CHILDKILLER_REPUTATION) == 0)) then begin
+            set_global_var(GVAR_CHAMPION_REPUTATION, 1);
+            set_global_var(GVAR_BERSERKER_REPUTATION, 0);
         end
-        if (metarule(51, self_obj) == 2) then begin
+        if (critter_kill_type(self_obj) == KILL_TYPE_children_kills) then begin
             if (source_obj == dude_obj) then begin
-                set_global_var(2, 0);
-                set_global_var(1, global_var(1) + 1);
-                set_global_var(0, global_var(0) + -15);
+                set_global_var(GVAR_CHAMPION_REPUTATION, 0);
+                set_global_var(GVAR_CHILDKILLER_REPUTATION, global_var(GVAR_CHILDKILLER_REPUTATION) + 1);
+                set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + -15);
                 debug_msg("Player gains " + -15 + " Karma Points.");
-                set_global_var(37, 0);
-                set_global_var(38, 0);
-                set_global_var(39, 0);
-                set_global_var(40, 0);
-                set_global_var(41, 0);
-                set_global_var(42, 0);
-                set_global_var(43, 0);
-                set_global_var(44, 0);
-                set_global_var(45, 0);
-                if (global_var(0) >= 1000) then begin
-                    set_global_var(37, 1);
+                set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+                set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+                set_global_var(GVAR_KARMA_DEFENDER, 0);
+                set_global_var(GVAR_KARMA_WANDERER, 0);
+                set_global_var(GVAR_KARMA_BETRAYER, 0);
+                set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+                set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+                set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+                    set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
                 end
                 else begin
-                    if (global_var(0) >= 750) then begin
-                        set_global_var(38, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
                     end
                     else begin
-                        if (global_var(0) >= 500) then begin
-                            set_global_var(39, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                            set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                         end
                         else begin
-                            if (global_var(0) >= 250) then begin
-                                set_global_var(40, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                                set_global_var(GVAR_KARMA_DEFENDER, 1);
                             end
                             else begin
-                                if (global_var(0) > -250) then begin
-                                    set_global_var(41, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                                    set_global_var(GVAR_KARMA_WANDERER, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -500) then begin
-                                        set_global_var(42, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                        set_global_var(GVAR_KARMA_BETRAYER, 1);
                                     end
                                     else begin
-                                        if (global_var(0) > -750) then begin
-                                            set_global_var(43, 1);
+                                        if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                            set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                         end
                                         else begin
-                                            if (global_var(0) > -1000) then begin
-                                                set_global_var(44, 1);
+                                            if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                                set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                             end
                                             else begin
-                                                set_global_var(45, 1);
+                                                set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                             end
                                         end
                                     end
@@ -294,58 +294,58 @@ begin
                         end
                     end
                 end
-                set_global_var(52, global_var(52) + -8);
+                set_global_var(GVAR_TOWN_REP_MUTANT_CAMP, global_var(GVAR_TOWN_REP_MUTANT_CAMP) + -8);
                 debug_msg("Added " + -8 + " to Town Rep");
             end
         end
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(5) > (2 * global_var(4))) or (global_var(3) == 1))) then begin
-            set_global_var(3, 1);
-            set_global_var(2, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
+            set_global_var(GVAR_BERSERKER_REPUTATION, 1);
+            set_global_var(GVAR_CHAMPION_REPUTATION, 0);
         end
-        set_global_var(0, global_var(0) + 0);
+        set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + 0);
         debug_msg("Player gains " + 0 + " Karma Points.");
-        set_global_var(37, 0);
-        set_global_var(38, 0);
-        set_global_var(39, 0);
-        set_global_var(40, 0);
-        set_global_var(41, 0);
-        set_global_var(42, 0);
-        set_global_var(43, 0);
-        set_global_var(44, 0);
-        set_global_var(45, 0);
-        if (global_var(0) >= 1000) then begin
-            set_global_var(37, 1);
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+        set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+        set_global_var(GVAR_KARMA_DEFENDER, 0);
+        set_global_var(GVAR_KARMA_WANDERER, 0);
+        set_global_var(GVAR_KARMA_BETRAYER, 0);
+        set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+        set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+            set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
         end
         else begin
-            if (global_var(0) >= 750) then begin
-                set_global_var(38, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
             end
             else begin
-                if (global_var(0) >= 500) then begin
-                    set_global_var(39, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                 end
                 else begin
-                    if (global_var(0) >= 250) then begin
-                        set_global_var(40, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                        set_global_var(GVAR_KARMA_DEFENDER, 1);
                     end
                     else begin
-                        if (global_var(0) > -250) then begin
-                            set_global_var(41, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                            set_global_var(GVAR_KARMA_WANDERER, 1);
                         end
                         else begin
-                            if (global_var(0) > -500) then begin
-                                set_global_var(42, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                set_global_var(GVAR_KARMA_BETRAYER, 1);
                             end
                             else begin
-                                if (global_var(0) > -750) then begin
-                                    set_global_var(43, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -1000) then begin
-                                        set_global_var(44, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                     end
                                     else begin
-                                        set_global_var(45, 1);
+                                        set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                     end
                                 end
                             end
@@ -362,10 +362,10 @@ procedure look_at_p_proc
 begin
     script_overrides;
     if (local_var(5) == 1) then begin
-        display_msg(message_str(1767, 150));
+        display_mstr(150);
     end
     else begin
-        display_msg(message_str(1767, 100));
+        display_mstr(100);
     end
 end
 
@@ -373,10 +373,10 @@ procedure description_p_proc
 begin
     script_overrides;
     if (local_var(5) == 1) then begin
-        display_msg(message_str(1767, 250));
+        display_mstr(250);
     end
     else begin
-        display_msg(message_str(1767, 200));
+        display_mstr(200);
     end
 end
 
@@ -387,7 +387,7 @@ end
 procedure damage_p_proc
 begin
     if (source_obj == dude_obj) then begin
-        if ((map_var(2) < 3) and (global_var(1030) > 0) and (global_var(1030) != 3) and (map_var(4) < 7)) then begin
+        if ((map_var(2) < 3) and (global_var(GVAR_MC_SEBASTIAN_NOT_CONVINCED) > 0) and (global_var(GVAR_MC_SEBASTIAN_NOT_CONVINCED) != 3) and (map_var(4) < 7)) then begin
             set_map_var(2, map_var(2) + 1);
         end
         else begin
@@ -395,34 +395,34 @@ begin
         end
         if (map_var(2) >= 3) then begin
             set_local_var(4, 1);
-            if (not((global_var(906) bwand 32) != 0)) then begin
-                set_global_var(906, global_var(906) bwor 32);
+            if (not((global_var(GVAR_MC_MUTANT_CAMP_EVENTS) bwand 32) != 0)) then begin
+                set_global_var(GVAR_MC_MUTANT_CAMP_EVENTS, global_var(GVAR_MC_MUTANT_CAMP_EVENTS) bwor 32);
             end
-            if (global_var(907) != 4) then begin
-                set_global_var(907, 0);
+            if (global_var(GVAR_MC_QUEST_BRING_SEBASTIAN) != 4) then begin
+                set_global_var(GVAR_MC_QUEST_BRING_SEBASTIAN, 0);
             end
-            set_global_var(1015, global_var(1015) bwand (-1 - 262144));
-            set_global_var(1018, global_var(1018) bwand (-1 - 67108864));
-            set_global_var(1015, global_var(1015) bwand (-1 - 524288));
-            set_global_var(1020, global_var(1020) bwand (-1 - 2048));
-            set_global_var(1020, global_var(1020) bwand (-1 - 4096));
+            set_global_var(GVAR_PTR_KERI_REACTIONS, global_var(GVAR_PTR_KERI_REACTIONS) bwand (-1 - 262144));
+            set_global_var(GVAR_PTR_GABRIEL_REACTIONS, global_var(GVAR_PTR_GABRIEL_REACTIONS) bwand (-1 - 67108864));
+            set_global_var(GVAR_PTR_KERI_REACTIONS, global_var(GVAR_PTR_KERI_REACTIONS) bwand (-1 - 524288));
+            set_global_var(GVAR_PTR_LYSTRA_REACTIONS_2, global_var(GVAR_PTR_LYSTRA_REACTIONS_2) bwand (-1 - 2048));
+            set_global_var(GVAR_PTR_LYSTRA_REACTIONS_2, global_var(GVAR_PTR_LYSTRA_REACTIONS_2) bwand (-1 - 4096));
         end
     end
 end
 
 procedure map_enter_p_proc
 begin
-    if ((map_var(2) < 3) and (global_var(1030) > 0) and (global_var(1030) != 3) and (map_var(4) < 7)) then begin
-        critter_add_trait(self_obj, 1, 6, 0);
+    if ((map_var(2) < 3) and (global_var(GVAR_MC_SEBASTIAN_NOT_CONVINCED) > 0) and (global_var(GVAR_MC_SEBASTIAN_NOT_CONVINCED) != 3) and (map_var(4) < 7)) then begin
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 0);
     end
     else begin
-        critter_add_trait(self_obj, 1, 6, 268);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 268);
     end
-    critter_add_trait(self_obj, 1, 5, 47);
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_SUPER_MUTANT);
     if (map_var(2) < 3) then begin
         add_timer_event(self_obj, game_ticks(300), 1);
     end
-    if (((local_var(6) bwand 1) != 0) and (map_var(2) < 3) and not(metarule(22, 0)) and (global_var(907) != 2)) then begin
+    if (((local_var(6) bwand 1) != 0) and (map_var(2) < 3) and not(is_loading_game) and (global_var(GVAR_MC_QUEST_BRING_SEBASTIAN) != 2)) then begin
         critter_attempt_placement(self_obj, 15737, 0);
     end
 end
@@ -433,13 +433,13 @@ end
 
 procedure combat_p_proc
 begin
-    if ((has_trait(1, self_obj, 6) == 0) and (map_var(2) >= 3)) then begin
-        critter_add_trait(self_obj, 1, 6, 268);
+    if ((has_trait(TRAIT_OBJECT, self_obj, OBJECT_TEAM_NUM) == 0) and (map_var(2) >= 3)) then begin
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 268);
     end
     if (fixed_param == 4) then begin
         probability := random(0, 13);
         if ((probability >= 13) and not(critter_is_fleeing(self_obj))) then begin
-            float_msg(self_obj, message_str(1767, random(400, 402)), 2);
+            float_msg(self_obj, mstr(random(400, 402)), FLOAT_MSG_RED);
         end
     end
 end
@@ -448,7 +448,7 @@ procedure timed_event_p_proc
 begin
     if (fixed_param == 1) then begin
         if (not(combat_is_initialized)) then begin
-            float_msg(self_obj, message_str(1767, 300), 8);
+            float_msg(self_obj, mstr(300), FLOAT_MSG_YELLOW);
         end
         if (map_var(2) < 3) then begin
             add_timer_event(self_obj, game_ticks(300), 1);
@@ -460,9 +460,9 @@ procedure Node998
 begin
     set_local_var(4, 1);
     set_map_var(2, 3);
-    set_global_var(1015, global_var(1015) bwand (-1 - 262144));
-    set_global_var(1018, global_var(1018) bwand (-1 - 67108864));
-    set_global_var(1020, global_var(1020) bwand (-1 - 2048));
+    set_global_var(GVAR_PTR_KERI_REACTIONS, global_var(GVAR_PTR_KERI_REACTIONS) bwand (-1 - 262144));
+    set_global_var(GVAR_PTR_GABRIEL_REACTIONS, global_var(GVAR_PTR_GABRIEL_REACTIONS) bwand (-1 - 67108864));
+    set_global_var(GVAR_PTR_LYSTRA_REACTIONS_2, global_var(GVAR_PTR_LYSTRA_REACTIONS_2) bwand (-1 - 2048));
 end
 
 procedure Node999
@@ -472,329 +472,328 @@ end
 
 procedure Node500
 begin
-    gsay_reply(1767, 500);
+    Reply(500);
     if ((local_var(6) bwand 2) == 0) then begin
-        giq_option(4, 1767, 501, Node550, 50);
+        giq_option(4, NAME, 501, Node550, NEUTRAL_REACTION);
     end
-    giq_option(4, 1767, 502, Node650, 50);
-    giq_option(4, 1767, 503, Node750, 50);
-    giq_option(4, 1767, 504, Node850, 50);
-    giq_option(4, 1767, 505, Node999, 50);
+    giq_option(4, NAME, 502, Node650, NEUTRAL_REACTION);
+    giq_option(4, NAME, 503, Node750, NEUTRAL_REACTION);
+    giq_option(4, NAME, 504, Node850, NEUTRAL_REACTION);
+    giq_option(4, NAME, 505, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node550
 begin
-    gsay_reply(1767, 550);
+    Reply(550);
     if ((local_var(6) bwand 2) == 0) then begin
-        giq_option(4, 1767, 551, Node600, 50);
+        giq_option(4, NAME, 551, Node600, NEUTRAL_REACTION);
     end
-    giq_option(4, 1767, 552, Node999, 50);
+    giq_option(4, NAME, 552, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node600
 begin
-    gsay_reply(1767, 600);
+    Reply(600);
     set_local_var(5, 1);
     set_local_var(6, local_var(6) bwor 2);
-    giq_option(4, 1767, 601, Node650, 50);
-    giq_option(4, 1767, 602, Node750, 50);
-    giq_option(4, 1767, 603, Node850, 50);
-    giq_option(4, 1767, 604, Node999, 50);
+    giq_option(4, NAME, 601, Node650, NEUTRAL_REACTION);
+    giq_option(4, NAME, 602, Node750, NEUTRAL_REACTION);
+    giq_option(4, NAME, 603, Node850, NEUTRAL_REACTION);
+    giq_option(4, NAME, 604, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node650
 begin
-    gsay_reply(1767, 650);
-    giq_option(4, 1767, 651, Node700, 50);
-    giq_option(4, 1767, 652, Node999, 50);
+    Reply(650);
+    giq_option(4, NAME, 651, Node700, NEUTRAL_REACTION);
+    giq_option(4, NAME, 652, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node700
 begin
-    gsay_reply(1767, 700);
+    Reply(700);
     set_local_var(5, 1);
     if ((local_var(6) bwand 2) == 0) then begin
-        giq_option(4, 1767, 701, Node550, 50);
+        giq_option(4, NAME, 701, Node550, NEUTRAL_REACTION);
     end
-    giq_option(4, 1767, 702, Node750, 50);
-    giq_option(4, 1767, 703, Node850, 50);
-    giq_option(4, 1767, 704, Node999, 50);
+    giq_option(4, NAME, 702, Node750, NEUTRAL_REACTION);
+    giq_option(4, NAME, 703, Node850, NEUTRAL_REACTION);
+    giq_option(4, NAME, 704, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node750
 begin
-    gsay_reply(1767, 750);
-    giq_option(4, 1767, 751, Node800, 50);
-    giq_option(4, 1767, 752, Node999, 50);
+    Reply(750);
+    giq_option(4, NAME, 751, Node800, NEUTRAL_REACTION);
+    giq_option(4, NAME, 752, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node800
 begin
-    gsay_reply(1767, 800);
+    Reply(800);
     set_local_var(5, 1);
-    giq_option(4, 1767, 801, Node1000, 50);
-    giq_option(4, 1767, 802, Node999, 50);
+    giq_option(4, NAME, 801, Node1000, NEUTRAL_REACTION);
+    giq_option(4, NAME, 802, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node850
 begin
-    gsay_reply(1767, 850);
-    giq_option(4, 1767, 851, Node900, 50);
-    giq_option(4, 1767, 852, Node999, 50);
+    Reply(850);
+    giq_option(4, NAME, 851, Node900, NEUTRAL_REACTION);
+    giq_option(4, NAME, 852, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node900
 begin
-    gsay_reply(1767, 900);
-    giq_option(4, 1767, 901, Node950, 50);
-    giq_option(4, 1767, 902, Node999, 50);
+    Reply(900);
+    giq_option(4, NAME, 901, Node950, NEUTRAL_REACTION);
+    giq_option(4, NAME, 902, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node950
 begin
-    gsay_reply(1767, 950);
-    giq_option(4, 1767, 951, Node999, 50);
+    Reply(950);
+    giq_option(4, NAME, 951, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node1000
 begin
-    gsay_reply(1767, 1000);
-    giq_option(4, 1767, 1001, Node1050, 50);
-    giq_option(4, 1767, 1002, Node999, 50);
+    Reply(1000);
+    giq_option(4, NAME, 1001, Node1050, NEUTRAL_REACTION);
+    giq_option(4, NAME, 1002, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node1050
 begin
-    gsay_reply(1767, 1050);
+    Reply(1050);
     set_local_var(5, 1);
-    giq_option(4, 1767, 1051, Node1100, 50);
-    giq_option(4, 1767, 1052, Node1150, 50);
-    if (has_skill(dude_obj, 14) >= 80) then begin
-        giq_option(4, 1767, 1053, Node1800, 50);
+    giq_option(4, NAME, 1051, Node1100, NEUTRAL_REACTION);
+    giq_option(4, NAME, 1052, Node1150, NEUTRAL_REACTION);
+    if (dude_skill(SKILL_SPEECH) >= 80) then begin
+        giq_option(4, NAME, 1053, Node1800, NEUTRAL_REACTION);
     end
-    giq_option(4, 1767, 1054, Node999, 50);
+    giq_option(4, NAME, 1054, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node1100
 begin
-    gsay_reply(1767, 1100);
+    Reply(1100);
     set_local_var(5, 1);
-    giq_option(4, 1767, 1101, Node1150, 50);
-    if (has_skill(dude_obj, 14) >= 80) then begin
-        giq_option(4, 1767, 1102, Node1800, 50);
+    giq_option(4, NAME, 1101, Node1150, NEUTRAL_REACTION);
+    if (dude_skill(SKILL_SPEECH) >= 80) then begin
+        giq_option(4, NAME, 1102, Node1800, NEUTRAL_REACTION);
     end
-    giq_option(4, 1767, 1103, Node999, 50);
+    giq_option(4, NAME, 1103, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node1150
 begin
-    gsay_reply(1767, 1150);
+    Reply(1150);
     set_local_var(5, 1);
-    giq_option(4, 1767, 1151, Node1100, 50);
-    if (has_skill(dude_obj, 14) >= 80) then begin
-        giq_option(4, 1767, 1152, Node1800, 50);
+    giq_option(4, NAME, 1151, Node1100, NEUTRAL_REACTION);
+    if (dude_skill(SKILL_SPEECH) >= 80) then begin
+        giq_option(4, NAME, 1152, Node1800, NEUTRAL_REACTION);
     end
-    giq_option(4, 1767, 1153, Node999, 50);
+    giq_option(4, NAME, 1153, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node1200
 begin
-    gsay_reply(1767, 1200);
-    giq_option(4, 1767, 1205, Node1250, 50);
+    Reply(1200);
+    giq_option(4, NAME, 1205, Node1250, NEUTRAL_REACTION);
     if ((local_var(6) bwand 2) == 0) then begin
-        giq_option(4, 1767, 1201, Node550, 50);
+        giq_option(4, NAME, 1201, Node550, NEUTRAL_REACTION);
     end
-    giq_option(4, 1767, 1202, Node650, 50);
-    giq_option(4, 1767, 1203, Node750, 50);
-    giq_option(4, 1767, 1204, Node850, 50);
-    giq_option(4, 1767, 1206, Node999, 50);
+    giq_option(4, NAME, 1202, Node650, NEUTRAL_REACTION);
+    giq_option(4, NAME, 1203, Node750, NEUTRAL_REACTION);
+    giq_option(4, NAME, 1204, Node850, NEUTRAL_REACTION);
+    giq_option(4, NAME, 1206, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node1250
 begin
-    gsay_reply(1767, 1250);
-    giq_option(4, 1767, 1251, Node1300, 50);
-    giq_option(4, 1767, 1252, Node999, 50);
+    Reply(1250);
+    giq_option(4, NAME, 1251, Node1300, NEUTRAL_REACTION);
+    giq_option(4, NAME, 1252, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node1300
 begin
-    gsay_reply(1767, 1300);
+    Reply(1300);
     set_local_var(5, 1);
-    giq_option(4, 1767, 1301, Node1350, 50);
-    giq_option(4, 1767, 1302, Node999, 50);
+    giq_option(4, NAME, 1301, Node1350, NEUTRAL_REACTION);
+    giq_option(4, NAME, 1302, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node1350
 begin
-    gsay_reply(1767, 1350);
-    giq_option(4, 1767, 1351, unblock_path, 50);
+    Reply(1350);
+    giq_option(4, NAME, 1351, unblock_path, NEUTRAL_REACTION);
 end
 
 procedure Node1400
 begin
-    if (get_critter_stat(dude_obj, 6) != 10) then begin
-        gsay_reply(1767, 1400);
+    if (dude_luck != 10) then begin
+        Reply(1400);
     end
     else begin
-        gsay_reply(1767, 1401);
+        Reply(1401);
     end
-    if (get_critter_stat(dude_obj, 6) != 10) then begin
-        giq_option(-3, 1767, message_str(1767, 1402) + obj_name(dude_obj) + message_str(1767, 1403), Node1450, 50);
+    if (dude_luck != 10) then begin
+        giq_option(-3, NAME, mstr(1402) + dude_name + mstr(1403), Node1450, NEUTRAL_REACTION);
     end
     else begin
-        giq_option(-3, 1767, message_str(1767, 1402) + obj_name(dude_obj) + message_str(1767, 1403) + message_str(1767, 1404) + obj_name(dude_obj) + message_str(1767, 1405), Node1450, 50);
+        giq_option(-3, NAME, mstr(1402) + dude_name + mstr(1403) + mstr(1404) + dude_name + mstr(1405), Node1450, NEUTRAL_REACTION);
     end
-    if (get_critter_stat(dude_obj, 6) != 10) then begin
-        giq_option(-3, 1767, 1406, Node999, 50);
+    if (dude_luck != 10) then begin
+        giq_option(-3, NAME, 1406, Node999, NEUTRAL_REACTION);
     end
     else begin
-        giq_option(-3, 1767, 1407, Node999, 50);
+        giq_option(-3, NAME, 1407, Node999, NEUTRAL_REACTION);
     end
 end
 
 procedure Node1450
 begin
-    if (get_critter_stat(dude_obj, 6) != 10) then begin
-        gsay_reply(1767, 1450);
+    if (dude_luck != 10) then begin
+        Reply(1450);
     end
     else begin
-        gsay_reply(1767, 1451);
+        Reply(1451);
     end
     set_local_var(5, 1);
-    if (get_critter_stat(dude_obj, 6) != 10) then begin
-        giq_option(-3, 1767, 1452, Node1500, 50);
+    if (dude_luck != 10) then begin
+        giq_option(-3, NAME, 1452, Node1500, NEUTRAL_REACTION);
     end
     else begin
-        giq_option(-3, 1767, 1453, Node1500, 50);
+        giq_option(-3, NAME, 1453, Node1500, NEUTRAL_REACTION);
     end
-    if (get_critter_stat(dude_obj, 6) != 10) then begin
-        giq_option(-3, 1767, 1454, Node999, 50);
+    if (dude_luck != 10) then begin
+        giq_option(-3, NAME, 1454, Node999, NEUTRAL_REACTION);
     end
     else begin
-        giq_option(-3, 1767, 1455, Node999, 50);
+        giq_option(-3, NAME, 1455, Node999, NEUTRAL_REACTION);
     end
 end
 
 procedure Node1500
 begin
-    if (get_critter_stat(dude_obj, 6) != 10) then begin
-        gsay_reply(1767, 1500);
+    if (dude_luck != 10) then begin
+        Reply(1500);
     end
     else begin
-        gsay_reply(1767, 1501);
+        Reply(1501);
     end
     set_local_var(5, 1);
-    if (get_critter_stat(dude_obj, 6) != 10) then begin
-        giq_option(-3, 1767, 1502, Node1550, 50);
+    if (dude_luck != 10) then begin
+        giq_option(-3, NAME, 1502, Node1550, NEUTRAL_REACTION);
     end
     else begin
-        giq_option(-3, 1767, 1503, Node1550, 50);
+        giq_option(-3, NAME, 1503, Node1550, NEUTRAL_REACTION);
     end
-    if (get_critter_stat(dude_obj, 6) != 10) then begin
-        giq_option(-3, 1767, 1504, Node999, 50);
+    if (dude_luck != 10) then begin
+        giq_option(-3, NAME, 1504, Node999, NEUTRAL_REACTION);
     end
     else begin
-        giq_option(-3, 1767, 1505, Node999, 50);
+        giq_option(-3, NAME, 1505, Node999, NEUTRAL_REACTION);
     end
 end
 
 procedure Node1550
 begin
-    if (get_critter_stat(dude_obj, 6) != 10) then begin
-        gsay_reply(1767, 1550);
+    if (dude_luck != 10) then begin
+        Reply(1550);
     end
     else begin
-        gsay_reply(1767, 1551);
+        Reply(1551);
     end
     set_local_var(5, 1);
-    if (get_critter_stat(dude_obj, 6) != 10) then begin
-        giq_option(-3, 1767, 1552, Node1600, 50);
+    if (dude_luck != 10) then begin
+        giq_option(-3, NAME, 1552, Node1600, NEUTRAL_REACTION);
     end
     else begin
-        giq_option(-3, 1767, 1553, Node1600, 50);
+        giq_option(-3, NAME, 1553, Node1600, NEUTRAL_REACTION);
     end
-    if (get_critter_stat(dude_obj, 6) != 10) then begin
-        giq_option(-3, 1767, 1554, Node999, 50);
+    if (dude_luck != 10) then begin
+        giq_option(-3, NAME, 1554, Node999, NEUTRAL_REACTION);
     end
     else begin
-        giq_option(-3, 1767, 1555, Node999, 50);
+        giq_option(-3, NAME, 1555, Node999, NEUTRAL_REACTION);
     end
 end
 
 procedure Node1600
 begin
-    if (get_critter_stat(dude_obj, 6) != 10) then begin
-        gsay_reply(1767, 1600);
+    if (dude_luck != 10) then begin
+        Reply(1600);
     end
     else begin
-        gsay_reply(1767, 1601);
+        Reply(1601);
     end
-    if (get_critter_stat(dude_obj, 6) != 10) then begin
-        giq_option(-3, 1767, 1602, Node1650, 50);
+    if (dude_luck != 10) then begin
+        giq_option(-3, NAME, 1602, Node1650, NEUTRAL_REACTION);
     end
     else begin
-        giq_option(-3, 1767, 1603, Node1650, 50);
+        giq_option(-3, NAME, 1603, Node1650, NEUTRAL_REACTION);
     end
 end
 
 procedure Node1650
 begin
-    if (get_critter_stat(dude_obj, 6) != 10) then begin
-        gsay_reply(1767, 1650);
+    if (dude_luck != 10) then begin
+        Reply(1650);
     end
     else begin
-        gsay_reply(1767, 1651);
+        Reply(1651);
     end
-    if (get_critter_stat(dude_obj, 6) != 10) then begin
-        giq_option(-3, 1767, 1652, unblock_path, 50);
+    if (dude_luck != 10) then begin
+        giq_option(-3, NAME, 1652, unblock_path, NEUTRAL_REACTION);
     end
     else begin
-        giq_option(-3, 1767, 1653, unblock_path, 50);
+        giq_option(-3, NAME, 1653, unblock_path, NEUTRAL_REACTION);
     end
 end
 
 procedure Node1700
 begin
-    gsay_reply(1767, 1700);
-    giq_option(4, 1767, 1701, Node999, 50);
+    Reply(1700);
+    giq_option(4, NAME, 1701, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node1750
 begin
-    if (get_critter_stat(dude_obj, 6) != 10) then begin
-        gsay_reply(1767, 1750);
+    if (dude_luck != 10) then begin
+        Reply(1750);
     end
     else begin
-        gsay_reply(1767, 1751);
+        Reply(1751);
     end
-    if (get_critter_stat(dude_obj, 6) != 10) then begin
-        giq_option(-3, 1767, 1752, Node999, 50);
+    if (dude_luck != 10) then begin
+        giq_option(-3, NAME, 1752, Node999, NEUTRAL_REACTION);
     end
     else begin
-        giq_option(-3, 1767, 1753, Node999, 50);
+        giq_option(-3, NAME, 1753, Node999, NEUTRAL_REACTION);
     end
 end
 
 procedure Node1800
 begin
-    gsay_reply(1767, 1800);
-    giq_option(4, 1767, 1801, Node1850, 50);
+    Reply(1800);
+    giq_option(4, NAME, 1801, Node1850, NEUTRAL_REACTION);
 end
 
 procedure Node1850
 begin
-    gsay_reply(1767, 1850);
-    giq_option(4, 1767, 1851, unblock_path, 50);
+    Reply(1850);
+    giq_option(4, NAME, 1851, unblock_path, NEUTRAL_REACTION);
 end
 
 procedure unblock_path
 begin
     set_local_var(5, 1);
     set_local_var(6, local_var(6) bwor 1);
-    set_global_var(1015, global_var(1015) bwand (-1 - 262144));
-    set_global_var(1018, global_var(1018) bwand (-1 - 67108864));
-    set_global_var(1020, global_var(1020) bwand (-1 - 2048));
+    set_global_var(GVAR_PTR_KERI_REACTIONS, global_var(GVAR_PTR_KERI_REACTIONS) bwand (-1 - 262144));
+    set_global_var(GVAR_PTR_GABRIEL_REACTIONS, global_var(GVAR_PTR_GABRIEL_REACTIONS) bwand (-1 - 67108864));
+    set_global_var(GVAR_PTR_LYSTRA_REACTIONS_2, global_var(GVAR_PTR_LYSTRA_REACTIONS_2) bwand (-1 - 2048));
     call Node999();
 end
-
