diff --git a/Mods/Resurrection/Original/MUTANT.ssl b/Mods/Resurrection/ReDefine/MUTANT.ssl
index 1da8ea4..42fe5de 100644
--- a/Mods/Resurrection/Original/MUTANT.ssl
+++ b/Mods/Resurrection/ReDefine/MUTANT.ssl
@@ -60,12 +60,12 @@ begin
             return 1;
         end
     end
-    if ((party_member_obj(16777779) != 0) and ((global_var(910) bwand 32) == 0)) then begin
+    if ((party_member_obj(16777779) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 32) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777779)) <= 5) then begin
             return 1;
         end
     end
-    if ((party_member_obj(16777811) != 0) and ((global_var(910) bwand 64) == 0)) then begin
+    if ((party_member_obj(16777811) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 64) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777811)) <= 5) then begin
             return 1;
         end
@@ -80,7 +80,7 @@ begin
             return 1;
         end
     end
-    if ((party_member_obj(16777814) != 0) and ((global_var(910) bwand 128) == 0)) then begin
+    if ((party_member_obj(16777814) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 128) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777814)) <= 5) then begin
             return 1;
         end
@@ -137,7 +137,7 @@ end
 
 procedure map_enter_p_proc
 begin
-    if (elevation(dude_obj) == 1) then begin
+    if (dude_elevation == 1) then begin
         set_light_level(50);
     end
     else begin
@@ -218,32 +218,32 @@ begin
             end
         end
     end
-    if (not((global_var(906) bwand 1) != 0)) then begin
-        set_global_var(906, global_var(906) bwor 1);
+    if (not((global_var(GVAR_MC_MUTANT_CAMP_EVENTS) bwand 1) != 0)) then begin
+        set_global_var(GVAR_MC_MUTANT_CAMP_EVENTS, global_var(GVAR_MC_MUTANT_CAMP_EVENTS) bwor 1);
     end
-    if (not(metarule(22, 0))) then begin
-        if (metarule(14, 0)) then begin
-            display_msg(message_str(1771, 500));
+    if (not(is_loading_game)) then begin
+        if (map_first_run) then begin
+            display_mstr(500);
         end
-        if ((global_var(907) == 2) and (map_var(2) < 3)) then begin
-            gfade_out(1);
+        if ((global_var(GVAR_MC_QUEST_BRING_SEBASTIAN) == 2) and (map_var(2) < 3)) then begin
+            fade_out;
             game_time_advance(3 * (24 * (60 * (60 * 10))));
             add_timer_event(self_obj, 10, 3);
-            if (global_var(1030) == 1) then begin
+            if (global_var(GVAR_MC_SEBASTIAN_NOT_CONVINCED) == 1) then begin
                 add_timer_event(self_obj, 20, 2);
             end
             else begin
                 add_timer_event(self_obj, 20, 1);
             end
-            set_global_var(1015, global_var(1015) bwand (-1 - 524288));
-            set_global_var(1020, global_var(1020) bwand (-1 - 4096));
+            set_global_var(GVAR_PTR_KERI_REACTIONS, global_var(GVAR_PTR_KERI_REACTIONS) bwand (-1 - 524288));
+            set_global_var(GVAR_PTR_LYSTRA_REACTIONS_2, global_var(GVAR_PTR_LYSTRA_REACTIONS_2) bwand (-1 - 4096));
         end
     end
 end
 
 procedure map_update_p_proc
 begin
-    if (elevation(dude_obj) == 1) then begin
+    if (dude_elevation == 1) then begin
         set_light_level(50);
     end
     else begin
@@ -328,15 +328,15 @@ end
 
 procedure map_exit_p_proc
 begin
-    if ((global_var(1030) > 0) and (global_var(1030) < 3) and (map_var(4) < 7)) then begin
+    if ((global_var(GVAR_MC_SEBASTIAN_NOT_CONVINCED) > 0) and (global_var(GVAR_MC_SEBASTIAN_NOT_CONVINCED) < 3) and (map_var(4) < 7)) then begin
         set_map_var(2, 3);
         kill_critter(mutant1, 61);
         kill_critter(mutant2, 61);
         kill_critter(mutant3, 61);
         kill_critter(Marry, 61);
     end
-    if (not((global_var(906) bwand 64) != 0) and (map_var(3) >= (13 - 1)) and ((global_var(906) bwand 2) != 0)) then begin
-        set_global_var(906, global_var(906) bwor 64);
+    if (not((global_var(GVAR_MC_MUTANT_CAMP_EVENTS) bwand 64) != 0) and (map_var(3) >= (13 - 1)) and ((global_var(GVAR_MC_MUTANT_CAMP_EVENTS) bwand 2) != 0)) then begin
+        set_global_var(GVAR_MC_MUTANT_CAMP_EVENTS, global_var(GVAR_MC_MUTANT_CAMP_EVENTS) bwor 64);
     end
 end
 
@@ -345,103 +345,103 @@ begin
     if (fixed_param == 1) then begin
         sequence := sequence + 1;
         if (sequence == 1) then begin
-            gfade_in(1);
+            fade_in;
             move_to(dude_obj, 14923, 0);
             anim(dude_obj, 1000, 1);
             if (party_member_obj(16777729) != 0) then begin
                 if (party_member_obj(16777729) != 0) then begin
-                    if (has_trait(1, party_member_obj(16777729), 666)) then begin
-                        critter_attempt_placement(party_member_obj(16777729), tile_num_in_direction(tile_num(dude_obj), (has_trait(1, dude_obj, 10) + 3) % 6, 4), elevation(dude_obj));
+                    if (obj_is_visible_flag(party_member_obj(16777729))) then begin
+                        critter_attempt_placement(party_member_obj(16777729), tile_num_in_direction(dude_tile, (obj_get_rot(dude_obj) + 3) % 6, 4), dude_elevation);
                     end
                 end
             end
-            if ((party_member_obj(16777779) != 0) and ((global_var(910) bwand 32) == 0)) then begin
+            if ((party_member_obj(16777779) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 32) == 0)) then begin
                 if (party_member_obj(16777779) != 0) then begin
-                    if (has_trait(1, party_member_obj(16777779), 666)) then begin
-                        critter_attempt_placement(party_member_obj(16777779), tile_num_in_direction(tile_num(dude_obj), (has_trait(1, dude_obj, 10) + 3) % 6, 4), elevation(dude_obj));
+                    if (obj_is_visible_flag(party_member_obj(16777779))) then begin
+                        critter_attempt_placement(party_member_obj(16777779), tile_num_in_direction(dude_tile, (obj_get_rot(dude_obj) + 3) % 6, 4), dude_elevation);
                     end
                 end
             end
-            if ((party_member_obj(16777811) != 0) and ((global_var(910) bwand 64) == 0)) then begin
+            if ((party_member_obj(16777811) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 64) == 0)) then begin
                 if (party_member_obj(16777811) != 0) then begin
-                    if (has_trait(1, party_member_obj(16777811), 666)) then begin
-                        critter_attempt_placement(party_member_obj(16777811), tile_num_in_direction(tile_num(dude_obj), (has_trait(1, dude_obj, 10) + 3) % 6, 4), elevation(dude_obj));
+                    if (obj_is_visible_flag(party_member_obj(16777811))) then begin
+                        critter_attempt_placement(party_member_obj(16777811), tile_num_in_direction(dude_tile, (obj_get_rot(dude_obj) + 3) % 6, 4), dude_elevation);
                     end
                 end
             end
             if (party_member_obj(16777812) != 0) then begin
                 if (party_member_obj(16777812) != 0) then begin
-                    if (has_trait(1, party_member_obj(16777812), 666)) then begin
-                        critter_attempt_placement(party_member_obj(16777812), tile_num_in_direction(tile_num(dude_obj), (has_trait(1, dude_obj, 10) + 3) % 6, 4), elevation(dude_obj));
+                    if (obj_is_visible_flag(party_member_obj(16777812))) then begin
+                        critter_attempt_placement(party_member_obj(16777812), tile_num_in_direction(dude_tile, (obj_get_rot(dude_obj) + 3) % 6, 4), dude_elevation);
                     end
                 end
             end
-            if ((party_member_obj(16777814) != 0) and ((global_var(910) bwand 128) == 0)) then begin
+            if ((party_member_obj(16777814) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 128) == 0)) then begin
                 if (party_member_obj(16777814) != 0) then begin
-                    if (has_trait(1, party_member_obj(16777814), 666)) then begin
-                        critter_attempt_placement(party_member_obj(16777814), tile_num_in_direction(tile_num(dude_obj), (has_trait(1, dude_obj, 10) + 3) % 6, 4), elevation(dude_obj));
+                    if (obj_is_visible_flag(party_member_obj(16777814))) then begin
+                        critter_attempt_placement(party_member_obj(16777814), tile_num_in_direction(dude_tile, (obj_get_rot(dude_obj) + 3) % 6, 4), dude_elevation);
                     end
                 end
             end
             if (party_member_obj(16777748) != 0) then begin
                 if (party_member_obj(16777748) != 0) then begin
-                    if (has_trait(1, party_member_obj(16777748), 666)) then begin
-                        critter_attempt_placement(party_member_obj(16777748), tile_num_in_direction(tile_num(dude_obj), (has_trait(1, dude_obj, 10) + 3) % 6, 4), elevation(dude_obj));
+                    if (obj_is_visible_flag(party_member_obj(16777748))) then begin
+                        critter_attempt_placement(party_member_obj(16777748), tile_num_in_direction(dude_tile, (obj_get_rot(dude_obj) + 3) % 6, 4), dude_elevation);
                     end
                 end
             end
             if (party_member_obj(16778017) != 0) then begin
                 if (party_member_obj(16778017) != 0) then begin
-                    if (has_trait(1, party_member_obj(16778017), 666)) then begin
-                        critter_attempt_placement(party_member_obj(16778017), tile_num_in_direction(tile_num(dude_obj), (has_trait(1, dude_obj, 10) + 3) % 6, 4), elevation(dude_obj));
+                    if (obj_is_visible_flag(party_member_obj(16778017))) then begin
+                        critter_attempt_placement(party_member_obj(16778017), tile_num_in_direction(dude_tile, (obj_get_rot(dude_obj) + 3) % 6, 4), dude_elevation);
                     end
                 end
             end
             if (party_member_obj(16778016) != 0) then begin
                 if (party_member_obj(16778016) != 0) then begin
-                    if (has_trait(1, party_member_obj(16778016), 666)) then begin
-                        critter_attempt_placement(party_member_obj(16778016), tile_num_in_direction(tile_num(dude_obj), (has_trait(1, dude_obj, 10) + 3) % 6, 4), elevation(dude_obj));
+                    if (obj_is_visible_flag(party_member_obj(16778016))) then begin
+                        critter_attempt_placement(party_member_obj(16778016), tile_num_in_direction(dude_tile, (obj_get_rot(dude_obj) + 3) % 6, 4), dude_elevation);
                     end
                 end
             end
             if (party_member_obj(16778049) != 0) then begin
                 if (party_member_obj(16778049) != 0) then begin
-                    if (has_trait(1, party_member_obj(16778049), 666)) then begin
-                        critter_attempt_placement(party_member_obj(16778049), tile_num_in_direction(tile_num(dude_obj), (has_trait(1, dude_obj, 10) + 3) % 6, 4), elevation(dude_obj));
+                    if (obj_is_visible_flag(party_member_obj(16778049))) then begin
+                        critter_attempt_placement(party_member_obj(16778049), tile_num_in_direction(dude_tile, (obj_get_rot(dude_obj) + 3) % 6, 4), dude_elevation);
                     end
                 end
             end
             if (party_member_obj(16778050) != 0) then begin
                 if (party_member_obj(16778050) != 0) then begin
-                    if (has_trait(1, party_member_obj(16778050), 666)) then begin
-                        critter_attempt_placement(party_member_obj(16778050), tile_num_in_direction(tile_num(dude_obj), (has_trait(1, dude_obj, 10) + 3) % 6, 4), elevation(dude_obj));
+                    if (obj_is_visible_flag(party_member_obj(16778050))) then begin
+                        critter_attempt_placement(party_member_obj(16778050), tile_num_in_direction(dude_tile, (obj_get_rot(dude_obj) + 3) % 6, 4), dude_elevation);
                     end
                 end
             end
             if (party_member_obj(16778051) != 0) then begin
                 if (party_member_obj(16778051) != 0) then begin
-                    if (has_trait(1, party_member_obj(16778051), 666)) then begin
-                        critter_attempt_placement(party_member_obj(16778051), tile_num_in_direction(tile_num(dude_obj), (has_trait(1, dude_obj, 10) + 3) % 6, 4), elevation(dude_obj));
+                    if (obj_is_visible_flag(party_member_obj(16778051))) then begin
+                        critter_attempt_placement(party_member_obj(16778051), tile_num_in_direction(dude_tile, (obj_get_rot(dude_obj) + 3) % 6, 4), dude_elevation);
                     end
                 end
             end
             if (party_member_obj(16777294) != 0) then begin
                 if (party_member_obj(16777294) != 0) then begin
-                    if (has_trait(1, party_member_obj(16777294), 666)) then begin
-                        critter_attempt_placement(party_member_obj(16777294), tile_num_in_direction(tile_num(dude_obj), (has_trait(1, dude_obj, 10) + 3) % 6, 4), elevation(dude_obj));
+                    if (obj_is_visible_flag(party_member_obj(16777294))) then begin
+                        critter_attempt_placement(party_member_obj(16777294), tile_num_in_direction(dude_tile, (obj_get_rot(dude_obj) + 3) % 6, 4), dude_elevation);
                     end
                 end
             end
             if (party_member_obj(16777729) != 0) then begin
                 anim(party_member_obj(16777729), 1000, 1);
             end
-            if ((party_member_obj(16777779) != 0) and ((global_var(910) bwand 32) == 0)) then begin
+            if ((party_member_obj(16777779) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 32) == 0)) then begin
                 anim(party_member_obj(16777779), 1000, 1);
             end
-            if ((party_member_obj(16777811) != 0) and ((global_var(910) bwand 64) == 0)) then begin
+            if ((party_member_obj(16777811) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 64) == 0)) then begin
                 anim(party_member_obj(16777811), 1000, 1);
             end
-            if ((party_member_obj(16777814) != 0) and ((global_var(910) bwand 128) == 0)) then begin
+            if ((party_member_obj(16777814) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 128) == 0)) then begin
                 anim(party_member_obj(16777814), 1000, 1);
             end
             if (party_member_obj(16778017) != 0) then begin
@@ -462,12 +462,12 @@ begin
             if (party_member_obj(16777812) != 0) then begin
                 anim(party_member_obj(16777812), 1000, 1);
             end
-            metarule3(108, 15506, 0, 0);
+            tile_set_center(15506);
             game_ui_disable;
             add_timer_event(self_obj, 5, 1);
         end
         if (sequence == 2) then begin
-            metarule3(108, 15506, 0, 0);
+            tile_set_center(15506);
             reg_anim_func(2, dude_obj);
             reg_anim_func(2, sebastian);
             reg_anim_func(2, lovec1);
@@ -507,7 +507,7 @@ begin
             add_timer_event(self_obj, game_ticks(2), 1);
         end
         if (sequence == 9) then begin
-            float_msg(sebastian, message_str(1771, 100), 2);
+            float_msg(sebastian, mstr(100), FLOAT_MSG_RED);
             add_timer_event(self_obj, game_ticks(4), 1);
         end
         if (sequence == 10) then begin
@@ -515,13 +515,13 @@ begin
             if (party_member_obj(16777729) != 0) then begin
                 anim(party_member_obj(16777729), 1000, 1);
             end
-            if ((party_member_obj(16777779) != 0) and ((global_var(910) bwand 32) == 0)) then begin
+            if ((party_member_obj(16777779) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 32) == 0)) then begin
                 anim(party_member_obj(16777779), 1000, 1);
             end
-            if ((party_member_obj(16777811) != 0) and ((global_var(910) bwand 64) == 0)) then begin
+            if ((party_member_obj(16777811) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 64) == 0)) then begin
                 anim(party_member_obj(16777811), 1000, 1);
             end
-            if ((party_member_obj(16777814) != 0) and ((global_var(910) bwand 128) == 0)) then begin
+            if ((party_member_obj(16777814) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 128) == 0)) then begin
                 anim(party_member_obj(16777814), 1000, 1);
             end
             if (party_member_obj(16778017) != 0) then begin
@@ -543,7 +543,7 @@ begin
                 anim(party_member_obj(16777812), 1000, 1);
             end
             anim(sebastian, 1000, 4);
-            float_msg(sebastian, message_str(1771, 101), 2);
+            float_msg(sebastian, mstr(101), FLOAT_MSG_RED);
             anim(mutant3, 1000, 1);
             add_timer_event(self_obj, game_ticks(7), 1);
         end
@@ -554,7 +554,7 @@ begin
             if (not(tile_contains_pid_obj(16918, 0, 16777985))) then begin
                 animate_move_obj_to_tile(Marry, 16918, 0);
             end
-            float_msg(Dernus, message_str(1771, 200), 8);
+            float_msg(Dernus, mstr(200), FLOAT_MSG_YELLOW);
             add_timer_event(self_obj, 4, 1);
         end
         if (sequence == 12) then begin
@@ -562,7 +562,7 @@ begin
             add_timer_event(self_obj, game_ticks(6), 1);
         end
         if (sequence == 13) then begin
-            float_msg(Dernus, message_str(1771, 201), 8);
+            float_msg(Dernus, mstr(201), FLOAT_MSG_YELLOW);
             add_timer_event(self_obj, game_ticks(5), 1);
         end
         if (sequence == 14) then begin
@@ -572,15 +572,15 @@ begin
             if (not(tile_contains_pid_obj(16918, 0, 16777985))) then begin
                 animate_move_obj_to_tile(Marry, 16918, 0);
             end
-            float_msg(sebastian, message_str(1771, 300), 2);
+            float_msg(sebastian, mstr(300), FLOAT_MSG_RED);
             add_timer_event(self_obj, game_ticks(5), 1);
         end
         if (sequence == 15) then begin
-            float_msg(sebastian, message_str(1771, 301), 2);
+            float_msg(sebastian, mstr(301), FLOAT_MSG_RED);
             add_timer_event(self_obj, game_ticks(8), 1);
         end
         if (sequence == 16) then begin
-            float_msg(Dernus, message_str(1771, 400), 8);
+            float_msg(Dernus, mstr(400), FLOAT_MSG_YELLOW);
             add_timer_event(self_obj, game_ticks(4), 1);
         end
         if (sequence == 17) then begin
@@ -637,8 +637,8 @@ begin
             add_timer_event(self_obj, game_ticks(2), 1);
         end
         if (sequence == 19) then begin
-            set_global_var(867, global_var(867) bwor 1);
-            set_global_var(907, 3);
+            set_global_var(GVAR_AQ_ALBUQ_DEATHS, global_var(GVAR_AQ_ALBUQ_DEATHS) bwor 1);
+            set_global_var(GVAR_MC_QUEST_BRING_SEBASTIAN, 3);
             add_timer_event(self_obj, game_ticks(1), 1);
         end
         if (sequence == 20) then begin
@@ -649,103 +649,103 @@ begin
     if (fixed_param == 2) then begin
         sequence := sequence + 1;
         if (sequence == 1) then begin
-            gfade_in(1);
+            fade_in;
             move_to(dude_obj, 14923, 0);
             anim(dude_obj, 1000, 1);
             if (party_member_obj(16777729) != 0) then begin
                 if (party_member_obj(16777729) != 0) then begin
-                    if (has_trait(1, party_member_obj(16777729), 666)) then begin
-                        critter_attempt_placement(party_member_obj(16777729), tile_num_in_direction(tile_num(dude_obj), (has_trait(1, dude_obj, 10) + 3) % 6, 4), elevation(dude_obj));
+                    if (obj_is_visible_flag(party_member_obj(16777729))) then begin
+                        critter_attempt_placement(party_member_obj(16777729), tile_num_in_direction(dude_tile, (obj_get_rot(dude_obj) + 3) % 6, 4), dude_elevation);
                     end
                 end
             end
-            if ((party_member_obj(16777779) != 0) and ((global_var(910) bwand 32) == 0)) then begin
+            if ((party_member_obj(16777779) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 32) == 0)) then begin
                 if (party_member_obj(16777779) != 0) then begin
-                    if (has_trait(1, party_member_obj(16777779), 666)) then begin
-                        critter_attempt_placement(party_member_obj(16777779), tile_num_in_direction(tile_num(dude_obj), (has_trait(1, dude_obj, 10) + 3) % 6, 4), elevation(dude_obj));
+                    if (obj_is_visible_flag(party_member_obj(16777779))) then begin
+                        critter_attempt_placement(party_member_obj(16777779), tile_num_in_direction(dude_tile, (obj_get_rot(dude_obj) + 3) % 6, 4), dude_elevation);
                     end
                 end
             end
-            if ((party_member_obj(16777811) != 0) and ((global_var(910) bwand 64) == 0)) then begin
+            if ((party_member_obj(16777811) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 64) == 0)) then begin
                 if (party_member_obj(16777811) != 0) then begin
-                    if (has_trait(1, party_member_obj(16777811), 666)) then begin
-                        critter_attempt_placement(party_member_obj(16777811), tile_num_in_direction(tile_num(dude_obj), (has_trait(1, dude_obj, 10) + 3) % 6, 4), elevation(dude_obj));
+                    if (obj_is_visible_flag(party_member_obj(16777811))) then begin
+                        critter_attempt_placement(party_member_obj(16777811), tile_num_in_direction(dude_tile, (obj_get_rot(dude_obj) + 3) % 6, 4), dude_elevation);
                     end
                 end
             end
             if (party_member_obj(16777812) != 0) then begin
                 if (party_member_obj(16777812) != 0) then begin
-                    if (has_trait(1, party_member_obj(16777812), 666)) then begin
-                        critter_attempt_placement(party_member_obj(16777812), tile_num_in_direction(tile_num(dude_obj), (has_trait(1, dude_obj, 10) + 3) % 6, 4), elevation(dude_obj));
+                    if (obj_is_visible_flag(party_member_obj(16777812))) then begin
+                        critter_attempt_placement(party_member_obj(16777812), tile_num_in_direction(dude_tile, (obj_get_rot(dude_obj) + 3) % 6, 4), dude_elevation);
                     end
                 end
             end
-            if ((party_member_obj(16777814) != 0) and ((global_var(910) bwand 128) == 0)) then begin
+            if ((party_member_obj(16777814) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 128) == 0)) then begin
                 if (party_member_obj(16777814) != 0) then begin
-                    if (has_trait(1, party_member_obj(16777814), 666)) then begin
-                        critter_attempt_placement(party_member_obj(16777814), tile_num_in_direction(tile_num(dude_obj), (has_trait(1, dude_obj, 10) + 3) % 6, 4), elevation(dude_obj));
+                    if (obj_is_visible_flag(party_member_obj(16777814))) then begin
+                        critter_attempt_placement(party_member_obj(16777814), tile_num_in_direction(dude_tile, (obj_get_rot(dude_obj) + 3) % 6, 4), dude_elevation);
                     end
                 end
             end
             if (party_member_obj(16777748) != 0) then begin
                 if (party_member_obj(16777748) != 0) then begin
-                    if (has_trait(1, party_member_obj(16777748), 666)) then begin
-                        critter_attempt_placement(party_member_obj(16777748), tile_num_in_direction(tile_num(dude_obj), (has_trait(1, dude_obj, 10) + 3) % 6, 4), elevation(dude_obj));
+                    if (obj_is_visible_flag(party_member_obj(16777748))) then begin
+                        critter_attempt_placement(party_member_obj(16777748), tile_num_in_direction(dude_tile, (obj_get_rot(dude_obj) + 3) % 6, 4), dude_elevation);
                     end
                 end
             end
             if (party_member_obj(16778017) != 0) then begin
                 if (party_member_obj(16778017) != 0) then begin
-                    if (has_trait(1, party_member_obj(16778017), 666)) then begin
-                        critter_attempt_placement(party_member_obj(16778017), tile_num_in_direction(tile_num(dude_obj), (has_trait(1, dude_obj, 10) + 3) % 6, 4), elevation(dude_obj));
+                    if (obj_is_visible_flag(party_member_obj(16778017))) then begin
+                        critter_attempt_placement(party_member_obj(16778017), tile_num_in_direction(dude_tile, (obj_get_rot(dude_obj) + 3) % 6, 4), dude_elevation);
                     end
                 end
             end
             if (party_member_obj(16778016) != 0) then begin
                 if (party_member_obj(16778016) != 0) then begin
-                    if (has_trait(1, party_member_obj(16778016), 666)) then begin
-                        critter_attempt_placement(party_member_obj(16778016), tile_num_in_direction(tile_num(dude_obj), (has_trait(1, dude_obj, 10) + 3) % 6, 4), elevation(dude_obj));
+                    if (obj_is_visible_flag(party_member_obj(16778016))) then begin
+                        critter_attempt_placement(party_member_obj(16778016), tile_num_in_direction(dude_tile, (obj_get_rot(dude_obj) + 3) % 6, 4), dude_elevation);
                     end
                 end
             end
             if (party_member_obj(16778049) != 0) then begin
                 if (party_member_obj(16778049) != 0) then begin
-                    if (has_trait(1, party_member_obj(16778049), 666)) then begin
-                        critter_attempt_placement(party_member_obj(16778049), tile_num_in_direction(tile_num(dude_obj), (has_trait(1, dude_obj, 10) + 3) % 6, 4), elevation(dude_obj));
+                    if (obj_is_visible_flag(party_member_obj(16778049))) then begin
+                        critter_attempt_placement(party_member_obj(16778049), tile_num_in_direction(dude_tile, (obj_get_rot(dude_obj) + 3) % 6, 4), dude_elevation);
                     end
                 end
             end
             if (party_member_obj(16778050) != 0) then begin
                 if (party_member_obj(16778050) != 0) then begin
-                    if (has_trait(1, party_member_obj(16778050), 666)) then begin
-                        critter_attempt_placement(party_member_obj(16778050), tile_num_in_direction(tile_num(dude_obj), (has_trait(1, dude_obj, 10) + 3) % 6, 4), elevation(dude_obj));
+                    if (obj_is_visible_flag(party_member_obj(16778050))) then begin
+                        critter_attempt_placement(party_member_obj(16778050), tile_num_in_direction(dude_tile, (obj_get_rot(dude_obj) + 3) % 6, 4), dude_elevation);
                     end
                 end
             end
             if (party_member_obj(16778051) != 0) then begin
                 if (party_member_obj(16778051) != 0) then begin
-                    if (has_trait(1, party_member_obj(16778051), 666)) then begin
-                        critter_attempt_placement(party_member_obj(16778051), tile_num_in_direction(tile_num(dude_obj), (has_trait(1, dude_obj, 10) + 3) % 6, 4), elevation(dude_obj));
+                    if (obj_is_visible_flag(party_member_obj(16778051))) then begin
+                        critter_attempt_placement(party_member_obj(16778051), tile_num_in_direction(dude_tile, (obj_get_rot(dude_obj) + 3) % 6, 4), dude_elevation);
                     end
                 end
             end
             if (party_member_obj(16777294) != 0) then begin
                 if (party_member_obj(16777294) != 0) then begin
-                    if (has_trait(1, party_member_obj(16777294), 666)) then begin
-                        critter_attempt_placement(party_member_obj(16777294), tile_num_in_direction(tile_num(dude_obj), (has_trait(1, dude_obj, 10) + 3) % 6, 4), elevation(dude_obj));
+                    if (obj_is_visible_flag(party_member_obj(16777294))) then begin
+                        critter_attempt_placement(party_member_obj(16777294), tile_num_in_direction(dude_tile, (obj_get_rot(dude_obj) + 3) % 6, 4), dude_elevation);
                     end
                 end
             end
             if (party_member_obj(16777729) != 0) then begin
                 anim(party_member_obj(16777729), 1000, 1);
             end
-            if ((party_member_obj(16777779) != 0) and ((global_var(910) bwand 32) == 0)) then begin
+            if ((party_member_obj(16777779) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 32) == 0)) then begin
                 anim(party_member_obj(16777779), 1000, 1);
             end
-            if ((party_member_obj(16777811) != 0) and ((global_var(910) bwand 64) == 0)) then begin
+            if ((party_member_obj(16777811) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 64) == 0)) then begin
                 anim(party_member_obj(16777811), 1000, 1);
             end
-            if ((party_member_obj(16777814) != 0) and ((global_var(910) bwand 128) == 0)) then begin
+            if ((party_member_obj(16777814) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 128) == 0)) then begin
                 anim(party_member_obj(16777814), 1000, 1);
             end
             if (party_member_obj(16778017) != 0) then begin
@@ -766,12 +766,12 @@ begin
             if (party_member_obj(16777812) != 0) then begin
                 anim(party_member_obj(16777812), 1000, 1);
             end
-            metarule3(108, 15506, 0, 0);
+            tile_set_center(15506);
             game_ui_disable;
             add_timer_event(self_obj, 5, 2);
         end
         if (sequence == 2) then begin
-            metarule3(108, 15506, 0, 0);
+            tile_set_center(15506);
             reg_anim_func(2, dude_obj);
             reg_anim_func(2, sebastian);
             reg_anim_func(2, lovec1);
@@ -815,7 +815,7 @@ begin
         if (sequence == 8) then begin
             animate_move_obj_to_tile(lovec5b, 14922, 0);
             animate_move_obj_to_tile(lovec7b, 16922, 0);
-            float_msg(sebastian, message_str(1771, 1000), 2);
+            float_msg(sebastian, mstr(1000), FLOAT_MSG_RED);
             add_timer_event(self_obj, game_ticks(1), 2);
         end
         if (sequence == 9) then begin
@@ -827,13 +827,13 @@ begin
             if (party_member_obj(16777729) != 0) then begin
                 anim(party_member_obj(16777729), 1000, 1);
             end
-            if ((party_member_obj(16777779) != 0) and ((global_var(910) bwand 32) == 0)) then begin
+            if ((party_member_obj(16777779) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 32) == 0)) then begin
                 anim(party_member_obj(16777779), 1000, 1);
             end
-            if ((party_member_obj(16777811) != 0) and ((global_var(910) bwand 64) == 0)) then begin
+            if ((party_member_obj(16777811) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 64) == 0)) then begin
                 anim(party_member_obj(16777811), 1000, 1);
             end
-            if ((party_member_obj(16777814) != 0) and ((global_var(910) bwand 128) == 0)) then begin
+            if ((party_member_obj(16777814) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 128) == 0)) then begin
                 anim(party_member_obj(16777814), 1000, 1);
             end
             if (party_member_obj(16778017) != 0) then begin
@@ -855,12 +855,12 @@ begin
                 anim(party_member_obj(16777812), 1000, 1);
             end
             anim(sebastian, 1000, 5);
-            float_msg(sebastian, message_str(1771, 1001 + (get_critter_stat(dude_obj, 34) == 1)), 2);
+            float_msg(sebastian, mstr(1001 + (dude_gender == 1)), FLOAT_MSG_RED);
             anim(mutant3, 1000, 1);
             add_timer_event(self_obj, game_ticks(7), 2);
         end
         if (sequence == 11) then begin
-            float_msg(sebastian, message_str(1771, 1003 + (get_critter_stat(dude_obj, 34) == 1)), 2);
+            float_msg(sebastian, mstr(1003 + (dude_gender == 1)), FLOAT_MSG_RED);
             add_timer_event(self_obj, game_ticks(7), 2);
         end
         if (sequence == 12) then begin
@@ -870,7 +870,7 @@ begin
             if (not(tile_contains_pid_obj(16918, 0, 16777985))) then begin
                 animate_move_obj_to_tile(Marry, 16918, 0);
             end
-            float_msg(Dernus, message_str(1771, 1005), 8);
+            float_msg(Dernus, mstr(1005), FLOAT_MSG_YELLOW);
             add_timer_event(self_obj, 4, 2);
         end
         if (sequence == 13) then begin
@@ -879,7 +879,7 @@ begin
         end
         if (sequence == 14) then begin
             anim(Dernus, 1000, 4);
-            float_msg(Dernus, message_str(1771, 1006), 8);
+            float_msg(Dernus, mstr(1006), FLOAT_MSG_YELLOW);
             add_timer_event(self_obj, game_ticks(9), 2);
         end
         if (sequence == 15) then begin
@@ -890,11 +890,11 @@ begin
                 animate_move_obj_to_tile(Marry, 16918, 0);
             end
             anim(Dernus, 1000, 3);
-            float_msg(sebastian, message_str(1771, 1007), 2);
+            float_msg(sebastian, mstr(1007), FLOAT_MSG_RED);
             add_timer_event(self_obj, game_ticks(6), 2);
         end
         if (sequence == 16) then begin
-            float_msg(Dernus, message_str(1771, 1008), 8);
+            float_msg(Dernus, mstr(1008), FLOAT_MSG_YELLOW);
             add_timer_event(self_obj, game_ticks(4), 2);
         end
         if (sequence == 17) then begin
@@ -916,9 +916,9 @@ begin
             add_timer_event(self_obj, game_ticks(1), 2);
         end
         if (sequence == 19) then begin
-            set_global_var(867, global_var(867) bwor 1);
-            set_global_var(907, 3);
-            set_global_var(1030, 2);
+            set_global_var(GVAR_AQ_ALBUQ_DEATHS, global_var(GVAR_AQ_ALBUQ_DEATHS) bwor 1);
+            set_global_var(GVAR_MC_QUEST_BRING_SEBASTIAN, 3);
+            set_global_var(GVAR_MC_SEBASTIAN_NOT_CONVINCED, 2);
             game_ui_enable;
             attack_setup(lovec1, mutant1);
         end
@@ -930,15 +930,15 @@ end
 
 procedure generate_hunters
 begin
-    sebastian := create_object_sid(16777949, 15120, 0, -1);
-    lovec1 := create_object_sid(16778017, 0, 0, 558);
+    sebastian := create_object(16777949, 15120, 0);
+    lovec1 := create_object_sid(16778017, 0, 0, SCRIPT_MHUNTER);
     critter_attempt_placement(lovec1, 14721, 0);
-    critter_add_trait(lovec1, 1, 6, 290);
-    critter_add_trait(lovec1, 1, 5, 3);
-    lovec2 := create_object_sid(16778016, 0, 0, 558);
+    critter_add_trait(lovec1, TRAIT_OBJECT, OBJECT_TEAM_NUM, 290);
+    critter_add_trait(lovec1, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_TOUGH_GUARD_NO_TALKING);
+    lovec2 := create_object_sid(16778016, 0, 0, SCRIPT_MHUNTER);
     critter_attempt_placement(lovec2, 15322, 0);
-    critter_add_trait(lovec2, 1, 6, 290);
-    critter_add_trait(lovec2, 1, 5, 3);
+    critter_add_trait(lovec2, TRAIT_OBJECT, OBJECT_TEAM_NUM, 290);
+    critter_add_trait(lovec2, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_TOUGH_GUARD_NO_TALKING);
     move_to(Dernus, 15095, 0);
     move_to(mutant1, 13898, 0);
     move_to(mutant2, 16294, 0);
@@ -947,15 +947,15 @@ begin
     move_to(mutant6, 16893, 0);
     move_to(mutant7, 14098, 0);
     move_to(mutant8, 16694, 0);
-    restock_obj := create_object_sid(596, 0, 0, -1);
+    restock_obj := create_object(596, 0, 0);
     add_mult_objs_to_inven(sebastian, restock_obj, 1);
-    restock_obj := create_object_sid(35, 0, 0, -1);
+    restock_obj := create_object(35, 0, 0);
     add_mult_objs_to_inven(sebastian, restock_obj, 2);
-    restock_obj := create_object_sid(13, 0, 0, -1);
+    restock_obj := create_object(13, 0, 0);
     add_mult_objs_to_inven(lovec1, restock_obj, 1);
-    restock_obj := create_object_sid(14, 0, 0, -1);
+    restock_obj := create_object(14, 0, 0);
     add_mult_objs_to_inven(lovec1, restock_obj, 3);
-    restock_obj := create_object_sid(115, 0, 0, -1);
+    restock_obj := create_object(115, 0, 0);
     add_mult_objs_to_inven(lovec2, restock_obj, 1);
     wield_obj_critter(sebastian, obj_carrying_pid_obj(sebastian, 596));
     wield_obj_critter(lovec1, obj_carrying_pid_obj(lovec1, 13));
@@ -968,48 +968,48 @@ begin
     anim(mutant2, 1000, 4);
     anim(mutant3, 1000, 1);
     anim(Marry, 1000, 3);
-    if (global_var(1030) == 1) then begin
-        lovec3b := create_object_sid(16778017, 0, 0, 558);
+    if (global_var(GVAR_MC_SEBASTIAN_NOT_CONVINCED) == 1) then begin
+        lovec3b := create_object_sid(16778017, 0, 0, SCRIPT_MHUNTER);
         critter_attempt_placement(lovec3b, 16154, 0);
-        critter_add_trait(lovec3b, 1, 6, 290);
-        critter_add_trait(lovec3b, 1, 5, 3);
-        lovec4b := create_object_sid(16778018, 0, 0, 558);
+        critter_add_trait(lovec3b, TRAIT_OBJECT, OBJECT_TEAM_NUM, 290);
+        critter_add_trait(lovec3b, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_TOUGH_GUARD_NO_TALKING);
+        lovec4b := create_object_sid(16778018, 0, 0, SCRIPT_MHUNTER);
         critter_attempt_placement(lovec4b, 16754, 0);
-        critter_add_trait(lovec4b, 1, 6, 290);
-        critter_add_trait(lovec4b, 1, 5, 3);
-        lovec5b := create_object_sid(16778018, 0, 0, 558);
+        critter_add_trait(lovec4b, TRAIT_OBJECT, OBJECT_TEAM_NUM, 290);
+        critter_add_trait(lovec4b, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_TOUGH_GUARD_NO_TALKING);
+        lovec5b := create_object_sid(16778018, 0, 0, SCRIPT_MHUNTER);
         critter_attempt_placement(lovec5b, 17354, 0);
-        critter_add_trait(lovec5b, 1, 6, 290);
-        critter_add_trait(lovec5b, 1, 5, 3);
-        lovec6b := create_object_sid(16778016, 0, 0, 558);
+        critter_add_trait(lovec5b, TRAIT_OBJECT, OBJECT_TEAM_NUM, 290);
+        critter_add_trait(lovec5b, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_TOUGH_GUARD_NO_TALKING);
+        lovec6b := create_object_sid(16778016, 0, 0, SCRIPT_MHUNTER);
         critter_attempt_placement(lovec6b, 18557, 0);
-        critter_add_trait(lovec6b, 1, 6, 290);
-        critter_add_trait(lovec6b, 1, 5, 3);
-        lovec7b := create_object_sid(16778018, 0, 0, 558);
+        critter_add_trait(lovec6b, TRAIT_OBJECT, OBJECT_TEAM_NUM, 290);
+        critter_add_trait(lovec6b, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_TOUGH_GUARD_NO_TALKING);
+        lovec7b := create_object_sid(16778018, 0, 0, SCRIPT_MHUNTER);
         critter_attempt_placement(lovec7b, 18157, 0);
-        critter_add_trait(lovec7b, 1, 6, 290);
-        critter_add_trait(lovec7b, 1, 5, 3);
-        restock_obj := create_object_sid(603, 0, 0, -1);
+        critter_add_trait(lovec7b, TRAIT_OBJECT, OBJECT_TEAM_NUM, 290);
+        critter_add_trait(lovec7b, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_TOUGH_GUARD_NO_TALKING);
+        restock_obj := create_object(603, 0, 0);
         add_mult_objs_to_inven(lovec3b, restock_obj, 1);
-        restock_obj := create_object_sid(363, 0, 0, -1);
+        restock_obj := create_object(363, 0, 0);
         add_mult_objs_to_inven(lovec3b, restock_obj, 1);
-        restock_obj := create_object_sid(11, 0, 0, -1);
+        restock_obj := create_object(11, 0, 0);
         add_mult_objs_to_inven(lovec4b, restock_obj, 1);
-        restock_obj := create_object_sid(115, 0, 0, -1);
+        restock_obj := create_object(115, 0, 0);
         add_mult_objs_to_inven(lovec5b, restock_obj, 1);
-        restock_obj := create_object_sid(603, 0, 0, -1);
+        restock_obj := create_object(603, 0, 0);
         add_mult_objs_to_inven(lovec6b, restock_obj, 1);
-        restock_obj := create_object_sid(363, 0, 0, -1);
+        restock_obj := create_object(363, 0, 0);
         add_mult_objs_to_inven(lovec6b, restock_obj, 1);
-        restock_obj := create_object_sid(13, 0, 0, -1);
+        restock_obj := create_object(13, 0, 0);
         add_mult_objs_to_inven(lovec7b, restock_obj, 1);
-        restock_obj := create_object_sid(14, 0, 0, -1);
+        restock_obj := create_object(14, 0, 0);
         add_mult_objs_to_inven(lovec7b, restock_obj, 5);
-        restock_obj := create_object_sid(144, 0, 0, -1);
+        restock_obj := create_object(144, 0, 0);
         add_mult_objs_to_inven(lovec4b, restock_obj, 1);
-        restock_obj := create_object_sid(144, 0, 0, -1);
+        restock_obj := create_object(144, 0, 0);
         add_mult_objs_to_inven(lovec5b, restock_obj, 1);
-        restock_obj := create_object_sid(144, 0, 0, -1);
+        restock_obj := create_object(144, 0, 0);
         add_mult_objs_to_inven(lovec7b, restock_obj, 1);
         wield_obj_critter(lovec3b, obj_carrying_pid_obj(lovec3b, 603));
         wield_obj_critter(lovec4b, obj_carrying_pid_obj(lovec4b, 11));
@@ -1023,4 +1023,3 @@ begin
         anim(lovec7b, 1000, 0);
     end
 end
-
