diff --git a/Mods/Resurrection/Original/NHMRAT.ssl b/Mods/Resurrection/ReDefine/NHMRAT.ssl
index 2488f55..dc50a6c 100644
--- a/Mods/Resurrection/Original/NHMRAT.ssl
+++ b/Mods/Resurrection/ReDefine/NHMRAT.ssl
@@ -41,12 +41,12 @@ begin
             return 1;
         end
     end
-    if ((party_member_obj(16777779) != 0) and ((global_var(910) bwand 32) == 0)) then begin
+    if ((party_member_obj(16777779) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 32) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777779)) <= 5) then begin
             return 1;
         end
     end
-    if ((party_member_obj(16777811) != 0) and ((global_var(910) bwand 64) == 0)) then begin
+    if ((party_member_obj(16777811) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 64) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777811)) <= 5) then begin
             return 1;
         end
@@ -61,7 +61,7 @@ begin
             return 1;
         end
     end
-    if ((party_member_obj(16777814) != 0) and ((global_var(910) bwand 128) == 0)) then begin
+    if ((party_member_obj(16777814) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 128) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777814)) <= 5) then begin
             return 1;
         end
@@ -118,20 +118,20 @@ end
 
 procedure map_enter_p_proc
 begin
-    critter_add_trait(self_obj, 1, 6, 3);
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 3);
     if (not(local_var(4))) then begin
         set_local_var(4, tile_num(self_obj));
     end
     add_timer_event(self_obj, game_ticks(random(1, 5)), 0);
-    if (metarule(14, 0)) then begin
-        set_global_var(699, global_var(699) + 1);
+    if (map_first_run) then begin
+        set_global_var(GVAR_NH_CAVE_CRITTERS, global_var(GVAR_NH_CAVE_CRITTERS) + 1);
     end
 end
 
 procedure critter_p_proc
 begin
     if (obj_can_see_obj(self_obj, dude_obj)) then begin
-        attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+        attack(dude_obj);
     end
 end
 
@@ -156,15 +156,14 @@ end
 procedure damage_p_proc
 begin
     if (source_obj == dude_obj) then begin
-        attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+        attack(dude_obj);
     end
 end
 
 procedure destroy_p_proc
 begin
-    set_global_var(699, global_var(699) - 1);
-    if (global_var(699) == 0) then begin
+    set_global_var(GVAR_NH_CAVE_CRITTERS, global_var(GVAR_NH_CAVE_CRITTERS) - 1);
+    if (global_var(GVAR_NH_CAVE_CRITTERS) == 0) then begin
         add_timer_event(cave_obj, 0, 0);
     end
 end
-
