diff --git a/Mods/Resurrection/Original/NHOSWALB.ssl b/Mods/Resurrection/ReDefine/NHOSWALB.ssl
index bce0ed2..acbf305 100644
--- a/Mods/Resurrection/Original/NHOSWALB.ssl
+++ b/Mods/Resurrection/ReDefine/NHOSWALB.ssl
@@ -57,12 +57,12 @@ begin
             return 1;
         end
     end
-    if ((party_member_obj(16777779) != 0) and ((global_var(910) bwand 32) == 0)) then begin
+    if ((party_member_obj(16777779) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 32) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777779)) <= 5) then begin
             return 1;
         end
     end
-    if ((party_member_obj(16777811) != 0) and ((global_var(910) bwand 64) == 0)) then begin
+    if ((party_member_obj(16777811) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 64) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777811)) <= 5) then begin
             return 1;
         end
@@ -77,7 +77,7 @@ begin
             return 1;
         end
     end
-    if ((party_member_obj(16777814) != 0) and ((global_var(910) bwand 128) == 0)) then begin
+    if ((party_member_obj(16777814) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 128) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777814)) <= 5) then begin
             return 1;
         end
@@ -134,7 +134,7 @@ end
 
 procedure map_enter_p_proc
 begin
-    critter_add_trait(self_obj, 1, 6, 211);
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 211);
 end
 
 procedure talk_p_proc
@@ -149,11 +149,11 @@ end
 procedure critter_p_proc
 begin
     if (fleeing and not(anim_busy(self_obj))) then begin
-        metarule3(108, tile_num(dude_obj), 0, 0);
+        tile_set_center(dude_tile);
         game_ui_enable;
         destroy_object(self_obj);
     end
-    if (obj_can_see_obj(self_obj, dude_obj) and (tile_distance(tile_num(self_obj), tile_num(dude_obj)) < 8)) then begin
+    if (obj_can_see_obj(self_obj, dude_obj) and (tile_distance(tile_num(self_obj), dude_tile) < 8)) then begin
         dialogue_system_enter;
     end
 end
@@ -161,9 +161,9 @@ end
 procedure damage_p_proc
 begin
     if (source_obj == dude_obj) then begin
-        set_global_var(700, 1);
+        set_global_var(GVAR_ENEMY_NEWHOPE, 1);
     end
-    set_global_var(701, get_critter_stat(self_obj, 7) - get_critter_stat(self_obj, 35));
+    set_global_var(GVAR_NH_OSWALD_DMG, get_critter_stat(self_obj, STAT_max_hp) - get_critter_stat(self_obj, STAT_current_hp));
 end
 
 procedure combat_p_proc
@@ -183,7 +183,7 @@ end
 procedure look_at_p_proc
 begin
     script_overrides;
-    display_msg(message_str(1309, 50));
+    display_mstr(50);
 end
 
 procedure description_p_proc
@@ -199,28 +199,27 @@ end
 
 procedure Node100
 begin
-    gsay_reply(1309, message_str(1309, 100));
-    giq_option(4, 1309, 101, Node150, 50);
-    giq_option(-3, 1309, 102, Node200, 50);
+    Reply(mstr(100));
+    giq_option(4, NAME, 101, Node150, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 102, Node200, NEUTRAL_REACTION);
 end
 
 procedure Node150
 begin
-    gsay_reply(1309, message_str(1309, 150));
-    giq_option(4, 1309, 103, Node999, 50);
+    Reply(mstr(150));
+    giq_option(4, NAME, 103, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node200
 begin
-    gsay_reply(1309, message_str(1309, 200 + (get_critter_stat(dude_obj, 34) == 1)));
-    giq_option(-3, 1309, 103, Node999, 50);
+    Reply(mstr(200 + (dude_gender == 1)));
+    giq_option(-3, NAME, 103, Node999, NEUTRAL_REACTION);
 end
 
 procedure timed_event_p_proc
 begin
-    metarule3(108, 17714, 0, 0);
+    tile_set_center(17714);
     animate_move_obj_to_tile(self_obj, 18332, 1);
     fleeing := 1;
     game_ui_disable;
 end
-
