diff --git a/Mods/Resurrection/Original/PCOM1.ssl b/Mods/Resurrection/ReDefine/PCOM1.ssl
index 397912b..dec2a80 100644
--- a/Mods/Resurrection/Original/PCOM1.ssl
+++ b/Mods/Resurrection/ReDefine/PCOM1.ssl
@@ -73,12 +73,12 @@ begin
             return 1;
         end
     end
-    if ((party_member_obj(16777779) != 0) and ((global_var(910) bwand 32) == 0)) then begin
+    if ((party_member_obj(16777779) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 32) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777779)) <= 5) then begin
             return 1;
         end
     end
-    if ((party_member_obj(16777811) != 0) and ((global_var(910) bwand 64) == 0)) then begin
+    if ((party_member_obj(16777811) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 64) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777811)) <= 5) then begin
             return 1;
         end
@@ -93,7 +93,7 @@ begin
             return 1;
         end
     end
-    if ((party_member_obj(16777814) != 0) and ((global_var(910) bwand 128) == 0)) then begin
+    if ((party_member_obj(16777814) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 128) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777814)) <= 5) then begin
             return 1;
         end
@@ -152,16 +152,16 @@ procedure use_p_proc
 begin
     script_overrides;
     if (combat_is_initialized) then begin
-        display_msg(message_str(195, 301));
+        display_mstr(301);
     end
     else begin
-        if (global_var(919) == 0) then begin
-            display_msg(message_str(195, 300));
+        if (global_var(GVAR_PROGEMA_STATUS) == 0) then begin
+            display_mstr(300);
         end
         else begin
             start_gdialog(195, self_obj, 4, -1, -1);
             gsay_start;
-            if (global_var(919) == 1) then begin
+            if (global_var(GVAR_PROGEMA_STATUS) == 1) then begin
                 call Node500();
             end
             else begin
@@ -176,17 +176,17 @@ end
 procedure look_at_p_proc
 begin
     script_overrides;
-    display_msg(message_str(195, 100));
+    display_mstr(100);
 end
 
 procedure description_p_proc
 begin
     script_overrides;
-    if (global_var(919) >= 1) then begin
-        display_msg(message_str(195, 201));
+    if (global_var(GVAR_PROGEMA_STATUS) >= 1) then begin
+        display_mstr(201);
     end
     else begin
-        display_msg(message_str(195, 200));
+        display_mstr(200);
     end
 end
 
@@ -217,310 +217,310 @@ end
 procedure Node500
 begin
     if ((local_var(0) bwand 1) != 0) then begin
-        gsay_reply(195, message_str(195, 500) + message_str(195, 501));
+        Reply(mstr(500) + mstr(501));
     end
     else begin
-        gsay_reply(195, 500);
+        Reply(500);
     end
-    giq_option(4, 195, 502, Node550, 50);
-    giq_option(4, 195, 503, Node950, 50);
+    giq_option(4, NAME, 502, Node550, NEUTRAL_REACTION);
+    giq_option(4, NAME, 503, Node950, NEUTRAL_REACTION);
     if ((local_var(0) bwand 2) == 0) then begin
-        giq_option(4, 195, 504, Node1550, 50);
+        giq_option(4, NAME, 504, Node1550, NEUTRAL_REACTION);
     end
     if ((local_var(0) bwand 2) != 0) then begin
-        giq_option(4, 195, 505, Change_to_robot, 50);
+        giq_option(4, NAME, 505, Change_to_robot, NEUTRAL_REACTION);
     end
     if ((local_var(0) bwand 4) == 0) then begin
-        giq_option(4, 195, 506, Node1750, 50);
+        giq_option(4, NAME, 506, Node1750, NEUTRAL_REACTION);
     end
-    giq_option(4, 195, 507, Node999, 50);
-    giq_option(-3, 195, 508, Node999, 50);
+    giq_option(4, NAME, 507, Node999, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 508, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node550
 begin
-    gsay_reply(195, 550);
+    Reply(550);
     if ((local_var(0) bwand 1) != 0) then begin
-        giq_option(4, 195, 551, Node750, 50);
-        giq_option(4, 195, 552, Node850, 50);
+        giq_option(4, NAME, 551, Node750, NEUTRAL_REACTION);
+        giq_option(4, NAME, 552, Node850, NEUTRAL_REACTION);
     end
     else begin
-        giq_option(4, 195, 551, Node600, 50);
-        giq_option(4, 195, 552, Node600, 50);
+        giq_option(4, NAME, 551, Node600, NEUTRAL_REACTION);
+        giq_option(4, NAME, 552, Node600, NEUTRAL_REACTION);
     end
-    giq_option(4, 195, 553, Node950, 50);
-    giq_option(4, 195, 554, Node1800, 50);
-    giq_option(4, 195, 555, Node999, 50);
+    giq_option(4, NAME, 553, Node950, NEUTRAL_REACTION);
+    giq_option(4, NAME, 554, Node1800, NEUTRAL_REACTION);
+    giq_option(4, NAME, 555, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node600
 begin
-    gsay_reply(195, 600);
-    if (has_skill(dude_obj, 12) > 90) then begin
-        giq_option(4, 195, 601, Node700, 50);
+    Reply(600);
+    if (dude_skill(SKILL_SCIENCE) > 90) then begin
+        giq_option(4, NAME, 601, Node700, NEUTRAL_REACTION);
     end
     else begin
-        giq_option(4, 195, 601, Node650, 50);
+        giq_option(4, NAME, 601, Node650, NEUTRAL_REACTION);
     end
-    giq_option(4, 195, 602, Node1800, 50);
-    giq_option(4, 195, 603, Node999, 50);
+    giq_option(4, NAME, 602, Node1800, NEUTRAL_REACTION);
+    giq_option(4, NAME, 603, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node650
 begin
-    gsay_reply(195, 650);
-    giq_option(4, 195, 651, Node1800, 50);
-    giq_option(4, 195, 652, Node999, 50);
+    Reply(650);
+    giq_option(4, NAME, 651, Node1800, NEUTRAL_REACTION);
+    giq_option(4, NAME, 652, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node700
 begin
-    gsay_reply(195, 700);
+    Reply(700);
     set_local_var(0, local_var(0) bwor 1);
-    giq_option(4, 195, 701, Node750, 50);
-    giq_option(4, 195, 702, Node850, 50);
-    giq_option(4, 195, 703, Node950, 50);
-    giq_option(4, 195, 704, Node1800, 50);
-    giq_option(4, 195, 705, Node999, 50);
+    giq_option(4, NAME, 701, Node750, NEUTRAL_REACTION);
+    giq_option(4, NAME, 702, Node850, NEUTRAL_REACTION);
+    giq_option(4, NAME, 703, Node950, NEUTRAL_REACTION);
+    giq_option(4, NAME, 704, Node1800, NEUTRAL_REACTION);
+    giq_option(4, NAME, 705, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node750
 begin
-    gsay_reply(195, 750);
-    set_global_var(1025, global_var(1025) bwor 1);
-    giq_option(4, 195, 751, Node800, 50);
-    giq_option(4, 195, 752, Node850, 50);
-    giq_option(4, 195, 753, Node950, 50);
-    giq_option(4, 195, 754, Node1800, 50);
-    giq_option(4, 195, 755, Node999, 50);
+    Reply(750);
+    set_global_var(GVAR_PROGEMA_ABOUT_ALIENS, global_var(GVAR_PROGEMA_ABOUT_ALIENS) bwor 1);
+    giq_option(4, NAME, 751, Node800, NEUTRAL_REACTION);
+    giq_option(4, NAME, 752, Node850, NEUTRAL_REACTION);
+    giq_option(4, NAME, 753, Node950, NEUTRAL_REACTION);
+    giq_option(4, NAME, 754, Node1800, NEUTRAL_REACTION);
+    giq_option(4, NAME, 755, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node800
 begin
-    gsay_reply(195, 800);
-    giq_option(4, 195, 801, Node850, 50);
-    giq_option(4, 195, 802, Node950, 50);
-    giq_option(4, 195, 803, Node1800, 50);
-    giq_option(4, 195, 804, Node999, 50);
+    Reply(800);
+    giq_option(4, NAME, 801, Node850, NEUTRAL_REACTION);
+    giq_option(4, NAME, 802, Node950, NEUTRAL_REACTION);
+    giq_option(4, NAME, 803, Node1800, NEUTRAL_REACTION);
+    giq_option(4, NAME, 804, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node850
 begin
-    gsay_reply(195, 850);
-    giq_option(4, 195, 851, Node900, 50);
-    giq_option(4, 195, 852, Node750, 50);
-    giq_option(4, 195, 853, Node950, 50);
-    giq_option(4, 195, 854, Node1800, 50);
-    giq_option(4, 195, 855, Node999, 50);
+    Reply(850);
+    giq_option(4, NAME, 851, Node900, NEUTRAL_REACTION);
+    giq_option(4, NAME, 852, Node750, NEUTRAL_REACTION);
+    giq_option(4, NAME, 853, Node950, NEUTRAL_REACTION);
+    giq_option(4, NAME, 854, Node1800, NEUTRAL_REACTION);
+    giq_option(4, NAME, 855, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node900
 begin
-    gsay_reply(195, 900);
-    giq_option(4, 195, 901, Node750, 50);
-    giq_option(4, 195, 902, Node950, 50);
-    giq_option(4, 195, 903, Node1800, 50);
-    giq_option(4, 195, 904, Node999, 50);
+    Reply(900);
+    giq_option(4, NAME, 901, Node750, NEUTRAL_REACTION);
+    giq_option(4, NAME, 902, Node950, NEUTRAL_REACTION);
+    giq_option(4, NAME, 903, Node1800, NEUTRAL_REACTION);
+    giq_option(4, NAME, 904, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node950
 begin
-    gsay_reply(195, 950);
-    giq_option(4, 195, 951, Node1000, 50);
-    giq_option(4, 195, 952, Node550, 50);
-    giq_option(4, 195, 953, Node1800, 50);
-    giq_option(4, 195, 954, Node999, 50);
+    Reply(950);
+    giq_option(4, NAME, 951, Node1000, NEUTRAL_REACTION);
+    giq_option(4, NAME, 952, Node550, NEUTRAL_REACTION);
+    giq_option(4, NAME, 953, Node1800, NEUTRAL_REACTION);
+    giq_option(4, NAME, 954, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node1000
 begin
-    gsay_reply(195, 1000);
-    giq_option(4, 195, 1001, Node1050, 50);
-    giq_option(4, 195, 1002, Node550, 50);
-    giq_option(4, 195, 1003, Node1800, 50);
-    giq_option(4, 195, 1004, Node999, 50);
+    Reply(1000);
+    giq_option(4, NAME, 1001, Node1050, NEUTRAL_REACTION);
+    giq_option(4, NAME, 1002, Node550, NEUTRAL_REACTION);
+    giq_option(4, NAME, 1003, Node1800, NEUTRAL_REACTION);
+    giq_option(4, NAME, 1004, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node1050
 begin
-    gsay_reply(195, 1050);
-    giq_option(4, 195, 1001, Node1100, 50);
-    giq_option(4, 195, 1002, Node550, 50);
-    giq_option(4, 195, 1003, Node1800, 50);
-    giq_option(4, 195, 1004, Node999, 50);
+    Reply(1050);
+    giq_option(4, NAME, 1001, Node1100, NEUTRAL_REACTION);
+    giq_option(4, NAME, 1002, Node550, NEUTRAL_REACTION);
+    giq_option(4, NAME, 1003, Node1800, NEUTRAL_REACTION);
+    giq_option(4, NAME, 1004, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node1100
 begin
-    gsay_reply(195, 1100);
-    giq_option(4, 195, 1001, Node1150, 50);
-    giq_option(4, 195, 1002, Node550, 50);
-    giq_option(4, 195, 1003, Node1800, 50);
-    giq_option(4, 195, 1004, Node999, 50);
+    Reply(1100);
+    giq_option(4, NAME, 1001, Node1150, NEUTRAL_REACTION);
+    giq_option(4, NAME, 1002, Node550, NEUTRAL_REACTION);
+    giq_option(4, NAME, 1003, Node1800, NEUTRAL_REACTION);
+    giq_option(4, NAME, 1004, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node1150
 begin
-    gsay_reply(195, 1150);
-    giq_option(4, 195, 1001, Node1200, 50);
-    giq_option(4, 195, 1002, Node550, 50);
-    giq_option(4, 195, 1003, Node1800, 50);
-    giq_option(4, 195, 1004, Node999, 50);
+    Reply(1150);
+    giq_option(4, NAME, 1001, Node1200, NEUTRAL_REACTION);
+    giq_option(4, NAME, 1002, Node550, NEUTRAL_REACTION);
+    giq_option(4, NAME, 1003, Node1800, NEUTRAL_REACTION);
+    giq_option(4, NAME, 1004, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node1200
 begin
-    gsay_reply(195, 1200);
-    giq_option(4, 195, 1001, Node1250, 50);
-    giq_option(4, 195, 1002, Node550, 50);
-    giq_option(4, 195, 1003, Node1800, 50);
-    giq_option(4, 195, 1004, Node999, 50);
+    Reply(1200);
+    giq_option(4, NAME, 1001, Node1250, NEUTRAL_REACTION);
+    giq_option(4, NAME, 1002, Node550, NEUTRAL_REACTION);
+    giq_option(4, NAME, 1003, Node1800, NEUTRAL_REACTION);
+    giq_option(4, NAME, 1004, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node1250
 begin
-    gsay_reply(195, 1250);
-    giq_option(4, 195, 1001, Node1300, 50);
-    giq_option(4, 195, 1002, Node550, 50);
-    giq_option(4, 195, 1003, Node1800, 50);
-    giq_option(4, 195, 1004, Node999, 50);
+    Reply(1250);
+    giq_option(4, NAME, 1001, Node1300, NEUTRAL_REACTION);
+    giq_option(4, NAME, 1002, Node550, NEUTRAL_REACTION);
+    giq_option(4, NAME, 1003, Node1800, NEUTRAL_REACTION);
+    giq_option(4, NAME, 1004, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node1300
 begin
-    gsay_reply(195, 1300);
-    giq_option(4, 195, 1001, Node1350, 50);
-    giq_option(4, 195, 1002, Node550, 50);
-    giq_option(4, 195, 1003, Node1800, 50);
-    giq_option(4, 195, 1004, Node999, 50);
+    Reply(1300);
+    giq_option(4, NAME, 1001, Node1350, NEUTRAL_REACTION);
+    giq_option(4, NAME, 1002, Node550, NEUTRAL_REACTION);
+    giq_option(4, NAME, 1003, Node1800, NEUTRAL_REACTION);
+    giq_option(4, NAME, 1004, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node1350
 begin
-    gsay_reply(195, 1350);
-    giq_option(4, 195, 1001, Node1400, 50);
-    giq_option(4, 195, 1002, Node550, 50);
-    giq_option(4, 195, 1003, Node1800, 50);
-    giq_option(4, 195, 1004, Node999, 50);
+    Reply(1350);
+    giq_option(4, NAME, 1001, Node1400, NEUTRAL_REACTION);
+    giq_option(4, NAME, 1002, Node550, NEUTRAL_REACTION);
+    giq_option(4, NAME, 1003, Node1800, NEUTRAL_REACTION);
+    giq_option(4, NAME, 1004, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node1400
 begin
-    gsay_reply(195, 1400);
-    giq_option(4, 195, 1001, Node1450, 50);
-    giq_option(4, 195, 1002, Node550, 50);
-    giq_option(4, 195, 1003, Node1800, 50);
-    giq_option(4, 195, 1004, Node999, 50);
+    Reply(1400);
+    giq_option(4, NAME, 1001, Node1450, NEUTRAL_REACTION);
+    giq_option(4, NAME, 1002, Node550, NEUTRAL_REACTION);
+    giq_option(4, NAME, 1003, Node1800, NEUTRAL_REACTION);
+    giq_option(4, NAME, 1004, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node1450
 begin
-    gsay_reply(195, 1450);
-    set_global_var(1025, global_var(1025) bwor 2);
-    giq_option(4, 195, 1001, Node1500, 50);
-    giq_option(4, 195, 1002, Node550, 50);
-    giq_option(4, 195, 1003, Node1800, 50);
-    giq_option(4, 195, 1004, Node999, 50);
+    Reply(1450);
+    set_global_var(GVAR_PROGEMA_ABOUT_ALIENS, global_var(GVAR_PROGEMA_ABOUT_ALIENS) bwor 2);
+    giq_option(4, NAME, 1001, Node1500, NEUTRAL_REACTION);
+    giq_option(4, NAME, 1002, Node550, NEUTRAL_REACTION);
+    giq_option(4, NAME, 1003, Node1800, NEUTRAL_REACTION);
+    giq_option(4, NAME, 1004, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node1500
 begin
-    gsay_reply(195, 1500);
-    giq_option(4, 195, 1501, Node550, 50);
-    giq_option(4, 195, 1502, Node1800, 50);
-    giq_option(4, 195, 1503, Node999, 50);
+    Reply(1500);
+    giq_option(4, NAME, 1501, Node550, NEUTRAL_REACTION);
+    giq_option(4, NAME, 1502, Node1800, NEUTRAL_REACTION);
+    giq_option(4, NAME, 1503, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node1550
 begin
-    gsay_reply(195, 1550);
-    if (has_skill(dude_obj, 12) > 100) then begin
-        giq_option(4, 195, 1551, Node1650, 50);
+    Reply(1550);
+    if (dude_skill(SKILL_SCIENCE) > 100) then begin
+        giq_option(4, NAME, 1551, Node1650, NEUTRAL_REACTION);
     end
     else begin
-        giq_option(4, 195, 1551, Node1600, 50);
+        giq_option(4, NAME, 1551, Node1600, NEUTRAL_REACTION);
     end
-    giq_option(4, 195, 1552, Node1800, 50);
-    giq_option(4, 195, 1553, Node999, 50);
+    giq_option(4, NAME, 1552, Node1800, NEUTRAL_REACTION);
+    giq_option(4, NAME, 1553, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node1600
 begin
-    gsay_reply(195, 1600);
-    giq_option(4, 195, 1601, Node1800, 50);
-    giq_option(4, 195, 1602, Node999, 50);
+    Reply(1600);
+    giq_option(4, NAME, 1601, Node1800, NEUTRAL_REACTION);
+    giq_option(4, NAME, 1602, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node1650
 begin
-    gsay_reply(195, 1650 + (get_critter_stat(dude_obj, 34) == 1));
-    giq_option(4, 195, 1652, Node1700, 50);
-    giq_option(4, 195, 1653, Node1800, 50);
-    giq_option(4, 195, 1654, Node999, 50);
+    Reply(1650 + (dude_gender == 1));
+    giq_option(4, NAME, 1652, Node1700, NEUTRAL_REACTION);
+    giq_option(4, NAME, 1653, Node1800, NEUTRAL_REACTION);
+    giq_option(4, NAME, 1654, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node1700
 begin
-    gsay_reply(195, 1700 + (get_critter_stat(dude_obj, 34) == 1));
+    Reply(1700 + (dude_gender == 1));
     set_local_var(0, local_var(0) bwor 2);
     game_time_advance(random(10, 15) * 60 * 10);
-    giq_option(4, 195, 1702, Change_to_robot, 50);
+    giq_option(4, NAME, 1702, Change_to_robot, NEUTRAL_REACTION);
 end
 
 procedure Node1750
 begin
-    gsay_reply(195, 1750);
+    Reply(1750);
     set_local_var(0, local_var(0) bwor 4);
-    giq_option(4, 195, 1751, Node550, 50);
-    giq_option(4, 195, 1752, Node950, 50);
+    giq_option(4, NAME, 1751, Node550, NEUTRAL_REACTION);
+    giq_option(4, NAME, 1752, Node950, NEUTRAL_REACTION);
     if ((local_var(0) bwand 2) == 0) then begin
-        giq_option(4, 195, 1753, Node1550, 50);
+        giq_option(4, NAME, 1753, Node1550, NEUTRAL_REACTION);
     end
     if ((local_var(0) bwand 2) != 0) then begin
-        giq_option(4, 195, 1754, Change_to_robot, 50);
+        giq_option(4, NAME, 1754, Change_to_robot, NEUTRAL_REACTION);
     end
-    giq_option(4, 195, 1755, Node999, 50);
+    giq_option(4, NAME, 1755, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node1800
 begin
     if ((local_var(0) bwand 1) != 0) then begin
-        gsay_reply(195, message_str(195, 1800) + message_str(195, 1801));
+        Reply(mstr(1800) + mstr(1801));
     end
     else begin
-        gsay_reply(195, 1800);
+        Reply(1800);
     end
-    giq_option(4, 195, 1802, Node550, 50);
-    giq_option(4, 195, 1803, Node950, 50);
+    giq_option(4, NAME, 1802, Node550, NEUTRAL_REACTION);
+    giq_option(4, NAME, 1803, Node950, NEUTRAL_REACTION);
     if ((local_var(0) bwand 2) == 0) then begin
-        giq_option(4, 195, 1804, Node1550, 50);
+        giq_option(4, NAME, 1804, Node1550, NEUTRAL_REACTION);
     end
     if ((local_var(0) bwand 2) != 0) then begin
-        giq_option(4, 195, 1805, Change_to_robot, 50);
+        giq_option(4, NAME, 1805, Change_to_robot, NEUTRAL_REACTION);
     end
     if ((local_var(0) bwand 4) == 0) then begin
-        giq_option(4, 195, 1806, Node1750, 50);
+        giq_option(4, NAME, 1806, Node1750, NEUTRAL_REACTION);
     end
-    giq_option(4, 195, 1807, Node999, 50);
+    giq_option(4, NAME, 1807, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node1850
 begin
-    gsay_reply(195, 1850);
-    giq_option(4, 195, 1851, Node999, 50);
-    giq_option(-3, 195, 1852, Node999, 50);
+    Reply(1850);
+    giq_option(4, NAME, 1851, Node999, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 1852, Node999, NEUTRAL_REACTION);
 end
 
 procedure Change_to_robot
 begin
-    gfade_out(1);
+    fade_out;
     game_ui_disable;
     set_map_var(2, obj_art_fid(dude_obj));
-    set_map_var(8, obj_pid(critter_inven_obj(dude_obj, 0)));
-    set_map_var(9, obj_pid(critter_inven_obj(dude_obj, 2)));
-    set_map_var(10, obj_pid(critter_inven_obj(dude_obj, 1)));
-    set_global_var(923, 1);
-    metarule3(107, dude_obj, 16777269, 0);
+    set_map_var(8, obj_pid(critter_inven_obj(dude_obj, INVEN_TYPE_WORN)));
+    set_map_var(9, obj_pid(critter_inven_obj(dude_obj, INVEN_TYPE_LEFT_HAND)));
+    set_map_var(10, obj_pid(critter_inven_obj(dude_obj, INVEN_TYPE_RIGHT_HAND)));
+    set_global_var(GVAR_PROGEMA_DUDE_IS_ROBOT, 1);
+    art_change_fid_num(dude_obj, 16777269);
     anim(dude_obj, 1000, 2);
     move_obj_inven_to_obj(dude_obj, tile_contains_pid_obj(23641, 1, 129));
     if (map_var(3) == 0) then begin
@@ -529,4 +529,3 @@ begin
     move_to(dude_obj, map_var(3), 2);
     add_timer_event(progema1_map, game_ticks(1), 5);
 end
-
