diff --git a/Mods/Resurrection/Original/PDEAD.ssl b/Mods/Resurrection/ReDefine/PDEAD.ssl
index 709ff35..d89b161 100644
--- a/Mods/Resurrection/Original/PDEAD.ssl
+++ b/Mods/Resurrection/ReDefine/PDEAD.ssl
@@ -38,12 +38,12 @@ begin
             return 1;
         end
     end
-    if ((party_member_obj(16777779) != 0) and ((global_var(910) bwand 32) == 0)) then begin
+    if ((party_member_obj(16777779) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 32) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777779)) <= 5) then begin
             return 1;
         end
     end
-    if ((party_member_obj(16777811) != 0) and ((global_var(910) bwand 64) == 0)) then begin
+    if ((party_member_obj(16777811) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 64) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777811)) <= 5) then begin
             return 1;
         end
@@ -58,7 +58,7 @@ begin
             return 1;
         end
     end
-    if ((party_member_obj(16777814) != 0) and ((global_var(910) bwand 128) == 0)) then begin
+    if ((party_member_obj(16777814) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 128) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777814)) <= 5) then begin
             return 1;
         end
@@ -124,7 +124,7 @@ end
 procedure description_p_proc
 begin
     script_overrides;
-    display_msg(message_str(190, 100));
+    display_mstr(100);
 end
 
 procedure map_enter_p_proc
@@ -134,4 +134,3 @@ end
 procedure destroy_p_proc
 begin
 end
-
