diff --git a/Mods/Resurrection/Original/PGENERTR.ssl b/Mods/Resurrection/ReDefine/PGENERTR.ssl
index 2dbd785..8319739 100644
--- a/Mods/Resurrection/Original/PGENERTR.ssl
+++ b/Mods/Resurrection/ReDefine/PGENERTR.ssl
@@ -46,12 +46,12 @@ begin
             return 1;
         end
     end
-    if ((party_member_obj(16777779) != 0) and ((global_var(910) bwand 32) == 0)) then begin
+    if ((party_member_obj(16777779) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 32) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777779)) <= 5) then begin
             return 1;
         end
     end
-    if ((party_member_obj(16777811) != 0) and ((global_var(910) bwand 64) == 0)) then begin
+    if ((party_member_obj(16777811) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 64) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777811)) <= 5) then begin
             return 1;
         end
@@ -66,7 +66,7 @@ begin
             return 1;
         end
     end
-    if ((party_member_obj(16777814) != 0) and ((global_var(910) bwand 128) == 0)) then begin
+    if ((party_member_obj(16777814) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 128) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777814)) <= 5) then begin
             return 1;
         end
@@ -129,30 +129,30 @@ procedure description_p_proc
 begin
     if (local_var(0) == 1) then begin
         script_overrides;
-        display_msg(message_str(196, 104));
+        display_mstr(104);
     end
     if (local_var(0) == 3) then begin
         script_overrides;
         if (local_var(1) == 1) then begin
-            display_msg(message_str(196, 103));
+            display_mstr(103);
         end
         else begin
-            if (has_skill(dude_obj, 13) > 100) then begin
-                display_msg(message_str(196, 101));
+            if (dude_skill(SKILL_REPAIR) > 100) then begin
+                display_mstr(101);
             end
             else begin
-                display_msg(message_str(196, 100));
+                display_mstr(100);
             end
         end
     end
     if (local_var(0) == 2) then begin
         if (local_var(1) == 0) then begin
             script_overrides;
-            if (has_skill(dude_obj, 13) > 100) then begin
-                display_msg(message_str(196, 102));
+            if (dude_skill(SKILL_REPAIR) > 100) then begin
+                display_mstr(102);
             end
             else begin
-                display_msg(message_str(196, 100));
+                display_mstr(100);
             end
         end
     end
@@ -164,7 +164,7 @@ end
 
 procedure map_enter_p_proc
 begin
-    if (metarule(14, 0)) then begin
+    if (map_first_run) then begin
         if (tile_num(self_obj) == 21475) then begin
             set_local_var(0, 1);
         end
@@ -232,13 +232,13 @@ end
 
 procedure Repairing_WRONG_generator
 begin
-    if (has_skill(dude_obj, 13) > 70) then begin
+    if (dude_skill(SKILL_REPAIR) > 70) then begin
         set_local_var(1, 1);
-        critter_dmg(dude_obj, random(1, 4), 0);
-        display_msg(message_str(196, 152 + (get_critter_stat(dude_obj, 34) == 1)));
+        critter_damage(dude_obj, random(1, 4));
+        display_mstr(152 + (dude_gender == 1));
     end
     else begin
-        display_msg(message_str(196, 150 + (get_critter_stat(dude_obj, 34) == 1)));
+        display_mstr(150 + (dude_gender == 1));
     end
 end
 
@@ -248,26 +248,25 @@ begin
     LVar0 := roll_vs_skill(dude_obj, 13, -30);
     if (is_success(LVar0)) then begin
         set_local_var(1, 1);
-        set_global_var(919, 1);
-        display_msg(message_str(196, 156));
-        created_light := create_object_sid(33554573, 20866, 1, 197);
+        set_global_var(GVAR_PROGEMA_STATUS, 1);
+        display_mstr(156);
+        created_light := create_object_sid(33554573, 20866, 1, SCRIPT_PLIGHT1);
         obj_set_light_level(created_light, 0, 8);
         add_timer_event(created_light, 8, 1);
-        created_light := create_object_sid(33554573, 14275, 1, 197);
+        created_light := create_object_sid(33554573, 14275, 1, SCRIPT_PLIGHT1);
         obj_set_light_level(created_light, 0, 8);
         add_timer_event(created_light, 8, 1);
-        created_light := create_object_sid(33554573, 15494, 1, 197);
+        created_light := create_object_sid(33554573, 15494, 1, SCRIPT_PLIGHT1);
         obj_set_light_level(created_light, 0, 8);
         add_timer_event(created_light, 8, 1);
     end
     else begin
         if (is_critical(LVar0)) then begin
-            critter_dmg(dude_obj, random(1, 4), 0);
-            display_msg(message_str(196, 154));
+            critter_damage(dude_obj, random(1, 4));
+            display_mstr(154);
         end
         else begin
-            display_msg(message_str(196, 155));
+            display_mstr(155);
         end
     end
 end
-
