diff --git a/Mods/Resurrection/Original/PROBO3.ssl b/Mods/Resurrection/ReDefine/PROBO3.ssl
index 4acff27..9dc5024 100644
--- a/Mods/Resurrection/Original/PROBO3.ssl
+++ b/Mods/Resurrection/ReDefine/PROBO3.ssl
@@ -45,12 +45,12 @@ begin
             return 1;
         end
     end
-    if ((party_member_obj(16777779) != 0) and ((global_var(910) bwand 32) == 0)) then begin
+    if ((party_member_obj(16777779) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 32) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777779)) <= 5) then begin
             return 1;
         end
     end
-    if ((party_member_obj(16777811) != 0) and ((global_var(910) bwand 64) == 0)) then begin
+    if ((party_member_obj(16777811) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 64) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777811)) <= 5) then begin
             return 1;
         end
@@ -65,7 +65,7 @@ begin
             return 1;
         end
     end
-    if ((party_member_obj(16777814) != 0) and ((global_var(910) bwand 128) == 0)) then begin
+    if ((party_member_obj(16777814) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 128) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777814)) <= 5) then begin
             return 1;
         end
@@ -123,7 +123,7 @@ end
 procedure critter_p_proc
 begin
     if (obj_can_see_obj(self_obj, dude_obj)) then begin
-        attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+        attack(dude_obj);
     end
 end
 
@@ -146,16 +146,16 @@ end
 procedure damage_p_proc
 begin
     if (source_obj == dude_obj) then begin
-        attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+        attack(dude_obj);
     end
 end
 
 procedure map_enter_p_proc
 begin
-    critter_add_trait(self_obj, 1, 6, 270);
-    critter_add_trait(self_obj, 1, 5, 31);
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 270);
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_SECURITY_BOT);
     starttile := tile_num(self_obj);
-    metarule3(100, self_obj, 1, 0);
+    rm_fixed_timer_event(self_obj, 1);
     add_timer_event(self_obj, game_ticks(random(2, 6)), 1);
 end
 
@@ -168,21 +168,21 @@ begin
             animate_move_obj_to_tile(self_obj, tile_num_in_direction(starttile, random(0, 5), random(1, 4)), 0);
             reg_anim_func(3, 0);
         end
-        metarule3(100, self_obj, 1, 0);
+        rm_fixed_timer_event(self_obj, 1);
         add_timer_event(self_obj, game_ticks(random(2, 6)), 1);
     end
     if (fixed_param == 2) then begin
-        critter_add_trait(self_obj, 1, 6, 270);
-        critter_add_trait(self_obj, 1, 5, 31);
-        if (cur_map_index == 100) then begin
-            restock_obj := create_object_sid(23, 0, 0, -1);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 270);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_SECURITY_BOT);
+        if (CUR_MAP_PROGEMA2) then begin
+            restock_obj := create_object(23, 0, 0);
             add_mult_objs_to_inven(self_obj, restock_obj, 1);
             if (obj_carrying_pid_obj(self_obj, 23)) then begin
-                wield_obj_critter(self_obj, obj_carrying_pid_obj(self_obj, 23));
+                wield_obj(obj_carrying_pid_obj(self_obj, 23));
             end
         end
         starttile := tile_num(self_obj);
-        metarule3(100, self_obj, 1, 0);
+        rm_fixed_timer_event(self_obj, 1);
         add_timer_event(self_obj, game_ticks(random(2, 6)), 1);
     end
 end
@@ -190,4 +190,3 @@ end
 procedure use_obj_on_p_proc
 begin
 end
-
