diff --git a/Mods/Resurrection/Original/RH_JLSEV.ssl b/Mods/Resurrection/ReDefine/RH_JLSEV.ssl
index 3dca7d8..fd936bd 100644
--- a/Mods/Resurrection/Original/RH_JLSEV.ssl
+++ b/Mods/Resurrection/ReDefine/RH_JLSEV.ssl
@@ -60,12 +60,12 @@ begin
             return 1;
         end
     end
-    if ((party_member_obj(16777779) != 0) and ((global_var(910) bwand 32) == 0)) then begin
+    if ((party_member_obj(16777779) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 32) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777779)) <= 5) then begin
             return 1;
         end
     end
-    if ((party_member_obj(16777811) != 0) and ((global_var(910) bwand 64) == 0)) then begin
+    if ((party_member_obj(16777811) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 64) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777811)) <= 5) then begin
             return 1;
         end
@@ -80,7 +80,7 @@ begin
             return 1;
         end
     end
-    if ((party_member_obj(16777814) != 0) and ((global_var(910) bwand 128) == 0)) then begin
+    if ((party_member_obj(16777814) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 128) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777814)) <= 5) then begin
             return 1;
         end
@@ -138,42 +138,42 @@ end
 procedure critter_p_proc
 begin
     if (not(combat_is_initialized)) then begin
-        if ((global_var(798) != 1) and (Team_Changed == 0)) then begin
-            critter_add_trait(self_obj, 1, 6, 0);
-            critter_add_trait(self_obj, 1, 5, 22);
+        if ((global_var(GVAR_RH_ALEXA_DEAD) != 1) and (Team_Changed == 0)) then begin
+            critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 0);
+            critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_TOUGH_GUARD);
         end
-        if ((global_var(798) == 1) and (Team_Changed == 0)) then begin
+        if ((global_var(GVAR_RH_ALEXA_DEAD) == 1) and (Team_Changed == 0)) then begin
             Waiting_time := game_time;
-            critter_rm_trait(self_obj, 1, 6, 0);
-            critter_add_trait(self_obj, 1, 6, 235);
-            critter_add_trait(self_obj, 1, 5, 22);
+            critter_rm_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 0);
+            critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 235);
+            critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_TOUGH_GUARD);
             Team_Changed := 1;
         end
     end
-    if ((global_var(805) == 5) or (global_var(801) == 4) and not(combat_is_initialized)) then begin
+    if ((global_var(GVAR_RH_Q_QUEST_MARIE) == 5) or (global_var(GVAR_RH_Q_JULIAN_QUEST3) == 4) and not(combat_is_initialized)) then begin
         if (tile_in_tile_rect(14717, 14691, 18717, 18691, tile_num(self_obj))) then begin
             if (obj_can_see_obj(self_obj, attack_target1_obj)) then begin
-                attack_complex(attack_target1_obj, 0, 1, 0, 0, 30000, 0, 0);
+                attack(attack_target1_obj);
             end
             if (obj_can_see_obj(self_obj, attack_target2_obj)) then begin
-                attack_complex(attack_target2_obj, 0, 1, 0, 0, 30000, 0, 0);
+                attack(attack_target2_obj);
             end
             if (obj_can_see_obj(self_obj, attack_target3_obj)) then begin
-                attack_complex(attack_target3_obj, 0, 1, 0, 0, 30000, 0, 0);
+                attack(attack_target3_obj);
             end
             if (obj_can_see_obj(self_obj, attack_target4_obj)) then begin
-                attack_complex(attack_target4_obj, 0, 1, 0, 0, 30000, 0, 0);
+                attack(attack_target4_obj);
             end
             if (obj_can_see_obj(self_obj, attack_target5_obj)) then begin
-                attack_complex(attack_target5_obj, 0, 1, 0, 0, 30000, 0, 0);
+                attack(attack_target5_obj);
             end
             if (obj_can_see_obj(self_obj, alexa_obj)) then begin
-                attack_complex(alexa_obj, 0, 1, 0, 0, 30000, 0, 0);
+                attack(alexa_obj);
             end
         end
     end
-    if ((global_var(805) < 5) and (obj_on_screen(self_obj) == 1) and (map_var(9) == 0) and (global_var(798) == 1) and (global_var(797) != 1) and (global_var(818) == 0) and not(combat_is_initialized)) then begin
-        float_msg(self_obj, message_str(1524, 300), 2);
+    if ((global_var(GVAR_RH_Q_QUEST_MARIE) < 5) and (obj_on_screen(self_obj) == 1) and (map_var(9) == 0) and (global_var(GVAR_RH_ALEXA_DEAD) == 1) and (global_var(GVAR_RH_JULIAN_DEAD) != 1) and (global_var(GVAR_RH_LEONARD_SEARCH) == 0) and not(combat_is_initialized)) then begin
+        float_msg(self_obj, mstr(300), FLOAT_MSG_RED);
         set_map_var(9, 1);
     end
     if (tile_in_tile_rect(13721, 13679, 20721, 20679, tile_num(self_obj))) then begin
@@ -191,9 +191,9 @@ begin
         Sektor2 := 0;
         Sektor3 := 0;
     end
-    if (not(combat_is_initialized) and (global_var(798) != 1) and (global_var(820) == 1) and (get_critter_stat(dude_obj, 35) > 0)) then begin
+    if (not(combat_is_initialized) and (global_var(GVAR_RH_ALEXA_DEAD) != 1) and (global_var(GVAR_RH_ENEMY_LEONARD) == 1) and (get_critter_stat(dude_obj, STAT_current_hp) > 0)) then begin
         if (Sektor1 == 1) then begin
-            if (get_critter_stat(self_obj, 34) == 0) then begin
+            if (self_gender == 0) then begin
                 animate_move_obj_to_tile(self_obj, 24698, 0);
             end
             else begin
@@ -201,7 +201,7 @@ begin
             end
         end
         if (Sektor2 == 1) then begin
-            if (get_critter_stat(self_obj, 34) == 0) then begin
+            if (self_gender == 0) then begin
                 animate_move_obj_to_tile(self_obj, 18301, 0);
             end
             else begin
@@ -209,7 +209,7 @@ begin
             end
         end
         if (Sektor3 == 1) then begin
-            if (get_critter_stat(self_obj, 34) == 0) then begin
+            if (self_gender == 0) then begin
                 animate_move_obj_to_tile(self_obj, 14105, 0);
             end
             else begin
@@ -217,7 +217,7 @@ begin
             end
         end
     end
-    if (not(combat_is_initialized) and (global_var(798) == 1) and ((global_var(801) == 8) or (global_var(818) == 2) or (global_var(818) == 3)) and (get_critter_stat(dude_obj, 35) > 0)) then begin
+    if (not(combat_is_initialized) and (global_var(GVAR_RH_ALEXA_DEAD) == 1) and ((global_var(GVAR_RH_Q_JULIAN_QUEST3) == 8) or (global_var(GVAR_RH_LEONARD_SEARCH) == 2) or (global_var(GVAR_RH_LEONARD_SEARCH) == 3)) and (get_critter_stat(dude_obj, STAT_current_hp) > 0)) then begin
         if (Sektor3 == 1) then begin
             animate_move_obj_to_tile(self_obj, 21304, 0);
         end
@@ -231,40 +231,40 @@ begin
             destroy_object(self_obj);
         end
     end
-    if ((global_var(819) == 1) and obj_can_see_obj(self_obj, dude_obj) and ((game_time - Waiting_time) > (10 * 1))) then begin
-        attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+    if ((global_var(GVAR_RH_ENEMY_JULIAN) == 1) and obj_can_see_obj(self_obj, dude_obj) and ((game_time - Waiting_time) > (10 * 1))) then begin
+        attack(dude_obj);
     end
 end
 
 procedure pickup_p_proc
 begin
     if (source_obj == dude_obj) then begin
-        set_global_var(819, 1);
+        set_global_var(GVAR_RH_ENEMY_JULIAN, 1);
     end
 end
 
 procedure talk_p_proc
 begin
     if (combat_is_initialized == 0) then begin
-        if ((global_var(797) == 1) and (Projed_Smycku == 0)) then begin
-            float_msg(self_obj, message_str(1524, 360), 2);
+        if ((global_var(GVAR_RH_JULIAN_DEAD) == 1) and (Projed_Smycku == 0)) then begin
+            float_msg(self_obj, mstr(360), FLOAT_MSG_RED);
             Projed_Smycku := 1;
         end
         if (Projed_Smycku == 0) then begin
-            if ((global_var(798) == 1) and (global_var(797) != 1) and (global_var(805) == 5) and (global_var(818) < 2) and (Projed_Smycku == 0)) then begin
-                float_msg(self_obj, message_str(1524, 303), 8);
+            if ((global_var(GVAR_RH_ALEXA_DEAD) == 1) and (global_var(GVAR_RH_JULIAN_DEAD) != 1) and (global_var(GVAR_RH_Q_QUEST_MARIE) == 5) and (global_var(GVAR_RH_LEONARD_SEARCH) < 2) and (Projed_Smycku == 0)) then begin
+                float_msg(self_obj, mstr(303), FLOAT_MSG_YELLOW);
                 Projed_Smycku := 1;
             end
-            if ((global_var(798) == 1) and (global_var(797) != 1) and (global_var(805) < 5) and (global_var(801) == 4) and (global_var(818) >= 1) and (Projed_Smycku == 0)) then begin
-                float_msg(self_obj, message_str(1524, 302), 8);
+            if ((global_var(GVAR_RH_ALEXA_DEAD) == 1) and (global_var(GVAR_RH_JULIAN_DEAD) != 1) and (global_var(GVAR_RH_Q_QUEST_MARIE) < 5) and (global_var(GVAR_RH_Q_JULIAN_QUEST3) == 4) and (global_var(GVAR_RH_LEONARD_SEARCH) >= 1) and (Projed_Smycku == 0)) then begin
+                float_msg(self_obj, mstr(302), FLOAT_MSG_YELLOW);
                 Projed_Smycku := 1;
             end
-            if ((global_var(798) == 1) and (global_var(797) != 1) and (global_var(805) < 5) and (global_var(801) == 4) and (global_var(818) < 1) and (Projed_Smycku == 0)) then begin
-                float_msg(self_obj, message_str(1524, 301), 8);
+            if ((global_var(GVAR_RH_ALEXA_DEAD) == 1) and (global_var(GVAR_RH_JULIAN_DEAD) != 1) and (global_var(GVAR_RH_Q_QUEST_MARIE) < 5) and (global_var(GVAR_RH_Q_JULIAN_QUEST3) == 4) and (global_var(GVAR_RH_LEONARD_SEARCH) < 1) and (Projed_Smycku == 0)) then begin
+                float_msg(self_obj, mstr(301), FLOAT_MSG_YELLOW);
                 Projed_Smycku := 1;
             end
             if (Projed_Smycku == 0) then begin
-                float_msg(self_obj, message_str(1524, random(304, 305)), 8);
+                float_msg(self_obj, mstr(random(304, 305)), FLOAT_MSG_YELLOW);
                 Projed_Smycku := 1;
             end
         end
@@ -274,65 +274,65 @@ end
 
 procedure destroy_p_proc
 begin
-    if (get_critter_stat(self_obj, 34) == 0) then begin
-        set_global_var(821, global_var(821) + 1);
+    if (self_gender == 0) then begin
+        set_global_var(GVAR_RH_KILLS_MAN, global_var(GVAR_RH_KILLS_MAN) + 1);
     end
-    if (get_critter_stat(self_obj, 34) == 1) then begin
-        set_global_var(822, global_var(822) + 1);
+    if (self_gender == 1) then begin
+        set_global_var(GVAR_RH_KILLS_WOMAN, global_var(GVAR_RH_KILLS_WOMAN) + 1);
     end
     if (source_obj == dude_obj) then begin
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(4) > (3 * global_var(5))) or (global_var(2) == 1)) and (global_var(1) == 0)) then begin
-            set_global_var(2, 1);
-            set_global_var(3, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(4) > (3 * global_var(GVAR_GOOD_MONSTER))) or (global_var(GVAR_CHAMPION_REPUTATION) == 1)) and (global_var(GVAR_CHILDKILLER_REPUTATION) == 0)) then begin
+            set_global_var(GVAR_CHAMPION_REPUTATION, 1);
+            set_global_var(GVAR_BERSERKER_REPUTATION, 0);
         end
-        if (metarule(51, self_obj) == 2) then begin
+        if (critter_kill_type(self_obj) == KILL_TYPE_children_kills) then begin
             if (source_obj == dude_obj) then begin
-                set_global_var(2, 0);
-                set_global_var(1, global_var(1) + 1);
-                set_global_var(0, global_var(0) + -15);
+                set_global_var(GVAR_CHAMPION_REPUTATION, 0);
+                set_global_var(GVAR_CHILDKILLER_REPUTATION, global_var(GVAR_CHILDKILLER_REPUTATION) + 1);
+                set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + -15);
                 debug_msg("Player gains " + -15 + " Karma Points.");
-                set_global_var(37, 0);
-                set_global_var(38, 0);
-                set_global_var(39, 0);
-                set_global_var(40, 0);
-                set_global_var(41, 0);
-                set_global_var(42, 0);
-                set_global_var(43, 0);
-                set_global_var(44, 0);
-                set_global_var(45, 0);
-                if (global_var(0) >= 1000) then begin
-                    set_global_var(37, 1);
+                set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+                set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+                set_global_var(GVAR_KARMA_DEFENDER, 0);
+                set_global_var(GVAR_KARMA_WANDERER, 0);
+                set_global_var(GVAR_KARMA_BETRAYER, 0);
+                set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+                set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+                set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+                    set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
                 end
                 else begin
-                    if (global_var(0) >= 750) then begin
-                        set_global_var(38, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
                     end
                     else begin
-                        if (global_var(0) >= 500) then begin
-                            set_global_var(39, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                            set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                         end
                         else begin
-                            if (global_var(0) >= 250) then begin
-                                set_global_var(40, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                                set_global_var(GVAR_KARMA_DEFENDER, 1);
                             end
                             else begin
-                                if (global_var(0) > -250) then begin
-                                    set_global_var(41, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                                    set_global_var(GVAR_KARMA_WANDERER, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -500) then begin
-                                        set_global_var(42, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                        set_global_var(GVAR_KARMA_BETRAYER, 1);
                                     end
                                     else begin
-                                        if (global_var(0) > -750) then begin
-                                            set_global_var(43, 1);
+                                        if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                            set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                         end
                                         else begin
-                                            if (global_var(0) > -1000) then begin
-                                                set_global_var(44, 1);
+                                            if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                                set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                             end
                                             else begin
-                                                set_global_var(45, 1);
+                                                set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                             end
                                         end
                                     end
@@ -341,58 +341,58 @@ begin
                         end
                     end
                 end
-                set_global_var(49, global_var(49) + -8);
+                set_global_var(GVAR_TOWN_REP_RATHOLE, global_var(GVAR_TOWN_REP_RATHOLE) + -8);
                 debug_msg("Added " + -8 + " to Town Rep");
             end
         end
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(5) > (2 * global_var(4))) or (global_var(3) == 1))) then begin
-            set_global_var(3, 1);
-            set_global_var(2, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
+            set_global_var(GVAR_BERSERKER_REPUTATION, 1);
+            set_global_var(GVAR_CHAMPION_REPUTATION, 0);
         end
-        set_global_var(0, global_var(0) + 0);
+        set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + 0);
         debug_msg("Player gains " + 0 + " Karma Points.");
-        set_global_var(37, 0);
-        set_global_var(38, 0);
-        set_global_var(39, 0);
-        set_global_var(40, 0);
-        set_global_var(41, 0);
-        set_global_var(42, 0);
-        set_global_var(43, 0);
-        set_global_var(44, 0);
-        set_global_var(45, 0);
-        if (global_var(0) >= 1000) then begin
-            set_global_var(37, 1);
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+        set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+        set_global_var(GVAR_KARMA_DEFENDER, 0);
+        set_global_var(GVAR_KARMA_WANDERER, 0);
+        set_global_var(GVAR_KARMA_BETRAYER, 0);
+        set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+        set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+            set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
         end
         else begin
-            if (global_var(0) >= 750) then begin
-                set_global_var(38, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
             end
             else begin
-                if (global_var(0) >= 500) then begin
-                    set_global_var(39, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                 end
                 else begin
-                    if (global_var(0) >= 250) then begin
-                        set_global_var(40, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                        set_global_var(GVAR_KARMA_DEFENDER, 1);
                     end
                     else begin
-                        if (global_var(0) > -250) then begin
-                            set_global_var(41, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                            set_global_var(GVAR_KARMA_WANDERER, 1);
                         end
                         else begin
-                            if (global_var(0) > -500) then begin
-                                set_global_var(42, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                set_global_var(GVAR_KARMA_BETRAYER, 1);
                             end
                             else begin
-                                if (global_var(0) > -750) then begin
-                                    set_global_var(43, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -1000) then begin
-                                        set_global_var(44, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                     end
                                     else begin
-                                        set_global_var(45, 1);
+                                        set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                     end
                                 end
                             end
@@ -411,17 +411,17 @@ begin
             restock_obj := obj_carrying_pid_obj(self_obj, 40);
             if (obj_type(self_obj) == 1) then begin
                 restock_obj := obj_carrying_pid_obj(self_obj, 40);
-                if (critter_inven_obj(self_obj, 0) == restock_obj) then begin
-                    if (obj_item_subtype(critter_inven_obj(self_obj, 0)) == 0) then begin
-                        restock_obj := critter_inven_obj(self_obj, 0);
+                if (critter_inven_obj(self_obj, INVEN_TYPE_WORN) == restock_obj) then begin
+                    if (obj_item_subtype(critter_inven_obj(self_obj, INVEN_TYPE_WORN)) == 0) then begin
+                        restock_obj := critter_inven_obj(self_obj, INVEN_TYPE_WORN);
                         debug_msg("armour pid == " + obj_pid(restock_obj));
                         rm_obj_from_inven(self_obj, restock_obj);
                         add_obj_to_inven(self_obj, restock_obj);
                     end
                 end
                 else begin
-                    if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                        metarule(43, self_obj);
+                    if ((critter_inven_obj(self_obj, INVEN_TYPE_LEFT_HAND) == restock_obj) or (critter_inven_obj(self_obj, INVEN_TYPE_RIGHT_HAND) == restock_obj)) then begin
+                        inven_unwield(self_obj);
                     end
                 end
             end
@@ -451,17 +451,17 @@ begin
             restock_obj := obj_carrying_pid_obj(self_obj, 95);
             if (obj_type(self_obj) == 1) then begin
                 restock_obj := obj_carrying_pid_obj(self_obj, 95);
-                if (critter_inven_obj(self_obj, 0) == restock_obj) then begin
-                    if (obj_item_subtype(critter_inven_obj(self_obj, 0)) == 0) then begin
-                        restock_obj := critter_inven_obj(self_obj, 0);
+                if (critter_inven_obj(self_obj, INVEN_TYPE_WORN) == restock_obj) then begin
+                    if (obj_item_subtype(critter_inven_obj(self_obj, INVEN_TYPE_WORN)) == 0) then begin
+                        restock_obj := critter_inven_obj(self_obj, INVEN_TYPE_WORN);
                         debug_msg("armour pid == " + obj_pid(restock_obj));
                         rm_obj_from_inven(self_obj, restock_obj);
                         add_obj_to_inven(self_obj, restock_obj);
                     end
                 end
                 else begin
-                    if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                        metarule(43, self_obj);
+                    if ((critter_inven_obj(self_obj, INVEN_TYPE_LEFT_HAND) == restock_obj) or (critter_inven_obj(self_obj, INVEN_TYPE_RIGHT_HAND) == restock_obj)) then begin
+                        inven_unwield(self_obj);
                     end
                 end
             end
@@ -491,17 +491,17 @@ begin
             restock_obj := obj_carrying_pid_obj(self_obj, 34);
             if (obj_type(self_obj) == 1) then begin
                 restock_obj := obj_carrying_pid_obj(self_obj, 34);
-                if (critter_inven_obj(self_obj, 0) == restock_obj) then begin
-                    if (obj_item_subtype(critter_inven_obj(self_obj, 0)) == 0) then begin
-                        restock_obj := critter_inven_obj(self_obj, 0);
+                if (critter_inven_obj(self_obj, INVEN_TYPE_WORN) == restock_obj) then begin
+                    if (obj_item_subtype(critter_inven_obj(self_obj, INVEN_TYPE_WORN)) == 0) then begin
+                        restock_obj := critter_inven_obj(self_obj, INVEN_TYPE_WORN);
                         debug_msg("armour pid == " + obj_pid(restock_obj));
                         rm_obj_from_inven(self_obj, restock_obj);
                         add_obj_to_inven(self_obj, restock_obj);
                     end
                 end
                 else begin
-                    if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                        metarule(43, self_obj);
+                    if ((critter_inven_obj(self_obj, INVEN_TYPE_LEFT_HAND) == restock_obj) or (critter_inven_obj(self_obj, INVEN_TYPE_RIGHT_HAND) == restock_obj)) then begin
+                        inven_unwield(self_obj);
                     end
                 end
             end
@@ -531,17 +531,17 @@ begin
             restock_obj := obj_carrying_pid_obj(self_obj, 29);
             if (obj_type(self_obj) == 1) then begin
                 restock_obj := obj_carrying_pid_obj(self_obj, 29);
-                if (critter_inven_obj(self_obj, 0) == restock_obj) then begin
-                    if (obj_item_subtype(critter_inven_obj(self_obj, 0)) == 0) then begin
-                        restock_obj := critter_inven_obj(self_obj, 0);
+                if (critter_inven_obj(self_obj, INVEN_TYPE_WORN) == restock_obj) then begin
+                    if (obj_item_subtype(critter_inven_obj(self_obj, INVEN_TYPE_WORN)) == 0) then begin
+                        restock_obj := critter_inven_obj(self_obj, INVEN_TYPE_WORN);
                         debug_msg("armour pid == " + obj_pid(restock_obj));
                         rm_obj_from_inven(self_obj, restock_obj);
                         add_obj_to_inven(self_obj, restock_obj);
                     end
                 end
                 else begin
-                    if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                        metarule(43, self_obj);
+                    if ((critter_inven_obj(self_obj, INVEN_TYPE_LEFT_HAND) == restock_obj) or (critter_inven_obj(self_obj, INVEN_TYPE_RIGHT_HAND) == restock_obj)) then begin
+                        inven_unwield(self_obj);
                     end
                 end
             end
@@ -571,17 +571,17 @@ begin
             restock_obj := obj_carrying_pid_obj(self_obj, 29);
             if (obj_type(self_obj) == 1) then begin
                 restock_obj := obj_carrying_pid_obj(self_obj, 29);
-                if (critter_inven_obj(self_obj, 0) == restock_obj) then begin
-                    if (obj_item_subtype(critter_inven_obj(self_obj, 0)) == 0) then begin
-                        restock_obj := critter_inven_obj(self_obj, 0);
+                if (critter_inven_obj(self_obj, INVEN_TYPE_WORN) == restock_obj) then begin
+                    if (obj_item_subtype(critter_inven_obj(self_obj, INVEN_TYPE_WORN)) == 0) then begin
+                        restock_obj := critter_inven_obj(self_obj, INVEN_TYPE_WORN);
                         debug_msg("armour pid == " + obj_pid(restock_obj));
                         rm_obj_from_inven(self_obj, restock_obj);
                         add_obj_to_inven(self_obj, restock_obj);
                     end
                 end
                 else begin
-                    if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                        metarule(43, self_obj);
+                    if ((critter_inven_obj(self_obj, INVEN_TYPE_LEFT_HAND) == restock_obj) or (critter_inven_obj(self_obj, INVEN_TYPE_RIGHT_HAND) == restock_obj)) then begin
+                        inven_unwield(self_obj);
                     end
                 end
             end
@@ -611,17 +611,17 @@ begin
             restock_obj := obj_carrying_pid_obj(self_obj, 30);
             if (obj_type(self_obj) == 1) then begin
                 restock_obj := obj_carrying_pid_obj(self_obj, 30);
-                if (critter_inven_obj(self_obj, 0) == restock_obj) then begin
-                    if (obj_item_subtype(critter_inven_obj(self_obj, 0)) == 0) then begin
-                        restock_obj := critter_inven_obj(self_obj, 0);
+                if (critter_inven_obj(self_obj, INVEN_TYPE_WORN) == restock_obj) then begin
+                    if (obj_item_subtype(critter_inven_obj(self_obj, INVEN_TYPE_WORN)) == 0) then begin
+                        restock_obj := critter_inven_obj(self_obj, INVEN_TYPE_WORN);
                         debug_msg("armour pid == " + obj_pid(restock_obj));
                         rm_obj_from_inven(self_obj, restock_obj);
                         add_obj_to_inven(self_obj, restock_obj);
                     end
                 end
                 else begin
-                    if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                        metarule(43, self_obj);
+                    if ((critter_inven_obj(self_obj, INVEN_TYPE_LEFT_HAND) == restock_obj) or (critter_inven_obj(self_obj, INVEN_TYPE_RIGHT_HAND) == restock_obj)) then begin
+                        inven_unwield(self_obj);
                     end
                 end
             end
@@ -651,17 +651,17 @@ begin
             restock_obj := obj_carrying_pid_obj(self_obj, 30);
             if (obj_type(self_obj) == 1) then begin
                 restock_obj := obj_carrying_pid_obj(self_obj, 30);
-                if (critter_inven_obj(self_obj, 0) == restock_obj) then begin
-                    if (obj_item_subtype(critter_inven_obj(self_obj, 0)) == 0) then begin
-                        restock_obj := critter_inven_obj(self_obj, 0);
+                if (critter_inven_obj(self_obj, INVEN_TYPE_WORN) == restock_obj) then begin
+                    if (obj_item_subtype(critter_inven_obj(self_obj, INVEN_TYPE_WORN)) == 0) then begin
+                        restock_obj := critter_inven_obj(self_obj, INVEN_TYPE_WORN);
                         debug_msg("armour pid == " + obj_pid(restock_obj));
                         rm_obj_from_inven(self_obj, restock_obj);
                         add_obj_to_inven(self_obj, restock_obj);
                     end
                 end
                 else begin
-                    if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                        metarule(43, self_obj);
+                    if ((critter_inven_obj(self_obj, INVEN_TYPE_LEFT_HAND) == restock_obj) or (critter_inven_obj(self_obj, INVEN_TYPE_RIGHT_HAND) == restock_obj)) then begin
+                        inven_unwield(self_obj);
                     end
                 end
             end
@@ -687,22 +687,22 @@ end
 procedure look_at_p_proc
 begin
     script_overrides;
-    if (get_critter_stat(self_obj, 34) == 0) then begin
-        display_msg(message_str(1524, 100));
+    if (self_gender == 0) then begin
+        display_mstr(100);
     end
-    if (get_critter_stat(self_obj, 34) == 1) then begin
-        display_msg(message_str(1524, 101));
+    if (self_gender == 1) then begin
+        display_mstr(101);
     end
 end
 
 procedure description_p_proc
 begin
     script_overrides;
-    if (get_critter_stat(self_obj, 34) == 0) then begin
-        display_msg(message_str(1524, 200));
+    if (self_gender == 0) then begin
+        display_mstr(200);
     end
-    if (get_critter_stat(self_obj, 34) == 1) then begin
-        display_msg(message_str(1524, 201));
+    if (self_gender == 1) then begin
+        display_mstr(201);
     end
 end
 
@@ -714,22 +714,22 @@ procedure damage_p_proc
 begin
     if (source_obj == dude_obj) then begin
         set_map_var(11, map_var(11) + 1);
-        if ((map_var(11) >= 3) and (global_var(819) != 1)) then begin
-            set_global_var(819, 1);
-            float_msg(self_obj, message_str(1524, 401), 2);
+        if ((map_var(11) >= 3) and (global_var(GVAR_RH_ENEMY_JULIAN) != 1)) then begin
+            set_global_var(GVAR_RH_ENEMY_JULIAN, 1);
+            float_msg(self_obj, mstr(401), FLOAT_MSG_RED);
         end
     end
 end
 
 procedure map_enter_p_proc
 begin
-    if ((global_var(819) != 1) and (global_var(798) != 1)) then begin
-        critter_add_trait(self_obj, 1, 5, 22);
-        critter_add_trait(self_obj, 1, 6, 0);
+    if ((global_var(GVAR_RH_ENEMY_JULIAN) != 1) and (global_var(GVAR_RH_ALEXA_DEAD) != 1)) then begin
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_TOUGH_GUARD);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 0);
     end
     else begin
-        critter_add_trait(self_obj, 1, 5, 22);
-        critter_add_trait(self_obj, 1, 6, 235);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_TOUGH_GUARD);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 235);
     end
 end
 
@@ -740,12 +740,11 @@ end
 
 procedure combat_p_proc
 begin
-    if (global_var(819) == 1) then begin
-        critter_add_trait(self_obj, 1, 6, 235);
+    if (global_var(GVAR_RH_ENEMY_JULIAN) == 1) then begin
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 235);
     end
 end
 
 procedure push_p_proc
 begin
 end
-
