diff --git a/Mods/Resurrection/Original/RH_OBYVD.ssl b/Mods/Resurrection/ReDefine/RH_OBYVD.ssl
index 854a9e3..779183b 100644
--- a/Mods/Resurrection/Original/RH_OBYVD.ssl
+++ b/Mods/Resurrection/ReDefine/RH_OBYVD.ssl
@@ -48,12 +48,12 @@ begin
             return 1;
         end
     end
-    if ((party_member_obj(16777779) != 0) and ((global_var(910) bwand 32) == 0)) then begin
+    if ((party_member_obj(16777779) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 32) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777779)) <= 5) then begin
             return 1;
         end
     end
-    if ((party_member_obj(16777811) != 0) and ((global_var(910) bwand 64) == 0)) then begin
+    if ((party_member_obj(16777811) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 64) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777811)) <= 5) then begin
             return 1;
         end
@@ -68,7 +68,7 @@ begin
             return 1;
         end
     end
-    if ((party_member_obj(16777814) != 0) and ((global_var(910) bwand 128) == 0)) then begin
+    if ((party_member_obj(16777814) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 128) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777814)) <= 5) then begin
             return 1;
         end
@@ -130,13 +130,13 @@ end
 procedure look_at_p_proc
 begin
     script_overrides;
-    display_msg(message_str(1491, 100));
+    display_mstr(100);
 end
 
 procedure description_p_proc
 begin
     script_overrides;
-    display_msg(message_str(1491, 200));
+    display_mstr(200);
 end
 
 procedure use_skill_on_p_proc
@@ -149,7 +149,7 @@ end
 
 procedure damage_p_proc
 begin
-    if (party_member_obj(obj_pid(source_obj)) != 0) then begin
+    if (obj_in_party(source_obj)) then begin
         set_local_var(6, 1);
     end
 end
@@ -158,59 +158,59 @@ procedure destroy_p_proc
 begin
     if (source_obj == dude_obj) then begin
         if (source_obj == dude_obj) then begin
-            set_global_var(5, global_var(5) + 1);
-            if (((global_var(4) + global_var(5)) >= 25) and ((global_var(4) > (3 * global_var(5))) or (global_var(2) == 1)) and (global_var(1) == 0)) then begin
-                set_global_var(2, 1);
-                set_global_var(3, 0);
+            set_global_var(GVAR_GOOD_MONSTER, global_var(GVAR_GOOD_MONSTER) + 1);
+            if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(4) > (3 * global_var(GVAR_GOOD_MONSTER))) or (global_var(GVAR_CHAMPION_REPUTATION) == 1)) and (global_var(GVAR_CHILDKILLER_REPUTATION) == 0)) then begin
+                set_global_var(GVAR_CHAMPION_REPUTATION, 1);
+                set_global_var(GVAR_BERSERKER_REPUTATION, 0);
             end
-            if (metarule(51, self_obj) == 2) then begin
+            if (critter_kill_type(self_obj) == KILL_TYPE_children_kills) then begin
                 if (source_obj == dude_obj) then begin
-                    set_global_var(2, 0);
-                    set_global_var(1, global_var(1) + 1);
-                    set_global_var(0, global_var(0) + -15);
+                    set_global_var(GVAR_CHAMPION_REPUTATION, 0);
+                    set_global_var(GVAR_CHILDKILLER_REPUTATION, global_var(GVAR_CHILDKILLER_REPUTATION) + 1);
+                    set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + -15);
                     debug_msg("Player gains " + -15 + " Karma Points.");
-                    set_global_var(37, 0);
-                    set_global_var(38, 0);
-                    set_global_var(39, 0);
-                    set_global_var(40, 0);
-                    set_global_var(41, 0);
-                    set_global_var(42, 0);
-                    set_global_var(43, 0);
-                    set_global_var(44, 0);
-                    set_global_var(45, 0);
-                    if (global_var(0) >= 1000) then begin
-                        set_global_var(37, 1);
+                    set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+                    set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+                    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+                    set_global_var(GVAR_KARMA_DEFENDER, 0);
+                    set_global_var(GVAR_KARMA_WANDERER, 0);
+                    set_global_var(GVAR_KARMA_BETRAYER, 0);
+                    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+                    set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+                    set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+                        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
                     end
                     else begin
-                        if (global_var(0) >= 750) then begin
-                            set_global_var(38, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                            set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
                         end
                         else begin
-                            if (global_var(0) >= 500) then begin
-                                set_global_var(39, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                                set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                             end
                             else begin
-                                if (global_var(0) >= 250) then begin
-                                    set_global_var(40, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                                    set_global_var(GVAR_KARMA_DEFENDER, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -250) then begin
-                                        set_global_var(41, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                                        set_global_var(GVAR_KARMA_WANDERER, 1);
                                     end
                                     else begin
-                                        if (global_var(0) > -500) then begin
-                                            set_global_var(42, 1);
+                                        if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                            set_global_var(GVAR_KARMA_BETRAYER, 1);
                                         end
                                         else begin
-                                            if (global_var(0) > -750) then begin
-                                                set_global_var(43, 1);
+                                            if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                                set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                             end
                                             else begin
-                                                if (global_var(0) > -1000) then begin
-                                                    set_global_var(44, 1);
+                                                if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                                    set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                                 end
                                                 else begin
-                                                    set_global_var(45, 1);
+                                                    set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                                 end
                                             end
                                         end
@@ -219,58 +219,58 @@ begin
                             end
                         end
                     end
-                    set_global_var(49, global_var(49) + -8);
+                    set_global_var(GVAR_TOWN_REP_RATHOLE, global_var(GVAR_TOWN_REP_RATHOLE) + -8);
                     debug_msg("Added " + -8 + " to Town Rep");
                 end
             end
-            if (((global_var(4) + global_var(5)) >= 25) and ((global_var(5) > (2 * global_var(4))) or (global_var(3) == 1))) then begin
-                set_global_var(3, 1);
-                set_global_var(2, 0);
+            if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
+                set_global_var(GVAR_BERSERKER_REPUTATION, 1);
+                set_global_var(GVAR_CHAMPION_REPUTATION, 0);
             end
-            set_global_var(0, global_var(0) + -10);
+            set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + -10);
             debug_msg("Player gains " + -10 + " Karma Points.");
-            set_global_var(37, 0);
-            set_global_var(38, 0);
-            set_global_var(39, 0);
-            set_global_var(40, 0);
-            set_global_var(41, 0);
-            set_global_var(42, 0);
-            set_global_var(43, 0);
-            set_global_var(44, 0);
-            set_global_var(45, 0);
-            if (global_var(0) >= 1000) then begin
-                set_global_var(37, 1);
+            set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+            set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+            set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+            set_global_var(GVAR_KARMA_DEFENDER, 0);
+            set_global_var(GVAR_KARMA_WANDERER, 0);
+            set_global_var(GVAR_KARMA_BETRAYER, 0);
+            set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+            set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+            set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+                set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
             end
             else begin
-                if (global_var(0) >= 750) then begin
-                    set_global_var(38, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                    set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
                 end
                 else begin
-                    if (global_var(0) >= 500) then begin
-                        set_global_var(39, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                        set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                     end
                     else begin
-                        if (global_var(0) >= 250) then begin
-                            set_global_var(40, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                            set_global_var(GVAR_KARMA_DEFENDER, 1);
                         end
                         else begin
-                            if (global_var(0) > -250) then begin
-                                set_global_var(41, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                                set_global_var(GVAR_KARMA_WANDERER, 1);
                             end
                             else begin
-                                if (global_var(0) > -500) then begin
-                                    set_global_var(42, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                    set_global_var(GVAR_KARMA_BETRAYER, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -750) then begin
-                                        set_global_var(43, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                        set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                     end
                                     else begin
-                                        if (global_var(0) > -1000) then begin
-                                            set_global_var(44, 1);
+                                        if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                            set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                         end
                                         else begin
-                                            set_global_var(45, 1);
+                                            set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                         end
                                     end
                                 end
@@ -279,7 +279,7 @@ begin
                     end
                 end
             end
-            set_global_var(49, global_var(49) + -5);
+            set_global_var(GVAR_TOWN_REP_RATHOLE, global_var(GVAR_TOWN_REP_RATHOLE) + -5);
             debug_msg("Added " + -5 + " to Town Rep");
         end
     end
@@ -287,23 +287,23 @@ end
 
 procedure map_enter_p_proc
 begin
-    if (not(cur_map_index == 8)) then begin
-        critter_add_trait(self_obj, 1, 6, 237);
-        critter_add_trait(self_obj, 1, 5, 15);
+    if (not(CUR_MAP_RH_SEV)) then begin
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 237);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_CHILD);
         if (local_var(7) == 0) then begin
             set_local_var(7, tile_num(self_obj));
         end
         add_timer_event(self_obj, game_ticks(random(6, 24)), 20);
     end
     else begin
-        critter_add_trait(self_obj, 1, 6, 240);
-        critter_add_trait(self_obj, 1, 5, 15);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 240);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_CHILD);
         if (local_var(7) == 0) then begin
             set_local_var(7, tile_num(self_obj));
         end
     end
-    if (not(metarule(22, 0))) then begin
-        if ((global_var(804) == 4) or (global_var(805) == 5) or (global_var(801) == 4)) then begin
+    if (not(is_loading_game)) then begin
+        if ((global_var(GVAR_RH_Q_ALEXA_QUEST3) == 4) or (global_var(GVAR_RH_Q_QUEST_MARIE) == 5) or (global_var(GVAR_RH_Q_JULIAN_QUEST3) == 4)) then begin
             set_local_var(6, 0);
             set_local_var(5, 0);
         end
@@ -316,7 +316,7 @@ end
 
 procedure talk_p_proc
 begin
-    if (cur_map_index == 8) then begin
+    if (CUR_MAP_RH_SEV) then begin
         if (art_anim(obj_art_fid(self_obj)) != 1) then begin
             if (tile_num(self_obj) == local_var(7)) then begin
                 animate_move_obj_to_tile(self_obj, 25724, 1);
@@ -332,7 +332,7 @@ begin
         end
     end
     else begin
-        animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(self_obj), rotation_to_tile(tile_num(dude_obj), tile_num(self_obj)), random(3, 7)), 1);
+        animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(self_obj), rotation_to_tile(dude_tile, tile_num(self_obj)), random(3, 7)), 1);
     end
 end
 
@@ -347,10 +347,10 @@ begin
             if (anim_busy(self_obj) == 0) then begin
                 if ((tile_distance_objs(self_obj, dude_obj) < 8) or obj_can_see_obj(self_obj, dude_obj)) then begin
                     if (anim_busy(self_obj) == 0) then begin
-                        global_temp := rotation_to_tile(tile_num(dude_obj), tile_num(self_obj));
+                        global_temp := rotation_to_tile(dude_tile, tile_num(self_obj));
                         animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(self_obj), global_temp, random(3, 10)), 1);
                         global_temp := (global_temp + 1) % 6;
-                        while ((anim_busy(self_obj) == 0) and (global_temp != rotation_to_tile(tile_num(dude_obj), tile_num(self_obj)))) do begin
+                        while ((anim_busy(self_obj) == 0) and (global_temp != rotation_to_tile(dude_tile, tile_num(self_obj)))) do begin
                             debug_msg("flee loop: rot == " + global_temp);
                             animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(self_obj), global_temp, random(3, 10)), 1);
                             global_temp := (global_temp + 1) % 6;
@@ -360,7 +360,7 @@ begin
             end
         end
         else begin
-            attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+            attack(dude_obj);
         end
     end
     if (not(combat_is_initialized) and (local_var(5) == 0) and (tile_num(self_obj) != local_var(7)) and (tile_num(self_obj) != 25724) and (tile_num(self_obj) != 25721) and (art_anim(obj_art_fid(self_obj)) != 1) and (art_anim(obj_art_fid(self_obj)) != 1)) then begin
@@ -386,4 +386,3 @@ begin
         add_timer_event(self_obj, game_ticks(random(6, 24)), 20);
     end
 end
-
