diff --git a/Mods/Resurrection/Original/RH_STEAV.ssl b/Mods/Resurrection/ReDefine/RH_STEAV.ssl
index 278bbd9..336e789 100644
--- a/Mods/Resurrection/Original/RH_STEAV.ssl
+++ b/Mods/Resurrection/ReDefine/RH_STEAV.ssl
@@ -84,12 +84,12 @@ begin
             return 1;
         end
     end
-    if ((party_member_obj(16777779) != 0) and ((global_var(910) bwand 32) == 0)) then begin
+    if ((party_member_obj(16777779) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 32) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777779)) <= 5) then begin
             return 1;
         end
     end
-    if ((party_member_obj(16777811) != 0) and ((global_var(910) bwand 64) == 0)) then begin
+    if ((party_member_obj(16777811) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 64) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777811)) <= 5) then begin
             return 1;
         end
@@ -104,7 +104,7 @@ begin
             return 1;
         end
     end
-    if ((party_member_obj(16777814) != 0) and ((global_var(910) bwand 128) == 0)) then begin
+    if ((party_member_obj(16777814) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 128) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777814)) <= 5) then begin
             return 1;
         end
@@ -167,10 +167,10 @@ begin
             if (anim_busy(self_obj) == 0) then begin
                 if ((tile_distance_objs(self_obj, dude_obj) < 8) or obj_can_see_obj(self_obj, dude_obj)) then begin
                     if (anim_busy(self_obj) == 0) then begin
-                        global_temp := rotation_to_tile(tile_num(dude_obj), tile_num(self_obj));
+                        global_temp := rotation_to_tile(dude_tile, tile_num(self_obj));
                         animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(self_obj), global_temp, random(3, 10)), 1);
                         global_temp := (global_temp + 1) % 6;
-                        while ((anim_busy(self_obj) == 0) and (global_temp != rotation_to_tile(tile_num(dude_obj), tile_num(self_obj)))) do begin
+                        while ((anim_busy(self_obj) == 0) and (global_temp != rotation_to_tile(dude_tile, tile_num(self_obj)))) do begin
                             debug_msg("flee loop: rot == " + global_temp);
                             animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(self_obj), global_temp, random(3, 10)), 1);
                             global_temp := (global_temp + 1) % 6;
@@ -180,46 +180,46 @@ begin
             end
         end
         else begin
-            attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+            attack(dude_obj);
         end
     end
-    if ((tile_num(dude_obj) == 27867) or (tile_num(dude_obj) == 27667) and (map_var(8) != 1) and not(using_skill(dude_obj, 8))) then begin
+    if ((dude_tile == 27867) or (dude_tile == 27667) and (map_var(8) != 1) and not(using_skill(dude_obj, 8))) then begin
         if (not(combat_is_initialized)) then begin
-            float_msg(self_obj, message_str(1488, 306), 2);
+            float_msg(self_obj, mstr(306), FLOAT_MSG_RED);
         end
         set_local_var(4, 1);
     end
-    if ((local_var(7) != 1) and (map_var(8) != 1) and ((global_var(910) bwand 1) != 0) and (party_member_obj(16777729) != 0) and obj_can_see_obj(self_obj, dude_obj)) then begin
+    if ((local_var(7) != 1) and (map_var(8) != 1) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 1) != 0) and (party_member_obj(16777729) != 0) and obj_can_see_obj(self_obj, dude_obj)) then begin
         if (see_time < game_time) then begin
             if (using_skill(dude_obj, 8) and is_success(roll_vs_skill(dude_obj, 8, -55))) then begin
             end
             else begin
-                float_msg(self_obj, message_str(1488, 352), 2);
+                float_msg(self_obj, mstr(352), FLOAT_MSG_RED);
                 call Node998();
             end
             see_time := game_time + (10 * 1);
         end
     end
-    if (((global_var(910) bwand 131072) != 0) and not((global_var(910) bwand 1) != 0) and (map_var(9) == 0)) then begin
+    if (((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 131072) != 0) and not((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 1) != 0) and (map_var(9) == 0)) then begin
         if ((local_var(7) != 1) and not(combat_is_initialized)) then begin
             if (barking_float == 0) then begin
-                float_msg(self_obj, message_str(1488, 300), 2);
+                float_msg(self_obj, mstr(300), FLOAT_MSG_RED);
             end
             if (barking_float == 1) then begin
-                float_msg(self_obj, message_str(1488, 301), 2);
+                float_msg(self_obj, mstr(301), FLOAT_MSG_RED);
             end
             if (barking_float == 2) then begin
-                float_msg(self_obj, message_str(1488, 302), 2);
+                float_msg(self_obj, mstr(302), FLOAT_MSG_RED);
             end
             if (barking_float == 3) then begin
-                float_msg(self_obj, message_str(1488, 303), 2);
+                float_msg(self_obj, mstr(303), FLOAT_MSG_RED);
             end
         end
         barking_float := barking_float + 1;
         if (barking_float >= 4) then begin
             barking_float := 0;
         end
-        set_global_var(910, global_var(910) bwand (-1 - 131072));
+        set_global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS, global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand (-1 - 131072));
     end
 end
 
@@ -232,21 +232,21 @@ end
 
 procedure talk_p_proc
 begin
-    if ((local_var(5) == 1) and (Projed_Smycku == 0) and ((local_var(7) == 1) or (map_var(8) == 1)) and ((global_var(910) bwand 1) != 0) and (party_member_obj(16777729) != 0)) then begin
-        float_msg(self_obj, message_str(1488, 350), 8);
+    if ((local_var(5) == 1) and (Projed_Smycku == 0) and ((local_var(7) == 1) or (map_var(8) == 1)) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 1) != 0) and (party_member_obj(16777729) != 0)) then begin
+        float_msg(self_obj, mstr(350), FLOAT_MSG_YELLOW);
         Projed_Smycku := 1;
     end
-    if ((local_var(5) == 1) and (Projed_Smycku == 0) and ((local_var(7) == 1) or (map_var(8) == 1)) and ((global_var(910) bwand 1) != 0) and not(party_member_obj(16777729) != 0)) then begin
-        float_msg(self_obj, message_str(1488, 351), 8);
+    if ((local_var(5) == 1) and (Projed_Smycku == 0) and ((local_var(7) == 1) or (map_var(8) == 1)) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 1) != 0) and not(party_member_obj(16777729) != 0)) then begin
+        float_msg(self_obj, mstr(351), FLOAT_MSG_YELLOW);
         Projed_Smycku := 1;
     end
     if ((local_var(5) == 1) and (Projed_Smycku == 0) and (local_var(7) == 1)) then begin
-        float_msg(self_obj, message_str(1488, 353), 8);
+        float_msg(self_obj, mstr(353), FLOAT_MSG_YELLOW);
         Projed_Smycku := 1;
     end
     if ((local_var(5) != 1) and (Projed_Smycku == 0) and (local_var(7) != 1)) then begin
-        anim(dude_obj, 1000, rotation_to_tile(tile_num(dude_obj), tile_num(self_obj)));
-        anim(self_obj, 1000, rotation_to_tile(tile_num(self_obj), tile_num(dude_obj)));
+        anim(dude_obj, 1000, rotation_to_tile(dude_tile, tile_num(self_obj)));
+        anim(self_obj, 1000, rotation_to_tile(tile_num(self_obj), dude_tile));
         start_gdialog(1488, self_obj, 4, -1, -1);
         gsay_start;
         call Node500();
@@ -256,8 +256,8 @@ begin
         Projed_Smycku := 1;
     end
     if ((local_var(5) == 1) and (Projed_Smycku == 0) and (local_var(7) != 1)) then begin
-        anim(dude_obj, 1000, rotation_to_tile(tile_num(dude_obj), tile_num(self_obj)));
-        anim(self_obj, 1000, rotation_to_tile(tile_num(self_obj), tile_num(dude_obj)));
+        anim(dude_obj, 1000, rotation_to_tile(dude_tile, tile_num(self_obj)));
+        anim(self_obj, 1000, rotation_to_tile(tile_num(self_obj), dude_tile));
         start_gdialog(1488, self_obj, 4, -1, -1);
         gsay_start;
         call Node550();
@@ -269,49 +269,49 @@ begin
     if ((map_var(9) == 1) and (Projed_Smycku == 1)) then begin
         gfade_out(600);
         set_map_var(8, 1);
-        set_global_var(0, global_var(0) - 10);
-        set_global_var(37, 0);
-        set_global_var(38, 0);
-        set_global_var(39, 0);
-        set_global_var(40, 0);
-        set_global_var(41, 0);
-        set_global_var(42, 0);
-        set_global_var(43, 0);
-        set_global_var(44, 0);
-        set_global_var(45, 0);
-        if (global_var(0) >= 1000) then begin
-            set_global_var(37, 1);
+        set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) - 10);
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+        set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+        set_global_var(GVAR_KARMA_DEFENDER, 0);
+        set_global_var(GVAR_KARMA_WANDERER, 0);
+        set_global_var(GVAR_KARMA_BETRAYER, 0);
+        set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+        set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+            set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
         end
         else begin
-            if (global_var(0) >= 750) then begin
-                set_global_var(38, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
             end
             else begin
-                if (global_var(0) >= 500) then begin
-                    set_global_var(39, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                 end
                 else begin
-                    if (global_var(0) >= 250) then begin
-                        set_global_var(40, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                        set_global_var(GVAR_KARMA_DEFENDER, 1);
                     end
                     else begin
-                        if (global_var(0) > -250) then begin
-                            set_global_var(41, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                            set_global_var(GVAR_KARMA_WANDERER, 1);
                         end
                         else begin
-                            if (global_var(0) > -500) then begin
-                                set_global_var(42, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                set_global_var(GVAR_KARMA_BETRAYER, 1);
                             end
                             else begin
-                                if (global_var(0) > -750) then begin
-                                    set_global_var(43, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -1000) then begin
-                                        set_global_var(44, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                     end
                                     else begin
-                                        set_global_var(45, 1);
+                                        set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                     end
                                 end
                             end
@@ -328,60 +328,60 @@ end
 procedure destroy_p_proc
 begin
     set_map_var(8, 1);
-    set_global_var(49, global_var(49) - 5);
+    set_global_var(GVAR_TOWN_REP_RATHOLE, global_var(GVAR_TOWN_REP_RATHOLE) - 5);
     if (source_obj == dude_obj) then begin
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(4) > (3 * global_var(5))) or (global_var(2) == 1)) and (global_var(1) == 0)) then begin
-            set_global_var(2, 1);
-            set_global_var(3, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(4) > (3 * global_var(GVAR_GOOD_MONSTER))) or (global_var(GVAR_CHAMPION_REPUTATION) == 1)) and (global_var(GVAR_CHILDKILLER_REPUTATION) == 0)) then begin
+            set_global_var(GVAR_CHAMPION_REPUTATION, 1);
+            set_global_var(GVAR_BERSERKER_REPUTATION, 0);
         end
-        if (metarule(51, self_obj) == 2) then begin
+        if (critter_kill_type(self_obj) == KILL_TYPE_children_kills) then begin
             if (source_obj == dude_obj) then begin
-                set_global_var(2, 0);
-                set_global_var(1, global_var(1) + 1);
-                set_global_var(0, global_var(0) + -15);
+                set_global_var(GVAR_CHAMPION_REPUTATION, 0);
+                set_global_var(GVAR_CHILDKILLER_REPUTATION, global_var(GVAR_CHILDKILLER_REPUTATION) + 1);
+                set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + -15);
                 debug_msg("Player gains " + -15 + " Karma Points.");
-                set_global_var(37, 0);
-                set_global_var(38, 0);
-                set_global_var(39, 0);
-                set_global_var(40, 0);
-                set_global_var(41, 0);
-                set_global_var(42, 0);
-                set_global_var(43, 0);
-                set_global_var(44, 0);
-                set_global_var(45, 0);
-                if (global_var(0) >= 1000) then begin
-                    set_global_var(37, 1);
+                set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+                set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+                set_global_var(GVAR_KARMA_DEFENDER, 0);
+                set_global_var(GVAR_KARMA_WANDERER, 0);
+                set_global_var(GVAR_KARMA_BETRAYER, 0);
+                set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+                set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+                set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+                    set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
                 end
                 else begin
-                    if (global_var(0) >= 750) then begin
-                        set_global_var(38, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
                     end
                     else begin
-                        if (global_var(0) >= 500) then begin
-                            set_global_var(39, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                            set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                         end
                         else begin
-                            if (global_var(0) >= 250) then begin
-                                set_global_var(40, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                                set_global_var(GVAR_KARMA_DEFENDER, 1);
                             end
                             else begin
-                                if (global_var(0) > -250) then begin
-                                    set_global_var(41, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                                    set_global_var(GVAR_KARMA_WANDERER, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -500) then begin
-                                        set_global_var(42, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                        set_global_var(GVAR_KARMA_BETRAYER, 1);
                                     end
                                     else begin
-                                        if (global_var(0) > -750) then begin
-                                            set_global_var(43, 1);
+                                        if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                            set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                         end
                                         else begin
-                                            if (global_var(0) > -1000) then begin
-                                                set_global_var(44, 1);
+                                            if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                                set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                             end
                                             else begin
-                                                set_global_var(45, 1);
+                                                set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                             end
                                         end
                                     end
@@ -390,58 +390,58 @@ begin
                         end
                     end
                 end
-                set_global_var(49, global_var(49) + -8);
+                set_global_var(GVAR_TOWN_REP_RATHOLE, global_var(GVAR_TOWN_REP_RATHOLE) + -8);
                 debug_msg("Added " + -8 + " to Town Rep");
             end
         end
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(5) > (2 * global_var(4))) or (global_var(3) == 1))) then begin
-            set_global_var(3, 1);
-            set_global_var(2, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
+            set_global_var(GVAR_BERSERKER_REPUTATION, 1);
+            set_global_var(GVAR_CHAMPION_REPUTATION, 0);
         end
-        set_global_var(0, global_var(0) + 0);
+        set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + 0);
         debug_msg("Player gains " + 0 + " Karma Points.");
-        set_global_var(37, 0);
-        set_global_var(38, 0);
-        set_global_var(39, 0);
-        set_global_var(40, 0);
-        set_global_var(41, 0);
-        set_global_var(42, 0);
-        set_global_var(43, 0);
-        set_global_var(44, 0);
-        set_global_var(45, 0);
-        if (global_var(0) >= 1000) then begin
-            set_global_var(37, 1);
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+        set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+        set_global_var(GVAR_KARMA_DEFENDER, 0);
+        set_global_var(GVAR_KARMA_WANDERER, 0);
+        set_global_var(GVAR_KARMA_BETRAYER, 0);
+        set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+        set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+            set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
         end
         else begin
-            if (global_var(0) >= 750) then begin
-                set_global_var(38, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
             end
             else begin
-                if (global_var(0) >= 500) then begin
-                    set_global_var(39, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                 end
                 else begin
-                    if (global_var(0) >= 250) then begin
-                        set_global_var(40, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                        set_global_var(GVAR_KARMA_DEFENDER, 1);
                     end
                     else begin
-                        if (global_var(0) > -250) then begin
-                            set_global_var(41, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                            set_global_var(GVAR_KARMA_WANDERER, 1);
                         end
                         else begin
-                            if (global_var(0) > -500) then begin
-                                set_global_var(42, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                set_global_var(GVAR_KARMA_BETRAYER, 1);
                             end
                             else begin
-                                if (global_var(0) > -750) then begin
-                                    set_global_var(43, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -1000) then begin
-                                        set_global_var(44, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                     end
                                     else begin
-                                        set_global_var(45, 1);
+                                        set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                     end
                                 end
                             end
@@ -457,10 +457,10 @@ procedure look_at_p_proc
 begin
     script_overrides;
     if (local_var(6) == 1) then begin
-        display_msg(message_str(1488, 150));
+        display_mstr(150);
     end
     else begin
-        display_msg(message_str(1488, 100));
+        display_mstr(100);
     end
 end
 
@@ -468,10 +468,10 @@ procedure description_p_proc
 begin
     script_overrides;
     if (local_var(6) == 1) then begin
-        display_msg(message_str(1488, 250));
+        display_mstr(250);
     end
     else begin
-        display_msg(message_str(1488, 200));
+        display_mstr(200);
     end
 end
 
@@ -488,10 +488,10 @@ end
 
 procedure map_enter_p_proc
 begin
-    critter_add_trait(self_obj, 1, 6, 238);
-    critter_add_trait(self_obj, 1, 5, 24);
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 238);
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_TOUGH_CITIZEN);
     if (local_var(8) != 1) then begin
-        restock_obj := create_object_sid(563, 0, 0, -1);
+        restock_obj := create_object(563, 0, 0);
         add_mult_objs_to_inven(self_obj, restock_obj, 1);
         set_local_var(8, 1);
     end
@@ -503,19 +503,19 @@ begin
         if (door_counter == 1) then begin
             door_counter := 2;
             if (not(combat_is_initialized)) then begin
-                float_msg(self_obj, message_str(1488, 305), 2);
+                float_msg(self_obj, mstr(305), FLOAT_MSG_RED);
             end
             call Node998();
         end
         if (door_counter == 0) then begin
             door_counter := 1;
             if (not(combat_is_initialized)) then begin
-                float_msg(self_obj, message_str(1488, 304), 2);
+                float_msg(self_obj, mstr(304), FLOAT_MSG_RED);
             end
         end
     end
     if ((fixed_param == 3) and (local_var(7) != 1)) then begin
-        float_msg(self_obj, message_str(1488, 305), 2);
+        float_msg(self_obj, mstr(305), FLOAT_MSG_RED);
         call Node998();
     end
 end
@@ -532,190 +532,190 @@ end
 
 procedure Node500
 begin
-    gsay_reply(1488, 500);
-    if ((has_skill(dude_obj, 14) > 75) and (map_var(9) == 0) and (map_var(5) != 1) and (map_var(10) != 1) and (global_var(819) != 1) and (global_var(803) == 4) and (global_var(800) != 8) and (map_var(12) == 1) and (local_var(10) == 1)) then begin
-        giq_option(8, 1488, 558, Node1750, 50);
+    Reply(500);
+    if ((dude_skill(SKILL_SPEECH) > 75) and (map_var(9) == 0) and (map_var(5) != 1) and (map_var(10) != 1) and (global_var(GVAR_RH_ENEMY_JULIAN) != 1) and (global_var(GVAR_RH_Q_ALEXA_QUEST2) == 4) and (global_var(GVAR_RH_Q_JULIAN_QUEST2) != 8) and (map_var(12) == 1) and (local_var(10) == 1)) then begin
+        giq_option(8, NAME, 558, Node1750, NEUTRAL_REACTION);
     end
-    if (get_critter_stat(dude_obj, 34) == 0) then begin
-        giq_option(4, 1488, 504, Node700, 50);
+    if (dude_gender == 0) then begin
+        giq_option(4, NAME, 504, Node700, NEUTRAL_REACTION);
     end
-    if (get_critter_stat(dude_obj, 34) == 1) then begin
-        giq_option(4, 1488, 505, Node700, 50);
+    if (dude_gender == 1) then begin
+        giq_option(4, NAME, 505, Node700, NEUTRAL_REACTION);
     end
-    giq_option(4, 1488, 502, Node600, 50);
-    giq_option(4, 1488, 503, Node650, 50);
-    if (global_var(0) < 10) then begin
-        giq_option(4, 1488, 506, Node998, 50);
+    giq_option(4, NAME, 502, Node600, NEUTRAL_REACTION);
+    giq_option(4, NAME, 503, Node650, NEUTRAL_REACTION);
+    if (global_var(GVAR_PLAYER_REPUTATION) < 10) then begin
+        giq_option(4, NAME, 506, Node998, NEUTRAL_REACTION);
     end
-    giq_option(4, 1488, 501, Node999, 50);
-    giq_option(-3, 1488, 507, Node750, 50);
-    giq_option(-3, 1488, 508, Node900, 50);
-    giq_option(-3, 1488, 509, Node999, 50);
+    giq_option(4, NAME, 501, Node999, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 507, Node750, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 508, Node900, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 509, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node550
 begin
-    gsay_reply(1488, 550);
-    if ((has_skill(dude_obj, 14) > 75) and (map_var(9) == 0) and (map_var(5) != 1) and (map_var(10) != 1) and (global_var(819) != 1) and (global_var(803) == 4) and (global_var(800) != 8) and (map_var(12) == 1) and (local_var(10) == 1)) then begin
-        giq_option(8, 1488, 558, Node1750, 50);
+    Reply(550);
+    if ((dude_skill(SKILL_SPEECH) > 75) and (map_var(9) == 0) and (map_var(5) != 1) and (map_var(10) != 1) and (global_var(GVAR_RH_ENEMY_JULIAN) != 1) and (global_var(GVAR_RH_Q_ALEXA_QUEST2) == 4) and (global_var(GVAR_RH_Q_JULIAN_QUEST2) != 8) and (map_var(12) == 1) and (local_var(10) == 1)) then begin
+        giq_option(8, NAME, 558, Node1750, NEUTRAL_REACTION);
     end
     if (local_var(6) != 1) then begin
-        giq_option(4, 1488, message_str(1488, 551) + obj_name(dude_obj), Node700, 50);
+        giq_option(4, NAME, mstr(551) + dude_name, Node700, NEUTRAL_REACTION);
     end
-    giq_option(4, 1488, 552, Node600, 50);
-    giq_option(4, 1488, 553, Node650, 50);
-    if (global_var(0) < 10) then begin
-        giq_option(4, 1488, 554, Node998, 50);
+    giq_option(4, NAME, 552, Node600, NEUTRAL_REACTION);
+    giq_option(4, NAME, 553, Node650, NEUTRAL_REACTION);
+    if (global_var(GVAR_PLAYER_REPUTATION) < 10) then begin
+        giq_option(4, NAME, 554, Node998, NEUTRAL_REACTION);
     end
-    giq_option(4, 1488, 555, Node999, 50);
-    giq_option(-3, 1488, 556, Node750, 50);
-    giq_option(-3, 1488, 557, Node999, 50);
+    giq_option(4, NAME, 555, Node999, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 556, Node750, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 557, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node600
 begin
-    gsay_reply(1488, 600);
+    Reply(600);
     if (local_var(9) != 1) then begin
-        giq_option(7, 1488, 603, Node1050, 50);
+        giq_option(7, NAME, 603, Node1050, NEUTRAL_REACTION);
     end
-    giq_option(4, 1488, 601, Node950, 50);
-    giq_option(4, 1488, 602, Node1000, 50);
-    giq_option(4, 1488, 604, Node999, 50);
+    giq_option(4, NAME, 601, Node950, NEUTRAL_REACTION);
+    giq_option(4, NAME, 602, Node1000, NEUTRAL_REACTION);
+    giq_option(4, NAME, 604, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node650
 begin
-    gsay_reply(1488, 650);
-    giq_option(4, 1488, 651, Node600, 50);
-    if (global_var(0) < 10) then begin
-        giq_option(4, 1488, 652, Node998, 50);
+    Reply(650);
+    giq_option(4, NAME, 651, Node600, NEUTRAL_REACTION);
+    if (global_var(GVAR_PLAYER_REPUTATION) < 10) then begin
+        giq_option(4, NAME, 652, Node998, NEUTRAL_REACTION);
     end
-    giq_option(4, 1488, 653, Node999, 50);
+    giq_option(4, NAME, 653, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node700
 begin
-    if (get_critter_stat(dude_obj, 34) == 0) then begin
-        gsay_reply(1488, 700);
+    if (dude_gender == 0) then begin
+        Reply(700);
     end
-    if (get_critter_stat(dude_obj, 34) == 1) then begin
-        gsay_reply(1488, 701);
+    if (dude_gender == 1) then begin
+        Reply(701);
     end
     set_local_var(6, 1);
-    giq_option(4, 1488, 702, Node600, 50);
-    if (global_var(0) < 10) then begin
-        giq_option(4, 1488, 704, Node998, 50);
+    giq_option(4, NAME, 702, Node600, NEUTRAL_REACTION);
+    if (global_var(GVAR_PLAYER_REPUTATION) < 10) then begin
+        giq_option(4, NAME, 704, Node998, NEUTRAL_REACTION);
     end
-    giq_option(4, 1488, 703, Node999, 50);
+    giq_option(4, NAME, 703, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node750
 begin
-    gsay_reply(1488, 750);
-    giq_option(-3, 1488, 751, Node800, 50);
-    giq_option(-3, 1488, 752, Node850, 50);
-    giq_option(-3, 1488, 753, Node999, 50);
+    Reply(750);
+    giq_option(-3, NAME, 751, Node800, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 752, Node850, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 753, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node800
 begin
-    gsay_reply(1488, 800);
-    giq_option(-3, 1488, 801, Node998, 50);
-    giq_option(-3, 1488, 802, Node999, 50);
+    Reply(800);
+    giq_option(-3, NAME, 801, Node998, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 802, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node850
 begin
-    gsay_reply(1488, 850);
-    giq_option(-3, 1488, 851, Node998, 50);
-    giq_option(-3, 1488, 852, Node999, 50);
+    Reply(850);
+    giq_option(-3, NAME, 851, Node998, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 852, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node900
 begin
-    gsay_reply(1488, 900);
-    giq_option(-3, 1488, 901, Node998, 50);
-    giq_option(-3, 1488, 902, Node750, 50);
-    giq_option(-3, 1488, 903, Node999, 50);
+    Reply(900);
+    giq_option(-3, NAME, 901, Node998, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 902, Node750, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 903, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node950
 begin
-    gsay_reply(1488, 950);
-    giq_option(4, 1488, 951, Node1300, 50);
-    giq_option(4, 1488, 952, Node1350, 50);
-    giq_option(4, 1488, 953, Node999, 50);
+    Reply(950);
+    giq_option(4, NAME, 951, Node1300, NEUTRAL_REACTION);
+    giq_option(4, NAME, 952, Node1350, NEUTRAL_REACTION);
+    giq_option(4, NAME, 953, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node1000
 begin
-    gsay_reply(1488, 1000);
+    Reply(1000);
     if (local_var(10) == 0) then begin
         set_local_var(10, 1);
     end
-    giq_option(4, 1488, 1003, Node1450, 50);
-    giq_option(4, 1488, 1001, Node1300, 50);
-    giq_option(4, 1488, 1002, Node1350, 50);
-    giq_option(4, 1488, 1004, Node999, 50);
-    if ((has_skill(dude_obj, 14) > 75) and (map_var(9) == 0) and (map_var(5) != 1) and (map_var(10) != 1) and (global_var(819) != 1) and (global_var(803) == 4) and (global_var(800) != 8) and (map_var(12) == 1) and (local_var(10) == 1)) then begin
-        giq_option(8, 1488, 1005, Node1750, 50);
+    giq_option(4, NAME, 1003, Node1450, NEUTRAL_REACTION);
+    giq_option(4, NAME, 1001, Node1300, NEUTRAL_REACTION);
+    giq_option(4, NAME, 1002, Node1350, NEUTRAL_REACTION);
+    giq_option(4, NAME, 1004, Node999, NEUTRAL_REACTION);
+    if ((dude_skill(SKILL_SPEECH) > 75) and (map_var(9) == 0) and (map_var(5) != 1) and (map_var(10) != 1) and (global_var(GVAR_RH_ENEMY_JULIAN) != 1) and (global_var(GVAR_RH_Q_ALEXA_QUEST2) == 4) and (global_var(GVAR_RH_Q_JULIAN_QUEST2) != 8) and (map_var(12) == 1) and (local_var(10) == 1)) then begin
+        giq_option(8, NAME, 1005, Node1750, NEUTRAL_REACTION);
     end
 end
 
 procedure Node1050
 begin
-    gsay_reply(1488, 1050);
+    Reply(1050);
     set_local_var(9, 1);
-    giq_option(8, 1488, 1051, Node1100, 50);
-    if (get_critter_stat(dude_obj, 34) == 0) then begin
-        giq_option(4, 1488, 1052, Node1600, 50);
+    giq_option(8, NAME, 1051, Node1100, NEUTRAL_REACTION);
+    if (dude_gender == 0) then begin
+        giq_option(4, NAME, 1052, Node1600, NEUTRAL_REACTION);
     end
-    if (get_critter_stat(dude_obj, 34) == 1) then begin
-        giq_option(4, 1488, 1053, Node1600, 50);
+    if (dude_gender == 1) then begin
+        giq_option(4, NAME, 1053, Node1600, NEUTRAL_REACTION);
     end
 end
 
 procedure Node1100
 begin
-    gsay_reply(1488, 1100);
-    if (has_skill(dude_obj, 14) <= 65) then begin
-        giq_option(7, 1488, 1102, Node1150, 50);
+    Reply(1100);
+    if (dude_skill(SKILL_SPEECH) <= 65) then begin
+        giq_option(7, NAME, 1102, Node1150, NEUTRAL_REACTION);
     end
-    if (has_skill(dude_obj, 14) > 65) then begin
-        giq_option(7, 1488, 1102, Node1200, 50);
+    if (dude_skill(SKILL_SPEECH) > 65) then begin
+        giq_option(7, NAME, 1102, Node1200, NEUTRAL_REACTION);
     end
-    giq_option(4, 1488, 1101, Node1300, 50);
-    giq_option(4, 1488, 1103, Node1400, 50);
-    giq_option(4, 1488, 1104, Node999, 50);
+    giq_option(4, NAME, 1101, Node1300, NEUTRAL_REACTION);
+    giq_option(4, NAME, 1103, Node1400, NEUTRAL_REACTION);
+    giq_option(4, NAME, 1104, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node1150
 begin
-    gsay_reply(1488, 1150);
-    giq_option(4, 1488, 1151, Node1300, 50);
-    giq_option(4, 1488, 1152, Node999, 50);
+    Reply(1150);
+    giq_option(4, NAME, 1151, Node1300, NEUTRAL_REACTION);
+    giq_option(4, NAME, 1152, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node1200
 begin
-    gsay_reply(1488, 1200);
-    giq_option(7, 1488, 1201, Node1250, 50);
-    giq_option(4, 1488, 1202, Node1700, 50);
+    Reply(1200);
+    giq_option(7, NAME, 1201, Node1250, NEUTRAL_REACTION);
+    giq_option(4, NAME, 1202, Node1700, NEUTRAL_REACTION);
 end
 
 procedure Node1250
 begin
     if (obj_is_carrying_obj_pid(self_obj, 563) >= 1) then begin
-        gsay_reply(1488, 1250);
+        Reply(1250);
     end
     if (obj_is_carrying_obj_pid(self_obj, 563) < 1) then begin
-        gsay_reply(1488, 1251);
+        Reply(1251);
     end
     set_local_var(7, 1);
     set_map_var(8, 1);
-    set_global_var(49, global_var(49) + 3);
+    set_global_var(GVAR_TOWN_REP_RATHOLE, global_var(GVAR_TOWN_REP_RATHOLE) + 3);
     if (obj_is_carrying_obj_pid(self_obj, 563) >= 1) then begin
-        restock_obj := create_object_sid(563, 0, 0, -1);
+        restock_obj := create_object(563, 0, 0);
         add_mult_objs_to_inven(dude_obj, restock_obj, 1);
         removed_qty := obj_is_carrying_obj_pid(self_obj, 563);
         if (1 < removed_qty) then begin
@@ -725,17 +725,17 @@ begin
             restock_obj := obj_carrying_pid_obj(self_obj, 563);
             if (obj_type(self_obj) == 1) then begin
                 restock_obj := obj_carrying_pid_obj(self_obj, 563);
-                if (critter_inven_obj(self_obj, 0) == restock_obj) then begin
-                    if (obj_item_subtype(critter_inven_obj(self_obj, 0)) == 0) then begin
-                        restock_obj := critter_inven_obj(self_obj, 0);
+                if (critter_inven_obj(self_obj, INVEN_TYPE_WORN) == restock_obj) then begin
+                    if (obj_item_subtype(critter_inven_obj(self_obj, INVEN_TYPE_WORN)) == 0) then begin
+                        restock_obj := critter_inven_obj(self_obj, INVEN_TYPE_WORN);
                         debug_msg("armour pid == " + obj_pid(restock_obj));
                         rm_obj_from_inven(self_obj, restock_obj);
                         add_obj_to_inven(self_obj, restock_obj);
                     end
                 end
                 else begin
-                    if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                        metarule(43, self_obj);
+                    if ((critter_inven_obj(self_obj, INVEN_TYPE_LEFT_HAND) == restock_obj) or (critter_inven_obj(self_obj, INVEN_TYPE_RIGHT_HAND) == restock_obj)) then begin
+                        inven_unwield(self_obj);
                     end
                 end
             end
@@ -756,94 +756,94 @@ begin
         restock_obj := 0;
         restock_amt := 0;
     end
-    giq_option(7, 1488, 1252, Node999, 50);
+    giq_option(7, NAME, 1252, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node1300
 begin
-    if (get_critter_stat(dude_obj, 34) == 0) then begin
-        gsay_reply(1488, 1300);
+    if (dude_gender == 0) then begin
+        Reply(1300);
     end
-    if (get_critter_stat(dude_obj, 34) == 1) then begin
-        gsay_reply(1488, 1301);
+    if (dude_gender == 1) then begin
+        Reply(1301);
     end
     if (local_var(10) == 0) then begin
         set_local_var(10, 1);
     end
-    giq_option(4, 1488, 1302, Node1400, 50);
-    if (global_var(0) < 10) then begin
-        giq_option(4, 1488, 1303, Node998, 50);
+    giq_option(4, NAME, 1302, Node1400, NEUTRAL_REACTION);
+    if (global_var(GVAR_PLAYER_REPUTATION) < 10) then begin
+        giq_option(4, NAME, 1303, Node998, NEUTRAL_REACTION);
     end
 end
 
 procedure Node1350
 begin
-    gsay_reply(1488, 1350);
+    Reply(1350);
     if (local_var(10) == 0) then begin
         set_local_var(10, 1);
     end
-    if (global_var(0) < 10) then begin
-        giq_option(4, 1488, 1352, Node998, 50);
+    if (global_var(GVAR_PLAYER_REPUTATION) < 10) then begin
+        giq_option(4, NAME, 1352, Node998, NEUTRAL_REACTION);
     end
-    giq_option(4, 1488, 1351, Node1400, 50);
-    giq_option(4, 1488, 1353, Node999, 50);
+    giq_option(4, NAME, 1351, Node1400, NEUTRAL_REACTION);
+    giq_option(4, NAME, 1353, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node1400
 begin
-    gsay_reply(1488, 1400);
-    if (get_critter_stat(dude_obj, 34) == 0) then begin
-        giq_option(4, 1488, 1401, Node1450, 50);
+    Reply(1400);
+    if (dude_gender == 0) then begin
+        giq_option(4, NAME, 1401, Node1450, NEUTRAL_REACTION);
     end
-    if (get_critter_stat(dude_obj, 34) == 1) then begin
-        giq_option(4, 1488, 1402, Node1450, 50);
+    if (dude_gender == 1) then begin
+        giq_option(4, NAME, 1402, Node1450, NEUTRAL_REACTION);
     end
-    if (global_var(0) < 10) then begin
-        giq_option(4, 1488, 1403, Node998, 50);
+    if (global_var(GVAR_PLAYER_REPUTATION) < 10) then begin
+        giq_option(4, NAME, 1403, Node998, NEUTRAL_REACTION);
     end
-    giq_option(4, 1488, 1404, Node999, 50);
+    giq_option(4, NAME, 1404, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node1450
 begin
-    gsay_reply(1488, 1450);
+    Reply(1450);
     if (item_caps_total(dude_obj) > 99) then begin
-        giq_option(4, 1488, 1451, Node1500, 50);
+        giq_option(4, NAME, 1451, Node1500, NEUTRAL_REACTION);
     end
     if (item_caps_total(dude_obj) > 74) then begin
-        if ((has_skill(dude_obj, 14) > 50) and (has_skill(dude_obj, 15) > 25)) then begin
-            giq_option(4, 1488, 1452, Node1550_75, 50);
+        if ((dude_skill(SKILL_SPEECH) > 50) and (dude_skill(SKILL_BARTER) > 25)) then begin
+            giq_option(4, NAME, 1452, Node1550_75, NEUTRAL_REACTION);
         end
         else begin
-            giq_option(4, 1488, 1452, Node1650, 50);
+            giq_option(4, NAME, 1452, Node1650, NEUTRAL_REACTION);
         end
     end
     if (item_caps_total(dude_obj) > 49) then begin
-        if ((has_skill(dude_obj, 14) > 80) and (has_skill(dude_obj, 15) > 35)) then begin
-            giq_option(4, 1488, 1453, Node1550_50, 50);
+        if ((dude_skill(SKILL_SPEECH) > 80) and (dude_skill(SKILL_BARTER) > 35)) then begin
+            giq_option(4, NAME, 1453, Node1550_50, NEUTRAL_REACTION);
         end
         else begin
-            giq_option(4, 1488, 1453, Node1650, 50);
+            giq_option(4, NAME, 1453, Node1650, NEUTRAL_REACTION);
         end
     end
-    giq_option(4, 1488, 1454, Node999, 50);
+    giq_option(4, NAME, 1454, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node1500
 begin
     if (obj_is_carrying_obj_pid(self_obj, 563) >= 1) then begin
-        gsay_reply(1488, 1500);
+        Reply(1500);
     end
     if (obj_is_carrying_obj_pid(self_obj, 563) < 1) then begin
-        gsay_reply(1488, 1501);
+        Reply(1501);
     end
     item_caps_adjust(dude_obj, -100);
     item_caps_adjust(self_obj, 100);
     set_local_var(7, 1);
     set_map_var(8, 1);
-    set_global_var(49, global_var(49) + 3);
+    set_global_var(GVAR_TOWN_REP_RATHOLE, global_var(GVAR_TOWN_REP_RATHOLE) + 3);
     if (obj_is_carrying_obj_pid(self_obj, 563) >= 1) then begin
-        restock_obj := create_object_sid(563, 0, 0, -1);
+        restock_obj := create_object(563, 0, 0);
         add_mult_objs_to_inven(dude_obj, restock_obj, 1);
         removed_qty := obj_is_carrying_obj_pid(self_obj, 563);
         if (1 < removed_qty) then begin
@@ -853,17 +853,17 @@ begin
             restock_obj := obj_carrying_pid_obj(self_obj, 563);
             if (obj_type(self_obj) == 1) then begin
                 restock_obj := obj_carrying_pid_obj(self_obj, 563);
-                if (critter_inven_obj(self_obj, 0) == restock_obj) then begin
-                    if (obj_item_subtype(critter_inven_obj(self_obj, 0)) == 0) then begin
-                        restock_obj := critter_inven_obj(self_obj, 0);
+                if (critter_inven_obj(self_obj, INVEN_TYPE_WORN) == restock_obj) then begin
+                    if (obj_item_subtype(critter_inven_obj(self_obj, INVEN_TYPE_WORN)) == 0) then begin
+                        restock_obj := critter_inven_obj(self_obj, INVEN_TYPE_WORN);
                         debug_msg("armour pid == " + obj_pid(restock_obj));
                         rm_obj_from_inven(self_obj, restock_obj);
                         add_obj_to_inven(self_obj, restock_obj);
                     end
                 end
                 else begin
-                    if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                        metarule(43, self_obj);
+                    if ((critter_inven_obj(self_obj, INVEN_TYPE_LEFT_HAND) == restock_obj) or (critter_inven_obj(self_obj, INVEN_TYPE_RIGHT_HAND) == restock_obj)) then begin
+                        inven_unwield(self_obj);
                     end
                 end
             end
@@ -884,22 +884,22 @@ begin
         restock_obj := 0;
         restock_amt := 0;
     end
-    giq_option(4, 1488, 1502, Node999, 50);
+    giq_option(4, NAME, 1502, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node1550
 begin
     if (obj_is_carrying_obj_pid(self_obj, 563) >= 1) then begin
-        gsay_reply(1488, 1550);
+        Reply(1550);
     end
     if (obj_is_carrying_obj_pid(self_obj, 563) < 1) then begin
-        gsay_reply(1488, 1551);
+        Reply(1551);
     end
     set_local_var(7, 1);
     set_map_var(8, 1);
-    set_global_var(49, global_var(49) + 3);
+    set_global_var(GVAR_TOWN_REP_RATHOLE, global_var(GVAR_TOWN_REP_RATHOLE) + 3);
     if (obj_is_carrying_obj_pid(self_obj, 563) >= 1) then begin
-        restock_obj := create_object_sid(563, 0, 0, -1);
+        restock_obj := create_object(563, 0, 0);
         add_mult_objs_to_inven(dude_obj, restock_obj, 1);
         removed_qty := obj_is_carrying_obj_pid(self_obj, 563);
         if (1 < removed_qty) then begin
@@ -909,17 +909,17 @@ begin
             restock_obj := obj_carrying_pid_obj(self_obj, 563);
             if (obj_type(self_obj) == 1) then begin
                 restock_obj := obj_carrying_pid_obj(self_obj, 563);
-                if (critter_inven_obj(self_obj, 0) == restock_obj) then begin
-                    if (obj_item_subtype(critter_inven_obj(self_obj, 0)) == 0) then begin
-                        restock_obj := critter_inven_obj(self_obj, 0);
+                if (critter_inven_obj(self_obj, INVEN_TYPE_WORN) == restock_obj) then begin
+                    if (obj_item_subtype(critter_inven_obj(self_obj, INVEN_TYPE_WORN)) == 0) then begin
+                        restock_obj := critter_inven_obj(self_obj, INVEN_TYPE_WORN);
                         debug_msg("armour pid == " + obj_pid(restock_obj));
                         rm_obj_from_inven(self_obj, restock_obj);
                         add_obj_to_inven(self_obj, restock_obj);
                     end
                 end
                 else begin
-                    if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                        metarule(43, self_obj);
+                    if ((critter_inven_obj(self_obj, INVEN_TYPE_LEFT_HAND) == restock_obj) or (critter_inven_obj(self_obj, INVEN_TYPE_RIGHT_HAND) == restock_obj)) then begin
+                        inven_unwield(self_obj);
                     end
                 end
             end
@@ -940,47 +940,47 @@ begin
         restock_obj := 0;
         restock_amt := 0;
     end
-    giq_option(4, 1488, 1552, Node999, 50);
+    giq_option(4, NAME, 1552, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node1600
 begin
-    gsay_reply(1488, 1600);
-    giq_option(4, 1488, 1601, Node999, 50);
+    Reply(1600);
+    giq_option(4, NAME, 1601, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node1650
 begin
-    gsay_reply(1488, 1650);
-    giq_option(4, 1488, 1651, Node1450, 50);
-    giq_option(4, 1488, 1652, Node999, 50);
+    Reply(1650);
+    giq_option(4, NAME, 1651, Node1450, NEUTRAL_REACTION);
+    giq_option(4, NAME, 1652, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node1700
 begin
-    gsay_reply(1488, 1700);
-    giq_option(4, 1488, 1701, Node999, 50);
+    Reply(1700);
+    giq_option(4, NAME, 1701, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node1750
 begin
-    gsay_reply(1488, 1750);
-    giq_option(4, 1488, 1751, Node1800, 50);
-    giq_option(4, 1488, 1752, Node1850, 50);
+    Reply(1750);
+    giq_option(4, NAME, 1751, Node1800, NEUTRAL_REACTION);
+    giq_option(4, NAME, 1752, Node1850, NEUTRAL_REACTION);
 end
 
 procedure Node1800
 begin
-    gsay_reply(1488, 1800);
+    Reply(1800);
     set_map_var(9, 1);
-    giq_option(4, 1488, 1801, Node999, 50);
+    giq_option(4, NAME, 1801, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node1850
 begin
-    gsay_reply(1488, 1850);
+    Reply(1850);
     set_local_var(10, 2);
-    giq_option(4, 1488, 1851, Node999, 50);
+    giq_option(4, NAME, 1851, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node1550_75
@@ -996,4 +996,3 @@ begin
     item_caps_adjust(self_obj, 50);
     call Node1550();
 end
-
