diff --git a/Mods/Resurrection/Original/RNDSPAM.ssl b/Mods/Resurrection/ReDefine/RNDSPAM.ssl
index fdbea1e..3958085 100644
--- a/Mods/Resurrection/Original/RNDSPAM.ssl
+++ b/Mods/Resurrection/ReDefine/RNDSPAM.ssl
@@ -33,8 +33,8 @@ procedure map_update_p_proc;
 procedure put_critter(variable arg0, variable arg1, variable arg2)
 begin
     variable LVar3 := 0;
-    LVar3 := create_object_sid(arg0, arg1, elevation(dude_obj), arg2);
-    critter_attempt_placement(LVar3, arg1, elevation(dude_obj));
+    LVar3 := create_object_sid(arg0, arg1, dude_elevation, arg2);
+    critter_attempt_placement(LVar3, arg1, dude_elevation);
 end
 
 procedure checkPartyMembersNearDoor
@@ -44,12 +44,12 @@ begin
             return 1;
         end
     end
-    if ((party_member_obj(16777779) != 0) and ((global_var(910) bwand 32) == 0)) then begin
+    if ((party_member_obj(16777779) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 32) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777779)) <= 5) then begin
             return 1;
         end
     end
-    if ((party_member_obj(16777811) != 0) and ((global_var(910) bwand 64) == 0)) then begin
+    if ((party_member_obj(16777811) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 64) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777811)) <= 5) then begin
             return 1;
         end
@@ -64,7 +64,7 @@ begin
             return 1;
         end
     end
-    if ((party_member_obj(16777814) != 0) and ((global_var(910) bwand 128) == 0)) then begin
+    if ((party_member_obj(16777814) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 128) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777814)) <= 5) then begin
             return 1;
         end
@@ -121,11 +121,11 @@ end
 
 procedure map_enter_p_proc
 begin
-    if (metarule(14, 0)) then begin
-        set_global_var(1040, 1);
-        display_msg(message_str(1599, 100));
+    if (map_first_run) then begin
+        set_global_var(GVAR_SPECIAL_ENCOUNTER_SPAMMERS, 1);
+        display_mstr(100);
     end
-    if (elevation(dude_obj) == 0) then begin
+    if (dude_elevation == 0) then begin
         if ((get_month >= 3) and (get_month < 5)) then begin
             if ((game_time_hour >= 600) and (game_time_hour < (600 + 100))) then begin
                 set_light_level(game_time_hour - 600 + 40);
@@ -207,7 +207,7 @@ end
 
 procedure map_update_p_proc
 begin
-    if (elevation(dude_obj) == 0) then begin
+    if (dude_elevation == 0) then begin
         if ((get_month >= 3) and (get_month < 5)) then begin
             if ((game_time_hour >= 600) and (game_time_hour < (600 + 100))) then begin
                 set_light_level(game_time_hour - 600 + 40);
@@ -286,4 +286,3 @@ begin
         end
     end
 end
-
