diff --git a/Mods/Resurrection/Original/RNDSG.ssl b/Mods/Resurrection/ReDefine/RNDSG.ssl
index fc3d623..5142ff8 100644
--- a/Mods/Resurrection/Original/RNDSG.ssl
+++ b/Mods/Resurrection/ReDefine/RNDSG.ssl
@@ -42,12 +42,12 @@ begin
             return 1;
         end
     end
-    if ((party_member_obj(16777779) != 0) and ((global_var(910) bwand 32) == 0)) then begin
+    if ((party_member_obj(16777779) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 32) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777779)) <= 5) then begin
             return 1;
         end
     end
-    if ((party_member_obj(16777811) != 0) and ((global_var(910) bwand 64) == 0)) then begin
+    if ((party_member_obj(16777811) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 64) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777811)) <= 5) then begin
             return 1;
         end
@@ -62,7 +62,7 @@ begin
             return 1;
         end
     end
-    if ((party_member_obj(16777814) != 0) and ((global_var(910) bwand 128) == 0)) then begin
+    if ((party_member_obj(16777814) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 128) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777814)) <= 5) then begin
             return 1;
         end
@@ -123,13 +123,13 @@ end
 
 procedure map_enter_p_proc
 begin
-    if (metarule(14, 0)) then begin
-        set_global_var(1039, 1);
+    if (map_first_run) then begin
+        set_global_var(GVAR_SPECIAL_ENCOUNTER_STARGATE, 1);
         set_map_var(2, 1);
-        display_msg(message_str(1593, 100 + (get_critter_stat(dude_obj, 34) == 1)));
+        display_mstr(100 + (dude_gender == 1));
         add_timer_event(self_obj, game_ticks(1), 1);
     end
-    if (elevation(dude_obj) == 0) then begin
+    if (dude_elevation == 0) then begin
         if ((get_month >= 3) and (get_month < 5)) then begin
             if ((game_time_hour >= 600) and (game_time_hour < (600 + 100))) then begin
                 set_light_level(game_time_hour - 600 + 40);
@@ -211,7 +211,7 @@ end
 
 procedure map_update_p_proc
 begin
-    if (elevation(dude_obj) == 0) then begin
+    if (dude_elevation == 0) then begin
         if ((get_month >= 3) and (get_month < 5)) then begin
             if ((game_time_hour >= 600) and (game_time_hour < (600 + 100))) then begin
                 set_light_level(game_time_hour - 600 + 40);
@@ -294,80 +294,79 @@ end
 procedure timed_event_p_proc
 begin
     if (fixed_param == 1) then begin
-        float_msg(sg_carter, message_str(1593, 102), 8);
+        float_msg(sg_carter, mstr(102), FLOAT_MSG_YELLOW);
         add_timer_event(self_obj, game_ticks(5), 2);
     end
     if (fixed_param == 2) then begin
-        float_msg(sg_carter, message_str(1593, 103), 8);
+        float_msg(sg_carter, mstr(103), FLOAT_MSG_YELLOW);
         add_timer_event(self_obj, game_ticks(7), 3);
     end
     if (fixed_param == 3) then begin
-        float_msg(sg_oneill, message_str(1593, 104), 8);
+        float_msg(sg_oneill, mstr(104), FLOAT_MSG_YELLOW);
         add_timer_event(self_obj, game_ticks(1), 4);
     end
     if (fixed_param == 4) then begin
-        float_msg(sg_carter, message_str(1593, 105), 8);
+        float_msg(sg_carter, mstr(105), FLOAT_MSG_YELLOW);
         add_timer_event(self_obj, game_ticks(6), 5);
     end
     if (fixed_param == 5) then begin
-        float_msg(sg_oneill, message_str(1593, 106), 8);
+        float_msg(sg_oneill, mstr(106), FLOAT_MSG_YELLOW);
         add_timer_event(self_obj, game_ticks(1), 6);
     end
     if (fixed_param == 6) then begin
-        float_msg(sg_carter, message_str(1593, 107), 8);
+        float_msg(sg_carter, mstr(107), FLOAT_MSG_YELLOW);
         add_timer_event(self_obj, game_ticks(3), 7);
     end
     if (fixed_param == 7) then begin
-        float_msg(sg_oneill, message_str(1593, 108), 8);
+        float_msg(sg_oneill, mstr(108), FLOAT_MSG_YELLOW);
         add_timer_event(self_obj, game_ticks(1), 8);
     end
     if (fixed_param == 8) then begin
-        float_msg(sg_carter, message_str(1593, 109), 8);
+        float_msg(sg_carter, mstr(109), FLOAT_MSG_YELLOW);
         add_timer_event(self_obj, game_ticks(2), 9);
     end
     if (fixed_param == 9) then begin
-        float_msg(sg_oneill, message_str(1593, 110), 8);
+        float_msg(sg_oneill, mstr(110), FLOAT_MSG_YELLOW);
         add_timer_event(self_obj, game_ticks(2), 10);
     end
     if (fixed_param == 10) then begin
-        float_msg(sg_drjack, message_str(1593, 111), 8);
+        float_msg(sg_drjack, mstr(111), FLOAT_MSG_YELLOW);
         add_timer_event(self_obj, game_ticks(2), 11);
     end
     if (fixed_param == 11) then begin
-        float_msg(sg_oneill, message_str(1593, 112), 8);
+        float_msg(sg_oneill, mstr(112), FLOAT_MSG_YELLOW);
         add_timer_event(self_obj, game_ticks(2), 12);
     end
     if (fixed_param == 12) then begin
-        float_msg(sg_carter, message_str(1593, 113), 8);
+        float_msg(sg_carter, mstr(113), FLOAT_MSG_YELLOW);
         add_timer_event(self_obj, game_ticks(2), 13);
     end
     if (fixed_param == 13) then begin
-        float_msg(sg_oneill, message_str(1593, 114), 8);
+        float_msg(sg_oneill, mstr(114), FLOAT_MSG_YELLOW);
         add_timer_event(self_obj, game_ticks(1), 14);
     end
     if (fixed_param == 14) then begin
-        float_msg(sg_carter, message_str(1593, 115), 8);
+        float_msg(sg_carter, mstr(115), FLOAT_MSG_YELLOW);
         add_timer_event(self_obj, game_ticks(1), 15);
     end
     if (fixed_param == 15) then begin
-        float_msg(sg_oneill, message_str(1593, 116), 8);
+        float_msg(sg_oneill, mstr(116), FLOAT_MSG_YELLOW);
         add_timer_event(self_obj, game_ticks(2), 16);
     end
     if (fixed_param == 16) then begin
-        float_msg(sg_carter, message_str(1593, 117), 8);
+        float_msg(sg_carter, mstr(117), FLOAT_MSG_YELLOW);
         add_timer_event(self_obj, game_ticks(2), 17);
     end
     if (fixed_param == 17) then begin
-        float_msg(sg_oneill, message_str(1593, 118), 8);
+        float_msg(sg_oneill, mstr(118), FLOAT_MSG_YELLOW);
         add_timer_event(self_obj, game_ticks(3), 18);
     end
     if (fixed_param == 18) then begin
-        float_msg(sg_oneill, 0, 0);
+        float_msg_clear(sg_oneill);
         add_timer_event(self_obj, game_ticks(2), 19);
     end
     if (fixed_param == 19) then begin
-        float_msg(sg_oneill, message_str(1593, 119), 8);
+        float_msg(sg_oneill, mstr(119), FLOAT_MSG_YELLOW);
         set_map_var(2, 0);
     end
 end
-
