diff --git a/Mods/Resurrection/Original/SBARMCBX.ssl b/Mods/Resurrection/ReDefine/SBARMCBX.ssl
index 9a70a54..8a576cb 100644
--- a/Mods/Resurrection/Original/SBARMCBX.ssl
+++ b/Mods/Resurrection/ReDefine/SBARMCBX.ssl
@@ -52,12 +52,12 @@ begin
             return 1;
         end
     end
-    if ((party_member_obj(16777779) != 0) and ((global_var(910) bwand 32) == 0)) then begin
+    if ((party_member_obj(16777779) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 32) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777779)) <= 5) then begin
             return 1;
         end
     end
-    if ((party_member_obj(16777811) != 0) and ((global_var(910) bwand 64) == 0)) then begin
+    if ((party_member_obj(16777811) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 64) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777811)) <= 5) then begin
             return 1;
         end
@@ -72,7 +72,7 @@ begin
             return 1;
         end
     end
-    if ((party_member_obj(16777814) != 0) and ((global_var(910) bwand 128) == 0)) then begin
+    if ((party_member_obj(16777814) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 128) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777814)) <= 5) then begin
             return 1;
         end
@@ -129,7 +129,7 @@ end
 
 procedure map_enter_p_proc
 begin
-    if (metarule(22, 0) == 0) then begin
+    if (is_loading_game == 0) then begin
         if (0) then begin
             obj_close(self_obj);
         end
@@ -145,7 +145,7 @@ end
 
 procedure use_p_proc
 begin
-    if (barmcas and (global_var(770) != 1) and (not(using_skill(source_obj, 8)) or not(is_success(roll_vs_skill(source_obj, 8, -20))))) then begin
+    if (barmcas and (global_var(GVAR_ENEMY_RODRIGUES) != 1) and (not(using_skill(source_obj, 8)) or not(is_success(roll_vs_skill(source_obj, 8, -20))))) then begin
         script_overrides;
         add_timer_event(barmcas, 0, 10);
     end
@@ -155,13 +155,13 @@ procedure damage_p_proc
 begin
     if (obj_is_locked(self_obj)) then begin
         if (target_obj != 0) then begin
-            if (metarule(49, target_obj) == 6) then begin
+            if (weapon_dmg_type(target_obj) == 6) then begin
                 if (1) then begin
-                    display_msg(message_str(1189, 300));
+                    display_msg(message_str(SCRIPT_ZILOCKER, 300));
                     obj_unlock(self_obj);
                 end
                 else begin
-                    display_msg(message_str(1189, 301));
+                    display_msg(message_str(SCRIPT_ZILOCKER, 301));
                 end
             end
         end
@@ -174,7 +174,7 @@ begin
                 attack_setup(0, user_obj);
             end
             else begin
-                float_msg(0, message_str(14, 5400), 8);
+                float_msg(0, g_mstr(5400), FLOAT_MSG_YELLOW);
             end
             set_local_var(0, local_var(0) bwor 1024);
         end
@@ -198,7 +198,7 @@ end
 procedure use_obj_on_p_proc
 begin
     user_obj := source_obj;
-    if (barmcas and (global_var(770) != 1) and (not(using_skill(source_obj, 8)) or not(is_success(roll_vs_skill(source_obj, 8, -20))))) then begin
+    if (barmcas and (global_var(GVAR_ENEMY_RODRIGUES) != 1) and (not(using_skill(source_obj, 8)) or not(is_success(roll_vs_skill(source_obj, 8, -20))))) then begin
         script_overrides;
         add_timer_event(barmcas, 0, 10);
     end
@@ -210,19 +210,19 @@ begin
                     if (is_success(roll_vs_skill(user_obj, 9, -20 + 20))) then begin
                         script_overrides;
                         if (user_obj == dude_obj) then begin
-                            display_msg(message_str(1189, 100));
+                            display_msg(message_str(SCRIPT_ZILOCKER, 100));
                             if (((local_var(0) bwand 512) != 0) == 0) then begin
                                 give_exp_points(25);
-                                display_msg(message_str(14, 100) + 25 + message_str(14, 101));
+                                display_msg(g_mstr(100) + 25 + g_mstr(101));
                             end
                             set_local_var(0, local_var(0) bwor 512);
                         end
                         else begin
                             if (user_obj != 0) then begin
-                                display_msg(obj_name(user_obj) + message_str(1189, 200));
+                                display_msg(obj_name(user_obj) + message_str(SCRIPT_ZILOCKER, 200));
                                 if (((local_var(0) bwand 512) != 0) == 0) then begin
                                     give_exp_points(25);
-                                    display_msg(message_str(14, 100) + 25 + message_str(14, 101));
+                                    display_msg(g_mstr(100) + 25 + g_mstr(101));
                                 end
                                 set_local_var(0, local_var(0) bwor 512);
                             end
@@ -233,11 +233,11 @@ begin
                     else begin
                         script_overrides;
                         if (user_obj == dude_obj) then begin
-                            display_msg(message_str(1189, 101));
+                            display_msg(message_str(SCRIPT_ZILOCKER, 101));
                         end
                         else begin
                             if (user_obj != 0) then begin
-                                display_msg(obj_name(user_obj) + message_str(1189, 201));
+                                display_msg(obj_name(user_obj) + message_str(SCRIPT_ZILOCKER, 201));
                             end
                         end
                         debug_msg("unlock_fail_msg");
@@ -251,19 +251,19 @@ begin
                         if (is_success(roll_vs_skill(user_obj, 9, -20 + 20))) then begin
                             script_overrides;
                             if (user_obj == dude_obj) then begin
-                                display_msg(message_str(1189, 100));
+                                display_msg(message_str(SCRIPT_ZILOCKER, 100));
                                 if (((local_var(0) bwand 512) != 0) == 0) then begin
                                     give_exp_points(25);
-                                    display_msg(message_str(14, 100) + 25 + message_str(14, 101));
+                                    display_msg(g_mstr(100) + 25 + g_mstr(101));
                                 end
                                 set_local_var(0, local_var(0) bwor 512);
                             end
                             else begin
                                 if (user_obj != 0) then begin
-                                    display_msg(obj_name(user_obj) + message_str(1189, 200));
+                                    display_msg(obj_name(user_obj) + message_str(SCRIPT_ZILOCKER, 200));
                                     if (((local_var(0) bwand 512) != 0) == 0) then begin
                                         give_exp_points(25);
-                                        display_msg(message_str(14, 100) + 25 + message_str(14, 101));
+                                        display_msg(g_mstr(100) + 25 + g_mstr(101));
                                     end
                                     set_local_var(0, local_var(0) bwor 512);
                                 end
@@ -274,11 +274,11 @@ begin
                         else begin
                             script_overrides;
                             if (user_obj == dude_obj) then begin
-                                display_msg(message_str(1189, 101));
+                                display_msg(message_str(SCRIPT_ZILOCKER, 101));
                             end
                             else begin
                                 if (user_obj != 0) then begin
-                                    display_msg(obj_name(user_obj) + message_str(1189, 201));
+                                    display_msg(obj_name(user_obj) + message_str(SCRIPT_ZILOCKER, 201));
                                 end
                             end
                             debug_msg("unlock_fail_msg");
@@ -290,19 +290,19 @@ begin
                         script_overrides;
                         if (is_success(do_check(user_obj, 0, 0))) then begin
                             if (user_obj == dude_obj) then begin
-                                display_msg(message_str(1189, 400));
+                                display_msg(message_str(SCRIPT_ZILOCKER, 400));
                                 if (((local_var(0) bwand 512) != 0) == 0) then begin
                                     give_exp_points(25);
-                                    display_msg(message_str(14, 100) + 25 + message_str(14, 101));
+                                    display_msg(g_mstr(100) + 25 + g_mstr(101));
                                 end
                                 set_local_var(0, local_var(0) bwor 512);
                             end
                             else begin
                                 if (user_obj != 0) then begin
-                                    display_msg(obj_name(user_obj) + message_str(1189, 500));
+                                    display_msg(obj_name(user_obj) + message_str(SCRIPT_ZILOCKER, 500));
                                     if (((local_var(0) bwand 512) != 0) == 0) then begin
                                         give_exp_points(25);
-                                        display_msg(message_str(14, 100) + 25 + message_str(14, 101));
+                                        display_msg(g_mstr(100) + 25 + g_mstr(101));
                                     end
                                     set_local_var(0, local_var(0) bwor 512);
                                 end
@@ -312,11 +312,11 @@ begin
                         end
                         else begin
                             if (user_obj == dude_obj) then begin
-                                display_msg(message_str(1189, 401));
+                                display_msg(message_str(SCRIPT_ZILOCKER, 401));
                             end
                             else begin
                                 if (user_obj != 0) then begin
-                                    display_msg(obj_name(user_obj) + message_str(1189, 501));
+                                    display_msg(obj_name(user_obj) + message_str(SCRIPT_ZILOCKER, 501));
                                 end
                             end
                             debug_msg("pry_fail_msg");
@@ -332,19 +332,19 @@ begin
                     if (is_success(roll_vs_skill(user_obj, 9, -20 + 40))) then begin
                         script_overrides;
                         if (user_obj == dude_obj) then begin
-                            display_msg(message_str(1189, 102));
+                            display_msg(message_str(SCRIPT_ZILOCKER, 102));
                             if (((local_var(0) bwand 512) != 0) == 0) then begin
                                 give_exp_points(25);
-                                display_msg(message_str(14, 100) + 25 + message_str(14, 101));
+                                display_msg(g_mstr(100) + 25 + g_mstr(101));
                             end
                             set_local_var(0, local_var(0) bwor 512);
                         end
                         else begin
                             if (user_obj != 0) then begin
-                                display_msg(obj_name(user_obj) + message_str(1189, 202));
+                                display_msg(obj_name(user_obj) + message_str(SCRIPT_ZILOCKER, 202));
                                 if (((local_var(0) bwand 512) != 0) == 0) then begin
                                     give_exp_points(25);
-                                    display_msg(message_str(14, 100) + 25 + message_str(14, 101));
+                                    display_msg(g_mstr(100) + 25 + g_mstr(101));
                                 end
                                 set_local_var(0, local_var(0) bwor 512);
                             end
@@ -355,11 +355,11 @@ begin
                     else begin
                         script_overrides;
                         if (user_obj == dude_obj) then begin
-                            display_msg(message_str(1189, 103));
+                            display_msg(message_str(SCRIPT_ZILOCKER, 103));
                         end
                         else begin
                             if (user_obj != 0) then begin
-                                display_msg(obj_name(user_obj) + message_str(1189, 203));
+                                display_msg(obj_name(user_obj) + message_str(SCRIPT_ZILOCKER, 203));
                             end
                         end
                         debug_msg("lock_fail_msg");
@@ -373,19 +373,19 @@ begin
                         if (is_success(roll_vs_skill(user_obj, 9, -20 + 40))) then begin
                             script_overrides;
                             if (user_obj == dude_obj) then begin
-                                display_msg(message_str(1189, 102));
+                                display_msg(message_str(SCRIPT_ZILOCKER, 102));
                                 if (((local_var(0) bwand 512) != 0) == 0) then begin
                                     give_exp_points(25);
-                                    display_msg(message_str(14, 100) + 25 + message_str(14, 101));
+                                    display_msg(g_mstr(100) + 25 + g_mstr(101));
                                 end
                                 set_local_var(0, local_var(0) bwor 512);
                             end
                             else begin
                                 if (user_obj != 0) then begin
-                                    display_msg(obj_name(user_obj) + message_str(1189, 202));
+                                    display_msg(obj_name(user_obj) + message_str(SCRIPT_ZILOCKER, 202));
                                     if (((local_var(0) bwand 512) != 0) == 0) then begin
                                         give_exp_points(25);
-                                        display_msg(message_str(14, 100) + 25 + message_str(14, 101));
+                                        display_msg(g_mstr(100) + 25 + g_mstr(101));
                                     end
                                     set_local_var(0, local_var(0) bwor 512);
                                 end
@@ -396,11 +396,11 @@ begin
                         else begin
                             script_overrides;
                             if (user_obj == dude_obj) then begin
-                                display_msg(message_str(1189, 103));
+                                display_msg(message_str(SCRIPT_ZILOCKER, 103));
                             end
                             else begin
                                 if (user_obj != 0) then begin
-                                    display_msg(obj_name(user_obj) + message_str(1189, 203));
+                                    display_msg(obj_name(user_obj) + message_str(SCRIPT_ZILOCKER, 203));
                                 end
                             end
                             debug_msg("lock_fail_msg");
@@ -415,7 +415,7 @@ end
 procedure use_skill_on_p_proc
 begin
     user_obj := source_obj;
-    if (barmcas and (action_being_used == 9) and (global_var(770) != 1) and (not(using_skill(source_obj, 8)) or not(is_success(roll_vs_skill(source_obj, 8, -20))))) then begin
+    if (barmcas and (action_being_used == 9) and (global_var(GVAR_ENEMY_RODRIGUES) != 1) and (not(using_skill(source_obj, 8)) or not(is_success(roll_vs_skill(source_obj, 8, -20))))) then begin
         script_overrides;
         add_timer_event(barmcas, 0, 10);
     end
@@ -427,19 +427,19 @@ begin
                     if (is_success(roll_vs_skill(user_obj, 9, -20))) then begin
                         script_overrides;
                         if (user_obj == dude_obj) then begin
-                            display_msg(message_str(1189, 100));
+                            display_msg(message_str(SCRIPT_ZILOCKER, 100));
                             if (((local_var(0) bwand 512) != 0) == 0) then begin
                                 give_exp_points(25);
-                                display_msg(message_str(14, 100) + 25 + message_str(14, 101));
+                                display_msg(g_mstr(100) + 25 + g_mstr(101));
                             end
                             set_local_var(0, local_var(0) bwor 512);
                         end
                         else begin
                             if (user_obj != 0) then begin
-                                display_msg(obj_name(user_obj) + message_str(1189, 200));
+                                display_msg(obj_name(user_obj) + message_str(SCRIPT_ZILOCKER, 200));
                                 if (((local_var(0) bwand 512) != 0) == 0) then begin
                                     give_exp_points(25);
-                                    display_msg(message_str(14, 100) + 25 + message_str(14, 101));
+                                    display_msg(g_mstr(100) + 25 + g_mstr(101));
                                 end
                                 set_local_var(0, local_var(0) bwor 512);
                             end
@@ -450,11 +450,11 @@ begin
                     else begin
                         script_overrides;
                         if (user_obj == dude_obj) then begin
-                            display_msg(message_str(1189, 101));
+                            display_msg(message_str(SCRIPT_ZILOCKER, 101));
                         end
                         else begin
                             if (user_obj != 0) then begin
-                                display_msg(obj_name(user_obj) + message_str(1189, 201));
+                                display_msg(obj_name(user_obj) + message_str(SCRIPT_ZILOCKER, 201));
                             end
                         end
                         debug_msg("unlock_fail_msg");
@@ -466,19 +466,19 @@ begin
                     if (is_success(roll_vs_skill(user_obj, 9, -20))) then begin
                         script_overrides;
                         if (user_obj == dude_obj) then begin
-                            display_msg(message_str(1189, 102));
+                            display_msg(message_str(SCRIPT_ZILOCKER, 102));
                             if (((local_var(0) bwand 512) != 0) == 0) then begin
                                 give_exp_points(25);
-                                display_msg(message_str(14, 100) + 25 + message_str(14, 101));
+                                display_msg(g_mstr(100) + 25 + g_mstr(101));
                             end
                             set_local_var(0, local_var(0) bwor 512);
                         end
                         else begin
                             if (user_obj != 0) then begin
-                                display_msg(obj_name(user_obj) + message_str(1189, 202));
+                                display_msg(obj_name(user_obj) + message_str(SCRIPT_ZILOCKER, 202));
                                 if (((local_var(0) bwand 512) != 0) == 0) then begin
                                     give_exp_points(25);
-                                    display_msg(message_str(14, 100) + 25 + message_str(14, 101));
+                                    display_msg(g_mstr(100) + 25 + g_mstr(101));
                                 end
                                 set_local_var(0, local_var(0) bwor 512);
                             end
@@ -489,11 +489,11 @@ begin
                     else begin
                         script_overrides;
                         if (user_obj == dude_obj) then begin
-                            display_msg(message_str(1189, 103));
+                            display_msg(message_str(SCRIPT_ZILOCKER, 103));
                         end
                         else begin
                             if (user_obj != 0) then begin
-                                display_msg(obj_name(user_obj) + message_str(1189, 203));
+                                display_msg(obj_name(user_obj) + message_str(SCRIPT_ZILOCKER, 203));
                             end
                         end
                         debug_msg("lock_fail_msg");
@@ -515,4 +515,3 @@ end
 procedure timed_event_p_proc
 begin
 end
-
