diff --git a/Mods/Resurrection/Original/SEDCENT.ssl b/Mods/Resurrection/ReDefine/SEDCENT.ssl
index 6c67d18..e5ff5c6 100644
--- a/Mods/Resurrection/Original/SEDCENT.ssl
+++ b/Mods/Resurrection/ReDefine/SEDCENT.ssl
@@ -62,12 +62,12 @@ begin
             return 1;
         end
     end
-    if ((party_member_obj(16777779) != 0) and ((global_var(910) bwand 32) == 0)) then begin
+    if ((party_member_obj(16777779) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 32) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777779)) <= 5) then begin
             return 1;
         end
     end
-    if ((party_member_obj(16777811) != 0) and ((global_var(910) bwand 64) == 0)) then begin
+    if ((party_member_obj(16777811) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 64) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777811)) <= 5) then begin
             return 1;
         end
@@ -82,7 +82,7 @@ begin
             return 1;
         end
     end
-    if ((party_member_obj(16777814) != 0) and ((global_var(910) bwand 128) == 0)) then begin
+    if ((party_member_obj(16777814) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 128) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777814)) <= 5) then begin
             return 1;
         end
@@ -220,17 +220,17 @@ begin
             end
         end
     end
-    if (not(metarule(22, 0)) and (global_var(776) != 0)) then begin
-        if (((3 * (30 * (24 * (60 * (60 * 10))))) + global_var(776)) <= game_time) then begin
-            set_global_var(776, 0);
+    if (not(is_loading_game) and (global_var(GVAR_SEDIT_GHOULS_KILLER) != 0)) then begin
+        if (((3 * (30 * (24 * (60 * (60 * 10))))) + global_var(GVAR_SEDIT_GHOULS_KILLER)) <= game_time) then begin
+            set_global_var(GVAR_SEDIT_GHOULS_KILLER, 0);
         end
         else begin
-            set_global_var(769, 1);
+            set_global_var(GVAR_ENEMY_SEDIT, 1);
         end
     end
     police := tile_contains_pid_obj(14673, 0, 16777754);
-    if (not(metarule(22, 0))) then begin
-        if (global_var(749) == 1) then begin
+    if (not(is_loading_game)) then begin
+        if (global_var(GVAR_DUDE_NICOL_MEETING) == 1) then begin
             set_obj_visibility(police, 1);
         end
         else begin
@@ -323,7 +323,6 @@ procedure timed_event_p_proc
 begin
     if (fixed_param == 1) then begin
         game_ui_enable;
-        gfade_in(1);
+        fade_in;
     end
 end
-
