diff --git a/Mods/Resurrection/Original/SEDTUNL.ssl b/Mods/Resurrection/ReDefine/SEDTUNL.ssl
index 0b6daf1..f0d3d39 100644
--- a/Mods/Resurrection/Original/SEDTUNL.ssl
+++ b/Mods/Resurrection/ReDefine/SEDTUNL.ssl
@@ -39,12 +39,12 @@ begin
             return 1;
         end
     end
-    if ((party_member_obj(16777779) != 0) and ((global_var(910) bwand 32) == 0)) then begin
+    if ((party_member_obj(16777779) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 32) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777779)) <= 5) then begin
             return 1;
         end
     end
-    if ((party_member_obj(16777811) != 0) and ((global_var(910) bwand 64) == 0)) then begin
+    if ((party_member_obj(16777811) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 64) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777811)) <= 5) then begin
             return 1;
         end
@@ -59,7 +59,7 @@ begin
             return 1;
         end
     end
-    if ((party_member_obj(16777814) != 0) and ((global_var(910) bwand 128) == 0)) then begin
+    if ((party_member_obj(16777814) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 128) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777814)) <= 5) then begin
             return 1;
         end
@@ -120,7 +120,7 @@ end
 
 procedure map_enter_p_proc
 begin
-    if (elevation(dude_obj) == 1) then begin
+    if (dude_elevation == 1) then begin
         set_light_level(50);
     end
     else begin
@@ -201,27 +201,27 @@ begin
             end
         end
     end
-    if (metarule(14, 0)) then begin
-        set_global_var(1060, 1);
+    if (map_first_run) then begin
+        set_global_var(GVAR_SEDIT_CAVE_KNOW, 1);
     end
-    if (not(metarule(22, 0)) and (elevation(dude_obj) == 0) and (global_var(776) != 0)) then begin
-        if (((3 * (30 * (24 * (60 * (60 * 10))))) + global_var(776)) <= game_time) then begin
-            set_global_var(776, 0);
+    if (not(is_loading_game) and (dude_elevation == 0) and (global_var(GVAR_SEDIT_GHOULS_KILLER) != 0)) then begin
+        if (((3 * (30 * (24 * (60 * (60 * 10))))) + global_var(GVAR_SEDIT_GHOULS_KILLER)) <= game_time) then begin
+            set_global_var(GVAR_SEDIT_GHOULS_KILLER, 0);
         end
         else begin
-            set_global_var(769, 1);
+            set_global_var(GVAR_ENEMY_SEDIT, 1);
         end
     end
-    if ((global_var(795) == 1) and ((global_var(743) == 3) or (global_var(736) == 3)) and (map_var(0) != 1)) then begin
+    if ((global_var(GVAR_S_MARVIN_LIE) == 1) and ((global_var(GVAR_QUEST_S_FREDERYK_MEETING) == 3) or (global_var(GVAR_QUEST_S_NICOL_MARVIN) == 3)) and (map_var(0) != 1)) then begin
         set_map_var(0, 1);
-        Dead_marvin := create_object_sid(16777769, 18305, 1, -1);
+        Dead_marvin := create_object(16777769, 18305, 1);
         kill_critter(Dead_marvin, 35);
     end
 end
 
 procedure map_update_p_proc
 begin
-    if (elevation(dude_obj) == 1) then begin
+    if (dude_elevation == 1) then begin
         set_light_level(50);
     end
     else begin
@@ -303,4 +303,3 @@ begin
         end
     end
 end
-
