diff --git a/Mods/Resurrection/Original/SEDZTR.ssl b/Mods/Resurrection/ReDefine/SEDZTR.ssl
index 8b4a4b2..b08d03f 100644
--- a/Mods/Resurrection/Original/SEDZTR.ssl
+++ b/Mods/Resurrection/ReDefine/SEDZTR.ssl
@@ -59,12 +59,12 @@ begin
             return 1;
         end
     end
-    if ((party_member_obj(16777779) != 0) and ((global_var(910) bwand 32) == 0)) then begin
+    if ((party_member_obj(16777779) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 32) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777779)) <= 5) then begin
             return 1;
         end
     end
-    if ((party_member_obj(16777811) != 0) and ((global_var(910) bwand 64) == 0)) then begin
+    if ((party_member_obj(16777811) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 64) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777811)) <= 5) then begin
             return 1;
         end
@@ -79,7 +79,7 @@ begin
             return 1;
         end
     end
-    if ((party_member_obj(16777814) != 0) and ((global_var(910) bwand 128) == 0)) then begin
+    if ((party_member_obj(16777814) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 128) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777814)) <= 5) then begin
             return 1;
         end
@@ -141,11 +141,11 @@ end
 
 procedure map_enter_p_proc
 begin
-    if (elevation(dude_obj) == 1) then begin
+    if (dude_elevation == 1) then begin
         set_light_level(60);
     end
     else begin
-        if (elevation(dude_obj) == 2) then begin
+        if (dude_elevation == 2) then begin
             set_light_level(50);
         end
         else begin
@@ -227,23 +227,23 @@ begin
             end
         end
     end
-    if ((global_var(712) == 1) and (global_var(793) == 3)) then begin
-        set_global_var(712, 2);
+    if ((global_var(GVAR_QUEST_S_RODRIGUES_GHOULS) == 1) and (global_var(GVAR_SEDIT_DEAD_GHOULS) == 3)) then begin
+        set_global_var(GVAR_QUEST_S_RODRIGUES_GHOULS, 2);
     end
-    if (global_var(1016) == 0) then begin
+    if (global_var(GVAR_S_RAIDERS_KILLED) == 0) then begin
         lRaidersAlive := 0;
-        metarule3(100, self_obj, 1, 0);
+        rm_fixed_timer_event(self_obj, 1);
         add_timer_event(self_obj, 30, 1);
     end
 end
 
 procedure map_update_p_proc
 begin
-    if (elevation(dude_obj) == 1) then begin
+    if (dude_elevation == 1) then begin
         set_light_level(60);
     end
     else begin
-        if (elevation(dude_obj) == 2) then begin
+        if (dude_elevation == 2) then begin
             set_light_level(50);
         end
         else begin
@@ -325,11 +325,11 @@ begin
             end
         end
     end
-    if ((global_var(712) == 1) and (global_var(793) == 3)) then begin
-        set_global_var(712, 2);
+    if ((global_var(GVAR_QUEST_S_RODRIGUES_GHOULS) == 1) and (global_var(GVAR_SEDIT_DEAD_GHOULS) == 3)) then begin
+        set_global_var(GVAR_QUEST_S_RODRIGUES_GHOULS, 2);
     end
-    if ((map_var(7) == 1) or (critter_state(sanchez) == 1) and ((map_var(8) == 1) or (rodrigues == -1) or (critter_state(rodrigues) == 1)) and (global_var(792) != 2)) then begin
-        set_global_var(792, 2);
+    if ((map_var(7) == 1) or (critter_state(sanchez) == 1) and ((map_var(8) == 1) or (rodrigues == -1) or (critter_state(rodrigues) == 1)) and (global_var(GVAR_SEDIT_DEAD_GANGSTERS) != 2)) then begin
+        set_global_var(GVAR_SEDIT_DEAD_GANGSTERS, 2);
         set_map_var(7, 1);
         set_map_var(8, 1);
     end
@@ -339,18 +339,18 @@ procedure timed_event_p_proc
 begin
     if (fixed_param == 1) then begin
         if (lRaidersAlive == 0) then begin
-            if (global_var(1016) == 0) then begin
-                set_global_var(1016, 1);
+            if (global_var(GVAR_S_RAIDERS_KILLED) == 0) then begin
+                set_global_var(GVAR_S_RAIDERS_KILLED, 1);
                 set_map_var(4, 16);
-                if (global_var(746) == 1) then begin
-                    set_global_var(746, 2);
+                if (global_var(GVAR_QUEST_S_FREDERYK_RAIDERS) == 1) then begin
+                    set_global_var(GVAR_QUEST_S_FREDERYK_RAIDERS, 2);
                 end
-                set_global_var(747, 0);
-                set_global_var(1015, global_var(1015) bwand (-1 - 64));
-                display_msg(message_str(1465, 450 + (get_critter_stat(dude_obj, 34) == 1)));
+                set_global_var(GVAR_DEAL_WITH_RAIDERS, 0);
+                set_global_var(GVAR_PTR_KERI_REACTIONS, global_var(GVAR_PTR_KERI_REACTIONS) bwand (-1 - 64));
+                display_mstr(450 + (dude_gender == 1));
             end
         end
-        if (global_var(1016) == 0) then begin
+        if (global_var(GVAR_S_RAIDERS_KILLED) == 0) then begin
             add_timer_event(self_obj, 30, 1);
             lRaidersAlive := 0;
         end
@@ -359,4 +359,3 @@ begin
         lRaidersAlive := 1;
     end
 end
-
