diff --git a/Mods/Resurrection/Original/SFIGHFLO.ssl b/Mods/Resurrection/ReDefine/SFIGHFLO.ssl
index 4e2e8bd..6c438be 100644
--- a/Mods/Resurrection/Original/SFIGHFLO.ssl
+++ b/Mods/Resurrection/ReDefine/SFIGHFLO.ssl
@@ -45,12 +45,12 @@ begin
             return 1;
         end
     end
-    if ((party_member_obj(16777779) != 0) and ((global_var(910) bwand 32) == 0)) then begin
+    if ((party_member_obj(16777779) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 32) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777779)) <= 5) then begin
             return 1;
         end
     end
-    if ((party_member_obj(16777811) != 0) and ((global_var(910) bwand 64) == 0)) then begin
+    if ((party_member_obj(16777811) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 64) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777811)) <= 5) then begin
             return 1;
         end
@@ -65,7 +65,7 @@ begin
             return 1;
         end
     end
-    if ((party_member_obj(16777814) != 0) and ((global_var(910) bwand 128) == 0)) then begin
+    if ((party_member_obj(16777814) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 128) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777814)) <= 5) then begin
             return 1;
         end
@@ -126,8 +126,8 @@ end
 
 procedure map_enter_p_proc
 begin
-    critter_add_trait(self_obj, 1, 6, 188);
-    critter_add_trait(self_obj, 1, 5, 132);
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 188);
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_FLOATER);
     if (obj_pid(self_obj) == 16778048) then begin
         set_local_var(6, 1);
     end
@@ -136,14 +136,14 @@ end
 procedure critter_p_proc
 begin
     if ((local_var(5) == 2) and obj_can_see_obj(self_obj, dude_obj)) then begin
-        attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+        attack(dude_obj);
     end
 end
 
 procedure timed_event_p_proc
 begin
     if (fixed_param == 1) then begin
-        attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+        attack(dude_obj);
         set_local_var(5, 2);
         set_local_var(7, 1);
     end
@@ -153,7 +153,7 @@ procedure look_at_p_proc
 begin
     if (local_var(6) == 1) then begin
         script_overrides;
-        display_msg(message_str(1335, 100));
+        display_mstr(100);
     end
 end
 
@@ -164,18 +164,18 @@ begin
         LVar0 := random(0, 2);
         script_overrides;
         if (LVar0 == 0) then begin
-            display_msg(message_str(1335, 201));
+            display_mstr(201);
         end
         if (LVar0 == 1) then begin
-            display_msg(message_str(1335, 202));
+            display_mstr(202);
         end
         if (LVar0 == 2) then begin
-            display_msg(message_str(1335, 203));
+            display_mstr(203);
         end
     end
     else begin
         script_overrides;
-        display_msg(message_str(1335, 200));
+        display_mstr(200);
     end
 end
 
@@ -190,7 +190,7 @@ procedure damage_p_proc
 begin
     if (source_obj == dude_obj) then begin
         if (local_var(7) != 1) then begin
-            set_global_var(770, 1);
+            set_global_var(GVAR_ENEMY_RODRIGUES, 1);
         end
     end
 end
@@ -198,4 +198,3 @@ end
 procedure map_update_p_proc
 begin
 end
-
