diff --git a/Mods/Resurrection/Original/SFRENOTI.ssl b/Mods/Resurrection/ReDefine/SFRENOTI.ssl
index 74e402d..29aba31 100644
--- a/Mods/Resurrection/Original/SFRENOTI.ssl
+++ b/Mods/Resurrection/ReDefine/SFRENOTI.ssl
@@ -40,12 +40,12 @@ begin
             return 1;
         end
     end
-    if ((party_member_obj(16777779) != 0) and ((global_var(910) bwand 32) == 0)) then begin
+    if ((party_member_obj(16777779) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 32) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777779)) <= 5) then begin
             return 1;
         end
     end
-    if ((party_member_obj(16777811) != 0) and ((global_var(910) bwand 64) == 0)) then begin
+    if ((party_member_obj(16777811) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 64) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777811)) <= 5) then begin
             return 1;
         end
@@ -60,7 +60,7 @@ begin
             return 1;
         end
     end
-    if ((party_member_obj(16777814) != 0) and ((global_var(910) bwand 128) == 0)) then begin
+    if ((party_member_obj(16777814) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 128) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777814)) <= 5) then begin
             return 1;
         end
@@ -121,12 +121,12 @@ end
 
 procedure map_enter_p_proc
 begin
-    if ((global_var(745) == 1) and (global_var(754) == 2)) then begin
-        create_object_sid(16777763, 19300, 0, 1385);
+    if ((global_var(GVAR_QUEST_S_FREDERYK_NOTICE) == 1) and (global_var(GVAR_JEREMY_DEAL) == 2)) then begin
+        create_object_sid(16777763, 19300, 0, SCRIPT_STIM);
         add_timer_event(self_obj, game_ticks(1), 1);
     end
-    if ((global_var(745) == 1) and (global_var(754) < 2)) then begin
-        create_object_sid(16777764, 19300, 0, 1384);
+    if ((global_var(GVAR_QUEST_S_FREDERYK_NOTICE) == 1) and (global_var(GVAR_JEREMY_DEAL) < 2)) then begin
+        create_object_sid(16777764, 19300, 0, SCRIPT_SJEREMY);
         add_timer_event(self_obj, game_ticks(1), 2);
     end
     if ((get_month >= 3) and (get_month < 5)) then begin
@@ -292,180 +292,179 @@ begin
     if (fixed_param == 1) then begin
         if (party_member_obj(16777729) != 0) then begin
             if (party_member_obj(16777729) != 0) then begin
-                if (has_trait(1, party_member_obj(16777729), 666)) then begin
+                if (obj_is_visible_flag(party_member_obj(16777729))) then begin
                     critter_attempt_placement(party_member_obj(16777729), 20500, 0);
                 end
             end
         end
-        if ((party_member_obj(16777779) != 0) and ((global_var(910) bwand 32) == 0)) then begin
+        if ((party_member_obj(16777779) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 32) == 0)) then begin
             if (party_member_obj(16777779) != 0) then begin
-                if (has_trait(1, party_member_obj(16777779), 666)) then begin
+                if (obj_is_visible_flag(party_member_obj(16777779))) then begin
                     critter_attempt_placement(party_member_obj(16777779), 20500, 0);
                 end
             end
         end
-        if ((party_member_obj(16777811) != 0) and ((global_var(910) bwand 64) == 0)) then begin
+        if ((party_member_obj(16777811) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 64) == 0)) then begin
             if (party_member_obj(16777811) != 0) then begin
-                if (has_trait(1, party_member_obj(16777811), 666)) then begin
+                if (obj_is_visible_flag(party_member_obj(16777811))) then begin
                     critter_attempt_placement(party_member_obj(16777811), 20500, 0);
                 end
             end
         end
         if (party_member_obj(16777812) != 0) then begin
             if (party_member_obj(16777812) != 0) then begin
-                if (has_trait(1, party_member_obj(16777812), 666)) then begin
+                if (obj_is_visible_flag(party_member_obj(16777812))) then begin
                     critter_attempt_placement(party_member_obj(16777812), 20500, 0);
                 end
             end
         end
-        if ((party_member_obj(16777814) != 0) and ((global_var(910) bwand 128) == 0)) then begin
+        if ((party_member_obj(16777814) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 128) == 0)) then begin
             if (party_member_obj(16777814) != 0) then begin
-                if (has_trait(1, party_member_obj(16777814), 666)) then begin
+                if (obj_is_visible_flag(party_member_obj(16777814))) then begin
                     critter_attempt_placement(party_member_obj(16777814), 20500, 0);
                 end
             end
         end
         if (party_member_obj(16777748) != 0) then begin
             if (party_member_obj(16777748) != 0) then begin
-                if (has_trait(1, party_member_obj(16777748), 666)) then begin
+                if (obj_is_visible_flag(party_member_obj(16777748))) then begin
                     critter_attempt_placement(party_member_obj(16777748), 20500, 0);
                 end
             end
         end
         if (party_member_obj(16778017) != 0) then begin
             if (party_member_obj(16778017) != 0) then begin
-                if (has_trait(1, party_member_obj(16778017), 666)) then begin
+                if (obj_is_visible_flag(party_member_obj(16778017))) then begin
                     critter_attempt_placement(party_member_obj(16778017), 20500, 0);
                 end
             end
         end
         if (party_member_obj(16778016) != 0) then begin
             if (party_member_obj(16778016) != 0) then begin
-                if (has_trait(1, party_member_obj(16778016), 666)) then begin
+                if (obj_is_visible_flag(party_member_obj(16778016))) then begin
                     critter_attempt_placement(party_member_obj(16778016), 20500, 0);
                 end
             end
         end
         if (party_member_obj(16778049) != 0) then begin
             if (party_member_obj(16778049) != 0) then begin
-                if (has_trait(1, party_member_obj(16778049), 666)) then begin
+                if (obj_is_visible_flag(party_member_obj(16778049))) then begin
                     critter_attempt_placement(party_member_obj(16778049), 20500, 0);
                 end
             end
         end
         if (party_member_obj(16778050) != 0) then begin
             if (party_member_obj(16778050) != 0) then begin
-                if (has_trait(1, party_member_obj(16778050), 666)) then begin
+                if (obj_is_visible_flag(party_member_obj(16778050))) then begin
                     critter_attempt_placement(party_member_obj(16778050), 20500, 0);
                 end
             end
         end
         if (party_member_obj(16778051) != 0) then begin
             if (party_member_obj(16778051) != 0) then begin
-                if (has_trait(1, party_member_obj(16778051), 666)) then begin
+                if (obj_is_visible_flag(party_member_obj(16778051))) then begin
                     critter_attempt_placement(party_member_obj(16778051), 20500, 0);
                 end
             end
         end
         if (party_member_obj(16777293) != 0) then begin
             if (party_member_obj(16777293) != 0) then begin
-                if (has_trait(1, party_member_obj(16777293), 666)) then begin
+                if (obj_is_visible_flag(party_member_obj(16777293))) then begin
                     critter_attempt_placement(party_member_obj(16777293), 20500, 0);
                 end
             end
         end
         game_ui_enable;
-        gfade_in(1);
+        fade_in;
         add_timer_event(Tim, 1, 11);
     end
     if (fixed_param == 2) then begin
         if (party_member_obj(16777729) != 0) then begin
             if (party_member_obj(16777729) != 0) then begin
-                if (has_trait(1, party_member_obj(16777729), 666)) then begin
+                if (obj_is_visible_flag(party_member_obj(16777729))) then begin
                     critter_attempt_placement(party_member_obj(16777729), 20500, 0);
                 end
             end
         end
-        if ((party_member_obj(16777779) != 0) and ((global_var(910) bwand 32) == 0)) then begin
+        if ((party_member_obj(16777779) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 32) == 0)) then begin
             if (party_member_obj(16777779) != 0) then begin
-                if (has_trait(1, party_member_obj(16777779), 666)) then begin
+                if (obj_is_visible_flag(party_member_obj(16777779))) then begin
                     critter_attempt_placement(party_member_obj(16777779), 20500, 0);
                 end
             end
         end
-        if ((party_member_obj(16777811) != 0) and ((global_var(910) bwand 64) == 0)) then begin
+        if ((party_member_obj(16777811) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 64) == 0)) then begin
             if (party_member_obj(16777811) != 0) then begin
-                if (has_trait(1, party_member_obj(16777811), 666)) then begin
+                if (obj_is_visible_flag(party_member_obj(16777811))) then begin
                     critter_attempt_placement(party_member_obj(16777811), 20500, 0);
                 end
             end
         end
         if (party_member_obj(16777812) != 0) then begin
             if (party_member_obj(16777812) != 0) then begin
-                if (has_trait(1, party_member_obj(16777812), 666)) then begin
+                if (obj_is_visible_flag(party_member_obj(16777812))) then begin
                     critter_attempt_placement(party_member_obj(16777812), 20500, 0);
                 end
             end
         end
-        if ((party_member_obj(16777814) != 0) and ((global_var(910) bwand 128) == 0)) then begin
+        if ((party_member_obj(16777814) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 128) == 0)) then begin
             if (party_member_obj(16777814) != 0) then begin
-                if (has_trait(1, party_member_obj(16777814), 666)) then begin
+                if (obj_is_visible_flag(party_member_obj(16777814))) then begin
                     critter_attempt_placement(party_member_obj(16777814), 20500, 0);
                 end
             end
         end
         if (party_member_obj(16777748) != 0) then begin
             if (party_member_obj(16777748) != 0) then begin
-                if (has_trait(1, party_member_obj(16777748), 666)) then begin
+                if (obj_is_visible_flag(party_member_obj(16777748))) then begin
                     critter_attempt_placement(party_member_obj(16777748), 20500, 0);
                 end
             end
         end
         if (party_member_obj(16778017) != 0) then begin
             if (party_member_obj(16778017) != 0) then begin
-                if (has_trait(1, party_member_obj(16778017), 666)) then begin
+                if (obj_is_visible_flag(party_member_obj(16778017))) then begin
                     critter_attempt_placement(party_member_obj(16778017), 20500, 0);
                 end
             end
         end
         if (party_member_obj(16778016) != 0) then begin
             if (party_member_obj(16778016) != 0) then begin
-                if (has_trait(1, party_member_obj(16778016), 666)) then begin
+                if (obj_is_visible_flag(party_member_obj(16778016))) then begin
                     critter_attempt_placement(party_member_obj(16778016), 20500, 0);
                 end
             end
         end
         if (party_member_obj(16778049) != 0) then begin
             if (party_member_obj(16778049) != 0) then begin
-                if (has_trait(1, party_member_obj(16778049), 666)) then begin
+                if (obj_is_visible_flag(party_member_obj(16778049))) then begin
                     critter_attempt_placement(party_member_obj(16778049), 20500, 0);
                 end
             end
         end
         if (party_member_obj(16778050) != 0) then begin
             if (party_member_obj(16778050) != 0) then begin
-                if (has_trait(1, party_member_obj(16778050), 666)) then begin
+                if (obj_is_visible_flag(party_member_obj(16778050))) then begin
                     critter_attempt_placement(party_member_obj(16778050), 20500, 0);
                 end
             end
         end
         if (party_member_obj(16778051) != 0) then begin
             if (party_member_obj(16778051) != 0) then begin
-                if (has_trait(1, party_member_obj(16778051), 666)) then begin
+                if (obj_is_visible_flag(party_member_obj(16778051))) then begin
                     critter_attempt_placement(party_member_obj(16778051), 20500, 0);
                 end
             end
         end
         if (party_member_obj(16777293) != 0) then begin
             if (party_member_obj(16777293) != 0) then begin
-                if (has_trait(1, party_member_obj(16777293), 666)) then begin
+                if (obj_is_visible_flag(party_member_obj(16777293))) then begin
                     critter_attempt_placement(party_member_obj(16777293), 20500, 0);
                 end
             end
         end
         game_ui_enable;
-        gfade_in(1);
+        fade_in;
         add_timer_event(Jeremy, 1, 10);
     end
 end
-
