diff --git a/Mods/Resurrection/Original/SGABRIEL.ssl b/Mods/Resurrection/ReDefine/SGABRIEL.ssl
index ee11553..4084c89 100644
--- a/Mods/Resurrection/Original/SGABRIEL.ssl
+++ b/Mods/Resurrection/ReDefine/SGABRIEL.ssl
@@ -93,12 +93,12 @@ begin
             return 1;
         end
     end
-    if ((party_member_obj(16777779) != 0) and ((global_var(910) bwand 32) == 0)) then begin
+    if ((party_member_obj(16777779) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 32) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777779)) <= 5) then begin
             return 1;
         end
     end
-    if ((party_member_obj(16777811) != 0) and ((global_var(910) bwand 64) == 0)) then begin
+    if ((party_member_obj(16777811) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 64) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777811)) <= 5) then begin
             return 1;
         end
@@ -113,7 +113,7 @@ begin
             return 1;
         end
     end
-    if ((party_member_obj(16777814) != 0) and ((global_var(910) bwand 128) == 0)) then begin
+    if ((party_member_obj(16777814) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 128) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777814)) <= 5) then begin
             return 1;
         end
@@ -175,10 +175,10 @@ begin
             if (anim_busy(self_obj) == 0) then begin
                 if ((tile_distance_objs(self_obj, dude_obj) < 8) or obj_can_see_obj(self_obj, dude_obj)) then begin
                     if (anim_busy(self_obj) == 0) then begin
-                        global_temp := rotation_to_tile(tile_num(dude_obj), tile_num(self_obj));
+                        global_temp := rotation_to_tile(dude_tile, tile_num(self_obj));
                         animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(self_obj), global_temp, random(3, 10)), 1);
                         global_temp := (global_temp + 1) % 6;
-                        while ((anim_busy(self_obj) == 0) and (global_temp != rotation_to_tile(tile_num(dude_obj), tile_num(self_obj)))) do begin
+                        while ((anim_busy(self_obj) == 0) and (global_temp != rotation_to_tile(dude_tile, tile_num(self_obj)))) do begin
                             debug_msg("flee loop: rot == " + global_temp);
                             animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(self_obj), global_temp, random(3, 10)), 1);
                             global_temp := (global_temp + 1) % 6;
@@ -188,18 +188,18 @@ begin
             end
         end
         else begin
-            attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+            attack(dude_obj);
         end
     end
-    if ((global_var(770) == 1) and obj_can_see_obj(self_obj, dude_obj)) then begin
+    if ((global_var(GVAR_ENEMY_RODRIGUES) == 1) and obj_can_see_obj(self_obj, dude_obj)) then begin
         if (critter_is_fleeing(self_obj)) then begin
             if (anim_busy(self_obj) == 0) then begin
                 if ((tile_distance_objs(self_obj, dude_obj) < 8) or obj_can_see_obj(self_obj, dude_obj)) then begin
                     if (anim_busy(self_obj) == 0) then begin
-                        global_temp := rotation_to_tile(tile_num(dude_obj), tile_num(self_obj));
+                        global_temp := rotation_to_tile(dude_tile, tile_num(self_obj));
                         animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(self_obj), global_temp, random(3, 10)), 1);
                         global_temp := (global_temp + 1) % 6;
-                        while ((anim_busy(self_obj) == 0) and (global_temp != rotation_to_tile(tile_num(dude_obj), tile_num(self_obj)))) do begin
+                        while ((anim_busy(self_obj) == 0) and (global_temp != rotation_to_tile(dude_tile, tile_num(self_obj)))) do begin
                             debug_msg("flee loop: rot == " + global_temp);
                             animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(self_obj), global_temp, random(3, 10)), 1);
                             global_temp := (global_temp + 1) % 6;
@@ -209,7 +209,7 @@ begin
             end
         end
         else begin
-            attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+            attack(dude_obj);
         end
     end
 end
@@ -218,7 +218,7 @@ procedure pickup_p_proc
 begin
     if (source_obj == dude_obj) then begin
         set_local_var(4, 1);
-        attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+        attack(dude_obj);
     end
 end
 
@@ -226,27 +226,27 @@ procedure talk_p_proc
 begin
     projed_smycku := 0;
     if (((local_var(5) bwand 4) != 0) and (projed_smycku == 0)) then begin
-        if (get_critter_stat(dude_obj, 34) == 0) then begin
-            float_msg(self_obj, message_str(41, 300), 2);
+        if (dude_gender == 0) then begin
+            float_msg(self_obj, message_str(SCRIPT_GABRIEL, 300), FLOAT_MSG_RED);
         end
         else begin
-            float_msg(self_obj, message_str(41, 301), 2);
+            float_msg(self_obj, message_str(SCRIPT_GABRIEL, 301), FLOAT_MSG_RED);
         end
         projed_smycku := 1;
         call Node998();
     end
     if (((local_var(5) bwand 16) != 0) and (projed_smycku == 0)) then begin
-        float_msg(self_obj, message_str(41, 302), 2);
+        float_msg(self_obj, message_str(SCRIPT_GABRIEL, 302), FLOAT_MSG_RED);
         projed_smycku := 1;
         call Node998();
     end
     if (projed_smycku == 0) then begin
-        anim(dude_obj, 1000, rotation_to_tile(tile_num(dude_obj), tile_num(self_obj)));
-        anim(self_obj, 1000, rotation_to_tile(tile_num(self_obj), tile_num(dude_obj)));
+        anim(dude_obj, 1000, rotation_to_tile(dude_tile, tile_num(self_obj)));
+        anim(self_obj, 1000, rotation_to_tile(tile_num(self_obj), dude_tile));
         start_gdialog(41, self_obj, 4, -1, -1);
         gsay_start;
         if (((local_var(5) bwand 8) != 0) and (projed_smycku == 0)) then begin
-            if (get_critter_stat(dude_obj, 34) == 0) then begin
+            if (dude_gender == 0) then begin
                 call Node1500();
             end
             else begin
@@ -255,7 +255,7 @@ begin
             projed_smycku := 1;
         end
         if (((local_var(5) bwand 2) != 0) and (projed_smycku == 0)) then begin
-            if (get_critter_stat(dude_obj, 34) == 0) then begin
+            if (dude_gender == 0) then begin
                 call Node1300();
             end
             else begin
@@ -264,7 +264,7 @@ begin
             projed_smycku := 1;
         end
         if (((local_var(5) bwand 1) != 0) and (projed_smycku == 0)) then begin
-            if (get_critter_stat(dude_obj, 34) == 0) then begin
+            if (dude_gender == 0) then begin
                 call Node1250();
             end
             else begin
@@ -273,7 +273,7 @@ begin
             projed_smycku := 1;
         end
         if (projed_smycku == 0) then begin
-            if (get_critter_stat(dude_obj, 34) == 0) then begin
+            if (dude_gender == 0) then begin
                 call Node500();
             end
             else begin
@@ -291,55 +291,55 @@ begin
     if (destroy_after_join == 0) then begin
         set_local_var(4, 1);
         if (source_obj == dude_obj) then begin
-            set_global_var(4, global_var(4) + 1);
-            if (metarule(51, self_obj) == 2) then begin
+            set_global_var(GVAR_BAD_MONSTER, global_var(GVAR_BAD_MONSTER) + 1);
+            if (critter_kill_type(self_obj) == KILL_TYPE_children_kills) then begin
                 if (source_obj == dude_obj) then begin
-                    set_global_var(2, 0);
-                    set_global_var(1, global_var(1) + 1);
-                    set_global_var(0, global_var(0) + -15);
+                    set_global_var(GVAR_CHAMPION_REPUTATION, 0);
+                    set_global_var(GVAR_CHILDKILLER_REPUTATION, global_var(GVAR_CHILDKILLER_REPUTATION) + 1);
+                    set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + -15);
                     debug_msg("Player gains " + -15 + " Karma Points.");
-                    set_global_var(37, 0);
-                    set_global_var(38, 0);
-                    set_global_var(39, 0);
-                    set_global_var(40, 0);
-                    set_global_var(41, 0);
-                    set_global_var(42, 0);
-                    set_global_var(43, 0);
-                    set_global_var(44, 0);
-                    set_global_var(45, 0);
-                    if (global_var(0) >= 1000) then begin
-                        set_global_var(37, 1);
+                    set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+                    set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+                    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+                    set_global_var(GVAR_KARMA_DEFENDER, 0);
+                    set_global_var(GVAR_KARMA_WANDERER, 0);
+                    set_global_var(GVAR_KARMA_BETRAYER, 0);
+                    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+                    set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+                    set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+                        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
                     end
                     else begin
-                        if (global_var(0) >= 750) then begin
-                            set_global_var(38, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                            set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
                         end
                         else begin
-                            if (global_var(0) >= 500) then begin
-                                set_global_var(39, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                                set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                             end
                             else begin
-                                if (global_var(0) >= 250) then begin
-                                    set_global_var(40, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                                    set_global_var(GVAR_KARMA_DEFENDER, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -250) then begin
-                                        set_global_var(41, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                                        set_global_var(GVAR_KARMA_WANDERER, 1);
                                     end
                                     else begin
-                                        if (global_var(0) > -500) then begin
-                                            set_global_var(42, 1);
+                                        if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                            set_global_var(GVAR_KARMA_BETRAYER, 1);
                                         end
                                         else begin
-                                            if (global_var(0) > -750) then begin
-                                                set_global_var(43, 1);
+                                            if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                                set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                             end
                                             else begin
-                                                if (global_var(0) > -1000) then begin
-                                                    set_global_var(44, 1);
+                                                if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                                    set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                                 end
                                                 else begin
-                                                    set_global_var(45, 1);
+                                                    set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                                 end
                                             end
                                         end
@@ -350,58 +350,58 @@ begin
                     end
                 end
             end
-            if (((global_var(4) + global_var(5)) >= 25) and ((global_var(5) > (2 * global_var(4))) or (global_var(3) == 1))) then begin
-                set_global_var(3, 1);
-                set_global_var(2, 0);
+            if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
+                set_global_var(GVAR_BERSERKER_REPUTATION, 1);
+                set_global_var(GVAR_CHAMPION_REPUTATION, 0);
             end
-            if (((global_var(4) + global_var(5)) >= 25) and ((global_var(4) > (3 * global_var(5))) or (global_var(2) == 1)) and (global_var(1) == 0)) then begin
-                set_global_var(2, 1);
-                set_global_var(3, 0);
+            if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(4) > (3 * global_var(GVAR_GOOD_MONSTER))) or (global_var(GVAR_CHAMPION_REPUTATION) == 1)) and (global_var(GVAR_CHILDKILLER_REPUTATION) == 0)) then begin
+                set_global_var(GVAR_CHAMPION_REPUTATION, 1);
+                set_global_var(GVAR_BERSERKER_REPUTATION, 0);
             end
-            set_global_var(0, global_var(0) + 5);
+            set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + 5);
             debug_msg("Player gains " + 5 + " Karma Points.");
-            set_global_var(37, 0);
-            set_global_var(38, 0);
-            set_global_var(39, 0);
-            set_global_var(40, 0);
-            set_global_var(41, 0);
-            set_global_var(42, 0);
-            set_global_var(43, 0);
-            set_global_var(44, 0);
-            set_global_var(45, 0);
-            if (global_var(0) >= 1000) then begin
-                set_global_var(37, 1);
+            set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+            set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+            set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+            set_global_var(GVAR_KARMA_DEFENDER, 0);
+            set_global_var(GVAR_KARMA_WANDERER, 0);
+            set_global_var(GVAR_KARMA_BETRAYER, 0);
+            set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+            set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+            set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+                set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
             end
             else begin
-                if (global_var(0) >= 750) then begin
-                    set_global_var(38, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                    set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
                 end
                 else begin
-                    if (global_var(0) >= 500) then begin
-                        set_global_var(39, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                        set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                     end
                     else begin
-                        if (global_var(0) >= 250) then begin
-                            set_global_var(40, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                            set_global_var(GVAR_KARMA_DEFENDER, 1);
                         end
                         else begin
-                            if (global_var(0) > -250) then begin
-                                set_global_var(41, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                                set_global_var(GVAR_KARMA_WANDERER, 1);
                             end
                             else begin
-                                if (global_var(0) > -500) then begin
-                                    set_global_var(42, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                    set_global_var(GVAR_KARMA_BETRAYER, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -750) then begin
-                                        set_global_var(43, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                        set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                     end
                                     else begin
-                                        if (global_var(0) > -1000) then begin
-                                            set_global_var(44, 1);
+                                        if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                            set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                         end
                                         else begin
-                                            set_global_var(45, 1);
+                                            set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                         end
                                     end
                                 end
@@ -418,10 +418,10 @@ procedure look_at_p_proc
 begin
     script_overrides;
     if ((local_var(5) bwand 2) != 0) then begin
-        display_msg(message_str(41, 150));
+        display_msg(message_str(SCRIPT_GABRIEL, 150));
     end
     else begin
-        display_msg(message_str(41, 100));
+        display_msg(message_str(SCRIPT_GABRIEL, 100));
     end
 end
 
@@ -429,10 +429,10 @@ procedure description_p_proc
 begin
     script_overrides;
     if ((local_var(5) bwand 2) != 0) then begin
-        display_msg(message_str(41, 249));
+        display_msg(message_str(SCRIPT_GABRIEL, 249));
     end
     else begin
-        display_msg(message_str(41, 200));
+        display_msg(message_str(SCRIPT_GABRIEL, 200));
     end
 end
 
@@ -446,16 +446,16 @@ begin
         set_local_var(4, 1);
     end
     if (random(1, 12) == 6) then begin
-        float_msg(self_obj, message_str(41, random(359, 362)), 2);
+        float_msg(self_obj, message_str(SCRIPT_GABRIEL, random(359, 362)), FLOAT_MSG_RED);
     end
 end
 
 procedure map_enter_p_proc
 begin
-    critter_add_trait(self_obj, 1, 6, 213);
-    if (metarule(14, 0)) then begin
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 213);
+    if (map_first_run) then begin
         if (obj_carrying_pid_obj(self_obj, 1)) then begin
-            wield_obj_critter(self_obj, obj_carrying_pid_obj(self_obj, 1));
+            wield_obj(obj_carrying_pid_obj(self_obj, 1));
         end
     end
     add_timer_event(self_obj, game_ticks(random(90, 120)), 2);
@@ -482,10 +482,10 @@ begin
             if (anim_busy(self_obj) == 0) then begin
                 if ((tile_distance_objs(self_obj, dude_obj) < 8) or obj_can_see_obj(self_obj, dude_obj)) then begin
                     if (anim_busy(self_obj) == 0) then begin
-                        global_temp := rotation_to_tile(tile_num(dude_obj), tile_num(self_obj));
+                        global_temp := rotation_to_tile(dude_tile, tile_num(self_obj));
                         animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(self_obj), global_temp, random(3, 10)), 1);
                         global_temp := (global_temp + 1) % 6;
-                        while ((anim_busy(self_obj) == 0) and (global_temp != rotation_to_tile(tile_num(dude_obj), tile_num(self_obj)))) do begin
+                        while ((anim_busy(self_obj) == 0) and (global_temp != rotation_to_tile(dude_tile, tile_num(self_obj)))) do begin
                             debug_msg("flee loop: rot == " + global_temp);
                             animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(self_obj), global_temp, random(3, 10)), 1);
                             global_temp := (global_temp + 1) % 6;
@@ -495,7 +495,7 @@ begin
             end
         end
         else begin
-            attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+            attack(dude_obj);
         end
     end
 end
@@ -503,8 +503,8 @@ end
 procedure combat_p_proc
 begin
     if (fixed_param == 4) then begin
-        if ((local_var(4) == 1) or (global_var(770) == 1) and (random(1, 10) == 5)) then begin
-            float_msg(self_obj, message_str(41, random(350, 358)), 2);
+        if ((local_var(4) == 1) or (global_var(GVAR_ENEMY_RODRIGUES) == 1) and (random(1, 10) == 5)) then begin
+            float_msg(self_obj, message_str(SCRIPT_GABRIEL, random(350, 358)), FLOAT_MSG_RED);
         end
     end
 end
@@ -517,140 +517,140 @@ end
 
 procedure Node999
 begin
-    if (not((global_var(910) bwand 8) != 0)) then begin
+    if (not((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 8) != 0)) then begin
         anim(self_obj, 1000, 5);
     end
 end
 
 procedure Node500
 begin
-    gsay_reply(41, 500);
-    giq_option(4, 41, 501, Node550, 50);
-    giq_option(4, 41, 502, Node999, 50);
-    giq_option(-3, 41, 503, Node2400, 50);
+    gsay_reply(SCRIPT_GABRIEL, 500);
+    giq_option(4, SCRIPT_GABRIEL, 501, Node550, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_GABRIEL, 502, Node999, NEUTRAL_REACTION);
+    giq_option(-3, SCRIPT_GABRIEL, 503, Node2400, NEUTRAL_REACTION);
 end
 
 procedure Node550
 begin
-    gsay_reply(41, 550);
+    gsay_reply(SCRIPT_GABRIEL, 550);
     set_local_var(5, local_var(5) bwor 1);
-    giq_option(4, 41, 551, Node600, 50);
+    giq_option(4, SCRIPT_GABRIEL, 551, Node600, NEUTRAL_REACTION);
     if (item_caps_total(dude_obj) >= 10) then begin
-        if (global_var(0) > -50) then begin
-            giq_option(4, 41, 552, Node700, 50);
+        if (global_var(GVAR_PLAYER_REPUTATION) > -50) then begin
+            giq_option(4, SCRIPT_GABRIEL, 552, Node700, NEUTRAL_REACTION);
         end
         else begin
-            giq_option(4, 41, 552, Node650, 50);
+            giq_option(4, SCRIPT_GABRIEL, 552, Node650, NEUTRAL_REACTION);
         end
     end
-    giq_option(4, 41, 553, Node999, 50);
+    giq_option(4, SCRIPT_GABRIEL, 553, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node600
 begin
-    gsay_reply(41, 600);
+    gsay_reply(SCRIPT_GABRIEL, 600);
     if (item_caps_total(dude_obj) >= 10) then begin
-        if (global_var(0) > -50) then begin
-            giq_option(4, 41, 601, Node700, 50);
+        if (global_var(GVAR_PLAYER_REPUTATION) > -50) then begin
+            giq_option(4, SCRIPT_GABRIEL, 601, Node700, NEUTRAL_REACTION);
         end
         else begin
-            giq_option(4, 41, 601, Node650, 50);
+            giq_option(4, SCRIPT_GABRIEL, 601, Node650, NEUTRAL_REACTION);
         end
     end
-    giq_option(4, 41, 602, Node750, 50);
-    giq_option(4, 41, 603, Node999, 50);
+    giq_option(4, SCRIPT_GABRIEL, 602, Node750, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_GABRIEL, 603, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node650
 begin
-    gsay_reply(41, 650);
+    gsay_reply(SCRIPT_GABRIEL, 650);
     set_local_var(5, local_var(5) bwor 2);
-    giq_option(4, 41, 651, Bridge_800, 50);
+    giq_option(4, SCRIPT_GABRIEL, 651, Bridge_800, NEUTRAL_REACTION);
 end
 
 procedure Node700
 begin
-    gsay_reply(41, 700);
-    giq_option(4, 41, 701, Node750, 50);
-    giq_option(4, 41, 702, Node999, 50);
+    gsay_reply(SCRIPT_GABRIEL, 700);
+    giq_option(4, SCRIPT_GABRIEL, 701, Node750, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_GABRIEL, 702, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node750
 begin
-    gsay_reply(41, 750);
-    giq_option(4, 41, 751, Node998, 50);
-    giq_option(4, 41, 752, Node999, 50);
+    gsay_reply(SCRIPT_GABRIEL, 750);
+    giq_option(4, SCRIPT_GABRIEL, 751, Node998, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_GABRIEL, 752, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node800
 begin
-    gsay_reply(41, 800);
-    giq_option(4, 41, 801, Node850, 50);
+    gsay_reply(SCRIPT_GABRIEL, 800);
+    giq_option(4, SCRIPT_GABRIEL, 801, Node850, NEUTRAL_REACTION);
 end
 
 procedure Node850
 begin
-    gsay_reply(41, 850);
-    giq_option(4, 41, 851, Node900, 50);
-    giq_option(4, 41, 852, Node1000, 50);
-    giq_option(4, 41, 853, Node950, 50);
-    giq_option(4, 41, 854, Node999, 50);
+    gsay_reply(SCRIPT_GABRIEL, 850);
+    giq_option(4, SCRIPT_GABRIEL, 851, Node900, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_GABRIEL, 852, Node1000, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_GABRIEL, 853, Node950, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_GABRIEL, 854, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node900
 begin
-    gsay_reply(41, 900);
-    giq_option(4, 41, 901, Node1000, 50);
-    giq_option(4, 41, 902, Node950, 50);
-    giq_option(4, 41, 903, Node999, 50);
+    gsay_reply(SCRIPT_GABRIEL, 900);
+    giq_option(4, SCRIPT_GABRIEL, 901, Node1000, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_GABRIEL, 902, Node950, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_GABRIEL, 903, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node950
 begin
-    gsay_reply(41, 950);
-    set_global_var(0, global_var(0) - 1);
-    set_global_var(37, 0);
-    set_global_var(38, 0);
-    set_global_var(39, 0);
-    set_global_var(40, 0);
-    set_global_var(41, 0);
-    set_global_var(42, 0);
-    set_global_var(43, 0);
-    set_global_var(44, 0);
-    set_global_var(45, 0);
-    if (global_var(0) >= 1000) then begin
-        set_global_var(37, 1);
+    gsay_reply(SCRIPT_GABRIEL, 950);
+    set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) - 1);
+    set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+    set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+    set_global_var(GVAR_KARMA_DEFENDER, 0);
+    set_global_var(GVAR_KARMA_WANDERER, 0);
+    set_global_var(GVAR_KARMA_BETRAYER, 0);
+    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+    set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+    set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+    if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
     end
     else begin
-        if (global_var(0) >= 750) then begin
-            set_global_var(38, 1);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+            set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
         end
         else begin
-            if (global_var(0) >= 500) then begin
-                set_global_var(39, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
             end
             else begin
-                if (global_var(0) >= 250) then begin
-                    set_global_var(40, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                    set_global_var(GVAR_KARMA_DEFENDER, 1);
                 end
                 else begin
-                    if (global_var(0) > -250) then begin
-                        set_global_var(41, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                        set_global_var(GVAR_KARMA_WANDERER, 1);
                     end
                     else begin
-                        if (global_var(0) > -500) then begin
-                            set_global_var(42, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                            set_global_var(GVAR_KARMA_BETRAYER, 1);
                         end
                         else begin
-                            if (global_var(0) > -750) then begin
-                                set_global_var(43, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                             end
                             else begin
-                                if (global_var(0) > -1000) then begin
-                                    set_global_var(44, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                    set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                 end
                                 else begin
-                                    set_global_var(45, 1);
+                                    set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                 end
                             end
                         end
@@ -659,66 +659,66 @@ begin
             end
         end
     end
-    giq_option(4, 41, 951, Node1100, 50);
-    giq_option(4, 41, 952, Node1050, 50);
+    giq_option(4, SCRIPT_GABRIEL, 951, Node1100, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_GABRIEL, 952, Node1050, NEUTRAL_REACTION);
 end
 
 procedure Node1000
 begin
-    gsay_reply(41, 1000);
+    gsay_reply(SCRIPT_GABRIEL, 1000);
     set_local_var(5, local_var(5) bwor 4);
-    giq_option(4, 41, 1001, Node998, 50);
-    giq_option(4, 41, 1002, Node999, 50);
+    giq_option(4, SCRIPT_GABRIEL, 1001, Node998, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_GABRIEL, 1002, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node1050
 begin
-    gsay_reply(41, 1050);
+    gsay_reply(SCRIPT_GABRIEL, 1050);
     set_local_var(5, local_var(5) bwor 4);
-    set_global_var(0, global_var(0) + 1);
+    set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + 1);
     debug_msg("Player gains " + 1 + " Karma Points.");
-    set_global_var(37, 0);
-    set_global_var(38, 0);
-    set_global_var(39, 0);
-    set_global_var(40, 0);
-    set_global_var(41, 0);
-    set_global_var(42, 0);
-    set_global_var(43, 0);
-    set_global_var(44, 0);
-    set_global_var(45, 0);
-    if (global_var(0) >= 1000) then begin
-        set_global_var(37, 1);
+    set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+    set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+    set_global_var(GVAR_KARMA_DEFENDER, 0);
+    set_global_var(GVAR_KARMA_WANDERER, 0);
+    set_global_var(GVAR_KARMA_BETRAYER, 0);
+    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+    set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+    set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+    if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
     end
     else begin
-        if (global_var(0) >= 750) then begin
-            set_global_var(38, 1);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+            set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
         end
         else begin
-            if (global_var(0) >= 500) then begin
-                set_global_var(39, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
             end
             else begin
-                if (global_var(0) >= 250) then begin
-                    set_global_var(40, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                    set_global_var(GVAR_KARMA_DEFENDER, 1);
                 end
                 else begin
-                    if (global_var(0) > -250) then begin
-                        set_global_var(41, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                        set_global_var(GVAR_KARMA_WANDERER, 1);
                     end
                     else begin
-                        if (global_var(0) > -500) then begin
-                            set_global_var(42, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                            set_global_var(GVAR_KARMA_BETRAYER, 1);
                         end
                         else begin
-                            if (global_var(0) > -750) then begin
-                                set_global_var(43, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                             end
                             else begin
-                                if (global_var(0) > -1000) then begin
-                                    set_global_var(44, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                    set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                 end
                                 else begin
-                                    set_global_var(45, 1);
+                                    set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                 end
                             end
                         end
@@ -727,232 +727,232 @@ begin
             end
         end
     end
-    giq_option(4, 41, 1051, Node998, 50);
-    giq_option(4, 41, 1052, Node999, 50);
+    giq_option(4, SCRIPT_GABRIEL, 1051, Node998, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_GABRIEL, 1052, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node1100
 begin
-    gsay_reply(41, 1100);
+    gsay_reply(SCRIPT_GABRIEL, 1100);
     set_local_var(5, local_var(5) bwor 8);
-    if (((metarule(16, 0) - (((global_var(910) bwand 32) != 0) and not((global_var(632) bwand 2) != 0)) - (((global_var(910) bwand 64) != 0) and not((global_var(632) bwand 4) != 0)) - (((global_var(910) bwand 128) != 0) and not((global_var(632) bwand 32) != 0)) - 1) >= (floor(get_critter_stat(dude_obj, 3) / 2) + has_trait(0, dude_obj, 98))) or ((metarule(16, 0) - (((global_var(910) bwand 32) != 0) and not((global_var(632) bwand 2) != 0)) - (((global_var(910) bwand 64) != 0) and not((global_var(632) bwand 4) != 0)) - (((global_var(910) bwand 128) != 0) and not((global_var(632) bwand 32) != 0)) - 1) >= 5)) then begin
-        giq_option(4, 41, 1101, Node1200, 50);
+    if (((party_member_count(DONT_LIST_HIDDEN_MEMBERS) - (((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 32) != 0) and not((global_var(GVAR_PARTY_MEMBERS_HIDDEN) bwand 2) != 0)) - (((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 64) != 0) and not((global_var(GVAR_PARTY_MEMBERS_HIDDEN) bwand 4) != 0)) - (((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 128) != 0) and not((global_var(GVAR_PARTY_MEMBERS_HIDDEN) bwand 32) != 0)) - 1) >= (floor(dude_charisma / 2) + has_trait(TRAIT_PERK, dude_obj, PERK_magnetic_personality_perk))) or ((party_member_count(DONT_LIST_HIDDEN_MEMBERS) - (((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 32) != 0) and not((global_var(GVAR_PARTY_MEMBERS_HIDDEN) bwand 2) != 0)) - (((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 64) != 0) and not((global_var(GVAR_PARTY_MEMBERS_HIDDEN) bwand 4) != 0)) - (((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 128) != 0) and not((global_var(GVAR_PARTY_MEMBERS_HIDDEN) bwand 32) != 0)) - 1) >= 5)) then begin
+        giq_option(4, SCRIPT_GABRIEL, 1101, Node1200, NEUTRAL_REACTION);
     end
     else begin
-        giq_option(4, 41, 1101, JOIN_TO_PARTY, 50);
+        giq_option(4, SCRIPT_GABRIEL, 1101, JOIN_TO_PARTY, NEUTRAL_REACTION);
     end
-    giq_option(4, 41, 1102, Node1150, 50);
+    giq_option(4, SCRIPT_GABRIEL, 1102, Node1150, NEUTRAL_REACTION);
 end
 
 procedure Node1150
 begin
-    gsay_reply(41, 1150);
-    giq_option(4, 41, 1151, Node999, 50);
+    gsay_reply(SCRIPT_GABRIEL, 1150);
+    giq_option(4, SCRIPT_GABRIEL, 1151, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node1200
 begin
-    gsay_reply(41, 1200);
-    giq_option(4, 41, 1201, Node999, 50);
+    gsay_reply(SCRIPT_GABRIEL, 1200);
+    giq_option(4, SCRIPT_GABRIEL, 1201, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node1250
 begin
-    gsay_reply(41, 1250);
+    gsay_reply(SCRIPT_GABRIEL, 1250);
     if (item_caps_total(dude_obj) >= 10) then begin
-        if (global_var(0) > -50) then begin
-            giq_option(4, 41, 1251, Node700, 50);
+        if (global_var(GVAR_PLAYER_REPUTATION) > -50) then begin
+            giq_option(4, SCRIPT_GABRIEL, 1251, Node700, NEUTRAL_REACTION);
         end
         else begin
-            giq_option(4, 41, 1251, Node650, 50);
+            giq_option(4, SCRIPT_GABRIEL, 1251, Node650, NEUTRAL_REACTION);
         end
     end
-    giq_option(4, 41, 1252, Node999, 50);
-    giq_option(-3, 41, 1253, Node2400, 50);
+    giq_option(4, SCRIPT_GABRIEL, 1252, Node999, NEUTRAL_REACTION);
+    giq_option(-3, SCRIPT_GABRIEL, 1253, Node2400, NEUTRAL_REACTION);
 end
 
 procedure Node1300
 begin
-    gsay_reply(41, 1300);
-    if (global_var(0) > -50) then begin
-        giq_option(4, 41, 1301, Node1450, 50);
+    gsay_reply(SCRIPT_GABRIEL, 1300);
+    if (global_var(GVAR_PLAYER_REPUTATION) > -50) then begin
+        giq_option(4, SCRIPT_GABRIEL, 1301, Node1450, NEUTRAL_REACTION);
     end
     else begin
-        giq_option(4, 41, 1301, Node1350, 50);
+        giq_option(4, SCRIPT_GABRIEL, 1301, Node1350, NEUTRAL_REACTION);
     end
-    giq_option(4, 41, 1302, Node999, 50);
-    giq_option(-3, 41, 1303, Node2400, 50);
+    giq_option(4, SCRIPT_GABRIEL, 1302, Node999, NEUTRAL_REACTION);
+    giq_option(-3, SCRIPT_GABRIEL, 1303, Node2400, NEUTRAL_REACTION);
 end
 
 procedure Node1350
 begin
-    gsay_reply(41, 1350);
+    gsay_reply(SCRIPT_GABRIEL, 1350);
     if (item_caps_total(dude_obj) >= 10) then begin
-        giq_option(4, 41, 1351, Node1400, 50);
+        giq_option(4, SCRIPT_GABRIEL, 1351, Node1400, NEUTRAL_REACTION);
     end
     else begin
-        giq_option(4, 41, 1352, Node999, 50);
+        giq_option(4, SCRIPT_GABRIEL, 1352, Node999, NEUTRAL_REACTION);
     end
     if (item_caps_total(dude_obj) >= 10) then begin
-        giq_option(4, 41, 1353, Node999, 50);
+        giq_option(4, SCRIPT_GABRIEL, 1353, Node999, NEUTRAL_REACTION);
     end
 end
 
 procedure Node1400
 begin
-    gsay_reply(41, 1400);
+    gsay_reply(SCRIPT_GABRIEL, 1400);
     item_caps_adjust(dude_obj, -10);
-    giq_option(4, 41, 1401, Node950, 50);
-    giq_option(4, 41, 1402, Node999, 50);
+    giq_option(4, SCRIPT_GABRIEL, 1401, Node950, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_GABRIEL, 1402, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node1450
 begin
-    gsay_reply(41, 1450);
-    giq_option(4, 41, 1451, Node750, 50);
-    giq_option(4, 41, 1452, Node999, 50);
+    gsay_reply(SCRIPT_GABRIEL, 1450);
+    giq_option(4, SCRIPT_GABRIEL, 1451, Node750, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_GABRIEL, 1452, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node1500
 begin
-    gsay_reply(41, 1500);
-    if (global_var(0) > -50) then begin
-        giq_option(4, 41, 1501, Node2420, 50);
+    gsay_reply(SCRIPT_GABRIEL, 1500);
+    if (global_var(GVAR_PLAYER_REPUTATION) > -50) then begin
+        giq_option(4, SCRIPT_GABRIEL, 1501, Node2420, NEUTRAL_REACTION);
     end
     else begin
-        if (((metarule(16, 0) - (((global_var(910) bwand 32) != 0) and not((global_var(632) bwand 2) != 0)) - (((global_var(910) bwand 64) != 0) and not((global_var(632) bwand 4) != 0)) - (((global_var(910) bwand 128) != 0) and not((global_var(632) bwand 32) != 0)) - 1) >= (floor(get_critter_stat(dude_obj, 3) / 2) + has_trait(0, dude_obj, 98))) or ((metarule(16, 0) - (((global_var(910) bwand 32) != 0) and not((global_var(632) bwand 2) != 0)) - (((global_var(910) bwand 64) != 0) and not((global_var(632) bwand 4) != 0)) - (((global_var(910) bwand 128) != 0) and not((global_var(632) bwand 32) != 0)) - 1) >= 5)) then begin
-            giq_option(4, 41, 1501, Node1200, 50);
+        if (((party_member_count(DONT_LIST_HIDDEN_MEMBERS) - (((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 32) != 0) and not((global_var(GVAR_PARTY_MEMBERS_HIDDEN) bwand 2) != 0)) - (((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 64) != 0) and not((global_var(GVAR_PARTY_MEMBERS_HIDDEN) bwand 4) != 0)) - (((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 128) != 0) and not((global_var(GVAR_PARTY_MEMBERS_HIDDEN) bwand 32) != 0)) - 1) >= (floor(dude_charisma / 2) + has_trait(TRAIT_PERK, dude_obj, PERK_magnetic_personality_perk))) or ((party_member_count(DONT_LIST_HIDDEN_MEMBERS) - (((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 32) != 0) and not((global_var(GVAR_PARTY_MEMBERS_HIDDEN) bwand 2) != 0)) - (((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 64) != 0) and not((global_var(GVAR_PARTY_MEMBERS_HIDDEN) bwand 4) != 0)) - (((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 128) != 0) and not((global_var(GVAR_PARTY_MEMBERS_HIDDEN) bwand 32) != 0)) - 1) >= 5)) then begin
+            giq_option(4, SCRIPT_GABRIEL, 1501, Node1200, NEUTRAL_REACTION);
         end
         else begin
-            giq_option(4, 41, 1501, JOIN_TO_PARTY, 50);
+            giq_option(4, SCRIPT_GABRIEL, 1501, JOIN_TO_PARTY, NEUTRAL_REACTION);
         end
     end
-    giq_option(4, 41, 1502, Node999, 50);
-    giq_option(-3, 41, 1503, Node999, 50);
+    giq_option(4, SCRIPT_GABRIEL, 1502, Node999, NEUTRAL_REACTION);
+    giq_option(-3, SCRIPT_GABRIEL, 1503, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node1550
 begin
-    gsay_reply(41, 1550);
-    giq_option(4, 41, 1551, Node1600, 50);
-    giq_option(4, 41, 1552, Node999, 50);
-    giq_option(-3, 41, 1553, Node2400, 50);
+    gsay_reply(SCRIPT_GABRIEL, 1550);
+    giq_option(4, SCRIPT_GABRIEL, 1551, Node1600, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_GABRIEL, 1552, Node999, NEUTRAL_REACTION);
+    giq_option(-3, SCRIPT_GABRIEL, 1553, Node2400, NEUTRAL_REACTION);
 end
 
 procedure Node1600
 begin
-    gsay_reply(41, 1600);
+    gsay_reply(SCRIPT_GABRIEL, 1600);
     set_local_var(5, local_var(5) bwor 1);
-    giq_option(4, 41, 1601, Node1650, 50);
+    giq_option(4, SCRIPT_GABRIEL, 1601, Node1650, NEUTRAL_REACTION);
     if (item_caps_total(dude_obj) >= 10) then begin
-        if (global_var(0) > -50) then begin
-            giq_option(4, 41, 1602, Node1750, 50);
+        if (global_var(GVAR_PLAYER_REPUTATION) > -50) then begin
+            giq_option(4, SCRIPT_GABRIEL, 1602, Node1750, NEUTRAL_REACTION);
         end
         else begin
-            giq_option(4, 41, 1602, Node1700, 50);
+            giq_option(4, SCRIPT_GABRIEL, 1602, Node1700, NEUTRAL_REACTION);
         end
     end
-    giq_option(4, 41, 1603, Node999, 50);
+    giq_option(4, SCRIPT_GABRIEL, 1603, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node1650
 begin
-    gsay_reply(41, 1650);
+    gsay_reply(SCRIPT_GABRIEL, 1650);
     if (item_caps_total(dude_obj) >= 10) then begin
-        if (global_var(0) > -50) then begin
-            giq_option(4, 41, 1651, Node1750, 50);
+        if (global_var(GVAR_PLAYER_REPUTATION) > -50) then begin
+            giq_option(4, SCRIPT_GABRIEL, 1651, Node1750, NEUTRAL_REACTION);
         end
         else begin
-            giq_option(4, 41, 1651, Node1700, 50);
+            giq_option(4, SCRIPT_GABRIEL, 1651, Node1700, NEUTRAL_REACTION);
         end
     end
-    giq_option(4, 41, 1652, Node750, 50);
-    giq_option(4, 41, 1653, Node999, 50);
+    giq_option(4, SCRIPT_GABRIEL, 1652, Node750, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_GABRIEL, 1653, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node1700
 begin
-    gsay_reply(41, 1700);
+    gsay_reply(SCRIPT_GABRIEL, 1700);
     set_local_var(5, local_var(5) bwor 2);
-    giq_option(4, 41, 1701, Bridge_800, 50);
+    giq_option(4, SCRIPT_GABRIEL, 1701, Bridge_800, NEUTRAL_REACTION);
 end
 
 procedure Node1750
 begin
-    gsay_reply(41, 1750);
-    giq_option(4, 41, 1751, Node750, 50);
-    giq_option(4, 41, 1752, Node999, 50);
+    gsay_reply(SCRIPT_GABRIEL, 1750);
+    giq_option(4, SCRIPT_GABRIEL, 1751, Node750, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_GABRIEL, 1752, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node1800
 begin
-    gsay_reply(41, 1800);
-    giq_option(4, 41, 1801, Node1850, 50);
+    gsay_reply(SCRIPT_GABRIEL, 1800);
+    giq_option(4, SCRIPT_GABRIEL, 1801, Node1850, NEUTRAL_REACTION);
 end
 
 procedure Node1850
 begin
-    gsay_reply(41, 1850);
-    giq_option(4, 41, 1851, Node1900, 50);
-    giq_option(4, 41, 1852, Node2000, 50);
-    giq_option(4, 41, 1853, Node1950, 50);
-    giq_option(4, 41, 1854, Node999, 50);
+    gsay_reply(SCRIPT_GABRIEL, 1850);
+    giq_option(4, SCRIPT_GABRIEL, 1851, Node1900, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_GABRIEL, 1852, Node2000, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_GABRIEL, 1853, Node1950, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_GABRIEL, 1854, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node1900
 begin
-    gsay_reply(41, 1900);
-    giq_option(4, 41, 1901, Node2000, 50);
-    giq_option(4, 41, 1902, Node1950, 50);
-    giq_option(4, 41, 1903, Node999, 50);
+    gsay_reply(SCRIPT_GABRIEL, 1900);
+    giq_option(4, SCRIPT_GABRIEL, 1901, Node2000, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_GABRIEL, 1902, Node1950, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_GABRIEL, 1903, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node1950
 begin
-    gsay_reply(41, 1950);
-    set_global_var(0, global_var(0) - 1);
-    set_global_var(37, 0);
-    set_global_var(38, 0);
-    set_global_var(39, 0);
-    set_global_var(40, 0);
-    set_global_var(41, 0);
-    set_global_var(42, 0);
-    set_global_var(43, 0);
-    set_global_var(44, 0);
-    set_global_var(45, 0);
-    if (global_var(0) >= 1000) then begin
-        set_global_var(37, 1);
+    gsay_reply(SCRIPT_GABRIEL, 1950);
+    set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) - 1);
+    set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+    set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+    set_global_var(GVAR_KARMA_DEFENDER, 0);
+    set_global_var(GVAR_KARMA_WANDERER, 0);
+    set_global_var(GVAR_KARMA_BETRAYER, 0);
+    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+    set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+    set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+    if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
     end
     else begin
-        if (global_var(0) >= 750) then begin
-            set_global_var(38, 1);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+            set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
         end
         else begin
-            if (global_var(0) >= 500) then begin
-                set_global_var(39, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
             end
             else begin
-                if (global_var(0) >= 250) then begin
-                    set_global_var(40, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                    set_global_var(GVAR_KARMA_DEFENDER, 1);
                 end
                 else begin
-                    if (global_var(0) > -250) then begin
-                        set_global_var(41, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                        set_global_var(GVAR_KARMA_WANDERER, 1);
                     end
                     else begin
-                        if (global_var(0) > -500) then begin
-                            set_global_var(42, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                            set_global_var(GVAR_KARMA_BETRAYER, 1);
                         end
                         else begin
-                            if (global_var(0) > -750) then begin
-                                set_global_var(43, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                             end
                             else begin
-                                if (global_var(0) > -1000) then begin
-                                    set_global_var(44, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                    set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                 end
                                 else begin
-                                    set_global_var(45, 1);
+                                    set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                 end
                             end
                         end
@@ -961,58 +961,58 @@ begin
             end
         end
     end
-    giq_option(4, 41, 1951, Node1100, 50);
-    giq_option(4, 41, 1952, Node2050, 50);
+    giq_option(4, SCRIPT_GABRIEL, 1951, Node1100, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_GABRIEL, 1952, Node2050, NEUTRAL_REACTION);
 end
 
 procedure Node2000
 begin
-    gsay_reply(41, 2000);
+    gsay_reply(SCRIPT_GABRIEL, 2000);
     set_local_var(5, local_var(5) bwor 4);
-    set_global_var(0, global_var(0) + 1);
+    set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + 1);
     debug_msg("Player gains " + 1 + " Karma Points.");
-    set_global_var(37, 0);
-    set_global_var(38, 0);
-    set_global_var(39, 0);
-    set_global_var(40, 0);
-    set_global_var(41, 0);
-    set_global_var(42, 0);
-    set_global_var(43, 0);
-    set_global_var(44, 0);
-    set_global_var(45, 0);
-    if (global_var(0) >= 1000) then begin
-        set_global_var(37, 1);
+    set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+    set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+    set_global_var(GVAR_KARMA_DEFENDER, 0);
+    set_global_var(GVAR_KARMA_WANDERER, 0);
+    set_global_var(GVAR_KARMA_BETRAYER, 0);
+    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+    set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+    set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+    if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
     end
     else begin
-        if (global_var(0) >= 750) then begin
-            set_global_var(38, 1);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+            set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
         end
         else begin
-            if (global_var(0) >= 500) then begin
-                set_global_var(39, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
             end
             else begin
-                if (global_var(0) >= 250) then begin
-                    set_global_var(40, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                    set_global_var(GVAR_KARMA_DEFENDER, 1);
                 end
                 else begin
-                    if (global_var(0) > -250) then begin
-                        set_global_var(41, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                        set_global_var(GVAR_KARMA_WANDERER, 1);
                     end
                     else begin
-                        if (global_var(0) > -500) then begin
-                            set_global_var(42, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                            set_global_var(GVAR_KARMA_BETRAYER, 1);
                         end
                         else begin
-                            if (global_var(0) > -750) then begin
-                                set_global_var(43, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                             end
                             else begin
-                                if (global_var(0) > -1000) then begin
-                                    set_global_var(44, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                    set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                 end
                                 else begin
-                                    set_global_var(45, 1);
+                                    set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                 end
                             end
                         end
@@ -1021,117 +1021,117 @@ begin
             end
         end
     end
-    giq_option(4, 41, 2001, Node998, 50);
-    giq_option(4, 41, 2002, Node999, 50);
+    giq_option(4, SCRIPT_GABRIEL, 2001, Node998, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_GABRIEL, 2002, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node2050
 begin
-    gsay_reply(41, 2050);
+    gsay_reply(SCRIPT_GABRIEL, 2050);
     set_local_var(5, local_var(5) bwor 4);
-    giq_option(4, 41, 2051, Node998, 50);
-    giq_option(4, 41, 2052, Node999, 50);
+    giq_option(4, SCRIPT_GABRIEL, 2051, Node998, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_GABRIEL, 2052, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node2100
 begin
-    gsay_reply(41, 2100);
+    gsay_reply(SCRIPT_GABRIEL, 2100);
     if (item_caps_total(dude_obj) >= 10) then begin
-        if (global_var(0) > -50) then begin
-            giq_option(4, 41, 2101, Node1750, 50);
+        if (global_var(GVAR_PLAYER_REPUTATION) > -50) then begin
+            giq_option(4, SCRIPT_GABRIEL, 2101, Node1750, NEUTRAL_REACTION);
         end
         else begin
-            giq_option(4, 41, 2101, Node1700, 50);
+            giq_option(4, SCRIPT_GABRIEL, 2101, Node1700, NEUTRAL_REACTION);
         end
     end
-    giq_option(4, 41, 2102, Node999, 50);
-    giq_option(-3, 41, 2103, Node2400, 50);
+    giq_option(4, SCRIPT_GABRIEL, 2102, Node999, NEUTRAL_REACTION);
+    giq_option(-3, SCRIPT_GABRIEL, 2103, Node2400, NEUTRAL_REACTION);
 end
 
 procedure Node2150
 begin
-    gsay_reply(41, 2150);
-    if (global_var(0) > -50) then begin
-        giq_option(4, 41, 2151, Node2300, 50);
+    gsay_reply(SCRIPT_GABRIEL, 2150);
+    if (global_var(GVAR_PLAYER_REPUTATION) > -50) then begin
+        giq_option(4, SCRIPT_GABRIEL, 2151, Node2300, NEUTRAL_REACTION);
     end
     else begin
-        giq_option(4, 41, 2151, Node2200, 50);
+        giq_option(4, SCRIPT_GABRIEL, 2151, Node2200, NEUTRAL_REACTION);
     end
-    giq_option(4, 41, 2152, Node999, 50);
-    giq_option(-3, 41, 2153, Node2400, 50);
+    giq_option(4, SCRIPT_GABRIEL, 2152, Node999, NEUTRAL_REACTION);
+    giq_option(-3, SCRIPT_GABRIEL, 2153, Node2400, NEUTRAL_REACTION);
 end
 
 procedure Node2200
 begin
-    gsay_reply(41, 2200);
+    gsay_reply(SCRIPT_GABRIEL, 2200);
     if (item_caps_total(dude_obj) >= 10) then begin
-        giq_option(4, 41, 2201, Node2250, 50);
+        giq_option(4, SCRIPT_GABRIEL, 2201, Node2250, NEUTRAL_REACTION);
     end
     else begin
-        giq_option(4, 41, 2202, Node999, 50);
+        giq_option(4, SCRIPT_GABRIEL, 2202, Node999, NEUTRAL_REACTION);
     end
     if (item_caps_total(dude_obj) >= 10) then begin
-        giq_option(4, 41, 2203, Node999, 50);
+        giq_option(4, SCRIPT_GABRIEL, 2203, Node999, NEUTRAL_REACTION);
     end
 end
 
 procedure Node2250
 begin
-    gsay_reply(41, 2250);
-    giq_option(4, 41, 2251, Node1950, 50);
-    giq_option(4, 41, 2252, Node999, 50);
+    gsay_reply(SCRIPT_GABRIEL, 2250);
+    giq_option(4, SCRIPT_GABRIEL, 2251, Node1950, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_GABRIEL, 2252, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node2300
 begin
-    gsay_reply(41, 2300);
-    giq_option(4, 41, 2301, Node750, 50);
-    giq_option(4, 41, 2302, Node999, 50);
+    gsay_reply(SCRIPT_GABRIEL, 2300);
+    giq_option(4, SCRIPT_GABRIEL, 2301, Node750, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_GABRIEL, 2302, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node2350
 begin
-    gsay_reply(41, 2350);
-    if (global_var(0) > -50) then begin
-        giq_option(4, 41, 2351, Node2420, 50);
+    gsay_reply(SCRIPT_GABRIEL, 2350);
+    if (global_var(GVAR_PLAYER_REPUTATION) > -50) then begin
+        giq_option(4, SCRIPT_GABRIEL, 2351, Node2420, NEUTRAL_REACTION);
     end
     else begin
-        if (((metarule(16, 0) - (((global_var(910) bwand 32) != 0) and not((global_var(632) bwand 2) != 0)) - (((global_var(910) bwand 64) != 0) and not((global_var(632) bwand 4) != 0)) - (((global_var(910) bwand 128) != 0) and not((global_var(632) bwand 32) != 0)) - 1) >= (floor(get_critter_stat(dude_obj, 3) / 2) + has_trait(0, dude_obj, 98))) or ((metarule(16, 0) - (((global_var(910) bwand 32) != 0) and not((global_var(632) bwand 2) != 0)) - (((global_var(910) bwand 64) != 0) and not((global_var(632) bwand 4) != 0)) - (((global_var(910) bwand 128) != 0) and not((global_var(632) bwand 32) != 0)) - 1) >= 5)) then begin
-            giq_option(4, 41, 2351, Node1200, 50);
+        if (((party_member_count(DONT_LIST_HIDDEN_MEMBERS) - (((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 32) != 0) and not((global_var(GVAR_PARTY_MEMBERS_HIDDEN) bwand 2) != 0)) - (((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 64) != 0) and not((global_var(GVAR_PARTY_MEMBERS_HIDDEN) bwand 4) != 0)) - (((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 128) != 0) and not((global_var(GVAR_PARTY_MEMBERS_HIDDEN) bwand 32) != 0)) - 1) >= (floor(dude_charisma / 2) + has_trait(TRAIT_PERK, dude_obj, PERK_magnetic_personality_perk))) or ((party_member_count(DONT_LIST_HIDDEN_MEMBERS) - (((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 32) != 0) and not((global_var(GVAR_PARTY_MEMBERS_HIDDEN) bwand 2) != 0)) - (((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 64) != 0) and not((global_var(GVAR_PARTY_MEMBERS_HIDDEN) bwand 4) != 0)) - (((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 128) != 0) and not((global_var(GVAR_PARTY_MEMBERS_HIDDEN) bwand 32) != 0)) - 1) >= 5)) then begin
+            giq_option(4, SCRIPT_GABRIEL, 2351, Node1200, NEUTRAL_REACTION);
         end
         else begin
-            giq_option(4, 41, 2351, JOIN_TO_PARTY, 50);
+            giq_option(4, SCRIPT_GABRIEL, 2351, JOIN_TO_PARTY, NEUTRAL_REACTION);
         end
     end
-    giq_option(4, 41, 2352, Node999, 50);
-    giq_option(-3, 41, 2353, Node999, 50);
+    giq_option(4, SCRIPT_GABRIEL, 2352, Node999, NEUTRAL_REACTION);
+    giq_option(-3, SCRIPT_GABRIEL, 2353, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node2400
 begin
-    gsay_reply(41, 2400);
+    gsay_reply(SCRIPT_GABRIEL, 2400);
     set_local_var(5, local_var(5) bwor 16);
-    giq_option(-3, 41, 2401, Node999, 50);
+    giq_option(-3, SCRIPT_GABRIEL, 2401, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node2420
 begin
-    gsay_reply(41, 2420 + (get_critter_stat(dude_obj, 34) == 1));
-    giq_option(4, 41, 2422, Node999, 50);
+    gsay_reply(SCRIPT_GABRIEL, 2420 + (dude_gender == 1));
+    giq_option(4, SCRIPT_GABRIEL, 2422, Node999, NEUTRAL_REACTION);
 end
 
 procedure JOIN_TO_PARTY
 begin
-    gabriel_partym := create_object_sid(16777811, tile_num(self_obj), elevation(self_obj), 41);
-    anim(gabriel_partym, 1000, has_trait(1, self_obj, 10));
+    gabriel_partym := create_object_sid(16777811, tile_num(self_obj), elevation(self_obj), SCRIPT_GABRIEL);
+    anim(gabriel_partym, 1000, obj_get_rot(self_obj));
     party_add(gabriel_partym);
-    critter_add_trait(gabriel_partym, 1, 6, 0);
+    critter_add_trait(gabriel_partym, TRAIT_OBJECT, OBJECT_TEAM_NUM, 0);
     move_obj_inven_to_obj(self_obj, gabriel_partym);
     wield_obj_critter(gabriel_partym, obj_carrying_pid_obj(gabriel_partym, 1));
-    if (not((global_var(910) bwand 8) != 0)) then begin
-        set_global_var(910, global_var(910) bwor 8);
+    if (not((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 8) != 0)) then begin
+        set_global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS, global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwor 8);
         give_exp_points(100);
-        display_msg(message_str(14, 100) + 100 + message_str(14, 101));
+        display_msg(g_mstr(100) + 100 + g_mstr(101));
     end
     destroy_after_join := 1;
     add_timer_event(self_obj, 2, 1);
@@ -1141,11 +1141,10 @@ end
 procedure Bridge_800
 begin
     item_caps_adjust(dude_obj, -10);
-    if (get_critter_stat(dude_obj, 34) == 0) then begin
+    if (dude_gender == 0) then begin
         call Node800();
     end
     else begin
         call Node1800();
     end
 end
-
