diff --git a/Mods/Resurrection/Original/SMANTIS.ssl b/Mods/Resurrection/ReDefine/SMANTIS.ssl
index 0eadf58..6a01d01 100644
--- a/Mods/Resurrection/Original/SMANTIS.ssl
+++ b/Mods/Resurrection/ReDefine/SMANTIS.ssl
@@ -43,12 +43,12 @@ begin
             return 1;
         end
     end
-    if ((party_member_obj(16777779) != 0) and ((global_var(910) bwand 32) == 0)) then begin
+    if ((party_member_obj(16777779) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 32) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777779)) <= 5) then begin
             return 1;
         end
     end
-    if ((party_member_obj(16777811) != 0) and ((global_var(910) bwand 64) == 0)) then begin
+    if ((party_member_obj(16777811) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 64) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777811)) <= 5) then begin
             return 1;
         end
@@ -63,7 +63,7 @@ begin
             return 1;
         end
     end
-    if ((party_member_obj(16777814) != 0) and ((global_var(910) bwand 128) == 0)) then begin
+    if ((party_member_obj(16777814) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 128) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777814)) <= 5) then begin
             return 1;
         end
@@ -121,22 +121,22 @@ end
 procedure critter_p_proc
 begin
     if (obj_can_see_obj(self_obj, dude_obj)) then begin
-        attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+        attack(dude_obj);
     end
 end
 
 procedure destroy_p_proc
 begin
     set_map_var(0, map_var(0) + 1);
-    if ((global_var(786) == 1) and (map_var(0) >= 9)) then begin
-        display_msg(message_str(1450, 200 + (get_critter_stat(dude_obj, 34) == 1)));
+    if ((global_var(GVAR_QUEST_S_PETERS_MANTISTS) == 1) and (map_var(0) >= 9)) then begin
+        display_mstr(200 + (dude_gender == 1));
     end
 end
 
 procedure look_at_p_proc
 begin
     script_overrides;
-    display_msg(message_str(1450, 100));
+    display_mstr(100);
 end
 
 procedure description_p_proc
@@ -153,7 +153,7 @@ end
 procedure damage_p_proc
 begin
     if (source_obj == dude_obj) then begin
-        attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+        attack(dude_obj);
     end
 end
 
@@ -162,9 +162,9 @@ begin
     if (local_var(4) == 0) then begin
         set_local_var(4, tile_num(self_obj));
     end
-    critter_add_trait(self_obj, 1, 6, 5);
-    critter_add_trait(self_obj, 1, 5, 9);
-    metarule3(100, self_obj, 0, 0);
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 5);
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_MANTIS);
+    rm_fixed_timer_event(self_obj, 0);
     add_timer_event(self_obj, game_ticks(random(1, 5)), 0);
 end
 
@@ -181,7 +181,7 @@ begin
         end
         reg_anim_func(3, 0);
     end
-    metarule3(100, self_obj, 0, 0);
+    rm_fixed_timer_event(self_obj, 0);
     add_timer_event(self_obj, game_ticks(random(10, 15)), 0);
 end
 
@@ -191,7 +191,6 @@ begin
         script_overrides;
         rm_obj_from_inven(source_obj, obj_being_used_with);
         destroy_object(obj_being_used_with);
-        critter_dmg(self_obj, get_critter_stat(self_obj, 35) + 10, 6);
+        critter_dmg(self_obj, get_critter_stat(self_obj, STAT_current_hp) + 10, DMG_explosion);
     end
 end
-
