diff --git a/Mods/Resurrection/Original/SPATEBOX.ssl b/Mods/Resurrection/ReDefine/SPATEBOX.ssl
index 23ada2a..46a39d6 100644
--- a/Mods/Resurrection/Original/SPATEBOX.ssl
+++ b/Mods/Resurrection/ReDefine/SPATEBOX.ssl
@@ -53,12 +53,12 @@ begin
             return 1;
         end
     end
-    if ((party_member_obj(16777779) != 0) and ((global_var(910) bwand 32) == 0)) then begin
+    if ((party_member_obj(16777779) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 32) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777779)) <= 5) then begin
             return 1;
         end
     end
-    if ((party_member_obj(16777811) != 0) and ((global_var(910) bwand 64) == 0)) then begin
+    if ((party_member_obj(16777811) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 64) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777811)) <= 5) then begin
             return 1;
         end
@@ -73,7 +73,7 @@ begin
             return 1;
         end
     end
-    if ((party_member_obj(16777814) != 0) and ((global_var(910) bwand 128) == 0)) then begin
+    if ((party_member_obj(16777814) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 128) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777814)) <= 5) then begin
             return 1;
         end
@@ -131,7 +131,7 @@ end
 
 procedure map_enter_p_proc
 begin
-    if (metarule(22, 0) == 0) then begin
+    if (is_loading_game == 0) then begin
         if (1) then begin
             obj_close(self_obj);
         end
@@ -149,7 +149,7 @@ begin
             if (obj_is_carrying_obj_pid(self_obj, 49) < restock_amt) then begin
                 if (100 >= random(1, 100)) then begin
                     if (obj_is_carrying_obj_pid(self_obj, 49) < restock_amt) then begin
-                        restock_obj := create_object_sid(49, 0, 0, -1);
+                        restock_obj := create_object(49, 0, 0);
                         add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 49));
                     end
                     else begin
@@ -162,17 +162,17 @@ begin
                                 restock_obj := obj_carrying_pid_obj(self_obj, 49);
                                 if (obj_type(self_obj) == 1) then begin
                                     restock_obj := obj_carrying_pid_obj(self_obj, 49);
-                                    if (critter_inven_obj(self_obj, 0) == restock_obj) then begin
-                                        if (obj_item_subtype(critter_inven_obj(self_obj, 0)) == 0) then begin
-                                            restock_obj := critter_inven_obj(self_obj, 0);
+                                    if (critter_inven_obj(self_obj, INVEN_TYPE_WORN) == restock_obj) then begin
+                                        if (obj_item_subtype(critter_inven_obj(self_obj, INVEN_TYPE_WORN)) == 0) then begin
+                                            restock_obj := critter_inven_obj(self_obj, INVEN_TYPE_WORN);
                                             debug_msg("armour pid == " + obj_pid(restock_obj));
                                             rm_obj_from_inven(self_obj, restock_obj);
                                             add_obj_to_inven(self_obj, restock_obj);
                                         end
                                     end
                                     else begin
-                                        if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                            metarule(43, self_obj);
+                                        if ((critter_inven_obj(self_obj, INVEN_TYPE_LEFT_HAND) == restock_obj) or (critter_inven_obj(self_obj, INVEN_TYPE_RIGHT_HAND) == restock_obj)) then begin
+                                            inven_unwield(self_obj);
                                         end
                                     end
                                 end
@@ -201,7 +201,7 @@ begin
             end
             else begin
                 if (obj_is_carrying_obj_pid(self_obj, 49) < restock_amt) then begin
-                    restock_obj := create_object_sid(49, 0, 0, -1);
+                    restock_obj := create_object(49, 0, 0);
                     add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 49));
                 end
                 else begin
@@ -214,17 +214,17 @@ begin
                             restock_obj := obj_carrying_pid_obj(self_obj, 49);
                             if (obj_type(self_obj) == 1) then begin
                                 restock_obj := obj_carrying_pid_obj(self_obj, 49);
-                                if (critter_inven_obj(self_obj, 0) == restock_obj) then begin
-                                    if (obj_item_subtype(critter_inven_obj(self_obj, 0)) == 0) then begin
-                                        restock_obj := critter_inven_obj(self_obj, 0);
+                                if (critter_inven_obj(self_obj, INVEN_TYPE_WORN) == restock_obj) then begin
+                                    if (obj_item_subtype(critter_inven_obj(self_obj, INVEN_TYPE_WORN)) == 0) then begin
+                                        restock_obj := critter_inven_obj(self_obj, INVEN_TYPE_WORN);
                                         debug_msg("armour pid == " + obj_pid(restock_obj));
                                         rm_obj_from_inven(self_obj, restock_obj);
                                         add_obj_to_inven(self_obj, restock_obj);
                                     end
                                 end
                                 else begin
-                                    if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                        metarule(43, self_obj);
+                                    if ((critter_inven_obj(self_obj, INVEN_TYPE_LEFT_HAND) == restock_obj) or (critter_inven_obj(self_obj, INVEN_TYPE_RIGHT_HAND) == restock_obj)) then begin
+                                        inven_unwield(self_obj);
                                     end
                                 end
                             end
@@ -254,7 +254,7 @@ begin
             if (obj_is_carrying_obj_pid(self_obj, 40) < restock_amt) then begin
                 if (100 >= random(1, 100)) then begin
                     if (obj_is_carrying_obj_pid(self_obj, 40) < restock_amt) then begin
-                        restock_obj := create_object_sid(40, 0, 0, -1);
+                        restock_obj := create_object(40, 0, 0);
                         add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 40));
                     end
                     else begin
@@ -267,17 +267,17 @@ begin
                                 restock_obj := obj_carrying_pid_obj(self_obj, 40);
                                 if (obj_type(self_obj) == 1) then begin
                                     restock_obj := obj_carrying_pid_obj(self_obj, 40);
-                                    if (critter_inven_obj(self_obj, 0) == restock_obj) then begin
-                                        if (obj_item_subtype(critter_inven_obj(self_obj, 0)) == 0) then begin
-                                            restock_obj := critter_inven_obj(self_obj, 0);
+                                    if (critter_inven_obj(self_obj, INVEN_TYPE_WORN) == restock_obj) then begin
+                                        if (obj_item_subtype(critter_inven_obj(self_obj, INVEN_TYPE_WORN)) == 0) then begin
+                                            restock_obj := critter_inven_obj(self_obj, INVEN_TYPE_WORN);
                                             debug_msg("armour pid == " + obj_pid(restock_obj));
                                             rm_obj_from_inven(self_obj, restock_obj);
                                             add_obj_to_inven(self_obj, restock_obj);
                                         end
                                     end
                                     else begin
-                                        if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                            metarule(43, self_obj);
+                                        if ((critter_inven_obj(self_obj, INVEN_TYPE_LEFT_HAND) == restock_obj) or (critter_inven_obj(self_obj, INVEN_TYPE_RIGHT_HAND) == restock_obj)) then begin
+                                            inven_unwield(self_obj);
                                         end
                                     end
                                 end
@@ -306,7 +306,7 @@ begin
             end
             else begin
                 if (obj_is_carrying_obj_pid(self_obj, 40) < restock_amt) then begin
-                    restock_obj := create_object_sid(40, 0, 0, -1);
+                    restock_obj := create_object(40, 0, 0);
                     add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 40));
                 end
                 else begin
@@ -319,17 +319,17 @@ begin
                             restock_obj := obj_carrying_pid_obj(self_obj, 40);
                             if (obj_type(self_obj) == 1) then begin
                                 restock_obj := obj_carrying_pid_obj(self_obj, 40);
-                                if (critter_inven_obj(self_obj, 0) == restock_obj) then begin
-                                    if (obj_item_subtype(critter_inven_obj(self_obj, 0)) == 0) then begin
-                                        restock_obj := critter_inven_obj(self_obj, 0);
+                                if (critter_inven_obj(self_obj, INVEN_TYPE_WORN) == restock_obj) then begin
+                                    if (obj_item_subtype(critter_inven_obj(self_obj, INVEN_TYPE_WORN)) == 0) then begin
+                                        restock_obj := critter_inven_obj(self_obj, INVEN_TYPE_WORN);
                                         debug_msg("armour pid == " + obj_pid(restock_obj));
                                         rm_obj_from_inven(self_obj, restock_obj);
                                         add_obj_to_inven(self_obj, restock_obj);
                                     end
                                 end
                                 else begin
-                                    if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                        metarule(43, self_obj);
+                                    if ((critter_inven_obj(self_obj, INVEN_TYPE_LEFT_HAND) == restock_obj) or (critter_inven_obj(self_obj, INVEN_TYPE_RIGHT_HAND) == restock_obj)) then begin
+                                        inven_unwield(self_obj);
                                     end
                                 end
                             end
@@ -359,7 +359,7 @@ begin
             if (obj_is_carrying_obj_pid(self_obj, 87) < restock_amt) then begin
                 if (100 >= random(1, 100)) then begin
                     if (obj_is_carrying_obj_pid(self_obj, 87) < restock_amt) then begin
-                        restock_obj := create_object_sid(87, 0, 0, -1);
+                        restock_obj := create_object(87, 0, 0);
                         add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 87));
                     end
                     else begin
@@ -372,17 +372,17 @@ begin
                                 restock_obj := obj_carrying_pid_obj(self_obj, 87);
                                 if (obj_type(self_obj) == 1) then begin
                                     restock_obj := obj_carrying_pid_obj(self_obj, 87);
-                                    if (critter_inven_obj(self_obj, 0) == restock_obj) then begin
-                                        if (obj_item_subtype(critter_inven_obj(self_obj, 0)) == 0) then begin
-                                            restock_obj := critter_inven_obj(self_obj, 0);
+                                    if (critter_inven_obj(self_obj, INVEN_TYPE_WORN) == restock_obj) then begin
+                                        if (obj_item_subtype(critter_inven_obj(self_obj, INVEN_TYPE_WORN)) == 0) then begin
+                                            restock_obj := critter_inven_obj(self_obj, INVEN_TYPE_WORN);
                                             debug_msg("armour pid == " + obj_pid(restock_obj));
                                             rm_obj_from_inven(self_obj, restock_obj);
                                             add_obj_to_inven(self_obj, restock_obj);
                                         end
                                     end
                                     else begin
-                                        if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                            metarule(43, self_obj);
+                                        if ((critter_inven_obj(self_obj, INVEN_TYPE_LEFT_HAND) == restock_obj) or (critter_inven_obj(self_obj, INVEN_TYPE_RIGHT_HAND) == restock_obj)) then begin
+                                            inven_unwield(self_obj);
                                         end
                                     end
                                 end
@@ -411,7 +411,7 @@ begin
             end
             else begin
                 if (obj_is_carrying_obj_pid(self_obj, 87) < restock_amt) then begin
-                    restock_obj := create_object_sid(87, 0, 0, -1);
+                    restock_obj := create_object(87, 0, 0);
                     add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 87));
                 end
                 else begin
@@ -424,17 +424,17 @@ begin
                             restock_obj := obj_carrying_pid_obj(self_obj, 87);
                             if (obj_type(self_obj) == 1) then begin
                                 restock_obj := obj_carrying_pid_obj(self_obj, 87);
-                                if (critter_inven_obj(self_obj, 0) == restock_obj) then begin
-                                    if (obj_item_subtype(critter_inven_obj(self_obj, 0)) == 0) then begin
-                                        restock_obj := critter_inven_obj(self_obj, 0);
+                                if (critter_inven_obj(self_obj, INVEN_TYPE_WORN) == restock_obj) then begin
+                                    if (obj_item_subtype(critter_inven_obj(self_obj, INVEN_TYPE_WORN)) == 0) then begin
+                                        restock_obj := critter_inven_obj(self_obj, INVEN_TYPE_WORN);
                                         debug_msg("armour pid == " + obj_pid(restock_obj));
                                         rm_obj_from_inven(self_obj, restock_obj);
                                         add_obj_to_inven(self_obj, restock_obj);
                                     end
                                 end
                                 else begin
-                                    if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                        metarule(43, self_obj);
+                                    if ((critter_inven_obj(self_obj, INVEN_TYPE_LEFT_HAND) == restock_obj) or (critter_inven_obj(self_obj, INVEN_TYPE_RIGHT_HAND) == restock_obj)) then begin
+                                        inven_unwield(self_obj);
                                     end
                                 end
                             end
@@ -464,7 +464,7 @@ begin
             if (obj_is_carrying_obj_pid(self_obj, 53) < restock_amt) then begin
                 if (100 >= random(1, 100)) then begin
                     if (obj_is_carrying_obj_pid(self_obj, 53) < restock_amt) then begin
-                        restock_obj := create_object_sid(53, 0, 0, -1);
+                        restock_obj := create_object(53, 0, 0);
                         add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 53));
                     end
                     else begin
@@ -477,17 +477,17 @@ begin
                                 restock_obj := obj_carrying_pid_obj(self_obj, 53);
                                 if (obj_type(self_obj) == 1) then begin
                                     restock_obj := obj_carrying_pid_obj(self_obj, 53);
-                                    if (critter_inven_obj(self_obj, 0) == restock_obj) then begin
-                                        if (obj_item_subtype(critter_inven_obj(self_obj, 0)) == 0) then begin
-                                            restock_obj := critter_inven_obj(self_obj, 0);
+                                    if (critter_inven_obj(self_obj, INVEN_TYPE_WORN) == restock_obj) then begin
+                                        if (obj_item_subtype(critter_inven_obj(self_obj, INVEN_TYPE_WORN)) == 0) then begin
+                                            restock_obj := critter_inven_obj(self_obj, INVEN_TYPE_WORN);
                                             debug_msg("armour pid == " + obj_pid(restock_obj));
                                             rm_obj_from_inven(self_obj, restock_obj);
                                             add_obj_to_inven(self_obj, restock_obj);
                                         end
                                     end
                                     else begin
-                                        if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                            metarule(43, self_obj);
+                                        if ((critter_inven_obj(self_obj, INVEN_TYPE_LEFT_HAND) == restock_obj) or (critter_inven_obj(self_obj, INVEN_TYPE_RIGHT_HAND) == restock_obj)) then begin
+                                            inven_unwield(self_obj);
                                         end
                                     end
                                 end
@@ -516,7 +516,7 @@ begin
             end
             else begin
                 if (obj_is_carrying_obj_pid(self_obj, 53) < restock_amt) then begin
-                    restock_obj := create_object_sid(53, 0, 0, -1);
+                    restock_obj := create_object(53, 0, 0);
                     add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 53));
                 end
                 else begin
@@ -529,17 +529,17 @@ begin
                             restock_obj := obj_carrying_pid_obj(self_obj, 53);
                             if (obj_type(self_obj) == 1) then begin
                                 restock_obj := obj_carrying_pid_obj(self_obj, 53);
-                                if (critter_inven_obj(self_obj, 0) == restock_obj) then begin
-                                    if (obj_item_subtype(critter_inven_obj(self_obj, 0)) == 0) then begin
-                                        restock_obj := critter_inven_obj(self_obj, 0);
+                                if (critter_inven_obj(self_obj, INVEN_TYPE_WORN) == restock_obj) then begin
+                                    if (obj_item_subtype(critter_inven_obj(self_obj, INVEN_TYPE_WORN)) == 0) then begin
+                                        restock_obj := critter_inven_obj(self_obj, INVEN_TYPE_WORN);
                                         debug_msg("armour pid == " + obj_pid(restock_obj));
                                         rm_obj_from_inven(self_obj, restock_obj);
                                         add_obj_to_inven(self_obj, restock_obj);
                                     end
                                 end
                                 else begin
-                                    if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                        metarule(43, self_obj);
+                                    if ((critter_inven_obj(self_obj, INVEN_TYPE_LEFT_HAND) == restock_obj) or (critter_inven_obj(self_obj, INVEN_TYPE_RIGHT_HAND) == restock_obj)) then begin
+                                        inven_unwield(self_obj);
                                     end
                                 end
                             end
@@ -569,7 +569,7 @@ begin
             if (obj_is_carrying_obj_pid(self_obj, 110) < restock_amt) then begin
                 if (100 >= random(1, 100)) then begin
                     if (obj_is_carrying_obj_pid(self_obj, 110) < restock_amt) then begin
-                        restock_obj := create_object_sid(110, 0, 0, -1);
+                        restock_obj := create_object(110, 0, 0);
                         add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 110));
                     end
                     else begin
@@ -582,17 +582,17 @@ begin
                                 restock_obj := obj_carrying_pid_obj(self_obj, 110);
                                 if (obj_type(self_obj) == 1) then begin
                                     restock_obj := obj_carrying_pid_obj(self_obj, 110);
-                                    if (critter_inven_obj(self_obj, 0) == restock_obj) then begin
-                                        if (obj_item_subtype(critter_inven_obj(self_obj, 0)) == 0) then begin
-                                            restock_obj := critter_inven_obj(self_obj, 0);
+                                    if (critter_inven_obj(self_obj, INVEN_TYPE_WORN) == restock_obj) then begin
+                                        if (obj_item_subtype(critter_inven_obj(self_obj, INVEN_TYPE_WORN)) == 0) then begin
+                                            restock_obj := critter_inven_obj(self_obj, INVEN_TYPE_WORN);
                                             debug_msg("armour pid == " + obj_pid(restock_obj));
                                             rm_obj_from_inven(self_obj, restock_obj);
                                             add_obj_to_inven(self_obj, restock_obj);
                                         end
                                     end
                                     else begin
-                                        if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                            metarule(43, self_obj);
+                                        if ((critter_inven_obj(self_obj, INVEN_TYPE_LEFT_HAND) == restock_obj) or (critter_inven_obj(self_obj, INVEN_TYPE_RIGHT_HAND) == restock_obj)) then begin
+                                            inven_unwield(self_obj);
                                         end
                                     end
                                 end
@@ -621,7 +621,7 @@ begin
             end
             else begin
                 if (obj_is_carrying_obj_pid(self_obj, 110) < restock_amt) then begin
-                    restock_obj := create_object_sid(110, 0, 0, -1);
+                    restock_obj := create_object(110, 0, 0);
                     add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 110));
                 end
                 else begin
@@ -634,17 +634,17 @@ begin
                             restock_obj := obj_carrying_pid_obj(self_obj, 110);
                             if (obj_type(self_obj) == 1) then begin
                                 restock_obj := obj_carrying_pid_obj(self_obj, 110);
-                                if (critter_inven_obj(self_obj, 0) == restock_obj) then begin
-                                    if (obj_item_subtype(critter_inven_obj(self_obj, 0)) == 0) then begin
-                                        restock_obj := critter_inven_obj(self_obj, 0);
+                                if (critter_inven_obj(self_obj, INVEN_TYPE_WORN) == restock_obj) then begin
+                                    if (obj_item_subtype(critter_inven_obj(self_obj, INVEN_TYPE_WORN)) == 0) then begin
+                                        restock_obj := critter_inven_obj(self_obj, INVEN_TYPE_WORN);
                                         debug_msg("armour pid == " + obj_pid(restock_obj));
                                         rm_obj_from_inven(self_obj, restock_obj);
                                         add_obj_to_inven(self_obj, restock_obj);
                                     end
                                 end
                                 else begin
-                                    if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                        metarule(43, self_obj);
+                                    if ((critter_inven_obj(self_obj, INVEN_TYPE_LEFT_HAND) == restock_obj) or (critter_inven_obj(self_obj, INVEN_TYPE_RIGHT_HAND) == restock_obj)) then begin
+                                        inven_unwield(self_obj);
                                     end
                                 end
                             end
@@ -674,7 +674,7 @@ begin
             if (obj_is_carrying_obj_pid(self_obj, 109) < restock_amt) then begin
                 if (100 >= random(1, 100)) then begin
                     if (obj_is_carrying_obj_pid(self_obj, 109) < restock_amt) then begin
-                        restock_obj := create_object_sid(109, 0, 0, -1);
+                        restock_obj := create_object(109, 0, 0);
                         add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 109));
                     end
                     else begin
@@ -687,17 +687,17 @@ begin
                                 restock_obj := obj_carrying_pid_obj(self_obj, 109);
                                 if (obj_type(self_obj) == 1) then begin
                                     restock_obj := obj_carrying_pid_obj(self_obj, 109);
-                                    if (critter_inven_obj(self_obj, 0) == restock_obj) then begin
-                                        if (obj_item_subtype(critter_inven_obj(self_obj, 0)) == 0) then begin
-                                            restock_obj := critter_inven_obj(self_obj, 0);
+                                    if (critter_inven_obj(self_obj, INVEN_TYPE_WORN) == restock_obj) then begin
+                                        if (obj_item_subtype(critter_inven_obj(self_obj, INVEN_TYPE_WORN)) == 0) then begin
+                                            restock_obj := critter_inven_obj(self_obj, INVEN_TYPE_WORN);
                                             debug_msg("armour pid == " + obj_pid(restock_obj));
                                             rm_obj_from_inven(self_obj, restock_obj);
                                             add_obj_to_inven(self_obj, restock_obj);
                                         end
                                     end
                                     else begin
-                                        if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                            metarule(43, self_obj);
+                                        if ((critter_inven_obj(self_obj, INVEN_TYPE_LEFT_HAND) == restock_obj) or (critter_inven_obj(self_obj, INVEN_TYPE_RIGHT_HAND) == restock_obj)) then begin
+                                            inven_unwield(self_obj);
                                         end
                                     end
                                 end
@@ -726,7 +726,7 @@ begin
             end
             else begin
                 if (obj_is_carrying_obj_pid(self_obj, 109) < restock_amt) then begin
-                    restock_obj := create_object_sid(109, 0, 0, -1);
+                    restock_obj := create_object(109, 0, 0);
                     add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 109));
                 end
                 else begin
@@ -739,17 +739,17 @@ begin
                             restock_obj := obj_carrying_pid_obj(self_obj, 109);
                             if (obj_type(self_obj) == 1) then begin
                                 restock_obj := obj_carrying_pid_obj(self_obj, 109);
-                                if (critter_inven_obj(self_obj, 0) == restock_obj) then begin
-                                    if (obj_item_subtype(critter_inven_obj(self_obj, 0)) == 0) then begin
-                                        restock_obj := critter_inven_obj(self_obj, 0);
+                                if (critter_inven_obj(self_obj, INVEN_TYPE_WORN) == restock_obj) then begin
+                                    if (obj_item_subtype(critter_inven_obj(self_obj, INVEN_TYPE_WORN)) == 0) then begin
+                                        restock_obj := critter_inven_obj(self_obj, INVEN_TYPE_WORN);
                                         debug_msg("armour pid == " + obj_pid(restock_obj));
                                         rm_obj_from_inven(self_obj, restock_obj);
                                         add_obj_to_inven(self_obj, restock_obj);
                                     end
                                 end
                                 else begin
-                                    if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                        metarule(43, self_obj);
+                                    if ((critter_inven_obj(self_obj, INVEN_TYPE_LEFT_HAND) == restock_obj) or (critter_inven_obj(self_obj, INVEN_TYPE_RIGHT_HAND) == restock_obj)) then begin
+                                        inven_unwield(self_obj);
                                     end
                                 end
                             end
@@ -779,7 +779,7 @@ begin
             if (obj_is_carrying_obj_pid(self_obj, 48) < restock_amt) then begin
                 if (100 >= random(1, 100)) then begin
                     if (obj_is_carrying_obj_pid(self_obj, 48) < restock_amt) then begin
-                        restock_obj := create_object_sid(48, 0, 0, -1);
+                        restock_obj := create_object(48, 0, 0);
                         add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 48));
                     end
                     else begin
@@ -792,17 +792,17 @@ begin
                                 restock_obj := obj_carrying_pid_obj(self_obj, 48);
                                 if (obj_type(self_obj) == 1) then begin
                                     restock_obj := obj_carrying_pid_obj(self_obj, 48);
-                                    if (critter_inven_obj(self_obj, 0) == restock_obj) then begin
-                                        if (obj_item_subtype(critter_inven_obj(self_obj, 0)) == 0) then begin
-                                            restock_obj := critter_inven_obj(self_obj, 0);
+                                    if (critter_inven_obj(self_obj, INVEN_TYPE_WORN) == restock_obj) then begin
+                                        if (obj_item_subtype(critter_inven_obj(self_obj, INVEN_TYPE_WORN)) == 0) then begin
+                                            restock_obj := critter_inven_obj(self_obj, INVEN_TYPE_WORN);
                                             debug_msg("armour pid == " + obj_pid(restock_obj));
                                             rm_obj_from_inven(self_obj, restock_obj);
                                             add_obj_to_inven(self_obj, restock_obj);
                                         end
                                     end
                                     else begin
-                                        if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                            metarule(43, self_obj);
+                                        if ((critter_inven_obj(self_obj, INVEN_TYPE_LEFT_HAND) == restock_obj) or (critter_inven_obj(self_obj, INVEN_TYPE_RIGHT_HAND) == restock_obj)) then begin
+                                            inven_unwield(self_obj);
                                         end
                                     end
                                 end
@@ -831,7 +831,7 @@ begin
             end
             else begin
                 if (obj_is_carrying_obj_pid(self_obj, 48) < restock_amt) then begin
-                    restock_obj := create_object_sid(48, 0, 0, -1);
+                    restock_obj := create_object(48, 0, 0);
                     add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 48));
                 end
                 else begin
@@ -844,17 +844,17 @@ begin
                             restock_obj := obj_carrying_pid_obj(self_obj, 48);
                             if (obj_type(self_obj) == 1) then begin
                                 restock_obj := obj_carrying_pid_obj(self_obj, 48);
-                                if (critter_inven_obj(self_obj, 0) == restock_obj) then begin
-                                    if (obj_item_subtype(critter_inven_obj(self_obj, 0)) == 0) then begin
-                                        restock_obj := critter_inven_obj(self_obj, 0);
+                                if (critter_inven_obj(self_obj, INVEN_TYPE_WORN) == restock_obj) then begin
+                                    if (obj_item_subtype(critter_inven_obj(self_obj, INVEN_TYPE_WORN)) == 0) then begin
+                                        restock_obj := critter_inven_obj(self_obj, INVEN_TYPE_WORN);
                                         debug_msg("armour pid == " + obj_pid(restock_obj));
                                         rm_obj_from_inven(self_obj, restock_obj);
                                         add_obj_to_inven(self_obj, restock_obj);
                                     end
                                 end
                                 else begin
-                                    if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                        metarule(43, self_obj);
+                                    if ((critter_inven_obj(self_obj, INVEN_TYPE_LEFT_HAND) == restock_obj) or (critter_inven_obj(self_obj, INVEN_TYPE_RIGHT_HAND) == restock_obj)) then begin
+                                        inven_unwield(self_obj);
                                     end
                                 end
                             end
@@ -880,11 +880,11 @@ begin
                     end
                 end
             end
-            restock_amt := random(1, 1);
+            restock_amt := 1;
             if (obj_is_carrying_obj_pid(self_obj, 80) < restock_amt) then begin
                 if (5 >= random(1, 100)) then begin
                     if (obj_is_carrying_obj_pid(self_obj, 80) < restock_amt) then begin
-                        restock_obj := create_object_sid(80, 0, 0, -1);
+                        restock_obj := create_object(80, 0, 0);
                         add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 80));
                     end
                     else begin
@@ -897,17 +897,17 @@ begin
                                 restock_obj := obj_carrying_pid_obj(self_obj, 80);
                                 if (obj_type(self_obj) == 1) then begin
                                     restock_obj := obj_carrying_pid_obj(self_obj, 80);
-                                    if (critter_inven_obj(self_obj, 0) == restock_obj) then begin
-                                        if (obj_item_subtype(critter_inven_obj(self_obj, 0)) == 0) then begin
-                                            restock_obj := critter_inven_obj(self_obj, 0);
+                                    if (critter_inven_obj(self_obj, INVEN_TYPE_WORN) == restock_obj) then begin
+                                        if (obj_item_subtype(critter_inven_obj(self_obj, INVEN_TYPE_WORN)) == 0) then begin
+                                            restock_obj := critter_inven_obj(self_obj, INVEN_TYPE_WORN);
                                             debug_msg("armour pid == " + obj_pid(restock_obj));
                                             rm_obj_from_inven(self_obj, restock_obj);
                                             add_obj_to_inven(self_obj, restock_obj);
                                         end
                                     end
                                     else begin
-                                        if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                            metarule(43, self_obj);
+                                        if ((critter_inven_obj(self_obj, INVEN_TYPE_LEFT_HAND) == restock_obj) or (critter_inven_obj(self_obj, INVEN_TYPE_RIGHT_HAND) == restock_obj)) then begin
+                                            inven_unwield(self_obj);
                                         end
                                     end
                                 end
@@ -936,7 +936,7 @@ begin
             end
             else begin
                 if (obj_is_carrying_obj_pid(self_obj, 80) < restock_amt) then begin
-                    restock_obj := create_object_sid(80, 0, 0, -1);
+                    restock_obj := create_object(80, 0, 0);
                     add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 80));
                 end
                 else begin
@@ -949,17 +949,17 @@ begin
                             restock_obj := obj_carrying_pid_obj(self_obj, 80);
                             if (obj_type(self_obj) == 1) then begin
                                 restock_obj := obj_carrying_pid_obj(self_obj, 80);
-                                if (critter_inven_obj(self_obj, 0) == restock_obj) then begin
-                                    if (obj_item_subtype(critter_inven_obj(self_obj, 0)) == 0) then begin
-                                        restock_obj := critter_inven_obj(self_obj, 0);
+                                if (critter_inven_obj(self_obj, INVEN_TYPE_WORN) == restock_obj) then begin
+                                    if (obj_item_subtype(critter_inven_obj(self_obj, INVEN_TYPE_WORN)) == 0) then begin
+                                        restock_obj := critter_inven_obj(self_obj, INVEN_TYPE_WORN);
                                         debug_msg("armour pid == " + obj_pid(restock_obj));
                                         rm_obj_from_inven(self_obj, restock_obj);
                                         add_obj_to_inven(self_obj, restock_obj);
                                     end
                                 end
                                 else begin
-                                    if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                        metarule(43, self_obj);
+                                    if ((critter_inven_obj(self_obj, INVEN_TYPE_LEFT_HAND) == restock_obj) or (critter_inven_obj(self_obj, INVEN_TYPE_RIGHT_HAND) == restock_obj)) then begin
+                                        inven_unwield(self_obj);
                                     end
                                 end
                             end
@@ -985,11 +985,11 @@ begin
                     end
                 end
             end
-            restock_amt := random(1, 1);
+            restock_amt := 1;
             if (obj_is_carrying_obj_pid(self_obj, 47) < restock_amt) then begin
                 if (40 >= random(1, 100)) then begin
                     if (obj_is_carrying_obj_pid(self_obj, 47) < restock_amt) then begin
-                        restock_obj := create_object_sid(47, 0, 0, -1);
+                        restock_obj := create_object(47, 0, 0);
                         add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 47));
                     end
                     else begin
@@ -1002,17 +1002,17 @@ begin
                                 restock_obj := obj_carrying_pid_obj(self_obj, 47);
                                 if (obj_type(self_obj) == 1) then begin
                                     restock_obj := obj_carrying_pid_obj(self_obj, 47);
-                                    if (critter_inven_obj(self_obj, 0) == restock_obj) then begin
-                                        if (obj_item_subtype(critter_inven_obj(self_obj, 0)) == 0) then begin
-                                            restock_obj := critter_inven_obj(self_obj, 0);
+                                    if (critter_inven_obj(self_obj, INVEN_TYPE_WORN) == restock_obj) then begin
+                                        if (obj_item_subtype(critter_inven_obj(self_obj, INVEN_TYPE_WORN)) == 0) then begin
+                                            restock_obj := critter_inven_obj(self_obj, INVEN_TYPE_WORN);
                                             debug_msg("armour pid == " + obj_pid(restock_obj));
                                             rm_obj_from_inven(self_obj, restock_obj);
                                             add_obj_to_inven(self_obj, restock_obj);
                                         end
                                     end
                                     else begin
-                                        if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                            metarule(43, self_obj);
+                                        if ((critter_inven_obj(self_obj, INVEN_TYPE_LEFT_HAND) == restock_obj) or (critter_inven_obj(self_obj, INVEN_TYPE_RIGHT_HAND) == restock_obj)) then begin
+                                            inven_unwield(self_obj);
                                         end
                                     end
                                 end
@@ -1041,7 +1041,7 @@ begin
             end
             else begin
                 if (obj_is_carrying_obj_pid(self_obj, 47) < restock_amt) then begin
-                    restock_obj := create_object_sid(47, 0, 0, -1);
+                    restock_obj := create_object(47, 0, 0);
                     add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 47));
                 end
                 else begin
@@ -1054,17 +1054,17 @@ begin
                             restock_obj := obj_carrying_pid_obj(self_obj, 47);
                             if (obj_type(self_obj) == 1) then begin
                                 restock_obj := obj_carrying_pid_obj(self_obj, 47);
-                                if (critter_inven_obj(self_obj, 0) == restock_obj) then begin
-                                    if (obj_item_subtype(critter_inven_obj(self_obj, 0)) == 0) then begin
-                                        restock_obj := critter_inven_obj(self_obj, 0);
+                                if (critter_inven_obj(self_obj, INVEN_TYPE_WORN) == restock_obj) then begin
+                                    if (obj_item_subtype(critter_inven_obj(self_obj, INVEN_TYPE_WORN)) == 0) then begin
+                                        restock_obj := critter_inven_obj(self_obj, INVEN_TYPE_WORN);
                                         debug_msg("armour pid == " + obj_pid(restock_obj));
                                         rm_obj_from_inven(self_obj, restock_obj);
                                         add_obj_to_inven(self_obj, restock_obj);
                                     end
                                 end
                                 else begin
-                                    if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                        metarule(43, self_obj);
+                                    if ((critter_inven_obj(self_obj, INVEN_TYPE_LEFT_HAND) == restock_obj) or (critter_inven_obj(self_obj, INVEN_TYPE_RIGHT_HAND) == restock_obj)) then begin
+                                        inven_unwield(self_obj);
                                     end
                                 end
                             end
@@ -1090,11 +1090,11 @@ begin
                     end
                 end
             end
-            restock_amt := random(1, 1);
+            restock_amt := 1;
             if (obj_is_carrying_obj_pid(self_obj, 117) < restock_amt) then begin
                 if (10 >= random(1, 100)) then begin
                     if (obj_is_carrying_obj_pid(self_obj, 117) < restock_amt) then begin
-                        restock_obj := create_object_sid(117, 0, 0, -1);
+                        restock_obj := create_object(117, 0, 0);
                         add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 117));
                     end
                     else begin
@@ -1107,17 +1107,17 @@ begin
                                 restock_obj := obj_carrying_pid_obj(self_obj, 117);
                                 if (obj_type(self_obj) == 1) then begin
                                     restock_obj := obj_carrying_pid_obj(self_obj, 117);
-                                    if (critter_inven_obj(self_obj, 0) == restock_obj) then begin
-                                        if (obj_item_subtype(critter_inven_obj(self_obj, 0)) == 0) then begin
-                                            restock_obj := critter_inven_obj(self_obj, 0);
+                                    if (critter_inven_obj(self_obj, INVEN_TYPE_WORN) == restock_obj) then begin
+                                        if (obj_item_subtype(critter_inven_obj(self_obj, INVEN_TYPE_WORN)) == 0) then begin
+                                            restock_obj := critter_inven_obj(self_obj, INVEN_TYPE_WORN);
                                             debug_msg("armour pid == " + obj_pid(restock_obj));
                                             rm_obj_from_inven(self_obj, restock_obj);
                                             add_obj_to_inven(self_obj, restock_obj);
                                         end
                                     end
                                     else begin
-                                        if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                            metarule(43, self_obj);
+                                        if ((critter_inven_obj(self_obj, INVEN_TYPE_LEFT_HAND) == restock_obj) or (critter_inven_obj(self_obj, INVEN_TYPE_RIGHT_HAND) == restock_obj)) then begin
+                                            inven_unwield(self_obj);
                                         end
                                     end
                                 end
@@ -1146,7 +1146,7 @@ begin
             end
             else begin
                 if (obj_is_carrying_obj_pid(self_obj, 117) < restock_amt) then begin
-                    restock_obj := create_object_sid(117, 0, 0, -1);
+                    restock_obj := create_object(117, 0, 0);
                     add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 117));
                 end
                 else begin
@@ -1159,17 +1159,17 @@ begin
                             restock_obj := obj_carrying_pid_obj(self_obj, 117);
                             if (obj_type(self_obj) == 1) then begin
                                 restock_obj := obj_carrying_pid_obj(self_obj, 117);
-                                if (critter_inven_obj(self_obj, 0) == restock_obj) then begin
-                                    if (obj_item_subtype(critter_inven_obj(self_obj, 0)) == 0) then begin
-                                        restock_obj := critter_inven_obj(self_obj, 0);
+                                if (critter_inven_obj(self_obj, INVEN_TYPE_WORN) == restock_obj) then begin
+                                    if (obj_item_subtype(critter_inven_obj(self_obj, INVEN_TYPE_WORN)) == 0) then begin
+                                        restock_obj := critter_inven_obj(self_obj, INVEN_TYPE_WORN);
                                         debug_msg("armour pid == " + obj_pid(restock_obj));
                                         rm_obj_from_inven(self_obj, restock_obj);
                                         add_obj_to_inven(self_obj, restock_obj);
                                     end
                                 end
                                 else begin
-                                    if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                        metarule(43, self_obj);
+                                    if ((critter_inven_obj(self_obj, INVEN_TYPE_LEFT_HAND) == restock_obj) or (critter_inven_obj(self_obj, INVEN_TYPE_RIGHT_HAND) == restock_obj)) then begin
+                                        inven_unwield(self_obj);
                                     end
                                 end
                             end
@@ -1195,11 +1195,11 @@ begin
                     end
                 end
             end
-            restock_amt := random(1, 1);
+            restock_amt := 1;
             if (obj_is_carrying_obj_pid(self_obj, 272) < restock_amt) then begin
                 if (10 >= random(1, 100)) then begin
                     if (obj_is_carrying_obj_pid(self_obj, 272) < restock_amt) then begin
-                        restock_obj := create_object_sid(272, 0, 0, -1);
+                        restock_obj := create_object(272, 0, 0);
                         add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 272));
                     end
                     else begin
@@ -1212,17 +1212,17 @@ begin
                                 restock_obj := obj_carrying_pid_obj(self_obj, 272);
                                 if (obj_type(self_obj) == 1) then begin
                                     restock_obj := obj_carrying_pid_obj(self_obj, 272);
-                                    if (critter_inven_obj(self_obj, 0) == restock_obj) then begin
-                                        if (obj_item_subtype(critter_inven_obj(self_obj, 0)) == 0) then begin
-                                            restock_obj := critter_inven_obj(self_obj, 0);
+                                    if (critter_inven_obj(self_obj, INVEN_TYPE_WORN) == restock_obj) then begin
+                                        if (obj_item_subtype(critter_inven_obj(self_obj, INVEN_TYPE_WORN)) == 0) then begin
+                                            restock_obj := critter_inven_obj(self_obj, INVEN_TYPE_WORN);
                                             debug_msg("armour pid == " + obj_pid(restock_obj));
                                             rm_obj_from_inven(self_obj, restock_obj);
                                             add_obj_to_inven(self_obj, restock_obj);
                                         end
                                     end
                                     else begin
-                                        if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                            metarule(43, self_obj);
+                                        if ((critter_inven_obj(self_obj, INVEN_TYPE_LEFT_HAND) == restock_obj) or (critter_inven_obj(self_obj, INVEN_TYPE_RIGHT_HAND) == restock_obj)) then begin
+                                            inven_unwield(self_obj);
                                         end
                                     end
                                 end
@@ -1251,7 +1251,7 @@ begin
             end
             else begin
                 if (obj_is_carrying_obj_pid(self_obj, 272) < restock_amt) then begin
-                    restock_obj := create_object_sid(272, 0, 0, -1);
+                    restock_obj := create_object(272, 0, 0);
                     add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 272));
                 end
                 else begin
@@ -1264,17 +1264,17 @@ begin
                             restock_obj := obj_carrying_pid_obj(self_obj, 272);
                             if (obj_type(self_obj) == 1) then begin
                                 restock_obj := obj_carrying_pid_obj(self_obj, 272);
-                                if (critter_inven_obj(self_obj, 0) == restock_obj) then begin
-                                    if (obj_item_subtype(critter_inven_obj(self_obj, 0)) == 0) then begin
-                                        restock_obj := critter_inven_obj(self_obj, 0);
+                                if (critter_inven_obj(self_obj, INVEN_TYPE_WORN) == restock_obj) then begin
+                                    if (obj_item_subtype(critter_inven_obj(self_obj, INVEN_TYPE_WORN)) == 0) then begin
+                                        restock_obj := critter_inven_obj(self_obj, INVEN_TYPE_WORN);
                                         debug_msg("armour pid == " + obj_pid(restock_obj));
                                         rm_obj_from_inven(self_obj, restock_obj);
                                         add_obj_to_inven(self_obj, restock_obj);
                                     end
                                 end
                                 else begin
-                                    if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                        metarule(43, self_obj);
+                                    if ((critter_inven_obj(self_obj, INVEN_TYPE_LEFT_HAND) == restock_obj) or (critter_inven_obj(self_obj, INVEN_TYPE_RIGHT_HAND) == restock_obj)) then begin
+                                        inven_unwield(self_obj);
                                     end
                                 end
                             end
@@ -1300,11 +1300,11 @@ begin
                     end
                 end
             end
-            restock_amt := random(1, 1);
+            restock_amt := 1;
             if (obj_is_carrying_obj_pid(self_obj, 271) < restock_amt) then begin
                 if (10 >= random(1, 100)) then begin
                     if (obj_is_carrying_obj_pid(self_obj, 271) < restock_amt) then begin
-                        restock_obj := create_object_sid(271, 0, 0, -1);
+                        restock_obj := create_object(271, 0, 0);
                         add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 271));
                     end
                     else begin
@@ -1317,17 +1317,17 @@ begin
                                 restock_obj := obj_carrying_pid_obj(self_obj, 271);
                                 if (obj_type(self_obj) == 1) then begin
                                     restock_obj := obj_carrying_pid_obj(self_obj, 271);
-                                    if (critter_inven_obj(self_obj, 0) == restock_obj) then begin
-                                        if (obj_item_subtype(critter_inven_obj(self_obj, 0)) == 0) then begin
-                                            restock_obj := critter_inven_obj(self_obj, 0);
+                                    if (critter_inven_obj(self_obj, INVEN_TYPE_WORN) == restock_obj) then begin
+                                        if (obj_item_subtype(critter_inven_obj(self_obj, INVEN_TYPE_WORN)) == 0) then begin
+                                            restock_obj := critter_inven_obj(self_obj, INVEN_TYPE_WORN);
                                             debug_msg("armour pid == " + obj_pid(restock_obj));
                                             rm_obj_from_inven(self_obj, restock_obj);
                                             add_obj_to_inven(self_obj, restock_obj);
                                         end
                                     end
                                     else begin
-                                        if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                            metarule(43, self_obj);
+                                        if ((critter_inven_obj(self_obj, INVEN_TYPE_LEFT_HAND) == restock_obj) or (critter_inven_obj(self_obj, INVEN_TYPE_RIGHT_HAND) == restock_obj)) then begin
+                                            inven_unwield(self_obj);
                                         end
                                     end
                                 end
@@ -1356,7 +1356,7 @@ begin
             end
             else begin
                 if (obj_is_carrying_obj_pid(self_obj, 271) < restock_amt) then begin
-                    restock_obj := create_object_sid(271, 0, 0, -1);
+                    restock_obj := create_object(271, 0, 0);
                     add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 271));
                 end
                 else begin
@@ -1369,17 +1369,17 @@ begin
                             restock_obj := obj_carrying_pid_obj(self_obj, 271);
                             if (obj_type(self_obj) == 1) then begin
                                 restock_obj := obj_carrying_pid_obj(self_obj, 271);
-                                if (critter_inven_obj(self_obj, 0) == restock_obj) then begin
-                                    if (obj_item_subtype(critter_inven_obj(self_obj, 0)) == 0) then begin
-                                        restock_obj := critter_inven_obj(self_obj, 0);
+                                if (critter_inven_obj(self_obj, INVEN_TYPE_WORN) == restock_obj) then begin
+                                    if (obj_item_subtype(critter_inven_obj(self_obj, INVEN_TYPE_WORN)) == 0) then begin
+                                        restock_obj := critter_inven_obj(self_obj, INVEN_TYPE_WORN);
                                         debug_msg("armour pid == " + obj_pid(restock_obj));
                                         rm_obj_from_inven(self_obj, restock_obj);
                                         add_obj_to_inven(self_obj, restock_obj);
                                     end
                                 end
                                 else begin
-                                    if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                        metarule(43, self_obj);
+                                    if ((critter_inven_obj(self_obj, INVEN_TYPE_LEFT_HAND) == restock_obj) or (critter_inven_obj(self_obj, INVEN_TYPE_RIGHT_HAND) == restock_obj)) then begin
+                                        inven_unwield(self_obj);
                                     end
                                 end
                             end
@@ -1409,7 +1409,7 @@ begin
             if (obj_is_carrying_obj_pid(self_obj, 318) < restock_amt) then begin
                 if (60 >= random(1, 100)) then begin
                     if (obj_is_carrying_obj_pid(self_obj, 318) < restock_amt) then begin
-                        restock_obj := create_object_sid(318, 0, 0, -1);
+                        restock_obj := create_object(318, 0, 0);
                         add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 318));
                     end
                     else begin
@@ -1422,17 +1422,17 @@ begin
                                 restock_obj := obj_carrying_pid_obj(self_obj, 318);
                                 if (obj_type(self_obj) == 1) then begin
                                     restock_obj := obj_carrying_pid_obj(self_obj, 318);
-                                    if (critter_inven_obj(self_obj, 0) == restock_obj) then begin
-                                        if (obj_item_subtype(critter_inven_obj(self_obj, 0)) == 0) then begin
-                                            restock_obj := critter_inven_obj(self_obj, 0);
+                                    if (critter_inven_obj(self_obj, INVEN_TYPE_WORN) == restock_obj) then begin
+                                        if (obj_item_subtype(critter_inven_obj(self_obj, INVEN_TYPE_WORN)) == 0) then begin
+                                            restock_obj := critter_inven_obj(self_obj, INVEN_TYPE_WORN);
                                             debug_msg("armour pid == " + obj_pid(restock_obj));
                                             rm_obj_from_inven(self_obj, restock_obj);
                                             add_obj_to_inven(self_obj, restock_obj);
                                         end
                                     end
                                     else begin
-                                        if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                            metarule(43, self_obj);
+                                        if ((critter_inven_obj(self_obj, INVEN_TYPE_LEFT_HAND) == restock_obj) or (critter_inven_obj(self_obj, INVEN_TYPE_RIGHT_HAND) == restock_obj)) then begin
+                                            inven_unwield(self_obj);
                                         end
                                     end
                                 end
@@ -1461,7 +1461,7 @@ begin
             end
             else begin
                 if (obj_is_carrying_obj_pid(self_obj, 318) < restock_amt) then begin
-                    restock_obj := create_object_sid(318, 0, 0, -1);
+                    restock_obj := create_object(318, 0, 0);
                     add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 318));
                 end
                 else begin
@@ -1474,17 +1474,17 @@ begin
                             restock_obj := obj_carrying_pid_obj(self_obj, 318);
                             if (obj_type(self_obj) == 1) then begin
                                 restock_obj := obj_carrying_pid_obj(self_obj, 318);
-                                if (critter_inven_obj(self_obj, 0) == restock_obj) then begin
-                                    if (obj_item_subtype(critter_inven_obj(self_obj, 0)) == 0) then begin
-                                        restock_obj := critter_inven_obj(self_obj, 0);
+                                if (critter_inven_obj(self_obj, INVEN_TYPE_WORN) == restock_obj) then begin
+                                    if (obj_item_subtype(critter_inven_obj(self_obj, INVEN_TYPE_WORN)) == 0) then begin
+                                        restock_obj := critter_inven_obj(self_obj, INVEN_TYPE_WORN);
                                         debug_msg("armour pid == " + obj_pid(restock_obj));
                                         rm_obj_from_inven(self_obj, restock_obj);
                                         add_obj_to_inven(self_obj, restock_obj);
                                     end
                                 end
                                 else begin
-                                    if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                        metarule(43, self_obj);
+                                    if ((critter_inven_obj(self_obj, INVEN_TYPE_LEFT_HAND) == restock_obj) or (critter_inven_obj(self_obj, INVEN_TYPE_RIGHT_HAND) == restock_obj)) then begin
+                                        inven_unwield(self_obj);
                                     end
                                 end
                             end
@@ -1535,7 +1535,7 @@ begin
             if (caught_tamper == 0) then begin
                 if (1) then begin
                     if (0 != 0) then begin
-                        if (has_trait(1, 0, 666) and ((critter_state(0) bwand 1) == 0)) then begin
+                        if (obj_is_visible_flag(0) and ((critter_state(0) bwand 1) == 0)) then begin
                             if (obj_can_see_obj(0, user_obj)) then begin
                                 caught_tamper := 1;
                             end
@@ -1550,7 +1550,7 @@ begin
                         attack_setup(0, user_obj);
                     end
                     else begin
-                        float_msg(0, message_str(14, 5400), 8);
+                        float_msg(0, g_mstr(5400), FLOAT_MSG_YELLOW);
                     end
                     set_local_var(0, local_var(0) bwor 1024);
                 end
@@ -1563,13 +1563,13 @@ procedure damage_p_proc
 begin
     if (obj_is_locked(self_obj)) then begin
         if (target_obj != 0) then begin
-            if (metarule(49, target_obj) == 6) then begin
+            if (weapon_dmg_type(target_obj) == 6) then begin
                 if (1) then begin
-                    display_msg(message_str(1189, 300));
+                    display_msg(message_str(SCRIPT_ZILOCKER, 300));
                     obj_unlock(self_obj);
                 end
                 else begin
-                    display_msg(message_str(1189, 301));
+                    display_msg(message_str(SCRIPT_ZILOCKER, 301));
                 end
             end
         end
@@ -1582,7 +1582,7 @@ begin
                 attack_setup(0, user_obj);
             end
             else begin
-                float_msg(0, message_str(14, 5400), 8);
+                float_msg(0, g_mstr(5400), FLOAT_MSG_YELLOW);
             end
             set_local_var(0, local_var(0) bwor 1024);
         end
@@ -1620,7 +1620,7 @@ begin
             if (caught_tamper == 0) then begin
                 if (1) then begin
                     if (0 != 0) then begin
-                        if (has_trait(1, 0, 666) and ((critter_state(0) bwand 1) == 0)) then begin
+                        if (obj_is_visible_flag(0) and ((critter_state(0) bwand 1) == 0)) then begin
                             if (obj_can_see_obj(0, user_obj)) then begin
                                 caught_tamper := 1;
                             end
@@ -1635,7 +1635,7 @@ begin
                         attack_setup(0, user_obj);
                     end
                     else begin
-                        float_msg(0, message_str(14, 5400), 8);
+                        float_msg(0, g_mstr(5400), FLOAT_MSG_YELLOW);
                     end
                     set_local_var(0, local_var(0) bwor 1024);
                 end
@@ -1648,19 +1648,19 @@ begin
                             if (is_success(roll_vs_skill(user_obj, 9, -75 + 20))) then begin
                                 script_overrides;
                                 if (user_obj == dude_obj) then begin
-                                    display_msg(message_str(1189, 100));
+                                    display_msg(message_str(SCRIPT_ZILOCKER, 100));
                                     if (((local_var(0) bwand 512) != 0) == 0) then begin
                                         give_exp_points(25);
-                                        display_msg(message_str(14, 100) + 25 + message_str(14, 101));
+                                        display_msg(g_mstr(100) + 25 + g_mstr(101));
                                     end
                                     set_local_var(0, local_var(0) bwor 512);
                                 end
                                 else begin
                                     if (user_obj != 0) then begin
-                                        display_msg(obj_name(user_obj) + message_str(1189, 200));
+                                        display_msg(obj_name(user_obj) + message_str(SCRIPT_ZILOCKER, 200));
                                         if (((local_var(0) bwand 512) != 0) == 0) then begin
                                             give_exp_points(25);
-                                            display_msg(message_str(14, 100) + 25 + message_str(14, 101));
+                                            display_msg(g_mstr(100) + 25 + g_mstr(101));
                                         end
                                         set_local_var(0, local_var(0) bwor 512);
                                     end
@@ -1671,11 +1671,11 @@ begin
                             else begin
                                 script_overrides;
                                 if (user_obj == dude_obj) then begin
-                                    display_msg(message_str(1189, 101));
+                                    display_msg(message_str(SCRIPT_ZILOCKER, 101));
                                 end
                                 else begin
                                     if (user_obj != 0) then begin
-                                        display_msg(obj_name(user_obj) + message_str(1189, 201));
+                                        display_msg(obj_name(user_obj) + message_str(SCRIPT_ZILOCKER, 201));
                                     end
                                 end
                                 debug_msg("unlock_fail_msg");
@@ -1689,19 +1689,19 @@ begin
                                 if (is_success(roll_vs_skill(user_obj, 9, -75 + 20))) then begin
                                     script_overrides;
                                     if (user_obj == dude_obj) then begin
-                                        display_msg(message_str(1189, 100));
+                                        display_msg(message_str(SCRIPT_ZILOCKER, 100));
                                         if (((local_var(0) bwand 512) != 0) == 0) then begin
                                             give_exp_points(25);
-                                            display_msg(message_str(14, 100) + 25 + message_str(14, 101));
+                                            display_msg(g_mstr(100) + 25 + g_mstr(101));
                                         end
                                         set_local_var(0, local_var(0) bwor 512);
                                     end
                                     else begin
                                         if (user_obj != 0) then begin
-                                            display_msg(obj_name(user_obj) + message_str(1189, 200));
+                                            display_msg(obj_name(user_obj) + message_str(SCRIPT_ZILOCKER, 200));
                                             if (((local_var(0) bwand 512) != 0) == 0) then begin
                                                 give_exp_points(25);
-                                                display_msg(message_str(14, 100) + 25 + message_str(14, 101));
+                                                display_msg(g_mstr(100) + 25 + g_mstr(101));
                                             end
                                             set_local_var(0, local_var(0) bwor 512);
                                         end
@@ -1712,11 +1712,11 @@ begin
                                 else begin
                                     script_overrides;
                                     if (user_obj == dude_obj) then begin
-                                        display_msg(message_str(1189, 101));
+                                        display_msg(message_str(SCRIPT_ZILOCKER, 101));
                                     end
                                     else begin
                                         if (user_obj != 0) then begin
-                                            display_msg(obj_name(user_obj) + message_str(1189, 201));
+                                            display_msg(obj_name(user_obj) + message_str(SCRIPT_ZILOCKER, 201));
                                         end
                                     end
                                     debug_msg("unlock_fail_msg");
@@ -1728,19 +1728,19 @@ begin
                                 script_overrides;
                                 if (is_success(do_check(user_obj, 0, 0))) then begin
                                     if (user_obj == dude_obj) then begin
-                                        display_msg(message_str(1189, 400));
+                                        display_msg(message_str(SCRIPT_ZILOCKER, 400));
                                         if (((local_var(0) bwand 512) != 0) == 0) then begin
                                             give_exp_points(25);
-                                            display_msg(message_str(14, 100) + 25 + message_str(14, 101));
+                                            display_msg(g_mstr(100) + 25 + g_mstr(101));
                                         end
                                         set_local_var(0, local_var(0) bwor 512);
                                     end
                                     else begin
                                         if (user_obj != 0) then begin
-                                            display_msg(obj_name(user_obj) + message_str(1189, 500));
+                                            display_msg(obj_name(user_obj) + message_str(SCRIPT_ZILOCKER, 500));
                                             if (((local_var(0) bwand 512) != 0) == 0) then begin
                                                 give_exp_points(25);
-                                                display_msg(message_str(14, 100) + 25 + message_str(14, 101));
+                                                display_msg(g_mstr(100) + 25 + g_mstr(101));
                                             end
                                             set_local_var(0, local_var(0) bwor 512);
                                         end
@@ -1750,11 +1750,11 @@ begin
                                 end
                                 else begin
                                     if (user_obj == dude_obj) then begin
-                                        display_msg(message_str(1189, 401));
+                                        display_msg(message_str(SCRIPT_ZILOCKER, 401));
                                     end
                                     else begin
                                         if (user_obj != 0) then begin
-                                            display_msg(obj_name(user_obj) + message_str(1189, 501));
+                                            display_msg(obj_name(user_obj) + message_str(SCRIPT_ZILOCKER, 501));
                                         end
                                     end
                                     debug_msg("pry_fail_msg");
@@ -1770,19 +1770,19 @@ begin
                             if (is_success(roll_vs_skill(user_obj, 9, -75 + 40))) then begin
                                 script_overrides;
                                 if (user_obj == dude_obj) then begin
-                                    display_msg(message_str(1189, 102));
+                                    display_msg(message_str(SCRIPT_ZILOCKER, 102));
                                     if (((local_var(0) bwand 512) != 0) == 0) then begin
                                         give_exp_points(25);
-                                        display_msg(message_str(14, 100) + 25 + message_str(14, 101));
+                                        display_msg(g_mstr(100) + 25 + g_mstr(101));
                                     end
                                     set_local_var(0, local_var(0) bwor 512);
                                 end
                                 else begin
                                     if (user_obj != 0) then begin
-                                        display_msg(obj_name(user_obj) + message_str(1189, 202));
+                                        display_msg(obj_name(user_obj) + message_str(SCRIPT_ZILOCKER, 202));
                                         if (((local_var(0) bwand 512) != 0) == 0) then begin
                                             give_exp_points(25);
-                                            display_msg(message_str(14, 100) + 25 + message_str(14, 101));
+                                            display_msg(g_mstr(100) + 25 + g_mstr(101));
                                         end
                                         set_local_var(0, local_var(0) bwor 512);
                                     end
@@ -1793,11 +1793,11 @@ begin
                             else begin
                                 script_overrides;
                                 if (user_obj == dude_obj) then begin
-                                    display_msg(message_str(1189, 103));
+                                    display_msg(message_str(SCRIPT_ZILOCKER, 103));
                                 end
                                 else begin
                                     if (user_obj != 0) then begin
-                                        display_msg(obj_name(user_obj) + message_str(1189, 203));
+                                        display_msg(obj_name(user_obj) + message_str(SCRIPT_ZILOCKER, 203));
                                     end
                                 end
                                 debug_msg("lock_fail_msg");
@@ -1811,19 +1811,19 @@ begin
                                 if (is_success(roll_vs_skill(user_obj, 9, -75 + 40))) then begin
                                     script_overrides;
                                     if (user_obj == dude_obj) then begin
-                                        display_msg(message_str(1189, 102));
+                                        display_msg(message_str(SCRIPT_ZILOCKER, 102));
                                         if (((local_var(0) bwand 512) != 0) == 0) then begin
                                             give_exp_points(25);
-                                            display_msg(message_str(14, 100) + 25 + message_str(14, 101));
+                                            display_msg(g_mstr(100) + 25 + g_mstr(101));
                                         end
                                         set_local_var(0, local_var(0) bwor 512);
                                     end
                                     else begin
                                         if (user_obj != 0) then begin
-                                            display_msg(obj_name(user_obj) + message_str(1189, 202));
+                                            display_msg(obj_name(user_obj) + message_str(SCRIPT_ZILOCKER, 202));
                                             if (((local_var(0) bwand 512) != 0) == 0) then begin
                                                 give_exp_points(25);
-                                                display_msg(message_str(14, 100) + 25 + message_str(14, 101));
+                                                display_msg(g_mstr(100) + 25 + g_mstr(101));
                                             end
                                             set_local_var(0, local_var(0) bwor 512);
                                         end
@@ -1834,11 +1834,11 @@ begin
                                 else begin
                                     script_overrides;
                                     if (user_obj == dude_obj) then begin
-                                        display_msg(message_str(1189, 103));
+                                        display_msg(message_str(SCRIPT_ZILOCKER, 103));
                                     end
                                     else begin
                                         if (user_obj != 0) then begin
-                                            display_msg(obj_name(user_obj) + message_str(1189, 203));
+                                            display_msg(obj_name(user_obj) + message_str(SCRIPT_ZILOCKER, 203));
                                         end
                                     end
                                     debug_msg("lock_fail_msg");
@@ -1873,7 +1873,7 @@ begin
             if (caught_tamper == 0) then begin
                 if (1) then begin
                     if (0 != 0) then begin
-                        if (has_trait(1, 0, 666) and ((critter_state(0) bwand 1) == 0)) then begin
+                        if (obj_is_visible_flag(0) and ((critter_state(0) bwand 1) == 0)) then begin
                             if (obj_can_see_obj(0, user_obj)) then begin
                                 caught_tamper := 1;
                             end
@@ -1888,7 +1888,7 @@ begin
                         attack_setup(0, user_obj);
                     end
                     else begin
-                        float_msg(0, message_str(14, 5400), 8);
+                        float_msg(0, g_mstr(5400), FLOAT_MSG_YELLOW);
                     end
                     set_local_var(0, local_var(0) bwor 1024);
                 end
@@ -1901,19 +1901,19 @@ begin
                             if (is_success(roll_vs_skill(user_obj, 9, -75))) then begin
                                 script_overrides;
                                 if (user_obj == dude_obj) then begin
-                                    display_msg(message_str(1189, 100));
+                                    display_msg(message_str(SCRIPT_ZILOCKER, 100));
                                     if (((local_var(0) bwand 512) != 0) == 0) then begin
                                         give_exp_points(25);
-                                        display_msg(message_str(14, 100) + 25 + message_str(14, 101));
+                                        display_msg(g_mstr(100) + 25 + g_mstr(101));
                                     end
                                     set_local_var(0, local_var(0) bwor 512);
                                 end
                                 else begin
                                     if (user_obj != 0) then begin
-                                        display_msg(obj_name(user_obj) + message_str(1189, 200));
+                                        display_msg(obj_name(user_obj) + message_str(SCRIPT_ZILOCKER, 200));
                                         if (((local_var(0) bwand 512) != 0) == 0) then begin
                                             give_exp_points(25);
-                                            display_msg(message_str(14, 100) + 25 + message_str(14, 101));
+                                            display_msg(g_mstr(100) + 25 + g_mstr(101));
                                         end
                                         set_local_var(0, local_var(0) bwor 512);
                                     end
@@ -1924,11 +1924,11 @@ begin
                             else begin
                                 script_overrides;
                                 if (user_obj == dude_obj) then begin
-                                    display_msg(message_str(1189, 101));
+                                    display_msg(message_str(SCRIPT_ZILOCKER, 101));
                                 end
                                 else begin
                                     if (user_obj != 0) then begin
-                                        display_msg(obj_name(user_obj) + message_str(1189, 201));
+                                        display_msg(obj_name(user_obj) + message_str(SCRIPT_ZILOCKER, 201));
                                     end
                                 end
                                 debug_msg("unlock_fail_msg");
@@ -1940,19 +1940,19 @@ begin
                             if (is_success(roll_vs_skill(user_obj, 9, -75))) then begin
                                 script_overrides;
                                 if (user_obj == dude_obj) then begin
-                                    display_msg(message_str(1189, 102));
+                                    display_msg(message_str(SCRIPT_ZILOCKER, 102));
                                     if (((local_var(0) bwand 512) != 0) == 0) then begin
                                         give_exp_points(25);
-                                        display_msg(message_str(14, 100) + 25 + message_str(14, 101));
+                                        display_msg(g_mstr(100) + 25 + g_mstr(101));
                                     end
                                     set_local_var(0, local_var(0) bwor 512);
                                 end
                                 else begin
                                     if (user_obj != 0) then begin
-                                        display_msg(obj_name(user_obj) + message_str(1189, 202));
+                                        display_msg(obj_name(user_obj) + message_str(SCRIPT_ZILOCKER, 202));
                                         if (((local_var(0) bwand 512) != 0) == 0) then begin
                                             give_exp_points(25);
-                                            display_msg(message_str(14, 100) + 25 + message_str(14, 101));
+                                            display_msg(g_mstr(100) + 25 + g_mstr(101));
                                         end
                                         set_local_var(0, local_var(0) bwor 512);
                                     end
@@ -1963,11 +1963,11 @@ begin
                             else begin
                                 script_overrides;
                                 if (user_obj == dude_obj) then begin
-                                    display_msg(message_str(1189, 103));
+                                    display_msg(message_str(SCRIPT_ZILOCKER, 103));
                                 end
                                 else begin
                                     if (user_obj != 0) then begin
-                                        display_msg(obj_name(user_obj) + message_str(1189, 203));
+                                        display_msg(obj_name(user_obj) + message_str(SCRIPT_ZILOCKER, 203));
                                     end
                                 end
                                 debug_msg("lock_fail_msg");
@@ -1983,4 +1983,3 @@ end
 procedure timed_event_p_proc
 begin
 end
-
