diff --git a/Mods/Resurrection/Original/SPOLICE2.ssl b/Mods/Resurrection/ReDefine/SPOLICE2.ssl
index fc431cb..7c189dc 100644
--- a/Mods/Resurrection/Original/SPOLICE2.ssl
+++ b/Mods/Resurrection/ReDefine/SPOLICE2.ssl
@@ -52,12 +52,12 @@ begin
             return 1;
         end
     end
-    if ((party_member_obj(16777779) != 0) and ((global_var(910) bwand 32) == 0)) then begin
+    if ((party_member_obj(16777779) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 32) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777779)) <= 5) then begin
             return 1;
         end
     end
-    if ((party_member_obj(16777811) != 0) and ((global_var(910) bwand 64) == 0)) then begin
+    if ((party_member_obj(16777811) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 64) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777811)) <= 5) then begin
             return 1;
         end
@@ -72,7 +72,7 @@ begin
             return 1;
         end
     end
-    if ((party_member_obj(16777814) != 0) and ((global_var(910) bwand 128) == 0)) then begin
+    if ((party_member_obj(16777814) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 128) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777814)) <= 5) then begin
             return 1;
         end
@@ -125,7 +125,7 @@ end
 
 procedure start
 begin
-    if ((cur_map_index == 10) and (tile_num(self_obj) == 14712)) then begin
+    if ((CUR_MAP_SED_CENT) and (tile_num(self_obj) == 14712)) then begin
         police_door_guard := self_obj;
     end
 end
@@ -133,21 +133,21 @@ end
 procedure critter_p_proc
 begin
     if ((local_var(5) == 1) and obj_can_see_obj(self_obj, dude_obj)) then begin
-        attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+        attack(dude_obj);
     end
-    if (global_var(769) == 1) then begin
+    if (global_var(GVAR_ENEMY_SEDIT) == 1) then begin
         if (obj_can_see_obj(self_obj, dude_obj)) then begin
-            attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+            attack(dude_obj);
         end
     end
-    if ((cur_map_index == 11) and (tile_num(self_obj) == 16273)) then begin
-        if ((global_var(724) >= 5) and (global_var(724) <= 6)) then begin
-            if (has_trait(1, self_obj, 666) == 1) then begin
+    if ((CUR_MAP_SED_MHUB) and (tile_num(self_obj) == 16273)) then begin
+        if ((global_var(GVAR_QUEST_A_DAVID_KIDNAP_COMPLICITY) >= 5) and (global_var(GVAR_QUEST_A_DAVID_KIDNAP_COMPLICITY) <= 6)) then begin
+            if (obj_is_visible_flag(self_obj) == 1) then begin
                 set_obj_visibility(self_obj, 1);
             end
         end
         else begin
-            if (has_trait(1, self_obj, 666) == 0) then begin
+            if (obj_is_visible_flag(self_obj) == 0) then begin
                 set_obj_visibility(self_obj, 0);
             end
         end
@@ -157,14 +157,14 @@ end
 procedure combat_p_proc
 begin
     if (fixed_param == 4) then begin
-        if ((get_critter_stat(dude_obj, 34) == 0) and (get_critter_stat(dude_obj, 6) == 10) and (combat_cry != 41) and (global_var(769) == 1) and has_trait(1, self_obj, 666)) then begin
+        if ((dude_gender == 0) and (dude_luck == 10) and (combat_cry != 41) and (global_var(GVAR_ENEMY_SEDIT) == 1) and obj_is_visible_flag(self_obj)) then begin
             combat_cry := random(1, 40);
             if (combat_cry == 13) then begin
-                float_msg(self_obj, message_str(1410, 400), 8);
+                float_msg(self_obj, mstr(400), FLOAT_MSG_YELLOW);
                 combat_cry := 41;
             end
             if (combat_cry == 23) then begin
-                float_msg(self_obj, message_str(1410, 401), 8);
+                float_msg(self_obj, mstr(401), FLOAT_MSG_YELLOW);
                 combat_cry := 41;
             end
         end
@@ -175,24 +175,24 @@ procedure pickup_p_proc
 begin
     if (source_obj == dude_obj) then begin
         if (random(1, 2) == 1) then begin
-            if (get_critter_stat(self_obj, 34) == 0) then begin
-                float_msg(self_obj, message_str(1410, 350 + (get_critter_stat(dude_obj, 34) == 1)), 2);
+            if (self_gender == 0) then begin
+                float_msg(self_obj, mstr(350 + (dude_gender == 1)), FLOAT_MSG_RED);
             end
-            if (get_critter_stat(self_obj, 34) == 1) then begin
-                float_msg(self_obj, message_str(1410, 353 + (get_critter_stat(dude_obj, 34) == 1)), 2);
+            if (self_gender == 1) then begin
+                float_msg(self_obj, mstr(353 + (dude_gender == 1)), FLOAT_MSG_RED);
             end
         end
         else begin
-            float_msg(self_obj, message_str(1410, 352), 2);
+            float_msg(self_obj, mstr(352), FLOAT_MSG_RED);
         end
         set_local_var(5, 1);
-        attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+        attack(dude_obj);
     end
 end
 
 procedure talk_p_proc
 begin
-    if ((global_var(790) != 3) and (global_var(738) != 4) and (global_var(722) != 1)) then begin
+    if ((global_var(GVAR_QUEST_S_MIMI_RODRIGUES) != 3) and (global_var(GVAR_QUEST_S_NICOL_CRATER) != 4) and (global_var(GVAR_SEDIT_TYLER_IS_SHERIFF) != 1)) then begin
         call Police_talking();
     end
     else begin
@@ -202,21 +202,21 @@ begin
         else begin
             while (dialog_loop == 0) do begin
                 talking_float := random(1, 3);
-                if ((talking_float == 1) and (global_var(790) == 3)) then begin
-                    float_msg(self_obj, message_str(1410, random(321, 322)), 8);
+                if ((talking_float == 1) and (global_var(GVAR_QUEST_S_MIMI_RODRIGUES) == 3)) then begin
+                    float_msg(self_obj, mstr(random(321, 322)), FLOAT_MSG_YELLOW);
                     dialog_loop := 1;
                 end
-                if ((talking_float == 2) and (global_var(738) == 4)) then begin
-                    if (get_critter_stat(self_obj, 34) == 0) then begin
-                        float_msg(self_obj, message_str(1410, random(323, 324)), 8);
+                if ((talking_float == 2) and (global_var(GVAR_QUEST_S_NICOL_CRATER) == 4)) then begin
+                    if (self_gender == 0) then begin
+                        float_msg(self_obj, mstr(random(323, 324)), FLOAT_MSG_YELLOW);
                     end
-                    if (get_critter_stat(self_obj, 34) == 1) then begin
-                        float_msg(self_obj, message_str(1410, random(331, 332)), 8);
+                    if (self_gender == 1) then begin
+                        float_msg(self_obj, mstr(random(331, 332)), FLOAT_MSG_YELLOW);
                     end
                     dialog_loop := 1;
                 end
-                if ((talking_float == 3) and (global_var(722) == 1)) then begin
-                    float_msg(self_obj, message_str(1410, random(325, 330)), 8);
+                if ((talking_float == 3) and (global_var(GVAR_SEDIT_TYLER_IS_SHERIFF) == 1)) then begin
+                    float_msg(self_obj, mstr(random(325, 330)), FLOAT_MSG_YELLOW);
                     dialog_loop := 1;
                 end
             end
@@ -228,45 +228,45 @@ end
 procedure look_at_p_proc
 begin
     script_overrides;
-    if (get_critter_stat(self_obj, 34) == 0) then begin
-        display_msg(message_str(1410, 100));
+    if (self_gender == 0) then begin
+        display_mstr(100);
     end
-    if (get_critter_stat(self_obj, 34) == 1) then begin
-        display_msg(message_str(1410, 101));
+    if (self_gender == 1) then begin
+        display_mstr(101);
     end
 end
 
 procedure description_p_proc
 begin
     script_overrides;
-    if (get_critter_stat(self_obj, 34) == 0) then begin
-        display_msg(message_str(1410, 200));
+    if (self_gender == 0) then begin
+        display_mstr(200);
     end
-    if (get_critter_stat(self_obj, 34) == 1) then begin
-        display_msg(message_str(1410, 201));
+    if (self_gender == 1) then begin
+        display_mstr(201);
     end
 end
 
 procedure timed_event_p_proc
 begin
     if (fixed_param == 10) then begin
-        float_msg(self_obj, message_str(1410, 355), 2);
+        float_msg(self_obj, mstr(355), FLOAT_MSG_RED);
     end
     if (fixed_param == 11) then begin
         set_local_var(5, 1);
-        set_global_var(769, 1);
+        set_global_var(GVAR_ENEMY_SEDIT, 1);
     end
 end
 
 procedure map_enter_p_proc
 begin
-    critter_add_trait(self_obj, 1, 6, 212);
-    critter_add_trait(self_obj, 1, 5, 146);
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 212);
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_SEDIT_POLICE);
     if (local_var(6) == 0) then begin
         set_local_var(6, tile_num(self_obj));
     end
-    if (not(metarule(22, 0)) and (cur_map_index == 10) and (tile_num(self_obj) == 16287)) then begin
-        if (global_var(749) == 1) then begin
+    if (not(is_loading_game) and (CUR_MAP_SED_CENT) and (tile_num(self_obj) == 16287)) then begin
+        if (global_var(GVAR_DUDE_NICOL_MEETING) == 1) then begin
             set_obj_visibility(self_obj, 1);
         end
         else begin
@@ -278,43 +278,42 @@ end
 procedure map_exit_p_proc
 begin
     if (tile_num(self_obj) != local_var(6)) then begin
-        if (global_var(769) != 1) then begin
+        if (global_var(GVAR_ENEMY_SEDIT) != 1) then begin
             move_to(self_obj, local_var(6), elevation(self_obj));
         end
     end
-    float_msg(self_obj, 0, 0);
+    float_msg_clear(self_obj);
 end
 
 procedure damage_p_proc
 begin
     if (source_obj == dude_obj) then begin
         set_local_var(5, 1);
-        set_global_var(769, 1);
+        set_global_var(GVAR_ENEMY_SEDIT, 1);
     end
 end
 
 procedure destroy_p_proc
 begin
-    set_global_var(48, global_var(48) + -5);
+    set_global_var(GVAR_TOWN_REP_SEDIT, global_var(GVAR_TOWN_REP_SEDIT) + -5);
 end
 
 procedure Police_talking
 begin
     if (party_member_obj(16777812) != 0) then begin
-        float_msg(self_obj, message_str(1410, random(316, 320)), 8);
+        float_msg(self_obj, mstr(random(316, 320)), FLOAT_MSG_YELLOW);
     end
     else begin
-        if ((obj_pid(critter_inven_obj(dude_obj, 0)) == 3) or (obj_pid(critter_inven_obj(dude_obj, 0)) == 348) or (obj_pid(critter_inven_obj(dude_obj, 0)) == 349) or (obj_pid(critter_inven_obj(dude_obj, 0)) == 232)) then begin
-            float_msg(self_obj, message_str(1410, random(312, 315)), 8);
+        if ((obj_pid(critter_inven_obj(dude_obj, INVEN_TYPE_WORN)) == 3) or (obj_pid(critter_inven_obj(dude_obj, INVEN_TYPE_WORN)) == 348) or (obj_pid(critter_inven_obj(dude_obj, INVEN_TYPE_WORN)) == 349) or (obj_pid(critter_inven_obj(dude_obj, INVEN_TYPE_WORN)) == 232)) then begin
+            float_msg(self_obj, mstr(random(312, 315)), FLOAT_MSG_YELLOW);
         end
         else begin
-            if ((get_critter_stat(dude_obj, 34) == 1) and (get_critter_stat(self_obj, 34) == 0) and ((get_critter_stat(dude_obj, 3) > 6) or has_trait(2, dude_obj, 13))) then begin
-                float_msg(self_obj, message_str(1410, random(307, 311)), 8);
+            if ((dude_gender == 1) and (self_gender == 0) and ((dude_charisma > 6) or has_trait(TRAIT_TRAIT, dude_obj, TRAIT_sex_appeal))) then begin
+                float_msg(self_obj, mstr(random(307, 311)), FLOAT_MSG_YELLOW);
             end
             else begin
-                float_msg(self_obj, message_str(1410, random(300, 306)), 8);
+                float_msg(self_obj, mstr(random(300, 306)), FLOAT_MSG_YELLOW);
             end
         end
     end
 end
-
