diff --git a/Mods/Resurrection/Original/SPOLICE4.ssl b/Mods/Resurrection/ReDefine/SPOLICE4.ssl
index e1972aa..c247691 100644
--- a/Mods/Resurrection/Original/SPOLICE4.ssl
+++ b/Mods/Resurrection/ReDefine/SPOLICE4.ssl
@@ -47,12 +47,12 @@ begin
             return 1;
         end
     end
-    if ((party_member_obj(16777779) != 0) and ((global_var(910) bwand 32) == 0)) then begin
+    if ((party_member_obj(16777779) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 32) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777779)) <= 5) then begin
             return 1;
         end
     end
-    if ((party_member_obj(16777811) != 0) and ((global_var(910) bwand 64) == 0)) then begin
+    if ((party_member_obj(16777811) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 64) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777811)) <= 5) then begin
             return 1;
         end
@@ -67,7 +67,7 @@ begin
             return 1;
         end
     end
-    if ((party_member_obj(16777814) != 0) and ((global_var(910) bwand 128) == 0)) then begin
+    if ((party_member_obj(16777814) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 128) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777814)) <= 5) then begin
             return 1;
         end
@@ -126,11 +126,11 @@ procedure critter_p_proc
 begin
     if ((local_var(5) == 2) and obj_can_see_obj(self_obj, dude_obj)) then begin
         set_local_var(5, 1);
-        attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+        attack(dude_obj);
     end
     if (local_var(4) == 1) then begin
         if (obj_can_see_obj(self_obj, dude_obj) or obj_can_hear_obj(self_obj, dude_obj)) then begin
-            attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+            attack(dude_obj);
         end
     end
 end
@@ -144,8 +144,8 @@ end
 
 procedure talk_p_proc
 begin
-    anim(dude_obj, 1000, rotation_to_tile(tile_num(dude_obj), tile_num(self_obj)));
-    anim(self_obj, 1000, rotation_to_tile(tile_num(self_obj), tile_num(dude_obj)));
+    anim(dude_obj, 1000, rotation_to_tile(dude_tile, tile_num(self_obj)));
+    anim(self_obj, 1000, rotation_to_tile(tile_num(self_obj), dude_tile));
     start_gdialog(1412, self_obj, 4, -1, -1);
     gsay_start;
     call Node500();
@@ -156,13 +156,13 @@ end
 procedure look_at_p_proc
 begin
     script_overrides;
-    display_msg(message_str(1412, 100));
+    display_mstr(100);
 end
 
 procedure description_p_proc
 begin
     script_overrides;
-    display_msg(message_str(1412, 200));
+    display_mstr(200);
 end
 
 procedure timed_event_p_proc
@@ -180,8 +180,8 @@ end
 
 procedure map_enter_p_proc
 begin
-    critter_add_trait(self_obj, 1, 6, 233);
-    critter_add_trait(self_obj, 1, 5, 3);
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 233);
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_TOUGH_GUARD_NO_TALKING);
     add_timer_event(self_obj, game_ticks(random(8, 14)), 1);
 end
 
@@ -209,27 +209,26 @@ end
 
 procedure Node500
 begin
-    gsay_reply(1412, 500 + (get_critter_stat(dude_obj, 34) == 1));
-    giq_option(7, 1412, 502, Node550, 50);
+    Reply(500 + (dude_gender == 1));
+    giq_option(7, NAME, 502, Node550, NEUTRAL_REACTION);
     if (obj_carrying_pid_obj(dude_obj, 559)) then begin
-        giq_option(4, 1412, 503, Node600, 50);
+        giq_option(4, NAME, 503, Node600, NEUTRAL_REACTION);
     end
-    giq_option(4, 1412, 504, Node998, 50);
-    giq_option(-3, 1412, 505, Node998, 50);
+    giq_option(4, NAME, 504, Node998, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 505, Node998, NEUTRAL_REACTION);
 end
 
 procedure Node550
 begin
-    gsay_reply(1412, 550);
+    Reply(550);
     if (obj_carrying_pid_obj(dude_obj, 559)) then begin
-        giq_option(4, 1412, 551, Node600, 50);
+        giq_option(4, NAME, 551, Node600, NEUTRAL_REACTION);
     end
-    giq_option(4, 1412, 552, Node998, 50);
+    giq_option(4, NAME, 552, Node998, NEUTRAL_REACTION);
 end
 
 procedure Node600
 begin
-    gsay_reply(1412, 600);
-    giq_option(4, 1412, 601, Node998, 50);
+    Reply(600);
+    giq_option(4, NAME, 601, Node998, NEUTRAL_REACTION);
 end
-
