diff --git a/Mods/Resurrection/Original/SPOLICE7.ssl b/Mods/Resurrection/ReDefine/SPOLICE7.ssl
index 8571bc0..80e1838 100644
--- a/Mods/Resurrection/Original/SPOLICE7.ssl
+++ b/Mods/Resurrection/ReDefine/SPOLICE7.ssl
@@ -49,12 +49,12 @@ begin
             return 1;
         end
     end
-    if ((party_member_obj(16777779) != 0) and ((global_var(910) bwand 32) == 0)) then begin
+    if ((party_member_obj(16777779) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 32) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777779)) <= 5) then begin
             return 1;
         end
     end
-    if ((party_member_obj(16777811) != 0) and ((global_var(910) bwand 64) == 0)) then begin
+    if ((party_member_obj(16777811) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 64) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777811)) <= 5) then begin
             return 1;
         end
@@ -69,7 +69,7 @@ begin
             return 1;
         end
     end
-    if ((party_member_obj(16777814) != 0) and ((global_var(910) bwand 128) == 0)) then begin
+    if ((party_member_obj(16777814) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 128) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777814)) <= 5) then begin
             return 1;
         end
@@ -128,11 +128,11 @@ procedure critter_p_proc
 begin
     if ((local_var(5) == 2) and obj_can_see_obj(self_obj, dude_obj)) then begin
         set_local_var(5, 1);
-        attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+        attack(dude_obj);
     end
-    if (global_var(774) > 2) then begin
+    if (global_var(GVAR_ENEMY_LOST_QUARTER) > 2) then begin
         if (obj_can_see_obj(self_obj, dude_obj)) then begin
-            attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+            attack(dude_obj);
         end
     end
     if (map_var(9) > 0) then begin
@@ -160,24 +160,24 @@ end
 procedure combat_p_proc
 begin
     if (fixed_param == 4) then begin
-        if ((global_var(770) == 1) or (global_var(774) > 0) and (global_var(774) < 3)) then begin
+        if ((global_var(GVAR_ENEMY_RODRIGUES) == 1) or (global_var(GVAR_ENEMY_LOST_QUARTER) > 0) and (global_var(GVAR_ENEMY_LOST_QUARTER) < 3)) then begin
             critter_set_flee_state(self_obj, 1);
         end
         else begin
-            if (global_var(774) > 2) then begin
+            if (global_var(GVAR_ENEMY_LOST_QUARTER) > 2) then begin
                 if (critter_is_fleeing(self_obj)) then begin
-                    attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+                    attack(dude_obj);
                 end
             end
         end
-        if ((get_critter_stat(dude_obj, 34) == 0) and (get_critter_stat(dude_obj, 6) == 10) and (combat_cry != 41) and (global_var(769) == 1)) then begin
+        if ((dude_gender == 0) and (dude_luck == 10) and (combat_cry != 41) and (global_var(GVAR_ENEMY_SEDIT) == 1)) then begin
             combat_cry := random(1, 40);
             if (combat_cry == 13) then begin
-                float_msg(self_obj, message_str(1432, 400), 8);
+                float_msg(self_obj, mstr(400), FLOAT_MSG_YELLOW);
                 combat_cry := 41;
             end
             if (combat_cry == 23) then begin
-                float_msg(self_obj, message_str(1432, 401), 8);
+                float_msg(self_obj, mstr(401), FLOAT_MSG_YELLOW);
                 combat_cry := 41;
             end
         end
@@ -187,36 +187,36 @@ end
 procedure pickup_p_proc
 begin
     if (source_obj == dude_obj) then begin
-        float_msg(self_obj, message_str(1432, 350 + (get_critter_stat(dude_obj, 34) == 1)), 2);
+        float_msg(self_obj, mstr(350 + (dude_gender == 1)), FLOAT_MSG_RED);
         set_local_var(5, 2);
     end
 end
 
 procedure talk_p_proc
 begin
-    float_msg(self_obj, message_str(1432, random(308, 315)), 8);
+    float_msg(self_obj, mstr(random(308, 315)), FLOAT_MSG_YELLOW);
 end
 
 procedure look_at_p_proc
 begin
     script_overrides;
-    display_msg(message_str(1432, 100));
+    display_mstr(100);
 end
 
 procedure description_p_proc
 begin
     script_overrides;
-    display_msg(message_str(1432, 200));
+    display_mstr(200);
 end
 
 procedure timed_event_p_proc
 begin
     if (fixed_param == 1) then begin
         if (not(combat_is_initialized) and (map_var(9) == 0)) then begin
-            float_msg(self_obj, message_str(1432, random(300, 307)), 8);
+            float_msg(self_obj, mstr(random(300, 307)), FLOAT_MSG_YELLOW);
         end
-        if (global_var(792) != 2) then begin
-            metarule3(100, self_obj, 1, 0);
+        if (global_var(GVAR_SEDIT_DEAD_GANGSTERS) != 2) then begin
+            rm_fixed_timer_event(self_obj, 1);
             add_timer_event(self_obj, game_ticks(random(7, 14)), 1);
         end
     end
@@ -224,16 +224,16 @@ end
 
 procedure map_enter_p_proc
 begin
-    critter_add_trait(self_obj, 1, 6, 212);
-    critter_add_trait(self_obj, 1, 5, 146);
-    metarule3(100, self_obj, 1, 0);
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 212);
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_SEDIT_POLICE);
+    rm_fixed_timer_event(self_obj, 1);
     add_timer_event(self_obj, game_ticks(random(7, 14)), 1);
 end
 
 procedure map_exit_p_proc
 begin
-    float_msg(self_obj, 0, 0);
-    if (global_var(792) == 2) then begin
+    float_msg_clear(self_obj);
+    if (global_var(GVAR_SEDIT_DEAD_GANGSTERS) == 2) then begin
         destroy_object(self_obj);
     end
 end
@@ -241,9 +241,9 @@ end
 procedure damage_p_proc
 begin
     if (source_obj == dude_obj) then begin
-        set_global_var(770, 1);
-        if (global_var(774) < 3) then begin
-            set_global_var(774, global_var(774) + 1);
+        set_global_var(GVAR_ENEMY_RODRIGUES, 1);
+        if (global_var(GVAR_ENEMY_LOST_QUARTER) < 3) then begin
+            set_global_var(GVAR_ENEMY_LOST_QUARTER, global_var(GVAR_ENEMY_LOST_QUARTER) + 1);
         end
     end
 end
@@ -251,4 +251,3 @@ end
 procedure destroy_p_proc
 begin
 end
-
