diff --git a/Mods/Resurrection/Original/SSABRINA.ssl b/Mods/Resurrection/ReDefine/SSABRINA.ssl
index e7afe96..8f5588d 100644
--- a/Mods/Resurrection/Original/SSABRINA.ssl
+++ b/Mods/Resurrection/ReDefine/SSABRINA.ssl
@@ -90,12 +90,12 @@ begin
             return 1;
         end
     end
-    if ((party_member_obj(16777779) != 0) and ((global_var(910) bwand 32) == 0)) then begin
+    if ((party_member_obj(16777779) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 32) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777779)) <= 5) then begin
             return 1;
         end
     end
-    if ((party_member_obj(16777811) != 0) and ((global_var(910) bwand 64) == 0)) then begin
+    if ((party_member_obj(16777811) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 64) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777811)) <= 5) then begin
             return 1;
         end
@@ -110,7 +110,7 @@ begin
             return 1;
         end
     end
-    if ((party_member_obj(16777814) != 0) and ((global_var(910) bwand 128) == 0)) then begin
+    if ((party_member_obj(16777814) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 128) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777814)) <= 5) then begin
             return 1;
         end
@@ -171,16 +171,16 @@ begin
     if ((local_var(5) == 2) and obj_can_see_obj(self_obj, dude_obj)) then begin
         set_local_var(5, 1);
         Sabrina := -1;
-        attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+        attack(dude_obj);
     end
-    if (tile_num(dude_obj) == 17274) then begin
+    if (dude_tile == 17274) then begin
         set_local_var(5, 1);
         Sabrina := -1;
-        attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+        attack(dude_obj);
     end
-    if (global_var(769) == 1) then begin
+    if (global_var(GVAR_ENEMY_SEDIT) == 1) then begin
         if (obj_can_see_obj(self_obj, dude_obj) or obj_can_hear_obj(self_obj, dude_obj)) then begin
-            attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+            attack(dude_obj);
         end
     end
 end
@@ -196,8 +196,8 @@ procedure talk_p_proc
 begin
     weapon := 0;
     if (local_var(4) != 1) then begin
-        anim(dude_obj, 1000, rotation_to_tile(tile_num(dude_obj), tile_num(self_obj)));
-        anim(self_obj, 1000, rotation_to_tile(tile_num(self_obj), tile_num(dude_obj)));
+        anim(dude_obj, 1000, rotation_to_tile(dude_tile, tile_num(self_obj)));
+        anim(self_obj, 1000, rotation_to_tile(tile_num(self_obj), dude_tile));
         if (Open_hours() == 1) then begin
             start_gdialog(1370, self_obj, 4, -1, -1);
             gsay_start;
@@ -215,8 +215,8 @@ begin
         end
     end
     else begin
-        anim(dude_obj, 1000, rotation_to_tile(tile_num(dude_obj), tile_num(self_obj)));
-        anim(self_obj, 1000, rotation_to_tile(tile_num(self_obj), tile_num(dude_obj)));
+        anim(dude_obj, 1000, rotation_to_tile(dude_tile, tile_num(self_obj)));
+        anim(self_obj, 1000, rotation_to_tile(tile_num(self_obj), dude_tile));
         start_gdialog(1370, self_obj, 4, -1, -1);
         gsay_start;
         if (Open_hours() == 1) then begin
@@ -244,10 +244,10 @@ procedure look_at_p_proc
 begin
     script_overrides;
     if (local_var(4) == 1) then begin
-        display_msg(message_str(1370, 150));
+        display_mstr(150);
     end
     else begin
-        display_msg(message_str(1370, 100));
+        display_mstr(100);
     end
 end
 
@@ -256,10 +256,10 @@ begin
     script_overrides;
     set_local_var(11, 1);
     if (local_var(4) == 1) then begin
-        display_msg(message_str(1370, 250));
+        display_mstr(250);
     end
     else begin
-        display_msg(message_str(1370, 200));
+        display_mstr(200);
     end
 end
 
@@ -268,21 +268,21 @@ begin
     if (fixed_param == 10) then begin
         set_local_var(5, 1);
         Sabrina := -1;
-        attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+        attack(dude_obj);
     end
 end
 
 procedure map_enter_p_proc
 begin
-    critter_add_trait(self_obj, 1, 6, 212);
-    critter_add_trait(self_obj, 1, 5, 52);
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 212);
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_TOUGH_CITIZEN_NO_TALKING);
 end
 
 procedure damage_p_proc
 begin
     if (source_obj == dude_obj) then begin
         call Node998();
-        set_global_var(769, 1);
+        set_global_var(GVAR_ENEMY_SEDIT, 1);
     end
 end
 
@@ -290,59 +290,59 @@ procedure destroy_p_proc
 begin
     Sabrina := -1;
     if (source_obj == dude_obj) then begin
-        set_global_var(5, global_var(5) + 1);
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(4) > (3 * global_var(5))) or (global_var(2) == 1)) and (global_var(1) == 0)) then begin
-            set_global_var(2, 1);
-            set_global_var(3, 0);
+        set_global_var(GVAR_GOOD_MONSTER, global_var(GVAR_GOOD_MONSTER) + 1);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(4) > (3 * global_var(GVAR_GOOD_MONSTER))) or (global_var(GVAR_CHAMPION_REPUTATION) == 1)) and (global_var(GVAR_CHILDKILLER_REPUTATION) == 0)) then begin
+            set_global_var(GVAR_CHAMPION_REPUTATION, 1);
+            set_global_var(GVAR_BERSERKER_REPUTATION, 0);
         end
-        if (metarule(51, self_obj) == 2) then begin
+        if (critter_kill_type(self_obj) == KILL_TYPE_children_kills) then begin
             if (source_obj == dude_obj) then begin
-                set_global_var(2, 0);
-                set_global_var(1, global_var(1) + 1);
-                set_global_var(0, global_var(0) + -15);
+                set_global_var(GVAR_CHAMPION_REPUTATION, 0);
+                set_global_var(GVAR_CHILDKILLER_REPUTATION, global_var(GVAR_CHILDKILLER_REPUTATION) + 1);
+                set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + -15);
                 debug_msg("Player gains " + -15 + " Karma Points.");
-                set_global_var(37, 0);
-                set_global_var(38, 0);
-                set_global_var(39, 0);
-                set_global_var(40, 0);
-                set_global_var(41, 0);
-                set_global_var(42, 0);
-                set_global_var(43, 0);
-                set_global_var(44, 0);
-                set_global_var(45, 0);
-                if (global_var(0) >= 1000) then begin
-                    set_global_var(37, 1);
+                set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+                set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+                set_global_var(GVAR_KARMA_DEFENDER, 0);
+                set_global_var(GVAR_KARMA_WANDERER, 0);
+                set_global_var(GVAR_KARMA_BETRAYER, 0);
+                set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+                set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+                set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+                    set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
                 end
                 else begin
-                    if (global_var(0) >= 750) then begin
-                        set_global_var(38, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
                     end
                     else begin
-                        if (global_var(0) >= 500) then begin
-                            set_global_var(39, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                            set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                         end
                         else begin
-                            if (global_var(0) >= 250) then begin
-                                set_global_var(40, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                                set_global_var(GVAR_KARMA_DEFENDER, 1);
                             end
                             else begin
-                                if (global_var(0) > -250) then begin
-                                    set_global_var(41, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                                    set_global_var(GVAR_KARMA_WANDERER, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -500) then begin
-                                        set_global_var(42, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                        set_global_var(GVAR_KARMA_BETRAYER, 1);
                                     end
                                     else begin
-                                        if (global_var(0) > -750) then begin
-                                            set_global_var(43, 1);
+                                        if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                            set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                         end
                                         else begin
-                                            if (global_var(0) > -1000) then begin
-                                                set_global_var(44, 1);
+                                            if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                                set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                             end
                                             else begin
-                                                set_global_var(45, 1);
+                                                set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                             end
                                         end
                                     end
@@ -351,58 +351,58 @@ begin
                         end
                     end
                 end
-                set_global_var(48, global_var(48) + -8);
+                set_global_var(GVAR_TOWN_REP_SEDIT, global_var(GVAR_TOWN_REP_SEDIT) + -8);
                 debug_msg("Added " + -8 + " to Town Rep");
             end
         end
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(5) > (2 * global_var(4))) or (global_var(3) == 1))) then begin
-            set_global_var(3, 1);
-            set_global_var(2, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
+            set_global_var(GVAR_BERSERKER_REPUTATION, 1);
+            set_global_var(GVAR_CHAMPION_REPUTATION, 0);
         end
-        set_global_var(0, global_var(0) + -10);
+        set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + -10);
         debug_msg("Player gains " + -10 + " Karma Points.");
-        set_global_var(37, 0);
-        set_global_var(38, 0);
-        set_global_var(39, 0);
-        set_global_var(40, 0);
-        set_global_var(41, 0);
-        set_global_var(42, 0);
-        set_global_var(43, 0);
-        set_global_var(44, 0);
-        set_global_var(45, 0);
-        if (global_var(0) >= 1000) then begin
-            set_global_var(37, 1);
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+        set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+        set_global_var(GVAR_KARMA_DEFENDER, 0);
+        set_global_var(GVAR_KARMA_WANDERER, 0);
+        set_global_var(GVAR_KARMA_BETRAYER, 0);
+        set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+        set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+            set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
         end
         else begin
-            if (global_var(0) >= 750) then begin
-                set_global_var(38, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
             end
             else begin
-                if (global_var(0) >= 500) then begin
-                    set_global_var(39, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                 end
                 else begin
-                    if (global_var(0) >= 250) then begin
-                        set_global_var(40, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                        set_global_var(GVAR_KARMA_DEFENDER, 1);
                     end
                     else begin
-                        if (global_var(0) > -250) then begin
-                            set_global_var(41, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                            set_global_var(GVAR_KARMA_WANDERER, 1);
                         end
                         else begin
-                            if (global_var(0) > -500) then begin
-                                set_global_var(42, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                set_global_var(GVAR_KARMA_BETRAYER, 1);
                             end
                             else begin
-                                if (global_var(0) > -750) then begin
-                                    set_global_var(43, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -1000) then begin
-                                        set_global_var(44, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                     end
                                     else begin
-                                        set_global_var(45, 1);
+                                        set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                     end
                                 end
                             end
@@ -411,7 +411,7 @@ begin
                 end
             end
         end
-        set_global_var(48, global_var(48) + -5);
+        set_global_var(GVAR_TOWN_REP_SEDIT, global_var(GVAR_TOWN_REP_SEDIT) + -5);
         debug_msg("Added " + -5 + " to Town Rep");
     end
 end
@@ -436,17 +436,17 @@ begin
         restock_obj := obj_carrying_pid_obj(self_obj, 4);
         if (obj_type(self_obj) == 1) then begin
             restock_obj := obj_carrying_pid_obj(self_obj, 4);
-            if (critter_inven_obj(self_obj, 0) == restock_obj) then begin
-                if (obj_item_subtype(critter_inven_obj(self_obj, 0)) == 0) then begin
-                    restock_obj := critter_inven_obj(self_obj, 0);
+            if (critter_inven_obj(self_obj, INVEN_TYPE_WORN) == restock_obj) then begin
+                if (obj_item_subtype(critter_inven_obj(self_obj, INVEN_TYPE_WORN)) == 0) then begin
+                    restock_obj := critter_inven_obj(self_obj, INVEN_TYPE_WORN);
                     debug_msg("armour pid == " + obj_pid(restock_obj));
                     rm_obj_from_inven(self_obj, restock_obj);
                     add_obj_to_inven(self_obj, restock_obj);
                 end
             end
             else begin
-                if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                    metarule(43, self_obj);
+                if ((critter_inven_obj(self_obj, INVEN_TYPE_LEFT_HAND) == restock_obj) or (critter_inven_obj(self_obj, INVEN_TYPE_RIGHT_HAND) == restock_obj)) then begin
+                    inven_unwield(self_obj);
                 end
             end
         end
@@ -473,293 +473,293 @@ end
 
 procedure Node500
 begin
-    gsay_reply(1370, 500);
+    Reply(500);
     if (local_var(7) != 1) then begin
-        giq_option(4, 1370, message_str(1370, 501) + obj_name(dude_obj), Node700, 50);
+        giq_option(4, NAME, mstr(501) + dude_name, Node700, NEUTRAL_REACTION);
     end
-    giq_option(4, 1370, 502, Node550, 50);
-    if ((global_var(725) == 3) and (local_var(8) != 1)) then begin
-        if (get_critter_stat(dude_obj, 4) > 5) then begin
-            giq_option(4, 1370, 503, Node850, 50);
+    giq_option(4, NAME, 502, Node550, NEUTRAL_REACTION);
+    if ((global_var(GVAR_QUEST_S_NICOL_NECKLACE) == 3) and (local_var(8) != 1)) then begin
+        if (dude_iq > 5) then begin
+            giq_option(4, NAME, 503, Node850, NEUTRAL_REACTION);
         end
         else begin
-            if (global_var(719) >= 2) then begin
-                giq_option(4, 1370, 507, Node900, 50);
+            if (global_var(GVAR_QUEST_S_TYLER_NICOL) >= 2) then begin
+                giq_option(4, NAME, 507, Node900, NEUTRAL_REACTION);
             end
             else begin
-                giq_option(4, 1370, 504, Node900, 50);
+                giq_option(4, NAME, 504, Node900, NEUTRAL_REACTION);
             end
         end
     end
-    giq_option(-3, 1370, 505, Node950, 50);
-    giq_option(4, 1370, 506, Node999, 50);
+    giq_option(-3, NAME, 505, Node950, NEUTRAL_REACTION);
+    giq_option(4, NAME, 506, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node550
 begin
-    gsay_reply(1370, 550);
-    giq_option(4, 1370, 551, Barter, 50);
-    giq_option(4, 1370, 552, Choose_upgrade_weapon, 50);
-    giq_option(6, 1370, 553, Node600, 50);
-    giq_option(4, 1370, 554, Node999, 50);
+    Reply(550);
+    giq_option(4, NAME, 551, Barter, NEUTRAL_REACTION);
+    giq_option(4, NAME, 552, Choose_upgrade_weapon, NEUTRAL_REACTION);
+    giq_option(6, NAME, 553, Node600, NEUTRAL_REACTION);
+    giq_option(4, NAME, 554, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node600
 begin
-    gsay_reply(1370, 600);
-    giq_option(4, 1370, 601, Barter, 50);
-    giq_option(4, 1370, 602, Choose_upgrade_weapon, 50);
+    Reply(600);
+    giq_option(4, NAME, 601, Barter, NEUTRAL_REACTION);
+    giq_option(4, NAME, 602, Choose_upgrade_weapon, NEUTRAL_REACTION);
     if (local_var(6) != 1) then begin
-        giq_option(6, 1370, 603, Node650, 50);
+        giq_option(6, NAME, 603, Node650, NEUTRAL_REACTION);
     end
-    giq_option(4, 1370, 604, Node999, 50);
+    giq_option(4, NAME, 604, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node650
 begin
-    gsay_reply(1370, 650);
+    Reply(650);
     set_local_var(6, 1);
-    giq_option(4, 1370, 651, Barter, 50);
-    giq_option(4, 1370, 652, Choose_upgrade_weapon, 50);
-    giq_option(4, 1370, 653, Node999, 50);
+    giq_option(4, NAME, 651, Barter, NEUTRAL_REACTION);
+    giq_option(4, NAME, 652, Choose_upgrade_weapon, NEUTRAL_REACTION);
+    giq_option(4, NAME, 653, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node700
 begin
-    gsay_reply(1370, 700);
+    Reply(700);
     set_local_var(7, 1);
     if (local_var(11) == 1) then begin
-        giq_option(4, 1370, 701, Node750, 50);
+        giq_option(4, NAME, 701, Node750, NEUTRAL_REACTION);
     end
-    if (get_critter_stat(dude_obj, 34) == 0) then begin
-        giq_option(4, 1370, 702, Node800, 50);
+    if (dude_gender == 0) then begin
+        giq_option(4, NAME, 702, Node800, NEUTRAL_REACTION);
     end
-    giq_option(4, 1370, 703, Node550, 50);
+    giq_option(4, NAME, 703, Node550, NEUTRAL_REACTION);
 end
 
 procedure Node750
 begin
-    gsay_reply(1370, 750);
-    if (get_critter_stat(dude_obj, 34) == 0) then begin
-        giq_option(4, 1370, 751, Node550, 50);
-        giq_option(4, 1370, 753, Node800, 50);
+    Reply(750);
+    if (dude_gender == 0) then begin
+        giq_option(4, NAME, 751, Node550, NEUTRAL_REACTION);
+        giq_option(4, NAME, 753, Node800, NEUTRAL_REACTION);
     end
-    if (get_critter_stat(dude_obj, 34) == 1) then begin
-        giq_option(4, 1370, 752, Node550, 50);
+    if (dude_gender == 1) then begin
+        giq_option(4, NAME, 752, Node550, NEUTRAL_REACTION);
     end
 end
 
 procedure Node800
 begin
-    gsay_reply(1370, 800);
-    giq_option(4, 1370, 801, Node550, 50);
-    giq_option(4, 1370, 802, Node998, 50);
+    Reply(800);
+    giq_option(4, NAME, 801, Node550, NEUTRAL_REACTION);
+    giq_option(4, NAME, 802, Node998, NEUTRAL_REACTION);
 end
 
 procedure Node850
 begin
-    gsay_reply(1370, 850);
-    if (global_var(719) >= 2) then begin
-        giq_option(4, 1370, 853 + (get_critter_stat(dude_obj, 34) == 1), Node900, 50);
+    Reply(850);
+    if (global_var(GVAR_QUEST_S_TYLER_NICOL) >= 2) then begin
+        giq_option(4, NAME, 853 + (dude_gender == 1), Node900, NEUTRAL_REACTION);
     end
     else begin
-        giq_option(4, 1370, 851 + (get_critter_stat(dude_obj, 34) == 1), Node900, 50);
+        giq_option(4, NAME, 851 + (dude_gender == 1), Node900, NEUTRAL_REACTION);
     end
 end
 
 procedure Node900
 begin
-    gsay_reply(1370, 900);
+    Reply(900);
     set_local_var(8, 1);
-    giq_option(4, 1370, 901, Barter, 50);
-    giq_option(4, 1370, 902, Choose_upgrade_weapon, 50);
-    giq_option(4, 1370, 903, Node999, 50);
+    giq_option(4, NAME, 901, Barter, NEUTRAL_REACTION);
+    giq_option(4, NAME, 902, Choose_upgrade_weapon, NEUTRAL_REACTION);
+    giq_option(4, NAME, 903, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node950
 begin
-    gsay_reply(1370, 950);
-    giq_option(-3, 1370, 951, Barter, 50);
-    giq_option(-3, 1370, 952, Node999, 50);
+    Reply(950);
+    giq_option(-3, NAME, 951, Barter, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 952, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node1000
 begin
-    gsay_reply(1370, 1000);
+    Reply(1000);
     if (local_var(7) != 1) then begin
-        giq_option(4, 1370, message_str(1370, 1001) + obj_name(dude_obj), Node700, 50);
+        giq_option(4, NAME, mstr(1001) + dude_name, Node700, NEUTRAL_REACTION);
     end
-    giq_option(4, 1370, 1002, Node550, 50);
-    if ((global_var(725) == 3) and (local_var(8) != 1)) then begin
-        if (get_critter_stat(dude_obj, 4) > 5) then begin
-            giq_option(4, 1370, 1003, Node850, 50);
+    giq_option(4, NAME, 1002, Node550, NEUTRAL_REACTION);
+    if ((global_var(GVAR_QUEST_S_NICOL_NECKLACE) == 3) and (local_var(8) != 1)) then begin
+        if (dude_iq > 5) then begin
+            giq_option(4, NAME, 1003, Node850, NEUTRAL_REACTION);
         end
         else begin
-            if (global_var(719) >= 2) then begin
-                giq_option(4, 1370, 1007, Node900, 50);
+            if (global_var(GVAR_QUEST_S_TYLER_NICOL) >= 2) then begin
+                giq_option(4, NAME, 1007, Node900, NEUTRAL_REACTION);
             end
             else begin
-                giq_option(4, 1370, 1004, Node900, 50);
+                giq_option(4, NAME, 1004, Node900, NEUTRAL_REACTION);
             end
         end
     end
-    giq_option(-3, 1370, 1005, Node950, 50);
-    giq_option(4, 1370, 1006, Node999, 50);
+    giq_option(-3, NAME, 1005, Node950, NEUTRAL_REACTION);
+    giq_option(4, NAME, 1006, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node1050
 begin
-    gsay_reply(1370, 1050);
-    giq_option(4, 1370, 1051, Node999, 50);
-    giq_option(-3, 1370, 1052, Node999, 50);
+    Reply(1050);
+    giq_option(4, NAME, 1051, Node999, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 1052, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node1100
 begin
-    gsay_reply(1370, 1100);
-    giq_option(4, 1370, 1101, Node999swap, 50);
+    Reply(1100);
+    giq_option(4, NAME, 1101, Node999swap, NEUTRAL_REACTION);
 end
 
 procedure Node1150
 begin
-    gsay_reply(1370, 1150);
-    giq_option(-3, 1370, 1151, Node999swap, 50);
+    Reply(1150);
+    giq_option(-3, NAME, 1151, Node999swap, NEUTRAL_REACTION);
 end
 
 procedure Node1200
 begin
-    gsay_reply(1370, 1200);
-    giq_option(4, 1370, 1201, Node999, 50);
+    Reply(1200);
+    giq_option(4, NAME, 1201, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node1250
 begin
-    gsay_reply(1370, 1250);
+    Reply(1250);
     weapon := 1;
     if (item_caps_total(dude_obj) >= 2000) then begin
-        giq_option(4, 1370, 1251, Node1600, 50);
-        giq_option(4, 1370, 1253, Choose_upgrade_weapon, 50);
+        giq_option(4, NAME, 1251, Node1600, NEUTRAL_REACTION);
+        giq_option(4, NAME, 1253, Choose_upgrade_weapon, NEUTRAL_REACTION);
     end
     else begin
-        giq_option(4, 1370, 1252, Choose_upgrade_weapon, 50);
+        giq_option(4, NAME, 1252, Choose_upgrade_weapon, NEUTRAL_REACTION);
     end
-    giq_option(4, 1370, 1254, Node999, 50);
+    giq_option(4, NAME, 1254, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node1300
 begin
-    gsay_reply(1370, 1300);
+    Reply(1300);
     weapon := 2;
     if (item_caps_total(dude_obj) >= 2000) then begin
-        giq_option(4, 1370, 1301, Node1500, 50);
-        giq_option(4, 1370, 1303, Choose_upgrade_weapon, 50);
+        giq_option(4, NAME, 1301, Node1500, NEUTRAL_REACTION);
+        giq_option(4, NAME, 1303, Choose_upgrade_weapon, NEUTRAL_REACTION);
     end
     else begin
-        giq_option(4, 1370, 1302, Choose_upgrade_weapon, 50);
+        giq_option(4, NAME, 1302, Choose_upgrade_weapon, NEUTRAL_REACTION);
     end
-    giq_option(4, 1370, 1304, Node999, 50);
+    giq_option(4, NAME, 1304, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node1350
 begin
-    gsay_reply(1370, 1350);
+    Reply(1350);
     weapon := 3;
     if (item_caps_total(dude_obj) >= 1500) then begin
-        giq_option(4, 1370, 1351, Node1500, 50);
-        giq_option(4, 1370, 1353, Choose_upgrade_weapon, 50);
+        giq_option(4, NAME, 1351, Node1500, NEUTRAL_REACTION);
+        giq_option(4, NAME, 1353, Choose_upgrade_weapon, NEUTRAL_REACTION);
     end
     else begin
-        giq_option(4, 1370, 1352, Choose_upgrade_weapon, 50);
+        giq_option(4, NAME, 1352, Choose_upgrade_weapon, NEUTRAL_REACTION);
     end
-    giq_option(4, 1370, 1354, Node999, 50);
+    giq_option(4, NAME, 1354, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node1400
 begin
-    gsay_reply(1370, 1400);
+    Reply(1400);
     weapon := 4;
     if (item_caps_total(dude_obj) >= 4000) then begin
-        giq_option(4, 1370, 1401, Node1650, 50);
-        giq_option(4, 1370, 1403, Choose_upgrade_weapon, 50);
+        giq_option(4, NAME, 1401, Node1650, NEUTRAL_REACTION);
+        giq_option(4, NAME, 1403, Choose_upgrade_weapon, NEUTRAL_REACTION);
     end
     else begin
-        giq_option(4, 1370, 1402, Choose_upgrade_weapon, 50);
+        giq_option(4, NAME, 1402, Choose_upgrade_weapon, NEUTRAL_REACTION);
     end
-    giq_option(4, 1370, 1404, Node999, 50);
+    giq_option(4, NAME, 1404, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node1410
 begin
-    gsay_reply(1370, 1410);
+    Reply(1410);
     weapon := 5;
     if (item_caps_total(dude_obj) >= 3000) then begin
-        giq_option(4, 1370, 1411, Node1600, 50);
-        giq_option(4, 1370, 1413, Choose_upgrade_weapon, 50);
+        giq_option(4, NAME, 1411, Node1600, NEUTRAL_REACTION);
+        giq_option(4, NAME, 1413, Choose_upgrade_weapon, NEUTRAL_REACTION);
     end
     else begin
-        giq_option(4, 1370, 1412, Choose_upgrade_weapon, 50);
+        giq_option(4, NAME, 1412, Choose_upgrade_weapon, NEUTRAL_REACTION);
     end
-    giq_option(4, 1370, 1414, Node999, 50);
+    giq_option(4, NAME, 1414, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node1420
 begin
-    gsay_reply(1370, 1420);
+    Reply(1420);
     weapon := 6;
     if (item_caps_total(dude_obj) >= 2000) then begin
-        giq_option(4, 1370, 1421, Node1550, 50);
-        giq_option(4, 1370, 1423, Choose_upgrade_weapon, 50);
+        giq_option(4, NAME, 1421, Node1550, NEUTRAL_REACTION);
+        giq_option(4, NAME, 1423, Choose_upgrade_weapon, NEUTRAL_REACTION);
     end
     else begin
-        giq_option(4, 1370, 1422, Choose_upgrade_weapon, 50);
+        giq_option(4, NAME, 1422, Choose_upgrade_weapon, NEUTRAL_REACTION);
     end
-    giq_option(4, 1370, 1424, Node999, 50);
+    giq_option(4, NAME, 1424, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node1430
 begin
-    gsay_reply(1370, 1430);
+    Reply(1430);
     weapon := 7;
     if (item_caps_total(dude_obj) >= 4000) then begin
-        giq_option(4, 1370, 1431, Node1650, 50);
-        giq_option(4, 1370, 1433, Choose_upgrade_weapon, 50);
+        giq_option(4, NAME, 1431, Node1650, NEUTRAL_REACTION);
+        giq_option(4, NAME, 1433, Choose_upgrade_weapon, NEUTRAL_REACTION);
     end
     else begin
-        giq_option(4, 1370, 1432, Choose_upgrade_weapon, 50);
+        giq_option(4, NAME, 1432, Choose_upgrade_weapon, NEUTRAL_REACTION);
     end
-    giq_option(4, 1370, 1434, Node999, 50);
+    giq_option(4, NAME, 1434, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node1450
 begin
-    gsay_reply(1370, 1450);
+    Reply(1450);
     if (obj_carrying_pid_obj(dude_obj, 18) and (item_caps_total(dude_obj) >= 2000)) then begin
-        giq_option(4, 1370, 1451, UpgradeDesert, 50);
+        giq_option(4, NAME, 1451, UpgradeDesert, NEUTRAL_REACTION);
     end
     if (obj_carrying_pid_obj(dude_obj, 10) and (item_caps_total(dude_obj) >= 2000)) then begin
-        giq_option(4, 1370, 1452, UpgradeHunter, 50);
+        giq_option(4, NAME, 1452, UpgradeHunter, NEUTRAL_REACTION);
     end
     if (obj_carrying_pid_obj(dude_obj, 94) and (item_caps_total(dude_obj) >= 1500)) then begin
-        giq_option(4, 1370, 1453, UpgradeDoubleShotgun, 50);
+        giq_option(4, NAME, 1453, UpgradeDoubleShotgun, NEUTRAL_REACTION);
     end
     if (obj_carrying_pid_obj(dude_obj, 23) and (item_caps_total(dude_obj) >= 4000)) then begin
-        giq_option(4, 1370, 1454, UpgradeAssault, 50);
+        giq_option(4, NAME, 1454, UpgradeAssault, NEUTRAL_REACTION);
     end
     if (obj_carrying_pid_obj(dude_obj, 143) and (item_caps_total(dude_obj) >= 3000)) then begin
-        giq_option(4, 1370, 1455, UpgradeSniper, 50);
+        giq_option(4, NAME, 1455, UpgradeSniper, NEUTRAL_REACTION);
     end
     if (obj_carrying_pid_obj(dude_obj, 313) and (item_caps_total(dude_obj) >= 2000)) then begin
-        giq_option(4, 1370, 1456, Node1550, 50);
+        giq_option(4, NAME, 1456, Node1550, NEUTRAL_REACTION);
     end
     if (obj_carrying_pid_obj(dude_obj, 242) and (item_caps_total(dude_obj) >= 4000)) then begin
-        giq_option(4, 1370, 1457, UpgradeCombatShotgun, 50);
+        giq_option(4, NAME, 1457, UpgradeCombatShotgun, NEUTRAL_REACTION);
     end
-    giq_option(4, 1370, 1458, Node999, 50);
+    giq_option(4, NAME, 1458, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node1500
 begin
-    gsay_reply(1370, 1500);
+    Reply(1500);
     if (weapon == 3) then begin
         item_caps_adjust(dude_obj, -1500);
         removed_qty := obj_is_carrying_obj_pid(dude_obj, 94);
@@ -770,17 +770,17 @@ begin
             restock_obj := obj_carrying_pid_obj(dude_obj, 94);
             if (obj_type(self_obj) == 1) then begin
                 restock_obj := obj_carrying_pid_obj(dude_obj, 94);
-                if (critter_inven_obj(self_obj, 0) == restock_obj) then begin
-                    if (obj_item_subtype(critter_inven_obj(dude_obj, 0)) == 0) then begin
-                        restock_obj := critter_inven_obj(dude_obj, 0);
+                if (critter_inven_obj(self_obj, INVEN_TYPE_WORN) == restock_obj) then begin
+                    if (obj_item_subtype(critter_inven_obj(dude_obj, INVEN_TYPE_WORN)) == 0) then begin
+                        restock_obj := critter_inven_obj(dude_obj, INVEN_TYPE_WORN);
                         debug_msg("armour pid == " + obj_pid(restock_obj));
                         rm_obj_from_inven(dude_obj, restock_obj);
                         add_obj_to_inven(dude_obj, restock_obj);
                     end
                 end
                 else begin
-                    if ((critter_inven_obj(dude_obj, 2) == restock_obj) or (critter_inven_obj(dude_obj, 1) == restock_obj)) then begin
-                        metarule(43, dude_obj);
+                    if ((critter_inven_obj(dude_obj, INVEN_TYPE_LEFT_HAND) == restock_obj) or (critter_inven_obj(dude_obj, INVEN_TYPE_RIGHT_HAND) == restock_obj)) then begin
+                        inven_unwield(dude_obj);
                     end
                 end
             end
@@ -812,17 +812,17 @@ begin
             restock_obj := obj_carrying_pid_obj(dude_obj, 10);
             if (obj_type(self_obj) == 1) then begin
                 restock_obj := obj_carrying_pid_obj(dude_obj, 10);
-                if (critter_inven_obj(self_obj, 0) == restock_obj) then begin
-                    if (obj_item_subtype(critter_inven_obj(dude_obj, 0)) == 0) then begin
-                        restock_obj := critter_inven_obj(dude_obj, 0);
+                if (critter_inven_obj(self_obj, INVEN_TYPE_WORN) == restock_obj) then begin
+                    if (obj_item_subtype(critter_inven_obj(dude_obj, INVEN_TYPE_WORN)) == 0) then begin
+                        restock_obj := critter_inven_obj(dude_obj, INVEN_TYPE_WORN);
                         debug_msg("armour pid == " + obj_pid(restock_obj));
                         rm_obj_from_inven(dude_obj, restock_obj);
                         add_obj_to_inven(dude_obj, restock_obj);
                     end
                 end
                 else begin
-                    if ((critter_inven_obj(dude_obj, 2) == restock_obj) or (critter_inven_obj(dude_obj, 1) == restock_obj)) then begin
-                        metarule(43, dude_obj);
+                    if ((critter_inven_obj(dude_obj, INVEN_TYPE_LEFT_HAND) == restock_obj) or (critter_inven_obj(dude_obj, INVEN_TYPE_RIGHT_HAND) == restock_obj)) then begin
+                        inven_unwield(dude_obj);
                     end
                 end
             end
@@ -845,12 +845,12 @@ begin
         set_local_var(10, 287);
     end
     set_local_var(9, game_time + (24 * (60 * (60 * 10))));
-    giq_option(4, 1370, 1501, Node999, 50);
+    giq_option(4, NAME, 1501, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node1550
 begin
-    gsay_reply(1370, 1550);
+    Reply(1550);
     item_caps_adjust(dude_obj, -2000);
     removed_qty := obj_is_carrying_obj_pid(dude_obj, 313);
     if (1 < removed_qty) then begin
@@ -860,17 +860,17 @@ begin
         restock_obj := obj_carrying_pid_obj(dude_obj, 313);
         if (obj_type(self_obj) == 1) then begin
             restock_obj := obj_carrying_pid_obj(dude_obj, 313);
-            if (critter_inven_obj(self_obj, 0) == restock_obj) then begin
-                if (obj_item_subtype(critter_inven_obj(dude_obj, 0)) == 0) then begin
-                    restock_obj := critter_inven_obj(dude_obj, 0);
+            if (critter_inven_obj(self_obj, INVEN_TYPE_WORN) == restock_obj) then begin
+                if (obj_item_subtype(critter_inven_obj(dude_obj, INVEN_TYPE_WORN)) == 0) then begin
+                    restock_obj := critter_inven_obj(dude_obj, INVEN_TYPE_WORN);
                     debug_msg("armour pid == " + obj_pid(restock_obj));
                     rm_obj_from_inven(dude_obj, restock_obj);
                     add_obj_to_inven(dude_obj, restock_obj);
                 end
             end
             else begin
-                if ((critter_inven_obj(dude_obj, 2) == restock_obj) or (critter_inven_obj(dude_obj, 1) == restock_obj)) then begin
-                    metarule(43, dude_obj);
+                if ((critter_inven_obj(dude_obj, INVEN_TYPE_LEFT_HAND) == restock_obj) or (critter_inven_obj(dude_obj, INVEN_TYPE_RIGHT_HAND) == restock_obj)) then begin
+                    inven_unwield(dude_obj);
                 end
             end
         end
@@ -892,12 +892,12 @@ begin
     restock_amt := 0;
     set_local_var(10, 398);
     set_local_var(9, game_time + (2 * (24 * (60 * (60 * 10)))));
-    giq_option(4, 1370, 1551, Node999, 50);
+    giq_option(4, NAME, 1551, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node1600
 begin
-    gsay_reply(1370, 1600);
+    Reply(1600);
     if (weapon == 5) then begin
         item_caps_adjust(dude_obj, -3000);
         removed_qty := obj_is_carrying_obj_pid(dude_obj, 143);
@@ -908,17 +908,17 @@ begin
             restock_obj := obj_carrying_pid_obj(dude_obj, 143);
             if (obj_type(self_obj) == 1) then begin
                 restock_obj := obj_carrying_pid_obj(dude_obj, 143);
-                if (critter_inven_obj(self_obj, 0) == restock_obj) then begin
-                    if (obj_item_subtype(critter_inven_obj(dude_obj, 0)) == 0) then begin
-                        restock_obj := critter_inven_obj(dude_obj, 0);
+                if (critter_inven_obj(self_obj, INVEN_TYPE_WORN) == restock_obj) then begin
+                    if (obj_item_subtype(critter_inven_obj(dude_obj, INVEN_TYPE_WORN)) == 0) then begin
+                        restock_obj := critter_inven_obj(dude_obj, INVEN_TYPE_WORN);
                         debug_msg("armour pid == " + obj_pid(restock_obj));
                         rm_obj_from_inven(dude_obj, restock_obj);
                         add_obj_to_inven(dude_obj, restock_obj);
                     end
                 end
                 else begin
-                    if ((critter_inven_obj(dude_obj, 2) == restock_obj) or (critter_inven_obj(dude_obj, 1) == restock_obj)) then begin
-                        metarule(43, dude_obj);
+                    if ((critter_inven_obj(dude_obj, INVEN_TYPE_LEFT_HAND) == restock_obj) or (critter_inven_obj(dude_obj, INVEN_TYPE_RIGHT_HAND) == restock_obj)) then begin
+                        inven_unwield(dude_obj);
                     end
                 end
             end
@@ -950,17 +950,17 @@ begin
             restock_obj := obj_carrying_pid_obj(dude_obj, 18);
             if (obj_type(self_obj) == 1) then begin
                 restock_obj := obj_carrying_pid_obj(dude_obj, 18);
-                if (critter_inven_obj(self_obj, 0) == restock_obj) then begin
-                    if (obj_item_subtype(critter_inven_obj(dude_obj, 0)) == 0) then begin
-                        restock_obj := critter_inven_obj(dude_obj, 0);
+                if (critter_inven_obj(self_obj, INVEN_TYPE_WORN) == restock_obj) then begin
+                    if (obj_item_subtype(critter_inven_obj(dude_obj, INVEN_TYPE_WORN)) == 0) then begin
+                        restock_obj := critter_inven_obj(dude_obj, INVEN_TYPE_WORN);
                         debug_msg("armour pid == " + obj_pid(restock_obj));
                         rm_obj_from_inven(dude_obj, restock_obj);
                         add_obj_to_inven(dude_obj, restock_obj);
                     end
                 end
                 else begin
-                    if ((critter_inven_obj(dude_obj, 2) == restock_obj) or (critter_inven_obj(dude_obj, 1) == restock_obj)) then begin
-                        metarule(43, dude_obj);
+                    if ((critter_inven_obj(dude_obj, INVEN_TYPE_LEFT_HAND) == restock_obj) or (critter_inven_obj(dude_obj, INVEN_TYPE_RIGHT_HAND) == restock_obj)) then begin
+                        inven_unwield(dude_obj);
                     end
                 end
             end
@@ -983,12 +983,12 @@ begin
         set_local_var(10, 404);
     end
     set_local_var(9, game_time + (3 * (24 * (60 * (60 * 10)))));
-    giq_option(4, 1370, 1601, Node999, 50);
+    giq_option(4, NAME, 1601, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node1650
 begin
-    gsay_reply(1370, 1650);
+    Reply(1650);
     if (weapon == 7) then begin
         item_caps_adjust(dude_obj, -4000);
         removed_qty := obj_is_carrying_obj_pid(dude_obj, 242);
@@ -999,17 +999,17 @@ begin
             restock_obj := obj_carrying_pid_obj(dude_obj, 242);
             if (obj_type(self_obj) == 1) then begin
                 restock_obj := obj_carrying_pid_obj(dude_obj, 242);
-                if (critter_inven_obj(self_obj, 0) == restock_obj) then begin
-                    if (obj_item_subtype(critter_inven_obj(dude_obj, 0)) == 0) then begin
-                        restock_obj := critter_inven_obj(dude_obj, 0);
+                if (critter_inven_obj(self_obj, INVEN_TYPE_WORN) == restock_obj) then begin
+                    if (obj_item_subtype(critter_inven_obj(dude_obj, INVEN_TYPE_WORN)) == 0) then begin
+                        restock_obj := critter_inven_obj(dude_obj, INVEN_TYPE_WORN);
                         debug_msg("armour pid == " + obj_pid(restock_obj));
                         rm_obj_from_inven(dude_obj, restock_obj);
                         add_obj_to_inven(dude_obj, restock_obj);
                     end
                 end
                 else begin
-                    if ((critter_inven_obj(dude_obj, 2) == restock_obj) or (critter_inven_obj(dude_obj, 1) == restock_obj)) then begin
-                        metarule(43, dude_obj);
+                    if ((critter_inven_obj(dude_obj, INVEN_TYPE_LEFT_HAND) == restock_obj) or (critter_inven_obj(dude_obj, INVEN_TYPE_RIGHT_HAND) == restock_obj)) then begin
+                        inven_unwield(dude_obj);
                     end
                 end
             end
@@ -1041,17 +1041,17 @@ begin
             restock_obj := obj_carrying_pid_obj(dude_obj, 23);
             if (obj_type(self_obj) == 1) then begin
                 restock_obj := obj_carrying_pid_obj(dude_obj, 23);
-                if (critter_inven_obj(self_obj, 0) == restock_obj) then begin
-                    if (obj_item_subtype(critter_inven_obj(dude_obj, 0)) == 0) then begin
-                        restock_obj := critter_inven_obj(dude_obj, 0);
+                if (critter_inven_obj(self_obj, INVEN_TYPE_WORN) == restock_obj) then begin
+                    if (obj_item_subtype(critter_inven_obj(dude_obj, INVEN_TYPE_WORN)) == 0) then begin
+                        restock_obj := critter_inven_obj(dude_obj, INVEN_TYPE_WORN);
                         debug_msg("armour pid == " + obj_pid(restock_obj));
                         rm_obj_from_inven(dude_obj, restock_obj);
                         add_obj_to_inven(dude_obj, restock_obj);
                     end
                 end
                 else begin
-                    if ((critter_inven_obj(dude_obj, 2) == restock_obj) or (critter_inven_obj(dude_obj, 1) == restock_obj)) then begin
-                        metarule(43, dude_obj);
+                    if ((critter_inven_obj(dude_obj, INVEN_TYPE_LEFT_HAND) == restock_obj) or (critter_inven_obj(dude_obj, INVEN_TYPE_RIGHT_HAND) == restock_obj)) then begin
+                        inven_unwield(dude_obj);
                     end
                 end
             end
@@ -1074,37 +1074,37 @@ begin
         set_local_var(10, 405);
     end
     set_local_var(9, game_time + (4 * (24 * (60 * (60 * 10)))));
-    giq_option(4, 1370, 1651, Node999, 50);
+    giq_option(4, NAME, 1651, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node1700
 begin
-    gsay_reply(1370, 1700);
-    giq_option(4, 1370, 1701, Barter, 50);
-    giq_option(4, 1370, 1702, Node999, 50);
-    giq_option(-3, 1370, 1703, Barter, 50);
-    giq_option(-3, 1370, 1704, Node999, 50);
+    Reply(1700);
+    giq_option(4, NAME, 1701, Barter, NEUTRAL_REACTION);
+    giq_option(4, NAME, 1702, Node999, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 1703, Barter, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 1704, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node1750
 begin
-    gsay_reply(1370, 1750);
-    restock_obj := create_object_sid(local_var(10), 0, 0, -1);
+    Reply(1750);
+    restock_obj := create_object(local_var(10), 0, 0);
     add_mult_objs_to_inven(dude_obj, restock_obj, 1);
     set_local_var(10, 0);
-    giq_option(4, 1370, 1751, Node999, 50);
-    giq_option(-3, 1370, 1752, Node999, 50);
+    giq_option(4, NAME, 1751, Node999, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 1752, Node999, NEUTRAL_REACTION);
 end
 
 procedure Barter
 begin
     table_swap := tile_contains_pid_obj(16473, 0, 168);
     move_obj_inven_to_obj(Sabrina, table_swap);
-    restock_obj := create_object_sid(4, 0, 0, -1);
+    restock_obj := create_object(4, 0, 0);
     add_mult_objs_to_inven(self_obj, restock_obj, 1);
     move_obj_inven_to_obj(Sabrina_table, Sabrina);
     if (local_var(8) == 1) then begin
-        if ((party_member_obj(16777779) != 0) and ((global_var(910) bwand 32) == 0)) then begin
+        if ((party_member_obj(16777779) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 32) == 0)) then begin
             gdialog_mod_barter(50 - 15);
         end
         else begin
@@ -1112,14 +1112,14 @@ begin
         end
     end
     else begin
-        if ((party_member_obj(16777779) != 0) and ((global_var(910) bwand 32) == 0)) then begin
+        if ((party_member_obj(16777779) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 32) == 0)) then begin
             gdialog_mod_barter(75 - 15);
         end
         else begin
             gdialog_mod_barter(75);
         end
     end
-    if ((get_critter_stat(dude_obj, 4) + has_trait(0, dude_obj, 49)) <= 3) then begin
+    if ((dude_iq + has_trait(TRAIT_PERK, dude_obj, PERK_smooth_talker)) <= 3) then begin
         call Node1150();
     end
     else begin
@@ -1217,4 +1217,3 @@ begin
     weapon := 4;
     call Node1650();
 end
-
