diff --git a/Mods/Resurrection/Original/SSALEMT2.ssl b/Mods/Resurrection/ReDefine/SSALEMT2.ssl
index de80597..dea3df4 100644
--- a/Mods/Resurrection/Original/SSALEMT2.ssl
+++ b/Mods/Resurrection/ReDefine/SSALEMT2.ssl
@@ -46,12 +46,12 @@ begin
             return 1;
         end
     end
-    if ((party_member_obj(16777779) != 0) and ((global_var(910) bwand 32) == 0)) then begin
+    if ((party_member_obj(16777779) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 32) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777779)) <= 5) then begin
             return 1;
         end
     end
-    if ((party_member_obj(16777811) != 0) and ((global_var(910) bwand 64) == 0)) then begin
+    if ((party_member_obj(16777811) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 64) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777811)) <= 5) then begin
             return 1;
         end
@@ -66,7 +66,7 @@ begin
             return 1;
         end
     end
-    if ((party_member_obj(16777814) != 0) and ((global_var(910) bwand 128) == 0)) then begin
+    if ((party_member_obj(16777814) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 128) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777814)) <= 5) then begin
             return 1;
         end
@@ -124,7 +124,7 @@ end
 
 procedure use_p_proc
 begin
-    if (salem and (salem != -1) and (global_var(769) != 1)) then begin
+    if (salem and (salem != -1) and (global_var(GVAR_ENEMY_SEDIT) != 1)) then begin
         script_overrides;
         add_timer_event(salem, 0, 10);
     end
@@ -140,14 +140,14 @@ end
 
 procedure use_skill_on_p_proc
 begin
-    if (salem and (salem != -1) and (global_var(769) != 1)) then begin
+    if (salem and (salem != -1) and (global_var(GVAR_ENEMY_SEDIT) != 1)) then begin
         call use_p_proc();
     end
 end
 
 procedure use_obj_on_p_proc
 begin
-    if (salem and (salem != -1) and (global_var(769) != 1)) then begin
+    if (salem and (salem != -1) and (global_var(GVAR_ENEMY_SEDIT) != 1)) then begin
         call use_p_proc();
     end
 end
@@ -167,4 +167,3 @@ end
 procedure timed_event_p_proc
 begin
 end
-
