diff --git a/Mods/Resurrection/Original/SSECRTDR.ssl b/Mods/Resurrection/ReDefine/SSECRTDR.ssl
index 8b2bd70..cdb477f 100644
--- a/Mods/Resurrection/Original/SSECRTDR.ssl
+++ b/Mods/Resurrection/ReDefine/SSECRTDR.ssl
@@ -64,12 +64,12 @@ begin
             return 1;
         end
     end
-    if ((party_member_obj(16777779) != 0) and ((global_var(910) bwand 32) == 0)) then begin
+    if ((party_member_obj(16777779) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 32) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777779)) <= 5) then begin
             return 1;
         end
     end
-    if ((party_member_obj(16777811) != 0) and ((global_var(910) bwand 64) == 0)) then begin
+    if ((party_member_obj(16777811) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 64) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777811)) <= 5) then begin
             return 1;
         end
@@ -84,7 +84,7 @@ begin
             return 1;
         end
     end
-    if ((party_member_obj(16777814) != 0) and ((global_var(910) bwand 128) == 0)) then begin
+    if ((party_member_obj(16777814) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 128) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777814)) <= 5) then begin
             return 1;
         end
@@ -143,30 +143,30 @@ procedure use_p_proc
 begin
     if (obj_is_locked(self_obj)) then begin
         script_overrides;
-        display_msg(message_str(1367, 203));
+        display_mstr(203);
     end
 end
 
 procedure look_at_p_proc
 begin
     script_overrides;
-    if (get_critter_stat(dude_obj, 1) > 4) then begin
-        display_msg(message_str(1367, 101));
+    if (dude_perception > 4) then begin
+        display_mstr(101);
     end
     else begin
-        display_msg(message_str(1367, 100));
+        display_mstr(100);
     end
 end
 
 procedure description_p_proc
 begin
-    if (get_critter_stat(dude_obj, 1) > 4) then begin
+    if (dude_perception > 4) then begin
         script_overrides;
         if (obj_is_locked(self_obj)) then begin
-            display_msg(message_str(1367, 102));
+            display_mstr(102);
         end
         else begin
-            display_msg(message_str(1367, 103));
+            display_mstr(103);
         end
     end
 end
@@ -191,10 +191,10 @@ begin
             end
             else begin
                 if (source_obj == dude_obj) then begin
-                    display_msg(message_str(1367, 195));
+                    display_mstr(195);
                 end
                 else begin
-                    display_msg(obj_name(source_obj) + message_str(1367, 200));
+                    display_msg(obj_name(source_obj) + mstr(200));
                 end
             end
         end
@@ -284,12 +284,12 @@ end
 procedure damage_p_proc
 begin
     variable LVar0 := 0;
-    if (metarule(49, target_obj) == 6) then begin
+    if (weapon_dmg_type(target_obj) == 6) then begin
         if (0 == 0) then begin
             if (local_var(0) == 0) then begin
                 set_local_var(0, 1);
                 obj_lock(self_obj);
-                LVar0 := random(20, 20);
+                LVar0 := 20;
                 explosion(tile_num(self_obj), elevation(self_obj), LVar0);
                 call destroy_p_proc();
                 destroy_object(self_obj);
@@ -306,7 +306,7 @@ begin
                 set_local_var(3, local_var(3) + 1);
                 if (local_var(0) == 0) then begin
                     obj_unlock(self_obj);
-                    LVar0 := random(20, 20);
+                    LVar0 := 20;
                     explosion(tile_num(self_obj), elevation(self_obj), LVar0);
                 end
                 else begin
@@ -321,14 +321,14 @@ begin
         end
     end
     else begin
-        display_msg(message_str(1367, 193));
+        display_mstr(193);
     end
 end
 
 procedure map_enter_p_proc
 begin
     obj_close(self_obj);
-    if (metarule(22, 0) == 0) then begin
+    if (is_loading_game == 0) then begin
         if (local_var(2) == 0) then begin
             set_local_var(2, 1);
             obj_unlock(self_obj);
@@ -372,36 +372,36 @@ begin
                 if (is_critical(LVar1)) then begin
                     if (is_success(LVar2)) then begin
                         if (is_critical(LVar2)) then begin
-                            display_msg(message_str(1367, 102));
+                            display_mstr(102);
                         end
                         else begin
-                            display_msg(message_str(1367, 103));
+                            display_mstr(103);
                         end
                     end
                     else begin
                         if (is_critical(LVar2)) then begin
-                            display_msg(message_str(1367, 104));
+                            display_mstr(104);
                         end
                         else begin
-                            display_msg(message_str(1367, 105));
+                            display_mstr(105);
                         end
                     end
                 end
                 else begin
                     if (is_success(LVar2)) then begin
                         if (is_critical(LVar2)) then begin
-                            display_msg(message_str(1367, 106));
+                            display_mstr(106);
                         end
                         else begin
-                            display_msg(message_str(1367, 107));
+                            display_mstr(107);
                         end
                     end
                     else begin
                         if (is_critical(LVar2)) then begin
-                            display_msg(message_str(1367, 108));
+                            display_mstr(108);
                         end
                         else begin
-                            display_msg(message_str(1367, 109));
+                            display_mstr(109);
                         end
                     end
                 end
@@ -410,36 +410,36 @@ begin
                 if (is_critical(LVar1)) then begin
                     if (is_success(LVar2)) then begin
                         if (is_critical(LVar2)) then begin
-                            display_msg(message_str(1367, 110));
+                            display_mstr(110);
                         end
                         else begin
-                            display_msg(message_str(1367, 111));
+                            display_mstr(111);
                         end
                     end
                     else begin
                         if (is_critical(LVar2)) then begin
-                            display_msg(message_str(1367, 112));
+                            display_mstr(112);
                         end
                         else begin
-                            display_msg(message_str(1367, 113));
+                            display_mstr(113);
                         end
                     end
                 end
                 else begin
                     if (is_success(LVar2)) then begin
                         if (is_critical(LVar2)) then begin
-                            display_msg(message_str(1367, 114));
+                            display_mstr(114);
                         end
                         else begin
-                            display_msg(message_str(1367, 115));
+                            display_mstr(115);
                         end
                     end
                     else begin
                         if (is_critical(LVar2)) then begin
-                            display_msg(message_str(1367, 116));
+                            display_mstr(116);
                         end
                         else begin
-                            display_msg(message_str(1367, 117));
+                            display_mstr(117);
                         end
                     end
                 end
@@ -451,36 +451,36 @@ begin
                 if (is_critical(LVar1)) then begin
                     if (is_success(LVar2)) then begin
                         if (is_critical(LVar2)) then begin
-                            display_msg(message_str(1367, 118));
+                            display_mstr(118);
                         end
                         else begin
-                            display_msg(message_str(1367, 119));
+                            display_mstr(119);
                         end
                     end
                     else begin
                         if (is_critical(LVar2)) then begin
-                            display_msg(message_str(1367, 120));
+                            display_mstr(120);
                         end
                         else begin
-                            display_msg(message_str(1367, 121));
+                            display_mstr(121);
                         end
                     end
                 end
                 else begin
                     if (is_success(LVar2)) then begin
                         if (is_critical(LVar2)) then begin
-                            display_msg(message_str(1367, 122));
+                            display_mstr(122);
                         end
                         else begin
-                            display_msg(message_str(1367, 123));
+                            display_mstr(123);
                         end
                     end
                     else begin
                         if (is_critical(LVar2)) then begin
-                            display_msg(message_str(1367, 124));
+                            display_mstr(124);
                         end
                         else begin
-                            display_msg(message_str(1367, 125));
+                            display_mstr(125);
                         end
                     end
                 end
@@ -489,36 +489,36 @@ begin
                 if (is_critical(LVar1)) then begin
                     if (is_success(LVar2)) then begin
                         if (is_critical(LVar2)) then begin
-                            display_msg(message_str(1367, 126));
+                            display_mstr(126);
                         end
                         else begin
-                            display_msg(message_str(1367, 127));
+                            display_mstr(127);
                         end
                     end
                     else begin
                         if (is_critical(LVar2)) then begin
-                            display_msg(message_str(1367, 128));
+                            display_mstr(128);
                         end
                         else begin
-                            display_msg(message_str(1367, 129));
+                            display_mstr(129);
                         end
                     end
                 end
                 else begin
                     if (is_success(LVar2)) then begin
                         if (is_critical(LVar2)) then begin
-                            display_msg(message_str(1367, 130));
+                            display_mstr(130);
                         end
                         else begin
-                            display_msg(message_str(1367, 131));
+                            display_mstr(131);
                         end
                     end
                     else begin
                         if (is_critical(LVar2)) then begin
-                            display_msg(message_str(1367, 132));
+                            display_mstr(132);
                         end
                         else begin
-                            display_msg(message_str(1367, 133));
+                            display_mstr(133);
                         end
                     end
                 end
@@ -532,36 +532,36 @@ begin
                 if (is_critical(LVar1)) then begin
                     if (is_success(LVar2)) then begin
                         if (is_critical(LVar2)) then begin
-                            display_msg(message_str(1367, 134));
+                            display_mstr(134);
                         end
                         else begin
-                            display_msg(message_str(1367, 135));
+                            display_mstr(135);
                         end
                     end
                     else begin
                         if (is_critical(LVar2)) then begin
-                            display_msg(message_str(1367, 136));
+                            display_mstr(136);
                         end
                         else begin
-                            display_msg(message_str(1367, 137));
+                            display_mstr(137);
                         end
                     end
                 end
                 else begin
                     if (is_success(LVar2)) then begin
                         if (is_critical(LVar2)) then begin
-                            display_msg(message_str(1367, 138));
+                            display_mstr(138);
                         end
                         else begin
-                            display_msg(message_str(1367, 139));
+                            display_mstr(139);
                         end
                     end
                     else begin
                         if (is_critical(LVar2)) then begin
-                            display_msg(message_str(1367, 140));
+                            display_mstr(140);
                         end
                         else begin
-                            display_msg(message_str(1367, 141));
+                            display_mstr(141);
                         end
                     end
                 end
@@ -570,36 +570,36 @@ begin
                 if (is_critical(LVar1)) then begin
                     if (is_success(LVar2)) then begin
                         if (is_critical(LVar2)) then begin
-                            display_msg(message_str(1367, 142));
+                            display_mstr(142);
                         end
                         else begin
-                            display_msg(message_str(1367, 143));
+                            display_mstr(143);
                         end
                     end
                     else begin
                         if (is_critical(LVar2)) then begin
-                            display_msg(message_str(1367, 144));
+                            display_mstr(144);
                         end
                         else begin
-                            display_msg(message_str(1367, 145));
+                            display_mstr(145);
                         end
                     end
                 end
                 else begin
                     if (is_success(LVar2)) then begin
                         if (is_critical(LVar2)) then begin
-                            display_msg(message_str(1367, 146));
+                            display_mstr(146);
                         end
                         else begin
-                            display_msg(message_str(1367, 147));
+                            display_mstr(147);
                         end
                     end
                     else begin
                         if (is_critical(LVar2)) then begin
-                            display_msg(message_str(1367, 148));
+                            display_mstr(148);
                         end
                         else begin
-                            display_msg(message_str(1367, 149));
+                            display_mstr(149);
                         end
                     end
                 end
@@ -611,36 +611,36 @@ begin
                 if (is_critical(LVar1)) then begin
                     if (is_success(LVar2)) then begin
                         if (is_critical(LVar2)) then begin
-                            display_msg(message_str(1367, 150));
+                            display_mstr(150);
                         end
                         else begin
-                            display_msg(message_str(1367, 151));
+                            display_mstr(151);
                         end
                     end
                     else begin
                         if (is_critical(LVar2)) then begin
-                            display_msg(message_str(1367, 152));
+                            display_mstr(152);
                         end
                         else begin
-                            display_msg(message_str(1367, 153));
+                            display_mstr(153);
                         end
                     end
                 end
                 else begin
                     if (is_success(LVar2)) then begin
                         if (is_critical(LVar2)) then begin
-                            display_msg(message_str(1367, 154));
+                            display_mstr(154);
                         end
                         else begin
-                            display_msg(message_str(1367, 155));
+                            display_mstr(155);
                         end
                     end
                     else begin
                         if (is_critical(LVar2)) then begin
-                            display_msg(message_str(1367, 156));
+                            display_mstr(156);
                         end
                         else begin
-                            display_msg(message_str(1367, 157));
+                            display_mstr(157);
                         end
                     end
                 end
@@ -649,36 +649,36 @@ begin
                 if (is_critical(LVar1)) then begin
                     if (is_success(LVar2)) then begin
                         if (is_critical(LVar2)) then begin
-                            display_msg(message_str(1367, 158));
+                            display_mstr(158);
                         end
                         else begin
-                            display_msg(message_str(1367, 159));
+                            display_mstr(159);
                         end
                     end
                     else begin
                         if (is_critical(LVar2)) then begin
-                            display_msg(message_str(1367, 160));
+                            display_mstr(160);
                         end
                         else begin
-                            display_msg(message_str(1367, 161));
+                            display_mstr(161);
                         end
                     end
                 end
                 else begin
                     if (is_success(LVar2)) then begin
                         if (is_critical(LVar2)) then begin
-                            display_msg(message_str(1367, 162));
+                            display_mstr(162);
                         end
                         else begin
-                            display_msg(message_str(1367, 163));
+                            display_mstr(163);
                         end
                     end
                     else begin
                         if (is_critical(LVar2)) then begin
-                            display_msg(message_str(1367, 164));
+                            display_mstr(164);
                         end
                         else begin
-                            display_msg(message_str(1367, 165));
+                            display_mstr(165);
                         end
                     end
                 end
@@ -697,36 +697,36 @@ begin
         if (is_critical(LVar0)) then begin
             if (is_success(LVar1)) then begin
                 if (is_critical(LVar1)) then begin
-                    display_msg(message_str(1367, 104));
+                    display_mstr(104);
                 end
                 else begin
-                    display_msg(message_str(1367, 108));
+                    display_mstr(108);
                 end
             end
             else begin
                 if (is_critical(LVar1)) then begin
-                    display_msg(message_str(1367, 112));
+                    display_mstr(112);
                 end
                 else begin
-                    display_msg(message_str(1367, 116));
+                    display_mstr(116);
                 end
             end
         end
         else begin
             if (is_success(LVar1)) then begin
                 if (is_critical(LVar1)) then begin
-                    display_msg(message_str(1367, 120));
+                    display_mstr(120);
                 end
                 else begin
-                    display_msg(message_str(1367, 124));
+                    display_mstr(124);
                 end
             end
             else begin
                 if (is_critical(LVar1)) then begin
-                    display_msg(message_str(1367, 128));
+                    display_mstr(128);
                 end
                 else begin
-                    display_msg(message_str(1367, 132));
+                    display_mstr(132);
                 end
             end
         end
@@ -735,36 +735,36 @@ begin
         if (is_critical(LVar0)) then begin
             if (is_success(LVar1)) then begin
                 if (is_critical(LVar1)) then begin
-                    display_msg(message_str(1367, 136));
+                    display_mstr(136);
                 end
                 else begin
-                    display_msg(message_str(1367, 140));
+                    display_mstr(140);
                 end
             end
             else begin
                 if (is_critical(LVar1)) then begin
-                    display_msg(message_str(1367, 144));
+                    display_mstr(144);
                 end
                 else begin
-                    display_msg(message_str(1367, 148));
+                    display_mstr(148);
                 end
             end
         end
         else begin
             if (is_success(LVar1)) then begin
                 if (is_critical(LVar1)) then begin
-                    display_msg(message_str(1367, 152));
+                    display_mstr(152);
                 end
                 else begin
-                    display_msg(message_str(1367, 156));
+                    display_mstr(156);
                 end
             end
             else begin
                 if (is_critical(LVar1)) then begin
-                    display_msg(message_str(1367, 160));
+                    display_mstr(160);
                 end
                 else begin
-                    display_msg(message_str(1367, 164));
+                    display_mstr(164);
                 end
             end
         end
@@ -781,36 +781,36 @@ begin
         if (is_critical(LVar0)) then begin
             if (is_success(LVar1)) then begin
                 if (is_critical(LVar1)) then begin
-                    display_msg(message_str(1367, 114));
+                    display_mstr(114);
                 end
                 else begin
-                    display_msg(message_str(1367, 115));
+                    display_mstr(115);
                 end
             end
             else begin
                 if (is_critical(LVar1)) then begin
-                    display_msg(message_str(1367, 116));
+                    display_mstr(116);
                 end
                 else begin
-                    display_msg(message_str(1367, 117));
+                    display_mstr(117);
                 end
             end
         end
         else begin
             if (is_success(LVar1)) then begin
                 if (is_critical(LVar1)) then begin
-                    display_msg(message_str(1367, 130));
+                    display_mstr(130);
                 end
                 else begin
-                    display_msg(message_str(1367, 131));
+                    display_mstr(131);
                 end
             end
             else begin
                 if (is_critical(LVar1)) then begin
-                    display_msg(message_str(1367, 132));
+                    display_mstr(132);
                 end
                 else begin
-                    display_msg(message_str(1367, 133));
+                    display_mstr(133);
                 end
             end
         end
@@ -819,36 +819,36 @@ begin
         if (is_critical(LVar0)) then begin
             if (is_success(LVar1)) then begin
                 if (is_critical(LVar1)) then begin
-                    display_msg(message_str(1367, 146));
+                    display_mstr(146);
                 end
                 else begin
-                    display_msg(message_str(1367, 147));
+                    display_mstr(147);
                 end
             end
             else begin
                 if (is_critical(LVar1)) then begin
-                    display_msg(message_str(1367, 148));
+                    display_mstr(148);
                 end
                 else begin
-                    display_msg(message_str(1367, 149));
+                    display_mstr(149);
                 end
             end
         end
         else begin
             if (is_success(LVar1)) then begin
                 if (is_critical(LVar1)) then begin
-                    display_msg(message_str(1367, 162));
+                    display_mstr(162);
                 end
                 else begin
-                    display_msg(message_str(1367, 163));
+                    display_mstr(163);
                 end
             end
             else begin
                 if (is_critical(LVar1)) then begin
-                    display_msg(message_str(1367, 164));
+                    display_mstr(164);
                 end
                 else begin
-                    display_msg(message_str(1367, 165));
+                    display_mstr(165);
                 end
             end
         end
@@ -867,24 +867,24 @@ begin
         if (local_var(1) == 0) then begin
             set_local_var(1, 1);
             if (source_obj == dude_obj) then begin
-                display_msg(message_str(1367, 194));
+                display_mstr(194);
             end
             else begin
-                display_msg(message_str(1367, 199));
+                display_mstr(199);
             end
             if (local_var(5) == 0) then begin
                 set_local_var(5, 1);
                 give_exp_points(25);
-                display_msg(message_str(14, 100) + 25 + message_str(14, 101));
+                display_msg(g_mstr(100) + 25 + g_mstr(101));
             end
         end
         else begin
             set_local_var(0, 1);
             if (source_obj == dude_obj) then begin
-                display_msg(message_str(1367, 196));
+                display_mstr(196);
             end
             else begin
-                display_msg(obj_name(source_obj) + message_str(1367, 201));
+                display_msg(obj_name(source_obj) + mstr(201));
             end
         end
     end
@@ -895,14 +895,14 @@ begin
         else begin
             if (local_var(1) == 0) then begin
                 if (source_obj == dude_obj) then begin
-                    display_msg(message_str(1367, 195));
+                    display_mstr(195);
                 end
                 else begin
-                    display_msg(obj_name(source_obj) + message_str(1367, 200));
+                    display_msg(obj_name(source_obj) + mstr(200));
                 end
             end
             else begin
-                display_msg(obj_name(source_obj) + message_str(1367, 202));
+                display_msg(obj_name(source_obj) + mstr(202));
             end
         end
     end
@@ -926,33 +926,33 @@ begin
     if (is_success(Locks_Roll)) then begin
         obj_unlock(self_obj);
         if (source_obj == dude_obj) then begin
-            display_msg(message_str(1367, 170));
+            display_mstr(170);
         end
         else begin
-            display_msg(obj_name(source_obj) + message_str(1367, 173));
+            display_msg(obj_name(source_obj) + mstr(173));
         end
         if (local_var(4) == 0) then begin
             set_local_var(4, 1);
             give_exp_points(25);
-            display_msg(message_str(14, 100) + 25 + message_str(14, 101));
+            display_msg(g_mstr(100) + 25 + g_mstr(101));
         end
     end
     else begin
         if (is_critical(Locks_Roll)) then begin
             jam_lock(self_obj);
             if (source_obj == dude_obj) then begin
-                display_msg(message_str(1367, 171));
+                display_mstr(171);
             end
             else begin
-                display_msg(obj_name(source_obj) + message_str(1367, 174));
+                display_msg(obj_name(source_obj) + mstr(174));
             end
         end
         else begin
             if (source_obj == dude_obj) then begin
-                display_msg(message_str(1367, 172));
+                display_mstr(172);
             end
             else begin
-                display_msg(obj_name(source_obj) + message_str(1367, 175));
+                display_msg(obj_name(source_obj) + mstr(175));
             end
         end
     end
@@ -970,23 +970,23 @@ begin
     if (is_success(Locks_Roll)) then begin
         obj_lock(self_obj);
         if (source_obj == dude_obj) then begin
-            display_msg(message_str(1367, 189));
+            display_mstr(189);
         end
         else begin
-            display_msg(obj_name(source_obj) + message_str(1367, 191));
+            display_msg(obj_name(source_obj) + mstr(191));
         end
         if (local_var(4) == 0) then begin
             set_local_var(4, 1);
             give_exp_points(25);
-            display_msg(message_str(14, 100) + 25 + message_str(14, 101));
+            display_msg(g_mstr(100) + 25 + g_mstr(101));
         end
     end
     else begin
         if (source_obj == dude_obj) then begin
-            display_msg(message_str(1367, 190));
+            display_mstr(190);
         end
         else begin
-            display_msg(obj_name(source_obj) + message_str(1367, 192));
+            display_msg(obj_name(source_obj) + mstr(192));
         end
     end
 end
@@ -1019,10 +1019,10 @@ begin
     if (obj_is_open(self_obj)) then begin
         script_overrides;
         if (source_obj == dude_obj) then begin
-            display_msg(message_str(1367, 206));
+            display_mstr(206);
         end
         else begin
-            display_msg(obj_name(source_obj) + message_str(1367, 207));
+            display_msg(obj_name(source_obj) + mstr(207));
         end
     end
     else begin
@@ -1031,10 +1031,10 @@ begin
         if (is_success(Traps_Roll)) then begin
             set_local_var(0, 0);
             if (source_obj == dude_obj) then begin
-                display_msg(message_str(1367, 208));
+                display_mstr(208);
             end
             else begin
-                display_msg(obj_name(source_obj) + message_str(1367, 210));
+                display_msg(obj_name(source_obj) + mstr(210));
             end
         end
         else begin
@@ -1043,10 +1043,10 @@ begin
             end
             else begin
                 if (source_obj == dude_obj) then begin
-                    display_msg(message_str(1367, 209));
+                    display_mstr(209);
                 end
                 else begin
-                    display_msg(obj_name(source_obj) + message_str(1367, 211));
+                    display_msg(obj_name(source_obj) + mstr(211));
                 end
             end
         end
@@ -1060,19 +1060,19 @@ begin
     if (is_success(LVar0)) then begin
         obj_unlock(self_obj);
         if (source_obj == dude_obj) then begin
-            if (get_critter_stat(dude_obj, 34) == 0) then begin
-                display_msg(message_str(1367, 176));
+            if (dude_gender == 0) then begin
+                display_mstr(176);
             end
             else begin
-                display_msg(message_str(1367, 1076));
+                display_mstr(1076);
             end
         end
         else begin
-            if (get_critter_stat(source_obj, 34) == 0) then begin
-                display_msg(obj_name(source_obj) + message_str(1367, 181));
+            if (get_critter_stat(source_obj, STAT_gender) == 0) then begin
+                display_msg(obj_name(source_obj) + mstr(181));
             end
             else begin
-                display_msg(obj_name(source_obj) + message_str(1367, 1081));
+                display_msg(obj_name(source_obj) + mstr(1081));
             end
         end
     end
@@ -1081,46 +1081,46 @@ begin
             critter_dmg(source_obj, 2, 0 bwor 256);
             if (source_obj == dude_obj) then begin
                 if (2 == 1) then begin
-                    display_msg(message_str(1367, 177));
+                    display_mstr(177);
                 end
                 else begin
-                    display_msg(message_str(1367, 178) + 2 + message_str(1367, 179));
+                    display_msg(mstr(178) + 2 + mstr(179));
                 end
             end
             else begin
-                if (get_critter_stat(source_obj, 34) == 0) then begin
+                if (get_critter_stat(source_obj, STAT_gender) == 0) then begin
                     if (2 == 1) then begin
-                        display_msg(message_str(1367, 182));
+                        display_mstr(182);
                     end
                     else begin
-                        display_msg(message_str(1367, 183) + 2 + message_str(1367, 184));
+                        display_msg(mstr(183) + 2 + mstr(184));
                     end
                 end
                 else begin
                     if (2 == 1) then begin
-                        display_msg(message_str(1367, 186));
+                        display_mstr(186);
                     end
                     else begin
-                        display_msg(message_str(1367, 187) + 2 + message_str(1367, 188));
+                        display_msg(mstr(187) + 2 + mstr(188));
                     end
                 end
             end
         end
         else begin
             if (source_obj == dude_obj) then begin
-                if (get_critter_stat(dude_obj, 34) == 0) then begin
-                    display_msg(message_str(1367, 180));
+                if (dude_gender == 0) then begin
+                    display_mstr(180);
                 end
                 else begin
-                    display_msg(message_str(1367, 1080));
+                    display_mstr(1080);
                 end
             end
             else begin
-                if (get_critter_stat(source_obj, 34) == 0) then begin
-                    display_msg(obj_name(source_obj) + message_str(1367, 185));
+                if (get_critter_stat(source_obj, STAT_gender) == 0) then begin
+                    display_msg(obj_name(source_obj) + mstr(185));
                 end
                 else begin
-                    display_msg(obj_name(source_obj) + message_str(1367, 1085));
+                    display_msg(obj_name(source_obj) + mstr(1085));
                 end
             end
         end
@@ -1130,14 +1130,14 @@ end
 procedure Damage_Critter
 begin
     variable LVar0 := 0;
-    LVar0 := random(20, 20);
+    LVar0 := 20;
     if (source_obj == dude_obj) then begin
         critter_dmg(dude_obj, LVar0, 6 bwor 256);
-        display_msg(message_str(1367, 166) + LVar0 + message_str(1367, 167));
+        display_msg(mstr(166) + LVar0 + mstr(167));
     end
     else begin
         critter_dmg(source_obj, LVar0, 6 bwor 256);
-        display_msg(obj_name(source_obj) + message_str(1367, 168) + LVar0 + message_str(1367, 169));
+        display_msg(obj_name(source_obj) + mstr(168) + LVar0 + mstr(169));
     end
     set_local_var(0, 1);
 end
@@ -1158,4 +1158,3 @@ begin
         end
     end
 end
-
