diff --git a/Mods/Resurrection/Original/SSTAPLE.ssl b/Mods/Resurrection/ReDefine/SSTAPLE.ssl
index 41dc4f1..447f3a0 100644
--- a/Mods/Resurrection/Original/SSTAPLE.ssl
+++ b/Mods/Resurrection/ReDefine/SSTAPLE.ssl
@@ -92,12 +92,12 @@ begin
             return 1;
         end
     end
-    if ((party_member_obj(16777779) != 0) and ((global_var(910) bwand 32) == 0)) then begin
+    if ((party_member_obj(16777779) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 32) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777779)) <= 5) then begin
             return 1;
         end
     end
-    if ((party_member_obj(16777811) != 0) and ((global_var(910) bwand 64) == 0)) then begin
+    if ((party_member_obj(16777811) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 64) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777811)) <= 5) then begin
             return 1;
         end
@@ -112,7 +112,7 @@ begin
             return 1;
         end
     end
-    if ((party_member_obj(16777814) != 0) and ((global_var(910) bwand 128) == 0)) then begin
+    if ((party_member_obj(16777814) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 128) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777814)) <= 5) then begin
             return 1;
         end
@@ -172,11 +172,11 @@ procedure critter_p_proc
 begin
     if ((local_var(5) == 2) and obj_can_see_obj(self_obj, dude_obj)) then begin
         set_local_var(5, 1);
-        attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+        attack(dude_obj);
     end
-    if (global_var(769) == 1) then begin
+    if (global_var(GVAR_ENEMY_SEDIT) == 1) then begin
         if (obj_can_see_obj(self_obj, dude_obj) or obj_can_hear_obj(self_obj, dude_obj)) then begin
-            attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+            attack(dude_obj);
         end
     end
     if (local_var(18) == 1) then begin
@@ -190,7 +190,7 @@ end
 procedure pickup_p_proc
 begin
     if (source_obj == dude_obj) then begin
-        float_msg(self_obj, message_str(1468, 300), 2);
+        float_msg(self_obj, mstr(300), FLOAT_MSG_RED);
         set_local_var(5, 2);
     end
 end
@@ -198,8 +198,8 @@ end
 procedure talk_p_proc
 begin
     if (local_var(18) == 1) then begin
-        anim(dude_obj, 1000, rotation_to_tile(tile_num(dude_obj), tile_num(self_obj)));
-        anim(self_obj, 1000, rotation_to_tile(tile_num(self_obj), tile_num(dude_obj)));
+        anim(dude_obj, 1000, rotation_to_tile(dude_tile, tile_num(self_obj)));
+        anim(self_obj, 1000, rotation_to_tile(tile_num(self_obj), dude_tile));
         start_gdialog(1468, self_obj, 4, -1, -1);
         gsay_start;
         call Node1850();
@@ -208,19 +208,19 @@ begin
     end
     else begin
         if (local_var(4) != 1) then begin
-            anim(dude_obj, 1000, rotation_to_tile(tile_num(dude_obj), tile_num(self_obj)));
-            anim(self_obj, 1000, rotation_to_tile(tile_num(self_obj), tile_num(dude_obj)));
+            anim(dude_obj, 1000, rotation_to_tile(dude_tile, tile_num(self_obj)));
+            anim(self_obj, 1000, rotation_to_tile(tile_num(self_obj), dude_tile));
             start_gdialog(1468, self_obj, 4, -1, -1);
             gsay_start;
             call Node500();
             gsay_end;
             end_dialogue;
             set_local_var(4, 1);
-            set_global_var(764, 1);
+            set_global_var(GVAR_STAPLETON_SPEECH, 1);
         end
         else begin
-            anim(dude_obj, 1000, rotation_to_tile(tile_num(dude_obj), tile_num(self_obj)));
-            anim(self_obj, 1000, rotation_to_tile(tile_num(self_obj), tile_num(dude_obj)));
+            anim(dude_obj, 1000, rotation_to_tile(dude_tile, tile_num(self_obj)));
+            anim(self_obj, 1000, rotation_to_tile(tile_num(self_obj), dude_tile));
             start_gdialog(1468, self_obj, 4, -1, -1);
             gsay_start;
             if (local_var(16) == 0) then begin
@@ -244,10 +244,10 @@ procedure look_at_p_proc
 begin
     script_overrides;
     if (local_var(4) == 1) then begin
-        display_msg(message_str(1468, 150));
+        display_mstr(150);
     end
     else begin
-        display_msg(message_str(1468, 100));
+        display_mstr(100);
     end
 end
 
@@ -255,10 +255,10 @@ procedure description_p_proc
 begin
     script_overrides;
     if (local_var(4) == 1) then begin
-        display_msg(message_str(1468, 250));
+        display_mstr(250);
     end
     else begin
-        display_msg(message_str(1468, 200));
+        display_mstr(200);
     end
 end
 
@@ -266,7 +266,7 @@ procedure timed_event_p_proc
 begin
     if (fixed_param == 1) then begin
         time1 := game_time_hour;
-        metarule3(100, self_obj, 2, 0);
+        rm_fixed_timer_event(self_obj, 2);
         add_timer_event(self_obj, game_ticks(60), 2);
     end
     if (fixed_param == 2) then begin
@@ -274,7 +274,7 @@ begin
         if (((time2 - time1) > 15) or ((time2 - time1) < -15)) then begin
             call Time_check();
         end
-        metarule3(100, self_obj, 1, 0);
+        rm_fixed_timer_event(self_obj, 1);
         add_timer_event(self_obj, game_ticks(1), 1);
     end
     if (fixed_param == 10) then begin
@@ -284,12 +284,12 @@ end
 
 procedure map_enter_p_proc
 begin
-    critter_add_trait(self_obj, 1, 6, 212);
-    critter_add_trait(self_obj, 1, 5, 2);
-    if (not(metarule(22, 0))) then begin
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 212);
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_PEASANTS_NO_TALKING);
+    if (not(is_loading_game)) then begin
         call Time_check();
     end
-    metarule3(100, self_obj, 1, 0);
+    rm_fixed_timer_event(self_obj, 1);
     add_timer_event(self_obj, game_ticks(1), 1);
 end
 
@@ -297,7 +297,7 @@ procedure damage_p_proc
 begin
     if (source_obj == dude_obj) then begin
         call Node998();
-        set_global_var(769, 1);
+        set_global_var(GVAR_ENEMY_SEDIT, 1);
     end
 end
 
@@ -305,59 +305,59 @@ procedure destroy_p_proc
 begin
     stapleton := -1;
     if (source_obj == dude_obj) then begin
-        set_global_var(5, global_var(5) + 1);
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(4) > (3 * global_var(5))) or (global_var(2) == 1)) and (global_var(1) == 0)) then begin
-            set_global_var(2, 1);
-            set_global_var(3, 0);
+        set_global_var(GVAR_GOOD_MONSTER, global_var(GVAR_GOOD_MONSTER) + 1);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(4) > (3 * global_var(GVAR_GOOD_MONSTER))) or (global_var(GVAR_CHAMPION_REPUTATION) == 1)) and (global_var(GVAR_CHILDKILLER_REPUTATION) == 0)) then begin
+            set_global_var(GVAR_CHAMPION_REPUTATION, 1);
+            set_global_var(GVAR_BERSERKER_REPUTATION, 0);
         end
-        if (metarule(51, self_obj) == 2) then begin
+        if (critter_kill_type(self_obj) == KILL_TYPE_children_kills) then begin
             if (source_obj == dude_obj) then begin
-                set_global_var(2, 0);
-                set_global_var(1, global_var(1) + 1);
-                set_global_var(0, global_var(0) + -15);
+                set_global_var(GVAR_CHAMPION_REPUTATION, 0);
+                set_global_var(GVAR_CHILDKILLER_REPUTATION, global_var(GVAR_CHILDKILLER_REPUTATION) + 1);
+                set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + -15);
                 debug_msg("Player gains " + -15 + " Karma Points.");
-                set_global_var(37, 0);
-                set_global_var(38, 0);
-                set_global_var(39, 0);
-                set_global_var(40, 0);
-                set_global_var(41, 0);
-                set_global_var(42, 0);
-                set_global_var(43, 0);
-                set_global_var(44, 0);
-                set_global_var(45, 0);
-                if (global_var(0) >= 1000) then begin
-                    set_global_var(37, 1);
+                set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+                set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+                set_global_var(GVAR_KARMA_DEFENDER, 0);
+                set_global_var(GVAR_KARMA_WANDERER, 0);
+                set_global_var(GVAR_KARMA_BETRAYER, 0);
+                set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+                set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+                set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+                    set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
                 end
                 else begin
-                    if (global_var(0) >= 750) then begin
-                        set_global_var(38, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
                     end
                     else begin
-                        if (global_var(0) >= 500) then begin
-                            set_global_var(39, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                            set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                         end
                         else begin
-                            if (global_var(0) >= 250) then begin
-                                set_global_var(40, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                                set_global_var(GVAR_KARMA_DEFENDER, 1);
                             end
                             else begin
-                                if (global_var(0) > -250) then begin
-                                    set_global_var(41, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                                    set_global_var(GVAR_KARMA_WANDERER, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -500) then begin
-                                        set_global_var(42, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                        set_global_var(GVAR_KARMA_BETRAYER, 1);
                                     end
                                     else begin
-                                        if (global_var(0) > -750) then begin
-                                            set_global_var(43, 1);
+                                        if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                            set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                         end
                                         else begin
-                                            if (global_var(0) > -1000) then begin
-                                                set_global_var(44, 1);
+                                            if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                                set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                             end
                                             else begin
-                                                set_global_var(45, 1);
+                                                set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                             end
                                         end
                                     end
@@ -366,58 +366,58 @@ begin
                         end
                     end
                 end
-                set_global_var(48, global_var(48) + -8);
+                set_global_var(GVAR_TOWN_REP_SEDIT, global_var(GVAR_TOWN_REP_SEDIT) + -8);
                 debug_msg("Added " + -8 + " to Town Rep");
             end
         end
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(5) > (2 * global_var(4))) or (global_var(3) == 1))) then begin
-            set_global_var(3, 1);
-            set_global_var(2, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
+            set_global_var(GVAR_BERSERKER_REPUTATION, 1);
+            set_global_var(GVAR_CHAMPION_REPUTATION, 0);
         end
-        set_global_var(0, global_var(0) + -10);
+        set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + -10);
         debug_msg("Player gains " + -10 + " Karma Points.");
-        set_global_var(37, 0);
-        set_global_var(38, 0);
-        set_global_var(39, 0);
-        set_global_var(40, 0);
-        set_global_var(41, 0);
-        set_global_var(42, 0);
-        set_global_var(43, 0);
-        set_global_var(44, 0);
-        set_global_var(45, 0);
-        if (global_var(0) >= 1000) then begin
-            set_global_var(37, 1);
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+        set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+        set_global_var(GVAR_KARMA_DEFENDER, 0);
+        set_global_var(GVAR_KARMA_WANDERER, 0);
+        set_global_var(GVAR_KARMA_BETRAYER, 0);
+        set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+        set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+            set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
         end
         else begin
-            if (global_var(0) >= 750) then begin
-                set_global_var(38, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
             end
             else begin
-                if (global_var(0) >= 500) then begin
-                    set_global_var(39, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                 end
                 else begin
-                    if (global_var(0) >= 250) then begin
-                        set_global_var(40, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                        set_global_var(GVAR_KARMA_DEFENDER, 1);
                     end
                     else begin
-                        if (global_var(0) > -250) then begin
-                            set_global_var(41, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                            set_global_var(GVAR_KARMA_WANDERER, 1);
                         end
                         else begin
-                            if (global_var(0) > -500) then begin
-                                set_global_var(42, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                set_global_var(GVAR_KARMA_BETRAYER, 1);
                             end
                             else begin
-                                if (global_var(0) > -750) then begin
-                                    set_global_var(43, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -1000) then begin
-                                        set_global_var(44, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                     end
                                     else begin
-                                        set_global_var(45, 1);
+                                        set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                     end
                                 end
                             end
@@ -426,7 +426,7 @@ begin
                 end
             end
         end
-        set_global_var(48, global_var(48) + -5);
+        set_global_var(GVAR_TOWN_REP_SEDIT, global_var(GVAR_TOWN_REP_SEDIT) + -5);
         debug_msg("Added " + -5 + " to Town Rep");
     end
 end
@@ -445,80 +445,80 @@ procedure Node999move
 begin
     move_obj_inven_to_obj(stapleton, staple_shelf_trade);
     move_obj_inven_to_obj(staple_shelf_swap, stapleton);
-    wield_obj_critter(self_obj, obj_carrying_pid_obj(self_obj, 18));
+    wield_obj(obj_carrying_pid_obj(self_obj, 18));
     anim(self_obj, 1000, 4);
 end
 
 procedure Node500
 begin
-    gsay_reply(1468, 500);
-    giq_option(4, 1468, 501, Node550, 50);
-    giq_option(4, 1468, 502, Node998, 50);
-    giq_option(-3, 1468, 503, Node750, 50);
-    giq_option(4, 1468, 504, Node999, 50);
+    Reply(500);
+    giq_option(4, NAME, 501, Node550, NEUTRAL_REACTION);
+    giq_option(4, NAME, 502, Node998, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 503, Node750, NEUTRAL_REACTION);
+    giq_option(4, NAME, 504, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node550
 begin
-    gsay_reply(1468, 550);
-    if (get_critter_stat(dude_obj, 34) == 0) then begin
-        giq_option(4, 1468, 551, Node650, 50);
+    Reply(550);
+    if (dude_gender == 0) then begin
+        giq_option(4, NAME, 551, Node650, NEUTRAL_REACTION);
     end
-    if (get_critter_stat(dude_obj, 34) == 1) then begin
-        giq_option(4, 1468, 551, Node700, 50);
+    if (dude_gender == 1) then begin
+        giq_option(4, NAME, 551, Node700, NEUTRAL_REACTION);
     end
-    giq_option(4, 1468, 552, Node600, 50);
-    giq_option(4, 1468, 553, Node998, 50);
-    giq_option(4, 1468, 554, Node999, 50);
+    giq_option(4, NAME, 552, Node600, NEUTRAL_REACTION);
+    giq_option(4, NAME, 553, Node998, NEUTRAL_REACTION);
+    giq_option(4, NAME, 554, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node600
 begin
-    gsay_reply(1468, 600);
+    Reply(600);
     set_local_var(16, 2);
-    set_global_var(0, global_var(0) - 1);
-    set_global_var(37, 0);
-    set_global_var(38, 0);
-    set_global_var(39, 0);
-    set_global_var(40, 0);
-    set_global_var(41, 0);
-    set_global_var(42, 0);
-    set_global_var(43, 0);
-    set_global_var(44, 0);
-    set_global_var(45, 0);
-    if (global_var(0) >= 1000) then begin
-        set_global_var(37, 1);
+    set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) - 1);
+    set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+    set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+    set_global_var(GVAR_KARMA_DEFENDER, 0);
+    set_global_var(GVAR_KARMA_WANDERER, 0);
+    set_global_var(GVAR_KARMA_BETRAYER, 0);
+    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+    set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+    set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+    if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
     end
     else begin
-        if (global_var(0) >= 750) then begin
-            set_global_var(38, 1);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+            set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
         end
         else begin
-            if (global_var(0) >= 500) then begin
-                set_global_var(39, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
             end
             else begin
-                if (global_var(0) >= 250) then begin
-                    set_global_var(40, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                    set_global_var(GVAR_KARMA_DEFENDER, 1);
                 end
                 else begin
-                    if (global_var(0) > -250) then begin
-                        set_global_var(41, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                        set_global_var(GVAR_KARMA_WANDERER, 1);
                     end
                     else begin
-                        if (global_var(0) > -500) then begin
-                            set_global_var(42, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                            set_global_var(GVAR_KARMA_BETRAYER, 1);
                         end
                         else begin
-                            if (global_var(0) > -750) then begin
-                                set_global_var(43, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                             end
                             else begin
-                                if (global_var(0) > -1000) then begin
-                                    set_global_var(44, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                    set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                 end
                                 else begin
-                                    set_global_var(45, 1);
+                                    set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                 end
                             end
                         end
@@ -527,471 +527,471 @@ begin
             end
         end
     end
-    giq_option(4, 1468, 601, Node998, 50);
-    if (get_critter_stat(dude_obj, 34) == 0) then begin
-        giq_option(4, 1468, 602, Node999, 50);
+    giq_option(4, NAME, 601, Node998, NEUTRAL_REACTION);
+    if (dude_gender == 0) then begin
+        giq_option(4, NAME, 602, Node999, NEUTRAL_REACTION);
     end
-    if (get_critter_stat(dude_obj, 34) == 1) then begin
-        giq_option(4, 1468, 603, Node999, 50);
+    if (dude_gender == 1) then begin
+        giq_option(4, NAME, 603, Node999, NEUTRAL_REACTION);
     end
 end
 
 procedure Node650
 begin
-    gsay_reply(1468, 650);
-    giq_option(4, 1468, 651, Node800, 50);
-    giq_option(4, 1468, 652, Node600, 50);
+    Reply(650);
+    giq_option(4, NAME, 651, Node800, NEUTRAL_REACTION);
+    giq_option(4, NAME, 652, Node600, NEUTRAL_REACTION);
 end
 
 procedure Node700
 begin
-    gsay_reply(1468, 700);
-    giq_option(4, 1468, 701, Node800, 50);
-    giq_option(4, 1468, 702, Node600, 50);
+    Reply(700);
+    giq_option(4, NAME, 701, Node800, NEUTRAL_REACTION);
+    giq_option(4, NAME, 702, Node600, NEUTRAL_REACTION);
 end
 
 procedure Node750
 begin
-    gsay_reply(1468, 750);
+    Reply(750);
     set_local_var(16, 2);
-    giq_option(-3, 1468, 751, Node998, 50);
-    giq_option(-3, 1468, 752, Node999, 50);
+    giq_option(-3, NAME, 751, Node998, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 752, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node800
 begin
-    gsay_reply(1468, 800);
+    Reply(800);
     set_local_var(16, 1);
-    giq_option(4, 1468, 801, Barter, 50);
+    giq_option(4, NAME, 801, Barter, NEUTRAL_REACTION);
     if (obj_carrying_pid_obj(dude_obj, 309) and (local_var(6) != 1)) then begin
-        giq_option(4, 1468, 802, Node850, 50);
+        giq_option(4, NAME, 802, Node850, NEUTRAL_REACTION);
     end
     if (local_var(7) != 1) then begin
-        giq_option(4, 1468, 803, Node900, 50);
+        giq_option(4, NAME, 803, Node900, NEUTRAL_REACTION);
     end
     if (local_var(8) != 1) then begin
-        giq_option(4, 1468, 804, Node1000, 50);
+        giq_option(4, NAME, 804, Node1000, NEUTRAL_REACTION);
     end
-    giq_option(4, 1468, 805, Node999, 50);
+    giq_option(4, NAME, 805, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node850
 begin
-    gsay_reply(1468, 850);
+    Reply(850);
     set_local_var(6, 1);
-    giq_option(4, 1468, 851, Node1150, 50);
-    giq_option(4, 1468, 852, Barter, 50);
+    giq_option(4, NAME, 851, Node1150, NEUTRAL_REACTION);
+    giq_option(4, NAME, 852, Barter, NEUTRAL_REACTION);
     if ((local_var(10) == 1) and (local_var(17) != 3)) then begin
-        giq_option(4, 1468, 853, Node1650, 50);
+        giq_option(4, NAME, 853, Node1650, NEUTRAL_REACTION);
     end
     if (local_var(7) != 1) then begin
-        giq_option(4, 1468, 854, Node900, 50);
+        giq_option(4, NAME, 854, Node900, NEUTRAL_REACTION);
     end
     if (local_var(8) != 1) then begin
-        giq_option(4, 1468, 855, Node1000, 50);
+        giq_option(4, NAME, 855, Node1000, NEUTRAL_REACTION);
     end
-    giq_option(4, 1468, 856, Node999, 50);
+    giq_option(4, NAME, 856, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node900
 begin
-    gsay_reply(1468, 900);
+    Reply(900);
     set_local_var(7, 1);
-    giq_option(6, 1468, 901, Node950, 50);
-    giq_option(4, 1468, 902, Barter, 50);
+    giq_option(6, NAME, 901, Node950, NEUTRAL_REACTION);
+    giq_option(4, NAME, 902, Barter, NEUTRAL_REACTION);
     if ((local_var(10) == 1) and (local_var(17) != 3)) then begin
-        giq_option(4, 1468, 903, Node1650, 50);
+        giq_option(4, NAME, 903, Node1650, NEUTRAL_REACTION);
     end
     if (obj_carrying_pid_obj(dude_obj, 309) and (local_var(6) != 1)) then begin
-        giq_option(4, 1468, 904, Node850, 50);
+        giq_option(4, NAME, 904, Node850, NEUTRAL_REACTION);
     end
     if (local_var(8) != 1) then begin
-        giq_option(4, 1468, 905, Node1000, 50);
+        giq_option(4, NAME, 905, Node1000, NEUTRAL_REACTION);
     end
-    giq_option(4, 1468, 906, Node999, 50);
+    giq_option(4, NAME, 906, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node950
 begin
-    gsay_reply(1468, 950);
-    giq_option(4, 1468, 951, Barter, 50);
+    Reply(950);
+    giq_option(4, NAME, 951, Barter, NEUTRAL_REACTION);
     if ((local_var(10) == 1) and (local_var(17) != 3)) then begin
-        giq_option(4, 1468, 952, Node1650, 50);
+        giq_option(4, NAME, 952, Node1650, NEUTRAL_REACTION);
     end
     if (obj_carrying_pid_obj(dude_obj, 309) and (local_var(6) != 1)) then begin
-        giq_option(4, 1468, 953, Node850, 50);
+        giq_option(4, NAME, 953, Node850, NEUTRAL_REACTION);
     end
     if (local_var(8) != 1) then begin
-        giq_option(4, 1468, 954, Node1000, 50);
+        giq_option(4, NAME, 954, Node1000, NEUTRAL_REACTION);
     end
-    giq_option(4, 1468, 955, Node999, 50);
+    giq_option(4, NAME, 955, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node1000
 begin
-    gsay_reply(1468, 1000);
+    Reply(1000);
     set_local_var(8, 1);
-    if (get_critter_stat(dude_obj, 34) == 0) then begin
-        giq_option(4, 1468, 1001, Node1050, 50);
+    if (dude_gender == 0) then begin
+        giq_option(4, NAME, 1001, Node1050, NEUTRAL_REACTION);
     end
-    if (get_critter_stat(dude_obj, 34) == 1) then begin
-        giq_option(4, 1468, 1002, Node1050, 50);
+    if (dude_gender == 1) then begin
+        giq_option(4, NAME, 1002, Node1050, NEUTRAL_REACTION);
     end
-    giq_option(4, 1468, 1003, Barter, 50);
+    giq_option(4, NAME, 1003, Barter, NEUTRAL_REACTION);
     if ((local_var(10) == 1) and (local_var(17) != 3)) then begin
-        giq_option(4, 1468, 1004, Node1650, 50);
+        giq_option(4, NAME, 1004, Node1650, NEUTRAL_REACTION);
     end
     if (obj_carrying_pid_obj(dude_obj, 309) and (local_var(6) != 1)) then begin
-        giq_option(4, 1468, 1005, Node850, 50);
+        giq_option(4, NAME, 1005, Node850, NEUTRAL_REACTION);
     end
     if (local_var(7) != 1) then begin
-        giq_option(4, 1468, 1006, Node900, 50);
+        giq_option(4, NAME, 1006, Node900, NEUTRAL_REACTION);
     end
-    giq_option(4, 1468, 1007, Node999, 50);
+    giq_option(4, NAME, 1007, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node1050
 begin
-    gsay_reply(1468, 1050);
+    Reply(1050);
     set_local_var(8, 1);
-    giq_option(7, 1468, 1051, Node1100, 50);
-    giq_option(8, 1468, 1052, Node1200, 50);
-    giq_option(4, 1468, 1053, Barter, 50);
+    giq_option(7, NAME, 1051, Node1100, NEUTRAL_REACTION);
+    giq_option(8, NAME, 1052, Node1200, NEUTRAL_REACTION);
+    giq_option(4, NAME, 1053, Barter, NEUTRAL_REACTION);
     if ((local_var(10) == 1) and (local_var(17) != 3)) then begin
-        giq_option(4, 1468, 1054, Node1650, 50);
+        giq_option(4, NAME, 1054, Node1650, NEUTRAL_REACTION);
     end
     if (obj_carrying_pid_obj(dude_obj, 309) and (local_var(6) != 1)) then begin
-        giq_option(4, 1468, 1055, Node850, 50);
+        giq_option(4, NAME, 1055, Node850, NEUTRAL_REACTION);
     end
     if (local_var(7) != 1) then begin
-        giq_option(4, 1468, 1056, Node900, 50);
+        giq_option(4, NAME, 1056, Node900, NEUTRAL_REACTION);
     end
-    giq_option(4, 1468, 1057, Node999, 50);
+    giq_option(4, NAME, 1057, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node1100
 begin
-    gsay_reply(1468, 1100);
-    giq_option(8, 1468, 1101, Node1200, 50);
-    giq_option(4, 1468, 1102, Barter, 50);
+    Reply(1100);
+    giq_option(8, NAME, 1101, Node1200, NEUTRAL_REACTION);
+    giq_option(4, NAME, 1102, Barter, NEUTRAL_REACTION);
     if ((local_var(10) == 1) and (local_var(17) != 3)) then begin
-        giq_option(4, 1468, 1103, Node1650, 50);
+        giq_option(4, NAME, 1103, Node1650, NEUTRAL_REACTION);
     end
     if (obj_carrying_pid_obj(dude_obj, 309) and (local_var(6) != 1)) then begin
-        giq_option(4, 1468, 1104, Node850, 50);
+        giq_option(4, NAME, 1104, Node850, NEUTRAL_REACTION);
     end
     if (local_var(7) != 1) then begin
-        giq_option(4, 1468, 1105, Node900, 50);
+        giq_option(4, NAME, 1105, Node900, NEUTRAL_REACTION);
     end
-    giq_option(4, 1468, 1106, Node999, 50);
+    giq_option(4, NAME, 1106, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node1150
 begin
-    gsay_reply(1468, 1150);
-    giq_option(4, 1468, 1151, Node1050, 50);
-    giq_option(4, 1468, 1152, Barter, 50);
+    Reply(1150);
+    giq_option(4, NAME, 1151, Node1050, NEUTRAL_REACTION);
+    giq_option(4, NAME, 1152, Barter, NEUTRAL_REACTION);
     if ((local_var(10) == 1) and (local_var(17) != 3)) then begin
-        giq_option(4, 1468, 1153, Node1650, 50);
+        giq_option(4, NAME, 1153, Node1650, NEUTRAL_REACTION);
     end
     if (local_var(7) != 1) then begin
-        giq_option(4, 1468, 1154, Node900, 50);
+        giq_option(4, NAME, 1154, Node900, NEUTRAL_REACTION);
     end
-    giq_option(4, 1468, 1155, Node999, 50);
+    giq_option(4, NAME, 1155, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node1200
 begin
-    gsay_reply(1468, 1200);
-    giq_option(4, 1468, 1201, Node1100, 50);
-    giq_option(4, 1468, 1202, Barter, 50);
+    Reply(1200);
+    giq_option(4, NAME, 1201, Node1100, NEUTRAL_REACTION);
+    giq_option(4, NAME, 1202, Barter, NEUTRAL_REACTION);
     if ((local_var(10) == 1) and (local_var(17) != 3)) then begin
-        giq_option(4, 1468, 1203, Node1650, 50);
+        giq_option(4, NAME, 1203, Node1650, NEUTRAL_REACTION);
     end
     if (obj_carrying_pid_obj(dude_obj, 309) and (local_var(6) != 1)) then begin
-        giq_option(4, 1468, 1204, Node850, 50);
+        giq_option(4, NAME, 1204, Node850, NEUTRAL_REACTION);
     end
     if (local_var(7) != 1) then begin
-        giq_option(4, 1468, 1205, Node900, 50);
+        giq_option(4, NAME, 1205, Node900, NEUTRAL_REACTION);
     end
-    giq_option(4, 1468, 1206, Node999, 50);
+    giq_option(4, NAME, 1206, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node1250
 begin
-    gsay_reply(1468, 1250);
+    Reply(1250);
     if (local_var(9) != 1) then begin
-        giq_option(4, 1468, 1251, Node1300move, 50);
+        giq_option(4, NAME, 1251, Node1300move, NEUTRAL_REACTION);
     end
-    giq_option(4, 1468, 1252, Node1350move, 50);
-    giq_option(4, 1468, 1253, Node999move, 50);
+    giq_option(4, NAME, 1252, Node1350move, NEUTRAL_REACTION);
+    giq_option(4, NAME, 1253, Node999move, NEUTRAL_REACTION);
 end
 
 procedure Node1300
 begin
-    gsay_reply(1468, 1300);
+    Reply(1300);
     set_local_var(9, 1);
-    giq_option(4, 1468, 1301, Node1350, 50);
-    giq_option(4, 1468, 1302, Node999, 50);
+    giq_option(4, NAME, 1301, Node1350, NEUTRAL_REACTION);
+    giq_option(4, NAME, 1302, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node1350
 begin
-    gsay_reply(1468, 1350);
-    giq_option(4, 1468, 1351, Barter, 50);
+    Reply(1350);
+    giq_option(4, NAME, 1351, Barter, NEUTRAL_REACTION);
     if ((local_var(10) == 1) and (local_var(17) != 3)) then begin
-        giq_option(4, 1468, 1352, Node1650, 50);
+        giq_option(4, NAME, 1352, Node1650, NEUTRAL_REACTION);
     end
     if (obj_carrying_pid_obj(dude_obj, 309) and (local_var(6) != 1)) then begin
-        giq_option(4, 1468, 1353, Node850, 50);
+        giq_option(4, NAME, 1353, Node850, NEUTRAL_REACTION);
     end
     if (local_var(7) != 1) then begin
-        giq_option(4, 1468, 1354, Node900, 50);
+        giq_option(4, NAME, 1354, Node900, NEUTRAL_REACTION);
     end
     if (local_var(8) != 1) then begin
-        giq_option(4, 1468, 1355, Node1000, 50);
+        giq_option(4, NAME, 1355, Node1000, NEUTRAL_REACTION);
     end
-    giq_option(4, 1468, 1356, Node999, 50);
+    giq_option(4, NAME, 1356, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node1400
 begin
-    gsay_reply(1468, 1400);
-    giq_option(4, 1468, 1401, Barter, 50);
+    Reply(1400);
+    giq_option(4, NAME, 1401, Barter, NEUTRAL_REACTION);
     if ((local_var(10) == 1) and (local_var(17) != 3)) then begin
-        giq_option(4, 1468, 1402, Node1650, 50);
+        giq_option(4, NAME, 1402, Node1650, NEUTRAL_REACTION);
     end
     if (obj_carrying_pid_obj(dude_obj, 309) and (local_var(6) != 1)) then begin
-        giq_option(4, 1468, 1403, Node850, 50);
+        giq_option(4, NAME, 1403, Node850, NEUTRAL_REACTION);
     end
     if (local_var(7) != 1) then begin
-        giq_option(4, 1468, 1404, Node900, 50);
+        giq_option(4, NAME, 1404, Node900, NEUTRAL_REACTION);
     end
     if (local_var(8) != 1) then begin
-        giq_option(4, 1468, 1405, Node1000, 50);
+        giq_option(4, NAME, 1405, Node1000, NEUTRAL_REACTION);
     end
-    giq_option(4, 1468, 1406, Node999, 50);
-    giq_option(-3, 1468, 1407, Node750, 50);
+    giq_option(4, NAME, 1406, Node999, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 1407, Node750, NEUTRAL_REACTION);
 end
 
 procedure Node1450
 begin
-    gsay_reply(1468, 1450);
+    Reply(1450);
     if (is_success(roll_vs_skill(dude_obj, 14, 0))) then begin
-        giq_option(4, 1468, 1451, Node1550, 50);
+        giq_option(4, NAME, 1451, Node1550, NEUTRAL_REACTION);
     end
     else begin
-        giq_option(4, 1468, 1451, Node1500, 50);
+        giq_option(4, NAME, 1451, Node1500, NEUTRAL_REACTION);
     end
-    giq_option(4, 1468, 1452, Node998, 50);
-    giq_option(4, 1468, 1453, Node999, 50);
-    giq_option(-3, 1468, 1454, Node999, 50);
+    giq_option(4, NAME, 1452, Node998, NEUTRAL_REACTION);
+    giq_option(4, NAME, 1453, Node999, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 1454, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node1500
 begin
-    gsay_reply(1468, 1500);
-    giq_option(4, 1468, 1501, Node998, 50);
-    giq_option(4, 1468, 1502, Node999, 50);
+    Reply(1500);
+    giq_option(4, NAME, 1501, Node998, NEUTRAL_REACTION);
+    giq_option(4, NAME, 1502, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node1550
 begin
-    gsay_reply(1468, 1550);
+    Reply(1550);
     set_local_var(16, 1);
-    giq_option(4, 1468, 1551, Barter, 50);
+    giq_option(4, NAME, 1551, Barter, NEUTRAL_REACTION);
     if ((local_var(10) == 1) and (local_var(17) != 3)) then begin
-        giq_option(4, 1468, 1552, Node1650, 50);
+        giq_option(4, NAME, 1552, Node1650, NEUTRAL_REACTION);
     end
     if (obj_carrying_pid_obj(dude_obj, 309) and (local_var(6) != 1)) then begin
-        giq_option(4, 1468, 1553, Node850, 50);
+        giq_option(4, NAME, 1553, Node850, NEUTRAL_REACTION);
     end
     if (local_var(7) != 1) then begin
-        giq_option(4, 1468, 1554, Node900, 50);
+        giq_option(4, NAME, 1554, Node900, NEUTRAL_REACTION);
     end
     if (local_var(8) != 1) then begin
-        giq_option(4, 1468, 1555, Node1000, 50);
+        giq_option(4, NAME, 1555, Node1000, NEUTRAL_REACTION);
     end
-    giq_option(4, 1468, 1556, Node999, 50);
+    giq_option(4, NAME, 1556, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node1580
 begin
-    gsay_reply(1468, 1580);
-    giq_option(4, 1468, 1581, Node1600, 50);
+    Reply(1580);
+    giq_option(4, NAME, 1581, Node1600, NEUTRAL_REACTION);
 end
 
 procedure Node1600
 begin
-    gsay_reply(1468, 1600);
+    Reply(1600);
     set_local_var(10, 1);
-    giq_option(4, 1468, 1601, Node1650move, 50);
+    giq_option(4, NAME, 1601, Node1650move, NEUTRAL_REACTION);
     if (local_var(9) != 1) then begin
-        giq_option(4, 1468, 1602, Node1300move, 50);
+        giq_option(4, NAME, 1602, Node1300move, NEUTRAL_REACTION);
     end
-    giq_option(4, 1468, 1603, Node1350move, 50);
-    giq_option(4, 1468, 1604, Node999move, 50);
+    giq_option(4, NAME, 1603, Node1350move, NEUTRAL_REACTION);
+    giq_option(4, NAME, 1604, Node999move, NEUTRAL_REACTION);
 end
 
 procedure Node1650
 begin
-    gsay_reply(1468, 1650);
+    Reply(1650);
     if (item_caps_total(dude_obj) >= 600) then begin
         if (local_var(11) != 1) then begin
-            giq_option(4, 1468, 1651, Node1700, 50);
+            giq_option(4, NAME, 1651, Node1700, NEUTRAL_REACTION);
         end
         if (local_var(12) != 1) then begin
-            giq_option(4, 1468, 1652, Node1750, 50);
+            giq_option(4, NAME, 1652, Node1750, NEUTRAL_REACTION);
         end
         if (local_var(13) != 1) then begin
-            giq_option(4, 1468, 1653, Node1800a, 50);
+            giq_option(4, NAME, 1653, Node1800a, NEUTRAL_REACTION);
         end
         if (local_var(14) != 1) then begin
-            giq_option(4, 1468, 1654, Node1800b, 50);
+            giq_option(4, NAME, 1654, Node1800b, NEUTRAL_REACTION);
         end
         if (local_var(15) != 1) then begin
-            giq_option(4, 1468, 1655, Node1800c, 50);
+            giq_option(4, NAME, 1655, Node1800c, NEUTRAL_REACTION);
         end
-        giq_option(4, 1468, 1656, Node1350, 50);
-        giq_option(4, 1468, 1657, Node999, 50);
+        giq_option(4, NAME, 1656, Node1350, NEUTRAL_REACTION);
+        giq_option(4, NAME, 1657, Node999, NEUTRAL_REACTION);
     end
     else begin
-        giq_option(4, 1468, 1658, Node1350, 50);
-        giq_option(4, 1468, 1659, Node999, 50);
+        giq_option(4, NAME, 1658, Node1350, NEUTRAL_REACTION);
+        giq_option(4, NAME, 1659, Node999, NEUTRAL_REACTION);
     end
 end
 
 procedure Node1700
 begin
-    gsay_reply(1468, 1700);
+    Reply(1700);
     set_local_var(11, 1);
     if (item_caps_total(dude_obj) >= 600) then begin
         if (local_var(12) != 1) then begin
-            giq_option(4, 1468, 1701, Node1750, 50);
+            giq_option(4, NAME, 1701, Node1750, NEUTRAL_REACTION);
         end
         if (local_var(13) != 1) then begin
-            giq_option(4, 1468, 1702, Node1800a, 50);
+            giq_option(4, NAME, 1702, Node1800a, NEUTRAL_REACTION);
         end
         if (local_var(14) != 1) then begin
-            giq_option(4, 1468, 1703, Node1800b, 50);
+            giq_option(4, NAME, 1703, Node1800b, NEUTRAL_REACTION);
         end
         if (local_var(15) != 1) then begin
-            giq_option(4, 1468, 1704, Node1800c, 50);
+            giq_option(4, NAME, 1704, Node1800c, NEUTRAL_REACTION);
         end
-        giq_option(4, 1468, 1705, Node1350, 50);
-        giq_option(4, 1468, 1706, Node999, 50);
+        giq_option(4, NAME, 1705, Node1350, NEUTRAL_REACTION);
+        giq_option(4, NAME, 1706, Node999, NEUTRAL_REACTION);
     end
     else begin
-        giq_option(4, 1468, 1707, Node1350, 50);
-        giq_option(4, 1468, 1708, Node999, 50);
+        giq_option(4, NAME, 1707, Node1350, NEUTRAL_REACTION);
+        giq_option(4, NAME, 1708, Node999, NEUTRAL_REACTION);
     end
 end
 
 procedure Node1750
 begin
-    gsay_reply(1468, 1750);
+    Reply(1750);
     set_local_var(12, 1);
     if (item_caps_total(dude_obj) >= 600) then begin
         if (local_var(11) != 1) then begin
-            giq_option(4, 1468, 1751, Node1700, 50);
+            giq_option(4, NAME, 1751, Node1700, NEUTRAL_REACTION);
         end
         if (local_var(13) != 1) then begin
-            giq_option(4, 1468, 1752, Node1800a, 50);
+            giq_option(4, NAME, 1752, Node1800a, NEUTRAL_REACTION);
         end
         if (local_var(14) != 1) then begin
-            giq_option(4, 1468, 1753, Node1800b, 50);
+            giq_option(4, NAME, 1753, Node1800b, NEUTRAL_REACTION);
         end
         if (local_var(15) != 1) then begin
-            giq_option(4, 1468, 1754, Node1800c, 50);
+            giq_option(4, NAME, 1754, Node1800c, NEUTRAL_REACTION);
         end
-        giq_option(4, 1468, 1755, Node1350, 50);
-        giq_option(4, 1468, 1756, Node999, 50);
+        giq_option(4, NAME, 1755, Node1350, NEUTRAL_REACTION);
+        giq_option(4, NAME, 1756, Node999, NEUTRAL_REACTION);
     end
     else begin
-        giq_option(4, 1468, 1757, Node1350, 50);
-        giq_option(4, 1468, 1758, Node999, 50);
+        giq_option(4, NAME, 1757, Node1350, NEUTRAL_REACTION);
+        giq_option(4, NAME, 1758, Node999, NEUTRAL_REACTION);
     end
 end
 
 procedure Node1800a
 begin
-    gsay_reply(1468, 1800);
+    Reply(1800);
     set_local_var(13, 1);
     item_caps_adjust(dude_obj, -600);
     item_caps_adjust(staple_shelf_trade, 600);
-    add_obj_to_inven(dude_obj, create_object_sid(552, 0, 0, 1469));
+    add_obj_to_inven(dude_obj, create_object_sid(552, 0, 0, SCRIPT_SDACHD));
     set_local_var(17, local_var(17) + 1);
-    giq_option(4, 1468, 1801, Barter, 50);
+    giq_option(4, NAME, 1801, Barter, NEUTRAL_REACTION);
     if ((local_var(10) == 1) and (local_var(17) != 3)) then begin
-        giq_option(4, 1468, 1802, Node1650, 50);
+        giq_option(4, NAME, 1802, Node1650, NEUTRAL_REACTION);
     end
     if (obj_carrying_pid_obj(dude_obj, 309) and (local_var(6) != 1)) then begin
-        giq_option(4, 1468, 1803, Node850, 50);
+        giq_option(4, NAME, 1803, Node850, NEUTRAL_REACTION);
     end
     if (local_var(7) != 1) then begin
-        giq_option(4, 1468, 1804, Node900, 50);
+        giq_option(4, NAME, 1804, Node900, NEUTRAL_REACTION);
     end
     if (local_var(8) != 1) then begin
-        giq_option(4, 1468, 1805, Node1000, 50);
+        giq_option(4, NAME, 1805, Node1000, NEUTRAL_REACTION);
     end
-    giq_option(4, 1468, 1806, Node999, 50);
+    giq_option(4, NAME, 1806, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node1800b
 begin
-    gsay_reply(1468, 1800);
+    Reply(1800);
     set_local_var(14, 1);
     item_caps_adjust(dude_obj, -600);
     item_caps_adjust(staple_shelf_trade, 600);
-    add_obj_to_inven(dude_obj, create_object_sid(553, 0, 0, 1379));
+    add_obj_to_inven(dude_obj, create_object_sid(553, 0, 0, SCRIPT_SLOCVHD));
     set_local_var(17, local_var(17) + 1);
-    giq_option(4, 1468, 1801, Barter, 50);
+    giq_option(4, NAME, 1801, Barter, NEUTRAL_REACTION);
     if ((local_var(10) == 1) and (local_var(17) != 3)) then begin
-        giq_option(4, 1468, 1802, Node1650, 50);
+        giq_option(4, NAME, 1802, Node1650, NEUTRAL_REACTION);
     end
     if (obj_carrying_pid_obj(dude_obj, 309) and (local_var(6) != 1)) then begin
-        giq_option(4, 1468, 1803, Node850, 50);
+        giq_option(4, NAME, 1803, Node850, NEUTRAL_REACTION);
     end
     if (local_var(7) != 1) then begin
-        giq_option(4, 1468, 1804, Node900, 50);
+        giq_option(4, NAME, 1804, Node900, NEUTRAL_REACTION);
     end
     if (local_var(8) != 1) then begin
-        giq_option(4, 1468, 1805, Node1000, 50);
+        giq_option(4, NAME, 1805, Node1000, NEUTRAL_REACTION);
     end
-    giq_option(4, 1468, 1806, Node999, 50);
+    giq_option(4, NAME, 1806, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node1800c
 begin
-    gsay_reply(1468, 1800);
+    Reply(1800);
     set_local_var(15, 1);
     item_caps_adjust(dude_obj, -600);
     item_caps_adjust(staple_shelf_trade, 600);
-    add_obj_to_inven(dude_obj, create_object_sid(554, 0, 0, 1378));
+    add_obj_to_inven(dude_obj, create_object_sid(554, 0, 0, SCRIPT_SMUTACHD));
     set_local_var(17, local_var(17) + 1);
-    giq_option(4, 1468, 1801, Barter, 50);
+    giq_option(4, NAME, 1801, Barter, NEUTRAL_REACTION);
     if ((local_var(10) == 1) and (local_var(17) != 3)) then begin
-        giq_option(4, 1468, 1802, Node1650, 50);
+        giq_option(4, NAME, 1802, Node1650, NEUTRAL_REACTION);
     end
     if (obj_carrying_pid_obj(dude_obj, 309) and (local_var(6) != 1)) then begin
-        giq_option(4, 1468, 1803, Node850, 50);
+        giq_option(4, NAME, 1803, Node850, NEUTRAL_REACTION);
     end
     if (local_var(7) != 1) then begin
-        giq_option(4, 1468, 1804, Node900, 50);
+        giq_option(4, NAME, 1804, Node900, NEUTRAL_REACTION);
     end
     if (local_var(8) != 1) then begin
-        giq_option(4, 1468, 1805, Node1000, 50);
+        giq_option(4, NAME, 1805, Node1000, NEUTRAL_REACTION);
     end
-    giq_option(4, 1468, 1806, Node999, 50);
+    giq_option(4, NAME, 1806, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node1850
 begin
-    gsay_reply(1468, 1850);
-    giq_option(4, 1468, 1851, ThrowOutFromShop, 50);
-    giq_option(4, 1468, 1852, Node998, 50);
-    giq_option(-3, 1468, 1853, ThrowOutFromShop, 50);
-    giq_option(-3, 1468, 1854, Node998, 50);
+    Reply(1850);
+    giq_option(4, NAME, 1851, ThrowOutFromShop, NEUTRAL_REACTION);
+    giq_option(4, NAME, 1852, Node998, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 1853, ThrowOutFromShop, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 1854, Node998, NEUTRAL_REACTION);
 end
 
 procedure ThrowOutFromShop
@@ -1000,85 +1000,85 @@ begin
     move_to(dude_obj, 17903, 0);
     if (party_member_obj(16777729) != 0) then begin
         if (party_member_obj(16777729) != 0) then begin
-            if (has_trait(1, party_member_obj(16777729), 666)) then begin
-                critter_attempt_placement(party_member_obj(16777729), tile_num_in_direction(tile_num(dude_obj), (has_trait(1, dude_obj, 10) + 3) % 6, 1), 0);
+            if (obj_is_visible_flag(party_member_obj(16777729))) then begin
+                critter_attempt_placement(party_member_obj(16777729), tile_num_in_direction(dude_tile, (obj_get_rot(dude_obj) + 3) % 6, 1), 0);
             end
         end
     end
-    if ((party_member_obj(16777779) != 0) and ((global_var(910) bwand 32) == 0)) then begin
+    if ((party_member_obj(16777779) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 32) == 0)) then begin
         if (party_member_obj(16777779) != 0) then begin
-            if (has_trait(1, party_member_obj(16777779), 666)) then begin
-                critter_attempt_placement(party_member_obj(16777779), tile_num_in_direction(tile_num(dude_obj), (has_trait(1, dude_obj, 10) + 3) % 6, 1), 0);
+            if (obj_is_visible_flag(party_member_obj(16777779))) then begin
+                critter_attempt_placement(party_member_obj(16777779), tile_num_in_direction(dude_tile, (obj_get_rot(dude_obj) + 3) % 6, 1), 0);
             end
         end
     end
-    if ((party_member_obj(16777811) != 0) and ((global_var(910) bwand 64) == 0)) then begin
+    if ((party_member_obj(16777811) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 64) == 0)) then begin
         if (party_member_obj(16777811) != 0) then begin
-            if (has_trait(1, party_member_obj(16777811), 666)) then begin
-                critter_attempt_placement(party_member_obj(16777811), tile_num_in_direction(tile_num(dude_obj), (has_trait(1, dude_obj, 10) + 3) % 6, 1), 0);
+            if (obj_is_visible_flag(party_member_obj(16777811))) then begin
+                critter_attempt_placement(party_member_obj(16777811), tile_num_in_direction(dude_tile, (obj_get_rot(dude_obj) + 3) % 6, 1), 0);
             end
         end
     end
     if (party_member_obj(16777812) != 0) then begin
         if (party_member_obj(16777812) != 0) then begin
-            if (has_trait(1, party_member_obj(16777812), 666)) then begin
-                critter_attempt_placement(party_member_obj(16777812), tile_num_in_direction(tile_num(dude_obj), (has_trait(1, dude_obj, 10) + 3) % 6, 1), 0);
+            if (obj_is_visible_flag(party_member_obj(16777812))) then begin
+                critter_attempt_placement(party_member_obj(16777812), tile_num_in_direction(dude_tile, (obj_get_rot(dude_obj) + 3) % 6, 1), 0);
             end
         end
     end
-    if ((party_member_obj(16777814) != 0) and ((global_var(910) bwand 128) == 0)) then begin
+    if ((party_member_obj(16777814) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 128) == 0)) then begin
         if (party_member_obj(16777814) != 0) then begin
-            if (has_trait(1, party_member_obj(16777814), 666)) then begin
-                critter_attempt_placement(party_member_obj(16777814), tile_num_in_direction(tile_num(dude_obj), (has_trait(1, dude_obj, 10) + 3) % 6, 1), 0);
+            if (obj_is_visible_flag(party_member_obj(16777814))) then begin
+                critter_attempt_placement(party_member_obj(16777814), tile_num_in_direction(dude_tile, (obj_get_rot(dude_obj) + 3) % 6, 1), 0);
             end
         end
     end
     if (party_member_obj(16777748) != 0) then begin
         if (party_member_obj(16777748) != 0) then begin
-            if (has_trait(1, party_member_obj(16777748), 666)) then begin
-                critter_attempt_placement(party_member_obj(16777748), tile_num_in_direction(tile_num(dude_obj), (has_trait(1, dude_obj, 10) + 3) % 6, 1), 0);
+            if (obj_is_visible_flag(party_member_obj(16777748))) then begin
+                critter_attempt_placement(party_member_obj(16777748), tile_num_in_direction(dude_tile, (obj_get_rot(dude_obj) + 3) % 6, 1), 0);
             end
         end
     end
     if (party_member_obj(16778017) != 0) then begin
         if (party_member_obj(16778017) != 0) then begin
-            if (has_trait(1, party_member_obj(16778017), 666)) then begin
-                critter_attempt_placement(party_member_obj(16778017), tile_num_in_direction(tile_num(dude_obj), (has_trait(1, dude_obj, 10) + 3) % 6, 1), 0);
+            if (obj_is_visible_flag(party_member_obj(16778017))) then begin
+                critter_attempt_placement(party_member_obj(16778017), tile_num_in_direction(dude_tile, (obj_get_rot(dude_obj) + 3) % 6, 1), 0);
             end
         end
     end
     if (party_member_obj(16778016) != 0) then begin
         if (party_member_obj(16778016) != 0) then begin
-            if (has_trait(1, party_member_obj(16778016), 666)) then begin
-                critter_attempt_placement(party_member_obj(16778016), tile_num_in_direction(tile_num(dude_obj), (has_trait(1, dude_obj, 10) + 3) % 6, 1), 0);
+            if (obj_is_visible_flag(party_member_obj(16778016))) then begin
+                critter_attempt_placement(party_member_obj(16778016), tile_num_in_direction(dude_tile, (obj_get_rot(dude_obj) + 3) % 6, 1), 0);
             end
         end
     end
     if (party_member_obj(16778049) != 0) then begin
         if (party_member_obj(16778049) != 0) then begin
-            if (has_trait(1, party_member_obj(16778049), 666)) then begin
-                critter_attempt_placement(party_member_obj(16778049), tile_num_in_direction(tile_num(dude_obj), (has_trait(1, dude_obj, 10) + 3) % 6, 1), 0);
+            if (obj_is_visible_flag(party_member_obj(16778049))) then begin
+                critter_attempt_placement(party_member_obj(16778049), tile_num_in_direction(dude_tile, (obj_get_rot(dude_obj) + 3) % 6, 1), 0);
             end
         end
     end
     if (party_member_obj(16778050) != 0) then begin
         if (party_member_obj(16778050) != 0) then begin
-            if (has_trait(1, party_member_obj(16778050), 666)) then begin
-                critter_attempt_placement(party_member_obj(16778050), tile_num_in_direction(tile_num(dude_obj), (has_trait(1, dude_obj, 10) + 3) % 6, 1), 0);
+            if (obj_is_visible_flag(party_member_obj(16778050))) then begin
+                critter_attempt_placement(party_member_obj(16778050), tile_num_in_direction(dude_tile, (obj_get_rot(dude_obj) + 3) % 6, 1), 0);
             end
         end
     end
     if (party_member_obj(16778051) != 0) then begin
         if (party_member_obj(16778051) != 0) then begin
-            if (has_trait(1, party_member_obj(16778051), 666)) then begin
-                critter_attempt_placement(party_member_obj(16778051), tile_num_in_direction(tile_num(dude_obj), (has_trait(1, dude_obj, 10) + 3) % 6, 1), 0);
+            if (obj_is_visible_flag(party_member_obj(16778051))) then begin
+                critter_attempt_placement(party_member_obj(16778051), tile_num_in_direction(dude_tile, (obj_get_rot(dude_obj) + 3) % 6, 1), 0);
             end
         end
     end
     if (party_member_obj(16777293) != 0) then begin
         if (party_member_obj(16777293) != 0) then begin
-            if (has_trait(1, party_member_obj(16777293), 666)) then begin
-                critter_attempt_placement(party_member_obj(16777293), tile_num_in_direction(tile_num(dude_obj), (has_trait(1, dude_obj, 10) + 3) % 6, 1), 0);
+            if (obj_is_visible_flag(party_member_obj(16777293))) then begin
+                critter_attempt_placement(party_member_obj(16777293), tile_num_in_direction(dude_tile, (obj_get_rot(dude_obj) + 3) % 6, 1), 0);
             end
         end
     end
@@ -1092,7 +1092,7 @@ procedure Barter
 begin
     move_obj_inven_to_obj(stapleton, staple_shelf_swap);
     move_obj_inven_to_obj(staple_shelf_trade, stapleton);
-    if ((party_member_obj(16777779) != 0) and ((global_var(910) bwand 32) == 0)) then begin
+    if ((party_member_obj(16777779) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 32) == 0)) then begin
         gdialog_mod_barter(120 - 15);
     end
     else begin
@@ -1130,7 +1130,7 @@ procedure Node1300move
 begin
     move_obj_inven_to_obj(stapleton, staple_shelf_trade);
     move_obj_inven_to_obj(staple_shelf_swap, stapleton);
-    wield_obj_critter(self_obj, obj_carrying_pid_obj(self_obj, 18));
+    wield_obj(obj_carrying_pid_obj(self_obj, 18));
     call Node1300();
 end
 
@@ -1138,7 +1138,7 @@ procedure Node1350move
 begin
     move_obj_inven_to_obj(stapleton, staple_shelf_trade);
     move_obj_inven_to_obj(staple_shelf_swap, stapleton);
-    wield_obj_critter(self_obj, obj_carrying_pid_obj(self_obj, 18));
+    wield_obj(obj_carrying_pid_obj(self_obj, 18));
     call Node1350();
 end
 
@@ -1146,7 +1146,6 @@ procedure Node1650move
 begin
     move_obj_inven_to_obj(stapleton, staple_shelf_trade);
     move_obj_inven_to_obj(staple_shelf_swap, stapleton);
-    wield_obj_critter(self_obj, obj_carrying_pid_obj(self_obj, 18));
+    wield_obj(obj_carrying_pid_obj(self_obj, 18));
     call Node1650();
 end
-
