diff --git a/Mods/Resurrection/Original/STYLERM.ssl b/Mods/Resurrection/ReDefine/STYLERM.ssl
index 36ee4ae..723e759 100644
--- a/Mods/Resurrection/Original/STYLERM.ssl
+++ b/Mods/Resurrection/ReDefine/STYLERM.ssl
@@ -99,12 +99,12 @@ begin
             return 1;
         end
     end
-    if ((party_member_obj(16777779) != 0) and ((global_var(910) bwand 32) == 0)) then begin
+    if ((party_member_obj(16777779) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 32) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777779)) <= 5) then begin
             return 1;
         end
     end
-    if ((party_member_obj(16777811) != 0) and ((global_var(910) bwand 64) == 0)) then begin
+    if ((party_member_obj(16777811) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 64) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777811)) <= 5) then begin
             return 1;
         end
@@ -119,7 +119,7 @@ begin
             return 1;
         end
     end
-    if ((party_member_obj(16777814) != 0) and ((global_var(910) bwand 128) == 0)) then begin
+    if ((party_member_obj(16777814) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 128) == 0)) then begin
         if (tile_distance_objs(self_obj, party_member_obj(16777814)) <= 5) then begin
             return 1;
         end
@@ -177,42 +177,42 @@ end
 procedure critter_p_proc
 begin
     if (local_var(5) == 2) then begin
-        if (global_var(781) == 1) then begin
-            if (get_critter_stat(dude_obj, 34) == 0) then begin
-                float_msg(self_obj, message_str(1341, 300), 2);
+        if (global_var(GVAR_KEVIN_DESTROY) == 1) then begin
+            if (dude_gender == 0) then begin
+                float_msg(self_obj, mstr(300), FLOAT_MSG_RED);
             end
             else begin
-                float_msg(self_obj, message_str(1341, 301), 2);
+                float_msg(self_obj, mstr(301), FLOAT_MSG_RED);
             end
         end
         set_local_var(5, 1);
-        attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+        attack(dude_obj);
     end
-    if (global_var(769) == 1) then begin
+    if (global_var(GVAR_ENEMY_SEDIT) == 1) then begin
         if (obj_can_see_obj(self_obj, dude_obj) or obj_can_hear_obj(self_obj, dude_obj)) then begin
-            if (global_var(781) == 1) then begin
-                if (get_critter_stat(dude_obj, 34) == 0) then begin
-                    float_msg(self_obj, message_str(1341, 300), 2);
+            if (global_var(GVAR_KEVIN_DESTROY) == 1) then begin
+                if (dude_gender == 0) then begin
+                    float_msg(self_obj, mstr(300), FLOAT_MSG_RED);
                 end
                 else begin
-                    float_msg(self_obj, message_str(1341, 301), 2);
+                    float_msg(self_obj, mstr(301), FLOAT_MSG_RED);
                 end
             end
-            attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+            attack(dude_obj);
         end
     end
-    if ((global_var(720) == 3) and (global_var(722) != 1) and (((24 * (60 * (60 * 10))) + local_var(14)) <= game_time)) then begin
-        set_global_var(722, 1);
+    if ((global_var(GVAR_QUEST_S_TYLER_SHERIFF) == 3) and (global_var(GVAR_SEDIT_TYLER_IS_SHERIFF) != 1) and (((24 * (60 * (60 * 10))) + local_var(14)) <= game_time)) then begin
+        set_global_var(GVAR_SEDIT_TYLER_IS_SHERIFF, 1);
     end
 end
 
 procedure combat_p_proc
 begin
     if (fixed_param == 4) then begin
-        if ((get_critter_stat(dude_obj, 6) == 10) and (combat_cry != 14) and (global_var(769) == 1)) then begin
+        if ((dude_luck == 10) and (combat_cry != 14) and (global_var(GVAR_ENEMY_SEDIT) == 1)) then begin
             combat_cry := random(1, 13);
             if (combat_cry == 10) then begin
-                float_msg(self_obj, message_str(1341, 400), 8);
+                float_msg(self_obj, mstr(400), FLOAT_MSG_YELLOW);
                 combat_cry := 14;
             end
         end
@@ -228,59 +228,59 @@ end
 
 procedure talk_p_proc
 begin
-    if (global_var(781) == 1) then begin
-        float_msg(self_obj, message_str(1341, 300 + (get_critter_stat(dude_obj, 34) == 1)), 2);
+    if (global_var(GVAR_KEVIN_DESTROY) == 1) then begin
+        float_msg(self_obj, mstr(300 + (dude_gender == 1)), FLOAT_MSG_RED);
     end
     else begin
-        anim(dude_obj, 1000, rotation_to_tile(tile_num(dude_obj), tile_num(self_obj)));
-        anim(self_obj, 1000, rotation_to_tile(tile_num(self_obj), tile_num(dude_obj)));
+        anim(dude_obj, 1000, rotation_to_tile(dude_tile, tile_num(self_obj)));
+        anim(self_obj, 1000, rotation_to_tile(tile_num(self_obj), dude_tile));
         start_gdialog(1341, self_obj, 4, -1, -1);
         gsay_start;
-        if ((global_var(733) == 2) and ((get_critter_stat(dude_obj, 4) + has_trait(0, dude_obj, 49)) > 3) and (local_var(11) != 1) and (global_var(723) == 0) and (global_var(724) == 0) and (dialog_loop == 0)) then begin
+        if ((global_var(GVAR_QUEST_S_RODRIGUES_TYLER) == 2) and ((dude_iq + has_trait(TRAIT_PERK, dude_obj, PERK_smooth_talker)) > 3) and (local_var(11) != 1) and (global_var(GVAR_QUEST_S_DAVID_KIDNAP_INVESTIGATION) == 0) and (global_var(GVAR_QUEST_A_DAVID_KIDNAP_COMPLICITY) == 0) and (dialog_loop == 0)) then begin
             dialog_loop := 1;
-            if (get_critter_stat(dude_obj, 34) == 0) then begin
+            if (dude_gender == 0) then begin
                 call Node1200();
             end
-            if (get_critter_stat(dude_obj, 34) == 1) then begin
+            if (dude_gender == 1) then begin
                 call Node1250();
             end
         end
-        if ((global_var(718) == 1) and (dialog_loop == 0)) then begin
+        if ((global_var(GVAR_QUEST_S_TYLER_POISON) == 1) and (dialog_loop == 0)) then begin
             dialog_loop := 1;
             call Node1650();
         end
-        if ((global_var(719) == 1) or (global_var(720) == 1) and (dialog_loop == 0)) then begin
+        if ((global_var(GVAR_QUEST_S_TYLER_NICOL) == 1) or (global_var(GVAR_QUEST_S_TYLER_SHERIFF) == 1) and (dialog_loop == 0)) then begin
             dialog_loop := 1;
             call Node1850();
         end
-        if ((global_var(719) == 2) and (dialog_loop == 0)) then begin
+        if ((global_var(GVAR_QUEST_S_TYLER_NICOL) == 2) and (dialog_loop == 0)) then begin
             dialog_loop := 1;
-            if ((get_critter_stat(dude_obj, 4) + has_trait(0, dude_obj, 49)) <= 3) then begin
+            if ((dude_iq + has_trait(TRAIT_PERK, dude_obj, PERK_smooth_talker)) <= 3) then begin
                 call Node2400();
             end
             else begin
                 call Node1900();
             end
         end
-        if ((global_var(720) == 2) and (dialog_loop == 0)) then begin
+        if ((global_var(GVAR_QUEST_S_TYLER_SHERIFF) == 2) and (dialog_loop == 0)) then begin
             dialog_loop := 1;
-            if ((get_critter_stat(dude_obj, 4) + has_trait(0, dude_obj, 49)) <= 3) then begin
+            if ((dude_iq + has_trait(TRAIT_PERK, dude_obj, PERK_smooth_talker)) <= 3) then begin
                 call Node2400();
             end
             else begin
-                if (get_critter_stat(dude_obj, 34) == 0) then begin
+                if (dude_gender == 0) then begin
                     call Node2100();
                 end
-                if (get_critter_stat(dude_obj, 34) == 1) then begin
+                if (dude_gender == 1) then begin
                     call Node2150();
                 end
             end
         end
-        if ((global_var(720) == 3) and (global_var(722) != 1) and (dialog_loop == 0)) then begin
+        if ((global_var(GVAR_QUEST_S_TYLER_SHERIFF) == 3) and (global_var(GVAR_SEDIT_TYLER_IS_SHERIFF) != 1) and (dialog_loop == 0)) then begin
             dialog_loop := 1;
             call Node2450();
         end
-        if ((global_var(720) == 3) and (global_var(722) == 1) and (dialog_loop == 0)) then begin
+        if ((global_var(GVAR_QUEST_S_TYLER_SHERIFF) == 3) and (global_var(GVAR_SEDIT_TYLER_IS_SHERIFF) == 1) and (dialog_loop == 0)) then begin
             dialog_loop := 1;
             call Node2200();
         end
@@ -296,7 +296,7 @@ begin
         end_dialogue;
         dialog_loop := 0;
         set_local_var(4, 1);
-        set_global_var(760, 1);
+        set_global_var(GVAR_TYLER_SPEECH, 1);
     end
 end
 
@@ -304,15 +304,15 @@ procedure look_at_p_proc
 begin
     script_overrides;
     if (local_var(4) == 1) then begin
-        if (global_var(722) == 1) then begin
-            display_msg(message_str(1341, 170));
+        if (global_var(GVAR_SEDIT_TYLER_IS_SHERIFF) == 1) then begin
+            display_mstr(170);
         end
         else begin
-            display_msg(message_str(1341, 150));
+            display_mstr(150);
         end
     end
     else begin
-        display_msg(message_str(1341, 100));
+        display_mstr(100);
     end
 end
 
@@ -320,15 +320,15 @@ procedure description_p_proc
 begin
     script_overrides;
     if (local_var(4) == 1) then begin
-        if (global_var(722) == 1) then begin
-            display_msg(message_str(1341, 270));
+        if (global_var(GVAR_SEDIT_TYLER_IS_SHERIFF) == 1) then begin
+            display_mstr(270);
         end
         else begin
-            display_msg(message_str(1341, 250));
+            display_mstr(250);
         end
     end
     else begin
-        display_msg(message_str(1341, 200));
+        display_mstr(200);
     end
 end
 
@@ -339,49 +339,49 @@ begin
         if (prison_sequence == 0) then begin
             game_ui_disable;
             gfade_in(5);
-            display_msg(message_str(1341, 5000));
+            display_mstr(5000);
             add_timer_event(self_obj, game_ticks(6), 2);
         end
         if (prison_sequence == 1) then begin
             game_time_advance(game_ticks(16 * 60 * 60));
             gfade_in(5);
-            display_msg(message_str(1341, 5016));
-            display_msg(message_str(1341, 5001));
+            display_mstr(5016);
+            display_mstr(5001);
             add_timer_event(self_obj, game_ticks(6), 2);
         end
         if (prison_sequence == 2) then begin
             game_time_advance(game_ticks(8 * 60 * 60));
             gfade_in(5);
-            display_msg(message_str(1341, 5008));
-            if (get_critter_stat(dude_obj, 34) == 0) then begin
-                display_msg(message_str(1341, 5002));
+            display_mstr(5008);
+            if (dude_gender == 0) then begin
+                display_mstr(5002);
             end
-            if (get_critter_stat(dude_obj, 34) == 1) then begin
-                display_msg(message_str(1341, 5003));
+            if (dude_gender == 1) then begin
+                display_mstr(5003);
             end
             add_timer_event(self_obj, game_ticks(6), 2);
         end
         if (prison_sequence == 3) then begin
             game_time_advance(game_ticks(8 * 60 * 60));
             gfade_in(5);
-            display_msg(message_str(1341, 5008));
-            display_msg(message_str(1341, 5004));
+            display_mstr(5008);
+            display_mstr(5004);
             add_timer_event(self_obj, game_ticks(6), 2);
         end
         if (prison_sequence == 4) then begin
             game_time_advance(game_ticks(16 * 60 * 60));
-            if (get_critter_stat(dude_obj, 35) > 2) then begin
-                critter_dmg(dude_obj, 2, 0);
+            if (get_critter_stat(dude_obj, STAT_current_hp) > 2) then begin
+                critter_damage(dude_obj, 2);
             end
             move_to(dude_obj, 14306, 0);
             anim(dude_obj, 1000, 2);
-            set_global_var(632, global_var(632) bwand (-1 - 64));
+            set_global_var(GVAR_PARTY_MEMBERS_HIDDEN, global_var(GVAR_PARTY_MEMBERS_HIDDEN) bwand (-1 - 64));
             if (party_member_obj(16777729) != 0) then begin
                 if (party_member_obj(16777729)) then begin
-                    if (has_trait(1, party_member_obj(16777729), 666) == 0) then begin
-                        if ((global_var(632) bwand 1) != 0) then begin
+                    if (obj_is_visible_flag(party_member_obj(16777729)) == 0) then begin
+                        if ((global_var(GVAR_PARTY_MEMBERS_HIDDEN) bwand 1) != 0) then begin
                             debug_msg("unhiding party member " + obj_name(party_member_obj(16777729)));
-                            set_global_var(632, global_var(632) bwand (-1 - 1));
+                            set_global_var(GVAR_PARTY_MEMBERS_HIDDEN, global_var(GVAR_PARTY_MEMBERS_HIDDEN) bwand (-1 - 1));
                             set_obj_visibility(party_member_obj(16777729), 0);
                         end
                         else begin
@@ -390,18 +390,18 @@ begin
                     end
                     else begin
                         debug_msg("unhiding party member " + obj_name(party_member_obj(16777729)));
-                        set_global_var(632, global_var(632) bwand (-1 - 1));
+                        set_global_var(GVAR_PARTY_MEMBERS_HIDDEN, global_var(GVAR_PARTY_MEMBERS_HIDDEN) bwand (-1 - 1));
                         set_obj_visibility(party_member_obj(16777729), 0);
                         debug_msg("PARTY.H:::: party member wasn't hidden.. clearing my state");
                     end
                 end
             end
-            if ((party_member_obj(16777779) != 0) and ((global_var(910) bwand 32) == 0)) then begin
+            if ((party_member_obj(16777779) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 32) == 0)) then begin
                 if (party_member_obj(16777779)) then begin
-                    if (has_trait(1, party_member_obj(16777779), 666) == 0) then begin
-                        if ((global_var(632) bwand 2) != 0) then begin
+                    if (obj_is_visible_flag(party_member_obj(16777779)) == 0) then begin
+                        if ((global_var(GVAR_PARTY_MEMBERS_HIDDEN) bwand 2) != 0) then begin
                             debug_msg("unhiding party member " + obj_name(party_member_obj(16777779)));
-                            set_global_var(632, global_var(632) bwand (-1 - 2));
+                            set_global_var(GVAR_PARTY_MEMBERS_HIDDEN, global_var(GVAR_PARTY_MEMBERS_HIDDEN) bwand (-1 - 2));
                             set_obj_visibility(party_member_obj(16777779), 0);
                         end
                         else begin
@@ -410,18 +410,18 @@ begin
                     end
                     else begin
                         debug_msg("unhiding party member " + obj_name(party_member_obj(16777779)));
-                        set_global_var(632, global_var(632) bwand (-1 - 2));
+                        set_global_var(GVAR_PARTY_MEMBERS_HIDDEN, global_var(GVAR_PARTY_MEMBERS_HIDDEN) bwand (-1 - 2));
                         set_obj_visibility(party_member_obj(16777779), 0);
                         debug_msg("PARTY.H:::: party member wasn't hidden.. clearing my state");
                     end
                 end
             end
-            if ((party_member_obj(16777811) != 0) and ((global_var(910) bwand 64) == 0)) then begin
+            if ((party_member_obj(16777811) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 64) == 0)) then begin
                 if (party_member_obj(16777811)) then begin
-                    if (has_trait(1, party_member_obj(16777811), 666) == 0) then begin
-                        if ((global_var(632) bwand 4) != 0) then begin
+                    if (obj_is_visible_flag(party_member_obj(16777811)) == 0) then begin
+                        if ((global_var(GVAR_PARTY_MEMBERS_HIDDEN) bwand 4) != 0) then begin
                             debug_msg("unhiding party member " + obj_name(party_member_obj(16777811)));
-                            set_global_var(632, global_var(632) bwand (-1 - 4));
+                            set_global_var(GVAR_PARTY_MEMBERS_HIDDEN, global_var(GVAR_PARTY_MEMBERS_HIDDEN) bwand (-1 - 4));
                             set_obj_visibility(party_member_obj(16777811), 0);
                         end
                         else begin
@@ -430,7 +430,7 @@ begin
                     end
                     else begin
                         debug_msg("unhiding party member " + obj_name(party_member_obj(16777811)));
-                        set_global_var(632, global_var(632) bwand (-1 - 4));
+                        set_global_var(GVAR_PARTY_MEMBERS_HIDDEN, global_var(GVAR_PARTY_MEMBERS_HIDDEN) bwand (-1 - 4));
                         set_obj_visibility(party_member_obj(16777811), 0);
                         debug_msg("PARTY.H:::: party member wasn't hidden.. clearing my state");
                     end
@@ -438,10 +438,10 @@ begin
             end
             if (party_member_obj(16777812) != 0) then begin
                 if (party_member_obj(16777812)) then begin
-                    if (has_trait(1, party_member_obj(16777812), 666) == 0) then begin
-                        if ((global_var(632) bwand 8) != 0) then begin
+                    if (obj_is_visible_flag(party_member_obj(16777812)) == 0) then begin
+                        if ((global_var(GVAR_PARTY_MEMBERS_HIDDEN) bwand 8) != 0) then begin
                             debug_msg("unhiding party member " + obj_name(party_member_obj(16777812)));
-                            set_global_var(632, global_var(632) bwand (-1 - 8));
+                            set_global_var(GVAR_PARTY_MEMBERS_HIDDEN, global_var(GVAR_PARTY_MEMBERS_HIDDEN) bwand (-1 - 8));
                             set_obj_visibility(party_member_obj(16777812), 0);
                         end
                         else begin
@@ -450,7 +450,7 @@ begin
                     end
                     else begin
                         debug_msg("unhiding party member " + obj_name(party_member_obj(16777812)));
-                        set_global_var(632, global_var(632) bwand (-1 - 8));
+                        set_global_var(GVAR_PARTY_MEMBERS_HIDDEN, global_var(GVAR_PARTY_MEMBERS_HIDDEN) bwand (-1 - 8));
                         set_obj_visibility(party_member_obj(16777812), 0);
                         debug_msg("PARTY.H:::: party member wasn't hidden.. clearing my state");
                     end
@@ -458,10 +458,10 @@ begin
             end
             if (party_member_obj(16777813) != 0) then begin
                 if (party_member_obj(16777813)) then begin
-                    if (has_trait(1, party_member_obj(16777813), 666) == 0) then begin
-                        if ((global_var(632) bwand 16) != 0) then begin
+                    if (obj_is_visible_flag(party_member_obj(16777813)) == 0) then begin
+                        if ((global_var(GVAR_PARTY_MEMBERS_HIDDEN) bwand 16) != 0) then begin
                             debug_msg("unhiding party member " + obj_name(party_member_obj(16777813)));
-                            set_global_var(632, global_var(632) bwand (-1 - 16));
+                            set_global_var(GVAR_PARTY_MEMBERS_HIDDEN, global_var(GVAR_PARTY_MEMBERS_HIDDEN) bwand (-1 - 16));
                             set_obj_visibility(party_member_obj(16777813), 0);
                         end
                         else begin
@@ -470,18 +470,18 @@ begin
                     end
                     else begin
                         debug_msg("unhiding party member " + obj_name(party_member_obj(16777813)));
-                        set_global_var(632, global_var(632) bwand (-1 - 16));
+                        set_global_var(GVAR_PARTY_MEMBERS_HIDDEN, global_var(GVAR_PARTY_MEMBERS_HIDDEN) bwand (-1 - 16));
                         set_obj_visibility(party_member_obj(16777813), 0);
                         debug_msg("PARTY.H:::: party member wasn't hidden.. clearing my state");
                     end
                 end
             end
-            if ((party_member_obj(16777814) != 0) and ((global_var(910) bwand 128) == 0)) then begin
+            if ((party_member_obj(16777814) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 128) == 0)) then begin
                 if (party_member_obj(16777814)) then begin
-                    if (has_trait(1, party_member_obj(16777814), 666) == 0) then begin
-                        if ((global_var(632) bwand 32) != 0) then begin
+                    if (obj_is_visible_flag(party_member_obj(16777814)) == 0) then begin
+                        if ((global_var(GVAR_PARTY_MEMBERS_HIDDEN) bwand 32) != 0) then begin
                             debug_msg("unhiding party member " + obj_name(party_member_obj(16777814)));
-                            set_global_var(632, global_var(632) bwand (-1 - 32));
+                            set_global_var(GVAR_PARTY_MEMBERS_HIDDEN, global_var(GVAR_PARTY_MEMBERS_HIDDEN) bwand (-1 - 32));
                             set_obj_visibility(party_member_obj(16777814), 0);
                         end
                         else begin
@@ -490,7 +490,7 @@ begin
                     end
                     else begin
                         debug_msg("unhiding party member " + obj_name(party_member_obj(16777814)));
-                        set_global_var(632, global_var(632) bwand (-1 - 32));
+                        set_global_var(GVAR_PARTY_MEMBERS_HIDDEN, global_var(GVAR_PARTY_MEMBERS_HIDDEN) bwand (-1 - 32));
                         set_obj_visibility(party_member_obj(16777814), 0);
                         debug_msg("PARTY.H:::: party member wasn't hidden.. clearing my state");
                     end
@@ -506,15 +506,15 @@ begin
                 end
                 item_caps_adjust(dude_obj, -LVar0);
             end
-            set_global_var(48, global_var(48) - 2);
+            set_global_var(GVAR_TOWN_REP_SEDIT, global_var(GVAR_TOWN_REP_SEDIT) - 2);
             game_ui_enable;
             gfade_in(5);
-            display_msg(message_str(1341, 5016));
-            if (get_critter_stat(dude_obj, 34) == 0) then begin
-                display_msg(message_str(1341, 5005));
+            display_mstr(5016);
+            if (dude_gender == 0) then begin
+                display_mstr(5005);
             end
-            if (get_critter_stat(dude_obj, 34) == 1) then begin
-                display_msg(message_str(1341, 5006));
+            if (dude_gender == 1) then begin
+                display_mstr(5006);
             end
         end
         prison_sequence := prison_sequence + 1;
@@ -527,9 +527,9 @@ end
 
 procedure map_enter_p_proc
 begin
-    critter_add_trait(self_obj, 1, 6, 212);
-    critter_add_trait(self_obj, 1, 5, 146);
-    Tyler_hp := get_critter_stat(self_obj, 35);
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 212);
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_SEDIT_POLICE);
+    Tyler_hp := get_critter_stat(self_obj, STAT_current_hp);
     if (local_var(15) == 0) then begin
         set_local_var(15, tile_num(self_obj));
     end
@@ -538,7 +538,7 @@ end
 procedure map_exit_p_proc
 begin
     if (tile_num(self_obj) != local_var(15)) then begin
-        if (global_var(769) != 1) then begin
+        if (global_var(GVAR_ENEMY_SEDIT) != 1) then begin
             move_to(self_obj, local_var(15), elevation(self_obj));
             anim(self_obj, 1000, 2);
         end
@@ -549,73 +549,73 @@ procedure damage_p_proc
 begin
     if (source_obj == dude_obj) then begin
         call Node998();
-        set_global_var(769, 1);
+        set_global_var(GVAR_ENEMY_SEDIT, 1);
     end
-    if (((Tyler_hp - get_critter_stat(self_obj, 35)) > 18) and not(critter_state(self_obj) bwand 1)) then begin
-        if (get_critter_stat(dude_obj, 34) == 0) then begin
-            float_msg(self_obj, message_str(1341, random(401, 402)), 2);
+    if (((Tyler_hp - get_critter_stat(self_obj, STAT_current_hp)) > 18) and not(critter_state(self_obj) bwand 1)) then begin
+        if (dude_gender == 0) then begin
+            float_msg(self_obj, mstr(random(401, 402)), FLOAT_MSG_RED);
         end
-        if (get_critter_stat(dude_obj, 34) == 1) then begin
-            float_msg(self_obj, message_str(1341, random(403, 404)), 2);
+        if (dude_gender == 1) then begin
+            float_msg(self_obj, mstr(random(403, 404)), FLOAT_MSG_RED);
         end
     end
-    Tyler_hp := get_critter_stat(self_obj, 35);
+    Tyler_hp := get_critter_stat(self_obj, STAT_current_hp);
 end
 
 procedure destroy_p_proc
 begin
-    set_global_var(976, 1);
-    set_global_var(1018, global_var(1018) bwand (-1 - 16384));
+    set_global_var(GVAR_S_TYLER_DEAD, 1);
+    set_global_var(GVAR_PTR_GABRIEL_REACTIONS, global_var(GVAR_PTR_GABRIEL_REACTIONS) bwand (-1 - 16384));
     if (source_obj == dude_obj) then begin
-        set_global_var(4, global_var(4) + 1);
-        if (metarule(51, self_obj) == 2) then begin
+        set_global_var(GVAR_BAD_MONSTER, global_var(GVAR_BAD_MONSTER) + 1);
+        if (critter_kill_type(self_obj) == KILL_TYPE_children_kills) then begin
             if (source_obj == dude_obj) then begin
-                set_global_var(2, 0);
-                set_global_var(1, global_var(1) + 1);
-                set_global_var(0, global_var(0) + -15);
+                set_global_var(GVAR_CHAMPION_REPUTATION, 0);
+                set_global_var(GVAR_CHILDKILLER_REPUTATION, global_var(GVAR_CHILDKILLER_REPUTATION) + 1);
+                set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + -15);
                 debug_msg("Player gains " + -15 + " Karma Points.");
-                set_global_var(37, 0);
-                set_global_var(38, 0);
-                set_global_var(39, 0);
-                set_global_var(40, 0);
-                set_global_var(41, 0);
-                set_global_var(42, 0);
-                set_global_var(43, 0);
-                set_global_var(44, 0);
-                set_global_var(45, 0);
-                if (global_var(0) >= 1000) then begin
-                    set_global_var(37, 1);
+                set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+                set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+                set_global_var(GVAR_KARMA_DEFENDER, 0);
+                set_global_var(GVAR_KARMA_WANDERER, 0);
+                set_global_var(GVAR_KARMA_BETRAYER, 0);
+                set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+                set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+                set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+                    set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
                 end
                 else begin
-                    if (global_var(0) >= 750) then begin
-                        set_global_var(38, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
                     end
                     else begin
-                        if (global_var(0) >= 500) then begin
-                            set_global_var(39, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                            set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                         end
                         else begin
-                            if (global_var(0) >= 250) then begin
-                                set_global_var(40, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                                set_global_var(GVAR_KARMA_DEFENDER, 1);
                             end
                             else begin
-                                if (global_var(0) > -250) then begin
-                                    set_global_var(41, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                                    set_global_var(GVAR_KARMA_WANDERER, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -500) then begin
-                                        set_global_var(42, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                        set_global_var(GVAR_KARMA_BETRAYER, 1);
                                     end
                                     else begin
-                                        if (global_var(0) > -750) then begin
-                                            set_global_var(43, 1);
+                                        if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                            set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                         end
                                         else begin
-                                            if (global_var(0) > -1000) then begin
-                                                set_global_var(44, 1);
+                                            if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                                set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                             end
                                             else begin
-                                                set_global_var(45, 1);
+                                                set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                             end
                                         end
                                     end
@@ -624,62 +624,62 @@ begin
                         end
                     end
                 end
-                set_global_var(48, global_var(48) + -8);
+                set_global_var(GVAR_TOWN_REP_SEDIT, global_var(GVAR_TOWN_REP_SEDIT) + -8);
                 debug_msg("Added " + -8 + " to Town Rep");
             end
         end
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(5) > (2 * global_var(4))) or (global_var(3) == 1))) then begin
-            set_global_var(3, 1);
-            set_global_var(2, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
+            set_global_var(GVAR_BERSERKER_REPUTATION, 1);
+            set_global_var(GVAR_CHAMPION_REPUTATION, 0);
         end
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(4) > (3 * global_var(5))) or (global_var(2) == 1)) and (global_var(1) == 0)) then begin
-            set_global_var(2, 1);
-            set_global_var(3, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(4) > (3 * global_var(GVAR_GOOD_MONSTER))) or (global_var(GVAR_CHAMPION_REPUTATION) == 1)) and (global_var(GVAR_CHILDKILLER_REPUTATION) == 0)) then begin
+            set_global_var(GVAR_CHAMPION_REPUTATION, 1);
+            set_global_var(GVAR_BERSERKER_REPUTATION, 0);
         end
-        set_global_var(0, global_var(0) + 5);
+        set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + 5);
         debug_msg("Player gains " + 5 + " Karma Points.");
-        set_global_var(37, 0);
-        set_global_var(38, 0);
-        set_global_var(39, 0);
-        set_global_var(40, 0);
-        set_global_var(41, 0);
-        set_global_var(42, 0);
-        set_global_var(43, 0);
-        set_global_var(44, 0);
-        set_global_var(45, 0);
-        if (global_var(0) >= 1000) then begin
-            set_global_var(37, 1);
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+        set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+        set_global_var(GVAR_KARMA_DEFENDER, 0);
+        set_global_var(GVAR_KARMA_WANDERER, 0);
+        set_global_var(GVAR_KARMA_BETRAYER, 0);
+        set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+        set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+            set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
         end
         else begin
-            if (global_var(0) >= 750) then begin
-                set_global_var(38, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
             end
             else begin
-                if (global_var(0) >= 500) then begin
-                    set_global_var(39, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                 end
                 else begin
-                    if (global_var(0) >= 250) then begin
-                        set_global_var(40, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                        set_global_var(GVAR_KARMA_DEFENDER, 1);
                     end
                     else begin
-                        if (global_var(0) > -250) then begin
-                            set_global_var(41, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                            set_global_var(GVAR_KARMA_WANDERER, 1);
                         end
                         else begin
-                            if (global_var(0) > -500) then begin
-                                set_global_var(42, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                set_global_var(GVAR_KARMA_BETRAYER, 1);
                             end
                             else begin
-                                if (global_var(0) > -750) then begin
-                                    set_global_var(43, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -1000) then begin
-                                        set_global_var(44, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                     end
                                     else begin
-                                        set_global_var(45, 1);
+                                        set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                     end
                                 end
                             end
@@ -688,7 +688,7 @@ begin
                 end
             end
         end
-        set_global_var(48, global_var(48) + -5);
+        set_global_var(GVAR_TOWN_REP_SEDIT, global_var(GVAR_TOWN_REP_SEDIT) + -5);
         debug_msg("Added " + 2 + " to Town Rep");
     end
 end
@@ -705,270 +705,270 @@ end
 
 procedure Node500
 begin
-    gsay_reply(1341, 500);
-    if ((get_critter_stat(dude_obj, 34) == 0) and (local_var(6) != 1)) then begin
-        giq_option(4, 1341, 501, Node550, 50);
+    Reply(500);
+    if ((dude_gender == 0) and (local_var(6) != 1)) then begin
+        giq_option(4, NAME, 501, Node550, NEUTRAL_REACTION);
     end
-    if ((get_critter_stat(dude_obj, 34) == 1) and (local_var(7) != 1)) then begin
-        giq_option(4, 1341, 502, Node600, 50);
+    if ((dude_gender == 1) and (local_var(7) != 1)) then begin
+        giq_option(4, NAME, 502, Node600, NEUTRAL_REACTION);
     end
-    giq_option(4, 1341, 503, Node650, 50);
-    giq_option(4, 1341, 504, Node700, 50);
-    giq_option(-3, 1341, 505, Node1000, 50);
-    if ((global_var(718) == 0) and (local_var(11) == 1)) then begin
-        if (get_critter_stat(dude_obj, 34) == 0) then begin
-            giq_option(4, 1341, 506, Node1300, 50);
+    giq_option(4, NAME, 503, Node650, NEUTRAL_REACTION);
+    giq_option(4, NAME, 504, Node700, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 505, Node1000, NEUTRAL_REACTION);
+    if ((global_var(GVAR_QUEST_S_TYLER_POISON) == 0) and (local_var(11) == 1)) then begin
+        if (dude_gender == 0) then begin
+            giq_option(4, NAME, 506, Node1300, NEUTRAL_REACTION);
         end
-        if (get_critter_stat(dude_obj, 34) == 1) then begin
-            giq_option(4, 1341, 507, Node1350, 50);
+        if (dude_gender == 1) then begin
+            giq_option(4, NAME, 507, Node1350, NEUTRAL_REACTION);
         end
     end
-    if ((global_var(719) == 0) and (local_var(12) == 1)) then begin
-        giq_option(4, 1341, 508, Node1750, 50);
+    if ((global_var(GVAR_QUEST_S_TYLER_NICOL) == 0) and (local_var(12) == 1)) then begin
+        giq_option(4, NAME, 508, Node1750, NEUTRAL_REACTION);
     end
-    if ((global_var(720) == 0) and (local_var(13) == 1)) then begin
-        if (global_var(732) == 3) then begin
-            giq_option(4, 1341, 509, Node1970, 50);
+    if ((global_var(GVAR_QUEST_S_TYLER_SHERIFF) == 0) and (local_var(13) == 1)) then begin
+        if (global_var(GVAR_QUEST_S_KEVIN_MEETING) == 3) then begin
+            giq_option(4, NAME, 509, Node1970, NEUTRAL_REACTION);
         end
         else begin
-            giq_option(4, 1341, 509, Node1950, 50);
+            giq_option(4, NAME, 509, Node1950, NEUTRAL_REACTION);
         end
     end
-    giq_option(4, 1341, 510, Node999, 50);
+    giq_option(4, NAME, 510, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node550
 begin
-    gsay_reply(1341, 550);
+    Reply(550);
     set_local_var(6, 1);
-    giq_option(4, 1341, 551, Node650, 50);
-    giq_option(4, 1341, 552, Node700, 50);
-    giq_option(4, 1341, 553, Node999, 50);
+    giq_option(4, NAME, 551, Node650, NEUTRAL_REACTION);
+    giq_option(4, NAME, 552, Node700, NEUTRAL_REACTION);
+    giq_option(4, NAME, 553, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node600
 begin
-    gsay_reply(1341, 600);
+    Reply(600);
     set_local_var(7, 1);
-    giq_option(4, 1341, 601, Node650, 50);
-    giq_option(4, 1341, 602, Node700, 50);
-    giq_option(4, 1341, 603, Node999, 50);
+    giq_option(4, NAME, 601, Node650, NEUTRAL_REACTION);
+    giq_option(4, NAME, 602, Node700, NEUTRAL_REACTION);
+    giq_option(4, NAME, 603, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node650
 begin
-    gsay_reply(1341, 650);
+    Reply(650);
     if (local_var(10) != 1) then begin
-        giq_option(4, 1341, 651, Node750, 50);
+        giq_option(4, NAME, 651, Node750, NEUTRAL_REACTION);
     end
     else begin
-        giq_option(4, 1341, 651, Node950, 50);
+        giq_option(4, NAME, 651, Node950, NEUTRAL_REACTION);
     end
-    if ((global_var(733) == 2) and (local_var(8) != 1)) then begin
-        if ((global_var(731) == 1) or (global_var(724) == 12)) then begin
-            giq_option(4, 1341, 652 + (get_critter_stat(dude_obj, 34) == 1), Node2350, 50);
+    if ((global_var(GVAR_QUEST_S_RODRIGUES_TYLER) == 2) and (local_var(8) != 1)) then begin
+        if ((global_var(GVAR_DAVID_DESTROY) == 1) or (global_var(GVAR_QUEST_A_DAVID_KIDNAP_COMPLICITY) == 12)) then begin
+            giq_option(4, NAME, 652 + (dude_gender == 1), Node2350, NEUTRAL_REACTION);
         end
         else begin
-            if (global_var(724) >= 4) then begin
-                giq_option(4, 1341, 652 + (get_critter_stat(dude_obj, 34) == 1), Node2250, 50);
+            if (global_var(GVAR_QUEST_A_DAVID_KIDNAP_COMPLICITY) >= 4) then begin
+                giq_option(4, NAME, 652 + (dude_gender == 1), Node2250, NEUTRAL_REACTION);
             end
             else begin
-                giq_option(4, 1341, 652 + (get_critter_stat(dude_obj, 34) == 1), Node2270, 50);
+                giq_option(4, NAME, 652 + (dude_gender == 1), Node2270, NEUTRAL_REACTION);
             end
         end
     end
-    giq_option(4, 1341, 654, Node999, 50);
+    giq_option(4, NAME, 654, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node700
 begin
-    gsay_reply(1341, 700);
+    Reply(700);
     if (local_var(10) != 1) then begin
-        giq_option(4, 1341, 701, Node750, 50);
+        giq_option(4, NAME, 701, Node750, NEUTRAL_REACTION);
     end
     else begin
-        giq_option(4, 1341, 701, Node950, 50);
+        giq_option(4, NAME, 701, Node950, NEUTRAL_REACTION);
     end
-    giq_option(4, 1341, 702, Node999, 50);
+    giq_option(4, NAME, 702, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node750
 begin
-    gsay_reply(1341, 750);
-    if (get_critter_stat(dude_obj, 34) == 0) then begin
-        giq_option(4, 1341, 751, Node800, 50);
+    Reply(750);
+    if (dude_gender == 0) then begin
+        giq_option(4, NAME, 751, Node800, NEUTRAL_REACTION);
     end
-    if (get_critter_stat(dude_obj, 34) == 1) then begin
-        giq_option(4, 1341, 751, Node850, 50);
+    if (dude_gender == 1) then begin
+        giq_option(4, NAME, 751, Node850, NEUTRAL_REACTION);
     end
-    giq_option(4, 1341, 752, Node999, 50);
+    giq_option(4, NAME, 752, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node800
 begin
-    gsay_reply(1341, 800);
-    giq_option(4, 1341, 801, Dude_arrest, 50);
-    giq_option(4, 1341, 802, Node998, 50);
+    Reply(800);
+    giq_option(4, NAME, 801, Dude_arrest, NEUTRAL_REACTION);
+    giq_option(4, NAME, 802, Node998, NEUTRAL_REACTION);
 end
 
 procedure Node850
 begin
-    gsay_reply(1341, 850);
-    giq_option(4, 1341, 851, Dude_arrest, 50);
-    giq_option(4, 1341, 852, Node998, 50);
+    Reply(850);
+    giq_option(4, NAME, 851, Dude_arrest, NEUTRAL_REACTION);
+    giq_option(4, NAME, 852, Node998, NEUTRAL_REACTION);
 end
 
 procedure Node900
 begin
-    gsay_reply(1341, 900);
-    if ((get_critter_stat(dude_obj, 34) == 0) and (local_var(6) != 1)) then begin
-        giq_option(4, 1341, 901, Node550, 50);
+    Reply(900);
+    if ((dude_gender == 0) and (local_var(6) != 1)) then begin
+        giq_option(4, NAME, 901, Node550, NEUTRAL_REACTION);
     end
-    if ((get_critter_stat(dude_obj, 34) == 1) and (local_var(7) != 1)) then begin
-        giq_option(4, 1341, 902, Node600, 50);
+    if ((dude_gender == 1) and (local_var(7) != 1)) then begin
+        giq_option(4, NAME, 902, Node600, NEUTRAL_REACTION);
     end
-    giq_option(4, 1341, 903, Node650, 50);
-    giq_option(4, 1341, 904, Node700, 50);
-    giq_option(-3, 1341, 905, Node1150, 50);
-    if ((global_var(718) == 0) and (local_var(11) == 1)) then begin
-        if (get_critter_stat(dude_obj, 34) == 0) then begin
-            giq_option(4, 1341, 906, Node1300, 50);
+    giq_option(4, NAME, 903, Node650, NEUTRAL_REACTION);
+    giq_option(4, NAME, 904, Node700, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 905, Node1150, NEUTRAL_REACTION);
+    if ((global_var(GVAR_QUEST_S_TYLER_POISON) == 0) and (local_var(11) == 1)) then begin
+        if (dude_gender == 0) then begin
+            giq_option(4, NAME, 906, Node1300, NEUTRAL_REACTION);
         end
-        if (get_critter_stat(dude_obj, 34) == 1) then begin
-            giq_option(4, 1341, 907, Node1350, 50);
+        if (dude_gender == 1) then begin
+            giq_option(4, NAME, 907, Node1350, NEUTRAL_REACTION);
         end
     end
-    if ((global_var(719) == 0) and (local_var(12) == 1)) then begin
-        giq_option(4, 1341, 908, Node1750, 50);
+    if ((global_var(GVAR_QUEST_S_TYLER_NICOL) == 0) and (local_var(12) == 1)) then begin
+        giq_option(4, NAME, 908, Node1750, NEUTRAL_REACTION);
     end
-    if ((global_var(720) == 0) and (local_var(13) == 1)) then begin
-        if (global_var(732) == 3) then begin
-            giq_option(4, 1341, 909, Node1970, 50);
+    if ((global_var(GVAR_QUEST_S_TYLER_SHERIFF) == 0) and (local_var(13) == 1)) then begin
+        if (global_var(GVAR_QUEST_S_KEVIN_MEETING) == 3) then begin
+            giq_option(4, NAME, 909, Node1970, NEUTRAL_REACTION);
         end
         else begin
-            giq_option(4, 1341, 909, Node1950, 50);
+            giq_option(4, NAME, 909, Node1950, NEUTRAL_REACTION);
         end
     end
-    giq_option(4, 1341, 910, Node999, 50);
+    giq_option(4, NAME, 910, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node950
 begin
-    gsay_reply(1341, 950);
-    giq_option(4, 1341, 951, Node998, 50);
-    giq_option(4, 1341, 952, Node999, 50);
+    Reply(950);
+    giq_option(4, NAME, 951, Node998, NEUTRAL_REACTION);
+    giq_option(4, NAME, 952, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node1000
 begin
-    gsay_reply(1341, 1000);
-    if (get_critter_stat(dude_obj, 34) == 0) then begin
-        giq_option(-3, 1341, 1001, Node1050, 50);
+    Reply(1000);
+    if (dude_gender == 0) then begin
+        giq_option(-3, NAME, 1001, Node1050, NEUTRAL_REACTION);
     end
-    if (get_critter_stat(dude_obj, 34) == 1) then begin
-        giq_option(-3, 1341, 1001, Node1100, 50);
+    if (dude_gender == 1) then begin
+        giq_option(-3, NAME, 1001, Node1100, NEUTRAL_REACTION);
     end
-    giq_option(-3, 1341, 1002, Node999, 50);
+    giq_option(-3, NAME, 1002, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node1050
 begin
-    gsay_reply(1341, 1050);
-    giq_option(-3, 1341, 1051, Dude_arrest, 50);
-    giq_option(-3, 1341, 1052, Node998, 50);
+    Reply(1050);
+    giq_option(-3, NAME, 1051, Dude_arrest, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 1052, Node998, NEUTRAL_REACTION);
 end
 
 procedure Node1100
 begin
-    gsay_reply(1341, 1100);
-    giq_option(-3, 1341, 1101, Dude_arrest, 50);
-    giq_option(-3, 1341, 1102, Node998, 50);
+    Reply(1100);
+    giq_option(-3, NAME, 1101, Dude_arrest, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 1102, Node998, NEUTRAL_REACTION);
 end
 
 procedure Node1150
 begin
-    gsay_reply(1341, 1150);
-    giq_option(-3, 1341, 1151, Node998, 50);
-    giq_option(-3, 1341, 1152, Node999, 50);
+    Reply(1150);
+    giq_option(-3, NAME, 1151, Node998, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 1152, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node1200
 begin
-    gsay_reply(1341, 1200);
-    giq_option(4, 1341, 1201, Node1300, 50);
+    Reply(1200);
+    giq_option(4, NAME, 1201, Node1300, NEUTRAL_REACTION);
 end
 
 procedure Node1250
 begin
-    gsay_reply(1341, 1250);
-    giq_option(4, 1341, 1251, Node1350, 50);
+    Reply(1250);
+    giq_option(4, NAME, 1251, Node1350, NEUTRAL_REACTION);
 end
 
 procedure Node1300
 begin
-    gsay_reply(1341, 1300);
+    Reply(1300);
     set_local_var(11, 1);
-    giq_option(4, 1341, 1301, Node1400, 50);
-    giq_option(4, 1341, 1302, Node999, 50);
+    giq_option(4, NAME, 1301, Node1400, NEUTRAL_REACTION);
+    giq_option(4, NAME, 1302, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node1350
 begin
-    gsay_reply(1341, 1350);
+    Reply(1350);
     set_local_var(11, 1);
-    giq_option(4, 1341, 1351, Node1450, 50);
-    giq_option(4, 1341, 1352, Node999, 50);
+    giq_option(4, NAME, 1351, Node1450, NEUTRAL_REACTION);
+    giq_option(4, NAME, 1352, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node1400
 begin
-    gsay_reply(1341, 1400);
-    giq_option(4, 1341, 1401, Node1550, 50);
-    giq_option(7, 1341, 1402, Node1500, 50);
+    Reply(1400);
+    giq_option(4, NAME, 1401, Node1550, NEUTRAL_REACTION);
+    giq_option(7, NAME, 1402, Node1500, NEUTRAL_REACTION);
 end
 
 procedure Node1450
 begin
-    gsay_reply(1341, 1450);
-    giq_option(4, 1341, 1451, Node1550, 50);
-    giq_option(7, 1341, 1452, Node1500, 50);
+    Reply(1450);
+    giq_option(4, NAME, 1451, Node1550, NEUTRAL_REACTION);
+    giq_option(7, NAME, 1452, Node1500, NEUTRAL_REACTION);
 end
 
 procedure Node1500
 begin
-    gsay_reply(1341, 1500);
-    giq_option(4, 1341, 1501, Node1600, 50);
+    Reply(1500);
+    giq_option(4, NAME, 1501, Node1600, NEUTRAL_REACTION);
 end
 
 procedure Node1550
 begin
-    gsay_reply(1341, 1550);
-    set_global_var(718, 1);
-    giq_option(4, 1341, 1551, Node999, 50);
+    Reply(1550);
+    set_global_var(GVAR_QUEST_S_TYLER_POISON, 1);
+    giq_option(4, NAME, 1551, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node1600
 begin
-    gsay_reply(1341, 1600);
-    set_global_var(718, 1);
-    giq_option(4, 1341, 1601, Node999, 50);
+    Reply(1600);
+    set_global_var(GVAR_QUEST_S_TYLER_POISON, 1);
+    giq_option(4, NAME, 1601, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node1650
 begin
-    gsay_reply(1341, 1650);
+    Reply(1650);
     if (obj_carrying_pid_obj(dude_obj, 538)) then begin
-        giq_option(4, 1341, 1652, Node1700, 50);
+        giq_option(4, NAME, 1652, Node1700, NEUTRAL_REACTION);
     end
     else begin
-        giq_option(4, 1341, 1651, Node999, 50);
+        giq_option(4, NAME, 1651, Node999, NEUTRAL_REACTION);
     end
-    giq_option(-3, 1341, 1653, Node999, 50);
+    giq_option(-3, NAME, 1653, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node1700
 begin
-    gsay_reply(1341, message_str(1341, 1700) + 500 + message_str(1341, 1701));
+    Reply(mstr(1700) + 500 + mstr(1701));
     removed_qty := obj_is_carrying_obj_pid(dude_obj, 538);
     if (1 < removed_qty) then begin
         removed_qty := 1;
@@ -977,17 +977,17 @@ begin
         restock_obj := obj_carrying_pid_obj(dude_obj, 538);
         if (obj_type(self_obj) == 1) then begin
             restock_obj := obj_carrying_pid_obj(dude_obj, 538);
-            if (critter_inven_obj(self_obj, 0) == restock_obj) then begin
-                if (obj_item_subtype(critter_inven_obj(dude_obj, 0)) == 0) then begin
-                    restock_obj := critter_inven_obj(dude_obj, 0);
+            if (critter_inven_obj(self_obj, INVEN_TYPE_WORN) == restock_obj) then begin
+                if (obj_item_subtype(critter_inven_obj(dude_obj, INVEN_TYPE_WORN)) == 0) then begin
+                    restock_obj := critter_inven_obj(dude_obj, INVEN_TYPE_WORN);
                     debug_msg("armour pid == " + obj_pid(restock_obj));
                     rm_obj_from_inven(dude_obj, restock_obj);
                     add_obj_to_inven(dude_obj, restock_obj);
                 end
             end
             else begin
-                if ((critter_inven_obj(dude_obj, 2) == restock_obj) or (critter_inven_obj(dude_obj, 1) == restock_obj)) then begin
-                    metarule(43, dude_obj);
+                if ((critter_inven_obj(dude_obj, INVEN_TYPE_LEFT_HAND) == restock_obj) or (critter_inven_obj(dude_obj, INVEN_TYPE_RIGHT_HAND) == restock_obj)) then begin
+                    inven_unwield(dude_obj);
                 end
             end
         end
@@ -1008,51 +1008,51 @@ begin
     restock_obj := 0;
     restock_amt := 0;
     item_caps_adjust(dude_obj, 500);
-    set_global_var(718, 2);
+    set_global_var(GVAR_QUEST_S_TYLER_POISON, 2);
     set_local_var(12, 1);
-    set_global_var(0, global_var(0) - 5);
-    set_global_var(37, 0);
-    set_global_var(38, 0);
-    set_global_var(39, 0);
-    set_global_var(40, 0);
-    set_global_var(41, 0);
-    set_global_var(42, 0);
-    set_global_var(43, 0);
-    set_global_var(44, 0);
-    set_global_var(45, 0);
-    if (global_var(0) >= 1000) then begin
-        set_global_var(37, 1);
+    set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) - 5);
+    set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+    set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+    set_global_var(GVAR_KARMA_DEFENDER, 0);
+    set_global_var(GVAR_KARMA_WANDERER, 0);
+    set_global_var(GVAR_KARMA_BETRAYER, 0);
+    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+    set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+    set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+    if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
     end
     else begin
-        if (global_var(0) >= 750) then begin
-            set_global_var(38, 1);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+            set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
         end
         else begin
-            if (global_var(0) >= 500) then begin
-                set_global_var(39, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
             end
             else begin
-                if (global_var(0) >= 250) then begin
-                    set_global_var(40, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                    set_global_var(GVAR_KARMA_DEFENDER, 1);
                 end
                 else begin
-                    if (global_var(0) > -250) then begin
-                        set_global_var(41, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                        set_global_var(GVAR_KARMA_WANDERER, 1);
                     end
                     else begin
-                        if (global_var(0) > -500) then begin
-                            set_global_var(42, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                            set_global_var(GVAR_KARMA_BETRAYER, 1);
                         end
                         else begin
-                            if (global_var(0) > -750) then begin
-                                set_global_var(43, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                             end
                             else begin
-                                if (global_var(0) > -1000) then begin
-                                    set_global_var(44, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                    set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                 end
                                 else begin
-                                    set_global_var(45, 1);
+                                    set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                 end
                             end
                         end
@@ -1062,80 +1062,80 @@ begin
         end
     end
     give_exp_points(500);
-    display_msg(message_str(14, 100) + 500 + message_str(14, 101));
-    giq_option(4, 1341, 1702, Node1750, 50);
-    giq_option(4, 1341, 1703, Node999, 50);
+    display_msg(g_mstr(100) + 500 + g_mstr(101));
+    giq_option(4, NAME, 1702, Node1750, NEUTRAL_REACTION);
+    giq_option(4, NAME, 1703, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node1750
 begin
-    gsay_reply(1341, 1750);
-    giq_option(4, 1341, 1751, Node1800, 50);
+    Reply(1750);
+    giq_option(4, NAME, 1751, Node1800, NEUTRAL_REACTION);
 end
 
 procedure Node1800
 begin
-    gsay_reply(1341, 1800 + (get_critter_stat(dude_obj, 34) == 1));
-    set_global_var(719, 1);
-    giq_option(4, 1341, 1802, Node999, 50);
+    Reply(1800 + (dude_gender == 1));
+    set_global_var(GVAR_QUEST_S_TYLER_NICOL, 1);
+    giq_option(4, NAME, 1802, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node1850
 begin
-    gsay_reply(1341, 1850);
-    giq_option(4, 1341, 1851, Node999, 50);
-    giq_option(-3, 1341, 1852, Node999, 50);
+    Reply(1850);
+    giq_option(4, NAME, 1851, Node999, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 1852, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node1900
 begin
-    gsay_reply(1341, message_str(1341, 1900) + 2000 + message_str(1341, 1901));
+    Reply(mstr(1900) + 2000 + mstr(1901));
     item_caps_adjust(dude_obj, 2000);
-    set_global_var(719, 3);
+    set_global_var(GVAR_QUEST_S_TYLER_NICOL, 3);
     set_local_var(13, 1);
-    set_global_var(0, global_var(0) - 10);
-    set_global_var(37, 0);
-    set_global_var(38, 0);
-    set_global_var(39, 0);
-    set_global_var(40, 0);
-    set_global_var(41, 0);
-    set_global_var(42, 0);
-    set_global_var(43, 0);
-    set_global_var(44, 0);
-    set_global_var(45, 0);
-    if (global_var(0) >= 1000) then begin
-        set_global_var(37, 1);
+    set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) - 10);
+    set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+    set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+    set_global_var(GVAR_KARMA_DEFENDER, 0);
+    set_global_var(GVAR_KARMA_WANDERER, 0);
+    set_global_var(GVAR_KARMA_BETRAYER, 0);
+    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+    set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+    set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+    if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
     end
     else begin
-        if (global_var(0) >= 750) then begin
-            set_global_var(38, 1);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+            set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
         end
         else begin
-            if (global_var(0) >= 500) then begin
-                set_global_var(39, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
             end
             else begin
-                if (global_var(0) >= 250) then begin
-                    set_global_var(40, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                    set_global_var(GVAR_KARMA_DEFENDER, 1);
                 end
                 else begin
-                    if (global_var(0) > -250) then begin
-                        set_global_var(41, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                        set_global_var(GVAR_KARMA_WANDERER, 1);
                     end
                     else begin
-                        if (global_var(0) > -500) then begin
-                            set_global_var(42, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                            set_global_var(GVAR_KARMA_BETRAYER, 1);
                         end
                         else begin
-                            if (global_var(0) > -750) then begin
-                                set_global_var(43, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                             end
                             else begin
-                                if (global_var(0) > -1000) then begin
-                                    set_global_var(44, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                    set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                 end
                                 else begin
-                                    set_global_var(45, 1);
+                                    set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                 end
                             end
                         end
@@ -1145,100 +1145,100 @@ begin
         end
     end
     give_exp_points(2000);
-    display_msg(message_str(14, 100) + 2000 + message_str(14, 101));
-    if (global_var(732) == 3) then begin
-        giq_option(4, 1341, 1902, Node1970, 50);
+    display_msg(g_mstr(100) + 2000 + g_mstr(101));
+    if (global_var(GVAR_QUEST_S_KEVIN_MEETING) == 3) then begin
+        giq_option(4, NAME, 1902, Node1970, NEUTRAL_REACTION);
     end
     else begin
-        giq_option(4, 1341, 1902, Node1950, 50);
+        giq_option(4, NAME, 1902, Node1950, NEUTRAL_REACTION);
     end
-    giq_option(4, 1341, 1903, Node999, 50);
+    giq_option(4, NAME, 1903, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node1950
 begin
-    gsay_reply(1341, 1950);
-    giq_option(4, 1341, 1951, Node2050, 50);
-    giq_option(7, 1341, 1952, Node2000, 50);
+    Reply(1950);
+    giq_option(4, NAME, 1951, Node2050, NEUTRAL_REACTION);
+    giq_option(7, NAME, 1952, Node2000, NEUTRAL_REACTION);
 end
 
 procedure Node1970
 begin
-    gsay_reply(1341, 1970);
-    giq_option(4, 1341, 1971, Node2050, 50);
-    giq_option(7, 1341, 1972, Node2000, 50);
+    Reply(1970);
+    giq_option(4, NAME, 1971, Node2050, NEUTRAL_REACTION);
+    giq_option(7, NAME, 1972, Node2000, NEUTRAL_REACTION);
 end
 
 procedure Node2000
 begin
-    gsay_reply(1341, 2000);
-    giq_option(4, 1341, 2001, Node2050, 50);
+    Reply(2000);
+    giq_option(4, NAME, 2001, Node2050, NEUTRAL_REACTION);
 end
 
 procedure Node2050
 begin
-    gsay_reply(1341, 2050);
-    set_global_var(720, 1);
-    restock_obj := create_object_sid(537, 0, 0, -1);
+    Reply(2050);
+    set_global_var(GVAR_QUEST_S_TYLER_SHERIFF, 1);
+    restock_obj := create_object(537, 0, 0);
     add_mult_objs_to_inven(dude_obj, restock_obj, 1);
     add_timer_event(Carpenter_Bruce, 1, 13);
-    if ((party_member_obj(16777811) != 0) and ((global_var(910) bwand 64) == 0)) then begin
-        set_global_var(1018, global_var(1018) bwor 16384);
+    if ((party_member_obj(16777811) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 64) == 0)) then begin
+        set_global_var(GVAR_PTR_GABRIEL_REACTIONS, global_var(GVAR_PTR_GABRIEL_REACTIONS) bwor 16384);
     end
-    giq_option(4, 1341, 2051, Node999, 50);
+    giq_option(4, NAME, 2051, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node2100
 begin
-    gsay_reply(1341, message_str(1341, 2100) + 4000 + message_str(1341, 2101));
+    Reply(mstr(2100) + 4000 + mstr(2101));
     item_caps_adjust(dude_obj, 4000);
-    set_global_var(720, 3);
-    set_global_var(1018, global_var(1018) bwand (-1 - 16384));
+    set_global_var(GVAR_QUEST_S_TYLER_SHERIFF, 3);
+    set_global_var(GVAR_PTR_GABRIEL_REACTIONS, global_var(GVAR_PTR_GABRIEL_REACTIONS) bwand (-1 - 16384));
     set_local_var(14, game_time);
     destroy_object(Carpenter_Bruce);
-    set_global_var(0, global_var(0) - 20);
-    set_global_var(37, 0);
-    set_global_var(38, 0);
-    set_global_var(39, 0);
-    set_global_var(40, 0);
-    set_global_var(41, 0);
-    set_global_var(42, 0);
-    set_global_var(43, 0);
-    set_global_var(44, 0);
-    set_global_var(45, 0);
-    if (global_var(0) >= 1000) then begin
-        set_global_var(37, 1);
+    set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) - 20);
+    set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+    set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+    set_global_var(GVAR_KARMA_DEFENDER, 0);
+    set_global_var(GVAR_KARMA_WANDERER, 0);
+    set_global_var(GVAR_KARMA_BETRAYER, 0);
+    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+    set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+    set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+    if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
     end
     else begin
-        if (global_var(0) >= 750) then begin
-            set_global_var(38, 1);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+            set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
         end
         else begin
-            if (global_var(0) >= 500) then begin
-                set_global_var(39, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
             end
             else begin
-                if (global_var(0) >= 250) then begin
-                    set_global_var(40, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                    set_global_var(GVAR_KARMA_DEFENDER, 1);
                 end
                 else begin
-                    if (global_var(0) > -250) then begin
-                        set_global_var(41, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                        set_global_var(GVAR_KARMA_WANDERER, 1);
                     end
                     else begin
-                        if (global_var(0) > -500) then begin
-                            set_global_var(42, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                            set_global_var(GVAR_KARMA_BETRAYER, 1);
                         end
                         else begin
-                            if (global_var(0) > -750) then begin
-                                set_global_var(43, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                             end
                             else begin
-                                if (global_var(0) > -1000) then begin
-                                    set_global_var(44, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                    set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                 end
                                 else begin
-                                    set_global_var(45, 1);
+                                    set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                 end
                             end
                         end
@@ -1247,63 +1247,63 @@ begin
             end
         end
     end
-    set_global_var(48, global_var(48) - 15);
+    set_global_var(GVAR_TOWN_REP_SEDIT, global_var(GVAR_TOWN_REP_SEDIT) - 15);
     give_exp_points(3000);
-    display_msg(message_str(14, 100) + 3000 + message_str(14, 101));
-    giq_option(4, 1341, 2102, Node999, 50);
+    display_msg(g_mstr(100) + 3000 + g_mstr(101));
+    giq_option(4, NAME, 2102, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node2150
 begin
-    gsay_reply(1341, message_str(1341, 2150) + 4000 + message_str(1341, 2151));
+    Reply(mstr(2150) + 4000 + mstr(2151));
     item_caps_adjust(dude_obj, 4000);
-    set_global_var(720, 3);
-    set_global_var(1018, global_var(1018) bwand (-1 - 16384));
+    set_global_var(GVAR_QUEST_S_TYLER_SHERIFF, 3);
+    set_global_var(GVAR_PTR_GABRIEL_REACTIONS, global_var(GVAR_PTR_GABRIEL_REACTIONS) bwand (-1 - 16384));
     set_local_var(14, game_time);
     destroy_object(Carpenter_Bruce);
-    set_global_var(0, global_var(0) - 20);
-    set_global_var(37, 0);
-    set_global_var(38, 0);
-    set_global_var(39, 0);
-    set_global_var(40, 0);
-    set_global_var(41, 0);
-    set_global_var(42, 0);
-    set_global_var(43, 0);
-    set_global_var(44, 0);
-    set_global_var(45, 0);
-    if (global_var(0) >= 1000) then begin
-        set_global_var(37, 1);
+    set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) - 20);
+    set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+    set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+    set_global_var(GVAR_KARMA_DEFENDER, 0);
+    set_global_var(GVAR_KARMA_WANDERER, 0);
+    set_global_var(GVAR_KARMA_BETRAYER, 0);
+    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+    set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+    set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+    if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
     end
     else begin
-        if (global_var(0) >= 750) then begin
-            set_global_var(38, 1);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+            set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
         end
         else begin
-            if (global_var(0) >= 500) then begin
-                set_global_var(39, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
             end
             else begin
-                if (global_var(0) >= 250) then begin
-                    set_global_var(40, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                    set_global_var(GVAR_KARMA_DEFENDER, 1);
                 end
                 else begin
-                    if (global_var(0) > -250) then begin
-                        set_global_var(41, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                        set_global_var(GVAR_KARMA_WANDERER, 1);
                     end
                     else begin
-                        if (global_var(0) > -500) then begin
-                            set_global_var(42, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                            set_global_var(GVAR_KARMA_BETRAYER, 1);
                         end
                         else begin
-                            if (global_var(0) > -750) then begin
-                                set_global_var(43, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                             end
                             else begin
-                                if (global_var(0) > -1000) then begin
-                                    set_global_var(44, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                    set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                 end
                                 else begin
-                                    set_global_var(45, 1);
+                                    set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                 end
                             end
                         end
@@ -1312,62 +1312,62 @@ begin
             end
         end
     end
-    set_global_var(48, global_var(48) - 15);
+    set_global_var(GVAR_TOWN_REP_SEDIT, global_var(GVAR_TOWN_REP_SEDIT) - 15);
     give_exp_points(3000);
-    display_msg(message_str(14, 100) + 3000 + message_str(14, 101));
-    giq_option(4, 1341, 2152, Node999, 50);
+    display_msg(g_mstr(100) + 3000 + g_mstr(101));
+    giq_option(4, NAME, 2152, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node2200
 begin
-    gsay_reply(1341, 2200);
-    if ((global_var(712) == 1) and (global_var(793) == 0) and (local_var(9) != 1)) then begin
-        giq_option(4, 1341, 2201, Node2300, 50);
+    Reply(2200);
+    if ((global_var(GVAR_QUEST_S_RODRIGUES_GHOULS) == 1) and (global_var(GVAR_SEDIT_DEAD_GHOULS) == 0) and (local_var(9) != 1)) then begin
+        giq_option(4, NAME, 2201, Node2300, NEUTRAL_REACTION);
     end
-    giq_option(4, 1341, 2202, Node999, 50);
-    giq_option(-3, 1341, 2203, Node999, 50);
+    giq_option(4, NAME, 2202, Node999, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 2203, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node2250
 begin
-    gsay_reply(1341, 2250);
+    Reply(2250);
     set_local_var(8, 1);
-    giq_option(4, 1341, 2251, Node999, 50);
+    giq_option(4, NAME, 2251, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node2270
 begin
-    gsay_reply(1341, 2270 + (get_critter_stat(dude_obj, 34) == 1));
+    Reply(2270 + (dude_gender == 1));
     set_local_var(8, 1);
-    giq_option(4, 1341, 2272, Node999, 50);
+    giq_option(4, NAME, 2272, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node2300
 begin
-    gsay_reply(1341, 2300);
+    Reply(2300);
     set_local_var(9, 1);
-    set_global_var(721, 1);
-    giq_option(4, 1341, 2301, Node999, 50);
+    set_global_var(GVAR_FALCON_POLICEMEN_OFF, 1);
+    giq_option(4, NAME, 2301, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node2350
 begin
-    gsay_reply(1341, 2350);
+    Reply(2350);
     set_local_var(8, 1);
-    giq_option(4, 1341, 2351, Node999, 50);
+    giq_option(4, NAME, 2351, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node2400
 begin
-    gsay_reply(1341, 2400);
-    giq_option(-3, 1341, 2401, Node999, 50);
+    Reply(2400);
+    giq_option(-3, NAME, 2401, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node2450
 begin
-    gsay_reply(1341, 2450);
-    giq_option(4, 1341, 2451, Node999, 50);
-    giq_option(-3, 1341, 2452, Node999, 50);
+    Reply(2450);
+    giq_option(4, NAME, 2451, Node999, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 2452, Node999, NEUTRAL_REACTION);
 end
 
 procedure Dude_arrest
@@ -1375,16 +1375,16 @@ begin
     game_ui_disable;
     gfade_out(5);
     set_local_var(10, 1);
-    tmp_locker := create_object_sid(128, 9629, 0, -1);
+    tmp_locker := create_object(128, 9629, 0);
     move_obj_inven_to_obj(dude_obj, tmp_locker);
     move_to(dude_obj, 13902, 0);
     anim(dude_obj, 1000, 3);
-    set_global_var(632, global_var(632) bwor 64);
+    set_global_var(GVAR_PARTY_MEMBERS_HIDDEN, global_var(GVAR_PARTY_MEMBERS_HIDDEN) bwor 64);
     if (party_member_obj(16777729) != 0) then begin
         if (party_member_obj(16777729)) then begin
-            if (has_trait(1, party_member_obj(16777729), 666)) then begin
+            if (obj_is_visible_flag(party_member_obj(16777729))) then begin
                 debug_msg("set party member hidden: " + obj_name(party_member_obj(16777729)));
-                set_global_var(632, global_var(632) bwor 1);
+                set_global_var(GVAR_PARTY_MEMBERS_HIDDEN, global_var(GVAR_PARTY_MEMBERS_HIDDEN) bwor 1);
                 set_obj_visibility(party_member_obj(16777729), 1);
             end
             else begin
@@ -1392,11 +1392,11 @@ begin
             end
         end
     end
-    if ((party_member_obj(16777779) != 0) and ((global_var(910) bwand 32) == 0)) then begin
+    if ((party_member_obj(16777779) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 32) == 0)) then begin
         if (party_member_obj(16777779)) then begin
-            if (has_trait(1, party_member_obj(16777779), 666)) then begin
+            if (obj_is_visible_flag(party_member_obj(16777779))) then begin
                 debug_msg("set party member hidden: " + obj_name(party_member_obj(16777779)));
-                set_global_var(632, global_var(632) bwor 2);
+                set_global_var(GVAR_PARTY_MEMBERS_HIDDEN, global_var(GVAR_PARTY_MEMBERS_HIDDEN) bwor 2);
                 set_obj_visibility(party_member_obj(16777779), 1);
             end
             else begin
@@ -1404,11 +1404,11 @@ begin
             end
         end
     end
-    if ((party_member_obj(16777811) != 0) and ((global_var(910) bwand 64) == 0)) then begin
+    if ((party_member_obj(16777811) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 64) == 0)) then begin
         if (party_member_obj(16777811)) then begin
-            if (has_trait(1, party_member_obj(16777811), 666)) then begin
+            if (obj_is_visible_flag(party_member_obj(16777811))) then begin
                 debug_msg("set party member hidden: " + obj_name(party_member_obj(16777811)));
-                set_global_var(632, global_var(632) bwor 4);
+                set_global_var(GVAR_PARTY_MEMBERS_HIDDEN, global_var(GVAR_PARTY_MEMBERS_HIDDEN) bwor 4);
                 set_obj_visibility(party_member_obj(16777811), 1);
             end
             else begin
@@ -1418,9 +1418,9 @@ begin
     end
     if (party_member_obj(16777812) != 0) then begin
         if (party_member_obj(16777812)) then begin
-            if (has_trait(1, party_member_obj(16777812), 666)) then begin
+            if (obj_is_visible_flag(party_member_obj(16777812))) then begin
                 debug_msg("set party member hidden: " + obj_name(party_member_obj(16777812)));
-                set_global_var(632, global_var(632) bwor 8);
+                set_global_var(GVAR_PARTY_MEMBERS_HIDDEN, global_var(GVAR_PARTY_MEMBERS_HIDDEN) bwor 8);
                 set_obj_visibility(party_member_obj(16777812), 1);
             end
             else begin
@@ -1430,9 +1430,9 @@ begin
     end
     if (party_member_obj(16777813) != 0) then begin
         if (party_member_obj(16777813)) then begin
-            if (has_trait(1, party_member_obj(16777813), 666)) then begin
+            if (obj_is_visible_flag(party_member_obj(16777813))) then begin
                 debug_msg("set party member hidden: " + obj_name(party_member_obj(16777813)));
-                set_global_var(632, global_var(632) bwor 16);
+                set_global_var(GVAR_PARTY_MEMBERS_HIDDEN, global_var(GVAR_PARTY_MEMBERS_HIDDEN) bwor 16);
                 set_obj_visibility(party_member_obj(16777813), 1);
             end
             else begin
@@ -1440,11 +1440,11 @@ begin
             end
         end
     end
-    if ((party_member_obj(16777814) != 0) and ((global_var(910) bwand 128) == 0)) then begin
+    if ((party_member_obj(16777814) != 0) and ((global_var(GVAR_PTR_PARTY_MEMBER_ACTIONS) bwand 128) == 0)) then begin
         if (party_member_obj(16777814)) then begin
-            if (has_trait(1, party_member_obj(16777814), 666)) then begin
+            if (obj_is_visible_flag(party_member_obj(16777814))) then begin
                 debug_msg("set party member hidden: " + obj_name(party_member_obj(16777814)));
-                set_global_var(632, global_var(632) bwor 32);
+                set_global_var(GVAR_PARTY_MEMBERS_HIDDEN, global_var(GVAR_PARTY_MEMBERS_HIDDEN) bwor 32);
                 set_obj_visibility(party_member_obj(16777814), 1);
             end
             else begin
@@ -1453,10 +1453,9 @@ begin
         end
     end
     game_time_advance(game_ticks(5 * 60));
-    if (get_critter_stat(dude_obj, 35) > 10) then begin
-        critter_dmg(dude_obj, get_critter_stat(dude_obj, 35) - 10, 0);
+    if (get_critter_stat(dude_obj, STAT_current_hp) > 10) then begin
+        critter_damage(dude_obj, get_critter_stat(dude_obj, STAT_current_hp) - 10);
     end
     add_timer_event(prison_door1, 0, 10);
     add_timer_event(self_obj, game_ticks(3), 1);
 end
-
