diff --git a/Games/Fallout2.Steam/Original/AIBROC.ssl b/Games/Fallout2.Steam/ReDefine/AIBROC.ssl
index c0553ac..fe3a3ca 100644
--- a/Games/Fallout2.Steam/Original/AIBROC.ssl
+++ b/Games/Fallout2.Steam/ReDefine/AIBROC.ssl
@@ -138,22 +138,21 @@ end
 
 procedure pickup_p_proc
 begin
-    if (cur_map_index == 35) then begin
+    if (CUR_MAP_ARGARDEN) then begin
         set_map_var(3, map_var(4) - 1);
     end
 end
 
 procedure destroy_p_proc
 begin
-    if (cur_map_index == 35) then begin
+    if (CUR_MAP_ARGARDEN) then begin
         set_map_var(3, map_var(4) - 1);
     end
 end
 
 procedure drop_p_proc
 begin
-    if (cur_map_index == 35) then begin
+    if (CUR_MAP_ARGARDEN) then begin
         set_map_var(3, map_var(4) + 1);
     end
 end
-
