diff --git a/Games/Fallout2.Steam/Original/AICHEST.ssl b/Games/Fallout2.Steam/ReDefine/AICHEST.ssl
index e9d8616..5f7239a 100644
--- a/Games/Fallout2.Steam/Original/AICHEST.ssl
+++ b/Games/Fallout2.Steam/ReDefine/AICHEST.ssl
@@ -142,13 +142,13 @@ end
 procedure look_at_p_proc
 begin
     script_overrides;
-    display_msg(message_str(24, 101));
+    display_mstr(101);
 end
 
 procedure description_p_proc
 begin
     script_overrides;
-    display_msg(message_str(24, 101));
+    display_mstr(101);
 end
 
 procedure damage_p_proc
@@ -160,7 +160,7 @@ end
 procedure pickup_p_proc
 begin
     if (is_success(roll_vs_skill(dude_obj, 10, -20)) and (map_var(0) < 6)) then begin
-        display_msg(message_str(24, 104));
+        display_mstr(104);
     end
     else begin
         script_overrides;
@@ -171,7 +171,7 @@ end
 procedure use_p_proc
 begin
     if (is_success(roll_vs_skill(dude_obj, 10, -20)) and (map_var(0) < 6)) then begin
-        display_msg(message_str(24, 104));
+        display_mstr(104);
     end
     else begin
         script_overrides;
@@ -182,11 +182,10 @@ end
 procedure use_skill_on_p_proc
 begin
     if (is_success(roll_vs_skill(dude_obj, 10, -20)) and (map_var(0) < 6)) then begin
-        display_msg(message_str(24, 104));
+        display_mstr(104);
     end
     else begin
         script_overrides;
         set_map_var(0, map_var(0) + 1);
     end
 end
-
