diff --git a/Games/Fallout2.Steam/Original/AIXANDER.ssl b/Games/Fallout2.Steam/ReDefine/AIXANDER.ssl
index 6856bf3..aa5da9d 100644
--- a/Games/Fallout2.Steam/Original/AIXANDER.ssl
+++ b/Games/Fallout2.Steam/ReDefine/AIXANDER.ssl
@@ -138,22 +138,21 @@ end
 
 procedure pickup_p_proc
 begin
-    if (cur_map_index == 35) then begin
+    if (CUR_MAP_ARGARDEN) then begin
         set_map_var(3, map_var(3) - 1);
     end
 end
 
 procedure destroy_p_proc
 begin
-    if (cur_map_index == 35) then begin
+    if (CUR_MAP_ARGARDEN) then begin
         set_map_var(3, map_var(3) - 1);
     end
 end
 
 procedure drop_p_proc
 begin
-    if (cur_map_index == 35) then begin
+    if (CUR_MAP_ARGARDEN) then begin
         set_map_var(3, map_var(3) + 1);
     end
 end
-
