diff --git a/Games/Fallout2.Steam/Original/ARCAVES.ssl b/Games/Fallout2.Steam/ReDefine/ARCAVES.ssl
index cd3e799..5ff59a6 100644
--- a/Games/Fallout2.Steam/Original/ARCAVES.ssl
+++ b/Games/Fallout2.Steam/ReDefine/ARCAVES.ssl
@@ -139,15 +139,14 @@ end
 
 procedure map_enter_p_proc
 begin
-    if (metarule(14, 0)) then begin
-        display_msg(message_str(25, 100));
+    if (map_first_run) then begin
+        display_mstr(100);
     end
     set_light_level(50);
-    set_global_var(27, 0);
+    set_global_var(GVAR_LOAD_MAP_INDEX, 0);
 end
 
 procedure map_update_p_proc
 begin
     set_light_level(50);
 end
-
