diff --git a/Games/Fallout2.Steam/Original/ARGARDEN.ssl b/Games/Fallout2.Steam/ReDefine/ARGARDEN.ssl
index 08000bc..ce46b0f 100644
--- a/Games/Fallout2.Steam/Original/ARGARDEN.ssl
+++ b/Games/Fallout2.Steam/ReDefine/ARGARDEN.ssl
@@ -224,24 +224,24 @@ begin
         call Make_Geckos();
         call Make_Plants();
         if (not(tile_contains_obj_pid(24298, 0, 272))) then begin
-            LVar0 := create_object_sid(272, 0, 0, -1);
+            LVar0 := create_object(272, 0, 0);
             move_to(LVar0, 24298, 0);
         end
     end
-    if (((game_time - map_var(0)) >= (24 * (60 * (60 * 10)))) and not(metarule(22, 0))) then begin
+    if (((game_time - map_var(0)) >= (24 * (60 * (60 * 10)))) and not(is_loading_game)) then begin
         set_map_var(0, game_time);
         call Make_Geckos();
         call Make_Plants();
         if (not(tile_contains_obj_pid(24298, 0, 272))) then begin
-            LVar0 := create_object_sid(272, 0, 0, -1);
+            LVar0 := create_object(272, 0, 0);
             move_to(LVar0, 24298, 0);
         end
     end
-    if ((global_var(191) == 3) and (map_var(6) == 0)) then begin
+    if ((global_var(GVAR_ARROYO_DOG) == 3) and (map_var(6) == 0)) then begin
         set_map_var(6, 1);
         call Kill_Brat();
     end
-    set_global_var(27, 0);
+    set_global_var(GVAR_LOAD_MAP_INDEX, 0);
 end
 
 procedure map_update_p_proc
@@ -354,7 +354,7 @@ begin
                     end
                 end
             end
-            LVar4 := create_object_sid(16777296, 0, 0, 200);
+            LVar4 := create_object_sid(16777296, 0, 0, SCRIPT_ACGECKO);
             while (LVar5 <= 0) do begin
                 LVar5 := tile_num_in_direction(map_var(5), random(0, 5), random(0, 10));
             end
@@ -387,7 +387,7 @@ begin
                         end
                     end
                 end
-                LVar4 := create_object_sid(16777296, 0, 0, 200);
+                LVar4 := create_object_sid(16777296, 0, 0, SCRIPT_ACGECKO);
                 while (LVar5 <= 0) do begin
                     LVar5 := tile_num_in_direction(map_var(5), random(0, 5), random(0, 10));
                 end
@@ -421,7 +421,7 @@ begin
                     end
                 end
             end
-            LVar4 := create_object_sid(16777297, 0, 0, 200);
+            LVar4 := create_object_sid(16777297, 0, 0, SCRIPT_ACGECKO);
             while (LVar5 <= 0) do begin
                 LVar5 := tile_num_in_direction(map_var(5), random(0, 5), random(0, 10));
             end
@@ -454,7 +454,7 @@ begin
                         end
                     end
                 end
-                LVar4 := create_object_sid(16777297, 0, 0, 200);
+                LVar4 := create_object_sid(16777297, 0, 0, SCRIPT_ACGECKO);
                 while (LVar5 <= 0) do begin
                     LVar5 := tile_num_in_direction(map_var(5), random(0, 5), random(0, 10));
                 end
@@ -496,7 +496,7 @@ begin
                 end
             end
         end
-        LVar4 := create_object_sid(272, 0, 0, -1);
+        LVar4 := create_object(272, 0, 0);
         LVar5 := tile_num_in_direction(map_var(5), random(0, 5), random(0, 10));
         move_to(LVar4, LVar5, 0);
         LVar0 := LVar0 + 1;
@@ -522,7 +522,7 @@ begin
                 end
             end
         end
-        LVar4 := create_object_sid(271, 0, 0, -1);
+        LVar4 := create_object(271, 0, 0);
         LVar5 := tile_num_in_direction(map_var(5), random(0, 5), random(0, 10));
         move_to(LVar4, LVar5, 0);
         LVar2 := LVar2 + 1;
@@ -536,13 +536,13 @@ begin
     variable LVar2 := 0;
     variable LVar3 := 0;
     kill_critter_type(16777225, 0);
-    LVar1 := create_object_sid(16777219, 0, 0, -1);
-    LVar2 := create_object_sid(16777297, 0, 0, -1);
-    LVar3 := create_object_sid(7, 0, 0, -1);
+    LVar1 := create_object(16777219, 0, 0);
+    LVar2 := create_object(16777297, 0, 0);
+    LVar3 := create_object(7, 0, 0);
     critter_attempt_placement(LVar1, 23894, 0);
     critter_attempt_placement(LVar2, 23491, 0);
     if (map_var(7) == 0) then begin
-        LVar0 := create_object_sid(16777225, 0, 0, -1);
+        LVar0 := create_object(16777225, 0, 0);
         critter_attempt_placement(LVar0, 23491, 0);
     end
     anim(LVar2, 1000, rotation_to_tile(tile_num(LVar2), tile_num(LVar1)));
@@ -551,4 +551,3 @@ begin
     kill_critter(LVar1, 62);
     kill_critter(LVar2, 61);
 end
-
