diff --git a/Games/Fallout2.Steam/Original/ATDEATH.ssl b/Games/Fallout2.Steam/ReDefine/ATDEATH.ssl
index 58553d8..e92da19 100644
--- a/Games/Fallout2.Steam/Original/ATDEATH.ssl
+++ b/Games/Fallout2.Steam/ReDefine/ATDEATH.ssl
@@ -136,9 +136,8 @@ end
 
 procedure spatial_p_proc
 begin
-    if ((local_var(0) == 0) and (global_var(191) == 3)) then begin
+    if ((local_var(0) == 0) and (global_var(GVAR_ARROYO_DOG) == 3)) then begin
         set_local_var(0, 1);
-        display_msg(message_str(14, 920));
+        display_msg(g_mstr(920));
     end
 end
-
