diff --git a/Games/Fallout2.Steam/Original/BCBRAHMN.ssl b/Games/Fallout2.Steam/ReDefine/BCBRAHMN.ssl
index 45bbd6f..7e0e800 100644
--- a/Games/Fallout2.Steam/Original/BCBRAHMN.ssl
+++ b/Games/Fallout2.Steam/ReDefine/BCBRAHMN.ssl
@@ -151,24 +151,24 @@ procedure pickup_p_proc
 begin
     if (source_obj == dude_obj) then begin
         if (3 == 1) then begin
-            attack_complex(source_obj, 0, 1, 0, 0, 30000, 0, 0);
+            attack(source_obj);
         end
         else begin
             if (3 == 2) then begin
-                float_msg(self_obj, message_str(14, random(3325, 3329)), 0);
+                float_msg(self_obj, g_mstr(random(3325, 3329)), FLOAT_MSG_NORMAL);
                 if (-1 > -1) then begin
                     if (local_var(-1) >= 1) then begin
-                        attack_complex(source_obj, 0, 1, 0, 0, 30000, 0, 0);
+                        attack(source_obj);
                     end
                     set_local_var(-1, local_var(-1) + 1);
                 end
             end
             else begin
                 if (3 == 3) then begin
-                    float_msg(self_obj, message_str(14, random(3050, 3064)), 0);
+                    float_msg(self_obj, g_mstr(random(3050, 3064)), FLOAT_MSG_NORMAL);
                     if (-1 > -1) then begin
                         if (local_var(-1) >= 1) then begin
-                            attack_complex(source_obj, 0, 1, 0, 0, 30000, 0, 0);
+                            attack(source_obj);
                         end
                         set_local_var(-1, local_var(-1) + 1);
                     end
@@ -283,13 +283,13 @@ end
 procedure look_at_p_proc
 begin
     script_overrides;
-    display_msg(message_str(203, 100));
+    display_msg(message_str(SCRIPT_ZCBRAHMN, 100));
 end
 
 procedure description_p_proc
 begin
     script_overrides;
-    display_msg(message_str(203, 105));
+    display_msg(message_str(SCRIPT_ZCBRAHMN, 105));
 end
 
 procedure use_skill_on_p_proc
@@ -297,10 +297,10 @@ begin
     if (action_being_used == 12) then begin
         script_overrides;
         if (is_success(roll_vs_skill(dude_obj, action_being_used, 0))) then begin
-            display_msg(message_str(203, 101));
+            display_msg(message_str(SCRIPT_ZCBRAHMN, 101));
         end
         else begin
-            display_msg(message_str(203, 102));
+            display_msg(message_str(SCRIPT_ZCBRAHMN, 102));
         end
     end
     else begin
@@ -316,13 +316,13 @@ end
 
 procedure map_enter_p_proc
 begin
-    if (elevation(dude_obj) == 1) then begin
+    if (dude_elevation == 1) then begin
         set_obj_visibility(self_obj, 0);
     end
     else begin
         set_obj_visibility(self_obj, 1);
     end
-    critter_add_trait(self_obj, 1, 6, 113);
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 113);
 end
 
 procedure timed_event_p_proc
@@ -352,4 +352,3 @@ begin
         end
     end
 end
-
