diff --git a/Games/Fallout2.Steam/Original/BCDALIA.ssl b/Games/Fallout2.Steam/ReDefine/BCDALIA.ssl
index 77f26e4..dbef6ad 100644
--- a/Games/Fallout2.Steam/Original/BCDALIA.ssl
+++ b/Games/Fallout2.Steam/ReDefine/BCDALIA.ssl
@@ -185,24 +185,24 @@ procedure pickup_p_proc
 begin
     if (source_obj == dude_obj) then begin
         if (1 == 1) then begin
-            attack_complex(source_obj, 0, 1, 0, 0, 30000, 0, 0);
+            attack(source_obj);
         end
         else begin
             if (1 == 2) then begin
-                float_msg(self_obj, message_str(14, random(3325, 3329)), 0);
+                float_msg(self_obj, g_mstr(random(3325, 3329)), FLOAT_MSG_NORMAL);
                 if (7 > -1) then begin
                     if (local_var(7) >= 1) then begin
-                        attack_complex(source_obj, 0, 1, 0, 0, 30000, 0, 0);
+                        attack(source_obj);
                     end
                     set_local_var(7, local_var(7) + 1);
                 end
             end
             else begin
                 if (1 == 3) then begin
-                    float_msg(self_obj, message_str(14, random(3050, 3064)), 0);
+                    float_msg(self_obj, g_mstr(random(3050, 3064)), FLOAT_MSG_NORMAL);
                     if (7 > -1) then begin
                         if (local_var(7) >= 1) then begin
-                            attack_complex(source_obj, 0, 1, 0, 0, 30000, 0, 0);
+                            attack(source_obj);
                         end
                         set_local_var(7, local_var(7) + 1);
                     end
@@ -310,10 +310,10 @@ begin
             if (anim_busy(self_obj) == 0) then begin
                 if ((tile_distance_objs(self_obj, dude_obj) < 8) or obj_can_see_obj(self_obj, dude_obj)) then begin
                     if (anim_busy(self_obj) == 0) then begin
-                        global_temp := rotation_to_tile(tile_num(dude_obj), tile_num(self_obj));
+                        global_temp := rotation_to_tile(dude_tile, tile_num(self_obj));
                         animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(self_obj), global_temp, random(3, 10)), 1);
                         global_temp := (global_temp + 1) % 6;
-                        while ((anim_busy(self_obj) == 0) and (global_temp != rotation_to_tile(tile_num(dude_obj), tile_num(self_obj)))) do begin
+                        while ((anim_busy(self_obj) == 0) and (global_temp != rotation_to_tile(dude_tile, tile_num(self_obj)))) do begin
                             debug_msg("flee loop: rot == " + global_temp);
                             animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(self_obj), global_temp, random(3, 10)), 1);
                             global_temp := (global_temp + 1) % 6;
@@ -323,7 +323,7 @@ begin
             end
         end
         else begin
-            attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+            attack(dude_obj);
         end
     end
 end
@@ -333,48 +333,48 @@ begin
     Evil_Critter := 0;
     Slavery_Tolerant := 2;
     Karma_Perception := 1;
-    set_global_var(37, 0);
-    set_global_var(38, 0);
-    set_global_var(39, 0);
-    set_global_var(40, 0);
-    set_global_var(41, 0);
-    set_global_var(42, 0);
-    set_global_var(43, 0);
-    set_global_var(44, 0);
-    set_global_var(45, 0);
-    if (global_var(0) >= 1000) then begin
-        set_global_var(37, 1);
+    set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+    set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+    set_global_var(GVAR_KARMA_DEFENDER, 0);
+    set_global_var(GVAR_KARMA_WANDERER, 0);
+    set_global_var(GVAR_KARMA_BETRAYER, 0);
+    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+    set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+    set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+    if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
     end
     else begin
-        if (global_var(0) >= 750) then begin
-            set_global_var(38, 1);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+            set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
         end
         else begin
-            if (global_var(0) >= 500) then begin
-                set_global_var(39, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
             end
             else begin
-                if (global_var(0) >= 250) then begin
-                    set_global_var(40, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                    set_global_var(GVAR_KARMA_DEFENDER, 1);
                 end
                 else begin
-                    if (global_var(0) > -250) then begin
-                        set_global_var(41, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                        set_global_var(GVAR_KARMA_WANDERER, 1);
                     end
                     else begin
-                        if (global_var(0) > -500) then begin
-                            set_global_var(42, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                            set_global_var(GVAR_KARMA_BETRAYER, 1);
                         end
                         else begin
-                            if (global_var(0) > -750) then begin
-                                set_global_var(43, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                             end
                             else begin
-                                if (global_var(0) > -1000) then begin
-                                    set_global_var(44, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                    set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                 end
                                 else begin
-                                    set_global_var(45, 1);
+                                    set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                 end
                             end
                         end
@@ -385,7 +385,7 @@ begin
     end
     if (local_var(2) == 0) then begin
         set_local_var(2, 1);
-        set_local_var(3, (get_critter_stat(dude_obj, 3) - get_critter_stat(self_obj, 3)) * 5);
+        set_local_var(3, (dude_charisma - self_charisma) * 5);
         debug_msg("Base Reaction == " + local_var(3));
     end
     Static_Reaction := 0;
@@ -397,14 +397,14 @@ begin
     else begin
         reaction_bonus_karma := -1 * (-1 * reaction_bonus_karma / 1000);
     end
-    if (global_var(294) >= 0) then begin
-        reaction_bonus_town_rep := global_var(294) / 2;
+    if (global_var(GVAR_TOWN_REP_VAULT_15) >= 0) then begin
+        reaction_bonus_town_rep := global_var(GVAR_TOWN_REP_VAULT_15) / 2;
     end
     else begin
-        reaction_bonus_town_rep := -1 * (-1 * global_var(294) / 2);
+        reaction_bonus_town_rep := -1 * (-1 * global_var(GVAR_TOWN_REP_VAULT_15) / 2);
     end
-    if (has_trait(2, dude_obj, 13)) then begin
-        if (get_critter_stat(dude_obj, 34) == get_critter_stat(self_obj, 34)) then begin
+    if (has_trait(TRAIT_TRAIT, dude_obj, TRAIT_sex_appeal)) then begin
+        if (dude_gender == self_gender) then begin
             Static_Reaction := Static_Reaction + -20;
         end
         else begin
@@ -420,48 +420,48 @@ begin
         debug_msg("Karma Reaction Bonus == " + Static_Reaction);
         Static_Reaction := Static_Reaction - reaction_bonus_town_rep;
         debug_msg("Town Rep Bonus == " + Static_Reaction);
-        set_global_var(37, 0);
-        set_global_var(38, 0);
-        set_global_var(39, 0);
-        set_global_var(40, 0);
-        set_global_var(41, 0);
-        set_global_var(42, 0);
-        set_global_var(43, 0);
-        set_global_var(44, 0);
-        set_global_var(45, 0);
-        if (global_var(0) >= 1000) then begin
-            set_global_var(37, 1);
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+        set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+        set_global_var(GVAR_KARMA_DEFENDER, 0);
+        set_global_var(GVAR_KARMA_WANDERER, 0);
+        set_global_var(GVAR_KARMA_BETRAYER, 0);
+        set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+        set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+            set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
         end
         else begin
-            if (global_var(0) >= 750) then begin
-                set_global_var(38, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
             end
             else begin
-                if (global_var(0) >= 500) then begin
-                    set_global_var(39, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                 end
                 else begin
-                    if (global_var(0) >= 250) then begin
-                        set_global_var(40, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                        set_global_var(GVAR_KARMA_DEFENDER, 1);
                     end
                     else begin
-                        if (global_var(0) > -250) then begin
-                            set_global_var(41, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                            set_global_var(GVAR_KARMA_WANDERER, 1);
                         end
                         else begin
-                            if (global_var(0) > -500) then begin
-                                set_global_var(42, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                set_global_var(GVAR_KARMA_BETRAYER, 1);
                             end
                             else begin
-                                if (global_var(0) > -750) then begin
-                                    set_global_var(43, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -1000) then begin
-                                        set_global_var(44, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                     end
                                     else begin
-                                        set_global_var(45, 1);
+                                        set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                     end
                                 end
                             end
@@ -470,35 +470,35 @@ begin
                 end
             end
         end
-        if (global_var(37) == 1) then begin
+        if (global_var(GVAR_KARMA_HOLY_WARRIOR) == 1) then begin
             Static_Reaction := Static_Reaction - 20;
         end
         else begin
-            if (global_var(38) == 1) then begin
+            if (global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES) == 1) then begin
                 Static_Reaction := Static_Reaction - 15;
             end
             else begin
-                if (global_var(39) == 1) then begin
+                if (global_var(GVAR_KARMA_SHIELD_OF_HOPE) == 1) then begin
                     Static_Reaction := Static_Reaction - 10;
                 end
                 else begin
-                    if (global_var(40) == 1) then begin
+                    if (global_var(GVAR_KARMA_DEFENDER) == 1) then begin
                         Static_Reaction := Static_Reaction - 5;
                     end
                     else begin
-                        if (global_var(41) == 1) then begin
+                        if (global_var(GVAR_KARMA_WANDERER) == 1) then begin
                             Static_Reaction := Static_Reaction - 0;
                         end
                         else begin
-                            if (global_var(42) == 1) then begin
+                            if (global_var(GVAR_KARMA_BETRAYER) == 1) then begin
                                 Static_Reaction := Static_Reaction - -5;
                             end
                             else begin
-                                if (global_var(43) == 1) then begin
+                                if (global_var(GVAR_KARMA_SWORD_OF_DESPAIR) == 1) then begin
                                     Static_Reaction := Static_Reaction - -10;
                                 end
                                 else begin
-                                    if (global_var(44) == 1) then begin
+                                    if (global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE) == 1) then begin
                                         Static_Reaction := Static_Reaction - -15;
                                     end
                                     else begin
@@ -518,48 +518,48 @@ begin
         debug_msg("Karma Reaction Bonus == " + Static_Reaction);
         Static_Reaction := Static_Reaction + reaction_bonus_town_rep;
         debug_msg("Town Rep Bonus == " + Static_Reaction);
-        set_global_var(37, 0);
-        set_global_var(38, 0);
-        set_global_var(39, 0);
-        set_global_var(40, 0);
-        set_global_var(41, 0);
-        set_global_var(42, 0);
-        set_global_var(43, 0);
-        set_global_var(44, 0);
-        set_global_var(45, 0);
-        if (global_var(0) >= 1000) then begin
-            set_global_var(37, 1);
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+        set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+        set_global_var(GVAR_KARMA_DEFENDER, 0);
+        set_global_var(GVAR_KARMA_WANDERER, 0);
+        set_global_var(GVAR_KARMA_BETRAYER, 0);
+        set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+        set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+            set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
         end
         else begin
-            if (global_var(0) >= 750) then begin
-                set_global_var(38, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
             end
             else begin
-                if (global_var(0) >= 500) then begin
-                    set_global_var(39, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                 end
                 else begin
-                    if (global_var(0) >= 250) then begin
-                        set_global_var(40, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                        set_global_var(GVAR_KARMA_DEFENDER, 1);
                     end
                     else begin
-                        if (global_var(0) > -250) then begin
-                            set_global_var(41, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                            set_global_var(GVAR_KARMA_WANDERER, 1);
                         end
                         else begin
-                            if (global_var(0) > -500) then begin
-                                set_global_var(42, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                set_global_var(GVAR_KARMA_BETRAYER, 1);
                             end
                             else begin
-                                if (global_var(0) > -750) then begin
-                                    set_global_var(43, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -1000) then begin
-                                        set_global_var(44, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                     end
                                     else begin
-                                        set_global_var(45, 1);
+                                        set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                     end
                                 end
                             end
@@ -568,35 +568,35 @@ begin
                 end
             end
         end
-        if (global_var(37) == 1) then begin
+        if (global_var(GVAR_KARMA_HOLY_WARRIOR) == 1) then begin
             Static_Reaction := Static_Reaction + 20;
         end
         else begin
-            if (global_var(38) == 1) then begin
+            if (global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES) == 1) then begin
                 Static_Reaction := Static_Reaction + 15;
             end
             else begin
-                if (global_var(39) == 1) then begin
+                if (global_var(GVAR_KARMA_SHIELD_OF_HOPE) == 1) then begin
                     Static_Reaction := Static_Reaction + 10;
                 end
                 else begin
-                    if (global_var(40) == 1) then begin
+                    if (global_var(GVAR_KARMA_DEFENDER) == 1) then begin
                         Static_Reaction := Static_Reaction + 5;
                     end
                     else begin
-                        if (global_var(41) == 1) then begin
+                        if (global_var(GVAR_KARMA_WANDERER) == 1) then begin
                             Static_Reaction := Static_Reaction + 0;
                         end
                         else begin
-                            if (global_var(42) == 1) then begin
+                            if (global_var(GVAR_KARMA_BETRAYER) == 1) then begin
                                 Static_Reaction := Static_Reaction + -5;
                             end
                             else begin
-                                if (global_var(43) == 1) then begin
+                                if (global_var(GVAR_KARMA_SWORD_OF_DESPAIR) == 1) then begin
                                     Static_Reaction := Static_Reaction + -10;
                                 end
                                 else begin
-                                    if (global_var(44) == 1) then begin
+                                    if (global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE) == 1) then begin
                                         Static_Reaction := Static_Reaction + -15;
                                     end
                                     else begin
@@ -612,28 +612,28 @@ begin
         debug_msg("Good Person Reaction Bonus == " + Static_Reaction);
     end
     if (Slavery_Tolerant == 0) then begin
-        if (global_var(11) == 1) then begin
+        if (global_var(GVAR_REPUTATION_SLAVER) == 1) then begin
             Static_Reaction := Static_Reaction + (2 * -25);
         end
     end
     else begin
         if (Slavery_Tolerant == 1) then begin
-            if (global_var(11) == 1) then begin
+            if (global_var(GVAR_REPUTATION_SLAVER) == 1) then begin
                 Static_Reaction := Static_Reaction + -25;
             end
         end
         else begin
             if (Slavery_Tolerant == 3) then begin
-                if (global_var(11) == 1) then begin
+                if (global_var(GVAR_REPUTATION_SLAVER) == 1) then begin
                     Static_Reaction := Static_Reaction - -25;
                 end
             end
         end
     end
     debug_msg("Slaver + Aligned Reaction == " + Static_Reaction);
-    Static_Reaction := Static_Reaction + (10 * has_trait(0, dude_obj, 10));
+    Static_Reaction := Static_Reaction + (10 * has_trait(TRAIT_PERK, dude_obj, PERK_presence));
     debug_msg("Presence Reaction == " + Static_Reaction);
-    if (global_var(1) >= 2) then begin
+    if (global_var(GVAR_CHILDKILLER_REPUTATION) >= 2) then begin
         Static_Reaction := Static_Reaction + -30;
     end
     debug_msg("Childkiller Reaction == " + Static_Reaction);
@@ -643,8 +643,8 @@ begin
         call Node007();
     end
     else begin
-        if (global_var(293) bwand 1) then begin
-            set_global_var(288, global_var(288) bwor 4);
+        if (global_var(GVAR_V15_SEED_STATUS) bwand 1) then begin
+            set_global_var(GVAR_V13_V15_DALIA_STATE, global_var(GVAR_V13_V15_DALIA_STATE) bwor 4);
             start_gdialog(555, self_obj, 4, -1, -1);
             gsay_start;
             call Node008();
@@ -653,7 +653,7 @@ begin
         end
         else begin
             if (local_var(4) == 0) then begin
-                set_global_var(288, global_var(288) bwor 4);
+                set_global_var(GVAR_V13_V15_DALIA_STATE, global_var(GVAR_V13_V15_DALIA_STATE) bwor 4);
                 start_gdialog(555, self_obj, 4, -1, -1);
                 gsay_start;
                 call Node001();
@@ -675,59 +675,59 @@ end
 procedure destroy_p_proc
 begin
     if (source_obj == dude_obj) then begin
-        set_global_var(5, global_var(5) + 1);
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(4) > (3 * global_var(5))) or (global_var(2) == 1)) and (global_var(1) == 0)) then begin
-            set_global_var(2, 1);
-            set_global_var(3, 0);
+        set_global_var(GVAR_GOOD_MONSTER, global_var(GVAR_GOOD_MONSTER) + 1);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(4) > (3 * global_var(GVAR_GOOD_MONSTER))) or (global_var(GVAR_CHAMPION_REPUTATION) == 1)) and (global_var(GVAR_CHILDKILLER_REPUTATION) == 0)) then begin
+            set_global_var(GVAR_CHAMPION_REPUTATION, 1);
+            set_global_var(GVAR_BERSERKER_REPUTATION, 0);
         end
-        if (metarule(51, self_obj) == 2) then begin
+        if (critter_kill_type(self_obj) == KILL_TYPE_children_kills) then begin
             if (source_obj == dude_obj) then begin
-                set_global_var(2, 0);
-                set_global_var(1, global_var(1) + 1);
-                set_global_var(0, global_var(0) + -15);
+                set_global_var(GVAR_CHAMPION_REPUTATION, 0);
+                set_global_var(GVAR_CHILDKILLER_REPUTATION, global_var(GVAR_CHILDKILLER_REPUTATION) + 1);
+                set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + -15);
                 debug_msg("Player gains " + -15 + " Karma Points.");
-                set_global_var(37, 0);
-                set_global_var(38, 0);
-                set_global_var(39, 0);
-                set_global_var(40, 0);
-                set_global_var(41, 0);
-                set_global_var(42, 0);
-                set_global_var(43, 0);
-                set_global_var(44, 0);
-                set_global_var(45, 0);
-                if (global_var(0) >= 1000) then begin
-                    set_global_var(37, 1);
+                set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+                set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+                set_global_var(GVAR_KARMA_DEFENDER, 0);
+                set_global_var(GVAR_KARMA_WANDERER, 0);
+                set_global_var(GVAR_KARMA_BETRAYER, 0);
+                set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+                set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+                set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+                    set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
                 end
                 else begin
-                    if (global_var(0) >= 750) then begin
-                        set_global_var(38, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
                     end
                     else begin
-                        if (global_var(0) >= 500) then begin
-                            set_global_var(39, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                            set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                         end
                         else begin
-                            if (global_var(0) >= 250) then begin
-                                set_global_var(40, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                                set_global_var(GVAR_KARMA_DEFENDER, 1);
                             end
                             else begin
-                                if (global_var(0) > -250) then begin
-                                    set_global_var(41, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                                    set_global_var(GVAR_KARMA_WANDERER, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -500) then begin
-                                        set_global_var(42, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                        set_global_var(GVAR_KARMA_BETRAYER, 1);
                                     end
                                     else begin
-                                        if (global_var(0) > -750) then begin
-                                            set_global_var(43, 1);
+                                        if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                            set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                         end
                                         else begin
-                                            if (global_var(0) > -1000) then begin
-                                                set_global_var(44, 1);
+                                            if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                                set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                             end
                                             else begin
-                                                set_global_var(45, 1);
+                                                set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                             end
                                         end
                                     end
@@ -736,58 +736,58 @@ begin
                         end
                     end
                 end
-                set_global_var(294, global_var(294) + -8);
+                set_global_var(GVAR_TOWN_REP_VAULT_15, global_var(GVAR_TOWN_REP_VAULT_15) + -8);
                 debug_msg("Added " + -8 + " to Town Rep");
             end
         end
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(5) > (2 * global_var(4))) or (global_var(3) == 1))) then begin
-            set_global_var(3, 1);
-            set_global_var(2, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
+            set_global_var(GVAR_BERSERKER_REPUTATION, 1);
+            set_global_var(GVAR_CHAMPION_REPUTATION, 0);
         end
-        set_global_var(0, global_var(0) + -10);
+        set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + -10);
         debug_msg("Player gains " + -10 + " Karma Points.");
-        set_global_var(37, 0);
-        set_global_var(38, 0);
-        set_global_var(39, 0);
-        set_global_var(40, 0);
-        set_global_var(41, 0);
-        set_global_var(42, 0);
-        set_global_var(43, 0);
-        set_global_var(44, 0);
-        set_global_var(45, 0);
-        if (global_var(0) >= 1000) then begin
-            set_global_var(37, 1);
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+        set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+        set_global_var(GVAR_KARMA_DEFENDER, 0);
+        set_global_var(GVAR_KARMA_WANDERER, 0);
+        set_global_var(GVAR_KARMA_BETRAYER, 0);
+        set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+        set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+            set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
         end
         else begin
-            if (global_var(0) >= 750) then begin
-                set_global_var(38, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
             end
             else begin
-                if (global_var(0) >= 500) then begin
-                    set_global_var(39, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                 end
                 else begin
-                    if (global_var(0) >= 250) then begin
-                        set_global_var(40, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                        set_global_var(GVAR_KARMA_DEFENDER, 1);
                     end
                     else begin
-                        if (global_var(0) > -250) then begin
-                            set_global_var(41, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                            set_global_var(GVAR_KARMA_WANDERER, 1);
                         end
                         else begin
-                            if (global_var(0) > -500) then begin
-                                set_global_var(42, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                set_global_var(GVAR_KARMA_BETRAYER, 1);
                             end
                             else begin
-                                if (global_var(0) > -750) then begin
-                                    set_global_var(43, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -1000) then begin
-                                        set_global_var(44, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                     end
                                     else begin
-                                        set_global_var(45, 1);
+                                        set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                     end
                                 end
                             end
@@ -796,27 +796,27 @@ begin
                 end
             end
         end
-        set_global_var(294, global_var(294) + -5);
+        set_global_var(GVAR_TOWN_REP_VAULT_15, global_var(GVAR_TOWN_REP_VAULT_15) + -5);
         debug_msg("Added " + -5 + " to Town Rep");
     end
-    set_global_var(288, global_var(288) bwor 1);
+    set_global_var(GVAR_V13_V15_DALIA_STATE, global_var(GVAR_V13_V15_DALIA_STATE) bwor 1);
 end
 
 procedure look_at_p_proc
 begin
     script_overrides;
     if (local_var(4) == 0) then begin
-        display_msg(message_str(555, 100));
+        display_mstr(100);
     end
     else begin
-        display_msg(message_str(555, 101));
+        display_mstr(101);
     end
 end
 
 procedure description_p_proc
 begin
     script_overrides;
-    display_msg(message_str(555, 102));
+    display_mstr(102);
 end
 
 procedure use_skill_on_p_proc
@@ -832,12 +832,12 @@ end
 
 procedure map_enter_p_proc
 begin
-    if (elevation(self_obj) != elevation(dude_obj)) then begin
-        move_to(self_obj, tile_num(self_obj), elevation(dude_obj));
+    if (elevation(self_obj) != dude_elevation) then begin
+        move_to(self_obj, tile_num(self_obj), dude_elevation);
     end
     Only_Once := 0;
-    critter_add_trait(self_obj, 1, 6, 116);
-    critter_add_trait(self_obj, 1, 5, 22);
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 116);
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_TOUGH_GUARD);
 end
 
 procedure timed_event_p_proc
@@ -859,92 +859,92 @@ end
 
 procedure Node001
 begin
-    gsay_reply(555, 103);
+    Reply(103);
     if (q1 == 0) then begin
-        giq_option(-3, 555, 104, Node002a, 50);
-        giq_option(4, 555, 109, Node002a, 50);
+        giq_option(-3, NAME, 104, Node002a, NEUTRAL_REACTION);
+        giq_option(4, NAME, 109, Node002a, NEUTRAL_REACTION);
     end
     if (q2 == 0) then begin
-        giq_option(-3, 555, 105, Node003a, 50);
-        giq_option(4, 555, 110, Node003a, 50);
+        giq_option(-3, NAME, 105, Node003a, NEUTRAL_REACTION);
+        giq_option(4, NAME, 110, Node003a, NEUTRAL_REACTION);
     end
     if (q3 == 0) then begin
-        giq_option(-3, 555, 106, Node004a, 50);
-        giq_option(4, 555, 111, Node004a, 50);
+        giq_option(-3, NAME, 106, Node004a, NEUTRAL_REACTION);
+        giq_option(4, NAME, 111, Node004a, NEUTRAL_REACTION);
     end
-    if ((q4 == 0) and (global_var(288) bwand 8)) then begin
-        giq_option(-3, 555, 107, Node005a, 50);
-        giq_option(4, 555, 112, Node005a, 50);
+    if ((q4 == 0) and (global_var(GVAR_V13_V15_DALIA_STATE) bwand 8)) then begin
+        giq_option(-3, NAME, 107, Node005a, NEUTRAL_REACTION);
+        giq_option(4, NAME, 112, Node005a, NEUTRAL_REACTION);
     end
-    giq_option(-3, 555, 108, Node999, 50);
-    giq_option(4, 555, 113, Node999, 50);
+    giq_option(-3, NAME, 108, Node999, NEUTRAL_REACTION);
+    giq_option(4, NAME, 113, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node002
 begin
-    gsay_reply(555, 114);
-    giq_option(-3, 555, 115, Node001, 50);
-    giq_option(-3, 555, 116, Node999, 50);
-    giq_option(4, 555, 117, Node001, 50);
-    giq_option(4, 555, 118, Node999, 50);
+    Reply(114);
+    giq_option(-3, NAME, 115, Node001, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 116, Node999, NEUTRAL_REACTION);
+    giq_option(4, NAME, 117, Node001, NEUTRAL_REACTION);
+    giq_option(4, NAME, 118, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node003
 begin
-    gsay_reply(555, 119);
-    giq_option(-3, 555, 120, Node999, 50);
-    giq_option(-3, 555, 121, Node001, 50);
-    giq_option(4, 555, 122, Node999, 50);
-    giq_option(4, 555, 123, Node001, 50);
+    Reply(119);
+    giq_option(-3, NAME, 120, Node999, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 121, Node001, NEUTRAL_REACTION);
+    giq_option(4, NAME, 122, Node999, NEUTRAL_REACTION);
+    giq_option(4, NAME, 123, Node001, NEUTRAL_REACTION);
 end
 
 procedure Node004
 begin
-    set_global_var(288, global_var(288) bwor 8);
-    gsay_reply(555, 124);
-    giq_option(-3, 555, 125, Node999, 50);
-    giq_option(-3, 555, 126, Node001, 50);
-    giq_option(4, 555, 127, Node001, 50);
-    giq_option(4, 555, 128, Node999, 50);
+    set_global_var(GVAR_V13_V15_DALIA_STATE, global_var(GVAR_V13_V15_DALIA_STATE) bwor 8);
+    Reply(124);
+    giq_option(-3, NAME, 125, Node999, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 126, Node001, NEUTRAL_REACTION);
+    giq_option(4, NAME, 127, Node001, NEUTRAL_REACTION);
+    giq_option(4, NAME, 128, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node005
 begin
-    gsay_reply(555, 129);
-    giq_option(-3, 555, 130, Node001, 50);
-    giq_option(-3, 555, 131, Node999, 50);
-    giq_option(4, 555, 132, Node001, 50);
-    giq_option(4, 555, 133, Node999, 50);
+    Reply(129);
+    giq_option(-3, NAME, 130, Node001, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 131, Node999, NEUTRAL_REACTION);
+    giq_option(4, NAME, 132, Node001, NEUTRAL_REACTION);
+    giq_option(4, NAME, 133, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node006
 begin
-    float_msg(self_obj, message_str(555, 134), 0);
+    float_msg(self_obj, mstr(134), FLOAT_MSG_NORMAL);
 end
 
 procedure Node007
 begin
-    float_msg(self_obj, message_str(555, 137), 0);
+    float_msg(self_obj, mstr(137), FLOAT_MSG_NORMAL);
 end
 
 procedure Node008
 begin
-    gsay_reply(555, 140);
-    giq_option(-3, 555, 141, Node009, 50);
-    giq_option(-3, 555, 142, Node999, 50);
-    giq_option(4, 555, 143, Node999, 50);
-    giq_option(4, 555, 144, Node009, 50);
+    Reply(140);
+    giq_option(-3, NAME, 141, Node009, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 142, Node999, NEUTRAL_REACTION);
+    giq_option(4, NAME, 143, Node999, NEUTRAL_REACTION);
+    giq_option(4, NAME, 144, Node009, NEUTRAL_REACTION);
 end
 
 procedure Node009
 begin
-    gsay_reply(555, 145);
-    giq_option(-3, 555, 146, Node012, 50);
-    giq_option(-3, 555, 147, Node999, 50);
-    giq_option(-3, 555, 148, Node011, 50);
-    giq_option(4, 555, 149, Node012, 50);
-    giq_option(4, 555, 150, Node999, 50);
-    giq_option(4, 555, 151, Node011, 50);
+    Reply(145);
+    giq_option(-3, NAME, 146, Node012, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 147, Node999, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 148, Node011, NEUTRAL_REACTION);
+    giq_option(4, NAME, 149, Node012, NEUTRAL_REACTION);
+    giq_option(4, NAME, 150, Node999, NEUTRAL_REACTION);
+    giq_option(4, NAME, 151, Node011, NEUTRAL_REACTION);
 end
 
 procedure Node010
@@ -954,24 +954,24 @@ end
 procedure Node011
 begin
     doTransport := 1;
-    gsay_reply(555, 152);
-    giq_option(-3, 555, 153, Node999, 50);
-    giq_option(4, 555, 154, Node999, 50);
+    Reply(152);
+    giq_option(-3, NAME, 153, Node999, NEUTRAL_REACTION);
+    giq_option(4, NAME, 154, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node012
 begin
-    gsay_reply(555, 155);
-    giq_option(0, 555, message_str(14, 350), Node999, 50);
+    Reply(155);
+    giq_option(0, NAME, g_mstr(350), Node999, NEUTRAL_REACTION);
 end
 
 procedure do_teleport
 begin
-    gfade_out(1);
-    set_global_var(288, global_var(288) bwor 2);
+    fade_out;
+    set_global_var(GVAR_V13_V15_DALIA_STATE, global_var(GVAR_V13_V15_DALIA_STATE) bwor 2);
     set_obj_visibility(self_obj, 1);
     add_timer_event(self_obj, 0, 1);
-    gfade_in(1);
+    fade_in;
 end
 
 procedure Node002a
@@ -997,4 +997,3 @@ begin
     q4 := 1;
     call Node005();
 end
-
