diff --git a/Games/Fallout2.Steam/Original/BLKROM.ssl b/Games/Fallout2.Steam/ReDefine/BLKROM.ssl
index f701689..7549738 100644
--- a/Games/Fallout2.Steam/Original/BLKROM.ssl
+++ b/Games/Fallout2.Steam/ReDefine/BLKROM.ssl
@@ -3,18 +3,17 @@ procedure start;
 
 procedure start
 begin
-    if (script_action == 2) then begin
+    if (script_action == spatial_proc) then begin
         if (obj_type(source_obj) == dude_obj) then begin
             if (local_var(4) == 0) then begin
-                if (get_critter_stat(dude_obj, 4) > 6) then begin
-                    display_msg(message_str(189, 100));
+                if (dude_iq > 6) then begin
+                    display_msg(message_str(SCRIPT_GSTCAV1, 100));
                 end
                 else begin
-                    display_msg(message_str(189, 101));
+                    display_msg(message_str(SCRIPT_GSTCAV1, 101));
                 end
                 set_local_var(4, 1);
             end
         end
     end
 end
-
