diff --git a/Games/Fallout2.Steam/Original/BROKEN2.ssl b/Games/Fallout2.Steam/ReDefine/BROKEN2.ssl
index 34af127..7ddfd8c 100644
--- a/Games/Fallout2.Steam/Original/BROKEN2.ssl
+++ b/Games/Fallout2.Steam/ReDefine/BROKEN2.ssl
@@ -159,13 +159,13 @@ begin
     override_map_start(21144 % 200, 21144 / 200, 0, 0);
     set_map_var(23, 1);
     Damage_Timer_Started := 0;
-    if (metarule(14, 0)) then begin
-        display_msg(message_str(899, 100));
+    if (map_first_run) then begin
+        display_mstr(100);
     end
-    if (global_var(443) == 2) then begin
-        set_global_var(443, 3);
+    if (global_var(GVAR_JAIL_BREAK) == 2) then begin
+        set_global_var(GVAR_JAIL_BREAK, 3);
     end
-    if (elevation(dude_obj) == 0) then begin
+    if (dude_elevation == 0) then begin
         if ((get_month >= 3) and (get_month < 5)) then begin
             if ((game_time_hour >= 600) and (game_time_hour < (600 + 100))) then begin
                 set_light_level(game_time_hour - 600 + 40);
@@ -246,17 +246,17 @@ begin
     else begin
         set_light_level(50);
     end
-    if ((global_var(302) < 7) and (Damage_Timer_Started == 0)) then begin
+    if ((global_var(GVAR_BH_MINE) < 7) and (Damage_Timer_Started == 0)) then begin
         Damage_Timer_Started := 1;
         add_timer_event(self_obj, game_ticks(20), 2);
     end
-    set_global_var(27, 0);
+    set_global_var(GVAR_LOAD_MAP_INDEX, 0);
 end
 
 procedure map_update_p_proc
 begin
     set_map_var(23, 1);
-    if (elevation(dude_obj) == 0) then begin
+    if (dude_elevation == 0) then begin
         if ((get_month >= 3) and (get_month < 5)) then begin
             if ((game_time_hour >= 600) and (game_time_hour < (600 + 100))) then begin
                 set_light_level(game_time_hour - 600 + 40);
@@ -343,9 +343,9 @@ begin
         move_to(dude_obj, 14095, 0);
         critter_attempt_placement(dude_obj, 14095, 0);
         move_to(broken_hills_cavein, 9091, 0);
-        create_object_sid(33554499, 9493, 0, -1);
-        create_object_sid(33554499, 9292, 0, -1);
-        create_object_sid(33554499, 9491, 0, -1);
+        create_object(33554499, 9493, 0);
+        create_object(33554499, 9292, 0);
+        create_object(33554499, 9491, 0);
         add_timer_event(self_obj, game_ticks(2), 1);
     end
 end
@@ -357,8 +357,8 @@ begin
     end
     else begin
         if (fixed_param == 2) then begin
-            if (global_var(302) < 7) then begin
-                if (elevation(dude_obj) == 2) then begin
+            if (global_var(GVAR_BH_MINE) < 7) then begin
+                if (dude_elevation == 2) then begin
                     call CauseDamage();
                 end
                 add_timer_event(self_obj, game_ticks(20), 2);
@@ -370,75 +370,74 @@ end
 procedure CauseDamage
 begin
     variable LVar0 := 0;
-    if (not((obj_pid(critter_inven_obj(dude_obj, 0)) == 3) or (obj_pid(critter_inven_obj(dude_obj, 0)) == 232) or (obj_pid(critter_inven_obj(dude_obj, 0)) == 348) or (obj_pid(critter_inven_obj(dude_obj, 0)) == 349))) then begin
+    if (not((obj_pid(critter_inven_obj(dude_obj, INVEN_TYPE_WORN)) == 3) or (obj_pid(critter_inven_obj(dude_obj, INVEN_TYPE_WORN)) == 232) or (obj_pid(critter_inven_obj(dude_obj, INVEN_TYPE_WORN)) == 348) or (obj_pid(critter_inven_obj(dude_obj, INVEN_TYPE_WORN)) == 349))) then begin
         damage := random(10, 15);
         critter_dmg(dude_obj, damage, 0 bwor 512 bwor 256);
     end
     if (party_member_obj(16777278) != 0) then begin
-        if (not((obj_pid(critter_inven_obj(party_member_obj(16777278), 0)) == 3) or (obj_pid(critter_inven_obj(party_member_obj(16777278), 0)) == 232) or (obj_pid(critter_inven_obj(party_member_obj(16777278), 0)) == 348) or (obj_pid(critter_inven_obj(party_member_obj(16777278), 0)) == 349))) then begin
+        if (not((obj_pid(critter_inven_obj(party_member_obj(16777278), INVEN_TYPE_WORN)) == 3) or (obj_pid(critter_inven_obj(party_member_obj(16777278), INVEN_TYPE_WORN)) == 232) or (obj_pid(critter_inven_obj(party_member_obj(16777278), INVEN_TYPE_WORN)) == 348) or (obj_pid(critter_inven_obj(party_member_obj(16777278), INVEN_TYPE_WORN)) == 349))) then begin
             damage := random(10, 15);
             critter_dmg(party_member_obj(16777278), damage, 0 bwor 512 bwor 256);
         end
     end
     if (party_member_obj(16777376) != 0) then begin
-        if (not((obj_pid(critter_inven_obj(party_member_obj(16777376), 0)) == 3) or (obj_pid(critter_inven_obj(party_member_obj(16777376), 0)) == 232) or (obj_pid(critter_inven_obj(party_member_obj(16777376), 0)) == 348) or (obj_pid(critter_inven_obj(party_member_obj(16777376), 0)) == 349))) then begin
+        if (not((obj_pid(critter_inven_obj(party_member_obj(16777376), INVEN_TYPE_WORN)) == 3) or (obj_pid(critter_inven_obj(party_member_obj(16777376), INVEN_TYPE_WORN)) == 232) or (obj_pid(critter_inven_obj(party_member_obj(16777376), INVEN_TYPE_WORN)) == 348) or (obj_pid(critter_inven_obj(party_member_obj(16777376), INVEN_TYPE_WORN)) == 349))) then begin
             damage := random(10, 15);
             critter_dmg(party_member_obj(16777376), damage, 0 bwor 512 bwor 256);
         end
     end
     if (party_member_obj(16777377) != 0) then begin
-        if (not((obj_pid(critter_inven_obj(party_member_obj(16777377), 0)) == 3) or (obj_pid(critter_inven_obj(party_member_obj(16777377), 0)) == 232) or (obj_pid(critter_inven_obj(party_member_obj(16777377), 0)) == 348) or (obj_pid(critter_inven_obj(party_member_obj(16777377), 0)) == 349))) then begin
+        if (not((obj_pid(critter_inven_obj(party_member_obj(16777377), INVEN_TYPE_WORN)) == 3) or (obj_pid(critter_inven_obj(party_member_obj(16777377), INVEN_TYPE_WORN)) == 232) or (obj_pid(critter_inven_obj(party_member_obj(16777377), INVEN_TYPE_WORN)) == 348) or (obj_pid(critter_inven_obj(party_member_obj(16777377), INVEN_TYPE_WORN)) == 349))) then begin
             damage := random(10, 15);
             critter_dmg(party_member_obj(16777377), damage, 0 bwor 512 bwor 256);
         end
     end
     if (party_member_obj(16777305) != 0) then begin
-        if (not((obj_pid(critter_inven_obj(party_member_obj(16777305), 0)) == 3) or (obj_pid(critter_inven_obj(party_member_obj(16777305), 0)) == 232) or (obj_pid(critter_inven_obj(party_member_obj(16777305), 0)) == 348) or (obj_pid(critter_inven_obj(party_member_obj(16777305), 0)) == 349))) then begin
+        if (not((obj_pid(critter_inven_obj(party_member_obj(16777305), INVEN_TYPE_WORN)) == 3) or (obj_pid(critter_inven_obj(party_member_obj(16777305), INVEN_TYPE_WORN)) == 232) or (obj_pid(critter_inven_obj(party_member_obj(16777305), INVEN_TYPE_WORN)) == 348) or (obj_pid(critter_inven_obj(party_member_obj(16777305), INVEN_TYPE_WORN)) == 349))) then begin
             damage := random(10, 15);
             critter_dmg(party_member_obj(16777305), damage, 0 bwor 512 bwor 256);
         end
     end
     if (party_member_obj(16777313) != 0) then begin
-        if (not((obj_pid(critter_inven_obj(party_member_obj(16777313), 0)) == 3) or (obj_pid(critter_inven_obj(party_member_obj(16777313), 0)) == 232) or (obj_pid(critter_inven_obj(party_member_obj(16777313), 0)) == 348) or (obj_pid(critter_inven_obj(party_member_obj(16777313), 0)) == 349))) then begin
+        if (not((obj_pid(critter_inven_obj(party_member_obj(16777313), INVEN_TYPE_WORN)) == 3) or (obj_pid(critter_inven_obj(party_member_obj(16777313), INVEN_TYPE_WORN)) == 232) or (obj_pid(critter_inven_obj(party_member_obj(16777313), INVEN_TYPE_WORN)) == 348) or (obj_pid(critter_inven_obj(party_member_obj(16777313), INVEN_TYPE_WORN)) == 349))) then begin
             damage := random(10, 15);
             critter_dmg(party_member_obj(16777313), damage, 0 bwor 512 bwor 256);
         end
     end
     if (party_member_obj(16777323) != 0) then begin
-        if (not((obj_pid(critter_inven_obj(party_member_obj(16777323), 0)) == 3) or (obj_pid(critter_inven_obj(party_member_obj(16777323), 0)) == 232) or (obj_pid(critter_inven_obj(party_member_obj(16777323), 0)) == 348) or (obj_pid(critter_inven_obj(party_member_obj(16777323), 0)) == 349))) then begin
+        if (not((obj_pid(critter_inven_obj(party_member_obj(16777323), INVEN_TYPE_WORN)) == 3) or (obj_pid(critter_inven_obj(party_member_obj(16777323), INVEN_TYPE_WORN)) == 232) or (obj_pid(critter_inven_obj(party_member_obj(16777323), INVEN_TYPE_WORN)) == 348) or (obj_pid(critter_inven_obj(party_member_obj(16777323), INVEN_TYPE_WORN)) == 349))) then begin
             damage := random(10, 15);
             critter_dmg(party_member_obj(16777323), damage, 0 bwor 512 bwor 256);
         end
     end
     if (party_member_obj(16777352) != 0) then begin
-        if (not((obj_pid(critter_inven_obj(party_member_obj(16777352), 0)) == 3) or (obj_pid(critter_inven_obj(party_member_obj(16777352), 0)) == 232) or (obj_pid(critter_inven_obj(party_member_obj(16777352), 0)) == 348) or (obj_pid(critter_inven_obj(party_member_obj(16777352), 0)) == 349))) then begin
+        if (not((obj_pid(critter_inven_obj(party_member_obj(16777352), INVEN_TYPE_WORN)) == 3) or (obj_pid(critter_inven_obj(party_member_obj(16777352), INVEN_TYPE_WORN)) == 232) or (obj_pid(critter_inven_obj(party_member_obj(16777352), INVEN_TYPE_WORN)) == 348) or (obj_pid(critter_inven_obj(party_member_obj(16777352), INVEN_TYPE_WORN)) == 349))) then begin
             damage := random(10, 15);
             critter_dmg(party_member_obj(16777352), damage, 0 bwor 512 bwor 256);
         end
     end
     if (party_member_obj(16777378) != 0) then begin
-        if (not((obj_pid(critter_inven_obj(party_member_obj(16777378), 0)) == 3) or (obj_pid(critter_inven_obj(party_member_obj(16777378), 0)) == 232) or (obj_pid(critter_inven_obj(party_member_obj(16777378), 0)) == 348) or (obj_pid(critter_inven_obj(party_member_obj(16777378), 0)) == 349))) then begin
+        if (not((obj_pid(critter_inven_obj(party_member_obj(16777378), INVEN_TYPE_WORN)) == 3) or (obj_pid(critter_inven_obj(party_member_obj(16777378), INVEN_TYPE_WORN)) == 232) or (obj_pid(critter_inven_obj(party_member_obj(16777378), INVEN_TYPE_WORN)) == 348) or (obj_pid(critter_inven_obj(party_member_obj(16777378), INVEN_TYPE_WORN)) == 349))) then begin
             damage := random(10, 15);
             critter_dmg(party_member_obj(16777378), damage, 0 bwor 512 bwor 256);
         end
     end
     if (party_member_obj(16777368) != 0) then begin
-        if (not((obj_pid(critter_inven_obj(party_member_obj(16777368), 0)) == 3) or (obj_pid(critter_inven_obj(party_member_obj(16777368), 0)) == 232) or (obj_pid(critter_inven_obj(party_member_obj(16777368), 0)) == 348) or (obj_pid(critter_inven_obj(party_member_obj(16777368), 0)) == 349))) then begin
+        if (not((obj_pid(critter_inven_obj(party_member_obj(16777368), INVEN_TYPE_WORN)) == 3) or (obj_pid(critter_inven_obj(party_member_obj(16777368), INVEN_TYPE_WORN)) == 232) or (obj_pid(critter_inven_obj(party_member_obj(16777368), INVEN_TYPE_WORN)) == 348) or (obj_pid(critter_inven_obj(party_member_obj(16777368), INVEN_TYPE_WORN)) == 349))) then begin
             damage := random(10, 15);
             critter_dmg(party_member_obj(16777368), damage, 0 bwor 512 bwor 256);
         end
     end
     if (party_member_obj(16777379) != 0) then begin
-        if (not((obj_pid(critter_inven_obj(party_member_obj(16777379), 0)) == 3) or (obj_pid(critter_inven_obj(party_member_obj(16777379), 0)) == 232) or (obj_pid(critter_inven_obj(party_member_obj(16777379), 0)) == 348) or (obj_pid(critter_inven_obj(party_member_obj(16777379), 0)) == 349))) then begin
+        if (not((obj_pid(critter_inven_obj(party_member_obj(16777379), INVEN_TYPE_WORN)) == 3) or (obj_pid(critter_inven_obj(party_member_obj(16777379), INVEN_TYPE_WORN)) == 232) or (obj_pid(critter_inven_obj(party_member_obj(16777379), INVEN_TYPE_WORN)) == 348) or (obj_pid(critter_inven_obj(party_member_obj(16777379), INVEN_TYPE_WORN)) == 349))) then begin
             damage := random(10, 15);
             critter_dmg(party_member_obj(16777379), damage, 0 bwor 512 bwor 256);
         end
     end
     if (party_member_obj(16777380) != 0) then begin
-        if (not((obj_pid(critter_inven_obj(party_member_obj(16777380), 0)) == 3) or (obj_pid(critter_inven_obj(party_member_obj(16777380), 0)) == 232) or (obj_pid(critter_inven_obj(party_member_obj(16777380), 0)) == 348) or (obj_pid(critter_inven_obj(party_member_obj(16777380), 0)) == 349))) then begin
+        if (not((obj_pid(critter_inven_obj(party_member_obj(16777380), INVEN_TYPE_WORN)) == 3) or (obj_pid(critter_inven_obj(party_member_obj(16777380), INVEN_TYPE_WORN)) == 232) or (obj_pid(critter_inven_obj(party_member_obj(16777380), INVEN_TYPE_WORN)) == 348) or (obj_pid(critter_inven_obj(party_member_obj(16777380), INVEN_TYPE_WORN)) == 349))) then begin
             damage := random(10, 15);
             critter_dmg(party_member_obj(16777380), damage, 0 bwor 512 bwor 256);
         end
     end
 end
-
