diff --git a/Games/Fallout2.Steam/Original/BROKEN1.ssl b/Games/Fallout2.Steam/ReDefine/BROKEN1.ssl
index dda6d8f..156e8ed 100644
--- a/Games/Fallout2.Steam/Original/BROKEN1.ssl
+++ b/Games/Fallout2.Steam/ReDefine/BROKEN1.ssl
@@ -156,110 +156,110 @@ end
 procedure map_enter_p_proc
 begin
     override_map_start(17947 % 200, 17947 / 200, 0, 3);
-    if (metarule(14, 0)) then begin
-        display_msg(message_str(675, 100));
+    if (map_first_run) then begin
+        display_mstr(100);
     end
     debug_msg("Check_Create_Car(" + 17752 + "," + 0 + ")");
-    if (not(metarule(22, 0))) then begin
-        if (global_var(18) != 0) then begin
-            debug_msg("car_current_town == " + metarule(30, 0) + " / map_get_load_area == " + metarule3(111, 0, 0, 0));
-            if ((metarule(30, 0) == metarule3(111, 0, 0, 0)) or ((26 <= metarule(30, 0)) and (metarule(30, 0) <= 29) and (metarule3(111, 0, 0, 0) == -1))) then begin
+    if (not(is_loading_game)) then begin
+        if (global_var(GVAR_PLAYER_GOT_CAR) != 0) then begin
+            debug_msg("car_current_town == " + car_current_town + " / map_get_load_area == " + map_get_load_area);
+            if ((car_current_town == map_get_load_area) or ((26 <= car_current_town) and (car_current_town <= 29) and (map_get_load_area == -1))) then begin
                 debug_msg("Place_Car(" + 17752 + "," + 0 + ")");
-                debug_msg("   GVAR_CAR_PLACED_TILE == " + global_var(633));
+                debug_msg("   GVAR_CAR_PLACED_TILE == " + global_var(GVAR_CAR_PLACED_TILE));
                 debug_msg("   Car_At_Loc == " + (tile_contains_pid_obj(17752, 0, 33555441) != 0));
-                if ((global_var(633) <= 0) or (17752 == global_var(633))) then begin
-                    set_global_var(633, 17752);
+                if ((global_var(GVAR_CAR_PLACED_TILE) <= 0) or (17752 == global_var(GVAR_CAR_PLACED_TILE))) then begin
+                    set_global_var(GVAR_CAR_PLACED_TILE, 17752);
                     if (not(tile_contains_pid_obj(17752, 0, 33555441) != 0)) then begin
-                        Scenery_Creation := create_object_sid(33555441, 17752, 0, 304);
+                        Scenery_Creation := create_object_sid(33555441, 17752, 0, SCRIPT_ZSDRVCAR);
                         Scenery_Creation_Hex := 17752;
                         Scenery_Creation_Count := 0;
                         while (Scenery_Creation_Count < 2) do begin
                             Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 1, 1);
-                            Scenery_Creation := create_object_sid(33554499, Scenery_Creation_Hex, 0, -1);
+                            Scenery_Creation := create_object(33554499, Scenery_Creation_Hex, 0);
                             Scenery_Creation_Count := Scenery_Creation_Count + 1;
                         end
                         Scenery_Creation_Count := 0;
                         while (Scenery_Creation_Count < 2) do begin
                             Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 2, 1);
-                            Scenery_Creation := create_object_sid(33554499, Scenery_Creation_Hex, 0, -1);
+                            Scenery_Creation := create_object(33554499, Scenery_Creation_Hex, 0);
                             Scenery_Creation_Count := Scenery_Creation_Count + 1;
                         end
                         Scenery_Creation_Count := 0;
                         while (Scenery_Creation_Count < 2) do begin
                             Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 3, 1);
-                            Scenery_Creation := create_object_sid(33554499, Scenery_Creation_Hex, 0, -1);
+                            Scenery_Creation := create_object(33554499, Scenery_Creation_Hex, 0);
                             Scenery_Creation_Count := Scenery_Creation_Count + 1;
                         end
                         Scenery_Creation_Count := 0;
                         while (Scenery_Creation_Count < 2) do begin
                             Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 4, 1);
-                            Scenery_Creation := create_object_sid(33554499, Scenery_Creation_Hex, 0, -1);
+                            Scenery_Creation := create_object(33554499, Scenery_Creation_Hex, 0);
                             Scenery_Creation_Count := Scenery_Creation_Count + 1;
                         end
                         Scenery_Creation_Count := 0;
                         while (Scenery_Creation_Count < 1) do begin
                             Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 5, 1);
-                            Scenery_Creation := create_object_sid(33554499, Scenery_Creation_Hex, 0, -1);
+                            Scenery_Creation := create_object(33554499, Scenery_Creation_Hex, 0);
                             Scenery_Creation_Count := Scenery_Creation_Count + 1;
                         end
                         Scenery_Creation_Count := 0;
                         while (Scenery_Creation_Count < 1) do begin
                             Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 4, 1);
-                            Scenery_Creation := create_object_sid(33554499, Scenery_Creation_Hex, 0, -1);
+                            Scenery_Creation := create_object(33554499, Scenery_Creation_Hex, 0);
                             Scenery_Creation_Count := Scenery_Creation_Count + 1;
                         end
                         Scenery_Creation_Count := 0;
                         while (Scenery_Creation_Count < 1) do begin
                             Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 5, 1);
-                            Scenery_Creation := create_object_sid(33554499, Scenery_Creation_Hex, 0, -1);
+                            Scenery_Creation := create_object(33554499, Scenery_Creation_Hex, 0);
                             Scenery_Creation_Count := Scenery_Creation_Count + 1;
                         end
                         Scenery_Creation_Count := 0;
                         while (Scenery_Creation_Count < 1) do begin
                             Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 4, 1);
-                            Scenery_Creation := create_object_sid(33554499, Scenery_Creation_Hex, 0, -1);
+                            Scenery_Creation := create_object(33554499, Scenery_Creation_Hex, 0);
                             Scenery_Creation_Count := Scenery_Creation_Count + 1;
                         end
                         Scenery_Creation_Count := 0;
                         while (Scenery_Creation_Count < 1) do begin
                             Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 5, 1);
-                            Scenery_Creation := create_object_sid(33554499, Scenery_Creation_Hex, 0, -1);
+                            Scenery_Creation := create_object(33554499, Scenery_Creation_Hex, 0);
                             Scenery_Creation_Count := Scenery_Creation_Count + 1;
                         end
                         Scenery_Creation_Count := 0;
                         while (Scenery_Creation_Count < 1) do begin
                             Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 4, 1);
-                            Scenery_Creation := create_object_sid(33554499, Scenery_Creation_Hex, 0, -1);
+                            Scenery_Creation := create_object(33554499, Scenery_Creation_Hex, 0);
                             Scenery_Creation_Count := Scenery_Creation_Count + 1;
                         end
                         Scenery_Creation_Count := 0;
                         while (Scenery_Creation_Count < 1) do begin
                             Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 0, 1);
-                            Scenery_Creation := create_object_sid(33554499, Scenery_Creation_Hex, 0, -1);
+                            Scenery_Creation := create_object(33554499, Scenery_Creation_Hex, 0);
                             Scenery_Creation_Count := Scenery_Creation_Count + 1;
                         end
                         Scenery_Creation_Count := 0;
                         while (Scenery_Creation_Count < 1) do begin
                             Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 5, 1);
-                            Scenery_Creation := create_object_sid(33554499, Scenery_Creation_Hex, 0, -1);
+                            Scenery_Creation := create_object(33554499, Scenery_Creation_Hex, 0);
                             Scenery_Creation_Count := Scenery_Creation_Count + 1;
                         end
                         Scenery_Creation_Count := 0;
                         while (Scenery_Creation_Count < 2) do begin
                             Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 0, 1);
-                            Scenery_Creation := create_object_sid(33554499, Scenery_Creation_Hex, 0, -1);
+                            Scenery_Creation := create_object(33554499, Scenery_Creation_Hex, 0);
                             Scenery_Creation_Count := Scenery_Creation_Count + 1;
                         end
                         Scenery_Creation_Count := 0;
                         while (Scenery_Creation_Count < 1) do begin
                             Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 1, 1);
-                            Scenery_Creation := create_object_sid(33554499, Scenery_Creation_Hex, 0, -1);
+                            Scenery_Creation := create_object(33554499, Scenery_Creation_Hex, 0);
                             Scenery_Creation_Count := Scenery_Creation_Count + 1;
                         end
                         Scenery_Creation_Count := 0;
                         while (Scenery_Creation_Count < 1) do begin
                             Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 2, 1);
-                            Scenery_Creation := create_object_sid(33554499, Scenery_Creation_Hex, 0, -1);
+                            Scenery_Creation := create_object(33554499, Scenery_Creation_Hex, 0);
                             Scenery_Creation_Count := Scenery_Creation_Count + 1;
                         end
                     end
@@ -272,8 +272,8 @@ begin
                             Scenery_Creation_Ptr := party_member_obj(455);
                         end
                         else begin
-                            if (global_var(18) == 0) then begin
-                                Scenery_Creation_Ptr := create_object_sid(455, 0, 0, 920);
+                            if (global_var(GVAR_PLAYER_GOT_CAR) == 0) then begin
+                                Scenery_Creation_Ptr := create_object_sid(455, 0, 0, SCRIPT_ZICRTRNK);
                             end
                         end
                         if (Scenery_Creation_Ptr != 0) then begin
@@ -282,13 +282,13 @@ begin
                             Scenery_Creation_Count := 0;
                             while (Scenery_Creation_Count < 1) do begin
                                 Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 1, 1);
-                                Scenery_Creation := create_object_sid(33554499, Scenery_Creation_Hex, 0, -1);
+                                Scenery_Creation := create_object(33554499, Scenery_Creation_Hex, 0);
                                 Scenery_Creation_Count := Scenery_Creation_Count + 1;
                             end
                             Scenery_Creation_Count := 0;
                             while (Scenery_Creation_Count < 1) do begin
                                 Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 2, 1);
-                                Scenery_Creation := create_object_sid(33554499, Scenery_Creation_Hex, 0, -1);
+                                Scenery_Creation := create_object(33554499, Scenery_Creation_Hex, 0);
                                 Scenery_Creation_Count := Scenery_Creation_Count + 1;
                             end
                         end
@@ -444,7 +444,7 @@ begin
     else begin
         debug_msg("sorry, can't place the car, the game's loading");
     end
-    if (elevation(dude_obj) == 0) then begin
+    if (dude_elevation == 0) then begin
         if ((get_month >= 3) and (get_month < 5)) then begin
             if ((game_time_hour >= 600) and (game_time_hour < (600 + 100))) then begin
                 set_light_level(game_time_hour - 600 + 40);
@@ -525,12 +525,12 @@ begin
     else begin
         set_light_level(50);
     end
-    set_global_var(27, 0);
+    set_global_var(GVAR_LOAD_MAP_INDEX, 0);
 end
 
 procedure map_update_p_proc
 begin
-    if (elevation(dude_obj) == 0) then begin
+    if (dude_elevation == 0) then begin
         if ((get_month >= 3) and (get_month < 5)) then begin
             if ((game_time_hour >= 600) and (game_time_hour < (600 + 100))) then begin
                 set_light_level(game_time_hour - 600 + 40);
@@ -617,4 +617,3 @@ procedure map_exit_p_proc
 begin
     set_map_var(23, 0);
 end
-
