diff --git a/Games/Fallout2.Steam/Original/BV2VAULT.ssl b/Games/Fallout2.Steam/ReDefine/BV2VAULT.ssl
index dfad6f7..4d51078 100644
--- a/Games/Fallout2.Steam/Original/BV2VAULT.ssl
+++ b/Games/Fallout2.Steam/ReDefine/BV2VAULT.ssl
@@ -3,8 +3,7 @@ procedure start;
 
 procedure start
 begin
-    if (script_action == 6) then begin
+    if (script_action == use_proc) then begin
         load_map("VAULTBUR.map", 0);
     end
 end
-
