diff --git a/Games/Fallout2.Steam/Original/BVAIR.ssl b/Games/Fallout2.Steam/ReDefine/BVAIR.ssl
index f528991..150fa6c 100644
--- a/Games/Fallout2.Steam/Original/BVAIR.ssl
+++ b/Games/Fallout2.Steam/ReDefine/BVAIR.ssl
@@ -3,15 +3,14 @@ procedure start;
 
 procedure start
 begin
-    if (script_action == 2) then begin
+    if (script_action == spatial_proc) then begin
         if (obj_type(source_obj) == dude_obj) then begin
             if (local_var(0) == 0) then begin
                 set_local_var(0, 1);
                 if (is_success(do_check(dude_obj, 4, 0))) then begin
-                    display_msg(message_str(190, 100));
+                    display_msg(message_str(SCRIPT_GSTCAV2, 100));
                 end
             end
         end
     end
 end
-
