diff --git a/Games/Fallout2.Steam/Original/CAVE06.ssl b/Games/Fallout2.Steam/ReDefine/CAVE06.ssl
index d70c202..728af7b 100644
--- a/Games/Fallout2.Steam/Original/CAVE06.ssl
+++ b/Games/Fallout2.Steam/ReDefine/CAVE06.ssl
@@ -133,113 +133,113 @@ end
 procedure map_enter_p_proc
 begin
     set_light_level(50);
-    if ((map_var(0) == 0) and (global_var(8) != 0) and (metarule(30, 0) == 11)) then begin
+    if ((map_var(0) == 0) and (global_var(GVAR_KNOWLEDGE_HEALING_POWDER) != 0) and (car_current_town == 11)) then begin
         set_map_var(0, 1);
-        if (not(metarule(22, 0))) then begin
-            Scenery_Creation := create_object_sid(33555441, 19500, 0, 304);
+        if (not(is_loading_game)) then begin
+            Scenery_Creation := create_object_sid(33555441, 19500, 0, SCRIPT_ZSDRVCAR);
             Scenery_Creation_Hex := 19500;
             Scenery_Creation_Count := 0;
             while (Scenery_Creation_Count < 2) do begin
                 Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 1, 1);
-                Scenery_Creation := create_object_sid(33554499, Scenery_Creation_Hex, 0, -1);
+                Scenery_Creation := create_object(33554499, Scenery_Creation_Hex, 0);
                 Scenery_Creation_Count := Scenery_Creation_Count + 1;
                 debug_msg("Putting Blocking Hex at " + Scenery_Creation_Hex);
             end
             Scenery_Creation_Count := 0;
             while (Scenery_Creation_Count < 2) do begin
                 Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 2, 1);
-                Scenery_Creation := create_object_sid(33554499, Scenery_Creation_Hex, 0, -1);
+                Scenery_Creation := create_object(33554499, Scenery_Creation_Hex, 0);
                 Scenery_Creation_Count := Scenery_Creation_Count + 1;
                 debug_msg("Putting Blocking Hex at " + Scenery_Creation_Hex);
             end
             Scenery_Creation_Count := 0;
             while (Scenery_Creation_Count < 2) do begin
                 Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 3, 1);
-                Scenery_Creation := create_object_sid(33554499, Scenery_Creation_Hex, 0, -1);
+                Scenery_Creation := create_object(33554499, Scenery_Creation_Hex, 0);
                 Scenery_Creation_Count := Scenery_Creation_Count + 1;
                 debug_msg("Putting Blocking Hex at " + Scenery_Creation_Hex);
             end
             Scenery_Creation_Count := 0;
             while (Scenery_Creation_Count < 2) do begin
                 Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 4, 1);
-                Scenery_Creation := create_object_sid(33554499, Scenery_Creation_Hex, 0, -1);
+                Scenery_Creation := create_object(33554499, Scenery_Creation_Hex, 0);
                 Scenery_Creation_Count := Scenery_Creation_Count + 1;
                 debug_msg("Putting Blocking Hex at " + Scenery_Creation_Hex);
             end
             Scenery_Creation_Count := 0;
             while (Scenery_Creation_Count < 1) do begin
                 Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 5, 1);
-                Scenery_Creation := create_object_sid(33554499, Scenery_Creation_Hex, 0, -1);
+                Scenery_Creation := create_object(33554499, Scenery_Creation_Hex, 0);
                 Scenery_Creation_Count := Scenery_Creation_Count + 1;
                 debug_msg("Putting Blocking Hex at " + Scenery_Creation_Hex);
             end
             Scenery_Creation_Count := 0;
             while (Scenery_Creation_Count < 1) do begin
                 Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 4, 1);
-                Scenery_Creation := create_object_sid(33554499, Scenery_Creation_Hex, 0, -1);
+                Scenery_Creation := create_object(33554499, Scenery_Creation_Hex, 0);
                 Scenery_Creation_Count := Scenery_Creation_Count + 1;
                 debug_msg("Putting Blocking Hex at " + Scenery_Creation_Hex);
             end
             Scenery_Creation_Count := 0;
             while (Scenery_Creation_Count < 1) do begin
                 Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 5, 1);
-                Scenery_Creation := create_object_sid(33554499, Scenery_Creation_Hex, 0, -1);
+                Scenery_Creation := create_object(33554499, Scenery_Creation_Hex, 0);
                 Scenery_Creation_Count := Scenery_Creation_Count + 1;
                 debug_msg("Putting Blocking Hex at " + Scenery_Creation_Hex);
             end
             Scenery_Creation_Count := 0;
             while (Scenery_Creation_Count < 1) do begin
                 Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 4, 1);
-                Scenery_Creation := create_object_sid(33554499, Scenery_Creation_Hex, 0, -1);
+                Scenery_Creation := create_object(33554499, Scenery_Creation_Hex, 0);
                 Scenery_Creation_Count := Scenery_Creation_Count + 1;
                 debug_msg("Putting Blocking Hex at " + Scenery_Creation_Hex);
             end
             Scenery_Creation_Count := 0;
             while (Scenery_Creation_Count < 1) do begin
                 Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 5, 1);
-                Scenery_Creation := create_object_sid(33554499, Scenery_Creation_Hex, 0, -1);
+                Scenery_Creation := create_object(33554499, Scenery_Creation_Hex, 0);
                 Scenery_Creation_Count := Scenery_Creation_Count + 1;
                 debug_msg("Putting Blocking Hex at " + Scenery_Creation_Hex);
             end
             Scenery_Creation_Count := 0;
             while (Scenery_Creation_Count < 1) do begin
                 Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 4, 1);
-                Scenery_Creation := create_object_sid(33554499, Scenery_Creation_Hex, 0, -1);
+                Scenery_Creation := create_object(33554499, Scenery_Creation_Hex, 0);
                 Scenery_Creation_Count := Scenery_Creation_Count + 1;
                 debug_msg("Putting Blocking Hex at " + Scenery_Creation_Hex);
             end
             Scenery_Creation_Count := 0;
             while (Scenery_Creation_Count < 1) do begin
                 Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 0, 1);
-                Scenery_Creation := create_object_sid(33554499, Scenery_Creation_Hex, 0, -1);
+                Scenery_Creation := create_object(33554499, Scenery_Creation_Hex, 0);
                 Scenery_Creation_Count := Scenery_Creation_Count + 1;
                 debug_msg("Putting Blocking Hex at " + Scenery_Creation_Hex);
             end
             Scenery_Creation_Count := 0;
             while (Scenery_Creation_Count < 1) do begin
                 Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 5, 1);
-                Scenery_Creation := create_object_sid(33554499, Scenery_Creation_Hex, 0, -1);
+                Scenery_Creation := create_object(33554499, Scenery_Creation_Hex, 0);
                 Scenery_Creation_Count := Scenery_Creation_Count + 1;
                 debug_msg("Putting Blocking Hex at " + Scenery_Creation_Hex);
             end
             Scenery_Creation_Count := 0;
             while (Scenery_Creation_Count < 2) do begin
                 Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 0, 1);
-                Scenery_Creation := create_object_sid(33554499, Scenery_Creation_Hex, 0, -1);
+                Scenery_Creation := create_object(33554499, Scenery_Creation_Hex, 0);
                 Scenery_Creation_Count := Scenery_Creation_Count + 1;
                 debug_msg("Putting Blocking Hex at " + Scenery_Creation_Hex);
             end
             Scenery_Creation_Count := 0;
             while (Scenery_Creation_Count < 1) do begin
                 Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 1, 1);
-                Scenery_Creation := create_object_sid(33554499, Scenery_Creation_Hex, 0, -1);
+                Scenery_Creation := create_object(33554499, Scenery_Creation_Hex, 0);
                 Scenery_Creation_Count := Scenery_Creation_Count + 1;
                 debug_msg("Putting Blocking Hex at " + Scenery_Creation_Hex);
             end
             Scenery_Creation_Count := 0;
             while (Scenery_Creation_Count < 1) do begin
                 Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 2, 1);
-                Scenery_Creation := create_object_sid(33554499, Scenery_Creation_Hex, 0, -1);
+                Scenery_Creation := create_object(33554499, Scenery_Creation_Hex, 0);
                 Scenery_Creation_Count := Scenery_Creation_Count + 1;
                 debug_msg("Putting Blocking Hex at " + Scenery_Creation_Hex);
             end
@@ -247,28 +247,28 @@ begin
                 move_to(party_member_obj(455), Scenery_Creation_Hex, 0);
             end
             else begin
-                create_object_sid(455, Scenery_Creation_Hex, 0, 920);
+                create_object_sid(455, Scenery_Creation_Hex, 0, SCRIPT_ZICRTRNK);
             end
             Scenery_Creation_Count := 0;
             while (Scenery_Creation_Count < 1) do begin
                 Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 1, 1);
-                Scenery_Creation := create_object_sid(33554499, Scenery_Creation_Hex, 0, -1);
+                Scenery_Creation := create_object(33554499, Scenery_Creation_Hex, 0);
                 Scenery_Creation_Count := Scenery_Creation_Count + 1;
                 debug_msg("Putting Blocking Hex at " + Scenery_Creation_Hex);
             end
             Scenery_Creation_Count := 0;
             while (Scenery_Creation_Count < 1) do begin
                 Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 2, 1);
-                Scenery_Creation := create_object_sid(33554499, Scenery_Creation_Hex, 0, -1);
+                Scenery_Creation := create_object(33554499, Scenery_Creation_Hex, 0);
                 Scenery_Creation_Count := Scenery_Creation_Count + 1;
                 debug_msg("Putting Blocking Hex at " + Scenery_Creation_Hex);
             end
         end
     end
     else begin
-        if ((map_var(0) == 0) and (metarule(30, 0) != 11)) then begin
+        if ((map_var(0) == 0) and (car_current_town != 11)) then begin
             set_map_var(0, 1);
-            if (not(metarule(22, 0))) then begin
+            if (not(is_loading_game)) then begin
                 Scenery_Creation_Ptr := tile_contains_pid_obj(19500, 0, 33555441);
                 if (tile_contains_obj_pid(19500, 0, 33555441)) then begin
                     destroy_object(Scenery_Creation_Ptr);
@@ -413,16 +413,16 @@ begin
             end
         end
     end
-    if (not(metarule(22, 0))) then begin
-        if (global_var(213) == 1) then begin
-            set_global_var(213, 2);
-            if (metarule(17, 45) != 0) then begin
+    if (not(is_loading_game)) then begin
+        if (global_var(GVAR_NCR_FAKE_VAULT13_MAP) == 1) then begin
+            set_global_var(GVAR_NCR_FAKE_VAULT13_MAP, 2);
+            if (town_known(45) != MARK_STATE_UNKNOWN) then begin
                 debug_msg(" unmark_on_map(" + 45 + ")");
-                mark_area_known(0, 45, 0);
+                mark_area_known(MARK_TYPE_TOWN, 45, MARK_STATE_UNKNOWN);
             end
-            if (metarule(17, 46) == 0) then begin
+            if (town_known(46) == MARK_STATE_UNKNOWN) then begin
                 debug_msg("  mark_on_map(" + 46 + ")");
-                mark_area_known(0, 46, 1);
+                mark_area_known(MARK_TYPE_TOWN, 46, MARK_STATE_KNOWN);
             end
         end
     end
@@ -432,4 +432,3 @@ procedure map_update_p_proc
 begin
     set_light_level(50);
 end
-
