diff --git a/Games/Fallout2.Steam/Original/CICRISDR.ssl b/Games/Fallout2.Steam/ReDefine/CICRISDR.ssl
index 495aab5..4b7b701 100644
--- a/Games/Fallout2.Steam/Original/CICRISDR.ssl
+++ b/Games/Fallout2.Steam/ReDefine/CICRISDR.ssl
@@ -139,7 +139,7 @@ end
 
 procedure use_p_proc
 begin
-    if ((tile_num(dude_obj) == 23300) or (tile_num(dude_obj) == 23501) or (tile_num(dude_obj) == 23301) or (tile_num(dude_obj) == 23302)) then begin
+    if ((dude_tile == 23300) or (dude_tile == 23501) or (dude_tile == 23301) or (dude_tile == 23302)) then begin
         obj_unlock(self_obj);
     end
     else begin
@@ -166,4 +166,3 @@ begin
     obj_close(self_obj);
     obj_lock(self_obj);
 end
-
