diff --git a/Games/Fallout2.Steam/Original/COCDOOR.ssl b/Games/Fallout2.Steam/ReDefine/COCDOOR.ssl
index 0feee35..6a8f90b 100644
--- a/Games/Fallout2.Steam/Original/COCDOOR.ssl
+++ b/Games/Fallout2.Steam/ReDefine/COCDOOR.ssl
@@ -4,7 +4,7 @@ procedure use_p_proc;
 
 procedure start
 begin
-    if (script_action == 6) then begin
+    if (script_action == use_proc) then begin
         debug_msg("in check for use_proc");
         call use_p_proc();
     end
@@ -20,4 +20,3 @@ begin
         move_to(dude_obj, 22499, 1);
     end
 end
-
