diff --git a/Games/Fallout2.Steam/Original/CODEDISK.ssl b/Games/Fallout2.Steam/ReDefine/CODEDISK.ssl
index b8f95fe..0035183 100644
--- a/Games/Fallout2.Steam/Original/CODEDISK.ssl
+++ b/Games/Fallout2.Steam/ReDefine/CODEDISK.ssl
@@ -3,24 +3,23 @@ procedure start;
 
 procedure start
 begin
-    if ((script_action == 21) or (script_action == 3)) then begin
+    if ((script_action == look_at_proc) or (script_action == description_proc)) then begin
         script_overrides;
-        display_msg(message_str(350, 100));
+        display_msg(message_str(SCRIPT_NCWRITEE, 100));
     end
     else begin
-        if (script_action == 6) then begin
+        if (script_action == use_proc) then begin
             script_overrides;
             if (local_var(0) == 1) then begin
-                display_msg(message_str(350, 101));
+                display_msg(message_str(SCRIPT_NCWRITEE, 101));
             end
             else begin
                 set_local_var(0, 1);
-                display_msg(message_str(350, 102));
-                set_global_var(167, 1);
-                display_msg(message_str(766, 103) + "100" + message_str(766, 104));
+                display_msg(message_str(SCRIPT_NCWRITEE, 102));
+                set_global_var(GVAR_GECKO_SKEETER_STATUS, 1);
+                display_msg(message_str(SCRIPT_ECELRON, 103) + "100" + message_str(SCRIPT_ECELRON, 104));
                 give_exp_points(100);
             end
         end
     end
 end
-
