diff --git a/Games/Fallout2.Steam/Original/CTK9PART.ssl b/Games/Fallout2.Steam/ReDefine/CTK9PART.ssl
index 470bcf5..d98f6f6 100644
--- a/Games/Fallout2.Steam/Original/CTK9PART.ssl
+++ b/Games/Fallout2.Steam/ReDefine/CTK9PART.ssl
@@ -134,11 +134,10 @@ procedure spatial_p_proc
 begin
     if ((local_var(0) == 0) and (obj_is_carrying_obj_pid(dude_obj, 492) > 0)) then begin
         set_local_var(0, 1);
-        display_msg(message_str(1289, 100));
+        display_mstr(100);
     end
 end
 
 procedure start
 begin
 end
-
