diff --git a/Games/Fallout2.Steam/Original/DCCAESAR.ssl b/Games/Fallout2.Steam/ReDefine/DCCAESAR.ssl
index f355126..36891ee 100644
--- a/Games/Fallout2.Steam/Original/DCCAESAR.ssl
+++ b/Games/Fallout2.Steam/ReDefine/DCCAESAR.ssl
@@ -211,13 +211,13 @@ procedure critter_p_proc
 begin
     if (obj_can_see_obj(self_obj, dude_obj) and (local_var(4) == 2)) then begin
         set_local_var(4, 1);
-        attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+        attack(dude_obj);
         set_map_var(7, 1);
     end
     else begin
         if (critter_is_fleeing(self_obj)) then begin
             if ((tile_distance_objs(self_obj, dude_obj) < 8) or obj_can_see_obj(self_obj, dude_obj)) then begin
-                animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(self_obj), rotation_to_tile(tile_num(dude_obj), tile_num(self_obj)), random(6, 10)), 1);
+                animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(self_obj), rotation_to_tile(dude_tile, tile_num(self_obj)), random(6, 10)), 1);
             end
         end
         else begin
@@ -225,7 +225,7 @@ begin
                 set_map_var(7, 1);
             end
             else begin
-                if ((tile_distance_objs(self_obj, dude_obj) > 5) and (has_trait(1, self_obj, 10) != 2)) then begin
+                if ((tile_distance_objs(self_obj, dude_obj) > 5) and (obj_get_rot(self_obj) != 2)) then begin
                     anim(self_obj, 1000, 2);
                 end
             end
@@ -245,7 +245,7 @@ begin
     script_overrides;
     if (local_var(2) == 0) then begin
         set_local_var(2, 1);
-        set_local_var(3, (get_critter_stat(dude_obj, 3) - get_critter_stat(self_obj, 3)) * 5);
+        set_local_var(3, (dude_charisma - self_charisma) * 5);
         debug_msg("Base Reaction == " + local_var(3));
     end
     Static_Reaction := 0;
@@ -257,14 +257,14 @@ begin
     else begin
         reaction_bonus_karma := -1 * (-1 * reaction_bonus_karma / 1000);
     end
-    if (global_var(49) >= 0) then begin
-        reaction_bonus_town_rep := global_var(49) / 2;
+    if (global_var(GVAR_TOWN_REP_THE_DEN) >= 0) then begin
+        reaction_bonus_town_rep := global_var(GVAR_TOWN_REP_THE_DEN) / 2;
     end
     else begin
-        reaction_bonus_town_rep := -1 * (-1 * global_var(49) / 2);
+        reaction_bonus_town_rep := -1 * (-1 * global_var(GVAR_TOWN_REP_THE_DEN) / 2);
     end
-    if (has_trait(2, dude_obj, 13)) then begin
-        if (get_critter_stat(dude_obj, 34) == get_critter_stat(self_obj, 34)) then begin
+    if (has_trait(TRAIT_TRAIT, dude_obj, TRAIT_sex_appeal)) then begin
+        if (dude_gender == self_gender) then begin
             Static_Reaction := Static_Reaction + -20;
         end
         else begin
@@ -280,48 +280,48 @@ begin
         debug_msg("Karma Reaction Bonus == " + Static_Reaction);
         Static_Reaction := Static_Reaction - reaction_bonus_town_rep;
         debug_msg("Town Rep Bonus == " + Static_Reaction);
-        set_global_var(37, 0);
-        set_global_var(38, 0);
-        set_global_var(39, 0);
-        set_global_var(40, 0);
-        set_global_var(41, 0);
-        set_global_var(42, 0);
-        set_global_var(43, 0);
-        set_global_var(44, 0);
-        set_global_var(45, 0);
-        if (global_var(0) >= 1000) then begin
-            set_global_var(37, 1);
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+        set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+        set_global_var(GVAR_KARMA_DEFENDER, 0);
+        set_global_var(GVAR_KARMA_WANDERER, 0);
+        set_global_var(GVAR_KARMA_BETRAYER, 0);
+        set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+        set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+            set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
         end
         else begin
-            if (global_var(0) >= 750) then begin
-                set_global_var(38, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
             end
             else begin
-                if (global_var(0) >= 500) then begin
-                    set_global_var(39, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                 end
                 else begin
-                    if (global_var(0) >= 250) then begin
-                        set_global_var(40, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                        set_global_var(GVAR_KARMA_DEFENDER, 1);
                     end
                     else begin
-                        if (global_var(0) > -250) then begin
-                            set_global_var(41, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                            set_global_var(GVAR_KARMA_WANDERER, 1);
                         end
                         else begin
-                            if (global_var(0) > -500) then begin
-                                set_global_var(42, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                set_global_var(GVAR_KARMA_BETRAYER, 1);
                             end
                             else begin
-                                if (global_var(0) > -750) then begin
-                                    set_global_var(43, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -1000) then begin
-                                        set_global_var(44, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                     end
                                     else begin
-                                        set_global_var(45, 1);
+                                        set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                     end
                                 end
                             end
@@ -330,35 +330,35 @@ begin
                 end
             end
         end
-        if (global_var(37) == 1) then begin
+        if (global_var(GVAR_KARMA_HOLY_WARRIOR) == 1) then begin
             Static_Reaction := Static_Reaction - 20;
         end
         else begin
-            if (global_var(38) == 1) then begin
+            if (global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES) == 1) then begin
                 Static_Reaction := Static_Reaction - 15;
             end
             else begin
-                if (global_var(39) == 1) then begin
+                if (global_var(GVAR_KARMA_SHIELD_OF_HOPE) == 1) then begin
                     Static_Reaction := Static_Reaction - 10;
                 end
                 else begin
-                    if (global_var(40) == 1) then begin
+                    if (global_var(GVAR_KARMA_DEFENDER) == 1) then begin
                         Static_Reaction := Static_Reaction - 5;
                     end
                     else begin
-                        if (global_var(41) == 1) then begin
+                        if (global_var(GVAR_KARMA_WANDERER) == 1) then begin
                             Static_Reaction := Static_Reaction - 0;
                         end
                         else begin
-                            if (global_var(42) == 1) then begin
+                            if (global_var(GVAR_KARMA_BETRAYER) == 1) then begin
                                 Static_Reaction := Static_Reaction - -5;
                             end
                             else begin
-                                if (global_var(43) == 1) then begin
+                                if (global_var(GVAR_KARMA_SWORD_OF_DESPAIR) == 1) then begin
                                     Static_Reaction := Static_Reaction - -10;
                                 end
                                 else begin
-                                    if (global_var(44) == 1) then begin
+                                    if (global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE) == 1) then begin
                                         Static_Reaction := Static_Reaction - -15;
                                     end
                                     else begin
@@ -378,48 +378,48 @@ begin
         debug_msg("Karma Reaction Bonus == " + Static_Reaction);
         Static_Reaction := Static_Reaction + reaction_bonus_town_rep;
         debug_msg("Town Rep Bonus == " + Static_Reaction);
-        set_global_var(37, 0);
-        set_global_var(38, 0);
-        set_global_var(39, 0);
-        set_global_var(40, 0);
-        set_global_var(41, 0);
-        set_global_var(42, 0);
-        set_global_var(43, 0);
-        set_global_var(44, 0);
-        set_global_var(45, 0);
-        if (global_var(0) >= 1000) then begin
-            set_global_var(37, 1);
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+        set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+        set_global_var(GVAR_KARMA_DEFENDER, 0);
+        set_global_var(GVAR_KARMA_WANDERER, 0);
+        set_global_var(GVAR_KARMA_BETRAYER, 0);
+        set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+        set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+            set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
         end
         else begin
-            if (global_var(0) >= 750) then begin
-                set_global_var(38, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
             end
             else begin
-                if (global_var(0) >= 500) then begin
-                    set_global_var(39, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                 end
                 else begin
-                    if (global_var(0) >= 250) then begin
-                        set_global_var(40, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                        set_global_var(GVAR_KARMA_DEFENDER, 1);
                     end
                     else begin
-                        if (global_var(0) > -250) then begin
-                            set_global_var(41, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                            set_global_var(GVAR_KARMA_WANDERER, 1);
                         end
                         else begin
-                            if (global_var(0) > -500) then begin
-                                set_global_var(42, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                set_global_var(GVAR_KARMA_BETRAYER, 1);
                             end
                             else begin
-                                if (global_var(0) > -750) then begin
-                                    set_global_var(43, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -1000) then begin
-                                        set_global_var(44, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                     end
                                     else begin
-                                        set_global_var(45, 1);
+                                        set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                     end
                                 end
                             end
@@ -428,35 +428,35 @@ begin
                 end
             end
         end
-        if (global_var(37) == 1) then begin
+        if (global_var(GVAR_KARMA_HOLY_WARRIOR) == 1) then begin
             Static_Reaction := Static_Reaction + 20;
         end
         else begin
-            if (global_var(38) == 1) then begin
+            if (global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES) == 1) then begin
                 Static_Reaction := Static_Reaction + 15;
             end
             else begin
-                if (global_var(39) == 1) then begin
+                if (global_var(GVAR_KARMA_SHIELD_OF_HOPE) == 1) then begin
                     Static_Reaction := Static_Reaction + 10;
                 end
                 else begin
-                    if (global_var(40) == 1) then begin
+                    if (global_var(GVAR_KARMA_DEFENDER) == 1) then begin
                         Static_Reaction := Static_Reaction + 5;
                     end
                     else begin
-                        if (global_var(41) == 1) then begin
+                        if (global_var(GVAR_KARMA_WANDERER) == 1) then begin
                             Static_Reaction := Static_Reaction + 0;
                         end
                         else begin
-                            if (global_var(42) == 1) then begin
+                            if (global_var(GVAR_KARMA_BETRAYER) == 1) then begin
                                 Static_Reaction := Static_Reaction + -5;
                             end
                             else begin
-                                if (global_var(43) == 1) then begin
+                                if (global_var(GVAR_KARMA_SWORD_OF_DESPAIR) == 1) then begin
                                     Static_Reaction := Static_Reaction + -10;
                                 end
                                 else begin
-                                    if (global_var(44) == 1) then begin
+                                    if (global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE) == 1) then begin
                                         Static_Reaction := Static_Reaction + -15;
                                     end
                                     else begin
@@ -472,28 +472,28 @@ begin
         debug_msg("Good Person Reaction Bonus == " + Static_Reaction);
     end
     if (Slavery_Tolerant == 0) then begin
-        if (global_var(11) == 1) then begin
+        if (global_var(GVAR_REPUTATION_SLAVER) == 1) then begin
             Static_Reaction := Static_Reaction + (2 * -25);
         end
     end
     else begin
         if (Slavery_Tolerant == 1) then begin
-            if (global_var(11) == 1) then begin
+            if (global_var(GVAR_REPUTATION_SLAVER) == 1) then begin
                 Static_Reaction := Static_Reaction + -25;
             end
         end
         else begin
             if (Slavery_Tolerant == 3) then begin
-                if (global_var(11) == 1) then begin
+                if (global_var(GVAR_REPUTATION_SLAVER) == 1) then begin
                     Static_Reaction := Static_Reaction - -25;
                 end
             end
         end
     end
     debug_msg("Slaver + Aligned Reaction == " + Static_Reaction);
-    Static_Reaction := Static_Reaction + (10 * has_trait(0, dude_obj, 10));
+    Static_Reaction := Static_Reaction + (10 * has_trait(TRAIT_PERK, dude_obj, PERK_presence));
     debug_msg("Presence Reaction == " + Static_Reaction);
-    if (global_var(1) >= 2) then begin
+    if (global_var(GVAR_CHILDKILLER_REPUTATION) >= 2) then begin
         Static_Reaction := Static_Reaction + -30;
     end
     debug_msg("Childkiller Reaction == " + Static_Reaction);
@@ -501,15 +501,15 @@ begin
     debug_msg("Initial Reaction == " + local_var(0));
     start_gdialog(46, self_obj, 4, -1, -1);
     gsay_start;
-    if ((local_var(4) > 0) or global_var(14) or global_var(15) or ((global_var(13) bwand 2) != 0) or ((global_var(13) bwand 4) != 0)) then begin
+    if ((local_var(4) > 0) or global_var(GVAR_ENEMY_DEN) or global_var(GVAR_EXILE_DEN) or ((global_var(GVAR_DEN_MOM_STATUS) bwand 2) != 0) or ((global_var(GVAR_DEN_MOM_STATUS) bwand 4) != 0)) then begin
         call Node001();
     end
     else begin
-        if (global_var(30) == 2) then begin
+        if (global_var(GVAR_SLAVE_RUN) == 2) then begin
             call Node037();
         end
         else begin
-            if (global_var(30) != 0) then begin
+            if (global_var(GVAR_SLAVE_RUN) != 0) then begin
                 call Node039();
             end
             else begin
@@ -521,7 +521,7 @@ begin
                         call Node003();
                     end
                     else begin
-                        if (global_var(11) or (local_var(5) and (local_var(1) >= 2))) then begin
+                        if (global_var(GVAR_REPUTATION_SLAVER) or (local_var(5) and (local_var(1) >= 2))) then begin
                             call Node006();
                         end
                         else begin
@@ -535,66 +535,66 @@ begin
     gsay_end;
     end_dialogue;
     set_local_var(5, local_var(5) + 1);
-    set_global_var(298, global_var(298) bwor 1);
-    set_global_var(298, global_var(298) bwor 32);
+    set_global_var(GVAR_DEN_CEASAR_STATUS, global_var(GVAR_DEN_CEASAR_STATUS) bwor 1);
+    set_global_var(GVAR_DEN_CEASAR_STATUS, global_var(GVAR_DEN_CEASAR_STATUS) bwor 32);
 end
 
 procedure destroy_p_proc
 begin
     if (source_obj == dude_obj) then begin
-        set_global_var(4, global_var(4) + 1);
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(5) > (2 * global_var(4))) or (global_var(3) == 1))) then begin
-            set_global_var(3, 1);
-            set_global_var(2, 0);
+        set_global_var(GVAR_BAD_MONSTER, global_var(GVAR_BAD_MONSTER) + 1);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
+            set_global_var(GVAR_BERSERKER_REPUTATION, 1);
+            set_global_var(GVAR_CHAMPION_REPUTATION, 0);
         end
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(4) > (3 * global_var(5))) or (global_var(2) == 1)) and (global_var(1) == 0)) then begin
-            set_global_var(2, 1);
-            set_global_var(3, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(4) > (3 * global_var(GVAR_GOOD_MONSTER))) or (global_var(GVAR_CHAMPION_REPUTATION) == 1)) and (global_var(GVAR_CHILDKILLER_REPUTATION) == 0)) then begin
+            set_global_var(GVAR_CHAMPION_REPUTATION, 1);
+            set_global_var(GVAR_BERSERKER_REPUTATION, 0);
         end
-        set_global_var(0, global_var(0) + 5);
+        set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + 5);
         debug_msg("Player gains " + 5 + " Karma Points.");
-        set_global_var(37, 0);
-        set_global_var(38, 0);
-        set_global_var(39, 0);
-        set_global_var(40, 0);
-        set_global_var(41, 0);
-        set_global_var(42, 0);
-        set_global_var(43, 0);
-        set_global_var(44, 0);
-        set_global_var(45, 0);
-        if (global_var(0) >= 1000) then begin
-            set_global_var(37, 1);
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+        set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+        set_global_var(GVAR_KARMA_DEFENDER, 0);
+        set_global_var(GVAR_KARMA_WANDERER, 0);
+        set_global_var(GVAR_KARMA_BETRAYER, 0);
+        set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+        set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+            set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
         end
         else begin
-            if (global_var(0) >= 750) then begin
-                set_global_var(38, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
             end
             else begin
-                if (global_var(0) >= 500) then begin
-                    set_global_var(39, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                 end
                 else begin
-                    if (global_var(0) >= 250) then begin
-                        set_global_var(40, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                        set_global_var(GVAR_KARMA_DEFENDER, 1);
                     end
                     else begin
-                        if (global_var(0) > -250) then begin
-                            set_global_var(41, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                            set_global_var(GVAR_KARMA_WANDERER, 1);
                         end
                         else begin
-                            if (global_var(0) > -500) then begin
-                                set_global_var(42, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                set_global_var(GVAR_KARMA_BETRAYER, 1);
                             end
                             else begin
-                                if (global_var(0) > -750) then begin
-                                    set_global_var(43, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -1000) then begin
-                                        set_global_var(44, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                     end
                                     else begin
-                                        set_global_var(45, 1);
+                                        set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                     end
                                 end
                             end
@@ -603,25 +603,25 @@ begin
                 end
             end
         end
-        set_global_var(49, global_var(49) + 2);
+        set_global_var(GVAR_TOWN_REP_THE_DEN, global_var(GVAR_TOWN_REP_THE_DEN) + 2);
         debug_msg("Added " + 2 + " to Town Rep");
     end
     if (source_obj == dude_obj) then begin
-        set_global_var(298, global_var(298) bwor 4);
+        set_global_var(GVAR_DEN_CEASAR_STATUS, global_var(GVAR_DEN_CEASAR_STATUS) bwor 4);
     end
-    set_global_var(298, global_var(298) bwor 512);
+    set_global_var(GVAR_DEN_CEASAR_STATUS, global_var(GVAR_DEN_CEASAR_STATUS) bwor 512);
 end
 
 procedure look_at_p_proc
 begin
     script_overrides;
-    display_msg(message_str(46, 100));
+    display_msg(message_str(SCRIPT_DCMETZGE, 100));
 end
 
 procedure description_p_proc
 begin
     script_overrides;
-    display_msg(message_str(46, 150));
+    display_msg(message_str(SCRIPT_DCMETZGE, 150));
 end
 
 procedure use_skill_on_p_proc
@@ -632,7 +632,7 @@ procedure damage_p_proc
 begin
     if (source_obj == dude_obj) then begin
         call Node998();
-        set_global_var(298, global_var(298) bwor 2);
+        set_global_var(GVAR_DEN_CEASAR_STATUS, global_var(GVAR_DEN_CEASAR_STATUS) bwor 2);
     end
 end
 
@@ -640,8 +640,8 @@ procedure map_enter_p_proc
 begin
     if (only_once == 0) then begin
         only_once := 1;
-        critter_add_trait(self_obj, 1, 6, 15);
-        critter_add_trait(self_obj, 1, 5, 16);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 15);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_SLAVER);
         if (local_var(4) == 1) then begin
             set_local_var(4, 2);
         end
@@ -698,33 +698,33 @@ begin
             end
         end
     end
-    gsay_reply(46, 152);
-    giq_option(1, 46, 153, Node998, 50);
+    gsay_reply(SCRIPT_DCMETZGE, 152);
+    giq_option(1, SCRIPT_DCMETZGE, 153, Node998, NEUTRAL_REACTION);
 end
 
 procedure Node002
 begin
-    gsay_reply(46, 154);
-    giq_option(1, 46, 155, Node999, 50);
+    gsay_reply(SCRIPT_DCMETZGE, 154);
+    giq_option(1, SCRIPT_DCMETZGE, 155, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node003
 begin
-    gsay_reply(46, 156);
-    giq_option(4, 46, 157, Node996, 50);
-    giq_option(5, 46, 158, Node995, 50);
-    giq_option(4, 46, 159, Node001, 50);
-    giq_option(4, 46, 160, Node008, 50);
-    giq_option(4, 46, 161, Node999, 50);
-    giq_option(-3, 46, 162, Node999, 50);
+    gsay_reply(SCRIPT_DCMETZGE, 156);
+    giq_option(4, SCRIPT_DCMETZGE, 157, Node996, NEUTRAL_REACTION);
+    giq_option(5, SCRIPT_DCMETZGE, 158, Node995, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_DCMETZGE, 159, Node001, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_DCMETZGE, 160, Node008, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_DCMETZGE, 161, Node999, NEUTRAL_REACTION);
+    giq_option(-3, SCRIPT_DCMETZGE, 162, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node004
 begin
-    gsay_reply(46, 163);
-    giq_option(4, 46, 164, Node994, 50);
-    giq_option(4, 46, 165, Node999, 50);
-    giq_option(4, 46, 166, Node993, 50);
+    gsay_reply(SCRIPT_DCMETZGE, 163);
+    giq_option(4, SCRIPT_DCMETZGE, 164, Node994, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_DCMETZGE, 165, Node999, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_DCMETZGE, 166, Node993, NEUTRAL_REACTION);
 end
 
 procedure Node005
@@ -771,25 +771,25 @@ begin
             end
         end
     end
-    gsay_reply(46, 167);
-    giq_option(4, 46, 168, Node009, 50);
-    giq_option(4, 46, 169, Node999, 50);
+    gsay_reply(SCRIPT_DCMETZGE, 167);
+    giq_option(4, SCRIPT_DCMETZGE, 168, Node009, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_DCMETZGE, 169, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node006
 begin
-    gsay_reply(46, 170);
-    giq_option(4, 46, 171, Node009, 50);
-    giq_option(4, 46, 172, Node999, 50);
-    giq_option(-3, 46, 173, Node999, 50);
+    gsay_reply(SCRIPT_DCMETZGE, 170);
+    giq_option(4, SCRIPT_DCMETZGE, 171, Node009, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_DCMETZGE, 172, Node999, NEUTRAL_REACTION);
+    giq_option(-3, SCRIPT_DCMETZGE, 173, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node007
 begin
-    gsay_reply(46, 174);
-    giq_option(4, 46, 175, Node009, 50);
-    giq_option(4, 46, 176, Node999, 50);
-    giq_option(-3, 46, 177, Node999, 50);
+    gsay_reply(SCRIPT_DCMETZGE, 174);
+    giq_option(4, SCRIPT_DCMETZGE, 175, Node009, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_DCMETZGE, 176, Node999, NEUTRAL_REACTION);
+    giq_option(-3, SCRIPT_DCMETZGE, 177, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node008
@@ -837,131 +837,131 @@ begin
             end
         end
     end
-    gsay_reply(46, 178);
-    giq_option(4, 46, 179, Node001, 50);
-    giq_option(4, 46, 180, Node991, 50);
-    giq_option(4, 46, 181, Node999, 50);
+    gsay_reply(SCRIPT_DCMETZGE, 178);
+    giq_option(4, SCRIPT_DCMETZGE, 179, Node001, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_DCMETZGE, 180, Node991, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_DCMETZGE, 181, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node009
 begin
-    gsay_reply(46, 182);
-    if (((global_var(19) bwand 1) != 0) and ((party_member_obj(16777278) != 0) == 0) and ((((global_var(19) bwand 128) != 0) or ((global_var(298) bwand 512) != 0)) == 0)) then begin
-        giq_option(4, 46, 183, Node024, 50);
+    gsay_reply(SCRIPT_DCMETZGE, 182);
+    if (((global_var(GVAR_DEN_VIC_STATUS) bwand 1) != 0) and ((party_member_obj(16777278) != 0) == 0) and ((((global_var(GVAR_DEN_VIC_STATUS) bwand 128) != 0) or ((global_var(GVAR_DEN_CEASAR_STATUS) bwand 512) != 0)) == 0)) then begin
+        giq_option(4, SCRIPT_DCMETZGE, 183, Node024, NEUTRAL_REACTION);
     end
-    if (global_var(11) == 0) then begin
-        giq_option(4, 46, 184, Node013, 50);
+    if (global_var(GVAR_REPUTATION_SLAVER) == 0) then begin
+        giq_option(4, SCRIPT_DCMETZGE, 184, Node013, NEUTRAL_REACTION);
     end
     if ((party_member_obj(16777379) != 0) or (party_member_obj(16777380) != 0)) then begin
         if (party_member_obj(16777379) != 0) then begin
-            giq_option(4, 46, 185, Node018, 50);
+            giq_option(4, SCRIPT_DCMETZGE, 185, Node018, NEUTRAL_REACTION);
         end
         if (party_member_obj(16777380) != 0) then begin
-            giq_option(4, 46, 186, Node018, 50);
+            giq_option(4, SCRIPT_DCMETZGE, 186, Node018, NEUTRAL_REACTION);
         end
     end
     if ((metarule(16, 0) - ((party_member_obj(16777379) != 0) or (party_member_obj(16777380) != 0))) > 2) then begin
-        giq_option(4, 46, 187, Node021, 50);
+        giq_option(4, SCRIPT_DCMETZGE, 187, Node021, NEUTRAL_REACTION);
     end
-    if (global_var(11) == 0) then begin
-        giq_option(4, 46, 188, Node011, 50);
+    if (global_var(GVAR_REPUTATION_SLAVER) == 0) then begin
+        giq_option(4, SCRIPT_DCMETZGE, 188, Node011, NEUTRAL_REACTION);
     end
     else begin
-        giq_option(4, 46, 189, Node016, 50);
-        giq_option(4, 46, 330, Node038, 50);
+        giq_option(4, SCRIPT_DCMETZGE, 189, Node016, NEUTRAL_REACTION);
+        giq_option(4, SCRIPT_DCMETZGE, 330, Node038, NEUTRAL_REACTION);
     end
-    if ((metarule(17, 11) != 0) == 0) then begin
-        giq_option(4, 46, 191, Node015, 50);
+    if ((town_known(11) != 0) == 0) then begin
+        giq_option(4, SCRIPT_DCMETZGE, 191, Node015, NEUTRAL_REACTION);
     end
-    giq_option(4, 46, 193, Node999, 50);
+    giq_option(4, SCRIPT_DCMETZGE, 193, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node010
 begin
-    gsay_reply(46, random(195, 201));
-    giq_option(4, 46, 202, Node009, 50);
-    giq_option(4, 46, 203, Node008, 50);
-    giq_option(4, 46, 204, Node999, 50);
+    gsay_reply(SCRIPT_DCMETZGE, random(195, 201));
+    giq_option(4, SCRIPT_DCMETZGE, 202, Node009, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_DCMETZGE, 203, Node008, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_DCMETZGE, 204, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node011
 begin
-    gsay_reply(46, 205);
-    giq_option(4, 46, 206, Node013, 50);
-    giq_option(4, 46, 207, Node008, 50);
-    giq_option(4, 46, 208, Node012, 50);
+    gsay_reply(SCRIPT_DCMETZGE, 205);
+    giq_option(4, SCRIPT_DCMETZGE, 206, Node013, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_DCMETZGE, 207, Node008, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_DCMETZGE, 208, Node012, NEUTRAL_REACTION);
 end
 
 procedure Node012
 begin
-    gsay_reply(46, 209);
-    giq_option(4, 46, 210, Node013, 50);
-    giq_option(4, 46, 211, Node008, 50);
-    giq_option(4, 46, 212, Node009, 50);
-    giq_option(4, 46, 213, Node999, 50);
+    gsay_reply(SCRIPT_DCMETZGE, 209);
+    giq_option(4, SCRIPT_DCMETZGE, 210, Node013, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_DCMETZGE, 211, Node008, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_DCMETZGE, 212, Node009, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_DCMETZGE, 213, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node013
 begin
-    gsay_reply(46, 214);
-    giq_option(4, 46, 215, Node014, 50);
-    giq_option(4, 46, 216, Node009, 50);
-    giq_option(4, 46, 217, Node999, 50);
+    gsay_reply(SCRIPT_DCMETZGE, 214);
+    giq_option(4, SCRIPT_DCMETZGE, 215, Node014, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_DCMETZGE, 216, Node009, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_DCMETZGE, 217, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node014
 begin
-    gfade_out(1);
-    anim(self_obj, 1000, rotation_to_tile(tile_num(self_obj), tile_num(dude_obj)));
-    set_global_var(11, 1);
-    set_global_var(287, 0);
+    fade_out;
+    anim(self_obj, 1000, rotation_to_tile(tile_num(self_obj), dude_tile));
+    set_global_var(GVAR_REPUTATION_SLAVER, 1);
+    set_global_var(GVAR_DEN_SLAVER_WARNINGS, 0);
     gfade_in(10);
-    gsay_reply(46, 218);
-    giq_option(4, 46, 219, Node009, 50);
-    giq_option(4, 46, 220, Node999, 50);
+    gsay_reply(SCRIPT_DCMETZGE, 218);
+    giq_option(4, SCRIPT_DCMETZGE, 219, Node009, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_DCMETZGE, 220, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node015
 begin
-    gsay_reply(46, 221);
-    giq_option(4, 46, 222, Node990, 50);
-    giq_option(4, 46, 223, Node009, 50);
-    giq_option(4, 46, 224, Node999, 50);
+    gsay_reply(SCRIPT_DCMETZGE, 221);
+    giq_option(4, SCRIPT_DCMETZGE, 222, Node990, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_DCMETZGE, 223, Node009, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_DCMETZGE, 224, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node016
 begin
-    gsay_reply(46, 225);
-    if (metarule(17, 7) == 0) then begin
+    gsay_reply(SCRIPT_DCMETZGE, 225);
+    if (town_known(7) == 0) then begin
         debug_msg("  mark_on_map(" + 7 + ")");
         mark_area_known(0, 7, 1);
     end
-    if ((metarule(17, 4) != 0) == 0) then begin
-        giq_option(4, 46, 226, Node017, 50);
+    if ((town_known(4) != 0) == 0) then begin
+        giq_option(4, SCRIPT_DCMETZGE, 226, Node017, NEUTRAL_REACTION);
     end
-    giq_option(4, 46, 227, Node009, 50);
-    giq_option(4, 46, 228, Node999, 50);
+    giq_option(4, SCRIPT_DCMETZGE, 227, Node009, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_DCMETZGE, 228, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node017
 begin
-    gsay_reply(46, 229);
-    if (metarule(17, 4) == 0) then begin
+    gsay_reply(SCRIPT_DCMETZGE, 229);
+    if (town_known(4) == 0) then begin
         debug_msg("  mark_on_map(" + 4 + ")");
         mark_area_known(0, 4, 1);
     end
-    giq_option(4, 46, 230, Node009, 50);
-    giq_option(4, 46, 231, Node999, 50);
+    giq_option(4, SCRIPT_DCMETZGE, 230, Node009, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_DCMETZGE, 231, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node018
 begin
     slave_value := 0;
     slave_value := slave_value + 100;
-    gsay_reply(46, message_str(46, 232) + slave_value + message_str(46, 233));
-    giq_option(4, 46, 234, Node989, 50);
-    giq_option(4, 46, 235, Node020, 50);
-    giq_option(4, 46, 236, Node988, 50);
+    gsay_reply(SCRIPT_DCMETZGE, message_str(SCRIPT_DCMETZGE, 232) + slave_value + message_str(SCRIPT_DCMETZGE, 233));
+    giq_option(4, SCRIPT_DCMETZGE, 234, Node989, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_DCMETZGE, 235, Node020, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_DCMETZGE, 236, Node988, NEUTRAL_REACTION);
 end
 
 procedure Node019
@@ -969,50 +969,50 @@ begin
     slave_value := slave_value * 2;
     item_caps_adjust(self_obj, slave_value);
     item_caps_adjust(dude_obj, -slave_value);
-    set_global_var(0, global_var(0) + -20);
+    set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + -20);
     debug_msg("Player gains " + -20 + " Karma Points.");
-    set_global_var(37, 0);
-    set_global_var(38, 0);
-    set_global_var(39, 0);
-    set_global_var(40, 0);
-    set_global_var(41, 0);
-    set_global_var(42, 0);
-    set_global_var(43, 0);
-    set_global_var(44, 0);
-    set_global_var(45, 0);
-    if (global_var(0) >= 1000) then begin
-        set_global_var(37, 1);
+    set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+    set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+    set_global_var(GVAR_KARMA_DEFENDER, 0);
+    set_global_var(GVAR_KARMA_WANDERER, 0);
+    set_global_var(GVAR_KARMA_BETRAYER, 0);
+    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+    set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+    set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+    if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
     end
     else begin
-        if (global_var(0) >= 750) then begin
-            set_global_var(38, 1);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+            set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
         end
         else begin
-            if (global_var(0) >= 500) then begin
-                set_global_var(39, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
             end
             else begin
-                if (global_var(0) >= 250) then begin
-                    set_global_var(40, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                    set_global_var(GVAR_KARMA_DEFENDER, 1);
                 end
                 else begin
-                    if (global_var(0) > -250) then begin
-                        set_global_var(41, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                        set_global_var(GVAR_KARMA_WANDERER, 1);
                     end
                     else begin
-                        if (global_var(0) > -500) then begin
-                            set_global_var(42, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                            set_global_var(GVAR_KARMA_BETRAYER, 1);
                         end
                         else begin
-                            if (global_var(0) > -750) then begin
-                                set_global_var(43, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                             end
                             else begin
-                                if (global_var(0) > -1000) then begin
-                                    set_global_var(44, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                    set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                 end
                                 else begin
-                                    set_global_var(45, 1);
+                                    set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                 end
                             end
                         end
@@ -1021,86 +1021,86 @@ begin
             end
         end
     end
-    gsay_reply(46, 237);
-    giq_option(4, 46, 238, Node009, 50);
-    giq_option(4, 46, 239, Node999, 50);
+    gsay_reply(SCRIPT_DCMETZGE, 237);
+    giq_option(4, SCRIPT_DCMETZGE, 238, Node009, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_DCMETZGE, 239, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node020
 begin
-    gsay_reply(46, 240);
-    giq_option(4, 46, 241, Node009, 50);
-    giq_option(4, 46, 242, Node999, 50);
+    gsay_reply(SCRIPT_DCMETZGE, 240);
+    giq_option(4, SCRIPT_DCMETZGE, 241, Node009, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_DCMETZGE, 242, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node021
 begin
-    gsay_reply(46, 243);
-    giq_option(4, 46, 244, Node020, 50);
+    gsay_reply(SCRIPT_DCMETZGE, 243);
+    giq_option(4, SCRIPT_DCMETZGE, 244, Node020, NEUTRAL_REACTION);
 end
 
 procedure Node022
 begin
     slave_value := 0;
-    gsay_reply(46, message_str(46, 245) + slave_value + message_str(46, 246));
-    giq_option(4, 46, 247, Node987, 50);
-    giq_option(4, 46, 248, Node020, 50);
-    giq_option(4, 46, 249, Node986, 50);
+    gsay_reply(SCRIPT_DCMETZGE, message_str(SCRIPT_DCMETZGE, 245) + slave_value + message_str(SCRIPT_DCMETZGE, 246));
+    giq_option(4, SCRIPT_DCMETZGE, 247, Node987, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_DCMETZGE, 248, Node020, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_DCMETZGE, 249, Node986, NEUTRAL_REACTION);
 end
 
 procedure Node023
 begin
     slave_value := slave_value * 2;
-    gsay_reply(46, message_str(46, 250) + slave_value + message_str(46, 251));
-    giq_option(4, 46, 252, Node969, 50);
-    giq_option(4, 46, 253, Node985, 50);
-    giq_option(4, 46, 254, Node984, 50);
+    gsay_reply(SCRIPT_DCMETZGE, message_str(SCRIPT_DCMETZGE, 250) + slave_value + message_str(SCRIPT_DCMETZGE, 251));
+    giq_option(4, SCRIPT_DCMETZGE, 252, Node969, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_DCMETZGE, 253, Node985, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_DCMETZGE, 254, Node984, NEUTRAL_REACTION);
 end
 
 procedure Node024
 begin
-    gsay_reply(46, 255);
-    giq_option(4, 46, 256, Node983, 50);
-    if ((((game_time - global_var(29)) >= (24 * (60 * (60 * 10)))) and (global_var(29) != 0)) == 0) then begin
-        giq_option(4, 46, 257, Node028, 50);
+    gsay_reply(SCRIPT_DCMETZGE, 255);
+    giq_option(4, SCRIPT_DCMETZGE, 256, Node983, NEUTRAL_REACTION);
+    if ((((game_time - global_var(GVAR_VIC_DEVICE)) >= (24 * (60 * (60 * 10)))) and (global_var(GVAR_VIC_DEVICE) != 0)) == 0) then begin
+        giq_option(4, SCRIPT_DCMETZGE, 257, Node028, NEUTRAL_REACTION);
     end
-    giq_option(4, 46, 258, Node020, 50);
+    giq_option(4, SCRIPT_DCMETZGE, 258, Node020, NEUTRAL_REACTION);
 end
 
 procedure Node025
 begin
-    gsay_reply(46, 259);
-    giq_option(4, 46, 260, Node028, 50);
-    giq_option(4, 46, 261, Node020, 50);
+    gsay_reply(SCRIPT_DCMETZGE, 259);
+    giq_option(4, SCRIPT_DCMETZGE, 260, Node028, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_DCMETZGE, 261, Node020, NEUTRAL_REACTION);
 end
 
 procedure Node026
 begin
     slave_value := 2000;
-    gsay_reply(46, 262);
-    giq_option(4, 46, 263, Node982, 50);
-    giq_option(4, 46, 264, Node020, 50);
+    gsay_reply(SCRIPT_DCMETZGE, 262);
+    giq_option(4, SCRIPT_DCMETZGE, 263, Node982, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_DCMETZGE, 264, Node020, NEUTRAL_REACTION);
 end
 
 procedure Node027
 begin
     slave_value := 1000;
-    gsay_reply(46, 265);
-    giq_option(4, 46, 266, Node982, 50);
-    giq_option(4, 46, 267, Node020, 50);
+    gsay_reply(SCRIPT_DCMETZGE, 265);
+    giq_option(4, SCRIPT_DCMETZGE, 266, Node982, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_DCMETZGE, 267, Node020, NEUTRAL_REACTION);
 end
 
 procedure Node028
 begin
-    gsay_reply(46, 268);
-    giq_option(4, 46, 269, Node020, 50);
+    gsay_reply(SCRIPT_DCMETZGE, 268);
+    giq_option(4, SCRIPT_DCMETZGE, 269, Node020, NEUTRAL_REACTION);
 end
 
 procedure Node029
 begin
-    gsay_reply(46, 270);
-    giq_option(4, 46, 271, Node009, 50);
-    giq_option(4, 46, 272, Node999, 50);
+    gsay_reply(SCRIPT_DCMETZGE, 270);
+    giq_option(4, SCRIPT_DCMETZGE, 271, Node009, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_DCMETZGE, 272, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node037
@@ -1147,23 +1147,23 @@ begin
             end
         end
     end
-    set_global_var(30, 0);
-    gsay_reply(46, 307);
-    giq_option(4, 46, 308, Node998, 50);
+    set_global_var(GVAR_SLAVE_RUN, 0);
+    gsay_reply(SCRIPT_DCMETZGE, 307);
+    giq_option(4, SCRIPT_DCMETZGE, 308, Node998, NEUTRAL_REACTION);
 end
 
 procedure Node038
 begin
-    gsay_reply(46, 309);
-    giq_option(4, 46, 311, Node968, 50);
-    giq_option(4, 46, 310, Node020, 50);
+    gsay_reply(SCRIPT_DCMETZGE, 309);
+    giq_option(4, SCRIPT_DCMETZGE, 311, Node968, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_DCMETZGE, 310, Node020, NEUTRAL_REACTION);
 end
 
 procedure Node039
 begin
-    set_global_var(30, 0);
-    gsay_reply(46, 312);
-    giq_option(4, 46, 313, Node967, 50);
+    set_global_var(GVAR_SLAVE_RUN, 0);
+    gsay_reply(SCRIPT_DCMETZGE, 312);
+    giq_option(4, SCRIPT_DCMETZGE, 313, Node967, NEUTRAL_REACTION);
 end
 
 procedure Node040
@@ -1211,8 +1211,8 @@ begin
             end
         end
     end
-    gsay_reply(46, 331);
-    giq_option(4, 46, 314, Node966, 50);
+    gsay_reply(SCRIPT_DCMETZGE, 331);
+    giq_option(4, SCRIPT_DCMETZGE, 314, Node966, NEUTRAL_REACTION);
 end
 
 procedure Node041
@@ -1260,56 +1260,56 @@ begin
             end
         end
     end
-    gsay_reply(46, 315);
-    giq_option(4, 46, 316, Node965, 50);
+    gsay_reply(SCRIPT_DCMETZGE, 315);
+    giq_option(4, SCRIPT_DCMETZGE, 316, Node965, NEUTRAL_REACTION);
 end
 
 procedure Node042
 begin
-    set_global_var(0, global_var(0) + 5);
+    set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + 5);
     debug_msg("Player gains " + 5 + " Karma Points.");
-    set_global_var(37, 0);
-    set_global_var(38, 0);
-    set_global_var(39, 0);
-    set_global_var(40, 0);
-    set_global_var(41, 0);
-    set_global_var(42, 0);
-    set_global_var(43, 0);
-    set_global_var(44, 0);
-    set_global_var(45, 0);
-    if (global_var(0) >= 1000) then begin
-        set_global_var(37, 1);
+    set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+    set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+    set_global_var(GVAR_KARMA_DEFENDER, 0);
+    set_global_var(GVAR_KARMA_WANDERER, 0);
+    set_global_var(GVAR_KARMA_BETRAYER, 0);
+    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+    set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+    set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+    if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
     end
     else begin
-        if (global_var(0) >= 750) then begin
-            set_global_var(38, 1);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+            set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
         end
         else begin
-            if (global_var(0) >= 500) then begin
-                set_global_var(39, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
             end
             else begin
-                if (global_var(0) >= 250) then begin
-                    set_global_var(40, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                    set_global_var(GVAR_KARMA_DEFENDER, 1);
                 end
                 else begin
-                    if (global_var(0) > -250) then begin
-                        set_global_var(41, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                        set_global_var(GVAR_KARMA_WANDERER, 1);
                     end
                     else begin
-                        if (global_var(0) > -500) then begin
-                            set_global_var(42, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                            set_global_var(GVAR_KARMA_BETRAYER, 1);
                         end
                         else begin
-                            if (global_var(0) > -750) then begin
-                                set_global_var(43, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                             end
                             else begin
-                                if (global_var(0) > -1000) then begin
-                                    set_global_var(44, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                    set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                 end
                                 else begin
-                                    set_global_var(45, 1);
+                                    set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                 end
                             end
                         end
@@ -1318,8 +1318,8 @@ begin
             end
         end
     end
-    gsay_reply(46, 317);
-    giq_option(4, 46, 318, Node964, 50);
+    gsay_reply(SCRIPT_DCMETZGE, 317);
+    giq_option(4, SCRIPT_DCMETZGE, 318, Node964, NEUTRAL_REACTION);
 end
 
 procedure Node043
@@ -1368,50 +1368,50 @@ begin
             end
         end
     end
-    set_global_var(0, global_var(0) + -30);
+    set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + -30);
     debug_msg("Player gains " + -30 + " Karma Points.");
-    set_global_var(37, 0);
-    set_global_var(38, 0);
-    set_global_var(39, 0);
-    set_global_var(40, 0);
-    set_global_var(41, 0);
-    set_global_var(42, 0);
-    set_global_var(43, 0);
-    set_global_var(44, 0);
-    set_global_var(45, 0);
-    if (global_var(0) >= 1000) then begin
-        set_global_var(37, 1);
+    set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+    set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+    set_global_var(GVAR_KARMA_DEFENDER, 0);
+    set_global_var(GVAR_KARMA_WANDERER, 0);
+    set_global_var(GVAR_KARMA_BETRAYER, 0);
+    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+    set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+    set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+    if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
     end
     else begin
-        if (global_var(0) >= 750) then begin
-            set_global_var(38, 1);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+            set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
         end
         else begin
-            if (global_var(0) >= 500) then begin
-                set_global_var(39, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
             end
             else begin
-                if (global_var(0) >= 250) then begin
-                    set_global_var(40, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                    set_global_var(GVAR_KARMA_DEFENDER, 1);
                 end
                 else begin
-                    if (global_var(0) > -250) then begin
-                        set_global_var(41, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                        set_global_var(GVAR_KARMA_WANDERER, 1);
                     end
                     else begin
-                        if (global_var(0) > -500) then begin
-                            set_global_var(42, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                            set_global_var(GVAR_KARMA_BETRAYER, 1);
                         end
                         else begin
-                            if (global_var(0) > -750) then begin
-                                set_global_var(43, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                             end
                             else begin
-                                if (global_var(0) > -1000) then begin
-                                    set_global_var(44, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                    set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                 end
                                 else begin
-                                    set_global_var(45, 1);
+                                    set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                 end
                             end
                         end
@@ -1420,24 +1420,24 @@ begin
             end
         end
     end
-    LVar0 := 100 * global_var(31);
+    LVar0 := 100 * global_var(GVAR_SLAVES_COUNT);
     if (LVar0 > 0) then begin
         item_caps_adjust(dude_obj, LVar0);
-        gsay_reply(46, message_str(46, 332) + LVar0 + message_str(46, 333));
+        gsay_reply(SCRIPT_DCMETZGE, message_str(SCRIPT_DCMETZGE, 332) + LVar0 + message_str(SCRIPT_DCMETZGE, 333));
     end
     else begin
-        gsay_reply(46, 319);
+        gsay_reply(SCRIPT_DCMETZGE, 319);
     end
-    giq_option(4, 46, 320, Node020, 50);
-    giq_option(4, 46, 321, Node963, 50);
+    giq_option(4, SCRIPT_DCMETZGE, 320, Node020, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_DCMETZGE, 321, Node963, NEUTRAL_REACTION);
 end
 
 procedure Node044
 begin
-    gsay_reply(46, 322);
-    giq_option(4, 46, 323, Node962, 50);
-    giq_option(4, 46, 324, Node961, 50);
-    giq_option(4, 46, 325, Node963, 50);
+    gsay_reply(SCRIPT_DCMETZGE, 322);
+    giq_option(4, SCRIPT_DCMETZGE, 323, Node962, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_DCMETZGE, 324, Node961, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_DCMETZGE, 325, Node963, NEUTRAL_REACTION);
 end
 
 procedure Node045
@@ -1485,10 +1485,10 @@ begin
             end
         end
     end
-    gsay_reply(46, 326);
-    giq_option(4, 46, 327, Node961, 50);
-    giq_option(4, 46, 328, Node962, 50);
-    giq_option(4, 46, 329, Node963, 50);
+    gsay_reply(SCRIPT_DCMETZGE, 326);
+    giq_option(4, SCRIPT_DCMETZGE, 327, Node961, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_DCMETZGE, 328, Node962, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_DCMETZGE, 329, Node963, NEUTRAL_REACTION);
 end
 
 procedure Node998
@@ -1624,7 +1624,7 @@ end
 
 procedure Node990
 begin
-    if (global_var(11)) then begin
+    if (global_var(GVAR_REPUTATION_SLAVER)) then begin
         call Node017();
     end
     else begin
@@ -1635,50 +1635,50 @@ end
 procedure Node989
 begin
     item_caps_adjust(dude_obj, slave_value);
-    set_global_var(0, global_var(0) + -15);
+    set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + -15);
     debug_msg("Player gains " + -15 + " Karma Points.");
-    set_global_var(37, 0);
-    set_global_var(38, 0);
-    set_global_var(39, 0);
-    set_global_var(40, 0);
-    set_global_var(41, 0);
-    set_global_var(42, 0);
-    set_global_var(43, 0);
-    set_global_var(44, 0);
-    set_global_var(45, 0);
-    if (global_var(0) >= 1000) then begin
-        set_global_var(37, 1);
+    set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+    set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+    set_global_var(GVAR_KARMA_DEFENDER, 0);
+    set_global_var(GVAR_KARMA_WANDERER, 0);
+    set_global_var(GVAR_KARMA_BETRAYER, 0);
+    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+    set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+    set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+    if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
     end
     else begin
-        if (global_var(0) >= 750) then begin
-            set_global_var(38, 1);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+            set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
         end
         else begin
-            if (global_var(0) >= 500) then begin
-                set_global_var(39, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
             end
             else begin
-                if (global_var(0) >= 250) then begin
-                    set_global_var(40, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                    set_global_var(GVAR_KARMA_DEFENDER, 1);
                 end
                 else begin
-                    if (global_var(0) > -250) then begin
-                        set_global_var(41, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                        set_global_var(GVAR_KARMA_WANDERER, 1);
                     end
                     else begin
-                        if (global_var(0) > -500) then begin
-                            set_global_var(42, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                            set_global_var(GVAR_KARMA_BETRAYER, 1);
                         end
                         else begin
-                            if (global_var(0) > -750) then begin
-                                set_global_var(43, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                             end
                             else begin
-                                if (global_var(0) > -1000) then begin
-                                    set_global_var(44, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                    set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                 end
                                 else begin
-                                    set_global_var(45, 1);
+                                    set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                 end
                             end
                         end
@@ -1703,50 +1703,50 @@ end
 procedure Node987
 begin
     item_caps_adjust(dude_obj, slave_value);
-    set_global_var(0, global_var(0) + -15);
+    set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + -15);
     debug_msg("Player gains " + -15 + " Karma Points.");
-    set_global_var(37, 0);
-    set_global_var(38, 0);
-    set_global_var(39, 0);
-    set_global_var(40, 0);
-    set_global_var(41, 0);
-    set_global_var(42, 0);
-    set_global_var(43, 0);
-    set_global_var(44, 0);
-    set_global_var(45, 0);
-    if (global_var(0) >= 1000) then begin
-        set_global_var(37, 1);
+    set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+    set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+    set_global_var(GVAR_KARMA_DEFENDER, 0);
+    set_global_var(GVAR_KARMA_WANDERER, 0);
+    set_global_var(GVAR_KARMA_BETRAYER, 0);
+    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+    set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+    set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+    if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
     end
     else begin
-        if (global_var(0) >= 750) then begin
-            set_global_var(38, 1);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+            set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
         end
         else begin
-            if (global_var(0) >= 500) then begin
-                set_global_var(39, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
             end
             else begin
-                if (global_var(0) >= 250) then begin
-                    set_global_var(40, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                    set_global_var(GVAR_KARMA_DEFENDER, 1);
                 end
                 else begin
-                    if (global_var(0) > -250) then begin
-                        set_global_var(41, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                        set_global_var(GVAR_KARMA_WANDERER, 1);
                     end
                     else begin
-                        if (global_var(0) > -500) then begin
-                            set_global_var(42, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                            set_global_var(GVAR_KARMA_BETRAYER, 1);
                         end
                         else begin
-                            if (global_var(0) > -750) then begin
-                                set_global_var(43, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                             end
                             else begin
-                                if (global_var(0) > -1000) then begin
-                                    set_global_var(44, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                    set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                 end
                                 else begin
-                                    set_global_var(45, 1);
+                                    set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                 end
                             end
                         end
@@ -1828,7 +1828,7 @@ end
 
 procedure Node983
 begin
-    if (((game_time - global_var(29)) >= (24 * (60 * (60 * 10)))) and (global_var(29) != 0)) then begin
+    if (((game_time - global_var(GVAR_VIC_DEVICE)) >= (24 * (60 * (60 * 10)))) and (global_var(GVAR_VIC_DEVICE) != 0)) then begin
         if (is_success(roll_vs_skill(dude_obj, 15, 0))) then begin
             call Node027();
         end
@@ -1846,52 +1846,52 @@ begin
     if (item_caps_total(dude_obj) >= slave_value) then begin
         item_caps_adjust(self_obj, slave_value);
         item_caps_adjust(dude_obj, -slave_value);
-        if ((((global_var(19) bwand 128) != 0) or ((global_var(298) bwand 512) != 0)) == 0) then begin
-            set_global_var(19, global_var(19) bwor 128);
-            set_global_var(0, global_var(0) + 10);
+        if ((((global_var(GVAR_DEN_VIC_STATUS) bwand 128) != 0) or ((global_var(GVAR_DEN_CEASAR_STATUS) bwand 512) != 0)) == 0) then begin
+            set_global_var(GVAR_DEN_VIC_STATUS, global_var(GVAR_DEN_VIC_STATUS) bwor 128);
+            set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + 10);
             debug_msg("Player gains " + 10 + " Karma Points.");
-            set_global_var(37, 0);
-            set_global_var(38, 0);
-            set_global_var(39, 0);
-            set_global_var(40, 0);
-            set_global_var(41, 0);
-            set_global_var(42, 0);
-            set_global_var(43, 0);
-            set_global_var(44, 0);
-            set_global_var(45, 0);
-            if (global_var(0) >= 1000) then begin
-                set_global_var(37, 1);
+            set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+            set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+            set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+            set_global_var(GVAR_KARMA_DEFENDER, 0);
+            set_global_var(GVAR_KARMA_WANDERER, 0);
+            set_global_var(GVAR_KARMA_BETRAYER, 0);
+            set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+            set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+            set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+                set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
             end
             else begin
-                if (global_var(0) >= 750) then begin
-                    set_global_var(38, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                    set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
                 end
                 else begin
-                    if (global_var(0) >= 500) then begin
-                        set_global_var(39, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                        set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                     end
                     else begin
-                        if (global_var(0) >= 250) then begin
-                            set_global_var(40, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                            set_global_var(GVAR_KARMA_DEFENDER, 1);
                         end
                         else begin
-                            if (global_var(0) > -250) then begin
-                                set_global_var(41, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                                set_global_var(GVAR_KARMA_WANDERER, 1);
                             end
                             else begin
-                                if (global_var(0) > -500) then begin
-                                    set_global_var(42, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                    set_global_var(GVAR_KARMA_BETRAYER, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -750) then begin
-                                        set_global_var(43, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                        set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                     end
                                     else begin
-                                        if (global_var(0) > -1000) then begin
-                                            set_global_var(44, 1);
+                                        if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                            set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                         end
                                         else begin
-                                            set_global_var(45, 1);
+                                            set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                         end
                                     end
                                 end
@@ -1901,7 +1901,7 @@ begin
                 end
             end
         end
-        set_global_var(12, 1);
+        set_global_var(GVAR_REPUTATION_SLAVE_OWNER, 1);
         call Node020();
     end
     else begin
@@ -1924,50 +1924,50 @@ end
 procedure Node969
 begin
     item_caps_adjust(dude_obj, slave_value);
-    set_global_var(0, global_var(0) + -20);
+    set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + -20);
     debug_msg("Player gains " + -20 + " Karma Points.");
-    set_global_var(37, 0);
-    set_global_var(38, 0);
-    set_global_var(39, 0);
-    set_global_var(40, 0);
-    set_global_var(41, 0);
-    set_global_var(42, 0);
-    set_global_var(43, 0);
-    set_global_var(44, 0);
-    set_global_var(45, 0);
-    if (global_var(0) >= 1000) then begin
-        set_global_var(37, 1);
+    set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+    set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+    set_global_var(GVAR_KARMA_DEFENDER, 0);
+    set_global_var(GVAR_KARMA_WANDERER, 0);
+    set_global_var(GVAR_KARMA_BETRAYER, 0);
+    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+    set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+    set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+    if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
     end
     else begin
-        if (global_var(0) >= 750) then begin
-            set_global_var(38, 1);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+            set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
         end
         else begin
-            if (global_var(0) >= 500) then begin
-                set_global_var(39, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
             end
             else begin
-                if (global_var(0) >= 250) then begin
-                    set_global_var(40, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                    set_global_var(GVAR_KARMA_DEFENDER, 1);
                 end
                 else begin
-                    if (global_var(0) > -250) then begin
-                        set_global_var(41, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                        set_global_var(GVAR_KARMA_WANDERER, 1);
                     end
                     else begin
-                        if (global_var(0) > -500) then begin
-                            set_global_var(42, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                            set_global_var(GVAR_KARMA_BETRAYER, 1);
                         end
                         else begin
-                            if (global_var(0) > -750) then begin
-                                set_global_var(43, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                             end
                             else begin
-                                if (global_var(0) > -1000) then begin
-                                    set_global_var(44, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                    set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                 end
                                 else begin
-                                    set_global_var(45, 1);
+                                    set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                 end
                             end
                         end
@@ -1987,7 +1987,7 @@ end
 
 procedure Node967
 begin
-    if (global_var(34) > 0) then begin
+    if (global_var(GVAR_SLAVERS_LOST) > 0) then begin
         call Node040();
     end
     else begin
@@ -1997,7 +1997,7 @@ end
 
 procedure Node966
 begin
-    if (global_var(33) > 0) then begin
+    if (global_var(GVAR_SLAVES_LOST) > 0) then begin
         call Node041();
     end
     else begin
@@ -2007,7 +2007,7 @@ end
 
 procedure Node965
 begin
-    if (global_var(31) == 0) then begin
+    if (global_var(GVAR_SLAVES_COUNT) == 0) then begin
         call Node042();
     end
     else begin
@@ -2019,11 +2019,11 @@ procedure Node964
 begin
     variable LVar0 := 0;
     LVar0 := 0;
-    if ((global_var(34) >= 3) or (global_var(33) >= 5)) then begin
+    if ((global_var(GVAR_SLAVERS_LOST) >= 3) or (global_var(GVAR_SLAVES_LOST) >= 5)) then begin
         LVar0 := 1;
     end
     else begin
-        LVar0 := global_var(31) - global_var(34) - global_var(33);
+        LVar0 := global_var(GVAR_SLAVES_COUNT) - global_var(GVAR_SLAVERS_LOST) - global_var(GVAR_SLAVES_LOST);
         if (LVar0 >= 5) then begin
             LVar0 := 2;
         end
@@ -2125,4 +2125,3 @@ end
 procedure Node999
 begin
 end
-
