diff --git a/Games/Fallout2.Steam/Original/DCFRED.ssl b/Games/Fallout2.Steam/ReDefine/DCFRED.ssl
index 958a9c6..5efab38 100644
--- a/Games/Fallout2.Steam/Original/DCFRED.ssl
+++ b/Games/Fallout2.Steam/ReDefine/DCFRED.ssl
@@ -207,10 +207,10 @@ begin
                 if (anim_busy(self_obj) == 0) then begin
                     if ((tile_distance_objs(self_obj, dude_obj) < 8) or obj_can_see_obj(self_obj, dude_obj)) then begin
                         if (anim_busy(self_obj) == 0) then begin
-                            global_temp := rotation_to_tile(tile_num(dude_obj), tile_num(self_obj));
+                            global_temp := rotation_to_tile(dude_tile, tile_num(self_obj));
                             animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(self_obj), global_temp, random(3, 10)), 1);
                             global_temp := (global_temp + 1) % 6;
-                            while ((anim_busy(self_obj) == 0) and (global_temp != rotation_to_tile(tile_num(dude_obj), tile_num(self_obj)))) do begin
+                            while ((anim_busy(self_obj) == 0) and (global_temp != rotation_to_tile(dude_tile, tile_num(self_obj)))) do begin
                                 debug_msg("flee loop: rot == " + global_temp);
                                 animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(self_obj), global_temp, random(3, 10)), 1);
                                 global_temp := (global_temp + 1) % 6;
@@ -220,7 +220,7 @@ begin
                 end
             end
             else begin
-                attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+                attack(dude_obj);
             end
         end
         else begin
@@ -278,8 +278,8 @@ begin
                         call Node005();
                     end
                     else begin
-                        if ((global_var(371) == 1) and ((global_var(446) bwand 256) != 0)) then begin
-                            if ((global_var(446) bwand 16384) != 0) then begin
+                        if ((global_var(GVAR_DEN_BECKY_JOB) == 1) and ((global_var(GVAR_DEN_FLAG_2) bwand 256) != 0)) then begin
+                            if ((global_var(GVAR_DEN_FLAG_2) bwand 16384) != 0) then begin
                                 if (in_dialog == 0) then begin
                                     in_dialog := 1;
                                     start_gdialog(916, self_obj, 4, -1, -1);
@@ -321,57 +321,57 @@ end
 
 procedure destroy_p_proc
 begin
-    set_global_var(445, global_var(445) bwor 1024);
-    item_caps_adjust(self_obj, global_var(464));
+    set_global_var(GVAR_DEN_FLAG_1, global_var(GVAR_DEN_FLAG_1) bwor 1024);
+    item_caps_adjust(self_obj, global_var(GVAR_FRED_MONEY));
     if (source_obj == dude_obj) then begin
-        set_global_var(5, global_var(5) + 1);
-        if (metarule(51, self_obj) == 2) then begin
+        set_global_var(GVAR_GOOD_MONSTER, global_var(GVAR_GOOD_MONSTER) + 1);
+        if (critter_kill_type(self_obj) == KILL_TYPE_children_kills) then begin
             if (source_obj == dude_obj) then begin
-                set_global_var(2, 0);
-                set_global_var(1, global_var(1) + 1);
-                set_global_var(0, global_var(0) - -15);
-                set_global_var(37, 0);
-                set_global_var(38, 0);
-                set_global_var(39, 0);
-                set_global_var(40, 0);
-                set_global_var(41, 0);
-                set_global_var(42, 0);
-                set_global_var(43, 0);
-                set_global_var(44, 0);
-                set_global_var(45, 0);
-                if (global_var(0) >= 1000) then begin
-                    set_global_var(37, 1);
+                set_global_var(GVAR_CHAMPION_REPUTATION, 0);
+                set_global_var(GVAR_CHILDKILLER_REPUTATION, global_var(GVAR_CHILDKILLER_REPUTATION) + 1);
+                set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) - -15);
+                set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+                set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+                set_global_var(GVAR_KARMA_DEFENDER, 0);
+                set_global_var(GVAR_KARMA_WANDERER, 0);
+                set_global_var(GVAR_KARMA_BETRAYER, 0);
+                set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+                set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+                set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+                    set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
                 end
                 else begin
-                    if (global_var(0) >= 750) then begin
-                        set_global_var(38, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
                     end
                     else begin
-                        if (global_var(0) >= 500) then begin
-                            set_global_var(39, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                            set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                         end
                         else begin
-                            if (global_var(0) >= 250) then begin
-                                set_global_var(40, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                                set_global_var(GVAR_KARMA_DEFENDER, 1);
                             end
                             else begin
-                                if (global_var(0) > -250) then begin
-                                    set_global_var(41, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                                    set_global_var(GVAR_KARMA_WANDERER, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -500) then begin
-                                        set_global_var(42, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                        set_global_var(GVAR_KARMA_BETRAYER, 1);
                                     end
                                     else begin
-                                        if (global_var(0) > -750) then begin
-                                            set_global_var(43, 1);
+                                        if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                            set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                         end
                                         else begin
-                                            if (global_var(0) > -1000) then begin
-                                                set_global_var(44, 1);
+                                            if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                                set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                             end
                                             else begin
-                                                set_global_var(45, 1);
+                                                set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                             end
                                         end
                                     end
@@ -382,58 +382,58 @@ begin
                 end
             end
         end
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(4) > (3 * global_var(5))) or (global_var(2) == 1)) and (global_var(1) == 0)) then begin
-            set_global_var(2, 1);
-            set_global_var(3, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(4) > (3 * global_var(GVAR_GOOD_MONSTER))) or (global_var(GVAR_CHAMPION_REPUTATION) == 1)) and (global_var(GVAR_CHILDKILLER_REPUTATION) == 0)) then begin
+            set_global_var(GVAR_CHAMPION_REPUTATION, 1);
+            set_global_var(GVAR_BERSERKER_REPUTATION, 0);
         end
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(5) > (2 * global_var(4))) or (global_var(3) == 1))) then begin
-            set_global_var(3, 1);
-            set_global_var(2, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
+            set_global_var(GVAR_BERSERKER_REPUTATION, 1);
+            set_global_var(GVAR_CHAMPION_REPUTATION, 0);
         end
-        set_global_var(0, global_var(0) + -10);
+        set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + -10);
         debug_msg("Player gains " + -10 + " Karma Points.");
-        set_global_var(37, 0);
-        set_global_var(38, 0);
-        set_global_var(39, 0);
-        set_global_var(40, 0);
-        set_global_var(41, 0);
-        set_global_var(42, 0);
-        set_global_var(43, 0);
-        set_global_var(44, 0);
-        set_global_var(45, 0);
-        if (global_var(0) >= 1000) then begin
-            set_global_var(37, 1);
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+        set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+        set_global_var(GVAR_KARMA_DEFENDER, 0);
+        set_global_var(GVAR_KARMA_WANDERER, 0);
+        set_global_var(GVAR_KARMA_BETRAYER, 0);
+        set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+        set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+            set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
         end
         else begin
-            if (global_var(0) >= 750) then begin
-                set_global_var(38, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
             end
             else begin
-                if (global_var(0) >= 500) then begin
-                    set_global_var(39, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                 end
                 else begin
-                    if (global_var(0) >= 250) then begin
-                        set_global_var(40, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                        set_global_var(GVAR_KARMA_DEFENDER, 1);
                     end
                     else begin
-                        if (global_var(0) > -250) then begin
-                            set_global_var(41, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                            set_global_var(GVAR_KARMA_WANDERER, 1);
                         end
                         else begin
-                            if (global_var(0) > -500) then begin
-                                set_global_var(42, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                set_global_var(GVAR_KARMA_BETRAYER, 1);
                             end
                             else begin
-                                if (global_var(0) > -750) then begin
-                                    set_global_var(43, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -1000) then begin
-                                        set_global_var(44, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                     end
                                     else begin
-                                        set_global_var(45, 1);
+                                        set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                     end
                                 end
                             end
@@ -448,13 +448,13 @@ end
 procedure look_at_p_proc
 begin
     script_overrides;
-    display_msg(message_str(916, 100 + ((local_var(4) bwand 1) != 0) + (obj_pid(self_obj) == 16777310)));
+    display_mstr(100 + ((local_var(4) bwand 1) != 0) + (obj_pid(self_obj) == 16777310));
 end
 
 procedure description_p_proc
 begin
     script_overrides;
-    display_msg(message_str(916, 150 + (obj_pid(self_obj) == 16777310)));
+    display_mstr(150 + (obj_pid(self_obj) == 16777310));
     set_local_var(4, local_var(4) bwor 4);
 end
 
@@ -476,28 +476,28 @@ begin
     if (local_var(5) == 0) then begin
         set_local_var(5, tile_num(self_obj));
     end
-    if (metarule(22, 0) == 0) then begin
+    if (is_loading_game == 0) then begin
         if (obj_pid(self_obj) != 16777310) then begin
             if ((game_time >= local_var(6)) and (local_var(6) != 0)) then begin
-                if ((global_var(447) bwand 8) != 0) then begin
-                    set_global_var(464, 2000);
+                if ((global_var(GVAR_DEN_FLAG_3) bwand 8) != 0) then begin
+                    set_global_var(GVAR_FRED_MONEY, 2000);
                 end
-                set_global_var(447, global_var(447) bwor 8);
-                if (has_trait(1, self_obj, 666) == 1) then begin
+                set_global_var(GVAR_DEN_FLAG_3, global_var(GVAR_DEN_FLAG_3) bwor 8);
+                if (obj_is_visible_flag(self_obj) == 1) then begin
                     set_obj_visibility(self_obj, 1);
                 end
-                critter_attempt_placement(self_obj, global_var(46), elevation(self_obj));
-                LVar0 := create_object_sid(16777310, 0, 0, 916);
+                critter_attempt_placement(self_obj, global_var(GVAR_MAP_EXIT_TILE), elevation(self_obj));
+                LVar0 := create_object_sid(16777310, 0, 0, NAME);
                 critter_attempt_placement(LVar0, local_var(5), elevation(self_obj));
                 destroy_object(self_obj);
             end
         end
-        metarule3(100, self_obj, 1, 0);
+        rm_fixed_timer_event(self_obj, 1);
         add_timer_event(self_obj, game_ticks(random(15, 90)), 1);
         anim(self_obj, 1000, 3);
     end
-    critter_add_trait(self_obj, 1, 6, 9);
-    critter_add_trait(self_obj, 1, 5, 33);
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 9);
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_COWARD);
 end
 
 procedure map_update_p_proc
@@ -530,12 +530,12 @@ begin
                                 animate_move_obj_to_tile(self_obj, step_tile, 0);
                             end
                             if (anim_busy(self_obj) == 0) then begin
-                                if (metarule3(106, tile_num_in_direction(tile_num(self_obj), has_trait(1, self_obj, 10), 1), elevation(self_obj), 0) != 0) then begin
+                                if (tile_get_next_critter(tile_num_in_direction(tile_num(self_obj), obj_get_rot(self_obj), 1), elevation(self_obj), 0) != 0) then begin
                                     if (anim_busy(self_obj) == 0) then begin
-                                        global_temp := rotation_to_tile(tile_num_in_direction(tile_num(self_obj), rotation_to_tile(tile_num_in_direction(tile_num(self_obj), has_trait(1, self_obj, 10), 1), tile_num(self_obj)), 1), tile_num(self_obj));
+                                        global_temp := rotation_to_tile(tile_num_in_direction(tile_num(self_obj), rotation_to_tile(tile_num_in_direction(tile_num(self_obj), obj_get_rot(self_obj), 1), tile_num(self_obj)), 1), tile_num(self_obj));
                                         animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(self_obj), global_temp, random(2, 5)), 0);
                                         global_temp := (global_temp + 1) % 6;
-                                        while ((anim_busy(self_obj) == 0) and (global_temp != rotation_to_tile(tile_num_in_direction(tile_num(self_obj), rotation_to_tile(tile_num_in_direction(tile_num(self_obj), has_trait(1, self_obj, 10), 1), tile_num(self_obj)), 1), tile_num(self_obj)))) do begin
+                                        while ((anim_busy(self_obj) == 0) and (global_temp != rotation_to_tile(tile_num_in_direction(tile_num(self_obj), rotation_to_tile(tile_num_in_direction(tile_num(self_obj), obj_get_rot(self_obj), 1), tile_num(self_obj)), 1), tile_num(self_obj)))) do begin
                                             debug_msg("flee loop: rot == " + global_temp);
                                             animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(self_obj), global_temp, random(2, 5)), 0);
                                             global_temp := (global_temp + 1) % 6;
@@ -580,12 +580,12 @@ begin
                                         animate_move_obj_to_tile(self_obj, step_tile, 0);
                                     end
                                     if (anim_busy(self_obj) == 0) then begin
-                                        if (metarule3(106, tile_num_in_direction(tile_num(self_obj), has_trait(1, self_obj, 10), 1), elevation(self_obj), 0) != 0) then begin
+                                        if (tile_get_next_critter(tile_num_in_direction(tile_num(self_obj), obj_get_rot(self_obj), 1), elevation(self_obj), 0) != 0) then begin
                                             if (anim_busy(self_obj) == 0) then begin
-                                                global_temp := rotation_to_tile(tile_num_in_direction(tile_num(self_obj), rotation_to_tile(tile_num_in_direction(tile_num(self_obj), has_trait(1, self_obj, 10), 1), tile_num(self_obj)), 1), tile_num(self_obj));
+                                                global_temp := rotation_to_tile(tile_num_in_direction(tile_num(self_obj), rotation_to_tile(tile_num_in_direction(tile_num(self_obj), obj_get_rot(self_obj), 1), tile_num(self_obj)), 1), tile_num(self_obj));
                                                 animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(self_obj), global_temp, random(2, 5)), 0);
                                                 global_temp := (global_temp + 1) % 6;
-                                                while ((anim_busy(self_obj) == 0) and (global_temp != rotation_to_tile(tile_num_in_direction(tile_num(self_obj), rotation_to_tile(tile_num_in_direction(tile_num(self_obj), has_trait(1, self_obj, 10), 1), tile_num(self_obj)), 1), tile_num(self_obj)))) do begin
+                                                while ((anim_busy(self_obj) == 0) and (global_temp != rotation_to_tile(tile_num_in_direction(tile_num(self_obj), rotation_to_tile(tile_num_in_direction(tile_num(self_obj), obj_get_rot(self_obj), 1), tile_num(self_obj)), 1), tile_num(self_obj)))) do begin
                                                     debug_msg("flee loop: rot == " + global_temp);
                                                     animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(self_obj), global_temp, random(2, 5)), 0);
                                                     global_temp := (global_temp + 1) % 6;
@@ -608,7 +608,7 @@ begin
                 end
             end
         end
-        metarule3(100, self_obj, 1, 0);
+        rm_fixed_timer_event(self_obj, 1);
         add_timer_event(self_obj, game_ticks(random(15, 90)), 1);
     end
 end
@@ -624,80 +624,80 @@ end
 
 procedure Node001
 begin
-    float_msg(self_obj, message_str(916, random(200, 207)), 8);
+    float_msg(self_obj, mstr(random(200, 207)), FLOAT_MSG_YELLOW);
 end
 
 procedure Node002
 begin
-    float_msg(self_obj, message_str(916, random(221 - (get_critter_stat(dude_obj, 34) == 0), 227 + (get_critter_stat(dude_obj, 34) == 1))), 8);
+    float_msg(self_obj, mstr(random(221 - (dude_gender == 0), 227 + (dude_gender == 1))), FLOAT_MSG_YELLOW);
 end
 
 procedure Node003
 begin
-    float_msg(self_obj, message_str(916, random(240, 247)), 8);
+    float_msg(self_obj, mstr(random(240, 247)), FLOAT_MSG_YELLOW);
 end
 
 procedure Node004
 begin
-    float_msg(self_obj, message_str(916, random(260, 264)), 8);
+    float_msg(self_obj, mstr(random(260, 264)), FLOAT_MSG_YELLOW);
 end
 
 procedure Node005
 begin
-    float_msg(self_obj, message_str(916, random(275, 281)), 8);
+    float_msg(self_obj, mstr(random(275, 281)), FLOAT_MSG_YELLOW);
 end
 
 procedure Node006
 begin
-    set_global_var(446, global_var(446) bwor 32768);
-    gsay_reply(916, 295 + ((local_var(4) bwand 512) != 0));
+    set_global_var(GVAR_DEN_FLAG_2, global_var(GVAR_DEN_FLAG_2) bwor 32768);
+    Reply(295 + ((local_var(4) bwand 512) != 0));
     set_local_var(4, local_var(4) bwor 512);
-    giq_option(4, 916, 297, Node018, 50);
-    giq_option(4, 916, 298, Node999, 50);
-    giq_option(-3, 916, 299, Node017, 50);
-    giq_option(-3, 916, 300, Node999, 50);
+    giq_option(4, NAME, 297, Node018, NEUTRAL_REACTION);
+    giq_option(4, NAME, 298, Node999, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 299, Node017, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 300, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node007
 begin
-    gsay_reply(916, 310);
+    Reply(310);
     set_local_var(4, local_var(4) bwor 1024);
-    giq_option(4, 916, 312, Node008, 49);
-    giq_option(4, 916, 313, Node009, 50);
-    giq_option(-3, 916, 314, Node009, 50);
-    giq_option(-3, 916, 315, Node009, 50);
+    giq_option(4, NAME, 312, Node008, GOOD_REACTION);
+    giq_option(4, NAME, 313, Node009, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 314, Node009, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 315, Node009, NEUTRAL_REACTION);
 end
 
 procedure Node007b
 begin
-    float_msg(self_obj, message_str(916, 311), 8);
+    float_msg(self_obj, mstr(311), FLOAT_MSG_YELLOW);
 end
 
 procedure Node008
 begin
-    gsay_reply(916, 320);
-    giq_option(4, 916, 321, Node990, 50);
-    giq_option(4, 916, 322, Node010, 50);
+    Reply(320);
+    giq_option(4, NAME, 321, Node990, NEUTRAL_REACTION);
+    giq_option(4, NAME, 322, Node010, NEUTRAL_REACTION);
 end
 
 procedure Node009
 begin
-    gsay_reply(916, 330);
-    giq_option(4, 916, 331, Node990, 50);
-    giq_option(4, 916, 332, Node010, 51);
-    giq_option(-3, 916, 333, Node990, 50);
+    Reply(330);
+    giq_option(4, NAME, 331, Node990, NEUTRAL_REACTION);
+    giq_option(4, NAME, 332, Node010, BAD_REACTION);
+    giq_option(-3, NAME, 333, Node990, NEUTRAL_REACTION);
 end
 
 procedure Node010
 begin
-    gsay_reply(916, 340);
-    giq_option(4, 916, 341, Node990, 49);
+    Reply(340);
+    giq_option(4, NAME, 341, Node990, GOOD_REACTION);
     if (local_var(7) == 2) then begin
-        giq_option(4, 916, 342, Node011, 51);
+        giq_option(4, NAME, 342, Node011, BAD_REACTION);
     end
     else begin
         if (local_var(7) == 3) then begin
-            giq_option(4, 916, 343, Node011, 51);
+            giq_option(4, NAME, 343, Node011, BAD_REACTION);
         end
     end
 end
@@ -705,91 +705,91 @@ end
 procedure Node011
 begin
     item_caps_adjust(dude_obj, 500);
-    gsay_reply(916, 350);
-    giq_option(4, 916, 351, Node990, 49);
-    giq_option(4, 916, 352, Node990, 51);
+    Reply(350);
+    giq_option(4, NAME, 351, Node990, GOOD_REACTION);
+    giq_option(4, NAME, 352, Node990, BAD_REACTION);
 end
 
 procedure Node012
 begin
-    gsay_reply(916, 360);
-    giq_option(4, 916, 361, Node014, 50);
-    giq_option(4, 916, 362, Node014, 50);
-    giq_option(-3, 916, 363, Node014, 50);
+    Reply(360);
+    giq_option(4, NAME, 361, Node014, NEUTRAL_REACTION);
+    giq_option(4, NAME, 362, Node014, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 363, Node014, NEUTRAL_REACTION);
 end
 
 procedure Node013
 begin
-    gsay_reply(916, 370);
-    giq_option(4, 916, 371, Node015, 50);
-    giq_option(4, 916, 372, Node015, 50);
-    giq_option(-3, 916, 373, Node015, 50);
+    Reply(370);
+    giq_option(4, NAME, 371, Node015, NEUTRAL_REACTION);
+    giq_option(4, NAME, 372, Node015, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 373, Node015, NEUTRAL_REACTION);
 end
 
 procedure Node014
 begin
-    gsay_reply(916, 380);
-    giq_option(4, 916, 381, Node016, 49);
-    giq_option(4, 916, 382, Node989, 50);
-    giq_option(4, 916, 383, Node989, 50);
-    giq_option(-3, 916, 384, Node989, 50);
+    Reply(380);
+    giq_option(4, NAME, 381, Node016, GOOD_REACTION);
+    giq_option(4, NAME, 382, Node989, NEUTRAL_REACTION);
+    giq_option(4, NAME, 383, Node989, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 384, Node989, NEUTRAL_REACTION);
 end
 
 procedure Node015
 begin
-    gsay_reply(916, 390);
-    giq_option(4, 916, 391, Node016, 49);
-    giq_option(4, 916, 392, Node988, 50);
-    giq_option(4, 916, 393, Node988, 50);
-    giq_option(-3, 916, 394, Node988, 50);
+    Reply(390);
+    giq_option(4, NAME, 391, Node016, GOOD_REACTION);
+    giq_option(4, NAME, 392, Node988, NEUTRAL_REACTION);
+    giq_option(4, NAME, 393, Node988, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 394, Node988, NEUTRAL_REACTION);
 end
 
 procedure Node016
 begin
-    set_global_var(0, global_var(0) + 3);
+    set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + 3);
     debug_msg("Player gains " + 3 + " Karma Points.");
-    set_global_var(37, 0);
-    set_global_var(38, 0);
-    set_global_var(39, 0);
-    set_global_var(40, 0);
-    set_global_var(41, 0);
-    set_global_var(42, 0);
-    set_global_var(43, 0);
-    set_global_var(44, 0);
-    set_global_var(45, 0);
-    if (global_var(0) >= 1000) then begin
-        set_global_var(37, 1);
+    set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+    set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+    set_global_var(GVAR_KARMA_DEFENDER, 0);
+    set_global_var(GVAR_KARMA_WANDERER, 0);
+    set_global_var(GVAR_KARMA_BETRAYER, 0);
+    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+    set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+    set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+    if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
     end
     else begin
-        if (global_var(0) >= 750) then begin
-            set_global_var(38, 1);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+            set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
         end
         else begin
-            if (global_var(0) >= 500) then begin
-                set_global_var(39, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
             end
             else begin
-                if (global_var(0) >= 250) then begin
-                    set_global_var(40, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                    set_global_var(GVAR_KARMA_DEFENDER, 1);
                 end
                 else begin
-                    if (global_var(0) > -250) then begin
-                        set_global_var(41, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                        set_global_var(GVAR_KARMA_WANDERER, 1);
                     end
                     else begin
-                        if (global_var(0) > -500) then begin
-                            set_global_var(42, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                            set_global_var(GVAR_KARMA_BETRAYER, 1);
                         end
                         else begin
-                            if (global_var(0) > -750) then begin
-                                set_global_var(43, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                             end
                             else begin
-                                if (global_var(0) > -1000) then begin
-                                    set_global_var(44, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                    set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                 end
                                 else begin
-                                    set_global_var(45, 1);
+                                    set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                 end
                             end
                         end
@@ -798,63 +798,63 @@ begin
             end
         end
     end
-    gsay_reply(916, 400);
-    giq_option(4, 916, 401, Node999, 50);
+    Reply(400);
+    giq_option(4, NAME, 401, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node017
 begin
-    gsay_reply(916, 410 + ((local_var(4) bwand 2048) != 0));
+    Reply(410 + ((local_var(4) bwand 2048) != 0));
     set_local_var(4, local_var(4) bwor 2048);
-    giq_option(-3, 916, 412, Node999, 50);
+    giq_option(-3, NAME, 412, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node018
 begin
-    gsay_reply(916, 420 + ((local_var(4) bwand 4096) != 0));
+    Reply(420 + ((local_var(4) bwand 4096) != 0));
     set_local_var(4, local_var(4) bwor 4096);
-    giq_option(4, 916, 422, Node020, 51);
-    if (get_critter_stat(dude_obj, 4) < 7) then begin
-        giq_option(4, 916, 423, Node033, 50);
+    giq_option(4, NAME, 422, Node020, BAD_REACTION);
+    if (dude_iq < 7) then begin
+        giq_option(4, NAME, 423, Node033, NEUTRAL_REACTION);
     end
-    giq_option(7, 916, 424, Node019, 50);
+    giq_option(7, NAME, 424, Node019, NEUTRAL_REACTION);
 end
 
 procedure Node019
 begin
-    gsay_reply(916, 430);
-    giq_option(4, 916, 431, Node021, 51);
-    giq_option(4, 916, 432, Node022, 50);
+    Reply(430);
+    giq_option(4, NAME, 431, Node021, BAD_REACTION);
+    giq_option(4, NAME, 432, Node022, NEUTRAL_REACTION);
 end
 
 procedure Node020
 begin
-    gsay_reply(916, 440);
-    giq_option(4, 916, 441, Node999, 50);
-    giq_option(1, 916, 442, Node999, 50);
+    Reply(440);
+    giq_option(4, NAME, 441, Node999, NEUTRAL_REACTION);
+    giq_option(1, NAME, 442, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node021
 begin
-    gsay_reply(916, 450);
-    giq_option(4, 916, 451, Node999, 51);
-    giq_option(4, 916, 452, Node999, 50);
+    Reply(450);
+    giq_option(4, NAME, 451, Node999, BAD_REACTION);
+    giq_option(4, NAME, 452, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node022
 begin
-    set_global_var(446, global_var(446) bwor 16384);
-    gsay_reply(916, 460);
-    giq_option(4, 916, 461, Node023, 50);
-    giq_option(4, 916, 462, Node023, 50);
+    set_global_var(GVAR_DEN_FLAG_2, global_var(GVAR_DEN_FLAG_2) bwor 16384);
+    Reply(460);
+    giq_option(4, NAME, 461, Node023, NEUTRAL_REACTION);
+    giq_option(4, NAME, 462, Node023, NEUTRAL_REACTION);
 end
 
 procedure Node023
 begin
-    gsay_reply(916, 470);
-    giq_option(4, 916, 471, Node999, 51);
-    giq_option(4, 916, 472, Node987, 51);
-    giq_option(4, 916, 473, Node026, 50);
+    Reply(470);
+    giq_option(4, NAME, 471, Node999, BAD_REACTION);
+    giq_option(4, NAME, 472, Node987, BAD_REACTION);
+    giq_option(4, NAME, 473, Node026, NEUTRAL_REACTION);
 end
 
 procedure Node024
@@ -862,19 +862,19 @@ begin
     if (1 > local_var(7)) then begin
         set_local_var(7, 1);
     end
-    gsay_reply(916, 480);
-    giq_option(4, 916, 481, Node026, 49);
-    giq_option(4, 916, 482, Node986, 51);
-    giq_option(4, 916, 483, Node986, 50);
+    Reply(480);
+    giq_option(4, NAME, 481, Node026, GOOD_REACTION);
+    giq_option(4, NAME, 482, Node986, BAD_REACTION);
+    giq_option(4, NAME, 483, Node986, NEUTRAL_REACTION);
 end
 
 procedure Node025
 begin
-    set_global_var(464, 100);
-    gsay_reply(916, 490);
-    giq_option(4, 916, 491, Node026, 49);
-    giq_option(4, 916, 492, Node029, 51);
-    giq_option(4, 916, 493, Node999, 50);
+    set_global_var(GVAR_FRED_MONEY, 100);
+    Reply(490);
+    giq_option(4, NAME, 491, Node026, GOOD_REACTION);
+    giq_option(4, NAME, 492, Node029, BAD_REACTION);
+    giq_option(4, NAME, 493, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node026
@@ -882,17 +882,17 @@ begin
     if (2 > local_var(7)) then begin
         set_local_var(7, 2);
     end
-    set_global_var(464, 100);
+    set_global_var(GVAR_FRED_MONEY, 100);
     item_caps_adjust(dude_obj, 100);
-    gsay_reply(916, 500 + (get_critter_stat(dude_obj, 34) == 1));
-    giq_option(4, 916, 502, Node028, 50);
-    giq_option(4, 916, 503, Node027, 50);
+    Reply(500 + (dude_gender == 1));
+    giq_option(4, NAME, 502, Node028, NEUTRAL_REACTION);
+    giq_option(4, NAME, 503, Node027, NEUTRAL_REACTION);
 end
 
 procedure Node027
 begin
-    gsay_reply(916, 510);
-    giq_option(4, 916, 511, Node028, 50);
+    Reply(510);
+    giq_option(4, NAME, 511, Node028, NEUTRAL_REACTION);
 end
 
 procedure Node028
@@ -900,24 +900,24 @@ begin
     if (local_var(6) == 0) then begin
         set_local_var(6, game_time + (30 * (24 * (60 * (60 * 10))) * random(3, 6)));
     end
-    gsay_reply(916, 520);
-    giq_option(4, 916, 521, Node030, 49);
-    giq_option(4, 916, 522, Node999, 50);
+    Reply(520);
+    giq_option(4, NAME, 521, Node030, GOOD_REACTION);
+    giq_option(4, NAME, 522, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node029
 begin
-    gsay_reply(916, 530);
-    giq_option(4, 916, 531, Node026, 49);
-    giq_option(4, 916, 532, Node999, 51);
+    Reply(530);
+    giq_option(4, NAME, 531, Node026, GOOD_REACTION);
+    giq_option(4, NAME, 532, Node999, BAD_REACTION);
 end
 
 procedure Node030
 begin
-    gsay_reply(916, 540);
-    giq_option(4, 916, 541, Node031, 49);
-    giq_option(5, 916, 542, Node032, 50);
-    giq_option(4, 916, 543, Node999, 51);
+    Reply(540);
+    giq_option(4, NAME, 541, Node031, GOOD_REACTION);
+    giq_option(5, NAME, 542, Node032, NEUTRAL_REACTION);
+    giq_option(4, NAME, 543, Node999, BAD_REACTION);
 end
 
 procedure Node031
@@ -926,9 +926,9 @@ begin
     if (3 > local_var(7)) then begin
         set_local_var(7, 3);
     end
-    set_global_var(464, 150);
-    gsay_reply(916, 550);
-    giq_option(4, 916, 551, Node999, 50);
+    set_global_var(GVAR_FRED_MONEY, 150);
+    Reply(550);
+    giq_option(4, NAME, 551, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node032
@@ -936,47 +936,47 @@ begin
     if (3 > local_var(7)) then begin
         set_local_var(7, 3);
     end
-    gsay_reply(916, 560);
-    giq_option(4, 916, 561, Node031, 49);
-    giq_option(4, 916, 562, Node999, 51);
+    Reply(560);
+    giq_option(4, NAME, 561, Node031, GOOD_REACTION);
+    giq_option(4, NAME, 562, Node999, BAD_REACTION);
 end
 
 procedure Node033
 begin
-    gsay_reply(916, 570 + ((local_var(4) bwand 8192) != 0));
+    Reply(570 + ((local_var(4) bwand 8192) != 0));
     set_local_var(4, local_var(4) bwor 8192);
-    giq_option(4, 916, 572, Node034, 50);
-    giq_option(4, 916, 573, Node035, 51);
-    giq_option(4, 916, 574, Node985, 50);
+    giq_option(4, NAME, 572, Node034, NEUTRAL_REACTION);
+    giq_option(4, NAME, 573, Node035, BAD_REACTION);
+    giq_option(4, NAME, 574, Node985, NEUTRAL_REACTION);
 end
 
 procedure Node034
 begin
-    gsay_reply(916, 580);
-    giq_option(4, 916, 581, Node022, 50);
-    giq_option(4, 916, 582, Node022, 51);
+    Reply(580);
+    giq_option(4, NAME, 581, Node022, NEUTRAL_REACTION);
+    giq_option(4, NAME, 582, Node022, BAD_REACTION);
 end
 
 procedure Node035
 begin
-    gsay_reply(916, 590);
-    giq_option(4, 916, 591, Node022, 51);
-    giq_option(4, 916, 592, Node022, 51);
+    Reply(590);
+    giq_option(4, NAME, 591, Node022, BAD_REACTION);
+    giq_option(4, NAME, 592, Node022, BAD_REACTION);
 end
 
 procedure Node036
 begin
-    gsay_reply(916, 600);
-    giq_option(4, 916, 601, Node999, 50);
-    giq_option(4, 916, 602, Node999, 50);
+    Reply(600);
+    giq_option(4, NAME, 601, Node999, NEUTRAL_REACTION);
+    giq_option(4, NAME, 602, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node037
 begin
-    gsay_reply(916, 610);
-    giq_option(4, 916, 611, Node999, 51);
-    giq_option(4, 916, 612, Node987, 51);
-    giq_option(4, 916, 613, Node026, 50);
+    Reply(610);
+    giq_option(4, NAME, 611, Node999, BAD_REACTION);
+    giq_option(4, NAME, 612, Node987, BAD_REACTION);
+    giq_option(4, NAME, 613, Node026, NEUTRAL_REACTION);
 end
 
 procedure Node990
@@ -997,11 +997,11 @@ end
 procedure Node988
 begin
     item_caps_adjust(dude_obj, 2000);
-    restock_obj := create_object_sid(39, 0, 0, -1);
+    restock_obj := create_object(39, 0, 0);
     add_mult_objs_to_inven(dude_obj, restock_obj, 5);
-    restock_obj := create_object_sid(38, 0, 0, -1);
+    restock_obj := create_object(38, 0, 0);
     add_mult_objs_to_inven(dude_obj, restock_obj, 5);
-    restock_obj := create_object_sid(26, 0, 0, -1);
+    restock_obj := create_object(26, 0, 0);
     add_mult_objs_to_inven(dude_obj, restock_obj, 1);
 end
 
@@ -1018,7 +1018,7 @@ end
 procedure Node986
 begin
     item_caps_adjust(dude_obj, 200);
-    set_global_var(464, 0);
+    set_global_var(GVAR_FRED_MONEY, 0);
 end
 
 procedure Node985
@@ -1030,4 +1030,3 @@ begin
         call Node036();
     end
 end
-
