diff --git a/Games/Fallout2.Steam/Original/DCRUNSLV.ssl b/Games/Fallout2.Steam/ReDefine/DCRUNSLV.ssl
index bbe6293..8c162c6 100644
--- a/Games/Fallout2.Steam/Original/DCRUNSLV.ssl
+++ b/Games/Fallout2.Steam/ReDefine/DCRUNSLV.ssl
@@ -193,24 +193,24 @@ begin
         if ((critter_state(self_obj) bwand 2) == 0) then begin
             if ((tile_distance_objs(self_obj, dude_obj) < 8) or obj_can_see_obj(self_obj, dude_obj)) then begin
                 if (anim_busy(self_obj) == 0) then begin
-                    GVar10 := rotation_to_tile(tile_num(dude_obj), tile_num(self_obj));
+                    GVar10 := rotation_to_tile(dude_tile, tile_num(self_obj));
                     animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(self_obj), GVar10, random(3, 10)), 1);
                     GVar10 := (GVar10 + 1) % 6;
-                    while ((anim_busy(self_obj) == 0) and (GVar10 != rotation_to_tile(tile_num(dude_obj), tile_num(self_obj)))) do begin
+                    while ((anim_busy(self_obj) == 0) and (GVar10 != rotation_to_tile(dude_tile, tile_num(self_obj)))) do begin
                         debug_msg("flee loop: rot == " + GVar10);
                         animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(self_obj), GVar10, random(3, 10)), 1);
                         GVar10 := (GVar10 + 1) % 6;
                     end
                 end
             end
-            if (((global_var(446) bwand 33554432) != 0) == 0) then begin
+            if (((global_var(GVAR_DEN_FLAG_2) bwand 33554432) != 0) == 0) then begin
                 set_local_var(0, local_var(0) bwor 512);
-                set_global_var(446, global_var(446) bwor 33554432);
+                set_global_var(GVAR_DEN_FLAG_2, global_var(GVAR_DEN_FLAG_2) bwor 33554432);
                 if ((obj_pid(self_obj) == 16777219) or (obj_pid(self_obj) == 16777220)) then begin
-                    float_msg(self_obj, message_str(932, random(300, 308)), 8);
+                    float_msg(self_obj, mstr(random(300, 308)), FLOAT_MSG_YELLOW);
                 end
                 else begin
-                    float_msg(self_obj, message_str(932, random(350, 363)), 8);
+                    float_msg(self_obj, mstr(random(350, 363)), FLOAT_MSG_YELLOW);
                 end
             end
         end
@@ -219,7 +219,7 @@ begin
         if (anim_busy(self_obj) == 0) then begin
             if ((local_var(0) bwand 512) != 0) then begin
                 set_local_var(0, local_var(0) bwand (-1 - 512));
-                set_global_var(446, global_var(446) bwand (-1 - 33554432));
+                set_global_var(GVAR_DEN_FLAG_2, global_var(GVAR_DEN_FLAG_2) bwand (-1 - 33554432));
             end
         end
     end
@@ -237,22 +237,22 @@ begin
     if ((critter_state(self_obj) bwand 2) == 0) then begin
         if ((tile_distance_objs(self_obj, dude_obj) < 8) or obj_can_see_obj(self_obj, dude_obj)) then begin
             if (anim_busy(self_obj) == 0) then begin
-                GVar10 := rotation_to_tile(tile_num(dude_obj), tile_num(self_obj));
+                GVar10 := rotation_to_tile(dude_tile, tile_num(self_obj));
                 animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(self_obj), GVar10, random(3, 10)), 1);
                 GVar10 := (GVar10 + 1) % 6;
-                while ((anim_busy(self_obj) == 0) and (GVar10 != rotation_to_tile(tile_num(dude_obj), tile_num(self_obj)))) do begin
+                while ((anim_busy(self_obj) == 0) and (GVar10 != rotation_to_tile(dude_tile, tile_num(self_obj)))) do begin
                     debug_msg("flee loop: rot == " + GVar10);
                     animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(self_obj), GVar10, random(3, 10)), 1);
                     GVar10 := (GVar10 + 1) % 6;
                 end
             end
         end
-        set_global_var(446, global_var(446) bwor 33554432);
+        set_global_var(GVAR_DEN_FLAG_2, global_var(GVAR_DEN_FLAG_2) bwor 33554432);
         if ((obj_pid(self_obj) == 16777219) or (obj_pid(self_obj) == 16777220)) then begin
-            float_msg(self_obj, message_str(932, random(300, 308)), 8);
+            float_msg(self_obj, mstr(random(300, 308)), FLOAT_MSG_YELLOW);
         end
         else begin
-            float_msg(self_obj, message_str(932, random(350, 363)), 8);
+            float_msg(self_obj, mstr(random(350, 363)), FLOAT_MSG_YELLOW);
         end
         set_local_var(0, local_var(0) bwor 1);
     end
@@ -263,58 +263,58 @@ begin
     if (club_slave_object == self_obj) then begin
         club_slave_object := 0;
     end
-    set_global_var(445, global_var(445) bwor 67108864);
-    set_global_var(33, global_var(33) + 1);
-    set_global_var(31, global_var(31) - 1);
+    set_global_var(GVAR_DEN_FLAG_1, global_var(GVAR_DEN_FLAG_1) bwor 67108864);
+    set_global_var(GVAR_SLAVES_LOST, global_var(GVAR_SLAVES_LOST) + 1);
+    set_global_var(GVAR_SLAVES_COUNT, global_var(GVAR_SLAVES_COUNT) - 1);
     if (source_obj == dude_obj) then begin
-        set_global_var(5, global_var(5) + 1);
-        if (metarule(51, self_obj) == 2) then begin
+        set_global_var(GVAR_GOOD_MONSTER, global_var(GVAR_GOOD_MONSTER) + 1);
+        if (critter_kill_type(self_obj) == KILL_TYPE_children_kills) then begin
             if (source_obj == dude_obj) then begin
-                set_global_var(2, 0);
-                set_global_var(1, global_var(1) + 1);
-                set_global_var(0, global_var(0) - -15);
-                set_global_var(37, 0);
-                set_global_var(38, 0);
-                set_global_var(39, 0);
-                set_global_var(40, 0);
-                set_global_var(41, 0);
-                set_global_var(42, 0);
-                set_global_var(43, 0);
-                set_global_var(44, 0);
-                set_global_var(45, 0);
-                if (global_var(0) >= 1000) then begin
-                    set_global_var(37, 1);
+                set_global_var(GVAR_CHAMPION_REPUTATION, 0);
+                set_global_var(GVAR_CHILDKILLER_REPUTATION, global_var(GVAR_CHILDKILLER_REPUTATION) + 1);
+                set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) - -15);
+                set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+                set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+                set_global_var(GVAR_KARMA_DEFENDER, 0);
+                set_global_var(GVAR_KARMA_WANDERER, 0);
+                set_global_var(GVAR_KARMA_BETRAYER, 0);
+                set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+                set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+                set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+                    set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
                 end
                 else begin
-                    if (global_var(0) >= 750) then begin
-                        set_global_var(38, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
                     end
                     else begin
-                        if (global_var(0) >= 500) then begin
-                            set_global_var(39, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                            set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                         end
                         else begin
-                            if (global_var(0) >= 250) then begin
-                                set_global_var(40, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                                set_global_var(GVAR_KARMA_DEFENDER, 1);
                             end
                             else begin
-                                if (global_var(0) > -250) then begin
-                                    set_global_var(41, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                                    set_global_var(GVAR_KARMA_WANDERER, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -500) then begin
-                                        set_global_var(42, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                        set_global_var(GVAR_KARMA_BETRAYER, 1);
                                     end
                                     else begin
-                                        if (global_var(0) > -750) then begin
-                                            set_global_var(43, 1);
+                                        if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                            set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                         end
                                         else begin
-                                            if (global_var(0) > -1000) then begin
-                                                set_global_var(44, 1);
+                                            if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                                set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                             end
                                             else begin
-                                                set_global_var(45, 1);
+                                                set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                             end
                                         end
                                     end
@@ -325,58 +325,58 @@ begin
                 end
             end
         end
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(4) > (3 * global_var(5))) or (global_var(2) == 1)) and (global_var(1) == 0)) then begin
-            set_global_var(2, 1);
-            set_global_var(3, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(4) > (3 * global_var(GVAR_GOOD_MONSTER))) or (global_var(GVAR_CHAMPION_REPUTATION) == 1)) and (global_var(GVAR_CHILDKILLER_REPUTATION) == 0)) then begin
+            set_global_var(GVAR_CHAMPION_REPUTATION, 1);
+            set_global_var(GVAR_BERSERKER_REPUTATION, 0);
         end
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(5) > (2 * global_var(4))) or (global_var(3) == 1))) then begin
-            set_global_var(3, 1);
-            set_global_var(2, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
+            set_global_var(GVAR_BERSERKER_REPUTATION, 1);
+            set_global_var(GVAR_CHAMPION_REPUTATION, 0);
         end
-        set_global_var(0, global_var(0) + -10);
+        set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + -10);
         debug_msg("Player gains " + -10 + " Karma Points.");
-        set_global_var(37, 0);
-        set_global_var(38, 0);
-        set_global_var(39, 0);
-        set_global_var(40, 0);
-        set_global_var(41, 0);
-        set_global_var(42, 0);
-        set_global_var(43, 0);
-        set_global_var(44, 0);
-        set_global_var(45, 0);
-        if (global_var(0) >= 1000) then begin
-            set_global_var(37, 1);
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+        set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+        set_global_var(GVAR_KARMA_DEFENDER, 0);
+        set_global_var(GVAR_KARMA_WANDERER, 0);
+        set_global_var(GVAR_KARMA_BETRAYER, 0);
+        set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+        set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+            set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
         end
         else begin
-            if (global_var(0) >= 750) then begin
-                set_global_var(38, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
             end
             else begin
-                if (global_var(0) >= 500) then begin
-                    set_global_var(39, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                 end
                 else begin
-                    if (global_var(0) >= 250) then begin
-                        set_global_var(40, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                        set_global_var(GVAR_KARMA_DEFENDER, 1);
                     end
                     else begin
-                        if (global_var(0) > -250) then begin
-                            set_global_var(41, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                            set_global_var(GVAR_KARMA_WANDERER, 1);
                         end
                         else begin
-                            if (global_var(0) > -500) then begin
-                                set_global_var(42, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                set_global_var(GVAR_KARMA_BETRAYER, 1);
                             end
                             else begin
-                                if (global_var(0) > -750) then begin
-                                    set_global_var(43, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -1000) then begin
-                                        set_global_var(44, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                     end
                                     else begin
-                                        set_global_var(45, 1);
+                                        set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                     end
                                 end
                             end
@@ -392,10 +392,10 @@ procedure look_at_p_proc
 begin
     script_overrides;
     if ((obj_pid(self_obj) == 16777219) or (obj_pid(self_obj) == 16777220)) then begin
-        display_msg(message_str(932, 100 + (get_critter_stat(self_obj, 34) == 1)));
+        display_mstr(100 + (self_gender == 1));
     end
     else begin
-        display_msg(message_str(932, 200 + (10 * (get_critter_stat(self_obj, 34) == 1)) + local_var(1)));
+        display_mstr(200 + (10 * (self_gender == 1)) + local_var(1));
     end
 end
 
@@ -403,10 +403,10 @@ procedure description_p_proc
 begin
     script_overrides;
     if ((obj_pid(self_obj) == 16777219) or (obj_pid(self_obj) == 16777220)) then begin
-        display_msg(message_str(932, 100 + (get_critter_stat(self_obj, 34) == 1)));
+        display_mstr(100 + (self_gender == 1));
     end
     else begin
-        display_msg(message_str(932, 200 + (10 * (get_critter_stat(self_obj, 34) == 1)) + local_var(1)));
+        display_mstr(200 + (10 * (self_gender == 1)) + local_var(1));
     end
     set_local_var(0, local_var(0) bwor 4);
 end
@@ -436,10 +436,10 @@ begin
             critter_set_flee_state(self_obj, 1);
         end
     end
-    if (metarule(22, 0) == 0) then begin
-        metarule3(100, self_obj, 0, 0);
+    if (is_loading_game == 0) then begin
+        rm_fixed_timer_event(self_obj, 0);
         add_timer_event(self_obj, game_ticks(random(10, 30)), 0);
-        metarule3(100, self_obj, 1, 0);
+        rm_fixed_timer_event(self_obj, 1);
         add_timer_event(self_obj, game_ticks(random(15, 60)), 1);
     end
 end
@@ -456,24 +456,24 @@ begin
             if (combat_is_initialized == 0) then begin
                 if ((critter_state(self_obj) bwand 2) == 0) then begin
                     if ((obj_pid(self_obj) == 16777219) or (obj_pid(self_obj) == 16777220)) then begin
-                        float_msg(self_obj, message_str(932, random(270, 276)), 8);
+                        float_msg(self_obj, mstr(random(270, 276)), FLOAT_MSG_YELLOW);
                     end
                     else begin
-                        float_msg(self_obj, message_str(932, random(280, 285)), 8);
+                        float_msg(self_obj, mstr(random(280, 285)), FLOAT_MSG_YELLOW);
                     end
                 end
             end
         end
-        metarule3(100, self_obj, 0, 0);
+        rm_fixed_timer_event(self_obj, 0);
         add_timer_event(self_obj, game_ticks(random(10, 30)), 0);
         if ((critter_state(self_obj) bwand 2) == 0) then begin
             if (club_slave_object == 0) then begin
-                set_global_var(446, global_var(446) bwand (-1 - 67108864));
+                set_global_var(GVAR_DEN_FLAG_2, global_var(GVAR_DEN_FLAG_2) bwand (-1 - 67108864));
                 club_slave_object := self_obj;
             end
             else begin
                 if ((critter_state(club_slave_object) bwand 2) or (tile_distance_objs(self_obj, dude_obj) < tile_distance_objs(club_slave_object, dude_obj))) then begin
-                    set_global_var(446, global_var(446) bwand (-1 - 67108864));
+                    set_global_var(GVAR_DEN_FLAG_2, global_var(GVAR_DEN_FLAG_2) bwand (-1 - 67108864));
                     club_slave_object := self_obj;
                 end
             end
@@ -493,7 +493,7 @@ begin
                             set_local_var(2, tile_num(self_obj));
                         end
                         if (tile_distance(tile_num(self_obj), local_var(2)) > 10) then begin
-                            animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(self_obj), (has_trait(1, self_obj, 10) + random(5, 7)) % 6, random(5, 10)), 0);
+                            animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(self_obj), (obj_get_rot(self_obj) + random(5, 7)) % 6, random(5, 10)), 0);
                         end
                         else begin
                             animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(self_obj), rotation_to_tile(tile_num(self_obj), local_var(2)), random(5, 10)), 0);
@@ -501,16 +501,16 @@ begin
                     end
                 end
             end
-            metarule3(100, self_obj, 1, 0);
+            rm_fixed_timer_event(self_obj, 1);
             add_timer_event(self_obj, game_ticks(random(15, 60)), 1);
             if ((critter_state(self_obj) bwand 2) == 0) then begin
                 if (club_slave_object == 0) then begin
-                    set_global_var(446, global_var(446) bwand (-1 - 67108864));
+                    set_global_var(GVAR_DEN_FLAG_2, global_var(GVAR_DEN_FLAG_2) bwand (-1 - 67108864));
                     club_slave_object := self_obj;
                 end
                 else begin
                     if ((critter_state(club_slave_object) bwand 2) or (tile_distance_objs(self_obj, dude_obj) < tile_distance_objs(club_slave_object, dude_obj))) then begin
-                        set_global_var(446, global_var(446) bwand (-1 - 67108864));
+                        set_global_var(GVAR_DEN_FLAG_2, global_var(GVAR_DEN_FLAG_2) bwand (-1 - 67108864));
                         club_slave_object := self_obj;
                     end
                 end
@@ -548,12 +548,12 @@ procedure combat_p_proc
 begin
     if ((critter_state(self_obj) bwand 2) == 0) then begin
         if (club_slave_object == 0) then begin
-            set_global_var(446, global_var(446) bwand (-1 - 67108864));
+            set_global_var(GVAR_DEN_FLAG_2, global_var(GVAR_DEN_FLAG_2) bwand (-1 - 67108864));
             club_slave_object := self_obj;
         end
         else begin
             if ((critter_state(club_slave_object) bwand 2) or (tile_distance_objs(self_obj, dude_obj) < tile_distance_objs(club_slave_object, dude_obj))) then begin
-                set_global_var(446, global_var(446) bwand (-1 - 67108864));
+                set_global_var(GVAR_DEN_FLAG_2, global_var(GVAR_DEN_FLAG_2) bwand (-1 - 67108864));
                 club_slave_object := self_obj;
             end
         end
@@ -585,12 +585,12 @@ procedure map_update_p_proc
 begin
     if ((critter_state(self_obj) bwand 2) == 0) then begin
         if (club_slave_object == 0) then begin
-            set_global_var(446, global_var(446) bwand (-1 - 67108864));
+            set_global_var(GVAR_DEN_FLAG_2, global_var(GVAR_DEN_FLAG_2) bwand (-1 - 67108864));
             club_slave_object := self_obj;
         end
         else begin
             if ((critter_state(club_slave_object) bwand 2) or (tile_distance_objs(self_obj, dude_obj) < tile_distance_objs(club_slave_object, dude_obj))) then begin
-                set_global_var(446, global_var(446) bwand (-1 - 67108864));
+                set_global_var(GVAR_DEN_FLAG_2, global_var(GVAR_DEN_FLAG_2) bwand (-1 - 67108864));
                 club_slave_object := self_obj;
             end
         end
@@ -601,4 +601,3 @@ begin
         end
     end
 end
-
