diff --git a/Games/Fallout2.Steam/Original/DCSLVGRT.ssl b/Games/Fallout2.Steam/ReDefine/DCSLVGRT.ssl
index e00f1ef..0dbb93b 100644
--- a/Games/Fallout2.Steam/Original/DCSLVGRT.ssl
+++ b/Games/Fallout2.Steam/ReDefine/DCSLVGRT.ssl
@@ -180,15 +180,15 @@ end
 procedure critter_p_proc
 begin
     if (obj_can_see_obj(self_obj, dude_obj)) then begin
-        if (((local_var(4) bwand 2) != 0) or ((global_var(446) bwand 16777216) != 0)) then begin
+        if (((local_var(4) bwand 2) != 0) or ((global_var(GVAR_DEN_FLAG_2) bwand 16777216) != 0)) then begin
             if (critter_is_fleeing(self_obj)) then begin
                 if (anim_busy(self_obj) == 0) then begin
                     if ((tile_distance_objs(self_obj, dude_obj) < 8) or obj_can_see_obj(self_obj, dude_obj)) then begin
                         if (anim_busy(self_obj) == 0) then begin
-                            global_temp := rotation_to_tile(tile_num(dude_obj), tile_num(self_obj));
+                            global_temp := rotation_to_tile(dude_tile, tile_num(self_obj));
                             animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(self_obj), global_temp, random(3, 10)), 1);
                             global_temp := (global_temp + 1) % 6;
-                            while ((anim_busy(self_obj) == 0) and (global_temp != rotation_to_tile(tile_num(dude_obj), tile_num(self_obj)))) do begin
+                            while ((anim_busy(self_obj) == 0) and (global_temp != rotation_to_tile(dude_tile, tile_num(self_obj)))) do begin
                                 debug_msg("flee loop: rot == " + global_temp);
                                 animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(self_obj), global_temp, random(3, 10)), 1);
                                 global_temp := (global_temp + 1) % 6;
@@ -198,7 +198,7 @@ begin
                 end
             end
             else begin
-                attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+                attack(dude_obj);
             end
         end
         else begin
@@ -218,7 +218,7 @@ begin
             animate_move_obj_to_tile(self_obj, local_var(5), 0);
         end
         else begin
-            if (has_trait(1, self_obj, 10) != local_var(6)) then begin
+            if (obj_get_rot(self_obj) != local_var(6)) then begin
                 anim(self_obj, 1000, local_var(6));
             end
         end
@@ -235,16 +235,16 @@ end
 procedure talk_p_proc
 begin
     from_where_to_go := 0;
-    if (((local_var(4) bwand 2) != 0) or ((global_var(446) bwand 16777216) != 0)) then begin
+    if (((local_var(4) bwand 2) != 0) or ((global_var(GVAR_DEN_FLAG_2) bwand 16777216) != 0)) then begin
         call Node001();
         if (critter_is_fleeing(self_obj)) then begin
             if (anim_busy(self_obj) == 0) then begin
                 if ((tile_distance_objs(self_obj, dude_obj) < 8) or obj_can_see_obj(self_obj, dude_obj)) then begin
                     if (anim_busy(self_obj) == 0) then begin
-                        global_temp := rotation_to_tile(tile_num(dude_obj), tile_num(self_obj));
+                        global_temp := rotation_to_tile(dude_tile, tile_num(self_obj));
                         animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(self_obj), global_temp, random(3, 10)), 1);
                         global_temp := (global_temp + 1) % 6;
-                        while ((anim_busy(self_obj) == 0) and (global_temp != rotation_to_tile(tile_num(dude_obj), tile_num(self_obj)))) do begin
+                        while ((anim_busy(self_obj) == 0) and (global_temp != rotation_to_tile(dude_tile, tile_num(self_obj)))) do begin
                             debug_msg("flee loop: rot == " + global_temp);
                             animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(self_obj), global_temp, random(3, 10)), 1);
                             global_temp := (global_temp + 1) % 6;
@@ -254,7 +254,7 @@ begin
             end
         end
         else begin
-            attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+            attack(dude_obj);
         end
     end
     else begin
@@ -294,54 +294,54 @@ end
 procedure destroy_p_proc
 begin
     if (source_obj == dude_obj) then begin
-        set_global_var(4, global_var(4) + 1);
-        if (metarule(51, self_obj) == 2) then begin
+        set_global_var(GVAR_BAD_MONSTER, global_var(GVAR_BAD_MONSTER) + 1);
+        if (critter_kill_type(self_obj) == KILL_TYPE_children_kills) then begin
             if (source_obj == dude_obj) then begin
-                set_global_var(2, 0);
-                set_global_var(1, global_var(1) + 1);
-                set_global_var(0, global_var(0) - -15);
-                set_global_var(37, 0);
-                set_global_var(38, 0);
-                set_global_var(39, 0);
-                set_global_var(40, 0);
-                set_global_var(41, 0);
-                set_global_var(42, 0);
-                set_global_var(43, 0);
-                set_global_var(44, 0);
-                set_global_var(45, 0);
-                if (global_var(0) >= 1000) then begin
-                    set_global_var(37, 1);
+                set_global_var(GVAR_CHAMPION_REPUTATION, 0);
+                set_global_var(GVAR_CHILDKILLER_REPUTATION, global_var(GVAR_CHILDKILLER_REPUTATION) + 1);
+                set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) - -15);
+                set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+                set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+                set_global_var(GVAR_KARMA_DEFENDER, 0);
+                set_global_var(GVAR_KARMA_WANDERER, 0);
+                set_global_var(GVAR_KARMA_BETRAYER, 0);
+                set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+                set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+                set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+                    set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
                 end
                 else begin
-                    if (global_var(0) >= 750) then begin
-                        set_global_var(38, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
                     end
                     else begin
-                        if (global_var(0) >= 500) then begin
-                            set_global_var(39, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                            set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                         end
                         else begin
-                            if (global_var(0) >= 250) then begin
-                                set_global_var(40, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                                set_global_var(GVAR_KARMA_DEFENDER, 1);
                             end
                             else begin
-                                if (global_var(0) > -250) then begin
-                                    set_global_var(41, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                                    set_global_var(GVAR_KARMA_WANDERER, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -500) then begin
-                                        set_global_var(42, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                        set_global_var(GVAR_KARMA_BETRAYER, 1);
                                     end
                                     else begin
-                                        if (global_var(0) > -750) then begin
-                                            set_global_var(43, 1);
+                                        if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                            set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                         end
                                         else begin
-                                            if (global_var(0) > -1000) then begin
-                                                set_global_var(44, 1);
+                                            if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                                set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                             end
                                             else begin
-                                                set_global_var(45, 1);
+                                                set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                             end
                                         end
                                     end
@@ -352,58 +352,58 @@ begin
                 end
             end
         end
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(5) > (2 * global_var(4))) or (global_var(3) == 1))) then begin
-            set_global_var(3, 1);
-            set_global_var(2, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
+            set_global_var(GVAR_BERSERKER_REPUTATION, 1);
+            set_global_var(GVAR_CHAMPION_REPUTATION, 0);
         end
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(4) > (3 * global_var(5))) or (global_var(2) == 1)) and (global_var(1) == 0)) then begin
-            set_global_var(2, 1);
-            set_global_var(3, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(4) > (3 * global_var(GVAR_GOOD_MONSTER))) or (global_var(GVAR_CHAMPION_REPUTATION) == 1)) and (global_var(GVAR_CHILDKILLER_REPUTATION) == 0)) then begin
+            set_global_var(GVAR_CHAMPION_REPUTATION, 1);
+            set_global_var(GVAR_BERSERKER_REPUTATION, 0);
         end
-        set_global_var(0, global_var(0) + 5);
+        set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + 5);
         debug_msg("Player gains " + 5 + " Karma Points.");
-        set_global_var(37, 0);
-        set_global_var(38, 0);
-        set_global_var(39, 0);
-        set_global_var(40, 0);
-        set_global_var(41, 0);
-        set_global_var(42, 0);
-        set_global_var(43, 0);
-        set_global_var(44, 0);
-        set_global_var(45, 0);
-        if (global_var(0) >= 1000) then begin
-            set_global_var(37, 1);
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+        set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+        set_global_var(GVAR_KARMA_DEFENDER, 0);
+        set_global_var(GVAR_KARMA_WANDERER, 0);
+        set_global_var(GVAR_KARMA_BETRAYER, 0);
+        set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+        set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+            set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
         end
         else begin
-            if (global_var(0) >= 750) then begin
-                set_global_var(38, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
             end
             else begin
-                if (global_var(0) >= 500) then begin
-                    set_global_var(39, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                 end
                 else begin
-                    if (global_var(0) >= 250) then begin
-                        set_global_var(40, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                        set_global_var(GVAR_KARMA_DEFENDER, 1);
                     end
                     else begin
-                        if (global_var(0) > -250) then begin
-                            set_global_var(41, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                            set_global_var(GVAR_KARMA_WANDERER, 1);
                         end
                         else begin
-                            if (global_var(0) > -500) then begin
-                                set_global_var(42, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                set_global_var(GVAR_KARMA_BETRAYER, 1);
                             end
                             else begin
-                                if (global_var(0) > -750) then begin
-                                    set_global_var(43, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -1000) then begin
-                                        set_global_var(44, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                     end
                                     else begin
-                                        set_global_var(45, 1);
+                                        set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                     end
                                 end
                             end
@@ -418,13 +418,13 @@ end
 procedure look_at_p_proc
 begin
     script_overrides;
-    display_msg(message_str(915, 100));
+    display_mstr(100);
 end
 
 procedure description_p_proc
 begin
     script_overrides;
-    display_msg(message_str(915, 150));
+    display_mstr(150);
     set_local_var(4, local_var(4) bwor 4);
 end
 
@@ -437,23 +437,23 @@ begin
     if (source_obj == dude_obj) then begin
         set_local_var(4, local_var(4) bwor 8);
         set_local_var(4, local_var(4) bwor 2);
-        set_global_var(446, global_var(446) bwor 16777216);
+        set_global_var(GVAR_DEN_FLAG_2, global_var(GVAR_DEN_FLAG_2) bwor 16777216);
     end
 end
 
 procedure map_enter_p_proc
 begin
-    if (metarule(22, 0) == 0) then begin
-        if (metarule(14, 0)) then begin
-            critter_add_trait(self_obj, 1, 6, 12);
-            critter_add_trait(self_obj, 1, 5, 16);
+    if (is_loading_game == 0) then begin
+        if (map_first_run) then begin
+            critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 12);
+            critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_SLAVER);
             set_local_var(5, tile_num(self_obj));
-            set_local_var(6, has_trait(1, self_obj, 10));
+            set_local_var(6, obj_get_rot(self_obj));
         end
         if (tile_num(self_obj) != local_var(5)) then begin
             critter_attempt_placement(self_obj, local_var(5), elevation(self_obj));
         end
-        if (has_trait(1, self_obj, 10) != local_var(6)) then begin
+        if (obj_get_rot(self_obj) != local_var(6)) then begin
             anim(self_obj, 1000, local_var(6));
         end
     end
@@ -475,7 +475,7 @@ end
 procedure Node998
 begin
     set_local_var(4, local_var(4) bwor 2);
-    set_global_var(446, global_var(446) bwor 16777216);
+    set_global_var(GVAR_DEN_FLAG_2, global_var(GVAR_DEN_FLAG_2) bwor 16777216);
 end
 
 procedure Node999
@@ -484,253 +484,252 @@ end
 
 procedure Node001
 begin
-    gsay_reply(915, 200);
-    giq_option(4, 915, 201, Node004, 50);
-    giq_option(4, 915, 202, Node004, 50);
-    giq_option(-3, 915, 203, Node004, 50);
-    giq_option(-3, 915, 204, Node007, 50);
+    Reply(200);
+    giq_option(4, NAME, 201, Node004, NEUTRAL_REACTION);
+    giq_option(4, NAME, 202, Node004, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 203, Node004, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 204, Node007, NEUTRAL_REACTION);
 end
 
 procedure Node002
 begin
-    set_global_var(445, global_var(445) bwor 32768);
+    set_global_var(GVAR_DEN_FLAG_1, global_var(GVAR_DEN_FLAG_1) bwor 32768);
     if ((local_var(4) bwand 1024) != 0) then begin
-        gsay_message(915, 217, 51);
+        gsay_message(NAME, 217, BAD_REACTION);
     end
     else begin
         set_local_var(4, local_var(4) bwor 1024);
-        gsay_reply(915, 215 + (global_var(0) >= 0));
-        if ((global_var(11) == 1) == 0) then begin
-            giq_option(4, 915, 218, Node010, 50);
+        Reply(215 + (global_var(GVAR_PLAYER_REPUTATION) >= 0));
+        if ((global_var(GVAR_REPUTATION_SLAVER) == 1) == 0) then begin
+            giq_option(4, NAME, 218, Node010, NEUTRAL_REACTION);
         end
-        giq_option(4, 915, 219, Node004, 50);
-        giq_option(4, 915, 220, Node999, 50);
-        giq_option(-3, 915, 221, Node999, 50);
-        giq_option(-3, 915, 222, Node999, 50);
+        giq_option(4, NAME, 219, Node004, NEUTRAL_REACTION);
+        giq_option(4, NAME, 220, Node999, NEUTRAL_REACTION);
+        giq_option(-3, NAME, 221, Node999, NEUTRAL_REACTION);
+        giq_option(-3, NAME, 222, Node999, NEUTRAL_REACTION);
     end
 end
 
 procedure Node003
 begin
-    float_msg(self_obj, message_str(915, 230), 8);
+    float_msg(self_obj, mstr(230), FLOAT_MSG_YELLOW);
 end
 
 procedure Node004
 begin
     if (((local_var(4) bwand 2048) != 0) == 0) then begin
         set_local_var(4, local_var(4) bwor 2048);
-        gsay_reply(915, 240 + (get_critter_stat(dude_obj, 34) == 1));
+        Reply(240 + (dude_gender == 1));
     end
     else begin
-        if ((global_var(11) == 1) and (((local_var(4) bwand 4096) != 0) == 0)) then begin
-            gsay_reply(915, 242 + (get_critter_stat(dude_obj, 34) == 1));
+        if ((global_var(GVAR_REPUTATION_SLAVER) == 1) and (((local_var(4) bwand 4096) != 0) == 0)) then begin
+            Reply(242 + (dude_gender == 1));
             set_local_var(4, local_var(4) bwor 4096);
         end
         else begin
             if (from_where_to_go) then begin
-                gsay_reply(915, 244 + (get_critter_stat(dude_obj, 34) == 1));
+                Reply(244 + (dude_gender == 1));
             end
             else begin
-                gsay_reply(915, 246);
+                Reply(246);
             end
         end
     end
     from_where_to_go := 0;
-    if (global_var(11) == 1) then begin
-        giq_option(4, 915, 247, Node990, 49);
-        giq_option(4, 915, 248, Node011, 50);
-        giq_option(-3, 915, 254, Node990, 49);
-        giq_option(-3, 915, 256, Node005, 49);
+    if (global_var(GVAR_REPUTATION_SLAVER) == 1) then begin
+        giq_option(4, NAME, 247, Node990, GOOD_REACTION);
+        giq_option(4, NAME, 248, Node011, NEUTRAL_REACTION);
+        giq_option(-3, NAME, 254, Node990, GOOD_REACTION);
+        giq_option(-3, NAME, 256, Node005, GOOD_REACTION);
     end
     if (((local_var(4) bwand 8192) != 0) == 0) then begin
-        giq_option(4, 915, 249, Node006, 50);
+        giq_option(4, NAME, 249, Node006, NEUTRAL_REACTION);
     end
-    if ((((global_var(445) bwand 32) != 0) == 0) and (((global_var(445) bwand 524288) != 0) == 0) and (((global_var(445) bwand 134217728) != 0) == 0)) then begin
-        if (global_var(452) == 2) then begin
-            giq_option(4, 915, 250, Node018, 50);
+    if ((((global_var(GVAR_DEN_FLAG_1) bwand 32) != 0) == 0) and (((global_var(GVAR_DEN_FLAG_1) bwand 524288) != 0) == 0) and (((global_var(GVAR_DEN_FLAG_1) bwand 134217728) != 0) == 0)) then begin
+        if (global_var(GVAR_DEN_VIC_KNOWN) == 2) then begin
+            giq_option(4, NAME, 250, Node018, NEUTRAL_REACTION);
         end
         else begin
-            if (global_var(452) == 1) then begin
-                giq_option(4, 915, 251, Node018, 50);
+            if (global_var(GVAR_DEN_VIC_KNOWN) == 1) then begin
+                giq_option(4, NAME, 251, Node018, NEUTRAL_REACTION);
             end
         end
     end
     if (((local_var(4) bwand 524288) != 0) == 0) then begin
-        giq_option(4, 915, 252, Node017, 50);
+        giq_option(4, NAME, 252, Node017, NEUTRAL_REACTION);
     end
-    giq_option(4, 915, 253, Node999, 50);
-    if (((global_var(445) bwand 64) != 0) == 0) then begin
-        giq_option(-3, 915, 255, Node015, 50);
+    giq_option(4, NAME, 253, Node999, NEUTRAL_REACTION);
+    if (((global_var(GVAR_DEN_FLAG_1) bwand 64) != 0) == 0) then begin
+        giq_option(-3, NAME, 255, Node015, NEUTRAL_REACTION);
     end
-    giq_option(-3, 915, 257, Node007, 50);
+    giq_option(-3, NAME, 257, Node007, NEUTRAL_REACTION);
 end
 
 procedure Node005
 begin
-    set_global_var(445, global_var(445) bwor 32768);
-    gsay_reply(915, 270 + (get_critter_stat(dude_obj, 34) == 1));
-    giq_option(1, 915, 272, Node999, 50);
-    giq_option(1, 915, 273, Node999, 50);
-    giq_option(1, 915, 274, Node999, 50);
+    set_global_var(GVAR_DEN_FLAG_1, global_var(GVAR_DEN_FLAG_1) bwor 32768);
+    Reply(270 + (dude_gender == 1));
+    giq_option(1, NAME, 272, Node999, NEUTRAL_REACTION);
+    giq_option(1, NAME, 273, Node999, NEUTRAL_REACTION);
+    giq_option(1, NAME, 274, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node006
 begin
-    set_global_var(445, global_var(445) bwor 32768);
+    set_global_var(GVAR_DEN_FLAG_1, global_var(GVAR_DEN_FLAG_1) bwor 32768);
     set_local_var(4, local_var(4) bwor 8192);
-    gsay_reply(915, 280);
-    if ((global_var(11) == 1) == 0) then begin
-        giq_option(4, 915, 281, Node010, 50);
+    Reply(280);
+    if ((global_var(GVAR_REPUTATION_SLAVER) == 1) == 0) then begin
+        giq_option(4, NAME, 281, Node010, NEUTRAL_REACTION);
     end
-    giq_option(4, 915, 282, Node004, 50);
-    if ((((global_var(445) bwand 32) != 0) == 0) and (((global_var(445) bwand 524288) != 0) == 0) and (((global_var(445) bwand 134217728) != 0) == 0)) then begin
-        if (global_var(452) == 2) then begin
-            giq_option(4, 915, 284, Node008, 50);
+    giq_option(4, NAME, 282, Node004, NEUTRAL_REACTION);
+    if ((((global_var(GVAR_DEN_FLAG_1) bwand 32) != 0) == 0) and (((global_var(GVAR_DEN_FLAG_1) bwand 524288) != 0) == 0) and (((global_var(GVAR_DEN_FLAG_1) bwand 134217728) != 0) == 0)) then begin
+        if (global_var(GVAR_DEN_VIC_KNOWN) == 2) then begin
+            giq_option(4, NAME, 284, Node008, NEUTRAL_REACTION);
         end
     end
-    giq_option(4, 915, 283, Node999, 50);
+    giq_option(4, NAME, 283, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node007
 begin
-    gsay_reply(915, 290 + (get_critter_stat(dude_obj, 34) == 1));
-    giq_option(1, 915, 292, Node999, 50);
-    giq_option(1, 915, 293, Node999, 50);
+    Reply(290 + (dude_gender == 1));
+    giq_option(1, NAME, 292, Node999, NEUTRAL_REACTION);
+    giq_option(1, NAME, 293, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node008
 begin
-    set_global_var(445, global_var(445) bwor 32768);
-    gsay_reply(915, 300 + ((global_var(446) bwand 4194304) != 0));
-    if ((global_var(11) == 1) == 0) then begin
-        giq_option(4, 915, 302, Node010, 50);
-        giq_option(4, 915, 303, Node009, 50);
+    set_global_var(GVAR_DEN_FLAG_1, global_var(GVAR_DEN_FLAG_1) bwor 32768);
+    Reply(300 + ((global_var(GVAR_DEN_FLAG_2) bwand 4194304) != 0));
+    if ((global_var(GVAR_REPUTATION_SLAVER) == 1) == 0) then begin
+        giq_option(4, NAME, 302, Node010, NEUTRAL_REACTION);
+        giq_option(4, NAME, 303, Node009, NEUTRAL_REACTION);
     end
-    giq_option(4, 915, 304, Node999, 50);
+    giq_option(4, NAME, 304, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node009
 begin
-    gsay_reply(915, 310 + ((local_var(4) bwand 16384) != 0));
+    Reply(310 + ((local_var(4) bwand 16384) != 0));
     set_local_var(4, local_var(4) bwor 16384);
-    giq_option(4, 915, 312, Node999, 50);
-    giq_option(4, 915, 313, Node999, 49);
+    giq_option(4, NAME, 312, Node999, NEUTRAL_REACTION);
+    giq_option(4, NAME, 313, Node999, GOOD_REACTION);
 end
 
 procedure Node010
 begin
-    gsay_reply(915, 320 + ((local_var(4) bwand 32768) != 0));
+    Reply(320 + ((local_var(4) bwand 32768) != 0));
     set_local_var(4, local_var(4) bwor 32768);
-    giq_option(4, 915, 322, Node004, 50);
-    giq_option(4, 915, 323, Node999, 50);
+    giq_option(4, NAME, 322, Node004, NEUTRAL_REACTION);
+    giq_option(4, NAME, 323, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node011
 begin
-    gsay_reply(915, 330 + ((local_var(4) bwand 65536) != 0));
+    Reply(330 + ((local_var(4) bwand 65536) != 0));
     set_local_var(4, local_var(4) bwor 65536);
-    giq_option(4, 915, 332, Node012, 51);
-    giq_option(4, 915, 333, Node004, 50);
-    giq_option(4, 915, 334, Node999, 50);
+    giq_option(4, NAME, 332, Node012, BAD_REACTION);
+    giq_option(4, NAME, 333, Node004, NEUTRAL_REACTION);
+    giq_option(4, NAME, 334, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node012
 begin
     if ((local_var(4) bwand 131072) != 0) then begin
-        gsay_reply(915, 342);
-        giq_option(4, 915, 343, Node998, 51);
-        giq_option(4, 915, 344, Node998, 51);
+        Reply(342);
+        giq_option(4, NAME, 343, Node998, BAD_REACTION);
+        giq_option(4, NAME, 344, Node998, BAD_REACTION);
     end
     else begin
-        gsay_reply(915, 340 + (get_critter_stat(dude_obj, 34) == 1));
+        Reply(340 + (dude_gender == 1));
         set_local_var(4, local_var(4) bwor 131072);
-        giq_option(4, 915, 343, Node012, 51);
-        giq_option(4, 915, 344, Node014, 50);
-        giq_option(4, 915, 345, Node013, 51);
+        giq_option(4, NAME, 343, Node012, BAD_REACTION);
+        giq_option(4, NAME, 344, Node014, NEUTRAL_REACTION);
+        giq_option(4, NAME, 345, Node013, BAD_REACTION);
     end
 end
 
 procedure Node013
 begin
-    gsay_reply(915, 350);
-    giq_option(4, 915, 351, Node012, 51);
-    giq_option(4, 915, 352, Node012, 51);
-    giq_option(4, 915, 353, Node014, 50);
-    giq_option(4, 915, 354, Node014, 50);
+    Reply(350);
+    giq_option(4, NAME, 351, Node012, BAD_REACTION);
+    giq_option(4, NAME, 352, Node012, BAD_REACTION);
+    giq_option(4, NAME, 353, Node014, NEUTRAL_REACTION);
+    giq_option(4, NAME, 354, Node014, NEUTRAL_REACTION);
 end
 
 procedure Node014
 begin
-    gsay_reply(915, 360);
-    giq_option(4, 915, 361, Node999, 50);
+    Reply(360);
+    giq_option(4, NAME, 361, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node015
 begin
-    gsay_reply(915, 370);
-    giq_option(1, 915, 371, Node999, 50);
+    Reply(370);
+    giq_option(1, NAME, 371, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node016
 begin
     if ((local_var(4) bwand 262144) != 0) then begin
-        gsay_message(915, 382, 50);
+        gsay_message(NAME, 382, NEUTRAL_REACTION);
     end
     else begin
         set_local_var(4, local_var(4) bwor 262144);
-        gsay_reply(915, 380 + (get_critter_stat(dude_obj, 34) == 1));
-        if ((global_var(11) == 1) == 0) then begin
-            giq_option(4, 915, 383, Node010, 50);
+        Reply(380 + (dude_gender == 1));
+        if ((global_var(GVAR_REPUTATION_SLAVER) == 1) == 0) then begin
+            giq_option(4, NAME, 383, Node010, NEUTRAL_REACTION);
         end
-        giq_option(4, 915, 384, Node004, 50);
-        giq_option(4, 915, 385, Node999, 50);
-        giq_option(-3, 915, 386, Node999, 50);
-        giq_option(-3, 915, 387, Node999, 50);
+        giq_option(4, NAME, 384, Node004, NEUTRAL_REACTION);
+        giq_option(4, NAME, 385, Node999, NEUTRAL_REACTION);
+        giq_option(-3, NAME, 386, Node999, NEUTRAL_REACTION);
+        giq_option(-3, NAME, 387, Node999, NEUTRAL_REACTION);
     end
 end
 
 procedure Node017
 begin
     set_local_var(4, local_var(4) bwor 524288);
-    gsay_reply(915, 400);
-    giq_option(4, 915, 401, Node004, 50);
-    giq_option(4, 915, 402, Node999, 50);
+    Reply(400);
+    giq_option(4, NAME, 401, Node004, NEUTRAL_REACTION);
+    giq_option(4, NAME, 402, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node018
 begin
     if ((local_var(4) bwand 1048576) != 0) then begin
-        gsay_reply(915, 412);
+        Reply(412);
     end
     else begin
         set_local_var(4, local_var(4) bwor 1048576);
-        gsay_reply(915, 410 + (global_var(452) == 2));
+        Reply(410 + (global_var(GVAR_DEN_VIC_KNOWN) == 2));
     end
-    if ((global_var(11) == 1) == 0) then begin
-        giq_option(4, 915, 413, Node010, 50);
+    if ((global_var(GVAR_REPUTATION_SLAVER) == 1) == 0) then begin
+        giq_option(4, NAME, 413, Node010, NEUTRAL_REACTION);
     end
-    giq_option(4, 915, 414, Node004, 50);
-    giq_option(4, 915, 415, Node008, 50);
-    giq_option(4, 915, 416, Node999, 50);
+    giq_option(4, NAME, 414, Node004, NEUTRAL_REACTION);
+    giq_option(4, NAME, 415, Node008, NEUTRAL_REACTION);
+    giq_option(4, NAME, 416, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node019
 begin
-    if (global_var(11) == 1) then begin
-        float_msg(self_obj, message_str(915, random(430, 434)), 8);
+    if (global_var(GVAR_REPUTATION_SLAVER) == 1) then begin
+        float_msg(self_obj, mstr(random(430, 434)), FLOAT_MSG_YELLOW);
     end
     else begin
-        float_msg(self_obj, message_str(915, random(435, 439)), 8);
+        float_msg(self_obj, mstr(random(435, 439)), FLOAT_MSG_YELLOW);
     end
 end
 
 procedure Node990
 begin
-    if (get_critter_stat(dude_obj, 34) == 0) then begin
+    if (dude_gender == 0) then begin
         call Node002();
     end
     else begin
         call Node016();
     end
 end
-
