diff --git a/Games/Fallout2.Steam/Original/DEADCOMP.ssl b/Games/Fallout2.Steam/ReDefine/DEADCOMP.ssl
index ace32a4..7e91b60 100644
--- a/Games/Fallout2.Steam/Original/DEADCOMP.ssl
+++ b/Games/Fallout2.Steam/ReDefine/DEADCOMP.ssl
@@ -4,7 +4,7 @@ procedure do_stuff;
 
 procedure start
 begin
-    if (script_action == 6) then begin
+    if (script_action == use_proc) then begin
         script_overrides;
         call do_stuff();
     end
@@ -12,6 +12,5 @@ end
 
 procedure do_stuff
 begin
-    display_msg(message_str(62, 100));
+    display_msg(message_str(SCRIPT_DIANNGRV, 100));
 end
-
