diff --git a/Games/Fallout2.Steam/Original/DIBONES.ssl b/Games/Fallout2.Steam/ReDefine/DIBONES.ssl
index ccea218..62754bb 100644
--- a/Games/Fallout2.Steam/Original/DIBONES.ssl
+++ b/Games/Fallout2.Steam/ReDefine/DIBONES.ssl
@@ -122,12 +122,11 @@ end
 procedure look_at_p_proc
 begin
     script_overrides;
-    display_msg(message_str(65, 100));
+    display_mstr(100);
 end
 
 procedure description_p_proc
 begin
     script_overrides;
-    display_msg(message_str(65, 100));
+    display_mstr(100);
 end
-
