diff --git a/Games/Fallout2.Steam/Original/DISMIBOX.ssl b/Games/Fallout2.Steam/ReDefine/DISMIBOX.ssl
index b310b26..42b24ab 100644
--- a/Games/Fallout2.Steam/Original/DISMIBOX.ssl
+++ b/Games/Fallout2.Steam/ReDefine/DISMIBOX.ssl
@@ -140,13 +140,13 @@ end
 procedure map_enter_p_proc
 begin
     den_smitty_box_obj := self_obj;
-    if (metarule(22, 0) == 0) then begin
+    if (is_loading_game == 0) then begin
         if (local_var(0) < game_time) then begin
             restock_amt := random(0, 1);
             if (obj_is_carrying_obj_pid(self_obj, 226) < restock_amt) then begin
                 if (100 >= random(1, 100)) then begin
                     if (obj_is_carrying_obj_pid(self_obj, 226) < restock_amt) then begin
-                        restock_obj := create_object_sid(226, 0, 0, -1);
+                        restock_obj := create_object(226, 0, 0);
                         add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 226));
                     end
                     else begin
@@ -169,7 +169,7 @@ begin
                                     end
                                     else begin
                                         if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                            metarule(43, self_obj);
+                                            inven_unwield(self_obj);
                                         end
                                     end
                                 end
@@ -187,7 +187,7 @@ begin
             end
             else begin
                 if (obj_is_carrying_obj_pid(self_obj, 226) < restock_amt) then begin
-                    restock_obj := create_object_sid(226, 0, 0, -1);
+                    restock_obj := create_object(226, 0, 0);
                     add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 226));
                 end
                 else begin
@@ -210,7 +210,7 @@ begin
                                 end
                                 else begin
                                     if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                        metarule(43, self_obj);
+                                        inven_unwield(self_obj);
                                     end
                                 end
                             end
@@ -229,7 +229,7 @@ begin
             if (obj_is_carrying_obj_pid(self_obj, 124) < restock_amt) then begin
                 if (100 >= random(1, 100)) then begin
                     if (obj_is_carrying_obj_pid(self_obj, 124) < restock_amt) then begin
-                        restock_obj := create_object_sid(124, 0, 0, -1);
+                        restock_obj := create_object(124, 0, 0);
                         add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 124));
                     end
                     else begin
@@ -252,7 +252,7 @@ begin
                                     end
                                     else begin
                                         if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                            metarule(43, self_obj);
+                                            inven_unwield(self_obj);
                                         end
                                     end
                                 end
@@ -270,7 +270,7 @@ begin
             end
             else begin
                 if (obj_is_carrying_obj_pid(self_obj, 124) < restock_amt) then begin
-                    restock_obj := create_object_sid(124, 0, 0, -1);
+                    restock_obj := create_object(124, 0, 0);
                     add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 124));
                 end
                 else begin
@@ -293,7 +293,7 @@ begin
                                 end
                                 else begin
                                     if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                        metarule(43, self_obj);
+                                        inven_unwield(self_obj);
                                     end
                                 end
                             end
@@ -312,7 +312,7 @@ begin
             if (obj_is_carrying_obj_pid(self_obj, 308) < restock_amt) then begin
                 if (40 >= random(1, 100)) then begin
                     if (obj_is_carrying_obj_pid(self_obj, 308) < restock_amt) then begin
-                        restock_obj := create_object_sid(308, 0, 0, -1);
+                        restock_obj := create_object(308, 0, 0);
                         add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 308));
                     end
                     else begin
@@ -335,7 +335,7 @@ begin
                                     end
                                     else begin
                                         if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                            metarule(43, self_obj);
+                                            inven_unwield(self_obj);
                                         end
                                     end
                                 end
@@ -353,7 +353,7 @@ begin
             end
             else begin
                 if (obj_is_carrying_obj_pid(self_obj, 308) < restock_amt) then begin
-                    restock_obj := create_object_sid(308, 0, 0, -1);
+                    restock_obj := create_object(308, 0, 0);
                     add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 308));
                 end
                 else begin
@@ -376,7 +376,7 @@ begin
                                 end
                                 else begin
                                     if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                        metarule(43, self_obj);
+                                        inven_unwield(self_obj);
                                     end
                                 end
                             end
@@ -395,7 +395,7 @@ begin
             if (obj_is_carrying_obj_pid(self_obj, 4) < restock_amt) then begin
                 if (40 >= random(1, 100)) then begin
                     if (obj_is_carrying_obj_pid(self_obj, 4) < restock_amt) then begin
-                        restock_obj := create_object_sid(4, 0, 0, -1);
+                        restock_obj := create_object(4, 0, 0);
                         add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 4));
                     end
                     else begin
@@ -418,7 +418,7 @@ begin
                                     end
                                     else begin
                                         if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                            metarule(43, self_obj);
+                                            inven_unwield(self_obj);
                                         end
                                     end
                                 end
@@ -436,7 +436,7 @@ begin
             end
             else begin
                 if (obj_is_carrying_obj_pid(self_obj, 4) < restock_amt) then begin
-                    restock_obj := create_object_sid(4, 0, 0, -1);
+                    restock_obj := create_object(4, 0, 0);
                     add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 4));
                 end
                 else begin
@@ -459,7 +459,7 @@ begin
                                 end
                                 else begin
                                     if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                        metarule(43, self_obj);
+                                        inven_unwield(self_obj);
                                     end
                                 end
                             end
@@ -478,7 +478,7 @@ begin
             if (obj_is_carrying_obj_pid(self_obj, 159) < restock_amt) then begin
                 if (40 >= random(1, 100)) then begin
                     if (obj_is_carrying_obj_pid(self_obj, 159) < restock_amt) then begin
-                        restock_obj := create_object_sid(159, 0, 0, -1);
+                        restock_obj := create_object(159, 0, 0);
                         add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 159));
                     end
                     else begin
@@ -501,7 +501,7 @@ begin
                                     end
                                     else begin
                                         if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                            metarule(43, self_obj);
+                                            inven_unwield(self_obj);
                                         end
                                     end
                                 end
@@ -519,7 +519,7 @@ begin
             end
             else begin
                 if (obj_is_carrying_obj_pid(self_obj, 159) < restock_amt) then begin
-                    restock_obj := create_object_sid(159, 0, 0, -1);
+                    restock_obj := create_object(159, 0, 0);
                     add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 159));
                 end
                 else begin
@@ -542,7 +542,7 @@ begin
                                 end
                                 else begin
                                     if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                        metarule(43, self_obj);
+                                        inven_unwield(self_obj);
                                     end
                                 end
                             end
@@ -561,7 +561,7 @@ begin
             if (obj_is_carrying_obj_pid(self_obj, 39) < restock_amt) then begin
                 if (40 >= random(1, 100)) then begin
                     if (obj_is_carrying_obj_pid(self_obj, 39) < restock_amt) then begin
-                        restock_obj := create_object_sid(39, 0, 0, -1);
+                        restock_obj := create_object(39, 0, 0);
                         add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 39));
                     end
                     else begin
@@ -584,7 +584,7 @@ begin
                                     end
                                     else begin
                                         if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                            metarule(43, self_obj);
+                                            inven_unwield(self_obj);
                                         end
                                     end
                                 end
@@ -602,7 +602,7 @@ begin
             end
             else begin
                 if (obj_is_carrying_obj_pid(self_obj, 39) < restock_amt) then begin
-                    restock_obj := create_object_sid(39, 0, 0, -1);
+                    restock_obj := create_object(39, 0, 0);
                     add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 39));
                 end
                 else begin
@@ -625,7 +625,7 @@ begin
                                 end
                                 else begin
                                     if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                        metarule(43, self_obj);
+                                        inven_unwield(self_obj);
                                     end
                                 end
                             end
@@ -644,4 +644,3 @@ begin
         end
     end
 end
-
