diff --git a/Games/Fallout2.Steam/Original/DISTILL.ssl b/Games/Fallout2.Steam/ReDefine/DISTILL.ssl
index 46e4101..6b251b6 100644
--- a/Games/Fallout2.Steam/Original/DISTILL.ssl
+++ b/Games/Fallout2.Steam/ReDefine/DISTILL.ssl
@@ -149,16 +149,16 @@ end
 procedure look_at_p_proc
 begin
     script_overrides;
-    set_global_var(446, global_var(446) bwor 134217728);
-    if ((global_var(446) bwand 268435456) != 0) then begin
-        display_msg(message_str(68, 102));
+    set_global_var(GVAR_DEN_FLAG_2, global_var(GVAR_DEN_FLAG_2) bwor 134217728);
+    if ((global_var(GVAR_DEN_FLAG_2) bwand 268435456) != 0) then begin
+        display_mstr(102);
     end
     else begin
-        if (global_var(101) == 1) then begin
-            display_msg(message_str(68, 101));
+        if (global_var(GVAR_QUEST_MAGGIE_STILL) == 1) then begin
+            display_mstr(101);
         end
         else begin
-            display_msg(message_str(68, 100));
+            display_mstr(100);
         end
     end
 end
@@ -166,33 +166,33 @@ end
 procedure description_p_proc
 begin
     script_overrides;
-    set_global_var(446, global_var(446) bwor 134217728);
-    if ((global_var(446) bwand 268435456) != 0) then begin
-        display_msg(message_str(68, 102));
+    set_global_var(GVAR_DEN_FLAG_2, global_var(GVAR_DEN_FLAG_2) bwor 134217728);
+    if ((global_var(GVAR_DEN_FLAG_2) bwand 268435456) != 0) then begin
+        display_mstr(102);
     end
     else begin
-        if ((global_var(101) == 0) == 0) then begin
-            display_msg(message_str(68, 101));
+        if ((global_var(GVAR_QUEST_MAGGIE_STILL) == 0) == 0) then begin
+            display_mstr(101);
         end
         else begin
-            display_msg(message_str(68, 100));
+            display_mstr(100);
         end
     end
 end
 
 procedure damage_p_proc
 begin
-    if (((global_var(446) bwand 268435456) != 0) == 0) then begin
+    if (((global_var(GVAR_DEN_FLAG_2) bwand 268435456) != 0) == 0) then begin
         gfade_out(10);
-        if (global_var(101) == 2) then begin
-            if (3 > global_var(101)) then begin
-                set_global_var(101, 3);
+        if (global_var(GVAR_QUEST_MAGGIE_STILL) == 2) then begin
+            if (3 > global_var(GVAR_QUEST_MAGGIE_STILL)) then begin
+                set_global_var(GVAR_QUEST_MAGGIE_STILL, 3);
             end
-            debug_msg("set_still_quest(" + 3 + ") == " + global_var(101));
+            debug_msg("set_still_quest(" + 3 + ") == " + global_var(GVAR_QUEST_MAGGIE_STILL));
         end
-        set_global_var(447, global_var(447) bwor 2048);
-        set_global_var(446, global_var(446) bwor 268435456);
-        display_msg(message_str(68, 102));
+        set_global_var(GVAR_DEN_FLAG_3, global_var(GVAR_DEN_FLAG_3) bwor 2048);
+        set_global_var(GVAR_DEN_FLAG_2, global_var(GVAR_DEN_FLAG_2) bwor 268435456);
+        display_mstr(102);
         anim(self_obj, 1010, 1);
         gfade_in(10);
     end
@@ -200,7 +200,7 @@ end
 
 procedure map_enter_p_proc
 begin
-    if ((global_var(446) bwand 268435456) != 0) then begin
+    if ((global_var(GVAR_DEN_FLAG_2) bwand 268435456) != 0) then begin
         anim(self_obj, 1010, 1);
     end
     else begin
@@ -216,17 +216,16 @@ procedure use_obj_on_p_proc
 begin
     if ((obj_pid(obj_being_used_with) == 384) or (obj_pid(obj_being_used_with) == 20) or (obj_pid(obj_being_used_with) == 75)) then begin
         gfade_out(10);
-        if (global_var(101) == 2) then begin
-            if (3 > global_var(101)) then begin
-                set_global_var(101, 3);
+        if (global_var(GVAR_QUEST_MAGGIE_STILL) == 2) then begin
+            if (3 > global_var(GVAR_QUEST_MAGGIE_STILL)) then begin
+                set_global_var(GVAR_QUEST_MAGGIE_STILL, 3);
             end
-            debug_msg("set_still_quest(" + 3 + ") == " + global_var(101));
+            debug_msg("set_still_quest(" + 3 + ") == " + global_var(GVAR_QUEST_MAGGIE_STILL));
         end
-        set_global_var(447, global_var(447) bwor 2048);
-        set_global_var(446, global_var(446) bwor 268435456);
-        display_msg(message_str(68, 102));
+        set_global_var(GVAR_DEN_FLAG_3, global_var(GVAR_DEN_FLAG_3) bwor 2048);
+        set_global_var(GVAR_DEN_FLAG_2, global_var(GVAR_DEN_FLAG_2) bwor 268435456);
+        display_mstr(102);
         anim(self_obj, 1010, 1);
         gfade_in(10);
     end
 end
-
