diff --git a/Games/Fallout2.Steam/Original/ECBESS.ssl b/Games/Fallout2.Steam/ReDefine/ECBESS.ssl
index 423d944..1c45501 100644
--- a/Games/Fallout2.Steam/Original/ECBESS.ssl
+++ b/Games/Fallout2.Steam/ReDefine/ECBESS.ssl
@@ -149,13 +149,13 @@ end
 
 procedure look_at_p_proc
 begin
-    float_msg(dude_obj, message_str(644, 100), 0);
+    float_msg(dude_obj, mstr(100), FLOAT_MSG_NORMAL);
 end
 
 procedure description_p_proc
 begin
     script_overrides;
-    display_msg(message_str(644, 101));
+    display_mstr(101);
 end
 
 procedure use_skill_on_p_proc
@@ -185,4 +185,3 @@ end
 procedure timed_event_p_proc
 begin
 end
-
