diff --git a/Games/Fallout2.Steam/Original/ECCENTUR.ssl b/Games/Fallout2.Steam/ReDefine/ECCENTUR.ssl
index 8f762c9..ea1bb7a 100644
--- a/Games/Fallout2.Steam/Original/ECCENTUR.ssl
+++ b/Games/Fallout2.Steam/ReDefine/ECCENTUR.ssl
@@ -144,7 +144,7 @@ end
 procedure critter_p_proc
 begin
     if (obj_can_see_obj(self_obj, dude_obj)) then begin
-        attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+        attack(dude_obj);
     end
     if (not(anim_busy(self_obj))) then begin
         reg_anim_func(2, self_obj);
@@ -156,8 +156,8 @@ end
 
 procedure map_enter_p_proc
 begin
-    critter_add_trait(self_obj, 1, 6, 173);
-    critter_add_trait(self_obj, 1, 5, 133);
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 173);
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_CENTAUR);
     starttile := tile_num(self_obj);
     reg_anim_func(2, self_obj);
     reg_anim_func(1, 1);
@@ -179,8 +179,7 @@ end
 
 procedure destroy_p_proc
 begin
-    if (cur_map_index == 94) then begin
+    if (CUR_MAP_BHRNDDST) then begin
         set_map_var(1, map_var(1) - 1);
     end
 end
-
