diff --git a/Games/Fallout2.Steam/Original/ECCOWBMB.ssl b/Games/Fallout2.Steam/ReDefine/ECCOWBMB.ssl
index 8bb48f3..9a48a4a 100644
--- a/Games/Fallout2.Steam/Original/ECCOWBMB.ssl
+++ b/Games/Fallout2.Steam/ReDefine/ECCOWBMB.ssl
@@ -181,7 +181,7 @@ end
 
 procedure map_enter_p_proc
 begin
-    critter_add_trait(self_obj, 1, 6, 143);
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 143);
     add_timer_event(self_obj, random(10, 50), 1);
 end
 
@@ -201,7 +201,7 @@ end
 procedure combat_p_proc
 begin
     if (not(doExplosion())) then begin
-        attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+        attack(dude_obj);
     end
 end
 
@@ -211,7 +211,7 @@ begin
         if (not(anim_busy(self_obj))) then begin
             reg_anim_func(2, self_obj);
             reg_anim_func(1, 1);
-            animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(dude_obj), random(0, 5), 1), 1);
+            animate_move_obj_to_tile(self_obj, tile_num_in_direction(dude_tile, random(0, 5), 1), 1);
             reg_anim_func(3, 0);
         end
     end
@@ -220,7 +220,7 @@ end
 procedure timed_event_p_proc
 begin
     if (fixed_param == 1) then begin
-        float_msg(self_obj, message_str(203, 201 + random(0, 5)), 2);
+        float_msg(self_obj, message_str(SCRIPT_ZCBRAHMN, 201 + random(0, 5)), FLOAT_MSG_RED);
         add_timer_event(self_obj, random(10, 50), 1);
     end
 end
@@ -233,4 +233,3 @@ begin
     end
     return 0;
 end
-
