diff --git a/Games/Fallout2.Steam/Original/ECGRDFRV.ssl b/Games/Fallout2.Steam/ReDefine/ECGRDFRV.ssl
index 4c9f083..dd7a8ee 100644
--- a/Games/Fallout2.Steam/Original/ECGRDFRV.ssl
+++ b/Games/Fallout2.Steam/ReDefine/ECGRDFRV.ssl
@@ -133,8 +133,8 @@ end
 
 procedure spatial_p_proc
 begin
-    if (global_var(602) == 0) then begin
-        set_global_var(602, 1);
+    if (global_var(GVAR_ENTERED_GUARDIAN) == 0) then begin
+        set_global_var(GVAR_ENTERED_GUARDIAN, 1);
         gfade_out(600);
         move_to(dude_obj, 17512, 1);
         critter_attempt_placement(dude_obj, 17512, 1);
@@ -150,4 +150,3 @@ end
 procedure start
 begin
 end
-
