diff --git a/Games/Fallout2.Steam/Original/ECHUNTER.ssl b/Games/Fallout2.Steam/ReDefine/ECHUNTER.ssl
index fe918d0..282dc01 100644
--- a/Games/Fallout2.Steam/Original/ECHUNTER.ssl
+++ b/Games/Fallout2.Steam/ReDefine/ECHUNTER.ssl
@@ -139,24 +139,23 @@ end
 
 procedure talk_p_proc
 begin
-    float_msg(self_obj, message_str(484, random(101, 107)), 8);
+    float_msg(self_obj, mstr(random(101, 107)), FLOAT_MSG_YELLOW);
 end
 
 procedure map_enter_p_proc
 begin
-    critter_add_trait(self_obj, 1, 6, 146);
-    critter_add_trait(self_obj, 1, 5, 38);
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 146);
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_BERSERKER);
 end
 
 procedure description_p_proc
 begin
     script_overrides;
-    display_msg(message_str(484, 100));
+    display_mstr(100);
 end
 
 procedure look_at_p_proc
 begin
     script_overrides;
-    display_msg(message_str(484, 100));
+    display_mstr(100);
 end
-
