diff --git a/Games/Fallout2.Steam/Original/ECMANTIS.ssl b/Games/Fallout2.Steam/ReDefine/ECMANTIS.ssl
index b9e0f70..b72f2e9 100644
--- a/Games/Fallout2.Steam/Original/ECMANTIS.ssl
+++ b/Games/Fallout2.Steam/ReDefine/ECMANTIS.ssl
@@ -30,7 +30,7 @@ procedure critter_p_proc;
 
 procedure start
 begin
-    critter_add_trait(self_obj, 1, 6, 142);
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 142);
 end
 
 procedure checkPartyMembersNearDoor
@@ -140,7 +140,7 @@ end
 
 procedure destroy_p_proc
 begin
-    if (cur_map_index == 94) then begin
+    if (CUR_MAP_BHRNDDST) then begin
         set_map_var(1, map_var(1) - 1);
     end
 end
@@ -148,7 +148,6 @@ end
 procedure critter_p_proc
 begin
     if (obj_can_see_obj(self_obj, dude_obj) and not(critter_is_fleeing(self_obj))) then begin
-        attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+        attack(dude_obj);
     end
 end
-
