diff --git a/Games/Fallout2.Steam/Original/ECMELPET.ssl b/Games/Fallout2.Steam/ReDefine/ECMELPET.ssl
index fbc69d2..5ed1311 100644
--- a/Games/Fallout2.Steam/Original/ECMELPET.ssl
+++ b/Games/Fallout2.Steam/ReDefine/ECMELPET.ssl
@@ -151,7 +151,7 @@ end
 procedure critter_p_proc
 begin
     if (obj_can_see_obj(self_obj, dude_obj)) then begin
-        attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+        attack(dude_obj);
     end
 end
 
@@ -181,21 +181,21 @@ end
 procedure damage_p_proc
 begin
     if (source_obj == dude_obj) then begin
-        attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+        attack(dude_obj);
     end
 end
 
 procedure map_enter_p_proc
 begin
     variable LVar0 := 0;
-    if (metarule(14, 0)) then begin
+    if (map_first_run) then begin
         set_local_var(1, tile_num(self_obj));
         LVar0 := ((obj_pid(self_obj) == 16777461) * 0) + ((obj_pid(self_obj) == 16777457) * 1) + ((obj_pid(self_obj) == 16777463) * 2) + ((obj_pid(self_obj) == 16777333) * 3);
         set_map_var(LVar0, map_var(LVar0) + 1);
         set_local_var(0, map_var(LVar0));
-        debug_msg("" + obj_name(self_obj) + ": Place Num == " + local_var(0) + ". Map Var (" + LVar0 + ") == " + map_var(LVar0) + " , pid_to_ai(" + obj_pid(self_obj) + ") == " + (((obj_pid(self_obj) == 16777461) * 26) + ((obj_pid(self_obj) == 16777457) * 144) + ((obj_pid(self_obj) == 16777463) * 132) + ((obj_pid(self_obj) == 16777333) * 48)));
-        critter_add_trait(self_obj, 1, 6, 67);
-        critter_add_trait(self_obj, 1, 5, ((obj_pid(self_obj) == 16777461) * 26) + ((obj_pid(self_obj) == 16777457) * 144) + ((obj_pid(self_obj) == 16777463) * 132) + ((obj_pid(self_obj) == 16777333) * 48));
+        debug_msg("" + self_name + ": Place Num == " + local_var(0) + ". Map Var (" + LVar0 + ") == " + map_var(LVar0) + " , pid_to_ai(" + obj_pid(self_obj) + ") == " + (((obj_pid(self_obj) == 16777461) * 26) + ((obj_pid(self_obj) == 16777457) * 144) + ((obj_pid(self_obj) == 16777463) * 132) + ((obj_pid(self_obj) == 16777333) * 48)));
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 67);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, ((obj_pid(self_obj) == 16777461) * 26) + ((obj_pid(self_obj) == 16777457) * 144) + ((obj_pid(self_obj) == 16777463) * 132) + ((obj_pid(self_obj) == 16777333) * 48));
     end
 end
 
@@ -208,7 +208,7 @@ begin
     variable LVar0 := 0;
     variable LVar1 := 0;
     variable LVar2 := 0;
-    if (has_trait(1, self_obj, 666) and ((critter_state(self_obj) bwand 1) == 0)) then begin
+    if (obj_is_visible_flag(self_obj) and ((critter_state(self_obj) bwand 1) == 0)) then begin
         if (fixed_param == 2) then begin
             if (target_obj == dude_obj) then begin
                 if (obj_pid(self_obj) == 16777333) then begin
@@ -228,8 +228,8 @@ begin
                         if (((tile_contains_pid_obj(LVar2, elevation(self_obj), 33554499) != 0) == 0) and ((tile_contains_pid_obj(tile_num_in_direction(LVar2, 0, 1), elevation(self_obj), 33554499) != 0) == 0) and ((tile_contains_pid_obj(tile_num_in_direction(LVar2, 1, 1), elevation(self_obj), 33554499) != 0) == 0) and ((tile_contains_pid_obj(tile_num_in_direction(LVar2, 2, 1), elevation(self_obj), 33554499) != 0) == 0) and ((tile_contains_pid_obj(tile_num_in_direction(LVar2, 3, 1), elevation(self_obj), 33554499) != 0) == 0) and ((tile_contains_pid_obj(tile_num_in_direction(LVar2, 4, 1), elevation(self_obj), 33554499) != 0) == 0) and ((tile_contains_pid_obj(tile_num_in_direction(LVar2, 5, 1), elevation(self_obj), 33554499) != 0) == 0)) then begin
                             debug_msg("moving pet");
                             move_to(self_obj, LVar2, elevation(self_obj));
-                            anim(self_obj, 1000, rotation_to_tile(tile_num(self_obj), tile_num(dude_obj)));
-                            attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+                            anim(self_obj, 1000, rotation_to_tile(tile_num(self_obj), dude_tile));
+                            attack(dude_obj);
                         end
                         else begin
                             debug_msg("pet can't be released, clump is blocked");
@@ -245,4 +245,3 @@ end
 procedure use_obj_on_p_proc
 begin
 end
-
