diff --git a/Games/Fallout2.Steam/Original/ECPARIAH.ssl b/Games/Fallout2.Steam/ReDefine/ECPARIAH.ssl
index add8398..6fa62c5 100644
--- a/Games/Fallout2.Steam/Original/ECPARIAH.ssl
+++ b/Games/Fallout2.Steam/ReDefine/ECPARIAH.ssl
@@ -167,13 +167,13 @@ procedure critter_p_proc
 begin
     variable LVar0 := 0;
     if (party_member_obj(16777600) != 0) then begin
-        if (((local_var(9) != 0) == 0) and ((global_var(398) != 0) == 0)) then begin
+        if (((local_var(9) != 0) == 0) and ((global_var(GVAR_PARTY_NO_FOLLOW) != 0) == 0)) then begin
             if (local_var(10) == 0) then begin
                 set_local_var(10, 6);
             end
             if (tile_distance_objs(self_obj, dude_obj) > (3 * local_var(10) / 2)) then begin
                 if (anim_busy(self_obj) == 0) then begin
-                    dest_tile := tile_num_in_direction(tile_num_in_direction(tile_num(dude_obj), rotation_to_tile(tile_num(dude_obj), tile_num(self_obj)), local_var(10)), random(0, 5), random(0, 2));
+                    dest_tile := tile_num_in_direction(tile_num_in_direction(dude_tile, rotation_to_tile(dude_tile, tile_num(self_obj)), local_var(10)), random(0, 5), random(0, 2));
                     if (tile_distance_objs(self_obj, dude_obj) > (3 * local_var(10) / 2 * 2)) then begin
                         animate_move_obj_to_tile(self_obj, dest_tile, 1);
                     end
@@ -182,27 +182,27 @@ begin
                     end
                 end
                 else begin
-                    if (tile_distance(tile_num(self_obj), tile_num(dude_obj)) < tile_distance(tile_num(self_obj), dest_tile)) then begin
+                    if (tile_distance(tile_num(self_obj), dude_tile) < tile_distance(tile_num(self_obj), dest_tile)) then begin
                         reg_anim_func(2, self_obj);
                     end
                 end
             end
         end
-        if ((get_critter_stat(dude_obj, 6) > (1 + has_trait(2, self_obj, 15))) and (local_var(7) == 0)) then begin
+        if ((dude_luck > (1 + has_trait(TRAIT_TRAIT, self_obj, TRAIT_gifted))) and (local_var(7) == 0)) then begin
             if ((local_var(7) < 10) and (local_var(7) != 0)) then begin
-                set_local_var(7, local_var(7) + get_critter_stat(dude_obj, 6) - 1);
+                set_local_var(7, local_var(7) + dude_luck - 1);
             end
             else begin
-                set_local_var(7, get_critter_stat(dude_obj, 6));
+                set_local_var(7, dude_luck);
             end
-            LVar0 := 1 - get_critter_stat(dude_obj, 6);
-            if (has_trait(2, self_obj, 15)) then begin
+            LVar0 := 1 - dude_luck;
+            if (has_trait(TRAIT_TRAIT, self_obj, TRAIT_gifted)) then begin
                 LVar0 := LVar0 + 1;
             end
             debug_msg("ECPARIA: luck adjust " + LVar0);
-            set_critter_stat(dude_obj, 6, LVar0);
+            set_critter_stat(dude_obj, STAT_lu, LVar0);
             debug_msg("ECPARIA: dude's luck wasn't 1: true value: " + local_var(7));
-            debug_msg("ECPARIA: dude_luck: " + get_critter_stat(dude_obj, 6));
+            debug_msg("ECPARIA: dude_luck: " + dude_luck);
         end
     end
     else begin
@@ -211,21 +211,21 @@ begin
                 if (not(is_success(do_check(dude_obj, 6, 0)))) then begin
                     debug_msg("ECPARIAH: added self to party");
                     tmp_add := 1;
-                    display_msg(message_str(1098, 103));
+                    display_mstr(103);
                     if (not(critter_state(self_obj) bwand 1)) then begin
                         if ((critter_state(self_obj) bwand 1) == 0) then begin
                             if (local_var(10) == 0) then begin
                                 set_local_var(10, 6);
                             end
                             set_local_var(9, 0);
-                            if (has_trait(1, self_obj, 6) != 0) then begin
-                                set_local_var(11, has_trait(1, self_obj, 6));
+                            if (has_trait(TRAIT_OBJECT, self_obj, OBJECT_TEAM_NUM) != 0) then begin
+                                set_local_var(11, has_trait(TRAIT_OBJECT, self_obj, OBJECT_TEAM_NUM));
                             end
-                            critter_add_trait(self_obj, 1, 6, 0);
+                            critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 0);
                             party_add(self_obj);
                         end
-                        debug_msg("join party: " + obj_name(self_obj));
-                        critter_add_trait(dude_obj, 0, 118, 1);
+                        debug_msg("join party: " + self_name);
+                        critter_add_trait(dude_obj, TRAIT_PERK, PERK_jinxed_perk, 1);
                     end
                 end
                 else begin
@@ -261,54 +261,54 @@ end
 procedure destroy_p_proc
 begin
     if (source_obj == dude_obj) then begin
-        set_global_var(4, global_var(4) + 1);
-        if (metarule(51, self_obj) == 2) then begin
+        set_global_var(GVAR_BAD_MONSTER, global_var(GVAR_BAD_MONSTER) + 1);
+        if (critter_kill_type(self_obj) == KILL_TYPE_children_kills) then begin
             if (source_obj == dude_obj) then begin
-                set_global_var(2, 0);
-                set_global_var(1, global_var(1) + 1);
-                set_global_var(0, global_var(0) - -15);
-                set_global_var(37, 0);
-                set_global_var(38, 0);
-                set_global_var(39, 0);
-                set_global_var(40, 0);
-                set_global_var(41, 0);
-                set_global_var(42, 0);
-                set_global_var(43, 0);
-                set_global_var(44, 0);
-                set_global_var(45, 0);
-                if (global_var(0) >= 1000) then begin
-                    set_global_var(37, 1);
+                set_global_var(GVAR_CHAMPION_REPUTATION, 0);
+                set_global_var(GVAR_CHILDKILLER_REPUTATION, global_var(GVAR_CHILDKILLER_REPUTATION) + 1);
+                set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) - -15);
+                set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+                set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+                set_global_var(GVAR_KARMA_DEFENDER, 0);
+                set_global_var(GVAR_KARMA_WANDERER, 0);
+                set_global_var(GVAR_KARMA_BETRAYER, 0);
+                set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+                set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+                set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+                    set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
                 end
                 else begin
-                    if (global_var(0) >= 750) then begin
-                        set_global_var(38, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
                     end
                     else begin
-                        if (global_var(0) >= 500) then begin
-                            set_global_var(39, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                            set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                         end
                         else begin
-                            if (global_var(0) >= 250) then begin
-                                set_global_var(40, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                                set_global_var(GVAR_KARMA_DEFENDER, 1);
                             end
                             else begin
-                                if (global_var(0) > -250) then begin
-                                    set_global_var(41, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                                    set_global_var(GVAR_KARMA_WANDERER, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -500) then begin
-                                        set_global_var(42, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                        set_global_var(GVAR_KARMA_BETRAYER, 1);
                                     end
                                     else begin
-                                        if (global_var(0) > -750) then begin
-                                            set_global_var(43, 1);
+                                        if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                            set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                         end
                                         else begin
-                                            if (global_var(0) > -1000) then begin
-                                                set_global_var(44, 1);
+                                            if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                                set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                             end
                                             else begin
-                                                set_global_var(45, 1);
+                                                set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                             end
                                         end
                                     end
@@ -319,58 +319,58 @@ begin
                 end
             end
         end
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(5) > (2 * global_var(4))) or (global_var(3) == 1))) then begin
-            set_global_var(3, 1);
-            set_global_var(2, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
+            set_global_var(GVAR_BERSERKER_REPUTATION, 1);
+            set_global_var(GVAR_CHAMPION_REPUTATION, 0);
         end
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(4) > (3 * global_var(5))) or (global_var(2) == 1)) and (global_var(1) == 0)) then begin
-            set_global_var(2, 1);
-            set_global_var(3, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(4) > (3 * global_var(GVAR_GOOD_MONSTER))) or (global_var(GVAR_CHAMPION_REPUTATION) == 1)) and (global_var(GVAR_CHILDKILLER_REPUTATION) == 0)) then begin
+            set_global_var(GVAR_CHAMPION_REPUTATION, 1);
+            set_global_var(GVAR_BERSERKER_REPUTATION, 0);
         end
-        set_global_var(0, global_var(0) + 5);
+        set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + 5);
         debug_msg("Player gains " + 5 + " Karma Points.");
-        set_global_var(37, 0);
-        set_global_var(38, 0);
-        set_global_var(39, 0);
-        set_global_var(40, 0);
-        set_global_var(41, 0);
-        set_global_var(42, 0);
-        set_global_var(43, 0);
-        set_global_var(44, 0);
-        set_global_var(45, 0);
-        if (global_var(0) >= 1000) then begin
-            set_global_var(37, 1);
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+        set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+        set_global_var(GVAR_KARMA_DEFENDER, 0);
+        set_global_var(GVAR_KARMA_WANDERER, 0);
+        set_global_var(GVAR_KARMA_BETRAYER, 0);
+        set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+        set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+            set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
         end
         else begin
-            if (global_var(0) >= 750) then begin
-                set_global_var(38, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
             end
             else begin
-                if (global_var(0) >= 500) then begin
-                    set_global_var(39, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                 end
                 else begin
-                    if (global_var(0) >= 250) then begin
-                        set_global_var(40, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                        set_global_var(GVAR_KARMA_DEFENDER, 1);
                     end
                     else begin
-                        if (global_var(0) > -250) then begin
-                            set_global_var(41, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                            set_global_var(GVAR_KARMA_WANDERER, 1);
                         end
                         else begin
-                            if (global_var(0) > -500) then begin
-                                set_global_var(42, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                set_global_var(GVAR_KARMA_BETRAYER, 1);
                             end
                             else begin
-                                if (global_var(0) > -750) then begin
-                                    set_global_var(43, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -1000) then begin
-                                        set_global_var(44, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                     end
                                     else begin
-                                        set_global_var(45, 1);
+                                        set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                     end
                                 end
                             end
@@ -380,13 +380,13 @@ begin
             end
         end
     end
-    critter_add_trait(dude_obj, 0, 118, 0);
+    critter_add_trait(dude_obj, TRAIT_PERK, PERK_jinxed_perk, 0);
     if (local_var(7)) then begin
         debug_msg("ECPARIAH: setting luck back to : " + local_var(7));
         if (local_var(7) > 10) then begin
             set_local_var(7, 10);
         end
-        set_critter_stat(dude_obj, 6, local_var(7) - (1 + has_trait(2, self_obj, 15)));
+        set_critter_stat(dude_obj, STAT_lu, local_var(7) - (1 + has_trait(TRAIT_TRAIT, self_obj, TRAIT_gifted)));
     end
 end
 
@@ -394,17 +394,17 @@ procedure look_at_p_proc
 begin
     script_overrides;
     if (local_var(4) == 0) then begin
-        display_msg(message_str(1098, 100));
+        display_mstr(100);
     end
     else begin
-        display_msg(message_str(1098, 101));
+        display_mstr(101);
     end
 end
 
 procedure description_p_proc
 begin
     script_overrides;
-    display_msg(message_str(1098, 102));
+    display_mstr(102);
 end
 
 procedure use_skill_on_p_proc
@@ -417,10 +417,10 @@ begin
         if (not(anim_busy(self_obj))) then begin
             if ((tile_distance_objs(self_obj, dude_obj) < 8) or obj_can_see_obj(self_obj, dude_obj)) then begin
                 if (anim_busy(self_obj) == 0) then begin
-                    global_temp := rotation_to_tile(tile_num(dude_obj), tile_num(self_obj));
+                    global_temp := rotation_to_tile(dude_tile, tile_num(self_obj));
                     animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(self_obj), global_temp, random(3, 10)), 1);
                     global_temp := (global_temp + 1) % 6;
-                    while ((anim_busy(self_obj) == 0) and (global_temp != rotation_to_tile(tile_num(dude_obj), tile_num(self_obj)))) do begin
+                    while ((anim_busy(self_obj) == 0) and (global_temp != rotation_to_tile(dude_tile, tile_num(self_obj)))) do begin
                         debug_msg("flee loop: rot == " + global_temp);
                         animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(self_obj), global_temp, random(3, 10)), 1);
                         global_temp := (global_temp + 1) % 6;
@@ -434,16 +434,16 @@ end
 procedure map_enter_p_proc
 begin
     Only_Once := 0;
-    if ((cur_map_index == 19) or (cur_map_index == 37) or (cur_map_index == 38)) then begin
-        if ((elevation(self_obj) != elevation(dude_obj)) and (local_var(9) != 0)) then begin
-            move_to(self_obj, tile_num(self_obj), elevation(dude_obj));
+    if ((CUR_MAP_MODINN) or (CUR_MAP_V15ENT) or (CUR_MAP_V15ENTB)) then begin
+        if ((elevation(self_obj) != dude_elevation) and (local_var(9) != 0)) then begin
+            move_to(self_obj, tile_num(self_obj), dude_elevation);
         end
     end
     if (party_member_obj(16777600) != 0) then begin
-        critter_add_trait(self_obj, 1, 6, 0);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 0);
     end
     else begin
-        critter_add_trait(self_obj, 1, 6, 80);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 80);
     end
 end
 
@@ -451,9 +451,9 @@ procedure map_update_p_proc
 begin
     if (combat_is_initialized == 0) then begin
         if (anim_busy(self_obj) == 0) then begin
-            if (party_member_obj(obj_pid(self_obj)) != 0) then begin
-                if (metarule3(109, self_obj, 0, 0) == 1) then begin
-                    if ((100 * get_critter_stat(self_obj, 35) / get_critter_stat(self_obj, 7)) <= 60) then begin
+            if (obj_in_party(self_obj)) then begin
+                if (chem_use_level(self_obj) == 1) then begin
+                    if ((100 * get_critter_stat(self_obj, STAT_current_hp) / get_critter_stat(self_obj, STAT_max_hp)) <= 60) then begin
                         global_temp := 1;
                         if (obj_is_carrying_obj_pid(self_obj, 144)) then begin
                             if ((global_temp == 0) and (combat_is_initialized == 0)) then begin
@@ -465,7 +465,7 @@ begin
                             end
                             PartyHealingItem := obj_carrying_pid_obj(self_obj, 144);
                             use_obj_on_obj(PartyHealingItem, self_obj);
-                            debug_msg(obj_name(self_obj) + " used super stimpak.");
+                            debug_msg(self_name + " used super stimpak.");
                         end
                         else begin
                             if (obj_is_carrying_obj_pid(self_obj, 40)) then begin
@@ -477,7 +477,7 @@ begin
                                     gfade_in(600);
                                 end
                                 use_obj_on_obj(obj_carrying_pid_obj(self_obj, 40), self_obj);
-                                debug_msg(obj_name(self_obj) + " used stimpak.");
+                                debug_msg(self_name + " used stimpak.");
                             end
                             else begin
                                 if (obj_is_carrying_obj_pid(self_obj, 273)) then begin
@@ -489,7 +489,7 @@ begin
                                         gfade_in(600);
                                     end
                                     use_obj_on_obj(obj_carrying_pid_obj(self_obj, 273), self_obj);
-                                    debug_msg(obj_name(self_obj) + " used healing powder.");
+                                    debug_msg(self_name + " used healing powder.");
                                 end
                                 else begin
                                     if (obj_is_carrying_obj_pid(self_obj, 91)) then begin
@@ -501,7 +501,7 @@ begin
                                             gfade_in(600);
                                         end
                                         use_obj_on_obj(obj_carrying_pid_obj(self_obj, 91), self_obj);
-                                        debug_msg(obj_name(self_obj) + " used doctor's bag kit.");
+                                        debug_msg(self_name + " used doctor's bag kit.");
                                     end
                                     else begin
                                         if (obj_is_carrying_obj_pid(self_obj, 47)) then begin
@@ -513,7 +513,7 @@ begin
                                                 gfade_in(600);
                                             end
                                             use_obj_on_obj(obj_carrying_pid_obj(self_obj, 47), self_obj);
-                                            debug_msg(obj_name(self_obj) + " used First Aid kit.");
+                                            debug_msg(self_name + " used First Aid kit.");
                                         end
                                     end
                                 end
@@ -522,8 +522,8 @@ begin
                     end
                 end
                 else begin
-                    if (metarule3(109, self_obj, 0, 0) == 2) then begin
-                        if ((100 * get_critter_stat(self_obj, 35) / get_critter_stat(self_obj, 7)) <= 30) then begin
+                    if (chem_use_level(self_obj) == 2) then begin
+                        if ((100 * get_critter_stat(self_obj, STAT_current_hp) / get_critter_stat(self_obj, STAT_max_hp)) <= 30) then begin
                             global_temp := 1;
                             if (obj_is_carrying_obj_pid(self_obj, 144)) then begin
                                 if ((global_temp == 0) and (combat_is_initialized == 0)) then begin
@@ -535,7 +535,7 @@ begin
                                 end
                                 PartyHealingItem := obj_carrying_pid_obj(self_obj, 144);
                                 use_obj_on_obj(PartyHealingItem, self_obj);
-                                debug_msg(obj_name(self_obj) + " used super stimpak.");
+                                debug_msg(self_name + " used super stimpak.");
                             end
                             else begin
                                 if (obj_is_carrying_obj_pid(self_obj, 40)) then begin
@@ -547,7 +547,7 @@ begin
                                         gfade_in(600);
                                     end
                                     use_obj_on_obj(obj_carrying_pid_obj(self_obj, 40), self_obj);
-                                    debug_msg(obj_name(self_obj) + " used stimpak.");
+                                    debug_msg(self_name + " used stimpak.");
                                 end
                                 else begin
                                     if (obj_is_carrying_obj_pid(self_obj, 273)) then begin
@@ -559,7 +559,7 @@ begin
                                             gfade_in(600);
                                         end
                                         use_obj_on_obj(obj_carrying_pid_obj(self_obj, 273), self_obj);
-                                        debug_msg(obj_name(self_obj) + " used healing powder.");
+                                        debug_msg(self_name + " used healing powder.");
                                     end
                                     else begin
                                         if (obj_is_carrying_obj_pid(self_obj, 91)) then begin
@@ -571,7 +571,7 @@ begin
                                                 gfade_in(600);
                                             end
                                             use_obj_on_obj(obj_carrying_pid_obj(self_obj, 91), self_obj);
-                                            debug_msg(obj_name(self_obj) + " used doctor's bag kit.");
+                                            debug_msg(self_name + " used doctor's bag kit.");
                                         end
                                         else begin
                                             if (obj_is_carrying_obj_pid(self_obj, 47)) then begin
@@ -583,7 +583,7 @@ begin
                                                     gfade_in(600);
                                                 end
                                                 use_obj_on_obj(obj_carrying_pid_obj(self_obj, 47), self_obj);
-                                                debug_msg(obj_name(self_obj) + " used First Aid kit.");
+                                                debug_msg(self_name + " used First Aid kit.");
                                             end
                                         end
                                     end
@@ -592,8 +592,8 @@ begin
                         end
                     end
                     else begin
-                        if (metarule3(109, self_obj, 0, 0) == 3) then begin
-                            if ((100 * get_critter_stat(self_obj, 35) / get_critter_stat(self_obj, 7)) <= 50) then begin
+                        if (chem_use_level(self_obj) == 3) then begin
+                            if ((100 * get_critter_stat(self_obj, STAT_current_hp) / get_critter_stat(self_obj, STAT_max_hp)) <= 50) then begin
                                 global_temp := 1;
                                 if (obj_is_carrying_obj_pid(self_obj, 144)) then begin
                                     if ((global_temp == 0) and (combat_is_initialized == 0)) then begin
@@ -605,7 +605,7 @@ begin
                                     end
                                     PartyHealingItem := obj_carrying_pid_obj(self_obj, 144);
                                     use_obj_on_obj(PartyHealingItem, self_obj);
-                                    debug_msg(obj_name(self_obj) + " used super stimpak.");
+                                    debug_msg(self_name + " used super stimpak.");
                                 end
                                 else begin
                                     if (obj_is_carrying_obj_pid(self_obj, 40)) then begin
@@ -617,7 +617,7 @@ begin
                                             gfade_in(600);
                                         end
                                         use_obj_on_obj(obj_carrying_pid_obj(self_obj, 40), self_obj);
-                                        debug_msg(obj_name(self_obj) + " used stimpak.");
+                                        debug_msg(self_name + " used stimpak.");
                                     end
                                     else begin
                                         if (obj_is_carrying_obj_pid(self_obj, 273)) then begin
@@ -629,7 +629,7 @@ begin
                                                 gfade_in(600);
                                             end
                                             use_obj_on_obj(obj_carrying_pid_obj(self_obj, 273), self_obj);
-                                            debug_msg(obj_name(self_obj) + " used healing powder.");
+                                            debug_msg(self_name + " used healing powder.");
                                         end
                                         else begin
                                             if (obj_is_carrying_obj_pid(self_obj, 91)) then begin
@@ -641,7 +641,7 @@ begin
                                                     gfade_in(600);
                                                 end
                                                 use_obj_on_obj(obj_carrying_pid_obj(self_obj, 91), self_obj);
-                                                debug_msg(obj_name(self_obj) + " used doctor's bag kit.");
+                                                debug_msg(self_name + " used doctor's bag kit.");
                                             end
                                             else begin
                                                 if (obj_is_carrying_obj_pid(self_obj, 47)) then begin
@@ -653,7 +653,7 @@ begin
                                                         gfade_in(600);
                                                     end
                                                     use_obj_on_obj(obj_carrying_pid_obj(self_obj, 47), self_obj);
-                                                    debug_msg(obj_name(self_obj) + " used First Aid kit.");
+                                                    debug_msg(self_name + " used First Aid kit.");
                                                 end
                                             end
                                         end
@@ -662,8 +662,8 @@ begin
                             end
                         end
                         else begin
-                            if (metarule3(109, self_obj, 0, 0) == 4) then begin
-                                if ((100 * get_critter_stat(self_obj, 35) / get_critter_stat(self_obj, 7)) <= 50) then begin
+                            if (chem_use_level(self_obj) == 4) then begin
+                                if ((100 * get_critter_stat(self_obj, STAT_current_hp) / get_critter_stat(self_obj, STAT_max_hp)) <= 50) then begin
                                     global_temp := 1;
                                     if (obj_is_carrying_obj_pid(self_obj, 144)) then begin
                                         if ((global_temp == 0) and (combat_is_initialized == 0)) then begin
@@ -675,7 +675,7 @@ begin
                                         end
                                         PartyHealingItem := obj_carrying_pid_obj(self_obj, 144);
                                         use_obj_on_obj(PartyHealingItem, self_obj);
-                                        debug_msg(obj_name(self_obj) + " used super stimpak.");
+                                        debug_msg(self_name + " used super stimpak.");
                                     end
                                     else begin
                                         if (obj_is_carrying_obj_pid(self_obj, 40)) then begin
@@ -687,7 +687,7 @@ begin
                                                 gfade_in(600);
                                             end
                                             use_obj_on_obj(obj_carrying_pid_obj(self_obj, 40), self_obj);
-                                            debug_msg(obj_name(self_obj) + " used stimpak.");
+                                            debug_msg(self_name + " used stimpak.");
                                         end
                                         else begin
                                             if (obj_is_carrying_obj_pid(self_obj, 273)) then begin
@@ -699,7 +699,7 @@ begin
                                                     gfade_in(600);
                                                 end
                                                 use_obj_on_obj(obj_carrying_pid_obj(self_obj, 273), self_obj);
-                                                debug_msg(obj_name(self_obj) + " used healing powder.");
+                                                debug_msg(self_name + " used healing powder.");
                                             end
                                             else begin
                                                 if (obj_is_carrying_obj_pid(self_obj, 91)) then begin
@@ -711,7 +711,7 @@ begin
                                                         gfade_in(600);
                                                     end
                                                     use_obj_on_obj(obj_carrying_pid_obj(self_obj, 91), self_obj);
-                                                    debug_msg(obj_name(self_obj) + " used doctor's bag kit.");
+                                                    debug_msg(self_name + " used doctor's bag kit.");
                                                 end
                                                 else begin
                                                     if (obj_is_carrying_obj_pid(self_obj, 47)) then begin
@@ -723,7 +723,7 @@ begin
                                                             gfade_in(600);
                                                         end
                                                         use_obj_on_obj(obj_carrying_pid_obj(self_obj, 47), self_obj);
-                                                        debug_msg(obj_name(self_obj) + " used First Aid kit.");
+                                                        debug_msg(self_name + " used First Aid kit.");
                                                     end
                                                 end
                                             end
@@ -798,4 +798,3 @@ end
 procedure timed_event_p_proc
 begin
 end
-
