diff --git a/Games/Fallout2.Steam/Original/ECPDOG.ssl b/Games/Fallout2.Steam/ReDefine/ECPDOG.ssl
index 50ee570..a54ecc3 100644
--- a/Games/Fallout2.Steam/Original/ECPDOG.ssl
+++ b/Games/Fallout2.Steam/ReDefine/ECPDOG.ssl
@@ -173,23 +173,23 @@ end
 
 procedure destroy_p_proc
 begin
-    display_msg(message_str(717, 200));
+    display_mstr(200);
 end
 
 procedure look_at_p_proc
 begin
     script_overrides;
-    display_msg(message_str(717, 100));
+    display_mstr(100);
 end
 
 procedure description_p_proc
 begin
     script_overrides;
     if (local_var(4) == 0) then begin
-        display_msg(message_str(717, 101));
+        display_mstr(101);
     end
     else begin
-        display_msg(message_str(717, 102));
+        display_mstr(102);
     end
 end
 
@@ -199,12 +199,12 @@ begin
     LVar0 := action_being_used;
     if (LVar0 == 6) then begin
         script_overrides;
-        display_msg(message_str(717, 201));
+        display_mstr(201);
     end
     else begin
         if (LVar0 == 7) then begin
             script_overrides;
-            display_msg(message_str(717, 202));
+            display_mstr(202);
         end
     end
 end
@@ -218,8 +218,8 @@ begin
     Only_Once := 0;
     if (not(0)) then begin
         starttile := tile_num(self_obj);
-        critter_add_trait(self_obj, 1, 6, 158);
-        critter_add_trait(self_obj, 1, 5, 14);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 158);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_PEASANTS);
         add_timer_event(self_obj, game_ticks(random(20, 30)), 1);
     end
 end
@@ -235,7 +235,7 @@ end
 
 procedure Node001
 begin
-    float_msg(self_obj, message_str(717, 110 + random(0, 5)), 0);
+    float_msg(self_obj, mstr(110 + random(0, 5)), FLOAT_MSG_NORMAL);
 end
 
 procedure timed_event_p_proc
@@ -257,19 +257,18 @@ begin
     LVar0 := obj_pid(obj_being_used_with);
     if (LVar0 == 91) then begin
         script_overrides;
-        display_msg(message_str(717, 202));
+        display_mstr(202);
     end
     else begin
         if ((LVar0 == 40) or (LVar0 == 47) or (LVar0 == 144) or (LVar0 == 273)) then begin
             script_overrides;
-            display_msg(message_str(717, 201));
+            display_mstr(201);
         end
         else begin
             if (LVar0 == 48) then begin
-                float_msg(self_obj, message_str(717, 114), 2);
-                critter_dmg(self_obj, random(1, 5), 0);
+                float_msg(self_obj, mstr(114), FLOAT_MSG_RED);
+                critter_damage(self_obj, random(1, 5));
             end
         end
     end
 end
-
