diff --git a/Games/Fallout2.Steam/Original/ECPLANT.ssl b/Games/Fallout2.Steam/ReDefine/ECPLANT.ssl
index d71784b..afc0282 100644
--- a/Games/Fallout2.Steam/Original/ECPLANT.ssl
+++ b/Games/Fallout2.Steam/ReDefine/ECPLANT.ssl
@@ -30,15 +30,15 @@ variable removed_qty;
 
 procedure start
 begin
-    critter_add_trait(self_obj, 1, 6, 121);
-    critter_add_trait(self_obj, 1, 5, 5);
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 121);
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_SPORE_PLANT);
 end
 
 procedure map_enter_p_proc
 begin
     variable LVar0 := 0;
-    if (metarule(14, 0)) then begin
-        LVar0 := create_object_sid(365, 0, 0, -1);
+    if (map_first_run) then begin
+        LVar0 := create_object(365, 0, 0);
         add_mult_objs_to_inven(self_obj, LVar0, 10);
     end
 end
@@ -46,13 +46,13 @@ end
 procedure critter_p_proc
 begin
     if (obj_can_see_obj(self_obj, dude_obj)) then begin
-        attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+        attack(dude_obj);
     end
 end
 
 procedure destroy_p_proc
 begin
-    if (cur_map_index == 94) then begin
+    if (CUR_MAP_BHRNDDST) then begin
         set_map_var(1, map_var(1) - 1);
     end
 end
@@ -161,4 +161,3 @@ begin
     end
     return 0;
 end
-
