diff --git a/Games/Fallout2.Steam/Original/ECRAT.ssl b/Games/Fallout2.Steam/ReDefine/ECRAT.ssl
index f8b70e7..6091353 100644
--- a/Games/Fallout2.Steam/Original/ECRAT.ssl
+++ b/Games/Fallout2.Steam/ReDefine/ECRAT.ssl
@@ -143,7 +143,7 @@ end
 
 procedure destroy_p_proc
 begin
-    if (cur_map_index == 94) then begin
+    if (CUR_MAP_BHRNDDST) then begin
         set_map_var(1, map_var(1) - 1);
     end
 end
@@ -151,7 +151,7 @@ end
 procedure critter_p_proc
 begin
     if (obj_can_see_obj(self_obj, dude_obj)) then begin
-        attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+        attack(dude_obj);
     end
     if (not(anim_busy(self_obj))) then begin
         reg_anim_func(2, self_obj);
@@ -163,7 +163,7 @@ end
 
 procedure map_enter_p_proc
 begin
-    critter_add_trait(self_obj, 1, 6, 124);
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 124);
     starttile := tile_num(self_obj);
     reg_anim_func(2, self_obj);
     reg_anim_func(1, 1);
@@ -174,4 +174,3 @@ end
 procedure combat_p_proc
 begin
 end
-
