diff --git a/Games/Fallout2.Steam/Original/ECVORRAT.ssl b/Games/Fallout2.Steam/ReDefine/ECVORRAT.ssl
index 3117b6b..4f41763 100644
--- a/Games/Fallout2.Steam/Original/ECVORRAT.ssl
+++ b/Games/Fallout2.Steam/ReDefine/ECVORRAT.ssl
@@ -163,7 +163,7 @@ end
 
 procedure map_enter_p_proc
 begin
-    critter_add_trait(self_obj, 1, 6, 124);
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 124);
     starttile := tile_num(self_obj);
     reg_anim_func(2, self_obj);
     reg_anim_func(1, 1);
@@ -181,7 +181,7 @@ begin
         if (obj_pid(target_obj) == 421) then begin
             script_overrides;
             i_vorpal_obj := -1;
-            critter_dmg(self_obj, 99999, 0);
+            critter_damage(self_obj, 99999);
         end
     end
 end
@@ -193,25 +193,25 @@ end
 
 procedure checkAttack
 begin
-    if (elevation(self_obj) == elevation(dude_obj)) then begin
+    if (elevation(self_obj) == dude_elevation) then begin
         if (i_arthur_obj != -1) then begin
-            attack_complex(i_arthur_obj, 0, 1, 0, 0, 30000, 0, 0);
+            attack(i_arthur_obj);
         end
         else begin
             if (i_robin_obj != -1) then begin
-                attack_complex(i_robin_obj, 0, 1, 0, 0, 30000, 0, 0);
+                attack(i_robin_obj);
             end
             else begin
                 if (i_launcelot_obj != -1) then begin
-                    attack_complex(i_launcelot_obj, 0, 1, 0, 0, 30000, 0, 0);
+                    attack(i_launcelot_obj);
                 end
                 else begin
                     if (i_galahad_obj != -1) then begin
-                        attack_complex(i_bedemir_obj, 0, 1, 0, 0, 30000, 0, 0);
+                        attack(i_bedemir_obj);
                     end
                     else begin
                         if (obj_can_see_obj(self_obj, dude_obj)) then begin
-                            attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+                            attack(dude_obj);
                         end
                     end
                 end
@@ -219,4 +219,3 @@ begin
         end
     end
 end
-
