diff --git a/Games/Fallout2.Steam/Original/ENCRETR.ssl b/Games/Fallout2.Steam/ReDefine/ENCRETR.ssl
index dbdc194..a74bc61 100644
--- a/Games/Fallout2.Steam/Original/ENCRETR.ssl
+++ b/Games/Fallout2.Steam/ReDefine/ENCRETR.ssl
@@ -143,42 +143,41 @@ end
 procedure map_enter_p_proc
 begin
     set_light_level(60);
-    set_global_var(27, 0);
+    set_global_var(GVAR_LOAD_MAP_INDEX, 0);
 end
 
 procedure map_update_p_proc
 begin
     set_light_level(60);
-    if (global_var(498) != 0) then begin
-        Enclave_Countdown_Variable := game_time - global_var(498);
+    if (global_var(GVAR_ENCLAVE_COUNTDOWN) != 0) then begin
+        Enclave_Countdown_Variable := game_time - global_var(GVAR_ENCLAVE_COUNTDOWN);
         if (Enclave_Countdown_Variable >= (10 * (60 * 10))) then begin
             play_gmovie(11);
-            metarule(13, 0);
+            signal_end_game;
         end
         else begin
             Enclave_Minute := ((10 * (60 * 10)) - Enclave_Countdown_Variable) / (60 * 10);
             Enclave_Second := ((10 * (60 * 10)) - Enclave_Countdown_Variable) % (60 * 10) / 10;
-            display_msg(message_str(14, 1012));
+            display_msg(g_mstr(1012));
             if (Enclave_Minute == 1) then begin
                 if (Enclave_Second == 1) then begin
-                    display_msg(Enclave_Minute + message_str(146, 114) + Enclave_Second + message_str(146, 115));
+                    display_msg(Enclave_Minute + message_str(SCRIPT_DEPOLV1, 114) + Enclave_Second + message_str(SCRIPT_DEPOLV1, 115));
                 end
                 else begin
-                    display_msg(Enclave_Minute + message_str(146, 114) + Enclave_Second + message_str(146, 113));
+                    display_msg(Enclave_Minute + message_str(SCRIPT_DEPOLV1, 114) + Enclave_Second + message_str(SCRIPT_DEPOLV1, 113));
                 end
             end
             else begin
                 if (Enclave_Second == 1) then begin
-                    display_msg(Enclave_Minute + message_str(146, 112) + Enclave_Second + message_str(146, 115));
+                    display_msg(Enclave_Minute + message_str(SCRIPT_DEPOLV1, 112) + Enclave_Second + message_str(SCRIPT_DEPOLV1, 115));
                 end
                 else begin
-                    display_msg(Enclave_Minute + message_str(146, 112) + Enclave_Second + message_str(146, 113));
+                    display_msg(Enclave_Minute + message_str(SCRIPT_DEPOLV1, 112) + Enclave_Second + message_str(SCRIPT_DEPOLV1, 113));
                 end
             end
         end
     end
-    if (global_var(433) == 2) then begin
-        display_msg(message_str(14, 1013));
+    if (global_var(GVAR_ENCLAVE_ALARM) == 2) then begin
+        display_msg(g_mstr(1013));
     end
 end
-
