diff --git a/Games/Fallout2.Steam/Original/ETENTER.ssl b/Games/Fallout2.Steam/ReDefine/ETENTER.ssl
index d453f8b..a8e7dd4 100644
--- a/Games/Fallout2.Steam/Original/ETENTER.ssl
+++ b/Games/Fallout2.Steam/ReDefine/ETENTER.ssl
@@ -136,11 +136,10 @@ end
 
 procedure spatial_p_proc
 begin
-    if (tile_distance(tile_num(self_obj), tile_num(dude_obj)) < 8) then begin
-        if (((global_var(215) bwand 2) != 0) == 0) then begin
-            display_msg(message_str(884, 100));
+    if (tile_distance(tile_num(self_obj), dude_tile) < 8) then begin
+        if (((global_var(GVAR_MILITARY_BASE_FLAGS) bwand 2) != 0) == 0) then begin
+            display_mstr(100);
         end
         destroy_object(self_obj);
     end
 end
-
