diff --git a/Games/Fallout2.Steam/Original/FCCHEMST.ssl b/Games/Fallout2.Steam/ReDefine/FCCHEMST.ssl
index be9a08d..9e0d2e8 100644
--- a/Games/Fallout2.Steam/Original/FCCHEMST.ssl
+++ b/Games/Fallout2.Steam/ReDefine/FCCHEMST.ssl
@@ -195,10 +195,10 @@ begin
             if (anim_busy(self_obj) == 0) then begin
                 if ((tile_distance_objs(self_obj, dude_obj) < 8) or obj_can_see_obj(self_obj, dude_obj)) then begin
                     if (anim_busy(self_obj) == 0) then begin
-                        global_temp := rotation_to_tile(tile_num(dude_obj), tile_num(self_obj));
+                        global_temp := rotation_to_tile(dude_tile, tile_num(self_obj));
                         animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(self_obj), global_temp, random(3, 10)), 1);
                         global_temp := (global_temp + 1) % 6;
-                        while ((anim_busy(self_obj) == 0) and (global_temp != rotation_to_tile(tile_num(dude_obj), tile_num(self_obj)))) do begin
+                        while ((anim_busy(self_obj) == 0) and (global_temp != rotation_to_tile(dude_tile, tile_num(self_obj)))) do begin
                             debug_msg("flee loop: rot == " + global_temp);
                             animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(self_obj), global_temp, random(3, 10)), 1);
                             global_temp := (global_temp + 1) % 6;
@@ -208,7 +208,7 @@ begin
             end
         end
         else begin
-            attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+            attack(dude_obj);
         end
     end
 end
@@ -225,48 +225,48 @@ begin
     Evil_Critter := 0;
     Slavery_Tolerant := 1;
     Karma_Perception := 20;
-    set_global_var(37, 0);
-    set_global_var(38, 0);
-    set_global_var(39, 0);
-    set_global_var(40, 0);
-    set_global_var(41, 0);
-    set_global_var(42, 0);
-    set_global_var(43, 0);
-    set_global_var(44, 0);
-    set_global_var(45, 0);
-    if (global_var(0) >= 1000) then begin
-        set_global_var(37, 1);
+    set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+    set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+    set_global_var(GVAR_KARMA_DEFENDER, 0);
+    set_global_var(GVAR_KARMA_WANDERER, 0);
+    set_global_var(GVAR_KARMA_BETRAYER, 0);
+    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+    set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+    set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+    if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
     end
     else begin
-        if (global_var(0) >= 750) then begin
-            set_global_var(38, 1);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+            set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
         end
         else begin
-            if (global_var(0) >= 500) then begin
-                set_global_var(39, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
             end
             else begin
-                if (global_var(0) >= 250) then begin
-                    set_global_var(40, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                    set_global_var(GVAR_KARMA_DEFENDER, 1);
                 end
                 else begin
-                    if (global_var(0) > -250) then begin
-                        set_global_var(41, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                        set_global_var(GVAR_KARMA_WANDERER, 1);
                     end
                     else begin
-                        if (global_var(0) > -500) then begin
-                            set_global_var(42, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                            set_global_var(GVAR_KARMA_BETRAYER, 1);
                         end
                         else begin
-                            if (global_var(0) > -750) then begin
-                                set_global_var(43, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                             end
                             else begin
-                                if (global_var(0) > -1000) then begin
-                                    set_global_var(44, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                    set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                 end
                                 else begin
-                                    set_global_var(45, 1);
+                                    set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                 end
                             end
                         end
@@ -277,7 +277,7 @@ begin
     end
     if (local_var(2) == 0) then begin
         set_local_var(2, 1);
-        set_local_var(3, (get_critter_stat(dude_obj, 3) - get_critter_stat(self_obj, 3)) * 5);
+        set_local_var(3, (dude_charisma - self_charisma) * 5);
         debug_msg("Base Reaction == " + local_var(3));
     end
     Static_Reaction := 0;
@@ -289,14 +289,14 @@ begin
     else begin
         reaction_bonus_karma := -1 * (-1 * reaction_bonus_karma / 1000);
     end
-    if (global_var(61) >= 0) then begin
-        reaction_bonus_town_rep := global_var(61) / 2;
+    if (global_var(GVAR_TOWN_REP_SAN_FRANCISCO) >= 0) then begin
+        reaction_bonus_town_rep := global_var(GVAR_TOWN_REP_SAN_FRANCISCO) / 2;
     end
     else begin
-        reaction_bonus_town_rep := -1 * (-1 * global_var(61) / 2);
+        reaction_bonus_town_rep := -1 * (-1 * global_var(GVAR_TOWN_REP_SAN_FRANCISCO) / 2);
     end
-    if (has_trait(2, dude_obj, 13)) then begin
-        if (get_critter_stat(dude_obj, 34) == get_critter_stat(self_obj, 34)) then begin
+    if (has_trait(TRAIT_TRAIT, dude_obj, TRAIT_sex_appeal)) then begin
+        if (dude_gender == self_gender) then begin
             Static_Reaction := Static_Reaction + -20;
         end
         else begin
@@ -312,48 +312,48 @@ begin
         debug_msg("Karma Reaction Bonus == " + Static_Reaction);
         Static_Reaction := Static_Reaction - reaction_bonus_town_rep;
         debug_msg("Town Rep Bonus == " + Static_Reaction);
-        set_global_var(37, 0);
-        set_global_var(38, 0);
-        set_global_var(39, 0);
-        set_global_var(40, 0);
-        set_global_var(41, 0);
-        set_global_var(42, 0);
-        set_global_var(43, 0);
-        set_global_var(44, 0);
-        set_global_var(45, 0);
-        if (global_var(0) >= 1000) then begin
-            set_global_var(37, 1);
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+        set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+        set_global_var(GVAR_KARMA_DEFENDER, 0);
+        set_global_var(GVAR_KARMA_WANDERER, 0);
+        set_global_var(GVAR_KARMA_BETRAYER, 0);
+        set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+        set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+            set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
         end
         else begin
-            if (global_var(0) >= 750) then begin
-                set_global_var(38, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
             end
             else begin
-                if (global_var(0) >= 500) then begin
-                    set_global_var(39, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                 end
                 else begin
-                    if (global_var(0) >= 250) then begin
-                        set_global_var(40, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                        set_global_var(GVAR_KARMA_DEFENDER, 1);
                     end
                     else begin
-                        if (global_var(0) > -250) then begin
-                            set_global_var(41, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                            set_global_var(GVAR_KARMA_WANDERER, 1);
                         end
                         else begin
-                            if (global_var(0) > -500) then begin
-                                set_global_var(42, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                set_global_var(GVAR_KARMA_BETRAYER, 1);
                             end
                             else begin
-                                if (global_var(0) > -750) then begin
-                                    set_global_var(43, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -1000) then begin
-                                        set_global_var(44, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                     end
                                     else begin
-                                        set_global_var(45, 1);
+                                        set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                     end
                                 end
                             end
@@ -362,35 +362,35 @@ begin
                 end
             end
         end
-        if (global_var(37) == 1) then begin
+        if (global_var(GVAR_KARMA_HOLY_WARRIOR) == 1) then begin
             Static_Reaction := Static_Reaction - 20;
         end
         else begin
-            if (global_var(38) == 1) then begin
+            if (global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES) == 1) then begin
                 Static_Reaction := Static_Reaction - 15;
             end
             else begin
-                if (global_var(39) == 1) then begin
+                if (global_var(GVAR_KARMA_SHIELD_OF_HOPE) == 1) then begin
                     Static_Reaction := Static_Reaction - 10;
                 end
                 else begin
-                    if (global_var(40) == 1) then begin
+                    if (global_var(GVAR_KARMA_DEFENDER) == 1) then begin
                         Static_Reaction := Static_Reaction - 5;
                     end
                     else begin
-                        if (global_var(41) == 1) then begin
+                        if (global_var(GVAR_KARMA_WANDERER) == 1) then begin
                             Static_Reaction := Static_Reaction - 0;
                         end
                         else begin
-                            if (global_var(42) == 1) then begin
+                            if (global_var(GVAR_KARMA_BETRAYER) == 1) then begin
                                 Static_Reaction := Static_Reaction - -5;
                             end
                             else begin
-                                if (global_var(43) == 1) then begin
+                                if (global_var(GVAR_KARMA_SWORD_OF_DESPAIR) == 1) then begin
                                     Static_Reaction := Static_Reaction - -10;
                                 end
                                 else begin
-                                    if (global_var(44) == 1) then begin
+                                    if (global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE) == 1) then begin
                                         Static_Reaction := Static_Reaction - -15;
                                     end
                                     else begin
@@ -410,48 +410,48 @@ begin
         debug_msg("Karma Reaction Bonus == " + Static_Reaction);
         Static_Reaction := Static_Reaction + reaction_bonus_town_rep;
         debug_msg("Town Rep Bonus == " + Static_Reaction);
-        set_global_var(37, 0);
-        set_global_var(38, 0);
-        set_global_var(39, 0);
-        set_global_var(40, 0);
-        set_global_var(41, 0);
-        set_global_var(42, 0);
-        set_global_var(43, 0);
-        set_global_var(44, 0);
-        set_global_var(45, 0);
-        if (global_var(0) >= 1000) then begin
-            set_global_var(37, 1);
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+        set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+        set_global_var(GVAR_KARMA_DEFENDER, 0);
+        set_global_var(GVAR_KARMA_WANDERER, 0);
+        set_global_var(GVAR_KARMA_BETRAYER, 0);
+        set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+        set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+            set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
         end
         else begin
-            if (global_var(0) >= 750) then begin
-                set_global_var(38, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
             end
             else begin
-                if (global_var(0) >= 500) then begin
-                    set_global_var(39, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                 end
                 else begin
-                    if (global_var(0) >= 250) then begin
-                        set_global_var(40, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                        set_global_var(GVAR_KARMA_DEFENDER, 1);
                     end
                     else begin
-                        if (global_var(0) > -250) then begin
-                            set_global_var(41, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                            set_global_var(GVAR_KARMA_WANDERER, 1);
                         end
                         else begin
-                            if (global_var(0) > -500) then begin
-                                set_global_var(42, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                set_global_var(GVAR_KARMA_BETRAYER, 1);
                             end
                             else begin
-                                if (global_var(0) > -750) then begin
-                                    set_global_var(43, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -1000) then begin
-                                        set_global_var(44, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                     end
                                     else begin
-                                        set_global_var(45, 1);
+                                        set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                     end
                                 end
                             end
@@ -460,35 +460,35 @@ begin
                 end
             end
         end
-        if (global_var(37) == 1) then begin
+        if (global_var(GVAR_KARMA_HOLY_WARRIOR) == 1) then begin
             Static_Reaction := Static_Reaction + 20;
         end
         else begin
-            if (global_var(38) == 1) then begin
+            if (global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES) == 1) then begin
                 Static_Reaction := Static_Reaction + 15;
             end
             else begin
-                if (global_var(39) == 1) then begin
+                if (global_var(GVAR_KARMA_SHIELD_OF_HOPE) == 1) then begin
                     Static_Reaction := Static_Reaction + 10;
                 end
                 else begin
-                    if (global_var(40) == 1) then begin
+                    if (global_var(GVAR_KARMA_DEFENDER) == 1) then begin
                         Static_Reaction := Static_Reaction + 5;
                     end
                     else begin
-                        if (global_var(41) == 1) then begin
+                        if (global_var(GVAR_KARMA_WANDERER) == 1) then begin
                             Static_Reaction := Static_Reaction + 0;
                         end
                         else begin
-                            if (global_var(42) == 1) then begin
+                            if (global_var(GVAR_KARMA_BETRAYER) == 1) then begin
                                 Static_Reaction := Static_Reaction + -5;
                             end
                             else begin
-                                if (global_var(43) == 1) then begin
+                                if (global_var(GVAR_KARMA_SWORD_OF_DESPAIR) == 1) then begin
                                     Static_Reaction := Static_Reaction + -10;
                                 end
                                 else begin
-                                    if (global_var(44) == 1) then begin
+                                    if (global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE) == 1) then begin
                                         Static_Reaction := Static_Reaction + -15;
                                     end
                                     else begin
@@ -504,38 +504,38 @@ begin
         debug_msg("Good Person Reaction Bonus == " + Static_Reaction);
     end
     if (Slavery_Tolerant == 0) then begin
-        if (global_var(11) == 1) then begin
+        if (global_var(GVAR_REPUTATION_SLAVER) == 1) then begin
             Static_Reaction := Static_Reaction + (2 * -25);
         end
     end
     else begin
         if (Slavery_Tolerant == 1) then begin
-            if (global_var(11) == 1) then begin
+            if (global_var(GVAR_REPUTATION_SLAVER) == 1) then begin
                 Static_Reaction := Static_Reaction + -25;
             end
         end
         else begin
             if (Slavery_Tolerant == 3) then begin
-                if (global_var(11) == 1) then begin
+                if (global_var(GVAR_REPUTATION_SLAVER) == 1) then begin
                     Static_Reaction := Static_Reaction - -25;
                 end
             end
         end
     end
     debug_msg("Slaver + Aligned Reaction == " + Static_Reaction);
-    Static_Reaction := Static_Reaction + (10 * has_trait(0, dude_obj, 10));
+    Static_Reaction := Static_Reaction + (10 * has_trait(TRAIT_PERK, dude_obj, PERK_presence));
     debug_msg("Presence Reaction == " + Static_Reaction);
-    if (global_var(1) >= 2) then begin
+    if (global_var(GVAR_CHILDKILLER_REPUTATION) >= 2) then begin
         Static_Reaction := Static_Reaction + -30;
     end
     debug_msg("Childkiller Reaction == " + Static_Reaction);
     set_local_var(0, local_var(3) + Static_Reaction);
     debug_msg("Initial Reaction == " + local_var(0));
-    if ((global_var(361) bwand 1) != 0) then begin
+    if ((global_var(GVAR_SAN_FRAN_FLAGS) bwand 1) != 0) then begin
         call Node017();
     end
     else begin
-        if (global_var(444) == 2) then begin
+        if (global_var(GVAR_SAN_FRAN_ARMOR) == 2) then begin
             start_gdialog(892, self_obj, 4, -1, -1);
             gsay_start;
             call Node018();
@@ -543,7 +543,7 @@ begin
             end_dialogue;
         end
         else begin
-            if ((global_var(444) == 1) or (global_var(444) == 4) and obj_carrying_pid_obj(dude_obj, 232)) then begin
+            if ((global_var(GVAR_SAN_FRAN_ARMOR) == 1) or (global_var(GVAR_SAN_FRAN_ARMOR) == 4) and obj_carrying_pid_obj(dude_obj, 232)) then begin
                 start_gdialog(892, self_obj, 4, -1, -1);
                 gsay_start;
                 call Node013();
@@ -551,11 +551,11 @@ begin
                 end_dialogue;
             end
             else begin
-                if ((global_var(444) == 5) or (global_var(444) == 6)) then begin
+                if ((global_var(GVAR_SAN_FRAN_ARMOR) == 5) or (global_var(GVAR_SAN_FRAN_ARMOR) == 6)) then begin
                     call Node021();
                 end
                 else begin
-                    if (global_var(444) == 7) then begin
+                    if (global_var(GVAR_SAN_FRAN_ARMOR) == 7) then begin
                         call Node022();
                     end
                     else begin
@@ -584,59 +584,59 @@ end
 procedure destroy_p_proc
 begin
     if (source_obj == dude_obj) then begin
-        set_global_var(5, global_var(5) + 1);
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(4) > (3 * global_var(5))) or (global_var(2) == 1)) and (global_var(1) == 0)) then begin
-            set_global_var(2, 1);
-            set_global_var(3, 0);
+        set_global_var(GVAR_GOOD_MONSTER, global_var(GVAR_GOOD_MONSTER) + 1);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(4) > (3 * global_var(GVAR_GOOD_MONSTER))) or (global_var(GVAR_CHAMPION_REPUTATION) == 1)) and (global_var(GVAR_CHILDKILLER_REPUTATION) == 0)) then begin
+            set_global_var(GVAR_CHAMPION_REPUTATION, 1);
+            set_global_var(GVAR_BERSERKER_REPUTATION, 0);
         end
-        if (metarule(51, self_obj) == 2) then begin
+        if (critter_kill_type(self_obj) == KILL_TYPE_children_kills) then begin
             if (source_obj == dude_obj) then begin
-                set_global_var(2, 0);
-                set_global_var(1, global_var(1) + 1);
-                set_global_var(0, global_var(0) + -15);
+                set_global_var(GVAR_CHAMPION_REPUTATION, 0);
+                set_global_var(GVAR_CHILDKILLER_REPUTATION, global_var(GVAR_CHILDKILLER_REPUTATION) + 1);
+                set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + -15);
                 debug_msg("Player gains " + -15 + " Karma Points.");
-                set_global_var(37, 0);
-                set_global_var(38, 0);
-                set_global_var(39, 0);
-                set_global_var(40, 0);
-                set_global_var(41, 0);
-                set_global_var(42, 0);
-                set_global_var(43, 0);
-                set_global_var(44, 0);
-                set_global_var(45, 0);
-                if (global_var(0) >= 1000) then begin
-                    set_global_var(37, 1);
+                set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+                set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+                set_global_var(GVAR_KARMA_DEFENDER, 0);
+                set_global_var(GVAR_KARMA_WANDERER, 0);
+                set_global_var(GVAR_KARMA_BETRAYER, 0);
+                set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+                set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+                set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+                    set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
                 end
                 else begin
-                    if (global_var(0) >= 750) then begin
-                        set_global_var(38, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
                     end
                     else begin
-                        if (global_var(0) >= 500) then begin
-                            set_global_var(39, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                            set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                         end
                         else begin
-                            if (global_var(0) >= 250) then begin
-                                set_global_var(40, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                                set_global_var(GVAR_KARMA_DEFENDER, 1);
                             end
                             else begin
-                                if (global_var(0) > -250) then begin
-                                    set_global_var(41, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                                    set_global_var(GVAR_KARMA_WANDERER, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -500) then begin
-                                        set_global_var(42, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                        set_global_var(GVAR_KARMA_BETRAYER, 1);
                                     end
                                     else begin
-                                        if (global_var(0) > -750) then begin
-                                            set_global_var(43, 1);
+                                        if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                            set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                         end
                                         else begin
-                                            if (global_var(0) > -1000) then begin
-                                                set_global_var(44, 1);
+                                            if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                                set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                             end
                                             else begin
-                                                set_global_var(45, 1);
+                                                set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                             end
                                         end
                                     end
@@ -645,58 +645,58 @@ begin
                         end
                     end
                 end
-                set_global_var(61, global_var(61) + -8);
+                set_global_var(GVAR_TOWN_REP_SAN_FRANCISCO, global_var(GVAR_TOWN_REP_SAN_FRANCISCO) + -8);
                 debug_msg("Added " + -8 + " to Town Rep");
             end
         end
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(5) > (2 * global_var(4))) or (global_var(3) == 1))) then begin
-            set_global_var(3, 1);
-            set_global_var(2, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
+            set_global_var(GVAR_BERSERKER_REPUTATION, 1);
+            set_global_var(GVAR_CHAMPION_REPUTATION, 0);
         end
-        set_global_var(0, global_var(0) + -10);
+        set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + -10);
         debug_msg("Player gains " + -10 + " Karma Points.");
-        set_global_var(37, 0);
-        set_global_var(38, 0);
-        set_global_var(39, 0);
-        set_global_var(40, 0);
-        set_global_var(41, 0);
-        set_global_var(42, 0);
-        set_global_var(43, 0);
-        set_global_var(44, 0);
-        set_global_var(45, 0);
-        if (global_var(0) >= 1000) then begin
-            set_global_var(37, 1);
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+        set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+        set_global_var(GVAR_KARMA_DEFENDER, 0);
+        set_global_var(GVAR_KARMA_WANDERER, 0);
+        set_global_var(GVAR_KARMA_BETRAYER, 0);
+        set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+        set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+            set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
         end
         else begin
-            if (global_var(0) >= 750) then begin
-                set_global_var(38, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
             end
             else begin
-                if (global_var(0) >= 500) then begin
-                    set_global_var(39, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                 end
                 else begin
-                    if (global_var(0) >= 250) then begin
-                        set_global_var(40, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                        set_global_var(GVAR_KARMA_DEFENDER, 1);
                     end
                     else begin
-                        if (global_var(0) > -250) then begin
-                            set_global_var(41, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                            set_global_var(GVAR_KARMA_WANDERER, 1);
                         end
                         else begin
-                            if (global_var(0) > -500) then begin
-                                set_global_var(42, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                set_global_var(GVAR_KARMA_BETRAYER, 1);
                             end
                             else begin
-                                if (global_var(0) > -750) then begin
-                                    set_global_var(43, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -1000) then begin
-                                        set_global_var(44, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                     end
                                     else begin
-                                        set_global_var(45, 1);
+                                        set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                     end
                                 end
                             end
@@ -705,7 +705,7 @@ begin
                 end
             end
         end
-        set_global_var(61, global_var(61) + -5);
+        set_global_var(GVAR_TOWN_REP_SAN_FRANCISCO, global_var(GVAR_TOWN_REP_SAN_FRANCISCO) + -5);
         debug_msg("Added " + -5 + " to Town Rep");
     end
 end
@@ -713,13 +713,13 @@ end
 procedure look_at_p_proc
 begin
     script_overrides;
-    display_msg(message_str(892, 100));
+    display_mstr(100);
 end
 
 procedure description_p_proc
 begin
     script_overrides;
-    display_msg(message_str(892, 101));
+    display_mstr(101);
 end
 
 procedure use_skill_on_p_proc
@@ -728,17 +728,17 @@ end
 
 procedure damage_p_proc
 begin
-    if (party_member_obj(obj_pid(source_obj)) != 0) then begin
+    if (obj_in_party(source_obj)) then begin
         set_local_var(6, 1);
-        set_global_var(361, global_var(361) bwor 1);
+        set_global_var(GVAR_SAN_FRAN_FLAGS, global_var(GVAR_SAN_FRAN_FLAGS) bwor 1);
     end
 end
 
 procedure map_enter_p_proc
 begin
     Only_Once := 0;
-    critter_add_trait(self_obj, 1, 6, 167);
-    critter_add_trait(self_obj, 1, 5, 153);
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 167);
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_SHI);
 end
 
 procedure Node998
@@ -752,199 +752,199 @@ end
 
 procedure Node001
 begin
-    gsay_reply(892, 103);
-    giq_option(-3, 892, 104, Node002, 50);
-    giq_option(4, 892, 105, Node003a, 51);
-    giq_option(4, 892, message_str(892, 106) + " " + obj_name(dude_obj) + message_str(892, 171), Node004, 50);
-    giq_option(4, 892, 107, Node005a, 51);
-    giq_option(4, 892, 108, Node999, 50);
+    Reply(103);
+    giq_option(-3, NAME, 104, Node002, NEUTRAL_REACTION);
+    giq_option(4, NAME, 105, Node003a, BAD_REACTION);
+    giq_option(4, NAME, mstr(106) + " " + dude_name + mstr(171), Node004, NEUTRAL_REACTION);
+    giq_option(4, NAME, 107, Node005a, BAD_REACTION);
+    giq_option(4, NAME, 108, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node002
 begin
     if (n2 < 2) then begin
-        gsay_reply(892, 109);
-        giq_option(1, 892, 111, Node999, 50);
+        Reply(109);
+        giq_option(1, NAME, 111, Node999, NEUTRAL_REACTION);
     end
     else begin
         callGuards := 1;
-        gsay_reply(892, 110);
-        giq_option(1, 892, 111, Node999, 50);
+        Reply(110);
+        giq_option(1, NAME, 111, Node999, NEUTRAL_REACTION);
     end
     n2 := n2 + 1;
 end
 
 procedure Node003
 begin
-    gsay_reply(892, 112);
-    giq_option(4, 892, 113, NodeGuards, 50);
-    giq_option(4, 892, 114, Node999, 50);
+    Reply(112);
+    giq_option(4, NAME, 113, NodeGuards, NEUTRAL_REACTION);
+    giq_option(4, NAME, 114, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node004
 begin
-    gsay_reply(892, message_str(892, 115) + " " + obj_name(dude_obj) + message_str(892, 172));
-    if (((global_var(363) bwand 1) != 0) and not((global_var(363) bwand 8) != 0)) then begin
-        giq_option(4, 892, 116, Node96a, 50);
+    Reply(mstr(115) + " " + dude_name + mstr(172));
+    if (((global_var(GVAR_SAN_FRAN_TANKER) bwand 1) != 0) and not((global_var(GVAR_SAN_FRAN_TANKER) bwand 8) != 0)) then begin
+        giq_option(4, NAME, 116, Node96a, NEUTRAL_REACTION);
     end
-    giq_option(4, 892, 117, Node007, 50);
-    if (((global_var(361) bwand 65536) != 0) and not((global_var(361) bwand 32768) != 0)) then begin
-        giq_option(4, 892, 118, Node108a, 50);
+    giq_option(4, NAME, 117, Node007, NEUTRAL_REACTION);
+    if (((global_var(GVAR_SAN_FRAN_FLAGS) bwand 65536) != 0) and not((global_var(GVAR_SAN_FRAN_FLAGS) bwand 32768) != 0)) then begin
+        giq_option(4, NAME, 118, Node108a, NEUTRAL_REACTION);
     end
-    giq_option(4, 892, 119, Node999, 50);
+    giq_option(4, NAME, 119, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node005
 begin
-    gsay_reply(892, 120);
-    if (((global_var(363) bwand 1) != 0) and not((global_var(363) bwand 8) != 0)) then begin
-        giq_option(4, 892, 121, Node96a, 50);
+    Reply(120);
+    if (((global_var(GVAR_SAN_FRAN_TANKER) bwand 1) != 0) and not((global_var(GVAR_SAN_FRAN_TANKER) bwand 8) != 0)) then begin
+        giq_option(4, NAME, 121, Node96a, NEUTRAL_REACTION);
     end
-    giq_option(4, 892, 122, Node007, 50);
-    if (((global_var(361) bwand 65536) != 0) and not((global_var(361) bwand 32768) != 0)) then begin
-        giq_option(4, 892, 123, Node108a, 50);
+    giq_option(4, NAME, 122, Node007, NEUTRAL_REACTION);
+    if (((global_var(GVAR_SAN_FRAN_FLAGS) bwand 65536) != 0) and not((global_var(GVAR_SAN_FRAN_FLAGS) bwand 32768) != 0)) then begin
+        giq_option(4, NAME, 123, Node108a, NEUTRAL_REACTION);
     end
-    giq_option(4, 892, 124, Node999, 50);
+    giq_option(4, NAME, 124, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node006
 begin
-    set_global_var(444, 1);
-    gsay_reply(892, 125);
-    giq_option(4, 892, 126, Node011a, 50);
-    giq_option(4, 892, 127, Node012, 50);
+    set_global_var(GVAR_SAN_FRAN_ARMOR, 1);
+    Reply(125);
+    giq_option(4, NAME, 126, Node011a, NEUTRAL_REACTION);
+    giq_option(4, NAME, 127, Node012, NEUTRAL_REACTION);
     if (obj_carrying_pid_obj(dude_obj, 232)) then begin
-        giq_option(4, 892, 128, Node013, 50);
+        giq_option(4, NAME, 128, Node013, NEUTRAL_REACTION);
     end
 end
 
 procedure Node007
 begin
-    gsay_reply(892, 129);
-    if (((global_var(363) bwand 1) != 0) and not((global_var(363) bwand 8) != 0)) then begin
-        giq_option(4, 892, 130, Node96a, 50);
+    Reply(129);
+    if (((global_var(GVAR_SAN_FRAN_TANKER) bwand 1) != 0) and not((global_var(GVAR_SAN_FRAN_TANKER) bwand 8) != 0)) then begin
+        giq_option(4, NAME, 130, Node96a, NEUTRAL_REACTION);
     end
-    if (((global_var(361) bwand 65536) != 0) and not((global_var(361) bwand 32768) != 0)) then begin
-        giq_option(4, 892, 131, Node108a, 50);
+    if (((global_var(GVAR_SAN_FRAN_FLAGS) bwand 65536) != 0) and not((global_var(GVAR_SAN_FRAN_FLAGS) bwand 32768) != 0)) then begin
+        giq_option(4, NAME, 131, Node108a, NEUTRAL_REACTION);
     end
-    giq_option(4, 892, 132, Node999, 50);
+    giq_option(4, NAME, 132, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node008
 begin
-    set_global_var(444, 1);
-    gsay_reply(892, 133);
-    giq_option(4, 892, 134, Node011a, 50);
-    giq_option(4, 892, 135, Node012, 50);
+    set_global_var(GVAR_SAN_FRAN_ARMOR, 1);
+    Reply(133);
+    giq_option(4, NAME, 134, Node011a, NEUTRAL_REACTION);
+    giq_option(4, NAME, 135, Node012, NEUTRAL_REACTION);
     if (obj_carrying_pid_obj(dude_obj, 232)) then begin
-        giq_option(4, 892, 136, Node013, 50);
+        giq_option(4, NAME, 136, Node013, NEUTRAL_REACTION);
     end
 end
 
 procedure Node009
 begin
-    gsay_reply(892, 137);
-    giq_option(4, 892, 138, Node999, 50);
+    Reply(137);
+    giq_option(4, NAME, 138, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node010
 begin
-    gsay_reply(892, 139);
-    giq_option(4, 892, 140, Node999, 50);
+    Reply(139);
+    giq_option(4, NAME, 140, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node011
 begin
-    gsay_reply(892, 141);
-    giq_option(4, 892, 142, Node999, 50);
+    Reply(141);
+    giq_option(4, NAME, 142, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node012
 begin
-    gsay_reply(892, 143);
-    giq_option(4, 892, 144, Node999, 50);
+    Reply(143);
+    giq_option(4, NAME, 144, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node013
 begin
-    gsay_reply(892, 145);
-    if (((global_var(361) bwand 65536) != 0) and not((global_var(361) bwand 32768) != 0)) then begin
-        giq_option(4, 892, 146, Node1014, 50);
+    Reply(145);
+    if (((global_var(GVAR_SAN_FRAN_FLAGS) bwand 65536) != 0) and not((global_var(GVAR_SAN_FRAN_FLAGS) bwand 32768) != 0)) then begin
+        giq_option(4, NAME, 146, Node1014, NEUTRAL_REACTION);
     end
-    if (((global_var(363) bwand 1) != 0) and not((global_var(363) bwand 8) != 0)) then begin
-        giq_option(4, 892, 147, Node0914, 50);
+    if (((global_var(GVAR_SAN_FRAN_TANKER) bwand 1) != 0) and not((global_var(GVAR_SAN_FRAN_TANKER) bwand 8) != 0)) then begin
+        giq_option(4, NAME, 147, Node0914, NEUTRAL_REACTION);
     end
-    giq_option(4, 892, message_str(892, 148) + 20000 + message_str(892, 173), Node015a, 50);
-    giq_option(4, 892, 149, Node016, 50);
+    giq_option(4, NAME, mstr(148) + 20000 + mstr(173), Node015a, NEUTRAL_REACTION);
+    giq_option(4, NAME, 149, Node016, NEUTRAL_REACTION);
 end
 
 procedure Node014
 begin
-    gsay_reply(892, 150);
-    giq_option(4, 892, 151, Node999, 50);
+    Reply(150);
+    giq_option(4, NAME, 151, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node015
 begin
-    gsay_reply(892, 152);
-    giq_option(4, 892, 153, Node999, 50);
+    Reply(152);
+    giq_option(4, NAME, 153, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node016
 begin
-    gsay_reply(892, 154);
-    giq_option(4, 892, 155, Node999, 50);
-    giq_option(4, 892, 156, Node999a, 51);
+    Reply(154);
+    giq_option(4, NAME, 155, Node999, NEUTRAL_REACTION);
+    giq_option(4, NAME, 156, Node999a, BAD_REACTION);
 end
 
 procedure Node017
 begin
-    float_msg(self_obj, message_str(892, 157), 2);
+    float_msg(self_obj, mstr(157), FLOAT_MSG_RED);
     call Node998();
 end
 
 procedure Node018
 begin
-    gsay_reply(892, 158);
-    giq_option(4, 892, 159, Node019, 50);
+    Reply(158);
+    giq_option(4, NAME, 159, Node019, NEUTRAL_REACTION);
     if (obj_carrying_pid_obj(dude_obj, 232)) then begin
-        giq_option(4, 892, 128, Node013, 50);
+        giq_option(4, NAME, 128, Node013, NEUTRAL_REACTION);
     end
 end
 
 procedure Node019
 begin
-    gsay_reply(892, 160);
-    giq_option(4, 892, 161, Node020, 50);
-    giq_option(4, 892, 162, Node999, 50);
+    Reply(160);
+    giq_option(4, NAME, 161, Node020, NEUTRAL_REACTION);
+    giq_option(4, NAME, 162, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node020
 begin
     if (local_var(1) <= -2) then begin
-        gsay_reply(892, 163);
+        Reply(163);
     end
     else begin
-        gsay_reply(892, 164);
+        Reply(164);
     end
-    giq_option(4, 892, 165, Node007, 50);
-    if ((global_var(444) == 0) and (((global_var(363) bwand 1) != 0) and not((global_var(363) bwand 8) != 0))) then begin
-        giq_option(4, 892, 166, Node96a, 50);
+    giq_option(4, NAME, 165, Node007, NEUTRAL_REACTION);
+    if ((global_var(GVAR_SAN_FRAN_ARMOR) == 0) and (((global_var(GVAR_SAN_FRAN_TANKER) bwand 1) != 0) and not((global_var(GVAR_SAN_FRAN_TANKER) bwand 8) != 0))) then begin
+        giq_option(4, NAME, 166, Node96a, NEUTRAL_REACTION);
     end
-    if ((global_var(444) == 0) and (((global_var(361) bwand 65536) != 0) and not((global_var(361) bwand 32768) != 0))) then begin
-        giq_option(4, 892, 167, Node108a, 50);
+    if ((global_var(GVAR_SAN_FRAN_ARMOR) == 0) and (((global_var(GVAR_SAN_FRAN_FLAGS) bwand 65536) != 0) and not((global_var(GVAR_SAN_FRAN_FLAGS) bwand 32768) != 0))) then begin
+        giq_option(4, NAME, 167, Node108a, NEUTRAL_REACTION);
     end
-    giq_option(4, 892, 168, Node999, 50);
+    giq_option(4, NAME, 168, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node021
 begin
-    float_msg(self_obj, message_str(892, 169), 0);
+    float_msg(self_obj, mstr(169), FLOAT_MSG_NORMAL);
 end
 
 procedure Node022
 begin
-    float_msg(self_obj, message_str(892, 174), 0);
+    float_msg(self_obj, mstr(174), FLOAT_MSG_NORMAL);
 end
 
 procedure Node003a
@@ -1051,7 +1051,7 @@ end
 
 procedure Node96a
 begin
-    if ((global_var(361) bwand 32768) != 0) then begin
+    if ((global_var(GVAR_SAN_FRAN_FLAGS) bwand 32768) != 0) then begin
         call Node009();
     end
     else begin
@@ -1061,7 +1061,7 @@ end
 
 procedure Node108a
 begin
-    if ((global_var(363) bwand 8) != 0) then begin
+    if ((global_var(GVAR_SAN_FRAN_TANKER) bwand 8) != 0) then begin
         call Node010();
     end
     else begin
@@ -1071,64 +1071,64 @@ end
 
 procedure Node011a
 begin
-    if (global_var(539) == 0) then begin
-        set_global_var(539, 1);
+    if (global_var(GVAR_SAN_FRAN_ARMOR_QST) == 0) then begin
+        set_global_var(GVAR_SAN_FRAN_ARMOR_QST, 1);
     end
-    set_global_var(444, 2);
+    set_global_var(GVAR_SAN_FRAN_ARMOR, 2);
     call Node011();
 end
 
 procedure Node1014
 begin
     variable LVar0 := 0;
-    if ((global_var(363) bwand 8) != 0) then begin
+    if ((global_var(GVAR_SAN_FRAN_TANKER) bwand 8) != 0) then begin
         call Node010();
     end
     else begin
-        set_global_var(0, global_var(0) + -25);
+        set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + -25);
         debug_msg("Player gains " + -25 + " Karma Points.");
-        set_global_var(37, 0);
-        set_global_var(38, 0);
-        set_global_var(39, 0);
-        set_global_var(40, 0);
-        set_global_var(41, 0);
-        set_global_var(42, 0);
-        set_global_var(43, 0);
-        set_global_var(44, 0);
-        set_global_var(45, 0);
-        if (global_var(0) >= 1000) then begin
-            set_global_var(37, 1);
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+        set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+        set_global_var(GVAR_KARMA_DEFENDER, 0);
+        set_global_var(GVAR_KARMA_WANDERER, 0);
+        set_global_var(GVAR_KARMA_BETRAYER, 0);
+        set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+        set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+            set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
         end
         else begin
-            if (global_var(0) >= 750) then begin
-                set_global_var(38, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
             end
             else begin
-                if (global_var(0) >= 500) then begin
-                    set_global_var(39, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                 end
                 else begin
-                    if (global_var(0) >= 250) then begin
-                        set_global_var(40, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                        set_global_var(GVAR_KARMA_DEFENDER, 1);
                     end
                     else begin
-                        if (global_var(0) > -250) then begin
-                            set_global_var(41, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                            set_global_var(GVAR_KARMA_WANDERER, 1);
                         end
                         else begin
-                            if (global_var(0) > -500) then begin
-                                set_global_var(42, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                set_global_var(GVAR_KARMA_BETRAYER, 1);
                             end
                             else begin
-                                if (global_var(0) > -750) then begin
-                                    set_global_var(43, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -1000) then begin
-                                        set_global_var(44, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                     end
                                     else begin
-                                        set_global_var(45, 1);
+                                        set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                     end
                                 end
                             end
@@ -1138,56 +1138,56 @@ begin
             end
         end
         give_exp_points(6000);
-        display_msg(message_str(14, 100) + 6000 + message_str(14, 101));
-        set_global_var(61, global_var(61) + 1);
-        set_global_var(361, global_var(361) bwor 32768);
-        set_global_var(535, 2);
+        display_msg(g_mstr(100) + 6000 + g_mstr(101));
+        set_global_var(GVAR_TOWN_REP_SAN_FRANCISCO, global_var(GVAR_TOWN_REP_SAN_FRANCISCO) + 1);
+        set_global_var(GVAR_SAN_FRAN_FLAGS, global_var(GVAR_SAN_FRAN_FLAGS) bwor 32768);
+        set_global_var(GVAR_SAN_FRAN_ELRON_GAS_QST, 2);
         if (3 != 0) then begin
-            set_global_var(0, global_var(0) + 3);
+            set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + 3);
         end
         debug_msg("Player gains " + 3 + " Karma Points.");
-        set_global_var(37, 0);
-        set_global_var(38, 0);
-        set_global_var(39, 0);
-        set_global_var(40, 0);
-        set_global_var(41, 0);
-        set_global_var(42, 0);
-        set_global_var(43, 0);
-        set_global_var(44, 0);
-        set_global_var(45, 0);
-        if (global_var(0) >= 1000) then begin
-            set_global_var(37, 1);
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+        set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+        set_global_var(GVAR_KARMA_DEFENDER, 0);
+        set_global_var(GVAR_KARMA_WANDERER, 0);
+        set_global_var(GVAR_KARMA_BETRAYER, 0);
+        set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+        set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+            set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
         end
         else begin
-            if (global_var(0) >= 750) then begin
-                set_global_var(38, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
             end
             else begin
-                if (global_var(0) >= 500) then begin
-                    set_global_var(39, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                 end
                 else begin
-                    if (global_var(0) >= 250) then begin
-                        set_global_var(40, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                        set_global_var(GVAR_KARMA_DEFENDER, 1);
                     end
                     else begin
-                        if (global_var(0) > -250) then begin
-                            set_global_var(41, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                            set_global_var(GVAR_KARMA_WANDERER, 1);
                         end
                         else begin
-                            if (global_var(0) > -500) then begin
-                                set_global_var(42, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                set_global_var(GVAR_KARMA_BETRAYER, 1);
                             end
                             else begin
-                                if (global_var(0) > -750) then begin
-                                    set_global_var(43, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -1000) then begin
-                                        set_global_var(44, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                     end
                                     else begin
-                                        set_global_var(45, 1);
+                                        set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                     end
                                 end
                             end
@@ -1197,14 +1197,14 @@ begin
             end
         end
         if (5 != 0) then begin
-            set_global_var(61, global_var(61) + 1);
+            set_global_var(GVAR_TOWN_REP_SAN_FRANCISCO, global_var(GVAR_TOWN_REP_SAN_FRANCISCO) + 1);
         end
         if (5000 != 0) then begin
             give_exp_points(5000);
         end
-        display_msg(message_str(14, 100) + 5000 + message_str(14, 101));
-        set_global_var(539, 2);
-        set_global_var(444, 5);
+        display_msg(g_mstr(100) + 5000 + g_mstr(101));
+        set_global_var(GVAR_SAN_FRAN_ARMOR_QST, 2);
+        set_global_var(GVAR_SAN_FRAN_ARMOR, 5);
         LVar0 := obj_carrying_pid_obj(dude_obj, 232);
         rm_obj_from_inven(dude_obj, LVar0);
         add_obj_to_inven(self_obj, LVar0);
@@ -1215,63 +1215,63 @@ end
 procedure Node0914
 begin
     variable LVar0 := 0;
-    if ((global_var(361) bwand 32768) != 0) then begin
+    if ((global_var(GVAR_SAN_FRAN_FLAGS) bwand 32768) != 0) then begin
         call Node009();
     end
     else begin
-        set_global_var(363, global_var(363) bwor 8);
-        if (((global_var(363) bwand 8) != 0) and ((global_var(363) bwand 4) != 0)) then begin
-            if (global_var(494) < 7) then begin
-                set_global_var(494, 7);
+        set_global_var(GVAR_SAN_FRAN_TANKER, global_var(GVAR_SAN_FRAN_TANKER) bwor 8);
+        if (((global_var(GVAR_SAN_FRAN_TANKER) bwand 8) != 0) and ((global_var(GVAR_SAN_FRAN_TANKER) bwand 4) != 0)) then begin
+            if (global_var(GVAR_FALLOUT_2) < 7) then begin
+                set_global_var(GVAR_FALLOUT_2, 7);
             end
         end
         if (892 != 950) then begin
             if (5 != 0) then begin
-                set_global_var(0, global_var(0) + 5);
+                set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + 5);
             end
             debug_msg("Player gains " + 5 + " Karma Points.");
-            set_global_var(37, 0);
-            set_global_var(38, 0);
-            set_global_var(39, 0);
-            set_global_var(40, 0);
-            set_global_var(41, 0);
-            set_global_var(42, 0);
-            set_global_var(43, 0);
-            set_global_var(44, 0);
-            set_global_var(45, 0);
-            if (global_var(0) >= 1000) then begin
-                set_global_var(37, 1);
+            set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+            set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+            set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+            set_global_var(GVAR_KARMA_DEFENDER, 0);
+            set_global_var(GVAR_KARMA_WANDERER, 0);
+            set_global_var(GVAR_KARMA_BETRAYER, 0);
+            set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+            set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+            set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+                set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
             end
             else begin
-                if (global_var(0) >= 750) then begin
-                    set_global_var(38, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                    set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
                 end
                 else begin
-                    if (global_var(0) >= 500) then begin
-                        set_global_var(39, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                        set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                     end
                     else begin
-                        if (global_var(0) >= 250) then begin
-                            set_global_var(40, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                            set_global_var(GVAR_KARMA_DEFENDER, 1);
                         end
                         else begin
-                            if (global_var(0) > -250) then begin
-                                set_global_var(41, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                                set_global_var(GVAR_KARMA_WANDERER, 1);
                             end
                             else begin
-                                if (global_var(0) > -500) then begin
-                                    set_global_var(42, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                    set_global_var(GVAR_KARMA_BETRAYER, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -750) then begin
-                                        set_global_var(43, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                        set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                     end
                                     else begin
-                                        if (global_var(0) > -1000) then begin
-                                            set_global_var(44, 1);
+                                        if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                            set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                         end
                                         else begin
-                                            set_global_var(45, 1);
+                                            set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                         end
                                     end
                                 end
@@ -1281,60 +1281,60 @@ begin
                 end
             end
             if (0 != 0) then begin
-                set_global_var(61, global_var(61) + 1);
+                set_global_var(GVAR_TOWN_REP_SAN_FRANCISCO, global_var(GVAR_TOWN_REP_SAN_FRANCISCO) + 1);
             end
             if (5000 != 0) then begin
                 give_exp_points(5000);
             end
-            display_msg(message_str(14, 100) + 5000 + message_str(14, 101));
+            display_msg(g_mstr(100) + 5000 + g_mstr(101));
         end
-        set_global_var(532, 2);
+        set_global_var(GVAR_SAN_FRAN_FUEL_TANK_QST, 2);
         if (3 != 0) then begin
-            set_global_var(0, global_var(0) + 3);
+            set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + 3);
         end
         debug_msg("Player gains " + 3 + " Karma Points.");
-        set_global_var(37, 0);
-        set_global_var(38, 0);
-        set_global_var(39, 0);
-        set_global_var(40, 0);
-        set_global_var(41, 0);
-        set_global_var(42, 0);
-        set_global_var(43, 0);
-        set_global_var(44, 0);
-        set_global_var(45, 0);
-        if (global_var(0) >= 1000) then begin
-            set_global_var(37, 1);
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+        set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+        set_global_var(GVAR_KARMA_DEFENDER, 0);
+        set_global_var(GVAR_KARMA_WANDERER, 0);
+        set_global_var(GVAR_KARMA_BETRAYER, 0);
+        set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+        set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+            set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
         end
         else begin
-            if (global_var(0) >= 750) then begin
-                set_global_var(38, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
             end
             else begin
-                if (global_var(0) >= 500) then begin
-                    set_global_var(39, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                 end
                 else begin
-                    if (global_var(0) >= 250) then begin
-                        set_global_var(40, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                        set_global_var(GVAR_KARMA_DEFENDER, 1);
                     end
                     else begin
-                        if (global_var(0) > -250) then begin
-                            set_global_var(41, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                            set_global_var(GVAR_KARMA_WANDERER, 1);
                         end
                         else begin
-                            if (global_var(0) > -500) then begin
-                                set_global_var(42, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                set_global_var(GVAR_KARMA_BETRAYER, 1);
                             end
                             else begin
-                                if (global_var(0) > -750) then begin
-                                    set_global_var(43, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -1000) then begin
-                                        set_global_var(44, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                     end
                                     else begin
-                                        set_global_var(45, 1);
+                                        set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                     end
                                 end
                             end
@@ -1344,14 +1344,14 @@ begin
             end
         end
         if (5 != 0) then begin
-            set_global_var(61, global_var(61) + 1);
+            set_global_var(GVAR_TOWN_REP_SAN_FRANCISCO, global_var(GVAR_TOWN_REP_SAN_FRANCISCO) + 1);
         end
         if (5000 != 0) then begin
             give_exp_points(5000);
         end
-        display_msg(message_str(14, 100) + 5000 + message_str(14, 101));
-        set_global_var(539, 2);
-        set_global_var(444, 6);
+        display_msg(g_mstr(100) + 5000 + g_mstr(101));
+        set_global_var(GVAR_SAN_FRAN_ARMOR_QST, 2);
+        set_global_var(GVAR_SAN_FRAN_ARMOR, 6);
         LVar0 := obj_carrying_pid_obj(dude_obj, 232);
         rm_obj_from_inven(dude_obj, LVar0);
         destroy_object(LVar0);
@@ -1363,51 +1363,51 @@ procedure Node015a
 begin
     variable LVar0 := 0;
     if (3 != 0) then begin
-        set_global_var(0, global_var(0) + 3);
+        set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + 3);
     end
     debug_msg("Player gains " + 3 + " Karma Points.");
-    set_global_var(37, 0);
-    set_global_var(38, 0);
-    set_global_var(39, 0);
-    set_global_var(40, 0);
-    set_global_var(41, 0);
-    set_global_var(42, 0);
-    set_global_var(43, 0);
-    set_global_var(44, 0);
-    set_global_var(45, 0);
-    if (global_var(0) >= 1000) then begin
-        set_global_var(37, 1);
+    set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+    set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+    set_global_var(GVAR_KARMA_DEFENDER, 0);
+    set_global_var(GVAR_KARMA_WANDERER, 0);
+    set_global_var(GVAR_KARMA_BETRAYER, 0);
+    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+    set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+    set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+    if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
     end
     else begin
-        if (global_var(0) >= 750) then begin
-            set_global_var(38, 1);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+            set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
         end
         else begin
-            if (global_var(0) >= 500) then begin
-                set_global_var(39, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
             end
             else begin
-                if (global_var(0) >= 250) then begin
-                    set_global_var(40, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                    set_global_var(GVAR_KARMA_DEFENDER, 1);
                 end
                 else begin
-                    if (global_var(0) > -250) then begin
-                        set_global_var(41, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                        set_global_var(GVAR_KARMA_WANDERER, 1);
                     end
                     else begin
-                        if (global_var(0) > -500) then begin
-                            set_global_var(42, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                            set_global_var(GVAR_KARMA_BETRAYER, 1);
                         end
                         else begin
-                            if (global_var(0) > -750) then begin
-                                set_global_var(43, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                             end
                             else begin
-                                if (global_var(0) > -1000) then begin
-                                    set_global_var(44, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                    set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                 end
                                 else begin
-                                    set_global_var(45, 1);
+                                    set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                 end
                             end
                         end
@@ -1417,14 +1417,14 @@ begin
         end
     end
     if (5 != 0) then begin
-        set_global_var(61, global_var(61) + 1);
+        set_global_var(GVAR_TOWN_REP_SAN_FRANCISCO, global_var(GVAR_TOWN_REP_SAN_FRANCISCO) + 1);
     end
     if (5000 != 0) then begin
         give_exp_points(5000);
     end
-    display_msg(message_str(14, 100) + 5000 + message_str(14, 101));
-    set_global_var(539, 2);
-    set_global_var(444, 7);
+    display_msg(g_mstr(100) + 5000 + g_mstr(101));
+    set_global_var(GVAR_SAN_FRAN_ARMOR_QST, 2);
+    set_global_var(GVAR_SAN_FRAN_ARMOR, 7);
     LVar0 := obj_carrying_pid_obj(dude_obj, 232);
     rm_obj_from_inven(dude_obj, LVar0);
     add_obj_to_inven(self_obj, LVar0);
@@ -1479,4 +1479,3 @@ begin
     end
     call Node999();
 end
-
