diff --git a/Games/Fallout2.Steam/Original/FITGUNTB.ssl b/Games/Fallout2.Steam/ReDefine/FITGUNTB.ssl
index 735647d..1f5e793 100644
--- a/Games/Fallout2.Steam/Original/FITGUNTB.ssl
+++ b/Games/Fallout2.Steam/ReDefine/FITGUNTB.ssl
@@ -192,7 +192,7 @@ begin
                 if (obj_is_carrying_obj_pid(self_obj, 41) < restock_amt) then begin
                     if (80 >= random(1, 100)) then begin
                         if (obj_is_carrying_obj_pid(self_obj, 41) < restock_amt) then begin
-                            restock_obj := create_object_sid(41, 0, 0, -1);
+                            restock_obj := create_object(41, 0, 0);
                             add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 41));
                         end
                         else begin
@@ -215,7 +215,7 @@ begin
                                         end
                                         else begin
                                             if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                                metarule(43, self_obj);
+                                                inven_unwield(self_obj);
                                             end
                                         end
                                     end
@@ -233,7 +233,7 @@ begin
                 end
                 else begin
                     if (obj_is_carrying_obj_pid(self_obj, 41) < restock_amt) then begin
-                        restock_obj := create_object_sid(41, 0, 0, -1);
+                        restock_obj := create_object(41, 0, 0);
                         add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 41));
                     end
                     else begin
@@ -256,7 +256,7 @@ begin
                                     end
                                     else begin
                                         if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                            metarule(43, self_obj);
+                                            inven_unwield(self_obj);
                                         end
                                     end
                                 end
@@ -275,7 +275,7 @@ begin
                 if (obj_is_carrying_obj_pid(self_obj, 234) < restock_amt) then begin
                     if (80 >= random(1, 100)) then begin
                         if (obj_is_carrying_obj_pid(self_obj, 234) < restock_amt) then begin
-                            restock_obj := create_object_sid(234, 0, 0, -1);
+                            restock_obj := create_object(234, 0, 0);
                             add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 234));
                         end
                         else begin
@@ -298,7 +298,7 @@ begin
                                         end
                                         else begin
                                             if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                                metarule(43, self_obj);
+                                                inven_unwield(self_obj);
                                             end
                                         end
                                     end
@@ -316,7 +316,7 @@ begin
                 end
                 else begin
                     if (obj_is_carrying_obj_pid(self_obj, 234) < restock_amt) then begin
-                        restock_obj := create_object_sid(234, 0, 0, -1);
+                        restock_obj := create_object(234, 0, 0);
                         add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 234));
                     end
                     else begin
@@ -339,7 +339,7 @@ begin
                                     end
                                     else begin
                                         if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                            metarule(43, self_obj);
+                                            inven_unwield(self_obj);
                                         end
                                     end
                                 end
@@ -358,7 +358,7 @@ begin
                 if (obj_is_carrying_obj_pid(self_obj, 407) < restock_amt) then begin
                     if (100 >= random(1, 100)) then begin
                         if (obj_is_carrying_obj_pid(self_obj, 407) < restock_amt) then begin
-                            restock_obj := create_object_sid(407, 0, 0, -1);
+                            restock_obj := create_object(407, 0, 0);
                             add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 407));
                         end
                         else begin
@@ -381,7 +381,7 @@ begin
                                         end
                                         else begin
                                             if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                                metarule(43, self_obj);
+                                                inven_unwield(self_obj);
                                             end
                                         end
                                     end
@@ -399,7 +399,7 @@ begin
                 end
                 else begin
                     if (obj_is_carrying_obj_pid(self_obj, 407) < restock_amt) then begin
-                        restock_obj := create_object_sid(407, 0, 0, -1);
+                        restock_obj := create_object(407, 0, 0);
                         add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 407));
                     end
                     else begin
@@ -422,7 +422,7 @@ begin
                                     end
                                     else begin
                                         if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                            metarule(43, self_obj);
+                                            inven_unwield(self_obj);
                                         end
                                     end
                                 end
@@ -441,7 +441,7 @@ begin
                 if (obj_is_carrying_obj_pid(self_obj, 20) < restock_amt) then begin
                     if (100 >= random(1, 100)) then begin
                         if (obj_is_carrying_obj_pid(self_obj, 20) < restock_amt) then begin
-                            restock_obj := create_object_sid(20, 0, 0, -1);
+                            restock_obj := create_object(20, 0, 0);
                             add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 20));
                         end
                         else begin
@@ -464,7 +464,7 @@ begin
                                         end
                                         else begin
                                             if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                                metarule(43, self_obj);
+                                                inven_unwield(self_obj);
                                             end
                                         end
                                     end
@@ -482,7 +482,7 @@ begin
                 end
                 else begin
                     if (obj_is_carrying_obj_pid(self_obj, 20) < restock_amt) then begin
-                        restock_obj := create_object_sid(20, 0, 0, -1);
+                        restock_obj := create_object(20, 0, 0);
                         add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 20));
                     end
                     else begin
@@ -505,7 +505,7 @@ begin
                                     end
                                     else begin
                                         if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                            metarule(43, self_obj);
+                                            inven_unwield(self_obj);
                                         end
                                     end
                                 end
@@ -524,7 +524,7 @@ begin
                 if (obj_is_carrying_obj_pid(self_obj, 383) < restock_amt) then begin
                     if (80 >= random(1, 100)) then begin
                         if (obj_is_carrying_obj_pid(self_obj, 383) < restock_amt) then begin
-                            restock_obj := create_object_sid(383, 0, 0, -1);
+                            restock_obj := create_object(383, 0, 0);
                             add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 383));
                         end
                         else begin
@@ -547,7 +547,7 @@ begin
                                         end
                                         else begin
                                             if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                                metarule(43, self_obj);
+                                                inven_unwield(self_obj);
                                             end
                                         end
                                     end
@@ -565,7 +565,7 @@ begin
                 end
                 else begin
                     if (obj_is_carrying_obj_pid(self_obj, 383) < restock_amt) then begin
-                        restock_obj := create_object_sid(383, 0, 0, -1);
+                        restock_obj := create_object(383, 0, 0);
                         add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 383));
                     end
                     else begin
@@ -588,7 +588,7 @@ begin
                                     end
                                     else begin
                                         if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                            metarule(43, self_obj);
+                                            inven_unwield(self_obj);
                                         end
                                     end
                                 end
@@ -607,7 +607,7 @@ begin
                 if (obj_is_carrying_obj_pid(self_obj, 384) < restock_amt) then begin
                     if (50 >= random(1, 100)) then begin
                         if (obj_is_carrying_obj_pid(self_obj, 384) < restock_amt) then begin
-                            restock_obj := create_object_sid(384, 0, 0, -1);
+                            restock_obj := create_object(384, 0, 0);
                             add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 384));
                         end
                         else begin
@@ -630,7 +630,7 @@ begin
                                         end
                                         else begin
                                             if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                                metarule(43, self_obj);
+                                                inven_unwield(self_obj);
                                             end
                                         end
                                     end
@@ -648,7 +648,7 @@ begin
                 end
                 else begin
                     if (obj_is_carrying_obj_pid(self_obj, 384) < restock_amt) then begin
-                        restock_obj := create_object_sid(384, 0, 0, -1);
+                        restock_obj := create_object(384, 0, 0);
                         add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 384));
                     end
                     else begin
@@ -671,7 +671,7 @@ begin
                                     end
                                     else begin
                                         if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                            metarule(43, self_obj);
+                                            inven_unwield(self_obj);
                                         end
                                     end
                                 end
@@ -690,7 +690,7 @@ begin
                 if (obj_is_carrying_obj_pid(self_obj, 40) < restock_amt) then begin
                     if (80 >= random(1, 100)) then begin
                         if (obj_is_carrying_obj_pid(self_obj, 40) < restock_amt) then begin
-                            restock_obj := create_object_sid(40, 0, 0, -1);
+                            restock_obj := create_object(40, 0, 0);
                             add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 40));
                         end
                         else begin
@@ -713,7 +713,7 @@ begin
                                         end
                                         else begin
                                             if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                                metarule(43, self_obj);
+                                                inven_unwield(self_obj);
                                             end
                                         end
                                     end
@@ -731,7 +731,7 @@ begin
                 end
                 else begin
                     if (obj_is_carrying_obj_pid(self_obj, 40) < restock_amt) then begin
-                        restock_obj := create_object_sid(40, 0, 0, -1);
+                        restock_obj := create_object(40, 0, 0);
                         add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 40));
                     end
                     else begin
@@ -754,7 +754,7 @@ begin
                                     end
                                     else begin
                                         if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                            metarule(43, self_obj);
+                                            inven_unwield(self_obj);
                                         end
                                     end
                                 end
@@ -773,7 +773,7 @@ begin
                 if (obj_is_carrying_obj_pid(self_obj, 47) < restock_amt) then begin
                     if (60 >= random(1, 100)) then begin
                         if (obj_is_carrying_obj_pid(self_obj, 47) < restock_amt) then begin
-                            restock_obj := create_object_sid(47, 0, 0, -1);
+                            restock_obj := create_object(47, 0, 0);
                             add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 47));
                         end
                         else begin
@@ -796,7 +796,7 @@ begin
                                         end
                                         else begin
                                             if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                                metarule(43, self_obj);
+                                                inven_unwield(self_obj);
                                             end
                                         end
                                     end
@@ -814,7 +814,7 @@ begin
                 end
                 else begin
                     if (obj_is_carrying_obj_pid(self_obj, 47) < restock_amt) then begin
-                        restock_obj := create_object_sid(47, 0, 0, -1);
+                        restock_obj := create_object(47, 0, 0);
                         add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 47));
                     end
                     else begin
@@ -837,7 +837,7 @@ begin
                                     end
                                     else begin
                                         if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                            metarule(43, self_obj);
+                                            inven_unwield(self_obj);
                                         end
                                     end
                                 end
@@ -856,7 +856,7 @@ begin
                 if (obj_is_carrying_obj_pid(self_obj, 48) < restock_amt) then begin
                     if (60 >= random(1, 100)) then begin
                         if (obj_is_carrying_obj_pid(self_obj, 48) < restock_amt) then begin
-                            restock_obj := create_object_sid(48, 0, 0, -1);
+                            restock_obj := create_object(48, 0, 0);
                             add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 48));
                         end
                         else begin
@@ -879,7 +879,7 @@ begin
                                         end
                                         else begin
                                             if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                                metarule(43, self_obj);
+                                                inven_unwield(self_obj);
                                             end
                                         end
                                     end
@@ -897,7 +897,7 @@ begin
                 end
                 else begin
                     if (obj_is_carrying_obj_pid(self_obj, 48) < restock_amt) then begin
-                        restock_obj := create_object_sid(48, 0, 0, -1);
+                        restock_obj := create_object(48, 0, 0);
                         add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 48));
                     end
                     else begin
@@ -920,7 +920,7 @@ begin
                                     end
                                     else begin
                                         if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                            metarule(43, self_obj);
+                                            inven_unwield(self_obj);
                                         end
                                     end
                                 end
@@ -939,7 +939,7 @@ begin
                 if (obj_is_carrying_obj_pid(self_obj, 53) < restock_amt) then begin
                     if (50 >= random(1, 100)) then begin
                         if (obj_is_carrying_obj_pid(self_obj, 53) < restock_amt) then begin
-                            restock_obj := create_object_sid(53, 0, 0, -1);
+                            restock_obj := create_object(53, 0, 0);
                             add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 53));
                         end
                         else begin
@@ -962,7 +962,7 @@ begin
                                         end
                                         else begin
                                             if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                                metarule(43, self_obj);
+                                                inven_unwield(self_obj);
                                             end
                                         end
                                     end
@@ -980,7 +980,7 @@ begin
                 end
                 else begin
                     if (obj_is_carrying_obj_pid(self_obj, 53) < restock_amt) then begin
-                        restock_obj := create_object_sid(53, 0, 0, -1);
+                        restock_obj := create_object(53, 0, 0);
                         add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 53));
                     end
                     else begin
@@ -1003,7 +1003,7 @@ begin
                                     end
                                     else begin
                                         if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                            metarule(43, self_obj);
+                                            inven_unwield(self_obj);
                                         end
                                     end
                                 end
@@ -1022,7 +1022,7 @@ begin
                 if (obj_is_carrying_obj_pid(self_obj, 71) < restock_amt) then begin
                     if (60 >= random(1, 100)) then begin
                         if (obj_is_carrying_obj_pid(self_obj, 71) < restock_amt) then begin
-                            restock_obj := create_object_sid(71, 0, 0, -1);
+                            restock_obj := create_object(71, 0, 0);
                             add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 71));
                         end
                         else begin
@@ -1045,7 +1045,7 @@ begin
                                         end
                                         else begin
                                             if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                                metarule(43, self_obj);
+                                                inven_unwield(self_obj);
                                             end
                                         end
                                     end
@@ -1063,7 +1063,7 @@ begin
                 end
                 else begin
                     if (obj_is_carrying_obj_pid(self_obj, 71) < restock_amt) then begin
-                        restock_obj := create_object_sid(71, 0, 0, -1);
+                        restock_obj := create_object(71, 0, 0);
                         add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 71));
                     end
                     else begin
@@ -1086,7 +1086,7 @@ begin
                                     end
                                     else begin
                                         if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                            metarule(43, self_obj);
+                                            inven_unwield(self_obj);
                                         end
                                     end
                                 end
@@ -1105,7 +1105,7 @@ begin
                 if (obj_is_carrying_obj_pid(self_obj, 87) < restock_amt) then begin
                     if (50 >= random(1, 100)) then begin
                         if (obj_is_carrying_obj_pid(self_obj, 87) < restock_amt) then begin
-                            restock_obj := create_object_sid(87, 0, 0, -1);
+                            restock_obj := create_object(87, 0, 0);
                             add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 87));
                         end
                         else begin
@@ -1128,7 +1128,7 @@ begin
                                         end
                                         else begin
                                             if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                                metarule(43, self_obj);
+                                                inven_unwield(self_obj);
                                             end
                                         end
                                     end
@@ -1146,7 +1146,7 @@ begin
                 end
                 else begin
                     if (obj_is_carrying_obj_pid(self_obj, 87) < restock_amt) then begin
-                        restock_obj := create_object_sid(87, 0, 0, -1);
+                        restock_obj := create_object(87, 0, 0);
                         add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 87));
                     end
                     else begin
@@ -1169,7 +1169,7 @@ begin
                                     end
                                     else begin
                                         if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                            metarule(43, self_obj);
+                                            inven_unwield(self_obj);
                                         end
                                     end
                                 end
@@ -1188,7 +1188,7 @@ begin
                 if (obj_is_carrying_obj_pid(self_obj, 110) < restock_amt) then begin
                     if (50 >= random(1, 100)) then begin
                         if (obj_is_carrying_obj_pid(self_obj, 110) < restock_amt) then begin
-                            restock_obj := create_object_sid(110, 0, 0, -1);
+                            restock_obj := create_object(110, 0, 0);
                             add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 110));
                         end
                         else begin
@@ -1211,7 +1211,7 @@ begin
                                         end
                                         else begin
                                             if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                                metarule(43, self_obj);
+                                                inven_unwield(self_obj);
                                             end
                                         end
                                     end
@@ -1229,7 +1229,7 @@ begin
                 end
                 else begin
                     if (obj_is_carrying_obj_pid(self_obj, 110) < restock_amt) then begin
-                        restock_obj := create_object_sid(110, 0, 0, -1);
+                        restock_obj := create_object(110, 0, 0);
                         add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 110));
                     end
                     else begin
@@ -1252,7 +1252,7 @@ begin
                                     end
                                     else begin
                                         if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                            metarule(43, self_obj);
+                                            inven_unwield(self_obj);
                                         end
                                     end
                                 end
@@ -1271,7 +1271,7 @@ begin
                 if (obj_is_carrying_obj_pid(self_obj, 144) < restock_amt) then begin
                     if (30 >= random(1, 100)) then begin
                         if (obj_is_carrying_obj_pid(self_obj, 144) < restock_amt) then begin
-                            restock_obj := create_object_sid(144, 0, 0, -1);
+                            restock_obj := create_object(144, 0, 0);
                             add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 144));
                         end
                         else begin
@@ -1294,7 +1294,7 @@ begin
                                         end
                                         else begin
                                             if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                                metarule(43, self_obj);
+                                                inven_unwield(self_obj);
                                             end
                                         end
                                     end
@@ -1312,7 +1312,7 @@ begin
                 end
                 else begin
                     if (obj_is_carrying_obj_pid(self_obj, 144) < restock_amt) then begin
-                        restock_obj := create_object_sid(144, 0, 0, -1);
+                        restock_obj := create_object(144, 0, 0);
                         add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 144));
                     end
                     else begin
@@ -1335,7 +1335,7 @@ begin
                                     end
                                     else begin
                                         if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                            metarule(43, self_obj);
+                                            inven_unwield(self_obj);
                                         end
                                     end
                                 end
@@ -1354,7 +1354,7 @@ begin
                 if (obj_is_carrying_obj_pid(self_obj, 260) < restock_amt) then begin
                     if (30 >= random(1, 100)) then begin
                         if (obj_is_carrying_obj_pid(self_obj, 260) < restock_amt) then begin
-                            restock_obj := create_object_sid(260, 0, 0, -1);
+                            restock_obj := create_object(260, 0, 0);
                             add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 260));
                         end
                         else begin
@@ -1377,7 +1377,7 @@ begin
                                         end
                                         else begin
                                             if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                                metarule(43, self_obj);
+                                                inven_unwield(self_obj);
                                             end
                                         end
                                     end
@@ -1395,7 +1395,7 @@ begin
                 end
                 else begin
                     if (obj_is_carrying_obj_pid(self_obj, 260) < restock_amt) then begin
-                        restock_obj := create_object_sid(260, 0, 0, -1);
+                        restock_obj := create_object(260, 0, 0);
                         add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 260));
                     end
                     else begin
@@ -1418,7 +1418,7 @@ begin
                                     end
                                     else begin
                                         if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                            metarule(43, self_obj);
+                                            inven_unwield(self_obj);
                                         end
                                     end
                                 end
@@ -1437,7 +1437,7 @@ begin
                 if (obj_is_carrying_obj_pid(self_obj, 284) < restock_amt) then begin
                     if (45 >= random(1, 100)) then begin
                         if (obj_is_carrying_obj_pid(self_obj, 284) < restock_amt) then begin
-                            restock_obj := create_object_sid(284, 0, 0, -1);
+                            restock_obj := create_object(284, 0, 0);
                             add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 284));
                         end
                         else begin
@@ -1460,7 +1460,7 @@ begin
                                         end
                                         else begin
                                             if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                                metarule(43, self_obj);
+                                                inven_unwield(self_obj);
                                             end
                                         end
                                     end
@@ -1478,7 +1478,7 @@ begin
                 end
                 else begin
                     if (obj_is_carrying_obj_pid(self_obj, 284) < restock_amt) then begin
-                        restock_obj := create_object_sid(284, 0, 0, -1);
+                        restock_obj := create_object(284, 0, 0);
                         add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 284));
                     end
                     else begin
@@ -1501,7 +1501,7 @@ begin
                                     end
                                     else begin
                                         if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                            metarule(43, self_obj);
+                                            inven_unwield(self_obj);
                                         end
                                     end
                                 end
@@ -1537,4 +1537,3 @@ end
 procedure timed_event_p_proc
 begin
 end
-
