diff --git a/Games/Fallout2.Steam/Original/FLIP.ssl b/Games/Fallout2.Steam/ReDefine/FLIP.ssl
index 2ad4659..b8dc852 100644
--- a/Games/Fallout2.Steam/Original/FLIP.ssl
+++ b/Games/Fallout2.Steam/ReDefine/FLIP.ssl
@@ -43,36 +43,36 @@ variable exit_line;
 procedure start
 begin
     if (not(initialized)) then begin
-        critter_add_trait(self_obj, 1, 6, 43);
-        critter_add_trait(self_obj, 1, 5, 4);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 43);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_ARROYO_SHAMAN);
         initialized := 1;
     end
     else begin
-        if (script_action == 13) then begin
+        if (script_action == combat_proc) then begin
             call combat_p_proc();
         end
         else begin
-            if (script_action == 12) then begin
+            if (script_action == critter_proc) then begin
                 call critter_p_proc();
             end
             else begin
-                if (script_action == 18) then begin
+                if (script_action == destroy_proc) then begin
                     call destroy_p_proc();
                 end
                 else begin
-                    if (script_action == 21) then begin
+                    if (script_action == look_at_proc) then begin
                         call look_at_p_proc();
                     end
                     else begin
-                        if (script_action == 4) then begin
+                        if (script_action == pickup_proc) then begin
                             call pickup_p_proc();
                         end
                         else begin
-                            if (script_action == 11) then begin
+                            if (script_action == talk_proc) then begin
                                 call talk_p_proc();
                             end
                             else begin
-                                if (script_action == 22) then begin
+                                if (script_action == timed_event_proc) then begin
                                     call timed_event_p_proc();
                                 end
                             end
@@ -90,8 +90,8 @@ begin
         round_counter := round_counter + 1;
     end
     if (round_counter > 2) then begin
-        if (global_var(146) == 0) then begin
-            set_global_var(146, 1);
+        if (global_var(GVAR_ENEMY_VAULT_VILLAGE) == 0) then begin
+            set_global_var(GVAR_ENEMY_VAULT_VILLAGE, 1);
             set_map_var(7, 1);
         end
     end
@@ -101,11 +101,11 @@ procedure critter_p_proc
 begin
     if (hostile) then begin
         hostile := 0;
-        attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+        attack(dude_obj);
     end
     else begin
         if (obj_can_see_obj(self_obj, dude_obj) and (tile_distance_objs(self_obj, dude_obj) < 12)) then begin
-            if (global_var(54)) then begin
+            if (global_var(GVAR_TOWN_REP_BROKEN_HILLS)) then begin
                 call Flip00();
             end
             else begin
@@ -127,17 +127,17 @@ procedure destroy_p_proc
 begin
     set_map_var(8, 3);
     if (source_obj == dude_obj) then begin
-        if (((global_var(160) + global_var(159)) >= 25) and ((global_var(159) > (2 * global_var(160))) or (global_var(156) == 1))) then begin
-            set_global_var(156, 1);
-            set_global_var(157, 0);
+        if (((global_var(GVAR_GECKO_SKEETER_PART) + global_var(GVAR_GECKO_REQ_FORM)) >= 25) and ((global_var(GVAR_GECKO_REQ_FORM) > (2 * global_var(GVAR_GECKO_SKEETER_PART))) or (global_var(GVAR_SIERRA_LOCKOUT) == 1))) then begin
+            set_global_var(GVAR_SIERRA_LOCKOUT, 1);
+            set_global_var(GVAR_SIERRA_PASSWORD, 0);
         end
-        if (((global_var(160) + global_var(159)) >= 25) and ((global_var(160) > (3 * global_var(159))) or (global_var(157) == 1))) then begin
-            set_global_var(157, 1);
-            set_global_var(156, 0);
+        if (((global_var(GVAR_GECKO_SKEETER_PART) + global_var(GVAR_GECKO_REQ_FORM)) >= 25) and ((global_var(160) > (3 * global_var(GVAR_GECKO_REQ_FORM))) or (global_var(GVAR_SIERRA_PASSWORD) == 1))) then begin
+            set_global_var(GVAR_SIERRA_PASSWORD, 1);
+            set_global_var(GVAR_SIERRA_LOCKOUT, 0);
         end
-        set_global_var(160, global_var(160) + 1);
-        if ((global_var(160) % 6) == 0) then begin
-            set_global_var(155, global_var(155) + 1);
+        set_global_var(GVAR_GECKO_SKEETER_PART, global_var(GVAR_GECKO_SKEETER_PART) + 1);
+        if ((global_var(GVAR_GECKO_SKEETER_PART) % 6) == 0) then begin
+            set_global_var(GVAR_BRAIN_BOT_BRAIN, global_var(GVAR_BRAIN_BOT_BRAIN) + 1);
         end
     end
 end
@@ -146,10 +146,10 @@ procedure look_at_p_proc
 begin
     script_overrides;
     if (local_var(4)) then begin
-        display_msg(message_str(367, 100));
+        display_msg(message_str(SCRIPT_ESMINCRT, 100));
     end
     else begin
-        display_msg(message_str(367, 101));
+        display_msg(message_str(SCRIPT_ESMINCRT, 101));
     end
 end
 
@@ -194,30 +194,30 @@ end
 
 procedure Flip00
 begin
-    float_msg(self_obj, message_str(367, 102), 0);
+    float_msg(self_obj, message_str(SCRIPT_ESMINCRT, 102), FLOAT_MSG_NORMAL);
     call FlipCombat();
 end
 
 procedure Flip01
 begin
     add_timer_event(self_obj, game_ticks(30), 1);
-    gsay_message(367, 103, 50);
+    gsay_message(SCRIPT_ESMINCRT, 103, NEUTRAL_REACTION);
 end
 
 procedure Flip02
 begin
     set_local_var(4, 1);
-    gsay_reply(367, 104);
-    giq_option(4, 367, 105, FlipCombat, 50);
-    giq_option(4, 367, 106, transit, 50);
-    giq_option(-3, 367, 107, FlipCombat, 50);
+    gsay_reply(SCRIPT_ESMINCRT, 104);
+    giq_option(4, SCRIPT_ESMINCRT, 105, FlipCombat, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_ESMINCRT, 106, transit, NEUTRAL_REACTION);
+    giq_option(-3, SCRIPT_ESMINCRT, 107, FlipCombat, NEUTRAL_REACTION);
 end
 
 procedure Flip03
 begin
-    gsay_reply(367, 108);
-    giq_option(4, 367, 109, transit, 50);
-    giq_option(4, 367, 110, Flip03a, 50);
+    gsay_reply(SCRIPT_ESMINCRT, 108);
+    giq_option(4, SCRIPT_ESMINCRT, 109, transit, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_ESMINCRT, 110, Flip03a, NEUTRAL_REACTION);
 end
 
 procedure Flip03a
@@ -232,13 +232,13 @@ end
 
 procedure Flip04
 begin
-    gsay_message(367, 111, 50);
+    gsay_message(SCRIPT_ESMINCRT, 111, NEUTRAL_REACTION);
     call FlipCombat();
 end
 
 procedure Flip05
 begin
-    gsay_message(367, 112, 50);
+    gsay_message(SCRIPT_ESMINCRT, 112, NEUTRAL_REACTION);
     call FlipCombat();
 end
 
@@ -249,7 +249,7 @@ end
 
 procedure transit
 begin
-    set_global_var(57, 2);
+    set_global_var(GVAR_TOWN_REP_NCR, 2);
     load_map(32, 5);
 end
 
@@ -270,31 +270,31 @@ begin
         set_local_var(0, 50);
         set_local_var(1, 2);
         set_local_var(2, 1);
-        set_local_var(0, local_var(0) + (5 * get_critter_stat(dude_obj, 3)) - 25);
-        set_local_var(0, local_var(0) + (10 * has_trait(0, dude_obj, 10)));
-        if (has_trait(0, dude_obj, 39)) then begin
-            if (global_var(155) > 0) then begin
-                set_local_var(0, local_var(0) + global_var(155));
+        set_local_var(0, local_var(0) + (5 * dude_charisma) - 25);
+        set_local_var(0, local_var(0) + (10 * has_trait(TRAIT_PERK, dude_obj, PERK_presence)));
+        if (has_trait(TRAIT_PERK, dude_obj, PERK_cult_of_personality)) then begin
+            if (global_var(GVAR_BRAIN_BOT_BRAIN) > 0) then begin
+                set_local_var(0, local_var(0) + global_var(GVAR_BRAIN_BOT_BRAIN));
             end
             else begin
-                set_local_var(0, local_var(0) - global_var(155));
+                set_local_var(0, local_var(0) - global_var(GVAR_BRAIN_BOT_BRAIN));
             end
         end
         else begin
             if (local_var(3) == 1) then begin
-                set_local_var(0, local_var(0) - global_var(155));
+                set_local_var(0, local_var(0) - global_var(GVAR_BRAIN_BOT_BRAIN));
             end
             else begin
-                set_local_var(0, local_var(0) + global_var(155));
+                set_local_var(0, local_var(0) + global_var(GVAR_BRAIN_BOT_BRAIN));
             end
         end
-        if (global_var(158) > 2) then begin
+        if (global_var(GVAR_GECKO_ECON_DISK) > 2) then begin
             set_local_var(0, local_var(0) - 30);
         end
-        if (((global_var(160) + global_var(159)) >= 25) and ((global_var(160) > (3 * global_var(159))) or (global_var(157) == 1))) then begin
+        if (((global_var(GVAR_GECKO_SKEETER_PART) + global_var(GVAR_GECKO_REQ_FORM)) >= 25) and ((global_var(160) > (3 * global_var(GVAR_GECKO_REQ_FORM))) or (global_var(GVAR_SIERRA_PASSWORD) == 1))) then begin
             set_local_var(0, local_var(0) + 20);
         end
-        if (((global_var(160) + global_var(159)) >= 25) and ((global_var(159) > (2 * global_var(160))) or (global_var(156) == 1))) then begin
+        if (((global_var(GVAR_GECKO_SKEETER_PART) + global_var(GVAR_GECKO_REQ_FORM)) >= 25) and ((global_var(GVAR_GECKO_REQ_FORM) > (2 * global_var(GVAR_GECKO_SKEETER_PART))) or (global_var(GVAR_SIERRA_LOCKOUT) == 1))) then begin
             set_local_var(0, local_var(0) - 20);
         end
         call ReactToLevel();
@@ -387,6 +387,5 @@ end
 
 procedure Goodbyes
 begin
-    exit_line := message_str(634, random(100, 105));
+    exit_line := message_str(SCRIPT_BSCOMP3, random(100, 105));
 end
-
