diff --git a/Games/Fallout2.Steam/Original/FSELCOMP.ssl b/Games/Fallout2.Steam/ReDefine/FSELCOMP.ssl
index 2314007..0cc2533 100644
--- a/Games/Fallout2.Steam/Original/FSELCOMP.ssl
+++ b/Games/Fallout2.Steam/ReDefine/FSELCOMP.ssl
@@ -180,17 +180,17 @@ procedure look_at_p_proc
 begin
     script_overrides;
     if (local_var(4) == 0) then begin
-        display_msg(message_str(978, 100));
+        display_mstr(100);
     end
     else begin
-        display_msg(message_str(978, 101));
+        display_mstr(101);
     end
 end
 
 procedure description_p_proc
 begin
     script_overrides;
-    display_msg(message_str(978, 102));
+    display_mstr(102);
 end
 
 procedure use_skill_on_p_proc
@@ -200,7 +200,7 @@ end
 
 procedure use_obj_on_p_proc
 begin
-    display_msg(message_str(978, 132));
+    display_mstr(132);
 end
 
 procedure damage_p_proc
@@ -214,11 +214,11 @@ end
 procedure use_p_proc
 begin
     script_overrides;
-    if (not((global_var(361) bwand 524288) != 0)) then begin
+    if (not((global_var(GVAR_SAN_FRAN_FLAGS) bwand 524288) != 0)) then begin
         dialogue_system_enter;
     end
     else begin
-        display_msg(message_str(978, 103));
+        display_mstr(103);
     end
 end
 
@@ -232,82 +232,81 @@ end
 
 procedure Node001
 begin
-    display_msg(message_str(978, 103));
+    display_mstr(103);
 end
 
 procedure Node002
 begin
-    gsay_reply(978, 104);
-    giq_option(1, 978, 105, Node999, 50);
+    Reply(104);
+    giq_option(1, NAME, 105, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node003
 begin
     if (local_var(4) == 0) then begin
-        gsay_reply(978, 106);
+        Reply(106);
     end
     else begin
-        gsay_reply(978, 107);
+        Reply(107);
     end
-    giq_option(1, 978, 108, Node004, 50);
-    giq_option(1, 978, 109, Node008, 50);
-    giq_option(1, 978, 110, Node999, 50);
+    giq_option(1, NAME, 108, Node004, NEUTRAL_REACTION);
+    giq_option(1, NAME, 109, Node008, NEUTRAL_REACTION);
+    giq_option(1, NAME, 110, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node004
 begin
-    gsay_reply(978, 111);
-    giq_option(1, 978, 112, Node005, 50);
-    giq_option(1, 978, 113, Node006, 50);
-    giq_option(1, 978, 114, Node007, 50);
-    giq_option(1, 978, 115, Node003, 50);
+    Reply(111);
+    giq_option(1, NAME, 112, Node005, NEUTRAL_REACTION);
+    giq_option(1, NAME, 113, Node006, NEUTRAL_REACTION);
+    giq_option(1, NAME, 114, Node007, NEUTRAL_REACTION);
+    giq_option(1, NAME, 115, Node003, NEUTRAL_REACTION);
 end
 
 procedure Node005
 begin
-    gsay_reply(978, 116);
-    giq_option(1, 978, 117, Node004, 50);
-    giq_option(1, 978, 118, Node999, 50);
+    Reply(116);
+    giq_option(1, NAME, 117, Node004, NEUTRAL_REACTION);
+    giq_option(1, NAME, 118, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node006
 begin
-    gsay_reply(978, 119);
-    giq_option(1, 978, 120, Node004, 50);
-    giq_option(1, 978, 121, Node999, 50);
+    Reply(119);
+    giq_option(1, NAME, 120, Node004, NEUTRAL_REACTION);
+    giq_option(1, NAME, 121, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node007
 begin
-    gsay_reply(978, 122);
-    giq_option(1, 978, 123, Node004, 50);
-    giq_option(1, 978, 124, Node999, 50);
+    Reply(122);
+    giq_option(1, NAME, 123, Node004, NEUTRAL_REACTION);
+    giq_option(1, NAME, 124, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node008
 begin
-    gsay_reply(978, 125);
-    giq_option(1, 978, 126, Node009, 50);
-    giq_option(1, 978, 127, Node010, 50);
-    giq_option(1, 978, 128, Node003, 50);
-    giq_option(1, 978, 129, Node999, 50);
+    Reply(125);
+    giq_option(1, NAME, 126, Node009, NEUTRAL_REACTION);
+    giq_option(1, NAME, 127, Node010, NEUTRAL_REACTION);
+    giq_option(1, NAME, 128, Node003, NEUTRAL_REACTION);
+    giq_option(1, NAME, 129, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node009
 begin
-    gsay_reply(978, 130);
-    giq_option(1, 978, 131, Node999, 50);
+    Reply(130);
+    giq_option(1, NAME, 131, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node010
 begin
-    if ((global_var(361) bwand 32768) != 0) then begin
-        gsay_reply(978, message_str(978, 133) + " " + message_str(978, 135));
+    if ((global_var(GVAR_SAN_FRAN_FLAGS) bwand 32768) != 0) then begin
+        Reply(mstr(133) + " " + mstr(135));
     end
     else begin
-        gsay_reply(978, message_str(978, 134) + " " + message_str(978, 135));
+        Reply(mstr(134) + " " + mstr(135));
     end
-    giq_option(4, 978, 135, Node008, 50);
-    giq_option(4, 978, 136, Node999, 50);
+    giq_option(4, NAME, 135, Node008, NEUTRAL_REACTION);
+    giq_option(4, NAME, 136, Node999, NEUTRAL_REACTION);
 end
-
