diff --git a/Games/Fallout2.Steam/Original/GCRDESK.ssl b/Games/Fallout2.Steam/ReDefine/GCRDESK.ssl
index b0dfa5d..9b421fa 100644
--- a/Games/Fallout2.Steam/Original/GCRDESK.ssl
+++ b/Games/Fallout2.Steam/ReDefine/GCRDESK.ssl
@@ -168,14 +168,14 @@ end
 procedure critter_p_proc
 begin
     if (obj_can_see_obj(self_obj, dude_obj)) then begin
-        if (global_var(395) == 1) then begin
+        if (global_var(GVAR_GECKO_MELTDOWN) == 1) then begin
             set_local_var(4, 1);
-            attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+            attack(dude_obj);
         end
         else begin
             if ((local_var(4) == 2) or (local_var(5) == 1)) then begin
                 set_local_var(4, 1);
-                attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+                attack(dude_obj);
             end
         end
     end
@@ -190,7 +190,7 @@ end
 
 procedure talk_p_proc
 begin
-    if ((global_var(238) == 1) or (local_var(5) == 1)) then begin
+    if ((global_var(GVAR_ENEMY_GECKO) == 1) or (local_var(5) == 1)) then begin
         call Node998();
     end
     else begin
@@ -215,59 +215,59 @@ end
 procedure destroy_p_proc
 begin
     if (source_obj == dude_obj) then begin
-        set_global_var(5, global_var(5) + 1);
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(4) > (3 * global_var(5))) or (global_var(2) == 1)) and (global_var(1) == 0)) then begin
-            set_global_var(2, 1);
-            set_global_var(3, 0);
+        set_global_var(GVAR_GOOD_MONSTER, global_var(GVAR_GOOD_MONSTER) + 1);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(4) > (3 * global_var(GVAR_GOOD_MONSTER))) or (global_var(GVAR_CHAMPION_REPUTATION) == 1)) and (global_var(GVAR_CHILDKILLER_REPUTATION) == 0)) then begin
+            set_global_var(GVAR_CHAMPION_REPUTATION, 1);
+            set_global_var(GVAR_BERSERKER_REPUTATION, 0);
         end
-        if (metarule(51, self_obj) == 2) then begin
+        if (critter_kill_type(self_obj) == KILL_TYPE_children_kills) then begin
             if (source_obj == dude_obj) then begin
-                set_global_var(2, 0);
-                set_global_var(1, global_var(1) + 1);
-                set_global_var(0, global_var(0) + -15);
+                set_global_var(GVAR_CHAMPION_REPUTATION, 0);
+                set_global_var(GVAR_CHILDKILLER_REPUTATION, global_var(GVAR_CHILDKILLER_REPUTATION) + 1);
+                set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + -15);
                 debug_msg("Player gains " + -15 + " Karma Points.");
-                set_global_var(37, 0);
-                set_global_var(38, 0);
-                set_global_var(39, 0);
-                set_global_var(40, 0);
-                set_global_var(41, 0);
-                set_global_var(42, 0);
-                set_global_var(43, 0);
-                set_global_var(44, 0);
-                set_global_var(45, 0);
-                if (global_var(0) >= 1000) then begin
-                    set_global_var(37, 1);
+                set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+                set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+                set_global_var(GVAR_KARMA_DEFENDER, 0);
+                set_global_var(GVAR_KARMA_WANDERER, 0);
+                set_global_var(GVAR_KARMA_BETRAYER, 0);
+                set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+                set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+                set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+                    set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
                 end
                 else begin
-                    if (global_var(0) >= 750) then begin
-                        set_global_var(38, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
                     end
                     else begin
-                        if (global_var(0) >= 500) then begin
-                            set_global_var(39, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                            set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                         end
                         else begin
-                            if (global_var(0) >= 250) then begin
-                                set_global_var(40, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                                set_global_var(GVAR_KARMA_DEFENDER, 1);
                             end
                             else begin
-                                if (global_var(0) > -250) then begin
-                                    set_global_var(41, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                                    set_global_var(GVAR_KARMA_WANDERER, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -500) then begin
-                                        set_global_var(42, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                        set_global_var(GVAR_KARMA_BETRAYER, 1);
                                     end
                                     else begin
-                                        if (global_var(0) > -750) then begin
-                                            set_global_var(43, 1);
+                                        if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                            set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                         end
                                         else begin
-                                            if (global_var(0) > -1000) then begin
-                                                set_global_var(44, 1);
+                                            if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                                set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                             end
                                             else begin
-                                                set_global_var(45, 1);
+                                                set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                             end
                                         end
                                     end
@@ -276,58 +276,58 @@ begin
                         end
                     end
                 end
-                set_global_var(51, global_var(51) + -8);
+                set_global_var(GVAR_TOWN_REP_GECKO, global_var(GVAR_TOWN_REP_GECKO) + -8);
                 debug_msg("Added " + -8 + " to Town Rep");
             end
         end
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(5) > (2 * global_var(4))) or (global_var(3) == 1))) then begin
-            set_global_var(3, 1);
-            set_global_var(2, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
+            set_global_var(GVAR_BERSERKER_REPUTATION, 1);
+            set_global_var(GVAR_CHAMPION_REPUTATION, 0);
         end
-        set_global_var(0, global_var(0) + -10);
+        set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + -10);
         debug_msg("Player gains " + -10 + " Karma Points.");
-        set_global_var(37, 0);
-        set_global_var(38, 0);
-        set_global_var(39, 0);
-        set_global_var(40, 0);
-        set_global_var(41, 0);
-        set_global_var(42, 0);
-        set_global_var(43, 0);
-        set_global_var(44, 0);
-        set_global_var(45, 0);
-        if (global_var(0) >= 1000) then begin
-            set_global_var(37, 1);
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+        set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+        set_global_var(GVAR_KARMA_DEFENDER, 0);
+        set_global_var(GVAR_KARMA_WANDERER, 0);
+        set_global_var(GVAR_KARMA_BETRAYER, 0);
+        set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+        set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+            set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
         end
         else begin
-            if (global_var(0) >= 750) then begin
-                set_global_var(38, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
             end
             else begin
-                if (global_var(0) >= 500) then begin
-                    set_global_var(39, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                 end
                 else begin
-                    if (global_var(0) >= 250) then begin
-                        set_global_var(40, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                        set_global_var(GVAR_KARMA_DEFENDER, 1);
                     end
                     else begin
-                        if (global_var(0) > -250) then begin
-                            set_global_var(41, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                            set_global_var(GVAR_KARMA_WANDERER, 1);
                         end
                         else begin
-                            if (global_var(0) > -500) then begin
-                                set_global_var(42, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                set_global_var(GVAR_KARMA_BETRAYER, 1);
                             end
                             else begin
-                                if (global_var(0) > -750) then begin
-                                    set_global_var(43, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -1000) then begin
-                                        set_global_var(44, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                     end
                                     else begin
-                                        set_global_var(45, 1);
+                                        set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                     end
                                 end
                             end
@@ -336,7 +336,7 @@ begin
                 end
             end
         end
-        set_global_var(51, global_var(51) + -5);
+        set_global_var(GVAR_TOWN_REP_GECKO, global_var(GVAR_TOWN_REP_GECKO) + -5);
         debug_msg("Added " + -5 + " to Town Rep");
     end
 end
@@ -344,13 +344,13 @@ end
 procedure look_at_p_proc
 begin
     script_overrides;
-    display_msg(message_str(140, 100));
+    display_mstr(100);
 end
 
 procedure description_p_proc
 begin
     script_overrides;
-    display_msg(message_str(140, 101));
+    display_mstr(101);
 end
 
 procedure use_skill_on_p_proc
@@ -361,115 +361,115 @@ procedure damage_p_proc
 begin
     if (source_obj == dude_obj) then begin
         set_local_var(5, 1);
-        set_global_var(238, 1);
+        set_global_var(GVAR_ENEMY_GECKO, 1);
     end
 end
 
 procedure map_enter_p_proc
 begin
     Only_Once := 0;
-    critter_add_trait(self_obj, 1, 6, 49);
-    critter_add_trait(self_obj, 1, 5, 22);
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 49);
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_TOUGH_GUARD);
 end
 
 procedure Node001
 begin
-    if (global_var(51) < 5) then begin
-        gsay_reply(140, 110);
+    if (global_var(GVAR_TOWN_REP_GECKO) < 5) then begin
+        Reply(110);
     end
     else begin
-        gsay_reply(140, 111);
+        Reply(111);
     end
-    giq_option(-3, 140, 112, Node999, 50);
-    giq_option(4, 140, 113, Node003, 50);
-    giq_option(4, 140, 114, Node999, 50);
+    giq_option(-3, NAME, 112, Node999, NEUTRAL_REACTION);
+    giq_option(4, NAME, 113, Node003, NEUTRAL_REACTION);
+    giq_option(4, NAME, 114, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node002
 begin
-    gsay_reply(140, 120);
-    giq_option(-3, 140, 121, Node999, 50);
-    giq_option(4, 140, 122, Node003, 50);
+    Reply(120);
+    giq_option(-3, NAME, 121, Node999, NEUTRAL_REACTION);
+    giq_option(4, NAME, 122, Node003, NEUTRAL_REACTION);
     if (local_var(7) == 0) then begin
-        giq_option(4, 140, 123, Node007, 50);
+        giq_option(4, NAME, 123, Node007, NEUTRAL_REACTION);
     end
-    giq_option(4, 140, 124, Node999, 50);
+    giq_option(4, NAME, 124, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node02a
 begin
-    gsay_reply(140, 300);
-    giq_option(4, 140, 301, Node003, 50);
+    Reply(300);
+    giq_option(4, NAME, 301, Node003, NEUTRAL_REACTION);
     if (local_var(7) == 0) then begin
-        giq_option(4, 140, 123, Node007, 50);
+        giq_option(4, NAME, 123, Node007, NEUTRAL_REACTION);
     end
-    giq_option(4, 140, 194, Node999, 50);
+    giq_option(4, NAME, 194, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node003
 begin
-    gsay_reply(140, 130);
-    giq_option(4, 140, 131, Node008, 50);
-    giq_option(4, 140, 132, Node006, 50);
-    giq_option(4, 140, 133, Node009, 50);
-    giq_option(4, 140, 134, Node005, 50);
-    giq_option(4, 140, 135, Node999, 50);
+    Reply(130);
+    giq_option(4, NAME, 131, Node008, NEUTRAL_REACTION);
+    giq_option(4, NAME, 132, Node006, NEUTRAL_REACTION);
+    giq_option(4, NAME, 133, Node009, NEUTRAL_REACTION);
+    giq_option(4, NAME, 134, Node005, NEUTRAL_REACTION);
+    giq_option(4, NAME, 135, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node004
 begin
-    gsay_reply(140, 140);
-    giq_option(4, 140, 141, Node002, 50);
-    giq_option(4, 140, 142, Node999, 50);
+    Reply(140);
+    giq_option(4, NAME, 141, Node002, NEUTRAL_REACTION);
+    giq_option(4, NAME, 142, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node005
 begin
-    gsay_reply(140, 150);
-    giq_option(4, 140, 151, Node009, 50);
-    giq_option(4, 140, 152, Node002, 50);
-    giq_option(4, 140, 153, Node004, 50);
-    giq_option(4, 140, 154, Node999, 50);
+    Reply(150);
+    giq_option(4, NAME, 151, Node009, NEUTRAL_REACTION);
+    giq_option(4, NAME, 152, Node002, NEUTRAL_REACTION);
+    giq_option(4, NAME, 153, Node004, NEUTRAL_REACTION);
+    giq_option(4, NAME, 154, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node006
 begin
-    gsay_reply(140, 160);
-    giq_option(4, 140, 161, Node003, 50);
-    giq_option(4, 140, 162, Node999, 50);
+    Reply(160);
+    giq_option(4, NAME, 161, Node003, NEUTRAL_REACTION);
+    giq_option(4, NAME, 162, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node007
 begin
     set_local_var(7, 1);
-    gsay_reply(140, 170);
-    giq_option(4, 140, 171, Node002, 50);
-    giq_option(6, 140, 172, Node010, 50);
-    giq_option(4, 140, 173, Node999, 50);
+    Reply(170);
+    giq_option(4, NAME, 171, Node002, NEUTRAL_REACTION);
+    giq_option(6, NAME, 172, Node010, NEUTRAL_REACTION);
+    giq_option(4, NAME, 173, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node008
 begin
-    gsay_reply(140, 180);
-    giq_option(4, 140, 181, Node02a, 50);
-    giq_option(4, 140, 182, Node004, 50);
-    giq_option(4, 140, 183, Node999, 50);
+    Reply(180);
+    giq_option(4, NAME, 181, Node02a, NEUTRAL_REACTION);
+    giq_option(4, NAME, 182, Node004, NEUTRAL_REACTION);
+    giq_option(4, NAME, 183, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node009
 begin
-    gsay_reply(140, 190);
-    giq_option(4, 140, 191, Node009, 50);
-    giq_option(4, 140, 192, Node002, 50);
-    giq_option(4, 140, 193, Node004, 50);
-    giq_option(4, 140, 194, Node999, 50);
+    Reply(190);
+    giq_option(4, NAME, 191, Node009, NEUTRAL_REACTION);
+    giq_option(4, NAME, 192, Node002, NEUTRAL_REACTION);
+    giq_option(4, NAME, 193, Node004, NEUTRAL_REACTION);
+    giq_option(4, NAME, 194, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node010
 begin
-    gsay_reply(140, 200);
-    giq_option(4, 140, 201, Node009, 50);
-    giq_option(4, 140, 202, Node002, 50);
+    Reply(200);
+    giq_option(4, NAME, 201, Node009, NEUTRAL_REACTION);
+    giq_option(4, NAME, 202, Node002, NEUTRAL_REACTION);
 end
 
 procedure Node998
@@ -480,4 +480,3 @@ end
 procedure Node999
 begin
 end
-
