diff --git a/Games/Fallout2.Steam/Original/GCSKEETR.ssl b/Games/Fallout2.Steam/ReDefine/GCSKEETR.ssl
index ef83f1d..64e3219 100644
--- a/Games/Fallout2.Steam/Original/GCSKEETR.ssl
+++ b/Games/Fallout2.Steam/ReDefine/GCSKEETR.ssl
@@ -224,14 +224,14 @@ end
 
 procedure critter_p_proc
 begin
-    if ((global_var(395) == 1) and obj_can_see_obj(self_obj, dude_obj)) then begin
+    if ((global_var(GVAR_GECKO_MELTDOWN) == 1) and obj_can_see_obj(self_obj, dude_obj)) then begin
         set_local_var(5, 1);
-        attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+        attack(dude_obj);
     end
     else begin
         if ((local_var(5) == 2) and obj_can_see_obj(self_obj, dude_obj)) then begin
             set_local_var(5, 1);
-            attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+            attack(dude_obj);
         end
     end
 end
@@ -247,7 +247,7 @@ procedure talk_p_proc
 begin
     if (local_var(2) == 0) then begin
         set_local_var(2, 1);
-        set_local_var(3, (get_critter_stat(dude_obj, 3) - get_critter_stat(self_obj, 3)) * 5);
+        set_local_var(3, (dude_charisma - self_charisma) * 5);
         debug_msg("Base Reaction == " + local_var(3));
     end
     Static_Reaction := 0;
@@ -259,14 +259,14 @@ begin
     else begin
         reaction_bonus_karma := -1 * (-1 * reaction_bonus_karma / 1000);
     end
-    if (global_var(50) >= 0) then begin
-        reaction_bonus_town_rep := global_var(50) / 2;
+    if (global_var(GVAR_TOWN_REP_VAULT_CITY) >= 0) then begin
+        reaction_bonus_town_rep := global_var(GVAR_TOWN_REP_VAULT_CITY) / 2;
     end
     else begin
-        reaction_bonus_town_rep := -1 * (-1 * global_var(50) / 2);
+        reaction_bonus_town_rep := -1 * (-1 * global_var(GVAR_TOWN_REP_VAULT_CITY) / 2);
     end
-    if (has_trait(2, dude_obj, 13)) then begin
-        if (get_critter_stat(dude_obj, 34) == get_critter_stat(self_obj, 34)) then begin
+    if (has_trait(TRAIT_TRAIT, dude_obj, TRAIT_sex_appeal)) then begin
+        if (dude_gender == self_gender) then begin
             Static_Reaction := Static_Reaction + -20;
         end
         else begin
@@ -282,48 +282,48 @@ begin
         debug_msg("Karma Reaction Bonus == " + Static_Reaction);
         Static_Reaction := Static_Reaction - reaction_bonus_town_rep;
         debug_msg("Town Rep Bonus == " + Static_Reaction);
-        set_global_var(37, 0);
-        set_global_var(38, 0);
-        set_global_var(39, 0);
-        set_global_var(40, 0);
-        set_global_var(41, 0);
-        set_global_var(42, 0);
-        set_global_var(43, 0);
-        set_global_var(44, 0);
-        set_global_var(45, 0);
-        if (global_var(0) >= 1000) then begin
-            set_global_var(37, 1);
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+        set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+        set_global_var(GVAR_KARMA_DEFENDER, 0);
+        set_global_var(GVAR_KARMA_WANDERER, 0);
+        set_global_var(GVAR_KARMA_BETRAYER, 0);
+        set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+        set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+            set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
         end
         else begin
-            if (global_var(0) >= 750) then begin
-                set_global_var(38, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
             end
             else begin
-                if (global_var(0) >= 500) then begin
-                    set_global_var(39, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                 end
                 else begin
-                    if (global_var(0) >= 250) then begin
-                        set_global_var(40, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                        set_global_var(GVAR_KARMA_DEFENDER, 1);
                     end
                     else begin
-                        if (global_var(0) > -250) then begin
-                            set_global_var(41, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                            set_global_var(GVAR_KARMA_WANDERER, 1);
                         end
                         else begin
-                            if (global_var(0) > -500) then begin
-                                set_global_var(42, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                set_global_var(GVAR_KARMA_BETRAYER, 1);
                             end
                             else begin
-                                if (global_var(0) > -750) then begin
-                                    set_global_var(43, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -1000) then begin
-                                        set_global_var(44, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                     end
                                     else begin
-                                        set_global_var(45, 1);
+                                        set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                     end
                                 end
                             end
@@ -332,35 +332,35 @@ begin
                 end
             end
         end
-        if (global_var(37) == 1) then begin
+        if (global_var(GVAR_KARMA_HOLY_WARRIOR) == 1) then begin
             Static_Reaction := Static_Reaction - 20;
         end
         else begin
-            if (global_var(38) == 1) then begin
+            if (global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES) == 1) then begin
                 Static_Reaction := Static_Reaction - 15;
             end
             else begin
-                if (global_var(39) == 1) then begin
+                if (global_var(GVAR_KARMA_SHIELD_OF_HOPE) == 1) then begin
                     Static_Reaction := Static_Reaction - 10;
                 end
                 else begin
-                    if (global_var(40) == 1) then begin
+                    if (global_var(GVAR_KARMA_DEFENDER) == 1) then begin
                         Static_Reaction := Static_Reaction - 5;
                     end
                     else begin
-                        if (global_var(41) == 1) then begin
+                        if (global_var(GVAR_KARMA_WANDERER) == 1) then begin
                             Static_Reaction := Static_Reaction - 0;
                         end
                         else begin
-                            if (global_var(42) == 1) then begin
+                            if (global_var(GVAR_KARMA_BETRAYER) == 1) then begin
                                 Static_Reaction := Static_Reaction - -5;
                             end
                             else begin
-                                if (global_var(43) == 1) then begin
+                                if (global_var(GVAR_KARMA_SWORD_OF_DESPAIR) == 1) then begin
                                     Static_Reaction := Static_Reaction - -10;
                                 end
                                 else begin
-                                    if (global_var(44) == 1) then begin
+                                    if (global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE) == 1) then begin
                                         Static_Reaction := Static_Reaction - -15;
                                     end
                                     else begin
@@ -380,48 +380,48 @@ begin
         debug_msg("Karma Reaction Bonus == " + Static_Reaction);
         Static_Reaction := Static_Reaction + reaction_bonus_town_rep;
         debug_msg("Town Rep Bonus == " + Static_Reaction);
-        set_global_var(37, 0);
-        set_global_var(38, 0);
-        set_global_var(39, 0);
-        set_global_var(40, 0);
-        set_global_var(41, 0);
-        set_global_var(42, 0);
-        set_global_var(43, 0);
-        set_global_var(44, 0);
-        set_global_var(45, 0);
-        if (global_var(0) >= 1000) then begin
-            set_global_var(37, 1);
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+        set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+        set_global_var(GVAR_KARMA_DEFENDER, 0);
+        set_global_var(GVAR_KARMA_WANDERER, 0);
+        set_global_var(GVAR_KARMA_BETRAYER, 0);
+        set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+        set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+            set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
         end
         else begin
-            if (global_var(0) >= 750) then begin
-                set_global_var(38, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
             end
             else begin
-                if (global_var(0) >= 500) then begin
-                    set_global_var(39, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                 end
                 else begin
-                    if (global_var(0) >= 250) then begin
-                        set_global_var(40, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                        set_global_var(GVAR_KARMA_DEFENDER, 1);
                     end
                     else begin
-                        if (global_var(0) > -250) then begin
-                            set_global_var(41, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                            set_global_var(GVAR_KARMA_WANDERER, 1);
                         end
                         else begin
-                            if (global_var(0) > -500) then begin
-                                set_global_var(42, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                set_global_var(GVAR_KARMA_BETRAYER, 1);
                             end
                             else begin
-                                if (global_var(0) > -750) then begin
-                                    set_global_var(43, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -1000) then begin
-                                        set_global_var(44, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                     end
                                     else begin
-                                        set_global_var(45, 1);
+                                        set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                     end
                                 end
                             end
@@ -430,35 +430,35 @@ begin
                 end
             end
         end
-        if (global_var(37) == 1) then begin
+        if (global_var(GVAR_KARMA_HOLY_WARRIOR) == 1) then begin
             Static_Reaction := Static_Reaction + 20;
         end
         else begin
-            if (global_var(38) == 1) then begin
+            if (global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES) == 1) then begin
                 Static_Reaction := Static_Reaction + 15;
             end
             else begin
-                if (global_var(39) == 1) then begin
+                if (global_var(GVAR_KARMA_SHIELD_OF_HOPE) == 1) then begin
                     Static_Reaction := Static_Reaction + 10;
                 end
                 else begin
-                    if (global_var(40) == 1) then begin
+                    if (global_var(GVAR_KARMA_DEFENDER) == 1) then begin
                         Static_Reaction := Static_Reaction + 5;
                     end
                     else begin
-                        if (global_var(41) == 1) then begin
+                        if (global_var(GVAR_KARMA_WANDERER) == 1) then begin
                             Static_Reaction := Static_Reaction + 0;
                         end
                         else begin
-                            if (global_var(42) == 1) then begin
+                            if (global_var(GVAR_KARMA_BETRAYER) == 1) then begin
                                 Static_Reaction := Static_Reaction + -5;
                             end
                             else begin
-                                if (global_var(43) == 1) then begin
+                                if (global_var(GVAR_KARMA_SWORD_OF_DESPAIR) == 1) then begin
                                     Static_Reaction := Static_Reaction + -10;
                                 end
                                 else begin
-                                    if (global_var(44) == 1) then begin
+                                    if (global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE) == 1) then begin
                                         Static_Reaction := Static_Reaction + -15;
                                     end
                                     else begin
@@ -474,28 +474,28 @@ begin
         debug_msg("Good Person Reaction Bonus == " + Static_Reaction);
     end
     if (Slavery_Tolerant == 0) then begin
-        if (global_var(11) == 1) then begin
+        if (global_var(GVAR_REPUTATION_SLAVER) == 1) then begin
             Static_Reaction := Static_Reaction + (2 * -25);
         end
     end
     else begin
         if (Slavery_Tolerant == 1) then begin
-            if (global_var(11) == 1) then begin
+            if (global_var(GVAR_REPUTATION_SLAVER) == 1) then begin
                 Static_Reaction := Static_Reaction + -25;
             end
         end
         else begin
             if (Slavery_Tolerant == 3) then begin
-                if (global_var(11) == 1) then begin
+                if (global_var(GVAR_REPUTATION_SLAVER) == 1) then begin
                     Static_Reaction := Static_Reaction - -25;
                 end
             end
         end
     end
     debug_msg("Slaver + Aligned Reaction == " + Static_Reaction);
-    Static_Reaction := Static_Reaction + (10 * has_trait(0, dude_obj, 10));
+    Static_Reaction := Static_Reaction + (10 * has_trait(TRAIT_PERK, dude_obj, PERK_presence));
     debug_msg("Presence Reaction == " + Static_Reaction);
-    if (global_var(1) >= 2) then begin
+    if (global_var(GVAR_CHILDKILLER_REPUTATION) >= 2) then begin
         Static_Reaction := Static_Reaction + -30;
     end
     debug_msg("Childkiller Reaction == " + Static_Reaction);
@@ -505,7 +505,7 @@ begin
         call Node998();
     end
     else begin
-        if (get_critter_stat(dude_obj, 4) <= 3) then begin
+        if (dude_is_stupid) then begin
             call Node001();
         end
         else begin
@@ -541,7 +541,7 @@ begin
                             end_dialogue;
                         end
                         else begin
-                            if (global_var(160) == 1) then begin
+                            if (global_var(GVAR_GECKO_SKEETER_PART) == 1) then begin
                                 start_gdialog(136, self_obj, 4, -1, -1);
                                 gsay_start;
                                 call Node008();
@@ -549,7 +549,7 @@ begin
                                 end_dialogue;
                             end
                             else begin
-                                if (global_var(83) == 9) then begin
+                                if (global_var(GVAR_VAULT_PLANT_STATUS) == 9) then begin
                                     start_gdialog(136, self_obj, 4, -1, -1);
                                     gsay_start;
                                     call Node007();
@@ -557,7 +557,7 @@ begin
                                     end_dialogue;
                                 end
                                 else begin
-                                    if (global_var(83) == 15) then begin
+                                    if (global_var(GVAR_VAULT_PLANT_STATUS) == 15) then begin
                                         start_gdialog(136, self_obj, 4, -1, -1);
                                         gsay_start;
                                         call Node007();
@@ -584,63 +584,63 @@ end
 procedure destroy_p_proc
 begin
     if (local_var(7) < 2) then begin
-        item := create_object_sid(253, 0, 0, -1);
+        item := create_object(253, 0, 0);
         add_obj_to_inven(self_obj, item);
     end
     if (source_obj == dude_obj) then begin
-        set_global_var(5, global_var(5) + 1);
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(4) > (3 * global_var(5))) or (global_var(2) == 1)) and (global_var(1) == 0)) then begin
-            set_global_var(2, 1);
-            set_global_var(3, 0);
+        set_global_var(GVAR_GOOD_MONSTER, global_var(GVAR_GOOD_MONSTER) + 1);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(4) > (3 * global_var(GVAR_GOOD_MONSTER))) or (global_var(GVAR_CHAMPION_REPUTATION) == 1)) and (global_var(GVAR_CHILDKILLER_REPUTATION) == 0)) then begin
+            set_global_var(GVAR_CHAMPION_REPUTATION, 1);
+            set_global_var(GVAR_BERSERKER_REPUTATION, 0);
         end
-        if (metarule(51, self_obj) == 2) then begin
+        if (critter_kill_type(self_obj) == KILL_TYPE_children_kills) then begin
             if (source_obj == dude_obj) then begin
-                set_global_var(2, 0);
-                set_global_var(1, global_var(1) + 1);
-                set_global_var(0, global_var(0) + -15);
+                set_global_var(GVAR_CHAMPION_REPUTATION, 0);
+                set_global_var(GVAR_CHILDKILLER_REPUTATION, global_var(GVAR_CHILDKILLER_REPUTATION) + 1);
+                set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + -15);
                 debug_msg("Player gains " + -15 + " Karma Points.");
-                set_global_var(37, 0);
-                set_global_var(38, 0);
-                set_global_var(39, 0);
-                set_global_var(40, 0);
-                set_global_var(41, 0);
-                set_global_var(42, 0);
-                set_global_var(43, 0);
-                set_global_var(44, 0);
-                set_global_var(45, 0);
-                if (global_var(0) >= 1000) then begin
-                    set_global_var(37, 1);
+                set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+                set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+                set_global_var(GVAR_KARMA_DEFENDER, 0);
+                set_global_var(GVAR_KARMA_WANDERER, 0);
+                set_global_var(GVAR_KARMA_BETRAYER, 0);
+                set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+                set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+                set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+                    set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
                 end
                 else begin
-                    if (global_var(0) >= 750) then begin
-                        set_global_var(38, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
                     end
                     else begin
-                        if (global_var(0) >= 500) then begin
-                            set_global_var(39, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                            set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                         end
                         else begin
-                            if (global_var(0) >= 250) then begin
-                                set_global_var(40, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                                set_global_var(GVAR_KARMA_DEFENDER, 1);
                             end
                             else begin
-                                if (global_var(0) > -250) then begin
-                                    set_global_var(41, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                                    set_global_var(GVAR_KARMA_WANDERER, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -500) then begin
-                                        set_global_var(42, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                        set_global_var(GVAR_KARMA_BETRAYER, 1);
                                     end
                                     else begin
-                                        if (global_var(0) > -750) then begin
-                                            set_global_var(43, 1);
+                                        if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                            set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                         end
                                         else begin
-                                            if (global_var(0) > -1000) then begin
-                                                set_global_var(44, 1);
+                                            if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                                set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                             end
                                             else begin
-                                                set_global_var(45, 1);
+                                                set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                             end
                                         end
                                     end
@@ -649,58 +649,58 @@ begin
                         end
                     end
                 end
-                set_global_var(50, global_var(50) + -8);
+                set_global_var(GVAR_TOWN_REP_VAULT_CITY, global_var(GVAR_TOWN_REP_VAULT_CITY) + -8);
                 debug_msg("Added " + -8 + " to Town Rep");
             end
         end
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(5) > (2 * global_var(4))) or (global_var(3) == 1))) then begin
-            set_global_var(3, 1);
-            set_global_var(2, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
+            set_global_var(GVAR_BERSERKER_REPUTATION, 1);
+            set_global_var(GVAR_CHAMPION_REPUTATION, 0);
         end
-        set_global_var(0, global_var(0) + -10);
+        set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + -10);
         debug_msg("Player gains " + -10 + " Karma Points.");
-        set_global_var(37, 0);
-        set_global_var(38, 0);
-        set_global_var(39, 0);
-        set_global_var(40, 0);
-        set_global_var(41, 0);
-        set_global_var(42, 0);
-        set_global_var(43, 0);
-        set_global_var(44, 0);
-        set_global_var(45, 0);
-        if (global_var(0) >= 1000) then begin
-            set_global_var(37, 1);
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+        set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+        set_global_var(GVAR_KARMA_DEFENDER, 0);
+        set_global_var(GVAR_KARMA_WANDERER, 0);
+        set_global_var(GVAR_KARMA_BETRAYER, 0);
+        set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+        set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+            set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
         end
         else begin
-            if (global_var(0) >= 750) then begin
-                set_global_var(38, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
             end
             else begin
-                if (global_var(0) >= 500) then begin
-                    set_global_var(39, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                 end
                 else begin
-                    if (global_var(0) >= 250) then begin
-                        set_global_var(40, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                        set_global_var(GVAR_KARMA_DEFENDER, 1);
                     end
                     else begin
-                        if (global_var(0) > -250) then begin
-                            set_global_var(41, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                            set_global_var(GVAR_KARMA_WANDERER, 1);
                         end
                         else begin
-                            if (global_var(0) > -500) then begin
-                                set_global_var(42, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                set_global_var(GVAR_KARMA_BETRAYER, 1);
                             end
                             else begin
-                                if (global_var(0) > -750) then begin
-                                    set_global_var(43, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -1000) then begin
-                                        set_global_var(44, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                     end
                                     else begin
-                                        set_global_var(45, 1);
+                                        set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                     end
                                 end
                             end
@@ -709,7 +709,7 @@ begin
                 end
             end
         end
-        set_global_var(50, global_var(50) + -5);
+        set_global_var(GVAR_TOWN_REP_VAULT_CITY, global_var(GVAR_TOWN_REP_VAULT_CITY) + -5);
         debug_msg("Added " + -5 + " to Town Rep");
     end
 end
@@ -717,13 +717,13 @@ end
 procedure look_at_p_proc
 begin
     script_overrides;
-    display_msg(message_str(136, 100));
+    display_mstr(100);
 end
 
 procedure description_p_proc
 begin
     script_overrides;
-    display_msg(message_str(136, 101));
+    display_mstr(101);
 end
 
 procedure use_skill_on_p_proc
@@ -734,15 +734,15 @@ procedure damage_p_proc
 begin
     if (source_obj == dude_obj) then begin
         set_local_var(6, 1);
-        set_global_var(238, 1);
+        set_global_var(GVAR_ENEMY_GECKO, 1);
     end
 end
 
 procedure map_enter_p_proc
 begin
     Only_Once := 0;
-    critter_add_trait(self_obj, 1, 6, 49);
-    critter_add_trait(self_obj, 1, 5, 24);
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 49);
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_TOUGH_CITIZEN);
 end
 
 procedure combat
@@ -751,7 +751,7 @@ end
 
 procedure Node910
 begin
-    if (global_var(83) == 9) then begin
+    if (global_var(GVAR_VAULT_PLANT_STATUS) == 9) then begin
         call Node008();
     end
     else begin
@@ -801,10 +801,10 @@ begin
         set_local_var(8, 2);
     end
     else begin
-        item_caps_adjust(dude_obj, -(3000 * (250 / (160 + has_skill(dude_obj, 15)))));
-        item_caps_adjust(self_obj, 3000 * (250 / (160 + has_skill(dude_obj, 15))));
+        item_caps_adjust(dude_obj, -(3000 * (250 / (160 + dude_skill(SKILL_BARTER)))));
+        item_caps_adjust(self_obj, 3000 * (250 / (160 + dude_skill(SKILL_BARTER))));
     end
-    metarule(43, dude_obj);
+    inven_unwield(dude_obj);
     removed_qty := obj_is_carrying_obj_pid(dude_obj, 160);
     if (1 < removed_qty) then begin
         removed_qty := 1;
@@ -813,17 +813,17 @@ begin
         restock_obj := obj_carrying_pid_obj(dude_obj, 160);
         if (obj_type(self_obj) == 1) then begin
             restock_obj := obj_carrying_pid_obj(dude_obj, 160);
-            if (critter_inven_obj(self_obj, 0) == restock_obj) then begin
-                if (obj_item_subtype(critter_inven_obj(dude_obj, 0)) == 0) then begin
-                    restock_obj := critter_inven_obj(dude_obj, 0);
+            if (critter_inven_obj(self_obj, INVEN_TYPE_WORN) == restock_obj) then begin
+                if (obj_item_subtype(critter_inven_obj(dude_obj, INVEN_TYPE_WORN)) == 0) then begin
+                    restock_obj := critter_inven_obj(dude_obj, INVEN_TYPE_WORN);
                     debug_msg("armour pid == " + obj_pid(restock_obj));
                     rm_obj_from_inven(dude_obj, restock_obj);
                     add_obj_to_inven(dude_obj, restock_obj);
                 end
             end
             else begin
-                if ((critter_inven_obj(dude_obj, 2) == restock_obj) or (critter_inven_obj(dude_obj, 1) == restock_obj)) then begin
-                    metarule(43, dude_obj);
+                if ((critter_inven_obj(dude_obj, INVEN_TYPE_LEFT_HAND) == restock_obj) or (critter_inven_obj(dude_obj, INVEN_TYPE_RIGHT_HAND) == restock_obj)) then begin
+                    inven_unwield(dude_obj);
                 end
             end
         end
@@ -832,7 +832,7 @@ begin
     end
     restock_obj := 0;
     restock_amt := 0;
-    restock_obj := create_object_sid(399, 0, 0, -1);
+    restock_obj := create_object(399, 0, 0);
     add_mult_objs_to_inven(dude_obj, restock_obj, 1);
     call Node137();
 end
@@ -843,10 +843,10 @@ begin
         set_local_var(8, 2);
     end
     else begin
-        item_caps_adjust(dude_obj, -(12000 * (250 / (160 + has_skill(dude_obj, 15)))));
-        item_caps_adjust(self_obj, 12000 * (250 / (160 + has_skill(dude_obj, 15))));
+        item_caps_adjust(dude_obj, -(12000 * (250 / (160 + dude_skill(SKILL_BARTER)))));
+        item_caps_adjust(self_obj, 12000 * (250 / (160 + dude_skill(SKILL_BARTER))));
     end
-    metarule(43, dude_obj);
+    inven_unwield(dude_obj);
     removed_qty := obj_is_carrying_obj_pid(dude_obj, 351);
     if (1 < removed_qty) then begin
         removed_qty := 1;
@@ -855,17 +855,17 @@ begin
         restock_obj := obj_carrying_pid_obj(dude_obj, 351);
         if (obj_type(self_obj) == 1) then begin
             restock_obj := obj_carrying_pid_obj(dude_obj, 351);
-            if (critter_inven_obj(self_obj, 0) == restock_obj) then begin
-                if (obj_item_subtype(critter_inven_obj(dude_obj, 0)) == 0) then begin
-                    restock_obj := critter_inven_obj(dude_obj, 0);
+            if (critter_inven_obj(self_obj, INVEN_TYPE_WORN) == restock_obj) then begin
+                if (obj_item_subtype(critter_inven_obj(dude_obj, INVEN_TYPE_WORN)) == 0) then begin
+                    restock_obj := critter_inven_obj(dude_obj, INVEN_TYPE_WORN);
                     debug_msg("armour pid == " + obj_pid(restock_obj));
                     rm_obj_from_inven(dude_obj, restock_obj);
                     add_obj_to_inven(dude_obj, restock_obj);
                 end
             end
             else begin
-                if ((critter_inven_obj(dude_obj, 2) == restock_obj) or (critter_inven_obj(dude_obj, 1) == restock_obj)) then begin
-                    metarule(43, dude_obj);
+                if ((critter_inven_obj(dude_obj, INVEN_TYPE_LEFT_HAND) == restock_obj) or (critter_inven_obj(dude_obj, INVEN_TYPE_RIGHT_HAND) == restock_obj)) then begin
+                    inven_unwield(dude_obj);
                 end
             end
         end
@@ -874,7 +874,7 @@ begin
     end
     restock_obj := 0;
     restock_amt := 0;
-    restock_obj := create_object_sid(403, 0, 0, -1);
+    restock_obj := create_object(403, 0, 0);
     add_mult_objs_to_inven(dude_obj, restock_obj, 1);
     call Node137();
 end
@@ -885,10 +885,10 @@ begin
         set_local_var(8, 2);
     end
     else begin
-        item_caps_adjust(dude_obj, -(6000 * (250 / (160 + has_skill(dude_obj, 15)))));
-        item_caps_adjust(self_obj, 6000 * (250 / (160 + has_skill(dude_obj, 15))));
+        item_caps_adjust(dude_obj, -(6000 * (250 / (160 + dude_skill(SKILL_BARTER)))));
+        item_caps_adjust(self_obj, 6000 * (250 / (160 + dude_skill(SKILL_BARTER))));
     end
-    metarule(43, dude_obj);
+    inven_unwield(dude_obj);
     removed_qty := obj_is_carrying_obj_pid(dude_obj, 23);
     if (1 < removed_qty) then begin
         removed_qty := 1;
@@ -897,17 +897,17 @@ begin
         restock_obj := obj_carrying_pid_obj(dude_obj, 23);
         if (obj_type(self_obj) == 1) then begin
             restock_obj := obj_carrying_pid_obj(dude_obj, 23);
-            if (critter_inven_obj(self_obj, 0) == restock_obj) then begin
-                if (obj_item_subtype(critter_inven_obj(dude_obj, 0)) == 0) then begin
-                    restock_obj := critter_inven_obj(dude_obj, 0);
+            if (critter_inven_obj(self_obj, INVEN_TYPE_WORN) == restock_obj) then begin
+                if (obj_item_subtype(critter_inven_obj(dude_obj, INVEN_TYPE_WORN)) == 0) then begin
+                    restock_obj := critter_inven_obj(dude_obj, INVEN_TYPE_WORN);
                     debug_msg("armour pid == " + obj_pid(restock_obj));
                     rm_obj_from_inven(dude_obj, restock_obj);
                     add_obj_to_inven(dude_obj, restock_obj);
                 end
             end
             else begin
-                if ((critter_inven_obj(dude_obj, 2) == restock_obj) or (critter_inven_obj(dude_obj, 1) == restock_obj)) then begin
-                    metarule(43, dude_obj);
+                if ((critter_inven_obj(dude_obj, INVEN_TYPE_LEFT_HAND) == restock_obj) or (critter_inven_obj(dude_obj, INVEN_TYPE_RIGHT_HAND) == restock_obj)) then begin
+                    inven_unwield(dude_obj);
                 end
             end
         end
@@ -916,7 +916,7 @@ begin
     end
     restock_obj := 0;
     restock_amt := 0;
-    restock_obj := create_object_sid(405, 0, 0, -1);
+    restock_obj := create_object(405, 0, 0);
     add_mult_objs_to_inven(dude_obj, restock_obj, 1);
     call Node137();
 end
@@ -927,10 +927,10 @@ begin
         set_local_var(8, 2);
     end
     else begin
-        item_caps_adjust(dude_obj, -(4000 * (250 / (160 + has_skill(dude_obj, 15)))));
-        item_caps_adjust(self_obj, 4000 * (250 / (160 + has_skill(dude_obj, 15))));
+        item_caps_adjust(dude_obj, -(4000 * (250 / (160 + dude_skill(SKILL_BARTER)))));
+        item_caps_adjust(self_obj, 4000 * (250 / (160 + dude_skill(SKILL_BARTER))));
     end
-    metarule(43, dude_obj);
+    inven_unwield(dude_obj);
     removed_qty := obj_is_carrying_obj_pid(dude_obj, 10);
     if (1 < removed_qty) then begin
         removed_qty := 1;
@@ -939,17 +939,17 @@ begin
         restock_obj := obj_carrying_pid_obj(dude_obj, 10);
         if (obj_type(self_obj) == 1) then begin
             restock_obj := obj_carrying_pid_obj(dude_obj, 10);
-            if (critter_inven_obj(self_obj, 0) == restock_obj) then begin
-                if (obj_item_subtype(critter_inven_obj(dude_obj, 0)) == 0) then begin
-                    restock_obj := critter_inven_obj(dude_obj, 0);
+            if (critter_inven_obj(self_obj, INVEN_TYPE_WORN) == restock_obj) then begin
+                if (obj_item_subtype(critter_inven_obj(dude_obj, INVEN_TYPE_WORN)) == 0) then begin
+                    restock_obj := critter_inven_obj(dude_obj, INVEN_TYPE_WORN);
                     debug_msg("armour pid == " + obj_pid(restock_obj));
                     rm_obj_from_inven(dude_obj, restock_obj);
                     add_obj_to_inven(dude_obj, restock_obj);
                 end
             end
             else begin
-                if ((critter_inven_obj(dude_obj, 2) == restock_obj) or (critter_inven_obj(dude_obj, 1) == restock_obj)) then begin
-                    metarule(43, dude_obj);
+                if ((critter_inven_obj(dude_obj, INVEN_TYPE_LEFT_HAND) == restock_obj) or (critter_inven_obj(dude_obj, INVEN_TYPE_RIGHT_HAND) == restock_obj)) then begin
+                    inven_unwield(dude_obj);
                 end
             end
         end
@@ -958,7 +958,7 @@ begin
     end
     restock_obj := 0;
     restock_amt := 0;
-    restock_obj := create_object_sid(287, 0, 0, -1);
+    restock_obj := create_object(287, 0, 0);
     add_mult_objs_to_inven(dude_obj, restock_obj, 1);
     call Node137();
 end
@@ -969,10 +969,10 @@ begin
         set_local_var(8, 2);
     end
     else begin
-        item_caps_adjust(dude_obj, -(3000 * (250 / (160 + has_skill(dude_obj, 15)))));
-        item_caps_adjust(self_obj, 3000 * (250 / (160 + has_skill(dude_obj, 15))));
+        item_caps_adjust(dude_obj, -(3000 * (250 / (160 + dude_skill(SKILL_BARTER)))));
+        item_caps_adjust(self_obj, 3000 * (250 / (160 + dude_skill(SKILL_BARTER))));
     end
-    metarule(43, dude_obj);
+    inven_unwield(dude_obj);
     removed_qty := obj_is_carrying_obj_pid(dude_obj, 18);
     if (1 < removed_qty) then begin
         removed_qty := 1;
@@ -981,17 +981,17 @@ begin
         restock_obj := obj_carrying_pid_obj(dude_obj, 18);
         if (obj_type(self_obj) == 1) then begin
             restock_obj := obj_carrying_pid_obj(dude_obj, 18);
-            if (critter_inven_obj(self_obj, 0) == restock_obj) then begin
-                if (obj_item_subtype(critter_inven_obj(dude_obj, 0)) == 0) then begin
-                    restock_obj := critter_inven_obj(dude_obj, 0);
+            if (critter_inven_obj(self_obj, INVEN_TYPE_WORN) == restock_obj) then begin
+                if (obj_item_subtype(critter_inven_obj(dude_obj, INVEN_TYPE_WORN)) == 0) then begin
+                    restock_obj := critter_inven_obj(dude_obj, INVEN_TYPE_WORN);
                     debug_msg("armour pid == " + obj_pid(restock_obj));
                     rm_obj_from_inven(dude_obj, restock_obj);
                     add_obj_to_inven(dude_obj, restock_obj);
                 end
             end
             else begin
-                if ((critter_inven_obj(dude_obj, 2) == restock_obj) or (critter_inven_obj(dude_obj, 1) == restock_obj)) then begin
-                    metarule(43, dude_obj);
+                if ((critter_inven_obj(dude_obj, INVEN_TYPE_LEFT_HAND) == restock_obj) or (critter_inven_obj(dude_obj, INVEN_TYPE_RIGHT_HAND) == restock_obj)) then begin
+                    inven_unwield(dude_obj);
                 end
             end
         end
@@ -1000,7 +1000,7 @@ begin
     end
     restock_obj := 0;
     restock_amt := 0;
-    restock_obj := create_object_sid(404, 0, 0, -1);
+    restock_obj := create_object(404, 0, 0);
     add_mult_objs_to_inven(dude_obj, restock_obj, 1);
     call Node137();
 end
@@ -1011,10 +1011,10 @@ begin
         set_local_var(8, 2);
     end
     else begin
-        item_caps_adjust(dude_obj, -(2000 * (250 / (160 + has_skill(dude_obj, 15)))));
-        item_caps_adjust(self_obj, 2000 * (250 / (160 + has_skill(dude_obj, 15))));
+        item_caps_adjust(dude_obj, -(2000 * (250 / (160 + dude_skill(SKILL_BARTER)))));
+        item_caps_adjust(self_obj, 2000 * (250 / (160 + dude_skill(SKILL_BARTER))));
     end
-    metarule(43, dude_obj);
+    inven_unwield(dude_obj);
     removed_qty := obj_is_carrying_obj_pid(dude_obj, 313);
     if (1 < removed_qty) then begin
         removed_qty := 1;
@@ -1023,17 +1023,17 @@ begin
         restock_obj := obj_carrying_pid_obj(dude_obj, 313);
         if (obj_type(self_obj) == 1) then begin
             restock_obj := obj_carrying_pid_obj(dude_obj, 313);
-            if (critter_inven_obj(self_obj, 0) == restock_obj) then begin
-                if (obj_item_subtype(critter_inven_obj(dude_obj, 0)) == 0) then begin
-                    restock_obj := critter_inven_obj(dude_obj, 0);
+            if (critter_inven_obj(self_obj, INVEN_TYPE_WORN) == restock_obj) then begin
+                if (obj_item_subtype(critter_inven_obj(dude_obj, INVEN_TYPE_WORN)) == 0) then begin
+                    restock_obj := critter_inven_obj(dude_obj, INVEN_TYPE_WORN);
                     debug_msg("armour pid == " + obj_pid(restock_obj));
                     rm_obj_from_inven(dude_obj, restock_obj);
                     add_obj_to_inven(dude_obj, restock_obj);
                 end
             end
             else begin
-                if ((critter_inven_obj(dude_obj, 2) == restock_obj) or (critter_inven_obj(dude_obj, 1) == restock_obj)) then begin
-                    metarule(43, dude_obj);
+                if ((critter_inven_obj(dude_obj, INVEN_TYPE_LEFT_HAND) == restock_obj) or (critter_inven_obj(dude_obj, INVEN_TYPE_RIGHT_HAND) == restock_obj)) then begin
+                    inven_unwield(dude_obj);
                 end
             end
         end
@@ -1042,14 +1042,14 @@ begin
     end
     restock_obj := 0;
     restock_amt := 0;
-    restock_obj := create_object_sid(398, 0, 0, -1);
+    restock_obj := create_object(398, 0, 0);
     add_mult_objs_to_inven(dude_obj, restock_obj, 1);
     call Node137();
 end
 
 procedure Node995
 begin
-    if (get_critter_stat(dude_obj, 3) >= 7) then begin
+    if (dude_charisma >= 7) then begin
         set_local_var(3, 100);
         set_local_var(1, 4);
     end
@@ -1067,7 +1067,7 @@ end
 
 procedure Node997
 begin
-    if (get_critter_stat(dude_obj, 3) >= 8) then begin
+    if (dude_charisma >= 8) then begin
         call Node004();
     end
     else begin
@@ -1086,35 +1086,35 @@ end
 
 procedure Node001
 begin
-    float_msg(self_obj, message_str(136, random(110, 114)), 2);
+    float_msg(self_obj, mstr(random(110, 114)), FLOAT_MSG_RED);
 end
 
 procedure Node002
 begin
-    gsay_reply(136, 120);
-    giq_option(4, 136, 121, Node003, 50);
-    giq_option(4, 136, 122, Node004, 50);
+    Reply(120);
+    giq_option(4, NAME, 121, Node003, NEUTRAL_REACTION);
+    giq_option(4, NAME, 122, Node004, NEUTRAL_REACTION);
 end
 
 procedure Node003
 begin
-    gsay_reply(136, 130);
-    giq_option(4, 136, 131, Node004, 50);
+    Reply(130);
+    giq_option(4, NAME, 131, Node004, NEUTRAL_REACTION);
 end
 
 procedure Node004
 begin
-    gsay_reply(136, 140);
-    giq_option(4, 136, 141, Node005, 50);
-    giq_option(4, 136, 142, Node005, 50);
-    giq_option(4, 136, 143, Node999, 50);
-    giq_option(4, 136, 144, Node009, 50);
+    Reply(140);
+    giq_option(4, NAME, 141, Node005, NEUTRAL_REACTION);
+    giq_option(4, NAME, 142, Node005, NEUTRAL_REACTION);
+    giq_option(4, NAME, 143, Node999, NEUTRAL_REACTION);
+    giq_option(4, NAME, 144, Node009, NEUTRAL_REACTION);
 end
 
 procedure Node005
 begin
-    gsay_reply(136, 150);
-    giq_option(4, 136, 151, Node021, 50);
+    Reply(150);
+    giq_option(4, NAME, 151, Node021, NEUTRAL_REACTION);
 end
 
 procedure Node006
@@ -1123,274 +1123,274 @@ begin
         item := obj_carrying_pid_obj(dude_obj, 307);
         rm_obj_from_inven(dude_obj, item);
         destroy_object(item);
-        set_global_var(160, 2);
-        set_global_var(51, global_var(51) + 5);
-        display_msg(message_str(136, 700));
+        set_global_var(GVAR_GECKO_SKEETER_PART, 2);
+        set_global_var(GVAR_TOWN_REP_GECKO, global_var(GVAR_TOWN_REP_GECKO) + 5);
+        display_mstr(700);
         give_exp_points(350);
-        display_msg(message_str(14, 100) + 350 + message_str(14, 101));
+        display_msg(g_mstr(100) + 350 + g_mstr(101));
         set_local_var(8, 1);
     end
-    gsay_reply(136, 160);
-    giq_option(1, 136, 161, Node110, 50);
-    giq_option(4, 136, 162, Node999, 50);
+    Reply(160);
+    giq_option(1, NAME, 161, Node110, NEUTRAL_REACTION);
+    giq_option(4, NAME, 162, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node007
 begin
-    gsay_reply(136, 170);
-    giq_option(4, 136, 171, Node009, 50);
-    giq_option(4, 136, 172, Node999, 50);
+    Reply(170);
+    giq_option(4, NAME, 171, Node009, NEUTRAL_REACTION);
+    giq_option(4, NAME, 172, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node008
 begin
-    gsay_reply(136, 180);
-    giq_option(4, 136, 181, Node009, 50);
-    giq_option(4, 136, 182, Node999, 50);
+    Reply(180);
+    giq_option(4, NAME, 181, Node009, NEUTRAL_REACTION);
+    giq_option(4, NAME, 182, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node009
 begin
     if (local_var(9) == 0) then begin
         set_local_var(9, 1);
-        gsay_reply(136, 190);
+        Reply(190);
     end
     else begin
-        gsay_reply(136, 192);
+        Reply(192);
     end
-    if (global_var(233)) then begin
+    if (global_var(GVAR_VAULT13_FOUND_GECK)) then begin
     end
     else begin
-        giq_option(4, 136, 200, Node016, 50);
+        giq_option(4, NAME, 200, Node016, NEUTRAL_REACTION);
     end
-    if ((global_var(393) == 0) and (global_var(162) > 0)) then begin
-        giq_option(4, 136, 201, Node002, 50);
+    if ((global_var(GVAR_QUEST_SUPER_REPAIR_KIT) == 0) and (global_var(GVAR_DEN_SMITTY_PART) > 0)) then begin
+        giq_option(4, NAME, 201, Node002, NEUTRAL_REACTION);
     end
-    giq_option(4, 136, 202, Node110, 50);
-    giq_option(4, 136, 203, Node015, 50);
-    giq_option(4, 136, 204, Node999, 50);
+    giq_option(4, NAME, 202, Node110, NEUTRAL_REACTION);
+    giq_option(4, NAME, 203, Node015, NEUTRAL_REACTION);
+    giq_option(4, NAME, 204, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node010
 begin
-    gsay_reply(136, 210);
-    giq_option(1, 136, 211, Node920, 50);
-    giq_option(1, 136, 212, Node922, 50);
-    giq_option(1, 136, 213, Node924, 50);
-    giq_option(1, 136, 214, Node926, 50);
-    giq_option(1, 136, 215, Node928, 50);
-    giq_option(1, 136, 216, Node930, 50);
-    giq_option(1, 136, 217, Node035, 50);
-    giq_option(1, 136, 218, Node999, 50);
+    Reply(210);
+    giq_option(1, NAME, 211, Node920, NEUTRAL_REACTION);
+    giq_option(1, NAME, 212, Node922, NEUTRAL_REACTION);
+    giq_option(1, NAME, 213, Node924, NEUTRAL_REACTION);
+    giq_option(1, NAME, 214, Node926, NEUTRAL_REACTION);
+    giq_option(1, NAME, 215, Node928, NEUTRAL_REACTION);
+    giq_option(1, NAME, 216, Node930, NEUTRAL_REACTION);
+    giq_option(1, NAME, 217, Node035, NEUTRAL_REACTION);
+    giq_option(1, NAME, 218, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node011
 begin
-    gsay_reply(136, 220);
-    giq_option(1, 136, 222, Node034, 50);
-    giq_option(1, 136, 223, Node034, 50);
-    giq_option(1, 136, 224, Node034, 50);
-    giq_option(1, 136, 225, Node110, 50);
-    giq_option(1, 136, 226, Node999, 50);
+    Reply(220);
+    giq_option(1, NAME, 222, Node034, NEUTRAL_REACTION);
+    giq_option(1, NAME, 223, Node034, NEUTRAL_REACTION);
+    giq_option(1, NAME, 224, Node034, NEUTRAL_REACTION);
+    giq_option(1, NAME, 225, Node110, NEUTRAL_REACTION);
+    giq_option(1, NAME, 226, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node012
 begin
-    gsay_reply(136, 230);
-    giq_option(1, 136, 231, Node110, 50);
-    giq_option(1, 136, 232, Node019, 50);
-    giq_option(1, 136, 233, Node999, 50);
+    Reply(230);
+    giq_option(1, NAME, 231, Node110, NEUTRAL_REACTION);
+    giq_option(1, NAME, 232, Node019, NEUTRAL_REACTION);
+    giq_option(1, NAME, 233, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node013
 begin
-    gsay_reply(136, 240);
-    giq_option(1, 136, 241, Node110, 50);
-    giq_option(1, 136, 242, Node999, 50);
+    Reply(240);
+    giq_option(1, NAME, 241, Node110, NEUTRAL_REACTION);
+    giq_option(1, NAME, 242, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node014
 begin
-    gsay_reply(136, 250);
-    giq_option(1, 136, 251, Node110, 50);
-    giq_option(1, 136, 252, Node999, 50);
+    Reply(250);
+    giq_option(1, NAME, 251, Node110, NEUTRAL_REACTION);
+    giq_option(1, NAME, 252, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node015
 begin
-    gsay_reply(136, 260);
-    giq_option(4, 136, 261, Node009, 50);
-    giq_option(4, 136, 262, Node999, 50);
+    Reply(260);
+    giq_option(4, NAME, 261, Node009, NEUTRAL_REACTION);
+    giq_option(4, NAME, 262, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node016
 begin
-    gsay_reply(136, 270);
-    giq_option(4, 136, 271, Node017, 50);
-    giq_option(4, 136, 272, Node009, 50);
-    giq_option(4, 136, 273, Node999, 50);
+    Reply(270);
+    giq_option(4, NAME, 271, Node017, NEUTRAL_REACTION);
+    giq_option(4, NAME, 272, Node009, NEUTRAL_REACTION);
+    giq_option(4, NAME, 273, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node017
 begin
-    gsay_reply(136, 280);
-    giq_option(4, 136, 281, Node018, 50);
+    Reply(280);
+    giq_option(4, NAME, 281, Node018, NEUTRAL_REACTION);
 end
 
 procedure Node018
 begin
-    gsay_reply(136, 290);
-    giq_option(4, 136, 291, Node009, 50);
-    giq_option(4, 136, 292, Node999, 50);
+    Reply(290);
+    giq_option(4, NAME, 291, Node009, NEUTRAL_REACTION);
+    giq_option(4, NAME, 292, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node019
 begin
-    gsay_reply(136, 300);
-    giq_option(1, 136, 301, Node020, 50);
-    giq_option(1, 136, 302, Node110, 50);
-    giq_option(1, 136, 303, Node999, 49);
+    Reply(300);
+    giq_option(1, NAME, 301, Node020, NEUTRAL_REACTION);
+    giq_option(1, NAME, 302, Node110, NEUTRAL_REACTION);
+    giq_option(1, NAME, 303, Node999, GOOD_REACTION);
 end
 
 procedure Node020
 begin
-    gsay_reply(136, 310);
-    giq_option(1, 136, 311, Node999, 50);
-    giq_option(1, 136, 312, Node110, 50);
-    giq_option(1, 136, 313, Node999, 49);
+    Reply(310);
+    giq_option(1, NAME, 311, Node999, NEUTRAL_REACTION);
+    giq_option(1, NAME, 312, Node110, NEUTRAL_REACTION);
+    giq_option(1, NAME, 313, Node999, GOOD_REACTION);
 end
 
 procedure Node021
 begin
-    gsay_reply(136, 320);
-    giq_option(4, 136, 321, Node022, 50);
+    Reply(320);
+    giq_option(4, NAME, 321, Node022, NEUTRAL_REACTION);
 end
 
 procedure Node022
 begin
-    set_global_var(393, 1);
+    set_global_var(GVAR_QUEST_SUPER_REPAIR_KIT, 1);
     set_local_var(7, 1);
-    gsay_reply(136, 330);
-    giq_option(4, 136, 331, Node023, 50);
+    Reply(330);
+    giq_option(4, NAME, 331, Node023, NEUTRAL_REACTION);
 end
 
 procedure Node023
 begin
-    gsay_reply(136, 340);
-    giq_option(4, 136, 341, Node009, 50);
-    giq_option(4, 136, 342, Node024, 50);
-    giq_option(4, 136, 343, Node999, 50);
+    Reply(340);
+    giq_option(4, NAME, 341, Node009, NEUTRAL_REACTION);
+    giq_option(4, NAME, 342, Node024, NEUTRAL_REACTION);
+    giq_option(4, NAME, 343, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node024
 begin
-    gsay_reply(136, 360);
-    giq_option(4, 136, 361, Node009, 50);
-    giq_option(4, 136, 362, combat, 51);
-    giq_option(7, 136, 363, Node999, 50);
+    Reply(360);
+    giq_option(4, NAME, 361, Node009, NEUTRAL_REACTION);
+    giq_option(4, NAME, 362, combat, BAD_REACTION);
+    giq_option(7, NAME, 363, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node025
 begin
     if (local_var(7) == 1) then begin
-        gsay_reply(136, 371);
+        Reply(371);
     end
     else begin
-        gsay_reply(136, 370);
+        Reply(370);
     end
-    giq_option(4, 136, 372, Node915, 50);
-    giq_option(4, 136, 373, Node999, 50);
+    giq_option(4, NAME, 372, Node915, NEUTRAL_REACTION);
+    giq_option(4, NAME, 373, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node026
 begin
     call Node150();
     if (combyn == 0) then begin
-        gsay_reply(136, 380);
+        Reply(380);
     end
     else begin
-        gsay_reply(136, 381);
-        giq_option(4, 136, 382, Node027, 50);
+        Reply(381);
+        giq_option(4, NAME, 382, Node027, NEUTRAL_REACTION);
     end
-    giq_option(4, 136, 383, Node009, 50);
-    giq_option(4, 136, 384, Node999, 50);
+    giq_option(4, NAME, 383, Node009, NEUTRAL_REACTION);
+    giq_option(4, NAME, 384, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node027
 begin
-    gsay_reply(136, 390);
-    giq_option(4, 136, 391, Node028, 50);
+    Reply(390);
+    giq_option(4, NAME, 391, Node028, NEUTRAL_REACTION);
 end
 
 procedure Node028
 begin
-    gsay_reply(136, 400);
-    giq_option(4, 136, 401, Node029, 50);
+    Reply(400);
+    giq_option(4, NAME, 401, Node029, NEUTRAL_REACTION);
 end
 
 procedure Node029
 begin
-    gsay_reply(136, 410);
-    giq_option(4, 136, 411, Node030, 50);
+    Reply(410);
+    giq_option(4, NAME, 411, Node030, NEUTRAL_REACTION);
 end
 
 procedure Node030
 begin
-    gsay_reply(136, 420);
-    giq_option(4, 136, 421, Node031, 50);
+    Reply(420);
+    giq_option(4, NAME, 421, Node031, NEUTRAL_REACTION);
 end
 
 procedure Node031
 begin
-    set_global_var(160, 1);
-    gsay_reply(136, 430);
-    giq_option(4, 136, 431, Node999, 50);
-    giq_option(4, 136, 432, Node999, 50);
+    set_global_var(GVAR_GECKO_SKEETER_PART, 1);
+    Reply(430);
+    giq_option(4, NAME, 431, Node999, NEUTRAL_REACTION);
+    giq_option(4, NAME, 432, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node032
 begin
-    gsay_reply(136, 440);
-    giq_option(4, 136, 441, Node110, 50);
-    giq_option(4, 136, 442, Node999, 50);
+    Reply(440);
+    giq_option(4, NAME, 441, Node110, NEUTRAL_REACTION);
+    giq_option(4, NAME, 442, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node034
 begin
-    gsay_reply(136, 460);
-    giq_option(1, 136, 461, Node940, 50);
-    giq_option(1, 136, 462, Node110, 50);
-    giq_option(1, 136, 463, Node999, 50);
+    Reply(460);
+    giq_option(1, NAME, 461, Node940, NEUTRAL_REACTION);
+    giq_option(1, NAME, 462, Node110, NEUTRAL_REACTION);
+    giq_option(1, NAME, 463, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node035
 begin
-    gsay_reply(136, 470);
-    giq_option(1, 136, 471, Node110, 50);
-    giq_option(1, 136, 472, Node999, 50);
+    Reply(470);
+    giq_option(1, NAME, 471, Node110, NEUTRAL_REACTION);
+    giq_option(1, NAME, 472, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node036
 begin
-    gsay_reply(136, 500);
-    giq_option(1, 136, 501, Node999, 50);
-    giq_option(1, 136, 502, Node999, 50);
+    Reply(500);
+    giq_option(1, NAME, 501, Node999, NEUTRAL_REACTION);
+    giq_option(1, NAME, 502, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node110
 begin
-    gsay_reply(136, 600);
+    Reply(600);
     if (obj_is_carrying_obj_pid(dude_obj, 18) or obj_is_carrying_obj_pid(dude_obj, 313)) then begin
-        giq_option(4, 136, 601, Node111, 50);
+        giq_option(4, NAME, 601, Node111, NEUTRAL_REACTION);
     end
     if (obj_is_carrying_obj_pid(dude_obj, 10) or obj_is_carrying_obj_pid(dude_obj, 23) or obj_is_carrying_obj_pid(dude_obj, 351)) then begin
-        giq_option(4, 136, 602, Node112, 50);
+        giq_option(4, NAME, 602, Node112, NEUTRAL_REACTION);
     end
     if (obj_is_carrying_obj_pid(dude_obj, 160)) then begin
-        giq_option(4, 136, 603, Node113, 50);
+        giq_option(4, NAME, 603, Node113, NEUTRAL_REACTION);
     end
-    giq_option(4, 136, 604, Node009, 50);
-    giq_option(4, 136, 605, Node999, 50);
+    giq_option(4, NAME, 604, Node009, NEUTRAL_REACTION);
+    giq_option(4, NAME, 605, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node111
@@ -1399,44 +1399,44 @@ begin
     variable LVar1 := 0;
     if (local_var(8) == 1) then begin
         if (obj_is_carrying_obj_pid(dude_obj, 313)) then begin
-            LVar0 := LVar0 + (message_str(136, 680) + " ");
+            LVar0 := LVar0 + (mstr(680) + " ");
         end
         if (obj_is_carrying_obj_pid(dude_obj, 18)) then begin
-            LVar0 := LVar0 + message_str(136, 681);
+            LVar0 := LVar0 + mstr(681);
         end
-        gsay_reply(136, LVar0);
+        Reply(LVar0);
         if (obj_is_carrying_obj_pid(dude_obj, 313)) then begin
-            giq_option(4, 136, 612, Node987, 50);
+            giq_option(4, NAME, 612, Node987, NEUTRAL_REACTION);
             LVar1 := 1;
         end
         if (obj_is_carrying_obj_pid(dude_obj, 18)) then begin
-            giq_option(4, 136, 613, Node986, 50);
+            giq_option(4, NAME, 613, Node986, NEUTRAL_REACTION);
             LVar1 := 1;
         end
     end
     else begin
         if (obj_is_carrying_obj_pid(dude_obj, 313)) then begin
-            LVar0 := LVar0 + (message_str(136, 610) + (2000 * (250 / (160 + has_skill(dude_obj, 15)))) + message_str(136, 5500) + " ");
+            LVar0 := LVar0 + (mstr(610) + (2000 * (250 / (160 + dude_skill(SKILL_BARTER)))) + mstr(5500) + " ");
         end
         if (obj_is_carrying_obj_pid(dude_obj, 18)) then begin
-            LVar0 := LVar0 + (message_str(136, 611) + (3000 * (250 / (160 + has_skill(dude_obj, 15)))) + message_str(136, 5500));
+            LVar0 := LVar0 + (mstr(611) + (3000 * (250 / (160 + dude_skill(SKILL_BARTER)))) + mstr(5500));
         end
-        gsay_reply(136, LVar0);
-        if (obj_is_carrying_obj_pid(dude_obj, 313) and (item_caps_total(dude_obj) >= (2000 * (250 / (160 + has_skill(dude_obj, 15)))))) then begin
-            giq_option(4, 136, 612, Node987, 50);
+        Reply(LVar0);
+        if (obj_is_carrying_obj_pid(dude_obj, 313) and (item_caps_total(dude_obj) >= (2000 * (250 / (160 + dude_skill(SKILL_BARTER)))))) then begin
+            giq_option(4, NAME, 612, Node987, NEUTRAL_REACTION);
             LVar1 := 1;
         end
-        if (obj_is_carrying_obj_pid(dude_obj, 18) and (item_caps_total(dude_obj) >= (3000 * (250 / (160 + has_skill(dude_obj, 15)))))) then begin
-            giq_option(4, 136, 613, Node986, 50);
+        if (obj_is_carrying_obj_pid(dude_obj, 18) and (item_caps_total(dude_obj) >= (3000 * (250 / (160 + dude_skill(SKILL_BARTER)))))) then begin
+            giq_option(4, NAME, 613, Node986, NEUTRAL_REACTION);
             LVar1 := 1;
         end
         if (LVar1 == 0) then begin
-            giq_option(4, 136, 617, Node999, 50);
+            giq_option(4, NAME, 617, Node999, NEUTRAL_REACTION);
         end
     end
-    giq_option(4, 136, 614, Node110, 50);
-    giq_option(4, 136, 615, Node009, 50);
-    giq_option(4, 136, 616, Node999, 50);
+    giq_option(4, NAME, 614, Node110, NEUTRAL_REACTION);
+    giq_option(4, NAME, 615, Node009, NEUTRAL_REACTION);
+    giq_option(4, NAME, 616, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node112
@@ -1444,104 +1444,104 @@ begin
     variable LVar0 := 0;
     variable LVar1 := 0;
     if (local_var(8) == 1) then begin
-        LVar0 := message_str(136, 620);
+        LVar0 := mstr(620);
         if (obj_is_carrying_obj_pid(dude_obj, 10)) then begin
-            LVar0 := LVar0 + (message_str(136, 682) + " ");
+            LVar0 := LVar0 + (mstr(682) + " ");
         end
         if (obj_is_carrying_obj_pid(dude_obj, 23)) then begin
-            LVar0 := LVar0 + (message_str(136, 683) + " ");
+            LVar0 := LVar0 + (mstr(683) + " ");
         end
         if (obj_is_carrying_obj_pid(dude_obj, 351)) then begin
-            LVar0 := LVar0 + message_str(136, 684);
+            LVar0 := LVar0 + mstr(684);
         end
-        gsay_reply(136, LVar0);
+        Reply(LVar0);
         if (obj_is_carrying_obj_pid(dude_obj, 10)) then begin
-            giq_option(4, 136, 624, Node985, 50);
+            giq_option(4, NAME, 624, Node985, NEUTRAL_REACTION);
         end
         if (obj_is_carrying_obj_pid(dude_obj, 23)) then begin
-            giq_option(4, 136, 625, Node984, 50);
+            giq_option(4, NAME, 625, Node984, NEUTRAL_REACTION);
         end
         if (obj_is_carrying_obj_pid(dude_obj, 351)) then begin
-            giq_option(4, 136, 626, Node983, 50);
+            giq_option(4, NAME, 626, Node983, NEUTRAL_REACTION);
         end
     end
     else begin
-        LVar0 := message_str(136, 620);
+        LVar0 := mstr(620);
         if (obj_is_carrying_obj_pid(dude_obj, 10)) then begin
-            LVar0 := LVar0 + (message_str(136, 621) + (4000 * (250 / (160 + has_skill(dude_obj, 15)))) + message_str(136, 5500) + " ");
+            LVar0 := LVar0 + (mstr(621) + (4000 * (250 / (160 + dude_skill(SKILL_BARTER)))) + mstr(5500) + " ");
         end
         if (obj_is_carrying_obj_pid(dude_obj, 23)) then begin
-            LVar0 := LVar0 + (message_str(136, 622) + (6000 * (250 / (160 + has_skill(dude_obj, 15)))) + message_str(136, 5500) + " ");
+            LVar0 := LVar0 + (mstr(622) + (6000 * (250 / (160 + dude_skill(SKILL_BARTER)))) + mstr(5500) + " ");
         end
         if (obj_is_carrying_obj_pid(dude_obj, 351)) then begin
-            LVar0 := LVar0 + (message_str(136, 623) + (12000 * (250 / (160 + has_skill(dude_obj, 15)))) + message_str(136, 5500));
+            LVar0 := LVar0 + (mstr(623) + (12000 * (250 / (160 + dude_skill(SKILL_BARTER)))) + mstr(5500));
         end
-        gsay_reply(136, LVar0);
-        if (obj_is_carrying_obj_pid(dude_obj, 10) and (item_caps_total(dude_obj) >= (4000 * (250 / (160 + has_skill(dude_obj, 15)))))) then begin
-            giq_option(4, 136, 624, Node985, 50);
+        Reply(LVar0);
+        if (obj_is_carrying_obj_pid(dude_obj, 10) and (item_caps_total(dude_obj) >= (4000 * (250 / (160 + dude_skill(SKILL_BARTER)))))) then begin
+            giq_option(4, NAME, 624, Node985, NEUTRAL_REACTION);
             LVar1 := 1;
         end
-        if (obj_is_carrying_obj_pid(dude_obj, 23) and (item_caps_total(dude_obj) >= (6000 * (250 / (160 + has_skill(dude_obj, 15)))))) then begin
-            giq_option(4, 136, 625, Node984, 50);
+        if (obj_is_carrying_obj_pid(dude_obj, 23) and (item_caps_total(dude_obj) >= (6000 * (250 / (160 + dude_skill(SKILL_BARTER)))))) then begin
+            giq_option(4, NAME, 625, Node984, NEUTRAL_REACTION);
             LVar1 := 1;
         end
-        if (obj_is_carrying_obj_pid(dude_obj, 351) and (item_caps_total(dude_obj) >= (12000 * (250 / (160 + has_skill(dude_obj, 15)))))) then begin
-            giq_option(4, 136, 626, Node983, 50);
+        if (obj_is_carrying_obj_pid(dude_obj, 351) and (item_caps_total(dude_obj) >= (12000 * (250 / (160 + dude_skill(SKILL_BARTER)))))) then begin
+            giq_option(4, NAME, 626, Node983, NEUTRAL_REACTION);
             LVar1 := 1;
         end
         if (LVar1 == 0) then begin
-            giq_option(4, 136, 617, Node999, 50);
+            giq_option(4, NAME, 617, Node999, NEUTRAL_REACTION);
         end
     end
-    giq_option(4, 136, 614, Node110, 50);
-    giq_option(4, 136, 615, Node009, 50);
-    giq_option(4, 136, 616, Node999, 50);
+    giq_option(4, NAME, 614, Node110, NEUTRAL_REACTION);
+    giq_option(4, NAME, 615, Node009, NEUTRAL_REACTION);
+    giq_option(4, NAME, 616, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node113
 begin
     if (local_var(8) == 1) then begin
-        gsay_reply(136, 685);
+        Reply(685);
     end
     else begin
-        gsay_reply(136, message_str(136, 630) + (3000 * (250 / (160 + has_skill(dude_obj, 15)))) + message_str(136, 5500));
+        Reply(mstr(630) + (3000 * (250 / (160 + dude_skill(SKILL_BARTER)))) + mstr(5500));
     end
-    if ((item_caps_total(dude_obj) >= (3000 * (250 / (160 + has_skill(dude_obj, 15))))) or (local_var(8) == 1)) then begin
-        giq_option(4, 136, 631, Node982, 50);
+    if ((item_caps_total(dude_obj) >= (3000 * (250 / (160 + dude_skill(SKILL_BARTER))))) or (local_var(8) == 1)) then begin
+        giq_option(4, NAME, 631, Node982, NEUTRAL_REACTION);
     end
     else begin
-        giq_option(4, 136, 617, Node999, 50);
+        giq_option(4, NAME, 617, Node999, NEUTRAL_REACTION);
     end
-    giq_option(4, 136, 614, Node110, 50);
-    giq_option(4, 136, 615, Node009, 50);
-    giq_option(4, 136, 616, Node999, 50);
+    giq_option(4, NAME, 614, Node110, NEUTRAL_REACTION);
+    giq_option(4, NAME, 615, Node009, NEUTRAL_REACTION);
+    giq_option(4, NAME, 616, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node137
 begin
     game_time_advance(game_ticks(3 * 3600));
-    gsay_reply(136, 640);
-    giq_option(4, 136, 641, Node138, 50);
+    Reply(640);
+    giq_option(4, NAME, 641, Node138, NEUTRAL_REACTION);
 end
 
 procedure Node138
 begin
-    gsay_reply(136, 650);
-    giq_option(4, 136, 651, Node139, 50);
+    Reply(650);
+    giq_option(4, NAME, 651, Node139, NEUTRAL_REACTION);
 end
 
 procedure Node139
 begin
-    gsay_reply(136, 660);
-    giq_option(4, 136, 661, Node140, 50);
+    Reply(660);
+    giq_option(4, NAME, 661, Node140, NEUTRAL_REACTION);
 end
 
 procedure Node140
 begin
-    gsay_reply(136, 670);
-    giq_option(4, 136, 671, Node110, 50);
-    giq_option(4, 136, 672, Node009, 50);
-    giq_option(4, 136, 673, Node999, 50);
+    Reply(670);
+    giq_option(4, NAME, 671, Node110, NEUTRAL_REACTION);
+    giq_option(4, NAME, 672, Node009, NEUTRAL_REACTION);
+    giq_option(4, NAME, 673, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node150
@@ -1574,21 +1574,20 @@ begin
         item := obj_carrying_pid_obj(dude_obj, 308);
         rm_obj_from_inven(dude_obj, item);
         add_obj_to_inven(self_obj, item);
-        item := create_object_sid(253, 0, 0, -1);
+        item := create_object(253, 0, 0);
         add_obj_to_inven(dude_obj, item);
-        set_global_var(393, 2);
-        set_global_var(51, global_var(51) + 5);
-        display_msg(message_str(136, 710));
+        set_global_var(GVAR_QUEST_SUPER_REPAIR_KIT, 2);
+        set_global_var(GVAR_TOWN_REP_GECKO, global_var(GVAR_TOWN_REP_GECKO) + 5);
+        display_mstr(710);
         give_exp_points(250);
-        display_msg(message_str(14, 100) + 250 + message_str(14, 101));
+        display_msg(g_mstr(100) + 250 + g_mstr(101));
     end
     call Node026();
 end
 
 procedure Node033
 begin
-    gsay_reply(136, 450);
-    giq_option(1, 136, 451, Node110, 50);
-    giq_option(1, 136, 452, Node999, 50);
+    Reply(450);
+    giq_option(1, NAME, 451, Node110, NEUTRAL_REACTION);
+    giq_option(1, NAME, 452, Node999, NEUTRAL_REACTION);
 end
-
