diff --git a/Games/Fallout2.Steam/Original/GCWANDER.ssl b/Games/Fallout2.Steam/ReDefine/GCWANDER.ssl
index a843b5d..f74ed6b 100644
--- a/Games/Fallout2.Steam/Original/GCWANDER.ssl
+++ b/Games/Fallout2.Steam/ReDefine/GCWANDER.ssl
@@ -172,14 +172,14 @@ end
 
 procedure critter_p_proc
 begin
-    if ((global_var(395) == 1) and obj_can_see_obj(self_obj, dude_obj)) then begin
+    if ((global_var(GVAR_GECKO_MELTDOWN) == 1) and obj_can_see_obj(self_obj, dude_obj)) then begin
         set_local_var(4, 1);
-        attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+        attack(dude_obj);
     end
     else begin
         if ((local_var(4) == 2) or (local_var(5) == 1) and obj_can_see_obj(self_obj, dude_obj)) then begin
             set_local_var(4, 1);
-            attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+            attack(dude_obj);
         end
         else begin
             if ((random(0, 100) == 1) and (In_Timed_Event == 0)) then begin
@@ -199,16 +199,16 @@ end
 
 procedure talk_p_proc
 begin
-    if ((global_var(238) == 1) or (local_var(5) == 1)) then begin
+    if ((global_var(GVAR_ENEMY_GECKO) == 1) or (local_var(5) == 1)) then begin
         call Node998();
     end
     else begin
-        if (global_var(83) == 8) then begin
+        if (global_var(GVAR_VAULT_PLANT_STATUS) == 8) then begin
             call Node998();
         end
         else begin
-            if (cur_map_index == 31) then begin
-                if ((random(1, 10) == 1) and (global_var(161) == 1)) then begin
+            if (CUR_MAP_GECKSETL) then begin
+                if ((random(1, 10) == 1) and (global_var(GVAR_GECKO_ANKH) == 1)) then begin
                     call Node011();
                 end
                 else begin
@@ -228,15 +228,15 @@ begin
                                     call gobar();
                                 end
                                 else begin
-                                    if (global_var(83) == 15) then begin
+                                    if (global_var(GVAR_VAULT_PLANT_STATUS) == 15) then begin
                                         call Node010();
                                     end
                                     else begin
-                                        if (global_var(83) == 9) then begin
+                                        if (global_var(GVAR_VAULT_PLANT_STATUS) == 9) then begin
                                             call Node001();
                                         end
                                         else begin
-                                            if (global_var(51) > 9) then begin
+                                            if (global_var(GVAR_TOWN_REP_GECKO) > 9) then begin
                                                 call Node002();
                                             end
                                             else begin
@@ -251,8 +251,8 @@ begin
                 end
             end
             else begin
-                if (cur_map_index == 33) then begin
-                    if ((random(1, 10) == 1) and (global_var(161) == 1)) then begin
+                if (CUR_MAP_GECKJUNK) then begin
+                    if ((random(1, 10) == 1) and (global_var(GVAR_GECKO_ANKH) == 1)) then begin
                         call Node011();
                     end
                     else begin
@@ -268,15 +268,15 @@ begin
                                     call Node005();
                                 end
                                 else begin
-                                    if (global_var(83) == 15) then begin
+                                    if (global_var(GVAR_VAULT_PLANT_STATUS) == 15) then begin
                                         call Node010();
                                     end
                                     else begin
-                                        if (global_var(83) == 9) then begin
+                                        if (global_var(GVAR_VAULT_PLANT_STATUS) == 9) then begin
                                             call Node001();
                                         end
                                         else begin
-                                            if (global_var(51) > 9) then begin
+                                            if (global_var(GVAR_TOWN_REP_GECKO) > 9) then begin
                                                 call Node002();
                                             end
                                             else begin
@@ -297,59 +297,59 @@ end
 procedure destroy_p_proc
 begin
     if (source_obj == dude_obj) then begin
-        set_global_var(5, global_var(5) + 1);
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(4) > (3 * global_var(5))) or (global_var(2) == 1)) and (global_var(1) == 0)) then begin
-            set_global_var(2, 1);
-            set_global_var(3, 0);
+        set_global_var(GVAR_GOOD_MONSTER, global_var(GVAR_GOOD_MONSTER) + 1);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(4) > (3 * global_var(GVAR_GOOD_MONSTER))) or (global_var(GVAR_CHAMPION_REPUTATION) == 1)) and (global_var(GVAR_CHILDKILLER_REPUTATION) == 0)) then begin
+            set_global_var(GVAR_CHAMPION_REPUTATION, 1);
+            set_global_var(GVAR_BERSERKER_REPUTATION, 0);
         end
-        if (metarule(51, self_obj) == 2) then begin
+        if (critter_kill_type(self_obj) == KILL_TYPE_children_kills) then begin
             if (source_obj == dude_obj) then begin
-                set_global_var(2, 0);
-                set_global_var(1, global_var(1) + 1);
-                set_global_var(0, global_var(0) + -15);
+                set_global_var(GVAR_CHAMPION_REPUTATION, 0);
+                set_global_var(GVAR_CHILDKILLER_REPUTATION, global_var(GVAR_CHILDKILLER_REPUTATION) + 1);
+                set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + -15);
                 debug_msg("Player gains " + -15 + " Karma Points.");
-                set_global_var(37, 0);
-                set_global_var(38, 0);
-                set_global_var(39, 0);
-                set_global_var(40, 0);
-                set_global_var(41, 0);
-                set_global_var(42, 0);
-                set_global_var(43, 0);
-                set_global_var(44, 0);
-                set_global_var(45, 0);
-                if (global_var(0) >= 1000) then begin
-                    set_global_var(37, 1);
+                set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+                set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+                set_global_var(GVAR_KARMA_DEFENDER, 0);
+                set_global_var(GVAR_KARMA_WANDERER, 0);
+                set_global_var(GVAR_KARMA_BETRAYER, 0);
+                set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+                set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+                set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+                    set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
                 end
                 else begin
-                    if (global_var(0) >= 750) then begin
-                        set_global_var(38, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
                     end
                     else begin
-                        if (global_var(0) >= 500) then begin
-                            set_global_var(39, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                            set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                         end
                         else begin
-                            if (global_var(0) >= 250) then begin
-                                set_global_var(40, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                                set_global_var(GVAR_KARMA_DEFENDER, 1);
                             end
                             else begin
-                                if (global_var(0) > -250) then begin
-                                    set_global_var(41, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                                    set_global_var(GVAR_KARMA_WANDERER, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -500) then begin
-                                        set_global_var(42, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                        set_global_var(GVAR_KARMA_BETRAYER, 1);
                                     end
                                     else begin
-                                        if (global_var(0) > -750) then begin
-                                            set_global_var(43, 1);
+                                        if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                            set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                         end
                                         else begin
-                                            if (global_var(0) > -1000) then begin
-                                                set_global_var(44, 1);
+                                            if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                                set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                             end
                                             else begin
-                                                set_global_var(45, 1);
+                                                set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                             end
                                         end
                                     end
@@ -358,58 +358,58 @@ begin
                         end
                     end
                 end
-                set_global_var(51, global_var(51) + -8);
+                set_global_var(GVAR_TOWN_REP_GECKO, global_var(GVAR_TOWN_REP_GECKO) + -8);
                 debug_msg("Added " + -8 + " to Town Rep");
             end
         end
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(5) > (2 * global_var(4))) or (global_var(3) == 1))) then begin
-            set_global_var(3, 1);
-            set_global_var(2, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
+            set_global_var(GVAR_BERSERKER_REPUTATION, 1);
+            set_global_var(GVAR_CHAMPION_REPUTATION, 0);
         end
-        set_global_var(0, global_var(0) + -10);
+        set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + -10);
         debug_msg("Player gains " + -10 + " Karma Points.");
-        set_global_var(37, 0);
-        set_global_var(38, 0);
-        set_global_var(39, 0);
-        set_global_var(40, 0);
-        set_global_var(41, 0);
-        set_global_var(42, 0);
-        set_global_var(43, 0);
-        set_global_var(44, 0);
-        set_global_var(45, 0);
-        if (global_var(0) >= 1000) then begin
-            set_global_var(37, 1);
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+        set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+        set_global_var(GVAR_KARMA_DEFENDER, 0);
+        set_global_var(GVAR_KARMA_WANDERER, 0);
+        set_global_var(GVAR_KARMA_BETRAYER, 0);
+        set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+        set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+            set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
         end
         else begin
-            if (global_var(0) >= 750) then begin
-                set_global_var(38, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
             end
             else begin
-                if (global_var(0) >= 500) then begin
-                    set_global_var(39, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                 end
                 else begin
-                    if (global_var(0) >= 250) then begin
-                        set_global_var(40, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                        set_global_var(GVAR_KARMA_DEFENDER, 1);
                     end
                     else begin
-                        if (global_var(0) > -250) then begin
-                            set_global_var(41, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                            set_global_var(GVAR_KARMA_WANDERER, 1);
                         end
                         else begin
-                            if (global_var(0) > -500) then begin
-                                set_global_var(42, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                set_global_var(GVAR_KARMA_BETRAYER, 1);
                             end
                             else begin
-                                if (global_var(0) > -750) then begin
-                                    set_global_var(43, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -1000) then begin
-                                        set_global_var(44, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                     end
                                     else begin
-                                        set_global_var(45, 1);
+                                        set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                     end
                                 end
                             end
@@ -418,7 +418,7 @@ begin
                 end
             end
         end
-        set_global_var(51, global_var(51) + -5);
+        set_global_var(GVAR_TOWN_REP_GECKO, global_var(GVAR_TOWN_REP_GECKO) + -5);
         debug_msg("Added " + -5 + " to Town Rep");
     end
 end
@@ -426,13 +426,13 @@ end
 procedure look_at_p_proc
 begin
     script_overrides;
-    display_msg(message_str(145, 100));
+    display_msg(message_str(SCRIPT_GCFOLK, 100));
 end
 
 procedure description_p_proc
 begin
     script_overrides;
-    display_msg(message_str(145, 101));
+    display_msg(message_str(SCRIPT_GCFOLK, 101));
 end
 
 procedure use_skill_on_p_proc
@@ -443,7 +443,7 @@ procedure damage_p_proc
 begin
     if (source_obj == dude_obj) then begin
         set_local_var(5, 1);
-        set_global_var(238, 1);
+        set_global_var(GVAR_ENEMY_GECKO, 1);
     end
 end
 
@@ -458,8 +458,8 @@ end
 procedure map_enter_p_proc
 begin
     Only_Once := 0;
-    critter_add_trait(self_obj, 1, 6, 49);
-    critter_add_trait(self_obj, 1, 5, 24);
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 49);
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_TOUGH_CITIZEN);
 end
 
 procedure push_p_proc
@@ -468,62 +468,62 @@ end
 
 procedure Node001
 begin
-    float_msg(self_obj, message_str(145, random(120, 132)), 8);
+    float_msg(self_obj, message_str(SCRIPT_GCFOLK, random(120, 132)), FLOAT_MSG_YELLOW);
 end
 
 procedure Node002
 begin
-    float_msg(self_obj, message_str(145, random(140, 151)), 8);
+    float_msg(self_obj, message_str(SCRIPT_GCFOLK, random(140, 151)), FLOAT_MSG_YELLOW);
 end
 
 procedure Node003
 begin
-    float_msg(self_obj, message_str(145, random(160, 169)), 8);
+    float_msg(self_obj, message_str(SCRIPT_GCFOLK, random(160, 169)), FLOAT_MSG_YELLOW);
 end
 
 procedure Node004
 begin
-    float_msg(self_obj, message_str(145, random(180, 189)), 8);
+    float_msg(self_obj, message_str(SCRIPT_GCFOLK, random(180, 189)), FLOAT_MSG_YELLOW);
 end
 
 procedure Node005
 begin
-    float_msg(self_obj, message_str(145, random(200, 209)), 8);
+    float_msg(self_obj, message_str(SCRIPT_GCFOLK, random(200, 209)), FLOAT_MSG_YELLOW);
 end
 
 procedure Node006
 begin
-    float_msg(self_obj, message_str(145, random(220, 229)), 8);
+    float_msg(self_obj, message_str(SCRIPT_GCFOLK, random(220, 229)), FLOAT_MSG_YELLOW);
 end
 
 procedure Node007
 begin
-    float_msg(self_obj, message_str(145, random(240, 249)), 8);
+    float_msg(self_obj, message_str(SCRIPT_GCFOLK, random(240, 249)), FLOAT_MSG_YELLOW);
 end
 
 procedure Node008
 begin
-    float_msg(self_obj, message_str(145, random(260, 269)), 8);
+    float_msg(self_obj, message_str(SCRIPT_GCFOLK, random(260, 269)), FLOAT_MSG_YELLOW);
 end
 
 procedure Node009
 begin
-    float_msg(self_obj, message_str(145, random(280, 291)), 8);
+    float_msg(self_obj, message_str(SCRIPT_GCFOLK, random(280, 291)), FLOAT_MSG_YELLOW);
 end
 
 procedure Node010
 begin
-    float_msg(self_obj, message_str(145, random(300, 309)), 8);
+    float_msg(self_obj, message_str(SCRIPT_GCFOLK, random(300, 309)), FLOAT_MSG_YELLOW);
 end
 
 procedure Node011
 begin
-    float_msg(self_obj, message_str(145, random(320, 329)), 8);
+    float_msg(self_obj, message_str(SCRIPT_GCFOLK, random(320, 329)), FLOAT_MSG_YELLOW);
 end
 
 procedure Node012
 begin
-    float_msg(self_obj, message_str(145, random(340, 347)), 8);
+    float_msg(self_obj, message_str(SCRIPT_GCFOLK, random(340, 347)), FLOAT_MSG_YELLOW);
 end
 
 procedure gobar
@@ -556,4 +556,3 @@ end
 procedure Node999
 begin
 end
-
