diff --git a/Games/Fallout2.Steam/Original/GECKJUNK.ssl b/Games/Fallout2.Steam/ReDefine/GECKJUNK.ssl
index 24e95eb..440c9a1 100644
--- a/Games/Fallout2.Steam/Original/GECKJUNK.ssl
+++ b/Games/Fallout2.Steam/ReDefine/GECKJUNK.ssl
@@ -141,11 +141,11 @@ end
 
 procedure map_enter_p_proc
 begin
-    if (metarule(14, 0)) then begin
-        display_msg(message_str(859, 100));
+    if (map_first_run) then begin
+        display_mstr(100);
     end
-    if (global_var(395) == 2) then begin
-        if (((game_time / (10 * 60 * 60)) - global_var(430)) > 24) then begin
+    if (global_var(GVAR_GECKO_MELTDOWN) == 2) then begin
+        if (((game_time / (10 * 60 * 60)) - global_var(GVAR_GECKO_TIMER_MELTDOWN)) > 24) then begin
             call ghost_town();
         end
     end
@@ -225,7 +225,7 @@ begin
             end
         end
     end
-    set_global_var(27, 0);
+    set_global_var(GVAR_LOAD_MAP_INDEX, 0);
 end
 
 procedure map_update_p_proc
@@ -325,4 +325,3 @@ begin
         kill_critter_type(16777323, 0);
     end
 end
-
