diff --git a/Games/Fallout2.Steam/Original/GSVALVE.ssl b/Games/Fallout2.Steam/ReDefine/GSVALVE.ssl
index f58d601..40e9521 100644
--- a/Games/Fallout2.Steam/Original/GSVALVE.ssl
+++ b/Games/Fallout2.Steam/ReDefine/GSVALVE.ssl
@@ -152,23 +152,23 @@ begin
         if (is_success(do_check(dude_obj, 0, 0))) then begin
             set_local_var(0, local_var(0) + 1);
             if (local_var(0) == 1) then begin
-                display_msg(message_str(847, 111));
+                display_mstr(111);
             end
             else begin
                 if (local_var(0) == 2) then begin
-                    display_msg(message_str(847, 112));
+                    display_mstr(112);
                 end
                 else begin
                     if (local_var(0) > 2) then begin
-                        set_global_var(238, 1);
-                        set_global_var(395, 1);
-                        display_msg(message_str(847, 113));
+                        set_global_var(GVAR_ENEMY_GECKO, 1);
+                        set_global_var(GVAR_GECKO_MELTDOWN, 1);
+                        display_mstr(113);
                     end
                 end
             end
         end
         else begin
-            display_msg(message_str(847, 110));
+            display_mstr(110);
         end
     end
 end
@@ -185,13 +185,13 @@ end
 procedure look_at_p_proc
 begin
     script_overrides;
-    display_msg(message_str(847, 100));
+    display_mstr(100);
 end
 
 procedure description_p_proc
 begin
     script_overrides;
-    display_msg(message_str(847, 101));
+    display_mstr(101);
 end
 
 procedure use_skill_on_p_proc
@@ -203,11 +203,11 @@ begin
             call repair_it();
         end
         else begin
-            display_msg(message_str(847, 120));
+            display_mstr(120);
         end
     end
     else begin
-        display_msg(message_str(847, 121));
+        display_mstr(121);
     end
 end
 
@@ -217,56 +217,56 @@ begin
     if (is_success(repair_check)) then begin
         item := obj_carrying_pid_obj(dude_obj, 258);
         rm_obj_from_inven(dude_obj, item);
-        set_global_var(82, 9);
-        set_global_var(83, 9);
-        display_msg(message_str(847, 122));
+        set_global_var(GVAR_VAULT_GECKO_PLANT, 9);
+        set_global_var(GVAR_VAULT_PLANT_STATUS, 9);
+        display_mstr(122);
         give_exp_points(4000);
-        display_msg(message_str(14, 100) + 4000 + message_str(14, 101));
-        set_global_var(51, global_var(51) + 30);
-        set_global_var(0, global_var(0) + 50);
+        display_msg(g_mstr(100) + 4000 + g_mstr(101));
+        set_global_var(GVAR_TOWN_REP_GECKO, global_var(GVAR_TOWN_REP_GECKO) + 30);
+        set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + 50);
         debug_msg("Player gains " + 50 + " Karma Points.");
-        set_global_var(37, 0);
-        set_global_var(38, 0);
-        set_global_var(39, 0);
-        set_global_var(40, 0);
-        set_global_var(41, 0);
-        set_global_var(42, 0);
-        set_global_var(43, 0);
-        set_global_var(44, 0);
-        set_global_var(45, 0);
-        if (global_var(0) >= 1000) then begin
-            set_global_var(37, 1);
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+        set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+        set_global_var(GVAR_KARMA_DEFENDER, 0);
+        set_global_var(GVAR_KARMA_WANDERER, 0);
+        set_global_var(GVAR_KARMA_BETRAYER, 0);
+        set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+        set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+            set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
         end
         else begin
-            if (global_var(0) >= 750) then begin
-                set_global_var(38, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
             end
             else begin
-                if (global_var(0) >= 500) then begin
-                    set_global_var(39, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                 end
                 else begin
-                    if (global_var(0) >= 250) then begin
-                        set_global_var(40, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                        set_global_var(GVAR_KARMA_DEFENDER, 1);
                     end
                     else begin
-                        if (global_var(0) > -250) then begin
-                            set_global_var(41, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                            set_global_var(GVAR_KARMA_WANDERER, 1);
                         end
                         else begin
-                            if (global_var(0) > -500) then begin
-                                set_global_var(42, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                set_global_var(GVAR_KARMA_BETRAYER, 1);
                             end
                             else begin
-                                if (global_var(0) > -750) then begin
-                                    set_global_var(43, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -1000) then begin
-                                        set_global_var(44, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                     end
                                     else begin
-                                        set_global_var(45, 1);
+                                        set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                     end
                                 end
                             end
@@ -277,11 +277,10 @@ begin
         end
     end
     else begin
-        display_msg(message_str(847, 123));
+        display_mstr(123);
     end
 end
 
 procedure Node999
 begin
 end
-
