diff --git a/Games/Fallout2.Steam/Original/HCENCHB.ssl b/Games/Fallout2.Steam/ReDefine/HCENCHB.ssl
index a3e0a7c..9839fd0 100644
--- a/Games/Fallout2.Steam/Original/HCENCHB.ssl
+++ b/Games/Fallout2.Steam/ReDefine/HCENCHB.ssl
@@ -166,9 +166,9 @@ end
 
 procedure critter_p_proc
 begin
-    if ((global_var(309) == 1) or (local_var(5) == 2) and obj_can_see_obj(self_obj, dude_obj)) then begin
+    if ((global_var(GVAR_ENEMY_BROKEN_HILLS) == 1) or (local_var(5) == 2) and obj_can_see_obj(self_obj, dude_obj)) then begin
         set_local_var(5, 1);
-        attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+        attack(dude_obj);
     end
 end
 
@@ -183,7 +183,7 @@ procedure talk_p_proc
 begin
     if (local_var(2) == 0) then begin
         set_local_var(2, 1);
-        set_local_var(3, (get_critter_stat(dude_obj, 3) - get_critter_stat(self_obj, 3)) * 5);
+        set_local_var(3, (dude_charisma - self_charisma) * 5);
         debug_msg("Base Reaction == " + local_var(3));
     end
     Static_Reaction := 0;
@@ -195,14 +195,14 @@ begin
     else begin
         reaction_bonus_karma := -1 * (-1 * reaction_bonus_karma / 1000);
     end
-    if (global_var(54) >= 0) then begin
-        reaction_bonus_town_rep := global_var(54) / 2;
+    if (global_var(GVAR_TOWN_REP_BROKEN_HILLS) >= 0) then begin
+        reaction_bonus_town_rep := global_var(GVAR_TOWN_REP_BROKEN_HILLS) / 2;
     end
     else begin
-        reaction_bonus_town_rep := -1 * (-1 * global_var(54) / 2);
+        reaction_bonus_town_rep := -1 * (-1 * global_var(GVAR_TOWN_REP_BROKEN_HILLS) / 2);
     end
-    if (has_trait(2, dude_obj, 13)) then begin
-        if (get_critter_stat(dude_obj, 34) == get_critter_stat(self_obj, 34)) then begin
+    if (has_trait(TRAIT_TRAIT, dude_obj, TRAIT_sex_appeal)) then begin
+        if (dude_gender == self_gender) then begin
             Static_Reaction := Static_Reaction + -20;
         end
         else begin
@@ -218,48 +218,48 @@ begin
         debug_msg("Karma Reaction Bonus == " + Static_Reaction);
         Static_Reaction := Static_Reaction - reaction_bonus_town_rep;
         debug_msg("Town Rep Bonus == " + Static_Reaction);
-        set_global_var(37, 0);
-        set_global_var(38, 0);
-        set_global_var(39, 0);
-        set_global_var(40, 0);
-        set_global_var(41, 0);
-        set_global_var(42, 0);
-        set_global_var(43, 0);
-        set_global_var(44, 0);
-        set_global_var(45, 0);
-        if (global_var(0) >= 1000) then begin
-            set_global_var(37, 1);
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+        set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+        set_global_var(GVAR_KARMA_DEFENDER, 0);
+        set_global_var(GVAR_KARMA_WANDERER, 0);
+        set_global_var(GVAR_KARMA_BETRAYER, 0);
+        set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+        set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+            set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
         end
         else begin
-            if (global_var(0) >= 750) then begin
-                set_global_var(38, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
             end
             else begin
-                if (global_var(0) >= 500) then begin
-                    set_global_var(39, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                 end
                 else begin
-                    if (global_var(0) >= 250) then begin
-                        set_global_var(40, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                        set_global_var(GVAR_KARMA_DEFENDER, 1);
                     end
                     else begin
-                        if (global_var(0) > -250) then begin
-                            set_global_var(41, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                            set_global_var(GVAR_KARMA_WANDERER, 1);
                         end
                         else begin
-                            if (global_var(0) > -500) then begin
-                                set_global_var(42, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                set_global_var(GVAR_KARMA_BETRAYER, 1);
                             end
                             else begin
-                                if (global_var(0) > -750) then begin
-                                    set_global_var(43, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -1000) then begin
-                                        set_global_var(44, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                     end
                                     else begin
-                                        set_global_var(45, 1);
+                                        set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                     end
                                 end
                             end
@@ -268,35 +268,35 @@ begin
                 end
             end
         end
-        if (global_var(37) == 1) then begin
+        if (global_var(GVAR_KARMA_HOLY_WARRIOR) == 1) then begin
             Static_Reaction := Static_Reaction - 20;
         end
         else begin
-            if (global_var(38) == 1) then begin
+            if (global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES) == 1) then begin
                 Static_Reaction := Static_Reaction - 15;
             end
             else begin
-                if (global_var(39) == 1) then begin
+                if (global_var(GVAR_KARMA_SHIELD_OF_HOPE) == 1) then begin
                     Static_Reaction := Static_Reaction - 10;
                 end
                 else begin
-                    if (global_var(40) == 1) then begin
+                    if (global_var(GVAR_KARMA_DEFENDER) == 1) then begin
                         Static_Reaction := Static_Reaction - 5;
                     end
                     else begin
-                        if (global_var(41) == 1) then begin
+                        if (global_var(GVAR_KARMA_WANDERER) == 1) then begin
                             Static_Reaction := Static_Reaction - 0;
                         end
                         else begin
-                            if (global_var(42) == 1) then begin
+                            if (global_var(GVAR_KARMA_BETRAYER) == 1) then begin
                                 Static_Reaction := Static_Reaction - -5;
                             end
                             else begin
-                                if (global_var(43) == 1) then begin
+                                if (global_var(GVAR_KARMA_SWORD_OF_DESPAIR) == 1) then begin
                                     Static_Reaction := Static_Reaction - -10;
                                 end
                                 else begin
-                                    if (global_var(44) == 1) then begin
+                                    if (global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE) == 1) then begin
                                         Static_Reaction := Static_Reaction - -15;
                                     end
                                     else begin
@@ -316,48 +316,48 @@ begin
         debug_msg("Karma Reaction Bonus == " + Static_Reaction);
         Static_Reaction := Static_Reaction + reaction_bonus_town_rep;
         debug_msg("Town Rep Bonus == " + Static_Reaction);
-        set_global_var(37, 0);
-        set_global_var(38, 0);
-        set_global_var(39, 0);
-        set_global_var(40, 0);
-        set_global_var(41, 0);
-        set_global_var(42, 0);
-        set_global_var(43, 0);
-        set_global_var(44, 0);
-        set_global_var(45, 0);
-        if (global_var(0) >= 1000) then begin
-            set_global_var(37, 1);
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+        set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+        set_global_var(GVAR_KARMA_DEFENDER, 0);
+        set_global_var(GVAR_KARMA_WANDERER, 0);
+        set_global_var(GVAR_KARMA_BETRAYER, 0);
+        set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+        set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+            set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
         end
         else begin
-            if (global_var(0) >= 750) then begin
-                set_global_var(38, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
             end
             else begin
-                if (global_var(0) >= 500) then begin
-                    set_global_var(39, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                 end
                 else begin
-                    if (global_var(0) >= 250) then begin
-                        set_global_var(40, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                        set_global_var(GVAR_KARMA_DEFENDER, 1);
                     end
                     else begin
-                        if (global_var(0) > -250) then begin
-                            set_global_var(41, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                            set_global_var(GVAR_KARMA_WANDERER, 1);
                         end
                         else begin
-                            if (global_var(0) > -500) then begin
-                                set_global_var(42, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                set_global_var(GVAR_KARMA_BETRAYER, 1);
                             end
                             else begin
-                                if (global_var(0) > -750) then begin
-                                    set_global_var(43, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -1000) then begin
-                                        set_global_var(44, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                     end
                                     else begin
-                                        set_global_var(45, 1);
+                                        set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                     end
                                 end
                             end
@@ -366,35 +366,35 @@ begin
                 end
             end
         end
-        if (global_var(37) == 1) then begin
+        if (global_var(GVAR_KARMA_HOLY_WARRIOR) == 1) then begin
             Static_Reaction := Static_Reaction + 20;
         end
         else begin
-            if (global_var(38) == 1) then begin
+            if (global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES) == 1) then begin
                 Static_Reaction := Static_Reaction + 15;
             end
             else begin
-                if (global_var(39) == 1) then begin
+                if (global_var(GVAR_KARMA_SHIELD_OF_HOPE) == 1) then begin
                     Static_Reaction := Static_Reaction + 10;
                 end
                 else begin
-                    if (global_var(40) == 1) then begin
+                    if (global_var(GVAR_KARMA_DEFENDER) == 1) then begin
                         Static_Reaction := Static_Reaction + 5;
                     end
                     else begin
-                        if (global_var(41) == 1) then begin
+                        if (global_var(GVAR_KARMA_WANDERER) == 1) then begin
                             Static_Reaction := Static_Reaction + 0;
                         end
                         else begin
-                            if (global_var(42) == 1) then begin
+                            if (global_var(GVAR_KARMA_BETRAYER) == 1) then begin
                                 Static_Reaction := Static_Reaction + -5;
                             end
                             else begin
-                                if (global_var(43) == 1) then begin
+                                if (global_var(GVAR_KARMA_SWORD_OF_DESPAIR) == 1) then begin
                                     Static_Reaction := Static_Reaction + -10;
                                 end
                                 else begin
-                                    if (global_var(44) == 1) then begin
+                                    if (global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE) == 1) then begin
                                         Static_Reaction := Static_Reaction + -15;
                                     end
                                     else begin
@@ -410,28 +410,28 @@ begin
         debug_msg("Good Person Reaction Bonus == " + Static_Reaction);
     end
     if (Slavery_Tolerant == 0) then begin
-        if (global_var(11) == 1) then begin
+        if (global_var(GVAR_REPUTATION_SLAVER) == 1) then begin
             Static_Reaction := Static_Reaction + (2 * -25);
         end
     end
     else begin
         if (Slavery_Tolerant == 1) then begin
-            if (global_var(11) == 1) then begin
+            if (global_var(GVAR_REPUTATION_SLAVER) == 1) then begin
                 Static_Reaction := Static_Reaction + -25;
             end
         end
         else begin
             if (Slavery_Tolerant == 3) then begin
-                if (global_var(11) == 1) then begin
+                if (global_var(GVAR_REPUTATION_SLAVER) == 1) then begin
                     Static_Reaction := Static_Reaction - -25;
                 end
             end
         end
     end
     debug_msg("Slaver + Aligned Reaction == " + Static_Reaction);
-    Static_Reaction := Static_Reaction + (10 * has_trait(0, dude_obj, 10));
+    Static_Reaction := Static_Reaction + (10 * has_trait(TRAIT_PERK, dude_obj, PERK_presence));
     debug_msg("Presence Reaction == " + Static_Reaction);
-    if (global_var(1) >= 2) then begin
+    if (global_var(GVAR_CHILDKILLER_REPUTATION) >= 2) then begin
         Static_Reaction := Static_Reaction + -30;
     end
     debug_msg("Childkiller Reaction == " + Static_Reaction);
@@ -441,7 +441,7 @@ begin
         call Node998();
     end
     else begin
-        if (global_var(309) == 1) then begin
+        if (global_var(GVAR_ENEMY_BROKEN_HILLS) == 1) then begin
             start_gdialog(592, self_obj, 4, -1, -1);
             gsay_start;
             call Node006();
@@ -455,7 +455,7 @@ begin
                 call Node001();
             end
             else begin
-                if (get_critter_stat(dude_obj, 4) <= 2) then begin
+                if (dude_iq <= 2) then begin
                     call Node002();
                 end
                 else begin
@@ -471,59 +471,59 @@ end
 procedure destroy_p_proc
 begin
     if (source_obj == dude_obj) then begin
-        set_global_var(5, global_var(5) + 1);
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(4) > (3 * global_var(5))) or (global_var(2) == 1)) and (global_var(1) == 0)) then begin
-            set_global_var(2, 1);
-            set_global_var(3, 0);
+        set_global_var(GVAR_GOOD_MONSTER, global_var(GVAR_GOOD_MONSTER) + 1);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(4) > (3 * global_var(GVAR_GOOD_MONSTER))) or (global_var(GVAR_CHAMPION_REPUTATION) == 1)) and (global_var(GVAR_CHILDKILLER_REPUTATION) == 0)) then begin
+            set_global_var(GVAR_CHAMPION_REPUTATION, 1);
+            set_global_var(GVAR_BERSERKER_REPUTATION, 0);
         end
-        if (metarule(51, self_obj) == 2) then begin
+        if (critter_kill_type(self_obj) == KILL_TYPE_children_kills) then begin
             if (source_obj == dude_obj) then begin
-                set_global_var(2, 0);
-                set_global_var(1, global_var(1) + 1);
-                set_global_var(0, global_var(0) + -15);
+                set_global_var(GVAR_CHAMPION_REPUTATION, 0);
+                set_global_var(GVAR_CHILDKILLER_REPUTATION, global_var(GVAR_CHILDKILLER_REPUTATION) + 1);
+                set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + -15);
                 debug_msg("Player gains " + -15 + " Karma Points.");
-                set_global_var(37, 0);
-                set_global_var(38, 0);
-                set_global_var(39, 0);
-                set_global_var(40, 0);
-                set_global_var(41, 0);
-                set_global_var(42, 0);
-                set_global_var(43, 0);
-                set_global_var(44, 0);
-                set_global_var(45, 0);
-                if (global_var(0) >= 1000) then begin
-                    set_global_var(37, 1);
+                set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+                set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+                set_global_var(GVAR_KARMA_DEFENDER, 0);
+                set_global_var(GVAR_KARMA_WANDERER, 0);
+                set_global_var(GVAR_KARMA_BETRAYER, 0);
+                set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+                set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+                set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+                    set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
                 end
                 else begin
-                    if (global_var(0) >= 750) then begin
-                        set_global_var(38, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
                     end
                     else begin
-                        if (global_var(0) >= 500) then begin
-                            set_global_var(39, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                            set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                         end
                         else begin
-                            if (global_var(0) >= 250) then begin
-                                set_global_var(40, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                                set_global_var(GVAR_KARMA_DEFENDER, 1);
                             end
                             else begin
-                                if (global_var(0) > -250) then begin
-                                    set_global_var(41, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                                    set_global_var(GVAR_KARMA_WANDERER, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -500) then begin
-                                        set_global_var(42, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                        set_global_var(GVAR_KARMA_BETRAYER, 1);
                                     end
                                     else begin
-                                        if (global_var(0) > -750) then begin
-                                            set_global_var(43, 1);
+                                        if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                            set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                         end
                                         else begin
-                                            if (global_var(0) > -1000) then begin
-                                                set_global_var(44, 1);
+                                            if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                                set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                             end
                                             else begin
-                                                set_global_var(45, 1);
+                                                set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                             end
                                         end
                                     end
@@ -532,58 +532,58 @@ begin
                         end
                     end
                 end
-                set_global_var(54, global_var(54) + -8);
+                set_global_var(GVAR_TOWN_REP_BROKEN_HILLS, global_var(GVAR_TOWN_REP_BROKEN_HILLS) + -8);
                 debug_msg("Added " + -8 + " to Town Rep");
             end
         end
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(5) > (2 * global_var(4))) or (global_var(3) == 1))) then begin
-            set_global_var(3, 1);
-            set_global_var(2, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
+            set_global_var(GVAR_BERSERKER_REPUTATION, 1);
+            set_global_var(GVAR_CHAMPION_REPUTATION, 0);
         end
-        set_global_var(0, global_var(0) + -10);
+        set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + -10);
         debug_msg("Player gains " + -10 + " Karma Points.");
-        set_global_var(37, 0);
-        set_global_var(38, 0);
-        set_global_var(39, 0);
-        set_global_var(40, 0);
-        set_global_var(41, 0);
-        set_global_var(42, 0);
-        set_global_var(43, 0);
-        set_global_var(44, 0);
-        set_global_var(45, 0);
-        if (global_var(0) >= 1000) then begin
-            set_global_var(37, 1);
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+        set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+        set_global_var(GVAR_KARMA_DEFENDER, 0);
+        set_global_var(GVAR_KARMA_WANDERER, 0);
+        set_global_var(GVAR_KARMA_BETRAYER, 0);
+        set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+        set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+            set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
         end
         else begin
-            if (global_var(0) >= 750) then begin
-                set_global_var(38, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
             end
             else begin
-                if (global_var(0) >= 500) then begin
-                    set_global_var(39, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                 end
                 else begin
-                    if (global_var(0) >= 250) then begin
-                        set_global_var(40, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                        set_global_var(GVAR_KARMA_DEFENDER, 1);
                     end
                     else begin
-                        if (global_var(0) > -250) then begin
-                            set_global_var(41, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                            set_global_var(GVAR_KARMA_WANDERER, 1);
                         end
                         else begin
-                            if (global_var(0) > -500) then begin
-                                set_global_var(42, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                set_global_var(GVAR_KARMA_BETRAYER, 1);
                             end
                             else begin
-                                if (global_var(0) > -750) then begin
-                                    set_global_var(43, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -1000) then begin
-                                        set_global_var(44, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                     end
                                     else begin
-                                        set_global_var(45, 1);
+                                        set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                     end
                                 end
                             end
@@ -592,7 +592,7 @@ begin
                 end
             end
         end
-        set_global_var(54, global_var(54) + -5);
+        set_global_var(GVAR_TOWN_REP_BROKEN_HILLS, global_var(GVAR_TOWN_REP_BROKEN_HILLS) + -5);
         debug_msg("Added " + -5 + " to Town Rep");
     end
 end
@@ -600,13 +600,13 @@ end
 procedure look_at_p_proc
 begin
     script_overrides;
-    display_msg(message_str(592, 100));
+    display_mstr(100);
 end
 
 procedure description_p_proc
 begin
     script_overrides;
-    display_msg(message_str(592, 101));
+    display_mstr(101);
 end
 
 procedure use_skill_on_p_proc
@@ -617,7 +617,7 @@ procedure damage_p_proc
 begin
     if (source_obj == dude_obj) then begin
         set_local_var(6, 1);
-        set_global_var(309, 1);
+        set_global_var(GVAR_ENEMY_BROKEN_HILLS, 1);
     end
 end
 
@@ -626,16 +626,16 @@ begin
     variable LVar0 := 0;
     variable LVar1 := 0;
     Only_Once := 0;
-    critter_add_trait(self_obj, 1, 6, 120);
-    critter_add_trait(self_obj, 1, 5, 12);
-    if (metarule(14, 0)) then begin
-        LVar0 := create_object_sid(242, 0, 0, -1);
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 120);
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_GENERIC_GUARDS);
+    if (map_first_run) then begin
+        LVar0 := create_object(242, 0, 0);
         add_obj_to_inven(self_obj, LVar0);
         if (random(1, 100) < 11) then begin
-            LVar1 := create_object_sid(40, 0, 0, -1);
+            LVar1 := create_object(40, 0, 0);
             add_obj_to_inven(self_obj, LVar1);
         end
-        wield_obj_critter(self_obj, LVar0);
+        wield_obj(LVar0);
     end
 end
 
@@ -645,7 +645,7 @@ end
 
 procedure Node995
 begin
-    if (get_critter_stat(dude_obj, 3) >= 7) then begin
+    if (dude_charisma >= 7) then begin
         set_local_var(3, 100);
         set_local_var(1, 4);
     end
@@ -663,7 +663,7 @@ end
 
 procedure Node997
 begin
-    if (get_critter_stat(dude_obj, 3) >= 8) then begin
+    if (dude_charisma >= 8) then begin
         call Node004();
     end
     else begin
@@ -682,44 +682,43 @@ end
 
 procedure Node001
 begin
-    gsay_reply(592, 103);
-    giq_option(-1, 592, 104, Node002, 50);
-    if ((get_critter_stat(dude_obj, 4) > 1) and (get_critter_stat(dude_obj, 4) < 3)) then begin
-        giq_option(2, 592, 105, Node002, 50);
+    Reply(103);
+    giq_option(-1, NAME, 104, Node002, NEUTRAL_REACTION);
+    if ((dude_iq > 1) and (dude_iq < 3)) then begin
+        giq_option(2, NAME, 105, Node002, NEUTRAL_REACTION);
     end
-    giq_option(4, 592, 106, Node003, 50);
-    giq_option(4, 592, 107, Node004, 50);
+    giq_option(4, NAME, 106, Node003, NEUTRAL_REACTION);
+    giq_option(4, NAME, 107, Node004, NEUTRAL_REACTION);
 end
 
 procedure Node002
 begin
-    gsay_reply(592, 108);
-    giq_option(1, 592, 109, Node999, 50);
+    Reply(108);
+    giq_option(1, NAME, 109, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node003
 begin
-    gsay_reply(592, 110);
-    giq_option(4, 592, 111, Node003, 50);
-    giq_option(4, 592, 112, Node999, 50);
+    Reply(110);
+    giq_option(4, NAME, 111, Node003, NEUTRAL_REACTION);
+    giq_option(4, NAME, 112, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node004
 begin
-    gsay_reply(592, 113);
-    giq_option(4, 592, 114, Node003, 50);
-    giq_option(4, 592, 115, Node999, 50);
+    Reply(113);
+    giq_option(4, NAME, 114, Node003, NEUTRAL_REACTION);
+    giq_option(4, NAME, 115, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node005
 begin
-    gsay_reply(592, 116);
-    giq_option(4, 592, 117, Node999, 50);
+    Reply(116);
+    giq_option(4, NAME, 117, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node006
 begin
-    gsay_reply(592, 118);
-    giq_option(4, 592, 119, Node998, 50);
+    Reply(118);
+    giq_option(4, NAME, 119, Node998, NEUTRAL_REACTION);
 end
-
