diff --git a/Games/Fallout2.Steam/Original/HCTYPHON.ssl b/Games/Fallout2.Steam/ReDefine/HCTYPHON.ssl
index 4d184ec..87de60d 100644
--- a/Games/Fallout2.Steam/Original/HCTYPHON.ssl
+++ b/Games/Fallout2.Steam/ReDefine/HCTYPHON.ssl
@@ -197,24 +197,24 @@ procedure pickup_p_proc
 begin
     if (source_obj == dude_obj) then begin
         if (3 == 1) then begin
-            attack_complex(source_obj, 0, 1, 0, 0, 30000, 0, 0);
+            attack(source_obj);
         end
         else begin
             if (3 == 2) then begin
-                float_msg(self_obj, message_str(14, random(3325, 3329)), 0);
+                float_msg(self_obj, g_mstr(random(3325, 3329)), FLOAT_MSG_NORMAL);
                 if (7 > -1) then begin
                     if (local_var(7) >= 1) then begin
-                        attack_complex(source_obj, 0, 1, 0, 0, 30000, 0, 0);
+                        attack(source_obj);
                     end
                     set_local_var(7, local_var(7) + 1);
                 end
             end
             else begin
                 if (3 == 3) then begin
-                    float_msg(self_obj, message_str(14, random(3050, 3064)), 0);
+                    float_msg(self_obj, g_mstr(random(3050, 3064)), FLOAT_MSG_NORMAL);
                     if (7 > -1) then begin
                         if (local_var(7) >= 1) then begin
-                            attack_complex(source_obj, 0, 1, 0, 0, 30000, 0, 0);
+                            attack(source_obj);
                         end
                         set_local_var(7, local_var(7) + 1);
                     end
@@ -238,16 +238,16 @@ end
 
 procedure critter_p_proc
 begin
-    if ((global_var(309) == 1) or (local_var(5) != 0) and obj_can_see_obj(self_obj, dude_obj)) then begin
+    if ((global_var(GVAR_ENEMY_BROKEN_HILLS) == 1) or (local_var(5) != 0) and obj_can_see_obj(self_obj, dude_obj)) then begin
         set_local_var(5, 1);
         if (critter_is_fleeing(self_obj)) then begin
             if (anim_busy(self_obj) == 0) then begin
                 if ((tile_distance_objs(self_obj, dude_obj) < 8) or obj_can_see_obj(self_obj, dude_obj)) then begin
                     if (anim_busy(self_obj) == 0) then begin
-                        global_temp := rotation_to_tile(tile_num(dude_obj), tile_num(self_obj));
+                        global_temp := rotation_to_tile(dude_tile, tile_num(self_obj));
                         animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(self_obj), global_temp, random(3, 10)), 1);
                         global_temp := (global_temp + 1) % 6;
-                        while ((anim_busy(self_obj) == 0) and (global_temp != rotation_to_tile(tile_num(dude_obj), tile_num(self_obj)))) do begin
+                        while ((anim_busy(self_obj) == 0) and (global_temp != rotation_to_tile(dude_tile, tile_num(self_obj)))) do begin
                             debug_msg("flee loop: rot == " + global_temp);
                             animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(self_obj), global_temp, random(3, 10)), 1);
                             global_temp := (global_temp + 1) % 6;
@@ -257,7 +257,7 @@ begin
             end
         end
         else begin
-            attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+            attack(dude_obj);
         end
     end
 end
@@ -267,48 +267,48 @@ begin
     Evil_Critter := 0;
     Slavery_Tolerant := 2;
     Karma_Perception := 10;
-    set_global_var(37, 0);
-    set_global_var(38, 0);
-    set_global_var(39, 0);
-    set_global_var(40, 0);
-    set_global_var(41, 0);
-    set_global_var(42, 0);
-    set_global_var(43, 0);
-    set_global_var(44, 0);
-    set_global_var(45, 0);
-    if (global_var(0) >= 1000) then begin
-        set_global_var(37, 1);
+    set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+    set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+    set_global_var(GVAR_KARMA_DEFENDER, 0);
+    set_global_var(GVAR_KARMA_WANDERER, 0);
+    set_global_var(GVAR_KARMA_BETRAYER, 0);
+    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+    set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+    set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+    if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
     end
     else begin
-        if (global_var(0) >= 750) then begin
-            set_global_var(38, 1);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+            set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
         end
         else begin
-            if (global_var(0) >= 500) then begin
-                set_global_var(39, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
             end
             else begin
-                if (global_var(0) >= 250) then begin
-                    set_global_var(40, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                    set_global_var(GVAR_KARMA_DEFENDER, 1);
                 end
                 else begin
-                    if (global_var(0) > -250) then begin
-                        set_global_var(41, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                        set_global_var(GVAR_KARMA_WANDERER, 1);
                     end
                     else begin
-                        if (global_var(0) > -500) then begin
-                            set_global_var(42, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                            set_global_var(GVAR_KARMA_BETRAYER, 1);
                         end
                         else begin
-                            if (global_var(0) > -750) then begin
-                                set_global_var(43, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                             end
                             else begin
-                                if (global_var(0) > -1000) then begin
-                                    set_global_var(44, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                    set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                 end
                                 else begin
-                                    set_global_var(45, 1);
+                                    set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                 end
                             end
                         end
@@ -319,7 +319,7 @@ begin
     end
     if (local_var(2) == 0) then begin
         set_local_var(2, 1);
-        set_local_var(3, (get_critter_stat(dude_obj, 3) - get_critter_stat(self_obj, 3)) * 5);
+        set_local_var(3, (dude_charisma - self_charisma) * 5);
         debug_msg("Base Reaction == " + local_var(3));
     end
     Static_Reaction := 0;
@@ -331,14 +331,14 @@ begin
     else begin
         reaction_bonus_karma := -1 * (-1 * reaction_bonus_karma / 1000);
     end
-    if (global_var(54) >= 0) then begin
-        reaction_bonus_town_rep := global_var(54) / 2;
+    if (global_var(GVAR_TOWN_REP_BROKEN_HILLS) >= 0) then begin
+        reaction_bonus_town_rep := global_var(GVAR_TOWN_REP_BROKEN_HILLS) / 2;
     end
     else begin
-        reaction_bonus_town_rep := -1 * (-1 * global_var(54) / 2);
+        reaction_bonus_town_rep := -1 * (-1 * global_var(GVAR_TOWN_REP_BROKEN_HILLS) / 2);
     end
-    if (has_trait(2, dude_obj, 13)) then begin
-        if (get_critter_stat(dude_obj, 34) == get_critter_stat(self_obj, 34)) then begin
+    if (has_trait(TRAIT_TRAIT, dude_obj, TRAIT_sex_appeal)) then begin
+        if (dude_gender == self_gender) then begin
             Static_Reaction := Static_Reaction + -20;
         end
         else begin
@@ -354,48 +354,48 @@ begin
         debug_msg("Karma Reaction Bonus == " + Static_Reaction);
         Static_Reaction := Static_Reaction - reaction_bonus_town_rep;
         debug_msg("Town Rep Bonus == " + Static_Reaction);
-        set_global_var(37, 0);
-        set_global_var(38, 0);
-        set_global_var(39, 0);
-        set_global_var(40, 0);
-        set_global_var(41, 0);
-        set_global_var(42, 0);
-        set_global_var(43, 0);
-        set_global_var(44, 0);
-        set_global_var(45, 0);
-        if (global_var(0) >= 1000) then begin
-            set_global_var(37, 1);
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+        set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+        set_global_var(GVAR_KARMA_DEFENDER, 0);
+        set_global_var(GVAR_KARMA_WANDERER, 0);
+        set_global_var(GVAR_KARMA_BETRAYER, 0);
+        set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+        set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+            set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
         end
         else begin
-            if (global_var(0) >= 750) then begin
-                set_global_var(38, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
             end
             else begin
-                if (global_var(0) >= 500) then begin
-                    set_global_var(39, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                 end
                 else begin
-                    if (global_var(0) >= 250) then begin
-                        set_global_var(40, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                        set_global_var(GVAR_KARMA_DEFENDER, 1);
                     end
                     else begin
-                        if (global_var(0) > -250) then begin
-                            set_global_var(41, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                            set_global_var(GVAR_KARMA_WANDERER, 1);
                         end
                         else begin
-                            if (global_var(0) > -500) then begin
-                                set_global_var(42, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                set_global_var(GVAR_KARMA_BETRAYER, 1);
                             end
                             else begin
-                                if (global_var(0) > -750) then begin
-                                    set_global_var(43, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -1000) then begin
-                                        set_global_var(44, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                     end
                                     else begin
-                                        set_global_var(45, 1);
+                                        set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                     end
                                 end
                             end
@@ -404,35 +404,35 @@ begin
                 end
             end
         end
-        if (global_var(37) == 1) then begin
+        if (global_var(GVAR_KARMA_HOLY_WARRIOR) == 1) then begin
             Static_Reaction := Static_Reaction - 20;
         end
         else begin
-            if (global_var(38) == 1) then begin
+            if (global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES) == 1) then begin
                 Static_Reaction := Static_Reaction - 15;
             end
             else begin
-                if (global_var(39) == 1) then begin
+                if (global_var(GVAR_KARMA_SHIELD_OF_HOPE) == 1) then begin
                     Static_Reaction := Static_Reaction - 10;
                 end
                 else begin
-                    if (global_var(40) == 1) then begin
+                    if (global_var(GVAR_KARMA_DEFENDER) == 1) then begin
                         Static_Reaction := Static_Reaction - 5;
                     end
                     else begin
-                        if (global_var(41) == 1) then begin
+                        if (global_var(GVAR_KARMA_WANDERER) == 1) then begin
                             Static_Reaction := Static_Reaction - 0;
                         end
                         else begin
-                            if (global_var(42) == 1) then begin
+                            if (global_var(GVAR_KARMA_BETRAYER) == 1) then begin
                                 Static_Reaction := Static_Reaction - -5;
                             end
                             else begin
-                                if (global_var(43) == 1) then begin
+                                if (global_var(GVAR_KARMA_SWORD_OF_DESPAIR) == 1) then begin
                                     Static_Reaction := Static_Reaction - -10;
                                 end
                                 else begin
-                                    if (global_var(44) == 1) then begin
+                                    if (global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE) == 1) then begin
                                         Static_Reaction := Static_Reaction - -15;
                                     end
                                     else begin
@@ -452,48 +452,48 @@ begin
         debug_msg("Karma Reaction Bonus == " + Static_Reaction);
         Static_Reaction := Static_Reaction + reaction_bonus_town_rep;
         debug_msg("Town Rep Bonus == " + Static_Reaction);
-        set_global_var(37, 0);
-        set_global_var(38, 0);
-        set_global_var(39, 0);
-        set_global_var(40, 0);
-        set_global_var(41, 0);
-        set_global_var(42, 0);
-        set_global_var(43, 0);
-        set_global_var(44, 0);
-        set_global_var(45, 0);
-        if (global_var(0) >= 1000) then begin
-            set_global_var(37, 1);
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+        set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+        set_global_var(GVAR_KARMA_DEFENDER, 0);
+        set_global_var(GVAR_KARMA_WANDERER, 0);
+        set_global_var(GVAR_KARMA_BETRAYER, 0);
+        set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+        set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+            set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
         end
         else begin
-            if (global_var(0) >= 750) then begin
-                set_global_var(38, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
             end
             else begin
-                if (global_var(0) >= 500) then begin
-                    set_global_var(39, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                 end
                 else begin
-                    if (global_var(0) >= 250) then begin
-                        set_global_var(40, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                        set_global_var(GVAR_KARMA_DEFENDER, 1);
                     end
                     else begin
-                        if (global_var(0) > -250) then begin
-                            set_global_var(41, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                            set_global_var(GVAR_KARMA_WANDERER, 1);
                         end
                         else begin
-                            if (global_var(0) > -500) then begin
-                                set_global_var(42, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                set_global_var(GVAR_KARMA_BETRAYER, 1);
                             end
                             else begin
-                                if (global_var(0) > -750) then begin
-                                    set_global_var(43, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -1000) then begin
-                                        set_global_var(44, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                     end
                                     else begin
-                                        set_global_var(45, 1);
+                                        set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                     end
                                 end
                             end
@@ -502,35 +502,35 @@ begin
                 end
             end
         end
-        if (global_var(37) == 1) then begin
+        if (global_var(GVAR_KARMA_HOLY_WARRIOR) == 1) then begin
             Static_Reaction := Static_Reaction + 20;
         end
         else begin
-            if (global_var(38) == 1) then begin
+            if (global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES) == 1) then begin
                 Static_Reaction := Static_Reaction + 15;
             end
             else begin
-                if (global_var(39) == 1) then begin
+                if (global_var(GVAR_KARMA_SHIELD_OF_HOPE) == 1) then begin
                     Static_Reaction := Static_Reaction + 10;
                 end
                 else begin
-                    if (global_var(40) == 1) then begin
+                    if (global_var(GVAR_KARMA_DEFENDER) == 1) then begin
                         Static_Reaction := Static_Reaction + 5;
                     end
                     else begin
-                        if (global_var(41) == 1) then begin
+                        if (global_var(GVAR_KARMA_WANDERER) == 1) then begin
                             Static_Reaction := Static_Reaction + 0;
                         end
                         else begin
-                            if (global_var(42) == 1) then begin
+                            if (global_var(GVAR_KARMA_BETRAYER) == 1) then begin
                                 Static_Reaction := Static_Reaction + -5;
                             end
                             else begin
-                                if (global_var(43) == 1) then begin
+                                if (global_var(GVAR_KARMA_SWORD_OF_DESPAIR) == 1) then begin
                                     Static_Reaction := Static_Reaction + -10;
                                 end
                                 else begin
-                                    if (global_var(44) == 1) then begin
+                                    if (global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE) == 1) then begin
                                         Static_Reaction := Static_Reaction + -15;
                                     end
                                     else begin
@@ -546,34 +546,34 @@ begin
         debug_msg("Good Person Reaction Bonus == " + Static_Reaction);
     end
     if (Slavery_Tolerant == 0) then begin
-        if (global_var(11) == 1) then begin
+        if (global_var(GVAR_REPUTATION_SLAVER) == 1) then begin
             Static_Reaction := Static_Reaction + (2 * -25);
         end
     end
     else begin
         if (Slavery_Tolerant == 1) then begin
-            if (global_var(11) == 1) then begin
+            if (global_var(GVAR_REPUTATION_SLAVER) == 1) then begin
                 Static_Reaction := Static_Reaction + -25;
             end
         end
         else begin
             if (Slavery_Tolerant == 3) then begin
-                if (global_var(11) == 1) then begin
+                if (global_var(GVAR_REPUTATION_SLAVER) == 1) then begin
                     Static_Reaction := Static_Reaction - -25;
                 end
             end
         end
     end
     debug_msg("Slaver + Aligned Reaction == " + Static_Reaction);
-    Static_Reaction := Static_Reaction + (10 * has_trait(0, dude_obj, 10));
+    Static_Reaction := Static_Reaction + (10 * has_trait(TRAIT_PERK, dude_obj, PERK_presence));
     debug_msg("Presence Reaction == " + Static_Reaction);
-    if (global_var(1) >= 2) then begin
+    if (global_var(GVAR_CHILDKILLER_REPUTATION) >= 2) then begin
         Static_Reaction := Static_Reaction + -30;
     end
     debug_msg("Childkiller Reaction == " + Static_Reaction);
     set_local_var(0, local_var(3) + Static_Reaction);
     debug_msg("Initial Reaction == " + local_var(0));
-    if (global_var(598) == 8) then begin
+    if (global_var(GVAR_TYPHON_QUEST_STATUS) == 8) then begin
         start_gdialog(1186, self_obj, 4, -1, -1);
         gsay_start;
         call Node026();
@@ -581,7 +581,7 @@ begin
         end_dialogue;
     end
     else begin
-        if ((global_var(598) == 6) or (global_var(598) == 7)) then begin
+        if ((global_var(GVAR_TYPHON_QUEST_STATUS) == 6) or (global_var(GVAR_TYPHON_QUEST_STATUS) == 7)) then begin
             start_gdialog(1186, self_obj, 4, -1, -1);
             gsay_start;
             call Node024();
@@ -589,7 +589,7 @@ begin
             end_dialogue;
         end
         else begin
-            if ((global_var(598) == 4) and (local_var(8) != 0) and ((game_time - local_var(8)) >= (3 * (60 * (60 * 10)))) or (global_var(598) == 5)) then begin
+            if ((global_var(GVAR_TYPHON_QUEST_STATUS) == 4) and (local_var(8) != 0) and ((game_time - local_var(8)) >= (3 * (60 * (60 * 10)))) or (global_var(GVAR_TYPHON_QUEST_STATUS) == 5)) then begin
                 start_gdialog(1186, self_obj, 4, -1, -1);
                 gsay_start;
                 call Node021();
@@ -597,7 +597,7 @@ begin
                 end_dialogue;
             end
             else begin
-                if ((global_var(598) == 4) and (local_var(8) != 0) and ((game_time - local_var(8)) < (3 * (60 * (60 * 10))))) then begin
+                if ((global_var(GVAR_TYPHON_QUEST_STATUS) == 4) and (local_var(8) != 0) and ((game_time - local_var(8)) < (3 * (60 * (60 * 10))))) then begin
                     start_gdialog(1186, self_obj, 4, -1, -1);
                     gsay_start;
                     call Node020();
@@ -605,7 +605,7 @@ begin
                     end_dialogue;
                 end
                 else begin
-                    if ((global_var(598) == 2) and (local_var(9) != 0) and ((game_time - local_var(9)) >= (60 * (60 * 10))) or (global_var(598) == 3)) then begin
+                    if ((global_var(GVAR_TYPHON_QUEST_STATUS) == 2) and (local_var(9) != 0) and ((game_time - local_var(9)) >= (60 * (60 * 10))) or (global_var(GVAR_TYPHON_QUEST_STATUS) == 3)) then begin
                         start_gdialog(1186, self_obj, 4, -1, -1);
                         gsay_start;
                         call Node018();
@@ -613,7 +613,7 @@ begin
                         end_dialogue;
                     end
                     else begin
-                        if ((global_var(598) == 2) and (local_var(9) != 0) and ((game_time - local_var(9)) < (60 * (60 * 10)))) then begin
+                        if ((global_var(GVAR_TYPHON_QUEST_STATUS) == 2) and (local_var(9) != 0) and ((game_time - local_var(9)) < (60 * (60 * 10)))) then begin
                             start_gdialog(1186, self_obj, 4, -1, -1);
                             gsay_start;
                             call Node008();
@@ -621,7 +621,7 @@ begin
                             end_dialogue;
                         end
                         else begin
-                            if (global_var(598) == 1) then begin
+                            if (global_var(GVAR_TYPHON_QUEST_STATUS) == 1) then begin
                                 start_gdialog(1186, self_obj, 4, -1, -1);
                                 gsay_start;
                                 call Node016();
@@ -652,59 +652,59 @@ end
 procedure destroy_p_proc
 begin
     if (source_obj == dude_obj) then begin
-        set_global_var(5, global_var(5) + 1);
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(4) > (3 * global_var(5))) or (global_var(2) == 1)) and (global_var(1) == 0)) then begin
-            set_global_var(2, 1);
-            set_global_var(3, 0);
+        set_global_var(GVAR_GOOD_MONSTER, global_var(GVAR_GOOD_MONSTER) + 1);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(4) > (3 * global_var(GVAR_GOOD_MONSTER))) or (global_var(GVAR_CHAMPION_REPUTATION) == 1)) and (global_var(GVAR_CHILDKILLER_REPUTATION) == 0)) then begin
+            set_global_var(GVAR_CHAMPION_REPUTATION, 1);
+            set_global_var(GVAR_BERSERKER_REPUTATION, 0);
         end
-        if (metarule(51, self_obj) == 2) then begin
+        if (critter_kill_type(self_obj) == KILL_TYPE_children_kills) then begin
             if (source_obj == dude_obj) then begin
-                set_global_var(2, 0);
-                set_global_var(1, global_var(1) + 1);
-                set_global_var(0, global_var(0) + -15);
+                set_global_var(GVAR_CHAMPION_REPUTATION, 0);
+                set_global_var(GVAR_CHILDKILLER_REPUTATION, global_var(GVAR_CHILDKILLER_REPUTATION) + 1);
+                set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + -15);
                 debug_msg("Player gains " + -15 + " Karma Points.");
-                set_global_var(37, 0);
-                set_global_var(38, 0);
-                set_global_var(39, 0);
-                set_global_var(40, 0);
-                set_global_var(41, 0);
-                set_global_var(42, 0);
-                set_global_var(43, 0);
-                set_global_var(44, 0);
-                set_global_var(45, 0);
-                if (global_var(0) >= 1000) then begin
-                    set_global_var(37, 1);
+                set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+                set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+                set_global_var(GVAR_KARMA_DEFENDER, 0);
+                set_global_var(GVAR_KARMA_WANDERER, 0);
+                set_global_var(GVAR_KARMA_BETRAYER, 0);
+                set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+                set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+                set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+                    set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
                 end
                 else begin
-                    if (global_var(0) >= 750) then begin
-                        set_global_var(38, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
                     end
                     else begin
-                        if (global_var(0) >= 500) then begin
-                            set_global_var(39, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                            set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                         end
                         else begin
-                            if (global_var(0) >= 250) then begin
-                                set_global_var(40, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                                set_global_var(GVAR_KARMA_DEFENDER, 1);
                             end
                             else begin
-                                if (global_var(0) > -250) then begin
-                                    set_global_var(41, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                                    set_global_var(GVAR_KARMA_WANDERER, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -500) then begin
-                                        set_global_var(42, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                        set_global_var(GVAR_KARMA_BETRAYER, 1);
                                     end
                                     else begin
-                                        if (global_var(0) > -750) then begin
-                                            set_global_var(43, 1);
+                                        if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                            set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                         end
                                         else begin
-                                            if (global_var(0) > -1000) then begin
-                                                set_global_var(44, 1);
+                                            if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                                set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                             end
                                             else begin
-                                                set_global_var(45, 1);
+                                                set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                             end
                                         end
                                     end
@@ -713,58 +713,58 @@ begin
                         end
                     end
                 end
-                set_global_var(54, global_var(54) + -8);
+                set_global_var(GVAR_TOWN_REP_BROKEN_HILLS, global_var(GVAR_TOWN_REP_BROKEN_HILLS) + -8);
                 debug_msg("Added " + -8 + " to Town Rep");
             end
         end
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(5) > (2 * global_var(4))) or (global_var(3) == 1))) then begin
-            set_global_var(3, 1);
-            set_global_var(2, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
+            set_global_var(GVAR_BERSERKER_REPUTATION, 1);
+            set_global_var(GVAR_CHAMPION_REPUTATION, 0);
         end
-        set_global_var(0, global_var(0) + -10);
+        set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + -10);
         debug_msg("Player gains " + -10 + " Karma Points.");
-        set_global_var(37, 0);
-        set_global_var(38, 0);
-        set_global_var(39, 0);
-        set_global_var(40, 0);
-        set_global_var(41, 0);
-        set_global_var(42, 0);
-        set_global_var(43, 0);
-        set_global_var(44, 0);
-        set_global_var(45, 0);
-        if (global_var(0) >= 1000) then begin
-            set_global_var(37, 1);
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+        set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+        set_global_var(GVAR_KARMA_DEFENDER, 0);
+        set_global_var(GVAR_KARMA_WANDERER, 0);
+        set_global_var(GVAR_KARMA_BETRAYER, 0);
+        set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+        set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+            set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
         end
         else begin
-            if (global_var(0) >= 750) then begin
-                set_global_var(38, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
             end
             else begin
-                if (global_var(0) >= 500) then begin
-                    set_global_var(39, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                 end
                 else begin
-                    if (global_var(0) >= 250) then begin
-                        set_global_var(40, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                        set_global_var(GVAR_KARMA_DEFENDER, 1);
                     end
                     else begin
-                        if (global_var(0) > -250) then begin
-                            set_global_var(41, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                            set_global_var(GVAR_KARMA_WANDERER, 1);
                         end
                         else begin
-                            if (global_var(0) > -500) then begin
-                                set_global_var(42, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                set_global_var(GVAR_KARMA_BETRAYER, 1);
                             end
                             else begin
-                                if (global_var(0) > -750) then begin
-                                    set_global_var(43, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -1000) then begin
-                                        set_global_var(44, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                     end
                                     else begin
-                                        set_global_var(45, 1);
+                                        set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                     end
                                 end
                             end
@@ -773,7 +773,7 @@ begin
                 end
             end
         end
-        set_global_var(54, global_var(54) + -5);
+        set_global_var(GVAR_TOWN_REP_BROKEN_HILLS, global_var(GVAR_TOWN_REP_BROKEN_HILLS) + -5);
         debug_msg("Added " + -5 + " to Town Rep");
     end
 end
@@ -782,17 +782,17 @@ procedure look_at_p_proc
 begin
     script_overrides;
     if (local_var(4) == 0) then begin
-        display_msg(message_str(1186, 100));
+        display_mstr(100);
     end
     else begin
-        display_msg(message_str(1186, 101));
+        display_mstr(101);
     end
 end
 
 procedure description_p_proc
 begin
     script_overrides;
-    display_msg(message_str(1186, 102));
+    display_mstr(102);
 end
 
 procedure use_skill_on_p_proc
@@ -801,26 +801,26 @@ end
 
 procedure damage_p_proc
 begin
-    if (party_member_obj(obj_pid(source_obj)) != 0) then begin
+    if (obj_in_party(source_obj)) then begin
         set_local_var(6, 1);
-        set_global_var(309, 1);
+        set_global_var(GVAR_ENEMY_BROKEN_HILLS, 1);
     end
 end
 
 procedure map_enter_p_proc
 begin
     Only_Once := 0;
-    critter_add_trait(self_obj, 1, 6, 120);
-    critter_add_trait(self_obj, 1, 5, 33);
-    if (not(metarule(22, 0))) then begin
-        if ((global_var(598) >= 6) and (map_var(24) == 2) and (tile_num(self_obj) != 18482)) then begin
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 120);
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_COWARD);
+    if (not(is_loading_game)) then begin
+        if ((global_var(GVAR_TYPHON_QUEST_STATUS) >= 6) and (map_var(24) == 2) and (tile_num(self_obj) != 18482)) then begin
             move_to(self_obj, 18482, 0);
             critter_attempt_placement(self_obj, 18482, 0);
             anim(self_obj, 1000, 5);
             add_timer_event(self_obj, game_ticks(random(5, 10)), 0);
         end
         else begin
-            if ((global_var(598) >= 6) and (map_var(24) == 2) and (tile_num(self_obj) == 18482)) then begin
+            if ((global_var(GVAR_TYPHON_QUEST_STATUS) >= 6) and (map_var(24) == 2) and (tile_num(self_obj) == 18482)) then begin
                 add_timer_event(self_obj, game_ticks(random(5, 10)), 0);
             end
         end
@@ -832,20 +832,20 @@ begin
     if (fixed_param == 0) then begin
         current_dialog := random(0, 21);
         if ((current_dialog == 1) or (current_dialog == 2) or (current_dialog == 6) or (current_dialog == 8) or (current_dialog == 11) or (current_dialog == 14) or (current_dialog == 17)) then begin
-            float_msg(self_obj, message_str(1186, 300 + (current_dialog * 2)), 8);
+            float_msg(self_obj, mstr(300 + (current_dialog * 2)), FLOAT_MSG_YELLOW);
         end
         else begin
-            float_msg(seymour_obj, message_str(1186, 300 + (current_dialog * 2)), 3);
+            float_msg(seymour_obj, mstr(300 + (current_dialog * 2)), FLOAT_MSG_GREEN);
         end
         add_timer_event(self_obj, game_ticks(3), 1);
     end
     else begin
         if (fixed_param == 1) then begin
             if ((current_dialog == 1) or (current_dialog == 2) or (current_dialog == 6) or (current_dialog == 8) or (current_dialog == 11) or (current_dialog == 14) or (current_dialog == 17)) then begin
-                float_msg(seymour_obj, message_str(1186, 300 + (current_dialog * 2) + 1), 3);
+                float_msg(seymour_obj, mstr(300 + (current_dialog * 2) + 1), FLOAT_MSG_GREEN);
             end
             else begin
-                float_msg(self_obj, message_str(1186, 300 + (current_dialog * 2) + 1), 8);
+                float_msg(self_obj, mstr(300 + (current_dialog * 2) + 1), FLOAT_MSG_YELLOW);
             end
             add_timer_event(self_obj, game_ticks(random(10, 15)), 0);
         end
@@ -863,70 +863,70 @@ end
 
 procedure Node001
 begin
-    gsay_reply(1186, 103);
-    giq_option(-3, 1186, 104, Node002, 50);
-    giq_option(4, 1186, 105, Node012, 50);
-    giq_option(4, 1186, 106, Node013, 50);
-    giq_option(4, 1186, 107, Node014, 50);
-    giq_option(1, 1186, 108, Node999, 50);
+    Reply(103);
+    giq_option(-3, NAME, 104, Node002, NEUTRAL_REACTION);
+    giq_option(4, NAME, 105, Node012, NEUTRAL_REACTION);
+    giq_option(4, NAME, 106, Node013, NEUTRAL_REACTION);
+    giq_option(4, NAME, 107, Node014, NEUTRAL_REACTION);
+    giq_option(1, NAME, 108, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node002
 begin
-    gsay_reply(1186, 109);
-    giq_option(-3, 1186, 110, Node003, 50);
-    giq_option(-3, 1186, 111, Node004, 50);
+    Reply(109);
+    giq_option(-3, NAME, 110, Node003, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 111, Node004, NEUTRAL_REACTION);
 end
 
 procedure Node003
 begin
-    gsay_reply(1186, 112);
-    giq_option(-3, 1186, 113, Node005, 50);
-    giq_option(-3, 1186, 114, Node004, 50);
-    giq_option(4, 1186, 115, Node009, 50);
-    giq_option(4, 1186, 116, Node006, 50);
-    giq_option(4, 1186, 117, Node011, 50);
-    giq_option(1, 1186, 118, Node999, 50);
+    Reply(112);
+    giq_option(-3, NAME, 113, Node005, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 114, Node004, NEUTRAL_REACTION);
+    giq_option(4, NAME, 115, Node009, NEUTRAL_REACTION);
+    giq_option(4, NAME, 116, Node006, NEUTRAL_REACTION);
+    giq_option(4, NAME, 117, Node011, NEUTRAL_REACTION);
+    giq_option(1, NAME, 118, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node004
 begin
-    gsay_reply(1186, 119);
-    giq_option(-3, 1186, 120, Node999, 50);
-    giq_option(1, 1186, 120, Node999, 50);
+    Reply(119);
+    giq_option(-3, NAME, 120, Node999, NEUTRAL_REACTION);
+    giq_option(1, NAME, 120, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node005
 begin
-    gsay_reply(1186, 121);
-    giq_option(-3, 1186, 122, Node006, 50);
-    giq_option(-3, 1186, 123, Node004, 50);
-    giq_option(4, 1186, 124, Node006, 50);
-    giq_option(4, 1186, 125, Node004, 50);
+    Reply(121);
+    giq_option(-3, NAME, 122, Node006, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 123, Node004, NEUTRAL_REACTION);
+    giq_option(4, NAME, 124, Node006, NEUTRAL_REACTION);
+    giq_option(4, NAME, 125, Node004, NEUTRAL_REACTION);
 end
 
 procedure Node006
 begin
-    if (global_var(598) == 0) then begin
-        set_global_var(598, 1);
+    if (global_var(GVAR_TYPHON_QUEST_STATUS) == 0) then begin
+        set_global_var(GVAR_TYPHON_QUEST_STATUS, 1);
     end
-    gsay_reply(1186, 126);
+    Reply(126);
     if (obj_is_carrying_obj_pid(dude_obj, 225) > 0) then begin
-        giq_option(-3, 1186, 127, Node006a, 50);
+        giq_option(-3, NAME, 127, Node006a, NEUTRAL_REACTION);
     end
-    giq_option(-3, 1186, 128, Node999, 50);
-    giq_option(-3, 1186, 129, Node004, 50);
+    giq_option(-3, NAME, 128, Node999, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 129, Node004, NEUTRAL_REACTION);
     if (obj_is_carrying_obj_pid(dude_obj, 225) > 0) then begin
-        giq_option(4, 1186, 130, Node006a, 50);
+        giq_option(4, NAME, 130, Node006a, NEUTRAL_REACTION);
     end
-    giq_option(4, 1186, 131, Node999, 50);
-    giq_option(4, 1186, 132, Node004, 50);
+    giq_option(4, NAME, 131, Node999, NEUTRAL_REACTION);
+    giq_option(4, NAME, 132, Node004, NEUTRAL_REACTION);
 end
 
 procedure Node006a
 begin
     variable LVar0 := 0;
-    set_global_var(598, 2);
+    set_global_var(GVAR_TYPHON_QUEST_STATUS, 2);
     LVar0 := obj_carrying_pid_obj(dude_obj, 225);
     rm_obj_from_inven(dude_obj, LVar0);
     add_obj_to_inven(self_obj, LVar0);
@@ -936,114 +936,114 @@ end
 procedure Node007
 begin
     set_local_var(9, game_time);
-    gsay_reply(1186, 133);
-    giq_option(1, 1186, 134, Node999, 50);
+    Reply(133);
+    giq_option(1, NAME, 134, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node008
 begin
-    gsay_message(1186, 135, 50);
+    gsay_message(NAME, 135, NEUTRAL_REACTION);
 end
 
 procedure Node009
 begin
-    gsay_reply(1186, 137);
-    giq_option(4, 1186, 138, Node005, 50);
-    giq_option(4, 1186, 139, Node004, 50);
+    Reply(137);
+    giq_option(4, NAME, 138, Node005, NEUTRAL_REACTION);
+    giq_option(4, NAME, 139, Node004, NEUTRAL_REACTION);
 end
 
 procedure Node010
 begin
-    gsay_reply(1186, 140);
-    giq_option(-3, 1186, 141, Node005, 50);
-    giq_option(-3, 1186, 142, Node004, 50);
-    giq_option(4, 1186, 143, Node009, 50);
-    giq_option(4, 1186, 144, Node005, 50);
-    giq_option(4, 1186, 145, Node011, 50);
-    giq_option(4, 1186, 146, Node999, 50);
+    Reply(140);
+    giq_option(-3, NAME, 141, Node005, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 142, Node004, NEUTRAL_REACTION);
+    giq_option(4, NAME, 143, Node009, NEUTRAL_REACTION);
+    giq_option(4, NAME, 144, Node005, NEUTRAL_REACTION);
+    giq_option(4, NAME, 145, Node011, NEUTRAL_REACTION);
+    giq_option(4, NAME, 146, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node011
 begin
-    gsay_reply(1186, 147);
-    giq_option(4, 1186, 148, Node012, 50);
-    giq_option(4, 1186, 149, Node014, 50);
-    giq_option(4, 1186, 150, Node013, 50);
-    if (global_var(598) == 0) then begin
-        giq_option(4, 1186, 151, Node003, 50);
+    Reply(147);
+    giq_option(4, NAME, 148, Node012, NEUTRAL_REACTION);
+    giq_option(4, NAME, 149, Node014, NEUTRAL_REACTION);
+    giq_option(4, NAME, 150, Node013, NEUTRAL_REACTION);
+    if (global_var(GVAR_TYPHON_QUEST_STATUS) == 0) then begin
+        giq_option(4, NAME, 151, Node003, NEUTRAL_REACTION);
     end
 end
 
 procedure Node012
 begin
-    gsay_reply(1186, 152);
-    giq_option(4, 1186, 153, Node011, 50);
-    giq_option(4, 1186, 154, Node999, 50);
+    Reply(152);
+    giq_option(4, NAME, 153, Node011, NEUTRAL_REACTION);
+    giq_option(4, NAME, 154, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node013
 begin
-    gsay_reply(1186, 155);
-    giq_option(4, 1186, 156, Node015, 50);
-    giq_option(4, 1186, 157, Node015, 50);
-    giq_option(4, 1186, 158, Node011, 50);
-    giq_option(4, 1186, 159, Node999, 50);
+    Reply(155);
+    giq_option(4, NAME, 156, Node015, NEUTRAL_REACTION);
+    giq_option(4, NAME, 157, Node015, NEUTRAL_REACTION);
+    giq_option(4, NAME, 158, Node011, NEUTRAL_REACTION);
+    giq_option(4, NAME, 159, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node014
 begin
-    gsay_reply(1186, 160);
-    giq_option(4, 1186, 161, Node011, 50);
-    giq_option(4, 1186, 162, Node999, 50);
+    Reply(160);
+    giq_option(4, NAME, 161, Node011, NEUTRAL_REACTION);
+    giq_option(4, NAME, 162, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node015
 begin
-    gsay_reply(1186, 163);
-    giq_option(4, 1186, 164, Node011, 50);
-    giq_option(4, 1186, 165, Node999, 50);
+    Reply(163);
+    giq_option(4, NAME, 164, Node011, NEUTRAL_REACTION);
+    giq_option(4, NAME, 165, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node016
 begin
-    gsay_reply(1186, 166);
+    Reply(166);
     if (obj_is_carrying_obj_pid(dude_obj, 225) > 0) then begin
-        giq_option(-3, 1186, 167, Node006a, 50);
+        giq_option(-3, NAME, 167, Node006a, NEUTRAL_REACTION);
     end
-    giq_option(-3, 1186, 168, Node017, 50);
+    giq_option(-3, NAME, 168, Node017, NEUTRAL_REACTION);
     if (obj_is_carrying_obj_pid(dude_obj, 225) > 0) then begin
-        giq_option(4, 1186, 169, Node006a, 50);
+        giq_option(4, NAME, 169, Node006a, NEUTRAL_REACTION);
     end
-    giq_option(4, 1186, 170, Node011, 50);
-    giq_option(4, 1186, 171, Node017, 50);
+    giq_option(4, NAME, 170, Node011, NEUTRAL_REACTION);
+    giq_option(4, NAME, 171, Node017, NEUTRAL_REACTION);
 end
 
 procedure Node017
 begin
-    gsay_message(1186, 172, 50);
+    gsay_message(NAME, 172, NEUTRAL_REACTION);
 end
 
 procedure Node018
 begin
-    set_global_var(598, 3);
-    gsay_reply(1186, 174);
+    set_global_var(GVAR_TYPHON_QUEST_STATUS, 3);
+    Reply(174);
     if (obj_is_carrying_obj_pid(dude_obj, 508) > 0) then begin
-        giq_option(-3, 1186, 177, Node018a, 50);
+        giq_option(-3, NAME, 177, Node018a, NEUTRAL_REACTION);
     end
-    giq_option(-3, 1186, 176, Node004, 50);
-    giq_option(-3, 1186, 175, Node017, 50);
+    giq_option(-3, NAME, 176, Node004, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 175, Node017, NEUTRAL_REACTION);
     if (obj_is_carrying_obj_pid(dude_obj, 508) > 0) then begin
-        giq_option(4, 1186, 181, Node018a, 50);
+        giq_option(4, NAME, 181, Node018a, NEUTRAL_REACTION);
     end
-    giq_option(4, 1186, 178, Node011, 50);
-    giq_option(4, 1186, 179, Node017, 50);
-    giq_option(4, 1186, 180, Node004, 50);
+    giq_option(4, NAME, 178, Node011, NEUTRAL_REACTION);
+    giq_option(4, NAME, 179, Node017, NEUTRAL_REACTION);
+    giq_option(4, NAME, 180, Node004, NEUTRAL_REACTION);
 end
 
 procedure Node018a
 begin
     variable LVar0 := 0;
-    set_global_var(598, 4);
+    set_global_var(GVAR_TYPHON_QUEST_STATUS, 4);
     LVar0 := obj_carrying_pid_obj(dude_obj, 508);
     rm_obj_from_inven(dude_obj, LVar0);
     add_obj_to_inven(self_obj, LVar0);
@@ -1053,44 +1053,44 @@ end
 procedure Node019
 begin
     set_local_var(8, game_time);
-    gsay_message(1186, 182, 50);
+    gsay_message(NAME, 182, NEUTRAL_REACTION);
 end
 
 procedure Node020
 begin
-    gsay_message(1186, 184, 50);
+    gsay_message(NAME, 184, NEUTRAL_REACTION);
 end
 
 procedure Node021
 begin
     variable LVar0 := 0;
     variable LVar1 := 0;
-    set_global_var(598, 5);
+    set_global_var(GVAR_TYPHON_QUEST_STATUS, 5);
     LVar1 := obj_is_carrying_obj_pid(self_obj, 508);
     if (LVar1 > 0) then begin
         LVar0 := obj_carrying_pid_obj(self_obj, 508);
         rm_obj_from_inven(self_obj, LVar0);
-        LVar0 := create_object_sid(509, 0, 0, -1);
+        LVar0 := create_object(509, 0, 0);
         add_obj_to_inven(self_obj, LVar0);
     end
-    gsay_reply(1186, 186);
+    Reply(186);
     if (obj_is_carrying_obj_pid(dude_obj, 469) > 0) then begin
-        giq_option(-3, 1186, 189, Node021a, 50);
+        giq_option(-3, NAME, 189, Node021a, NEUTRAL_REACTION);
     end
-    giq_option(-3, 1186, 188, Node004, 50);
-    giq_option(-3, 1186, 187, Node999, 50);
+    giq_option(-3, NAME, 188, Node004, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 187, Node999, NEUTRAL_REACTION);
     if (obj_is_carrying_obj_pid(dude_obj, 469) > 0) then begin
-        giq_option(4, 1186, 190, Node021a, 50);
+        giq_option(4, NAME, 190, Node021a, NEUTRAL_REACTION);
     end
-    giq_option(4, 1186, 191, Node022, 50);
-    giq_option(4, 1186, 193, Node004, 50);
-    giq_option(4, 1186, 192, Node999, 50);
+    giq_option(4, NAME, 191, Node022, NEUTRAL_REACTION);
+    giq_option(4, NAME, 193, Node004, NEUTRAL_REACTION);
+    giq_option(4, NAME, 192, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node021a
 begin
     variable LVar0 := 0;
-    set_global_var(598, 6);
+    set_global_var(GVAR_TYPHON_QUEST_STATUS, 6);
     LVar0 := obj_carrying_pid_obj(dude_obj, 469);
     rm_obj_from_inven(dude_obj, LVar0);
     add_obj_to_inven(self_obj, LVar0);
@@ -1099,60 +1099,60 @@ end
 
 procedure Node022
 begin
-    gsay_reply(1186, 194);
-    giq_option(4, 1186, 195, Node999, 50);
-    giq_option(4, 1186, 196, Node004, 50);
+    Reply(194);
+    giq_option(4, NAME, 195, Node999, NEUTRAL_REACTION);
+    giq_option(4, NAME, 196, Node004, NEUTRAL_REACTION);
 end
 
 procedure Node023
 begin
-    display_msg(message_str(1186, 224));
+    display_mstr(224);
     give_exp_points(2000);
-    display_msg(message_str(14, 100) + 2000 + message_str(14, 101));
-    set_global_var(0, global_var(0) + 2);
+    display_msg(g_mstr(100) + 2000 + g_mstr(101));
+    set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + 2);
     debug_msg("Player gains " + 2 + " Karma Points.");
-    set_global_var(37, 0);
-    set_global_var(38, 0);
-    set_global_var(39, 0);
-    set_global_var(40, 0);
-    set_global_var(41, 0);
-    set_global_var(42, 0);
-    set_global_var(43, 0);
-    set_global_var(44, 0);
-    set_global_var(45, 0);
-    if (global_var(0) >= 1000) then begin
-        set_global_var(37, 1);
+    set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+    set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+    set_global_var(GVAR_KARMA_DEFENDER, 0);
+    set_global_var(GVAR_KARMA_WANDERER, 0);
+    set_global_var(GVAR_KARMA_BETRAYER, 0);
+    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+    set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+    set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+    if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
     end
     else begin
-        if (global_var(0) >= 750) then begin
-            set_global_var(38, 1);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+            set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
         end
         else begin
-            if (global_var(0) >= 500) then begin
-                set_global_var(39, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
             end
             else begin
-                if (global_var(0) >= 250) then begin
-                    set_global_var(40, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                    set_global_var(GVAR_KARMA_DEFENDER, 1);
                 end
                 else begin
-                    if (global_var(0) > -250) then begin
-                        set_global_var(41, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                        set_global_var(GVAR_KARMA_WANDERER, 1);
                     end
                     else begin
-                        if (global_var(0) > -500) then begin
-                            set_global_var(42, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                            set_global_var(GVAR_KARMA_BETRAYER, 1);
                         end
                         else begin
-                            if (global_var(0) > -750) then begin
-                                set_global_var(43, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                             end
                             else begin
-                                if (global_var(0) > -1000) then begin
-                                    set_global_var(44, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                    set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                 end
                                 else begin
-                                    set_global_var(45, 1);
+                                    set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                 end
                             end
                         end
@@ -1161,113 +1161,113 @@ begin
             end
         end
     end
-    set_global_var(54, global_var(54) + 5);
-    gsay_reply(1186, 197);
-    giq_option(-3, 1186, 221, Node023a, 50);
-    giq_option(4, 1186, 222, Node023a, 50);
+    set_global_var(GVAR_TOWN_REP_BROKEN_HILLS, global_var(GVAR_TOWN_REP_BROKEN_HILLS) + 5);
+    Reply(197);
+    giq_option(-3, NAME, 221, Node023a, NEUTRAL_REACTION);
+    giq_option(4, NAME, 222, Node023a, NEUTRAL_REACTION);
 end
 
 procedure Node023a
 begin
-    gsay_reply(1186, 223);
-    giq_option(-3, 1186, 198, Node999, 50);
-    giq_option(4, 1186, 199, Node011, 50);
-    giq_option(4, 1186, 200, Node999, 50);
+    Reply(223);
+    giq_option(-3, NAME, 198, Node999, NEUTRAL_REACTION);
+    giq_option(4, NAME, 199, Node011, NEUTRAL_REACTION);
+    giq_option(4, NAME, 200, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node024
 begin
-    gsay_reply(1186, 201);
-    giq_option(-3, 1186, 202, Node025, 50);
-    giq_option(4, 1186, 203, Node025, 50);
-    giq_option(4, 1186, 204, Node011, 50);
-    giq_option(4, 1186, 205, Node999, 50);
+    Reply(201);
+    giq_option(-3, NAME, 202, Node025, NEUTRAL_REACTION);
+    giq_option(4, NAME, 203, Node025, NEUTRAL_REACTION);
+    giq_option(4, NAME, 204, Node011, NEUTRAL_REACTION);
+    giq_option(4, NAME, 205, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node025
 begin
-    gsay_reply(1186, 206);
-    giq_option(-3, 1186, 207, Node999, 50);
-    giq_option(4, 1186, 208, Node999, 50);
+    Reply(206);
+    giq_option(-3, NAME, 207, Node999, NEUTRAL_REACTION);
+    giq_option(4, NAME, 208, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node026
 begin
-    gsay_reply(1186, 209);
+    Reply(209);
     if (local_var(10) == 0) then begin
-        giq_option(-3, 1186, 210, Node027, 50);
+        giq_option(-3, NAME, 210, Node027, NEUTRAL_REACTION);
     end
     else begin
-        giq_option(-3, 1186, 214, Node999, 50);
+        giq_option(-3, NAME, 214, Node999, NEUTRAL_REACTION);
     end
     if (local_var(10) == 0) then begin
-        giq_option(4, 1186, 211, Node027, 50);
+        giq_option(4, NAME, 211, Node027, NEUTRAL_REACTION);
     end
     else begin
-        giq_option(4, 1186, 212, Node011, 50);
-        giq_option(4, 1186, 213, Node999, 50);
+        giq_option(4, NAME, 212, Node011, NEUTRAL_REACTION);
+        giq_option(4, NAME, 213, Node999, NEUTRAL_REACTION);
     end
 end
 
 procedure Node027
 begin
     set_local_var(10, 1);
-    gsay_reply(1186, 215);
-    giq_option(-3, 1186, 216, Node999, 50);
-    giq_option(4, 1186, 217, Node998, 50);
-    giq_option(4, 1186, 218, Node027a, 50);
-    giq_option(4, 1186, 219, Node999, 50);
-    giq_option(4, 1186, 220, Node027b, 50);
+    Reply(215);
+    giq_option(-3, NAME, 216, Node999, NEUTRAL_REACTION);
+    giq_option(4, NAME, 217, Node998, NEUTRAL_REACTION);
+    giq_option(4, NAME, 218, Node027a, NEUTRAL_REACTION);
+    giq_option(4, NAME, 219, Node999, NEUTRAL_REACTION);
+    giq_option(4, NAME, 220, Node027b, NEUTRAL_REACTION);
 end
 
 procedure Node027a
 begin
     if (local_var(11) == 0) then begin
         set_local_var(11, 1);
-        set_global_var(0, global_var(0) + 2);
+        set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + 2);
         debug_msg("Player gains " + 2 + " Karma Points.");
-        set_global_var(37, 0);
-        set_global_var(38, 0);
-        set_global_var(39, 0);
-        set_global_var(40, 0);
-        set_global_var(41, 0);
-        set_global_var(42, 0);
-        set_global_var(43, 0);
-        set_global_var(44, 0);
-        set_global_var(45, 0);
-        if (global_var(0) >= 1000) then begin
-            set_global_var(37, 1);
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+        set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+        set_global_var(GVAR_KARMA_DEFENDER, 0);
+        set_global_var(GVAR_KARMA_WANDERER, 0);
+        set_global_var(GVAR_KARMA_BETRAYER, 0);
+        set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+        set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+            set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
         end
         else begin
-            if (global_var(0) >= 750) then begin
-                set_global_var(38, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
             end
             else begin
-                if (global_var(0) >= 500) then begin
-                    set_global_var(39, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                 end
                 else begin
-                    if (global_var(0) >= 250) then begin
-                        set_global_var(40, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                        set_global_var(GVAR_KARMA_DEFENDER, 1);
                     end
                     else begin
-                        if (global_var(0) > -250) then begin
-                            set_global_var(41, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                            set_global_var(GVAR_KARMA_WANDERER, 1);
                         end
                         else begin
-                            if (global_var(0) > -500) then begin
-                                set_global_var(42, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                set_global_var(GVAR_KARMA_BETRAYER, 1);
                             end
                             else begin
-                                if (global_var(0) > -750) then begin
-                                    set_global_var(43, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -1000) then begin
-                                        set_global_var(44, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                     end
                                     else begin
-                                        set_global_var(45, 1);
+                                        set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                     end
                                 end
                             end
@@ -1284,50 +1284,50 @@ procedure Node027b
 begin
     if (local_var(11) == 0) then begin
         set_local_var(11, 1);
-        set_global_var(0, global_var(0) + 2);
+        set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + 2);
         debug_msg("Player gains " + 2 + " Karma Points.");
-        set_global_var(37, 0);
-        set_global_var(38, 0);
-        set_global_var(39, 0);
-        set_global_var(40, 0);
-        set_global_var(41, 0);
-        set_global_var(42, 0);
-        set_global_var(43, 0);
-        set_global_var(44, 0);
-        set_global_var(45, 0);
-        if (global_var(0) >= 1000) then begin
-            set_global_var(37, 1);
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+        set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+        set_global_var(GVAR_KARMA_DEFENDER, 0);
+        set_global_var(GVAR_KARMA_WANDERER, 0);
+        set_global_var(GVAR_KARMA_BETRAYER, 0);
+        set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+        set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+            set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
         end
         else begin
-            if (global_var(0) >= 750) then begin
-                set_global_var(38, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
             end
             else begin
-                if (global_var(0) >= 500) then begin
-                    set_global_var(39, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                 end
                 else begin
-                    if (global_var(0) >= 250) then begin
-                        set_global_var(40, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                        set_global_var(GVAR_KARMA_DEFENDER, 1);
                     end
                     else begin
-                        if (global_var(0) > -250) then begin
-                            set_global_var(41, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                            set_global_var(GVAR_KARMA_WANDERER, 1);
                         end
                         else begin
-                            if (global_var(0) > -500) then begin
-                                set_global_var(42, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                set_global_var(GVAR_KARMA_BETRAYER, 1);
                             end
                             else begin
-                                if (global_var(0) > -750) then begin
-                                    set_global_var(43, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -1000) then begin
-                                        set_global_var(44, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                     end
                                     else begin
-                                        set_global_var(45, 1);
+                                        set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                     end
                                 end
                             end
@@ -1338,4 +1338,3 @@ begin
         end
     end
 end
-
