diff --git a/Games/Fallout2.Steam/Original/HICAVEIN.ssl b/Games/Fallout2.Steam/ReDefine/HICAVEIN.ssl
index fa747d6..0afc5c8 100644
--- a/Games/Fallout2.Steam/Original/HICAVEIN.ssl
+++ b/Games/Fallout2.Steam/ReDefine/HICAVEIN.ssl
@@ -143,13 +143,13 @@ end
 procedure look_at_p_proc
 begin
     script_overrides;
-    display_msg(message_str(1069, 100));
+    display_mstr(100);
 end
 
 procedure description_p_proc
 begin
     script_overrides;
-    display_msg(message_str(1069, 101));
+    display_mstr(101);
 end
 
 procedure map_enter_p_proc
@@ -161,4 +161,3 @@ procedure map_update_p_proc
 begin
     broken_hills_cavein := self_obj;
 end
-
