diff --git a/Games/Fallout2.Steam/Original/HICONSOL.ssl b/Games/Fallout2.Steam/ReDefine/HICONSOL.ssl
index 731ed1e..03da0fb 100644
--- a/Games/Fallout2.Steam/Original/HICONSOL.ssl
+++ b/Games/Fallout2.Steam/ReDefine/HICONSOL.ssl
@@ -140,27 +140,27 @@ end
 procedure look_at_p_proc
 begin
     script_overrides;
-    display_msg(message_str(1134, 100));
+    display_mstr(100);
 end
 
 procedure description_p_proc
 begin
     script_overrides;
-    display_msg(message_str(1134, 101));
+    display_mstr(101);
 end
 
 procedure use_p_proc
 begin
     script_overrides;
-    if (global_var(542) <= 2) then begin
-        display_msg(message_str(1134, 102));
+    if (global_var(GVAR_BH_POWER) <= 2) then begin
+        display_mstr(102);
     end
     else begin
-        if ((global_var(542) > 2) and (global_var(542) < 6)) then begin
-            display_msg(message_str(1134, 103));
+        if ((global_var(GVAR_BH_POWER) > 2) and (global_var(GVAR_BH_POWER) < 6)) then begin
+            display_mstr(103);
         end
         else begin
-            display_msg(message_str(1134, 104));
+            display_mstr(104);
         end
     end
 end
@@ -171,64 +171,64 @@ begin
     variable LVar1 := 0;
     LVar0 := action_being_used;
     if (LVar0 == 12) then begin
-        if (global_var(542) <= 2) then begin
+        if (global_var(GVAR_BH_POWER) <= 2) then begin
             script_overrides;
-            display_msg(message_str(1134, 105));
+            display_mstr(105);
         end
         else begin
-            if ((global_var(542) > 2) and (global_var(542) < 6)) then begin
+            if ((global_var(GVAR_BH_POWER) > 2) and (global_var(GVAR_BH_POWER) < 6)) then begin
                 LVar1 := roll_vs_skill(source_obj, 12, -30);
                 if (is_success(LVar1)) then begin
                     if (is_critical(LVar1)) then begin
                         script_overrides;
-                        set_global_var(542, 6);
-                        display_msg(message_str(1134, 109));
+                        set_global_var(GVAR_BH_POWER, 6);
+                        display_mstr(109);
                         give_exp_points(1500);
-                        display_msg(message_str(14, 100) + 1500 + message_str(14, 101));
-                        set_global_var(0, global_var(0) + 10);
+                        display_msg(g_mstr(100) + 1500 + g_mstr(101));
+                        set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + 10);
                         debug_msg("Player gains " + 10 + " Karma Points.");
-                        set_global_var(37, 0);
-                        set_global_var(38, 0);
-                        set_global_var(39, 0);
-                        set_global_var(40, 0);
-                        set_global_var(41, 0);
-                        set_global_var(42, 0);
-                        set_global_var(43, 0);
-                        set_global_var(44, 0);
-                        set_global_var(45, 0);
-                        if (global_var(0) >= 1000) then begin
-                            set_global_var(37, 1);
+                        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+                        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+                        set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+                        set_global_var(GVAR_KARMA_DEFENDER, 0);
+                        set_global_var(GVAR_KARMA_WANDERER, 0);
+                        set_global_var(GVAR_KARMA_BETRAYER, 0);
+                        set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+                        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+                        set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+                        if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+                            set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
                         end
                         else begin
-                            if (global_var(0) >= 750) then begin
-                                set_global_var(38, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
                             end
                             else begin
-                                if (global_var(0) >= 500) then begin
-                                    set_global_var(39, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                                    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                                 end
                                 else begin
-                                    if (global_var(0) >= 250) then begin
-                                        set_global_var(40, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                                        set_global_var(GVAR_KARMA_DEFENDER, 1);
                                     end
                                     else begin
-                                        if (global_var(0) > -250) then begin
-                                            set_global_var(41, 1);
+                                        if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                                            set_global_var(GVAR_KARMA_WANDERER, 1);
                                         end
                                         else begin
-                                            if (global_var(0) > -500) then begin
-                                                set_global_var(42, 1);
+                                            if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                                set_global_var(GVAR_KARMA_BETRAYER, 1);
                                             end
                                             else begin
-                                                if (global_var(0) > -750) then begin
-                                                    set_global_var(43, 1);
+                                                if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                                    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                                 end
                                                 else begin
-                                                    if (global_var(0) > -1000) then begin
-                                                        set_global_var(44, 1);
+                                                    if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                                        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                                     end
                                                     else begin
-                                                        set_global_var(45, 1);
+                                                        set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                                     end
                                                 end
                                             end
@@ -237,34 +237,33 @@ begin
                                 end
                             end
                         end
-                        set_global_var(54, global_var(54) + 4);
+                        set_global_var(GVAR_TOWN_REP_BROKEN_HILLS, global_var(GVAR_TOWN_REP_BROKEN_HILLS) + 4);
                     end
                     else begin
                         script_overrides;
-                        set_global_var(542, 6);
-                        display_msg(message_str(1134, 108));
+                        set_global_var(GVAR_BH_POWER, 6);
+                        display_mstr(108);
                         give_exp_points(1000);
-                        display_msg(message_str(14, 100) + 1000 + message_str(14, 101));
-                        set_global_var(54, global_var(54) + 2);
+                        display_msg(g_mstr(100) + 1000 + g_mstr(101));
+                        set_global_var(GVAR_TOWN_REP_BROKEN_HILLS, global_var(GVAR_TOWN_REP_BROKEN_HILLS) + 2);
                     end
                 end
                 else begin
                     if (is_critical(LVar1)) then begin
                         script_overrides;
-                        display_msg(message_str(1134, 107));
-                        critter_dmg(source_obj, random(10, 15), 4);
+                        display_mstr(107);
+                        critter_dmg(source_obj, random(10, 15), DMG_electrical);
                     end
                     else begin
                         script_overrides;
-                        display_msg(message_str(1134, 106));
+                        display_mstr(106);
                     end
                 end
             end
             else begin
                 script_overrides;
-                display_msg(message_str(1134, 110));
+                display_mstr(110);
             end
         end
     end
 end
-
