diff --git a/Games/Fallout2.Steam/Original/HIDOCBOX.ssl b/Games/Fallout2.Steam/ReDefine/HIDOCBOX.ssl
index b09690b..cad0934 100644
--- a/Games/Fallout2.Steam/Original/HIDOCBOX.ssl
+++ b/Games/Fallout2.Steam/ReDefine/HIDOCBOX.ssl
@@ -141,18 +141,18 @@ end
 procedure map_enter_p_proc
 begin
     variable LVar0 := 0;
-    if (metarule(14, 0)) then begin
-        LVar0 := create_object_sid(80, 0, 0, -1);
+    if (map_first_run) then begin
+        LVar0 := create_object(80, 0, 0);
         add_obj_to_inven(self_obj, LVar0);
     end
     broken_hills_doc_box := self_obj;
-    if (metarule(22, 0) == 0) then begin
+    if (is_loading_game == 0) then begin
         if (local_var(0) < game_time) then begin
             restock_amt := random(200, 750);
             if (obj_is_carrying_obj_pid(self_obj, 41) < restock_amt) then begin
                 if (100 >= random(1, 100)) then begin
                     if (obj_is_carrying_obj_pid(self_obj, 41) < restock_amt) then begin
-                        restock_obj := create_object_sid(41, 0, 0, -1);
+                        restock_obj := create_object(41, 0, 0);
                         add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 41));
                     end
                     else begin
@@ -175,7 +175,7 @@ begin
                                     end
                                     else begin
                                         if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                            metarule(43, self_obj);
+                                            inven_unwield(self_obj);
                                         end
                                     end
                                 end
@@ -193,7 +193,7 @@ begin
             end
             else begin
                 if (obj_is_carrying_obj_pid(self_obj, 41) < restock_amt) then begin
-                    restock_obj := create_object_sid(41, 0, 0, -1);
+                    restock_obj := create_object(41, 0, 0);
                     add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 41));
                 end
                 else begin
@@ -216,7 +216,7 @@ begin
                                 end
                                 else begin
                                     if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                        metarule(43, self_obj);
+                                        inven_unwield(self_obj);
                                     end
                                 end
                             end
@@ -235,7 +235,7 @@ begin
             if (obj_is_carrying_obj_pid(self_obj, 47) < restock_amt) then begin
                 if (25 >= random(1, 100)) then begin
                     if (obj_is_carrying_obj_pid(self_obj, 47) < restock_amt) then begin
-                        restock_obj := create_object_sid(47, 0, 0, -1);
+                        restock_obj := create_object(47, 0, 0);
                         add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 47));
                     end
                     else begin
@@ -258,7 +258,7 @@ begin
                                     end
                                     else begin
                                         if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                            metarule(43, self_obj);
+                                            inven_unwield(self_obj);
                                         end
                                     end
                                 end
@@ -276,7 +276,7 @@ begin
             end
             else begin
                 if (obj_is_carrying_obj_pid(self_obj, 47) < restock_amt) then begin
-                    restock_obj := create_object_sid(47, 0, 0, -1);
+                    restock_obj := create_object(47, 0, 0);
                     add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 47));
                 end
                 else begin
@@ -299,7 +299,7 @@ begin
                                 end
                                 else begin
                                     if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                        metarule(43, self_obj);
+                                        inven_unwield(self_obj);
                                     end
                                 end
                             end
@@ -318,7 +318,7 @@ begin
             if (obj_is_carrying_obj_pid(self_obj, 273) < restock_amt) then begin
                 if (40 >= random(1, 100)) then begin
                     if (obj_is_carrying_obj_pid(self_obj, 273) < restock_amt) then begin
-                        restock_obj := create_object_sid(273, 0, 0, -1);
+                        restock_obj := create_object(273, 0, 0);
                         add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 273));
                     end
                     else begin
@@ -341,7 +341,7 @@ begin
                                     end
                                     else begin
                                         if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                            metarule(43, self_obj);
+                                            inven_unwield(self_obj);
                                         end
                                     end
                                 end
@@ -359,7 +359,7 @@ begin
             end
             else begin
                 if (obj_is_carrying_obj_pid(self_obj, 273) < restock_amt) then begin
-                    restock_obj := create_object_sid(273, 0, 0, -1);
+                    restock_obj := create_object(273, 0, 0);
                     add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 273));
                 end
                 else begin
@@ -382,7 +382,7 @@ begin
                                 end
                                 else begin
                                     if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                        metarule(43, self_obj);
+                                        inven_unwield(self_obj);
                                     end
                                 end
                             end
@@ -401,7 +401,7 @@ begin
             if (obj_is_carrying_obj_pid(self_obj, 311) < restock_amt) then begin
                 if (80 >= random(1, 100)) then begin
                     if (obj_is_carrying_obj_pid(self_obj, 311) < restock_amt) then begin
-                        restock_obj := create_object_sid(311, 0, 0, -1);
+                        restock_obj := create_object(311, 0, 0);
                         add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 311));
                     end
                     else begin
@@ -424,7 +424,7 @@ begin
                                     end
                                     else begin
                                         if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                            metarule(43, self_obj);
+                                            inven_unwield(self_obj);
                                         end
                                     end
                                 end
@@ -442,7 +442,7 @@ begin
             end
             else begin
                 if (obj_is_carrying_obj_pid(self_obj, 311) < restock_amt) then begin
-                    restock_obj := create_object_sid(311, 0, 0, -1);
+                    restock_obj := create_object(311, 0, 0);
                     add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 311));
                 end
                 else begin
@@ -465,7 +465,7 @@ begin
                                 end
                                 else begin
                                     if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                        metarule(43, self_obj);
+                                        inven_unwield(self_obj);
                                     end
                                 end
                             end
@@ -484,7 +484,7 @@ begin
             if (obj_is_carrying_obj_pid(self_obj, 40) < restock_amt) then begin
                 if (70 >= random(1, 100)) then begin
                     if (obj_is_carrying_obj_pid(self_obj, 40) < restock_amt) then begin
-                        restock_obj := create_object_sid(40, 0, 0, -1);
+                        restock_obj := create_object(40, 0, 0);
                         add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 40));
                     end
                     else begin
@@ -507,7 +507,7 @@ begin
                                     end
                                     else begin
                                         if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                            metarule(43, self_obj);
+                                            inven_unwield(self_obj);
                                         end
                                     end
                                 end
@@ -525,7 +525,7 @@ begin
             end
             else begin
                 if (obj_is_carrying_obj_pid(self_obj, 40) < restock_amt) then begin
-                    restock_obj := create_object_sid(40, 0, 0, -1);
+                    restock_obj := create_object(40, 0, 0);
                     add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 40));
                 end
                 else begin
@@ -548,7 +548,7 @@ begin
                                 end
                                 else begin
                                     if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                        metarule(43, self_obj);
+                                        inven_unwield(self_obj);
                                     end
                                 end
                             end
@@ -567,7 +567,7 @@ begin
             if (obj_is_carrying_obj_pid(self_obj, 144) < restock_amt) then begin
                 if (50 >= random(1, 100)) then begin
                     if (obj_is_carrying_obj_pid(self_obj, 144) < restock_amt) then begin
-                        restock_obj := create_object_sid(144, 0, 0, -1);
+                        restock_obj := create_object(144, 0, 0);
                         add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 144));
                     end
                     else begin
@@ -590,7 +590,7 @@ begin
                                     end
                                     else begin
                                         if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                            metarule(43, self_obj);
+                                            inven_unwield(self_obj);
                                         end
                                     end
                                 end
@@ -608,7 +608,7 @@ begin
             end
             else begin
                 if (obj_is_carrying_obj_pid(self_obj, 144) < restock_amt) then begin
-                    restock_obj := create_object_sid(144, 0, 0, -1);
+                    restock_obj := create_object(144, 0, 0);
                     add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 144));
                 end
                 else begin
@@ -631,7 +631,7 @@ begin
                                 end
                                 else begin
                                     if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                        metarule(43, self_obj);
+                                        inven_unwield(self_obj);
                                     end
                                 end
                             end
@@ -650,7 +650,7 @@ begin
             if (obj_is_carrying_obj_pid(self_obj, 109) < restock_amt) then begin
                 if (75 >= random(1, 100)) then begin
                     if (obj_is_carrying_obj_pid(self_obj, 109) < restock_amt) then begin
-                        restock_obj := create_object_sid(109, 0, 0, -1);
+                        restock_obj := create_object(109, 0, 0);
                         add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 109));
                     end
                     else begin
@@ -673,7 +673,7 @@ begin
                                     end
                                     else begin
                                         if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                            metarule(43, self_obj);
+                                            inven_unwield(self_obj);
                                         end
                                     end
                                 end
@@ -691,7 +691,7 @@ begin
             end
             else begin
                 if (obj_is_carrying_obj_pid(self_obj, 109) < restock_amt) then begin
-                    restock_obj := create_object_sid(109, 0, 0, -1);
+                    restock_obj := create_object(109, 0, 0);
                     add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 109));
                 end
                 else begin
@@ -714,7 +714,7 @@ begin
                                 end
                                 else begin
                                     if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                        metarule(43, self_obj);
+                                        inven_unwield(self_obj);
                                     end
                                 end
                             end
@@ -733,7 +733,7 @@ begin
             if (obj_is_carrying_obj_pid(self_obj, 48) < restock_amt) then begin
                 if (65 >= random(1, 100)) then begin
                     if (obj_is_carrying_obj_pid(self_obj, 48) < restock_amt) then begin
-                        restock_obj := create_object_sid(48, 0, 0, -1);
+                        restock_obj := create_object(48, 0, 0);
                         add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 48));
                     end
                     else begin
@@ -756,7 +756,7 @@ begin
                                     end
                                     else begin
                                         if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                            metarule(43, self_obj);
+                                            inven_unwield(self_obj);
                                         end
                                     end
                                 end
@@ -774,7 +774,7 @@ begin
             end
             else begin
                 if (obj_is_carrying_obj_pid(self_obj, 48) < restock_amt) then begin
-                    restock_obj := create_object_sid(48, 0, 0, -1);
+                    restock_obj := create_object(48, 0, 0);
                     add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 48));
                 end
                 else begin
@@ -797,7 +797,7 @@ begin
                                 end
                                 else begin
                                     if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                        metarule(43, self_obj);
+                                        inven_unwield(self_obj);
                                     end
                                 end
                             end
@@ -821,4 +821,3 @@ procedure map_update_p_proc
 begin
     broken_hills_doc_box := self_obj;
 end
-
