diff --git a/Games/Fallout2.Steam/Original/HIJACBOX.ssl b/Games/Fallout2.Steam/ReDefine/HIJACBOX.ssl
index 46eed0d..a3cf8b8 100644
--- a/Games/Fallout2.Steam/Original/HIJACBOX.ssl
+++ b/Games/Fallout2.Steam/ReDefine/HIJACBOX.ssl
@@ -141,13 +141,13 @@ end
 procedure map_enter_p_proc
 begin
     broken_hills_jacob_box := self_obj;
-    if (metarule(22, 0) == 0) then begin
+    if (is_loading_game == 0) then begin
         if (local_var(0) < game_time) then begin
             restock_amt := random(200, 1250);
             if (obj_is_carrying_obj_pid(self_obj, 41) < restock_amt) then begin
                 if (100 >= random(1, 100)) then begin
                     if (obj_is_carrying_obj_pid(self_obj, 41) < restock_amt) then begin
-                        restock_obj := create_object_sid(41, 0, 0, -1);
+                        restock_obj := create_object(41, 0, 0);
                         add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 41));
                     end
                     else begin
@@ -170,7 +170,7 @@ begin
                                     end
                                     else begin
                                         if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                            metarule(43, self_obj);
+                                            inven_unwield(self_obj);
                                         end
                                     end
                                 end
@@ -188,7 +188,7 @@ begin
             end
             else begin
                 if (obj_is_carrying_obj_pid(self_obj, 41) < restock_amt) then begin
-                    restock_obj := create_object_sid(41, 0, 0, -1);
+                    restock_obj := create_object(41, 0, 0);
                     add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 41));
                 end
                 else begin
@@ -211,7 +211,7 @@ begin
                                 end
                                 else begin
                                     if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                        metarule(43, self_obj);
+                                        inven_unwield(self_obj);
                                     end
                                 end
                             end
@@ -230,7 +230,7 @@ begin
             if (obj_is_carrying_obj_pid(self_obj, 87) < restock_amt) then begin
                 if (25 >= random(1, 100)) then begin
                     if (obj_is_carrying_obj_pid(self_obj, 87) < restock_amt) then begin
-                        restock_obj := create_object_sid(87, 0, 0, -1);
+                        restock_obj := create_object(87, 0, 0);
                         add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 87));
                     end
                     else begin
@@ -253,7 +253,7 @@ begin
                                     end
                                     else begin
                                         if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                            metarule(43, self_obj);
+                                            inven_unwield(self_obj);
                                         end
                                     end
                                 end
@@ -271,7 +271,7 @@ begin
             end
             else begin
                 if (obj_is_carrying_obj_pid(self_obj, 87) < restock_amt) then begin
-                    restock_obj := create_object_sid(87, 0, 0, -1);
+                    restock_obj := create_object(87, 0, 0);
                     add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 87));
                 end
                 else begin
@@ -294,7 +294,7 @@ begin
                                 end
                                 else begin
                                     if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                        metarule(43, self_obj);
+                                        inven_unwield(self_obj);
                                     end
                                 end
                             end
@@ -313,7 +313,7 @@ begin
             if (obj_is_carrying_obj_pid(self_obj, 273) < restock_amt) then begin
                 if (80 >= random(1, 100)) then begin
                     if (obj_is_carrying_obj_pid(self_obj, 273) < restock_amt) then begin
-                        restock_obj := create_object_sid(273, 0, 0, -1);
+                        restock_obj := create_object(273, 0, 0);
                         add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 273));
                     end
                     else begin
@@ -336,7 +336,7 @@ begin
                                     end
                                     else begin
                                         if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                            metarule(43, self_obj);
+                                            inven_unwield(self_obj);
                                         end
                                     end
                                 end
@@ -354,7 +354,7 @@ begin
             end
             else begin
                 if (obj_is_carrying_obj_pid(self_obj, 273) < restock_amt) then begin
-                    restock_obj := create_object_sid(273, 0, 0, -1);
+                    restock_obj := create_object(273, 0, 0);
                     add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 273));
                 end
                 else begin
@@ -377,7 +377,7 @@ begin
                                 end
                                 else begin
                                     if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                        metarule(43, self_obj);
+                                        inven_unwield(self_obj);
                                     end
                                 end
                             end
@@ -396,7 +396,7 @@ begin
             if (obj_is_carrying_obj_pid(self_obj, 259) < restock_amt) then begin
                 if (50 >= random(1, 100)) then begin
                     if (obj_is_carrying_obj_pid(self_obj, 259) < restock_amt) then begin
-                        restock_obj := create_object_sid(259, 0, 0, -1);
+                        restock_obj := create_object(259, 0, 0);
                         add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 259));
                     end
                     else begin
@@ -419,7 +419,7 @@ begin
                                     end
                                     else begin
                                         if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                            metarule(43, self_obj);
+                                            inven_unwield(self_obj);
                                         end
                                     end
                                 end
@@ -437,7 +437,7 @@ begin
             end
             else begin
                 if (obj_is_carrying_obj_pid(self_obj, 259) < restock_amt) then begin
-                    restock_obj := create_object_sid(259, 0, 0, -1);
+                    restock_obj := create_object(259, 0, 0);
                     add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 259));
                 end
                 else begin
@@ -460,7 +460,7 @@ begin
                                 end
                                 else begin
                                     if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                        metarule(43, self_obj);
+                                        inven_unwield(self_obj);
                                     end
                                 end
                             end
@@ -479,7 +479,7 @@ begin
             if (obj_is_carrying_obj_pid(self_obj, 159) < restock_amt) then begin
                 if (60 >= random(1, 100)) then begin
                     if (obj_is_carrying_obj_pid(self_obj, 159) < restock_amt) then begin
-                        restock_obj := create_object_sid(159, 0, 0, -1);
+                        restock_obj := create_object(159, 0, 0);
                         add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 159));
                     end
                     else begin
@@ -502,7 +502,7 @@ begin
                                     end
                                     else begin
                                         if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                            metarule(43, self_obj);
+                                            inven_unwield(self_obj);
                                         end
                                     end
                                 end
@@ -520,7 +520,7 @@ begin
             end
             else begin
                 if (obj_is_carrying_obj_pid(self_obj, 159) < restock_amt) then begin
-                    restock_obj := create_object_sid(159, 0, 0, -1);
+                    restock_obj := create_object(159, 0, 0);
                     add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 159));
                 end
                 else begin
@@ -543,7 +543,7 @@ begin
                                 end
                                 else begin
                                     if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                        metarule(43, self_obj);
+                                        inven_unwield(self_obj);
                                     end
                                 end
                             end
@@ -562,7 +562,7 @@ begin
             if (obj_is_carrying_obj_pid(self_obj, 110) < restock_amt) then begin
                 if (50 >= random(1, 100)) then begin
                     if (obj_is_carrying_obj_pid(self_obj, 110) < restock_amt) then begin
-                        restock_obj := create_object_sid(110, 0, 0, -1);
+                        restock_obj := create_object(110, 0, 0);
                         add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 110));
                     end
                     else begin
@@ -585,7 +585,7 @@ begin
                                     end
                                     else begin
                                         if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                            metarule(43, self_obj);
+                                            inven_unwield(self_obj);
                                         end
                                     end
                                 end
@@ -603,7 +603,7 @@ begin
             end
             else begin
                 if (obj_is_carrying_obj_pid(self_obj, 110) < restock_amt) then begin
-                    restock_obj := create_object_sid(110, 0, 0, -1);
+                    restock_obj := create_object(110, 0, 0);
                     add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 110));
                 end
                 else begin
@@ -626,7 +626,7 @@ begin
                                 end
                                 else begin
                                     if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                        metarule(43, self_obj);
+                                        inven_unwield(self_obj);
                                     end
                                 end
                             end
@@ -645,7 +645,7 @@ begin
             if (obj_is_carrying_obj_pid(self_obj, 48) < restock_amt) then begin
                 if (50 >= random(1, 100)) then begin
                     if (obj_is_carrying_obj_pid(self_obj, 48) < restock_amt) then begin
-                        restock_obj := create_object_sid(48, 0, 0, -1);
+                        restock_obj := create_object(48, 0, 0);
                         add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 48));
                     end
                     else begin
@@ -668,7 +668,7 @@ begin
                                     end
                                     else begin
                                         if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                            metarule(43, self_obj);
+                                            inven_unwield(self_obj);
                                         end
                                     end
                                 end
@@ -686,7 +686,7 @@ begin
             end
             else begin
                 if (obj_is_carrying_obj_pid(self_obj, 48) < restock_amt) then begin
-                    restock_obj := create_object_sid(48, 0, 0, -1);
+                    restock_obj := create_object(48, 0, 0);
                     add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 48));
                 end
                 else begin
@@ -709,7 +709,7 @@ begin
                                 end
                                 else begin
                                     if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                        metarule(43, self_obj);
+                                        inven_unwield(self_obj);
                                     end
                                 end
                             end
@@ -728,7 +728,7 @@ begin
             if (obj_is_carrying_obj_pid(self_obj, 109) < restock_amt) then begin
                 if (50 >= random(1, 100)) then begin
                     if (obj_is_carrying_obj_pid(self_obj, 109) < restock_amt) then begin
-                        restock_obj := create_object_sid(109, 0, 0, -1);
+                        restock_obj := create_object(109, 0, 0);
                         add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 109));
                     end
                     else begin
@@ -751,7 +751,7 @@ begin
                                     end
                                     else begin
                                         if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                            metarule(43, self_obj);
+                                            inven_unwield(self_obj);
                                         end
                                     end
                                 end
@@ -769,7 +769,7 @@ begin
             end
             else begin
                 if (obj_is_carrying_obj_pid(self_obj, 109) < restock_amt) then begin
-                    restock_obj := create_object_sid(109, 0, 0, -1);
+                    restock_obj := create_object(109, 0, 0);
                     add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 109));
                 end
                 else begin
@@ -792,7 +792,7 @@ begin
                                 end
                                 else begin
                                     if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                        metarule(43, self_obj);
+                                        inven_unwield(self_obj);
                                     end
                                 end
                             end
@@ -811,7 +811,7 @@ begin
             if (obj_is_carrying_obj_pid(self_obj, 40) < restock_amt) then begin
                 if (50 >= random(1, 100)) then begin
                     if (obj_is_carrying_obj_pid(self_obj, 40) < restock_amt) then begin
-                        restock_obj := create_object_sid(40, 0, 0, -1);
+                        restock_obj := create_object(40, 0, 0);
                         add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 40));
                     end
                     else begin
@@ -834,7 +834,7 @@ begin
                                     end
                                     else begin
                                         if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                            metarule(43, self_obj);
+                                            inven_unwield(self_obj);
                                         end
                                     end
                                 end
@@ -852,7 +852,7 @@ begin
             end
             else begin
                 if (obj_is_carrying_obj_pid(self_obj, 40) < restock_amt) then begin
-                    restock_obj := create_object_sid(40, 0, 0, -1);
+                    restock_obj := create_object(40, 0, 0);
                     add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 40));
                 end
                 else begin
@@ -875,7 +875,7 @@ begin
                                 end
                                 else begin
                                     if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                        metarule(43, self_obj);
+                                        inven_unwield(self_obj);
                                     end
                                 end
                             end
@@ -894,7 +894,7 @@ begin
             if (obj_is_carrying_obj_pid(self_obj, 144) < restock_amt) then begin
                 if (30 >= random(1, 100)) then begin
                     if (obj_is_carrying_obj_pid(self_obj, 144) < restock_amt) then begin
-                        restock_obj := create_object_sid(144, 0, 0, -1);
+                        restock_obj := create_object(144, 0, 0);
                         add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 144));
                     end
                     else begin
@@ -917,7 +917,7 @@ begin
                                     end
                                     else begin
                                         if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                            metarule(43, self_obj);
+                                            inven_unwield(self_obj);
                                         end
                                     end
                                 end
@@ -935,7 +935,7 @@ begin
             end
             else begin
                 if (obj_is_carrying_obj_pid(self_obj, 144) < restock_amt) then begin
-                    restock_obj := create_object_sid(144, 0, 0, -1);
+                    restock_obj := create_object(144, 0, 0);
                     add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 144));
                 end
                 else begin
@@ -958,7 +958,7 @@ begin
                                 end
                                 else begin
                                     if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                        metarule(43, self_obj);
+                                        inven_unwield(self_obj);
                                     end
                                 end
                             end
@@ -982,4 +982,3 @@ procedure map_update_p_proc
 begin
     broken_hills_jacob_box := self_obj;
 end
-
