diff --git a/Games/Fallout2.Steam/Original/HIMANHOL.ssl b/Games/Fallout2.Steam/ReDefine/HIMANHOL.ssl
index f2a52f8..e566f84 100644
--- a/Games/Fallout2.Steam/Original/HIMANHOL.ssl
+++ b/Games/Fallout2.Steam/ReDefine/HIMANHOL.ssl
@@ -144,7 +144,7 @@ begin
     if (map_var(23) == 1) then begin
         if (local_var(0) == 0) then begin
             set_local_var(0, 1);
-            display_msg(message_str(995, 100));
+            display_mstr(100);
         end
         move_to(dude_obj, 11513, 1);
     end
@@ -152,7 +152,7 @@ begin
         if (map_var(22) == 1) then begin
             if (local_var(0) == 0) then begin
                 set_local_var(0, 1);
-                display_msg(message_str(995, 100));
+                display_mstr(100);
             end
             move_to(dude_obj, 11513, 1);
         end
@@ -162,7 +162,7 @@ begin
                     if (is_success(roll_vs_skill(dude_obj, 8, -50))) then begin
                         if (local_var(0) == 0) then begin
                             set_local_var(0, 1);
-                            display_msg(message_str(995, 100));
+                            display_mstr(100);
                         end
                         move_to(dude_obj, 11513, 1);
                     end
@@ -174,7 +174,7 @@ begin
                     if (is_success(roll_vs_skill(dude_obj, 8, -20))) then begin
                         if (local_var(0) == 0) then begin
                             set_local_var(0, 1);
-                            display_msg(message_str(995, 100));
+                            display_mstr(100);
                         end
                         move_to(dude_obj, 11513, 1);
                     end
@@ -201,4 +201,3 @@ end
 procedure map_update_p_proc
 begin
 end
-
