diff --git a/Games/Fallout2.Steam/Original/HINOTE.ssl b/Games/Fallout2.Steam/ReDefine/HINOTE.ssl
index 7b3f2fb..8e8d93e 100644
--- a/Games/Fallout2.Steam/Original/HINOTE.ssl
+++ b/Games/Fallout2.Steam/ReDefine/HINOTE.ssl
@@ -137,19 +137,18 @@ end
 
 procedure description_p_proc
 begin
-    if (global_var(524) == 0) then begin
-        set_global_var(524, 1);
+    if (global_var(GVAR_READ_FRANCIS_NOTE) == 0) then begin
+        set_global_var(GVAR_READ_FRANCIS_NOTE, 1);
     end
-    float_msg(dude_obj, message_str(1094, 101), 7);
+    float_msg(dude_obj, mstr(101), FLOAT_MSG_LIGHT_RED);
 end
 
 procedure use_p_proc
 begin
     script_overrides;
-    if (global_var(524) == 0) then begin
-        set_global_var(524, 1);
+    if (global_var(GVAR_READ_FRANCIS_NOTE) == 0) then begin
+        set_global_var(GVAR_READ_FRANCIS_NOTE, 1);
     end
-    display_msg(message_str(1094, 100));
-    float_msg(dude_obj, message_str(1094, 101), 7);
+    display_mstr(100);
+    float_msg(dude_obj, mstr(101), FLOAT_MSG_LIGHT_RED);
 end
-
