diff --git a/Games/Fallout2.Steam/Original/HIOUTBOX.ssl b/Games/Fallout2.Steam/ReDefine/HIOUTBOX.ssl
index 0d73967..0c58c23 100644
--- a/Games/Fallout2.Steam/Original/HIOUTBOX.ssl
+++ b/Games/Fallout2.Steam/ReDefine/HIOUTBOX.ssl
@@ -141,13 +141,13 @@ end
 procedure map_enter_p_proc
 begin
     broken_hills_outfitter_box := self_obj;
-    if (metarule(22, 0) == 0) then begin
+    if (is_loading_game == 0) then begin
         if (local_var(0) < game_time) then begin
             restock_amt := random(100, 500);
             if (obj_is_carrying_obj_pid(self_obj, 41) < restock_amt) then begin
                 if (100 >= random(1, 100)) then begin
                     if (obj_is_carrying_obj_pid(self_obj, 41) < restock_amt) then begin
-                        restock_obj := create_object_sid(41, 0, 0, -1);
+                        restock_obj := create_object(41, 0, 0);
                         add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 41));
                     end
                     else begin
@@ -170,7 +170,7 @@ begin
                                     end
                                     else begin
                                         if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                            metarule(43, self_obj);
+                                            inven_unwield(self_obj);
                                         end
                                     end
                                 end
@@ -188,7 +188,7 @@ begin
             end
             else begin
                 if (obj_is_carrying_obj_pid(self_obj, 41) < restock_amt) then begin
-                    restock_obj := create_object_sid(41, 0, 0, -1);
+                    restock_obj := create_object(41, 0, 0);
                     add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 41));
                 end
                 else begin
@@ -211,7 +211,7 @@ begin
                                 end
                                 else begin
                                     if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                        metarule(43, self_obj);
+                                        inven_unwield(self_obj);
                                     end
                                 end
                             end
@@ -230,7 +230,7 @@ begin
             if (obj_is_carrying_obj_pid(self_obj, 34) < restock_amt) then begin
                 if (25 >= random(1, 100)) then begin
                     if (obj_is_carrying_obj_pid(self_obj, 34) < restock_amt) then begin
-                        restock_obj := create_object_sid(34, 0, 0, -1);
+                        restock_obj := create_object(34, 0, 0);
                         add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 34));
                     end
                     else begin
@@ -253,7 +253,7 @@ begin
                                     end
                                     else begin
                                         if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                            metarule(43, self_obj);
+                                            inven_unwield(self_obj);
                                         end
                                     end
                                 end
@@ -271,7 +271,7 @@ begin
             end
             else begin
                 if (obj_is_carrying_obj_pid(self_obj, 34) < restock_amt) then begin
-                    restock_obj := create_object_sid(34, 0, 0, -1);
+                    restock_obj := create_object(34, 0, 0);
                     add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 34));
                 end
                 else begin
@@ -294,7 +294,7 @@ begin
                                 end
                                 else begin
                                     if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                        metarule(43, self_obj);
+                                        inven_unwield(self_obj);
                                     end
                                 end
                             end
@@ -313,7 +313,7 @@ begin
             if (obj_is_carrying_obj_pid(self_obj, 357) < restock_amt) then begin
                 if (25 >= random(1, 100)) then begin
                     if (obj_is_carrying_obj_pid(self_obj, 357) < restock_amt) then begin
-                        restock_obj := create_object_sid(357, 0, 0, -1);
+                        restock_obj := create_object(357, 0, 0);
                         add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 357));
                     end
                     else begin
@@ -336,7 +336,7 @@ begin
                                     end
                                     else begin
                                         if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                            metarule(43, self_obj);
+                                            inven_unwield(self_obj);
                                         end
                                     end
                                 end
@@ -354,7 +354,7 @@ begin
             end
             else begin
                 if (obj_is_carrying_obj_pid(self_obj, 357) < restock_amt) then begin
-                    restock_obj := create_object_sid(357, 0, 0, -1);
+                    restock_obj := create_object(357, 0, 0);
                     add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 357));
                 end
                 else begin
@@ -377,7 +377,7 @@ begin
                                 end
                                 else begin
                                     if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                        metarule(43, self_obj);
+                                        inven_unwield(self_obj);
                                     end
                                 end
                             end
@@ -396,7 +396,7 @@ begin
             if (obj_is_carrying_obj_pid(self_obj, 30) < restock_amt) then begin
                 if (30 >= random(1, 100)) then begin
                     if (obj_is_carrying_obj_pid(self_obj, 30) < restock_amt) then begin
-                        restock_obj := create_object_sid(30, 0, 0, -1);
+                        restock_obj := create_object(30, 0, 0);
                         add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 30));
                     end
                     else begin
@@ -419,7 +419,7 @@ begin
                                     end
                                     else begin
                                         if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                            metarule(43, self_obj);
+                                            inven_unwield(self_obj);
                                         end
                                     end
                                 end
@@ -437,7 +437,7 @@ begin
             end
             else begin
                 if (obj_is_carrying_obj_pid(self_obj, 30) < restock_amt) then begin
-                    restock_obj := create_object_sid(30, 0, 0, -1);
+                    restock_obj := create_object(30, 0, 0);
                     add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 30));
                 end
                 else begin
@@ -460,7 +460,7 @@ begin
                                 end
                                 else begin
                                     if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                        metarule(43, self_obj);
+                                        inven_unwield(self_obj);
                                     end
                                 end
                             end
@@ -479,7 +479,7 @@ begin
             if (obj_is_carrying_obj_pid(self_obj, 29) < restock_amt) then begin
                 if (40 >= random(1, 100)) then begin
                     if (obj_is_carrying_obj_pid(self_obj, 29) < restock_amt) then begin
-                        restock_obj := create_object_sid(29, 0, 0, -1);
+                        restock_obj := create_object(29, 0, 0);
                         add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 29));
                     end
                     else begin
@@ -502,7 +502,7 @@ begin
                                     end
                                     else begin
                                         if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                            metarule(43, self_obj);
+                                            inven_unwield(self_obj);
                                         end
                                     end
                                 end
@@ -520,7 +520,7 @@ begin
             end
             else begin
                 if (obj_is_carrying_obj_pid(self_obj, 29) < restock_amt) then begin
-                    restock_obj := create_object_sid(29, 0, 0, -1);
+                    restock_obj := create_object(29, 0, 0);
                     add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 29));
                 end
                 else begin
@@ -543,7 +543,7 @@ begin
                                 end
                                 else begin
                                     if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                        metarule(43, self_obj);
+                                        inven_unwield(self_obj);
                                     end
                                 end
                             end
@@ -562,7 +562,7 @@ begin
             if (obj_is_carrying_obj_pid(self_obj, 8) < restock_amt) then begin
                 if (65 >= random(1, 100)) then begin
                     if (obj_is_carrying_obj_pid(self_obj, 8) < restock_amt) then begin
-                        restock_obj := create_object_sid(8, 0, 0, -1);
+                        restock_obj := create_object(8, 0, 0);
                         add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 8));
                     end
                     else begin
@@ -585,7 +585,7 @@ begin
                                     end
                                     else begin
                                         if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                            metarule(43, self_obj);
+                                            inven_unwield(self_obj);
                                         end
                                     end
                                 end
@@ -603,7 +603,7 @@ begin
             end
             else begin
                 if (obj_is_carrying_obj_pid(self_obj, 8) < restock_amt) then begin
-                    restock_obj := create_object_sid(8, 0, 0, -1);
+                    restock_obj := create_object(8, 0, 0);
                     add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 8));
                 end
                 else begin
@@ -626,7 +626,7 @@ begin
                                 end
                                 else begin
                                     if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                        metarule(43, self_obj);
+                                        inven_unwield(self_obj);
                                     end
                                 end
                             end
@@ -645,7 +645,7 @@ begin
             if (obj_is_carrying_obj_pid(self_obj, 9) < restock_amt) then begin
                 if (45 >= random(1, 100)) then begin
                     if (obj_is_carrying_obj_pid(self_obj, 9) < restock_amt) then begin
-                        restock_obj := create_object_sid(9, 0, 0, -1);
+                        restock_obj := create_object(9, 0, 0);
                         add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 9));
                     end
                     else begin
@@ -668,7 +668,7 @@ begin
                                     end
                                     else begin
                                         if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                            metarule(43, self_obj);
+                                            inven_unwield(self_obj);
                                         end
                                     end
                                 end
@@ -686,7 +686,7 @@ begin
             end
             else begin
                 if (obj_is_carrying_obj_pid(self_obj, 9) < restock_amt) then begin
-                    restock_obj := create_object_sid(9, 0, 0, -1);
+                    restock_obj := create_object(9, 0, 0);
                     add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 9));
                 end
                 else begin
@@ -709,7 +709,7 @@ begin
                                 end
                                 else begin
                                     if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                        metarule(43, self_obj);
+                                        inven_unwield(self_obj);
                                     end
                                 end
                             end
@@ -728,7 +728,7 @@ begin
             if (obj_is_carrying_obj_pid(self_obj, 95) < restock_amt) then begin
                 if (50 >= random(1, 100)) then begin
                     if (obj_is_carrying_obj_pid(self_obj, 95) < restock_amt) then begin
-                        restock_obj := create_object_sid(95, 0, 0, -1);
+                        restock_obj := create_object(95, 0, 0);
                         add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 95));
                     end
                     else begin
@@ -751,7 +751,7 @@ begin
                                     end
                                     else begin
                                         if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                            metarule(43, self_obj);
+                                            inven_unwield(self_obj);
                                         end
                                     end
                                 end
@@ -769,7 +769,7 @@ begin
             end
             else begin
                 if (obj_is_carrying_obj_pid(self_obj, 95) < restock_amt) then begin
-                    restock_obj := create_object_sid(95, 0, 0, -1);
+                    restock_obj := create_object(95, 0, 0);
                     add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 95));
                 end
                 else begin
@@ -792,7 +792,7 @@ begin
                                 end
                                 else begin
                                     if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                        metarule(43, self_obj);
+                                        inven_unwield(self_obj);
                                     end
                                 end
                             end
@@ -811,7 +811,7 @@ begin
             if (obj_is_carrying_obj_pid(self_obj, 35) < restock_amt) then begin
                 if (35 >= random(1, 100)) then begin
                     if (obj_is_carrying_obj_pid(self_obj, 35) < restock_amt) then begin
-                        restock_obj := create_object_sid(35, 0, 0, -1);
+                        restock_obj := create_object(35, 0, 0);
                         add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 35));
                     end
                     else begin
@@ -834,7 +834,7 @@ begin
                                     end
                                     else begin
                                         if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                            metarule(43, self_obj);
+                                            inven_unwield(self_obj);
                                         end
                                     end
                                 end
@@ -852,7 +852,7 @@ begin
             end
             else begin
                 if (obj_is_carrying_obj_pid(self_obj, 35) < restock_amt) then begin
-                    restock_obj := create_object_sid(35, 0, 0, -1);
+                    restock_obj := create_object(35, 0, 0);
                     add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 35));
                 end
                 else begin
@@ -875,7 +875,7 @@ begin
                                 end
                                 else begin
                                     if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                        metarule(43, self_obj);
+                                        inven_unwield(self_obj);
                                     end
                                 end
                             end
@@ -894,7 +894,7 @@ begin
             if (obj_is_carrying_obj_pid(self_obj, 363) < restock_amt) then begin
                 if (40 >= random(1, 100)) then begin
                     if (obj_is_carrying_obj_pid(self_obj, 363) < restock_amt) then begin
-                        restock_obj := create_object_sid(363, 0, 0, -1);
+                        restock_obj := create_object(363, 0, 0);
                         add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 363));
                     end
                     else begin
@@ -917,7 +917,7 @@ begin
                                     end
                                     else begin
                                         if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                            metarule(43, self_obj);
+                                            inven_unwield(self_obj);
                                         end
                                     end
                                 end
@@ -935,7 +935,7 @@ begin
             end
             else begin
                 if (obj_is_carrying_obj_pid(self_obj, 363) < restock_amt) then begin
-                    restock_obj := create_object_sid(363, 0, 0, -1);
+                    restock_obj := create_object(363, 0, 0);
                     add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 363));
                 end
                 else begin
@@ -958,7 +958,7 @@ begin
                                 end
                                 else begin
                                     if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                        metarule(43, self_obj);
+                                        inven_unwield(self_obj);
                                     end
                                 end
                             end
@@ -977,7 +977,7 @@ begin
             if (obj_is_carrying_obj_pid(self_obj, 79) < restock_amt) then begin
                 if (85 >= random(1, 100)) then begin
                     if (obj_is_carrying_obj_pid(self_obj, 79) < restock_amt) then begin
-                        restock_obj := create_object_sid(79, 0, 0, -1);
+                        restock_obj := create_object(79, 0, 0);
                         add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 79));
                     end
                     else begin
@@ -1000,7 +1000,7 @@ begin
                                     end
                                     else begin
                                         if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                            metarule(43, self_obj);
+                                            inven_unwield(self_obj);
                                         end
                                     end
                                 end
@@ -1018,7 +1018,7 @@ begin
             end
             else begin
                 if (obj_is_carrying_obj_pid(self_obj, 79) < restock_amt) then begin
-                    restock_obj := create_object_sid(79, 0, 0, -1);
+                    restock_obj := create_object(79, 0, 0);
                     add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 79));
                 end
                 else begin
@@ -1041,7 +1041,7 @@ begin
                                 end
                                 else begin
                                     if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                        metarule(43, self_obj);
+                                        inven_unwield(self_obj);
                                     end
                                 end
                             end
@@ -1060,7 +1060,7 @@ begin
             if (obj_is_carrying_obj_pid(self_obj, 10) < restock_amt) then begin
                 if (25 >= random(1, 100)) then begin
                     if (obj_is_carrying_obj_pid(self_obj, 10) < restock_amt) then begin
-                        restock_obj := create_object_sid(10, 0, 0, -1);
+                        restock_obj := create_object(10, 0, 0);
                         add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 10));
                     end
                     else begin
@@ -1083,7 +1083,7 @@ begin
                                     end
                                     else begin
                                         if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                            metarule(43, self_obj);
+                                            inven_unwield(self_obj);
                                         end
                                     end
                                 end
@@ -1101,7 +1101,7 @@ begin
             end
             else begin
                 if (obj_is_carrying_obj_pid(self_obj, 10) < restock_amt) then begin
-                    restock_obj := create_object_sid(10, 0, 0, -1);
+                    restock_obj := create_object(10, 0, 0);
                     add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 10));
                 end
                 else begin
@@ -1124,7 +1124,7 @@ begin
                                 end
                                 else begin
                                     if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                        metarule(43, self_obj);
+                                        inven_unwield(self_obj);
                                     end
                                 end
                             end
@@ -1143,7 +1143,7 @@ begin
             if (obj_is_carrying_obj_pid(self_obj, 127) < restock_amt) then begin
                 if (70 >= random(1, 100)) then begin
                     if (obj_is_carrying_obj_pid(self_obj, 127) < restock_amt) then begin
-                        restock_obj := create_object_sid(127, 0, 0, -1);
+                        restock_obj := create_object(127, 0, 0);
                         add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 127));
                     end
                     else begin
@@ -1166,7 +1166,7 @@ begin
                                     end
                                     else begin
                                         if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                            metarule(43, self_obj);
+                                            inven_unwield(self_obj);
                                         end
                                     end
                                 end
@@ -1184,7 +1184,7 @@ begin
             end
             else begin
                 if (obj_is_carrying_obj_pid(self_obj, 127) < restock_amt) then begin
-                    restock_obj := create_object_sid(127, 0, 0, -1);
+                    restock_obj := create_object(127, 0, 0);
                     add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 127));
                 end
                 else begin
@@ -1207,7 +1207,7 @@ begin
                                 end
                                 else begin
                                     if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                        metarule(43, self_obj);
+                                        inven_unwield(self_obj);
                                     end
                                 end
                             end
@@ -1226,7 +1226,7 @@ begin
             if (obj_is_carrying_obj_pid(self_obj, 94) < restock_amt) then begin
                 if (50 >= random(1, 100)) then begin
                     if (obj_is_carrying_obj_pid(self_obj, 94) < restock_amt) then begin
-                        restock_obj := create_object_sid(94, 0, 0, -1);
+                        restock_obj := create_object(94, 0, 0);
                         add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 94));
                     end
                     else begin
@@ -1249,7 +1249,7 @@ begin
                                     end
                                     else begin
                                         if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                            metarule(43, self_obj);
+                                            inven_unwield(self_obj);
                                         end
                                     end
                                 end
@@ -1267,7 +1267,7 @@ begin
             end
             else begin
                 if (obj_is_carrying_obj_pid(self_obj, 94) < restock_amt) then begin
-                    restock_obj := create_object_sid(94, 0, 0, -1);
+                    restock_obj := create_object(94, 0, 0);
                     add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 94));
                 end
                 else begin
@@ -1290,7 +1290,7 @@ begin
                                 end
                                 else begin
                                     if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                        metarule(43, self_obj);
+                                        inven_unwield(self_obj);
                                     end
                                 end
                             end
@@ -1309,7 +1309,7 @@ begin
             if (obj_is_carrying_obj_pid(self_obj, 289) < restock_amt) then begin
                 if (70 >= random(1, 100)) then begin
                     if (obj_is_carrying_obj_pid(self_obj, 289) < restock_amt) then begin
-                        restock_obj := create_object_sid(289, 0, 0, -1);
+                        restock_obj := create_object(289, 0, 0);
                         add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 289));
                     end
                     else begin
@@ -1332,7 +1332,7 @@ begin
                                     end
                                     else begin
                                         if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                            metarule(43, self_obj);
+                                            inven_unwield(self_obj);
                                         end
                                     end
                                 end
@@ -1350,7 +1350,7 @@ begin
             end
             else begin
                 if (obj_is_carrying_obj_pid(self_obj, 289) < restock_amt) then begin
-                    restock_obj := create_object_sid(289, 0, 0, -1);
+                    restock_obj := create_object(289, 0, 0);
                     add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 289));
                 end
                 else begin
@@ -1373,7 +1373,7 @@ begin
                                 end
                                 else begin
                                     if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                        metarule(43, self_obj);
+                                        inven_unwield(self_obj);
                                     end
                                 end
                             end
@@ -1392,7 +1392,7 @@ begin
             if (obj_is_carrying_obj_pid(self_obj, 38) < restock_amt) then begin
                 if (25 >= random(1, 100)) then begin
                     if (obj_is_carrying_obj_pid(self_obj, 38) < restock_amt) then begin
-                        restock_obj := create_object_sid(38, 0, 0, -1);
+                        restock_obj := create_object(38, 0, 0);
                         add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 38));
                     end
                     else begin
@@ -1415,7 +1415,7 @@ begin
                                     end
                                     else begin
                                         if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                            metarule(43, self_obj);
+                                            inven_unwield(self_obj);
                                         end
                                     end
                                 end
@@ -1433,7 +1433,7 @@ begin
             end
             else begin
                 if (obj_is_carrying_obj_pid(self_obj, 38) < restock_amt) then begin
-                    restock_obj := create_object_sid(38, 0, 0, -1);
+                    restock_obj := create_object(38, 0, 0);
                     add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 38));
                 end
                 else begin
@@ -1456,7 +1456,7 @@ begin
                                 end
                                 else begin
                                     if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                        metarule(43, self_obj);
+                                        inven_unwield(self_obj);
                                     end
                                 end
                             end
@@ -1475,7 +1475,7 @@ begin
             if (obj_is_carrying_obj_pid(self_obj, 75) < restock_amt) then begin
                 if (35 >= random(1, 100)) then begin
                     if (obj_is_carrying_obj_pid(self_obj, 75) < restock_amt) then begin
-                        restock_obj := create_object_sid(75, 0, 0, -1);
+                        restock_obj := create_object(75, 0, 0);
                         add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 75));
                     end
                     else begin
@@ -1498,7 +1498,7 @@ begin
                                     end
                                     else begin
                                         if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                            metarule(43, self_obj);
+                                            inven_unwield(self_obj);
                                         end
                                     end
                                 end
@@ -1516,7 +1516,7 @@ begin
             end
             else begin
                 if (obj_is_carrying_obj_pid(self_obj, 75) < restock_amt) then begin
-                    restock_obj := create_object_sid(75, 0, 0, -1);
+                    restock_obj := create_object(75, 0, 0);
                     add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 75));
                 end
                 else begin
@@ -1539,7 +1539,7 @@ begin
                                 end
                                 else begin
                                     if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                        metarule(43, self_obj);
+                                        inven_unwield(self_obj);
                                     end
                                 end
                             end
@@ -1563,4 +1563,3 @@ procedure map_update_p_proc
 begin
     broken_hills_outfitter_box := self_obj;
 end
-
