diff --git a/Games/Fallout2.Steam/Original/JAKE.ssl b/Games/Fallout2.Steam/ReDefine/JAKE.ssl
index da2e606..67aa499 100644
--- a/Games/Fallout2.Steam/Original/JAKE.ssl
+++ b/Games/Fallout2.Steam/ReDefine/JAKE.ssl
@@ -35,35 +35,35 @@ variable initialized;
 procedure start
 begin
     if (not(initialized)) then begin
-        critter_add_trait(self_obj, 1, 5, 34);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_TORR);
         initialized := 1;
     end
     else begin
-        if (script_action == 12) then begin
+        if (script_action == critter_proc) then begin
             call critter_p_proc();
         end
         else begin
-            if (script_action == 22) then begin
+            if (script_action == timed_event_proc) then begin
                 call timed_event_p_proc();
             end
             else begin
-                if (script_action == 18) then begin
+                if (script_action == destroy_proc) then begin
                     call destroy_p_proc();
                 end
                 else begin
-                    if (script_action == 21) then begin
+                    if (script_action == look_at_proc) then begin
                         call look_at_p_proc();
                     end
                     else begin
-                        if (script_action == 23) then begin
+                        if (script_action == map_update_proc) then begin
                             call map_update_p_proc();
                         end
                         else begin
-                            if (script_action == 4) then begin
+                            if (script_action == pickup_proc) then begin
                                 call pickup_p_proc();
                             end
                             else begin
-                                if (script_action == 11) then begin
+                                if (script_action == talk_proc) then begin
                                     call talk_p_proc();
                                 end
                             end
@@ -78,34 +78,34 @@ end
 procedure critter_p_proc
 begin
     if (obj_can_see_obj(self_obj, dude_obj) and (map_var(1) == 1)) then begin
-        float_msg(self_obj, message_str(268, 143), 3);
+        float_msg(self_obj, message_str(SCRIPT_KSCVDOOR, 143), FLOAT_MSG_GREEN);
         animate_move_obj_to_tile(self_obj, 15484, 0);
         add_timer_event(self_obj, game_ticks(15), 1);
     end
     if (obj_can_see_obj(self_obj, dude_obj) and (map_var(1) == 2)) then begin
-        float_msg(self_obj, message_str(268, 144), 3);
+        float_msg(self_obj, message_str(SCRIPT_KSCVDOOR, 144), FLOAT_MSG_GREEN);
         hostile := 1;
     end
     if (hostile) then begin
         hostile := 0;
-        attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+        attack(dude_obj);
     end
 end
 
 procedure destroy_p_proc
 begin
     if (source_obj == dude_obj) then begin
-        if (((global_var(160) + global_var(159)) >= 25) and ((global_var(159) > (2 * global_var(160))) or (global_var(156) == 1))) then begin
-            set_global_var(156, 1);
-            set_global_var(157, 0);
+        if (((global_var(GVAR_GECKO_SKEETER_PART) + global_var(GVAR_GECKO_REQ_FORM)) >= 25) and ((global_var(GVAR_GECKO_REQ_FORM) > (2 * global_var(GVAR_GECKO_SKEETER_PART))) or (global_var(GVAR_SIERRA_LOCKOUT) == 1))) then begin
+            set_global_var(GVAR_SIERRA_LOCKOUT, 1);
+            set_global_var(GVAR_SIERRA_PASSWORD, 0);
         end
-        if (((global_var(160) + global_var(159)) >= 25) and ((global_var(160) > (3 * global_var(159))) or (global_var(157) == 1))) then begin
-            set_global_var(157, 1);
-            set_global_var(156, 0);
+        if (((global_var(GVAR_GECKO_SKEETER_PART) + global_var(GVAR_GECKO_REQ_FORM)) >= 25) and ((global_var(160) > (3 * global_var(GVAR_GECKO_REQ_FORM))) or (global_var(GVAR_SIERRA_PASSWORD) == 1))) then begin
+            set_global_var(GVAR_SIERRA_PASSWORD, 1);
+            set_global_var(GVAR_SIERRA_LOCKOUT, 0);
         end
-        set_global_var(160, global_var(160) + 1);
-        if ((global_var(160) % 6) == 0) then begin
-            set_global_var(155, global_var(155) + 1);
+        set_global_var(GVAR_GECKO_SKEETER_PART, global_var(GVAR_GECKO_SKEETER_PART) + 1);
+        if ((global_var(GVAR_GECKO_SKEETER_PART) % 6) == 0) then begin
+            set_global_var(GVAR_BRAIN_BOT_BRAIN, global_var(GVAR_BRAIN_BOT_BRAIN) + 1);
         end
     end
 end
@@ -114,16 +114,16 @@ procedure look_at_p_proc
 begin
     script_overrides;
     if (local_var(4) == 1) then begin
-        display_msg(message_str(268, 100));
+        display_msg(message_str(SCRIPT_KSCVDOOR, 100));
     end
     else begin
-        display_msg(message_str(268, 101));
+        display_msg(message_str(SCRIPT_KSCVDOOR, 101));
     end
 end
 
 procedure map_update_p_proc
 begin
-    if (global_var(129) == 2) then begin
+    if (global_var(GVAR_MODOC_JONNY_HOME) == 2) then begin
         kill_critter(self_obj, 57);
     end
 end
@@ -143,15 +143,15 @@ begin
             call Jake14();
         end
         else begin
-            if (global_var(132)) then begin
+            if (global_var(GVAR_MUTATE)) then begin
                 call Jake17();
             end
             else begin
-                if (global_var(133) == 2) then begin
+                if (global_var(GVAR_MUTATE_WHEN) == 2) then begin
                     call Jake16();
                 end
                 else begin
-                    if (global_var(133) == 1) then begin
+                    if (global_var(GVAR_MUTATE_WHEN) == 1) then begin
                         call Jake15();
                     end
                     else begin
@@ -167,125 +167,125 @@ end
 
 procedure timed_event_p_proc
 begin
-    float_msg(self_obj, message_str(268, 143), 3);
+    float_msg(self_obj, message_str(SCRIPT_KSCVDOOR, 143), FLOAT_MSG_GREEN);
     set_map_var(1, 2);
 end
 
 procedure Jake00
 begin
-    gsay_reply(268, 102);
-    giq_option(-3, 268, 103, Jake01, 50);
-    giq_option(4, 268, 104, Jake02, 50);
-    giq_option(4, 268, 105, Jake12, 50);
-    giq_option(4, 268, 106, Jake13, 50);
-    if ((global_var(129) == 1) and (obj_is_carrying_obj_pid(dude_obj, 72) != 0)) then begin
-        giq_option(4, 268, 141, Jake18, 50);
+    gsay_reply(SCRIPT_KSCVDOOR, 102);
+    giq_option(-3, SCRIPT_KSCVDOOR, 103, Jake01, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_KSCVDOOR, 104, Jake02, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_KSCVDOOR, 105, Jake12, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_KSCVDOOR, 106, Jake13, NEUTRAL_REACTION);
+    if ((global_var(GVAR_MODOC_JONNY_HOME) == 1) and (obj_is_carrying_obj_pid(dude_obj, 72) != 0)) then begin
+        giq_option(4, SCRIPT_KSCVDOOR, 141, Jake18, NEUTRAL_REACTION);
     end
 end
 
 procedure Jake01
 begin
-    gsay_message(268, 107, 50);
+    gsay_message(SCRIPT_KSCVDOOR, 107, NEUTRAL_REACTION);
 end
 
 procedure Jake02
 begin
-    gsay_reply(268, 108);
-    giq_option(4, 268, 109, Jake03, 50);
-    giq_option(4, 268, 110, JakeEnd, 50);
-    giq_option(4, 268, 111, Jake04, 50);
+    gsay_reply(SCRIPT_KSCVDOOR, 108);
+    giq_option(4, SCRIPT_KSCVDOOR, 109, Jake03, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_KSCVDOOR, 110, JakeEnd, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_KSCVDOOR, 111, Jake04, NEUTRAL_REACTION);
 end
 
 procedure Jake03
 begin
-    gsay_reply(268, 112);
-    giq_option(4, 268, 113, JakeEnd, 50);
+    gsay_reply(SCRIPT_KSCVDOOR, 112);
+    giq_option(4, SCRIPT_KSCVDOOR, 113, JakeEnd, NEUTRAL_REACTION);
 end
 
 procedure Jake04
 begin
-    gsay_reply(268, 114);
-    giq_option(4, 268, 115, Jake05, 50);
-    giq_option(4, 268, 116, Jake06, 50);
+    gsay_reply(SCRIPT_KSCVDOOR, 114);
+    giq_option(4, SCRIPT_KSCVDOOR, 115, Jake05, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_KSCVDOOR, 116, Jake06, NEUTRAL_REACTION);
 end
 
 procedure Jake05
 begin
-    gsay_message(268, 117, 50);
+    gsay_message(SCRIPT_KSCVDOOR, 117, NEUTRAL_REACTION);
     set_local_var(5, 1);
 end
 
 procedure Jake06
 begin
-    gsay_reply(268, 118);
-    giq_option(4, 268, 119, Jake07, 50);
-    giq_option(4, 268, 120, Jake11, 50);
+    gsay_reply(SCRIPT_KSCVDOOR, 118);
+    giq_option(4, SCRIPT_KSCVDOOR, 119, Jake07, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_KSCVDOOR, 120, Jake11, NEUTRAL_REACTION);
 end
 
 procedure Jake07
 begin
-    gsay_reply(268, 121);
-    giq_option(4, 268, 122, Jake08, 50);
-    giq_option(4, 268, 123, Jake09, 50);
-    giq_option(4, 268, 124, Jake10, 50);
+    gsay_reply(SCRIPT_KSCVDOOR, 121);
+    giq_option(4, SCRIPT_KSCVDOOR, 122, Jake08, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_KSCVDOOR, 123, Jake09, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_KSCVDOOR, 124, Jake10, NEUTRAL_REACTION);
 end
 
 procedure Jake08
 begin
-    set_global_var(133, 1);
-    gsay_message(268, 125, 50);
+    set_global_var(GVAR_MUTATE_WHEN, 1);
+    gsay_message(SCRIPT_KSCVDOOR, 125, NEUTRAL_REACTION);
 end
 
 procedure Jake09
 begin
-    gsay_reply(268, 126);
-    giq_option(4, 268, 127, Jake08, 50);
-    giq_option(4, 268, 128, Jake10, 50);
+    gsay_reply(SCRIPT_KSCVDOOR, 126);
+    giq_option(4, SCRIPT_KSCVDOOR, 127, Jake08, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_KSCVDOOR, 128, Jake10, NEUTRAL_REACTION);
 end
 
 procedure Jake10
 begin
-    gsay_message(268, 129, 50);
+    gsay_message(SCRIPT_KSCVDOOR, 129, NEUTRAL_REACTION);
     call JakeCombat();
 end
 
 procedure Jake11
 begin
-    gsay_message(268, 130, 50);
+    gsay_message(SCRIPT_KSCVDOOR, 130, NEUTRAL_REACTION);
 end
 
 procedure Jake12
 begin
-    gsay_reply(268, 131);
-    giq_option(4, 268, 132, Jake07, 50);
-    giq_option(4, 268, 133, Jake08, 50);
-    giq_option(4, 268, 134, Jake09, 50);
-    giq_option(4, 268, 135, Jake10, 50);
+    gsay_reply(SCRIPT_KSCVDOOR, 131);
+    giq_option(4, SCRIPT_KSCVDOOR, 132, Jake07, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_KSCVDOOR, 133, Jake08, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_KSCVDOOR, 134, Jake09, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_KSCVDOOR, 135, Jake10, NEUTRAL_REACTION);
 end
 
 procedure Jake13
 begin
-    gsay_message(268, 136, 50);
+    gsay_message(SCRIPT_KSCVDOOR, 136, NEUTRAL_REACTION);
 end
 
 procedure Jake14
 begin
-    gsay_message(268, 137, 50);
+    gsay_message(SCRIPT_KSCVDOOR, 137, NEUTRAL_REACTION);
 end
 
 procedure Jake15
 begin
-    gsay_message(268, 138, 50);
+    gsay_message(SCRIPT_KSCVDOOR, 138, NEUTRAL_REACTION);
 end
 
 procedure Jake16
 begin
-    gsay_message(268, 139, 50);
+    gsay_message(SCRIPT_KSCVDOOR, 139, NEUTRAL_REACTION);
 end
 
 procedure Jake17
 begin
-    gsay_message(268, 140, 50);
+    gsay_message(SCRIPT_KSCVDOOR, 140, NEUTRAL_REACTION);
 end
 
 procedure Jake18
@@ -294,7 +294,7 @@ begin
     LVar0 := obj_carrying_pid_obj(dude_obj, 72);
     rm_obj_from_inven(dude_obj, LVar0);
     destroy_object(LVar0);
-    gsay_message(268, 142, 50);
+    gsay_message(SCRIPT_KSCVDOOR, 142, NEUTRAL_REACTION);
 end
 
 procedure JakeCombat
@@ -305,4 +305,3 @@ end
 procedure JakeEnd
 begin
 end
-
