diff --git a/Games/Fallout2.Steam/Original/JASON.ssl b/Games/Fallout2.Steam/ReDefine/JASON.ssl
index 16f2b00..b15b515 100644
--- a/Games/Fallout2.Steam/Original/JASON.ssl
+++ b/Games/Fallout2.Steam/ReDefine/JASON.ssl
@@ -35,23 +35,23 @@ variable initialized;
 procedure start
 begin
     if (not(initialized)) then begin
-        critter_add_trait(self_obj, 1, 6, 49);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 49);
         initialized := 1;
     end
     else begin
-        if (script_action == 12) then begin
+        if (script_action == critter_proc) then begin
             call critter_p_proc();
         end
         else begin
-            if (script_action == 18) then begin
+            if (script_action == destroy_proc) then begin
                 call destroy_p_proc();
             end
             else begin
-                if (script_action == 21) then begin
+                if (script_action == look_at_proc) then begin
                     call look_at_p_proc();
                 end
                 else begin
-                    if (script_action == 11) then begin
+                    if (script_action == talk_proc) then begin
                         call talk_p_proc();
                     end
                 end
@@ -70,17 +70,17 @@ end
 procedure destroy_p_proc
 begin
     if (source_obj == dude_obj) then begin
-        if (((global_var(160) + global_var(159)) >= 25) and ((global_var(159) > (2 * global_var(160))) or (global_var(156) == 1))) then begin
-            set_global_var(156, 1);
-            set_global_var(157, 0);
+        if (((global_var(GVAR_GECKO_SKEETER_PART) + global_var(GVAR_GECKO_REQ_FORM)) >= 25) and ((global_var(GVAR_GECKO_REQ_FORM) > (2 * global_var(GVAR_GECKO_SKEETER_PART))) or (global_var(GVAR_SIERRA_LOCKOUT) == 1))) then begin
+            set_global_var(GVAR_SIERRA_LOCKOUT, 1);
+            set_global_var(GVAR_SIERRA_PASSWORD, 0);
         end
-        if (((global_var(160) + global_var(159)) >= 25) and ((global_var(160) > (3 * global_var(159))) or (global_var(157) == 1))) then begin
-            set_global_var(157, 1);
-            set_global_var(156, 0);
+        if (((global_var(GVAR_GECKO_SKEETER_PART) + global_var(GVAR_GECKO_REQ_FORM)) >= 25) and ((global_var(160) > (3 * global_var(GVAR_GECKO_REQ_FORM))) or (global_var(GVAR_SIERRA_PASSWORD) == 1))) then begin
+            set_global_var(GVAR_SIERRA_PASSWORD, 1);
+            set_global_var(GVAR_SIERRA_LOCKOUT, 0);
         end
-        set_global_var(159, global_var(159) + 1);
-        if ((global_var(159) % 2) == 0) then begin
-            set_global_var(155, global_var(155) - 1);
+        set_global_var(GVAR_GECKO_REQ_FORM, global_var(GVAR_GECKO_REQ_FORM) + 1);
+        if ((global_var(GVAR_GECKO_REQ_FORM) % 2) == 0) then begin
+            set_global_var(GVAR_BRAIN_BOT_BRAIN, global_var(GVAR_BRAIN_BOT_BRAIN) - 1);
         end
     end
 end
@@ -89,10 +89,10 @@ procedure look_at_p_proc
 begin
     script_overrides;
     if (not(local_var(1))) then begin
-        display_msg(message_str(382, 101));
+        display_msg(message_str(SCRIPT_SIFFIELD, 101));
     end
     else begin
-        display_msg(message_str(382, 100));
+        display_msg(message_str(SCRIPT_SIFFIELD, 100));
     end
 end
 
@@ -100,7 +100,7 @@ procedure talk_p_proc
 begin
     start_gdialog(382, self_obj, 4, -1, -1);
     gsay_start;
-    if (cur_map_index == 45) then begin
+    if (CUR_MAP_NCR4) then begin
         if (following) then begin
             call Jason23();
         end
@@ -125,161 +125,160 @@ end
 
 procedure Jason01
 begin
-    gsay_reply(382, 102);
-    giq_option(4, 382, 103, Jason05, 50);
-    giq_option(4, 382, 104, Jason13, 50);
-    giq_option(4, 382, 105, Jason08, 50);
-    giq_option(4, 382, 106, Jason07, 50);
-    giq_option(-3, 382, 107, Jason02, 50);
+    gsay_reply(SCRIPT_SIFFIELD, 102);
+    giq_option(4, SCRIPT_SIFFIELD, 103, Jason05, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_SIFFIELD, 104, Jason13, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_SIFFIELD, 105, Jason08, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_SIFFIELD, 106, Jason07, NEUTRAL_REACTION);
+    giq_option(-3, SCRIPT_SIFFIELD, 107, Jason02, NEUTRAL_REACTION);
 end
 
 procedure Jason02
 begin
-    gsay_message(382, 108, 50);
+    gsay_message(SCRIPT_SIFFIELD, 108, NEUTRAL_REACTION);
 end
 
 procedure Jason03
 begin
-    gsay_message(382, 109, 50);
-    gsay_message(382, 110, 50);
+    gsay_message(SCRIPT_SIFFIELD, 109, NEUTRAL_REACTION);
+    gsay_message(SCRIPT_SIFFIELD, 110, NEUTRAL_REACTION);
 end
 
 procedure Jason04
 begin
-    gsay_message(382, 111, 50);
-    gsay_message(382, 112, 50);
+    gsay_message(SCRIPT_SIFFIELD, 111, NEUTRAL_REACTION);
+    gsay_message(SCRIPT_SIFFIELD, 112, NEUTRAL_REACTION);
 end
 
 procedure Jason05
 begin
-    gsay_reply(382, 113);
-    giq_option(4, 382, 114, Jason06, 50);
-    giq_option(4, 382, 115, Jason14, 50);
+    gsay_reply(SCRIPT_SIFFIELD, 113);
+    giq_option(4, SCRIPT_SIFFIELD, 114, Jason06, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_SIFFIELD, 115, Jason14, NEUTRAL_REACTION);
 end
 
 procedure Jason06
 begin
-    gsay_reply(382, 116);
-    giq_option(4, 382, 117, Jason07, 50);
-    giq_option(4, 382, 118, Jason08, 50);
-    giq_option(6, 382, 119, Jason15, 50);
+    gsay_reply(SCRIPT_SIFFIELD, 116);
+    giq_option(4, SCRIPT_SIFFIELD, 117, Jason07, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_SIFFIELD, 118, Jason08, NEUTRAL_REACTION);
+    giq_option(6, SCRIPT_SIFFIELD, 119, Jason15, NEUTRAL_REACTION);
 end
 
 procedure Jason07
 begin
-    gsay_reply(382, 120);
-    giq_option(4, 382, 121, Jason12, 50);
-    giq_option(4, 382, 122, Jason08, 50);
+    gsay_reply(SCRIPT_SIFFIELD, 120);
+    giq_option(4, SCRIPT_SIFFIELD, 121, Jason12, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_SIFFIELD, 122, Jason08, NEUTRAL_REACTION);
 end
 
 procedure Jason08
 begin
-    gsay_reply(382, 123);
-    giq_option(4, 382, 124, Jason12, 50);
+    gsay_reply(SCRIPT_SIFFIELD, 123);
+    giq_option(4, SCRIPT_SIFFIELD, 124, Jason12, NEUTRAL_REACTION);
 end
 
 procedure Jason09
 begin
-    gsay_reply(382, 125);
-    giq_option(4, 382, 126, Jason10, 50);
-    giq_option(4, 382, 127, Jason11, 50);
+    gsay_reply(SCRIPT_SIFFIELD, 125);
+    giq_option(4, SCRIPT_SIFFIELD, 126, Jason10, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_SIFFIELD, 127, Jason11, NEUTRAL_REACTION);
 end
 
 procedure Jason10
 begin
-    gsay_message(382, 128, 50);
+    gsay_message(SCRIPT_SIFFIELD, 128, NEUTRAL_REACTION);
 end
 
 procedure Jason11
 begin
-    gsay_message(382, 129, 50);
+    gsay_message(SCRIPT_SIFFIELD, 129, NEUTRAL_REACTION);
 end
 
 procedure Jason12
 begin
-    gsay_message(382, 130, 50);
+    gsay_message(SCRIPT_SIFFIELD, 130, NEUTRAL_REACTION);
 end
 
 procedure Jason13
 begin
-    gsay_message(382, 131, 50);
+    gsay_message(SCRIPT_SIFFIELD, 131, NEUTRAL_REACTION);
     set_local_var(0, 1);
 end
 
 procedure Jason14
 begin
-    gsay_message(382, 132, 50);
+    gsay_message(SCRIPT_SIFFIELD, 132, NEUTRAL_REACTION);
     call Jason10();
 end
 
 procedure Jason15
 begin
-    gsay_message(382, 133, 50);
+    gsay_message(SCRIPT_SIFFIELD, 133, NEUTRAL_REACTION);
     call Jason11();
 end
 
 procedure Jason16
 begin
-    gsay_reply(382, 134);
-    giq_option(4, 382, 135, Jason08, 50);
-    giq_option(4, 382, 136, Jason17, 50);
+    gsay_reply(SCRIPT_SIFFIELD, 134);
+    giq_option(4, SCRIPT_SIFFIELD, 135, Jason08, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_SIFFIELD, 136, Jason17, NEUTRAL_REACTION);
 end
 
 procedure Jason17
 begin
-    gsay_message(382, 137, 50);
+    gsay_message(SCRIPT_SIFFIELD, 137, NEUTRAL_REACTION);
 end
 
 procedure Jason18
 begin
-    gsay_message(382, 138, 50);
+    gsay_message(SCRIPT_SIFFIELD, 138, NEUTRAL_REACTION);
 end
 
 procedure Jason19
 begin
-    gsay_reply(382, 139);
-    giq_option(-3, 382, 140, Jason20, 50);
-    giq_option(4, 382, 141, Jason20, 50);
-    giq_option(4, 382, 142, Jason21, 50);
-    giq_option(4, 382, 143, Jason22, 50);
+    gsay_reply(SCRIPT_SIFFIELD, 139);
+    giq_option(-3, SCRIPT_SIFFIELD, 140, Jason20, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_SIFFIELD, 141, Jason20, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_SIFFIELD, 142, Jason21, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_SIFFIELD, 143, Jason22, NEUTRAL_REACTION);
 end
 
 procedure Jason20
 begin
-    gsay_message(382, 144, 50);
-    critter_add_trait(self_obj, 1, 6, 0);
+    gsay_message(SCRIPT_SIFFIELD, 144, NEUTRAL_REACTION);
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 0);
     following := 1;
 end
 
 procedure Jason21
 begin
-    gsay_message(382, 145, 50);
+    gsay_message(SCRIPT_SIFFIELD, 145, NEUTRAL_REACTION);
 end
 
 procedure Jason22
 begin
-    gsay_reply(382, 146);
-    giq_option(4, 382, 147, Jason20, 50);
-    giq_option(4, 382, 148, Jason21, 50);
+    gsay_reply(SCRIPT_SIFFIELD, 146);
+    giq_option(4, SCRIPT_SIFFIELD, 147, Jason20, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_SIFFIELD, 148, Jason21, NEUTRAL_REACTION);
 end
 
 procedure Jason23
 begin
-    gsay_message(382, 149, 50);
+    gsay_message(SCRIPT_SIFFIELD, 149, NEUTRAL_REACTION);
 end
 
 procedure follow_player
 begin
     if (tile_distance_objs(self_obj, dude_obj) > 7) then begin
-        animate_move_obj_to_tile(self_obj, tile_num(dude_obj), 1);
+        animate_move_obj_to_tile(self_obj, dude_tile, 1);
     end
     else begin
         if (tile_distance_objs(self_obj, dude_obj) > 3) then begin
-            animate_move_obj_to_tile(self_obj, tile_num(dude_obj), 0);
+            animate_move_obj_to_tile(self_obj, dude_tile, 0);
         end
         else begin
             animate_move_obj_to_tile(self_obj, tile_num(self_obj), 0);
         end
     end
 end
-
