diff --git a/Games/Fallout2.Steam/Original/KCBOB.ssl b/Games/Fallout2.Steam/ReDefine/KCBOB.ssl
index 237fe18..cc30eef 100644
--- a/Games/Fallout2.Steam/Original/KCBOB.ssl
+++ b/Games/Fallout2.Steam/ReDefine/KCBOB.ssl
@@ -199,7 +199,7 @@ procedure critter_p_proc
 begin
     if ((local_var(5) == 2) and obj_can_see_obj(self_obj, dude_obj)) then begin
         set_local_var(5, 1);
-        attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+        attack(dude_obj);
     end
 end
 
@@ -214,7 +214,7 @@ procedure talk_p_proc
 begin
     if (local_var(2) == 0) then begin
         set_local_var(2, 1);
-        set_local_var(3, (get_critter_stat(dude_obj, 3) - get_critter_stat(self_obj, 3)) * 5);
+        set_local_var(3, (dude_charisma - self_charisma) * 5);
         debug_msg("Base Reaction == " + local_var(3));
     end
     Static_Reaction := 0;
@@ -226,14 +226,14 @@ begin
     else begin
         reaction_bonus_karma := -1 * (-1 * reaction_bonus_karma / 1000);
     end
-    if (global_var(48) >= 0) then begin
-        reaction_bonus_town_rep := global_var(48) / 2;
+    if (global_var(GVAR_TOWN_REP_KLAMATH) >= 0) then begin
+        reaction_bonus_town_rep := global_var(GVAR_TOWN_REP_KLAMATH) / 2;
     end
     else begin
-        reaction_bonus_town_rep := -1 * (-1 * global_var(48) / 2);
+        reaction_bonus_town_rep := -1 * (-1 * global_var(GVAR_TOWN_REP_KLAMATH) / 2);
     end
-    if (has_trait(2, dude_obj, 13)) then begin
-        if (get_critter_stat(dude_obj, 34) == get_critter_stat(self_obj, 34)) then begin
+    if (has_trait(TRAIT_TRAIT, dude_obj, TRAIT_sex_appeal)) then begin
+        if (dude_gender == self_gender) then begin
             Static_Reaction := Static_Reaction + -20;
         end
         else begin
@@ -249,48 +249,48 @@ begin
         debug_msg("Karma Reaction Bonus == " + Static_Reaction);
         Static_Reaction := Static_Reaction - reaction_bonus_town_rep;
         debug_msg("Town Rep Bonus == " + Static_Reaction);
-        set_global_var(37, 0);
-        set_global_var(38, 0);
-        set_global_var(39, 0);
-        set_global_var(40, 0);
-        set_global_var(41, 0);
-        set_global_var(42, 0);
-        set_global_var(43, 0);
-        set_global_var(44, 0);
-        set_global_var(45, 0);
-        if (global_var(0) >= 1000) then begin
-            set_global_var(37, 1);
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+        set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+        set_global_var(GVAR_KARMA_DEFENDER, 0);
+        set_global_var(GVAR_KARMA_WANDERER, 0);
+        set_global_var(GVAR_KARMA_BETRAYER, 0);
+        set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+        set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+            set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
         end
         else begin
-            if (global_var(0) >= 750) then begin
-                set_global_var(38, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
             end
             else begin
-                if (global_var(0) >= 500) then begin
-                    set_global_var(39, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                 end
                 else begin
-                    if (global_var(0) >= 250) then begin
-                        set_global_var(40, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                        set_global_var(GVAR_KARMA_DEFENDER, 1);
                     end
                     else begin
-                        if (global_var(0) > -250) then begin
-                            set_global_var(41, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                            set_global_var(GVAR_KARMA_WANDERER, 1);
                         end
                         else begin
-                            if (global_var(0) > -500) then begin
-                                set_global_var(42, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                set_global_var(GVAR_KARMA_BETRAYER, 1);
                             end
                             else begin
-                                if (global_var(0) > -750) then begin
-                                    set_global_var(43, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -1000) then begin
-                                        set_global_var(44, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                     end
                                     else begin
-                                        set_global_var(45, 1);
+                                        set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                     end
                                 end
                             end
@@ -299,35 +299,35 @@ begin
                 end
             end
         end
-        if (global_var(37) == 1) then begin
+        if (global_var(GVAR_KARMA_HOLY_WARRIOR) == 1) then begin
             Static_Reaction := Static_Reaction - 20;
         end
         else begin
-            if (global_var(38) == 1) then begin
+            if (global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES) == 1) then begin
                 Static_Reaction := Static_Reaction - 15;
             end
             else begin
-                if (global_var(39) == 1) then begin
+                if (global_var(GVAR_KARMA_SHIELD_OF_HOPE) == 1) then begin
                     Static_Reaction := Static_Reaction - 10;
                 end
                 else begin
-                    if (global_var(40) == 1) then begin
+                    if (global_var(GVAR_KARMA_DEFENDER) == 1) then begin
                         Static_Reaction := Static_Reaction - 5;
                     end
                     else begin
-                        if (global_var(41) == 1) then begin
+                        if (global_var(GVAR_KARMA_WANDERER) == 1) then begin
                             Static_Reaction := Static_Reaction - 0;
                         end
                         else begin
-                            if (global_var(42) == 1) then begin
+                            if (global_var(GVAR_KARMA_BETRAYER) == 1) then begin
                                 Static_Reaction := Static_Reaction - -5;
                             end
                             else begin
-                                if (global_var(43) == 1) then begin
+                                if (global_var(GVAR_KARMA_SWORD_OF_DESPAIR) == 1) then begin
                                     Static_Reaction := Static_Reaction - -10;
                                 end
                                 else begin
-                                    if (global_var(44) == 1) then begin
+                                    if (global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE) == 1) then begin
                                         Static_Reaction := Static_Reaction - -15;
                                     end
                                     else begin
@@ -347,48 +347,48 @@ begin
         debug_msg("Karma Reaction Bonus == " + Static_Reaction);
         Static_Reaction := Static_Reaction + reaction_bonus_town_rep;
         debug_msg("Town Rep Bonus == " + Static_Reaction);
-        set_global_var(37, 0);
-        set_global_var(38, 0);
-        set_global_var(39, 0);
-        set_global_var(40, 0);
-        set_global_var(41, 0);
-        set_global_var(42, 0);
-        set_global_var(43, 0);
-        set_global_var(44, 0);
-        set_global_var(45, 0);
-        if (global_var(0) >= 1000) then begin
-            set_global_var(37, 1);
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+        set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+        set_global_var(GVAR_KARMA_DEFENDER, 0);
+        set_global_var(GVAR_KARMA_WANDERER, 0);
+        set_global_var(GVAR_KARMA_BETRAYER, 0);
+        set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+        set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+            set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
         end
         else begin
-            if (global_var(0) >= 750) then begin
-                set_global_var(38, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
             end
             else begin
-                if (global_var(0) >= 500) then begin
-                    set_global_var(39, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                 end
                 else begin
-                    if (global_var(0) >= 250) then begin
-                        set_global_var(40, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                        set_global_var(GVAR_KARMA_DEFENDER, 1);
                     end
                     else begin
-                        if (global_var(0) > -250) then begin
-                            set_global_var(41, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                            set_global_var(GVAR_KARMA_WANDERER, 1);
                         end
                         else begin
-                            if (global_var(0) > -500) then begin
-                                set_global_var(42, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                set_global_var(GVAR_KARMA_BETRAYER, 1);
                             end
                             else begin
-                                if (global_var(0) > -750) then begin
-                                    set_global_var(43, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -1000) then begin
-                                        set_global_var(44, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                     end
                                     else begin
-                                        set_global_var(45, 1);
+                                        set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                     end
                                 end
                             end
@@ -397,35 +397,35 @@ begin
                 end
             end
         end
-        if (global_var(37) == 1) then begin
+        if (global_var(GVAR_KARMA_HOLY_WARRIOR) == 1) then begin
             Static_Reaction := Static_Reaction + 20;
         end
         else begin
-            if (global_var(38) == 1) then begin
+            if (global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES) == 1) then begin
                 Static_Reaction := Static_Reaction + 15;
             end
             else begin
-                if (global_var(39) == 1) then begin
+                if (global_var(GVAR_KARMA_SHIELD_OF_HOPE) == 1) then begin
                     Static_Reaction := Static_Reaction + 10;
                 end
                 else begin
-                    if (global_var(40) == 1) then begin
+                    if (global_var(GVAR_KARMA_DEFENDER) == 1) then begin
                         Static_Reaction := Static_Reaction + 5;
                     end
                     else begin
-                        if (global_var(41) == 1) then begin
+                        if (global_var(GVAR_KARMA_WANDERER) == 1) then begin
                             Static_Reaction := Static_Reaction + 0;
                         end
                         else begin
-                            if (global_var(42) == 1) then begin
+                            if (global_var(GVAR_KARMA_BETRAYER) == 1) then begin
                                 Static_Reaction := Static_Reaction + -5;
                             end
                             else begin
-                                if (global_var(43) == 1) then begin
+                                if (global_var(GVAR_KARMA_SWORD_OF_DESPAIR) == 1) then begin
                                     Static_Reaction := Static_Reaction + -10;
                                 end
                                 else begin
-                                    if (global_var(44) == 1) then begin
+                                    if (global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE) == 1) then begin
                                         Static_Reaction := Static_Reaction + -15;
                                     end
                                     else begin
@@ -441,42 +441,42 @@ begin
         debug_msg("Good Person Reaction Bonus == " + Static_Reaction);
     end
     if (Slavery_Tolerant == 0) then begin
-        if (global_var(11) == 1) then begin
+        if (global_var(GVAR_REPUTATION_SLAVER) == 1) then begin
             Static_Reaction := Static_Reaction + (2 * -25);
         end
     end
     else begin
         if (Slavery_Tolerant == 1) then begin
-            if (global_var(11) == 1) then begin
+            if (global_var(GVAR_REPUTATION_SLAVER) == 1) then begin
                 Static_Reaction := Static_Reaction + -25;
             end
         end
         else begin
             if (Slavery_Tolerant == 3) then begin
-                if (global_var(11) == 1) then begin
+                if (global_var(GVAR_REPUTATION_SLAVER) == 1) then begin
                     Static_Reaction := Static_Reaction - -25;
                 end
             end
         end
     end
     debug_msg("Slaver + Aligned Reaction == " + Static_Reaction);
-    Static_Reaction := Static_Reaction + (10 * has_trait(0, dude_obj, 10));
+    Static_Reaction := Static_Reaction + (10 * has_trait(TRAIT_PERK, dude_obj, PERK_presence));
     debug_msg("Presence Reaction == " + Static_Reaction);
-    if (global_var(1) >= 2) then begin
+    if (global_var(GVAR_CHILDKILLER_REPUTATION) >= 2) then begin
         Static_Reaction := Static_Reaction + -30;
     end
     debug_msg("Childkiller Reaction == " + Static_Reaction);
     set_local_var(0, local_var(3) + Static_Reaction);
     debug_msg("Initial Reaction == " + local_var(0));
-    if ((global_var(68) == 1) or (local_var(6) == 1)) then begin
+    if ((global_var(GVAR_ENEMY_KLAMATH) == 1) or (local_var(6) == 1)) then begin
         call Node998();
     end
     else begin
         if (local_var(13) == 1) then begin
-            display_msg(message_str(74, 550));
+            display_mstr(550);
         end
         else begin
-            if (global_var(198) == 4) then begin
+            if (global_var(GVAR_STILL_STATUS) == 4) then begin
                 start_gdialog(74, self_obj, 4, -1, -1);
                 gsay_start;
                 call Node017();
@@ -484,7 +484,7 @@ begin
                 end_dialogue;
             end
             else begin
-                if (get_critter_stat(dude_obj, 4) <= (0 - -3)) then begin
+                if (dude_iq <= (0 - -3)) then begin
                     start_gdialog(74, self_obj, 4, -1, -1);
                     gsay_start;
                     call Node001();
@@ -492,7 +492,7 @@ begin
                     end_dialogue;
                 end
                 else begin
-                    if (global_var(198) == 5) then begin
+                    if (global_var(GVAR_STILL_STATUS) == 5) then begin
                         start_gdialog(74, self_obj, 4, -1, -1);
                         gsay_start;
                         call Node040();
@@ -500,7 +500,7 @@ begin
                         end_dialogue;
                     end
                     else begin
-                        if (global_var(198) == 2) then begin
+                        if (global_var(GVAR_STILL_STATUS) == 2) then begin
                             start_gdialog(74, self_obj, 4, -1, -1);
                             gsay_start;
                             call Node018();
@@ -508,7 +508,7 @@ begin
                             end_dialogue;
                         end
                         else begin
-                            if (global_var(198) == 3) then begin
+                            if (global_var(GVAR_STILL_STATUS) == 3) then begin
                                 start_gdialog(74, self_obj, 4, -1, -1);
                                 gsay_start;
                                 call Node029();
@@ -516,11 +516,11 @@ begin
                                 end_dialogue;
                             end
                             else begin
-                                if (global_var(198) == 1) then begin
+                                if (global_var(GVAR_STILL_STATUS) == 1) then begin
                                     start_gdialog(74, self_obj, 4, -1, -1);
                                     gsay_start;
-                                    if (((game_time / (10 * 60 * 60)) - global_var(458)) > 24) then begin
-                                        set_global_var(198, 3);
+                                    if (((game_time / (10 * 60 * 60)) - global_var(GVAR_STILL_START)) > 24) then begin
+                                        set_global_var(GVAR_STILL_STATUS, 3);
                                         call Node029();
                                     end
                                     else begin
@@ -530,7 +530,7 @@ begin
                                     end_dialogue;
                                 end
                                 else begin
-                                    if (global_var(global_var(48)) <= -3) then begin
+                                    if (global_var(global_var(GVAR_TOWN_REP_KLAMATH)) <= -3) then begin
                                         start_gdialog(74, self_obj, 4, -1, -1);
                                         gsay_start;
                                         call Node004();
@@ -538,7 +538,7 @@ begin
                                         end_dialogue;
                                     end
                                     else begin
-                                        if (global_var(global_var(48)) >= -2) then begin
+                                        if (global_var(global_var(GVAR_TOWN_REP_KLAMATH)) >= -2) then begin
                                             start_gdialog(74, self_obj, 4, -1, -1);
                                             gsay_start;
                                             if (local_var(4) == 0) then begin
@@ -565,59 +565,59 @@ end
 procedure destroy_p_proc
 begin
     if (source_obj == dude_obj) then begin
-        set_global_var(5, global_var(5) + 1);
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(4) > (3 * global_var(5))) or (global_var(2) == 1)) and (global_var(1) == 0)) then begin
-            set_global_var(2, 1);
-            set_global_var(3, 0);
+        set_global_var(GVAR_GOOD_MONSTER, global_var(GVAR_GOOD_MONSTER) + 1);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(4) > (3 * global_var(GVAR_GOOD_MONSTER))) or (global_var(GVAR_CHAMPION_REPUTATION) == 1)) and (global_var(GVAR_CHILDKILLER_REPUTATION) == 0)) then begin
+            set_global_var(GVAR_CHAMPION_REPUTATION, 1);
+            set_global_var(GVAR_BERSERKER_REPUTATION, 0);
         end
-        if (metarule(51, self_obj) == 2) then begin
+        if (critter_kill_type(self_obj) == KILL_TYPE_children_kills) then begin
             if (source_obj == dude_obj) then begin
-                set_global_var(2, 0);
-                set_global_var(1, global_var(1) + 1);
-                set_global_var(0, global_var(0) + -15);
+                set_global_var(GVAR_CHAMPION_REPUTATION, 0);
+                set_global_var(GVAR_CHILDKILLER_REPUTATION, global_var(GVAR_CHILDKILLER_REPUTATION) + 1);
+                set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + -15);
                 debug_msg("Player gains " + -15 + " Karma Points.");
-                set_global_var(37, 0);
-                set_global_var(38, 0);
-                set_global_var(39, 0);
-                set_global_var(40, 0);
-                set_global_var(41, 0);
-                set_global_var(42, 0);
-                set_global_var(43, 0);
-                set_global_var(44, 0);
-                set_global_var(45, 0);
-                if (global_var(0) >= 1000) then begin
-                    set_global_var(37, 1);
+                set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+                set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+                set_global_var(GVAR_KARMA_DEFENDER, 0);
+                set_global_var(GVAR_KARMA_WANDERER, 0);
+                set_global_var(GVAR_KARMA_BETRAYER, 0);
+                set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+                set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+                set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+                    set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
                 end
                 else begin
-                    if (global_var(0) >= 750) then begin
-                        set_global_var(38, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
                     end
                     else begin
-                        if (global_var(0) >= 500) then begin
-                            set_global_var(39, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                            set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                         end
                         else begin
-                            if (global_var(0) >= 250) then begin
-                                set_global_var(40, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                                set_global_var(GVAR_KARMA_DEFENDER, 1);
                             end
                             else begin
-                                if (global_var(0) > -250) then begin
-                                    set_global_var(41, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                                    set_global_var(GVAR_KARMA_WANDERER, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -500) then begin
-                                        set_global_var(42, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                        set_global_var(GVAR_KARMA_BETRAYER, 1);
                                     end
                                     else begin
-                                        if (global_var(0) > -750) then begin
-                                            set_global_var(43, 1);
+                                        if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                            set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                         end
                                         else begin
-                                            if (global_var(0) > -1000) then begin
-                                                set_global_var(44, 1);
+                                            if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                                set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                             end
                                             else begin
-                                                set_global_var(45, 1);
+                                                set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                             end
                                         end
                                     end
@@ -626,58 +626,58 @@ begin
                         end
                     end
                 end
-                set_global_var(48, global_var(48) + -8);
+                set_global_var(GVAR_TOWN_REP_KLAMATH, global_var(GVAR_TOWN_REP_KLAMATH) + -8);
                 debug_msg("Added " + -8 + " to Town Rep");
             end
         end
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(5) > (2 * global_var(4))) or (global_var(3) == 1))) then begin
-            set_global_var(3, 1);
-            set_global_var(2, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
+            set_global_var(GVAR_BERSERKER_REPUTATION, 1);
+            set_global_var(GVAR_CHAMPION_REPUTATION, 0);
         end
-        set_global_var(0, global_var(0) + -10);
+        set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + -10);
         debug_msg("Player gains " + -10 + " Karma Points.");
-        set_global_var(37, 0);
-        set_global_var(38, 0);
-        set_global_var(39, 0);
-        set_global_var(40, 0);
-        set_global_var(41, 0);
-        set_global_var(42, 0);
-        set_global_var(43, 0);
-        set_global_var(44, 0);
-        set_global_var(45, 0);
-        if (global_var(0) >= 1000) then begin
-            set_global_var(37, 1);
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+        set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+        set_global_var(GVAR_KARMA_DEFENDER, 0);
+        set_global_var(GVAR_KARMA_WANDERER, 0);
+        set_global_var(GVAR_KARMA_BETRAYER, 0);
+        set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+        set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+            set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
         end
         else begin
-            if (global_var(0) >= 750) then begin
-                set_global_var(38, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
             end
             else begin
-                if (global_var(0) >= 500) then begin
-                    set_global_var(39, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                 end
                 else begin
-                    if (global_var(0) >= 250) then begin
-                        set_global_var(40, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                        set_global_var(GVAR_KARMA_DEFENDER, 1);
                     end
                     else begin
-                        if (global_var(0) > -250) then begin
-                            set_global_var(41, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                            set_global_var(GVAR_KARMA_WANDERER, 1);
                         end
                         else begin
-                            if (global_var(0) > -500) then begin
-                                set_global_var(42, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                set_global_var(GVAR_KARMA_BETRAYER, 1);
                             end
                             else begin
-                                if (global_var(0) > -750) then begin
-                                    set_global_var(43, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -1000) then begin
-                                        set_global_var(44, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                     end
                                     else begin
-                                        set_global_var(45, 1);
+                                        set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                     end
                                 end
                             end
@@ -686,7 +686,7 @@ begin
                 end
             end
         end
-        set_global_var(48, global_var(48) + -5);
+        set_global_var(GVAR_TOWN_REP_KLAMATH, global_var(GVAR_TOWN_REP_KLAMATH) + -5);
         debug_msg("Added " + -5 + " to Town Rep");
     end
 end
@@ -694,13 +694,13 @@ end
 procedure look_at_p_proc
 begin
     script_overrides;
-    display_msg(message_str(74, 100));
+    display_mstr(100);
 end
 
 procedure description_p_proc
 begin
     script_overrides;
-    display_msg(message_str(74, 101));
+    display_mstr(101);
 end
 
 procedure use_skill_on_p_proc
@@ -711,15 +711,15 @@ procedure damage_p_proc
 begin
     if (source_obj == dude_obj) then begin
         set_local_var(6, 1);
-        set_global_var(68, 1);
+        set_global_var(GVAR_ENEMY_KLAMATH, 1);
     end
 end
 
 procedure map_enter_p_proc
 begin
     Only_Once := 0;
-    critter_add_trait(self_obj, 1, 6, 43);
-    critter_add_trait(self_obj, 1, 5, 24);
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 43);
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_TOUGH_CITIZEN);
 end
 
 procedure push_p_proc
@@ -731,7 +731,7 @@ procedure Node910
 begin
     if (item_caps_total(dude_obj) > 4) then begin
         item_caps_adjust(dude_obj, -5);
-        if (global_var(198) == 0) then begin
+        if (global_var(GVAR_STILL_STATUS) == 0) then begin
             call Node007();
         end
         else begin
@@ -755,19 +755,19 @@ end
 
 procedure Node950
 begin
-    set_global_var(198, 5);
+    set_global_var(GVAR_STILL_STATUS, 5);
     item_caps_adjust(dude_obj, 50);
 end
 
 procedure Node955
 begin
-    set_global_var(198, 5);
+    set_global_var(GVAR_STILL_STATUS, 5);
     item_caps_adjust(dude_obj, 55);
 end
 
 procedure Node995
 begin
-    if (get_critter_stat(dude_obj, 3) >= 7) then begin
+    if (dude_charisma >= 7) then begin
         set_local_var(3, 100);
         set_local_var(1, 4);
     end
@@ -785,7 +785,7 @@ end
 
 procedure Node997
 begin
-    if (get_critter_stat(dude_obj, 3) >= 8) then begin
+    if (dude_charisma >= 8) then begin
         call Node004();
     end
     else begin
@@ -804,95 +804,95 @@ end
 
 procedure Node001
 begin
-    if (get_critter_stat(dude_obj, 34) == 0) then begin
-        gsay_reply(74, 110);
+    if (dude_gender == 0) then begin
+        Reply(110);
     end
     else begin
-        gsay_reply(74, 111);
+        Reply(111);
     end
-    giq_option(-3, 74, 112, Node002, 50);
-    giq_option(-3, 74, 113, Node002, 50);
-    giq_option(-3, 74, message_str(74, 114) + obj_name(dude_obj), Node002, 50);
-    giq_option(-3, 74, 115, Node003, 50);
+    giq_option(-3, NAME, 112, Node002, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 113, Node002, NEUTRAL_REACTION);
+    giq_option(-3, NAME, mstr(114) + dude_name, Node002, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 115, Node003, NEUTRAL_REACTION);
 end
 
 procedure Node002
 begin
-    if (get_critter_stat(dude_obj, 34) == 0) then begin
-        gsay_message(74, 120, 50);
+    if (dude_gender == 0) then begin
+        gsay_message(NAME, 120, NEUTRAL_REACTION);
     end
     else begin
-        gsay_message(74, 121, 50);
+        gsay_message(NAME, 121, NEUTRAL_REACTION);
     end
 end
 
 procedure Node003
 begin
-    if (get_critter_stat(dude_obj, 34) == 0) then begin
-        gsay_message(74, 130, 50);
+    if (dude_gender == 0) then begin
+        gsay_message(NAME, 130, NEUTRAL_REACTION);
     end
     else begin
-        gsay_message(74, 131, 50);
+        gsay_message(NAME, 131, NEUTRAL_REACTION);
     end
 end
 
 procedure Node004
 begin
-    if (get_critter_stat(dude_obj, 34) == 0) then begin
-        gsay_message(74, 140, 50);
+    if (dude_gender == 0) then begin
+        gsay_message(NAME, 140, NEUTRAL_REACTION);
     end
     else begin
-        gsay_message(74, 141, 50);
+        gsay_message(NAME, 141, NEUTRAL_REACTION);
     end
 end
 
 procedure Node005
 begin
-    gsay_reply(74, 150);
-    giq_option(4, 74, message_str(74, 151) + obj_name(dude_obj) + message_str(74, 152), Node006, 50);
-    giq_option(4, 74, message_str(74, 153) + obj_name(dude_obj) + message_str(74, 154), Node006, 50);
-    giq_option(4, 74, 155, Node003, 50);
+    Reply(150);
+    giq_option(4, NAME, mstr(151) + dude_name + mstr(152), Node006, NEUTRAL_REACTION);
+    giq_option(4, NAME, mstr(153) + dude_name + mstr(154), Node006, NEUTRAL_REACTION);
+    giq_option(4, NAME, 155, Node003, NEUTRAL_REACTION);
 end
 
 procedure Node006
 begin
-    gsay_reply(74, 160);
-    giq_option(4, 74, 161, Node910, 50);
-    giq_option(4, 74, 162, Node003, 50);
-    giq_option(4, 74, 163, Node004, 50);
-    if (global_var(198) == 0) then begin
-        giq_option(7, 74, 164, Node920, 50);
+    Reply(160);
+    giq_option(4, NAME, 161, Node910, NEUTRAL_REACTION);
+    giq_option(4, NAME, 162, Node003, NEUTRAL_REACTION);
+    giq_option(4, NAME, 163, Node004, NEUTRAL_REACTION);
+    if (global_var(GVAR_STILL_STATUS) == 0) then begin
+        giq_option(7, NAME, 164, Node920, NEUTRAL_REACTION);
     end
 end
 
 procedure Node007
 begin
-    gsay_reply(74, 170);
-    giq_option(4, 74, 171, Node003, 50);
-    giq_option(4, 74, 172, Node009, 50);
-    giq_option(4, 74, 173, Node010, 50);
-    giq_option(6, 74, 174, Node010, 50);
-    giq_option(6, 74, 175, Node019, 50);
+    Reply(170);
+    giq_option(4, NAME, 171, Node003, NEUTRAL_REACTION);
+    giq_option(4, NAME, 172, Node009, NEUTRAL_REACTION);
+    giq_option(4, NAME, 173, Node010, NEUTRAL_REACTION);
+    giq_option(6, NAME, 174, Node010, NEUTRAL_REACTION);
+    giq_option(6, NAME, 175, Node019, NEUTRAL_REACTION);
 end
 
 procedure Node008
 begin
-    gsay_reply(74, 180);
-    giq_option(-3, 74, 181, Node003, 50);
+    Reply(180);
+    giq_option(-3, NAME, 181, Node003, NEUTRAL_REACTION);
 end
 
 procedure Node009
 begin
-    gsay_reply(74, 190);
-    giq_option(4, 74, 191, Node012, 50);
-    giq_option(4, 74, 192, Node003, 50);
+    Reply(190);
+    giq_option(4, NAME, 191, Node012, NEUTRAL_REACTION);
+    giq_option(4, NAME, 192, Node003, NEUTRAL_REACTION);
 end
 
 procedure Node010
 begin
-    gsay_reply(74, 200);
-    giq_option(4, 74, 201, Node009, 50);
-    giq_option(4, 74, 202, Node999, 50);
+    Reply(200);
+    giq_option(4, NAME, 201, Node009, NEUTRAL_REACTION);
+    giq_option(4, NAME, 202, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node011
@@ -900,30 +900,30 @@ begin
     if (obj_is_carrying_obj_pid(self_obj, 125)) then begin
         mybooze := obj_carrying_pid_obj(self_obj, 125);
         rm_obj_from_inven(self_obj, mybooze);
-        item := create_object_sid(125, 0, 0, -1);
+        item := create_object(125, 0, 0);
         add_obj_to_inven(dude_obj, item);
     end
-    gsay_reply(74, 210);
-    giq_option(-3, 74, 181, Node009, 50);
+    Reply(210);
+    giq_option(-3, NAME, 181, Node009, NEUTRAL_REACTION);
 end
 
 procedure Node012
 begin
-    gsay_reply(74, 220);
-    giq_option(4, 74, 221, Node013, 50);
-    giq_option(4, 74, 222, Node003, 50);
+    Reply(220);
+    giq_option(4, NAME, 221, Node013, NEUTRAL_REACTION);
+    giq_option(4, NAME, 222, Node003, NEUTRAL_REACTION);
     if (local_var(12) == 0) then begin
-        giq_option(4, 74, 223, Node014, 50);
+        giq_option(4, NAME, 223, Node014, NEUTRAL_REACTION);
     end
 end
 
 procedure Node013
 begin
-    set_global_var(458, game_time / (10 * 60 * 60));
-    set_global_var(198, 1);
-    gsay_reply(74, 230);
-    giq_option(4, 74, 231, Node999, 50);
-    giq_option(4, 74, 232, Node019, 50);
+    set_global_var(GVAR_STILL_START, game_time / (10 * 60 * 60));
+    set_global_var(GVAR_STILL_STATUS, 1);
+    Reply(230);
+    giq_option(4, NAME, 231, Node999, NEUTRAL_REACTION);
+    giq_option(4, NAME, 232, Node019, NEUTRAL_REACTION);
 end
 
 procedure Node014
@@ -931,83 +931,83 @@ begin
     set_local_var(12, 1);
     if (is_success(roll_vs_skill(dude_obj, 15, 0))) then begin
         set_local_var(11, 1);
-        gsay_reply(74, 240);
+        Reply(240);
     end
     else begin
-        gsay_reply(74, 241);
+        Reply(241);
     end
-    giq_option(4, 74, 242, Node012, 50);
-    giq_option(4, 74, 243, Node003, 50);
+    giq_option(4, NAME, 242, Node012, NEUTRAL_REACTION);
+    giq_option(4, NAME, 243, Node003, NEUTRAL_REACTION);
 end
 
 procedure Node015
 begin
-    if (get_critter_stat(dude_obj, 34) == 0) then begin
-        gsay_reply(74, 250);
+    if (dude_gender == 0) then begin
+        Reply(250);
     end
     else begin
-        gsay_reply(74, 251);
+        Reply(251);
     end
-    giq_option(4, 74, 252, Node006, 50);
-    giq_option(4, 74, 253, Node003, 50);
+    giq_option(4, NAME, 252, Node006, NEUTRAL_REACTION);
+    giq_option(4, NAME, 253, Node003, NEUTRAL_REACTION);
 end
 
 procedure Node016
 begin
-    gsay_message(74, 260, 50);
+    gsay_message(NAME, 260, NEUTRAL_REACTION);
 end
 
 procedure Node017
 begin
-    gsay_message(74, 270, 50);
+    gsay_message(NAME, 270, NEUTRAL_REACTION);
 end
 
 procedure Node018
 begin
-    set_global_var(0, global_var(0) + 15);
+    set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + 15);
     debug_msg("Player gains " + 15 + " Karma Points.");
-    set_global_var(37, 0);
-    set_global_var(38, 0);
-    set_global_var(39, 0);
-    set_global_var(40, 0);
-    set_global_var(41, 0);
-    set_global_var(42, 0);
-    set_global_var(43, 0);
-    set_global_var(44, 0);
-    set_global_var(45, 0);
-    if (global_var(0) >= 1000) then begin
-        set_global_var(37, 1);
+    set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+    set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+    set_global_var(GVAR_KARMA_DEFENDER, 0);
+    set_global_var(GVAR_KARMA_WANDERER, 0);
+    set_global_var(GVAR_KARMA_BETRAYER, 0);
+    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+    set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+    set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+    if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
     end
     else begin
-        if (global_var(0) >= 750) then begin
-            set_global_var(38, 1);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+            set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
         end
         else begin
-            if (global_var(0) >= 500) then begin
-                set_global_var(39, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
             end
             else begin
-                if (global_var(0) >= 250) then begin
-                    set_global_var(40, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                    set_global_var(GVAR_KARMA_DEFENDER, 1);
                 end
                 else begin
-                    if (global_var(0) > -250) then begin
-                        set_global_var(41, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                        set_global_var(GVAR_KARMA_WANDERER, 1);
                     end
                     else begin
-                        if (global_var(0) > -500) then begin
-                            set_global_var(42, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                            set_global_var(GVAR_KARMA_BETRAYER, 1);
                         end
                         else begin
-                            if (global_var(0) > -750) then begin
-                                set_global_var(43, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                             end
                             else begin
-                                if (global_var(0) > -1000) then begin
-                                    set_global_var(44, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                    set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                 end
                                 else begin
-                                    set_global_var(45, 1);
+                                    set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                 end
                             end
                         end
@@ -1018,131 +1018,130 @@ begin
     end
     if (local_var(11) == 1) then begin
         call Node955();
-        gsay_reply(74, 281);
-        giq_option(4, 74, 282, Node019, 50);
-        giq_option(4, 74, 283, Node003, 50);
+        Reply(281);
+        giq_option(4, NAME, 282, Node019, NEUTRAL_REACTION);
+        giq_option(4, NAME, 283, Node003, NEUTRAL_REACTION);
     end
     else begin
         call Node950();
-        gsay_reply(74, 280);
-        giq_option(4, 74, 282, Node019, 50);
-        giq_option(4, 74, 283, Node003, 50);
+        Reply(280);
+        giq_option(4, NAME, 282, Node019, NEUTRAL_REACTION);
+        giq_option(4, NAME, 283, Node003, NEUTRAL_REACTION);
     end
 end
 
 procedure Node019
 begin
-    gsay_reply(74, 290);
-    giq_option(4, 74, 291, Node021, 50);
-    giq_option(4, 74, 292, Node023, 50);
-    giq_option(4, 74, 293, Node024, 50);
-    giq_option(4, 74, 294, Node025, 50);
-    giq_option(4, 74, 295, Node026, 50);
-    giq_option(4, 74, 296, Node031, 50);
-    giq_option(4, 74, 297, Node003, 50);
+    Reply(290);
+    giq_option(4, NAME, 291, Node021, NEUTRAL_REACTION);
+    giq_option(4, NAME, 292, Node023, NEUTRAL_REACTION);
+    giq_option(4, NAME, 293, Node024, NEUTRAL_REACTION);
+    giq_option(4, NAME, 294, Node025, NEUTRAL_REACTION);
+    giq_option(4, NAME, 295, Node026, NEUTRAL_REACTION);
+    giq_option(4, NAME, 296, Node031, NEUTRAL_REACTION);
+    giq_option(4, NAME, 297, Node003, NEUTRAL_REACTION);
 end
 
 procedure Node020
 begin
-    gsay_reply(74, 310);
-    giq_option(4, 74, 311, Node006, 50);
-    giq_option(4, 74, 312, Node003, 50);
+    Reply(310);
+    giq_option(4, NAME, 311, Node006, NEUTRAL_REACTION);
+    giq_option(4, NAME, 312, Node003, NEUTRAL_REACTION);
 end
 
 procedure Node021
 begin
-    gsay_reply(74, 320);
-    giq_option(4, 74, 321, Node022, 50);
+    Reply(320);
+    giq_option(4, NAME, 321, Node022, NEUTRAL_REACTION);
 end
 
 procedure Node022
 begin
-    gsay_reply(74, 330);
-    giq_option(4, 74, 331, Node019, 50);
-    giq_option(4, 74, 332, Node003, 50);
+    Reply(330);
+    giq_option(4, NAME, 331, Node019, NEUTRAL_REACTION);
+    giq_option(4, NAME, 332, Node003, NEUTRAL_REACTION);
 end
 
 procedure Node023
 begin
-    gsay_reply(74, 340);
-    giq_option(4, 74, 341, Node019, 50);
-    giq_option(4, 74, 342, Node003, 50);
+    Reply(340);
+    giq_option(4, NAME, 341, Node019, NEUTRAL_REACTION);
+    giq_option(4, NAME, 342, Node003, NEUTRAL_REACTION);
 end
 
 procedure Node024
 begin
-    gsay_reply(74, 350);
-    giq_option(4, 74, 351, Node019, 50);
-    giq_option(4, 74, 352, Node003, 50);
+    Reply(350);
+    giq_option(4, NAME, 351, Node019, NEUTRAL_REACTION);
+    giq_option(4, NAME, 352, Node003, NEUTRAL_REACTION);
 end
 
 procedure Node025
 begin
-    if (metarule(17, 1) == 0) then begin
+    if (town_known(1) == MARK_STATE_UNKNOWN) then begin
         debug_msg("  mark_on_map(" + 1 + ")");
-        mark_area_known(0, 1, 1);
+        mark_area_known(MARK_TYPE_TOWN, 1, MARK_STATE_KNOWN);
     end
-    gsay_reply(74, 360);
-    giq_option(4, 74, 361, Node019, 50);
-    giq_option(4, 74, 362, Node003, 50);
+    Reply(360);
+    giq_option(4, NAME, 361, Node019, NEUTRAL_REACTION);
+    giq_option(4, NAME, 362, Node003, NEUTRAL_REACTION);
 end
 
 procedure Node026
 begin
-    gsay_reply(74, 370);
-    giq_option(4, 74, 371, Node027, 50);
-    giq_option(4, 74, 372, Node028, 50);
-    giq_option(4, 74, 373, Node019, 50);
-    giq_option(4, 74, 374, Node003, 50);
+    Reply(370);
+    giq_option(4, NAME, 371, Node027, NEUTRAL_REACTION);
+    giq_option(4, NAME, 372, Node028, NEUTRAL_REACTION);
+    giq_option(4, NAME, 373, Node019, NEUTRAL_REACTION);
+    giq_option(4, NAME, 374, Node003, NEUTRAL_REACTION);
 end
 
 procedure Node027
 begin
-    gsay_reply(74, 380);
-    giq_option(4, 74, 381, Node028, 50);
-    giq_option(4, 74, 382, Node019, 50);
-    giq_option(4, 74, 383, Node003, 50);
+    Reply(380);
+    giq_option(4, NAME, 381, Node028, NEUTRAL_REACTION);
+    giq_option(4, NAME, 382, Node019, NEUTRAL_REACTION);
+    giq_option(4, NAME, 383, Node003, NEUTRAL_REACTION);
 end
 
 procedure Node028
 begin
-    gsay_reply(74, 390);
-    giq_option(4, 74, 391, Node027, 50);
-    giq_option(4, 74, 392, Node019, 50);
-    giq_option(4, 74, 393, Node003, 50);
+    Reply(390);
+    giq_option(4, NAME, 391, Node027, NEUTRAL_REACTION);
+    giq_option(4, NAME, 392, Node019, NEUTRAL_REACTION);
+    giq_option(4, NAME, 393, Node003, NEUTRAL_REACTION);
 end
 
 procedure Node029
 begin
-    gsay_reply(74, 400);
-    giq_option(4, 74, 401, Node030, 50);
-    giq_option(4, 74, 402, Node030, 50);
-    giq_option(4, 74, 403, Node999, 50);
-    giq_option(7, 74, 404, Node030, 50);
+    Reply(400);
+    giq_option(4, NAME, 401, Node030, NEUTRAL_REACTION);
+    giq_option(4, NAME, 402, Node030, NEUTRAL_REACTION);
+    giq_option(4, NAME, 403, Node999, NEUTRAL_REACTION);
+    giq_option(7, NAME, 404, Node030, NEUTRAL_REACTION);
 end
 
 procedure Node030
 begin
-    gsay_message(74, 410, 50);
+    gsay_message(NAME, 410, NEUTRAL_REACTION);
 end
 
 procedure Node031
 begin
-    gsay_reply(74, 420);
-    giq_option(4, 74, 421, Node032, 50);
+    Reply(420);
+    giq_option(4, NAME, 421, Node032, NEUTRAL_REACTION);
 end
 
 procedure Node032
 begin
-    gsay_reply(74, 430);
-    giq_option(4, 74, 431, Node019, 50);
-    giq_option(4, 74, 432, Node003, 50);
+    Reply(430);
+    giq_option(4, NAME, 431, Node019, NEUTRAL_REACTION);
+    giq_option(4, NAME, 432, Node003, NEUTRAL_REACTION);
 end
 
 procedure Node040
 begin
-    gsay_reply(74, 500);
-    giq_option(4, 74, 501, Node019, 50);
-    giq_option(4, 74, 502, Node999, 50);
+    Reply(500);
+    giq_option(4, NAME, 501, Node019, NEUTRAL_REACTION);
+    giq_option(4, NAME, 502, Node999, NEUTRAL_REACTION);
 end
-
