diff --git a/Games/Fallout2.Steam/Original/KCDUNTON.ssl b/Games/Fallout2.Steam/ReDefine/KCDUNTON.ssl
index 5400675..7f3c7b7 100644
--- a/Games/Fallout2.Steam/Original/KCDUNTON.ssl
+++ b/Games/Fallout2.Steam/ReDefine/KCDUNTON.ssl
@@ -278,17 +278,17 @@ end
 procedure critter_p_proc
 begin
     if (active) then begin
-        if ((global_var(204) == 1) and obj_can_see_obj(self_obj, dude_obj)) then begin
-            attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+        if ((global_var(GVAR_ENEMY_DUNTON) == 1) and obj_can_see_obj(self_obj, dude_obj)) then begin
+            attack(dude_obj);
         end
         else begin
-            if (cur_map_index == 9) then begin
+            if (CUR_MAP_KLADWTWN) then begin
                 call checkarea();
                 if ((area == 2) and (map_var(17) > 0) and obj_can_see_obj(self_obj, dude_obj)) then begin
                     set_map_var(17, 0);
-                    float_msg(self_obj, message_str(87, 600), 2);
-                    set_global_var(204, 1);
-                    attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+                    float_msg(self_obj, mstr(600), FLOAT_MSG_RED);
+                    set_global_var(GVAR_ENEMY_DUNTON, 1);
+                    attack(dude_obj);
                 end
                 else begin
                     needtomove := 0;
@@ -307,8 +307,8 @@ begin
                 end
             end
             else begin
-                if ((global_var(76) == 1) or (global_var(75) == 1) or (map_var(5) >= map_var(19))) then begin
-                    set_global_var(184, 1);
+                if ((global_var(GVAR_RUSTLE_FAIL) == 1) or (global_var(GVAR_RUSTLE_FAIL_VIOLENT) == 1) or (map_var(5) >= map_var(19))) then begin
+                    set_global_var(GVAR_RUSTLE_OVER, 1);
                     dest := tile_num_in_direction(tile_num(self_obj), 5, 12);
                     animate_move_obj_to_tile(self_obj, dest, 1);
                     add_timer_event(self_obj, game_ticks(2), 1);
@@ -341,7 +341,7 @@ procedure pickup_p_proc
 begin
     if (active) then begin
         if (source_obj == dude_obj) then begin
-            set_global_var(204, 1);
+            set_global_var(GVAR_ENEMY_DUNTON, 1);
         end
     end
 end
@@ -352,7 +352,7 @@ begin
         script_overrides;
         if (local_var(2) == 0) then begin
             set_local_var(2, 1);
-            set_local_var(3, (get_critter_stat(dude_obj, 3) - get_critter_stat(self_obj, 3)) * 5);
+            set_local_var(3, (dude_charisma - self_charisma) * 5);
             debug_msg("Base Reaction == " + local_var(3));
         end
         Static_Reaction := 0;
@@ -364,14 +364,14 @@ begin
         else begin
             reaction_bonus_karma := -1 * (-1 * reaction_bonus_karma / 1000);
         end
-        if (global_var(48) >= 0) then begin
-            reaction_bonus_town_rep := global_var(48) / 2;
+        if (global_var(GVAR_TOWN_REP_KLAMATH) >= 0) then begin
+            reaction_bonus_town_rep := global_var(GVAR_TOWN_REP_KLAMATH) / 2;
         end
         else begin
-            reaction_bonus_town_rep := -1 * (-1 * global_var(48) / 2);
+            reaction_bonus_town_rep := -1 * (-1 * global_var(GVAR_TOWN_REP_KLAMATH) / 2);
         end
-        if (has_trait(2, dude_obj, 13)) then begin
-            if (get_critter_stat(dude_obj, 34) == get_critter_stat(self_obj, 34)) then begin
+        if (has_trait(TRAIT_TRAIT, dude_obj, TRAIT_sex_appeal)) then begin
+            if (dude_gender == self_gender) then begin
                 Static_Reaction := Static_Reaction + -20;
             end
             else begin
@@ -387,48 +387,48 @@ begin
             debug_msg("Karma Reaction Bonus == " + Static_Reaction);
             Static_Reaction := Static_Reaction - reaction_bonus_town_rep;
             debug_msg("Town Rep Bonus == " + Static_Reaction);
-            set_global_var(37, 0);
-            set_global_var(38, 0);
-            set_global_var(39, 0);
-            set_global_var(40, 0);
-            set_global_var(41, 0);
-            set_global_var(42, 0);
-            set_global_var(43, 0);
-            set_global_var(44, 0);
-            set_global_var(45, 0);
-            if (global_var(0) >= 1000) then begin
-                set_global_var(37, 1);
+            set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+            set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+            set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+            set_global_var(GVAR_KARMA_DEFENDER, 0);
+            set_global_var(GVAR_KARMA_WANDERER, 0);
+            set_global_var(GVAR_KARMA_BETRAYER, 0);
+            set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+            set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+            set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+                set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
             end
             else begin
-                if (global_var(0) >= 750) then begin
-                    set_global_var(38, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                    set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
                 end
                 else begin
-                    if (global_var(0) >= 500) then begin
-                        set_global_var(39, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                        set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                     end
                     else begin
-                        if (global_var(0) >= 250) then begin
-                            set_global_var(40, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                            set_global_var(GVAR_KARMA_DEFENDER, 1);
                         end
                         else begin
-                            if (global_var(0) > -250) then begin
-                                set_global_var(41, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                                set_global_var(GVAR_KARMA_WANDERER, 1);
                             end
                             else begin
-                                if (global_var(0) > -500) then begin
-                                    set_global_var(42, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                    set_global_var(GVAR_KARMA_BETRAYER, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -750) then begin
-                                        set_global_var(43, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                        set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                     end
                                     else begin
-                                        if (global_var(0) > -1000) then begin
-                                            set_global_var(44, 1);
+                                        if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                            set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                         end
                                         else begin
-                                            set_global_var(45, 1);
+                                            set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                         end
                                     end
                                 end
@@ -437,35 +437,35 @@ begin
                     end
                 end
             end
-            if (global_var(37) == 1) then begin
+            if (global_var(GVAR_KARMA_HOLY_WARRIOR) == 1) then begin
                 Static_Reaction := Static_Reaction - 20;
             end
             else begin
-                if (global_var(38) == 1) then begin
+                if (global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES) == 1) then begin
                     Static_Reaction := Static_Reaction - 15;
                 end
                 else begin
-                    if (global_var(39) == 1) then begin
+                    if (global_var(GVAR_KARMA_SHIELD_OF_HOPE) == 1) then begin
                         Static_Reaction := Static_Reaction - 10;
                     end
                     else begin
-                        if (global_var(40) == 1) then begin
+                        if (global_var(GVAR_KARMA_DEFENDER) == 1) then begin
                             Static_Reaction := Static_Reaction - 5;
                         end
                         else begin
-                            if (global_var(41) == 1) then begin
+                            if (global_var(GVAR_KARMA_WANDERER) == 1) then begin
                                 Static_Reaction := Static_Reaction - 0;
                             end
                             else begin
-                                if (global_var(42) == 1) then begin
+                                if (global_var(GVAR_KARMA_BETRAYER) == 1) then begin
                                     Static_Reaction := Static_Reaction - -5;
                                 end
                                 else begin
-                                    if (global_var(43) == 1) then begin
+                                    if (global_var(GVAR_KARMA_SWORD_OF_DESPAIR) == 1) then begin
                                         Static_Reaction := Static_Reaction - -10;
                                     end
                                     else begin
-                                        if (global_var(44) == 1) then begin
+                                        if (global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE) == 1) then begin
                                             Static_Reaction := Static_Reaction - -15;
                                         end
                                         else begin
@@ -485,48 +485,48 @@ begin
             debug_msg("Karma Reaction Bonus == " + Static_Reaction);
             Static_Reaction := Static_Reaction + reaction_bonus_town_rep;
             debug_msg("Town Rep Bonus == " + Static_Reaction);
-            set_global_var(37, 0);
-            set_global_var(38, 0);
-            set_global_var(39, 0);
-            set_global_var(40, 0);
-            set_global_var(41, 0);
-            set_global_var(42, 0);
-            set_global_var(43, 0);
-            set_global_var(44, 0);
-            set_global_var(45, 0);
-            if (global_var(0) >= 1000) then begin
-                set_global_var(37, 1);
+            set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+            set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+            set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+            set_global_var(GVAR_KARMA_DEFENDER, 0);
+            set_global_var(GVAR_KARMA_WANDERER, 0);
+            set_global_var(GVAR_KARMA_BETRAYER, 0);
+            set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+            set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+            set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+                set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
             end
             else begin
-                if (global_var(0) >= 750) then begin
-                    set_global_var(38, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                    set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
                 end
                 else begin
-                    if (global_var(0) >= 500) then begin
-                        set_global_var(39, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                        set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                     end
                     else begin
-                        if (global_var(0) >= 250) then begin
-                            set_global_var(40, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                            set_global_var(GVAR_KARMA_DEFENDER, 1);
                         end
                         else begin
-                            if (global_var(0) > -250) then begin
-                                set_global_var(41, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                                set_global_var(GVAR_KARMA_WANDERER, 1);
                             end
                             else begin
-                                if (global_var(0) > -500) then begin
-                                    set_global_var(42, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                    set_global_var(GVAR_KARMA_BETRAYER, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -750) then begin
-                                        set_global_var(43, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                        set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                     end
                                     else begin
-                                        if (global_var(0) > -1000) then begin
-                                            set_global_var(44, 1);
+                                        if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                            set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                         end
                                         else begin
-                                            set_global_var(45, 1);
+                                            set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                         end
                                     end
                                 end
@@ -535,35 +535,35 @@ begin
                     end
                 end
             end
-            if (global_var(37) == 1) then begin
+            if (global_var(GVAR_KARMA_HOLY_WARRIOR) == 1) then begin
                 Static_Reaction := Static_Reaction + 20;
             end
             else begin
-                if (global_var(38) == 1) then begin
+                if (global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES) == 1) then begin
                     Static_Reaction := Static_Reaction + 15;
                 end
                 else begin
-                    if (global_var(39) == 1) then begin
+                    if (global_var(GVAR_KARMA_SHIELD_OF_HOPE) == 1) then begin
                         Static_Reaction := Static_Reaction + 10;
                     end
                     else begin
-                        if (global_var(40) == 1) then begin
+                        if (global_var(GVAR_KARMA_DEFENDER) == 1) then begin
                             Static_Reaction := Static_Reaction + 5;
                         end
                         else begin
-                            if (global_var(41) == 1) then begin
+                            if (global_var(GVAR_KARMA_WANDERER) == 1) then begin
                                 Static_Reaction := Static_Reaction + 0;
                             end
                             else begin
-                                if (global_var(42) == 1) then begin
+                                if (global_var(GVAR_KARMA_BETRAYER) == 1) then begin
                                     Static_Reaction := Static_Reaction + -5;
                                 end
                                 else begin
-                                    if (global_var(43) == 1) then begin
+                                    if (global_var(GVAR_KARMA_SWORD_OF_DESPAIR) == 1) then begin
                                         Static_Reaction := Static_Reaction + -10;
                                     end
                                     else begin
-                                        if (global_var(44) == 1) then begin
+                                        if (global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE) == 1) then begin
                                             Static_Reaction := Static_Reaction + -15;
                                         end
                                         else begin
@@ -579,39 +579,39 @@ begin
             debug_msg("Good Person Reaction Bonus == " + Static_Reaction);
         end
         if (Slavery_Tolerant == 0) then begin
-            if (global_var(11) == 1) then begin
+            if (global_var(GVAR_REPUTATION_SLAVER) == 1) then begin
                 Static_Reaction := Static_Reaction + (2 * -25);
             end
         end
         else begin
             if (Slavery_Tolerant == 1) then begin
-                if (global_var(11) == 1) then begin
+                if (global_var(GVAR_REPUTATION_SLAVER) == 1) then begin
                     Static_Reaction := Static_Reaction + -25;
                 end
             end
             else begin
                 if (Slavery_Tolerant == 3) then begin
-                    if (global_var(11) == 1) then begin
+                    if (global_var(GVAR_REPUTATION_SLAVER) == 1) then begin
                         Static_Reaction := Static_Reaction - -25;
                     end
                 end
             end
         end
         debug_msg("Slaver + Aligned Reaction == " + Static_Reaction);
-        Static_Reaction := Static_Reaction + (10 * has_trait(0, dude_obj, 10));
+        Static_Reaction := Static_Reaction + (10 * has_trait(TRAIT_PERK, dude_obj, PERK_presence));
         debug_msg("Presence Reaction == " + Static_Reaction);
-        if (global_var(1) >= 2) then begin
+        if (global_var(GVAR_CHILDKILLER_REPUTATION) >= 2) then begin
             Static_Reaction := Static_Reaction + -30;
         end
         debug_msg("Childkiller Reaction == " + Static_Reaction);
         set_local_var(0, local_var(3) + Static_Reaction);
         debug_msg("Initial Reaction == " + local_var(0));
-        if (cur_map_index == 9) then begin
+        if (CUR_MAP_KLADWTWN) then begin
             move_obj_inven_to_obj(klam_dunton_box_obj, self_obj);
         end
         call checkarea();
         who_am_i := 1 + random(0, 1);
-        if (global_var(204) == 1) then begin
+        if (global_var(GVAR_ENEMY_DUNTON) == 1) then begin
             call Node998();
         end
         else begin
@@ -619,7 +619,7 @@ begin
                 call Node035();
             end
             else begin
-                if (get_critter_stat(dude_obj, 4) <= (0 - -3)) then begin
+                if (dude_iq <= (0 - -3)) then begin
                     start_gdialog(87, self_obj, 4, -1, -1);
                     gsay_start;
                     call Node032();
@@ -630,19 +630,19 @@ begin
                     if (area == 3) then begin
                         start_gdialog(87, self_obj, 4, -1, -1);
                         gsay_start;
-                        if (global_var(76) == 1) then begin
+                        if (global_var(GVAR_RUSTLE_FAIL) == 1) then begin
                             call Node007();
                         end
                         else begin
-                            if (global_var(77) == 1) then begin
+                            if (global_var(GVAR_RUSTLE_SUCCESS) == 1) then begin
                                 call Node008();
                             end
                             else begin
-                                if (global_var(180) == 1) then begin
+                                if (global_var(GVAR_RUSTLE_REFUSE) == 1) then begin
                                     call Node012();
                                 end
                                 else begin
-                                    if (global_var(179) == 1) then begin
+                                    if (global_var(GVAR_RUSTLE_ACCEPT) == 1) then begin
                                         call Node025();
                                     end
                                     else begin
@@ -655,7 +655,7 @@ begin
                         end_dialogue;
                     end
                     else begin
-                        if (global_var(74) == 1) then begin
+                        if (global_var(GVAR_DUNTONS_ANGRY) == 1) then begin
                             start_gdialog(87, self_obj, 4, -1, -1);
                             gsay_start;
                             call Node001();
@@ -663,7 +663,7 @@ begin
                             end_dialogue;
                         end
                         else begin
-                            if (global_var(75) == 1) then begin
+                            if (global_var(GVAR_RUSTLE_FAIL_VIOLENT) == 1) then begin
                                 start_gdialog(87, self_obj, 4, -1, -1);
                                 gsay_start;
                                 call Node004();
@@ -671,7 +671,7 @@ begin
                                 end_dialogue;
                             end
                             else begin
-                                if (global_var(76) == 1) then begin
+                                if (global_var(GVAR_RUSTLE_FAIL) == 1) then begin
                                     start_gdialog(87, self_obj, 4, -1, -1);
                                     gsay_start;
                                     call Node007();
@@ -679,7 +679,7 @@ begin
                                     end_dialogue;
                                 end
                                 else begin
-                                    if (global_var(77) == 1) then begin
+                                    if (global_var(GVAR_RUSTLE_SUCCESS) == 1) then begin
                                         start_gdialog(87, self_obj, 4, -1, -1);
                                         gsay_start;
                                         call Node008();
@@ -687,7 +687,7 @@ begin
                                         end_dialogue;
                                     end
                                     else begin
-                                        if (global_var(180) == 1) then begin
+                                        if (global_var(GVAR_RUSTLE_REFUSE) == 1) then begin
                                             start_gdialog(87, self_obj, 4, -1, -1);
                                             gsay_start;
                                             call Node012();
@@ -695,7 +695,7 @@ begin
                                             end_dialogue;
                                         end
                                         else begin
-                                            if ((global_var(179) == 1) and (global_var(184) == 0)) then begin
+                                            if ((global_var(GVAR_RUSTLE_ACCEPT) == 1) and (global_var(GVAR_RUSTLE_OVER) == 0)) then begin
                                                 start_gdialog(87, self_obj, 4, -1, -1);
                                                 gsay_start;
                                                 call Node014();
@@ -703,7 +703,7 @@ begin
                                                 end_dialogue;
                                             end
                                             else begin
-                                                if ((global_var(179) == 1) and (global_var(184) == 1) and (local_var(10) == 0)) then begin
+                                                if ((global_var(GVAR_RUSTLE_ACCEPT) == 1) and (global_var(GVAR_RUSTLE_OVER) == 1) and (local_var(10) == 0)) then begin
                                                     set_local_var(10, 1);
                                                     start_gdialog(87, self_obj, 4, -1, -1);
                                                     gsay_start;
@@ -742,7 +742,7 @@ begin
                 end
             end
         end
-        if (cur_map_index == 9) then begin
+        if (CUR_MAP_KLADWTWN) then begin
             move_obj_inven_to_obj(self_obj, klam_dunton_box_obj);
         end
     end
@@ -751,59 +751,59 @@ end
 procedure destroy_p_proc
 begin
     if (active) then begin
-        if (cur_map_index == 9) then begin
+        if (CUR_MAP_KLADWTWN) then begin
             move_obj_inven_to_obj(klam_dunton_box_obj, self_obj);
         end
         if (source_obj == dude_obj) then begin
-            set_global_var(4, global_var(4) + 1);
-            if (metarule(51, self_obj) == 2) then begin
+            set_global_var(GVAR_BAD_MONSTER, global_var(GVAR_BAD_MONSTER) + 1);
+            if (critter_kill_type(self_obj) == KILL_TYPE_children_kills) then begin
                 if (source_obj == dude_obj) then begin
-                    set_global_var(2, 0);
-                    set_global_var(1, global_var(1) + 1);
-                    set_global_var(0, global_var(0) + -15);
+                    set_global_var(GVAR_CHAMPION_REPUTATION, 0);
+                    set_global_var(GVAR_CHILDKILLER_REPUTATION, global_var(GVAR_CHILDKILLER_REPUTATION) + 1);
+                    set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + -15);
                     debug_msg("Player gains " + -15 + " Karma Points.");
-                    set_global_var(37, 0);
-                    set_global_var(38, 0);
-                    set_global_var(39, 0);
-                    set_global_var(40, 0);
-                    set_global_var(41, 0);
-                    set_global_var(42, 0);
-                    set_global_var(43, 0);
-                    set_global_var(44, 0);
-                    set_global_var(45, 0);
-                    if (global_var(0) >= 1000) then begin
-                        set_global_var(37, 1);
+                    set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+                    set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+                    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+                    set_global_var(GVAR_KARMA_DEFENDER, 0);
+                    set_global_var(GVAR_KARMA_WANDERER, 0);
+                    set_global_var(GVAR_KARMA_BETRAYER, 0);
+                    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+                    set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+                    set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+                        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
                     end
                     else begin
-                        if (global_var(0) >= 750) then begin
-                            set_global_var(38, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                            set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
                         end
                         else begin
-                            if (global_var(0) >= 500) then begin
-                                set_global_var(39, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                                set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                             end
                             else begin
-                                if (global_var(0) >= 250) then begin
-                                    set_global_var(40, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                                    set_global_var(GVAR_KARMA_DEFENDER, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -250) then begin
-                                        set_global_var(41, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                                        set_global_var(GVAR_KARMA_WANDERER, 1);
                                     end
                                     else begin
-                                        if (global_var(0) > -500) then begin
-                                            set_global_var(42, 1);
+                                        if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                            set_global_var(GVAR_KARMA_BETRAYER, 1);
                                         end
                                         else begin
-                                            if (global_var(0) > -750) then begin
-                                                set_global_var(43, 1);
+                                            if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                                set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                             end
                                             else begin
-                                                if (global_var(0) > -1000) then begin
-                                                    set_global_var(44, 1);
+                                                if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                                    set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                                 end
                                                 else begin
-                                                    set_global_var(45, 1);
+                                                    set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                                 end
                                             end
                                         end
@@ -812,62 +812,62 @@ begin
                             end
                         end
                     end
-                    set_global_var(48, global_var(48) + -8);
+                    set_global_var(GVAR_TOWN_REP_KLAMATH, global_var(GVAR_TOWN_REP_KLAMATH) + -8);
                     debug_msg("Added " + -8 + " to Town Rep");
                 end
             end
-            if (((global_var(4) + global_var(5)) >= 25) and ((global_var(5) > (2 * global_var(4))) or (global_var(3) == 1))) then begin
-                set_global_var(3, 1);
-                set_global_var(2, 0);
+            if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
+                set_global_var(GVAR_BERSERKER_REPUTATION, 1);
+                set_global_var(GVAR_CHAMPION_REPUTATION, 0);
             end
-            if (((global_var(4) + global_var(5)) >= 25) and ((global_var(4) > (3 * global_var(5))) or (global_var(2) == 1)) and (global_var(1) == 0)) then begin
-                set_global_var(2, 1);
-                set_global_var(3, 0);
+            if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(4) > (3 * global_var(GVAR_GOOD_MONSTER))) or (global_var(GVAR_CHAMPION_REPUTATION) == 1)) and (global_var(GVAR_CHILDKILLER_REPUTATION) == 0)) then begin
+                set_global_var(GVAR_CHAMPION_REPUTATION, 1);
+                set_global_var(GVAR_BERSERKER_REPUTATION, 0);
             end
-            set_global_var(0, global_var(0) + 5);
+            set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + 5);
             debug_msg("Player gains " + 5 + " Karma Points.");
-            set_global_var(37, 0);
-            set_global_var(38, 0);
-            set_global_var(39, 0);
-            set_global_var(40, 0);
-            set_global_var(41, 0);
-            set_global_var(42, 0);
-            set_global_var(43, 0);
-            set_global_var(44, 0);
-            set_global_var(45, 0);
-            if (global_var(0) >= 1000) then begin
-                set_global_var(37, 1);
+            set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+            set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+            set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+            set_global_var(GVAR_KARMA_DEFENDER, 0);
+            set_global_var(GVAR_KARMA_WANDERER, 0);
+            set_global_var(GVAR_KARMA_BETRAYER, 0);
+            set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+            set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+            set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+                set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
             end
             else begin
-                if (global_var(0) >= 750) then begin
-                    set_global_var(38, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                    set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
                 end
                 else begin
-                    if (global_var(0) >= 500) then begin
-                        set_global_var(39, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                        set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                     end
                     else begin
-                        if (global_var(0) >= 250) then begin
-                            set_global_var(40, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                            set_global_var(GVAR_KARMA_DEFENDER, 1);
                         end
                         else begin
-                            if (global_var(0) > -250) then begin
-                                set_global_var(41, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                                set_global_var(GVAR_KARMA_WANDERER, 1);
                             end
                             else begin
-                                if (global_var(0) > -500) then begin
-                                    set_global_var(42, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                    set_global_var(GVAR_KARMA_BETRAYER, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -750) then begin
-                                        set_global_var(43, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                        set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                     end
                                     else begin
-                                        if (global_var(0) > -1000) then begin
-                                            set_global_var(44, 1);
+                                        if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                            set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                         end
                                         else begin
-                                            set_global_var(45, 1);
+                                            set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                         end
                                     end
                                 end
@@ -876,10 +876,10 @@ begin
                     end
                 end
             end
-            set_global_var(48, global_var(48) + 2);
+            set_global_var(GVAR_TOWN_REP_KLAMATH, global_var(GVAR_TOWN_REP_KLAMATH) + 2);
             debug_msg("Added " + 2 + " to Town Rep");
         end
-        set_global_var(188, 1);
+        set_global_var(GVAR_DUNTON_DEAD, 1);
     end
 end
 
@@ -887,7 +887,7 @@ procedure look_at_p_proc
 begin
     if (active) then begin
         script_overrides;
-        display_msg(message_str(87, 100));
+        display_mstr(100);
     end
 end
 
@@ -895,7 +895,7 @@ procedure description_p_proc
 begin
     if (active) then begin
         script_overrides;
-        display_msg(message_str(87, 101));
+        display_mstr(101);
     end
 end
 
@@ -907,7 +907,7 @@ procedure damage_p_proc
 begin
     if (active) then begin
         if (source_obj == dude_obj) then begin
-            set_global_var(204, 1);
+            set_global_var(GVAR_ENEMY_DUNTON, 1);
         end
     end
 end
@@ -915,7 +915,7 @@ end
 procedure combat_p_proc
 begin
     if (obj_is_carrying_obj_pid(self_obj, dunton_weapon) == 0) then begin
-        item := create_object_sid(dunton_weapon, 0, 0, -1);
+        item := create_object(dunton_weapon, 0, 0);
         add_obj_to_inven(self_obj, item);
     end
 end
@@ -923,16 +923,16 @@ end
 procedure map_enter_p_proc
 begin
     Only_Once := 0;
-    critter_add_trait(self_obj, 1, 5, 126);
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_DUNTON);
     dunton_weapon := 234;
-    if (cur_map_index == 9) then begin
-        critter_add_trait(self_obj, 1, 6, 43);
+    if (CUR_MAP_KLADWTWN) then begin
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 43);
         set_local_var(5, 0);
         active := 1;
     end
     else begin
-        if (cur_map_index == 14) then begin
-            critter_add_trait(self_obj, 1, 6, 35);
+        if (CUR_MAP_KLAGRAZ) then begin
+            critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 35);
             call rustle_status();
             call should_i_be_here();
         end
@@ -948,7 +948,7 @@ procedure checkarea
 begin
     here := tile_num(self_obj);
     area := 0;
-    if (cur_map_index == 14) then begin
+    if (CUR_MAP_KLAGRAZ) then begin
         area := 3;
     end
     else begin
@@ -995,36 +995,36 @@ end
 
 procedure duntonbarter
 begin
-    gdialog_mod_barter(0);
+    gdialog_barter;
     set_local_var(6, 1);
-    gsay_message(87, "", 50);
+    gsay_message(NAME, "", NEUTRAL_REACTION);
     call Node013();
 end
 
 procedure rustle_status
 begin
-    if (global_var(179) == 1) then begin
+    if (global_var(GVAR_RUSTLE_ACCEPT) == 1) then begin
         rustle_triggered := 1;
     end
-    if (global_var(180) == 1) then begin
+    if (global_var(GVAR_RUSTLE_REFUSE) == 1) then begin
         rustle_triggered := 1;
     end
-    if (global_var(182) == 1) then begin
+    if (global_var(GVAR_TORR_GUARD_STATUS) == 1) then begin
         rustle_triggered := 1;
     end
-    if (global_var(77) == 1) then begin
+    if (global_var(GVAR_RUSTLE_SUCCESS) == 1) then begin
         rustle_triggered := 0;
     end
-    if (global_var(76) == 1) then begin
+    if (global_var(GVAR_RUSTLE_FAIL) == 1) then begin
         rustle_triggered := 0;
     end
-    if (global_var(70) == 1) then begin
+    if (global_var(GVAR_TORR_DEAD) == 1) then begin
         rustle_triggered := 0;
     end
-    if (global_var(188) == 1) then begin
+    if (global_var(GVAR_DUNTON_DEAD) == 1) then begin
         rustle_triggered := 0;
     end
-    if (global_var(184) == 1) then begin
+    if (global_var(GVAR_RUSTLE_OVER) == 1) then begin
         rustle_triggered := 0;
     end
 end
@@ -1045,7 +1045,7 @@ end
 procedure combat
 begin
     debug_msg("COMBAT");
-    set_global_var(204, 1);
+    set_global_var(GVAR_ENEMY_DUNTON, 1);
 end
 
 procedure scamper
@@ -1054,7 +1054,7 @@ begin
         set_map_var(3, 1);
         call rustle_status();
         gfade_out(900);
-        if ((global_var(184) == 1) and (cur_map_index == 14)) then begin
+        if ((global_var(GVAR_RUSTLE_OVER) == 1) and (CUR_MAP_KLAGRAZ)) then begin
             reg_anim_func(2, self_obj);
             move_to(self_obj, 0, 0);
         end
@@ -1537,14 +1537,14 @@ end
 
 procedure Node938
 begin
-    gsay_reply(87, 510);
-    giq_option(1, 87, 520, Node016, 50);
+    Reply(510);
+    giq_option(1, NAME, 520, Node016, NEUTRAL_REACTION);
 end
 
 procedure Node939
 begin
-    gsay_reply(87, 500);
-    giq_option(1, 87, 520, Node017, 50);
+    Reply(500);
+    giq_option(1, NAME, 520, Node017, NEUTRAL_REACTION);
 end
 
 procedure Node940
@@ -1694,7 +1694,7 @@ begin
         set_map_var(7, 1);
         if (local_var(8) == 0) then begin
             set_local_var(8, 1);
-            set_global_var(48, global_var(48) + -5);
+            set_global_var(GVAR_TOWN_REP_KLAMATH, global_var(GVAR_TOWN_REP_KLAMATH) + -5);
         end
         call Node938();
     end
@@ -1702,7 +1702,7 @@ begin
         set_map_var(8, 1);
         if (local_var(9) == 0) then begin
             set_local_var(9, 1);
-            set_global_var(48, global_var(48) + -2);
+            set_global_var(GVAR_TOWN_REP_KLAMATH, global_var(GVAR_TOWN_REP_KLAMATH) + -2);
         end
         call Node939();
     end
@@ -2454,15 +2454,15 @@ end
 
 procedure Node973
 begin
-    set_global_var(102, 1);
-    set_global_var(179, 1);
+    set_global_var(GVAR_QUEST_KILL_EVIL_PLANTS, 1);
+    set_global_var(GVAR_RUSTLE_ACCEPT, 1);
     call Node003();
 end
 
 procedure Node975
 begin
-    set_global_var(102, 1);
-    set_global_var(179, 1);
+    set_global_var(GVAR_QUEST_KILL_EVIL_PLANTS, 1);
+    set_global_var(GVAR_RUSTLE_ACCEPT, 1);
     if (area != 3) then begin
         gfade_out(600);
         game_time_advance(game_ticks(900));
@@ -2706,8 +2706,8 @@ begin
             end
         end
     end
-    set_global_var(102, 1);
-    set_global_var(179, 1);
+    set_global_var(GVAR_QUEST_KILL_EVIL_PLANTS, 1);
+    set_global_var(GVAR_RUSTLE_ACCEPT, 1);
 end
 
 procedure Node990
@@ -2760,22 +2760,22 @@ end
 
 procedure Node991
 begin
-    gsay_reply(87, 198);
-    giq_option(4, 87, 202, Node975, 49);
-    giq_option(4, 87, 204, Node993, 51);
+    Reply(198);
+    giq_option(4, NAME, 202, Node975, GOOD_REACTION);
+    giq_option(4, NAME, 204, Node993, BAD_REACTION);
 end
 
 procedure Node992
 begin
-    gsay_reply(87, message_str(87, 198) + message_str(87, 601));
-    giq_option(4, 87, 202, Node975, 49);
-    giq_option(4, 87, 204, Node993, 51);
+    Reply(mstr(198) + mstr(601));
+    giq_option(4, NAME, 202, Node975, GOOD_REACTION);
+    giq_option(4, NAME, 204, Node993, BAD_REACTION);
 end
 
 procedure Node993
 begin
     if (from_14) then begin
-        set_global_var(180, 1);
+        set_global_var(GVAR_RUSTLE_REFUSE, 1);
     end
     else begin
         set_local_var(3, local_var(3) + -10);
@@ -2827,52 +2827,52 @@ end
 
 procedure Node994
 begin
-    set_global_var(76, 1);
-    set_global_var(182, 2);
-    set_global_var(0, global_var(0) + 50);
+    set_global_var(GVAR_RUSTLE_FAIL, 1);
+    set_global_var(GVAR_TORR_GUARD_STATUS, 2);
+    set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + 50);
     debug_msg("Player gains " + 50 + " Karma Points.");
-    set_global_var(37, 0);
-    set_global_var(38, 0);
-    set_global_var(39, 0);
-    set_global_var(40, 0);
-    set_global_var(41, 0);
-    set_global_var(42, 0);
-    set_global_var(43, 0);
-    set_global_var(44, 0);
-    set_global_var(45, 0);
-    if (global_var(0) >= 1000) then begin
-        set_global_var(37, 1);
-    end
-    else begin
-        if (global_var(0) >= 750) then begin
-            set_global_var(38, 1);
+    set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+    set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+    set_global_var(GVAR_KARMA_DEFENDER, 0);
+    set_global_var(GVAR_KARMA_WANDERER, 0);
+    set_global_var(GVAR_KARMA_BETRAYER, 0);
+    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+    set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+    set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+    if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
+    end
+    else begin
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+            set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
         end
         else begin
-            if (global_var(0) >= 500) then begin
-                set_global_var(39, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
             end
             else begin
-                if (global_var(0) >= 250) then begin
-                    set_global_var(40, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                    set_global_var(GVAR_KARMA_DEFENDER, 1);
                 end
                 else begin
-                    if (global_var(0) > -250) then begin
-                        set_global_var(41, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                        set_global_var(GVAR_KARMA_WANDERER, 1);
                     end
                     else begin
-                        if (global_var(0) > -500) then begin
-                            set_global_var(42, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                            set_global_var(GVAR_KARMA_BETRAYER, 1);
                         end
                         else begin
-                            if (global_var(0) > -750) then begin
-                                set_global_var(43, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                             end
                             else begin
-                                if (global_var(0) > -1000) then begin
-                                    set_global_var(44, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                    set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                 end
                                 else begin
-                                    set_global_var(45, 1);
+                                    set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                 end
                             end
                         end
@@ -2886,7 +2886,7 @@ end
 
 procedure Node995
 begin
-    if (get_critter_stat(dude_obj, 3) >= 7) then begin
+    if (dude_charisma >= 7) then begin
         set_local_var(3, 100);
         set_local_var(1, 4);
     end
@@ -2904,7 +2904,7 @@ end
 
 procedure Node997
 begin
-    if (get_critter_stat(dude_obj, 3) >= 8) then begin
+    if (dude_charisma >= 8) then begin
         call Node004();
     end
     else begin
@@ -2914,7 +2914,7 @@ end
 
 procedure Node998
 begin
-    set_global_var(204, 1);
+    set_global_var(GVAR_ENEMY_DUNTON, 1);
 end
 
 procedure Node999
@@ -2923,76 +2923,76 @@ end
 
 procedure Node001
 begin
-    gsay_reply(87, 110);
-    giq_option(4, 87, 111, Node936, 49);
-    giq_option(4, 87, 112, Node999, 50);
-    giq_option(4, 87, 113, Node932, 51);
-    giq_option(4, 87, 114, Node950, 51);
+    Reply(110);
+    giq_option(4, NAME, 111, Node936, GOOD_REACTION);
+    giq_option(4, NAME, 112, Node999, NEUTRAL_REACTION);
+    giq_option(4, NAME, 113, Node932, BAD_REACTION);
+    giq_option(4, NAME, 114, Node950, BAD_REACTION);
 end
 
 procedure Node002
 begin
-    gsay_reply(87, 120);
-    giq_option(4, 87, 121, Node961, 49);
-    giq_option(4, 87, 122, Node999, 50);
-    giq_option(4, 87, 123, combat, 51);
+    Reply(120);
+    giq_option(4, NAME, 121, Node961, GOOD_REACTION);
+    giq_option(4, NAME, 122, Node999, NEUTRAL_REACTION);
+    giq_option(4, NAME, 123, combat, BAD_REACTION);
 end
 
 procedure Node003
 begin
     if (from_18) then begin
-        gsay_reply(87, 132);
+        Reply(132);
     end
     else begin
-        if (global_var(48) < 5) then begin
-            gsay_reply(87, 130);
+        if (global_var(GVAR_TOWN_REP_KLAMATH) < 5) then begin
+            Reply(130);
         end
         else begin
-            gsay_reply(87, 131);
+            Reply(131);
         end
     end
-    giq_option(4, 87, 133, Node019, 50);
+    giq_option(4, NAME, 133, Node019, NEUTRAL_REACTION);
     if (area == 2) then begin
-        giq_option(4, 87, 134, Node005, 50);
+        giq_option(4, NAME, 134, Node005, NEUTRAL_REACTION);
     end
     if (local_var(7) == 0) then begin
         set_local_var(7, 1);
-        giq_option(4, 87, 135, Node006, 50);
+        giq_option(4, NAME, 135, Node006, NEUTRAL_REACTION);
     end
-    giq_option(4, 87, 136, Node011, 50);
-    giq_option(4, 87, 137, Node020, 50);
-    giq_option(4, 87, 138, Node999, 50);
+    giq_option(4, NAME, 136, Node011, NEUTRAL_REACTION);
+    giq_option(4, NAME, 137, Node020, NEUTRAL_REACTION);
+    giq_option(4, NAME, 138, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node004
 begin
-    gsay_reply(87, 140);
-    if (global_var(179) == 1) then begin
-        giq_option(4, 87, 141, Node999, 50);
-        giq_option(4, 87, 142, Node980, 51);
+    Reply(140);
+    if (global_var(GVAR_RUSTLE_ACCEPT) == 1) then begin
+        giq_option(4, NAME, 141, Node999, NEUTRAL_REACTION);
+        giq_option(4, NAME, 142, Node980, BAD_REACTION);
     end
-    giq_option(4, 87, 143, Node990, 51);
+    giq_option(4, NAME, 143, Node990, BAD_REACTION);
 end
 
 procedure Node005
 begin
-    gsay_reply(87, 150);
-    giq_option(4, 87, 151, Node937, 50);
-    giq_option(4, 87, 152, Node003, 50);
-    giq_option(4, 87, 153, Node999, 50);
+    Reply(150);
+    giq_option(4, NAME, 151, Node937, NEUTRAL_REACTION);
+    giq_option(4, NAME, 152, Node003, NEUTRAL_REACTION);
+    giq_option(4, NAME, 153, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node006
 begin
-    if (global_var(180) == 1) then begin
+    if (global_var(GVAR_RUSTLE_REFUSE) == 1) then begin
         call Node06a();
     end
     else begin
-        if (global_var(184) == 1) then begin
+        if (global_var(GVAR_RUSTLE_OVER) == 1) then begin
             call Node06b();
         end
         else begin
-            if (global_var(48) > 4) then begin
+            if (global_var(GVAR_TOWN_REP_KLAMATH) > 4) then begin
                 call Node06c();
             end
             else begin
@@ -3004,135 +3004,135 @@ end
 
 procedure Node06a
 begin
-    gsay_reply(87, 160);
-    if ((global_var(48) > 4) and (area == 3)) then begin
-        giq_option(4, 87, 531, Node999, 50);
+    Reply(160);
+    if ((global_var(GVAR_TOWN_REP_KLAMATH) > 4) and (area == 3)) then begin
+        giq_option(4, NAME, 531, Node999, NEUTRAL_REACTION);
     end
     if (area != 3) then begin
-        giq_option(4, 87, 166, Node003, 50);
+        giq_option(4, NAME, 166, Node003, NEUTRAL_REACTION);
     end
-    if (global_var(184) == 0) then begin
-        giq_option(4, 87, 167, Node948, 49);
+    if (global_var(GVAR_RUSTLE_OVER) == 0) then begin
+        giq_option(4, NAME, 167, Node948, GOOD_REACTION);
     end
-    giq_option(4, 87, 168, Node999, 50);
+    giq_option(4, NAME, 168, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node06b
 begin
-    gsay_reply(87, 161);
-    giq_option(4, 87, 169, Node999, 50);
+    Reply(161);
+    giq_option(4, NAME, 169, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node06c
 begin
     if (area == 3) then begin
-        gsay_reply(87, 530);
+        Reply(530);
     end
     else begin
-        gsay_reply(87, 161);
+        Reply(161);
     end
-    giq_option(4, 87, 169, Node999, 50);
+    giq_option(4, NAME, 169, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node06d
 begin
-    gsay_reply(87, 162);
+    Reply(162);
     if (area != 3) then begin
-        giq_option(4, 87, 163, Node945, 50);
-        giq_option(4, 87, 164, Node946, 50);
+        giq_option(4, NAME, 163, Node945, NEUTRAL_REACTION);
+        giq_option(4, NAME, 164, Node946, NEUTRAL_REACTION);
     end
-    giq_option(4, 87, 165, Node010, 50);
+    giq_option(4, NAME, 165, Node010, NEUTRAL_REACTION);
 end
 
 procedure Node007
 begin
-    gsay_reply(87, 170);
-    giq_option(4, 87, 171, Node955, 49);
+    Reply(170);
+    giq_option(4, NAME, 171, Node955, GOOD_REACTION);
     if (area != 3) then begin
-        giq_option(4, 87, 172, Node003, 50);
+        giq_option(4, NAME, 172, Node003, NEUTRAL_REACTION);
     end
-    giq_option(4, 87, 173, combat, 51);
+    giq_option(4, NAME, 173, combat, BAD_REACTION);
 end
 
 procedure Node008
 begin
-    if (global_var(181) == 1) then begin
-        gsay_reply(87, 180);
+    if (global_var(GVAR_RUSTLE_REWARD) == 1) then begin
+        Reply(180);
     end
     else begin
-        gsay_reply(87, 181);
+        Reply(181);
     end
-    if ((global_var(181) == 1) and (area != 3)) then begin
-        giq_option(4, 87, 182, Node983, 49);
+    if ((global_var(GVAR_RUSTLE_REWARD) == 1) and (area != 3)) then begin
+        giq_option(4, NAME, 182, Node983, GOOD_REACTION);
     end
-    if (global_var(181) == 0) then begin
-        giq_option(4, 87, 183, Node985, 49);
+    if (global_var(GVAR_RUSTLE_REWARD) == 0) then begin
+        giq_option(4, NAME, 183, Node985, GOOD_REACTION);
     end
-    giq_option(4, 87, 184, Node988, 51);
-    giq_option(4, 87, 185, Node999, 50);
+    giq_option(4, NAME, 184, Node988, BAD_REACTION);
+    giq_option(4, NAME, 185, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node009
 begin
-    item := create_object_sid(284, 0, 0, -1);
+    item := create_object(284, 0, 0);
     add_obj_to_inven(dude_obj, item);
-    item := create_object_sid(284, 0, 0, -1);
+    item := create_object(284, 0, 0);
     add_obj_to_inven(dude_obj, item);
-    item := create_object_sid(284, 0, 0, -1);
+    item := create_object(284, 0, 0);
     add_obj_to_inven(dude_obj, item);
-    item := create_object_sid(284, 0, 0, -1);
+    item := create_object(284, 0, 0);
     add_obj_to_inven(dude_obj, item);
-    item := create_object_sid(284, 0, 0, -1);
+    item := create_object(284, 0, 0);
     add_obj_to_inven(dude_obj, item);
     item_caps_adjust(dude_obj, 50);
-    set_global_var(181, 1);
-    display_msg(message_str(87, 700));
+    set_global_var(GVAR_RUSTLE_REWARD, 1);
+    display_mstr(700);
     give_exp_points(150);
-    display_msg(message_str(14, 100) + 150 + message_str(14, 101));
-    set_global_var(0, global_var(0) + -60);
+    display_msg(g_mstr(100) + 150 + g_mstr(101));
+    set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + -60);
     debug_msg("Player gains " + -60 + " Karma Points.");
-    set_global_var(37, 0);
-    set_global_var(38, 0);
-    set_global_var(39, 0);
-    set_global_var(40, 0);
-    set_global_var(41, 0);
-    set_global_var(42, 0);
-    set_global_var(43, 0);
-    set_global_var(44, 0);
-    set_global_var(45, 0);
-    if (global_var(0) >= 1000) then begin
-        set_global_var(37, 1);
-    end
-    else begin
-        if (global_var(0) >= 750) then begin
-            set_global_var(38, 1);
+    set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+    set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+    set_global_var(GVAR_KARMA_DEFENDER, 0);
+    set_global_var(GVAR_KARMA_WANDERER, 0);
+    set_global_var(GVAR_KARMA_BETRAYER, 0);
+    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+    set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+    set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+    if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
+    end
+    else begin
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+            set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
         end
         else begin
-            if (global_var(0) >= 500) then begin
-                set_global_var(39, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
             end
             else begin
-                if (global_var(0) >= 250) then begin
-                    set_global_var(40, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                    set_global_var(GVAR_KARMA_DEFENDER, 1);
                 end
                 else begin
-                    if (global_var(0) > -250) then begin
-                        set_global_var(41, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                        set_global_var(GVAR_KARMA_WANDERER, 1);
                     end
                     else begin
-                        if (global_var(0) > -500) then begin
-                            set_global_var(42, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                            set_global_var(GVAR_KARMA_BETRAYER, 1);
                         end
                         else begin
-                            if (global_var(0) > -750) then begin
-                                set_global_var(43, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                             end
                             else begin
-                                if (global_var(0) > -1000) then begin
-                                    set_global_var(44, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                    set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                 end
                                 else begin
-                                    set_global_var(45, 1);
+                                    set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                 end
                             end
                         end
@@ -3141,11 +3141,11 @@ begin
             end
         end
     end
-    gsay_reply(87, 190);
+    Reply(190);
     if (area != 3) then begin
-        giq_option(4, 87, 191, Node003, 50);
+        giq_option(4, NAME, 191, Node003, NEUTRAL_REACTION);
     end
-    giq_option(4, 87, 192, Node999, 50);
+    giq_option(4, NAME, 192, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node010
@@ -3164,23 +3164,23 @@ begin
     if (who_am_i == 1) then begin
         k := random(1, 5);
         if (k == 1) then begin
-            gsay_reply(87, 208);
+            Reply(208);
         end
         else begin
             if (k == 2) then begin
-                gsay_reply(87, 210);
+                Reply(210);
             end
             else begin
                 if (k == 3) then begin
-                    gsay_reply(87, 212);
+                    Reply(212);
                 end
                 else begin
                     if (k == 4) then begin
-                        gsay_reply(87, 214);
+                        Reply(214);
                     end
                     else begin
                         if (k == 5) then begin
-                            gsay_reply(87, 216);
+                            Reply(216);
                         end
                     end
                 end
@@ -3190,225 +3190,225 @@ begin
     else begin
         k := random(1, 5);
         if (k == 1) then begin
-            gsay_reply(87, 209);
+            Reply(209);
         end
         else begin
             if (k == 2) then begin
-                gsay_reply(87, 211);
+                Reply(211);
             end
             else begin
                 if (k == 3) then begin
-                    gsay_reply(87, 213);
+                    Reply(213);
                 end
                 else begin
                     if (k == 4) then begin
-                        gsay_reply(87, 215);
+                        Reply(215);
                     end
                     else begin
                         if (k == 5) then begin
-                            gsay_reply(87, 217);
+                            Reply(217);
                         end
                     end
                 end
             end
         end
     end
-    giq_option(4, 87, 218, Node935, 49);
-    giq_option(4, 87, 219, Node003, 50);
-    giq_option(4, 87, 220, Node931, 51);
-    giq_option(4, 87, 221, Node934, 51);
+    giq_option(4, NAME, 218, Node935, GOOD_REACTION);
+    giq_option(4, NAME, 219, Node003, NEUTRAL_REACTION);
+    giq_option(4, NAME, 220, Node931, BAD_REACTION);
+    giq_option(4, NAME, 221, Node934, BAD_REACTION);
     if (from_10) then begin
-        giq_option(4, 87, 222, Node010, 50);
+        giq_option(4, NAME, 222, Node010, NEUTRAL_REACTION);
     end
 end
 
 procedure Node012
 begin
-    set_global_var(180, 1);
-    gsay_reply(87, 230);
-    giq_option(4, 87, 231, Node960, 49);
+    set_global_var(GVAR_RUSTLE_REFUSE, 1);
+    Reply(230);
+    giq_option(4, NAME, 231, Node960, GOOD_REACTION);
     if (area != 3) then begin
-        giq_option(4, 87, 232, Node003, 50);
+        giq_option(4, NAME, 232, Node003, NEUTRAL_REACTION);
     end
-    giq_option(4, 87, 233, Node941, 51);
+    giq_option(4, NAME, 233, Node941, BAD_REACTION);
 end
 
 procedure Node013
 begin
-    gsay_reply(87, 240);
-    giq_option(4, 87, 241, Node937, 50);
-    giq_option(4, 87, 242, Node003, 50);
-    giq_option(4, 87, 243, Node981, 50);
-    giq_option(4, 87, 244, Node951, 50);
+    Reply(240);
+    giq_option(4, NAME, 241, Node937, NEUTRAL_REACTION);
+    giq_option(4, NAME, 242, Node003, NEUTRAL_REACTION);
+    giq_option(4, NAME, 243, Node981, NEUTRAL_REACTION);
+    giq_option(4, NAME, 244, Node951, NEUTRAL_REACTION);
 end
 
 procedure Node014
 begin
     from_14 := 1;
-    gsay_reply(87, 250);
-    giq_option(4, 87, 251, Node971, 51);
+    Reply(250);
+    giq_option(4, NAME, 251, Node971, BAD_REACTION);
     if (area != 3) then begin
-        giq_option(4, 87, 252, Node003, 50);
+        giq_option(4, NAME, 252, Node003, NEUTRAL_REACTION);
     end
-    giq_option(4, 87, 253, Node999, 50);
+    giq_option(4, NAME, 253, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node015
 begin
     if (map_var(7) == 1) then begin
-        gsay_reply(87, 259);
+        Reply(259);
     end
     else begin
         if (map_var(8) == 1) then begin
-            gsay_reply(87, 260);
+            Reply(260);
         end
         else begin
-            if (get_critter_stat(dude_obj, 0) < 5) then begin
-                if (global_var(48) > 4) then begin
-                    gsay_reply(87, 261);
+            if (dude_strength < 5) then begin
+                if (global_var(GVAR_TOWN_REP_KLAMATH) > 4) then begin
+                    Reply(261);
                 end
                 else begin
                     if (who_am_i == 1) then begin
-                        gsay_reply(87, 262);
+                        Reply(262);
                     end
                     else begin
-                        gsay_reply(87, 263);
+                        Reply(263);
                     end
                 end
             end
             else begin
                 if (who_am_i == 1) then begin
-                    if (global_var(48) > 4) then begin
-                        gsay_reply(87, 264);
+                    if (global_var(GVAR_TOWN_REP_KLAMATH) > 4) then begin
+                        Reply(264);
                     end
                     else begin
-                        gsay_reply(87, 266);
+                        Reply(266);
                     end
                 end
                 else begin
-                    if (global_var(48) > 4) then begin
-                        gsay_reply(87, 265);
+                    if (global_var(GVAR_TOWN_REP_KLAMATH) > 4) then begin
+                        Reply(265);
                     end
                     else begin
-                        gsay_reply(87, 267);
+                        Reply(267);
                     end
                 end
             end
         end
     end
-    giq_option(4, 87, 268, Node944, 49);
-    giq_option(4, 87, 269, Node018, 50);
-    giq_option(4, 87, 270, Node930, 51);
-    giq_option(4, 87, 271, Node999, 50);
+    giq_option(4, NAME, 268, Node944, GOOD_REACTION);
+    giq_option(4, NAME, 269, Node018, NEUTRAL_REACTION);
+    giq_option(4, NAME, 270, Node930, BAD_REACTION);
+    giq_option(4, NAME, 271, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node016
 begin
-    gsay_reply(87, 280);
-    giq_option(4, 87, 281, Node967, 49);
-    giq_option(4, 87, 282, Node999, 50);
-    giq_option(4, 87, 283, Node930, 51);
-    giq_option(4, 87, 284, Node941, 51);
+    Reply(280);
+    giq_option(4, NAME, 281, Node967, GOOD_REACTION);
+    giq_option(4, NAME, 282, Node999, NEUTRAL_REACTION);
+    giq_option(4, NAME, 283, Node930, BAD_REACTION);
+    giq_option(4, NAME, 284, Node941, BAD_REACTION);
 end
 
 procedure Node017
 begin
     if (map_var(9) == 1) then begin
-        gsay_reply(87, 291);
+        Reply(291);
     end
     else begin
         set_map_var(9, 1);
-        gsay_reply(87, 290);
+        Reply(290);
     end
-    giq_option(4, 87, 292, Node944, 49);
-    giq_option(4, 87, 293, Node003, 50);
-    giq_option(4, 87, 294, Node999, 50);
+    giq_option(4, NAME, 292, Node944, GOOD_REACTION);
+    giq_option(4, NAME, 293, Node003, NEUTRAL_REACTION);
+    giq_option(4, NAME, 294, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node018
 begin
     from_18 := 1;
-    gsay_reply(87, 300);
-    giq_option(4, 87, 301, Node935, 49);
-    giq_option(4, 87, 302, Node003, 50);
-    giq_option(4, 87, 303, Node941, 51);
-    giq_option(4, 87, 304, Node942, 51);
+    Reply(300);
+    giq_option(4, NAME, 301, Node935, GOOD_REACTION);
+    giq_option(4, NAME, 302, Node003, NEUTRAL_REACTION);
+    giq_option(4, NAME, 303, Node941, BAD_REACTION);
+    giq_option(4, NAME, 304, Node942, BAD_REACTION);
 end
 
 procedure Node019
 begin
-    gsay_reply(87, 310);
-    giq_option(4, 87, 311, Node003, 50);
-    giq_option(4, 87, 312, Node999, 50);
+    Reply(310);
+    giq_option(4, NAME, 311, Node003, NEUTRAL_REACTION);
+    giq_option(4, NAME, 312, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node020
 begin
-    gsay_reply(87, 320);
-    giq_option(4, 87, 321, Node003, 50);
-    giq_option(4, 87, 322, Node999, 50);
+    Reply(320);
+    giq_option(4, NAME, 321, Node003, NEUTRAL_REACTION);
+    giq_option(4, NAME, 322, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node021
 begin
     if (map_var(10) == 0) then begin
         set_map_var(10, 1);
-        gsay_reply(87, 330);
+        Reply(330);
     end
     else begin
-        gsay_reply(87, 331);
+        Reply(331);
     end
-    giq_option(4, 87, 332, Node952, 49);
-    giq_option(4, 87, 333, Node964, 49);
-    giq_option(4, 87, 334, Node003, 50);
-    giq_option(4, 87, 335, Node999, 50);
+    giq_option(4, NAME, 332, Node952, GOOD_REACTION);
+    giq_option(4, NAME, 333, Node964, GOOD_REACTION);
+    giq_option(4, NAME, 334, Node003, NEUTRAL_REACTION);
+    giq_option(4, NAME, 335, Node999, NEUTRAL_REACTION);
     if (is_success(do_check(dude_obj, 1, 0))) then begin
-        giq_option(4, 87, 336, Node022, 51);
+        giq_option(4, NAME, 336, Node022, BAD_REACTION);
     end
 end
 
 procedure Node022
 begin
-    gsay_reply(87, 340);
-    giq_option(4, 87, 341, Node972, 51);
-    giq_option(4, 87, 342, Node999, 50);
-    giq_option(4, 87, 343, Node021, 50);
+    Reply(340);
+    giq_option(4, NAME, 341, Node972, BAD_REACTION);
+    giq_option(4, NAME, 342, Node999, NEUTRAL_REACTION);
+    giq_option(4, NAME, 343, Node021, NEUTRAL_REACTION);
 end
 
 procedure Node023
 begin
     if (map_var(11) == 0) then begin
         set_map_var(11, 1);
-        gsay_reply(87, 350);
+        Reply(350);
     end
     else begin
         k := random(1, 7);
         if (k == 1) then begin
-            gsay_reply(87, 351);
+            Reply(351);
         end
         else begin
             if (k == 2) then begin
-                gsay_reply(87, 352);
+                Reply(352);
             end
             else begin
                 if (k == 3) then begin
-                    gsay_reply(87, 353);
+                    Reply(353);
                 end
                 else begin
                     if (k == 4) then begin
-                        gsay_reply(87, 354);
+                        Reply(354);
                     end
                     else begin
                         if (k == 5) then begin
-                            gsay_reply(87, 355);
+                            Reply(355);
                         end
                         else begin
                             if (k == 6) then begin
-                                gsay_reply(87, 356);
+                                Reply(356);
                             end
                             else begin
                                 if (k == 7) then begin
-                                    gsay_reply(87, 357);
+                                    Reply(357);
                                 end
                             end
                         end
@@ -3417,127 +3417,127 @@ begin
             end
         end
     end
-    giq_option(4, 87, 358, Node963, 49);
-    giq_option(4, 87, 359, Node023, 49);
-    giq_option(4, 87, 360, Node999, 50);
-    giq_option(4, 87, 361, Node966, 51);
+    giq_option(4, NAME, 358, Node963, GOOD_REACTION);
+    giq_option(4, NAME, 359, Node023, GOOD_REACTION);
+    giq_option(4, NAME, 360, Node999, NEUTRAL_REACTION);
+    giq_option(4, NAME, 361, Node966, BAD_REACTION);
 end
 
 procedure Node024
 begin
-    if (global_var(180) == 1) then begin
-        gsay_reply(87, 370);
+    if (global_var(GVAR_RUSTLE_REFUSE) == 1) then begin
+        Reply(370);
     end
     else begin
-        gsay_reply(87, 371);
+        Reply(371);
     end
-    giq_option(4, 87, 372, Node006, 50);
-    giq_option(4, 87, 373, Node941, 50);
+    giq_option(4, NAME, 372, Node006, NEUTRAL_REACTION);
+    giq_option(4, NAME, 373, Node941, NEUTRAL_REACTION);
     if (map_var(12) == 1) then begin
-        giq_option(4, 87, 374, Node026, 50);
+        giq_option(4, NAME, 374, Node026, NEUTRAL_REACTION);
     end
 end
 
 procedure Node025
 begin
-    gsay_reply(87, 380);
-    giq_option(4, 87, 381, Node999, 50);
-    giq_option(4, 87, 382, Node921, 50);
-    giq_option(4, 87, 383, Node027, 50);
+    Reply(380);
+    giq_option(4, NAME, 381, Node999, NEUTRAL_REACTION);
+    giq_option(4, NAME, 382, Node921, NEUTRAL_REACTION);
+    giq_option(4, NAME, 383, Node027, NEUTRAL_REACTION);
     if (map_var(12) == 1) then begin
-        giq_option(4, 87, 384, Node026, 50);
+        giq_option(4, NAME, 384, Node026, NEUTRAL_REACTION);
     end
 end
 
 procedure Node026
 begin
-    gsay_reply(87, 390);
-    giq_option(4, 87, 391, Node922, 51);
-    giq_option(4, 87, 392, Node999, 50);
+    Reply(390);
+    giq_option(4, NAME, 391, Node922, BAD_REACTION);
+    giq_option(4, NAME, 392, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node027
 begin
     if (map_var(4) == 1) then begin
-        gsay_reply(87, message_str(87, 400) + message_str(87, 405));
+        Reply(mstr(400) + mstr(405));
     end
     else begin
-        gsay_reply(87, message_str(87, 404) + message_str(87, 505));
+        Reply(mstr(404) + mstr(505));
     end
-    giq_option(4, 87, 401, Node989, 49);
-    giq_option(4, 87, 402, Node923, 51);
-    giq_option(4, 87, 403, Node921, 50);
+    giq_option(4, NAME, 401, Node989, GOOD_REACTION);
+    giq_option(4, NAME, 402, Node923, BAD_REACTION);
+    giq_option(4, NAME, 403, Node921, NEUTRAL_REACTION);
 end
 
 procedure Node028
 begin
-    gsay_reply(87, 410);
-    giq_option(4, 87, 411, Node994, 50);
+    Reply(410);
+    giq_option(4, NAME, 411, Node994, NEUTRAL_REACTION);
 end
 
 procedure Node029
 begin
     if (map_var(4) == 1) then begin
-        gsay_reply(87, message_str(87, 420) + message_str(87, 424));
+        Reply(mstr(420) + mstr(424));
     end
     else begin
-        gsay_reply(87, message_str(87, 420) + message_str(87, 533));
+        Reply(mstr(420) + mstr(533));
     end
-    giq_option(4, 87, 421, Node960, 49);
-    giq_option(4, 87, 422, Node999, 51);
-    giq_option(4, 87, 423, Node942, 50);
+    giq_option(4, NAME, 421, Node960, GOOD_REACTION);
+    giq_option(4, NAME, 422, Node999, BAD_REACTION);
+    giq_option(4, NAME, 423, Node942, NEUTRAL_REACTION);
 end
 
 procedure Node030
 begin
-    gsay_reply(87, 430);
-    giq_option(4, 87, 431, Node924, 51);
-    giq_option(4, 87, 432, Node999, 50);
+    Reply(430);
+    giq_option(4, NAME, 431, Node924, BAD_REACTION);
+    giq_option(4, NAME, 432, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node031
 begin
-    gsay_reply(87, 440);
-    if (global_var(184) == 0) then begin
-        giq_option(4, 87, 441, Node961, 49);
+    Reply(440);
+    if (global_var(GVAR_RUSTLE_OVER) == 0) then begin
+        giq_option(4, NAME, 441, Node961, GOOD_REACTION);
     end
-    giq_option(4, 87, 442, Node999, 50);
-    giq_option(4, 87, 443, Node940, 51);
-    giq_option(4, 87, 444, Node942, 50);
+    giq_option(4, NAME, 442, Node999, NEUTRAL_REACTION);
+    giq_option(4, NAME, 443, Node940, BAD_REACTION);
+    giq_option(4, NAME, 444, Node942, NEUTRAL_REACTION);
 end
 
 procedure Node032
 begin
     if (who_am_i == 1) then begin
-        gsay_reply(87, message_str(87, 449) + message_str(87, 450));
+        Reply(mstr(449) + mstr(450));
     end
     else begin
-        gsay_reply(87, message_str(87, 449) + message_str(87, 451));
+        Reply(mstr(449) + mstr(451));
     end
-    giq_option(-3, 87, 452, Node033, 51);
-    giq_option(-3, 87, obj_name(dude_obj) + message_str(87, 453), Node034, 50);
+    giq_option(-3, NAME, 452, Node033, BAD_REACTION);
+    giq_option(-3, NAME, dude_name + mstr(453), Node034, NEUTRAL_REACTION);
 end
 
 procedure Node033
 begin
-    gsay_reply(87, 460);
-    giq_option(-3, 87, 461, Node999, 50);
+    Reply(460);
+    giq_option(-3, NAME, 461, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node034
 begin
-    gsay_reply(87, 470);
-    giq_option(-3, 87, message_str(87, 471) + obj_name(dude_obj) + message_str(87, 472), Node999, 50);
+    Reply(470);
+    giq_option(-3, NAME, mstr(471) + dude_name + mstr(472), Node999, NEUTRAL_REACTION);
 end
 
 procedure Node035
 begin
-    float_msg(self_obj, message_str(87, 490), 2);
+    float_msg(self_obj, mstr(490), FLOAT_MSG_RED);
 end
 
 procedure Node036
 begin
-    float_msg(self_obj, message_str(87, 530), 2);
+    float_msg(self_obj, mstr(530), FLOAT_MSG_RED);
 end
 
 procedure Node981
@@ -3546,4 +3546,3 @@ begin
         call Node011();
     end
 end
-
