diff --git a/Games/Fallout2.Steam/Original/KCSLIM.ssl b/Games/Fallout2.Steam/ReDefine/KCSLIM.ssl
index 2d44746..11a2b45 100644
--- a/Games/Fallout2.Steam/Original/KCSLIM.ssl
+++ b/Games/Fallout2.Steam/ReDefine/KCSLIM.ssl
@@ -204,7 +204,7 @@ procedure critter_p_proc
 begin
     if ((local_var(5) == 2) and obj_can_see_obj(self_obj, dude_obj)) then begin
         set_local_var(5, 1);
-        attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+        attack(dude_obj);
     end
 end
 
@@ -219,7 +219,7 @@ procedure talk_p_proc
 begin
     if (local_var(2) == 0) then begin
         set_local_var(2, 1);
-        set_local_var(3, (get_critter_stat(dude_obj, 3) - get_critter_stat(self_obj, 3)) * 5);
+        set_local_var(3, (dude_charisma - self_charisma) * 5);
         debug_msg("Base Reaction == " + local_var(3));
     end
     Static_Reaction := 0;
@@ -231,14 +231,14 @@ begin
     else begin
         reaction_bonus_karma := -1 * (-1 * reaction_bonus_karma / 1000);
     end
-    if (global_var(48) >= 0) then begin
-        reaction_bonus_town_rep := global_var(48) / 2;
+    if (global_var(GVAR_TOWN_REP_KLAMATH) >= 0) then begin
+        reaction_bonus_town_rep := global_var(GVAR_TOWN_REP_KLAMATH) / 2;
     end
     else begin
-        reaction_bonus_town_rep := -1 * (-1 * global_var(48) / 2);
+        reaction_bonus_town_rep := -1 * (-1 * global_var(GVAR_TOWN_REP_KLAMATH) / 2);
     end
-    if (has_trait(2, dude_obj, 13)) then begin
-        if (get_critter_stat(dude_obj, 34) == get_critter_stat(self_obj, 34)) then begin
+    if (has_trait(TRAIT_TRAIT, dude_obj, TRAIT_sex_appeal)) then begin
+        if (dude_gender == self_gender) then begin
             Static_Reaction := Static_Reaction + -20;
         end
         else begin
@@ -254,48 +254,48 @@ begin
         debug_msg("Karma Reaction Bonus == " + Static_Reaction);
         Static_Reaction := Static_Reaction - reaction_bonus_town_rep;
         debug_msg("Town Rep Bonus == " + Static_Reaction);
-        set_global_var(37, 0);
-        set_global_var(38, 0);
-        set_global_var(39, 0);
-        set_global_var(40, 0);
-        set_global_var(41, 0);
-        set_global_var(42, 0);
-        set_global_var(43, 0);
-        set_global_var(44, 0);
-        set_global_var(45, 0);
-        if (global_var(0) >= 1000) then begin
-            set_global_var(37, 1);
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+        set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+        set_global_var(GVAR_KARMA_DEFENDER, 0);
+        set_global_var(GVAR_KARMA_WANDERER, 0);
+        set_global_var(GVAR_KARMA_BETRAYER, 0);
+        set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+        set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+            set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
         end
         else begin
-            if (global_var(0) >= 750) then begin
-                set_global_var(38, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
             end
             else begin
-                if (global_var(0) >= 500) then begin
-                    set_global_var(39, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                 end
                 else begin
-                    if (global_var(0) >= 250) then begin
-                        set_global_var(40, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                        set_global_var(GVAR_KARMA_DEFENDER, 1);
                     end
                     else begin
-                        if (global_var(0) > -250) then begin
-                            set_global_var(41, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                            set_global_var(GVAR_KARMA_WANDERER, 1);
                         end
                         else begin
-                            if (global_var(0) > -500) then begin
-                                set_global_var(42, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                set_global_var(GVAR_KARMA_BETRAYER, 1);
                             end
                             else begin
-                                if (global_var(0) > -750) then begin
-                                    set_global_var(43, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -1000) then begin
-                                        set_global_var(44, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                     end
                                     else begin
-                                        set_global_var(45, 1);
+                                        set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                     end
                                 end
                             end
@@ -304,35 +304,35 @@ begin
                 end
             end
         end
-        if (global_var(37) == 1) then begin
+        if (global_var(GVAR_KARMA_HOLY_WARRIOR) == 1) then begin
             Static_Reaction := Static_Reaction - 20;
         end
         else begin
-            if (global_var(38) == 1) then begin
+            if (global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES) == 1) then begin
                 Static_Reaction := Static_Reaction - 15;
             end
             else begin
-                if (global_var(39) == 1) then begin
+                if (global_var(GVAR_KARMA_SHIELD_OF_HOPE) == 1) then begin
                     Static_Reaction := Static_Reaction - 10;
                 end
                 else begin
-                    if (global_var(40) == 1) then begin
+                    if (global_var(GVAR_KARMA_DEFENDER) == 1) then begin
                         Static_Reaction := Static_Reaction - 5;
                     end
                     else begin
-                        if (global_var(41) == 1) then begin
+                        if (global_var(GVAR_KARMA_WANDERER) == 1) then begin
                             Static_Reaction := Static_Reaction - 0;
                         end
                         else begin
-                            if (global_var(42) == 1) then begin
+                            if (global_var(GVAR_KARMA_BETRAYER) == 1) then begin
                                 Static_Reaction := Static_Reaction - -5;
                             end
                             else begin
-                                if (global_var(43) == 1) then begin
+                                if (global_var(GVAR_KARMA_SWORD_OF_DESPAIR) == 1) then begin
                                     Static_Reaction := Static_Reaction - -10;
                                 end
                                 else begin
-                                    if (global_var(44) == 1) then begin
+                                    if (global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE) == 1) then begin
                                         Static_Reaction := Static_Reaction - -15;
                                     end
                                     else begin
@@ -352,48 +352,48 @@ begin
         debug_msg("Karma Reaction Bonus == " + Static_Reaction);
         Static_Reaction := Static_Reaction + reaction_bonus_town_rep;
         debug_msg("Town Rep Bonus == " + Static_Reaction);
-        set_global_var(37, 0);
-        set_global_var(38, 0);
-        set_global_var(39, 0);
-        set_global_var(40, 0);
-        set_global_var(41, 0);
-        set_global_var(42, 0);
-        set_global_var(43, 0);
-        set_global_var(44, 0);
-        set_global_var(45, 0);
-        if (global_var(0) >= 1000) then begin
-            set_global_var(37, 1);
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+        set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+        set_global_var(GVAR_KARMA_DEFENDER, 0);
+        set_global_var(GVAR_KARMA_WANDERER, 0);
+        set_global_var(GVAR_KARMA_BETRAYER, 0);
+        set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+        set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+            set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
         end
         else begin
-            if (global_var(0) >= 750) then begin
-                set_global_var(38, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
             end
             else begin
-                if (global_var(0) >= 500) then begin
-                    set_global_var(39, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                 end
                 else begin
-                    if (global_var(0) >= 250) then begin
-                        set_global_var(40, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                        set_global_var(GVAR_KARMA_DEFENDER, 1);
                     end
                     else begin
-                        if (global_var(0) > -250) then begin
-                            set_global_var(41, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                            set_global_var(GVAR_KARMA_WANDERER, 1);
                         end
                         else begin
-                            if (global_var(0) > -500) then begin
-                                set_global_var(42, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                set_global_var(GVAR_KARMA_BETRAYER, 1);
                             end
                             else begin
-                                if (global_var(0) > -750) then begin
-                                    set_global_var(43, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -1000) then begin
-                                        set_global_var(44, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                     end
                                     else begin
-                                        set_global_var(45, 1);
+                                        set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                     end
                                 end
                             end
@@ -402,35 +402,35 @@ begin
                 end
             end
         end
-        if (global_var(37) == 1) then begin
+        if (global_var(GVAR_KARMA_HOLY_WARRIOR) == 1) then begin
             Static_Reaction := Static_Reaction + 20;
         end
         else begin
-            if (global_var(38) == 1) then begin
+            if (global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES) == 1) then begin
                 Static_Reaction := Static_Reaction + 15;
             end
             else begin
-                if (global_var(39) == 1) then begin
+                if (global_var(GVAR_KARMA_SHIELD_OF_HOPE) == 1) then begin
                     Static_Reaction := Static_Reaction + 10;
                 end
                 else begin
-                    if (global_var(40) == 1) then begin
+                    if (global_var(GVAR_KARMA_DEFENDER) == 1) then begin
                         Static_Reaction := Static_Reaction + 5;
                     end
                     else begin
-                        if (global_var(41) == 1) then begin
+                        if (global_var(GVAR_KARMA_WANDERER) == 1) then begin
                             Static_Reaction := Static_Reaction + 0;
                         end
                         else begin
-                            if (global_var(42) == 1) then begin
+                            if (global_var(GVAR_KARMA_BETRAYER) == 1) then begin
                                 Static_Reaction := Static_Reaction + -5;
                             end
                             else begin
-                                if (global_var(43) == 1) then begin
+                                if (global_var(GVAR_KARMA_SWORD_OF_DESPAIR) == 1) then begin
                                     Static_Reaction := Static_Reaction + -10;
                                 end
                                 else begin
-                                    if (global_var(44) == 1) then begin
+                                    if (global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE) == 1) then begin
                                         Static_Reaction := Static_Reaction + -15;
                                     end
                                     else begin
@@ -446,42 +446,42 @@ begin
         debug_msg("Good Person Reaction Bonus == " + Static_Reaction);
     end
     if (Slavery_Tolerant == 0) then begin
-        if (global_var(11) == 1) then begin
+        if (global_var(GVAR_REPUTATION_SLAVER) == 1) then begin
             Static_Reaction := Static_Reaction + (2 * -25);
         end
     end
     else begin
         if (Slavery_Tolerant == 1) then begin
-            if (global_var(11) == 1) then begin
+            if (global_var(GVAR_REPUTATION_SLAVER) == 1) then begin
                 Static_Reaction := Static_Reaction + -25;
             end
         end
         else begin
             if (Slavery_Tolerant == 3) then begin
-                if (global_var(11) == 1) then begin
+                if (global_var(GVAR_REPUTATION_SLAVER) == 1) then begin
                     Static_Reaction := Static_Reaction - -25;
                 end
             end
         end
     end
     debug_msg("Slaver + Aligned Reaction == " + Static_Reaction);
-    Static_Reaction := Static_Reaction + (10 * has_trait(0, dude_obj, 10));
+    Static_Reaction := Static_Reaction + (10 * has_trait(TRAIT_PERK, dude_obj, PERK_presence));
     debug_msg("Presence Reaction == " + Static_Reaction);
-    if (global_var(1) >= 2) then begin
+    if (global_var(GVAR_CHILDKILLER_REPUTATION) >= 2) then begin
         Static_Reaction := Static_Reaction + -30;
     end
     debug_msg("Childkiller Reaction == " + Static_Reaction);
     set_local_var(0, local_var(3) + Static_Reaction);
     debug_msg("Initial Reaction == " + local_var(0));
-    if ((global_var(68) == 1) or (local_var(6) == 1)) then begin
+    if ((global_var(GVAR_ENEMY_KLAMATH) == 1) or (local_var(6) == 1)) then begin
         call Node998();
     end
     else begin
-        if (global_var(390) == 2) then begin
+        if (global_var(GVAR_QUEST_RAT_GOD) == 2) then begin
             call Node050();
         end
         else begin
-            if (get_critter_stat(dude_obj, 4) < 4) then begin
+            if (dude_iq < 4) then begin
                 start_gdialog(85, self_obj, 4, -1, -1);
                 gsay_start;
                 call Node006();
@@ -517,59 +517,59 @@ end
 procedure destroy_p_proc
 begin
     if (source_obj == dude_obj) then begin
-        set_global_var(5, global_var(5) + 1);
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(4) > (3 * global_var(5))) or (global_var(2) == 1)) and (global_var(1) == 0)) then begin
-            set_global_var(2, 1);
-            set_global_var(3, 0);
+        set_global_var(GVAR_GOOD_MONSTER, global_var(GVAR_GOOD_MONSTER) + 1);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(4) > (3 * global_var(GVAR_GOOD_MONSTER))) or (global_var(GVAR_CHAMPION_REPUTATION) == 1)) and (global_var(GVAR_CHILDKILLER_REPUTATION) == 0)) then begin
+            set_global_var(GVAR_CHAMPION_REPUTATION, 1);
+            set_global_var(GVAR_BERSERKER_REPUTATION, 0);
         end
-        if (metarule(51, self_obj) == 2) then begin
+        if (critter_kill_type(self_obj) == KILL_TYPE_children_kills) then begin
             if (source_obj == dude_obj) then begin
-                set_global_var(2, 0);
-                set_global_var(1, global_var(1) + 1);
-                set_global_var(0, global_var(0) + -15);
+                set_global_var(GVAR_CHAMPION_REPUTATION, 0);
+                set_global_var(GVAR_CHILDKILLER_REPUTATION, global_var(GVAR_CHILDKILLER_REPUTATION) + 1);
+                set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + -15);
                 debug_msg("Player gains " + -15 + " Karma Points.");
-                set_global_var(37, 0);
-                set_global_var(38, 0);
-                set_global_var(39, 0);
-                set_global_var(40, 0);
-                set_global_var(41, 0);
-                set_global_var(42, 0);
-                set_global_var(43, 0);
-                set_global_var(44, 0);
-                set_global_var(45, 0);
-                if (global_var(0) >= 1000) then begin
-                    set_global_var(37, 1);
+                set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+                set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+                set_global_var(GVAR_KARMA_DEFENDER, 0);
+                set_global_var(GVAR_KARMA_WANDERER, 0);
+                set_global_var(GVAR_KARMA_BETRAYER, 0);
+                set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+                set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+                set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+                    set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
                 end
                 else begin
-                    if (global_var(0) >= 750) then begin
-                        set_global_var(38, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
                     end
                     else begin
-                        if (global_var(0) >= 500) then begin
-                            set_global_var(39, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                            set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                         end
                         else begin
-                            if (global_var(0) >= 250) then begin
-                                set_global_var(40, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                                set_global_var(GVAR_KARMA_DEFENDER, 1);
                             end
                             else begin
-                                if (global_var(0) > -250) then begin
-                                    set_global_var(41, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                                    set_global_var(GVAR_KARMA_WANDERER, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -500) then begin
-                                        set_global_var(42, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                        set_global_var(GVAR_KARMA_BETRAYER, 1);
                                     end
                                     else begin
-                                        if (global_var(0) > -750) then begin
-                                            set_global_var(43, 1);
+                                        if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                            set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                         end
                                         else begin
-                                            if (global_var(0) > -1000) then begin
-                                                set_global_var(44, 1);
+                                            if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                                set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                             end
                                             else begin
-                                                set_global_var(45, 1);
+                                                set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                             end
                                         end
                                     end
@@ -578,58 +578,58 @@ begin
                         end
                     end
                 end
-                set_global_var(48, global_var(48) + -8);
+                set_global_var(GVAR_TOWN_REP_KLAMATH, global_var(GVAR_TOWN_REP_KLAMATH) + -8);
                 debug_msg("Added " + -8 + " to Town Rep");
             end
         end
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(5) > (2 * global_var(4))) or (global_var(3) == 1))) then begin
-            set_global_var(3, 1);
-            set_global_var(2, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
+            set_global_var(GVAR_BERSERKER_REPUTATION, 1);
+            set_global_var(GVAR_CHAMPION_REPUTATION, 0);
         end
-        set_global_var(0, global_var(0) + -10);
+        set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + -10);
         debug_msg("Player gains " + -10 + " Karma Points.");
-        set_global_var(37, 0);
-        set_global_var(38, 0);
-        set_global_var(39, 0);
-        set_global_var(40, 0);
-        set_global_var(41, 0);
-        set_global_var(42, 0);
-        set_global_var(43, 0);
-        set_global_var(44, 0);
-        set_global_var(45, 0);
-        if (global_var(0) >= 1000) then begin
-            set_global_var(37, 1);
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+        set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+        set_global_var(GVAR_KARMA_DEFENDER, 0);
+        set_global_var(GVAR_KARMA_WANDERER, 0);
+        set_global_var(GVAR_KARMA_BETRAYER, 0);
+        set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+        set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+            set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
         end
         else begin
-            if (global_var(0) >= 750) then begin
-                set_global_var(38, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
             end
             else begin
-                if (global_var(0) >= 500) then begin
-                    set_global_var(39, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                 end
                 else begin
-                    if (global_var(0) >= 250) then begin
-                        set_global_var(40, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                        set_global_var(GVAR_KARMA_DEFENDER, 1);
                     end
                     else begin
-                        if (global_var(0) > -250) then begin
-                            set_global_var(41, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                            set_global_var(GVAR_KARMA_WANDERER, 1);
                         end
                         else begin
-                            if (global_var(0) > -500) then begin
-                                set_global_var(42, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                set_global_var(GVAR_KARMA_BETRAYER, 1);
                             end
                             else begin
-                                if (global_var(0) > -750) then begin
-                                    set_global_var(43, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -1000) then begin
-                                        set_global_var(44, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                     end
                                     else begin
-                                        set_global_var(45, 1);
+                                        set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                     end
                                 end
                             end
@@ -638,7 +638,7 @@ begin
                 end
             end
         end
-        set_global_var(48, global_var(48) + -5);
+        set_global_var(GVAR_TOWN_REP_KLAMATH, global_var(GVAR_TOWN_REP_KLAMATH) + -5);
         debug_msg("Added " + -5 + " to Town Rep");
     end
 end
@@ -646,13 +646,13 @@ end
 procedure look_at_p_proc
 begin
     script_overrides;
-    display_msg(message_str(85, 100));
+    display_mstr(100);
 end
 
 procedure description_p_proc
 begin
     script_overrides;
-    display_msg(message_str(85, 101));
+    display_mstr(101);
 end
 
 procedure use_skill_on_p_proc
@@ -663,15 +663,15 @@ procedure damage_p_proc
 begin
     if (source_obj == dude_obj) then begin
         set_local_var(6, 1);
-        set_global_var(68, 1);
+        set_global_var(GVAR_ENEMY_KLAMATH, 1);
     end
 end
 
 procedure map_enter_p_proc
 begin
     Only_Once := 0;
-    critter_add_trait(self_obj, 1, 6, 43);
-    critter_add_trait(self_obj, 1, 5, 24);
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 43);
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_TOUGH_CITIZEN);
 end
 
 procedure push_p_proc
@@ -685,7 +685,7 @@ begin
         hadkey := 1;
     end
     set_local_var(10, 1);
-    gdialog_mod_barter(0);
+    gdialog_barter;
     if (obj_carrying_pid_obj(dude_obj, 298) and (hadkey == 0)) then begin
         set_map_var(2, 1);
         call Node013();
@@ -697,7 +697,7 @@ end
 
 procedure Node920
 begin
-    if (global_var(48) > 4) then begin
+    if (global_var(GVAR_TOWN_REP_KLAMATH) > 4) then begin
         call Node011();
     end
     else begin
@@ -728,7 +728,7 @@ end
 
 procedure Node960
 begin
-    if (has_skill(dude_obj, 17) > 28) then begin
+    if (dude_skill(SKILL_OUTDOORSMAN) > 28) then begin
         call Node018();
     end
     else begin
@@ -761,17 +761,17 @@ end
 
 procedure Node990
 begin
-    if (global_var(48) < 7) then begin
+    if (global_var(GVAR_TOWN_REP_KLAMATH) < 7) then begin
         item_caps_adjust(dude_obj, 100);
     end
     else begin
         item_caps_adjust(dude_obj, 50);
     end
-    learn := 2 * get_critter_stat(dude_obj, 4);
+    learn := 2 * dude_iq;
     if (learn < 7) then begin
         learn := 7;
     end
-    critter_mod_skill(dude_obj, 17, learn);
+    critter_mod_skill(dude_obj, SKILL_OUTDOORSMAN, learn);
     gfade_out(60 * 10);
     game_time_advance(8 * (60 * (60 * 10)));
     gfade_in(60 * 10);
@@ -780,7 +780,7 @@ end
 
 procedure Node995
 begin
-    if (get_critter_stat(dude_obj, 3) >= 7) then begin
+    if (dude_charisma >= 7) then begin
         set_local_var(3, 100);
         set_local_var(1, 4);
     end
@@ -798,7 +798,7 @@ end
 
 procedure Node997
 begin
-    if (get_critter_stat(dude_obj, 3) >= 8) then begin
+    if (dude_charisma >= 8) then begin
         call Node004();
     end
     else begin
@@ -817,141 +817,141 @@ end
 
 procedure Node001
 begin
-    if (get_critter_stat(dude_obj, 34) == 0) then begin
-        gsay_reply(85, 110);
+    if (dude_gender == 0) then begin
+        Reply(110);
     end
     else begin
-        gsay_reply(85, 111);
+        Reply(111);
     end
-    if (global_var(390) == 0) then begin
-        giq_option(4, 85, 390, Node023, 50);
+    if (global_var(GVAR_QUEST_RAT_GOD) == 0) then begin
+        giq_option(4, NAME, 390, Node023, NEUTRAL_REACTION);
     end
-    giq_option(4, 85, 112, Node003, 50);
-    giq_option(4, 85, 113, Node002, 50);
-    giq_option(4, 85, 114, Node999, 50);
+    giq_option(4, NAME, 112, Node003, NEUTRAL_REACTION);
+    giq_option(4, NAME, 113, Node002, NEUTRAL_REACTION);
+    giq_option(4, NAME, 114, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node002
 begin
-    gsay_message(85, 120, 50);
+    gsay_message(NAME, 120, NEUTRAL_REACTION);
 end
 
 procedure Node003
 begin
-    gsay_message(85, 130, 50);
+    gsay_message(NAME, 130, NEUTRAL_REACTION);
 end
 
 procedure Node004
 begin
-    if (global_var(48) < 5) then begin
-        gsay_reply(85, 140);
+    if (global_var(GVAR_TOWN_REP_KLAMATH) < 5) then begin
+        Reply(140);
     end
     else begin
-        gsay_reply(85, 141);
+        Reply(141);
     end
-    giq_option(4, 85, message_str(85, 142) + obj_name(dude_obj) + message_str(85, 143), Node008, 50);
-    giq_option(4, 85, 144, Node009, 50);
-    giq_option(4, 85, 145, Node999, 50);
+    giq_option(4, NAME, mstr(142) + dude_name + mstr(143), Node008, NEUTRAL_REACTION);
+    giq_option(4, NAME, 144, Node009, NEUTRAL_REACTION);
+    giq_option(4, NAME, 145, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node005
 begin
-    gsay_reply(85, 150);
-    giq_option(-3, 85, 151, Node006, 50);
-    giq_option(-3, 85, 152, Node006, 50);
-    giq_option(3, 85, message_str(85, 153) + obj_name(dude_obj) + message_str(85, 154), Node006, 50);
+    Reply(150);
+    giq_option(-3, NAME, 151, Node006, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 152, Node006, NEUTRAL_REACTION);
+    giq_option(3, NAME, mstr(153) + dude_name + mstr(154), Node006, NEUTRAL_REACTION);
 end
 
 procedure Node006
 begin
-    if (global_var(390) == 2) then begin
-        gsay_reply(85, 161);
+    if (global_var(GVAR_QUEST_RAT_GOD) == 2) then begin
+        Reply(161);
     end
     else begin
-        gsay_reply(85, 160);
+        Reply(160);
     end
-    giq_option(-3, 85, 162, Node06a, 50);
-    giq_option(-3, 85, 163, Node999, 50);
+    giq_option(-3, NAME, 162, Node06a, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 163, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node06a
 begin
-    if (global_var(390) == 0) then begin
-        set_global_var(390, 1);
+    if (global_var(GVAR_QUEST_RAT_GOD) == 0) then begin
+        set_global_var(GVAR_QUEST_RAT_GOD, 1);
         call Node11a();
     end
 end
 
 procedure Node007
 begin
-    if (global_var(48) < 5) then begin
-        gsay_reply(85, 170);
+    if (global_var(GVAR_TOWN_REP_KLAMATH) < 5) then begin
+        Reply(170);
     end
     else begin
-        gsay_reply(85, 171);
+        Reply(171);
     end
     if (obj_is_carrying_obj_pid(dude_obj, 298)) then begin
     end
     else begin
-        giq_option(4, 85, 172, Node010, 50);
+        giq_option(4, NAME, 172, Node010, NEUTRAL_REACTION);
     end
-    giq_option(4, 85, 173, Node008, 50);
-    giq_option(4, 85, 174, Node999, 50);
+    giq_option(4, NAME, 173, Node008, NEUTRAL_REACTION);
+    giq_option(4, NAME, 174, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node008
 begin
     if (local_var(7) < 1) then begin
         set_local_var(7, 1);
-        gsay_reply(85, 180);
+        Reply(180);
     end
     else begin
-        gsay_reply(85, 181);
-    end
-    giq_option(4, 85, 182, Node015, 50);
-    giq_option(4, 85, 183, Node016, 50);
-    giq_option(4, 85, 184, Node009, 50);
-    giq_option(4, 85, 185, Node022, 50);
-    giq_option(4, 85, 186, Node023, 50);
-    giq_option(4, 85, 187, Node999, 50);
+        Reply(181);
+    end
+    giq_option(4, NAME, 182, Node015, NEUTRAL_REACTION);
+    giq_option(4, NAME, 183, Node016, NEUTRAL_REACTION);
+    giq_option(4, NAME, 184, Node009, NEUTRAL_REACTION);
+    giq_option(4, NAME, 185, Node022, NEUTRAL_REACTION);
+    giq_option(4, NAME, 186, Node023, NEUTRAL_REACTION);
+    giq_option(4, NAME, 187, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node009
 begin
-    gsay_reply(85, 190);
-    giq_option(4, 85, 191, Node008, 50);
-    giq_option(4, 85, 192, Node999, 50);
+    Reply(190);
+    giq_option(4, NAME, 191, Node008, NEUTRAL_REACTION);
+    giq_option(4, NAME, 192, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node010
 begin
     if (local_var(8) < 1) then begin
         set_local_var(8, 1);
-        gsay_reply(85, 200);
+        Reply(200);
     end
     else begin
-        gsay_reply(85, 201);
+        Reply(201);
     end
-    giq_option(4, 85, 202, Node920, 50);
-    giq_option(4, 85, 203, Node920, 50);
-    giq_option(4, 85, 204, Node920, 50);
-    if (get_critter_stat(dude_obj, 34) == 1) then begin
-        giq_option(4, 85, 205, Node025, 50);
+    giq_option(4, NAME, 202, Node920, NEUTRAL_REACTION);
+    giq_option(4, NAME, 203, Node920, NEUTRAL_REACTION);
+    giq_option(4, NAME, 204, Node920, NEUTRAL_REACTION);
+    if (dude_gender == 1) then begin
+        giq_option(4, NAME, 205, Node025, NEUTRAL_REACTION);
     end
-    giq_option(4, 85, 206, Node999, 50);
+    giq_option(4, NAME, 206, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node011
 begin
     call Node11a();
     if (from_25 == 1) then begin
-        gsay_reply(85, 211);
+        Reply(211);
     end
     else begin
-        gsay_reply(85, 210);
+        Reply(210);
     end
-    giq_option(4, 85, 212, Node008, 50);
-    giq_option(4, 85, 213, Node999, 50);
+    giq_option(4, NAME, 212, Node008, NEUTRAL_REACTION);
+    giq_option(4, NAME, 213, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node11a
@@ -959,7 +959,7 @@ begin
     if (obj_is_carrying_obj_pid(self_obj, 298)) then begin
         mykey := obj_carrying_pid_obj(self_obj, 298);
         rm_obj_from_inven(self_obj, mykey);
-        item := create_object_sid(298, 0, 0, -1);
+        item := create_object(298, 0, 0);
         add_obj_to_inven(dude_obj, item);
     end
     set_map_var(2, 1);
@@ -967,172 +967,171 @@ end
 
 procedure Node012
 begin
-    gsay_reply(85, 220);
-    giq_option(4, 85, 221, slimbarter, 50);
-    giq_option(4, 85, 222, Node950, 50);
-    giq_option(7, 85, 223, Node950, 50);
-    giq_option(4, 85, 224, Node999, 50);
+    Reply(220);
+    giq_option(4, NAME, 221, slimbarter, NEUTRAL_REACTION);
+    giq_option(4, NAME, 222, Node950, NEUTRAL_REACTION);
+    giq_option(7, NAME, 223, Node950, NEUTRAL_REACTION);
+    giq_option(4, NAME, 224, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node013
 begin
-    gsay_reply(85, 230);
-    giq_option(4, 85, 231, Node999, 50);
-    giq_option(4, 85, 232, Node999, 50);
+    Reply(230);
+    giq_option(4, NAME, 231, Node999, NEUTRAL_REACTION);
+    giq_option(4, NAME, 232, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node014
 begin
-    gsay_reply(85, 240);
-    giq_option(4, 85, 241, slimbarter, 50);
-    giq_option(4, 85, 242, Node008, 50);
-    giq_option(4, 85, 243, Node999, 50);
+    Reply(240);
+    giq_option(4, NAME, 241, slimbarter, NEUTRAL_REACTION);
+    giq_option(4, NAME, 242, Node008, NEUTRAL_REACTION);
+    giq_option(4, NAME, 243, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node015
 begin
-    gsay_reply(85, 250);
-    giq_option(4, 85, 251, Node008, 50);
+    Reply(250);
+    giq_option(4, NAME, 251, Node008, NEUTRAL_REACTION);
     if (obj_is_carrying_obj_pid(dude_obj, 298)) then begin
     end
     else begin
-        giq_option(4, 85, 252, Node010, 50);
+        giq_option(4, NAME, 252, Node010, NEUTRAL_REACTION);
     end
-    giq_option(4, 85, 253, Node999, 50);
+    giq_option(4, NAME, 253, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node016
 begin
-    gsay_reply(85, 260);
-    giq_option(1, 85, 261, Node960, 50);
+    Reply(260);
+    giq_option(1, NAME, 261, Node960, NEUTRAL_REACTION);
 end
 
 procedure Node017
 begin
-    gsay_reply(85, 270);
-    giq_option(4, 85, 271, Node008, 50);
-    giq_option(4, 85, 272, Node970, 50);
-    giq_option(4, 85, 273, Node999, 50);
+    Reply(270);
+    giq_option(4, NAME, 271, Node008, NEUTRAL_REACTION);
+    giq_option(4, NAME, 272, Node970, NEUTRAL_REACTION);
+    giq_option(4, NAME, 273, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node018
 begin
-    gsay_reply(85, 280);
-    giq_option(4, 85, 281, Node019, 50);
-    giq_option(4, 85, 282, Node008, 50);
-    giq_option(4, 85, 283, Node999, 50);
+    Reply(280);
+    giq_option(4, NAME, 281, Node019, NEUTRAL_REACTION);
+    giq_option(4, NAME, 282, Node008, NEUTRAL_REACTION);
+    giq_option(4, NAME, 283, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node019
 begin
-    gsay_reply(85, 290);
-    giq_option(4, 85, 291, Node020, 50);
+    Reply(290);
+    giq_option(4, NAME, 291, Node020, NEUTRAL_REACTION);
 end
 
 procedure Node020
 begin
-    gsay_reply(85, 300);
-    giq_option(4, 85, 301, Node021, 50);
+    Reply(300);
+    giq_option(4, NAME, 301, Node021, NEUTRAL_REACTION);
 end
 
 procedure Node021
 begin
-    set_global_var(465, 1);
-    gsay_reply(85, 310);
-    giq_option(4, 85, 311, Node008, 50);
-    giq_option(4, 85, 312, Node999, 50);
+    set_global_var(GVAR_CAN_ASK_ARDIN_ABOUT_SMILEY, 1);
+    Reply(310);
+    giq_option(4, NAME, 311, Node008, NEUTRAL_REACTION);
+    giq_option(4, NAME, 312, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node022
 begin
-    gsay_reply(85, 320);
-    giq_option(4, 85, 321, Node008, 50);
-    giq_option(4, 85, 322, Node999, 50);
+    Reply(320);
+    giq_option(4, NAME, 321, Node008, NEUTRAL_REACTION);
+    giq_option(4, NAME, 322, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node023
 begin
-    if (global_var(390) == 2) then begin
-        gsay_reply(85, 336);
+    if (global_var(GVAR_QUEST_RAT_GOD) == 2) then begin
+        Reply(336);
     end
     else begin
-        gsay_reply(85, 335);
+        Reply(335);
     end
-    if (global_var(390) == 0) then begin
-        giq_option(4, 85, 337, Node024, 50);
+    if (global_var(GVAR_QUEST_RAT_GOD) == 0) then begin
+        giq_option(4, NAME, 337, Node024, NEUTRAL_REACTION);
     end
-    giq_option(4, 85, 311, Node008, 50);
-    giq_option(4, 85, 312, Node999, 50);
+    giq_option(4, NAME, 311, Node008, NEUTRAL_REACTION);
+    giq_option(4, NAME, 312, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node024
 begin
-    gsay_reply(85, 340);
+    Reply(340);
     if (obj_is_carrying_obj_pid(dude_obj, 298)) then begin
     end
     else begin
-        giq_option(4, 85, 341, Node24a, 50);
+        giq_option(4, NAME, 341, Node24a, NEUTRAL_REACTION);
     end
-    giq_option(4, 85, 342, Node008, 50);
-    giq_option(4, 85, 343, Node999, 50);
+    giq_option(4, NAME, 342, Node008, NEUTRAL_REACTION);
+    giq_option(4, NAME, 343, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node24a
 begin
-    set_global_var(390, 1);
+    set_global_var(GVAR_QUEST_RAT_GOD, 1);
     call Node010();
 end
 
 procedure Node025
 begin
     from_25 := 1;
-    if (get_critter_stat(dude_obj, 3) > 5) then begin
-        gsay_reply(85, 351);
+    if (dude_charisma > 5) then begin
+        Reply(351);
     end
     else begin
-        gsay_reply(85, 350);
+        Reply(350);
     end
-    giq_option(4, 85, 353, Node980, 50);
-    giq_option(4, 85, 354, Node010, 50);
-    giq_option(4, 85, 355, Node999, 50);
+    giq_option(4, NAME, 353, Node980, NEUTRAL_REACTION);
+    giq_option(4, NAME, 354, Node010, NEUTRAL_REACTION);
+    giq_option(4, NAME, 355, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node026
 begin
     set_local_var(9, 1);
-    if (global_var(48) < 7) then begin
-        gsay_reply(85, 360);
+    if (global_var(GVAR_TOWN_REP_KLAMATH) < 7) then begin
+        Reply(360);
         if (item_caps_total(dude_obj) >= 100) then begin
-            giq_option(4, 85, 362, Node990, 50);
+            giq_option(4, NAME, 362, Node990, NEUTRAL_REACTION);
         end
     end
     else begin
-        gsay_reply(85, 361);
+        Reply(361);
         if (item_caps_total(dude_obj) >= 50) then begin
-            giq_option(4, 85, 362, Node990, 50);
+            giq_option(4, NAME, 362, Node990, NEUTRAL_REACTION);
         end
     end
-    giq_option(4, 85, 363, Node008, 50);
-    giq_option(4, 85, 364, Node999, 50);
+    giq_option(4, NAME, 363, Node008, NEUTRAL_REACTION);
+    giq_option(4, NAME, 364, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node027
 begin
-    gsay_reply(85, 370);
-    giq_option(4, 85, 371, Node008, 50);
-    giq_option(4, 85, 372, Node999, 50);
+    Reply(370);
+    giq_option(4, NAME, 371, Node008, NEUTRAL_REACTION);
+    giq_option(4, NAME, 372, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node028
 begin
-    gsay_reply(85, 380);
-    giq_option(4, 85, 381, Node018, 50);
-    giq_option(4, 85, 382, Node008, 50);
-    giq_option(4, 85, 383, Node999, 50);
+    Reply(380);
+    giq_option(4, NAME, 381, Node018, NEUTRAL_REACTION);
+    giq_option(4, NAME, 382, Node008, NEUTRAL_REACTION);
+    giq_option(4, NAME, 383, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node050
 begin
-    float_msg(self_obj, message_str(85, random(400, 405)), 8);
+    float_msg(self_obj, mstr(random(400, 405)), FLOAT_MSG_YELLOW);
 end
-
