diff --git a/Games/Fallout2.Steam/Original/MCFARREL.ssl b/Games/Fallout2.Steam/ReDefine/MCFARREL.ssl
index 389d946..a9f5568 100644
--- a/Games/Fallout2.Steam/Original/MCFARREL.ssl
+++ b/Games/Fallout2.Steam/ReDefine/MCFARREL.ssl
@@ -184,22 +184,22 @@ end
 
 procedure start
 begin
-    if (cur_map_index == 19) then begin
+    if (CUR_MAP_MODINN) then begin
         modoc_farrel_obj := self_obj;
     end
 end
 
 procedure critter_p_proc
 begin
-    if ((local_var(4) != 0) or (global_var(52) <= -1000) and obj_can_see_obj(self_obj, dude_obj)) then begin
+    if ((local_var(4) != 0) or (global_var(GVAR_TOWN_REP_MODOC) <= -1000) and obj_can_see_obj(self_obj, dude_obj)) then begin
         if (critter_is_fleeing(self_obj)) then begin
             if (anim_busy(self_obj) == 0) then begin
                 if ((tile_distance_objs(self_obj, dude_obj) < 8) or obj_can_see_obj(self_obj, dude_obj)) then begin
                     if (anim_busy(self_obj) == 0) then begin
-                        global_temp := rotation_to_tile(tile_num(dude_obj), tile_num(self_obj));
+                        global_temp := rotation_to_tile(dude_tile, tile_num(self_obj));
                         animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(self_obj), global_temp, random(3, 10)), 1);
                         global_temp := (global_temp + 1) % 6;
-                        while ((anim_busy(self_obj) == 0) and (global_temp != rotation_to_tile(tile_num(dude_obj), tile_num(self_obj)))) do begin
+                        while ((anim_busy(self_obj) == 0) and (global_temp != rotation_to_tile(dude_tile, tile_num(self_obj)))) do begin
                             debug_msg("flee loop: rot == " + global_temp);
                             animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(self_obj), global_temp, random(3, 10)), 1);
                             global_temp := (global_temp + 1) % 6;
@@ -209,22 +209,22 @@ begin
             end
         end
         else begin
-            attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+            attack(dude_obj);
         end
     end
     else begin
-        if (cur_map_index == 24) then begin
-            if (has_trait(1, self_obj, 666)) then begin
+        if (CUR_MAP_GSTFARM) then begin
+            if (obj_is_visible_flag(self_obj)) then begin
                 if (obj_can_see_obj(self_obj, ghost_farm_slag_obj)) then begin
-                    attack_complex(ghost_farm_slag_obj, 0, 1, 0, 0, 30000, 0, 0);
+                    attack(ghost_farm_slag_obj);
                 end
                 else begin
                     if (tile_distance_objs(self_obj, dude_obj) > 12) then begin
                         if (tile_distance_objs(self_obj, dude_obj) > (12 * 2)) then begin
-                            animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(dude_obj), (has_trait(1, dude_obj, 10) + 3) % 6, 9), 1);
+                            animate_move_obj_to_tile(self_obj, tile_num_in_direction(dude_tile, (obj_get_rot(dude_obj) + 3) % 6, 9), 1);
                         end
                         else begin
-                            animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(dude_obj), (has_trait(1, dude_obj, 10) + 3) % 6, 9), 0);
+                            animate_move_obj_to_tile(self_obj, tile_num_in_direction(dude_tile, (obj_get_rot(dude_obj) + 3) % 6, 9), 0);
                         end
                     end
                 end
@@ -233,11 +233,11 @@ begin
         else begin
             if (local_var(6) < ((map_var(0) - 1) * (map_var(0) >= 1))) then begin
                 if (local_var(6) < 3) then begin
-                    float_msg(self_obj, message_str(100, 150 + local_var(6)), 8);
+                    float_msg(self_obj, mstr(150 + local_var(6)), FLOAT_MSG_YELLOW);
                 end
                 set_local_var(6, local_var(6) + 1);
                 if (local_var(6) >= 3) then begin
-                    critter_add_trait(self_obj, 1, 5, 183);
+                    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_PEASANT_KAMAKAZI);
                     call Node998();
                 end
             end
@@ -254,8 +254,8 @@ end
 
 procedure talk_p_proc
 begin
-    if (cur_map_index == 24) then begin
-        float_msg(self_obj, message_str(195, random(1000, 1003)), 2);
+    if (CUR_MAP_GSTFARM) then begin
+        float_msg(self_obj, message_str(SCRIPT_MODMAIN, random(1000, 1003)), FLOAT_MSG_RED);
     end
     else begin
         if (local_var(7) == 0) then begin
@@ -280,64 +280,64 @@ end
 
 procedure destroy_p_proc
 begin
-    if (cur_map_index == 19) then begin
+    if (CUR_MAP_MODINN) then begin
         modoc_farrel_obj := 0;
     end
-    set_global_var(297, global_var(297) bwor 65536);
+    set_global_var(GVAR_MODOC_GENERIC_FLAG_1, global_var(GVAR_MODOC_GENERIC_FLAG_1) bwor 65536);
     if (source_obj == dude_obj) then begin
-        set_global_var(5, global_var(5) + 1);
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(4) > (3 * global_var(5))) or (global_var(2) == 1)) and (global_var(1) == 0)) then begin
-            set_global_var(2, 1);
-            set_global_var(3, 0);
+        set_global_var(GVAR_GOOD_MONSTER, global_var(GVAR_GOOD_MONSTER) + 1);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(4) > (3 * global_var(GVAR_GOOD_MONSTER))) or (global_var(GVAR_CHAMPION_REPUTATION) == 1)) and (global_var(GVAR_CHILDKILLER_REPUTATION) == 0)) then begin
+            set_global_var(GVAR_CHAMPION_REPUTATION, 1);
+            set_global_var(GVAR_BERSERKER_REPUTATION, 0);
         end
-        if (metarule(51, self_obj) == 2) then begin
+        if (critter_kill_type(self_obj) == KILL_TYPE_children_kills) then begin
             if (source_obj == dude_obj) then begin
-                set_global_var(2, 0);
-                set_global_var(1, global_var(1) + 1);
-                set_global_var(0, global_var(0) + -15);
+                set_global_var(GVAR_CHAMPION_REPUTATION, 0);
+                set_global_var(GVAR_CHILDKILLER_REPUTATION, global_var(GVAR_CHILDKILLER_REPUTATION) + 1);
+                set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + -15);
                 debug_msg("Player gains " + -15 + " Karma Points.");
-                set_global_var(37, 0);
-                set_global_var(38, 0);
-                set_global_var(39, 0);
-                set_global_var(40, 0);
-                set_global_var(41, 0);
-                set_global_var(42, 0);
-                set_global_var(43, 0);
-                set_global_var(44, 0);
-                set_global_var(45, 0);
-                if (global_var(0) >= 1000) then begin
-                    set_global_var(37, 1);
+                set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+                set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+                set_global_var(GVAR_KARMA_DEFENDER, 0);
+                set_global_var(GVAR_KARMA_WANDERER, 0);
+                set_global_var(GVAR_KARMA_BETRAYER, 0);
+                set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+                set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+                set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+                    set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
                 end
                 else begin
-                    if (global_var(0) >= 750) then begin
-                        set_global_var(38, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
                     end
                     else begin
-                        if (global_var(0) >= 500) then begin
-                            set_global_var(39, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                            set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                         end
                         else begin
-                            if (global_var(0) >= 250) then begin
-                                set_global_var(40, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                                set_global_var(GVAR_KARMA_DEFENDER, 1);
                             end
                             else begin
-                                if (global_var(0) > -250) then begin
-                                    set_global_var(41, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                                    set_global_var(GVAR_KARMA_WANDERER, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -500) then begin
-                                        set_global_var(42, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                        set_global_var(GVAR_KARMA_BETRAYER, 1);
                                     end
                                     else begin
-                                        if (global_var(0) > -750) then begin
-                                            set_global_var(43, 1);
+                                        if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                            set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                         end
                                         else begin
-                                            if (global_var(0) > -1000) then begin
-                                                set_global_var(44, 1);
+                                            if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                                set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                             end
                                             else begin
-                                                set_global_var(45, 1);
+                                                set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                             end
                                         end
                                     end
@@ -346,58 +346,58 @@ begin
                         end
                     end
                 end
-                set_global_var(52, global_var(52) + -8);
+                set_global_var(GVAR_TOWN_REP_MODOC, global_var(GVAR_TOWN_REP_MODOC) + -8);
                 debug_msg("Added " + -8 + " to Town Rep");
             end
         end
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(5) > (2 * global_var(4))) or (global_var(3) == 1))) then begin
-            set_global_var(3, 1);
-            set_global_var(2, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
+            set_global_var(GVAR_BERSERKER_REPUTATION, 1);
+            set_global_var(GVAR_CHAMPION_REPUTATION, 0);
         end
-        set_global_var(0, global_var(0) + -10);
+        set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + -10);
         debug_msg("Player gains " + -10 + " Karma Points.");
-        set_global_var(37, 0);
-        set_global_var(38, 0);
-        set_global_var(39, 0);
-        set_global_var(40, 0);
-        set_global_var(41, 0);
-        set_global_var(42, 0);
-        set_global_var(43, 0);
-        set_global_var(44, 0);
-        set_global_var(45, 0);
-        if (global_var(0) >= 1000) then begin
-            set_global_var(37, 1);
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+        set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+        set_global_var(GVAR_KARMA_DEFENDER, 0);
+        set_global_var(GVAR_KARMA_WANDERER, 0);
+        set_global_var(GVAR_KARMA_BETRAYER, 0);
+        set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+        set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+            set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
         end
         else begin
-            if (global_var(0) >= 750) then begin
-                set_global_var(38, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
             end
             else begin
-                if (global_var(0) >= 500) then begin
-                    set_global_var(39, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                 end
                 else begin
-                    if (global_var(0) >= 250) then begin
-                        set_global_var(40, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                        set_global_var(GVAR_KARMA_DEFENDER, 1);
                     end
                     else begin
-                        if (global_var(0) > -250) then begin
-                            set_global_var(41, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                            set_global_var(GVAR_KARMA_WANDERER, 1);
                         end
                         else begin
-                            if (global_var(0) > -500) then begin
-                                set_global_var(42, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                set_global_var(GVAR_KARMA_BETRAYER, 1);
                             end
                             else begin
-                                if (global_var(0) > -750) then begin
-                                    set_global_var(43, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -1000) then begin
-                                        set_global_var(44, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                     end
                                     else begin
-                                        set_global_var(45, 1);
+                                        set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                     end
                                 end
                             end
@@ -406,7 +406,7 @@ begin
                 end
             end
         end
-        set_global_var(52, global_var(52) + -5);
+        set_global_var(GVAR_TOWN_REP_MODOC, global_var(GVAR_TOWN_REP_MODOC) + -5);
         debug_msg("Added " + -5 + " to Town Rep");
     end
 end
@@ -415,17 +415,17 @@ procedure look_at_p_proc
 begin
     script_overrides;
     if (local_var(5) > 0) then begin
-        display_msg(message_str(100, 101));
+        display_mstr(101);
     end
     else begin
-        display_msg(message_str(100, 100));
+        display_mstr(100);
     end
 end
 
 procedure description_p_proc
 begin
     script_overrides;
-    display_msg(message_str(100, 102));
+    display_mstr(102);
 end
 
 procedure use_skill_on_p_proc
@@ -435,8 +435,8 @@ end
 procedure damage_p_proc
 begin
     if (source_obj == dude_obj) then begin
-        if ((global_var(52) <= -1000) == 0) then begin
-            set_global_var(52, -1000);
+        if ((global_var(GVAR_TOWN_REP_MODOC) <= -1000) == 0) then begin
+            set_global_var(GVAR_TOWN_REP_MODOC, -1000);
         end
         call Node998();
     end
@@ -444,25 +444,25 @@ end
 
 procedure map_enter_p_proc
 begin
-    if ((global_var(297) bwand 65536) != 0) then begin
+    if ((global_var(GVAR_MODOC_GENERIC_FLAG_1) bwand 65536) != 0) then begin
         destroy_object(self_obj);
     end
-    if ((cur_map_index == 19) and (elevation(self_obj) != ((global_var(297) bwand 1) != 0))) then begin
-        move_to(self_obj, tile_num(self_obj), (global_var(297) bwand 1) != 0);
+    if ((CUR_MAP_MODINN) and (elevation(self_obj) != ((global_var(GVAR_MODOC_GENERIC_FLAG_1) bwand 1) != 0))) then begin
+        move_to(self_obj, tile_num(self_obj), (global_var(GVAR_MODOC_GENERIC_FLAG_1) bwand 1) != 0);
     end
-    if (cur_map_index == 19) then begin
+    if (CUR_MAP_MODINN) then begin
         modoc_farrel_obj := self_obj;
     end
-    if (metarule(14, 0)) then begin
-        if (cur_map_index != 24) then begin
-            critter_add_trait(self_obj, 1, 6, 28);
+    if (map_first_run) then begin
+        if (cur_map_index != MAP_GSTFARM) then begin
+            critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 28);
         end
-        critter_add_trait(self_obj, 1, 5, 14);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_PEASANTS);
         restock_amt := random(2, 2);
         if (obj_is_carrying_obj_pid(self_obj, 284) < restock_amt) then begin
             if (100 >= random(1, 100)) then begin
                 if (obj_is_carrying_obj_pid(self_obj, 284) < restock_amt) then begin
-                    restock_obj := create_object_sid(284, 0, 0, -1);
+                    restock_obj := create_object(284, 0, 0);
                     add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 284));
                 end
                 else begin
@@ -475,17 +475,17 @@ begin
                             restock_obj := obj_carrying_pid_obj(self_obj, 284);
                             if (obj_type(self_obj) == 1) then begin
                                 restock_obj := obj_carrying_pid_obj(self_obj, 284);
-                                if (critter_inven_obj(self_obj, 0) == restock_obj) then begin
-                                    if (obj_item_subtype(critter_inven_obj(self_obj, 0)) == 0) then begin
-                                        restock_obj := critter_inven_obj(self_obj, 0);
+                                if (critter_inven_obj(self_obj, INVEN_TYPE_WORN) == restock_obj) then begin
+                                    if (obj_item_subtype(critter_inven_obj(self_obj, INVEN_TYPE_WORN)) == 0) then begin
+                                        restock_obj := critter_inven_obj(self_obj, INVEN_TYPE_WORN);
                                         debug_msg("armour pid == " + obj_pid(restock_obj));
                                         rm_obj_from_inven(self_obj, restock_obj);
                                         add_obj_to_inven(self_obj, restock_obj);
                                     end
                                 end
                                 else begin
-                                    if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                        metarule(43, self_obj);
+                                    if ((critter_inven_obj(self_obj, INVEN_TYPE_LEFT_HAND) == restock_obj) or (critter_inven_obj(self_obj, INVEN_TYPE_RIGHT_HAND) == restock_obj)) then begin
+                                        inven_unwield(self_obj);
                                     end
                                 end
                             end
@@ -503,7 +503,7 @@ begin
         end
         else begin
             if (obj_is_carrying_obj_pid(self_obj, 284) < restock_amt) then begin
-                restock_obj := create_object_sid(284, 0, 0, -1);
+                restock_obj := create_object(284, 0, 0);
                 add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 284));
             end
             else begin
@@ -516,17 +516,17 @@ begin
                         restock_obj := obj_carrying_pid_obj(self_obj, 284);
                         if (obj_type(self_obj) == 1) then begin
                             restock_obj := obj_carrying_pid_obj(self_obj, 284);
-                            if (critter_inven_obj(self_obj, 0) == restock_obj) then begin
-                                if (obj_item_subtype(critter_inven_obj(self_obj, 0)) == 0) then begin
-                                    restock_obj := critter_inven_obj(self_obj, 0);
+                            if (critter_inven_obj(self_obj, INVEN_TYPE_WORN) == restock_obj) then begin
+                                if (obj_item_subtype(critter_inven_obj(self_obj, INVEN_TYPE_WORN)) == 0) then begin
+                                    restock_obj := critter_inven_obj(self_obj, INVEN_TYPE_WORN);
                                     debug_msg("armour pid == " + obj_pid(restock_obj));
                                     rm_obj_from_inven(self_obj, restock_obj);
                                     add_obj_to_inven(self_obj, restock_obj);
                                 end
                             end
                             else begin
-                                if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                    metarule(43, self_obj);
+                                if ((critter_inven_obj(self_obj, INVEN_TYPE_LEFT_HAND) == restock_obj) or (critter_inven_obj(self_obj, INVEN_TYPE_RIGHT_HAND) == restock_obj)) then begin
+                                    inven_unwield(self_obj);
                                 end
                             end
                         end
@@ -542,27 +542,27 @@ begin
             end
         end
     end
-    if (metarule(22, 0) == 0) then begin
-        if ((party_member_obj(obj_pid(self_obj)) != 0) == 0) then begin
-            if (cur_map_index == 24) then begin
-                if (((global_var(402) bwand 1024) != 0) and (global_var(310) <= 0) or ((global_var(112) == 2) or (((global_var(403) - (game_time / (10 * 60 * 60 * 24))) <= 0) or ((global_var(403) - (game_time / (10 * 60 * 60 * 24))) > 31) and (global_var(403) > 0)))) then begin
+    if (is_loading_game == 0) then begin
+        if ((obj_in_party(self_obj)) == 0) then begin
+            if (CUR_MAP_GSTFARM) then begin
+                if (((global_var(GVAR_MODOC_GENERIC_FLAG_2) bwand 1024) != 0) and (global_var(GVAR_SLAG_CNT) <= 0) or ((global_var(GVAR_MODOC_GHOST_FARM_SEED) == 2) or (((global_var(GVAR_SLAUGHTER_SLAG_TIME) - (game_time / (10 * 60 * 60 * 24))) <= 0) or ((global_var(GVAR_SLAUGHTER_SLAG_TIME) - (game_time / (10 * 60 * 60 * 24))) > 31) and (global_var(GVAR_SLAUGHTER_SLAG_TIME) > 0)))) then begin
                     debug_msg("no need for attackers");
-                    if (has_trait(1, self_obj, 666) == 1) then begin
+                    if (obj_is_visible_flag(self_obj) == 1) then begin
                         set_obj_visibility(self_obj, 1);
                     end
                     destroy_object(self_obj);
                 end
                 else begin
-                    if (global_var(112) == 1) then begin
+                    if (global_var(GVAR_MODOC_GHOST_FARM_SEED) == 1) then begin
                         debug_msg("showing need for attackers");
-                        critter_add_trait(self_obj, 1, 6, 0);
-                        if (has_trait(1, self_obj, 666) == 0) then begin
+                        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 0);
+                        if (obj_is_visible_flag(self_obj) == 0) then begin
                             set_obj_visibility(self_obj, 0);
                         end
                     end
                     else begin
                         debug_msg("hiding need for attackers");
-                        if (has_trait(1, self_obj, 666) == 1) then begin
+                        if (obj_is_visible_flag(self_obj) == 1) then begin
                             set_obj_visibility(self_obj, 1);
                         end
                     end
@@ -574,7 +574,7 @@ end
 
 procedure map_update_p_proc
 begin
-    if (cur_map_index == 19) then begin
+    if (CUR_MAP_MODINN) then begin
         modoc_farrel_obj := self_obj;
     end
 end
@@ -583,17 +583,17 @@ procedure timed_event_p_proc
 begin
     variable LVar0 := 0;
     if (fixed_param == 2) then begin
-        float_msg(self_obj, " ", 8);
+        float_msg(self_obj, " ", FLOAT_MSG_YELLOW);
     end
     else begin
         if (modoc_cornelius_obj != 0) then begin
-            if (cur_map_index == 19) then begin
+            if (CUR_MAP_MODINN) then begin
                 modoc_farrel_obj := self_obj;
             end
             if ((local_var(7) <= 0) or (local_var(7) > 3)) then begin
                 set_local_var(7, 1);
             end
-            float_msg(self_obj, message_str(100, 500 + local_var(7)), 8 + ((2 - 8) * has_trait(0, dude_obj, 22)));
+            float_msg(self_obj, mstr(500 + local_var(7)), 8 + ((2 - 8) * has_trait(TRAIT_PERK, dude_obj, PERK_empathy)));
             set_local_var(7, local_var(7) + 1);
             LVar0 := random(3, 5);
             add_timer_event(self_obj, game_ticks(LVar0) - 1, 2);
@@ -607,14 +607,14 @@ end
 
 procedure combat_p_proc
 begin
-    if (global_var(52) <= -1000) then begin
-        if (has_trait(1, self_obj, 6) == 0) then begin
-            critter_add_trait(self_obj, 1, 6, 28);
+    if (global_var(GVAR_TOWN_REP_MODOC) <= -1000) then begin
+        if (has_trait(TRAIT_OBJECT, self_obj, OBJECT_TEAM_NUM) == 0) then begin
+            critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 28);
         end
     end
     else begin
-        if (cur_map_index == 24) then begin
-            critter_add_trait(self_obj, 1, 6, 0);
+        if (CUR_MAP_GSTFARM) then begin
+            critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 0);
         end
     end
 end
@@ -622,196 +622,196 @@ end
 procedure Node000
 begin
     debug_msg("Node000");
-    gsay_reply(100, 153);
-    if ((global_var(105) >= 4) and (global_var(105) < 7) and ((global_var(106) >= 4) == 0) and ((obj_is_carrying_obj_pid(dude_obj, 257) > 0) == 0)) then begin
-        giq_option(4, 100, 154, Node010, 50);
+    Reply(153);
+    if ((global_var(GVAR_DUDE_STOMACH) >= 4) and (global_var(GVAR_DUDE_STOMACH) < 7) and ((global_var(GVAR_MODOC_FAMILY_FEUD_SEED_ONE) >= 4) == 0) and ((obj_is_carrying_obj_pid(dude_obj, 257) > 0) == 0)) then begin
+        giq_option(4, NAME, 154, Node010, NEUTRAL_REACTION);
     end
-    if (global_var(117)) then begin
-        giq_option(4, 100, 155, Node011, 50);
+    if (global_var(GVAR_MODOC_DOGS_ALIVE)) then begin
+        giq_option(4, NAME, 155, Node011, NEUTRAL_REACTION);
     end
-    giq_option(4, 100, 156, Node017, 50);
-    if ((global_var(110) >= 4) == 0) then begin
-        giq_option(4, 100, 157, Node018, 50);
+    giq_option(4, NAME, 156, Node017, NEUTRAL_REACTION);
+    if ((global_var(GVAR_MODOC_KARL_SEED) >= 4) == 0) then begin
+        giq_option(4, NAME, 157, Node018, NEUTRAL_REACTION);
     end
-    if ((global_var(105) == 0) or (global_var(106) == 0)) then begin
-        giq_option(4, 100, 158, Node023, 50);
+    if ((global_var(GVAR_DUDE_STOMACH) == 0) or (global_var(GVAR_MODOC_FAMILY_FEUD_SEED_ONE) == 0)) then begin
+        giq_option(4, NAME, 158, Node023, NEUTRAL_REACTION);
     end
-    giq_option(4, 100, message_str(14, 209), Node999, 50);
-    giq_option(-3, 100, message_str(14, 302), Node026, 50);
+    giq_option(4, NAME, g_mstr(209), Node999, NEUTRAL_REACTION);
+    giq_option(-3, NAME, g_mstr(302), Node026, NEUTRAL_REACTION);
 end
 
 procedure Node001
 begin
     debug_msg("Node001");
-    gsay_reply(100, 159);
-    if ((global_var(105) >= 4) and (global_var(105) < 7) and ((global_var(106) >= 4) == 0) and ((obj_is_carrying_obj_pid(dude_obj, 257) > 0) == 0)) then begin
-        giq_option(4, 100, 154, Node010, 50);
+    Reply(159);
+    if ((global_var(GVAR_DUDE_STOMACH) >= 4) and (global_var(GVAR_DUDE_STOMACH) < 7) and ((global_var(GVAR_MODOC_FAMILY_FEUD_SEED_ONE) >= 4) == 0) and ((obj_is_carrying_obj_pid(dude_obj, 257) > 0) == 0)) then begin
+        giq_option(4, NAME, 154, Node010, NEUTRAL_REACTION);
     end
-    if (global_var(117)) then begin
-        giq_option(4, 100, 155, Node011, 50);
+    if (global_var(GVAR_MODOC_DOGS_ALIVE)) then begin
+        giq_option(4, NAME, 155, Node011, NEUTRAL_REACTION);
     end
     if (map_var(0) >= 1) then begin
-        giq_option(4, 100, 162, Node016, 50);
+        giq_option(4, NAME, 162, Node016, NEUTRAL_REACTION);
     end
-    giq_option(4, 100, 156, Node017, 50);
-    if ((global_var(110) >= 4) == 0) then begin
-        giq_option(4, 100, 157, Node018, 50);
+    giq_option(4, NAME, 156, Node017, NEUTRAL_REACTION);
+    if ((global_var(GVAR_MODOC_KARL_SEED) >= 4) == 0) then begin
+        giq_option(4, NAME, 157, Node018, NEUTRAL_REACTION);
     end
-    if ((global_var(110) >= 4) and (global_var(110) < 7)) then begin
+    if ((global_var(GVAR_MODOC_KARL_SEED) >= 4) and (global_var(GVAR_MODOC_KARL_SEED) < 7)) then begin
         debug_msg("/nvermin_hunter_seed_active");
-        if ((global_var(297) bwand 128) != 0) then begin
+        if ((global_var(GVAR_MODOC_GENERIC_FLAG_1) bwand 128) != 0) then begin
             debug_msg("/nall_rats_killed");
-            giq_option(4, 100, 166, Node021, 50);
-            giq_option(-3, 100, 169, Node029, 50);
+            giq_option(4, NAME, 166, Node021, NEUTRAL_REACTION);
+            giq_option(-3, NAME, 169, Node029, NEUTRAL_REACTION);
         end
         else begin
-            giq_option(4, 100, 165, Node994, 50);
-            giq_option(-3, 100, 170, Node994, 50);
+            giq_option(4, NAME, 165, Node994, NEUTRAL_REACTION);
+            giq_option(-3, NAME, 170, Node994, NEUTRAL_REACTION);
         end
     end
-    if ((global_var(105) == 0) or (global_var(106) == 0)) then begin
-        giq_option(4, 100, 167, Node023, 50);
+    if ((global_var(GVAR_DUDE_STOMACH) == 0) or (global_var(GVAR_MODOC_FAMILY_FEUD_SEED_ONE) == 0)) then begin
+        giq_option(4, NAME, 167, Node023, NEUTRAL_REACTION);
     end
-    if ((global_var(106) >= 4) and (global_var(106) < 7) and (obj_is_carrying_obj_pid(dude_obj, 257) > 0)) then begin
-        giq_option(4, 100, 168, Node025, 50);
+    if ((global_var(GVAR_MODOC_FAMILY_FEUD_SEED_ONE) >= 4) and (global_var(GVAR_MODOC_FAMILY_FEUD_SEED_ONE) < 7) and (obj_is_carrying_obj_pid(dude_obj, 257) > 0)) then begin
+        giq_option(4, NAME, 168, Node025, NEUTRAL_REACTION);
     end
-    giq_option(4, 100, message_str(14, 209), Node999, 50);
-    giq_option(-3, 100, message_str(14, 302), Node995, 50);
+    giq_option(4, NAME, g_mstr(209), Node999, NEUTRAL_REACTION);
+    giq_option(-3, NAME, g_mstr(302), Node995, NEUTRAL_REACTION);
 end
 
 procedure Node010
 begin
     debug_msg("Node010");
-    gsay_reply(100, 171);
-    giq_option(4, 100, 172, Node999, 50);
+    Reply(171);
+    giq_option(4, NAME, 172, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node011
 begin
     debug_msg("Node011");
-    gsay_reply(100, 173);
-    giq_option(4, 100, 174, Node015, 50);
-    giq_option(4, 100, 175, Node012, 50);
-    giq_option(4, 100, 176, Node014, 50);
-    giq_option(4, 100, message_str(14, 209), Node999, 50);
+    Reply(173);
+    giq_option(4, NAME, 174, Node015, NEUTRAL_REACTION);
+    giq_option(4, NAME, 175, Node012, NEUTRAL_REACTION);
+    giq_option(4, NAME, 176, Node014, NEUTRAL_REACTION);
+    giq_option(4, NAME, g_mstr(209), Node999, NEUTRAL_REACTION);
 end
 
 procedure Node012
 begin
     debug_msg("Node012");
-    gsay_reply(100, 177);
-    giq_option(4, 100, 178, Node013, 50);
-    giq_option(4, 100, message_str(14, 209), Node999, 50);
+    Reply(177);
+    giq_option(4, NAME, 178, Node013, NEUTRAL_REACTION);
+    giq_option(4, NAME, g_mstr(209), Node999, NEUTRAL_REACTION);
 end
 
 procedure Node013
 begin
     debug_msg("Node013");
-    gsay_reply(100, 179);
-    giq_option(4, 100, 180, Node014, 50);
-    giq_option(4, 100, message_str(14, 209), Node999, 50);
+    Reply(179);
+    giq_option(4, NAME, 180, Node014, NEUTRAL_REACTION);
+    giq_option(4, NAME, g_mstr(209), Node999, NEUTRAL_REACTION);
 end
 
 procedure Node014
 begin
     debug_msg("Node014");
-    gsay_reply(100, 181);
-    giq_option(4, 100, 182, Node001, 50);
-    giq_option(4, 100, message_str(14, 209), Node999, 50);
+    Reply(181);
+    giq_option(4, NAME, 182, Node001, NEUTRAL_REACTION);
+    giq_option(4, NAME, g_mstr(209), Node999, NEUTRAL_REACTION);
 end
 
 procedure Node015
 begin
     debug_msg("Node015");
-    gsay_reply(100, 183);
-    giq_option(4, 100, message_str(14, 350), Node999, 50);
+    Reply(183);
+    giq_option(4, NAME, g_mstr(350), Node999, NEUTRAL_REACTION);
 end
 
 procedure Node016
 begin
     debug_msg("Node016");
-    gsay_reply(100, 184);
-    giq_option(4, 100, message_str(14, 350), Node999, 50);
+    Reply(184);
+    giq_option(4, NAME, g_mstr(350), Node999, NEUTRAL_REACTION);
 end
 
 procedure Node017
 begin
     debug_msg("Node017");
-    gsay_reply(100, 185);
-    giq_option(4, 100, 182, Node001, 50);
-    giq_option(4, 100, message_str(14, 209), Node999, 50);
+    Reply(185);
+    giq_option(4, NAME, 182, Node001, NEUTRAL_REACTION);
+    giq_option(4, NAME, g_mstr(209), Node999, NEUTRAL_REACTION);
 end
 
 procedure Node018
 begin
     debug_msg("Node018");
-    if ((global_var(105) >= 4) and (global_var(105) < 7) or ((global_var(106) >= 4) and (global_var(106) < 7))) then begin
-        gsay_reply(100, 187);
-        giq_option(4, 100, 191, Node022, 50);
+    if ((global_var(GVAR_DUDE_STOMACH) >= 4) and (global_var(GVAR_DUDE_STOMACH) < 7) or ((global_var(GVAR_MODOC_FAMILY_FEUD_SEED_ONE) >= 4) and (global_var(GVAR_MODOC_FAMILY_FEUD_SEED_ONE) < 7))) then begin
+        Reply(187);
+        giq_option(4, NAME, 191, Node022, NEUTRAL_REACTION);
     end
     else begin
-        gsay_reply(100, 188);
+        Reply(188);
     end
-    giq_option(4, 100, 189, Node019, 50);
-    giq_option(4, 100, 190, Node020, 50);
-    giq_option(4, 100, 182, Node001, 50);
-    giq_option(4, 100, message_str(14, 209), Node999, 50);
+    giq_option(4, NAME, 189, Node019, NEUTRAL_REACTION);
+    giq_option(4, NAME, 190, Node020, NEUTRAL_REACTION);
+    giq_option(4, NAME, 182, Node001, NEUTRAL_REACTION);
+    giq_option(4, NAME, g_mstr(209), Node999, NEUTRAL_REACTION);
 end
 
 procedure Node019
 begin
     debug_msg("Node019");
-    gsay_reply(100, 193);
-    giq_option(4, 100, 194, Node020, 50);
-    giq_option(4, 100, 195, Node999, 50);
+    Reply(193);
+    giq_option(4, NAME, 194, Node020, NEUTRAL_REACTION);
+    giq_option(4, NAME, 195, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node020
 begin
     debug_msg("Node020");
-    gsay_reply(100, 196);
-    giq_option(4, 100, message_str(14, 350), Node994, 50);
+    Reply(196);
+    giq_option(4, NAME, g_mstr(350), Node994, NEUTRAL_REACTION);
 end
 
 procedure Node021
 begin
     debug_msg("Node021");
-    if ((global_var(110) >= 4) == 0) then begin
-        set_global_var(110, 3);
+    if ((global_var(GVAR_MODOC_KARL_SEED) >= 4) == 0) then begin
+        set_global_var(GVAR_MODOC_KARL_SEED, 3);
     end
     else begin
-        if ((global_var(110) >= 7) == 0) then begin
-            set_global_var(110, 8);
+        if ((global_var(GVAR_MODOC_KARL_SEED) >= 7) == 0) then begin
+            set_global_var(GVAR_MODOC_KARL_SEED, 8);
         end
     end
-    set_global_var(52, global_var(52) + 1);
-    gsay_reply(100, 197);
-    giq_option(4, 100, 198, Node001, 50);
-    giq_option(4, 100, message_str(14, 209), Node999, 50);
+    set_global_var(GVAR_TOWN_REP_MODOC, global_var(GVAR_TOWN_REP_MODOC) + 1);
+    Reply(197);
+    giq_option(4, NAME, 198, Node001, NEUTRAL_REACTION);
+    giq_option(4, NAME, g_mstr(209), Node999, NEUTRAL_REACTION);
 end
 
 procedure Node022
 begin
     debug_msg("Node022");
-    gsay_reply(100, 199);
-    giq_option(4, 100, message_str(14, 350), Node999, 50);
+    Reply(199);
+    giq_option(4, NAME, g_mstr(350), Node999, NEUTRAL_REACTION);
 end
 
 procedure Node023
 begin
     debug_msg("Node023");
-    gsay_reply(100, 200);
-    giq_option(4, 100, 201, Node993, 50);
-    giq_option(4, 100, 202, Node001, 50);
-    giq_option(4, 100, message_str(14, 209), Node999, 50);
+    Reply(200);
+    giq_option(4, NAME, 201, Node993, NEUTRAL_REACTION);
+    giq_option(4, NAME, 202, Node001, NEUTRAL_REACTION);
+    giq_option(4, NAME, g_mstr(209), Node999, NEUTRAL_REACTION);
 end
 
 procedure Node024
 begin
     debug_msg("Node024");
-    gsay_reply(100, 203);
-    giq_option(4, 100, 182, Node001, 50);
-    giq_option(4, 100, message_str(14, 209), Node999, 50);
+    Reply(203);
+    giq_option(4, NAME, 182, Node001, NEUTRAL_REACTION);
+    giq_option(4, NAME, g_mstr(209), Node999, NEUTRAL_REACTION);
 end
 
 procedure Node025
@@ -825,17 +825,17 @@ begin
         restock_obj := obj_carrying_pid_obj(dude_obj, 257);
         if (obj_type(self_obj) == 1) then begin
             restock_obj := obj_carrying_pid_obj(dude_obj, 257);
-            if (critter_inven_obj(self_obj, 0) == restock_obj) then begin
-                if (obj_item_subtype(critter_inven_obj(dude_obj, 0)) == 0) then begin
-                    restock_obj := critter_inven_obj(dude_obj, 0);
+            if (critter_inven_obj(self_obj, INVEN_TYPE_WORN) == restock_obj) then begin
+                if (obj_item_subtype(critter_inven_obj(dude_obj, INVEN_TYPE_WORN)) == 0) then begin
+                    restock_obj := critter_inven_obj(dude_obj, INVEN_TYPE_WORN);
                     debug_msg("armour pid == " + obj_pid(restock_obj));
                     rm_obj_from_inven(dude_obj, restock_obj);
                     add_obj_to_inven(dude_obj, restock_obj);
                 end
             end
             else begin
-                if ((critter_inven_obj(dude_obj, 2) == restock_obj) or (critter_inven_obj(dude_obj, 1) == restock_obj)) then begin
-                    metarule(43, dude_obj);
+                if ((critter_inven_obj(dude_obj, INVEN_TYPE_LEFT_HAND) == restock_obj) or (critter_inven_obj(dude_obj, INVEN_TYPE_RIGHT_HAND) == restock_obj)) then begin
+                    inven_unwield(dude_obj);
                 end
             end
         end
@@ -844,66 +844,66 @@ begin
     end
     restock_obj := 0;
     restock_amt := 0;
-    if ((global_var(105) >= 4) == 0) then begin
-        set_global_var(105, 3);
+    if ((global_var(GVAR_DUDE_STOMACH) >= 4) == 0) then begin
+        set_global_var(GVAR_DUDE_STOMACH, 3);
     end
     else begin
-        if ((global_var(105) >= 7) == 0) then begin
-            set_global_var(105, 8);
+        if ((global_var(GVAR_DUDE_STOMACH) >= 7) == 0) then begin
+            set_global_var(GVAR_DUDE_STOMACH, 8);
         end
     end
-    if ((global_var(106) >= 4) == 0) then begin
-        set_global_var(106, 3);
+    if ((global_var(GVAR_MODOC_FAMILY_FEUD_SEED_ONE) >= 4) == 0) then begin
+        set_global_var(GVAR_MODOC_FAMILY_FEUD_SEED_ONE, 3);
     end
     else begin
-        if ((global_var(106) >= 7) == 0) then begin
-            set_global_var(106, 8);
+        if ((global_var(GVAR_MODOC_FAMILY_FEUD_SEED_ONE) >= 7) == 0) then begin
+            set_global_var(GVAR_MODOC_FAMILY_FEUD_SEED_ONE, 8);
         end
     end
-    set_global_var(0, global_var(0) + 10);
+    set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + 10);
     debug_msg("Player gains " + 10 + " Karma Points.");
-    set_global_var(37, 0);
-    set_global_var(38, 0);
-    set_global_var(39, 0);
-    set_global_var(40, 0);
-    set_global_var(41, 0);
-    set_global_var(42, 0);
-    set_global_var(43, 0);
-    set_global_var(44, 0);
-    set_global_var(45, 0);
-    if (global_var(0) >= 1000) then begin
-        set_global_var(37, 1);
+    set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+    set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+    set_global_var(GVAR_KARMA_DEFENDER, 0);
+    set_global_var(GVAR_KARMA_WANDERER, 0);
+    set_global_var(GVAR_KARMA_BETRAYER, 0);
+    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+    set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+    set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+    if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
     end
     else begin
-        if (global_var(0) >= 750) then begin
-            set_global_var(38, 1);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+            set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
         end
         else begin
-            if (global_var(0) >= 500) then begin
-                set_global_var(39, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
             end
             else begin
-                if (global_var(0) >= 250) then begin
-                    set_global_var(40, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                    set_global_var(GVAR_KARMA_DEFENDER, 1);
                 end
                 else begin
-                    if (global_var(0) > -250) then begin
-                        set_global_var(41, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                        set_global_var(GVAR_KARMA_WANDERER, 1);
                     end
                     else begin
-                        if (global_var(0) > -500) then begin
-                            set_global_var(42, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                            set_global_var(GVAR_KARMA_BETRAYER, 1);
                         end
                         else begin
-                            if (global_var(0) > -750) then begin
-                                set_global_var(43, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                             end
                             else begin
-                                if (global_var(0) > -1000) then begin
-                                    set_global_var(44, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                    set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                 end
                                 else begin
-                                    set_global_var(45, 1);
+                                    set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                 end
                             end
                         end
@@ -912,51 +912,51 @@ begin
             end
         end
     end
-    display_msg(message_str(100, 600));
+    display_mstr(600);
     give_exp_points(1500);
-    display_msg(message_str(14, 100) + 1500 + message_str(14, 101));
-    gsay_reply(100, 205);
-    giq_option(4, 100, message_str(14, 350), Node999, 50);
+    display_msg(g_mstr(100) + 1500 + g_mstr(101));
+    Reply(205);
+    giq_option(4, NAME, g_mstr(350), Node999, NEUTRAL_REACTION);
 end
 
 procedure Node026
 begin
     debug_msg("Node026");
-    gsay_reply(100, 206);
-    giq_option(-3, 100, 207, Node027, 50);
-    giq_option(-3, 100, 208, Node994, 50);
-    giq_option(-3, 100, 209, Node999, 50);
+    Reply(206);
+    giq_option(-3, NAME, 207, Node027, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 208, Node994, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 209, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node027
 begin
     debug_msg("Node027");
-    gsay_reply(100, 210);
-    giq_option(-3, 100, message_str(14, 350), Node999, 50);
+    Reply(210);
+    giq_option(-3, NAME, g_mstr(350), Node999, NEUTRAL_REACTION);
 end
 
 procedure Node028
 begin
     debug_msg("Node028");
-    gsay_reply(100, 211 + (get_critter_stat(dude_obj, 34) == 1));
-    giq_option(-3, 100, message_str(14, 350), Node994, 50);
+    Reply(211 + (dude_gender == 1));
+    giq_option(-3, NAME, g_mstr(350), Node994, NEUTRAL_REACTION);
 end
 
 procedure Node029
 begin
     debug_msg("Node029");
-    add_obj_to_inven(dude_obj, create_object_sid(378, 0, 0, -1));
-    set_global_var(52, global_var(52) + 1);
-    if ((global_var(110) >= 4) == 0) then begin
-        set_global_var(110, 3);
+    add_obj_to_inven(dude_obj, create_object(378, 0, 0));
+    set_global_var(GVAR_TOWN_REP_MODOC, global_var(GVAR_TOWN_REP_MODOC) + 1);
+    if ((global_var(GVAR_MODOC_KARL_SEED) >= 4) == 0) then begin
+        set_global_var(GVAR_MODOC_KARL_SEED, 3);
     end
     else begin
-        if ((global_var(110) >= 7) == 0) then begin
-            set_global_var(110, 8);
+        if ((global_var(GVAR_MODOC_KARL_SEED) >= 7) == 0) then begin
+            set_global_var(GVAR_MODOC_KARL_SEED, 8);
         end
     end
-    gsay_reply(100, 213);
-    giq_option(-3, 100, message_str(14, 350), Node999, 50);
+    Reply(213);
+    giq_option(-3, NAME, g_mstr(350), Node999, NEUTRAL_REACTION);
 end
 
 procedure Node999
@@ -975,7 +975,7 @@ end
 
 procedure Node995
 begin
-    if (global_var(110) == 0) then begin
+    if (global_var(GVAR_MODOC_KARL_SEED) == 0) then begin
         call Node026();
     end
     else begin
@@ -985,17 +985,16 @@ end
 
 procedure Node994
 begin
-    if (global_var(110) < 4) then begin
-        set_global_var(110, 4);
+    if (global_var(GVAR_MODOC_KARL_SEED) < 4) then begin
+        set_global_var(GVAR_MODOC_KARL_SEED, 4);
     end
     call Node996();
 end
 
 procedure Node993
 begin
-    if (global_var(106) < 4) then begin
-        set_global_var(106, 4);
+    if (global_var(GVAR_MODOC_FAMILY_FEUD_SEED_ONE) < 4) then begin
+        set_global_var(GVAR_MODOC_FAMILY_FEUD_SEED_ONE, 4);
     end
     call Node024();
 end
-
