diff --git a/Games/Fallout2.Steam/Original/MCROSE.ssl b/Games/Fallout2.Steam/ReDefine/MCROSE.ssl
index ce82bb5..e10533b 100644
--- a/Games/Fallout2.Steam/Original/MCROSE.ssl
+++ b/Games/Fallout2.Steam/ReDefine/MCROSE.ssl
@@ -199,15 +199,15 @@ end
 procedure critter_p_proc
 begin
     if (obj_can_see_obj(self_obj, dude_obj)) then begin
-        if (((local_var(5) bwand 2) != 0) or (global_var(52) <= -1000)) then begin
+        if (((local_var(5) bwand 2) != 0) or (global_var(GVAR_TOWN_REP_MODOC) <= -1000)) then begin
             if (critter_is_fleeing(self_obj)) then begin
                 if (anim_busy(self_obj) == 0) then begin
                     if ((tile_distance_objs(self_obj, dude_obj) < 8) or obj_can_see_obj(self_obj, dude_obj)) then begin
                         if (anim_busy(self_obj) == 0) then begin
-                            global_temp := rotation_to_tile(tile_num(dude_obj), tile_num(self_obj));
+                            global_temp := rotation_to_tile(dude_tile, tile_num(self_obj));
                             animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(self_obj), global_temp, random(3, 10)), 1);
                             global_temp := (global_temp + 1) % 6;
-                            while ((anim_busy(self_obj) == 0) and (global_temp != rotation_to_tile(tile_num(dude_obj), tile_num(self_obj)))) do begin
+                            while ((anim_busy(self_obj) == 0) and (global_temp != rotation_to_tile(dude_tile, tile_num(self_obj)))) do begin
                                 debug_msg("flee loop: rot == " + global_temp);
                                 animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(self_obj), global_temp, random(3, 10)), 1);
                                 global_temp := (global_temp + 1) % 6;
@@ -217,7 +217,7 @@ begin
                 end
             end
             else begin
-                attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+                attack(dude_obj);
             end
         end
     end
@@ -233,14 +233,14 @@ end
 procedure talk_p_proc
 begin
     script_overrides;
-    if ((global_var(297) bwand 16384) != 0) then begin
-        float_msg(self_obj, message_str(108, 150), 8);
+    if ((global_var(GVAR_MODOC_GENERIC_FLAG_1) bwand 16384) != 0) then begin
+        float_msg(self_obj, mstr(150), FLOAT_MSG_YELLOW);
     end
     else begin
-        set_global_var(297, global_var(297) bwor 32);
+        set_global_var(GVAR_MODOC_GENERIC_FLAG_1, global_var(GVAR_MODOC_GENERIC_FLAG_1) bwor 32);
         start_gdialog(108, self_obj, 4, -1, -1);
         gsay_start;
-        if (global_var(111) >= 7) then begin
+        if (global_var(GVAR_MODOC_VERMIN_HUNTER_SEED) >= 7) then begin
             call Node002();
         end
         else begin
@@ -260,59 +260,59 @@ end
 procedure destroy_p_proc
 begin
     if (source_obj == dude_obj) then begin
-        set_global_var(5, global_var(5) + 1);
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(4) > (3 * global_var(5))) or (global_var(2) == 1)) and (global_var(1) == 0)) then begin
-            set_global_var(2, 1);
-            set_global_var(3, 0);
+        set_global_var(GVAR_GOOD_MONSTER, global_var(GVAR_GOOD_MONSTER) + 1);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(4) > (3 * global_var(GVAR_GOOD_MONSTER))) or (global_var(GVAR_CHAMPION_REPUTATION) == 1)) and (global_var(GVAR_CHILDKILLER_REPUTATION) == 0)) then begin
+            set_global_var(GVAR_CHAMPION_REPUTATION, 1);
+            set_global_var(GVAR_BERSERKER_REPUTATION, 0);
         end
-        if (metarule(51, self_obj) == 2) then begin
+        if (critter_kill_type(self_obj) == KILL_TYPE_children_kills) then begin
             if (source_obj == dude_obj) then begin
-                set_global_var(2, 0);
-                set_global_var(1, global_var(1) + 1);
-                set_global_var(0, global_var(0) + -15);
+                set_global_var(GVAR_CHAMPION_REPUTATION, 0);
+                set_global_var(GVAR_CHILDKILLER_REPUTATION, global_var(GVAR_CHILDKILLER_REPUTATION) + 1);
+                set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + -15);
                 debug_msg("Player gains " + -15 + " Karma Points.");
-                set_global_var(37, 0);
-                set_global_var(38, 0);
-                set_global_var(39, 0);
-                set_global_var(40, 0);
-                set_global_var(41, 0);
-                set_global_var(42, 0);
-                set_global_var(43, 0);
-                set_global_var(44, 0);
-                set_global_var(45, 0);
-                if (global_var(0) >= 1000) then begin
-                    set_global_var(37, 1);
+                set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+                set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+                set_global_var(GVAR_KARMA_DEFENDER, 0);
+                set_global_var(GVAR_KARMA_WANDERER, 0);
+                set_global_var(GVAR_KARMA_BETRAYER, 0);
+                set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+                set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+                set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+                    set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
                 end
                 else begin
-                    if (global_var(0) >= 750) then begin
-                        set_global_var(38, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
                     end
                     else begin
-                        if (global_var(0) >= 500) then begin
-                            set_global_var(39, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                            set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                         end
                         else begin
-                            if (global_var(0) >= 250) then begin
-                                set_global_var(40, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                                set_global_var(GVAR_KARMA_DEFENDER, 1);
                             end
                             else begin
-                                if (global_var(0) > -250) then begin
-                                    set_global_var(41, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                                    set_global_var(GVAR_KARMA_WANDERER, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -500) then begin
-                                        set_global_var(42, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                        set_global_var(GVAR_KARMA_BETRAYER, 1);
                                     end
                                     else begin
-                                        if (global_var(0) > -750) then begin
-                                            set_global_var(43, 1);
+                                        if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                            set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                         end
                                         else begin
-                                            if (global_var(0) > -1000) then begin
-                                                set_global_var(44, 1);
+                                            if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                                set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                             end
                                             else begin
-                                                set_global_var(45, 1);
+                                                set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                             end
                                         end
                                     end
@@ -321,58 +321,58 @@ begin
                         end
                     end
                 end
-                set_global_var(52, global_var(52) + -8);
+                set_global_var(GVAR_TOWN_REP_MODOC, global_var(GVAR_TOWN_REP_MODOC) + -8);
                 debug_msg("Added " + -8 + " to Town Rep");
             end
         end
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(5) > (2 * global_var(4))) or (global_var(3) == 1))) then begin
-            set_global_var(3, 1);
-            set_global_var(2, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
+            set_global_var(GVAR_BERSERKER_REPUTATION, 1);
+            set_global_var(GVAR_CHAMPION_REPUTATION, 0);
         end
-        set_global_var(0, global_var(0) + -10);
+        set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + -10);
         debug_msg("Player gains " + -10 + " Karma Points.");
-        set_global_var(37, 0);
-        set_global_var(38, 0);
-        set_global_var(39, 0);
-        set_global_var(40, 0);
-        set_global_var(41, 0);
-        set_global_var(42, 0);
-        set_global_var(43, 0);
-        set_global_var(44, 0);
-        set_global_var(45, 0);
-        if (global_var(0) >= 1000) then begin
-            set_global_var(37, 1);
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+        set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+        set_global_var(GVAR_KARMA_DEFENDER, 0);
+        set_global_var(GVAR_KARMA_WANDERER, 0);
+        set_global_var(GVAR_KARMA_BETRAYER, 0);
+        set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+        set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+            set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
         end
         else begin
-            if (global_var(0) >= 750) then begin
-                set_global_var(38, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
             end
             else begin
-                if (global_var(0) >= 500) then begin
-                    set_global_var(39, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                 end
                 else begin
-                    if (global_var(0) >= 250) then begin
-                        set_global_var(40, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                        set_global_var(GVAR_KARMA_DEFENDER, 1);
                     end
                     else begin
-                        if (global_var(0) > -250) then begin
-                            set_global_var(41, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                            set_global_var(GVAR_KARMA_WANDERER, 1);
                         end
                         else begin
-                            if (global_var(0) > -500) then begin
-                                set_global_var(42, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                set_global_var(GVAR_KARMA_BETRAYER, 1);
                             end
                             else begin
-                                if (global_var(0) > -750) then begin
-                                    set_global_var(43, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -1000) then begin
-                                        set_global_var(44, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                     end
                                     else begin
-                                        set_global_var(45, 1);
+                                        set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                     end
                                 end
                             end
@@ -381,7 +381,7 @@ begin
                 end
             end
         end
-        set_global_var(52, global_var(52) + -5);
+        set_global_var(GVAR_TOWN_REP_MODOC, global_var(GVAR_TOWN_REP_MODOC) + -5);
         debug_msg("Added " + -5 + " to Town Rep");
     end
     modoc_rose_obj := 0;
@@ -390,13 +390,13 @@ end
 procedure look_at_p_proc
 begin
     script_overrides;
-    display_msg(message_str(108, 100 + ((local_var(5) bwand 1) != 0)));
+    display_mstr(100 + ((local_var(5) bwand 1) != 0));
 end
 
 procedure description_p_proc
 begin
     script_overrides;
-    display_msg(message_str(108, 102));
+    display_mstr(102);
 end
 
 procedure use_skill_on_p_proc
@@ -406,8 +406,8 @@ end
 procedure damage_p_proc
 begin
     if (source_obj == dude_obj) then begin
-        if ((global_var(52) <= -1000) == 0) then begin
-            set_global_var(52, -1000);
+        if ((global_var(GVAR_TOWN_REP_MODOC) <= -1000) == 0) then begin
+            set_global_var(GVAR_TOWN_REP_MODOC, -1000);
         end
         call Node998();
     end
@@ -416,20 +416,20 @@ end
 procedure map_enter_p_proc
 begin
     modoc_rose_obj := self_obj;
-    if ((cur_map_index == 19) and (elevation(self_obj) != ((global_var(297) bwand 1) != 0))) then begin
-        move_to(self_obj, tile_num(self_obj), (global_var(297) bwand 1) != 0);
+    if ((CUR_MAP_MODINN) and (elevation(self_obj) != ((global_var(GVAR_MODOC_GENERIC_FLAG_1) bwand 1) != 0))) then begin
+        move_to(self_obj, tile_num(self_obj), (global_var(GVAR_MODOC_GENERIC_FLAG_1) bwand 1) != 0);
     end
     if (only_once == 0) then begin
         only_once := 1;
-        critter_add_trait(self_obj, 1, 6, 28);
-        critter_add_trait(self_obj, 1, 5, 14);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 28);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_PEASANTS);
     end
-    if (metarule(14, 0)) then begin
+    if (map_first_run) then begin
         restock_amt := random(20, 20);
         if (obj_is_carrying_obj_pid(self_obj, 41) < restock_amt) then begin
             if (100 >= random(1, 100)) then begin
                 if (obj_is_carrying_obj_pid(self_obj, 41) < restock_amt) then begin
-                    restock_obj := create_object_sid(41, 0, 0, -1);
+                    restock_obj := create_object(41, 0, 0);
                     add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 41));
                 end
                 else begin
@@ -452,7 +452,7 @@ begin
                                 end
                                 else begin
                                     if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                        metarule(43, self_obj);
+                                        inven_unwield(self_obj);
                                     end
                                 end
                             end
@@ -470,7 +470,7 @@ begin
         end
         else begin
             if (obj_is_carrying_obj_pid(self_obj, 41) < restock_amt) then begin
-                restock_obj := create_object_sid(41, 0, 0, -1);
+                restock_obj := create_object(41, 0, 0);
                 add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 41));
             end
             else begin
@@ -493,7 +493,7 @@ begin
                             end
                             else begin
                                 if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                    metarule(43, self_obj);
+                                    inven_unwield(self_obj);
                                 end
                             end
                         end
@@ -519,150 +519,150 @@ end
 
 procedure Node000
 begin
-    if (get_critter_stat(dude_obj, 34) == 0) then begin
-        gsay_reply(108, 154);
+    if (dude_gender == 0) then begin
+        Reply(154);
     end
     else begin
-        gsay_reply(108, 155);
+        Reply(155);
     end
-    giq_option(4, 108, 156, Node010, 50);
-    giq_option(4, 108, 157, Node032, 50);
-    giq_option(4, 108, 158, Node023, 50);
-    giq_option(4, 108, message_str(14, random(375, 376)), Node999, 50);
-    giq_option(-3, 108, 159, Node996, 50);
-    giq_option(-3, 108, message_str(14, random(300, 303)), Node999, 50);
+    giq_option(4, NAME, 156, Node010, NEUTRAL_REACTION);
+    giq_option(4, NAME, 157, Node032, NEUTRAL_REACTION);
+    giq_option(4, NAME, 158, Node023, NEUTRAL_REACTION);
+    giq_option(4, NAME, g_mstr(random(375, 376)), Node999, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 159, Node996, NEUTRAL_REACTION);
+    giq_option(-3, NAME, g_mstr(random(300, 303)), Node999, NEUTRAL_REACTION);
 end
 
 procedure Node001
 begin
-    gsay_reply(108, 160);
-    giq_option(4, 108, 156, Node010, 50);
-    giq_option(4, 108, 157, Node032, 50);
-    giq_option(4, 108, 158, Node023, 50);
-    giq_option(4, 108, message_str(14, random(375, 376)), Node999, 50);
-    giq_option(-3, 108, 159, Node996, 50);
-    giq_option(-3, 108, message_str(14, random(300, 303)), Node999, 50);
+    Reply(160);
+    giq_option(4, NAME, 156, Node010, NEUTRAL_REACTION);
+    giq_option(4, NAME, 157, Node032, NEUTRAL_REACTION);
+    giq_option(4, NAME, 158, Node023, NEUTRAL_REACTION);
+    giq_option(4, NAME, g_mstr(random(375, 376)), Node999, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 159, Node996, NEUTRAL_REACTION);
+    giq_option(-3, NAME, g_mstr(random(300, 303)), Node999, NEUTRAL_REACTION);
 end
 
 procedure Node002
 begin
     food_cost := 0;
-    if ((global_var(402) bwand 262144) != 0) then begin
-        gsay_reply(108, 166);
+    if ((global_var(GVAR_MODOC_GENERIC_FLAG_2) bwand 262144) != 0) then begin
+        Reply(166);
     end
     else begin
-        set_global_var(402, global_var(402) bwor 262144);
-        gsay_reply(108, 165);
+        set_global_var(GVAR_MODOC_GENERIC_FLAG_2, global_var(GVAR_MODOC_GENERIC_FLAG_2) bwor 262144);
+        Reply(165);
     end
-    giq_option(4, 108, 167, Node995, 50);
-    giq_option(4, 108, message_str(14, random(375, 376)), Node999, 50);
+    giq_option(4, NAME, 167, Node995, NEUTRAL_REACTION);
+    giq_option(4, NAME, g_mstr(random(375, 376)), Node999, NEUTRAL_REACTION);
 end
 
 procedure Node010
 begin
-    if (get_critter_stat(dude_obj, 34) == 0) then begin
-        gsay_reply(108, 168);
+    if (dude_gender == 0) then begin
+        Reply(168);
     end
     else begin
-        gsay_reply(108, 169);
+        Reply(169);
     end
-    giq_option(4, 108, 170, Node994, 50);
-    giq_option(4, 108, 171, Node993, 50);
-    if (((global_var(111) >= 7) == 0) and (((local_var(5) bwand 512) != 0) == 0)) then begin
-        giq_option(4, 108, 172, Node033, 50);
+    giq_option(4, NAME, 170, Node994, NEUTRAL_REACTION);
+    giq_option(4, NAME, 171, Node993, NEUTRAL_REACTION);
+    if (((global_var(GVAR_MODOC_VERMIN_HUNTER_SEED) >= 7) == 0) and (((local_var(5) bwand 512) != 0) == 0)) then begin
+        giq_option(4, NAME, 172, Node033, NEUTRAL_REACTION);
     end
-    giq_option(4, 108, 173, Node991, 50);
-    giq_option(4, 108, 174, Node990, 50);
-    giq_option(4, 108, 175, Node989, 50);
-    giq_option(4, 108, 176, Node999, 50);
+    giq_option(4, NAME, 173, Node991, NEUTRAL_REACTION);
+    giq_option(4, NAME, 174, Node990, NEUTRAL_REACTION);
+    giq_option(4, NAME, 175, Node989, NEUTRAL_REACTION);
+    giq_option(4, NAME, 176, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node011
 begin
-    gsay_reply(108, 177);
+    Reply(177);
     if (item_caps_total(dude_obj) >= 3) then begin
-        giq_option(4, 108, 178, Node010, 50);
+        giq_option(4, NAME, 178, Node010, NEUTRAL_REACTION);
     end
-    giq_option(4, 108, 179, Node023, 50);
-    giq_option(4, 108, message_str(14, random(375, 376)), Node999, 50);
+    giq_option(4, NAME, 179, Node023, NEUTRAL_REACTION);
+    giq_option(4, NAME, g_mstr(random(375, 376)), Node999, NEUTRAL_REACTION);
 end
 
 procedure Node012
 begin
     if (local_var(4) > 36) then begin
-        gsay_reply(108, 180);
-        giq_option(4, 108, 182, Node022, 50);
+        Reply(180);
+        giq_option(4, NAME, 182, Node022, NEUTRAL_REACTION);
     end
     else begin
-        gsay_reply(108, 181);
-        giq_option(4, 108, 183, Node013, 50);
-        giq_option(4, 108, 184, Node022, 50);
+        Reply(181);
+        giq_option(4, NAME, 183, Node013, NEUTRAL_REACTION);
+        giq_option(4, NAME, 184, Node022, NEUTRAL_REACTION);
     end
 end
 
 procedure Node013
 begin
-    gsay_reply(108, 185);
-    giq_option(4, 108, 186, Node022, 50);
-    giq_option(4, 108, 187, Node988, 50);
+    Reply(185);
+    giq_option(4, NAME, 186, Node022, NEUTRAL_REACTION);
+    giq_option(4, NAME, 187, Node988, NEUTRAL_REACTION);
 end
 
 procedure Node014
 begin
     item_caps_adjust(self_obj, food_cost);
     item_caps_adjust(dude_obj, -food_cost);
-    gsay_reply(108, 188);
-    giq_option(4, 108, message_str(14, 352), Node015, 50);
+    Reply(188);
+    giq_option(4, NAME, g_mstr(352), Node015, NEUTRAL_REACTION);
 end
 
 procedure Node015
 begin
-    gsay_reply(108, 189);
-    giq_option(4, 108, message_str(14, 352), Node016, 50);
+    Reply(189);
+    giq_option(4, NAME, g_mstr(352), Node016, NEUTRAL_REACTION);
 end
 
 procedure Node016
 begin
-    gsay_reply(108, 190);
-    giq_option(4, 108, 191, Node017, 50);
+    Reply(190);
+    giq_option(4, NAME, 191, Node017, NEUTRAL_REACTION);
 end
 
 procedure Node017
 begin
-    gsay_reply(108, 192);
-    giq_option(4, 108, 193, Node018, 50);
+    Reply(192);
+    giq_option(4, NAME, 193, Node018, NEUTRAL_REACTION);
 end
 
 procedure Node018
 begin
-    gsay_reply(108, 194);
-    giq_option(4, 108, 195, Node019, 50);
+    Reply(194);
+    giq_option(4, NAME, 195, Node019, NEUTRAL_REACTION);
 end
 
 procedure Node019
 begin
     set_local_var(4, 36 + 1);
-    gsay_reply(108, 196);
-    giq_option(4, 108, 197, Node020, 50);
+    Reply(196);
+    giq_option(4, NAME, 197, Node020, NEUTRAL_REACTION);
 end
 
 procedure Node020
 begin
     poison(dude_obj, 1);
     item_caps_adjust(dude_obj, 20);
-    gsay_reply(108, 198);
-    giq_option(4, 108, 199, Node999, 50);
+    Reply(198);
+    giq_option(4, NAME, 199, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node021
 begin
     item_caps_adjust(self_obj, food_cost);
     item_caps_adjust(dude_obj, -food_cost);
-    critter_heal(dude_obj, get_critter_stat(dude_obj, 7) - get_critter_stat(dude_obj, 35));
+    critter_heal(dude_obj, get_critter_stat(dude_obj, STAT_max_hp) - get_critter_stat(dude_obj, STAT_current_hp));
     poison(dude_obj, -get_poison(dude_obj));
     set_local_var(5, local_var(5) bwor 1024);
-    gsay_reply(108, 200);
-    giq_option(4, 108, 201, Node999, 50);
+    Reply(200);
+    giq_option(4, NAME, 201, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node022
@@ -671,114 +671,114 @@ begin
     item_caps_adjust(dude_obj, -food_cost);
     if (food_cost == 1000) then begin
         if (((local_var(5) bwand 512) != 0) == 0) then begin
-            restock_obj := create_object_sid(378, 0, 0, -1);
+            restock_obj := create_object(378, 0, 0);
             add_mult_objs_to_inven(dude_obj, restock_obj, 1);
         end
         set_local_var(5, local_var(5) bwor 512);
-        gsay_reply(108, 300);
+        Reply(300);
     end
     else begin
-        gsay_reply(108, 202);
+        Reply(202);
     end
-    giq_option(4, 108, 203, Node024, 50);
-    giq_option(4, 108, message_str(14, random(375, 376)), Node999, 50);
+    giq_option(4, NAME, 203, Node024, NEUTRAL_REACTION);
+    giq_option(4, NAME, g_mstr(random(375, 376)), Node999, NEUTRAL_REACTION);
 end
 
 procedure Node023
 begin
-    gsay_reply(108, 204);
-    giq_option(4, 108, 205, Node010, 50);
-    giq_option(4, 108, message_str(14, random(375, 376)), Node999, 50);
+    Reply(204);
+    giq_option(4, NAME, 205, Node010, NEUTRAL_REACTION);
+    giq_option(4, NAME, g_mstr(random(375, 376)), Node999, NEUTRAL_REACTION);
 end
 
 procedure Node024
 begin
-    gsay_reply(108, 206);
-    if ((global_var(105) >= 4) and (global_var(105) < 7) or ((global_var(106) >= 4) and (global_var(106) < 7))) then begin
-        giq_option(4, 108, 207, Node025, 50);
+    Reply(206);
+    if ((global_var(GVAR_DUDE_STOMACH) >= 4) and (global_var(GVAR_DUDE_STOMACH) < 7) or ((global_var(GVAR_MODOC_FAMILY_FEUD_SEED_ONE) >= 4) and (global_var(GVAR_MODOC_FAMILY_FEUD_SEED_ONE) < 7))) then begin
+        giq_option(4, NAME, 207, Node025, NEUTRAL_REACTION);
     end
-    giq_option(4, 108, 208, Node027, 50);
-    if ((global_var(111) >= 4) and (global_var(111) < 7)) then begin
-        giq_option(4, 108, 209, Node028, 50);
+    giq_option(4, NAME, 208, Node027, NEUTRAL_REACTION);
+    if ((global_var(GVAR_MODOC_VERMIN_HUNTER_SEED) >= 4) and (global_var(GVAR_MODOC_VERMIN_HUNTER_SEED) < 7)) then begin
+        giq_option(4, NAME, 209, Node028, NEUTRAL_REACTION);
     end
     if ((local_var(5) bwand 1024) != 0) then begin
-        giq_option(6, 108, 210, Node029, 50);
+        giq_option(6, NAME, 210, Node029, NEUTRAL_REACTION);
     end
-    giq_option(4, 108, 211, Node030, 50);
-    giq_option(4, 108, message_str(14, random(375, 376)), Node999, 50);
+    giq_option(4, NAME, 211, Node030, NEUTRAL_REACTION);
+    giq_option(4, NAME, g_mstr(random(375, 376)), Node999, NEUTRAL_REACTION);
 end
 
 procedure Node025
 begin
-    gsay_reply(108, 212);
-    giq_option(4, 108, message_str(14, 353), Node026, 50);
+    Reply(212);
+    giq_option(4, NAME, g_mstr(353), Node026, NEUTRAL_REACTION);
 end
 
 procedure Node026
 begin
-    gsay_reply(108, 213);
-    giq_option(4, 108, 214, Node023, 50);
-    giq_option(4, 108, 215, Node010, 50);
-    giq_option(4, 108, message_str(14, random(375, 376)), Node999, 50);
+    Reply(213);
+    giq_option(4, NAME, 214, Node023, NEUTRAL_REACTION);
+    giq_option(4, NAME, 215, Node010, NEUTRAL_REACTION);
+    giq_option(4, NAME, g_mstr(random(375, 376)), Node999, NEUTRAL_REACTION);
 end
 
 procedure Node027
 begin
-    gsay_reply(108, 216);
-    giq_option(4, 108, message_str(14, 353), Node026, 50);
+    Reply(216);
+    giq_option(4, NAME, g_mstr(353), Node026, NEUTRAL_REACTION);
 end
 
 procedure Node028
 begin
-    gsay_reply(108, 217);
-    giq_option(4, 108, message_str(14, 353), Node026, 50);
+    Reply(217);
+    giq_option(4, NAME, g_mstr(353), Node026, NEUTRAL_REACTION);
 end
 
 procedure Node029
 begin
-    gsay_reply(108, 218);
-    giq_option(4, 108, message_str(14, 353), Node026, 50);
+    Reply(218);
+    giq_option(4, NAME, g_mstr(353), Node026, NEUTRAL_REACTION);
 end
 
 procedure Node030
 begin
-    gsay_reply(108, 220);
-    giq_option(4, 108, message_str(14, 353), Node026, 50);
+    Reply(220);
+    giq_option(4, NAME, g_mstr(353), Node026, NEUTRAL_REACTION);
 end
 
 procedure Node031
 begin
-    critter_heal(dude_obj, get_critter_stat(dude_obj, 7) - get_critter_stat(dude_obj, 35));
+    critter_heal(dude_obj, get_critter_stat(dude_obj, STAT_max_hp) - get_critter_stat(dude_obj, STAT_current_hp));
     poison(dude_obj, -get_poison(dude_obj));
-    gsay_reply(108, 221);
-    giq_option(1, 108, message_str(14, random(202, 203)), Node026, 50);
+    Reply(221);
+    giq_option(1, NAME, g_mstr(random(202, 203)), Node026, NEUTRAL_REACTION);
 end
 
 procedure Node032
 begin
-    gsay_reply(108, 222);
-    giq_option(4, 108, 156, Node010, 50);
-    giq_option(4, 108, 158, Node023, 50);
-    giq_option(4, 108, message_str(14, random(375, 376)), Node999, 50);
+    Reply(222);
+    giq_option(4, NAME, 156, Node010, NEUTRAL_REACTION);
+    giq_option(4, NAME, 158, Node023, NEUTRAL_REACTION);
+    giq_option(4, NAME, g_mstr(random(375, 376)), Node999, NEUTRAL_REACTION);
 end
 
 procedure Node033
 begin
     food_cost := 1000;
-    gsay_reply(108, 225);
-    giq_option(4, 108, 226, Node987, 50);
-    giq_option(4, 108, 227, Node010, 50);
-    giq_option(4, 108, message_str(14, random(375, 376)), Node999, 50);
+    Reply(225);
+    giq_option(4, NAME, 226, Node987, NEUTRAL_REACTION);
+    giq_option(4, NAME, 227, Node010, NEUTRAL_REACTION);
+    giq_option(4, NAME, g_mstr(random(375, 376)), Node999, NEUTRAL_REACTION);
 end
 
 procedure Node034
 begin
-    gsay_reply(108, 228);
-    if ((global_var(111) >= 7) == 0) then begin
-        giq_option(4, 108, 229, Node010, 50);
+    Reply(228);
+    if ((global_var(GVAR_MODOC_VERMIN_HUNTER_SEED) >= 7) == 0) then begin
+        giq_option(4, NAME, 229, Node010, NEUTRAL_REACTION);
     end
-    giq_option(4, 108, 230, Node999, 50);
-    giq_option(-3, 108, 231, Node999, 50);
+    giq_option(4, NAME, 230, Node999, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 231, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node999
@@ -792,7 +792,7 @@ end
 
 procedure Node996
 begin
-    if ((global_var(297) bwand 2) != 0) then begin
+    if ((global_var(GVAR_MODOC_GENERIC_FLAG_1) bwand 2) != 0) then begin
         call Node034();
     end
     else begin
@@ -802,7 +802,7 @@ end
 
 procedure Node995
 begin
-    if ((global_var(297) bwand 2) != 0) then begin
+    if ((global_var(GVAR_MODOC_GENERIC_FLAG_1) bwand 2) != 0) then begin
         call Node034();
     end
     else begin
@@ -823,7 +823,7 @@ end
 
 procedure Node993
 begin
-    if ((global_var(297) bwand 2) != 0) then begin
+    if ((global_var(GVAR_MODOC_GENERIC_FLAG_1) bwand 2) != 0) then begin
         call Node034();
     end
     else begin
@@ -895,8 +895,8 @@ end
 procedure timed_event_p_proc
 begin
     if (map_var(1) <= 3) then begin
-        float_msg(self_obj, message_str(108, 151 + map_var(1)), 8);
-        if (((global_var(402) bwand 262144) != 0) == 0) then begin
+        float_msg(self_obj, mstr(151 + map_var(1)), FLOAT_MSG_YELLOW);
+        if (((global_var(GVAR_MODOC_GENERIC_FLAG_2) bwand 262144) != 0) == 0) then begin
             set_map_var(1, map_var(1) + 1);
         end
     end
@@ -904,4 +904,3 @@ begin
         call Node998();
     end
 end
-
