diff --git a/Games/Fallout2.Steam/Original/MICOINBG.ssl b/Games/Fallout2.Steam/ReDefine/MICOINBG.ssl
index ba1577b..57156b2 100644
--- a/Games/Fallout2.Steam/Original/MICOINBG.ssl
+++ b/Games/Fallout2.Steam/ReDefine/MICOINBG.ssl
@@ -140,55 +140,55 @@ end
 procedure pickup_p_proc
 begin
     if (source_obj != 0) then begin
-        if ((source_obj == dude_obj) or (party_member_obj(obj_pid(source_obj)) != 0)) then begin
-            if (((global_var(297) bwand 1073741824) != 0) == 0) then begin
-                float_msg(dude_obj, message_str(195, 610), 8);
+        if ((source_obj == dude_obj) or (obj_in_party(source_obj))) then begin
+            if (((global_var(GVAR_MODOC_GENERIC_FLAG_1) bwand 1073741824) != 0) == 0) then begin
+                float_msg(dude_obj, message_str(SCRIPT_MODMAIN, 610), FLOAT_MSG_YELLOW);
             end
-            set_global_var(297, global_var(297) bwor 1073741824);
-            set_global_var(0, global_var(0) + -1);
+            set_global_var(GVAR_MODOC_GENERIC_FLAG_1, global_var(GVAR_MODOC_GENERIC_FLAG_1) bwor 1073741824);
+            set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + -1);
             debug_msg("Player gains " + -1 + " Karma Points.");
-            set_global_var(37, 0);
-            set_global_var(38, 0);
-            set_global_var(39, 0);
-            set_global_var(40, 0);
-            set_global_var(41, 0);
-            set_global_var(42, 0);
-            set_global_var(43, 0);
-            set_global_var(44, 0);
-            set_global_var(45, 0);
-            if (global_var(0) >= 1000) then begin
-                set_global_var(37, 1);
+            set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+            set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+            set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+            set_global_var(GVAR_KARMA_DEFENDER, 0);
+            set_global_var(GVAR_KARMA_WANDERER, 0);
+            set_global_var(GVAR_KARMA_BETRAYER, 0);
+            set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+            set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+            set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+                set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
             end
             else begin
-                if (global_var(0) >= 750) then begin
-                    set_global_var(38, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                    set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
                 end
                 else begin
-                    if (global_var(0) >= 500) then begin
-                        set_global_var(39, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                        set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                     end
                     else begin
-                        if (global_var(0) >= 250) then begin
-                            set_global_var(40, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                            set_global_var(GVAR_KARMA_DEFENDER, 1);
                         end
                         else begin
-                            if (global_var(0) > -250) then begin
-                                set_global_var(41, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                                set_global_var(GVAR_KARMA_WANDERER, 1);
                             end
                             else begin
-                                if (global_var(0) > -500) then begin
-                                    set_global_var(42, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                    set_global_var(GVAR_KARMA_BETRAYER, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -750) then begin
-                                        set_global_var(43, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                        set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                     end
                                     else begin
-                                        if (global_var(0) > -1000) then begin
-                                            set_global_var(44, 1);
+                                        if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                            set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                         end
                                         else begin
-                                            set_global_var(45, 1);
+                                            set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                         end
                                     end
                                 end
@@ -206,16 +206,15 @@ end
 procedure look_at_p_proc
 begin
     script_overrides;
-    display_msg(message_str(195, 600));
+    display_msg(message_str(SCRIPT_MODMAIN, 600));
 end
 
 procedure description_p_proc
 begin
     script_overrides;
-    display_msg(message_str(195, 601));
+    display_msg(message_str(SCRIPT_MODMAIN, 601));
 end
 
 procedure timed_event_p_proc
 begin
 end
-
