diff --git a/Games/Fallout2.Steam/Original/MODBRAH.ssl b/Games/Fallout2.Steam/ReDefine/MODBRAH.ssl
index 302123f..01a3b9a 100644
--- a/Games/Fallout2.Steam/Original/MODBRAH.ssl
+++ b/Games/Fallout2.Steam/ReDefine/MODBRAH.ssl
@@ -146,8 +146,8 @@ procedure map_enter_p_proc
 begin
     last_update := game_time;
     override_map_start(19524 % 200, 19524 / 200, 0, 5);
-    set_global_var(46, 29736);
-    set_global_var(67, 0);
+    set_global_var(GVAR_MAP_EXIT_TILE, 29736);
+    set_global_var(GVAR_MAP_NEXT_TILE, 0);
     if ((get_month >= 3) and (get_month < 5)) then begin
         if ((game_time_hour >= 600) and (game_time_hour < (600 + 100))) then begin
             set_light_level(game_time_hour - 600 + 40);
@@ -224,26 +224,26 @@ begin
             end
         end
     end
-    if (metarule(22, 0) == 0) then begin
-        if (global_var(99) == 0) then begin
-            set_global_var(99, game_time + ((24 * (60 * (60 * 10)) * (game_time_hour > 2259)) + (((2259 / 100) - (game_time_hour / 100)) * (60 * (60 * 10))) + (((2259 % 100) - (game_time_hour % 100)) * (60 * 10))));
-            if (global_var(99) == 0) then begin
-                set_global_var(99, game_time + (60 * 10));
+    if (is_loading_game == 0) then begin
+        if (global_var(GVAR_MODOC_KILL_ALL_BRAHMIN_TIME) == 0) then begin
+            set_global_var(GVAR_MODOC_KILL_ALL_BRAHMIN_TIME, game_time + ((24 * (60 * (60 * 10)) * (game_time_hour > 2259)) + (((2259 / 100) - (game_time_hour / 100)) * (60 * (60 * 10))) + (((2259 % 100) - (game_time_hour % 100)) * (60 * 10))));
+            if (global_var(GVAR_MODOC_KILL_ALL_BRAHMIN_TIME) == 0) then begin
+                set_global_var(GVAR_MODOC_KILL_ALL_BRAHMIN_TIME, game_time + (60 * 10));
             end
         end
         else begin
-            if ((global_var(99) - game_time) < 0) then begin
-                if (((global_var(116) < 0) == 0) and (global_var(99) != 0)) then begin
-                    if ((global_var(115) < 0) == 0) then begin
-                        set_global_var(115, -1);
+            if ((global_var(GVAR_MODOC_KILL_ALL_BRAHMIN_TIME) - game_time) < 0) then begin
+                if (((global_var(GVAR_MODOC_BRAHMIN_ALIVE) < 0) == 0) and (global_var(GVAR_MODOC_KILL_ALL_BRAHMIN_TIME) != 0)) then begin
+                    if ((global_var(GVAR_JONNY_TILE) < 0) == 0) then begin
+                        set_global_var(GVAR_JONNY_TILE, -1);
                     end
-                    if (cur_map_index == 20) then begin
+                    if (CUR_MAP_MODBRAH) then begin
                         kill_critter_type(16777226, random(61, 62));
                         if (map_var(0) == 0) then begin
-                            if (global_var(115) < 0) then begin
-                                set_map_var(0, random(0, global_var(116) / 2));
+                            if (global_var(GVAR_JONNY_TILE) < 0) then begin
+                                set_map_var(0, random(0, global_var(GVAR_MODOC_BRAHMIN_ALIVE) / 2));
                                 while (map_var(0) > 0) do begin
-                                    temp_dog := create_object_sid(16777225, 0, 0, 206);
+                                    temp_dog := create_object_sid(16777225, 0, 0, SCRIPT_MCATKDOG);
                                     critter_attempt_placement(temp_dog, tile_num_in_direction(16915, random(0, 5), random(2, 10)), 0);
                                     set_map_var(0, map_var(0) - 1);
                                 end
@@ -260,22 +260,22 @@ end
 
 procedure map_exit_p_proc
 begin
-    if ((((global_var(402) bwand 4) != 0) or (global_var(388) == 4)) == 0) then begin
-        if (((global_var(402) bwand 1024) != 0) and (global_var(310) <= 0) or ((global_var(112) == 2) or (((global_var(403) - (game_time / (10 * 60 * 60 * 24))) <= 0) or ((global_var(403) - (game_time / (10 * 60 * 60 * 24))) > 31) and (global_var(403) > 0)))) then begin
-            set_global_var(402, global_var(402) bwor 4);
+    if ((((global_var(GVAR_MODOC_GENERIC_FLAG_2) bwand 4) != 0) or (global_var(GVAR_MODOC_ENDINGS) == 4)) == 0) then begin
+        if (((global_var(GVAR_MODOC_GENERIC_FLAG_2) bwand 1024) != 0) and (global_var(GVAR_SLAG_CNT) <= 0) or ((global_var(GVAR_MODOC_GHOST_FARM_SEED) == 2) or (((global_var(GVAR_SLAUGHTER_SLAG_TIME) - (game_time / (10 * 60 * 60 * 24))) <= 0) or ((global_var(GVAR_SLAUGHTER_SLAG_TIME) - (game_time / (10 * 60 * 60 * 24))) > 31) and (global_var(GVAR_SLAUGHTER_SLAG_TIME) > 0)))) then begin
+            set_global_var(GVAR_MODOC_GENERIC_FLAG_2, global_var(GVAR_MODOC_GENERIC_FLAG_2) bwor 4);
         end
     end
-    if (((global_var(116) < 0) == 0) and (global_var(99) != 0)) then begin
-        if ((global_var(115) < 0) == 0) then begin
-            set_global_var(115, -1);
+    if (((global_var(GVAR_MODOC_BRAHMIN_ALIVE) < 0) == 0) and (global_var(GVAR_MODOC_KILL_ALL_BRAHMIN_TIME) != 0)) then begin
+        if ((global_var(GVAR_JONNY_TILE) < 0) == 0) then begin
+            set_global_var(GVAR_JONNY_TILE, -1);
         end
-        if (cur_map_index == 20) then begin
+        if (CUR_MAP_MODBRAH) then begin
             kill_critter_type(16777226, random(61, 62));
             if (map_var(0) == 0) then begin
-                if (global_var(115) < 0) then begin
-                    set_map_var(0, random(0, global_var(116) / 2));
+                if (global_var(GVAR_JONNY_TILE) < 0) then begin
+                    set_map_var(0, random(0, global_var(GVAR_MODOC_BRAHMIN_ALIVE) / 2));
                     while (map_var(0) > 0) do begin
-                        temp_dog := create_object_sid(16777225, 0, 0, 206);
+                        temp_dog := create_object_sid(16777225, 0, 0, SCRIPT_MCATKDOG);
                         critter_attempt_placement(temp_dog, tile_num_in_direction(16915, random(0, 5), random(2, 10)), 0);
                         set_map_var(0, map_var(0) - 1);
                     end
@@ -366,18 +366,18 @@ begin
         end
     end
     if (((game_time - last_update) > (15 * (60 * 10))) and (last_update != 0)) then begin
-        if ((global_var(99) - game_time) < 0) then begin
-            if (((global_var(116) < 0) == 0) and (global_var(99) != 0)) then begin
-                if ((global_var(115) < 0) == 0) then begin
-                    set_global_var(115, -1);
+        if ((global_var(GVAR_MODOC_KILL_ALL_BRAHMIN_TIME) - game_time) < 0) then begin
+            if (((global_var(GVAR_MODOC_BRAHMIN_ALIVE) < 0) == 0) and (global_var(GVAR_MODOC_KILL_ALL_BRAHMIN_TIME) != 0)) then begin
+                if ((global_var(GVAR_JONNY_TILE) < 0) == 0) then begin
+                    set_global_var(GVAR_JONNY_TILE, -1);
                 end
-                if (cur_map_index == 20) then begin
+                if (CUR_MAP_MODBRAH) then begin
                     kill_critter_type(16777226, random(61, 62));
                     if (map_var(0) == 0) then begin
-                        if (global_var(115) < 0) then begin
-                            set_map_var(0, random(0, global_var(116) / 2));
+                        if (global_var(GVAR_JONNY_TILE) < 0) then begin
+                            set_map_var(0, random(0, global_var(GVAR_MODOC_BRAHMIN_ALIVE) / 2));
                             while (map_var(0) > 0) do begin
-                                temp_dog := create_object_sid(16777225, 0, 0, 206);
+                                temp_dog := create_object_sid(16777225, 0, 0, SCRIPT_MCATKDOG);
                                 critter_attempt_placement(temp_dog, tile_num_in_direction(16915, random(0, 5), random(2, 10)), 0);
                                 set_map_var(0, map_var(0) - 1);
                             end
@@ -391,6 +391,5 @@ begin
     end
     last_update := game_time;
     debug_msg("  game_time_hour == " + game_time_hour);
-    debug_msg("  countdown in ticks == " + (global_var(99) - game_time));
+    debug_msg("  countdown in ticks == " + (global_var(GVAR_MODOC_KILL_ALL_BRAHMIN_TIME) - game_time));
 end
-
