diff --git a/Games/Fallout2.Steam/Original/NCANOUNC.ssl b/Games/Fallout2.Steam/ReDefine/NCANOUNC.ssl
index d1fb72a..61caf4e 100644
--- a/Games/Fallout2.Steam/Original/NCANOUNC.ssl
+++ b/Games/Fallout2.Steam/ReDefine/NCANOUNC.ssl
@@ -167,53 +167,53 @@ end
 procedure ChooseFromList
 begin
     variable LVar0 := 0;
-    gsay_reply(1086, stored_msg);
-    if (message_str(353, top_msg - 1) != "") then begin
-        giq_option(1, 1086, message_str(353, 150), CLICK_PREV, 50);
+    Reply(stored_msg);
+    if (message_str(SCRIPT_NEWR1, top_msg - 1) != "") then begin
+        giq_option(1, NAME, message_str(SCRIPT_NEWR1, 150), CLICK_PREV, NEUTRAL_REACTION);
     end
     if (LVar0 == 0) then begin
-        if (message_str(353, top_msg + 0) != "") then begin
-            giq_option(1, 1086, message_str(353, top_msg + 0), CLICK_0, 50);
+        if (message_str(SCRIPT_NEWR1, top_msg + 0) != "") then begin
+            giq_option(1, NAME, message_str(SCRIPT_NEWR1, top_msg + 0), CLICK_0, NEUTRAL_REACTION);
         end
         else begin
             LVar0 := 1;
         end
     end
     if (LVar0 == 0) then begin
-        if (message_str(353, top_msg + 1) != "") then begin
-            giq_option(1, 1086, message_str(353, top_msg + 1), CLICK_1, 50);
+        if (message_str(SCRIPT_NEWR1, top_msg + 1) != "") then begin
+            giq_option(1, NAME, message_str(SCRIPT_NEWR1, top_msg + 1), CLICK_1, NEUTRAL_REACTION);
         end
         else begin
             LVar0 := 1;
         end
     end
     if (LVar0 == 0) then begin
-        if (message_str(353, top_msg + 2) != "") then begin
-            giq_option(1, 1086, message_str(353, top_msg + 2), CLICK_2, 50);
+        if (message_str(SCRIPT_NEWR1, top_msg + 2) != "") then begin
+            giq_option(1, NAME, message_str(SCRIPT_NEWR1, top_msg + 2), CLICK_2, NEUTRAL_REACTION);
         end
         else begin
             LVar0 := 1;
         end
     end
     if (LVar0 == 0) then begin
-        if (message_str(353, top_msg + 3) != "") then begin
-            giq_option(1, 1086, message_str(353, top_msg + 3), CLICK_3, 50);
+        if (message_str(SCRIPT_NEWR1, top_msg + 3) != "") then begin
+            giq_option(1, NAME, message_str(SCRIPT_NEWR1, top_msg + 3), CLICK_3, NEUTRAL_REACTION);
         end
         else begin
             LVar0 := 1;
         end
     end
     if (LVar0 == 0) then begin
-        if (message_str(353, top_msg + 4) != "") then begin
-            giq_option(1, 1086, message_str(353, top_msg + 4), CLICK_4, 50);
+        if (message_str(SCRIPT_NEWR1, top_msg + 4) != "") then begin
+            giq_option(1, NAME, message_str(SCRIPT_NEWR1, top_msg + 4), CLICK_4, NEUTRAL_REACTION);
         end
         else begin
             LVar0 := 1;
         end
     end
     if (LVar0 == 0) then begin
-        if (message_str(353, top_msg + 5) != "") then begin
-            giq_option(1, 1086, message_str(353, 151), CLICK_MORE, 50);
+        if (message_str(SCRIPT_NEWR1, top_msg + 5) != "") then begin
+            giq_option(1, NAME, message_str(SCRIPT_NEWR1, 151), CLICK_MORE, NEUTRAL_REACTION);
         end
     end
 end
@@ -267,21 +267,21 @@ end
 
 procedure critter_p_proc
 begin
-    if (has_trait(1, self_obj, 666)) then begin
+    if (obj_is_visible_flag(self_obj)) then begin
         if (allow_anim == 0) then begin
             reg_anim_func(2, dude_obj);
         end
         else begin
             if (combat_is_initialized == 0) then begin
-                if (cur_map_index == 55) then begin
+                if (CUR_MAP_NEWR2) then begin
                     if (announcer_obj != 0) then begin
                         last_timed_obj := announcer_obj;
                         last_timed_param := last_timed_param;
-                        metarule3(100, announcer_obj, 48, 0);
+                        rm_fixed_timer_event(announcer_obj, 48);
                         add_timer_event(announcer_obj, game_ticks(0), 48);
                     end
                 end
-                set_global_var(694, global_var(694) bwor 2);
+                set_global_var(GVAR_NEW_RENO_FLAG_4, global_var(GVAR_NEW_RENO_FLAG_4) bwor 2);
             end
         end
     end
@@ -294,30 +294,30 @@ end
 
 procedure damage_p_proc
 begin
-    if (cur_map_index == 55) then begin
+    if (CUR_MAP_NEWR2) then begin
         if (announcer_obj != 0) then begin
             last_timed_obj := announcer_obj;
             last_timed_param := last_timed_param;
-            metarule3(100, announcer_obj, 48, 0);
+            rm_fixed_timer_event(announcer_obj, 48);
             add_timer_event(announcer_obj, game_ticks(0), 48);
         end
     end
-    set_global_var(694, global_var(694) bwor 2);
-    if (get_critter_stat(self_obj, 35) < get_critter_stat(self_obj, 7)) then begin
-        critter_heal(self_obj, get_critter_stat(self_obj, 7) - get_critter_stat(self_obj, 35));
+    set_global_var(GVAR_NEW_RENO_FLAG_4, global_var(GVAR_NEW_RENO_FLAG_4) bwor 2);
+    if (get_critter_stat(self_obj, STAT_current_hp) < get_critter_stat(self_obj, STAT_max_hp)) then begin
+        critter_heal(self_obj, get_critter_stat(self_obj, STAT_max_hp) - get_critter_stat(self_obj, STAT_current_hp));
     end
 end
 
 procedure map_enter_p_proc
 begin
     announcer_obj := self_obj;
-    if (metarule(22, 0) == 0) then begin
-        if (metarule(14, 0)) then begin
+    if (is_loading_game == 0) then begin
+        if (map_first_run) then begin
             set_local_var(1, tile_num(self_obj));
-            critter_add_trait(self_obj, 1, 6, 54);
-            critter_add_trait(self_obj, 1, 5, 49);
+            critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 54);
+            critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_LONER_CITIZEN);
         end
-        if (has_trait(1, self_obj, 666) == 1) then begin
+        if (obj_is_visible_flag(self_obj) == 1) then begin
             set_obj_visibility(self_obj, 1);
         end
     end
@@ -328,13 +328,13 @@ begin
     variable LVar0 := 0;
     variable LVar1 := 0;
     if (fixed_param == 48) then begin
-        if ((global_var(694) bwand 2) != 0) then begin
+        if ((global_var(GVAR_NEW_RENO_FLAG_4) bwand 2) != 0) then begin
             if (combat_is_initialized == 0) then begin
                 game_ui_disable;
             end
             else begin
                 terminate_combat;
-                metarule3(100, self_obj, 48, 0);
+                rm_fixed_timer_event(self_obj, 48);
                 add_timer_event(self_obj, 3, 48);
             end
         end
@@ -344,103 +344,103 @@ begin
     end
     else begin
         if (fixed_param == 110) then begin
-            float_msg(self_obj, " ", 8);
+            float_msg(self_obj, " ", FLOAT_MSG_YELLOW);
             debug_msg("start boxing round: start");
             gfade_out(10);
             game_time_advance(60 * 10);
-            if (has_trait(1, announcer_obj, 666) == 1) then begin
+            if (obj_is_visible_flag(announcer_obj) == 1) then begin
                 set_obj_visibility(announcer_obj, 1);
             end
-            if (has_trait(1, ring_girl_obj, 666) == 1) then begin
+            if (obj_is_visible_flag(ring_girl_obj) == 1) then begin
                 set_obj_visibility(ring_girl_obj, 1);
             end
-            float_msg(dude_obj, message_str(1087, 100) + global_var(506) + message_str(1087, 101), -2);
+            float_msg(dude_obj, message_str(SCRIPT_NCRNGGRL, 100) + global_var(GVAR_NEW_RENO_ROUND_NUMBER) + message_str(SCRIPT_NCRNGGRL, 101), FLOAT_MSG_WARNING);
             debug_msg("move_boxers_to_corners: start");
             move_to(dude_obj, 24479, elevation(self_obj));
             critter_injure(dude_obj, 2 bwor 8388608);
-            if ((prize_fighter_obj != 0) and (((global_var(495) bwand 8192) != 0) == 0)) then begin
+            if ((prize_fighter_obj != 0) and (((global_var(GVAR_NEW_RENO_FLAG_3) bwand 8192) != 0) == 0)) then begin
                 move_to(prize_fighter_obj, 26471, elevation(self_obj));
                 animate_stand_obj(prize_fighter_obj);
                 critter_injure(prize_fighter_obj, 2 bwor 8388608);
-                anim(dude_obj, 1000, rotation_to_tile(tile_num(dude_obj), tile_num(prize_fighter_obj)));
-                anim(prize_fighter_obj, 1000, rotation_to_tile(tile_num(prize_fighter_obj), tile_num(dude_obj)));
+                anim(dude_obj, 1000, rotation_to_tile(dude_tile, tile_num(prize_fighter_obj)));
+                anim(prize_fighter_obj, 1000, rotation_to_tile(tile_num(prize_fighter_obj), dude_tile));
             end
             animate_stand_obj(dude_obj);
             debug_msg("move_boxers_to_corners: end");
-            set_global_var(507, game_time + (60 * 10));
+            set_global_var(GVAR_NEW_RENO_ROUND_TIME, game_time + (60 * 10));
             gfade_in(10);
-            if (cur_map_index == 55) then begin
+            if (CUR_MAP_NEWR2) then begin
                 if (announcer_obj != 0) then begin
                     last_timed_obj := announcer_obj;
                     last_timed_param := last_timed_param;
-                    metarule3(100, announcer_obj, 48, 0);
+                    rm_fixed_timer_event(announcer_obj, 48);
                     add_timer_event(announcer_obj, game_ticks(0), 48);
                 end
             end
-            set_global_var(694, global_var(694) bwand (-1 - 2));
+            set_global_var(GVAR_NEW_RENO_FLAG_4, global_var(GVAR_NEW_RENO_FLAG_4) bwand (-1 - 2));
             play_sfx("boxbel1");
             attack_setup(prize_fighter_obj, dude_obj);
             debug_msg("start boxing round: end");
         end
         else begin
-            if (has_trait(1, self_obj, 666)) then begin
+            if (obj_is_visible_flag(self_obj)) then begin
                 debug_msg("announcer timed event, fixed param == " + fixed_param);
-                debug_msg("  fight_status == " + (global_var(510) + (-2 * global_var(510) * (global_var(510) < 0))));
+                debug_msg("  fight_status == " + (global_var(GVAR_NEW_RENO_FIGHT_STATUS) + (-2 * global_var(GVAR_NEW_RENO_FIGHT_STATUS) * (global_var(GVAR_NEW_RENO_FIGHT_STATUS) < 0))));
                 if ((103 <= fixed_param) and (fixed_param <= 105)) then begin
-                    float_msg(self_obj, message_str(1086, fixed_param + (local_var(2) % 4 * 20)), 8);
+                    float_msg(self_obj, mstr(fixed_param + (local_var(2) % 4 * 20)), FLOAT_MSG_YELLOW);
                     last_timed_obj := announcer_obj;
                     last_timed_param := last_timed_param;
-                    metarule3(100, announcer_obj, fixed_param + 1, 0);
+                    rm_fixed_timer_event(announcer_obj, fixed_param + 1);
                     add_timer_event(announcer_obj, game_ticks(3), fixed_param + 1);
                 end
                 else begin
                     if (fixed_param == 106) then begin
                         set_local_var(2, local_var(2) + 1);
-                        float_msg(self_obj, message_str(1086, 106) + (100 + ((get_critter_stat(dude_obj, 34) == 0) * 50) + (get_critter_stat(dude_obj, 0) * 5) - ((get_critter_stat(dude_obj, 5) + get_critter_stat(dude_obj, 2)) / 3)) + message_str(1086, 1106) + message_str(353, 400 + global_var(231)) + message_str(1086, 2106), 8);
-                        anim(self_obj, 1000, rotation_to_tile(tile_num(self_obj), tile_num(dude_obj)));
+                        float_msg(self_obj, mstr(106) + (100 + ((dude_gender == 0) * 50) + (dude_strength * 5) - ((get_critter_stat(dude_obj, STAT_ag) + dude_endurance) / 3)) + mstr(1106) + message_str(SCRIPT_NEWR1, 400 + global_var(GVAR_NEW_RENO_PRIZEFIGHTER)) + mstr(2106), FLOAT_MSG_YELLOW);
+                        anim(self_obj, 1000, rotation_to_tile(tile_num(self_obj), dude_tile));
                         last_timed_obj := announcer_obj;
                         last_timed_param := last_timed_param;
-                        metarule3(100, announcer_obj, fixed_param + 1, 0);
+                        rm_fixed_timer_event(announcer_obj, fixed_param + 1);
                         add_timer_event(announcer_obj, game_ticks(8), fixed_param + 1);
                     end
                     else begin
                         if (fixed_param == 107) then begin
                             if (prize_fighter_obj != 0) then begin
-                                float_msg(self_obj, message_str(1086, 107) + (100 + ((get_critter_stat(prize_fighter_obj, 34) == 0) * 50) + (get_critter_stat(prize_fighter_obj, 0) * 5) - ((get_critter_stat(prize_fighter_obj, 5) + get_critter_stat(prize_fighter_obj, 2)) / 3)) + message_str(1086, 1107) + message_str(353, 700 + (50 * ((global_var(495) bwand 4096) != 0)) + global_var(504)) + message_str(1086, 2107), 8);
+                                float_msg(self_obj, mstr(107) + (100 + ((get_critter_stat(prize_fighter_obj, STAT_gender) == 0) * 50) + (get_critter_stat(prize_fighter_obj, STAT_st) * 5) - ((get_critter_stat(prize_fighter_obj, STAT_ag) + get_critter_stat(prize_fighter_obj, STAT_en)) / 3)) + mstr(1107) + message_str(SCRIPT_NEWR1, 700 + (50 * ((global_var(GVAR_NEW_RENO_FLAG_3) bwand 4096) != 0)) + global_var(GVAR_NEW_RENO_FIGHT_LEVEL)) + mstr(2107), FLOAT_MSG_YELLOW);
                                 anim(self_obj, 1000, rotation_to_tile(tile_num(self_obj), tile_num(prize_fighter_obj)));
                             end
                             last_timed_obj := announcer_obj;
                             last_timed_param := last_timed_param;
-                            metarule3(100, announcer_obj, fixed_param + 1, 0);
+                            rm_fixed_timer_event(announcer_obj, fixed_param + 1);
                             add_timer_event(announcer_obj, game_ticks(8), fixed_param + 1);
                         end
                         else begin
                             if (fixed_param == 201) then begin
-                                if ((global_var(510) + (-2 * global_var(510) * (global_var(510) < 0))) != 4) then begin
-                                    float_msg(self_obj, message_str(1086, 200 + (global_var(510) + (-2 * global_var(510) * (global_var(510) < 0)))), 8);
+                                if ((global_var(GVAR_NEW_RENO_FIGHT_STATUS) + (-2 * global_var(GVAR_NEW_RENO_FIGHT_STATUS) * (global_var(GVAR_NEW_RENO_FIGHT_STATUS) < 0))) != 4) then begin
+                                    float_msg(self_obj, mstr(200 + (global_var(GVAR_NEW_RENO_FIGHT_STATUS) + (-2 * global_var(GVAR_NEW_RENO_FIGHT_STATUS) * (global_var(GVAR_NEW_RENO_FIGHT_STATUS) < 0)))), FLOAT_MSG_YELLOW);
                                 end
                                 else begin
-                                    if (global_var(510) > 0) then begin
-                                        LVar0 := global_var(508);
-                                        LVar1 := global_var(509);
+                                    if (global_var(GVAR_NEW_RENO_FIGHT_STATUS) > 0) then begin
+                                        LVar0 := global_var(GVAR_NEW_RENO_DUDE_SCORE);
+                                        LVar1 := global_var(GVAR_NEW_RENO_BOXER_SCORE);
                                     end
                                     else begin
-                                        LVar0 := global_var(509);
-                                        LVar1 := global_var(508);
+                                        LVar0 := global_var(GVAR_NEW_RENO_BOXER_SCORE);
+                                        LVar1 := global_var(GVAR_NEW_RENO_DUDE_SCORE);
                                     end
                                     LVar1 := LVar1 - (LVar1 == LVar0);
-                                    float_msg(self_obj, message_str(1086, 204) + LVar0 + message_str(1086, 205) + LVar1 + message_str(1086, 206), 8);
+                                    float_msg(self_obj, mstr(204) + LVar0 + mstr(205) + LVar1 + mstr(206), FLOAT_MSG_YELLOW);
                                 end
                                 last_timed_obj := announcer_obj;
                                 last_timed_param := last_timed_param;
-                                metarule3(100, announcer_obj, 210, 0);
+                                rm_fixed_timer_event(announcer_obj, 210);
                                 add_timer_event(announcer_obj, game_ticks(4), 210);
                             end
                             else begin
                                 if (fixed_param == 210) then begin
-                                    if (global_var(510) > 0) then begin
+                                    if (global_var(GVAR_NEW_RENO_FIGHT_STATUS) > 0) then begin
                                         allow_anim := 1;
-                                        float_msg(self_obj, message_str(353, 400 + global_var(231)) + message_str(1086, 210), 8);
+                                        float_msg(self_obj, message_str(SCRIPT_NEWR1, 400 + global_var(GVAR_NEW_RENO_PRIZEFIGHTER)) + mstr(210), FLOAT_MSG_YELLOW);
                                         reg_anim_func(2, dude_obj);
                                         reg_anim_func(1, 1);
                                         reg_anim_animate(dude_obj, 16, -1);
@@ -448,7 +448,7 @@ begin
                                         reg_anim_func(3, 0);
                                     end
                                     else begin
-                                        float_msg(self_obj, message_str(353, 700 + (50 * ((global_var(495) bwand 4096) != 0)) + global_var(504)) + message_str(1086, 210), 8);
+                                        float_msg(self_obj, message_str(SCRIPT_NEWR1, 700 + (50 * ((global_var(GVAR_NEW_RENO_FLAG_3) bwand 4096) != 0)) + global_var(GVAR_NEW_RENO_FIGHT_LEVEL)) + mstr(210), FLOAT_MSG_YELLOW);
                                         reg_anim_func(2, prize_fighter_obj);
                                         reg_anim_func(1, 1);
                                         reg_anim_animate(prize_fighter_obj, 16, -1);
@@ -457,27 +457,27 @@ begin
                                     end
                                     last_timed_obj := announcer_obj;
                                     last_timed_param := last_timed_param;
-                                    metarule3(100, announcer_obj, fixed_param + 1, 0);
+                                    rm_fixed_timer_event(announcer_obj, fixed_param + 1);
                                     add_timer_event(announcer_obj, game_ticks(4), fixed_param + 1);
                                 end
                                 else begin
                                     if (fixed_param == 211) then begin
                                         debug_msg("cleanup boxing match: start");
                                         gfade_out(10);
-                                        if (cur_map_index == 55) then begin
+                                        if (CUR_MAP_NEWR2) then begin
                                             if (announcer_obj != 0) then begin
                                                 last_timed_obj := announcer_obj;
                                                 last_timed_param := last_timed_param;
-                                                metarule3(100, announcer_obj, 48, 0);
+                                                rm_fixed_timer_event(announcer_obj, 48);
                                                 add_timer_event(announcer_obj, game_ticks(0), 48);
                                             end
                                         end
-                                        set_global_var(694, global_var(694) bwor 2);
-                                        set_global_var(495, global_var(495) bwand (-1 - 536870912));
+                                        set_global_var(GVAR_NEW_RENO_FLAG_4, global_var(GVAR_NEW_RENO_FLAG_4) bwor 2);
+                                        set_global_var(GVAR_NEW_RENO_FLAG_3, global_var(GVAR_NEW_RENO_FLAG_3) bwand (-1 - 536870912));
                                         if (prize_fighter_obj != 0) then begin
                                             destroy_object(prize_fighter_obj);
                                         end
-                                        if (has_trait(1, announcer_obj, 666) == 1) then begin
+                                        if (obj_is_visible_flag(announcer_obj) == 1) then begin
                                             set_obj_visibility(announcer_obj, 1);
                                         end
                                         removed_qty := obj_is_carrying_obj_pid(dude_obj, 497);
@@ -488,17 +488,17 @@ begin
                                             restock_obj := obj_carrying_pid_obj(dude_obj, 497);
                                             if (obj_type(self_obj) == 1) then begin
                                                 restock_obj := obj_carrying_pid_obj(dude_obj, 497);
-                                                if (critter_inven_obj(self_obj, 0) == restock_obj) then begin
-                                                    if (obj_item_subtype(critter_inven_obj(dude_obj, 0)) == 0) then begin
-                                                        restock_obj := critter_inven_obj(dude_obj, 0);
+                                                if (critter_inven_obj(self_obj, INVEN_TYPE_WORN) == restock_obj) then begin
+                                                    if (obj_item_subtype(critter_inven_obj(dude_obj, INVEN_TYPE_WORN)) == 0) then begin
+                                                        restock_obj := critter_inven_obj(dude_obj, INVEN_TYPE_WORN);
                                                         debug_msg("armour pid == " + obj_pid(restock_obj));
                                                         rm_obj_from_inven(dude_obj, restock_obj);
                                                         add_obj_to_inven(dude_obj, restock_obj);
                                                     end
                                                 end
                                                 else begin
-                                                    if ((critter_inven_obj(dude_obj, 2) == restock_obj) or (critter_inven_obj(dude_obj, 1) == restock_obj)) then begin
-                                                        metarule(43, dude_obj);
+                                                    if ((critter_inven_obj(dude_obj, INVEN_TYPE_LEFT_HAND) == restock_obj) or (critter_inven_obj(dude_obj, INVEN_TYPE_RIGHT_HAND) == restock_obj)) then begin
+                                                        inven_unwield(dude_obj);
                                                     end
                                                 end
                                             end
@@ -515,17 +515,17 @@ begin
                                             restock_obj := obj_carrying_pid_obj(dude_obj, 496);
                                             if (obj_type(self_obj) == 1) then begin
                                                 restock_obj := obj_carrying_pid_obj(dude_obj, 496);
-                                                if (critter_inven_obj(self_obj, 0) == restock_obj) then begin
-                                                    if (obj_item_subtype(critter_inven_obj(dude_obj, 0)) == 0) then begin
-                                                        restock_obj := critter_inven_obj(dude_obj, 0);
+                                                if (critter_inven_obj(self_obj, INVEN_TYPE_WORN) == restock_obj) then begin
+                                                    if (obj_item_subtype(critter_inven_obj(dude_obj, INVEN_TYPE_WORN)) == 0) then begin
+                                                        restock_obj := critter_inven_obj(dude_obj, INVEN_TYPE_WORN);
                                                         debug_msg("armour pid == " + obj_pid(restock_obj));
                                                         rm_obj_from_inven(dude_obj, restock_obj);
                                                         add_obj_to_inven(dude_obj, restock_obj);
                                                     end
                                                 end
                                                 else begin
-                                                    if ((critter_inven_obj(dude_obj, 2) == restock_obj) or (critter_inven_obj(dude_obj, 1) == restock_obj)) then begin
-                                                        metarule(43, dude_obj);
+                                                    if ((critter_inven_obj(dude_obj, INVEN_TYPE_LEFT_HAND) == restock_obj) or (critter_inven_obj(dude_obj, INVEN_TYPE_RIGHT_HAND) == restock_obj)) then begin
+                                                        inven_unwield(dude_obj);
                                                     end
                                                 end
                                             end
@@ -535,30 +535,30 @@ begin
                                         restock_obj := 0;
                                         restock_amt := 0;
                                         move_obj_inven_to_obj(new_reno_temp_box, dude_obj);
-                                        critter_attempt_placement(dude_obj, tile_num_in_direction(tile_num(stuart_little_obj), has_trait(1, stuart_little_obj, 10), 1), elevation(self_obj));
-                                        anim(dude_obj, 1000, rotation_to_tile(tile_num(dude_obj), tile_num(stuart_little_obj)));
-                                        anim(stuart_little_obj, 1000, rotation_to_tile(tile_num(stuart_little_obj), tile_num(dude_obj)));
-                                        if (has_trait(1, ring_girl_obj, 666) == 1) then begin
+                                        critter_attempt_placement(dude_obj, tile_num_in_direction(tile_num(stuart_little_obj), obj_get_rot(stuart_little_obj), 1), elevation(self_obj));
+                                        anim(dude_obj, 1000, rotation_to_tile(dude_tile, tile_num(stuart_little_obj)));
+                                        anim(stuart_little_obj, 1000, rotation_to_tile(tile_num(stuart_little_obj), dude_tile));
+                                        if (obj_is_visible_flag(ring_girl_obj) == 1) then begin
                                             set_obj_visibility(ring_girl_obj, 1);
                                         end
-                                        if (has_trait(1, announcer_obj, 666) == 1) then begin
+                                        if (obj_is_visible_flag(announcer_obj) == 1) then begin
                                             set_obj_visibility(announcer_obj, 1);
                                         end
                                         last_timed_obj := stuart_little_obj;
                                         last_timed_param := last_timed_param;
-                                        metarule3(100, stuart_little_obj, 34, 0);
+                                        rm_fixed_timer_event(stuart_little_obj, 34);
                                         add_timer_event(stuart_little_obj, game_ticks(2), 34);
                                         game_time_advance(60 * 10);
                                         debug_msg("cleanup boxing match: end");
                                     end
                                     else begin
-                                        if (has_trait(1, self_obj, 10) != 3) then begin
+                                        if (obj_get_rot(self_obj) != 3) then begin
                                             anim(self_obj, 1000, 3);
                                         end
-                                        float_msg(self_obj, message_str(1086, fixed_param), 8);
+                                        float_msg(self_obj, mstr(fixed_param), FLOAT_MSG_YELLOW);
                                         last_timed_obj := announcer_obj;
                                         last_timed_param := last_timed_param;
-                                        metarule3(100, announcer_obj, fixed_param + 1, 0);
+                                        rm_fixed_timer_event(announcer_obj, fixed_param + 1);
                                         add_timer_event(announcer_obj, game_ticks(4), fixed_param + 1);
                                     end
                                 end
@@ -578,8 +578,7 @@ end
 
 procedure combat_p_proc
 begin
-    if (has_trait(1, self_obj, 666)) then begin
+    if (obj_is_visible_flag(self_obj)) then begin
         terminate_combat;
     end
 end
-
