diff --git a/Games/Fallout2.Steam/Original/NCANOUN.ssl b/Games/Fallout2.Steam/ReDefine/NCANOUN.ssl
index 1cbc087..2575241 100644
--- a/Games/Fallout2.Steam/Original/NCANOUN.ssl
+++ b/Games/Fallout2.Steam/ReDefine/NCANOUN.ssl
@@ -151,53 +151,53 @@ end
 procedure ChooseFromList
 begin
     variable LVar0 := 0;
-    gsay_reply(668, stored_msg);
-    if (message_str(353, top_msg - 1) != "") then begin
-        giq_option(1, 668, message_str(353, 150), CLICK_PREV, 50);
+    gsay_reply(SCRIPT_NTLYDROM, stored_msg);
+    if (message_str(SCRIPT_NEWR1, top_msg - 1) != "") then begin
+        giq_option(1, SCRIPT_NTLYDROM, message_str(SCRIPT_NEWR1, 150), CLICK_PREV, NEUTRAL_REACTION);
     end
     if (LVar0 == 0) then begin
-        if (message_str(353, top_msg + 0) != "") then begin
-            giq_option(1, 668, message_str(353, top_msg + 0), CLICK_0, 50);
+        if (message_str(SCRIPT_NEWR1, top_msg + 0) != "") then begin
+            giq_option(1, SCRIPT_NTLYDROM, message_str(SCRIPT_NEWR1, top_msg + 0), CLICK_0, NEUTRAL_REACTION);
         end
         else begin
             LVar0 := 1;
         end
     end
     if (LVar0 == 0) then begin
-        if (message_str(353, top_msg + 1) != "") then begin
-            giq_option(1, 668, message_str(353, top_msg + 1), CLICK_1, 50);
+        if (message_str(SCRIPT_NEWR1, top_msg + 1) != "") then begin
+            giq_option(1, SCRIPT_NTLYDROM, message_str(SCRIPT_NEWR1, top_msg + 1), CLICK_1, NEUTRAL_REACTION);
         end
         else begin
             LVar0 := 1;
         end
     end
     if (LVar0 == 0) then begin
-        if (message_str(353, top_msg + 2) != "") then begin
-            giq_option(1, 668, message_str(353, top_msg + 2), CLICK_2, 50);
+        if (message_str(SCRIPT_NEWR1, top_msg + 2) != "") then begin
+            giq_option(1, SCRIPT_NTLYDROM, message_str(SCRIPT_NEWR1, top_msg + 2), CLICK_2, NEUTRAL_REACTION);
         end
         else begin
             LVar0 := 1;
         end
     end
     if (LVar0 == 0) then begin
-        if (message_str(353, top_msg + 3) != "") then begin
-            giq_option(1, 668, message_str(353, top_msg + 3), CLICK_3, 50);
+        if (message_str(SCRIPT_NEWR1, top_msg + 3) != "") then begin
+            giq_option(1, SCRIPT_NTLYDROM, message_str(SCRIPT_NEWR1, top_msg + 3), CLICK_3, NEUTRAL_REACTION);
         end
         else begin
             LVar0 := 1;
         end
     end
     if (LVar0 == 0) then begin
-        if (message_str(353, top_msg + 4) != "") then begin
-            giq_option(1, 668, message_str(353, top_msg + 4), CLICK_4, 50);
+        if (message_str(SCRIPT_NEWR1, top_msg + 4) != "") then begin
+            giq_option(1, SCRIPT_NTLYDROM, message_str(SCRIPT_NEWR1, top_msg + 4), CLICK_4, NEUTRAL_REACTION);
         end
         else begin
             LVar0 := 1;
         end
     end
     if (LVar0 == 0) then begin
-        if (message_str(353, top_msg + 5) != "") then begin
-            giq_option(1, 668, message_str(353, 151), CLICK_MORE, 50);
+        if (message_str(SCRIPT_NEWR1, top_msg + 5) != "") then begin
+            giq_option(1, SCRIPT_NTLYDROM, message_str(SCRIPT_NEWR1, 151), CLICK_MORE, NEUTRAL_REACTION);
         end
     end
 end
@@ -252,14 +252,13 @@ procedure spatial_p_proc
 begin
     if (local_var(0) == 0) then begin
         if (source_obj == dude_obj) then begin
-            if ((global_var(314) bwand 16777216) != 0) then begin
-                if (tile_in_tile_rect(12899, 12895, 14499, 14495, tile_num(dude_obj)) and (cur_map_index == 54) and ((elevation(dude_obj) == 2) or (-1 == 2))) then begin
+            if ((global_var(GVAR_NEW_RENO_FLAG_1) bwand 16777216) != 0) then begin
+                if (tile_in_tile_rect(12899, 12895, 14499, 14495, dude_tile) and (CUR_MAP_NEWR1) and ((dude_elevation == 2) or (-1 == 2))) then begin
                     set_local_var(0, 1);
-                    display_msg(message_str(668, 100));
-                    float_msg(dude_obj, message_str(668, 200), 8);
+                    display_msg(message_str(SCRIPT_NTLYDROM, 100));
+                    float_msg(dude_obj, message_str(SCRIPT_NTLYDROM, 200), FLOAT_MSG_YELLOW);
                 end
             end
         end
     end
 end
-
