diff --git a/Games/Fallout2.Steam/Original/NCELDRID.ssl b/Games/Fallout2.Steam/ReDefine/NCELDRID.ssl
index 0499e7f..1b75bc0 100644
--- a/Games/Fallout2.Steam/Original/NCELDRID.ssl
+++ b/Games/Fallout2.Steam/ReDefine/NCELDRID.ssl
@@ -229,53 +229,53 @@ end
 procedure ChooseFromList
 begin
     variable LVar0 := 0;
-    gsay_reply(426, stored_msg);
-    if (message_str(353, top_msg - 1) != "") then begin
-        giq_option(1, 426, message_str(353, 150), CLICK_PREV, 50);
+    Reply(stored_msg);
+    if (message_str(SCRIPT_NEWR1, top_msg - 1) != "") then begin
+        giq_option(1, NAME, message_str(SCRIPT_NEWR1, 150), CLICK_PREV, NEUTRAL_REACTION);
     end
     if (LVar0 == 0) then begin
-        if (message_str(353, top_msg + 0) != "") then begin
-            giq_option(1, 426, message_str(353, top_msg + 0), CLICK_0, 50);
+        if (message_str(SCRIPT_NEWR1, top_msg + 0) != "") then begin
+            giq_option(1, NAME, message_str(SCRIPT_NEWR1, top_msg + 0), CLICK_0, NEUTRAL_REACTION);
         end
         else begin
             LVar0 := 1;
         end
     end
     if (LVar0 == 0) then begin
-        if (message_str(353, top_msg + 1) != "") then begin
-            giq_option(1, 426, message_str(353, top_msg + 1), CLICK_1, 50);
+        if (message_str(SCRIPT_NEWR1, top_msg + 1) != "") then begin
+            giq_option(1, NAME, message_str(SCRIPT_NEWR1, top_msg + 1), CLICK_1, NEUTRAL_REACTION);
         end
         else begin
             LVar0 := 1;
         end
     end
     if (LVar0 == 0) then begin
-        if (message_str(353, top_msg + 2) != "") then begin
-            giq_option(1, 426, message_str(353, top_msg + 2), CLICK_2, 50);
+        if (message_str(SCRIPT_NEWR1, top_msg + 2) != "") then begin
+            giq_option(1, NAME, message_str(SCRIPT_NEWR1, top_msg + 2), CLICK_2, NEUTRAL_REACTION);
         end
         else begin
             LVar0 := 1;
         end
     end
     if (LVar0 == 0) then begin
-        if (message_str(353, top_msg + 3) != "") then begin
-            giq_option(1, 426, message_str(353, top_msg + 3), CLICK_3, 50);
+        if (message_str(SCRIPT_NEWR1, top_msg + 3) != "") then begin
+            giq_option(1, NAME, message_str(SCRIPT_NEWR1, top_msg + 3), CLICK_3, NEUTRAL_REACTION);
         end
         else begin
             LVar0 := 1;
         end
     end
     if (LVar0 == 0) then begin
-        if (message_str(353, top_msg + 4) != "") then begin
-            giq_option(1, 426, message_str(353, top_msg + 4), CLICK_4, 50);
+        if (message_str(SCRIPT_NEWR1, top_msg + 4) != "") then begin
+            giq_option(1, NAME, message_str(SCRIPT_NEWR1, top_msg + 4), CLICK_4, NEUTRAL_REACTION);
         end
         else begin
             LVar0 := 1;
         end
     end
     if (LVar0 == 0) then begin
-        if (message_str(353, top_msg + 5) != "") then begin
-            giq_option(1, 426, message_str(353, 151), CLICK_MORE, 50);
+        if (message_str(SCRIPT_NEWR1, top_msg + 5) != "") then begin
+            giq_option(1, NAME, message_str(SCRIPT_NEWR1, 151), CLICK_MORE, NEUTRAL_REACTION);
         end
     end
 end
@@ -331,17 +331,17 @@ procedure critter_p_proc
 begin
     if (obj_can_see_obj(self_obj, dude_obj)) then begin
         if (anim_busy(self_obj) == 0) then begin
-            anim(self_obj, 1000, rotation_to_tile(tile_num(self_obj), tile_num(dude_obj)));
+            anim(self_obj, 1000, rotation_to_tile(tile_num(self_obj), dude_tile));
         end
         if ((local_var(4) bwand 2) != 0) then begin
             if (critter_is_fleeing(self_obj)) then begin
                 if (anim_busy(self_obj) == 0) then begin
                     if ((tile_distance_objs(self_obj, dude_obj) < 8) or obj_can_see_obj(self_obj, dude_obj)) then begin
                         if (anim_busy(self_obj) == 0) then begin
-                            global_temp := rotation_to_tile(tile_num(dude_obj), tile_num(self_obj));
+                            global_temp := rotation_to_tile(dude_tile, tile_num(self_obj));
                             animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(self_obj), global_temp, random(3, 10)), 1);
                             global_temp := (global_temp + 1) % 6;
-                            while ((anim_busy(self_obj) == 0) and (global_temp != rotation_to_tile(tile_num(dude_obj), tile_num(self_obj)))) do begin
+                            while ((anim_busy(self_obj) == 0) and (global_temp != rotation_to_tile(dude_tile, tile_num(self_obj)))) do begin
                                 debug_msg("flee loop: rot == " + global_temp);
                                 animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(self_obj), global_temp, random(3, 10)), 1);
                                 global_temp := (global_temp + 1) % 6;
@@ -351,32 +351,32 @@ begin
                 end
             end
             else begin
-                attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+                attack(dude_obj);
             end
         end
         else begin
-            if (tile_in_tile_rect(22334, 22306, 29534, 29506, tile_num(dude_obj)) and (cur_map_index == 56) and ((elevation(dude_obj) == 0) or (-1 == 0))) then begin
-                if ((game_time_hour <= 600) or (game_time_hour >= 1800) and ((global_var(495) bwand 134217728) != 0)) then begin
+            if (tile_in_tile_rect(22334, 22306, 29534, 29506, dude_tile) and (CUR_MAP_NEWR3) and ((dude_elevation == 0) or (-1 == 0))) then begin
+                if ((game_time_hour <= 600) or (game_time_hour >= 1800) and ((global_var(GVAR_NEW_RENO_FLAG_3) bwand 134217728) != 0)) then begin
                     if (((local_var(4) bwand 65536) != 0) == 0) then begin
                         set_local_var(4, local_var(4) bwor 65536);
                         call Node030();
-                        set_global_var(219, game_time + 60);
+                        set_global_var(GVAR_NEW_RENO_WARNING_TIMER, game_time + 60);
                     end
                     else begin
-                        if (global_var(219) <= game_time) then begin
+                        if (global_var(GVAR_NEW_RENO_WARNING_TIMER) <= game_time) then begin
                             call Node998();
                         end
                     end
                 end
                 else begin
-                    if ((tile_in_tile_rect(26125, 26115, 29725, 29715, tile_num(dude_obj)) and (cur_map_index == 56) and ((elevation(dude_obj) == 0) or (-1 == 0))) == 0) then begin
+                    if ((tile_in_tile_rect(26125, 26115, 29725, 29715, dude_tile) and (CUR_MAP_NEWR3) and ((dude_elevation == 0) or (-1 == 0))) == 0) then begin
                         if (((local_var(4) bwand 32768) != 0) == 0) then begin
                             set_local_var(4, local_var(4) bwor 32768);
                             call Node029();
-                            set_global_var(219, game_time + 60);
+                            set_global_var(GVAR_NEW_RENO_WARNING_TIMER, game_time + 60);
                         end
                         else begin
-                            if (global_var(219) <= game_time) then begin
+                            if (global_var(GVAR_NEW_RENO_WARNING_TIMER) <= game_time) then begin
                                 call Node998();
                             end
                         end
@@ -388,7 +388,7 @@ begin
     else begin
         if (anim_busy(self_obj) == 0) then begin
             if ((game_time_hour >= 600) and (game_time_hour <= 2100)) then begin
-                set_global_var(495, global_var(495) bwand (-1 - 134217728));
+                set_global_var(GVAR_NEW_RENO_FLAG_3, global_var(GVAR_NEW_RENO_FLAG_3) bwand (-1 - 134217728));
                 if (tile_num(self_obj) != 26719) then begin
                     if ((tile_num(self_obj) != 26719) and (26719 != 0)) then begin
                         if ((dest_tile == 26719) and (step_tile < 0)) then begin
@@ -404,12 +404,12 @@ begin
                                     animate_move_obj_to_tile(self_obj, step_tile, 0);
                                 end
                                 if (anim_busy(self_obj) == 0) then begin
-                                    if (metarule3(106, tile_num_in_direction(tile_num(self_obj), has_trait(1, self_obj, 10), 1), elevation(self_obj), 0) != 0) then begin
+                                    if (tile_get_next_critter(tile_num_in_direction(tile_num(self_obj), obj_get_rot(self_obj), 1), elevation(self_obj), 0) != 0) then begin
                                         if (anim_busy(self_obj) == 0) then begin
-                                            global_temp := rotation_to_tile(tile_num_in_direction(tile_num(self_obj), rotation_to_tile(tile_num_in_direction(tile_num(self_obj), has_trait(1, self_obj, 10), 1), tile_num(self_obj)), 1), tile_num(self_obj));
+                                            global_temp := rotation_to_tile(tile_num_in_direction(tile_num(self_obj), rotation_to_tile(tile_num_in_direction(tile_num(self_obj), obj_get_rot(self_obj), 1), tile_num(self_obj)), 1), tile_num(self_obj));
                                             animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(self_obj), global_temp, random(2, 5)), 0);
                                             global_temp := (global_temp + 1) % 6;
-                                            while ((anim_busy(self_obj) == 0) and (global_temp != rotation_to_tile(tile_num_in_direction(tile_num(self_obj), rotation_to_tile(tile_num_in_direction(tile_num(self_obj), has_trait(1, self_obj, 10), 1), tile_num(self_obj)), 1), tile_num(self_obj)))) do begin
+                                            while ((anim_busy(self_obj) == 0) and (global_temp != rotation_to_tile(tile_num_in_direction(tile_num(self_obj), rotation_to_tile(tile_num_in_direction(tile_num(self_obj), obj_get_rot(self_obj), 1), tile_num(self_obj)), 1), tile_num(self_obj)))) do begin
                                                 debug_msg("flee loop: rot == " + global_temp);
                                                 animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(self_obj), global_temp, random(2, 5)), 0);
                                                 global_temp := (global_temp + 1) % 6;
@@ -447,12 +447,12 @@ begin
                                         animate_move_obj_to_tile(self_obj, step_tile, 0);
                                     end
                                     if (anim_busy(self_obj) == 0) then begin
-                                        if (metarule3(106, tile_num_in_direction(tile_num(self_obj), has_trait(1, self_obj, 10), 1), elevation(self_obj), 0) != 0) then begin
+                                        if (tile_get_next_critter(tile_num_in_direction(tile_num(self_obj), obj_get_rot(self_obj), 1), elevation(self_obj), 0) != 0) then begin
                                             if (anim_busy(self_obj) == 0) then begin
-                                                global_temp := rotation_to_tile(tile_num_in_direction(tile_num(self_obj), rotation_to_tile(tile_num_in_direction(tile_num(self_obj), has_trait(1, self_obj, 10), 1), tile_num(self_obj)), 1), tile_num(self_obj));
+                                                global_temp := rotation_to_tile(tile_num_in_direction(tile_num(self_obj), rotation_to_tile(tile_num_in_direction(tile_num(self_obj), obj_get_rot(self_obj), 1), tile_num(self_obj)), 1), tile_num(self_obj));
                                                 animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(self_obj), global_temp, random(2, 5)), 0);
                                                 global_temp := (global_temp + 1) % 6;
-                                                while ((anim_busy(self_obj) == 0) and (global_temp != rotation_to_tile(tile_num_in_direction(tile_num(self_obj), rotation_to_tile(tile_num_in_direction(tile_num(self_obj), has_trait(1, self_obj, 10), 1), tile_num(self_obj)), 1), tile_num(self_obj)))) do begin
+                                                while ((anim_busy(self_obj) == 0) and (global_temp != rotation_to_tile(tile_num_in_direction(tile_num(self_obj), rotation_to_tile(tile_num_in_direction(tile_num(self_obj), obj_get_rot(self_obj), 1), tile_num(self_obj)), 1), tile_num(self_obj)))) do begin
                                                     debug_msg("flee loop: rot == " + global_temp);
                                                     animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(self_obj), global_temp, random(2, 5)), 0);
                                                     global_temp := (global_temp + 1) % 6;
@@ -472,18 +472,18 @@ begin
                             end
                         end
                         else begin
-                            float_msg(self_obj, message_str(426, 700), 8);
+                            float_msg(self_obj, mstr(700), FLOAT_MSG_YELLOW);
                             last_timed_obj := self_obj;
                             last_timed_param := last_timed_param;
-                            metarule3(100, self_obj, 1, 0);
+                            rm_fixed_timer_event(self_obj, 1);
                             add_timer_event(self_obj, game_ticks(25) + random(0, game_ticks(35 - 25)), 1);
                             set_local_var(6, 0);
                         end
                     end
                 end
                 else begin
-                    if ((tile_in_tile_rect(22519, 22507, 24319, 24307, tile_num(self_obj)) and (cur_map_index == 56) and ((elevation(self_obj) == 0) or (-1 == 0))) == 0) then begin
-                        set_global_var(495, global_var(495) bwor 134217728);
+                    if ((tile_in_tile_rect(22519, 22507, 24319, 24307, tile_num(self_obj)) and (CUR_MAP_NEWR3) and ((elevation(self_obj) == 0) or (-1 == 0))) == 0) then begin
+                        set_global_var(GVAR_NEW_RENO_FLAG_3, global_var(GVAR_NEW_RENO_FLAG_3) bwor 134217728);
                         if ((tile_num(self_obj) != 23110) and (23110 != 0)) then begin
                             if ((dest_tile == 23110) and (step_tile < 0)) then begin
                                 step_tile := step_tile + 1;
@@ -498,12 +498,12 @@ begin
                                         animate_move_obj_to_tile(self_obj, step_tile, 0);
                                     end
                                     if (anim_busy(self_obj) == 0) then begin
-                                        if (metarule3(106, tile_num_in_direction(tile_num(self_obj), has_trait(1, self_obj, 10), 1), elevation(self_obj), 0) != 0) then begin
+                                        if (tile_get_next_critter(tile_num_in_direction(tile_num(self_obj), obj_get_rot(self_obj), 1), elevation(self_obj), 0) != 0) then begin
                                             if (anim_busy(self_obj) == 0) then begin
-                                                global_temp := rotation_to_tile(tile_num_in_direction(tile_num(self_obj), rotation_to_tile(tile_num_in_direction(tile_num(self_obj), has_trait(1, self_obj, 10), 1), tile_num(self_obj)), 1), tile_num(self_obj));
+                                                global_temp := rotation_to_tile(tile_num_in_direction(tile_num(self_obj), rotation_to_tile(tile_num_in_direction(tile_num(self_obj), obj_get_rot(self_obj), 1), tile_num(self_obj)), 1), tile_num(self_obj));
                                                 animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(self_obj), global_temp, random(2, 5)), 0);
                                                 global_temp := (global_temp + 1) % 6;
-                                                while ((anim_busy(self_obj) == 0) and (global_temp != rotation_to_tile(tile_num_in_direction(tile_num(self_obj), rotation_to_tile(tile_num_in_direction(tile_num(self_obj), has_trait(1, self_obj, 10), 1), tile_num(self_obj)), 1), tile_num(self_obj)))) do begin
+                                                while ((anim_busy(self_obj) == 0) and (global_temp != rotation_to_tile(tile_num_in_direction(tile_num(self_obj), rotation_to_tile(tile_num_in_direction(tile_num(self_obj), obj_get_rot(self_obj), 1), tile_num(self_obj)), 1), tile_num(self_obj)))) do begin
                                                     debug_msg("flee loop: rot == " + global_temp);
                                                     animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(self_obj), global_temp, random(2, 5)), 0);
                                                     global_temp := (global_temp + 1) % 6;
@@ -538,7 +538,7 @@ end
 
 procedure talk_p_proc
 begin
-    if ((global_var(495) bwand 134217728) != 0) then begin
+    if ((global_var(GVAR_NEW_RENO_FLAG_3) bwand 134217728) != 0) then begin
         call Node030();
     end
     else begin
@@ -561,7 +561,7 @@ begin
         start_gdialog(426, self_obj, 4, -1, -1);
         gsay_start;
         if ((local_var(4) bwand 4096) != 0) then begin
-            if (global_var(494) == 8) then begin
+            if (global_var(GVAR_FALLOUT_2) == 8) then begin
                 call Node005();
             end
             else begin
@@ -592,7 +592,7 @@ begin
     end
     last_timed_obj := self_obj;
     last_timed_param := last_timed_param;
-    metarule3(100, self_obj, 1, 0);
+    rm_fixed_timer_event(self_obj, 1);
     add_timer_event(self_obj, game_ticks(25) + random(0, game_ticks(35 - 25)), 1);
 end
 
@@ -603,22 +603,22 @@ begin
     move_obj_inven_to_obj(new_reno_eldridge_box_2, self_obj);
     if (((local_var(4) bwand 512) != 0) == 0) then begin
         set_local_var(4, local_var(4) bwor 512);
-        restock_obj := create_object_sid(356, 0, 0, -1);
+        restock_obj := create_object(356, 0, 0);
         add_mult_objs_to_inven(self_obj, restock_obj, 1);
     end
-    set_global_var(345, global_var(345) bwor 4194304);
+    set_global_var(GVAR_NEW_RENO_FLAG_2, global_var(GVAR_NEW_RENO_FLAG_2) bwor 4194304);
 end
 
 procedure look_at_p_proc
 begin
     script_overrides;
-    display_msg(message_str(426, 100));
+    display_mstr(100);
 end
 
 procedure description_p_proc
 begin
     script_overrides;
-    display_msg(message_str(426, 150 + (((local_var(4) bwand 1) != 0) * ((local_var(4) bwand 4) != 0))));
+    display_mstr(150 + (((local_var(4) bwand 1) != 0) * ((local_var(4) bwand 4) != 0)));
     set_local_var(4, local_var(4) bwor 4);
 end
 
@@ -637,13 +637,13 @@ end
 procedure map_enter_p_proc
 begin
     new_reno_eldridge_obj := self_obj;
-    if (metarule(22, 0) == 0) then begin
-        if (metarule(14, 0)) then begin
-            critter_add_trait(self_obj, 1, 6, 157);
-            critter_add_trait(self_obj, 1, 5, 22);
+    if (is_loading_game == 0) then begin
+        if (map_first_run) then begin
+            critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 157);
+            critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_TOUGH_GUARD);
             last_timed_obj := self_obj;
             last_timed_param := last_timed_param;
-            metarule3(100, self_obj, 1, 0);
+            rm_fixed_timer_event(self_obj, 1);
             add_timer_event(self_obj, game_ticks(25) + random(0, game_ticks(35 - 25)), 1);
         end
     end
@@ -676,13 +676,13 @@ begin
         else begin
             if ((game_time_hour >= 600) and (game_time_hour <= 2100)) then begin
                 set_local_var(6, 0);
-                if ((combat_is_initialized == 0) and (global_var(219) <= game_time)) then begin
+                if ((combat_is_initialized == 0) and (global_var(GVAR_NEW_RENO_WARNING_TIMER) <= game_time)) then begin
                     if (obj_can_see_obj(self_obj, dude_obj)) then begin
-                        if (global_var(494) == 8) then begin
+                        if (global_var(GVAR_FALLOUT_2) == 8) then begin
                             call Node001();
                         end
                         else begin
-                            if ((obj_pid(critter_inven_obj(dude_obj, 0)) == 3) or (obj_pid(critter_inven_obj(dude_obj, 0)) == 348) or (obj_pid(critter_inven_obj(dude_obj, 0)) == 349) or (obj_pid(critter_inven_obj(dude_obj, 0)) == 232)) then begin
+                            if ((obj_pid(critter_inven_obj(dude_obj, INVEN_TYPE_WORN)) == 3) or (obj_pid(critter_inven_obj(dude_obj, INVEN_TYPE_WORN)) == 348) or (obj_pid(critter_inven_obj(dude_obj, INVEN_TYPE_WORN)) == 349) or (obj_pid(critter_inven_obj(dude_obj, INVEN_TYPE_WORN)) == 232)) then begin
                                 call Node002();
                             end
                             else begin
@@ -696,7 +696,7 @@ begin
     end
     last_timed_obj := self_obj;
     last_timed_param := last_timed_param;
-    metarule3(100, self_obj, 1, 0);
+    rm_fixed_timer_event(self_obj, 1);
     add_timer_event(self_obj, game_ticks(25) + random(0, game_ticks(35 - 25)), 1);
 end
 
@@ -713,76 +713,76 @@ end
 
 procedure Node001
 begin
-    float_msg(self_obj, message_str(426, random(200, 204)), 8);
+    float_msg(self_obj, mstr(random(200, 204)), FLOAT_MSG_YELLOW);
 end
 
 procedure Node002
 begin
-    float_msg(self_obj, message_str(426, random(210, 214)), 8);
+    float_msg(self_obj, mstr(random(210, 214)), FLOAT_MSG_YELLOW);
 end
 
 procedure Node003
 begin
-    float_msg(self_obj, message_str(426, random(220, 234)), 8);
+    float_msg(self_obj, mstr(random(220, 234)), FLOAT_MSG_YELLOW);
 end
 
 procedure Node004
 begin
-    if (global_var(494) == 8) then begin
-        gsay_reply(426, 240);
+    if (global_var(GVAR_FALLOUT_2) == 8) then begin
+        Reply(240);
     end
     else begin
-        gsay_reply(426, 241 + (get_critter_stat(dude_obj, 34) == 1));
+        Reply(241 + (dude_gender == 1));
     end
     set_local_var(4, local_var(4) bwor 4096);
-    giq_option(1, 426, 243 + ((get_critter_stat(dude_obj, 4) <= 3) == 0), Node990, 50);
+    giq_option(1, NAME, 243 + ((dude_is_stupid) == 0), Node990, NEUTRAL_REACTION);
 end
 
 procedure Node005
 begin
     return_node := "Node005";
     if (prev_node == 0) then begin
-        gsay_reply(426, message_str(426, 250) + message_str(426, 251));
+        Reply(mstr(250) + mstr(251));
     end
     else begin
-        gsay_reply(426, 251);
+        Reply(251);
     end
     set_local_var(5, local_var(5) + 200);
     set_local_var(5, local_var(5) - (local_var(5) % 200));
     gdialog_set_barter_mod(local_var(5));
-    giq_option(-3, 426, 252, Node007, 50);
-    giq_option(-3, 426, 253, Node999, 50);
-    giq_option(4, 426, 254, Node974, 50);
+    giq_option(-3, NAME, 252, Node007, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 253, Node999, NEUTRAL_REACTION);
+    giq_option(4, NAME, 254, Node974, NEUTRAL_REACTION);
     if (((local_var(4) bwand 512) != 0) == 0) then begin
-        giq_option(4, 426, 255, Node018, 50);
+        giq_option(4, NAME, 255, Node018, NEUTRAL_REACTION);
     end
-    if (global_var(230) > 0) then begin
+    if (global_var(GVAR_NEW_RENO_MADE_MAN) > 0) then begin
         if ((local_var(4) bwand 1024) != 0) then begin
-            giq_option(4, 426, 262, Node975, 50);
+            giq_option(4, NAME, 262, Node975, NEUTRAL_REACTION);
         end
         else begin
-            giq_option(4, 426, message_str(426, 256) + message_str(353, 200 + (global_var(230) / 100)) + message_str(426, 5256), Node026, 50);
+            giq_option(4, NAME, mstr(256) + message_str(SCRIPT_NEWR1, 200 + (global_var(GVAR_NEW_RENO_MADE_MAN) / 100)) + mstr(5256), Node026, NEUTRAL_REACTION);
         end
     end
     else begin
-        if (global_var(349) == 1) then begin
-            giq_option(4, 426, 257, Node989, 50);
+        if (global_var(GVAR_NEW_RENO_JULES_ELDRIDGE) == 1) then begin
+            giq_option(4, NAME, 257, Node989, NEUTRAL_REACTION);
         end
     end
-    if (global_var(549) == 0) then begin
-        if (((global_var(314) bwand 16384) != 0) and ((global_var(230) / 100) != 1)) then begin
-            giq_option(4, 426, 258, Node023, 50);
+    if (global_var(GVAR_NEW_RENO_ELDRIDGE_PISTOL_QUEST) == 0) then begin
+        if (((global_var(GVAR_NEW_RENO_FLAG_1) bwand 16384) != 0) and ((global_var(GVAR_NEW_RENO_MADE_MAN) / 100) != 1)) then begin
+            giq_option(4, NAME, 258, Node023, NEUTRAL_REACTION);
         end
     end
     else begin
-        if (global_var(549) == 1) then begin
+        if (global_var(GVAR_NEW_RENO_ELDRIDGE_PISTOL_QUEST) == 1) then begin
             if (obj_is_carrying_obj_pid(dude_obj, 16)) then begin
-                giq_option(4, 426, 259, Node025, 50);
+                giq_option(4, NAME, 259, Node025, NEUTRAL_REACTION);
             end
         end
     end
-    giq_option(4, 426, 260, Node998, 50);
-    giq_option(4, 426, 261, Node999, 51);
+    giq_option(4, NAME, 260, Node998, NEUTRAL_REACTION);
+    giq_option(4, NAME, 261, Node999, BAD_REACTION);
     prev_node := 5;
 end
 
@@ -790,103 +790,103 @@ procedure Node006
 begin
     return_node := "Node006";
     if (((local_var(4) bwand 8192) != 0) == 0) then begin
-        gsay_reply(426, 270);
+        Reply(270);
     end
     else begin
-        gsay_reply(426, 272 + (prev_node != 0));
+        Reply(272 + (prev_node != 0));
     end
-    giq_option(-3, 426, 274, Node007, 50);
-    giq_option(-3, 426, 253, Node999, 50);
-    giq_option(4, 426, 276 + ((local_var(4) bwand 8192) != 0), Node974, 50);
+    giq_option(-3, NAME, 274, Node007, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 253, Node999, NEUTRAL_REACTION);
+    giq_option(4, NAME, 276 + ((local_var(4) bwand 8192) != 0), Node974, NEUTRAL_REACTION);
     if (((local_var(4) bwand 512) != 0) == 0) then begin
-        giq_option(4, 426, 255, Node018, 50);
+        giq_option(4, NAME, 255, Node018, NEUTRAL_REACTION);
     end
-    if (global_var(230) > 0) then begin
+    if (global_var(GVAR_NEW_RENO_MADE_MAN) > 0) then begin
         if ((local_var(4) bwand 1024) != 0) then begin
-            giq_option(4, 426, 286, Node975, 50);
+            giq_option(4, NAME, 286, Node975, NEUTRAL_REACTION);
         end
         else begin
-            giq_option(4, 426, message_str(426, 256) + message_str(353, 200 + (global_var(230) / 100)) + message_str(426, 5256), Node026, 50);
+            giq_option(4, NAME, mstr(256) + message_str(SCRIPT_NEWR1, 200 + (global_var(GVAR_NEW_RENO_MADE_MAN) / 100)) + mstr(5256), Node026, NEUTRAL_REACTION);
         end
     end
     else begin
-        if (global_var(349) == 1) then begin
-            giq_option(4, 426, 257, Node989, 50);
+        if (global_var(GVAR_NEW_RENO_JULES_ELDRIDGE) == 1) then begin
+            giq_option(4, NAME, 257, Node989, NEUTRAL_REACTION);
         end
     end
-    if (global_var(549) == 0) then begin
-        if (((global_var(314) bwand 16384) != 0) and ((global_var(230) / 100) != 1)) then begin
-            giq_option(4, 426, 258, Node023, 50);
+    if (global_var(GVAR_NEW_RENO_ELDRIDGE_PISTOL_QUEST) == 0) then begin
+        if (((global_var(GVAR_NEW_RENO_FLAG_1) bwand 16384) != 0) and ((global_var(GVAR_NEW_RENO_MADE_MAN) / 100) != 1)) then begin
+            giq_option(4, NAME, 258, Node023, NEUTRAL_REACTION);
         end
     end
     else begin
-        if (global_var(549) == 1) then begin
+        if (global_var(GVAR_NEW_RENO_ELDRIDGE_PISTOL_QUEST) == 1) then begin
             if (obj_is_carrying_obj_pid(dude_obj, 16)) then begin
-                giq_option(4, 426, 259, Node025, 50);
+                giq_option(4, NAME, 259, Node025, NEUTRAL_REACTION);
             end
         end
     end
-    giq_option(4, 426, 283, Node988, 50);
-    giq_option(4, 426, 284 + ((local_var(4) bwand 8192) != 0), Node999, 51);
+    giq_option(4, NAME, 283, Node988, NEUTRAL_REACTION);
+    giq_option(4, NAME, 284 + ((local_var(4) bwand 8192) != 0), Node999, BAD_REACTION);
     set_local_var(4, local_var(4) bwor 8192);
 end
 
 procedure Node007
 begin
     return_node := "Node007";
-    gsay_reply(426, 300 + ((global_var(494) == 8) == 0));
-    giq_option(-3, 426, 302, Node974, 50);
+    Reply(300 + ((global_var(GVAR_FALLOUT_2) == 8) == 0));
+    giq_option(-3, NAME, 302, Node974, NEUTRAL_REACTION);
     if (((local_var(4) bwand 512) != 0) == 0) then begin
-        giq_option(-3, 426, 303, Node018, 50);
+        giq_option(-3, NAME, 303, Node018, NEUTRAL_REACTION);
     end
-    giq_option(-3, 426, 304, Node999, 50);
+    giq_option(-3, NAME, 304, Node999, NEUTRAL_REACTION);
     prev_node := 7;
 end
 
 procedure Node008
 begin
     return_node := "Node008";
-    if (global_var(349) == 1) then begin
-        set_global_var(349, 2);
+    if (global_var(GVAR_NEW_RENO_JULES_ELDRIDGE) == 1) then begin
+        set_global_var(GVAR_NEW_RENO_JULES_ELDRIDGE, 2);
         set_local_var(5, local_var(5) + 10);
         set_local_var(5, local_var(5) - (local_var(5) % 200));
         gdialog_set_barter_mod(local_var(5));
     end
-    gsay_reply(426, message_str(426, 310) + message_str(426, 311 + (get_critter_stat(dude_obj, 34) == 1)));
-    giq_option(4, 426, 277, Node974, 50);
+    Reply(mstr(310) + mstr(311 + (dude_gender == 1)));
+    giq_option(4, NAME, 277, Node974, NEUTRAL_REACTION);
     if (((local_var(4) bwand 512) != 0) == 0) then begin
-        giq_option(4, 426, 255, Node018, 50);
+        giq_option(4, NAME, 255, Node018, NEUTRAL_REACTION);
     end
-    if (global_var(230) > 0) then begin
+    if (global_var(GVAR_NEW_RENO_MADE_MAN) > 0) then begin
         if ((local_var(4) bwand 1024) != 0) then begin
-            giq_option(4, 426, 286, Node975, 50);
+            giq_option(4, NAME, 286, Node975, NEUTRAL_REACTION);
         end
         else begin
-            giq_option(4, 426, message_str(426, 256) + message_str(353, 200 + (global_var(230) / 100)) + message_str(426, 5256), Node026, 50);
+            giq_option(4, NAME, mstr(256) + message_str(SCRIPT_NEWR1, 200 + (global_var(GVAR_NEW_RENO_MADE_MAN) / 100)) + mstr(5256), Node026, NEUTRAL_REACTION);
         end
     end
-    if (global_var(549) == 0) then begin
-        if (((global_var(314) bwand 16384) != 0) and ((global_var(230) / 100) != 1)) then begin
-            giq_option(4, 426, 258, Node023, 50);
+    if (global_var(GVAR_NEW_RENO_ELDRIDGE_PISTOL_QUEST) == 0) then begin
+        if (((global_var(GVAR_NEW_RENO_FLAG_1) bwand 16384) != 0) and ((global_var(GVAR_NEW_RENO_MADE_MAN) / 100) != 1)) then begin
+            giq_option(4, NAME, 258, Node023, NEUTRAL_REACTION);
         end
     end
     else begin
-        if (global_var(549) == 1) then begin
+        if (global_var(GVAR_NEW_RENO_ELDRIDGE_PISTOL_QUEST) == 1) then begin
             if (obj_is_carrying_obj_pid(dude_obj, 16)) then begin
-                giq_option(4, 426, 259, Node025, 50);
+                giq_option(4, NAME, 259, Node025, NEUTRAL_REACTION);
             end
         end
     end
-    giq_option(4, 426, 283, Node988, 50);
-    giq_option(4, 426, 285, Node999, 51);
+    giq_option(4, NAME, 283, Node988, NEUTRAL_REACTION);
+    giq_option(4, NAME, 285, Node999, BAD_REACTION);
     prev_node := 8;
 end
 
 procedure Node009
 begin
-    gsay_reply(426, 330);
-    giq_option(4, 426, 331, Node006, 50);
-    giq_option(4, 426, 285, Node999, 50);
+    Reply(330);
+    giq_option(4, NAME, 331, Node006, NEUTRAL_REACTION);
+    giq_option(4, NAME, 285, Node999, NEUTRAL_REACTION);
     prev_node := 9;
 end
 
@@ -905,23 +905,23 @@ begin
         LVar2 := 1;
     end
     if (LVar0 or LVar1 or LVar2) then begin
-        gsay_reply(426, 340);
+        Reply(340);
         if (LVar0) then begin
-            giq_option(4, 426, 341, Node011, 50);
+            giq_option(4, NAME, 341, Node011, NEUTRAL_REACTION);
         end
         if (LVar1) then begin
-            giq_option(4, 426, 342, Node012, 50);
+            giq_option(4, NAME, 342, Node012, NEUTRAL_REACTION);
         end
         if (LVar2) then begin
-            giq_option(4, 426, 343, Node013, 50);
+            giq_option(4, NAME, 343, Node013, NEUTRAL_REACTION);
         end
-        giq_option(4, 426, 344, Node006, 50);
-        giq_option(4, 426, 345, Node999, 50);
+        giq_option(4, NAME, 344, Node006, NEUTRAL_REACTION);
+        giq_option(4, NAME, 345, Node999, NEUTRAL_REACTION);
     end
     else begin
-        gsay_reply(426, 1340);
-        giq_option(4, 426, 1345, Node006, 50);
-        giq_option(4, 426, 1346, Node999, 50);
+        Reply(1340);
+        giq_option(4, NAME, 1345, Node006, NEUTRAL_REACTION);
+        giq_option(4, NAME, 1346, Node999, NEUTRAL_REACTION);
     end
     prev_node := 10;
 end
@@ -931,26 +931,26 @@ begin
     variable LVar0 := "";
     variable LVar1 := 0;
     if (obj_is_carrying_obj_pid(dude_obj, 313)) then begin
-        LVar0 := LVar0 + (message_str(426, 350) + (2000 * (250 / (160 + has_skill(dude_obj, 15)))) + message_str(426, 5500) + " ");
+        LVar0 := LVar0 + (mstr(350) + (2000 * (250 / (160 + dude_skill(SKILL_BARTER)))) + mstr(5500) + " ");
     end
     if (obj_is_carrying_obj_pid(dude_obj, 18)) then begin
-        LVar0 := LVar0 + (message_str(426, 351) + (3000 * (250 / (160 + has_skill(dude_obj, 15)))) + message_str(426, 5500));
+        LVar0 := LVar0 + (mstr(351) + (3000 * (250 / (160 + dude_skill(SKILL_BARTER)))) + mstr(5500));
     end
-    gsay_reply(426, LVar0);
-    if (obj_is_carrying_obj_pid(dude_obj, 313) and (item_caps_total(dude_obj) >= (2000 * (250 / (160 + has_skill(dude_obj, 15)))))) then begin
-        giq_option(4, 426, 352, Node987, 50);
+    Reply(LVar0);
+    if (obj_is_carrying_obj_pid(dude_obj, 313) and (item_caps_total(dude_obj) >= (2000 * (250 / (160 + dude_skill(SKILL_BARTER)))))) then begin
+        giq_option(4, NAME, 352, Node987, NEUTRAL_REACTION);
         LVar1 := 1;
     end
-    if (obj_is_carrying_obj_pid(dude_obj, 18) and (item_caps_total(dude_obj) >= (3000 * (250 / (160 + has_skill(dude_obj, 15)))))) then begin
-        giq_option(4, 426, 353, Node986, 50);
+    if (obj_is_carrying_obj_pid(dude_obj, 18) and (item_caps_total(dude_obj) >= (3000 * (250 / (160 + dude_skill(SKILL_BARTER)))))) then begin
+        giq_option(4, NAME, 353, Node986, NEUTRAL_REACTION);
         LVar1 := 1;
     end
     if (LVar1 == 0) then begin
-        giq_option(4, 426, 357, Node999, 50);
+        giq_option(4, NAME, 357, Node999, NEUTRAL_REACTION);
     end
-    giq_option(4, 426, 354, Node010, 50);
-    giq_option(4, 426, 355, Node006, 50);
-    giq_option(4, 426, 356, Node999, 50);
+    giq_option(4, NAME, 354, Node010, NEUTRAL_REACTION);
+    giq_option(4, NAME, 355, Node006, NEUTRAL_REACTION);
+    giq_option(4, NAME, 356, Node999, NEUTRAL_REACTION);
     prev_node := 11;
 end
 
@@ -958,57 +958,57 @@ procedure Node012
 begin
     variable LVar0 := 0;
     variable LVar1 := 0;
-    LVar0 := message_str(426, 365);
+    LVar0 := mstr(365);
     if (obj_is_carrying_obj_pid(dude_obj, 10)) then begin
-        LVar0 := LVar0 + (message_str(426, 366) + (4000 * (250 / (160 + has_skill(dude_obj, 15)))) + message_str(426, 5500) + " ");
+        LVar0 := LVar0 + (mstr(366) + (4000 * (250 / (160 + dude_skill(SKILL_BARTER)))) + mstr(5500) + " ");
     end
     if (obj_is_carrying_obj_pid(dude_obj, 23)) then begin
-        LVar0 := LVar0 + (message_str(426, 367) + (6000 * (250 / (160 + has_skill(dude_obj, 15)))) + message_str(426, 5500) + " ");
+        LVar0 := LVar0 + (mstr(367) + (6000 * (250 / (160 + dude_skill(SKILL_BARTER)))) + mstr(5500) + " ");
     end
     if (obj_is_carrying_obj_pid(dude_obj, 351)) then begin
-        LVar0 := LVar0 + (message_str(426, 368) + (12000 * (250 / (160 + has_skill(dude_obj, 15)))) + message_str(426, 5500));
+        LVar0 := LVar0 + (mstr(368) + (12000 * (250 / (160 + dude_skill(SKILL_BARTER)))) + mstr(5500));
     end
-    gsay_reply(426, LVar0);
-    if (obj_is_carrying_obj_pid(dude_obj, 10) and (item_caps_total(dude_obj) >= (4000 * (250 / (160 + has_skill(dude_obj, 15)))))) then begin
-        giq_option(4, 426, 369, Node985, 50);
+    Reply(LVar0);
+    if (obj_is_carrying_obj_pid(dude_obj, 10) and (item_caps_total(dude_obj) >= (4000 * (250 / (160 + dude_skill(SKILL_BARTER)))))) then begin
+        giq_option(4, NAME, 369, Node985, NEUTRAL_REACTION);
         LVar1 := 1;
     end
-    if (obj_is_carrying_obj_pid(dude_obj, 23) and (item_caps_total(dude_obj) >= (6000 * (250 / (160 + has_skill(dude_obj, 15)))))) then begin
-        giq_option(4, 426, 370, Node984, 50);
+    if (obj_is_carrying_obj_pid(dude_obj, 23) and (item_caps_total(dude_obj) >= (6000 * (250 / (160 + dude_skill(SKILL_BARTER)))))) then begin
+        giq_option(4, NAME, 370, Node984, NEUTRAL_REACTION);
         LVar1 := 1;
     end
-    if (obj_is_carrying_obj_pid(dude_obj, 351) and (item_caps_total(dude_obj) >= (12000 * (250 / (160 + has_skill(dude_obj, 15)))))) then begin
-        giq_option(4, 426, 371, Node983, 50);
+    if (obj_is_carrying_obj_pid(dude_obj, 351) and (item_caps_total(dude_obj) >= (12000 * (250 / (160 + dude_skill(SKILL_BARTER)))))) then begin
+        giq_option(4, NAME, 371, Node983, NEUTRAL_REACTION);
         LVar1 := 1;
     end
     if (LVar1 == 0) then begin
-        giq_option(4, 426, 357, Node999, 50);
+        giq_option(4, NAME, 357, Node999, NEUTRAL_REACTION);
     end
-    giq_option(4, 426, 354, Node010, 50);
-    giq_option(4, 426, 355, Node006, 50);
-    giq_option(4, 426, 356, Node999, 50);
+    giq_option(4, NAME, 354, Node010, NEUTRAL_REACTION);
+    giq_option(4, NAME, 355, Node006, NEUTRAL_REACTION);
+    giq_option(4, NAME, 356, Node999, NEUTRAL_REACTION);
     prev_node := 12;
 end
 
 procedure Node013
 begin
-    gsay_reply(426, message_str(426, 385) + (3000 * (250 / (160 + has_skill(dude_obj, 15)))) + message_str(426, 5500));
-    if (item_caps_total(dude_obj) >= (3000 * (250 / (160 + has_skill(dude_obj, 15))))) then begin
-        giq_option(4, 426, 386, Node982, 50);
+    Reply(mstr(385) + (3000 * (250 / (160 + dude_skill(SKILL_BARTER)))) + mstr(5500));
+    if (item_caps_total(dude_obj) >= (3000 * (250 / (160 + dude_skill(SKILL_BARTER))))) then begin
+        giq_option(4, NAME, 386, Node982, NEUTRAL_REACTION);
     end
     else begin
-        giq_option(4, 426, 357, Node999, 50);
+        giq_option(4, NAME, 357, Node999, NEUTRAL_REACTION);
     end
-    giq_option(4, 426, 354, Node010, 50);
-    giq_option(4, 426, 355, Node006, 50);
-    giq_option(4, 426, 356, Node999, 50);
+    giq_option(4, NAME, 354, Node010, NEUTRAL_REACTION);
+    giq_option(4, NAME, 355, Node006, NEUTRAL_REACTION);
+    giq_option(4, NAME, 356, Node999, NEUTRAL_REACTION);
     prev_node := 13;
 end
 
 procedure Node014
 begin
-    gsay_reply(426, 395);
-    giq_option(4, 426, 396, Node017, 50);
+    Reply(395);
+    giq_option(4, NAME, 396, Node017, NEUTRAL_REACTION);
     prev_node := 14;
 end
 
@@ -1016,9 +1016,9 @@ procedure Node015
 begin
     gfade_out(5);
     game_time_advance(60 * (60 * 10));
-    gsay_reply(426, 400);
+    Reply(400);
     gfade_in(5);
-    giq_option(4, 426, 401, Node016, 50);
+    giq_option(4, NAME, 401, Node016, NEUTRAL_REACTION);
     prev_node := 15;
 end
 
@@ -1026,9 +1026,9 @@ procedure Node016
 begin
     gfade_out(5);
     game_time_advance(60 * (60 * 10));
-    gsay_reply(426, 405);
+    Reply(405);
     gfade_in(5);
-    giq_option(4, 426, 406, Node017, 50);
+    giq_option(4, NAME, 406, Node017, NEUTRAL_REACTION);
     prev_node := 16;
 end
 
@@ -1036,65 +1036,65 @@ procedure Node017
 begin
     gfade_out(5);
     game_time_advance(60 * (60 * 10));
-    gsay_reply(426, 410);
+    Reply(410);
     gfade_in(5);
-    giq_option(4, 426, 411, Node010, 50);
-    giq_option(4, 426, 412, Node006, 50);
-    giq_option(4, 426, 413, Node999, 50);
+    giq_option(4, NAME, 411, Node010, NEUTRAL_REACTION);
+    giq_option(4, NAME, 412, Node006, NEUTRAL_REACTION);
+    giq_option(4, NAME, 413, Node999, NEUTRAL_REACTION);
     prev_node := 17;
 end
 
 procedure Node018
 begin
-    gsay_reply(426, 420);
-    giq_option(-3, 426, 302, Node019, 50);
-    giq_option(-3, 426, 304, Node019, 50);
-    giq_option(4, 426, 423, Node019, 50);
-    giq_option(4, 426, 424, Node006, 50);
-    giq_option(4, 426, 425, Node999, 50);
+    Reply(420);
+    giq_option(-3, NAME, 302, Node019, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 304, Node019, NEUTRAL_REACTION);
+    giq_option(4, NAME, 423, Node019, NEUTRAL_REACTION);
+    giq_option(4, NAME, 424, Node006, NEUTRAL_REACTION);
+    giq_option(4, NAME, 425, Node999, NEUTRAL_REACTION);
     prev_node := 18;
 end
 
 procedure Node019
 begin
-    gsay_reply(426, 430 + (get_critter_stat(dude_obj, 4) > 3) + (get_critter_stat(dude_obj, 4) > 6));
-    giq_option(-3, 426, 433, Node020, 50);
-    giq_option(-3, 426, 434, Node007, 50);
-    giq_option(-3, 426, 435, Node999, 50);
-    giq_option(4, 426, 436 + (global_var(350) == 1), Node020, 50);
-    giq_option(4, 426, 438, Node006, 50);
-    giq_option(4, 426, 439, Node999, 51);
+    Reply(430 + (dude_iq > 3) + (dude_iq > 6));
+    giq_option(-3, NAME, 433, Node020, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 434, Node007, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 435, Node999, NEUTRAL_REACTION);
+    giq_option(4, NAME, 436 + (global_var(GVAR_GRUTHAR_DSTATUS) == 1), Node020, NEUTRAL_REACTION);
+    giq_option(4, NAME, 438, Node006, NEUTRAL_REACTION);
+    giq_option(4, NAME, 439, Node999, BAD_REACTION);
     prev_node := 19;
 end
 
 procedure Node020
 begin
-    gsay_reply(426, 450 + (prev_node != 19));
+    Reply(450 + (prev_node != 19));
     if (item_caps_total(dude_obj) >= 3000) then begin
-        giq_option(-3, 426, 452, Node973, 50);
-        giq_option(4, 426, 386, Node973, 50);
+        giq_option(-3, NAME, 452, Node973, NEUTRAL_REACTION);
+        giq_option(4, NAME, 386, Node973, NEUTRAL_REACTION);
     end
-    giq_option(-3, 426, 434, Node007, 50);
-    giq_option(-3, 426, 454, Node981, 50);
-    giq_option(-3, 426, 455, Node999, 50);
+    giq_option(-3, NAME, 434, Node007, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 454, Node981, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 455, Node999, NEUTRAL_REACTION);
     if ((prev_node == 19) and (item_caps_total(dude_obj) >= 1500)) then begin
-        giq_option(4, 426, 457, Node980, 51);
+        giq_option(4, NAME, 457, Node980, BAD_REACTION);
     end
-    giq_option(4, 426, 458, Node006, 50);
-    giq_option(4, 426, 459, Node999, 50);
+    giq_option(4, NAME, 458, Node006, NEUTRAL_REACTION);
+    giq_option(4, NAME, 459, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node021
 begin
-    gsay_reply(426, 470);
+    Reply(470);
     if (item_caps_total(dude_obj) >= 2500) then begin
-        giq_option(4, 426, 471, Node979, 50);
+        giq_option(4, NAME, 471, Node979, NEUTRAL_REACTION);
     end
     else begin
-        giq_option(4, 426, 474, Node999, 50);
+        giq_option(4, NAME, 474, Node999, NEUTRAL_REACTION);
     end
-    giq_option(4, 426, 472, Node006, 50);
-    giq_option(4, 426, 473, Node999, 50);
+    giq_option(4, NAME, 472, Node006, NEUTRAL_REACTION);
+    giq_option(4, NAME, 473, Node999, NEUTRAL_REACTION);
     prev_node := 21;
 end
 
@@ -1103,36 +1103,36 @@ begin
     return_node := "Node022";
     if (((local_var(4) bwand 512) != 0) == 0) then begin
         set_local_var(4, local_var(4) bwor 512);
-        restock_obj := create_object_sid(356, 0, 0, -1);
+        restock_obj := create_object(356, 0, 0);
         add_mult_objs_to_inven(dude_obj, restock_obj, 1);
     end
-    gsay_reply(426, message_str(426, 480 + ((local_var(4) bwand 2048) != 0)) + message_str(426, 482));
-    giq_option(-3, 426, 302, Node974, 50);
-    giq_option(-3, 426, 304, Node999, 50);
-    giq_option(4, 426, 485, Node006, 50);
-    giq_option(4, 426, 486, Node999, 51);
+    Reply(mstr(480 + ((local_var(4) bwand 2048) != 0)) + mstr(482));
+    giq_option(-3, NAME, 302, Node974, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 304, Node999, NEUTRAL_REACTION);
+    giq_option(4, NAME, 485, Node006, NEUTRAL_REACTION);
+    giq_option(4, NAME, 486, Node999, BAD_REACTION);
     prev_node := 22;
 end
 
 procedure Node023
 begin
-    gsay_reply(426, 495);
-    giq_option(4, 426, 496, Node024, 50);
-    giq_option(4, 426, 497, Node006, 50);
-    giq_option(4, 426, 498, Node999, 50);
+    Reply(495);
+    giq_option(4, NAME, 496, Node024, NEUTRAL_REACTION);
+    giq_option(4, NAME, 497, Node006, NEUTRAL_REACTION);
+    giq_option(4, NAME, 498, Node999, NEUTRAL_REACTION);
     prev_node := 23;
 end
 
 procedure Node024
 begin
-    gsay_reply(426, 505);
+    Reply(505);
     if (obj_is_carrying_obj_pid(dude_obj, 16)) then begin
-        giq_option(4, 426, 506, Node025, 49);
+        giq_option(4, NAME, 506, Node025, GOOD_REACTION);
     end
-    giq_option(4, 426, 507, Node978, 50);
-    giq_option(4, 426, 508, Node006, 50);
-    giq_option(4, 426, 509, Node977, 50);
-    giq_option(4, 426, 510, Node999, 50);
+    giq_option(4, NAME, 507, Node978, NEUTRAL_REACTION);
+    giq_option(4, NAME, 508, Node006, NEUTRAL_REACTION);
+    giq_option(4, NAME, 509, Node977, NEUTRAL_REACTION);
+    giq_option(4, NAME, 510, Node999, NEUTRAL_REACTION);
     prev_node := 24;
 end
 
@@ -1148,14 +1148,14 @@ begin
         add_mult_objs_to_inven(new_reno_temp_box, restock_obj, restock_amt);
     end
     item_caps_adjust(dude_obj, 1500);
-    display_msg(message_str(426, 800));
+    display_mstr(800);
     give_exp_points(500);
-    display_msg(message_str(14, 100) + 500 + message_str(14, 101));
-    set_global_var(55, global_var(55) + 1);
-    set_global_var(549, 2);
-    gsay_reply(426, message_str(426, 520 + (prev_node != 24)) + message_str(426, 522));
-    giq_option(4, 426, 523, Node006, 50);
-    giq_option(4, 426, 524, Node999, 50);
+    display_msg(g_mstr(100) + 500 + g_mstr(101));
+    set_global_var(GVAR_TOWN_REP_NEW_RENO, global_var(GVAR_TOWN_REP_NEW_RENO) + 1);
+    set_global_var(GVAR_NEW_RENO_ELDRIDGE_PISTOL_QUEST, 2);
+    Reply(mstr(520 + (prev_node != 24)) + mstr(522));
+    giq_option(4, NAME, 523, Node006, NEUTRAL_REACTION);
+    giq_option(4, NAME, 524, Node999, NEUTRAL_REACTION);
     prev_node := 25;
 end
 
@@ -1168,41 +1168,41 @@ begin
         gdialog_set_barter_mod(local_var(5));
         set_local_var(4, local_var(4) bwor 1024);
     end
-    gsay_reply(426, 530);
-    giq_option(4, 426, 386, Node975, 50);
-    giq_option(4, 426, 532, Node006, 50);
-    giq_option(4, 426, 533, Node999, 50);
+    Reply(530);
+    giq_option(4, NAME, 386, Node975, NEUTRAL_REACTION);
+    giq_option(4, NAME, 532, Node006, NEUTRAL_REACTION);
+    giq_option(4, NAME, 533, Node999, NEUTRAL_REACTION);
     prev_node := 26;
 end
 
 procedure Node027
 begin
-    float_msg(self_obj, message_str(426, random(545, 550)), 8 + ((2 - 8) * has_trait(0, dude_obj, 22)));
+    float_msg(self_obj, mstr(random(545, 550)), 8 + ((2 - 8) * has_trait(TRAIT_PERK, dude_obj, PERK_empathy)));
 end
 
 procedure Node028
 begin
-    float_msg(self_obj, message_str(426, random(560, 564)), 8 + ((2 - 8) * has_trait(0, dude_obj, 22)));
+    float_msg(self_obj, mstr(random(560, 564)), 8 + ((2 - 8) * has_trait(TRAIT_PERK, dude_obj, PERK_empathy)));
 end
 
 procedure Node029
 begin
-    float_msg(self_obj, message_str(426, random(575, 578)), 8 + ((2 - 8) * has_trait(0, dude_obj, 22)));
+    float_msg(self_obj, mstr(random(575, 578)), 8 + ((2 - 8) * has_trait(TRAIT_PERK, dude_obj, PERK_empathy)));
 end
 
 procedure Node030
 begin
     animate_stand_obj(self_obj);
-    float_msg(self_obj, message_str(426, 600 + (get_critter_stat(dude_obj, 34) == 1)) + message_str(426, 610), 8);
+    float_msg(self_obj, mstr(600 + (dude_gender == 1)) + mstr(610), FLOAT_MSG_YELLOW);
 end
 
 procedure Node990
 begin
-    if (global_var(494) == 8) then begin
+    if (global_var(GVAR_FALLOUT_2) == 8) then begin
         call Node005();
     end
     else begin
-        if (get_critter_stat(dude_obj, 4) <= 3) then begin
+        if (dude_is_stupid) then begin
             call Node007();
         end
         else begin
@@ -1218,7 +1218,7 @@ end
 
 procedure Node988
 begin
-    if ((global_var(345) bwand 1048576) != 0) then begin
+    if ((global_var(GVAR_NEW_RENO_FLAG_2) bwand 1048576) != 0) then begin
         call Node009();
     end
     else begin
@@ -1228,8 +1228,8 @@ end
 
 procedure Node987
 begin
-    item_caps_adjust(dude_obj, -(2000 * (250 / (160 + has_skill(dude_obj, 15)))));
-    item_caps_adjust(self_obj, 2000 * (250 / (160 + has_skill(dude_obj, 15))));
+    item_caps_adjust(dude_obj, -(2000 * (250 / (160 + dude_skill(SKILL_BARTER)))));
+    item_caps_adjust(self_obj, 2000 * (250 / (160 + dude_skill(SKILL_BARTER))));
     removed_qty := obj_is_carrying_obj_pid(dude_obj, 313);
     if (1 < removed_qty) then begin
         removed_qty := 1;
@@ -1238,17 +1238,17 @@ begin
         restock_obj := obj_carrying_pid_obj(dude_obj, 313);
         if (obj_type(self_obj) == 1) then begin
             restock_obj := obj_carrying_pid_obj(dude_obj, 313);
-            if (critter_inven_obj(self_obj, 0) == restock_obj) then begin
-                if (obj_item_subtype(critter_inven_obj(dude_obj, 0)) == 0) then begin
-                    restock_obj := critter_inven_obj(dude_obj, 0);
+            if (critter_inven_obj(self_obj, INVEN_TYPE_WORN) == restock_obj) then begin
+                if (obj_item_subtype(critter_inven_obj(dude_obj, INVEN_TYPE_WORN)) == 0) then begin
+                    restock_obj := critter_inven_obj(dude_obj, INVEN_TYPE_WORN);
                     debug_msg("armour pid == " + obj_pid(restock_obj));
                     rm_obj_from_inven(dude_obj, restock_obj);
                     add_obj_to_inven(dude_obj, restock_obj);
                 end
             end
             else begin
-                if ((critter_inven_obj(dude_obj, 2) == restock_obj) or (critter_inven_obj(dude_obj, 1) == restock_obj)) then begin
-                    metarule(43, dude_obj);
+                if ((critter_inven_obj(dude_obj, INVEN_TYPE_LEFT_HAND) == restock_obj) or (critter_inven_obj(dude_obj, INVEN_TYPE_RIGHT_HAND) == restock_obj)) then begin
+                    inven_unwield(dude_obj);
                 end
             end
         end
@@ -1257,15 +1257,15 @@ begin
     end
     restock_obj := 0;
     restock_amt := 0;
-    restock_obj := create_object_sid(398, 0, 0, -1);
+    restock_obj := create_object(398, 0, 0);
     add_mult_objs_to_inven(dude_obj, restock_obj, 1);
     call Node014();
 end
 
 procedure Node986
 begin
-    item_caps_adjust(dude_obj, -(3000 * (250 / (160 + has_skill(dude_obj, 15)))));
-    item_caps_adjust(self_obj, 3000 * (250 / (160 + has_skill(dude_obj, 15))));
+    item_caps_adjust(dude_obj, -(3000 * (250 / (160 + dude_skill(SKILL_BARTER)))));
+    item_caps_adjust(self_obj, 3000 * (250 / (160 + dude_skill(SKILL_BARTER))));
     removed_qty := obj_is_carrying_obj_pid(dude_obj, 18);
     if (1 < removed_qty) then begin
         removed_qty := 1;
@@ -1274,17 +1274,17 @@ begin
         restock_obj := obj_carrying_pid_obj(dude_obj, 18);
         if (obj_type(self_obj) == 1) then begin
             restock_obj := obj_carrying_pid_obj(dude_obj, 18);
-            if (critter_inven_obj(self_obj, 0) == restock_obj) then begin
-                if (obj_item_subtype(critter_inven_obj(dude_obj, 0)) == 0) then begin
-                    restock_obj := critter_inven_obj(dude_obj, 0);
+            if (critter_inven_obj(self_obj, INVEN_TYPE_WORN) == restock_obj) then begin
+                if (obj_item_subtype(critter_inven_obj(dude_obj, INVEN_TYPE_WORN)) == 0) then begin
+                    restock_obj := critter_inven_obj(dude_obj, INVEN_TYPE_WORN);
                     debug_msg("armour pid == " + obj_pid(restock_obj));
                     rm_obj_from_inven(dude_obj, restock_obj);
                     add_obj_to_inven(dude_obj, restock_obj);
                 end
             end
             else begin
-                if ((critter_inven_obj(dude_obj, 2) == restock_obj) or (critter_inven_obj(dude_obj, 1) == restock_obj)) then begin
-                    metarule(43, dude_obj);
+                if ((critter_inven_obj(dude_obj, INVEN_TYPE_LEFT_HAND) == restock_obj) or (critter_inven_obj(dude_obj, INVEN_TYPE_RIGHT_HAND) == restock_obj)) then begin
+                    inven_unwield(dude_obj);
                 end
             end
         end
@@ -1293,15 +1293,15 @@ begin
     end
     restock_obj := 0;
     restock_amt := 0;
-    restock_obj := create_object_sid(404, 0, 0, -1);
+    restock_obj := create_object(404, 0, 0);
     add_mult_objs_to_inven(dude_obj, restock_obj, 1);
     call Node014();
 end
 
 procedure Node985
 begin
-    item_caps_adjust(dude_obj, -(4000 * (250 / (160 + has_skill(dude_obj, 15)))));
-    item_caps_adjust(self_obj, 4000 * (250 / (160 + has_skill(dude_obj, 15))));
+    item_caps_adjust(dude_obj, -(4000 * (250 / (160 + dude_skill(SKILL_BARTER)))));
+    item_caps_adjust(self_obj, 4000 * (250 / (160 + dude_skill(SKILL_BARTER))));
     removed_qty := obj_is_carrying_obj_pid(dude_obj, 10);
     if (1 < removed_qty) then begin
         removed_qty := 1;
@@ -1310,17 +1310,17 @@ begin
         restock_obj := obj_carrying_pid_obj(dude_obj, 10);
         if (obj_type(self_obj) == 1) then begin
             restock_obj := obj_carrying_pid_obj(dude_obj, 10);
-            if (critter_inven_obj(self_obj, 0) == restock_obj) then begin
-                if (obj_item_subtype(critter_inven_obj(dude_obj, 0)) == 0) then begin
-                    restock_obj := critter_inven_obj(dude_obj, 0);
+            if (critter_inven_obj(self_obj, INVEN_TYPE_WORN) == restock_obj) then begin
+                if (obj_item_subtype(critter_inven_obj(dude_obj, INVEN_TYPE_WORN)) == 0) then begin
+                    restock_obj := critter_inven_obj(dude_obj, INVEN_TYPE_WORN);
                     debug_msg("armour pid == " + obj_pid(restock_obj));
                     rm_obj_from_inven(dude_obj, restock_obj);
                     add_obj_to_inven(dude_obj, restock_obj);
                 end
             end
             else begin
-                if ((critter_inven_obj(dude_obj, 2) == restock_obj) or (critter_inven_obj(dude_obj, 1) == restock_obj)) then begin
-                    metarule(43, dude_obj);
+                if ((critter_inven_obj(dude_obj, INVEN_TYPE_LEFT_HAND) == restock_obj) or (critter_inven_obj(dude_obj, INVEN_TYPE_RIGHT_HAND) == restock_obj)) then begin
+                    inven_unwield(dude_obj);
                 end
             end
         end
@@ -1329,15 +1329,15 @@ begin
     end
     restock_obj := 0;
     restock_amt := 0;
-    restock_obj := create_object_sid(287, 0, 0, -1);
+    restock_obj := create_object(287, 0, 0);
     add_mult_objs_to_inven(dude_obj, restock_obj, 1);
     call Node014();
 end
 
 procedure Node984
 begin
-    item_caps_adjust(dude_obj, -(6000 * (250 / (160 + has_skill(dude_obj, 15)))));
-    item_caps_adjust(self_obj, 6000 * (250 / (160 + has_skill(dude_obj, 15))));
+    item_caps_adjust(dude_obj, -(6000 * (250 / (160 + dude_skill(SKILL_BARTER)))));
+    item_caps_adjust(self_obj, 6000 * (250 / (160 + dude_skill(SKILL_BARTER))));
     removed_qty := obj_is_carrying_obj_pid(dude_obj, 23);
     if (1 < removed_qty) then begin
         removed_qty := 1;
@@ -1346,17 +1346,17 @@ begin
         restock_obj := obj_carrying_pid_obj(dude_obj, 23);
         if (obj_type(self_obj) == 1) then begin
             restock_obj := obj_carrying_pid_obj(dude_obj, 23);
-            if (critter_inven_obj(self_obj, 0) == restock_obj) then begin
-                if (obj_item_subtype(critter_inven_obj(dude_obj, 0)) == 0) then begin
-                    restock_obj := critter_inven_obj(dude_obj, 0);
+            if (critter_inven_obj(self_obj, INVEN_TYPE_WORN) == restock_obj) then begin
+                if (obj_item_subtype(critter_inven_obj(dude_obj, INVEN_TYPE_WORN)) == 0) then begin
+                    restock_obj := critter_inven_obj(dude_obj, INVEN_TYPE_WORN);
                     debug_msg("armour pid == " + obj_pid(restock_obj));
                     rm_obj_from_inven(dude_obj, restock_obj);
                     add_obj_to_inven(dude_obj, restock_obj);
                 end
             end
             else begin
-                if ((critter_inven_obj(dude_obj, 2) == restock_obj) or (critter_inven_obj(dude_obj, 1) == restock_obj)) then begin
-                    metarule(43, dude_obj);
+                if ((critter_inven_obj(dude_obj, INVEN_TYPE_LEFT_HAND) == restock_obj) or (critter_inven_obj(dude_obj, INVEN_TYPE_RIGHT_HAND) == restock_obj)) then begin
+                    inven_unwield(dude_obj);
                 end
             end
         end
@@ -1365,15 +1365,15 @@ begin
     end
     restock_obj := 0;
     restock_amt := 0;
-    restock_obj := create_object_sid(405, 0, 0, -1);
+    restock_obj := create_object(405, 0, 0);
     add_mult_objs_to_inven(dude_obj, restock_obj, 1);
     call Node014();
 end
 
 procedure Node983
 begin
-    item_caps_adjust(dude_obj, -(12000 * (250 / (160 + has_skill(dude_obj, 15)))));
-    item_caps_adjust(self_obj, 12000 * (250 / (160 + has_skill(dude_obj, 15))));
+    item_caps_adjust(dude_obj, -(12000 * (250 / (160 + dude_skill(SKILL_BARTER)))));
+    item_caps_adjust(self_obj, 12000 * (250 / (160 + dude_skill(SKILL_BARTER))));
     removed_qty := obj_is_carrying_obj_pid(dude_obj, 351);
     if (1 < removed_qty) then begin
         removed_qty := 1;
@@ -1382,17 +1382,17 @@ begin
         restock_obj := obj_carrying_pid_obj(dude_obj, 351);
         if (obj_type(self_obj) == 1) then begin
             restock_obj := obj_carrying_pid_obj(dude_obj, 351);
-            if (critter_inven_obj(self_obj, 0) == restock_obj) then begin
-                if (obj_item_subtype(critter_inven_obj(dude_obj, 0)) == 0) then begin
-                    restock_obj := critter_inven_obj(dude_obj, 0);
+            if (critter_inven_obj(self_obj, INVEN_TYPE_WORN) == restock_obj) then begin
+                if (obj_item_subtype(critter_inven_obj(dude_obj, INVEN_TYPE_WORN)) == 0) then begin
+                    restock_obj := critter_inven_obj(dude_obj, INVEN_TYPE_WORN);
                     debug_msg("armour pid == " + obj_pid(restock_obj));
                     rm_obj_from_inven(dude_obj, restock_obj);
                     add_obj_to_inven(dude_obj, restock_obj);
                 end
             end
             else begin
-                if ((critter_inven_obj(dude_obj, 2) == restock_obj) or (critter_inven_obj(dude_obj, 1) == restock_obj)) then begin
-                    metarule(43, dude_obj);
+                if ((critter_inven_obj(dude_obj, INVEN_TYPE_LEFT_HAND) == restock_obj) or (critter_inven_obj(dude_obj, INVEN_TYPE_RIGHT_HAND) == restock_obj)) then begin
+                    inven_unwield(dude_obj);
                 end
             end
         end
@@ -1401,15 +1401,15 @@ begin
     end
     restock_obj := 0;
     restock_amt := 0;
-    restock_obj := create_object_sid(403, 0, 0, -1);
+    restock_obj := create_object(403, 0, 0);
     add_mult_objs_to_inven(dude_obj, restock_obj, 1);
     call Node014();
 end
 
 procedure Node982
 begin
-    item_caps_adjust(dude_obj, -(3000 * (250 / (160 + has_skill(dude_obj, 15)))));
-    item_caps_adjust(self_obj, 3000 * (250 / (160 + has_skill(dude_obj, 15))));
+    item_caps_adjust(dude_obj, -(3000 * (250 / (160 + dude_skill(SKILL_BARTER)))));
+    item_caps_adjust(self_obj, 3000 * (250 / (160 + dude_skill(SKILL_BARTER))));
     removed_qty := obj_is_carrying_obj_pid(dude_obj, 160);
     if (1 < removed_qty) then begin
         removed_qty := 1;
@@ -1418,17 +1418,17 @@ begin
         restock_obj := obj_carrying_pid_obj(dude_obj, 160);
         if (obj_type(self_obj) == 1) then begin
             restock_obj := obj_carrying_pid_obj(dude_obj, 160);
-            if (critter_inven_obj(self_obj, 0) == restock_obj) then begin
-                if (obj_item_subtype(critter_inven_obj(dude_obj, 0)) == 0) then begin
-                    restock_obj := critter_inven_obj(dude_obj, 0);
+            if (critter_inven_obj(self_obj, INVEN_TYPE_WORN) == restock_obj) then begin
+                if (obj_item_subtype(critter_inven_obj(dude_obj, INVEN_TYPE_WORN)) == 0) then begin
+                    restock_obj := critter_inven_obj(dude_obj, INVEN_TYPE_WORN);
                     debug_msg("armour pid == " + obj_pid(restock_obj));
                     rm_obj_from_inven(dude_obj, restock_obj);
                     add_obj_to_inven(dude_obj, restock_obj);
                 end
             end
             else begin
-                if ((critter_inven_obj(dude_obj, 2) == restock_obj) or (critter_inven_obj(dude_obj, 1) == restock_obj)) then begin
-                    metarule(43, dude_obj);
+                if ((critter_inven_obj(dude_obj, INVEN_TYPE_LEFT_HAND) == restock_obj) or (critter_inven_obj(dude_obj, INVEN_TYPE_RIGHT_HAND) == restock_obj)) then begin
+                    inven_unwield(dude_obj);
                 end
             end
         end
@@ -1437,7 +1437,7 @@ begin
     end
     restock_obj := 0;
     restock_amt := 0;
-    restock_obj := create_object_sid(399, 0, 0, -1);
+    restock_obj := create_object(399, 0, 0);
     add_mult_objs_to_inven(dude_obj, restock_obj, 1);
     call Node014();
 end
@@ -1446,7 +1446,7 @@ procedure Node981
 begin
     if (((local_var(4) bwand 512) != 0) == 0) then begin
         set_local_var(4, local_var(4) bwor 512);
-        restock_obj := create_object_sid(356, 0, 0, -1);
+        restock_obj := create_object(356, 0, 0);
         add_mult_objs_to_inven(self_obj, restock_obj, 1);
     end
     drop_obj(restock_obj);
@@ -1457,7 +1457,7 @@ procedure Node980
 begin
     variable LVar0 := 0;
     LVar0 := roll_vs_skill(dude_obj, 15, 0);
-    if ((get_critter_stat(dude_obj, 3) > 8) or (is_success(LVar0) and is_critical(LVar0))) then begin
+    if ((dude_charisma > 8) or (is_success(LVar0) and is_critical(LVar0))) then begin
         set_local_var(4, local_var(4) bwor 2048);
         item_caps_adjust(dude_obj, -1500);
         item_caps_adjust(self_obj, 1500);
@@ -1482,13 +1482,13 @@ end
 
 procedure Node978
 begin
-    set_global_var(549, 1);
+    set_global_var(GVAR_NEW_RENO_ELDRIDGE_PISTOL_QUEST, 1);
     call Node006();
 end
 
 procedure Node977
 begin
-    set_global_var(549, 1);
+    set_global_var(GVAR_NEW_RENO_ELDRIDGE_PISTOL_QUEST, 1);
 end
 
 procedure Node975
@@ -1508,7 +1508,7 @@ begin
             current_box := 0;
         end
     end
-    gdialog_mod_barter(0);
+    gdialog_barter;
     call return_node();
 end
 
@@ -1529,7 +1529,7 @@ begin
             current_box := 0;
         end
     end
-    gdialog_mod_barter(0);
+    gdialog_barter;
     call return_node();
 end
 
@@ -1539,4 +1539,3 @@ begin
     item_caps_adjust(self_obj, 3000);
     call Node022();
 end
-
