diff --git a/Games/Fallout2.Steam/Original/NCHSSAL2.ssl b/Games/Fallout2.Steam/ReDefine/NCHSSAL2.ssl
index 61352b2..0ca1622 100644
--- a/Games/Fallout2.Steam/Original/NCHSSAL2.ssl
+++ b/Games/Fallout2.Steam/ReDefine/NCHSSAL2.ssl
@@ -212,53 +212,53 @@ end
 procedure ChooseFromList
 begin
     variable LVar0 := 0;
-    gsay_reply(1032, GVar22);
-    if (message_str(353, GVar24 - 1) != "") then begin
-        giq_option(1, 1032, message_str(353, 150), CLICK_PREV, 50);
+    gsay_reply(SCRIPT_NEWRVB, GVar22);
+    if (message_str(SCRIPT_NEWR1, GVar24 - 1) != "") then begin
+        giq_option(1, SCRIPT_NEWRVB, message_str(SCRIPT_NEWR1, 150), CLICK_PREV, NEUTRAL_REACTION);
     end
     if (LVar0 == 0) then begin
-        if (message_str(353, GVar24 + 0) != "") then begin
-            giq_option(1, 1032, message_str(353, GVar24 + 0), CLICK_0, 50);
+        if (message_str(SCRIPT_NEWR1, GVar24 + 0) != "") then begin
+            giq_option(1, SCRIPT_NEWRVB, message_str(SCRIPT_NEWR1, GVar24 + 0), CLICK_0, NEUTRAL_REACTION);
         end
         else begin
             LVar0 := 1;
         end
     end
     if (LVar0 == 0) then begin
-        if (message_str(353, GVar24 + 1) != "") then begin
-            giq_option(1, 1032, message_str(353, GVar24 + 1), CLICK_1, 50);
+        if (message_str(SCRIPT_NEWR1, GVar24 + 1) != "") then begin
+            giq_option(1, SCRIPT_NEWRVB, message_str(SCRIPT_NEWR1, GVar24 + 1), CLICK_1, NEUTRAL_REACTION);
         end
         else begin
             LVar0 := 1;
         end
     end
     if (LVar0 == 0) then begin
-        if (message_str(353, GVar24 + 2) != "") then begin
-            giq_option(1, 1032, message_str(353, GVar24 + 2), CLICK_2, 50);
+        if (message_str(SCRIPT_NEWR1, GVar24 + 2) != "") then begin
+            giq_option(1, SCRIPT_NEWRVB, message_str(SCRIPT_NEWR1, GVar24 + 2), CLICK_2, NEUTRAL_REACTION);
         end
         else begin
             LVar0 := 1;
         end
     end
     if (LVar0 == 0) then begin
-        if (message_str(353, GVar24 + 3) != "") then begin
-            giq_option(1, 1032, message_str(353, GVar24 + 3), CLICK_3, 50);
+        if (message_str(SCRIPT_NEWR1, GVar24 + 3) != "") then begin
+            giq_option(1, SCRIPT_NEWRVB, message_str(SCRIPT_NEWR1, GVar24 + 3), CLICK_3, NEUTRAL_REACTION);
         end
         else begin
             LVar0 := 1;
         end
     end
     if (LVar0 == 0) then begin
-        if (message_str(353, GVar24 + 4) != "") then begin
-            giq_option(1, 1032, message_str(353, GVar24 + 4), CLICK_4, 50);
+        if (message_str(SCRIPT_NEWR1, GVar24 + 4) != "") then begin
+            giq_option(1, SCRIPT_NEWRVB, message_str(SCRIPT_NEWR1, GVar24 + 4), CLICK_4, NEUTRAL_REACTION);
         end
         else begin
             LVar0 := 1;
         end
     end
     if (LVar0 == 0) then begin
-        if (message_str(353, GVar24 + 5) != "") then begin
-            giq_option(1, 1032, message_str(353, 151), CLICK_MORE, 50);
+        if (message_str(SCRIPT_NEWR1, GVar24 + 5) != "") then begin
+            giq_option(1, SCRIPT_NEWRVB, message_str(SCRIPT_NEWR1, 151), CLICK_MORE, NEUTRAL_REACTION);
         end
     end
 end
@@ -313,15 +313,15 @@ end
 procedure critter_p_proc
 begin
     if (obj_can_see_obj(self_obj, dude_obj)) then begin
-        if (((local_var(0) bwand 2) != 0) or (((global_var(314) bwand 32768) != 0) or (global_var(494) == 8))) then begin
+        if (((local_var(0) bwand 2) != 0) or (((global_var(GVAR_NEW_RENO_FLAG_1) bwand 32768) != 0) or (global_var(GVAR_FALLOUT_2) == 8))) then begin
             if (critter_is_fleeing(self_obj)) then begin
                 if (anim_busy(self_obj) == 0) then begin
                     if ((tile_distance_objs(self_obj, dude_obj) < 8) or obj_can_see_obj(self_obj, dude_obj)) then begin
                         if (anim_busy(self_obj) == 0) then begin
-                            GVar10 := rotation_to_tile(tile_num(dude_obj), tile_num(self_obj));
+                            GVar10 := rotation_to_tile(dude_tile, tile_num(self_obj));
                             animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(self_obj), GVar10, random(3, 10)), 1);
                             GVar10 := (GVar10 + 1) % 6;
-                            while ((anim_busy(self_obj) == 0) and (GVar10 != rotation_to_tile(tile_num(dude_obj), tile_num(self_obj)))) do begin
+                            while ((anim_busy(self_obj) == 0) and (GVar10 != rotation_to_tile(dude_tile, tile_num(self_obj)))) do begin
                                 debug_msg("flee loop: rot == " + GVar10);
                                 animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(self_obj), GVar10, random(3, 10)), 1);
                                 GVar10 := (GVar10 + 1) % 6;
@@ -331,10 +331,10 @@ begin
                 end
             end
             else begin
-                attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+                attack(dude_obj);
             end
         end
-        if (global_var(313) == 0) then begin
+        if (global_var(GVAR_NEW_RENO_GUARD_ASSIGNMENT) == 0) then begin
             call Node998();
         end
     end
@@ -342,14 +342,14 @@ end
 
 procedure pickup_p_proc
 begin
-    if ((source_obj == dude_obj) or (party_member_obj(obj_pid(source_obj)) != 0)) then begin
+    if ((source_obj == dude_obj) or (obj_in_party(source_obj))) then begin
         call Node998();
     end
 end
 
 procedure talk_p_proc
 begin
-    if (global_var(313) == 0) then begin
+    if (global_var(GVAR_NEW_RENO_GUARD_ASSIGNMENT) == 0) then begin
         call Node998();
     end
     else begin
@@ -361,55 +361,55 @@ end
 procedure destroy_p_proc
 begin
     if (source_obj == dude_obj) then begin
-        set_global_var(4, global_var(4) + 1);
-        if (metarule(51, self_obj) == 2) then begin
+        set_global_var(GVAR_BAD_MONSTER, global_var(GVAR_BAD_MONSTER) + 1);
+        if (critter_kill_type(self_obj) == KILL_TYPE_children_kills) then begin
             if (source_obj == dude_obj) then begin
-                set_global_var(2, 0);
-                set_global_var(1, global_var(1) + 1);
-                set_global_var(0, global_var(0) + -15);
+                set_global_var(GVAR_CHAMPION_REPUTATION, 0);
+                set_global_var(GVAR_CHILDKILLER_REPUTATION, global_var(GVAR_CHILDKILLER_REPUTATION) + 1);
+                set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + -15);
                 debug_msg("Player gains " + -15 + " Karma Points.");
-                set_global_var(37, 0);
-                set_global_var(38, 0);
-                set_global_var(39, 0);
-                set_global_var(40, 0);
-                set_global_var(41, 0);
-                set_global_var(42, 0);
-                set_global_var(43, 0);
-                set_global_var(44, 0);
-                set_global_var(45, 0);
-                if (global_var(0) >= 1000) then begin
-                    set_global_var(37, 1);
+                set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+                set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+                set_global_var(GVAR_KARMA_DEFENDER, 0);
+                set_global_var(GVAR_KARMA_WANDERER, 0);
+                set_global_var(GVAR_KARMA_BETRAYER, 0);
+                set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+                set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+                set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+                    set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
                 end
                 else begin
-                    if (global_var(0) >= 750) then begin
-                        set_global_var(38, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
                     end
                     else begin
-                        if (global_var(0) >= 500) then begin
-                            set_global_var(39, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                            set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                         end
                         else begin
-                            if (global_var(0) >= 250) then begin
-                                set_global_var(40, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                                set_global_var(GVAR_KARMA_DEFENDER, 1);
                             end
                             else begin
-                                if (global_var(0) > -250) then begin
-                                    set_global_var(41, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                                    set_global_var(GVAR_KARMA_WANDERER, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -500) then begin
-                                        set_global_var(42, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                        set_global_var(GVAR_KARMA_BETRAYER, 1);
                                     end
                                     else begin
-                                        if (global_var(0) > -750) then begin
-                                            set_global_var(43, 1);
+                                        if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                            set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                         end
                                         else begin
-                                            if (global_var(0) > -1000) then begin
-                                                set_global_var(44, 1);
+                                            if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                                set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                             end
                                             else begin
-                                                set_global_var(45, 1);
+                                                set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                             end
                                         end
                                     end
@@ -418,62 +418,62 @@ begin
                         end
                     end
                 end
-                set_global_var(55, global_var(55) + -8);
+                set_global_var(GVAR_TOWN_REP_NEW_RENO, global_var(GVAR_TOWN_REP_NEW_RENO) + -8);
                 debug_msg("Added " + -8 + " to Town Rep");
             end
         end
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(5) > (2 * global_var(4))) or (global_var(3) == 1))) then begin
-            set_global_var(3, 1);
-            set_global_var(2, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
+            set_global_var(GVAR_BERSERKER_REPUTATION, 1);
+            set_global_var(GVAR_CHAMPION_REPUTATION, 0);
         end
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(4) > (3 * global_var(5))) or (global_var(2) == 1)) and (global_var(1) == 0)) then begin
-            set_global_var(2, 1);
-            set_global_var(3, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(4) > (3 * global_var(GVAR_GOOD_MONSTER))) or (global_var(GVAR_CHAMPION_REPUTATION) == 1)) and (global_var(GVAR_CHILDKILLER_REPUTATION) == 0)) then begin
+            set_global_var(GVAR_CHAMPION_REPUTATION, 1);
+            set_global_var(GVAR_BERSERKER_REPUTATION, 0);
         end
-        set_global_var(0, global_var(0) + 5);
+        set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + 5);
         debug_msg("Player gains " + 5 + " Karma Points.");
-        set_global_var(37, 0);
-        set_global_var(38, 0);
-        set_global_var(39, 0);
-        set_global_var(40, 0);
-        set_global_var(41, 0);
-        set_global_var(42, 0);
-        set_global_var(43, 0);
-        set_global_var(44, 0);
-        set_global_var(45, 0);
-        if (global_var(0) >= 1000) then begin
-            set_global_var(37, 1);
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+        set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+        set_global_var(GVAR_KARMA_DEFENDER, 0);
+        set_global_var(GVAR_KARMA_WANDERER, 0);
+        set_global_var(GVAR_KARMA_BETRAYER, 0);
+        set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+        set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+            set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
         end
         else begin
-            if (global_var(0) >= 750) then begin
-                set_global_var(38, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
             end
             else begin
-                if (global_var(0) >= 500) then begin
-                    set_global_var(39, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                 end
                 else begin
-                    if (global_var(0) >= 250) then begin
-                        set_global_var(40, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                        set_global_var(GVAR_KARMA_DEFENDER, 1);
                     end
                     else begin
-                        if (global_var(0) > -250) then begin
-                            set_global_var(41, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                            set_global_var(GVAR_KARMA_WANDERER, 1);
                         end
                         else begin
-                            if (global_var(0) > -500) then begin
-                                set_global_var(42, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                set_global_var(GVAR_KARMA_BETRAYER, 1);
                             end
                             else begin
-                                if (global_var(0) > -750) then begin
-                                    set_global_var(43, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -1000) then begin
-                                        set_global_var(44, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                     end
                                     else begin
-                                        set_global_var(45, 1);
+                                        set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                     end
                                 end
                             end
@@ -482,7 +482,7 @@ begin
                 end
             end
         end
-        set_global_var(55, global_var(55) + 2);
+        set_global_var(GVAR_TOWN_REP_NEW_RENO, global_var(GVAR_TOWN_REP_NEW_RENO) + 2);
         debug_msg("Added " + 2 + " to Town Rep");
     end
     salvatore_guard_2_obj := 0;
@@ -512,14 +512,14 @@ end
 procedure map_enter_p_proc
 begin
     salvatore_guard_2_obj := self_obj;
-    if (metarule(22, 0) == 0) then begin
-        if (metarule(14, 0)) then begin
+    if (is_loading_game == 0) then begin
+        if (map_first_run) then begin
             set_local_var(1, tile_num(self_obj));
-            critter_add_trait(self_obj, 1, 6, 59);
-            critter_add_trait(self_obj, 1, 5, 22);
+            critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 59);
+            critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_TOUGH_GUARD);
         end
         else begin
-            if (((global_var(314) bwand 32768) != 0) or (global_var(494) == 8)) then begin
+            if (((global_var(GVAR_NEW_RENO_FLAG_1) bwand 32768) != 0) or (global_var(GVAR_FALLOUT_2) == 8)) then begin
                 destroy_object(self_obj);
             end
             else begin
@@ -533,68 +533,68 @@ end
 
 procedure map_exit_p_proc
 begin
-    if (((global_var(314) bwand 32768) != 0) or (global_var(494) == 8)) then begin
+    if (((global_var(GVAR_NEW_RENO_FLAG_1) bwand 32768) != 0) or (global_var(GVAR_FALLOUT_2) == 8)) then begin
         if (1 == 1) then begin
-            set_global_var(315, global_var(315) bwand (-1 - 128));
-            if (global_var(316) >= 2) then begin
-                set_global_var(316, 7);
+            set_global_var(GVAR_NEW_RENO_SALVATORE, global_var(GVAR_NEW_RENO_SALVATORE) bwand (-1 - 128));
+            if (global_var(GVAR_NEW_RENO_TRIBUTE) >= 2) then begin
+                set_global_var(GVAR_NEW_RENO_TRIBUTE, 7);
             end
-            if (global_var(312) >= 2) then begin
-                set_global_var(312, 4);
+            if (global_var(GVAR_NEW_RENO_TRACK_LLOYD) >= 2) then begin
+                set_global_var(GVAR_NEW_RENO_TRACK_LLOYD, 4);
             end
-            if (global_var(313) >= 2) then begin
-                set_global_var(313, 4);
+            if (global_var(GVAR_NEW_RENO_GUARD_ASSIGNMENT) >= 2) then begin
+                set_global_var(GVAR_NEW_RENO_GUARD_ASSIGNMENT, 4);
             end
         end
         else begin
             if (1 == 2) then begin
-                set_global_var(255, global_var(255) bwand (-1 - 128));
+                set_global_var(GVAR_NEW_RENO_BISHOP, global_var(GVAR_NEW_RENO_BISHOP) bwand (-1 - 128));
                 if (2 == 1) then begin
-                    if (global_var(548) == 0) then begin
-                        set_global_var(0, global_var(0) + -5);
+                    if (global_var(GVAR_NEW_RENO_CARLSON_SNUFF_PIP) == 0) then begin
+                        set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + -5);
                         debug_msg("Player gains " + -5 + " Karma Points.");
-                        set_global_var(37, 0);
-                        set_global_var(38, 0);
-                        set_global_var(39, 0);
-                        set_global_var(40, 0);
-                        set_global_var(41, 0);
-                        set_global_var(42, 0);
-                        set_global_var(43, 0);
-                        set_global_var(44, 0);
-                        set_global_var(45, 0);
-                        if (global_var(0) >= 1000) then begin
-                            set_global_var(37, 1);
+                        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+                        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+                        set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+                        set_global_var(GVAR_KARMA_DEFENDER, 0);
+                        set_global_var(GVAR_KARMA_WANDERER, 0);
+                        set_global_var(GVAR_KARMA_BETRAYER, 0);
+                        set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+                        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+                        set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+                        if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+                            set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
                         end
                         else begin
-                            if (global_var(0) >= 750) then begin
-                                set_global_var(38, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
                             end
                             else begin
-                                if (global_var(0) >= 500) then begin
-                                    set_global_var(39, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                                    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                                 end
                                 else begin
-                                    if (global_var(0) >= 250) then begin
-                                        set_global_var(40, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                                        set_global_var(GVAR_KARMA_DEFENDER, 1);
                                     end
                                     else begin
-                                        if (global_var(0) > -250) then begin
-                                            set_global_var(41, 1);
+                                        if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                                            set_global_var(GVAR_KARMA_WANDERER, 1);
                                         end
                                         else begin
-                                            if (global_var(0) > -500) then begin
-                                                set_global_var(42, 1);
+                                            if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                                set_global_var(GVAR_KARMA_BETRAYER, 1);
                                             end
                                             else begin
-                                                if (global_var(0) > -750) then begin
-                                                    set_global_var(43, 1);
+                                                if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                                    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                                 end
                                                 else begin
-                                                    if (global_var(0) > -1000) then begin
-                                                        set_global_var(44, 1);
+                                                    if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                                        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                                     end
                                                     else begin
-                                                        set_global_var(45, 1);
+                                                        set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                                     end
                                                 end
                                             end
@@ -604,61 +604,61 @@ begin
                             end
                         end
                     end
-                    set_global_var(175, global_var(175) bwor 1);
+                    set_global_var(GVAR_NEWRENO_SNUFF_CARLSON, global_var(GVAR_NEWRENO_SNUFF_CARLSON) bwor 1);
                 end
-                if ((2 == 2) and (global_var(548) == 0)) then begin
+                if ((2 == 2) and (global_var(GVAR_NEW_RENO_CARLSON_SNUFF_PIP) == 0)) then begin
                 end
                 else begin
-                    if (2 > global_var(548)) then begin
-                        set_global_var(548, 2);
+                    if (2 > global_var(GVAR_NEW_RENO_CARLSON_SNUFF_PIP)) then begin
+                        set_global_var(GVAR_NEW_RENO_CARLSON_SNUFF_PIP, 2);
                     end
                 end
                 if (2 == 1) then begin
-                    if (global_var(547) == 0) then begin
-                        set_global_var(0, global_var(0) + -5);
+                    if (global_var(GVAR_NEW_RENO_WESTIN_SNUFF_PIP) == 0) then begin
+                        set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + -5);
                         debug_msg("Player gains " + -5 + " Karma Points.");
-                        set_global_var(37, 0);
-                        set_global_var(38, 0);
-                        set_global_var(39, 0);
-                        set_global_var(40, 0);
-                        set_global_var(41, 0);
-                        set_global_var(42, 0);
-                        set_global_var(43, 0);
-                        set_global_var(44, 0);
-                        set_global_var(45, 0);
-                        if (global_var(0) >= 1000) then begin
-                            set_global_var(37, 1);
+                        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+                        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+                        set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+                        set_global_var(GVAR_KARMA_DEFENDER, 0);
+                        set_global_var(GVAR_KARMA_WANDERER, 0);
+                        set_global_var(GVAR_KARMA_BETRAYER, 0);
+                        set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+                        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+                        set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+                        if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+                            set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
                         end
                         else begin
-                            if (global_var(0) >= 750) then begin
-                                set_global_var(38, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
                             end
                             else begin
-                                if (global_var(0) >= 500) then begin
-                                    set_global_var(39, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                                    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                                 end
                                 else begin
-                                    if (global_var(0) >= 250) then begin
-                                        set_global_var(40, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                                        set_global_var(GVAR_KARMA_DEFENDER, 1);
                                     end
                                     else begin
-                                        if (global_var(0) > -250) then begin
-                                            set_global_var(41, 1);
+                                        if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                                            set_global_var(GVAR_KARMA_WANDERER, 1);
                                         end
                                         else begin
-                                            if (global_var(0) > -500) then begin
-                                                set_global_var(42, 1);
+                                            if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                                set_global_var(GVAR_KARMA_BETRAYER, 1);
                                             end
                                             else begin
-                                                if (global_var(0) > -750) then begin
-                                                    set_global_var(43, 1);
+                                                if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                                    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                                 end
                                                 else begin
-                                                    if (global_var(0) > -1000) then begin
-                                                        set_global_var(44, 1);
+                                                    if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                                        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                                     end
                                                     else begin
-                                                        set_global_var(45, 1);
+                                                        set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                                     end
                                                 end
                                             end
@@ -668,81 +668,81 @@ begin
                             end
                         end
                     end
-                    set_global_var(174, global_var(174) bwor 1);
+                    set_global_var(GVAR_NEWRENO_SNUFF_WESTIN, global_var(GVAR_NEWRENO_SNUFF_WESTIN) bwor 1);
                 end
-                if ((2 >= 2) and (global_var(547) == 0)) then begin
+                if ((2 >= 2) and (global_var(GVAR_NEW_RENO_WESTIN_SNUFF_PIP) == 0)) then begin
                 end
                 else begin
-                    if (2 > global_var(547)) then begin
-                        set_global_var(547, 2);
+                    if (2 > global_var(GVAR_NEW_RENO_WESTIN_SNUFF_PIP)) then begin
+                        set_global_var(GVAR_NEW_RENO_WESTIN_SNUFF_PIP, 2);
                     end
                 end
             end
             else begin
                 if (1 == 3) then begin
-                    if ((global_var(354) == 0) and (3 == 3)) then begin
+                    if ((global_var(GVAR_NEW_RENO_DELIVERY) == 0) and (3 == 3)) then begin
                     end
                     else begin
-                        if (3 > global_var(354)) then begin
-                            set_global_var(354, 3);
+                        if (3 > global_var(GVAR_NEW_RENO_DELIVERY)) then begin
+                            set_global_var(GVAR_NEW_RENO_DELIVERY, 3);
                         end
                     end
-                    debug_msg("delivery == " + global_var(354));
-                    if ((3 == 3) and (global_var(355) == 0)) then begin
+                    debug_msg("delivery == " + global_var(GVAR_NEW_RENO_DELIVERY));
+                    if ((3 == 3) and (global_var(GVAR_NEW_RENO_EXTORTION_BROS) == 0)) then begin
                     end
                     else begin
-                        set_global_var(355, 3);
+                        set_global_var(GVAR_NEW_RENO_EXTORTION_BROS, 3);
                     end
-                    if ((2 == 2) and (global_var(356) == 0)) then begin
+                    if ((2 == 2) and (global_var(GVAR_NEW_RENO_ASSASSINATION) == 0)) then begin
                     end
                     else begin
-                        set_global_var(356, 2);
+                        set_global_var(GVAR_NEW_RENO_ASSASSINATION, 2);
                     end
                 end
                 else begin
                     if (1 == 4) then begin
-                        if (0 > global_var(343)) then begin
-                            set_global_var(343, 0);
+                        if (0 > global_var(GVAR_NEW_RENO_SAD)) then begin
+                            set_global_var(GVAR_NEW_RENO_SAD, 0);
                         end
-                        if (global_var(286) == 1) then begin
+                        if (global_var(GVAR_NEW_RENO_WRIGHT_MYSTERY) == 1) then begin
                             if (2 == 3) then begin
-                                set_global_var(55, global_var(55) + 1);
-                                set_global_var(55, global_var(55) + 1);
+                                set_global_var(GVAR_TOWN_REP_NEW_RENO, global_var(GVAR_TOWN_REP_NEW_RENO) + 1);
+                                set_global_var(GVAR_TOWN_REP_NEW_RENO, global_var(GVAR_TOWN_REP_NEW_RENO) + 1);
                             end
                             else begin
                                 if (2 == 2) then begin
-                                    set_global_var(55, global_var(55) + 1);
+                                    set_global_var(GVAR_TOWN_REP_NEW_RENO, global_var(GVAR_TOWN_REP_NEW_RENO) + 1);
                                 end
                             end
                         end
-                        set_global_var(286, 2);
-                        set_global_var(285, global_var(285) bwand (-1 - 1));
+                        set_global_var(GVAR_NEW_RENO_WRIGHT_MYSTERY, 2);
+                        set_global_var(GVAR_NEW_RENO_WRIGHT_FLAGS, global_var(GVAR_NEW_RENO_WRIGHT_FLAGS) bwand (-1 - 1));
                     end
                 end
             end
         end
         set_global_var(251 + 1 - 1, global_var(251 + 1 - 1) bwor 1);
         if (1 != 0) then begin
-            if (1 == (global_var(230) / 100)) then begin
-                set_global_var(55, global_var(55) + 1);
+            if (1 == (global_var(GVAR_NEW_RENO_MADE_MAN) / 100)) then begin
+                set_global_var(GVAR_TOWN_REP_NEW_RENO, global_var(GVAR_TOWN_REP_NEW_RENO) + 1);
                 if (0 == 1) then begin
                     set_global_var(591 + 1 - 1, 1);
-                    set_global_var(230, (1 * 100) + (global_var(230) % 100));
+                    set_global_var(GVAR_NEW_RENO_MADE_MAN, (1 * 100) + (global_var(GVAR_NEW_RENO_MADE_MAN) % 100));
                 end
                 else begin
                     if (0 == 2) then begin
                         set_global_var(591 + 2 - 1, 1);
-                        set_global_var(230, (2 * 100) + (global_var(230) % 100));
+                        set_global_var(GVAR_NEW_RENO_MADE_MAN, (2 * 100) + (global_var(GVAR_NEW_RENO_MADE_MAN) % 100));
                     end
                     else begin
                         if (0 == 3) then begin
                             set_global_var(591 + 3 - 1, 1);
-                            set_global_var(230, (3 * 100) + (global_var(230) % 100));
+                            set_global_var(GVAR_NEW_RENO_MADE_MAN, (3 * 100) + (global_var(GVAR_NEW_RENO_MADE_MAN) % 100));
                         end
                         else begin
                             if (0 == 4) then begin
                                 set_global_var(591 + 4 - 1, 1);
-                                set_global_var(230, (4 * 100) + (global_var(230) % 100));
+                                set_global_var(GVAR_NEW_RENO_MADE_MAN, (4 * 100) + (global_var(GVAR_NEW_RENO_MADE_MAN) % 100));
                             end
                             else begin
                                 if (0 == 0) then begin
@@ -750,7 +750,7 @@ begin
                                     set_global_var(591 + 2 - 1, 0);
                                     set_global_var(591 + 3 - 1, 0);
                                     set_global_var(591 + 4 - 1, 0);
-                                    set_global_var(230, (0 * 100) + 0);
+                                    set_global_var(GVAR_NEW_RENO_MADE_MAN, (0 * 100) + 0);
                                 end
                             end
                         end
@@ -758,15 +758,15 @@ begin
                 end
             end
         end
-        debug_msg("called attack family(" + message_str(353, 200 + 1) + ")");
+        debug_msg("called attack family(" + message_str(SCRIPT_NEWR1, 200 + 1) + ")");
     end
 end
 
 procedure timed_event_p_proc
 begin
-    debug_msg("secret trade event: name == " + obj_name(self_obj) + " / fixed == " + fixed_param);
+    debug_msg("secret trade event: name == " + self_name + " / fixed == " + fixed_param);
     if (fixed_param == 1) then begin
-        float_msg(self_obj, " ", 8);
+        float_msg(self_obj, " ", FLOAT_MSG_YELLOW);
     end
     else begin
         if (fixed_param == 23) then begin
@@ -791,80 +791,80 @@ begin
         end
         else begin
             if (fixed_param >= 350) then begin
-                float_msg(self_obj, message_str(1032, fixed_param), 8);
+                float_msg(self_obj, message_str(SCRIPT_NEWRVB, fixed_param), FLOAT_MSG_YELLOW);
                 if (obj_can_see_obj(self_obj, dude_obj) == 0) then begin
-                    anim(dude_obj, 1000, rotation_to_tile(tile_num(dude_obj), tile_num(self_obj)));
+                    anim(dude_obj, 1000, rotation_to_tile(dude_tile, tile_num(self_obj)));
                 end
                 if (fixed_param == 350) then begin
-                    set_global_var(313, 4);
+                    set_global_var(GVAR_NEW_RENO_GUARD_ASSIGNMENT, 4);
                     if (salvatore_guard_2_obj != 0) then begin
-                        metarule3(100, salvatore_guard_2_obj, 351, 0);
+                        rm_fixed_timer_event(salvatore_guard_2_obj, 351);
                         add_timer_event(salvatore_guard_2_obj, 10 * 3, 351);
                     end
-                    metarule3(100, self_obj, 1, 0);
+                    rm_fixed_timer_event(self_obj, 1);
                     add_timer_event(self_obj, (10 * 3) - 5, 1);
                 end
                 else begin
                     if (fixed_param == 351) then begin
                         if (power_armor_1_obj != 0) then begin
-                            metarule3(100, power_armor_1_obj, 352, 0);
+                            rm_fixed_timer_event(power_armor_1_obj, 352);
                             add_timer_event(power_armor_1_obj, 10 * 3, 352);
                         end
-                        metarule3(100, self_obj, 1, 0);
+                        rm_fixed_timer_event(self_obj, 1);
                         add_timer_event(self_obj, (10 * 3) - 5, 1);
                     end
                     else begin
                         if (fixed_param == 352) then begin
                             if (power_armor_2_obj != 0) then begin
-                                metarule3(100, power_armor_2_obj, 353, 0);
+                                rm_fixed_timer_event(power_armor_2_obj, 353);
                                 add_timer_event(power_armor_2_obj, 10 * 3, 353);
                             end
-                            metarule3(100, self_obj, 1, 0);
+                            rm_fixed_timer_event(self_obj, 1);
                             add_timer_event(self_obj, (10 * 3) - 5, 1);
                         end
                         else begin
                             if (fixed_param == 353) then begin
                                 debug_msg("helicopter_script_node998");
                                 set_local_var(0, local_var(0) bwor 2);
-                                if ((global_var(313) == 0) == 0) then begin
-                                    set_global_var(313, 4);
+                                if ((global_var(GVAR_NEW_RENO_GUARD_ASSIGNMENT) == 0) == 0) then begin
+                                    set_global_var(GVAR_NEW_RENO_GUARD_ASSIGNMENT, 4);
                                 end
-                                if ((((global_var(314) bwand 32768) != 0) or (global_var(494) == 8)) == 0) then begin
-                                    if ((((global_var(314) bwand 32768) != 0) or (global_var(494) == 8)) == 0) then begin
-                                        if (((1 == 1) and (global_var(134) <= -1) or ((1 == 2) and (global_var(135) <= -1)) or ((1 == 3) and (global_var(136) <= -1)) or ((1 == 4) and (global_var(216) <= -1))) == 0) then begin
+                                if ((((global_var(GVAR_NEW_RENO_FLAG_1) bwand 32768) != 0) or (global_var(GVAR_FALLOUT_2) == 8)) == 0) then begin
+                                    if ((((global_var(GVAR_NEW_RENO_FLAG_1) bwand 32768) != 0) or (global_var(GVAR_FALLOUT_2) == 8)) == 0) then begin
+                                        if (((1 == 1) and (global_var(GVAR_SALVATORE_FAMILY_COUNTER) <= -1) or ((1 == 2) and (global_var(GVAR_BISHOP_FAMILY_COUNTER) <= -1)) or ((1 == 3) and (global_var(GVAR_MORDINO_FAMILY_COUNTER) <= -1)) or ((1 == 4) and (global_var(GVAR_WRIGHT_FAMILY_COUNTER) <= -1))) == 0) then begin
                                             if (1 == 1) then begin
-                                                set_global_var(134, global_var(134) - 20);
+                                                set_global_var(GVAR_SALVATORE_FAMILY_COUNTER, global_var(GVAR_SALVATORE_FAMILY_COUNTER) - 20);
                                             end
                                             else begin
                                                 if (1 == 2) then begin
-                                                    set_global_var(135, global_var(135) - 20);
+                                                    set_global_var(GVAR_BISHOP_FAMILY_COUNTER, global_var(GVAR_BISHOP_FAMILY_COUNTER) - 20);
                                                 end
                                                 else begin
                                                     if (1 == 3) then begin
-                                                        set_global_var(136, global_var(136) - 20);
+                                                        set_global_var(GVAR_MORDINO_FAMILY_COUNTER, global_var(GVAR_MORDINO_FAMILY_COUNTER) - 20);
                                                     end
                                                     else begin
                                                         if (1 == 4) then begin
-                                                            set_global_var(216, global_var(216) - 20);
+                                                            set_global_var(GVAR_WRIGHT_FAMILY_COUNTER, global_var(GVAR_WRIGHT_FAMILY_COUNTER) - 20);
                                                         end
                                                     end
                                                 end
                                             end
                                         end
                                         debug_msg("\nfamily standings: ");
-                                        debug_msg("\n  Salvatore == " + global_var(134));
-                                        debug_msg("\n  Bishop == " + global_var(135));
-                                        debug_msg("\n  Mordino == " + global_var(136));
-                                        debug_msg("\n  Wright == " + global_var(216));
+                                        debug_msg("\n  Salvatore == " + global_var(GVAR_SALVATORE_FAMILY_COUNTER));
+                                        debug_msg("\n  Bishop == " + global_var(GVAR_BISHOP_FAMILY_COUNTER));
+                                        debug_msg("\n  Mordino == " + global_var(GVAR_MORDINO_FAMILY_COUNTER));
+                                        debug_msg("\n  Wright == " + global_var(GVAR_WRIGHT_FAMILY_COUNTER));
                                     end
-                                    set_global_var(314, global_var(314) bwor 32768);
-                                    if (global_var(313) == 0) then begin
-                                        metarule3(100, self_obj, 250, 0);
+                                    set_global_var(GVAR_NEW_RENO_FLAG_1, global_var(GVAR_NEW_RENO_FLAG_1) bwor 32768);
+                                    if (global_var(GVAR_NEW_RENO_GUARD_ASSIGNMENT) == 0) then begin
+                                        rm_fixed_timer_event(self_obj, 250);
                                         add_timer_event(self_obj, 0, 250);
                                     end
                                 end
                                 if (combat_is_initialized == 0) then begin
-                                    attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+                                    attack(dude_obj);
                                 end
                             end
                         end
@@ -872,65 +872,65 @@ begin
                 end
                 debug_msg("helicopter_script_node998");
                 set_local_var(0, local_var(0) bwor 2);
-                if ((global_var(313) == 0) == 0) then begin
-                    set_global_var(313, 4);
+                if ((global_var(GVAR_NEW_RENO_GUARD_ASSIGNMENT) == 0) == 0) then begin
+                    set_global_var(GVAR_NEW_RENO_GUARD_ASSIGNMENT, 4);
                 end
-                if ((((global_var(314) bwand 32768) != 0) or (global_var(494) == 8)) == 0) then begin
-                    if ((((global_var(314) bwand 32768) != 0) or (global_var(494) == 8)) == 0) then begin
-                        if (((1 == 1) and (global_var(134) <= -1) or ((1 == 2) and (global_var(135) <= -1)) or ((1 == 3) and (global_var(136) <= -1)) or ((1 == 4) and (global_var(216) <= -1))) == 0) then begin
+                if ((((global_var(GVAR_NEW_RENO_FLAG_1) bwand 32768) != 0) or (global_var(GVAR_FALLOUT_2) == 8)) == 0) then begin
+                    if ((((global_var(GVAR_NEW_RENO_FLAG_1) bwand 32768) != 0) or (global_var(GVAR_FALLOUT_2) == 8)) == 0) then begin
+                        if (((1 == 1) and (global_var(GVAR_SALVATORE_FAMILY_COUNTER) <= -1) or ((1 == 2) and (global_var(GVAR_BISHOP_FAMILY_COUNTER) <= -1)) or ((1 == 3) and (global_var(GVAR_MORDINO_FAMILY_COUNTER) <= -1)) or ((1 == 4) and (global_var(GVAR_WRIGHT_FAMILY_COUNTER) <= -1))) == 0) then begin
                             if (1 == 1) then begin
-                                set_global_var(134, global_var(134) - 20);
+                                set_global_var(GVAR_SALVATORE_FAMILY_COUNTER, global_var(GVAR_SALVATORE_FAMILY_COUNTER) - 20);
                             end
                             else begin
                                 if (1 == 2) then begin
-                                    set_global_var(135, global_var(135) - 20);
+                                    set_global_var(GVAR_BISHOP_FAMILY_COUNTER, global_var(GVAR_BISHOP_FAMILY_COUNTER) - 20);
                                 end
                                 else begin
                                     if (1 == 3) then begin
-                                        set_global_var(136, global_var(136) - 20);
+                                        set_global_var(GVAR_MORDINO_FAMILY_COUNTER, global_var(GVAR_MORDINO_FAMILY_COUNTER) - 20);
                                     end
                                     else begin
                                         if (1 == 4) then begin
-                                            set_global_var(216, global_var(216) - 20);
+                                            set_global_var(GVAR_WRIGHT_FAMILY_COUNTER, global_var(GVAR_WRIGHT_FAMILY_COUNTER) - 20);
                                         end
                                     end
                                 end
                             end
                         end
                         debug_msg("\nfamily standings: ");
-                        debug_msg("\n  Salvatore == " + global_var(134));
-                        debug_msg("\n  Bishop == " + global_var(135));
-                        debug_msg("\n  Mordino == " + global_var(136));
-                        debug_msg("\n  Wright == " + global_var(216));
+                        debug_msg("\n  Salvatore == " + global_var(GVAR_SALVATORE_FAMILY_COUNTER));
+                        debug_msg("\n  Bishop == " + global_var(GVAR_BISHOP_FAMILY_COUNTER));
+                        debug_msg("\n  Mordino == " + global_var(GVAR_MORDINO_FAMILY_COUNTER));
+                        debug_msg("\n  Wright == " + global_var(GVAR_WRIGHT_FAMILY_COUNTER));
                     end
-                    set_global_var(314, global_var(314) bwor 32768);
-                    if (global_var(313) == 0) then begin
-                        metarule3(100, self_obj, 250, 0);
+                    set_global_var(GVAR_NEW_RENO_FLAG_1, global_var(GVAR_NEW_RENO_FLAG_1) bwor 32768);
+                    if (global_var(GVAR_NEW_RENO_GUARD_ASSIGNMENT) == 0) then begin
+                        rm_fixed_timer_event(self_obj, 250);
                         add_timer_event(self_obj, 0, 250);
                     end
                 end
                 if (combat_is_initialized == 0) then begin
-                    attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+                    attack(dude_obj);
                 end
             end
             else begin
-                if (((((global_var(314) bwand 32768) != 0) or (global_var(494) == 8)) == 0) and ((global_var(313) == 4) == 0)) then begin
+                if (((((global_var(GVAR_NEW_RENO_FLAG_1) bwand 32768) != 0) or (global_var(GVAR_FALLOUT_2) == 8)) == 0) and ((global_var(GVAR_NEW_RENO_GUARD_ASSIGNMENT) == 4) == 0)) then begin
                     GVar27 := 1;
                     if (fixed_param == 200) then begin
                         if (salvatore_guard_2_obj != 0) then begin
-                            metarule3(100, salvatore_guard_2_obj, 201, 0);
+                            rm_fixed_timer_event(salvatore_guard_2_obj, 201);
                             add_timer_event(salvatore_guard_2_obj, 10 * 3, 201);
                         end
-                        metarule3(100, self_obj, 1, 0);
+                        rm_fixed_timer_event(self_obj, 1);
                         add_timer_event(self_obj, (10 * 3) - 5, 1);
                     end
                     else begin
                         if (fixed_param == 201) then begin
                             if (power_armor_1_obj != 0) then begin
-                                metarule3(100, power_armor_1_obj, 202, 0);
+                                rm_fixed_timer_event(power_armor_1_obj, 202);
                                 add_timer_event(power_armor_1_obj, 10 * 3, 202);
                             end
-                            metarule3(100, self_obj, 1, 0);
+                            rm_fixed_timer_event(self_obj, 1);
                             add_timer_event(self_obj, (10 * 3) - 5, 1);
                         end
                         else begin
@@ -953,124 +953,124 @@ begin
                                 end
                                 if (GVar27) then begin
                                     if (salvatore_guard_2_obj != 0) then begin
-                                        metarule3(100, salvatore_guard_2_obj, 203, 0);
+                                        rm_fixed_timer_event(salvatore_guard_2_obj, 203);
                                         add_timer_event(salvatore_guard_2_obj, 10 * 5, 203);
                                     end
-                                    metarule3(100, self_obj, 1, 0);
+                                    rm_fixed_timer_event(self_obj, 1);
                                     add_timer_event(self_obj, (10 * 5) - 5, 1);
                                 end
                             end
                             else begin
                                 if (fixed_param == 203) then begin
                                     if (power_armor_1_obj != 0) then begin
-                                        metarule3(100, power_armor_1_obj, 204, 0);
+                                        rm_fixed_timer_event(power_armor_1_obj, 204);
                                         add_timer_event(power_armor_1_obj, 10 * 5, 204);
                                     end
-                                    metarule3(100, self_obj, 1, 0);
+                                    rm_fixed_timer_event(self_obj, 1);
                                     add_timer_event(self_obj, (10 * 5) - 5, 1);
                                 end
                                 else begin
                                     if (fixed_param == 204) then begin
                                         if (power_armor_2_obj != 0) then begin
-                                            metarule3(100, power_armor_2_obj, 205, 0);
+                                            rm_fixed_timer_event(power_armor_2_obj, 205);
                                             add_timer_event(power_armor_2_obj, 10 * 5, 205);
                                         end
-                                        metarule3(100, self_obj, 1, 0);
+                                        rm_fixed_timer_event(self_obj, 1);
                                         add_timer_event(self_obj, (10 * 5) - 5, 1);
                                     end
                                     else begin
                                         if (fixed_param == 205) then begin
                                             if (salvatore_guard_2_obj != 0) then begin
-                                                metarule3(100, salvatore_guard_2_obj, 206, 0);
+                                                rm_fixed_timer_event(salvatore_guard_2_obj, 206);
                                                 add_timer_event(salvatore_guard_2_obj, 10 * 7, 206);
                                             end
-                                            metarule3(100, self_obj, 1, 0);
+                                            rm_fixed_timer_event(self_obj, 1);
                                             add_timer_event(self_obj, (10 * 7) - 5, 1);
                                         end
                                         else begin
                                             if (fixed_param == 206) then begin
                                                 if (power_armor_1_obj != 0) then begin
-                                                    metarule3(100, power_armor_1_obj, 207, 0);
+                                                    rm_fixed_timer_event(power_armor_1_obj, 207);
                                                     add_timer_event(power_armor_1_obj, 10 * 5, 207);
                                                 end
-                                                metarule3(100, self_obj, 1, 0);
+                                                rm_fixed_timer_event(self_obj, 1);
                                                 add_timer_event(self_obj, (10 * 5) - 5, 1);
                                             end
                                             else begin
                                                 if (fixed_param == 207) then begin
                                                     if (power_armor_2_obj != 0) then begin
-                                                        metarule3(100, power_armor_2_obj, 208, 0);
+                                                        rm_fixed_timer_event(power_armor_2_obj, 208);
                                                         add_timer_event(power_armor_2_obj, 10 * 5, 208);
                                                     end
-                                                    metarule3(100, self_obj, 1, 0);
+                                                    rm_fixed_timer_event(self_obj, 1);
                                                     add_timer_event(self_obj, (10 * 5) - 5, 1);
                                                 end
                                                 else begin
                                                     if (fixed_param == 208) then begin
                                                         if (salvatore_guard_2_obj != 0) then begin
-                                                            metarule3(100, salvatore_guard_2_obj, 209, 0);
+                                                            rm_fixed_timer_event(salvatore_guard_2_obj, 209);
                                                             add_timer_event(salvatore_guard_2_obj, 10 * 7, 209);
                                                         end
-                                                        metarule3(100, self_obj, 1, 0);
+                                                        rm_fixed_timer_event(self_obj, 1);
                                                         add_timer_event(self_obj, (10 * 7) - 5, 1);
                                                     end
                                                     else begin
                                                         if (fixed_param == 209) then begin
                                                             if (power_armor_1_obj != 0) then begin
-                                                                metarule3(100, power_armor_1_obj, 210, 0);
+                                                                rm_fixed_timer_event(power_armor_1_obj, 210);
                                                                 add_timer_event(power_armor_1_obj, 10 * 5, 210);
                                                             end
-                                                            metarule3(100, self_obj, 1, 0);
+                                                            rm_fixed_timer_event(self_obj, 1);
                                                             add_timer_event(self_obj, (10 * 5) - 5, 1);
                                                         end
                                                         else begin
                                                             if (fixed_param == 210) then begin
                                                                 if (power_armor_2_obj != 0) then begin
-                                                                    metarule3(100, power_armor_2_obj, 211, 0);
+                                                                    rm_fixed_timer_event(power_armor_2_obj, 211);
                                                                     add_timer_event(power_armor_2_obj, 10 * 3, 211);
                                                                 end
-                                                                metarule3(100, self_obj, 1, 0);
+                                                                rm_fixed_timer_event(self_obj, 1);
                                                                 add_timer_event(self_obj, (10 * 3) - 5, 1);
                                                                 GVar19 := handy_obj;
                                                                 GVar20 := GVar20;
-                                                                metarule3(100, handy_obj, 0, 0);
+                                                                rm_fixed_timer_event(handy_obj, 0);
                                                                 add_timer_event(handy_obj, 5, 0);
                                                             end
                                                             else begin
                                                                 if (fixed_param == 211) then begin
                                                                     if (handy_obj != 0) then begin
-                                                                        metarule3(100, handy_obj, 1, 0);
+                                                                        rm_fixed_timer_event(handy_obj, 1);
                                                                         add_timer_event(handy_obj, 0, 1);
                                                                     end
                                                                     if (salvatore_guard_2_obj != 0) then begin
-                                                                        metarule3(100, salvatore_guard_2_obj, 212, 0);
+                                                                        rm_fixed_timer_event(salvatore_guard_2_obj, 212);
                                                                         add_timer_event(salvatore_guard_2_obj, 10 * 3, 212);
                                                                     end
-                                                                    metarule3(100, self_obj, 1, 0);
+                                                                    rm_fixed_timer_event(self_obj, 1);
                                                                     add_timer_event(self_obj, (10 * 3) - 5, 1);
                                                                 end
                                                                 else begin
                                                                     if (fixed_param == 212) then begin
                                                                         if (salvatore_guard_2_obj != 0) then begin
-                                                                            metarule3(100, salvatore_guard_2_obj, 213, 0);
+                                                                            rm_fixed_timer_event(salvatore_guard_2_obj, 213);
                                                                             add_timer_event(salvatore_guard_2_obj, 10 * 7, 213);
                                                                         end
-                                                                        metarule3(100, self_obj, 1, 0);
+                                                                        rm_fixed_timer_event(self_obj, 1);
                                                                         add_timer_event(self_obj, (10 * 7) - 5, 1);
                                                                     end
                                                                     else begin
                                                                         if (fixed_param == 213) then begin
-                                                                            set_global_var(345, global_var(345) bwor 128);
-                                                                            if ((global_var(313) == 0) == 0) then begin
-                                                                                set_global_var(313, 3);
+                                                                            set_global_var(GVAR_NEW_RENO_FLAG_2, global_var(GVAR_NEW_RENO_FLAG_2) bwor 128);
+                                                                            if ((global_var(GVAR_NEW_RENO_GUARD_ASSIGNMENT) == 0) == 0) then begin
+                                                                                set_global_var(GVAR_NEW_RENO_GUARD_ASSIGNMENT, 3);
                                                                             end
                                                                             else begin
-                                                                                if (((global_var(495) bwand --2147483648) != 0) == 0) then begin
-                                                                                    display_msg(message_str(353, 1550));
+                                                                                if (((global_var(GVAR_NEW_RENO_FLAG_3) bwand --2147483648) != 0) == 0) then begin
+                                                                                    display_msg(message_str(SCRIPT_NEWR1, 1550));
                                                                                     give_exp_points(1500);
-                                                                                    display_msg(message_str(14, 100) + 1500 + message_str(14, 101));
+                                                                                    display_msg(g_mstr(100) + 1500 + g_mstr(101));
                                                                                 end
-                                                                                set_global_var(495, global_var(495) bwor --2147483648);
+                                                                                set_global_var(GVAR_NEW_RENO_FLAG_3, global_var(GVAR_NEW_RENO_FLAG_3) bwor --2147483648);
                                                                             end
                                                                         end
                                                                     end
@@ -1087,125 +1087,125 @@ begin
                         end
                     end
                     if (GVar27) then begin
-                        float_msg(self_obj, message_str(1032, fixed_param), 8);
+                        float_msg(self_obj, message_str(SCRIPT_NEWRVB, fixed_param), FLOAT_MSG_YELLOW);
                     end
                     else begin
-                        metarule3(100, self_obj, fixed_param, 0);
+                        rm_fixed_timer_event(self_obj, fixed_param);
                         add_timer_event(self_obj, 10, fixed_param);
                     end
                 end
                 else begin
-                    float_msg(self_obj, message_str(1032, fixed_param), 8);
+                    float_msg(self_obj, message_str(SCRIPT_NEWRVB, fixed_param), FLOAT_MSG_YELLOW);
                     if (fixed_param == 250) then begin
                         if (salvatore_guard_1_obj != 0) then begin
-                            metarule3(100, salvatore_guard_1_obj, 251, 0);
+                            rm_fixed_timer_event(salvatore_guard_1_obj, 251);
                             add_timer_event(salvatore_guard_1_obj, 10 * 3, 251);
                         end
-                        metarule3(100, self_obj, 1, 0);
+                        rm_fixed_timer_event(self_obj, 1);
                         add_timer_event(self_obj, (10 * 3) - 5, 1);
                     end
                     else begin
                         if (fixed_param == 251) then begin
                             if (salvatore_guard_2_obj != 0) then begin
-                                metarule3(100, salvatore_guard_2_obj, 252, 0);
+                                rm_fixed_timer_event(salvatore_guard_2_obj, 252);
                                 add_timer_event(salvatore_guard_2_obj, 10 * 3, 252);
                             end
-                            metarule3(100, self_obj, 1, 0);
+                            rm_fixed_timer_event(self_obj, 1);
                             add_timer_event(self_obj, (10 * 3) - 5, 1);
                         end
                         else begin
                             if (fixed_param == 252) then begin
                                 if (power_armor_1_obj != 0) then begin
-                                    metarule3(100, power_armor_1_obj, 253, 0);
+                                    rm_fixed_timer_event(power_armor_1_obj, 253);
                                     add_timer_event(power_armor_1_obj, 10 * 3, 253);
                                 end
-                                metarule3(100, self_obj, 1, 0);
+                                rm_fixed_timer_event(self_obj, 1);
                                 add_timer_event(self_obj, (10 * 3) - 5, 1);
                             end
                             else begin
                                 if (fixed_param == 253) then begin
                                     if (power_armor_2_obj != 0) then begin
-                                        metarule3(100, power_armor_2_obj, 254, 0);
+                                        rm_fixed_timer_event(power_armor_2_obj, 254);
                                         add_timer_event(power_armor_2_obj, 10 * 3, 254);
                                     end
-                                    metarule3(100, self_obj, 1, 0);
+                                    rm_fixed_timer_event(self_obj, 1);
                                     add_timer_event(self_obj, (10 * 3) - 5, 1);
                                 end
                                 else begin
                                     if (fixed_param >= 350) then begin
                                         if (obj_can_see_obj(self_obj, dude_obj) == 0) then begin
-                                            anim(dude_obj, 1000, rotation_to_tile(tile_num(dude_obj), tile_num(self_obj)));
+                                            anim(dude_obj, 1000, rotation_to_tile(dude_tile, tile_num(self_obj)));
                                         end
                                         if (fixed_param == 350) then begin
-                                            set_global_var(313, 4);
+                                            set_global_var(GVAR_NEW_RENO_GUARD_ASSIGNMENT, 4);
                                             if (salvatore_guard_2_obj != 0) then begin
-                                                metarule3(100, salvatore_guard_2_obj, 351, 0);
+                                                rm_fixed_timer_event(salvatore_guard_2_obj, 351);
                                                 add_timer_event(salvatore_guard_2_obj, 10 * 3, 351);
                                             end
-                                            metarule3(100, self_obj, 1, 0);
+                                            rm_fixed_timer_event(self_obj, 1);
                                             add_timer_event(self_obj, (10 * 3) - 5, 1);
                                         end
                                         else begin
                                             if (fixed_param == 351) then begin
                                                 if (power_armor_1_obj != 0) then begin
-                                                    metarule3(100, power_armor_1_obj, 352, 0);
+                                                    rm_fixed_timer_event(power_armor_1_obj, 352);
                                                     add_timer_event(power_armor_1_obj, 10 * 3, 352);
                                                 end
-                                                metarule3(100, self_obj, 1, 0);
+                                                rm_fixed_timer_event(self_obj, 1);
                                                 add_timer_event(self_obj, (10 * 3) - 5, 1);
                                             end
                                             else begin
                                                 if (fixed_param == 352) then begin
                                                     if (power_armor_2_obj != 0) then begin
-                                                        metarule3(100, power_armor_2_obj, 353, 0);
+                                                        rm_fixed_timer_event(power_armor_2_obj, 353);
                                                         add_timer_event(power_armor_2_obj, 10 * 3, 353);
                                                     end
-                                                    metarule3(100, self_obj, 1, 0);
+                                                    rm_fixed_timer_event(self_obj, 1);
                                                     add_timer_event(self_obj, (10 * 3) - 5, 1);
                                                 end
                                                 else begin
                                                     if (fixed_param == 353) then begin
                                                         debug_msg("helicopter_script_node998");
                                                         set_local_var(0, local_var(0) bwor 2);
-                                                        if ((global_var(313) == 0) == 0) then begin
-                                                            set_global_var(313, 4);
+                                                        if ((global_var(GVAR_NEW_RENO_GUARD_ASSIGNMENT) == 0) == 0) then begin
+                                                            set_global_var(GVAR_NEW_RENO_GUARD_ASSIGNMENT, 4);
                                                         end
-                                                        if ((((global_var(314) bwand 32768) != 0) or (global_var(494) == 8)) == 0) then begin
-                                                            if ((((global_var(314) bwand 32768) != 0) or (global_var(494) == 8)) == 0) then begin
-                                                                if (((1 == 1) and (global_var(134) <= -1) or ((1 == 2) and (global_var(135) <= -1)) or ((1 == 3) and (global_var(136) <= -1)) or ((1 == 4) and (global_var(216) <= -1))) == 0) then begin
+                                                        if ((((global_var(GVAR_NEW_RENO_FLAG_1) bwand 32768) != 0) or (global_var(GVAR_FALLOUT_2) == 8)) == 0) then begin
+                                                            if ((((global_var(GVAR_NEW_RENO_FLAG_1) bwand 32768) != 0) or (global_var(GVAR_FALLOUT_2) == 8)) == 0) then begin
+                                                                if (((1 == 1) and (global_var(GVAR_SALVATORE_FAMILY_COUNTER) <= -1) or ((1 == 2) and (global_var(GVAR_BISHOP_FAMILY_COUNTER) <= -1)) or ((1 == 3) and (global_var(GVAR_MORDINO_FAMILY_COUNTER) <= -1)) or ((1 == 4) and (global_var(GVAR_WRIGHT_FAMILY_COUNTER) <= -1))) == 0) then begin
                                                                     if (1 == 1) then begin
-                                                                        set_global_var(134, global_var(134) - 20);
+                                                                        set_global_var(GVAR_SALVATORE_FAMILY_COUNTER, global_var(GVAR_SALVATORE_FAMILY_COUNTER) - 20);
                                                                     end
                                                                     else begin
                                                                         if (1 == 2) then begin
-                                                                            set_global_var(135, global_var(135) - 20);
+                                                                            set_global_var(GVAR_BISHOP_FAMILY_COUNTER, global_var(GVAR_BISHOP_FAMILY_COUNTER) - 20);
                                                                         end
                                                                         else begin
                                                                             if (1 == 3) then begin
-                                                                                set_global_var(136, global_var(136) - 20);
+                                                                                set_global_var(GVAR_MORDINO_FAMILY_COUNTER, global_var(GVAR_MORDINO_FAMILY_COUNTER) - 20);
                                                                             end
                                                                             else begin
                                                                                 if (1 == 4) then begin
-                                                                                    set_global_var(216, global_var(216) - 20);
+                                                                                    set_global_var(GVAR_WRIGHT_FAMILY_COUNTER, global_var(GVAR_WRIGHT_FAMILY_COUNTER) - 20);
                                                                                 end
                                                                             end
                                                                         end
                                                                     end
                                                                 end
                                                                 debug_msg("\nfamily standings: ");
-                                                                debug_msg("\n  Salvatore == " + global_var(134));
-                                                                debug_msg("\n  Bishop == " + global_var(135));
-                                                                debug_msg("\n  Mordino == " + global_var(136));
-                                                                debug_msg("\n  Wright == " + global_var(216));
+                                                                debug_msg("\n  Salvatore == " + global_var(GVAR_SALVATORE_FAMILY_COUNTER));
+                                                                debug_msg("\n  Bishop == " + global_var(GVAR_BISHOP_FAMILY_COUNTER));
+                                                                debug_msg("\n  Mordino == " + global_var(GVAR_MORDINO_FAMILY_COUNTER));
+                                                                debug_msg("\n  Wright == " + global_var(GVAR_WRIGHT_FAMILY_COUNTER));
                                                             end
-                                                            set_global_var(314, global_var(314) bwor 32768);
-                                                            if (global_var(313) == 0) then begin
-                                                                metarule3(100, self_obj, 250, 0);
+                                                            set_global_var(GVAR_NEW_RENO_FLAG_1, global_var(GVAR_NEW_RENO_FLAG_1) bwor 32768);
+                                                            if (global_var(GVAR_NEW_RENO_GUARD_ASSIGNMENT) == 0) then begin
+                                                                rm_fixed_timer_event(self_obj, 250);
                                                                 add_timer_event(self_obj, 0, 250);
                                                             end
                                                         end
                                                         if (combat_is_initialized == 0) then begin
-                                                            attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+                                                            attack(dude_obj);
                                                         end
                                                     end
                                                 end
@@ -1213,52 +1213,52 @@ begin
                                         end
                                         debug_msg("helicopter_script_node998");
                                         set_local_var(0, local_var(0) bwor 2);
-                                        if ((global_var(313) == 0) == 0) then begin
-                                            set_global_var(313, 4);
+                                        if ((global_var(GVAR_NEW_RENO_GUARD_ASSIGNMENT) == 0) == 0) then begin
+                                            set_global_var(GVAR_NEW_RENO_GUARD_ASSIGNMENT, 4);
                                         end
-                                        if ((((global_var(314) bwand 32768) != 0) or (global_var(494) == 8)) == 0) then begin
-                                            if ((((global_var(314) bwand 32768) != 0) or (global_var(494) == 8)) == 0) then begin
-                                                if (((1 == 1) and (global_var(134) <= -1) or ((1 == 2) and (global_var(135) <= -1)) or ((1 == 3) and (global_var(136) <= -1)) or ((1 == 4) and (global_var(216) <= -1))) == 0) then begin
+                                        if ((((global_var(GVAR_NEW_RENO_FLAG_1) bwand 32768) != 0) or (global_var(GVAR_FALLOUT_2) == 8)) == 0) then begin
+                                            if ((((global_var(GVAR_NEW_RENO_FLAG_1) bwand 32768) != 0) or (global_var(GVAR_FALLOUT_2) == 8)) == 0) then begin
+                                                if (((1 == 1) and (global_var(GVAR_SALVATORE_FAMILY_COUNTER) <= -1) or ((1 == 2) and (global_var(GVAR_BISHOP_FAMILY_COUNTER) <= -1)) or ((1 == 3) and (global_var(GVAR_MORDINO_FAMILY_COUNTER) <= -1)) or ((1 == 4) and (global_var(GVAR_WRIGHT_FAMILY_COUNTER) <= -1))) == 0) then begin
                                                     if (1 == 1) then begin
-                                                        set_global_var(134, global_var(134) - 20);
+                                                        set_global_var(GVAR_SALVATORE_FAMILY_COUNTER, global_var(GVAR_SALVATORE_FAMILY_COUNTER) - 20);
                                                     end
                                                     else begin
                                                         if (1 == 2) then begin
-                                                            set_global_var(135, global_var(135) - 20);
+                                                            set_global_var(GVAR_BISHOP_FAMILY_COUNTER, global_var(GVAR_BISHOP_FAMILY_COUNTER) - 20);
                                                         end
                                                         else begin
                                                             if (1 == 3) then begin
-                                                                set_global_var(136, global_var(136) - 20);
+                                                                set_global_var(GVAR_MORDINO_FAMILY_COUNTER, global_var(GVAR_MORDINO_FAMILY_COUNTER) - 20);
                                                             end
                                                             else begin
                                                                 if (1 == 4) then begin
-                                                                    set_global_var(216, global_var(216) - 20);
+                                                                    set_global_var(GVAR_WRIGHT_FAMILY_COUNTER, global_var(GVAR_WRIGHT_FAMILY_COUNTER) - 20);
                                                                 end
                                                             end
                                                         end
                                                     end
                                                 end
                                                 debug_msg("\nfamily standings: ");
-                                                debug_msg("\n  Salvatore == " + global_var(134));
-                                                debug_msg("\n  Bishop == " + global_var(135));
-                                                debug_msg("\n  Mordino == " + global_var(136));
-                                                debug_msg("\n  Wright == " + global_var(216));
+                                                debug_msg("\n  Salvatore == " + global_var(GVAR_SALVATORE_FAMILY_COUNTER));
+                                                debug_msg("\n  Bishop == " + global_var(GVAR_BISHOP_FAMILY_COUNTER));
+                                                debug_msg("\n  Mordino == " + global_var(GVAR_MORDINO_FAMILY_COUNTER));
+                                                debug_msg("\n  Wright == " + global_var(GVAR_WRIGHT_FAMILY_COUNTER));
                                             end
-                                            set_global_var(314, global_var(314) bwor 32768);
-                                            if (global_var(313) == 0) then begin
-                                                metarule3(100, self_obj, 250, 0);
+                                            set_global_var(GVAR_NEW_RENO_FLAG_1, global_var(GVAR_NEW_RENO_FLAG_1) bwor 32768);
+                                            if (global_var(GVAR_NEW_RENO_GUARD_ASSIGNMENT) == 0) then begin
+                                                rm_fixed_timer_event(self_obj, 250);
                                                 add_timer_event(self_obj, 0, 250);
                                             end
                                         end
                                         if (combat_is_initialized == 0) then begin
-                                            attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+                                            attack(dude_obj);
                                         end
                                     end
                                     else begin
                                         if (fixed_param >= 300) then begin
                                             if (obj_can_see_obj(self_obj, dude_obj) == 0) then begin
-                                                anim(dude_obj, 1000, rotation_to_tile(tile_num(dude_obj), tile_num(self_obj)));
-                                                metarule3(100, self_obj, 23, 0);
+                                                anim(dude_obj, 1000, rotation_to_tile(dude_tile, tile_num(self_obj)));
+                                                rm_fixed_timer_event(self_obj, 23);
                                                 add_timer_event(self_obj, 10 * 3, 23);
                                             end
                                         end
@@ -1281,49 +1281,48 @@ procedure Node998
 begin
     debug_msg("helicopter_script_node998");
     set_local_var(0, local_var(0) bwor 2);
-    if ((global_var(313) == 0) == 0) then begin
-        set_global_var(313, 4);
+    if ((global_var(GVAR_NEW_RENO_GUARD_ASSIGNMENT) == 0) == 0) then begin
+        set_global_var(GVAR_NEW_RENO_GUARD_ASSIGNMENT, 4);
     end
-    if ((((global_var(314) bwand 32768) != 0) or (global_var(494) == 8)) == 0) then begin
-        if ((((global_var(314) bwand 32768) != 0) or (global_var(494) == 8)) == 0) then begin
-            if (((1 == 1) and (global_var(134) <= -1) or ((1 == 2) and (global_var(135) <= -1)) or ((1 == 3) and (global_var(136) <= -1)) or ((1 == 4) and (global_var(216) <= -1))) == 0) then begin
+    if ((((global_var(GVAR_NEW_RENO_FLAG_1) bwand 32768) != 0) or (global_var(GVAR_FALLOUT_2) == 8)) == 0) then begin
+        if ((((global_var(GVAR_NEW_RENO_FLAG_1) bwand 32768) != 0) or (global_var(GVAR_FALLOUT_2) == 8)) == 0) then begin
+            if (((1 == 1) and (global_var(GVAR_SALVATORE_FAMILY_COUNTER) <= -1) or ((1 == 2) and (global_var(GVAR_BISHOP_FAMILY_COUNTER) <= -1)) or ((1 == 3) and (global_var(GVAR_MORDINO_FAMILY_COUNTER) <= -1)) or ((1 == 4) and (global_var(GVAR_WRIGHT_FAMILY_COUNTER) <= -1))) == 0) then begin
                 if (1 == 1) then begin
-                    set_global_var(134, global_var(134) - 20);
+                    set_global_var(GVAR_SALVATORE_FAMILY_COUNTER, global_var(GVAR_SALVATORE_FAMILY_COUNTER) - 20);
                 end
                 else begin
                     if (1 == 2) then begin
-                        set_global_var(135, global_var(135) - 20);
+                        set_global_var(GVAR_BISHOP_FAMILY_COUNTER, global_var(GVAR_BISHOP_FAMILY_COUNTER) - 20);
                     end
                     else begin
                         if (1 == 3) then begin
-                            set_global_var(136, global_var(136) - 20);
+                            set_global_var(GVAR_MORDINO_FAMILY_COUNTER, global_var(GVAR_MORDINO_FAMILY_COUNTER) - 20);
                         end
                         else begin
                             if (1 == 4) then begin
-                                set_global_var(216, global_var(216) - 20);
+                                set_global_var(GVAR_WRIGHT_FAMILY_COUNTER, global_var(GVAR_WRIGHT_FAMILY_COUNTER) - 20);
                             end
                         end
                     end
                 end
             end
             debug_msg("\nfamily standings: ");
-            debug_msg("\n  Salvatore == " + global_var(134));
-            debug_msg("\n  Bishop == " + global_var(135));
-            debug_msg("\n  Mordino == " + global_var(136));
-            debug_msg("\n  Wright == " + global_var(216));
+            debug_msg("\n  Salvatore == " + global_var(GVAR_SALVATORE_FAMILY_COUNTER));
+            debug_msg("\n  Bishop == " + global_var(GVAR_BISHOP_FAMILY_COUNTER));
+            debug_msg("\n  Mordino == " + global_var(GVAR_MORDINO_FAMILY_COUNTER));
+            debug_msg("\n  Wright == " + global_var(GVAR_WRIGHT_FAMILY_COUNTER));
         end
-        set_global_var(314, global_var(314) bwor 32768);
-        if (global_var(313) == 0) then begin
-            metarule3(100, self_obj, 250, 0);
+        set_global_var(GVAR_NEW_RENO_FLAG_1, global_var(GVAR_NEW_RENO_FLAG_1) bwor 32768);
+        if (global_var(GVAR_NEW_RENO_GUARD_ASSIGNMENT) == 0) then begin
+            rm_fixed_timer_event(self_obj, 250);
             add_timer_event(self_obj, 0, 250);
         end
     end
     if (combat_is_initialized == 0) then begin
-        attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+        attack(dude_obj);
     end
 end
 
 procedure Node999
 begin
 end
-
