diff --git a/Games/Fallout2.Steam/Original/NCJULES.ssl b/Games/Fallout2.Steam/ReDefine/NCJULES.ssl
index dbd8583..e7f6178 100644
--- a/Games/Fallout2.Steam/Original/NCJULES.ssl
+++ b/Games/Fallout2.Steam/ReDefine/NCJULES.ssl
@@ -273,53 +273,53 @@ end
 procedure ChooseFromList
 begin
     variable LVar0 := 0;
-    gsay_reply(451, stored_msg);
-    if (message_str(353, top_msg - 1) != "") then begin
-        giq_option(1, 451, message_str(353, 150), CLICK_PREV, 50);
+    Reply(stored_msg);
+    if (message_str(SCRIPT_NEWR1, top_msg - 1) != "") then begin
+        giq_option(1, NAME, message_str(SCRIPT_NEWR1, 150), CLICK_PREV, NEUTRAL_REACTION);
     end
     if (LVar0 == 0) then begin
-        if (message_str(353, top_msg + 0) != "") then begin
-            giq_option(1, 451, message_str(353, top_msg + 0), CLICK_0, 50);
+        if (message_str(SCRIPT_NEWR1, top_msg + 0) != "") then begin
+            giq_option(1, NAME, message_str(SCRIPT_NEWR1, top_msg + 0), CLICK_0, NEUTRAL_REACTION);
         end
         else begin
             LVar0 := 1;
         end
     end
     if (LVar0 == 0) then begin
-        if (message_str(353, top_msg + 1) != "") then begin
-            giq_option(1, 451, message_str(353, top_msg + 1), CLICK_1, 50);
+        if (message_str(SCRIPT_NEWR1, top_msg + 1) != "") then begin
+            giq_option(1, NAME, message_str(SCRIPT_NEWR1, top_msg + 1), CLICK_1, NEUTRAL_REACTION);
         end
         else begin
             LVar0 := 1;
         end
     end
     if (LVar0 == 0) then begin
-        if (message_str(353, top_msg + 2) != "") then begin
-            giq_option(1, 451, message_str(353, top_msg + 2), CLICK_2, 50);
+        if (message_str(SCRIPT_NEWR1, top_msg + 2) != "") then begin
+            giq_option(1, NAME, message_str(SCRIPT_NEWR1, top_msg + 2), CLICK_2, NEUTRAL_REACTION);
         end
         else begin
             LVar0 := 1;
         end
     end
     if (LVar0 == 0) then begin
-        if (message_str(353, top_msg + 3) != "") then begin
-            giq_option(1, 451, message_str(353, top_msg + 3), CLICK_3, 50);
+        if (message_str(SCRIPT_NEWR1, top_msg + 3) != "") then begin
+            giq_option(1, NAME, message_str(SCRIPT_NEWR1, top_msg + 3), CLICK_3, NEUTRAL_REACTION);
         end
         else begin
             LVar0 := 1;
         end
     end
     if (LVar0 == 0) then begin
-        if (message_str(353, top_msg + 4) != "") then begin
-            giq_option(1, 451, message_str(353, top_msg + 4), CLICK_4, 50);
+        if (message_str(SCRIPT_NEWR1, top_msg + 4) != "") then begin
+            giq_option(1, NAME, message_str(SCRIPT_NEWR1, top_msg + 4), CLICK_4, NEUTRAL_REACTION);
         end
         else begin
             LVar0 := 1;
         end
     end
     if (LVar0 == 0) then begin
-        if (message_str(353, top_msg + 5) != "") then begin
-            giq_option(1, 451, message_str(353, 151), CLICK_MORE, 50);
+        if (message_str(SCRIPT_NEWR1, top_msg + 5) != "") then begin
+            giq_option(1, NAME, message_str(SCRIPT_NEWR1, 151), CLICK_MORE, NEUTRAL_REACTION);
         end
     end
 end
@@ -376,8 +376,8 @@ begin
         if ((local_var(4) bwand 2) != 0) then begin
             if (critter_is_fleeing(self_obj)) then begin
                 if (anim_busy(self_obj) == 0) then begin
-                    if (tile_distance(tile_num(self_obj), global_var(46)) > 2) then begin
-                        animate_move_obj_to_tile(self_obj, global_var(46), 1);
+                    if (tile_distance(tile_num(self_obj), global_var(GVAR_MAP_EXIT_TILE)) > 2) then begin
+                        animate_move_obj_to_tile(self_obj, global_var(GVAR_MAP_EXIT_TILE), 1);
                     end
                     else begin
                         destroy_object(self_obj);
@@ -389,10 +389,10 @@ begin
                     if (anim_busy(self_obj) == 0) then begin
                         if ((tile_distance_objs(self_obj, dude_obj) < 8) or obj_can_see_obj(self_obj, dude_obj)) then begin
                             if (anim_busy(self_obj) == 0) then begin
-                                global_temp := rotation_to_tile(tile_num(dude_obj), tile_num(self_obj));
+                                global_temp := rotation_to_tile(dude_tile, tile_num(self_obj));
                                 animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(self_obj), global_temp, random(3, 10)), 1);
                                 global_temp := (global_temp + 1) % 6;
-                                while ((anim_busy(self_obj) == 0) and (global_temp != rotation_to_tile(tile_num(dude_obj), tile_num(self_obj)))) do begin
+                                while ((anim_busy(self_obj) == 0) and (global_temp != rotation_to_tile(dude_tile, tile_num(self_obj)))) do begin
                                     debug_msg("flee loop: rot == " + global_temp);
                                     animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(self_obj), global_temp, random(3, 10)), 1);
                                     global_temp := (global_temp + 1) % 6;
@@ -402,7 +402,7 @@ begin
                     end
                 end
                 else begin
-                    attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+                    attack(dude_obj);
                 end
             end
         end
@@ -411,7 +411,7 @@ begin
                 set_local_var(4, local_var(4) bwor 512);
                 last_timed_obj := self_obj;
                 last_timed_param := last_timed_param;
-                metarule3(100, self_obj, 1, 0);
+                rm_fixed_timer_event(self_obj, 1);
                 add_timer_event(self_obj, game_ticks(1), 1);
             end
         end
@@ -431,11 +431,11 @@ begin
     move_obj_inven_to_obj(new_reno_jules_box, self_obj);
     start_gdialog(451, self_obj, 4, -1, -1);
     gsay_start;
-    if ((global_var(345) bwand 2048) != 0) then begin
+    if ((global_var(GVAR_NEW_RENO_FLAG_2) bwand 2048) != 0) then begin
         call Node081();
     end
     else begin
-        if ((global_var(345) bwand 2) != 0) then begin
+        if ((global_var(GVAR_NEW_RENO_FLAG_2) bwand 2) != 0) then begin
             call Node017();
         end
         else begin
@@ -449,62 +449,62 @@ begin
     set_local_var(4, local_var(4) bwor 1);
     last_timed_obj := self_obj;
     last_timed_param := last_timed_param;
-    metarule3(100, self_obj, 1, 0);
+    rm_fixed_timer_event(self_obj, 1);
     add_timer_event(self_obj, game_ticks(25) + random(0, game_ticks(35 - 25)), 1);
 end
 
 procedure destroy_p_proc
 begin
     if (source_obj == dude_obj) then begin
-        set_global_var(4, global_var(4) + 1);
-        if (metarule(51, self_obj) == 2) then begin
+        set_global_var(GVAR_BAD_MONSTER, global_var(GVAR_BAD_MONSTER) + 1);
+        if (critter_kill_type(self_obj) == KILL_TYPE_children_kills) then begin
             if (source_obj == dude_obj) then begin
-                set_global_var(2, 0);
-                set_global_var(1, global_var(1) + 1);
-                set_global_var(0, global_var(0) + -15);
+                set_global_var(GVAR_CHAMPION_REPUTATION, 0);
+                set_global_var(GVAR_CHILDKILLER_REPUTATION, global_var(GVAR_CHILDKILLER_REPUTATION) + 1);
+                set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + -15);
                 debug_msg("Player gains " + -15 + " Karma Points.");
-                set_global_var(37, 0);
-                set_global_var(38, 0);
-                set_global_var(39, 0);
-                set_global_var(40, 0);
-                set_global_var(41, 0);
-                set_global_var(42, 0);
-                set_global_var(43, 0);
-                set_global_var(44, 0);
-                set_global_var(45, 0);
-                if (global_var(0) >= 1000) then begin
-                    set_global_var(37, 1);
+                set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+                set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+                set_global_var(GVAR_KARMA_DEFENDER, 0);
+                set_global_var(GVAR_KARMA_WANDERER, 0);
+                set_global_var(GVAR_KARMA_BETRAYER, 0);
+                set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+                set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+                set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+                    set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
                 end
                 else begin
-                    if (global_var(0) >= 750) then begin
-                        set_global_var(38, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
                     end
                     else begin
-                        if (global_var(0) >= 500) then begin
-                            set_global_var(39, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                            set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                         end
                         else begin
-                            if (global_var(0) >= 250) then begin
-                                set_global_var(40, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                                set_global_var(GVAR_KARMA_DEFENDER, 1);
                             end
                             else begin
-                                if (global_var(0) > -250) then begin
-                                    set_global_var(41, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                                    set_global_var(GVAR_KARMA_WANDERER, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -500) then begin
-                                        set_global_var(42, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                        set_global_var(GVAR_KARMA_BETRAYER, 1);
                                     end
                                     else begin
-                                        if (global_var(0) > -750) then begin
-                                            set_global_var(43, 1);
+                                        if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                            set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                         end
                                         else begin
-                                            if (global_var(0) > -1000) then begin
-                                                set_global_var(44, 1);
+                                            if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                                set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                             end
                                             else begin
-                                                set_global_var(45, 1);
+                                                set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                             end
                                         end
                                     end
@@ -513,62 +513,62 @@ begin
                         end
                     end
                 end
-                set_global_var(55, global_var(55) + -8);
+                set_global_var(GVAR_TOWN_REP_NEW_RENO, global_var(GVAR_TOWN_REP_NEW_RENO) + -8);
                 debug_msg("Added " + -8 + " to Town Rep");
             end
         end
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(5) > (2 * global_var(4))) or (global_var(3) == 1))) then begin
-            set_global_var(3, 1);
-            set_global_var(2, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
+            set_global_var(GVAR_BERSERKER_REPUTATION, 1);
+            set_global_var(GVAR_CHAMPION_REPUTATION, 0);
         end
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(4) > (3 * global_var(5))) or (global_var(2) == 1)) and (global_var(1) == 0)) then begin
-            set_global_var(2, 1);
-            set_global_var(3, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(4) > (3 * global_var(GVAR_GOOD_MONSTER))) or (global_var(GVAR_CHAMPION_REPUTATION) == 1)) and (global_var(GVAR_CHILDKILLER_REPUTATION) == 0)) then begin
+            set_global_var(GVAR_CHAMPION_REPUTATION, 1);
+            set_global_var(GVAR_BERSERKER_REPUTATION, 0);
         end
-        set_global_var(0, global_var(0) + 5);
+        set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + 5);
         debug_msg("Player gains " + 5 + " Karma Points.");
-        set_global_var(37, 0);
-        set_global_var(38, 0);
-        set_global_var(39, 0);
-        set_global_var(40, 0);
-        set_global_var(41, 0);
-        set_global_var(42, 0);
-        set_global_var(43, 0);
-        set_global_var(44, 0);
-        set_global_var(45, 0);
-        if (global_var(0) >= 1000) then begin
-            set_global_var(37, 1);
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+        set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+        set_global_var(GVAR_KARMA_DEFENDER, 0);
+        set_global_var(GVAR_KARMA_WANDERER, 0);
+        set_global_var(GVAR_KARMA_BETRAYER, 0);
+        set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+        set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+            set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
         end
         else begin
-            if (global_var(0) >= 750) then begin
-                set_global_var(38, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
             end
             else begin
-                if (global_var(0) >= 500) then begin
-                    set_global_var(39, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                 end
                 else begin
-                    if (global_var(0) >= 250) then begin
-                        set_global_var(40, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                        set_global_var(GVAR_KARMA_DEFENDER, 1);
                     end
                     else begin
-                        if (global_var(0) > -250) then begin
-                            set_global_var(41, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                            set_global_var(GVAR_KARMA_WANDERER, 1);
                         end
                         else begin
-                            if (global_var(0) > -500) then begin
-                                set_global_var(42, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                set_global_var(GVAR_KARMA_BETRAYER, 1);
                             end
                             else begin
-                                if (global_var(0) > -750) then begin
-                                    set_global_var(43, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -1000) then begin
-                                        set_global_var(44, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                     end
                                     else begin
-                                        set_global_var(45, 1);
+                                        set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                     end
                                 end
                             end
@@ -577,11 +577,11 @@ begin
                 end
             end
         end
-        set_global_var(55, global_var(55) + 2);
+        set_global_var(GVAR_TOWN_REP_NEW_RENO, global_var(GVAR_TOWN_REP_NEW_RENO) + 2);
         debug_msg("Added " + 2 + " to Town Rep");
     end
-    set_global_var(314, global_var(314) bwor 1073741824);
-    restock_obj := create_object_sid(429, 0, 0, -1);
+    set_global_var(GVAR_NEW_RENO_FLAG_1, global_var(GVAR_NEW_RENO_FLAG_1) bwor 1073741824);
+    restock_obj := create_object(429, 0, 0);
     add_mult_objs_to_inven(self_obj, restock_obj, 1);
     move_obj_inven_to_obj(new_reno_jules_box, self_obj);
 end
@@ -589,13 +589,13 @@ end
 procedure look_at_p_proc
 begin
     script_overrides;
-    display_msg(message_str(451, 151));
+    display_mstr(151);
 end
 
 procedure description_p_proc
 begin
     script_overrides;
-    display_msg(message_str(451, 150 + ((local_var(4) bwand 1) != 0)));
+    display_mstr(150 + ((local_var(4) bwand 1) != 0));
     set_local_var(4, local_var(4) bwor 4);
 end
 
@@ -613,17 +613,17 @@ end
 
 procedure map_enter_p_proc
 begin
-    if (metarule(22, 0) == 0) then begin
-        if (metarule(14, 0)) then begin
-            critter_add_trait(self_obj, 1, 6, 156);
-            critter_add_trait(self_obj, 1, 5, 22);
-            set_local_var(5, has_trait(1, self_obj, 10));
+    if (is_loading_game == 0) then begin
+        if (map_first_run) then begin
+            critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 156);
+            critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_TOUGH_GUARD);
+            set_local_var(5, obj_get_rot(self_obj));
         end
         if (((local_var(4) bwand 1024) != 0) == 0) then begin
-            if (metarule(30, 0) == 7) then begin
+            if (car_current_town == 7) then begin
                 last_timed_obj := self_obj;
                 last_timed_param := last_timed_param;
-                metarule3(100, self_obj, 12, 0);
+                rm_fixed_timer_event(self_obj, 12);
                 add_timer_event(self_obj, game_ticks(1), 12);
             end
         end
@@ -639,7 +639,7 @@ end
 
 procedure map_update_p_proc
 begin
-    if ((global_var(314) bwand 1073741824) != 0) then begin
+    if ((global_var(GVAR_NEW_RENO_FLAG_1) bwand 1073741824) != 0) then begin
         call destroy_p_proc();
         destroy_object(self_obj);
     end
@@ -650,8 +650,8 @@ begin
     if (fixed_param == 1) then begin
         if (combat_is_initialized == 0) then begin
             if (obj_can_see_obj(self_obj, dude_obj)) then begin
-                if (((global_var(345) bwand 2) != 0) == 0) then begin
-                    if (global_var(11) == 1) then begin
+                if (((global_var(GVAR_NEW_RENO_FLAG_2) bwand 2) != 0) == 0) then begin
+                    if (global_var(GVAR_REPUTATION_SLAVER) == 1) then begin
                         call Node001();
                     end
                     else begin
@@ -659,24 +659,24 @@ begin
                     end
                 end
                 else begin
-                    if (global_var(494) == 8) then begin
+                    if (global_var(GVAR_FALLOUT_2) == 8) then begin
                         call Node003();
                     end
                     else begin
-                        if ((obj_pid(critter_inven_obj(dude_obj, 0)) == 3) or (obj_pid(critter_inven_obj(dude_obj, 0)) == 348) or (obj_pid(critter_inven_obj(dude_obj, 0)) == 349) or (obj_pid(critter_inven_obj(dude_obj, 0)) == 232)) then begin
+                        if ((obj_pid(critter_inven_obj(dude_obj, INVEN_TYPE_WORN)) == 3) or (obj_pid(critter_inven_obj(dude_obj, INVEN_TYPE_WORN)) == 348) or (obj_pid(critter_inven_obj(dude_obj, INVEN_TYPE_WORN)) == 349) or (obj_pid(critter_inven_obj(dude_obj, INVEN_TYPE_WORN)) == 232)) then begin
                             call Node004();
                         end
                         else begin
-                            if (global_var(230) > 0) then begin
+                            if (global_var(GVAR_NEW_RENO_MADE_MAN) > 0) then begin
                                 call Node005();
                             end
                             else begin
-                                if (global_var(259) > 0) then begin
+                                if (global_var(GVAR_NEW_RENO_HAS_REP_PRIZEFIGHTER) > 0) then begin
                                     call Node007();
                                 end
                                 else begin
-                                    if (global_var(232) > 0) then begin
-                                        if (get_critter_stat(dude_obj, 34) == 0) then begin
+                                    if (global_var(GVAR_NEW_RENO_PORN_STAR) > 0) then begin
+                                        if (dude_gender == 0) then begin
                                             call Node009();
                                         end
                                         else begin
@@ -722,19 +722,19 @@ begin
             anim(self_obj, 1000, rotation_to_tile(tile_num(self_obj), 25321));
             if (obj_can_see_obj(self_obj, dude_obj) or obj_can_see_obj(dude_obj, self_obj) and obj_on_screen(self_obj)) then begin
                 set_local_var(4, local_var(4) bwor 1024);
-                float_msg(self_obj, message_str(451, 2000), 8);
+                float_msg(self_obj, mstr(2000), FLOAT_MSG_YELLOW);
             end
             else begin
                 last_timed_obj := self_obj;
                 last_timed_param := last_timed_param;
-                metarule3(100, self_obj, 12, 0);
+                rm_fixed_timer_event(self_obj, 12);
                 add_timer_event(self_obj, game_ticks(1), 12);
             end
         end
     end
     last_timed_obj := self_obj;
     last_timed_param := last_timed_param;
-    metarule3(100, self_obj, 1, 0);
+    rm_fixed_timer_event(self_obj, 1);
     add_timer_event(self_obj, game_ticks(15) + random(0, game_ticks(30 - 15)), 1);
 end
 
@@ -751,923 +751,923 @@ end
 
 procedure Node001
 begin
-    float_msg(self_obj, message_str(451, random(200, 206)), 8);
+    float_msg(self_obj, mstr(random(200, 206)), FLOAT_MSG_YELLOW);
 end
 
 procedure Node002
 begin
-    float_msg(self_obj, message_str(451, random(220, 226)), 8);
+    float_msg(self_obj, mstr(random(220, 226)), FLOAT_MSG_YELLOW);
 end
 
 procedure Node003
 begin
-    float_msg(self_obj, message_str(451, random(240, 246)), 8);
+    float_msg(self_obj, mstr(random(240, 246)), FLOAT_MSG_YELLOW);
 end
 
 procedure Node004
 begin
-    float_msg(self_obj, message_str(451, random(260, 263)), 8);
+    float_msg(self_obj, mstr(random(260, 263)), FLOAT_MSG_YELLOW);
 end
 
 procedure Node005
 begin
-    float_msg(self_obj, message_str(451, random(282 - (2 * (get_critter_stat(dude_obj, 34) == 0)), 283 + (2 * (get_critter_stat(dude_obj, 34) == 1)))), 8);
+    float_msg(self_obj, mstr(random(282 - (2 * (dude_gender == 0)), 283 + (2 * (dude_gender == 1)))), FLOAT_MSG_YELLOW);
 end
 
 procedure Node007
 begin
-    float_msg(self_obj, message_str(451, random(303 - (3 * (get_critter_stat(dude_obj, 34) == 0)), 303 + (3 * (get_critter_stat(dude_obj, 34) == 1)))), 8);
+    float_msg(self_obj, mstr(random(303 - (3 * (dude_gender == 0)), 303 + (3 * (dude_gender == 1)))), FLOAT_MSG_YELLOW);
 end
 
 procedure Node009
 begin
-    float_msg(self_obj, message_str(451, random(320, 325)), 8);
+    float_msg(self_obj, mstr(random(320, 325)), FLOAT_MSG_YELLOW);
 end
 
 procedure Node010
 begin
-    float_msg(self_obj, message_str(451, random(340, 350)), 8);
+    float_msg(self_obj, mstr(random(340, 350)), FLOAT_MSG_YELLOW);
 end
 
 procedure Node011
 begin
-    float_msg(self_obj, message_str(451, random(365, 370)), 8);
+    float_msg(self_obj, mstr(random(365, 370)), FLOAT_MSG_YELLOW);
 end
 
 procedure Node012
 begin
-    float_msg(self_obj, message_str(451, random(380, 383)), 8);
+    float_msg(self_obj, mstr(random(380, 383)), FLOAT_MSG_YELLOW);
 end
 
 procedure Node013
 begin
-    float_msg(self_obj, message_str(451, random(390, 392)), 8);
+    float_msg(self_obj, mstr(random(390, 392)), FLOAT_MSG_YELLOW);
 end
 
 procedure Node014
 begin
-    float_msg(self_obj, message_str(451, random(400, 404)), 8);
+    float_msg(self_obj, mstr(random(400, 404)), FLOAT_MSG_YELLOW);
 end
 
 procedure Node015
 begin
-    float_msg(self_obj, message_str(451, random(415 - (5 * (get_critter_stat(dude_obj, 34) == 0)), 423 + (6 * (get_critter_stat(dude_obj, 34) == 1)))), 8);
+    float_msg(self_obj, mstr(random(415 - (5 * (dude_gender == 0)), 423 + (6 * (dude_gender == 1)))), FLOAT_MSG_YELLOW);
 end
 
 procedure Node016
 begin
-    set_global_var(345, global_var(345) bwor 2);
-    gsay_reply(451, message_str(451, 450) + message_str(451, 451 + (get_critter_stat(dude_obj, 34) == 1)));
-    giq_option(-3, 451, 453, Node017, 50);
-    giq_option(4, 451, 454, Node017, 50);
+    set_global_var(GVAR_NEW_RENO_FLAG_2, global_var(GVAR_NEW_RENO_FLAG_2) bwor 2);
+    Reply(mstr(450) + mstr(451 + (dude_gender == 1)));
+    giq_option(-3, NAME, 453, Node017, NEUTRAL_REACTION);
+    giq_option(4, NAME, 454, Node017, NEUTRAL_REACTION);
 end
 
 procedure Node017
 begin
     if ((local_var(4) bwand 4096) != 0) then begin
-        gsay_reply(451, 462);
-        giq_option(-3, 451, 463, Node018, 50);
-        giq_option(-3, 451, 464, Node999, 50);
-        if ((global_var(286) == 1) and (((global_var(314) bwand 64) != 0) == 0)) then begin
-            giq_option(4, 451, 467, Node072, 51);
+        Reply(462);
+        giq_option(-3, NAME, 463, Node018, NEUTRAL_REACTION);
+        giq_option(-3, NAME, 464, Node999, NEUTRAL_REACTION);
+        if ((global_var(GVAR_NEW_RENO_WRIGHT_MYSTERY) == 1) and (((global_var(GVAR_NEW_RENO_FLAG_1) bwand 64) != 0) == 0)) then begin
+            giq_option(4, NAME, 467, Node072, BAD_REACTION);
         end
-        if (global_var(348) == 1) then begin
-            giq_option(4, 451, 468, Node076, 50);
+        if (global_var(GVAR_NEW_RENO_STOLEN_CAR) == 1) then begin
+            giq_option(4, NAME, 468, Node076, NEUTRAL_REACTION);
         end
-        giq_option(4, 451, 466, Node019, 50);
-        giq_option(7, 451, 470, Node075, 51);
-        giq_option(4, 451, 472, Node999, 50);
+        giq_option(4, NAME, 466, Node019, NEUTRAL_REACTION);
+        giq_option(7, NAME, 470, Node075, BAD_REACTION);
+        giq_option(4, NAME, 472, Node999, NEUTRAL_REACTION);
     end
     else begin
         set_local_var(4, local_var(4) bwor 4096);
-        gsay_reply(451, 460 + ((get_critter_stat(dude_obj, 34) == 1) and (get_critter_stat(dude_obj, 3) > 5)));
-        giq_option(-3, 451, 463, Node018, 50);
-        giq_option(-3, 451, 464, Node999, 50);
-        if ((global_var(286) == 1) and (((global_var(314) bwand 64) != 0) == 0)) then begin
-            giq_option(4, 451, 467, Node072, 51);
+        Reply(460 + ((dude_gender == 1) and (dude_charisma > 5)));
+        giq_option(-3, NAME, 463, Node018, NEUTRAL_REACTION);
+        giq_option(-3, NAME, 464, Node999, NEUTRAL_REACTION);
+        if ((global_var(GVAR_NEW_RENO_WRIGHT_MYSTERY) == 1) and (((global_var(GVAR_NEW_RENO_FLAG_1) bwand 64) != 0) == 0)) then begin
+            giq_option(4, NAME, 467, Node072, BAD_REACTION);
         end
-        if (global_var(348) == 1) then begin
-            giq_option(4, 451, 468, Node076, 50);
+        if (global_var(GVAR_NEW_RENO_STOLEN_CAR) == 1) then begin
+            giq_option(4, NAME, 468, Node076, NEUTRAL_REACTION);
         end
-        giq_option(4, 451, 465, Node019, 50);
-        giq_option(7, 451, 469, Node075, 51);
-        giq_option(4, 451, 471, Node999, 50);
+        giq_option(4, NAME, 465, Node019, NEUTRAL_REACTION);
+        giq_option(7, NAME, 469, Node075, BAD_REACTION);
+        giq_option(4, NAME, 471, Node999, NEUTRAL_REACTION);
     end
 end
 
 procedure Node018
 begin
-    gsay_reply(451, 480 + (get_critter_stat(dude_obj, 34) == 1));
-    giq_option(-3, 451, 482, Node999, 50);
+    Reply(480 + (dude_gender == 1));
+    giq_option(-3, NAME, 482, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node019
 begin
     if ((local_var(4) bwand 8192) != 0) then begin
-        gsay_reply(451, 493);
+        Reply(493);
     end
     else begin
         set_local_var(4, local_var(4) bwor 8192);
-        gsay_reply(451, 490 + ((1 + (get_critter_stat(dude_obj, 3) > 5)) * (get_critter_stat(dude_obj, 34) == 1)));
+        Reply(490 + ((1 + (dude_charisma > 5)) * (dude_gender == 1)));
     end
-    giq_option(4, 451, 494, Node020, 50);
-    giq_option(4, 451, 495, Node054, 50);
-    giq_option(4, 451, 496, Node065, 50);
-    giq_option(4, 451, 497, Node066, 50);
-    giq_option(4, 451, 498, Node067, 50);
-    if (global_var(494) == 0) then begin
-        giq_option(4, 451, 499, Node068, 50);
+    giq_option(4, NAME, 494, Node020, NEUTRAL_REACTION);
+    giq_option(4, NAME, 495, Node054, NEUTRAL_REACTION);
+    giq_option(4, NAME, 496, Node065, NEUTRAL_REACTION);
+    giq_option(4, NAME, 497, Node066, NEUTRAL_REACTION);
+    giq_option(4, NAME, 498, Node067, NEUTRAL_REACTION);
+    if (global_var(GVAR_FALLOUT_2) == 0) then begin
+        giq_option(4, NAME, 499, Node068, NEUTRAL_REACTION);
     end
     else begin
-        if (global_var(494) == 4) then begin
-            giq_option(4, 451, 500, Node070, 50);
+        if (global_var(GVAR_FALLOUT_2) == 4) then begin
+            giq_option(4, NAME, 500, Node070, NEUTRAL_REACTION);
         end
         else begin
-            if (global_var(494) == 5) then begin
-                giq_option(4, 451, 501, Node071, 50);
+            if (global_var(GVAR_FALLOUT_2) == 5) then begin
+                giq_option(4, NAME, 501, Node071, NEUTRAL_REACTION);
             end
         end
     end
-    if (global_var(233) == 0) then begin
-        giq_option(4, 451, 502, Node069, 50);
+    if (global_var(GVAR_VAULT13_FOUND_GECK) == 0) then begin
+        giq_option(4, NAME, 502, Node069, NEUTRAL_REACTION);
     end
-    giq_option(4, 451, 503, Node999, 50);
+    giq_option(4, NAME, 503, Node999, NEUTRAL_REACTION);
     prev_node := 19;
 end
 
 procedure Node020
 begin
-    set_global_var(345, global_var(345) bwor 16);
-    set_global_var(345, global_var(345) bwor 32);
-    set_global_var(255, global_var(255) bwor 1);
-    set_global_var(285, global_var(285) bwor 256);
-    gsay_reply(451, 515 + ((local_var(4) bwand 16384) != 0));
+    set_global_var(GVAR_NEW_RENO_FLAG_2, global_var(GVAR_NEW_RENO_FLAG_2) bwor 16);
+    set_global_var(GVAR_NEW_RENO_FLAG_2, global_var(GVAR_NEW_RENO_FLAG_2) bwor 32);
+    set_global_var(GVAR_NEW_RENO_BISHOP, global_var(GVAR_NEW_RENO_BISHOP) bwor 1);
+    set_global_var(GVAR_NEW_RENO_WRIGHT_FLAGS, global_var(GVAR_NEW_RENO_WRIGHT_FLAGS) bwor 256);
+    Reply(515 + ((local_var(4) bwand 16384) != 0));
     set_local_var(4, local_var(4) bwor 16384);
-    giq_option(4, 451, 517, Node024, 50);
-    giq_option(4, 451, 518, Node021, 50);
-    giq_option(4, 451, 519, Node019, 50);
-    giq_option(4, 451, 520, Node999, 50);
+    giq_option(4, NAME, 517, Node024, NEUTRAL_REACTION);
+    giq_option(4, NAME, 518, Node021, NEUTRAL_REACTION);
+    giq_option(4, NAME, 519, Node019, NEUTRAL_REACTION);
+    giq_option(4, NAME, 520, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node021
 begin
-    gsay_reply(451, 525);
-    giq_option(4, 451, 517, Node024, 50);
-    giq_option(4, 451, 527, Node022, 50);
-    giq_option(4, 451, 528, Node019, 50);
-    giq_option(4, 451, 520, Node999, 50);
+    Reply(525);
+    giq_option(4, NAME, 517, Node024, NEUTRAL_REACTION);
+    giq_option(4, NAME, 527, Node022, NEUTRAL_REACTION);
+    giq_option(4, NAME, 528, Node019, NEUTRAL_REACTION);
+    giq_option(4, NAME, 520, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node022
 begin
-    gsay_reply(451, 535);
-    giq_option(4, 451, 517, Node024, 50);
-    giq_option(4, 451, 537, Node023, 50);
-    giq_option(4, 451, 528, Node019, 50);
-    giq_option(4, 451, 520, Node999, 50);
+    Reply(535);
+    giq_option(4, NAME, 517, Node024, NEUTRAL_REACTION);
+    giq_option(4, NAME, 537, Node023, NEUTRAL_REACTION);
+    giq_option(4, NAME, 528, Node019, NEUTRAL_REACTION);
+    giq_option(4, NAME, 520, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node023
 begin
-    if (metarule(17, 30) == 0) then begin
+    if (town_known(30) == MARK_STATE_UNKNOWN) then begin
         debug_msg("  mark_on_map(" + 30 + ")");
-        mark_area_known(0, 30, 1);
+        mark_area_known(MARK_TYPE_TOWN, 30, MARK_STATE_KNOWN);
     end
-    gsay_reply(451, 545);
-    giq_option(4, 451, 517, Node024, 50);
-    giq_option(4, 451, 547, Node019, 50);
-    giq_option(4, 451, 548, Node999, 50);
+    Reply(545);
+    giq_option(4, NAME, 517, Node024, NEUTRAL_REACTION);
+    giq_option(4, NAME, 547, Node019, NEUTRAL_REACTION);
+    giq_option(4, NAME, 548, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node024
 begin
-    gsay_reply(451, 555 + ((local_var(4) bwand 32768) != 0));
+    Reply(555 + ((local_var(4) bwand 32768) != 0));
     set_local_var(4, local_var(4) bwor 32768);
-    if (has_skill(dude_obj, 15) > 90) then begin
+    if (dude_skill(SKILL_BARTER) > 90) then begin
         if (item_caps_total(dude_obj) >= 25) then begin
             info_cost := 25;
-            giq_option(4, 451, 557, Node990, 50);
+            giq_option(4, NAME, 557, Node990, NEUTRAL_REACTION);
         end
         else begin
-            giq_option(4, 451, 562, Node999, 50);
+            giq_option(4, NAME, 562, Node999, NEUTRAL_REACTION);
         end
     end
     else begin
-        if (has_skill(dude_obj, 15) > 75) then begin
+        if (dude_skill(SKILL_BARTER) > 75) then begin
             if (item_caps_total(dude_obj) >= 50) then begin
                 info_cost := 50;
-                giq_option(4, 451, 558, Node990, 50);
+                giq_option(4, NAME, 558, Node990, NEUTRAL_REACTION);
             end
             else begin
-                giq_option(4, 451, 562, Node999, 50);
+                giq_option(4, NAME, 562, Node999, NEUTRAL_REACTION);
             end
         end
         else begin
-            if (has_skill(dude_obj, 15) > 40) then begin
+            if (dude_skill(SKILL_BARTER) > 40) then begin
                 if (item_caps_total(dude_obj) >= 75) then begin
                     info_cost := 75;
-                    giq_option(4, 451, 559, Node990, 50);
+                    giq_option(4, NAME, 559, Node990, NEUTRAL_REACTION);
                 end
                 else begin
-                    giq_option(4, 451, 562, Node999, 50);
+                    giq_option(4, NAME, 562, Node999, NEUTRAL_REACTION);
                 end
             end
             else begin
                 if (item_caps_total(dude_obj) >= 100) then begin
                     info_cost := 100;
-                    giq_option(4, 451, 560, Node990, 50);
+                    giq_option(4, NAME, 560, Node990, NEUTRAL_REACTION);
                 end
                 else begin
-                    giq_option(4, 451, 562, Node999, 50);
+                    giq_option(4, NAME, 562, Node999, NEUTRAL_REACTION);
                 end
             end
         end
     end
-    giq_option(4, 451, 561, Node019, 50);
-    giq_option(4, 451, 563, Node999, 50);
+    giq_option(4, NAME, 561, Node019, NEUTRAL_REACTION);
+    giq_option(4, NAME, 563, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node025
 begin
-    gsay_reply(451, 575 + ((local_var(4) bwand 2048) != 0));
+    Reply(575 + ((local_var(4) bwand 2048) != 0));
     set_local_var(4, local_var(4) bwor 2048);
-    giq_option(4, 451, 577, Node026, 50);
-    giq_option(4, 451, 578, Node034, 50);
-    giq_option(4, 451, 579, Node048, 50);
-    giq_option(4, 451, 580, Node051, 50);
-    giq_option(4, 451, 581, Node019, 50);
-    giq_option(4, 451, 582, Node999, 50);
+    giq_option(4, NAME, 577, Node026, NEUTRAL_REACTION);
+    giq_option(4, NAME, 578, Node034, NEUTRAL_REACTION);
+    giq_option(4, NAME, 579, Node048, NEUTRAL_REACTION);
+    giq_option(4, NAME, 580, Node051, NEUTRAL_REACTION);
+    giq_option(4, NAME, 581, Node019, NEUTRAL_REACTION);
+    giq_option(4, NAME, 582, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node026
 begin
-    gsay_reply(451, 590);
-    giq_option(4, 451, 591, Node027, 50);
-    if (has_trait(0, dude_obj, 18) or has_skill(dude_obj, 12) or (get_critter_stat(dude_obj, 4) >= 8)) then begin
-        giq_option(4, 451, 592, Node027, 50);
+    Reply(590);
+    giq_option(4, NAME, 591, Node027, NEUTRAL_REACTION);
+    if (has_trait(TRAIT_PERK, dude_obj, PERK_educated) or dude_skill(SKILL_SCIENCE) or (dude_iq >= 8)) then begin
+        giq_option(4, NAME, 592, Node027, NEUTRAL_REACTION);
     end
-    giq_option(4, 451, 594, Node024, 50);
-    giq_option(4, 451, 581, Node019, 50);
-    giq_option(4, 451, 596, Node999, 50);
+    giq_option(4, NAME, 594, Node024, NEUTRAL_REACTION);
+    giq_option(4, NAME, 581, Node019, NEUTRAL_REACTION);
+    giq_option(4, NAME, 596, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node027
 begin
-    gsay_reply(451, 615);
-    giq_option(4, 451, 616, Node028, 50);
-    giq_option(4, 451, 594, Node024, 50);
-    giq_option(4, 451, 581, Node019, 50);
-    giq_option(4, 451, 520, Node999, 50);
+    Reply(615);
+    giq_option(4, NAME, 616, Node028, NEUTRAL_REACTION);
+    giq_option(4, NAME, 594, Node024, NEUTRAL_REACTION);
+    giq_option(4, NAME, 581, Node019, NEUTRAL_REACTION);
+    giq_option(4, NAME, 520, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node028
 begin
-    gsay_reply(451, 625);
-    giq_option(4, 451, 626, Node054, 50);
-    giq_option(4, 451, 627, Node029, 50);
-    giq_option(4, 451, 628, Node024, 50);
-    giq_option(4, 451, 581, Node019, 50);
-    giq_option(4, 451, 520, Node999, 50);
+    Reply(625);
+    giq_option(4, NAME, 626, Node054, NEUTRAL_REACTION);
+    giq_option(4, NAME, 627, Node029, NEUTRAL_REACTION);
+    giq_option(4, NAME, 628, Node024, NEUTRAL_REACTION);
+    giq_option(4, NAME, 581, Node019, NEUTRAL_REACTION);
+    giq_option(4, NAME, 520, Node999, NEUTRAL_REACTION);
     prev_node := 28;
 end
 
 procedure Node029
 begin
-    gsay_reply(451, 635 + (prev_node != 28));
-    giq_option(4, 451, 637, Node030, 50);
-    giq_option(4, 451, 638, Node031, 50);
-    giq_option(6, 451, 639, Node032, 50);
-    giq_option(4, 451, 628, Node024, 50);
-    giq_option(4, 451, 581, Node019, 50);
-    giq_option(4, 451, 520, Node999, 50);
+    Reply(635 + (prev_node != 28));
+    giq_option(4, NAME, 637, Node030, NEUTRAL_REACTION);
+    giq_option(4, NAME, 638, Node031, NEUTRAL_REACTION);
+    giq_option(6, NAME, 639, Node032, NEUTRAL_REACTION);
+    giq_option(4, NAME, 628, Node024, NEUTRAL_REACTION);
+    giq_option(4, NAME, 581, Node019, NEUTRAL_REACTION);
+    giq_option(4, NAME, 520, Node999, NEUTRAL_REACTION);
     prev_node := 29;
 end
 
 procedure Node030
 begin
     node_string := "Node030";
-    gsay_reply(451, 650);
-    giq_option(4, 451, 651, Node989, 50);
-    giq_option(4, 451, 652, Node029, 50);
-    giq_option(4, 451, 628, Node024, 50);
-    giq_option(4, 451, 581, Node019, 50);
-    giq_option(4, 451, 520, Node999, 50);
+    Reply(650);
+    giq_option(4, NAME, 651, Node989, NEUTRAL_REACTION);
+    giq_option(4, NAME, 652, Node029, NEUTRAL_REACTION);
+    giq_option(4, NAME, 628, Node024, NEUTRAL_REACTION);
+    giq_option(4, NAME, 581, Node019, NEUTRAL_REACTION);
+    giq_option(4, NAME, 520, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node031
 begin
-    set_global_var(284, global_var(284) bwor 1);
-    gsay_reply(451, 660);
-    giq_option(4, 451, 652, Node029, 50);
-    giq_option(4, 451, 662, Node057, 50);
-    giq_option(4, 451, 628, Node024, 50);
-    giq_option(4, 451, 581, Node019, 50);
-    giq_option(4, 451, 520, Node999, 50);
+    set_global_var(GVAR_NEW_RENO_MYRON, global_var(GVAR_NEW_RENO_MYRON) bwor 1);
+    Reply(660);
+    giq_option(4, NAME, 652, Node029, NEUTRAL_REACTION);
+    giq_option(4, NAME, 662, Node057, NEUTRAL_REACTION);
+    giq_option(4, NAME, 628, Node024, NEUTRAL_REACTION);
+    giq_option(4, NAME, 581, Node019, NEUTRAL_REACTION);
+    giq_option(4, NAME, 520, Node999, NEUTRAL_REACTION);
     prev_node := 31;
 end
 
 procedure Node032
 begin
-    gsay_reply(451, 670);
-    giq_option(4, 451, 652, Node029, 50);
-    giq_option(4, 451, 672, Node033, 50);
-    giq_option(4, 451, 628, Node024, 50);
-    giq_option(4, 451, 581, Node019, 50);
-    giq_option(4, 451, 520, Node999, 50);
+    Reply(670);
+    giq_option(4, NAME, 652, Node029, NEUTRAL_REACTION);
+    giq_option(4, NAME, 672, Node033, NEUTRAL_REACTION);
+    giq_option(4, NAME, 628, Node024, NEUTRAL_REACTION);
+    giq_option(4, NAME, 581, Node019, NEUTRAL_REACTION);
+    giq_option(4, NAME, 520, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node033
 begin
-    if (metarule(17, 13) == 0) then begin
+    if (town_known(13) == MARK_STATE_UNKNOWN) then begin
         debug_msg("  mark_on_map(" + 13 + ")");
-        mark_area_known(0, 13, 1);
+        mark_area_known(MARK_TYPE_TOWN, 13, MARK_STATE_KNOWN);
     end
-    gsay_reply(451, 685);
-    giq_option(4, 451, 652, Node029, 50);
-    giq_option(4, 451, 628, Node024, 50);
-    giq_option(4, 451, 581, Node019, 50);
-    giq_option(4, 451, 520, Node999, 50);
+    Reply(685);
+    giq_option(4, NAME, 652, Node029, NEUTRAL_REACTION);
+    giq_option(4, NAME, 628, Node024, NEUTRAL_REACTION);
+    giq_option(4, NAME, 581, Node019, NEUTRAL_REACTION);
+    giq_option(4, NAME, 520, Node999, NEUTRAL_REACTION);
     prev_node := 33;
 end
 
 procedure Node034
 begin
-    gsay_reply(451, 695);
-    giq_option(5, 451, 696, Node035, 50);
-    giq_option(4, 451, 628, Node024, 50);
-    giq_option(4, 451, 581, Node019, 50);
-    giq_option(4, 451, 520, Node999, 50);
+    Reply(695);
+    giq_option(5, NAME, 696, Node035, NEUTRAL_REACTION);
+    giq_option(4, NAME, 628, Node024, NEUTRAL_REACTION);
+    giq_option(4, NAME, 581, Node019, NEUTRAL_REACTION);
+    giq_option(4, NAME, 520, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node035
 begin
-    gsay_reply(451, 710);
-    giq_option(5, 451, 711, Node036, 50);
-    giq_option(5, 451, 712, Node036, 50);
-    giq_option(4, 451, 628, Node024, 50);
-    giq_option(4, 451, 581, Node019, 50);
-    giq_option(4, 451, 715, Node999, 50);
+    Reply(710);
+    giq_option(5, NAME, 711, Node036, NEUTRAL_REACTION);
+    giq_option(5, NAME, 712, Node036, NEUTRAL_REACTION);
+    giq_option(4, NAME, 628, Node024, NEUTRAL_REACTION);
+    giq_option(4, NAME, 581, Node019, NEUTRAL_REACTION);
+    giq_option(4, NAME, 715, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node036
 begin
-    set_global_var(314, global_var(314) bwor 16384);
-    gsay_reply(451, 725);
-    giq_option(6, 451, 726, Node037, 50);
-    giq_option(4, 451, 727, Node024, 50);
-    giq_option(4, 451, 581, Node019, 50);
-    giq_option(4, 451, 520, Node999, 50);
+    set_global_var(GVAR_NEW_RENO_FLAG_1, global_var(GVAR_NEW_RENO_FLAG_1) bwor 16384);
+    Reply(725);
+    giq_option(6, NAME, 726, Node037, NEUTRAL_REACTION);
+    giq_option(4, NAME, 727, Node024, NEUTRAL_REACTION);
+    giq_option(4, NAME, 581, Node019, NEUTRAL_REACTION);
+    giq_option(4, NAME, 520, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node037
 begin
-    gsay_reply(451, 735);
-    giq_option(8, 451, 736, Node041, 50);
-    giq_option(7, 451, 737, Node038, 50);
-    giq_option(4, 451, 628, Node024, 50);
-    giq_option(4, 451, 581, Node019, 50);
-    giq_option(4, 451, 520, Node999, 50);
+    Reply(735);
+    giq_option(8, NAME, 736, Node041, NEUTRAL_REACTION);
+    giq_option(7, NAME, 737, Node038, NEUTRAL_REACTION);
+    giq_option(4, NAME, 628, Node024, NEUTRAL_REACTION);
+    giq_option(4, NAME, 581, Node019, NEUTRAL_REACTION);
+    giq_option(4, NAME, 520, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node038
 begin
-    gsay_reply(451, 750);
-    giq_option(7, 451, 751, Node039, 50);
-    giq_option(4, 451, 628, Node024, 50);
-    giq_option(4, 451, 581, Node019, 50);
-    giq_option(4, 451, 520, Node999, 50);
+    Reply(750);
+    giq_option(7, NAME, 751, Node039, NEUTRAL_REACTION);
+    giq_option(4, NAME, 628, Node024, NEUTRAL_REACTION);
+    giq_option(4, NAME, 581, Node019, NEUTRAL_REACTION);
+    giq_option(4, NAME, 520, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node039
 begin
-    gsay_reply(451, 765);
-    giq_option(8, 451, 766, Node041, 50);
-    giq_option(7, 451, 767, Node040, 50);
-    giq_option(4, 451, 628, Node024, 50);
-    giq_option(4, 451, 581, Node019, 50);
-    giq_option(4, 451, 520, Node999, 50);
+    Reply(765);
+    giq_option(8, NAME, 766, Node041, NEUTRAL_REACTION);
+    giq_option(7, NAME, 767, Node040, NEUTRAL_REACTION);
+    giq_option(4, NAME, 628, Node024, NEUTRAL_REACTION);
+    giq_option(4, NAME, 581, Node019, NEUTRAL_REACTION);
+    giq_option(4, NAME, 520, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node040
 begin
-    gsay_reply(451, 780);
-    giq_option(4, 451, 781, Node065, 49);
-    giq_option(4, 451, 782, Node024, 50);
-    giq_option(4, 451, 581, Node019, 50);
-    giq_option(4, 451, 520, Node999, 50);
+    Reply(780);
+    giq_option(4, NAME, 781, Node065, GOOD_REACTION);
+    giq_option(4, NAME, 782, Node024, NEUTRAL_REACTION);
+    giq_option(4, NAME, 581, Node019, NEUTRAL_REACTION);
+    giq_option(4, NAME, 520, Node999, NEUTRAL_REACTION);
     prev_node := 40;
 end
 
 procedure Node041
 begin
-    gsay_reply(451, 795);
-    giq_option(7, 451, 796, Node042, 50);
-    giq_option(4, 451, 782, Node024, 50);
-    giq_option(4, 451, 581, Node019, 50);
-    giq_option(4, 451, 520, Node999, 50);
+    Reply(795);
+    giq_option(7, NAME, 796, Node042, NEUTRAL_REACTION);
+    giq_option(4, NAME, 782, Node024, NEUTRAL_REACTION);
+    giq_option(4, NAME, 581, Node019, NEUTRAL_REACTION);
+    giq_option(4, NAME, 520, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node042
 begin
-    if (metarule(17, 48) == 0) then begin
+    if (town_known(48) == MARK_STATE_UNKNOWN) then begin
         debug_msg("  mark_on_map(" + 48 + ")");
-        mark_area_known(0, 48, 1);
+        mark_area_known(MARK_TYPE_TOWN, 48, MARK_STATE_KNOWN);
     end
-    gsay_reply(451, 810);
-    giq_option(4, 451, 811, Node043, 50);
-    giq_option(4, 451, 782, Node024, 50);
-    giq_option(4, 451, 581, Node019, 50);
-    giq_option(4, 451, 520, Node999, 50);
+    Reply(810);
+    giq_option(4, NAME, 811, Node043, NEUTRAL_REACTION);
+    giq_option(4, NAME, 782, Node024, NEUTRAL_REACTION);
+    giq_option(4, NAME, 581, Node019, NEUTRAL_REACTION);
+    giq_option(4, NAME, 520, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node043
 begin
-    gsay_reply(451, 825);
-    giq_option(7, 451, 826, Node045, 50);
-    giq_option(7, 451, 827, Node044, 50);
-    giq_option(4, 451, 828, Node046, 50);
-    giq_option(4, 451, 782, Node024, 50);
-    giq_option(4, 451, 581, Node019, 50);
-    giq_option(4, 451, 520, Node999, 50);
+    Reply(825);
+    giq_option(7, NAME, 826, Node045, NEUTRAL_REACTION);
+    giq_option(7, NAME, 827, Node044, NEUTRAL_REACTION);
+    giq_option(4, NAME, 828, Node046, NEUTRAL_REACTION);
+    giq_option(4, NAME, 782, Node024, NEUTRAL_REACTION);
+    giq_option(4, NAME, 581, Node019, NEUTRAL_REACTION);
+    giq_option(4, NAME, 520, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node044
 begin
-    gsay_reply(451, message_str(451, 840) + message_str(451, 841 + (global_var(11) == 1)) + message_str(451, 843));
-    giq_option(7, 451, 826, Node045, 50);
-    giq_option(4, 451, 828, Node046, 50);
-    giq_option(4, 451, 782, Node024, 50);
-    giq_option(4, 451, 581, Node019, 50);
-    giq_option(4, 451, 520, Node999, 50);
+    Reply(mstr(840) + mstr(841 + (global_var(GVAR_REPUTATION_SLAVER) == 1)) + mstr(843));
+    giq_option(7, NAME, 826, Node045, NEUTRAL_REACTION);
+    giq_option(4, NAME, 828, Node046, NEUTRAL_REACTION);
+    giq_option(4, NAME, 782, Node024, NEUTRAL_REACTION);
+    giq_option(4, NAME, 581, Node019, NEUTRAL_REACTION);
+    giq_option(4, NAME, 520, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node045
 begin
-    gsay_reply(451, 860);
-    giq_option(4, 451, 828, Node046, 50);
-    giq_option(7, 451, 827, Node044, 50);
-    giq_option(4, 451, 782, Node024, 50);
-    giq_option(4, 451, 581, Node019, 50);
-    giq_option(4, 451, 520, Node999, 50);
+    Reply(860);
+    giq_option(4, NAME, 828, Node046, NEUTRAL_REACTION);
+    giq_option(7, NAME, 827, Node044, NEUTRAL_REACTION);
+    giq_option(4, NAME, 782, Node024, NEUTRAL_REACTION);
+    giq_option(4, NAME, 581, Node019, NEUTRAL_REACTION);
+    giq_option(4, NAME, 520, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node046
 begin
-    gsay_reply(451, 880);
-    giq_option(7, 451, 881, Node047, 50);
-    giq_option(7, 451, 827, Node044, 50);
-    giq_option(7, 451, 826, Node045, 50);
-    giq_option(4, 451, 782, Node024, 50);
-    giq_option(4, 451, 581, Node019, 50);
-    giq_option(4, 451, 886, Node999, 50);
+    Reply(880);
+    giq_option(7, NAME, 881, Node047, NEUTRAL_REACTION);
+    giq_option(7, NAME, 827, Node044, NEUTRAL_REACTION);
+    giq_option(7, NAME, 826, Node045, NEUTRAL_REACTION);
+    giq_option(4, NAME, 782, Node024, NEUTRAL_REACTION);
+    giq_option(4, NAME, 581, Node019, NEUTRAL_REACTION);
+    giq_option(4, NAME, 886, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node047
 begin
-    gsay_reply(451, 895);
-    giq_option(7, 451, 827, Node044, 50);
-    giq_option(7, 451, 826, Node045, 50);
-    giq_option(4, 451, 782, Node024, 50);
-    giq_option(4, 451, 581, Node019, 50);
-    giq_option(4, 451, 520, Node999, 50);
+    Reply(895);
+    giq_option(7, NAME, 827, Node044, NEUTRAL_REACTION);
+    giq_option(7, NAME, 826, Node045, NEUTRAL_REACTION);
+    giq_option(4, NAME, 782, Node024, NEUTRAL_REACTION);
+    giq_option(4, NAME, 581, Node019, NEUTRAL_REACTION);
+    giq_option(4, NAME, 520, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node048
 begin
-    gsay_reply(451, 910);
-    giq_option(4, 451, 911 + (global_var(0) >= 0), Node049, 50);
-    giq_option(4, 451, 782, Node024, 50);
-    giq_option(4, 451, 581, Node019, 50);
-    giq_option(4, 451, 520, Node999, 50);
+    Reply(910);
+    giq_option(4, NAME, 911 + (global_var(GVAR_PLAYER_REPUTATION) >= 0), Node049, NEUTRAL_REACTION);
+    giq_option(4, NAME, 782, Node024, NEUTRAL_REACTION);
+    giq_option(4, NAME, 581, Node019, NEUTRAL_REACTION);
+    giq_option(4, NAME, 520, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node049
 begin
-    gsay_reply(451, 925);
-    giq_option(4, 451, 926 + (get_critter_stat(dude_obj, 34) == 1), Node050, 50);
-    giq_option(4, 451, 782, Node024, 50);
-    giq_option(4, 451, 581, Node019, 50);
-    giq_option(4, 451, 520, Node999, 50);
+    Reply(925);
+    giq_option(4, NAME, 926 + (dude_gender == 1), Node050, NEUTRAL_REACTION);
+    giq_option(4, NAME, 782, Node024, NEUTRAL_REACTION);
+    giq_option(4, NAME, 581, Node019, NEUTRAL_REACTION);
+    giq_option(4, NAME, 520, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node050
 begin
-    gsay_reply(451, 935 + (get_critter_stat(dude_obj, 34) == 1));
-    giq_option(4, 451, 937, Node024, 50);
-    giq_option(4, 451, 581, Node019, 50);
-    giq_option(4, 451, 520, Node999, 50);
+    Reply(935 + (dude_gender == 1));
+    giq_option(4, NAME, 937, Node024, NEUTRAL_REACTION);
+    giq_option(4, NAME, 581, Node019, NEUTRAL_REACTION);
+    giq_option(4, NAME, 520, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node051
 begin
-    gsay_reply(451, 945 + (global_var(11) == 1));
-    giq_option(4, 451, 947, Node052, 50);
-    giq_option(4, 451, 782, Node024, 50);
-    giq_option(4, 451, 581, Node019, 50);
-    giq_option(4, 451, 520, Node999, 50);
+    Reply(945 + (global_var(GVAR_REPUTATION_SLAVER) == 1));
+    giq_option(4, NAME, 947, Node052, NEUTRAL_REACTION);
+    giq_option(4, NAME, 782, Node024, NEUTRAL_REACTION);
+    giq_option(4, NAME, 581, Node019, NEUTRAL_REACTION);
+    giq_option(4, NAME, 520, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node052
 begin
-    gsay_reply(451, 960 + (global_var(11) == 1));
-    giq_option(4, 451, 962, Node053, 50);
-    giq_option(4, 451, 782, Node024, 50);
-    giq_option(4, 451, 581, Node019, 50);
-    giq_option(4, 451, 520, Node999, 50);
+    Reply(960 + (global_var(GVAR_REPUTATION_SLAVER) == 1));
+    giq_option(4, NAME, 962, Node053, NEUTRAL_REACTION);
+    giq_option(4, NAME, 782, Node024, NEUTRAL_REACTION);
+    giq_option(4, NAME, 581, Node019, NEUTRAL_REACTION);
+    giq_option(4, NAME, 520, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node053
 begin
-    set_global_var(285, global_var(285) bwor 256);
-    gsay_reply(451, 975);
-    giq_option(4, 451, 782, Node024, 50);
-    giq_option(4, 451, 581, Node019, 50);
-    giq_option(4, 451, 520, Node999, 50);
+    set_global_var(GVAR_NEW_RENO_WRIGHT_FLAGS, global_var(GVAR_NEW_RENO_WRIGHT_FLAGS) bwor 256);
+    Reply(975);
+    giq_option(4, NAME, 782, Node024, NEUTRAL_REACTION);
+    giq_option(4, NAME, 581, Node019, NEUTRAL_REACTION);
+    giq_option(4, NAME, 520, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node054
 begin
     node_string := "Node054";
-    gsay_reply(451, message_str(451, 985 + ((1 + (get_critter_stat(dude_obj, 3) > 5)) * (get_critter_stat(dude_obj, 34) == 1))) + message_str(451, 988));
-    giq_option(4, 451, 989, Node989, 49);
-    giq_option(4, 451, 990, Node055, 50);
-    giq_option(4, 451, 991, Node019, 50);
-    giq_option(4, 451, 992, Node999, 50);
+    Reply(mstr(985 + ((1 + (dude_charisma > 5)) * (dude_gender == 1))) + mstr(988));
+    giq_option(4, NAME, 989, Node989, GOOD_REACTION);
+    giq_option(4, NAME, 990, Node055, NEUTRAL_REACTION);
+    giq_option(4, NAME, 991, Node019, NEUTRAL_REACTION);
+    giq_option(4, NAME, 992, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node055
 begin
     node_string := "Node055";
-    gsay_reply(451, 1005 + (get_critter_stat(dude_obj, 34) == 1));
-    giq_option(4, 451, 989, Node989, 49);
-    giq_option(5, 451, 1008, Node056, 50);
-    giq_option(4, 451, 991, Node019, 50);
-    giq_option(4, 451, 992, Node999, 50);
+    Reply(1005 + (dude_gender == 1));
+    giq_option(4, NAME, 989, Node989, GOOD_REACTION);
+    giq_option(5, NAME, 1008, Node056, NEUTRAL_REACTION);
+    giq_option(4, NAME, 991, Node019, NEUTRAL_REACTION);
+    giq_option(4, NAME, 992, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node056
 begin
     node_string := "Node056";
-    gsay_reply(451, 1025);
-    giq_option(4, 451, 1026, Node989, 49);
-    giq_option(6, 451, 1027, Node057, 50);
-    giq_option(4, 451, 547, Node019, 50);
-    giq_option(4, 451, 992, Node999, 50);
+    Reply(1025);
+    giq_option(4, NAME, 1026, Node989, GOOD_REACTION);
+    giq_option(6, NAME, 1027, Node057, NEUTRAL_REACTION);
+    giq_option(4, NAME, 547, Node019, NEUTRAL_REACTION);
+    giq_option(4, NAME, 992, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node057
 begin
-    set_global_var(345, global_var(345) bwor 16);
-    set_global_var(284, global_var(284) bwor 1);
-    gsay_reply(451, 1035);
-    giq_option(4, 451, 1036, Node058, 50);
-    giq_option(4, 451, 547, Node019, 50);
-    giq_option(4, 451, 992, Node999, 50);
+    set_global_var(GVAR_NEW_RENO_FLAG_2, global_var(GVAR_NEW_RENO_FLAG_2) bwor 16);
+    set_global_var(GVAR_NEW_RENO_MYRON, global_var(GVAR_NEW_RENO_MYRON) bwor 1);
+    Reply(1035);
+    giq_option(4, NAME, 1036, Node058, NEUTRAL_REACTION);
+    giq_option(4, NAME, 547, Node019, NEUTRAL_REACTION);
+    giq_option(4, NAME, 992, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node058
 begin
-    if (metarule(17, 48) == 0) then begin
+    if (town_known(48) == MARK_STATE_UNKNOWN) then begin
         debug_msg("  mark_on_map(" + 48 + ")");
-        mark_area_known(0, 48, 1);
+        mark_area_known(MARK_TYPE_TOWN, 48, MARK_STATE_KNOWN);
     end
-    gsay_reply(451, 1045);
-    giq_option(4, 451, 1046, Node059, 50);
-    giq_option(4, 451, 547, Node019, 50);
-    giq_option(4, 451, 992, Node999, 50);
+    Reply(1045);
+    giq_option(4, NAME, 1046, Node059, NEUTRAL_REACTION);
+    giq_option(4, NAME, 547, Node019, NEUTRAL_REACTION);
+    giq_option(4, NAME, 992, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node059
 begin
-    gsay_reply(451, 1055);
-    giq_option(4, 451, 1056, Node988, 51);
-    giq_option(4, 451, 547, Node019, 50);
-    giq_option(4, 451, 992, Node999, 50);
+    Reply(1055);
+    giq_option(4, NAME, 1056, Node988, BAD_REACTION);
+    giq_option(4, NAME, 547, Node019, NEUTRAL_REACTION);
+    giq_option(4, NAME, 992, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node060
 begin
-    gsay_reply(451, 1065);
-    giq_option(4, 451, 1066, Node061, 50);
+    Reply(1065);
+    giq_option(4, NAME, 1066, Node061, NEUTRAL_REACTION);
 end
 
 procedure Node061
 begin
-    if (get_critter_stat(dude_obj, 34) == 0) then begin
-        gsay_reply(451, 1075);
-        giq_option(4, 451, 1078, Node064, 50);
+    if (dude_gender == 0) then begin
+        Reply(1075);
+        giq_option(4, NAME, 1078, Node064, NEUTRAL_REACTION);
     end
     else begin
-        gsay_reply(451, 1076 + (get_critter_stat(dude_obj, 3) > 5));
-        giq_option(4, 451, 1079 + (get_critter_stat(dude_obj, 3) <= 5), Node063, 50);
+        Reply(1076 + (dude_charisma > 5));
+        giq_option(4, NAME, 1079 + (dude_charisma <= 5), Node063, NEUTRAL_REACTION);
     end
-    giq_option(4, 451, 1081, Node062, 50);
-    giq_option(4, 451, 547, Node019, 50);
-    giq_option(4, 451, 1083, Node999, 50);
+    giq_option(4, NAME, 1081, Node062, NEUTRAL_REACTION);
+    giq_option(4, NAME, 547, Node019, NEUTRAL_REACTION);
+    giq_option(4, NAME, 1083, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node062
 begin
-    if (metarule(17, 30) == 0) then begin
+    if (town_known(30) == MARK_STATE_UNKNOWN) then begin
         debug_msg("  mark_on_map(" + 30 + ")");
-        mark_area_known(0, 30, 1);
+        mark_area_known(MARK_TYPE_TOWN, 30, MARK_STATE_KNOWN);
     end
-    gsay_reply(451, 545);
-    if (get_critter_stat(dude_obj, 34) == 0) then begin
-        giq_option(4, 451, 1101, Node064, 50);
+    Reply(545);
+    if (dude_gender == 0) then begin
+        giq_option(4, NAME, 1101, Node064, NEUTRAL_REACTION);
     end
     else begin
-        giq_option(4, 451, 1102 + (get_critter_stat(dude_obj, 3) <= 5), Node063, 50);
+        giq_option(4, NAME, 1102 + (dude_charisma <= 5), Node063, NEUTRAL_REACTION);
     end
-    giq_option(4, 451, 547, Node019, 50);
-    giq_option(4, 451, 548, Node999, 50);
+    giq_option(4, NAME, 547, Node019, NEUTRAL_REACTION);
+    giq_option(4, NAME, 548, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node063
 begin
-    gsay_reply(451, message_str(451, 1115 + (get_critter_stat(dude_obj, 3) > 5)) + message_str(451, 1117));
-    giq_option(4, 451, 1118, Node062, 50);
-    giq_option(4, 451, 1119, Node019, 50);
-    giq_option(4, 451, 1120 + (get_critter_stat(dude_obj, 4) >= 7), Node999, 50);
+    Reply(mstr(1115 + (dude_charisma > 5)) + mstr(1117));
+    giq_option(4, NAME, 1118, Node062, NEUTRAL_REACTION);
+    giq_option(4, NAME, 1119, Node019, NEUTRAL_REACTION);
+    giq_option(4, NAME, 1120 + (dude_iq >= 7), Node999, NEUTRAL_REACTION);
 end
 
 procedure Node064
 begin
-    gsay_reply(451, 1130);
-    giq_option(4, 451, 1118, Node062, 50);
-    giq_option(4, 451, 1132, Node019, 50);
-    giq_option(4, 451, 1120, Node999, 50);
+    Reply(1130);
+    giq_option(4, NAME, 1118, Node062, NEUTRAL_REACTION);
+    giq_option(4, NAME, 1132, Node019, NEUTRAL_REACTION);
+    giq_option(4, NAME, 1120, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node065
 begin
-    if (global_var(349) == 0) then begin
-        set_global_var(349, 1);
+    if (global_var(GVAR_NEW_RENO_JULES_ELDRIDGE) == 0) then begin
+        set_global_var(GVAR_NEW_RENO_JULES_ELDRIDGE, 1);
     end
     if (prev_node == 40) then begin
-        gsay_reply(451, 1145);
+        Reply(1145);
     end
     else begin
-        gsay_reply(451, 1146 + ((1 + (get_critter_stat(dude_obj, 3) < 5)) * (get_critter_stat(dude_obj, 34) == 1)));
+        Reply(1146 + ((1 + (dude_charisma < 5)) * (dude_gender == 1)));
     end
-    giq_option(4, 451, 1132, Node019, 50);
-    giq_option(4, 451, 1150, Node999, 50);
+    giq_option(4, NAME, 1132, Node019, NEUTRAL_REACTION);
+    giq_option(4, NAME, 1150, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node066
 begin
-    set_global_var(314, global_var(314) bwor 128);
-    gsay_reply(451, 1165);
-    giq_option(4, 451, 1166, Node019, 50);
-    giq_option(4, 451, 1167, Node999, 50);
+    set_global_var(GVAR_NEW_RENO_FLAG_1, global_var(GVAR_NEW_RENO_FLAG_1) bwor 128);
+    Reply(1165);
+    giq_option(4, NAME, 1166, Node019, NEUTRAL_REACTION);
+    giq_option(4, NAME, 1167, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node067
 begin
-    if (get_critter_stat(dude_obj, 34) == 0) then begin
-        if (global_var(347) == 0) then begin
-            set_global_var(347, 1);
+    if (dude_gender == 0) then begin
+        if (global_var(GVAR_NEW_RENO_JULES_KITTY) == 0) then begin
+            set_global_var(GVAR_NEW_RENO_JULES_KITTY, 1);
         end
-        gsay_reply(451, 1175);
-        giq_option(4, 451, 1166, Node019, 51);
-        giq_option(4, 451, 1180, Node999, 50);
+        Reply(1175);
+        giq_option(4, NAME, 1166, Node019, BAD_REACTION);
+        giq_option(4, NAME, 1180, Node999, NEUTRAL_REACTION);
     end
     else begin
-        if (get_critter_stat(dude_obj, 3) < 6) then begin
-            gsay_reply(451, 1176);
-            giq_option(4, 451, 1166, Node019, 51);
-            giq_option(4, 451, 1180, Node999, 50);
+        if (dude_charisma < 6) then begin
+            Reply(1176);
+            giq_option(4, NAME, 1166, Node019, BAD_REACTION);
+            giq_option(4, NAME, 1180, Node999, NEUTRAL_REACTION);
         end
         else begin
-            if (global_var(347) == 0) then begin
-                set_global_var(347, 1);
+            if (global_var(GVAR_NEW_RENO_JULES_KITTY) == 0) then begin
+                set_global_var(GVAR_NEW_RENO_JULES_KITTY, 1);
             end
-            gsay_reply(451, 1177);
-            giq_option(4, 451, 1178, Node019, 51);
-            giq_option(4, 451, 1181, Node999, 50);
+            Reply(1177);
+            giq_option(4, NAME, 1178, Node019, BAD_REACTION);
+            giq_option(4, NAME, 1181, Node999, NEUTRAL_REACTION);
         end
     end
 end
 
 procedure Node068
 begin
-    if (metarule(17, 4) == 0) then begin
+    if (town_known(4) == MARK_STATE_UNKNOWN) then begin
         debug_msg("  mark_on_map(" + 4 + ")");
-        mark_area_known(0, 4, 1);
+        mark_area_known(MARK_TYPE_TOWN, 4, MARK_STATE_KNOWN);
     end
-    gsay_reply(451, 1190);
-    giq_option(4, 451, 1191, Node019, 50);
-    giq_option(4, 451, 1192, Node999, 50);
+    Reply(1190);
+    giq_option(4, NAME, 1191, Node019, NEUTRAL_REACTION);
+    giq_option(4, NAME, 1192, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node069
 begin
-    gsay_reply(451, 1200);
-    giq_option(4, 451, 1201, Node019, 50);
-    giq_option(4, 451, 1202, Node999, 50);
+    Reply(1200);
+    giq_option(4, NAME, 1201, Node019, NEUTRAL_REACTION);
+    giq_option(4, NAME, 1202, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node070
 begin
-    gsay_reply(451, 1210);
-    giq_option(4, 451, 1211, Node019, 50);
-    giq_option(4, 451, 1202, Node999, 50);
+    Reply(1210);
+    giq_option(4, NAME, 1211, Node019, NEUTRAL_REACTION);
+    giq_option(4, NAME, 1202, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node071
 begin
-    gsay_reply(451, 1220);
-    giq_option(4, 451, 1221, Node019, 50);
-    giq_option(4, 451, 1202, Node999, 50);
+    Reply(1220);
+    giq_option(4, NAME, 1221, Node019, NEUTRAL_REACTION);
+    giq_option(4, NAME, 1202, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node072
 begin
-    gsay_reply(451, 1230);
-    if ((global_var(314) bwand 512) != 0) then begin
-        giq_option(4, 451, 1231, Node089, 51);
+    Reply(1230);
+    if ((global_var(GVAR_NEW_RENO_FLAG_1) bwand 512) != 0) then begin
+        giq_option(4, NAME, 1231, Node089, BAD_REACTION);
     end
-    giq_option(4, 451, 1232, Node073, 50);
+    giq_option(4, NAME, 1232, Node073, NEUTRAL_REACTION);
     if (obj_is_carrying_obj_pid(dude_obj, 416)) then begin
-        giq_option(4, 451, 1233, Node086, 50);
+        giq_option(4, NAME, 1233, Node086, NEUTRAL_REACTION);
     end
-    if ((global_var(314) bwand 1024) != 0) then begin
-        giq_option(4, 451, 1234, Node085, 50);
+    if ((global_var(GVAR_NEW_RENO_FLAG_1) bwand 1024) != 0) then begin
+        giq_option(4, NAME, 1234, Node085, NEUTRAL_REACTION);
     end
-    giq_option(4, 451, 528, Node019, 50);
-    giq_option(4, 451, 1202, Node999, 50);
+    giq_option(4, NAME, 528, Node019, NEUTRAL_REACTION);
+    giq_option(4, NAME, 1202, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node073
 begin
-    set_global_var(314, global_var(314) bwor 128);
-    gsay_reply(451, 1250);
-    if ((global_var(314) bwand 512) != 0) then begin
-        giq_option(4, 451, 1251, Node089, 51);
+    set_global_var(GVAR_NEW_RENO_FLAG_1, global_var(GVAR_NEW_RENO_FLAG_1) bwor 128);
+    Reply(1250);
+    if ((global_var(GVAR_NEW_RENO_FLAG_1) bwand 512) != 0) then begin
+        giq_option(4, NAME, 1251, Node089, BAD_REACTION);
     end
-    giq_option(6, 451, 1252, Node074, 50);
-    giq_option(4, 451, 1253, Node019, 50);
-    giq_option(4, 451, 1254, Node999, 50);
+    giq_option(6, NAME, 1252, Node074, NEUTRAL_REACTION);
+    giq_option(4, NAME, 1253, Node019, NEUTRAL_REACTION);
+    giq_option(4, NAME, 1254, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node074
 begin
-    set_global_var(345, global_var(345) bwor 67108864);
-    if (global_var(543) == 0) then begin
-        set_global_var(543, 1);
+    set_global_var(GVAR_NEW_RENO_FLAG_2, global_var(GVAR_NEW_RENO_FLAG_2) bwor 67108864);
+    if (global_var(GVAR_NEW_RENO_SUSPECT_JJJ) == 0) then begin
+        set_global_var(GVAR_NEW_RENO_SUSPECT_JJJ, 1);
     end
-    if (global_var(545) == 0) then begin
-        set_global_var(545, 1);
+    if (global_var(GVAR_NEW_RENO_SUSPECT_LIL_JESUS) == 0) then begin
+        set_global_var(GVAR_NEW_RENO_SUSPECT_LIL_JESUS, 1);
     end
-    if (global_var(546) == 0) then begin
-        set_global_var(546, 1);
+    if (global_var(GVAR_NEW_RENO_SUSPECT_RENESCO) == 0) then begin
+        set_global_var(GVAR_NEW_RENO_SUSPECT_RENESCO, 1);
     end
-    gsay_reply(451, 1270);
-    if ((global_var(314) bwand 512) != 0) then begin
-        giq_option(4, 451, 1251, Node089, 51);
+    Reply(1270);
+    if ((global_var(GVAR_NEW_RENO_FLAG_1) bwand 512) != 0) then begin
+        giq_option(4, NAME, 1251, Node089, BAD_REACTION);
     end
-    giq_option(4, 451, 1272, Node083, 50);
-    giq_option(4, 451, 1273, Node084, 50);
-    giq_option(4, 451, 1274, Node019, 50);
-    giq_option(4, 451, 1275, Node999, 50);
-    giq_option(4, 451, 1254, Node999, 50);
+    giq_option(4, NAME, 1272, Node083, NEUTRAL_REACTION);
+    giq_option(4, NAME, 1273, Node084, NEUTRAL_REACTION);
+    giq_option(4, NAME, 1274, Node019, NEUTRAL_REACTION);
+    giq_option(4, NAME, 1275, Node999, NEUTRAL_REACTION);
+    giq_option(4, NAME, 1254, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node075
 begin
-    gsay_reply(451, 1290);
-    giq_option(4, 451, 1291, Node019, 50);
-    giq_option(4, 451, 1292, Node999, 50);
+    Reply(1290);
+    giq_option(4, NAME, 1291, Node019, NEUTRAL_REACTION);
+    giq_option(4, NAME, 1292, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node076
 begin
-    gsay_reply(451, 1305);
-    giq_option(4, 451, 1306, Node078, 51);
-    giq_option(4, 451, 1308, Node987, 50);
-    giq_option(4, 451, 1310, Node019, 50);
-    giq_option(4, 451, 1311, Node999, 50);
+    Reply(1305);
+    giq_option(4, NAME, 1306, Node078, BAD_REACTION);
+    giq_option(4, NAME, 1308, Node987, NEUTRAL_REACTION);
+    giq_option(4, NAME, 1310, Node019, NEUTRAL_REACTION);
+    giq_option(4, NAME, 1311, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node077
 begin
-    gsay_reply(451, message_str(451, 1320 + (get_critter_stat(dude_obj, 34) == 1)) + message_str(451, 1322));
-    if ((global_var(348) == 1) and (((global_var(345) bwand 8192) != 0) or ((global_var(345) bwand 4096) != 0))) then begin
-        giq_option(4, 451, 1323, Node078, 51);
+    Reply(mstr(1320 + (dude_gender == 1)) + mstr(1322));
+    if ((global_var(GVAR_NEW_RENO_STOLEN_CAR) == 1) and (((global_var(GVAR_NEW_RENO_FLAG_2) bwand 8192) != 0) or ((global_var(GVAR_NEW_RENO_FLAG_2) bwand 4096) != 0))) then begin
+        giq_option(4, NAME, 1323, Node078, BAD_REACTION);
     end
-    giq_option(4, 451, 1324, Node019, 50);
-    giq_option(4, 451, 1202, Node999, 50);
+    giq_option(4, NAME, 1324, Node019, NEUTRAL_REACTION);
+    giq_option(4, NAME, 1202, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node078
 begin
-    if (((global_var(345) bwand 8192) != 0) and (((global_var(345) bwand 16384) != 0) == 0)) then begin
-        gsay_reply(451, message_str(451, 1340 + (get_critter_stat(dude_obj, 34) == 0)) + message_str(451, 1342) + message_str(451, 1343));
+    if (((global_var(GVAR_NEW_RENO_FLAG_2) bwand 8192) != 0) and (((global_var(GVAR_NEW_RENO_FLAG_2) bwand 16384) != 0) == 0)) then begin
+        Reply(mstr(1340 + (dude_gender == 0)) + mstr(1342) + mstr(1343));
     end
     else begin
-        gsay_reply(451, message_str(451, 1340 + (get_critter_stat(dude_obj, 34) == 0)) + message_str(451, 1342));
+        Reply(mstr(1340 + (dude_gender == 0)) + mstr(1342));
     end
-    giq_option(4, 451, 1344, Node079, 51);
-    if (((global_var(345) bwand 8192) != 0) or ((global_var(345) bwand 4096) != 0)) then begin
-        giq_option(4, 451, 1345, Node079, 51);
+    giq_option(4, NAME, 1344, Node079, BAD_REACTION);
+    if (((global_var(GVAR_NEW_RENO_FLAG_2) bwand 8192) != 0) or ((global_var(GVAR_NEW_RENO_FLAG_2) bwand 4096) != 0)) then begin
+        giq_option(4, NAME, 1345, Node079, BAD_REACTION);
     end
-    giq_option(4, 451, 1346, Node999, 50);
+    giq_option(4, NAME, 1346, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node079
 begin
-    gsay_reply(451, message_str(451, 1360 + (get_critter_stat(dude_obj, 34) == 1)) + message_str(451, 1362));
-    giq_option(4, 451, 1363, Node080, 51);
-    giq_option(4, 451, 1364, Node999, 50);
+    Reply(mstr(1360 + (dude_gender == 1)) + mstr(1362));
+    giq_option(4, NAME, 1363, Node080, BAD_REACTION);
+    giq_option(4, NAME, 1364, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node080
 begin
-    gsay_reply(451, 1375);
-    if (((global_var(345) bwand 8192) != 0) and (((global_var(345) bwand 16384) != 0) == 0)) then begin
-        giq_option(4, 451, 1376 + (get_critter_stat(dude_obj, 4) < 7), Node986, 51);
+    Reply(1375);
+    if (((global_var(GVAR_NEW_RENO_FLAG_2) bwand 8192) != 0) and (((global_var(GVAR_NEW_RENO_FLAG_2) bwand 16384) != 0) == 0)) then begin
+        giq_option(4, NAME, 1376 + (dude_iq < 7), Node986, BAD_REACTION);
     end
-    giq_option(4, 451, 1378, Node985, 50);
+    giq_option(4, NAME, 1378, Node985, NEUTRAL_REACTION);
 end
 
 procedure Node081
 begin
-    gsay_reply(451, 1385 + ((global_var(345) bwand 2) != 0));
-    giq_option(-3, 451, 463, Node018, 50);
-    giq_option(-3, 451, 464, Node999, 50);
-    giq_option(4, 451, 1389, Node082, 50);
-    giq_option(4, 451, 1390 + (((global_var(345) bwand 32768) != 0) == 0), Node997, 50);
-    giq_option(4, 451, 472, Node999, 50);
+    Reply(1385 + ((global_var(GVAR_NEW_RENO_FLAG_2) bwand 2) != 0));
+    giq_option(-3, NAME, 463, Node018, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 464, Node999, NEUTRAL_REACTION);
+    giq_option(4, NAME, 1389, Node082, NEUTRAL_REACTION);
+    giq_option(4, NAME, 1390 + (((global_var(GVAR_NEW_RENO_FLAG_2) bwand 32768) != 0) == 0), Node997, NEUTRAL_REACTION);
+    giq_option(4, NAME, 472, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node082
 begin
-    gsay_reply(451, 1405);
-    giq_option(4, 451, 1406 + (((global_var(345) bwand 32768) != 0) == 0), Node997, 50);
-    giq_option(4, 451, 1408, Node999, 50);
+    Reply(1405);
+    giq_option(4, NAME, 1406 + (((global_var(GVAR_NEW_RENO_FLAG_2) bwand 32768) != 0) == 0), Node997, NEUTRAL_REACTION);
+    giq_option(4, NAME, 1408, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node083
 begin
-    set_global_var(345, global_var(345) bwor 65536);
-    gsay_reply(451, 1415);
-    giq_option(4, 451, 1416, Node084, 50);
-    giq_option(4, 451, 1417, Node019, 50);
-    giq_option(4, 451, 1275, Node999, 50);
-    giq_option(4, 451, 1408, Node999, 50);
+    set_global_var(GVAR_NEW_RENO_FLAG_2, global_var(GVAR_NEW_RENO_FLAG_2) bwor 65536);
+    Reply(1415);
+    giq_option(4, NAME, 1416, Node084, NEUTRAL_REACTION);
+    giq_option(4, NAME, 1417, Node019, NEUTRAL_REACTION);
+    giq_option(4, NAME, 1275, Node999, NEUTRAL_REACTION);
+    giq_option(4, NAME, 1408, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node084
 begin
-    gsay_reply(451, 1425);
-    giq_option(4, 451, 1426, Node083, 50);
-    giq_option(4, 451, 1427, Node019, 50);
-    giq_option(4, 451, 1275, Node999, 50);
-    giq_option(4, 451, 1408, Node999, 50);
+    Reply(1425);
+    giq_option(4, NAME, 1426, Node083, NEUTRAL_REACTION);
+    giq_option(4, NAME, 1427, Node019, NEUTRAL_REACTION);
+    giq_option(4, NAME, 1275, Node999, NEUTRAL_REACTION);
+    giq_option(4, NAME, 1408, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node085
 begin
-    gsay_reply(451, 1435);
-    giq_option(4, 451, 1436, Node073, 50);
+    Reply(1435);
+    giq_option(4, NAME, 1436, Node073, NEUTRAL_REACTION);
     if (obj_is_carrying_obj_pid(dude_obj, 416)) then begin
-        giq_option(4, 451, 1437, Node086, 50);
+        giq_option(4, NAME, 1437, Node086, NEUTRAL_REACTION);
     end
-    giq_option(4, 451, 1427, Node019, 50);
-    giq_option(4, 451, 1439, Node999, 50);
+    giq_option(4, NAME, 1427, Node019, NEUTRAL_REACTION);
+    giq_option(4, NAME, 1439, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node086
 begin
-    gsay_reply(451, 1450);
-    giq_option(4, 451, 1451, Node087, 50);
-    giq_option(4, 451, 1452, Node088, 50);
-    giq_option(4, 451, 1453, Node019, 50);
-    giq_option(4, 451, 1439, Node999, 50);
+    Reply(1450);
+    giq_option(4, NAME, 1451, Node087, NEUTRAL_REACTION);
+    giq_option(4, NAME, 1452, Node088, NEUTRAL_REACTION);
+    giq_option(4, NAME, 1453, Node019, NEUTRAL_REACTION);
+    giq_option(4, NAME, 1439, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node087
 begin
-    set_global_var(345, global_var(345) bwor 65536);
-    gsay_reply(451, 1470);
-    giq_option(4, 451, 1471, Node088, 50);
-    giq_option(4, 451, 1417, Node019, 50);
-    giq_option(4, 451, 1473, Node999, 50);
+    set_global_var(GVAR_NEW_RENO_FLAG_2, global_var(GVAR_NEW_RENO_FLAG_2) bwor 65536);
+    Reply(1470);
+    giq_option(4, NAME, 1471, Node088, NEUTRAL_REACTION);
+    giq_option(4, NAME, 1417, Node019, NEUTRAL_REACTION);
+    giq_option(4, NAME, 1473, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node088
 begin
-    gsay_reply(451, 1485);
-    giq_option(4, 451, 1486, Node087, 50);
-    giq_option(4, 451, 1487, Node019, 50);
-    giq_option(4, 451, 1473, Node999, 50);
+    Reply(1485);
+    giq_option(4, NAME, 1486, Node087, NEUTRAL_REACTION);
+    giq_option(4, NAME, 1487, Node019, NEUTRAL_REACTION);
+    giq_option(4, NAME, 1473, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node089
 begin
-    gsay_reply(451, 1500);
-    giq_option(4, 451, 1501, Node019, 50);
-    giq_option(4, 451, 1502, Node999, 50);
+    Reply(1500);
+    giq_option(4, NAME, 1501, Node019, NEUTRAL_REACTION);
+    giq_option(4, NAME, 1502, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node997
@@ -1685,13 +1685,13 @@ end
 procedure Node989
 begin
     debug_msg("barter to " + node_string);
-    gdialog_mod_barter(0);
+    gdialog_barter;
     call node_string();
 end
 
 procedure Node988
 begin
-    if (get_critter_stat(dude_obj, 34) == 0) then begin
+    if (dude_gender == 0) then begin
         call Node061();
     end
     else begin
@@ -1701,7 +1701,7 @@ end
 
 procedure Node987
 begin
-    if (has_trait(2, dude_obj, 1) or (get_critter_stat(dude_obj, 0) > 8) or ((obj_pid(critter_inven_obj(dude_obj, 0)) == 3) or (obj_pid(critter_inven_obj(dude_obj, 0)) == 348) or (obj_pid(critter_inven_obj(dude_obj, 0)) == 349) or (obj_pid(critter_inven_obj(dude_obj, 0)) == 232)) or has_trait(2, dude_obj, 5) or (((global_var(4) + global_var(5)) >= 25) and ((global_var(5) > (2 * global_var(4))) or (global_var(3) == 1)))) then begin
+    if (has_trait(TRAIT_TRAIT, dude_obj, TRAIT_bruiser) or (dude_strength > 8) or ((obj_pid(critter_inven_obj(dude_obj, INVEN_TYPE_WORN)) == 3) or (obj_pid(critter_inven_obj(dude_obj, INVEN_TYPE_WORN)) == 348) or (obj_pid(critter_inven_obj(dude_obj, INVEN_TYPE_WORN)) == 349) or (obj_pid(critter_inven_obj(dude_obj, INVEN_TYPE_WORN)) == 232)) or has_trait(TRAIT_TRAIT, dude_obj, TRAIT_kamikaze) or (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1)))) then begin
         call Node078();
     end
     else begin
@@ -1711,51 +1711,51 @@ end
 
 procedure Node986
 begin
-    if (((global_var(495) bwand 16384) != 0) == 0) then begin
-        set_global_var(0, global_var(0) + 5);
+    if (((global_var(GVAR_NEW_RENO_FLAG_3) bwand 16384) != 0) == 0) then begin
+        set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + 5);
         debug_msg("Player gains " + 5 + " Karma Points.");
-        set_global_var(37, 0);
-        set_global_var(38, 0);
-        set_global_var(39, 0);
-        set_global_var(40, 0);
-        set_global_var(41, 0);
-        set_global_var(42, 0);
-        set_global_var(43, 0);
-        set_global_var(44, 0);
-        set_global_var(45, 0);
-        if (global_var(0) >= 1000) then begin
-            set_global_var(37, 1);
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+        set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+        set_global_var(GVAR_KARMA_DEFENDER, 0);
+        set_global_var(GVAR_KARMA_WANDERER, 0);
+        set_global_var(GVAR_KARMA_BETRAYER, 0);
+        set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+        set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+            set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
         end
         else begin
-            if (global_var(0) >= 750) then begin
-                set_global_var(38, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
             end
             else begin
-                if (global_var(0) >= 500) then begin
-                    set_global_var(39, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                 end
                 else begin
-                    if (global_var(0) >= 250) then begin
-                        set_global_var(40, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                        set_global_var(GVAR_KARMA_DEFENDER, 1);
                     end
                     else begin
-                        if (global_var(0) > -250) then begin
-                            set_global_var(41, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                            set_global_var(GVAR_KARMA_WANDERER, 1);
                         end
                         else begin
-                            if (global_var(0) > -500) then begin
-                                set_global_var(42, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                set_global_var(GVAR_KARMA_BETRAYER, 1);
                             end
                             else begin
-                                if (global_var(0) > -750) then begin
-                                    set_global_var(43, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -1000) then begin
-                                        set_global_var(44, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                     end
                                     else begin
-                                        set_global_var(45, 1);
+                                        set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                     end
                                 end
                             end
@@ -1764,9 +1764,9 @@ begin
                 end
             end
         end
-        set_global_var(55, global_var(55) + 1);
+        set_global_var(GVAR_TOWN_REP_NEW_RENO, global_var(GVAR_TOWN_REP_NEW_RENO) + 1);
     end
-    set_global_var(495, global_var(495) bwor 16384);
+    set_global_var(GVAR_NEW_RENO_FLAG_3, global_var(GVAR_NEW_RENO_FLAG_3) bwor 16384);
     load_map(58, 999);
 end
 
@@ -1774,4 +1774,3 @@ procedure Node985
 begin
     load_map(58, 888);
 end
-
