diff --git a/Games/Fallout2.Steam/Original/NCR1.ssl b/Games/Fallout2.Steam/ReDefine/NCR1.ssl
index e1ce79c..2b46bad 100644
--- a/Games/Fallout2.Steam/Original/NCR1.ssl
+++ b/Games/Fallout2.Steam/ReDefine/NCR1.ssl
@@ -172,11 +172,11 @@ end
 procedure map_enter_p_proc
 begin
     variable LVar0 := 0;
-    if (metarule(14, 0)) then begin
+    if (map_first_run) then begin
     end
-    if (global_var(27) == 5) then begin
-        gfade_in(1);
-        set_global_var(global_var(27), 0);
+    if (global_var(GVAR_LOAD_MAP_INDEX) == 5) then begin
+        fade_in;
+        set_global_var(global_var(GVAR_LOAD_MAP_INDEX), 0);
         override_map_start(12282 % 200, 12282 / 200, 0, 2);
     end
     if ((get_month >= 3) and (get_month < 5)) then begin
@@ -255,12 +255,12 @@ begin
             end
         end
     end
-    if ((global_var(516) == 0) and (global_var(382) == 1) and (map_var(3) == 0)) then begin
+    if ((global_var(GVAR_STANWELL_PAYOUT) == 0) and (global_var(GVAR_STANWELL_STATUS) == 1) and (map_var(3) == 0)) then begin
         set_map_var(3, map_var(3) + 1);
-        LVar0 := create_object_sid(16777260, 0, 0, 811);
+        LVar0 := create_object_sid(16777260, 0, 0, SCRIPT_RCNORD);
         critter_attempt_placement(LVar0, 10913, 0);
     end
-    set_global_var(27, 0);
+    set_global_var(GVAR_LOAD_MAP_INDEX, 0);
 end
 
 procedure map_update_p_proc
@@ -346,4 +346,3 @@ end
 procedure Initial_Inven
 begin
 end
-
