diff --git a/Games/Fallout2.Steam/Original/NCRNGGRL.ssl b/Games/Fallout2.Steam/ReDefine/NCRNGGRL.ssl
index 2c31d40..2f140d8 100644
--- a/Games/Fallout2.Steam/Original/NCRNGGRL.ssl
+++ b/Games/Fallout2.Steam/ReDefine/NCRNGGRL.ssl
@@ -197,53 +197,53 @@ end
 procedure ChooseFromList
 begin
     variable LVar0 := 0;
-    gsay_reply(1087, GVar22);
-    if (message_str(353, GVar24 - 1) != "") then begin
-        giq_option(1, 1087, message_str(353, 150), CLICK_PREV, 50);
+    Reply(GVar22);
+    if (message_str(SCRIPT_NEWR1, GVar24 - 1) != "") then begin
+        giq_option(1, NAME, message_str(SCRIPT_NEWR1, 150), CLICK_PREV, NEUTRAL_REACTION);
     end
     if (LVar0 == 0) then begin
-        if (message_str(353, GVar24 + 0) != "") then begin
-            giq_option(1, 1087, message_str(353, GVar24 + 0), CLICK_0, 50);
+        if (message_str(SCRIPT_NEWR1, GVar24 + 0) != "") then begin
+            giq_option(1, NAME, message_str(SCRIPT_NEWR1, GVar24 + 0), CLICK_0, NEUTRAL_REACTION);
         end
         else begin
             LVar0 := 1;
         end
     end
     if (LVar0 == 0) then begin
-        if (message_str(353, GVar24 + 1) != "") then begin
-            giq_option(1, 1087, message_str(353, GVar24 + 1), CLICK_1, 50);
+        if (message_str(SCRIPT_NEWR1, GVar24 + 1) != "") then begin
+            giq_option(1, NAME, message_str(SCRIPT_NEWR1, GVar24 + 1), CLICK_1, NEUTRAL_REACTION);
         end
         else begin
             LVar0 := 1;
         end
     end
     if (LVar0 == 0) then begin
-        if (message_str(353, GVar24 + 2) != "") then begin
-            giq_option(1, 1087, message_str(353, GVar24 + 2), CLICK_2, 50);
+        if (message_str(SCRIPT_NEWR1, GVar24 + 2) != "") then begin
+            giq_option(1, NAME, message_str(SCRIPT_NEWR1, GVar24 + 2), CLICK_2, NEUTRAL_REACTION);
         end
         else begin
             LVar0 := 1;
         end
     end
     if (LVar0 == 0) then begin
-        if (message_str(353, GVar24 + 3) != "") then begin
-            giq_option(1, 1087, message_str(353, GVar24 + 3), CLICK_3, 50);
+        if (message_str(SCRIPT_NEWR1, GVar24 + 3) != "") then begin
+            giq_option(1, NAME, message_str(SCRIPT_NEWR1, GVar24 + 3), CLICK_3, NEUTRAL_REACTION);
         end
         else begin
             LVar0 := 1;
         end
     end
     if (LVar0 == 0) then begin
-        if (message_str(353, GVar24 + 4) != "") then begin
-            giq_option(1, 1087, message_str(353, GVar24 + 4), CLICK_4, 50);
+        if (message_str(SCRIPT_NEWR1, GVar24 + 4) != "") then begin
+            giq_option(1, NAME, message_str(SCRIPT_NEWR1, GVar24 + 4), CLICK_4, NEUTRAL_REACTION);
         end
         else begin
             LVar0 := 1;
         end
     end
     if (LVar0 == 0) then begin
-        if (message_str(353, GVar24 + 5) != "") then begin
-            giq_option(1, 1087, message_str(353, 151), CLICK_MORE, 50);
+        if (message_str(SCRIPT_NEWR1, GVar24 + 5) != "") then begin
+            giq_option(1, NAME, message_str(SCRIPT_NEWR1, 151), CLICK_MORE, NEUTRAL_REACTION);
         end
     end
 end
@@ -302,8 +302,8 @@ begin
             animate_move_obj_to_tile(self_obj, local_var(1), 0);
         end
         else begin
-            if (has_trait(1, self_obj, 666)) then begin
-                float_msg(self_obj, message_str(1087, 100) + global_var(506) + message_str(1087, 101), 8);
+            if (obj_is_visible_flag(self_obj)) then begin
+                float_msg(self_obj, mstr(100) + global_var(GVAR_NEW_RENO_ROUND_NUMBER) + mstr(101), FLOAT_MSG_YELLOW);
                 if (tile_num(self_obj) == 24672) then begin
                     set_local_var(1, 24678);
                 end
@@ -333,13 +333,13 @@ end
 procedure map_enter_p_proc
 begin
     ring_girl_obj := self_obj;
-    if (metarule(22, 0) == 0) then begin
-        if (metarule(14, 0)) then begin
+    if (is_loading_game == 0) then begin
+        if (map_first_run) then begin
             set_local_var(1, tile_num(self_obj));
-            critter_add_trait(self_obj, 1, 6, 54);
-            critter_add_trait(self_obj, 1, 5, 49);
+            critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 54);
+            critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_LONER_CITIZEN);
         end
-        if (has_trait(1, self_obj, 666) == 1) then begin
+        if (obj_is_visible_flag(self_obj) == 1) then begin
             set_obj_visibility(self_obj, 1);
         end
     end
@@ -353,4 +353,3 @@ procedure map_update_p_proc
 begin
     ring_girl_obj := self_obj;
 end
-
