diff --git a/Games/Fallout2.Steam/Original/NCSTULIT.ssl b/Games/Fallout2.Steam/ReDefine/NCSTULIT.ssl
index 1fd69f1..33c52a9 100644
--- a/Games/Fallout2.Steam/Original/NCSTULIT.ssl
+++ b/Games/Fallout2.Steam/ReDefine/NCSTULIT.ssl
@@ -239,53 +239,53 @@ end
 procedure ChooseFromList
 begin
     variable LVar0 := 0;
-    gsay_reply(444, stored_msg);
-    if (message_str(353, top_msg - 1) != "") then begin
-        giq_option(1, 444, message_str(353, 150), CLICK_PREV, 50);
+    Reply(stored_msg);
+    if (message_str(SCRIPT_NEWR1, top_msg - 1) != "") then begin
+        giq_option(1, NAME, message_str(SCRIPT_NEWR1, 150), CLICK_PREV, NEUTRAL_REACTION);
     end
     if (LVar0 == 0) then begin
-        if (message_str(353, top_msg + 0) != "") then begin
-            giq_option(1, 444, message_str(353, top_msg + 0), CLICK_0, 50);
+        if (message_str(SCRIPT_NEWR1, top_msg + 0) != "") then begin
+            giq_option(1, NAME, message_str(SCRIPT_NEWR1, top_msg + 0), CLICK_0, NEUTRAL_REACTION);
         end
         else begin
             LVar0 := 1;
         end
     end
     if (LVar0 == 0) then begin
-        if (message_str(353, top_msg + 1) != "") then begin
-            giq_option(1, 444, message_str(353, top_msg + 1), CLICK_1, 50);
+        if (message_str(SCRIPT_NEWR1, top_msg + 1) != "") then begin
+            giq_option(1, NAME, message_str(SCRIPT_NEWR1, top_msg + 1), CLICK_1, NEUTRAL_REACTION);
         end
         else begin
             LVar0 := 1;
         end
     end
     if (LVar0 == 0) then begin
-        if (message_str(353, top_msg + 2) != "") then begin
-            giq_option(1, 444, message_str(353, top_msg + 2), CLICK_2, 50);
+        if (message_str(SCRIPT_NEWR1, top_msg + 2) != "") then begin
+            giq_option(1, NAME, message_str(SCRIPT_NEWR1, top_msg + 2), CLICK_2, NEUTRAL_REACTION);
         end
         else begin
             LVar0 := 1;
         end
     end
     if (LVar0 == 0) then begin
-        if (message_str(353, top_msg + 3) != "") then begin
-            giq_option(1, 444, message_str(353, top_msg + 3), CLICK_3, 50);
+        if (message_str(SCRIPT_NEWR1, top_msg + 3) != "") then begin
+            giq_option(1, NAME, message_str(SCRIPT_NEWR1, top_msg + 3), CLICK_3, NEUTRAL_REACTION);
         end
         else begin
             LVar0 := 1;
         end
     end
     if (LVar0 == 0) then begin
-        if (message_str(353, top_msg + 4) != "") then begin
-            giq_option(1, 444, message_str(353, top_msg + 4), CLICK_4, 50);
+        if (message_str(SCRIPT_NEWR1, top_msg + 4) != "") then begin
+            giq_option(1, NAME, message_str(SCRIPT_NEWR1, top_msg + 4), CLICK_4, NEUTRAL_REACTION);
         end
         else begin
             LVar0 := 1;
         end
     end
     if (LVar0 == 0) then begin
-        if (message_str(353, top_msg + 5) != "") then begin
-            giq_option(1, 444, message_str(353, 151), CLICK_MORE, 50);
+        if (message_str(SCRIPT_NEWR1, top_msg + 5) != "") then begin
+            giq_option(1, NAME, message_str(SCRIPT_NEWR1, 151), CLICK_MORE, NEUTRAL_REACTION);
         end
     end
 end
@@ -340,15 +340,15 @@ end
 procedure critter_p_proc
 begin
     if (obj_can_see_obj(self_obj, dude_obj)) then begin
-        if (((local_var(5) bwand 2) != 0) or ((global_var(495) bwand 8) != 0)) then begin
+        if (((local_var(5) bwand 2) != 0) or ((global_var(GVAR_NEW_RENO_FLAG_3) bwand 8) != 0)) then begin
             if (critter_is_fleeing(self_obj)) then begin
                 if (anim_busy(self_obj) == 0) then begin
                     if ((tile_distance_objs(self_obj, dude_obj) < 8) or obj_can_see_obj(self_obj, dude_obj)) then begin
                         if (anim_busy(self_obj) == 0) then begin
-                            global_temp := rotation_to_tile(tile_num(dude_obj), tile_num(self_obj));
+                            global_temp := rotation_to_tile(dude_tile, tile_num(self_obj));
                             animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(self_obj), global_temp, random(3, 10)), 1);
                             global_temp := (global_temp + 1) % 6;
-                            while ((anim_busy(self_obj) == 0) and (global_temp != rotation_to_tile(tile_num(dude_obj), tile_num(self_obj)))) do begin
+                            while ((anim_busy(self_obj) == 0) and (global_temp != rotation_to_tile(dude_tile, tile_num(self_obj)))) do begin
                                 debug_msg("flee loop: rot == " + global_temp);
                                 animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(self_obj), global_temp, random(3, 10)), 1);
                                 global_temp := (global_temp + 1) % 6;
@@ -358,7 +358,7 @@ begin
                 end
             end
             else begin
-                attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+                attack(dude_obj);
             end
         end
     end
@@ -390,8 +390,8 @@ begin
         forced_node := 0;
     end
     else begin
-        if (global_var(494) == 8) then begin
-            if (global_var(259) > 0) then begin
+        if (global_var(GVAR_FALLOUT_2) == 8) then begin
+            if (global_var(GVAR_NEW_RENO_HAS_REP_PRIZEFIGHTER) > 0) then begin
                 call Node001();
             end
             else begin
@@ -410,7 +410,7 @@ begin
             end
         end
         else begin
-            if (global_var(259) > 0) then begin
+            if (global_var(GVAR_NEW_RENO_HAS_REP_PRIZEFIGHTER) > 0) then begin
                 call Node003();
             end
             else begin
@@ -447,66 +447,66 @@ begin
     end
     last_timed_obj := self_obj;
     last_timed_param := last_timed_param;
-    metarule3(100, self_obj, 1, 0);
+    rm_fixed_timer_event(self_obj, 1);
     add_timer_event(self_obj, game_ticks(25) + random(0, game_ticks(35 - 25)), 1);
 end
 
 procedure destroy_p_proc
 begin
     if (source_obj == dude_obj) then begin
-        set_global_var(5, global_var(5) + 1);
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(4) > (3 * global_var(5))) or (global_var(2) == 1)) and (global_var(1) == 0)) then begin
-            set_global_var(2, 1);
-            set_global_var(3, 0);
+        set_global_var(GVAR_GOOD_MONSTER, global_var(GVAR_GOOD_MONSTER) + 1);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(4) > (3 * global_var(GVAR_GOOD_MONSTER))) or (global_var(GVAR_CHAMPION_REPUTATION) == 1)) and (global_var(GVAR_CHILDKILLER_REPUTATION) == 0)) then begin
+            set_global_var(GVAR_CHAMPION_REPUTATION, 1);
+            set_global_var(GVAR_BERSERKER_REPUTATION, 0);
         end
-        if (metarule(51, self_obj) == 2) then begin
+        if (critter_kill_type(self_obj) == KILL_TYPE_children_kills) then begin
             if (source_obj == dude_obj) then begin
-                set_global_var(2, 0);
-                set_global_var(1, global_var(1) + 1);
-                set_global_var(0, global_var(0) + -15);
+                set_global_var(GVAR_CHAMPION_REPUTATION, 0);
+                set_global_var(GVAR_CHILDKILLER_REPUTATION, global_var(GVAR_CHILDKILLER_REPUTATION) + 1);
+                set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + -15);
                 debug_msg("Player gains " + -15 + " Karma Points.");
-                set_global_var(37, 0);
-                set_global_var(38, 0);
-                set_global_var(39, 0);
-                set_global_var(40, 0);
-                set_global_var(41, 0);
-                set_global_var(42, 0);
-                set_global_var(43, 0);
-                set_global_var(44, 0);
-                set_global_var(45, 0);
-                if (global_var(0) >= 1000) then begin
-                    set_global_var(37, 1);
+                set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+                set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+                set_global_var(GVAR_KARMA_DEFENDER, 0);
+                set_global_var(GVAR_KARMA_WANDERER, 0);
+                set_global_var(GVAR_KARMA_BETRAYER, 0);
+                set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+                set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+                set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+                    set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
                 end
                 else begin
-                    if (global_var(0) >= 750) then begin
-                        set_global_var(38, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
                     end
                     else begin
-                        if (global_var(0) >= 500) then begin
-                            set_global_var(39, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                            set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                         end
                         else begin
-                            if (global_var(0) >= 250) then begin
-                                set_global_var(40, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                                set_global_var(GVAR_KARMA_DEFENDER, 1);
                             end
                             else begin
-                                if (global_var(0) > -250) then begin
-                                    set_global_var(41, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                                    set_global_var(GVAR_KARMA_WANDERER, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -500) then begin
-                                        set_global_var(42, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                        set_global_var(GVAR_KARMA_BETRAYER, 1);
                                     end
                                     else begin
-                                        if (global_var(0) > -750) then begin
-                                            set_global_var(43, 1);
+                                        if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                            set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                         end
                                         else begin
-                                            if (global_var(0) > -1000) then begin
-                                                set_global_var(44, 1);
+                                            if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                                set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                             end
                                             else begin
-                                                set_global_var(45, 1);
+                                                set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                             end
                                         end
                                     end
@@ -515,58 +515,58 @@ begin
                         end
                     end
                 end
-                set_global_var(55, global_var(55) + -8);
+                set_global_var(GVAR_TOWN_REP_NEW_RENO, global_var(GVAR_TOWN_REP_NEW_RENO) + -8);
                 debug_msg("Added " + -8 + " to Town Rep");
             end
         end
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(5) > (2 * global_var(4))) or (global_var(3) == 1))) then begin
-            set_global_var(3, 1);
-            set_global_var(2, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
+            set_global_var(GVAR_BERSERKER_REPUTATION, 1);
+            set_global_var(GVAR_CHAMPION_REPUTATION, 0);
         end
-        set_global_var(0, global_var(0) + -10);
+        set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + -10);
         debug_msg("Player gains " + -10 + " Karma Points.");
-        set_global_var(37, 0);
-        set_global_var(38, 0);
-        set_global_var(39, 0);
-        set_global_var(40, 0);
-        set_global_var(41, 0);
-        set_global_var(42, 0);
-        set_global_var(43, 0);
-        set_global_var(44, 0);
-        set_global_var(45, 0);
-        if (global_var(0) >= 1000) then begin
-            set_global_var(37, 1);
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+        set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+        set_global_var(GVAR_KARMA_DEFENDER, 0);
+        set_global_var(GVAR_KARMA_WANDERER, 0);
+        set_global_var(GVAR_KARMA_BETRAYER, 0);
+        set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+        set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+            set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
         end
         else begin
-            if (global_var(0) >= 750) then begin
-                set_global_var(38, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
             end
             else begin
-                if (global_var(0) >= 500) then begin
-                    set_global_var(39, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                 end
                 else begin
-                    if (global_var(0) >= 250) then begin
-                        set_global_var(40, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                        set_global_var(GVAR_KARMA_DEFENDER, 1);
                     end
                     else begin
-                        if (global_var(0) > -250) then begin
-                            set_global_var(41, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                            set_global_var(GVAR_KARMA_WANDERER, 1);
                         end
                         else begin
-                            if (global_var(0) > -500) then begin
-                                set_global_var(42, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                set_global_var(GVAR_KARMA_BETRAYER, 1);
                             end
                             else begin
-                                if (global_var(0) > -750) then begin
-                                    set_global_var(43, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -1000) then begin
-                                        set_global_var(44, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                     end
                                     else begin
-                                        set_global_var(45, 1);
+                                        set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                     end
                                 end
                             end
@@ -575,7 +575,7 @@ begin
                 end
             end
         end
-        set_global_var(55, global_var(55) + 2);
+        set_global_var(GVAR_TOWN_REP_NEW_RENO, global_var(GVAR_TOWN_REP_NEW_RENO) + 2);
         debug_msg("Added " + -5 + " to Town Rep");
     end
 end
@@ -583,17 +583,17 @@ end
 procedure look_at_p_proc
 begin
     script_overrides;
-    display_msg(message_str(444, 100 + ((local_var(5) bwand 1) != 0)));
+    display_mstr(100 + ((local_var(5) bwand 1) != 0));
 end
 
 procedure description_p_proc
 begin
     script_overrides;
     if ((local_var(5) bwand 1) != 0) then begin
-        display_msg(message_str(444, random(151, 152)));
+        display_mstr(random(151, 152));
     end
     else begin
-        display_msg(message_str(444, 150));
+        display_mstr(150);
     end
     set_local_var(5, local_var(5) bwor 4);
 end
@@ -613,19 +613,19 @@ end
 procedure map_enter_p_proc
 begin
     stuart_little_obj := self_obj;
-    if (metarule(22, 0) == 0) then begin
-        if (((global_var(495) bwand 2048) != 0) and ((global_var(259) > 0) == 0)) then begin
+    if (is_loading_game == 0) then begin
+        if (((global_var(GVAR_NEW_RENO_FLAG_3) bwand 2048) != 0) and ((global_var(GVAR_NEW_RENO_HAS_REP_PRIZEFIGHTER) > 0) == 0)) then begin
             destroy_object(self_obj);
         end
         else begin
-            if (metarule(14, 0)) then begin
-                critter_add_trait(self_obj, 1, 6, 54);
-                critter_add_trait(self_obj, 1, 5, 24);
+            if (map_first_run) then begin
+                critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 54);
+                critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_TOUGH_CITIZEN);
             end
         end
         last_timed_obj := self_obj;
         last_timed_param := last_timed_param;
-        metarule3(100, self_obj, 1, 0);
+        rm_fixed_timer_event(self_obj, 1);
         add_timer_event(self_obj, game_ticks(25) + random(0, game_ticks(35 - 25)), 1);
     end
 end
@@ -633,86 +633,86 @@ end
 procedure timed_event_p_proc
 begin
     if (fixed_param == 34) then begin
-        debug_msg("stu: fight_status_raw == " + global_var(510));
-        if (cur_map_index == 55) then begin
+        debug_msg("stu: fight_status_raw == " + global_var(GVAR_NEW_RENO_FIGHT_STATUS));
+        if (CUR_MAP_NEWR2) then begin
             if (announcer_obj != 0) then begin
                 last_timed_obj := announcer_obj;
                 last_timed_param := last_timed_param;
-                metarule3(100, announcer_obj, 48, 0);
+                rm_fixed_timer_event(announcer_obj, 48);
                 add_timer_event(announcer_obj, game_ticks(0), 48);
             end
         end
-        set_global_var(694, global_var(694) bwand (-1 - 2));
-        if (has_trait(1, ring_girl_obj, 666) == 1) then begin
+        set_global_var(GVAR_NEW_RENO_FLAG_4, global_var(GVAR_NEW_RENO_FLAG_4) bwand (-1 - 2));
+        if (obj_is_visible_flag(ring_girl_obj) == 1) then begin
             set_obj_visibility(ring_girl_obj, 1);
         end
-        if (has_trait(1, announcer_obj, 666) == 1) then begin
+        if (obj_is_visible_flag(announcer_obj) == 1) then begin
             set_obj_visibility(announcer_obj, 1);
         end
         gfade_in(10);
-        if (global_var(510) > 0) then begin
-            if (global_var(503) != 0) then begin
-                if (global_var(504) == 2) then begin
-                    item_caps_adjust(dude_obj, (100 - global_var(503)) * 800 / 100);
-                    item_caps_adjust(self_obj, global_var(503) * 800 / 100);
+        if (global_var(GVAR_NEW_RENO_FIGHT_STATUS) > 0) then begin
+            if (global_var(GVAR_NEW_RENO_STUART_DEAL) != 0) then begin
+                if (global_var(GVAR_NEW_RENO_FIGHT_LEVEL) == 2) then begin
+                    item_caps_adjust(dude_obj, (100 - global_var(GVAR_NEW_RENO_STUART_DEAL)) * 800 / 100);
+                    item_caps_adjust(self_obj, global_var(GVAR_NEW_RENO_STUART_DEAL) * 800 / 100);
                 end
                 else begin
-                    if (global_var(504) == 3) then begin
-                        item_caps_adjust(dude_obj, (100 - global_var(503)) * 2000 / 100);
-                        item_caps_adjust(self_obj, global_var(503) * 2000 / 100);
+                    if (global_var(GVAR_NEW_RENO_FIGHT_LEVEL) == 3) then begin
+                        item_caps_adjust(dude_obj, (100 - global_var(GVAR_NEW_RENO_STUART_DEAL)) * 2000 / 100);
+                        item_caps_adjust(self_obj, global_var(GVAR_NEW_RENO_STUART_DEAL) * 2000 / 100);
                     end
                     else begin
-                        item_caps_adjust(dude_obj, (100 - global_var(503)) * 100 / 100);
-                        item_caps_adjust(self_obj, global_var(503) * 100 / 100);
+                        item_caps_adjust(dude_obj, (100 - global_var(GVAR_NEW_RENO_STUART_DEAL)) * 100 / 100);
+                        item_caps_adjust(self_obj, global_var(GVAR_NEW_RENO_STUART_DEAL) * 100 / 100);
                     end
                 end
             end
-            if (global_var(504) <= 3) then begin
-                display_msg(message_str(353, 1400 + global_var(504)));
-                set_global_var(504, global_var(504) + 1);
-                if (global_var(504) == 1) then begin
+            if (global_var(GVAR_NEW_RENO_FIGHT_LEVEL) <= 3) then begin
+                display_msg(message_str(SCRIPT_NEWR1, 1400 + global_var(GVAR_NEW_RENO_FIGHT_LEVEL)));
+                set_global_var(GVAR_NEW_RENO_FIGHT_LEVEL, global_var(GVAR_NEW_RENO_FIGHT_LEVEL) + 1);
+                if (global_var(GVAR_NEW_RENO_FIGHT_LEVEL) == 1) then begin
                     give_exp_points(500);
-                    display_msg(message_str(14, 100) + 500 + message_str(14, 101));
-                    set_global_var(55, global_var(55) + 1);
+                    display_msg(g_mstr(100) + 500 + g_mstr(101));
+                    set_global_var(GVAR_TOWN_REP_NEW_RENO, global_var(GVAR_TOWN_REP_NEW_RENO) + 1);
                 end
                 else begin
-                    if (global_var(504) == 2) then begin
+                    if (global_var(GVAR_NEW_RENO_FIGHT_LEVEL) == 2) then begin
                         give_exp_points(750);
-                        display_msg(message_str(14, 100) + 750 + message_str(14, 101));
-                        set_global_var(55, global_var(55) + 1);
+                        display_msg(g_mstr(100) + 750 + g_mstr(101));
+                        set_global_var(GVAR_TOWN_REP_NEW_RENO, global_var(GVAR_TOWN_REP_NEW_RENO) + 1);
                     end
                     else begin
-                        if (global_var(504) == 3) then begin
+                        if (global_var(GVAR_NEW_RENO_FIGHT_LEVEL) == 3) then begin
                             give_exp_points(1000);
-                            display_msg(message_str(14, 100) + 1000 + message_str(14, 101));
-                            set_global_var(55, global_var(55) + 1);
+                            display_msg(g_mstr(100) + 1000 + g_mstr(101));
+                            set_global_var(GVAR_TOWN_REP_NEW_RENO, global_var(GVAR_TOWN_REP_NEW_RENO) + 1);
                         end
                         else begin
-                            if (global_var(504) == 4) then begin
+                            if (global_var(GVAR_NEW_RENO_FIGHT_LEVEL) == 4) then begin
                                 give_exp_points(2500);
-                                display_msg(message_str(14, 100) + 2500 + message_str(14, 101));
-                                set_global_var(55, global_var(55) + 1);
-                                critter_mod_skill(dude_obj, 3, 5);
-                                global_temp := get_critter_stat(dude_obj, 24) + 5;
-                                set_critter_stat(dude_obj, 24, global_temp);
-                                set_global_var(259, 1);
+                                display_msg(g_mstr(100) + 2500 + g_mstr(101));
+                                set_global_var(GVAR_TOWN_REP_NEW_RENO, global_var(GVAR_TOWN_REP_NEW_RENO) + 1);
+                                critter_mod_skill(dude_obj, SKILL_UNARMED_COMBAT, 5);
+                                global_temp := get_critter_stat(dude_obj, STAT_dmg_resist) + 5;
+                                set_critter_stat(dude_obj, STAT_dmg_resist, global_temp);
+                                set_global_var(GVAR_NEW_RENO_HAS_REP_PRIZEFIGHTER, 1);
                             end
                         end
                     end
                 end
             end
-            debug_msg("inc_fight_level == " + global_var(504));
-            if (global_var(259) > 0) then begin
+            debug_msg("inc_fight_level == " + global_var(GVAR_NEW_RENO_FIGHT_LEVEL));
+            if (global_var(GVAR_NEW_RENO_HAS_REP_PRIZEFIGHTER) > 0) then begin
                 dialogue_system_enter;
             end
             else begin
-                debug_msg("force_dialog_start(" + forced_node + "): " + obj_name(self_obj));
+                debug_msg("force_dialog_start(" + forced_node + "): " + self_name);
                 forced_node := "Node017";
                 dialogue_system_enter;
             end
         end
         else begin
-            debug_msg("force_dialog_start(" + forced_node + "): " + obj_name(self_obj));
+            debug_msg("force_dialog_start(" + forced_node + "): " + self_name);
             forced_node := "Node014";
             dialogue_system_enter;
         end
@@ -721,100 +721,100 @@ begin
         if (fixed_param == 46) then begin
             debug_msg("start boxing match: start");
             gfade_out(10);
-            set_global_var(495, global_var(495) bwor 536870912);
-            if (cur_map_index == 55) then begin
+            set_global_var(GVAR_NEW_RENO_FLAG_3, global_var(GVAR_NEW_RENO_FLAG_3) bwor 536870912);
+            if (CUR_MAP_NEWR2) then begin
                 if (announcer_obj != 0) then begin
                     last_timed_obj := announcer_obj;
                     last_timed_param := last_timed_param;
-                    metarule3(100, announcer_obj, 48, 0);
+                    rm_fixed_timer_event(announcer_obj, 48);
                     add_timer_event(announcer_obj, game_ticks(0), 48);
                 end
             end
-            set_global_var(694, global_var(694) bwor 2);
+            set_global_var(GVAR_NEW_RENO_FLAG_4, global_var(GVAR_NEW_RENO_FLAG_4) bwor 2);
             reg_anim_func(2, dude_obj);
             move_obj_inven_to_obj(dude_obj, new_reno_temp_box);
-            set_global_var(508, 0);
-            set_global_var(509, 0);
-            set_global_var(495, global_var(495) bwand (-1 - 8192));
-            set_global_var(507, 0);
+            set_global_var(GVAR_NEW_RENO_DUDE_SCORE, 0);
+            set_global_var(GVAR_NEW_RENO_BOXER_SCORE, 0);
+            set_global_var(GVAR_NEW_RENO_FLAG_3, global_var(GVAR_NEW_RENO_FLAG_3) bwand (-1 - 8192));
+            set_global_var(GVAR_NEW_RENO_ROUND_TIME, 0);
             if (0 != 0) then begin
-                if (global_var(510) == 0) then begin
-                    set_global_var(510, 0);
+                if (global_var(GVAR_NEW_RENO_FIGHT_STATUS) == 0) then begin
+                    set_global_var(GVAR_NEW_RENO_FIGHT_STATUS, 0);
                     last_timed_obj := announcer_obj;
                     last_timed_param := last_timed_param;
-                    metarule3(100, announcer_obj, 200, 0);
+                    rm_fixed_timer_event(announcer_obj, 200);
                     add_timer_event(announcer_obj, game_ticks(5), 200);
                 end
             end
             else begin
-                set_global_var(510, 0);
+                set_global_var(GVAR_NEW_RENO_FIGHT_STATUS, 0);
             end
-            debug_msg("set_fight_status(" + 0 + ") == " + global_var(510));
+            debug_msg("set_fight_status(" + 0 + ") == " + global_var(GVAR_NEW_RENO_FIGHT_STATUS));
             if (game_time_hour < 2100) then begin
                 game_time_advance((24 * (60 * (60 * 10)) * (game_time_hour > 2100)) + (((2100 / 100) - (game_time_hour / 100)) * (60 * (60 * 10))) + (((2100 % 100) - (game_time_hour % 100)) * (60 * 10)));
             end
-            set_global_var(506, 1);
+            set_global_var(GVAR_NEW_RENO_ROUND_NUMBER, 1);
             if (fighting_boxer_1_obj != 0) then begin
-                if (has_trait(1, fighting_boxer_1_obj, 666) == 1) then begin
+                if (obj_is_visible_flag(fighting_boxer_1_obj) == 1) then begin
                     set_obj_visibility(fighting_boxer_1_obj, 1);
                 end
             end
             if (fighting_boxer_2_obj != 0) then begin
-                if (has_trait(1, fighting_boxer_2_obj, 666) == 1) then begin
+                if (obj_is_visible_flag(fighting_boxer_2_obj) == 1) then begin
                     set_obj_visibility(fighting_boxer_2_obj, 1);
                 end
             end
             if (obj_carrying_pid_obj(new_reno_temp_box, 293)) then begin
-                restock_obj := create_object_sid(497, 0, 0, -1);
+                restock_obj := create_object(497, 0, 0);
                 add_mult_objs_to_inven(dude_obj, restock_obj, 1);
                 wield_obj_critter(dude_obj, restock_obj);
             end
             else begin
                 if (obj_carrying_pid_obj(new_reno_temp_box, 292)) then begin
-                    restock_obj := create_object_sid(496, 0, 0, -1);
+                    restock_obj := create_object(496, 0, 0);
                     add_mult_objs_to_inven(dude_obj, restock_obj, 1);
                     wield_obj_critter(dude_obj, restock_obj);
                 end
             end
             move_to(dude_obj, tile_num(announcer_obj), elevation(self_obj));
-            set_global_var(495, global_var(495) bwor 4096);
-            if (global_var(504) == 1) then begin
-                if (((global_var(495) bwand 512) != 0) == 0) then begin
+            set_global_var(GVAR_NEW_RENO_FLAG_3, global_var(GVAR_NEW_RENO_FLAG_3) bwor 4096);
+            if (global_var(GVAR_NEW_RENO_FIGHT_LEVEL) == 1) then begin
+                if (((global_var(GVAR_NEW_RENO_FLAG_3) bwand 512) != 0) == 0) then begin
                     prize_fighter_obj := 16777649;
                     if (mckneely_obj != 0) then begin
-                        if (has_trait(1, mckneely_obj, 666) == 1) then begin
+                        if (obj_is_visible_flag(mckneely_obj) == 1) then begin
                             set_obj_visibility(mckneely_obj, 1);
                         end
                     end
                 end
                 else begin
                     prize_fighter_obj := 16777645;
-                    set_global_var(495, global_var(495) bwand (-1 - 4096));
+                    set_global_var(GVAR_NEW_RENO_FLAG_3, global_var(GVAR_NEW_RENO_FLAG_3) bwand (-1 - 4096));
                 end
             end
             else begin
-                if (global_var(504) == 2) then begin
-                    if (((global_var(495) bwand 1024) != 0) == 0) then begin
+                if (global_var(GVAR_NEW_RENO_FIGHT_LEVEL) == 2) then begin
+                    if (((global_var(GVAR_NEW_RENO_FLAG_3) bwand 1024) != 0) == 0) then begin
                         prize_fighter_obj := 16777648;
                         if (hollyfeld_obj != 0) then begin
-                            if (has_trait(1, hollyfeld_obj, 666) == 1) then begin
+                            if (obj_is_visible_flag(hollyfeld_obj) == 1) then begin
                                 set_obj_visibility(hollyfeld_obj, 1);
                             end
                         end
                     end
                     else begin
                         prize_fighter_obj := 16777643;
-                        set_global_var(495, global_var(495) bwand (-1 - 4096));
+                        set_global_var(GVAR_NEW_RENO_FLAG_3, global_var(GVAR_NEW_RENO_FLAG_3) bwand (-1 - 4096));
                     end
                 end
                 else begin
-                    if (global_var(504) == 3) then begin
-                        if (((global_var(495) bwand 2048) != 0) == 0) then begin
+                    if (global_var(GVAR_NEW_RENO_FIGHT_LEVEL) == 3) then begin
+                        if (((global_var(GVAR_NEW_RENO_FLAG_3) bwand 2048) != 0) == 0) then begin
                             prize_fighter_obj := 16777647;
                         end
                         else begin
                             prize_fighter_obj := 16777643;
-                            set_global_var(495, global_var(495) bwand (-1 - 4096));
+                            set_global_var(GVAR_NEW_RENO_FLAG_3, global_var(GVAR_NEW_RENO_FLAG_3) bwand (-1 - 4096));
                         end
                     end
                     else begin
@@ -822,33 +822,33 @@ begin
                     end
                 end
             end
-            prize_fighter_obj := create_object_sid(prize_fighter_obj, 0, 0, 1088);
-            restock_obj := create_object_sid(489, 0, 0, -1);
+            prize_fighter_obj := create_object_sid(prize_fighter_obj, 0, 0, SCRIPT_NCPRZFTR);
+            restock_obj := create_object(489, 0, 0);
             add_mult_objs_to_inven(prize_fighter_obj, restock_obj, 1);
-            critter_add_trait(prize_fighter_obj, 1, 6, 202);
-            critter_add_trait(prize_fighter_obj, 1, 5, 40);
+            critter_add_trait(prize_fighter_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 202);
+            critter_add_trait(prize_fighter_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_BOXER);
             debug_msg("creating boxer");
             debug_msg("move_boxers_to_corners: start");
             move_to(dude_obj, 24479, elevation(self_obj));
             critter_injure(dude_obj, 2 bwor 8388608);
-            if ((prize_fighter_obj != 0) and (((global_var(495) bwand 8192) != 0) == 0)) then begin
+            if ((prize_fighter_obj != 0) and (((global_var(GVAR_NEW_RENO_FLAG_3) bwand 8192) != 0) == 0)) then begin
                 move_to(prize_fighter_obj, 26471, elevation(self_obj));
                 animate_stand_obj(prize_fighter_obj);
                 critter_injure(prize_fighter_obj, 2 bwor 8388608);
-                anim(dude_obj, 1000, rotation_to_tile(tile_num(dude_obj), tile_num(prize_fighter_obj)));
-                anim(prize_fighter_obj, 1000, rotation_to_tile(tile_num(prize_fighter_obj), tile_num(dude_obj)));
+                anim(dude_obj, 1000, rotation_to_tile(dude_tile, tile_num(prize_fighter_obj)));
+                anim(prize_fighter_obj, 1000, rotation_to_tile(tile_num(prize_fighter_obj), dude_tile));
             end
             animate_stand_obj(dude_obj);
             debug_msg("move_boxers_to_corners: end");
-            if (has_trait(1, announcer_obj, 666) == 0) then begin
+            if (obj_is_visible_flag(announcer_obj) == 0) then begin
                 set_obj_visibility(announcer_obj, 0);
             end
-            if (has_trait(1, ring_girl_obj, 666) == 0) then begin
+            if (obj_is_visible_flag(ring_girl_obj) == 0) then begin
                 set_obj_visibility(ring_girl_obj, 0);
             end
             last_timed_obj := announcer_obj;
             last_timed_param := last_timed_param;
-            metarule3(100, announcer_obj, 100, 0);
+            rm_fixed_timer_event(announcer_obj, 100);
             add_timer_event(announcer_obj, game_ticks(5), 100);
             gfade_in(10);
             debug_msg("start boxing match: end");
@@ -856,8 +856,8 @@ begin
         else begin
             if (fixed_param == 1) then begin
                 if (combat_is_initialized == 0) then begin
-                    if (global_var(494) == 8) then begin
-                        if (global_var(259) > 0) then begin
+                    if (global_var(GVAR_FALLOUT_2) == 8) then begin
+                        if (global_var(GVAR_NEW_RENO_HAS_REP_PRIZEFIGHTER) > 0) then begin
                             call Node001();
                         end
                         else begin
@@ -865,7 +865,7 @@ begin
                         end
                     end
                     else begin
-                        if (global_var(259) > 0) then begin
+                        if (global_var(GVAR_NEW_RENO_HAS_REP_PRIZEFIGHTER) > 0) then begin
                             call Node003();
                         end
                         else begin
@@ -875,7 +875,7 @@ begin
                 end
                 last_timed_obj := self_obj;
                 last_timed_param := last_timed_param;
-                metarule3(100, self_obj, 1, 0);
+                rm_fixed_timer_event(self_obj, 1);
                 add_timer_event(self_obj, game_ticks(25) + random(0, game_ticks(35 - 25)), 1);
             end
         end
@@ -895,321 +895,321 @@ end
 
 procedure Node001
 begin
-    float_msg(self_obj, message_str(444, random(200, 202)), 8);
+    float_msg(self_obj, mstr(random(200, 202)), FLOAT_MSG_YELLOW);
 end
 
 procedure Node002
 begin
-    float_msg(self_obj, message_str(444, random(210, 212)), 8);
+    float_msg(self_obj, mstr(random(210, 212)), FLOAT_MSG_YELLOW);
 end
 
 procedure Node003
 begin
-    float_msg(self_obj, message_str(444, random(222 - (2 * (get_critter_stat(dude_obj, 34) == 0)), 224 + (2 * (get_critter_stat(dude_obj, 34) == 1)))), 8);
+    float_msg(self_obj, mstr(random(222 - (2 * (dude_gender == 0)), 224 + (2 * (dude_gender == 1)))), FLOAT_MSG_YELLOW);
 end
 
 procedure Node005
 begin
-    float_msg(self_obj, message_str(444, random(230, 233)), 8);
+    float_msg(self_obj, mstr(random(230, 233)), FLOAT_MSG_YELLOW);
 end
 
 procedure Node006
 begin
-    if ((global_var(494) == 8) == 0) then begin
-        gsay_reply(444, message_str(444, 240) + message_str(444, 241 + (get_critter_stat(dude_obj, 34) == 1)));
+    if ((global_var(GVAR_FALLOUT_2) == 8) == 0) then begin
+        Reply(mstr(240) + mstr(241 + (dude_gender == 1)));
     end
     else begin
-        gsay_reply(444, message_str(444, 240) + message_str(444, 243));
+        Reply(mstr(240) + mstr(243));
     end
-    giq_option(-3, 444, 244, Node990, 50);
-    giq_option(-3, 444, 245, Node990, 51);
-    giq_option(-3, 444, 246, Node999, 50);
-    giq_option(4, 444, 247, Node022, 50);
-    giq_option(4, 444, 248, Node999, 50);
+    giq_option(-3, NAME, 244, Node990, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 245, Node990, BAD_REACTION);
+    giq_option(-3, NAME, 246, Node999, NEUTRAL_REACTION);
+    giq_option(4, NAME, 247, Node022, NEUTRAL_REACTION);
+    giq_option(4, NAME, 248, Node999, NEUTRAL_REACTION);
     prev_node := 6;
 end
 
 procedure Node007
 begin
-    gsay_reply(444, 250);
-    giq_option(-3, 444, 251, Node999, 50);
+    Reply(250);
+    giq_option(-3, NAME, 251, Node999, NEUTRAL_REACTION);
     prev_node := 7;
 end
 
 procedure Node008
 begin
-    gsay_reply(444, 260);
-    giq_option(-3, 444, 261, Node009, 51);
-    giq_option(-3, 444, 262, Node009, 51);
-    giq_option(-3, 444, 363, Node999, 50);
+    Reply(260);
+    giq_option(-3, NAME, 261, Node009, BAD_REACTION);
+    giq_option(-3, NAME, 262, Node009, BAD_REACTION);
+    giq_option(-3, NAME, 363, Node999, NEUTRAL_REACTION);
     prev_node := 8;
 end
 
 procedure Node009
 begin
-    gsay_reply(444, 270);
-    giq_option(-3, 444, 271, Node010, 51);
-    giq_option(-3, 444, 272, Node011, 49);
-    giq_option(-3, 444, 273, Node018, 50);
-    giq_option(-3, 444, 274, Node020, 51);
-    giq_option(-3, 444, 275, Node999, 50);
+    Reply(270);
+    giq_option(-3, NAME, 271, Node010, BAD_REACTION);
+    giq_option(-3, NAME, 272, Node011, GOOD_REACTION);
+    giq_option(-3, NAME, 273, Node018, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 274, Node020, BAD_REACTION);
+    giq_option(-3, NAME, 275, Node999, NEUTRAL_REACTION);
     prev_node := 9;
 end
 
 procedure Node010
 begin
-    gsay_reply(444, 280);
-    giq_option(-3, 444, 281, Node011, 50);
-    giq_option(-3, 444, 273, Node018, 50);
-    giq_option(-3, 444, 283, Node020, 50);
-    giq_option(-3, 444, 363, Node999, 50);
+    Reply(280);
+    giq_option(-3, NAME, 281, Node011, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 273, Node018, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 283, Node020, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 363, Node999, NEUTRAL_REACTION);
     prev_node := 10;
 end
 
 procedure Node011
 begin
     set_local_var(5, local_var(5) bwor 1);
-    gsay_reply(444, 290);
-    giq_option(-3, 444, message_str(444, 291) + obj_name(dude_obj) + message_str(444, 1322), Node998, 51);
-    giq_option(-3, 444, 292, Node989, 50);
-    giq_option(-3, 444, 293, Node984, 51);
-    giq_option(-3, 444, message_str(353, 498), Node988, 50);
-    giq_option(-3, 444, message_str(353, 499), Node987, 50);
-    giq_option(-3, 444, 296, Node999, 50);
+    Reply(290);
+    giq_option(-3, NAME, mstr(291) + dude_name + mstr(1322), Node998, BAD_REACTION);
+    giq_option(-3, NAME, 292, Node989, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 293, Node984, BAD_REACTION);
+    giq_option(-3, NAME, message_str(SCRIPT_NEWR1, 498), Node988, NEUTRAL_REACTION);
+    giq_option(-3, NAME, message_str(SCRIPT_NEWR1, 499), Node987, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 296, Node999, NEUTRAL_REACTION);
     prev_node := 11;
 end
 
 procedure Node012
 begin
-    debug_msg("GVAR_NEW_RENO_PRIZEFIGHTER == " + global_var(231) + " / dude_prizefighter_number == " + global_var(231) + " / prizefighter_name == " + message_str(353, 400 + global_var(231)));
-    if (global_var(231) == 497) then begin
-        gsay_reply(444, 300);
+    debug_msg("GVAR_NEW_RENO_PRIZEFIGHTER == " + global_var(GVAR_NEW_RENO_PRIZEFIGHTER) + " / dude_prizefighter_number == " + global_var(GVAR_NEW_RENO_PRIZEFIGHTER) + " / prizefighter_name == " + message_str(SCRIPT_NEWR1, 400 + global_var(GVAR_NEW_RENO_PRIZEFIGHTER)));
+    if (global_var(GVAR_NEW_RENO_PRIZEFIGHTER) == 497) then begin
+        Reply(300);
     end
     else begin
-        if (global_var(231) == 496) then begin
-            gsay_reply(444, 301);
+        if (global_var(GVAR_NEW_RENO_PRIZEFIGHTER) == 496) then begin
+            Reply(301);
         end
         else begin
-            gsay_reply(444, message_str(444, 302) + message_str(353, 400 + global_var(231)) + message_str(444, 1302));
+            Reply(mstr(302) + message_str(SCRIPT_NEWR1, 400 + global_var(GVAR_NEW_RENO_PRIZEFIGHTER)) + mstr(1302));
         end
     end
-    giq_option(-3, 444, 391, Node986, 50);
-    giq_option(-3, 444, 305, Node999, 50);
+    giq_option(-3, NAME, 391, Node986, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 305, Node999, NEUTRAL_REACTION);
     prev_node := 12;
 end
 
 procedure Node013
 begin
-    gsay_reply(444, 310);
-    giq_option(4, 444, 311, Node040, 51);
+    Reply(310);
+    giq_option(4, NAME, 311, Node040, BAD_REACTION);
     prev_node := 13;
 end
 
 procedure Node014
 begin
     set_local_var(5, local_var(5) bwor 1024);
-    gsay_reply(444, message_str(444, 329) + message_str(444, 320 + (get_critter_stat(dude_obj, 34) == 1)));
-    giq_option(-3, 444, message_str(444, 322) + obj_name(dude_obj) + message_str(444, 1322), Node998, 51);
-    giq_option(-3, 444, 323, Node015, 50);
-    giq_option(-3, 444, 324, Node999, 50);
-    giq_option(-3, 444, 363, Node999, 50);
-    giq_option(-3, 444, 326, Node998, 51);
-    giq_option(4, 444, 327, Node985, 50);
-    giq_option(4, 444, 328, Node999, 50);
+    Reply(mstr(329) + mstr(320 + (dude_gender == 1)));
+    giq_option(-3, NAME, mstr(322) + dude_name + mstr(1322), Node998, BAD_REACTION);
+    giq_option(-3, NAME, 323, Node015, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 324, Node999, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 363, Node999, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 326, Node998, BAD_REACTION);
+    giq_option(4, NAME, 327, Node985, NEUTRAL_REACTION);
+    giq_option(4, NAME, 328, Node999, NEUTRAL_REACTION);
     prev_node := 14;
 end
 
 procedure Node015
 begin
-    gsay_reply(444, message_str(444, 337) + message_str(444, 330 + (get_critter_stat(dude_obj, 34) == 1)) + message_str(444, 332));
-    giq_option(-3, 444, 333, Node998, 51);
-    giq_option(-3, 444, 334, Node999, 50);
-    giq_option(4, 444, 326, Node998, 51);
-    giq_option(4, 444, 336, Node999, 50);
+    Reply(mstr(337) + mstr(330 + (dude_gender == 1)) + mstr(332));
+    giq_option(-3, NAME, 333, Node998, BAD_REACTION);
+    giq_option(-3, NAME, 334, Node999, NEUTRAL_REACTION);
+    giq_option(4, NAME, 326, Node998, BAD_REACTION);
+    giq_option(4, NAME, 336, Node999, NEUTRAL_REACTION);
     prev_node := 15;
 end
 
 procedure Node016
 begin
-    gsay_reply(444, message_str(444, 345) + message_str(444, 340 + (global_var(503) == 75)) + message_str(444, 346));
-    set_global_var(503, 75);
-    giq_option(4, 444, 342, Node979, 50);
-    giq_option(4, 444, 343, Node983, 50);
-    giq_option(4, 444, 344, Node999, 51);
+    Reply(mstr(345) + mstr(340 + (global_var(GVAR_NEW_RENO_STUART_DEAL) == 75)) + mstr(346));
+    set_global_var(GVAR_NEW_RENO_STUART_DEAL, 75);
+    giq_option(4, NAME, 342, Node979, NEUTRAL_REACTION);
+    giq_option(4, NAME, 343, Node983, NEUTRAL_REACTION);
+    giq_option(4, NAME, 344, Node999, BAD_REACTION);
     prev_node := 16;
 end
 
 procedure Node017
 begin
-    gsay_reply(444, message_str(444, 350) + message_str(353, 400 + global_var(231)) + message_str(444, 1350));
-    if (((global_var(4) + global_var(5)) >= 25) and ((global_var(5) > (2 * global_var(4))) or (global_var(3) == 1))) then begin
-        giq_option(-3, 444, 351, Node021, 50);
-    end
-    giq_option(-3, 444, 352, Node999, 50);
-    giq_option(-3, 444, 353, Node018, 50);
-    giq_option(-3, 444, 363, Node999, 50);
-    giq_option(4, 444, 355, Node021, 50);
-    giq_option(4, 444, 356, Node999, 50);
-    giq_option(4, 444, 357, Node020, 51);
-    giq_option(4, 444, 336, Node999, 50);
+    Reply(mstr(350) + message_str(SCRIPT_NEWR1, 400 + global_var(GVAR_NEW_RENO_PRIZEFIGHTER)) + mstr(1350));
+    if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
+        giq_option(-3, NAME, 351, Node021, NEUTRAL_REACTION);
+    end
+    giq_option(-3, NAME, 352, Node999, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 353, Node018, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 363, Node999, NEUTRAL_REACTION);
+    giq_option(4, NAME, 355, Node021, NEUTRAL_REACTION);
+    giq_option(4, NAME, 356, Node999, NEUTRAL_REACTION);
+    giq_option(4, NAME, 357, Node020, BAD_REACTION);
+    giq_option(4, NAME, 336, Node999, NEUTRAL_REACTION);
     prev_node := 17;
 end
 
 procedure Node018
 begin
-    gsay_reply(444, 360);
-    giq_option(-3, 444, 361, Node019, 50);
-    giq_option(-3, 444, 362, Node019, 50);
-    giq_option(-3, 444, 363, Node999, 50);
+    Reply(360);
+    giq_option(-3, NAME, 361, Node019, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 362, Node019, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 363, Node999, NEUTRAL_REACTION);
     prev_node := 18;
 end
 
 procedure Node019
 begin
-    gsay_reply(444, message_str(444, 370) + message_str(444, 371 + (get_critter_stat(dude_obj, 34) == 1)) + message_str(444, 373));
-    giq_option(-3, 444, 374, Node982, 50);
+    Reply(mstr(370) + mstr(371 + (dude_gender == 1)) + mstr(373));
+    giq_option(-3, NAME, 374, Node982, NEUTRAL_REACTION);
     if ((local_var(5) bwand 512) != 0) then begin
-        giq_option(-3, 444, 375, Node999, 50);
+        giq_option(-3, NAME, 375, Node999, NEUTRAL_REACTION);
     end
-    giq_option(-3, 444, 376, Node020, 50);
-    giq_option(-3, 444, 363, Node999, 50);
+    giq_option(-3, NAME, 376, Node020, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 363, Node999, NEUTRAL_REACTION);
     prev_node := 19;
 end
 
 procedure Node020
 begin
-    gsay_reply(444, 380);
-    giq_option(-3, 444, 363, Node999, 50);
+    Reply(380);
+    giq_option(-3, NAME, 363, Node999, NEUTRAL_REACTION);
     prev_node := 20;
 end
 
 procedure Node021
 begin
-    gsay_reply(444, 390);
-    giq_option(1, 444, 391, Node986, 50);
+    Reply(390);
+    giq_option(1, NAME, 391, Node986, NEUTRAL_REACTION);
     prev_node := 21;
 end
 
 procedure Node022
 begin
     if (prev_node == 6) then begin
-        gsay_reply(444, message_str(444, 400) + message_str(444, 401 + (get_critter_stat(dude_obj, 34) == 1)) + message_str(444, 411 + ((get_critter_stat(dude_obj, 34) == 1) and (get_critter_stat(dude_obj, 3) > 5))));
-        if ((get_critter_stat(dude_obj, 34) == 1) and (get_critter_stat(dude_obj, 3) > 5)) then begin
-            giq_option(4, 444, 409, Node013, 50);
+        Reply(mstr(400) + mstr(401 + (dude_gender == 1)) + mstr(411 + ((dude_gender == 1) and (dude_charisma > 5))));
+        if ((dude_gender == 1) and (dude_charisma > 5)) then begin
+            giq_option(4, NAME, 409, Node013, NEUTRAL_REACTION);
         end
     end
     else begin
-        gsay_reply(444, 404);
+        Reply(404);
     end
     set_local_var(5, local_var(5) bwor 1);
-    giq_option(4, 444, message_str(444, 405) + obj_name(dude_obj) + message_str(444, 1405), Node023, 50);
-    giq_option(4, 444, message_str(444, 405) + obj_name(dude_obj) + message_str(444, 1406), Node978, 50);
-    giq_option(4, 444, 407, Node034, 51);
-    giq_option(9, 444, 408, Node033, 50);
-    giq_option(4, 444, 410, Node999, 50);
+    giq_option(4, NAME, mstr(405) + dude_name + mstr(1405), Node023, NEUTRAL_REACTION);
+    giq_option(4, NAME, mstr(405) + dude_name + mstr(1406), Node978, NEUTRAL_REACTION);
+    giq_option(4, NAME, 407, Node034, BAD_REACTION);
+    giq_option(9, NAME, 408, Node033, NEUTRAL_REACTION);
+    giq_option(4, NAME, 410, Node999, NEUTRAL_REACTION);
     prev_node := 22;
 end
 
 procedure Node023
 begin
-    if ((get_critter_stat(dude_obj, 34) == 1) and (get_critter_stat(dude_obj, 3) > 5)) then begin
-        gsay_reply(444, message_str(444, 420) + obj_name(dude_obj) + message_str(444, 422) + obj_name(dude_obj) + message_str(444, 1422) + obj_name(dude_obj) + message_str(444, 3422) + message_str(444, 423));
+    if ((dude_gender == 1) and (dude_charisma > 5)) then begin
+        Reply(mstr(420) + dude_name + mstr(422) + dude_name + mstr(1422) + dude_name + mstr(3422) + mstr(423));
     end
     else begin
-        gsay_reply(444, message_str(444, 420) + message_str(444, 421) + obj_name(dude_obj) + message_str(444, 1421) + message_str(444, 423));
+        Reply(mstr(420) + mstr(421) + dude_name + mstr(1421) + mstr(423));
     end
     set_local_var(5, local_var(5) bwor 1);
-    giq_option(4, 444, 424, Node978, 50);
-    giq_option(4, 444, 410, Node999, 50);
+    giq_option(4, NAME, 424, Node978, NEUTRAL_REACTION);
+    giq_option(4, NAME, 410, Node999, NEUTRAL_REACTION);
     prev_node := 23;
 end
 
 procedure Node024
 begin
-    if (global_var(494) == 8) then begin
-        gsay_reply(444, 430);
+    if (global_var(GVAR_FALLOUT_2) == 8) then begin
+        Reply(430);
     end
     else begin
         if ((local_var(5) bwand 1024) != 0) then begin
-            gsay_reply(444, message_str(444, 432) + message_str(353, 400 + global_var(231)) + message_str(444, 1432));
+            Reply(mstr(432) + message_str(SCRIPT_NEWR1, 400 + global_var(GVAR_NEW_RENO_PRIZEFIGHTER)) + mstr(1432));
         end
         else begin
-            gsay_reply(444, message_str(444, 431) + message_str(353, 400 + global_var(231)) + message_str(444, 1431));
+            Reply(mstr(431) + message_str(SCRIPT_NEWR1, 400 + global_var(GVAR_NEW_RENO_PRIZEFIGHTER)) + mstr(1431));
         end
     end
     if (((local_var(5) bwand 1024) != 0) == 0) then begin
-        giq_option(4, 444, 435, Node021, 50);
+        giq_option(4, NAME, 435, Node021, NEUTRAL_REACTION);
     end
     else begin
-        if ((global_var(494) == 8) == 0) then begin
-            giq_option(4, 444, 436, Node976, 50);
+        if ((global_var(GVAR_FALLOUT_2) == 8) == 0) then begin
+            giq_option(4, NAME, 436, Node976, NEUTRAL_REACTION);
         end
     end
-    giq_option(4, 444, 437, Node999, 50);
-    giq_option(-3, 444, 433, Node977, 50);
-    giq_option(-3, 444, 434, Node999, 50);
+    giq_option(4, NAME, 437, Node999, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 433, Node977, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 434, Node999, NEUTRAL_REACTION);
     prev_node := 24;
 end
 
 procedure Node025
 begin
-    if (get_critter_stat(dude_obj, 34) == 0) then begin
-        gsay_reply(444, 440);
+    if (dude_gender == 0) then begin
+        Reply(440);
     end
     else begin
-        if (get_critter_stat(dude_obj, 3) < 6) then begin
-            gsay_reply(444, 441);
-            giq_option(4, 444, 444, Node981, 51);
+        if (dude_charisma < 6) then begin
+            Reply(441);
+            giq_option(4, NAME, 444, Node981, BAD_REACTION);
         end
         else begin
-            gsay_reply(444, 442);
+            Reply(442);
         end
-        if (has_skill(dude_obj, 14) >= 50) then begin
-            giq_option(6, 444, 452, Node975, 50);
+        if (dude_skill(SKILL_SPEECH) >= 50) then begin
+            giq_option(6, NAME, 452, Node975, NEUTRAL_REACTION);
         end
     end
-    giq_option(4, 444, 445, Node981, 50);
-    giq_option(4, 444, 446, Node999, 50);
+    giq_option(4, NAME, 445, Node981, NEUTRAL_REACTION);
+    giq_option(4, NAME, 446, Node999, NEUTRAL_REACTION);
     prev_node := 25;
 end
 
 procedure Node026
 begin
-    gsay_reply(444, 450 + (get_critter_stat(dude_obj, 34) == 0));
-    if (has_skill(dude_obj, 14) >= 50) then begin
-        giq_option(6, 444, 452 + (get_critter_stat(dude_obj, 34) == 0), Node975, 49);
+    Reply(450 + (dude_gender == 0));
+    if (dude_skill(SKILL_SPEECH) >= 50) then begin
+        giq_option(6, NAME, 452 + (dude_gender == 0), Node975, GOOD_REACTION);
     end
-    giq_option(4, 444, 454, Node999, 50);
+    giq_option(4, NAME, 454, Node999, NEUTRAL_REACTION);
     prev_node := 26;
 end
 
 procedure Node027
 begin
-    gsay_reply(444, 460 + (get_critter_stat(dude_obj, 34) == 0));
+    Reply(460 + (dude_gender == 0));
     if (prev_node == 27) then begin
-        if ((has_skill(dude_obj, 14) >= 50) and (get_critter_stat(dude_obj, 34) == 1) and (get_critter_stat(dude_obj, 3) > 5)) then begin
-            giq_option(6, 444, 462, Node975, 49);
+        if ((dude_skill(SKILL_SPEECH) >= 50) and (dude_gender == 1) and (dude_charisma > 5)) then begin
+            giq_option(6, NAME, 462, Node975, GOOD_REACTION);
         end
     end
-    giq_option(4, 444, 454, Node999, 50);
+    giq_option(4, NAME, 454, Node999, NEUTRAL_REACTION);
     prev_node := 27;
 end
 
 procedure Node028
 begin
-    set_global_var(503, 50);
-    gsay_reply(444, 470 + ((get_critter_stat(dude_obj, 34) == 0) and (get_critter_stat(dude_obj, 0) > 5)));
-    giq_option(4, 444, 472, Node030, 50);
-    giq_option(4, 444, 473, Node980, 51);
-    giq_option(4, 444, 454, Node999, 51);
+    set_global_var(GVAR_NEW_RENO_STUART_DEAL, 50);
+    Reply(470 + ((dude_gender == 0) and (dude_strength > 5)));
+    giq_option(4, NAME, 472, Node030, NEUTRAL_REACTION);
+    giq_option(4, NAME, 473, Node980, BAD_REACTION);
+    giq_option(4, NAME, 454, Node999, BAD_REACTION);
     prev_node := 28;
 end
 
 procedure Node029
 begin
-    set_global_var(503, 25);
-    gsay_reply(444, 480);
-    giq_option(4, 444, 481, Node030, 50);
+    set_global_var(GVAR_NEW_RENO_STUART_DEAL, 25);
+    Reply(480);
+    giq_option(4, NAME, 481, Node030, NEUTRAL_REACTION);
     prev_node := 29;
 end
 
@@ -1217,135 +1217,135 @@ procedure Node030
 begin
     var_num := 231;
     top_msg := 400 + 1;
-    stored_msg := message_str(444, 490) + obj_name(dude_obj) + message_str(444, 1490) + obj_name(dude_obj) + message_str(444, 2490);
+    stored_msg := mstr(490) + dude_name + mstr(1490) + dude_name + mstr(2490);
     stored_node := "Node031";
     call ChooseFromList();
 end
 
 procedure Node031
 begin
-    if (global_var(231) == 11) then begin
-        gsay_reply(444, 500);
+    if (global_var(GVAR_NEW_RENO_PRIZEFIGHTER) == 11) then begin
+        Reply(500);
     end
     else begin
-        if (global_var(231) == 12) then begin
-            gsay_reply(444, 501);
+        if (global_var(GVAR_NEW_RENO_PRIZEFIGHTER) == 12) then begin
+            Reply(501);
         end
         else begin
-            if (global_var(231) == 13) then begin
-                gsay_reply(444, 502);
+            if (global_var(GVAR_NEW_RENO_PRIZEFIGHTER) == 13) then begin
+                Reply(502);
             end
             else begin
-                if (global_var(231) == 14) then begin
-                    gsay_reply(444, 503);
+                if (global_var(GVAR_NEW_RENO_PRIZEFIGHTER) == 14) then begin
+                    Reply(503);
                 end
                 else begin
-                    if (global_var(231) == 15) then begin
-                        gsay_reply(444, 504);
+                    if (global_var(GVAR_NEW_RENO_PRIZEFIGHTER) == 15) then begin
+                        Reply(504);
                     end
                     else begin
-                        gsay_reply(444, message_str(444, 505) + message_str(353, 400 + global_var(231)) + message_str(444, 422) + message_str(353, 400 + global_var(231)) + message_str(444, 2505));
+                        Reply(mstr(505) + message_str(SCRIPT_NEWR1, 400 + global_var(GVAR_NEW_RENO_PRIZEFIGHTER)) + mstr(422) + message_str(SCRIPT_NEWR1, 400 + global_var(GVAR_NEW_RENO_PRIZEFIGHTER)) + mstr(2505));
                     end
                 end
             end
         end
     end
-    giq_option(1, 444, 506, Node986, 50);
+    giq_option(1, NAME, 506, Node986, NEUTRAL_REACTION);
     prev_node := 31;
 end
 
 procedure Node032
 begin
-    gsay_reply(444, 510);
-    giq_option(4, 444, 511, Node030, 51);
-    giq_option(4, 444, 512, Node030, 50);
-    giq_option(4, 444, 513, Node030, 50);
-    giq_option(4, 444, 514, Node999, 51);
+    Reply(510);
+    giq_option(4, NAME, 511, Node030, BAD_REACTION);
+    giq_option(4, NAME, 512, Node030, NEUTRAL_REACTION);
+    giq_option(4, NAME, 513, Node030, NEUTRAL_REACTION);
+    giq_option(4, NAME, 514, Node999, BAD_REACTION);
     prev_node := 32;
 end
 
 procedure Node033
 begin
     if (prev_node == 39) then begin
-        gsay_reply(444, 521);
-        giq_option(9, 444, 408, Node033, 50);
+        Reply(521);
+        giq_option(9, NAME, 408, Node033, NEUTRAL_REACTION);
     end
     else begin
-        gsay_reply(444, 520);
-        giq_option(4, 444, 407, Node034, 51);
+        Reply(520);
+        giq_option(4, NAME, 407, Node034, BAD_REACTION);
     end
-    giq_option(4, 444, message_str(444, 405) + obj_name(dude_obj) + message_str(444, 1405), Node023, 50);
-    giq_option(4, 444, message_str(444, 405) + obj_name(dude_obj) + message_str(444, 1406), Node978, 50);
-    giq_option(4, 444, 526, Node999, 50);
+    giq_option(4, NAME, mstr(405) + dude_name + mstr(1405), Node023, NEUTRAL_REACTION);
+    giq_option(4, NAME, mstr(405) + dude_name + mstr(1406), Node978, NEUTRAL_REACTION);
+    giq_option(4, NAME, 526, Node999, NEUTRAL_REACTION);
     prev_node := 33;
 end
 
 procedure Node034
 begin
-    gsay_reply(444, 530);
-    giq_option(4, 444, 531, Node035, 51);
+    Reply(530);
+    giq_option(4, NAME, 531, Node035, BAD_REACTION);
     prev_node := 34;
 end
 
 procedure Node035
 begin
-    gsay_reply(444, 540);
-    giq_option(4, 444, 541, Node036, 51);
+    Reply(540);
+    giq_option(4, NAME, 541, Node036, BAD_REACTION);
     prev_node := 35;
 end
 
 procedure Node036
 begin
-    gsay_reply(444, 550);
-    giq_option(4, 444, 551, Node037, 50);
+    Reply(550);
+    giq_option(4, NAME, 551, Node037, NEUTRAL_REACTION);
     prev_node := 36;
 end
 
 procedure Node037
 begin
-    gsay_reply(444, 560);
-    giq_option(4, 444, 561, Node038, 50);
+    Reply(560);
+    giq_option(4, NAME, 561, Node038, NEUTRAL_REACTION);
     prev_node := 37;
 end
 
 procedure Node038
 begin
-    gsay_reply(444, 570);
-    giq_option(4, 444, 571, Node041, 50);
+    Reply(570);
+    giq_option(4, NAME, 571, Node041, NEUTRAL_REACTION);
     prev_node := 38;
 end
 
 procedure Node039
 begin
-    gsay_reply(444, 580);
-    giq_option(4, 444, 581, Node033, 50);
+    Reply(580);
+    giq_option(4, NAME, 581, Node033, NEUTRAL_REACTION);
     prev_node := 39;
 end
 
 procedure Node040
 begin
-    gsay_reply(444, 590);
-    giq_option(4, 444, 591, Node022, 50);
+    Reply(590);
+    giq_option(4, NAME, 591, Node022, NEUTRAL_REACTION);
     prev_node := 40;
 end
 
 procedure Node041
 begin
-    gsay_reply(444, 600 + (get_critter_stat(dude_obj, 34) == 1));
-    giq_option(4, 444, 602, Node042, 50);
+    Reply(600 + (dude_gender == 1));
+    giq_option(4, NAME, 602, Node042, NEUTRAL_REACTION);
     prev_node := 41;
 end
 
 procedure Node042
 begin
-    gsay_reply(444, 610 + (get_critter_stat(dude_obj, 34) == 1));
-    giq_option(4, 444, 612, Node039, 50);
+    Reply(610 + (dude_gender == 1));
+    giq_option(4, NAME, 612, Node039, NEUTRAL_REACTION);
     prev_node := 42;
 end
 
 procedure Node990
 begin
-    if ((get_critter_stat(dude_obj, 34) == 1) or (get_critter_stat(dude_obj, 0) < 6)) then begin
+    if ((dude_gender == 1) or (dude_strength < 6)) then begin
         call Node007();
     end
     else begin
@@ -1355,33 +1355,33 @@ end
 
 procedure Node989
 begin
-    set_global_var(global_var(231), 497);
+    set_global_var(global_var(GVAR_NEW_RENO_PRIZEFIGHTER), 497);
     call Node012();
 end
 
 procedure Node988
 begin
-    set_global_var(global_var(231), 498);
+    set_global_var(global_var(GVAR_NEW_RENO_PRIZEFIGHTER), 498);
     call Node012();
 end
 
 procedure Node987
 begin
-    set_global_var(global_var(231), 499);
+    set_global_var(global_var(GVAR_NEW_RENO_PRIZEFIGHTER), 499);
     call Node012();
 end
 
 procedure Node986
 begin
     if (((local_var(5) bwand 2048) != 0) == 0) then begin
-        restock_obj := create_object_sid(292, 0, 0, -1);
+        restock_obj := create_object(292, 0, 0);
         add_mult_objs_to_inven(dude_obj, restock_obj, 1);
     end
     set_local_var(5, local_var(5) bwor 2048);
     set_local_var(5, local_var(5) bwor 512);
     last_timed_obj := self_obj;
     last_timed_param := last_timed_param;
-    metarule3(100, self_obj, 46, 0);
+    rm_fixed_timer_event(self_obj, 46);
     add_timer_event(self_obj, game_ticks(0), 46);
 end
 
@@ -1398,14 +1398,14 @@ end
 
 procedure Node984
 begin
-    set_global_var(global_var(231), 496);
+    set_global_var(global_var(GVAR_NEW_RENO_PRIZEFIGHTER), 496);
     call Node012();
 end
 
 procedure Node983
 begin
     set_local_var(5, local_var(5) bwand (-1 - 1024));
-    set_global_var(503, 75);
+    set_global_var(GVAR_NEW_RENO_STUART_DEAL, 75);
 end
 
 procedure Node982
@@ -1431,8 +1431,8 @@ end
 
 procedure Node980
 begin
-    if ((has_skill(dude_obj, 15) > 75) and (has_skill(dude_obj, 14) > 75)) then begin
-        set_global_var(503, 25);
+    if ((dude_skill(SKILL_BARTER) > 75) and (dude_skill(SKILL_SPEECH) > 75)) then begin
+        set_global_var(GVAR_NEW_RENO_STUART_DEAL, 25);
         call Node029();
     end
     else begin
@@ -1448,7 +1448,7 @@ end
 
 procedure Node978
 begin
-    if ((get_critter_stat(dude_obj, 34) == 1) or (get_critter_stat(dude_obj, 0) < 6)) then begin
+    if ((dude_gender == 1) or (dude_strength < 6)) then begin
         call Node025();
     end
     else begin
@@ -1458,7 +1458,7 @@ end
 
 procedure Node977
 begin
-    if (((global_var(494) == 8) == 0) and ((local_var(5) bwand 1024) != 0)) then begin
+    if (((global_var(GVAR_FALLOUT_2) == 8) == 0) and ((local_var(5) bwand 1024) != 0)) then begin
         call Node015();
     end
     else begin
@@ -1468,7 +1468,7 @@ end
 
 procedure Node976
 begin
-    if ((global_var(494) == 8) or is_success(roll_vs_skill(dude_obj, 14, 0))) then begin
+    if ((global_var(GVAR_FALLOUT_2) == 8) or is_success(roll_vs_skill(dude_obj, 14, 0))) then begin
         call Node015();
     end
     else begin
@@ -1485,4 +1485,3 @@ begin
         call Node027();
     end
 end
-
