diff --git a/Games/Fallout2.Steam/Original/NCWRIPLY.ssl b/Games/Fallout2.Steam/ReDefine/NCWRIPLY.ssl
index a4a8e91..a8069f3 100644
--- a/Games/Fallout2.Steam/Original/NCWRIPLY.ssl
+++ b/Games/Fallout2.Steam/ReDefine/NCWRIPLY.ssl
@@ -220,53 +220,53 @@ end
 procedure ChooseFromList
 begin
     variable LVar0 := 0;
-    gsay_reply(447, stored_msg);
-    if (message_str(353, top_msg - 1) != "") then begin
-        giq_option(1, 447, message_str(353, 150), CLICK_PREV, 50);
+    Reply(stored_msg);
+    if (message_str(SCRIPT_NEWR1, top_msg - 1) != "") then begin
+        giq_option(1, NAME, message_str(SCRIPT_NEWR1, 150), CLICK_PREV, NEUTRAL_REACTION);
     end
     if (LVar0 == 0) then begin
-        if (message_str(353, top_msg + 0) != "") then begin
-            giq_option(1, 447, message_str(353, top_msg + 0), CLICK_0, 50);
+        if (message_str(SCRIPT_NEWR1, top_msg + 0) != "") then begin
+            giq_option(1, NAME, message_str(SCRIPT_NEWR1, top_msg + 0), CLICK_0, NEUTRAL_REACTION);
         end
         else begin
             LVar0 := 1;
         end
     end
     if (LVar0 == 0) then begin
-        if (message_str(353, top_msg + 1) != "") then begin
-            giq_option(1, 447, message_str(353, top_msg + 1), CLICK_1, 50);
+        if (message_str(SCRIPT_NEWR1, top_msg + 1) != "") then begin
+            giq_option(1, NAME, message_str(SCRIPT_NEWR1, top_msg + 1), CLICK_1, NEUTRAL_REACTION);
         end
         else begin
             LVar0 := 1;
         end
     end
     if (LVar0 == 0) then begin
-        if (message_str(353, top_msg + 2) != "") then begin
-            giq_option(1, 447, message_str(353, top_msg + 2), CLICK_2, 50);
+        if (message_str(SCRIPT_NEWR1, top_msg + 2) != "") then begin
+            giq_option(1, NAME, message_str(SCRIPT_NEWR1, top_msg + 2), CLICK_2, NEUTRAL_REACTION);
         end
         else begin
             LVar0 := 1;
         end
     end
     if (LVar0 == 0) then begin
-        if (message_str(353, top_msg + 3) != "") then begin
-            giq_option(1, 447, message_str(353, top_msg + 3), CLICK_3, 50);
+        if (message_str(SCRIPT_NEWR1, top_msg + 3) != "") then begin
+            giq_option(1, NAME, message_str(SCRIPT_NEWR1, top_msg + 3), CLICK_3, NEUTRAL_REACTION);
         end
         else begin
             LVar0 := 1;
         end
     end
     if (LVar0 == 0) then begin
-        if (message_str(353, top_msg + 4) != "") then begin
-            giq_option(1, 447, message_str(353, top_msg + 4), CLICK_4, 50);
+        if (message_str(SCRIPT_NEWR1, top_msg + 4) != "") then begin
+            giq_option(1, NAME, message_str(SCRIPT_NEWR1, top_msg + 4), CLICK_4, NEUTRAL_REACTION);
         end
         else begin
             LVar0 := 1;
         end
     end
     if (LVar0 == 0) then begin
-        if (message_str(353, top_msg + 5) != "") then begin
-            giq_option(1, 447, message_str(353, 151), CLICK_MORE, 50);
+        if (message_str(SCRIPT_NEWR1, top_msg + 5) != "") then begin
+            giq_option(1, NAME, message_str(SCRIPT_NEWR1, 151), CLICK_MORE, NEUTRAL_REACTION);
         end
     end
 end
@@ -325,10 +325,10 @@ begin
                 if (anim_busy(self_obj) == 0) then begin
                     if ((tile_distance_objs(self_obj, dude_obj) < 8) or obj_can_see_obj(self_obj, dude_obj)) then begin
                         if (anim_busy(self_obj) == 0) then begin
-                            global_temp := rotation_to_tile(tile_num(dude_obj), tile_num(self_obj));
+                            global_temp := rotation_to_tile(dude_tile, tile_num(self_obj));
                             animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(self_obj), global_temp, random(3, 10)), 1);
                             global_temp := (global_temp + 1) % 6;
-                            while ((anim_busy(self_obj) == 0) and (global_temp != rotation_to_tile(tile_num(dude_obj), tile_num(self_obj)))) do begin
+                            while ((anim_busy(self_obj) == 0) and (global_temp != rotation_to_tile(dude_tile, tile_num(self_obj)))) do begin
                                 debug_msg("flee loop: rot == " + global_temp);
                                 animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(self_obj), global_temp, random(3, 10)), 1);
                                 global_temp := (global_temp + 1) % 6;
@@ -338,17 +338,17 @@ begin
                 end
             end
             else begin
-                attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+                attack(dude_obj);
             end
         end
         else begin
             if ((local_var(4) bwand 2048) != 0) then begin
                 if ((tile_distance_objs(self_obj, dude_obj) < 8) or obj_can_see_obj(self_obj, dude_obj)) then begin
                     if (anim_busy(self_obj) == 0) then begin
-                        global_temp := rotation_to_tile(tile_num(dude_obj), tile_num(self_obj));
+                        global_temp := rotation_to_tile(dude_tile, tile_num(self_obj));
                         animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(self_obj), global_temp, random(3, 10)), 1);
                         global_temp := (global_temp + 1) % 6;
-                        while ((anim_busy(self_obj) == 0) and (global_temp != rotation_to_tile(tile_num(dude_obj), tile_num(self_obj)))) do begin
+                        while ((anim_busy(self_obj) == 0) and (global_temp != rotation_to_tile(dude_tile, tile_num(self_obj)))) do begin
                             debug_msg("flee loop: rot == " + global_temp);
                             animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(self_obj), global_temp, random(3, 10)), 1);
                             global_temp := (global_temp + 1) % 6;
@@ -379,12 +379,12 @@ begin
                                                         animate_move_obj_to_tile(self_obj, step_tile, 1);
                                                     end
                                                     if (anim_busy(self_obj) == 0) then begin
-                                                        if (metarule3(106, tile_num_in_direction(tile_num(self_obj), has_trait(1, self_obj, 10), 1), elevation(self_obj), 0) != 0) then begin
+                                                        if (tile_get_next_critter(tile_num_in_direction(tile_num(self_obj), obj_get_rot(self_obj), 1), elevation(self_obj), 0) != 0) then begin
                                                             if (anim_busy(self_obj) == 0) then begin
-                                                                global_temp := rotation_to_tile(tile_num_in_direction(tile_num(self_obj), rotation_to_tile(tile_num_in_direction(tile_num(self_obj), has_trait(1, self_obj, 10), 1), tile_num(self_obj)), 1), tile_num(self_obj));
+                                                                global_temp := rotation_to_tile(tile_num_in_direction(tile_num(self_obj), rotation_to_tile(tile_num_in_direction(tile_num(self_obj), obj_get_rot(self_obj), 1), tile_num(self_obj)), 1), tile_num(self_obj));
                                                                 animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(self_obj), global_temp, random(2, 5)), 0);
                                                                 global_temp := (global_temp + 1) % 6;
-                                                                while ((anim_busy(self_obj) == 0) and (global_temp != rotation_to_tile(tile_num_in_direction(tile_num(self_obj), rotation_to_tile(tile_num_in_direction(tile_num(self_obj), has_trait(1, self_obj, 10), 1), tile_num(self_obj)), 1), tile_num(self_obj)))) do begin
+                                                                while ((anim_busy(self_obj) == 0) and (global_temp != rotation_to_tile(tile_num_in_direction(tile_num(self_obj), rotation_to_tile(tile_num_in_direction(tile_num(self_obj), obj_get_rot(self_obj), 1), tile_num(self_obj)), 1), tile_num(self_obj)))) do begin
                                                                     debug_msg("flee loop: rot == " + global_temp);
                                                                     animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(self_obj), global_temp, random(2, 5)), 0);
                                                                     global_temp := (global_temp + 1) % 6;
@@ -405,7 +405,7 @@ begin
                                         end
                                     end
                                     else begin
-                                        if (rotation_to_tile(tile_num(self_obj), tile_num(new_reno_wright_teen_obj)) != has_trait(1, self_obj, 10)) then begin
+                                        if (rotation_to_tile(tile_num(self_obj), tile_num(new_reno_wright_teen_obj)) != obj_get_rot(self_obj)) then begin
                                             if (anim_busy(self_obj) == 0) then begin
                                                 anim(self_obj, 1000, rotation_to_tile(tile_num(self_obj), tile_num(new_reno_wright_teen_obj)));
                                             end
@@ -417,10 +417,10 @@ begin
                         if (set_flee == 0) then begin
                             if ((tile_distance_objs(self_obj, dude_obj) < 8) or obj_can_see_obj(self_obj, dude_obj)) then begin
                                 if (anim_busy(self_obj) == 0) then begin
-                                    global_temp := rotation_to_tile(tile_num(dude_obj), tile_num(self_obj));
+                                    global_temp := rotation_to_tile(dude_tile, tile_num(self_obj));
                                     animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(self_obj), global_temp, random(3, 10)), 1);
                                     global_temp := (global_temp + 1) % 6;
-                                    while ((anim_busy(self_obj) == 0) and (global_temp != rotation_to_tile(tile_num(dude_obj), tile_num(self_obj)))) do begin
+                                    while ((anim_busy(self_obj) == 0) and (global_temp != rotation_to_tile(dude_tile, tile_num(self_obj)))) do begin
                                         debug_msg("flee loop: rot == " + global_temp);
                                         animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(self_obj), global_temp, random(3, 10)), 1);
                                         global_temp := (global_temp + 1) % 6;
@@ -448,7 +448,7 @@ end
 procedure talk_p_proc
 begin
     script_overrides;
-    if (global_var(494) == 8) then begin
+    if (global_var(GVAR_FALLOUT_2) == 8) then begin
         call Node004();
     end
     else begin
@@ -513,48 +513,48 @@ begin
                     Evil_Critter := 1;
                     Slavery_Tolerant := 0;
                     Karma_Perception := 50;
-                    set_global_var(37, 0);
-                    set_global_var(38, 0);
-                    set_global_var(39, 0);
-                    set_global_var(40, 0);
-                    set_global_var(41, 0);
-                    set_global_var(42, 0);
-                    set_global_var(43, 0);
-                    set_global_var(44, 0);
-                    set_global_var(45, 0);
-                    if (global_var(0) >= 1000) then begin
-                        set_global_var(37, 1);
+                    set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+                    set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+                    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+                    set_global_var(GVAR_KARMA_DEFENDER, 0);
+                    set_global_var(GVAR_KARMA_WANDERER, 0);
+                    set_global_var(GVAR_KARMA_BETRAYER, 0);
+                    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+                    set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+                    set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+                        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
                     end
                     else begin
-                        if (global_var(0) >= 750) then begin
-                            set_global_var(38, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                            set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
                         end
                         else begin
-                            if (global_var(0) >= 500) then begin
-                                set_global_var(39, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                                set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                             end
                             else begin
-                                if (global_var(0) >= 250) then begin
-                                    set_global_var(40, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                                    set_global_var(GVAR_KARMA_DEFENDER, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -250) then begin
-                                        set_global_var(41, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                                        set_global_var(GVAR_KARMA_WANDERER, 1);
                                     end
                                     else begin
-                                        if (global_var(0) > -500) then begin
-                                            set_global_var(42, 1);
+                                        if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                            set_global_var(GVAR_KARMA_BETRAYER, 1);
                                         end
                                         else begin
-                                            if (global_var(0) > -750) then begin
-                                                set_global_var(43, 1);
+                                            if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                                set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                             end
                                             else begin
-                                                if (global_var(0) > -1000) then begin
-                                                    set_global_var(44, 1);
+                                                if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                                    set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                                 end
                                                 else begin
-                                                    set_global_var(45, 1);
+                                                    set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                                 end
                                             end
                                         end
@@ -565,7 +565,7 @@ begin
                     end
                     if (local_var(2) == 0) then begin
                         set_local_var(2, 1);
-                        set_local_var(3, (get_critter_stat(dude_obj, 3) - get_critter_stat(self_obj, 3)) * 5);
+                        set_local_var(3, (dude_charisma - self_charisma) * 5);
                         debug_msg("Base Reaction == " + local_var(3));
                     end
                     Static_Reaction := 0;
@@ -577,14 +577,14 @@ begin
                     else begin
                         reaction_bonus_karma := -1 * (-1 * reaction_bonus_karma / 1000);
                     end
-                    if (global_var(55) >= 0) then begin
-                        reaction_bonus_town_rep := global_var(55) / 2;
+                    if (global_var(GVAR_TOWN_REP_NEW_RENO) >= 0) then begin
+                        reaction_bonus_town_rep := global_var(GVAR_TOWN_REP_NEW_RENO) / 2;
                     end
                     else begin
-                        reaction_bonus_town_rep := -1 * (-1 * global_var(55) / 2);
+                        reaction_bonus_town_rep := -1 * (-1 * global_var(GVAR_TOWN_REP_NEW_RENO) / 2);
                     end
-                    if (has_trait(2, dude_obj, 13)) then begin
-                        if (get_critter_stat(dude_obj, 34) == get_critter_stat(self_obj, 34)) then begin
+                    if (has_trait(TRAIT_TRAIT, dude_obj, TRAIT_sex_appeal)) then begin
+                        if (dude_gender == self_gender) then begin
                             Static_Reaction := Static_Reaction + -20;
                         end
                         else begin
@@ -600,48 +600,48 @@ begin
                         debug_msg("Karma Reaction Bonus == " + Static_Reaction);
                         Static_Reaction := Static_Reaction - reaction_bonus_town_rep;
                         debug_msg("Town Rep Bonus == " + Static_Reaction);
-                        set_global_var(37, 0);
-                        set_global_var(38, 0);
-                        set_global_var(39, 0);
-                        set_global_var(40, 0);
-                        set_global_var(41, 0);
-                        set_global_var(42, 0);
-                        set_global_var(43, 0);
-                        set_global_var(44, 0);
-                        set_global_var(45, 0);
-                        if (global_var(0) >= 1000) then begin
-                            set_global_var(37, 1);
+                        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+                        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+                        set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+                        set_global_var(GVAR_KARMA_DEFENDER, 0);
+                        set_global_var(GVAR_KARMA_WANDERER, 0);
+                        set_global_var(GVAR_KARMA_BETRAYER, 0);
+                        set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+                        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+                        set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+                        if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+                            set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
                         end
                         else begin
-                            if (global_var(0) >= 750) then begin
-                                set_global_var(38, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
                             end
                             else begin
-                                if (global_var(0) >= 500) then begin
-                                    set_global_var(39, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                                    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                                 end
                                 else begin
-                                    if (global_var(0) >= 250) then begin
-                                        set_global_var(40, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                                        set_global_var(GVAR_KARMA_DEFENDER, 1);
                                     end
                                     else begin
-                                        if (global_var(0) > -250) then begin
-                                            set_global_var(41, 1);
+                                        if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                                            set_global_var(GVAR_KARMA_WANDERER, 1);
                                         end
                                         else begin
-                                            if (global_var(0) > -500) then begin
-                                                set_global_var(42, 1);
+                                            if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                                set_global_var(GVAR_KARMA_BETRAYER, 1);
                                             end
                                             else begin
-                                                if (global_var(0) > -750) then begin
-                                                    set_global_var(43, 1);
+                                                if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                                    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                                 end
                                                 else begin
-                                                    if (global_var(0) > -1000) then begin
-                                                        set_global_var(44, 1);
+                                                    if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                                        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                                     end
                                                     else begin
-                                                        set_global_var(45, 1);
+                                                        set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                                     end
                                                 end
                                             end
@@ -650,35 +650,35 @@ begin
                                 end
                             end
                         end
-                        if (global_var(37) == 1) then begin
+                        if (global_var(GVAR_KARMA_HOLY_WARRIOR) == 1) then begin
                             Static_Reaction := Static_Reaction - 20;
                         end
                         else begin
-                            if (global_var(38) == 1) then begin
+                            if (global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES) == 1) then begin
                                 Static_Reaction := Static_Reaction - 15;
                             end
                             else begin
-                                if (global_var(39) == 1) then begin
+                                if (global_var(GVAR_KARMA_SHIELD_OF_HOPE) == 1) then begin
                                     Static_Reaction := Static_Reaction - 10;
                                 end
                                 else begin
-                                    if (global_var(40) == 1) then begin
+                                    if (global_var(GVAR_KARMA_DEFENDER) == 1) then begin
                                         Static_Reaction := Static_Reaction - 5;
                                     end
                                     else begin
-                                        if (global_var(41) == 1) then begin
+                                        if (global_var(GVAR_KARMA_WANDERER) == 1) then begin
                                             Static_Reaction := Static_Reaction - 0;
                                         end
                                         else begin
-                                            if (global_var(42) == 1) then begin
+                                            if (global_var(GVAR_KARMA_BETRAYER) == 1) then begin
                                                 Static_Reaction := Static_Reaction - -5;
                                             end
                                             else begin
-                                                if (global_var(43) == 1) then begin
+                                                if (global_var(GVAR_KARMA_SWORD_OF_DESPAIR) == 1) then begin
                                                     Static_Reaction := Static_Reaction - -10;
                                                 end
                                                 else begin
-                                                    if (global_var(44) == 1) then begin
+                                                    if (global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE) == 1) then begin
                                                         Static_Reaction := Static_Reaction - -15;
                                                     end
                                                     else begin
@@ -698,48 +698,48 @@ begin
                         debug_msg("Karma Reaction Bonus == " + Static_Reaction);
                         Static_Reaction := Static_Reaction + reaction_bonus_town_rep;
                         debug_msg("Town Rep Bonus == " + Static_Reaction);
-                        set_global_var(37, 0);
-                        set_global_var(38, 0);
-                        set_global_var(39, 0);
-                        set_global_var(40, 0);
-                        set_global_var(41, 0);
-                        set_global_var(42, 0);
-                        set_global_var(43, 0);
-                        set_global_var(44, 0);
-                        set_global_var(45, 0);
-                        if (global_var(0) >= 1000) then begin
-                            set_global_var(37, 1);
+                        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+                        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+                        set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+                        set_global_var(GVAR_KARMA_DEFENDER, 0);
+                        set_global_var(GVAR_KARMA_WANDERER, 0);
+                        set_global_var(GVAR_KARMA_BETRAYER, 0);
+                        set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+                        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+                        set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+                        if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+                            set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
                         end
                         else begin
-                            if (global_var(0) >= 750) then begin
-                                set_global_var(38, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
                             end
                             else begin
-                                if (global_var(0) >= 500) then begin
-                                    set_global_var(39, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                                    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                                 end
                                 else begin
-                                    if (global_var(0) >= 250) then begin
-                                        set_global_var(40, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                                        set_global_var(GVAR_KARMA_DEFENDER, 1);
                                     end
                                     else begin
-                                        if (global_var(0) > -250) then begin
-                                            set_global_var(41, 1);
+                                        if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                                            set_global_var(GVAR_KARMA_WANDERER, 1);
                                         end
                                         else begin
-                                            if (global_var(0) > -500) then begin
-                                                set_global_var(42, 1);
+                                            if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                                set_global_var(GVAR_KARMA_BETRAYER, 1);
                                             end
                                             else begin
-                                                if (global_var(0) > -750) then begin
-                                                    set_global_var(43, 1);
+                                                if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                                    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                                 end
                                                 else begin
-                                                    if (global_var(0) > -1000) then begin
-                                                        set_global_var(44, 1);
+                                                    if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                                        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                                     end
                                                     else begin
-                                                        set_global_var(45, 1);
+                                                        set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                                     end
                                                 end
                                             end
@@ -748,35 +748,35 @@ begin
                                 end
                             end
                         end
-                        if (global_var(37) == 1) then begin
+                        if (global_var(GVAR_KARMA_HOLY_WARRIOR) == 1) then begin
                             Static_Reaction := Static_Reaction + 20;
                         end
                         else begin
-                            if (global_var(38) == 1) then begin
+                            if (global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES) == 1) then begin
                                 Static_Reaction := Static_Reaction + 15;
                             end
                             else begin
-                                if (global_var(39) == 1) then begin
+                                if (global_var(GVAR_KARMA_SHIELD_OF_HOPE) == 1) then begin
                                     Static_Reaction := Static_Reaction + 10;
                                 end
                                 else begin
-                                    if (global_var(40) == 1) then begin
+                                    if (global_var(GVAR_KARMA_DEFENDER) == 1) then begin
                                         Static_Reaction := Static_Reaction + 5;
                                     end
                                     else begin
-                                        if (global_var(41) == 1) then begin
+                                        if (global_var(GVAR_KARMA_WANDERER) == 1) then begin
                                             Static_Reaction := Static_Reaction + 0;
                                         end
                                         else begin
-                                            if (global_var(42) == 1) then begin
+                                            if (global_var(GVAR_KARMA_BETRAYER) == 1) then begin
                                                 Static_Reaction := Static_Reaction + -5;
                                             end
                                             else begin
-                                                if (global_var(43) == 1) then begin
+                                                if (global_var(GVAR_KARMA_SWORD_OF_DESPAIR) == 1) then begin
                                                     Static_Reaction := Static_Reaction + -10;
                                                 end
                                                 else begin
-                                                    if (global_var(44) == 1) then begin
+                                                    if (global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE) == 1) then begin
                                                         Static_Reaction := Static_Reaction + -15;
                                                     end
                                                     else begin
@@ -792,34 +792,34 @@ begin
                         debug_msg("Good Person Reaction Bonus == " + Static_Reaction);
                     end
                     if (Slavery_Tolerant == 0) then begin
-                        if (global_var(11) == 1) then begin
+                        if (global_var(GVAR_REPUTATION_SLAVER) == 1) then begin
                             Static_Reaction := Static_Reaction + (2 * -25);
                         end
                     end
                     else begin
                         if (Slavery_Tolerant == 1) then begin
-                            if (global_var(11) == 1) then begin
+                            if (global_var(GVAR_REPUTATION_SLAVER) == 1) then begin
                                 Static_Reaction := Static_Reaction + -25;
                             end
                         end
                         else begin
                             if (Slavery_Tolerant == 3) then begin
-                                if (global_var(11) == 1) then begin
+                                if (global_var(GVAR_REPUTATION_SLAVER) == 1) then begin
                                     Static_Reaction := Static_Reaction - -25;
                                 end
                             end
                         end
                     end
                     debug_msg("Slaver + Aligned Reaction == " + Static_Reaction);
-                    Static_Reaction := Static_Reaction + (10 * has_trait(0, dude_obj, 10));
+                    Static_Reaction := Static_Reaction + (10 * has_trait(TRAIT_PERK, dude_obj, PERK_presence));
                     debug_msg("Presence Reaction == " + Static_Reaction);
-                    if (global_var(1) >= 2) then begin
+                    if (global_var(GVAR_CHILDKILLER_REPUTATION) >= 2) then begin
                         Static_Reaction := Static_Reaction + -30;
                     end
                     debug_msg("Childkiller Reaction == " + Static_Reaction);
                     set_local_var(0, local_var(3) + Static_Reaction);
                     debug_msg("Initial Reaction == " + local_var(0));
-                    if ((local_var(1) < 0) or (global_var(11) == 1) or ((global_var(230) > 0) and ((global_var(230) / 100) != 4))) then begin
+                    if ((local_var(1) < 0) or (global_var(GVAR_REPUTATION_SLAVER) == 1) or ((global_var(GVAR_NEW_RENO_MADE_MAN) > 0) and ((global_var(GVAR_NEW_RENO_MADE_MAN) / 100) != 4))) then begin
                         call Node006();
                     end
                     else begin
@@ -846,59 +846,59 @@ end
 procedure destroy_p_proc
 begin
     if (source_obj == dude_obj) then begin
-        set_global_var(5, global_var(5) + 1);
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(4) > (3 * global_var(5))) or (global_var(2) == 1)) and (global_var(1) == 0)) then begin
-            set_global_var(2, 1);
-            set_global_var(3, 0);
+        set_global_var(GVAR_GOOD_MONSTER, global_var(GVAR_GOOD_MONSTER) + 1);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(4) > (3 * global_var(GVAR_GOOD_MONSTER))) or (global_var(GVAR_CHAMPION_REPUTATION) == 1)) and (global_var(GVAR_CHILDKILLER_REPUTATION) == 0)) then begin
+            set_global_var(GVAR_CHAMPION_REPUTATION, 1);
+            set_global_var(GVAR_BERSERKER_REPUTATION, 0);
         end
-        if (metarule(51, self_obj) == 2) then begin
+        if (critter_kill_type(self_obj) == KILL_TYPE_children_kills) then begin
             if (source_obj == dude_obj) then begin
-                set_global_var(2, 0);
-                set_global_var(1, global_var(1) + 1);
-                set_global_var(0, global_var(0) + -15);
+                set_global_var(GVAR_CHAMPION_REPUTATION, 0);
+                set_global_var(GVAR_CHILDKILLER_REPUTATION, global_var(GVAR_CHILDKILLER_REPUTATION) + 1);
+                set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + -15);
                 debug_msg("Player gains " + -15 + " Karma Points.");
-                set_global_var(37, 0);
-                set_global_var(38, 0);
-                set_global_var(39, 0);
-                set_global_var(40, 0);
-                set_global_var(41, 0);
-                set_global_var(42, 0);
-                set_global_var(43, 0);
-                set_global_var(44, 0);
-                set_global_var(45, 0);
-                if (global_var(0) >= 1000) then begin
-                    set_global_var(37, 1);
+                set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+                set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+                set_global_var(GVAR_KARMA_DEFENDER, 0);
+                set_global_var(GVAR_KARMA_WANDERER, 0);
+                set_global_var(GVAR_KARMA_BETRAYER, 0);
+                set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+                set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+                set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+                    set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
                 end
                 else begin
-                    if (global_var(0) >= 750) then begin
-                        set_global_var(38, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
                     end
                     else begin
-                        if (global_var(0) >= 500) then begin
-                            set_global_var(39, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                            set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                         end
                         else begin
-                            if (global_var(0) >= 250) then begin
-                                set_global_var(40, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                                set_global_var(GVAR_KARMA_DEFENDER, 1);
                             end
                             else begin
-                                if (global_var(0) > -250) then begin
-                                    set_global_var(41, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                                    set_global_var(GVAR_KARMA_WANDERER, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -500) then begin
-                                        set_global_var(42, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                        set_global_var(GVAR_KARMA_BETRAYER, 1);
                                     end
                                     else begin
-                                        if (global_var(0) > -750) then begin
-                                            set_global_var(43, 1);
+                                        if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                            set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                         end
                                         else begin
-                                            if (global_var(0) > -1000) then begin
-                                                set_global_var(44, 1);
+                                            if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                                set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                             end
                                             else begin
-                                                set_global_var(45, 1);
+                                                set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                             end
                                         end
                                     end
@@ -907,58 +907,58 @@ begin
                         end
                     end
                 end
-                set_global_var(55, global_var(55) + -8);
+                set_global_var(GVAR_TOWN_REP_NEW_RENO, global_var(GVAR_TOWN_REP_NEW_RENO) + -8);
                 debug_msg("Added " + -8 + " to Town Rep");
             end
         end
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(5) > (2 * global_var(4))) or (global_var(3) == 1))) then begin
-            set_global_var(3, 1);
-            set_global_var(2, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
+            set_global_var(GVAR_BERSERKER_REPUTATION, 1);
+            set_global_var(GVAR_CHAMPION_REPUTATION, 0);
         end
-        set_global_var(0, global_var(0) + -10);
+        set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + -10);
         debug_msg("Player gains " + -10 + " Karma Points.");
-        set_global_var(37, 0);
-        set_global_var(38, 0);
-        set_global_var(39, 0);
-        set_global_var(40, 0);
-        set_global_var(41, 0);
-        set_global_var(42, 0);
-        set_global_var(43, 0);
-        set_global_var(44, 0);
-        set_global_var(45, 0);
-        if (global_var(0) >= 1000) then begin
-            set_global_var(37, 1);
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+        set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+        set_global_var(GVAR_KARMA_DEFENDER, 0);
+        set_global_var(GVAR_KARMA_WANDERER, 0);
+        set_global_var(GVAR_KARMA_BETRAYER, 0);
+        set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+        set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+            set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
         end
         else begin
-            if (global_var(0) >= 750) then begin
-                set_global_var(38, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
             end
             else begin
-                if (global_var(0) >= 500) then begin
-                    set_global_var(39, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                 end
                 else begin
-                    if (global_var(0) >= 250) then begin
-                        set_global_var(40, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                        set_global_var(GVAR_KARMA_DEFENDER, 1);
                     end
                     else begin
-                        if (global_var(0) > -250) then begin
-                            set_global_var(41, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                            set_global_var(GVAR_KARMA_WANDERER, 1);
                         end
                         else begin
-                            if (global_var(0) > -500) then begin
-                                set_global_var(42, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                set_global_var(GVAR_KARMA_BETRAYER, 1);
                             end
                             else begin
-                                if (global_var(0) > -750) then begin
-                                    set_global_var(43, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -1000) then begin
-                                        set_global_var(44, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                     end
                                     else begin
-                                        set_global_var(45, 1);
+                                        set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                     end
                                 end
                             end
@@ -967,7 +967,7 @@ begin
                 end
             end
         end
-        set_global_var(55, global_var(55) + -5);
+        set_global_var(GVAR_TOWN_REP_NEW_RENO, global_var(GVAR_TOWN_REP_NEW_RENO) + -5);
         debug_msg("Added " + -5 + " to Town Rep");
     end
 end
@@ -975,17 +975,17 @@ end
 procedure look_at_p_proc
 begin
     script_overrides;
-    display_msg(message_str(447, 100));
+    display_mstr(100);
 end
 
 procedure description_p_proc
 begin
     script_overrides;
-    if (global_var(494) == 8) then begin
-        display_msg(message_str(447, random(150, 152)));
+    if (global_var(GVAR_FALLOUT_2) == 8) then begin
+        display_mstr(random(150, 152));
     end
     else begin
-        display_msg(message_str(447, random(200, 201)));
+        display_mstr(random(200, 201));
     end
     set_local_var(4, local_var(4) bwor 4);
 end
@@ -1004,35 +1004,35 @@ end
 
 procedure map_enter_p_proc
 begin
-    if (metarule(22, 0) == 0) then begin
-        if (metarule(14, 0)) then begin
+    if (is_loading_game == 0) then begin
+        if (map_first_run) then begin
             set_local_var(5, tile_num(self_obj));
-            critter_add_trait(self_obj, 1, 6, 66);
-            critter_add_trait(self_obj, 1, 5, 15);
+            critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 66);
+            critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_CHILD);
         end
-        if ((global_var(285) bwand 1024) != 0) then begin
+        if ((global_var(GVAR_NEW_RENO_WRIGHT_FLAGS) bwand 1024) != 0) then begin
             destroy_object(self_obj);
         end
         else begin
-            if ((global_var(285) bwand 4) != 0) then begin
+            if ((global_var(GVAR_NEW_RENO_WRIGHT_FLAGS) bwand 4) != 0) then begin
                 kill_critter(self_obj, random(61, 62));
             end
             else begin
                 if ((800 < game_time_hour) and (game_time_hour <= 2100)) then begin
-                    if (has_trait(1, self_obj, 666) == 0) then begin
+                    if (obj_is_visible_flag(self_obj) == 0) then begin
                         set_obj_visibility(self_obj, 0);
                     end
                     last_timed_obj := self_obj;
                     last_timed_param := last_timed_param;
-                    metarule3(100, self_obj, 1, 0);
+                    rm_fixed_timer_event(self_obj, 1);
                     add_timer_event(self_obj, game_ticks(10) + random(0, game_ticks(30 - 10)), 1);
                     last_timed_obj := self_obj;
                     last_timed_param := last_timed_param;
-                    metarule3(100, self_obj, 21, 0);
+                    rm_fixed_timer_event(self_obj, 21);
                     add_timer_event(self_obj, game_ticks(2) + random(0, game_ticks(5 - 2)), 21);
                 end
                 else begin
-                    if (has_trait(1, self_obj, 666) == 1) then begin
+                    if (obj_is_visible_flag(self_obj) == 1) then begin
                         set_obj_visibility(self_obj, 1);
                     end
                 end
@@ -1044,7 +1044,7 @@ end
 procedure timed_event_p_proc
 begin
     variable LVar0 := 0;
-    if (has_trait(1, self_obj, 666)) then begin
+    if (obj_is_visible_flag(self_obj)) then begin
         if (fixed_param == 1) then begin
             if (combat_is_initialized == 0) then begin
                 if (obj_can_see_obj(self_obj, dude_obj) and ((local_var(4) bwand 2048) != 0)) then begin
@@ -1053,7 +1053,7 @@ begin
                     end
                 end
                 else begin
-                    if (global_var(494) == 8) then begin
+                    if (global_var(GVAR_FALLOUT_2) == 8) then begin
                         call Node002();
                     end
                     else begin
@@ -1063,7 +1063,7 @@ begin
             end
             last_timed_obj := self_obj;
             last_timed_param := last_timed_param;
-            metarule3(100, self_obj, 1, 0);
+            rm_fixed_timer_event(self_obj, 1);
             add_timer_event(self_obj, game_ticks(10) + random(0, game_ticks(30 - 10)), 1);
         end
         else begin
@@ -1074,7 +1074,7 @@ begin
                             if (tile_distance_objs(new_reno_wright_teen_obj, self_obj) <= 15) then begin
                                 last_timed_obj := new_reno_wright_teen_obj;
                                 last_timed_param := last_timed_param;
-                                metarule3(100, new_reno_wright_teen_obj, 19, 0);
+                                rm_fixed_timer_event(new_reno_wright_teen_obj, 19);
                                 add_timer_event(new_reno_wright_teen_obj, 0, 19);
                             end
                         end
@@ -1102,7 +1102,7 @@ begin
                             end
                             else begin
                                 if (tile_distance_objs(self_obj, wright_child_playing_obj) > 1) then begin
-                                    animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(wright_child_playing_obj), (has_trait(1, wright_child_playing_obj, 10) + 3) % 6, 1), 1);
+                                    animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(wright_child_playing_obj), (obj_get_rot(wright_child_playing_obj) + 3) % 6, 1), 1);
                                 end
                             end
                         end
@@ -1110,7 +1110,7 @@ begin
                 end
                 last_timed_obj := self_obj;
                 last_timed_param := last_timed_param;
-                metarule3(100, self_obj, 21, 0);
+                rm_fixed_timer_event(self_obj, 21);
                 add_timer_event(self_obj, game_ticks(2) + random(0, game_ticks(5 - 2)), 21);
             end
         end
@@ -1130,34 +1130,34 @@ end
 
 procedure Node002
 begin
-    float_msg(self_obj, message_str(447, random(210, 228)), 8);
+    float_msg(self_obj, mstr(random(210, 228)), FLOAT_MSG_YELLOW);
 end
 
 procedure Node003
 begin
-    float_msg(self_obj, message_str(447, random(240, 255)), 8);
+    float_msg(self_obj, mstr(random(240, 255)), FLOAT_MSG_YELLOW);
 end
 
 procedure Node004
 begin
     floater_patch_msg := random(270, 282);
     if ((floater_patch_msg == 277) or (floater_patch_msg == 278) or (floater_patch_msg == 279) or (floater_patch_msg == 282)) then begin
-        float_msg(self_obj, message_str(447, floater_patch_msg) + obj_name(dude_obj) + message_str(447, floater_patch_msg + 5000), 8);
+        float_msg(self_obj, mstr(floater_patch_msg) + dude_name + mstr(floater_patch_msg + 5000), FLOAT_MSG_YELLOW);
     end
     else begin
-        float_msg(self_obj, message_str(447, floater_patch_msg), 8);
+        float_msg(self_obj, mstr(floater_patch_msg), FLOAT_MSG_YELLOW);
     end
 end
 
 procedure Node005
 begin
-    float_msg(self_obj, message_str(447, random(300, 302)), 8);
+    float_msg(self_obj, mstr(random(300, 302)), FLOAT_MSG_YELLOW);
     if ((tile_distance_objs(self_obj, dude_obj) < 8) or obj_can_see_obj(self_obj, dude_obj)) then begin
         if (anim_busy(self_obj) == 0) then begin
-            global_temp := rotation_to_tile(tile_num(dude_obj), tile_num(self_obj));
+            global_temp := rotation_to_tile(dude_tile, tile_num(self_obj));
             animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(self_obj), global_temp, random(3, 10)), 1);
             global_temp := (global_temp + 1) % 6;
-            while ((anim_busy(self_obj) == 0) and (global_temp != rotation_to_tile(tile_num(dude_obj), tile_num(self_obj)))) do begin
+            while ((anim_busy(self_obj) == 0) and (global_temp != rotation_to_tile(dude_tile, tile_num(self_obj)))) do begin
                 debug_msg("flee loop: rot == " + global_temp);
                 animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(self_obj), global_temp, random(3, 10)), 1);
                 global_temp := (global_temp + 1) % 6;
@@ -1168,13 +1168,13 @@ end
 
 procedure Node006
 begin
-    float_msg(self_obj, message_str(447, random(310, 312)), 8);
+    float_msg(self_obj, mstr(random(310, 312)), FLOAT_MSG_YELLOW);
     if ((tile_distance_objs(self_obj, dude_obj) < 8) or obj_can_see_obj(self_obj, dude_obj)) then begin
         if (anim_busy(self_obj) == 0) then begin
-            global_temp := rotation_to_tile(tile_num(dude_obj), tile_num(self_obj));
+            global_temp := rotation_to_tile(dude_tile, tile_num(self_obj));
             animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(self_obj), global_temp, random(3, 10)), 1);
             global_temp := (global_temp + 1) % 6;
-            while ((anim_busy(self_obj) == 0) and (global_temp != rotation_to_tile(tile_num(dude_obj), tile_num(self_obj)))) do begin
+            while ((anim_busy(self_obj) == 0) and (global_temp != rotation_to_tile(dude_tile, tile_num(self_obj)))) do begin
                 debug_msg("flee loop: rot == " + global_temp);
                 animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(self_obj), global_temp, random(3, 10)), 1);
                 global_temp := (global_temp + 1) % 6;
@@ -1185,124 +1185,124 @@ end
 
 procedure Node007
 begin
-    gsay_reply(447, 320);
-    giq_option(-3, 447, message_str(447, 321) + obj_name(dude_obj) + message_str(447, 5321), Node008, 50);
-    giq_option(-3, 447, 322, Node999, 50);
-    giq_option(4, 447, 323, Node017, 51);
-    giq_option(4, 447, 324, Node020, 50);
-    giq_option(4, 447, 325, Node032, 50);
-    giq_option(4, 447, 326, Node999, 50);
+    Reply(320);
+    giq_option(-3, NAME, mstr(321) + dude_name + mstr(5321), Node008, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 322, Node999, NEUTRAL_REACTION);
+    giq_option(4, NAME, 323, Node017, BAD_REACTION);
+    giq_option(4, NAME, 324, Node020, NEUTRAL_REACTION);
+    giq_option(4, NAME, 325, Node032, NEUTRAL_REACTION);
+    giq_option(4, NAME, 326, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node008
 begin
-    gsay_reply(447, 330);
-    giq_option(-3, 447, 331, Node009, 50);
-    giq_option(-3, 447, obj_name(dude_obj) + message_str(447, 332), Node009, 50);
-    giq_option(-3, 447, 333, Node010, 50);
-    giq_option(-3, 447, message_str(447, 334) + obj_name(dude_obj) + message_str(447, 5334), Node010, 51);
-    giq_option(-3, 447, 335, Node999, 50);
+    Reply(330);
+    giq_option(-3, NAME, 331, Node009, NEUTRAL_REACTION);
+    giq_option(-3, NAME, dude_name + mstr(332), Node009, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 333, Node010, NEUTRAL_REACTION);
+    giq_option(-3, NAME, mstr(334) + dude_name + mstr(5334), Node010, BAD_REACTION);
+    giq_option(-3, NAME, 335, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node009
 begin
-    gsay_reply(447, 340);
-    giq_option(-3, 447, 341, Node990, 50);
-    giq_option(-3, 447, 342, Node999, 50);
-    giq_option(4, 447, 343, Node999, 50);
+    Reply(340);
+    giq_option(-3, NAME, 341, Node990, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 342, Node999, NEUTRAL_REACTION);
+    giq_option(4, NAME, 343, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node010
 begin
-    gsay_reply(447, 350);
-    giq_option(-3, 447, 351, Node011, 51);
-    giq_option(-3, 447, 352, Node013, 51);
-    giq_option(-3, 447, message_str(447, 353) + obj_name(dude_obj) + message_str(447, 5353), Node013, 51);
-    giq_option(-3, 447, message_str(447, 354) + obj_name(dude_obj) + message_str(447, 5354) + obj_name(dude_obj) + message_str(447, 6354) + obj_name(dude_obj) + message_str(447, 7354), Node015, 51);
-    if (global_var(1) >= 2) then begin
-        giq_option(-3, 447, obj_name(dude_obj) + message_str(447, 355), Node019, 51);
+    Reply(350);
+    giq_option(-3, NAME, 351, Node011, BAD_REACTION);
+    giq_option(-3, NAME, 352, Node013, BAD_REACTION);
+    giq_option(-3, NAME, mstr(353) + dude_name + mstr(5353), Node013, BAD_REACTION);
+    giq_option(-3, NAME, mstr(354) + dude_name + mstr(5354) + dude_name + mstr(6354) + dude_name + mstr(7354), Node015, BAD_REACTION);
+    if (global_var(GVAR_CHILDKILLER_REPUTATION) >= 2) then begin
+        giq_option(-3, NAME, dude_name + mstr(355), Node019, BAD_REACTION);
     end
-    giq_option(-3, 447, 356, Node999, 50);
+    giq_option(-3, NAME, 356, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node011
 begin
-    gsay_reply(447, 360);
-    giq_option(-3, 447, 351, Node012, 51);
-    giq_option(-3, 447, 362, Node014, 50);
-    giq_option(-3, 447, 363, Node014, 50);
-    giq_option(-3, 447, 364, Node999, 50);
+    Reply(360);
+    giq_option(-3, NAME, 351, Node012, BAD_REACTION);
+    giq_option(-3, NAME, 362, Node014, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 363, Node014, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 364, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node012
 begin
-    gsay_reply(447, message_str(447, 370 + ((800 < game_time_hour) and (game_time_hour <= 2100))) + message_str(447, 372));
-    giq_option(-3, 447, 373, Node036, 50);
-    giq_option(-3, 447, 374, Node999, 50);
+    Reply(mstr(370 + ((800 < game_time_hour) and (game_time_hour <= 2100))) + mstr(372));
+    giq_option(-3, NAME, 373, Node036, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 374, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node013
 begin
-    gsay_reply(447, 380);
-    giq_option(-3, 447, 381, Node019, 51);
-    giq_option(-3, 447, message_str(447, 382) + obj_name(dude_obj) + message_str(447, 5382) + obj_name(dude_obj) + message_str(447, 6382), Node019, 51);
-    giq_option(-3, 447, 383, Node999, 50);
-    giq_option(4, 447, 384, Node999, 51);
-    giq_option(4, 447, 343, Node999, 50);
+    Reply(380);
+    giq_option(-3, NAME, 381, Node019, BAD_REACTION);
+    giq_option(-3, NAME, mstr(382) + dude_name + mstr(5382) + dude_name + mstr(6382), Node019, BAD_REACTION);
+    giq_option(-3, NAME, 383, Node999, NEUTRAL_REACTION);
+    giq_option(4, NAME, 384, Node999, BAD_REACTION);
+    giq_option(4, NAME, 343, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node014
 begin
-    gsay_reply(447, 390);
-    giq_option(-3, 447, 391, Node016, 50);
-    giq_option(-3, 447, 356, Node999, 50);
+    Reply(390);
+    giq_option(-3, NAME, 391, Node016, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 356, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node015
 begin
-    gsay_reply(447, message_str(447, 400 + ((800 < game_time_hour) and (game_time_hour <= 2100))) + message_str(447, 412));
-    giq_option(-3, 447, 403, Node013, 51);
-    giq_option(-3, 447, 404, Node036, 50);
-    giq_option(-3, 447, 405, Node999, 50);
+    Reply(mstr(400 + ((800 < game_time_hour) and (game_time_hour <= 2100))) + mstr(412));
+    giq_option(-3, NAME, 403, Node013, BAD_REACTION);
+    giq_option(-3, NAME, 404, Node036, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 405, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node016
 begin
-    gsay_reply(447, message_str(447, 410 + ((800 < game_time_hour) and (game_time_hour <= 2100))) + message_str(447, 412));
-    giq_option(-3, 447, 404, Node036, 50);
-    giq_option(-3, 447, 405, Node999, 50);
+    Reply(mstr(410 + ((800 < game_time_hour) and (game_time_hour <= 2100))) + mstr(412));
+    giq_option(-3, NAME, 404, Node036, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 405, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node017
 begin
-    gsay_reply(447, 420);
-    if (global_var(0) >= 0) then begin
-        giq_option(4, 447, 421, Node018, 50);
+    Reply(420);
+    if (global_var(GVAR_PLAYER_REPUTATION) >= 0) then begin
+        giq_option(4, NAME, 421, Node018, NEUTRAL_REACTION);
     end
-    giq_option(4, 447, 422, Node019, 51);
-    giq_option(4, 447, 423, Node999, 50);
+    giq_option(4, NAME, 422, Node019, BAD_REACTION);
+    giq_option(4, NAME, 423, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node018
 begin
-    gsay_reply(447, 430);
-    if (get_critter_stat(dude_obj, 3) > 5) then begin
-        giq_option(4, 447, 432, Node020, 50);
+    Reply(430);
+    if (dude_charisma > 5) then begin
+        giq_option(4, NAME, 432, Node020, NEUTRAL_REACTION);
     end
     else begin
-        giq_option(4, 447, 324, Node020, 50);
+        giq_option(4, NAME, 324, Node020, NEUTRAL_REACTION);
     end
-    giq_option(4, 447, 433, Node032, 50);
-    giq_option(4, 447, 434, Node999, 50);
+    giq_option(4, NAME, 433, Node032, NEUTRAL_REACTION);
+    giq_option(4, NAME, 434, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node019
 begin
     set_local_var(4, local_var(4) bwor 2048);
     set_local_var(4, local_var(4) bwor 512);
-    gsay_reply(447, 440 + (get_critter_stat(dude_obj, 4) > 3));
-    giq_option(-3, 447, 442, Node999, 50);
-    giq_option(4, 447, 443, Node999, 50);
+    Reply(440 + (dude_iq > 3));
+    giq_option(-3, NAME, 442, Node999, NEUTRAL_REACTION);
+    giq_option(4, NAME, 443, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node020
@@ -1349,151 +1349,151 @@ begin
         end
     end
     if (the_pistol != 0) then begin
-        gsay_reply(447, message_str(447, 450) + message_str(447, 451));
-        giq_option(4, 447, 452, Node021, 51);
-        giq_option(4, 447, 453, Node021, 51);
-        giq_option(4, 447, 454, Node028, 49);
-        giq_option(4, 447, 456, Node021, 50);
-        giq_option(4, 447, 458, Node021, 50);
+        Reply(mstr(450) + mstr(451));
+        giq_option(4, NAME, 452, Node021, BAD_REACTION);
+        giq_option(4, NAME, 453, Node021, BAD_REACTION);
+        giq_option(4, NAME, 454, Node028, GOOD_REACTION);
+        giq_option(4, NAME, 456, Node021, NEUTRAL_REACTION);
+        giq_option(4, NAME, 458, Node021, NEUTRAL_REACTION);
     end
     else begin
-        gsay_reply(447, 450);
-        giq_option(4, 447, 456, Node022, 50);
-        giq_option(4, 447, 457, Node999, 50);
+        Reply(450);
+        giq_option(4, NAME, 456, Node022, NEUTRAL_REACTION);
+        giq_option(4, NAME, 457, Node999, NEUTRAL_REACTION);
     end
 end
 
 procedure Node021
 begin
-    gsay_reply(447, 470);
-    if (global_var(0) >= 0) then begin
-        giq_option(4, 447, 471, Node021, 51);
+    Reply(470);
+    if (global_var(GVAR_PLAYER_REPUTATION) >= 0) then begin
+        giq_option(4, NAME, 471, Node021, BAD_REACTION);
     end
     else begin
-        giq_option(4, 447, 473, Node013, 51);
+        giq_option(4, NAME, 473, Node013, BAD_REACTION);
     end
-    giq_option(4, 447, 472, Node013, 51);
-    giq_option(4, 447, 474, Node028, 49);
-    giq_option(4, 447, 476, Node999, 50);
+    giq_option(4, NAME, 472, Node013, BAD_REACTION);
+    giq_option(4, NAME, 474, Node028, GOOD_REACTION);
+    giq_option(4, NAME, 476, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node022
 begin
-    gsay_reply(447, 485);
-    if (get_critter_stat(dude_obj, 3) > 5) then begin
-        giq_option(4, 447, 486, Node987, 49);
+    Reply(485);
+    if (dude_charisma > 5) then begin
+        giq_option(4, NAME, 486, Node987, GOOD_REACTION);
     end
-    giq_option(4, 447, 487, Node986, 50);
-    giq_option(4, 447, 488, Node026, 51);
-    giq_option(4, 447, 489, Node999, 51);
-    giq_option(4, 447, 490, Node999, 50);
+    giq_option(4, NAME, 487, Node986, NEUTRAL_REACTION);
+    giq_option(4, NAME, 488, Node026, BAD_REACTION);
+    giq_option(4, NAME, 489, Node999, BAD_REACTION);
+    giq_option(4, NAME, 490, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node023
 begin
-    gsay_reply(447, 500);
-    giq_option(4, 447, 501, Node035, 50);
-    giq_option(4, 447, 502, Node035, 50);
-    giq_option(4, 447, 503, Node999, 50);
+    Reply(500);
+    giq_option(4, NAME, 501, Node035, NEUTRAL_REACTION);
+    giq_option(4, NAME, 502, Node035, NEUTRAL_REACTION);
+    giq_option(4, NAME, 503, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node024
 begin
-    gsay_reply(447, message_str(447, 510 + ((800 < game_time_hour) and (game_time_hour <= 2100))) + message_str(447, 412));
-    if (((global_var(495) bwand 65536) != 0) and (((global_var(314) bwand 32768) != 0) or (global_var(494) == 8))) then begin
-        giq_option(4, 447, 515, Node025, 50);
+    Reply(mstr(510 + ((800 < game_time_hour) and (game_time_hour <= 2100))) + mstr(412));
+    if (((global_var(GVAR_NEW_RENO_FLAG_3) bwand 65536) != 0) and (((global_var(GVAR_NEW_RENO_FLAG_1) bwand 32768) != 0) or (global_var(GVAR_FALLOUT_2) == 8))) then begin
+        giq_option(4, NAME, 515, Node025, NEUTRAL_REACTION);
     end
     else begin
-        if ((((global_var(314) bwand 32768) != 0) or (global_var(494) == 8)) == 0) then begin
-            giq_option(1, 447, 513 + (get_critter_stat(dude_obj, 4) < 9), Node036, 50);
+        if ((((global_var(GVAR_NEW_RENO_FLAG_1) bwand 32768) != 0) or (global_var(GVAR_FALLOUT_2) == 8)) == 0) then begin
+            giq_option(1, NAME, 513 + (dude_iq < 9), Node036, NEUTRAL_REACTION);
         end
     end
-    giq_option(4, 447, 516, Node034, 50);
-    giq_option(4, 447, 326, Node999, 50);
+    giq_option(4, NAME, 516, Node034, NEUTRAL_REACTION);
+    giq_option(4, NAME, 326, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node025
 begin
-    gsay_reply(447, 530);
-    if (get_critter_stat(dude_obj, 4) < 8) then begin
-        giq_option(4, 447, 531, Node034, 50);
+    Reply(530);
+    if (dude_iq < 8) then begin
+        giq_option(4, NAME, 531, Node034, NEUTRAL_REACTION);
     end
     else begin
-        giq_option(4, 447, 532, Node034, 50);
+        giq_option(4, NAME, 532, Node034, NEUTRAL_REACTION);
     end
-    giq_option(4, 447, 533, Node034, 51);
+    giq_option(4, NAME, 533, Node034, BAD_REACTION);
 end
 
 procedure Node026
 begin
-    gsay_reply(447, 540);
-    giq_option(4, 447, 541, Node035, 50);
-    giq_option(4, 447, 542, Node035, 50);
-    giq_option(4, 447, 543, Node999, 50);
+    Reply(540);
+    giq_option(4, NAME, 541, Node035, NEUTRAL_REACTION);
+    giq_option(4, NAME, 542, Node035, NEUTRAL_REACTION);
+    giq_option(4, NAME, 543, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node027
 begin
-    gsay_reply(447, message_str(447, 550 + ((800 < game_time_hour) and (game_time_hour <= 2100))) + message_str(447, 552));
-    if (((global_var(495) bwand 65536) != 0) and (((global_var(314) bwand 32768) != 0) or (global_var(494) == 8))) then begin
-        giq_option(4, 447, 515, Node025, 50);
+    Reply(mstr(550 + ((800 < game_time_hour) and (game_time_hour <= 2100))) + mstr(552));
+    if (((global_var(GVAR_NEW_RENO_FLAG_3) bwand 65536) != 0) and (((global_var(GVAR_NEW_RENO_FLAG_1) bwand 32768) != 0) or (global_var(GVAR_FALLOUT_2) == 8))) then begin
+        giq_option(4, NAME, 515, Node025, NEUTRAL_REACTION);
     end
     else begin
-        if ((((global_var(314) bwand 32768) != 0) or (global_var(494) == 8)) == 0) then begin
-            giq_option(1, 447, 514, Node036, 50);
+        if ((((global_var(GVAR_NEW_RENO_FLAG_1) bwand 32768) != 0) or (global_var(GVAR_FALLOUT_2) == 8)) == 0) then begin
+            giq_option(1, NAME, 514, Node036, NEUTRAL_REACTION);
         end
     end
-    giq_option(4, 447, 516, Node034, 50);
-    giq_option(4, 447, 556, Node999, 50);
+    giq_option(4, NAME, 516, Node034, NEUTRAL_REACTION);
+    giq_option(4, NAME, 556, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node028
 begin
     if (((local_var(4) bwand 1024) != 0) == 0) then begin
-        set_global_var(0, global_var(0) + -3);
+        set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + -3);
         debug_msg("Player gains " + -3 + " Karma Points.");
-        set_global_var(37, 0);
-        set_global_var(38, 0);
-        set_global_var(39, 0);
-        set_global_var(40, 0);
-        set_global_var(41, 0);
-        set_global_var(42, 0);
-        set_global_var(43, 0);
-        set_global_var(44, 0);
-        set_global_var(45, 0);
-        if (global_var(0) >= 1000) then begin
-            set_global_var(37, 1);
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+        set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+        set_global_var(GVAR_KARMA_DEFENDER, 0);
+        set_global_var(GVAR_KARMA_WANDERER, 0);
+        set_global_var(GVAR_KARMA_BETRAYER, 0);
+        set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+        set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+            set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
         end
         else begin
-            if (global_var(0) >= 750) then begin
-                set_global_var(38, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
             end
             else begin
-                if (global_var(0) >= 500) then begin
-                    set_global_var(39, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                 end
                 else begin
-                    if (global_var(0) >= 250) then begin
-                        set_global_var(40, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                        set_global_var(GVAR_KARMA_DEFENDER, 1);
                     end
                     else begin
-                        if (global_var(0) > -250) then begin
-                            set_global_var(41, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                            set_global_var(GVAR_KARMA_WANDERER, 1);
                         end
                         else begin
-                            if (global_var(0) > -500) then begin
-                                set_global_var(42, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                set_global_var(GVAR_KARMA_BETRAYER, 1);
                             end
                             else begin
-                                if (global_var(0) > -750) then begin
-                                    set_global_var(43, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -1000) then begin
-                                        set_global_var(44, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                     end
                                     else begin
-                                        set_global_var(45, 1);
+                                        set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                     end
                                 end
                             end
@@ -1513,76 +1513,76 @@ begin
         restock_trash := rm_mult_objs_from_inven(dude_obj, restock_obj, restock_amt);
         add_mult_objs_to_inven(self_obj, restock_obj, restock_amt);
     end
-    gsay_reply(447, message_str(447, 560) + message_str(447, 561));
-    giq_option(4, 447, 562, Node029, 51);
-    giq_option(4, 447, 563, Node985, 49);
-    giq_option(4, 447, 564, Node031, 50);
-    giq_option(4, 447, 565, Node999, 50);
+    Reply(mstr(560) + mstr(561));
+    giq_option(4, NAME, 562, Node029, BAD_REACTION);
+    giq_option(4, NAME, 563, Node985, GOOD_REACTION);
+    giq_option(4, NAME, 564, Node031, NEUTRAL_REACTION);
+    giq_option(4, NAME, 565, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node029
 begin
-    gsay_reply(447, 570);
-    giq_option(4, 447, 571, Node030, 51);
-    giq_option(4, 447, 572, Node030, 50);
-    giq_option(4, 447, 573, Node999, 50);
+    Reply(570);
+    giq_option(4, NAME, 571, Node030, BAD_REACTION);
+    giq_option(4, NAME, 572, Node030, NEUTRAL_REACTION);
+    giq_option(4, NAME, 573, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node030
 begin
     set_local_var(4, local_var(4) bwor 2048);
-    gsay_reply(447, 580);
-    giq_option(1, 447, 581, Node999, 51);
-    giq_option(1, 447, 277, Node999, 50);
+    Reply(580);
+    giq_option(1, NAME, 581, Node999, BAD_REACTION);
+    giq_option(1, NAME, 277, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node031
 begin
-    if (((global_var(285) bwand 2048) != 0) == 0) then begin
-        set_global_var(0, global_var(0) + -5);
+    if (((global_var(GVAR_NEW_RENO_WRIGHT_FLAGS) bwand 2048) != 0) == 0) then begin
+        set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + -5);
         debug_msg("Player gains " + -5 + " Karma Points.");
-        set_global_var(37, 0);
-        set_global_var(38, 0);
-        set_global_var(39, 0);
-        set_global_var(40, 0);
-        set_global_var(41, 0);
-        set_global_var(42, 0);
-        set_global_var(43, 0);
-        set_global_var(44, 0);
-        set_global_var(45, 0);
-        if (global_var(0) >= 1000) then begin
-            set_global_var(37, 1);
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+        set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+        set_global_var(GVAR_KARMA_DEFENDER, 0);
+        set_global_var(GVAR_KARMA_WANDERER, 0);
+        set_global_var(GVAR_KARMA_BETRAYER, 0);
+        set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+        set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+            set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
         end
         else begin
-            if (global_var(0) >= 750) then begin
-                set_global_var(38, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
             end
             else begin
-                if (global_var(0) >= 500) then begin
-                    set_global_var(39, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                 end
                 else begin
-                    if (global_var(0) >= 250) then begin
-                        set_global_var(40, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                        set_global_var(GVAR_KARMA_DEFENDER, 1);
                     end
                     else begin
-                        if (global_var(0) > -250) then begin
-                            set_global_var(41, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                            set_global_var(GVAR_KARMA_WANDERER, 1);
                         end
                         else begin
-                            if (global_var(0) > -500) then begin
-                                set_global_var(42, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                set_global_var(GVAR_KARMA_BETRAYER, 1);
                             end
                             else begin
-                                if (global_var(0) > -750) then begin
-                                    set_global_var(43, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -1000) then begin
-                                        set_global_var(44, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                     end
                                     else begin
-                                        set_global_var(45, 1);
+                                        set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                     end
                                 end
                             end
@@ -1592,72 +1592,72 @@ begin
             end
         end
     end
-    set_global_var(285, global_var(285) bwor 2048);
+    set_global_var(GVAR_NEW_RENO_WRIGHT_FLAGS, global_var(GVAR_NEW_RENO_WRIGHT_FLAGS) bwor 2048);
     set_local_var(4, local_var(4) bwor 2048);
     set_local_var(4, local_var(4) bwor 4096);
-    gsay_reply(447, 590);
-    giq_option(1, 447, 591, Node999, 50);
-    giq_option(1, 447, 592, Node999, 50);
+    Reply(590);
+    giq_option(1, NAME, 591, Node999, NEUTRAL_REACTION);
+    giq_option(1, NAME, 592, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node032
 begin
-    gsay_reply(447, 600);
-    giq_option(4, 447, 323, Node017, 51);
-    giq_option(4, 447, 324, Node020, 50);
-    giq_option(4, 447, 603, Node999, 50);
+    Reply(600);
+    giq_option(4, NAME, 323, Node017, BAD_REACTION);
+    giq_option(4, NAME, 324, Node020, NEUTRAL_REACTION);
+    giq_option(4, NAME, 603, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node033
 begin
-    gsay_reply(447, 610);
-    giq_option(1, 447, 343, Node999, 50);
+    Reply(610);
+    giq_option(1, NAME, 343, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node034
 begin
-    gsay_reply(447, 620);
-    giq_option(1, 447, 343, Node999, 50);
+    Reply(620);
+    giq_option(1, NAME, 343, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node035
 begin
-    gsay_reply(447, 630);
-    giq_option(1, 447, 631, Node999, 50);
-    giq_option(4, 447, 632, Node999, 51);
+    Reply(630);
+    giq_option(1, NAME, 631, Node999, NEUTRAL_REACTION);
+    giq_option(4, NAME, 632, Node999, BAD_REACTION);
 end
 
 procedure Node036
 begin
-    gsay_reply(447, 640 + (((game_time_hour <= 600) or (game_time_hour >= 1800)) == 0));
-    giq_option(1, 447, 642, Node984, 50);
+    Reply(640 + (((game_time_hour <= 600) or (game_time_hour >= 1800)) == 0));
+    giq_option(1, NAME, 642, Node984, NEUTRAL_REACTION);
 end
 
 procedure Node037
 begin
     if ((tile_distance_objs(self_obj, dude_obj) < 8) or obj_can_see_obj(self_obj, dude_obj)) then begin
         if (anim_busy(self_obj) == 0) then begin
-            global_temp := rotation_to_tile(tile_num(dude_obj), tile_num(self_obj));
+            global_temp := rotation_to_tile(dude_tile, tile_num(self_obj));
             animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(self_obj), global_temp, random(3, 10)), 1);
             global_temp := (global_temp + 1) % 6;
-            while ((anim_busy(self_obj) == 0) and (global_temp != rotation_to_tile(tile_num(dude_obj), tile_num(self_obj)))) do begin
+            while ((anim_busy(self_obj) == 0) and (global_temp != rotation_to_tile(dude_tile, tile_num(self_obj)))) do begin
                 debug_msg("flee loop: rot == " + global_temp);
                 animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(self_obj), global_temp, random(3, 10)), 1);
                 global_temp := (global_temp + 1) % 6;
             end
         end
     end
-    float_msg(self_obj, message_str(447, 701 - (get_critter_stat(dude_obj, 34) == 0)), 8);
+    float_msg(self_obj, mstr(701 - (dude_gender == 0)), FLOAT_MSG_YELLOW);
 end
 
 procedure Node038
 begin
-    float_msg(self_obj, message_str(447, 711 - (get_critter_stat(dude_obj, 34) == 0)), 8);
+    float_msg(self_obj, mstr(711 - (dude_gender == 0)), FLOAT_MSG_YELLOW);
 end
 
 procedure Node039
 begin
-    float_msg(self_obj, message_str(447, 721 - (get_critter_stat(dude_obj, 34) == 0)), 8);
+    float_msg(self_obj, mstr(721 - (dude_gender == 0)), FLOAT_MSG_YELLOW);
 end
 
 procedure map_exit_p_proc
@@ -1704,10 +1704,10 @@ begin
         end
     end
     if (the_pistol != 0) then begin
-        set_global_var(285, global_var(285) bwor 4);
+        set_global_var(GVAR_NEW_RENO_WRIGHT_FLAGS, global_var(GVAR_NEW_RENO_WRIGHT_FLAGS) bwor 4);
         if ((local_var(4) bwand 4096) != 0) then begin
-            set_global_var(285, global_var(285) bwor 1024);
-            set_global_var(523, the_pistol);
+            set_global_var(GVAR_NEW_RENO_WRIGHT_FLAGS, global_var(GVAR_NEW_RENO_WRIGHT_FLAGS) bwor 1024);
+            set_global_var(GVAR_NEW_RENO_KILL_DADDY_WEAPON, the_pistol);
         end
     end
 end
@@ -1721,7 +1721,7 @@ end
 
 procedure Node987
 begin
-    if ((get_critter_stat(dude_obj, 3) > 6) or is_success(roll_vs_skill(dude_obj, 14, 0)) or has_trait(0, dude_obj, 22) or (local_var(1) > 0)) then begin
+    if ((dude_charisma > 6) or is_success(roll_vs_skill(dude_obj, 14, 0)) or has_trait(TRAIT_PERK, dude_obj, PERK_empathy) or (local_var(1) > 0)) then begin
         call Node024();
     end
     else begin
@@ -1731,7 +1731,7 @@ end
 
 procedure Node986
 begin
-    if ((get_critter_stat(dude_obj, 3) > 8) or is_success(roll_vs_skill(dude_obj, 14, -20)) or has_trait(0, dude_obj, 22) or (local_var(1) > 2)) then begin
+    if ((dude_charisma > 8) or is_success(roll_vs_skill(dude_obj, 14, -20)) or has_trait(TRAIT_PERK, dude_obj, PERK_empathy) or (local_var(1) > 2)) then begin
         call Node027();
     end
     else begin
@@ -1750,4 +1750,3 @@ procedure Node984
 begin
     load_map(147, 0);
 end
-
