diff --git a/Games/Fallout2.Steam/Original/NEWR1.ssl b/Games/Fallout2.Steam/ReDefine/NEWR1.ssl
index e2b9cb4..3bc48f9 100644
--- a/Games/Fallout2.Steam/Original/NEWR1.ssl
+++ b/Games/Fallout2.Steam/ReDefine/NEWR1.ssl
@@ -164,53 +164,53 @@ end
 procedure ChooseFromList
 begin
     variable LVar0 := 0;
-    gsay_reply(353, stored_msg);
-    if (message_str(353, top_msg - 1) != "") then begin
-        giq_option(1, 353, message_str(353, 150), CLICK_PREV, 50);
+    Reply(stored_msg);
+    if (mstr(top_msg - 1) != "") then begin
+        giq_option(1, NAME, mstr(150), CLICK_PREV, NEUTRAL_REACTION);
     end
     if (LVar0 == 0) then begin
-        if (message_str(353, top_msg + 0) != "") then begin
-            giq_option(1, 353, message_str(353, top_msg + 0), CLICK_0, 50);
+        if (mstr(top_msg + 0) != "") then begin
+            giq_option(1, NAME, mstr(top_msg + 0), CLICK_0, NEUTRAL_REACTION);
         end
         else begin
             LVar0 := 1;
         end
     end
     if (LVar0 == 0) then begin
-        if (message_str(353, top_msg + 1) != "") then begin
-            giq_option(1, 353, message_str(353, top_msg + 1), CLICK_1, 50);
+        if (mstr(top_msg + 1) != "") then begin
+            giq_option(1, NAME, mstr(top_msg + 1), CLICK_1, NEUTRAL_REACTION);
         end
         else begin
             LVar0 := 1;
         end
     end
     if (LVar0 == 0) then begin
-        if (message_str(353, top_msg + 2) != "") then begin
-            giq_option(1, 353, message_str(353, top_msg + 2), CLICK_2, 50);
+        if (mstr(top_msg + 2) != "") then begin
+            giq_option(1, NAME, mstr(top_msg + 2), CLICK_2, NEUTRAL_REACTION);
         end
         else begin
             LVar0 := 1;
         end
     end
     if (LVar0 == 0) then begin
-        if (message_str(353, top_msg + 3) != "") then begin
-            giq_option(1, 353, message_str(353, top_msg + 3), CLICK_3, 50);
+        if (mstr(top_msg + 3) != "") then begin
+            giq_option(1, NAME, mstr(top_msg + 3), CLICK_3, NEUTRAL_REACTION);
         end
         else begin
             LVar0 := 1;
         end
     end
     if (LVar0 == 0) then begin
-        if (message_str(353, top_msg + 4) != "") then begin
-            giq_option(1, 353, message_str(353, top_msg + 4), CLICK_4, 50);
+        if (mstr(top_msg + 4) != "") then begin
+            giq_option(1, NAME, mstr(top_msg + 4), CLICK_4, NEUTRAL_REACTION);
         end
         else begin
             LVar0 := 1;
         end
     end
     if (LVar0 == 0) then begin
-        if (message_str(353, top_msg + 5) != "") then begin
-            giq_option(1, 353, message_str(353, 151), CLICK_MORE, 50);
+        if (mstr(top_msg + 5) != "") then begin
+            giq_option(1, NAME, mstr(151), CLICK_MORE, NEUTRAL_REACTION);
         end
     end
 end
@@ -264,25 +264,25 @@ end
 procedure map_enter_p_proc
 begin
     variable LVar0 := 0;
-    if (metarule(22, 0) == 0) then begin
-        if (((global_var(495) bwand 131072) != 0) == 0) then begin
-            set_global_var(495, global_var(495) bwor 131072);
-            if ((global_var(518) == 0) and (global_var(383) == 1)) then begin
-                LVar0 := create_object_sid(16777252, 0, 0, 811);
+    if (is_loading_game == 0) then begin
+        if (((global_var(GVAR_NEW_RENO_FLAG_3) bwand 131072) != 0) == 0) then begin
+            set_global_var(GVAR_NEW_RENO_FLAG_3, global_var(GVAR_NEW_RENO_FLAG_3) bwor 131072);
+            if ((global_var(GVAR_SAVINE_PAYOUT) == 0) and (global_var(GVAR_SAVINELLI_STATUS) == 1)) then begin
+                LVar0 := create_object_sid(16777252, 0, 0, SCRIPT_RCNORD);
                 critter_attempt_placement(LVar0, 27507, 0);
             end
         end
-        if (metarule(14, 0)) then begin
-            display_msg(message_str(353, 100));
+        if (map_first_run) then begin
+            display_mstr(100);
         end
-        if ((global_var(495) bwand 67108864) != 0) then begin
-            set_global_var(495, global_var(495) bwand (-1 - 67108864));
-            if (global_var(348) < 1) then begin
-                set_global_var(348, 1);
+        if ((global_var(GVAR_NEW_RENO_FLAG_3) bwand 67108864) != 0) then begin
+            set_global_var(GVAR_NEW_RENO_FLAG_3, global_var(GVAR_NEW_RENO_FLAG_3) bwand (-1 - 67108864));
+            if (global_var(GVAR_NEW_RENO_STOLEN_CAR) < 1) then begin
+                set_global_var(GVAR_NEW_RENO_STOLEN_CAR, 1);
             end
-            debug_msg("stolen_car state == " + global_var(348));
+            debug_msg("stolen_car state == " + global_var(GVAR_NEW_RENO_STOLEN_CAR));
         end
-        if ((global_var(348) == 1) or (global_var(348) == 2) or ((global_var(495) bwand 67108864) != 0)) then begin
+        if ((global_var(GVAR_NEW_RENO_STOLEN_CAR) == 1) or (global_var(GVAR_NEW_RENO_STOLEN_CAR) == 2) or ((global_var(GVAR_NEW_RENO_FLAG_3) bwand 67108864) != 0)) then begin
             debug_msg("Dest_Car_Loc(" + 25321 + "," + 0 + ")");
             Scenery_Creation_Ptr := tile_contains_pid_obj(25321, 0, 33555441);
             if (Scenery_Creation_Ptr != 0) then begin
@@ -420,106 +420,106 @@ begin
         end
         else begin
             debug_msg("Check_Create_Car(" + 25321 + "," + 0 + ")");
-            if (not(metarule(22, 0))) then begin
-                if (global_var(18) != 0) then begin
-                    debug_msg("car_current_town == " + metarule(30, 0) + " / map_get_load_area == " + metarule3(111, 0, 0, 0));
-                    if ((metarule(30, 0) == metarule3(111, 0, 0, 0)) or ((26 <= metarule(30, 0)) and (metarule(30, 0) <= 29) and (metarule3(111, 0, 0, 0) == -1))) then begin
+            if (not(is_loading_game)) then begin
+                if (global_var(GVAR_PLAYER_GOT_CAR) != 0) then begin
+                    debug_msg("car_current_town == " + car_current_town + " / map_get_load_area == " + map_get_load_area);
+                    if ((car_current_town == map_get_load_area) or ((26 <= car_current_town) and (car_current_town <= 29) and (map_get_load_area == -1))) then begin
                         debug_msg("Place_Car(" + 25321 + "," + 0 + ")");
-                        debug_msg("   GVAR_CAR_PLACED_TILE == " + global_var(633));
+                        debug_msg("   GVAR_CAR_PLACED_TILE == " + global_var(GVAR_CAR_PLACED_TILE));
                         debug_msg("   Car_At_Loc == " + (tile_contains_pid_obj(25321, 0, 33555441) != 0));
-                        if ((global_var(633) <= 0) or (25321 == global_var(633))) then begin
-                            set_global_var(633, 25321);
+                        if ((global_var(GVAR_CAR_PLACED_TILE) <= 0) or (25321 == global_var(GVAR_CAR_PLACED_TILE))) then begin
+                            set_global_var(GVAR_CAR_PLACED_TILE, 25321);
                             if (not(tile_contains_pid_obj(25321, 0, 33555441) != 0)) then begin
-                                Scenery_Creation := create_object_sid(33555441, 25321, 0, 304);
+                                Scenery_Creation := create_object_sid(33555441, 25321, 0, SCRIPT_ZSDRVCAR);
                                 Scenery_Creation_Hex := 25321;
                                 Scenery_Creation_Count := 0;
                                 while (Scenery_Creation_Count < 2) do begin
                                     Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 1, 1);
-                                    Scenery_Creation := create_object_sid(33554499, Scenery_Creation_Hex, 0, -1);
+                                    Scenery_Creation := create_object(33554499, Scenery_Creation_Hex, 0);
                                     Scenery_Creation_Count := Scenery_Creation_Count + 1;
                                 end
                                 Scenery_Creation_Count := 0;
                                 while (Scenery_Creation_Count < 2) do begin
                                     Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 2, 1);
-                                    Scenery_Creation := create_object_sid(33554499, Scenery_Creation_Hex, 0, -1);
+                                    Scenery_Creation := create_object(33554499, Scenery_Creation_Hex, 0);
                                     Scenery_Creation_Count := Scenery_Creation_Count + 1;
                                 end
                                 Scenery_Creation_Count := 0;
                                 while (Scenery_Creation_Count < 2) do begin
                                     Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 3, 1);
-                                    Scenery_Creation := create_object_sid(33554499, Scenery_Creation_Hex, 0, -1);
+                                    Scenery_Creation := create_object(33554499, Scenery_Creation_Hex, 0);
                                     Scenery_Creation_Count := Scenery_Creation_Count + 1;
                                 end
                                 Scenery_Creation_Count := 0;
                                 while (Scenery_Creation_Count < 2) do begin
                                     Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 4, 1);
-                                    Scenery_Creation := create_object_sid(33554499, Scenery_Creation_Hex, 0, -1);
+                                    Scenery_Creation := create_object(33554499, Scenery_Creation_Hex, 0);
                                     Scenery_Creation_Count := Scenery_Creation_Count + 1;
                                 end
                                 Scenery_Creation_Count := 0;
                                 while (Scenery_Creation_Count < 1) do begin
                                     Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 5, 1);
-                                    Scenery_Creation := create_object_sid(33554499, Scenery_Creation_Hex, 0, -1);
+                                    Scenery_Creation := create_object(33554499, Scenery_Creation_Hex, 0);
                                     Scenery_Creation_Count := Scenery_Creation_Count + 1;
                                 end
                                 Scenery_Creation_Count := 0;
                                 while (Scenery_Creation_Count < 1) do begin
                                     Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 4, 1);
-                                    Scenery_Creation := create_object_sid(33554499, Scenery_Creation_Hex, 0, -1);
+                                    Scenery_Creation := create_object(33554499, Scenery_Creation_Hex, 0);
                                     Scenery_Creation_Count := Scenery_Creation_Count + 1;
                                 end
                                 Scenery_Creation_Count := 0;
                                 while (Scenery_Creation_Count < 1) do begin
                                     Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 5, 1);
-                                    Scenery_Creation := create_object_sid(33554499, Scenery_Creation_Hex, 0, -1);
+                                    Scenery_Creation := create_object(33554499, Scenery_Creation_Hex, 0);
                                     Scenery_Creation_Count := Scenery_Creation_Count + 1;
                                 end
                                 Scenery_Creation_Count := 0;
                                 while (Scenery_Creation_Count < 1) do begin
                                     Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 4, 1);
-                                    Scenery_Creation := create_object_sid(33554499, Scenery_Creation_Hex, 0, -1);
+                                    Scenery_Creation := create_object(33554499, Scenery_Creation_Hex, 0);
                                     Scenery_Creation_Count := Scenery_Creation_Count + 1;
                                 end
                                 Scenery_Creation_Count := 0;
                                 while (Scenery_Creation_Count < 1) do begin
                                     Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 5, 1);
-                                    Scenery_Creation := create_object_sid(33554499, Scenery_Creation_Hex, 0, -1);
+                                    Scenery_Creation := create_object(33554499, Scenery_Creation_Hex, 0);
                                     Scenery_Creation_Count := Scenery_Creation_Count + 1;
                                 end
                                 Scenery_Creation_Count := 0;
                                 while (Scenery_Creation_Count < 1) do begin
                                     Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 4, 1);
-                                    Scenery_Creation := create_object_sid(33554499, Scenery_Creation_Hex, 0, -1);
+                                    Scenery_Creation := create_object(33554499, Scenery_Creation_Hex, 0);
                                     Scenery_Creation_Count := Scenery_Creation_Count + 1;
                                 end
                                 Scenery_Creation_Count := 0;
                                 while (Scenery_Creation_Count < 1) do begin
                                     Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 0, 1);
-                                    Scenery_Creation := create_object_sid(33554499, Scenery_Creation_Hex, 0, -1);
+                                    Scenery_Creation := create_object(33554499, Scenery_Creation_Hex, 0);
                                     Scenery_Creation_Count := Scenery_Creation_Count + 1;
                                 end
                                 Scenery_Creation_Count := 0;
                                 while (Scenery_Creation_Count < 1) do begin
                                     Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 5, 1);
-                                    Scenery_Creation := create_object_sid(33554499, Scenery_Creation_Hex, 0, -1);
+                                    Scenery_Creation := create_object(33554499, Scenery_Creation_Hex, 0);
                                     Scenery_Creation_Count := Scenery_Creation_Count + 1;
                                 end
                                 Scenery_Creation_Count := 0;
                                 while (Scenery_Creation_Count < 2) do begin
                                     Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 0, 1);
-                                    Scenery_Creation := create_object_sid(33554499, Scenery_Creation_Hex, 0, -1);
+                                    Scenery_Creation := create_object(33554499, Scenery_Creation_Hex, 0);
                                     Scenery_Creation_Count := Scenery_Creation_Count + 1;
                                 end
                                 Scenery_Creation_Count := 0;
                                 while (Scenery_Creation_Count < 1) do begin
                                     Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 1, 1);
-                                    Scenery_Creation := create_object_sid(33554499, Scenery_Creation_Hex, 0, -1);
+                                    Scenery_Creation := create_object(33554499, Scenery_Creation_Hex, 0);
                                     Scenery_Creation_Count := Scenery_Creation_Count + 1;
                                 end
                                 Scenery_Creation_Count := 0;
                                 while (Scenery_Creation_Count < 1) do begin
                                     Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 2, 1);
-                                    Scenery_Creation := create_object_sid(33554499, Scenery_Creation_Hex, 0, -1);
+                                    Scenery_Creation := create_object(33554499, Scenery_Creation_Hex, 0);
                                     Scenery_Creation_Count := Scenery_Creation_Count + 1;
                                 end
                             end
@@ -532,8 +532,8 @@ begin
                                     Scenery_Creation_Ptr := party_member_obj(455);
                                 end
                                 else begin
-                                    if (global_var(18) == 0) then begin
-                                        Scenery_Creation_Ptr := create_object_sid(455, 0, 0, 920);
+                                    if (global_var(GVAR_PLAYER_GOT_CAR) == 0) then begin
+                                        Scenery_Creation_Ptr := create_object_sid(455, 0, 0, SCRIPT_ZICRTRNK);
                                     end
                                 end
                                 if (Scenery_Creation_Ptr != 0) then begin
@@ -542,13 +542,13 @@ begin
                                     Scenery_Creation_Count := 0;
                                     while (Scenery_Creation_Count < 1) do begin
                                         Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 1, 1);
-                                        Scenery_Creation := create_object_sid(33554499, Scenery_Creation_Hex, 0, -1);
+                                        Scenery_Creation := create_object(33554499, Scenery_Creation_Hex, 0);
                                         Scenery_Creation_Count := Scenery_Creation_Count + 1;
                                     end
                                     Scenery_Creation_Count := 0;
                                     while (Scenery_Creation_Count < 1) do begin
                                         Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 2, 1);
-                                        Scenery_Creation := create_object_sid(33554499, Scenery_Creation_Hex, 0, -1);
+                                        Scenery_Creation := create_object(33554499, Scenery_Creation_Hex, 0);
                                         Scenery_Creation_Count := Scenery_Creation_Count + 1;
                                     end
                                 end
@@ -706,13 +706,13 @@ begin
             end
         end
     end
-    set_global_var(46, 33107);
-    set_global_var(67, 6105);
-    if (elevation(dude_obj) == 1) then begin
+    set_global_var(GVAR_MAP_EXIT_TILE, 33107);
+    set_global_var(GVAR_MAP_NEXT_TILE, 6105);
+    if (dude_elevation == 1) then begin
         set_light_level(50);
     end
     else begin
-        if (elevation(dude_obj) == 2) then begin
+        if (dude_elevation == 2) then begin
             set_light_level(60);
         end
         else begin
@@ -802,11 +802,11 @@ end
 
 procedure map_update_p_proc
 begin
-    if (elevation(dude_obj) == 1) then begin
+    if (dude_elevation == 1) then begin
         set_light_level(50);
     end
     else begin
-        if (elevation(dude_obj) == 2) then begin
+        if (dude_elevation == 2) then begin
             set_light_level(60);
         end
         else begin
@@ -889,4 +889,3 @@ begin
         end
     end
 end
-
