diff --git a/Games/Fallout2.Steam/Original/NEWR2.ssl b/Games/Fallout2.Steam/ReDefine/NEWR2.ssl
index 8917b14..22c82ab 100644
--- a/Games/Fallout2.Steam/Original/NEWR2.ssl
+++ b/Games/Fallout2.Steam/ReDefine/NEWR2.ssl
@@ -187,53 +187,53 @@ end
 procedure ChooseFromList
 begin
     variable LVar0 := 0;
-    gsay_reply(354, stored_msg);
-    if (message_str(353, top_msg - 1) != "") then begin
-        giq_option(1, 354, message_str(353, 150), CLICK_PREV, 50);
+    Reply(stored_msg);
+    if (message_str(SCRIPT_NEWR1, top_msg - 1) != "") then begin
+        giq_option(1, NAME, message_str(SCRIPT_NEWR1, 150), CLICK_PREV, NEUTRAL_REACTION);
     end
     if (LVar0 == 0) then begin
-        if (message_str(353, top_msg + 0) != "") then begin
-            giq_option(1, 354, message_str(353, top_msg + 0), CLICK_0, 50);
+        if (message_str(SCRIPT_NEWR1, top_msg + 0) != "") then begin
+            giq_option(1, NAME, message_str(SCRIPT_NEWR1, top_msg + 0), CLICK_0, NEUTRAL_REACTION);
         end
         else begin
             LVar0 := 1;
         end
     end
     if (LVar0 == 0) then begin
-        if (message_str(353, top_msg + 1) != "") then begin
-            giq_option(1, 354, message_str(353, top_msg + 1), CLICK_1, 50);
+        if (message_str(SCRIPT_NEWR1, top_msg + 1) != "") then begin
+            giq_option(1, NAME, message_str(SCRIPT_NEWR1, top_msg + 1), CLICK_1, NEUTRAL_REACTION);
         end
         else begin
             LVar0 := 1;
         end
     end
     if (LVar0 == 0) then begin
-        if (message_str(353, top_msg + 2) != "") then begin
-            giq_option(1, 354, message_str(353, top_msg + 2), CLICK_2, 50);
+        if (message_str(SCRIPT_NEWR1, top_msg + 2) != "") then begin
+            giq_option(1, NAME, message_str(SCRIPT_NEWR1, top_msg + 2), CLICK_2, NEUTRAL_REACTION);
         end
         else begin
             LVar0 := 1;
         end
     end
     if (LVar0 == 0) then begin
-        if (message_str(353, top_msg + 3) != "") then begin
-            giq_option(1, 354, message_str(353, top_msg + 3), CLICK_3, 50);
+        if (message_str(SCRIPT_NEWR1, top_msg + 3) != "") then begin
+            giq_option(1, NAME, message_str(SCRIPT_NEWR1, top_msg + 3), CLICK_3, NEUTRAL_REACTION);
         end
         else begin
             LVar0 := 1;
         end
     end
     if (LVar0 == 0) then begin
-        if (message_str(353, top_msg + 4) != "") then begin
-            giq_option(1, 354, message_str(353, top_msg + 4), CLICK_4, 50);
+        if (message_str(SCRIPT_NEWR1, top_msg + 4) != "") then begin
+            giq_option(1, NAME, message_str(SCRIPT_NEWR1, top_msg + 4), CLICK_4, NEUTRAL_REACTION);
         end
         else begin
             LVar0 := 1;
         end
     end
     if (LVar0 == 0) then begin
-        if (message_str(353, top_msg + 5) != "") then begin
-            giq_option(1, 354, message_str(353, 151), CLICK_MORE, 50);
+        if (message_str(SCRIPT_NEWR1, top_msg + 5) != "") then begin
+            giq_option(1, NAME, message_str(SCRIPT_NEWR1, 151), CLICK_MORE, NEUTRAL_REACTION);
         end
     end
 end
@@ -286,109 +286,109 @@ end
 
 procedure map_enter_p_proc
 begin
-    set_global_var(46, 34103);
-    set_global_var(67, 6106);
+    set_global_var(GVAR_MAP_EXIT_TILE, 34103);
+    set_global_var(GVAR_MAP_NEXT_TILE, 6106);
     debug_msg("Check_Create_Car(" + 9509 + "," + 0 + ")");
-    if (not(metarule(22, 0))) then begin
-        if (global_var(18) != 0) then begin
-            debug_msg("car_current_town == " + metarule(30, 0) + " / map_get_load_area == " + metarule3(111, 0, 0, 0));
-            if ((metarule(30, 0) == metarule3(111, 0, 0, 0)) or ((26 <= metarule(30, 0)) and (metarule(30, 0) <= 29) and (metarule3(111, 0, 0, 0) == -1))) then begin
+    if (not(is_loading_game)) then begin
+        if (global_var(GVAR_PLAYER_GOT_CAR) != 0) then begin
+            debug_msg("car_current_town == " + car_current_town + " / map_get_load_area == " + map_get_load_area);
+            if ((car_current_town == map_get_load_area) or ((26 <= car_current_town) and (car_current_town <= 29) and (map_get_load_area == -1))) then begin
                 debug_msg("Place_Car(" + 9509 + "," + 0 + ")");
-                debug_msg("   GVAR_CAR_PLACED_TILE == " + global_var(633));
+                debug_msg("   GVAR_CAR_PLACED_TILE == " + global_var(GVAR_CAR_PLACED_TILE));
                 debug_msg("   Car_At_Loc == " + (tile_contains_pid_obj(9509, 0, 33555441) != 0));
-                if ((global_var(633) <= 0) or (9509 == global_var(633))) then begin
-                    set_global_var(633, 9509);
+                if ((global_var(GVAR_CAR_PLACED_TILE) <= 0) or (9509 == global_var(GVAR_CAR_PLACED_TILE))) then begin
+                    set_global_var(GVAR_CAR_PLACED_TILE, 9509);
                     if (not(tile_contains_pid_obj(9509, 0, 33555441) != 0)) then begin
-                        Scenery_Creation := create_object_sid(33555441, 9509, 0, 304);
+                        Scenery_Creation := create_object_sid(33555441, 9509, 0, SCRIPT_ZSDRVCAR);
                         Scenery_Creation_Hex := 9509;
                         Scenery_Creation_Count := 0;
                         while (Scenery_Creation_Count < 2) do begin
                             Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 1, 1);
-                            Scenery_Creation := create_object_sid(33554499, Scenery_Creation_Hex, 0, -1);
+                            Scenery_Creation := create_object(33554499, Scenery_Creation_Hex, 0);
                             Scenery_Creation_Count := Scenery_Creation_Count + 1;
                         end
                         Scenery_Creation_Count := 0;
                         while (Scenery_Creation_Count < 2) do begin
                             Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 2, 1);
-                            Scenery_Creation := create_object_sid(33554499, Scenery_Creation_Hex, 0, -1);
+                            Scenery_Creation := create_object(33554499, Scenery_Creation_Hex, 0);
                             Scenery_Creation_Count := Scenery_Creation_Count + 1;
                         end
                         Scenery_Creation_Count := 0;
                         while (Scenery_Creation_Count < 2) do begin
                             Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 3, 1);
-                            Scenery_Creation := create_object_sid(33554499, Scenery_Creation_Hex, 0, -1);
+                            Scenery_Creation := create_object(33554499, Scenery_Creation_Hex, 0);
                             Scenery_Creation_Count := Scenery_Creation_Count + 1;
                         end
                         Scenery_Creation_Count := 0;
                         while (Scenery_Creation_Count < 2) do begin
                             Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 4, 1);
-                            Scenery_Creation := create_object_sid(33554499, Scenery_Creation_Hex, 0, -1);
+                            Scenery_Creation := create_object(33554499, Scenery_Creation_Hex, 0);
                             Scenery_Creation_Count := Scenery_Creation_Count + 1;
                         end
                         Scenery_Creation_Count := 0;
                         while (Scenery_Creation_Count < 1) do begin
                             Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 5, 1);
-                            Scenery_Creation := create_object_sid(33554499, Scenery_Creation_Hex, 0, -1);
+                            Scenery_Creation := create_object(33554499, Scenery_Creation_Hex, 0);
                             Scenery_Creation_Count := Scenery_Creation_Count + 1;
                         end
                         Scenery_Creation_Count := 0;
                         while (Scenery_Creation_Count < 1) do begin
                             Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 4, 1);
-                            Scenery_Creation := create_object_sid(33554499, Scenery_Creation_Hex, 0, -1);
+                            Scenery_Creation := create_object(33554499, Scenery_Creation_Hex, 0);
                             Scenery_Creation_Count := Scenery_Creation_Count + 1;
                         end
                         Scenery_Creation_Count := 0;
                         while (Scenery_Creation_Count < 1) do begin
                             Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 5, 1);
-                            Scenery_Creation := create_object_sid(33554499, Scenery_Creation_Hex, 0, -1);
+                            Scenery_Creation := create_object(33554499, Scenery_Creation_Hex, 0);
                             Scenery_Creation_Count := Scenery_Creation_Count + 1;
                         end
                         Scenery_Creation_Count := 0;
                         while (Scenery_Creation_Count < 1) do begin
                             Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 4, 1);
-                            Scenery_Creation := create_object_sid(33554499, Scenery_Creation_Hex, 0, -1);
+                            Scenery_Creation := create_object(33554499, Scenery_Creation_Hex, 0);
                             Scenery_Creation_Count := Scenery_Creation_Count + 1;
                         end
                         Scenery_Creation_Count := 0;
                         while (Scenery_Creation_Count < 1) do begin
                             Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 5, 1);
-                            Scenery_Creation := create_object_sid(33554499, Scenery_Creation_Hex, 0, -1);
+                            Scenery_Creation := create_object(33554499, Scenery_Creation_Hex, 0);
                             Scenery_Creation_Count := Scenery_Creation_Count + 1;
                         end
                         Scenery_Creation_Count := 0;
                         while (Scenery_Creation_Count < 1) do begin
                             Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 4, 1);
-                            Scenery_Creation := create_object_sid(33554499, Scenery_Creation_Hex, 0, -1);
+                            Scenery_Creation := create_object(33554499, Scenery_Creation_Hex, 0);
                             Scenery_Creation_Count := Scenery_Creation_Count + 1;
                         end
                         Scenery_Creation_Count := 0;
                         while (Scenery_Creation_Count < 1) do begin
                             Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 0, 1);
-                            Scenery_Creation := create_object_sid(33554499, Scenery_Creation_Hex, 0, -1);
+                            Scenery_Creation := create_object(33554499, Scenery_Creation_Hex, 0);
                             Scenery_Creation_Count := Scenery_Creation_Count + 1;
                         end
                         Scenery_Creation_Count := 0;
                         while (Scenery_Creation_Count < 1) do begin
                             Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 5, 1);
-                            Scenery_Creation := create_object_sid(33554499, Scenery_Creation_Hex, 0, -1);
+                            Scenery_Creation := create_object(33554499, Scenery_Creation_Hex, 0);
                             Scenery_Creation_Count := Scenery_Creation_Count + 1;
                         end
                         Scenery_Creation_Count := 0;
                         while (Scenery_Creation_Count < 2) do begin
                             Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 0, 1);
-                            Scenery_Creation := create_object_sid(33554499, Scenery_Creation_Hex, 0, -1);
+                            Scenery_Creation := create_object(33554499, Scenery_Creation_Hex, 0);
                             Scenery_Creation_Count := Scenery_Creation_Count + 1;
                         end
                         Scenery_Creation_Count := 0;
                         while (Scenery_Creation_Count < 1) do begin
                             Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 1, 1);
-                            Scenery_Creation := create_object_sid(33554499, Scenery_Creation_Hex, 0, -1);
+                            Scenery_Creation := create_object(33554499, Scenery_Creation_Hex, 0);
                             Scenery_Creation_Count := Scenery_Creation_Count + 1;
                         end
                         Scenery_Creation_Count := 0;
                         while (Scenery_Creation_Count < 1) do begin
                             Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 2, 1);
-                            Scenery_Creation := create_object_sid(33554499, Scenery_Creation_Hex, 0, -1);
+                            Scenery_Creation := create_object(33554499, Scenery_Creation_Hex, 0);
                             Scenery_Creation_Count := Scenery_Creation_Count + 1;
                         end
                     end
@@ -401,8 +401,8 @@ begin
                             Scenery_Creation_Ptr := party_member_obj(455);
                         end
                         else begin
-                            if (global_var(18) == 0) then begin
-                                Scenery_Creation_Ptr := create_object_sid(455, 0, 0, 920);
+                            if (global_var(GVAR_PLAYER_GOT_CAR) == 0) then begin
+                                Scenery_Creation_Ptr := create_object_sid(455, 0, 0, SCRIPT_ZICRTRNK);
                             end
                         end
                         if (Scenery_Creation_Ptr != 0) then begin
@@ -411,13 +411,13 @@ begin
                             Scenery_Creation_Count := 0;
                             while (Scenery_Creation_Count < 1) do begin
                                 Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 1, 1);
-                                Scenery_Creation := create_object_sid(33554499, Scenery_Creation_Hex, 0, -1);
+                                Scenery_Creation := create_object(33554499, Scenery_Creation_Hex, 0);
                                 Scenery_Creation_Count := Scenery_Creation_Count + 1;
                             end
                             Scenery_Creation_Count := 0;
                             while (Scenery_Creation_Count < 1) do begin
                                 Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 2, 1);
-                                Scenery_Creation := create_object_sid(33554499, Scenery_Creation_Hex, 0, -1);
+                                Scenery_Creation := create_object(33554499, Scenery_Creation_Hex, 0);
                                 Scenery_Creation_Count := Scenery_Creation_Count + 1;
                             end
                         end
@@ -573,7 +573,7 @@ begin
     else begin
         debug_msg("sorry, can't place the car, the game's loading");
     end
-    if (elevation(dude_obj) == 1) then begin
+    if (dude_elevation == 1) then begin
         set_light_level(60);
     end
     else begin
@@ -658,7 +658,7 @@ end
 
 procedure map_update_p_proc
 begin
-    if (elevation(dude_obj) == 1) then begin
+    if (dude_elevation == 1) then begin
         set_light_level(60);
     end
     else begin
@@ -739,15 +739,15 @@ begin
             end
         end
     end
-    if ((global_var(632) bwand 262144) != 0) then begin
-        debug_msg("all_party_hidden: elevation == " + elevation(dude_obj));
-        if (elevation(dude_obj) < 2) then begin
-            set_global_var(632, global_var(632) bwand (-1 - 262144));
+    if ((global_var(GVAR_PARTY_MEMBERS_HIDDEN) bwand 262144) != 0) then begin
+        debug_msg("all_party_hidden: elevation == " + dude_elevation);
+        if (dude_elevation < 2) then begin
+            set_global_var(GVAR_PARTY_MEMBERS_HIDDEN, global_var(GVAR_PARTY_MEMBERS_HIDDEN) bwand (-1 - 262144));
             if (party_member_obj(16777278)) then begin
-                if (has_trait(1, party_member_obj(16777278), 666) == 0) then begin
-                    if ((global_var(632) bwand 1) != 0) then begin
+                if (obj_is_visible_flag(party_member_obj(16777278)) == 0) then begin
+                    if ((global_var(GVAR_PARTY_MEMBERS_HIDDEN) bwand 1) != 0) then begin
                         debug_msg("unhiding party member " + obj_name(party_member_obj(16777278)));
-                        set_global_var(632, global_var(632) bwand (-1 - 1));
+                        set_global_var(GVAR_PARTY_MEMBERS_HIDDEN, global_var(GVAR_PARTY_MEMBERS_HIDDEN) bwand (-1 - 1));
                         set_obj_visibility(party_member_obj(16777278), 0);
                     end
                     else begin
@@ -756,16 +756,16 @@ begin
                 end
                 else begin
                     debug_msg("unhiding party member " + obj_name(party_member_obj(16777278)));
-                    set_global_var(632, global_var(632) bwand (-1 - 1));
+                    set_global_var(GVAR_PARTY_MEMBERS_HIDDEN, global_var(GVAR_PARTY_MEMBERS_HIDDEN) bwand (-1 - 1));
                     set_obj_visibility(party_member_obj(16777278), 0);
                     debug_msg("PARTY.H:::: party member wasn't hidden.. clearing my state");
                 end
             end
             if (party_member_obj(16777376)) then begin
-                if (has_trait(1, party_member_obj(16777376), 666) == 0) then begin
-                    if ((global_var(632) bwand 2) != 0) then begin
+                if (obj_is_visible_flag(party_member_obj(16777376)) == 0) then begin
+                    if ((global_var(GVAR_PARTY_MEMBERS_HIDDEN) bwand 2) != 0) then begin
                         debug_msg("unhiding party member " + obj_name(party_member_obj(16777376)));
-                        set_global_var(632, global_var(632) bwand (-1 - 2));
+                        set_global_var(GVAR_PARTY_MEMBERS_HIDDEN, global_var(GVAR_PARTY_MEMBERS_HIDDEN) bwand (-1 - 2));
                         set_obj_visibility(party_member_obj(16777376), 0);
                     end
                     else begin
@@ -774,16 +774,16 @@ begin
                 end
                 else begin
                     debug_msg("unhiding party member " + obj_name(party_member_obj(16777376)));
-                    set_global_var(632, global_var(632) bwand (-1 - 2));
+                    set_global_var(GVAR_PARTY_MEMBERS_HIDDEN, global_var(GVAR_PARTY_MEMBERS_HIDDEN) bwand (-1 - 2));
                     set_obj_visibility(party_member_obj(16777376), 0);
                     debug_msg("PARTY.H:::: party member wasn't hidden.. clearing my state");
                 end
             end
             if (party_member_obj(16777377)) then begin
-                if (has_trait(1, party_member_obj(16777377), 666) == 0) then begin
-                    if ((global_var(632) bwand 4) != 0) then begin
+                if (obj_is_visible_flag(party_member_obj(16777377)) == 0) then begin
+                    if ((global_var(GVAR_PARTY_MEMBERS_HIDDEN) bwand 4) != 0) then begin
                         debug_msg("unhiding party member " + obj_name(party_member_obj(16777377)));
-                        set_global_var(632, global_var(632) bwand (-1 - 4));
+                        set_global_var(GVAR_PARTY_MEMBERS_HIDDEN, global_var(GVAR_PARTY_MEMBERS_HIDDEN) bwand (-1 - 4));
                         set_obj_visibility(party_member_obj(16777377), 0);
                     end
                     else begin
@@ -792,16 +792,16 @@ begin
                 end
                 else begin
                     debug_msg("unhiding party member " + obj_name(party_member_obj(16777377)));
-                    set_global_var(632, global_var(632) bwand (-1 - 4));
+                    set_global_var(GVAR_PARTY_MEMBERS_HIDDEN, global_var(GVAR_PARTY_MEMBERS_HIDDEN) bwand (-1 - 4));
                     set_obj_visibility(party_member_obj(16777377), 0);
                     debug_msg("PARTY.H:::: party member wasn't hidden.. clearing my state");
                 end
             end
             if (party_member_obj(16777305)) then begin
-                if (has_trait(1, party_member_obj(16777305), 666) == 0) then begin
-                    if ((global_var(632) bwand 8) != 0) then begin
+                if (obj_is_visible_flag(party_member_obj(16777305)) == 0) then begin
+                    if ((global_var(GVAR_PARTY_MEMBERS_HIDDEN) bwand 8) != 0) then begin
                         debug_msg("unhiding party member " + obj_name(party_member_obj(16777305)));
-                        set_global_var(632, global_var(632) bwand (-1 - 8));
+                        set_global_var(GVAR_PARTY_MEMBERS_HIDDEN, global_var(GVAR_PARTY_MEMBERS_HIDDEN) bwand (-1 - 8));
                         set_obj_visibility(party_member_obj(16777305), 0);
                     end
                     else begin
@@ -810,16 +810,16 @@ begin
                 end
                 else begin
                     debug_msg("unhiding party member " + obj_name(party_member_obj(16777305)));
-                    set_global_var(632, global_var(632) bwand (-1 - 8));
+                    set_global_var(GVAR_PARTY_MEMBERS_HIDDEN, global_var(GVAR_PARTY_MEMBERS_HIDDEN) bwand (-1 - 8));
                     set_obj_visibility(party_member_obj(16777305), 0);
                     debug_msg("PARTY.H:::: party member wasn't hidden.. clearing my state");
                 end
             end
             if (party_member_obj(16777313)) then begin
-                if (has_trait(1, party_member_obj(16777313), 666) == 0) then begin
-                    if ((global_var(632) bwand 16) != 0) then begin
+                if (obj_is_visible_flag(party_member_obj(16777313)) == 0) then begin
+                    if ((global_var(GVAR_PARTY_MEMBERS_HIDDEN) bwand 16) != 0) then begin
                         debug_msg("unhiding party member " + obj_name(party_member_obj(16777313)));
-                        set_global_var(632, global_var(632) bwand (-1 - 16));
+                        set_global_var(GVAR_PARTY_MEMBERS_HIDDEN, global_var(GVAR_PARTY_MEMBERS_HIDDEN) bwand (-1 - 16));
                         set_obj_visibility(party_member_obj(16777313), 0);
                     end
                     else begin
@@ -828,16 +828,16 @@ begin
                 end
                 else begin
                     debug_msg("unhiding party member " + obj_name(party_member_obj(16777313)));
-                    set_global_var(632, global_var(632) bwand (-1 - 16));
+                    set_global_var(GVAR_PARTY_MEMBERS_HIDDEN, global_var(GVAR_PARTY_MEMBERS_HIDDEN) bwand (-1 - 16));
                     set_obj_visibility(party_member_obj(16777313), 0);
                     debug_msg("PARTY.H:::: party member wasn't hidden.. clearing my state");
                 end
             end
             if (party_member_obj(16777323)) then begin
-                if (has_trait(1, party_member_obj(16777323), 666) == 0) then begin
-                    if ((global_var(632) bwand 32) != 0) then begin
+                if (obj_is_visible_flag(party_member_obj(16777323)) == 0) then begin
+                    if ((global_var(GVAR_PARTY_MEMBERS_HIDDEN) bwand 32) != 0) then begin
                         debug_msg("unhiding party member " + obj_name(party_member_obj(16777323)));
-                        set_global_var(632, global_var(632) bwand (-1 - 32));
+                        set_global_var(GVAR_PARTY_MEMBERS_HIDDEN, global_var(GVAR_PARTY_MEMBERS_HIDDEN) bwand (-1 - 32));
                         set_obj_visibility(party_member_obj(16777323), 0);
                     end
                     else begin
@@ -846,16 +846,16 @@ begin
                 end
                 else begin
                     debug_msg("unhiding party member " + obj_name(party_member_obj(16777323)));
-                    set_global_var(632, global_var(632) bwand (-1 - 32));
+                    set_global_var(GVAR_PARTY_MEMBERS_HIDDEN, global_var(GVAR_PARTY_MEMBERS_HIDDEN) bwand (-1 - 32));
                     set_obj_visibility(party_member_obj(16777323), 0);
                     debug_msg("PARTY.H:::: party member wasn't hidden.. clearing my state");
                 end
             end
             if (party_member_obj(16777352)) then begin
-                if (has_trait(1, party_member_obj(16777352), 666) == 0) then begin
-                    if ((global_var(632) bwand 64) != 0) then begin
+                if (obj_is_visible_flag(party_member_obj(16777352)) == 0) then begin
+                    if ((global_var(GVAR_PARTY_MEMBERS_HIDDEN) bwand 64) != 0) then begin
                         debug_msg("unhiding party member " + obj_name(party_member_obj(16777352)));
-                        set_global_var(632, global_var(632) bwand (-1 - 64));
+                        set_global_var(GVAR_PARTY_MEMBERS_HIDDEN, global_var(GVAR_PARTY_MEMBERS_HIDDEN) bwand (-1 - 64));
                         set_obj_visibility(party_member_obj(16777352), 0);
                     end
                     else begin
@@ -864,16 +864,16 @@ begin
                 end
                 else begin
                     debug_msg("unhiding party member " + obj_name(party_member_obj(16777352)));
-                    set_global_var(632, global_var(632) bwand (-1 - 64));
+                    set_global_var(GVAR_PARTY_MEMBERS_HIDDEN, global_var(GVAR_PARTY_MEMBERS_HIDDEN) bwand (-1 - 64));
                     set_obj_visibility(party_member_obj(16777352), 0);
                     debug_msg("PARTY.H:::: party member wasn't hidden.. clearing my state");
                 end
             end
             if (party_member_obj(16777378)) then begin
-                if (has_trait(1, party_member_obj(16777378), 666) == 0) then begin
-                    if ((global_var(632) bwand 128) != 0) then begin
+                if (obj_is_visible_flag(party_member_obj(16777378)) == 0) then begin
+                    if ((global_var(GVAR_PARTY_MEMBERS_HIDDEN) bwand 128) != 0) then begin
                         debug_msg("unhiding party member " + obj_name(party_member_obj(16777378)));
-                        set_global_var(632, global_var(632) bwand (-1 - 128));
+                        set_global_var(GVAR_PARTY_MEMBERS_HIDDEN, global_var(GVAR_PARTY_MEMBERS_HIDDEN) bwand (-1 - 128));
                         set_obj_visibility(party_member_obj(16777378), 0);
                     end
                     else begin
@@ -882,16 +882,16 @@ begin
                 end
                 else begin
                     debug_msg("unhiding party member " + obj_name(party_member_obj(16777378)));
-                    set_global_var(632, global_var(632) bwand (-1 - 128));
+                    set_global_var(GVAR_PARTY_MEMBERS_HIDDEN, global_var(GVAR_PARTY_MEMBERS_HIDDEN) bwand (-1 - 128));
                     set_obj_visibility(party_member_obj(16777378), 0);
                     debug_msg("PARTY.H:::: party member wasn't hidden.. clearing my state");
                 end
             end
             if (party_member_obj(16777368)) then begin
-                if (has_trait(1, party_member_obj(16777368), 666) == 0) then begin
-                    if ((global_var(632) bwand 256) != 0) then begin
+                if (obj_is_visible_flag(party_member_obj(16777368)) == 0) then begin
+                    if ((global_var(GVAR_PARTY_MEMBERS_HIDDEN) bwand 256) != 0) then begin
                         debug_msg("unhiding party member " + obj_name(party_member_obj(16777368)));
-                        set_global_var(632, global_var(632) bwand (-1 - 256));
+                        set_global_var(GVAR_PARTY_MEMBERS_HIDDEN, global_var(GVAR_PARTY_MEMBERS_HIDDEN) bwand (-1 - 256));
                         set_obj_visibility(party_member_obj(16777368), 0);
                     end
                     else begin
@@ -900,16 +900,16 @@ begin
                 end
                 else begin
                     debug_msg("unhiding party member " + obj_name(party_member_obj(16777368)));
-                    set_global_var(632, global_var(632) bwand (-1 - 256));
+                    set_global_var(GVAR_PARTY_MEMBERS_HIDDEN, global_var(GVAR_PARTY_MEMBERS_HIDDEN) bwand (-1 - 256));
                     set_obj_visibility(party_member_obj(16777368), 0);
                     debug_msg("PARTY.H:::: party member wasn't hidden.. clearing my state");
                 end
             end
             if (party_member_obj(16777379)) then begin
-                if (has_trait(1, party_member_obj(16777379), 666) == 0) then begin
-                    if ((global_var(632) bwand 512) != 0) then begin
+                if (obj_is_visible_flag(party_member_obj(16777379)) == 0) then begin
+                    if ((global_var(GVAR_PARTY_MEMBERS_HIDDEN) bwand 512) != 0) then begin
                         debug_msg("unhiding party member " + obj_name(party_member_obj(16777379)));
-                        set_global_var(632, global_var(632) bwand (-1 - 512));
+                        set_global_var(GVAR_PARTY_MEMBERS_HIDDEN, global_var(GVAR_PARTY_MEMBERS_HIDDEN) bwand (-1 - 512));
                         set_obj_visibility(party_member_obj(16777379), 0);
                     end
                     else begin
@@ -918,16 +918,16 @@ begin
                 end
                 else begin
                     debug_msg("unhiding party member " + obj_name(party_member_obj(16777379)));
-                    set_global_var(632, global_var(632) bwand (-1 - 512));
+                    set_global_var(GVAR_PARTY_MEMBERS_HIDDEN, global_var(GVAR_PARTY_MEMBERS_HIDDEN) bwand (-1 - 512));
                     set_obj_visibility(party_member_obj(16777379), 0);
                     debug_msg("PARTY.H:::: party member wasn't hidden.. clearing my state");
                 end
             end
             if (party_member_obj(16777380)) then begin
-                if (has_trait(1, party_member_obj(16777380), 666) == 0) then begin
-                    if ((global_var(632) bwand 1024) != 0) then begin
+                if (obj_is_visible_flag(party_member_obj(16777380)) == 0) then begin
+                    if ((global_var(GVAR_PARTY_MEMBERS_HIDDEN) bwand 1024) != 0) then begin
                         debug_msg("unhiding party member " + obj_name(party_member_obj(16777380)));
-                        set_global_var(632, global_var(632) bwand (-1 - 1024));
+                        set_global_var(GVAR_PARTY_MEMBERS_HIDDEN, global_var(GVAR_PARTY_MEMBERS_HIDDEN) bwand (-1 - 1024));
                         set_obj_visibility(party_member_obj(16777380), 0);
                     end
                     else begin
@@ -936,16 +936,16 @@ begin
                 end
                 else begin
                     debug_msg("unhiding party member " + obj_name(party_member_obj(16777380)));
-                    set_global_var(632, global_var(632) bwand (-1 - 1024));
+                    set_global_var(GVAR_PARTY_MEMBERS_HIDDEN, global_var(GVAR_PARTY_MEMBERS_HIDDEN) bwand (-1 - 1024));
                     set_obj_visibility(party_member_obj(16777380), 0);
                     debug_msg("PARTY.H:::: party member wasn't hidden.. clearing my state");
                 end
             end
             if (party_member_obj(16777295)) then begin
-                if (has_trait(1, party_member_obj(16777295), 666) == 0) then begin
-                    if ((global_var(632) bwand 2048) != 0) then begin
+                if (obj_is_visible_flag(party_member_obj(16777295)) == 0) then begin
+                    if ((global_var(GVAR_PARTY_MEMBERS_HIDDEN) bwand 2048) != 0) then begin
                         debug_msg("unhiding party member " + obj_name(party_member_obj(16777295)));
-                        set_global_var(632, global_var(632) bwand (-1 - 2048));
+                        set_global_var(GVAR_PARTY_MEMBERS_HIDDEN, global_var(GVAR_PARTY_MEMBERS_HIDDEN) bwand (-1 - 2048));
                         set_obj_visibility(party_member_obj(16777295), 0);
                     end
                     else begin
@@ -954,16 +954,16 @@ begin
                 end
                 else begin
                     debug_msg("unhiding party member " + obj_name(party_member_obj(16777295)));
-                    set_global_var(632, global_var(632) bwand (-1 - 2048));
+                    set_global_var(GVAR_PARTY_MEMBERS_HIDDEN, global_var(GVAR_PARTY_MEMBERS_HIDDEN) bwand (-1 - 2048));
                     set_obj_visibility(party_member_obj(16777295), 0);
                     debug_msg("PARTY.H:::: party member wasn't hidden.. clearing my state");
                 end
             end
             if (party_member_obj(16777558)) then begin
-                if (has_trait(1, party_member_obj(16777558), 666) == 0) then begin
-                    if ((global_var(632) bwand 4096) != 0) then begin
+                if (obj_is_visible_flag(party_member_obj(16777558)) == 0) then begin
+                    if ((global_var(GVAR_PARTY_MEMBERS_HIDDEN) bwand 4096) != 0) then begin
                         debug_msg("unhiding party member " + obj_name(party_member_obj(16777558)));
-                        set_global_var(632, global_var(632) bwand (-1 - 4096));
+                        set_global_var(GVAR_PARTY_MEMBERS_HIDDEN, global_var(GVAR_PARTY_MEMBERS_HIDDEN) bwand (-1 - 4096));
                         set_obj_visibility(party_member_obj(16777558), 0);
                     end
                     else begin
@@ -972,16 +972,16 @@ begin
                 end
                 else begin
                     debug_msg("unhiding party member " + obj_name(party_member_obj(16777558)));
-                    set_global_var(632, global_var(632) bwand (-1 - 4096));
+                    set_global_var(GVAR_PARTY_MEMBERS_HIDDEN, global_var(GVAR_PARTY_MEMBERS_HIDDEN) bwand (-1 - 4096));
                     set_obj_visibility(party_member_obj(16777558), 0);
                     debug_msg("PARTY.H:::: party member wasn't hidden.. clearing my state");
                 end
             end
             if (party_member_obj(16777600)) then begin
-                if (has_trait(1, party_member_obj(16777600), 666) == 0) then begin
-                    if ((global_var(632) bwand 8192) != 0) then begin
+                if (obj_is_visible_flag(party_member_obj(16777600)) == 0) then begin
+                    if ((global_var(GVAR_PARTY_MEMBERS_HIDDEN) bwand 8192) != 0) then begin
                         debug_msg("unhiding party member " + obj_name(party_member_obj(16777600)));
-                        set_global_var(632, global_var(632) bwand (-1 - 8192));
+                        set_global_var(GVAR_PARTY_MEMBERS_HIDDEN, global_var(GVAR_PARTY_MEMBERS_HIDDEN) bwand (-1 - 8192));
                         set_obj_visibility(party_member_obj(16777600), 0);
                     end
                     else begin
@@ -990,16 +990,16 @@ begin
                 end
                 else begin
                     debug_msg("unhiding party member " + obj_name(party_member_obj(16777600)));
-                    set_global_var(632, global_var(632) bwand (-1 - 8192));
+                    set_global_var(GVAR_PARTY_MEMBERS_HIDDEN, global_var(GVAR_PARTY_MEMBERS_HIDDEN) bwand (-1 - 8192));
                     set_obj_visibility(party_member_obj(16777600), 0);
                     debug_msg("PARTY.H:::: party member wasn't hidden.. clearing my state");
                 end
             end
             if (party_member_obj(16777687)) then begin
-                if (has_trait(1, party_member_obj(16777687), 666) == 0) then begin
-                    if ((global_var(632) bwand 16384) != 0) then begin
+                if (obj_is_visible_flag(party_member_obj(16777687)) == 0) then begin
+                    if ((global_var(GVAR_PARTY_MEMBERS_HIDDEN) bwand 16384) != 0) then begin
                         debug_msg("unhiding party member " + obj_name(party_member_obj(16777687)));
-                        set_global_var(632, global_var(632) bwand (-1 - 16384));
+                        set_global_var(GVAR_PARTY_MEMBERS_HIDDEN, global_var(GVAR_PARTY_MEMBERS_HIDDEN) bwand (-1 - 16384));
                         set_obj_visibility(party_member_obj(16777687), 0);
                     end
                     else begin
@@ -1008,16 +1008,16 @@ begin
                 end
                 else begin
                     debug_msg("unhiding party member " + obj_name(party_member_obj(16777687)));
-                    set_global_var(632, global_var(632) bwand (-1 - 16384));
+                    set_global_var(GVAR_PARTY_MEMBERS_HIDDEN, global_var(GVAR_PARTY_MEMBERS_HIDDEN) bwand (-1 - 16384));
                     set_obj_visibility(party_member_obj(16777687), 0);
                     debug_msg("PARTY.H:::: party member wasn't hidden.. clearing my state");
                 end
             end
             if (party_member_obj(16777597)) then begin
-                if (has_trait(1, party_member_obj(16777597), 666) == 0) then begin
-                    if ((global_var(632) bwand 32768) != 0) then begin
+                if (obj_is_visible_flag(party_member_obj(16777597)) == 0) then begin
+                    if ((global_var(GVAR_PARTY_MEMBERS_HIDDEN) bwand 32768) != 0) then begin
                         debug_msg("unhiding party member " + obj_name(party_member_obj(16777597)));
-                        set_global_var(632, global_var(632) bwand (-1 - 32768));
+                        set_global_var(GVAR_PARTY_MEMBERS_HIDDEN, global_var(GVAR_PARTY_MEMBERS_HIDDEN) bwand (-1 - 32768));
                         set_obj_visibility(party_member_obj(16777597), 0);
                     end
                     else begin
@@ -1026,16 +1026,16 @@ begin
                 end
                 else begin
                     debug_msg("unhiding party member " + obj_name(party_member_obj(16777597)));
-                    set_global_var(632, global_var(632) bwand (-1 - 32768));
+                    set_global_var(GVAR_PARTY_MEMBERS_HIDDEN, global_var(GVAR_PARTY_MEMBERS_HIDDEN) bwand (-1 - 32768));
                     set_obj_visibility(party_member_obj(16777597), 0);
                     debug_msg("PARTY.H:::: party member wasn't hidden.. clearing my state");
                 end
             end
             if (party_member_obj(16777596)) then begin
-                if (has_trait(1, party_member_obj(16777596), 666) == 0) then begin
-                    if ((global_var(632) bwand 65536) != 0) then begin
+                if (obj_is_visible_flag(party_member_obj(16777596)) == 0) then begin
+                    if ((global_var(GVAR_PARTY_MEMBERS_HIDDEN) bwand 65536) != 0) then begin
                         debug_msg("unhiding party member " + obj_name(party_member_obj(16777596)));
-                        set_global_var(632, global_var(632) bwand (-1 - 65536));
+                        set_global_var(GVAR_PARTY_MEMBERS_HIDDEN, global_var(GVAR_PARTY_MEMBERS_HIDDEN) bwand (-1 - 65536));
                         set_obj_visibility(party_member_obj(16777596), 0);
                     end
                     else begin
@@ -1044,16 +1044,16 @@ begin
                 end
                 else begin
                     debug_msg("unhiding party member " + obj_name(party_member_obj(16777596)));
-                    set_global_var(632, global_var(632) bwand (-1 - 65536));
+                    set_global_var(GVAR_PARTY_MEMBERS_HIDDEN, global_var(GVAR_PARTY_MEMBERS_HIDDEN) bwand (-1 - 65536));
                     set_obj_visibility(party_member_obj(16777596), 0);
                     debug_msg("PARTY.H:::: party member wasn't hidden.. clearing my state");
                 end
             end
             if (party_member_obj(16777595)) then begin
-                if (has_trait(1, party_member_obj(16777595), 666) == 0) then begin
-                    if ((global_var(632) bwand 131072) != 0) then begin
+                if (obj_is_visible_flag(party_member_obj(16777595)) == 0) then begin
+                    if ((global_var(GVAR_PARTY_MEMBERS_HIDDEN) bwand 131072) != 0) then begin
                         debug_msg("unhiding party member " + obj_name(party_member_obj(16777595)));
-                        set_global_var(632, global_var(632) bwand (-1 - 131072));
+                        set_global_var(GVAR_PARTY_MEMBERS_HIDDEN, global_var(GVAR_PARTY_MEMBERS_HIDDEN) bwand (-1 - 131072));
                         set_obj_visibility(party_member_obj(16777595), 0);
                     end
                     else begin
@@ -1062,137 +1062,136 @@ begin
                 end
                 else begin
                     debug_msg("unhiding party member " + obj_name(party_member_obj(16777595)));
-                    set_global_var(632, global_var(632) bwand (-1 - 131072));
+                    set_global_var(GVAR_PARTY_MEMBERS_HIDDEN, global_var(GVAR_PARTY_MEMBERS_HIDDEN) bwand (-1 - 131072));
                     set_obj_visibility(party_member_obj(16777595), 0);
                     debug_msg("PARTY.H:::: party member wasn't hidden.. clearing my state");
                 end
             end
             if (party_member_obj(16777278) != 0) then begin
-                if (has_trait(1, party_member_obj(16777278), 666)) then begin
-                    critter_attempt_placement(party_member_obj(16777278), tile_num(dude_obj), elevation(dude_obj));
+                if (obj_is_visible_flag(party_member_obj(16777278))) then begin
+                    critter_attempt_placement(party_member_obj(16777278), dude_tile, dude_elevation);
                 end
             end
             if (party_member_obj(16777376) != 0) then begin
-                if (has_trait(1, party_member_obj(16777376), 666)) then begin
-                    critter_attempt_placement(party_member_obj(16777376), tile_num(dude_obj), elevation(dude_obj));
+                if (obj_is_visible_flag(party_member_obj(16777376))) then begin
+                    critter_attempt_placement(party_member_obj(16777376), dude_tile, dude_elevation);
                 end
             end
             if (party_member_obj(16777377) != 0) then begin
-                if (has_trait(1, party_member_obj(16777377), 666)) then begin
-                    critter_attempt_placement(party_member_obj(16777377), tile_num(dude_obj), elevation(dude_obj));
+                if (obj_is_visible_flag(party_member_obj(16777377))) then begin
+                    critter_attempt_placement(party_member_obj(16777377), dude_tile, dude_elevation);
                 end
             end
             if (party_member_obj(16777305) != 0) then begin
-                if (has_trait(1, party_member_obj(16777305), 666)) then begin
-                    critter_attempt_placement(party_member_obj(16777305), tile_num(dude_obj), elevation(dude_obj));
+                if (obj_is_visible_flag(party_member_obj(16777305))) then begin
+                    critter_attempt_placement(party_member_obj(16777305), dude_tile, dude_elevation);
                 end
             end
             if (party_member_obj(16777313) != 0) then begin
-                if (has_trait(1, party_member_obj(16777313), 666)) then begin
-                    critter_attempt_placement(party_member_obj(16777313), tile_num(dude_obj), elevation(dude_obj));
+                if (obj_is_visible_flag(party_member_obj(16777313))) then begin
+                    critter_attempt_placement(party_member_obj(16777313), dude_tile, dude_elevation);
                 end
             end
             if (party_member_obj(16777323) != 0) then begin
-                if (has_trait(1, party_member_obj(16777323), 666)) then begin
-                    critter_attempt_placement(party_member_obj(16777323), tile_num(dude_obj), elevation(dude_obj));
+                if (obj_is_visible_flag(party_member_obj(16777323))) then begin
+                    critter_attempt_placement(party_member_obj(16777323), dude_tile, dude_elevation);
                 end
             end
             if (party_member_obj(16777352) != 0) then begin
-                if (has_trait(1, party_member_obj(16777352), 666)) then begin
-                    critter_attempt_placement(party_member_obj(16777352), tile_num(dude_obj), elevation(dude_obj));
+                if (obj_is_visible_flag(party_member_obj(16777352))) then begin
+                    critter_attempt_placement(party_member_obj(16777352), dude_tile, dude_elevation);
                 end
             end
             if (party_member_obj(16777378) != 0) then begin
-                if (has_trait(1, party_member_obj(16777378), 666)) then begin
-                    critter_attempt_placement(party_member_obj(16777378), tile_num(dude_obj), elevation(dude_obj));
+                if (obj_is_visible_flag(party_member_obj(16777378))) then begin
+                    critter_attempt_placement(party_member_obj(16777378), dude_tile, dude_elevation);
                 end
             end
             if (party_member_obj(16777368) != 0) then begin
-                if (has_trait(1, party_member_obj(16777368), 666)) then begin
-                    critter_attempt_placement(party_member_obj(16777368), tile_num(dude_obj), elevation(dude_obj));
+                if (obj_is_visible_flag(party_member_obj(16777368))) then begin
+                    critter_attempt_placement(party_member_obj(16777368), dude_tile, dude_elevation);
                 end
             end
             if (party_member_obj(16777379) != 0) then begin
-                if (has_trait(1, party_member_obj(16777379), 666)) then begin
-                    critter_attempt_placement(party_member_obj(16777379), tile_num(dude_obj), elevation(dude_obj));
+                if (obj_is_visible_flag(party_member_obj(16777379))) then begin
+                    critter_attempt_placement(party_member_obj(16777379), dude_tile, dude_elevation);
                 end
             end
             if (party_member_obj(16777380) != 0) then begin
-                if (has_trait(1, party_member_obj(16777380), 666)) then begin
-                    critter_attempt_placement(party_member_obj(16777380), tile_num(dude_obj), elevation(dude_obj));
+                if (obj_is_visible_flag(party_member_obj(16777380))) then begin
+                    critter_attempt_placement(party_member_obj(16777380), dude_tile, dude_elevation);
                 end
             end
             if (party_member_obj(16777295) != 0) then begin
-                if (has_trait(1, party_member_obj(16777295), 666)) then begin
-                    critter_attempt_placement(party_member_obj(16777295), tile_num(dude_obj), elevation(dude_obj));
+                if (obj_is_visible_flag(party_member_obj(16777295))) then begin
+                    critter_attempt_placement(party_member_obj(16777295), dude_tile, dude_elevation);
                 end
             end
             if (party_member_obj(455) != 0) then begin
-                if (has_trait(1, party_member_obj(455), 666)) then begin
-                    critter_attempt_placement(party_member_obj(455), tile_num(dude_obj), elevation(dude_obj));
+                if (obj_is_visible_flag(party_member_obj(455))) then begin
+                    critter_attempt_placement(party_member_obj(455), dude_tile, dude_elevation);
                 end
             end
             if (party_member_obj(16777558) != 0) then begin
-                if (has_trait(1, party_member_obj(16777558), 666)) then begin
-                    critter_attempt_placement(party_member_obj(16777558), tile_num(dude_obj), elevation(dude_obj));
+                if (obj_is_visible_flag(party_member_obj(16777558))) then begin
+                    critter_attempt_placement(party_member_obj(16777558), dude_tile, dude_elevation);
                 end
             end
             if (party_member_obj(16777600) != 0) then begin
-                if (has_trait(1, party_member_obj(16777600), 666)) then begin
-                    critter_attempt_placement(party_member_obj(16777600), tile_num(dude_obj), elevation(dude_obj));
+                if (obj_is_visible_flag(party_member_obj(16777600))) then begin
+                    critter_attempt_placement(party_member_obj(16777600), dude_tile, dude_elevation);
                 end
             end
             if (party_member_obj(16777687) != 0) then begin
-                if (has_trait(1, party_member_obj(16777687), 666)) then begin
-                    critter_attempt_placement(party_member_obj(16777687), tile_num(dude_obj), elevation(dude_obj));
+                if (obj_is_visible_flag(party_member_obj(16777687))) then begin
+                    critter_attempt_placement(party_member_obj(16777687), dude_tile, dude_elevation);
                 end
             end
             if (party_member_obj(16777597) != 0) then begin
-                if (has_trait(1, party_member_obj(16777597), 666)) then begin
-                    critter_attempt_placement(party_member_obj(16777597), tile_num(dude_obj), elevation(dude_obj));
+                if (obj_is_visible_flag(party_member_obj(16777597))) then begin
+                    critter_attempt_placement(party_member_obj(16777597), dude_tile, dude_elevation);
                 end
             end
             if (party_member_obj(16777596) != 0) then begin
-                if (has_trait(1, party_member_obj(16777596), 666)) then begin
-                    critter_attempt_placement(party_member_obj(16777596), tile_num(dude_obj), elevation(dude_obj));
+                if (obj_is_visible_flag(party_member_obj(16777596))) then begin
+                    critter_attempt_placement(party_member_obj(16777596), dude_tile, dude_elevation);
                 end
             end
             if (party_member_obj(16777595) != 0) then begin
-                if (has_trait(1, party_member_obj(16777595), 666)) then begin
-                    critter_attempt_placement(party_member_obj(16777595), tile_num(dude_obj), elevation(dude_obj));
+                if (obj_is_visible_flag(party_member_obj(16777595))) then begin
+                    critter_attempt_placement(party_member_obj(16777595), dude_tile, dude_elevation);
                 end
             end
             if (party_member_obj(16777381) != 0) then begin
-                if (has_trait(1, party_member_obj(16777381), 666)) then begin
-                    critter_attempt_placement(party_member_obj(16777381), tile_num(dude_obj), elevation(dude_obj));
+                if (obj_is_visible_flag(party_member_obj(16777381))) then begin
+                    critter_attempt_placement(party_member_obj(16777381), dude_tile, dude_elevation);
                 end
             end
             if (party_member_obj(16777407) != 0) then begin
-                if (has_trait(1, party_member_obj(16777407), 666)) then begin
-                    critter_attempt_placement(party_member_obj(16777407), tile_num(dude_obj), elevation(dude_obj));
+                if (obj_is_visible_flag(party_member_obj(16777407))) then begin
+                    critter_attempt_placement(party_member_obj(16777407), dude_tile, dude_elevation);
                 end
             end
             if (party_member_obj(16777411) != 0) then begin
-                if (has_trait(1, party_member_obj(16777411), 666)) then begin
-                    critter_attempt_placement(party_member_obj(16777411), tile_num(dude_obj), elevation(dude_obj));
+                if (obj_is_visible_flag(party_member_obj(16777411))) then begin
+                    critter_attempt_placement(party_member_obj(16777411), dude_tile, dude_elevation);
                 end
             end
             if (party_member_obj(16777412) != 0) then begin
-                if (has_trait(1, party_member_obj(16777412), 666)) then begin
-                    critter_attempt_placement(party_member_obj(16777412), tile_num(dude_obj), elevation(dude_obj));
+                if (obj_is_visible_flag(party_member_obj(16777412))) then begin
+                    critter_attempt_placement(party_member_obj(16777412), dude_tile, dude_elevation);
                 end
             end
             if (party_member_obj(16777413) != 0) then begin
-                if (has_trait(1, party_member_obj(16777413), 666)) then begin
-                    critter_attempt_placement(party_member_obj(16777413), tile_num(dude_obj), elevation(dude_obj));
+                if (obj_is_visible_flag(party_member_obj(16777413))) then begin
+                    critter_attempt_placement(party_member_obj(16777413), dude_tile, dude_elevation);
                 end
             end
             if (party_member_obj(16777481) != 0) then begin
-                if (has_trait(1, party_member_obj(16777481), 666)) then begin
-                    critter_attempt_placement(party_member_obj(16777481), tile_num(dude_obj), elevation(dude_obj));
+                if (obj_is_visible_flag(party_member_obj(16777481))) then begin
+                    critter_attempt_placement(party_member_obj(16777481), dude_tile, dude_elevation);
                 end
             end
         end
     end
 end
-
