diff --git a/Games/Fallout2.Steam/Original/NEWRST.ssl b/Games/Fallout2.Steam/ReDefine/NEWRST.ssl
index 1b1e0cb..b06b88c 100644
--- a/Games/Fallout2.Steam/Original/NEWRST.ssl
+++ b/Games/Fallout2.Steam/ReDefine/NEWRST.ssl
@@ -158,53 +158,53 @@ end
 procedure ChooseFromList
 begin
     variable LVar0 := 0;
-    gsay_reply(357, stored_msg);
-    if (message_str(353, top_msg - 1) != "") then begin
-        giq_option(1, 357, message_str(353, 150), CLICK_PREV, 50);
+    Reply(stored_msg);
+    if (message_str(SCRIPT_NEWR1, top_msg - 1) != "") then begin
+        giq_option(1, NAME, message_str(SCRIPT_NEWR1, 150), CLICK_PREV, NEUTRAL_REACTION);
     end
     if (LVar0 == 0) then begin
-        if (message_str(353, top_msg + 0) != "") then begin
-            giq_option(1, 357, message_str(353, top_msg + 0), CLICK_0, 50);
+        if (message_str(SCRIPT_NEWR1, top_msg + 0) != "") then begin
+            giq_option(1, NAME, message_str(SCRIPT_NEWR1, top_msg + 0), CLICK_0, NEUTRAL_REACTION);
         end
         else begin
             LVar0 := 1;
         end
     end
     if (LVar0 == 0) then begin
-        if (message_str(353, top_msg + 1) != "") then begin
-            giq_option(1, 357, message_str(353, top_msg + 1), CLICK_1, 50);
+        if (message_str(SCRIPT_NEWR1, top_msg + 1) != "") then begin
+            giq_option(1, NAME, message_str(SCRIPT_NEWR1, top_msg + 1), CLICK_1, NEUTRAL_REACTION);
         end
         else begin
             LVar0 := 1;
         end
     end
     if (LVar0 == 0) then begin
-        if (message_str(353, top_msg + 2) != "") then begin
-            giq_option(1, 357, message_str(353, top_msg + 2), CLICK_2, 50);
+        if (message_str(SCRIPT_NEWR1, top_msg + 2) != "") then begin
+            giq_option(1, NAME, message_str(SCRIPT_NEWR1, top_msg + 2), CLICK_2, NEUTRAL_REACTION);
         end
         else begin
             LVar0 := 1;
         end
     end
     if (LVar0 == 0) then begin
-        if (message_str(353, top_msg + 3) != "") then begin
-            giq_option(1, 357, message_str(353, top_msg + 3), CLICK_3, 50);
+        if (message_str(SCRIPT_NEWR1, top_msg + 3) != "") then begin
+            giq_option(1, NAME, message_str(SCRIPT_NEWR1, top_msg + 3), CLICK_3, NEUTRAL_REACTION);
         end
         else begin
             LVar0 := 1;
         end
     end
     if (LVar0 == 0) then begin
-        if (message_str(353, top_msg + 4) != "") then begin
-            giq_option(1, 357, message_str(353, top_msg + 4), CLICK_4, 50);
+        if (message_str(SCRIPT_NEWR1, top_msg + 4) != "") then begin
+            giq_option(1, NAME, message_str(SCRIPT_NEWR1, top_msg + 4), CLICK_4, NEUTRAL_REACTION);
         end
         else begin
             LVar0 := 1;
         end
     end
     if (LVar0 == 0) then begin
-        if (message_str(353, top_msg + 5) != "") then begin
-            giq_option(1, 357, message_str(353, 151), CLICK_MORE, 50);
+        if (message_str(SCRIPT_NEWR1, top_msg + 5) != "") then begin
+            giq_option(1, NAME, message_str(SCRIPT_NEWR1, 151), CLICK_MORE, NEUTRAL_REACTION);
         end
     end
 end
@@ -257,108 +257,108 @@ end
 
 procedure map_enter_p_proc
 begin
-    if (metarule(22, 0) == 0) then begin
+    if (is_loading_game == 0) then begin
         debug_msg("Check_Create_Car(" + 26506 + "," + 0 + ")");
-        if (not(metarule(22, 0))) then begin
-            if (global_var(18) != 0) then begin
-                debug_msg("car_current_town == " + metarule(30, 0) + " / map_get_load_area == " + metarule3(111, 0, 0, 0));
-                if ((metarule(30, 0) == metarule3(111, 0, 0, 0)) or ((26 <= metarule(30, 0)) and (metarule(30, 0) <= 29) and (metarule3(111, 0, 0, 0) == -1))) then begin
+        if (not(is_loading_game)) then begin
+            if (global_var(GVAR_PLAYER_GOT_CAR) != 0) then begin
+                debug_msg("car_current_town == " + car_current_town + " / map_get_load_area == " + map_get_load_area);
+                if ((car_current_town == map_get_load_area) or ((26 <= car_current_town) and (car_current_town <= 29) and (map_get_load_area == -1))) then begin
                     debug_msg("Place_Car(" + 26506 + "," + 0 + ")");
-                    debug_msg("   GVAR_CAR_PLACED_TILE == " + global_var(633));
+                    debug_msg("   GVAR_CAR_PLACED_TILE == " + global_var(GVAR_CAR_PLACED_TILE));
                     debug_msg("   Car_At_Loc == " + (tile_contains_pid_obj(26506, 0, 33555441) != 0));
-                    if ((global_var(633) <= 0) or (26506 == global_var(633))) then begin
-                        set_global_var(633, 26506);
+                    if ((global_var(GVAR_CAR_PLACED_TILE) <= 0) or (26506 == global_var(GVAR_CAR_PLACED_TILE))) then begin
+                        set_global_var(GVAR_CAR_PLACED_TILE, 26506);
                         if (not(tile_contains_pid_obj(26506, 0, 33555441) != 0)) then begin
-                            Scenery_Creation := create_object_sid(33555441, 26506, 0, 304);
+                            Scenery_Creation := create_object_sid(33555441, 26506, 0, SCRIPT_ZSDRVCAR);
                             Scenery_Creation_Hex := 26506;
                             Scenery_Creation_Count := 0;
                             while (Scenery_Creation_Count < 2) do begin
                                 Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 1, 1);
-                                Scenery_Creation := create_object_sid(33554499, Scenery_Creation_Hex, 0, -1);
+                                Scenery_Creation := create_object(33554499, Scenery_Creation_Hex, 0);
                                 Scenery_Creation_Count := Scenery_Creation_Count + 1;
                             end
                             Scenery_Creation_Count := 0;
                             while (Scenery_Creation_Count < 2) do begin
                                 Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 2, 1);
-                                Scenery_Creation := create_object_sid(33554499, Scenery_Creation_Hex, 0, -1);
+                                Scenery_Creation := create_object(33554499, Scenery_Creation_Hex, 0);
                                 Scenery_Creation_Count := Scenery_Creation_Count + 1;
                             end
                             Scenery_Creation_Count := 0;
                             while (Scenery_Creation_Count < 2) do begin
                                 Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 3, 1);
-                                Scenery_Creation := create_object_sid(33554499, Scenery_Creation_Hex, 0, -1);
+                                Scenery_Creation := create_object(33554499, Scenery_Creation_Hex, 0);
                                 Scenery_Creation_Count := Scenery_Creation_Count + 1;
                             end
                             Scenery_Creation_Count := 0;
                             while (Scenery_Creation_Count < 2) do begin
                                 Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 4, 1);
-                                Scenery_Creation := create_object_sid(33554499, Scenery_Creation_Hex, 0, -1);
+                                Scenery_Creation := create_object(33554499, Scenery_Creation_Hex, 0);
                                 Scenery_Creation_Count := Scenery_Creation_Count + 1;
                             end
                             Scenery_Creation_Count := 0;
                             while (Scenery_Creation_Count < 1) do begin
                                 Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 5, 1);
-                                Scenery_Creation := create_object_sid(33554499, Scenery_Creation_Hex, 0, -1);
+                                Scenery_Creation := create_object(33554499, Scenery_Creation_Hex, 0);
                                 Scenery_Creation_Count := Scenery_Creation_Count + 1;
                             end
                             Scenery_Creation_Count := 0;
                             while (Scenery_Creation_Count < 1) do begin
                                 Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 4, 1);
-                                Scenery_Creation := create_object_sid(33554499, Scenery_Creation_Hex, 0, -1);
+                                Scenery_Creation := create_object(33554499, Scenery_Creation_Hex, 0);
                                 Scenery_Creation_Count := Scenery_Creation_Count + 1;
                             end
                             Scenery_Creation_Count := 0;
                             while (Scenery_Creation_Count < 1) do begin
                                 Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 5, 1);
-                                Scenery_Creation := create_object_sid(33554499, Scenery_Creation_Hex, 0, -1);
+                                Scenery_Creation := create_object(33554499, Scenery_Creation_Hex, 0);
                                 Scenery_Creation_Count := Scenery_Creation_Count + 1;
                             end
                             Scenery_Creation_Count := 0;
                             while (Scenery_Creation_Count < 1) do begin
                                 Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 4, 1);
-                                Scenery_Creation := create_object_sid(33554499, Scenery_Creation_Hex, 0, -1);
+                                Scenery_Creation := create_object(33554499, Scenery_Creation_Hex, 0);
                                 Scenery_Creation_Count := Scenery_Creation_Count + 1;
                             end
                             Scenery_Creation_Count := 0;
                             while (Scenery_Creation_Count < 1) do begin
                                 Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 5, 1);
-                                Scenery_Creation := create_object_sid(33554499, Scenery_Creation_Hex, 0, -1);
+                                Scenery_Creation := create_object(33554499, Scenery_Creation_Hex, 0);
                                 Scenery_Creation_Count := Scenery_Creation_Count + 1;
                             end
                             Scenery_Creation_Count := 0;
                             while (Scenery_Creation_Count < 1) do begin
                                 Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 4, 1);
-                                Scenery_Creation := create_object_sid(33554499, Scenery_Creation_Hex, 0, -1);
+                                Scenery_Creation := create_object(33554499, Scenery_Creation_Hex, 0);
                                 Scenery_Creation_Count := Scenery_Creation_Count + 1;
                             end
                             Scenery_Creation_Count := 0;
                             while (Scenery_Creation_Count < 1) do begin
                                 Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 0, 1);
-                                Scenery_Creation := create_object_sid(33554499, Scenery_Creation_Hex, 0, -1);
+                                Scenery_Creation := create_object(33554499, Scenery_Creation_Hex, 0);
                                 Scenery_Creation_Count := Scenery_Creation_Count + 1;
                             end
                             Scenery_Creation_Count := 0;
                             while (Scenery_Creation_Count < 1) do begin
                                 Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 5, 1);
-                                Scenery_Creation := create_object_sid(33554499, Scenery_Creation_Hex, 0, -1);
+                                Scenery_Creation := create_object(33554499, Scenery_Creation_Hex, 0);
                                 Scenery_Creation_Count := Scenery_Creation_Count + 1;
                             end
                             Scenery_Creation_Count := 0;
                             while (Scenery_Creation_Count < 2) do begin
                                 Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 0, 1);
-                                Scenery_Creation := create_object_sid(33554499, Scenery_Creation_Hex, 0, -1);
+                                Scenery_Creation := create_object(33554499, Scenery_Creation_Hex, 0);
                                 Scenery_Creation_Count := Scenery_Creation_Count + 1;
                             end
                             Scenery_Creation_Count := 0;
                             while (Scenery_Creation_Count < 1) do begin
                                 Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 1, 1);
-                                Scenery_Creation := create_object_sid(33554499, Scenery_Creation_Hex, 0, -1);
+                                Scenery_Creation := create_object(33554499, Scenery_Creation_Hex, 0);
                                 Scenery_Creation_Count := Scenery_Creation_Count + 1;
                             end
                             Scenery_Creation_Count := 0;
                             while (Scenery_Creation_Count < 1) do begin
                                 Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 2, 1);
-                                Scenery_Creation := create_object_sid(33554499, Scenery_Creation_Hex, 0, -1);
+                                Scenery_Creation := create_object(33554499, Scenery_Creation_Hex, 0);
                                 Scenery_Creation_Count := Scenery_Creation_Count + 1;
                             end
                         end
@@ -371,8 +371,8 @@ begin
                                 Scenery_Creation_Ptr := party_member_obj(455);
                             end
                             else begin
-                                if (global_var(18) == 0) then begin
-                                    Scenery_Creation_Ptr := create_object_sid(455, 0, 0, 920);
+                                if (global_var(GVAR_PLAYER_GOT_CAR) == 0) then begin
+                                    Scenery_Creation_Ptr := create_object_sid(455, 0, 0, SCRIPT_ZICRTRNK);
                                 end
                             end
                             if (Scenery_Creation_Ptr != 0) then begin
@@ -381,13 +381,13 @@ begin
                                 Scenery_Creation_Count := 0;
                                 while (Scenery_Creation_Count < 1) do begin
                                     Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 1, 1);
-                                    Scenery_Creation := create_object_sid(33554499, Scenery_Creation_Hex, 0, -1);
+                                    Scenery_Creation := create_object(33554499, Scenery_Creation_Hex, 0);
                                     Scenery_Creation_Count := Scenery_Creation_Count + 1;
                                 end
                                 Scenery_Creation_Count := 0;
                                 while (Scenery_Creation_Count < 1) do begin
                                     Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 2, 1);
-                                    Scenery_Creation := create_object_sid(33554499, Scenery_Creation_Hex, 0, -1);
+                                    Scenery_Creation := create_object(33554499, Scenery_Creation_Hex, 0);
                                     Scenery_Creation_Count := Scenery_Creation_Count + 1;
                                 end
                             end
@@ -543,20 +543,20 @@ begin
         else begin
             debug_msg("sorry, can't place the car, the game's loading");
         end
-        if (metarule(14, 0)) then begin
-            if (metarule(17, 48) == 0) then begin
+        if (map_first_run) then begin
+            if (town_known(48) == 0) then begin
                 debug_msg("  mark_on_map(" + 48 + ")");
                 mark_area_known(0, 48, 1);
             end
-            display_msg(message_str(353, 103));
+            display_msg(message_str(SCRIPT_NEWR1, 103));
         end
         override_map_start(25915 % 200, 25915 / 200, 0, (5 + random(0, 1)) % 6);
-        set_global_var(46, 0);
-        set_global_var(67, 0);
+        set_global_var(GVAR_MAP_EXIT_TILE, 0);
+        set_global_var(GVAR_MAP_NEXT_TILE, 0);
     end
-    set_global_var(46, 29326);
-    set_global_var(67, 0);
-    if (elevation(dude_obj) == 1) then begin
+    set_global_var(GVAR_MAP_EXIT_TILE, 29326);
+    set_global_var(GVAR_MAP_NEXT_TILE, 0);
+    if (dude_elevation == 1) then begin
         set_light_level(50);
     end
     else begin
@@ -641,7 +641,7 @@ end
 
 procedure map_update_p_proc
 begin
-    if (elevation(dude_obj) == 1) then begin
+    if (dude_elevation == 1) then begin
         set_light_level(50);
     end
     else begin
@@ -726,7 +726,6 @@ end
 
 procedure map_exit_p_proc
 begin
-    set_global_var(284, global_var(284) bwand (-1 - 16384));
-    set_global_var(314, global_var(314) bwand (-1 - 2));
+    set_global_var(GVAR_NEW_RENO_MYRON, global_var(GVAR_NEW_RENO_MYRON) bwand (-1 - 16384));
+    set_global_var(GVAR_NEW_RENO_FLAG_1, global_var(GVAR_NEW_RENO_FLAG_1) bwand (-1 - 2));
 end
-
