diff --git a/Games/Fallout2.Steam/Original/NIPRIGRV.ssl b/Games/Fallout2.Steam/ReDefine/NIPRIGRV.ssl
index 7f12716..dd63228 100644
--- a/Games/Fallout2.Steam/Original/NIPRIGRV.ssl
+++ b/Games/Fallout2.Steam/ReDefine/NIPRIGRV.ssl
@@ -76,10 +76,10 @@ begin
         if (local_var(2) != 0) then begin
             if (is_success(roll_vs_skill(source_obj, 11, -20))) then begin
                 if (source_obj == dude_obj) then begin
-                    display_msg(message_str(69, 515));
+                    display_msg(message_str(SCRIPT_ZIGENGRV, 515));
                 end
                 else begin
-                    display_msg(obj_name(source_obj) + message_str(69, 520));
+                    display_msg(obj_name(source_obj) + message_str(SCRIPT_ZIGENGRV, 520));
                 end
             end
             else begin
@@ -100,19 +100,19 @@ begin
     if (((33555669 <= obj_pid(self_obj)) and (obj_pid(self_obj) <= 33555674) or ((33555445 <= obj_pid(self_obj)) and (obj_pid(self_obj) <= 33555448))) == 0) then begin
         if ((local_var(2) != 0) and is_success(roll_vs_skill(source_obj, 11, -20))) then begin
             if (source_obj == dude_obj) then begin
-                display_msg(message_str(69, 515));
+                display_msg(message_str(SCRIPT_ZIGENGRV, 515));
             end
             else begin
-                display_msg(obj_name(source_obj) + message_str(69, 520));
+                display_msg(obj_name(source_obj) + message_str(SCRIPT_ZIGENGRV, 520));
             end
         end
         else begin
-            display_msg(message_str(69, 102 + (local_var(0) == -1)));
+            display_msg(message_str(SCRIPT_ZIGENGRV, 102 + (local_var(0) == -1)));
         end
     end
     else begin
         debug_msg("grave site message #" + local_var(0));
-        display_msg(message_str(69, 275 + ((33555669 <= obj_pid(self_obj)) and (obj_pid(self_obj) <= 33555674))) + message_str(69, local_var(0)));
+        display_msg(message_str(SCRIPT_ZIGENGRV, 275 + ((33555669 <= obj_pid(self_obj)) and (obj_pid(self_obj) <= 33555674))) + message_str(SCRIPT_ZIGENGRV, local_var(0)));
     end
 end
 
@@ -120,10 +120,10 @@ procedure def_look_at_p_proc
 begin
     script_overrides;
     if (((33555669 <= obj_pid(self_obj)) and (obj_pid(self_obj) <= 33555674) or ((33555445 <= obj_pid(self_obj)) and (obj_pid(self_obj) <= 33555448))) == 0) then begin
-        display_msg(message_str(69, 100 + (local_var(0) == -1)));
+        display_msg(message_str(SCRIPT_ZIGENGRV, 100 + (local_var(0) == -1)));
     end
     else begin
-        display_msg(message_str(69, 200 + ((33555669 <= obj_pid(self_obj)) and (obj_pid(self_obj) <= 33555674))));
+        display_msg(message_str(SCRIPT_ZIGENGRV, 200 + ((33555669 <= obj_pid(self_obj)) and (obj_pid(self_obj) <= 33555674))));
     end
 end
 
@@ -131,12 +131,12 @@ procedure grave_description_proc
 begin
     if ((33555669 <= obj_pid(self_obj)) and (obj_pid(self_obj) <= 33555674) or ((33555445 <= obj_pid(self_obj)) and (obj_pid(self_obj) <= 33555448))) then begin
         script_overrides;
-        display_msg(message_str(652, 200));
+        display_mstr(200);
     end
     else begin
         if (local_var(0) == -1) then begin
             script_overrides;
-            display_msg(message_str(652, 100 + ((global_var(345) bwand 1) != 0)));
+            display_mstr(100 + ((global_var(GVAR_NEW_RENO_FLAG_2) bwand 1) != 0));
         end
         else begin
             call def_description_p_proc();
@@ -146,27 +146,27 @@ end
 
 procedure grave_use_proc
 begin
-    if (((global_var(345) bwand 1) != 0) and (local_var(0) == -1)) then begin
-        if (((global_var(495) bwand 8388608) != 0) == 0) then begin
+    if (((global_var(GVAR_NEW_RENO_FLAG_2) bwand 1) != 0) and (local_var(0) == -1)) then begin
+        if (((global_var(GVAR_NEW_RENO_FLAG_3) bwand 8388608) != 0) == 0) then begin
             script_overrides;
-            float_msg(dude_obj, message_str(652, 255 + (get_critter_stat(dude_obj, 4) <= 3)), 8);
-            display_msg(message_str(652, 250));
+            float_msg(dude_obj, mstr(255 + (dude_is_stupid)), FLOAT_MSG_YELLOW);
+            display_mstr(250);
             give_exp_points(500);
-            display_msg(message_str(14, 100) + 500 + message_str(14, 101));
-            if (((global_var(495) bwand 8388608) != 0) == 0) then begin
+            display_msg(g_mstr(100) + 500 + g_mstr(101));
+            if (((global_var(GVAR_NEW_RENO_FLAG_3) bwand 8388608) != 0) == 0) then begin
                 gfade_out(10);
                 game_time_advance(60 * 10 * 15);
                 script_overrides;
                 give_exp_points(500);
-                display_msg(message_str(14, 100) + 500 + message_str(14, 101));
-                if (metarule(17, 8) == 0) then begin
+                display_msg(g_mstr(100) + 500 + g_mstr(101));
+                if (town_known(8) == MARK_STATE_UNKNOWN) then begin
                     debug_msg("  mark_on_map(" + 8 + ")");
-                    mark_area_known(0, 8, 1);
+                    mark_area_known(MARK_TYPE_TOWN, 8, MARK_STATE_KNOWN);
                 end
-                display_msg(message_str(652, 150));
+                display_mstr(150);
                 gfade_in(10);
             end
-            set_global_var(495, global_var(495) bwor 8388608);
+            set_global_var(GVAR_NEW_RENO_FLAG_3, global_var(GVAR_NEW_RENO_FLAG_3) bwor 8388608);
         end
     end
     else begin
@@ -180,7 +180,7 @@ begin
         call def_look_at_p_proc();
     end
     else begin
-        display_msg(message_str(652, 200));
+        display_mstr(200);
     end
 end
 
@@ -336,10 +336,10 @@ begin
         if (local_var(2) != 0) then begin
             if (is_success(roll_vs_skill(source_obj, 11, -20))) then begin
                 if (source_obj == dude_obj) then begin
-                    display_msg(message_str(69, 515));
+                    display_msg(message_str(SCRIPT_ZIGENGRV, 515));
                 end
                 else begin
-                    display_msg(obj_name(source_obj) + message_str(69, 520));
+                    display_msg(obj_name(source_obj) + message_str(SCRIPT_ZIGENGRV, 520));
                 end
             end
             else begin
@@ -361,10 +361,10 @@ begin
         if (local_var(2) != 0) then begin
             if (is_success(roll_vs_skill(source_obj, 11, -20))) then begin
                 if (source_obj == dude_obj) then begin
-                    display_msg(message_str(69, 515));
+                    display_msg(message_str(SCRIPT_ZIGENGRV, 515));
                 end
                 else begin
-                    display_msg(obj_name(source_obj) + message_str(69, 520));
+                    display_msg(obj_name(source_obj) + message_str(SCRIPT_ZIGENGRV, 520));
                 end
             end
             else begin
@@ -373,7 +373,7 @@ begin
         end
         else begin
             if (obj_pid(obj_being_used_with) == 289) then begin
-                gfade_out(1);
+                fade_out;
                 if (local_var(0) == -1) then begin
                     set_local_var(0, 1);
                     play_sfx("iccntnre");
@@ -386,50 +386,50 @@ begin
                     if (no_karma_penalty == 0) then begin
                         if (local_var(1) == 0) then begin
                             set_local_var(1, 1);
-                            set_global_var(0, global_var(0) + -5);
+                            set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + -5);
                             debug_msg("Player gains " + -5 + " Karma Points.");
-                            set_global_var(37, 0);
-                            set_global_var(38, 0);
-                            set_global_var(39, 0);
-                            set_global_var(40, 0);
-                            set_global_var(41, 0);
-                            set_global_var(42, 0);
-                            set_global_var(43, 0);
-                            set_global_var(44, 0);
-                            set_global_var(45, 0);
-                            if (global_var(0) >= 1000) then begin
-                                set_global_var(37, 1);
+                            set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+                            set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+                            set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+                            set_global_var(GVAR_KARMA_DEFENDER, 0);
+                            set_global_var(GVAR_KARMA_WANDERER, 0);
+                            set_global_var(GVAR_KARMA_BETRAYER, 0);
+                            set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+                            set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+                            set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+                            if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+                                set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
                             end
                             else begin
-                                if (global_var(0) >= 750) then begin
-                                    set_global_var(38, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                                    set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
                                 end
                                 else begin
-                                    if (global_var(0) >= 500) then begin
-                                        set_global_var(39, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                                        set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                                     end
                                     else begin
-                                        if (global_var(0) >= 250) then begin
-                                            set_global_var(40, 1);
+                                        if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                                            set_global_var(GVAR_KARMA_DEFENDER, 1);
                                         end
                                         else begin
-                                            if (global_var(0) > -250) then begin
-                                                set_global_var(41, 1);
+                                            if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                                                set_global_var(GVAR_KARMA_WANDERER, 1);
                                             end
                                             else begin
-                                                if (global_var(0) > -500) then begin
-                                                    set_global_var(42, 1);
+                                                if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                                    set_global_var(GVAR_KARMA_BETRAYER, 1);
                                                 end
                                                 else begin
-                                                    if (global_var(0) > -750) then begin
-                                                        set_global_var(43, 1);
+                                                    if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                                        set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                                     end
                                                     else begin
-                                                        if (global_var(0) > -1000) then begin
-                                                            set_global_var(44, 1);
+                                                        if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                                            set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                                         end
                                                         else begin
-                                                            set_global_var(45, 1);
+                                                            set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                                         end
                                                     end
                                                 end
@@ -438,13 +438,13 @@ begin
                                     end
                                 end
                             end
-                            set_global_var(319, global_var(319) + 1);
+                            set_global_var(GVAR_GRAVES_UNEARTHED, global_var(GVAR_GRAVES_UNEARTHED) + 1);
                         end
                     end
                 end
-                display_msg(message_str(69, 150 + (local_var(0) == -1)));
-                game_time_advance(60 * 10 * (15 + (random(30, 45) / get_critter_stat(dude_obj, 0))));
-                gfade_in(1);
+                display_msg(message_str(SCRIPT_ZIGENGRV, 150 + (local_var(0) == -1)));
+                game_time_advance(60 * 10 * (15 + (random(30, 45) / dude_strength)));
+                fade_in;
             end
             else begin
                 if ((obj_pid(obj_being_used_with) == 51) or (obj_pid(obj_being_used_with) == 85)) then begin
@@ -464,10 +464,10 @@ begin
         end
         else begin
             if (source_obj == dude_obj) then begin
-                display_msg(message_str(69, 516));
+                display_msg(message_str(SCRIPT_ZIGENGRV, 516));
             end
             else begin
-                display_msg(obj_name(source_obj) + message_str(69, 521));
+                display_msg(obj_name(source_obj) + message_str(SCRIPT_ZIGENGRV, 521));
             end
         end
     end
@@ -483,10 +483,10 @@ begin
             set_local_var(2, obj_pid(obj_being_used_with));
             LVar0 := rm_mult_objs_from_inven(source_obj, obj_being_used_with, 1);
             if (source_obj == dude_obj) then begin
-                display_msg(message_str(69, 500));
+                display_msg(message_str(SCRIPT_ZIGENGRV, 500));
             end
             else begin
-                display_msg(obj_name(source_obj) + message_str(69, 510));
+                display_msg(obj_name(source_obj) + message_str(SCRIPT_ZIGENGRV, 510));
             end
         end
         else begin
@@ -497,20 +497,20 @@ begin
             end
             else begin
                 if (source_obj == dude_obj) then begin
-                    display_msg(message_str(69, 501));
+                    display_msg(message_str(SCRIPT_ZIGENGRV, 501));
                 end
                 else begin
-                    display_msg(obj_name(source_obj) + message_str(69, 511));
+                    display_msg(obj_name(source_obj) + message_str(SCRIPT_ZIGENGRV, 511));
                 end
             end
         end
     end
     else begin
         if (source_obj == dude_obj) then begin
-            display_msg(message_str(69, 501));
+            display_msg(message_str(SCRIPT_ZIGENGRV, 501));
         end
         else begin
-            display_msg(obj_name(source_obj) + message_str(69, 511));
+            display_msg(obj_name(source_obj) + message_str(SCRIPT_ZIGENGRV, 511));
         end
     end
 end
@@ -533,16 +533,16 @@ begin
     if (is_success(trap_roll)) then begin
         set_local_var(2, 0);
         if (source_obj == dude_obj) then begin
-            display_msg(message_str(69, 525));
+            display_msg(message_str(SCRIPT_ZIGENGRV, 525));
         end
         else begin
-            display_msg(obj_name(source_obj) + message_str(69, 530));
+            display_msg(obj_name(source_obj) + message_str(SCRIPT_ZIGENGRV, 530));
         end
         if (local_var(3) == 1) then begin
             if (local_var(4) == 0) then begin
                 set_local_var(4, 1);
                 give_exp_points(25);
-                display_msg(message_str(14, 100) + 25 + message_str(14, 101));
+                display_msg(g_mstr(100) + 25 + g_mstr(101));
             end
         end
         else begin
@@ -556,18 +556,18 @@ begin
         else begin
             if (local_var(3) == 0) then begin
                 if (source_obj == dude_obj) then begin
-                    display_msg(message_str(69, 516));
+                    display_msg(message_str(SCRIPT_ZIGENGRV, 516));
                 end
                 else begin
-                    display_msg(obj_name(source_obj) + message_str(69, 521));
+                    display_msg(obj_name(source_obj) + message_str(SCRIPT_ZIGENGRV, 521));
                 end
             end
             else begin
                 if (source_obj == dude_obj) then begin
-                    display_msg(message_str(69, 526));
+                    display_msg(message_str(SCRIPT_ZIGENGRV, 526));
                 end
                 else begin
-                    display_msg(obj_name(source_obj) + message_str(69, 531));
+                    display_msg(obj_name(source_obj) + message_str(SCRIPT_ZIGENGRV, 531));
                 end
             end
         end
@@ -606,53 +606,53 @@ end
 procedure ChooseFromList
 begin
     variable LVar0 := 0;
-    gsay_reply(69, stored_msg);
-    if (message_str(353, top_msg - 1) != "") then begin
-        giq_option(1, 69, message_str(353, 150), CLICK_PREV, 50);
+    gsay_reply(SCRIPT_ZIGENGRV, stored_msg);
+    if (message_str(SCRIPT_NEWR1, top_msg - 1) != "") then begin
+        giq_option(1, SCRIPT_ZIGENGRV, message_str(SCRIPT_NEWR1, 150), CLICK_PREV, NEUTRAL_REACTION);
     end
     if (LVar0 == 0) then begin
-        if (message_str(353, top_msg + 0) != "") then begin
-            giq_option(1, 69, message_str(353, top_msg + 0), CLICK_0, 50);
+        if (message_str(SCRIPT_NEWR1, top_msg + 0) != "") then begin
+            giq_option(1, SCRIPT_ZIGENGRV, message_str(SCRIPT_NEWR1, top_msg + 0), CLICK_0, NEUTRAL_REACTION);
         end
         else begin
             LVar0 := 1;
         end
     end
     if (LVar0 == 0) then begin
-        if (message_str(353, top_msg + 1) != "") then begin
-            giq_option(1, 69, message_str(353, top_msg + 1), CLICK_1, 50);
+        if (message_str(SCRIPT_NEWR1, top_msg + 1) != "") then begin
+            giq_option(1, SCRIPT_ZIGENGRV, message_str(SCRIPT_NEWR1, top_msg + 1), CLICK_1, NEUTRAL_REACTION);
         end
         else begin
             LVar0 := 1;
         end
     end
     if (LVar0 == 0) then begin
-        if (message_str(353, top_msg + 2) != "") then begin
-            giq_option(1, 69, message_str(353, top_msg + 2), CLICK_2, 50);
+        if (message_str(SCRIPT_NEWR1, top_msg + 2) != "") then begin
+            giq_option(1, SCRIPT_ZIGENGRV, message_str(SCRIPT_NEWR1, top_msg + 2), CLICK_2, NEUTRAL_REACTION);
         end
         else begin
             LVar0 := 1;
         end
     end
     if (LVar0 == 0) then begin
-        if (message_str(353, top_msg + 3) != "") then begin
-            giq_option(1, 69, message_str(353, top_msg + 3), CLICK_3, 50);
+        if (message_str(SCRIPT_NEWR1, top_msg + 3) != "") then begin
+            giq_option(1, SCRIPT_ZIGENGRV, message_str(SCRIPT_NEWR1, top_msg + 3), CLICK_3, NEUTRAL_REACTION);
         end
         else begin
             LVar0 := 1;
         end
     end
     if (LVar0 == 0) then begin
-        if (message_str(353, top_msg + 4) != "") then begin
-            giq_option(1, 69, message_str(353, top_msg + 4), CLICK_4, 50);
+        if (message_str(SCRIPT_NEWR1, top_msg + 4) != "") then begin
+            giq_option(1, SCRIPT_ZIGENGRV, message_str(SCRIPT_NEWR1, top_msg + 4), CLICK_4, NEUTRAL_REACTION);
         end
         else begin
             LVar0 := 1;
         end
     end
     if (LVar0 == 0) then begin
-        if (message_str(353, top_msg + 5) != "") then begin
-            giq_option(1, 69, message_str(353, 151), CLICK_MORE, 50);
+        if (message_str(SCRIPT_NEWR1, top_msg + 5) != "") then begin
+            giq_option(1, SCRIPT_ZIGENGRV, message_str(SCRIPT_NEWR1, 151), CLICK_MORE, NEUTRAL_REACTION);
         end
     end
 end
@@ -698,4 +698,3 @@ begin
     top_msg := top_msg + 5;
     call ChooseFromList();
 end
-
