diff --git a/Games/Fallout2.Steam/Original/NITOILET.ssl b/Games/Fallout2.Steam/ReDefine/NITOILET.ssl
index 5d4fa17..a364801 100644
--- a/Games/Fallout2.Steam/Original/NITOILET.ssl
+++ b/Games/Fallout2.Steam/ReDefine/NITOILET.ssl
@@ -170,23 +170,23 @@ end
 procedure talk_p_proc
 begin
     variable LVar0 := 0;
-    if (global_var(243) == 0) then begin
-        float_msg(dude_obj, message_str(1196, 100), 8);
+    if (global_var(GVAR_8_BALL_TOILET_SECRET) == 0) then begin
+        float_msg(dude_obj, mstr(100), FLOAT_MSG_YELLOW);
     end
     else begin
-        if ((global_var(243) == 1) and (local_var(0) == 0)) then begin
+        if ((global_var(GVAR_8_BALL_TOILET_SECRET) == 1) and (local_var(0) == 0)) then begin
             set_local_var(0, 1);
-            LVar0 := create_object_sid(25, 0, 0, -1);
+            LVar0 := create_object(25, 0, 0);
             add_obj_to_inven(dude_obj, LVar0);
-            LVar0 := create_object_sid(27, 0, 0, -1);
+            LVar0 := create_object(27, 0, 0);
             add_obj_to_inven(dude_obj, LVar0);
-            LVar0 := create_object_sid(26, 0, 0, -1);
+            LVar0 := create_object(26, 0, 0);
             add_obj_to_inven(dude_obj, LVar0);
-            float_msg(dude_obj, message_str(1196, 101), 8);
+            float_msg(dude_obj, mstr(101), FLOAT_MSG_YELLOW);
         end
         else begin
-            if ((global_var(243) == 1) and (local_var(0) == 1)) then begin
-                float_msg(dude_obj, message_str(1196, 102), 8);
+            if ((global_var(GVAR_8_BALL_TOILET_SECRET) == 1) and (local_var(0) == 1)) then begin
+                float_msg(dude_obj, mstr(102), FLOAT_MSG_YELLOW);
             end
         end
     end
@@ -195,4 +195,3 @@ end
 procedure timed_event_p_proc
 begin
 end
-
