diff --git a/Games/Fallout2.Steam/Original/NIWRIGRV.ssl b/Games/Fallout2.Steam/ReDefine/NIWRIGRV.ssl
index 61cfd8c..6d86352 100644
--- a/Games/Fallout2.Steam/Original/NIWRIGRV.ssl
+++ b/Games/Fallout2.Steam/ReDefine/NIWRIGRV.ssl
@@ -76,10 +76,10 @@ begin
         if (local_var(2) != 0) then begin
             if (is_success(roll_vs_skill(source_obj, 11, -20))) then begin
                 if (source_obj == dude_obj) then begin
-                    display_msg(message_str(69, 515));
+                    display_msg(message_str(SCRIPT_ZIGENGRV, 515));
                 end
                 else begin
-                    display_msg(obj_name(source_obj) + message_str(69, 520));
+                    display_msg(obj_name(source_obj) + message_str(SCRIPT_ZIGENGRV, 520));
                 end
             end
             else begin
@@ -100,19 +100,19 @@ begin
     if (((33555669 <= obj_pid(self_obj)) and (obj_pid(self_obj) <= 33555674) or ((33555445 <= obj_pid(self_obj)) and (obj_pid(self_obj) <= 33555448))) == 0) then begin
         if ((local_var(2) != 0) and is_success(roll_vs_skill(source_obj, 11, -20))) then begin
             if (source_obj == dude_obj) then begin
-                display_msg(message_str(69, 515));
+                display_msg(message_str(SCRIPT_ZIGENGRV, 515));
             end
             else begin
-                display_msg(obj_name(source_obj) + message_str(69, 520));
+                display_msg(obj_name(source_obj) + message_str(SCRIPT_ZIGENGRV, 520));
             end
         end
         else begin
-            display_msg(message_str(69, 102 + (local_var(0) == -1)));
+            display_msg(message_str(SCRIPT_ZIGENGRV, 102 + (local_var(0) == -1)));
         end
     end
     else begin
         debug_msg("grave site message #" + local_var(0));
-        display_msg(message_str(69, 275 + ((33555669 <= obj_pid(self_obj)) and (obj_pid(self_obj) <= 33555674))) + message_str(69, local_var(0)));
+        display_msg(message_str(SCRIPT_ZIGENGRV, 275 + ((33555669 <= obj_pid(self_obj)) and (obj_pid(self_obj) <= 33555674))) + message_str(SCRIPT_ZIGENGRV, local_var(0)));
     end
 end
 
@@ -120,10 +120,10 @@ procedure def_look_at_p_proc
 begin
     script_overrides;
     if (((33555669 <= obj_pid(self_obj)) and (obj_pid(self_obj) <= 33555674) or ((33555445 <= obj_pid(self_obj)) and (obj_pid(self_obj) <= 33555448))) == 0) then begin
-        display_msg(message_str(69, 100 + (local_var(0) == -1)));
+        display_msg(message_str(SCRIPT_ZIGENGRV, 100 + (local_var(0) == -1)));
     end
     else begin
-        display_msg(message_str(69, 200 + ((33555669 <= obj_pid(self_obj)) and (obj_pid(self_obj) <= 33555674))));
+        display_msg(message_str(SCRIPT_ZIGENGRV, 200 + ((33555669 <= obj_pid(self_obj)) and (obj_pid(self_obj) <= 33555674))));
     end
 end
 
@@ -131,20 +131,20 @@ procedure grave_description_proc
 begin
     if ((33555669 <= obj_pid(self_obj)) and (obj_pid(self_obj) <= 33555674) or ((33555445 <= obj_pid(self_obj)) and (obj_pid(self_obj) <= 33555448))) then begin
         script_overrides;
-        display_msg(message_str(654, 100));
+        display_mstr(100);
     end
     else begin
-        if ((global_var(286) == 1) and (local_var(0) == -1)) then begin
-            if (((global_var(314) bwand 1024) != 0) == 0) then begin
+        if ((global_var(GVAR_NEW_RENO_WRIGHT_MYSTERY) == 1) and (local_var(0) == -1)) then begin
+            if (((global_var(GVAR_NEW_RENO_FLAG_1) bwand 1024) != 0) == 0) then begin
                 script_overrides;
-                display_msg(message_str(654, 201));
-                display_msg(message_str(654, 300));
-                float_msg(self_obj, message_str(654, 202), 8);
-                if (((global_var(314) bwand 2048) != 0) == 0) then begin
+                display_mstr(201);
+                display_mstr(300);
+                float_msg(self_obj, mstr(202), FLOAT_MSG_YELLOW);
+                if (((global_var(GVAR_NEW_RENO_FLAG_1) bwand 2048) != 0) == 0) then begin
                     give_exp_points(500);
-                    display_msg(message_str(14, 100) + 500 + message_str(14, 101));
+                    display_msg(g_mstr(100) + 500 + g_mstr(101));
                 end
-                set_global_var(314, global_var(314) bwor 1024);
+                set_global_var(GVAR_NEW_RENO_FLAG_1, global_var(GVAR_NEW_RENO_FLAG_1) bwor 1024);
             end
             else begin
                 call def_description_p_proc();
@@ -158,17 +158,17 @@ end
 
 procedure grave_use_proc
 begin
-    if ((global_var(286) == 1) and (local_var(0) == -1)) then begin
-        if (((global_var(314) bwand 1024) != 0) == 0) then begin
+    if ((global_var(GVAR_NEW_RENO_WRIGHT_MYSTERY) == 1) and (local_var(0) == -1)) then begin
+        if (((global_var(GVAR_NEW_RENO_FLAG_1) bwand 1024) != 0) == 0) then begin
             script_overrides;
-            display_msg(message_str(654, 201));
-            display_msg(message_str(654, 300));
-            float_msg(self_obj, message_str(654, 202), 8);
-            if (((global_var(314) bwand 2048) != 0) == 0) then begin
+            display_mstr(201);
+            display_mstr(300);
+            float_msg(self_obj, mstr(202), FLOAT_MSG_YELLOW);
+            if (((global_var(GVAR_NEW_RENO_FLAG_1) bwand 2048) != 0) == 0) then begin
                 give_exp_points(500);
-                display_msg(message_str(14, 100) + 500 + message_str(14, 101));
+                display_msg(g_mstr(100) + 500 + g_mstr(101));
             end
-            set_global_var(314, global_var(314) bwor 1024);
+            set_global_var(GVAR_NEW_RENO_FLAG_1, global_var(GVAR_NEW_RENO_FLAG_1) bwor 1024);
         end
         else begin
             call def_use_p_proc();
@@ -185,7 +185,7 @@ begin
         call def_look_at_p_proc();
     end
     else begin
-        display_msg(message_str(654, 200));
+        display_mstr(200);
     end
 end
 
@@ -341,10 +341,10 @@ begin
         if (local_var(2) != 0) then begin
             if (is_success(roll_vs_skill(source_obj, 11, -20))) then begin
                 if (source_obj == dude_obj) then begin
-                    display_msg(message_str(69, 515));
+                    display_msg(message_str(SCRIPT_ZIGENGRV, 515));
                 end
                 else begin
-                    display_msg(obj_name(source_obj) + message_str(69, 520));
+                    display_msg(obj_name(source_obj) + message_str(SCRIPT_ZIGENGRV, 520));
                 end
             end
             else begin
@@ -366,10 +366,10 @@ begin
         if (local_var(2) != 0) then begin
             if (is_success(roll_vs_skill(source_obj, 11, -20))) then begin
                 if (source_obj == dude_obj) then begin
-                    display_msg(message_str(69, 515));
+                    display_msg(message_str(SCRIPT_ZIGENGRV, 515));
                 end
                 else begin
-                    display_msg(obj_name(source_obj) + message_str(69, 520));
+                    display_msg(obj_name(source_obj) + message_str(SCRIPT_ZIGENGRV, 520));
                 end
             end
             else begin
@@ -378,7 +378,7 @@ begin
         end
         else begin
             if (obj_pid(obj_being_used_with) == 289) then begin
-                gfade_out(1);
+                fade_out;
                 if (local_var(0) == -1) then begin
                     set_local_var(0, 1);
                     play_sfx("iccntnre");
@@ -391,50 +391,50 @@ begin
                     if (no_karma_penalty == 0) then begin
                         if (local_var(1) == 0) then begin
                             set_local_var(1, 1);
-                            set_global_var(0, global_var(0) + -5);
+                            set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + -5);
                             debug_msg("Player gains " + -5 + " Karma Points.");
-                            set_global_var(37, 0);
-                            set_global_var(38, 0);
-                            set_global_var(39, 0);
-                            set_global_var(40, 0);
-                            set_global_var(41, 0);
-                            set_global_var(42, 0);
-                            set_global_var(43, 0);
-                            set_global_var(44, 0);
-                            set_global_var(45, 0);
-                            if (global_var(0) >= 1000) then begin
-                                set_global_var(37, 1);
+                            set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+                            set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+                            set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+                            set_global_var(GVAR_KARMA_DEFENDER, 0);
+                            set_global_var(GVAR_KARMA_WANDERER, 0);
+                            set_global_var(GVAR_KARMA_BETRAYER, 0);
+                            set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+                            set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+                            set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+                            if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+                                set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
                             end
                             else begin
-                                if (global_var(0) >= 750) then begin
-                                    set_global_var(38, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                                    set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
                                 end
                                 else begin
-                                    if (global_var(0) >= 500) then begin
-                                        set_global_var(39, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                                        set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                                     end
                                     else begin
-                                        if (global_var(0) >= 250) then begin
-                                            set_global_var(40, 1);
+                                        if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                                            set_global_var(GVAR_KARMA_DEFENDER, 1);
                                         end
                                         else begin
-                                            if (global_var(0) > -250) then begin
-                                                set_global_var(41, 1);
+                                            if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                                                set_global_var(GVAR_KARMA_WANDERER, 1);
                                             end
                                             else begin
-                                                if (global_var(0) > -500) then begin
-                                                    set_global_var(42, 1);
+                                                if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                                    set_global_var(GVAR_KARMA_BETRAYER, 1);
                                                 end
                                                 else begin
-                                                    if (global_var(0) > -750) then begin
-                                                        set_global_var(43, 1);
+                                                    if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                                        set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                                     end
                                                     else begin
-                                                        if (global_var(0) > -1000) then begin
-                                                            set_global_var(44, 1);
+                                                        if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                                            set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                                         end
                                                         else begin
-                                                            set_global_var(45, 1);
+                                                            set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                                         end
                                                     end
                                                 end
@@ -443,13 +443,13 @@ begin
                                     end
                                 end
                             end
-                            set_global_var(319, global_var(319) + 1);
+                            set_global_var(GVAR_GRAVES_UNEARTHED, global_var(GVAR_GRAVES_UNEARTHED) + 1);
                         end
                     end
                 end
-                display_msg(message_str(69, 150 + (local_var(0) == -1)));
-                game_time_advance(60 * 10 * (15 + (random(30, 45) / get_critter_stat(dude_obj, 0))));
-                gfade_in(1);
+                display_msg(message_str(SCRIPT_ZIGENGRV, 150 + (local_var(0) == -1)));
+                game_time_advance(60 * 10 * (15 + (random(30, 45) / dude_strength)));
+                fade_in;
             end
             else begin
                 if ((obj_pid(obj_being_used_with) == 51) or (obj_pid(obj_being_used_with) == 85)) then begin
@@ -469,10 +469,10 @@ begin
         end
         else begin
             if (source_obj == dude_obj) then begin
-                display_msg(message_str(69, 516));
+                display_msg(message_str(SCRIPT_ZIGENGRV, 516));
             end
             else begin
-                display_msg(obj_name(source_obj) + message_str(69, 521));
+                display_msg(obj_name(source_obj) + message_str(SCRIPT_ZIGENGRV, 521));
             end
         end
     end
@@ -488,10 +488,10 @@ begin
             set_local_var(2, obj_pid(obj_being_used_with));
             LVar0 := rm_mult_objs_from_inven(source_obj, obj_being_used_with, 1);
             if (source_obj == dude_obj) then begin
-                display_msg(message_str(69, 500));
+                display_msg(message_str(SCRIPT_ZIGENGRV, 500));
             end
             else begin
-                display_msg(obj_name(source_obj) + message_str(69, 510));
+                display_msg(obj_name(source_obj) + message_str(SCRIPT_ZIGENGRV, 510));
             end
         end
         else begin
@@ -502,20 +502,20 @@ begin
             end
             else begin
                 if (source_obj == dude_obj) then begin
-                    display_msg(message_str(69, 501));
+                    display_msg(message_str(SCRIPT_ZIGENGRV, 501));
                 end
                 else begin
-                    display_msg(obj_name(source_obj) + message_str(69, 511));
+                    display_msg(obj_name(source_obj) + message_str(SCRIPT_ZIGENGRV, 511));
                 end
             end
         end
     end
     else begin
         if (source_obj == dude_obj) then begin
-            display_msg(message_str(69, 501));
+            display_msg(message_str(SCRIPT_ZIGENGRV, 501));
         end
         else begin
-            display_msg(obj_name(source_obj) + message_str(69, 511));
+            display_msg(obj_name(source_obj) + message_str(SCRIPT_ZIGENGRV, 511));
         end
     end
 end
@@ -538,16 +538,16 @@ begin
     if (is_success(trap_roll)) then begin
         set_local_var(2, 0);
         if (source_obj == dude_obj) then begin
-            display_msg(message_str(69, 525));
+            display_msg(message_str(SCRIPT_ZIGENGRV, 525));
         end
         else begin
-            display_msg(obj_name(source_obj) + message_str(69, 530));
+            display_msg(obj_name(source_obj) + message_str(SCRIPT_ZIGENGRV, 530));
         end
         if (local_var(3) == 1) then begin
             if (local_var(4) == 0) then begin
                 set_local_var(4, 1);
                 give_exp_points(25);
-                display_msg(message_str(14, 100) + 25 + message_str(14, 101));
+                display_msg(g_mstr(100) + 25 + g_mstr(101));
             end
         end
         else begin
@@ -561,18 +561,18 @@ begin
         else begin
             if (local_var(3) == 0) then begin
                 if (source_obj == dude_obj) then begin
-                    display_msg(message_str(69, 516));
+                    display_msg(message_str(SCRIPT_ZIGENGRV, 516));
                 end
                 else begin
-                    display_msg(obj_name(source_obj) + message_str(69, 521));
+                    display_msg(obj_name(source_obj) + message_str(SCRIPT_ZIGENGRV, 521));
                 end
             end
             else begin
                 if (source_obj == dude_obj) then begin
-                    display_msg(message_str(69, 526));
+                    display_msg(message_str(SCRIPT_ZIGENGRV, 526));
                 end
                 else begin
-                    display_msg(obj_name(source_obj) + message_str(69, 531));
+                    display_msg(obj_name(source_obj) + message_str(SCRIPT_ZIGENGRV, 531));
                 end
             end
         end
@@ -611,53 +611,53 @@ end
 procedure ChooseFromList
 begin
     variable LVar0 := 0;
-    gsay_reply(69, stored_msg);
-    if (message_str(353, top_msg - 1) != "") then begin
-        giq_option(1, 69, message_str(353, 150), CLICK_PREV, 50);
+    gsay_reply(SCRIPT_ZIGENGRV, stored_msg);
+    if (message_str(SCRIPT_NEWR1, top_msg - 1) != "") then begin
+        giq_option(1, SCRIPT_ZIGENGRV, message_str(SCRIPT_NEWR1, 150), CLICK_PREV, NEUTRAL_REACTION);
     end
     if (LVar0 == 0) then begin
-        if (message_str(353, top_msg + 0) != "") then begin
-            giq_option(1, 69, message_str(353, top_msg + 0), CLICK_0, 50);
+        if (message_str(SCRIPT_NEWR1, top_msg + 0) != "") then begin
+            giq_option(1, SCRIPT_ZIGENGRV, message_str(SCRIPT_NEWR1, top_msg + 0), CLICK_0, NEUTRAL_REACTION);
         end
         else begin
             LVar0 := 1;
         end
     end
     if (LVar0 == 0) then begin
-        if (message_str(353, top_msg + 1) != "") then begin
-            giq_option(1, 69, message_str(353, top_msg + 1), CLICK_1, 50);
+        if (message_str(SCRIPT_NEWR1, top_msg + 1) != "") then begin
+            giq_option(1, SCRIPT_ZIGENGRV, message_str(SCRIPT_NEWR1, top_msg + 1), CLICK_1, NEUTRAL_REACTION);
         end
         else begin
             LVar0 := 1;
         end
     end
     if (LVar0 == 0) then begin
-        if (message_str(353, top_msg + 2) != "") then begin
-            giq_option(1, 69, message_str(353, top_msg + 2), CLICK_2, 50);
+        if (message_str(SCRIPT_NEWR1, top_msg + 2) != "") then begin
+            giq_option(1, SCRIPT_ZIGENGRV, message_str(SCRIPT_NEWR1, top_msg + 2), CLICK_2, NEUTRAL_REACTION);
         end
         else begin
             LVar0 := 1;
         end
     end
     if (LVar0 == 0) then begin
-        if (message_str(353, top_msg + 3) != "") then begin
-            giq_option(1, 69, message_str(353, top_msg + 3), CLICK_3, 50);
+        if (message_str(SCRIPT_NEWR1, top_msg + 3) != "") then begin
+            giq_option(1, SCRIPT_ZIGENGRV, message_str(SCRIPT_NEWR1, top_msg + 3), CLICK_3, NEUTRAL_REACTION);
         end
         else begin
             LVar0 := 1;
         end
     end
     if (LVar0 == 0) then begin
-        if (message_str(353, top_msg + 4) != "") then begin
-            giq_option(1, 69, message_str(353, top_msg + 4), CLICK_4, 50);
+        if (message_str(SCRIPT_NEWR1, top_msg + 4) != "") then begin
+            giq_option(1, SCRIPT_ZIGENGRV, message_str(SCRIPT_NEWR1, top_msg + 4), CLICK_4, NEUTRAL_REACTION);
         end
         else begin
             LVar0 := 1;
         end
     end
     if (LVar0 == 0) then begin
-        if (message_str(353, top_msg + 5) != "") then begin
-            giq_option(1, 69, message_str(353, 151), CLICK_MORE, 50);
+        if (message_str(SCRIPT_NEWR1, top_msg + 5) != "") then begin
+            giq_option(1, SCRIPT_ZIGENGRV, message_str(SCRIPT_NEWR1, 151), CLICK_MORE, NEUTRAL_REACTION);
         end
     end
 end
@@ -703,4 +703,3 @@ begin
     top_msg := top_msg + 5;
     call ChooseFromList();
 end
-
