diff --git a/Games/Fallout2.Steam/Original/OCGRUTHA.ssl b/Games/Fallout2.Steam/ReDefine/OCGRUTHA.ssl
index df98cb4..ec9e642 100644
--- a/Games/Fallout2.Steam/Original/OCGRUTHA.ssl
+++ b/Games/Fallout2.Steam/ReDefine/OCGRUTHA.ssl
@@ -223,24 +223,24 @@ procedure pickup_p_proc
 begin
     if (source_obj == dude_obj) then begin
         if (1 == 1) then begin
-            attack_complex(source_obj, 0, 1, 0, 0, 30000, 0, 0);
+            attack(source_obj);
         end
         else begin
             if (1 == 2) then begin
-                float_msg(self_obj, message_str(14, random(3325, 3329)), 0);
+                float_msg(self_obj, g_mstr(random(3325, 3329)), FLOAT_MSG_NORMAL);
                 if (18 > -1) then begin
                     if (local_var(18) >= 1) then begin
-                        attack_complex(source_obj, 0, 1, 0, 0, 30000, 0, 0);
+                        attack(source_obj);
                     end
                     set_local_var(18, local_var(18) + 1);
                 end
             end
             else begin
                 if (1 == 3) then begin
-                    float_msg(self_obj, message_str(14, random(3050, 3064)), 0);
+                    float_msg(self_obj, g_mstr(random(3050, 3064)), FLOAT_MSG_NORMAL);
                     if (18 > -1) then begin
                         if (local_var(18) >= 1) then begin
-                            attack_complex(source_obj, 0, 1, 0, 0, 30000, 0, 0);
+                            attack(source_obj);
                         end
                         set_local_var(18, local_var(18) + 1);
                     end
@@ -338,8 +338,8 @@ end
 
 procedure start
 begin
-    if ((global_var(582) bwand 1) != 0) then begin
-        set_global_var(429, global_var(429) bwor 2);
+    if ((global_var(GVAR_VIGNETTE_SEQUENCE) bwand 1) != 0) then begin
+        set_global_var(GVAR_V13_STATUS_FLAGS, global_var(GVAR_V13_STATUS_FLAGS) bwor 2);
         destroy_object(self_obj);
     end
     else begin
@@ -358,10 +358,10 @@ begin
             if (anim_busy(self_obj) == 0) then begin
                 if ((tile_distance_objs(self_obj, dude_obj) < 8) or obj_can_see_obj(self_obj, dude_obj)) then begin
                     if (anim_busy(self_obj) == 0) then begin
-                        global_temp := rotation_to_tile(tile_num(dude_obj), tile_num(self_obj));
+                        global_temp := rotation_to_tile(dude_tile, tile_num(self_obj));
                         animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(self_obj), global_temp, random(3, 10)), 1);
                         global_temp := (global_temp + 1) % 6;
-                        while ((anim_busy(self_obj) == 0) and (global_temp != rotation_to_tile(tile_num(dude_obj), tile_num(self_obj)))) do begin
+                        while ((anim_busy(self_obj) == 0) and (global_temp != rotation_to_tile(dude_tile, tile_num(self_obj)))) do begin
                             debug_msg("flee loop: rot == " + global_temp);
                             animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(self_obj), global_temp, random(3, 10)), 1);
                             global_temp := (global_temp + 1) % 6;
@@ -371,10 +371,10 @@ begin
             end
         end
         else begin
-            attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+            attack(dude_obj);
         end
     end
-    LVar1 := global_var(350);
+    LVar1 := global_var(GVAR_GRUTHAR_DSTATUS);
     if ((LVar1 == 5) or (LVar1 == 4) or (LVar1 == 0)) then begin
         if (local_var(7) == 0) then begin
             if (obj_can_see_obj(self_obj, dude_obj)) then begin
@@ -395,7 +395,7 @@ begin
     if (i_matt_obj != -1) then begin
         if (obj_can_see_obj(self_obj, i_matt_obj)) then begin
             if ((elevation(i_matt_obj) == 0) and (elevation(self_obj) == 0)) then begin
-                attack_complex(i_matt_obj, 0, 1, 0, 0, 30000, 0, 0);
+                attack(i_matt_obj);
             end
         end
     end
@@ -421,48 +421,48 @@ begin
     Evil_Critter := 0;
     Slavery_Tolerant := 2;
     Karma_Perception := 10;
-    set_global_var(37, 0);
-    set_global_var(38, 0);
-    set_global_var(39, 0);
-    set_global_var(40, 0);
-    set_global_var(41, 0);
-    set_global_var(42, 0);
-    set_global_var(43, 0);
-    set_global_var(44, 0);
-    set_global_var(45, 0);
-    if (global_var(0) >= 1000) then begin
-        set_global_var(37, 1);
+    set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+    set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+    set_global_var(GVAR_KARMA_DEFENDER, 0);
+    set_global_var(GVAR_KARMA_WANDERER, 0);
+    set_global_var(GVAR_KARMA_BETRAYER, 0);
+    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+    set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+    set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+    if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
     end
     else begin
-        if (global_var(0) >= 750) then begin
-            set_global_var(38, 1);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+            set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
         end
         else begin
-            if (global_var(0) >= 500) then begin
-                set_global_var(39, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
             end
             else begin
-                if (global_var(0) >= 250) then begin
-                    set_global_var(40, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                    set_global_var(GVAR_KARMA_DEFENDER, 1);
                 end
                 else begin
-                    if (global_var(0) > -250) then begin
-                        set_global_var(41, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                        set_global_var(GVAR_KARMA_WANDERER, 1);
                     end
                     else begin
-                        if (global_var(0) > -500) then begin
-                            set_global_var(42, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                            set_global_var(GVAR_KARMA_BETRAYER, 1);
                         end
                         else begin
-                            if (global_var(0) > -750) then begin
-                                set_global_var(43, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                             end
                             else begin
-                                if (global_var(0) > -1000) then begin
-                                    set_global_var(44, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                    set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                 end
                                 else begin
-                                    set_global_var(45, 1);
+                                    set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                 end
                             end
                         end
@@ -473,7 +473,7 @@ begin
     end
     if (local_var(2) == 0) then begin
         set_local_var(2, 1);
-        set_local_var(3, (get_critter_stat(dude_obj, 3) - get_critter_stat(self_obj, 3)) * 5);
+        set_local_var(3, (dude_charisma - self_charisma) * 5);
         debug_msg("Base Reaction == " + local_var(3));
     end
     Static_Reaction := 0;
@@ -485,14 +485,14 @@ begin
     else begin
         reaction_bonus_karma := -1 * (-1 * reaction_bonus_karma / 1000);
     end
-    if (global_var(59) >= 0) then begin
-        reaction_bonus_town_rep := global_var(59) / 2;
+    if (global_var(GVAR_TOWN_REP_VAULT_13) >= 0) then begin
+        reaction_bonus_town_rep := global_var(GVAR_TOWN_REP_VAULT_13) / 2;
     end
     else begin
-        reaction_bonus_town_rep := -1 * (-1 * global_var(59) / 2);
+        reaction_bonus_town_rep := -1 * (-1 * global_var(GVAR_TOWN_REP_VAULT_13) / 2);
     end
-    if (has_trait(2, dude_obj, 13)) then begin
-        if (get_critter_stat(dude_obj, 34) == get_critter_stat(self_obj, 34)) then begin
+    if (has_trait(TRAIT_TRAIT, dude_obj, TRAIT_sex_appeal)) then begin
+        if (dude_gender == self_gender) then begin
             Static_Reaction := Static_Reaction + -20;
         end
         else begin
@@ -508,48 +508,48 @@ begin
         debug_msg("Karma Reaction Bonus == " + Static_Reaction);
         Static_Reaction := Static_Reaction - reaction_bonus_town_rep;
         debug_msg("Town Rep Bonus == " + Static_Reaction);
-        set_global_var(37, 0);
-        set_global_var(38, 0);
-        set_global_var(39, 0);
-        set_global_var(40, 0);
-        set_global_var(41, 0);
-        set_global_var(42, 0);
-        set_global_var(43, 0);
-        set_global_var(44, 0);
-        set_global_var(45, 0);
-        if (global_var(0) >= 1000) then begin
-            set_global_var(37, 1);
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+        set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+        set_global_var(GVAR_KARMA_DEFENDER, 0);
+        set_global_var(GVAR_KARMA_WANDERER, 0);
+        set_global_var(GVAR_KARMA_BETRAYER, 0);
+        set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+        set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+            set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
         end
         else begin
-            if (global_var(0) >= 750) then begin
-                set_global_var(38, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
             end
             else begin
-                if (global_var(0) >= 500) then begin
-                    set_global_var(39, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                 end
                 else begin
-                    if (global_var(0) >= 250) then begin
-                        set_global_var(40, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                        set_global_var(GVAR_KARMA_DEFENDER, 1);
                     end
                     else begin
-                        if (global_var(0) > -250) then begin
-                            set_global_var(41, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                            set_global_var(GVAR_KARMA_WANDERER, 1);
                         end
                         else begin
-                            if (global_var(0) > -500) then begin
-                                set_global_var(42, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                set_global_var(GVAR_KARMA_BETRAYER, 1);
                             end
                             else begin
-                                if (global_var(0) > -750) then begin
-                                    set_global_var(43, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -1000) then begin
-                                        set_global_var(44, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                     end
                                     else begin
-                                        set_global_var(45, 1);
+                                        set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                     end
                                 end
                             end
@@ -558,35 +558,35 @@ begin
                 end
             end
         end
-        if (global_var(37) == 1) then begin
+        if (global_var(GVAR_KARMA_HOLY_WARRIOR) == 1) then begin
             Static_Reaction := Static_Reaction - 20;
         end
         else begin
-            if (global_var(38) == 1) then begin
+            if (global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES) == 1) then begin
                 Static_Reaction := Static_Reaction - 15;
             end
             else begin
-                if (global_var(39) == 1) then begin
+                if (global_var(GVAR_KARMA_SHIELD_OF_HOPE) == 1) then begin
                     Static_Reaction := Static_Reaction - 10;
                 end
                 else begin
-                    if (global_var(40) == 1) then begin
+                    if (global_var(GVAR_KARMA_DEFENDER) == 1) then begin
                         Static_Reaction := Static_Reaction - 5;
                     end
                     else begin
-                        if (global_var(41) == 1) then begin
+                        if (global_var(GVAR_KARMA_WANDERER) == 1) then begin
                             Static_Reaction := Static_Reaction - 0;
                         end
                         else begin
-                            if (global_var(42) == 1) then begin
+                            if (global_var(GVAR_KARMA_BETRAYER) == 1) then begin
                                 Static_Reaction := Static_Reaction - -5;
                             end
                             else begin
-                                if (global_var(43) == 1) then begin
+                                if (global_var(GVAR_KARMA_SWORD_OF_DESPAIR) == 1) then begin
                                     Static_Reaction := Static_Reaction - -10;
                                 end
                                 else begin
-                                    if (global_var(44) == 1) then begin
+                                    if (global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE) == 1) then begin
                                         Static_Reaction := Static_Reaction - -15;
                                     end
                                     else begin
@@ -606,48 +606,48 @@ begin
         debug_msg("Karma Reaction Bonus == " + Static_Reaction);
         Static_Reaction := Static_Reaction + reaction_bonus_town_rep;
         debug_msg("Town Rep Bonus == " + Static_Reaction);
-        set_global_var(37, 0);
-        set_global_var(38, 0);
-        set_global_var(39, 0);
-        set_global_var(40, 0);
-        set_global_var(41, 0);
-        set_global_var(42, 0);
-        set_global_var(43, 0);
-        set_global_var(44, 0);
-        set_global_var(45, 0);
-        if (global_var(0) >= 1000) then begin
-            set_global_var(37, 1);
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+        set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+        set_global_var(GVAR_KARMA_DEFENDER, 0);
+        set_global_var(GVAR_KARMA_WANDERER, 0);
+        set_global_var(GVAR_KARMA_BETRAYER, 0);
+        set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+        set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+            set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
         end
         else begin
-            if (global_var(0) >= 750) then begin
-                set_global_var(38, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
             end
             else begin
-                if (global_var(0) >= 500) then begin
-                    set_global_var(39, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                 end
                 else begin
-                    if (global_var(0) >= 250) then begin
-                        set_global_var(40, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                        set_global_var(GVAR_KARMA_DEFENDER, 1);
                     end
                     else begin
-                        if (global_var(0) > -250) then begin
-                            set_global_var(41, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                            set_global_var(GVAR_KARMA_WANDERER, 1);
                         end
                         else begin
-                            if (global_var(0) > -500) then begin
-                                set_global_var(42, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                set_global_var(GVAR_KARMA_BETRAYER, 1);
                             end
                             else begin
-                                if (global_var(0) > -750) then begin
-                                    set_global_var(43, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -1000) then begin
-                                        set_global_var(44, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                     end
                                     else begin
-                                        set_global_var(45, 1);
+                                        set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                     end
                                 end
                             end
@@ -656,35 +656,35 @@ begin
                 end
             end
         end
-        if (global_var(37) == 1) then begin
+        if (global_var(GVAR_KARMA_HOLY_WARRIOR) == 1) then begin
             Static_Reaction := Static_Reaction + 20;
         end
         else begin
-            if (global_var(38) == 1) then begin
+            if (global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES) == 1) then begin
                 Static_Reaction := Static_Reaction + 15;
             end
             else begin
-                if (global_var(39) == 1) then begin
+                if (global_var(GVAR_KARMA_SHIELD_OF_HOPE) == 1) then begin
                     Static_Reaction := Static_Reaction + 10;
                 end
                 else begin
-                    if (global_var(40) == 1) then begin
+                    if (global_var(GVAR_KARMA_DEFENDER) == 1) then begin
                         Static_Reaction := Static_Reaction + 5;
                     end
                     else begin
-                        if (global_var(41) == 1) then begin
+                        if (global_var(GVAR_KARMA_WANDERER) == 1) then begin
                             Static_Reaction := Static_Reaction + 0;
                         end
                         else begin
-                            if (global_var(42) == 1) then begin
+                            if (global_var(GVAR_KARMA_BETRAYER) == 1) then begin
                                 Static_Reaction := Static_Reaction + -5;
                             end
                             else begin
-                                if (global_var(43) == 1) then begin
+                                if (global_var(GVAR_KARMA_SWORD_OF_DESPAIR) == 1) then begin
                                     Static_Reaction := Static_Reaction + -10;
                                 end
                                 else begin
-                                    if (global_var(44) == 1) then begin
+                                    if (global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE) == 1) then begin
                                         Static_Reaction := Static_Reaction + -15;
                                     end
                                     else begin
@@ -700,28 +700,28 @@ begin
         debug_msg("Good Person Reaction Bonus == " + Static_Reaction);
     end
     if (Slavery_Tolerant == 0) then begin
-        if (global_var(11) == 1) then begin
+        if (global_var(GVAR_REPUTATION_SLAVER) == 1) then begin
             Static_Reaction := Static_Reaction + (2 * -25);
         end
     end
     else begin
         if (Slavery_Tolerant == 1) then begin
-            if (global_var(11) == 1) then begin
+            if (global_var(GVAR_REPUTATION_SLAVER) == 1) then begin
                 Static_Reaction := Static_Reaction + -25;
             end
         end
         else begin
             if (Slavery_Tolerant == 3) then begin
-                if (global_var(11) == 1) then begin
+                if (global_var(GVAR_REPUTATION_SLAVER) == 1) then begin
                     Static_Reaction := Static_Reaction - -25;
                 end
             end
         end
     end
     debug_msg("Slaver + Aligned Reaction == " + Static_Reaction);
-    Static_Reaction := Static_Reaction + (10 * has_trait(0, dude_obj, 10));
+    Static_Reaction := Static_Reaction + (10 * has_trait(TRAIT_PERK, dude_obj, PERK_presence));
     debug_msg("Presence Reaction == " + Static_Reaction);
-    if (global_var(1) >= 2) then begin
+    if (global_var(GVAR_CHILDKILLER_REPUTATION) >= 2) then begin
         Static_Reaction := Static_Reaction + -30;
     end
     debug_msg("Childkiller Reaction == " + Static_Reaction);
@@ -747,12 +747,12 @@ begin
             if (local_var(7) == 1) then begin
                 if (local_var(6)) then begin
                     if (obj_can_see_obj(self_obj, dude_obj)) then begin
-                        float_msg(self_obj, message_str(218, 202), 8);
+                        float_msg(self_obj, mstr(202), FLOAT_MSG_YELLOW);
                         set_local_var(5, 2);
                     end
                 end
                 else begin
-                    if (global_var(350) == 1) then begin
+                    if (global_var(GVAR_GRUTHAR_DSTATUS) == 1) then begin
                         start_gdialog(218, self_obj, 4, -1, -1);
                         gsay_start;
                         call Node015();
@@ -760,7 +760,7 @@ begin
                         end_dialogue;
                     end
                     else begin
-                        if (global_var(350) == 2) then begin
+                        if (global_var(GVAR_GRUTHAR_DSTATUS) == 2) then begin
                             start_gdialog(218, self_obj, 4, -1, -1);
                             gsay_start;
                             call Node017();
@@ -768,7 +768,7 @@ begin
                             end_dialogue;
                         end
                         else begin
-                            if (global_var(350) == 3) then begin
+                            if (global_var(GVAR_GRUTHAR_DSTATUS) == 3) then begin
                                 start_gdialog(218, self_obj, 4, -1, -1);
                                 gsay_start;
                                 call Node018();
@@ -776,7 +776,7 @@ begin
                                 end_dialogue;
                             end
                             else begin
-                                if (global_var(350) == 4) then begin
+                                if (global_var(GVAR_GRUTHAR_DSTATUS) == 4) then begin
                                     start_gdialog(218, self_obj, 4, -1, -1);
                                     gsay_start;
                                     call Node020();
@@ -784,7 +784,7 @@ begin
                                     end_dialogue;
                                 end
                                 else begin
-                                    if (global_var(350) == 5) then begin
+                                    if (global_var(GVAR_GRUTHAR_DSTATUS) == 5) then begin
                                         start_gdialog(218, self_obj, 4, -1, -1);
                                         gsay_start;
                                         call Node021();
@@ -792,7 +792,7 @@ begin
                                         end_dialogue;
                                     end
                                     else begin
-                                        if (global_var(350) == 6) then begin
+                                        if (global_var(GVAR_GRUTHAR_DSTATUS) == 6) then begin
                                             start_gdialog(218, self_obj, 4, -1, -1);
                                             gsay_start;
                                             call Node022();
@@ -800,11 +800,11 @@ begin
                                             end_dialogue;
                                         end
                                         else begin
-                                            if (global_var(350) == 7) then begin
+                                            if (global_var(GVAR_GRUTHAR_DSTATUS) == 7) then begin
                                                 call Node023();
                                             end
                                             else begin
-                                                if (global_var(350) == 8) then begin
+                                                if (global_var(GVAR_GRUTHAR_DSTATUS) == 8) then begin
                                                     call Node033();
                                                 end
                                                 else begin
@@ -835,59 +835,59 @@ end
 procedure destroy_p_proc
 begin
     if (source_obj == dude_obj) then begin
-        set_global_var(5, global_var(5) + 1);
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(4) > (3 * global_var(5))) or (global_var(2) == 1)) and (global_var(1) == 0)) then begin
-            set_global_var(2, 1);
-            set_global_var(3, 0);
+        set_global_var(GVAR_GOOD_MONSTER, global_var(GVAR_GOOD_MONSTER) + 1);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(4) > (3 * global_var(GVAR_GOOD_MONSTER))) or (global_var(GVAR_CHAMPION_REPUTATION) == 1)) and (global_var(GVAR_CHILDKILLER_REPUTATION) == 0)) then begin
+            set_global_var(GVAR_CHAMPION_REPUTATION, 1);
+            set_global_var(GVAR_BERSERKER_REPUTATION, 0);
         end
-        if (metarule(51, self_obj) == 2) then begin
+        if (critter_kill_type(self_obj) == KILL_TYPE_children_kills) then begin
             if (source_obj == dude_obj) then begin
-                set_global_var(2, 0);
-                set_global_var(1, global_var(1) + 1);
-                set_global_var(0, global_var(0) + -15);
+                set_global_var(GVAR_CHAMPION_REPUTATION, 0);
+                set_global_var(GVAR_CHILDKILLER_REPUTATION, global_var(GVAR_CHILDKILLER_REPUTATION) + 1);
+                set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + -15);
                 debug_msg("Player gains " + -15 + " Karma Points.");
-                set_global_var(37, 0);
-                set_global_var(38, 0);
-                set_global_var(39, 0);
-                set_global_var(40, 0);
-                set_global_var(41, 0);
-                set_global_var(42, 0);
-                set_global_var(43, 0);
-                set_global_var(44, 0);
-                set_global_var(45, 0);
-                if (global_var(0) >= 1000) then begin
-                    set_global_var(37, 1);
+                set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+                set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+                set_global_var(GVAR_KARMA_DEFENDER, 0);
+                set_global_var(GVAR_KARMA_WANDERER, 0);
+                set_global_var(GVAR_KARMA_BETRAYER, 0);
+                set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+                set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+                set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+                    set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
                 end
                 else begin
-                    if (global_var(0) >= 750) then begin
-                        set_global_var(38, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
                     end
                     else begin
-                        if (global_var(0) >= 500) then begin
-                            set_global_var(39, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                            set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                         end
                         else begin
-                            if (global_var(0) >= 250) then begin
-                                set_global_var(40, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                                set_global_var(GVAR_KARMA_DEFENDER, 1);
                             end
                             else begin
-                                if (global_var(0) > -250) then begin
-                                    set_global_var(41, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                                    set_global_var(GVAR_KARMA_WANDERER, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -500) then begin
-                                        set_global_var(42, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                        set_global_var(GVAR_KARMA_BETRAYER, 1);
                                     end
                                     else begin
-                                        if (global_var(0) > -750) then begin
-                                            set_global_var(43, 1);
+                                        if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                            set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                         end
                                         else begin
-                                            if (global_var(0) > -1000) then begin
-                                                set_global_var(44, 1);
+                                            if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                                set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                             end
                                             else begin
-                                                set_global_var(45, 1);
+                                                set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                             end
                                         end
                                     end
@@ -896,58 +896,58 @@ begin
                         end
                     end
                 end
-                set_global_var(59, global_var(59) + -8);
+                set_global_var(GVAR_TOWN_REP_VAULT_13, global_var(GVAR_TOWN_REP_VAULT_13) + -8);
                 debug_msg("Added " + -8 + " to Town Rep");
             end
         end
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(5) > (2 * global_var(4))) or (global_var(3) == 1))) then begin
-            set_global_var(3, 1);
-            set_global_var(2, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
+            set_global_var(GVAR_BERSERKER_REPUTATION, 1);
+            set_global_var(GVAR_CHAMPION_REPUTATION, 0);
         end
-        set_global_var(0, global_var(0) + -10);
+        set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + -10);
         debug_msg("Player gains " + -10 + " Karma Points.");
-        set_global_var(37, 0);
-        set_global_var(38, 0);
-        set_global_var(39, 0);
-        set_global_var(40, 0);
-        set_global_var(41, 0);
-        set_global_var(42, 0);
-        set_global_var(43, 0);
-        set_global_var(44, 0);
-        set_global_var(45, 0);
-        if (global_var(0) >= 1000) then begin
-            set_global_var(37, 1);
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+        set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+        set_global_var(GVAR_KARMA_DEFENDER, 0);
+        set_global_var(GVAR_KARMA_WANDERER, 0);
+        set_global_var(GVAR_KARMA_BETRAYER, 0);
+        set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+        set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+            set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
         end
         else begin
-            if (global_var(0) >= 750) then begin
-                set_global_var(38, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
             end
             else begin
-                if (global_var(0) >= 500) then begin
-                    set_global_var(39, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                 end
                 else begin
-                    if (global_var(0) >= 250) then begin
-                        set_global_var(40, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                        set_global_var(GVAR_KARMA_DEFENDER, 1);
                     end
                     else begin
-                        if (global_var(0) > -250) then begin
-                            set_global_var(41, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                            set_global_var(GVAR_KARMA_WANDERER, 1);
                         end
                         else begin
-                            if (global_var(0) > -500) then begin
-                                set_global_var(42, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                set_global_var(GVAR_KARMA_BETRAYER, 1);
                             end
                             else begin
-                                if (global_var(0) > -750) then begin
-                                    set_global_var(43, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -1000) then begin
-                                        set_global_var(44, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                     end
                                     else begin
-                                        set_global_var(45, 1);
+                                        set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                     end
                                 end
                             end
@@ -956,28 +956,28 @@ begin
                 end
             end
         end
-        set_global_var(59, global_var(59) + -5);
+        set_global_var(GVAR_TOWN_REP_VAULT_13, global_var(GVAR_TOWN_REP_VAULT_13) + -5);
         debug_msg("Added " + -5 + " to Town Rep");
     end
     i_gruthar_obj := -1;
-    set_global_var(429, global_var(429) bwor 2);
+    set_global_var(GVAR_V13_STATUS_FLAGS, global_var(GVAR_V13_STATUS_FLAGS) bwor 2);
 end
 
 procedure look_at_p_proc
 begin
     script_overrides;
     if (local_var(7) == 0) then begin
-        display_msg(message_str(218, 100));
+        display_mstr(100);
     end
     else begin
-        display_msg(message_str(218, 101));
+        display_mstr(101);
     end
 end
 
 procedure description_p_proc
 begin
     script_overrides;
-    display_msg(message_str(218, 102));
+    display_mstr(102);
 end
 
 procedure use_skill_on_p_proc
@@ -998,8 +998,8 @@ end
 procedure map_enter_p_proc
 begin
     Only_Once := 0;
-    critter_add_trait(self_obj, 1, 6, 114);
-    critter_add_trait(self_obj, 1, 5, 29);
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 114);
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_VAULT_DEATHCLAWS);
     if (map_var(4) == 0) then begin
         set_obj_visibility(self_obj, 1);
     end
@@ -1027,78 +1027,78 @@ procedure Node001
 begin
     if (local_var(7) == 0) then begin
         set_local_var(7, 1);
-        gsay_reply(218, 103);
+        Reply(103);
     end
     else begin
-        gsay_reply(218, 104);
+        Reply(104);
     end
-    if (global_var(350) != 4) then begin
-        giq_option(-3, 218, 105, Node002, 50);
+    if (global_var(GVAR_GRUTHAR_DSTATUS) != 4) then begin
+        giq_option(-3, NAME, 105, Node002, NEUTRAL_REACTION);
     end
     if (local_var(9) == 0) then begin
-        giq_option(4, 218, 106, Node025a, 50);
+        giq_option(4, NAME, 106, Node025a, NEUTRAL_REACTION);
     end
     if (local_var(8) == 0) then begin
-        giq_option(4, 218, 107, Node003a, 50);
+        giq_option(4, NAME, 107, Node003a, NEUTRAL_REACTION);
     end
     if (local_var(10) == 0) then begin
-        giq_option(4, 218, 108, Node005a, 50);
+        giq_option(4, NAME, 108, Node005a, NEUTRAL_REACTION);
     end
-    giq_option(4, 218, 109, Node006a, 50);
-    if (global_var(186)) then begin
-        giq_option(4, 218, 110, Node011, 50);
+    giq_option(4, NAME, 109, Node006a, NEUTRAL_REACTION);
+    if (global_var(GVAR_NCR_DEATHCLAW_DEN)) then begin
+        giq_option(4, NAME, 110, Node011, NEUTRAL_REACTION);
     end
-    if (global_var(185) and (global_var(186) == 0)) then begin
-        giq_option(4, 218, 111, Node013, 50);
+    if (global_var(GVAR_NCR_BRAHMIN_PROTECT) and (global_var(GVAR_NCR_DEATHCLAW_DEN) == 0)) then begin
+        giq_option(4, NAME, 111, Node013, NEUTRAL_REACTION);
     end
-    giq_option(4, 218, 112, Node031, 50);
+    giq_option(4, NAME, 112, Node031, NEUTRAL_REACTION);
 end
 
 procedure Node002
 begin
-    if ((get_critter_stat(dude_obj, 4) > 3) and (local_var(12) == 0)) then begin
-        gsay_reply(218, 113);
+    if ((dude_iq > 3) and (local_var(12) == 0)) then begin
+        Reply(113);
         set_local_var(12, 1);
-        giq_option(0, 218, message_str(14, 352), Node001, 50);
+        giq_option(0, NAME, g_mstr(352), Node001, NEUTRAL_REACTION);
     end
     else begin
-        if ((get_critter_stat(dude_obj, 4) > 3) and (local_var(12) == 1)) then begin
-            gsay_reply(218, 114);
-            set_global_var(350, 4);
-            giq_option(0, 218, message_str(14, 352), Node030, 50);
+        if ((dude_iq > 3) and (local_var(12) == 1)) then begin
+            Reply(114);
+            set_global_var(GVAR_GRUTHAR_DSTATUS, 4);
+            giq_option(0, NAME, g_mstr(352), Node030, NEUTRAL_REACTION);
         end
         else begin
-            set_global_var(350, 8);
-            gsay_reply(218, 115);
-            giq_option(-3, 218, 234, Node999, 50);
+            set_global_var(GVAR_GRUTHAR_DSTATUS, 8);
+            Reply(115);
+            giq_option(-3, NAME, 234, Node999, NEUTRAL_REACTION);
         end
     end
 end
 
 procedure Node003
 begin
-    gsay_reply(218, 116);
-    giq_option(4, 218, 117, Node001, 50);
-    giq_option(4, 218, 118, Node004, 50);
-    giq_option(4, 218, 119, Node026, 51);
+    Reply(116);
+    giq_option(4, NAME, 117, Node001, NEUTRAL_REACTION);
+    giq_option(4, NAME, 118, Node004, NEUTRAL_REACTION);
+    giq_option(4, NAME, 119, Node026, BAD_REACTION);
 end
 
 procedure Node004
 begin
-    set_global_var(429, global_var(429) bwor 4);
-    set_global_var(488, 1);
-    gsay_reply(218, 120);
-    giq_option(4, 218, 121, Node001, 49);
-    giq_option(4, 218, 122, Node030, 51);
-    giq_option(4, 218, 123, Node001, 50);
+    set_global_var(GVAR_V13_STATUS_FLAGS, global_var(GVAR_V13_STATUS_FLAGS) bwor 4);
+    set_global_var(GVAR_V13_GORIS_QST, 1);
+    Reply(120);
+    giq_option(4, NAME, 121, Node001, GOOD_REACTION);
+    giq_option(4, NAME, 122, Node030, BAD_REACTION);
+    giq_option(4, NAME, 123, Node001, NEUTRAL_REACTION);
 end
 
 procedure Node005
 begin
-    gsay_reply(218, 124);
-    giq_option(4, 218, 125, Node025, 51);
-    giq_option(4, 218, 126, Node001, 49);
-    giq_option(4, 218, 127, Node001, 50);
+    Reply(124);
+    giq_option(4, NAME, 125, Node025, BAD_REACTION);
+    giq_option(4, NAME, 126, Node001, GOOD_REACTION);
+    giq_option(4, NAME, 127, Node001, NEUTRAL_REACTION);
 end
 
 procedure Node006
@@ -1106,180 +1106,180 @@ begin
     variable LVar0 := 0;
     if (local_var(13) == 0) then begin
         set_local_var(13, 1);
-        gsay_reply(218, 128);
+        Reply(128);
         LVar0 := 1;
     end
     else begin
-        gsay_reply(218, 129);
+        Reply(129);
     end
-    giq_option(4, 218, 130, Node007, 50);
+    giq_option(4, NAME, 130, Node007, NEUTRAL_REACTION);
     if (LVar0 == 1) then begin
-        giq_option(4, 218, 131, Node027, 51);
+        giq_option(4, NAME, 131, Node027, BAD_REACTION);
     end
-    giq_option(4, 218, 132, Node030, 51);
-    giq_option(4, 218, 133, Node007a, 49);
+    giq_option(4, NAME, 132, Node030, BAD_REACTION);
+    giq_option(4, NAME, 133, Node007a, GOOD_REACTION);
 end
 
 procedure Node007
 begin
     set_local_var(15, 1);
     if (nice) then begin
-        gsay_reply(218, 134);
+        Reply(134);
     end
     else begin
-        gsay_reply(218, 135);
+        Reply(135);
     end
-    giq_option(4, 218, 136, Node028, 51);
-    giq_option(4, 218, 137, Node001, 50);
-    giq_option(4, 218, 138, Node008a, 49);
+    giq_option(4, NAME, 136, Node028, BAD_REACTION);
+    giq_option(4, NAME, 137, Node001, NEUTRAL_REACTION);
+    giq_option(4, NAME, 138, Node008a, GOOD_REACTION);
 end
 
 procedure Node008
 begin
-    if (not((local_var(1) <= -2) or (global_var(0) < 0) or (global_var(59) < 0) or (global_var(1) >= 2) or (((global_var(4) + global_var(5)) >= 25) and ((global_var(5) > (2 * global_var(4))) or (global_var(3) == 1))))) then begin
-        gsay_reply(218, 139);
+    if (not((local_var(1) <= -2) or (global_var(GVAR_PLAYER_REPUTATION) < 0) or (global_var(GVAR_TOWN_REP_VAULT_13) < 0) or (global_var(GVAR_CHILDKILLER_REPUTATION) >= 2) or (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))))) then begin
+        Reply(139);
     end
     else begin
-        gsay_reply(218, 140);
+        Reply(140);
     end
-    if (not((local_var(1) <= -2) or (global_var(0) < 0) or (global_var(59) < 0) or (global_var(1) >= 2) or (((global_var(4) + global_var(5)) >= 25) and ((global_var(5) > (2 * global_var(4))) or (global_var(3) == 1))))) then begin
-        giq_option(4, 218, 141, Node009, 50);
+    if (not((local_var(1) <= -2) or (global_var(GVAR_PLAYER_REPUTATION) < 0) or (global_var(GVAR_TOWN_REP_VAULT_13) < 0) or (global_var(GVAR_CHILDKILLER_REPUTATION) >= 2) or (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))))) then begin
+        giq_option(4, NAME, 141, Node009, NEUTRAL_REACTION);
     end
     else begin
-        giq_option(4, 218, 142, Node998, 51);
-        giq_option(4, 218, 143, Node001, 50);
-        giq_option(4, 218, 144, Node030, 49);
+        giq_option(4, NAME, 142, Node998, BAD_REACTION);
+        giq_option(4, NAME, 143, Node001, NEUTRAL_REACTION);
+        giq_option(4, NAME, 144, Node030, GOOD_REACTION);
     end
 end
 
 procedure Node009
 begin
-    gsay_reply(218, 145);
-    if (has_skill(dude_obj, 13) < 40) then begin
-        giq_option(4, 218, 146, Node014a, 50);
+    Reply(145);
+    if (dude_skill(SKILL_REPAIR) < 40) then begin
+        giq_option(4, NAME, 146, Node014a, NEUTRAL_REACTION);
     end
-    if ((has_skill(dude_obj, 13) > 39) and (has_skill(dude_obj, 13) < 60)) then begin
-        giq_option(4, 218, 147, Node010a, 50);
+    if ((dude_skill(SKILL_REPAIR) > 39) and (dude_skill(SKILL_REPAIR) < 60)) then begin
+        giq_option(4, NAME, 147, Node010a, NEUTRAL_REACTION);
     end
-    if (has_skill(dude_obj, 13) > 59) then begin
-        giq_option(4, 218, 148, Node010a, 50);
+    if (dude_skill(SKILL_REPAIR) > 59) then begin
+        giq_option(4, NAME, 148, Node010a, NEUTRAL_REACTION);
     end
-    giq_option(4, 218, 149, Node031a, 50);
-    giq_option(4, 218, 150, Node010a, 50);
-    giq_option(4, 218, 151, Node014a, 50);
+    giq_option(4, NAME, 149, Node031a, NEUTRAL_REACTION);
+    giq_option(4, NAME, 150, Node010a, NEUTRAL_REACTION);
+    giq_option(4, NAME, 151, Node014a, NEUTRAL_REACTION);
 end
 
 procedure Node010
 begin
-    gsay_reply(218, 152);
-    giq_option(4, 218, 153, Node999, 49);
-    giq_option(4, 218, 154, Node999, 50);
-    giq_option(4, 218, 155, Node999, 51);
+    Reply(152);
+    giq_option(4, NAME, 153, Node999, GOOD_REACTION);
+    giq_option(4, NAME, 154, Node999, NEUTRAL_REACTION);
+    giq_option(4, NAME, 155, Node999, BAD_REACTION);
 end
 
 procedure Node011
 begin
-    gsay_reply(218, 156);
-    giq_option(4, 218, 157, Node001, 50);
-    giq_option(4, 218, 158, Node012, 51);
-    giq_option(4, 218, 159, Node001, 49);
+    Reply(156);
+    giq_option(4, NAME, 157, Node001, NEUTRAL_REACTION);
+    giq_option(4, NAME, 158, Node012, BAD_REACTION);
+    giq_option(4, NAME, 159, Node001, GOOD_REACTION);
 end
 
 procedure Node012
 begin
-    gsay_reply(218, 160);
-    giq_option(4, 218, 161, Node998, 51);
-    giq_option(4, 218, 162, Node001, 49);
-    giq_option(4, 218, 163, Node001, 50);
+    Reply(160);
+    giq_option(4, NAME, 161, Node998, BAD_REACTION);
+    giq_option(4, NAME, 162, Node001, GOOD_REACTION);
+    giq_option(4, NAME, 163, Node001, NEUTRAL_REACTION);
 end
 
 procedure Node013
 begin
-    gsay_reply(218, 164);
-    giq_option(4, 218, 165, Node998, 51);
-    giq_option(4, 218, 166, Node008, 49);
-    giq_option(4, 218, 167, Node031, 50);
-    giq_option(4, 218, 168, Node001, 50);
+    Reply(164);
+    giq_option(4, NAME, 165, Node998, BAD_REACTION);
+    giq_option(4, NAME, 166, Node008, GOOD_REACTION);
+    giq_option(4, NAME, 167, Node031, NEUTRAL_REACTION);
+    giq_option(4, NAME, 168, Node001, NEUTRAL_REACTION);
 end
 
 procedure Node014
 begin
-    gsay_reply(218, 169);
-    giq_option(4, 218, 170, Node031, 50);
+    Reply(169);
+    giq_option(4, NAME, 170, Node031, NEUTRAL_REACTION);
 end
 
 procedure Node015
 begin
-    gsay_reply(218, message_str(218, 171) + " " + obj_name(dude_obj) + message_str(218, 231));
-    giq_option(4, 218, 172, Node999, 50);
-    giq_option(4, 218, 173, Node016a, 50);
+    Reply(mstr(171) + " " + dude_name + mstr(231));
+    giq_option(4, NAME, 172, Node999, NEUTRAL_REACTION);
+    giq_option(4, NAME, 173, Node016a, NEUTRAL_REACTION);
 end
 
 procedure Node016
 begin
-    gsay_reply(218, message_str(218, 174) + " " + obj_name(dude_obj) + message_str(218, 232));
-    giq_option(4, 218, 175, Node999, 50);
-    giq_option(4, 218, 176, Node999, 49);
+    Reply(mstr(174) + " " + dude_name + mstr(232));
+    giq_option(4, NAME, 175, Node999, NEUTRAL_REACTION);
+    giq_option(4, NAME, 176, Node999, GOOD_REACTION);
 end
 
 procedure Node017
 begin
-    gsay_reply(218, message_str(218, 177) + " " + obj_name(dude_obj) + message_str(218, 233));
-    giq_option(4, 218, 178, Node999, 50);
+    Reply(mstr(177) + " " + dude_name + mstr(233));
+    giq_option(4, NAME, 178, Node999, NEUTRAL_REACTION);
     if (obj_is_carrying_obj_pid(dude_obj, 356) > 0) then begin
-        giq_option(4, 218, 179, Node029, 49);
+        giq_option(4, NAME, 179, Node029, GOOD_REACTION);
     end
 end
 
 procedure Node018
 begin
-    set_global_var(350, 7);
+    set_global_var(GVAR_GRUTHAR_DSTATUS, 7);
     add_timer_event(dude_obj, 0, 4);
-    set_global_var(487, 2);
-    set_global_var(0, global_var(0) + 10);
+    set_global_var(GVAR_V13_COMP_QST, 2);
+    set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + 10);
     debug_msg("Player gains " + 10 + " Karma Points.");
-    set_global_var(37, 0);
-    set_global_var(38, 0);
-    set_global_var(39, 0);
-    set_global_var(40, 0);
-    set_global_var(41, 0);
-    set_global_var(42, 0);
-    set_global_var(43, 0);
-    set_global_var(44, 0);
-    set_global_var(45, 0);
-    if (global_var(0) >= 1000) then begin
-        set_global_var(37, 1);
+    set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+    set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+    set_global_var(GVAR_KARMA_DEFENDER, 0);
+    set_global_var(GVAR_KARMA_WANDERER, 0);
+    set_global_var(GVAR_KARMA_BETRAYER, 0);
+    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+    set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+    set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+    if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
     end
     else begin
-        if (global_var(0) >= 750) then begin
-            set_global_var(38, 1);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+            set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
         end
         else begin
-            if (global_var(0) >= 500) then begin
-                set_global_var(39, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
             end
             else begin
-                if (global_var(0) >= 250) then begin
-                    set_global_var(40, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                    set_global_var(GVAR_KARMA_DEFENDER, 1);
                 end
                 else begin
-                    if (global_var(0) > -250) then begin
-                        set_global_var(41, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                        set_global_var(GVAR_KARMA_WANDERER, 1);
                     end
                     else begin
-                        if (global_var(0) > -500) then begin
-                            set_global_var(42, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                            set_global_var(GVAR_KARMA_BETRAYER, 1);
                         end
                         else begin
-                            if (global_var(0) > -750) then begin
-                                set_global_var(43, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                             end
                             else begin
-                                if (global_var(0) > -1000) then begin
-                                    set_global_var(44, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                    set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                 end
                                 else begin
-                                    set_global_var(45, 1);
+                                    set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                 end
                             end
                         end
@@ -1288,136 +1288,136 @@ begin
             end
         end
     end
-    set_global_var(59, global_var(59) + 30);
+    set_global_var(GVAR_TOWN_REP_VAULT_13, global_var(GVAR_TOWN_REP_VAULT_13) + 30);
     give_exp_points(5000);
-    display_msg(message_str(14, 100) + 5000 + message_str(14, 101));
-    gsay_reply(218, 181);
+    display_msg(g_mstr(100) + 5000 + g_mstr(101));
+    Reply(181);
     if (local_var(15) == 1) then begin
-        giq_option(4, 218, 182, Node019, 50);
+        giq_option(4, NAME, 182, Node019, NEUTRAL_REACTION);
     end
     else begin
-        giq_option(4, 218, 183, Node019, 50);
+        giq_option(4, NAME, 183, Node019, NEUTRAL_REACTION);
     end
     if (checkGeck() == 0) then begin
-        giq_option(4, 218, 184, Node999, 50);
+        giq_option(4, NAME, 184, Node999, NEUTRAL_REACTION);
     end
 end
 
 procedure Node019
 begin
     if (checkGeck()) then begin
-        gsay_reply(218, 185);
+        Reply(185);
     end
     else begin
-        gsay_reply(218, 186);
+        Reply(186);
     end
-    giq_option(4, 218, 187, Node999a, 50);
+    giq_option(4, NAME, 187, Node999a, NEUTRAL_REACTION);
     if (checkGeck() == 0) then begin
-        giq_option(4, 218, 188, Node999, 50);
+        giq_option(4, NAME, 188, Node999, NEUTRAL_REACTION);
     end
 end
 
 procedure Node020
 begin
-    gsay_reply(218, 189);
-    giq_option(4, 218, 190, Node001, 49);
-    giq_option(4, 218, 191, Node031, 50);
-    giq_option(-3, 218, 192, Node032, 50);
+    Reply(189);
+    giq_option(4, NAME, 190, Node001, GOOD_REACTION);
+    giq_option(4, NAME, 191, Node031, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 192, Node032, NEUTRAL_REACTION);
 end
 
 procedure Node021
 begin
-    gsay_reply(218, 193);
-    giq_option(4, 218, 194, Node030, 51);
-    giq_option(4, 218, 195, Node030, 50);
-    giq_option(4, 218, 196, Node010a, 49);
+    Reply(193);
+    giq_option(4, NAME, 194, Node030, BAD_REACTION);
+    giq_option(4, NAME, 195, Node030, NEUTRAL_REACTION);
+    giq_option(4, NAME, 196, Node010a, GOOD_REACTION);
 end
 
 procedure Node022
 begin
-    gsay_reply(218, 197);
-    giq_option(4, 218, 198, Node999, 50);
+    Reply(197);
+    giq_option(4, NAME, 198, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node023
 begin
-    float_msg(self_obj, message_str(218, 199), 3);
+    float_msg(self_obj, mstr(199), FLOAT_MSG_GREEN);
 end
 
 procedure Node024
 begin
-    float_msg(self_obj, message_str(218, 202), 8);
+    float_msg(self_obj, mstr(202), FLOAT_MSG_YELLOW);
 end
 
 procedure Node025
 begin
-    gsay_reply(218, 205);
-    giq_option(4, 218, 206, Node001, 50);
-    giq_option(4, 218, 207, Node998, 51);
-    giq_option(4, 218, 208, Node001, 49);
-    giq_option(4, 218, 209, Node030, 50);
+    Reply(205);
+    giq_option(4, NAME, 206, Node001, NEUTRAL_REACTION);
+    giq_option(4, NAME, 207, Node998, BAD_REACTION);
+    giq_option(4, NAME, 208, Node001, GOOD_REACTION);
+    giq_option(4, NAME, 209, Node030, NEUTRAL_REACTION);
 end
 
 procedure Node026
 begin
-    gsay_reply(218, 210);
-    giq_option(4, 218, 211, Node998, 51);
-    giq_option(4, 218, 212, NodeTeleport, 50);
-    giq_option(4, 218, 213, Node001, 51);
+    Reply(210);
+    giq_option(4, NAME, 211, Node998, BAD_REACTION);
+    giq_option(4, NAME, 212, NodeTeleport, NEUTRAL_REACTION);
+    giq_option(4, NAME, 213, Node001, BAD_REACTION);
 end
 
 procedure Node027
 begin
-    gsay_reply(218, 214);
-    giq_option(4, 218, 215, Node001, 50);
-    giq_option(4, 218, 216, Node001, 49);
-    giq_option(4, 218, 217, Node001, 51);
+    Reply(214);
+    giq_option(4, NAME, 215, Node001, NEUTRAL_REACTION);
+    giq_option(4, NAME, 216, Node001, GOOD_REACTION);
+    giq_option(4, NAME, 217, Node001, BAD_REACTION);
 end
 
 procedure Node028
 begin
-    gsay_reply(218, 218);
-    giq_option(4, 218, 219, Node001, 49);
-    giq_option(4, 218, 220, Node998, 51);
-    giq_option(4, 218, 221, Node030, 50);
+    Reply(218);
+    giq_option(4, NAME, 219, Node001, GOOD_REACTION);
+    giq_option(4, NAME, 220, Node998, BAD_REACTION);
+    giq_option(4, NAME, 221, Node030, NEUTRAL_REACTION);
 end
 
 procedure Node029
 begin
-    gsay_reply(218, 222);
-    giq_option(4, 218, 223, Node999, 50);
+    Reply(222);
+    giq_option(4, NAME, 223, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node030
 begin
-    gsay_reply(218, 225);
-    giq_option(0, 218, message_str(14, 350), NodeTeleport, 50);
+    Reply(225);
+    giq_option(0, NAME, g_mstr(350), NodeTeleport, NEUTRAL_REACTION);
 end
 
 procedure Node031
 begin
-    gsay_reply(218, 226);
-    giq_option(0, 218, 227, NodeTeleport, 50);
+    Reply(226);
+    giq_option(0, NAME, 227, NodeTeleport, NEUTRAL_REACTION);
 end
 
 procedure Node032
 begin
-    gsay_reply(218, 229);
-    giq_option(0, 218, 230, NodeTeleport, 50);
+    Reply(229);
+    giq_option(0, NAME, 230, NodeTeleport, NEUTRAL_REACTION);
 end
 
 procedure Node033
 begin
-    float_msg(self_obj, message_str(218, 235), 0);
+    float_msg(self_obj, mstr(235), FLOAT_MSG_NORMAL);
 end
 
 procedure Node034
 begin
-    gsay_reply(218, 236);
-    giq_option(4, 218, 220, Node998, 51);
-    giq_option(-3, 218, 237, Node998, 51);
-    giq_option(4, 218, 238, Node999, 49);
-    giq_option(-3, 218, 239, Node999, 49);
+    Reply(236);
+    giq_option(4, NAME, 220, Node998, BAD_REACTION);
+    giq_option(-3, NAME, 237, Node998, BAD_REACTION);
+    giq_option(4, NAME, 238, Node999, GOOD_REACTION);
+    giq_option(-3, NAME, 239, Node999, GOOD_REACTION);
 end
 
 procedure checkGeck
@@ -1516,26 +1516,26 @@ end
 
 procedure Node014a
 begin
-    set_global_var(350, 5);
+    set_global_var(GVAR_GRUTHAR_DSTATUS, 5);
     call Node014();
 end
 
 procedure Node010a
 begin
-    set_global_var(350, 6);
-    set_global_var(487, 1);
+    set_global_var(GVAR_GRUTHAR_DSTATUS, 6);
+    set_global_var(GVAR_V13_COMP_QST, 1);
     call Node010();
 end
 
 procedure Node031a
 begin
-    set_global_var(350, 5);
+    set_global_var(GVAR_GRUTHAR_DSTATUS, 5);
     call Node030();
 end
 
 procedure Node016a
 begin
-    set_global_var(350, 2);
+    set_global_var(GVAR_GRUTHAR_DSTATUS, 2);
     call Node016();
 end
 
@@ -1543,9 +1543,8 @@ procedure Node999a
 begin
     variable LVar0 := 0;
     set_map_var(1, 1);
-    set_global_var(233, 3);
+    set_global_var(GVAR_VAULT13_FOUND_GECK, 3);
     LVar0 := checkGeck();
     rm_obj_from_inven(geck_container, LVar0);
     add_obj_to_inven(dude_obj, LVar0);
 end
-
