diff --git a/Games/Fallout2.Steam/Original/OCJIMMY.ssl b/Games/Fallout2.Steam/ReDefine/OCJIMMY.ssl
index aa6cbdd..bb9f25e 100644
--- a/Games/Fallout2.Steam/Original/OCJIMMY.ssl
+++ b/Games/Fallout2.Steam/ReDefine/OCJIMMY.ssl
@@ -260,24 +260,24 @@ procedure pickup_p_proc
 begin
     if (source_obj == dude_obj) then begin
         if (3 == 1) then begin
-            attack_complex(source_obj, 0, 1, 0, 0, 30000, 0, 0);
+            attack(source_obj);
         end
         else begin
             if (3 == 2) then begin
-                float_msg(self_obj, message_str(14, random(3325, 3329)), 0);
+                float_msg(self_obj, g_mstr(random(3325, 3329)), FLOAT_MSG_NORMAL);
                 if (7 > -1) then begin
                     if (local_var(7) >= 1) then begin
-                        attack_complex(source_obj, 0, 1, 0, 0, 30000, 0, 0);
+                        attack(source_obj);
                     end
                     set_local_var(7, local_var(7) + 1);
                 end
             end
             else begin
                 if (3 == 3) then begin
-                    float_msg(self_obj, message_str(14, random(3050, 3064)), 0);
+                    float_msg(self_obj, g_mstr(random(3050, 3064)), FLOAT_MSG_NORMAL);
                     if (7 > -1) then begin
                         if (local_var(7) >= 1) then begin
-                            attack_complex(source_obj, 0, 1, 0, 0, 30000, 0, 0);
+                            attack(source_obj);
                         end
                         set_local_var(7, local_var(7) + 1);
                     end
@@ -599,8 +599,8 @@ end
 
 procedure start
 begin
-    if ((global_var(582) bwand 1) != 0) then begin
-        global_temp := create_object_sid(128, 0, 0, -1);
+    if ((global_var(GVAR_VIGNETTE_SEQUENCE) bwand 1) != 0) then begin
+        global_temp := create_object(128, 0, 0);
         move_obj_inven_to_obj(self_obj, global_temp);
         destroy_object(global_temp);
         kill_critter(self_obj, 62);
@@ -612,7 +612,7 @@ begin
     if ((local_var(5) == 2) or map_var(5) and obj_can_see_obj(self_obj, dude_obj)) then begin
         set_local_var(5, 1);
         set_map_var(5, 1);
-        attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+        attack(dude_obj);
     end
 end
 
@@ -621,48 +621,48 @@ begin
     Evil_Critter := 0;
     Slavery_Tolerant := 2;
     Karma_Perception := 10;
-    set_global_var(37, 0);
-    set_global_var(38, 0);
-    set_global_var(39, 0);
-    set_global_var(40, 0);
-    set_global_var(41, 0);
-    set_global_var(42, 0);
-    set_global_var(43, 0);
-    set_global_var(44, 0);
-    set_global_var(45, 0);
-    if (global_var(0) >= 1000) then begin
-        set_global_var(37, 1);
+    set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+    set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+    set_global_var(GVAR_KARMA_DEFENDER, 0);
+    set_global_var(GVAR_KARMA_WANDERER, 0);
+    set_global_var(GVAR_KARMA_BETRAYER, 0);
+    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+    set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+    set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+    if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
     end
     else begin
-        if (global_var(0) >= 750) then begin
-            set_global_var(38, 1);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+            set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
         end
         else begin
-            if (global_var(0) >= 500) then begin
-                set_global_var(39, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
             end
             else begin
-                if (global_var(0) >= 250) then begin
-                    set_global_var(40, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                    set_global_var(GVAR_KARMA_DEFENDER, 1);
                 end
                 else begin
-                    if (global_var(0) > -250) then begin
-                        set_global_var(41, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                        set_global_var(GVAR_KARMA_WANDERER, 1);
                     end
                     else begin
-                        if (global_var(0) > -500) then begin
-                            set_global_var(42, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                            set_global_var(GVAR_KARMA_BETRAYER, 1);
                         end
                         else begin
-                            if (global_var(0) > -750) then begin
-                                set_global_var(43, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                             end
                             else begin
-                                if (global_var(0) > -1000) then begin
-                                    set_global_var(44, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                    set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                 end
                                 else begin
-                                    set_global_var(45, 1);
+                                    set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                 end
                             end
                         end
@@ -673,7 +673,7 @@ begin
     end
     if (local_var(2) == 0) then begin
         set_local_var(2, 1);
-        set_local_var(3, (get_critter_stat(dude_obj, 3) - get_critter_stat(self_obj, 3)) * 5);
+        set_local_var(3, (dude_charisma - self_charisma) * 5);
         debug_msg("Base Reaction == " + local_var(3));
     end
     Static_Reaction := 0;
@@ -685,14 +685,14 @@ begin
     else begin
         reaction_bonus_karma := -1 * (-1 * reaction_bonus_karma / 1000);
     end
-    if (global_var(59) >= 0) then begin
-        reaction_bonus_town_rep := global_var(59) / 2;
+    if (global_var(GVAR_TOWN_REP_VAULT_13) >= 0) then begin
+        reaction_bonus_town_rep := global_var(GVAR_TOWN_REP_VAULT_13) / 2;
     end
     else begin
-        reaction_bonus_town_rep := -1 * (-1 * global_var(59) / 2);
+        reaction_bonus_town_rep := -1 * (-1 * global_var(GVAR_TOWN_REP_VAULT_13) / 2);
     end
-    if (has_trait(2, dude_obj, 13)) then begin
-        if (get_critter_stat(dude_obj, 34) == get_critter_stat(self_obj, 34)) then begin
+    if (has_trait(TRAIT_TRAIT, dude_obj, TRAIT_sex_appeal)) then begin
+        if (dude_gender == self_gender) then begin
             Static_Reaction := Static_Reaction + -20;
         end
         else begin
@@ -708,48 +708,48 @@ begin
         debug_msg("Karma Reaction Bonus == " + Static_Reaction);
         Static_Reaction := Static_Reaction - reaction_bonus_town_rep;
         debug_msg("Town Rep Bonus == " + Static_Reaction);
-        set_global_var(37, 0);
-        set_global_var(38, 0);
-        set_global_var(39, 0);
-        set_global_var(40, 0);
-        set_global_var(41, 0);
-        set_global_var(42, 0);
-        set_global_var(43, 0);
-        set_global_var(44, 0);
-        set_global_var(45, 0);
-        if (global_var(0) >= 1000) then begin
-            set_global_var(37, 1);
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+        set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+        set_global_var(GVAR_KARMA_DEFENDER, 0);
+        set_global_var(GVAR_KARMA_WANDERER, 0);
+        set_global_var(GVAR_KARMA_BETRAYER, 0);
+        set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+        set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+            set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
         end
         else begin
-            if (global_var(0) >= 750) then begin
-                set_global_var(38, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
             end
             else begin
-                if (global_var(0) >= 500) then begin
-                    set_global_var(39, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                 end
                 else begin
-                    if (global_var(0) >= 250) then begin
-                        set_global_var(40, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                        set_global_var(GVAR_KARMA_DEFENDER, 1);
                     end
                     else begin
-                        if (global_var(0) > -250) then begin
-                            set_global_var(41, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                            set_global_var(GVAR_KARMA_WANDERER, 1);
                         end
                         else begin
-                            if (global_var(0) > -500) then begin
-                                set_global_var(42, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                set_global_var(GVAR_KARMA_BETRAYER, 1);
                             end
                             else begin
-                                if (global_var(0) > -750) then begin
-                                    set_global_var(43, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -1000) then begin
-                                        set_global_var(44, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                     end
                                     else begin
-                                        set_global_var(45, 1);
+                                        set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                     end
                                 end
                             end
@@ -758,35 +758,35 @@ begin
                 end
             end
         end
-        if (global_var(37) == 1) then begin
+        if (global_var(GVAR_KARMA_HOLY_WARRIOR) == 1) then begin
             Static_Reaction := Static_Reaction - 20;
         end
         else begin
-            if (global_var(38) == 1) then begin
+            if (global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES) == 1) then begin
                 Static_Reaction := Static_Reaction - 15;
             end
             else begin
-                if (global_var(39) == 1) then begin
+                if (global_var(GVAR_KARMA_SHIELD_OF_HOPE) == 1) then begin
                     Static_Reaction := Static_Reaction - 10;
                 end
                 else begin
-                    if (global_var(40) == 1) then begin
+                    if (global_var(GVAR_KARMA_DEFENDER) == 1) then begin
                         Static_Reaction := Static_Reaction - 5;
                     end
                     else begin
-                        if (global_var(41) == 1) then begin
+                        if (global_var(GVAR_KARMA_WANDERER) == 1) then begin
                             Static_Reaction := Static_Reaction - 0;
                         end
                         else begin
-                            if (global_var(42) == 1) then begin
+                            if (global_var(GVAR_KARMA_BETRAYER) == 1) then begin
                                 Static_Reaction := Static_Reaction - -5;
                             end
                             else begin
-                                if (global_var(43) == 1) then begin
+                                if (global_var(GVAR_KARMA_SWORD_OF_DESPAIR) == 1) then begin
                                     Static_Reaction := Static_Reaction - -10;
                                 end
                                 else begin
-                                    if (global_var(44) == 1) then begin
+                                    if (global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE) == 1) then begin
                                         Static_Reaction := Static_Reaction - -15;
                                     end
                                     else begin
@@ -806,48 +806,48 @@ begin
         debug_msg("Karma Reaction Bonus == " + Static_Reaction);
         Static_Reaction := Static_Reaction + reaction_bonus_town_rep;
         debug_msg("Town Rep Bonus == " + Static_Reaction);
-        set_global_var(37, 0);
-        set_global_var(38, 0);
-        set_global_var(39, 0);
-        set_global_var(40, 0);
-        set_global_var(41, 0);
-        set_global_var(42, 0);
-        set_global_var(43, 0);
-        set_global_var(44, 0);
-        set_global_var(45, 0);
-        if (global_var(0) >= 1000) then begin
-            set_global_var(37, 1);
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+        set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+        set_global_var(GVAR_KARMA_DEFENDER, 0);
+        set_global_var(GVAR_KARMA_WANDERER, 0);
+        set_global_var(GVAR_KARMA_BETRAYER, 0);
+        set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+        set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+            set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
         end
         else begin
-            if (global_var(0) >= 750) then begin
-                set_global_var(38, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
             end
             else begin
-                if (global_var(0) >= 500) then begin
-                    set_global_var(39, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                 end
                 else begin
-                    if (global_var(0) >= 250) then begin
-                        set_global_var(40, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                        set_global_var(GVAR_KARMA_DEFENDER, 1);
                     end
                     else begin
-                        if (global_var(0) > -250) then begin
-                            set_global_var(41, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                            set_global_var(GVAR_KARMA_WANDERER, 1);
                         end
                         else begin
-                            if (global_var(0) > -500) then begin
-                                set_global_var(42, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                set_global_var(GVAR_KARMA_BETRAYER, 1);
                             end
                             else begin
-                                if (global_var(0) > -750) then begin
-                                    set_global_var(43, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -1000) then begin
-                                        set_global_var(44, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                     end
                                     else begin
-                                        set_global_var(45, 1);
+                                        set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                     end
                                 end
                             end
@@ -856,35 +856,35 @@ begin
                 end
             end
         end
-        if (global_var(37) == 1) then begin
+        if (global_var(GVAR_KARMA_HOLY_WARRIOR) == 1) then begin
             Static_Reaction := Static_Reaction + 20;
         end
         else begin
-            if (global_var(38) == 1) then begin
+            if (global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES) == 1) then begin
                 Static_Reaction := Static_Reaction + 15;
             end
             else begin
-                if (global_var(39) == 1) then begin
+                if (global_var(GVAR_KARMA_SHIELD_OF_HOPE) == 1) then begin
                     Static_Reaction := Static_Reaction + 10;
                 end
                 else begin
-                    if (global_var(40) == 1) then begin
+                    if (global_var(GVAR_KARMA_DEFENDER) == 1) then begin
                         Static_Reaction := Static_Reaction + 5;
                     end
                     else begin
-                        if (global_var(41) == 1) then begin
+                        if (global_var(GVAR_KARMA_WANDERER) == 1) then begin
                             Static_Reaction := Static_Reaction + 0;
                         end
                         else begin
-                            if (global_var(42) == 1) then begin
+                            if (global_var(GVAR_KARMA_BETRAYER) == 1) then begin
                                 Static_Reaction := Static_Reaction + -5;
                             end
                             else begin
-                                if (global_var(43) == 1) then begin
+                                if (global_var(GVAR_KARMA_SWORD_OF_DESPAIR) == 1) then begin
                                     Static_Reaction := Static_Reaction + -10;
                                 end
                                 else begin
-                                    if (global_var(44) == 1) then begin
+                                    if (global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE) == 1) then begin
                                         Static_Reaction := Static_Reaction + -15;
                                     end
                                     else begin
@@ -900,28 +900,28 @@ begin
         debug_msg("Good Person Reaction Bonus == " + Static_Reaction);
     end
     if (Slavery_Tolerant == 0) then begin
-        if (global_var(11) == 1) then begin
+        if (global_var(GVAR_REPUTATION_SLAVER) == 1) then begin
             Static_Reaction := Static_Reaction + (2 * -25);
         end
     end
     else begin
         if (Slavery_Tolerant == 1) then begin
-            if (global_var(11) == 1) then begin
+            if (global_var(GVAR_REPUTATION_SLAVER) == 1) then begin
                 Static_Reaction := Static_Reaction + -25;
             end
         end
         else begin
             if (Slavery_Tolerant == 3) then begin
-                if (global_var(11) == 1) then begin
+                if (global_var(GVAR_REPUTATION_SLAVER) == 1) then begin
                     Static_Reaction := Static_Reaction - -25;
                 end
             end
         end
     end
     debug_msg("Slaver + Aligned Reaction == " + Static_Reaction);
-    Static_Reaction := Static_Reaction + (10 * has_trait(0, dude_obj, 10));
+    Static_Reaction := Static_Reaction + (10 * has_trait(TRAIT_PERK, dude_obj, PERK_presence));
     debug_msg("Presence Reaction == " + Static_Reaction);
-    if (global_var(1) >= 2) then begin
+    if (global_var(GVAR_CHILDKILLER_REPUTATION) >= 2) then begin
         Static_Reaction := Static_Reaction + -30;
     end
     debug_msg("Childkiller Reaction == " + Static_Reaction);
@@ -931,11 +931,11 @@ begin
         call Node019();
     end
     else begin
-        if (global_var(350) == 3) then begin
+        if (global_var(GVAR_GRUTHAR_DSTATUS) == 3) then begin
             call Node020();
         end
         else begin
-            if (global_var(350) == 7) then begin
+            if (global_var(GVAR_GRUTHAR_DSTATUS) == 7) then begin
                 call Node018();
             end
             else begin
@@ -962,59 +962,59 @@ end
 procedure destroy_p_proc
 begin
     if (source_obj == dude_obj) then begin
-        set_global_var(5, global_var(5) + 1);
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(4) > (3 * global_var(5))) or (global_var(2) == 1)) and (global_var(1) == 0)) then begin
-            set_global_var(2, 1);
-            set_global_var(3, 0);
+        set_global_var(GVAR_GOOD_MONSTER, global_var(GVAR_GOOD_MONSTER) + 1);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(4) > (3 * global_var(GVAR_GOOD_MONSTER))) or (global_var(GVAR_CHAMPION_REPUTATION) == 1)) and (global_var(GVAR_CHILDKILLER_REPUTATION) == 0)) then begin
+            set_global_var(GVAR_CHAMPION_REPUTATION, 1);
+            set_global_var(GVAR_BERSERKER_REPUTATION, 0);
         end
-        if (metarule(51, self_obj) == 2) then begin
+        if (critter_kill_type(self_obj) == KILL_TYPE_children_kills) then begin
             if (source_obj == dude_obj) then begin
-                set_global_var(2, 0);
-                set_global_var(1, global_var(1) + 1);
-                set_global_var(0, global_var(0) + -15);
+                set_global_var(GVAR_CHAMPION_REPUTATION, 0);
+                set_global_var(GVAR_CHILDKILLER_REPUTATION, global_var(GVAR_CHILDKILLER_REPUTATION) + 1);
+                set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + -15);
                 debug_msg("Player gains " + -15 + " Karma Points.");
-                set_global_var(37, 0);
-                set_global_var(38, 0);
-                set_global_var(39, 0);
-                set_global_var(40, 0);
-                set_global_var(41, 0);
-                set_global_var(42, 0);
-                set_global_var(43, 0);
-                set_global_var(44, 0);
-                set_global_var(45, 0);
-                if (global_var(0) >= 1000) then begin
-                    set_global_var(37, 1);
+                set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+                set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+                set_global_var(GVAR_KARMA_DEFENDER, 0);
+                set_global_var(GVAR_KARMA_WANDERER, 0);
+                set_global_var(GVAR_KARMA_BETRAYER, 0);
+                set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+                set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+                set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+                    set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
                 end
                 else begin
-                    if (global_var(0) >= 750) then begin
-                        set_global_var(38, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
                     end
                     else begin
-                        if (global_var(0) >= 500) then begin
-                            set_global_var(39, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                            set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                         end
                         else begin
-                            if (global_var(0) >= 250) then begin
-                                set_global_var(40, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                                set_global_var(GVAR_KARMA_DEFENDER, 1);
                             end
                             else begin
-                                if (global_var(0) > -250) then begin
-                                    set_global_var(41, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                                    set_global_var(GVAR_KARMA_WANDERER, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -500) then begin
-                                        set_global_var(42, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                        set_global_var(GVAR_KARMA_BETRAYER, 1);
                                     end
                                     else begin
-                                        if (global_var(0) > -750) then begin
-                                            set_global_var(43, 1);
+                                        if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                            set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                         end
                                         else begin
-                                            if (global_var(0) > -1000) then begin
-                                                set_global_var(44, 1);
+                                            if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                                set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                             end
                                             else begin
-                                                set_global_var(45, 1);
+                                                set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                             end
                                         end
                                     end
@@ -1023,58 +1023,58 @@ begin
                         end
                     end
                 end
-                set_global_var(59, global_var(59) + -8);
+                set_global_var(GVAR_TOWN_REP_VAULT_13, global_var(GVAR_TOWN_REP_VAULT_13) + -8);
                 debug_msg("Added " + -8 + " to Town Rep");
             end
         end
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(5) > (2 * global_var(4))) or (global_var(3) == 1))) then begin
-            set_global_var(3, 1);
-            set_global_var(2, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
+            set_global_var(GVAR_BERSERKER_REPUTATION, 1);
+            set_global_var(GVAR_CHAMPION_REPUTATION, 0);
         end
-        set_global_var(0, global_var(0) + -10);
+        set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + -10);
         debug_msg("Player gains " + -10 + " Karma Points.");
-        set_global_var(37, 0);
-        set_global_var(38, 0);
-        set_global_var(39, 0);
-        set_global_var(40, 0);
-        set_global_var(41, 0);
-        set_global_var(42, 0);
-        set_global_var(43, 0);
-        set_global_var(44, 0);
-        set_global_var(45, 0);
-        if (global_var(0) >= 1000) then begin
-            set_global_var(37, 1);
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+        set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+        set_global_var(GVAR_KARMA_DEFENDER, 0);
+        set_global_var(GVAR_KARMA_WANDERER, 0);
+        set_global_var(GVAR_KARMA_BETRAYER, 0);
+        set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+        set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+            set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
         end
         else begin
-            if (global_var(0) >= 750) then begin
-                set_global_var(38, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
             end
             else begin
-                if (global_var(0) >= 500) then begin
-                    set_global_var(39, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                 end
                 else begin
-                    if (global_var(0) >= 250) then begin
-                        set_global_var(40, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                        set_global_var(GVAR_KARMA_DEFENDER, 1);
                     end
                     else begin
-                        if (global_var(0) > -250) then begin
-                            set_global_var(41, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                            set_global_var(GVAR_KARMA_WANDERER, 1);
                         end
                         else begin
-                            if (global_var(0) > -500) then begin
-                                set_global_var(42, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                set_global_var(GVAR_KARMA_BETRAYER, 1);
                             end
                             else begin
-                                if (global_var(0) > -750) then begin
-                                    set_global_var(43, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -1000) then begin
-                                        set_global_var(44, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                     end
                                     else begin
-                                        set_global_var(45, 1);
+                                        set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                     end
                                 end
                             end
@@ -1083,7 +1083,7 @@ begin
                 end
             end
         end
-        set_global_var(59, global_var(59) + -5);
+        set_global_var(GVAR_TOWN_REP_VAULT_13, global_var(GVAR_TOWN_REP_VAULT_13) + -5);
         debug_msg("Added " + -5 + " to Town Rep");
     end
 end
@@ -1092,17 +1092,17 @@ procedure look_at_p_proc
 begin
     script_overrides;
     if (local_var(4) == 0) then begin
-        display_msg(message_str(531, 100));
+        display_mstr(100);
     end
     else begin
-        display_msg(message_str(531, 101));
+        display_mstr(101);
     end
 end
 
 procedure description_p_proc
 begin
     script_overrides;
-    display_msg(message_str(531, 102));
+    display_mstr(102);
 end
 
 procedure use_skill_on_p_proc
@@ -1120,8 +1120,8 @@ end
 procedure map_enter_p_proc
 begin
     Only_Once := 0;
-    critter_add_trait(self_obj, 1, 6, 114);
-    critter_add_trait(self_obj, 1, 5, 22);
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 114);
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_TOUGH_GUARD);
     numTvars := 4;
     cbFunc := "tvar_callback";
     tile0 := 23265;
@@ -1132,7 +1132,7 @@ begin
     lTime2 := game_ticks(1);
     tile3 := 23258;
     lTime3 := game_ticks(1);
-    if (not(metarule(22, 0))) then begin
+    if (not(is_loading_game)) then begin
         call tvar_Warp_To_Pos(0, game_ticks(random(10, 20)));
     end
     add_timer_event(self_obj, game_ticks(5), -1);
@@ -1160,114 +1160,114 @@ end
 
 procedure Node001
 begin
-    gsay_reply(531, 103);
-    giq_option(-3, 531, 104, Node002, 50);
-    giq_option(-3, 531, message_str(531, 105) + " " + obj_name(dude_obj) + message_str(531, 180), Node004, 50);
-    giq_option(-3, 531, 106, Node999, 50);
-    giq_option(4, 531, 107, Node002, 50);
-    giq_option(4, 531, message_str(531, 108) + " " + obj_name(dude_obj) + message_str(531, 181), Node004, 49);
-    giq_option(4, 531, 109, Node003, 51);
-    giq_option(4, 531, 110, Node999, 50);
+    Reply(103);
+    giq_option(-3, NAME, 104, Node002, NEUTRAL_REACTION);
+    giq_option(-3, NAME, mstr(105) + " " + dude_name + mstr(180), Node004, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 106, Node999, NEUTRAL_REACTION);
+    giq_option(4, NAME, 107, Node002, NEUTRAL_REACTION);
+    giq_option(4, NAME, mstr(108) + " " + dude_name + mstr(181), Node004, GOOD_REACTION);
+    giq_option(4, NAME, 109, Node003, BAD_REACTION);
+    giq_option(4, NAME, 110, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node002
 begin
-    gsay_reply(531, 111);
-    giq_option(-3, 531, 112, Node999, 50);
-    giq_option(4, 531, 113, Node999, 50);
-    giq_option(4, 531, 114, Node003, 51);
+    Reply(111);
+    giq_option(-3, NAME, 112, Node999, NEUTRAL_REACTION);
+    giq_option(4, NAME, 113, Node999, NEUTRAL_REACTION);
+    giq_option(4, NAME, 114, Node003, BAD_REACTION);
 end
 
 procedure Node003
 begin
-    gsay_reply(531, 115);
-    giq_option(4, 531, 116, Node999, 50);
+    Reply(115);
+    giq_option(4, NAME, 116, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node004
 begin
     if (local_var(9) == 0) then begin
         set_local_var(9, local_var(9) + 1);
-        gsay_reply(531, message_str(531, 117) + " " + obj_name(dude_obj) + message_str(531, 182));
+        Reply(mstr(117) + " " + dude_name + mstr(182));
     end
     else begin
-        gsay_reply(531, 118);
+        Reply(118);
     end
     if (not(q1)) then begin
-        giq_option(-3, 531, 119, Node005a, 50);
-        giq_option(4, 531, 126, Node005a, 50);
+        giq_option(-3, NAME, 119, Node005a, NEUTRAL_REACTION);
+        giq_option(4, NAME, 126, Node005a, NEUTRAL_REACTION);
     end
     if (not(q2)) then begin
-        giq_option(-3, 531, 120, Node009a, 50);
-        giq_option(4, 531, 127, Node009a, 50);
+        giq_option(-3, NAME, 120, Node009a, NEUTRAL_REACTION);
+        giq_option(4, NAME, 127, Node009a, NEUTRAL_REACTION);
     end
     if (not(q3)) then begin
-        giq_option(-3, 531, 121, Node010a, 50);
-        giq_option(4, 531, 128, Node010a, 50);
+        giq_option(-3, NAME, 121, Node010a, NEUTRAL_REACTION);
+        giq_option(4, NAME, 128, Node010a, NEUTRAL_REACTION);
     end
     if (not(q4)) then begin
-        giq_option(-3, 531, 122, Node013a, 50);
-        giq_option(4, 531, 129, Node013a, 50);
+        giq_option(-3, NAME, 122, Node013a, NEUTRAL_REACTION);
+        giq_option(4, NAME, 129, Node013a, NEUTRAL_REACTION);
     end
     if (not(q5)) then begin
-        giq_option(-3, 531, 123, Node014a, 50);
-        giq_option(4, 531, 130, Node014a, 50);
+        giq_option(-3, NAME, 123, Node014a, NEUTRAL_REACTION);
+        giq_option(4, NAME, 130, Node014a, NEUTRAL_REACTION);
     end
-    if ((global_var(350) == 2) or (global_var(350) == 1)) then begin
-        giq_option(-3, 531, 124, Node017, 50);
-        giq_option(4, 531, 131, Node017, 50);
+    if ((global_var(GVAR_GRUTHAR_DSTATUS) == 2) or (global_var(GVAR_GRUTHAR_DSTATUS) == 1)) then begin
+        giq_option(-3, NAME, 124, Node017, NEUTRAL_REACTION);
+        giq_option(4, NAME, 131, Node017, NEUTRAL_REACTION);
     end
-    giq_option(-3, 531, 125, Node999, 50);
-    giq_option(4, 531, 132, Node999, 50);
+    giq_option(-3, NAME, 125, Node999, NEUTRAL_REACTION);
+    giq_option(4, NAME, 132, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node005
 begin
-    gsay_reply(531, 133);
-    giq_option(-3, 531, 134, Node006, 50);
-    giq_option(4, 531, 135, Node006, 50);
-    if ((global_var(350) == 6) or (global_var(350) == 2)) then begin
-        giq_option(4, 531, 136, Node008, 50);
+    Reply(133);
+    giq_option(-3, NAME, 134, Node006, NEUTRAL_REACTION);
+    giq_option(4, NAME, 135, Node006, NEUTRAL_REACTION);
+    if ((global_var(GVAR_GRUTHAR_DSTATUS) == 6) or (global_var(GVAR_GRUTHAR_DSTATUS) == 2)) then begin
+        giq_option(4, NAME, 136, Node008, NEUTRAL_REACTION);
     end
 end
 
 procedure Node006
 begin
-    gsay_reply(531, 137);
-    giq_option(-3, 531, 138, Node007, 50);
-    giq_option(4, 531, 139, Node007, 50);
+    Reply(137);
+    giq_option(-3, NAME, 138, Node007, NEUTRAL_REACTION);
+    giq_option(4, NAME, 139, Node007, NEUTRAL_REACTION);
 end
 
 procedure Node007
 begin
-    gsay_reply(531, 140);
-    giq_option(-3, 531, 141, Node008, 50);
-    giq_option(4, 531, 142, Node008, 50);
-    if ((global_var(350) == 6) or (global_var(350) == 2)) then begin
-        giq_option(4, 531, 143, Node008, 50);
+    Reply(140);
+    giq_option(-3, NAME, 141, Node008, NEUTRAL_REACTION);
+    giq_option(4, NAME, 142, Node008, NEUTRAL_REACTION);
+    if ((global_var(GVAR_GRUTHAR_DSTATUS) == 6) or (global_var(GVAR_GRUTHAR_DSTATUS) == 2)) then begin
+        giq_option(4, NAME, 143, Node008, NEUTRAL_REACTION);
     end
 end
 
 procedure Node008
 begin
-    if (not(global_var(487))) then begin
-        set_global_var(487, 1);
+    if (not(global_var(GVAR_V13_COMP_QST))) then begin
+        set_global_var(GVAR_V13_COMP_QST, 1);
     end
-    if (get_critter_stat(dude_obj, 4) < 3) then begin
-        gsay_reply(531, 144);
+    if (dude_iq < 3) then begin
+        Reply(144);
     end
     else begin
-        gsay_reply(531, 145);
+        Reply(145);
     end
-    giq_option(-3, 531, 146, Node004, 50);
-    giq_option(4, 531, 147, Node004, 50);
+    giq_option(-3, NAME, 146, Node004, NEUTRAL_REACTION);
+    giq_option(4, NAME, 147, Node004, NEUTRAL_REACTION);
 end
 
 procedure Node009
 begin
-    gsay_reply(531, 148);
-    giq_option(-3, 531, 149, Node004, 50);
-    giq_option(4, 531, 150, Node004, 50);
+    Reply(148);
+    giq_option(-3, NAME, 149, Node004, NEUTRAL_REACTION);
+    giq_option(4, NAME, 150, Node004, NEUTRAL_REACTION);
 end
 
 procedure Node009a
@@ -1278,77 +1278,77 @@ end
 
 procedure Node010
 begin
-    gsay_reply(531, 151);
-    giq_option(-3, 531, 152, Node011, 50);
-    giq_option(-3, 531, 153, Node004, 50);
-    giq_option(4, 531, 154, Node011, 50);
-    giq_option(4, 531, 155, Node004, 50);
+    Reply(151);
+    giq_option(-3, NAME, 152, Node011, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 153, Node004, NEUTRAL_REACTION);
+    giq_option(4, NAME, 154, Node011, NEUTRAL_REACTION);
+    giq_option(4, NAME, 155, Node004, NEUTRAL_REACTION);
 end
 
 procedure Node011
 begin
-    gsay_reply(531, 156);
-    giq_option(-3, 531, 157, Node012, 50);
-    giq_option(4, 531, 158, Node012, 50);
+    Reply(156);
+    giq_option(-3, NAME, 157, Node012, NEUTRAL_REACTION);
+    giq_option(4, NAME, 158, Node012, NEUTRAL_REACTION);
 end
 
 procedure Node012
 begin
-    gsay_reply(531, 159);
-    giq_option(-3, 531, 160, Node004, 50);
-    giq_option(4, 531, 161, Node004, 50);
+    Reply(159);
+    giq_option(-3, NAME, 160, Node004, NEUTRAL_REACTION);
+    giq_option(4, NAME, 161, Node004, NEUTRAL_REACTION);
 end
 
 procedure Node013
 begin
-    gsay_reply(531, 162);
-    giq_option(-3, 531, 163, Node004, 50);
-    giq_option(4, 531, 164, Node004, 50);
+    Reply(162);
+    giq_option(-3, NAME, 163, Node004, NEUTRAL_REACTION);
+    giq_option(4, NAME, 164, Node004, NEUTRAL_REACTION);
 end
 
 procedure Node014
 begin
-    gsay_reply(531, 165);
-    giq_option(-3, 531, 166, Node015, 50);
-    giq_option(4, 531, 167, Node015, 50);
+    Reply(165);
+    giq_option(-3, NAME, 166, Node015, NEUTRAL_REACTION);
+    giq_option(4, NAME, 167, Node015, NEUTRAL_REACTION);
 end
 
 procedure Node015
 begin
-    gsay_reply(531, 168);
-    giq_option(-3, 531, 169, Node004, 50);
-    giq_option(4, 531, 170, Node004, 50);
+    Reply(168);
+    giq_option(-3, NAME, 169, Node004, NEUTRAL_REACTION);
+    giq_option(4, NAME, 170, Node004, NEUTRAL_REACTION);
 end
 
 procedure Node016
 begin
-    gsay_reply(531, 171);
-    giq_option(-3, 531, 172, Node004, 50);
-    giq_option(-3, 531, 173, Node999, 50);
-    giq_option(4, 531, 174, Node004, 50);
-    giq_option(4, 531, 175, Node999, 50);
+    Reply(171);
+    giq_option(-3, NAME, 172, Node004, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 173, Node999, NEUTRAL_REACTION);
+    giq_option(4, NAME, 174, Node004, NEUTRAL_REACTION);
+    giq_option(4, NAME, 175, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node017
 begin
-    gsay_reply(531, 176);
-    giq_option(1, 531, 177, Node004, 50);
+    Reply(176);
+    giq_option(1, NAME, 177, Node004, NEUTRAL_REACTION);
 end
 
 procedure Node018
 begin
-    float_msg(self_obj, message_str(531, 178), 0);
+    float_msg(self_obj, mstr(178), FLOAT_MSG_NORMAL);
 end
 
 procedure Node019
 begin
-    float_msg(self_obj, message_str(531, 179), 2);
+    float_msg(self_obj, mstr(179), FLOAT_MSG_RED);
     call Node998();
 end
 
 procedure Node020
 begin
-    float_msg(self_obj, message_str(531, 183), 0);
+    float_msg(self_obj, mstr(183), FLOAT_MSG_NORMAL);
 end
 
 procedure Node005a
@@ -1374,4 +1374,3 @@ begin
     q5 := 1;
     call Node014();
 end
-
