diff --git a/Games/Fallout2.Steam/Original/OSV13DR.ssl b/Games/Fallout2.Steam/ReDefine/OSV13DR.ssl
index 2b7a8eb..392fb33 100644
--- a/Games/Fallout2.Steam/Original/OSV13DR.ssl
+++ b/Games/Fallout2.Steam/ReDefine/OSV13DR.ssl
@@ -173,7 +173,7 @@ end
 procedure map_enter_p_proc
 begin
     i_vault13_door := self_obj;
-    if (not(metarule(22, 0))) then begin
+    if (not(is_loading_game)) then begin
         if (obj_is_open(self_obj)) then begin
             reg_anim_func(2, self_obj);
             reg_anim_func(1, 1);
@@ -195,4 +195,3 @@ end
 procedure timed_event_p_proc
 begin
 end
-
