diff --git a/Games/Fallout2.Steam/Original/PALGUARD.ssl b/Games/Fallout2.Steam/ReDefine/PALGUARD.ssl
index ed0082f..4b98844 100644
--- a/Games/Fallout2.Steam/Original/PALGUARD.ssl
+++ b/Games/Fallout2.Steam/ReDefine/PALGUARD.ssl
@@ -26,37 +26,37 @@ procedure start
 begin
     if (Only_Once == 0) then begin
         Only_Once := 1;
-        critter_add_trait(self_obj, 1, 6, 44);
-        critter_add_trait(self_obj, 1, 5, 65);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 44);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_PARTY_MARCUS_DEFENSIVE);
     end
-    if (script_action == 11) then begin
+    if (script_action == talk_proc) then begin
         call do_dialogue();
     end
     else begin
-        if (script_action == 18) then begin
+        if (script_action == destroy_proc) then begin
             if (source_obj == dude_obj) then begin
-                if (((global_var(160) + global_var(159)) >= 25) and ((global_var(159) > (2 * global_var(160))) or (global_var(156) == 1))) then begin
-                    set_global_var(156, 1);
-                    set_global_var(157, 0);
+                if (((global_var(GVAR_GECKO_SKEETER_PART) + global_var(GVAR_GECKO_REQ_FORM)) >= 25) and ((global_var(GVAR_GECKO_REQ_FORM) > (2 * global_var(GVAR_GECKO_SKEETER_PART))) or (global_var(GVAR_SIERRA_LOCKOUT) == 1))) then begin
+                    set_global_var(GVAR_SIERRA_LOCKOUT, 1);
+                    set_global_var(GVAR_SIERRA_PASSWORD, 0);
                 end
-                if (((global_var(160) + global_var(159)) >= 25) and ((global_var(160) > (3 * global_var(159))) or (global_var(157) == 1))) then begin
-                    set_global_var(157, 1);
-                    set_global_var(156, 0);
+                if (((global_var(GVAR_GECKO_SKEETER_PART) + global_var(GVAR_GECKO_REQ_FORM)) >= 25) and ((global_var(160) > (3 * global_var(GVAR_GECKO_REQ_FORM))) or (global_var(GVAR_SIERRA_PASSWORD) == 1))) then begin
+                    set_global_var(GVAR_SIERRA_PASSWORD, 1);
+                    set_global_var(GVAR_SIERRA_LOCKOUT, 0);
                 end
-                set_global_var(159, global_var(159) + 1);
-                if ((global_var(159) % 2) == 0) then begin
-                    set_global_var(155, global_var(155) - 1);
+                set_global_var(GVAR_GECKO_REQ_FORM, global_var(GVAR_GECKO_REQ_FORM) + 1);
+                if ((global_var(GVAR_GECKO_REQ_FORM) % 2) == 0) then begin
+                    set_global_var(GVAR_BRAIN_BOT_BRAIN, global_var(GVAR_BRAIN_BOT_BRAIN) - 1);
                 end
             end
         end
         else begin
-            if ((script_action == 21) or (script_action == 3)) then begin
+            if ((script_action == look_at_proc) or (script_action == description_proc)) then begin
                 script_overrides;
-                display_msg(message_str(317, 100));
+                display_msg(message_str(SCRIPT_KIBBOX, 100));
             end
             else begin
-                if (script_action == 22) then begin
-                    attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+                if (script_action == timed_event_proc) then begin
+                    attack(dude_obj);
                 end
             end
         end
@@ -86,28 +86,28 @@ begin
         line := 1;
     end
     if (line == 1) then begin
-        float_msg(self_obj, message_str(317, 101), 2);
+        float_msg(self_obj, message_str(SCRIPT_KIBBOX, 101), FLOAT_MSG_RED);
     end
     else begin
         if (line == 2) then begin
-            float_msg(self_obj, message_str(317, 102), 2);
+            float_msg(self_obj, message_str(SCRIPT_KIBBOX, 102), FLOAT_MSG_RED);
         end
         else begin
             if (line == 3) then begin
-                float_msg(self_obj, message_str(317, 103), 2);
+                float_msg(self_obj, message_str(SCRIPT_KIBBOX, 103), FLOAT_MSG_RED);
             end
             else begin
                 if (line == 4) then begin
-                    float_msg(self_obj, message_str(317, 104), 2);
+                    float_msg(self_obj, message_str(SCRIPT_KIBBOX, 104), FLOAT_MSG_RED);
                 end
                 else begin
                     if (line == 5) then begin
-                        float_msg(self_obj, message_str(317, 105), 2);
+                        float_msg(self_obj, message_str(SCRIPT_KIBBOX, 105), FLOAT_MSG_RED);
                     end
                     else begin
                         if (line == 6) then begin
-                            if (get_critter_stat(dude_obj, 34) == 1) then begin
-                                float_msg(self_obj, message_str(317, 106), 2);
+                            if (dude_gender == 1) then begin
+                                float_msg(self_obj, message_str(SCRIPT_KIBBOX, 106), FLOAT_MSG_RED);
                             end
                             else begin
                                 line := line + 1;
@@ -115,8 +115,8 @@ begin
                         end
                         else begin
                             if (line == 7) then begin
-                                if (global_var(108) == 2) then begin
-                                    float_msg(self_obj, message_str(317, 107), 2);
+                                if (global_var(GVAR_MODOC_BRAHMIN_SEED) == 2) then begin
+                                    float_msg(self_obj, message_str(SCRIPT_KIBBOX, 107), FLOAT_MSG_RED);
                                 end
                                 else begin
                                     line := line + 1;
@@ -124,8 +124,8 @@ begin
                             end
                             else begin
                                 if (line == 8) then begin
-                                    if (global_var(108) < 2) then begin
-                                        float_msg(self_obj, message_str(317, 108), 2);
+                                    if (global_var(GVAR_MODOC_BRAHMIN_SEED) < 2) then begin
+                                        float_msg(self_obj, message_str(SCRIPT_KIBBOX, 108), FLOAT_MSG_RED);
                                     end
                                     else begin
                                         line := line + 1;
@@ -133,12 +133,12 @@ begin
                                 end
                                 else begin
                                     if (line == 9) then begin
-                                        if (global_var(108) < 2) then begin
-                                            float_msg(self_obj, message_str(317, 109), 2);
+                                        if (global_var(GVAR_MODOC_BRAHMIN_SEED) < 2) then begin
+                                            float_msg(self_obj, message_str(SCRIPT_KIBBOX, 109), FLOAT_MSG_RED);
                                         end
                                         else begin
                                             line := 1;
-                                            float_msg(self_obj, message_str(317, 110), 2);
+                                            float_msg(self_obj, message_str(SCRIPT_KIBBOX, 110), FLOAT_MSG_RED);
                                         end
                                     end
                                 end
@@ -163,40 +163,40 @@ begin
         line := 1;
     end
     if (line == 1) then begin
-        float_msg(self_obj, message_str(317, 111), 2);
+        float_msg(self_obj, message_str(SCRIPT_KIBBOX, 111), FLOAT_MSG_RED);
     end
     else begin
         if (line == 2) then begin
-            float_msg(self_obj, message_str(317, 112), 2);
+            float_msg(self_obj, message_str(SCRIPT_KIBBOX, 112), FLOAT_MSG_RED);
         end
         else begin
             if (line == 3) then begin
-                float_msg(self_obj, message_str(317, 113), 2);
+                float_msg(self_obj, message_str(SCRIPT_KIBBOX, 113), FLOAT_MSG_RED);
             end
             else begin
                 if (line == 4) then begin
-                    float_msg(self_obj, message_str(317, 114), 2);
+                    float_msg(self_obj, message_str(SCRIPT_KIBBOX, 114), FLOAT_MSG_RED);
                 end
                 else begin
                     if (line == 5) then begin
-                        float_msg(self_obj, message_str(317, 115), 2);
+                        float_msg(self_obj, message_str(SCRIPT_KIBBOX, 115), FLOAT_MSG_RED);
                     end
                     else begin
                         if (line == 7) then begin
-                            if (global_var(108) == 2) then begin
-                                float_msg(self_obj, message_str(317, 116), 2);
+                            if (global_var(GVAR_MODOC_BRAHMIN_SEED) == 2) then begin
+                                float_msg(self_obj, message_str(SCRIPT_KIBBOX, 116), FLOAT_MSG_RED);
                             end
                             else begin
-                                float_msg(self_obj, message_str(317, 117), 2);
+                                float_msg(self_obj, message_str(SCRIPT_KIBBOX, 117), FLOAT_MSG_RED);
                             end
                         end
                         else begin
                             if (line == 7) then begin
-                                if (global_var(108) == 2) then begin
-                                    float_msg(self_obj, message_str(317, 118), 2);
+                                if (global_var(GVAR_MODOC_BRAHMIN_SEED) == 2) then begin
+                                    float_msg(self_obj, message_str(SCRIPT_KIBBOX, 118), FLOAT_MSG_RED);
                                 end
                                 else begin
-                                    float_msg(self_obj, message_str(317, 119), 2);
+                                    float_msg(self_obj, message_str(SCRIPT_KIBBOX, 119), FLOAT_MSG_RED);
                                 end
                             end
                         end
@@ -213,31 +213,31 @@ begin
         set_local_var(0, 50);
         set_local_var(1, 2);
         set_local_var(2, 1);
-        set_local_var(0, local_var(0) + (5 * get_critter_stat(dude_obj, 3)) - 25);
-        set_local_var(0, local_var(0) + (10 * has_trait(0, dude_obj, 10)));
-        if (has_trait(0, dude_obj, 39)) then begin
-            if (global_var(155) > 0) then begin
-                set_local_var(0, local_var(0) + global_var(155));
+        set_local_var(0, local_var(0) + (5 * dude_charisma) - 25);
+        set_local_var(0, local_var(0) + (10 * has_trait(TRAIT_PERK, dude_obj, PERK_presence)));
+        if (has_trait(TRAIT_PERK, dude_obj, PERK_cult_of_personality)) then begin
+            if (global_var(GVAR_BRAIN_BOT_BRAIN) > 0) then begin
+                set_local_var(0, local_var(0) + global_var(GVAR_BRAIN_BOT_BRAIN));
             end
             else begin
-                set_local_var(0, local_var(0) - global_var(155));
+                set_local_var(0, local_var(0) - global_var(GVAR_BRAIN_BOT_BRAIN));
             end
         end
         else begin
             if (local_var(3) == 1) then begin
-                set_local_var(0, local_var(0) - global_var(155));
+                set_local_var(0, local_var(0) - global_var(GVAR_BRAIN_BOT_BRAIN));
             end
             else begin
-                set_local_var(0, local_var(0) + global_var(155));
+                set_local_var(0, local_var(0) + global_var(GVAR_BRAIN_BOT_BRAIN));
             end
         end
-        if (global_var(158) > 2) then begin
+        if (global_var(GVAR_GECKO_ECON_DISK) > 2) then begin
             set_local_var(0, local_var(0) - 30);
         end
-        if (((global_var(160) + global_var(159)) >= 25) and ((global_var(160) > (3 * global_var(159))) or (global_var(157) == 1))) then begin
+        if (((global_var(GVAR_GECKO_SKEETER_PART) + global_var(GVAR_GECKO_REQ_FORM)) >= 25) and ((global_var(160) > (3 * global_var(GVAR_GECKO_REQ_FORM))) or (global_var(GVAR_SIERRA_PASSWORD) == 1))) then begin
             set_local_var(0, local_var(0) + 20);
         end
-        if (((global_var(160) + global_var(159)) >= 25) and ((global_var(159) > (2 * global_var(160))) or (global_var(156) == 1))) then begin
+        if (((global_var(GVAR_GECKO_SKEETER_PART) + global_var(GVAR_GECKO_REQ_FORM)) >= 25) and ((global_var(GVAR_GECKO_REQ_FORM) > (2 * global_var(GVAR_GECKO_SKEETER_PART))) or (global_var(GVAR_SIERRA_LOCKOUT) == 1))) then begin
             set_local_var(0, local_var(0) - 20);
         end
         call ReactToLevel();
@@ -330,6 +330,5 @@ end
 
 procedure Goodbyes
 begin
-    exit_line := message_str(634, random(100, 105));
+    exit_line := message_str(SCRIPT_BSCOMP3, random(100, 105));
 end
-
