diff --git a/Games/Fallout2.Steam/Original/QIBSTDR1.ssl b/Games/Fallout2.Steam/ReDefine/QIBSTDR1.ssl
index d56045d..9ff9606 100644
--- a/Games/Fallout2.Steam/Original/QIBSTDR1.ssl
+++ b/Games/Fallout2.Steam/ReDefine/QIBSTDR1.ssl
@@ -143,38 +143,38 @@ end
 procedure use_p_proc
 begin
     script_overrides;
-    display_msg(message_str(887, 103));
+    display_mstr(103);
 end
 
 procedure description_p_proc
 begin
     script_overrides;
-    display_msg(message_str(887, 102));
+    display_mstr(102);
 end
 
 procedure look_at_p_proc
 begin
     script_overrides;
-    display_msg(message_str(887, 102));
+    display_mstr(102);
 end
 
 procedure use_obj_on_p_proc
 begin
     if ((obj_pid(obj_being_used_with) == 411) or (obj_pid(obj_being_used_with) == 77)) then begin
         script_overrides;
-        display_msg(message_str(887, 100));
+        display_mstr(100);
     end
 end
 
 procedure damage_p_proc
 begin
     script_overrides;
-    display_msg(message_str(887, 101));
+    display_mstr(101);
 end
 
 procedure map_update_p_proc
 begin
-    if (global_var(435) == 1) then begin
+    if (global_var(GVAR_ENCLAVE_REACTOR) == 1) then begin
         obj_close(self_obj);
     end
     else begin
@@ -186,7 +186,6 @@ procedure use_skill_on_p_proc
 begin
     if (action_being_used == 9) then begin
         script_overrides;
-        display_msg(message_str(887, 100));
+        display_mstr(100);
     end
 end
-
