diff --git a/Games/Fallout2.Steam/Original/QIGECKDR.ssl b/Games/Fallout2.Steam/ReDefine/QIGECKDR.ssl
index 68840eb..4f788b3 100644
--- a/Games/Fallout2.Steam/Original/QIGECKDR.ssl
+++ b/Games/Fallout2.Steam/ReDefine/QIGECKDR.ssl
@@ -141,7 +141,7 @@ procedure use_p_proc
 begin
     if (not(obj_is_carrying_obj_pid(dude_obj, 366))) then begin
         script_overrides;
-        float_msg(dude_obj, message_str(14, 1016), 9);
+        float_msg(dude_obj, g_mstr(1016), FLOAT_MSG_WHITE);
     end
 end
 
@@ -165,8 +165,7 @@ begin
         script_overrides;
         explosion(tile_num(self_obj), elevation(self_obj), 60);
         set_local_var(0, 1);
-        display_msg(message_str(14, 1017));
+        display_msg(g_mstr(1017));
         destroy_object(self_obj);
     end
 end
-
