diff --git a/Games/Fallout2.Steam/Original/QIPZLTRM.ssl b/Games/Fallout2.Steam/ReDefine/QIPZLTRM.ssl
index 8d6bb88..ce8731d 100644
--- a/Games/Fallout2.Steam/Original/QIPZLTRM.ssl
+++ b/Games/Fallout2.Steam/ReDefine/QIPZLTRM.ssl
@@ -166,57 +166,57 @@ begin
     if ((action_being_used == 13) and (map_var(0) == 0) and (local_var(1) == map_var(1))) then begin
         set_map_var(0, 1);
         if (is_success(roll_vs_skill(dude_obj, 12, -20))) then begin
-            if ((has_skill(dude_obj, 12) > 99) and metarule(40, 12)) then begin
+            if ((dude_skill(SKILL_SCIENCE) > 99) and is_skill_tagged(SKILL_SCIENCE)) then begin
                 script_overrides;
-                display_msg(message_str(883, 110));
+                display_mstr(110);
                 call Egg_Science();
             end
             else begin
-                if ((has_skill(dude_obj, 13) > 99) and metarule(40, 13)) then begin
+                if ((dude_skill(SKILL_REPAIR) > 99) and is_skill_tagged(SKILL_REPAIR)) then begin
                     script_overrides;
-                    display_msg(message_str(883, 110));
+                    display_mstr(110);
                     call Egg_Repair();
                 end
                 else begin
-                    if ((has_skill(dude_obj, 17) > 99) and metarule(40, 17)) then begin
+                    if ((dude_skill(SKILL_OUTDOORSMAN) > 99) and is_skill_tagged(SKILL_OUTDOORSMAN)) then begin
                         script_overrides;
-                        display_msg(message_str(883, 110));
+                        display_mstr(110);
                         call Egg_Outdoorsman();
                     end
                     else begin
-                        if ((has_skill(dude_obj, 16) > 99) and metarule(40, 16)) then begin
+                        if ((dude_skill(SKILL_GAMBLING) > 99) and is_skill_tagged(SKILL_GAMBLING)) then begin
                             script_overrides;
-                            display_msg(message_str(883, 110));
+                            display_mstr(110);
                             call Egg_Gambling();
                         end
                         else begin
-                            if ((has_skill(dude_obj, 5) > 99) and metarule(40, 5)) then begin
+                            if ((dude_skill(SKILL_THROWING) > 99) and is_skill_tagged(SKILL_THROWING)) then begin
                                 script_overrides;
-                                display_msg(message_str(883, 110));
+                                display_mstr(110);
                                 call Egg_Throwing();
                             end
                             else begin
-                                if ((has_skill(dude_obj, 6) > 99) and metarule(40, 6)) then begin
+                                if ((dude_skill(SKILL_FIRST_AID) > 99) and is_skill_tagged(SKILL_FIRST_AID)) then begin
                                     script_overrides;
-                                    display_msg(message_str(883, 110));
+                                    display_mstr(110);
                                     call Egg_First_Aid();
                                 end
                                 else begin
-                                    if ((has_skill(dude_obj, 7) > 99) and metarule(40, 7)) then begin
+                                    if ((dude_skill(SKILL_DOCTOR) > 99) and is_skill_tagged(SKILL_DOCTOR)) then begin
                                         script_overrides;
-                                        display_msg(message_str(883, 110));
+                                        display_mstr(110);
                                         call Egg_Doctor();
                                     end
                                     else begin
-                                        if ((has_skill(dude_obj, 11) > 99) and metarule(40, 11)) then begin
+                                        if ((dude_skill(SKILL_TRAPS) > 99) and is_skill_tagged(SKILL_TRAPS)) then begin
                                             script_overrides;
-                                            display_msg(message_str(883, 110));
+                                            display_mstr(110);
                                             call Egg_Traps();
                                         end
                                         else begin
-                                            if ((has_skill(dude_obj, 14) > 99) and metarule(40, 14)) then begin
+                                            if ((dude_skill(SKILL_SPEECH) > 99) and is_skill_tagged(SKILL_SPEECH)) then begin
                                                 script_overrides;
-                                                display_msg(message_str(883, 110));
+                                                display_mstr(110);
                                                 call Egg_Conversant();
                                             end
                                         end
@@ -234,7 +234,7 @@ end
 procedure use_p_proc
 begin
     script_overrides;
-    display_msg(message_str(883, 102));
+    display_mstr(102);
     if (tile_num(self_obj) == 17911) then begin
         call Term1();
     end
@@ -283,13 +283,13 @@ end
 procedure look_at_p_proc
 begin
     script_overrides;
-    display_msg(message_str(883, 100));
+    display_mstr(100);
 end
 
 procedure description_p_proc
 begin
     script_overrides;
-    display_msg(message_str(883, 101));
+    display_mstr(101);
 end
 
 procedure map_enter_p_proc
@@ -305,7 +305,7 @@ begin
             debug_msg("MVAR_Egg_Counter == " + map_var(1));
         end
     end
-    if ((tile_num(self_obj) == 17911) and (map_var(2) == 0) and not(metarule(22, 0))) then begin
+    if ((tile_num(self_obj) == 17911) and (map_var(2) == 0) and not(is_loading_game)) then begin
         add_timer_event(self_obj, game_ticks(10), 1);
     end
 end
@@ -313,9 +313,9 @@ end
 procedure timed_event_p_proc
 begin
     if (map_var(2) == 0) then begin
-        display_msg(message_str(883, 120));
-        if (tile_in_tile_rect(17520, 17490, 23520, 23490, tile_num(dude_obj))) then begin
-            critter_dmg(dude_obj, random(20, 40), 4);
+        display_mstr(120);
+        if (tile_in_tile_rect(17520, 17490, 23520, 23490, dude_tile)) then begin
+            critter_dmg(dude_obj, random(20, 40), DMG_electrical);
         end
         call Zark_Party();
         add_timer_event(self_obj, game_ticks(10), 1);
@@ -758,62 +758,62 @@ procedure Zark_Party
 begin
     if (party_member_obj(16777278) != 0) then begin
         if (tile_in_tile_rect(17520, 17490, 23520, 23490, tile_num(party_member_obj(16777278)))) then begin
-            critter_dmg(party_member_obj(16777278), random(20, 40), 4);
+            critter_dmg(party_member_obj(16777278), random(20, 40), DMG_electrical);
         end
     end
     if (party_member_obj(16777376) != 0) then begin
         if (tile_in_tile_rect(17520, 17490, 23520, 23490, tile_num(party_member_obj(16777376)))) then begin
-            critter_dmg(party_member_obj(16777376), random(20, 40), 4);
+            critter_dmg(party_member_obj(16777376), random(20, 40), DMG_electrical);
         end
     end
     if (party_member_obj(16777377) != 0) then begin
         if (tile_in_tile_rect(17520, 17490, 23520, 23490, tile_num(party_member_obj(16777377)))) then begin
-            critter_dmg(party_member_obj(16777377), random(20, 40), 4);
+            critter_dmg(party_member_obj(16777377), random(20, 40), DMG_electrical);
         end
     end
     if (party_member_obj(16777305) != 0) then begin
         if (tile_in_tile_rect(17520, 17490, 23520, 23490, tile_num(party_member_obj(16777305)))) then begin
-            critter_dmg(party_member_obj(16777305), random(20, 40), 4);
+            critter_dmg(party_member_obj(16777305), random(20, 40), DMG_electrical);
         end
     end
     if (party_member_obj(16777313) != 0) then begin
         if (tile_in_tile_rect(17520, 17490, 23520, 23490, tile_num(party_member_obj(16777313)))) then begin
-            critter_dmg(party_member_obj(16777313), random(20, 40), 4);
+            critter_dmg(party_member_obj(16777313), random(20, 40), DMG_electrical);
         end
     end
     if (party_member_obj(16777323) != 0) then begin
         if (tile_in_tile_rect(17520, 17490, 23520, 23490, tile_num(party_member_obj(16777278)))) then begin
-            critter_dmg(party_member_obj(16777323), random(20, 40), 4);
+            critter_dmg(party_member_obj(16777323), random(20, 40), DMG_electrical);
         end
     end
     if (party_member_obj(16777352) != 0) then begin
         if (tile_in_tile_rect(17520, 17490, 23520, 23490, tile_num(party_member_obj(16777278)))) then begin
-            critter_dmg(party_member_obj(16777352), random(20, 40), 4);
+            critter_dmg(party_member_obj(16777352), random(20, 40), DMG_electrical);
         end
     end
     if (party_member_obj(16777378) != 0) then begin
         if (tile_in_tile_rect(17520, 17490, 23520, 23490, tile_num(party_member_obj(16777378)))) then begin
-            critter_dmg(party_member_obj(16777378), random(20, 40), 4);
+            critter_dmg(party_member_obj(16777378), random(20, 40), DMG_electrical);
         end
     end
     if (party_member_obj(16777368) != 0) then begin
         if (tile_in_tile_rect(17520, 17490, 23520, 23490, tile_num(party_member_obj(16777368)))) then begin
-            critter_dmg(party_member_obj(16777368), random(20, 40), 4);
+            critter_dmg(party_member_obj(16777368), random(20, 40), DMG_electrical);
         end
     end
     if (party_member_obj(16777379) != 0) then begin
         if (tile_in_tile_rect(17520, 17490, 23520, 23490, tile_num(party_member_obj(16777379)))) then begin
-            critter_dmg(party_member_obj(16777379), random(20, 40), 4);
+            critter_dmg(party_member_obj(16777379), random(20, 40), DMG_electrical);
         end
     end
     if (party_member_obj(16777380) != 0) then begin
         if (tile_in_tile_rect(17520, 17490, 23520, 23490, tile_num(party_member_obj(16777380)))) then begin
-            critter_dmg(party_member_obj(16777380), random(20, 40), 4);
+            critter_dmg(party_member_obj(16777380), random(20, 40), DMG_electrical);
         end
     end
     if (party_member_obj(16777295) != 0) then begin
         if (tile_in_tile_rect(17520, 17490, 23520, 23490, tile_num(party_member_obj(16777295)))) then begin
-            critter_dmg(party_member_obj(16777295), random(20, 40), 4);
+            critter_dmg(party_member_obj(16777295), random(20, 40), DMG_electrical);
         end
     end
 end
@@ -925,7 +925,7 @@ begin
     variable LVar0 := 0;
     debug_msg("Launched Repair Easter Egg!");
     set_map_var(2, 0);
-    LVar0 := create_object_sid(328, 0, 0, -1);
+    LVar0 := create_object(328, 0, 0);
     add_obj_to_inven(dude_obj, LVar0);
 end
 
@@ -933,7 +933,7 @@ procedure Egg_Outdoorsman
 begin
     variable LVar0 := 0;
     debug_msg("Launched Outdoorsman Easter Egg!");
-    LVar0 := create_object_sid(397, 0, 0, -1);
+    LVar0 := create_object(397, 0, 0);
     add_obj_to_inven(dude_obj, LVar0);
 end
 
@@ -941,9 +941,9 @@ procedure Egg_Gambling
 begin
     variable LVar0 := 0;
     debug_msg("Launched Gambling Easter Egg!");
-    LVar0 := create_object_sid(437, 0, 0, -1);
+    LVar0 := create_object(437, 0, 0);
     add_obj_to_inven(dude_obj, LVar0);
-    LVar0 := create_object_sid(326, 0, 0, -1);
+    LVar0 := create_object(326, 0, 0);
     add_obj_to_inven(dude_obj, LVar0);
     item_caps_adjust(dude_obj, random(2000, 5000));
 end
@@ -952,7 +952,7 @@ procedure Egg_Throwing
 begin
     variable LVar0 := 0;
     debug_msg("Launched Throwing Easter Egg!");
-    LVar0 := create_object_sid(26, 0, 0, -1);
+    LVar0 := create_object(26, 0, 0);
     add_mult_objs_to_inven(dude_obj, LVar0, random(10, 20));
 end
 
@@ -960,15 +960,15 @@ procedure Egg_First_Aid
 begin
     variable LVar0 := 0;
     debug_msg("Launched First Aid Easter Egg!");
-    LVar0 := create_object_sid(40, 0, 0, -1);
+    LVar0 := create_object(40, 0, 0);
     add_obj_to_inven(dude_obj, LVar0);
-    LVar0 := create_object_sid(53, 0, 0, -1);
+    LVar0 := create_object(53, 0, 0);
     add_obj_to_inven(dude_obj, LVar0);
-    LVar0 := create_object_sid(47, 0, 0, -1);
+    LVar0 := create_object(47, 0, 0);
     add_obj_to_inven(dude_obj, LVar0);
-    LVar0 := create_object_sid(87, 0, 0, -1);
+    LVar0 := create_object(87, 0, 0);
     add_obj_to_inven(dude_obj, LVar0);
-    LVar0 := create_object_sid(110, 0, 0, -1);
+    LVar0 := create_object(110, 0, 0);
     add_obj_to_inven(dude_obj, LVar0);
 end
 
@@ -976,21 +976,21 @@ procedure Egg_Doctor
 begin
     variable LVar0 := 0;
     debug_msg("Launched Doctor Easter Egg!");
-    LVar0 := create_object_sid(40, 0, 0, -1);
+    LVar0 := create_object(40, 0, 0);
     add_mult_objs_to_inven(dude_obj, LVar0, random(2, 4));
-    LVar0 := create_object_sid(91, 0, 0, -1);
+    LVar0 := create_object(91, 0, 0);
     add_mult_objs_to_inven(dude_obj, LVar0, random(1, 2));
-    LVar0 := create_object_sid(110, 0, 0, -1);
+    LVar0 := create_object(110, 0, 0);
     add_mult_objs_to_inven(dude_obj, LVar0, random(2, 4));
-    LVar0 := create_object_sid(144, 0, 0, -1);
+    LVar0 := create_object(144, 0, 0);
     add_mult_objs_to_inven(dude_obj, LVar0, random(2, 4));
-    LVar0 := create_object_sid(259, 0, 0, -1);
+    LVar0 := create_object(259, 0, 0);
     add_mult_objs_to_inven(dude_obj, LVar0, random(2, 4));
-    LVar0 := create_object_sid(334, 0, 0, -1);
+    LVar0 := create_object(334, 0, 0);
     add_mult_objs_to_inven(dude_obj, LVar0, 1);
-    LVar0 := create_object_sid(428, 0, 0, -1);
+    LVar0 := create_object(428, 0, 0);
     add_mult_objs_to_inven(dude_obj, LVar0, 1);
-    LVar0 := create_object_sid(87, 0, 0, -1);
+    LVar0 := create_object(87, 0, 0);
     add_mult_objs_to_inven(dude_obj, LVar0, random(2, 4));
 end
 
@@ -998,13 +998,13 @@ procedure Egg_Traps
 begin
     variable LVar0 := 0;
     debug_msg("Launched Traps Easter Egg!");
-    LVar0 := create_object_sid(59, 0, 0, -1);
+    LVar0 := create_object(59, 0, 0);
     add_mult_objs_to_inven(dude_obj, LVar0, 1);
-    LVar0 := create_object_sid(411, 0, 0, -1);
+    LVar0 := create_object(411, 0, 0);
     add_mult_objs_to_inven(dude_obj, LVar0, random(2, 4));
-    LVar0 := create_object_sid(51, 0, 0, -1);
+    LVar0 := create_object(51, 0, 0);
     add_mult_objs_to_inven(dude_obj, LVar0, random(2, 4));
-    LVar0 := create_object_sid(85, 0, 0, -1);
+    LVar0 := create_object(85, 0, 0);
     add_mult_objs_to_inven(dude_obj, LVar0, random(2, 4));
 end
 
@@ -1012,7 +1012,6 @@ procedure Egg_Conversant
 begin
     variable LVar0 := 0;
     debug_msg("Launched Speech Easter Egg!");
-    LVar0 := create_object_sid(120, 0, 0, -1);
+    LVar0 := create_object(120, 0, 0);
     add_obj_to_inven(dude_obj, LVar0);
 end
-
