diff --git a/Games/Fallout2.Steam/Original/RCCAMINT.ssl b/Games/Fallout2.Steam/ReDefine/RCCAMINT.ssl
index aed86ae..c277066 100644
--- a/Games/Fallout2.Steam/Original/RCCAMINT.ssl
+++ b/Games/Fallout2.Steam/ReDefine/RCCAMINT.ssl
@@ -181,10 +181,10 @@ end
 
 procedure critter_p_proc
 begin
-    if ((global_var(492) == 1) and obj_can_see_obj(self_obj, dude_obj)) then begin
-        attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+    if ((global_var(GVAR_ENEMY_REDDING) == 1) and obj_can_see_obj(self_obj, dude_obj)) then begin
+        attack(dude_obj);
     end
-    if (global_var(380) == 0) then begin
+    if (global_var(GVAR_BAR_BRAWL) == 0) then begin
         set_obj_visibility(self_obj, 1);
     end
     else begin
@@ -192,7 +192,7 @@ begin
     end
     if ((local_var(5) == 2) and obj_can_see_obj(self_obj, dude_obj)) then begin
         set_local_var(5, 1);
-        attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+        attack(dude_obj);
     end
 end
 
@@ -208,48 +208,48 @@ begin
     Evil_Critter := 0;
     Slavery_Tolerant := 2;
     Karma_Perception := 10;
-    set_global_var(37, 0);
-    set_global_var(38, 0);
-    set_global_var(39, 0);
-    set_global_var(40, 0);
-    set_global_var(41, 0);
-    set_global_var(42, 0);
-    set_global_var(43, 0);
-    set_global_var(44, 0);
-    set_global_var(45, 0);
-    if (global_var(0) >= 1000) then begin
-        set_global_var(37, 1);
+    set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+    set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+    set_global_var(GVAR_KARMA_DEFENDER, 0);
+    set_global_var(GVAR_KARMA_WANDERER, 0);
+    set_global_var(GVAR_KARMA_BETRAYER, 0);
+    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+    set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+    set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+    if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
     end
     else begin
-        if (global_var(0) >= 750) then begin
-            set_global_var(38, 1);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+            set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
         end
         else begin
-            if (global_var(0) >= 500) then begin
-                set_global_var(39, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
             end
             else begin
-                if (global_var(0) >= 250) then begin
-                    set_global_var(40, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                    set_global_var(GVAR_KARMA_DEFENDER, 1);
                 end
                 else begin
-                    if (global_var(0) > -250) then begin
-                        set_global_var(41, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                        set_global_var(GVAR_KARMA_WANDERER, 1);
                     end
                     else begin
-                        if (global_var(0) > -500) then begin
-                            set_global_var(42, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                            set_global_var(GVAR_KARMA_BETRAYER, 1);
                         end
                         else begin
-                            if (global_var(0) > -750) then begin
-                                set_global_var(43, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                             end
                             else begin
-                                if (global_var(0) > -1000) then begin
-                                    set_global_var(44, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                    set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                 end
                                 else begin
-                                    set_global_var(45, 1);
+                                    set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                 end
                             end
                         end
@@ -260,7 +260,7 @@ begin
     end
     if (local_var(2) == 0) then begin
         set_local_var(2, 1);
-        set_local_var(3, (get_critter_stat(dude_obj, 3) - get_critter_stat(self_obj, 3)) * 5);
+        set_local_var(3, (dude_charisma - self_charisma) * 5);
         debug_msg("Base Reaction == " + local_var(3));
     end
     Static_Reaction := 0;
@@ -272,14 +272,14 @@ begin
     else begin
         reaction_bonus_karma := -1 * (-1 * reaction_bonus_karma / 1000);
     end
-    if (global_var(56) >= 0) then begin
-        reaction_bonus_town_rep := global_var(56) / 2;
+    if (global_var(GVAR_TOWN_REP_REDDING) >= 0) then begin
+        reaction_bonus_town_rep := global_var(GVAR_TOWN_REP_REDDING) / 2;
     end
     else begin
-        reaction_bonus_town_rep := -1 * (-1 * global_var(56) / 2);
+        reaction_bonus_town_rep := -1 * (-1 * global_var(GVAR_TOWN_REP_REDDING) / 2);
     end
-    if (has_trait(2, dude_obj, 13)) then begin
-        if (get_critter_stat(dude_obj, 34) == get_critter_stat(self_obj, 34)) then begin
+    if (has_trait(TRAIT_TRAIT, dude_obj, TRAIT_sex_appeal)) then begin
+        if (dude_gender == self_gender) then begin
             Static_Reaction := Static_Reaction + -20;
         end
         else begin
@@ -295,48 +295,48 @@ begin
         debug_msg("Karma Reaction Bonus == " + Static_Reaction);
         Static_Reaction := Static_Reaction - reaction_bonus_town_rep;
         debug_msg("Town Rep Bonus == " + Static_Reaction);
-        set_global_var(37, 0);
-        set_global_var(38, 0);
-        set_global_var(39, 0);
-        set_global_var(40, 0);
-        set_global_var(41, 0);
-        set_global_var(42, 0);
-        set_global_var(43, 0);
-        set_global_var(44, 0);
-        set_global_var(45, 0);
-        if (global_var(0) >= 1000) then begin
-            set_global_var(37, 1);
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+        set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+        set_global_var(GVAR_KARMA_DEFENDER, 0);
+        set_global_var(GVAR_KARMA_WANDERER, 0);
+        set_global_var(GVAR_KARMA_BETRAYER, 0);
+        set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+        set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+            set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
         end
         else begin
-            if (global_var(0) >= 750) then begin
-                set_global_var(38, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
             end
             else begin
-                if (global_var(0) >= 500) then begin
-                    set_global_var(39, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                 end
                 else begin
-                    if (global_var(0) >= 250) then begin
-                        set_global_var(40, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                        set_global_var(GVAR_KARMA_DEFENDER, 1);
                     end
                     else begin
-                        if (global_var(0) > -250) then begin
-                            set_global_var(41, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                            set_global_var(GVAR_KARMA_WANDERER, 1);
                         end
                         else begin
-                            if (global_var(0) > -500) then begin
-                                set_global_var(42, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                set_global_var(GVAR_KARMA_BETRAYER, 1);
                             end
                             else begin
-                                if (global_var(0) > -750) then begin
-                                    set_global_var(43, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -1000) then begin
-                                        set_global_var(44, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                     end
                                     else begin
-                                        set_global_var(45, 1);
+                                        set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                     end
                                 end
                             end
@@ -345,35 +345,35 @@ begin
                 end
             end
         end
-        if (global_var(37) == 1) then begin
+        if (global_var(GVAR_KARMA_HOLY_WARRIOR) == 1) then begin
             Static_Reaction := Static_Reaction - 20;
         end
         else begin
-            if (global_var(38) == 1) then begin
+            if (global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES) == 1) then begin
                 Static_Reaction := Static_Reaction - 15;
             end
             else begin
-                if (global_var(39) == 1) then begin
+                if (global_var(GVAR_KARMA_SHIELD_OF_HOPE) == 1) then begin
                     Static_Reaction := Static_Reaction - 10;
                 end
                 else begin
-                    if (global_var(40) == 1) then begin
+                    if (global_var(GVAR_KARMA_DEFENDER) == 1) then begin
                         Static_Reaction := Static_Reaction - 5;
                     end
                     else begin
-                        if (global_var(41) == 1) then begin
+                        if (global_var(GVAR_KARMA_WANDERER) == 1) then begin
                             Static_Reaction := Static_Reaction - 0;
                         end
                         else begin
-                            if (global_var(42) == 1) then begin
+                            if (global_var(GVAR_KARMA_BETRAYER) == 1) then begin
                                 Static_Reaction := Static_Reaction - -5;
                             end
                             else begin
-                                if (global_var(43) == 1) then begin
+                                if (global_var(GVAR_KARMA_SWORD_OF_DESPAIR) == 1) then begin
                                     Static_Reaction := Static_Reaction - -10;
                                 end
                                 else begin
-                                    if (global_var(44) == 1) then begin
+                                    if (global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE) == 1) then begin
                                         Static_Reaction := Static_Reaction - -15;
                                     end
                                     else begin
@@ -393,48 +393,48 @@ begin
         debug_msg("Karma Reaction Bonus == " + Static_Reaction);
         Static_Reaction := Static_Reaction + reaction_bonus_town_rep;
         debug_msg("Town Rep Bonus == " + Static_Reaction);
-        set_global_var(37, 0);
-        set_global_var(38, 0);
-        set_global_var(39, 0);
-        set_global_var(40, 0);
-        set_global_var(41, 0);
-        set_global_var(42, 0);
-        set_global_var(43, 0);
-        set_global_var(44, 0);
-        set_global_var(45, 0);
-        if (global_var(0) >= 1000) then begin
-            set_global_var(37, 1);
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+        set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+        set_global_var(GVAR_KARMA_DEFENDER, 0);
+        set_global_var(GVAR_KARMA_WANDERER, 0);
+        set_global_var(GVAR_KARMA_BETRAYER, 0);
+        set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+        set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+            set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
         end
         else begin
-            if (global_var(0) >= 750) then begin
-                set_global_var(38, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
             end
             else begin
-                if (global_var(0) >= 500) then begin
-                    set_global_var(39, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                 end
                 else begin
-                    if (global_var(0) >= 250) then begin
-                        set_global_var(40, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                        set_global_var(GVAR_KARMA_DEFENDER, 1);
                     end
                     else begin
-                        if (global_var(0) > -250) then begin
-                            set_global_var(41, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                            set_global_var(GVAR_KARMA_WANDERER, 1);
                         end
                         else begin
-                            if (global_var(0) > -500) then begin
-                                set_global_var(42, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                set_global_var(GVAR_KARMA_BETRAYER, 1);
                             end
                             else begin
-                                if (global_var(0) > -750) then begin
-                                    set_global_var(43, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -1000) then begin
-                                        set_global_var(44, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                     end
                                     else begin
-                                        set_global_var(45, 1);
+                                        set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                     end
                                 end
                             end
@@ -443,35 +443,35 @@ begin
                 end
             end
         end
-        if (global_var(37) == 1) then begin
+        if (global_var(GVAR_KARMA_HOLY_WARRIOR) == 1) then begin
             Static_Reaction := Static_Reaction + 20;
         end
         else begin
-            if (global_var(38) == 1) then begin
+            if (global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES) == 1) then begin
                 Static_Reaction := Static_Reaction + 15;
             end
             else begin
-                if (global_var(39) == 1) then begin
+                if (global_var(GVAR_KARMA_SHIELD_OF_HOPE) == 1) then begin
                     Static_Reaction := Static_Reaction + 10;
                 end
                 else begin
-                    if (global_var(40) == 1) then begin
+                    if (global_var(GVAR_KARMA_DEFENDER) == 1) then begin
                         Static_Reaction := Static_Reaction + 5;
                     end
                     else begin
-                        if (global_var(41) == 1) then begin
+                        if (global_var(GVAR_KARMA_WANDERER) == 1) then begin
                             Static_Reaction := Static_Reaction + 0;
                         end
                         else begin
-                            if (global_var(42) == 1) then begin
+                            if (global_var(GVAR_KARMA_BETRAYER) == 1) then begin
                                 Static_Reaction := Static_Reaction + -5;
                             end
                             else begin
-                                if (global_var(43) == 1) then begin
+                                if (global_var(GVAR_KARMA_SWORD_OF_DESPAIR) == 1) then begin
                                     Static_Reaction := Static_Reaction + -10;
                                 end
                                 else begin
-                                    if (global_var(44) == 1) then begin
+                                    if (global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE) == 1) then begin
                                         Static_Reaction := Static_Reaction + -15;
                                     end
                                     else begin
@@ -487,34 +487,34 @@ begin
         debug_msg("Good Person Reaction Bonus == " + Static_Reaction);
     end
     if (Slavery_Tolerant == 0) then begin
-        if (global_var(11) == 1) then begin
+        if (global_var(GVAR_REPUTATION_SLAVER) == 1) then begin
             Static_Reaction := Static_Reaction + (2 * -25);
         end
     end
     else begin
         if (Slavery_Tolerant == 1) then begin
-            if (global_var(11) == 1) then begin
+            if (global_var(GVAR_REPUTATION_SLAVER) == 1) then begin
                 Static_Reaction := Static_Reaction + -25;
             end
         end
         else begin
             if (Slavery_Tolerant == 3) then begin
-                if (global_var(11) == 1) then begin
+                if (global_var(GVAR_REPUTATION_SLAVER) == 1) then begin
                     Static_Reaction := Static_Reaction - -25;
                 end
             end
         end
     end
     debug_msg("Slaver + Aligned Reaction == " + Static_Reaction);
-    Static_Reaction := Static_Reaction + (10 * has_trait(0, dude_obj, 10));
+    Static_Reaction := Static_Reaction + (10 * has_trait(TRAIT_PERK, dude_obj, PERK_presence));
     debug_msg("Presence Reaction == " + Static_Reaction);
-    if (global_var(1) >= 2) then begin
+    if (global_var(GVAR_CHILDKILLER_REPUTATION) >= 2) then begin
         Static_Reaction := Static_Reaction + -30;
     end
     debug_msg("Childkiller Reaction == " + Static_Reaction);
     set_local_var(0, local_var(3) + Static_Reaction);
     debug_msg("Initial Reaction == " + local_var(0));
-    if (tile_in_tile_rect(14923, 14900, 17923, 17900, tile_num(self_obj)) and (cur_map_index == 64)) then begin
+    if (tile_in_tile_rect(14923, 14900, 17923, 17900, tile_num(self_obj)) and (CUR_MAP_REDDOWN)) then begin
         start_gdialog(823, self_obj, 4, -1, -1);
         gsay_start;
         call Node005();
@@ -522,7 +522,7 @@ begin
         end_dialogue;
     end
     else begin
-        if (global_var(380) == 5) then begin
+        if (global_var(GVAR_BAR_BRAWL) == 5) then begin
             call Node003();
         end
         else begin
@@ -534,59 +534,59 @@ end
 procedure destroy_p_proc
 begin
     if (source_obj == dude_obj) then begin
-        set_global_var(5, global_var(5) + 1);
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(4) > (3 * global_var(5))) or (global_var(2) == 1)) and (global_var(1) == 0)) then begin
-            set_global_var(2, 1);
-            set_global_var(3, 0);
+        set_global_var(GVAR_GOOD_MONSTER, global_var(GVAR_GOOD_MONSTER) + 1);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(4) > (3 * global_var(GVAR_GOOD_MONSTER))) or (global_var(GVAR_CHAMPION_REPUTATION) == 1)) and (global_var(GVAR_CHILDKILLER_REPUTATION) == 0)) then begin
+            set_global_var(GVAR_CHAMPION_REPUTATION, 1);
+            set_global_var(GVAR_BERSERKER_REPUTATION, 0);
         end
-        if (metarule(51, self_obj) == 2) then begin
+        if (critter_kill_type(self_obj) == KILL_TYPE_children_kills) then begin
             if (source_obj == dude_obj) then begin
-                set_global_var(2, 0);
-                set_global_var(1, global_var(1) + 1);
-                set_global_var(0, global_var(0) + -15);
+                set_global_var(GVAR_CHAMPION_REPUTATION, 0);
+                set_global_var(GVAR_CHILDKILLER_REPUTATION, global_var(GVAR_CHILDKILLER_REPUTATION) + 1);
+                set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + -15);
                 debug_msg("Player gains " + -15 + " Karma Points.");
-                set_global_var(37, 0);
-                set_global_var(38, 0);
-                set_global_var(39, 0);
-                set_global_var(40, 0);
-                set_global_var(41, 0);
-                set_global_var(42, 0);
-                set_global_var(43, 0);
-                set_global_var(44, 0);
-                set_global_var(45, 0);
-                if (global_var(0) >= 1000) then begin
-                    set_global_var(37, 1);
+                set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+                set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+                set_global_var(GVAR_KARMA_DEFENDER, 0);
+                set_global_var(GVAR_KARMA_WANDERER, 0);
+                set_global_var(GVAR_KARMA_BETRAYER, 0);
+                set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+                set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+                set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+                    set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
                 end
                 else begin
-                    if (global_var(0) >= 750) then begin
-                        set_global_var(38, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
                     end
                     else begin
-                        if (global_var(0) >= 500) then begin
-                            set_global_var(39, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                            set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                         end
                         else begin
-                            if (global_var(0) >= 250) then begin
-                                set_global_var(40, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                                set_global_var(GVAR_KARMA_DEFENDER, 1);
                             end
                             else begin
-                                if (global_var(0) > -250) then begin
-                                    set_global_var(41, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                                    set_global_var(GVAR_KARMA_WANDERER, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -500) then begin
-                                        set_global_var(42, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                        set_global_var(GVAR_KARMA_BETRAYER, 1);
                                     end
                                     else begin
-                                        if (global_var(0) > -750) then begin
-                                            set_global_var(43, 1);
+                                        if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                            set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                         end
                                         else begin
-                                            if (global_var(0) > -1000) then begin
-                                                set_global_var(44, 1);
+                                            if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                                set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                             end
                                             else begin
-                                                set_global_var(45, 1);
+                                                set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                             end
                                         end
                                     end
@@ -595,58 +595,58 @@ begin
                         end
                     end
                 end
-                set_global_var(56, global_var(56) + -8);
+                set_global_var(GVAR_TOWN_REP_REDDING, global_var(GVAR_TOWN_REP_REDDING) + -8);
                 debug_msg("Added " + -8 + " to Town Rep");
             end
         end
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(5) > (2 * global_var(4))) or (global_var(3) == 1))) then begin
-            set_global_var(3, 1);
-            set_global_var(2, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
+            set_global_var(GVAR_BERSERKER_REPUTATION, 1);
+            set_global_var(GVAR_CHAMPION_REPUTATION, 0);
         end
-        set_global_var(0, global_var(0) + -10);
+        set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + -10);
         debug_msg("Player gains " + -10 + " Karma Points.");
-        set_global_var(37, 0);
-        set_global_var(38, 0);
-        set_global_var(39, 0);
-        set_global_var(40, 0);
-        set_global_var(41, 0);
-        set_global_var(42, 0);
-        set_global_var(43, 0);
-        set_global_var(44, 0);
-        set_global_var(45, 0);
-        if (global_var(0) >= 1000) then begin
-            set_global_var(37, 1);
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+        set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+        set_global_var(GVAR_KARMA_DEFENDER, 0);
+        set_global_var(GVAR_KARMA_WANDERER, 0);
+        set_global_var(GVAR_KARMA_BETRAYER, 0);
+        set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+        set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+            set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
         end
         else begin
-            if (global_var(0) >= 750) then begin
-                set_global_var(38, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
             end
             else begin
-                if (global_var(0) >= 500) then begin
-                    set_global_var(39, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                 end
                 else begin
-                    if (global_var(0) >= 250) then begin
-                        set_global_var(40, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                        set_global_var(GVAR_KARMA_DEFENDER, 1);
                     end
                     else begin
-                        if (global_var(0) > -250) then begin
-                            set_global_var(41, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                            set_global_var(GVAR_KARMA_WANDERER, 1);
                         end
                         else begin
-                            if (global_var(0) > -500) then begin
-                                set_global_var(42, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                set_global_var(GVAR_KARMA_BETRAYER, 1);
                             end
                             else begin
-                                if (global_var(0) > -750) then begin
-                                    set_global_var(43, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -1000) then begin
-                                        set_global_var(44, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                     end
                                     else begin
-                                        set_global_var(45, 1);
+                                        set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                     end
                                 end
                             end
@@ -655,11 +655,11 @@ begin
                 end
             end
         end
-        set_global_var(56, global_var(56) + -5);
+        set_global_var(GVAR_TOWN_REP_REDDING, global_var(GVAR_TOWN_REP_REDDING) + -5);
         debug_msg("Added " + -5 + " to Town Rep");
     end
-    set_global_var(380, 6);
-    set_global_var(492, 1);
+    set_global_var(GVAR_BAR_BRAWL, 6);
+    set_global_var(GVAR_ENEMY_REDDING, 1);
 end
 
 procedure look_at_p_proc
@@ -667,17 +667,17 @@ begin
     script_overrides;
     if (local_var(9) == 0) then begin
         set_local_var(9, local_var(9) + 1);
-        display_msg(message_str(823, 100));
+        display_mstr(100);
     end
     else begin
-        display_msg(message_str(823, 101));
+        display_mstr(101);
     end
 end
 
 procedure description_p_proc
 begin
     script_overrides;
-    display_msg(message_str(823, 102));
+    display_mstr(102);
 end
 
 procedure use_skill_on_p_proc
@@ -695,8 +695,8 @@ procedure map_enter_p_proc
 begin
     Caminetto_Ptr := self_obj;
     Only_Once := 0;
-    critter_add_trait(self_obj, 1, 6, 152);
-    critter_add_trait(self_obj, 1, 5, 140);
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 152);
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_REDDING_MINER);
 end
 
 procedure timed_event_p_proc
@@ -745,10 +745,10 @@ begin
     end
     else begin
         if ((LVar0 % 2) == 1) then begin
-            float_msg(Bette_Ptr, message_str(823, LVar0), 2);
+            float_msg(Bette_Ptr, mstr(LVar0), FLOAT_MSG_RED);
         end
         else begin
-            float_msg(Caminetto_Ptr, message_str(823, LVar0), 3);
+            float_msg(Caminetto_Ptr, mstr(LVar0), FLOAT_MSG_GREEN);
         end
         set_local_var(7, local_var(7) + 1);
         add_timer_event(self_obj, game_ticks(10), 1);
@@ -761,14 +761,14 @@ begin
     LVar0 := 131 + local_var(10);
     if (LVar0 > 133) then begin
         attack_setup(Caminetto_Ptr, dude_obj);
-        attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+        attack(dude_obj);
     end
     else begin
         if ((LVar0 % 2) == 1) then begin
-            float_msg(Bette_Ptr, message_str(823, LVar0), 2);
+            float_msg(Bette_Ptr, mstr(LVar0), FLOAT_MSG_RED);
         end
         else begin
-            float_msg(Caminetto_Ptr, message_str(823, LVar0), 3);
+            float_msg(Caminetto_Ptr, mstr(LVar0), FLOAT_MSG_GREEN);
         end
         set_local_var(7, local_var(7) + 1);
         add_timer_event(self_obj, game_ticks(10), 2);
@@ -777,7 +777,7 @@ end
 
 procedure Node003
 begin
-    float_msg(self_obj, message_str(823, random(134, 138)), 8);
+    float_msg(self_obj, mstr(random(134, 138)), FLOAT_MSG_YELLOW);
     if (local_var(12) == 0) then begin
         set_local_var(12, local_var(12) + 1);
         add_timer_event(self_obj, game_ticks(random(25, 35)), 3);
@@ -786,7 +786,7 @@ end
 
 procedure Node004
 begin
-    float_msg(self_obj, message_str(823, random(139, 143)), 8);
+    float_msg(self_obj, mstr(random(139, 143)), FLOAT_MSG_YELLOW);
     if (local_var(12) == 0) then begin
         set_local_var(12, local_var(12) + 1);
         add_timer_event(self_obj, game_ticks(random(25, 35)), 4);
@@ -795,92 +795,92 @@ end
 
 procedure Node005
 begin
-    gsay_reply(823, 144);
-    giq_option(4, 823, 145, Node008, 51);
-    giq_option(4, 823, 146, Node006, 50);
+    Reply(144);
+    giq_option(4, NAME, 145, Node008, BAD_REACTION);
+    giq_option(4, NAME, 146, Node006, NEUTRAL_REACTION);
 end
 
 procedure Node006
 begin
-    gsay_reply(823, 147);
-    giq_option(4, 823, 148, Node007, 50);
-    giq_option(4, 823, 149, Node008, 51);
+    Reply(147);
+    giq_option(4, NAME, 148, Node007, NEUTRAL_REACTION);
+    giq_option(4, NAME, 149, Node008, BAD_REACTION);
 end
 
 procedure Node007
 begin
-    gsay_reply(823, 150);
-    giq_option(4, 823, 151, Node008, 50);
-    giq_option(4, 823, 152, Node999, 49);
+    Reply(150);
+    giq_option(4, NAME, 151, Node008, NEUTRAL_REACTION);
+    giq_option(4, NAME, 152, Node999, GOOD_REACTION);
 end
 
 procedure Node008
 begin
-    gsay_reply(823, 153);
-    giq_option(4, 823, 154, Node012, 51);
-    giq_option(4, 823, 155, Node008a, 51);
-    giq_option(4, 823, 156, Node008b, 49);
+    Reply(153);
+    giq_option(4, NAME, 154, Node012, BAD_REACTION);
+    giq_option(4, NAME, 155, Node008a, BAD_REACTION);
+    giq_option(4, NAME, 156, Node008b, GOOD_REACTION);
 end
 
 procedure Node009
 begin
-    gsay_reply(823, 157);
-    giq_option(4, 823, 158, Node999, 49);
+    Reply(157);
+    giq_option(4, NAME, 158, Node999, GOOD_REACTION);
 end
 
 procedure Node010
 begin
-    gsay_reply(823, 159);
-    giq_option(4, 823, 160, Node010a, 51);
-    giq_option(4, 823, 161, Node999, 49);
+    Reply(159);
+    giq_option(4, NAME, 160, Node010a, BAD_REACTION);
+    giq_option(4, NAME, 161, Node999, GOOD_REACTION);
 end
 
 procedure Node011
 begin
-    gsay_reply(823, 162);
-    giq_option(4, 823, 163, Node008a, 51);
-    giq_option(4, 823, 164, Node011a, 51);
+    Reply(162);
+    giq_option(4, NAME, 163, Node008a, BAD_REACTION);
+    giq_option(4, NAME, 164, Node011a, BAD_REACTION);
 end
 
 procedure Node012
 begin
-    gsay_reply(823, 165);
-    giq_option(4, 823, 166, Node015, 49);
-    giq_option(4, 823, 167, Node008a, 51);
+    Reply(165);
+    giq_option(4, NAME, 166, Node015, GOOD_REACTION);
+    giq_option(4, NAME, 167, Node008a, BAD_REACTION);
 end
 
 procedure Node013
 begin
-    gsay_reply(823, 168);
-    giq_option(4, 823, 169, Node013a, 50);
-    giq_option(4, 823, 170, Node016, 49);
+    Reply(168);
+    giq_option(4, NAME, 169, Node013a, NEUTRAL_REACTION);
+    giq_option(4, NAME, 170, Node016, GOOD_REACTION);
 end
 
 procedure Node014
 begin
-    gsay_reply(823, 171);
-    giq_option(4, 823, 172, Node014a, 50);
-    giq_option(4, 823, 173, Node014b, 51);
-    giq_option(4, 823, 174, Node999, 49);
+    Reply(171);
+    giq_option(4, NAME, 172, Node014a, NEUTRAL_REACTION);
+    giq_option(4, NAME, 173, Node014b, BAD_REACTION);
+    giq_option(4, NAME, 174, Node999, GOOD_REACTION);
 end
 
 procedure Node015
 begin
-    gsay_reply(823, 175);
-    giq_option(4, 823, 176, Node008a, 50);
-    giq_option(4, 823, 177, Node999, 50);
+    Reply(175);
+    giq_option(4, NAME, 176, Node008a, NEUTRAL_REACTION);
+    giq_option(4, NAME, 177, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node016
 begin
-    gsay_reply(823, 178);
-    giq_option(4, 823, 179, Node011a, 50);
-    giq_option(4, 823, 180, Node998, 51);
+    Reply(178);
+    giq_option(4, NAME, 179, Node011a, NEUTRAL_REACTION);
+    giq_option(4, NAME, 180, Node998, BAD_REACTION);
 end
 
 procedure Node008a
 begin
-    if (is_success(roll_vs_skill(dude_obj, 14, 2 * get_pc_stat(1)))) then begin
+    if (is_success(roll_vs_skill(dude_obj, 14, 2 * get_pc_stat(PCSTAT_level)))) then begin
         call Node013();
     end
     else begin
@@ -900,7 +900,7 @@ end
 
 procedure Node010a
 begin
-    if (is_success(roll_vs_skill(dude_obj, 14, 2 * get_pc_stat(1)))) then begin
+    if (is_success(roll_vs_skill(dude_obj, 14, 2 * get_pc_stat(PCSTAT_level)))) then begin
         call Node011();
     end
     else begin
@@ -916,8 +916,8 @@ begin
     move_to(Caminetto_Ptr, 24081, 0);
     move_to(dude_obj, 23286, 0);
     give_exp_points(1500);
-    display_msg(message_str(14, 100) + 1500 + message_str(14, 101));
-    set_global_var(380, 5);
+    display_msg(g_mstr(100) + 1500 + g_mstr(101));
+    set_global_var(GVAR_BAR_BRAWL, 5);
 end
 
 procedure Node013a
@@ -929,13 +929,13 @@ begin
     set_obj_visibility(Bette_Ptr, 1);
     move_to(dude_obj, 23286, 0);
     give_exp_points(1500);
-    display_msg(message_str(14, 100) + 1500 + message_str(14, 101));
-    set_global_var(380, 3);
+    display_msg(g_mstr(100) + 1500 + g_mstr(101));
+    set_global_var(GVAR_BAR_BRAWL, 3);
 end
 
 procedure Node014a
 begin
-    if (is_success(roll_vs_skill(dude_obj, 14, 2 * get_pc_stat(1)))) then begin
+    if (is_success(roll_vs_skill(dude_obj, 14, 2 * get_pc_stat(PCSTAT_level)))) then begin
         call Node016();
     end
     else begin
@@ -945,11 +945,10 @@ end
 
 procedure Node014b
 begin
-    if (is_success(roll_vs_skill(dude_obj, 14, 2 * get_pc_stat(1)))) then begin
+    if (is_success(roll_vs_skill(dude_obj, 14, 2 * get_pc_stat(PCSTAT_level)))) then begin
         call Node016();
     end
     else begin
         call Node998();
     end
 end
-
