diff --git a/Games/Fallout2.Steam/Original/REDDOWN.ssl b/Games/Fallout2.Steam/ReDefine/REDDOWN.ssl
index 417ed23..ef06d08 100644
--- a/Games/Fallout2.Steam/Original/REDDOWN.ssl
+++ b/Games/Fallout2.Steam/ReDefine/REDDOWN.ssl
@@ -144,106 +144,106 @@ procedure map_enter_p_proc
 begin
     variable LVar0 := 0;
     debug_msg("Check_Create_Car(" + 18650 + "," + 0 + ")");
-    if (not(metarule(22, 0))) then begin
-        if (global_var(18) != 0) then begin
-            debug_msg("car_current_town == " + metarule(30, 0) + " / map_get_load_area == " + metarule3(111, 0, 0, 0));
-            if ((metarule(30, 0) == metarule3(111, 0, 0, 0)) or ((26 <= metarule(30, 0)) and (metarule(30, 0) <= 29) and (metarule3(111, 0, 0, 0) == -1))) then begin
+    if (not(is_loading_game)) then begin
+        if (global_var(GVAR_PLAYER_GOT_CAR) != 0) then begin
+            debug_msg("car_current_town == " + car_current_town + " / map_get_load_area == " + map_get_load_area);
+            if ((car_current_town == map_get_load_area) or ((26 <= car_current_town) and (car_current_town <= 29) and (map_get_load_area == -1))) then begin
                 debug_msg("Place_Car(" + 18650 + "," + 0 + ")");
-                debug_msg("   GVAR_CAR_PLACED_TILE == " + global_var(633));
+                debug_msg("   GVAR_CAR_PLACED_TILE == " + global_var(GVAR_CAR_PLACED_TILE));
                 debug_msg("   Car_At_Loc == " + (tile_contains_pid_obj(18650, 0, 33555441) != 0));
-                if ((global_var(633) <= 0) or (18650 == global_var(633))) then begin
-                    set_global_var(633, 18650);
+                if ((global_var(GVAR_CAR_PLACED_TILE) <= 0) or (18650 == global_var(GVAR_CAR_PLACED_TILE))) then begin
+                    set_global_var(GVAR_CAR_PLACED_TILE, 18650);
                     if (not(tile_contains_pid_obj(18650, 0, 33555441) != 0)) then begin
-                        Scenery_Creation := create_object_sid(33555441, 18650, 0, 304);
+                        Scenery_Creation := create_object_sid(33555441, 18650, 0, SCRIPT_ZSDRVCAR);
                         Scenery_Creation_Hex := 18650;
                         Scenery_Creation_Count := 0;
                         while (Scenery_Creation_Count < 2) do begin
                             Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 1, 1);
-                            Scenery_Creation := create_object_sid(33554499, Scenery_Creation_Hex, 0, -1);
+                            Scenery_Creation := create_object(33554499, Scenery_Creation_Hex, 0);
                             Scenery_Creation_Count := Scenery_Creation_Count + 1;
                         end
                         Scenery_Creation_Count := 0;
                         while (Scenery_Creation_Count < 2) do begin
                             Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 2, 1);
-                            Scenery_Creation := create_object_sid(33554499, Scenery_Creation_Hex, 0, -1);
+                            Scenery_Creation := create_object(33554499, Scenery_Creation_Hex, 0);
                             Scenery_Creation_Count := Scenery_Creation_Count + 1;
                         end
                         Scenery_Creation_Count := 0;
                         while (Scenery_Creation_Count < 2) do begin
                             Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 3, 1);
-                            Scenery_Creation := create_object_sid(33554499, Scenery_Creation_Hex, 0, -1);
+                            Scenery_Creation := create_object(33554499, Scenery_Creation_Hex, 0);
                             Scenery_Creation_Count := Scenery_Creation_Count + 1;
                         end
                         Scenery_Creation_Count := 0;
                         while (Scenery_Creation_Count < 2) do begin
                             Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 4, 1);
-                            Scenery_Creation := create_object_sid(33554499, Scenery_Creation_Hex, 0, -1);
+                            Scenery_Creation := create_object(33554499, Scenery_Creation_Hex, 0);
                             Scenery_Creation_Count := Scenery_Creation_Count + 1;
                         end
                         Scenery_Creation_Count := 0;
                         while (Scenery_Creation_Count < 1) do begin
                             Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 5, 1);
-                            Scenery_Creation := create_object_sid(33554499, Scenery_Creation_Hex, 0, -1);
+                            Scenery_Creation := create_object(33554499, Scenery_Creation_Hex, 0);
                             Scenery_Creation_Count := Scenery_Creation_Count + 1;
                         end
                         Scenery_Creation_Count := 0;
                         while (Scenery_Creation_Count < 1) do begin
                             Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 4, 1);
-                            Scenery_Creation := create_object_sid(33554499, Scenery_Creation_Hex, 0, -1);
+                            Scenery_Creation := create_object(33554499, Scenery_Creation_Hex, 0);
                             Scenery_Creation_Count := Scenery_Creation_Count + 1;
                         end
                         Scenery_Creation_Count := 0;
                         while (Scenery_Creation_Count < 1) do begin
                             Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 5, 1);
-                            Scenery_Creation := create_object_sid(33554499, Scenery_Creation_Hex, 0, -1);
+                            Scenery_Creation := create_object(33554499, Scenery_Creation_Hex, 0);
                             Scenery_Creation_Count := Scenery_Creation_Count + 1;
                         end
                         Scenery_Creation_Count := 0;
                         while (Scenery_Creation_Count < 1) do begin
                             Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 4, 1);
-                            Scenery_Creation := create_object_sid(33554499, Scenery_Creation_Hex, 0, -1);
+                            Scenery_Creation := create_object(33554499, Scenery_Creation_Hex, 0);
                             Scenery_Creation_Count := Scenery_Creation_Count + 1;
                         end
                         Scenery_Creation_Count := 0;
                         while (Scenery_Creation_Count < 1) do begin
                             Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 5, 1);
-                            Scenery_Creation := create_object_sid(33554499, Scenery_Creation_Hex, 0, -1);
+                            Scenery_Creation := create_object(33554499, Scenery_Creation_Hex, 0);
                             Scenery_Creation_Count := Scenery_Creation_Count + 1;
                         end
                         Scenery_Creation_Count := 0;
                         while (Scenery_Creation_Count < 1) do begin
                             Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 4, 1);
-                            Scenery_Creation := create_object_sid(33554499, Scenery_Creation_Hex, 0, -1);
+                            Scenery_Creation := create_object(33554499, Scenery_Creation_Hex, 0);
                             Scenery_Creation_Count := Scenery_Creation_Count + 1;
                         end
                         Scenery_Creation_Count := 0;
                         while (Scenery_Creation_Count < 1) do begin
                             Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 0, 1);
-                            Scenery_Creation := create_object_sid(33554499, Scenery_Creation_Hex, 0, -1);
+                            Scenery_Creation := create_object(33554499, Scenery_Creation_Hex, 0);
                             Scenery_Creation_Count := Scenery_Creation_Count + 1;
                         end
                         Scenery_Creation_Count := 0;
                         while (Scenery_Creation_Count < 1) do begin
                             Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 5, 1);
-                            Scenery_Creation := create_object_sid(33554499, Scenery_Creation_Hex, 0, -1);
+                            Scenery_Creation := create_object(33554499, Scenery_Creation_Hex, 0);
                             Scenery_Creation_Count := Scenery_Creation_Count + 1;
                         end
                         Scenery_Creation_Count := 0;
                         while (Scenery_Creation_Count < 2) do begin
                             Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 0, 1);
-                            Scenery_Creation := create_object_sid(33554499, Scenery_Creation_Hex, 0, -1);
+                            Scenery_Creation := create_object(33554499, Scenery_Creation_Hex, 0);
                             Scenery_Creation_Count := Scenery_Creation_Count + 1;
                         end
                         Scenery_Creation_Count := 0;
                         while (Scenery_Creation_Count < 1) do begin
                             Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 1, 1);
-                            Scenery_Creation := create_object_sid(33554499, Scenery_Creation_Hex, 0, -1);
+                            Scenery_Creation := create_object(33554499, Scenery_Creation_Hex, 0);
                             Scenery_Creation_Count := Scenery_Creation_Count + 1;
                         end
                         Scenery_Creation_Count := 0;
                         while (Scenery_Creation_Count < 1) do begin
                             Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 2, 1);
-                            Scenery_Creation := create_object_sid(33554499, Scenery_Creation_Hex, 0, -1);
+                            Scenery_Creation := create_object(33554499, Scenery_Creation_Hex, 0);
                             Scenery_Creation_Count := Scenery_Creation_Count + 1;
                         end
                     end
@@ -256,8 +256,8 @@ begin
                             Scenery_Creation_Ptr := party_member_obj(455);
                         end
                         else begin
-                            if (global_var(18) == 0) then begin
-                                Scenery_Creation_Ptr := create_object_sid(455, 0, 0, 920);
+                            if (global_var(GVAR_PLAYER_GOT_CAR) == 0) then begin
+                                Scenery_Creation_Ptr := create_object_sid(455, 0, 0, SCRIPT_ZICRTRNK);
                             end
                         end
                         if (Scenery_Creation_Ptr != 0) then begin
@@ -266,13 +266,13 @@ begin
                             Scenery_Creation_Count := 0;
                             while (Scenery_Creation_Count < 1) do begin
                                 Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 1, 1);
-                                Scenery_Creation := create_object_sid(33554499, Scenery_Creation_Hex, 0, -1);
+                                Scenery_Creation := create_object(33554499, Scenery_Creation_Hex, 0);
                                 Scenery_Creation_Count := Scenery_Creation_Count + 1;
                             end
                             Scenery_Creation_Count := 0;
                             while (Scenery_Creation_Count < 1) do begin
                                 Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 2, 1);
-                                Scenery_Creation := create_object_sid(33554499, Scenery_Creation_Hex, 0, -1);
+                                Scenery_Creation := create_object(33554499, Scenery_Creation_Hex, 0);
                                 Scenery_Creation_Count := Scenery_Creation_Count + 1;
                             end
                         end
@@ -428,9 +428,9 @@ begin
     else begin
         debug_msg("sorry, can't place the car, the game's loading");
     end
-    if (global_var(27) == 1) then begin
+    if (global_var(GVAR_LOAD_MAP_INDEX) == 1) then begin
         override_map_start(25686 % 200, 25686 / 200, 0, 0);
-        LVar0 := create_object_sid(16777488, 0, 0, 678);
+        LVar0 := create_object_sid(16777488, 0, 0, SCRIPT_RCHAKES);
         move_to(LVar0, 24280, 0);
     end
     else begin
@@ -512,7 +512,7 @@ begin
             end
         end
     end
-    set_global_var(27, 0);
+    set_global_var(GVAR_LOAD_MAP_INDEX, 0);
 end
 
 procedure map_update_p_proc
@@ -594,4 +594,3 @@ begin
         end
     end
 end
-
