diff --git a/Games/Fallout2.Steam/Original/REDWAN1.ssl b/Games/Fallout2.Steam/ReDefine/REDWAN1.ssl
index 0f4da5b..9a2d162 100644
--- a/Games/Fallout2.Steam/Original/REDWAN1.ssl
+++ b/Games/Fallout2.Steam/ReDefine/REDWAN1.ssl
@@ -138,11 +138,10 @@ end
 procedure map_enter_p_proc
 begin
     set_light_level(50);
-    set_global_var(27, 0);
+    set_global_var(GVAR_LOAD_MAP_INDEX, 0);
 end
 
 procedure map_update_p_proc
 begin
     set_light_level(50);
 end
-
