diff --git a/Games/Fallout2.Steam/Original/REDWAME.ssl b/Games/Fallout2.Steam/ReDefine/REDWAME.ssl
index 37d29f6..11b11c9 100644
--- a/Games/Fallout2.Steam/Original/REDWAME.ssl
+++ b/Games/Fallout2.Steam/ReDefine/REDWAME.ssl
@@ -216,83 +216,83 @@ begin
             end
         end
     end
-    if ((map_var(0) == 0) and (global_var(385) == 1)) then begin
+    if ((map_var(0) == 0) and (global_var(GVAR_FROG_MORTON) == 1)) then begin
         set_map_var(0, map_var(0) + 1);
-        LVar0 := create_object_sid(16777479, 0, 0, 949);
+        LVar0 := create_object_sid(16777479, 0, 0, SCRIPT_RCFROG);
         critter_attempt_placement(LVar0, 13294, 0);
-        LVar2 := create_object_sid(352, 0, 0, -1);
+        LVar2 := create_object(352, 0, 0);
         add_obj_to_inven(LVar0, LVar2);
         wield_obj_critter(LVar0, LVar2);
-        LVar2 := create_object_sid(359, 0, 0, -1);
+        LVar2 := create_object(359, 0, 0);
         add_obj_to_inven(LVar0, LVar2);
-        LVar0 := create_object_sid(16777400, 0, 0, 1168);
+        LVar0 := create_object_sid(16777400, 0, 0, SCRIPT_RCMOTGNG);
         critter_attempt_placement(LVar0, 23097, 0);
-        LVar2 := create_object_sid(385, 0, 0, -1);
+        LVar2 := create_object(385, 0, 0);
         add_obj_to_inven(LVar0, LVar2);
         wield_obj_critter(LVar0, LVar2);
-        LVar2 := create_object_sid(95, 0, 0, -1);
+        LVar2 := create_object(95, 0, 0);
         add_obj_to_inven(LVar0, LVar2);
-        LVar0 := create_object_sid(16777400, 0, 0, 1168);
+        LVar0 := create_object_sid(16777400, 0, 0, SCRIPT_RCMOTGNG);
         critter_attempt_placement(LVar0, 22494, 0);
-        LVar2 := create_object_sid(385, 0, 0, -1);
+        LVar2 := create_object(385, 0, 0);
         add_obj_to_inven(LVar0, LVar2);
         wield_obj_critter(LVar0, LVar2);
-        LVar2 := create_object_sid(95, 0, 0, -1);
+        LVar2 := create_object(95, 0, 0);
         add_obj_to_inven(LVar0, LVar2);
-        LVar0 := create_object_sid(16777400, 0, 0, 1168);
+        LVar0 := create_object_sid(16777400, 0, 0, SCRIPT_RCMOTGNG);
         critter_attempt_placement(LVar0, 24707, 0);
-        LVar2 := create_object_sid(242, 0, 0, -1);
+        LVar2 := create_object(242, 0, 0);
         add_obj_to_inven(LVar0, LVar2);
         wield_obj_critter(LVar0, LVar2);
-        LVar2 := create_object_sid(95, 0, 0, -1);
+        LVar2 := create_object(95, 0, 0);
         add_obj_to_inven(LVar0, LVar2);
-        LVar0 := create_object_sid(16777400, 0, 0, 1168);
+        LVar0 := create_object_sid(16777400, 0, 0, SCRIPT_RCMOTGNG);
         critter_attempt_placement(LVar0, 14290, 0);
-        LVar2 := create_object_sid(242, 0, 0, -1);
+        LVar2 := create_object(242, 0, 0);
         add_obj_to_inven(LVar0, LVar2);
         wield_obj_critter(LVar0, LVar2);
-        LVar2 := create_object_sid(95, 0, 0, -1);
+        LVar2 := create_object(95, 0, 0);
         add_obj_to_inven(LVar0, LVar2);
-        LVar0 := create_object_sid(16777400, 0, 0, 1168);
+        LVar0 := create_object_sid(16777400, 0, 0, SCRIPT_RCMOTGNG);
         LVar1 := tile_num_in_direction(14290, 1, random(2, 4));
         critter_attempt_placement(LVar0, LVar1, 0);
-        LVar2 := create_object_sid(242, 0, 0, -1);
+        LVar2 := create_object(242, 0, 0);
         add_obj_to_inven(LVar0, LVar2);
         wield_obj_critter(LVar0, LVar2);
-        LVar2 := create_object_sid(95, 0, 0, -1);
+        LVar2 := create_object(95, 0, 0);
         add_obj_to_inven(LVar0, LVar2);
-        LVar0 := create_object_sid(16777400, 0, 0, 1168);
+        LVar0 := create_object_sid(16777400, 0, 0, SCRIPT_RCMOTGNG);
         LVar1 := tile_num_in_direction(14290, 3, random(2, 4));
         critter_attempt_placement(LVar0, LVar1, 0);
-        LVar2 := create_object_sid(242, 0, 0, -1);
+        LVar2 := create_object(242, 0, 0);
         add_obj_to_inven(LVar0, LVar2);
         wield_obj_critter(LVar0, LVar2);
-        LVar2 := create_object_sid(95, 0, 0, -1);
+        LVar2 := create_object(95, 0, 0);
         add_obj_to_inven(LVar0, LVar2);
-        LVar0 := create_object_sid(16777400, 0, 0, 1168);
+        LVar0 := create_object_sid(16777400, 0, 0, SCRIPT_RCMOTGNG);
         LVar1 := tile_num_in_direction(14290, 5, random(2, 4));
         critter_attempt_placement(LVar0, LVar1, 0);
-        LVar2 := create_object_sid(242, 0, 0, -1);
+        LVar2 := create_object(242, 0, 0);
         add_obj_to_inven(LVar0, LVar2);
         wield_obj_critter(LVar0, LVar2);
-        LVar2 := create_object_sid(95, 0, 0, -1);
+        LVar2 := create_object(95, 0, 0);
         add_obj_to_inven(LVar0, LVar2);
-        LVar0 := create_object_sid(16777400, 0, 0, 1168);
+        LVar0 := create_object_sid(16777400, 0, 0, SCRIPT_RCMOTGNG);
         critter_attempt_placement(LVar0, 17102, 0);
-        LVar2 := create_object_sid(385, 0, 0, -1);
+        LVar2 := create_object(385, 0, 0);
         add_obj_to_inven(LVar0, LVar2);
         wield_obj_critter(LVar0, LVar2);
-        LVar2 := create_object_sid(95, 0, 0, -1);
+        LVar2 := create_object(95, 0, 0);
         add_obj_to_inven(LVar0, LVar2);
-        LVar0 := create_object_sid(16777400, 0, 0, 1168);
+        LVar0 := create_object_sid(16777400, 0, 0, SCRIPT_RCMOTGNG);
         critter_attempt_placement(LVar0, 16700, 0);
-        LVar2 := create_object_sid(385, 0, 0, -1);
+        LVar2 := create_object(385, 0, 0);
         add_obj_to_inven(LVar0, LVar2);
         wield_obj_critter(LVar0, LVar2);
-        LVar2 := create_object_sid(95, 0, 0, -1);
+        LVar2 := create_object(95, 0, 0);
         add_obj_to_inven(LVar0, LVar2);
     end
-    set_global_var(27, 0);
+    set_global_var(GVAR_LOAD_MAP_INDEX, 0);
 end
 
 procedure map_update_p_proc
@@ -374,4 +374,3 @@ begin
         end
     end
 end
-
