diff --git a/Games/Fallout2.Steam/Original/SCMIRA.ssl b/Games/Fallout2.Steam/ReDefine/SCMIRA.ssl
index c260c37..a1554ad 100644
--- a/Games/Fallout2.Steam/Original/SCMIRA.ssl
+++ b/Games/Fallout2.Steam/ReDefine/SCMIRA.ssl
@@ -192,7 +192,7 @@ procedure critter_p_proc
 begin
     if ((local_var(5) == 2) and obj_can_see_obj(self_obj, dude_obj)) then begin
         set_local_var(5, 1);
-        attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+        attack(dude_obj);
     end
 end
 
@@ -208,48 +208,48 @@ begin
     Evil_Critter := 1;
     Slavery_Tolerant := 2;
     Karma_Perception := 10;
-    set_global_var(37, 0);
-    set_global_var(38, 0);
-    set_global_var(39, 0);
-    set_global_var(40, 0);
-    set_global_var(41, 0);
-    set_global_var(42, 0);
-    set_global_var(43, 0);
-    set_global_var(44, 0);
-    set_global_var(45, 0);
-    if (global_var(0) >= 1000) then begin
-        set_global_var(37, 1);
+    set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+    set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+    set_global_var(GVAR_KARMA_DEFENDER, 0);
+    set_global_var(GVAR_KARMA_WANDERER, 0);
+    set_global_var(GVAR_KARMA_BETRAYER, 0);
+    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+    set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+    set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+    if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
     end
     else begin
-        if (global_var(0) >= 750) then begin
-            set_global_var(38, 1);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+            set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
         end
         else begin
-            if (global_var(0) >= 500) then begin
-                set_global_var(39, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
             end
             else begin
-                if (global_var(0) >= 250) then begin
-                    set_global_var(40, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                    set_global_var(GVAR_KARMA_DEFENDER, 1);
                 end
                 else begin
-                    if (global_var(0) > -250) then begin
-                        set_global_var(41, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                        set_global_var(GVAR_KARMA_WANDERER, 1);
                     end
                     else begin
-                        if (global_var(0) > -500) then begin
-                            set_global_var(42, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                            set_global_var(GVAR_KARMA_BETRAYER, 1);
                         end
                         else begin
-                            if (global_var(0) > -750) then begin
-                                set_global_var(43, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                             end
                             else begin
-                                if (global_var(0) > -1000) then begin
-                                    set_global_var(44, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                    set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                 end
                                 else begin
-                                    set_global_var(45, 1);
+                                    set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                 end
                             end
                         end
@@ -260,7 +260,7 @@ begin
     end
     if (local_var(2) == 0) then begin
         set_local_var(2, 1);
-        set_local_var(3, (get_critter_stat(dude_obj, 3) - get_critter_stat(self_obj, 3)) * 5);
+        set_local_var(3, (dude_charisma - self_charisma) * 5);
         debug_msg("Base Reaction == " + local_var(3));
     end
     Static_Reaction := 0;
@@ -272,14 +272,14 @@ begin
     else begin
         reaction_bonus_karma := -1 * (-1 * reaction_bonus_karma / 1000);
     end
-    if (global_var(57) >= 0) then begin
-        reaction_bonus_town_rep := global_var(57) / 2;
+    if (global_var(GVAR_TOWN_REP_NCR) >= 0) then begin
+        reaction_bonus_town_rep := global_var(GVAR_TOWN_REP_NCR) / 2;
     end
     else begin
-        reaction_bonus_town_rep := -1 * (-1 * global_var(57) / 2);
+        reaction_bonus_town_rep := -1 * (-1 * global_var(GVAR_TOWN_REP_NCR) / 2);
     end
-    if (has_trait(2, dude_obj, 13)) then begin
-        if (get_critter_stat(dude_obj, 34) == get_critter_stat(self_obj, 34)) then begin
+    if (has_trait(TRAIT_TRAIT, dude_obj, TRAIT_sex_appeal)) then begin
+        if (dude_gender == self_gender) then begin
             Static_Reaction := Static_Reaction + -20;
         end
         else begin
@@ -295,48 +295,48 @@ begin
         debug_msg("Karma Reaction Bonus == " + Static_Reaction);
         Static_Reaction := Static_Reaction - reaction_bonus_town_rep;
         debug_msg("Town Rep Bonus == " + Static_Reaction);
-        set_global_var(37, 0);
-        set_global_var(38, 0);
-        set_global_var(39, 0);
-        set_global_var(40, 0);
-        set_global_var(41, 0);
-        set_global_var(42, 0);
-        set_global_var(43, 0);
-        set_global_var(44, 0);
-        set_global_var(45, 0);
-        if (global_var(0) >= 1000) then begin
-            set_global_var(37, 1);
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+        set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+        set_global_var(GVAR_KARMA_DEFENDER, 0);
+        set_global_var(GVAR_KARMA_WANDERER, 0);
+        set_global_var(GVAR_KARMA_BETRAYER, 0);
+        set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+        set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+            set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
         end
         else begin
-            if (global_var(0) >= 750) then begin
-                set_global_var(38, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
             end
             else begin
-                if (global_var(0) >= 500) then begin
-                    set_global_var(39, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                 end
                 else begin
-                    if (global_var(0) >= 250) then begin
-                        set_global_var(40, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                        set_global_var(GVAR_KARMA_DEFENDER, 1);
                     end
                     else begin
-                        if (global_var(0) > -250) then begin
-                            set_global_var(41, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                            set_global_var(GVAR_KARMA_WANDERER, 1);
                         end
                         else begin
-                            if (global_var(0) > -500) then begin
-                                set_global_var(42, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                set_global_var(GVAR_KARMA_BETRAYER, 1);
                             end
                             else begin
-                                if (global_var(0) > -750) then begin
-                                    set_global_var(43, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -1000) then begin
-                                        set_global_var(44, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                     end
                                     else begin
-                                        set_global_var(45, 1);
+                                        set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                     end
                                 end
                             end
@@ -345,35 +345,35 @@ begin
                 end
             end
         end
-        if (global_var(37) == 1) then begin
+        if (global_var(GVAR_KARMA_HOLY_WARRIOR) == 1) then begin
             Static_Reaction := Static_Reaction - 20;
         end
         else begin
-            if (global_var(38) == 1) then begin
+            if (global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES) == 1) then begin
                 Static_Reaction := Static_Reaction - 15;
             end
             else begin
-                if (global_var(39) == 1) then begin
+                if (global_var(GVAR_KARMA_SHIELD_OF_HOPE) == 1) then begin
                     Static_Reaction := Static_Reaction - 10;
                 end
                 else begin
-                    if (global_var(40) == 1) then begin
+                    if (global_var(GVAR_KARMA_DEFENDER) == 1) then begin
                         Static_Reaction := Static_Reaction - 5;
                     end
                     else begin
-                        if (global_var(41) == 1) then begin
+                        if (global_var(GVAR_KARMA_WANDERER) == 1) then begin
                             Static_Reaction := Static_Reaction - 0;
                         end
                         else begin
-                            if (global_var(42) == 1) then begin
+                            if (global_var(GVAR_KARMA_BETRAYER) == 1) then begin
                                 Static_Reaction := Static_Reaction - -5;
                             end
                             else begin
-                                if (global_var(43) == 1) then begin
+                                if (global_var(GVAR_KARMA_SWORD_OF_DESPAIR) == 1) then begin
                                     Static_Reaction := Static_Reaction - -10;
                                 end
                                 else begin
-                                    if (global_var(44) == 1) then begin
+                                    if (global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE) == 1) then begin
                                         Static_Reaction := Static_Reaction - -15;
                                     end
                                     else begin
@@ -393,48 +393,48 @@ begin
         debug_msg("Karma Reaction Bonus == " + Static_Reaction);
         Static_Reaction := Static_Reaction + reaction_bonus_town_rep;
         debug_msg("Town Rep Bonus == " + Static_Reaction);
-        set_global_var(37, 0);
-        set_global_var(38, 0);
-        set_global_var(39, 0);
-        set_global_var(40, 0);
-        set_global_var(41, 0);
-        set_global_var(42, 0);
-        set_global_var(43, 0);
-        set_global_var(44, 0);
-        set_global_var(45, 0);
-        if (global_var(0) >= 1000) then begin
-            set_global_var(37, 1);
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+        set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+        set_global_var(GVAR_KARMA_DEFENDER, 0);
+        set_global_var(GVAR_KARMA_WANDERER, 0);
+        set_global_var(GVAR_KARMA_BETRAYER, 0);
+        set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+        set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+            set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
         end
         else begin
-            if (global_var(0) >= 750) then begin
-                set_global_var(38, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
             end
             else begin
-                if (global_var(0) >= 500) then begin
-                    set_global_var(39, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                 end
                 else begin
-                    if (global_var(0) >= 250) then begin
-                        set_global_var(40, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                        set_global_var(GVAR_KARMA_DEFENDER, 1);
                     end
                     else begin
-                        if (global_var(0) > -250) then begin
-                            set_global_var(41, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                            set_global_var(GVAR_KARMA_WANDERER, 1);
                         end
                         else begin
-                            if (global_var(0) > -500) then begin
-                                set_global_var(42, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                set_global_var(GVAR_KARMA_BETRAYER, 1);
                             end
                             else begin
-                                if (global_var(0) > -750) then begin
-                                    set_global_var(43, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -1000) then begin
-                                        set_global_var(44, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                     end
                                     else begin
-                                        set_global_var(45, 1);
+                                        set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                     end
                                 end
                             end
@@ -443,35 +443,35 @@ begin
                 end
             end
         end
-        if (global_var(37) == 1) then begin
+        if (global_var(GVAR_KARMA_HOLY_WARRIOR) == 1) then begin
             Static_Reaction := Static_Reaction + 20;
         end
         else begin
-            if (global_var(38) == 1) then begin
+            if (global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES) == 1) then begin
                 Static_Reaction := Static_Reaction + 15;
             end
             else begin
-                if (global_var(39) == 1) then begin
+                if (global_var(GVAR_KARMA_SHIELD_OF_HOPE) == 1) then begin
                     Static_Reaction := Static_Reaction + 10;
                 end
                 else begin
-                    if (global_var(40) == 1) then begin
+                    if (global_var(GVAR_KARMA_DEFENDER) == 1) then begin
                         Static_Reaction := Static_Reaction + 5;
                     end
                     else begin
-                        if (global_var(41) == 1) then begin
+                        if (global_var(GVAR_KARMA_WANDERER) == 1) then begin
                             Static_Reaction := Static_Reaction + 0;
                         end
                         else begin
-                            if (global_var(42) == 1) then begin
+                            if (global_var(GVAR_KARMA_BETRAYER) == 1) then begin
                                 Static_Reaction := Static_Reaction + -5;
                             end
                             else begin
-                                if (global_var(43) == 1) then begin
+                                if (global_var(GVAR_KARMA_SWORD_OF_DESPAIR) == 1) then begin
                                     Static_Reaction := Static_Reaction + -10;
                                 end
                                 else begin
-                                    if (global_var(44) == 1) then begin
+                                    if (global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE) == 1) then begin
                                         Static_Reaction := Static_Reaction + -15;
                                     end
                                     else begin
@@ -487,28 +487,28 @@ begin
         debug_msg("Good Person Reaction Bonus == " + Static_Reaction);
     end
     if (Slavery_Tolerant == 0) then begin
-        if (global_var(11) == 1) then begin
+        if (global_var(GVAR_REPUTATION_SLAVER) == 1) then begin
             Static_Reaction := Static_Reaction + (2 * -25);
         end
     end
     else begin
         if (Slavery_Tolerant == 1) then begin
-            if (global_var(11) == 1) then begin
+            if (global_var(GVAR_REPUTATION_SLAVER) == 1) then begin
                 Static_Reaction := Static_Reaction + -25;
             end
         end
         else begin
             if (Slavery_Tolerant == 3) then begin
-                if (global_var(11) == 1) then begin
+                if (global_var(GVAR_REPUTATION_SLAVER) == 1) then begin
                     Static_Reaction := Static_Reaction - -25;
                 end
             end
         end
     end
     debug_msg("Slaver + Aligned Reaction == " + Static_Reaction);
-    Static_Reaction := Static_Reaction + (10 * has_trait(0, dude_obj, 10));
+    Static_Reaction := Static_Reaction + (10 * has_trait(TRAIT_PERK, dude_obj, PERK_presence));
     debug_msg("Presence Reaction == " + Static_Reaction);
-    if (global_var(1) >= 2) then begin
+    if (global_var(GVAR_CHILDKILLER_REPUTATION) >= 2) then begin
         Static_Reaction := Static_Reaction + -30;
     end
     debug_msg("Childkiller Reaction == " + Static_Reaction);
@@ -522,7 +522,7 @@ begin
         end_dialogue;
     end
     else begin
-        if ((get_critter_stat(dude_obj, 4) <= 3) and (get_critter_stat(dude_obj, 0) >= 7)) then begin
+        if ((dude_is_stupid) and (dude_strength >= 7)) then begin
             start_gdialog(352, self_obj, 4, -1, -1);
             gsay_start;
             call Node002();
@@ -543,55 +543,55 @@ end
 procedure destroy_p_proc
 begin
     if (source_obj == dude_obj) then begin
-        set_global_var(4, global_var(4) + 1);
-        if (metarule(51, self_obj) == 2) then begin
+        set_global_var(GVAR_BAD_MONSTER, global_var(GVAR_BAD_MONSTER) + 1);
+        if (critter_kill_type(self_obj) == KILL_TYPE_children_kills) then begin
             if (source_obj == dude_obj) then begin
-                set_global_var(2, 0);
-                set_global_var(1, global_var(1) + 1);
-                set_global_var(0, global_var(0) + -15);
+                set_global_var(GVAR_CHAMPION_REPUTATION, 0);
+                set_global_var(GVAR_CHILDKILLER_REPUTATION, global_var(GVAR_CHILDKILLER_REPUTATION) + 1);
+                set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + -15);
                 debug_msg("Player gains " + -15 + " Karma Points.");
-                set_global_var(37, 0);
-                set_global_var(38, 0);
-                set_global_var(39, 0);
-                set_global_var(40, 0);
-                set_global_var(41, 0);
-                set_global_var(42, 0);
-                set_global_var(43, 0);
-                set_global_var(44, 0);
-                set_global_var(45, 0);
-                if (global_var(0) >= 1000) then begin
-                    set_global_var(37, 1);
+                set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+                set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+                set_global_var(GVAR_KARMA_DEFENDER, 0);
+                set_global_var(GVAR_KARMA_WANDERER, 0);
+                set_global_var(GVAR_KARMA_BETRAYER, 0);
+                set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+                set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+                set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+                    set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
                 end
                 else begin
-                    if (global_var(0) >= 750) then begin
-                        set_global_var(38, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
                     end
                     else begin
-                        if (global_var(0) >= 500) then begin
-                            set_global_var(39, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                            set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                         end
                         else begin
-                            if (global_var(0) >= 250) then begin
-                                set_global_var(40, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                                set_global_var(GVAR_KARMA_DEFENDER, 1);
                             end
                             else begin
-                                if (global_var(0) > -250) then begin
-                                    set_global_var(41, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                                    set_global_var(GVAR_KARMA_WANDERER, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -500) then begin
-                                        set_global_var(42, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                        set_global_var(GVAR_KARMA_BETRAYER, 1);
                                     end
                                     else begin
-                                        if (global_var(0) > -750) then begin
-                                            set_global_var(43, 1);
+                                        if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                            set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                         end
                                         else begin
-                                            if (global_var(0) > -1000) then begin
-                                                set_global_var(44, 1);
+                                            if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                                set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                             end
                                             else begin
-                                                set_global_var(45, 1);
+                                                set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                             end
                                         end
                                     end
@@ -600,62 +600,62 @@ begin
                         end
                     end
                 end
-                set_global_var(57, global_var(57) + -8);
+                set_global_var(GVAR_TOWN_REP_NCR, global_var(GVAR_TOWN_REP_NCR) + -8);
                 debug_msg("Added " + -8 + " to Town Rep");
             end
         end
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(5) > (2 * global_var(4))) or (global_var(3) == 1))) then begin
-            set_global_var(3, 1);
-            set_global_var(2, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
+            set_global_var(GVAR_BERSERKER_REPUTATION, 1);
+            set_global_var(GVAR_CHAMPION_REPUTATION, 0);
         end
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(4) > (3 * global_var(5))) or (global_var(2) == 1)) and (global_var(1) == 0)) then begin
-            set_global_var(2, 1);
-            set_global_var(3, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(4) > (3 * global_var(GVAR_GOOD_MONSTER))) or (global_var(GVAR_CHAMPION_REPUTATION) == 1)) and (global_var(GVAR_CHILDKILLER_REPUTATION) == 0)) then begin
+            set_global_var(GVAR_CHAMPION_REPUTATION, 1);
+            set_global_var(GVAR_BERSERKER_REPUTATION, 0);
         end
-        set_global_var(0, global_var(0) + 5);
+        set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + 5);
         debug_msg("Player gains " + 5 + " Karma Points.");
-        set_global_var(37, 0);
-        set_global_var(38, 0);
-        set_global_var(39, 0);
-        set_global_var(40, 0);
-        set_global_var(41, 0);
-        set_global_var(42, 0);
-        set_global_var(43, 0);
-        set_global_var(44, 0);
-        set_global_var(45, 0);
-        if (global_var(0) >= 1000) then begin
-            set_global_var(37, 1);
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+        set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+        set_global_var(GVAR_KARMA_DEFENDER, 0);
+        set_global_var(GVAR_KARMA_WANDERER, 0);
+        set_global_var(GVAR_KARMA_BETRAYER, 0);
+        set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+        set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+            set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
         end
         else begin
-            if (global_var(0) >= 750) then begin
-                set_global_var(38, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
             end
             else begin
-                if (global_var(0) >= 500) then begin
-                    set_global_var(39, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                 end
                 else begin
-                    if (global_var(0) >= 250) then begin
-                        set_global_var(40, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                        set_global_var(GVAR_KARMA_DEFENDER, 1);
                     end
                     else begin
-                        if (global_var(0) > -250) then begin
-                            set_global_var(41, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                            set_global_var(GVAR_KARMA_WANDERER, 1);
                         end
                         else begin
-                            if (global_var(0) > -500) then begin
-                                set_global_var(42, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                set_global_var(GVAR_KARMA_BETRAYER, 1);
                             end
                             else begin
-                                if (global_var(0) > -750) then begin
-                                    set_global_var(43, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -1000) then begin
-                                        set_global_var(44, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                     end
                                     else begin
-                                        set_global_var(45, 1);
+                                        set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                     end
                                 end
                             end
@@ -664,7 +664,7 @@ begin
                 end
             end
         end
-        set_global_var(57, global_var(57) + 2);
+        set_global_var(GVAR_TOWN_REP_NCR, global_var(GVAR_TOWN_REP_NCR) + 2);
         debug_msg("Added " + 2 + " to Town Rep");
     end
 end
@@ -673,17 +673,17 @@ procedure look_at_p_proc
 begin
     script_overrides;
     if (local_var(4) == 0) then begin
-        display_msg(message_str(352, 100));
+        display_mstr(100);
     end
     else begin
-        display_msg(message_str(352, 101));
+        display_mstr(101);
     end
 end
 
 procedure description_p_proc
 begin
     script_overrides;
-    display_msg(message_str(352, 102));
+    display_mstr(102);
 end
 
 procedure use_skill_on_p_proc
@@ -700,8 +700,8 @@ end
 procedure map_enter_p_proc
 begin
     Only_Once := 0;
-    critter_add_trait(self_obj, 1, 6, 56);
-    critter_add_trait(self_obj, 1, 5, 24);
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 56);
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_TOUGH_CITIZEN);
 end
 
 procedure Node998
@@ -717,60 +717,60 @@ procedure Node001
 begin
     if (local_var(9) == 0) then begin
         set_local_var(9, local_var(9) + 1);
-        gsay_reply(352, 103);
+        Reply(103);
     end
     else begin
-        gsay_reply(352, 104);
+        Reply(104);
     end
-    giq_option(-3, 352, 105, Node004, 50);
-    if (not(metarule(17, 11) != 0)) then begin
-        giq_option(-3, 352, 106, Node005, 50);
+    giq_option(-3, NAME, 105, Node004, NEUTRAL_REACTION);
+    if (not(town_known(11) != MARK_STATE_UNKNOWN)) then begin
+        giq_option(-3, NAME, 106, Node005, NEUTRAL_REACTION);
     end
-    if (global_var(233) == 0) then begin
-        giq_option(-3, 352, 107, Node005, 50);
+    if (global_var(GVAR_VAULT13_FOUND_GECK) == 0) then begin
+        giq_option(-3, NAME, 107, Node005, NEUTRAL_REACTION);
     end
-    giq_option(4, 352, 108, Node006, 50);
-    giq_option(4, 352, 109, Node007, 50);
-    giq_option(4, 352, 110, Node005, 50);
-    giq_option(4, 352, 111, Node008, 50);
-    if (((global_var(438) bwand 131072) != 0) or (global_var(217) == 5) and (local_var(8) == 0)) then begin
-        giq_option(4, 352, 186, Node025, 50);
-        giq_option(-3, 352, 187, Node025, 50);
+    giq_option(4, NAME, 108, Node006, NEUTRAL_REACTION);
+    giq_option(4, NAME, 109, Node007, NEUTRAL_REACTION);
+    giq_option(4, NAME, 110, Node005, NEUTRAL_REACTION);
+    giq_option(4, NAME, 111, Node008, NEUTRAL_REACTION);
+    if (((global_var(GVAR_NCR_GEN_FLAGS) bwand 131072) != 0) or (global_var(GVAR_NCR_MIRA_STATE) == 5) and (local_var(8) == 0)) then begin
+        giq_option(4, NAME, 186, Node025, NEUTRAL_REACTION);
+        giq_option(-3, NAME, 187, Node025, NEUTRAL_REACTION);
     end
 end
 
 procedure Node002
 begin
     if (local_var(4) == 0) then begin
-        gsay_reply(352, 112);
+        Reply(112);
     end
     else begin
-        gsay_reply(352, 113);
+        Reply(113);
     end
-    giq_option(1, 352, 114, NodeCheck, 50);
+    giq_option(1, NAME, 114, NodeCheck, NEUTRAL_REACTION);
 end
 
 procedure Node003
 begin
-    if ((global_var(438) bwand 32768) != 0) then begin
+    if ((global_var(GVAR_NCR_GEN_FLAGS) bwand 32768) != 0) then begin
         if (local_var(7) == 0) then begin
-            gsay_reply(352, 184);
-            giq_option(0, 352, message_str(14, 350), Node999, 50);
+            Reply(184);
+            giq_option(0, NAME, g_mstr(350), Node999, NEUTRAL_REACTION);
         end
         else begin
-            gsay_reply(352, 185);
-            giq_option(0, 352, message_str(14, 352), Node005, 50);
+            Reply(185);
+            giq_option(0, NAME, g_mstr(352), Node005, NEUTRAL_REACTION);
         end
     end
     else begin
         if (local_var(7) == 0) then begin
-            add_obj_to_inven(dude_obj, create_object_sid(125, tile_num(self_obj), 0, -1));
-            gsay_reply(352, 115);
-            giq_option(0, 352, message_str(14, 350), Node999, 50);
+            add_obj_to_inven(dude_obj, create_object(125, tile_num(self_obj), 0));
+            Reply(115);
+            giq_option(0, NAME, g_mstr(350), Node999, NEUTRAL_REACTION);
         end
         else begin
-            gsay_reply(352, 116);
-            giq_option(0, 352, message_str(14, 352), Node005, 50);
+            Reply(116);
+            giq_option(0, NAME, g_mstr(352), Node005, NEUTRAL_REACTION);
         end
     end
     set_local_var(7, local_var(7) + 1);
@@ -778,269 +778,269 @@ end
 
 procedure Node004
 begin
-    gsay_reply(352, 117);
+    Reply(117);
     if (item_caps_total(dude_obj) > 10) then begin
-        giq_option(1, 352, 118, Node9a, 50);
+        giq_option(1, NAME, 118, Node9a, NEUTRAL_REACTION);
     end
-    giq_option(1, 352, 119, Node010, 50);
+    giq_option(1, NAME, 119, Node010, NEUTRAL_REACTION);
 end
 
 procedure Node005
 begin
-    gsay_reply(352, 120);
+    Reply(120);
     if (item_caps_total(dude_obj) > 10) then begin
-        giq_option(1, 352, 121, Node9a, 50);
+        giq_option(1, NAME, 121, Node9a, NEUTRAL_REACTION);
     end
-    giq_option(1, 352, 122, Node010, 50);
+    giq_option(1, NAME, 122, Node010, NEUTRAL_REACTION);
 end
 
 procedure Node006
 begin
-    gsay_reply(352, 123);
+    Reply(123);
     if (item_caps_total(dude_obj) > 10) then begin
-        giq_option(4, 352, 124, Node9a, 50);
+        giq_option(4, NAME, 124, Node9a, NEUTRAL_REACTION);
     end
-    giq_option(4, 352, 125, Node010, 50);
+    giq_option(4, NAME, 125, Node010, NEUTRAL_REACTION);
 end
 
 procedure Node007
 begin
-    gsay_reply(352, 126);
+    Reply(126);
     if (item_caps_total(dude_obj) > 1) then begin
-        giq_option(4, 352, 127, Node9b, 50);
+        giq_option(4, NAME, 127, Node9b, NEUTRAL_REACTION);
     end
-    giq_option(4, 352, 128, Node010, 50);
+    giq_option(4, NAME, 128, Node010, NEUTRAL_REACTION);
 end
 
 procedure Node008
 begin
     variable LVar0 := 0;
     LVar0 := game_time_hour;
-    if (global_var(209) != 1) then begin
+    if (global_var(GVAR_NCR_SCMERK_STATUS) != 1) then begin
         if ((LVar0 >= 1600) and (LVar0 <= 2459) or ((LVar0 >= 0) and (LVar0 <= 759))) then begin
-            gsay_reply(352, 129);
+            Reply(129);
         end
         else begin
-            gsay_reply(352, 183);
+            Reply(183);
         end
     end
     else begin
-        gsay_reply(352, 130);
+        Reply(130);
     end
-    giq_option(4, 352, 131, Node999, 50);
+    giq_option(4, NAME, 131, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node009
 begin
-    gsay_reply(352, 132);
-    if (not(metarule(17, 11) != 0)) then begin
-        giq_option(-3, 352, 133, Node005, 50);
+    Reply(132);
+    if (not(town_known(11) != MARK_STATE_UNKNOWN)) then begin
+        giq_option(-3, NAME, 133, Node005, NEUTRAL_REACTION);
     end
-    if (global_var(233) == 0) then begin
-        giq_option(-3, 352, 134, Node005, 50);
+    if (global_var(GVAR_VAULT13_FOUND_GECK) == 0) then begin
+        giq_option(-3, NAME, 134, Node005, NEUTRAL_REACTION);
     end
-    giq_option(4, 352, 135, Node011, 50);
-    giq_option(-3, 352, 136, Node999, 50);
-    giq_option(4, 352, 137, Node999, 50);
+    giq_option(4, NAME, 135, Node011, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 136, Node999, NEUTRAL_REACTION);
+    giq_option(4, NAME, 137, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node010
 begin
-    gsay_reply(352, 138);
-    giq_option(0, 352, message_str(14, 350), Node999, 50);
+    Reply(138);
+    giq_option(0, NAME, g_mstr(350), Node999, NEUTRAL_REACTION);
 end
 
 procedure Node011
 begin
-    if ((get_critter_stat(dude_obj, 3) < 8) and (local_var(4) == 0)) then begin
-        gsay_reply(352, 139);
+    if ((dude_charisma < 8) and (local_var(4) == 0)) then begin
+        Reply(139);
     end
     else begin
-        if ((get_critter_stat(dude_obj, 3) > 7) and (local_var(4) == 0)) then begin
-            gsay_reply(352, 140);
+        if ((dude_charisma > 7) and (local_var(4) == 0)) then begin
+            Reply(140);
         end
         else begin
-            gsay_reply(352, 141);
+            Reply(141);
         end
     end
-    giq_option(4, 352, 142, Node012, 50);
-    if (not(metarule(17, 11) != 0)) then begin
-        giq_option(4, 352, 143, Node013, 50);
+    giq_option(4, NAME, 142, Node012, NEUTRAL_REACTION);
+    if (not(town_known(11) != MARK_STATE_UNKNOWN)) then begin
+        giq_option(4, NAME, 143, Node013, NEUTRAL_REACTION);
     end
-    if (global_var(233) == 0) then begin
-        giq_option(4, 352, 144, Node014, 50);
+    if (global_var(GVAR_VAULT13_FOUND_GECK) == 0) then begin
+        giq_option(4, NAME, 144, Node014, NEUTRAL_REACTION);
     end
-    if (global_var(209) == 2) then begin
-        giq_option(4, 352, 145, Node015, 50);
+    if (global_var(GVAR_NCR_SCMERK_STATUS) == 2) then begin
+        giq_option(4, NAME, 145, Node015, NEUTRAL_REACTION);
     end
-    if (global_var(217) == 4) then begin
-        giq_option(4, 352, 146, Node016, 50);
+    if (global_var(GVAR_NCR_MIRA_STATE) == 4) then begin
+        giq_option(4, NAME, 146, Node016, NEUTRAL_REACTION);
     end
-    giq_option(4, 352, 147, Node999, 50);
+    giq_option(4, NAME, 147, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node012
 begin
     if (Node12Repeat == 0) then begin
-        gsay_reply(352, 148);
+        Reply(148);
     end
     else begin
         if (Node12Repeat == 1) then begin
-            gsay_reply(352, 149);
+            Reply(149);
         end
         else begin
-            gsay_reply(352, 150);
+            Reply(150);
         end
     end
     if (Node12Repeat < 2) then begin
-        giq_option(1, 352, 151, Node12Rep, 50);
+        giq_option(1, NAME, 151, Node12Rep, NEUTRAL_REACTION);
     end
-    giq_option(4, 352, 152, Node011a, 50);
-    giq_option(4, 352, 153, Node999a, 50);
+    giq_option(4, NAME, 152, Node011a, NEUTRAL_REACTION);
+    giq_option(4, NAME, 153, Node999a, NEUTRAL_REACTION);
 end
 
 procedure Node013
 begin
-    gsay_reply(352, 154);
-    giq_option(-3, 352, 155, Node999, 50);
-    giq_option(4, 352, 156, Node011, 50);
-    giq_option(4, 352, 157, Node999, 50);
+    Reply(154);
+    giq_option(-3, NAME, 155, Node999, NEUTRAL_REACTION);
+    giq_option(4, NAME, 156, Node011, NEUTRAL_REACTION);
+    giq_option(4, NAME, 157, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node014
 begin
-    gsay_reply(352, 158);
-    giq_option(-3, 352, 159, Node017, 50);
-    giq_option(4, 352, 160, Node018, 50);
-    giq_option(4, 352, 161, Node011, 50);
-    giq_option(4, 352, 162, Node999, 50);
+    Reply(158);
+    giq_option(-3, NAME, 159, Node017, NEUTRAL_REACTION);
+    giq_option(4, NAME, 160, Node018, NEUTRAL_REACTION);
+    giq_option(4, NAME, 161, Node011, NEUTRAL_REACTION);
+    giq_option(4, NAME, 162, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node015
 begin
-    gsay_reply(352, 163);
-    giq_option(4, 352, 164, Node011, 50);
-    giq_option(4, 352, 165, Node999, 50);
+    Reply(163);
+    giq_option(4, NAME, 164, Node011, NEUTRAL_REACTION);
+    giq_option(4, NAME, 165, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node016
 begin
-    gsay_reply(352, 166);
-    giq_option(4, 352, 167, Node011, 50);
-    giq_option(4, 352, 168, Node999, 50);
+    Reply(166);
+    giq_option(4, NAME, 167, Node011, NEUTRAL_REACTION);
+    giq_option(4, NAME, 168, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node017
 begin
-    gsay_reply(352, 169);
-    giq_option(0, 352, message_str(14, 350), Node999, 50);
+    Reply(169);
+    giq_option(0, NAME, g_mstr(350), Node999, NEUTRAL_REACTION);
 end
 
 procedure Node018
 begin
-    gsay_reply(352, 170);
-    giq_option(4, 352, 171, Node011, 50);
-    giq_option(4, 352, 172, Node999, 50);
+    Reply(170);
+    giq_option(4, NAME, 171, Node011, NEUTRAL_REACTION);
+    giq_option(4, NAME, 172, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node019
 begin
-    gsay_reply(352, 173);
-    giq_option(1, 352, 174, Node021, 50);
-    giq_option(1, 352, 175, Node005, 50);
+    Reply(173);
+    giq_option(1, NAME, 174, Node021, NEUTRAL_REACTION);
+    giq_option(1, NAME, 175, Node005, NEUTRAL_REACTION);
 end
 
 procedure Node020
 begin
-    gsay_reply(352, 176);
-    giq_option(1, 352, 177, Node021, 50);
-    giq_option(1, 352, 178, Node024, 50);
+    Reply(176);
+    giq_option(1, NAME, 177, Node021, NEUTRAL_REACTION);
+    giq_option(1, NAME, 178, Node024, NEUTRAL_REACTION);
 end
 
 procedure Node021
 begin
-    set_global_var(218, 1);
-    set_global_var(217, 1);
-    gsay_reply(352, 179);
-    giq_option(0, 352, message_str(14, 350), Node999, 50);
+    set_global_var(GVAR_NCR_ROPE_KNOWN, 1);
+    set_global_var(GVAR_NCR_MIRA_STATE, 1);
+    Reply(179);
+    giq_option(0, NAME, g_mstr(350), Node999, NEUTRAL_REACTION);
 end
 
 procedure Node022
 begin
-    gsay_reply(352, 180);
-    giq_option(0, 352, message_str(14, 350), Node999, 50);
+    Reply(180);
+    giq_option(0, NAME, g_mstr(350), Node999, NEUTRAL_REACTION);
 end
 
 procedure Node023
 begin
-    set_global_var(217, 3);
-    add_obj_to_inven(dude_obj, create_object_sid(296, tile_num(self_obj), 0, -1));
-    gsay_reply(352, 181);
-    giq_option(0, 352, message_str(14, 350), Node999, 50);
+    set_global_var(GVAR_NCR_MIRA_STATE, 3);
+    add_obj_to_inven(dude_obj, create_object(296, tile_num(self_obj), 0));
+    Reply(181);
+    giq_option(0, NAME, g_mstr(350), Node999, NEUTRAL_REACTION);
 end
 
 procedure Node024
 begin
-    gsay_reply(352, 182);
-    giq_option(0, 352, message_str(14, 350), Node999, 50);
+    Reply(182);
+    giq_option(0, NAME, g_mstr(350), Node999, NEUTRAL_REACTION);
 end
 
 procedure Node025
 begin
     set_local_var(8, local_var(8) + 1);
-    gsay_reply(352, 188);
-    if ((global_var(438) bwand 131072) != 0) then begin
-        giq_option(-3, 352, 189, Node026, 50);
-        giq_option(4, 352, 191, Node026, 50);
+    Reply(188);
+    if ((global_var(GVAR_NCR_GEN_FLAGS) bwand 131072) != 0) then begin
+        giq_option(-3, NAME, 189, Node026, NEUTRAL_REACTION);
+        giq_option(4, NAME, 191, Node026, NEUTRAL_REACTION);
     end
     else begin
-        if (global_var(217) == 5) then begin
-            giq_option(-3, 352, 190, Node028, 50);
-            giq_option(4, 352, 192, Node027, 50);
+        if (global_var(GVAR_NCR_MIRA_STATE) == 5) then begin
+            giq_option(-3, NAME, 190, Node028, NEUTRAL_REACTION);
+            giq_option(4, NAME, 192, Node027, NEUTRAL_REACTION);
         end
     end
 end
 
 procedure Node026
 begin
-    gsay_reply(352, 193);
-    giq_option(-3, 352, 194, Node009, 50);
-    giq_option(-3, 352, 195, Node999, 50);
-    giq_option(4, 352, 196, Node011, 50);
-    giq_option(4, 352, 197, Node999, 50);
+    Reply(193);
+    giq_option(-3, NAME, 194, Node009, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 195, Node999, NEUTRAL_REACTION);
+    giq_option(4, NAME, 196, Node011, NEUTRAL_REACTION);
+    giq_option(4, NAME, 197, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node027
 begin
-    add_obj_to_inven(dude_obj, create_object_sid(296, tile_num(self_obj), 0, -1));
-    gsay_reply(352, 198);
-    giq_option(4, 352, 199, Node011, 50);
-    giq_option(4, 352, 200, Node999, 50);
+    add_obj_to_inven(dude_obj, create_object(296, tile_num(self_obj), 0));
+    Reply(198);
+    giq_option(4, NAME, 199, Node011, NEUTRAL_REACTION);
+    giq_option(4, NAME, 200, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node028
 begin
-    add_obj_to_inven(dude_obj, create_object_sid(296, tile_num(self_obj), 0, -1));
-    gsay_reply(352, 201);
-    giq_option(-3, 352, 202, Node009, 50);
-    giq_option(-3, 352, 203, Node999, 50);
+    add_obj_to_inven(dude_obj, create_object(296, tile_num(self_obj), 0));
+    Reply(201);
+    giq_option(-3, NAME, 202, Node009, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 203, Node999, NEUTRAL_REACTION);
 end
 
 procedure NodeCheck
 begin
-    if (global_var(217) == 1) then begin
+    if (global_var(GVAR_NCR_MIRA_STATE) == 1) then begin
         call Node022();
     end
     else begin
-        if (global_var(217) == 2) then begin
+        if (global_var(GVAR_NCR_MIRA_STATE) == 2) then begin
             call Node023();
         end
         else begin
-            if (global_var(217) == 3) then begin
+            if (global_var(GVAR_NCR_MIRA_STATE) == 3) then begin
                 call Node005();
             end
             else begin
-                if (global_var(217) == 5) then begin
+                if (global_var(GVAR_NCR_MIRA_STATE) == 5) then begin
                     call Node001();
                 end
                 else begin
@@ -1055,7 +1055,7 @@ procedure Node9a
 begin
     item_caps_adjust(dude_obj, -10);
     item_caps_adjust(self_obj, 10);
-    add_obj_to_inven(dude_obj, create_object_sid(125, tile_num(self_obj), 0, -1));
+    add_obj_to_inven(dude_obj, create_object(125, tile_num(self_obj), 0));
     call Node009();
 end
 
@@ -1063,7 +1063,7 @@ procedure Node9b
 begin
     item_caps_adjust(dude_obj, -1);
     item_caps_adjust(self_obj, 1);
-    add_obj_to_inven(dude_obj, create_object_sid(124, tile_num(self_obj), 0, -1));
+    add_obj_to_inven(dude_obj, create_object(124, tile_num(self_obj), 0));
     call Node009();
 end
 
@@ -1084,4 +1084,3 @@ begin
     Node12Repeat := 0;
     call Node999();
 end
-
