diff --git a/Games/Fallout2.Steam/Original/SFDOCK.ssl b/Games/Fallout2.Steam/ReDefine/SFDOCK.ssl
index 73c8929..3e7cb56 100644
--- a/Games/Fallout2.Steam/Original/SFDOCK.ssl
+++ b/Games/Fallout2.Steam/ReDefine/SFDOCK.ssl
@@ -213,8 +213,8 @@ begin
             end
         end
     end
-    if (global_var(499) == 2) then begin
-        set_global_var(499, 3);
+    if (global_var(GVAR_ENCLAVE_FRANK_DEAD) == 2) then begin
+        set_global_var(GVAR_ENCLAVE_FRANK_DEAD, 3);
     end
 end
 
@@ -297,4 +297,3 @@ begin
         end
     end
 end
-
