diff --git a/Games/Fallout2.Steam/Original/SFTANKER.ssl b/Games/Fallout2.Steam/ReDefine/SFTANKER.ssl
index 4876b9d..67680e6 100644
--- a/Games/Fallout2.Steam/Original/SFTANKER.ssl
+++ b/Games/Fallout2.Steam/ReDefine/SFTANKER.ssl
@@ -152,7 +152,7 @@ end
 
 procedure map_enter_p_proc
 begin
-    if (metarule(14, 0)) then begin
+    if (map_first_run) then begin
     end
     set_light_level(60);
 end
@@ -165,4 +165,3 @@ end
 procedure Initial_Inven
 begin
 end
-
