diff --git a/Games/Fallout2.Steam/Original/SHACK.ssl b/Games/Fallout2.Steam/ReDefine/SHACK.ssl
index 87ab2bf..6d7b1bd 100644
--- a/Games/Fallout2.Steam/Original/SHACK.ssl
+++ b/Games/Fallout2.Steam/ReDefine/SHACK.ssl
@@ -3,13 +3,12 @@ procedure start;
 
 procedure start
 begin
-    if (script_action == 2) then begin
+    if (script_action == spatial_proc) then begin
         if (source_obj == dude_obj) then begin
             if (local_var(0) == 0) then begin
                 set_local_var(0, 1);
-                display_msg(message_str(196, 100));
+                display_msg(message_str(SCRIPT_MODSHIT, 100));
             end
         end
     end
 end
-
