diff --git a/Games/Fallout2.Steam/Original/SIFAKHLO.ssl b/Games/Fallout2.Steam/ReDefine/SIFAKHLO.ssl
index 694653b..cb19565 100644
--- a/Games/Fallout2.Steam/Original/SIFAKHLO.ssl
+++ b/Games/Fallout2.Steam/ReDefine/SIFAKHLO.ssl
@@ -137,18 +137,17 @@ end
 
 procedure use_p_proc
 begin
-    if (global_var(214) != 1) then begin
-        set_global_var(214, 1);
-        display_msg(message_str(334, 102));
+    if (global_var(GVAR_NCR_FAKE_VAULT13_HOLODISK) != 1) then begin
+        set_global_var(GVAR_NCR_FAKE_VAULT13_HOLODISK, 1);
+        display_mstr(102);
     end
     else begin
-        display_msg(message_str(334, 101));
+        display_mstr(101);
     end
 end
 
 procedure description_p_proc
 begin
     script_overrides;
-    display_msg(message_str(334, 100));
+    display_mstr(100);
 end
-
