diff --git a/Games/Fallout2.Steam/Original/SIGCAOBJ.ssl b/Games/Fallout2.Steam/ReDefine/SIGCAOBJ.ssl
index e1ab109..5f1490c 100644
--- a/Games/Fallout2.Steam/Original/SIGCAOBJ.ssl
+++ b/Games/Fallout2.Steam/ReDefine/SIGCAOBJ.ssl
@@ -150,9 +150,9 @@ begin
     debug_msg("SIGCAOBJ: i_guard_captain_obj: " + i_guard_captain_obj);
     if (i_guard_captain_obj > 0) then begin
         if (obj_can_see_obj(i_guard_captain_obj, source_obj)) then begin
-            if (not((global_var(438) bwand 8192) != 0)) then begin
+            if (not((global_var(GVAR_NCR_GEN_FLAGS) bwand 8192) != 0)) then begin
                 script_overrides;
-                set_global_var(438, global_var(438) bwor 8192);
+                set_global_var(GVAR_NCR_GEN_FLAGS, global_var(GVAR_NCR_GEN_FLAGS) bwor 8192);
             end
         end
     end
@@ -189,4 +189,3 @@ end
 procedure timed_event_p_proc
 begin
 end
-
