diff --git a/Games/Fallout2.Steam/Original/SIPAINT.ssl b/Games/Fallout2.Steam/ReDefine/SIPAINT.ssl
index aced680..81c2755 100644
--- a/Games/Fallout2.Steam/Original/SIPAINT.ssl
+++ b/Games/Fallout2.Steam/ReDefine/SIPAINT.ssl
@@ -153,18 +153,18 @@ end
 procedure description_p_proc
 begin
     script_overrides;
-    if (not(metarule(17, 11) != 0)) then begin
-        if (metarule(17, 11) == 0) then begin
+    if (not(town_known(11) != MARK_STATE_UNKNOWN)) then begin
+        if (town_known(11) == MARK_STATE_UNKNOWN) then begin
             debug_msg("  mark_on_map(" + 11 + ")");
-            mark_area_known(0, 11, 1);
+            mark_area_known(MARK_TYPE_TOWN, 11, MARK_STATE_KNOWN);
         end
-        if (global_var(494) < 1) then begin
-            set_global_var(494, 1);
+        if (global_var(GVAR_FALLOUT_2) < 1) then begin
+            set_global_var(GVAR_FALLOUT_2, 1);
         end
-        display_msg(message_str(954, 100));
+        display_mstr(100);
     end
     else begin
-        display_msg(message_str(954, 100));
+        display_mstr(100);
     end
 end
 
@@ -191,4 +191,3 @@ end
 procedure timed_event_p_proc
 begin
 end
-
