diff --git a/Games/Fallout2.Steam/Original/SITERM.ssl b/Games/Fallout2.Steam/ReDefine/SITERM.ssl
index 6796188..2fa8c7d 100644
--- a/Games/Fallout2.Steam/Original/SITERM.ssl
+++ b/Games/Fallout2.Steam/ReDefine/SITERM.ssl
@@ -168,12 +168,12 @@ end
 
 procedure look_at_p_proc
 begin
-    display_msg(message_str(376, 100));
+    display_mstr(100);
 end
 
 procedure description_p_proc
 begin
-    display_msg(message_str(376, 101));
+    display_mstr(101);
 end
 
 procedure use_skill_on_p_proc
@@ -185,12 +185,12 @@ begin
                 call use_success();
             end
             else begin
-                display_msg(message_str(376, 201));
+                display_mstr(201);
                 call use_fail();
             end
         end
         else begin
-            display_msg(message_str(376, 205));
+            display_mstr(205);
             call use_fail();
         end
     end
@@ -198,7 +198,7 @@ end
 
 procedure use_obj_on_p_proc
 begin
-    display_msg(message_str(376, 204));
+    display_mstr(204);
 end
 
 procedure damage_p_proc
@@ -235,7 +235,7 @@ end
 procedure open_slave_doors
 begin
     debug_msg("SITERM: doors opening");
-    set_global_var(243, global_var(243) bwor 32);
+    set_global_var(GVAR_8_BALL_TOILET_SECRET, global_var(GVAR_8_BALL_TOILET_SECRET) bwor 32);
     set_map_var(0, 1);
     set_map_var(3, 1);
     if (tile_num(self_obj) == 17715) then begin
@@ -368,4 +368,3 @@ end
 procedure timed_event_p_proc
 begin
 end
-
