diff --git a/Games/Fallout2.Steam/Original/SIWESHOL.ssl b/Games/Fallout2.Steam/ReDefine/SIWESHOL.ssl
index 755e3d3..8b256d4 100644
--- a/Games/Fallout2.Steam/Original/SIWESHOL.ssl
+++ b/Games/Fallout2.Steam/ReDefine/SIWESHOL.ssl
@@ -137,18 +137,17 @@ end
 
 procedure use_p_proc
 begin
-    if (global_var(597) != 1) then begin
-        set_global_var(597, 1);
-        display_msg(message_str(1182, 102));
+    if (global_var(GVAR_NCR_WESTIN_HOLO_DOWNLOAD) != 1) then begin
+        set_global_var(GVAR_NCR_WESTIN_HOLO_DOWNLOAD, 1);
+        display_mstr(102);
     end
     else begin
-        display_msg(message_str(1182, 101));
+        display_mstr(101);
     end
 end
 
 procedure description_p_proc
 begin
     script_overrides;
-    display_msg(message_str(1182, 100));
+    display_mstr(100);
 end
-
