diff --git a/Games/Fallout2.Steam/Original/SSEMIENT.ssl b/Games/Fallout2.Steam/ReDefine/SSEMIENT.ssl
index 02552b3..64fbfe1 100644
--- a/Games/Fallout2.Steam/Original/SSEMIENT.ssl
+++ b/Games/Fallout2.Steam/ReDefine/SSEMIENT.ssl
@@ -149,7 +149,7 @@ end
 procedure use_p_proc
 begin
     script_overrides;
-    display_msg(message_str(229, 100));
+    display_msg(message_str(SCRIPT_WSEMIT1A, 100));
 end
 
 procedure use_skill_on_p_proc
@@ -168,34 +168,34 @@ begin
             if (local_var(0) == 1) then begin
                 set_map_var(6, 2);
                 if (source_obj == dude_obj) then begin
-                    display_msg(message_str(229, 112));
+                    display_msg(message_str(SCRIPT_WSEMIT1A, 112));
                 end
                 else begin
-                    display_msg(obj_name(source_obj) + message_str(229, 113));
+                    display_msg(obj_name(source_obj) + message_str(SCRIPT_WSEMIT1A, 113));
                 end
             end
             else begin
                 set_map_var(6, 1);
                 if (source_obj == dude_obj) then begin
-                    display_msg(message_str(229, 114));
+                    display_msg(message_str(SCRIPT_WSEMIT1A, 114));
                 end
                 else begin
-                    display_msg(obj_name(source_obj) + message_str(229, 115));
+                    display_msg(obj_name(source_obj) + message_str(SCRIPT_WSEMIT1A, 115));
                 end
             end
             if (local_var(1) == 0) then begin
                 set_local_var(1, 1);
                 give_exp_points(50);
-                display_msg(message_str(14, 100) + 50 + message_str(14, 101));
+                display_msg(g_mstr(100) + 50 + g_mstr(101));
             end
         end
         else begin
             if (source_obj == dude_obj) then begin
                 set_local_var(2, local_var(2) + 1);
-                display_msg(message_str(229, 109));
+                display_msg(message_str(SCRIPT_WSEMIT1A, 109));
             end
             else begin
-                display_msg(message_str(229, 110));
+                display_msg(message_str(SCRIPT_WSEMIT1A, 110));
             end
             if (local_var(2) == 3) then begin
                 if (-1 != -1) then begin
@@ -226,34 +226,34 @@ begin
             if (local_var(0) == 1) then begin
                 set_map_var(6, 2);
                 if (source_obj == dude_obj) then begin
-                    display_msg(message_str(229, 112));
+                    display_msg(message_str(SCRIPT_WSEMIT1A, 112));
                 end
                 else begin
-                    display_msg(obj_name(source_obj) + message_str(229, 113));
+                    display_msg(obj_name(source_obj) + message_str(SCRIPT_WSEMIT1A, 113));
                 end
             end
             else begin
                 set_map_var(6, 1);
                 if (source_obj == dude_obj) then begin
-                    display_msg(message_str(229, 114));
+                    display_msg(message_str(SCRIPT_WSEMIT1A, 114));
                 end
                 else begin
-                    display_msg(obj_name(source_obj) + message_str(229, 115));
+                    display_msg(obj_name(source_obj) + message_str(SCRIPT_WSEMIT1A, 115));
                 end
             end
             if (local_var(1) == 0) then begin
                 set_local_var(1, 1);
                 give_exp_points(50);
-                display_msg(message_str(14, 100) + 50 + message_str(14, 101));
+                display_msg(g_mstr(100) + 50 + g_mstr(101));
             end
         end
         else begin
             if (source_obj == dude_obj) then begin
                 set_local_var(2, local_var(2) + 1);
-                display_msg(message_str(229, 109));
+                display_msg(message_str(SCRIPT_WSEMIT1A, 109));
             end
             else begin
-                display_msg(message_str(229, 110));
+                display_msg(message_str(SCRIPT_WSEMIT1A, 110));
             end
             if (local_var(2) == 3) then begin
                 if (-1 != -1) then begin
@@ -318,7 +318,7 @@ begin
                     if (fieldPtr) then begin
                         destroy_object(fieldPtr);
                     end
-                    display_msg(message_str(229, 116));
+                    display_msg(message_str(SCRIPT_WSEMIT1A, 116));
                     set_local_var(0, 4);
                     i_entrace_emitter_obj := -1;
                     destroy_object(self_obj);
@@ -336,7 +336,7 @@ begin
                     if (fixed_param == 5) then begin
                         if (not(local_var(0) == 4) and (local_var(0) == 8)) then begin
                             if (game_time < local_var(3)) then begin
-                                if (has_trait(1, fieldPtr, 666)) then begin
+                                if (obj_is_visible_flag(fieldPtr)) then begin
                                     set_obj_visibility(fieldPtr, 1);
                                 end
                                 else begin
@@ -351,10 +351,10 @@ begin
                     end
                     else begin
                         if (fixed_param == 6) then begin
-                            if (not((global_var(234) == 0) and not((game_time_hour >= 600) and (game_time_hour < 2000)))) then begin
+                            if (not((global_var(GVAR_NCR_POWER_ON) == 0) and not((game_time_hour >= 600) and (game_time_hour < 2000)))) then begin
                                 set_local_var(4, 0);
                                 if (not(local_var(0) == 4)) then begin
-                                    if (has_trait(1, fieldPtr, 666)) then begin
+                                    if (obj_is_visible_flag(fieldPtr)) then begin
                                         set_obj_visibility(fieldPtr, 1);
                                     end
                                 end
@@ -362,7 +362,7 @@ begin
                             else begin
                                 set_local_var(4, 1);
                                 if (not(local_var(0) == 4) and (local_var(0) == 1)) then begin
-                                    if (not(has_trait(1, fieldPtr, 666))) then begin
+                                    if (not(obj_is_visible_flag(fieldPtr))) then begin
                                         set_obj_visibility(fieldPtr, 0);
                                     end
                                 end
@@ -427,10 +427,10 @@ begin
     if (not(local_var(0) == 4) and not(LVar0)) then begin
         fieldPtr := tile_contains_pid_obj(30334, 0, 33555980);
         debug_msg("SSEMIENT: fieldPtr: " + fieldPtr);
-        if (has_trait(1, fieldPtr, 666)) then begin
+        if (obj_is_visible_flag(fieldPtr)) then begin
             set_local_var(4, 1);
         end
-        if (not(metarule(22, 0))) then begin
+        if (not(is_loading_game)) then begin
             set_map_var(6, 1);
         end
         add_timer_event(self_obj, 0, 6);
@@ -450,4 +450,3 @@ procedure destroy_p_proc
 begin
     i_entrace_emitter_obj := -1;
 end
-
