diff --git a/Games/Fallout2.Steam/Original/TEST0.ssl b/Games/Fallout2.Steam/ReDefine/TEST0.ssl
index eff6093..2d7d506 100644
--- a/Games/Fallout2.Steam/Original/TEST0.ssl
+++ b/Games/Fallout2.Steam/ReDefine/TEST0.ssl
@@ -29,8 +29,7 @@ procedure talk_p_proc
 begin
     start_gdialog(2, self_obj, 4, -1, -1);
     gsay_start;
-    gsay_message(2, 100, 50);
+    gsay_message(NAME, 100, NEUTRAL_REACTION);
     gsay_end;
     end_dialogue;
 end
-
