diff --git a/Games/Fallout2.Steam/Original/THSUPMUT.ssl b/Games/Fallout2.Steam/ReDefine/THSUPMUT.ssl
index 7b12dac..4dbf40c 100644
--- a/Games/Fallout2.Steam/Original/THSUPMUT.ssl
+++ b/Games/Fallout2.Steam/ReDefine/THSUPMUT.ssl
@@ -24,23 +24,23 @@ procedure start
 begin
     if (ONLY_ONCE) then begin
         ONLY_ONCE := 0;
-        critter_add_trait(self_obj, 1, 6, 34);
-        critter_add_trait(self_obj, 1, 5, 49);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 34);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_LONER_CITIZEN);
     end
     else begin
-        if (script_action == 12) then begin
+        if (script_action == critter_proc) then begin
             call critter_p_proc();
         end
         else begin
-            if (script_action == 21) then begin
+            if (script_action == look_at_proc) then begin
                 call look_at_p_proc();
             end
             else begin
-                if (script_action == 4) then begin
+                if (script_action == pickup_proc) then begin
                     call pickup_p_proc();
                 end
                 else begin
-                    if (script_action == 11) then begin
+                    if (script_action == talk_proc) then begin
                         call talk_p_proc();
                     end
                 end
@@ -53,24 +53,24 @@ procedure critter_p_proc
 begin
     if (Hostile) then begin
         Hostile := 0;
-        attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+        attack(dude_obj);
     end
 end
 
 procedure destroy_p_proc
 begin
     if (source_obj == dude_obj) then begin
-        if (((global_var(160) + global_var(159)) >= 25) and ((global_var(159) > (2 * global_var(160))) or (global_var(156) == 1))) then begin
-            set_global_var(156, 1);
-            set_global_var(157, 0);
+        if (((global_var(GVAR_GECKO_SKEETER_PART) + global_var(GVAR_GECKO_REQ_FORM)) >= 25) and ((global_var(GVAR_GECKO_REQ_FORM) > (2 * global_var(GVAR_GECKO_SKEETER_PART))) or (global_var(GVAR_SIERRA_LOCKOUT) == 1))) then begin
+            set_global_var(GVAR_SIERRA_LOCKOUT, 1);
+            set_global_var(GVAR_SIERRA_PASSWORD, 0);
         end
-        if (((global_var(160) + global_var(159)) >= 25) and ((global_var(160) > (3 * global_var(159))) or (global_var(157) == 1))) then begin
-            set_global_var(157, 1);
-            set_global_var(156, 0);
+        if (((global_var(GVAR_GECKO_SKEETER_PART) + global_var(GVAR_GECKO_REQ_FORM)) >= 25) and ((global_var(160) > (3 * global_var(GVAR_GECKO_REQ_FORM))) or (global_var(GVAR_SIERRA_PASSWORD) == 1))) then begin
+            set_global_var(GVAR_SIERRA_PASSWORD, 1);
+            set_global_var(GVAR_SIERRA_LOCKOUT, 0);
         end
-        set_global_var(160, global_var(160) + 1);
-        if ((global_var(160) % 6) == 0) then begin
-            set_global_var(155, global_var(155) + 1);
+        set_global_var(GVAR_GECKO_SKEETER_PART, global_var(GVAR_GECKO_SKEETER_PART) + 1);
+        if ((global_var(GVAR_GECKO_SKEETER_PART) % 6) == 0) then begin
+            set_global_var(GVAR_BRAIN_BOT_BRAIN, global_var(GVAR_BRAIN_BOT_BRAIN) + 1);
         end
     end
 end
@@ -81,7 +81,7 @@ begin
     if (rndx == 0) then begin
         rndx := random(100, 109);
     end
-    display_msg(message_str(265, rndx));
+    display_msg(message_str(SCRIPT_KSBBOARD, rndx));
 end
 
 procedure pickup_p_proc
@@ -107,56 +107,56 @@ end
 procedure goto0
 begin
     set_local_var(0, 1);
-    gsay_reply(265, 110);
-    giq_option(-3, 265, 111, goto1, 50);
-    giq_option(4, 265, 112, combat, 50);
-    giq_option(4, 265, 113, goto2, 50);
-    giq_option(6, 265, 114, goto3, 50);
+    gsay_reply(SCRIPT_KSBBOARD, 110);
+    giq_option(-3, SCRIPT_KSBBOARD, 111, goto1, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_KSBBOARD, 112, combat, NEUTRAL_REACTION);
+    giq_option(4, SCRIPT_KSBBOARD, 113, goto2, NEUTRAL_REACTION);
+    giq_option(6, SCRIPT_KSBBOARD, 114, goto3, NEUTRAL_REACTION);
 end
 
 procedure goto1
 begin
-    gsay_message(265, 115, 50);
+    gsay_message(SCRIPT_KSBBOARD, 115, NEUTRAL_REACTION);
     call combat();
 end
 
 procedure goto2
 begin
-    gsay_message(265, 116, 50);
+    gsay_message(SCRIPT_KSBBOARD, 116, NEUTRAL_REACTION);
     call combat();
 end
 
 procedure goto3
 begin
-    gsay_reply(265, 117);
-    giq_option(6, 265, 118, goto4, 50);
-    giq_option(6, 265, 119, goto6, 50);
+    gsay_reply(SCRIPT_KSBBOARD, 117);
+    giq_option(6, SCRIPT_KSBBOARD, 118, goto4, NEUTRAL_REACTION);
+    giq_option(6, SCRIPT_KSBBOARD, 119, goto6, NEUTRAL_REACTION);
 end
 
 procedure goto4
 begin
-    gsay_reply(265, 120);
-    giq_option(6, 265, 121, combat, 50);
-    giq_option(6, 265, 122, combat, 50);
-    giq_option(8, 265, 123, goto5, 50);
+    gsay_reply(SCRIPT_KSBBOARD, 120);
+    giq_option(6, SCRIPT_KSBBOARD, 121, combat, NEUTRAL_REACTION);
+    giq_option(6, SCRIPT_KSBBOARD, 122, combat, NEUTRAL_REACTION);
+    giq_option(8, SCRIPT_KSBBOARD, 123, goto5, NEUTRAL_REACTION);
 end
 
 procedure goto5
 begin
-    gsay_reply(265, 124);
-    giq_option(8, 265, 125, combat, 50);
-    giq_option(8, 265, 126, combat, 50);
+    gsay_reply(SCRIPT_KSBBOARD, 124);
+    giq_option(8, SCRIPT_KSBBOARD, 125, combat, NEUTRAL_REACTION);
+    giq_option(8, SCRIPT_KSBBOARD, 126, combat, NEUTRAL_REACTION);
 end
 
 procedure goto6
 begin
-    gsay_message(265, 127, 50);
+    gsay_message(SCRIPT_KSBBOARD, 127, NEUTRAL_REACTION);
     call combat();
 end
 
 procedure goto7
 begin
-    gsay_message(265, 128, 50);
+    gsay_message(SCRIPT_KSBBOARD, 128, NEUTRAL_REACTION);
     call combat();
 end
 
@@ -168,4 +168,3 @@ end
 procedure gotoend
 begin
 end
-
