diff --git a/Games/Fallout2.Steam/Original/V15ENT.ssl b/Games/Fallout2.Steam/ReDefine/V15ENT.ssl
index 41a3ea3..382cd59 100644
--- a/Games/Fallout2.Steam/Original/V15ENT.ssl
+++ b/Games/Fallout2.Steam/ReDefine/V15ENT.ssl
@@ -218,17 +218,17 @@ begin
             end
         end
     end
-    debug_msg("V15SENT: Darion_deal " + global_var(475) + " diaron_dead " + global_var(172));
-    if ((global_var(475) == 2) and global_var(172)) then begin
+    debug_msg("V15SENT: Darion_deal " + global_var(GVAR_V15_NCR_DEAL) + " diaron_dead " + global_var(GVAR_NCR_V15_DARION_DEAD));
+    if ((global_var(GVAR_V15_NCR_DEAL) == 2) and global_var(GVAR_NCR_V15_DARION_DEAD)) then begin
         override_map_start(29525 % 200, 29525 / 200, 1, 0);
         set_map_start(29525 % 200, 29525 / 200, 1, 0);
     end
     else begin
         override_map_start(29525 % 200, 29525 / 200, 0, 0);
     end
-    debug_msg("V15ENT: map_var(MVAR_Made_Car): " + map_var(0) + " global_var(GVAR_PLAYER_GOT_CAR): " + global_var(18) + "V15AREA: " + 9);
-    if ((global_var(475) == 2) and global_var(172)) then begin
-        if (tile_contains_pid_obj(global_var(633), 0, 33555441) != 0) then begin
+    debug_msg("V15ENT: map_var(MVAR_Made_Car): " + map_var(0) + " global_var(GVAR_PLAYER_GOT_CAR): " + global_var(GVAR_PLAYER_GOT_CAR) + "V15AREA: " + 9);
+    if ((global_var(GVAR_V15_NCR_DEAL) == 2) and global_var(GVAR_NCR_V15_DARION_DEAD)) then begin
+        if (tile_contains_pid_obj(global_var(GVAR_CAR_PLACED_TILE), 0, 33555441) != 0) then begin
             debug_msg("Dest_Car_Loc(" + 26932 + "," + 0 + ")");
             Scenery_Creation_Ptr := tile_contains_pid_obj(26932, 0, 33555441);
             if (Scenery_Creation_Ptr != 0) then begin
@@ -363,103 +363,103 @@ begin
                     destroy_object(Scenery_Creation_Ptr);
                 end
             end
-            set_global_var(633, 0);
+            set_global_var(GVAR_CAR_PLACED_TILE, 0);
             debug_msg("Place_Car(" + 29718 + "," + 1 + ")");
-            debug_msg("   GVAR_CAR_PLACED_TILE == " + global_var(633));
+            debug_msg("   GVAR_CAR_PLACED_TILE == " + global_var(GVAR_CAR_PLACED_TILE));
             debug_msg("   Car_At_Loc == " + (tile_contains_pid_obj(29718, 1, 33555441) != 0));
-            if ((global_var(633) <= 0) or (29718 == global_var(633))) then begin
-                set_global_var(633, 29718);
+            if ((global_var(GVAR_CAR_PLACED_TILE) <= 0) or (29718 == global_var(GVAR_CAR_PLACED_TILE))) then begin
+                set_global_var(GVAR_CAR_PLACED_TILE, 29718);
                 if (not(tile_contains_pid_obj(29718, 1, 33555441) != 0)) then begin
-                    Scenery_Creation := create_object_sid(33555441, 29718, 1, 304);
+                    Scenery_Creation := create_object_sid(33555441, 29718, 1, SCRIPT_ZSDRVCAR);
                     Scenery_Creation_Hex := 29718;
                     Scenery_Creation_Count := 0;
                     while (Scenery_Creation_Count < 2) do begin
                         Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 1, 1);
-                        Scenery_Creation := create_object_sid(33554499, Scenery_Creation_Hex, 1, -1);
+                        Scenery_Creation := create_object(33554499, Scenery_Creation_Hex, 1);
                         Scenery_Creation_Count := Scenery_Creation_Count + 1;
                     end
                     Scenery_Creation_Count := 0;
                     while (Scenery_Creation_Count < 2) do begin
                         Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 2, 1);
-                        Scenery_Creation := create_object_sid(33554499, Scenery_Creation_Hex, 1, -1);
+                        Scenery_Creation := create_object(33554499, Scenery_Creation_Hex, 1);
                         Scenery_Creation_Count := Scenery_Creation_Count + 1;
                     end
                     Scenery_Creation_Count := 0;
                     while (Scenery_Creation_Count < 2) do begin
                         Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 3, 1);
-                        Scenery_Creation := create_object_sid(33554499, Scenery_Creation_Hex, 1, -1);
+                        Scenery_Creation := create_object(33554499, Scenery_Creation_Hex, 1);
                         Scenery_Creation_Count := Scenery_Creation_Count + 1;
                     end
                     Scenery_Creation_Count := 0;
                     while (Scenery_Creation_Count < 2) do begin
                         Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 4, 1);
-                        Scenery_Creation := create_object_sid(33554499, Scenery_Creation_Hex, 1, -1);
+                        Scenery_Creation := create_object(33554499, Scenery_Creation_Hex, 1);
                         Scenery_Creation_Count := Scenery_Creation_Count + 1;
                     end
                     Scenery_Creation_Count := 0;
                     while (Scenery_Creation_Count < 1) do begin
                         Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 5, 1);
-                        Scenery_Creation := create_object_sid(33554499, Scenery_Creation_Hex, 1, -1);
+                        Scenery_Creation := create_object(33554499, Scenery_Creation_Hex, 1);
                         Scenery_Creation_Count := Scenery_Creation_Count + 1;
                     end
                     Scenery_Creation_Count := 0;
                     while (Scenery_Creation_Count < 1) do begin
                         Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 4, 1);
-                        Scenery_Creation := create_object_sid(33554499, Scenery_Creation_Hex, 1, -1);
+                        Scenery_Creation := create_object(33554499, Scenery_Creation_Hex, 1);
                         Scenery_Creation_Count := Scenery_Creation_Count + 1;
                     end
                     Scenery_Creation_Count := 0;
                     while (Scenery_Creation_Count < 1) do begin
                         Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 5, 1);
-                        Scenery_Creation := create_object_sid(33554499, Scenery_Creation_Hex, 1, -1);
+                        Scenery_Creation := create_object(33554499, Scenery_Creation_Hex, 1);
                         Scenery_Creation_Count := Scenery_Creation_Count + 1;
                     end
                     Scenery_Creation_Count := 0;
                     while (Scenery_Creation_Count < 1) do begin
                         Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 4, 1);
-                        Scenery_Creation := create_object_sid(33554499, Scenery_Creation_Hex, 1, -1);
+                        Scenery_Creation := create_object(33554499, Scenery_Creation_Hex, 1);
                         Scenery_Creation_Count := Scenery_Creation_Count + 1;
                     end
                     Scenery_Creation_Count := 0;
                     while (Scenery_Creation_Count < 1) do begin
                         Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 5, 1);
-                        Scenery_Creation := create_object_sid(33554499, Scenery_Creation_Hex, 1, -1);
+                        Scenery_Creation := create_object(33554499, Scenery_Creation_Hex, 1);
                         Scenery_Creation_Count := Scenery_Creation_Count + 1;
                     end
                     Scenery_Creation_Count := 0;
                     while (Scenery_Creation_Count < 1) do begin
                         Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 4, 1);
-                        Scenery_Creation := create_object_sid(33554499, Scenery_Creation_Hex, 1, -1);
+                        Scenery_Creation := create_object(33554499, Scenery_Creation_Hex, 1);
                         Scenery_Creation_Count := Scenery_Creation_Count + 1;
                     end
                     Scenery_Creation_Count := 0;
                     while (Scenery_Creation_Count < 1) do begin
                         Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 0, 1);
-                        Scenery_Creation := create_object_sid(33554499, Scenery_Creation_Hex, 1, -1);
+                        Scenery_Creation := create_object(33554499, Scenery_Creation_Hex, 1);
                         Scenery_Creation_Count := Scenery_Creation_Count + 1;
                     end
                     Scenery_Creation_Count := 0;
                     while (Scenery_Creation_Count < 1) do begin
                         Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 5, 1);
-                        Scenery_Creation := create_object_sid(33554499, Scenery_Creation_Hex, 1, -1);
+                        Scenery_Creation := create_object(33554499, Scenery_Creation_Hex, 1);
                         Scenery_Creation_Count := Scenery_Creation_Count + 1;
                     end
                     Scenery_Creation_Count := 0;
                     while (Scenery_Creation_Count < 2) do begin
                         Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 0, 1);
-                        Scenery_Creation := create_object_sid(33554499, Scenery_Creation_Hex, 1, -1);
+                        Scenery_Creation := create_object(33554499, Scenery_Creation_Hex, 1);
                         Scenery_Creation_Count := Scenery_Creation_Count + 1;
                     end
                     Scenery_Creation_Count := 0;
                     while (Scenery_Creation_Count < 1) do begin
                         Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 1, 1);
-                        Scenery_Creation := create_object_sid(33554499, Scenery_Creation_Hex, 1, -1);
+                        Scenery_Creation := create_object(33554499, Scenery_Creation_Hex, 1);
                         Scenery_Creation_Count := Scenery_Creation_Count + 1;
                     end
                     Scenery_Creation_Count := 0;
                     while (Scenery_Creation_Count < 1) do begin
                         Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 2, 1);
-                        Scenery_Creation := create_object_sid(33554499, Scenery_Creation_Hex, 1, -1);
+                        Scenery_Creation := create_object(33554499, Scenery_Creation_Hex, 1);
                         Scenery_Creation_Count := Scenery_Creation_Count + 1;
                     end
                 end
@@ -472,8 +472,8 @@ begin
                         Scenery_Creation_Ptr := party_member_obj(455);
                     end
                     else begin
-                        if (global_var(18) == 0) then begin
-                            Scenery_Creation_Ptr := create_object_sid(455, 0, 0, 920);
+                        if (global_var(GVAR_PLAYER_GOT_CAR) == 0) then begin
+                            Scenery_Creation_Ptr := create_object_sid(455, 0, 0, SCRIPT_ZICRTRNK);
                         end
                     end
                     if (Scenery_Creation_Ptr != 0) then begin
@@ -482,13 +482,13 @@ begin
                         Scenery_Creation_Count := 0;
                         while (Scenery_Creation_Count < 1) do begin
                             Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 1, 1);
-                            Scenery_Creation := create_object_sid(33554499, Scenery_Creation_Hex, 1, -1);
+                            Scenery_Creation := create_object(33554499, Scenery_Creation_Hex, 1);
                             Scenery_Creation_Count := Scenery_Creation_Count + 1;
                         end
                         Scenery_Creation_Count := 0;
                         while (Scenery_Creation_Count < 1) do begin
                             Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 2, 1);
-                            Scenery_Creation := create_object_sid(33554499, Scenery_Creation_Hex, 1, -1);
+                            Scenery_Creation := create_object(33554499, Scenery_Creation_Hex, 1);
                             Scenery_Creation_Count := Scenery_Creation_Count + 1;
                         end
                     end
@@ -500,106 +500,106 @@ begin
         end
         else begin
             debug_msg("Check_Create_Car(" + 29718 + "," + 1 + ")");
-            if (not(metarule(22, 0))) then begin
-                if (global_var(18) != 0) then begin
-                    debug_msg("car_current_town == " + metarule(30, 0) + " / map_get_load_area == " + metarule3(111, 0, 0, 0));
-                    if ((metarule(30, 0) == metarule3(111, 0, 0, 0)) or ((26 <= metarule(30, 0)) and (metarule(30, 0) <= 29) and (metarule3(111, 0, 0, 0) == -1))) then begin
+            if (not(is_loading_game)) then begin
+                if (global_var(GVAR_PLAYER_GOT_CAR) != 0) then begin
+                    debug_msg("car_current_town == " + car_current_town + " / map_get_load_area == " + map_get_load_area);
+                    if ((car_current_town == map_get_load_area) or ((26 <= car_current_town) and (car_current_town <= 29) and (map_get_load_area == -1))) then begin
                         debug_msg("Place_Car(" + 29718 + "," + 1 + ")");
-                        debug_msg("   GVAR_CAR_PLACED_TILE == " + global_var(633));
+                        debug_msg("   GVAR_CAR_PLACED_TILE == " + global_var(GVAR_CAR_PLACED_TILE));
                         debug_msg("   Car_At_Loc == " + (tile_contains_pid_obj(29718, 1, 33555441) != 0));
-                        if ((global_var(633) <= 0) or (29718 == global_var(633))) then begin
-                            set_global_var(633, 29718);
+                        if ((global_var(GVAR_CAR_PLACED_TILE) <= 0) or (29718 == global_var(GVAR_CAR_PLACED_TILE))) then begin
+                            set_global_var(GVAR_CAR_PLACED_TILE, 29718);
                             if (not(tile_contains_pid_obj(29718, 1, 33555441) != 0)) then begin
-                                Scenery_Creation := create_object_sid(33555441, 29718, 1, 304);
+                                Scenery_Creation := create_object_sid(33555441, 29718, 1, SCRIPT_ZSDRVCAR);
                                 Scenery_Creation_Hex := 29718;
                                 Scenery_Creation_Count := 0;
                                 while (Scenery_Creation_Count < 2) do begin
                                     Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 1, 1);
-                                    Scenery_Creation := create_object_sid(33554499, Scenery_Creation_Hex, 1, -1);
+                                    Scenery_Creation := create_object(33554499, Scenery_Creation_Hex, 1);
                                     Scenery_Creation_Count := Scenery_Creation_Count + 1;
                                 end
                                 Scenery_Creation_Count := 0;
                                 while (Scenery_Creation_Count < 2) do begin
                                     Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 2, 1);
-                                    Scenery_Creation := create_object_sid(33554499, Scenery_Creation_Hex, 1, -1);
+                                    Scenery_Creation := create_object(33554499, Scenery_Creation_Hex, 1);
                                     Scenery_Creation_Count := Scenery_Creation_Count + 1;
                                 end
                                 Scenery_Creation_Count := 0;
                                 while (Scenery_Creation_Count < 2) do begin
                                     Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 3, 1);
-                                    Scenery_Creation := create_object_sid(33554499, Scenery_Creation_Hex, 1, -1);
+                                    Scenery_Creation := create_object(33554499, Scenery_Creation_Hex, 1);
                                     Scenery_Creation_Count := Scenery_Creation_Count + 1;
                                 end
                                 Scenery_Creation_Count := 0;
                                 while (Scenery_Creation_Count < 2) do begin
                                     Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 4, 1);
-                                    Scenery_Creation := create_object_sid(33554499, Scenery_Creation_Hex, 1, -1);
+                                    Scenery_Creation := create_object(33554499, Scenery_Creation_Hex, 1);
                                     Scenery_Creation_Count := Scenery_Creation_Count + 1;
                                 end
                                 Scenery_Creation_Count := 0;
                                 while (Scenery_Creation_Count < 1) do begin
                                     Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 5, 1);
-                                    Scenery_Creation := create_object_sid(33554499, Scenery_Creation_Hex, 1, -1);
+                                    Scenery_Creation := create_object(33554499, Scenery_Creation_Hex, 1);
                                     Scenery_Creation_Count := Scenery_Creation_Count + 1;
                                 end
                                 Scenery_Creation_Count := 0;
                                 while (Scenery_Creation_Count < 1) do begin
                                     Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 4, 1);
-                                    Scenery_Creation := create_object_sid(33554499, Scenery_Creation_Hex, 1, -1);
+                                    Scenery_Creation := create_object(33554499, Scenery_Creation_Hex, 1);
                                     Scenery_Creation_Count := Scenery_Creation_Count + 1;
                                 end
                                 Scenery_Creation_Count := 0;
                                 while (Scenery_Creation_Count < 1) do begin
                                     Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 5, 1);
-                                    Scenery_Creation := create_object_sid(33554499, Scenery_Creation_Hex, 1, -1);
+                                    Scenery_Creation := create_object(33554499, Scenery_Creation_Hex, 1);
                                     Scenery_Creation_Count := Scenery_Creation_Count + 1;
                                 end
                                 Scenery_Creation_Count := 0;
                                 while (Scenery_Creation_Count < 1) do begin
                                     Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 4, 1);
-                                    Scenery_Creation := create_object_sid(33554499, Scenery_Creation_Hex, 1, -1);
+                                    Scenery_Creation := create_object(33554499, Scenery_Creation_Hex, 1);
                                     Scenery_Creation_Count := Scenery_Creation_Count + 1;
                                 end
                                 Scenery_Creation_Count := 0;
                                 while (Scenery_Creation_Count < 1) do begin
                                     Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 5, 1);
-                                    Scenery_Creation := create_object_sid(33554499, Scenery_Creation_Hex, 1, -1);
+                                    Scenery_Creation := create_object(33554499, Scenery_Creation_Hex, 1);
                                     Scenery_Creation_Count := Scenery_Creation_Count + 1;
                                 end
                                 Scenery_Creation_Count := 0;
                                 while (Scenery_Creation_Count < 1) do begin
                                     Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 4, 1);
-                                    Scenery_Creation := create_object_sid(33554499, Scenery_Creation_Hex, 1, -1);
+                                    Scenery_Creation := create_object(33554499, Scenery_Creation_Hex, 1);
                                     Scenery_Creation_Count := Scenery_Creation_Count + 1;
                                 end
                                 Scenery_Creation_Count := 0;
                                 while (Scenery_Creation_Count < 1) do begin
                                     Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 0, 1);
-                                    Scenery_Creation := create_object_sid(33554499, Scenery_Creation_Hex, 1, -1);
+                                    Scenery_Creation := create_object(33554499, Scenery_Creation_Hex, 1);
                                     Scenery_Creation_Count := Scenery_Creation_Count + 1;
                                 end
                                 Scenery_Creation_Count := 0;
                                 while (Scenery_Creation_Count < 1) do begin
                                     Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 5, 1);
-                                    Scenery_Creation := create_object_sid(33554499, Scenery_Creation_Hex, 1, -1);
+                                    Scenery_Creation := create_object(33554499, Scenery_Creation_Hex, 1);
                                     Scenery_Creation_Count := Scenery_Creation_Count + 1;
                                 end
                                 Scenery_Creation_Count := 0;
                                 while (Scenery_Creation_Count < 2) do begin
                                     Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 0, 1);
-                                    Scenery_Creation := create_object_sid(33554499, Scenery_Creation_Hex, 1, -1);
+                                    Scenery_Creation := create_object(33554499, Scenery_Creation_Hex, 1);
                                     Scenery_Creation_Count := Scenery_Creation_Count + 1;
                                 end
                                 Scenery_Creation_Count := 0;
                                 while (Scenery_Creation_Count < 1) do begin
                                     Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 1, 1);
-                                    Scenery_Creation := create_object_sid(33554499, Scenery_Creation_Hex, 1, -1);
+                                    Scenery_Creation := create_object(33554499, Scenery_Creation_Hex, 1);
                                     Scenery_Creation_Count := Scenery_Creation_Count + 1;
                                 end
                                 Scenery_Creation_Count := 0;
                                 while (Scenery_Creation_Count < 1) do begin
                                     Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 2, 1);
-                                    Scenery_Creation := create_object_sid(33554499, Scenery_Creation_Hex, 1, -1);
+                                    Scenery_Creation := create_object(33554499, Scenery_Creation_Hex, 1);
                                     Scenery_Creation_Count := Scenery_Creation_Count + 1;
                                 end
                             end
@@ -612,8 +612,8 @@ begin
                                     Scenery_Creation_Ptr := party_member_obj(455);
                                 end
                                 else begin
-                                    if (global_var(18) == 0) then begin
-                                        Scenery_Creation_Ptr := create_object_sid(455, 0, 0, 920);
+                                    if (global_var(GVAR_PLAYER_GOT_CAR) == 0) then begin
+                                        Scenery_Creation_Ptr := create_object_sid(455, 0, 0, SCRIPT_ZICRTRNK);
                                     end
                                 end
                                 if (Scenery_Creation_Ptr != 0) then begin
@@ -622,13 +622,13 @@ begin
                                     Scenery_Creation_Count := 0;
                                     while (Scenery_Creation_Count < 1) do begin
                                         Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 1, 1);
-                                        Scenery_Creation := create_object_sid(33554499, Scenery_Creation_Hex, 1, -1);
+                                        Scenery_Creation := create_object(33554499, Scenery_Creation_Hex, 1);
                                         Scenery_Creation_Count := Scenery_Creation_Count + 1;
                                     end
                                     Scenery_Creation_Count := 0;
                                     while (Scenery_Creation_Count < 1) do begin
                                         Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 2, 1);
-                                        Scenery_Creation := create_object_sid(33554499, Scenery_Creation_Hex, 1, -1);
+                                        Scenery_Creation := create_object(33554499, Scenery_Creation_Hex, 1);
                                         Scenery_Creation_Count := Scenery_Creation_Count + 1;
                                     end
                                 end
@@ -788,106 +788,106 @@ begin
     end
     else begin
         debug_msg("Check_Create_Car(" + 26932 + "," + 0 + ")");
-        if (not(metarule(22, 0))) then begin
-            if (global_var(18) != 0) then begin
-                debug_msg("car_current_town == " + metarule(30, 0) + " / map_get_load_area == " + metarule3(111, 0, 0, 0));
-                if ((metarule(30, 0) == metarule3(111, 0, 0, 0)) or ((26 <= metarule(30, 0)) and (metarule(30, 0) <= 29) and (metarule3(111, 0, 0, 0) == -1))) then begin
+        if (not(is_loading_game)) then begin
+            if (global_var(GVAR_PLAYER_GOT_CAR) != 0) then begin
+                debug_msg("car_current_town == " + car_current_town + " / map_get_load_area == " + map_get_load_area);
+                if ((car_current_town == map_get_load_area) or ((26 <= car_current_town) and (car_current_town <= 29) and (map_get_load_area == -1))) then begin
                     debug_msg("Place_Car(" + 26932 + "," + 0 + ")");
-                    debug_msg("   GVAR_CAR_PLACED_TILE == " + global_var(633));
+                    debug_msg("   GVAR_CAR_PLACED_TILE == " + global_var(GVAR_CAR_PLACED_TILE));
                     debug_msg("   Car_At_Loc == " + (tile_contains_pid_obj(26932, 0, 33555441) != 0));
-                    if ((global_var(633) <= 0) or (26932 == global_var(633))) then begin
-                        set_global_var(633, 26932);
+                    if ((global_var(GVAR_CAR_PLACED_TILE) <= 0) or (26932 == global_var(GVAR_CAR_PLACED_TILE))) then begin
+                        set_global_var(GVAR_CAR_PLACED_TILE, 26932);
                         if (not(tile_contains_pid_obj(26932, 0, 33555441) != 0)) then begin
-                            Scenery_Creation := create_object_sid(33555441, 26932, 0, 304);
+                            Scenery_Creation := create_object_sid(33555441, 26932, 0, SCRIPT_ZSDRVCAR);
                             Scenery_Creation_Hex := 26932;
                             Scenery_Creation_Count := 0;
                             while (Scenery_Creation_Count < 2) do begin
                                 Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 1, 1);
-                                Scenery_Creation := create_object_sid(33554499, Scenery_Creation_Hex, 0, -1);
+                                Scenery_Creation := create_object(33554499, Scenery_Creation_Hex, 0);
                                 Scenery_Creation_Count := Scenery_Creation_Count + 1;
                             end
                             Scenery_Creation_Count := 0;
                             while (Scenery_Creation_Count < 2) do begin
                                 Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 2, 1);
-                                Scenery_Creation := create_object_sid(33554499, Scenery_Creation_Hex, 0, -1);
+                                Scenery_Creation := create_object(33554499, Scenery_Creation_Hex, 0);
                                 Scenery_Creation_Count := Scenery_Creation_Count + 1;
                             end
                             Scenery_Creation_Count := 0;
                             while (Scenery_Creation_Count < 2) do begin
                                 Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 3, 1);
-                                Scenery_Creation := create_object_sid(33554499, Scenery_Creation_Hex, 0, -1);
+                                Scenery_Creation := create_object(33554499, Scenery_Creation_Hex, 0);
                                 Scenery_Creation_Count := Scenery_Creation_Count + 1;
                             end
                             Scenery_Creation_Count := 0;
                             while (Scenery_Creation_Count < 2) do begin
                                 Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 4, 1);
-                                Scenery_Creation := create_object_sid(33554499, Scenery_Creation_Hex, 0, -1);
+                                Scenery_Creation := create_object(33554499, Scenery_Creation_Hex, 0);
                                 Scenery_Creation_Count := Scenery_Creation_Count + 1;
                             end
                             Scenery_Creation_Count := 0;
                             while (Scenery_Creation_Count < 1) do begin
                                 Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 5, 1);
-                                Scenery_Creation := create_object_sid(33554499, Scenery_Creation_Hex, 0, -1);
+                                Scenery_Creation := create_object(33554499, Scenery_Creation_Hex, 0);
                                 Scenery_Creation_Count := Scenery_Creation_Count + 1;
                             end
                             Scenery_Creation_Count := 0;
                             while (Scenery_Creation_Count < 1) do begin
                                 Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 4, 1);
-                                Scenery_Creation := create_object_sid(33554499, Scenery_Creation_Hex, 0, -1);
+                                Scenery_Creation := create_object(33554499, Scenery_Creation_Hex, 0);
                                 Scenery_Creation_Count := Scenery_Creation_Count + 1;
                             end
                             Scenery_Creation_Count := 0;
                             while (Scenery_Creation_Count < 1) do begin
                                 Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 5, 1);
-                                Scenery_Creation := create_object_sid(33554499, Scenery_Creation_Hex, 0, -1);
+                                Scenery_Creation := create_object(33554499, Scenery_Creation_Hex, 0);
                                 Scenery_Creation_Count := Scenery_Creation_Count + 1;
                             end
                             Scenery_Creation_Count := 0;
                             while (Scenery_Creation_Count < 1) do begin
                                 Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 4, 1);
-                                Scenery_Creation := create_object_sid(33554499, Scenery_Creation_Hex, 0, -1);
+                                Scenery_Creation := create_object(33554499, Scenery_Creation_Hex, 0);
                                 Scenery_Creation_Count := Scenery_Creation_Count + 1;
                             end
                             Scenery_Creation_Count := 0;
                             while (Scenery_Creation_Count < 1) do begin
                                 Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 5, 1);
-                                Scenery_Creation := create_object_sid(33554499, Scenery_Creation_Hex, 0, -1);
+                                Scenery_Creation := create_object(33554499, Scenery_Creation_Hex, 0);
                                 Scenery_Creation_Count := Scenery_Creation_Count + 1;
                             end
                             Scenery_Creation_Count := 0;
                             while (Scenery_Creation_Count < 1) do begin
                                 Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 4, 1);
-                                Scenery_Creation := create_object_sid(33554499, Scenery_Creation_Hex, 0, -1);
+                                Scenery_Creation := create_object(33554499, Scenery_Creation_Hex, 0);
                                 Scenery_Creation_Count := Scenery_Creation_Count + 1;
                             end
                             Scenery_Creation_Count := 0;
                             while (Scenery_Creation_Count < 1) do begin
                                 Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 0, 1);
-                                Scenery_Creation := create_object_sid(33554499, Scenery_Creation_Hex, 0, -1);
+                                Scenery_Creation := create_object(33554499, Scenery_Creation_Hex, 0);
                                 Scenery_Creation_Count := Scenery_Creation_Count + 1;
                             end
                             Scenery_Creation_Count := 0;
                             while (Scenery_Creation_Count < 1) do begin
                                 Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 5, 1);
-                                Scenery_Creation := create_object_sid(33554499, Scenery_Creation_Hex, 0, -1);
+                                Scenery_Creation := create_object(33554499, Scenery_Creation_Hex, 0);
                                 Scenery_Creation_Count := Scenery_Creation_Count + 1;
                             end
                             Scenery_Creation_Count := 0;
                             while (Scenery_Creation_Count < 2) do begin
                                 Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 0, 1);
-                                Scenery_Creation := create_object_sid(33554499, Scenery_Creation_Hex, 0, -1);
+                                Scenery_Creation := create_object(33554499, Scenery_Creation_Hex, 0);
                                 Scenery_Creation_Count := Scenery_Creation_Count + 1;
                             end
                             Scenery_Creation_Count := 0;
                             while (Scenery_Creation_Count < 1) do begin
                                 Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 1, 1);
-                                Scenery_Creation := create_object_sid(33554499, Scenery_Creation_Hex, 0, -1);
+                                Scenery_Creation := create_object(33554499, Scenery_Creation_Hex, 0);
                                 Scenery_Creation_Count := Scenery_Creation_Count + 1;
                             end
                             Scenery_Creation_Count := 0;
                             while (Scenery_Creation_Count < 1) do begin
                                 Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 2, 1);
-                                Scenery_Creation := create_object_sid(33554499, Scenery_Creation_Hex, 0, -1);
+                                Scenery_Creation := create_object(33554499, Scenery_Creation_Hex, 0);
                                 Scenery_Creation_Count := Scenery_Creation_Count + 1;
                             end
                         end
@@ -900,8 +900,8 @@ begin
                                 Scenery_Creation_Ptr := party_member_obj(455);
                             end
                             else begin
-                                if (global_var(18) == 0) then begin
-                                    Scenery_Creation_Ptr := create_object_sid(455, 0, 0, 920);
+                                if (global_var(GVAR_PLAYER_GOT_CAR) == 0) then begin
+                                    Scenery_Creation_Ptr := create_object_sid(455, 0, 0, SCRIPT_ZICRTRNK);
                                 end
                             end
                             if (Scenery_Creation_Ptr != 0) then begin
@@ -910,13 +910,13 @@ begin
                                 Scenery_Creation_Count := 0;
                                 while (Scenery_Creation_Count < 1) do begin
                                     Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 1, 1);
-                                    Scenery_Creation := create_object_sid(33554499, Scenery_Creation_Hex, 0, -1);
+                                    Scenery_Creation := create_object(33554499, Scenery_Creation_Hex, 0);
                                     Scenery_Creation_Count := Scenery_Creation_Count + 1;
                                 end
                                 Scenery_Creation_Count := 0;
                                 while (Scenery_Creation_Count < 1) do begin
                                     Scenery_Creation_Hex := tile_num_in_direction(Scenery_Creation_Hex, 2, 1);
-                                    Scenery_Creation := create_object_sid(33554499, Scenery_Creation_Hex, 0, -1);
+                                    Scenery_Creation := create_object(33554499, Scenery_Creation_Hex, 0);
                                     Scenery_Creation_Count := Scenery_Creation_Count + 1;
                                 end
                             end
@@ -1155,4 +1155,3 @@ begin
         end
     end
 end
-
