diff --git a/Games/Fallout2.Steam/Original/VANHAG.ssl b/Games/Fallout2.Steam/ReDefine/VANHAG.ssl
index d7895b6..b539152 100644
--- a/Games/Fallout2.Steam/Original/VANHAG.ssl
+++ b/Games/Fallout2.Steam/ReDefine/VANHAG.ssl
@@ -34,31 +34,31 @@ variable exit_line;
 procedure start
 begin
     if (not(initialized)) then begin
-        critter_add_trait(self_obj, 1, 6, 34);
-        critter_add_trait(self_obj, 1, 5, 69);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 34);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_PARTY_MACRAE_AGGRESSIVE);
         initialized := 1;
     end
-    if (script_action == 12) then begin
+    if (script_action == critter_proc) then begin
         call critter_p_proc();
     end
     else begin
-        if (script_action == 18) then begin
+        if (script_action == destroy_proc) then begin
             call destroy_p_proc();
         end
         else begin
-            if (script_action == 21) then begin
+            if (script_action == look_at_proc) then begin
                 call look_at_p_proc();
             end
             else begin
-                if (script_action == 4) then begin
+                if (script_action == pickup_proc) then begin
                     call pickup_p_proc();
                 end
                 else begin
-                    if (script_action == 11) then begin
+                    if (script_action == talk_proc) then begin
                         call talk_p_proc();
                     end
                     else begin
-                        if (script_action == 22) then begin
+                        if (script_action == timed_event_proc) then begin
                             call timed_event_p_proc();
                         end
                     end
@@ -73,9 +73,9 @@ begin
     if (tile_distance_objs(dude_obj, self_obj) < 12) then begin
         if (hostile) then begin
             hostile := 0;
-            attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+            attack(dude_obj);
         end
-        if (global_var(57) < 1) then begin
+        if (global_var(GVAR_TOWN_REP_NCR) < 1) then begin
             iseeu := 1;
         end
         if ((iseeu == 1) and (waiting == 0) and (Lt_ptr != 0) and (local_var(4) == 0)) then begin
@@ -89,17 +89,17 @@ procedure destroy_p_proc
 begin
     sequence := 10;
     if (source_obj == dude_obj) then begin
-        if (((global_var(160) + global_var(159)) >= 25) and ((global_var(159) > (2 * global_var(160))) or (global_var(156) == 1))) then begin
-            set_global_var(156, 1);
-            set_global_var(157, 0);
+        if (((global_var(GVAR_GECKO_SKEETER_PART) + global_var(GVAR_GECKO_REQ_FORM)) >= 25) and ((global_var(GVAR_GECKO_REQ_FORM) > (2 * global_var(GVAR_GECKO_SKEETER_PART))) or (global_var(GVAR_SIERRA_LOCKOUT) == 1))) then begin
+            set_global_var(GVAR_SIERRA_LOCKOUT, 1);
+            set_global_var(GVAR_SIERRA_PASSWORD, 0);
         end
-        if (((global_var(160) + global_var(159)) >= 25) and ((global_var(160) > (3 * global_var(159))) or (global_var(157) == 1))) then begin
-            set_global_var(157, 1);
-            set_global_var(156, 0);
+        if (((global_var(GVAR_GECKO_SKEETER_PART) + global_var(GVAR_GECKO_REQ_FORM)) >= 25) and ((global_var(160) > (3 * global_var(GVAR_GECKO_REQ_FORM))) or (global_var(GVAR_SIERRA_PASSWORD) == 1))) then begin
+            set_global_var(GVAR_SIERRA_PASSWORD, 1);
+            set_global_var(GVAR_SIERRA_LOCKOUT, 0);
         end
-        set_global_var(160, global_var(160) + 1);
-        if ((global_var(160) % 6) == 0) then begin
-            set_global_var(155, global_var(155) + 1);
+        set_global_var(GVAR_GECKO_SKEETER_PART, global_var(GVAR_GECKO_SKEETER_PART) + 1);
+        if ((global_var(GVAR_GECKO_SKEETER_PART) % 6) == 0) then begin
+            set_global_var(GVAR_BRAIN_BOT_BRAIN, global_var(GVAR_BRAIN_BOT_BRAIN) + 1);
         end
     end
 end
@@ -107,7 +107,7 @@ end
 procedure look_at_p_proc
 begin
     script_overrides;
-    display_msg(message_str(435, 100));
+    display_msg(message_str(SCRIPT_NCMASON, 100));
 end
 
 procedure pickup_p_proc
@@ -117,55 +117,55 @@ end
 
 procedure talk_p_proc
 begin
-    float_msg(self_obj, message_str(435, 101), 0);
+    float_msg(self_obj, message_str(SCRIPT_NCMASON, 101), FLOAT_MSG_NORMAL);
 end
 
 procedure timed_event_p_proc
 begin
     waiting := 0;
     if (sequence == 0) then begin
-        float_msg(Lt_ptr, message_str(435, 102), 0);
+        float_msg(Lt_ptr, message_str(SCRIPT_NCMASON, 102), FLOAT_MSG_NORMAL);
     end
     else begin
         if (sequence == 1) then begin
-            float_msg(self_obj, message_str(435, 103), 0);
+            float_msg(self_obj, message_str(SCRIPT_NCMASON, 103), FLOAT_MSG_NORMAL);
         end
         else begin
             if (sequence == 2) then begin
-                float_msg(Lt_ptr, message_str(435, 104), 0);
+                float_msg(Lt_ptr, message_str(SCRIPT_NCMASON, 104), FLOAT_MSG_NORMAL);
             end
             else begin
                 if (sequence == 3) then begin
-                    float_msg(self_obj, message_str(435, 105), 0);
+                    float_msg(self_obj, message_str(SCRIPT_NCMASON, 105), FLOAT_MSG_NORMAL);
                 end
                 else begin
                     if (sequence == 4) then begin
-                        float_msg(Lt_ptr, message_str(435, 106), 0);
+                        float_msg(Lt_ptr, message_str(SCRIPT_NCMASON, 106), FLOAT_MSG_NORMAL);
                     end
                     else begin
                         if (sequence == 5) then begin
-                            float_msg(self_obj, message_str(435, 107), 0);
+                            float_msg(self_obj, message_str(SCRIPT_NCMASON, 107), FLOAT_MSG_NORMAL);
                             clock := 6;
                         end
                         else begin
                             if (sequence == 6) then begin
-                                float_msg(Lt_ptr, message_str(435, 108), 0);
+                                float_msg(Lt_ptr, message_str(SCRIPT_NCMASON, 108), FLOAT_MSG_NORMAL);
                                 clock := 4;
                             end
                             else begin
                                 if (sequence == 7) then begin
-                                    float_msg(self_obj, message_str(435, 109), 0);
+                                    float_msg(self_obj, message_str(SCRIPT_NCMASON, 109), FLOAT_MSG_NORMAL);
                                 end
                                 else begin
                                     if (sequence == 8) then begin
-                                        float_msg(Lt_ptr, message_str(435, 110), 0);
+                                        float_msg(Lt_ptr, message_str(SCRIPT_NCMASON, 110), FLOAT_MSG_NORMAL);
                                     end
                                     else begin
                                         if (sequence == 9) then begin
-                                            float_msg(self_obj, message_str(435, 111), 0);
+                                            float_msg(self_obj, message_str(SCRIPT_NCMASON, 111), FLOAT_MSG_NORMAL);
                                             if (not(local_var(4))) then begin
                                                 give_exp_points(1000);
-                                                display_msg(message_str(435, 112));
+                                                display_msg(message_str(SCRIPT_NCMASON, 112));
                                                 set_local_var(4, 1);
                                             end
                                         end
@@ -187,31 +187,31 @@ begin
         set_local_var(0, 50);
         set_local_var(1, 2);
         set_local_var(2, 1);
-        set_local_var(0, local_var(0) + (5 * get_critter_stat(dude_obj, 3)) - 25);
-        set_local_var(0, local_var(0) + (10 * has_trait(0, dude_obj, 10)));
-        if (has_trait(0, dude_obj, 39)) then begin
-            if (global_var(155) > 0) then begin
-                set_local_var(0, local_var(0) + global_var(155));
+        set_local_var(0, local_var(0) + (5 * dude_charisma) - 25);
+        set_local_var(0, local_var(0) + (10 * has_trait(TRAIT_PERK, dude_obj, PERK_presence)));
+        if (has_trait(TRAIT_PERK, dude_obj, PERK_cult_of_personality)) then begin
+            if (global_var(GVAR_BRAIN_BOT_BRAIN) > 0) then begin
+                set_local_var(0, local_var(0) + global_var(GVAR_BRAIN_BOT_BRAIN));
             end
             else begin
-                set_local_var(0, local_var(0) - global_var(155));
+                set_local_var(0, local_var(0) - global_var(GVAR_BRAIN_BOT_BRAIN));
             end
         end
         else begin
             if (local_var(3) == 1) then begin
-                set_local_var(0, local_var(0) - global_var(155));
+                set_local_var(0, local_var(0) - global_var(GVAR_BRAIN_BOT_BRAIN));
             end
             else begin
-                set_local_var(0, local_var(0) + global_var(155));
+                set_local_var(0, local_var(0) + global_var(GVAR_BRAIN_BOT_BRAIN));
             end
         end
-        if (global_var(158) > 2) then begin
+        if (global_var(GVAR_GECKO_ECON_DISK) > 2) then begin
             set_local_var(0, local_var(0) - 30);
         end
-        if (((global_var(160) + global_var(159)) >= 25) and ((global_var(160) > (3 * global_var(159))) or (global_var(157) == 1))) then begin
+        if (((global_var(GVAR_GECKO_SKEETER_PART) + global_var(GVAR_GECKO_REQ_FORM)) >= 25) and ((global_var(160) > (3 * global_var(GVAR_GECKO_REQ_FORM))) or (global_var(GVAR_SIERRA_PASSWORD) == 1))) then begin
             set_local_var(0, local_var(0) + 20);
         end
-        if (((global_var(160) + global_var(159)) >= 25) and ((global_var(159) > (2 * global_var(160))) or (global_var(156) == 1))) then begin
+        if (((global_var(GVAR_GECKO_SKEETER_PART) + global_var(GVAR_GECKO_REQ_FORM)) >= 25) and ((global_var(GVAR_GECKO_REQ_FORM) > (2 * global_var(GVAR_GECKO_SKEETER_PART))) or (global_var(GVAR_SIERRA_LOCKOUT) == 1))) then begin
             set_local_var(0, local_var(0) - 20);
         end
         call ReactToLevel();
@@ -304,6 +304,5 @@ end
 
 procedure Goodbyes
 begin
-    exit_line := message_str(634, random(100, 105));
+    exit_line := message_str(SCRIPT_BSCOMP3, random(100, 105));
 end
-
