diff --git a/Games/Fallout2.Steam/Original/VCKOHL.ssl b/Games/Fallout2.Steam/ReDefine/VCKOHL.ssl
index 23c9338..20763bb 100644
--- a/Games/Fallout2.Steam/Original/VCKOHL.ssl
+++ b/Games/Fallout2.Steam/ReDefine/VCKOHL.ssl
@@ -182,9 +182,9 @@ end
 
 procedure critter_p_proc
 begin
-    if ((global_var(137) == 1) or (local_var(5) == 2) and obj_can_see_obj(self_obj, dude_obj)) then begin
+    if ((global_var(GVAR_ENEMY_VAULT_CITY) == 1) or (local_var(5) == 2) and obj_can_see_obj(self_obj, dude_obj)) then begin
         set_local_var(5, 1);
-        attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+        attack(dude_obj);
     end
 end
 
@@ -200,48 +200,48 @@ begin
     Evil_Critter := 0;
     Slavery_Tolerant := 2;
     Karma_Perception := 1;
-    set_global_var(37, 0);
-    set_global_var(38, 0);
-    set_global_var(39, 0);
-    set_global_var(40, 0);
-    set_global_var(41, 0);
-    set_global_var(42, 0);
-    set_global_var(43, 0);
-    set_global_var(44, 0);
-    set_global_var(45, 0);
-    if (global_var(0) >= 1000) then begin
-        set_global_var(37, 1);
+    set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+    set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+    set_global_var(GVAR_KARMA_DEFENDER, 0);
+    set_global_var(GVAR_KARMA_WANDERER, 0);
+    set_global_var(GVAR_KARMA_BETRAYER, 0);
+    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+    set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+    set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+    if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
     end
     else begin
-        if (global_var(0) >= 750) then begin
-            set_global_var(38, 1);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+            set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
         end
         else begin
-            if (global_var(0) >= 500) then begin
-                set_global_var(39, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
             end
             else begin
-                if (global_var(0) >= 250) then begin
-                    set_global_var(40, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                    set_global_var(GVAR_KARMA_DEFENDER, 1);
                 end
                 else begin
-                    if (global_var(0) > -250) then begin
-                        set_global_var(41, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                        set_global_var(GVAR_KARMA_WANDERER, 1);
                     end
                     else begin
-                        if (global_var(0) > -500) then begin
-                            set_global_var(42, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                            set_global_var(GVAR_KARMA_BETRAYER, 1);
                         end
                         else begin
-                            if (global_var(0) > -750) then begin
-                                set_global_var(43, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                             end
                             else begin
-                                if (global_var(0) > -1000) then begin
-                                    set_global_var(44, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                    set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                 end
                                 else begin
-                                    set_global_var(45, 1);
+                                    set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                 end
                             end
                         end
@@ -252,7 +252,7 @@ begin
     end
     if (local_var(2) == 0) then begin
         set_local_var(2, 1);
-        set_local_var(3, (get_critter_stat(dude_obj, 3) - get_critter_stat(self_obj, 3)) * 5);
+        set_local_var(3, (dude_charisma - self_charisma) * 5);
         debug_msg("Base Reaction == " + local_var(3));
     end
     Static_Reaction := 0;
@@ -264,14 +264,14 @@ begin
     else begin
         reaction_bonus_karma := -1 * (-1 * reaction_bonus_karma / 1000);
     end
-    if (global_var(50) >= 0) then begin
-        reaction_bonus_town_rep := global_var(50) / 2;
+    if (global_var(GVAR_TOWN_REP_VAULT_CITY) >= 0) then begin
+        reaction_bonus_town_rep := global_var(GVAR_TOWN_REP_VAULT_CITY) / 2;
     end
     else begin
-        reaction_bonus_town_rep := -1 * (-1 * global_var(50) / 2);
+        reaction_bonus_town_rep := -1 * (-1 * global_var(GVAR_TOWN_REP_VAULT_CITY) / 2);
     end
-    if (has_trait(2, dude_obj, 13)) then begin
-        if (get_critter_stat(dude_obj, 34) == get_critter_stat(self_obj, 34)) then begin
+    if (has_trait(TRAIT_TRAIT, dude_obj, TRAIT_sex_appeal)) then begin
+        if (dude_gender == self_gender) then begin
             Static_Reaction := Static_Reaction + -20;
         end
         else begin
@@ -287,48 +287,48 @@ begin
         debug_msg("Karma Reaction Bonus == " + Static_Reaction);
         Static_Reaction := Static_Reaction - reaction_bonus_town_rep;
         debug_msg("Town Rep Bonus == " + Static_Reaction);
-        set_global_var(37, 0);
-        set_global_var(38, 0);
-        set_global_var(39, 0);
-        set_global_var(40, 0);
-        set_global_var(41, 0);
-        set_global_var(42, 0);
-        set_global_var(43, 0);
-        set_global_var(44, 0);
-        set_global_var(45, 0);
-        if (global_var(0) >= 1000) then begin
-            set_global_var(37, 1);
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+        set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+        set_global_var(GVAR_KARMA_DEFENDER, 0);
+        set_global_var(GVAR_KARMA_WANDERER, 0);
+        set_global_var(GVAR_KARMA_BETRAYER, 0);
+        set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+        set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+            set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
         end
         else begin
-            if (global_var(0) >= 750) then begin
-                set_global_var(38, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
             end
             else begin
-                if (global_var(0) >= 500) then begin
-                    set_global_var(39, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                 end
                 else begin
-                    if (global_var(0) >= 250) then begin
-                        set_global_var(40, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                        set_global_var(GVAR_KARMA_DEFENDER, 1);
                     end
                     else begin
-                        if (global_var(0) > -250) then begin
-                            set_global_var(41, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                            set_global_var(GVAR_KARMA_WANDERER, 1);
                         end
                         else begin
-                            if (global_var(0) > -500) then begin
-                                set_global_var(42, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                set_global_var(GVAR_KARMA_BETRAYER, 1);
                             end
                             else begin
-                                if (global_var(0) > -750) then begin
-                                    set_global_var(43, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -1000) then begin
-                                        set_global_var(44, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                     end
                                     else begin
-                                        set_global_var(45, 1);
+                                        set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                     end
                                 end
                             end
@@ -337,35 +337,35 @@ begin
                 end
             end
         end
-        if (global_var(37) == 1) then begin
+        if (global_var(GVAR_KARMA_HOLY_WARRIOR) == 1) then begin
             Static_Reaction := Static_Reaction - 20;
         end
         else begin
-            if (global_var(38) == 1) then begin
+            if (global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES) == 1) then begin
                 Static_Reaction := Static_Reaction - 15;
             end
             else begin
-                if (global_var(39) == 1) then begin
+                if (global_var(GVAR_KARMA_SHIELD_OF_HOPE) == 1) then begin
                     Static_Reaction := Static_Reaction - 10;
                 end
                 else begin
-                    if (global_var(40) == 1) then begin
+                    if (global_var(GVAR_KARMA_DEFENDER) == 1) then begin
                         Static_Reaction := Static_Reaction - 5;
                     end
                     else begin
-                        if (global_var(41) == 1) then begin
+                        if (global_var(GVAR_KARMA_WANDERER) == 1) then begin
                             Static_Reaction := Static_Reaction - 0;
                         end
                         else begin
-                            if (global_var(42) == 1) then begin
+                            if (global_var(GVAR_KARMA_BETRAYER) == 1) then begin
                                 Static_Reaction := Static_Reaction - -5;
                             end
                             else begin
-                                if (global_var(43) == 1) then begin
+                                if (global_var(GVAR_KARMA_SWORD_OF_DESPAIR) == 1) then begin
                                     Static_Reaction := Static_Reaction - -10;
                                 end
                                 else begin
-                                    if (global_var(44) == 1) then begin
+                                    if (global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE) == 1) then begin
                                         Static_Reaction := Static_Reaction - -15;
                                     end
                                     else begin
@@ -385,48 +385,48 @@ begin
         debug_msg("Karma Reaction Bonus == " + Static_Reaction);
         Static_Reaction := Static_Reaction + reaction_bonus_town_rep;
         debug_msg("Town Rep Bonus == " + Static_Reaction);
-        set_global_var(37, 0);
-        set_global_var(38, 0);
-        set_global_var(39, 0);
-        set_global_var(40, 0);
-        set_global_var(41, 0);
-        set_global_var(42, 0);
-        set_global_var(43, 0);
-        set_global_var(44, 0);
-        set_global_var(45, 0);
-        if (global_var(0) >= 1000) then begin
-            set_global_var(37, 1);
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+        set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+        set_global_var(GVAR_KARMA_DEFENDER, 0);
+        set_global_var(GVAR_KARMA_WANDERER, 0);
+        set_global_var(GVAR_KARMA_BETRAYER, 0);
+        set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+        set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+            set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
         end
         else begin
-            if (global_var(0) >= 750) then begin
-                set_global_var(38, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
             end
             else begin
-                if (global_var(0) >= 500) then begin
-                    set_global_var(39, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                 end
                 else begin
-                    if (global_var(0) >= 250) then begin
-                        set_global_var(40, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                        set_global_var(GVAR_KARMA_DEFENDER, 1);
                     end
                     else begin
-                        if (global_var(0) > -250) then begin
-                            set_global_var(41, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                            set_global_var(GVAR_KARMA_WANDERER, 1);
                         end
                         else begin
-                            if (global_var(0) > -500) then begin
-                                set_global_var(42, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                set_global_var(GVAR_KARMA_BETRAYER, 1);
                             end
                             else begin
-                                if (global_var(0) > -750) then begin
-                                    set_global_var(43, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -1000) then begin
-                                        set_global_var(44, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                     end
                                     else begin
-                                        set_global_var(45, 1);
+                                        set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                     end
                                 end
                             end
@@ -435,35 +435,35 @@ begin
                 end
             end
         end
-        if (global_var(37) == 1) then begin
+        if (global_var(GVAR_KARMA_HOLY_WARRIOR) == 1) then begin
             Static_Reaction := Static_Reaction + 20;
         end
         else begin
-            if (global_var(38) == 1) then begin
+            if (global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES) == 1) then begin
                 Static_Reaction := Static_Reaction + 15;
             end
             else begin
-                if (global_var(39) == 1) then begin
+                if (global_var(GVAR_KARMA_SHIELD_OF_HOPE) == 1) then begin
                     Static_Reaction := Static_Reaction + 10;
                 end
                 else begin
-                    if (global_var(40) == 1) then begin
+                    if (global_var(GVAR_KARMA_DEFENDER) == 1) then begin
                         Static_Reaction := Static_Reaction + 5;
                     end
                     else begin
-                        if (global_var(41) == 1) then begin
+                        if (global_var(GVAR_KARMA_WANDERER) == 1) then begin
                             Static_Reaction := Static_Reaction + 0;
                         end
                         else begin
-                            if (global_var(42) == 1) then begin
+                            if (global_var(GVAR_KARMA_BETRAYER) == 1) then begin
                                 Static_Reaction := Static_Reaction + -5;
                             end
                             else begin
-                                if (global_var(43) == 1) then begin
+                                if (global_var(GVAR_KARMA_SWORD_OF_DESPAIR) == 1) then begin
                                     Static_Reaction := Static_Reaction + -10;
                                 end
                                 else begin
-                                    if (global_var(44) == 1) then begin
+                                    if (global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE) == 1) then begin
                                         Static_Reaction := Static_Reaction + -15;
                                     end
                                     else begin
@@ -479,28 +479,28 @@ begin
         debug_msg("Good Person Reaction Bonus == " + Static_Reaction);
     end
     if (Slavery_Tolerant == 0) then begin
-        if (global_var(11) == 1) then begin
+        if (global_var(GVAR_REPUTATION_SLAVER) == 1) then begin
             Static_Reaction := Static_Reaction + (2 * -25);
         end
     end
     else begin
         if (Slavery_Tolerant == 1) then begin
-            if (global_var(11) == 1) then begin
+            if (global_var(GVAR_REPUTATION_SLAVER) == 1) then begin
                 Static_Reaction := Static_Reaction + -25;
             end
         end
         else begin
             if (Slavery_Tolerant == 3) then begin
-                if (global_var(11) == 1) then begin
+                if (global_var(GVAR_REPUTATION_SLAVER) == 1) then begin
                     Static_Reaction := Static_Reaction - -25;
                 end
             end
         end
     end
     debug_msg("Slaver + Aligned Reaction == " + Static_Reaction);
-    Static_Reaction := Static_Reaction + (10 * has_trait(0, dude_obj, 10));
+    Static_Reaction := Static_Reaction + (10 * has_trait(TRAIT_PERK, dude_obj, PERK_presence));
     debug_msg("Presence Reaction == " + Static_Reaction);
-    if (global_var(1) >= 2) then begin
+    if (global_var(GVAR_CHILDKILLER_REPUTATION) >= 2) then begin
         Static_Reaction := Static_Reaction + -30;
     end
     debug_msg("Childkiller Reaction == " + Static_Reaction);
@@ -508,19 +508,19 @@ begin
     debug_msg("Initial Reaction == " + local_var(0));
     From_WTG := 1;
     From_17 := 0;
-    if (global_var(494) == 8) then begin
+    if (global_var(GVAR_FALLOUT_2) == 8) then begin
         call Node001();
     end
     else begin
-        if (global_var(79) == 6) then begin
+        if (global_var(GVAR_VAULT_CITIZEN) == 6) then begin
             call Node002();
         end
         else begin
-            if ((party_member_obj(16777377) != 0) and (global_var(79) != 5)) then begin
+            if ((party_member_obj(16777377) != 0) and (global_var(GVAR_VAULT_CITIZEN) != 5)) then begin
                 call Node003();
             end
             else begin
-                if ((party_member_obj(16777323) != 0) and (global_var(79) != 5)) then begin
+                if ((party_member_obj(16777323) != 0) and (global_var(GVAR_VAULT_CITIZEN) != 5)) then begin
                     call Node004();
                 end
                 else begin
@@ -553,59 +553,59 @@ end
 procedure destroy_p_proc
 begin
     if (source_obj == dude_obj) then begin
-        set_global_var(5, global_var(5) + 1);
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(4) > (3 * global_var(5))) or (global_var(2) == 1)) and (global_var(1) == 0)) then begin
-            set_global_var(2, 1);
-            set_global_var(3, 0);
+        set_global_var(GVAR_GOOD_MONSTER, global_var(GVAR_GOOD_MONSTER) + 1);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(4) > (3 * global_var(GVAR_GOOD_MONSTER))) or (global_var(GVAR_CHAMPION_REPUTATION) == 1)) and (global_var(GVAR_CHILDKILLER_REPUTATION) == 0)) then begin
+            set_global_var(GVAR_CHAMPION_REPUTATION, 1);
+            set_global_var(GVAR_BERSERKER_REPUTATION, 0);
         end
-        if (metarule(51, self_obj) == 2) then begin
+        if (critter_kill_type(self_obj) == KILL_TYPE_children_kills) then begin
             if (source_obj == dude_obj) then begin
-                set_global_var(2, 0);
-                set_global_var(1, global_var(1) + 1);
-                set_global_var(0, global_var(0) + -15);
+                set_global_var(GVAR_CHAMPION_REPUTATION, 0);
+                set_global_var(GVAR_CHILDKILLER_REPUTATION, global_var(GVAR_CHILDKILLER_REPUTATION) + 1);
+                set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + -15);
                 debug_msg("Player gains " + -15 + " Karma Points.");
-                set_global_var(37, 0);
-                set_global_var(38, 0);
-                set_global_var(39, 0);
-                set_global_var(40, 0);
-                set_global_var(41, 0);
-                set_global_var(42, 0);
-                set_global_var(43, 0);
-                set_global_var(44, 0);
-                set_global_var(45, 0);
-                if (global_var(0) >= 1000) then begin
-                    set_global_var(37, 1);
+                set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+                set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+                set_global_var(GVAR_KARMA_DEFENDER, 0);
+                set_global_var(GVAR_KARMA_WANDERER, 0);
+                set_global_var(GVAR_KARMA_BETRAYER, 0);
+                set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+                set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+                set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+                    set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
                 end
                 else begin
-                    if (global_var(0) >= 750) then begin
-                        set_global_var(38, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
                     end
                     else begin
-                        if (global_var(0) >= 500) then begin
-                            set_global_var(39, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                            set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                         end
                         else begin
-                            if (global_var(0) >= 250) then begin
-                                set_global_var(40, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                                set_global_var(GVAR_KARMA_DEFENDER, 1);
                             end
                             else begin
-                                if (global_var(0) > -250) then begin
-                                    set_global_var(41, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                                    set_global_var(GVAR_KARMA_WANDERER, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -500) then begin
-                                        set_global_var(42, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                        set_global_var(GVAR_KARMA_BETRAYER, 1);
                                     end
                                     else begin
-                                        if (global_var(0) > -750) then begin
-                                            set_global_var(43, 1);
+                                        if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                            set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                         end
                                         else begin
-                                            if (global_var(0) > -1000) then begin
-                                                set_global_var(44, 1);
+                                            if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                                set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                             end
                                             else begin
-                                                set_global_var(45, 1);
+                                                set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                             end
                                         end
                                     end
@@ -614,58 +614,58 @@ begin
                         end
                     end
                 end
-                set_global_var(50, global_var(50) + -8);
+                set_global_var(GVAR_TOWN_REP_VAULT_CITY, global_var(GVAR_TOWN_REP_VAULT_CITY) + -8);
                 debug_msg("Added " + -8 + " to Town Rep");
             end
         end
-        if (((global_var(4) + global_var(5)) >= 25) and ((global_var(5) > (2 * global_var(4))) or (global_var(3) == 1))) then begin
-            set_global_var(3, 1);
-            set_global_var(2, 0);
+        if (((global_var(GVAR_BAD_MONSTER) + global_var(GVAR_GOOD_MONSTER)) >= 25) and ((global_var(GVAR_GOOD_MONSTER) > (2 * global_var(GVAR_BAD_MONSTER))) or (global_var(GVAR_BERSERKER_REPUTATION) == 1))) then begin
+            set_global_var(GVAR_BERSERKER_REPUTATION, 1);
+            set_global_var(GVAR_CHAMPION_REPUTATION, 0);
         end
-        set_global_var(0, global_var(0) + -10);
+        set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + -10);
         debug_msg("Player gains " + -10 + " Karma Points.");
-        set_global_var(37, 0);
-        set_global_var(38, 0);
-        set_global_var(39, 0);
-        set_global_var(40, 0);
-        set_global_var(41, 0);
-        set_global_var(42, 0);
-        set_global_var(43, 0);
-        set_global_var(44, 0);
-        set_global_var(45, 0);
-        if (global_var(0) >= 1000) then begin
-            set_global_var(37, 1);
+        set_global_var(GVAR_KARMA_HOLY_WARRIOR, 0);
+        set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 0);
+        set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 0);
+        set_global_var(GVAR_KARMA_DEFENDER, 0);
+        set_global_var(GVAR_KARMA_WANDERER, 0);
+        set_global_var(GVAR_KARMA_BETRAYER, 0);
+        set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 0);
+        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 0);
+        set_global_var(GVAR_KARMA_DEMON_SPAWN, 0);
+        if (global_var(GVAR_PLAYER_REPUTATION) >= 1000) then begin
+            set_global_var(GVAR_KARMA_HOLY_WARRIOR, 1);
         end
         else begin
-            if (global_var(0) >= 750) then begin
-                set_global_var(38, 1);
+            if (global_var(GVAR_PLAYER_REPUTATION) >= 750) then begin
+                set_global_var(GVAR_KARMA_GUARDIAN_OF_THE_WASTES, 1);
             end
             else begin
-                if (global_var(0) >= 500) then begin
-                    set_global_var(39, 1);
+                if (global_var(GVAR_PLAYER_REPUTATION) >= 500) then begin
+                    set_global_var(GVAR_KARMA_SHIELD_OF_HOPE, 1);
                 end
                 else begin
-                    if (global_var(0) >= 250) then begin
-                        set_global_var(40, 1);
+                    if (global_var(GVAR_PLAYER_REPUTATION) >= 250) then begin
+                        set_global_var(GVAR_KARMA_DEFENDER, 1);
                     end
                     else begin
-                        if (global_var(0) > -250) then begin
-                            set_global_var(41, 1);
+                        if (global_var(GVAR_PLAYER_REPUTATION) > -250) then begin
+                            set_global_var(GVAR_KARMA_WANDERER, 1);
                         end
                         else begin
-                            if (global_var(0) > -500) then begin
-                                set_global_var(42, 1);
+                            if (global_var(GVAR_PLAYER_REPUTATION) > -500) then begin
+                                set_global_var(GVAR_KARMA_BETRAYER, 1);
                             end
                             else begin
-                                if (global_var(0) > -750) then begin
-                                    set_global_var(43, 1);
+                                if (global_var(GVAR_PLAYER_REPUTATION) > -750) then begin
+                                    set_global_var(GVAR_KARMA_SWORD_OF_DESPAIR, 1);
                                 end
                                 else begin
-                                    if (global_var(0) > -1000) then begin
-                                        set_global_var(44, 1);
+                                    if (global_var(GVAR_PLAYER_REPUTATION) > -1000) then begin
+                                        set_global_var(GVAR_KARMA_SCOURGE_OF_THE_WASTE, 1);
                                     end
                                     else begin
-                                        set_global_var(45, 1);
+                                        set_global_var(GVAR_KARMA_DEMON_SPAWN, 1);
                                     end
                                 end
                             end
@@ -674,7 +674,7 @@ begin
                 end
             end
         end
-        set_global_var(50, global_var(50) + -5);
+        set_global_var(GVAR_TOWN_REP_VAULT_CITY, global_var(GVAR_TOWN_REP_VAULT_CITY) + -5);
         debug_msg("Added " + -5 + " to Town Rep");
     end
 end
@@ -683,17 +683,17 @@ procedure look_at_p_proc
 begin
     script_overrides;
     if (local_var(4) == 0) then begin
-        display_msg(message_str(1128, 101));
+        display_mstr(101);
     end
     else begin
-        display_msg(message_str(1128, 100));
+        display_mstr(100);
     end
 end
 
 procedure description_p_proc
 begin
     script_overrides;
-    display_msg(message_str(1128, 101));
+    display_mstr(101);
 end
 
 procedure use_skill_on_p_proc
@@ -704,27 +704,27 @@ procedure damage_p_proc
 begin
     if (source_obj == dude_obj) then begin
         set_local_var(6, 1);
-        set_global_var(137, 1);
-        set_global_var(79, 6);
-        set_global_var(50, global_var(50) + -30);
-        set_global_var(81, 0);
+        set_global_var(GVAR_ENEMY_VAULT_CITY, 1);
+        set_global_var(GVAR_VAULT_CITIZEN, 6);
+        set_global_var(GVAR_TOWN_REP_VAULT_CITY, global_var(GVAR_TOWN_REP_VAULT_CITY) + -30);
+        set_global_var(GVAR_VAULT_CITIZENSHIP, 0);
     end
 end
 
 procedure map_enter_p_proc
 begin
     Only_Once := 0;
-    critter_add_trait(self_obj, 1, 6, 26);
-    critter_add_trait(self_obj, 1, 5, 33);
-    if (not(metarule(22, 0))) then begin
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 26);
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_COWARD);
+    if (not(is_loading_game)) then begin
         if ((game_time_hour > 700) and (game_time_hour < 1800)) then begin
-            if (not(has_trait(1, self_obj, 666))) then begin
+            if (not(obj_is_visible_flag(self_obj))) then begin
                 set_obj_visibility(self_obj, 0);
                 add_timer_event(self_obj, game_ticks(random(20, 40)), 1);
             end
         end
         else begin
-            if (has_trait(1, self_obj, 666)) then begin
+            if (obj_is_visible_flag(self_obj)) then begin
                 set_obj_visibility(self_obj, 1);
             end
         end
@@ -735,19 +735,19 @@ procedure timed_event_p_proc
 begin
     if (fixed_param == 1) then begin
         if (obj_can_see_obj(self_obj, dude_obj) and not(combat_is_initialized)) then begin
-            if (global_var(494) == 8) then begin
+            if (global_var(GVAR_FALLOUT_2) == 8) then begin
                 call Node001();
             end
             else begin
-                if (global_var(79) == 6) then begin
+                if (global_var(GVAR_VAULT_CITIZEN) == 6) then begin
                     call Node002();
                 end
                 else begin
-                    if ((party_member_obj(16777377) != 0) and (global_var(79) != 5)) then begin
+                    if ((party_member_obj(16777377) != 0) and (global_var(GVAR_VAULT_CITIZEN) != 5)) then begin
                         call Node003();
                     end
                     else begin
-                        if ((party_member_obj(16777323) != 0) and (global_var(79) != 5)) then begin
+                        if ((party_member_obj(16777323) != 0) and (global_var(GVAR_VAULT_CITIZEN) != 5)) then begin
                             call Node004();
                         end
                         else begin
@@ -774,208 +774,208 @@ end
 
 procedure Node001
 begin
-    float_msg(self_obj, message_str(1128, message_str(1128, 104) + obj_name(dude_obj) + message_str(1128, 193)), 8);
+    float_msg(self_obj, mstr(mstr(104) + dude_name + mstr(193)), FLOAT_MSG_YELLOW);
 end
 
 procedure Node002
 begin
-    float_msg(self_obj, message_str(1128, 105), 8);
+    float_msg(self_obj, mstr(105), FLOAT_MSG_YELLOW);
     call Node998();
 end
 
 procedure Node003
 begin
-    float_msg(self_obj, message_str(1128, 106), 8);
+    float_msg(self_obj, mstr(106), FLOAT_MSG_YELLOW);
     call Node998();
 end
 
 procedure Node004
 begin
-    float_msg(self_obj, message_str(1128, 107), 8);
+    float_msg(self_obj, mstr(107), FLOAT_MSG_YELLOW);
     call Node998();
 end
 
 procedure Node005
 begin
-    float_msg(self_obj, message_str(1128, random(108, 109)), 8);
+    float_msg(self_obj, mstr(random(108, 109)), FLOAT_MSG_YELLOW);
 end
 
 procedure Node006
 begin
-    float_msg(self_obj, message_str(1128, random(110, 113)), 8);
+    float_msg(self_obj, mstr(random(110, 113)), FLOAT_MSG_YELLOW);
 end
 
 procedure Node007
 begin
     set_local_var(9, 1);
     From_WTG := 0;
-    gsay_reply(1128, 114);
-    giq_option(-3, 1128, 115, Node008, 51);
-    giq_option(-3, 1128, 116, Node999, 50);
-    giq_option(4, 1128, 117, Node010, 50);
-    giq_option(4, 1128, 118, Node999, 50);
+    Reply(114);
+    giq_option(-3, NAME, 115, Node008, BAD_REACTION);
+    giq_option(-3, NAME, 116, Node999, NEUTRAL_REACTION);
+    giq_option(4, NAME, 117, Node010, NEUTRAL_REACTION);
+    giq_option(4, NAME, 118, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node008
 begin
-    gsay_reply(1128, 119);
-    giq_option(-3, 1128, 120, Node009, 50);
-    giq_option(-3, 1128, 121, Node999, 50);
+    Reply(119);
+    giq_option(-3, NAME, 120, Node009, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 121, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node009
 begin
-    gsay_reply(1128, 122);
-    giq_option(-3, 1128, 123, Node999, 50);
+    Reply(122);
+    giq_option(-3, NAME, 123, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node010
 begin
-    gsay_reply(1128, 124);
-    giq_option(4, 1128, 125, Node012, 50);
-    giq_option(4, 1128, 126, Node011, 50);
-    giq_option(4, 1128, 127, Node999, 50);
+    Reply(124);
+    giq_option(4, NAME, 125, Node012, NEUTRAL_REACTION);
+    giq_option(4, NAME, 126, Node011, NEUTRAL_REACTION);
+    giq_option(4, NAME, 127, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node011
 begin
-    gsay_reply(1128, 128);
-    giq_option(7, 1128, 129, Node025, 50);
-    giq_option(4, 1128, 130, Node018, 50);
-    giq_option(4, 1128, 131, Node012, 50);
-    giq_option(4, 1128, 132, Node999, 50);
+    Reply(128);
+    giq_option(7, NAME, 129, Node025, NEUTRAL_REACTION);
+    giq_option(4, NAME, 130, Node018, NEUTRAL_REACTION);
+    giq_option(4, NAME, 131, Node012, NEUTRAL_REACTION);
+    giq_option(4, NAME, 132, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node012
 begin
     if (From_WTG == 1) then begin
         From_WTG := 0;
-        gsay_reply(1128, 133);
+        Reply(133);
     end
     else begin
-        gsay_reply(1128, 134);
+        Reply(134);
     end
-    giq_option(-3, 1128, 115, Node008, 51);
-    giq_option(-3, 1128, 116, Node999, 50);
-    giq_option(4, 1128, 135, Node010, 50);
-    if (not(global_var(233))) then begin
-        giq_option(4, 1128, 136, Node013, 50);
+    giq_option(-3, NAME, 115, Node008, BAD_REACTION);
+    giq_option(-3, NAME, 116, Node999, NEUTRAL_REACTION);
+    giq_option(4, NAME, 135, Node010, NEUTRAL_REACTION);
+    if (not(global_var(GVAR_VAULT13_FOUND_GECK))) then begin
+        giq_option(4, NAME, 136, Node013, NEUTRAL_REACTION);
     end
-    giq_option(4, 1128, 137, Node014, 50);
-    giq_option(4, 1128, 138, Node999, 50);
+    giq_option(4, NAME, 137, Node014, NEUTRAL_REACTION);
+    giq_option(4, NAME, 138, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node013
 begin
-    gsay_reply(1128, 139);
-    giq_option(7, 1128, 140, Node016, 50);
-    giq_option(4, 1128, 141, Node016, 50);
-    giq_option(4, 1128, 142, Node012, 50);
-    giq_option(4, 1128, 143, Node999, 50);
+    Reply(139);
+    giq_option(7, NAME, 140, Node016, NEUTRAL_REACTION);
+    giq_option(4, NAME, 141, Node016, NEUTRAL_REACTION);
+    giq_option(4, NAME, 142, Node012, NEUTRAL_REACTION);
+    giq_option(4, NAME, 143, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node014
 begin
-    gsay_reply(1128, 144);
-    giq_option(4, 1128, 145, Node015, 50);
-    giq_option(4, 1128, 146, Node012, 50);
-    giq_option(4, 1128, 147, Node999, 50);
+    Reply(144);
+    giq_option(4, NAME, 145, Node015, NEUTRAL_REACTION);
+    giq_option(4, NAME, 146, Node012, NEUTRAL_REACTION);
+    giq_option(4, NAME, 147, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node015
 begin
-    gsay_reply(1128, 148);
-    giq_option(4, 1128, 149, Node017, 50);
-    giq_option(4, 1128, 150, Node012, 50);
-    giq_option(4, 1128, 151, Node999, 50);
+    Reply(148);
+    giq_option(4, NAME, 149, Node017, NEUTRAL_REACTION);
+    giq_option(4, NAME, 150, Node012, NEUTRAL_REACTION);
+    giq_option(4, NAME, 151, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node016
 begin
-    gsay_reply(1128, 152);
-    giq_option(4, 1128, 153, Node017, 50);
-    giq_option(4, 1128, 154, Node012, 50);
-    giq_option(4, 1128, 155, Node999, 50);
+    Reply(152);
+    giq_option(4, NAME, 153, Node017, NEUTRAL_REACTION);
+    giq_option(4, NAME, 154, Node012, NEUTRAL_REACTION);
+    giq_option(4, NAME, 155, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node017
 begin
-    if ((global_var(79) == 5) or (global_var(79) == 4) or (local_var(7) == 1)) then begin
-        gsay_reply(1128, 156);
+    if ((global_var(GVAR_VAULT_CITIZEN) == 5) or (global_var(GVAR_VAULT_CITIZEN) == 4) or (local_var(7) == 1)) then begin
+        Reply(156);
     end
     else begin
         if (From_17 == 1) then begin
             From_17 := 0;
             set_local_var(7, 1);
-            gsay_reply(1128, 157);
+            Reply(157);
         end
         else begin
-            gsay_reply(1128, 158);
+            Reply(158);
         end
     end
     From_17 := 1;
-    if ((global_var(79) != 5) and (global_var(79) != 4) and (local_var(7) != 1)) then begin
-        giq_option(4, 1128, 159, Node017, 50);
+    if ((global_var(GVAR_VAULT_CITIZEN) != 5) and (global_var(GVAR_VAULT_CITIZEN) != 4) and (local_var(7) != 1)) then begin
+        giq_option(4, NAME, 159, Node017, NEUTRAL_REACTION);
     end
-    giq_option(4, 1128, 160, Node012, 50);
-    giq_option(4, 1128, 161, Node999, 50);
+    giq_option(4, NAME, 160, Node012, NEUTRAL_REACTION);
+    giq_option(4, NAME, 161, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node018
 begin
-    gsay_reply(1128, 162);
-    if (get_critter_stat(dude_obj, 4) == 4) then begin
-        giq_option(4, 1128, 163, Node019, 50);
+    Reply(162);
+    if (dude_iq == 4) then begin
+        giq_option(4, NAME, 163, Node019, NEUTRAL_REACTION);
     end
-    giq_option(5, 1128, 164, Node019, 50);
-    giq_option(4, 1128, 165, Node012, 50);
-    giq_option(4, 1128, 166, Node999, 50);
+    giq_option(5, NAME, 164, Node019, NEUTRAL_REACTION);
+    giq_option(4, NAME, 165, Node012, NEUTRAL_REACTION);
+    giq_option(4, NAME, 166, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node019
 begin
-    if (get_critter_stat(dude_obj, 4) == 4) then begin
-        gsay_reply(1128, 167);
+    if (dude_iq == 4) then begin
+        Reply(167);
     end
     else begin
-        gsay_reply(1128, 168);
+        Reply(168);
     end
-    giq_option(4, 1128, 169, Node020, 50);
-    giq_option(4, 1128, 170, Node012, 50);
-    giq_option(4, 1128, 171, Node999, 50);
+    giq_option(4, NAME, 169, Node020, NEUTRAL_REACTION);
+    giq_option(4, NAME, 170, Node012, NEUTRAL_REACTION);
+    giq_option(4, NAME, 171, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node020
 begin
-    gsay_reply(1128, 172);
+    Reply(172);
     if (local_var(8) == 0) then begin
-        giq_option(7, 1128, 173, Node021, 49);
+        giq_option(7, NAME, 173, Node021, GOOD_REACTION);
     end
-    if ((get_critter_stat(dude_obj, 4) < 7) and (local_var(8) == 0) and has_trait(0, dude_obj, 81)) then begin
-        giq_option(4, 1128, 174, Node021, 49);
+    if ((dude_iq < 7) and (local_var(8) == 0) and has_trait(TRAIT_PERK, dude_obj, PERK_comprehension_perk)) then begin
+        giq_option(4, NAME, 174, Node021, GOOD_REACTION);
     end
-    giq_option(4, 1128, 175, Node012, 50);
-    giq_option(4, 1128, 176, Node999, 50);
+    giq_option(4, NAME, 175, Node012, NEUTRAL_REACTION);
+    giq_option(4, NAME, 176, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node021
 begin
-    gsay_reply(1128, 177);
-    giq_option(4, 1128, 178, Node022, 50);
-    giq_option(4, 1128, 179, Node012, 50);
-    giq_option(4, 1128, 180, Node999, 50);
+    Reply(177);
+    giq_option(4, NAME, 178, Node022, NEUTRAL_REACTION);
+    giq_option(4, NAME, 179, Node012, NEUTRAL_REACTION);
+    giq_option(4, NAME, 180, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node022
 begin
-    gsay_reply(1128, 181);
-    giq_option(4, 1128, 182, Node023, 50);
+    Reply(181);
+    giq_option(4, NAME, 182, Node023, NEUTRAL_REACTION);
 end
 
 procedure Node023
 begin
-    gsay_reply(1128, 183);
-    giq_option(4, 1128, 184, Node024, 50);
+    Reply(183);
+    giq_option(4, NAME, 184, Node024, NEUTRAL_REACTION);
 end
 
 procedure Node024
@@ -983,24 +983,23 @@ begin
     variable LVar0 := 0;
     variable LVar1 := 0;
     gfade_out(600);
-    set_global_var(50, global_var(50) + 1);
+    set_global_var(GVAR_TOWN_REP_VAULT_CITY, global_var(GVAR_TOWN_REP_VAULT_CITY) + 1);
     set_local_var(8, 1);
-    LVar0 := create_object_sid(73, 0, 0, -1);
+    LVar0 := create_object(73, 0, 0);
     add_obj_to_inven(dude_obj, LVar0);
-    LVar1 := create_object_sid(76, 0, 0, -1);
+    LVar1 := create_object(76, 0, 0);
     add_obj_to_inven(dude_obj, LVar1);
     gfade_in(600);
-    gsay_reply(1128, 185);
-    giq_option(4, 1128, 186, Node012, 50);
-    giq_option(4, 1128, 187, Node999, 50);
+    Reply(185);
+    giq_option(4, NAME, 186, Node012, NEUTRAL_REACTION);
+    giq_option(4, NAME, 187, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node025
 begin
-    gsay_reply(1128, 188);
-    giq_option(7, 1128, 189, Node018, 50);
-    giq_option(4, 1128, 190, Node017, 50);
-    giq_option(4, 1128, 191, Node012, 50);
-    giq_option(4, 1128, 192, Node999, 50);
+    Reply(188);
+    giq_option(7, NAME, 189, Node018, NEUTRAL_REACTION);
+    giq_option(4, NAME, 190, Node017, NEUTRAL_REACTION);
+    giq_option(4, NAME, 191, Node012, NEUTRAL_REACTION);
+    giq_option(4, NAME, 192, Node999, NEUTRAL_REACTION);
 end
-
