diff --git a/Games/Fallout2.Steam/Original/VIALOCOM.ssl b/Games/Fallout2.Steam/ReDefine/VIALOCOM.ssl
index b808fee..079f16a 100644
--- a/Games/Fallout2.Steam/Original/VIALOCOM.ssl
+++ b/Games/Fallout2.Steam/ReDefine/VIALOCOM.ssl
@@ -140,17 +140,17 @@ end
 procedure use_p_proc
 begin
     script_overrides;
-    if (global_var(401) == 3) then begin
-        display_msg(message_str(839, 108));
+    if (global_var(GVAR_VAULT_PIP) == 3) then begin
+        display_mstr(108);
         gfade_out(600);
         gfade_in(600);
-        float_msg(dude_obj, message_str(839, 109), 8);
-        display_msg(message_str(839, 110));
+        float_msg(dude_obj, mstr(109), FLOAT_MSG_YELLOW);
+        display_mstr(110);
         give_exp_points(20000);
-        display_msg(message_str(14, 100) + 20000 + message_str(14, 101));
+        display_msg(g_mstr(100) + 20000 + g_mstr(101));
     end
     else begin
-        display_msg(message_str(839, 107));
+        display_mstr(107);
     end
 end
 
@@ -166,30 +166,29 @@ begin
     script_overrides;
     if (local_var(0) == 0) then begin
         set_local_var(0, local_var(0) + 1);
-        if (global_var(401) <= 2) then begin
-            display_msg(message_str(839, 100));
+        if (global_var(GVAR_VAULT_PIP) <= 2) then begin
+            display_mstr(100);
         end
         else begin
-            display_msg(message_str(839, 103));
+            display_mstr(103);
         end
     end
     else begin
-        if (global_var(401) <= 2) then begin
-            display_msg(message_str(839, 101));
+        if (global_var(GVAR_VAULT_PIP) <= 2) then begin
+            display_mstr(101);
         end
         else begin
-            display_msg(message_str(839, 104));
+            display_mstr(104);
         end
     end
 end
 
 procedure description_p_proc
 begin
-    if (global_var(401) <= 2) then begin
-        display_msg(message_str(839, 102));
+    if (global_var(GVAR_VAULT_PIP) <= 2) then begin
+        display_mstr(102);
     end
     else begin
-        display_msg(message_str(839, 105));
+        display_mstr(105);
     end
 end
-
