diff --git a/Games/Fallout2.Steam/Original/VICENCOM.ssl b/Games/Fallout2.Steam/ReDefine/VICENCOM.ssl
index 31a16a2..1399e4c 100644
--- a/Games/Fallout2.Steam/Original/VICENCOM.ssl
+++ b/Games/Fallout2.Steam/ReDefine/VICENCOM.ssl
@@ -184,7 +184,7 @@ end
 
 procedure talk_p_proc
 begin
-    if (global_var(401) > 2) then begin
+    if (global_var(GVAR_VAULT_PIP) > 2) then begin
         call Node030();
     end
     else begin
@@ -199,26 +199,26 @@ end
 procedure look_at_p_proc
 begin
     script_overrides;
-    if (global_var(401) > 2) then begin
+    if (global_var(GVAR_VAULT_PIP) > 2) then begin
         display_msg(106);
     end
     else begin
-        if (get_critter_stat(dude_obj, 4) > 3) then begin
+        if (dude_iq > 3) then begin
             if (local_var(0) == 0) then begin
                 set_local_var(0, local_var(0) + 1);
-                display_msg(message_str(838, 100));
+                display_mstr(100);
             end
             else begin
-                display_msg(message_str(838, 101));
+                display_mstr(101);
             end
         end
         else begin
             if (local_var(0) == 0) then begin
                 set_local_var(0, local_var(0) + 1);
-                display_msg(message_str(838, 103));
+                display_mstr(103);
             end
             else begin
-                display_msg(message_str(838, 104));
+                display_mstr(104);
             end
         end
     end
@@ -227,11 +227,11 @@ end
 procedure description_p_proc
 begin
     script_overrides;
-    if (get_critter_stat(dude_obj, 4) > 3) then begin
-        display_msg(message_str(838, 102));
+    if (dude_iq > 3) then begin
+        display_mstr(102);
     end
     else begin
-        display_msg(message_str(838, 103));
+        display_mstr(103);
     end
 end
 
@@ -251,12 +251,12 @@ end
 procedure timed_event_p_proc
 begin
     if (fixed_param == 1) then begin
-        if ((global_var(400) != 2) and (global_var(401) < 3)) then begin
+        if ((global_var(GVAR_VAULT_CITY_VENT) != 2) and (global_var(GVAR_VAULT_PIP) < 3)) then begin
             call Node004();
             add_timer_event(self_obj, game_ticks(4), 1);
         end
         else begin
-            if (global_var(401) < 3) then begin
+            if (global_var(GVAR_VAULT_PIP) < 3) then begin
                 call Node005();
                 add_timer_event(self_obj, game_ticks(4), 2);
             end
@@ -285,10 +285,10 @@ begin
         set_local_var(1, 0);
     end
     LVar0 := 107 + local_var(1);
-    if ((global_var(400) != 2) and (LVar0 == 122)) then begin
-        set_global_var(400, 1);
+    if ((global_var(GVAR_VAULT_CITY_VENT) != 2) and (LVar0 == 122)) then begin
+        set_global_var(GVAR_VAULT_CITY_VENT, 1);
     end
-    float_msg(self_obj, message_str(838, LVar0), 8);
+    float_msg(self_obj, mstr(LVar0), FLOAT_MSG_YELLOW);
     set_local_var(1, local_var(1) + 1);
 end
 
@@ -299,56 +299,56 @@ begin
         set_local_var(2, 0);
     end
     LVar0 := 140 + local_var(2);
-    float_msg(self_obj, message_str(838, LVar0), 8);
+    float_msg(self_obj, mstr(LVar0), FLOAT_MSG_YELLOW);
     set_local_var(2, local_var(2) + 1);
 end
 
 procedure Node006
 begin
-    if (get_critter_stat(dude_obj, 4) < 4) then begin
-        gsay_reply(838, 166);
+    if (dude_iq < 4) then begin
+        Reply(166);
     end
     else begin
-        gsay_reply(838, 167);
+        Reply(167);
     end
-    giq_option(-3, 838, 168, Node007, 50);
-    giq_option(-3, 838, 169, Node008, 50);
-    giq_option(-3, 838, 170, Node999, 50);
-    giq_option(4, 838, 171, Node023, 50);
-    giq_option(4, 838, 172, Node024, 50);
-    if ((global_var(82) == 12) and obj_is_carrying_obj_pid(dude_obj, 303)) then begin
-        giq_option(4, 838, 173, Node031, 50);
+    giq_option(-3, NAME, 168, Node007, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 169, Node008, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 170, Node999, NEUTRAL_REACTION);
+    giq_option(4, NAME, 171, Node023, NEUTRAL_REACTION);
+    giq_option(4, NAME, 172, Node024, NEUTRAL_REACTION);
+    if ((global_var(GVAR_VAULT_GECKO_PLANT) == 12) and obj_is_carrying_obj_pid(dude_obj, 303)) then begin
+        giq_option(4, NAME, 173, Node031, NEUTRAL_REACTION);
     end
-    if (global_var(350) == 2) then begin
-        giq_option(4, 838, 174, Node010, 50);
+    if (global_var(GVAR_GRUTHAR_DSTATUS) == 2) then begin
+        giq_option(4, NAME, 174, Node010, NEUTRAL_REACTION);
     end
-    if (global_var(401) == 1) then begin
-        giq_option(4, 838, 175, Node014, 50);
+    if (global_var(GVAR_VAULT_PIP) == 1) then begin
+        giq_option(4, NAME, 175, Node014, NEUTRAL_REACTION);
     end
-    if ((global_var(401) >= 2) and (global_var(494) == 8)) then begin
-        giq_option(4, 838, 176, Node027, 50);
+    if ((global_var(GVAR_VAULT_PIP) >= 2) and (global_var(GVAR_FALLOUT_2) == 8)) then begin
+        giq_option(4, NAME, 176, Node027, NEUTRAL_REACTION);
     end
-    giq_option(4, 838, 177, Node006a, 50);
+    giq_option(4, NAME, 177, Node006a, NEUTRAL_REACTION);
 end
 
 procedure Node007
 begin
-    gsay_reply(838, 178);
-    giq_option(-3, 838, 179, Node008, 50);
-    giq_option(-3, 838, 180, Node999, 50);
+    Reply(178);
+    giq_option(-3, NAME, 179, Node008, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 180, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node008
 begin
-    gsay_reply(838, 181);
-    giq_option(-3, 838, 182, Node009, 50);
-    giq_option(-3, 838, 183, Node006b, 50);
+    Reply(181);
+    giq_option(-3, NAME, 182, Node009, NEUTRAL_REACTION);
+    giq_option(-3, NAME, 183, Node006b, NEUTRAL_REACTION);
 end
 
 procedure Node009
 begin
-    gsay_reply(838, 184);
-    giq_option(-3, 838, 272, Node009a, 50);
+    Reply(184);
+    giq_option(-3, NAME, 272, Node009a, NEUTRAL_REACTION);
 end
 
 procedure Node009a
@@ -358,269 +358,269 @@ begin
     gfade_in(600);
     if (local_var(4) == 0) then begin
         set_local_var(4, 1);
-        LVar0 := get_critter_stat(dude_obj, 28);
-        set_critter_stat(dude_obj, 28, LVar0 + 5);
-        critter_dmg(dude_obj, get_critter_stat(dude_obj, 35) - 1, 4 bwor 512);
+        LVar0 := get_critter_stat(dude_obj, STAT_dmg_resist_electrical);
+        set_critter_stat(dude_obj, STAT_dmg_resist_electrical, LVar0 + 5);
+        critter_dmg(dude_obj, get_critter_stat(dude_obj, STAT_current_hp) - 1, 4 bwor 512);
     end
-    gsay_reply(838, 185);
-    giq_option(-3, 838, 186, Node006b, 50);
+    Reply(185);
+    giq_option(-3, NAME, 186, Node006b, NEUTRAL_REACTION);
 end
 
 procedure Node010
 begin
-    gsay_reply(838, 187);
-    giq_option(4, 838, 188, Node010a, 50);
-    giq_option(4, 838, 189, Node006b, 50);
+    Reply(187);
+    giq_option(4, NAME, 188, Node010a, NEUTRAL_REACTION);
+    giq_option(4, NAME, 189, Node006b, NEUTRAL_REACTION);
 end
 
 procedure Node011
 begin
-    set_global_var(401, 1);
-    gsay_reply(838, 190);
-    giq_option(4, 838, 191, Node014, 50);
-    giq_option(4, 838, 192, Node999, 50);
+    set_global_var(GVAR_VAULT_PIP, 1);
+    Reply(190);
+    giq_option(4, NAME, 191, Node014, NEUTRAL_REACTION);
+    giq_option(4, NAME, 192, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node012
 begin
-    set_global_var(401, 1);
-    if (get_critter_stat(dude_obj, 4) < 4) then begin
-        gsay_reply(838, 193);
+    set_global_var(GVAR_VAULT_PIP, 1);
+    if (dude_iq < 4) then begin
+        Reply(193);
     end
     else begin
-        gsay_reply(838, 194);
+        Reply(194);
     end
-    giq_option(-3, 838, 195, Node013, 50);
-    giq_option(4, 838, 196, Node014, 50);
-    giq_option(4, 838, 197, Node999, 50);
+    giq_option(-3, NAME, 195, Node013, NEUTRAL_REACTION);
+    giq_option(4, NAME, 196, Node014, NEUTRAL_REACTION);
+    giq_option(4, NAME, 197, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node013
 begin
-    set_global_var(404, 1);
-    set_global_var(401, 2);
-    if (metarule(17, 5) == 0) then begin
+    set_global_var(GVAR_PIPBOY_TOUR_GUIDE, 1);
+    set_global_var(GVAR_VAULT_PIP, 2);
+    if (town_known(5) == 0) then begin
         debug_msg("  mark_on_map(" + 5 + ")");
         mark_area_known(0, 5, 1);
     end
-    if (metarule(17, 10) == 0) then begin
+    if (town_known(10) == 0) then begin
         debug_msg("  mark_on_map(" + 10 + ")");
         mark_area_known(0, 10, 1);
     end
-    if (metarule(17, 7) == 0) then begin
+    if (town_known(7) == 0) then begin
         debug_msg("  mark_on_map(" + 7 + ")");
         mark_area_known(0, 7, 1);
     end
-    if (metarule(17, 6) == 0) then begin
+    if (town_known(6) == 0) then begin
         debug_msg("  mark_on_map(" + 6 + ")");
         mark_area_known(0, 6, 1);
     end
-    if (metarule(17, 13) == 0) then begin
+    if (town_known(13) == 0) then begin
         debug_msg("  mark_on_map(" + 13 + ")");
         mark_area_known(0, 13, 1);
     end
-    if (metarule(17, 9) == 0) then begin
+    if (town_known(9) == 0) then begin
         debug_msg("  mark_on_map(" + 9 + ")");
         mark_area_known(0, 9, 1);
     end
-    gsay_reply(838, 198);
-    giq_option(-3, 838, 199, Node999, 50);
+    Reply(198);
+    giq_option(-3, NAME, 199, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node014
 begin
-    gsay_reply(838, 200);
-    giq_option(4, 838, 201, Node015, 50);
-    giq_option(4, 838, 202, Node999, 50);
+    Reply(200);
+    giq_option(4, NAME, 201, Node015, NEUTRAL_REACTION);
+    giq_option(4, NAME, 202, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node015
 begin
-    gsay_reply(838, 203);
-    giq_option(4, 838, 204, Node016, 50);
-    giq_option(4, 838, 205, Node999, 50);
+    Reply(203);
+    giq_option(4, NAME, 204, Node016, NEUTRAL_REACTION);
+    giq_option(4, NAME, 205, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node016
 begin
-    gsay_reply(838, 206);
-    giq_option(4, 838, 207, Node017, 50);
-    giq_option(4, 838, 208, Node017, 50);
-    giq_option(4, 838, 209, Node999, 50);
+    Reply(206);
+    giq_option(4, NAME, 207, Node017, NEUTRAL_REACTION);
+    giq_option(4, NAME, 208, Node017, NEUTRAL_REACTION);
+    giq_option(4, NAME, 209, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node017
 begin
-    gsay_reply(838, 210);
-    giq_option(4, 838, 211, Node018, 50);
-    giq_option(4, 838, 212, Node019, 50);
-    giq_option(4, 838, 213, Node999, 50);
+    Reply(210);
+    giq_option(4, NAME, 211, Node018, NEUTRAL_REACTION);
+    giq_option(4, NAME, 212, Node019, NEUTRAL_REACTION);
+    giq_option(4, NAME, 213, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node018
 begin
-    gsay_reply(838, 214);
-    giq_option(4, 838, 216, Node020, 50);
-    giq_option(4, 838, 217, Node999, 50);
+    Reply(214);
+    giq_option(4, NAME, 216, Node020, NEUTRAL_REACTION);
+    giq_option(4, NAME, 217, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node019
 begin
-    gsay_reply(838, message_str(838, 218) + message_str(838, 219));
-    giq_option(4, 838, 220, Node020, 50);
-    giq_option(4, 838, 221, Node020, 50);
+    Reply(mstr(218) + mstr(219));
+    giq_option(4, NAME, 220, Node020, NEUTRAL_REACTION);
+    giq_option(4, NAME, 221, Node020, NEUTRAL_REACTION);
 end
 
 procedure Node020
 begin
-    set_global_var(404, 1);
-    set_global_var(401, 2);
-    if (metarule(17, 5) == 0) then begin
+    set_global_var(GVAR_PIPBOY_TOUR_GUIDE, 1);
+    set_global_var(GVAR_VAULT_PIP, 2);
+    if (town_known(5) == 0) then begin
         debug_msg("  mark_on_map(" + 5 + ")");
         mark_area_known(0, 5, 1);
     end
-    if (metarule(17, 10) == 0) then begin
+    if (town_known(10) == 0) then begin
         debug_msg("  mark_on_map(" + 10 + ")");
         mark_area_known(0, 10, 1);
     end
-    if (metarule(17, 7) == 0) then begin
+    if (town_known(7) == 0) then begin
         debug_msg("  mark_on_map(" + 7 + ")");
         mark_area_known(0, 7, 1);
     end
-    if (metarule(17, 6) == 0) then begin
+    if (town_known(6) == 0) then begin
         debug_msg("  mark_on_map(" + 6 + ")");
         mark_area_known(0, 6, 1);
     end
-    if (metarule(17, 13) == 0) then begin
+    if (town_known(13) == 0) then begin
         debug_msg("  mark_on_map(" + 13 + ")");
         mark_area_known(0, 13, 1);
     end
-    if (metarule(17, 9) == 0) then begin
+    if (town_known(9) == 0) then begin
         debug_msg("  mark_on_map(" + 9 + ")");
         mark_area_known(0, 9, 1);
     end
-    gsay_reply(838, message_str(838, 222) + message_str(838, 223));
-    giq_option(4, 838, 224, Node021, 50);
-    giq_option(4, 838, 225, Node999, 50);
+    Reply(mstr(222) + mstr(223));
+    giq_option(4, NAME, 224, Node021, NEUTRAL_REACTION);
+    giq_option(4, NAME, 225, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node021
 begin
-    gsay_reply(838, 226);
-    giq_option(4, 838, 227, Node023, 50);
-    giq_option(4, 838, 228, Node024, 50);
-    if (global_var(82) == 9) then begin
-        giq_option(4, 838, 229, Node022, 50);
+    Reply(226);
+    giq_option(4, NAME, 227, Node023, NEUTRAL_REACTION);
+    giq_option(4, NAME, 228, Node024, NEUTRAL_REACTION);
+    if (global_var(GVAR_VAULT_GECKO_PLANT) == 9) then begin
+        giq_option(4, NAME, 229, Node022, NEUTRAL_REACTION);
     end
-    giq_option(4, 838, 230, Node006a, 50);
+    giq_option(4, NAME, 230, Node006a, NEUTRAL_REACTION);
 end
 
 procedure Node022
 begin
-    gsay_reply(838, 231);
-    giq_option(4, 838, 232, Node021, 50);
-    giq_option(4, 838, 233, Node006a, 50);
+    Reply(231);
+    giq_option(4, NAME, 232, Node021, NEUTRAL_REACTION);
+    giq_option(4, NAME, 233, Node006a, NEUTRAL_REACTION);
 end
 
 procedure Node023
 begin
-    if (metarule(17, 9) == 0) then begin
+    if (town_known(9) == 0) then begin
         debug_msg("  mark_on_map(" + 9 + ")");
         mark_area_known(0, 9, 1);
     end
     if (local_var(3) == 0) then begin
         set_local_var(3, local_var(3) + 1);
-        gsay_reply(838, message_str(838, 234) + " " + message_str(838, 235));
+        Reply(mstr(234) + " " + mstr(235));
     end
     else begin
-        gsay_reply(838, 234);
+        Reply(234);
     end
-    giq_option(4, 838, 236, Node021, 50);
-    giq_option(4, 838, 237, Node006a, 50);
+    giq_option(4, NAME, 236, Node021, NEUTRAL_REACTION);
+    giq_option(4, NAME, 237, Node006a, NEUTRAL_REACTION);
 end
 
 procedure Node024
 begin
-    gsay_reply(838, 238);
-    giq_option(4, 838, 239, Node025, 50);
+    Reply(238);
+    giq_option(4, NAME, 239, Node025, NEUTRAL_REACTION);
 end
 
 procedure Node025
 begin
-    gsay_reply(838, 240);
-    giq_option(4, 838, 241, Node026, 50);
-    giq_option(4, 838, 242, Node021, 50);
-    giq_option(4, 838, 243, Node006a, 50);
+    Reply(240);
+    giq_option(4, NAME, 241, Node026, NEUTRAL_REACTION);
+    giq_option(4, NAME, 242, Node021, NEUTRAL_REACTION);
+    giq_option(4, NAME, 243, Node006a, NEUTRAL_REACTION);
 end
 
 procedure Node026
 begin
-    gsay_reply(838, 244);
-    giq_option(4, 838, 245, Node021, 50);
-    giq_option(4, 838, 246, Node006a, 50);
+    Reply(244);
+    giq_option(4, NAME, 245, Node021, NEUTRAL_REACTION);
+    giq_option(4, NAME, 246, Node006a, NEUTRAL_REACTION);
 end
 
 procedure Node027
 begin
-    gsay_reply(838, 247);
-    if (global_var(0) >= 0) then begin
-        giq_option(4, 838, 248, Node028, 50);
+    Reply(247);
+    if (global_var(GVAR_PLAYER_REPUTATION) >= 0) then begin
+        giq_option(4, NAME, 248, Node028, NEUTRAL_REACTION);
     end
     else begin
-        giq_option(4, 838, 249, Node028, 50);
+        giq_option(4, NAME, 249, Node028, NEUTRAL_REACTION);
     end
 end
 
 procedure Node028
 begin
-    set_global_var(405, 1);
-    set_global_var(541, 1);
-    gsay_reply(838, 250);
-    if (global_var(0) >= 0) then begin
-        giq_option(4, 838, 251, Node029, 50);
+    set_global_var(GVAR_PIPBOY_CREDITS, 1);
+    set_global_var(GVAR_VAULT_CITY_DESIGNER_NOTES, 1);
+    Reply(250);
+    if (global_var(GVAR_PLAYER_REPUTATION) >= 0) then begin
+        giq_option(4, NAME, 251, Node029, NEUTRAL_REACTION);
     end
     else begin
-        giq_option(4, 838, 252, Node029, 50);
+        giq_option(4, NAME, 252, Node029, NEUTRAL_REACTION);
     end
 end
 
 procedure Node029
 begin
-    set_global_var(401, 3);
-    gsay_reply(838, 253);
-    if (global_var(0) >= 0) then begin
-        giq_option(4, 838, 254, Node999, 50);
+    set_global_var(GVAR_VAULT_PIP, 3);
+    Reply(253);
+    if (global_var(GVAR_PLAYER_REPUTATION) >= 0) then begin
+        giq_option(4, NAME, 254, Node999, NEUTRAL_REACTION);
     end
     else begin
-        giq_option(4, 838, 255, Node999, 50);
+        giq_option(4, NAME, 255, Node999, NEUTRAL_REACTION);
     end
 end
 
 procedure Node030
 begin
-    display_msg(message_str(838, 256));
+    display_mstr(256);
 end
 
 procedure Node031
 begin
-    gsay_reply(838, 258);
-    giq_option(4, 838, 263, Node032, 50);
-    giq_option(4, 838, 264, Node021, 50);
-    giq_option(4, 838, 265, Node006a, 50);
+    Reply(258);
+    giq_option(4, NAME, 263, Node032, NEUTRAL_REACTION);
+    giq_option(4, NAME, 264, Node021, NEUTRAL_REACTION);
+    giq_option(4, NAME, 265, Node006a, NEUTRAL_REACTION);
 end
 
 procedure Node032
 begin
     variable LVar0 := 0;
-    set_global_var(82, 13);
-    gsay_reply(838, 266);
-    giq_option(4, 838, 270, Node021, 50);
-    giq_option(4, 838, 271, Node006a, 50);
+    set_global_var(GVAR_VAULT_GECKO_PLANT, 13);
+    Reply(266);
+    giq_option(4, NAME, 270, Node021, NEUTRAL_REACTION);
+    giq_option(4, NAME, 271, Node006a, NEUTRAL_REACTION);
 end
 
 procedure Node006a
 begin
-    if ((global_var(401) == 0) and (get_critter_stat(dude_obj, 1) > 6)) then begin
+    if ((global_var(GVAR_VAULT_PIP) == 0) and (dude_perception > 6)) then begin
         call Node012();
     end
     else begin
@@ -630,7 +630,7 @@ end
 
 procedure Node006b
 begin
-    if (global_var(401) == 0) then begin
+    if (global_var(GVAR_VAULT_PIP) == 0) then begin
         call Node012();
     end
     else begin
@@ -640,11 +640,10 @@ end
 
 procedure Node010a
 begin
-    if (global_var(401) == 0) then begin
+    if (global_var(GVAR_VAULT_PIP) == 0) then begin
         call Node011();
     end
     else begin
         call Node021();
     end
 end
-
