diff --git a/Games/Fallout2.Steam/Original/VIEDWCH.ssl b/Games/Fallout2.Steam/ReDefine/VIEDWCH.ssl
index 4dd7e78..744640a 100644
--- a/Games/Fallout2.Steam/Original/VIEDWCH.ssl
+++ b/Games/Fallout2.Steam/ReDefine/VIEDWCH.ssl
@@ -153,11 +153,11 @@ end
 
 procedure map_enter_p_proc
 begin
-    if (metarule(22, 0) == 0) then begin
+    if (is_loading_game == 0) then begin
         if (0) then begin
             obj_close(self_obj);
         end
-        if (metarule(14, 0) and 1) then begin
+        if (map_first_run and 1) then begin
             obj_lock(self_obj);
         end
     end
@@ -175,7 +175,7 @@ begin
         if (caught_tamper == 0) then begin
             if (1) then begin
                 if (Ed_Ptr != 0) then begin
-                    if (has_trait(1, Ed_Ptr, 666) and ((critter_state(Ed_Ptr) bwand 1) == 0)) then begin
+                    if (obj_is_visible_flag(Ed_Ptr) and ((critter_state(Ed_Ptr) bwand 1) == 0)) then begin
                         if (obj_can_see_obj(Ed_Ptr, user_obj)) then begin
                             caught_tamper := 1;
                         end
@@ -190,7 +190,7 @@ begin
                     attack_setup(Ed_Ptr, user_obj);
                 end
                 else begin
-                    float_msg(Ed_Ptr, message_str(14, 5400), 8);
+                    float_msg(Ed_Ptr, g_mstr(5400), FLOAT_MSG_YELLOW);
                 end
                 set_local_var(0, local_var(0) bwor 1024);
             end
@@ -202,13 +202,13 @@ procedure damage_p_proc
 begin
     if (obj_is_locked(self_obj)) then begin
         if (target_obj != 0) then begin
-            if (metarule(49, target_obj) == 6) then begin
+            if (weapon_dmg_type(target_obj) == 6) then begin
                 if (1) then begin
-                    display_msg(message_str(1189, 300));
+                    display_msg(message_str(SCRIPT_ZILOCKER, 300));
                     obj_unlock(self_obj);
                 end
                 else begin
-                    display_msg(message_str(1189, 301));
+                    display_msg(message_str(SCRIPT_ZILOCKER, 301));
                 end
             end
         end
@@ -221,7 +221,7 @@ begin
                 attack_setup(Ed_Ptr, user_obj);
             end
             else begin
-                float_msg(Ed_Ptr, message_str(14, 5400), 8);
+                float_msg(Ed_Ptr, g_mstr(5400), FLOAT_MSG_YELLOW);
             end
             set_local_var(0, local_var(0) bwor 1024);
         end
@@ -251,7 +251,7 @@ begin
         if (caught_tamper == 0) then begin
             if (1) then begin
                 if (Ed_Ptr != 0) then begin
-                    if (has_trait(1, Ed_Ptr, 666) and ((critter_state(Ed_Ptr) bwand 1) == 0)) then begin
+                    if (obj_is_visible_flag(Ed_Ptr) and ((critter_state(Ed_Ptr) bwand 1) == 0)) then begin
                         if (obj_can_see_obj(Ed_Ptr, user_obj)) then begin
                             caught_tamper := 1;
                         end
@@ -266,7 +266,7 @@ begin
                     attack_setup(Ed_Ptr, user_obj);
                 end
                 else begin
-                    float_msg(Ed_Ptr, message_str(14, 5400), 8);
+                    float_msg(Ed_Ptr, g_mstr(5400), FLOAT_MSG_YELLOW);
                 end
                 set_local_var(0, local_var(0) bwor 1024);
             end
@@ -279,19 +279,19 @@ begin
                         if (is_success(roll_vs_skill(user_obj, 9, -20 + 20))) then begin
                             script_overrides;
                             if (user_obj == dude_obj) then begin
-                                display_msg(message_str(1189, 100));
+                                display_msg(message_str(SCRIPT_ZILOCKER, 100));
                                 if (((local_var(0) bwand 512) != 0) == 0) then begin
                                     give_exp_points(25);
-                                    display_msg(message_str(14, 100) + 25 + message_str(14, 101));
+                                    display_msg(g_mstr(100) + 25 + g_mstr(101));
                                 end
                                 set_local_var(0, local_var(0) bwor 512);
                             end
                             else begin
                                 if (user_obj != 0) then begin
-                                    display_msg(obj_name(user_obj) + message_str(1189, 200));
+                                    display_msg(obj_name(user_obj) + message_str(SCRIPT_ZILOCKER, 200));
                                     if (((local_var(0) bwand 512) != 0) == 0) then begin
                                         give_exp_points(25);
-                                        display_msg(message_str(14, 100) + 25 + message_str(14, 101));
+                                        display_msg(g_mstr(100) + 25 + g_mstr(101));
                                     end
                                     set_local_var(0, local_var(0) bwor 512);
                                 end
@@ -302,11 +302,11 @@ begin
                         else begin
                             script_overrides;
                             if (user_obj == dude_obj) then begin
-                                display_msg(message_str(1189, 101));
+                                display_msg(message_str(SCRIPT_ZILOCKER, 101));
                             end
                             else begin
                                 if (user_obj != 0) then begin
-                                    display_msg(obj_name(user_obj) + message_str(1189, 201));
+                                    display_msg(obj_name(user_obj) + message_str(SCRIPT_ZILOCKER, 201));
                                 end
                             end
                             debug_msg("unlock_fail_msg");
@@ -320,19 +320,19 @@ begin
                             if (is_success(roll_vs_skill(user_obj, 9, -20 + 20))) then begin
                                 script_overrides;
                                 if (user_obj == dude_obj) then begin
-                                    display_msg(message_str(1189, 100));
+                                    display_msg(message_str(SCRIPT_ZILOCKER, 100));
                                     if (((local_var(0) bwand 512) != 0) == 0) then begin
                                         give_exp_points(25);
-                                        display_msg(message_str(14, 100) + 25 + message_str(14, 101));
+                                        display_msg(g_mstr(100) + 25 + g_mstr(101));
                                     end
                                     set_local_var(0, local_var(0) bwor 512);
                                 end
                                 else begin
                                     if (user_obj != 0) then begin
-                                        display_msg(obj_name(user_obj) + message_str(1189, 200));
+                                        display_msg(obj_name(user_obj) + message_str(SCRIPT_ZILOCKER, 200));
                                         if (((local_var(0) bwand 512) != 0) == 0) then begin
                                             give_exp_points(25);
-                                            display_msg(message_str(14, 100) + 25 + message_str(14, 101));
+                                            display_msg(g_mstr(100) + 25 + g_mstr(101));
                                         end
                                         set_local_var(0, local_var(0) bwor 512);
                                     end
@@ -343,11 +343,11 @@ begin
                             else begin
                                 script_overrides;
                                 if (user_obj == dude_obj) then begin
-                                    display_msg(message_str(1189, 101));
+                                    display_msg(message_str(SCRIPT_ZILOCKER, 101));
                                 end
                                 else begin
                                     if (user_obj != 0) then begin
-                                        display_msg(obj_name(user_obj) + message_str(1189, 201));
+                                        display_msg(obj_name(user_obj) + message_str(SCRIPT_ZILOCKER, 201));
                                     end
                                 end
                                 debug_msg("unlock_fail_msg");
@@ -359,19 +359,19 @@ begin
                             script_overrides;
                             if (is_success(do_check(user_obj, 0, 0))) then begin
                                 if (user_obj == dude_obj) then begin
-                                    display_msg(message_str(1189, 400));
+                                    display_msg(message_str(SCRIPT_ZILOCKER, 400));
                                     if (((local_var(0) bwand 512) != 0) == 0) then begin
                                         give_exp_points(25);
-                                        display_msg(message_str(14, 100) + 25 + message_str(14, 101));
+                                        display_msg(g_mstr(100) + 25 + g_mstr(101));
                                     end
                                     set_local_var(0, local_var(0) bwor 512);
                                 end
                                 else begin
                                     if (user_obj != 0) then begin
-                                        display_msg(obj_name(user_obj) + message_str(1189, 500));
+                                        display_msg(obj_name(user_obj) + message_str(SCRIPT_ZILOCKER, 500));
                                         if (((local_var(0) bwand 512) != 0) == 0) then begin
                                             give_exp_points(25);
-                                            display_msg(message_str(14, 100) + 25 + message_str(14, 101));
+                                            display_msg(g_mstr(100) + 25 + g_mstr(101));
                                         end
                                         set_local_var(0, local_var(0) bwor 512);
                                     end
@@ -381,11 +381,11 @@ begin
                             end
                             else begin
                                 if (user_obj == dude_obj) then begin
-                                    display_msg(message_str(1189, 401));
+                                    display_msg(message_str(SCRIPT_ZILOCKER, 401));
                                 end
                                 else begin
                                     if (user_obj != 0) then begin
-                                        display_msg(obj_name(user_obj) + message_str(1189, 501));
+                                        display_msg(obj_name(user_obj) + message_str(SCRIPT_ZILOCKER, 501));
                                     end
                                 end
                                 debug_msg("pry_fail_msg");
@@ -401,19 +401,19 @@ begin
                         if (is_success(roll_vs_skill(user_obj, 9, -20 + 40))) then begin
                             script_overrides;
                             if (user_obj == dude_obj) then begin
-                                display_msg(message_str(1189, 102));
+                                display_msg(message_str(SCRIPT_ZILOCKER, 102));
                                 if (((local_var(0) bwand 512) != 0) == 0) then begin
                                     give_exp_points(25);
-                                    display_msg(message_str(14, 100) + 25 + message_str(14, 101));
+                                    display_msg(g_mstr(100) + 25 + g_mstr(101));
                                 end
                                 set_local_var(0, local_var(0) bwor 512);
                             end
                             else begin
                                 if (user_obj != 0) then begin
-                                    display_msg(obj_name(user_obj) + message_str(1189, 202));
+                                    display_msg(obj_name(user_obj) + message_str(SCRIPT_ZILOCKER, 202));
                                     if (((local_var(0) bwand 512) != 0) == 0) then begin
                                         give_exp_points(25);
-                                        display_msg(message_str(14, 100) + 25 + message_str(14, 101));
+                                        display_msg(g_mstr(100) + 25 + g_mstr(101));
                                     end
                                     set_local_var(0, local_var(0) bwor 512);
                                 end
@@ -424,11 +424,11 @@ begin
                         else begin
                             script_overrides;
                             if (user_obj == dude_obj) then begin
-                                display_msg(message_str(1189, 103));
+                                display_msg(message_str(SCRIPT_ZILOCKER, 103));
                             end
                             else begin
                                 if (user_obj != 0) then begin
-                                    display_msg(obj_name(user_obj) + message_str(1189, 203));
+                                    display_msg(obj_name(user_obj) + message_str(SCRIPT_ZILOCKER, 203));
                                 end
                             end
                             debug_msg("lock_fail_msg");
@@ -442,19 +442,19 @@ begin
                             if (is_success(roll_vs_skill(user_obj, 9, -20 + 40))) then begin
                                 script_overrides;
                                 if (user_obj == dude_obj) then begin
-                                    display_msg(message_str(1189, 102));
+                                    display_msg(message_str(SCRIPT_ZILOCKER, 102));
                                     if (((local_var(0) bwand 512) != 0) == 0) then begin
                                         give_exp_points(25);
-                                        display_msg(message_str(14, 100) + 25 + message_str(14, 101));
+                                        display_msg(g_mstr(100) + 25 + g_mstr(101));
                                     end
                                     set_local_var(0, local_var(0) bwor 512);
                                 end
                                 else begin
                                     if (user_obj != 0) then begin
-                                        display_msg(obj_name(user_obj) + message_str(1189, 202));
+                                        display_msg(obj_name(user_obj) + message_str(SCRIPT_ZILOCKER, 202));
                                         if (((local_var(0) bwand 512) != 0) == 0) then begin
                                             give_exp_points(25);
-                                            display_msg(message_str(14, 100) + 25 + message_str(14, 101));
+                                            display_msg(g_mstr(100) + 25 + g_mstr(101));
                                         end
                                         set_local_var(0, local_var(0) bwor 512);
                                     end
@@ -465,11 +465,11 @@ begin
                             else begin
                                 script_overrides;
                                 if (user_obj == dude_obj) then begin
-                                    display_msg(message_str(1189, 103));
+                                    display_msg(message_str(SCRIPT_ZILOCKER, 103));
                                 end
                                 else begin
                                     if (user_obj != 0) then begin
-                                        display_msg(obj_name(user_obj) + message_str(1189, 203));
+                                        display_msg(obj_name(user_obj) + message_str(SCRIPT_ZILOCKER, 203));
                                     end
                                 end
                                 debug_msg("lock_fail_msg");
@@ -502,7 +502,7 @@ begin
         if (caught_tamper == 0) then begin
             if (1) then begin
                 if (Ed_Ptr != 0) then begin
-                    if (has_trait(1, Ed_Ptr, 666) and ((critter_state(Ed_Ptr) bwand 1) == 0)) then begin
+                    if (obj_is_visible_flag(Ed_Ptr) and ((critter_state(Ed_Ptr) bwand 1) == 0)) then begin
                         if (obj_can_see_obj(Ed_Ptr, user_obj)) then begin
                             caught_tamper := 1;
                         end
@@ -517,7 +517,7 @@ begin
                     attack_setup(Ed_Ptr, user_obj);
                 end
                 else begin
-                    float_msg(Ed_Ptr, message_str(14, 5400), 8);
+                    float_msg(Ed_Ptr, g_mstr(5400), FLOAT_MSG_YELLOW);
                 end
                 set_local_var(0, local_var(0) bwor 1024);
             end
@@ -530,19 +530,19 @@ begin
                         if (is_success(roll_vs_skill(user_obj, 9, -20))) then begin
                             script_overrides;
                             if (user_obj == dude_obj) then begin
-                                display_msg(message_str(1189, 100));
+                                display_msg(message_str(SCRIPT_ZILOCKER, 100));
                                 if (((local_var(0) bwand 512) != 0) == 0) then begin
                                     give_exp_points(25);
-                                    display_msg(message_str(14, 100) + 25 + message_str(14, 101));
+                                    display_msg(g_mstr(100) + 25 + g_mstr(101));
                                 end
                                 set_local_var(0, local_var(0) bwor 512);
                             end
                             else begin
                                 if (user_obj != 0) then begin
-                                    display_msg(obj_name(user_obj) + message_str(1189, 200));
+                                    display_msg(obj_name(user_obj) + message_str(SCRIPT_ZILOCKER, 200));
                                     if (((local_var(0) bwand 512) != 0) == 0) then begin
                                         give_exp_points(25);
-                                        display_msg(message_str(14, 100) + 25 + message_str(14, 101));
+                                        display_msg(g_mstr(100) + 25 + g_mstr(101));
                                     end
                                     set_local_var(0, local_var(0) bwor 512);
                                 end
@@ -553,11 +553,11 @@ begin
                         else begin
                             script_overrides;
                             if (user_obj == dude_obj) then begin
-                                display_msg(message_str(1189, 101));
+                                display_msg(message_str(SCRIPT_ZILOCKER, 101));
                             end
                             else begin
                                 if (user_obj != 0) then begin
-                                    display_msg(obj_name(user_obj) + message_str(1189, 201));
+                                    display_msg(obj_name(user_obj) + message_str(SCRIPT_ZILOCKER, 201));
                                 end
                             end
                             debug_msg("unlock_fail_msg");
@@ -569,19 +569,19 @@ begin
                         if (is_success(roll_vs_skill(user_obj, 9, -20))) then begin
                             script_overrides;
                             if (user_obj == dude_obj) then begin
-                                display_msg(message_str(1189, 102));
+                                display_msg(message_str(SCRIPT_ZILOCKER, 102));
                                 if (((local_var(0) bwand 512) != 0) == 0) then begin
                                     give_exp_points(25);
-                                    display_msg(message_str(14, 100) + 25 + message_str(14, 101));
+                                    display_msg(g_mstr(100) + 25 + g_mstr(101));
                                 end
                                 set_local_var(0, local_var(0) bwor 512);
                             end
                             else begin
                                 if (user_obj != 0) then begin
-                                    display_msg(obj_name(user_obj) + message_str(1189, 202));
+                                    display_msg(obj_name(user_obj) + message_str(SCRIPT_ZILOCKER, 202));
                                     if (((local_var(0) bwand 512) != 0) == 0) then begin
                                         give_exp_points(25);
-                                        display_msg(message_str(14, 100) + 25 + message_str(14, 101));
+                                        display_msg(g_mstr(100) + 25 + g_mstr(101));
                                     end
                                     set_local_var(0, local_var(0) bwor 512);
                                 end
@@ -592,11 +592,11 @@ begin
                         else begin
                             script_overrides;
                             if (user_obj == dude_obj) then begin
-                                display_msg(message_str(1189, 103));
+                                display_msg(message_str(SCRIPT_ZILOCKER, 103));
                             end
                             else begin
                                 if (user_obj != 0) then begin
-                                    display_msg(obj_name(user_obj) + message_str(1189, 203));
+                                    display_msg(obj_name(user_obj) + message_str(SCRIPT_ZILOCKER, 203));
                                 end
                             end
                             debug_msg("lock_fail_msg");
@@ -607,4 +607,3 @@ begin
         end
     end
 end
-
