diff --git a/Games/Fallout2.Steam/Original/VIRANBOX.ssl b/Games/Fallout2.Steam/ReDefine/VIRANBOX.ssl
index c71da9e..4b1e80c 100644
--- a/Games/Fallout2.Steam/Original/VIRANBOX.ssl
+++ b/Games/Fallout2.Steam/ReDefine/VIRANBOX.ssl
@@ -141,13 +141,13 @@ end
 procedure map_enter_p_proc
 begin
     vault_city_randal_box := self_obj;
-    if (metarule(22, 0) == 0) then begin
+    if (is_loading_game == 0) then begin
         if (local_var(0) < game_time) then begin
             restock_amt := random(200, 600);
             if (obj_is_carrying_obj_pid(self_obj, 41) < restock_amt) then begin
                 if (100 >= random(1, 100)) then begin
                     if (obj_is_carrying_obj_pid(self_obj, 41) < restock_amt) then begin
-                        restock_obj := create_object_sid(41, 0, 0, -1);
+                        restock_obj := create_object(41, 0, 0);
                         add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 41));
                     end
                     else begin
@@ -170,7 +170,7 @@ begin
                                     end
                                     else begin
                                         if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                            metarule(43, self_obj);
+                                            inven_unwield(self_obj);
                                         end
                                     end
                                 end
@@ -188,7 +188,7 @@ begin
             end
             else begin
                 if (obj_is_carrying_obj_pid(self_obj, 41) < restock_amt) then begin
-                    restock_obj := create_object_sid(41, 0, 0, -1);
+                    restock_obj := create_object(41, 0, 0);
                     add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 41));
                 end
                 else begin
@@ -211,7 +211,7 @@ begin
                                 end
                                 else begin
                                     if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                        metarule(43, self_obj);
+                                        inven_unwield(self_obj);
                                     end
                                 end
                             end
@@ -230,7 +230,7 @@ begin
             if (obj_is_carrying_obj_pid(self_obj, 73) < restock_amt) then begin
                 if (75 >= random(1, 100)) then begin
                     if (obj_is_carrying_obj_pid(self_obj, 73) < restock_amt) then begin
-                        restock_obj := create_object_sid(73, 0, 0, -1);
+                        restock_obj := create_object(73, 0, 0);
                         add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 73));
                     end
                     else begin
@@ -253,7 +253,7 @@ begin
                                     end
                                     else begin
                                         if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                            metarule(43, self_obj);
+                                            inven_unwield(self_obj);
                                         end
                                     end
                                 end
@@ -271,7 +271,7 @@ begin
             end
             else begin
                 if (obj_is_carrying_obj_pid(self_obj, 73) < restock_amt) then begin
-                    restock_obj := create_object_sid(73, 0, 0, -1);
+                    restock_obj := create_object(73, 0, 0);
                     add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 73));
                 end
                 else begin
@@ -294,7 +294,7 @@ begin
                                 end
                                 else begin
                                     if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                        metarule(43, self_obj);
+                                        inven_unwield(self_obj);
                                     end
                                 end
                             end
@@ -313,7 +313,7 @@ begin
             if (obj_is_carrying_obj_pid(self_obj, 76) < restock_amt) then begin
                 if (75 >= random(1, 100)) then begin
                     if (obj_is_carrying_obj_pid(self_obj, 76) < restock_amt) then begin
-                        restock_obj := create_object_sid(76, 0, 0, -1);
+                        restock_obj := create_object(76, 0, 0);
                         add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 76));
                     end
                     else begin
@@ -336,7 +336,7 @@ begin
                                     end
                                     else begin
                                         if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                            metarule(43, self_obj);
+                                            inven_unwield(self_obj);
                                         end
                                     end
                                 end
@@ -354,7 +354,7 @@ begin
             end
             else begin
                 if (obj_is_carrying_obj_pid(self_obj, 76) < restock_amt) then begin
-                    restock_obj := create_object_sid(76, 0, 0, -1);
+                    restock_obj := create_object(76, 0, 0);
                     add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 76));
                 end
                 else begin
@@ -377,7 +377,7 @@ begin
                                 end
                                 else begin
                                     if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                        metarule(43, self_obj);
+                                        inven_unwield(self_obj);
                                     end
                                 end
                             end
@@ -396,7 +396,7 @@ begin
             if (obj_is_carrying_obj_pid(self_obj, 80) < restock_amt) then begin
                 if (75 >= random(1, 100)) then begin
                     if (obj_is_carrying_obj_pid(self_obj, 80) < restock_amt) then begin
-                        restock_obj := create_object_sid(80, 0, 0, -1);
+                        restock_obj := create_object(80, 0, 0);
                         add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 80));
                     end
                     else begin
@@ -419,7 +419,7 @@ begin
                                     end
                                     else begin
                                         if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                            metarule(43, self_obj);
+                                            inven_unwield(self_obj);
                                         end
                                     end
                                 end
@@ -437,7 +437,7 @@ begin
             end
             else begin
                 if (obj_is_carrying_obj_pid(self_obj, 80) < restock_amt) then begin
-                    restock_obj := create_object_sid(80, 0, 0, -1);
+                    restock_obj := create_object(80, 0, 0);
                     add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 80));
                 end
                 else begin
@@ -460,7 +460,7 @@ begin
                                 end
                                 else begin
                                     if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                        metarule(43, self_obj);
+                                        inven_unwield(self_obj);
                                     end
                                 end
                             end
@@ -479,7 +479,7 @@ begin
             if (obj_is_carrying_obj_pid(self_obj, 4) < restock_amt) then begin
                 if (100 >= random(1, 100)) then begin
                     if (obj_is_carrying_obj_pid(self_obj, 4) < restock_amt) then begin
-                        restock_obj := create_object_sid(4, 0, 0, -1);
+                        restock_obj := create_object(4, 0, 0);
                         add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 4));
                     end
                     else begin
@@ -502,7 +502,7 @@ begin
                                     end
                                     else begin
                                         if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                            metarule(43, self_obj);
+                                            inven_unwield(self_obj);
                                         end
                                     end
                                 end
@@ -520,7 +520,7 @@ begin
             end
             else begin
                 if (obj_is_carrying_obj_pid(self_obj, 4) < restock_amt) then begin
-                    restock_obj := create_object_sid(4, 0, 0, -1);
+                    restock_obj := create_object(4, 0, 0);
                     add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 4));
                 end
                 else begin
@@ -543,7 +543,7 @@ begin
                                 end
                                 else begin
                                     if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                        metarule(43, self_obj);
+                                        inven_unwield(self_obj);
                                     end
                                 end
                             end
@@ -562,7 +562,7 @@ begin
             if (obj_is_carrying_obj_pid(self_obj, 8) < restock_amt) then begin
                 if (100 >= random(1, 100)) then begin
                     if (obj_is_carrying_obj_pid(self_obj, 8) < restock_amt) then begin
-                        restock_obj := create_object_sid(8, 0, 0, -1);
+                        restock_obj := create_object(8, 0, 0);
                         add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 8));
                     end
                     else begin
@@ -585,7 +585,7 @@ begin
                                     end
                                     else begin
                                         if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                            metarule(43, self_obj);
+                                            inven_unwield(self_obj);
                                         end
                                     end
                                 end
@@ -603,7 +603,7 @@ begin
             end
             else begin
                 if (obj_is_carrying_obj_pid(self_obj, 8) < restock_amt) then begin
-                    restock_obj := create_object_sid(8, 0, 0, -1);
+                    restock_obj := create_object(8, 0, 0);
                     add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 8));
                 end
                 else begin
@@ -626,7 +626,7 @@ begin
                                 end
                                 else begin
                                     if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                        metarule(43, self_obj);
+                                        inven_unwield(self_obj);
                                     end
                                 end
                             end
@@ -645,7 +645,7 @@ begin
             if (obj_is_carrying_obj_pid(self_obj, 94) < restock_amt) then begin
                 if (25 >= random(1, 100)) then begin
                     if (obj_is_carrying_obj_pid(self_obj, 94) < restock_amt) then begin
-                        restock_obj := create_object_sid(94, 0, 0, -1);
+                        restock_obj := create_object(94, 0, 0);
                         add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 94));
                     end
                     else begin
@@ -668,7 +668,7 @@ begin
                                     end
                                     else begin
                                         if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                            metarule(43, self_obj);
+                                            inven_unwield(self_obj);
                                         end
                                     end
                                 end
@@ -686,7 +686,7 @@ begin
             end
             else begin
                 if (obj_is_carrying_obj_pid(self_obj, 94) < restock_amt) then begin
-                    restock_obj := create_object_sid(94, 0, 0, -1);
+                    restock_obj := create_object(94, 0, 0);
                     add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 94));
                 end
                 else begin
@@ -709,7 +709,7 @@ begin
                                 end
                                 else begin
                                     if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                        metarule(43, self_obj);
+                                        inven_unwield(self_obj);
                                     end
                                 end
                             end
@@ -728,7 +728,7 @@ begin
             if (obj_is_carrying_obj_pid(self_obj, 5) < restock_amt) then begin
                 if (25 >= random(1, 100)) then begin
                     if (obj_is_carrying_obj_pid(self_obj, 5) < restock_amt) then begin
-                        restock_obj := create_object_sid(5, 0, 0, -1);
+                        restock_obj := create_object(5, 0, 0);
                         add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 5));
                     end
                     else begin
@@ -751,7 +751,7 @@ begin
                                     end
                                     else begin
                                         if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                            metarule(43, self_obj);
+                                            inven_unwield(self_obj);
                                         end
                                     end
                                 end
@@ -769,7 +769,7 @@ begin
             end
             else begin
                 if (obj_is_carrying_obj_pid(self_obj, 5) < restock_amt) then begin
-                    restock_obj := create_object_sid(5, 0, 0, -1);
+                    restock_obj := create_object(5, 0, 0);
                     add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 5));
                 end
                 else begin
@@ -792,7 +792,7 @@ begin
                                 end
                                 else begin
                                     if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                        metarule(43, self_obj);
+                                        inven_unwield(self_obj);
                                     end
                                 end
                             end
@@ -811,7 +811,7 @@ begin
             if (obj_is_carrying_obj_pid(self_obj, 2) < restock_amt) then begin
                 if (25 >= random(1, 100)) then begin
                     if (obj_is_carrying_obj_pid(self_obj, 2) < restock_amt) then begin
-                        restock_obj := create_object_sid(2, 0, 0, -1);
+                        restock_obj := create_object(2, 0, 0);
                         add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 2));
                     end
                     else begin
@@ -834,7 +834,7 @@ begin
                                     end
                                     else begin
                                         if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                            metarule(43, self_obj);
+                                            inven_unwield(self_obj);
                                         end
                                     end
                                 end
@@ -852,7 +852,7 @@ begin
             end
             else begin
                 if (obj_is_carrying_obj_pid(self_obj, 2) < restock_amt) then begin
-                    restock_obj := create_object_sid(2, 0, 0, -1);
+                    restock_obj := create_object(2, 0, 0);
                     add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 2));
                 end
                 else begin
@@ -875,7 +875,7 @@ begin
                                 end
                                 else begin
                                     if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                        metarule(43, self_obj);
+                                        inven_unwield(self_obj);
                                     end
                                 end
                             end
@@ -894,7 +894,7 @@ begin
             if (obj_is_carrying_obj_pid(self_obj, 29) < restock_amt) then begin
                 if (100 >= random(1, 100)) then begin
                     if (obj_is_carrying_obj_pid(self_obj, 29) < restock_amt) then begin
-                        restock_obj := create_object_sid(29, 0, 0, -1);
+                        restock_obj := create_object(29, 0, 0);
                         add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 29));
                     end
                     else begin
@@ -917,7 +917,7 @@ begin
                                     end
                                     else begin
                                         if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                            metarule(43, self_obj);
+                                            inven_unwield(self_obj);
                                         end
                                     end
                                 end
@@ -935,7 +935,7 @@ begin
             end
             else begin
                 if (obj_is_carrying_obj_pid(self_obj, 29) < restock_amt) then begin
-                    restock_obj := create_object_sid(29, 0, 0, -1);
+                    restock_obj := create_object(29, 0, 0);
                     add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 29));
                 end
                 else begin
@@ -958,7 +958,7 @@ begin
                                 end
                                 else begin
                                     if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                        metarule(43, self_obj);
+                                        inven_unwield(self_obj);
                                     end
                                 end
                             end
@@ -977,7 +977,7 @@ begin
             if (obj_is_carrying_obj_pid(self_obj, 95) < restock_amt) then begin
                 if (100 >= random(1, 100)) then begin
                     if (obj_is_carrying_obj_pid(self_obj, 95) < restock_amt) then begin
-                        restock_obj := create_object_sid(95, 0, 0, -1);
+                        restock_obj := create_object(95, 0, 0);
                         add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 95));
                     end
                     else begin
@@ -1000,7 +1000,7 @@ begin
                                     end
                                     else begin
                                         if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                            metarule(43, self_obj);
+                                            inven_unwield(self_obj);
                                         end
                                     end
                                 end
@@ -1018,7 +1018,7 @@ begin
             end
             else begin
                 if (obj_is_carrying_obj_pid(self_obj, 95) < restock_amt) then begin
-                    restock_obj := create_object_sid(95, 0, 0, -1);
+                    restock_obj := create_object(95, 0, 0);
                     add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 95));
                 end
                 else begin
@@ -1041,7 +1041,7 @@ begin
                                 end
                                 else begin
                                     if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                        metarule(43, self_obj);
+                                        inven_unwield(self_obj);
                                     end
                                 end
                             end
@@ -1060,7 +1060,7 @@ begin
             if (obj_is_carrying_obj_pid(self_obj, 40) < restock_amt) then begin
                 if (100 >= random(1, 100)) then begin
                     if (obj_is_carrying_obj_pid(self_obj, 40) < restock_amt) then begin
-                        restock_obj := create_object_sid(40, 0, 0, -1);
+                        restock_obj := create_object(40, 0, 0);
                         add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 40));
                     end
                     else begin
@@ -1083,7 +1083,7 @@ begin
                                     end
                                     else begin
                                         if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                            metarule(43, self_obj);
+                                            inven_unwield(self_obj);
                                         end
                                     end
                                 end
@@ -1101,7 +1101,7 @@ begin
             end
             else begin
                 if (obj_is_carrying_obj_pid(self_obj, 40) < restock_amt) then begin
-                    restock_obj := create_object_sid(40, 0, 0, -1);
+                    restock_obj := create_object(40, 0, 0);
                     add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 40));
                 end
                 else begin
@@ -1124,7 +1124,7 @@ begin
                                 end
                                 else begin
                                     if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                        metarule(43, self_obj);
+                                        inven_unwield(self_obj);
                                     end
                                 end
                             end
@@ -1143,7 +1143,7 @@ begin
             if (obj_is_carrying_obj_pid(self_obj, 109) < restock_amt) then begin
                 if (100 >= random(1, 100)) then begin
                     if (obj_is_carrying_obj_pid(self_obj, 109) < restock_amt) then begin
-                        restock_obj := create_object_sid(109, 0, 0, -1);
+                        restock_obj := create_object(109, 0, 0);
                         add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 109));
                     end
                     else begin
@@ -1166,7 +1166,7 @@ begin
                                     end
                                     else begin
                                         if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                            metarule(43, self_obj);
+                                            inven_unwield(self_obj);
                                         end
                                     end
                                 end
@@ -1184,7 +1184,7 @@ begin
             end
             else begin
                 if (obj_is_carrying_obj_pid(self_obj, 109) < restock_amt) then begin
-                    restock_obj := create_object_sid(109, 0, 0, -1);
+                    restock_obj := create_object(109, 0, 0);
                     add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 109));
                 end
                 else begin
@@ -1207,7 +1207,7 @@ begin
                                 end
                                 else begin
                                     if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                        metarule(43, self_obj);
+                                        inven_unwield(self_obj);
                                     end
                                 end
                             end
@@ -1226,7 +1226,7 @@ begin
             if (obj_is_carrying_obj_pid(self_obj, 79) < restock_amt) then begin
                 if (50 >= random(1, 100)) then begin
                     if (obj_is_carrying_obj_pid(self_obj, 79) < restock_amt) then begin
-                        restock_obj := create_object_sid(79, 0, 0, -1);
+                        restock_obj := create_object(79, 0, 0);
                         add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 79));
                     end
                     else begin
@@ -1249,7 +1249,7 @@ begin
                                     end
                                     else begin
                                         if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                            metarule(43, self_obj);
+                                            inven_unwield(self_obj);
                                         end
                                     end
                                 end
@@ -1267,7 +1267,7 @@ begin
             end
             else begin
                 if (obj_is_carrying_obj_pid(self_obj, 79) < restock_amt) then begin
-                    restock_obj := create_object_sid(79, 0, 0, -1);
+                    restock_obj := create_object(79, 0, 0);
                     add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 79));
                 end
                 else begin
@@ -1290,7 +1290,7 @@ begin
                                 end
                                 else begin
                                     if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                        metarule(43, self_obj);
+                                        inven_unwield(self_obj);
                                     end
                                 end
                             end
@@ -1309,7 +1309,7 @@ begin
             if (obj_is_carrying_obj_pid(self_obj, 127) < restock_amt) then begin
                 if (50 >= random(1, 100)) then begin
                     if (obj_is_carrying_obj_pid(self_obj, 127) < restock_amt) then begin
-                        restock_obj := create_object_sid(127, 0, 0, -1);
+                        restock_obj := create_object(127, 0, 0);
                         add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 127));
                     end
                     else begin
@@ -1332,7 +1332,7 @@ begin
                                     end
                                     else begin
                                         if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                            metarule(43, self_obj);
+                                            inven_unwield(self_obj);
                                         end
                                     end
                                 end
@@ -1350,7 +1350,7 @@ begin
             end
             else begin
                 if (obj_is_carrying_obj_pid(self_obj, 127) < restock_amt) then begin
-                    restock_obj := create_object_sid(127, 0, 0, -1);
+                    restock_obj := create_object(127, 0, 0);
                     add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 127));
                 end
                 else begin
@@ -1373,7 +1373,7 @@ begin
                                 end
                                 else begin
                                     if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                        metarule(43, self_obj);
+                                        inven_unwield(self_obj);
                                     end
                                 end
                             end
@@ -1392,7 +1392,7 @@ begin
             if (obj_is_carrying_obj_pid(self_obj, 9) < restock_amt) then begin
                 if (100 >= random(1, 100)) then begin
                     if (obj_is_carrying_obj_pid(self_obj, 9) < restock_amt) then begin
-                        restock_obj := create_object_sid(9, 0, 0, -1);
+                        restock_obj := create_object(9, 0, 0);
                         add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 9));
                     end
                     else begin
@@ -1415,7 +1415,7 @@ begin
                                     end
                                     else begin
                                         if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                            metarule(43, self_obj);
+                                            inven_unwield(self_obj);
                                         end
                                     end
                                 end
@@ -1433,7 +1433,7 @@ begin
             end
             else begin
                 if (obj_is_carrying_obj_pid(self_obj, 9) < restock_amt) then begin
-                    restock_obj := create_object_sid(9, 0, 0, -1);
+                    restock_obj := create_object(9, 0, 0);
                     add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 9));
                 end
                 else begin
@@ -1456,7 +1456,7 @@ begin
                                 end
                                 else begin
                                     if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                        metarule(43, self_obj);
+                                        inven_unwield(self_obj);
                                     end
                                 end
                             end
@@ -1475,7 +1475,7 @@ begin
             if (obj_is_carrying_obj_pid(self_obj, 48) < restock_amt) then begin
                 if (100 >= random(1, 100)) then begin
                     if (obj_is_carrying_obj_pid(self_obj, 48) < restock_amt) then begin
-                        restock_obj := create_object_sid(48, 0, 0, -1);
+                        restock_obj := create_object(48, 0, 0);
                         add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 48));
                     end
                     else begin
@@ -1498,7 +1498,7 @@ begin
                                     end
                                     else begin
                                         if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                            metarule(43, self_obj);
+                                            inven_unwield(self_obj);
                                         end
                                     end
                                 end
@@ -1516,7 +1516,7 @@ begin
             end
             else begin
                 if (obj_is_carrying_obj_pid(self_obj, 48) < restock_amt) then begin
-                    restock_obj := create_object_sid(48, 0, 0, -1);
+                    restock_obj := create_object(48, 0, 0);
                     add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 48));
                 end
                 else begin
@@ -1539,7 +1539,7 @@ begin
                                 end
                                 else begin
                                     if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                        metarule(43, self_obj);
+                                        inven_unwield(self_obj);
                                     end
                                 end
                             end
@@ -1558,7 +1558,7 @@ begin
             if (obj_is_carrying_obj_pid(self_obj, 25) < restock_amt) then begin
                 if (100 >= random(1, 100)) then begin
                     if (obj_is_carrying_obj_pid(self_obj, 25) < restock_amt) then begin
-                        restock_obj := create_object_sid(25, 0, 0, -1);
+                        restock_obj := create_object(25, 0, 0);
                         add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 25));
                     end
                     else begin
@@ -1581,7 +1581,7 @@ begin
                                     end
                                     else begin
                                         if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                            metarule(43, self_obj);
+                                            inven_unwield(self_obj);
                                         end
                                     end
                                 end
@@ -1599,7 +1599,7 @@ begin
             end
             else begin
                 if (obj_is_carrying_obj_pid(self_obj, 25) < restock_amt) then begin
-                    restock_obj := create_object_sid(25, 0, 0, -1);
+                    restock_obj := create_object(25, 0, 0);
                     add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 25));
                 end
                 else begin
@@ -1622,7 +1622,7 @@ begin
                                 end
                                 else begin
                                     if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                        metarule(43, self_obj);
+                                        inven_unwield(self_obj);
                                     end
                                 end
                             end
@@ -1641,7 +1641,7 @@ begin
             if (obj_is_carrying_obj_pid(self_obj, 22) < restock_amt) then begin
                 if (100 >= random(1, 100)) then begin
                     if (obj_is_carrying_obj_pid(self_obj, 22) < restock_amt) then begin
-                        restock_obj := create_object_sid(22, 0, 0, -1);
+                        restock_obj := create_object(22, 0, 0);
                         add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 22));
                     end
                     else begin
@@ -1664,7 +1664,7 @@ begin
                                     end
                                     else begin
                                         if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                            metarule(43, self_obj);
+                                            inven_unwield(self_obj);
                                         end
                                     end
                                 end
@@ -1682,7 +1682,7 @@ begin
             end
             else begin
                 if (obj_is_carrying_obj_pid(self_obj, 22) < restock_amt) then begin
-                    restock_obj := create_object_sid(22, 0, 0, -1);
+                    restock_obj := create_object(22, 0, 0);
                     add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 22));
                 end
                 else begin
@@ -1705,7 +1705,7 @@ begin
                                 end
                                 else begin
                                     if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                        metarule(43, self_obj);
+                                        inven_unwield(self_obj);
                                     end
                                 end
                             end
@@ -1724,7 +1724,7 @@ begin
             if (obj_is_carrying_obj_pid(self_obj, 351) < restock_amt) then begin
                 if (25 >= random(1, 100)) then begin
                     if (obj_is_carrying_obj_pid(self_obj, 351) < restock_amt) then begin
-                        restock_obj := create_object_sid(351, 0, 0, -1);
+                        restock_obj := create_object(351, 0, 0);
                         add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 351));
                     end
                     else begin
@@ -1747,7 +1747,7 @@ begin
                                     end
                                     else begin
                                         if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                            metarule(43, self_obj);
+                                            inven_unwield(self_obj);
                                         end
                                     end
                                 end
@@ -1765,7 +1765,7 @@ begin
             end
             else begin
                 if (obj_is_carrying_obj_pid(self_obj, 351) < restock_amt) then begin
-                    restock_obj := create_object_sid(351, 0, 0, -1);
+                    restock_obj := create_object(351, 0, 0);
                     add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 351));
                 end
                 else begin
@@ -1788,7 +1788,7 @@ begin
                                 end
                                 else begin
                                     if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                        metarule(43, self_obj);
+                                        inven_unwield(self_obj);
                                     end
                                 end
                             end
@@ -1807,7 +1807,7 @@ begin
             if (obj_is_carrying_obj_pid(self_obj, 242) < restock_amt) then begin
                 if (25 >= random(1, 100)) then begin
                     if (obj_is_carrying_obj_pid(self_obj, 242) < restock_amt) then begin
-                        restock_obj := create_object_sid(242, 0, 0, -1);
+                        restock_obj := create_object(242, 0, 0);
                         add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 242));
                     end
                     else begin
@@ -1830,7 +1830,7 @@ begin
                                     end
                                     else begin
                                         if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                            metarule(43, self_obj);
+                                            inven_unwield(self_obj);
                                         end
                                     end
                                 end
@@ -1848,7 +1848,7 @@ begin
             end
             else begin
                 if (obj_is_carrying_obj_pid(self_obj, 242) < restock_amt) then begin
-                    restock_obj := create_object_sid(242, 0, 0, -1);
+                    restock_obj := create_object(242, 0, 0);
                     add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 242));
                 end
                 else begin
@@ -1871,7 +1871,7 @@ begin
                                 end
                                 else begin
                                     if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                        metarule(43, self_obj);
+                                        inven_unwield(self_obj);
                                     end
                                 end
                             end
@@ -1890,7 +1890,7 @@ begin
             if (obj_is_carrying_obj_pid(self_obj, 33) < restock_amt) then begin
                 if (100 >= random(1, 100)) then begin
                     if (obj_is_carrying_obj_pid(self_obj, 33) < restock_amt) then begin
-                        restock_obj := create_object_sid(33, 0, 0, -1);
+                        restock_obj := create_object(33, 0, 0);
                         add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 33));
                     end
                     else begin
@@ -1913,7 +1913,7 @@ begin
                                     end
                                     else begin
                                         if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                            metarule(43, self_obj);
+                                            inven_unwield(self_obj);
                                         end
                                     end
                                 end
@@ -1931,7 +1931,7 @@ begin
             end
             else begin
                 if (obj_is_carrying_obj_pid(self_obj, 33) < restock_amt) then begin
-                    restock_obj := create_object_sid(33, 0, 0, -1);
+                    restock_obj := create_object(33, 0, 0);
                     add_mult_objs_to_inven(self_obj, restock_obj, restock_amt - obj_is_carrying_obj_pid(self_obj, 33));
                 end
                 else begin
@@ -1954,7 +1954,7 @@ begin
                                 end
                                 else begin
                                     if ((critter_inven_obj(self_obj, 2) == restock_obj) or (critter_inven_obj(self_obj, 1) == restock_obj)) then begin
-                                        metarule(43, self_obj);
+                                        inven_unwield(self_obj);
                                     end
                                 end
                             end
@@ -1978,4 +1978,3 @@ procedure map_update_p_proc
 begin
     vault_city_randal_box := self_obj;
 end
-
