diff --git a/Games/Fallout2.Steam/Original/VIROCKS.ssl b/Games/Fallout2.Steam/ReDefine/VIROCKS.ssl
index e8eea80..3a42bfd 100644
--- a/Games/Fallout2.Steam/Original/VIROCKS.ssl
+++ b/Games/Fallout2.Steam/ReDefine/VIROCKS.ssl
@@ -148,18 +148,18 @@ begin
     variable LVar0 := 0;
     if (map_var(5) == 0) then begin
         script_overrides;
-        float_msg(dude_obj, message_str(981, 102), 8);
+        float_msg(dude_obj, mstr(102), FLOAT_MSG_YELLOW);
     end
     else begin
         if (map_var(5) == 1) then begin
-            LVar0 := create_object_sid(384, 0, 0, -1);
+            LVar0 := create_object(384, 0, 0);
             add_obj_to_inven(dude_obj, LVar0);
             set_map_var(5, 2);
-            float_msg(dude_obj, message_str(981, 103), 8);
+            float_msg(dude_obj, mstr(103), FLOAT_MSG_YELLOW);
         end
         else begin
             if (map_var(5) == 2) then begin
-                float_msg(dude_obj, message_str(981, 104), 8);
+                float_msg(dude_obj, mstr(104), FLOAT_MSG_YELLOW);
             end
         end
     end
@@ -168,13 +168,13 @@ end
 procedure look_at_p_proc
 begin
     script_overrides;
-    display_msg(message_str(981, 100));
+    display_mstr(100);
 end
 
 procedure description_p_proc
 begin
     script_overrides;
-    display_msg(message_str(981, 101));
+    display_mstr(101);
 end
 
 procedure use_skill_on_p_proc
@@ -204,4 +204,3 @@ end
 procedure timed_event_p_proc
 begin
 end
-
