diff --git a/Games/Fallout2.Steam/Original/VITERM.ssl b/Games/Fallout2.Steam/ReDefine/VITERM.ssl
index f23a392..f80be84 100644
--- a/Games/Fallout2.Steam/Original/VITERM.ssl
+++ b/Games/Fallout2.Steam/ReDefine/VITERM.ssl
@@ -146,7 +146,7 @@ end
 
 procedure use_p_proc
 begin
-    if ((global_var(599) == 1) and (local_var(0) == 0)) then begin
+    if ((global_var(GVAR_8_BALL_VAULT_TERMINAL) == 1) and (local_var(0) == 0)) then begin
         set_local_var(0, 1);
         add_timer_event(self_obj, 1, 1);
     end
@@ -181,31 +181,31 @@ begin
     if (fixed_param == 1) then begin
         if (Terminal_State == 0) then begin
             game_ui_disable;
-            float_msg(self_obj, message_str(1167, 101), 8);
+            float_msg(self_obj, mstr(101), FLOAT_MSG_YELLOW);
             Terminal_State := Terminal_State + 1;
             add_timer_event(self_obj, game_ticks(2), 1);
         end
         else begin
             if (Terminal_State == 1) then begin
-                float_msg(self_obj, message_str(1167, 102), 8);
+                float_msg(self_obj, mstr(102), FLOAT_MSG_YELLOW);
                 Terminal_State := Terminal_State + 1;
                 add_timer_event(self_obj, game_ticks(2), 1);
             end
             else begin
                 if (Terminal_State == 2) then begin
-                    float_msg(self_obj, message_str(1167, 103), 8);
+                    float_msg(self_obj, mstr(103), FLOAT_MSG_YELLOW);
                     Terminal_State := Terminal_State + 1;
                     add_timer_event(self_obj, game_ticks(3), 1);
                 end
                 else begin
                     if (Terminal_State == 3) then begin
-                        float_msg(self_obj, message_str(1167, 104), 8);
+                        float_msg(self_obj, mstr(104), FLOAT_MSG_YELLOW);
                         Terminal_State := Terminal_State + 1;
                         add_timer_event(self_obj, game_ticks(2), 1);
                     end
                     else begin
                         if (Terminal_State == 4) then begin
-                            float_msg(self_obj, message_str(1167, 105), 8);
+                            float_msg(self_obj, mstr(105), FLOAT_MSG_YELLOW);
                             Terminal_State := Terminal_State + 1;
                             add_timer_event(self_obj, game_ticks(3), 1);
                         end
@@ -220,14 +220,14 @@ begin
                             end
                             else begin
                                 if (Terminal_State == 6) then begin
-                                    float_msg(dude_obj, message_str(1167, 106), 3);
+                                    float_msg(dude_obj, mstr(106), FLOAT_MSG_GREEN);
                                     call Distribute_Booty();
                                     Terminal_State := Terminal_State + 1;
                                     add_timer_event(self_obj, game_ticks(2), 1);
                                 end
                                 else begin
                                     if (Terminal_State == 7) then begin
-                                        float_msg(self_obj, message_str(1167, 107), 8);
+                                        float_msg(self_obj, mstr(107), FLOAT_MSG_YELLOW);
                                         game_ui_enable;
                                     end
                                 end
@@ -243,9 +243,8 @@ end
 procedure Distribute_Booty
 begin
     variable LVar0 := 0;
-    LVar0 := create_object_sid(40, 0, 0, -1);
+    LVar0 := create_object(40, 0, 0);
     add_mult_objs_to_inven(dude_obj, LVar0, 2);
-    LVar0 := create_object_sid(144, 0, 0, -1);
+    LVar0 := create_object(144, 0, 0);
     add_obj_to_inven(dude_obj, LVar0);
 end
-
