diff --git a/Games/Fallout2.Steam/Original/VTALOMRK.ssl b/Games/Fallout2.Steam/ReDefine/VTALOMRK.ssl
index 919e4b9..032e73e 100644
--- a/Games/Fallout2.Steam/Original/VTALOMRK.ssl
+++ b/Games/Fallout2.Steam/ReDefine/VTALOMRK.ssl
@@ -142,9 +142,8 @@ end
 
 procedure spatial_p_proc
 begin
-    if ((global_var(401) == 3) and (floated == 0)) then begin
+    if ((global_var(GVAR_VAULT_PIP) == 3) and (floated == 0)) then begin
         floated := 1;
-        float_msg(dude_obj, message_str(839, 106), 8);
+        float_msg(dude_obj, message_str(SCRIPT_VIALOCOM, 106), FLOAT_MSG_YELLOW);
     end
 end
-
