diff --git a/Games/Fallout2.Steam/Original/WIEXPDSK.ssl b/Games/Fallout2.Steam/ReDefine/WIEXPDSK.ssl
index ccbdb64..8da6661 100644
--- a/Games/Fallout2.Steam/Original/WIEXPDSK.ssl
+++ b/Games/Fallout2.Steam/ReDefine/WIEXPDSK.ssl
@@ -137,18 +137,17 @@ end
 
 procedure use_p_proc
 begin
-    if (global_var(166) != 1) then begin
-        set_global_var(166, 1);
-        display_msg(message_str(151, 102));
+    if (global_var(GVAR_HOLODISK_SIERRA_EXP_LOG) != 1) then begin
+        set_global_var(GVAR_HOLODISK_SIERRA_EXP_LOG, 1);
+        display_msg(message_str(SCRIPT_WIEVCDSK, 102));
     end
     else begin
-        display_msg(message_str(151, 101));
+        display_msg(message_str(SCRIPT_WIEVCDSK, 101));
     end
 end
 
 procedure description_p_proc
 begin
     script_overrides;
-    display_msg(message_str(151, 100));
+    display_msg(message_str(SCRIPT_WIEVCDSK, 100));
 end
-
