diff --git a/Games/Fallout2.Steam/Original/WIMISDSK.ssl b/Games/Fallout2.Steam/ReDefine/WIMISDSK.ssl
index dde7f97..4b1a81e 100644
--- a/Games/Fallout2.Steam/Original/WIMISDSK.ssl
+++ b/Games/Fallout2.Steam/ReDefine/WIMISDSK.ssl
@@ -137,18 +137,17 @@ end
 
 procedure use_p_proc
 begin
-    if (global_var(521) != 1) then begin
-        set_global_var(521, 1);
-        display_msg(message_str(151, 102));
+    if (global_var(GVAR_SIERRA_MISSION_HOLODISK) != 1) then begin
+        set_global_var(GVAR_SIERRA_MISSION_HOLODISK, 1);
+        display_msg(message_str(SCRIPT_WIEVCDSK, 102));
     end
     else begin
-        display_msg(message_str(151, 101));
+        display_msg(message_str(SCRIPT_WIEVCDSK, 101));
     end
 end
 
 procedure description_p_proc
 begin
     script_overrides;
-    display_msg(message_str(151, 100));
+    display_msg(message_str(SCRIPT_WIEVCDSK, 100));
 end
-
