diff --git a/Games/Fallout2.Steam/Original/WIPNCHBG.ssl b/Games/Fallout2.Steam/ReDefine/WIPNCHBG.ssl
index 7cd7b14..adfbb88 100644
--- a/Games/Fallout2.Steam/Original/WIPNCHBG.ssl
+++ b/Games/Fallout2.Steam/ReDefine/WIPNCHBG.ssl
@@ -140,9 +140,8 @@ begin
         set_local_var(0, local_var(0) + 1);
         gfade_out(600);
         gfade_in(600);
-        critter_mod_skill(dude_obj, 3, 5);
+        critter_mod_skill(dude_obj, SKILL_UNARMED_COMBAT, 5);
         game_time_advance(60 * (60 * 10));
-        display_msg(message_str(14, 950));
+        display_msg(g_mstr(950));
     end
 end
-
