diff --git a/Games/Fallout2.Steam/Original/WIRETSCN.ssl b/Games/Fallout2.Steam/ReDefine/WIRETSCN.ssl
index f4ffeaa..ca428f4 100644
--- a/Games/Fallout2.Steam/Original/WIRETSCN.ssl
+++ b/Games/Fallout2.Steam/ReDefine/WIRETSCN.ssl
@@ -139,30 +139,30 @@ procedure use_p_proc
 begin
     variable LVar0 := 0;
     script_overrides;
-    set_global_var(151, 1);
-    set_global_var(154, 1);
-    if (cur_map_index == 28) then begin
-        if (elevation(dude_obj) == 0) then begin
+    set_global_var(GVAR_SIERRA_BASE_ALERT, 1);
+    set_global_var(GVAR_SIERRA_BASE_SECURITY, 1);
+    if (CUR_MAP_DEPOLVA) then begin
+        if (dude_elevation == 0) then begin
             set_map_var(25, 1);
         end
         else begin
-            if (elevation(dude_obj) == 1) then begin
+            if (dude_elevation == 1) then begin
                 set_map_var(26, 1);
             end
             else begin
                 set_map_var(27, 1);
             end
         end
-        display_msg(message_str(895, 101));
-        display_msg(message_str(895, 102));
-        LVar0 := 1 + elevation(dude_obj);
-        display_msg(message_str(895, 103) + LVar0 + ".");
+        display_mstr(101);
+        display_mstr(102);
+        LVar0 := 1 + dude_elevation;
+        display_msg(mstr(103) + LVar0 + ".");
     end
     else begin
         set_map_var(2, 1);
-        display_msg(message_str(895, 101));
-        display_msg(message_str(895, 102));
-        display_msg(message_str(895, 105));
+        display_mstr(101);
+        display_mstr(102);
+        display_mstr(105);
     end
 end
 
@@ -172,21 +172,21 @@ begin
     variable LVar1 := 0;
     LVar0 := obj_pid(obj_being_used_with);
     script_overrides;
-    if (cur_map_index == 28) then begin
+    if (CUR_MAP_DEPOLVA) then begin
         if ((LVar0 == 282) or (LVar0 == 281) and (tile_num(self_obj) != 19916)) then begin
             set_map_var(49, elevation(self_obj));
-            LVar1 := 1 + elevation(dude_obj);
-            display_msg(message_str(895, 100) + LVar1 + ".");
+            LVar1 := 1 + dude_elevation;
+            display_msg(mstr(100) + LVar1 + ".");
         end
         else begin
             if ((tile_num(self_obj) == 19916) and (LVar0 == 282)) then begin
                 set_map_var(48, elevation(self_obj));
-                LVar1 := 1 + elevation(dude_obj);
-                display_msg(message_str(895, 100) + LVar1 + ".");
+                LVar1 := 1 + dude_elevation;
+                display_msg(mstr(100) + LVar1 + ".");
             end
             else begin
                 if ((tile_num(self_obj) == 19916) and (LVar0 == 281)) then begin
-                    display_msg(message_str(895, 101));
+                    display_mstr(101);
                 end
             end
         end
@@ -194,13 +194,12 @@ begin
     else begin
         if (LVar0 == 282) then begin
             set_map_var(1, elevation(self_obj));
-            display_msg(message_str(895, 104));
+            display_mstr(104);
         end
         else begin
             if (LVar0 == 281) then begin
-                display_msg(message_str(895, 101));
+                display_mstr(101);
             end
         end
     end
 end
-
