diff --git a/Games/Fallout2.Steam/Original/WSPWRDWN.ssl b/Games/Fallout2.Steam/ReDefine/WSPWRDWN.ssl
index 0ca7081..b11adf1 100644
--- a/Games/Fallout2.Steam/Original/WSPWRDWN.ssl
+++ b/Games/Fallout2.Steam/ReDefine/WSPWRDWN.ssl
@@ -137,17 +137,16 @@ end
 procedure use_p_proc
 begin
     script_overrides;
-    if (global_var(153) == 0) then begin
-        set_global_var(153, 1);
-        display_msg(message_str(14, 912));
+    if (global_var(GVAR_SIERRA_BASE_POWER) == 0) then begin
+        set_global_var(GVAR_SIERRA_BASE_POWER, 1);
+        display_msg(g_mstr(912));
         play_sfx("pwrswtch");
         animate_stand_obj(self_obj);
     end
     else begin
-        set_global_var(153, 0);
-        display_msg(message_str(14, 913));
+        set_global_var(GVAR_SIERRA_BASE_POWER, 0);
+        display_msg(g_mstr(913));
         play_sfx("pwrswtch");
         animate_stand_reverse_obj(self_obj);
     end
 end
-
