diff --git a/Games/Fallout2.Steam/Original/WSTERM3C.ssl b/Games/Fallout2.Steam/ReDefine/WSTERM3C.ssl
index ec99ce6..448efeb 100644
--- a/Games/Fallout2.Steam/Original/WSTERM3C.ssl
+++ b/Games/Fallout2.Steam/ReDefine/WSTERM3C.ssl
@@ -146,12 +146,12 @@ end
 procedure use_p_proc
 begin
     variable LVar0 := 0;
-    if (get_critter_stat(source_obj, 4) < 4) then begin
+    if (get_critter_stat(source_obj, STAT_iq) < 4) then begin
         if (source_obj == dude_obj) then begin
-            display_msg(message_str(161, 102));
+            display_mstr(102);
         end
         else begin
-            display_msg(obj_name(source_obj) + message_str(161, 200));
+            display_msg(obj_name(source_obj) + mstr(200));
         end
     end
     else begin
@@ -180,13 +180,13 @@ end
 procedure look_p_proc
 begin
     script_overrides;
-    display_msg(message_str(161, 100));
+    display_mstr(100);
 end
 
 procedure description_p_proc
 begin
     script_overrides;
-    display_msg(message_str(161, 101));
+    display_mstr(101);
 end
 
 procedure push_p_proc
@@ -195,16 +195,16 @@ end
 
 procedure Node001
 begin
-    gsay_reply(161, message_str(161, 103) + message_str(161, 104));
-    giq_option(4, 161, 105, Node010, 50);
-    giq_option(4, 161, 106, Node011, 50);
-    giq_option(4, 161, 107, Node999, 50);
+    Reply(mstr(103) + mstr(104));
+    giq_option(4, NAME, 105, Node010, NEUTRAL_REACTION);
+    giq_option(4, NAME, 106, Node011, NEUTRAL_REACTION);
+    giq_option(4, NAME, 107, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node002
 begin
-    gsay_reply(161, 110);
-    giq_option(4, 161, 111, Node999, 50);
+    Reply(110);
+    giq_option(4, NAME, 111, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node010
@@ -212,41 +212,41 @@ begin
     if (map_var(72) == 1) then begin
         if (map_var(73) == 1) then begin
             if (map_var(74) == 1) then begin
-                gsay_reply(161, 113);
+                Reply(113);
             end
             else begin
-                gsay_reply(161, 117);
+                Reply(117);
             end
         end
         else begin
             if (map_var(74) == 1) then begin
-                gsay_reply(161, 115);
+                Reply(115);
             end
             else begin
-                gsay_reply(161, 119);
+                Reply(119);
             end
         end
     end
     else begin
         if (map_var(73) == 1) then begin
             if (map_var(74) == 1) then begin
-                gsay_reply(161, 114);
+                Reply(114);
             end
             else begin
-                gsay_reply(161, 118);
+                Reply(118);
             end
         end
         else begin
             if (map_var(74) == 1) then begin
-                gsay_reply(161, 116);
+                Reply(116);
             end
             else begin
-                gsay_reply(161, 120);
+                Reply(120);
             end
         end
     end
-    giq_option(4, 161, 153, Node011, 50);
-    giq_option(4, 161, 154, Node999, 50);
+    giq_option(4, NAME, 153, Node011, NEUTRAL_REACTION);
+    giq_option(4, NAME, 154, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node011
@@ -256,7 +256,7 @@ begin
     set_local_var(0, 1);
     if ((map_var(73) == 0) or (map_var(74) == 0) or (map_var(72) == 0)) then begin
         set_map_var(75, 1);
-        gsay_reply(161, message_str(161, 155) + message_str(161, 156));
+        Reply(mstr(155) + mstr(156));
     end
     else begin
         gfade_out(600);
@@ -264,52 +264,51 @@ begin
         if (tile_contains_obj_pid(8908, elevation(self_obj), 33555662)) then begin
             LVar0 := tile_contains_pid_obj(8908, elevation(self_obj), 33555662);
             destroy_object(LVar0);
-            if (global_var(155) == 4) then begin
-                LVar1 := create_object_sid(16777295, 8908, elevation(self_obj), 167);
+            if (global_var(GVAR_BRAIN_BOT_BRAIN) == 4) then begin
+                LVar1 := create_object_sid(16777295, 8908, elevation(self_obj), SCRIPT_WCBRNBOT);
                 debug_msg("Made Cybernetic Robobrain " + obj_pid(LVar1));
             end
             else begin
-                if (global_var(155) == 3) then begin
-                    LVar1 := create_object_sid(16777597, 8908, elevation(self_obj), 167);
+                if (global_var(GVAR_BRAIN_BOT_BRAIN) == 3) then begin
+                    LVar1 := create_object_sid(16777597, 8908, elevation(self_obj), SCRIPT_WCBRNBOT);
                     debug_msg("Made Human Robobrain " + obj_pid(LVar1));
                 end
                 else begin
-                    if (global_var(155) == 2) then begin
-                        LVar1 := create_object_sid(16777595, 8908, elevation(self_obj), 167);
+                    if (global_var(GVAR_BRAIN_BOT_BRAIN) == 2) then begin
+                        LVar1 := create_object_sid(16777595, 8908, elevation(self_obj), SCRIPT_WCBRNBOT);
                         debug_msg("Made Chimp Robobrain " + obj_pid(LVar1));
                     end
                     else begin
-                        if (global_var(155) == 1) then begin
-                            LVar1 := create_object_sid(16777596, 8908, elevation(self_obj), 167);
+                        if (global_var(GVAR_BRAIN_BOT_BRAIN) == 1) then begin
+                            LVar1 := create_object_sid(16777596, 8908, elevation(self_obj), SCRIPT_WCBRNBOT);
                             debug_msg("Made Abnormal Robobrain " + obj_pid(LVar1));
                         end
                     end
                 end
             end
         end
-        if (global_var(155) == 4) then begin
-            gsay_reply(161, 500);
+        if (global_var(GVAR_BRAIN_BOT_BRAIN) == 4) then begin
+            Reply(500);
         end
         else begin
-            if (global_var(155) == 3) then begin
-                gsay_reply(161, message_str(161, 164) + message_str(161, 165) + message_str(161, 166) + message_str(161, 167));
+            if (global_var(GVAR_BRAIN_BOT_BRAIN) == 3) then begin
+                Reply(mstr(164) + mstr(165) + mstr(166) + mstr(167));
             end
             else begin
-                if (global_var(155) == 2) then begin
-                    gsay_reply(161, message_str(161, 164) + message_str(161, 165) + message_str(161, 170) + message_str(161, 167));
+                if (global_var(GVAR_BRAIN_BOT_BRAIN) == 2) then begin
+                    Reply(mstr(164) + mstr(165) + mstr(170) + mstr(167));
                 end
                 else begin
-                    if (global_var(155) == 1) then begin
-                        gsay_reply(161, message_str(161, 164) + message_str(161, 165) + message_str(161, 174) + message_str(161, 167));
+                    if (global_var(GVAR_BRAIN_BOT_BRAIN) == 1) then begin
+                        Reply(mstr(164) + mstr(165) + mstr(174) + mstr(167));
                     end
                 end
             end
         end
     end
-    giq_option(4, 161, 176, Node999, 50);
+    giq_option(4, NAME, 176, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node999
 begin
 end
-
