diff --git a/Games/Fallout2.Steam/Original/ZCLRAT.ssl b/Games/Fallout2.Steam/ReDefine/ZCLRAT.ssl
index 61066b4..c7584ac 100644
--- a/Games/Fallout2.Steam/Original/ZCLRAT.ssl
+++ b/Games/Fallout2.Steam/ReDefine/ZCLRAT.ssl
@@ -145,7 +145,7 @@ end
 procedure critter_p_proc
 begin
     if (obj_can_see_obj(self_obj, dude_obj)) then begin
-        attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+        attack(dude_obj);
     end
 end
 
@@ -171,14 +171,14 @@ end
 procedure damage_p_proc
 begin
     if (source_obj == dude_obj) then begin
-        attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+        attack(dude_obj);
     end
 end
 
 procedure map_enter_p_proc
 begin
-    critter_add_trait(self_obj, 1, 6, 3);
-    critter_add_trait(self_obj, 1, 5, 7);
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 3);
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_RAT);
     add_timer_event(self_obj, game_ticks(random(1, 5)), 0);
 end
 
@@ -196,4 +196,3 @@ end
 procedure use_obj_on_p_proc
 begin
 end
-
