diff --git a/Games/Fallout2.Steam/Original/ZCLSCORP.ssl b/Games/Fallout2.Steam/ReDefine/ZCLSCORP.ssl
index 29a7008..816d113 100644
--- a/Games/Fallout2.Steam/Original/ZCLSCORP.ssl
+++ b/Games/Fallout2.Steam/ReDefine/ZCLSCORP.ssl
@@ -146,21 +146,21 @@ end
 procedure critter_p_proc
 begin
     if (obj_can_see_obj(self_obj, dude_obj)) then begin
-        attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+        attack(dude_obj);
     end
 end
 
 procedure destroy_p_proc
 begin
     variable LVar0 := 0;
-    LVar0 := create_object_sid(92, 0, 0, -1);
+    LVar0 := create_object(92, 0, 0);
     add_obj_to_inven(self_obj, LVar0);
 end
 
 procedure look_at_p_proc
 begin
     script_overrides;
-    display_msg(message_str(20, 100));
+    display_mstr(100);
 end
 
 procedure description_p_proc
@@ -177,14 +177,14 @@ end
 procedure damage_p_proc
 begin
     if (source_obj == dude_obj) then begin
-        attack_complex(dude_obj, 0, 1, 0, 0, 30000, 0, 0);
+        attack(dude_obj);
     end
 end
 
 procedure map_enter_p_proc
 begin
-    critter_add_trait(self_obj, 1, 6, 4);
-    critter_add_trait(self_obj, 1, 5, 8);
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 4);
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_SCORPION);
     add_timer_event(self_obj, game_ticks(random(1, 5)), 0);
 end
 
@@ -193,7 +193,7 @@ begin
     if ((combat_is_initialized == 0) and ((critter_state(self_obj) bwand 1) == 0)) then begin
         reg_anim_func(2, self_obj);
         reg_anim_func(1, 1);
-        if ((cur_map_index == 3) and (elevation(dude_obj) != elevation(self_obj))) then begin
+        if ((CUR_MAP_ARCAVES) and (dude_elevation != elevation(self_obj))) then begin
             debug_msg("I'm in Arroyo Caves and not on the same elevation as the dude.");
         end
         else begin
@@ -218,4 +218,3 @@ end
 procedure use_obj_on_p_proc
 begin
 end
-
