diff --git a/Games/Fallout2.Steam/Original/ZI8BALL.ssl b/Games/Fallout2.Steam/ReDefine/ZI8BALL.ssl
index 4755caa..4fedc18 100644
--- a/Games/Fallout2.Steam/Original/ZI8BALL.ssl
+++ b/Games/Fallout2.Steam/ReDefine/ZI8BALL.ssl
@@ -143,33 +143,33 @@ end
 procedure use_p_proc
 begin
     variable LVar0 := 0;
-    if (get_critter_stat(dude_obj, 6) <= 3) then begin
+    if (dude_luck <= 3) then begin
         LVar0 := random(199, 228);
     end
     else begin
-        if (get_critter_stat(dude_obj, 6) < 9) then begin
+        if (dude_luck < 9) then begin
             LVar0 := random(200, 228);
         end
         else begin
-            if (get_critter_stat(dude_obj, 6) >= 9) then begin
+            if (dude_luck >= 9) then begin
                 LVar0 := random(200, 243);
             end
         end
     end
     if (LVar0 == 232) then begin
-        set_global_var(243, 1);
+        set_global_var(GVAR_8_BALL_TOILET_SECRET, 1);
     end
     else begin
         if (LVar0 == 233) then begin
-            set_global_var(244, 1);
+            set_global_var(GVAR_8_BALL_TRASH_SECRET, 1);
         end
         else begin
             if (LVar0 == 243) then begin
-                set_global_var(599, 1);
+                set_global_var(GVAR_8_BALL_VAULT_TERMINAL, 1);
             end
         end
     end
-    display_msg(message_str(412, LVar0));
+    display_mstr(LVar0);
 end
 
 procedure timed_event_p_proc
@@ -195,4 +195,3 @@ end
 procedure description_p_proc
 begin
 end
-
