diff --git a/Games/Fallout2.Steam/Original/ZSELEV01.ssl b/Games/Fallout2.Steam/ReDefine/ZSELEV01.ssl
index e5193f9..34f01d9 100644
--- a/Games/Fallout2.Steam/Original/ZSELEV01.ssl
+++ b/Games/Fallout2.Steam/ReDefine/ZSELEV01.ssl
@@ -137,13 +137,12 @@ end
 procedure spatial_p_proc
 begin
     if (source_obj == dude_obj) then begin
-        debug_msg("cur_town == " + metarule(46, 0) + " / Generator_Fixed == " + ((global_var(215) bwand 1) != 0));
-        if ((metarule(46, 0) == 12) and (((global_var(215) bwand 1) != 0) == 0)) then begin
+        debug_msg("cur_town == " + cur_town + " / Generator_Fixed == " + ((global_var(GVAR_MILITARY_BASE_FLAGS) bwand 1) != 0));
+        if ((cur_town == 12) and (((global_var(GVAR_MILITARY_BASE_FLAGS) bwand 1) != 0) == 0)) then begin
             debug_msg("ELEVATOR IS OUT OF ACTION");
         end
         else begin
-            metarule(15, -1);
+            elevator(-1);
         end
     end
 end
-
