diff --git a/Mods/Olympus2207/Original/BLGNRTR.ssl b/Mods/Olympus2207/ReDefine/BLGNRTR.ssl
index 24006f4..d50f9a6 100644
--- a/Mods/Olympus2207/Original/BLGNRTR.ssl
+++ b/Mods/Olympus2207/ReDefine/BLGNRTR.ssl
@@ -28,15 +28,15 @@ procedure description_p_proc
 begin
     script_overrides;
     if (local_var(1) == 0) then begin
-        if ((has_skill(dude_obj, 12) >= 50) or (has_skill(dude_obj, 13) >= 40)) then begin
-            display_msg(message_str(525, 3));
+        if ((dude_skill(SKILL_SCIENCE) >= 50) or (dude_skill(SKILL_REPAIR) >= 40)) then begin
+            display_mstr(3);
         end
         else begin
-            display_msg(message_str(525, 4));
+            display_mstr(4);
         end
     end
     else begin
-        display_msg(message_str(525, 5));
+        display_mstr(5);
     end
 end
 
@@ -44,10 +44,10 @@ procedure look_at_p_proc
 begin
     script_overrides;
     if (local_var(1) == 0) then begin
-        display_msg(message_str(525, 1));
+        display_mstr(1);
     end
     else begin
-        display_msg(message_str(525, 2));
+        display_mstr(2);
     end
 end
 
@@ -58,12 +58,12 @@ end
 procedure use_p_proc
 begin
     if (local_var(1) == 0) then begin
-        if (random(0, 100) <= (35 - floor((((get_critter_stat(dude_obj, 6) - 4) * 4) - (difficulty_level * 4)) / 3))) then begin
-            critter_dmg(dude_obj, random(5, 10), 4);
-            display_msg(message_str(525, 10));
+        if (random(0, 100) <= (35 - floor((((dude_luck - 4) * 4) - (difficulty_level * 4)) / 3))) then begin
+            critter_dmg(dude_obj, random(5, 10), DMG_electrical);
+            display_mstr(10);
         end
         else begin
-            display_msg(message_str(525, 11));
+            display_mstr(11);
         end
     end
 end
@@ -74,13 +74,13 @@ begin
         script_overrides;
         if ((obj_pid(obj_being_used_with) == 57) or (obj_pid(obj_being_used_with) == 58)) then begin
             set_local_var(1, 1);
-            display_msg(message_str(525, 12));
-            give_exp_points(floor(50 / 5) + (get_critter_stat(dude_obj, 4) * floor(50 / 12)) + (get_critter_stat(dude_obj, 6) * floor(50 / 30)) + (floor(50 / 75) * (((get_critter_stat(dude_obj, 6) - 4) * 4) - (difficulty_level * 4))));
-            display_msg(message_str(10, random(140, 143)) + (floor(50 / 5) + (get_critter_stat(dude_obj, 4) * floor(50 / 12)) + (get_critter_stat(dude_obj, 6) * floor(50 / 30)) + (floor(50 / 75) * (((get_critter_stat(dude_obj, 6) - 4) * 4) - (difficulty_level * 4)))) + message_str(10, 144));
+            display_mstr(12);
+            give_exp_points(floor(50 / 5) + (dude_iq * floor(50 / 12)) + (dude_luck * floor(50 / 30)) + (floor(50 / 75) * (((dude_luck - 4) * 4) - (difficulty_level * 4))));
+            display_msg(message_str(SCRIPT_STTEXT, random(140, 143)) + (floor(50 / 5) + (dude_iq * floor(50 / 12)) + (dude_luck * floor(50 / 30)) + (floor(50 / 75) * (((dude_luck - 4) * 4) - (difficulty_level * 4)))) + message_str(SCRIPT_STTEXT, 144));
             call turn_on();
         end
         else begin
-            display_msg(message_str(10, random(60, 69)));
+            display_msg(message_str(SCRIPT_STTEXT, random(60, 69)));
         end
     end
 end
@@ -90,25 +90,25 @@ begin
     if (local_var(1) == 0) then begin
         script_overrides;
         if ((action_being_used == 13) or (action_being_used == 12)) then begin
-            if (has_skill(dude_obj, action_being_used) >= 75) then begin
+            if (dude_skill(action_being_used) >= 75) then begin
                 set_local_var(1, 1);
-                display_msg(message_str(525, 15));
-                give_exp_points(floor(50 / 5) + (get_critter_stat(dude_obj, 4) * floor(50 / 12)) + (get_critter_stat(dude_obj, 6) * floor(50 / 30)) + (floor(50 / 75) * (((get_critter_stat(dude_obj, 6) - 4) * 4) - (difficulty_level * 4))));
-                display_msg(message_str(10, random(140, 143)) + (floor(50 / 5) + (get_critter_stat(dude_obj, 4) * floor(50 / 12)) + (get_critter_stat(dude_obj, 6) * floor(50 / 30)) + (floor(50 / 75) * (((get_critter_stat(dude_obj, 6) - 4) * 4) - (difficulty_level * 4)))) + message_str(10, 144));
+                display_mstr(15);
+                give_exp_points(floor(50 / 5) + (dude_iq * floor(50 / 12)) + (dude_luck * floor(50 / 30)) + (floor(50 / 75) * (((dude_luck - 4) * 4) - (difficulty_level * 4))));
+                display_msg(message_str(SCRIPT_STTEXT, random(140, 143)) + (floor(50 / 5) + (dude_iq * floor(50 / 12)) + (dude_luck * floor(50 / 30)) + (floor(50 / 75) * (((dude_luck - 4) * 4) - (difficulty_level * 4)))) + message_str(SCRIPT_STTEXT, 144));
                 call turn_on();
             end
             else begin
-                if (has_skill(dude_obj, action_being_used) >= 55) then begin
+                if (dude_skill(action_being_used) >= 55) then begin
                     set_local_var(1, 1);
-                    critter_dmg(dude_obj, random(10, 15), 4);
-                    display_msg(message_str(525, 16));
-                    give_exp_points(floor(25 / 5) + (get_critter_stat(dude_obj, 4) * floor(25 / 12)) + (get_critter_stat(dude_obj, 6) * floor(25 / 30)) + (floor(25 / 75) * (((get_critter_stat(dude_obj, 6) - 4) * 4) - (difficulty_level * 4))));
-                    display_msg(message_str(10, random(140, 143)) + (floor(25 / 5) + (get_critter_stat(dude_obj, 4) * floor(25 / 12)) + (get_critter_stat(dude_obj, 6) * floor(25 / 30)) + (floor(25 / 75) * (((get_critter_stat(dude_obj, 6) - 4) * 4) - (difficulty_level * 4)))) + message_str(10, 144));
+                    critter_dmg(dude_obj, random(10, 15), DMG_electrical);
+                    display_mstr(16);
+                    give_exp_points(floor(25 / 5) + (dude_iq * floor(25 / 12)) + (dude_luck * floor(25 / 30)) + (floor(25 / 75) * (((dude_luck - 4) * 4) - (difficulty_level * 4))));
+                    display_msg(message_str(SCRIPT_STTEXT, random(140, 143)) + (floor(25 / 5) + (dude_iq * floor(25 / 12)) + (dude_luck * floor(25 / 30)) + (floor(25 / 75) * (((dude_luck - 4) * 4) - (difficulty_level * 4)))) + message_str(SCRIPT_STTEXT, 144));
                     call turn_on();
                 end
                 else begin
-                    critter_dmg(dude_obj, random(3, 5), 4);
-                    display_msg(message_str(525, 14));
+                    critter_dmg(dude_obj, random(3, 5), DMG_electrical);
+                    display_mstr(14);
                 end
             end
         end
@@ -129,22 +129,21 @@ end
 
 procedure turn_on
 begin
-    if (global_var(437) == 0) then begin
-        set_global_var(437, 1);
-        create_object_sid(33555079, 17302, 0, -1);
-        create_object_sid(33555079, 22130, 1, -1);
-        create_object_sid(3355471, 11521, 0, -1);
-        create_object_sid(3355471, 11314, 0, -1);
-        create_object_sid(3355471, 12705, 0, -1);
-        create_object_sid(3355471, 13298, 0, -1);
-        create_object_sid(3355471, 14906, 0, -1);
-        create_object_sid(3355471, 15117, 0, -1);
-        create_object_sid(3355471, 14122, 0, -1);
-        create_object_sid(3355471, 17919, 0, -1);
-        create_object_sid(3355471, 17706, 0, -1);
-        create_object_sid(3355471, 20720, 0, -1);
-        create_object_sid(3355471, 20308, 0, -1);
-        create_object_sid(3355471, 20700, 0, -1);
+    if (global_var(GVAR_DOC_JOHNSON_STATUS) == 0) then begin
+        set_global_var(GVAR_DOC_JOHNSON_STATUS, 1);
+        create_object(33555079, 17302, 0);
+        create_object(33555079, 22130, 1);
+        create_object(3355471, 11521, 0);
+        create_object(3355471, 11314, 0);
+        create_object(3355471, 12705, 0);
+        create_object(3355471, 13298, 0);
+        create_object(3355471, 14906, 0);
+        create_object(3355471, 15117, 0);
+        create_object(3355471, 14122, 0);
+        create_object(3355471, 17919, 0);
+        create_object(3355471, 17706, 0);
+        create_object(3355471, 20720, 0);
+        create_object(3355471, 20308, 0);
+        create_object(3355471, 20700, 0);
     end
 end
-
