diff --git a/Mods/Olympus2207/Original/DISK9.ssl b/Mods/Olympus2207/ReDefine/DISK9.ssl
index cb9b2f2..9f509da 100644
--- a/Mods/Olympus2207/Original/DISK9.ssl
+++ b/Mods/Olympus2207/ReDefine/DISK9.ssl
@@ -37,12 +37,12 @@ end
 
 procedure use_p_proc
 begin
-    if (global_var(111) == 0) then begin
+    if (global_var(GVAR_MODOC_VERMIN_HUNTER_SEED) == 0) then begin
         script_overrides;
-        mark_area_known(0, 16, 1);
-        set_global_var(111, 1);
-        display_msg(message_str(10, 850));
-        display_msg(message_str(10, 851));
+        mark_area_known(MARK_TYPE_TOWN, 16, MARK_STATE_KNOWN);
+        set_global_var(GVAR_MODOC_VERMIN_HUNTER_SEED, 1);
+        display_msg(message_str(SCRIPT_STTEXT, 850));
+        display_msg(message_str(SCRIPT_STTEXT, 851));
     end
 end
 
@@ -65,4 +65,3 @@ end
 procedure pickup_p_proc
 begin
 end
-
