diff --git a/Mods/Olympus2207/Original/FRBEAST.ssl b/Mods/Olympus2207/ReDefine/FRBEAST.ssl
index 61d31c3..14e0466 100644
--- a/Mods/Olympus2207/Original/FRBEAST.ssl
+++ b/Mods/Olympus2207/ReDefine/FRBEAST.ssl
@@ -33,11 +33,11 @@ begin
         set_local_var(3, tile_num(self_obj));
         set_local_var(1, 1);
         if ((obj_pid(self_obj) == 16777428) or (obj_pid(self_obj) == 16777429)) then begin
-            critter_add_trait(self_obj, 1, 6, 209);
-            critter_add_trait(self_obj, 1, 5, 67);
-            add_obj_to_inven(self_obj, create_object_sid(524, 1, 2, -1));
-            add_obj_to_inven(self_obj, create_object_sid(523, 1, 2, -1));
-            wield_obj_critter(self_obj, obj_carrying_pid_obj(self_obj, 523));
+            critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 209);
+            critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_GENERIC_BEASTIE);
+            add_obj_to_inven(self_obj, create_object(524, 1, 2));
+            add_obj_to_inven(self_obj, create_object(523, 1, 2));
+            wield_obj(obj_carrying_pid_obj(self_obj, 523));
         end
         max_distance := random(4, 8);
     end
@@ -90,19 +90,19 @@ begin
     if (source_obj == dude_obj) then begin
     end
     if (not(tile_contains_obj_pid(2, 2, 250))) then begin
-        stuff_container := create_object_sid(250, 2, 2, -1);
+        stuff_container := create_object(250, 2, 2);
     end
     else begin
         stuff_container := tile_contains_pid_obj(2, 2, 250);
     end
     move_obj_inven_to_obj(self_obj, stuff_container);
-    set_global_var(226, global_var(226) + 1);
-    if (global_var(226) == 4) then begin
+    set_global_var(GVAR_NCR_GATE_NIGHT, global_var(GVAR_NCR_GATE_NIGHT) + 1);
+    if (global_var(GVAR_NCR_GATE_NIGHT) == 4) then begin
         play_sfx("LEVELUP");
-        display_msg(message_str(10, 861));
+        display_msg(message_str(SCRIPT_STTEXT, 861));
     end
     else begin
-        display_msg(message_str(10, 864));
+        display_msg(message_str(SCRIPT_STTEXT, 864));
     end
 end
 
@@ -122,7 +122,7 @@ end
 
 procedure map_enter_p_proc
 begin
-    if ((global_var(183) > 0) and (not(metarule(22, 0)) == 1) and (local_var(7) == 0)) then begin
+    if ((global_var(GVAR_ARROYO_SPEAR) > 0) and (not(is_loading_game) == 1) and (local_var(7) == 0)) then begin
         move_to(self_obj, local_var(3), 0);
         set_local_var(7, 1);
     end
@@ -143,4 +143,3 @@ begin
         animation_is_called := 0;
     end
 end
-
