diff --git a/Mods/Olympus2207/Original/GL_HIGHLIGHTING.ssl b/Mods/Olympus2207/ReDefine/GL_HIGHLIGHTING.ssl
index 4323cb0..31a96b4 100644
--- a/Mods/Olympus2207/Original/GL_HIGHLIGHTING.ssl
+++ b/Mods/Olympus2207/ReDefine/GL_HIGHLIGHTING.ssl
@@ -209,7 +209,7 @@ begin
         LVar6 := array_key(LVar3, LVar5);
         LVar1 := get_array(LVar3, LVar6);
         if (LVar1 != sfall_func0("outlined_object")) then begin
-            if (elevation(LVar1) != elevation(dude_obj)) then begin
+            if (elevation(LVar1) != dude_elevation) then begin
                 continue;
             end
             LVar2 := alsoContainer andAlso (obj_item_subtype(LVar1) == 1);
@@ -226,7 +226,7 @@ begin
             LVar10 := array_key(LVar7, LVar9);
             LVar1 := get_array(LVar7, LVar10);
             if ((critter_state(LVar1) == 1) andAlso not(get_proto_data(obj_pid(LVar1), 32) bwand 32)) then begin
-                if (elevation(LVar1) != elevation(dude_obj)) then begin
+                if (elevation(LVar1) != dude_elevation) then begin
                     continue;
                 end
                 call ToggleHighlightObject(LVar1, arg0, 0);
@@ -298,8 +298,8 @@ begin
     variable LVar2 := 1;
     if (tile_distance_objs(dude_obj, arg0) > 2) then begin
         LVar1 := tile_num(arg0);
-        if (obj_blocking_tile(LVar1, elevation(dude_obj), 3)) then begin
-            LVar1 := tile_num_in_direction(LVar1, rotation_to_tile(LVar1, tile_num(dude_obj)), 1);
+        if (obj_blocking_tile(LVar1, dude_elevation, 3)) then begin
+            LVar1 := tile_num_in_direction(LVar1, rotation_to_tile(LVar1, dude_tile), 1);
         end
         LVar2 := obj_blocking_line(dude_obj, LVar1, 3) == 0;
     end
@@ -322,4 +322,3 @@ begin
         sfall_func2("set_outline", get_sfall_arg, 0);
     end
 end
-
