diff --git a/Mods/Olympus2207/Original/GL_NPCARMOR.ssl b/Mods/Olympus2207/ReDefine/GL_NPCARMOR.ssl
index 5737400..bbaeaf2 100644
--- a/Mods/Olympus2207/Original/GL_NPCARMOR.ssl
+++ b/Mods/Olympus2207/ReDefine/GL_NPCARMOR.ssl
@@ -45,10 +45,10 @@ begin
         if (LVar0 == dude_obj) then begin
             continue;
         end
-        LVar1 := critter_inven_obj(LVar0, 0);
+        LVar1 := critter_inven_obj(LVar0, INVEN_TYPE_WORN);
         if (LVar1) then begin
             if (simpleMode) then begin
-                if (get_critter_stat(LVar0, 34) == 1) then begin
+                if (get_critter_stat(LVar0, STAT_gender) == 1) then begin
                     LVar2 := get_proto_data(obj_pid(LVar1), 104);
                 end
                 else begin
@@ -61,7 +61,7 @@ begin
                     continue;
                 end
             end
-            metarule3(107, LVar0, LVar2, 0);
+            art_change_fid_num(LVar0, LVar2);
         end
     end
 end
@@ -134,7 +134,7 @@ begin
     variable LVar1 := 0;
     variable LVar2 := 0;
     variable LVar3 := 0;
-    LVar1 := inven_cmds(arg0, 13, 0);
+    LVar1 := inven_ptr(arg0, 0);
     while (LVar1) do begin
         if (obj_item_subtype(LVar1) == 3) then begin
             if ((unWieldWeapon == 0) orElse (unWieldWeapon != obj_pid(LVar1))) then begin
@@ -148,7 +148,7 @@ begin
             end
         end
         LVar3 := LVar3 + 1;
-        LVar1 := inven_cmds(arg0, 13, LVar3);
+        LVar1 := inven_ptr(arg0, LVar3);
     end
     unWieldWeapon := 0;
 end
@@ -189,7 +189,7 @@ begin
     if (LVar1 andAlso (LVar3 == 0)) then begin
         if (simpleMode) then begin
             if (LVar4) then begin
-                if (get_critter_stat(LVar0, 34) == 1) then begin
+                if (get_critter_stat(LVar0, STAT_gender) == 1) then begin
                     LVar2 := get_proto_data(obj_pid(LVar1), 104);
                 end
                 else begin
@@ -199,13 +199,13 @@ begin
             else begin
                 LVar2 := get_proto_data(obj_pid(LVar0), 8);
             end
-            metarule3(107, LVar0, LVar2, 0);
+            art_change_fid_num(LVar0, LVar2);
             return 0;
         end
         LVar2 := check_armor_change(LVar0, LVar1, LVar4);
         if (LVar2 != -1) then begin
             if (art_exists(LVar2)) then begin
-                metarule3(107, LVar0, LVar2, 0);
+                art_change_fid_num(LVar0, LVar2);
             end
             else begin
                 debug_msg("[Error] NPC armor mod. Invenwield missing FID: " + LVar2);
@@ -238,10 +238,10 @@ begin
         if ((LVar3 bwand 251658240) != 16777216) then begin
             return 0;
         end
-        LVar1 := critter_inven_obj(dude_obj, 0);
+        LVar1 := critter_inven_obj(dude_obj, INVEN_TYPE_WORN);
         if (simpleMode) then begin
             if (LVar1) then begin
-                if (get_critter_stat(dude_obj, 34) == 1) then begin
+                if (dude_gender == 1) then begin
                     LVar0 := get_proto_data(obj_pid(LVar1), 104);
                 end
                 else begin
@@ -377,4 +377,3 @@ begin
         debug_msg("NPC appearance armor mod: Done.");
     end
 end
-
