diff --git a/Mods/Olympus2207/Original/ITRADMTR.ssl b/Mods/Olympus2207/ReDefine/ITRADMTR.ssl
index c929cc8..9635409 100644
--- a/Mods/Olympus2207/Original/ITRADMTR.ssl
+++ b/Mods/Olympus2207/ReDefine/ITRADMTR.ssl
@@ -38,22 +38,22 @@ end
 procedure use_p_proc
 begin
     if ((get_game_mode bwand 4096) != 0) then begin
-        display_msg(message_str(12, random(10, 14)));
+        display_msg(message_str(SCRIPT_ITTEXT, random(10, 14)));
     end
     else begin
-        if (get_critter_stat(dude_obj, 37) == 0) then begin
-            display_msg(message_str(12, 160));
+        if (get_critter_stat(dude_obj, STAT_current_rad) == 0) then begin
+            display_msg(message_str(SCRIPT_ITTEXT, 160));
         end
         else begin
-            if ((get_critter_stat(dude_obj, 37) > 0) and (get_critter_stat(dude_obj, 37) <= 200)) then begin
-                display_msg(message_str(12, 161) + get_critter_stat(dude_obj, 37) + message_str(12, 164));
+            if ((get_critter_stat(dude_obj, STAT_current_rad) > 0) and (get_critter_stat(dude_obj, STAT_current_rad) <= 200)) then begin
+                display_msg(message_str(SCRIPT_ITTEXT, 161) + get_critter_stat(dude_obj, STAT_current_rad) + message_str(SCRIPT_ITTEXT, 164));
             end
             else begin
-                if ((get_critter_stat(dude_obj, 37) > 200) and (get_critter_stat(dude_obj, 37) <= 400)) then begin
-                    display_msg(message_str(12, 162) + get_critter_stat(dude_obj, 37) + message_str(12, 164));
+                if ((get_critter_stat(dude_obj, STAT_current_rad) > 200) and (get_critter_stat(dude_obj, STAT_current_rad) <= 400)) then begin
+                    display_msg(message_str(SCRIPT_ITTEXT, 162) + get_critter_stat(dude_obj, STAT_current_rad) + message_str(SCRIPT_ITTEXT, 164));
                 end
                 else begin
-                    display_msg(message_str(12, 163) + get_critter_stat(dude_obj, 37) + message_str(12, 164));
+                    display_msg(message_str(SCRIPT_ITTEXT, 163) + get_critter_stat(dude_obj, STAT_current_rad) + message_str(SCRIPT_ITTEXT, 164));
                 end
             end
         end
@@ -79,4 +79,3 @@ end
 procedure pickup_p_proc
 begin
 end
-
