diff --git a/Mods/Olympus2207/Original/LVESHTON.ssl b/Mods/Olympus2207/ReDefine/LVESHTON.ssl
index 9717961..5dc65ac 100644
--- a/Mods/Olympus2207/Original/LVESHTON.ssl
+++ b/Mods/Olympus2207/ReDefine/LVESHTON.ssl
@@ -63,52 +63,52 @@ begin
         set_local_var(4, random(70, 80));
         set_local_var(5, random(40, 50));
         set_local_var(6, random(-20, 10));
-        critter_add_trait(self_obj, 1, 6, 11);
-        critter_add_trait(self_obj, 1, 5, 32);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 11);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_GENTLEMAN_COWARD);
         if (random(0, 100) <= 75) then begin
-            add_mult_objs_to_inven(self_obj, create_object_sid(41, 1, 2, -1), random(1, 40));
+            add_mult_objs_to_inven(self_obj, create_object(41, 1, 2), random(1, 40));
         end
         if (random(0, 100) <= 45) then begin
-            add_obj_to_inven(self_obj, create_object_sid(415, 1, 2, -1));
+            add_obj_to_inven(self_obj, create_object(415, 1, 2));
         end
         if (random(0, 100) <= 40) then begin
-            add_obj_to_inven(self_obj, create_object_sid(419, 1, 2, -1));
+            add_obj_to_inven(self_obj, create_object(419, 1, 2));
         end
         if (random(0, 100) <= 30) then begin
-            add_obj_to_inven(self_obj, create_object_sid(434, 1, 2, -1));
+            add_obj_to_inven(self_obj, create_object(434, 1, 2));
         end
         if (random(0, 100) <= 20) then begin
-            add_obj_to_inven(self_obj, create_object_sid(145, 1, 2, -1));
+            add_obj_to_inven(self_obj, create_object(145, 1, 2));
         end
         if (random(0, 100) <= 15) then begin
-            add_obj_to_inven(self_obj, create_object_sid(223, 1, 2, -1));
+            add_obj_to_inven(self_obj, create_object(223, 1, 2));
         end
         if (random(0, 100) <= 15) then begin
-            add_obj_to_inven(self_obj, create_object_sid(39, 1, 2, -1));
+            add_obj_to_inven(self_obj, create_object(39, 1, 2));
         end
         if (random(0, 100) <= 15) then begin
-            add_obj_to_inven(self_obj, create_object_sid(82, 1, 2, -1));
+            add_obj_to_inven(self_obj, create_object(82, 1, 2));
         end
         if (random(0, 100) <= 10) then begin
-            add_obj_to_inven(self_obj, create_object_sid(228, 1, 2, -1));
+            add_obj_to_inven(self_obj, create_object(228, 1, 2));
         end
         if (random(0, 100) <= 10) then begin
-            add_obj_to_inven(self_obj, create_object_sid(410, 1, 2, -1));
+            add_obj_to_inven(self_obj, create_object(410, 1, 2));
         end
         if (random(0, 100) <= 10) then begin
-            add_obj_to_inven(self_obj, create_object_sid(431, 1, 2, -1));
+            add_obj_to_inven(self_obj, create_object(431, 1, 2));
         end
         if (random(0, 100) <= 5) then begin
-            add_obj_to_inven(self_obj, create_object_sid(110, 1, 2, -1));
+            add_obj_to_inven(self_obj, create_object(110, 1, 2));
         end
         if (random(0, 100) <= 5) then begin
-            add_obj_to_inven(self_obj, create_object_sid(323, 1, 2, -1));
+            add_obj_to_inven(self_obj, create_object(323, 1, 2));
         end
         if (random(0, 100) <= 5) then begin
-            add_obj_to_inven(self_obj, create_object_sid(239, 1, 2, -1));
+            add_obj_to_inven(self_obj, create_object(239, 1, 2));
         end
         if (random(0, 100) <= 5) then begin
-            add_obj_to_inven(self_obj, create_object_sid(235, 1, 2, -1));
+            add_obj_to_inven(self_obj, create_object(235, 1, 2));
         end
     end
 end
@@ -124,7 +124,7 @@ begin
         add_timer_event(self_obj, random(1000, 1000), 2);
     end
     if (obj_can_see_obj(self_obj, dude_obj)) then begin
-        if ((local_var(2) == 1) or (global_var(51) <= -31)) then begin
+        if ((local_var(2) == 1) or (global_var(GVAR_TOWN_REP_GECKO) <= -31)) then begin
             set_local_var(2, 0);
             attack_setup(self_obj, dude_obj);
         end
@@ -139,10 +139,10 @@ procedure description_p_proc
 begin
     script_overrides;
     if (local_var(7) == 0) then begin
-        display_msg(message_str(228, 2));
+        display_mstr(2);
     end
     else begin
-        display_msg(message_str(228, 4));
+        display_mstr(4);
     end
 end
 
@@ -150,10 +150,10 @@ procedure look_at_p_proc
 begin
     script_overrides;
     if (local_var(7) == 0) then begin
-        display_msg(message_str(228, 1));
+        display_mstr(1);
     end
     else begin
-        display_msg(message_str(228, 3));
+        display_mstr(3);
     end
 end
 
@@ -167,7 +167,7 @@ end
 procedure talk_p_proc
 begin
     if (not(anim_busy(self_obj))) then begin
-        anim(self_obj, 1000, rotation_to_tile(tile_num(self_obj), tile_num(dude_obj)));
+        anim(self_obj, 1000, rotation_to_tile(tile_num(self_obj), dude_tile));
     end
     start_gdialog(228, self_obj, -1, -1, -1);
     gsay_start;
@@ -183,13 +183,13 @@ end
 
 procedure destroy_p_proc
 begin
-    set_global_var(118, global_var(118) bwor 8388608);
-    if (global_var(156) == 1) then begin
-        set_global_var(156, 0);
+    set_global_var(GVAR_MODOC_TOOL_FLAG, global_var(GVAR_MODOC_TOOL_FLAG) bwor 8388608);
+    if (global_var(GVAR_SIERRA_LOCKOUT) == 1) then begin
+        set_global_var(GVAR_SIERRA_LOCKOUT, 0);
     end
     if (source_obj == dude_obj) then begin
-        set_global_var(0, global_var(0) - 1);
-        set_global_var(51, global_var(51) - 6);
+        set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) - 1);
+        set_global_var(GVAR_TOWN_REP_GECKO, global_var(GVAR_TOWN_REP_GECKO) - 6);
     end
 end
 
@@ -201,51 +201,51 @@ begin
             set_sfall_global(13, 1);
         end
         else begin
-            if (has_trait(2, dude_obj, 13)) then begin
-                if (global_var(81) < get_pc_stat(1)) then begin
+            if (has_trait(TRAIT_TRAIT, dude_obj, TRAIT_sex_appeal)) then begin
+                if (global_var(GVAR_VAULT_CITIZENSHIP) < get_pc_stat(PCSTAT_level)) then begin
                     if (is_success(roll_vs_skill(dude_obj, 10, 0))) then begin
-                        set_global_var(98, global_var(98) + 3);
+                        set_global_var(GVAR_QUEST_FIND_THOMAS, global_var(GVAR_QUEST_FIND_THOMAS) + 3);
                     end
                     else begin
-                        set_global_var(98, global_var(98) + 1);
+                        set_global_var(GVAR_QUEST_FIND_THOMAS, global_var(GVAR_QUEST_FIND_THOMAS) + 1);
                     end
-                    if (global_var(98) >= (2 * get_pc_stat(1))) then begin
-                        set_global_var(98, 0);
-                        set_global_var(81, global_var(81) + 1);
-                        critter_mod_skill(dude_obj, 10, 1);
-                        display_msg(message_str(10, 410));
+                    if (global_var(GVAR_QUEST_FIND_THOMAS) >= (2 * get_pc_stat(PCSTAT_level))) then begin
+                        set_global_var(GVAR_QUEST_FIND_THOMAS, 0);
+                        set_global_var(GVAR_VAULT_CITIZENSHIP, global_var(GVAR_VAULT_CITIZENSHIP) + 1);
+                        critter_mod_skill(dude_obj, SKILL_STEAL, 1);
+                        display_msg(message_str(SCRIPT_STTEXT, 410));
                     end
                 end
             end
             steal_perk := 0;
             steal_gloves := 0;
-            if (has_trait(0, dude_obj, 37) != 0) then begin
+            if (has_trait(TRAIT_PERK, dude_obj, PERK_pickpocket) != 0) then begin
                 steal_perk := steal_perk + 15;
             end
-            if (has_trait(0, dude_obj, 33) != 0) then begin
+            if (has_trait(TRAIT_PERK, dude_obj, PERK_master_thief) != 0) then begin
                 steal_perk := steal_perk + 30;
             end
-            if ((obj_pid(critter_inven_obj(dude_obj, 1)) == 245) or (obj_pid(critter_inven_obj(dude_obj, 2)) == 245)) then begin
+            if ((obj_pid(critter_inven_obj(dude_obj, INVEN_TYPE_RIGHT_HAND)) == 245) or (obj_pid(critter_inven_obj(dude_obj, INVEN_TYPE_LEFT_HAND)) == 245)) then begin
                 steal_gloves := 10;
             end
-            if (random(0, 100) <= ((has_skill(dude_obj, 10) / 2) + (((3 * get_critter_stat(dude_obj, 5)) + (((get_critter_stat(dude_obj, 6) - 4) * 4) - (difficulty_level * 4)) + (2 * steal_perk) + steal_gloves) / (local_var(4) / 25)))) then begin
-                display_msg(message_str(10, random(145, 148)));
+            if (random(0, 100) <= ((dude_skill(SKILL_STEAL) / 2) + (((3 * get_critter_stat(dude_obj, STAT_ag)) + (((dude_luck - 4) * 4) - (difficulty_level * 4)) + (2 * steal_perk) + steal_gloves) / (local_var(4) / 25)))) then begin
+                display_msg(message_str(SCRIPT_STTEXT, random(145, 148)));
             end
             else begin
                 script_overrides;
-                if (random(0, 100) <= (25 - floor((((get_critter_stat(dude_obj, 6) - 4) * 4) - (difficulty_level * 4)) / 3))) then begin
-                    float_msg(self_obj, message_str(10, random(206, 209)), 2);
+                if (random(0, 100) <= (25 - floor((((dude_luck - 4) * 4) - (difficulty_level * 4)) / 3))) then begin
+                    float_msg(self_obj, message_str(SCRIPT_STTEXT, random(206, 209)), FLOAT_MSG_RED);
                     set_local_var(2, 1);
                 end
                 else begin
-                    display_msg(message_str(10, random(200, 203)));
+                    display_msg(message_str(SCRIPT_STTEXT, random(200, 203)));
                 end
             end
         end
     end
     if (action_being_used == 12) then begin
         script_overrides;
-        display_msg(message_str(10, 204) + ((has_skill(dude_obj, 10) / 2) + (((3 * get_critter_stat(dude_obj, 5)) + (((get_critter_stat(dude_obj, 6) - 4) * 4) - (difficulty_level * 4)) + (2 * steal_perk) + steal_gloves) / (local_var(4) / 25))) + message_str(10, 205));
+        display_msg(message_str(SCRIPT_STTEXT, 204) + ((dude_skill(SKILL_STEAL) / 2) + (((3 * get_critter_stat(dude_obj, STAT_ag)) + (((dude_luck - 4) * 4) - (difficulty_level * 4)) + (2 * steal_perk) + steal_gloves) / (local_var(4) / 25))) + message_str(SCRIPT_STTEXT, 205));
     end
 end
 
@@ -278,375 +278,375 @@ procedure Node001
 begin
     set_local_var(7, 1);
     already_talk := 1;
-    gsay_reply(228, 10);
-    gsay_option(228, 11, Node002, 50);
-    gsay_option(228, 12, Node009, 50);
-    gsay_option(228, 13, Node011, 50);
+    Reply(10);
+    giq_option(1, NAME, 11, Node002, NEUTRAL_REACTION);
+    giq_option(1, NAME, 12, Node009, NEUTRAL_REACTION);
+    giq_option(1, NAME, 13, Node011, NEUTRAL_REACTION);
 end
 
 procedure Node002
 begin
-    gsay_reply(228, 20);
-    gsay_option(228, 21, Node003, 50);
-    gsay_option(228, 22, Node004, 50);
-    gsay_option(228, 23, Node005, 50);
+    Reply(20);
+    giq_option(1, NAME, 21, Node003, NEUTRAL_REACTION);
+    giq_option(1, NAME, 22, Node004, NEUTRAL_REACTION);
+    giq_option(1, NAME, 23, Node005, NEUTRAL_REACTION);
 end
 
 procedure Node003
 begin
-    gsay_reply(228, 30);
-    gsay_option(228, 31, Node004, 50);
-    gsay_option(228, 32, Node007, 50);
-    gsay_option(228, 33, Node005, 50);
+    Reply(30);
+    giq_option(1, NAME, 31, Node004, NEUTRAL_REACTION);
+    giq_option(1, NAME, 32, Node007, NEUTRAL_REACTION);
+    giq_option(1, NAME, 33, Node005, NEUTRAL_REACTION);
 end
 
 procedure Node004
 begin
-    gsay_reply(228, 40);
-    gsay_option(228, 41, Node006, 50);
-    gsay_option(228, 42, Node007, 50);
-    gsay_option(228, 43, Node003, 50);
+    Reply(40);
+    giq_option(1, NAME, 41, Node006, NEUTRAL_REACTION);
+    giq_option(1, NAME, 42, Node007, NEUTRAL_REACTION);
+    giq_option(1, NAME, 43, Node003, NEUTRAL_REACTION);
 end
 
 procedure Node005
 begin
-    gsay_reply(228, 50);
-    gsay_option(228, 51, Node006, 50);
-    gsay_option(228, 52, Node003, 50);
-    gsay_option(228, 53, Node007, 50);
+    Reply(50);
+    giq_option(1, NAME, 51, Node006, NEUTRAL_REACTION);
+    giq_option(1, NAME, 52, Node003, NEUTRAL_REACTION);
+    giq_option(1, NAME, 53, Node007, NEUTRAL_REACTION);
 end
 
 procedure Node006
 begin
-    gsay_reply(228, 60);
-    gsay_option(228, 61, Node012, 50);
-    gsay_option(228, 62, Node012, 50);
-    gsay_option(228, 63, Node013, 50);
-    gsay_option(228, 64, Node013, 50);
-    gsay_option(228, 65, Node013, 50);
-    gsay_option(228, 66, Node014, 50);
-    gsay_option(228, 67, Node014, 50);
+    Reply(60);
+    giq_option(1, NAME, 61, Node012, NEUTRAL_REACTION);
+    giq_option(1, NAME, 62, Node012, NEUTRAL_REACTION);
+    giq_option(1, NAME, 63, Node013, NEUTRAL_REACTION);
+    giq_option(1, NAME, 64, Node013, NEUTRAL_REACTION);
+    giq_option(1, NAME, 65, Node013, NEUTRAL_REACTION);
+    giq_option(1, NAME, 66, Node014, NEUTRAL_REACTION);
+    giq_option(1, NAME, 67, Node014, NEUTRAL_REACTION);
 end
 
 procedure Node007
 begin
-    gsay_reply(228, 70);
-    gsay_option(228, 71, Node003, 50);
-    gsay_option(228, 72, Node006, 50);
-    gsay_option(228, 73, Node008, 50);
+    Reply(70);
+    giq_option(1, NAME, 71, Node003, NEUTRAL_REACTION);
+    giq_option(1, NAME, 72, Node006, NEUTRAL_REACTION);
+    giq_option(1, NAME, 73, Node008, NEUTRAL_REACTION);
 end
 
 procedure Node008
 begin
-    gsay_reply(228, 80);
-    gsay_option(228, 81, Node998, 50);
+    Reply(80);
+    giq_option(1, NAME, 81, Node998, NEUTRAL_REACTION);
 end
 
 procedure Node009
 begin
-    gsay_reply(228, 90);
-    gsay_option(228, 91, Node003, 50);
-    gsay_option(228, 92, Node004, 50);
-    gsay_option(228, 93, Node005, 50);
+    Reply(90);
+    giq_option(1, NAME, 91, Node003, NEUTRAL_REACTION);
+    giq_option(1, NAME, 92, Node004, NEUTRAL_REACTION);
+    giq_option(1, NAME, 93, Node005, NEUTRAL_REACTION);
 end
 
 procedure Node010
 begin
     if (already_talk == 0) then begin
         already_talk := 1;
-        if (global_var(227) == 2) then begin
-            gsay_reply(228, 1001);
+        if (global_var(GVAR_NCR_ENCLAVE_INFO) == 2) then begin
+            Reply(1001);
         end
         else begin
-            gsay_reply(228, 1003);
+            Reply(1003);
         end
     end
     else begin
-        gsay_reply(228, 1002);
+        Reply(1002);
     end
-    if (global_var(160) == 1) then begin
-        gsay_option(228, 101, Node015, 50);
+    if (global_var(GVAR_GECKO_SKEETER_PART) == 1) then begin
+        giq_option(1, NAME, 101, Node015, NEUTRAL_REACTION);
     end
-    if ((global_var(152) == 1) and (global_var(228) == 0)) then begin
-        gsay_option(228, 102, Node016, 50);
+    if ((global_var(GVAR_SIERRA_BASE_ENEMY) == 1) and (global_var(GVAR_NCR_WESTIN_SEED) == 0)) then begin
+        giq_option(1, NAME, 102, Node016, NEUTRAL_REACTION);
     end
-    gsay_option(228, 103, Node017, 50);
-    gsay_option(228, 104, Node019, 50);
-    if (global_var(156) == 1) then begin
-        gsay_option(228, 105, Node026, 50);
+    giq_option(1, NAME, 103, Node017, NEUTRAL_REACTION);
+    giq_option(1, NAME, 104, Node019, NEUTRAL_REACTION);
+    if (global_var(GVAR_SIERRA_LOCKOUT) == 1) then begin
+        giq_option(1, NAME, 105, Node026, NEUTRAL_REACTION);
     end
-    gsay_option(228, 106, Node999, 50);
+    giq_option(1, NAME, 106, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node011
 begin
-    gsay_reply(228, 110);
-    gsay_option(228, 111, Node003, 50);
-    gsay_option(228, 112, Node004, 50);
-    gsay_option(228, 113, Node005, 50);
+    Reply(110);
+    giq_option(1, NAME, 111, Node003, NEUTRAL_REACTION);
+    giq_option(1, NAME, 112, Node004, NEUTRAL_REACTION);
+    giq_option(1, NAME, 113, Node005, NEUTRAL_REACTION);
 end
 
 procedure Node012
 begin
-    gsay_reply(228, 120);
-    gsay_option(228, 121, Node010, 50);
-    gsay_option(228, 122, Node999, 50);
+    Reply(120);
+    giq_option(1, NAME, 121, Node010, NEUTRAL_REACTION);
+    giq_option(1, NAME, 122, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node013
 begin
-    gsay_reply(228, 130);
-    gsay_option(228, 131, Node010, 50);
-    gsay_option(228, 132, Node999, 50);
+    Reply(130);
+    giq_option(1, NAME, 131, Node010, NEUTRAL_REACTION);
+    giq_option(1, NAME, 132, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node014
 begin
-    gsay_reply(228, 140);
-    gsay_option(228, 141, Node010, 50);
-    gsay_option(228, 142, Node999, 50);
+    Reply(140);
+    giq_option(1, NAME, 141, Node010, NEUTRAL_REACTION);
+    giq_option(1, NAME, 142, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node015
 begin
-    gsay_reply(228, 150);
-    gsay_option(228, 151, Node010, 50);
-    gsay_option(228, 152, Node999, 50);
+    Reply(150);
+    giq_option(1, NAME, 151, Node010, NEUTRAL_REACTION);
+    giq_option(1, NAME, 152, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node016
 begin
-    set_global_var(228, 1);
-    gsay_reply(228, 160);
-    gsay_option(228, 161, Node010, 50);
-    gsay_option(228, 162, Node999, 50);
+    set_global_var(GVAR_NCR_WESTIN_SEED, 1);
+    Reply(160);
+    giq_option(1, NAME, 161, Node010, NEUTRAL_REACTION);
+    giq_option(1, NAME, 162, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node017
 begin
-    gsay_reply(228, 170);
-    gsay_option(228, 171, Node018, 50);
+    Reply(170);
+    giq_option(1, NAME, 171, Node018, NEUTRAL_REACTION);
 end
 
 procedure Node018
 begin
-    gsay_reply(228, 180);
-    gsay_option(228, 181, Node020, 50);
-    gsay_option(228, 182, Node010, 50);
+    Reply(180);
+    giq_option(1, NAME, 181, Node020, NEUTRAL_REACTION);
+    giq_option(1, NAME, 182, Node010, NEUTRAL_REACTION);
 end
 
 procedure Node019
 begin
-    gsay_reply(228, 190);
-    gsay_option(228, 191, Node020, 50);
-    gsay_option(228, 192, Node010, 50);
+    Reply(190);
+    giq_option(1, NAME, 191, Node020, NEUTRAL_REACTION);
+    giq_option(1, NAME, 192, Node010, NEUTRAL_REACTION);
 end
 
 procedure Node020
 begin
-    gsay_reply(228, 200);
-    gsay_option(228, 201, Node021, 50);
-    gsay_option(228, 202, Node010, 50);
+    Reply(200);
+    giq_option(1, NAME, 201, Node021, NEUTRAL_REACTION);
+    giq_option(1, NAME, 202, Node010, NEUTRAL_REACTION);
 end
 
 procedure Node021
 begin
-    if (global_var(156) < 2) then begin
-        gsay_reply(228, 2101);
+    if (global_var(GVAR_SIERRA_LOCKOUT) < 2) then begin
+        Reply(2101);
     end
     else begin
-        if (global_var(156) == 2) then begin
-            gsay_reply(228, 2102);
+        if (global_var(GVAR_SIERRA_LOCKOUT) == 2) then begin
+            Reply(2102);
         end
         else begin
-            gsay_reply(228, 2103);
+            Reply(2103);
         end
     end
-    if ((global_var(151) == 1) and (global_var(203) != 5)) then begin
-        gsay_option(228, 211, Node022, 50);
+    if ((global_var(GVAR_SIERRA_BASE_ALERT) == 1) and (global_var(GVAR_ENEMY_TORR) != 5)) then begin
+        giq_option(1, NAME, 211, Node022, NEUTRAL_REACTION);
     end
-    if (global_var(156) == 0) then begin
-        gsay_option(228, 212, Node025, 50);
+    if (global_var(GVAR_SIERRA_LOCKOUT) == 0) then begin
+        giq_option(1, NAME, 212, Node025, NEUTRAL_REACTION);
     end
-    if (global_var(156) < 2) then begin
-        gsay_option(228, 213, Node023, 50);
+    if (global_var(GVAR_SIERRA_LOCKOUT) < 2) then begin
+        giq_option(1, NAME, 213, Node023, NEUTRAL_REACTION);
     end
-    gsay_option(228, 214, Node010, 50);
+    giq_option(1, NAME, 214, Node010, NEUTRAL_REACTION);
 end
 
 procedure Node022
 begin
-    gsay_reply(228, 220);
-    if (global_var(156) == 0) then begin
-        gsay_option(228, 221, Node025, 50);
+    Reply(220);
+    if (global_var(GVAR_SIERRA_LOCKOUT) == 0) then begin
+        giq_option(1, NAME, 221, Node025, NEUTRAL_REACTION);
     end
-    if (global_var(156) < 2) then begin
-        gsay_option(228, 222, Node023, 50);
+    if (global_var(GVAR_SIERRA_LOCKOUT) < 2) then begin
+        giq_option(1, NAME, 222, Node023, NEUTRAL_REACTION);
     end
-    gsay_option(228, 223, Node010, 50);
-    gsay_option(228, 224, Node999, 50);
+    giq_option(1, NAME, 223, Node010, NEUTRAL_REACTION);
+    giq_option(1, NAME, 224, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node023
 begin
-    if (global_var(156) == 0) then begin
-        set_global_var(156, 1);
+    if (global_var(GVAR_SIERRA_LOCKOUT) == 0) then begin
+        set_global_var(GVAR_SIERRA_LOCKOUT, 1);
     end
-    gsay_reply(228, 230);
-    if ((global_var(285) == 2) and (get_critter_stat(dude_obj, 34) == 0) and (get_critter_stat(dude_obj, 3) >= 6) and (has_skill(dude_obj, 14) >= 80)) then begin
-        gsay_option(228, 231, Node024, 50);
+    Reply(230);
+    if ((global_var(GVAR_NEW_RENO_WRIGHT_FLAGS) == 2) and (dude_gender == 0) and (dude_charisma >= 6) and (dude_skill(SKILL_SPEECH) >= 80)) then begin
+        giq_option(1, NAME, 231, Node024, NEUTRAL_REACTION);
     end
-    if ((global_var(285) == 2) and (get_critter_stat(dude_obj, 34) == 1) and (get_critter_stat(dude_obj, 3) >= 7) and (has_skill(dude_obj, 14) >= 90)) then begin
-        gsay_option(228, 231, Node024, 50);
+    if ((global_var(GVAR_NEW_RENO_WRIGHT_FLAGS) == 2) and (dude_gender == 1) and (dude_charisma >= 7) and (dude_skill(SKILL_SPEECH) >= 90)) then begin
+        giq_option(1, NAME, 231, Node024, NEUTRAL_REACTION);
     end
-    gsay_option(228, 232, Node025, 50);
-    gsay_option(228, 233, Node010, 50);
+    giq_option(1, NAME, 232, Node025, NEUTRAL_REACTION);
+    giq_option(1, NAME, 233, Node010, NEUTRAL_REACTION);
 end
 
 procedure Node024
 begin
-    gsay_reply(228, 240);
-    gsay_option(228, 241, Node010, 50);
+    Reply(240);
+    giq_option(1, NAME, 241, Node010, NEUTRAL_REACTION);
     if (item_caps_total(dude_obj) >= 350) then begin
-        gsay_option(228, 242, Node031, 50);
+        giq_option(1, NAME, 242, Node031, NEUTRAL_REACTION);
     end
-    gsay_option(228, 245, Node999, 50);
+    giq_option(1, NAME, 245, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node025
 begin
-    if (global_var(156) == 0) then begin
-        set_global_var(156, 1);
+    if (global_var(GVAR_SIERRA_LOCKOUT) == 0) then begin
+        set_global_var(GVAR_SIERRA_LOCKOUT, 1);
     end
-    gsay_reply(228, 250);
-    gsay_option(228, 251, Node999, 50);
+    Reply(250);
+    giq_option(1, NAME, 251, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node026
 begin
-    gsay_reply(228, 260);
-    if ((global_var(285) == 2) and (get_critter_stat(dude_obj, 34) == 0) and (get_critter_stat(dude_obj, 3) >= 6) and (has_skill(dude_obj, 14) >= 80)) then begin
-        gsay_option(228, 261, Node024, 50);
+    Reply(260);
+    if ((global_var(GVAR_NEW_RENO_WRIGHT_FLAGS) == 2) and (dude_gender == 0) and (dude_charisma >= 6) and (dude_skill(SKILL_SPEECH) >= 80)) then begin
+        giq_option(1, NAME, 261, Node024, NEUTRAL_REACTION);
     end
-    if ((global_var(285) == 2) and (get_critter_stat(dude_obj, 34) == 1) and (get_critter_stat(dude_obj, 3) >= 7) and (has_skill(dude_obj, 14) >= 90)) then begin
-        gsay_option(228, 261, Node024, 50);
+    if ((global_var(GVAR_NEW_RENO_WRIGHT_FLAGS) == 2) and (dude_gender == 1) and (dude_charisma >= 7) and (dude_skill(SKILL_SPEECH) >= 90)) then begin
+        giq_option(1, NAME, 261, Node024, NEUTRAL_REACTION);
     end
-    gsay_option(228, 262, Node010, 50);
-    if (global_var(243) == 1) then begin
-        gsay_option(228, 263, Node027, 50);
+    giq_option(1, NAME, 262, Node010, NEUTRAL_REACTION);
+    if (global_var(GVAR_8_BALL_TOILET_SECRET) == 1) then begin
+        giq_option(1, NAME, 263, Node027, NEUTRAL_REACTION);
     end
-    if (global_var(243) == 2) then begin
-        gsay_option(228, 265, Node027a, 50);
+    if (global_var(GVAR_8_BALL_TOILET_SECRET) == 2) then begin
+        giq_option(1, NAME, 265, Node027a, NEUTRAL_REACTION);
     end
-    if (global_var(156) == 1) then begin
-        gsay_option(228, 264, Node030, 50);
+    if (global_var(GVAR_SIERRA_LOCKOUT) == 1) then begin
+        giq_option(1, NAME, 264, Node030, NEUTRAL_REACTION);
     end
 end
 
 procedure Node027
 begin
-    gsay_reply(228, 270);
-    if (global_var(233) == 0) then begin
-        gsay_option(228, 271, Node028, 50);
+    Reply(270);
+    if (global_var(GVAR_VAULT13_FOUND_GECK) == 0) then begin
+        giq_option(1, NAME, 271, Node028, NEUTRAL_REACTION);
     end
-    if (global_var(234) == 0) then begin
-        gsay_option(228, 272, Node029, 50);
+    if (global_var(GVAR_NCR_POWER_ON) == 0) then begin
+        giq_option(1, NAME, 272, Node029, NEUTRAL_REACTION);
     end
 end
 
 procedure Node027a
 begin
-    if (global_var(156) == 1) then begin
-        set_global_var(156, 3);
+    if (global_var(GVAR_SIERRA_LOCKOUT) == 1) then begin
+        set_global_var(GVAR_SIERRA_LOCKOUT, 3);
     end
-    add_obj_to_inven(self_obj, create_object_sid(300, 1, 2, -1));
-    add_obj_to_inven(self_obj, create_object_sid(157, 1, 2, -1));
-    gsay_reply(228, 273);
-    gsay_option(228, 274, Node999, 50);
-    gsay_option(228, 275, Node999, 50);
+    add_obj_to_inven(self_obj, create_object(300, 1, 2));
+    add_obj_to_inven(self_obj, create_object(157, 1, 2));
+    Reply(273);
+    giq_option(1, NAME, 274, Node999, NEUTRAL_REACTION);
+    giq_option(1, NAME, 275, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node028
 begin
-    if (global_var(156) == 1) then begin
-        set_global_var(156, 2);
-        give_exp_points(floor(241) + (get_critter_stat(dude_obj, 4) * floor(100)) + (get_critter_stat(dude_obj, 6) * floor(40)) + (floor(16) * (((get_critter_stat(dude_obj, 6) - 4) * 4) - (difficulty_level * 4))));
-        display_msg(message_str(10, random(140, 143)) + (floor(241) + (get_critter_stat(dude_obj, 4) * floor(100)) + (get_critter_stat(dude_obj, 6) * floor(40)) + (floor(16) * (((get_critter_stat(dude_obj, 6) - 4) * 4) - (difficulty_level * 4)))) + message_str(10, 144));
-        if (global_var(285) == 3) then begin
+    if (global_var(GVAR_SIERRA_LOCKOUT) == 1) then begin
+        set_global_var(GVAR_SIERRA_LOCKOUT, 2);
+        give_exp_points(floor(241) + (dude_iq * floor(100)) + (dude_luck * floor(40)) + (floor(16) * (((dude_luck - 4) * 4) - (difficulty_level * 4))));
+        display_msg(message_str(SCRIPT_STTEXT, random(140, 143)) + (floor(241) + (dude_iq * floor(100)) + (dude_luck * floor(40)) + (floor(16) * (((dude_luck - 4) * 4) - (difficulty_level * 4)))) + message_str(SCRIPT_STTEXT, 144));
+        if (global_var(GVAR_NEW_RENO_WRIGHT_FLAGS) == 3) then begin
             item_caps_adjust(dude_obj, 250);
         end
         else begin
             item_caps_adjust(dude_obj, 150);
         end
-        add_obj_to_inven(dude_obj, create_object_sid(300, 1, 2, -1));
-        set_global_var(51, global_var(51) + 4);
-        set_global_var(233, 1);
+        add_obj_to_inven(dude_obj, create_object(300, 1, 2));
+        set_global_var(GVAR_TOWN_REP_GECKO, global_var(GVAR_TOWN_REP_GECKO) + 4);
+        set_global_var(GVAR_VAULT13_FOUND_GECK, 1);
     end
-    gsay_reply(228, 280);
-    gsay_option(228, 281, Node999, 50);
+    Reply(280);
+    giq_option(1, NAME, 281, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node029
 begin
-    if (global_var(156) == 1) then begin
-        set_global_var(156, 2);
-        give_exp_points(floor(241) + (get_critter_stat(dude_obj, 4) * floor(100)) + (get_critter_stat(dude_obj, 6) * floor(40)) + (floor(16) * (((get_critter_stat(dude_obj, 6) - 4) * 4) - (difficulty_level * 4))));
-        display_msg(message_str(10, random(140, 143)) + (floor(241) + (get_critter_stat(dude_obj, 4) * floor(100)) + (get_critter_stat(dude_obj, 6) * floor(40)) + (floor(16) * (((get_critter_stat(dude_obj, 6) - 4) * 4) - (difficulty_level * 4)))) + message_str(10, 144));
-        if (global_var(285) == 3) then begin
+    if (global_var(GVAR_SIERRA_LOCKOUT) == 1) then begin
+        set_global_var(GVAR_SIERRA_LOCKOUT, 2);
+        give_exp_points(floor(241) + (dude_iq * floor(100)) + (dude_luck * floor(40)) + (floor(16) * (((dude_luck - 4) * 4) - (difficulty_level * 4))));
+        display_msg(message_str(SCRIPT_STTEXT, random(140, 143)) + (floor(241) + (dude_iq * floor(100)) + (dude_luck * floor(40)) + (floor(16) * (((dude_luck - 4) * 4) - (difficulty_level * 4)))) + message_str(SCRIPT_STTEXT, 144));
+        if (global_var(GVAR_NEW_RENO_WRIGHT_FLAGS) == 3) then begin
             item_caps_adjust(dude_obj, 250);
         end
         else begin
             item_caps_adjust(dude_obj, 150);
         end
-        add_obj_to_inven(dude_obj, create_object_sid(157, 1, 2, -1));
-        set_global_var(51, global_var(51) + 4);
-        set_global_var(234, 1);
+        add_obj_to_inven(dude_obj, create_object(157, 1, 2));
+        set_global_var(GVAR_TOWN_REP_GECKO, global_var(GVAR_TOWN_REP_GECKO) + 4);
+        set_global_var(GVAR_NCR_POWER_ON, 1);
     end
-    gsay_reply(228, 290);
-    gsay_option(228, 291, Node999, 50);
+    Reply(290);
+    giq_option(1, NAME, 291, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node030
 begin
-    gsay_reply(228, 300);
-    gsay_option(228, 301, Node026, 50);
+    Reply(300);
+    giq_option(1, NAME, 301, Node026, NEUTRAL_REACTION);
 end
 
 procedure Node031
 begin
-    gsay_reply(228, 310);
-    if (global_var(233) == 0) then begin
-        gsay_option(228, 311, Node032, 50);
+    Reply(310);
+    if (global_var(GVAR_VAULT13_FOUND_GECK) == 0) then begin
+        giq_option(1, NAME, 311, Node032, NEUTRAL_REACTION);
     end
-    if (global_var(234) == 0) then begin
-        gsay_option(228, 312, Node033, 50);
+    if (global_var(GVAR_NCR_POWER_ON) == 0) then begin
+        giq_option(1, NAME, 312, Node033, NEUTRAL_REACTION);
     end
 end
 
 procedure Node032
 begin
-    if (global_var(156) == 1) then begin
-        set_global_var(156, 2);
-        give_exp_points(floor(241) + (get_critter_stat(dude_obj, 4) * floor(100)) + (get_critter_stat(dude_obj, 6) * floor(40)) + (floor(16) * (((get_critter_stat(dude_obj, 6) - 4) * 4) - (difficulty_level * 4))));
-        display_msg(message_str(10, random(140, 143)) + (floor(241) + (get_critter_stat(dude_obj, 4) * floor(100)) + (get_critter_stat(dude_obj, 6) * floor(40)) + (floor(16) * (((get_critter_stat(dude_obj, 6) - 4) * 4) - (difficulty_level * 4)))) + message_str(10, 144));
-        add_obj_to_inven(dude_obj, create_object_sid(300, 1, 2, -1));
-        set_global_var(233, 1);
+    if (global_var(GVAR_SIERRA_LOCKOUT) == 1) then begin
+        set_global_var(GVAR_SIERRA_LOCKOUT, 2);
+        give_exp_points(floor(241) + (dude_iq * floor(100)) + (dude_luck * floor(40)) + (floor(16) * (((dude_luck - 4) * 4) - (difficulty_level * 4))));
+        display_msg(message_str(SCRIPT_STTEXT, random(140, 143)) + (floor(241) + (dude_iq * floor(100)) + (dude_luck * floor(40)) + (floor(16) * (((dude_luck - 4) * 4) - (difficulty_level * 4)))) + message_str(SCRIPT_STTEXT, 144));
+        add_obj_to_inven(dude_obj, create_object(300, 1, 2));
+        set_global_var(GVAR_VAULT13_FOUND_GECK, 1);
     end
-    gsay_reply(228, 320);
-    gsay_option(228, 321, Node999, 50);
+    Reply(320);
+    giq_option(1, NAME, 321, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node033
 begin
-    if (global_var(156) == 1) then begin
-        set_global_var(156, 2);
-        give_exp_points(floor(241) + (get_critter_stat(dude_obj, 4) * floor(100)) + (get_critter_stat(dude_obj, 6) * floor(40)) + (floor(16) * (((get_critter_stat(dude_obj, 6) - 4) * 4) - (difficulty_level * 4))));
-        display_msg(message_str(10, random(140, 143)) + (floor(241) + (get_critter_stat(dude_obj, 4) * floor(100)) + (get_critter_stat(dude_obj, 6) * floor(40)) + (floor(16) * (((get_critter_stat(dude_obj, 6) - 4) * 4) - (difficulty_level * 4)))) + message_str(10, 144));
-        add_obj_to_inven(dude_obj, create_object_sid(157, 1, 2, -1));
-        set_global_var(234, 1);
+    if (global_var(GVAR_SIERRA_LOCKOUT) == 1) then begin
+        set_global_var(GVAR_SIERRA_LOCKOUT, 2);
+        give_exp_points(floor(241) + (dude_iq * floor(100)) + (dude_luck * floor(40)) + (floor(16) * (((dude_luck - 4) * 4) - (difficulty_level * 4))));
+        display_msg(message_str(SCRIPT_STTEXT, random(140, 143)) + (floor(241) + (dude_iq * floor(100)) + (dude_luck * floor(40)) + (floor(16) * (((dude_luck - 4) * 4) - (difficulty_level * 4)))) + message_str(SCRIPT_STTEXT, 144));
+        add_obj_to_inven(dude_obj, create_object(157, 1, 2));
+        set_global_var(GVAR_NCR_POWER_ON, 1);
     end
-    gsay_reply(228, 330);
-    gsay_option(228, 331, Node999, 50);
+    Reply(330);
+    giq_option(1, NAME, 331, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node999
@@ -659,4 +659,3 @@ begin
     already_talk := 0;
     set_map_var(0, 1);
 end
-
