diff --git a/Mods/Olympus2207/Original/LVMNDOOR.ssl b/Mods/Olympus2207/ReDefine/LVMNDOOR.ssl
index 1fcd30c..7d3ae38 100644
--- a/Mods/Olympus2207/Original/LVMNDOOR.ssl
+++ b/Mods/Olympus2207/ReDefine/LVMNDOOR.ssl
@@ -38,9 +38,9 @@ end
 
 procedure use_p_proc
 begin
-    if (global_var(155) < 2) then begin
+    if (global_var(GVAR_BRAIN_BOT_BRAIN) < 2) then begin
         script_overrides;
-        display_msg(message_str(10, 871));
+        display_msg(message_str(SCRIPT_STTEXT, 871));
     end
     else begin
         add_timer_event(self_obj, 15, 1);
@@ -82,4 +82,3 @@ end
 procedure talk_p_proc
 begin
 end
-
