diff --git a/Mods/Olympus2207/Original/LVZCUST1.ssl b/Mods/Olympus2207/ReDefine/LVZCUST1.ssl
index 5d10959..603e157 100644
--- a/Mods/Olympus2207/Original/LVZCUST1.ssl
+++ b/Mods/Olympus2207/ReDefine/LVZCUST1.ssl
@@ -38,75 +38,75 @@ procedure start
 begin
     if (local_var(1) == 0) then begin
         set_local_var(3, tile_num(self_obj));
-        set_local_var(7, has_trait(1, self_obj, 10));
+        set_local_var(7, obj_get_rot(self_obj));
         set_local_var(1, 1);
         set_local_var(4, random(70, 99));
         set_local_var(5, random(30, 80));
         set_local_var(6, random(-10, 10));
-        critter_add_trait(self_obj, 1, 6, 8);
-        critter_add_trait(self_obj, 1, 5, 21);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 8);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_SEARCHER);
         if (random(0, 100) <= 40) then begin
-            add_obj_to_inven(self_obj, create_object_sid(146, 1, 2, -1));
+            add_obj_to_inven(self_obj, create_object(146, 1, 2));
         end
         else begin
-            add_obj_to_inven(self_obj, create_object_sid(65, 1, 2, -1));
+            add_obj_to_inven(self_obj, create_object(65, 1, 2));
         end
         if (random(0, 100) <= 40) then begin
-            add_obj_to_inven(self_obj, create_object_sid(108, 1, 2, -1));
+            add_obj_to_inven(self_obj, create_object(108, 1, 2));
         end
         else begin
-            add_obj_to_inven(self_obj, create_object_sid(139, 1, 2, -1));
+            add_obj_to_inven(self_obj, create_object(139, 1, 2));
         end
         if (random(0, 100) <= 40) then begin
-            add_obj_to_inven(self_obj, create_object_sid(214, 1, 2, -1));
+            add_obj_to_inven(self_obj, create_object(214, 1, 2));
         end
         else begin
-            add_obj_to_inven(self_obj, create_object_sid(63, 1, 2, -1));
+            add_obj_to_inven(self_obj, create_object(63, 1, 2));
         end
         if (random(0, 100) <= 60) then begin
-            add_obj_to_inven(self_obj, create_object_sid(67, 1, 2, -1));
+            add_obj_to_inven(self_obj, create_object(67, 1, 2));
         end
         else begin
-            add_obj_to_inven(self_obj, create_object_sid(132, 1, 2, -1));
+            add_obj_to_inven(self_obj, create_object(132, 1, 2));
         end
         if (random(0, 100) <= 50) then begin
-            add_obj_to_inven(self_obj, create_object_sid(147, 1, 2, -1));
+            add_obj_to_inven(self_obj, create_object(147, 1, 2));
         end
         else begin
-            add_obj_to_inven(self_obj, create_object_sid(38, 1, 2, -1));
+            add_obj_to_inven(self_obj, create_object(38, 1, 2));
         end
         if (random(0, 100) <= 33) then begin
-            add_obj_to_inven(self_obj, create_object_sid(226, 1, 2, -1));
+            add_obj_to_inven(self_obj, create_object(226, 1, 2));
         end
         if (random(0, 100) <= 33) then begin
-            add_obj_to_inven(self_obj, create_object_sid(225, 1, 2, -1));
+            add_obj_to_inven(self_obj, create_object(225, 1, 2));
         end
         if (random(0, 100) <= 33) then begin
-            add_obj_to_inven(self_obj, create_object_sid(227, 1, 2, -1));
+            add_obj_to_inven(self_obj, create_object(227, 1, 2));
         end
         if (random(0, 100) <= 70) then begin
-            add_mult_objs_to_inven(self_obj, create_object_sid(2, 1, 2, -1), random(5, 40));
+            add_mult_objs_to_inven(self_obj, create_object(2, 1, 2), random(5, 40));
         end
         if (random(0, 100) <= 70) then begin
-            add_mult_objs_to_inven(self_obj, create_object_sid(5, 1, 2, -1), random(5, 40));
+            add_mult_objs_to_inven(self_obj, create_object(5, 1, 2), random(5, 40));
         end
         if (random(0, 100) <= 50) then begin
-            add_mult_objs_to_inven(self_obj, create_object_sid(7, 1, 2, -1), random(5, 40));
+            add_mult_objs_to_inven(self_obj, create_object(7, 1, 2), random(5, 40));
         end
         if (random(0, 100) <= 50) then begin
-            add_mult_objs_to_inven(self_obj, create_object_sid(9, 1, 2, -1), random(5, 40));
+            add_mult_objs_to_inven(self_obj, create_object(9, 1, 2), random(5, 40));
         end
         if (random(0, 100) <= 70) then begin
-            add_mult_objs_to_inven(self_obj, create_object_sid(14, 1, 2, -1), random(5, 40));
+            add_mult_objs_to_inven(self_obj, create_object(14, 1, 2), random(5, 40));
         end
         if (random(0, 100) <= 50) then begin
-            add_mult_objs_to_inven(self_obj, create_object_sid(33, 1, 2, -1), random(1, 3));
+            add_mult_objs_to_inven(self_obj, create_object(33, 1, 2), random(1, 3));
         end
         if (random(0, 100) <= 50) then begin
-            add_mult_objs_to_inven(self_obj, create_object_sid(25, 1, 2, -1), random(2, 6));
+            add_mult_objs_to_inven(self_obj, create_object(25, 1, 2), random(2, 6));
         end
         if (random(0, 100) <= 70) then begin
-            add_mult_objs_to_inven(self_obj, create_object_sid(20, 1, 2, -1), random(5, 40));
+            add_mult_objs_to_inven(self_obj, create_object(20, 1, 2), random(5, 40));
         end
     end
 end
@@ -117,7 +117,7 @@ begin
         animation_is_called := 1;
         add_timer_event(self_obj, random(29, 76), 1);
     end
-    if ((tile_num(self_obj) == local_var(3)) and (has_trait(1, self_obj, 10) != 0) and (anim_busy(self_obj) == 0)) then begin
+    if ((tile_num(self_obj) == local_var(3)) and (obj_get_rot(self_obj) != 0) and (anim_busy(self_obj) == 0)) then begin
         anim(self_obj, 1000, 0);
     end
     if ((ungry_count1 >= 10) or (ungry_count2 >= 10)) then begin
@@ -125,11 +125,11 @@ begin
         ungry_count2 := 0;
         set_local_var(2, 1);
     end
-    if ((ungry_count1 >= 0) and (tile_distance(tile_num(self_obj), tile_num(dude_obj)) > 3)) then begin
+    if ((ungry_count1 >= 0) and (tile_distance(tile_num(self_obj), dude_tile) > 3)) then begin
         ungry_count1 := 0;
     end
     if (obj_can_see_obj(self_obj, dude_obj)) then begin
-        if ((local_var(2) == 1) or (global_var(50) <= -31)) then begin
+        if ((local_var(2) == 1) or (global_var(GVAR_TOWN_REP_VAULT_CITY) <= -31)) then begin
             set_local_var(2, 0);
             attack_setup(self_obj, dude_obj);
         end
@@ -143,13 +143,13 @@ end
 procedure description_p_proc
 begin
     script_overrides;
-    display_msg(message_str(210, random(6, 9)));
+    display_mstr(random(6, 9));
 end
 
 procedure look_at_p_proc
 begin
     script_overrides;
-    display_msg(message_str(210, random(1, 5)));
+    display_mstr(random(1, 5));
 end
 
 procedure pickup_p_proc
@@ -164,17 +164,17 @@ begin
     script_overrides;
     ungry_count2 := ungry_count2 + random(2, 4);
     if (ungry_count2 <= 6) then begin
-        float_msg(self_obj, message_str(210, random(25, 29)), 0);
+        float_msg(self_obj, mstr(random(25, 29)), FLOAT_MSG_NORMAL);
     end
     else begin
-        float_msg(self_obj, message_str(210, random(25, 29)), 2);
+        float_msg(self_obj, mstr(random(25, 29)), FLOAT_MSG_RED);
     end
 end
 
 procedure talk_p_proc
 begin
     if (not(anim_busy(self_obj))) then begin
-        anim(self_obj, 1000, rotation_to_tile(tile_num(self_obj), tile_num(dude_obj)));
+        anim(self_obj, 1000, rotation_to_tile(tile_num(self_obj), dude_tile));
     end
     start_gdialog(210, self_obj, -1, -1, -1);
     gsay_start;
@@ -186,12 +186,12 @@ end
 procedure destroy_p_proc
 begin
     if (source_obj == dude_obj) then begin
-        set_global_var(0, global_var(0) - 1);
+        set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) - 1);
         if ((50 == 50) or (50 == 59)) then begin
-            set_global_var(50, global_var(50) - 3);
+            set_global_var(GVAR_TOWN_REP_VAULT_CITY, global_var(GVAR_TOWN_REP_VAULT_CITY) - 3);
         end
         else begin
-            set_global_var(50, global_var(50) - 6);
+            set_global_var(GVAR_TOWN_REP_VAULT_CITY, global_var(GVAR_TOWN_REP_VAULT_CITY) - 6);
         end
     end
 end
@@ -204,51 +204,51 @@ begin
             set_sfall_global(13, 1);
         end
         else begin
-            if (has_trait(2, dude_obj, 13)) then begin
-                if (global_var(81) < get_pc_stat(1)) then begin
+            if (has_trait(TRAIT_TRAIT, dude_obj, TRAIT_sex_appeal)) then begin
+                if (global_var(GVAR_VAULT_CITIZENSHIP) < get_pc_stat(PCSTAT_level)) then begin
                     if (is_success(roll_vs_skill(dude_obj, 10, 0))) then begin
-                        set_global_var(98, global_var(98) + 3);
+                        set_global_var(GVAR_QUEST_FIND_THOMAS, global_var(GVAR_QUEST_FIND_THOMAS) + 3);
                     end
                     else begin
-                        set_global_var(98, global_var(98) + 1);
+                        set_global_var(GVAR_QUEST_FIND_THOMAS, global_var(GVAR_QUEST_FIND_THOMAS) + 1);
                     end
-                    if (global_var(98) >= (2 * get_pc_stat(1))) then begin
-                        set_global_var(98, 0);
-                        set_global_var(81, global_var(81) + 1);
-                        critter_mod_skill(dude_obj, 10, 1);
-                        display_msg(message_str(10, 410));
+                    if (global_var(GVAR_QUEST_FIND_THOMAS) >= (2 * get_pc_stat(PCSTAT_level))) then begin
+                        set_global_var(GVAR_QUEST_FIND_THOMAS, 0);
+                        set_global_var(GVAR_VAULT_CITIZENSHIP, global_var(GVAR_VAULT_CITIZENSHIP) + 1);
+                        critter_mod_skill(dude_obj, SKILL_STEAL, 1);
+                        display_msg(message_str(SCRIPT_STTEXT, 410));
                     end
                 end
             end
             steal_perk := 0;
             steal_gloves := 0;
-            if (has_trait(0, dude_obj, 37) != 0) then begin
+            if (has_trait(TRAIT_PERK, dude_obj, PERK_pickpocket) != 0) then begin
                 steal_perk := steal_perk + 15;
             end
-            if (has_trait(0, dude_obj, 33) != 0) then begin
+            if (has_trait(TRAIT_PERK, dude_obj, PERK_master_thief) != 0) then begin
                 steal_perk := steal_perk + 30;
             end
-            if ((obj_pid(critter_inven_obj(dude_obj, 1)) == 245) or (obj_pid(critter_inven_obj(dude_obj, 2)) == 245)) then begin
+            if ((obj_pid(critter_inven_obj(dude_obj, INVEN_TYPE_RIGHT_HAND)) == 245) or (obj_pid(critter_inven_obj(dude_obj, INVEN_TYPE_LEFT_HAND)) == 245)) then begin
                 steal_gloves := 10;
             end
-            if (random(0, 100) <= ((has_skill(dude_obj, 10) / 2) + (((3 * get_critter_stat(dude_obj, 5)) + (((get_critter_stat(dude_obj, 6) - 4) * 4) - (difficulty_level * 4)) + (2 * steal_perk) + steal_gloves) / (local_var(4) / 25)))) then begin
-                display_msg(message_str(10, random(145, 148)));
+            if (random(0, 100) <= ((dude_skill(SKILL_STEAL) / 2) + (((3 * get_critter_stat(dude_obj, STAT_ag)) + (((dude_luck - 4) * 4) - (difficulty_level * 4)) + (2 * steal_perk) + steal_gloves) / (local_var(4) / 25)))) then begin
+                display_msg(message_str(SCRIPT_STTEXT, random(145, 148)));
             end
             else begin
                 script_overrides;
-                if (random(0, 100) <= (25 - floor((((get_critter_stat(dude_obj, 6) - 4) * 4) - (difficulty_level * 4)) / 3))) then begin
-                    float_msg(self_obj, message_str(10, random(206, 209)), 2);
+                if (random(0, 100) <= (25 - floor((((dude_luck - 4) * 4) - (difficulty_level * 4)) / 3))) then begin
+                    float_msg(self_obj, message_str(SCRIPT_STTEXT, random(206, 209)), FLOAT_MSG_RED);
                     set_local_var(2, 1);
                 end
                 else begin
-                    display_msg(message_str(10, random(200, 203)));
+                    display_msg(message_str(SCRIPT_STTEXT, random(200, 203)));
                 end
             end
         end
     end
     if (action_being_used == 12) then begin
         script_overrides;
-        display_msg(message_str(10, 204) + ((has_skill(dude_obj, 10) / 2) + (((3 * get_critter_stat(dude_obj, 5)) + (((get_critter_stat(dude_obj, 6) - 4) * 4) - (difficulty_level * 4)) + (2 * steal_perk) + steal_gloves) / (local_var(4) / 25))) + message_str(10, 205));
+        display_msg(message_str(SCRIPT_STTEXT, 204) + ((dude_skill(SKILL_STEAL) / 2) + (((3 * get_critter_stat(dude_obj, STAT_ag)) + (((dude_luck - 4) * 4) - (difficulty_level * 4)) + (2 * steal_perk) + steal_gloves) / (local_var(4) / 25))) + message_str(SCRIPT_STTEXT, 205));
     end
 end
 
@@ -262,7 +262,7 @@ end
 procedure map_enter_p_proc
 begin
     if (not(tile_contains_obj_pid(1, 2, 250))) then begin
-        trading_container := create_object_sid(250, 1, 2, -1);
+        trading_container := create_object(250, 1, 2);
     end
     else begin
         trading_container := tile_contains_pid_obj(1, 2, 250);
@@ -276,89 +276,89 @@ begin
     anim(self_obj, 1000, local_var(7));
     if (days_since_visited > 2) then begin
         move_obj_inven_to_obj(self_obj, trading_container);
-        add_mult_objs_to_inven(self_obj, create_object_sid(41, 1, 2, -1), random(10, 200));
+        add_mult_objs_to_inven(self_obj, create_object(41, 1, 2), random(10, 200));
         if (random(0, 100) <= 40) then begin
-            add_obj_to_inven(self_obj, create_object_sid(146, 1, 2, -1));
+            add_obj_to_inven(self_obj, create_object(146, 1, 2));
         end
         else begin
-            add_obj_to_inven(self_obj, create_object_sid(65, 1, 2, -1));
+            add_obj_to_inven(self_obj, create_object(65, 1, 2));
         end
         if (random(0, 100) <= 40) then begin
-            add_obj_to_inven(self_obj, create_object_sid(108, 1, 2, -1));
+            add_obj_to_inven(self_obj, create_object(108, 1, 2));
         end
         else begin
-            add_obj_to_inven(self_obj, create_object_sid(139, 1, 2, -1));
+            add_obj_to_inven(self_obj, create_object(139, 1, 2));
         end
         if (random(0, 100) <= 40) then begin
-            add_obj_to_inven(self_obj, create_object_sid(214, 1, 2, -1));
+            add_obj_to_inven(self_obj, create_object(214, 1, 2));
         end
         else begin
-            add_obj_to_inven(self_obj, create_object_sid(63, 1, 2, -1));
+            add_obj_to_inven(self_obj, create_object(63, 1, 2));
         end
         if (random(0, 100) <= 60) then begin
-            add_obj_to_inven(self_obj, create_object_sid(67, 1, 2, -1));
+            add_obj_to_inven(self_obj, create_object(67, 1, 2));
         end
         else begin
-            add_obj_to_inven(self_obj, create_object_sid(132, 1, 2, -1));
+            add_obj_to_inven(self_obj, create_object(132, 1, 2));
         end
         if (random(0, 100) <= 50) then begin
-            add_obj_to_inven(self_obj, create_object_sid(147, 1, 2, -1));
+            add_obj_to_inven(self_obj, create_object(147, 1, 2));
         end
         else begin
-            add_obj_to_inven(self_obj, create_object_sid(38, 1, 2, -1));
+            add_obj_to_inven(self_obj, create_object(38, 1, 2));
         end
         if (random(0, 100) <= 33) then begin
-            add_obj_to_inven(self_obj, create_object_sid(226, 1, 2, -1));
+            add_obj_to_inven(self_obj, create_object(226, 1, 2));
         end
         if (random(0, 100) <= 33) then begin
-            add_obj_to_inven(self_obj, create_object_sid(225, 1, 2, -1));
+            add_obj_to_inven(self_obj, create_object(225, 1, 2));
         end
         if (random(0, 100) <= 33) then begin
-            add_obj_to_inven(self_obj, create_object_sid(227, 1, 2, -1));
+            add_obj_to_inven(self_obj, create_object(227, 1, 2));
         end
         if (random(0, 100) <= 70) then begin
-            add_mult_objs_to_inven(self_obj, create_object_sid(2, 1, 2, -1), random(5, 40));
+            add_mult_objs_to_inven(self_obj, create_object(2, 1, 2), random(5, 40));
         end
         if (random(0, 100) <= 70) then begin
-            add_mult_objs_to_inven(self_obj, create_object_sid(5, 1, 2, -1), random(5, 40));
+            add_mult_objs_to_inven(self_obj, create_object(5, 1, 2), random(5, 40));
         end
         if (random(0, 100) <= 50) then begin
-            add_mult_objs_to_inven(self_obj, create_object_sid(7, 1, 2, -1), random(5, 40));
+            add_mult_objs_to_inven(self_obj, create_object(7, 1, 2), random(5, 40));
         end
         if (random(0, 100) <= 50) then begin
-            add_mult_objs_to_inven(self_obj, create_object_sid(9, 1, 2, -1), random(5, 40));
+            add_mult_objs_to_inven(self_obj, create_object(9, 1, 2), random(5, 40));
         end
         if (random(0, 100) <= 70) then begin
-            add_mult_objs_to_inven(self_obj, create_object_sid(14, 1, 2, -1), random(5, 40));
+            add_mult_objs_to_inven(self_obj, create_object(14, 1, 2), random(5, 40));
         end
         if (random(0, 100) <= 50) then begin
-            add_mult_objs_to_inven(self_obj, create_object_sid(33, 1, 2, -1), random(1, 3));
+            add_mult_objs_to_inven(self_obj, create_object(33, 1, 2), random(1, 3));
         end
         if (random(0, 100) <= 50) then begin
-            add_mult_objs_to_inven(self_obj, create_object_sid(25, 1, 2, -1), random(2, 6));
+            add_mult_objs_to_inven(self_obj, create_object(25, 1, 2), random(2, 6));
         end
         if (random(0, 100) <= 70) then begin
-            add_mult_objs_to_inven(self_obj, create_object_sid(20, 1, 2, -1), random(5, 40));
+            add_mult_objs_to_inven(self_obj, create_object(20, 1, 2), random(5, 40));
         end
     end
     max_distance := 8;
     if (random(0, 100) <= 20) then begin
-        metarule3(107, self_obj, 16777260, 0);
+        art_change_fid_num(self_obj, 16777260);
     end
     else begin
         if (random(0, 100) <= 25) then begin
-            metarule3(107, self_obj, 16777271, 0);
+            art_change_fid_num(self_obj, 16777271);
         end
         else begin
             if (random(0, 100) <= 33) then begin
-                metarule3(107, self_obj, 16777290, 0);
+                art_change_fid_num(self_obj, 16777290);
             end
             else begin
                 if (random(0, 100) <= 50) then begin
-                    metarule3(107, self_obj, 16777272, 0);
+                    art_change_fid_num(self_obj, 16777272);
                 end
                 else begin
-                    metarule3(107, self_obj, 16777246, 0);
+                    art_change_fid_num(self_obj, 16777246);
                 end
             end
         end
@@ -371,7 +371,7 @@ begin
         if (tile_num(self_obj) == local_var(3)) then begin
             if (random(0, 100) <= 80) then begin
                 if (random(0, 100) <= 60) then begin
-                    float_msg(self_obj, message_str(210, random(1030, 1039)), 0);
+                    float_msg(self_obj, mstr(random(1030, 1039)), FLOAT_MSG_NORMAL);
                 end
             end
             else begin
@@ -392,9 +392,9 @@ end
 
 procedure Node001
 begin
-    gsay_reply(210, random(10, 12));
-    gsay_option(210, random(13, 15), NodeBarter, 50);
-    gsay_option(210, random(16, 18), Node999, 50);
+    Reply(random(10, 12));
+    giq_option(1, NAME, random(13, 15), NodeBarter, NEUTRAL_REACTION);
+    giq_option(1, NAME, random(16, 18), Node999, NEUTRAL_REACTION);
 end
 
 procedure Node999
@@ -408,4 +408,3 @@ begin
     gdialog_mod_barter(barter_mod);
     call Node001();
 end
-
