diff --git a/Mods/Olympus2207/Original/LVZMNSTR.ssl b/Mods/Olympus2207/ReDefine/LVZMNSTR.ssl
index 747a113..dda7189 100644
--- a/Mods/Olympus2207/Original/LVZMNSTR.ssl
+++ b/Mods/Olympus2207/ReDefine/LVZMNSTR.ssl
@@ -33,10 +33,10 @@ begin
         set_local_var(3, tile_num(self_obj));
         set_local_var(1, 1);
         if (obj_pid(self_obj) == 16777419) then begin
-            critter_add_trait(self_obj, 1, 6, 203);
-            critter_add_trait(self_obj, 1, 5, 69);
-            add_obj_to_inven(self_obj, create_object_sid(518, 1, 2, -1));
-            wield_obj_critter(self_obj, obj_carrying_pid_obj(self_obj, 518));
+            critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 203);
+            critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_GENERIC_RAT);
+            add_obj_to_inven(self_obj, create_object(518, 1, 2));
+            wield_obj(obj_carrying_pid_obj(self_obj, 518));
         end
         max_distance := random(4, 8);
     end
@@ -64,13 +64,13 @@ end
 procedure description_p_proc
 begin
     script_overrides;
-    display_msg(message_str(413, 6));
+    display_mstr(6);
 end
 
 procedure look_at_p_proc
 begin
     script_overrides;
-    display_msg(message_str(413, random(1, 2)));
+    display_mstr(random(1, 2));
 end
 
 procedure pickup_p_proc
@@ -93,19 +93,19 @@ begin
     if (source_obj == dude_obj) then begin
     end
     if (not(tile_contains_obj_pid(2, 2, 250))) then begin
-        stuff_container := create_object_sid(250, 2, 2, -1);
+        stuff_container := create_object(250, 2, 2);
     end
     else begin
         stuff_container := tile_contains_pid_obj(2, 2, 250);
     end
     move_obj_inven_to_obj(self_obj, stuff_container);
-    set_global_var(245, global_var(245) + 1);
-    if (global_var(245) >= 15) then begin
-        if (global_var(244) == 2) then begin
-            display_msg(message_str(413, 12));
+    set_global_var(GVAR_NCR_GENERIC_STATE, global_var(GVAR_NCR_GENERIC_STATE) + 1);
+    if (global_var(GVAR_NCR_GENERIC_STATE) >= 15) then begin
+        if (global_var(GVAR_8_BALL_TRASH_SECRET) == 2) then begin
+            display_mstr(12);
         end
         else begin
-            display_msg(message_str(413, 10));
+            display_mstr(10);
         end
         play_sfx("LEVELUP");
     end
@@ -128,7 +128,7 @@ end
 procedure map_enter_p_proc
 begin
     max_distance := random(4, 8);
-    if (not(metarule(14, 0))) then begin
+    if (not(map_first_run)) then begin
         set_local_var(3, 13073);
     end
 end
@@ -147,4 +147,3 @@ begin
         animation_is_called := 0;
     end
 end
-
