diff --git a/Mods/Olympus2207/Original/LVZTURET.ssl b/Mods/Olympus2207/ReDefine/LVZTURET.ssl
index f0149ba..ad36070 100644
--- a/Mods/Olympus2207/Original/LVZTURET.ssl
+++ b/Mods/Olympus2207/ReDefine/LVZTURET.ssl
@@ -26,8 +26,8 @@ begin
     if (local_var(1) == 0) then begin
         set_local_var(3, tile_num(self_obj));
         set_local_var(1, 1);
-        critter_add_trait(self_obj, 1, 6, 11);
-        critter_add_trait(self_obj, 1, 5, 71);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 11);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_TURRET);
     end
 end
 
@@ -35,7 +35,7 @@ procedure critter_p_proc
 begin
     if (not(combat_is_initialized) and obj_can_see_obj(self_obj, dude_obj)) then begin
         if (not(anim_busy(self_obj))) then begin
-            anim(self_obj, 1000, rotation_to_tile(tile_num(self_obj), tile_num(dude_obj)));
+            anim(self_obj, 1000, rotation_to_tile(tile_num(self_obj), dude_tile));
         end
     end
     if (map_var(0) == 1) then begin
@@ -43,7 +43,7 @@ begin
         set_local_var(2, 1);
     end
     if (obj_can_see_obj(self_obj, dude_obj)) then begin
-        if ((local_var(2) == 1) or (global_var(51) <= -31)) then begin
+        if ((local_var(2) == 1) or (global_var(GVAR_TOWN_REP_GECKO) <= -31)) then begin
             set_local_var(2, 0);
             attack_setup(self_obj, dude_obj);
         end
@@ -80,7 +80,7 @@ end
 procedure damage_p_proc
 begin
     if (source_obj == dude_obj) then begin
-        critter_add_trait(self_obj, 1, 6, 12);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 12);
         set_local_var(2, 1);
     end
 end
@@ -92,4 +92,3 @@ end
 procedure timed_event_p_proc
 begin
 end
-
