diff --git a/Mods/Olympus2207/Original/NWHENRY.ssl b/Mods/Olympus2207/ReDefine/NWHENRY.ssl
index 98d7890..efec200 100644
--- a/Mods/Olympus2207/Original/NWHENRY.ssl
+++ b/Mods/Olympus2207/ReDefine/NWHENRY.ssl
@@ -50,19 +50,19 @@ begin
         set_local_var(4, random(50, 60));
         set_local_var(5, random(30, 40));
         set_local_var(6, random(-30, 20));
-        critter_add_trait(self_obj, 1, 6, 8);
-        critter_add_trait(self_obj, 1, 5, 7);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 8);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_MAN_WEAK);
     end
 end
 
 procedure critter_p_proc
 begin
     if (local_var(7) == 0) then begin
-        if ((tile_distance(tile_num(self_obj), tile_num(dude_obj)) < 13) and (anim_busy(self_obj) == 0)) then begin
-            animate_move_obj_to_tile(dude_obj, tile_num(dude_obj), 0);
-            animate_move_obj_to_tile(self_obj, tile_num(dude_obj), 1);
+        if ((tile_distance(tile_num(self_obj), dude_tile) < 13) and (anim_busy(self_obj) == 0)) then begin
+            animate_move_obj_to_tile(dude_obj, dude_tile, 0);
+            animate_move_obj_to_tile(self_obj, dude_tile, 1);
         end
-        if (tile_distance(tile_num(self_obj), tile_num(dude_obj)) < 4) then begin
+        if (tile_distance(tile_num(self_obj), dude_tile) < 4) then begin
             dialogue_system_enter;
         end
     end
@@ -75,7 +75,7 @@ begin
         add_timer_event(self_obj, random(1000, 1000), 2);
     end
     if (obj_can_see_obj(self_obj, dude_obj)) then begin
-        if ((local_var(2) == 1) or (global_var(50) <= -31)) then begin
+        if ((local_var(2) == 1) or (global_var(GVAR_TOWN_REP_VAULT_CITY) <= -31)) then begin
             set_local_var(2, 0);
             attack_setup(self_obj, dude_obj);
         end
@@ -90,10 +90,10 @@ procedure description_p_proc
 begin
     script_overrides;
     if (local_var(7) == 0) then begin
-        display_msg(message_str(510, 2));
+        display_mstr(2);
     end
     else begin
-        display_msg(message_str(510, 4));
+        display_mstr(4);
     end
 end
 
@@ -101,10 +101,10 @@ procedure look_at_p_proc
 begin
     script_overrides;
     if (local_var(7) == 0) then begin
-        display_msg(message_str(510, 1));
+        display_mstr(1);
     end
     else begin
-        display_msg(message_str(510, 3));
+        display_mstr(3);
     end
 end
 
@@ -122,7 +122,7 @@ end
 procedure talk_p_proc
 begin
     if (not(anim_busy(self_obj))) then begin
-        anim(self_obj, 1000, rotation_to_tile(tile_num(self_obj), tile_num(dude_obj)));
+        anim(self_obj, 1000, rotation_to_tile(tile_num(self_obj), dude_tile));
     end
     start_gdialog(510, self_obj, -1, -1, -1);
     gsay_start;
@@ -139,14 +139,14 @@ end
 
 procedure destroy_p_proc
 begin
-    set_global_var(429, global_var(429) bwor 32);
+    set_global_var(GVAR_V13_STATUS_FLAGS, global_var(GVAR_V13_STATUS_FLAGS) bwor 32);
     if (source_obj == dude_obj) then begin
-        set_global_var(0, global_var(0) - 1);
+        set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) - 1);
         if ((50 == 50) or (50 == 59)) then begin
-            set_global_var(50, global_var(50) - 3);
+            set_global_var(GVAR_TOWN_REP_VAULT_CITY, global_var(GVAR_TOWN_REP_VAULT_CITY) - 3);
         end
         else begin
-            set_global_var(50, global_var(50) - 6);
+            set_global_var(GVAR_TOWN_REP_VAULT_CITY, global_var(GVAR_TOWN_REP_VAULT_CITY) - 6);
         end
     end
 end
@@ -159,51 +159,51 @@ begin
             set_sfall_global(13, 1);
         end
         else begin
-            if (has_trait(2, dude_obj, 13)) then begin
-                if (global_var(81) < get_pc_stat(1)) then begin
+            if (has_trait(TRAIT_TRAIT, dude_obj, TRAIT_sex_appeal)) then begin
+                if (global_var(GVAR_VAULT_CITIZENSHIP) < get_pc_stat(PCSTAT_level)) then begin
                     if (is_success(roll_vs_skill(dude_obj, 10, 0))) then begin
-                        set_global_var(98, global_var(98) + 3);
+                        set_global_var(GVAR_QUEST_FIND_THOMAS, global_var(GVAR_QUEST_FIND_THOMAS) + 3);
                     end
                     else begin
-                        set_global_var(98, global_var(98) + 1);
+                        set_global_var(GVAR_QUEST_FIND_THOMAS, global_var(GVAR_QUEST_FIND_THOMAS) + 1);
                     end
-                    if (global_var(98) >= (2 * get_pc_stat(1))) then begin
-                        set_global_var(98, 0);
-                        set_global_var(81, global_var(81) + 1);
-                        critter_mod_skill(dude_obj, 10, 1);
-                        display_msg(message_str(10, 410));
+                    if (global_var(GVAR_QUEST_FIND_THOMAS) >= (2 * get_pc_stat(PCSTAT_level))) then begin
+                        set_global_var(GVAR_QUEST_FIND_THOMAS, 0);
+                        set_global_var(GVAR_VAULT_CITIZENSHIP, global_var(GVAR_VAULT_CITIZENSHIP) + 1);
+                        critter_mod_skill(dude_obj, SKILL_STEAL, 1);
+                        display_msg(message_str(SCRIPT_STTEXT, 410));
                     end
                 end
             end
             steal_perk := 0;
             steal_gloves := 0;
-            if (has_trait(0, dude_obj, 37) != 0) then begin
+            if (has_trait(TRAIT_PERK, dude_obj, PERK_pickpocket) != 0) then begin
                 steal_perk := steal_perk + 15;
             end
-            if (has_trait(0, dude_obj, 33) != 0) then begin
+            if (has_trait(TRAIT_PERK, dude_obj, PERK_master_thief) != 0) then begin
                 steal_perk := steal_perk + 30;
             end
-            if ((obj_pid(critter_inven_obj(dude_obj, 1)) == 245) or (obj_pid(critter_inven_obj(dude_obj, 2)) == 245)) then begin
+            if ((obj_pid(critter_inven_obj(dude_obj, INVEN_TYPE_RIGHT_HAND)) == 245) or (obj_pid(critter_inven_obj(dude_obj, INVEN_TYPE_LEFT_HAND)) == 245)) then begin
                 steal_gloves := 10;
             end
-            if (random(0, 100) <= ((has_skill(dude_obj, 10) / 2) + (((3 * get_critter_stat(dude_obj, 5)) + (((get_critter_stat(dude_obj, 6) - 4) * 4) - (difficulty_level * 4)) + (2 * steal_perk) + steal_gloves) / (local_var(4) / 25)))) then begin
-                display_msg(message_str(10, random(145, 148)));
+            if (random(0, 100) <= ((dude_skill(SKILL_STEAL) / 2) + (((3 * get_critter_stat(dude_obj, STAT_ag)) + (((dude_luck - 4) * 4) - (difficulty_level * 4)) + (2 * steal_perk) + steal_gloves) / (local_var(4) / 25)))) then begin
+                display_msg(message_str(SCRIPT_STTEXT, random(145, 148)));
             end
             else begin
                 script_overrides;
-                if (random(0, 100) <= (25 - floor((((get_critter_stat(dude_obj, 6) - 4) * 4) - (difficulty_level * 4)) / 3))) then begin
-                    float_msg(self_obj, message_str(10, random(206, 209)), 2);
+                if (random(0, 100) <= (25 - floor((((dude_luck - 4) * 4) - (difficulty_level * 4)) / 3))) then begin
+                    float_msg(self_obj, message_str(SCRIPT_STTEXT, random(206, 209)), FLOAT_MSG_RED);
                     set_local_var(2, 1);
                 end
                 else begin
-                    display_msg(message_str(10, random(200, 203)));
+                    display_msg(message_str(SCRIPT_STTEXT, random(200, 203)));
                 end
             end
         end
     end
     if (action_being_used == 12) then begin
         script_overrides;
-        display_msg(message_str(10, 204) + ((has_skill(dude_obj, 10) / 2) + (((3 * get_critter_stat(dude_obj, 5)) + (((get_critter_stat(dude_obj, 6) - 4) * 4) - (difficulty_level * 4)) + (2 * steal_perk) + steal_gloves) / (local_var(4) / 25))) + message_str(10, 205));
+        display_msg(message_str(SCRIPT_STTEXT, 204) + ((dude_skill(SKILL_STEAL) / 2) + (((3 * get_critter_stat(dude_obj, STAT_ag)) + (((dude_luck - 4) * 4) - (difficulty_level * 4)) + (2 * steal_perk) + steal_gloves) / (local_var(4) / 25))) + message_str(SCRIPT_STTEXT, 205));
     end
 end
 
@@ -237,106 +237,106 @@ procedure Node001
 begin
     set_local_var(7, 1);
     game_ui_enable;
-    gsay_reply(510, 10);
-    gsay_option(510, 11, Node002, 50);
-    gsay_option(510, 12, Node004, 50);
-    gsay_option(510, 13, Node005, 50);
-    gsay_option(510, 14, Node999, 50);
+    Reply(10);
+    giq_option(1, NAME, 11, Node002, NEUTRAL_REACTION);
+    giq_option(1, NAME, 12, Node004, NEUTRAL_REACTION);
+    giq_option(1, NAME, 13, Node005, NEUTRAL_REACTION);
+    giq_option(1, NAME, 14, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node002
 begin
-    gsay_reply(510, 20);
-    gsay_option(510, 21, Node005, 50);
-    gsay_option(510, 22, Node003, 50);
+    Reply(20);
+    giq_option(1, NAME, 21, Node005, NEUTRAL_REACTION);
+    giq_option(1, NAME, 22, Node003, NEUTRAL_REACTION);
 end
 
 procedure Node003
 begin
-    gsay_reply(510, 30);
-    gsay_option(510, 31, Node005, 50);
-    gsay_option(510, 32, Node999, 50);
+    Reply(30);
+    giq_option(1, NAME, 31, Node005, NEUTRAL_REACTION);
+    giq_option(1, NAME, 32, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node004
 begin
-    gsay_reply(510, 40);
-    gsay_option(510, 41, Node005, 50);
-    gsay_option(510, 42, Node003, 50);
+    Reply(40);
+    giq_option(1, NAME, 41, Node005, NEUTRAL_REACTION);
+    giq_option(1, NAME, 42, Node003, NEUTRAL_REACTION);
 end
 
 procedure Node005
 begin
-    gsay_reply(510, 50);
-    gsay_option(510, 51, Node006, 50);
-    gsay_option(510, 52, Node007, 50);
+    Reply(50);
+    giq_option(1, NAME, 51, Node006, NEUTRAL_REACTION);
+    giq_option(1, NAME, 52, Node007, NEUTRAL_REACTION);
 end
 
 procedure Node006
 begin
-    gsay_reply(510, 60);
-    gsay_option(510, 61, Node008, 50);
+    Reply(60);
+    giq_option(1, NAME, 61, Node008, NEUTRAL_REACTION);
 end
 
 procedure Node007
 begin
-    gsay_reply(510, 70);
-    gsay_option(510, 71, Node006, 50);
+    Reply(70);
+    giq_option(1, NAME, 71, Node006, NEUTRAL_REACTION);
 end
 
 procedure Node008
 begin
-    gsay_reply(510, 80);
-    gsay_option(510, 81, Node009, 50);
+    Reply(80);
+    giq_option(1, NAME, 81, Node009, NEUTRAL_REACTION);
 end
 
 procedure Node009
 begin
-    gsay_reply(510, 90);
-    gsay_option(510, 91, Node012, 50);
-    gsay_option(510, 92, Node012, 50);
-    gsay_option(510, 93, Node011, 50);
+    Reply(90);
+    giq_option(1, NAME, 91, Node012, NEUTRAL_REACTION);
+    giq_option(1, NAME, 92, Node012, NEUTRAL_REACTION);
+    giq_option(1, NAME, 93, Node011, NEUTRAL_REACTION);
 end
 
 procedure Node010
 begin
-    if (global_var(421) == 0) then begin
-        gsay_reply(510, 100);
-        gsay_option(510, 101, Node999, 50);
-        gsay_option(510, 102, Node005, 50);
+    if (global_var(GVAR_ENDGAME_MOVIE_VAULT_15) == 0) then begin
+        Reply(100);
+        giq_option(1, NAME, 101, Node999, NEUTRAL_REACTION);
+        giq_option(1, NAME, 102, Node005, NEUTRAL_REACTION);
     end
     else begin
-        if (global_var(421) == 1) then begin
-            gsay_reply(510, 103);
-            gsay_option(510, 104, Node999, 50);
+        if (global_var(GVAR_ENDGAME_MOVIE_VAULT_15) == 1) then begin
+            Reply(103);
+            giq_option(1, NAME, 104, Node999, NEUTRAL_REACTION);
         end
         else begin
-            gsay_reply(510, 105);
-            gsay_option(510, 106, Node999, 50);
+            Reply(105);
+            giq_option(1, NAME, 106, Node999, NEUTRAL_REACTION);
         end
     end
 end
 
 procedure Node011
 begin
-    gsay_reply(510, 110);
-    gsay_option(510, 111, NodeQA, 50);
-    gsay_option(510, 112, Node999, 50);
-    gsay_option(510, 113, Node999, 50);
+    Reply(110);
+    giq_option(1, NAME, 111, NodeQA, NEUTRAL_REACTION);
+    giq_option(1, NAME, 112, Node999, NEUTRAL_REACTION);
+    giq_option(1, NAME, 113, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node012
 begin
-    gsay_reply(510, 120);
-    gsay_option(510, 121, Node011, 50);
+    Reply(120);
+    giq_option(1, NAME, 121, Node011, NEUTRAL_REACTION);
 end
 
 procedure NodeQA
 begin
-    if (global_var(421) == 0) then begin
-        set_global_var(421, 1);
+    if (global_var(GVAR_ENDGAME_MOVIE_VAULT_15) == 0) then begin
+        set_global_var(GVAR_ENDGAME_MOVIE_VAULT_15, 1);
     end
-    mark_area_known(0, 23, 1);
+    mark_area_known(MARK_TYPE_TOWN, 23, MARK_STATE_KNOWN);
 end
 
 procedure Node999
@@ -355,4 +355,3 @@ begin
     gdialog_mod_barter(-(dude_objective_reputation + local_var(6)));
     call Node010();
 end
-
