diff --git a/Mods/Olympus2207/Original/NWSAM.ssl b/Mods/Olympus2207/ReDefine/NWSAM.ssl
index 9f28e18..63948e1 100644
--- a/Mods/Olympus2207/Original/NWSAM.ssl
+++ b/Mods/Olympus2207/ReDefine/NWSAM.ssl
@@ -111,70 +111,70 @@ begin
         set_local_var(4, random(60, 70));
         set_local_var(5, random(30, 40));
         set_local_var(6, random(-20, 10));
-        critter_add_trait(self_obj, 1, 6, 8);
-        critter_add_trait(self_obj, 1, 5, 19);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 8);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_UNDEAD_NORMAL);
         if (random(0, 100) <= 40) then begin
-            add_obj_to_inven(self_obj, create_object_sid(146, 1, 2, -1));
+            add_obj_to_inven(self_obj, create_object(146, 1, 2));
         end
         else begin
-            add_obj_to_inven(self_obj, create_object_sid(65, 1, 2, -1));
+            add_obj_to_inven(self_obj, create_object(65, 1, 2));
         end
         if (random(0, 100) <= 40) then begin
-            add_obj_to_inven(self_obj, create_object_sid(108, 1, 2, -1));
+            add_obj_to_inven(self_obj, create_object(108, 1, 2));
         end
         else begin
-            add_obj_to_inven(self_obj, create_object_sid(139, 1, 2, -1));
+            add_obj_to_inven(self_obj, create_object(139, 1, 2));
         end
         if (random(0, 100) <= 40) then begin
-            add_obj_to_inven(self_obj, create_object_sid(214, 1, 2, -1));
+            add_obj_to_inven(self_obj, create_object(214, 1, 2));
         end
         else begin
-            add_obj_to_inven(self_obj, create_object_sid(63, 1, 2, -1));
+            add_obj_to_inven(self_obj, create_object(63, 1, 2));
         end
         if (random(0, 100) <= 60) then begin
-            add_obj_to_inven(self_obj, create_object_sid(67, 1, 2, -1));
+            add_obj_to_inven(self_obj, create_object(67, 1, 2));
         end
         else begin
-            add_obj_to_inven(self_obj, create_object_sid(132, 1, 2, -1));
+            add_obj_to_inven(self_obj, create_object(132, 1, 2));
         end
         if (random(0, 100) <= 50) then begin
-            add_obj_to_inven(self_obj, create_object_sid(147, 1, 2, -1));
+            add_obj_to_inven(self_obj, create_object(147, 1, 2));
         end
         else begin
-            add_obj_to_inven(self_obj, create_object_sid(38, 1, 2, -1));
+            add_obj_to_inven(self_obj, create_object(38, 1, 2));
         end
         if (random(0, 100) <= 33) then begin
-            add_obj_to_inven(self_obj, create_object_sid(226, 1, 2, -1));
+            add_obj_to_inven(self_obj, create_object(226, 1, 2));
         end
         if (random(0, 100) <= 33) then begin
-            add_obj_to_inven(self_obj, create_object_sid(225, 1, 2, -1));
+            add_obj_to_inven(self_obj, create_object(225, 1, 2));
         end
         if (random(0, 100) <= 33) then begin
-            add_obj_to_inven(self_obj, create_object_sid(227, 1, 2, -1));
+            add_obj_to_inven(self_obj, create_object(227, 1, 2));
         end
         if (random(0, 100) <= 70) then begin
-            add_mult_objs_to_inven(self_obj, create_object_sid(2, 1, 2, -1), random(5, 40));
+            add_mult_objs_to_inven(self_obj, create_object(2, 1, 2), random(5, 40));
         end
         if (random(0, 100) <= 70) then begin
-            add_mult_objs_to_inven(self_obj, create_object_sid(5, 1, 2, -1), random(5, 40));
+            add_mult_objs_to_inven(self_obj, create_object(5, 1, 2), random(5, 40));
         end
         if (random(0, 100) <= 50) then begin
-            add_mult_objs_to_inven(self_obj, create_object_sid(7, 1, 2, -1), random(5, 40));
+            add_mult_objs_to_inven(self_obj, create_object(7, 1, 2), random(5, 40));
         end
         if (random(0, 100) <= 50) then begin
-            add_mult_objs_to_inven(self_obj, create_object_sid(9, 1, 2, -1), random(5, 40));
+            add_mult_objs_to_inven(self_obj, create_object(9, 1, 2), random(5, 40));
         end
         if (random(0, 100) <= 70) then begin
-            add_mult_objs_to_inven(self_obj, create_object_sid(14, 1, 2, -1), random(5, 40));
+            add_mult_objs_to_inven(self_obj, create_object(14, 1, 2), random(5, 40));
         end
         if (random(0, 100) <= 50) then begin
-            add_mult_objs_to_inven(self_obj, create_object_sid(33, 1, 2, -1), random(1, 3));
+            add_mult_objs_to_inven(self_obj, create_object(33, 1, 2), random(1, 3));
         end
         if (random(0, 100) <= 50) then begin
-            add_mult_objs_to_inven(self_obj, create_object_sid(25, 1, 2, -1), random(2, 6));
+            add_mult_objs_to_inven(self_obj, create_object(25, 1, 2), random(2, 6));
         end
         if (random(0, 100) <= 70) then begin
-            add_mult_objs_to_inven(self_obj, create_object_sid(20, 1, 2, -1), random(5, 40));
+            add_mult_objs_to_inven(self_obj, create_object(20, 1, 2), random(5, 40));
         end
         item_caps_adjust(self_obj, random(10, 30));
     end
@@ -188,10 +188,10 @@ begin
     end
     if (float_is_called == 0) then begin
         float_is_called := 1;
-        add_timer_event(self_obj, random(1000, 1000), 2);
+        add_timer_event(self_obj, 1000, 2);
     end
     if (obj_can_see_obj(self_obj, dude_obj)) then begin
-        if ((local_var(2) == 1) or (global_var(50) <= -31)) then begin
+        if ((local_var(2) == 1) or (global_var(GVAR_TOWN_REP_VAULT_CITY) <= -31)) then begin
             set_local_var(2, 0);
             attack_setup(self_obj, dude_obj);
         end
@@ -206,10 +206,10 @@ procedure description_p_proc
 begin
     script_overrides;
     if (local_var(7) == 0) then begin
-        display_msg(message_str(190, 2));
+        display_mstr(2);
     end
     else begin
-        display_msg(message_str(190, 4));
+        display_mstr(4);
     end
 end
 
@@ -217,10 +217,10 @@ procedure look_at_p_proc
 begin
     script_overrides;
     if (local_var(7) == 0) then begin
-        display_msg(message_str(190, 1));
+        display_mstr(1);
     end
     else begin
-        display_msg(message_str(190, 3));
+        display_mstr(3);
     end
 end
 
@@ -238,10 +238,10 @@ end
 procedure talk_p_proc
 begin
     if (not(anim_busy(self_obj))) then begin
-        anim(self_obj, 1000, rotation_to_tile(tile_num(self_obj), tile_num(dude_obj)));
+        anim(self_obj, 1000, rotation_to_tile(tile_num(self_obj), dude_tile));
     end
     if (local_var(10) == 1) then begin
-        float_msg(self_obj, message_str(190, random(751, 753)), 0);
+        float_msg(self_obj, mstr(random(751, 753)), FLOAT_MSG_NORMAL);
     end
     else begin
         start_gdialog(190, self_obj, 4, 10, -1);
@@ -260,18 +260,18 @@ end
 procedure destroy_p_proc
 begin
     if (source_obj == dude_obj) then begin
-        set_global_var(0, global_var(0) - 1);
+        set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) - 1);
         if ((50 == 50) or (50 == 59)) then begin
-            set_global_var(50, global_var(50) - 3);
+            set_global_var(GVAR_TOWN_REP_VAULT_CITY, global_var(GVAR_TOWN_REP_VAULT_CITY) - 3);
         end
         else begin
-            set_global_var(50, global_var(50) - 6);
+            set_global_var(GVAR_TOWN_REP_VAULT_CITY, global_var(GVAR_TOWN_REP_VAULT_CITY) - 6);
         end
     end
-    if (global_var(151) == 1) then begin
-        set_global_var(151, 0);
+    if (global_var(GVAR_SIERRA_BASE_ALERT) == 1) then begin
+        set_global_var(GVAR_SIERRA_BASE_ALERT, 0);
     end
-    set_global_var(69, global_var(69) bwor 536870912);
+    set_global_var(GVAR_TORR_HARMED, global_var(GVAR_TORR_HARMED) bwor 536870912);
 end
 
 procedure use_skill_on_p_proc
@@ -282,51 +282,51 @@ begin
             set_sfall_global(13, 1);
         end
         else begin
-            if (has_trait(2, dude_obj, 13)) then begin
-                if (global_var(81) < get_pc_stat(1)) then begin
+            if (has_trait(TRAIT_TRAIT, dude_obj, TRAIT_sex_appeal)) then begin
+                if (global_var(GVAR_VAULT_CITIZENSHIP) < get_pc_stat(PCSTAT_level)) then begin
                     if (is_success(roll_vs_skill(dude_obj, 10, 0))) then begin
-                        set_global_var(98, global_var(98) + 3);
+                        set_global_var(GVAR_QUEST_FIND_THOMAS, global_var(GVAR_QUEST_FIND_THOMAS) + 3);
                     end
                     else begin
-                        set_global_var(98, global_var(98) + 1);
+                        set_global_var(GVAR_QUEST_FIND_THOMAS, global_var(GVAR_QUEST_FIND_THOMAS) + 1);
                     end
-                    if (global_var(98) >= (2 * get_pc_stat(1))) then begin
-                        set_global_var(98, 0);
-                        set_global_var(81, global_var(81) + 1);
-                        critter_mod_skill(dude_obj, 10, 1);
-                        display_msg(message_str(10, 410));
+                    if (global_var(GVAR_QUEST_FIND_THOMAS) >= (2 * get_pc_stat(PCSTAT_level))) then begin
+                        set_global_var(GVAR_QUEST_FIND_THOMAS, 0);
+                        set_global_var(GVAR_VAULT_CITIZENSHIP, global_var(GVAR_VAULT_CITIZENSHIP) + 1);
+                        critter_mod_skill(dude_obj, SKILL_STEAL, 1);
+                        display_msg(message_str(SCRIPT_STTEXT, 410));
                     end
                 end
             end
             steal_perk := 0;
             steal_gloves := 0;
-            if (has_trait(0, dude_obj, 37) != 0) then begin
+            if (has_trait(TRAIT_PERK, dude_obj, PERK_pickpocket) != 0) then begin
                 steal_perk := steal_perk + 15;
             end
-            if (has_trait(0, dude_obj, 33) != 0) then begin
+            if (has_trait(TRAIT_PERK, dude_obj, PERK_master_thief) != 0) then begin
                 steal_perk := steal_perk + 30;
             end
-            if ((obj_pid(critter_inven_obj(dude_obj, 1)) == 245) or (obj_pid(critter_inven_obj(dude_obj, 2)) == 245)) then begin
+            if ((obj_pid(critter_inven_obj(dude_obj, INVEN_TYPE_RIGHT_HAND)) == 245) or (obj_pid(critter_inven_obj(dude_obj, INVEN_TYPE_LEFT_HAND)) == 245)) then begin
                 steal_gloves := 10;
             end
-            if (random(0, 100) <= ((has_skill(dude_obj, 10) / 2) + (((3 * get_critter_stat(dude_obj, 5)) + (((get_critter_stat(dude_obj, 6) - 4) * 4) - (difficulty_level * 4)) + (2 * steal_perk) + steal_gloves) / (local_var(4) / 25)))) then begin
-                display_msg(message_str(10, random(145, 148)));
+            if (random(0, 100) <= ((dude_skill(SKILL_STEAL) / 2) + (((3 * get_critter_stat(dude_obj, STAT_ag)) + (((dude_luck - 4) * 4) - (difficulty_level * 4)) + (2 * steal_perk) + steal_gloves) / (local_var(4) / 25)))) then begin
+                display_msg(message_str(SCRIPT_STTEXT, random(145, 148)));
             end
             else begin
                 script_overrides;
-                if (random(0, 100) <= (25 - floor((((get_critter_stat(dude_obj, 6) - 4) * 4) - (difficulty_level * 4)) / 3))) then begin
-                    float_msg(self_obj, message_str(10, random(206, 209)), 2);
+                if (random(0, 100) <= (25 - floor((((dude_luck - 4) * 4) - (difficulty_level * 4)) / 3))) then begin
+                    float_msg(self_obj, message_str(SCRIPT_STTEXT, random(206, 209)), FLOAT_MSG_RED);
                     set_local_var(2, 1);
                 end
                 else begin
-                    display_msg(message_str(10, random(200, 203)));
+                    display_msg(message_str(SCRIPT_STTEXT, random(200, 203)));
                 end
             end
         end
     end
     if (action_being_used == 12) then begin
         script_overrides;
-        display_msg(message_str(10, 204) + ((has_skill(dude_obj, 10) / 2) + (((3 * get_critter_stat(dude_obj, 5)) + (((get_critter_stat(dude_obj, 6) - 4) * 4) - (difficulty_level * 4)) + (2 * steal_perk) + steal_gloves) / (local_var(4) / 25))) + message_str(10, 205));
+        display_msg(message_str(SCRIPT_STTEXT, 204) + ((dude_skill(SKILL_STEAL) / 2) + (((3 * get_critter_stat(dude_obj, STAT_ag)) + (((dude_luck - 4) * 4) - (difficulty_level * 4)) + (2 * steal_perk) + steal_gloves) / (local_var(4) / 25))) + message_str(SCRIPT_STTEXT, 205));
     end
 end
 
@@ -343,7 +343,7 @@ end
 
 procedure map_enter_p_proc
 begin
-    if (global_var(279) == 1) then begin
+    if (global_var(GVAR_CARAVAN_GUARDS_TOTAL) == 1) then begin
         move_to(self_obj, 1, 2);
     end
     if (local_var(10) == 1) then begin
@@ -373,441 +373,441 @@ procedure Node001
 begin
     set_local_var(7, 1);
     already_talk := 1;
-    gsay_reply(190, 10);
-    gsay_option(190, 11, Node002, 50);
-    giq_option(1, 190, 12, Node003, 51);
+    Reply(10);
+    giq_option(1, NAME, 11, Node002, NEUTRAL_REACTION);
+    giq_option(1, NAME, 12, Node003, BAD_REACTION);
 end
 
 procedure Node002
 begin
-    gsay_reply(190, 20);
-    gsay_option(190, 21, Node010, 50);
+    Reply(20);
+    giq_option(1, NAME, 21, Node010, NEUTRAL_REACTION);
 end
 
 procedure Node003
 begin
-    gsay_reply(190, 30);
-    gsay_option(190, 31, Node010, 50);
-    gsay_option(190, 32, Node004, 50);
+    Reply(30);
+    giq_option(1, NAME, 31, Node010, NEUTRAL_REACTION);
+    giq_option(1, NAME, 32, Node004, NEUTRAL_REACTION);
 end
 
 procedure Node004
 begin
-    gsay_reply(190, 40);
-    gsay_option(190, 41, Node010, 50);
-    gsay_option(190, 42, Node010, 50);
+    Reply(40);
+    giq_option(1, NAME, 41, Node010, NEUTRAL_REACTION);
+    giq_option(1, NAME, 42, Node010, NEUTRAL_REACTION);
 end
 
 procedure Node010
 begin
     if (already_talk == 0) then begin
         already_talk := 1;
-        if (global_var(221) == 3) then begin
-            gsay_reply(190, 1002);
+        if (global_var(GVAR_HOLODISK_MB_LEVEL_1) == 3) then begin
+            Reply(1002);
         end
         else begin
-            if (global_var(221) == 2) then begin
-                gsay_reply(190, 1003);
+            if (global_var(GVAR_HOLODISK_MB_LEVEL_1) == 2) then begin
+                Reply(1003);
             end
             else begin
-                if (global_var(221) == 1) then begin
-                    gsay_reply(190, 1004);
+                if (global_var(GVAR_HOLODISK_MB_LEVEL_1) == 1) then begin
+                    Reply(1004);
                 end
                 else begin
-                    gsay_reply(190, 1001);
+                    Reply(1001);
                 end
             end
         end
     end
     else begin
-        gsay_reply(190, 1000);
+        Reply(1000);
     end
-    if (not(global_var(221) == 3)) then begin
-        gsay_option(190, 101, Node054, 50);
-        giq_option(1, 190, 103, Node011, 49);
-        gsay_option(190, 104, Node019, 50);
-        gsay_option(190, 105, Node020, 50);
+    if (not(global_var(GVAR_HOLODISK_MB_LEVEL_1) == 3)) then begin
+        giq_option(1, NAME, 101, Node054, NEUTRAL_REACTION);
+        giq_option(1, NAME, 103, Node011, GOOD_REACTION);
+        giq_option(1, NAME, 104, Node019, NEUTRAL_REACTION);
+        giq_option(1, NAME, 105, Node020, NEUTRAL_REACTION);
     end
-    if (global_var(151) == 1) then begin
-        gsay_option(190, 102, Node051, 50);
+    if (global_var(GVAR_SIERRA_BASE_ALERT) == 1) then begin
+        giq_option(1, NAME, 102, Node051, NEUTRAL_REACTION);
     end
-    if ((global_var(143) == 1) and (global_var(208) == 0)) then begin
-        gsay_option(190, 106, Node023, 50);
+    if ((global_var(GVAR_QUEST_VAULT_VILLAGE) == 1) and (global_var(GVAR_NCR_SCMERK_PERSONAL_ENEMY) == 0)) then begin
+        giq_option(1, NAME, 106, Node023, NEUTRAL_REACTION);
     end
-    if (global_var(144) == 1) then begin
-        giq_option(1, 190, 107, Node058, 51);
+    if (global_var(GVAR_VAULT_MONSTER_COUNT) == 1) then begin
+        giq_option(1, NAME, 107, Node058, BAD_REACTION);
     end
-    if (global_var(144) == 2) then begin
-        gsay_option(190, 108, Node059, 50);
+    if (global_var(GVAR_VAULT_MONSTER_COUNT) == 2) then begin
+        giq_option(1, NAME, 108, Node059, NEUTRAL_REACTION);
     end
-    gsay_option(190, 109, Node999, 50);
+    giq_option(1, NAME, 109, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node011
 begin
-    gsay_reply(190, 110);
-    gsay_option(190, 111, Node012, 50);
-    gsay_option(190, 112, Node016, 50);
-    gsay_option(190, 113, Node015, 50);
-    gsay_option(190, 114, Node020, 50);
-    gsay_option(190, 115, Node010, 50);
+    Reply(110);
+    giq_option(1, NAME, 111, Node012, NEUTRAL_REACTION);
+    giq_option(1, NAME, 112, Node016, NEUTRAL_REACTION);
+    giq_option(1, NAME, 113, Node015, NEUTRAL_REACTION);
+    giq_option(1, NAME, 114, Node020, NEUTRAL_REACTION);
+    giq_option(1, NAME, 115, Node010, NEUTRAL_REACTION);
 end
 
 procedure Node012
 begin
-    gsay_reply(190, 120);
-    gsay_option(190, 121, Node020, 50);
-    gsay_option(190, 122, Node015, 50);
-    gsay_option(190, 123, Node013, 50);
-    gsay_option(190, 124, Node010, 50);
+    Reply(120);
+    giq_option(1, NAME, 121, Node020, NEUTRAL_REACTION);
+    giq_option(1, NAME, 122, Node015, NEUTRAL_REACTION);
+    giq_option(1, NAME, 123, Node013, NEUTRAL_REACTION);
+    giq_option(1, NAME, 124, Node010, NEUTRAL_REACTION);
 end
 
 procedure Node013
 begin
-    gsay_reply(190, 130);
-    gsay_option(190, 131, Node018, 50);
-    gsay_option(190, 132, Node014, 50);
-    gsay_option(190, 133, Node010, 50);
+    Reply(130);
+    giq_option(1, NAME, 131, Node018, NEUTRAL_REACTION);
+    giq_option(1, NAME, 132, Node014, NEUTRAL_REACTION);
+    giq_option(1, NAME, 133, Node010, NEUTRAL_REACTION);
 end
 
 procedure Node014
 begin
-    gsay_reply(190, 140);
-    gsay_option(190, 141, Node018, 50);
-    gsay_option(190, 142, Node015, 50);
-    gsay_option(190, 143, Node010, 50);
+    Reply(140);
+    giq_option(1, NAME, 141, Node018, NEUTRAL_REACTION);
+    giq_option(1, NAME, 142, Node015, NEUTRAL_REACTION);
+    giq_option(1, NAME, 143, Node010, NEUTRAL_REACTION);
 end
 
 procedure Node015
 begin
-    gsay_reply(190, 150);
-    gsay_option(190, 151, Node018, 50);
-    gsay_option(190, 152, Node010, 50);
+    Reply(150);
+    giq_option(1, NAME, 151, Node018, NEUTRAL_REACTION);
+    giq_option(1, NAME, 152, Node010, NEUTRAL_REACTION);
 end
 
 procedure Node016
 begin
-    gsay_reply(190, 160);
-    gsay_option(190, 161, Node021, 50);
-    gsay_option(190, 162, Node023, 50);
-    gsay_option(190, 163, Node017, 50);
-    gsay_option(190, 164, Node010, 50);
+    Reply(160);
+    giq_option(1, NAME, 161, Node021, NEUTRAL_REACTION);
+    giq_option(1, NAME, 162, Node023, NEUTRAL_REACTION);
+    giq_option(1, NAME, 163, Node017, NEUTRAL_REACTION);
+    giq_option(1, NAME, 164, Node010, NEUTRAL_REACTION);
 end
 
 procedure Node017
 begin
-    gsay_reply(190, 170);
-    gsay_option(190, 171, Node018, 50);
+    Reply(170);
+    giq_option(1, NAME, 171, Node018, NEUTRAL_REACTION);
 end
 
 procedure Node018
 begin
-    gsay_reply(190, 180);
-    gsay_option(190, 181, Node022, 50);
-    gsay_option(190, 182, Node021, 50);
-    gsay_option(190, 183, Node010, 50);
+    Reply(180);
+    giq_option(1, NAME, 181, Node022, NEUTRAL_REACTION);
+    giq_option(1, NAME, 182, Node021, NEUTRAL_REACTION);
+    giq_option(1, NAME, 183, Node010, NEUTRAL_REACTION);
 end
 
 procedure Node019
 begin
-    gsay_reply(190, 190);
-    gsay_option(190, 191, Node023, 50);
-    gsay_option(190, 192, Node020, 50);
-    gsay_option(190, 193, Node010, 50);
+    Reply(190);
+    giq_option(1, NAME, 191, Node023, NEUTRAL_REACTION);
+    giq_option(1, NAME, 192, Node020, NEUTRAL_REACTION);
+    giq_option(1, NAME, 193, Node010, NEUTRAL_REACTION);
 end
 
 procedure Node020
 begin
-    gsay_reply(190, 200);
-    gsay_option(190, 201, Node010, 50);
+    Reply(200);
+    giq_option(1, NAME, 201, Node010, NEUTRAL_REACTION);
 end
 
 procedure Node021
 begin
-    gsay_reply(190, 210);
-    gsay_option(190, 211, Node018, 50);
-    gsay_option(190, 212, Node023, 50);
-    gsay_option(190, 213, Node022, 50);
-    gsay_option(190, 214, Node010, 50);
+    Reply(210);
+    giq_option(1, NAME, 211, Node018, NEUTRAL_REACTION);
+    giq_option(1, NAME, 212, Node023, NEUTRAL_REACTION);
+    giq_option(1, NAME, 213, Node022, NEUTRAL_REACTION);
+    giq_option(1, NAME, 214, Node010, NEUTRAL_REACTION);
 end
 
 procedure Node022
 begin
-    gsay_reply(190, 220);
-    giq_option(1, 190, 221, Node023, 51);
-    if (global_var(369) == 0) then begin
-        giq_option(1, 190, 222, Node027, 49);
+    Reply(220);
+    giq_option(1, NAME, 221, Node023, BAD_REACTION);
+    if (global_var(GVAR_DUMAR_STATUS) == 0) then begin
+        giq_option(1, NAME, 222, Node027, GOOD_REACTION);
     end
-    gsay_option(190, 223, Node010, 50);
+    giq_option(1, NAME, 223, Node010, NEUTRAL_REACTION);
 end
 
 procedure Node023
 begin
-    if (global_var(143) < 2) then begin
-        gsay_reply(190, 2301);
+    if (global_var(GVAR_QUEST_VAULT_VILLAGE) < 2) then begin
+        Reply(2301);
     end
     else begin
-        if ((global_var(143) == 2) and (local_var(8) == 0)) then begin
-            gsay_reply(190, 2302);
+        if ((global_var(GVAR_QUEST_VAULT_VILLAGE) == 2) and (local_var(8) == 0)) then begin
+            Reply(2302);
         end
         else begin
-            gsay_reply(190, 2303);
+            Reply(2303);
         end
     end
-    if (global_var(143) != 2) then begin
-        gsay_option(190, 231, Node024, 50);
+    if (global_var(GVAR_QUEST_VAULT_VILLAGE) != 2) then begin
+        giq_option(1, NAME, 231, Node024, NEUTRAL_REACTION);
     end
-    if (global_var(143) == 1) then begin
-        gsay_option(190, 232, Node025, 50);
+    if (global_var(GVAR_QUEST_VAULT_VILLAGE) == 1) then begin
+        giq_option(1, NAME, 232, Node025, NEUTRAL_REACTION);
     end
-    gsay_option(190, 233, Node010, 50);
+    giq_option(1, NAME, 233, Node010, NEUTRAL_REACTION);
 end
 
 procedure Node024
 begin
-    gsay_reply(190, 240);
-    if (global_var(143) == 1) then begin
-        giq_option(1, 190, 241, Node025, 51);
+    Reply(240);
+    if (global_var(GVAR_QUEST_VAULT_VILLAGE) == 1) then begin
+        giq_option(1, NAME, 241, Node025, BAD_REACTION);
     end
-    gsay_option(190, 242, Node010, 50);
+    giq_option(1, NAME, 242, Node010, NEUTRAL_REACTION);
 end
 
 procedure Node025
 begin
-    gsay_reply(190, 250);
-    gsay_option(190, 251, Node010, 50);
-    if (has_skill(dude_obj, 14) >= 75) then begin
-        giq_option(1, 190, 252, Node026, 49);
+    Reply(250);
+    giq_option(1, NAME, 251, Node010, NEUTRAL_REACTION);
+    if (dude_skill(SKILL_SPEECH) >= 75) then begin
+        giq_option(1, NAME, 252, Node026, GOOD_REACTION);
     end
-    if ((get_critter_stat(dude_obj, 34) == 1) and (global_var(223) == 1)) then begin
-        gsay_option(190, 253, Node047, 50);
+    if ((dude_gender == 1) and (global_var(GVAR_HOLODISK_MB_LEVEL_3) == 1)) then begin
+        giq_option(1, NAME, 253, Node047, NEUTRAL_REACTION);
     end
 end
 
 procedure Node026
 begin
     set_local_var(9, 1);
-    set_global_var(208, 1);
-    gsay_reply(190, 260);
-    giq_option(1, 190, 261, Node010, 49);
-    gsay_option(190, 262, Node999, 50);
+    set_global_var(GVAR_NCR_SCMERK_PERSONAL_ENEMY, 1);
+    Reply(260);
+    giq_option(1, NAME, 261, Node010, GOOD_REACTION);
+    giq_option(1, NAME, 262, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node027
 begin
-    gsay_reply(190, 270);
-    if (global_var(143) != 2) then begin
-        gsay_option(190, 271, Node028, 50);
+    Reply(270);
+    if (global_var(GVAR_QUEST_VAULT_VILLAGE) != 2) then begin
+        giq_option(1, NAME, 271, Node028, NEUTRAL_REACTION);
     end
-    if ((global_var(143) == 2) and (local_var(8) == 0)) then begin
-        gsay_option(190, 272, Node029, 50);
+    if ((global_var(GVAR_QUEST_VAULT_VILLAGE) == 2) and (local_var(8) == 0)) then begin
+        giq_option(1, NAME, 272, Node029, NEUTRAL_REACTION);
     end
-    if ((global_var(143) == 2) and (local_var(8) == 1)) then begin
-        giq_option(1, 190, 273, Node030, 51);
+    if ((global_var(GVAR_QUEST_VAULT_VILLAGE) == 2) and (local_var(8) == 1)) then begin
+        giq_option(1, NAME, 273, Node030, BAD_REACTION);
     end
 end
 
 procedure Node028
 begin
-    gsay_reply(190, 280);
-    gsay_option(190, 281, Node023, 50);
-    gsay_option(190, 282, Node010, 50);
-    gsay_option(190, 283, Node999, 50);
+    Reply(280);
+    giq_option(1, NAME, 281, Node023, NEUTRAL_REACTION);
+    giq_option(1, NAME, 282, Node010, NEUTRAL_REACTION);
+    giq_option(1, NAME, 283, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node029
 begin
-    set_global_var(369, 1);
-    gsay_reply(190, 290);
-    gsay_option(190, 291, Node010, 50);
+    set_global_var(GVAR_DUMAR_STATUS, 1);
+    Reply(290);
+    giq_option(1, NAME, 291, Node010, NEUTRAL_REACTION);
 end
 
 procedure Node030
 begin
-    gsay_reply(190, 300);
-    gsay_option(190, 301, Node031, 50);
+    Reply(300);
+    giq_option(1, NAME, 301, Node031, NEUTRAL_REACTION);
 end
 
 procedure Node031
 begin
-    gsay_reply(190, 310);
-    gsay_option(190, 311, Node033, 50);
-    gsay_option(190, 312, Node037, 50);
-    giq_option(1, 190, 313, Node032, 51);
-    if ((has_skill(dude_obj, 6) >= 75) or (has_skill(dude_obj, 7) >= 75)) then begin
-        giq_option(1, 190, 314, Node050, 49);
+    Reply(310);
+    giq_option(1, NAME, 311, Node033, NEUTRAL_REACTION);
+    giq_option(1, NAME, 312, Node037, NEUTRAL_REACTION);
+    giq_option(1, NAME, 313, Node032, BAD_REACTION);
+    if ((dude_skill(SKILL_FIRST_AID) >= 75) or (dude_skill(SKILL_DOCTOR) >= 75)) then begin
+        giq_option(1, NAME, 314, Node050, GOOD_REACTION);
     end
-    if ((has_skill(dude_obj, 13) >= 75) or (has_skill(dude_obj, 12) >= 75)) then begin
-        giq_option(1, 190, 315, Node040, 49);
+    if ((dude_skill(SKILL_REPAIR) >= 75) or (dude_skill(SKILL_SCIENCE) >= 75)) then begin
+        giq_option(1, NAME, 315, Node040, GOOD_REACTION);
     end
-    gsay_option(190, 316, Node041, 50);
+    giq_option(1, NAME, 316, Node041, NEUTRAL_REACTION);
 end
 
 procedure Node032
 begin
-    gsay_reply(190, 320);
-    gsay_option(190, 321, Node010, 50);
+    Reply(320);
+    giq_option(1, NAME, 321, Node010, NEUTRAL_REACTION);
 end
 
 procedure Node033
 begin
-    gsay_reply(190, 330);
-    if ((get_critter_stat(dude_obj, 0) >= 8) and (get_critter_stat(dude_obj, 5) >= 8)) then begin
-        giq_option(1, 190, 331, Node034, 49);
+    Reply(330);
+    if ((dude_strength >= 8) and (get_critter_stat(dude_obj, STAT_ag) >= 8)) then begin
+        giq_option(1, NAME, 331, Node034, GOOD_REACTION);
     end
     else begin
-        giq_option(1, 190, 332, Node035, 49);
+        giq_option(1, NAME, 332, Node035, GOOD_REACTION);
     end
-    gsay_option(190, 333, Node010, 50);
+    giq_option(1, NAME, 333, Node010, NEUTRAL_REACTION);
 end
 
 procedure Node034
 begin
-    gsay_reply(190, 340);
-    gsay_option(190, 341, Node036, 50);
+    Reply(340);
+    giq_option(1, NAME, 341, Node036, NEUTRAL_REACTION);
 end
 
 procedure Node035
 begin
-    gsay_reply(190, 350);
-    gsay_option(190, 351, Node010, 50);
+    Reply(350);
+    giq_option(1, NAME, 351, Node010, NEUTRAL_REACTION);
 end
 
 procedure Node036
 begin
-    set_global_var(369, 1);
+    set_global_var(GVAR_DUMAR_STATUS, 1);
     set_local_var(9, 1);
-    gsay_reply(190, 360);
-    gsay_option(190, 361, Node010, 50);
+    Reply(360);
+    giq_option(1, NAME, 361, Node010, NEUTRAL_REACTION);
 end
 
 procedure Node037
 begin
-    gsay_reply(190, 370);
-    if ((get_critter_stat(dude_obj, 4) >= 8) and (get_critter_stat(dude_obj, 1) >= 8)) then begin
-        giq_option(1, 190, 371, Node038, 49);
+    Reply(370);
+    if ((dude_iq >= 8) and (dude_perception >= 8)) then begin
+        giq_option(1, NAME, 371, Node038, GOOD_REACTION);
     end
     else begin
-        giq_option(1, 190, 372, Node039, 49);
+        giq_option(1, NAME, 372, Node039, GOOD_REACTION);
     end
-    gsay_option(190, 373, Node010, 50);
+    giq_option(1, NAME, 373, Node010, NEUTRAL_REACTION);
 end
 
 procedure Node038
 begin
-    gsay_reply(190, 380);
-    gsay_option(190, 381, Node036, 50);
+    Reply(380);
+    giq_option(1, NAME, 381, Node036, NEUTRAL_REACTION);
 end
 
 procedure Node039
 begin
-    gsay_reply(190, 390);
-    gsay_option(190, 391, Node010, 50);
+    Reply(390);
+    giq_option(1, NAME, 391, Node010, NEUTRAL_REACTION);
 end
 
 procedure Node040
 begin
-    gsay_reply(190, 400);
-    gsay_option(190, 401, Node036, 50);
+    Reply(400);
+    giq_option(1, NAME, 401, Node036, NEUTRAL_REACTION);
 end
 
 procedure Node041
 begin
-    gsay_reply(190, 410);
-    giq_option(1, 190, 411, Node042, 51);
+    Reply(410);
+    giq_option(1, NAME, 411, Node042, BAD_REACTION);
 end
 
 procedure Node042
 begin
-    gsay_reply(190, 420);
+    Reply(420);
     if (item_caps_total(dude_obj) >= 200) then begin
-        giq_option(1, 190, 421, Node043, 49);
+        giq_option(1, NAME, 421, Node043, GOOD_REACTION);
     end
-    giq_option(1, 190, 422, Node044, 51);
+    giq_option(1, NAME, 422, Node044, BAD_REACTION);
 end
 
 procedure Node043
 begin
     item_caps_adjust(dude_obj, -200);
     item_caps_adjust(self_obj, 200);
-    gsay_reply(190, 430);
-    gsay_option(190, 431, Node046, 50);
+    Reply(430);
+    giq_option(1, NAME, 431, Node046, NEUTRAL_REACTION);
 end
 
 procedure Node044
 begin
-    gsay_reply(190, 440);
-    gsay_option(190, 441, Node045, 50);
+    Reply(440);
+    giq_option(1, NAME, 441, Node045, NEUTRAL_REACTION);
 end
 
 procedure Node045
 begin
-    gsay_reply(190, 450);
-    gsay_option(190, 451, Node999, 50);
+    Reply(450);
+    giq_option(1, NAME, 451, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node046
 begin
-    set_global_var(369, 1);
-    gsay_reply(190, 460);
-    gsay_option(190, 461, Node010, 50);
+    set_global_var(GVAR_DUMAR_STATUS, 1);
+    Reply(460);
+    giq_option(1, NAME, 461, Node010, NEUTRAL_REACTION);
 end
 
 procedure Node047
 begin
-    gsay_reply(190, 470);
-    giq_option(1, 190, 471, Node048, 51);
+    Reply(470);
+    giq_option(1, NAME, 471, Node048, BAD_REACTION);
 end
 
 procedure Node048
 begin
-    gsay_reply(190, 480);
-    gsay_option(190, 481, Node049, 50);
+    Reply(480);
+    giq_option(1, NAME, 481, Node049, NEUTRAL_REACTION);
 end
 
 procedure Node049
 begin
     set_local_var(8, 1);
-    set_global_var(208, 1);
-    set_global_var(0, global_var(0) - 7);
-    gsay_reply(190, 490);
-    gsay_option(190, 491, Node010, 50);
+    set_global_var(GVAR_NCR_SCMERK_PERSONAL_ENEMY, 1);
+    set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) - 7);
+    Reply(490);
+    giq_option(1, NAME, 491, Node010, NEUTRAL_REACTION);
 end
 
 procedure Node050
 begin
-    gsay_reply(190, 500);
-    gsay_option(190, 501, Node036, 50);
+    Reply(500);
+    giq_option(1, NAME, 501, Node036, NEUTRAL_REACTION);
 end
 
 procedure Node051
 begin
-    gsay_reply(190, 510);
-    gsay_option(190, 511, Node010, 50);
-    if (global_var(203) == 5) then begin
-        gsay_option(190, 512, Node052, 50);
+    Reply(510);
+    giq_option(1, NAME, 511, Node010, NEUTRAL_REACTION);
+    if (global_var(GVAR_ENEMY_TORR) == 5) then begin
+        giq_option(1, NAME, 512, Node052, NEUTRAL_REACTION);
     end
 end
 
 procedure Node052
 begin
-    gsay_reply(190, 520);
-    giq_option(1, 190, 521, Node053, 49);
+    Reply(520);
+    giq_option(1, NAME, 521, Node053, GOOD_REACTION);
 end
 
 procedure Node053
 begin
-    if (global_var(151) < 2) then begin
-        set_global_var(151, 2);
-        give_exp_points(floor(518 / 5) + (get_critter_stat(dude_obj, 4) * floor(518 / 12)) + (get_critter_stat(dude_obj, 6) * floor(518 / 30)) + (floor(518 / 75) * (((get_critter_stat(dude_obj, 6) - 4) * 4) - (difficulty_level * 4))));
-        display_msg(message_str(10, random(140, 143)) + (floor(518 / 5) + (get_critter_stat(dude_obj, 4) * floor(518 / 12)) + (get_critter_stat(dude_obj, 6) * floor(518 / 30)) + (floor(518 / 75) * (((get_critter_stat(dude_obj, 6) - 4) * 4) - (difficulty_level * 4)))) + message_str(10, 144));
-        set_global_var(50, global_var(50) + 3);
+    if (global_var(GVAR_SIERRA_BASE_ALERT) < 2) then begin
+        set_global_var(GVAR_SIERRA_BASE_ALERT, 2);
+        give_exp_points(floor(518 / 5) + (dude_iq * floor(518 / 12)) + (dude_luck * floor(518 / 30)) + (floor(518 / 75) * (((dude_luck - 4) * 4) - (difficulty_level * 4))));
+        display_msg(message_str(SCRIPT_STTEXT, random(140, 143)) + (floor(518 / 5) + (dude_iq * floor(518 / 12)) + (dude_luck * floor(518 / 30)) + (floor(518 / 75) * (((dude_luck - 4) * 4) - (difficulty_level * 4)))) + message_str(SCRIPT_STTEXT, 144));
+        set_global_var(GVAR_TOWN_REP_VAULT_CITY, global_var(GVAR_TOWN_REP_VAULT_CITY) + 3);
     end
-    gsay_reply(190, 530);
-    gsay_option(190, 531, Node053a, 50);
-    gsay_option(190, 532, Node053b, 50);
-    if (global_var(221) == 3) then begin
-        gsay_option(190, 533, Node999, 50);
+    Reply(530);
+    giq_option(1, NAME, 531, Node053a, NEUTRAL_REACTION);
+    giq_option(1, NAME, 532, Node053b, NEUTRAL_REACTION);
+    if (global_var(GVAR_HOLODISK_MB_LEVEL_1) == 3) then begin
+        giq_option(1, NAME, 533, Node999, NEUTRAL_REACTION);
     end
 end
 
@@ -818,126 +818,126 @@ end
 
 procedure Node053b
 begin
-    set_global_var(0, global_var(0) + 20);
-    display_msg(message_str(190, 750));
+    set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + 20);
+    display_mstr(750);
 end
 
 procedure Node054
 begin
-    gsay_reply(190, 540);
-    if (global_var(151) == 0) then begin
-        gsay_option(190, 541, Node055, 50);
+    Reply(540);
+    if (global_var(GVAR_SIERRA_BASE_ALERT) == 0) then begin
+        giq_option(1, NAME, 541, Node055, NEUTRAL_REACTION);
     end
-    gsay_option(190, 542, Node010, 50);
+    giq_option(1, NAME, 542, Node010, NEUTRAL_REACTION);
 end
 
 procedure Node055
 begin
-    gsay_reply(190, 550);
-    gsay_option(190, 551, Node056, 50);
+    Reply(550);
+    giq_option(1, NAME, 551, Node056, NEUTRAL_REACTION);
 end
 
 procedure Node056
 begin
-    gsay_reply(190, 560);
-    gsay_option(190, 561, Node010, 50);
-    gsay_option(190, 562, Node057, 50);
+    Reply(560);
+    giq_option(1, NAME, 561, Node010, NEUTRAL_REACTION);
+    giq_option(1, NAME, 562, Node057, NEUTRAL_REACTION);
 end
 
 procedure Node057
 begin
-    set_global_var(151, 1);
-    gsay_reply(190, 570);
-    if (not(metarule(17, 5) == 1) or not(metarule(17, 10) == 1)) then begin
-        gsay_option(190, 571, Node057a, 50);
+    set_global_var(GVAR_SIERRA_BASE_ALERT, 1);
+    Reply(570);
+    if (not(town_known(5) == MARK_STATE_KNOWN) or not(town_known(10) == MARK_STATE_KNOWN)) then begin
+        giq_option(1, NAME, 571, Node057a, NEUTRAL_REACTION);
     end
     else begin
-        gsay_option(190, 572, Node999, 50);
+        giq_option(1, NAME, 572, Node999, NEUTRAL_REACTION);
     end
 end
 
 procedure Node057a
 begin
-    mark_area_known(0, 10, 1);
-    mark_area_known(0, 5, 1);
+    mark_area_known(MARK_TYPE_TOWN, 10, MARK_STATE_KNOWN);
+    mark_area_known(MARK_TYPE_TOWN, 5, MARK_STATE_KNOWN);
 end
 
 procedure Node058
 begin
-    gsay_reply(190, 580);
-    gsay_option(190, 581, Node010, 50);
-    gsay_option(190, 582, Node999, 50);
+    Reply(580);
+    giq_option(1, NAME, 581, Node010, NEUTRAL_REACTION);
+    giq_option(1, NAME, 582, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node059
 begin
-    if (global_var(221) == 3) then begin
-        gsay_reply(190, 590);
+    if (global_var(GVAR_HOLODISK_MB_LEVEL_1) == 3) then begin
+        Reply(590);
     end
-    if (global_var(221) == 2) then begin
-        gsay_reply(190, 591);
+    if (global_var(GVAR_HOLODISK_MB_LEVEL_1) == 2) then begin
+        Reply(591);
     end
-    if (global_var(221) == 1) then begin
-        gsay_reply(190, 592);
+    if (global_var(GVAR_HOLODISK_MB_LEVEL_1) == 1) then begin
+        Reply(592);
     end
-    if (global_var(221) == 3) then begin
-        gsay_option(190, 593, Node060, 50);
+    if (global_var(GVAR_HOLODISK_MB_LEVEL_1) == 3) then begin
+        giq_option(1, NAME, 593, Node060, NEUTRAL_REACTION);
     end
-    if (global_var(221) == 2) then begin
-        gsay_option(190, 594, Node073, 50);
+    if (global_var(GVAR_HOLODISK_MB_LEVEL_1) == 2) then begin
+        giq_option(1, NAME, 594, Node073, NEUTRAL_REACTION);
     end
-    if (global_var(221) == 1) then begin
-        gsay_option(190, 595, Node074, 50);
+    if (global_var(GVAR_HOLODISK_MB_LEVEL_1) == 1) then begin
+        giq_option(1, NAME, 595, Node074, NEUTRAL_REACTION);
     end
-    gsay_option(190, 596, Node999, 50);
+    giq_option(1, NAME, 596, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node060
 begin
-    gsay_reply(190, 600);
-    giq_option(1, 190, 601, Node061, 51);
-    giq_option(1, 190, 602, Node072, 49);
-    gsay_option(190, 603, Node066, 50);
+    Reply(600);
+    giq_option(1, NAME, 601, Node061, BAD_REACTION);
+    giq_option(1, NAME, 602, Node072, GOOD_REACTION);
+    giq_option(1, NAME, 603, Node066, NEUTRAL_REACTION);
 end
 
 procedure Node061
 begin
-    gsay_reply(190, 610);
-    giq_option(1, 190, 611, Node062, 49);
-    gsay_option(190, 612, Node063, 50);
-    giq_option(1, 190, 613, Node063, 51);
-    giq_option(1, 190, 614, Node065, 51);
+    Reply(610);
+    giq_option(1, NAME, 611, Node062, GOOD_REACTION);
+    giq_option(1, NAME, 612, Node063, NEUTRAL_REACTION);
+    giq_option(1, NAME, 613, Node063, BAD_REACTION);
+    giq_option(1, NAME, 614, Node065, BAD_REACTION);
 end
 
 procedure Node062
 begin
-    gsay_reply(190, 620);
-    gsay_option(190, 621, Node999, 50);
-    gsay_option(190, 622, Node999, 50);
+    Reply(620);
+    giq_option(1, NAME, 621, Node999, NEUTRAL_REACTION);
+    giq_option(1, NAME, 622, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node063
 begin
     set_local_var(10, 1);
-    gsay_reply(190, 630);
-    gsay_option(190, 631, Node999, 50);
-    gsay_option(190, 632, Node064, 50);
+    Reply(630);
+    giq_option(1, NAME, 631, Node999, NEUTRAL_REACTION);
+    giq_option(1, NAME, 632, Node064, NEUTRAL_REACTION);
 end
 
 procedure Node064
 begin
     set_local_var(10, 1);
-    gsay_reply(190, 640);
-    gsay_option(190, 641, Node999, 50);
+    Reply(640);
+    giq_option(1, NAME, 641, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node065
 begin
-    set_global_var(279, 1);
-    set_global_var(0, global_var(0) - 5);
-    gsay_reply(190, 650);
-    gsay_option(190, 651, Node065a, 50);
-    gsay_option(190, 652, Node065a, 50);
+    set_global_var(GVAR_CARAVAN_GUARDS_TOTAL, 1);
+    set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) - 5);
+    Reply(650);
+    giq_option(1, NAME, 651, Node065a, NEUTRAL_REACTION);
+    giq_option(1, NAME, 652, Node065a, NEUTRAL_REACTION);
 end
 
 procedure Node065a
@@ -949,61 +949,61 @@ end
 
 procedure Node066
 begin
-    gsay_reply(190, 660);
-    giq_option(1, 190, 661, Node067, 51);
-    gsay_option(190, 662, Node069, 50);
+    Reply(660);
+    giq_option(1, NAME, 661, Node067, BAD_REACTION);
+    giq_option(1, NAME, 662, Node069, NEUTRAL_REACTION);
 end
 
 procedure Node067
 begin
-    gsay_reply(190, 670);
-    gsay_option(190, 671, Node068, 50);
-    gsay_option(190, 672, Node063, 50);
+    Reply(670);
+    giq_option(1, NAME, 671, Node068, NEUTRAL_REACTION);
+    giq_option(1, NAME, 672, Node063, NEUTRAL_REACTION);
 end
 
 procedure Node068
 begin
-    gsay_reply(190, 680);
-    gsay_option(190, 681, Node999, 50);
+    Reply(680);
+    giq_option(1, NAME, 681, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node069
 begin
-    gsay_reply(190, 690);
-    gsay_option(190, 691, Node071, 50);
-    gsay_option(190, 692, Node070, 50);
-    gsay_option(190, 693, Node999, 50);
+    Reply(690);
+    giq_option(1, NAME, 691, Node071, NEUTRAL_REACTION);
+    giq_option(1, NAME, 692, Node070, NEUTRAL_REACTION);
+    giq_option(1, NAME, 693, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node070
 begin
-    gsay_reply(190, 700);
-    gsay_option(190, 701, Node063, 50);
+    Reply(700);
+    giq_option(1, NAME, 701, Node063, NEUTRAL_REACTION);
 end
 
 procedure Node071
 begin
-    gsay_reply(190, 710);
-    gsay_option(190, 711, Node063, 50);
-    gsay_option(190, 712, Node999, 50);
+    Reply(710);
+    giq_option(1, NAME, 711, Node063, NEUTRAL_REACTION);
+    giq_option(1, NAME, 712, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node072
 begin
-    gsay_reply(190, 720);
-    gsay_option(190, 721, Node999, 50);
+    Reply(720);
+    giq_option(1, NAME, 721, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node073
 begin
-    gsay_reply(190, 730);
-    gsay_option(190, 731, Node999, 50);
+    Reply(730);
+    giq_option(1, NAME, 731, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node074
 begin
-    gsay_reply(190, 740);
-    gsay_option(190, 741, Node999, 50);
+    Reply(740);
+    giq_option(1, NAME, 741, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node999
@@ -1022,4 +1022,3 @@ begin
     gdialog_mod_barter(-(dude_objective_reputation + local_var(6)));
     call Node010();
 end
-
