diff --git a/Mods/Olympus2207/Original/NWSAYREN.ssl b/Mods/Olympus2207/ReDefine/NWSAYREN.ssl
index c7122e1..44ee251 100644
--- a/Mods/Olympus2207/Original/NWSAYREN.ssl
+++ b/Mods/Olympus2207/ReDefine/NWSAYREN.ssl
@@ -53,44 +53,44 @@ begin
         set_local_var(4, random(60, 70));
         set_local_var(5, random(40, 50));
         set_local_var(6, random(-15, 0));
-        critter_add_trait(self_obj, 1, 6, 8);
-        critter_add_trait(self_obj, 1, 5, 7);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 8);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_MAN_WEAK);
         if (random(0, 100) <= 33) then begin
-            add_obj_to_inven(self_obj, create_object_sid(151, 1, 2, -1));
-            add_mult_objs_to_inven(self_obj, create_object_sid(9, 1, 2, -1), random(14, 26));
-            wield_obj_critter(self_obj, obj_carrying_pid_obj(self_obj, 151));
+            add_obj_to_inven(self_obj, create_object(151, 1, 2));
+            add_mult_objs_to_inven(self_obj, create_object(9, 1, 2), random(14, 26));
+            wield_obj(obj_carrying_pid_obj(self_obj, 151));
         end
         else begin
-            add_obj_to_inven(self_obj, create_object_sid(152, 1, 2, -1));
-            add_mult_objs_to_inven(self_obj, create_object_sid(2, 1, 2, -1), random(16, 29));
-            wield_obj_critter(self_obj, obj_carrying_pid_obj(self_obj, 152));
+            add_obj_to_inven(self_obj, create_object(152, 1, 2));
+            add_mult_objs_to_inven(self_obj, create_object(2, 1, 2), random(16, 29));
+            wield_obj(obj_carrying_pid_obj(self_obj, 152));
         end
         if (random(0, 100) <= 33) then begin
-            add_obj_to_inven(self_obj, create_object_sid(215, 1, 2, -1));
+            add_obj_to_inven(self_obj, create_object(215, 1, 2));
         end
         else begin
-            add_obj_to_inven(self_obj, create_object_sid(109, 1, 2, -1));
+            add_obj_to_inven(self_obj, create_object(109, 1, 2));
         end
         if (random(0, 100) <= 33) then begin
-            add_mult_objs_to_inven(self_obj, create_object_sid(41, 1, 2, -1), random(10, 40));
+            add_mult_objs_to_inven(self_obj, create_object(41, 1, 2), random(10, 40));
         end
         if (random(0, 100) <= 25) then begin
-            add_mult_objs_to_inven(self_obj, create_object_sid(220, 1, 2, -1), random(1, 3));
+            add_mult_objs_to_inven(self_obj, create_object(220, 1, 2), random(1, 3));
         end
         if (random(0, 100) <= 10) then begin
-            add_obj_to_inven(self_obj, create_object_sid(66, 1, 2, -1));
+            add_obj_to_inven(self_obj, create_object(66, 1, 2));
         end
         if (random(0, 100) <= 9) then begin
-            add_obj_to_inven(self_obj, create_object_sid(431, 1, 2, -1));
+            add_obj_to_inven(self_obj, create_object(431, 1, 2));
         end
         if (random(0, 100) <= 8) then begin
-            add_obj_to_inven(self_obj, create_object_sid(427, 1, 2, -1));
+            add_obj_to_inven(self_obj, create_object(427, 1, 2));
         end
         if (random(0, 100) <= 7) then begin
-            add_obj_to_inven(self_obj, create_object_sid(226, 1, 2, -1));
+            add_obj_to_inven(self_obj, create_object(226, 1, 2));
         end
         if (random(0, 100) <= 6) then begin
-            add_obj_to_inven(self_obj, create_object_sid(229, 1, 2, -1));
+            add_obj_to_inven(self_obj, create_object(229, 1, 2));
         end
     end
 end
@@ -98,17 +98,17 @@ end
 procedure critter_p_proc
 begin
     if (local_var(8) == 1) then begin
-        if ((tile_distance(tile_num(dude_obj), tile_num(self_obj)) > 6) and (tile_distance(tile_num(dude_obj), tile_num(self_obj)) <= 10)) then begin
-            animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(dude_obj), rotation_to_tile(tile_num(dude_obj), tile_num(self_obj)), random(3, 5)), 0);
+        if ((tile_distance(dude_tile, tile_num(self_obj)) > 6) and (tile_distance(dude_tile, tile_num(self_obj)) <= 10)) then begin
+            animate_move_obj_to_tile(self_obj, tile_num_in_direction(dude_tile, rotation_to_tile(dude_tile, tile_num(self_obj)), random(3, 5)), 0);
         end
         else begin
-            if (tile_distance(tile_num(dude_obj), tile_num(self_obj)) > 10) then begin
-                animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(dude_obj), rotation_to_tile(tile_num(dude_obj), tile_num(self_obj)), random(3, 5)), 1);
+            if (tile_distance(dude_tile, tile_num(self_obj)) > 10) then begin
+                animate_move_obj_to_tile(self_obj, tile_num_in_direction(dude_tile, rotation_to_tile(dude_tile, tile_num(self_obj)), random(3, 5)), 1);
             end
         end
     end
     if (obj_can_see_obj(self_obj, dude_obj)) then begin
-        if ((local_var(2) == 1) or (global_var(50) <= -31)) then begin
+        if ((local_var(2) == 1) or (global_var(GVAR_TOWN_REP_VAULT_CITY) <= -31)) then begin
             set_local_var(2, 0);
             attack_setup(self_obj, dude_obj);
         end
@@ -123,10 +123,10 @@ procedure description_p_proc
 begin
     script_overrides;
     if (local_var(7) == 0) then begin
-        display_msg(message_str(195, 2));
+        display_mstr(2);
     end
     else begin
-        display_msg(message_str(195, 4));
+        display_mstr(4);
     end
 end
 
@@ -134,10 +134,10 @@ procedure look_at_p_proc
 begin
     script_overrides;
     if (local_var(7) == 0) then begin
-        display_msg(message_str(195, 1));
+        display_mstr(1);
     end
     else begin
-        display_msg(message_str(195, 3));
+        display_mstr(3);
     end
 end
 
@@ -151,15 +151,15 @@ end
 procedure talk_p_proc
 begin
     if (not(anim_busy(self_obj))) then begin
-        anim(self_obj, 1000, rotation_to_tile(tile_num(self_obj), tile_num(dude_obj)));
+        anim(self_obj, 1000, rotation_to_tile(tile_num(self_obj), dude_tile));
     end
     if (local_var(8) == 0) then begin
         start_gdialog(195, self_obj, -1, -1, -1);
         gsay_start;
-        if (cur_map_index == 50) then begin
+        if (CUR_MAP_TIENTER) then begin
             call Node028();
         end
-        if (cur_map_index == 27) then begin
+        if (CUR_MAP_NWENTER) then begin
             if (local_var(7) == 0) then begin
                 call Node001();
             end
@@ -171,19 +171,19 @@ begin
         end_dialogue;
     end
     else begin
-        float_msg(self_obj, message_str(195, random(900, 903)), 0);
+        float_msg(self_obj, mstr(random(900, 903)), FLOAT_MSG_NORMAL);
     end
 end
 
 procedure destroy_p_proc
 begin
     if (source_obj == dude_obj) then begin
-        set_global_var(0, global_var(0) - 1);
-        set_global_var(50, global_var(50) - 3);
+        set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) - 1);
+        set_global_var(GVAR_TOWN_REP_VAULT_CITY, global_var(GVAR_TOWN_REP_VAULT_CITY) - 3);
     end
-    set_global_var(118, global_var(118) bwor 16);
-    if (global_var(146) == 1) then begin
-        set_global_var(146, 0);
+    set_global_var(GVAR_MODOC_TOOL_FLAG, global_var(GVAR_MODOC_TOOL_FLAG) bwor 16);
+    if (global_var(GVAR_ENEMY_VAULT_VILLAGE) == 1) then begin
+        set_global_var(GVAR_ENEMY_VAULT_VILLAGE, 0);
     end
 end
 
@@ -195,51 +195,51 @@ begin
             set_sfall_global(13, 1);
         end
         else begin
-            if (has_trait(2, dude_obj, 13)) then begin
-                if (global_var(81) < get_pc_stat(1)) then begin
+            if (has_trait(TRAIT_TRAIT, dude_obj, TRAIT_sex_appeal)) then begin
+                if (global_var(GVAR_VAULT_CITIZENSHIP) < get_pc_stat(PCSTAT_level)) then begin
                     if (is_success(roll_vs_skill(dude_obj, 10, 0))) then begin
-                        set_global_var(98, global_var(98) + 3);
+                        set_global_var(GVAR_QUEST_FIND_THOMAS, global_var(GVAR_QUEST_FIND_THOMAS) + 3);
                     end
                     else begin
-                        set_global_var(98, global_var(98) + 1);
+                        set_global_var(GVAR_QUEST_FIND_THOMAS, global_var(GVAR_QUEST_FIND_THOMAS) + 1);
                     end
-                    if (global_var(98) >= (2 * get_pc_stat(1))) then begin
-                        set_global_var(98, 0);
-                        set_global_var(81, global_var(81) + 1);
-                        critter_mod_skill(dude_obj, 10, 1);
-                        display_msg(message_str(10, 410));
+                    if (global_var(GVAR_QUEST_FIND_THOMAS) >= (2 * get_pc_stat(PCSTAT_level))) then begin
+                        set_global_var(GVAR_QUEST_FIND_THOMAS, 0);
+                        set_global_var(GVAR_VAULT_CITIZENSHIP, global_var(GVAR_VAULT_CITIZENSHIP) + 1);
+                        critter_mod_skill(dude_obj, SKILL_STEAL, 1);
+                        display_msg(message_str(SCRIPT_STTEXT, 410));
                     end
                 end
             end
             steal_perk := 0;
             steal_gloves := 0;
-            if (has_trait(0, dude_obj, 37) != 0) then begin
+            if (has_trait(TRAIT_PERK, dude_obj, PERK_pickpocket) != 0) then begin
                 steal_perk := steal_perk + 15;
             end
-            if (has_trait(0, dude_obj, 33) != 0) then begin
+            if (has_trait(TRAIT_PERK, dude_obj, PERK_master_thief) != 0) then begin
                 steal_perk := steal_perk + 30;
             end
-            if ((obj_pid(critter_inven_obj(dude_obj, 1)) == 245) or (obj_pid(critter_inven_obj(dude_obj, 2)) == 245)) then begin
+            if ((obj_pid(critter_inven_obj(dude_obj, INVEN_TYPE_RIGHT_HAND)) == 245) or (obj_pid(critter_inven_obj(dude_obj, INVEN_TYPE_LEFT_HAND)) == 245)) then begin
                 steal_gloves := 10;
             end
-            if (random(0, 100) <= ((has_skill(dude_obj, 10) / 2) + (((3 * get_critter_stat(dude_obj, 5)) + (((get_critter_stat(dude_obj, 6) - 4) * 4) - (difficulty_level * 4)) + (2 * steal_perk) + steal_gloves) / (local_var(4) / 25)))) then begin
-                display_msg(message_str(10, random(145, 148)));
+            if (random(0, 100) <= ((dude_skill(SKILL_STEAL) / 2) + (((3 * get_critter_stat(dude_obj, STAT_ag)) + (((dude_luck - 4) * 4) - (difficulty_level * 4)) + (2 * steal_perk) + steal_gloves) / (local_var(4) / 25)))) then begin
+                display_msg(message_str(SCRIPT_STTEXT, random(145, 148)));
             end
             else begin
                 script_overrides;
-                if (random(0, 100) <= (25 - floor((((get_critter_stat(dude_obj, 6) - 4) * 4) - (difficulty_level * 4)) / 3))) then begin
-                    float_msg(self_obj, message_str(10, random(206, 209)), 2);
+                if (random(0, 100) <= (25 - floor((((dude_luck - 4) * 4) - (difficulty_level * 4)) / 3))) then begin
+                    float_msg(self_obj, message_str(SCRIPT_STTEXT, random(206, 209)), FLOAT_MSG_RED);
                     set_local_var(2, 1);
                 end
                 else begin
-                    display_msg(message_str(10, random(200, 203)));
+                    display_msg(message_str(SCRIPT_STTEXT, random(200, 203)));
                 end
             end
         end
     end
     if (action_being_used == 12) then begin
         script_overrides;
-        display_msg(message_str(10, 204) + ((has_skill(dude_obj, 10) / 2) + (((3 * get_critter_stat(dude_obj, 5)) + (((get_critter_stat(dude_obj, 6) - 4) * 4) - (difficulty_level * 4)) + (2 * steal_perk) + steal_gloves) / (local_var(4) / 25))) + message_str(10, 205));
+        display_msg(message_str(SCRIPT_STTEXT, 204) + ((dude_skill(SKILL_STEAL) / 2) + (((3 * get_critter_stat(dude_obj, STAT_ag)) + (((dude_luck - 4) * 4) - (difficulty_level * 4)) + (2 * steal_perk) + steal_gloves) / (local_var(4) / 25))) + message_str(SCRIPT_STTEXT, 205));
     end
 end
 
@@ -256,13 +256,13 @@ end
 
 procedure map_enter_p_proc
 begin
-    if (not(metarule(22, 0)) == 1) then begin
-        if (cur_map_index == 50) then begin
-            if (global_var(205) == 1) then begin
+    if (not(is_loading_game) == 1) then begin
+        if (CUR_MAP_TIENTER) then begin
+            if (global_var(GVAR_ENEMY_SMILEY) == 1) then begin
                 move_to(self_obj, tile_num(self_obj), 2);
             end
             else begin
-                if ((global_var(146) != 1) or (global_var(205) == 3)) then begin
+                if ((global_var(GVAR_ENEMY_VAULT_VILLAGE) != 1) or (global_var(GVAR_ENEMY_SMILEY) == 3)) then begin
                     move_to(self_obj, tile_num(self_obj), 2);
                 end
                 else begin
@@ -270,9 +270,9 @@ begin
                 end
             end
         end
-        if (cur_map_index == 51) then begin
-            if (not(global_var(285) == 1)) then begin
-                set_global_var(205, 1);
+        if (CUR_MAP_TISHOP) then begin
+            if (not(global_var(GVAR_NEW_RENO_WRIGHT_FLAGS) == 1)) then begin
+                set_global_var(GVAR_ENEMY_SMILEY, 1);
                 set_local_var(8, 0);
                 party_remove(self_obj);
                 game_ui_disable;
@@ -281,8 +281,8 @@ begin
                 add_timer_event(self_obj, 55, 3);
             end
         end
-        if (cur_map_index == 27) then begin
-            if (((global_var(118) bwand 16) != 0) or (global_var(205) == 2)) then begin
+        if (CUR_MAP_NWENTER) then begin
+            if (((global_var(GVAR_MODOC_TOOL_FLAG) bwand 16) != 0) or (global_var(GVAR_ENEMY_SMILEY) == 2)) then begin
                 move_to(self_obj, tile_num(self_obj), 2);
             end
         end
@@ -297,7 +297,7 @@ procedure timed_event_p_proc
 begin
     variable LVar0 := 0;
     if (fixed_param == 1) then begin
-        float_msg(self_obj, message_str(195, 904), 0);
+        float_msg(self_obj, mstr(904), FLOAT_MSG_NORMAL);
     end
     if (fixed_param == 2) then begin
         animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(self_obj), 2, random(25, 25)), 1);
@@ -307,7 +307,7 @@ begin
         move_to(self_obj, tile_num(self_obj), 2);
         if (local_var(9) == 0) then begin
             set_local_var(9, 1);
-            LVar0 := create_object_sid(16777419, 24502, 0, -1);
+            LVar0 := create_object(16777419, 24502, 0);
             attack_setup(LVar0, dude_obj);
         end
     end
@@ -315,35 +315,35 @@ end
 
 procedure Node001
 begin
-    gsay_reply(195, 10);
-    if (get_critter_stat(dude_obj, 4) >= 7) then begin
-        gsay_option(195, 11, Node002, 50);
+    Reply(10);
+    if (dude_iq >= 7) then begin
+        giq_option(1, NAME, 11, Node002, NEUTRAL_REACTION);
     end
     else begin
-        if (get_critter_stat(dude_obj, 4) >= 4) then begin
-            gsay_option(195, 12, Node002, 50);
+        if (dude_iq >= 4) then begin
+            giq_option(1, NAME, 12, Node002, NEUTRAL_REACTION);
         end
         else begin
-            gsay_option(195, 13, Node002, 50);
+            giq_option(1, NAME, 13, Node002, NEUTRAL_REACTION);
         end
     end
 end
 
 procedure Node002
 begin
-    gsay_reply(195, 20);
-    if (get_critter_stat(dude_obj, 4) >= 7) then begin
-        gsay_option(195, 21, Node010a, 50);
+    Reply(20);
+    if (dude_iq >= 7) then begin
+        giq_option(1, NAME, 21, Node010a, NEUTRAL_REACTION);
     end
     else begin
-        if (get_critter_stat(dude_obj, 4) >= 4) then begin
-            gsay_option(195, 22, Node010a, 50);
+        if (dude_iq >= 4) then begin
+            giq_option(1, NAME, 22, Node010a, NEUTRAL_REACTION);
         end
         else begin
-            gsay_option(195, 23, Node010a, 50);
+            giq_option(1, NAME, 23, Node010a, NEUTRAL_REACTION);
         end
     end
-    gsay_option(195, 24, Node999, 50);
+    giq_option(1, NAME, 24, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node010a
@@ -357,214 +357,214 @@ procedure Node010
 begin
     if (already_talk == 0) then begin
         already_talk := 1;
-        gsay_reply(195, 100);
+        Reply(100);
     end
     else begin
-        gsay_reply(195, 101);
+        Reply(101);
     end
-    if (global_var(146) == 0) then begin
-        gsay_option(195, 102, Node011, 50);
-        if (not(get_critter_stat(dude_obj, 4)) <= 3) then begin
-            gsay_option(195, 103, Node018, 50);
+    if (global_var(GVAR_ENEMY_VAULT_VILLAGE) == 0) then begin
+        giq_option(1, NAME, 102, Node011, NEUTRAL_REACTION);
+        if (not(dude_iq) <= 3) then begin
+            giq_option(1, NAME, 103, Node018, NEUTRAL_REACTION);
         end
-        if (not(get_critter_stat(dude_obj, 4)) <= 3) then begin
-            gsay_option(195, 104, Node021, 50);
+        if (not(dude_iq) <= 3) then begin
+            giq_option(1, NAME, 104, Node021, NEUTRAL_REACTION);
         end
     end
-    if (global_var(146) == 1) then begin
-        if (not(get_critter_stat(dude_obj, 4)) <= 3) then begin
-            if (global_var(205) == 1) then begin
-                gsay_option(195, 105, Node023, 50);
+    if (global_var(GVAR_ENEMY_VAULT_VILLAGE) == 1) then begin
+        if (not(dude_iq) <= 3) then begin
+            if (global_var(GVAR_ENEMY_SMILEY) == 1) then begin
+                giq_option(1, NAME, 105, Node023, NEUTRAL_REACTION);
             end
-            if (global_var(205) == 3) then begin
-                gsay_option(195, 106, Node024, 50);
+            if (global_var(GVAR_ENEMY_SMILEY) == 3) then begin
+                giq_option(1, NAME, 106, Node024, NEUTRAL_REACTION);
             end
         end
         else begin
-            if (global_var(205) == 1) then begin
-                gsay_option(195, 107, Node022, 50);
+            if (global_var(GVAR_ENEMY_SMILEY) == 1) then begin
+                giq_option(1, NAME, 107, Node022, NEUTRAL_REACTION);
             end
-            if (global_var(205) == 3) then begin
-                gsay_option(195, 108, Node025, 50);
+            if (global_var(GVAR_ENEMY_SMILEY) == 3) then begin
+                giq_option(1, NAME, 108, Node025, NEUTRAL_REACTION);
             end
         end
     end
-    if ((global_var(213) == 0) and (global_var(202) == 1)) then begin
-        gsay_option(195, 117, Node011a, 50);
+    if ((global_var(GVAR_NCR_FAKE_VAULT13_MAP) == 0) and (global_var(GVAR_TORR_BRAHMIN_KILLED) == 1)) then begin
+        giq_option(1, NAME, 117, Node011a, NEUTRAL_REACTION);
     end
-    gsay_option(195, 109, Node999, 50);
+    giq_option(1, NAME, 109, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node011
 begin
-    gsay_reply(195, 113);
-    if (get_critter_stat(dude_obj, 4) >= 7) then begin
-        gsay_option(195, 110, Node012, 50);
+    Reply(113);
+    if (dude_iq >= 7) then begin
+        giq_option(1, NAME, 110, Node012, NEUTRAL_REACTION);
     end
-    if ((get_critter_stat(dude_obj, 4) >= 4) and (get_critter_stat(dude_obj, 4) <= 6)) then begin
-        gsay_option(195, 111, Node012, 50);
+    if ((dude_iq >= 4) and (dude_iq <= 6)) then begin
+        giq_option(1, NAME, 111, Node012, NEUTRAL_REACTION);
     end
-    if (get_critter_stat(dude_obj, 4) <= 3) then begin
-        gsay_option(195, 112, Node012, 50);
+    if (dude_is_stupid) then begin
+        giq_option(1, NAME, 112, Node012, NEUTRAL_REACTION);
     end
 end
 
 procedure Node011a
 begin
-    set_global_var(213, 1);
-    gsay_reply(195, 114);
-    gsay_option(195, 115, Node010, 50);
-    gsay_option(195, 116, Node999, 50);
+    set_global_var(GVAR_NCR_FAKE_VAULT13_MAP, 1);
+    Reply(114);
+    giq_option(1, NAME, 115, Node010, NEUTRAL_REACTION);
+    giq_option(1, NAME, 116, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node012
 begin
-    gsay_reply(195, 120);
-    if (get_critter_stat(dude_obj, 4) >= 7) then begin
-        gsay_option(195, 121, Node016, 50);
+    Reply(120);
+    if (dude_iq >= 7) then begin
+        giq_option(1, NAME, 121, Node016, NEUTRAL_REACTION);
     end
-    if ((get_critter_stat(dude_obj, 4) >= 4) and (get_critter_stat(dude_obj, 4) <= 6)) then begin
-        gsay_option(195, 122, Node016, 50);
+    if ((dude_iq >= 4) and (dude_iq <= 6)) then begin
+        giq_option(1, NAME, 122, Node016, NEUTRAL_REACTION);
     end
-    if (get_critter_stat(dude_obj, 4) >= 4) then begin
-        gsay_option(195, 123, Node015, 50);
+    if (dude_iq >= 4) then begin
+        giq_option(1, NAME, 123, Node015, NEUTRAL_REACTION);
     end
-    if (get_critter_stat(dude_obj, 4) <= 3) then begin
-        gsay_option(195, 124, Node013, 50);
+    if (dude_is_stupid) then begin
+        giq_option(1, NAME, 124, Node013, NEUTRAL_REACTION);
     end
-    if (get_critter_stat(dude_obj, 4) <= 3) then begin
-        gsay_option(195, 125, Node014, 50);
+    if (dude_is_stupid) then begin
+        giq_option(1, NAME, 125, Node014, NEUTRAL_REACTION);
     end
 end
 
 procedure Node013
 begin
-    gsay_reply(195, 130);
-    gsay_option(195, 131, Node027, 50);
-    gsay_option(195, 132, Node999, 50);
+    Reply(130);
+    giq_option(1, NAME, 131, Node027, NEUTRAL_REACTION);
+    giq_option(1, NAME, 132, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node014
 begin
-    gsay_reply(195, 140);
-    gsay_option(195, 141, Node999, 50);
+    Reply(140);
+    giq_option(1, NAME, 141, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node015
 begin
-    gsay_reply(195, 150);
-    gsay_option(195, 151, Node010, 50);
-    gsay_option(195, 152, Node999, 50);
+    Reply(150);
+    giq_option(1, NAME, 151, Node010, NEUTRAL_REACTION);
+    giq_option(1, NAME, 152, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node016
 begin
-    gsay_reply(195, 160);
-    gsay_option(195, 161, Node017, 50);
-    gsay_option(195, 162, Node010, 50);
-    gsay_option(195, 163, Node999, 50);
+    Reply(160);
+    giq_option(1, NAME, 161, Node017, NEUTRAL_REACTION);
+    giq_option(1, NAME, 162, Node010, NEUTRAL_REACTION);
+    giq_option(1, NAME, 163, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node017
 begin
-    gsay_reply(195, 170);
-    gsay_option(195, 171, Node010, 50);
-    gsay_option(195, 172, Node027, 50);
-    gsay_option(195, 173, Node999, 50);
+    Reply(170);
+    giq_option(1, NAME, 171, Node010, NEUTRAL_REACTION);
+    giq_option(1, NAME, 172, Node027, NEUTRAL_REACTION);
+    giq_option(1, NAME, 173, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node018
 begin
-    gsay_reply(195, 180);
-    gsay_option(195, 181, Node019, 50);
+    Reply(180);
+    giq_option(1, NAME, 181, Node019, NEUTRAL_REACTION);
 end
 
 procedure Node019
 begin
-    gsay_reply(195, 190);
-    gsay_option(195, 191, Node020, 50);
+    Reply(190);
+    giq_option(1, NAME, 191, Node020, NEUTRAL_REACTION);
 end
 
 procedure Node020
 begin
-    gsay_reply(195, 200);
-    gsay_option(195, 201, Node010, 50);
+    Reply(200);
+    giq_option(1, NAME, 201, Node010, NEUTRAL_REACTION);
 end
 
 procedure Node021
 begin
-    gsay_reply(195, 210);
-    gsay_option(195, 211, Node010, 50);
-    gsay_option(195, 212, Node999, 50);
+    Reply(210);
+    giq_option(1, NAME, 211, Node010, NEUTRAL_REACTION);
+    giq_option(1, NAME, 212, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node022
 begin
-    gsay_reply(195, 220);
-    gsay_option(195, 221, Node026, 50);
+    Reply(220);
+    giq_option(1, NAME, 221, Node026, NEUTRAL_REACTION);
     if (obj_carrying_pid_obj(dude_obj, 437)) then begin
-        gsay_option(195, 222, NodeQD, 50);
+        giq_option(1, NAME, 222, NodeQD, NEUTRAL_REACTION);
     end
 end
 
 procedure Node023
 begin
-    gsay_reply(195, 230);
-    gsay_option(195, 231, Node026, 50);
+    Reply(230);
+    giq_option(1, NAME, 231, Node026, NEUTRAL_REACTION);
     if (obj_carrying_pid_obj(dude_obj, 437)) then begin
-        gsay_option(195, 232, NodeQD, 50);
+        giq_option(1, NAME, 232, NodeQD, NEUTRAL_REACTION);
     end
 end
 
 procedure Node024
 begin
-    gsay_reply(195, 240);
-    gsay_option(195, 241, Node026, 50);
+    Reply(240);
+    giq_option(1, NAME, 241, Node026, NEUTRAL_REACTION);
     if (obj_carrying_pid_obj(dude_obj, 437)) then begin
-        gsay_option(195, 242, NodeQD, 50);
+        giq_option(1, NAME, 242, NodeQD, NEUTRAL_REACTION);
     end
 end
 
 procedure Node025
 begin
-    gsay_reply(195, 250);
-    gsay_option(195, 251, Node026, 50);
+    Reply(250);
+    giq_option(1, NAME, 251, Node026, NEUTRAL_REACTION);
     if (obj_carrying_pid_obj(dude_obj, 437)) then begin
-        gsay_option(195, 252, NodeQD, 50);
+        giq_option(1, NAME, 252, NodeQD, NEUTRAL_REACTION);
     end
 end
 
 procedure Node026
 begin
-    gsay_reply(195, 260);
-    gsay_option(195, 261, Node999, 50);
+    Reply(260);
+    giq_option(1, NAME, 261, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node027
 begin
-    mark_area_known(0, 10, 1);
-    set_global_var(146, 1);
-    gsay_reply(195, 270);
-    gsay_option(195, 271, Node999, 50);
+    mark_area_known(MARK_TYPE_TOWN, 10, MARK_STATE_KNOWN);
+    set_global_var(GVAR_ENEMY_VAULT_VILLAGE, 1);
+    Reply(270);
+    giq_option(1, NAME, 271, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node028
 begin
-    gsay_reply(195, 280);
-    if (not(get_critter_stat(dude_obj, 4)) <= 3) then begin
-        if (global_var(285) == 1) then begin
-            gsay_option(195, 281, NodeGo, 50);
+    Reply(280);
+    if (not(dude_iq) <= 3) then begin
+        if (global_var(GVAR_NEW_RENO_WRIGHT_FLAGS) == 1) then begin
+            giq_option(1, NAME, 281, NodeGo, NEUTRAL_REACTION);
         end
         else begin
-            gsay_option(195, 282, NodeGo, 50);
+            giq_option(1, NAME, 282, NodeGo, NEUTRAL_REACTION);
         end
     end
     else begin
-        if (global_var(285) == 1) then begin
-            gsay_option(195, 283, NodeGo, 50);
+        if (global_var(GVAR_NEW_RENO_WRIGHT_FLAGS) == 1) then begin
+            giq_option(1, NAME, 283, NodeGo, NEUTRAL_REACTION);
         end
         else begin
-            gsay_option(195, 284, NodeGo, 50);
+            giq_option(1, NAME, 284, NodeGo, NEUTRAL_REACTION);
         end
     end
 end
@@ -577,11 +577,11 @@ end
 
 procedure NodeQD
 begin
-    set_global_var(146, 2);
-    give_exp_points(floor(103) + (get_critter_stat(dude_obj, 4) * floor(43)) + (get_critter_stat(dude_obj, 6) * floor(17)) + (floor(6) * (((get_critter_stat(dude_obj, 6) - 4) * 4) - (difficulty_level * 4))));
-    display_msg(message_str(10, random(140, 143)) + (floor(103) + (get_critter_stat(dude_obj, 4) * floor(43)) + (get_critter_stat(dude_obj, 6) * floor(17)) + (floor(6) * (((get_critter_stat(dude_obj, 6) - 4) * 4) - (difficulty_level * 4)))) + message_str(10, 144));
+    set_global_var(GVAR_ENEMY_VAULT_VILLAGE, 2);
+    give_exp_points(floor(103) + (dude_iq * floor(43)) + (dude_luck * floor(17)) + (floor(6) * (((dude_luck - 4) * 4) - (difficulty_level * 4))));
+    display_msg(message_str(SCRIPT_STTEXT, random(140, 143)) + (floor(103) + (dude_iq * floor(43)) + (dude_luck * floor(17)) + (floor(6) * (((dude_luck - 4) * 4) - (difficulty_level * 4)))) + message_str(SCRIPT_STTEXT, 144));
     rm_obj_from_inven(dude_obj, obj_carrying_pid_obj(dude_obj, 437));
-    add_obj_to_inven(self_obj, create_object_sid(437, 1, 2, -1));
+    add_obj_to_inven(self_obj, create_object(437, 1, 2));
     call Node999();
 end
 
@@ -595,4 +595,3 @@ begin
     already_talk := 0;
     set_local_var(2, 1);
 end
-
