diff --git a/Mods/Olympus2207/Original/NWXVDRO.ssl b/Mods/Olympus2207/ReDefine/NWXVDRO.ssl
index a59be68..723f975 100644
--- a/Mods/Olympus2207/Original/NWXVDRO.ssl
+++ b/Mods/Olympus2207/ReDefine/NWXVDRO.ssl
@@ -25,25 +25,25 @@ end
 procedure description_p_proc
 begin
     script_overrides;
-    if ((global_var(150) == 1) and (map_var(1) == 0)) then begin
-        if (get_critter_stat(dude_obj, 1) >= 8) then begin
+    if ((global_var(GVAR_SIERRA_BASE_LEVEL_BREACH) == 1) and (map_var(1) == 0)) then begin
+        if (dude_perception >= 8) then begin
             set_map_var(1, 1);
-            set_global_var(207, 1);
-            display_msg(message_str(351, 6));
+            set_global_var(GVAR_NCR_SCMERK_HOSTILE, 1);
+            display_mstr(6);
         end
         else begin
-            display_msg(message_str(351, 5));
+            display_mstr(5);
         end
     end
     else begin
-        display_msg(message_str(351, 3));
+        display_mstr(3);
     end
 end
 
 procedure look_at_p_proc
 begin
     script_overrides;
-    display_msg(message_str(351, 1));
+    display_mstr(1);
 end
 
 procedure damage_p_proc
@@ -52,38 +52,38 @@ end
 
 procedure use_p_proc
 begin
-    display_msg(message_str(351, 1));
+    display_mstr(1);
 end
 
 procedure use_obj_on_p_proc
 begin
     script_overrides;
-    if ((obj_pid(obj_being_used_with) == 435) and (map_var(0) == 0) and (global_var(150) == 1) and (global_var(285) == 2)) then begin
+    if ((obj_pid(obj_being_used_with) == 435) and (map_var(0) == 0) and (global_var(GVAR_SIERRA_BASE_LEVEL_BREACH) == 1) and (global_var(GVAR_NEW_RENO_WRIGHT_FLAGS) == 2)) then begin
         set_map_var(0, game_time_in_seconds);
-        display_msg(message_str(351, 7));
+        display_mstr(7);
         play_sfx("LEVELUP");
-        give_exp_points(floor(50 / 5) + (get_critter_stat(dude_obj, 4) * floor(50 / 12)) + (get_critter_stat(dude_obj, 6) * floor(50 / 30)) + (floor(50 / 75) * (((get_critter_stat(dude_obj, 6) - 4) * 4) - (difficulty_level * 4))));
-        display_msg(message_str(10, random(140, 143)) + (floor(50 / 5) + (get_critter_stat(dude_obj, 4) * floor(50 / 12)) + (get_critter_stat(dude_obj, 6) * floor(50 / 30)) + (floor(50 / 75) * (((get_critter_stat(dude_obj, 6) - 4) * 4) - (difficulty_level * 4)))) + message_str(10, 144));
+        give_exp_points(floor(50 / 5) + (dude_iq * floor(50 / 12)) + (dude_luck * floor(50 / 30)) + (floor(50 / 75) * (((dude_luck - 4) * 4) - (difficulty_level * 4))));
+        display_msg(message_str(SCRIPT_STTEXT, random(140, 143)) + (floor(50 / 5) + (dude_iq * floor(50 / 12)) + (dude_luck * floor(50 / 30)) + (floor(50 / 75) * (((dude_luck - 4) * 4) - (difficulty_level * 4)))) + message_str(SCRIPT_STTEXT, 144));
     end
     else begin
-        display_msg(message_str(10, random(60, 69)));
+        display_msg(message_str(SCRIPT_STTEXT, random(60, 69)));
     end
 end
 
 procedure use_skill_on_p_proc
 begin
     script_overrides;
-    if ((action_being_used == 12) and (global_var(150) == 1) and (map_var(1) == 0)) then begin
-        if ((has_skill(dude_obj, 12) >= 45) and (get_critter_stat(dude_obj, 1) >= 6)) then begin
+    if ((action_being_used == 12) and (global_var(GVAR_SIERRA_BASE_LEVEL_BREACH) == 1) and (map_var(1) == 0)) then begin
+        if ((dude_skill(SKILL_SCIENCE) >= 45) and (dude_perception >= 6)) then begin
             set_map_var(1, 1);
-            display_msg(message_str(351, 6));
+            display_mstr(6);
         end
         else begin
-            display_msg(message_str(351, 5));
+            display_mstr(5);
         end
     end
     else begin
-        display_msg(message_str(10, random(60, 69)));
+        display_msg(message_str(SCRIPT_STTEXT, random(60, 69)));
     end
 end
 
@@ -94,4 +94,3 @@ end
 procedure map_update_p_proc
 begin
 end
-
