diff --git a/Mods/Olympus2207/Original/OBJ_DUDE.ssl b/Mods/Olympus2207/ReDefine/OBJ_DUDE.ssl
index 56de578..f1b974e 100644
--- a/Mods/Olympus2207/Original/OBJ_DUDE.ssl
+++ b/Mods/Olympus2207/ReDefine/OBJ_DUDE.ssl
@@ -54,63 +54,63 @@ end
 
 procedure critter_p_proc
 begin
-    if ((global_var(6) > 0) and ((obj_art_fid(dude_obj) == 16777217) or (obj_art_fid(dude_obj) == 16777218))) then begin
-        animate_move_obj_to_tile(dude_obj, tile_num(dude_obj), 0);
-    end
-    if (has_trait(2, dude_obj, 13) and (global_var(365) != get_pc_stat(1))) then begin
-        set_global_var(365, get_pc_stat(1));
-        set_global_var(71, 0);
-        set_global_var(72, 0);
-        set_global_var(73, 0);
-        set_global_var(74, 0);
-        set_global_var(75, 0);
-        set_global_var(76, 0);
-        set_global_var(77, 0);
-        set_global_var(78, 0);
-        set_global_var(79, 0);
-        set_global_var(80, 0);
-        set_global_var(81, 0);
-        set_global_var(82, 0);
-        set_global_var(83, 0);
-        set_global_var(84, 0);
-        set_global_var(85, 0);
-        set_global_var(86, 0);
-        set_global_var(87, 0);
-        set_global_var(88, 0);
-        critter_mod_skill(dude_obj, 8, floor(5 - (0.20000 * get_pc_stat(1))));
-        display_msg(message_str(10, 408));
-        critter_mod_skill(dude_obj, 12, floor(5 - (0.20000 * get_pc_stat(1))));
-        display_msg(message_str(10, 412));
-        critter_mod_skill(dude_obj, 15, floor(5 - (0.20000 * get_pc_stat(1))));
-        display_msg(message_str(10, 415));
-        critter_mod_skill(dude_obj, 17, floor(5 - (0.20000 * get_pc_stat(1))));
-        display_msg(message_str(10, 417));
+    if ((global_var(GVAR_PLAYER_MARRIED) > 0) and ((obj_art_fid(dude_obj) == 16777217) or (obj_art_fid(dude_obj) == 16777218))) then begin
+        animate_move_obj_to_tile(dude_obj, dude_tile, 0);
+    end
+    if (has_trait(TRAIT_TRAIT, dude_obj, TRAIT_sex_appeal) and (global_var(GVAR_SAN_FRAN_BADGER) != get_pc_stat(PCSTAT_level))) then begin
+        set_global_var(GVAR_SAN_FRAN_BADGER, get_pc_stat(PCSTAT_level));
+        set_global_var(GVAR_TORR_MISSING, 0);
+        set_global_var(GVAR_TORR_SEARCH_SUCCESS, 0);
+        set_global_var(GVAR_TRAPPER_RETURNED, 0);
+        set_global_var(GVAR_DUNTONS_ANGRY, 0);
+        set_global_var(GVAR_RUSTLE_FAIL_VIOLENT, 0);
+        set_global_var(GVAR_RUSTLE_FAIL, 0);
+        set_global_var(GVAR_RUSTLE_SUCCESS, 0);
+        set_global_var(GVAR_TORR_GUARD_SUCCESS, 0);
+        set_global_var(GVAR_VAULT_CITIZEN, 0);
+        set_global_var(GVAR_VAULT_PLOW_PROBLEM, 0);
+        set_global_var(GVAR_VAULT_CITIZENSHIP, 0);
+        set_global_var(GVAR_VAULT_GECKO_PLANT, 0);
+        set_global_var(GVAR_VAULT_PLANT_STATUS, 0);
+        set_global_var(GVAR_VAULT_REDDING_PROBLEM, 0);
+        set_global_var(GVAR_JET_QUEST, 0);
+        set_global_var(GVAR_DAY_PASS_SHOWN, 0);
+        set_global_var(GVAR_VAULT_CITIZEN_TEST, 0);
+        set_global_var(GVAR_VAULT_RAIDERS, 0);
+        critter_mod_skill(dude_obj, SKILL_SNEAK, floor(5 - (0.20000 * get_pc_stat(PCSTAT_level))));
+        display_msg(message_str(SCRIPT_STTEXT, 408));
+        critter_mod_skill(dude_obj, SKILL_SCIENCE, floor(5 - (0.20000 * get_pc_stat(PCSTAT_level))));
+        display_msg(message_str(SCRIPT_STTEXT, 412));
+        critter_mod_skill(dude_obj, SKILL_BARTER, floor(5 - (0.20000 * get_pc_stat(PCSTAT_level))));
+        display_msg(message_str(SCRIPT_STTEXT, 415));
+        critter_mod_skill(dude_obj, SKILL_OUTDOORSMAN, floor(5 - (0.20000 * get_pc_stat(PCSTAT_level))));
+        display_msg(message_str(SCRIPT_STTEXT, 417));
     end
     if (obj_carrying_pid_obj(dude_obj, 487) or obj_carrying_pid_obj(dude_obj, 488) or obj_carrying_pid_obj(dude_obj, 489) and (get_sfall_global_int(40) == 0)) then begin
         set_sfall_global(40, 1);
-        display_msg(message_str(78, 141));
+        display_msg(message_str(SCRIPT_HMSTEXT, 141));
     end
-    if ((get_pc_stat(1) >= 7) and (has_skill(dude_obj, 13) >= 70) and (get_sfall_global_int(36) == 0)) then begin
+    if ((get_pc_stat(PCSTAT_level) >= 7) and (dude_skill(SKILL_REPAIR) >= 70) and (get_sfall_global_int(36) == 0)) then begin
         set_sfall_global(36, 1);
         play_sfx("LEVELUP");
-        display_msg(message_str(78, 137));
+        display_msg(message_str(SCRIPT_HMSTEXT, 137));
     end
-    if ((global_var(10) != 0) and (global_var(13) == 0)) then begin
+    if ((global_var(GVAR_START_ARROYO_TRIAL) != 0) and (global_var(GVAR_DEN_MOM_STATUS) == 0)) then begin
         rm_obj_from_inven(dude_obj, obj_carrying_pid_obj(dude_obj, 43));
-        set_global_var(13, 1);
+        set_global_var(GVAR_DEN_MOM_STATUS, 1);
     end
-    if ((global_var(10) != 0) and (obj_carrying_pid_obj(dude_obj, 44) != 0)) then begin
+    if ((global_var(GVAR_START_ARROYO_TRIAL) != 0) and (obj_carrying_pid_obj(dude_obj, 44) != 0)) then begin
         rm_obj_from_inven(dude_obj, obj_carrying_pid_obj(dude_obj, 44));
-        add_obj_to_inven(dude_obj, create_object_sid(43, 1, 2, -1));
-        set_global_var(10, 0);
-        set_global_var(11, 0);
-        set_global_var(12, 0);
-        set_global_var(13, 0);
+        add_obj_to_inven(dude_obj, create_object(43, 1, 2));
+        set_global_var(GVAR_START_ARROYO_TRIAL, 0);
+        set_global_var(GVAR_REPUTATION_SLAVER, 0);
+        set_global_var(GVAR_REPUTATION_SLAVE_OWNER, 0);
+        set_global_var(GVAR_DEN_MOM_STATUS, 0);
     end
     else begin
         if (obj_carrying_pid_obj(dude_obj, 44) != 0) then begin
             rm_obj_from_inven(dude_obj, obj_carrying_pid_obj(dude_obj, 44));
-            add_obj_to_inven(dude_obj, create_object_sid(43, 1, 2, -1));
+            add_obj_to_inven(dude_obj, create_object(43, 1, 2));
         end
     end
     if (get_sfall_global_int(1) == 1) then begin
@@ -129,11 +129,11 @@ begin
         set_sfall_global(2, 2);
         call HMS_tip();
     end
-    if ((get_sfall_global_int(1) == 6) and (get_sfall_global_int(12) != 2) and (global_var(0) != 0)) then begin
+    if ((get_sfall_global_int(1) == 6) and (get_sfall_global_int(12) != 2) and (global_var(GVAR_PLAYER_REPUTATION) != 0)) then begin
         set_sfall_global(12, 2);
         call Karma_tip();
     end
-    if ((get_sfall_global_int(15) != 2) and (get_critter_stat(dude_obj, 37) > 0) and (cur_map_index > 11)) then begin
+    if ((get_sfall_global_int(15) != 2) and (get_critter_stat(dude_obj, STAT_current_rad) > 0) and (cur_map_index > MAP_TARTAR02)) then begin
         set_sfall_global(15, 2);
         call Radiation_tip();
     end
@@ -151,86 +151,86 @@ begin
         set_sfall_global(14, 2);
         call Kill_tip();
     end
-    if ((get_sfall_global_int(1) == 6) and (get_sfall_global_int(4) == 0) and (global_var(30) == 2) and (global_var(31) == 2) and (global_var(32) == 2) and (global_var(35) == 2)) then begin
+    if ((get_sfall_global_int(1) == 6) and (get_sfall_global_int(4) == 0) and (global_var(GVAR_SLAVE_RUN) == 2) and (global_var(GVAR_SLAVES_COUNT) == 2) and (global_var(GVAR_MAGGIE_STATUS) == 2) and (global_var(GVAR_PIP_BOY_ANNA_DIARY) == 2)) then begin
         set_sfall_global(4, 2);
         call All_Quests_tip();
     end
-    if (obj_carrying_pid_obj(dude_obj, 325) and (global_var(242) == 0)) then begin
-        set_global_var(242, 1);
-    end
-    if (obj_carrying_pid_obj(dude_obj, 93) and (global_var(341) == 0)) then begin
-        set_global_var(341, 1);
-    end
-    if (obj_carrying_pid_obj(dude_obj, 395) and (global_var(354) == 0)) then begin
-        set_global_var(354, 1);
-        display_msg(message_str(10, 1010));
-        set_critter_stat(dude_obj, 0, 1);
-    end
-    if (obj_carrying_pid_obj(dude_obj, 396) and (global_var(355) == 0)) then begin
-        set_global_var(355, 1);
-        display_msg(message_str(10, 1011));
-        set_critter_stat(dude_obj, 1, 1);
-    end
-    if (obj_carrying_pid_obj(dude_obj, 397) and (global_var(356) == 0)) then begin
-        set_global_var(356, 1);
-        display_msg(message_str(10, 1012));
-        set_critter_stat(dude_obj, 2, 1);
-    end
-    if (obj_carrying_pid_obj(dude_obj, 398) and (global_var(357) == 0)) then begin
-        set_global_var(357, 1);
-        display_msg(message_str(10, 1013));
-        set_critter_stat(dude_obj, 3, 1);
-    end
-    if (obj_carrying_pid_obj(dude_obj, 399) and (global_var(358) == 0)) then begin
-        set_global_var(358, 1);
-        display_msg(message_str(10, 1014));
-        set_critter_stat(dude_obj, 4, 1);
-    end
-    if (obj_carrying_pid_obj(dude_obj, 400) and (global_var(359) == 0)) then begin
-        set_global_var(359, 1);
-        display_msg(message_str(10, 1015));
-        set_critter_stat(dude_obj, 5, 1);
-    end
-    if (obj_carrying_pid_obj(dude_obj, 401) and (global_var(360) == 0)) then begin
-        set_global_var(360, 1);
-        display_msg(message_str(10, 1016));
-        set_critter_stat(dude_obj, 6, 1);
-    end
-    if (not(metarule(17, 11) == 1) and obj_carrying_pid_obj(dude_obj, 311) and obj_carrying_pid_obj(dude_obj, 312) and obj_carrying_pid_obj(dude_obj, 313) and obj_carrying_pid_obj(dude_obj, 314)) then begin
-        mark_area_known(0, 11, 1);
+    if (obj_carrying_pid_obj(dude_obj, 325) and (global_var(GVAR_NCR_JACK_STATE) == 0)) then begin
+        set_global_var(GVAR_NCR_JACK_STATE, 1);
+    end
+    if (obj_carrying_pid_obj(dude_obj, 93) and (global_var(GVAR_GOLD_CARAVAN) == 0)) then begin
+        set_global_var(GVAR_GOLD_CARAVAN, 1);
+    end
+    if (obj_carrying_pid_obj(dude_obj, 395) and (global_var(GVAR_NEW_RENO_DELIVERY) == 0)) then begin
+        set_global_var(GVAR_NEW_RENO_DELIVERY, 1);
+        display_msg(message_str(SCRIPT_STTEXT, 1010));
+        set_critter_stat(dude_obj, STAT_st, 1);
+    end
+    if (obj_carrying_pid_obj(dude_obj, 396) and (global_var(GVAR_NEW_RENO_EXTORTION_BROS) == 0)) then begin
+        set_global_var(GVAR_NEW_RENO_EXTORTION_BROS, 1);
+        display_msg(message_str(SCRIPT_STTEXT, 1011));
+        set_critter_stat(dude_obj, STAT_pe, 1);
+    end
+    if (obj_carrying_pid_obj(dude_obj, 397) and (global_var(GVAR_NEW_RENO_ASSASSINATION) == 0)) then begin
+        set_global_var(GVAR_NEW_RENO_ASSASSINATION, 1);
+        display_msg(message_str(SCRIPT_STTEXT, 1012));
+        set_critter_stat(dude_obj, STAT_en, 1);
+    end
+    if (obj_carrying_pid_obj(dude_obj, 398) and (global_var(GVAR_NEW_RENO_LIL_JESUS_REFERS) == 0)) then begin
+        set_global_var(GVAR_NEW_RENO_LIL_JESUS_REFERS, 1);
+        display_msg(message_str(SCRIPT_STTEXT, 1013));
+        set_critter_stat(dude_obj, STAT_ch, 1);
+    end
+    if (obj_carrying_pid_obj(dude_obj, 399) and (global_var(GVAR_SEX_COUNTER) == 0)) then begin
+        set_global_var(GVAR_SEX_COUNTER, 1);
+        display_msg(message_str(SCRIPT_STTEXT, 1014));
+        set_critter_stat(dude_obj, STAT_iq, 1);
+    end
+    if (obj_carrying_pid_obj(dude_obj, 400) and (global_var(GVAR_RND_SALES_NAME) == 0)) then begin
+        set_global_var(GVAR_RND_SALES_NAME, 1);
+        display_msg(message_str(SCRIPT_STTEXT, 1015));
+        set_critter_stat(dude_obj, STAT_ag, 1);
+    end
+    if (obj_carrying_pid_obj(dude_obj, 401) and (global_var(GVAR_RND_SALES_ENCOUNTER) == 0)) then begin
+        set_global_var(GVAR_RND_SALES_ENCOUNTER, 1);
+        display_msg(message_str(SCRIPT_STTEXT, 1016));
+        set_critter_stat(dude_obj, STAT_lu, 1);
+    end
+    if (not(town_known(11) == MARK_STATE_KNOWN) and obj_carrying_pid_obj(dude_obj, 311) and obj_carrying_pid_obj(dude_obj, 312) and obj_carrying_pid_obj(dude_obj, 313) and obj_carrying_pid_obj(dude_obj, 314)) then begin
+        mark_area_known(MARK_TYPE_TOWN, 11, MARK_STATE_KNOWN);
         if (obj_carrying_pid_obj(dude_obj, 393)) then begin
             set_sfall_global(18, 1);
             play_sfx("LEVELUP");
-            display_msg(message_str(61, 112));
+            display_msg(message_str(SCRIPT_ITNTBOOK, 112));
         end
     end
-    if (obj_carrying_pid_obj(dude_obj, 340) and (global_var(288) == 0)) then begin
-        set_global_var(288, 1);
+    if (obj_carrying_pid_obj(dude_obj, 340) and (global_var(GVAR_V13_V15_DALIA_STATE) == 0)) then begin
+        set_global_var(GVAR_V13_V15_DALIA_STATE, 1);
         if (obj_carrying_pid_obj(dude_obj, 393)) then begin
             set_sfall_global(22, 1);
             play_sfx("LEVELUP");
-            display_msg(message_str(61, 112));
+            display_msg(message_str(SCRIPT_ITNTBOOK, 112));
         end
     end
-    if (obj_carrying_pid_obj(dude_obj, 299) and (global_var(350) == 0)) then begin
-        set_global_var(350, 1);
+    if (obj_carrying_pid_obj(dude_obj, 299) and (global_var(GVAR_GRUTHAR_DSTATUS) == 0)) then begin
+        set_global_var(GVAR_GRUTHAR_DSTATUS, 1);
     end
-    if (obj_carrying_pid_obj(dude_obj, 301) and (global_var(349) == 0)) then begin
-        set_global_var(349, 1);
+    if (obj_carrying_pid_obj(dude_obj, 301) and (global_var(GVAR_NEW_RENO_JULES_ELDRIDGE) == 0)) then begin
+        set_global_var(GVAR_NEW_RENO_JULES_ELDRIDGE, 1);
     end
-    if (obj_carrying_pid_obj(dude_obj, 302) and (global_var(351) == 0)) then begin
-        set_global_var(351, 1);
+    if (obj_carrying_pid_obj(dude_obj, 302) and (global_var(GVAR_WHIRLY) == 0)) then begin
+        set_global_var(GVAR_WHIRLY, 1);
     end
-    if ((global_var(282) >= 15) and (global_var(264) == 0)) then begin
-        set_global_var(264, 1);
-        give_exp_points(floor(1000 / 5) + (get_critter_stat(dude_obj, 4) * floor(1000 / 12)) + (get_critter_stat(dude_obj, 6) * floor(1000 / 30)) + (floor(1000 / 75) * (((get_critter_stat(dude_obj, 6) - 4) * 4) - (difficulty_level * 4))));
-        display_msg(message_str(10, random(140, 143)) + (floor(1000 / 5) + (get_critter_stat(dude_obj, 4) * floor(1000 / 12)) + (get_critter_stat(dude_obj, 6) * floor(1000 / 30)) + (floor(1000 / 75) * (((get_critter_stat(dude_obj, 6) - 4) * 4) - (difficulty_level * 4)))) + message_str(10, 144));
-        critter_mod_skill(dude_obj, 16, 25);
+    if ((global_var(GVAR_CARAVAN_MASTERS_TOTAL) >= 15) and (global_var(GVAR_KARMA_HEFNER) == 0)) then begin
+        set_global_var(GVAR_KARMA_HEFNER, 1);
+        give_exp_points(floor(1000 / 5) + (dude_iq * floor(1000 / 12)) + (dude_luck * floor(1000 / 30)) + (floor(1000 / 75) * (((dude_luck - 4) * 4) - (difficulty_level * 4))));
+        display_msg(message_str(SCRIPT_STTEXT, random(140, 143)) + (floor(1000 / 5) + (dude_iq * floor(1000 / 12)) + (dude_luck * floor(1000 / 30)) + (floor(1000 / 75) * (((dude_luck - 4) * 4) - (difficulty_level * 4)))) + message_str(SCRIPT_STTEXT, 144));
+        critter_mod_skill(dude_obj, SKILL_GAMBLING, 25);
     end
-    if ((global_var(221) == 3) and (get_sfall_global_int(20) == 0) and obj_carrying_pid_obj(dude_obj, 393)) then begin
+    if ((global_var(GVAR_HOLODISK_MB_LEVEL_1) == 3) and (get_sfall_global_int(20) == 0) and obj_carrying_pid_obj(dude_obj, 393)) then begin
         set_sfall_global(20, 1);
         play_sfx("LEVELUP");
-        display_msg(message_str(61, 112));
+        display_msg(message_str(SCRIPT_ITNTBOOK, 112));
     end
     if (combat_is_initialized) then begin
         call close_tip();
@@ -239,23 +239,23 @@ end
 
 procedure map_enter_p_proc
 begin
-    if (global_var(7) == 1) then begin
-        set_global_var(7, 0);
+    if (global_var(GVAR_ENEMY_ARROYO) == 1) then begin
+        set_global_var(GVAR_ENEMY_ARROYO, 0);
     end
 end
 
 procedure combat_p_proc
 begin
-    if ((global_var(6) > 0) and ((obj_art_fid(dude_obj) == 16777217) or (obj_art_fid(dude_obj) == 16777218))) then begin
-        animate_move_obj_to_tile(dude_obj, tile_num(dude_obj), 0);
+    if ((global_var(GVAR_PLAYER_MARRIED) > 0) and ((obj_art_fid(dude_obj) == 16777217) or (obj_art_fid(dude_obj) == 16777218))) then begin
+        animate_move_obj_to_tile(dude_obj, dude_tile, 0);
     end
 end
 
 procedure map_update_p_proc
 begin
-    if (metarule(46, 0) > 0) then begin
-        if (global_var(47) < global_var(368)) then begin
-            set_global_var(47, global_var(368));
+    if (cur_town > 0) then begin
+        if (global_var(GVAR_TOWN_REP_ARROYO) < global_var(GVAR_KNOW_DOC_HOLIDAY)) then begin
+            set_global_var(GVAR_TOWN_REP_ARROYO, global_var(GVAR_KNOW_DOC_HOLIDAY));
         end
     end
 end
@@ -268,76 +268,76 @@ procedure use_skill_on_p_proc
 begin
     variable LVar0 := 1;
     if (action_being_used == 6) then begin
-        if (has_trait(2, dude_obj, 13)) then begin
-            if (global_var(77) < get_pc_stat(1)) then begin
-                if (metarule(40, 6)) then begin
+        if (has_trait(TRAIT_TRAIT, dude_obj, TRAIT_sex_appeal)) then begin
+            if (global_var(GVAR_RUSTLE_SUCCESS) < get_pc_stat(PCSTAT_level)) then begin
+                if (is_skill_tagged(SKILL_FIRST_AID)) then begin
                     LVar0 := 2;
                 end
                 if (is_success(roll_vs_skill(dude_obj, 6, 0))) then begin
-                    set_global_var(95, global_var(95) + 3);
+                    set_global_var(GVAR_QUEST_JET_QUEST, global_var(GVAR_QUEST_JET_QUEST) + 3);
                 end
                 else begin
-                    set_global_var(95, global_var(95) + 1);
+                    set_global_var(GVAR_QUEST_JET_QUEST, global_var(GVAR_QUEST_JET_QUEST) + 1);
                 end
-                if (global_var(95) >= (2 * get_pc_stat(1) * LVar0)) then begin
-                    set_global_var(95, 0);
-                    set_global_var(77, global_var(77) + 1);
-                    critter_mod_skill(dude_obj, 6, LVar0);
-                    display_msg(message_str(10, 406));
+                if (global_var(GVAR_QUEST_JET_QUEST) >= (2 * get_pc_stat(PCSTAT_level) * LVar0)) then begin
+                    set_global_var(GVAR_QUEST_JET_QUEST, 0);
+                    set_global_var(GVAR_RUSTLE_SUCCESS, global_var(GVAR_RUSTLE_SUCCESS) + 1);
+                    critter_mod_skill(dude_obj, SKILL_FIRST_AID, LVar0);
+                    display_msg(message_str(SCRIPT_STTEXT, 406));
                 end
             end
         end
     end
     if (action_being_used == 7) then begin
-        if (has_trait(2, dude_obj, 13)) then begin
-            if (global_var(78) < get_pc_stat(1)) then begin
-                if (metarule(40, 7)) then begin
+        if (has_trait(TRAIT_TRAIT, dude_obj, TRAIT_sex_appeal)) then begin
+            if (global_var(GVAR_TORR_GUARD_SUCCESS) < get_pc_stat(PCSTAT_level)) then begin
+                if (is_skill_tagged(SKILL_DOCTOR)) then begin
                     LVar0 := 2;
                 end
                 if (is_success(roll_vs_skill(dude_obj, 7, 0))) then begin
-                    set_global_var(96, global_var(96) + 3);
+                    set_global_var(GVAR_QUEST_RAIDERS, global_var(GVAR_QUEST_RAIDERS) + 3);
                 end
                 else begin
-                    set_global_var(96, global_var(96) + 1);
+                    set_global_var(GVAR_QUEST_RAIDERS, global_var(GVAR_QUEST_RAIDERS) + 1);
                 end
-                if (global_var(96) >= (2 * get_pc_stat(1) * LVar0)) then begin
-                    set_global_var(96, 0);
-                    set_global_var(78, global_var(78) + 1);
-                    critter_mod_skill(dude_obj, 7, LVar0);
-                    display_msg(message_str(10, 407));
+                if (global_var(GVAR_QUEST_RAIDERS) >= (2 * get_pc_stat(PCSTAT_level) * LVar0)) then begin
+                    set_global_var(GVAR_QUEST_RAIDERS, 0);
+                    set_global_var(GVAR_TORR_GUARD_SUCCESS, global_var(GVAR_TORR_GUARD_SUCCESS) + 1);
+                    critter_mod_skill(dude_obj, SKILL_DOCTOR, LVar0);
+                    display_msg(message_str(SCRIPT_STTEXT, 407));
                 end
             end
         end
     end
     if (action_being_used == 9) then begin
-        if (has_trait(2, dude_obj, 13)) then begin
+        if (has_trait(TRAIT_TRAIT, dude_obj, TRAIT_sex_appeal)) then begin
         end
     end
     if (action_being_used == 10) then begin
-        if (has_trait(2, dude_obj, 13)) then begin
+        if (has_trait(TRAIT_TRAIT, dude_obj, TRAIT_sex_appeal)) then begin
         end
     end
     if (action_being_used == 11) then begin
-        if (has_trait(2, dude_obj, 13)) then begin
+        if (has_trait(TRAIT_TRAIT, dude_obj, TRAIT_sex_appeal)) then begin
         end
     end
     if (action_being_used == 13) then begin
-        if (has_trait(2, dude_obj, 13)) then begin
-            if (global_var(84) < get_pc_stat(1)) then begin
-                if (metarule(40, 13)) then begin
+        if (has_trait(TRAIT_TRAIT, dude_obj, TRAIT_sex_appeal)) then begin
+            if (global_var(GVAR_VAULT_REDDING_PROBLEM) < get_pc_stat(PCSTAT_level)) then begin
+                if (is_skill_tagged(SKILL_REPAIR)) then begin
                     LVar0 := 2;
                 end
                 if (is_success(roll_vs_skill(dude_obj, 13, 0))) then begin
-                    set_global_var(100, global_var(100) + 2);
+                    set_global_var(GVAR_QUEST_VIC_DEVICE, global_var(GVAR_QUEST_VIC_DEVICE) + 2);
                 end
                 else begin
-                    set_global_var(100, global_var(100) + 1);
+                    set_global_var(GVAR_QUEST_VIC_DEVICE, global_var(GVAR_QUEST_VIC_DEVICE) + 1);
                 end
-                if (global_var(100) >= (get_pc_stat(1) * LVar0)) then begin
-                    set_global_var(100, 0);
-                    set_global_var(84, global_var(84) + 1);
-                    critter_mod_skill(dude_obj, 13, LVar0);
-                    display_msg(message_str(10, 413));
+                if (global_var(GVAR_QUEST_VIC_DEVICE) >= (get_pc_stat(PCSTAT_level) * LVar0)) then begin
+                    set_global_var(GVAR_QUEST_VIC_DEVICE, 0);
+                    set_global_var(GVAR_VAULT_REDDING_PROBLEM, global_var(GVAR_VAULT_REDDING_PROBLEM) + 1);
+                    critter_mod_skill(dude_obj, SKILL_REPAIR, LVar0);
+                    display_msg(message_str(SCRIPT_STTEXT, 413));
                 end
             end
         end
@@ -351,9 +351,9 @@ begin
     display("PCX/tipwndw.pcx");
     setFont(3);
     setTextColor(0.00000, 0.80000, 0.00000);
-    format(" " + message_str(91, 10), 39, 56, 529, 20, 2);
-    format(" " + message_str(91, 11), 39, 86, 529, 20, 2);
-    format(" " + message_str(91, 12), 19, 116, 529, 20, 2);
+    format(" " + message_str(SCRIPT_TIPTEXT, 10), 39, 56, 529, 20, 2);
+    format(" " + message_str(SCRIPT_TIPTEXT, 11), 39, 86, 529, 20, 2);
+    format(" " + message_str(SCRIPT_TIPTEXT, 12), 19, 116, 529, 20, 2);
     addButton("BTN_close", 445, 243, 135, 34);
     addButtonGfx("BTN_close", "PCX/tipcldwn.pcx", "PCX/tipclup.pcx", "PCX/tipclup.pcx");
     addButtonProc("BTN_close", vide, vide, vide, tips_no);
@@ -375,7 +375,7 @@ begin
     display("PCX/tipwndw.pcx");
     setFont(3);
     setTextColor(0.00000, 0.80000, 0.00000);
-    format(" " + message_str(91, 26), 39, 56, 529, 149, 2);
+    format(" " + message_str(SCRIPT_TIPTEXT, 26), 39, 56, 529, 149, 2);
     addButton("BTN_close", 445, 243, 135, 34);
     addButtonGfx("BTN_close", "PCX/tipcldwn.pcx", "PCX/tipclup.pcx", "PCX/tipclup.pcx");
     addButtonProc("BTN_close", vide, vide, vide, close_tip);
@@ -397,7 +397,7 @@ begin
     display("PCX/tipwndw.pcx");
     setFont(3);
     setTextColor(0.00000, 0.80000, 0.00000);
-    format(" " + message_str(91, 18), 39, 56, 529, 149, 2);
+    format(" " + message_str(SCRIPT_TIPTEXT, 18), 39, 56, 529, 149, 2);
     addButton("BTN_close", 445, 243, 135, 34);
     addButtonGfx("BTN_close", "PCX/tipcldwn.pcx", "PCX/tipclup.pcx", "PCX/tipclup.pcx");
     addButtonProc("BTN_close", vide, vide, vide, close_tip);
@@ -420,7 +420,7 @@ begin
     displayGfx("PCX/tipncstr.pcx", 220, 114, 160, 81);
     setFont(3);
     setTextColor(0.00000, 0.80000, 0.00000);
-    format(" " + message_str(91, 17), 39, 56, 529, 50, 2);
+    format(" " + message_str(SCRIPT_TIPTEXT, 17), 39, 56, 529, 50, 2);
     addButton("BTN_close", 445, 243, 135, 34);
     addButtonGfx("BTN_close", "PCX/tipcldwn.pcx", "PCX/tipclup.pcx", "PCX/tipclup.pcx");
     addButtonProc("BTN_close", vide, vide, vide, close_tip);
@@ -442,7 +442,7 @@ begin
     display("PCX/tipwndw.pcx");
     setFont(3);
     setTextColor(0.00000, 0.80000, 0.00000);
-    format(" " + message_str(91, 16), 39, 56, 529, 149, 2);
+    format(" " + message_str(SCRIPT_TIPTEXT, 16), 39, 56, 529, 149, 2);
     addButton("BTN_close", 445, 243, 135, 34);
     addButtonGfx("BTN_close", "PCX/tipcldwn.pcx", "PCX/tipclup.pcx", "PCX/tipclup.pcx");
     addButtonProc("BTN_close", vide, vide, vide, close_tip);
@@ -464,7 +464,7 @@ begin
     display("PCX/tipwndw.pcx");
     setFont(3);
     setTextColor(0.00000, 0.80000, 0.00000);
-    format(" " + message_str(91, 19), 39, 56, 529, 50, 2);
+    format(" " + message_str(SCRIPT_TIPTEXT, 19), 39, 56, 529, 50, 2);
     addButton("BTN_close", 445, 243, 135, 34);
     addButtonGfx("BTN_close", "PCX/tipcldwn.pcx", "PCX/tipclup.pcx", "PCX/tipclup.pcx");
     addButtonProc("BTN_close", vide, vide, vide, close_tip);
@@ -486,7 +486,7 @@ begin
     display("PCX/tipwndw.pcx");
     setFont(3);
     setTextColor(0.00000, 0.80000, 0.00000);
-    format(" " + message_str(91, 20), 39, 56, 529, 50, 2);
+    format(" " + message_str(SCRIPT_TIPTEXT, 20), 39, 56, 529, 50, 2);
     addButton("BTN_close", 445, 243, 135, 34);
     addButtonGfx("BTN_close", "PCX/tipcldwn.pcx", "PCX/tipclup.pcx", "PCX/tipclup.pcx");
     addButtonProc("BTN_close", vide, vide, vide, close_tip);
@@ -508,7 +508,7 @@ begin
     display("PCX/tipwndw.pcx");
     setFont(3);
     setTextColor(0.00000, 0.80000, 0.00000);
-    format(" " + message_str(91, 21), 39, 56, 529, 50, 2);
+    format(" " + message_str(SCRIPT_TIPTEXT, 21), 39, 56, 529, 50, 2);
     addButton("BTN_close", 445, 243, 135, 34);
     addButtonGfx("BTN_close", "PCX/tipcldwn.pcx", "PCX/tipclup.pcx", "PCX/tipclup.pcx");
     addButtonProc("BTN_close", vide, vide, vide, close_tip);
@@ -530,7 +530,7 @@ begin
     display("PCX/tipwndw.pcx");
     setFont(3);
     setTextColor(0.00000, 0.80000, 0.00000);
-    format(" " + message_str(91, 24), 39, 56, 529, 100, 2);
+    format(" " + message_str(SCRIPT_TIPTEXT, 24), 39, 56, 529, 100, 2);
     addButton("BTN_close", 445, 243, 135, 34);
     addButtonGfx("BTN_close", "PCX/tipcldwn.pcx", "PCX/tipclup.pcx", "PCX/tipclup.pcx");
     addButtonProc("BTN_close", vide, vide, vide, close_tip);
@@ -552,7 +552,7 @@ begin
     display("PCX/tipwndw.pcx");
     setFont(3);
     setTextColor(0.00000, 0.80000, 0.00000);
-    format(" " + message_str(91, 27), 39, 56, 529, 50, 2);
+    format(" " + message_str(SCRIPT_TIPTEXT, 27), 39, 56, 529, 50, 2);
     addButton("BTN_close", 445, 243, 135, 34);
     addButtonGfx("BTN_close", "PCX/tipcldwn.pcx", "PCX/tipclup.pcx", "PCX/tipclup.pcx");
     addButtonProc("BTN_close", vide, vide, vide, close_tip);
@@ -599,15 +599,15 @@ begin
     game_ui_enable;
     set_sfall_global(7, 3);
     set_sfall_global(4, 2);
-    set_global_var(27, 2);
-    set_global_var(28, 2);
-    set_global_var(29, 2);
-    set_global_var(30, 2);
-    set_global_var(31, 2);
-    set_global_var(32, 2);
-    set_global_var(33, 2);
-    set_global_var(34, 2);
-    set_global_var(35, 2);
+    set_global_var(GVAR_LOAD_MAP_INDEX, 2);
+    set_global_var(GVAR_RUNNING_BURNING_GUY, 2);
+    set_global_var(GVAR_VIC_DEVICE, 2);
+    set_global_var(GVAR_SLAVE_RUN, 2);
+    set_global_var(GVAR_SLAVES_COUNT, 2);
+    set_global_var(GVAR_MAGGIE_STATUS, 2);
+    set_global_var(GVAR_SLAVES_LOST, 2);
+    set_global_var(GVAR_SLAVERS_LOST, 2);
+    set_global_var(GVAR_PIP_BOY_ANNA_DIARY, 2);
     fadeOut(1500);
     load_map("tartar13", 1);
 end
@@ -619,12 +619,11 @@ end
 procedure description_p_proc
 begin
     script_overrides;
-    if (obj_name(dude_obj) == "deus") then begin
-        float_msg(self_obj, "How to enable GOD MODE?..", 0);
+    if (dude_name == "deus") then begin
+        float_msg(self_obj, "How to enable GOD MODE?..", FLOAT_MSG_NORMAL);
     end
 end
 
 procedure use_obj_on_p_proc
 begin
 end
-
