diff --git a/Mods/Olympus2207/Original/OLBOB.ssl b/Mods/Olympus2207/ReDefine/OLBOB.ssl
index e9ccdc3..c194a3d 100644
--- a/Mods/Olympus2207/Original/OLBOB.ssl
+++ b/Mods/Olympus2207/ReDefine/OLBOB.ssl
@@ -28,12 +28,12 @@ begin
     if (local_var(1) == 0) then begin
         set_local_var(3, tile_num(self_obj));
         set_local_var(1, 1);
-        set_local_var(7, has_trait(1, self_obj, 10));
+        set_local_var(7, obj_get_rot(self_obj));
         set_local_var(4, random(70, 80));
         set_local_var(5, random(0, 100));
         set_local_var(6, random(-5, 5));
-        critter_add_trait(self_obj, 1, 6, 39);
-        critter_add_trait(self_obj, 1, 5, 71);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 39);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_TURRET);
     end
 end
 
@@ -104,7 +104,6 @@ end
 procedure talk_p_proc
 begin
     if (not(combat_is_initialized)) then begin
-        float_msg(self_obj, message_str(449, random(321, 325)), 0);
+        float_msg(self_obj, message_str(SCRIPT_OLCRANK, random(321, 325)), FLOAT_MSG_NORMAL);
     end
 end
-
