diff --git a/Mods/Olympus2207/Original/OLKELLY.ssl b/Mods/Olympus2207/ReDefine/OLKELLY.ssl
index 178ed96..a6f69db 100644
--- a/Mods/Olympus2207/Original/OLKELLY.ssl
+++ b/Mods/Olympus2207/ReDefine/OLKELLY.ssl
@@ -80,8 +80,8 @@ begin
         set_local_var(4, random(30, 40));
         set_local_var(5, random(50, 60));
         set_local_var(6, random(-30, 20));
-        critter_add_trait(self_obj, 1, 6, 38);
-        critter_add_trait(self_obj, 1, 5, 33);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 38);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_BLONDIE_STUPID);
     end
 end
 
@@ -93,10 +93,10 @@ begin
     end
     if (float_is_called == 0) then begin
         float_is_called := 1;
-        add_timer_event(self_obj, random(1000, 1000), 2);
+        add_timer_event(self_obj, 1000, 2);
     end
     if (obj_can_see_obj(self_obj, dude_obj)) then begin
-        if ((local_var(2) == 1) or (global_var(59) <= -31)) then begin
+        if ((local_var(2) == 1) or (global_var(GVAR_TOWN_REP_VAULT_13) <= -31)) then begin
             set_local_var(2, 0);
             attack_setup(self_obj, dude_obj);
         end
@@ -111,10 +111,10 @@ procedure description_p_proc
 begin
     script_overrides;
     if (local_var(7) == 0) then begin
-        display_msg(message_str(519, 2));
+        display_mstr(2);
     end
     else begin
-        display_msg(message_str(519, 4));
+        display_mstr(4);
     end
 end
 
@@ -122,10 +122,10 @@ procedure look_at_p_proc
 begin
     script_overrides;
     if (local_var(7) == 0) then begin
-        display_msg(message_str(519, 1));
+        display_mstr(1);
     end
     else begin
-        display_msg(message_str(519, 3));
+        display_mstr(3);
     end
 end
 
@@ -143,20 +143,20 @@ end
 procedure talk_p_proc
 begin
     if (not(anim_busy(self_obj))) then begin
-        anim(self_obj, 1000, rotation_to_tile(tile_num(self_obj), tile_num(dude_obj)));
+        anim(self_obj, 1000, rotation_to_tile(tile_num(self_obj), dude_tile));
     end
-    if (global_var(432) == 2) then begin
-        float_msg(self_obj, message_str(519, random(42, 43)), 0);
-        animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(self_obj), rotation_to_tile(tile_num(dude_obj), tile_num(self_obj)), random(3, 5)), 1);
+    if (global_var(GVAR_ENCLAVE_GRANITE_JOINED) == 2) then begin
+        float_msg(self_obj, mstr(random(42, 43)), FLOAT_MSG_NORMAL);
+        animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(self_obj), rotation_to_tile(dude_tile, tile_num(self_obj)), random(3, 5)), 1);
     end
     else begin
-        if ((global_var(432) == 1) and (global_var(422) == 1)) then begin
-            float_msg(self_obj, message_str(519, random(40, 41)), 0);
+        if ((global_var(GVAR_ENCLAVE_GRANITE_JOINED) == 1) and (global_var(GVAR_ENDGAME_MOVIE_VAULT_13) == 1)) then begin
+            float_msg(self_obj, mstr(random(40, 41)), FLOAT_MSG_NORMAL);
         end
         else begin
             start_gdialog(519, self_obj, -1, -1, -1);
             gsay_start;
-            if (global_var(422) == 2) then begin
+            if (global_var(GVAR_ENDGAME_MOVIE_VAULT_13) == 2) then begin
                 call Node100();
             end
             else begin
@@ -176,12 +176,12 @@ end
 procedure destroy_p_proc
 begin
     if (source_obj == dude_obj) then begin
-        set_global_var(0, global_var(0) - 1);
+        set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) - 1);
         if ((59 == 50) or (59 == 59)) then begin
-            set_global_var(59, global_var(59) - 3);
+            set_global_var(GVAR_TOWN_REP_VAULT_13, global_var(GVAR_TOWN_REP_VAULT_13) - 3);
         end
         else begin
-            set_global_var(59, global_var(59) - 6);
+            set_global_var(GVAR_TOWN_REP_VAULT_13, global_var(GVAR_TOWN_REP_VAULT_13) - 6);
         end
     end
 end
@@ -194,51 +194,51 @@ begin
             set_sfall_global(13, 1);
         end
         else begin
-            if (has_trait(2, dude_obj, 13)) then begin
-                if (global_var(81) < get_pc_stat(1)) then begin
+            if (has_trait(TRAIT_TRAIT, dude_obj, TRAIT_sex_appeal)) then begin
+                if (global_var(GVAR_VAULT_CITIZENSHIP) < get_pc_stat(PCSTAT_level)) then begin
                     if (is_success(roll_vs_skill(dude_obj, 10, 0))) then begin
-                        set_global_var(98, global_var(98) + 3);
+                        set_global_var(GVAR_QUEST_FIND_THOMAS, global_var(GVAR_QUEST_FIND_THOMAS) + 3);
                     end
                     else begin
-                        set_global_var(98, global_var(98) + 1);
+                        set_global_var(GVAR_QUEST_FIND_THOMAS, global_var(GVAR_QUEST_FIND_THOMAS) + 1);
                     end
-                    if (global_var(98) >= (2 * get_pc_stat(1))) then begin
-                        set_global_var(98, 0);
-                        set_global_var(81, global_var(81) + 1);
-                        critter_mod_skill(dude_obj, 10, 1);
-                        display_msg(message_str(10, 410));
+                    if (global_var(GVAR_QUEST_FIND_THOMAS) >= (2 * get_pc_stat(PCSTAT_level))) then begin
+                        set_global_var(GVAR_QUEST_FIND_THOMAS, 0);
+                        set_global_var(GVAR_VAULT_CITIZENSHIP, global_var(GVAR_VAULT_CITIZENSHIP) + 1);
+                        critter_mod_skill(dude_obj, SKILL_STEAL, 1);
+                        display_msg(message_str(SCRIPT_STTEXT, 410));
                     end
                 end
             end
             steal_perk := 0;
             steal_gloves := 0;
-            if (has_trait(0, dude_obj, 37) != 0) then begin
+            if (has_trait(TRAIT_PERK, dude_obj, PERK_pickpocket) != 0) then begin
                 steal_perk := steal_perk + 15;
             end
-            if (has_trait(0, dude_obj, 33) != 0) then begin
+            if (has_trait(TRAIT_PERK, dude_obj, PERK_master_thief) != 0) then begin
                 steal_perk := steal_perk + 30;
             end
-            if ((obj_pid(critter_inven_obj(dude_obj, 1)) == 245) or (obj_pid(critter_inven_obj(dude_obj, 2)) == 245)) then begin
+            if ((obj_pid(critter_inven_obj(dude_obj, INVEN_TYPE_RIGHT_HAND)) == 245) or (obj_pid(critter_inven_obj(dude_obj, INVEN_TYPE_LEFT_HAND)) == 245)) then begin
                 steal_gloves := 10;
             end
-            if (random(0, 100) <= ((has_skill(dude_obj, 10) / 2) + (((3 * get_critter_stat(dude_obj, 5)) + (((get_critter_stat(dude_obj, 6) - 4) * 4) - (difficulty_level * 4)) + (2 * steal_perk) + steal_gloves) / (local_var(4) / 25)))) then begin
-                display_msg(message_str(10, random(145, 148)));
+            if (random(0, 100) <= ((dude_skill(SKILL_STEAL) / 2) + (((3 * get_critter_stat(dude_obj, STAT_ag)) + (((dude_luck - 4) * 4) - (difficulty_level * 4)) + (2 * steal_perk) + steal_gloves) / (local_var(4) / 25)))) then begin
+                display_msg(message_str(SCRIPT_STTEXT, random(145, 148)));
             end
             else begin
                 script_overrides;
-                if (random(0, 100) <= (25 - floor((((get_critter_stat(dude_obj, 6) - 4) * 4) - (difficulty_level * 4)) / 3))) then begin
-                    float_msg(self_obj, message_str(10, random(206, 209)), 2);
+                if (random(0, 100) <= (25 - floor((((dude_luck - 4) * 4) - (difficulty_level * 4)) / 3))) then begin
+                    float_msg(self_obj, message_str(SCRIPT_STTEXT, random(206, 209)), FLOAT_MSG_RED);
                     set_local_var(2, 1);
                 end
                 else begin
-                    display_msg(message_str(10, random(200, 203)));
+                    display_msg(message_str(SCRIPT_STTEXT, random(200, 203)));
                 end
             end
         end
     end
     if (action_being_used == 12) then begin
         script_overrides;
-        display_msg(message_str(10, 204) + ((has_skill(dude_obj, 10) / 2) + (((3 * get_critter_stat(dude_obj, 5)) + (((get_critter_stat(dude_obj, 6) - 4) * 4) - (difficulty_level * 4)) + (2 * steal_perk) + steal_gloves) / (local_var(4) / 25))) + message_str(10, 205));
+        display_msg(message_str(SCRIPT_STTEXT, 204) + ((dude_skill(SKILL_STEAL) / 2) + (((3 * get_critter_stat(dude_obj, STAT_ag)) + (((dude_luck - 4) * 4) - (difficulty_level * 4)) + (2 * steal_perk) + steal_gloves) / (local_var(4) / 25))) + message_str(SCRIPT_STTEXT, 205));
     end
 end
 
@@ -260,17 +260,17 @@ end
 procedure timed_event_p_proc
 begin
     if ((fixed_param == 1) and not(combat_is_initialized)) then begin
-        if ((global_var(432) == 0) and (map_var(0) != 0)) then begin
+        if ((global_var(GVAR_ENCLAVE_GRANITE_JOINED) == 0) and (map_var(0) != 0)) then begin
             if ((tile_distance(tile_num(self_obj), tile_num(map_var(0))) > 7) and (anim_busy(self_obj) == 0)) then begin
                 animate_move_obj_to_tile(self_obj, tile_num(map_var(0)), 1);
-                float_msg(self_obj, message_str(519, random(7, 8)), 0);
+                float_msg(self_obj, mstr(random(7, 8)), FLOAT_MSG_NORMAL);
             end
             if ((tile_distance(tile_num(self_obj), tile_num(map_var(0))) > 3) and (anim_busy(self_obj) == 0)) then begin
                 animate_move_obj_to_tile(self_obj, tile_num(map_var(0)), 0);
             end
             if ((tile_distance(tile_num(self_obj), tile_num(map_var(0))) < 2) and (local_var(9) == 0)) then begin
                 set_local_var(9, 1);
-                float_msg(map_var(0), message_str(519, 9), 0);
+                float_msg(map_var(0), mstr(9), FLOAT_MSG_NORMAL);
             end
             if ((tile_distance(tile_num(self_obj), tile_num(map_var(0))) > 2) and (local_var(9) == 1)) then begin
                 set_local_var(9, 0);
@@ -287,10 +287,10 @@ procedure Node001
 begin
     set_local_var(7, 1);
     already_talk := 1;
-    gsay_reply(519, 10);
-    gsay_option(519, 11, Node002, 50);
-    gsay_option(519, 12, Node004, 50);
-    gsay_option(519, 13, Node004, 50);
+    Reply(10);
+    giq_option(1, NAME, 11, Node002, NEUTRAL_REACTION);
+    giq_option(1, NAME, 12, Node004, NEUTRAL_REACTION);
+    giq_option(1, NAME, 13, Node004, NEUTRAL_REACTION);
 end
 
 procedure Node002
@@ -306,18 +306,18 @@ begin
             move_to(map_var(0), 29310, elevation(self_obj));
         end
     end
-    gsay_reply(519, 20);
-    gsay_option(519, 21, Node003, 50);
-    gsay_option(519, 22, Node010, 50);
-    if (global_var(422) == 1) then begin
-        gsay_option(519, 23, Node011, 50);
+    Reply(20);
+    giq_option(1, NAME, 21, Node003, NEUTRAL_REACTION);
+    giq_option(1, NAME, 22, Node010, NEUTRAL_REACTION);
+    if (global_var(GVAR_ENDGAME_MOVIE_VAULT_13) == 1) then begin
+        giq_option(1, NAME, 23, Node011, NEUTRAL_REACTION);
     end
 end
 
 procedure Node003
 begin
-    gsay_reply(519, 30);
-    gsay_option(519, 31, Node999, 50);
+    Reply(30);
+    giq_option(1, NAME, 31, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node004
@@ -333,11 +333,11 @@ begin
             move_to(map_var(0), 29310, elevation(self_obj));
         end
     end
-    gsay_reply(519, 20);
-    gsay_option(519, 21, Node003, 50);
-    gsay_option(519, 22, Node010, 50);
-    if (global_var(422) == 1) then begin
-        gsay_option(519, 23, Node011, 50);
+    Reply(20);
+    giq_option(1, NAME, 21, Node003, NEUTRAL_REACTION);
+    giq_option(1, NAME, 22, Node010, NEUTRAL_REACTION);
+    if (global_var(GVAR_ENDGAME_MOVIE_VAULT_13) == 1) then begin
+        giq_option(1, NAME, 23, Node011, NEUTRAL_REACTION);
     end
 end
 
@@ -361,281 +361,281 @@ begin
     end
     if (already_talk == 0) then begin
         already_talk := 1;
-        gsay_reply(519, 100);
+        Reply(100);
     end
     else begin
-        gsay_reply(519, 101);
+        Reply(101);
     end
-    if (global_var(422) == 1) then begin
-        gsay_option(519, 102, Node011, 50);
+    if (global_var(GVAR_ENDGAME_MOVIE_VAULT_13) == 1) then begin
+        giq_option(1, NAME, 102, Node011, NEUTRAL_REACTION);
     end
-    gsay_option(519, 103, Node044, 50);
-    gsay_option(519, 104, Node045, 50);
-    gsay_option(519, 105, Node045, 50);
-    gsay_option(519, 106, Node045, 50);
+    giq_option(1, NAME, 103, Node044, NEUTRAL_REACTION);
+    giq_option(1, NAME, 104, Node045, NEUTRAL_REACTION);
+    giq_option(1, NAME, 105, Node045, NEUTRAL_REACTION);
+    giq_option(1, NAME, 106, Node045, NEUTRAL_REACTION);
     if (already_talk == 0) then begin
-        gsay_option(519, 107, Node999, 50);
+        giq_option(1, NAME, 107, Node999, NEUTRAL_REACTION);
     end
-    gsay_option(519, 108, Node999, 50);
+    giq_option(1, NAME, 108, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node011
 begin
-    gsay_reply(519, 111);
-    if ((global_var(285) == 2) and (has_skill(dude_obj, 14) >= 80)) then begin
-        gsay_option(519, 112, Node042, 50);
+    Reply(111);
+    if ((global_var(GVAR_NEW_RENO_WRIGHT_FLAGS) == 2) and (dude_skill(SKILL_SPEECH) >= 80)) then begin
+        giq_option(1, NAME, 112, Node042, NEUTRAL_REACTION);
     end
-    if ((has_skill(dude_obj, 14) >= 60) and (local_var(10) == 0)) then begin
-        gsay_option(519, 113, Node021, 50);
+    if ((dude_skill(SKILL_SPEECH) >= 60) and (local_var(10) == 0)) then begin
+        giq_option(1, NAME, 113, Node021, NEUTRAL_REACTION);
     end
-    if (obj_carrying_pid_obj(dude_obj, 389) and (has_skill(dude_obj, 14) >= 70)) then begin
-        gsay_option(519, 114, Node016, 50);
+    if (obj_carrying_pid_obj(dude_obj, 389) and (dude_skill(SKILL_SPEECH) >= 70)) then begin
+        giq_option(1, NAME, 114, Node016, NEUTRAL_REACTION);
     end
-    if (global_var(431) == 1) then begin
-        gsay_option(519, 115, Node013, 50);
+    if (global_var(GVAR_ENCLAVE_POWER_PLANT) == 1) then begin
+        giq_option(1, NAME, 115, Node013, NEUTRAL_REACTION);
     end
-    gsay_option(519, 116, Node012, 50);
+    giq_option(1, NAME, 116, Node012, NEUTRAL_REACTION);
 end
 
 procedure Node012
 begin
-    gsay_reply(519, 120);
-    gsay_option(519, 121, Node999, 50);
+    Reply(120);
+    giq_option(1, NAME, 121, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node013
 begin
-    gsay_reply(519, 131);
-    gsay_option(519, 132, Node014, 50);
+    Reply(131);
+    giq_option(1, NAME, 132, Node014, NEUTRAL_REACTION);
 end
 
 procedure Node014
 begin
-    gsay_reply(519, 140);
-    gsay_option(519, 141, Node015, 50);
+    Reply(140);
+    giq_option(1, NAME, 141, Node015, NEUTRAL_REACTION);
 end
 
 procedure Node015
 begin
-    set_global_var(432, 1);
-    gsay_reply(519, 150);
-    gsay_option(519, 151, Node999, 50);
+    set_global_var(GVAR_ENCLAVE_GRANITE_JOINED, 1);
+    Reply(150);
+    giq_option(1, NAME, 151, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node016
 begin
     rm_obj_from_inven(dude_obj, obj_carrying_pid_obj(dude_obj, 389));
-    gsay_reply(519, 160);
-    gsay_option(519, 161, Node017, 50);
+    Reply(160);
+    giq_option(1, NAME, 161, Node017, NEUTRAL_REACTION);
 end
 
 procedure Node017
 begin
     fadeOut(500);
     fadeIn(500);
-    gsay_reply(519, 170);
-    gsay_option(519, 171, Node018, 50);
+    Reply(170);
+    giq_option(1, NAME, 171, Node018, NEUTRAL_REACTION);
 end
 
 procedure Node018
 begin
-    gsay_reply(519, 180);
-    gsay_option(519, 181, Node019, 50);
+    Reply(180);
+    giq_option(1, NAME, 181, Node019, NEUTRAL_REACTION);
 end
 
 procedure Node019
 begin
-    gsay_reply(519, 190);
-    gsay_option(519, 191, Node020, 50);
+    Reply(190);
+    giq_option(1, NAME, 191, Node020, NEUTRAL_REACTION);
 end
 
 procedure Node020
 begin
-    gsay_reply(519, 200);
-    gsay_option(519, 201, Node015, 50);
+    Reply(200);
+    giq_option(1, NAME, 201, Node015, NEUTRAL_REACTION);
 end
 
 procedure Node021
 begin
-    gsay_reply(519, 210);
-    gsay_option(519, 211, Node022, 50);
-    gsay_option(519, 212, Node023, 50);
-    gsay_option(519, 213, Node024, 50);
+    Reply(210);
+    giq_option(1, NAME, 211, Node022, NEUTRAL_REACTION);
+    giq_option(1, NAME, 212, Node023, NEUTRAL_REACTION);
+    giq_option(1, NAME, 213, Node024, NEUTRAL_REACTION);
 end
 
 procedure Node022
 begin
-    gsay_reply(519, 220);
-    gsay_option(519, 221, Node023, 50);
-    gsay_option(519, 222, Node026, 50);
+    Reply(220);
+    giq_option(1, NAME, 221, Node023, NEUTRAL_REACTION);
+    giq_option(1, NAME, 222, Node026, NEUTRAL_REACTION);
 end
 
 procedure Node023
 begin
-    gsay_reply(519, 230);
-    gsay_option(519, 231, Node012, 50);
+    Reply(230);
+    giq_option(1, NAME, 231, Node012, NEUTRAL_REACTION);
 end
 
 procedure Node024
 begin
-    gsay_reply(519, 240);
-    gsay_option(519, 241, Node025, 50);
+    Reply(240);
+    giq_option(1, NAME, 241, Node025, NEUTRAL_REACTION);
 end
 
 procedure Node025
 begin
-    gsay_reply(519, 250);
-    gsay_option(519, 251, Node012, 50);
+    Reply(250);
+    giq_option(1, NAME, 251, Node012, NEUTRAL_REACTION);
 end
 
 procedure Node026
 begin
-    gsay_reply(519, 260);
-    gsay_option(519, 261, Node027, 50);
+    Reply(260);
+    giq_option(1, NAME, 261, Node027, NEUTRAL_REACTION);
 end
 
 procedure Node027
 begin
-    gsay_reply(519, 270);
-    gsay_option(519, 271, Node028, 50);
+    Reply(270);
+    giq_option(1, NAME, 271, Node028, NEUTRAL_REACTION);
 end
 
 procedure Node028
 begin
-    gsay_reply(519, 280);
-    gsay_option(519, 281, Node031, 50);
-    gsay_option(519, 282, Node030, 50);
-    gsay_option(519, 283, Node029, 50);
+    Reply(280);
+    giq_option(1, NAME, 281, Node031, NEUTRAL_REACTION);
+    giq_option(1, NAME, 282, Node030, NEUTRAL_REACTION);
+    giq_option(1, NAME, 283, Node029, NEUTRAL_REACTION);
 end
 
 procedure Node029
 begin
-    gsay_reply(519, 290);
-    gsay_option(519, 291, Node012, 50);
+    Reply(290);
+    giq_option(1, NAME, 291, Node012, NEUTRAL_REACTION);
 end
 
 procedure Node030
 begin
-    gsay_reply(519, 300);
-    gsay_option(519, 301, Node012, 50);
+    Reply(300);
+    giq_option(1, NAME, 301, Node012, NEUTRAL_REACTION);
 end
 
 procedure Node031
 begin
-    gsay_reply(519, 310);
-    gsay_option(519, 311, Node033, 50);
-    gsay_option(519, 312, Node032, 50);
+    Reply(310);
+    giq_option(1, NAME, 311, Node033, NEUTRAL_REACTION);
+    giq_option(1, NAME, 312, Node032, NEUTRAL_REACTION);
 end
 
 procedure Node032
 begin
-    gsay_reply(519, 320);
-    gsay_option(519, 322, Node012, 50);
+    Reply(320);
+    giq_option(1, NAME, 322, Node012, NEUTRAL_REACTION);
 end
 
 procedure Node033
 begin
-    gsay_reply(519, 330);
-    gsay_option(519, 331, Node035, 50);
-    gsay_option(519, 332, Node034, 50);
+    Reply(330);
+    giq_option(1, NAME, 331, Node035, NEUTRAL_REACTION);
+    giq_option(1, NAME, 332, Node034, NEUTRAL_REACTION);
 end
 
 procedure Node034
 begin
-    gsay_reply(519, 340);
-    gsay_option(519, 341, Node012, 50);
+    Reply(340);
+    giq_option(1, NAME, 341, Node012, NEUTRAL_REACTION);
 end
 
 procedure Node035
 begin
-    gsay_reply(519, 350);
-    gsay_option(519, 351, Node036, 50);
+    Reply(350);
+    giq_option(1, NAME, 351, Node036, NEUTRAL_REACTION);
 end
 
 procedure Node036
 begin
-    gsay_reply(519, 360);
-    gsay_option(519, 361, Node041, 50);
-    gsay_option(519, 362, Node038, 50);
-    gsay_option(519, 363, Node037, 50);
+    Reply(360);
+    giq_option(1, NAME, 361, Node041, NEUTRAL_REACTION);
+    giq_option(1, NAME, 362, Node038, NEUTRAL_REACTION);
+    giq_option(1, NAME, 363, Node037, NEUTRAL_REACTION);
 end
 
 procedure Node037
 begin
-    gsay_reply(519, 370);
-    gsay_option(519, 371, Node012, 50);
+    Reply(370);
+    giq_option(1, NAME, 371, Node012, NEUTRAL_REACTION);
 end
 
 procedure Node038
 begin
-    gsay_reply(519, 380);
-    gsay_option(519, 381, Node039, 50);
+    Reply(380);
+    giq_option(1, NAME, 381, Node039, NEUTRAL_REACTION);
 end
 
 procedure Node039
 begin
-    gsay_reply(519, 390);
-    gsay_option(519, 391, Node040, 50);
+    Reply(390);
+    giq_option(1, NAME, 391, Node040, NEUTRAL_REACTION);
 end
 
 procedure Node040
 begin
-    gsay_reply(519, 400);
-    gsay_option(519, 401, Node012, 50);
+    Reply(400);
+    giq_option(1, NAME, 401, Node012, NEUTRAL_REACTION);
 end
 
 procedure Node041
 begin
-    set_global_var(432, 1);
-    gsay_reply(519, 410);
-    gsay_option(519, 411, Node999, 50);
+    set_global_var(GVAR_ENCLAVE_GRANITE_JOINED, 1);
+    Reply(410);
+    giq_option(1, NAME, 411, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node042
 begin
-    gsay_reply(519, 420);
-    gsay_option(519, 421, Node043, 50);
+    Reply(420);
+    giq_option(1, NAME, 421, Node043, NEUTRAL_REACTION);
 end
 
 procedure Node043
 begin
-    gsay_reply(519, 430);
-    gsay_option(519, 431, Node999, 50);
+    Reply(430);
+    giq_option(1, NAME, 431, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node044
 begin
-    gsay_reply(519, 440);
-    gsay_option(519, 441, Node010, 50);
-    gsay_option(519, 442, Node999, 50);
+    Reply(440);
+    giq_option(1, NAME, 441, Node010, NEUTRAL_REACTION);
+    giq_option(1, NAME, 442, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node045
 begin
-    gsay_reply(519, 450);
-    gsay_option(519, 451, Node010, 50);
-    gsay_option(519, 452, Node999, 50);
+    Reply(450);
+    giq_option(1, NAME, 451, Node010, NEUTRAL_REACTION);
+    giq_option(1, NAME, 452, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node100
 begin
-    set_global_var(432, 2);
-    gsay_reply(519, 1001);
-    gsay_option(519, 1002, Node101, 50);
+    set_global_var(GVAR_ENCLAVE_GRANITE_JOINED, 2);
+    Reply(1001);
+    giq_option(1, NAME, 1002, Node101, NEUTRAL_REACTION);
 end
 
 procedure Node101
 begin
-    gsay_reply(519, 1010);
-    gsay_option(519, 1011, Node999, 50);
-    gsay_option(519, 1012, Node102, 50);
+    Reply(1010);
+    giq_option(1, NAME, 1011, Node999, NEUTRAL_REACTION);
+    giq_option(1, NAME, 1012, Node102, NEUTRAL_REACTION);
 end
 
 procedure Node102
 begin
-    set_global_var(0, global_var(0) + 15);
-    gsay_reply(519, 1020);
-    gsay_option(519, 1021, Node999, 50);
+    set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + 15);
+    Reply(1020);
+    giq_option(1, NAME, 1021, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node999
@@ -654,4 +654,3 @@ begin
     gdialog_mod_barter(-(dude_objective_reputation + local_var(6)));
     call Node010();
 end
-
