diff --git a/Mods/Olympus2207/Original/OLKIRK.ssl b/Mods/Olympus2207/ReDefine/OLKIRK.ssl
index 6468566..3120df5 100644
--- a/Mods/Olympus2207/Original/OLKIRK.ssl
+++ b/Mods/Olympus2207/ReDefine/OLKIRK.ssl
@@ -70,8 +70,8 @@ begin
         set_local_var(5, random(60, 80));
         set_local_var(11, 1);
         set_local_var(6, random(-30, 20));
-        critter_add_trait(self_obj, 1, 6, 0);
-        critter_add_trait(self_obj, 1, 5, 3);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 0);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_TARTAR_MEN);
     end
 end
 
@@ -85,90 +85,90 @@ begin
         float_is_called := 1;
         add_timer_event(self_obj, random(200, 700), 2);
     end
-    if (global_var(265) == 1) then begin
+    if (global_var(GVAR_KARMA_RESERVED) == 1) then begin
         if (local_var(11) == 0) then begin
-            if ((tile_distance(tile_num(dude_obj), tile_num(self_obj)) > 3) and (tile_distance(tile_num(dude_obj), tile_num(self_obj)) <= 7)) then begin
-                animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(dude_obj), rotation_to_tile(tile_num(dude_obj), tile_num(self_obj)), random(2, 4)), 0);
+            if ((tile_distance(dude_tile, tile_num(self_obj)) > 3) and (tile_distance(dude_tile, tile_num(self_obj)) <= 7)) then begin
+                animate_move_obj_to_tile(self_obj, tile_num_in_direction(dude_tile, rotation_to_tile(dude_tile, tile_num(self_obj)), random(2, 4)), 0);
             end
             else begin
-                if (tile_distance(tile_num(dude_obj), tile_num(self_obj)) > 7) then begin
-                    animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(dude_obj), rotation_to_tile(tile_num(dude_obj), tile_num(self_obj)), random(2, 4)), 1);
+                if (tile_distance(dude_tile, tile_num(self_obj)) > 7) then begin
+                    animate_move_obj_to_tile(self_obj, tile_num_in_direction(dude_tile, rotation_to_tile(dude_tile, tile_num(self_obj)), random(2, 4)), 1);
                 end
             end
         end
         else begin
             if (local_var(11) == 1) then begin
-                if ((tile_distance(tile_num(dude_obj), tile_num(self_obj)) > 6) and (tile_distance(tile_num(dude_obj), tile_num(self_obj)) <= 10)) then begin
-                    animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(dude_obj), rotation_to_tile(tile_num(dude_obj), tile_num(self_obj)), random(3, 5)), 0);
+                if ((tile_distance(dude_tile, tile_num(self_obj)) > 6) and (tile_distance(dude_tile, tile_num(self_obj)) <= 10)) then begin
+                    animate_move_obj_to_tile(self_obj, tile_num_in_direction(dude_tile, rotation_to_tile(dude_tile, tile_num(self_obj)), random(3, 5)), 0);
                 end
                 else begin
-                    if (tile_distance(tile_num(dude_obj), tile_num(self_obj)) > 10) then begin
-                        animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(dude_obj), rotation_to_tile(tile_num(dude_obj), tile_num(self_obj)), random(3, 5)), 1);
+                    if (tile_distance(dude_tile, tile_num(self_obj)) > 10) then begin
+                        animate_move_obj_to_tile(self_obj, tile_num_in_direction(dude_tile, rotation_to_tile(dude_tile, tile_num(self_obj)), random(3, 5)), 1);
                     end
                 end
             end
             else begin
-                if ((tile_distance(tile_num(dude_obj), tile_num(self_obj)) > 10) and (tile_distance(tile_num(dude_obj), tile_num(self_obj)) <= 16)) then begin
-                    animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(dude_obj), rotation_to_tile(tile_num(dude_obj), tile_num(self_obj)), random(4, 6)), 0);
+                if ((tile_distance(dude_tile, tile_num(self_obj)) > 10) and (tile_distance(dude_tile, tile_num(self_obj)) <= 16)) then begin
+                    animate_move_obj_to_tile(self_obj, tile_num_in_direction(dude_tile, rotation_to_tile(dude_tile, tile_num(self_obj)), random(4, 6)), 0);
                 end
                 else begin
-                    if (tile_distance(tile_num(dude_obj), tile_num(self_obj)) > 16) then begin
-                        animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(dude_obj), rotation_to_tile(tile_num(dude_obj), tile_num(self_obj)), random(4, 6)), 1);
+                    if (tile_distance(dude_tile, tile_num(self_obj)) > 16) then begin
+                        animate_move_obj_to_tile(self_obj, tile_num_in_direction(dude_tile, rotation_to_tile(dude_tile, tile_num(self_obj)), random(4, 6)), 1);
                     end
                 end
             end
         end
     end
-    if (not(critter_inven_obj(self_obj, 0)) and (obj_art_fid(self_obj) != 16777226)) then begin
-        metarule3(107, self_obj, 16777226, 0);
+    if (not(critter_inven_obj(self_obj, INVEN_TYPE_WORN)) and (obj_art_fid(self_obj) != 16777226)) then begin
+        art_change_fid_num(self_obj, 16777226);
     end
-    if ((obj_pid(critter_inven_obj(self_obj, 0)) == 113) and (obj_art_fid(self_obj) != 16777246)) then begin
-        metarule3(107, self_obj, 16777246, 0);
+    if ((obj_pid(critter_inven_obj(self_obj, INVEN_TYPE_WORN)) == 113) and (obj_art_fid(self_obj) != 16777246)) then begin
+        art_change_fid_num(self_obj, 16777246);
     end
-    if ((obj_pid(critter_inven_obj(self_obj, 0)) == 114) and (obj_art_fid(self_obj) != 16777312)) then begin
-        metarule3(107, self_obj, 16777312, 0);
+    if ((obj_pid(critter_inven_obj(self_obj, INVEN_TYPE_WORN)) == 114) and (obj_art_fid(self_obj) != 16777312)) then begin
+        art_change_fid_num(self_obj, 16777312);
     end
-    if ((obj_pid(critter_inven_obj(self_obj, 0)) == 115) and (obj_art_fid(self_obj) != 16777244)) then begin
-        metarule3(107, self_obj, 16777244, 0);
+    if ((obj_pid(critter_inven_obj(self_obj, INVEN_TYPE_WORN)) == 115) and (obj_art_fid(self_obj) != 16777244)) then begin
+        art_change_fid_num(self_obj, 16777244);
     end
-    if ((obj_pid(critter_inven_obj(self_obj, 0)) == 116) and (obj_art_fid(self_obj) != 16777244)) then begin
-        metarule3(107, self_obj, 16777244, 0);
+    if ((obj_pid(critter_inven_obj(self_obj, INVEN_TYPE_WORN)) == 116) and (obj_art_fid(self_obj) != 16777244)) then begin
+        art_change_fid_num(self_obj, 16777244);
     end
-    if ((obj_pid(critter_inven_obj(self_obj, 0)) == 117) and (obj_art_fid(self_obj) != 16777240)) then begin
-        metarule3(107, self_obj, 16777240, 0);
+    if ((obj_pid(critter_inven_obj(self_obj, INVEN_TYPE_WORN)) == 117) and (obj_art_fid(self_obj) != 16777240)) then begin
+        art_change_fid_num(self_obj, 16777240);
     end
-    if ((obj_pid(critter_inven_obj(self_obj, 0)) == 118) and (obj_art_fid(self_obj) != 16777277)) then begin
-        metarule3(107, self_obj, 16777277, 0);
+    if ((obj_pid(critter_inven_obj(self_obj, INVEN_TYPE_WORN)) == 118) and (obj_art_fid(self_obj) != 16777277)) then begin
+        art_change_fid_num(self_obj, 16777277);
     end
-    if ((obj_pid(critter_inven_obj(self_obj, 0)) == 119) and (obj_art_fid(self_obj) != 16777250)) then begin
-        metarule3(107, self_obj, 16777250, 0);
+    if ((obj_pid(critter_inven_obj(self_obj, INVEN_TYPE_WORN)) == 119) and (obj_art_fid(self_obj) != 16777250)) then begin
+        art_change_fid_num(self_obj, 16777250);
     end
-    if ((obj_pid(critter_inven_obj(self_obj, 0)) == 120) or (obj_pid(critter_inven_obj(self_obj, 0)) == 121) or (obj_pid(critter_inven_obj(self_obj, 0)) == 122) or (obj_pid(critter_inven_obj(self_obj, 0)) == 123) and (obj_art_fid(self_obj) != 16777248)) then begin
-        metarule3(107, self_obj, 16777248, 0);
+    if ((obj_pid(critter_inven_obj(self_obj, INVEN_TYPE_WORN)) == 120) or (obj_pid(critter_inven_obj(self_obj, INVEN_TYPE_WORN)) == 121) or (obj_pid(critter_inven_obj(self_obj, INVEN_TYPE_WORN)) == 122) or (obj_pid(critter_inven_obj(self_obj, INVEN_TYPE_WORN)) == 123) and (obj_art_fid(self_obj) != 16777248)) then begin
+        art_change_fid_num(self_obj, 16777248);
     end
-    if ((obj_pid(critter_inven_obj(self_obj, 0)) == 124) and (obj_art_fid(self_obj) != 16777252)) then begin
-        metarule3(107, self_obj, 16777252, 0);
+    if ((obj_pid(critter_inven_obj(self_obj, INVEN_TYPE_WORN)) == 124) and (obj_art_fid(self_obj) != 16777252)) then begin
+        art_change_fid_num(self_obj, 16777252);
     end
-    if ((obj_pid(critter_inven_obj(self_obj, 0)) == 125) and (obj_art_fid(self_obj) != 16777253)) then begin
-        metarule3(107, self_obj, 16777253, 0);
+    if ((obj_pid(critter_inven_obj(self_obj, INVEN_TYPE_WORN)) == 125) and (obj_art_fid(self_obj) != 16777253)) then begin
+        art_change_fid_num(self_obj, 16777253);
     end
-    if ((obj_pid(critter_inven_obj(self_obj, 0)) == 126) and (obj_art_fid(self_obj) != 16777253)) then begin
-        metarule3(107, self_obj, 16777253, 0);
+    if ((obj_pid(critter_inven_obj(self_obj, INVEN_TYPE_WORN)) == 126) and (obj_art_fid(self_obj) != 16777253)) then begin
+        art_change_fid_num(self_obj, 16777253);
     end
-    if ((obj_pid(critter_inven_obj(self_obj, 0)) == 127) and (obj_art_fid(self_obj) != 16777254)) then begin
-        metarule3(107, self_obj, 16777254, 0);
+    if ((obj_pid(critter_inven_obj(self_obj, INVEN_TYPE_WORN)) == 127) and (obj_art_fid(self_obj) != 16777254)) then begin
+        art_change_fid_num(self_obj, 16777254);
     end
-    if ((obj_pid(critter_inven_obj(self_obj, 0)) == 128) and (obj_art_fid(self_obj) != 16777256)) then begin
-        metarule3(107, self_obj, 16777256, 0);
+    if ((obj_pid(critter_inven_obj(self_obj, INVEN_TYPE_WORN)) == 128) and (obj_art_fid(self_obj) != 16777256)) then begin
+        art_change_fid_num(self_obj, 16777256);
     end
-    if ((obj_pid(critter_inven_obj(self_obj, 0)) == 129) and (obj_art_fid(self_obj) != 16777255)) then begin
-        metarule3(107, self_obj, 16777255, 0);
+    if ((obj_pid(critter_inven_obj(self_obj, INVEN_TYPE_WORN)) == 129) and (obj_art_fid(self_obj) != 16777255)) then begin
+        art_change_fid_num(self_obj, 16777255);
     end
-    if ((obj_pid(critter_inven_obj(self_obj, 0)) == 391) and (obj_art_fid(self_obj) != 16777237)) then begin
-        metarule3(107, self_obj, 16777237, 0);
+    if ((obj_pid(critter_inven_obj(self_obj, INVEN_TYPE_WORN)) == 391) and (obj_art_fid(self_obj) != 16777237)) then begin
+        art_change_fid_num(self_obj, 16777237);
     end
     if (obj_can_see_obj(self_obj, dude_obj)) then begin
-        if ((local_var(2) == 1) or (global_var(47) <= -31)) then begin
+        if ((local_var(2) == 1) or (global_var(GVAR_TOWN_REP_ARROYO) <= -31)) then begin
             set_local_var(2, 0);
             attack_setup(self_obj, dude_obj);
         end
@@ -183,10 +183,10 @@ procedure description_p_proc
 begin
     script_overrides;
     if (local_var(7) == 0) then begin
-        display_msg(message_str(92, 2));
+        display_mstr(2);
     end
     else begin
-        display_msg(message_str(92, 4));
+        display_mstr(4);
     end
 end
 
@@ -194,10 +194,10 @@ procedure look_at_p_proc
 begin
     script_overrides;
     if (local_var(7) == 0) then begin
-        display_msg(message_str(92, 1));
+        display_mstr(1);
     end
     else begin
-        display_msg(message_str(92, 3));
+        display_mstr(3);
     end
 end
 
@@ -215,24 +215,24 @@ end
 procedure talk_p_proc
 begin
     if (not(anim_busy(self_obj))) then begin
-        anim(self_obj, 1000, rotation_to_tile(tile_num(self_obj), tile_num(dude_obj)));
+        anim(self_obj, 1000, rotation_to_tile(tile_num(self_obj), dude_tile));
     end
-    if ((local_var(14) == 1) and not(global_var(6) == 3)) then begin
-        float_msg(self_obj, message_str(92, random(1030, 1033)), 0);
+    if ((local_var(14) == 1) and not(global_var(GVAR_PLAYER_MARRIED) == 3)) then begin
+        float_msg(self_obj, mstr(random(1030, 1033)), FLOAT_MSG_NORMAL);
     end
     else begin
         if (local_var(13) == 1) then begin
             start_gdialog(92, self_obj, -1, -1, -1);
             gsay_start;
-            if (global_var(265) == 1) then begin
+            if (global_var(GVAR_KARMA_RESERVED) == 1) then begin
                 call DialogParty();
             end
             else begin
-                if (not(global_var(6) == 3)) then begin
+                if (not(global_var(GVAR_PLAYER_MARRIED) == 3)) then begin
                     call Node001();
                 end
                 else begin
-                    if (not((global_var(69) bwand 1) != 0) and (local_var(7) == 2) and (global_var(122) != 2)) then begin
+                    if (not((global_var(GVAR_TORR_HARMED) bwand 1) != 0) and (local_var(7) == 2) and (global_var(GVAR_MODOC_SLAUGHTER_FLAG) != 2)) then begin
                         call Node024();
                     end
                     else begin
@@ -244,23 +244,23 @@ begin
             end_dialogue;
         end
         else begin
-            float_msg(self_obj, message_str(92, random(1038, 1039)), 0);
+            float_msg(self_obj, mstr(random(1038, 1039)), FLOAT_MSG_NORMAL);
         end
     end
 end
 
 procedure destroy_p_proc
 begin
-    if (global_var(265) == 1) then begin
-        set_global_var(265, 0);
+    if (global_var(GVAR_KARMA_RESERVED) == 1) then begin
+        set_global_var(GVAR_KARMA_RESERVED, 0);
     end
     if (source_obj == dude_obj) then begin
-        set_global_var(0, global_var(0) - 1);
+        set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) - 1);
         if ((47 == 50) or (47 == 59)) then begin
-            set_global_var(47, global_var(47) - 3);
+            set_global_var(GVAR_TOWN_REP_ARROYO, global_var(GVAR_TOWN_REP_ARROYO) - 3);
         end
         else begin
-            set_global_var(47, global_var(47) - 6);
+            set_global_var(GVAR_TOWN_REP_ARROYO, global_var(GVAR_TOWN_REP_ARROYO) - 6);
         end
     end
 end
@@ -270,31 +270,31 @@ begin
     if (action_being_used == 10) then begin
         steal_perk := 0;
         steal_gloves := 0;
-        if (has_trait(0, dude_obj, 37) != 0) then begin
+        if (has_trait(TRAIT_PERK, dude_obj, PERK_pickpocket) != 0) then begin
             steal_perk := steal_perk + 15;
         end
-        if (has_trait(0, dude_obj, 33) != 0) then begin
+        if (has_trait(TRAIT_PERK, dude_obj, PERK_master_thief) != 0) then begin
             steal_perk := steal_perk + 30;
         end
-        if ((obj_pid(critter_inven_obj(dude_obj, 1)) == 245) or (obj_pid(critter_inven_obj(dude_obj, 2)) == 245)) then begin
+        if ((obj_pid(critter_inven_obj(dude_obj, INVEN_TYPE_RIGHT_HAND)) == 245) or (obj_pid(critter_inven_obj(dude_obj, INVEN_TYPE_LEFT_HAND)) == 245)) then begin
             steal_gloves := 10;
         end
-        if (random(0, 100) <= ((has_skill(dude_obj, 10) / 2) + (((3 * get_critter_stat(dude_obj, 5)) + (((get_critter_stat(dude_obj, 6) - 4) * 4) - (difficulty_level * 4)) + (2 * steal_perk) + steal_gloves) / (local_var(4) / 25)))) then begin
-            display_msg(message_str(10, random(145, 148)));
+        if (random(0, 100) <= ((dude_skill(SKILL_STEAL) / 2) + (((3 * get_critter_stat(dude_obj, STAT_ag)) + (((dude_luck - 4) * 4) - (difficulty_level * 4)) + (2 * steal_perk) + steal_gloves) / (local_var(4) / 25)))) then begin
+            display_msg(message_str(SCRIPT_STTEXT, random(145, 148)));
         end
         else begin
             script_overrides;
-            if (random(0, 100) <= (25 - floor((((get_critter_stat(dude_obj, 6) - 4) * 4) - (difficulty_level * 4)) / 3))) then begin
-                float_msg(self_obj, message_str(10, random(206, 209)), 2);
+            if (random(0, 100) <= (25 - floor((((dude_luck - 4) * 4) - (difficulty_level * 4)) / 3))) then begin
+                float_msg(self_obj, message_str(SCRIPT_STTEXT, random(206, 209)), FLOAT_MSG_RED);
             end
             else begin
-                display_msg(message_str(10, random(200, 203)));
+                display_msg(message_str(SCRIPT_STTEXT, random(200, 203)));
             end
         end
     end
     if (action_being_used == 12) then begin
         script_overrides;
-        display_msg(message_str(10, 204) + ((has_skill(dude_obj, 10) / 2) + (((3 * get_critter_stat(dude_obj, 5)) + (((get_critter_stat(dude_obj, 6) - 4) * 4) - (difficulty_level * 4)) + (2 * steal_perk) + steal_gloves) / (local_var(4) / 25))) + message_str(10, 205));
+        display_msg(message_str(SCRIPT_STTEXT, 204) + ((dude_skill(SKILL_STEAL) / 2) + (((3 * get_critter_stat(dude_obj, STAT_ag)) + (((dude_luck - 4) * 4) - (difficulty_level * 4)) + (2 * steal_perk) + steal_gloves) / (local_var(4) / 25))) + message_str(SCRIPT_STTEXT, 205));
     end
 end
 
@@ -325,48 +325,48 @@ end
 procedure Node001
 begin
     set_local_var(14, 1);
-    gsay_reply(92, message_str(92, 10) + obj_name(dude_obj) + message_str(92, 11));
-    gsay_option(92, 12, Node999, 50);
-    gsay_option(92, 13, Node999, 50);
+    Reply(mstr(10) + dude_name + mstr(11));
+    giq_option(1, NAME, 12, Node999, NEUTRAL_REACTION);
+    giq_option(1, NAME, 13, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node010
 begin
-    if (global_var(265) == 2) then begin
-        gsay_reply(92, 900);
-        gsay_option(92, 929, NodeParty, 50);
-        gsay_option(92, 919, Node999, 50);
+    if (global_var(GVAR_KARMA_RESERVED) == 2) then begin
+        Reply(900);
+        giq_option(1, NAME, 929, NodeParty, NEUTRAL_REACTION);
+        giq_option(1, NAME, 919, Node999, NEUTRAL_REACTION);
     end
     else begin
-        if ((global_var(122) == 2) and not((global_var(69) bwand 1) != 0)) then begin
-            gsay_reply(92, 104);
-            gsay_option(92, 103, Node999, 50);
-            gsay_option(92, 105, Node999, 50);
+        if ((global_var(GVAR_MODOC_SLAUGHTER_FLAG) == 2) and not((global_var(GVAR_TORR_HARMED) bwand 1) != 0)) then begin
+            Reply(104);
+            giq_option(1, NAME, 103, Node999, NEUTRAL_REACTION);
+            giq_option(1, NAME, 105, Node999, NEUTRAL_REACTION);
         end
         else begin
-            if ((global_var(69) bwand 1) != 0) then begin
+            if ((global_var(GVAR_TORR_HARMED) bwand 1) != 0) then begin
                 if (local_var(7) == 3) then begin
-                    gsay_reply(92, 223);
-                    gsay_option(92, 221, NodeParty, 50);
-                    gsay_option(92, 222, Node999, 50);
+                    Reply(223);
+                    giq_option(1, NAME, 221, NodeParty, NEUTRAL_REACTION);
+                    giq_option(1, NAME, 222, Node999, NEUTRAL_REACTION);
                 end
                 else begin
-                    gsay_reply(92, 200);
-                    gsay_option(92, 201, Node021, 50);
-                    gsay_option(92, 202, Node023, 50);
-                    gsay_option(92, 203, Node021, 50);
+                    Reply(200);
+                    giq_option(1, NAME, 201, Node021, NEUTRAL_REACTION);
+                    giq_option(1, NAME, 202, Node023, NEUTRAL_REACTION);
+                    giq_option(1, NAME, 203, Node021, NEUTRAL_REACTION);
                 end
             end
             else begin
                 if (local_var(7) == 0) then begin
-                    gsay_reply(92, obj_name(dude_obj) + message_str(92, 100));
-                    gsay_option(92, 101, Node011, 50);
-                    gsay_option(92, 102, Node012, 50);
-                    gsay_option(92, 103, Node999, 50);
+                    Reply(dude_name + mstr(100));
+                    giq_option(1, NAME, 101, Node011, NEUTRAL_REACTION);
+                    giq_option(1, NAME, 102, Node012, NEUTRAL_REACTION);
+                    giq_option(1, NAME, 103, Node999, NEUTRAL_REACTION);
                 end
                 if (local_var(7) == 1) then begin
-                    gsay_reply(92, 104);
-                    gsay_option(92, 105, Node999, 50);
+                    Reply(104);
+                    giq_option(1, NAME, 105, Node999, NEUTRAL_REACTION);
                 end
             end
         end
@@ -375,26 +375,26 @@ end
 
 procedure Node011
 begin
-    gsay_reply(92, 110);
-    gsay_option(92, 111, Node012, 50);
-    gsay_option(92, 112, Node012, 50);
+    Reply(110);
+    giq_option(1, NAME, 111, Node012, NEUTRAL_REACTION);
+    giq_option(1, NAME, 112, Node012, NEUTRAL_REACTION);
 end
 
 procedure Node012
 begin
     set_local_var(7, 1);
-    gsay_reply(92, 120);
-    gsay_option(92, 121, Node999, 50);
-    gsay_option(92, 122, Node999, 50);
-    gsay_option(92, 123, Node999, 50);
+    Reply(120);
+    giq_option(1, NAME, 121, Node999, NEUTRAL_REACTION);
+    giq_option(1, NAME, 122, Node999, NEUTRAL_REACTION);
+    giq_option(1, NAME, 123, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node021
 begin
-    gsay_reply(92, message_str(92, 210) + obj_name(dude_obj) + message_str(92, 211));
-    gsay_option(92, 212, Node0211, 50);
-    gsay_option(92, 213, Node0212, 50);
-    gsay_option(92, 214, Node999, 50);
+    Reply(mstr(210) + dude_name + mstr(211));
+    giq_option(1, NAME, 212, Node0211, NEUTRAL_REACTION);
+    giq_option(1, NAME, 213, Node0212, NEUTRAL_REACTION);
+    giq_option(1, NAME, 214, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node0211
@@ -411,48 +411,48 @@ end
 
 procedure Node022
 begin
-    if (random(0, 100) <= (floor((has_skill(dude_obj, 14) + (get_critter_stat(dude_obj, 3) * 10)) / 3) + persuade_mod)) then begin
+    if (random(0, 100) <= (floor((dude_skill(SKILL_SPEECH) + (dude_charisma * 10)) / 3) + persuade_mod)) then begin
         set_local_var(7, 3);
-        gsay_reply(92, 220);
-        gsay_option(92, 221, NodeParty, 50);
-        gsay_option(92, 222, Node999, 50);
+        Reply(220);
+        giq_option(1, NAME, 221, NodeParty, NEUTRAL_REACTION);
+        giq_option(1, NAME, 222, Node999, NEUTRAL_REACTION);
     end
     else begin
         set_local_var(13, 0);
-        gsay_reply(92, 234);
+        Reply(234);
         if (persuade_mod >= 25) then begin
-            gsay_option(92, 231, Node0212, 50);
+            giq_option(1, NAME, 231, Node0212, NEUTRAL_REACTION);
         end
-        gsay_option(92, 232, Node999, 50);
-        gsay_option(92, 233, Node999, 50);
+        giq_option(1, NAME, 232, Node999, NEUTRAL_REACTION);
+        giq_option(1, NAME, 233, Node999, NEUTRAL_REACTION);
     end
 end
 
 procedure Node023
 begin
-    gsay_reply(92, 230);
-    gsay_option(92, 231, Node0211, 50);
-    gsay_option(92, 232, Node999, 50);
-    gsay_option(92, 233, Node999, 50);
+    Reply(230);
+    giq_option(1, NAME, 231, Node0211, NEUTRAL_REACTION);
+    giq_option(1, NAME, 232, Node999, NEUTRAL_REACTION);
+    giq_option(1, NAME, 233, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node024
 begin
-    gsay_reply(92, 240);
+    Reply(240);
     if (obj_carrying_pid_obj(dude_obj, 335)) then begin
-        gsay_option(92, 242, Node999, 50);
+        giq_option(1, NAME, 242, Node999, NEUTRAL_REACTION);
     end
     else begin
-        gsay_option(92, 241, Node025, 50);
+        giq_option(1, NAME, 241, Node025, NEUTRAL_REACTION);
     end
-    gsay_option(92, 243, Node999, 50);
+    giq_option(1, NAME, 243, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node025
 begin
-    gsay_reply(92, 250);
-    gsay_option(92, 251, Node999, 50);
-    gsay_option(92, 252, Node999, 50);
+    Reply(250);
+    giq_option(1, NAME, 251, Node999, NEUTRAL_REACTION);
+    giq_option(1, NAME, 252, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node999
@@ -465,7 +465,7 @@ procedure NodeParty
 begin
     set_local_var(7, 4);
     party_add(self_obj);
-    set_global_var(265, 1);
+    set_global_var(GVAR_KARMA_RESERVED, 1);
     call DialogParty();
 end
 
@@ -473,58 +473,58 @@ procedure DialogParty
 begin
     if (already_talk == 0) then begin
         already_talk := 1;
-        gsay_reply(92, random(900, 904));
+        Reply(random(900, 904));
     end
     else begin
-        gsay_reply(92, 920);
+        Reply(920);
     end
-    if (get_critter_stat(self_obj, 35) < get_critter_stat(self_obj, 7)) then begin
-        gsay_option(92, 910, Healing, 50);
+    if (get_critter_stat(self_obj, STAT_current_hp) < get_critter_stat(self_obj, STAT_max_hp)) then begin
+        giq_option(1, NAME, 910, Healing, NEUTRAL_REACTION);
     end
-    gsay_option(92, 911, Armor, 50);
-    gsay_option(92, 912, Weapon, 50);
-    gsay_option(92, 913, Distance, 50);
-    gsay_option(92, 914, DialogParty2, 50);
-    gsay_option(92, 919, Node999, 50);
+    giq_option(1, NAME, 911, Armor, NEUTRAL_REACTION);
+    giq_option(1, NAME, 912, Weapon, NEUTRAL_REACTION);
+    giq_option(1, NAME, 913, Distance, NEUTRAL_REACTION);
+    giq_option(1, NAME, 914, DialogParty2, NEUTRAL_REACTION);
+    giq_option(1, NAME, 919, Node999, NEUTRAL_REACTION);
 end
 
 procedure Healing
 begin
-    if (get_critter_stat(self_obj, 35) >= (get_critter_stat(self_obj, 7) / 4 * 3)) then begin
-        gsay_reply(92, 930);
+    if (get_critter_stat(self_obj, STAT_current_hp) >= (get_critter_stat(self_obj, STAT_max_hp) / 4 * 3)) then begin
+        Reply(930);
     end
     else begin
-        if (get_critter_stat(self_obj, 35) >= (get_critter_stat(self_obj, 7) / 4 * 2)) then begin
-            gsay_reply(92, 931);
+        if (get_critter_stat(self_obj, STAT_current_hp) >= (get_critter_stat(self_obj, STAT_max_hp) / 4 * 2)) then begin
+            Reply(931);
         end
         else begin
-            if (get_critter_stat(self_obj, 35) >= (get_critter_stat(self_obj, 7) / 4)) then begin
-                gsay_reply(92, 932);
+            if (get_critter_stat(self_obj, STAT_current_hp) >= (get_critter_stat(self_obj, STAT_max_hp) / 4)) then begin
+                Reply(932);
             end
             else begin
-                gsay_reply(92, 933);
+                Reply(933);
             end
         end
     end
     if (obj_carrying_pid_obj(self_obj, 220) or obj_carrying_pid_obj(self_obj, 221) or obj_carrying_pid_obj(self_obj, 222) or obj_carrying_pid_obj(self_obj, 223)) then begin
-        gsay_option(92, 940, Healing_self, 50);
+        giq_option(1, NAME, 940, Healing_self, NEUTRAL_REACTION);
     end
     if (obj_carrying_pid_obj(dude_obj, 220)) then begin
-        gsay_option(92, message_str(92, 934) + message_str(92, 936) + message_str(92, 935), Healing_dude_binty, 50);
+        giq_option(1, NAME, mstr(934) + mstr(936) + mstr(935), Healing_dude_binty, NEUTRAL_REACTION);
     end
     if (obj_carrying_pid_obj(dude_obj, 221)) then begin
-        gsay_option(92, message_str(92, 934) + message_str(92, 937) + message_str(92, 935), Healing_dude_analgetic, 50);
+        giq_option(1, NAME, mstr(934) + mstr(937) + mstr(935), Healing_dude_analgetic, NEUTRAL_REACTION);
     end
     if (obj_carrying_pid_obj(dude_obj, 222)) then begin
-        gsay_option(92, message_str(92, 934) + message_str(92, 938) + message_str(92, 935), Healing_dude_heal_up, 50);
+        giq_option(1, NAME, mstr(934) + mstr(938) + mstr(935), Healing_dude_heal_up, NEUTRAL_REACTION);
     end
     if (obj_carrying_pid_obj(dude_obj, 223)) then begin
-        gsay_option(92, message_str(92, 934) + message_str(92, 939) + message_str(92, 935), Healing_dude_lazareth, 50);
+        giq_option(1, NAME, mstr(934) + mstr(939) + mstr(935), Healing_dude_lazareth, NEUTRAL_REACTION);
     end
     if (not(obj_carrying_pid_obj(self_obj, 220)) and not(obj_carrying_pid_obj(self_obj, 221)) and not(obj_carrying_pid_obj(self_obj, 222)) and not(obj_carrying_pid_obj(self_obj, 223)) and not(obj_carrying_pid_obj(dude_obj, 220)) and not(obj_carrying_pid_obj(dude_obj, 221)) and not(obj_carrying_pid_obj(dude_obj, 222)) and not(obj_carrying_pid_obj(dude_obj, 223))) then begin
-        gsay_option(92, 946, Node999, 50);
+        giq_option(1, NAME, 946, Node999, NEUTRAL_REACTION);
     end
-    gsay_option(92, 947, DialogParty, 50);
+    giq_option(1, NAME, 947, DialogParty, NEUTRAL_REACTION);
 end
 
 procedure Healing_self
@@ -545,160 +545,160 @@ begin
             end
         end
     end
-    gsay_reply(92, 941);
-    gsay_option(92, 918, DialogParty, 50);
+    Reply(941);
+    giq_option(1, NAME, 918, DialogParty, NEUTRAL_REACTION);
 end
 
 procedure Healing_dude_binty
 begin
     rm_obj_from_inven(dude_obj, obj_carrying_pid_obj(dude_obj, 220));
-    add_obj_to_inven(self_obj, create_object_sid(220, 1, 2, -1));
+    add_obj_to_inven(self_obj, create_object(220, 1, 2));
     use_obj_on_obj(obj_carrying_pid_obj(self_obj, 220), self_obj);
-    if (get_critter_stat(self_obj, 35) >= (get_critter_stat(self_obj, 7) / 4 * 3)) then begin
-        gsay_reply(92, 942);
+    if (get_critter_stat(self_obj, STAT_current_hp) >= (get_critter_stat(self_obj, STAT_max_hp) / 4 * 3)) then begin
+        Reply(942);
     end
     else begin
-        if (get_critter_stat(self_obj, 35) >= (get_critter_stat(self_obj, 7) / 4 * 2)) then begin
-            gsay_reply(92, 943);
+        if (get_critter_stat(self_obj, STAT_current_hp) >= (get_critter_stat(self_obj, STAT_max_hp) / 4 * 2)) then begin
+            Reply(943);
         end
         else begin
-            if (get_critter_stat(self_obj, 35) >= (get_critter_stat(self_obj, 7) / 4)) then begin
-                gsay_reply(92, 944);
+            if (get_critter_stat(self_obj, STAT_current_hp) >= (get_critter_stat(self_obj, STAT_max_hp) / 4)) then begin
+                Reply(944);
             end
             else begin
-                gsay_reply(92, 945);
+                Reply(945);
             end
         end
     end
-    gsay_option(92, 918, DialogParty, 50);
+    giq_option(1, NAME, 918, DialogParty, NEUTRAL_REACTION);
 end
 
 procedure Healing_dude_analgetic
 begin
     rm_obj_from_inven(dude_obj, obj_carrying_pid_obj(dude_obj, 221));
-    add_obj_to_inven(self_obj, create_object_sid(221, 1, 2, -1));
+    add_obj_to_inven(self_obj, create_object(221, 1, 2));
     use_obj_on_obj(obj_carrying_pid_obj(self_obj, 221), self_obj);
-    if (get_critter_stat(self_obj, 35) >= (get_critter_stat(self_obj, 7) / 4 * 3)) then begin
-        gsay_reply(92, 942);
+    if (get_critter_stat(self_obj, STAT_current_hp) >= (get_critter_stat(self_obj, STAT_max_hp) / 4 * 3)) then begin
+        Reply(942);
     end
     else begin
-        if (get_critter_stat(self_obj, 35) >= (get_critter_stat(self_obj, 7) / 4 * 2)) then begin
-            gsay_reply(92, 943);
+        if (get_critter_stat(self_obj, STAT_current_hp) >= (get_critter_stat(self_obj, STAT_max_hp) / 4 * 2)) then begin
+            Reply(943);
         end
         else begin
-            if (get_critter_stat(self_obj, 35) >= (get_critter_stat(self_obj, 7) / 4)) then begin
-                gsay_reply(92, 944);
+            if (get_critter_stat(self_obj, STAT_current_hp) >= (get_critter_stat(self_obj, STAT_max_hp) / 4)) then begin
+                Reply(944);
             end
             else begin
-                gsay_reply(92, 945);
+                Reply(945);
             end
         end
     end
-    gsay_option(92, 918, DialogParty, 50);
+    giq_option(1, NAME, 918, DialogParty, NEUTRAL_REACTION);
 end
 
 procedure Healing_dude_heal_up
 begin
     rm_obj_from_inven(dude_obj, obj_carrying_pid_obj(dude_obj, 222));
-    add_obj_to_inven(self_obj, create_object_sid(222, 1, 2, -1));
+    add_obj_to_inven(self_obj, create_object(222, 1, 2));
     use_obj_on_obj(obj_carrying_pid_obj(self_obj, 222), self_obj);
-    if (get_critter_stat(self_obj, 35) >= (get_critter_stat(self_obj, 7) / 4 * 3)) then begin
-        gsay_reply(92, 942);
+    if (get_critter_stat(self_obj, STAT_current_hp) >= (get_critter_stat(self_obj, STAT_max_hp) / 4 * 3)) then begin
+        Reply(942);
     end
     else begin
-        if (get_critter_stat(self_obj, 35) >= (get_critter_stat(self_obj, 7) / 4 * 2)) then begin
-            gsay_reply(92, 943);
+        if (get_critter_stat(self_obj, STAT_current_hp) >= (get_critter_stat(self_obj, STAT_max_hp) / 4 * 2)) then begin
+            Reply(943);
         end
         else begin
-            if (get_critter_stat(self_obj, 35) >= (get_critter_stat(self_obj, 7) / 4)) then begin
-                gsay_reply(92, 944);
+            if (get_critter_stat(self_obj, STAT_current_hp) >= (get_critter_stat(self_obj, STAT_max_hp) / 4)) then begin
+                Reply(944);
             end
             else begin
-                gsay_reply(92, 945);
+                Reply(945);
             end
         end
     end
-    gsay_option(92, 918, DialogParty, 50);
+    giq_option(1, NAME, 918, DialogParty, NEUTRAL_REACTION);
 end
 
 procedure Healing_dude_lazareth
 begin
     rm_obj_from_inven(dude_obj, obj_carrying_pid_obj(dude_obj, 223));
-    add_obj_to_inven(self_obj, create_object_sid(223, 1, 2, -1));
+    add_obj_to_inven(self_obj, create_object(223, 1, 2));
     use_obj_on_obj(obj_carrying_pid_obj(self_obj, 223), self_obj);
-    if (get_critter_stat(self_obj, 35) >= (get_critter_stat(self_obj, 7) / 4 * 3)) then begin
-        gsay_reply(92, 942);
+    if (get_critter_stat(self_obj, STAT_current_hp) >= (get_critter_stat(self_obj, STAT_max_hp) / 4 * 3)) then begin
+        Reply(942);
     end
     else begin
-        if (get_critter_stat(self_obj, 35) >= (get_critter_stat(self_obj, 7) / 4 * 2)) then begin
-            gsay_reply(92, 943);
+        if (get_critter_stat(self_obj, STAT_current_hp) >= (get_critter_stat(self_obj, STAT_max_hp) / 4 * 2)) then begin
+            Reply(943);
         end
         else begin
-            if (get_critter_stat(self_obj, 35) >= (get_critter_stat(self_obj, 7) / 4)) then begin
-                gsay_reply(92, 944);
+            if (get_critter_stat(self_obj, STAT_current_hp) >= (get_critter_stat(self_obj, STAT_max_hp) / 4)) then begin
+                Reply(944);
             end
             else begin
-                gsay_reply(92, 945);
+                Reply(945);
             end
         end
     end
-    gsay_option(92, 918, DialogParty, 50);
+    giq_option(1, NAME, 918, DialogParty, NEUTRAL_REACTION);
 end
 
 procedure Armor
 begin
-    if (critter_inven_obj(self_obj, 0) == 114) then begin
-        gsay_reply(92, random(948, 949));
+    if (critter_inven_obj(self_obj, INVEN_TYPE_WORN) == 114) then begin
+        Reply(random(948, 949));
     end
     else begin
-        gsay_reply(92, random(949, 950));
+        Reply(random(949, 950));
     end
-    if (critter_inven_obj(self_obj, 0) != 0) then begin
-        gsay_option(92, 951, Remove_armor, 50);
+    if (critter_inven_obj(self_obj, INVEN_TYPE_WORN) != 0) then begin
+        giq_option(1, NAME, 951, Remove_armor, NEUTRAL_REACTION);
     end
-    gsay_option(92, random(952, 955), New_armor, 50);
-    gsay_option(92, 918, DialogParty, 50);
+    giq_option(1, NAME, random(952, 955), New_armor, NEUTRAL_REACTION);
+    giq_option(1, NAME, 918, DialogParty, NEUTRAL_REACTION);
 end
 
 procedure Remove_armor
 begin
     variable LVar0 := 0;
-    if (critter_inven_obj(self_obj, 0) == 114) then begin
-        if ((has_skill(dude_obj, 14) * get_critter_stat(dude_obj, 3)) < 360) then begin
-            gsay_reply(92, random(907, 909));
-            gsay_option(92, 956, Remove_Aviator, 50);
-            gsay_option(92, 957, Node999, 50);
+    if (critter_inven_obj(self_obj, INVEN_TYPE_WORN) == 114) then begin
+        if ((dude_skill(SKILL_SPEECH) * dude_charisma) < 360) then begin
+            Reply(random(907, 909));
+            giq_option(1, NAME, 956, Remove_Aviator, NEUTRAL_REACTION);
+            giq_option(1, NAME, 957, Node999, NEUTRAL_REACTION);
         end
         else begin
-            LVar0 := critter_inven_obj(self_obj, 0);
+            LVar0 := critter_inven_obj(self_obj, INVEN_TYPE_WORN);
             rm_obj_from_inven(self_obj, LVar0);
             add_obj_to_inven(self_obj, LVar0);
-            gsay_reply(92, random(905, 906));
-            gsay_option(92, 918, DialogParty, 50);
+            Reply(random(905, 906));
+            giq_option(1, NAME, 918, DialogParty, NEUTRAL_REACTION);
         end
     end
     else begin
-        LVar0 := critter_inven_obj(self_obj, 0);
+        LVar0 := critter_inven_obj(self_obj, INVEN_TYPE_WORN);
         rm_obj_from_inven(self_obj, LVar0);
         add_obj_to_inven(self_obj, LVar0);
-        gsay_reply(92, random(905, 906));
-        gsay_option(92, 918, DialogParty, 50);
+        Reply(random(905, 906));
+        giq_option(1, NAME, 918, DialogParty, NEUTRAL_REACTION);
     end
 end
 
 procedure Remove_Aviator
 begin
     variable LVar0 := 0;
-    if ((has_skill(dude_obj, 14) * get_critter_stat(dude_obj, 3)) < 300) then begin
-        gsay_reply(92, 958);
-        gsay_option(92, 957, DialogParty, 50);
+    if ((dude_skill(SKILL_SPEECH) * dude_charisma) < 300) then begin
+        Reply(958);
+        giq_option(1, NAME, 957, DialogParty, NEUTRAL_REACTION);
     end
     else begin
-        LVar0 := critter_inven_obj(self_obj, 0);
+        LVar0 := critter_inven_obj(self_obj, INVEN_TYPE_WORN);
         rm_obj_from_inven(self_obj, LVar0);
         add_obj_to_inven(self_obj, LVar0);
-        gsay_reply(92, 959);
-        gsay_option(92, 918, DialogParty, 50);
+        Reply(959);
+        giq_option(1, NAME, 918, DialogParty, NEUTRAL_REACTION);
     end
 end
 
@@ -706,71 +706,71 @@ procedure New_armor
 begin
     variable LVar0 := 1;
     if (obj_carrying_pid_obj(self_obj, 114)) then begin
-        wield_obj_critter(self_obj, obj_carrying_pid_obj(self_obj, 114));
+        wield_obj(obj_carrying_pid_obj(self_obj, 114));
     end
     else begin
         if (obj_carrying_pid_obj(self_obj, 129)) then begin
-            wield_obj_critter(self_obj, obj_carrying_pid_obj(self_obj, 129));
+            wield_obj(obj_carrying_pid_obj(self_obj, 129));
         end
         else begin
             if (obj_carrying_pid_obj(self_obj, 128)) then begin
-                wield_obj_critter(self_obj, obj_carrying_pid_obj(self_obj, 128));
+                wield_obj(obj_carrying_pid_obj(self_obj, 128));
             end
             else begin
                 if (obj_carrying_pid_obj(self_obj, 127)) then begin
-                    wield_obj_critter(self_obj, obj_carrying_pid_obj(self_obj, 127));
+                    wield_obj(obj_carrying_pid_obj(self_obj, 127));
                 end
                 else begin
                     if (obj_carrying_pid_obj(self_obj, 126)) then begin
-                        wield_obj_critter(self_obj, obj_carrying_pid_obj(self_obj, 126));
+                        wield_obj(obj_carrying_pid_obj(self_obj, 126));
                     end
                     else begin
                         if (obj_carrying_pid_obj(self_obj, 125)) then begin
-                            wield_obj_critter(self_obj, obj_carrying_pid_obj(self_obj, 125));
+                            wield_obj(obj_carrying_pid_obj(self_obj, 125));
                         end
                         else begin
                             if (obj_carrying_pid_obj(self_obj, 122)) then begin
-                                wield_obj_critter(self_obj, obj_carrying_pid_obj(self_obj, 122));
+                                wield_obj(obj_carrying_pid_obj(self_obj, 122));
                             end
                             else begin
                                 if (obj_carrying_pid_obj(self_obj, 124)) then begin
-                                    wield_obj_critter(self_obj, obj_carrying_pid_obj(self_obj, 124));
+                                    wield_obj(obj_carrying_pid_obj(self_obj, 124));
                                 end
                                 else begin
                                     if (obj_carrying_pid_obj(self_obj, 121)) then begin
-                                        wield_obj_critter(self_obj, obj_carrying_pid_obj(self_obj, 121));
+                                        wield_obj(obj_carrying_pid_obj(self_obj, 121));
                                     end
                                     else begin
                                         if (obj_carrying_pid_obj(self_obj, 120)) then begin
-                                            wield_obj_critter(self_obj, obj_carrying_pid_obj(self_obj, 120));
+                                            wield_obj(obj_carrying_pid_obj(self_obj, 120));
                                         end
                                         else begin
                                             if (obj_carrying_pid_obj(self_obj, 119)) then begin
-                                                wield_obj_critter(self_obj, obj_carrying_pid_obj(self_obj, 119));
+                                                wield_obj(obj_carrying_pid_obj(self_obj, 119));
                                             end
                                             else begin
                                                 if (obj_carrying_pid_obj(self_obj, 118)) then begin
-                                                    wield_obj_critter(self_obj, obj_carrying_pid_obj(self_obj, 118));
+                                                    wield_obj(obj_carrying_pid_obj(self_obj, 118));
                                                 end
                                                 else begin
                                                     if (obj_carrying_pid_obj(self_obj, 117)) then begin
-                                                        wield_obj_critter(self_obj, obj_carrying_pid_obj(self_obj, 117));
+                                                        wield_obj(obj_carrying_pid_obj(self_obj, 117));
                                                     end
                                                     else begin
                                                         if (obj_carrying_pid_obj(self_obj, 116)) then begin
-                                                            wield_obj_critter(self_obj, obj_carrying_pid_obj(self_obj, 116));
+                                                            wield_obj(obj_carrying_pid_obj(self_obj, 116));
                                                         end
                                                         else begin
                                                             if (obj_carrying_pid_obj(self_obj, 115)) then begin
-                                                                wield_obj_critter(self_obj, obj_carrying_pid_obj(self_obj, 115));
+                                                                wield_obj(obj_carrying_pid_obj(self_obj, 115));
                                                             end
                                                             else begin
                                                                 if (obj_carrying_pid_obj(self_obj, 123)) then begin
-                                                                    wield_obj_critter(self_obj, obj_carrying_pid_obj(self_obj, 123));
+                                                                    wield_obj(obj_carrying_pid_obj(self_obj, 123));
                                                                 end
                                                                 else begin
                                                                     if (obj_carrying_pid_obj(self_obj, 113)) then begin
-                                                                        wield_obj_critter(self_obj, obj_carrying_pid_obj(self_obj, 113));
+                                                                        wield_obj(obj_carrying_pid_obj(self_obj, 113));
                                                                     end
                                                                     else begin
                                                                         LVar0 := 0;
@@ -792,33 +792,33 @@ begin
         end
     end
     if (LVar0 == 1) then begin
-        gsay_reply(92, random(905, 906));
+        Reply(random(905, 906));
     end
     else begin
-        gsay_reply(92, 921);
+        Reply(921);
     end
-    gsay_option(92, 918, DialogParty, 50);
+    giq_option(1, NAME, 918, DialogParty, NEUTRAL_REACTION);
 end
 
 procedure Weapon
 begin
-    gsay_reply(92, random(960, 961));
-    if ((critter_inven_obj(self_obj, 1) != 0) or (critter_inven_obj(self_obj, 2) != 0)) then begin
-        gsay_option(92, 962, Remove_Weapon, 50);
-    end
-    gsay_option(92, 963, Weapon_melee, 50);
-    gsay_option(92, 964, Weapon_throw, 50);
-    gsay_option(92, 965, Weapon_shoot, 50);
-    gsay_option(92, 966, Weapon_expl, 50);
-    gsay_option(92, 967, Weapon_best, 50);
-    gsay_option(92, 918, DialogParty, 50);
+    Reply(random(960, 961));
+    if ((critter_inven_obj(self_obj, INVEN_TYPE_RIGHT_HAND) != 0) or (critter_inven_obj(self_obj, INVEN_TYPE_LEFT_HAND) != 0)) then begin
+        giq_option(1, NAME, 962, Remove_Weapon, NEUTRAL_REACTION);
+    end
+    giq_option(1, NAME, 963, Weapon_melee, NEUTRAL_REACTION);
+    giq_option(1, NAME, 964, Weapon_throw, NEUTRAL_REACTION);
+    giq_option(1, NAME, 965, Weapon_shoot, NEUTRAL_REACTION);
+    giq_option(1, NAME, 966, Weapon_expl, NEUTRAL_REACTION);
+    giq_option(1, NAME, 967, Weapon_best, NEUTRAL_REACTION);
+    giq_option(1, NAME, 918, DialogParty, NEUTRAL_REACTION);
 end
 
 procedure Remove_Weapon
 begin
-    metarule(43, self_obj);
-    gsay_reply(92, random(905, 906));
-    gsay_option(92, 918, DialogParty, 50);
+    inven_unwield(self_obj);
+    Reply(random(905, 906));
+    giq_option(1, NAME, 918, DialogParty, NEUTRAL_REACTION);
 end
 
 procedure Weapon_melee
@@ -953,14 +953,14 @@ begin
         end
     end
     if (LVar0 != 0) then begin
-        wield_obj_critter(self_obj, LVar0);
-        gsay_reply(92, message_str(92, 968) + obj_name(LVar0) + message_str(92, 969));
+        wield_obj(LVar0);
+        Reply(mstr(968) + obj_name(LVar0) + mstr(969));
     end
     else begin
-        gsay_reply(92, 922);
-        gsay_option(92, 923, Weapon, 50);
+        Reply(922);
+        giq_option(1, NAME, 923, Weapon, NEUTRAL_REACTION);
     end
-    gsay_option(92, 918, DialogParty, 50);
+    giq_option(1, NAME, 918, DialogParty, NEUTRAL_REACTION);
 end
 
 procedure Weapon_throw
@@ -990,14 +990,14 @@ begin
         end
     end
     if (LVar0 != 0) then begin
-        wield_obj_critter(self_obj, LVar0);
-        gsay_reply(92, message_str(92, 968) + obj_name(LVar0) + message_str(92, 969));
+        wield_obj(LVar0);
+        Reply(mstr(968) + obj_name(LVar0) + mstr(969));
     end
     else begin
-        gsay_reply(92, 922);
-        gsay_option(92, 923, Weapon, 50);
+        Reply(922);
+        giq_option(1, NAME, 923, Weapon, NEUTRAL_REACTION);
     end
-    gsay_option(92, 918, DialogParty, 50);
+    giq_option(1, NAME, 918, DialogParty, NEUTRAL_REACTION);
 end
 
 procedure Weapon_shoot
@@ -1257,14 +1257,14 @@ begin
         end
     end
     if (LVar0 != 0) then begin
-        wield_obj_critter(self_obj, LVar0);
-        gsay_reply(92, message_str(92, 968) + obj_name(LVar0) + message_str(92, 969));
+        wield_obj(LVar0);
+        Reply(mstr(968) + obj_name(LVar0) + mstr(969));
     end
     else begin
-        gsay_reply(92, 922);
-        gsay_option(92, 923, Weapon, 50);
+        Reply(922);
+        giq_option(1, NAME, 923, Weapon, NEUTRAL_REACTION);
     end
-    gsay_option(92, 918, DialogParty, 50);
+    giq_option(1, NAME, 918, DialogParty, NEUTRAL_REACTION);
 end
 
 procedure Weapon_expl
@@ -1329,14 +1329,14 @@ begin
         end
     end
     if (LVar0 != 0) then begin
-        wield_obj_critter(self_obj, LVar0);
-        gsay_reply(92, message_str(92, 968) + obj_name(LVar0) + message_str(92, 969));
+        wield_obj(LVar0);
+        Reply(mstr(968) + obj_name(LVar0) + mstr(969));
     end
     else begin
-        gsay_reply(92, 922);
-        gsay_option(92, 923, Weapon, 50);
+        Reply(922);
+        giq_option(1, NAME, 923, Weapon, NEUTRAL_REACTION);
     end
-    gsay_option(92, 918, DialogParty, 50);
+    giq_option(1, NAME, 918, DialogParty, NEUTRAL_REACTION);
 end
 
 procedure Weapon_best
@@ -1551,54 +1551,54 @@ begin
         end
     end
     if (LVar0 != 0) then begin
-        wield_obj_critter(self_obj, LVar0);
-        gsay_reply(92, message_str(92, 968) + obj_name(LVar0) + message_str(92, 969));
+        wield_obj(LVar0);
+        Reply(mstr(968) + obj_name(LVar0) + mstr(969));
     end
     else begin
-        gsay_reply(92, 922);
-        gsay_option(92, 923, Weapon, 50);
+        Reply(922);
+        giq_option(1, NAME, 923, Weapon, NEUTRAL_REACTION);
     end
-    gsay_option(92, 918, DialogParty, 50);
+    giq_option(1, NAME, 918, DialogParty, NEUTRAL_REACTION);
 end
 
 procedure Distance
 begin
-    gsay_reply(92, random(970, 971));
-    gsay_option(92, 972, DistanceClose, 50);
-    gsay_option(92, 973, DistanceMedium, 50);
-    gsay_option(92, 974, DistanceFar, 50);
-    gsay_option(92, 918, DialogParty, 50);
+    Reply(random(970, 971));
+    giq_option(1, NAME, 972, DistanceClose, NEUTRAL_REACTION);
+    giq_option(1, NAME, 973, DistanceMedium, NEUTRAL_REACTION);
+    giq_option(1, NAME, 974, DistanceFar, NEUTRAL_REACTION);
+    giq_option(1, NAME, 918, DialogParty, NEUTRAL_REACTION);
 end
 
 procedure DistanceClose
 begin
     set_local_var(11, 0);
-    gsay_reply(92, random(905, 906));
-    gsay_option(92, 918, DialogParty, 50);
+    Reply(random(905, 906));
+    giq_option(1, NAME, 918, DialogParty, NEUTRAL_REACTION);
 end
 
 procedure DistanceMedium
 begin
     set_local_var(11, 1);
-    gsay_reply(92, random(905, 906));
-    gsay_option(92, 918, DialogParty, 50);
+    Reply(random(905, 906));
+    giq_option(1, NAME, 918, DialogParty, NEUTRAL_REACTION);
 end
 
 procedure DistanceFar
 begin
     set_local_var(11, 2);
-    gsay_reply(92, random(905, 906));
-    gsay_option(92, 918, DialogParty, 50);
+    Reply(random(905, 906));
+    giq_option(1, NAME, 918, DialogParty, NEUTRAL_REACTION);
 end
 
 procedure DialogParty2
 begin
-    gsay_reply(92, 920);
-    gsay_option(92, 915, AboutThisPlace, 50);
-    gsay_option(92, 916, NewIdeas, 50);
-    gsay_option(92, 917, BeforeCombat, 50);
-    gsay_option(92, 924, Waiting, 50);
-    gsay_option(92, 919, Node999, 50);
+    Reply(920);
+    giq_option(1, NAME, 915, AboutThisPlace, NEUTRAL_REACTION);
+    giq_option(1, NAME, 916, NewIdeas, NEUTRAL_REACTION);
+    giq_option(1, NAME, 917, BeforeCombat, NEUTRAL_REACTION);
+    giq_option(1, NAME, 924, Waiting, NEUTRAL_REACTION);
+    giq_option(1, NAME, 919, Node999, NEUTRAL_REACTION);
 end
 
 procedure AboutThisPlace
@@ -1616,10 +1616,9 @@ end
 procedure Waiting
 begin
     party_remove(self_obj);
-    set_global_var(265, 2);
-    gsay_reply(92, 925);
-    gsay_option(92, 926, Node999, 50);
-    gsay_option(92, 927, Node999, 50);
-    gsay_option(92, 928, Node999, 50);
+    set_global_var(GVAR_KARMA_RESERVED, 2);
+    Reply(925);
+    giq_option(1, NAME, 926, Node999, NEUTRAL_REACTION);
+    giq_option(1, NAME, 927, Node999, NEUTRAL_REACTION);
+    giq_option(1, NAME, 928, Node999, NEUTRAL_REACTION);
 end
-
