diff --git a/Mods/Olympus2207/Original/OLSPTLMN.ssl b/Mods/Olympus2207/ReDefine/OLSPTLMN.ssl
index 9786897..abc3d5c 100644
--- a/Mods/Olympus2207/Original/OLSPTLMN.ssl
+++ b/Mods/Olympus2207/ReDefine/OLSPTLMN.ssl
@@ -29,11 +29,10 @@ procedure spatial_p_proc
 begin
     if ((source_obj == dude_obj) and (local_var(0) == 0)) then begin
         set_local_var(0, 1);
-        if (global_var(370) == 0) then begin
-            set_global_var(370, 1);
+        if (global_var(GVAR_NEW_RENO_JET_SOURCE) == 0) then begin
+            set_global_var(GVAR_NEW_RENO_JET_SOURCE, 1);
             play_gmovie(6);
             load_map("olhorton", 1);
         end
     end
 end
-
