diff --git a/Mods/Olympus2207/Original/OLSPTLX.ssl b/Mods/Olympus2207/ReDefine/OLSPTLX.ssl
index e13446c..4bd5fbf 100644
--- a/Mods/Olympus2207/Original/OLSPTLX.ssl
+++ b/Mods/Olympus2207/ReDefine/OLSPTLX.ssl
@@ -27,7 +27,7 @@ end
 procedure map_update_p_proc
 begin
     if (map_var(0) == 0) then begin
-        metarule3(108, 18900, elevation(self_obj), 0);
+        metarule3(METARULE3_TILE_SET_CENTER, 18900, elevation(self_obj), 0);
         add_timer_event(self_obj, 2, 1);
         game_ui_disable;
         animate_move_obj_to_tile(dude_obj, 20100, 0 bwor 16);
@@ -38,7 +38,7 @@ procedure timed_event_p_proc
 begin
     if (fixed_param == 1) then begin
         if (camera_tile != 18900) then begin
-            metarule3(108, camera_tile, elevation(self_obj), 0);
+            metarule3(METARULE3_TILE_SET_CENTER, camera_tile, elevation(self_obj), 0);
             camera_tile := camera_tile - 400;
             add_timer_event(self_obj, 1, 1);
         end
@@ -52,4 +52,3 @@ begin
         animate_move_obj_to_tile(dude_obj, 19900, 0 bwor 16);
     end
 end
-
