diff --git a/Mods/Olympus2207/Original/OLZLAVSH.ssl b/Mods/Olympus2207/ReDefine/OLZLAVSH.ssl
index c926c05..8e23333 100644
--- a/Mods/Olympus2207/Original/OLZLAVSH.ssl
+++ b/Mods/Olympus2207/ReDefine/OLZLAVSH.ssl
@@ -37,8 +37,8 @@ begin
     if (local_var(1) == 0) then begin
         set_local_var(3, tile_num(self_obj));
         set_local_var(1, 1);
-        critter_add_trait(self_obj, 1, 6, 300);
-        critter_add_trait(self_obj, 1, 5, 15);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 300);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_GENERIC_COW);
         max_distance := random(2, 6);
     end
 end
@@ -62,13 +62,13 @@ end
 procedure description_p_proc
 begin
     script_overrides;
-    display_msg(message_str(115, random(6, 9)));
+    display_msg(message_str(SCRIPT_ZLAVACHE, random(6, 9)));
 end
 
 procedure look_at_p_proc
 begin
     script_overrides;
-    display_msg(message_str(115, random(1, 5)));
+    display_msg(message_str(SCRIPT_ZLAVACHE, random(1, 5)));
 end
 
 procedure pickup_p_proc
@@ -96,9 +96,9 @@ end
 procedure talk_p_proc
 begin
     if (not(anim_busy(self_obj))) then begin
-        anim(self_obj, 1000, rotation_to_tile(tile_num(self_obj), tile_num(dude_obj)));
+        anim(self_obj, 1000, rotation_to_tile(tile_num(self_obj), dude_tile));
     end
-    if (get_critter_stat(dude_obj, 4) <= 3) then begin
+    if (dude_is_stupid) then begin
         start_gdialog(115, self_obj, -1, -1, -1);
         gsay_start;
         call Node001();
@@ -106,7 +106,7 @@ begin
         end_dialogue;
     end
     else begin
-        float_msg(self_obj, message_str(115, random(1030, 1039)), 0);
+        float_msg(self_obj, message_str(SCRIPT_ZLAVACHE, random(1030, 1039)), FLOAT_MSG_NORMAL);
     end
 end
 
@@ -127,20 +127,20 @@ end
 procedure damage_p_proc
 begin
     if (source_obj == dude_obj) then begin
-        if ((cur_map_index >= 12) and (cur_map_index <= 17)) then begin
-            critter_add_trait(self_obj, 1, 6, 5);
+        if ((cur_map_index >= MAP_PNPASS) and (cur_map_index <= MAP_PNUNDER)) then begin
+            critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 5);
         end
-        if ((cur_map_index >= 18) and (cur_map_index <= 26)) then begin
-            critter_add_trait(self_obj, 1, 6, 7);
+        if ((cur_map_index >= MAP_SJENTER) and (cur_map_index <= MAP_SJCAVE)) then begin
+            critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 7);
         end
-        if ((cur_map_index >= 27) and (cur_map_index <= 31)) then begin
-            critter_add_trait(self_obj, 1, 6, 301);
+        if ((cur_map_index >= MAP_NWENTER) and (cur_map_index <= MAP_NWYARD)) then begin
+            critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 301);
         end
-        if ((cur_map_index >= 38) and (cur_map_index <= 43)) then begin
-            critter_add_trait(self_obj, 1, 6, 9);
+        if ((cur_map_index >= MAP_RBMAIN) and (cur_map_index <= MAP_RBFERM)) then begin
+            critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 9);
         end
-        if ((cur_map_index == 44) or (cur_map_index == 45)) then begin
-            critter_add_trait(self_obj, 1, 6, 10);
+        if ((CUR_MAP_OAENTER) or (CUR_MAP_OAMAIN)) then begin
+            critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 10);
         end
         set_local_var(2, 1);
     end
@@ -153,7 +153,7 @@ end
 
 procedure map_exit_p_proc
 begin
-    critter_add_trait(self_obj, 1, 6, 300);
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 300);
 end
 
 procedure timed_event_p_proc
@@ -171,25 +171,25 @@ end
 
 procedure Node001
 begin
-    gsay_reply(115, 10);
-    gsay_option(115, 11, Node002, 50);
-    gsay_option(115, 12, Node998, 50);
-    gsay_option(115, 13, Node999, 50);
+    gsay_reply(SCRIPT_ZLAVACHE, 10);
+    giq_option(1, SCRIPT_ZLAVACHE, 11, Node002, NEUTRAL_REACTION);
+    giq_option(1, SCRIPT_ZLAVACHE, 12, Node998, NEUTRAL_REACTION);
+    giq_option(1, SCRIPT_ZLAVACHE, 13, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node002
 begin
-    gsay_reply(115, 14);
-    gsay_option(115, 15, Node998, 50);
-    gsay_option(115, 16, Node003, 50);
-    gsay_option(115, 13, Node999, 50);
+    gsay_reply(SCRIPT_ZLAVACHE, 14);
+    giq_option(1, SCRIPT_ZLAVACHE, 15, Node998, NEUTRAL_REACTION);
+    giq_option(1, SCRIPT_ZLAVACHE, 16, Node003, NEUTRAL_REACTION);
+    giq_option(1, SCRIPT_ZLAVACHE, 13, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node003
 begin
-    gsay_reply(115, 17);
-    gsay_option(115, 19, Node998, 50);
-    gsay_option(115, 18, Node999, 50);
+    gsay_reply(SCRIPT_ZLAVACHE, 17);
+    giq_option(1, SCRIPT_ZLAVACHE, 19, Node998, NEUTRAL_REACTION);
+    giq_option(1, SCRIPT_ZLAVACHE, 18, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node998
@@ -200,4 +200,3 @@ end
 procedure Node999
 begin
 end
-
