diff --git a/Mods/Olympus2207/Original/OLZOCHK.ssl b/Mods/Olympus2207/ReDefine/OLZOCHK.ssl
index f662464..c34499b 100644
--- a/Mods/Olympus2207/Original/OLZOCHK.ssl
+++ b/Mods/Olympus2207/ReDefine/OLZOCHK.ssl
@@ -47,83 +47,83 @@ begin
     if (local_var(1) == 0) then begin
         set_local_var(3, tile_num(self_obj));
         set_local_var(1, 1);
-        set_local_var(8, has_trait(1, self_obj, 10));
+        set_local_var(8, obj_get_rot(self_obj));
         set_local_var(4, random(80, 90));
         set_local_var(5, random(0, 100));
         set_local_var(7, random(-5, 5));
-        critter_add_trait(self_obj, 1, 6, 38);
-        critter_add_trait(self_obj, 1, 5, 10);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 38);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_MAN_BERSERK);
         max_distance := random(10, 15);
         if (random(0, 100) <= 50) then begin
-            add_obj_to_inven(self_obj, create_object_sid(145, 1, 2, -1));
-            add_mult_objs_to_inven(self_obj, create_object_sid(28, 1, 2, -1), random(1, 3));
+            add_obj_to_inven(self_obj, create_object(145, 1, 2));
+            add_mult_objs_to_inven(self_obj, create_object(28, 1, 2), random(1, 3));
         end
         else begin
             if (random(0, 100) <= 50) then begin
-                add_obj_to_inven(self_obj, create_object_sid(141, 1, 2, -1));
-                add_mult_objs_to_inven(self_obj, create_object_sid(25, 1, 2, -1), random(2, 5));
+                add_obj_to_inven(self_obj, create_object(141, 1, 2));
+                add_mult_objs_to_inven(self_obj, create_object(25, 1, 2), random(2, 5));
             end
             else begin
-                add_obj_to_inven(self_obj, create_object_sid(134, 1, 2, -1));
-                add_mult_objs_to_inven(self_obj, create_object_sid(15, 1, 2, -1), random(4, 9));
+                add_obj_to_inven(self_obj, create_object(134, 1, 2));
+                add_mult_objs_to_inven(self_obj, create_object(15, 1, 2), random(4, 9));
             end
         end
         if (random(0, 100) <= 50) then begin
-            add_obj_to_inven(self_obj, create_object_sid(229, 1, 2, -1));
+            add_obj_to_inven(self_obj, create_object(229, 1, 2));
         end
         else begin
-            add_obj_to_inven(self_obj, create_object_sid(231, 1, 2, -1));
+            add_obj_to_inven(self_obj, create_object(231, 1, 2));
         end
         if (random(0, 100) <= 35) then begin
-            add_obj_to_inven(self_obj, create_object_sid(61, 1, 2, -1));
+            add_obj_to_inven(self_obj, create_object(61, 1, 2));
         end
         if (random(0, 100) <= 35) then begin
-            add_obj_to_inven(self_obj, create_object_sid(67, 1, 2, -1));
+            add_obj_to_inven(self_obj, create_object(67, 1, 2));
         end
         if (random(0, 100) <= 50) then begin
-            add_obj_to_inven(self_obj, create_object_sid(228, 1, 2, -1));
+            add_obj_to_inven(self_obj, create_object(228, 1, 2));
         end
         else begin
-            add_obj_to_inven(self_obj, create_object_sid(226, 1, 2, -1));
+            add_obj_to_inven(self_obj, create_object(226, 1, 2));
         end
         if (random(0, 100) <= 21) then begin
-            add_obj_to_inven(self_obj, create_object_sid(225, 1, 2, -1));
+            add_obj_to_inven(self_obj, create_object(225, 1, 2));
         end
         if (random(0, 100) <= 17) then begin
-            add_obj_to_inven(self_obj, create_object_sid(403, 1, 2, -1));
+            add_obj_to_inven(self_obj, create_object(403, 1, 2));
         end
         if (random(0, 100) <= 26) then begin
-            add_obj_to_inven(self_obj, create_object_sid(408, 1, 2, -1));
+            add_obj_to_inven(self_obj, create_object(408, 1, 2));
         end
         if (random(0, 100) <= 15) then begin
-            add_obj_to_inven(self_obj, create_object_sid(223, 1, 2, -1));
+            add_obj_to_inven(self_obj, create_object(223, 1, 2));
         end
         if (random(0, 100) <= 15) then begin
-            add_obj_to_inven(self_obj, create_object_sid(69, 1, 2, -1));
+            add_obj_to_inven(self_obj, create_object(69, 1, 2));
         end
         if (random(0, 100) <= 15) then begin
-            add_obj_to_inven(self_obj, create_object_sid(82, 1, 2, -1));
+            add_obj_to_inven(self_obj, create_object(82, 1, 2));
         end
         if (random(0, 100) <= 10) then begin
-            add_obj_to_inven(self_obj, create_object_sid(228, 1, 2, -1));
+            add_obj_to_inven(self_obj, create_object(228, 1, 2));
         end
         if (random(0, 100) <= 10) then begin
-            add_obj_to_inven(self_obj, create_object_sid(410, 1, 2, -1));
+            add_obj_to_inven(self_obj, create_object(410, 1, 2));
         end
         if (random(0, 100) <= 10) then begin
-            add_obj_to_inven(self_obj, create_object_sid(431, 1, 2, -1));
+            add_obj_to_inven(self_obj, create_object(431, 1, 2));
         end
         if (random(0, 100) <= 5) then begin
-            add_obj_to_inven(self_obj, create_object_sid(110, 1, 2, -1));
+            add_obj_to_inven(self_obj, create_object(110, 1, 2));
         end
         if (random(0, 100) <= 5) then begin
-            add_obj_to_inven(self_obj, create_object_sid(323, 1, 2, -1));
+            add_obj_to_inven(self_obj, create_object(323, 1, 2));
         end
         if (random(0, 100) <= 5) then begin
-            add_obj_to_inven(self_obj, create_object_sid(73, 1, 2, -1));
+            add_obj_to_inven(self_obj, create_object(73, 1, 2));
         end
         if (random(0, 100) <= 5) then begin
-            add_obj_to_inven(self_obj, create_object_sid(235, 1, 2, -1));
+            add_obj_to_inven(self_obj, create_object(235, 1, 2));
         end
     end
 end
@@ -135,13 +135,13 @@ begin
         add_timer_event(self_obj, random(59, 126), 1);
     end
     if (obj_can_see_obj(self_obj, dude_obj) == 1) then begin
-        if (not((obj_pid(critter_inven_obj(dude_obj, 0)) == 120) or (obj_pid(critter_inven_obj(dude_obj, 0)) == 121) or (obj_pid(critter_inven_obj(dude_obj, 0)) == 122) or (obj_pid(critter_inven_obj(dude_obj, 0)) == 123) or (obj_pid(critter_inven_obj(dude_obj, 0)) == 129) or (obj_pid(critter_inven_obj(dude_obj, 0)) == 391)) and not(obj_pid(critter_inven_obj(dude_obj, 0)) == 394)) then begin
+        if (not((obj_pid(critter_inven_obj(dude_obj, INVEN_TYPE_WORN)) == 120) or (obj_pid(critter_inven_obj(dude_obj, INVEN_TYPE_WORN)) == 121) or (obj_pid(critter_inven_obj(dude_obj, INVEN_TYPE_WORN)) == 122) or (obj_pid(critter_inven_obj(dude_obj, INVEN_TYPE_WORN)) == 123) or (obj_pid(critter_inven_obj(dude_obj, INVEN_TYPE_WORN)) == 129) or (obj_pid(critter_inven_obj(dude_obj, INVEN_TYPE_WORN)) == 391)) and not(obj_pid(critter_inven_obj(dude_obj, INVEN_TYPE_WORN)) == 394)) then begin
             set_local_var(2, 1);
-            float_msg(self_obj, message_str(394, random(97, 99)), 2);
+            float_msg(self_obj, message_str(SCRIPT_XDFNDPTR, random(97, 99)), FLOAT_MSG_RED);
         end
     end
     if (obj_can_see_obj(self_obj, dude_obj)) then begin
-        if ((local_var(2) == 1) or (global_var(59) <= -31)) then begin
+        if ((local_var(2) == 1) or (global_var(GVAR_TOWN_REP_VAULT_13) <= -31)) then begin
             set_local_var(2, 0);
             attack_setup(self_obj, dude_obj);
         end
@@ -156,10 +156,10 @@ procedure description_p_proc
 begin
     script_overrides;
     if (local_var(6) == 0) then begin
-        display_msg(message_str(450, 2));
+        display_mstr(2);
     end
     else begin
-        display_msg(message_str(450, 4));
+        display_mstr(4);
     end
 end
 
@@ -167,10 +167,10 @@ procedure look_at_p_proc
 begin
     script_overrides;
     if (local_var(6) == 0) then begin
-        display_msg(message_str(450, 1));
+        display_mstr(1);
     end
     else begin
-        display_msg(message_str(450, 3));
+        display_mstr(3);
     end
 end
 
@@ -188,9 +188,9 @@ end
 procedure talk_p_proc
 begin
     if (not(anim_busy(self_obj))) then begin
-        anim(self_obj, 1000, rotation_to_tile(tile_num(self_obj), tile_num(dude_obj)));
+        anim(self_obj, 1000, rotation_to_tile(tile_num(self_obj), dude_tile));
     end
-    if ((obj_pid(critter_inven_obj(dude_obj, 0)) == 120) or (obj_pid(critter_inven_obj(dude_obj, 0)) == 121) or (obj_pid(critter_inven_obj(dude_obj, 0)) == 122) or (obj_pid(critter_inven_obj(dude_obj, 0)) == 123) or (obj_pid(critter_inven_obj(dude_obj, 0)) == 129) or (obj_pid(critter_inven_obj(dude_obj, 0)) == 391) or (obj_pid(critter_inven_obj(dude_obj, 0)) == 394)) then begin
+    if ((obj_pid(critter_inven_obj(dude_obj, INVEN_TYPE_WORN)) == 120) or (obj_pid(critter_inven_obj(dude_obj, INVEN_TYPE_WORN)) == 121) or (obj_pid(critter_inven_obj(dude_obj, INVEN_TYPE_WORN)) == 122) or (obj_pid(critter_inven_obj(dude_obj, INVEN_TYPE_WORN)) == 123) or (obj_pid(critter_inven_obj(dude_obj, INVEN_TYPE_WORN)) == 129) or (obj_pid(critter_inven_obj(dude_obj, INVEN_TYPE_WORN)) == 391) or (obj_pid(critter_inven_obj(dude_obj, INVEN_TYPE_WORN)) == 394)) then begin
         start_gdialog(450, self_obj, -1, -1, -1);
         gsay_start;
         if (local_var(6) == 0) then begin
@@ -203,25 +203,25 @@ begin
         end_dialogue;
     end
     else begin
-        float_msg(self_obj, message_str(394, random(97, 99)), 2);
+        float_msg(self_obj, message_str(SCRIPT_XDFNDPTR, random(97, 99)), FLOAT_MSG_RED);
         set_local_var(2, 1);
     end
 end
 
 procedure destroy_p_proc
 begin
-    set_global_var(119, global_var(119) bwor 32768);
-    add_obj_to_inven(self_obj, create_object_sid(332, 1, 2, -1));
-    if (global_var(338) == 1) then begin
-        set_global_var(338, 0);
+    set_global_var(GVAR_MODOC_SLAUGHTER_BESS_TIME, global_var(GVAR_MODOC_SLAUGHTER_BESS_TIME) bwor 32768);
+    add_obj_to_inven(self_obj, create_object(332, 1, 2));
+    if (global_var(GVAR_CATTLE_DRIVE_CARAVAN) == 1) then begin
+        set_global_var(GVAR_CATTLE_DRIVE_CARAVAN, 0);
         party_remove(map_var(0));
     end
     megafon := tile_contains_pid_obj(25706, 0, 33557202);
-    metarule3(108, 25706, elevation(self_obj), 0);
-    float_msg(megafon, message_str(450, 111), 0);
+    metarule3(METARULE3_TILE_SET_CENTER, 25706, elevation(self_obj), 0);
+    float_msg(megafon, mstr(111), FLOAT_MSG_NORMAL);
     set_map_var(1, 1);
-    set_global_var(339, 1);
-    set_global_var(340, 1);
+    set_global_var(GVAR_MEDICINE_CARAVAN, 1);
+    set_global_var(GVAR_JET_CARAVAN, 1);
     set_light_level(30);
 end
 
@@ -233,51 +233,51 @@ begin
             set_sfall_global(13, 1);
         end
         else begin
-            if (has_trait(2, dude_obj, 13)) then begin
-                if (global_var(81) < get_pc_stat(1)) then begin
+            if (has_trait(TRAIT_TRAIT, dude_obj, TRAIT_sex_appeal)) then begin
+                if (global_var(GVAR_VAULT_CITIZENSHIP) < get_pc_stat(PCSTAT_level)) then begin
                     if (is_success(roll_vs_skill(dude_obj, 10, 0))) then begin
-                        set_global_var(98, global_var(98) + 3);
+                        set_global_var(GVAR_QUEST_FIND_THOMAS, global_var(GVAR_QUEST_FIND_THOMAS) + 3);
                     end
                     else begin
-                        set_global_var(98, global_var(98) + 1);
+                        set_global_var(GVAR_QUEST_FIND_THOMAS, global_var(GVAR_QUEST_FIND_THOMAS) + 1);
                     end
-                    if (global_var(98) >= (2 * get_pc_stat(1))) then begin
-                        set_global_var(98, 0);
-                        set_global_var(81, global_var(81) + 1);
-                        critter_mod_skill(dude_obj, 10, 1);
-                        display_msg(message_str(10, 410));
+                    if (global_var(GVAR_QUEST_FIND_THOMAS) >= (2 * get_pc_stat(PCSTAT_level))) then begin
+                        set_global_var(GVAR_QUEST_FIND_THOMAS, 0);
+                        set_global_var(GVAR_VAULT_CITIZENSHIP, global_var(GVAR_VAULT_CITIZENSHIP) + 1);
+                        critter_mod_skill(dude_obj, SKILL_STEAL, 1);
+                        display_msg(message_str(SCRIPT_STTEXT, 410));
                     end
                 end
             end
             steal_perk := 0;
             steal_gloves := 0;
-            if (has_trait(0, dude_obj, 37) != 0) then begin
+            if (has_trait(TRAIT_PERK, dude_obj, PERK_pickpocket) != 0) then begin
                 steal_perk := steal_perk + 15;
             end
-            if (has_trait(0, dude_obj, 33) != 0) then begin
+            if (has_trait(TRAIT_PERK, dude_obj, PERK_master_thief) != 0) then begin
                 steal_perk := steal_perk + 30;
             end
-            if ((obj_pid(critter_inven_obj(dude_obj, 1)) == 245) or (obj_pid(critter_inven_obj(dude_obj, 2)) == 245)) then begin
+            if ((obj_pid(critter_inven_obj(dude_obj, INVEN_TYPE_RIGHT_HAND)) == 245) or (obj_pid(critter_inven_obj(dude_obj, INVEN_TYPE_LEFT_HAND)) == 245)) then begin
                 steal_gloves := 10;
             end
-            if (random(0, 100) <= ((has_skill(dude_obj, 10) / 2) + (((3 * get_critter_stat(dude_obj, 5)) + (((get_critter_stat(dude_obj, 6) - 4) * 4) - (difficulty_level * 4)) + (2 * steal_perk) + steal_gloves) / (local_var(4) / 25)))) then begin
-                display_msg(message_str(10, random(145, 148)));
+            if (random(0, 100) <= ((dude_skill(SKILL_STEAL) / 2) + (((3 * get_critter_stat(dude_obj, STAT_ag)) + (((dude_luck - 4) * 4) - (difficulty_level * 4)) + (2 * steal_perk) + steal_gloves) / (local_var(4) / 25)))) then begin
+                display_msg(message_str(SCRIPT_STTEXT, random(145, 148)));
             end
             else begin
                 script_overrides;
-                if (random(0, 100) <= (25 - floor((((get_critter_stat(dude_obj, 6) - 4) * 4) - (difficulty_level * 4)) / 3))) then begin
-                    float_msg(self_obj, message_str(10, random(206, 209)), 2);
+                if (random(0, 100) <= (25 - floor((((dude_luck - 4) * 4) - (difficulty_level * 4)) / 3))) then begin
+                    float_msg(self_obj, message_str(SCRIPT_STTEXT, random(206, 209)), FLOAT_MSG_RED);
                     set_local_var(2, 1);
                 end
                 else begin
-                    display_msg(message_str(10, random(200, 203)));
+                    display_msg(message_str(SCRIPT_STTEXT, random(200, 203)));
                 end
             end
         end
     end
     if (action_being_used == 12) then begin
         script_overrides;
-        display_msg(message_str(10, 204) + ((has_skill(dude_obj, 10) / 2) + (((3 * get_critter_stat(dude_obj, 5)) + (((get_critter_stat(dude_obj, 6) - 4) * 4) - (difficulty_level * 4)) + (2 * steal_perk) + steal_gloves) / (local_var(4) / 25))) + message_str(10, 205));
+        display_msg(message_str(SCRIPT_STTEXT, 204) + ((dude_skill(SKILL_STEAL) / 2) + (((3 * get_critter_stat(dude_obj, STAT_ag)) + (((dude_luck - 4) * 4) - (difficulty_level * 4)) + (2 * steal_perk) + steal_gloves) / (local_var(4) / 25))) + message_str(SCRIPT_STTEXT, 205));
     end
 end
 
@@ -302,25 +302,25 @@ begin
     end
     if (fixed_param == 2) then begin
         critter_dmg(self_obj, 100, 0 bwor 256);
-        display_msg(message_str(450, 110));
-        set_global_var(119, global_var(119) bwor 32768);
-        add_obj_to_inven(self_obj, create_object_sid(332, 1, 2, -1));
-        if (global_var(338) == 1) then begin
-            set_global_var(338, 0);
+        display_mstr(110);
+        set_global_var(GVAR_MODOC_SLAUGHTER_BESS_TIME, global_var(GVAR_MODOC_SLAUGHTER_BESS_TIME) bwor 32768);
+        add_obj_to_inven(self_obj, create_object(332, 1, 2));
+        if (global_var(GVAR_CATTLE_DRIVE_CARAVAN) == 1) then begin
+            set_global_var(GVAR_CATTLE_DRIVE_CARAVAN, 0);
             party_remove(map_var(0));
         end
         megafon := tile_contains_pid_obj(25706, 0, 33557202);
-        metarule3(108, 25706, elevation(self_obj), 0);
-        float_msg(megafon, message_str(450, 111), 0);
+        metarule3(METARULE3_TILE_SET_CENTER, 25706, elevation(self_obj), 0);
+        float_msg(megafon, mstr(111), FLOAT_MSG_NORMAL);
         set_map_var(1, 1);
-        set_global_var(339, 1);
-        set_global_var(340, 1);
+        set_global_var(GVAR_MEDICINE_CARAVAN, 1);
+        set_global_var(GVAR_JET_CARAVAN, 1);
         set_light_level(30);
     end
     if (fixed_param == 3) then begin
         party_remove(map_var(0));
-        set_global_var(338, 0);
-        critter_add_trait(map_var(0), 1, 6, 666);
+        set_global_var(GVAR_CATTLE_DRIVE_CARAVAN, 0);
+        critter_add_trait(map_var(0), TRAIT_OBJECT, OBJECT_TEAM_NUM, 666);
         attack_setup(self_obj, map_var(0));
     end
 end
@@ -328,123 +328,123 @@ end
 procedure Node001
 begin
     set_local_var(6, 1);
-    if (global_var(338) == 1) then begin
-        gsay_reply(450, 11);
-        gsay_option(450, 15, Node005, 50);
-        gsay_option(450, 16, Node006, 50);
-        if ((global_var(424) == 1) and (global_var(436) == 0)) then begin
-            gsay_option(450, 23, Node011, 50);
+    if (global_var(GVAR_CATTLE_DRIVE_CARAVAN) == 1) then begin
+        Reply(11);
+        giq_option(1, NAME, 15, Node005, NEUTRAL_REACTION);
+        giq_option(1, NAME, 16, Node006, NEUTRAL_REACTION);
+        if ((global_var(GVAR_ENDGAME_MOVIE_SAN_FRAN_ELRON) == 1) and (global_var(GVAR_VAULT_LYNETTE_STATUS) == 0)) then begin
+            giq_option(1, NAME, 23, Node011, NEUTRAL_REACTION);
         end
-        if (has_skill(dude_obj, 14) >= 75) then begin
-            gsay_option(450, 17, Node008, 50);
+        if (dude_skill(SKILL_SPEECH) >= 75) then begin
+            giq_option(1, NAME, 17, Node008, NEUTRAL_REACTION);
         end
-        gsay_option(450, 19, Node999, 50);
+        giq_option(1, NAME, 19, Node999, NEUTRAL_REACTION);
     end
     else begin
-        gsay_reply(450, 10);
-        gsay_option(450, 12, Node002, 50);
-        gsay_option(450, 13, Node003, 50);
-        if ((global_var(424) == 1) and (global_var(436) == 0)) then begin
-            gsay_option(450, 23, Node011, 50);
-        end
-        gsay_option(450, 14, Node004, 50);
-        gsay_option(450, 18, Node999, 50);
+        Reply(10);
+        giq_option(1, NAME, 12, Node002, NEUTRAL_REACTION);
+        giq_option(1, NAME, 13, Node003, NEUTRAL_REACTION);
+        if ((global_var(GVAR_ENDGAME_MOVIE_SAN_FRAN_ELRON) == 1) and (global_var(GVAR_VAULT_LYNETTE_STATUS) == 0)) then begin
+            giq_option(1, NAME, 23, Node011, NEUTRAL_REACTION);
+        end
+        giq_option(1, NAME, 14, Node004, NEUTRAL_REACTION);
+        giq_option(1, NAME, 18, Node999, NEUTRAL_REACTION);
     end
 end
 
 procedure Node002
 begin
-    gsay_reply(450, 20);
-    gsay_option(450, 21, Node998, 50);
-    gsay_option(450, 22, Node999, 50);
+    Reply(20);
+    giq_option(1, NAME, 21, Node998, NEUTRAL_REACTION);
+    giq_option(1, NAME, 22, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node003
 begin
-    gsay_reply(450, 30);
-    gsay_option(450, 31, Node007, 50);
-    gsay_option(450, 32, Node002, 50);
-    gsay_option(450, 33, Node004, 50);
-    gsay_option(450, 34, Node999, 50);
+    Reply(30);
+    giq_option(1, NAME, 31, Node007, NEUTRAL_REACTION);
+    giq_option(1, NAME, 32, Node002, NEUTRAL_REACTION);
+    giq_option(1, NAME, 33, Node004, NEUTRAL_REACTION);
+    giq_option(1, NAME, 34, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node004
 begin
-    gsay_reply(450, 40);
-    gsay_option(450, 41, Node999, 50);
-    gsay_option(450, 42, Node999, 50);
+    Reply(40);
+    giq_option(1, NAME, 41, Node999, NEUTRAL_REACTION);
+    giq_option(1, NAME, 42, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node005
 begin
-    gsay_reply(450, 50);
-    gsay_option(450, 51, Node998, 50);
-    gsay_option(450, 52, Node999, 50);
+    Reply(50);
+    giq_option(1, NAME, 51, Node998, NEUTRAL_REACTION);
+    giq_option(1, NAME, 52, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node006
 begin
-    gsay_reply(450, 60);
-    gsay_option(450, 61, NodeDeath, 50);
-    gsay_option(450, 62, Node999, 50);
-    gsay_option(450, 63, Node998, 50);
+    Reply(60);
+    giq_option(1, NAME, 61, NodeDeath, NEUTRAL_REACTION);
+    giq_option(1, NAME, 62, Node999, NEUTRAL_REACTION);
+    giq_option(1, NAME, 63, Node998, NEUTRAL_REACTION);
 end
 
 procedure Node007
 begin
-    gsay_reply(450, 70);
-    gsay_option(450, 71, Node002, 50);
-    gsay_option(450, 72, Node004, 50);
-    gsay_option(450, 73, Node999, 50);
+    Reply(70);
+    giq_option(1, NAME, 71, Node002, NEUTRAL_REACTION);
+    giq_option(1, NAME, 72, Node004, NEUTRAL_REACTION);
+    giq_option(1, NAME, 73, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node008
 begin
-    gsay_reply(450, 80);
-    gsay_option(450, 81, Node009, 50);
+    Reply(80);
+    giq_option(1, NAME, 81, Node009, NEUTRAL_REACTION);
 end
 
 procedure Node009
 begin
-    gsay_reply(450, 90);
-    gsay_option(450, 91, NodeAttackChimera, 50);
+    Reply(90);
+    giq_option(1, NAME, 91, NodeAttackChimera, NEUTRAL_REACTION);
 end
 
 procedure Node010
 begin
-    if (global_var(338) == 1) then begin
-        gsay_reply(450, 11);
-        gsay_option(450, 15, Node005, 50);
-        if ((global_var(424) == 1) and (global_var(436) == 0)) then begin
-            gsay_option(450, 23, Node011, 50);
+    if (global_var(GVAR_CATTLE_DRIVE_CARAVAN) == 1) then begin
+        Reply(11);
+        giq_option(1, NAME, 15, Node005, NEUTRAL_REACTION);
+        if ((global_var(GVAR_ENDGAME_MOVIE_SAN_FRAN_ELRON) == 1) and (global_var(GVAR_VAULT_LYNETTE_STATUS) == 0)) then begin
+            giq_option(1, NAME, 23, Node011, NEUTRAL_REACTION);
         end
-        if (has_skill(dude_obj, 14) < 75) then begin
-            gsay_option(450, 16, Node006, 50);
+        if (dude_skill(SKILL_SPEECH) < 75) then begin
+            giq_option(1, NAME, 16, Node006, NEUTRAL_REACTION);
         end
         else begin
-            gsay_option(450, 17, Node008, 50);
+            giq_option(1, NAME, 17, Node008, NEUTRAL_REACTION);
         end
-        gsay_option(450, 19, Node999, 50);
+        giq_option(1, NAME, 19, Node999, NEUTRAL_REACTION);
     end
     else begin
-        gsay_reply(450, 10);
-        gsay_option(450, 12, Node002, 50);
-        if ((global_var(424) == 1) and (global_var(436) == 0)) then begin
-            gsay_option(450, 23, Node011, 50);
-        end
-        gsay_option(450, 13, Node003, 50);
-        gsay_option(450, 14, Node004, 50);
-        gsay_option(450, 18, Node999, 50);
+        Reply(10);
+        giq_option(1, NAME, 12, Node002, NEUTRAL_REACTION);
+        if ((global_var(GVAR_ENDGAME_MOVIE_SAN_FRAN_ELRON) == 1) and (global_var(GVAR_VAULT_LYNETTE_STATUS) == 0)) then begin
+            giq_option(1, NAME, 23, Node011, NEUTRAL_REACTION);
+        end
+        giq_option(1, NAME, 13, Node003, NEUTRAL_REACTION);
+        giq_option(1, NAME, 14, Node004, NEUTRAL_REACTION);
+        giq_option(1, NAME, 18, Node999, NEUTRAL_REACTION);
     end
 end
 
 procedure Node011
 begin
-    if (global_var(436) == 0) then begin
-        set_global_var(436, 1);
+    if (global_var(GVAR_VAULT_LYNETTE_STATUS) == 0) then begin
+        set_global_var(GVAR_VAULT_LYNETTE_STATUS, 1);
     end
-    gsay_reply(450, 24);
-    gsay_option(450, 25, Node999, 50);
+    Reply(24);
+    giq_option(1, NAME, 25, Node999, NEUTRAL_REACTION);
 end
 
 procedure NodeBarter
@@ -465,7 +465,7 @@ end
 
 procedure NodeDeath
 begin
-    if (get_critter_stat(dude_obj, 0) >= 9) then begin
+    if (dude_strength >= 9) then begin
         add_timer_event(self_obj, 8, 2);
     end
     else begin
@@ -477,4 +477,3 @@ procedure NodeAttackChimera
 begin
     add_timer_event(self_obj, 8, 3);
 end
-
