diff --git a/Mods/Olympus2207/Original/OLZSAFE.ssl b/Mods/Olympus2207/ReDefine/OLZSAFE.ssl
index cf0ef7a..caa3b7f 100644
--- a/Mods/Olympus2207/Original/OLZSAFE.ssl
+++ b/Mods/Olympus2207/ReDefine/OLZSAFE.ssl
@@ -40,16 +40,16 @@ end
 procedure description_p_proc
 begin
     script_overrides;
-    display_msg(message_str(458, 6));
-    if (has_skill(dude_obj, 12) >= 50) then begin
-        display_msg(message_str(458, 7));
+    display_msg(message_str(SCRIPT_OLZSOLPN, 6));
+    if (dude_skill(SKILL_SCIENCE) >= 50) then begin
+        display_msg(message_str(SCRIPT_OLZSOLPN, 7));
     end
 end
 
 procedure look_at_p_proc
 begin
     script_overrides;
-    display_msg(message_str(458, 5));
+    display_msg(message_str(SCRIPT_OLZSOLPN, 5));
 end
 
 procedure damage_p_proc
@@ -522,11 +522,11 @@ begin
         play_sfx("IACCUXX1");
         set_local_var(4, 1);
         call go_exit();
-        display_msg(message_str(458, 40));
+        display_msg(message_str(SCRIPT_OLZSOLPN, 40));
     end
     else begin
         play_sfx("IISXXXX1");
-        display_msg(message_str(458, 26));
+        display_msg(message_str(SCRIPT_OLZSOLPN, 26));
     end
     if (button_is_down == 1) then begin
         button_is_down := 0;
@@ -539,4 +539,3 @@ begin
     deleteWin("panel");
     game_ui_enable;
 end
-
