diff --git a/Mods/Olympus2207/Original/OLZTURSC.ssl b/Mods/Olympus2207/ReDefine/OLZTURSC.ssl
index a81d963..1d53f5f 100644
--- a/Mods/Olympus2207/Original/OLZTURSC.ssl
+++ b/Mods/Olympus2207/ReDefine/OLZTURSC.ssl
@@ -27,12 +27,12 @@ begin
     if (local_var(1) == 0) then begin
         set_local_var(3, tile_num(self_obj));
         set_local_var(1, 1);
-        set_local_var(7, has_trait(1, self_obj, 10));
+        set_local_var(7, obj_get_rot(self_obj));
         set_local_var(4, random(70, 80));
         set_local_var(5, random(0, 100));
         set_local_var(6, random(-5, 5));
-        critter_add_trait(self_obj, 1, 6, 38);
-        critter_add_trait(self_obj, 1, 5, 71);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 38);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_TURRET);
     end
 end
 
@@ -72,7 +72,7 @@ begin
     script_overrides;
     if (source_obj == dude_obj) then begin
         if (map_var(0) == 1) then begin
-            critter_add_trait(self_obj, 1, 6, 0);
+            critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 0);
             attack_setup(self_obj, tile_contains_pid_obj(17914, elevation(self_obj), 16777425));
         end
         else begin
@@ -100,7 +100,7 @@ end
 procedure map_enter_p_proc
 begin
     anim(self_obj, 1000, local_var(7));
-    critter_add_trait(self_obj, 1, 6, 38);
+    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 38);
 end
 
 procedure timed_event_p_proc
@@ -109,4 +109,3 @@ begin
         animation_is_called := 0;
     end
 end
-
