diff --git a/Mods/Olympus2207/Original/PNMARTA.ssl b/Mods/Olympus2207/ReDefine/PNMARTA.ssl
index 0bd215a..ea30cef 100644
--- a/Mods/Olympus2207/Original/PNMARTA.ssl
+++ b/Mods/Olympus2207/ReDefine/PNMARTA.ssl
@@ -53,61 +53,61 @@ begin
         set_local_var(4, random(50, 60));
         set_local_var(5, random(40, 50));
         set_local_var(6, random(-30, 20));
-        critter_add_trait(self_obj, 1, 6, 5);
-        critter_add_trait(self_obj, 1, 5, 11);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 5);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_WOMAN_WEAK);
         if (random(0, 100) <= 20) then begin
-            add_mult_objs_to_inven(self_obj, create_object_sid(41, 1, 2, -1), random(1, 10));
+            add_mult_objs_to_inven(self_obj, create_object(41, 1, 2), random(1, 10));
         end
         if (random(0, 100) <= 15) then begin
-            add_mult_objs_to_inven(self_obj, create_object_sid(220, 1, 2, -1), random(1, 2));
+            add_mult_objs_to_inven(self_obj, create_object(220, 1, 2), random(1, 2));
         end
         if (random(0, 100) <= 15) then begin
-            add_mult_objs_to_inven(self_obj, create_object_sid(217, 1, 2, -1), random(1, 2));
+            add_mult_objs_to_inven(self_obj, create_object(217, 1, 2), random(1, 2));
         end
         if (random(0, 100) <= 15) then begin
-            add_obj_to_inven(self_obj, create_object_sid(36, 1, 2, -1));
+            add_obj_to_inven(self_obj, create_object(36, 1, 2));
         end
         if (random(0, 100) <= 14) then begin
-            add_obj_to_inven(self_obj, create_object_sid(65, 1, 2, -1));
+            add_obj_to_inven(self_obj, create_object(65, 1, 2));
         end
         if (random(0, 100) <= 13) then begin
-            add_obj_to_inven(self_obj, create_object_sid(63, 1, 2, -1));
+            add_obj_to_inven(self_obj, create_object(63, 1, 2));
         end
         if (random(0, 100) <= 12) then begin
-            add_obj_to_inven(self_obj, create_object_sid(71, 1, 2, -1));
+            add_obj_to_inven(self_obj, create_object(71, 1, 2));
         end
         if (random(0, 100) <= 11) then begin
-            add_obj_to_inven(self_obj, create_object_sid(111, 1, 2, -1));
+            add_obj_to_inven(self_obj, create_object(111, 1, 2));
         end
         if (random(0, 100) <= 10) then begin
-            add_obj_to_inven(self_obj, create_object_sid(415, 1, 2, -1));
+            add_obj_to_inven(self_obj, create_object(415, 1, 2));
         end
         if (random(0, 100) <= 10) then begin
-            add_obj_to_inven(self_obj, create_object_sid(420, 1, 2, -1));
+            add_obj_to_inven(self_obj, create_object(420, 1, 2));
         end
         if (random(0, 100) <= 9) then begin
-            add_obj_to_inven(self_obj, create_object_sid(243, 1, 2, -1));
+            add_obj_to_inven(self_obj, create_object(243, 1, 2));
         end
         if (random(0, 100) <= 8) then begin
-            add_obj_to_inven(self_obj, create_object_sid(423, 1, 2, -1));
+            add_obj_to_inven(self_obj, create_object(423, 1, 2));
         end
         if (random(0, 100) <= 7) then begin
-            add_obj_to_inven(self_obj, create_object_sid(426, 1, 2, -1));
+            add_obj_to_inven(self_obj, create_object(426, 1, 2));
         end
         if (random(0, 100) <= 6) then begin
-            add_obj_to_inven(self_obj, create_object_sid(110, 1, 2, -1));
+            add_obj_to_inven(self_obj, create_object(110, 1, 2));
         end
         if (random(0, 100) <= 5) then begin
-            add_obj_to_inven(self_obj, create_object_sid(421, 1, 2, -1));
+            add_obj_to_inven(self_obj, create_object(421, 1, 2));
         end
         if (random(0, 100) <= 4) then begin
-            add_obj_to_inven(self_obj, create_object_sid(427, 1, 2, -1));
+            add_obj_to_inven(self_obj, create_object(427, 1, 2));
         end
         if (random(0, 100) <= 3) then begin
-            add_obj_to_inven(self_obj, create_object_sid(39, 1, 2, -1));
+            add_obj_to_inven(self_obj, create_object(39, 1, 2));
         end
         if (random(0, 100) <= 2) then begin
-            add_obj_to_inven(self_obj, create_object_sid(75, 1, 2, -1));
+            add_obj_to_inven(self_obj, create_object(75, 1, 2));
         end
     end
 end
@@ -116,14 +116,14 @@ procedure critter_p_proc
 begin
     if ((animation_is_called == 0) and (anim_busy(self_obj) == 0)) then begin
         animation_is_called := 1;
-        add_timer_event(self_obj, random(1000, 1000), 1);
+        add_timer_event(self_obj, 1000, 1);
     end
     if (float_is_called == 0) then begin
         float_is_called := 1;
-        add_timer_event(self_obj, random(1000, 1000), 2);
+        add_timer_event(self_obj, 1000, 2);
     end
     if (obj_can_see_obj(self_obj, dude_obj)) then begin
-        if ((local_var(2) == 1) or (global_var(49) <= -31)) then begin
+        if ((local_var(2) == 1) or (global_var(GVAR_TOWN_REP_THE_DEN) <= -31)) then begin
             set_local_var(2, 0);
             attack_setup(self_obj, dude_obj);
         end
@@ -138,10 +138,10 @@ procedure description_p_proc
 begin
     script_overrides;
     if (local_var(7) == 0) then begin
-        display_msg(message_str(130, 2));
+        display_mstr(2);
     end
     else begin
-        display_msg(message_str(130, 4));
+        display_mstr(4);
     end
 end
 
@@ -149,10 +149,10 @@ procedure look_at_p_proc
 begin
     script_overrides;
     if (local_var(7) == 0) then begin
-        display_msg(message_str(130, 1));
+        display_mstr(1);
     end
     else begin
-        display_msg(message_str(130, 3));
+        display_mstr(3);
     end
 end
 
@@ -169,19 +169,19 @@ end
 
 procedure talk_p_proc
 begin
-    if ((local_var(9) == 1) and (global_var(130) != 1)) then begin
-        display_msg(message_str(130, 500));
+    if ((local_var(9) == 1) and (global_var(GVAR_MODOC_SPOKE_PROTECTOR) != 1)) then begin
+        display_mstr(500);
     end
     else begin
         if (local_var(9) == 2) then begin
             if (not(anim_busy(self_obj))) then begin
-                anim(self_obj, 1000, rotation_to_tile(tile_num(self_obj), tile_num(dude_obj)));
+                anim(self_obj, 1000, rotation_to_tile(tile_num(self_obj), dude_tile));
             end
-            float_msg(self_obj, message_str(130, random(501, 505)), 0);
+            float_msg(self_obj, mstr(random(501, 505)), FLOAT_MSG_NORMAL);
         end
         else begin
             if (not(anim_busy(self_obj))) then begin
-                anim(self_obj, 1000, rotation_to_tile(tile_num(self_obj), tile_num(dude_obj)));
+                anim(self_obj, 1000, rotation_to_tile(tile_num(self_obj), dude_tile));
             end
             start_gdialog(130, self_obj, -1, -1, -1);
             gsay_start;
@@ -200,15 +200,15 @@ end
 procedure destroy_p_proc
 begin
     if (source_obj == dude_obj) then begin
-        set_global_var(0, global_var(0) - 1);
+        set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) - 1);
         if ((49 == 50) or (49 == 59)) then begin
-            set_global_var(49, global_var(49) - 3);
+            set_global_var(GVAR_TOWN_REP_THE_DEN, global_var(GVAR_TOWN_REP_THE_DEN) - 3);
         end
         else begin
-            set_global_var(49, global_var(49) - 6);
+            set_global_var(GVAR_TOWN_REP_THE_DEN, global_var(GVAR_TOWN_REP_THE_DEN) - 6);
         end
     end
-    set_global_var(119, global_var(119) bwor 1048576);
+    set_global_var(GVAR_MODOC_SLAUGHTER_BESS_TIME, global_var(GVAR_MODOC_SLAUGHTER_BESS_TIME) bwor 1048576);
 end
 
 procedure use_skill_on_p_proc
@@ -219,51 +219,51 @@ begin
             set_sfall_global(13, 1);
         end
         else begin
-            if (has_trait(2, dude_obj, 13)) then begin
-                if (global_var(81) < get_pc_stat(1)) then begin
+            if (has_trait(TRAIT_TRAIT, dude_obj, TRAIT_sex_appeal)) then begin
+                if (global_var(GVAR_VAULT_CITIZENSHIP) < get_pc_stat(PCSTAT_level)) then begin
                     if (is_success(roll_vs_skill(dude_obj, 10, 0))) then begin
-                        set_global_var(98, global_var(98) + 3);
+                        set_global_var(GVAR_QUEST_FIND_THOMAS, global_var(GVAR_QUEST_FIND_THOMAS) + 3);
                     end
                     else begin
-                        set_global_var(98, global_var(98) + 1);
+                        set_global_var(GVAR_QUEST_FIND_THOMAS, global_var(GVAR_QUEST_FIND_THOMAS) + 1);
                     end
-                    if (global_var(98) >= (2 * get_pc_stat(1))) then begin
-                        set_global_var(98, 0);
-                        set_global_var(81, global_var(81) + 1);
-                        critter_mod_skill(dude_obj, 10, 1);
-                        display_msg(message_str(10, 410));
+                    if (global_var(GVAR_QUEST_FIND_THOMAS) >= (2 * get_pc_stat(PCSTAT_level))) then begin
+                        set_global_var(GVAR_QUEST_FIND_THOMAS, 0);
+                        set_global_var(GVAR_VAULT_CITIZENSHIP, global_var(GVAR_VAULT_CITIZENSHIP) + 1);
+                        critter_mod_skill(dude_obj, SKILL_STEAL, 1);
+                        display_msg(message_str(SCRIPT_STTEXT, 410));
                     end
                 end
             end
             steal_perk := 0;
             steal_gloves := 0;
-            if (has_trait(0, dude_obj, 37) != 0) then begin
+            if (has_trait(TRAIT_PERK, dude_obj, PERK_pickpocket) != 0) then begin
                 steal_perk := steal_perk + 15;
             end
-            if (has_trait(0, dude_obj, 33) != 0) then begin
+            if (has_trait(TRAIT_PERK, dude_obj, PERK_master_thief) != 0) then begin
                 steal_perk := steal_perk + 30;
             end
-            if ((obj_pid(critter_inven_obj(dude_obj, 1)) == 245) or (obj_pid(critter_inven_obj(dude_obj, 2)) == 245)) then begin
+            if ((obj_pid(critter_inven_obj(dude_obj, INVEN_TYPE_RIGHT_HAND)) == 245) or (obj_pid(critter_inven_obj(dude_obj, INVEN_TYPE_LEFT_HAND)) == 245)) then begin
                 steal_gloves := 10;
             end
-            if (random(0, 100) <= ((has_skill(dude_obj, 10) / 2) + (((3 * get_critter_stat(dude_obj, 5)) + (((get_critter_stat(dude_obj, 6) - 4) * 4) - (difficulty_level * 4)) + (2 * steal_perk) + steal_gloves) / (local_var(4) / 25)))) then begin
-                display_msg(message_str(10, random(145, 148)));
+            if (random(0, 100) <= ((dude_skill(SKILL_STEAL) / 2) + (((3 * get_critter_stat(dude_obj, STAT_ag)) + (((dude_luck - 4) * 4) - (difficulty_level * 4)) + (2 * steal_perk) + steal_gloves) / (local_var(4) / 25)))) then begin
+                display_msg(message_str(SCRIPT_STTEXT, random(145, 148)));
             end
             else begin
                 script_overrides;
-                if (random(0, 100) <= (25 - floor((((get_critter_stat(dude_obj, 6) - 4) * 4) - (difficulty_level * 4)) / 3))) then begin
-                    float_msg(self_obj, message_str(10, random(206, 209)), 2);
+                if (random(0, 100) <= (25 - floor((((dude_luck - 4) * 4) - (difficulty_level * 4)) / 3))) then begin
+                    float_msg(self_obj, message_str(SCRIPT_STTEXT, random(206, 209)), FLOAT_MSG_RED);
                     set_local_var(2, 1);
                 end
                 else begin
-                    display_msg(message_str(10, random(200, 203)));
+                    display_msg(message_str(SCRIPT_STTEXT, random(200, 203)));
                 end
             end
         end
     end
     if (action_being_used == 12) then begin
         script_overrides;
-        display_msg(message_str(10, 204) + ((has_skill(dude_obj, 10) / 2) + (((3 * get_critter_stat(dude_obj, 5)) + (((get_critter_stat(dude_obj, 6) - 4) * 4) - (difficulty_level * 4)) + (2 * steal_perk) + steal_gloves) / (local_var(4) / 25))) + message_str(10, 205));
+        display_msg(message_str(SCRIPT_STTEXT, 204) + ((dude_skill(SKILL_STEAL) / 2) + (((3 * get_critter_stat(dude_obj, STAT_ag)) + (((dude_luck - 4) * 4) - (difficulty_level * 4)) + (2 * steal_perk) + steal_gloves) / (local_var(4) / 25))) + message_str(SCRIPT_STTEXT, 205));
     end
 end
 
@@ -280,7 +280,7 @@ end
 
 procedure map_enter_p_proc
 begin
-    if (not(metarule(22, 0)) == 1) then begin
+    if (not(is_loading_game) == 1) then begin
         if (local_var(9) == 1) then begin
             set_local_var(9, 2);
             if ((art_anim(obj_art_fid(self_obj)) == 48) or (art_anim(obj_art_fid(self_obj)) == 49)) then begin
@@ -312,152 +312,152 @@ end
 
 procedure Node001
 begin
-    gsay_reply(130, 10);
-    gsay_option(130, 11, Node002, 50);
-    gsay_option(130, message_str(130, 12) + obj_name(dude_obj) + message_str(130, 13), Node003, 50);
-    gsay_option(130, 14, Node999, 50);
+    Reply(10);
+    giq_option(1, NAME, 11, Node002, NEUTRAL_REACTION);
+    giq_option(1, NAME, mstr(12) + dude_name + mstr(13), Node003, NEUTRAL_REACTION);
+    giq_option(1, NAME, 14, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node002
 begin
     set_local_var(7, 1);
     already_talk := 1;
-    gsay_reply(130, 20);
-    gsay_option(130, 21, Node020, 50);
-    gsay_option(130, 22, Node010, 50);
-    gsay_option(130, 23, Node999, 50);
+    Reply(20);
+    giq_option(1, NAME, 21, Node020, NEUTRAL_REACTION);
+    giq_option(1, NAME, 22, Node010, NEUTRAL_REACTION);
+    giq_option(1, NAME, 23, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node003
 begin
     set_local_var(7, 1);
     already_talk := 1;
-    gsay_reply(130, message_str(130, 30) + obj_name(dude_obj) + message_str(130, 31));
-    gsay_option(130, 32, NodeQA, 50);
-    gsay_option(130, 33, Node999, 50);
+    Reply(mstr(30) + dude_name + mstr(31));
+    giq_option(1, NAME, 32, NodeQA, NEUTRAL_REACTION);
+    giq_option(1, NAME, 33, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node010
 begin
     if (already_talk == 0) then begin
         already_talk := 1;
-        gsay_reply(130, 100);
+        Reply(100);
     end
     else begin
-        gsay_reply(130, 101);
+        Reply(101);
     end
-    gsay_option(130, 102, Node011, 50);
-    if (global_var(129) == 0) then begin
-        gsay_option(130, 103, Node020, 50);
+    giq_option(1, NAME, 102, Node011, NEUTRAL_REACTION);
+    if (global_var(GVAR_MODOC_JONNY_HOME) == 0) then begin
+        giq_option(1, NAME, 103, Node020, NEUTRAL_REACTION);
     end
-    if ((global_var(129) > 0) and (local_var(8) == 0)) then begin
-        gsay_option(130, 104, Node030, 50);
+    if ((global_var(GVAR_MODOC_JONNY_HOME) > 0) and (local_var(8) == 0)) then begin
+        giq_option(1, NAME, 104, Node030, NEUTRAL_REACTION);
     end
-    if (global_var(130) == 1) then begin
-        gsay_option(130, 106, Node040, 50);
+    if (global_var(GVAR_MODOC_SPOKE_PROTECTOR) == 1) then begin
+        giq_option(1, NAME, 106, Node040, NEUTRAL_REACTION);
     end
-    gsay_option(130, 105, Node999, 50);
+    giq_option(1, NAME, 105, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node011
 begin
-    if (55 < ((get_critter_stat(dude_obj, 3) * 10) + (has_skill(dude_obj, 14) / 5))) then begin
-        gsay_reply(130, 110);
-        gsay_option(130, 111, Node012, 50);
-        gsay_option(130, 112, Node010, 50);
+    if (55 < ((dude_charisma * 10) + (dude_skill(SKILL_SPEECH) / 5))) then begin
+        Reply(110);
+        giq_option(1, NAME, 111, Node012, NEUTRAL_REACTION);
+        giq_option(1, NAME, 112, Node010, NEUTRAL_REACTION);
     end
     else begin
-        gsay_reply(130, 113);
-        gsay_option(130, 114, Node010, 50);
-        gsay_option(130, 115, Node999, 50);
+        Reply(113);
+        giq_option(1, NAME, 114, Node010, NEUTRAL_REACTION);
+        giq_option(1, NAME, 115, Node999, NEUTRAL_REACTION);
     end
 end
 
 procedure Node012
 begin
-    gsay_reply(130, 120);
-    gsay_option(130, 121, Node010, 50);
-    gsay_option(130, 122, Node010, 50);
+    Reply(120);
+    giq_option(1, NAME, 121, Node010, NEUTRAL_REACTION);
+    giq_option(1, NAME, 122, Node010, NEUTRAL_REACTION);
 end
 
 procedure Node020
 begin
-    gsay_reply(130, 200);
-    gsay_option(130, 201, NodeQA, 50);
-    gsay_option(130, 202, Node999, 50);
+    Reply(200);
+    giq_option(1, NAME, 201, NodeQA, NEUTRAL_REACTION);
+    giq_option(1, NAME, 202, Node999, NEUTRAL_REACTION);
 end
 
 procedure NodeQA
 begin
-    set_global_var(129, 1);
+    set_global_var(GVAR_MODOC_JONNY_HOME, 1);
 end
 
 procedure Node030
 begin
     if (map_var(0) == 4) then begin
         set_local_var(8, 1);
-        gsay_reply(130, 300);
-        gsay_option(130, 301, Node999, 50);
+        Reply(300);
+        giq_option(1, NAME, 301, Node999, NEUTRAL_REACTION);
     end
     else begin
-        gsay_reply(130, 302);
+        Reply(302);
         if (map_var(0) == 3) then begin
-            gsay_option(130, 303, Node031, 50);
+            giq_option(1, NAME, 303, Node031, NEUTRAL_REACTION);
         end
         if (map_var(0) == 2) then begin
-            gsay_option(130, 304, Node032, 50);
+            giq_option(1, NAME, 304, Node032, NEUTRAL_REACTION);
         end
         if (map_var(0) == 1) then begin
-            gsay_option(130, 305, Node033, 50);
+            giq_option(1, NAME, 305, Node033, NEUTRAL_REACTION);
         end
-        gsay_option(130, 306, Node999, 50);
+        giq_option(1, NAME, 306, Node999, NEUTRAL_REACTION);
     end
 end
 
 procedure Node031
 begin
-    add_obj_to_inven(dude_obj, create_object_sid(422, 1, 2, -1));
+    add_obj_to_inven(dude_obj, create_object(422, 1, 2));
     set_local_var(8, 1);
-    gsay_reply(130, 310);
-    gsay_option(130, 311, Node999, 50);
+    Reply(310);
+    giq_option(1, NAME, 311, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node032
 begin
-    add_obj_to_inven(dude_obj, create_object_sid(422, 1, 2, -1));
+    add_obj_to_inven(dude_obj, create_object(422, 1, 2));
     set_local_var(8, 1);
-    gsay_reply(130, 320);
-    gsay_option(130, 321, Node999, 50);
+    Reply(320);
+    giq_option(1, NAME, 321, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node033
 begin
     set_local_var(8, 1);
-    gsay_reply(130, 330);
-    gsay_option(130, 331, Node034, 50);
-    if ((has_skill(dude_obj, 16) >= 30) or (global_var(285) == 2)) then begin
-        gsay_option(130, 332, Node035, 50);
+    Reply(330);
+    giq_option(1, NAME, 331, Node034, NEUTRAL_REACTION);
+    if ((dude_skill(SKILL_GAMBLING) >= 30) or (global_var(GVAR_NEW_RENO_WRIGHT_FLAGS) == 2)) then begin
+        giq_option(1, NAME, 332, Node035, NEUTRAL_REACTION);
     end
-    gsay_option(130, 333, Node999, 50);
+    giq_option(1, NAME, 333, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node034
 begin
-    add_obj_to_inven(dude_obj, create_object_sid(422, 1, 2, -1));
-    gsay_reply(130, 340);
-    gsay_option(130, 341, Node999, 50);
+    add_obj_to_inven(dude_obj, create_object(422, 1, 2));
+    Reply(340);
+    giq_option(1, NAME, 341, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node035
 begin
-    gsay_reply(130, 350);
-    gsay_option(130, 351, Node036, 50);
-    gsay_option(130, 352, Node999, 50);
+    Reply(350);
+    giq_option(1, NAME, 351, Node036, NEUTRAL_REACTION);
+    giq_option(1, NAME, 352, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node036
 begin
-    set_global_var(282, global_var(282) + 1);
+    set_global_var(GVAR_CARAVAN_MASTERS_TOTAL, global_var(GVAR_CARAVAN_MASTERS_TOTAL) + 1);
     fadeOut(700);
     fadeIn(700);
     set_local_var(9, 1);
@@ -468,14 +468,14 @@ begin
     reg_anim_animate(self_obj, 48, -1);
     reg_anim_func(3, 0);
     critter_injure(self_obj, 2);
-    gsay_reply(130, 360);
-    gsay_option(130, 361, Node999, 50);
+    Reply(360);
+    giq_option(1, NAME, 361, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node040
 begin
-    gsay_reply(130, 400);
-    gsay_option(130, 401, Node999, 50);
+    Reply(400);
+    giq_option(1, NAME, 401, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node999
@@ -494,4 +494,3 @@ begin
     gdialog_mod_barter(-(dude_objective_reputation + local_var(6)));
     call Node010();
 end
-
