diff --git a/Mods/Olympus2207/Original/PNPEOPLE.ssl b/Mods/Olympus2207/ReDefine/PNPEOPLE.ssl
index feaaf3a..e579393 100644
--- a/Mods/Olympus2207/Original/PNPEOPLE.ssl
+++ b/Mods/Olympus2207/ReDefine/PNPEOPLE.ssl
@@ -31,17 +31,17 @@ begin
         set_local_var(1, 1);
         set_local_var(4, random(0, 100));
         set_local_var(5, random(0, 100));
-        critter_add_trait(self_obj, 1, 6, 5);
-        critter_add_trait(self_obj, 1, 5, 19);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 5);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_UNDEAD_NORMAL);
     end
 end
 
 procedure critter_p_proc
 begin
-    if ((global_var(126) == 13) and (elevation(self_obj) != 0)) then begin
+    if ((global_var(GVAR_HOSTILE_SLAVE_COUNT) == 13) and (elevation(self_obj) != 0)) then begin
         move_to(self_obj, tile_num(self_obj), 0);
     end
-    if ((global_var(126) != 13) and (elevation(self_obj) == 0)) then begin
+    if ((global_var(GVAR_HOSTILE_SLAVE_COUNT) != 13) and (elevation(self_obj) == 0)) then begin
         move_to(self_obj, tile_num(self_obj), 1);
     end
 end
@@ -84,10 +84,10 @@ end
 
 procedure map_enter_p_proc
 begin
-    if ((global_var(126) == 13) and (elevation(self_obj) != 0)) then begin
+    if ((global_var(GVAR_HOSTILE_SLAVE_COUNT) == 13) and (elevation(self_obj) != 0)) then begin
         move_to(self_obj, tile_num(self_obj), 0);
     end
-    if ((global_var(126) != 13) and (elevation(self_obj) == 0)) then begin
+    if ((global_var(GVAR_HOSTILE_SLAVE_COUNT) != 13) and (elevation(self_obj) == 0)) then begin
         move_to(self_obj, tile_num(self_obj), 1);
     end
 end
@@ -95,4 +95,3 @@ end
 procedure timed_event_p_proc
 begin
 end
-
