diff --git a/Mods/Olympus2207/Original/PNTTGARD.ssl b/Mods/Olympus2207/ReDefine/PNTTGARD.ssl
index f1f3d60..4ad77a1 100644
--- a/Mods/Olympus2207/Original/PNTTGARD.ssl
+++ b/Mods/Olympus2207/ReDefine/PNTTGARD.ssl
@@ -34,48 +34,48 @@ begin
         set_local_var(4, random(85, 95));
         set_local_var(5, random(10, 20));
         set_local_var(6, random(-30, 10));
-        critter_add_trait(self_obj, 1, 6, 3);
-        critter_add_trait(self_obj, 1, 5, 5);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 3);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_OLYMPUS_GUARD);
         if (random(0, 100) <= 50) then begin
-            add_obj_to_inven(self_obj, create_object_sid(185, 1, 2, -1));
-            add_mult_objs_to_inven(self_obj, create_object_sid(26, 1, 2, -1), random(1, 3));
-            wield_obj_critter(self_obj, obj_carrying_pid_obj(self_obj, 185));
+            add_obj_to_inven(self_obj, create_object(185, 1, 2));
+            add_mult_objs_to_inven(self_obj, create_object(26, 1, 2), random(1, 3));
+            wield_obj(obj_carrying_pid_obj(self_obj, 185));
         end
         else begin
             if (random(0, 100) <= 50) then begin
-                add_obj_to_inven(self_obj, create_object_sid(184, 1, 2, -1));
-                add_mult_objs_to_inven(self_obj, create_object_sid(17, 1, 2, -1), random(4, 8));
-                wield_obj_critter(self_obj, obj_carrying_pid_obj(self_obj, 184));
+                add_obj_to_inven(self_obj, create_object(184, 1, 2));
+                add_mult_objs_to_inven(self_obj, create_object(17, 1, 2), random(4, 8));
+                wield_obj(obj_carrying_pid_obj(self_obj, 184));
             end
             else begin
-                add_obj_to_inven(self_obj, create_object_sid(171, 1, 2, -1));
-                add_mult_objs_to_inven(self_obj, create_object_sid(4, 1, 2, -1), random(2, 5));
-                wield_obj_critter(self_obj, obj_carrying_pid_obj(self_obj, 171));
+                add_obj_to_inven(self_obj, create_object(171, 1, 2));
+                add_mult_objs_to_inven(self_obj, create_object(4, 1, 2), random(2, 5));
+                wield_obj(obj_carrying_pid_obj(self_obj, 171));
             end
         end
         if (random(0, 100) <= 33) then begin
-            add_obj_to_inven(self_obj, create_object_sid(221, 1, 2, -1));
+            add_obj_to_inven(self_obj, create_object(221, 1, 2));
         end
         else begin
-            add_obj_to_inven(self_obj, create_object_sid(112, 1, 2, -1));
+            add_obj_to_inven(self_obj, create_object(112, 1, 2));
         end
         if (random(0, 100) <= 25) then begin
-            add_mult_objs_to_inven(self_obj, create_object_sid(215, 1, 2, -1), random(1, 3));
+            add_mult_objs_to_inven(self_obj, create_object(215, 1, 2), random(1, 3));
         end
         if (random(0, 100) <= 10) then begin
-            add_obj_to_inven(self_obj, create_object_sid(64, 1, 2, -1));
+            add_obj_to_inven(self_obj, create_object(64, 1, 2));
         end
         if (random(0, 100) <= 9) then begin
-            add_obj_to_inven(self_obj, create_object_sid(236, 1, 2, -1));
+            add_obj_to_inven(self_obj, create_object(236, 1, 2));
         end
         if (random(0, 100) <= 8) then begin
-            add_obj_to_inven(self_obj, create_object_sid(226, 1, 2, -1));
+            add_obj_to_inven(self_obj, create_object(226, 1, 2));
         end
         if (random(0, 100) <= 7) then begin
-            add_obj_to_inven(self_obj, create_object_sid(229, 1, 2, -1));
+            add_obj_to_inven(self_obj, create_object(229, 1, 2));
         end
         if (random(0, 100) <= 6) then begin
-            add_obj_to_inven(self_obj, create_object_sid(230, 1, 2, -1));
+            add_obj_to_inven(self_obj, create_object(230, 1, 2));
         end
     end
 end
@@ -84,14 +84,14 @@ procedure critter_p_proc
 begin
     if ((animation_is_called == 0) and (anim_busy(self_obj) == 0)) then begin
         animation_is_called := 1;
-        add_timer_event(self_obj, random(1000, 1000), 1);
+        add_timer_event(self_obj, 1000, 1);
     end
     if (float_is_called == 0) then begin
         float_is_called := 1;
-        add_timer_event(self_obj, random(1000, 1000), 2);
+        add_timer_event(self_obj, 1000, 2);
     end
     if (obj_can_see_obj(self_obj, dude_obj)) then begin
-        if ((local_var(2) == 1) or (global_var(59) <= -31)) then begin
+        if ((local_var(2) == 1) or (global_var(GVAR_TOWN_REP_VAULT_13) <= -31)) then begin
             set_local_var(2, 0);
             attack_setup(self_obj, dude_obj);
         end
@@ -124,25 +124,25 @@ end
 procedure talk_p_proc
 begin
     if (not(anim_busy(self_obj))) then begin
-        anim(self_obj, 1000, rotation_to_tile(tile_num(self_obj), tile_num(dude_obj)));
+        anim(self_obj, 1000, rotation_to_tile(tile_num(self_obj), dude_tile));
     end
-    if ((obj_pid(critter_inven_obj(dude_obj, 0)) == 120) or (obj_pid(critter_inven_obj(dude_obj, 0)) == 121) or (obj_pid(critter_inven_obj(dude_obj, 0)) == 122) or (obj_pid(critter_inven_obj(dude_obj, 0)) == 123) or (obj_pid(critter_inven_obj(dude_obj, 0)) == 129) or (obj_pid(critter_inven_obj(dude_obj, 0)) == 391)) then begin
-        float_msg(self_obj, message_str(132, random(305, 309)), 0);
+    if ((obj_pid(critter_inven_obj(dude_obj, INVEN_TYPE_WORN)) == 120) or (obj_pid(critter_inven_obj(dude_obj, INVEN_TYPE_WORN)) == 121) or (obj_pid(critter_inven_obj(dude_obj, INVEN_TYPE_WORN)) == 122) or (obj_pid(critter_inven_obj(dude_obj, INVEN_TYPE_WORN)) == 123) or (obj_pid(critter_inven_obj(dude_obj, INVEN_TYPE_WORN)) == 129) or (obj_pid(critter_inven_obj(dude_obj, INVEN_TYPE_WORN)) == 391)) then begin
+        float_msg(self_obj, message_str(SCRIPT_PNSPOUK, random(305, 309)), FLOAT_MSG_NORMAL);
     end
     else begin
-        float_msg(self_obj, message_str(132, random(300, 304)), 0);
+        float_msg(self_obj, message_str(SCRIPT_PNSPOUK, random(300, 304)), FLOAT_MSG_NORMAL);
     end
 end
 
 procedure destroy_p_proc
 begin
     if (source_obj == dude_obj) then begin
-        set_global_var(0, global_var(0) - 1);
+        set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) - 1);
         if ((59 == 50) or (59 == 59)) then begin
-            set_global_var(59, global_var(59) - 3);
+            set_global_var(GVAR_TOWN_REP_VAULT_13, global_var(GVAR_TOWN_REP_VAULT_13) - 3);
         end
         else begin
-            set_global_var(59, global_var(59) - 6);
+            set_global_var(GVAR_TOWN_REP_VAULT_13, global_var(GVAR_TOWN_REP_VAULT_13) - 6);
         end
     end
 end
@@ -155,51 +155,51 @@ begin
             set_sfall_global(13, 1);
         end
         else begin
-            if (has_trait(2, dude_obj, 13)) then begin
-                if (global_var(81) < get_pc_stat(1)) then begin
+            if (has_trait(TRAIT_TRAIT, dude_obj, TRAIT_sex_appeal)) then begin
+                if (global_var(GVAR_VAULT_CITIZENSHIP) < get_pc_stat(PCSTAT_level)) then begin
                     if (is_success(roll_vs_skill(dude_obj, 10, 0))) then begin
-                        set_global_var(98, global_var(98) + 3);
+                        set_global_var(GVAR_QUEST_FIND_THOMAS, global_var(GVAR_QUEST_FIND_THOMAS) + 3);
                     end
                     else begin
-                        set_global_var(98, global_var(98) + 1);
+                        set_global_var(GVAR_QUEST_FIND_THOMAS, global_var(GVAR_QUEST_FIND_THOMAS) + 1);
                     end
-                    if (global_var(98) >= (2 * get_pc_stat(1))) then begin
-                        set_global_var(98, 0);
-                        set_global_var(81, global_var(81) + 1);
-                        critter_mod_skill(dude_obj, 10, 1);
-                        display_msg(message_str(10, 410));
+                    if (global_var(GVAR_QUEST_FIND_THOMAS) >= (2 * get_pc_stat(PCSTAT_level))) then begin
+                        set_global_var(GVAR_QUEST_FIND_THOMAS, 0);
+                        set_global_var(GVAR_VAULT_CITIZENSHIP, global_var(GVAR_VAULT_CITIZENSHIP) + 1);
+                        critter_mod_skill(dude_obj, SKILL_STEAL, 1);
+                        display_msg(message_str(SCRIPT_STTEXT, 410));
                     end
                 end
             end
             steal_perk := 0;
             steal_gloves := 0;
-            if (has_trait(0, dude_obj, 37) != 0) then begin
+            if (has_trait(TRAIT_PERK, dude_obj, PERK_pickpocket) != 0) then begin
                 steal_perk := steal_perk + 15;
             end
-            if (has_trait(0, dude_obj, 33) != 0) then begin
+            if (has_trait(TRAIT_PERK, dude_obj, PERK_master_thief) != 0) then begin
                 steal_perk := steal_perk + 30;
             end
-            if ((obj_pid(critter_inven_obj(dude_obj, 1)) == 245) or (obj_pid(critter_inven_obj(dude_obj, 2)) == 245)) then begin
+            if ((obj_pid(critter_inven_obj(dude_obj, INVEN_TYPE_RIGHT_HAND)) == 245) or (obj_pid(critter_inven_obj(dude_obj, INVEN_TYPE_LEFT_HAND)) == 245)) then begin
                 steal_gloves := 10;
             end
-            if (random(0, 100) <= ((has_skill(dude_obj, 10) / 2) + (((3 * get_critter_stat(dude_obj, 5)) + (((get_critter_stat(dude_obj, 6) - 4) * 4) - (difficulty_level * 4)) + (2 * steal_perk) + steal_gloves) / (local_var(4) / 25)))) then begin
-                display_msg(message_str(10, random(145, 148)));
+            if (random(0, 100) <= ((dude_skill(SKILL_STEAL) / 2) + (((3 * get_critter_stat(dude_obj, STAT_ag)) + (((dude_luck - 4) * 4) - (difficulty_level * 4)) + (2 * steal_perk) + steal_gloves) / (local_var(4) / 25)))) then begin
+                display_msg(message_str(SCRIPT_STTEXT, random(145, 148)));
             end
             else begin
                 script_overrides;
-                if (random(0, 100) <= (25 - floor((((get_critter_stat(dude_obj, 6) - 4) * 4) - (difficulty_level * 4)) / 3))) then begin
-                    float_msg(self_obj, message_str(10, random(206, 209)), 2);
+                if (random(0, 100) <= (25 - floor((((dude_luck - 4) * 4) - (difficulty_level * 4)) / 3))) then begin
+                    float_msg(self_obj, message_str(SCRIPT_STTEXT, random(206, 209)), FLOAT_MSG_RED);
                     set_local_var(2, 1);
                 end
                 else begin
-                    display_msg(message_str(10, random(200, 203)));
+                    display_msg(message_str(SCRIPT_STTEXT, random(200, 203)));
                 end
             end
         end
     end
     if (action_being_used == 12) then begin
         script_overrides;
-        display_msg(message_str(10, 204) + ((has_skill(dude_obj, 10) / 2) + (((3 * get_critter_stat(dude_obj, 5)) + (((get_critter_stat(dude_obj, 6) - 4) * 4) - (difficulty_level * 4)) + (2 * steal_perk) + steal_gloves) / (local_var(4) / 25))) + message_str(10, 205));
+        display_msg(message_str(SCRIPT_STTEXT, 204) + ((dude_skill(SKILL_STEAL) / 2) + (((3 * get_critter_stat(dude_obj, STAT_ag)) + (((dude_luck - 4) * 4) - (difficulty_level * 4)) + (2 * steal_perk) + steal_gloves) / (local_var(4) / 25))) + message_str(SCRIPT_STTEXT, 205));
     end
 end
 
@@ -227,4 +227,3 @@ begin
         float_is_called := 0;
     end
 end
-
