diff --git a/Mods/Olympus2207/Original/PNXRAT.ssl b/Mods/Olympus2207/ReDefine/PNXRAT.ssl
index 082c911..2ec1240 100644
--- a/Mods/Olympus2207/Original/PNXRAT.ssl
+++ b/Mods/Olympus2207/ReDefine/PNXRAT.ssl
@@ -33,10 +33,10 @@ begin
         set_local_var(3, tile_num(self_obj));
         set_local_var(1, 1);
         if (obj_pid(self_obj) == 16777421) then begin
-            critter_add_trait(self_obj, 1, 6, 205);
-            critter_add_trait(self_obj, 1, 5, 69);
-            add_obj_to_inven(self_obj, create_object_sid(521, 1, 2, -1));
-            wield_obj_critter(self_obj, obj_carrying_pid_obj(self_obj, 521));
+            critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 205);
+            critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_GENERIC_RAT);
+            add_obj_to_inven(self_obj, create_object(521, 1, 2));
+            wield_obj(obj_carrying_pid_obj(self_obj, 521));
         end
         max_distance := random(2, 6);
     end
@@ -89,7 +89,7 @@ begin
     if (source_obj == dude_obj) then begin
     end
     if (not(tile_contains_obj_pid(2, 2, 250))) then begin
-        stuff_container := create_object_sid(250, 2, 2, -1);
+        stuff_container := create_object(250, 2, 2);
     end
     else begin
         stuff_container := tile_contains_pid_obj(2, 2, 250);
@@ -131,4 +131,3 @@ begin
         animation_is_called := 0;
     end
 end
-
