diff --git a/Mods/Olympus2207/Original/RBKABAN.ssl b/Mods/Olympus2207/ReDefine/RBKABAN.ssl
index 780429f..df1c277 100644
--- a/Mods/Olympus2207/Original/RBKABAN.ssl
+++ b/Mods/Olympus2207/ReDefine/RBKABAN.ssl
@@ -55,89 +55,89 @@ begin
         set_local_var(4, random(85, 95));
         set_local_var(5, random(20, 30));
         set_local_var(6, random(-20, 10));
-        critter_add_trait(self_obj, 1, 6, 9);
-        critter_add_trait(self_obj, 1, 5, 25);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 9);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_RAIDER_BERSERK);
         if (random(0, 100) <= 15) then begin
-            add_obj_to_inven(self_obj, create_object_sid(438, 1, 2, -1));
+            add_obj_to_inven(self_obj, create_object(438, 1, 2));
         end
         if (random(0, 100) <= 10) then begin
-            add_mult_objs_to_inven(self_obj, create_object_sid(238, 1, 2, -1), random(1, 2));
+            add_mult_objs_to_inven(self_obj, create_object(238, 1, 2), random(1, 2));
         end
         if (random(0, 100) <= 10) then begin
-            add_mult_objs_to_inven(self_obj, create_object_sid(213, 1, 2, -1), random(1, 2));
+            add_mult_objs_to_inven(self_obj, create_object(213, 1, 2), random(1, 2));
         end
         if (random(0, 100) <= 10) then begin
-            add_obj_to_inven(self_obj, create_object_sid(233, 1, 2, -1));
+            add_obj_to_inven(self_obj, create_object(233, 1, 2));
         end
         if (random(0, 100) <= 10) then begin
-            add_obj_to_inven(self_obj, create_object_sid(420, 1, 2, -1));
+            add_obj_to_inven(self_obj, create_object(420, 1, 2));
         end
         if (random(0, 100) <= 9) then begin
-            add_obj_to_inven(self_obj, create_object_sid(236, 1, 2, -1));
+            add_obj_to_inven(self_obj, create_object(236, 1, 2));
         end
         if (random(0, 100) <= 8) then begin
-            add_obj_to_inven(self_obj, create_object_sid(218, 1, 2, -1));
+            add_obj_to_inven(self_obj, create_object(218, 1, 2));
         end
         if (random(0, 100) <= 6) then begin
-            add_obj_to_inven(self_obj, create_object_sid(440, 1, 2, -1));
+            add_obj_to_inven(self_obj, create_object(440, 1, 2));
         end
         if (random(0, 100) <= 5) then begin
-            add_obj_to_inven(self_obj, create_object_sid(244, 1, 2, -1));
+            add_obj_to_inven(self_obj, create_object(244, 1, 2));
         end
         if (random(0, 100) <= 5) then begin
-            add_obj_to_inven(self_obj, create_object_sid(428, 1, 2, -1));
+            add_obj_to_inven(self_obj, create_object(428, 1, 2));
         end
         if (random(0, 100) <= 5) then begin
-            add_obj_to_inven(self_obj, create_object_sid(422, 1, 2, -1));
+            add_obj_to_inven(self_obj, create_object(422, 1, 2));
         end
         if (random(0, 100) <= 4) then begin
-            add_obj_to_inven(self_obj, create_object_sid(423, 1, 2, -1));
+            add_obj_to_inven(self_obj, create_object(423, 1, 2));
         end
         if (random(0, 100) <= 3) then begin
-            add_obj_to_inven(self_obj, create_object_sid(426, 1, 2, -1));
+            add_obj_to_inven(self_obj, create_object(426, 1, 2));
         end
         if (random(0, 100) <= 2) then begin
-            add_obj_to_inven(self_obj, create_object_sid(433, 1, 2, -1));
+            add_obj_to_inven(self_obj, create_object(433, 1, 2));
         end
         if (random(0, 100) <= 2) then begin
-            add_obj_to_inven(self_obj, create_object_sid(421, 1, 2, -1));
+            add_obj_to_inven(self_obj, create_object(421, 1, 2));
         end
         if (random(0, 100) <= 33) then begin
-            add_obj_to_inven(self_obj, create_object_sid(162, 1, 2, -1));
-            add_mult_objs_to_inven(self_obj, create_object_sid(2, 1, 2, -1), random(21, 38));
-            wield_obj_critter(self_obj, obj_carrying_pid_obj(self_obj, 162));
+            add_obj_to_inven(self_obj, create_object(162, 1, 2));
+            add_mult_objs_to_inven(self_obj, create_object(2, 1, 2), random(21, 38));
+            wield_obj(obj_carrying_pid_obj(self_obj, 162));
         end
         else begin
-            add_obj_to_inven(self_obj, create_object_sid(180, 1, 2, -1));
-            add_mult_objs_to_inven(self_obj, create_object_sid(14, 1, 2, -1), random(10, 18));
-            wield_obj_critter(self_obj, obj_carrying_pid_obj(self_obj, 180));
+            add_obj_to_inven(self_obj, create_object(180, 1, 2));
+            add_mult_objs_to_inven(self_obj, create_object(14, 1, 2), random(10, 18));
+            wield_obj(obj_carrying_pid_obj(self_obj, 180));
         end
         if (random(0, 100) <= 33) then begin
-            add_obj_to_inven(self_obj, create_object_sid(229, 1, 2, -1));
+            add_obj_to_inven(self_obj, create_object(229, 1, 2));
         end
         else begin
-            add_obj_to_inven(self_obj, create_object_sid(65, 1, 2, -1));
+            add_obj_to_inven(self_obj, create_object(65, 1, 2));
         end
         if (random(0, 100) <= 33) then begin
-            add_mult_objs_to_inven(self_obj, create_object_sid(41, 1, 2, -1), random(10, 40));
+            add_mult_objs_to_inven(self_obj, create_object(41, 1, 2), random(10, 40));
         end
         if (random(0, 100) <= 25) then begin
-            add_mult_objs_to_inven(self_obj, create_object_sid(63, 1, 2, -1), random(2, 4));
+            add_mult_objs_to_inven(self_obj, create_object(63, 1, 2), random(2, 4));
         end
         if (random(0, 100) <= 10) then begin
-            add_obj_to_inven(self_obj, create_object_sid(233, 1, 2, -1));
+            add_obj_to_inven(self_obj, create_object(233, 1, 2));
         end
         if (random(0, 100) <= 9) then begin
-            add_obj_to_inven(self_obj, create_object_sid(434, 1, 2, -1));
+            add_obj_to_inven(self_obj, create_object(434, 1, 2));
         end
         if (random(0, 100) <= 8) then begin
-            add_obj_to_inven(self_obj, create_object_sid(236, 1, 2, -1));
+            add_obj_to_inven(self_obj, create_object(236, 1, 2));
         end
         if (random(0, 100) <= 7) then begin
-            add_obj_to_inven(self_obj, create_object_sid(222, 1, 2, -1));
+            add_obj_to_inven(self_obj, create_object(222, 1, 2));
         end
         if (random(0, 100) <= 6) then begin
-            add_obj_to_inven(self_obj, create_object_sid(230, 1, 2, -1));
+            add_obj_to_inven(self_obj, create_object(230, 1, 2));
         end
     end
 end
@@ -153,7 +153,7 @@ begin
         add_timer_event(self_obj, random(1000, 1000), 2);
     end
     if (obj_can_see_obj(self_obj, dude_obj)) then begin
-        if ((local_var(2) == 1) or (global_var(54) <= -31)) then begin
+        if ((local_var(2) == 1) or (global_var(GVAR_TOWN_REP_BROKEN_HILLS) <= -31)) then begin
             set_local_var(2, 0);
             attack_setup(self_obj, dude_obj);
         end
@@ -168,10 +168,10 @@ procedure description_p_proc
 begin
     script_overrides;
     if (local_var(7) == 0) then begin
-        display_msg(message_str(253, 2));
+        display_mstr(2);
     end
     else begin
-        display_msg(message_str(253, 4));
+        display_mstr(4);
     end
 end
 
@@ -179,10 +179,10 @@ procedure look_at_p_proc
 begin
     script_overrides;
     if (local_var(7) == 0) then begin
-        display_msg(message_str(253, 1));
+        display_mstr(1);
     end
     else begin
-        display_msg(message_str(253, 3));
+        display_mstr(3);
     end
 end
 
@@ -196,12 +196,12 @@ end
 procedure talk_p_proc
 begin
     if (not(anim_busy(self_obj))) then begin
-        anim(self_obj, 1000, rotation_to_tile(tile_num(self_obj), tile_num(dude_obj)));
+        anim(self_obj, 1000, rotation_to_tile(tile_num(self_obj), dude_tile));
     end
     start_gdialog(253, self_obj, -1, -1, -1);
     gsay_start;
-    if (global_var(305) == 2) then begin
-        set_global_var(305, 3);
+    if (global_var(GVAR_BH_MISSING) == 2) then begin
+        set_global_var(GVAR_BH_MISSING, 3);
         game_ui_enable;
         call Node011();
     end
@@ -219,13 +219,13 @@ end
 
 procedure destroy_p_proc
 begin
-    set_global_var(118, global_var(118) bwor 67108864);
-    if (global_var(175) == 1) then begin
-        set_global_var(175, 0);
+    set_global_var(GVAR_MODOC_TOOL_FLAG, global_var(GVAR_MODOC_TOOL_FLAG) bwor 67108864);
+    if (global_var(GVAR_NEWRENO_SNUFF_CARLSON) == 1) then begin
+        set_global_var(GVAR_NEWRENO_SNUFF_CARLSON, 0);
     end
     if (source_obj == dude_obj) then begin
-        set_global_var(0, global_var(0) - 1);
-        set_global_var(54, global_var(54) - 6);
+        set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) - 1);
+        set_global_var(GVAR_TOWN_REP_BROKEN_HILLS, global_var(GVAR_TOWN_REP_BROKEN_HILLS) - 6);
     end
 end
 
@@ -237,51 +237,51 @@ begin
             set_sfall_global(13, 1);
         end
         else begin
-            if (has_trait(2, dude_obj, 13)) then begin
-                if (global_var(81) < get_pc_stat(1)) then begin
+            if (has_trait(TRAIT_TRAIT, dude_obj, TRAIT_sex_appeal)) then begin
+                if (global_var(GVAR_VAULT_CITIZENSHIP) < get_pc_stat(PCSTAT_level)) then begin
                     if (is_success(roll_vs_skill(dude_obj, 10, 0))) then begin
-                        set_global_var(98, global_var(98) + 3);
+                        set_global_var(GVAR_QUEST_FIND_THOMAS, global_var(GVAR_QUEST_FIND_THOMAS) + 3);
                     end
                     else begin
-                        set_global_var(98, global_var(98) + 1);
+                        set_global_var(GVAR_QUEST_FIND_THOMAS, global_var(GVAR_QUEST_FIND_THOMAS) + 1);
                     end
-                    if (global_var(98) >= (2 * get_pc_stat(1))) then begin
-                        set_global_var(98, 0);
-                        set_global_var(81, global_var(81) + 1);
-                        critter_mod_skill(dude_obj, 10, 1);
-                        display_msg(message_str(10, 410));
+                    if (global_var(GVAR_QUEST_FIND_THOMAS) >= (2 * get_pc_stat(PCSTAT_level))) then begin
+                        set_global_var(GVAR_QUEST_FIND_THOMAS, 0);
+                        set_global_var(GVAR_VAULT_CITIZENSHIP, global_var(GVAR_VAULT_CITIZENSHIP) + 1);
+                        critter_mod_skill(dude_obj, SKILL_STEAL, 1);
+                        display_msg(message_str(SCRIPT_STTEXT, 410));
                     end
                 end
             end
             steal_perk := 0;
             steal_gloves := 0;
-            if (has_trait(0, dude_obj, 37) != 0) then begin
+            if (has_trait(TRAIT_PERK, dude_obj, PERK_pickpocket) != 0) then begin
                 steal_perk := steal_perk + 15;
             end
-            if (has_trait(0, dude_obj, 33) != 0) then begin
+            if (has_trait(TRAIT_PERK, dude_obj, PERK_master_thief) != 0) then begin
                 steal_perk := steal_perk + 30;
             end
-            if ((obj_pid(critter_inven_obj(dude_obj, 1)) == 245) or (obj_pid(critter_inven_obj(dude_obj, 2)) == 245)) then begin
+            if ((obj_pid(critter_inven_obj(dude_obj, INVEN_TYPE_RIGHT_HAND)) == 245) or (obj_pid(critter_inven_obj(dude_obj, INVEN_TYPE_LEFT_HAND)) == 245)) then begin
                 steal_gloves := 10;
             end
-            if (random(0, 100) <= ((has_skill(dude_obj, 10) / 2) + (((3 * get_critter_stat(dude_obj, 5)) + (((get_critter_stat(dude_obj, 6) - 4) * 4) - (difficulty_level * 4)) + (2 * steal_perk) + steal_gloves) / (local_var(4) / 25)))) then begin
-                display_msg(message_str(10, random(145, 148)));
+            if (random(0, 100) <= ((dude_skill(SKILL_STEAL) / 2) + (((3 * get_critter_stat(dude_obj, STAT_ag)) + (((dude_luck - 4) * 4) - (difficulty_level * 4)) + (2 * steal_perk) + steal_gloves) / (local_var(4) / 25)))) then begin
+                display_msg(message_str(SCRIPT_STTEXT, random(145, 148)));
             end
             else begin
                 script_overrides;
-                if (random(0, 100) <= (25 - floor((((get_critter_stat(dude_obj, 6) - 4) * 4) - (difficulty_level * 4)) / 3))) then begin
-                    float_msg(self_obj, message_str(10, random(206, 209)), 2);
+                if (random(0, 100) <= (25 - floor((((dude_luck - 4) * 4) - (difficulty_level * 4)) / 3))) then begin
+                    float_msg(self_obj, message_str(SCRIPT_STTEXT, random(206, 209)), FLOAT_MSG_RED);
                     set_local_var(2, 1);
                 end
                 else begin
-                    display_msg(message_str(10, random(200, 203)));
+                    display_msg(message_str(SCRIPT_STTEXT, random(200, 203)));
                 end
             end
         end
     end
     if (action_being_used == 12) then begin
         script_overrides;
-        display_msg(message_str(10, 204) + ((has_skill(dude_obj, 10) / 2) + (((3 * get_critter_stat(dude_obj, 5)) + (((get_critter_stat(dude_obj, 6) - 4) * 4) - (difficulty_level * 4)) + (2 * steal_perk) + steal_gloves) / (local_var(4) / 25))) + message_str(10, 205));
+        display_msg(message_str(SCRIPT_STTEXT, 204) + ((dude_skill(SKILL_STEAL) / 2) + (((3 * get_critter_stat(dude_obj, STAT_ag)) + (((dude_luck - 4) * 4) - (difficulty_level * 4)) + (2 * steal_perk) + steal_gloves) / (local_var(4) / 25))) + message_str(SCRIPT_STTEXT, 205));
     end
 end
 
@@ -298,13 +298,13 @@ end
 
 procedure map_enter_p_proc
 begin
-    if (((global_var(118) bwand 67108864) != 0) and (elevation(self_obj) != 2)) then begin
+    if (((global_var(GVAR_MODOC_TOOL_FLAG) bwand 67108864) != 0) and (elevation(self_obj) != 2)) then begin
         move_to(self_obj, tile_num(self_obj), 2);
     end
-    if (global_var(305) == 1) then begin
+    if (global_var(GVAR_BH_MISSING) == 1) then begin
         game_ui_disable;
-        set_global_var(305, 2);
-        move_to(dude_obj, tile_num_in_direction(tile_num(self_obj), 2, 2), elevation(dude_obj));
+        set_global_var(GVAR_BH_MISSING, 2);
+        move_to(dude_obj, tile_num_in_direction(tile_num(self_obj), 2, 2), dude_elevation);
         anim(dude_obj, 1000, 5);
         fadeIn(800);
         add_timer_event(self_obj, 8, 3);
@@ -328,73 +328,73 @@ procedure Node001
 begin
     set_local_var(7, 1);
     already_talk := 1;
-    gsay_reply(253, 10);
-    gsay_option(253, 11, Node024, 50);
-    gsay_option(253, 12, Node016, 50);
-    if ((global_var(170) == 1) and (global_var(374) == 0)) then begin
-        gsay_option(253, 13, Node025, 50);
+    Reply(10);
+    giq_option(1, NAME, 11, Node024, NEUTRAL_REACTION);
+    giq_option(1, NAME, 12, Node016, NEUTRAL_REACTION);
+    if ((global_var(GVAR_NCR_BEAT_HOSS) == 1) and (global_var(GVAR_DEN_FRED_STATUS) == 0)) then begin
+        giq_option(1, NAME, 13, Node025, NEUTRAL_REACTION);
     end
-    gsay_option(253, 14, Node999, 50);
+    giq_option(1, NAME, 14, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node010
 begin
     if (local_var(8) == 1) then begin
-        gsay_reply(253, 1001);
-        gsay_option(253, 101, Node023, 50);
-        gsay_option(253, 102, Node999, 50);
+        Reply(1001);
+        giq_option(1, NAME, 101, Node023, NEUTRAL_REACTION);
+        giq_option(1, NAME, 102, Node999, NEUTRAL_REACTION);
     end
     else begin
-        if (global_var(175) == 2) then begin
-            gsay_reply(253, 1003);
-            if ((global_var(170) == 1) and (global_var(374) == 0)) then begin
-                gsay_option(253, 107, Node025, 50);
+        if (global_var(GVAR_NEWRENO_SNUFF_CARLSON) == 2) then begin
+            Reply(1003);
+            if ((global_var(GVAR_NCR_BEAT_HOSS) == 1) and (global_var(GVAR_DEN_FRED_STATUS) == 0)) then begin
+                giq_option(1, NAME, 107, Node025, NEUTRAL_REACTION);
             end
-            gsay_option(253, 108, Node024, 50);
-            gsay_option(253, 109, Node999, 50);
+            giq_option(1, NAME, 108, Node024, NEUTRAL_REACTION);
+            giq_option(1, NAME, 109, Node999, NEUTRAL_REACTION);
         end
         else begin
-            if (global_var(306) == 1) then begin
+            if (global_var(GVAR_BH_MIGHTY_MAN) == 1) then begin
                 if ((game_time_in_seconds - local_var(9)) >= 2592000) then begin
                     if ((game_time_in_seconds - local_var(9)) >= 7776000) then begin
-                        gsay_reply(253, 1006);
-                        gsay_option(253, 1017, Node998, 50);
-                        gsay_option(253, 1018, Node998, 50);
+                        Reply(1006);
+                        giq_option(1, NAME, 1017, Node998, NEUTRAL_REACTION);
+                        giq_option(1, NAME, 1018, Node998, NEUTRAL_REACTION);
                     end
                     else begin
                         duty_to_pay := (game_time_in_seconds - local_var(9)) / 3600 / 24 * 12;
-                        gsay_reply(253, 1005);
+                        Reply(1005);
                         if (item_caps_total(dude_obj) >= duty_to_pay) then begin
-                            gsay_option(253, 1014, NodePayDuty, 50);
+                            giq_option(1, NAME, 1014, NodePayDuty, NEUTRAL_REACTION);
                         end
-                        gsay_option(253, 1015, Node023, 50);
-                        if ((global_var(170) == 1) and (global_var(374) == 0)) then begin
-                            gsay_option(253, 107, Node025, 50);
+                        giq_option(1, NAME, 1015, Node023, NEUTRAL_REACTION);
+                        if ((global_var(GVAR_NCR_BEAT_HOSS) == 1) and (global_var(GVAR_DEN_FRED_STATUS) == 0)) then begin
+                            giq_option(1, NAME, 107, Node025, NEUTRAL_REACTION);
                         end
-                        gsay_option(253, 1016, Node999, 50);
+                        giq_option(1, NAME, 1016, Node999, NEUTRAL_REACTION);
                     end
                 end
                 else begin
-                    gsay_reply(253, 1004);
-                    gsay_option(253, 1010, Node028, 50);
-                    if (global_var(175) == 0) then begin
-                        gsay_option(253, 1011, Node016, 50);
+                    Reply(1004);
+                    giq_option(1, NAME, 1010, Node028, NEUTRAL_REACTION);
+                    if (global_var(GVAR_NEWRENO_SNUFF_CARLSON) == 0) then begin
+                        giq_option(1, NAME, 1011, Node016, NEUTRAL_REACTION);
                     end
-                    if ((global_var(170) == 1) and (global_var(374) == 0)) then begin
-                        gsay_option(253, 1012, Node025, 50);
+                    if ((global_var(GVAR_NCR_BEAT_HOSS) == 1) and (global_var(GVAR_DEN_FRED_STATUS) == 0)) then begin
+                        giq_option(1, NAME, 1012, Node025, NEUTRAL_REACTION);
                     end
-                    gsay_option(253, 1013, Node999, 50);
+                    giq_option(1, NAME, 1013, Node999, NEUTRAL_REACTION);
                 end
             end
             else begin
-                gsay_reply(253, 1002);
-                gsay_option(253, 103, Node024, 50);
-                gsay_option(253, 104, Node016, 50);
-                if ((global_var(170) == 1) and (global_var(374) == 0)) then begin
-                    gsay_option(253, 105, Node025, 50);
+                Reply(1002);
+                giq_option(1, NAME, 103, Node024, NEUTRAL_REACTION);
+                giq_option(1, NAME, 104, Node016, NEUTRAL_REACTION);
+                if ((global_var(GVAR_NCR_BEAT_HOSS) == 1) and (global_var(GVAR_DEN_FRED_STATUS) == 0)) then begin
+                    giq_option(1, NAME, 105, Node025, NEUTRAL_REACTION);
                 end
-                if ((global_var(175) == 1) and (global_var(307) > 0)) then begin
-                    gsay_option(253, 106, Node020, 50);
+                if ((global_var(GVAR_NEWRENO_SNUFF_CARLSON) == 1) and (global_var(GVAR_BH_MINING) > 0)) then begin
+                    giq_option(1, NAME, 106, Node020, NEUTRAL_REACTION);
                 end
             end
         end
@@ -403,180 +403,180 @@ end
 
 procedure Node011
 begin
-    gsay_reply(253, 110);
-    if (global_var(285) == 3) then begin
-        gsay_option(253, 113, Node015, 50);
+    Reply(110);
+    if (global_var(GVAR_NEW_RENO_WRIGHT_FLAGS) == 3) then begin
+        giq_option(1, NAME, 113, Node015, NEUTRAL_REACTION);
     end
     else begin
-        gsay_option(253, 111, Node012, 50);
-        gsay_option(253, 112, Node012, 50);
+        giq_option(1, NAME, 111, Node012, NEUTRAL_REACTION);
+        giq_option(1, NAME, 112, Node012, NEUTRAL_REACTION);
     end
 end
 
 procedure Node012
 begin
-    gsay_reply(253, 1202);
-    gsay_option(253, 123, Node023, 50);
-    gsay_option(253, 124, Node013, 50);
+    Reply(1202);
+    giq_option(1, NAME, 123, Node023, NEUTRAL_REACTION);
+    giq_option(1, NAME, 124, Node013, NEUTRAL_REACTION);
 end
 
 procedure Node013
 begin
-    gsay_reply(253, 130);
-    gsay_option(253, 131, Node014, 50);
-    gsay_option(253, 132, Node023, 50);
-    gsay_option(253, 133, Node022, 50);
+    Reply(130);
+    giq_option(1, NAME, 131, Node014, NEUTRAL_REACTION);
+    giq_option(1, NAME, 132, Node023, NEUTRAL_REACTION);
+    giq_option(1, NAME, 133, Node022, NEUTRAL_REACTION);
 end
 
 procedure Node014
 begin
-    set_global_var(306, 1);
+    set_global_var(GVAR_BH_MIGHTY_MAN, 1);
     set_local_var(9, game_time_in_seconds);
-    gsay_reply(253, 140);
-    gsay_option(253, 141, Node999, 50);
-    gsay_option(253, 142, Node023, 50);
+    Reply(140);
+    giq_option(1, NAME, 141, Node999, NEUTRAL_REACTION);
+    giq_option(1, NAME, 142, Node023, NEUTRAL_REACTION);
 end
 
 procedure Node015
 begin
-    set_global_var(306, 1);
+    set_global_var(GVAR_BH_MIGHTY_MAN, 1);
     set_local_var(9, game_time_in_seconds);
-    gsay_reply(253, 150);
-    gsay_option(253, 151, Node999, 50);
+    Reply(150);
+    giq_option(1, NAME, 151, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node016
 begin
-    if (global_var(285) == 2) then begin
-        gsay_reply(253, 1601);
-        gsay_option(253, 161, Node017, 50);
+    if (global_var(GVAR_NEW_RENO_WRIGHT_FLAGS) == 2) then begin
+        Reply(1601);
+        giq_option(1, NAME, 161, Node017, NEUTRAL_REACTION);
     end
     else begin
-        gsay_reply(253, 1602);
-        gsay_option(253, 162, Node999, 50);
+        Reply(1602);
+        giq_option(1, NAME, 162, Node999, NEUTRAL_REACTION);
     end
 end
 
 procedure Node017
 begin
-    gsay_reply(253, 170);
-    gsay_option(253, 171, Node018, 50);
+    Reply(170);
+    giq_option(1, NAME, 171, Node018, NEUTRAL_REACTION);
 end
 
 procedure Node018
 begin
-    gsay_reply(253, 180);
-    gsay_option(253, 181, Node019, 50);
-    gsay_option(253, 182, Node999, 50);
+    Reply(180);
+    giq_option(1, NAME, 181, Node019, NEUTRAL_REACTION);
+    giq_option(1, NAME, 182, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node019
 begin
-    if (global_var(175) == 0) then begin
-        set_global_var(175, 1);
+    if (global_var(GVAR_NEWRENO_SNUFF_CARLSON) == 0) then begin
+        set_global_var(GVAR_NEWRENO_SNUFF_CARLSON, 1);
     end
-    gsay_reply(253, 190);
-    gsay_option(253, 191, Node999, 50);
+    Reply(190);
+    giq_option(1, NAME, 191, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node020
 begin
-    gsay_reply(253, 200);
-    gsay_option(253, 201, Node021, 50);
+    Reply(200);
+    giq_option(1, NAME, 201, Node021, NEUTRAL_REACTION);
 end
 
 procedure Node021
 begin
-    if (global_var(175) == 1) then begin
-        set_global_var(175, 2);
-        give_exp_points(floor(276) + (get_critter_stat(dude_obj, 4) * floor(115)) + (get_critter_stat(dude_obj, 6) * floor(46)) + (floor(18) * (((get_critter_stat(dude_obj, 6) - 4) * 4) - (difficulty_level * 4))));
-        display_msg(message_str(10, random(140, 143)) + (floor(276) + (get_critter_stat(dude_obj, 4) * floor(115)) + (get_critter_stat(dude_obj, 6) * floor(46)) + (floor(18) * (((get_critter_stat(dude_obj, 6) - 4) * 4) - (difficulty_level * 4)))) + message_str(10, 144));
-        add_obj_to_inven(dude_obj, create_object_sid(196, 1, 2, -1));
-        set_global_var(283, 1);
+    if (global_var(GVAR_NEWRENO_SNUFF_CARLSON) == 1) then begin
+        set_global_var(GVAR_NEWRENO_SNUFF_CARLSON, 2);
+        give_exp_points(floor(276) + (dude_iq * floor(115)) + (dude_luck * floor(46)) + (floor(18) * (((dude_luck - 4) * 4) - (difficulty_level * 4))));
+        display_msg(message_str(SCRIPT_STTEXT, random(140, 143)) + (floor(276) + (dude_iq * floor(115)) + (dude_luck * floor(46)) + (floor(18) * (((dude_luck - 4) * 4) - (difficulty_level * 4)))) + message_str(SCRIPT_STTEXT, 144));
+        add_obj_to_inven(dude_obj, create_object(196, 1, 2));
+        set_global_var(GVAR_CARAVAN_ENCOUNTERS_TOTAL, 1);
     end
-    gsay_reply(253, 210);
-    gsay_option(253, 211, Node999, 50);
+    Reply(210);
+    giq_option(1, NAME, 211, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node022
 begin
-    critter_heal(dude_obj, -(get_critter_stat(dude_obj, 35) / 2));
+    critter_heal(dude_obj, -(get_critter_stat(dude_obj, STAT_current_hp) / 2));
     set_local_var(8, 1);
-    gsay_reply(253, 220);
-    gsay_option(253, 221, Node998, 50);
-    gsay_option(253, 222, Node999, 50);
+    Reply(220);
+    giq_option(1, NAME, 221, Node998, NEUTRAL_REACTION);
+    giq_option(1, NAME, 222, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node023
 begin
-    gsay_reply(253, 230);
-    gsay_option(253, 231, Node998, 50);
-    gsay_option(253, 232, Node998, 50);
+    Reply(230);
+    giq_option(1, NAME, 231, Node998, NEUTRAL_REACTION);
+    giq_option(1, NAME, 232, Node998, NEUTRAL_REACTION);
 end
 
 procedure Node024
 begin
-    gsay_reply(253, 240);
-    gsay_option(253, 241, Node010, 50);
-    gsay_option(253, 242, Node999, 50);
+    Reply(240);
+    giq_option(1, NAME, 241, Node010, NEUTRAL_REACTION);
+    giq_option(1, NAME, 242, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node025
 begin
-    gsay_reply(253, 250);
-    gsay_option(253, 251, Node026, 50);
-    gsay_option(253, 252, Node999, 50);
+    Reply(250);
+    giq_option(1, NAME, 251, Node026, NEUTRAL_REACTION);
+    giq_option(1, NAME, 252, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node026
 begin
-    gsay_reply(253, 260);
-    if ((has_skill(dude_obj, 14) >= 50) or (global_var(175) == 2)) then begin
-        gsay_option(253, 261, Node027, 50);
+    Reply(260);
+    if ((dude_skill(SKILL_SPEECH) >= 50) or (global_var(GVAR_NEWRENO_SNUFF_CARLSON) == 2)) then begin
+        giq_option(1, NAME, 261, Node027, NEUTRAL_REACTION);
     end
-    gsay_option(253, 262, Node999, 50);
+    giq_option(1, NAME, 262, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node027
 begin
-    set_global_var(374, 1);
-    gsay_reply(253, 270);
-    gsay_option(253, 271, Node999, 50);
+    set_global_var(GVAR_DEN_FRED_STATUS, 1);
+    Reply(270);
+    giq_option(1, NAME, 271, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node028
 begin
-    gsay_reply(253, 280);
+    Reply(280);
     if ((((game_time_in_seconds - local_var(9)) / 3600 / 24) > 30) and (item_caps_total(dude_obj) >= 300)) then begin
-        gsay_option(253, 281, NodePay300, 50);
+        giq_option(1, NAME, 281, NodePay300, NEUTRAL_REACTION);
     end
-    gsay_option(253, 282, Node029, 50);
-    gsay_option(253, 283, Node030, 50);
+    giq_option(1, NAME, 282, Node029, NEUTRAL_REACTION);
+    giq_option(1, NAME, 283, Node030, NEUTRAL_REACTION);
 end
 
 procedure Node029
 begin
-    gsay_reply(253, 290);
-    if ((has_skill(dude_obj, 14) >= 80) or (((game_time_in_seconds - local_var(9)) / 3600 / 24) < 20)) then begin
-        gsay_option(253, 291, Node031, 50);
+    Reply(290);
+    if ((dude_skill(SKILL_SPEECH) >= 80) or (((game_time_in_seconds - local_var(9)) / 3600 / 24) < 20)) then begin
+        giq_option(1, NAME, 291, Node031, NEUTRAL_REACTION);
     end
     if (item_caps_total(dude_obj) >= 300) then begin
-        gsay_option(253, 292, NodePay300, 50);
+        giq_option(1, NAME, 292, NodePay300, NEUTRAL_REACTION);
     end
-    gsay_option(253, 293, Node030, 50);
+    giq_option(1, NAME, 293, Node030, NEUTRAL_REACTION);
 end
 
 procedure Node030
 begin
-    gsay_reply(253, 300);
-    gsay_option(253, 301, Node999, 50);
+    Reply(300);
+    giq_option(1, NAME, 301, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node031
 begin
     set_local_var(9, game_time_in_seconds);
-    gsay_reply(253, 310);
-    gsay_option(253, 311, Node999, 50);
+    Reply(310);
+    giq_option(1, NAME, 311, Node999, NEUTRAL_REACTION);
 end
 
 procedure NodePayDuty
@@ -603,4 +603,3 @@ begin
     already_talk := 0;
     set_local_var(2, 1);
 end
-
