diff --git a/Mods/Olympus2207/Original/RBPASTUH.ssl b/Mods/Olympus2207/ReDefine/RBPASTUH.ssl
index fd38dc1..4c9c418 100644
--- a/Mods/Olympus2207/Original/RBPASTUH.ssl
+++ b/Mods/Olympus2207/ReDefine/RBPASTUH.ssl
@@ -65,95 +65,95 @@ begin
         set_local_var(3, tile_num(self_obj));
         set_local_var(1, 1);
         set_local_var(4, random(75, 85));
-        set_local_var(9, has_trait(1, self_obj, 10));
+        set_local_var(9, obj_get_rot(self_obj));
         set_local_var(5, random(40, 50));
         set_local_var(6, random(-20, 10));
-        critter_add_trait(self_obj, 1, 6, 9);
-        critter_add_trait(self_obj, 1, 5, 25);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 9);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_RAIDER_BERSERK);
         if (random(0, 100) <= 15) then begin
-            add_obj_to_inven(self_obj, create_object_sid(438, 1, 2, -1));
+            add_obj_to_inven(self_obj, create_object(438, 1, 2));
         end
         if (random(0, 100) <= 10) then begin
-            add_mult_objs_to_inven(self_obj, create_object_sid(238, 1, 2, -1), random(1, 2));
+            add_mult_objs_to_inven(self_obj, create_object(238, 1, 2), random(1, 2));
         end
         if (random(0, 100) <= 10) then begin
-            add_mult_objs_to_inven(self_obj, create_object_sid(213, 1, 2, -1), random(1, 2));
+            add_mult_objs_to_inven(self_obj, create_object(213, 1, 2), random(1, 2));
         end
         if (random(0, 100) <= 10) then begin
-            add_obj_to_inven(self_obj, create_object_sid(233, 1, 2, -1));
+            add_obj_to_inven(self_obj, create_object(233, 1, 2));
         end
         if (random(0, 100) <= 10) then begin
-            add_obj_to_inven(self_obj, create_object_sid(420, 1, 2, -1));
+            add_obj_to_inven(self_obj, create_object(420, 1, 2));
         end
         if (random(0, 100) <= 9) then begin
-            add_obj_to_inven(self_obj, create_object_sid(236, 1, 2, -1));
+            add_obj_to_inven(self_obj, create_object(236, 1, 2));
         end
         if (random(0, 100) <= 8) then begin
-            add_obj_to_inven(self_obj, create_object_sid(218, 1, 2, -1));
+            add_obj_to_inven(self_obj, create_object(218, 1, 2));
         end
         if (random(0, 100) <= 6) then begin
-            add_obj_to_inven(self_obj, create_object_sid(440, 1, 2, -1));
+            add_obj_to_inven(self_obj, create_object(440, 1, 2));
         end
         if (random(0, 100) <= 5) then begin
-            add_obj_to_inven(self_obj, create_object_sid(244, 1, 2, -1));
+            add_obj_to_inven(self_obj, create_object(244, 1, 2));
         end
         if (random(0, 100) <= 5) then begin
-            add_obj_to_inven(self_obj, create_object_sid(428, 1, 2, -1));
+            add_obj_to_inven(self_obj, create_object(428, 1, 2));
         end
         if (random(0, 100) <= 5) then begin
-            add_obj_to_inven(self_obj, create_object_sid(422, 1, 2, -1));
+            add_obj_to_inven(self_obj, create_object(422, 1, 2));
         end
         if (random(0, 100) <= 4) then begin
-            add_obj_to_inven(self_obj, create_object_sid(423, 1, 2, -1));
+            add_obj_to_inven(self_obj, create_object(423, 1, 2));
         end
         if (random(0, 100) <= 3) then begin
-            add_obj_to_inven(self_obj, create_object_sid(426, 1, 2, -1));
+            add_obj_to_inven(self_obj, create_object(426, 1, 2));
         end
         if (random(0, 100) <= 2) then begin
-            add_obj_to_inven(self_obj, create_object_sid(433, 1, 2, -1));
+            add_obj_to_inven(self_obj, create_object(433, 1, 2));
         end
         if (random(0, 100) <= 2) then begin
-            add_obj_to_inven(self_obj, create_object_sid(421, 1, 2, -1));
+            add_obj_to_inven(self_obj, create_object(421, 1, 2));
         end
         if (random(0, 100) <= 40) then begin
-            add_obj_to_inven(self_obj, create_object_sid(164, 1, 2, -1));
-            add_mult_objs_to_inven(self_obj, create_object_sid(2, 1, 2, -1), random(21, 38));
+            add_obj_to_inven(self_obj, create_object(164, 1, 2));
+            add_mult_objs_to_inven(self_obj, create_object(2, 1, 2), random(21, 38));
         end
         else begin
             if (random(0, 100) <= 40) then begin
-                add_obj_to_inven(self_obj, create_object_sid(150, 1, 2, -1));
+                add_obj_to_inven(self_obj, create_object(150, 1, 2));
             end
             else begin
-                add_obj_to_inven(self_obj, create_object_sid(159, 1, 2, -1));
+                add_obj_to_inven(self_obj, create_object(159, 1, 2));
             end
-            add_mult_objs_to_inven(self_obj, create_object_sid(22, 1, 2, -1), random(2, 4));
+            add_mult_objs_to_inven(self_obj, create_object(22, 1, 2), random(2, 4));
         end
         if (random(0, 100) <= 33) then begin
-            add_obj_to_inven(self_obj, create_object_sid(112, 1, 2, -1));
+            add_obj_to_inven(self_obj, create_object(112, 1, 2));
         end
         else begin
-            add_obj_to_inven(self_obj, create_object_sid(109, 1, 2, -1));
+            add_obj_to_inven(self_obj, create_object(109, 1, 2));
         end
         if (random(0, 100) <= 33) then begin
-            add_mult_objs_to_inven(self_obj, create_object_sid(41, 1, 2, -1), random(5, 20));
+            add_mult_objs_to_inven(self_obj, create_object(41, 1, 2), random(5, 20));
         end
         if (random(0, 100) <= 25) then begin
-            add_mult_objs_to_inven(self_obj, create_object_sid(411, 1, 2, -1), random(3, 7));
+            add_mult_objs_to_inven(self_obj, create_object(411, 1, 2), random(3, 7));
         end
         if (random(0, 100) <= 10) then begin
-            add_obj_to_inven(self_obj, create_object_sid(233, 1, 2, -1));
+            add_obj_to_inven(self_obj, create_object(233, 1, 2));
         end
         if (random(0, 100) <= 9) then begin
-            add_obj_to_inven(self_obj, create_object_sid(243, 1, 2, -1));
+            add_obj_to_inven(self_obj, create_object(243, 1, 2));
         end
         if (random(0, 100) <= 8) then begin
-            add_obj_to_inven(self_obj, create_object_sid(237, 1, 2, -1));
+            add_obj_to_inven(self_obj, create_object(237, 1, 2));
         end
         if (random(0, 100) <= 7) then begin
-            add_obj_to_inven(self_obj, create_object_sid(234, 1, 2, -1));
+            add_obj_to_inven(self_obj, create_object(234, 1, 2));
         end
         if (random(0, 100) <= 6) then begin
-            add_obj_to_inven(self_obj, create_object_sid(232, 1, 2, -1));
+            add_obj_to_inven(self_obj, create_object(232, 1, 2));
         end
     end
 end
@@ -166,10 +166,10 @@ begin
     end
     if (float_is_called == 0) then begin
         float_is_called := 1;
-        add_timer_event(self_obj, random(1000, 1000), 2);
+        add_timer_event(self_obj, 1000, 2);
     end
     if (obj_can_see_obj(self_obj, dude_obj)) then begin
-        if ((local_var(2) == 1) or (global_var(54) <= -31)) then begin
+        if ((local_var(2) == 1) or (global_var(GVAR_TOWN_REP_BROKEN_HILLS) <= -31)) then begin
             set_local_var(2, 0);
             attack_setup(self_obj, dude_obj);
         end
@@ -184,10 +184,10 @@ procedure description_p_proc
 begin
     script_overrides;
     if (local_var(7) == 0) then begin
-        display_msg(message_str(254, 2));
+        display_mstr(2);
     end
     else begin
-        display_msg(message_str(254, 4));
+        display_mstr(4);
     end
 end
 
@@ -195,10 +195,10 @@ procedure look_at_p_proc
 begin
     script_overrides;
     if (local_var(7) == 0) then begin
-        display_msg(message_str(254, 1));
+        display_mstr(1);
     end
     else begin
-        display_msg(message_str(254, 3));
+        display_mstr(3);
     end
 end
 
@@ -217,11 +217,11 @@ end
 procedure talk_p_proc
 begin
     if (not(anim_busy(self_obj))) then begin
-        anim(self_obj, 1000, rotation_to_tile(tile_num(self_obj), tile_num(dude_obj)));
+        anim(self_obj, 1000, rotation_to_tile(tile_num(self_obj), dude_tile));
     end
     start_gdialog(254, self_obj, -1, -1, -1);
     gsay_start;
-    if ((local_var(13) == 0) and not((global_var(118) bwand 67108864) != 0)) then begin
+    if ((local_var(13) == 0) and not((global_var(GVAR_MODOC_TOOL_FLAG) bwand 67108864) != 0)) then begin
         set_local_var(13, 1);
         call Node023();
     end
@@ -239,30 +239,30 @@ end
 
 procedure destroy_p_proc
 begin
-    set_global_var(118, global_var(118) bwor 134217728);
-    if ((global_var(172) == 1) or (global_var(172) == 2)) then begin
-        set_global_var(172, 0);
+    set_global_var(GVAR_MODOC_TOOL_FLAG, global_var(GVAR_MODOC_TOOL_FLAG) bwor 134217728);
+    if ((global_var(GVAR_NCR_V15_DARION_DEAD) == 1) or (global_var(GVAR_NCR_V15_DARION_DEAD) == 2)) then begin
+        set_global_var(GVAR_NCR_V15_DARION_DEAD, 0);
     end
-    if (global_var(176) == 1) then begin
-        set_global_var(176, 0);
+    if (global_var(GVAR_VAULT13_CLEAR) == 1) then begin
+        set_global_var(GVAR_VAULT13_CLEAR, 0);
     end
-    if (global_var(171) == 1) then begin
-        set_global_var(171, 2);
-        give_exp_points(floor(250 / 5) + (get_critter_stat(dude_obj, 4) * floor(250 / 12)) + (get_critter_stat(dude_obj, 6) * floor(250 / 30)) + (floor(250 / 75) * (((get_critter_stat(dude_obj, 6) - 4) * 4) - (difficulty_level * 4))));
-        display_msg(message_str(10, random(140, 143)) + (floor(250 / 5) + (get_critter_stat(dude_obj, 4) * floor(250 / 12)) + (get_critter_stat(dude_obj, 6) * floor(250 / 30)) + (floor(250 / 75) * (((get_critter_stat(dude_obj, 6) - 4) * 4) - (difficulty_level * 4)))) + message_str(10, 144));
-        set_global_var(0, global_var(0) + 50);
+    if (global_var(GVAR_NCR_SQUAT_DEAL) == 1) then begin
+        set_global_var(GVAR_NCR_SQUAT_DEAL, 2);
+        give_exp_points(floor(250 / 5) + (dude_iq * floor(250 / 12)) + (dude_luck * floor(250 / 30)) + (floor(250 / 75) * (((dude_luck - 4) * 4) - (difficulty_level * 4))));
+        display_msg(message_str(SCRIPT_STTEXT, random(140, 143)) + (floor(250 / 5) + (dude_iq * floor(250 / 12)) + (dude_luck * floor(250 / 30)) + (floor(250 / 75) * (((dude_luck - 4) * 4) - (difficulty_level * 4)))) + message_str(SCRIPT_STTEXT, 144));
+        set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + 50);
     end
-    set_global_var(305, 1);
-    if (not((global_var(118) bwand 67108864) != 0)) then begin
+    set_global_var(GVAR_BH_MISSING, 1);
+    if (not((global_var(GVAR_MODOC_TOOL_FLAG) bwand 67108864) != 0)) then begin
         load_map("rbmain", -1);
     end
     if (source_obj == dude_obj) then begin
-        set_global_var(0, global_var(0) - 1);
+        set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) - 1);
         if ((54 == 50) or (54 == 59)) then begin
-            set_global_var(54, global_var(54) - 3);
+            set_global_var(GVAR_TOWN_REP_BROKEN_HILLS, global_var(GVAR_TOWN_REP_BROKEN_HILLS) - 3);
         end
         else begin
-            set_global_var(54, global_var(54) - 6);
+            set_global_var(GVAR_TOWN_REP_BROKEN_HILLS, global_var(GVAR_TOWN_REP_BROKEN_HILLS) - 6);
         end
     end
 end
@@ -275,51 +275,51 @@ begin
             set_sfall_global(13, 1);
         end
         else begin
-            if (has_trait(2, dude_obj, 13)) then begin
-                if (global_var(81) < get_pc_stat(1)) then begin
+            if (has_trait(TRAIT_TRAIT, dude_obj, TRAIT_sex_appeal)) then begin
+                if (global_var(GVAR_VAULT_CITIZENSHIP) < get_pc_stat(PCSTAT_level)) then begin
                     if (is_success(roll_vs_skill(dude_obj, 10, 0))) then begin
-                        set_global_var(98, global_var(98) + 3);
+                        set_global_var(GVAR_QUEST_FIND_THOMAS, global_var(GVAR_QUEST_FIND_THOMAS) + 3);
                     end
                     else begin
-                        set_global_var(98, global_var(98) + 1);
+                        set_global_var(GVAR_QUEST_FIND_THOMAS, global_var(GVAR_QUEST_FIND_THOMAS) + 1);
                     end
-                    if (global_var(98) >= (2 * get_pc_stat(1))) then begin
-                        set_global_var(98, 0);
-                        set_global_var(81, global_var(81) + 1);
-                        critter_mod_skill(dude_obj, 10, 1);
-                        display_msg(message_str(10, 410));
+                    if (global_var(GVAR_QUEST_FIND_THOMAS) >= (2 * get_pc_stat(PCSTAT_level))) then begin
+                        set_global_var(GVAR_QUEST_FIND_THOMAS, 0);
+                        set_global_var(GVAR_VAULT_CITIZENSHIP, global_var(GVAR_VAULT_CITIZENSHIP) + 1);
+                        critter_mod_skill(dude_obj, SKILL_STEAL, 1);
+                        display_msg(message_str(SCRIPT_STTEXT, 410));
                     end
                 end
             end
             steal_perk := 0;
             steal_gloves := 0;
-            if (has_trait(0, dude_obj, 37) != 0) then begin
+            if (has_trait(TRAIT_PERK, dude_obj, PERK_pickpocket) != 0) then begin
                 steal_perk := steal_perk + 15;
             end
-            if (has_trait(0, dude_obj, 33) != 0) then begin
+            if (has_trait(TRAIT_PERK, dude_obj, PERK_master_thief) != 0) then begin
                 steal_perk := steal_perk + 30;
             end
-            if ((obj_pid(critter_inven_obj(dude_obj, 1)) == 245) or (obj_pid(critter_inven_obj(dude_obj, 2)) == 245)) then begin
+            if ((obj_pid(critter_inven_obj(dude_obj, INVEN_TYPE_RIGHT_HAND)) == 245) or (obj_pid(critter_inven_obj(dude_obj, INVEN_TYPE_LEFT_HAND)) == 245)) then begin
                 steal_gloves := 10;
             end
-            if (random(0, 100) <= ((has_skill(dude_obj, 10) / 2) + (((3 * get_critter_stat(dude_obj, 5)) + (((get_critter_stat(dude_obj, 6) - 4) * 4) - (difficulty_level * 4)) + (2 * steal_perk) + steal_gloves) / (local_var(4) / 25)))) then begin
-                display_msg(message_str(10, random(145, 148)));
+            if (random(0, 100) <= ((dude_skill(SKILL_STEAL) / 2) + (((3 * get_critter_stat(dude_obj, STAT_ag)) + (((dude_luck - 4) * 4) - (difficulty_level * 4)) + (2 * steal_perk) + steal_gloves) / (local_var(4) / 25)))) then begin
+                display_msg(message_str(SCRIPT_STTEXT, random(145, 148)));
             end
             else begin
                 script_overrides;
-                if (random(0, 100) <= (25 - floor((((get_critter_stat(dude_obj, 6) - 4) * 4) - (difficulty_level * 4)) / 3))) then begin
-                    float_msg(self_obj, message_str(10, random(206, 209)), 2);
+                if (random(0, 100) <= (25 - floor((((dude_luck - 4) * 4) - (difficulty_level * 4)) / 3))) then begin
+                    float_msg(self_obj, message_str(SCRIPT_STTEXT, random(206, 209)), FLOAT_MSG_RED);
                     set_local_var(2, 1);
                 end
                 else begin
-                    display_msg(message_str(10, random(200, 203)));
+                    display_msg(message_str(SCRIPT_STTEXT, random(200, 203)));
                 end
             end
         end
     end
     if (action_being_used == 12) then begin
         script_overrides;
-        display_msg(message_str(10, 204) + ((has_skill(dude_obj, 10) / 2) + (((3 * get_critter_stat(dude_obj, 5)) + (((get_critter_stat(dude_obj, 6) - 4) * 4) - (difficulty_level * 4)) + (2 * steal_perk) + steal_gloves) / (local_var(4) / 25))) + message_str(10, 205));
+        display_msg(message_str(SCRIPT_STTEXT, 204) + ((dude_skill(SKILL_STEAL) / 2) + (((3 * get_critter_stat(dude_obj, STAT_ag)) + (((dude_luck - 4) * 4) - (difficulty_level * 4)) + (2 * steal_perk) + steal_gloves) / (local_var(4) / 25))) + message_str(SCRIPT_STTEXT, 205));
     end
 end
 
@@ -336,8 +336,8 @@ end
 
 procedure map_enter_p_proc
 begin
-    if (not(metarule(22, 0)) == 1) then begin
-        if (global_var(249) == 1) then begin
+    if (not(is_loading_game) == 1) then begin
+        if (global_var(GVAR_GECKO_BRAIN_DEAD) == 1) then begin
             set_local_var(7, 1);
         end
         if (tile_num(self_obj) != local_var(3)) then begin
@@ -348,8 +348,8 @@ begin
             set_local_var(11, 0);
             set_local_var(12, 0);
         end
-        if (metarule(14, 0)) then begin
-            if (not((global_var(118) bwand 67108864) != 0)) then begin
+        if (map_first_run) then begin
+            if (not((global_var(GVAR_MODOC_TOOL_FLAG) bwand 67108864) != 0)) then begin
                 move_to(self_obj, 23733, elevation(self_obj));
                 move_to(dude_obj, 23728, elevation(self_obj));
                 anim(self_obj, 1000, 2);
@@ -357,7 +357,7 @@ begin
                 add_timer_event(self_obj, 8, 10);
             end
             else begin
-                WHORE := tile_contains_pid_obj(24933, elevation(dude_obj), 16777366);
+                WHORE := tile_contains_pid_obj(24933, dude_elevation, 16777366);
                 move_to(WHORE, 2, 2);
             end
         end
@@ -366,7 +366,7 @@ end
 
 procedure timed_event_p_proc
 begin
-    WHORE := tile_contains_pid_obj(24933, elevation(dude_obj), 16777366);
+    WHORE := tile_contains_pid_obj(24933, dude_elevation, 16777366);
     if ((fixed_param == 1) and not(combat_is_initialized)) then begin
         if (tile_num(self_obj) == local_var(3)) then begin
             anim(self_obj, 1000, local_var(9));
@@ -384,8 +384,8 @@ begin
         reg_anim_func(3, 0);
         fadeOut(2000);
         add_timer_event(self_obj, 1, 4);
-        move_to(dude_obj, 19518, elevation(dude_obj));
-        set_global_var(282, global_var(282) + 1);
+        move_to(dude_obj, 19518, dude_elevation);
+        set_global_var(GVAR_CARAVAN_MASTERS_TOTAL, global_var(GVAR_CARAVAN_MASTERS_TOTAL) + 1);
         set_local_var(10, local_var(10) + 1);
     end
     if (fixed_param == 4) then begin
@@ -441,16 +441,16 @@ begin
         if (WHORE != 0) then begin
             move_to(WHORE, 3, 2);
         end
-        set_global_var(305, 1);
-        set_global_var(118, global_var(118) bwor 134217728);
-        if ((global_var(172) == 1) or (global_var(172) == 2)) then begin
-            set_global_var(172, 0);
+        set_global_var(GVAR_BH_MISSING, 1);
+        set_global_var(GVAR_MODOC_TOOL_FLAG, global_var(GVAR_MODOC_TOOL_FLAG) bwor 134217728);
+        if ((global_var(GVAR_NCR_V15_DARION_DEAD) == 1) or (global_var(GVAR_NCR_V15_DARION_DEAD) == 2)) then begin
+            set_global_var(GVAR_NCR_V15_DARION_DEAD, 0);
         end
-        if (global_var(176) == 1) then begin
-            set_global_var(176, 0);
+        if (global_var(GVAR_VAULT13_CLEAR) == 1) then begin
+            set_global_var(GVAR_VAULT13_CLEAR, 0);
         end
         play_sfx("LEVELUP");
-        display_msg(message_str(254, 285));
+        display_mstr(285);
         load_map("rbmain", -1);
     end
     if (fixed_param == 9) then begin
@@ -467,7 +467,7 @@ begin
     end
     if (fixed_param == 11) then begin
         fadeOut(1000);
-        set_global_var(176, 1);
+        set_global_var(GVAR_VAULT13_CLEAR, 1);
         fadeIn(1000);
     end
 end
@@ -476,89 +476,89 @@ procedure Node001
 begin
     set_local_var(7, 1);
     already_talk := 1;
-    gsay_reply(254, 10);
-    gsay_option(254, 11, Node015, 50);
-    gsay_option(254, 12, Node011, 50);
-    gsay_option(254, 13, Node016, 50);
-    gsay_option(254, 14, Node999, 50);
+    Reply(10);
+    giq_option(1, NAME, 11, Node015, NEUTRAL_REACTION);
+    giq_option(1, NAME, 12, Node011, NEUTRAL_REACTION);
+    giq_option(1, NAME, 13, Node016, NEUTRAL_REACTION);
+    giq_option(1, NAME, 14, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node010
 begin
-    if (global_var(249) == 1) then begin
-        gsay_reply(254, 1003);
-        gsay_option(254, 1012, Node021, 50);
-        if ((global_var(176) == 0) or ((global_var(172) == 0) and (not(global_var(119) bwand 8) != 0)) or ((global_var(176) == 2) and ((global_var(172) >= 2) or ((global_var(119) bwand 8) != 0)))) then begin
-            gsay_option(254, 1013, Node016, 50);
+    if (global_var(GVAR_GECKO_BRAIN_DEAD) == 1) then begin
+        Reply(1003);
+        giq_option(1, NAME, 1012, Node021, NEUTRAL_REACTION);
+        if ((global_var(GVAR_VAULT13_CLEAR) == 0) or ((global_var(GVAR_NCR_V15_DARION_DEAD) == 0) and (not(global_var(GVAR_MODOC_SLAUGHTER_BESS_TIME) bwand 8) != 0)) or ((global_var(GVAR_VAULT13_CLEAR) == 2) and ((global_var(GVAR_NCR_V15_DARION_DEAD) >= 2) or ((global_var(GVAR_MODOC_SLAUGHTER_BESS_TIME) bwand 8) != 0)))) then begin
+            giq_option(1, NAME, 1013, Node016, NEUTRAL_REACTION);
         end
-        gsay_option(254, 1014, Node022, 50);
-        gsay_option(254, 1015, Node015, 50);
-        if ((global_var(303) >= 4) and (global_var(176) == 1)) then begin
-            gsay_option(254, 1016, Node017, 50);
+        giq_option(1, NAME, 1014, Node022, NEUTRAL_REACTION);
+        giq_option(1, NAME, 1015, Node015, NEUTRAL_REACTION);
+        if ((global_var(GVAR_BH_JAIL) >= 4) and (global_var(GVAR_VAULT13_CLEAR) == 1)) then begin
+            giq_option(1, NAME, 1016, Node017, NEUTRAL_REACTION);
         end
-        if (global_var(172) == 2) then begin
-            gsay_option(254, 1017, Node018, 50);
+        if (global_var(GVAR_NCR_V15_DARION_DEAD) == 2) then begin
+            giq_option(1, NAME, 1017, Node018, NEUTRAL_REACTION);
         end
-        gsay_option(254, 1018, Node999, 50);
+        giq_option(1, NAME, 1018, Node999, NEUTRAL_REACTION);
     end
     else begin
-        if (global_var(249) == 2) then begin
-            gsay_reply(254, 1004);
-            gsay_option(254, 1019, Node021, 50);
-            if ((global_var(176) == 0) or ((global_var(172) == 0) and (not(global_var(119) bwand 8) != 0)) or ((global_var(176) == 2) and ((global_var(172) >= 2) or ((global_var(119) bwand 8) != 0)))) then begin
-                gsay_option(254, 1020, Node016, 50);
+        if (global_var(GVAR_GECKO_BRAIN_DEAD) == 2) then begin
+            Reply(1004);
+            giq_option(1, NAME, 1019, Node021, NEUTRAL_REACTION);
+            if ((global_var(GVAR_VAULT13_CLEAR) == 0) or ((global_var(GVAR_NCR_V15_DARION_DEAD) == 0) and (not(global_var(GVAR_MODOC_SLAUGHTER_BESS_TIME) bwand 8) != 0)) or ((global_var(GVAR_VAULT13_CLEAR) == 2) and ((global_var(GVAR_NCR_V15_DARION_DEAD) >= 2) or ((global_var(GVAR_MODOC_SLAUGHTER_BESS_TIME) bwand 8) != 0)))) then begin
+                giq_option(1, NAME, 1020, Node016, NEUTRAL_REACTION);
             end
-            gsay_option(254, 1021, Node015, 50);
-            if (global_var(172) == 2) then begin
-                gsay_option(254, 1022, Node018, 50);
+            giq_option(1, NAME, 1021, Node015, NEUTRAL_REACTION);
+            if (global_var(GVAR_NCR_V15_DARION_DEAD) == 2) then begin
+                giq_option(1, NAME, 1022, Node018, NEUTRAL_REACTION);
             end
-            if ((global_var(303) >= 4) and (global_var(176) == 1)) then begin
-                gsay_option(254, 1023, Node017, 50);
+            if ((global_var(GVAR_BH_JAIL) >= 4) and (global_var(GVAR_VAULT13_CLEAR) == 1)) then begin
+                giq_option(1, NAME, 1023, Node017, NEUTRAL_REACTION);
             end
-            gsay_option(254, 1024, Node999, 50);
+            giq_option(1, NAME, 1024, Node999, NEUTRAL_REACTION);
         end
         else begin
-            if (global_var(248) == 1) then begin
-                gsay_reply(254, 1005);
-                if (global_var(172) == 2) then begin
-                    gsay_option(254, 1025, Node018, 50);
+            if (global_var(GVAR_NEW_RENO_MCGEE_ATTACKED) == 1) then begin
+                Reply(1005);
+                if (global_var(GVAR_NCR_V15_DARION_DEAD) == 2) then begin
+                    giq_option(1, NAME, 1025, Node018, NEUTRAL_REACTION);
                 end
-                if ((global_var(176) == 0) or ((global_var(172) == 0) and (not(global_var(119) bwand 8) != 0)) or ((global_var(176) == 2) and ((global_var(172) >= 2) or ((global_var(119) bwand 8) != 0)))) then begin
-                    gsay_option(254, 1026, Node016, 50);
+                if ((global_var(GVAR_VAULT13_CLEAR) == 0) or ((global_var(GVAR_NCR_V15_DARION_DEAD) == 0) and (not(global_var(GVAR_MODOC_SLAUGHTER_BESS_TIME) bwand 8) != 0)) or ((global_var(GVAR_VAULT13_CLEAR) == 2) and ((global_var(GVAR_NCR_V15_DARION_DEAD) >= 2) or ((global_var(GVAR_MODOC_SLAUGHTER_BESS_TIME) bwand 8) != 0)))) then begin
+                    giq_option(1, NAME, 1026, Node016, NEUTRAL_REACTION);
                 end
-                if ((global_var(303) >= 4) and (global_var(176) == 1)) then begin
-                    gsay_option(254, 1027, Node017, 50);
+                if ((global_var(GVAR_BH_JAIL) >= 4) and (global_var(GVAR_VAULT13_CLEAR) == 1)) then begin
+                    giq_option(1, NAME, 1027, Node017, NEUTRAL_REACTION);
                 end
-                gsay_option(254, 1028, Node015, 50);
-                gsay_option(254, 1029, Node999, 50);
+                giq_option(1, NAME, 1028, Node015, NEUTRAL_REACTION);
+                giq_option(1, NAME, 1029, Node999, NEUTRAL_REACTION);
             end
             else begin
-                if ((global_var(302) >= 3) or (global_var(176) == 2)) then begin
-                    gsay_reply(254, 1002);
-                    gsay_option(254, 107, Node015, 50);
-                    gsay_option(254, 108, Node011, 50);
-                    if (global_var(172) == 2) then begin
-                        gsay_option(254, 109, Node018, 50);
+                if ((global_var(GVAR_BH_MINE) >= 3) or (global_var(GVAR_VAULT13_CLEAR) == 2)) then begin
+                    Reply(1002);
+                    giq_option(1, NAME, 107, Node015, NEUTRAL_REACTION);
+                    giq_option(1, NAME, 108, Node011, NEUTRAL_REACTION);
+                    if (global_var(GVAR_NCR_V15_DARION_DEAD) == 2) then begin
+                        giq_option(1, NAME, 109, Node018, NEUTRAL_REACTION);
                     end
-                    if ((global_var(176) == 0) or ((global_var(172) == 0) and (not(global_var(119) bwand 8) != 0)) or ((global_var(176) == 2) and ((global_var(172) >= 2) or ((global_var(119) bwand 8) != 0)))) then begin
-                        gsay_option(254, 1010, Node016, 50);
+                    if ((global_var(GVAR_VAULT13_CLEAR) == 0) or ((global_var(GVAR_NCR_V15_DARION_DEAD) == 0) and (not(global_var(GVAR_MODOC_SLAUGHTER_BESS_TIME) bwand 8) != 0)) or ((global_var(GVAR_VAULT13_CLEAR) == 2) and ((global_var(GVAR_NCR_V15_DARION_DEAD) >= 2) or ((global_var(GVAR_MODOC_SLAUGHTER_BESS_TIME) bwand 8) != 0)))) then begin
+                        giq_option(1, NAME, 1010, Node016, NEUTRAL_REACTION);
                     end
-                    gsay_option(254, 1011, Node999, 50);
+                    giq_option(1, NAME, 1011, Node999, NEUTRAL_REACTION);
                 end
                 else begin
-                    gsay_reply(254, 1001);
-                    gsay_option(254, 101, Node011, 50);
-                    gsay_option(254, 102, Node015, 50);
-                    if (global_var(172) == 2) then begin
-                        gsay_option(254, 103, Node018, 50);
+                    Reply(1001);
+                    giq_option(1, NAME, 101, Node011, NEUTRAL_REACTION);
+                    giq_option(1, NAME, 102, Node015, NEUTRAL_REACTION);
+                    if (global_var(GVAR_NCR_V15_DARION_DEAD) == 2) then begin
+                        giq_option(1, NAME, 103, Node018, NEUTRAL_REACTION);
                     end
-                    if ((global_var(303) >= 4) and (global_var(176) == 1)) then begin
-                        gsay_option(254, 104, Node017, 50);
+                    if ((global_var(GVAR_BH_JAIL) >= 4) and (global_var(GVAR_VAULT13_CLEAR) == 1)) then begin
+                        giq_option(1, NAME, 104, Node017, NEUTRAL_REACTION);
                     end
-                    if ((global_var(176) == 0) or ((global_var(172) == 0) and (not(global_var(119) bwand 8) != 0)) or ((global_var(176) == 2) and ((global_var(172) >= 2) or ((global_var(119) bwand 8) != 0)))) then begin
-                        gsay_option(254, 105, Node016, 50);
+                    if ((global_var(GVAR_VAULT13_CLEAR) == 0) or ((global_var(GVAR_NCR_V15_DARION_DEAD) == 0) and (not(global_var(GVAR_MODOC_SLAUGHTER_BESS_TIME) bwand 8) != 0)) or ((global_var(GVAR_VAULT13_CLEAR) == 2) and ((global_var(GVAR_NCR_V15_DARION_DEAD) >= 2) or ((global_var(GVAR_MODOC_SLAUGHTER_BESS_TIME) bwand 8) != 0)))) then begin
+                        giq_option(1, NAME, 105, Node016, NEUTRAL_REACTION);
                     end
-                    gsay_option(254, 106, Node999, 50);
+                    giq_option(1, NAME, 106, Node999, NEUTRAL_REACTION);
                 end
             end
         end
@@ -567,160 +567,160 @@ end
 
 procedure Node010a
 begin
-    if (global_var(172) == 0) then begin
-        set_global_var(172, 1);
+    if (global_var(GVAR_NCR_V15_DARION_DEAD) == 0) then begin
+        set_global_var(GVAR_NCR_V15_DARION_DEAD, 1);
     end
 end
 
 procedure Node011
 begin
-    gsay_reply(254, 110);
-    gsay_option(254, 111, Node019, 50);
+    Reply(110);
+    giq_option(1, NAME, 111, Node019, NEUTRAL_REACTION);
     if (item_caps_total(dude_obj) >= 10) then begin
-        gsay_option(254, 112, Node012, 50);
+        giq_option(1, NAME, 112, Node012, NEUTRAL_REACTION);
     end
     if (item_caps_total(dude_obj) >= 10) then begin
-        gsay_option(254, 113, Node013, 50);
+        giq_option(1, NAME, 113, Node013, NEUTRAL_REACTION);
     end
-    gsay_option(254, 114, Node020, 50);
-    gsay_option(254, 115, Node999, 50);
+    giq_option(1, NAME, 114, Node020, NEUTRAL_REACTION);
+    giq_option(1, NAME, 115, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node012
 begin
     item_caps_adjust(dude_obj, -10);
-    gsay_reply(254, 120);
+    Reply(120);
     if (item_caps_total(dude_obj) >= 10) then begin
-        gsay_option(254, 121, Node013, 50);
+        giq_option(1, NAME, 121, Node013, NEUTRAL_REACTION);
     end
-    gsay_option(254, 122, Node999, 50);
+    giq_option(1, NAME, 122, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node013
 begin
     item_caps_adjust(dude_obj, -10);
-    gsay_reply(254, 130);
+    Reply(130);
     if (item_caps_total(dude_obj) >= 10) then begin
-        gsay_option(254, 131, Node014, 50);
+        giq_option(1, NAME, 131, Node014, NEUTRAL_REACTION);
     end
     if (item_caps_total(dude_obj) >= 10) then begin
-        gsay_option(254, 132, Node012, 50);
+        giq_option(1, NAME, 132, Node012, NEUTRAL_REACTION);
     end
-    gsay_option(254, 133, Node999, 50);
+    giq_option(1, NAME, 133, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node014
 begin
     item_caps_adjust(dude_obj, -10);
-    if (global_var(304) == 0) then begin
-        set_global_var(304, 1);
+    if (global_var(GVAR_BH_CONSPIRACY) == 0) then begin
+        set_global_var(GVAR_BH_CONSPIRACY, 1);
     end
-    gsay_reply(254, 140);
-    gsay_option(254, 141, Node999, 50);
+    Reply(140);
+    giq_option(1, NAME, 141, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node015
 begin
-    if (global_var(249) == 1) then begin
-        gsay_reply(254, 1503);
-        gsay_option(254, 155, NodeSex, 50);
+    if (global_var(GVAR_GECKO_BRAIN_DEAD) == 1) then begin
+        Reply(1503);
+        giq_option(1, NAME, 155, NodeSex, NEUTRAL_REACTION);
     end
     else begin
         if (local_var(10) == 0) then begin
-            gsay_reply(254, 1501);
+            Reply(1501);
             if (item_caps_total(dude_obj) >= 27) then begin
-                gsay_option(254, 151, NodeSex, 50);
+                giq_option(1, NAME, 151, NodeSex, NEUTRAL_REACTION);
             end
-            gsay_option(254, 152, Node999, 50);
+            giq_option(1, NAME, 152, Node999, NEUTRAL_REACTION);
         end
         else begin
-            gsay_reply(254, 1502);
+            Reply(1502);
             if (item_caps_total(dude_obj) >= 27) then begin
-                gsay_option(254, 153, NodeSex, 50);
+                giq_option(1, NAME, 153, NodeSex, NEUTRAL_REACTION);
             end
-            gsay_option(254, 154, Node999, 50);
+            giq_option(1, NAME, 154, Node999, NEUTRAL_REACTION);
         end
     end
 end
 
 procedure Node016
 begin
-    if ((global_var(176) == 2) and ((global_var(172) >= 2) or ((global_var(119) bwand 8) != 0))) then begin
-        if (global_var(249) == 1) then begin
-            gsay_reply(254, 1605);
-            gsay_option(254, 1612, Node999, 50);
+    if ((global_var(GVAR_VAULT13_CLEAR) == 2) and ((global_var(GVAR_NCR_V15_DARION_DEAD) >= 2) or ((global_var(GVAR_MODOC_SLAUGHTER_BESS_TIME) bwand 8) != 0))) then begin
+        if (global_var(GVAR_GECKO_BRAIN_DEAD) == 1) then begin
+            Reply(1605);
+            giq_option(1, NAME, 1612, Node999, NEUTRAL_REACTION);
         end
         else begin
-            gsay_reply(254, 1604);
-            gsay_option(254, 1611, Node999, 50);
+            Reply(1604);
+            giq_option(1, NAME, 1611, Node999, NEUTRAL_REACTION);
         end
     end
     else begin
-        if (global_var(285) == 1) then begin
-            if (global_var(249) == 1) then begin
-                if (global_var(176) == 0) then begin
-                    gsay_reply(254, 1603);
-                    gsay_option(254, 167, QA_facecontrol, 50);
-                    gsay_option(254, 168, Node999, 50);
+        if (global_var(GVAR_NEW_RENO_WRIGHT_FLAGS) == 1) then begin
+            if (global_var(GVAR_GECKO_BRAIN_DEAD) == 1) then begin
+                if (global_var(GVAR_VAULT13_CLEAR) == 0) then begin
+                    Reply(1603);
+                    giq_option(1, NAME, 167, QA_facecontrol, NEUTRAL_REACTION);
+                    giq_option(1, NAME, 168, Node999, NEUTRAL_REACTION);
                 end
                 else begin
-                    if (global_var(172) == 0) then begin
-                        gsay_reply(254, 1604);
-                        gsay_option(254, 169, QA_sexy_brick, 50);
-                        gsay_option(254, 1610, Node999, 50);
+                    if (global_var(GVAR_NCR_V15_DARION_DEAD) == 0) then begin
+                        Reply(1604);
+                        giq_option(1, NAME, 169, QA_sexy_brick, NEUTRAL_REACTION);
+                        giq_option(1, NAME, 1610, Node999, NEUTRAL_REACTION);
                     end
                 end
             end
             else begin
-                if (global_var(248) == 1) then begin
-                    if (global_var(176) == 0) then begin
-                        gsay_reply(254, 1606);
-                        gsay_option(254, 1613, QA_facecontrol, 50);
-                        gsay_option(254, 1614, Node999, 50);
+                if (global_var(GVAR_NEW_RENO_MCGEE_ATTACKED) == 1) then begin
+                    if (global_var(GVAR_VAULT13_CLEAR) == 0) then begin
+                        Reply(1606);
+                        giq_option(1, NAME, 1613, QA_facecontrol, NEUTRAL_REACTION);
+                        giq_option(1, NAME, 1614, Node999, NEUTRAL_REACTION);
                     end
                     else begin
-                        if (global_var(172) == 0) then begin
-                            gsay_reply(254, 1607);
-                            gsay_option(254, 1615, QA_sexy_brick, 50);
-                            gsay_option(254, 1616, Node999, 50);
+                        if (global_var(GVAR_NCR_V15_DARION_DEAD) == 0) then begin
+                            Reply(1607);
+                            giq_option(1, NAME, 1615, QA_sexy_brick, NEUTRAL_REACTION);
+                            giq_option(1, NAME, 1616, Node999, NEUTRAL_REACTION);
                         end
                     end
                 end
                 else begin
-                    if (global_var(176) == 0) then begin
-                        gsay_reply(254, 1601);
-                        gsay_option(254, 161, QA_facecontrol, 50);
-                        gsay_option(254, 162, Node011, 50);
-                        gsay_option(254, 163, Node999, 50);
+                    if (global_var(GVAR_VAULT13_CLEAR) == 0) then begin
+                        Reply(1601);
+                        giq_option(1, NAME, 161, QA_facecontrol, NEUTRAL_REACTION);
+                        giq_option(1, NAME, 162, Node011, NEUTRAL_REACTION);
+                        giq_option(1, NAME, 163, Node999, NEUTRAL_REACTION);
                     end
                     else begin
-                        if (global_var(172) == 0) then begin
-                            gsay_reply(254, 1602);
-                            gsay_option(254, 164, QA_sexy_brick, 50);
-                            gsay_option(254, 166, Node999, 50);
+                        if (global_var(GVAR_NCR_V15_DARION_DEAD) == 0) then begin
+                            Reply(1602);
+                            giq_option(1, NAME, 164, QA_sexy_brick, NEUTRAL_REACTION);
+                            giq_option(1, NAME, 166, Node999, NEUTRAL_REACTION);
                         end
                     end
                 end
             end
         end
         else begin
-            if (global_var(249) == 1) then begin
-                if (global_var(172) == 0) then begin
-                    gsay_reply(254, 1604);
-                    gsay_option(254, 169, QA_sexy_brick, 50);
-                    gsay_option(254, 1610, Node999, 50);
+            if (global_var(GVAR_GECKO_BRAIN_DEAD) == 1) then begin
+                if (global_var(GVAR_NCR_V15_DARION_DEAD) == 0) then begin
+                    Reply(1604);
+                    giq_option(1, NAME, 169, QA_sexy_brick, NEUTRAL_REACTION);
+                    giq_option(1, NAME, 1610, Node999, NEUTRAL_REACTION);
                 end
             end
             else begin
-                if (global_var(172) == 0) then begin
-                    gsay_reply(254, 1602);
-                    gsay_option(254, 164, Node010a, 50);
-                    gsay_option(254, 165, Node011, 50);
-                    gsay_option(254, 166, Node999, 50);
+                if (global_var(GVAR_NCR_V15_DARION_DEAD) == 0) then begin
+                    Reply(1602);
+                    giq_option(1, NAME, 164, Node010a, NEUTRAL_REACTION);
+                    giq_option(1, NAME, 165, Node011, NEUTRAL_REACTION);
+                    giq_option(1, NAME, 166, Node999, NEUTRAL_REACTION);
                 end
                 else begin
-                    gsay_reply(254, 1604);
-                    gsay_option(254, 1611, Node999, 50);
+                    Reply(1604);
+                    giq_option(1, NAME, 1611, Node999, NEUTRAL_REACTION);
                 end
             end
         end
@@ -729,89 +729,89 @@ end
 
 procedure Node017
 begin
-    if (global_var(176) == 1) then begin
-        set_global_var(176, 2);
-        give_exp_points(floor(1208 / 5) + (get_critter_stat(dude_obj, 4) * floor(1208 / 12)) + (get_critter_stat(dude_obj, 6) * floor(1208 / 30)) + (floor(1208 / 75) * (((get_critter_stat(dude_obj, 6) - 4) * 4) - (difficulty_level * 4))));
-        display_msg(message_str(10, random(140, 143)) + (floor(1208 / 5) + (get_critter_stat(dude_obj, 4) * floor(1208 / 12)) + (get_critter_stat(dude_obj, 6) * floor(1208 / 30)) + (floor(1208 / 75) * (((get_critter_stat(dude_obj, 6) - 4) * 4) - (difficulty_level * 4)))) + message_str(10, 144));
-        add_obj_to_inven(dude_obj, create_object_sid(142, 1, 2, -1));
+    if (global_var(GVAR_VAULT13_CLEAR) == 1) then begin
+        set_global_var(GVAR_VAULT13_CLEAR, 2);
+        give_exp_points(floor(1208 / 5) + (dude_iq * floor(1208 / 12)) + (dude_luck * floor(1208 / 30)) + (floor(1208 / 75) * (((dude_luck - 4) * 4) - (difficulty_level * 4))));
+        display_msg(message_str(SCRIPT_STTEXT, random(140, 143)) + (floor(1208 / 5) + (dude_iq * floor(1208 / 12)) + (dude_luck * floor(1208 / 30)) + (floor(1208 / 75) * (((dude_luck - 4) * 4) - (difficulty_level * 4)))) + message_str(SCRIPT_STTEXT, 144));
+        add_obj_to_inven(dude_obj, create_object(142, 1, 2));
     end
-    if (not(global_var(249)) == 1) then begin
-        gsay_reply(254, 1701);
-        gsay_option(254, 171, Node999, 50);
+    if (not(global_var(GVAR_GECKO_BRAIN_DEAD)) == 1) then begin
+        Reply(1701);
+        giq_option(1, NAME, 171, Node999, NEUTRAL_REACTION);
     end
     else begin
-        gsay_reply(254, 1702);
-        gsay_option(254, 172, Node999, 50);
+        Reply(1702);
+        giq_option(1, NAME, 172, Node999, NEUTRAL_REACTION);
     end
 end
 
 procedure Node018
 begin
-    if (global_var(172) == 2) then begin
-        set_global_var(172, 3);
-        give_exp_points(floor(1035 / 5) + (get_critter_stat(dude_obj, 4) * floor(1035 / 12)) + (get_critter_stat(dude_obj, 6) * floor(1035 / 30)) + (floor(1035 / 75) * (((get_critter_stat(dude_obj, 6) - 4) * 4) - (difficulty_level * 4))));
-        display_msg(message_str(10, random(140, 143)) + (floor(1035 / 5) + (get_critter_stat(dude_obj, 4) * floor(1035 / 12)) + (get_critter_stat(dude_obj, 6) * floor(1035 / 30)) + (floor(1035 / 75) * (((get_critter_stat(dude_obj, 6) - 4) * 4) - (difficulty_level * 4)))) + message_str(10, 144));
-        add_obj_to_inven(dude_obj, create_object_sid(222, 1, 2, -1));
-        add_obj_to_inven(dude_obj, create_object_sid(229, 1, 2, -1));
-        add_obj_to_inven(dude_obj, create_object_sid(233, 1, 2, -1));
-        add_obj_to_inven(dude_obj, create_object_sid(236, 1, 2, -1));
+    if (global_var(GVAR_NCR_V15_DARION_DEAD) == 2) then begin
+        set_global_var(GVAR_NCR_V15_DARION_DEAD, 3);
+        give_exp_points(floor(1035 / 5) + (dude_iq * floor(1035 / 12)) + (dude_luck * floor(1035 / 30)) + (floor(1035 / 75) * (((dude_luck - 4) * 4) - (difficulty_level * 4))));
+        display_msg(message_str(SCRIPT_STTEXT, random(140, 143)) + (floor(1035 / 5) + (dude_iq * floor(1035 / 12)) + (dude_luck * floor(1035 / 30)) + (floor(1035 / 75) * (((dude_luck - 4) * 4) - (difficulty_level * 4)))) + message_str(SCRIPT_STTEXT, 144));
+        add_obj_to_inven(dude_obj, create_object(222, 1, 2));
+        add_obj_to_inven(dude_obj, create_object(229, 1, 2));
+        add_obj_to_inven(dude_obj, create_object(233, 1, 2));
+        add_obj_to_inven(dude_obj, create_object(236, 1, 2));
     end
-    if (not(global_var(249)) == 1) then begin
-        gsay_reply(254, 1801);
-        gsay_option(254, 181, Node999, 50);
+    if (not(global_var(GVAR_GECKO_BRAIN_DEAD)) == 1) then begin
+        Reply(1801);
+        giq_option(1, NAME, 181, Node999, NEUTRAL_REACTION);
     end
     else begin
-        gsay_reply(254, 1802);
-        gsay_option(254, 182, Node999, 50);
+        Reply(1802);
+        giq_option(1, NAME, 182, Node999, NEUTRAL_REACTION);
     end
 end
 
 procedure Node019
 begin
-    gsay_reply(254, 190);
-    gsay_option(254, 191, Node020, 50);
+    Reply(190);
+    giq_option(1, NAME, 191, Node020, NEUTRAL_REACTION);
     if (item_caps_total(dude_obj) >= 10) then begin
-        gsay_option(254, 192, Node013, 50);
+        giq_option(1, NAME, 192, Node013, NEUTRAL_REACTION);
     end
     if (item_caps_total(dude_obj) >= 10) then begin
-        gsay_option(254, 193, Node012, 50);
+        giq_option(1, NAME, 193, Node012, NEUTRAL_REACTION);
     end
-    gsay_option(254, 194, Node999, 50);
+    giq_option(1, NAME, 194, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node020
 begin
-    gsay_reply(254, 200);
-    gsay_option(254, 201, Node998, 50);
-    gsay_option(254, 202, Node998, 50);
+    Reply(200);
+    giq_option(1, NAME, 201, Node998, NEUTRAL_REACTION);
+    giq_option(1, NAME, 202, Node998, NEUTRAL_REACTION);
 end
 
 procedure Node021
 begin
-    if (global_var(249) == 1) then begin
-        gsay_reply(254, 2101);
-        gsay_option(254, 211, Node022, 50);
+    if (global_var(GVAR_GECKO_BRAIN_DEAD) == 1) then begin
+        Reply(2101);
+        giq_option(1, NAME, 211, Node022, NEUTRAL_REACTION);
     end
     else begin
-        gsay_reply(254, 2102);
-        gsay_option(254, 212, Node999, 50);
+        Reply(2102);
+        giq_option(1, NAME, 212, Node999, NEUTRAL_REACTION);
     end
 end
 
 procedure Node022
 begin
     if ((get_day == 5) and (local_var(11) == 0)) then begin
-        gsay_reply(254, 2201);
-        gsay_option(254, 221, Node022a, 50);
+        Reply(2201);
+        giq_option(1, NAME, 221, Node022a, NEUTRAL_REACTION);
     end
     else begin
         if ((get_day == 20) and (local_var(12) == 0)) then begin
-            gsay_reply(254, 2201);
-            gsay_option(254, 221, Node022a, 50);
+            Reply(2201);
+            giq_option(1, NAME, 221, Node022a, NEUTRAL_REACTION);
         end
         else begin
-            gsay_reply(254, 2202);
-            gsay_option(254, 222, Node999, 50);
+            Reply(2202);
+            giq_option(1, NAME, 222, Node999, NEUTRAL_REACTION);
         end
     end
 end
@@ -830,82 +830,82 @@ end
 
 procedure Node023
 begin
-    set_global_var(171, 1);
+    set_global_var(GVAR_NCR_SQUAT_DEAL, 1);
     if (local_var(7) == 0) then begin
-        gsay_reply(254, 2301);
-        gsay_option(254, 231, Node024, 50);
+        Reply(2301);
+        giq_option(1, NAME, 231, Node024, NEUTRAL_REACTION);
     end
     else begin
-        gsay_reply(254, 2302);
-        gsay_option(254, 232, Node024, 50);
+        Reply(2302);
+        giq_option(1, NAME, 232, Node024, NEUTRAL_REACTION);
     end
 end
 
 procedure Node024
 begin
-    gsay_reply(254, 240);
-    gsay_option(254, 241, Node025, 50);
-    gsay_option(254, 242, Node026, 50);
-    gsay_option(254, 243, Node027, 50);
+    Reply(240);
+    giq_option(1, NAME, 241, Node025, NEUTRAL_REACTION);
+    giq_option(1, NAME, 242, Node026, NEUTRAL_REACTION);
+    giq_option(1, NAME, 243, Node027, NEUTRAL_REACTION);
 end
 
 procedure Node025
 begin
-    set_global_var(171, 2);
-    give_exp_points(floor(200 / 5) + (get_critter_stat(dude_obj, 4) * floor(200 / 12)) + (get_critter_stat(dude_obj, 6) * floor(200 / 30)) + (floor(200 / 75) * (((get_critter_stat(dude_obj, 6) - 4) * 4) - (difficulty_level * 4))));
-    display_msg(message_str(10, random(140, 143)) + (floor(200 / 5) + (get_critter_stat(dude_obj, 4) * floor(200 / 12)) + (get_critter_stat(dude_obj, 6) * floor(200 / 30)) + (floor(200 / 75) * (((get_critter_stat(dude_obj, 6) - 4) * 4) - (difficulty_level * 4)))) + message_str(10, 144));
-    set_global_var(0, global_var(0) - 50);
-    gsay_reply(254, 250);
-    gsay_option(254, 251, NodePoorWhore, 50);
+    set_global_var(GVAR_NCR_SQUAT_DEAL, 2);
+    give_exp_points(floor(200 / 5) + (dude_iq * floor(200 / 12)) + (dude_luck * floor(200 / 30)) + (floor(200 / 75) * (((dude_luck - 4) * 4) - (difficulty_level * 4))));
+    display_msg(message_str(SCRIPT_STTEXT, random(140, 143)) + (floor(200 / 5) + (dude_iq * floor(200 / 12)) + (dude_luck * floor(200 / 30)) + (floor(200 / 75) * (((dude_luck - 4) * 4) - (difficulty_level * 4)))) + message_str(SCRIPT_STTEXT, 144));
+    set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) - 50);
+    Reply(250);
+    giq_option(1, NAME, 251, NodePoorWhore, NEUTRAL_REACTION);
 end
 
 procedure Node026
 begin
-    gsay_reply(254, 260);
-    gsay_option(254, 262, Node025, 50);
-    gsay_option(254, 263, Node027, 50);
+    Reply(260);
+    giq_option(1, NAME, 262, Node025, NEUTRAL_REACTION);
+    giq_option(1, NAME, 263, Node027, NEUTRAL_REACTION);
 end
 
 procedure Node027
 begin
-    if (global_var(285) == 3) then begin
-        set_global_var(171, 2);
-        give_exp_points(floor(250 / 5) + (get_critter_stat(dude_obj, 4) * floor(250 / 12)) + (get_critter_stat(dude_obj, 6) * floor(250 / 30)) + (floor(250 / 75) * (((get_critter_stat(dude_obj, 6) - 4) * 4) - (difficulty_level * 4))));
-        display_msg(message_str(10, random(140, 143)) + (floor(250 / 5) + (get_critter_stat(dude_obj, 4) * floor(250 / 12)) + (get_critter_stat(dude_obj, 6) * floor(250 / 30)) + (floor(250 / 75) * (((get_critter_stat(dude_obj, 6) - 4) * 4) - (difficulty_level * 4)))) + message_str(10, 144));
-        set_global_var(0, global_var(0) + 50);
-        gsay_reply(254, 2701);
-        gsay_option(254, 271, NodeKillPastuh, 50);
+    if (global_var(GVAR_NEW_RENO_WRIGHT_FLAGS) == 3) then begin
+        set_global_var(GVAR_NCR_SQUAT_DEAL, 2);
+        give_exp_points(floor(250 / 5) + (dude_iq * floor(250 / 12)) + (dude_luck * floor(250 / 30)) + (floor(250 / 75) * (((dude_luck - 4) * 4) - (difficulty_level * 4))));
+        display_msg(message_str(SCRIPT_STTEXT, random(140, 143)) + (floor(250 / 5) + (dude_iq * floor(250 / 12)) + (dude_luck * floor(250 / 30)) + (floor(250 / 75) * (((dude_luck - 4) * 4) - (difficulty_level * 4)))) + message_str(SCRIPT_STTEXT, 144));
+        set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + 50);
+        Reply(2701);
+        giq_option(1, NAME, 271, NodeKillPastuh, NEUTRAL_REACTION);
     end
     else begin
-        critter_add_trait(self_obj, 1, 6, 23);
-        gsay_reply(254, 2702);
-        gsay_option(254, 272, Node998, 50);
-        gsay_option(254, 273, Node998, 50);
-        if ((get_critter_stat(dude_obj, 0) >= 7) and (get_critter_stat(dude_obj, 5) >= 7) and (has_skill(dude_obj, 3) >= 110)) then begin
-            gsay_option(254, 274, Node028, 50);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 23);
+        Reply(2702);
+        giq_option(1, NAME, 272, Node998, NEUTRAL_REACTION);
+        giq_option(1, NAME, 273, Node998, NEUTRAL_REACTION);
+        if ((dude_strength >= 7) and (get_critter_stat(dude_obj, STAT_ag) >= 7) and (dude_skill(SKILL_UNARMED_COMBAT) >= 110)) then begin
+            giq_option(1, NAME, 274, Node028, NEUTRAL_REACTION);
         end
     end
 end
 
 procedure Node028
 begin
-    set_global_var(171, 2);
-    give_exp_points(floor(250 / 5) + (get_critter_stat(dude_obj, 4) * floor(250 / 12)) + (get_critter_stat(dude_obj, 6) * floor(250 / 30)) + (floor(250 / 75) * (((get_critter_stat(dude_obj, 6) - 4) * 4) - (difficulty_level * 4))));
-    display_msg(message_str(10, random(140, 143)) + (floor(250 / 5) + (get_critter_stat(dude_obj, 4) * floor(250 / 12)) + (get_critter_stat(dude_obj, 6) * floor(250 / 30)) + (floor(250 / 75) * (((get_critter_stat(dude_obj, 6) - 4) * 4) - (difficulty_level * 4)))) + message_str(10, 144));
-    set_global_var(0, global_var(0) + 50);
-    gsay_reply(254, 280);
-    gsay_option(254, 281, NodeKillPastuh, 50);
+    set_global_var(GVAR_NCR_SQUAT_DEAL, 2);
+    give_exp_points(floor(250 / 5) + (dude_iq * floor(250 / 12)) + (dude_luck * floor(250 / 30)) + (floor(250 / 75) * (((dude_luck - 4) * 4) - (difficulty_level * 4))));
+    display_msg(message_str(SCRIPT_STTEXT, random(140, 143)) + (floor(250 / 5) + (dude_iq * floor(250 / 12)) + (dude_luck * floor(250 / 30)) + (floor(250 / 75) * (((dude_luck - 4) * 4) - (difficulty_level * 4)))) + message_str(SCRIPT_STTEXT, 144));
+    set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) + 50);
+    Reply(280);
+    giq_option(1, NAME, 281, NodeKillPastuh, NEUTRAL_REACTION);
 end
 
 procedure NodeSex
 begin
-    if (not(global_var(249)) == 1) then begin
+    if (not(global_var(GVAR_GECKO_BRAIN_DEAD)) == 1) then begin
         item_caps_adjust(dude_obj, -27);
     end
     item_caps_adjust(self_obj, 27);
     add_timer_event(self_obj, 8, 3);
-    if ((get_critter_stat(dude_obj, 6) <= 9) and (get_critter_stat(dude_obj, 3) <= 8)) then begin
-        add_obj_to_inven(self_obj, create_object_sid(444, 1, 2, -1));
+    if ((dude_luck <= 9) and (dude_charisma <= 8)) then begin
+        add_obj_to_inven(self_obj, create_object(444, 1, 2));
         use_obj_on_obj(obj_carrying_pid_obj(self_obj, 444), dude_obj);
     end
     game_ui_disable;
@@ -913,15 +913,15 @@ end
 
 procedure QA_facecontrol
 begin
-    if (global_var(176) == 0) then begin
+    if (global_var(GVAR_VAULT13_CLEAR) == 0) then begin
         add_timer_event(self_obj, 8, 11);
     end
 end
 
 procedure QA_sexy_brick
 begin
-    if (global_var(172) == 0) then begin
-        set_global_var(172, 1);
+    if (global_var(GVAR_NCR_V15_DARION_DEAD) == 0) then begin
+        set_global_var(GVAR_NCR_V15_DARION_DEAD, 1);
     end
 end
 
@@ -944,7 +944,7 @@ end
 
 procedure Node998
 begin
-    WHORE := tile_contains_pid_obj(tile_num_in_direction(tile_num(dude_obj), (has_trait(1, dude_obj, 10) + 3) % 6, 1), elevation(dude_obj), 16777366);
+    WHORE := tile_contains_pid_obj(tile_num_in_direction(dude_tile, (obj_get_rot(dude_obj) + 3) % 6, 1), dude_elevation, 16777366);
     if (WHORE != 0) then begin
         move_to(WHORE, 3, 2);
     end
@@ -957,4 +957,3 @@ begin
     gdialog_mod_barter(-(dude_objective_reputation + local_var(6)));
     call Node010();
 end
-
