diff --git a/Mods/Olympus2207/Original/SJELYBUK.ssl b/Mods/Olympus2207/ReDefine/SJELYBUK.ssl
index 6f01f55..222b3a3 100644
--- a/Mods/Olympus2207/Original/SJELYBUK.ssl
+++ b/Mods/Olympus2207/ReDefine/SJELYBUK.ssl
@@ -26,22 +26,22 @@ end
 procedure description_p_proc
 begin
     script_overrides;
-    if (global_var(280) == 0) then begin
-        display_msg(message_str(390, 2));
+    if (global_var(GVAR_CARAVAN_CARTS_TOTAL) == 0) then begin
+        display_mstr(2);
     end
     else begin
-        display_msg(message_str(390, 4));
+        display_mstr(4);
     end
 end
 
 procedure look_at_p_proc
 begin
     script_overrides;
-    if (global_var(280) == 0) then begin
-        display_msg(message_str(390, 1));
+    if (global_var(GVAR_CARAVAN_CARTS_TOTAL) == 0) then begin
+        display_mstr(1);
     end
     else begin
-        display_msg(message_str(390, 3));
+        display_mstr(3);
     end
 end
 
@@ -51,16 +51,16 @@ end
 
 procedure use_p_proc
 begin
-    if (global_var(280) == 0) then begin
-        if (get_critter_stat(dude_obj, 4) >= 9) then begin
-            display_msg(message_str(390, 5));
-            set_global_var(280, 1);
-            give_exp_points(floor(100 / 5) + (get_critter_stat(dude_obj, 4) * floor(100 / 12)) + (get_critter_stat(dude_obj, 6) * floor(100 / 30)) + (floor(100 / 75) * (((get_critter_stat(dude_obj, 6) - 4) * 4) - (difficulty_level * 4))));
-            display_msg(message_str(10, random(140, 143)) + (floor(100 / 5) + (get_critter_stat(dude_obj, 4) * floor(100 / 12)) + (get_critter_stat(dude_obj, 6) * floor(100 / 30)) + (floor(100 / 75) * (((get_critter_stat(dude_obj, 6) - 4) * 4) - (difficulty_level * 4)))) + message_str(10, 144));
+    if (global_var(GVAR_CARAVAN_CARTS_TOTAL) == 0) then begin
+        if (dude_iq >= 9) then begin
+            display_mstr(5);
+            set_global_var(GVAR_CARAVAN_CARTS_TOTAL, 1);
+            give_exp_points(floor(100 / 5) + (dude_iq * floor(100 / 12)) + (dude_luck * floor(100 / 30)) + (floor(100 / 75) * (((dude_luck - 4) * 4) - (difficulty_level * 4))));
+            display_msg(message_str(SCRIPT_STTEXT, random(140, 143)) + (floor(100 / 5) + (dude_iq * floor(100 / 12)) + (dude_luck * floor(100 / 30)) + (floor(100 / 75) * (((dude_luck - 4) * 4) - (difficulty_level * 4)))) + message_str(SCRIPT_STTEXT, 144));
             play_sfx("LEVELUP");
         end
         else begin
-            display_msg(message_str(390, 6));
+            display_mstr(6);
         end
     end
 end
@@ -72,16 +72,16 @@ end
 procedure use_skill_on_p_proc
 begin
     script_overrides;
-    if ((global_var(280) == 0) and (action_being_used == has_skill(dude_obj, 12))) then begin
-        if (get_critter_stat(dude_obj, 4) >= 9) then begin
-            display_msg(message_str(390, 5));
-            set_global_var(280, 1);
-            give_exp_points(floor(100 / 5) + (get_critter_stat(dude_obj, 4) * floor(100 / 12)) + (get_critter_stat(dude_obj, 6) * floor(100 / 30)) + (floor(100 / 75) * (((get_critter_stat(dude_obj, 6) - 4) * 4) - (difficulty_level * 4))));
-            display_msg(message_str(10, random(140, 143)) + (floor(100 / 5) + (get_critter_stat(dude_obj, 4) * floor(100 / 12)) + (get_critter_stat(dude_obj, 6) * floor(100 / 30)) + (floor(100 / 75) * (((get_critter_stat(dude_obj, 6) - 4) * 4) - (difficulty_level * 4)))) + message_str(10, 144));
+    if ((global_var(GVAR_CARAVAN_CARTS_TOTAL) == 0) and (action_being_used == dude_skill(SKILL_SCIENCE))) then begin
+        if (dude_iq >= 9) then begin
+            display_mstr(5);
+            set_global_var(GVAR_CARAVAN_CARTS_TOTAL, 1);
+            give_exp_points(floor(100 / 5) + (dude_iq * floor(100 / 12)) + (dude_luck * floor(100 / 30)) + (floor(100 / 75) * (((dude_luck - 4) * 4) - (difficulty_level * 4))));
+            display_msg(message_str(SCRIPT_STTEXT, random(140, 143)) + (floor(100 / 5) + (dude_iq * floor(100 / 12)) + (dude_luck * floor(100 / 30)) + (floor(100 / 75) * (((dude_luck - 4) * 4) - (difficulty_level * 4)))) + message_str(SCRIPT_STTEXT, 144));
             play_sfx("LEVELUP");
         end
         else begin
-            display_msg(message_str(390, 6));
+            display_mstr(6);
         end
     end
 end
@@ -97,4 +97,3 @@ end
 procedure talk_p_proc
 begin
 end
-
