diff --git a/Mods/Olympus2207/Original/SJHARRY.ssl b/Mods/Olympus2207/ReDefine/SJHARRY.ssl
index 1589368..370e211 100644
--- a/Mods/Olympus2207/Original/SJHARRY.ssl
+++ b/Mods/Olympus2207/ReDefine/SJHARRY.ssl
@@ -57,109 +57,109 @@ begin
         set_local_var(4, random(65, 75));
         set_local_var(5, random(55, 65));
         set_local_var(6, random(-20, 10));
-        critter_add_trait(self_obj, 1, 6, 7);
-        critter_add_trait(self_obj, 1, 5, 26);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 7);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_COOK_BERSERK);
         if (random(0, 100) <= 15) then begin
-            add_obj_to_inven(self_obj, create_object_sid(438, 1, 2, -1));
+            add_obj_to_inven(self_obj, create_object(438, 1, 2));
         end
         if (random(0, 100) <= 10) then begin
-            add_mult_objs_to_inven(self_obj, create_object_sid(238, 1, 2, -1), random(1, 2));
+            add_mult_objs_to_inven(self_obj, create_object(238, 1, 2), random(1, 2));
         end
         if (random(0, 100) <= 10) then begin
-            add_mult_objs_to_inven(self_obj, create_object_sid(213, 1, 2, -1), random(1, 2));
+            add_mult_objs_to_inven(self_obj, create_object(213, 1, 2), random(1, 2));
         end
         if (random(0, 100) <= 10) then begin
-            add_obj_to_inven(self_obj, create_object_sid(233, 1, 2, -1));
+            add_obj_to_inven(self_obj, create_object(233, 1, 2));
         end
         if (random(0, 100) <= 10) then begin
-            add_obj_to_inven(self_obj, create_object_sid(420, 1, 2, -1));
+            add_obj_to_inven(self_obj, create_object(420, 1, 2));
         end
         if (random(0, 100) <= 9) then begin
-            add_obj_to_inven(self_obj, create_object_sid(236, 1, 2, -1));
+            add_obj_to_inven(self_obj, create_object(236, 1, 2));
         end
         if (random(0, 100) <= 8) then begin
-            add_obj_to_inven(self_obj, create_object_sid(218, 1, 2, -1));
+            add_obj_to_inven(self_obj, create_object(218, 1, 2));
         end
         if (random(0, 100) <= 6) then begin
-            add_obj_to_inven(self_obj, create_object_sid(440, 1, 2, -1));
+            add_obj_to_inven(self_obj, create_object(440, 1, 2));
         end
         if (random(0, 100) <= 5) then begin
-            add_obj_to_inven(self_obj, create_object_sid(244, 1, 2, -1));
+            add_obj_to_inven(self_obj, create_object(244, 1, 2));
         end
         if (random(0, 100) <= 5) then begin
-            add_obj_to_inven(self_obj, create_object_sid(428, 1, 2, -1));
+            add_obj_to_inven(self_obj, create_object(428, 1, 2));
         end
         if (random(0, 100) <= 5) then begin
-            add_obj_to_inven(self_obj, create_object_sid(422, 1, 2, -1));
+            add_obj_to_inven(self_obj, create_object(422, 1, 2));
         end
         if (random(0, 100) <= 4) then begin
-            add_obj_to_inven(self_obj, create_object_sid(423, 1, 2, -1));
+            add_obj_to_inven(self_obj, create_object(423, 1, 2));
         end
         if (random(0, 100) <= 3) then begin
-            add_obj_to_inven(self_obj, create_object_sid(426, 1, 2, -1));
+            add_obj_to_inven(self_obj, create_object(426, 1, 2));
         end
         if (random(0, 100) <= 2) then begin
-            add_obj_to_inven(self_obj, create_object_sid(433, 1, 2, -1));
+            add_obj_to_inven(self_obj, create_object(433, 1, 2));
         end
         if (random(0, 100) <= 2) then begin
-            add_obj_to_inven(self_obj, create_object_sid(421, 1, 2, -1));
+            add_obj_to_inven(self_obj, create_object(421, 1, 2));
         end
         if (random(0, 100) <= 20) then begin
-            add_mult_objs_to_inven(self_obj, create_object_sid(41, 1, 2, -1), random(1, 10));
+            add_mult_objs_to_inven(self_obj, create_object(41, 1, 2), random(1, 10));
         end
         if (random(0, 100) <= 15) then begin
-            add_mult_objs_to_inven(self_obj, create_object_sid(238, 1, 2, -1), random(1, 2));
+            add_mult_objs_to_inven(self_obj, create_object(238, 1, 2), random(1, 2));
         end
         if (random(0, 100) <= 15) then begin
-            add_mult_objs_to_inven(self_obj, create_object_sid(213, 1, 2, -1), random(1, 2));
+            add_mult_objs_to_inven(self_obj, create_object(213, 1, 2), random(1, 2));
         end
         if (random(0, 100) <= 15) then begin
-            add_obj_to_inven(self_obj, create_object_sid(35, 1, 2, -1));
+            add_obj_to_inven(self_obj, create_object(35, 1, 2));
         end
         if (random(0, 100) <= 14) then begin
-            add_obj_to_inven(self_obj, create_object_sid(65, 1, 2, -1));
+            add_obj_to_inven(self_obj, create_object(65, 1, 2));
         end
         if (random(0, 100) <= 13) then begin
-            add_obj_to_inven(self_obj, create_object_sid(218, 1, 2, -1));
+            add_obj_to_inven(self_obj, create_object(218, 1, 2));
         end
         if (random(0, 100) <= 12) then begin
-            add_obj_to_inven(self_obj, create_object_sid(70, 1, 2, -1));
+            add_obj_to_inven(self_obj, create_object(70, 1, 2));
         end
         if (random(0, 100) <= 11) then begin
-            add_obj_to_inven(self_obj, create_object_sid(106, 1, 2, -1));
+            add_obj_to_inven(self_obj, create_object(106, 1, 2));
         end
         if (random(0, 100) <= 10) then begin
-            add_obj_to_inven(self_obj, create_object_sid(244, 1, 2, -1));
+            add_obj_to_inven(self_obj, create_object(244, 1, 2));
         end
         if (random(0, 100) <= 10) then begin
-            add_obj_to_inven(self_obj, create_object_sid(242, 1, 2, -1));
+            add_obj_to_inven(self_obj, create_object(242, 1, 2));
         end
         if (random(0, 100) <= 10) then begin
-            add_obj_to_inven(self_obj, create_object_sid(414, 1, 2, -1));
+            add_obj_to_inven(self_obj, create_object(414, 1, 2));
         end
         if (random(0, 100) <= 10) then begin
-            add_obj_to_inven(self_obj, create_object_sid(422, 1, 2, -1));
+            add_obj_to_inven(self_obj, create_object(422, 1, 2));
         end
         if (random(0, 100) <= 9) then begin
-            add_obj_to_inven(self_obj, create_object_sid(423, 1, 2, -1));
+            add_obj_to_inven(self_obj, create_object(423, 1, 2));
         end
         if (random(0, 100) <= 8) then begin
-            add_obj_to_inven(self_obj, create_object_sid(426, 1, 2, -1));
+            add_obj_to_inven(self_obj, create_object(426, 1, 2));
         end
         if (random(0, 100) <= 7) then begin
-            add_obj_to_inven(self_obj, create_object_sid(433, 1, 2, -1));
+            add_obj_to_inven(self_obj, create_object(433, 1, 2));
         end
         if (random(0, 100) <= 6) then begin
-            add_obj_to_inven(self_obj, create_object_sid(421, 1, 2, -1));
+            add_obj_to_inven(self_obj, create_object(421, 1, 2));
         end
         if (random(0, 100) <= 5) then begin
-            add_obj_to_inven(self_obj, create_object_sid(427, 1, 2, -1));
+            add_obj_to_inven(self_obj, create_object(427, 1, 2));
         end
         if (random(0, 100) <= 4) then begin
-            add_obj_to_inven(self_obj, create_object_sid(236, 1, 2, -1));
+            add_obj_to_inven(self_obj, create_object(236, 1, 2));
         end
         if (random(0, 100) <= 3) then begin
-            add_obj_to_inven(self_obj, create_object_sid(74, 1, 2, -1));
+            add_obj_to_inven(self_obj, create_object(74, 1, 2));
         end
     end
 end
@@ -175,7 +175,7 @@ begin
         add_timer_event(self_obj, random(1000, 1000), 2);
     end
     if (obj_can_see_obj(self_obj, dude_obj)) then begin
-        if ((local_var(2) == 1) or (global_var(52) <= -31)) then begin
+        if ((local_var(2) == 1) or (global_var(GVAR_TOWN_REP_MODOC) <= -31)) then begin
             set_local_var(2, 0);
             attack_setup(self_obj, dude_obj);
         end
@@ -190,10 +190,10 @@ procedure description_p_proc
 begin
     script_overrides;
     if (local_var(7) == 0) then begin
-        display_msg(message_str(163, 2));
+        display_mstr(2);
     end
     else begin
-        display_msg(message_str(163, 4));
+        display_mstr(4);
     end
 end
 
@@ -201,10 +201,10 @@ procedure look_at_p_proc
 begin
     script_overrides;
     if (local_var(7) == 0) then begin
-        display_msg(message_str(163, 1));
+        display_mstr(1);
     end
     else begin
-        display_msg(message_str(163, 3));
+        display_mstr(3);
     end
 end
 
@@ -222,7 +222,7 @@ end
 procedure talk_p_proc
 begin
     if (not(anim_busy(self_obj))) then begin
-        anim(self_obj, 1000, rotation_to_tile(tile_num(self_obj), tile_num(dude_obj)));
+        anim(self_obj, 1000, rotation_to_tile(tile_num(self_obj), dude_tile));
     end
     start_gdialog(163, self_obj, -1, -1, -1);
     gsay_start;
@@ -238,17 +238,17 @@ end
 
 procedure destroy_p_proc
 begin
-    set_global_var(69, global_var(69) bwor 524288);
-    if (global_var(134) == 1) then begin
-        set_global_var(134, 0);
+    set_global_var(GVAR_TORR_HARMED, global_var(GVAR_TORR_HARMED) bwor 524288);
+    if (global_var(GVAR_SALVATORE_FAMILY_COUNTER) == 1) then begin
+        set_global_var(GVAR_SALVATORE_FAMILY_COUNTER, 0);
     end
     if (source_obj == dude_obj) then begin
-        set_global_var(0, global_var(0) - 1);
+        set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) - 1);
         if ((52 == 50) or (52 == 59)) then begin
-            set_global_var(52, global_var(52) - 3);
+            set_global_var(GVAR_TOWN_REP_MODOC, global_var(GVAR_TOWN_REP_MODOC) - 3);
         end
         else begin
-            set_global_var(52, global_var(52) - 6);
+            set_global_var(GVAR_TOWN_REP_MODOC, global_var(GVAR_TOWN_REP_MODOC) - 6);
         end
     end
 end
@@ -261,51 +261,51 @@ begin
             set_sfall_global(13, 1);
         end
         else begin
-            if (has_trait(2, dude_obj, 13)) then begin
-                if (global_var(81) < get_pc_stat(1)) then begin
+            if (has_trait(TRAIT_TRAIT, dude_obj, TRAIT_sex_appeal)) then begin
+                if (global_var(GVAR_VAULT_CITIZENSHIP) < get_pc_stat(PCSTAT_level)) then begin
                     if (is_success(roll_vs_skill(dude_obj, 10, 0))) then begin
-                        set_global_var(98, global_var(98) + 3);
+                        set_global_var(GVAR_QUEST_FIND_THOMAS, global_var(GVAR_QUEST_FIND_THOMAS) + 3);
                     end
                     else begin
-                        set_global_var(98, global_var(98) + 1);
+                        set_global_var(GVAR_QUEST_FIND_THOMAS, global_var(GVAR_QUEST_FIND_THOMAS) + 1);
                     end
-                    if (global_var(98) >= (2 * get_pc_stat(1))) then begin
-                        set_global_var(98, 0);
-                        set_global_var(81, global_var(81) + 1);
-                        critter_mod_skill(dude_obj, 10, 1);
-                        display_msg(message_str(10, 410));
+                    if (global_var(GVAR_QUEST_FIND_THOMAS) >= (2 * get_pc_stat(PCSTAT_level))) then begin
+                        set_global_var(GVAR_QUEST_FIND_THOMAS, 0);
+                        set_global_var(GVAR_VAULT_CITIZENSHIP, global_var(GVAR_VAULT_CITIZENSHIP) + 1);
+                        critter_mod_skill(dude_obj, SKILL_STEAL, 1);
+                        display_msg(message_str(SCRIPT_STTEXT, 410));
                     end
                 end
             end
             steal_perk := 0;
             steal_gloves := 0;
-            if (has_trait(0, dude_obj, 37) != 0) then begin
+            if (has_trait(TRAIT_PERK, dude_obj, PERK_pickpocket) != 0) then begin
                 steal_perk := steal_perk + 15;
             end
-            if (has_trait(0, dude_obj, 33) != 0) then begin
+            if (has_trait(TRAIT_PERK, dude_obj, PERK_master_thief) != 0) then begin
                 steal_perk := steal_perk + 30;
             end
-            if ((obj_pid(critter_inven_obj(dude_obj, 1)) == 245) or (obj_pid(critter_inven_obj(dude_obj, 2)) == 245)) then begin
+            if ((obj_pid(critter_inven_obj(dude_obj, INVEN_TYPE_RIGHT_HAND)) == 245) or (obj_pid(critter_inven_obj(dude_obj, INVEN_TYPE_LEFT_HAND)) == 245)) then begin
                 steal_gloves := 10;
             end
-            if (random(0, 100) <= ((has_skill(dude_obj, 10) / 2) + (((3 * get_critter_stat(dude_obj, 5)) + (((get_critter_stat(dude_obj, 6) - 4) * 4) - (difficulty_level * 4)) + (2 * steal_perk) + steal_gloves) / (local_var(4) / 25)))) then begin
-                display_msg(message_str(10, random(145, 148)));
+            if (random(0, 100) <= ((dude_skill(SKILL_STEAL) / 2) + (((3 * get_critter_stat(dude_obj, STAT_ag)) + (((dude_luck - 4) * 4) - (difficulty_level * 4)) + (2 * steal_perk) + steal_gloves) / (local_var(4) / 25)))) then begin
+                display_msg(message_str(SCRIPT_STTEXT, random(145, 148)));
             end
             else begin
                 script_overrides;
-                if (random(0, 100) <= (25 - floor((((get_critter_stat(dude_obj, 6) - 4) * 4) - (difficulty_level * 4)) / 3))) then begin
-                    float_msg(self_obj, message_str(10, random(206, 209)), 2);
+                if (random(0, 100) <= (25 - floor((((dude_luck - 4) * 4) - (difficulty_level * 4)) / 3))) then begin
+                    float_msg(self_obj, message_str(SCRIPT_STTEXT, random(206, 209)), FLOAT_MSG_RED);
                     set_local_var(2, 1);
                 end
                 else begin
-                    display_msg(message_str(10, random(200, 203)));
+                    display_msg(message_str(SCRIPT_STTEXT, random(200, 203)));
                 end
             end
         end
     end
     if (action_being_used == 12) then begin
         script_overrides;
-        display_msg(message_str(10, 204) + ((has_skill(dude_obj, 10) / 2) + (((3 * get_critter_stat(dude_obj, 5)) + (((get_critter_stat(dude_obj, 6) - 4) * 4) - (difficulty_level * 4)) + (2 * steal_perk) + steal_gloves) / (local_var(4) / 25))) + message_str(10, 205));
+        display_msg(message_str(SCRIPT_STTEXT, 204) + ((dude_skill(SKILL_STEAL) / 2) + (((3 * get_critter_stat(dude_obj, STAT_ag)) + (((dude_luck - 4) * 4) - (difficulty_level * 4)) + (2 * steal_perk) + steal_gloves) / (local_var(4) / 25))) + message_str(SCRIPT_STTEXT, 205));
     end
 end
 
@@ -342,157 +342,157 @@ procedure Node001
 begin
     set_local_var(7, 1);
     already_talk := 1;
-    gsay_reply(163, 10);
-    gsay_option(163, 11, Node002, 50);
-    gsay_option(163, 12, Node999, 50);
+    Reply(10);
+    giq_option(1, NAME, 11, Node002, NEUTRAL_REACTION);
+    giq_option(1, NAME, 12, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node002
 begin
-    gsay_reply(163, 24);
-    gsay_option(163, message_str(163, 20) + obj_name(dude_obj) + message_str(163, 21), Node003, 50);
-    gsay_option(163, 22, Node003, 50);
-    gsay_option(163, 23, Node005, 50);
+    Reply(24);
+    giq_option(1, NAME, mstr(20) + dude_name + mstr(21), Node003, NEUTRAL_REACTION);
+    giq_option(1, NAME, 22, Node003, NEUTRAL_REACTION);
+    giq_option(1, NAME, 23, Node005, NEUTRAL_REACTION);
 end
 
 procedure Node003
 begin
-    gsay_reply(163, 30);
-    gsay_option(163, 31, Node004, 50);
+    Reply(30);
+    giq_option(1, NAME, 31, Node004, NEUTRAL_REACTION);
 end
 
 procedure Node004
 begin
-    gsay_reply(163, 40);
-    gsay_option(163, 41, Node010, 50);
-    gsay_option(163, 42, Node999, 50);
+    Reply(40);
+    giq_option(1, NAME, 41, Node010, NEUTRAL_REACTION);
+    giq_option(1, NAME, 42, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node005
 begin
-    gsay_reply(163, 50);
-    gsay_option(163, 51, Node010, 50);
-    gsay_option(163, 52, Node999, 50);
+    Reply(50);
+    giq_option(1, NAME, 51, Node010, NEUTRAL_REACTION);
+    giq_option(1, NAME, 52, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node010
 begin
     if (already_talk == 0) then begin
         already_talk := 1;
-        gsay_reply(163, 100);
+        Reply(100);
     end
     else begin
-        gsay_reply(163, 101);
+        Reply(101);
     end
-    if (global_var(134) == 0) then begin
-        gsay_option(163, 102, Node011, 50);
+    if (global_var(GVAR_SALVATORE_FAMILY_COUNTER) == 0) then begin
+        giq_option(1, NAME, 102, Node011, NEUTRAL_REACTION);
     end
-    if (global_var(134) == 1) then begin
-        gsay_option(163, 103, Node015, 50);
+    if (global_var(GVAR_SALVATORE_FAMILY_COUNTER) == 1) then begin
+        giq_option(1, NAME, 103, Node015, NEUTRAL_REACTION);
     end
-    gsay_option(163, 104, Node019, 50);
-    gsay_option(163, 107, Node999, 50);
+    giq_option(1, NAME, 104, Node019, NEUTRAL_REACTION);
+    giq_option(1, NAME, 107, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node011
 begin
-    gsay_reply(163, 110);
-    gsay_option(163, 111, Node012, 50);
-    gsay_option(163, 112, Node012, 50);
-    gsay_option(163, 113, Node012, 50);
+    Reply(110);
+    giq_option(1, NAME, 111, Node012, NEUTRAL_REACTION);
+    giq_option(1, NAME, 112, Node012, NEUTRAL_REACTION);
+    giq_option(1, NAME, 113, Node012, NEUTRAL_REACTION);
 end
 
 procedure Node012
 begin
-    gsay_reply(163, 120);
-    gsay_option(163, 121, Node013, 50);
-    gsay_option(163, 122, Node014, 50);
-    gsay_option(163, 123, Node012a, 50);
-    gsay_option(163, 124, Node012b, 50);
+    Reply(120);
+    giq_option(1, NAME, 121, Node013, NEUTRAL_REACTION);
+    giq_option(1, NAME, 122, Node014, NEUTRAL_REACTION);
+    giq_option(1, NAME, 123, Node012a, NEUTRAL_REACTION);
+    giq_option(1, NAME, 124, Node012b, NEUTRAL_REACTION);
 end
 
 procedure Node012a
 begin
-    if (global_var(134) == 0) then begin
-        set_global_var(134, 1);
+    if (global_var(GVAR_SALVATORE_FAMILY_COUNTER) == 0) then begin
+        set_global_var(GVAR_SALVATORE_FAMILY_COUNTER, 1);
     end
     call Node010();
 end
 
 procedure Node012b
 begin
-    if (global_var(134) == 0) then begin
-        set_global_var(134, 1);
+    if (global_var(GVAR_SALVATORE_FAMILY_COUNTER) == 0) then begin
+        set_global_var(GVAR_SALVATORE_FAMILY_COUNTER, 1);
     end
 end
 
 procedure Node013
 begin
-    gsay_reply(163, 130);
-    gsay_option(163, 131, Node014, 50);
-    gsay_option(163, 132, Node012a, 50);
-    gsay_option(163, 133, Node012b, 50);
+    Reply(130);
+    giq_option(1, NAME, 131, Node014, NEUTRAL_REACTION);
+    giq_option(1, NAME, 132, Node012a, NEUTRAL_REACTION);
+    giq_option(1, NAME, 133, Node012b, NEUTRAL_REACTION);
 end
 
 procedure Node014
 begin
-    gsay_reply(163, 140);
-    gsay_option(163, 141, Node013, 50);
-    gsay_option(163, 142, Node012a, 50);
-    gsay_option(163, 143, Node012b, 50);
+    Reply(140);
+    giq_option(1, NAME, 141, Node013, NEUTRAL_REACTION);
+    giq_option(1, NAME, 142, Node012a, NEUTRAL_REACTION);
+    giq_option(1, NAME, 143, Node012b, NEUTRAL_REACTION);
 end
 
 procedure Node015
 begin
-    gsay_reply(163, 153);
+    Reply(153);
     if (obj_carrying_pid_obj(dude_obj, 306)) then begin
-        gsay_option(163, 150, Node016, 50);
+        giq_option(1, NAME, 150, Node016, NEUTRAL_REACTION);
     end
-    gsay_option(163, 151, Node017, 50);
-    gsay_option(163, 152, Node018, 50);
+    giq_option(1, NAME, 151, Node017, NEUTRAL_REACTION);
+    giq_option(1, NAME, 152, Node018, NEUTRAL_REACTION);
 end
 
 procedure Node016
 begin
     rm_obj_from_inven(dude_obj, obj_carrying_pid_obj(dude_obj, 306));
-    add_obj_to_inven(self_obj, create_object_sid(306, 1, 2, -1));
-    if (global_var(134) == 1) then begin
-        set_global_var(134, 2);
-        add_obj_to_inven(dude_obj, create_object_sid(68, 1, 2, -1));
-        give_exp_points(floor(518 / 5) + (get_critter_stat(dude_obj, 4) * floor(518 / 12)) + (get_critter_stat(dude_obj, 6) * floor(518 / 30)) + (floor(518 / 75) * (((get_critter_stat(dude_obj, 6) - 4) * 4) - (difficulty_level * 4))));
-        display_msg(message_str(10, random(140, 143)) + (floor(518 / 5) + (get_critter_stat(dude_obj, 4) * floor(518 / 12)) + (get_critter_stat(dude_obj, 6) * floor(518 / 30)) + (floor(518 / 75) * (((get_critter_stat(dude_obj, 6) - 4) * 4) - (difficulty_level * 4)))) + message_str(10, 144));
+    add_obj_to_inven(self_obj, create_object(306, 1, 2));
+    if (global_var(GVAR_SALVATORE_FAMILY_COUNTER) == 1) then begin
+        set_global_var(GVAR_SALVATORE_FAMILY_COUNTER, 2);
+        add_obj_to_inven(dude_obj, create_object(68, 1, 2));
+        give_exp_points(floor(518 / 5) + (dude_iq * floor(518 / 12)) + (dude_luck * floor(518 / 30)) + (floor(518 / 75) * (((dude_luck - 4) * 4) - (difficulty_level * 4))));
+        display_msg(message_str(SCRIPT_STTEXT, random(140, 143)) + (floor(518 / 5) + (dude_iq * floor(518 / 12)) + (dude_luck * floor(518 / 30)) + (floor(518 / 75) * (((dude_luck - 4) * 4) - (difficulty_level * 4)))) + message_str(SCRIPT_STTEXT, 144));
     end
-    gsay_reply(163, 160);
-    gsay_option(163, 161, Node010, 50);
-    gsay_option(163, 162, Node999, 50);
+    Reply(160);
+    giq_option(1, NAME, 161, Node010, NEUTRAL_REACTION);
+    giq_option(1, NAME, 162, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node017
 begin
-    gsay_reply(163, 170);
-    gsay_option(163, 171, Node010, 50);
-    gsay_option(163, 172, Node999, 50);
+    Reply(170);
+    giq_option(1, NAME, 171, Node010, NEUTRAL_REACTION);
+    giq_option(1, NAME, 172, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node018
 begin
-    gsay_reply(163, 180);
-    gsay_option(163, 181, Node010, 50);
-    gsay_option(163, 182, Node999, 50);
+    Reply(180);
+    giq_option(1, NAME, 181, Node010, NEUTRAL_REACTION);
+    giq_option(1, NAME, 182, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node019
 begin
-    gsay_reply(163, 190);
-    gsay_option(163, 191, Node020, 50);
-    gsay_option(163, 192, Node020, 50);
+    Reply(190);
+    giq_option(1, NAME, 191, Node020, NEUTRAL_REACTION);
+    giq_option(1, NAME, 192, Node020, NEUTRAL_REACTION);
 end
 
 procedure Node020
 begin
-    gsay_reply(163, 200);
-    gsay_option(163, 201, Node010, 50);
-    gsay_option(163, 202, Node999, 50);
+    Reply(200);
+    giq_option(1, NAME, 201, Node010, NEUTRAL_REACTION);
+    giq_option(1, NAME, 202, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node999
@@ -511,4 +511,3 @@ begin
     gdialog_mod_barter(-(dude_objective_reputation + local_var(6)));
     call Node010();
 end
-
