diff --git a/Mods/Olympus2207/Original/SJZSHLAG.ssl b/Mods/Olympus2207/ReDefine/SJZSHLAG.ssl
index 8eefed3..d6900a4 100644
--- a/Mods/Olympus2207/Original/SJZSHLAG.ssl
+++ b/Mods/Olympus2207/ReDefine/SJZSHLAG.ssl
@@ -42,7 +42,7 @@ begin
     if (source_obj == dude_obj) then begin
         if (local_var(1) == 0) then begin
             if (map_var(0) == 0) then begin
-                display_msg(message_str(10, random(40, 43)));
+                display_msg(message_str(SCRIPT_STTEXT, random(40, 43)));
             end
             else begin
                 set_local_var(1, 1);
@@ -57,18 +57,18 @@ begin
         end
     end
     else begin
-        animate_move_obj_to_tile(source_obj, tile_num_in_direction(tile_num(source_obj), (has_trait(1, source_obj, 10) + 3) % 6, 1), 0 bwor 16);
+        animate_move_obj_to_tile(source_obj, tile_num_in_direction(tile_num(source_obj), (obj_get_rot(source_obj) + 3) % 6, 1), 0 bwor 16);
     end
 end
 
 procedure use_obj_on_p_proc
 begin
-    display_msg(message_str(10, 130));
+    display_msg(message_str(SCRIPT_STTEXT, 130));
 end
 
 procedure use_skill_on_p_proc
 begin
-    display_msg(message_str(10, 130));
+    display_msg(message_str(SCRIPT_STTEXT, 130));
 end
 
 procedure map_enter_p_proc
@@ -114,4 +114,3 @@ begin
         end
     end
 end
-
