diff --git a/Mods/Olympus2207/Original/SJZSTNOK.ssl b/Mods/Olympus2207/ReDefine/SJZSTNOK.ssl
index d073c22..b7fbb33 100644
--- a/Mods/Olympus2207/Original/SJZSTNOK.ssl
+++ b/Mods/Olympus2207/ReDefine/SJZSTNOK.ssl
@@ -28,10 +28,10 @@ procedure description_p_proc
 begin
     script_overrides;
     if (map_var(0) == 0) then begin
-        display_msg(message_str(424, 3));
+        display_mstr(3);
     end
     else begin
-        display_msg(message_str(424, 4));
+        display_mstr(4);
     end
 end
 
@@ -39,10 +39,10 @@ procedure look_at_p_proc
 begin
     script_overrides;
     if (map_var(0) == 0) then begin
-        display_msg(message_str(424, 1));
+        display_mstr(1);
     end
     else begin
-        display_msg(message_str(424, 2));
+        display_mstr(2);
     end
 end
 
@@ -54,23 +54,23 @@ procedure use_p_proc
 begin
     script_overrides;
     if (map_var(0) == 0) then begin
-        if ((global_var(285) == 3) and (global_var(132) == 1)) then begin
-            display_msg(message_str(424, 7));
+        if ((global_var(GVAR_NEW_RENO_WRIGHT_FLAGS) == 3) and (global_var(GVAR_MUTATE) == 1)) then begin
+            display_mstr(7);
             set_map_var(0, 1);
             set_local_var(1, 1);
         end
         else begin
-            display_msg(message_str(424, random(5, 6)));
+            display_mstr(random(5, 6));
         end
     end
     else begin
-        display_msg(message_str(424, 8));
+        display_mstr(8);
     end
 end
 
 procedure use_obj_on_p_proc
 begin
-    display_msg(message_str(10, random(60, 69)));
+    display_msg(message_str(SCRIPT_STTEXT, random(60, 69)));
 end
 
 procedure use_skill_on_p_proc
@@ -92,4 +92,3 @@ end
 procedure timed_event_p_proc
 begin
 end
-
