diff --git a/Mods/Olympus2207/Original/SPPIRAT1.ssl b/Mods/Olympus2207/ReDefine/SPPIRAT1.ssl
index 9acb69f..b46ffb9 100644
--- a/Mods/Olympus2207/Original/SPPIRAT1.ssl
+++ b/Mods/Olympus2207/ReDefine/SPPIRAT1.ssl
@@ -34,152 +34,152 @@ begin
         set_local_var(1, 1);
         set_local_var(4, random(90, 100));
         set_local_var(5, random(0, 10));
-        critter_add_trait(self_obj, 1, 6, 52);
-        critter_add_trait(self_obj, 1, 5, 24);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 52);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_RAIDER_NORMAL);
         if (random(0, 100) <= 50) then begin
             if (random(0, 100) <= 60) then begin
                 if (random(0, 100) <= 33) then begin
-                    add_obj_to_inven(self_obj, create_object_sid(169, 1, 2, -1));
-                    add_mult_objs_to_inven(self_obj, create_object_sid(5, 1, 2, -1), random(21, 38));
-                    wield_obj_critter(self_obj, obj_carrying_pid_obj(self_obj, 169));
+                    add_obj_to_inven(self_obj, create_object(169, 1, 2));
+                    add_mult_objs_to_inven(self_obj, create_object(5, 1, 2), random(21, 38));
+                    wield_obj(obj_carrying_pid_obj(self_obj, 169));
                 end
                 else begin
-                    add_obj_to_inven(self_obj, create_object_sid(177, 1, 2, -1));
-                    add_mult_objs_to_inven(self_obj, create_object_sid(14, 1, 2, -1), random(10, 18));
-                    wield_obj_critter(self_obj, obj_carrying_pid_obj(self_obj, 177));
+                    add_obj_to_inven(self_obj, create_object(177, 1, 2));
+                    add_mult_objs_to_inven(self_obj, create_object(14, 1, 2), random(10, 18));
+                    wield_obj(obj_carrying_pid_obj(self_obj, 177));
                 end
                 if (random(0, 100) <= 33) then begin
-                    add_obj_to_inven(self_obj, create_object_sid(221, 1, 2, -1));
+                    add_obj_to_inven(self_obj, create_object(221, 1, 2));
                 end
                 else begin
-                    add_obj_to_inven(self_obj, create_object_sid(106, 1, 2, -1));
+                    add_obj_to_inven(self_obj, create_object(106, 1, 2));
                 end
                 if (random(0, 100) <= 33) then begin
-                    add_mult_objs_to_inven(self_obj, create_object_sid(41, 1, 2, -1), random(10, 40));
+                    add_mult_objs_to_inven(self_obj, create_object(41, 1, 2), random(10, 40));
                 end
                 if (random(0, 100) <= 25) then begin
-                    add_mult_objs_to_inven(self_obj, create_object_sid(63, 1, 2, -1), random(2, 4));
+                    add_mult_objs_to_inven(self_obj, create_object(63, 1, 2), random(2, 4));
                 end
                 if (random(0, 100) <= 10) then begin
-                    add_obj_to_inven(self_obj, create_object_sid(64, 1, 2, -1));
+                    add_obj_to_inven(self_obj, create_object(64, 1, 2));
                 end
                 if (random(0, 100) <= 9) then begin
-                    add_obj_to_inven(self_obj, create_object_sid(431, 1, 2, -1));
+                    add_obj_to_inven(self_obj, create_object(431, 1, 2));
                 end
                 if (random(0, 100) <= 8) then begin
-                    add_obj_to_inven(self_obj, create_object_sid(236, 1, 2, -1));
+                    add_obj_to_inven(self_obj, create_object(236, 1, 2));
                 end
                 if (random(0, 100) <= 7) then begin
-                    add_obj_to_inven(self_obj, create_object_sid(229, 1, 2, -1));
+                    add_obj_to_inven(self_obj, create_object(229, 1, 2));
                 end
                 if (random(0, 100) <= 6) then begin
-                    add_obj_to_inven(self_obj, create_object_sid(230, 1, 2, -1));
+                    add_obj_to_inven(self_obj, create_object(230, 1, 2));
                 end
             end
             else begin
                 if (random(0, 100) <= 33) then begin
-                    add_obj_to_inven(self_obj, create_object_sid(143, 1, 2, -1));
-                    add_mult_objs_to_inven(self_obj, create_object_sid(1, 1, 2, -1), random(1, 2));
-                    wield_obj_critter(self_obj, obj_carrying_pid_obj(self_obj, 143));
+                    add_obj_to_inven(self_obj, create_object(143, 1, 2));
+                    add_mult_objs_to_inven(self_obj, create_object(1, 1, 2), random(1, 2));
+                    wield_obj(obj_carrying_pid_obj(self_obj, 143));
                 end
                 else begin
-                    add_obj_to_inven(self_obj, create_object_sid(141, 1, 2, -1));
-                    add_mult_objs_to_inven(self_obj, create_object_sid(25, 1, 2, -1), random(2, 4));
-                    wield_obj_critter(self_obj, obj_carrying_pid_obj(self_obj, 141));
+                    add_obj_to_inven(self_obj, create_object(141, 1, 2));
+                    add_mult_objs_to_inven(self_obj, create_object(25, 1, 2), random(2, 4));
+                    wield_obj(obj_carrying_pid_obj(self_obj, 141));
                 end
                 if (random(0, 100) <= 33) then begin
-                    add_obj_to_inven(self_obj, create_object_sid(219, 1, 2, -1));
+                    add_obj_to_inven(self_obj, create_object(219, 1, 2));
                 end
                 else begin
-                    add_obj_to_inven(self_obj, create_object_sid(109, 1, 2, -1));
+                    add_obj_to_inven(self_obj, create_object(109, 1, 2));
                 end
                 if (random(0, 100) <= 20) then begin
-                    add_mult_objs_to_inven(self_obj, create_object_sid(41, 1, 2, -1), random(1, 10));
+                    add_mult_objs_to_inven(self_obj, create_object(41, 1, 2), random(1, 10));
                 end
                 if (random(0, 100) <= 10) then begin
-                    add_mult_objs_to_inven(self_obj, create_object_sid(220, 1, 2, -1), random(1, 2));
+                    add_mult_objs_to_inven(self_obj, create_object(220, 1, 2), random(1, 2));
                 end
                 if (random(0, 100) <= 10) then begin
-                    add_mult_objs_to_inven(self_obj, create_object_sid(49, 1, 2, -1), random(1, 2));
+                    add_mult_objs_to_inven(self_obj, create_object(49, 1, 2), random(1, 2));
                 end
                 if (random(0, 100) <= 15) then begin
-                    add_obj_to_inven(self_obj, create_object_sid(221, 1, 2, -1));
+                    add_obj_to_inven(self_obj, create_object(221, 1, 2));
                 end
                 if (random(0, 100) <= 14) then begin
-                    add_obj_to_inven(self_obj, create_object_sid(65, 1, 2, -1));
+                    add_obj_to_inven(self_obj, create_object(65, 1, 2));
                 end
                 if (random(0, 100) <= 12) then begin
-                    add_obj_to_inven(self_obj, create_object_sid(71, 1, 2, -1));
+                    add_obj_to_inven(self_obj, create_object(71, 1, 2));
                 end
                 if (random(0, 100) <= 11) then begin
-                    add_obj_to_inven(self_obj, create_object_sid(138, 1, 2, -1));
+                    add_obj_to_inven(self_obj, create_object(138, 1, 2));
                 end
                 if (random(0, 100) <= 10) then begin
-                    add_obj_to_inven(self_obj, create_object_sid(47, 1, 2, -1));
+                    add_obj_to_inven(self_obj, create_object(47, 1, 2));
                 end
                 if (random(0, 100) <= 10) then begin
-                    add_obj_to_inven(self_obj, create_object_sid(43, 1, 2, -1));
+                    add_obj_to_inven(self_obj, create_object(43, 1, 2));
                 end
                 if (random(0, 100) <= 9) then begin
-                    add_obj_to_inven(self_obj, create_object_sid(237, 1, 2, -1));
+                    add_obj_to_inven(self_obj, create_object(237, 1, 2));
                 end
                 if (random(0, 100) <= 8) then begin
-                    add_obj_to_inven(self_obj, create_object_sid(323, 1, 2, -1));
+                    add_obj_to_inven(self_obj, create_object(323, 1, 2));
                 end
                 if (random(0, 100) <= 7) then begin
-                    add_obj_to_inven(self_obj, create_object_sid(246, 1, 2, -1));
+                    add_obj_to_inven(self_obj, create_object(246, 1, 2));
                 end
                 if (random(0, 100) <= 6) then begin
-                    add_obj_to_inven(self_obj, create_object_sid(110, 1, 2, -1));
+                    add_obj_to_inven(self_obj, create_object(110, 1, 2));
                 end
                 if (random(0, 100) <= 5) then begin
-                    add_obj_to_inven(self_obj, create_object_sid(421, 1, 2, -1));
+                    add_obj_to_inven(self_obj, create_object(421, 1, 2));
                 end
                 if (random(0, 100) <= 4) then begin
-                    add_obj_to_inven(self_obj, create_object_sid(102, 1, 2, -1));
+                    add_obj_to_inven(self_obj, create_object(102, 1, 2));
                 end
                 if (random(0, 100) <= 3) then begin
-                    add_obj_to_inven(self_obj, create_object_sid(134, 1, 2, -1));
+                    add_obj_to_inven(self_obj, create_object(134, 1, 2));
                 end
             end
         end
         else begin
             if (random(0, 100) <= 33) then begin
-                add_obj_to_inven(self_obj, create_object_sid(151, 1, 2, -1));
-                add_mult_objs_to_inven(self_obj, create_object_sid(9, 1, 2, -1), random(14, 26));
-                wield_obj_critter(self_obj, obj_carrying_pid_obj(self_obj, 151));
+                add_obj_to_inven(self_obj, create_object(151, 1, 2));
+                add_mult_objs_to_inven(self_obj, create_object(9, 1, 2), random(14, 26));
+                wield_obj(obj_carrying_pid_obj(self_obj, 151));
             end
             else begin
-                add_obj_to_inven(self_obj, create_object_sid(152, 1, 2, -1));
-                add_mult_objs_to_inven(self_obj, create_object_sid(2, 1, 2, -1), random(16, 29));
-                wield_obj_critter(self_obj, obj_carrying_pid_obj(self_obj, 152));
+                add_obj_to_inven(self_obj, create_object(152, 1, 2));
+                add_mult_objs_to_inven(self_obj, create_object(2, 1, 2), random(16, 29));
+                wield_obj(obj_carrying_pid_obj(self_obj, 152));
             end
             if (random(0, 100) <= 33) then begin
-                add_obj_to_inven(self_obj, create_object_sid(215, 1, 2, -1));
+                add_obj_to_inven(self_obj, create_object(215, 1, 2));
             end
             else begin
-                add_obj_to_inven(self_obj, create_object_sid(109, 1, 2, -1));
+                add_obj_to_inven(self_obj, create_object(109, 1, 2));
             end
             if (random(0, 100) <= 33) then begin
-                add_mult_objs_to_inven(self_obj, create_object_sid(41, 1, 2, -1), random(10, 40));
+                add_mult_objs_to_inven(self_obj, create_object(41, 1, 2), random(10, 40));
             end
             if (random(0, 100) <= 25) then begin
-                add_mult_objs_to_inven(self_obj, create_object_sid(220, 1, 2, -1), random(1, 3));
+                add_mult_objs_to_inven(self_obj, create_object(220, 1, 2), random(1, 3));
             end
             if (random(0, 100) <= 10) then begin
-                add_obj_to_inven(self_obj, create_object_sid(66, 1, 2, -1));
+                add_obj_to_inven(self_obj, create_object(66, 1, 2));
             end
             if (random(0, 100) <= 9) then begin
-                add_obj_to_inven(self_obj, create_object_sid(431, 1, 2, -1));
+                add_obj_to_inven(self_obj, create_object(431, 1, 2));
             end
             if (random(0, 100) <= 8) then begin
-                add_obj_to_inven(self_obj, create_object_sid(427, 1, 2, -1));
+                add_obj_to_inven(self_obj, create_object(427, 1, 2));
             end
             if (random(0, 100) <= 7) then begin
-                add_obj_to_inven(self_obj, create_object_sid(226, 1, 2, -1));
+                add_obj_to_inven(self_obj, create_object(226, 1, 2));
             end
             if (random(0, 100) <= 6) then begin
-                add_obj_to_inven(self_obj, create_object_sid(229, 1, 2, -1));
+                add_obj_to_inven(self_obj, create_object(229, 1, 2));
             end
         end
         if (tile_num(self_obj) == 22500) then begin
@@ -202,36 +202,36 @@ begin
             if (random(0, 100) <= 50) then begin
                 if (random(0, 100) <= 50) then begin
                     if (random(0, 100) <= 50) then begin
-                        metarule3(107, self_obj, 16777244, 0);
+                        art_change_fid_num(self_obj, 16777244);
                     end
                     else begin
-                        metarule3(107, self_obj, 16777246, 0);
+                        art_change_fid_num(self_obj, 16777246);
                     end
                 end
                 else begin
                     if (random(0, 100) <= 50) then begin
-                        metarule3(107, self_obj, 16777250, 0);
+                        art_change_fid_num(self_obj, 16777250);
                     end
                     else begin
-                        metarule3(107, self_obj, 16777251, 0);
+                        art_change_fid_num(self_obj, 16777251);
                     end
                 end
             end
             else begin
                 if (random(0, 100) <= 50) then begin
                     if (random(0, 100) <= 50) then begin
-                        metarule3(107, self_obj, 16777275, 0);
+                        art_change_fid_num(self_obj, 16777275);
                     end
                     else begin
-                        metarule3(107, self_obj, 16777276, 0);
+                        art_change_fid_num(self_obj, 16777276);
                     end
                 end
                 else begin
                     if (random(0, 100) <= 50) then begin
-                        metarule3(107, self_obj, 16777277, 0);
+                        art_change_fid_num(self_obj, 16777277);
                     end
                     else begin
-                        metarule3(107, self_obj, 16777281, 0);
+                        art_change_fid_num(self_obj, 16777281);
                     end
                 end
             end
@@ -240,36 +240,36 @@ begin
             if (random(0, 100) <= 50) then begin
                 if (random(0, 100) <= 50) then begin
                     if (random(0, 100) <= 50) then begin
-                        metarule3(107, self_obj, 16777282, 0);
+                        art_change_fid_num(self_obj, 16777282);
                     end
                     else begin
-                        metarule3(107, self_obj, 16777283, 0);
+                        art_change_fid_num(self_obj, 16777283);
                     end
                 end
                 else begin
                     if (random(0, 100) <= 50) then begin
-                        metarule3(107, self_obj, 16777282, 0);
+                        art_change_fid_num(self_obj, 16777282);
                     end
                     else begin
-                        metarule3(107, self_obj, 16777281, 0);
+                        art_change_fid_num(self_obj, 16777281);
                     end
                 end
             end
             else begin
                 if (random(0, 100) <= 50) then begin
                     if (random(0, 100) <= 50) then begin
-                        metarule3(107, self_obj, 16777277, 0);
+                        art_change_fid_num(self_obj, 16777277);
                     end
                     else begin
-                        metarule3(107, self_obj, 16777276, 0);
+                        art_change_fid_num(self_obj, 16777276);
                     end
                 end
                 else begin
                     if (random(0, 100) <= 50) then begin
-                        metarule3(107, self_obj, 16777275, 0);
+                        art_change_fid_num(self_obj, 16777275);
                     end
                     else begin
-                        metarule3(107, self_obj, 16777251, 0);
+                        art_change_fid_num(self_obj, 16777251);
                     end
                 end
             end
@@ -296,24 +296,24 @@ begin
             set_local_var(10, 1);
         end
     end
-    if ((obj_pid(critter_inven_obj(dude_obj, 0)) == 120) or (obj_pid(critter_inven_obj(dude_obj, 0)) == 121) or (obj_pid(critter_inven_obj(dude_obj, 0)) == 122) or (obj_pid(critter_inven_obj(dude_obj, 0)) == 123) or (obj_pid(critter_inven_obj(dude_obj, 0)) == 129) or (obj_pid(critter_inven_obj(dude_obj, 0)) == 391) and (obj_can_see_obj(self_obj, dude_obj) == 1)) then begin
+    if ((obj_pid(critter_inven_obj(dude_obj, INVEN_TYPE_WORN)) == 120) or (obj_pid(critter_inven_obj(dude_obj, INVEN_TYPE_WORN)) == 121) or (obj_pid(critter_inven_obj(dude_obj, INVEN_TYPE_WORN)) == 122) or (obj_pid(critter_inven_obj(dude_obj, INVEN_TYPE_WORN)) == 123) or (obj_pid(critter_inven_obj(dude_obj, INVEN_TYPE_WORN)) == 129) or (obj_pid(critter_inven_obj(dude_obj, INVEN_TYPE_WORN)) == 391) and (obj_can_see_obj(self_obj, dude_obj) == 1)) then begin
         if (local_var(11) == 0) then begin
-            float_msg(self_obj, message_str(516, random(13, 15)), 2);
+            float_msg(self_obj, mstr(random(13, 15)), FLOAT_MSG_RED);
             set_local_var(11, 1);
         end
         set_local_var(2, 1);
     end
     else begin
-        if (global_var(0) > -100) then begin
+        if (global_var(GVAR_PLAYER_REPUTATION) > -100) then begin
             if (local_var(11) == 0) then begin
-                float_msg(self_obj, message_str(516, random(10, 12)), 2);
+                float_msg(self_obj, mstr(random(10, 12)), FLOAT_MSG_RED);
                 set_local_var(11, 1);
             end
             set_local_var(2, 1);
         end
     end
     if (obj_can_see_obj(self_obj, dude_obj)) then begin
-        if ((local_var(2) == 1) or (global_var(628) <= -31)) then begin
+        if ((local_var(2) == 1) or (global_var(GVAR_TOWN_REP_NAVARRO) <= -31)) then begin
             set_local_var(2, 0);
             attack_setup(self_obj, dude_obj);
         end
@@ -327,13 +327,13 @@ end
 procedure description_p_proc
 begin
     script_overrides;
-    display_msg(message_str(516, 2));
+    display_mstr(2);
 end
 
 procedure look_at_p_proc
 begin
     script_overrides;
-    display_msg(message_str(516, 1));
+    display_mstr(1);
 end
 
 procedure pickup_p_proc
@@ -347,16 +347,16 @@ procedure push_p_proc
 begin
     script_overrides;
     if (local_var(7) <= 3) then begin
-        float_msg(self_obj, message_str(516, random(16, 17)), 0);
+        float_msg(self_obj, mstr(random(16, 17)), FLOAT_MSG_NORMAL);
         set_local_var(7, local_var(7) + 1);
     end
     else begin
         if (local_var(7) == 4) then begin
-            float_msg(self_obj, message_str(516, random(16, 17)), 2);
+            float_msg(self_obj, mstr(random(16, 17)), FLOAT_MSG_RED);
             set_local_var(7, local_var(7) + 1);
         end
         else begin
-            float_msg(self_obj, message_str(516, random(18, 19)), 2);
+            float_msg(self_obj, mstr(random(18, 19)), FLOAT_MSG_RED);
             set_local_var(7, 0);
             set_local_var(2, 1);
         end
@@ -366,20 +366,20 @@ end
 procedure talk_p_proc
 begin
     if (not(anim_busy(self_obj))) then begin
-        anim(self_obj, 1000, rotation_to_tile(tile_num(self_obj), tile_num(dude_obj)));
+        anim(self_obj, 1000, rotation_to_tile(tile_num(self_obj), dude_tile));
     end
-    float_msg(self_obj, message_str(516, random(20, 22)), 0);
+    float_msg(self_obj, mstr(random(20, 22)), FLOAT_MSG_NORMAL);
 end
 
 procedure destroy_p_proc
 begin
     if (source_obj == dude_obj) then begin
-        set_global_var(0, global_var(0) - 1);
+        set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) - 1);
         if ((628 == 50) or (628 == 59)) then begin
-            set_global_var(628, global_var(628) - 3);
+            set_global_var(GVAR_TOWN_REP_NAVARRO, global_var(GVAR_TOWN_REP_NAVARRO) - 3);
         end
         else begin
-            set_global_var(628, global_var(628) - 6);
+            set_global_var(GVAR_TOWN_REP_NAVARRO, global_var(GVAR_TOWN_REP_NAVARRO) - 6);
         end
     end
 end
@@ -392,51 +392,51 @@ begin
             set_sfall_global(13, 1);
         end
         else begin
-            if (has_trait(2, dude_obj, 13)) then begin
-                if (global_var(81) < get_pc_stat(1)) then begin
+            if (has_trait(TRAIT_TRAIT, dude_obj, TRAIT_sex_appeal)) then begin
+                if (global_var(GVAR_VAULT_CITIZENSHIP) < get_pc_stat(PCSTAT_level)) then begin
                     if (is_success(roll_vs_skill(dude_obj, 10, 0))) then begin
-                        set_global_var(98, global_var(98) + 3);
+                        set_global_var(GVAR_QUEST_FIND_THOMAS, global_var(GVAR_QUEST_FIND_THOMAS) + 3);
                     end
                     else begin
-                        set_global_var(98, global_var(98) + 1);
+                        set_global_var(GVAR_QUEST_FIND_THOMAS, global_var(GVAR_QUEST_FIND_THOMAS) + 1);
                     end
-                    if (global_var(98) >= (2 * get_pc_stat(1))) then begin
-                        set_global_var(98, 0);
-                        set_global_var(81, global_var(81) + 1);
-                        critter_mod_skill(dude_obj, 10, 1);
-                        display_msg(message_str(10, 410));
+                    if (global_var(GVAR_QUEST_FIND_THOMAS) >= (2 * get_pc_stat(PCSTAT_level))) then begin
+                        set_global_var(GVAR_QUEST_FIND_THOMAS, 0);
+                        set_global_var(GVAR_VAULT_CITIZENSHIP, global_var(GVAR_VAULT_CITIZENSHIP) + 1);
+                        critter_mod_skill(dude_obj, SKILL_STEAL, 1);
+                        display_msg(message_str(SCRIPT_STTEXT, 410));
                     end
                 end
             end
             steal_perk := 0;
             steal_gloves := 0;
-            if (has_trait(0, dude_obj, 37) != 0) then begin
+            if (has_trait(TRAIT_PERK, dude_obj, PERK_pickpocket) != 0) then begin
                 steal_perk := steal_perk + 15;
             end
-            if (has_trait(0, dude_obj, 33) != 0) then begin
+            if (has_trait(TRAIT_PERK, dude_obj, PERK_master_thief) != 0) then begin
                 steal_perk := steal_perk + 30;
             end
-            if ((obj_pid(critter_inven_obj(dude_obj, 1)) == 245) or (obj_pid(critter_inven_obj(dude_obj, 2)) == 245)) then begin
+            if ((obj_pid(critter_inven_obj(dude_obj, INVEN_TYPE_RIGHT_HAND)) == 245) or (obj_pid(critter_inven_obj(dude_obj, INVEN_TYPE_LEFT_HAND)) == 245)) then begin
                 steal_gloves := 10;
             end
-            if (random(0, 100) <= ((has_skill(dude_obj, 10) / 2) + (((3 * get_critter_stat(dude_obj, 5)) + (((get_critter_stat(dude_obj, 6) - 4) * 4) - (difficulty_level * 4)) + (2 * steal_perk) + steal_gloves) / (local_var(4) / 25)))) then begin
-                display_msg(message_str(10, random(145, 148)));
+            if (random(0, 100) <= ((dude_skill(SKILL_STEAL) / 2) + (((3 * get_critter_stat(dude_obj, STAT_ag)) + (((dude_luck - 4) * 4) - (difficulty_level * 4)) + (2 * steal_perk) + steal_gloves) / (local_var(4) / 25)))) then begin
+                display_msg(message_str(SCRIPT_STTEXT, random(145, 148)));
             end
             else begin
                 script_overrides;
-                if (random(0, 100) <= (25 - floor((((get_critter_stat(dude_obj, 6) - 4) * 4) - (difficulty_level * 4)) / 3))) then begin
-                    float_msg(self_obj, message_str(10, random(206, 209)), 2);
+                if (random(0, 100) <= (25 - floor((((dude_luck - 4) * 4) - (difficulty_level * 4)) / 3))) then begin
+                    float_msg(self_obj, message_str(SCRIPT_STTEXT, random(206, 209)), FLOAT_MSG_RED);
                     set_local_var(2, 1);
                 end
                 else begin
-                    display_msg(message_str(10, random(200, 203)));
+                    display_msg(message_str(SCRIPT_STTEXT, random(200, 203)));
                 end
             end
         end
     end
     if (action_being_used == 12) then begin
         script_overrides;
-        display_msg(message_str(10, 204) + ((has_skill(dude_obj, 10) / 2) + (((3 * get_critter_stat(dude_obj, 5)) + (((get_critter_stat(dude_obj, 6) - 4) * 4) - (difficulty_level * 4)) + (2 * steal_perk) + steal_gloves) / (local_var(4) / 25))) + message_str(10, 205));
+        display_msg(message_str(SCRIPT_STTEXT, 204) + ((dude_skill(SKILL_STEAL) / 2) + (((3 * get_critter_stat(dude_obj, STAT_ag)) + (((dude_luck - 4) * 4) - (difficulty_level * 4)) + (2 * steal_perk) + steal_gloves) / (local_var(4) / 25))) + message_str(SCRIPT_STTEXT, 205));
     end
 end
 
@@ -450,7 +450,7 @@ end
 procedure map_enter_p_proc
 begin
     set_local_var(7, 0);
-    if (not(metarule(22, 0)) == 1) then begin
+    if (not(is_loading_game) == 1) then begin
         move_to(self_obj, local_var(3), elevation(self_obj));
         if (local_var(8) > 0) then begin
             set_local_var(10, 1);
@@ -506,11 +506,10 @@ end
 
 procedure Node001
 begin
-    gsay_reply(516, random(10, 12));
-    gsay_option(516, 13, Node999, 50);
+    Reply(random(10, 12));
+    giq_option(1, NAME, 13, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node999
 begin
 end
-
