diff --git a/Mods/Olympus2207/Original/TARKATT3.ssl b/Mods/Olympus2207/ReDefine/TARKATT3.ssl
index 6f11046..1a61446 100644
--- a/Mods/Olympus2207/Original/TARKATT3.ssl
+++ b/Mods/Olympus2207/ReDefine/TARKATT3.ssl
@@ -91,8 +91,8 @@ begin
         set_local_var(4, random(40, 70));
         set_local_var(5, random(50, 80));
         set_local_var(6, random(10, 30));
-        critter_add_trait(self_obj, 1, 6, 1);
-        critter_add_trait(self_obj, 1, 5, 4);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 1);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_TARTAR_WOMEN);
     end
 end
 
@@ -106,7 +106,7 @@ begin
         float_is_called := 1;
         add_timer_event(self_obj, random(200, 700), 2);
     end
-    if ((local_var(7) == 0) and (tile_distance(tile_num(self_obj), tile_num(dude_obj)) <= 2) and (elevation(self_obj) == 0)) then begin
+    if ((local_var(7) == 0) and (tile_distance(tile_num(self_obj), dude_tile) <= 2) and (elevation(self_obj) == 0)) then begin
         if (game_ui_is_disabled) then begin
             game_ui_enable;
         end
@@ -117,7 +117,7 @@ begin
         move_to(self_obj, 1, 1);
     end
     if (obj_can_see_obj(self_obj, dude_obj)) then begin
-        if ((local_var(2) == 1) or (global_var(47) <= -31)) then begin
+        if ((local_var(2) == 1) or (global_var(GVAR_TOWN_REP_ARROYO) <= -31)) then begin
             set_local_var(2, 0);
             attack_setup(self_obj, dude_obj);
         end
@@ -132,10 +132,10 @@ procedure description_p_proc
 begin
     script_overrides;
     if (local_var(7) == 0) then begin
-        display_msg(message_str(82, 2));
+        display_mstr(2);
     end
     else begin
-        display_msg(message_str(82, 4));
+        display_mstr(4);
     end
 end
 
@@ -143,10 +143,10 @@ procedure look_at_p_proc
 begin
     script_overrides;
     if (local_var(7) == 0) then begin
-        display_msg(message_str(82, 1));
+        display_mstr(1);
     end
     else begin
-        display_msg(message_str(82, 3));
+        display_mstr(3);
     end
 end
 
@@ -164,18 +164,18 @@ end
 procedure talk_p_proc
 begin
     if (not(anim_busy(self_obj))) then begin
-        anim(self_obj, 1000, rotation_to_tile(tile_num(self_obj), tile_num(dude_obj)));
+        anim(self_obj, 1000, rotation_to_tile(tile_num(self_obj), dude_tile));
     end
-    if ((global_var(6) == 2) and (local_var(7) == 1)) then begin
-        float_msg(self_obj, message_str(82, random(1030, 1033)), 0);
+    if ((global_var(GVAR_PLAYER_MARRIED) == 2) and (local_var(7) == 1)) then begin
+        float_msg(self_obj, mstr(random(1030, 1033)), FLOAT_MSG_NORMAL);
     end
     else begin
-        if ((global_var(6) == 3) and (local_var(9) == 1)) then begin
+        if ((global_var(GVAR_PLAYER_MARRIED) == 3) and (local_var(9) == 1)) then begin
             if (random(0, 100) <= 50) then begin
-                float_msg(self_obj, message_str(82, obj_name(dude_obj) + message_str(82, random(1034, 1036))), 0);
+                float_msg(self_obj, mstr(dude_name + mstr(random(1034, 1036))), FLOAT_MSG_NORMAL);
             end
             else begin
-                float_msg(self_obj, message_str(82, random(1037, 1039)), 0);
+                float_msg(self_obj, mstr(random(1037, 1039)), FLOAT_MSG_NORMAL);
             end
         end
         else begin
@@ -189,11 +189,11 @@ begin
                     call Node010();
                 end
                 else begin
-                    if (global_var(420) > 1) then begin
+                    if (global_var(GVAR_ENDGAME_MOVIE_NCR) > 1) then begin
                         call Node023();
                     end
                     else begin
-                        if ((obj_pid(critter_inven_obj(dude_obj, 0)) == 120) or (obj_pid(critter_inven_obj(dude_obj, 0)) == 121) or (obj_pid(critter_inven_obj(dude_obj, 0)) == 122) or (obj_pid(critter_inven_obj(dude_obj, 0)) == 123) or (obj_pid(critter_inven_obj(dude_obj, 0)) == 124) or (obj_pid(critter_inven_obj(dude_obj, 0)) == 125) or (obj_pid(critter_inven_obj(dude_obj, 0)) == 126) or (obj_pid(critter_inven_obj(dude_obj, 0)) == 127) or (obj_pid(critter_inven_obj(dude_obj, 0)) == 128) or (obj_pid(critter_inven_obj(dude_obj, 0)) == 129) or (obj_pid(critter_inven_obj(dude_obj, 0)) == 391) or (obj_pid(critter_inven_obj(dude_obj, 0)) == 394)) then begin
+                        if ((obj_pid(critter_inven_obj(dude_obj, INVEN_TYPE_WORN)) == 120) or (obj_pid(critter_inven_obj(dude_obj, INVEN_TYPE_WORN)) == 121) or (obj_pid(critter_inven_obj(dude_obj, INVEN_TYPE_WORN)) == 122) or (obj_pid(critter_inven_obj(dude_obj, INVEN_TYPE_WORN)) == 123) or (obj_pid(critter_inven_obj(dude_obj, INVEN_TYPE_WORN)) == 124) or (obj_pid(critter_inven_obj(dude_obj, INVEN_TYPE_WORN)) == 125) or (obj_pid(critter_inven_obj(dude_obj, INVEN_TYPE_WORN)) == 126) or (obj_pid(critter_inven_obj(dude_obj, INVEN_TYPE_WORN)) == 127) or (obj_pid(critter_inven_obj(dude_obj, INVEN_TYPE_WORN)) == 128) or (obj_pid(critter_inven_obj(dude_obj, INVEN_TYPE_WORN)) == 129) or (obj_pid(critter_inven_obj(dude_obj, INVEN_TYPE_WORN)) == 391) or (obj_pid(critter_inven_obj(dude_obj, INVEN_TYPE_WORN)) == 394)) then begin
                             call Node012();
                         end
                         else begin
@@ -211,12 +211,12 @@ end
 procedure destroy_p_proc
 begin
     if (source_obj == dude_obj) then begin
-        set_global_var(0, global_var(0) - 1);
+        set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) - 1);
         if ((47 == 50) or (47 == 59)) then begin
-            set_global_var(47, global_var(47) - 3);
+            set_global_var(GVAR_TOWN_REP_ARROYO, global_var(GVAR_TOWN_REP_ARROYO) - 3);
         end
         else begin
-            set_global_var(47, global_var(47) - 6);
+            set_global_var(GVAR_TOWN_REP_ARROYO, global_var(GVAR_TOWN_REP_ARROYO) - 6);
         end
     end
 end
@@ -229,51 +229,51 @@ begin
             set_sfall_global(13, 1);
         end
         else begin
-            if (has_trait(2, dude_obj, 13)) then begin
-                if (global_var(81) < get_pc_stat(1)) then begin
+            if (has_trait(TRAIT_TRAIT, dude_obj, TRAIT_sex_appeal)) then begin
+                if (global_var(GVAR_VAULT_CITIZENSHIP) < get_pc_stat(PCSTAT_level)) then begin
                     if (is_success(roll_vs_skill(dude_obj, 10, 0))) then begin
-                        set_global_var(98, global_var(98) + 3);
+                        set_global_var(GVAR_QUEST_FIND_THOMAS, global_var(GVAR_QUEST_FIND_THOMAS) + 3);
                     end
                     else begin
-                        set_global_var(98, global_var(98) + 1);
+                        set_global_var(GVAR_QUEST_FIND_THOMAS, global_var(GVAR_QUEST_FIND_THOMAS) + 1);
                     end
-                    if (global_var(98) >= (2 * get_pc_stat(1))) then begin
-                        set_global_var(98, 0);
-                        set_global_var(81, global_var(81) + 1);
-                        critter_mod_skill(dude_obj, 10, 1);
-                        display_msg(message_str(10, 410));
+                    if (global_var(GVAR_QUEST_FIND_THOMAS) >= (2 * get_pc_stat(PCSTAT_level))) then begin
+                        set_global_var(GVAR_QUEST_FIND_THOMAS, 0);
+                        set_global_var(GVAR_VAULT_CITIZENSHIP, global_var(GVAR_VAULT_CITIZENSHIP) + 1);
+                        critter_mod_skill(dude_obj, SKILL_STEAL, 1);
+                        display_msg(message_str(SCRIPT_STTEXT, 410));
                     end
                 end
             end
             steal_perk := 0;
             steal_gloves := 0;
-            if (has_trait(0, dude_obj, 37) != 0) then begin
+            if (has_trait(TRAIT_PERK, dude_obj, PERK_pickpocket) != 0) then begin
                 steal_perk := steal_perk + 15;
             end
-            if (has_trait(0, dude_obj, 33) != 0) then begin
+            if (has_trait(TRAIT_PERK, dude_obj, PERK_master_thief) != 0) then begin
                 steal_perk := steal_perk + 30;
             end
-            if ((obj_pid(critter_inven_obj(dude_obj, 1)) == 245) or (obj_pid(critter_inven_obj(dude_obj, 2)) == 245)) then begin
+            if ((obj_pid(critter_inven_obj(dude_obj, INVEN_TYPE_RIGHT_HAND)) == 245) or (obj_pid(critter_inven_obj(dude_obj, INVEN_TYPE_LEFT_HAND)) == 245)) then begin
                 steal_gloves := 10;
             end
-            if (random(0, 100) <= ((has_skill(dude_obj, 10) / 2) + (((3 * get_critter_stat(dude_obj, 5)) + (((get_critter_stat(dude_obj, 6) - 4) * 4) - (difficulty_level * 4)) + (2 * steal_perk) + steal_gloves) / (local_var(4) / 25)))) then begin
-                display_msg(message_str(10, random(145, 148)));
+            if (random(0, 100) <= ((dude_skill(SKILL_STEAL) / 2) + (((3 * get_critter_stat(dude_obj, STAT_ag)) + (((dude_luck - 4) * 4) - (difficulty_level * 4)) + (2 * steal_perk) + steal_gloves) / (local_var(4) / 25)))) then begin
+                display_msg(message_str(SCRIPT_STTEXT, random(145, 148)));
             end
             else begin
                 script_overrides;
-                if (random(0, 100) <= (25 - floor((((get_critter_stat(dude_obj, 6) - 4) * 4) - (difficulty_level * 4)) / 3))) then begin
-                    float_msg(self_obj, message_str(10, random(206, 209)), 2);
+                if (random(0, 100) <= (25 - floor((((dude_luck - 4) * 4) - (difficulty_level * 4)) / 3))) then begin
+                    float_msg(self_obj, message_str(SCRIPT_STTEXT, random(206, 209)), FLOAT_MSG_RED);
                     set_local_var(2, 1);
                 end
                 else begin
-                    display_msg(message_str(10, random(200, 203)));
+                    display_msg(message_str(SCRIPT_STTEXT, random(200, 203)));
                 end
             end
         end
     end
     if (action_being_used == 12) then begin
         script_overrides;
-        display_msg(message_str(10, 204) + ((has_skill(dude_obj, 10) / 2) + (((3 * get_critter_stat(dude_obj, 5)) + (((get_critter_stat(dude_obj, 6) - 4) * 4) - (difficulty_level * 4)) + (2 * steal_perk) + steal_gloves) / (local_var(4) / 25))) + message_str(10, 205));
+        display_msg(message_str(SCRIPT_STTEXT, 204) + ((dude_skill(SKILL_STEAL) / 2) + (((3 * get_critter_stat(dude_obj, STAT_ag)) + (((dude_luck - 4) * 4) - (difficulty_level * 4)) + (2 * steal_perk) + steal_gloves) / (local_var(4) / 25))) + message_str(SCRIPT_STTEXT, 205));
     end
 end
 
@@ -286,7 +286,7 @@ end
 
 procedure map_enter_p_proc
 begin
-    if (metarule(14, 0)) then begin
+    if (map_first_run) then begin
         game_ui_disable;
         reg_anim_func(2, self_obj);
         reg_anim_func(1, 1);
@@ -318,269 +318,269 @@ end
 procedure Node001x
 begin
     set_local_var(7, 1);
-    gsay_reply(82, message_str(82, 5) + obj_name(dude_obj) + message_str(82, 6));
-    gsay_option(82, 7, Node002x, 50);
-    gsay_option(82, 8, Node999, 50);
+    Reply(mstr(5) + dude_name + mstr(6));
+    giq_option(1, NAME, 7, Node002x, NEUTRAL_REACTION);
+    giq_option(1, NAME, 8, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node002x
 begin
-    gsay_reply(82, 9);
-    gsay_option(82, 8, Node999, 50);
+    Reply(9);
+    giq_option(1, NAME, 8, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node001
 begin
-    gsay_reply(82, 10);
-    gsay_option(82, 11, Node002, 50);
-    if (global_var(68) < 50) then begin
-        gsay_option(82, 12, Node002, 50);
+    Reply(10);
+    giq_option(1, NAME, 11, Node002, NEUTRAL_REACTION);
+    if (global_var(GVAR_ENEMY_KLAMATH) < 50) then begin
+        giq_option(1, NAME, 12, Node002, NEUTRAL_REACTION);
     end
 end
 
 procedure Node002
 begin
-    gsay_reply(82, 20);
-    gsay_option(82, 21, Node003, 50);
-    gsay_option(82, 22, Node004, 50);
-    gsay_option(82, 23, Node005, 50);
-    gsay_option(82, 24, Node003, 50);
+    Reply(20);
+    giq_option(1, NAME, 21, Node003, NEUTRAL_REACTION);
+    giq_option(1, NAME, 22, Node004, NEUTRAL_REACTION);
+    giq_option(1, NAME, 23, Node005, NEUTRAL_REACTION);
+    giq_option(1, NAME, 24, Node003, NEUTRAL_REACTION);
 end
 
 procedure Node003
 begin
-    if (global_var(68) < 50) then begin
-        gsay_reply(82, 30);
-        gsay_option(82, 32, Node006, 50);
+    if (global_var(GVAR_ENEMY_KLAMATH) < 50) then begin
+        Reply(30);
+        giq_option(1, NAME, 32, Node006, NEUTRAL_REACTION);
     end
     else begin
-        gsay_reply(82, 31);
-        gsay_option(82, 33, Node006, 50);
+        Reply(31);
+        giq_option(1, NAME, 33, Node006, NEUTRAL_REACTION);
     end
 end
 
 procedure Node004
 begin
-    gsay_reply(82, 40);
-    if (global_var(68) < 50) then begin
-        gsay_option(82, 41, Node003, 50);
+    Reply(40);
+    if (global_var(GVAR_ENEMY_KLAMATH) < 50) then begin
+        giq_option(1, NAME, 41, Node003, NEUTRAL_REACTION);
     end
     else begin
-        gsay_option(82, 42, Node003, 50);
+        giq_option(1, NAME, 42, Node003, NEUTRAL_REACTION);
     end
 end
 
 procedure Node005
 begin
-    gsay_reply(82, 50);
-    if (global_var(68) < 50) then begin
-        gsay_option(82, 51, Node003, 50);
+    Reply(50);
+    if (global_var(GVAR_ENEMY_KLAMATH) < 50) then begin
+        giq_option(1, NAME, 51, Node003, NEUTRAL_REACTION);
     end
     else begin
-        gsay_option(82, 52, Node003, 50);
+        giq_option(1, NAME, 52, Node003, NEUTRAL_REACTION);
     end
 end
 
 procedure Node006
 begin
-    gsay_reply(82, 60);
-    gsay_option(82, 61, Node007, 50);
+    Reply(60);
+    giq_option(1, NAME, 61, Node007, NEUTRAL_REACTION);
 end
 
 procedure Node007
 begin
     set_local_var(8, 1);
-    gsay_reply(82, 70);
-    gsay_option(82, 71, Node010, 50);
+    Reply(70);
+    giq_option(1, NAME, 71, Node010, NEUTRAL_REACTION);
 end
 
 procedure Node010
 begin
     if (already_talk == 0) then begin
         already_talk := 1;
-        gsay_reply(82, 100);
+        Reply(100);
     end
     else begin
-        gsay_reply(82, 101);
+        Reply(101);
     end
-    if (global_var(68) >= 50) then begin
-        gsay_option(82, 102, Node027, 50);
+    if (global_var(GVAR_ENEMY_KLAMATH) >= 50) then begin
+        giq_option(1, NAME, 102, Node027, NEUTRAL_REACTION);
     end
-    gsay_option(82, 103, Node030, 50);
-    if ((global_var(69) bwand 1) != 0) then begin
-        gsay_option(82, 104, Node035, 50);
+    giq_option(1, NAME, 103, Node030, NEUTRAL_REACTION);
+    if ((global_var(GVAR_TORR_HARMED) bwand 1) != 0) then begin
+        giq_option(1, NAME, 104, Node035, NEUTRAL_REACTION);
     end
-    gsay_option(82, 105, Node036, 50);
-    gsay_option(82, 106, Node055, 50);
-    if (not((global_var(69) bwand 1) != 0)) then begin
-        gsay_option(82, 107, Node011, 50);
+    giq_option(1, NAME, 105, Node036, NEUTRAL_REACTION);
+    giq_option(1, NAME, 106, Node055, NEUTRAL_REACTION);
+    if (not((global_var(GVAR_TORR_HARMED) bwand 1) != 0)) then begin
+        giq_option(1, NAME, 107, Node011, NEUTRAL_REACTION);
     end
-    gsay_option(82, 108, Node011, 50);
+    giq_option(1, NAME, 108, Node011, NEUTRAL_REACTION);
 end
 
 procedure Node011
 begin
-    gsay_reply(82, 110);
-    gsay_option(82, 111, Node999, 50);
+    Reply(110);
+    giq_option(1, NAME, 111, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node012
 begin
     set_local_var(8, 1);
-    gsay_reply(82, 120);
-    if (global_var(68) < 50) then begin
-        gsay_option(82, 121, Node013, 50);
-        gsay_option(82, 122, Node013, 50);
+    Reply(120);
+    if (global_var(GVAR_ENEMY_KLAMATH) < 50) then begin
+        giq_option(1, NAME, 121, Node013, NEUTRAL_REACTION);
+        giq_option(1, NAME, 122, Node013, NEUTRAL_REACTION);
     end
-    gsay_option(82, 123, Node018, 50);
-    if (global_var(68) >= 50) then begin
-        gsay_option(82, 124, Node013, 50);
-        gsay_option(82, 125, Node022, 50);
+    giq_option(1, NAME, 123, Node018, NEUTRAL_REACTION);
+    if (global_var(GVAR_ENEMY_KLAMATH) >= 50) then begin
+        giq_option(1, NAME, 124, Node013, NEUTRAL_REACTION);
+        giq_option(1, NAME, 125, Node022, NEUTRAL_REACTION);
     end
 end
 
 procedure Node013
 begin
-    gsay_reply(82, 130);
-    gsay_option(82, 131, Node014, 50);
-    gsay_option(82, 132, Node021, 50);
+    Reply(130);
+    giq_option(1, NAME, 131, Node014, NEUTRAL_REACTION);
+    giq_option(1, NAME, 132, Node021, NEUTRAL_REACTION);
 end
 
 procedure Node014
 begin
-    gsay_reply(82, 140);
-    gsay_option(82, 141, Node015, 50);
+    Reply(140);
+    giq_option(1, NAME, 141, Node015, NEUTRAL_REACTION);
 end
 
 procedure Node015
 begin
-    gsay_reply(82, 150);
-    gsay_option(82, 151, Node016, 50);
-    gsay_option(82, 152, Node016, 50);
-    gsay_option(82, 153, Node016, 50);
+    Reply(150);
+    giq_option(1, NAME, 151, Node016, NEUTRAL_REACTION);
+    giq_option(1, NAME, 152, Node016, NEUTRAL_REACTION);
+    giq_option(1, NAME, 153, Node016, NEUTRAL_REACTION);
 end
 
 procedure Node016
 begin
-    gsay_reply(82, 160);
-    gsay_option(82, 161, Node017, 50);
-    gsay_option(82, 162, Node017, 50);
-    gsay_option(82, 163, Node003, 50);
+    Reply(160);
+    giq_option(1, NAME, 161, Node017, NEUTRAL_REACTION);
+    giq_option(1, NAME, 162, Node017, NEUTRAL_REACTION);
+    giq_option(1, NAME, 163, Node003, NEUTRAL_REACTION);
 end
 
 procedure Node017
 begin
-    gsay_reply(82, 170);
-    gsay_option(82, 171, Node003, 50);
+    Reply(170);
+    giq_option(1, NAME, 171, Node003, NEUTRAL_REACTION);
 end
 
 procedure Node018
 begin
-    gsay_reply(82, 180);
-    gsay_option(82, 181, Node019, 50);
+    Reply(180);
+    giq_option(1, NAME, 181, Node019, NEUTRAL_REACTION);
 end
 
 procedure Node019
 begin
-    gsay_reply(82, 190);
-    gsay_option(82, 191, Node020, 50);
+    Reply(190);
+    giq_option(1, NAME, 191, Node020, NEUTRAL_REACTION);
 end
 
 procedure Node020
 begin
-    gsay_reply(82, 200);
-    gsay_option(82, 201, Node007, 50);
+    Reply(200);
+    giq_option(1, NAME, 201, Node007, NEUTRAL_REACTION);
 end
 
 procedure Node021
 begin
-    gsay_reply(82, 210);
-    gsay_option(82, 211, Node014, 50);
-    gsay_option(82, 212, Node014, 50);
+    Reply(210);
+    giq_option(1, NAME, 211, Node014, NEUTRAL_REACTION);
+    giq_option(1, NAME, 212, Node014, NEUTRAL_REACTION);
 end
 
 procedure Node022
 begin
-    gsay_reply(82, 220);
-    gsay_option(82, 221, Node018, 50);
+    Reply(220);
+    giq_option(1, NAME, 221, Node018, NEUTRAL_REACTION);
 end
 
 procedure Node023
 begin
     set_local_var(8, 1);
-    gsay_reply(82, 230);
-    gsay_option(82, 231, Node024, 50);
+    Reply(230);
+    giq_option(1, NAME, 231, Node024, NEUTRAL_REACTION);
 end
 
 procedure Node024
 begin
-    gsay_reply(82, 240);
-    gsay_option(82, 241, Node025, 50);
-    gsay_option(82, 242, Node025, 50);
-    gsay_option(82, 243, Node025, 50);
+    Reply(240);
+    giq_option(1, NAME, 241, Node025, NEUTRAL_REACTION);
+    giq_option(1, NAME, 242, Node025, NEUTRAL_REACTION);
+    giq_option(1, NAME, 243, Node025, NEUTRAL_REACTION);
 end
 
 procedure Node025
 begin
-    gsay_reply(82, 250);
-    gsay_option(82, 251, Node026, 50);
+    Reply(250);
+    giq_option(1, NAME, 251, Node026, NEUTRAL_REACTION);
 end
 
 procedure Node026
 begin
     set_local_var(10, 1);
-    gsay_reply(82, 260);
-    gsay_option(82, 261, Node999, 50);
+    Reply(260);
+    giq_option(1, NAME, 261, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node027
 begin
-    gsay_reply(82, 270);
-    gsay_option(82, 271, Node028, 50);
-    gsay_option(82, 272, Node029, 50);
+    Reply(270);
+    giq_option(1, NAME, 271, Node028, NEUTRAL_REACTION);
+    giq_option(1, NAME, 272, Node029, NEUTRAL_REACTION);
 end
 
 procedure Node028
 begin
-    gsay_reply(82, 280);
-    gsay_option(82, 281, Node010, 50);
+    Reply(280);
+    giq_option(1, NAME, 281, Node010, NEUTRAL_REACTION);
 end
 
 procedure Node029
 begin
-    gsay_reply(82, 290);
-    gsay_option(82, 291, Node010, 50);
+    Reply(290);
+    giq_option(1, NAME, 291, Node010, NEUTRAL_REACTION);
 end
 
 procedure Node030
 begin
     if ((game_time_in_seconds - local_var(11)) < (3600 * 24 * 7)) then begin
         set_local_var(11, game_time_in_seconds);
-        gsay_reply(82, 300);
-        gsay_option(82, 304, Node010, 50);
-        gsay_option(82, 305, Node010, 50);
+        Reply(300);
+        giq_option(1, NAME, 304, Node010, NEUTRAL_REACTION);
+        giq_option(1, NAME, 305, Node010, NEUTRAL_REACTION);
     end
     else begin
         set_local_var(11, game_time_in_seconds);
         if (local_var(12) == 0) then begin
             set_local_var(12, local_var(12) + 1);
-            gsay_reply(82, 301);
-            gsay_option(82, 306, Node031, 50);
-            gsay_option(82, 307, Node032, 50);
+            Reply(301);
+            giq_option(1, NAME, 306, Node031, NEUTRAL_REACTION);
+            giq_option(1, NAME, 307, Node032, NEUTRAL_REACTION);
         end
         else begin
             if (local_var(12) == 1) then begin
                 set_local_var(12, local_var(12) + 1);
-                gsay_reply(82, 302);
-                gsay_option(82, 308, Node031, 50);
-                gsay_option(82, 309, Node032, 50);
+                Reply(302);
+                giq_option(1, NAME, 308, Node031, NEUTRAL_REACTION);
+                giq_option(1, NAME, 309, Node032, NEUTRAL_REACTION);
             end
             else begin
                 if (local_var(12) == 3) then begin
                     set_local_var(12, local_var(12) + 1);
-                    gsay_reply(82, 303);
-                    gsay_option(82, 3010, Node032, 50);
-                    gsay_option(82, 3011, Node031, 50);
+                    Reply(303);
+                    giq_option(1, NAME, 3010, Node032, NEUTRAL_REACTION);
+                    giq_option(1, NAME, 3011, Node031, NEUTRAL_REACTION);
                 end
             end
         end
@@ -589,201 +589,201 @@ end
 
 procedure Node031
 begin
-    gsay_reply(82, 310);
-    gsay_option(82, 311, Node010, 50);
-    gsay_option(82, 312, Node999, 50);
+    Reply(310);
+    giq_option(1, NAME, 311, Node010, NEUTRAL_REACTION);
+    giq_option(1, NAME, 312, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node032
 begin
-    gsay_reply(82, 320);
-    gsay_option(82, 321, Node033, 50);
-    gsay_option(82, 322, Node034, 50);
-    gsay_option(82, 323, Node034, 50);
+    Reply(320);
+    giq_option(1, NAME, 321, Node033, NEUTRAL_REACTION);
+    giq_option(1, NAME, 322, Node034, NEUTRAL_REACTION);
+    giq_option(1, NAME, 323, Node034, NEUTRAL_REACTION);
 end
 
 procedure Node033
 begin
-    gsay_reply(82, 330);
-    gsay_option(82, 331, Node010, 50);
+    Reply(330);
+    giq_option(1, NAME, 331, Node010, NEUTRAL_REACTION);
 end
 
 procedure Node034
 begin
-    gsay_reply(82, 340);
-    gsay_option(82, 341, Node010, 50);
+    Reply(340);
+    giq_option(1, NAME, 341, Node010, NEUTRAL_REACTION);
 end
 
 procedure Node035
 begin
-    gsay_reply(82, 350);
-    gsay_option(82, 351, Node999, 50);
-    gsay_option(82, 352, Node010, 50);
+    Reply(350);
+    giq_option(1, NAME, 351, Node999, NEUTRAL_REACTION);
+    giq_option(1, NAME, 352, Node010, NEUTRAL_REACTION);
 end
 
 procedure Node036
 begin
-    gsay_reply(82, 360);
-    gsay_option(82, 361, Node037, 50);
-    if (global_var(244) == 2) then begin
-        gsay_option(82, 362, Node038, 50);
+    Reply(360);
+    giq_option(1, NAME, 361, Node037, NEUTRAL_REACTION);
+    if (global_var(GVAR_8_BALL_TRASH_SECRET) == 2) then begin
+        giq_option(1, NAME, 362, Node038, NEUTRAL_REACTION);
     end
-    if (global_var(144) >= 2) then begin
-        gsay_option(82, 363, Node039, 50);
+    if (global_var(GVAR_VAULT_MONSTER_COUNT) >= 2) then begin
+        giq_option(1, NAME, 363, Node039, NEUTRAL_REACTION);
     end
-    if (global_var(267) == 1) then begin
-        gsay_option(82, 364, Node044, 50);
+    if (global_var(GVAR_NCR_CATTLE_TIME_MIN) == 1) then begin
+        giq_option(1, NAME, 364, Node044, NEUTRAL_REACTION);
     end
-    if (global_var(248) >= 3) then begin
-        gsay_option(82, 365, Node047, 50);
+    if (global_var(GVAR_NEW_RENO_MCGEE_ATTACKED) >= 3) then begin
+        giq_option(1, NAME, 365, Node047, NEUTRAL_REACTION);
     end
-    if (global_var(173) >= 2) then begin
-        gsay_option(82, 366, Node048, 50);
+    if (global_var(GVAR_NCR_V15_DARION_DEAL) >= 2) then begin
+        giq_option(1, NAME, 366, Node048, NEUTRAL_REACTION);
     end
-    if (global_var(370) > 0) then begin
-        gsay_option(82, 367, Node050, 50);
+    if (global_var(GVAR_NEW_RENO_JET_SOURCE) > 0) then begin
+        giq_option(1, NAME, 367, Node050, NEUTRAL_REACTION);
     end
-    gsay_option(82, 368, Node010, 50);
+    giq_option(1, NAME, 368, Node010, NEUTRAL_REACTION);
 end
 
 procedure Node037
 begin
-    gsay_reply(82, 370);
-    gsay_option(82, 371, Node036, 50);
-    gsay_option(82, 372, Node010, 50);
+    Reply(370);
+    giq_option(1, NAME, 371, Node036, NEUTRAL_REACTION);
+    giq_option(1, NAME, 372, Node010, NEUTRAL_REACTION);
 end
 
 procedure Node038
 begin
-    gsay_reply(82, 380);
-    gsay_option(82, 381, Node041, 50);
-    gsay_option(82, 382, Node010, 50);
+    Reply(380);
+    giq_option(1, NAME, 381, Node041, NEUTRAL_REACTION);
+    giq_option(1, NAME, 382, Node010, NEUTRAL_REACTION);
 end
 
 procedure Node039
 begin
-    gsay_reply(82, 390);
-    gsay_option(82, 391, Node040, 50);
+    Reply(390);
+    giq_option(1, NAME, 391, Node040, NEUTRAL_REACTION);
 end
 
 procedure Node040
 begin
-    gsay_reply(82, 400);
-    if (global_var(221) == 2) then begin
-        gsay_option(82, 401, Node041, 50);
+    Reply(400);
+    if (global_var(GVAR_HOLODISK_MB_LEVEL_1) == 2) then begin
+        giq_option(1, NAME, 401, Node041, NEUTRAL_REACTION);
     end
-    if (global_var(221) == 3) then begin
-        gsay_option(82, 402, Node042, 50);
-        gsay_option(82, 403, Node042, 50);
+    if (global_var(GVAR_HOLODISK_MB_LEVEL_1) == 3) then begin
+        giq_option(1, NAME, 402, Node042, NEUTRAL_REACTION);
+        giq_option(1, NAME, 403, Node042, NEUTRAL_REACTION);
     end
 end
 
 procedure Node041
 begin
-    gsay_reply(82, 410);
-    gsay_option(82, 411, Node037, 50);
+    Reply(410);
+    giq_option(1, NAME, 411, Node037, NEUTRAL_REACTION);
 end
 
 procedure Node042
 begin
-    gsay_reply(82, 420);
-    gsay_option(82, 421, Node043, 50);
-    gsay_option(82, 422, Node037, 50);
+    Reply(420);
+    giq_option(1, NAME, 421, Node043, NEUTRAL_REACTION);
+    giq_option(1, NAME, 422, Node037, NEUTRAL_REACTION);
 end
 
 procedure Node043
 begin
-    gsay_reply(82, 430);
-    gsay_option(82, 431, Node037, 50);
-    gsay_option(82, 432, Node010, 50);
+    Reply(430);
+    giq_option(1, NAME, 431, Node037, NEUTRAL_REACTION);
+    giq_option(1, NAME, 432, Node010, NEUTRAL_REACTION);
 end
 
 procedure Node044
 begin
-    gsay_reply(82, 440);
-    gsay_option(82, 441, Node045, 50);
+    Reply(440);
+    giq_option(1, NAME, 441, Node045, NEUTRAL_REACTION);
 end
 
 procedure Node045
 begin
-    gsay_reply(82, 450);
-    gsay_option(82, 451, Node046, 50);
-    gsay_option(82, 452, Node046, 50);
+    Reply(450);
+    giq_option(1, NAME, 451, Node046, NEUTRAL_REACTION);
+    giq_option(1, NAME, 452, Node046, NEUTRAL_REACTION);
 end
 
 procedure Node046
 begin
-    gsay_reply(82, 460);
-    gsay_option(82, 461, Node036, 50);
-    gsay_option(82, 462, Node010, 50);
+    Reply(460);
+    giq_option(1, NAME, 461, Node036, NEUTRAL_REACTION);
+    giq_option(1, NAME, 462, Node010, NEUTRAL_REACTION);
 end
 
 procedure Node047
 begin
-    gsay_reply(82, 470);
-    gsay_option(82, 471, Node037, 50);
-    gsay_option(82, 472, Node010, 50);
+    Reply(470);
+    giq_option(1, NAME, 471, Node037, NEUTRAL_REACTION);
+    giq_option(1, NAME, 472, Node010, NEUTRAL_REACTION);
 end
 
 procedure Node048
 begin
-    gsay_reply(82, 480);
-    gsay_option(82, 481, Node049, 50);
+    Reply(480);
+    giq_option(1, NAME, 481, Node049, NEUTRAL_REACTION);
 end
 
 procedure Node049
 begin
-    if (has_skill(dude_obj, 14) > 199) then begin
-        gsay_reply(82, 490);
-        gsay_option(82, 492, Node037, 50);
+    if (dude_skill(SKILL_SPEECH) > 199) then begin
+        Reply(490);
+        giq_option(1, NAME, 492, Node037, NEUTRAL_REACTION);
     end
     else begin
-        gsay_reply(82, 491);
-        gsay_option(82, 493, Node036, 50);
-        gsay_option(82, 494, Node010, 50);
+        Reply(491);
+        giq_option(1, NAME, 493, Node036, NEUTRAL_REACTION);
+        giq_option(1, NAME, 494, Node010, NEUTRAL_REACTION);
     end
 end
 
 procedure Node050
 begin
-    gsay_reply(82, 500);
-    gsay_option(82, 501, Node051, 50);
+    Reply(500);
+    giq_option(1, NAME, 501, Node051, NEUTRAL_REACTION);
 end
 
 procedure Node051
 begin
-    gsay_reply(82, 510);
-    gsay_option(82, 511, Node052, 50);
-    gsay_option(82, 512, Node054, 50);
+    Reply(510);
+    giq_option(1, NAME, 511, Node052, NEUTRAL_REACTION);
+    giq_option(1, NAME, 512, Node054, NEUTRAL_REACTION);
 end
 
 procedure Node052
 begin
     set_local_var(9, 1);
-    gsay_reply(82, 520);
-    gsay_option(82, 521, Node053, 50);
-    gsay_option(82, 522, Node999, 50);
+    Reply(520);
+    giq_option(1, NAME, 521, Node053, NEUTRAL_REACTION);
+    giq_option(1, NAME, 522, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node053
 begin
     set_local_var(9, 1);
-    gsay_reply(82, 530);
-    gsay_option(82, 531, Node999, 50);
+    Reply(530);
+    giq_option(1, NAME, 531, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node054
 begin
     set_local_var(9, 1);
-    gsay_reply(82, 540);
-    gsay_option(82, 541, Node053, 50);
-    gsay_option(82, 542, Node999, 50);
+    Reply(540);
+    giq_option(1, NAME, 541, Node053, NEUTRAL_REACTION);
+    giq_option(1, NAME, 542, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node055
 begin
-    gsay_reply(82, 550);
-    gsay_option(82, 551, Node010, 50);
+    Reply(550);
+    giq_option(1, NAME, 551, Node010, NEUTRAL_REACTION);
 end
 
 procedure Node999
@@ -802,4 +802,3 @@ begin
     gdialog_mod_barter(-(dude_objective_reputation + local_var(6)));
     call Node010();
 end
-
