diff --git a/Mods/Olympus2207/Original/TBUZMCHN.ssl b/Mods/Olympus2207/ReDefine/TBUZMCHN.ssl
index 39dba89..c3c493e 100644
--- a/Mods/Olympus2207/Original/TBUZMCHN.ssl
+++ b/Mods/Olympus2207/ReDefine/TBUZMCHN.ssl
@@ -47,10 +47,10 @@ begin
     reg_anim_func(3, 0);
     if (local_var(1) == local_var(2)) then begin
         critter_dmg(dude_obj, random(1, 4), 0 bwor 256);
-        display_msg(message_str(12, 125));
+        display_msg(message_str(SCRIPT_ITTEXT, 125));
     end
     else begin
-        display_msg(message_str(12, 124));
+        display_msg(message_str(SCRIPT_ITTEXT, 124));
     end
 end
 
@@ -61,17 +61,17 @@ begin
     if ((obj_pid(object) == 486) and (map_var(0) == 0)) then begin
         rm_obj_from_inven(dude_obj, obj_carrying_pid_obj(dude_obj, 486));
         if (local_var(1) > 5) then begin
-            add_mult_objs_to_inven(dude_obj, create_object_sid(434, 1, 2, -1), local_var(1) - 5);
-            display_msg(message_str(12, 128) + (local_var(1) - 5) + message_str(12, 129));
+            add_mult_objs_to_inven(dude_obj, create_object(434, 1, 2), local_var(1) - 5);
+            display_msg(message_str(SCRIPT_ITTEXT, 128) + (local_var(1) - 5) + message_str(SCRIPT_ITTEXT, 129));
             set_local_var(1, 5);
         end
         set_local_var(2, 5);
         set_map_var(0, 1);
-        display_msg(message_str(12, 127));
+        display_msg(message_str(SCRIPT_ITTEXT, 127));
     end
     if (obj_pid(object) == 434) then begin
         if (local_var(1) == local_var(2)) then begin
-            display_msg(message_str(12, 120));
+            display_msg(message_str(SCRIPT_ITTEXT, 120));
         end
         else begin
             if (obj_is_carrying_obj_pid(dude_obj, 434) > (local_var(2) - local_var(1))) then begin
@@ -92,10 +92,10 @@ begin
             reg_anim_animate(dude_obj, 11, -1);
             reg_anim_func(3, 0);
             if (local_var(1) < local_var(2)) then begin
-                display_msg(message_str(12, 121) + (local_var(2) - local_var(1)) + message_str(12, 122));
+                display_msg(message_str(SCRIPT_ITTEXT, 121) + (local_var(2) - local_var(1)) + message_str(SCRIPT_ITTEXT, 122));
             end
             else begin
-                display_msg(message_str(12, 123));
+                display_msg(message_str(SCRIPT_ITTEXT, 123));
             end
         end
     end
@@ -108,11 +108,11 @@ begin
             reg_anim_animate(dude_obj, 11, -1);
             reg_anim_func(3, 0);
             rm_obj_from_inven(dude_obj, obj_carrying_pid_obj(dude_obj, 414));
-            add_obj_to_inven(dude_obj, create_object_sid(297, 1, 2, -1));
-            display_msg(message_str(12, 126));
+            add_obj_to_inven(dude_obj, create_object(297, 1, 2));
+            display_msg(message_str(SCRIPT_ITTEXT, 126));
         end
         else begin
-            display_msg(message_str(12, 124));
+            display_msg(message_str(SCRIPT_ITTEXT, 124));
         end
     end
 end
@@ -144,4 +144,3 @@ end
 procedure talk_p_proc
 begin
 end
-
