diff --git a/Mods/Olympus2207/Original/TDUDEBOX.ssl b/Mods/Olympus2207/ReDefine/TDUDEBOX.ssl
index a7a9ebd..013652d 100644
--- a/Mods/Olympus2207/Original/TDUDEBOX.ssl
+++ b/Mods/Olympus2207/ReDefine/TDUDEBOX.ssl
@@ -49,15 +49,15 @@ end
 
 procedure map_enter_p_proc
 begin
-    if (metarule(14, 0)) then begin
+    if (map_first_run) then begin
         if (obj_carrying_pid_obj(dude_obj, 393)) then begin
             rm_obj_from_inven(dude_obj, obj_carrying_pid_obj(dude_obj, 393));
             move_obj_inven_to_obj(dude_obj, self_obj);
-            add_obj_to_inven(dude_obj, create_object_sid(393, 1, 2, -1));
+            add_obj_to_inven(dude_obj, create_object(393, 1, 2));
         end
         else begin
             move_obj_inven_to_obj(dude_obj, self_obj);
-            add_obj_to_inven(dude_obj, create_object_sid(393, 1, 2, -1));
+            add_obj_to_inven(dude_obj, create_object(393, 1, 2));
         end
     end
 end
@@ -69,4 +69,3 @@ end
 procedure talk_p_proc
 begin
 end
-
