diff --git a/Mods/Olympus2207/Original/TDUDEBRT.ssl b/Mods/Olympus2207/ReDefine/TDUDEBRT.ssl
index 0ffe074..f01e697 100644
--- a/Mods/Olympus2207/Original/TDUDEBRT.ssl
+++ b/Mods/Olympus2207/ReDefine/TDUDEBRT.ssl
@@ -43,54 +43,54 @@ procedure start
 begin
     if (local_var(1) == 0) then begin
         if (random(0, 100) <= 25) then begin
-            add_mult_objs_to_inven(self_obj, create_object_sid(410, 1, 2, -1), random(1, 4));
+            add_mult_objs_to_inven(self_obj, create_object(410, 1, 2), random(1, 4));
         end
         if (random(0, 100) <= 15) then begin
-            add_mult_objs_to_inven(self_obj, create_object_sid(411, 1, 2, -1), random(1, 4));
+            add_mult_objs_to_inven(self_obj, create_object(411, 1, 2), random(1, 4));
         end
         if (random(0, 100) <= 10) then begin
-            add_obj_to_inven(self_obj, create_object_sid(416, 1, 2, -1));
+            add_obj_to_inven(self_obj, create_object(416, 1, 2));
         end
         if (random(0, 100) <= 10) then begin
-            add_obj_to_inven(self_obj, create_object_sid(417, 1, 2, -1));
+            add_obj_to_inven(self_obj, create_object(417, 1, 2));
         end
         if (random(0, 100) <= 10) then begin
-            add_obj_to_inven(self_obj, create_object_sid(418, 1, 2, -1));
+            add_obj_to_inven(self_obj, create_object(418, 1, 2));
         end
         if (random(0, 100) <= 15) then begin
-            add_obj_to_inven(self_obj, create_object_sid(424, 1, 2, -1));
+            add_obj_to_inven(self_obj, create_object(424, 1, 2));
         end
         if (random(0, 100) <= 15) then begin
-            add_obj_to_inven(self_obj, create_object_sid(422, 1, 2, -1));
+            add_obj_to_inven(self_obj, create_object(422, 1, 2));
         end
         if (random(0, 100) <= 15) then begin
-            add_obj_to_inven(self_obj, create_object_sid(432, 1, 2, -1));
+            add_obj_to_inven(self_obj, create_object(432, 1, 2));
         end
         if (random(0, 100) <= 15) then begin
-            add_obj_to_inven(self_obj, create_object_sid(429, 1, 2, -1));
+            add_obj_to_inven(self_obj, create_object(429, 1, 2));
         end
         if (random(0, 100) <= 25) then begin
-            add_mult_objs_to_inven(self_obj, create_object_sid(213, 1, 2, -1), random(1, 2));
+            add_mult_objs_to_inven(self_obj, create_object(213, 1, 2), random(1, 2));
         end
         if (random(0, 100) <= 10) then begin
-            add_obj_to_inven(self_obj, create_object_sid(217, 1, 2, -1));
+            add_obj_to_inven(self_obj, create_object(217, 1, 2));
         end
         if (random(0, 100) <= 10) then begin
-            add_obj_to_inven(self_obj, create_object_sid(108, 1, 2, -1));
+            add_obj_to_inven(self_obj, create_object(108, 1, 2));
         end
         if (random(0, 100) <= 5) then begin
-            add_obj_to_inven(self_obj, create_object_sid(39, 1, 2, -1));
+            add_obj_to_inven(self_obj, create_object(39, 1, 2));
         end
         if (random(0, 100) <= 15) then begin
-            add_obj_to_inven(self_obj, create_object_sid(220, 1, 2, -1));
+            add_obj_to_inven(self_obj, create_object(220, 1, 2));
         end
         set_local_var(3, tile_num(self_obj));
         set_local_var(1, 1);
         set_local_var(4, random(40, 60));
         set_local_var(5, random(50, 70));
         set_local_var(6, random(20, 35));
-        critter_add_trait(self_obj, 1, 6, 0);
-        critter_add_trait(self_obj, 1, 5, 1);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 0);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_LITTLE_BOY);
         max_distance := random(6, 8);
     end
 end
@@ -98,12 +98,12 @@ end
 procedure critter_p_proc
 begin
     if ((local_var(8) == 1) and (anim_busy(self_obj) == 0) and not(combat_is_initialized)) then begin
-        if ((tile_distance(tile_num(dude_obj), tile_num(self_obj)) > 3) and (tile_distance(tile_num(dude_obj), tile_num(self_obj)) <= 7)) then begin
-            animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(dude_obj), rotation_to_tile(tile_num(dude_obj), tile_num(self_obj)), random(2, 4)), 0);
+        if ((tile_distance(dude_tile, tile_num(self_obj)) > 3) and (tile_distance(dude_tile, tile_num(self_obj)) <= 7)) then begin
+            animate_move_obj_to_tile(self_obj, tile_num_in_direction(dude_tile, rotation_to_tile(dude_tile, tile_num(self_obj)), random(2, 4)), 0);
         end
         else begin
-            if (tile_distance(tile_num(dude_obj), tile_num(self_obj)) > 7) then begin
-                animate_move_obj_to_tile(self_obj, tile_num_in_direction(tile_num(dude_obj), rotation_to_tile(tile_num(dude_obj), tile_num(self_obj)), random(2, 4)), 1);
+            if (tile_distance(dude_tile, tile_num(self_obj)) > 7) then begin
+                animate_move_obj_to_tile(self_obj, tile_num_in_direction(dude_tile, rotation_to_tile(dude_tile, tile_num(self_obj)), random(2, 4)), 1);
             end
         end
     end
@@ -117,12 +117,12 @@ begin
         set_local_var(2, 0);
         attack_setup(self_obj, dude_obj);
     end
-    if ((global_var(28) == 2) and (local_var(8) == 1) and (local_var(9) == 0)) then begin
+    if ((global_var(GVAR_RUNNING_BURNING_GUY) == 2) and (local_var(8) == 1) and (local_var(9) == 0)) then begin
         set_local_var(8, 0);
-        give_exp_points(floor(240 / 5) + (get_critter_stat(dude_obj, 4) * floor(240 / 12)) + (get_critter_stat(dude_obj, 6) * floor(240 / 30)) + (floor(240 / 75) * (((get_critter_stat(dude_obj, 6) - 4) * 4) - (difficulty_level * 4))));
-        display_msg(message_str(10, random(140, 143)) + (floor(240 / 5) + (get_critter_stat(dude_obj, 4) * floor(240 / 12)) + (get_critter_stat(dude_obj, 6) * floor(240 / 30)) + (floor(240 / 75) * (((get_critter_stat(dude_obj, 6) - 4) * 4) - (difficulty_level * 4)))) + message_str(10, 144));
+        give_exp_points(floor(240 / 5) + (dude_iq * floor(240 / 12)) + (dude_luck * floor(240 / 30)) + (floor(240 / 75) * (((dude_luck - 4) * 4) - (difficulty_level * 4))));
+        display_msg(message_str(SCRIPT_STTEXT, random(140, 143)) + (floor(240 / 5) + (dude_iq * floor(240 / 12)) + (dude_luck * floor(240 / 30)) + (floor(240 / 75) * (((dude_luck - 4) * 4) - (difficulty_level * 4)))) + message_str(SCRIPT_STTEXT, 144));
         set_local_var(9, 1);
-        float_msg(self_obj, message_str(66, 1030), 3);
+        float_msg(self_obj, mstr(1030), FLOAT_MSG_GREEN);
     end
 end
 
@@ -134,10 +134,10 @@ procedure description_p_proc
 begin
     script_overrides;
     if (local_var(7) == 0) then begin
-        display_msg(message_str(66, 2));
+        display_mstr(2);
     end
     else begin
-        display_msg(message_str(66, 4));
+        display_mstr(4);
     end
 end
 
@@ -145,10 +145,10 @@ procedure look_at_p_proc
 begin
     script_overrides;
     if (local_var(7) == 0) then begin
-        display_msg(message_str(66, 1));
+        display_mstr(1);
     end
     else begin
-        display_msg(message_str(66, 3));
+        display_mstr(3);
     end
 end
 
@@ -187,26 +187,26 @@ begin
     if (action_being_used == 10) then begin
         steal_perk := 0;
         steal_gloves := 0;
-        if (has_trait(0, dude_obj, 37) != 0) then begin
+        if (has_trait(TRAIT_PERK, dude_obj, PERK_pickpocket) != 0) then begin
             steal_perk := steal_perk + 15;
         end
-        if (has_trait(0, dude_obj, 33) != 0) then begin
+        if (has_trait(TRAIT_PERK, dude_obj, PERK_master_thief) != 0) then begin
             steal_perk := steal_perk + 30;
         end
-        if ((obj_pid(critter_inven_obj(dude_obj, 1)) == 245) or (obj_pid(critter_inven_obj(dude_obj, 2)) == 245)) then begin
+        if ((obj_pid(critter_inven_obj(dude_obj, INVEN_TYPE_RIGHT_HAND)) == 245) or (obj_pid(critter_inven_obj(dude_obj, INVEN_TYPE_LEFT_HAND)) == 245)) then begin
             steal_gloves := 10;
         end
-        if (random(0, 100) <= ((has_skill(dude_obj, 10) / 2) + (((3 * get_critter_stat(dude_obj, 5)) + (((get_critter_stat(dude_obj, 6) - 4) * 4) - (difficulty_level * 4)) + (2 * steal_perk) + steal_gloves) / (local_var(4) / 25)))) then begin
-            display_msg(message_str(10, random(145, 148)));
+        if (random(0, 100) <= ((dude_skill(SKILL_STEAL) / 2) + (((3 * get_critter_stat(dude_obj, STAT_ag)) + (((dude_luck - 4) * 4) - (difficulty_level * 4)) + (2 * steal_perk) + steal_gloves) / (local_var(4) / 25)))) then begin
+            display_msg(message_str(SCRIPT_STTEXT, random(145, 148)));
         end
         else begin
             script_overrides;
-            display_msg(message_str(10, random(200, 203)));
+            display_msg(message_str(SCRIPT_STTEXT, random(200, 203)));
         end
     end
     if (action_being_used == 12) then begin
         script_overrides;
-        display_msg(message_str(10, 204) + ((has_skill(dude_obj, 10) / 2) + (((3 * get_critter_stat(dude_obj, 5)) + (((get_critter_stat(dude_obj, 6) - 4) * 4) - (difficulty_level * 4)) + (2 * steal_perk) + steal_gloves) / (local_var(4) / 25))) + message_str(10, 205));
+        display_msg(message_str(SCRIPT_STTEXT, 204) + ((dude_skill(SKILL_STEAL) / 2) + (((3 * get_critter_stat(dude_obj, STAT_ag)) + (((dude_luck - 4) * 4) - (difficulty_level * 4)) + (2 * steal_perk) + steal_gloves) / (local_var(4) / 25))) + message_str(SCRIPT_STTEXT, 205));
     end
 end
 
@@ -287,105 +287,105 @@ end
 
 procedure Node010
 begin
-    gsay_reply(66, 100);
-    gsay_option(66, 101, Node999, 50);
-    gsay_option(66, 102, Node011, 50);
-    if (global_var(28) == 0) then begin
-        gsay_option(66, 103, Node012, 50);
+    Reply(100);
+    giq_option(1, NAME, 101, Node999, NEUTRAL_REACTION);
+    giq_option(1, NAME, 102, Node011, NEUTRAL_REACTION);
+    if (global_var(GVAR_RUNNING_BURNING_GUY) == 0) then begin
+        giq_option(1, NAME, 103, Node012, NEUTRAL_REACTION);
     end
-    if ((global_var(27) == 1) and (obj_carrying_pid_obj(dude_obj, 290) == 0)) then begin
-        gsay_option(66, 104, Node016, 50);
+    if ((global_var(GVAR_LOAD_MAP_INDEX) == 1) and (obj_carrying_pid_obj(dude_obj, 290) == 0)) then begin
+        giq_option(1, NAME, 104, Node016, NEUTRAL_REACTION);
     end
-    if ((global_var(28) != 0) and (local_var(8) == 0) and (local_var(9) == 0)) then begin
-        gsay_option(66, 105, Node014, 50);
+    if ((global_var(GVAR_RUNNING_BURNING_GUY) != 0) and (local_var(8) == 0) and (local_var(9) == 0)) then begin
+        giq_option(1, NAME, 105, Node014, NEUTRAL_REACTION);
     end
 end
 
 procedure Node011
 begin
-    if (global_var(29) != 0) then begin
-        gsay_reply(66, 1101);
-        gsay_option(66, 111, Node999, 50);
+    if (global_var(GVAR_VIC_DEVICE) != 0) then begin
+        Reply(1101);
+        giq_option(1, NAME, 111, Node999, NEUTRAL_REACTION);
     end
     else begin
-        gsay_reply(66, 1102);
-        gsay_option(66, 112, Node999, 50);
+        Reply(1102);
+        giq_option(1, NAME, 112, Node999, NEUTRAL_REACTION);
     end
 end
 
 procedure Node012
 begin
-    gsay_reply(66, 120);
-    gsay_option(66, 121, Node999, 50);
-    gsay_option(66, 122, Node013, 50);
-    gsay_option(66, 123, Node013, 50);
+    Reply(120);
+    giq_option(1, NAME, 121, Node999, NEUTRAL_REACTION);
+    giq_option(1, NAME, 122, Node013, NEUTRAL_REACTION);
+    giq_option(1, NAME, 123, Node013, NEUTRAL_REACTION);
     if (local_var(8) == 0) then begin
-        gsay_option(66, 124, Node018, 50);
+        giq_option(1, NAME, 124, Node018, NEUTRAL_REACTION);
     end
 end
 
 procedure Node013
 begin
-    if (global_var(28) == 0) then begin
-        set_global_var(28, 1);
+    if (global_var(GVAR_RUNNING_BURNING_GUY) == 0) then begin
+        set_global_var(GVAR_RUNNING_BURNING_GUY, 1);
     end
-    gsay_reply(66, 130);
-    gsay_option(66, 131, Node999, 50);
-    gsay_option(66, 132, Node999, 50);
+    Reply(130);
+    giq_option(1, NAME, 131, Node999, NEUTRAL_REACTION);
+    giq_option(1, NAME, 132, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node014
 begin
-    gsay_reply(66, 140);
-    gsay_option(66, 141, Node999, 50);
-    if (global_var(28) >= 2) then begin
-        gsay_option(66, 142, Node015, 50);
+    Reply(140);
+    giq_option(1, NAME, 141, Node999, NEUTRAL_REACTION);
+    if (global_var(GVAR_RUNNING_BURNING_GUY) >= 2) then begin
+        giq_option(1, NAME, 142, Node015, NEUTRAL_REACTION);
     end
-    if ((local_var(8) == 0) and (global_var(28) == 1)) then begin
-        gsay_option(66, 143, Node018, 50);
+    if ((local_var(8) == 0) and (global_var(GVAR_RUNNING_BURNING_GUY) == 1)) then begin
+        giq_option(1, NAME, 143, Node018, NEUTRAL_REACTION);
     end
 end
 
 procedure Node015
 begin
-    give_exp_points(floor(300 / 5) + (get_critter_stat(dude_obj, 4) * floor(300 / 12)) + (get_critter_stat(dude_obj, 6) * floor(300 / 30)) + (floor(300 / 75) * (((get_critter_stat(dude_obj, 6) - 4) * 4) - (difficulty_level * 4))));
-    display_msg(message_str(10, random(140, 143)) + (floor(300 / 5) + (get_critter_stat(dude_obj, 4) * floor(300 / 12)) + (get_critter_stat(dude_obj, 6) * floor(300 / 30)) + (floor(300 / 75) * (((get_critter_stat(dude_obj, 6) - 4) * 4) - (difficulty_level * 4)))) + message_str(10, 144));
+    give_exp_points(floor(300 / 5) + (dude_iq * floor(300 / 12)) + (dude_luck * floor(300 / 30)) + (floor(300 / 75) * (((dude_luck - 4) * 4) - (difficulty_level * 4))));
+    display_msg(message_str(SCRIPT_STTEXT, random(140, 143)) + (floor(300 / 5) + (dude_iq * floor(300 / 12)) + (dude_luck * floor(300 / 30)) + (floor(300 / 75) * (((dude_luck - 4) * 4) - (difficulty_level * 4)))) + message_str(SCRIPT_STTEXT, 144));
     set_local_var(9, 1);
-    gsay_reply(66, 150);
-    gsay_option(66, 151, Node999, 50);
-    gsay_option(66, 152, Node999, 50);
+    Reply(150);
+    giq_option(1, NAME, 151, Node999, NEUTRAL_REACTION);
+    giq_option(1, NAME, 152, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node016
 begin
-    if (global_var(28) != 2) then begin
-        gsay_reply(66, 1601);
-        gsay_option(66, 161, Node017, 50);
+    if (global_var(GVAR_RUNNING_BURNING_GUY) != 2) then begin
+        Reply(1601);
+        giq_option(1, NAME, 161, Node017, NEUTRAL_REACTION);
     end
     else begin
-        gsay_reply(66, 1602);
-        gsay_option(66, 162, Node017, 50);
+        Reply(1602);
+        giq_option(1, NAME, 162, Node017, NEUTRAL_REACTION);
     end
 end
 
 procedure Node017
 begin
-    gsay_reply(66, 170);
-    if (global_var(28) == 0) then begin
-        gsay_option(66, 172, Node012, 50);
+    Reply(170);
+    if (global_var(GVAR_RUNNING_BURNING_GUY) == 0) then begin
+        giq_option(1, NAME, 172, Node012, NEUTRAL_REACTION);
     end
     else begin
-        gsay_option(66, 171, Node999, 50);
+        giq_option(1, NAME, 171, Node999, NEUTRAL_REACTION);
     end
 end
 
 procedure Node018
 begin
-    if (global_var(0) > 0) then begin
-        dude_objective_reputation := (global_var(47) * 30) + global_var(0) - (20 * global_var(1));
+    if (global_var(GVAR_PLAYER_REPUTATION) > 0) then begin
+        dude_objective_reputation := (global_var(GVAR_TOWN_REP_ARROYO) * 30) + global_var(GVAR_PLAYER_REPUTATION) - (20 * global_var(GVAR_CHILDKILLER_REPUTATION));
     end
     else begin
-        dude_objective_reputation := (global_var(47) * 30) + global_var(0) - (20 * global_var(1));
+        dude_objective_reputation := (global_var(GVAR_TOWN_REP_ARROYO) * 30) + global_var(GVAR_PLAYER_REPUTATION) - (20 * global_var(GVAR_CHILDKILLER_REPUTATION));
     end
     if ((dude_objective_reputation + local_var(6)) >= 300) then begin
         self_subjective_reaction := 1;
@@ -398,24 +398,24 @@ begin
             self_subjective_reaction := -1;
         end
     end
-    persuade_chance := floor(((get_critter_stat(dude_obj, 3) * 10) + has_skill(dude_obj, 14) + local_var(5)) / 5) + (10 * self_subjective_reaction);
+    persuade_chance := floor(((dude_charisma * 10) + dude_skill(SKILL_SPEECH) + local_var(5)) / 5) + (10 * self_subjective_reaction);
     if (random(0, 120) <= persuade_chance) then begin
-        gsay_reply(66, 1802);
-        gsay_option(66, 183, Node999, 50);
-        gsay_option(66, 184, NodeGoFight, 50);
+        Reply(1802);
+        giq_option(1, NAME, 183, Node999, NEUTRAL_REACTION);
+        giq_option(1, NAME, 184, NodeGoFight, NEUTRAL_REACTION);
     end
     else begin
-        gsay_reply(66, 1801);
-        gsay_option(66, 181, Node999, 50);
-        gsay_option(66, 182, Node999, 50);
+        Reply(1801);
+        giq_option(1, NAME, 181, Node999, NEUTRAL_REACTION);
+        giq_option(1, NAME, 182, Node999, NEUTRAL_REACTION);
     end
 end
 
 procedure Node019
 begin
-    gsay_reply(66, 190);
-    gsay_option(66, 191, Node999, 50);
-    gsay_option(66, 192, Node999, 50);
+    Reply(190);
+    giq_option(1, NAME, 191, Node999, NEUTRAL_REACTION);
+    giq_option(1, NAME, 192, Node999, NEUTRAL_REACTION);
 end
 
 procedure NodeGoFight
@@ -423,12 +423,11 @@ begin
     if (local_var(8) == 0) then begin
         set_local_var(8, 1);
     end
-    if (global_var(28) == 0) then begin
-        set_global_var(28, 1);
+    if (global_var(GVAR_RUNNING_BURNING_GUY) == 0) then begin
+        set_global_var(GVAR_RUNNING_BURNING_GUY, 1);
     end
 end
 
 procedure Node999
 begin
 end
-
