diff --git a/Mods/Olympus2207/Original/TDUDEMOM.ssl b/Mods/Olympus2207/ReDefine/TDUDEMOM.ssl
index 5fb47b8..37fbb52 100644
--- a/Mods/Olympus2207/Original/TDUDEMOM.ssl
+++ b/Mods/Olympus2207/ReDefine/TDUDEMOM.ssl
@@ -64,8 +64,8 @@ begin
         set_local_var(4, random(40, 70));
         set_local_var(5, random(60, 90));
         set_local_var(6, random(40, 45));
-        critter_add_trait(self_obj, 1, 6, 0);
-        critter_add_trait(self_obj, 1, 5, 4);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 0);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_TARTAR_WOMEN);
         max_distance := random(6, 8);
     end
 end
@@ -76,12 +76,12 @@ begin
         animation_is_called := 1;
         add_timer_event(self_obj, random(29, 76), 1);
     end
-    if ((global_var(27) == 2) and (global_var(29) == 2) and (global_var(28) >= 2) and (local_var(8) == 0) and (get_sfall_global_int(7) != 3)) then begin
+    if ((global_var(GVAR_LOAD_MAP_INDEX) == 2) and (global_var(GVAR_VIC_DEVICE) == 2) and (global_var(GVAR_RUNNING_BURNING_GUY) >= 2) and (local_var(8) == 0) and (get_sfall_global_int(7) != 3)) then begin
         set_local_var(8, 1);
-        create_message_window(message_str(10, 900));
+        create_message_window(message_str(SCRIPT_STTEXT, 900));
     end
     if (obj_can_see_obj(self_obj, dude_obj)) then begin
-        if ((local_var(2) == 1) or (global_var(47) <= -31)) then begin
+        if ((local_var(2) == 1) or (global_var(GVAR_TOWN_REP_ARROYO) <= -31)) then begin
             set_local_var(2, 0);
             attack_setup(self_obj, dude_obj);
         end
@@ -96,10 +96,10 @@ procedure description_p_proc
 begin
     script_overrides;
     if (local_var(7) == 0) then begin
-        display_msg(message_str(64, 2));
+        display_mstr(2);
     end
     else begin
-        display_msg(message_str(64, 4));
+        display_mstr(4);
     end
 end
 
@@ -107,10 +107,10 @@ procedure look_at_p_proc
 begin
     script_overrides;
     if (local_var(7) == 0) then begin
-        display_msg(message_str(64, 1));
+        display_mstr(1);
     end
     else begin
-        display_msg(message_str(64, 3));
+        display_mstr(3);
     end
 end
 
@@ -125,9 +125,9 @@ end
 procedure talk_p_proc
 begin
     if (not(anim_busy(self_obj))) then begin
-        anim(self_obj, 1000, rotation_to_tile(tile_num(self_obj), tile_num(dude_obj)));
+        anim(self_obj, 1000, rotation_to_tile(tile_num(self_obj), dude_tile));
     end
-    if ((global_var(27) == 2) and (global_var(28) >= 2) and (global_var(29) >= 2)) then begin
+    if ((global_var(GVAR_LOAD_MAP_INDEX) == 2) and (global_var(GVAR_RUNNING_BURNING_GUY) >= 2) and (global_var(GVAR_VIC_DEVICE) >= 2)) then begin
         start_gdialog(64, self_obj, 4, 5, -1);
         gsay_start;
         call Node015();
@@ -146,12 +146,12 @@ end
 procedure destroy_p_proc
 begin
     if (source_obj == dude_obj) then begin
-        set_global_var(0, global_var(0) - 1);
+        set_global_var(GVAR_PLAYER_REPUTATION, global_var(GVAR_PLAYER_REPUTATION) - 1);
         if ((47 == 50) or (47 == 59)) then begin
-            set_global_var(47, global_var(47) - 3);
+            set_global_var(GVAR_TOWN_REP_ARROYO, global_var(GVAR_TOWN_REP_ARROYO) - 3);
         end
         else begin
-            set_global_var(47, global_var(47) - 6);
+            set_global_var(GVAR_TOWN_REP_ARROYO, global_var(GVAR_TOWN_REP_ARROYO) - 6);
         end
     end
 end
@@ -161,26 +161,26 @@ begin
     if (action_being_used == 10) then begin
         steal_perk := 0;
         steal_gloves := 0;
-        if (has_trait(0, dude_obj, 37) != 0) then begin
+        if (has_trait(TRAIT_PERK, dude_obj, PERK_pickpocket) != 0) then begin
             steal_perk := steal_perk + 15;
         end
-        if (has_trait(0, dude_obj, 33) != 0) then begin
+        if (has_trait(TRAIT_PERK, dude_obj, PERK_master_thief) != 0) then begin
             steal_perk := steal_perk + 30;
         end
-        if ((obj_pid(critter_inven_obj(dude_obj, 1)) == 245) or (obj_pid(critter_inven_obj(dude_obj, 2)) == 245)) then begin
+        if ((obj_pid(critter_inven_obj(dude_obj, INVEN_TYPE_RIGHT_HAND)) == 245) or (obj_pid(critter_inven_obj(dude_obj, INVEN_TYPE_LEFT_HAND)) == 245)) then begin
             steal_gloves := 10;
         end
-        if (random(0, 100) <= ((has_skill(dude_obj, 10) / 2) + (((3 * get_critter_stat(dude_obj, 5)) + (((get_critter_stat(dude_obj, 6) - 4) * 4) - (difficulty_level * 4)) + (2 * steal_perk) + steal_gloves) / (local_var(4) / 25)))) then begin
-            display_msg(message_str(10, random(145, 148)));
+        if (random(0, 100) <= ((dude_skill(SKILL_STEAL) / 2) + (((3 * get_critter_stat(dude_obj, STAT_ag)) + (((dude_luck - 4) * 4) - (difficulty_level * 4)) + (2 * steal_perk) + steal_gloves) / (local_var(4) / 25)))) then begin
+            display_msg(message_str(SCRIPT_STTEXT, random(145, 148)));
         end
         else begin
             script_overrides;
-            display_msg(message_str(10, random(200, 203)));
+            display_msg(message_str(SCRIPT_STTEXT, random(200, 203)));
         end
     end
     if (action_being_used == 12) then begin
         script_overrides;
-        display_msg(message_str(10, 204) + ((has_skill(dude_obj, 10) / 2) + (((3 * get_critter_stat(dude_obj, 5)) + (((get_critter_stat(dude_obj, 6) - 4) * 4) - (difficulty_level * 4)) + (2 * steal_perk) + steal_gloves) / (local_var(4) / 25))) + message_str(10, 205));
+        display_msg(message_str(SCRIPT_STTEXT, 204) + ((dude_skill(SKILL_STEAL) / 2) + (((3 * get_critter_stat(dude_obj, STAT_ag)) + (((dude_luck - 4) * 4) - (difficulty_level * 4)) + (2 * steal_perk) + steal_gloves) / (local_var(4) / 25))) + message_str(SCRIPT_STTEXT, 205));
     end
 end
 
@@ -208,7 +208,7 @@ begin
             time_to_work := random(0, 5);
         end
     end
-    if (metarule(14, 0)) then begin
+    if (map_first_run) then begin
         dialogue_system_enter;
         game_time_advance(10 * 60 * 60 * 2);
     end
@@ -224,33 +224,33 @@ begin
     end
     if (fixed_param == 4) then begin
         set_local_var(9, 1);
-        if ((get_critter_stat(dude_obj, 4) <= 5) and (get_critter_stat(dude_obj, 2) >= 8) and (get_critter_stat(dude_obj, 0) >= 8)) then begin
-            set_global_var(285, 1);
+        if ((dude_iq <= 5) and (dude_endurance >= 8) and (dude_strength >= 8)) then begin
+            set_global_var(GVAR_NEW_RENO_WRIGHT_FLAGS, 1);
         end
         else begin
-            if ((get_critter_stat(dude_obj, 4) >= 7) and (get_critter_stat(dude_obj, 3) >= 8) and (get_critter_stat(dude_obj, 0) <= 5)) then begin
-                set_global_var(285, 2);
+            if ((dude_iq >= 7) and (dude_charisma >= 8) and (dude_strength <= 5)) then begin
+                set_global_var(GVAR_NEW_RENO_WRIGHT_FLAGS, 2);
             end
             else begin
-                if ((get_critter_stat(dude_obj, 4) <= 3) and (get_critter_stat(dude_obj, 6) >= 9)) then begin
-                    set_global_var(285, 3);
-                    critter_add_trait(dude_obj, 0, 3, 1);
+                if ((dude_is_stupid) and (dude_luck >= 9)) then begin
+                    set_global_var(GVAR_NEW_RENO_WRIGHT_FLAGS, 3);
+                    critter_add_trait(dude_obj, TRAIT_PERK, PERK_bonus_move, 1);
                 end
             end
         end
-        if (global_var(285) == 1) then begin
-            display_msg(message_str(10, 1));
+        if (global_var(GVAR_NEW_RENO_WRIGHT_FLAGS) == 1) then begin
+            display_msg(message_str(SCRIPT_STTEXT, 1));
         end
         else begin
-            if (global_var(285) == 2) then begin
-                display_msg(message_str(10, 2));
+            if (global_var(GVAR_NEW_RENO_WRIGHT_FLAGS) == 2) then begin
+                display_msg(message_str(SCRIPT_STTEXT, 2));
             end
             else begin
-                if (global_var(285) == 3) then begin
-                    display_msg(message_str(10, 3));
+                if (global_var(GVAR_NEW_RENO_WRIGHT_FLAGS) == 3) then begin
+                    display_msg(message_str(SCRIPT_STTEXT, 3));
                 end
                 else begin
-                    display_msg(message_str(10, 4));
+                    display_msg(message_str(SCRIPT_STTEXT, 4));
                 end
             end
         end
@@ -418,199 +418,199 @@ procedure Node010
 begin
     if (local_var(7) == 0) then begin
         set_local_var(7, 1);
-        add_obj_to_inven(dude_obj, create_object_sid(393, 1, 2, -1));
-        gsay_reply(64, 100);
-        giq_option(1, 64, 101, Node011a, 49);
-        giq_option(1, 64, 102, Node011a, 49);
-        if (get_critter_stat(dude_obj, 4) >= 7) then begin
-            giq_option(1, 64, 103, Node011b, 49);
+        add_obj_to_inven(dude_obj, create_object(393, 1, 2));
+        Reply(100);
+        giq_option(1, NAME, 101, Node011a, GOOD_REACTION);
+        giq_option(1, NAME, 102, Node011a, GOOD_REACTION);
+        if (dude_iq >= 7) then begin
+            giq_option(1, NAME, 103, Node011b, GOOD_REACTION);
         end
         else begin
-            giq_option(1, 64, 104, Node011a, 49);
+            giq_option(1, NAME, 104, Node011a, GOOD_REACTION);
         end
     end
     else begin
-        gsay_reply(64, 130);
-        gsay_option(64, 131, Node999, 50);
-        if ((global_var(27) == 1) and (obj_carrying_pid_obj(dude_obj, 290) == 0)) then begin
-            giq_option(1, 64, 132, Node014, 49);
+        Reply(130);
+        giq_option(1, NAME, 131, Node999, NEUTRAL_REACTION);
+        if ((global_var(GVAR_LOAD_MAP_INDEX) == 1) and (obj_carrying_pid_obj(dude_obj, 290) == 0)) then begin
+            giq_option(1, NAME, 132, Node014, GOOD_REACTION);
         end
-        giq_option(1, 64, 133, Node999, 49);
-        if ((global_var(27) == 0) or (global_var(28) == 0) or (global_var(29) == 0)) then begin
-            giq_option(1, 64, 134, Node023, 49);
+        giq_option(1, NAME, 133, Node999, GOOD_REACTION);
+        if ((global_var(GVAR_LOAD_MAP_INDEX) == 0) or (global_var(GVAR_RUNNING_BURNING_GUY) == 0) or (global_var(GVAR_VIC_DEVICE) == 0)) then begin
+            giq_option(1, NAME, 134, Node023, GOOD_REACTION);
         end
     end
 end
 
 procedure Node011a
 begin
-    gsay_reply(64, 1101);
-    gsay_option(64, 111, Node012a, 50);
-    gsay_option(64, 112, Node012b, 50);
+    Reply(1101);
+    giq_option(1, NAME, 111, Node012a, NEUTRAL_REACTION);
+    giq_option(1, NAME, 112, Node012b, NEUTRAL_REACTION);
 end
 
 procedure Node011b
 begin
-    gsay_reply(64, 1102);
-    gsay_option(64, 111, Node012a, 50);
-    gsay_option(64, 112, Node012b, 50);
+    Reply(1102);
+    giq_option(1, NAME, 111, Node012a, NEUTRAL_REACTION);
+    giq_option(1, NAME, 112, Node012b, NEUTRAL_REACTION);
 end
 
 procedure Node012a
 begin
-    add_obj_to_inven(dude_obj, create_object_sid(434, 1, 2, -1));
-    gsay_reply(64, 1201);
-    gsay_option(64, 121, Node999, 50);
+    add_obj_to_inven(dude_obj, create_object(434, 1, 2));
+    Reply(1201);
+    giq_option(1, NAME, 121, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node012b
 begin
-    add_obj_to_inven(dude_obj, create_object_sid(434, 1, 2, -1));
-    gsay_reply(64, 1202);
-    gsay_option(64, 121, Node999, 50);
+    add_obj_to_inven(dude_obj, create_object(434, 1, 2));
+    Reply(1202);
+    giq_option(1, NAME, 121, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node014
 begin
-    gsay_reply(64, 140);
-    gsay_option(64, 141, Node999, 50);
-    gsay_option(64, 142, Node999, 50);
+    Reply(140);
+    giq_option(1, NAME, 141, Node999, NEUTRAL_REACTION);
+    giq_option(1, NAME, 142, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node015
 begin
-    gsay_reply(64, 150);
-    gsay_option(64, 151, Node016a, 50);
-    gsay_option(64, 152, Node016b, 50);
+    Reply(150);
+    giq_option(1, NAME, 151, Node016a, NEUTRAL_REACTION);
+    giq_option(1, NAME, 152, Node016b, NEUTRAL_REACTION);
 end
 
 procedure Node016a
 begin
     if ((about_gods == 1) and (about_purgatoir == 1) and (about_childrens == 1)) then begin
-        gsay_reply(64, 1604);
-        gsay_option(64, 1605, NodeGoToBad, 50);
+        Reply(1604);
+        giq_option(1, NAME, 1605, NodeGoToBad, NEUTRAL_REACTION);
     end
     else begin
         if (talk_start == 0) then begin
             talk_start := 1;
-            gsay_reply(64, 1601);
+            Reply(1601);
         end
         else begin
-            gsay_reply(64, 1603);
+            Reply(1603);
         end
         if (about_gods == 0) then begin
-            gsay_option(64, 161, Node017, 50);
+            giq_option(1, NAME, 161, Node017, NEUTRAL_REACTION);
         end
         if (about_purgatoir == 0) then begin
-            gsay_option(64, 162, Node019, 50);
+            giq_option(1, NAME, 162, Node019, NEUTRAL_REACTION);
         end
         if (about_childrens == 0) then begin
-            gsay_option(64, 163, Node021, 50);
+            giq_option(1, NAME, 163, Node021, NEUTRAL_REACTION);
         end
-        gsay_option(64, 164, NodeGoToBad, 50);
+        giq_option(1, NAME, 164, NodeGoToBad, NEUTRAL_REACTION);
     end
 end
 
 procedure Node016b
 begin
     if ((about_gods == 1) and (about_purgatoir == 1) and (about_childrens == 1)) then begin
-        gsay_reply(64, 1604);
-        gsay_option(64, 1605, NodeGoToBad, 50);
+        Reply(1604);
+        giq_option(1, NAME, 1605, NodeGoToBad, NEUTRAL_REACTION);
     end
     else begin
         if (talk_start == 0) then begin
             talk_start := 1;
-            gsay_reply(64, 1602);
+            Reply(1602);
         end
         else begin
-            gsay_reply(64, 1603);
+            Reply(1603);
         end
         if (about_gods == 0) then begin
-            gsay_option(64, 161, Node017, 50);
+            giq_option(1, NAME, 161, Node017, NEUTRAL_REACTION);
         end
         if (about_purgatoir == 0) then begin
-            gsay_option(64, 162, Node019, 50);
+            giq_option(1, NAME, 162, Node019, NEUTRAL_REACTION);
         end
         if (about_childrens == 0) then begin
-            gsay_option(64, 163, Node021, 50);
+            giq_option(1, NAME, 163, Node021, NEUTRAL_REACTION);
         end
-        gsay_option(64, 164, NodeGoToBad, 50);
+        giq_option(1, NAME, 164, NodeGoToBad, NEUTRAL_REACTION);
     end
 end
 
 procedure Node017
 begin
     about_gods := 1;
-    gsay_reply(64, 170);
-    gsay_option(64, 171, Node018a, 50);
-    gsay_option(64, 172, Node018b, 50);
+    Reply(170);
+    giq_option(1, NAME, 171, Node018a, NEUTRAL_REACTION);
+    giq_option(1, NAME, 172, Node018b, NEUTRAL_REACTION);
 end
 
 procedure Node018a
 begin
-    gsay_reply(64, 1801);
-    gsay_option(64, 181, Node016a, 50);
-    gsay_option(64, 182, NodeGoToBad, 50);
+    Reply(1801);
+    giq_option(1, NAME, 181, Node016a, NEUTRAL_REACTION);
+    giq_option(1, NAME, 182, NodeGoToBad, NEUTRAL_REACTION);
 end
 
 procedure Node018b
 begin
-    gsay_reply(64, 1802);
-    gsay_option(64, 181, Node016a, 50);
-    gsay_option(64, 182, NodeGoToBad, 50);
+    Reply(1802);
+    giq_option(1, NAME, 181, Node016a, NEUTRAL_REACTION);
+    giq_option(1, NAME, 182, NodeGoToBad, NEUTRAL_REACTION);
 end
 
 procedure Node019
 begin
     about_purgatoir := 1;
-    gsay_reply(64, 190);
-    gsay_option(64, 191, Node020, 50);
+    Reply(190);
+    giq_option(1, NAME, 191, Node020, NEUTRAL_REACTION);
 end
 
 procedure Node020
 begin
-    gsay_reply(64, 200);
-    gsay_option(64, 201, Node016a, 50);
-    gsay_option(64, 202, NodeGoToBad, 50);
+    Reply(200);
+    giq_option(1, NAME, 201, Node016a, NEUTRAL_REACTION);
+    giq_option(1, NAME, 202, NodeGoToBad, NEUTRAL_REACTION);
 end
 
 procedure Node021
 begin
     about_childrens := 1;
-    gsay_reply(64, 210);
-    gsay_option(64, 211, Node022a, 50);
-    gsay_option(64, 212, Node022b, 50);
+    Reply(210);
+    giq_option(1, NAME, 211, Node022a, NEUTRAL_REACTION);
+    giq_option(1, NAME, 212, Node022b, NEUTRAL_REACTION);
 end
 
 procedure Node022a
 begin
-    gsay_reply(64, 2201);
-    gsay_option(64, 221, Node016a, 50);
-    gsay_option(64, 222, NodeGoToBad, 50);
+    Reply(2201);
+    giq_option(1, NAME, 221, Node016a, NEUTRAL_REACTION);
+    giq_option(1, NAME, 222, NodeGoToBad, NEUTRAL_REACTION);
 end
 
 procedure Node022b
 begin
-    gsay_reply(64, 2202);
-    gsay_option(64, 221, Node016a, 50);
-    gsay_option(64, 222, NodeGoToBad, 50);
+    Reply(2202);
+    giq_option(1, NAME, 221, Node016a, NEUTRAL_REACTION);
+    giq_option(1, NAME, 222, NodeGoToBad, NEUTRAL_REACTION);
 end
 
 procedure Node023
 begin
-    if (global_var(27) == 0) then begin
-        gsay_reply(64, 2301);
-        gsay_option(64, 231, Node999, 50);
+    if (global_var(GVAR_LOAD_MAP_INDEX) == 0) then begin
+        Reply(2301);
+        giq_option(1, NAME, 231, Node999, NEUTRAL_REACTION);
     end
     else begin
-        if (global_var(28) == 0) then begin
-            gsay_reply(64, 2302);
-            gsay_option(64, 232, Node999, 50);
+        if (global_var(GVAR_RUNNING_BURNING_GUY) == 0) then begin
+            Reply(2302);
+            giq_option(1, NAME, 232, Node999, NEUTRAL_REACTION);
         end
         else begin
-            if (global_var(29) == 0) then begin
-                gsay_reply(64, 2303);
-                gsay_option(64, 233, Node999, 50);
+            if (global_var(GVAR_VIC_DEVICE) == 0) then begin
+                Reply(2303);
+                giq_option(1, NAME, 233, Node999, NEUTRAL_REACTION);
             end
         end
     end
@@ -627,4 +627,3 @@ procedure NodeGoToBad
 begin
     add_timer_event(self_obj, 5, 2);
 end
-
