diff --git a/Mods/Olympus2207/Original/TDUDEBAD.ssl b/Mods/Olympus2207/ReDefine/TDUDEBAD.ssl
index e5dfd35..3a807cc 100644
--- a/Mods/Olympus2207/Original/TDUDEBAD.ssl
+++ b/Mods/Olympus2207/ReDefine/TDUDEBAD.ssl
@@ -43,7 +43,7 @@ end
 
 procedure use_p_proc
 begin
-    if ((global_var(30) == 2) and (global_var(31) == 2) and (global_var(32) == 2) and (global_var(35) == 2)) then begin
+    if ((global_var(GVAR_SLAVE_RUN) == 2) and (global_var(GVAR_SLAVES_COUNT) == 2) and (global_var(GVAR_MAGGIE_STATUS) == 2) and (global_var(GVAR_PIP_BOY_ANNA_DIARY) == 2)) then begin
         fadeOut(1500);
         load_map("tartar13", 1);
     end
@@ -68,4 +68,3 @@ end
 procedure talk_p_proc
 begin
 end
-
