diff --git a/Mods/Olympus2207/Original/TSPTLPRK.ssl b/Mods/Olympus2207/ReDefine/TSPTLPRK.ssl
index 09fa437..7b37ec1 100644
--- a/Mods/Olympus2207/Original/TSPTLPRK.ssl
+++ b/Mods/Olympus2207/ReDefine/TSPTLPRK.ssl
@@ -29,7 +29,6 @@ procedure spatial_p_proc
 begin
     if ((source_obj == dude_obj) and (map_var(0) == 0)) then begin
         set_map_var(0, 1);
-        display_msg(message_str(279, 12));
+        display_msg(message_str(SCRIPT_TBRDWALL, 12));
     end
 end
-
