diff --git a/Mods/Olympus2207/Original/TTLIFTGD.ssl b/Mods/Olympus2207/ReDefine/TTLIFTGD.ssl
index de0d9af..72ccb24 100644
--- a/Mods/Olympus2207/Original/TTLIFTGD.ssl
+++ b/Mods/Olympus2207/ReDefine/TTLIFTGD.ssl
@@ -38,61 +38,61 @@ procedure start
 begin
     if (local_var(1) == 0) then begin
         if (random(0, 100) <= 30) then begin
-            add_obj_to_inven(self_obj, create_object_sid(176, 1, 2, -1));
-            add_mult_objs_to_inven(self_obj, create_object_sid(11, 1, 2, -1), random(2, 3));
-            wield_obj_critter(self_obj, obj_carrying_pid_obj(self_obj, 176));
+            add_obj_to_inven(self_obj, create_object(176, 1, 2));
+            add_mult_objs_to_inven(self_obj, create_object(11, 1, 2), random(2, 3));
+            wield_obj(obj_carrying_pid_obj(self_obj, 176));
         end
         else begin
-            add_obj_to_inven(self_obj, create_object_sid(153, 1, 2, -1));
-            add_mult_objs_to_inven(self_obj, create_object_sid(19, 1, 2, -1), random(2, 3));
-            wield_obj_critter(self_obj, obj_carrying_pid_obj(self_obj, 153));
+            add_obj_to_inven(self_obj, create_object(153, 1, 2));
+            add_mult_objs_to_inven(self_obj, create_object(19, 1, 2), random(2, 3));
+            wield_obj(obj_carrying_pid_obj(self_obj, 153));
         end
         if (random(0, 100) <= 30) then begin
-            add_mult_objs_to_inven(self_obj, create_object_sid(221, 1, 2, -1), random(1, 2));
+            add_mult_objs_to_inven(self_obj, create_object(221, 1, 2), random(1, 2));
         end
         if (random(0, 100) <= 30) then begin
-            add_obj_to_inven(self_obj, create_object_sid(64, 1, 2, -1));
+            add_obj_to_inven(self_obj, create_object(64, 1, 2));
         end
         if (random(0, 100) <= 30) then begin
-            add_obj_to_inven(self_obj, create_object_sid(103, 1, 2, -1));
+            add_obj_to_inven(self_obj, create_object(103, 1, 2));
         end
         if (random(0, 100) <= 30) then begin
-            add_obj_to_inven(self_obj, create_object_sid(222, 1, 2, -1));
+            add_obj_to_inven(self_obj, create_object(222, 1, 2));
         end
         if (random(0, 100) <= 70) then begin
-            add_mult_objs_to_inven(self_obj, create_object_sid(41, 1, 2, -1), random(4, 30));
+            add_mult_objs_to_inven(self_obj, create_object(41, 1, 2), random(4, 30));
         end
-        set_critter_stat(self_obj, 0, random(1, 2));
-        set_critter_stat(self_obj, 1, random(-1, 0));
-        set_critter_stat(self_obj, 2, random(1, 2));
-        set_critter_stat(self_obj, 3, random(-3, -2));
-        set_critter_stat(self_obj, 4, random(-3, -2));
-        set_critter_stat(self_obj, 5, random(-1, 1));
-        set_critter_stat(self_obj, 6, random(-3, 3));
-        critter_mod_skill(self_obj, 0, random(10, 20));
-        critter_mod_skill(self_obj, 1, random(40, 60));
-        critter_mod_skill(self_obj, 2, random(50, 70));
-        critter_mod_skill(self_obj, 3, random(20, 25));
-        critter_mod_skill(self_obj, 4, random(20, 25));
-        critter_mod_skill(self_obj, 5, random(20, 30));
-        critter_mod_skill(self_obj, 6, random(0, 10));
-        critter_mod_skill(self_obj, 7, random(0, 10));
-        critter_mod_skill(self_obj, 8, random(0, 10));
-        critter_mod_skill(self_obj, 9, random(0, 10));
-        critter_mod_skill(self_obj, 10, random(0, 10));
-        critter_mod_skill(self_obj, 11, random(0, 10));
-        critter_mod_skill(self_obj, 12, random(0, 10));
-        critter_mod_skill(self_obj, 13, random(0, 10));
-        critter_mod_skill(self_obj, 14, random(0, 10));
-        critter_mod_skill(self_obj, 15, random(0, 10));
-        critter_mod_skill(self_obj, 16, random(0, 10));
-        critter_mod_skill(self_obj, 17, random(0, 10));
+        set_critter_stat(self_obj, STAT_st, random(1, 2));
+        set_critter_stat(self_obj, STAT_pe, random(-1, 0));
+        set_critter_stat(self_obj, STAT_en, random(1, 2));
+        set_critter_stat(self_obj, STAT_ch, random(-3, -2));
+        set_critter_stat(self_obj, STAT_iq, random(-3, -2));
+        set_critter_stat(self_obj, STAT_ag, random(-1, 1));
+        set_critter_stat(self_obj, STAT_lu, random(-3, 3));
+        critter_mod_skill(self_obj, SKILL_SMALL_GUNS, random(10, 20));
+        critter_mod_skill(self_obj, SKILL_BIG_GUNS, random(40, 60));
+        critter_mod_skill(self_obj, SKILL_ENERGY_WEAPONS, random(50, 70));
+        critter_mod_skill(self_obj, SKILL_UNARMED_COMBAT, random(20, 25));
+        critter_mod_skill(self_obj, SKILL_MELEE, random(20, 25));
+        critter_mod_skill(self_obj, SKILL_THROWING, random(20, 30));
+        critter_mod_skill(self_obj, SKILL_FIRST_AID, random(0, 10));
+        critter_mod_skill(self_obj, SKILL_DOCTOR, random(0, 10));
+        critter_mod_skill(self_obj, SKILL_SNEAK, random(0, 10));
+        critter_mod_skill(self_obj, SKILL_LOCKPICK, random(0, 10));
+        critter_mod_skill(self_obj, SKILL_STEAL, random(0, 10));
+        critter_mod_skill(self_obj, SKILL_TRAPS, random(0, 10));
+        critter_mod_skill(self_obj, SKILL_SCIENCE, random(0, 10));
+        critter_mod_skill(self_obj, SKILL_REPAIR, random(0, 10));
+        critter_mod_skill(self_obj, SKILL_SPEECH, random(0, 10));
+        critter_mod_skill(self_obj, SKILL_BARTER, random(0, 10));
+        critter_mod_skill(self_obj, SKILL_GAMBLING, random(0, 10));
+        critter_mod_skill(self_obj, SKILL_OUTDOORSMAN, random(0, 10));
         set_local_var(3, tile_num(self_obj));
         set_local_var(1, 1);
         set_local_var(4, random(60, 85));
         set_local_var(5, random(40, 100));
-        critter_add_trait(self_obj, 1, 6, 3);
-        critter_add_trait(self_obj, 1, 5, 5);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 3);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_OLYMPUS_GUARD);
     end
 end
 
@@ -102,15 +102,15 @@ begin
         event_is_called := 1;
         add_timer_event(self_obj, random(150, 450), 1);
     end
-    if ((global_var(6) == 3) and (obj_can_see_obj(self_obj, dude_obj) == 1)) then begin
-        if (not((obj_pid(critter_inven_obj(dude_obj, 0)) == 120) or (obj_pid(critter_inven_obj(dude_obj, 0)) == 121) or (obj_pid(critter_inven_obj(dude_obj, 0)) == 122) or (obj_pid(critter_inven_obj(dude_obj, 0)) == 123) or (obj_pid(critter_inven_obj(dude_obj, 0)) == 129) or (obj_pid(critter_inven_obj(dude_obj, 0)) == 391)) and not(obj_pid(critter_inven_obj(dude_obj, 0)) == 394)) then begin
+    if ((global_var(GVAR_PLAYER_MARRIED) == 3) and (obj_can_see_obj(self_obj, dude_obj) == 1)) then begin
+        if (not((obj_pid(critter_inven_obj(dude_obj, INVEN_TYPE_WORN)) == 120) or (obj_pid(critter_inven_obj(dude_obj, INVEN_TYPE_WORN)) == 121) or (obj_pid(critter_inven_obj(dude_obj, INVEN_TYPE_WORN)) == 122) or (obj_pid(critter_inven_obj(dude_obj, INVEN_TYPE_WORN)) == 123) or (obj_pid(critter_inven_obj(dude_obj, INVEN_TYPE_WORN)) == 129) or (obj_pid(critter_inven_obj(dude_obj, INVEN_TYPE_WORN)) == 391)) and not(obj_pid(critter_inven_obj(dude_obj, INVEN_TYPE_WORN)) == 394)) then begin
             set_local_var(2, 1);
-            float_msg(self_obj, message_str(394, random(97, 99)), 2);
+            float_msg(self_obj, message_str(SCRIPT_XDFNDPTR, random(97, 99)), FLOAT_MSG_RED);
         end
     end
-    if ((tile_distance(tile_num(self_obj), tile_num(dude_obj)) <= 4) and not(combat_is_initialized) and (local_var(8) == 0) and (local_var(9) == 0)) then begin
-        if (global_var(6) == 3) then begin
-            if ((obj_pid(critter_inven_obj(dude_obj, 0)) == 120) or (obj_pid(critter_inven_obj(dude_obj, 0)) == 121) or (obj_pid(critter_inven_obj(dude_obj, 0)) == 122) or (obj_pid(critter_inven_obj(dude_obj, 0)) == 123) or (obj_pid(critter_inven_obj(dude_obj, 0)) == 129) or (obj_pid(critter_inven_obj(dude_obj, 0)) == 391) or (obj_pid(critter_inven_obj(dude_obj, 0)) == 394)) then begin
+    if ((tile_distance(tile_num(self_obj), dude_tile) <= 4) and not(combat_is_initialized) and (local_var(8) == 0) and (local_var(9) == 0)) then begin
+        if (global_var(GVAR_PLAYER_MARRIED) == 3) then begin
+            if ((obj_pid(critter_inven_obj(dude_obj, INVEN_TYPE_WORN)) == 120) or (obj_pid(critter_inven_obj(dude_obj, INVEN_TYPE_WORN)) == 121) or (obj_pid(critter_inven_obj(dude_obj, INVEN_TYPE_WORN)) == 122) or (obj_pid(critter_inven_obj(dude_obj, INVEN_TYPE_WORN)) == 123) or (obj_pid(critter_inven_obj(dude_obj, INVEN_TYPE_WORN)) == 129) or (obj_pid(critter_inven_obj(dude_obj, INVEN_TYPE_WORN)) == 391) or (obj_pid(critter_inven_obj(dude_obj, INVEN_TYPE_WORN)) == 394)) then begin
                 dialogue_system_enter;
             end
             else begin
@@ -122,7 +122,7 @@ begin
         end
     end
     if (obj_can_see_obj(self_obj, dude_obj)) then begin
-        if ((local_var(2) == 1) or (global_var(59) <= -31)) then begin
+        if ((local_var(2) == 1) or (global_var(GVAR_TOWN_REP_VAULT_13) <= -31)) then begin
             set_local_var(2, 0);
             attack_setup(self_obj, dude_obj);
         end
@@ -136,13 +136,13 @@ end
 procedure description_p_proc
 begin
     script_overrides;
-    display_msg(message_str(27, random(6, 8)));
+    display_msg(message_str(SCRIPT_TGRDSTND, random(6, 8)));
 end
 
 procedure look_at_p_proc
 begin
     script_overrides;
-    display_msg(message_str(27, random(1, 5)));
+    display_msg(message_str(SCRIPT_TGRDSTND, random(1, 5)));
 end
 
 procedure pickup_p_proc
@@ -154,23 +154,23 @@ end
 
 procedure push_p_proc
 begin
-    if (global_var(6) == 0) then begin
+    if (global_var(GVAR_PLAYER_MARRIED) == 0) then begin
         script_overrides;
-        float_msg(self_obj, message_str(27, random(1030, 1039)), 0);
+        float_msg(self_obj, message_str(SCRIPT_TGRDSTND, random(1030, 1039)), FLOAT_MSG_NORMAL);
     end
     else begin
         script_overrides;
-        float_msg(self_obj, message_str(27, random(1060, 1064)), 0);
+        float_msg(self_obj, message_str(SCRIPT_TGRDSTND, random(1060, 1064)), FLOAT_MSG_NORMAL);
     end
 end
 
 procedure talk_p_proc
 begin
     if (not(anim_busy(self_obj))) then begin
-        anim(self_obj, 1000, rotation_to_tile(tile_num(self_obj), tile_num(dude_obj)));
+        anim(self_obj, 1000, rotation_to_tile(tile_num(self_obj), dude_tile));
     end
-    if (not(global_var(6) == 3)) then begin
-        float_msg(self_obj, message_str(27, random(1060, 1064)), 0);
+    if (not(global_var(GVAR_PLAYER_MARRIED) == 3)) then begin
+        float_msg(self_obj, message_str(SCRIPT_TGRDSTND, random(1060, 1064)), FLOAT_MSG_NORMAL);
         game_ui_disable;
         fadeOut(700);
         animate_move_obj_to_tile(dude_obj, 15289, 0 bwor 16);
@@ -183,7 +183,7 @@ begin
         if (local_var(8) == 0) then begin
             start_gdialog(27, self_obj, -1, -1, -1);
             gsay_start;
-            if ((obj_pid(critter_inven_obj(dude_obj, 0)) == 120) or (obj_pid(critter_inven_obj(dude_obj, 0)) == 121) or (obj_pid(critter_inven_obj(dude_obj, 0)) == 122) or (obj_pid(critter_inven_obj(dude_obj, 0)) == 123) or (obj_pid(critter_inven_obj(dude_obj, 0)) == 129) or (obj_pid(critter_inven_obj(dude_obj, 0)) == 391)) then begin
+            if ((obj_pid(critter_inven_obj(dude_obj, INVEN_TYPE_WORN)) == 120) or (obj_pid(critter_inven_obj(dude_obj, INVEN_TYPE_WORN)) == 121) or (obj_pid(critter_inven_obj(dude_obj, INVEN_TYPE_WORN)) == 122) or (obj_pid(critter_inven_obj(dude_obj, INVEN_TYPE_WORN)) == 123) or (obj_pid(critter_inven_obj(dude_obj, INVEN_TYPE_WORN)) == 129) or (obj_pid(critter_inven_obj(dude_obj, INVEN_TYPE_WORN)) == 391)) then begin
                 call Node001();
             end
             else begin
@@ -204,42 +204,42 @@ begin
     if (action_being_used == 10) then begin
         steal_perk := 0;
         steal_gloves := 0;
-        if (has_trait(0, dude_obj, 37) != 0) then begin
+        if (has_trait(TRAIT_PERK, dude_obj, PERK_pickpocket) != 0) then begin
             steal_perk := steal_perk + 15;
         end
-        if (has_trait(0, dude_obj, 33) != 0) then begin
+        if (has_trait(TRAIT_PERK, dude_obj, PERK_master_thief) != 0) then begin
             steal_perk := steal_perk + 30;
         end
-        if ((obj_pid(critter_inven_obj(dude_obj, 1)) == 245) or (obj_pid(critter_inven_obj(dude_obj, 2)) == 245)) then begin
+        if ((obj_pid(critter_inven_obj(dude_obj, INVEN_TYPE_RIGHT_HAND)) == 245) or (obj_pid(critter_inven_obj(dude_obj, INVEN_TYPE_LEFT_HAND)) == 245)) then begin
             steal_gloves := 10;
         end
-        if (random(0, 100) <= ((has_skill(dude_obj, 10) / 2) + (((3 * get_critter_stat(dude_obj, 5)) + (((get_critter_stat(dude_obj, 6) - 4) * 4) - (difficulty_level * 4)) + (2 * steal_perk) + steal_gloves) / (local_var(4) / 25)))) then begin
-            display_msg(message_str(10, random(145, 148)));
+        if (random(0, 100) <= ((dude_skill(SKILL_STEAL) / 2) + (((3 * get_critter_stat(dude_obj, STAT_ag)) + (((dude_luck - 4) * 4) - (difficulty_level * 4)) + (2 * steal_perk) + steal_gloves) / (local_var(4) / 25)))) then begin
+            display_msg(message_str(SCRIPT_STTEXT, random(145, 148)));
         end
         else begin
             script_overrides;
-            if (random(0, 100) <= (25 - floor((((get_critter_stat(dude_obj, 6) - 4) * 4) - (difficulty_level * 4)) / 3))) then begin
-                if ((global_var(6) == 0) and (local_var(7) == 0)) then begin
+            if (random(0, 100) <= (25 - floor((((dude_luck - 4) * 4) - (difficulty_level * 4)) / 3))) then begin
+                if ((global_var(GVAR_PLAYER_MARRIED) == 0) and (local_var(7) == 0)) then begin
                     set_local_var(7, 1);
-                    float_msg(self_obj, message_str(27, random(10, 12)), 2);
+                    float_msg(self_obj, message_str(SCRIPT_TGRDSTND, random(10, 12)), FLOAT_MSG_RED);
                 end
                 else begin
-                    float_msg(self_obj, message_str(10, random(206, 209)), 2);
+                    float_msg(self_obj, message_str(SCRIPT_STTEXT, random(206, 209)), FLOAT_MSG_RED);
                     set_local_var(2, 1);
                 end
             end
             else begin
-                display_msg(message_str(10, random(200, 203)));
+                display_msg(message_str(SCRIPT_STTEXT, random(200, 203)));
             end
         end
     end
     if (action_being_used == 12) then begin
         script_overrides;
         if (random(1, 2) == 1) then begin
-            display_msg(message_str(10, 204) + ((has_skill(dude_obj, 10) / 2) + (((3 * get_critter_stat(dude_obj, 5)) + (((get_critter_stat(dude_obj, 6) - 4) * 4) - (difficulty_level * 4)) + (2 * steal_perk) + steal_gloves) / (local_var(4) / 25))) + (30 - get_critter_stat(dude_obj, 4) - (2 * get_critter_stat(dude_obj, 1))) + message_str(10, 205));
+            display_msg(message_str(SCRIPT_STTEXT, 204) + ((dude_skill(SKILL_STEAL) / 2) + (((3 * get_critter_stat(dude_obj, STAT_ag)) + (((dude_luck - 4) * 4) - (difficulty_level * 4)) + (2 * steal_perk) + steal_gloves) / (local_var(4) / 25))) + (30 - dude_iq - (2 * dude_perception)) + message_str(SCRIPT_STTEXT, 205));
         end
         else begin
-            display_msg(message_str(10, 204) + ((has_skill(dude_obj, 10) / 2) + (((3 * get_critter_stat(dude_obj, 5)) + (((get_critter_stat(dude_obj, 6) - 4) * 4) - (difficulty_level * 4)) + (2 * steal_perk) + steal_gloves) / (local_var(4) / 25))) - (30 - get_critter_stat(dude_obj, 4) - (2 * get_critter_stat(dude_obj, 1))) + message_str(10, 205));
+            display_msg(message_str(SCRIPT_STTEXT, 204) + ((dude_skill(SKILL_STEAL) / 2) + (((3 * get_critter_stat(dude_obj, STAT_ag)) + (((dude_luck - 4) * 4) - (difficulty_level * 4)) + (2 * steal_perk) + steal_gloves) / (local_var(4) / 25))) - (30 - dude_iq - (2 * dude_perception)) + message_str(SCRIPT_STTEXT, 205));
         end
     end
 end
@@ -262,7 +262,7 @@ procedure timed_event_p_proc
 begin
     if ((fixed_param == 1) and not(combat_is_initialized)) then begin
         if (not(combat_is_initialized)) then begin
-            float_msg(self_obj, message_str(27, random(1050, 1059)), 0);
+            float_msg(self_obj, message_str(SCRIPT_TGRDSTND, random(1050, 1059)), FLOAT_MSG_NORMAL);
         end
         if ((local_var(3) == 15764) and (tile_num(self_obj) != local_var(3))) then begin
             animate_move_obj_to_tile(self_obj, local_var(3), 0 bwor 16);
@@ -282,59 +282,59 @@ end
 
 procedure Node001
 begin
-    gsay_reply(27, 100);
-    gsay_option(27, 101, GoBack, 50);
-    gsay_option(27, 102, Node998, 50);
+    gsay_reply(SCRIPT_TGRDSTND, 100);
+    giq_option(1, SCRIPT_TGRDSTND, 101, GoBack, NEUTRAL_REACTION);
+    giq_option(1, SCRIPT_TGRDSTND, 102, Node998, NEUTRAL_REACTION);
 end
 
 procedure Node002
 begin
-    gsay_reply(27, 103);
-    if (obj_carrying_pid_obj(dude_obj, 304) and (global_var(348) == 1)) then begin
-        gsay_option(27, 104, Node003, 50);
+    gsay_reply(SCRIPT_TGRDSTND, 103);
+    if (obj_carrying_pid_obj(dude_obj, 304) and (global_var(GVAR_NEW_RENO_STOLEN_CAR) == 1)) then begin
+        giq_option(1, SCRIPT_TGRDSTND, 104, Node003, NEUTRAL_REACTION);
     end
     else begin
-        gsay_option(27, 104, Node006, 50);
+        giq_option(1, SCRIPT_TGRDSTND, 104, Node006, NEUTRAL_REACTION);
     end
-    gsay_option(27, 108, GoBack, 50);
+    giq_option(1, SCRIPT_TGRDSTND, 108, GoBack, NEUTRAL_REACTION);
 end
 
 procedure Node003
 begin
-    gsay_reply(27, 105);
-    gsay_option(27, 106, Node004, 50);
-    gsay_option(27, 107, Node005, 50);
-    gsay_option(27, 108, GoBack, 50);
+    gsay_reply(SCRIPT_TGRDSTND, 105);
+    giq_option(1, SCRIPT_TGRDSTND, 106, Node004, NEUTRAL_REACTION);
+    giq_option(1, SCRIPT_TGRDSTND, 107, Node005, NEUTRAL_REACTION);
+    giq_option(1, SCRIPT_TGRDSTND, 108, GoBack, NEUTRAL_REACTION);
 end
 
 procedure Node004
 begin
     set_local_var(8, 1);
-    gsay_reply(27, 116);
-    gsay_option(27, 117, Node999, 50);
+    gsay_reply(SCRIPT_TGRDSTND, 116);
+    giq_option(1, SCRIPT_TGRDSTND, 117, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node005
 begin
-    gsay_reply(27, 109);
-    gsay_option(27, 110, Node998, 50);
+    gsay_reply(SCRIPT_TGRDSTND, 109);
+    giq_option(1, SCRIPT_TGRDSTND, 110, Node998, NEUTRAL_REACTION);
 end
 
 procedure Node006
 begin
-    gsay_reply(27, 111);
-    if ((get_critter_stat(dude_obj, 4) >= 8) or (has_skill(dude_obj, 14) >= 90)) then begin
-        gsay_option(27, 112, Node007, 50);
+    gsay_reply(SCRIPT_TGRDSTND, 111);
+    if ((dude_iq >= 8) or (dude_skill(SKILL_SPEECH) >= 90)) then begin
+        giq_option(1, SCRIPT_TGRDSTND, 112, Node007, NEUTRAL_REACTION);
     end
-    gsay_option(27, 114, Node005, 50);
-    gsay_option(27, 108, GoBack, 50);
+    giq_option(1, SCRIPT_TGRDSTND, 114, Node005, NEUTRAL_REACTION);
+    giq_option(1, SCRIPT_TGRDSTND, 108, GoBack, NEUTRAL_REACTION);
 end
 
 procedure Node007
 begin
     set_local_var(8, 1);
-    gsay_reply(27, 115);
-    gsay_option(27, 117, Node999, 50);
+    gsay_reply(SCRIPT_TGRDSTND, 115);
+    giq_option(1, SCRIPT_TGRDSTND, 117, Node999, NEUTRAL_REACTION);
 end
 
 procedure Node998
@@ -352,4 +352,3 @@ begin
     add_timer_event(self_obj, 8, 2);
     set_local_var(9, 1);
 end
-
