diff --git a/Mods/Olympus2207/Original/TTLIFTSC.ssl b/Mods/Olympus2207/ReDefine/TTLIFTSC.ssl
index b897753..0089156 100644
--- a/Mods/Olympus2207/Original/TTLIFTSC.ssl
+++ b/Mods/Olympus2207/ReDefine/TTLIFTSC.ssl
@@ -38,37 +38,37 @@ end
 procedure use_p_proc
 begin
     script_overrides;
-    display_msg(message_str(10, 869));
+    display_msg(message_str(SCRIPT_STTEXT, 869));
 end
 
 procedure use_obj_on_p_proc
 begin
     script_overrides;
     if (obj_pid(obj_being_used_with) == 303) then begin
-        if (cur_map_index == 4) then begin
+        if (CUR_MAP_TARTAR13) then begin
             if (local_var(0) == 0) then begin
                 set_local_var(0, 1);
-                if (global_var(370) == 0) then begin
-                    set_global_var(370, 1);
+                if (global_var(GVAR_NEW_RENO_JET_SOURCE) == 0) then begin
+                    set_global_var(GVAR_NEW_RENO_JET_SOURCE, 1);
                     play_gmovie(7);
                 end
             end
-            set_global_var(286, 1);
+            set_global_var(GVAR_NEW_RENO_WRIGHT_MYSTERY, 1);
             load_map("olhorton", -1);
         end
         else begin
-            if (cur_map_index == 56) then begin
-                set_global_var(286, 2);
+            if (CUR_MAP_OLHORTON) then begin
+                set_global_var(GVAR_NEW_RENO_WRIGHT_MYSTERY, 2);
                 load_map("tartar13", -1);
             end
         end
     end
     else begin
         if ((obj_pid(obj_being_used_with) == 330) or 331 or 332 or 333 or 334) then begin
-            display_msg(message_str(10, 870));
+            display_msg(message_str(SCRIPT_STTEXT, 870));
         end
         else begin
-            display_msg(message_str(10, random(60, 69)));
+            display_msg(message_str(SCRIPT_STTEXT, random(60, 69)));
         end
     end
 end
@@ -90,4 +90,3 @@ end
 procedure talk_p_proc
 begin
 end
-
