diff --git a/Mods/Olympus2207/Original/TWATSLVM.ssl b/Mods/Olympus2207/ReDefine/TWATSLVM.ssl
index e53bb52..59390e6 100644
--- a/Mods/Olympus2207/Original/TWATSLVM.ssl
+++ b/Mods/Olympus2207/ReDefine/TWATSLVM.ssl
@@ -29,8 +29,8 @@ begin
         set_local_var(1, 1);
         set_local_var(4, random(40, 55));
         set_local_var(5, random(0, 100));
-        critter_add_trait(self_obj, 1, 6, 2);
-        critter_add_trait(self_obj, 1, 5, 3);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 2);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_TARTAR_MEN);
         max_distance := random(20, 25);
         time_to_sleep := random(2200, 2300);
         time_to_work_1 := random(900, 915);
@@ -58,13 +58,13 @@ end
 procedure description_p_proc
 begin
     script_overrides;
-    display_msg(message_str(17, random(6, 9)));
+    display_msg(message_str(SCRIPT_TBIOSLVM, random(6, 9)));
 end
 
 procedure look_at_p_proc
 begin
     script_overrides;
-    display_msg(message_str(17, random(1, 5)));
+    display_msg(message_str(SCRIPT_TBIOSLVM, random(1, 5)));
 end
 
 procedure pickup_p_proc
@@ -81,14 +81,14 @@ end
 procedure talk_p_proc
 begin
     if ((art_anim(obj_art_fid(self_obj)) == 48) or (art_anim(obj_art_fid(self_obj)) == 49)) then begin
-        float_msg(self_obj, message_str(17, random(1010, 1013)), 0);
+        float_msg(self_obj, message_str(SCRIPT_TBIOSLVM, random(1010, 1013)), FLOAT_MSG_NORMAL);
     end
     else begin
         if (random(1, 2) == 1) then begin
-            float_msg(self_obj, message_str(17, random(1000, 1009)), 0);
+            float_msg(self_obj, message_str(SCRIPT_TBIOSLVM, random(1000, 1009)), FLOAT_MSG_NORMAL);
         end
         else begin
-            float_msg(self_obj, message_str(17, random(1000, 1009)), 8);
+            float_msg(self_obj, message_str(SCRIPT_TBIOSLVM, random(1000, 1009)), FLOAT_MSG_YELLOW);
         end
     end
 end
@@ -100,9 +100,9 @@ end
 procedure use_skill_on_p_proc
 begin
     if (action_being_used == 10) then begin
-        if (has_skill(dude_obj, 10) < (local_var(4) - ((get_critter_stat(dude_obj, 6) - 4) * 4))) then begin
+        if (dude_skill(SKILL_STEAL) < (local_var(4) - ((dude_luck - 4) * 4))) then begin
             script_overrides;
-            display_msg(message_str(10, random(200, 203)));
+            display_msg(message_str(SCRIPT_STTEXT, random(200, 203)));
         end
     end
 end
@@ -239,4 +239,3 @@ begin
         animation_is_called := 0;
     end
 end
-
