diff --git a/Mods/Olympus2207/Original/TWTRPUMP.ssl b/Mods/Olympus2207/ReDefine/TWTRPUMP.ssl
index b04183c..a7685a2 100644
--- a/Mods/Olympus2207/Original/TWTRPUMP.ssl
+++ b/Mods/Olympus2207/ReDefine/TWTRPUMP.ssl
@@ -21,7 +21,7 @@ procedure talk_p_proc;
 
 procedure start
 begin
-    if (global_var(31) == 2) then begin
+    if (global_var(GVAR_SLAVES_COUNT) == 2) then begin
         reg_anim_func(2, self_obj);
         reg_anim_func(1, 1);
         reg_anim_animate_forever(self_obj, 0);
@@ -31,17 +31,17 @@ end
 
 procedure description_p_proc
 begin
-    if (global_var(31) != 2) then begin
+    if (global_var(GVAR_SLAVES_COUNT) != 2) then begin
         script_overrides;
-        if (global_var(9) == 2) then begin
-            set_global_var(9, 3);
+        if (global_var(GVAR_KILL_EVIL_PLANTS) == 2) then begin
+            set_global_var(GVAR_KILL_EVIL_PLANTS, 3);
         end
-        if ((global_var(30) == 1) and (global_var(9) == 3)) then begin
-            set_global_var(30, 2);
-            display_msg(message_str(12, 143));
+        if ((global_var(GVAR_SLAVE_RUN) == 1) and (global_var(GVAR_KILL_EVIL_PLANTS) == 3)) then begin
+            set_global_var(GVAR_SLAVE_RUN, 2);
+            display_msg(message_str(SCRIPT_ITTEXT, 143));
         end
         else begin
-            display_msg(message_str(12, 140));
+            display_msg(message_str(SCRIPT_ITTEXT, 140));
         end
     end
 end
@@ -62,9 +62,9 @@ procedure use_obj_on_p_proc
 begin
     if (obj_pid(obj_being_used_with) == 296) then begin
         script_overrides;
-        set_global_var(31, 2);
+        set_global_var(GVAR_SLAVES_COUNT, 2);
         rm_obj_from_inven(dude_obj, obj_carrying_pid_obj(dude_obj, 296));
-        display_msg(message_str(12, 142));
+        display_msg(message_str(SCRIPT_ITTEXT, 142));
         reg_anim_func(2, self_obj);
         reg_anim_func(1, 1);
         reg_anim_animate_forever(self_obj, 0);
@@ -74,15 +74,15 @@ end
 
 procedure use_skill_on_p_proc
 begin
-    if ((action_being_used == 13) and (global_var(31) != 2)) then begin
+    if ((action_being_used == 13) and (global_var(GVAR_SLAVES_COUNT) != 2)) then begin
         script_overrides;
-        display_msg(message_str(12, 141));
+        display_msg(message_str(SCRIPT_ITTEXT, 141));
     end
 end
 
 procedure map_enter_p_proc
 begin
-    if (global_var(31) == 2) then begin
+    if (global_var(GVAR_SLAVES_COUNT) == 2) then begin
         reg_anim_func(2, self_obj);
         reg_anim_func(1, 1);
         reg_anim_animate_forever(self_obj, 0);
@@ -92,7 +92,7 @@ end
 
 procedure map_update_p_proc
 begin
-    if (global_var(31) == 2) then begin
+    if (global_var(GVAR_SLAVES_COUNT) == 2) then begin
         reg_anim_func(2, self_obj);
         reg_anim_func(1, 1);
         reg_anim_animate_forever(self_obj, 0);
@@ -103,4 +103,3 @@ end
 procedure talk_p_proc
 begin
 end
-
