diff --git a/Mods/Olympus2207/Original/WBPULT.ssl b/Mods/Olympus2207/ReDefine/WBPULT.ssl
index 90cb257..5345228 100644
--- a/Mods/Olympus2207/Original/WBPULT.ssl
+++ b/Mods/Olympus2207/ReDefine/WBPULT.ssl
@@ -27,10 +27,10 @@ procedure description_p_proc
 begin
     script_overrides;
     if ((map_var(0) == 1) or (map_var(1) == 1)) then begin
-        display_msg(message_str(307, 4));
+        display_mstr(4);
     end
     else begin
-        display_msg(message_str(307, 2));
+        display_mstr(2);
     end
 end
 
@@ -38,10 +38,10 @@ procedure look_at_p_proc
 begin
     script_overrides;
     if ((map_var(0) == 1) or (map_var(1) == 1)) then begin
-        display_msg(message_str(307, 3));
+        display_mstr(3);
     end
     else begin
-        display_msg(message_str(307, 1));
+        display_mstr(1);
     end
 end
 
@@ -50,11 +50,11 @@ begin
     if ((map_var(0) == 1) or (map_var(1) == 1)) then begin
     end
     else begin
-        if (metarule(49, target_obj) == 6) then begin
+        if (weapon_dmg_type(target_obj) == 6) then begin
             set_map_var(0, 1);
-            display_msg(message_str(307, 14));
-            give_exp_points(floor(30 / 5) + (get_critter_stat(dude_obj, 4) * floor(30 / 12)) + (get_critter_stat(dude_obj, 6) * floor(30 / 30)) + (floor(30 / 75) * (((get_critter_stat(dude_obj, 6) - 4) * 4) - (difficulty_level * 4))));
-            display_msg(message_str(10, random(140, 143)) + (floor(30 / 5) + (get_critter_stat(dude_obj, 4) * floor(30 / 12)) + (get_critter_stat(dude_obj, 6) * floor(30 / 30)) + (floor(30 / 75) * (((get_critter_stat(dude_obj, 6) - 4) * 4) - (difficulty_level * 4)))) + message_str(10, 144));
+            display_mstr(14);
+            give_exp_points(floor(30 / 5) + (dude_iq * floor(30 / 12)) + (dude_luck * floor(30 / 30)) + (floor(30 / 75) * (((dude_luck - 4) * 4) - (difficulty_level * 4))));
+            display_msg(message_str(SCRIPT_STTEXT, random(140, 143)) + (floor(30 / 5) + (dude_iq * floor(30 / 12)) + (dude_luck * floor(30 / 30)) + (floor(30 / 75) * (((dude_luck - 4) * 4) - (difficulty_level * 4)))) + message_str(SCRIPT_STTEXT, 144));
         end
     end
 end
@@ -65,8 +65,8 @@ begin
     end
     else begin
         script_overrides;
-        critter_dmg(dude_obj, random(3, 5), 4);
-        display_msg(message_str(307, random(5, 6)));
+        critter_dmg(dude_obj, random(3, 5), DMG_electrical);
+        display_mstr(random(5, 6));
     end
 end
 
@@ -81,23 +81,23 @@ begin
     else begin
         script_overrides;
         if ((action_being_used == 13) or (action_being_used == 9)) then begin
-            if (has_skill(dude_obj, action_being_used) >= 75) then begin
+            if (dude_skill(action_being_used) >= 75) then begin
                 set_map_var(0, 1);
-                display_msg(message_str(307, 12));
-                give_exp_points(floor(50 / 5) + (get_critter_stat(dude_obj, 4) * floor(50 / 12)) + (get_critter_stat(dude_obj, 6) * floor(50 / 30)) + (floor(50 / 75) * (((get_critter_stat(dude_obj, 6) - 4) * 4) - (difficulty_level * 4))));
-                display_msg(message_str(10, random(140, 143)) + (floor(50 / 5) + (get_critter_stat(dude_obj, 4) * floor(50 / 12)) + (get_critter_stat(dude_obj, 6) * floor(50 / 30)) + (floor(50 / 75) * (((get_critter_stat(dude_obj, 6) - 4) * 4) - (difficulty_level * 4)))) + message_str(10, 144));
+                display_mstr(12);
+                give_exp_points(floor(50 / 5) + (dude_iq * floor(50 / 12)) + (dude_luck * floor(50 / 30)) + (floor(50 / 75) * (((dude_luck - 4) * 4) - (difficulty_level * 4))));
+                display_msg(message_str(SCRIPT_STTEXT, random(140, 143)) + (floor(50 / 5) + (dude_iq * floor(50 / 12)) + (dude_luck * floor(50 / 30)) + (floor(50 / 75) * (((dude_luck - 4) * 4) - (difficulty_level * 4)))) + message_str(SCRIPT_STTEXT, 144));
             end
             else begin
-                if (has_skill(dude_obj, action_being_used) >= 55) then begin
+                if (dude_skill(action_being_used) >= 55) then begin
                     set_map_var(0, 1);
-                    critter_dmg(dude_obj, random(13, 15), 4);
-                    display_msg(message_str(307, 11));
-                    give_exp_points(floor(25 / 5) + (get_critter_stat(dude_obj, 4) * floor(25 / 12)) + (get_critter_stat(dude_obj, 6) * floor(25 / 30)) + (floor(25 / 75) * (((get_critter_stat(dude_obj, 6) - 4) * 4) - (difficulty_level * 4))));
-                    display_msg(message_str(10, random(140, 143)) + (floor(25 / 5) + (get_critter_stat(dude_obj, 4) * floor(25 / 12)) + (get_critter_stat(dude_obj, 6) * floor(25 / 30)) + (floor(25 / 75) * (((get_critter_stat(dude_obj, 6) - 4) * 4) - (difficulty_level * 4)))) + message_str(10, 144));
+                    critter_dmg(dude_obj, random(13, 15), DMG_electrical);
+                    display_mstr(11);
+                    give_exp_points(floor(25 / 5) + (dude_iq * floor(25 / 12)) + (dude_luck * floor(25 / 30)) + (floor(25 / 75) * (((dude_luck - 4) * 4) - (difficulty_level * 4))));
+                    display_msg(message_str(SCRIPT_STTEXT, random(140, 143)) + (floor(25 / 5) + (dude_iq * floor(25 / 12)) + (dude_luck * floor(25 / 30)) + (floor(25 / 75) * (((dude_luck - 4) * 4) - (difficulty_level * 4)))) + message_str(SCRIPT_STTEXT, 144));
                 end
                 else begin
-                    critter_dmg(dude_obj, random(3, 5), 4);
-                    display_msg(message_str(307, 10));
+                    critter_dmg(dude_obj, random(3, 5), DMG_electrical);
+                    display_mstr(10);
                 end
             end
         end
@@ -115,4 +115,3 @@ end
 procedure talk_p_proc
 begin
 end
-
