diff --git a/Mods/Olympus2207/Original/XCARIDR1.ssl b/Mods/Olympus2207/ReDefine/XCARIDR1.ssl
index ded27ec..90b300e 100644
--- a/Mods/Olympus2207/Original/XCARIDR1.ssl
+++ b/Mods/Olympus2207/ReDefine/XCARIDR1.ssl
@@ -29,95 +29,95 @@ begin
     if (local_var(1) == 0) then begin
         set_local_var(3, tile_num(self_obj));
         set_local_var(1, 1);
-        critter_add_trait(self_obj, 1, 6, 15);
-        critter_add_trait(self_obj, 1, 5, 25);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 15);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_RAIDER_BERSERK);
         if (random(0, 2) == 1) then begin
-            metarule3(107, self_obj, 16777281, 0);
+            art_change_fid_num(self_obj, 16777281);
         end
         else begin
             if (random(0, 1) == 1) then begin
-                metarule3(107, self_obj, 16777274, 0);
+                art_change_fid_num(self_obj, 16777274);
             end
         end
         if (random(0, 1) == 1) then begin
             if (random(0, 100) <= 40) then begin
-                add_obj_to_inven(self_obj, create_object_sid(164, 1, 2, -1));
-                add_mult_objs_to_inven(self_obj, create_object_sid(2, 1, 2, -1), random(21, 38));
+                add_obj_to_inven(self_obj, create_object(164, 1, 2));
+                add_mult_objs_to_inven(self_obj, create_object(2, 1, 2), random(21, 38));
             end
             else begin
                 if (random(0, 100) <= 40) then begin
-                    add_obj_to_inven(self_obj, create_object_sid(150, 1, 2, -1));
+                    add_obj_to_inven(self_obj, create_object(150, 1, 2));
                 end
                 else begin
-                    add_obj_to_inven(self_obj, create_object_sid(159, 1, 2, -1));
+                    add_obj_to_inven(self_obj, create_object(159, 1, 2));
                 end
-                add_mult_objs_to_inven(self_obj, create_object_sid(22, 1, 2, -1), random(2, 4));
+                add_mult_objs_to_inven(self_obj, create_object(22, 1, 2), random(2, 4));
             end
             if (random(0, 100) <= 33) then begin
-                add_obj_to_inven(self_obj, create_object_sid(112, 1, 2, -1));
+                add_obj_to_inven(self_obj, create_object(112, 1, 2));
             end
             else begin
-                add_obj_to_inven(self_obj, create_object_sid(109, 1, 2, -1));
+                add_obj_to_inven(self_obj, create_object(109, 1, 2));
             end
             if (random(0, 100) <= 33) then begin
-                add_mult_objs_to_inven(self_obj, create_object_sid(41, 1, 2, -1), random(5, 20));
+                add_mult_objs_to_inven(self_obj, create_object(41, 1, 2), random(5, 20));
             end
             if (random(0, 100) <= 25) then begin
-                add_mult_objs_to_inven(self_obj, create_object_sid(411, 1, 2, -1), random(3, 7));
+                add_mult_objs_to_inven(self_obj, create_object(411, 1, 2), random(3, 7));
             end
             if (random(0, 100) <= 10) then begin
-                add_obj_to_inven(self_obj, create_object_sid(233, 1, 2, -1));
+                add_obj_to_inven(self_obj, create_object(233, 1, 2));
             end
             if (random(0, 100) <= 9) then begin
-                add_obj_to_inven(self_obj, create_object_sid(243, 1, 2, -1));
+                add_obj_to_inven(self_obj, create_object(243, 1, 2));
             end
             if (random(0, 100) <= 8) then begin
-                add_obj_to_inven(self_obj, create_object_sid(237, 1, 2, -1));
+                add_obj_to_inven(self_obj, create_object(237, 1, 2));
             end
             if (random(0, 100) <= 7) then begin
-                add_obj_to_inven(self_obj, create_object_sid(234, 1, 2, -1));
+                add_obj_to_inven(self_obj, create_object(234, 1, 2));
             end
             if (random(0, 100) <= 6) then begin
-                add_obj_to_inven(self_obj, create_object_sid(232, 1, 2, -1));
+                add_obj_to_inven(self_obj, create_object(232, 1, 2));
             end
         end
         else begin
             if (random(0, 100) <= 33) then begin
-                add_obj_to_inven(self_obj, create_object_sid(169, 1, 2, -1));
-                add_mult_objs_to_inven(self_obj, create_object_sid(5, 1, 2, -1), random(21, 38));
-                wield_obj_critter(self_obj, obj_carrying_pid_obj(self_obj, 169));
+                add_obj_to_inven(self_obj, create_object(169, 1, 2));
+                add_mult_objs_to_inven(self_obj, create_object(5, 1, 2), random(21, 38));
+                wield_obj(obj_carrying_pid_obj(self_obj, 169));
             end
             else begin
-                add_obj_to_inven(self_obj, create_object_sid(177, 1, 2, -1));
-                add_mult_objs_to_inven(self_obj, create_object_sid(14, 1, 2, -1), random(10, 18));
-                wield_obj_critter(self_obj, obj_carrying_pid_obj(self_obj, 177));
+                add_obj_to_inven(self_obj, create_object(177, 1, 2));
+                add_mult_objs_to_inven(self_obj, create_object(14, 1, 2), random(10, 18));
+                wield_obj(obj_carrying_pid_obj(self_obj, 177));
             end
             if (random(0, 100) <= 33) then begin
-                add_obj_to_inven(self_obj, create_object_sid(221, 1, 2, -1));
+                add_obj_to_inven(self_obj, create_object(221, 1, 2));
             end
             else begin
-                add_obj_to_inven(self_obj, create_object_sid(106, 1, 2, -1));
+                add_obj_to_inven(self_obj, create_object(106, 1, 2));
             end
             if (random(0, 100) <= 33) then begin
-                add_mult_objs_to_inven(self_obj, create_object_sid(41, 1, 2, -1), random(10, 40));
+                add_mult_objs_to_inven(self_obj, create_object(41, 1, 2), random(10, 40));
             end
             if (random(0, 100) <= 25) then begin
-                add_mult_objs_to_inven(self_obj, create_object_sid(63, 1, 2, -1), random(2, 4));
+                add_mult_objs_to_inven(self_obj, create_object(63, 1, 2), random(2, 4));
             end
             if (random(0, 100) <= 10) then begin
-                add_obj_to_inven(self_obj, create_object_sid(64, 1, 2, -1));
+                add_obj_to_inven(self_obj, create_object(64, 1, 2));
             end
             if (random(0, 100) <= 9) then begin
-                add_obj_to_inven(self_obj, create_object_sid(431, 1, 2, -1));
+                add_obj_to_inven(self_obj, create_object(431, 1, 2));
             end
             if (random(0, 100) <= 8) then begin
-                add_obj_to_inven(self_obj, create_object_sid(236, 1, 2, -1));
+                add_obj_to_inven(self_obj, create_object(236, 1, 2));
             end
             if (random(0, 100) <= 7) then begin
-                add_obj_to_inven(self_obj, create_object_sid(229, 1, 2, -1));
+                add_obj_to_inven(self_obj, create_object(229, 1, 2));
             end
             if (random(0, 100) <= 6) then begin
-                add_obj_to_inven(self_obj, create_object_sid(230, 1, 2, -1));
+                add_obj_to_inven(self_obj, create_object(230, 1, 2));
             end
         end
     end
@@ -171,9 +171,9 @@ end
 
 procedure map_enter_p_proc
 begin
-    if (not(metarule(22, 0)) == 1) then begin
-        if (global_var(204) != 0) then begin
-            move_to(self_obj, tile_num_in_direction(tile_num_in_direction(tile_num(dude_obj), random(0, 2), random(6, 7)), random(0, 5), random(1, 2)), 0);
+    if (not(is_loading_game) == 1) then begin
+        if (global_var(GVAR_ENEMY_DUNTON) != 0) then begin
+            move_to(self_obj, tile_num_in_direction(tile_num_in_direction(dude_tile, random(0, 2), random(6, 7)), random(0, 5), random(1, 2)), 0);
             anim(self_obj, 1000, 4);
             add_timer_event(self_obj, random(12, 18), 1);
         end
@@ -186,4 +186,3 @@ begin
         attack_setup(self_obj, dude_obj);
     end
 end
-
