diff --git a/Mods/Olympus2207/Original/XRCANIBL.ssl b/Mods/Olympus2207/ReDefine/XRCANIBL.ssl
index 7210ebf..37198e9 100644
--- a/Mods/Olympus2207/Original/XRCANIBL.ssl
+++ b/Mods/Olympus2207/ReDefine/XRCANIBL.ssl
@@ -32,82 +32,82 @@ begin
         set_local_var(1, 1);
         set_local_var(4, random(60, 70));
         set_local_var(5, random(30, 40));
-        critter_add_trait(self_obj, 1, 6, 10);
-        if (get_critter_stat(self_obj, 34) == 0) then begin
-            critter_add_trait(self_obj, 1, 5, 41);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 10);
+        if (self_gender == 0) then begin
+            critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_PRIMITIVE_STUPID_MALE);
         end
         else begin
-            critter_add_trait(self_obj, 1, 5, 42);
+            critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_PRIMITIVE_STUPID_FEMALE);
         end
         if (random(0, 100) <= 20) then begin
-            add_mult_objs_to_inven(self_obj, create_object_sid(41, 1, 2, -1), random(1, 10));
+            add_mult_objs_to_inven(self_obj, create_object(41, 1, 2), random(1, 10));
         end
         if (random(0, 100) <= 15) then begin
-            add_mult_objs_to_inven(self_obj, create_object_sid(238, 1, 2, -1), random(1, 2));
+            add_mult_objs_to_inven(self_obj, create_object(238, 1, 2), random(1, 2));
         end
         if (random(0, 100) <= 15) then begin
-            add_mult_objs_to_inven(self_obj, create_object_sid(213, 1, 2, -1), random(1, 2));
+            add_mult_objs_to_inven(self_obj, create_object(213, 1, 2), random(1, 2));
         end
         if (random(0, 100) <= 15) then begin
-            add_obj_to_inven(self_obj, create_object_sid(35, 1, 2, -1));
+            add_obj_to_inven(self_obj, create_object(35, 1, 2));
         end
         if (random(0, 100) <= 14) then begin
-            add_obj_to_inven(self_obj, create_object_sid(65, 1, 2, -1));
+            add_obj_to_inven(self_obj, create_object(65, 1, 2));
         end
         if (random(0, 100) <= 13) then begin
-            add_obj_to_inven(self_obj, create_object_sid(218, 1, 2, -1));
+            add_obj_to_inven(self_obj, create_object(218, 1, 2));
         end
         if (random(0, 100) <= 12) then begin
-            add_obj_to_inven(self_obj, create_object_sid(70, 1, 2, -1));
+            add_obj_to_inven(self_obj, create_object(70, 1, 2));
         end
         if (random(0, 100) <= 11) then begin
-            add_obj_to_inven(self_obj, create_object_sid(106, 1, 2, -1));
+            add_obj_to_inven(self_obj, create_object(106, 1, 2));
         end
         if (random(0, 100) <= 10) then begin
-            add_obj_to_inven(self_obj, create_object_sid(244, 1, 2, -1));
+            add_obj_to_inven(self_obj, create_object(244, 1, 2));
         end
         if (random(0, 100) <= 10) then begin
-            add_obj_to_inven(self_obj, create_object_sid(242, 1, 2, -1));
+            add_obj_to_inven(self_obj, create_object(242, 1, 2));
         end
         if (random(0, 100) <= 10) then begin
-            add_obj_to_inven(self_obj, create_object_sid(414, 1, 2, -1));
+            add_obj_to_inven(self_obj, create_object(414, 1, 2));
         end
         if (random(0, 100) <= 10) then begin
-            add_obj_to_inven(self_obj, create_object_sid(422, 1, 2, -1));
+            add_obj_to_inven(self_obj, create_object(422, 1, 2));
         end
         if (random(0, 100) <= 9) then begin
-            add_obj_to_inven(self_obj, create_object_sid(423, 1, 2, -1));
+            add_obj_to_inven(self_obj, create_object(423, 1, 2));
         end
         if (random(0, 100) <= 8) then begin
-            add_obj_to_inven(self_obj, create_object_sid(426, 1, 2, -1));
+            add_obj_to_inven(self_obj, create_object(426, 1, 2));
         end
         if (random(0, 100) <= 7) then begin
-            add_obj_to_inven(self_obj, create_object_sid(433, 1, 2, -1));
+            add_obj_to_inven(self_obj, create_object(433, 1, 2));
         end
         if (random(0, 100) <= 6) then begin
-            add_obj_to_inven(self_obj, create_object_sid(421, 1, 2, -1));
+            add_obj_to_inven(self_obj, create_object(421, 1, 2));
         end
         if (random(0, 100) <= 5) then begin
-            add_obj_to_inven(self_obj, create_object_sid(427, 1, 2, -1));
+            add_obj_to_inven(self_obj, create_object(427, 1, 2));
         end
         if (random(0, 100) <= 4) then begin
-            add_obj_to_inven(self_obj, create_object_sid(236, 1, 2, -1));
+            add_obj_to_inven(self_obj, create_object(236, 1, 2));
         end
         if (random(0, 100) <= 3) then begin
-            add_obj_to_inven(self_obj, create_object_sid(74, 1, 2, -1));
+            add_obj_to_inven(self_obj, create_object(74, 1, 2));
         end
         i := random(1, 3);
         if (i == 1) then begin
-            add_obj_to_inven(self_obj, create_object_sid(88, 1, 2, -1));
-            wield_obj_critter(self_obj, obj_carrying_pid_obj(self_obj, 88));
+            add_obj_to_inven(self_obj, create_object(88, 1, 2));
+            wield_obj(obj_carrying_pid_obj(self_obj, 88));
         end
         if (i == 2) then begin
-            add_obj_to_inven(self_obj, create_object_sid(66, 1, 2, -1));
-            wield_obj_critter(self_obj, obj_carrying_pid_obj(self_obj, 66));
+            add_obj_to_inven(self_obj, create_object(66, 1, 2));
+            wield_obj(obj_carrying_pid_obj(self_obj, 66));
         end
         if (i == 3) then begin
-            add_obj_to_inven(self_obj, create_object_sid(71, 1, 2, -1));
-            wield_obj_critter(self_obj, obj_carrying_pid_obj(self_obj, 71));
+            add_obj_to_inven(self_obj, create_object(71, 1, 2));
+            wield_obj(obj_carrying_pid_obj(self_obj, 71));
         end
         max_distance := random(5, 8);
     end
@@ -120,14 +120,14 @@ begin
         add_timer_event(self_obj, random(40, 120), 1);
     end
     if ((obj_can_see_obj(self_obj, dude_obj) == 1) and not(combat_is_initialized)) then begin
-        if ((obj_pid(critter_inven_obj(dude_obj, 0)) == 129) or (obj_pid(critter_inven_obj(dude_obj, 0)) == 391) or (obj_pid(critter_inven_obj(dude_obj, 0)) == 127) or (obj_pid(critter_inven_obj(dude_obj, 0)) == 128)) then begin
+        if ((obj_pid(critter_inven_obj(dude_obj, INVEN_TYPE_WORN)) == 129) or (obj_pid(critter_inven_obj(dude_obj, INVEN_TYPE_WORN)) == 391) or (obj_pid(critter_inven_obj(dude_obj, INVEN_TYPE_WORN)) == 127) or (obj_pid(critter_inven_obj(dude_obj, INVEN_TYPE_WORN)) == 128)) then begin
         end
         else begin
             set_local_var(2, 1);
         end
     end
     if (obj_can_see_obj(self_obj, dude_obj)) then begin
-        if ((local_var(2) == 1) or (global_var(628) <= -31)) then begin
+        if ((local_var(2) == 1) or (global_var(GVAR_TOWN_REP_NAVARRO) <= -31)) then begin
             set_local_var(2, 0);
             attack_setup(self_obj, dude_obj);
         end
@@ -141,7 +141,7 @@ end
 procedure description_p_proc
 begin
     script_overrides;
-    display_msg(message_str(477, 1));
+    display_mstr(1);
 end
 
 procedure look_at_p_proc
@@ -162,10 +162,10 @@ end
 procedure talk_p_proc
 begin
     if (not(anim_busy(self_obj))) then begin
-        anim(self_obj, 1000, rotation_to_tile(tile_num(self_obj), tile_num(dude_obj)));
+        anim(self_obj, 1000, rotation_to_tile(tile_num(self_obj), dude_tile));
     end
-    if ((obj_pid(critter_inven_obj(dude_obj, 0)) == 129) or (obj_pid(critter_inven_obj(dude_obj, 0)) == 391) or (obj_pid(critter_inven_obj(dude_obj, 0)) == 127) or (obj_pid(critter_inven_obj(dude_obj, 0)) == 128)) then begin
-        float_msg(self_obj, message_str(477, random(10, 14)), 0);
+    if ((obj_pid(critter_inven_obj(dude_obj, INVEN_TYPE_WORN)) == 129) or (obj_pid(critter_inven_obj(dude_obj, INVEN_TYPE_WORN)) == 391) or (obj_pid(critter_inven_obj(dude_obj, INVEN_TYPE_WORN)) == 127) or (obj_pid(critter_inven_obj(dude_obj, INVEN_TYPE_WORN)) == 128)) then begin
+        float_msg(self_obj, mstr(random(10, 14)), FLOAT_MSG_NORMAL);
     end
     else begin
         set_local_var(2, 1);
@@ -184,51 +184,51 @@ begin
             set_sfall_global(13, 1);
         end
         else begin
-            if (has_trait(2, dude_obj, 13)) then begin
-                if (global_var(81) < get_pc_stat(1)) then begin
+            if (has_trait(TRAIT_TRAIT, dude_obj, TRAIT_sex_appeal)) then begin
+                if (global_var(GVAR_VAULT_CITIZENSHIP) < get_pc_stat(PCSTAT_level)) then begin
                     if (is_success(roll_vs_skill(dude_obj, 10, 0))) then begin
-                        set_global_var(98, global_var(98) + 3);
+                        set_global_var(GVAR_QUEST_FIND_THOMAS, global_var(GVAR_QUEST_FIND_THOMAS) + 3);
                     end
                     else begin
-                        set_global_var(98, global_var(98) + 1);
+                        set_global_var(GVAR_QUEST_FIND_THOMAS, global_var(GVAR_QUEST_FIND_THOMAS) + 1);
                     end
-                    if (global_var(98) >= (2 * get_pc_stat(1))) then begin
-                        set_global_var(98, 0);
-                        set_global_var(81, global_var(81) + 1);
-                        critter_mod_skill(dude_obj, 10, 1);
-                        display_msg(message_str(10, 410));
+                    if (global_var(GVAR_QUEST_FIND_THOMAS) >= (2 * get_pc_stat(PCSTAT_level))) then begin
+                        set_global_var(GVAR_QUEST_FIND_THOMAS, 0);
+                        set_global_var(GVAR_VAULT_CITIZENSHIP, global_var(GVAR_VAULT_CITIZENSHIP) + 1);
+                        critter_mod_skill(dude_obj, SKILL_STEAL, 1);
+                        display_msg(message_str(SCRIPT_STTEXT, 410));
                     end
                 end
             end
             steal_perk := 0;
             steal_gloves := 0;
-            if (has_trait(0, dude_obj, 37) != 0) then begin
+            if (has_trait(TRAIT_PERK, dude_obj, PERK_pickpocket) != 0) then begin
                 steal_perk := steal_perk + 15;
             end
-            if (has_trait(0, dude_obj, 33) != 0) then begin
+            if (has_trait(TRAIT_PERK, dude_obj, PERK_master_thief) != 0) then begin
                 steal_perk := steal_perk + 30;
             end
-            if ((obj_pid(critter_inven_obj(dude_obj, 1)) == 245) or (obj_pid(critter_inven_obj(dude_obj, 2)) == 245)) then begin
+            if ((obj_pid(critter_inven_obj(dude_obj, INVEN_TYPE_RIGHT_HAND)) == 245) or (obj_pid(critter_inven_obj(dude_obj, INVEN_TYPE_LEFT_HAND)) == 245)) then begin
                 steal_gloves := 10;
             end
-            if (random(0, 100) <= ((has_skill(dude_obj, 10) / 2) + (((3 * get_critter_stat(dude_obj, 5)) + (((get_critter_stat(dude_obj, 6) - 4) * 4) - (difficulty_level * 4)) + (2 * steal_perk) + steal_gloves) / (local_var(4) / 25)))) then begin
-                display_msg(message_str(10, random(145, 148)));
+            if (random(0, 100) <= ((dude_skill(SKILL_STEAL) / 2) + (((3 * get_critter_stat(dude_obj, STAT_ag)) + (((dude_luck - 4) * 4) - (difficulty_level * 4)) + (2 * steal_perk) + steal_gloves) / (local_var(4) / 25)))) then begin
+                display_msg(message_str(SCRIPT_STTEXT, random(145, 148)));
             end
             else begin
                 script_overrides;
-                if (random(0, 100) <= (25 - floor((((get_critter_stat(dude_obj, 6) - 4) * 4) - (difficulty_level * 4)) / 3))) then begin
-                    float_msg(self_obj, message_str(10, random(206, 209)), 2);
+                if (random(0, 100) <= (25 - floor((((dude_luck - 4) * 4) - (difficulty_level * 4)) / 3))) then begin
+                    float_msg(self_obj, message_str(SCRIPT_STTEXT, random(206, 209)), FLOAT_MSG_RED);
                     set_local_var(2, 1);
                 end
                 else begin
-                    display_msg(message_str(10, random(200, 203)));
+                    display_msg(message_str(SCRIPT_STTEXT, random(200, 203)));
                 end
             end
         end
     end
     if (action_being_used == 12) then begin
         script_overrides;
-        display_msg(message_str(10, 204) + ((has_skill(dude_obj, 10) / 2) + (((3 * get_critter_stat(dude_obj, 5)) + (((get_critter_stat(dude_obj, 6) - 4) * 4) - (difficulty_level * 4)) + (2 * steal_perk) + steal_gloves) / (local_var(4) / 25))) + message_str(10, 205));
+        display_msg(message_str(SCRIPT_STTEXT, 204) + ((dude_skill(SKILL_STEAL) / 2) + (((3 * get_critter_stat(dude_obj, STAT_ag)) + (((dude_luck - 4) * 4) - (difficulty_level * 4)) + (2 * steal_perk) + steal_gloves) / (local_var(4) / 25))) + message_str(SCRIPT_STTEXT, 205));
     end
 end
 
@@ -255,4 +255,3 @@ begin
         event_is_called := 0;
     end
 end
-
