diff --git a/Mods/Olympus2207/Original/YLVENTER.ssl b/Mods/Olympus2207/ReDefine/YLVENTER.ssl
index 8838eba..09652a1 100644
--- a/Mods/Olympus2207/Original/YLVENTER.ssl
+++ b/Mods/Olympus2207/ReDefine/YLVENTER.ssl
@@ -18,14 +18,14 @@ end
 
 procedure map_enter_p_proc
 begin
-    if (global_var(204) != 0) then begin
+    if (global_var(GVAR_ENEMY_DUNTON) != 0) then begin
         game_time_advance(game_ticks(3600 * random(4, 6)));
     end
-    if (metarule(14, 0)) then begin
-        display_msg(message_str(10, 300 + cur_map_index));
+    if (map_first_run) then begin
+        display_msg(message_str(SCRIPT_STTEXT, 300 + cur_map_index));
     end
-    if (metarule(14, 0) and not(metarule(17, 5) == 1)) then begin
-        mark_area_known(0, 5, 1);
+    if (map_first_run and not(town_known(5) == MARK_STATE_KNOWN)) then begin
+        mark_area_known(MARK_TYPE_TOWN, 5, MARK_STATE_KNOWN);
     end
     if ((get_month >= 3) and (get_month < 5)) then begin
         if ((game_time_hour >= 600) and (game_time_hour < (600 + 100))) then begin
@@ -184,4 +184,3 @@ begin
         end
     end
 end
-
