diff --git a/Mods/Olympus2207/Original/YNWFABRC.ssl b/Mods/Olympus2207/ReDefine/YNWFABRC.ssl
index dbbd601..ec708b6 100644
--- a/Mods/Olympus2207/Original/YNWFABRC.ssl
+++ b/Mods/Olympus2207/ReDefine/YNWFABRC.ssl
@@ -18,10 +18,10 @@ end
 
 procedure map_enter_p_proc
 begin
-    if (metarule(14, 0)) then begin
-        display_msg(message_str(10, 300 + cur_map_index));
+    if (map_first_run) then begin
+        display_msg(message_str(SCRIPT_STTEXT, 300 + cur_map_index));
     end
-    if (elevation(dude_obj) == 2) then begin
+    if (dude_elevation == 2) then begin
         set_light_level(50);
     end
     else begin
@@ -31,11 +31,10 @@ end
 
 procedure map_update_p_proc
 begin
-    if (elevation(dude_obj) == 2) then begin
+    if (dude_elevation == 2) then begin
         set_light_level(50);
     end
     else begin
         set_light_level(75);
     end
 end
-
