diff --git a/Mods/Olympus2207/Original/YOACAVE.ssl b/Mods/Olympus2207/ReDefine/YOACAVE.ssl
index afb24c2..3d11de1 100644
--- a/Mods/Olympus2207/Original/YOACAVE.ssl
+++ b/Mods/Olympus2207/ReDefine/YOACAVE.ssl
@@ -18,8 +18,8 @@ end
 
 procedure map_enter_p_proc
 begin
-    if (metarule(14, 0)) then begin
-        display_msg(message_str(10, 300 + cur_map_index));
+    if (map_first_run) then begin
+        display_msg(message_str(SCRIPT_STTEXT, 300 + cur_map_index));
     end
     set_light_level(30);
 end
@@ -28,4 +28,3 @@ procedure map_update_p_proc
 begin
     set_light_level(30);
 end
-
