diff --git a/Mods/Olympus2207/Original/YOLHRTON.ssl b/Mods/Olympus2207/ReDefine/YOLHRTON.ssl
index 382d81d..fd32e5b 100644
--- a/Mods/Olympus2207/Original/YOLHRTON.ssl
+++ b/Mods/Olympus2207/ReDefine/YOLHRTON.ssl
@@ -18,20 +18,20 @@ end
 
 procedure map_enter_p_proc
 begin
-    if (metarule(14, 0)) then begin
-        display_msg(message_str(10, 300 + cur_map_index));
+    if (map_first_run) then begin
+        display_msg(message_str(SCRIPT_STTEXT, 300 + cur_map_index));
     end
-    if (global_var(370) == 0) then begin
-        set_global_var(370, 1);
+    if (global_var(GVAR_NEW_RENO_JET_SOURCE) == 0) then begin
+        set_global_var(GVAR_NEW_RENO_JET_SOURCE, 1);
     end
-    if (global_var(286) == 1) then begin
-        set_global_var(286, 0);
+    if (global_var(GVAR_NEW_RENO_WRIGHT_MYSTERY) == 1) then begin
+        set_global_var(GVAR_NEW_RENO_WRIGHT_MYSTERY, 0);
         move_to(dude_obj, 20079, 0);
     end
     set_light_level(100);
-    if (global_var(287) != 0) then begin
-        move_to(dude_obj, global_var(287), 0);
-        set_global_var(287, 0);
+    if (global_var(GVAR_DEN_SLAVER_WARNINGS) != 0) then begin
+        move_to(dude_obj, global_var(GVAR_DEN_SLAVER_WARNINGS), 0);
+        set_global_var(GVAR_DEN_SLAVER_WARNINGS, 0);
         fadeIn(700);
         game_ui_enable;
     end
@@ -41,4 +41,3 @@ procedure map_update_p_proc
 begin
     set_light_level(100);
 end
-
