diff --git a/Mods/Olympus2207/Original/YOLTOP.ssl b/Mods/Olympus2207/ReDefine/YOLTOP.ssl
index efa11f8..3dc8333 100644
--- a/Mods/Olympus2207/Original/YOLTOP.ssl
+++ b/Mods/Olympus2207/ReDefine/YOLTOP.ssl
@@ -18,13 +18,13 @@ end
 
 procedure map_enter_p_proc
 begin
-    if (metarule(14, 0)) then begin
-        display_msg(message_str(10, 300 + cur_map_index));
+    if (map_first_run) then begin
+        display_msg(message_str(SCRIPT_STTEXT, 300 + cur_map_index));
     end
     set_light_level(100);
-    if (global_var(287) != 0) then begin
-        move_to(dude_obj, global_var(287), 0);
-        set_global_var(287, 0);
+    if (global_var(GVAR_DEN_SLAVER_WARNINGS) != 0) then begin
+        move_to(dude_obj, global_var(GVAR_DEN_SLAVER_WARNINGS), 0);
+        set_global_var(GVAR_DEN_SLAVER_WARNINGS, 0);
         fadeIn(700);
         game_ui_enable;
     end
@@ -34,4 +34,3 @@ procedure map_update_p_proc
 begin
     set_light_level(100);
 end
-
