diff --git a/Mods/Olympus2207/Original/YPSMARKT.ssl b/Mods/Olympus2207/ReDefine/YPSMARKT.ssl
index 07c6dff..94abaf9 100644
--- a/Mods/Olympus2207/Original/YPSMARKT.ssl
+++ b/Mods/Olympus2207/ReDefine/YPSMARKT.ssl
@@ -18,10 +18,10 @@ end
 
 procedure map_enter_p_proc
 begin
-    if (metarule(14, 0)) then begin
-        display_msg(message_str(10, 300 + cur_map_index));
+    if (map_first_run) then begin
+        display_msg(message_str(SCRIPT_STTEXT, 300 + cur_map_index));
     end
-    if (elevation(dude_obj) == 0) then begin
+    if (dude_elevation == 0) then begin
         if ((get_month >= 3) and (get_month < 5)) then begin
             if ((game_time_hour >= 600) and (game_time_hour < (600 + 100))) then begin
                 set_light_level(game_time_hour - 600 + 20);
@@ -106,7 +106,7 @@ end
 
 procedure map_update_p_proc
 begin
-    if (elevation(dude_obj) == 0) then begin
+    if (dude_elevation == 0) then begin
         if ((get_month >= 3) and (get_month < 5)) then begin
             if ((game_time_hour >= 600) and (game_time_hour < (600 + 100))) then begin
                 set_light_level(game_time_hour - 600 + 20);
@@ -188,4 +188,3 @@ begin
         set_light_level(30);
     end
 end
-
