diff --git a/Mods/Olympus2207/Original/YSJLABS.ssl b/Mods/Olympus2207/ReDefine/YSJLABS.ssl
index a6e2025..f6f8148 100644
--- a/Mods/Olympus2207/Original/YSJLABS.ssl
+++ b/Mods/Olympus2207/ReDefine/YSJLABS.ssl
@@ -18,15 +18,15 @@ end
 
 procedure map_enter_p_proc
 begin
-    if (metarule(14, 0)) then begin
-        display_msg(message_str(10, 300 + cur_map_index));
+    if (map_first_run) then begin
+        display_msg(message_str(SCRIPT_STTEXT, 300 + cur_map_index));
     end
-    if (metarule(14, 0) and metarule(19, 54) and (global_var(262) == 0)) then begin
-        set_global_var(262, 1);
-        give_exp_points(floor(2500 / 5) + (get_critter_stat(dude_obj, 4) * floor(2500 / 12)) + (get_critter_stat(dude_obj, 6) * floor(2500 / 30)) + (floor(2500 / 75) * (((get_critter_stat(dude_obj, 6) - 4) * 4) - (difficulty_level * 4))));
-        display_msg(message_str(10, random(140, 143)) + (floor(2500 / 5) + (get_critter_stat(dude_obj, 4) * floor(2500 / 12)) + (get_critter_stat(dude_obj, 6) * floor(2500 / 30)) + (floor(2500 / 75) * (((get_critter_stat(dude_obj, 6) - 4) * 4) - (difficulty_level * 4)))) + message_str(10, 144));
-        critter_mod_skill(dude_obj, 17, 25);
-        display_msg(message_str(10, 804));
+    if (map_first_run and map_is_known(54) and (global_var(GVAR_KARMA_INDIANA) == 0)) then begin
+        set_global_var(GVAR_KARMA_INDIANA, 1);
+        give_exp_points(floor(2500 / 5) + (dude_iq * floor(2500 / 12)) + (dude_luck * floor(2500 / 30)) + (floor(2500 / 75) * (((dude_luck - 4) * 4) - (difficulty_level * 4))));
+        display_msg(message_str(SCRIPT_STTEXT, random(140, 143)) + (floor(2500 / 5) + (dude_iq * floor(2500 / 12)) + (dude_luck * floor(2500 / 30)) + (floor(2500 / 75) * (((dude_luck - 4) * 4) - (difficulty_level * 4)))) + message_str(SCRIPT_STTEXT, 144));
+        critter_mod_skill(dude_obj, SKILL_OUTDOORSMAN, 25);
+        display_msg(message_str(SCRIPT_STTEXT, 804));
     end
     set_light_level(100);
 end
@@ -35,4 +35,3 @@ procedure map_update_p_proc
 begin
     set_light_level(100);
 end
-
