diff --git a/Mods/Olympus2207/Original/YSPHUNT.ssl b/Mods/Olympus2207/ReDefine/YSPHUNT.ssl
index 5926e6c..99f5f51 100644
--- a/Mods/Olympus2207/Original/YSPHUNT.ssl
+++ b/Mods/Olympus2207/ReDefine/YSPHUNT.ssl
@@ -18,29 +18,29 @@ end
 
 procedure map_enter_p_proc
 begin
-    if (metarule(14, 0)) then begin
-        if (get_pc_stat(1) <= 4) then begin
+    if (map_first_run) then begin
+        if (get_pc_stat(PCSTAT_level) <= 4) then begin
             set_map_var(0, 2);
-            create_object_sid(16777420, 20627, 0, -1);
-            create_object_sid(16777420, 21042, 0, -1);
+            create_object(16777420, 20627, 0);
+            create_object(16777420, 21042, 0);
         end
-        if ((get_pc_stat(1) > 4) and (get_pc_stat(1) <= 7)) then begin
+        if ((get_pc_stat(PCSTAT_level) > 4) and (get_pc_stat(PCSTAT_level) <= 7)) then begin
             set_map_var(0, 3);
-            create_object_sid(16777420, 20627, 0, -1);
-            create_object_sid(16777420, 21042, 0, -1);
-            create_object_sid(16777420, 15102, 0, -1);
+            create_object(16777420, 20627, 0);
+            create_object(16777420, 21042, 0);
+            create_object(16777420, 15102, 0);
         end
-        if (get_pc_stat(1) > 7) then begin
+        if (get_pc_stat(PCSTAT_level) > 7) then begin
             set_map_var(0, 4);
-            create_object_sid(16777420, 20627, 0, -1);
-            create_object_sid(16777420, 21042, 0, -1);
-            create_object_sid(16777420, 15102, 0, -1);
-            create_object_sid(16777420, 10543, 0, -1);
+            create_object(16777420, 20627, 0);
+            create_object(16777420, 21042, 0);
+            create_object(16777420, 15102, 0);
+            create_object(16777420, 10543, 0);
         end
     end
     fadeIn(500);
-    if (metarule(14, 0)) then begin
-        display_msg(message_str(10, 300 + cur_map_index));
+    if (map_first_run) then begin
+        display_msg(message_str(SCRIPT_STTEXT, 300 + cur_map_index));
     end
     if ((get_month >= 3) and (get_month < 5)) then begin
         if ((game_time_hour >= 600) and (game_time_hour < (600 + 100))) then begin
@@ -200,4 +200,3 @@ begin
         end
     end
 end
-
