diff --git a/Mods/Olympus2207/Original/YWBBUNKR.ssl b/Mods/Olympus2207/ReDefine/YWBBUNKR.ssl
index 4fd72af..3683b59 100644
--- a/Mods/Olympus2207/Original/YWBBUNKR.ssl
+++ b/Mods/Olympus2207/ReDefine/YWBBUNKR.ssl
@@ -18,8 +18,8 @@ end
 
 procedure map_enter_p_proc
 begin
-    if (metarule(14, 0)) then begin
-        display_msg(message_str(10, 300 + cur_map_index));
+    if (map_first_run) then begin
+        display_msg(message_str(SCRIPT_STTEXT, 300 + cur_map_index));
     end
     set_light_level(100);
 end
@@ -28,4 +28,3 @@ procedure map_update_p_proc
 begin
     set_light_level(100);
 end
-
