diff --git a/Mods/Olympus2207/Original/ZFIRZOLA.ssl b/Mods/Olympus2207/ReDefine/ZFIRZOLA.ssl
index 7deb7c2..585662e 100644
--- a/Mods/Olympus2207/Original/ZFIRZOLA.ssl
+++ b/Mods/Olympus2207/ReDefine/ZFIRZOLA.ssl
@@ -24,7 +24,7 @@ begin
     reg_anim_func(3, 0);
     if (days_since_visited > random(5, 7)) then begin
         if (tile_contains_obj_pid(tile_num_in_direction(tile_num(self_obj), 4, 1), elevation(self_obj), 440) == 0) then begin
-            create_object_sid(440, tile_num_in_direction(tile_num(self_obj), 4, 1), elevation(self_obj), -1);
+            create_object(440, tile_num_in_direction(tile_num(self_obj), 4, 1), elevation(self_obj));
         end
     end
 end
@@ -38,4 +38,3 @@ begin
         reg_anim_func(3, 0);
     end
 end
-
