diff --git a/Mods/Olympus2207/Original/ZHMJCKL.ssl b/Mods/Olympus2207/ReDefine/ZHMJCKL.ssl
index 4962c27..d089c20 100644
--- a/Mods/Olympus2207/Original/ZHMJCKL.ssl
+++ b/Mods/Olympus2207/ReDefine/ZHMJCKL.ssl
@@ -30,28 +30,28 @@ begin
     if (local_var(1) == 0) then begin
         set_local_var(3, tile_num(self_obj));
         set_local_var(1, 1);
-        if ((cur_map_index >= 18) and (cur_map_index <= 26) or (cur_map_index == 101)) then begin
-            critter_add_trait(self_obj, 1, 6, 7);
+        if ((cur_map_index >= MAP_SJENTER) and (cur_map_index <= MAP_SJCAVE) or (CUR_MAP_SJMAIN2)) then begin
+            critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 7);
         end
         else begin
-            if ((cur_map_index >= 27) and (cur_map_index <= 31)) then begin
-                critter_add_trait(self_obj, 1, 6, 8);
+            if ((cur_map_index >= MAP_NWENTER) and (cur_map_index <= MAP_NWYARD)) then begin
+                critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 8);
             end
             else begin
-                if ((cur_map_index >= 38) and (cur_map_index <= 43)) then begin
-                    critter_add_trait(self_obj, 1, 6, 9);
+                if ((cur_map_index >= MAP_RBMAIN) and (cur_map_index <= MAP_RBFERM)) then begin
+                    critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 9);
                 end
                 else begin
-                    if ((cur_map_index >= 44) and (cur_map_index <= 46)) then begin
-                        critter_add_trait(self_obj, 1, 6, 10);
+                    if ((cur_map_index >= MAP_OAENTER) and (cur_map_index <= MAP_OACAVE)) then begin
+                        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 10);
                     end
                     else begin
-                        critter_add_trait(self_obj, 1, 6, 202);
+                        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 202);
                     end
                 end
             end
         end
-        critter_add_trait(self_obj, 1, 5, 68);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_GENERIC_JACKAL);
         max_distance := random(24, 28);
     end
 end
@@ -91,16 +91,16 @@ end
 procedure push_p_proc
 begin
     script_overrides;
-    if ((random(0, 100) <= 60) and (tile_distance(tile_num(self_obj), tile_num(dude_obj)) <= 2)) then begin
+    if ((random(0, 100) <= 60) and (tile_distance(tile_num(self_obj), dude_tile) <= 2)) then begin
         if (not(anim_busy(self_obj))) then begin
-            anim(self_obj, 1000, rotation_to_tile(tile_num(self_obj), tile_num(dude_obj)));
+            anim(self_obj, 1000, rotation_to_tile(tile_num(self_obj), dude_tile));
         end
         reg_anim_func(2, self_obj);
         reg_anim_func(1, 1);
         reg_anim_animate(self_obj, 16, -1);
         play_sfx("MADDOGAQ");
         reg_anim_func(3, 0);
-        critter_dmg(dude_obj, random(1, 10), 0);
+        critter_damage(dude_obj, random(1, 10));
     end
     else begin
         play_sfx("MADDOGGR");
@@ -153,4 +153,3 @@ begin
         sound_is_called := 0;
     end
 end
-
