diff --git a/Mods/Olympus2207/Original/ZMINDLES.ssl b/Mods/Olympus2207/ReDefine/ZMINDLES.ssl
index 022a45a..e90f62a 100644
--- a/Mods/Olympus2207/Original/ZMINDLES.ssl
+++ b/Mods/Olympus2207/ReDefine/ZMINDLES.ssl
@@ -21,8 +21,8 @@ begin
     if (local_var(1) == 0) then begin
         set_local_var(3, tile_num(self_obj));
         set_local_var(1, 1);
-        critter_add_trait(self_obj, 1, 6, 200);
-        critter_add_trait(self_obj, 1, 5, 16);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_TEAM_NUM, 200);
+        critter_add_trait(self_obj, TRAIT_OBJECT, OBJECT_AI_PACKET, AI_GENERIC_MINDLESS);
         set_local_var(2, 1);
         critter_heal(self_obj, -(30 - (difficulty_level * 15)));
         max_distance := random(4, 8);
@@ -35,7 +35,7 @@ begin
         animation_is_called := 1;
         add_timer_event(self_obj, random(29, 76), 1);
     end
-    if (obj_can_see_obj(self_obj, dude_obj) and (tile_distance(tile_num(self_obj), tile_num(dude_obj)) <= 11)) then begin
+    if (obj_can_see_obj(self_obj, dude_obj) and (tile_distance(tile_num(self_obj), dude_tile) <= 11)) then begin
         if (local_var(2) == 1) then begin
             set_local_var(2, 0);
             attack_setup(self_obj, dude_obj);
@@ -72,7 +72,7 @@ end
 
 procedure destroy_p_proc
 begin
-    if (cur_map_index == 9) then begin
+    if (CUR_MAP_TARCHIST) then begin
         set_map_var(0, map_var(0) - 1);
     end
 end
@@ -108,4 +108,3 @@ begin
         animation_is_called := 0;
     end
 end
-
